<!DOCTYPE html><html lang="en"><head><title>Help for package ICBioMark</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ICBioMark}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ensembl_gene_lengths'><p>Gene Lengths from the Ensembl Database</p></a></li>
<li><a href='#example_first_pred_tmb'><p>First-Fit Predictive Model Fitting on Example Data</p></a></li>
<li><a href='#example_gen_model'><p>Generative Model from Simulated Data</p></a></li>
<li><a href='#example_maf_data'><p>Simulated MAF Data</p></a></li>
<li><a href='#example_predictions'><p>Example Predictions</p></a></li>
<li><a href='#example_refit_panel'><p>Refitted Predictive Model Fitted on Example Data</p></a></li>
<li><a href='#example_refit_range'><p>Refitted Predictive Models Fitted on Example Data</p></a></li>
<li><a href='#example_tables'><p>Mutation Matrices from Simulated Data</p></a></li>
<li><a href='#example_tib_tables'><p>Tumour Indel Burden of Example Train, Validation and Test Data.</p></a></li>
<li><a href='#example_tmb_tables'><p>Tumour Mutation Burden of Example Train, Validation and Test Data.</p></a></li>
<li><a href='#fit_gen_model'><p>Fit Generative Model</p></a></li>
<li><a href='#fit_gen_model_uninteract'><p>Fit Generative Model Without Gene/Variant Type-Specific Interactions</p></a></li>
<li><a href='#fit_gen_model_unisamp'><p>Fit Generative Model Without Sample-Specific Effects</p></a></li>
<li><a href='#generate_maf_data'><p>Generate mutation data.</p></a></li>
<li><a href='#get_auprc'><p>AUPRC Metrics for Predictions</p></a></li>
<li><a href='#get_biomarker_from_maf'><p>Produce a Table of Biomarker Values from a MAF</p></a></li>
<li><a href='#get_biomarker_tables'><p>Get True Biomarker Values on Training, Validation and Test Sets</p></a></li>
<li><a href='#get_gen_estimates'><p>Investigate Generative Model Comparisons</p></a></li>
<li><a href='#get_K'><p>Construct Bias Penalisation</p></a></li>
<li><a href='#get_mutation_dictionary'><p>Group and Filter Mutation Types</p></a></li>
<li><a href='#get_mutation_tables'><p>Produce Training, Validation and Test Matrices</p></a></li>
<li><a href='#get_p'><p>Construct Optimisation Parameters.</p></a></li>
<li><a href='#get_panels_from_fit'><p>Extract Panel Details from Group Lasso Fit</p></a></li>
<li><a href='#get_predictions'><p>Produce Predictions on an Unseen Dataset</p></a></li>
<li><a href='#get_r_squared'><p>R Squared Metrics for Predictions</p></a></li>
<li><a href='#get_stats'><p>Metrics for Predictive Performance</p></a></li>
<li><a href='#get_table_from_maf'><p>Produce a Mutation Matrix from a MAF</p></a></li>
<li><a href='#ICBioMark'><p>ICBioMark: A package for cost-effective design of gene panels to predict</p>
exome-wide biomarkers.</a></li>
<li><a href='#nsclc_maf'><p>Non-Small Cell Lung Cancer MAF Data</p></a></li>
<li><a href='#nsclc_survival'><p>Non-Small Cell Lung Cancer Survival and Clinical Data</p></a></li>
<li><a href='#pred_first_fit'><p>First-Fit Predicitve Model with Group Lasso</p></a></li>
<li><a href='#pred_intervals'><p>Produce Error Bounds for Predictions</p></a></li>
<li><a href='#pred_refit_panel'><p>Refitted Predictive Model for a Given Panel</p></a></li>
<li><a href='#pred_refit_range'><p>Get Refitted Predictive Models for a First-Fit Range of Panels</p></a></li>
<li><a href='#vis_model_fit'><p>Visualise Generative Model Fit</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Data-Driven Design of Targeted Gene Panels for Estimating
Immunotherapy Biomarkers</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the methodology proposed in 'Data-driven design of targeted gene panels for estimating immunotherapy biomarkers', Bradley and Cannings (2021) &lt;<a href="https://doi.org/10.48550/arXiv.2102.04296">doi:10.48550/arXiv.2102.04296</a>&gt;. This package allows the user to fit generative models of mutation from an annotated mutation dataset, and then further to produce tunable linear estimators of exome-wide biomarkers. It also contains functions to simulate mutation annotated format (MAF) data, as well as to analyse the output and performance of models.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 2.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, glmnet, Matrix, dplyr, purrr, latex2exp,
matrixStats, ggplot2, gglasso, PRROC</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-15 10:43:06 UTC; s1505825</td>
</tr>
<tr>
<td>Author:</td>
<td>Jacob R. Bradley <a href="https://orcid.org/0000-0003-1616-4969"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Timothy I. Cannings
    <a href="https://orcid.org/0000-0002-2111-4168"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jacob R. Bradley &lt;cobrbradley@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-15 12:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ensembl_gene_lengths'>Gene Lengths from the Ensembl Database</h2><span id='topic+ensembl_gene_lengths'></span>

<h3>Description</h3>

<p>Pre-imported length data from the Ensembl database for all genes on chromosomes 1-22, X and Y.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensembl_gene_lengths
</code></pre>


<h3>Format</h3>

<p>A dataframe with three columns:
</p>

<dl>
<dt>Hugo_Symbol</dt><dd><p>The names of all nuclear genes in humans for which ensembl entries with coding sequence lengths exist.</p>
</dd>
<dt>max_cds</dt><dd><p>The maximum coding sequence for each gene as given by the ensembl database.</p>
</dd>
<dt>Chromosome</dt><dd><p>The chromosome where each gene is located.</p>
</dd>
</dl>



<h3>Source</h3>

<p>See the folder data-raw.
</p>

<hr>
<h2 id='example_first_pred_tmb'>First-Fit Predictive Model Fitting on Example Data</h2><span id='topic+example_first_pred_tmb'></span>

<h3>Description</h3>

<p>An example output from the function pred_first_fit(), applied to pre-loaded example mutation data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_first_pred_tmb
</code></pre>


<h3>Format</h3>

<p>A list with six entries:
</p>

<dl>
<dt>fit</dt><dd><p>A gglasso fit.</p>
</dd>
<dt>panel_genes</dt><dd><p>A matrix where each row corresponds to a gene, each column to an iteration of the group
lasso with a different penalty factor, and the elements booleans specifying whether that gene was selected
to be included in that iteration.</p>
</dd>
<dt>panel_lengths</dt><dd><p>A vector giving total panel length for each gglasso iteration.</p>
</dd>
<dt>p</dt><dd><p>The vector of weights used in the optimisation procedure.</p>
</dd>
<dt>K</dt><dd><p>The bias penalty factor used in the optimisation procedure.</p>
</dd>
<dt>names</dt><dd><p>Gene and mutation type information as used when fitting the generative model.</p>
</dd>
</dl>


<hr>
<h2 id='example_gen_model'>Generative Model from Simulated Data</h2><span id='topic+example_gen_model'></span>

<h3>Description</h3>

<p>An example of the output produced by fit_gen_model() on simulated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_gen_model
</code></pre>


<h3>Format</h3>

<p>A list with two entries:
</p>

<dl>
<dt>fit</dt><dd><p>A glmnet fit object.</p>
</dd>
<dt>dev</dt><dd><p>A table containing the average deviance of each cross-validation fold, for each penalisation factor in fit$lambda.</p>
</dd>
<dt>s_min</dt><dd><p>The index of the regularisation penalty minimising average deviance across folds.</p>
</dd>
</dl>


<hr>
<h2 id='example_maf_data'>Simulated MAF Data</h2><span id='topic+example_maf_data'></span>

<h3>Description</h3>

<p>An example dataset generated by the function generate_maf_data(), with n_sample = 100 and n_genes = 20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_maf_data
</code></pre>


<h3>Format</h3>

<p>A list with two entries:
</p>

<dl>
<dt>maf</dt><dd><p>An annotated mutation dataframe with 3 columns and 1346 rows: </p>

<dl>
<dt>Tumor_Sample_Barcode</dt><dd><p>A sample id for each mutation.</p>
</dd>
<dt>Hugo_Symbol</dt><dd><p>The name of the gene location for each mutation.</p>
</dd>
<dt>Variant_Classification</dt><dd><p>The mutation type for each mutation.</p>
</dd></dl>
</dd>
<dt>gene_lengths</dt><dd><p>A data frame with two rows: </p>

<dl>
<dt>Hugo_Symbol</dt><dd><p>The name of each gene.</p>
</dd>
<dt>max_cds</dt><dd><p>The length of each gene, as defined by maximum coding sequence.</p>
</dd></dl>
</dd>
</dl>


<hr>
<h2 id='example_predictions'>Example Predictions</h2><span id='topic+example_predictions'></span>

<h3>Description</h3>

<p>An example output from use of the function get_predictions(), applied to the pre-loaded datasets example_refit_range and example_tables$val .
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_predictions
</code></pre>


<h3>Format</h3>

<p>A list with two entries:
</p>

<dl>
<dt>predictions</dt><dd><p>A a matrix containing a row for each sample and a column for each panel.</p>
</dd>
<dt>panel_lengths</dt><dd><p>A vector giving total panel lengths.</p>
</dd>
</dl>


<hr>
<h2 id='example_refit_panel'>Refitted Predictive Model Fitted on Example Data</h2><span id='topic+example_refit_panel'></span>

<h3>Description</h3>

<p>An example output from use of the function pred_refit_panel(), applied to example gene length data and generative model fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_refit_panel
</code></pre>


<h3>Format</h3>

<p>A list with three entries:
</p>

<dl>
<dt>fit</dt><dd><p>A list with a single element 'beta', a matrix with prediction weights.</p>
</dd>
<dt>panel_genes</dt><dd><p>A matrix (in this case with a single column) where each row corresponds to a gene,
and each entry corresponds to whether the gene is included in the panel.</p>
</dd>
<dt>panel_lengths</dt><dd><p>A vector of length 1 giving total panel length.</p>
</dd>
</dl>


<hr>
<h2 id='example_refit_range'>Refitted Predictive Models Fitted on Example Data</h2><span id='topic+example_refit_range'></span>

<h3>Description</h3>

<p>An example output from use of the function pred_refit_range(), applied to example gene length data and generative model fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_refit_range
</code></pre>


<h3>Format</h3>

<p>A list with six entries:
</p>

<dl>
<dt>fit</dt><dd><p>A list with a single element 'beta', a matrix with prediction weights.</p>
</dd>
<dt>panel_genes</dt><dd><p>A matrix where each row corresponds to a gene,
and each entry corresponds to whether the gene is included in the panel.</p>
</dd>
<dt>panel_lengths</dt><dd><p>A vector giving total panel length.</p>
</dd>
</dl>


<hr>
<h2 id='example_tables'>Mutation Matrices from Simulated Data</h2><span id='topic+example_tables'></span>

<h3>Description</h3>

<p>Mutation data extracted from the pre-loaded example mutation data example_maf_data, using the function get_mutation_tables().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_tables
</code></pre>


<h3>Format</h3>

<p>A list with three entries:
</p>

<dl>
<dt>train</dt><dd><p>An object 'train'.</p>
</dd>
<dt>val</dt><dd><p>An object 'val'.</p>
</dd>
<dt>test</dt><dd><p>An object 'test'.</p>
</dd>
</dl>

<p>Each of these three objects is a list with the following entries (for more detail see the documentation for the function get_table_from_maf()):
</p>

<dl>
<dt>matrix</dt><dd><p>A sparse matrix of mutations.</p>
</dd>
<dt>sample_list</dt><dd><p>A character vector of sample IDs, corresponding to the rows of the mutation matrix.</p>
</dd>
<dt>gene_list</dt><dd><p>A character vector of gene names.</p>
</dd>
<dt>mut_types_list</dt><dd><p>A character vector of mutation types.</p>
</dd>
<dt>colnames</dt><dd><p>A character vector of gene name/mutation type combinations (in each case separated by the character &quot;_&quot;), corresponding to the columns of the mutation matrix.</p>
</dd>
</dl>


<hr>
<h2 id='example_tib_tables'>Tumour Indel Burden of Example Train, Validation and Test Data.</h2><span id='topic+example_tib_tables'></span>

<h3>Description</h3>

<p>An example output produced by using the function get_biomarker_tables(), applied to the example MAF data pre-loaded in example_maf_data$maf.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_tib_tables
</code></pre>


<h3>Format</h3>

<p>A list with threeobjects: 'train', 'val' and 'test'. Each is a dataframe with two columns:
</p>

<dl>
<dt>Tumor_Sample_Barcode</dt><dd><p>A unique ID for each sample.</p>
</dd>
<dt>TIB</dt><dd><p>The value of Tumour Indel Burden for that sample.</p>
</dd>
</dl>


<hr>
<h2 id='example_tmb_tables'>Tumour Mutation Burden of Example Train, Validation and Test Data.</h2><span id='topic+example_tmb_tables'></span>

<h3>Description</h3>

<p>An example output produced by using the function get_biomarker_tables(), applied to the example MAF data pre-loaded in example_maf_data$maf.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_tmb_tables
</code></pre>


<h3>Format</h3>

<p>A list with threeobjects: 'train', 'val' and 'test'. Each is a dataframe with two columns:
</p>

<dl>
<dt>Tumor_Sample_Barcode</dt><dd><p>A unique ID for each sample.</p>
</dd>
<dt>TMB</dt><dd><p>The value of Tumour Mutation Burden for that sample.</p>
</dd>
</dl>


<hr>
<h2 id='fit_gen_model'>Fit Generative Model</h2><span id='topic+fit_gen_model'></span>

<h3>Description</h3>

<p>A function to fit a generative model to a mutation dataset. At its heart, requires a gene_lengths dataframe (for examples of the correct format for this see the pre-loaded datasets example_maf_data$gene_lengths and ensembl_gene_lengths), and a mutation dataset. This is best supplied through the 'table' argument, and constructed via the function get_mutation_tables().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_gen_model(
  gene_lengths,
  matrix = NULL,
  sample_list = NULL,
  gene_list = NULL,
  mut_types_list = NULL,
  col_names = NULL,
  table = NULL,
  nlambda = 100,
  n_folds = 10,
  maxit = 1e+09,
  seed_id = 1234,
  progress = FALSE,
  alt_model_type = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_gen_model_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A table with two columns: Hugo_Symbol and max_cds, providing the lengths of the genes to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_matrix">matrix</code></td>
<td>
<p>(Matrix::sparseMatrix)
A mutation matrix, such as produced by the function get_table_from_maf().</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_sample_list">sample_list</code></td>
<td>
<p>(character)
The set of samples to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
The set of genes to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_mut_types_list">mut_types_list</code></td>
<td>
<p>(character)
The set of mutation types to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_col_names">col_names</code></td>
<td>
<p>(character)
The column names of the 'matrix' parameter.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_table">table</code></td>
<td>
<p>(list)
Optional parameter combining matrix, sample_list, gene_list, mut_types_list, col_names, as is produced by the function get_tables().</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_nlambda">nlambda</code></td>
<td>
<p>(numeric)
The length of the vector of penalty weights, passed to the function glmnet::glmnet().</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_n_folds">n_folds</code></td>
<td>
<p>(numeric)
The number of cross-validation folds to employ.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_maxit">maxit</code></td>
<td>
<p>(numeric)
Technical parameter passed to the function glmnet::glmnet().</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_seed_id">seed_id</code></td>
<td>
<p>(numeric)
Input value for the function set.seed().</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_progress">progress</code></td>
<td>
<p>(logical)
Show progress bars and text.</p>
</td></tr>
<tr><td><code id="fit_gen_model_+3A_alt_model_type">alt_model_type</code></td>
<td>
<p>(character)
Used to call an alternative generative model type such as &quot;US&quot; (no sample-dependent parameters) or
&quot;UI&quot; (no gene/variant-type interactions).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list comprising three objects:
</p>

<ul>
<li><p> An object 'fit', a fitted glmnet model.
</p>
</li>
<li><p> A table 'dev', giving average deviances for each regularisation penalty factor and cross-validation fold.
</p>
</li>
<li><p> An integer 's_min', the index of the regularsisation penalty minimising cross-validation deviance.
</p>
</li>
<li><p> A list 'names', containing the sample, gene, and mutation type information of the training data.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_gen_model &lt;- fit_gen_model(example_maf_data$gene_lengths, table = example_tables$train)
print(names(example_gen_model))
</code></pre>

<hr>
<h2 id='fit_gen_model_uninteract'>Fit Generative Model Without Gene/Variant Type-Specific Interactions</h2><span id='topic+fit_gen_model_uninteract'></span>

<h3>Description</h3>

<p>A function to fit a generative model to a mutation dataset that does not incorporate gene/variant-specific effects.
Otherwise acts similarly to the function fit_gen_model().
</p>
<p>NOTE: fits produced by this model will not be compatible with predictive model fits downstream - it is purely for comparing
with full models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_gen_model_uninteract(
  gene_lengths,
  matrix = NULL,
  sample_list = NULL,
  gene_list = NULL,
  mut_types_list = NULL,
  col_names = NULL,
  table = NULL,
  nlambda = 100,
  n_folds = 10,
  maxit = 1e+09,
  seed_id = 1234,
  progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_gen_model_uninteract_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A table with two columns: Hugo_Symbol and max_cds, providing the lengths of the genes to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_matrix">matrix</code></td>
<td>
<p>(Matrix::sparseMatrix)
A mutation matrix, such as produced by the function get_table_from_maf().</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_sample_list">sample_list</code></td>
<td>
<p>(character)
The set of samples to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
The set of genes to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_mut_types_list">mut_types_list</code></td>
<td>
<p>(character)
The set of mutation types to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_col_names">col_names</code></td>
<td>
<p>(character)
The column names of the 'matrix' parameter.</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_table">table</code></td>
<td>
<p>(list)
Optional parameter combining matrix, sample_list, gene_list, mut_types_list, col_names, as is produced by the function get_tables().</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_nlambda">nlambda</code></td>
<td>
<p>(numeric)
The length of the vector of penalty weights, passed to the function glmnet::glmnet().</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_n_folds">n_folds</code></td>
<td>
<p>(numeric)
The number of cross-validation folds to employ.</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_maxit">maxit</code></td>
<td>
<p>(numeric)
Technical parameter passed to the function glmnet::glmnet().</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_seed_id">seed_id</code></td>
<td>
<p>(numeric)
Input value for the function set.seed().</p>
</td></tr>
<tr><td><code id="fit_gen_model_uninteract_+3A_progress">progress</code></td>
<td>
<p>(logical)
Show progress bars and text.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list comprising three objects:
</p>

<ul>
<li><p> An object 'fit', a fitted glmnet model.
</p>
</li>
<li><p> A table 'dev', giving average deviances for each regularisation penalty factor and cross-validation fold.
</p>
</li>
<li><p> An integer 's_min', the index of the regularsisation penalty minimising cross-validation deviance.
</p>
</li>
<li><p> A list 'names', containing the sample, gene, and mutation type information of the training data.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_gen_model_unisamp &lt;- fit_gen_model_unisamp(example_maf_data$gene_lengths,
                                                   table = example_tables$train)
print(names(example_gen_model))
</code></pre>

<hr>
<h2 id='fit_gen_model_unisamp'>Fit Generative Model Without Sample-Specific Effects</h2><span id='topic+fit_gen_model_unisamp'></span>

<h3>Description</h3>

<p>A function to fit a generative model to a mutation dataset that does not incorporate sample-specific effects.
Otherwise acts similarly to the function fit_gen_model().
</p>
<p>NOTE: fits produced by this model will not be compatible with predictive model fits downstream - it is purely for comparing
with full models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_gen_model_unisamp(
  gene_lengths,
  matrix = NULL,
  sample_list = NULL,
  gene_list = NULL,
  mut_types_list = NULL,
  col_names = NULL,
  table = NULL,
  nlambda = 100,
  n_folds = 10,
  maxit = 1e+09,
  seed_id = 1234,
  progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_gen_model_unisamp_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A table with two columns: Hugo_Symbol and max_cds, providing the lengths of the genes to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_matrix">matrix</code></td>
<td>
<p>(Matrix::sparseMatrix)
A mutation matrix, such as produced by the function get_table_from_maf().</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_sample_list">sample_list</code></td>
<td>
<p>(character)
The set of samples to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
The set of genes to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_mut_types_list">mut_types_list</code></td>
<td>
<p>(character)
The set of mutation types to be modelled.</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_col_names">col_names</code></td>
<td>
<p>(character)
The column names of the 'matrix' parameter.</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_table">table</code></td>
<td>
<p>(list)
Optional parameter combining matrix, sample_list, gene_list, mut_types_list, col_names, as is produced by the function get_tables().</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_nlambda">nlambda</code></td>
<td>
<p>(numeric)
The length of the vector of penalty weights, passed to the function glmnet::glmnet().</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_n_folds">n_folds</code></td>
<td>
<p>(numeric)
The number of cross-validation folds to employ.</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_maxit">maxit</code></td>
<td>
<p>(numeric)
Technical parameter passed to the function glmnet::glmnet().</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_seed_id">seed_id</code></td>
<td>
<p>(numeric)
Input value for the function set.seed().</p>
</td></tr>
<tr><td><code id="fit_gen_model_unisamp_+3A_progress">progress</code></td>
<td>
<p>(logical)
Show progress bars and text.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list comprising three objects:
</p>

<ul>
<li><p> An object 'fit', a fitted glmnet model.
</p>
</li>
<li><p> A table 'dev', giving average deviances for each regularisation penalty factor and cross-validation fold.
</p>
</li>
<li><p> An integer 's_min', the index of the regularsisation penalty minimising cross-validation deviance.
</p>
</li>
<li><p> A list 'names', containing the sample, gene, and mutation type information of the training data.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_gen_model_unisamp &lt;- fit_gen_model_unisamp(example_maf_data$gene_lengths,
                                                   table = example_tables$train)
print(names(example_gen_model))
</code></pre>

<hr>
<h2 id='generate_maf_data'>Generate mutation data.</h2><span id='topic+generate_maf_data'></span>

<h3>Description</h3>

<p>A function to randomly simulate an (abridged) annotated mutation file, containing information on sample of origin, gene and mutation type, as well as a dataframe of gene lengths.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_maf_data(
  n_samples = 100,
  n_genes = 20,
  mut_types = NULL,
  data_dist = NULL,
  sample_rates = NULL,
  gene_rates = NULL,
  gene_lengths = NULL,
  sample_rates_dist = NULL,
  gene_rates_dist = NULL,
  gene_lengths_dist = NULL,
  bmr_genes_prop = 0.7,
  output_rates = FALSE,
  seed_id = 1234
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate_maf_data_+3A_n_samples">n_samples</code></td>
<td>
<p>(numeric)
The number of samples to generate mutation data for - each will have a unique value in the 'Tumor_Sample_Barcode'
column of the simulated MAF table. Note that if no mutations are simulated for an example, they will not appear in the table.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_n_genes">n_genes</code></td>
<td>
<p>(numeric)
The number of genes to generate mutation data for - each will have a unique value in the 'Hugo_Symbol' column
of the simulated MAF table. A length will also be generated for each gene, and stored in the table 'gene_lengths'.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_mut_types">mut_types</code></td>
<td>
<p>(numeric)
A vector of positive values giving the relative average abundance of each mutation type. The names of each mutation
type are stored in the names attribute of the vector, and will form the entries of the column 'Variant_Classification'
in the output MAF table.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_data_dist">data_dist</code></td>
<td>
<p>(function)
Directly provide the probability distribution of mutations, as a function on n_samples, n_genes, mut_types, and gene_lengths.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_sample_rates">sample_rates</code></td>
<td>
<p>(numeric)
Directly provide sample-specific rates.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_gene_rates">gene_rates</code></td>
<td>
<p>(numeric)
Directly provide gene-specific rates.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(numeric)
Directly provide gene lengths, in the form of a vector of numerics with names attribute corresponding to gene names.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_sample_rates_dist">sample_rates_dist</code></td>
<td>
<p>(function)
Directly provide the distribution of sample-specific rates, as a function of the number of samples.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_gene_rates_dist">gene_rates_dist</code></td>
<td>
<p>(function)
Directly provide the distribution of gene-specific rates, as a function of the number of genes.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_gene_lengths_dist">gene_lengths_dist</code></td>
<td>
<p>(function)
Directly provide the distribution of gene lengths, as a function of the number of genes.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_bmr_genes_prop">bmr_genes_prop</code></td>
<td>
<p>(numeric)
The proportion of genes that follow the background mutation rate. If specified (as is automatic), this proportion of genes
will have gene-specific rates equal to 1. By setting to be NULL, can avoid applying this step.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_output_rates">output_rates</code></td>
<td>
<p>(logical)
If TRUE, will include the sample and gene rates in the output.</p>
</td></tr>
<tr><td><code id="generate_maf_data_+3A_seed_id">seed_id</code></td>
<td>
<p>(numeric)
Input value for the function set.seed().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements, 'maf' and 'gene_lengths'. These are (respectively):
</p>

<ul>
<li><p> A table with three columns: 'Tumor_Sample_Barcode', 'Hugo_Symbol' and 'Variant_Classification', listing the mutations occurring in
the simulated example.
gene_lengths (dataframe)
</p>
</li>
<li><p> A table with two rows: 'Hugo_Symbol' and 'gene_lengths'.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Generate some random data
data &lt;- generate_maf_data(n_samples = 10, n_genes = 20)
# See the first rows of the maf table.
print(head(data$maf))
# See the first rows of the gene_lengths table.
print(head(data$gene_lengths))
</code></pre>

<hr>
<h2 id='get_auprc'>AUPRC Metrics for Predictions</h2><span id='topic+get_auprc'></span>

<h3>Description</h3>

<p>A function to return AUPRC metrics for predictions vs actual values. Works well when piped to straight from get_predictions().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_auprc(predictions, biomarker_values, model = "", threshold = 300)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_auprc_+3A_predictions">predictions</code></td>
<td>
<p>(list)
A list with two elements, 'predictions' and 'panel_lengths',
as produced by the function get_predictions().</p>
</td></tr>
<tr><td><code id="get_auprc_+3A_biomarker_values">biomarker_values</code></td>
<td>
<p>(dataframe)
A dataframe with two columns, 'Tumor_Sample_Barcode' and a
column with the name of the biomarker in question containing values.</p>
</td></tr>
<tr><td><code id="get_auprc_+3A_model">model</code></td>
<td>
<p>(character)
The name of the model type producing these predictions.</p>
</td></tr>
<tr><td><code id="get_auprc_+3A_threshold">threshold</code></td>
<td>
<p>(numeric)
The threshold for biomarker high/low categorisation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with 5 columns:
</p>

<ul>
<li><p> panel_length: the length of each panel.
</p>
</li>
<li><p> model: the model that produced the predictions.
</p>
</li>
<li><p> biomarker: the name of the biomarker in question.
</p>
</li>
<li><p> stat: the AUPRC values for each panel.
</p>
</li>
<li><p> metric: a constant character &quot;AUPRC&quot;.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_auprc &lt;- get_auprc(predictions = get_predictions(example_refit_panel,
new_data = example_tables$val), biomarker_values = example_tmb_tables$val,
model = "Refitted T", threshold = 10)
</code></pre>

<hr>
<h2 id='get_biomarker_from_maf'>Produce a Table of Biomarker Values from a MAF</h2><span id='topic+get_biomarker_from_maf'></span>

<h3>Description</h3>

<p>A function to recover true biomarker values from a mutation annotation file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biomarker_from_maf(
  maf,
  biomarker = "TIB",
  sample_list = NULL,
  gene_list = NULL,
  biomarker_name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_biomarker_from_maf_+3A_maf">maf</code></td>
<td>
<p>(dataframe)
A table of annotated mutations containing the columns 'Tumor_Sample_Barcode', 'Hugo_Symbol', and 'Variant_Classification'.</p>
</td></tr>
<tr><td><code id="get_biomarker_from_maf_+3A_biomarker">biomarker</code></td>
<td>
<p>(character)
Which biomarker needs calculating? If &quot;TMB&quot; or &quot;TIB&quot;, then appropriate mutation types will be selected. Otherwise, will be interpreted as a vector of characters
denoting mutation types to include.</p>
</td></tr>
<tr><td><code id="get_biomarker_from_maf_+3A_sample_list">sample_list</code></td>
<td>
<p>(character)
Vector of characters giving a list of values of Tumor_Sample_Barcode to include.</p>
</td></tr>
<tr><td><code id="get_biomarker_from_maf_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
Vector of characters giving a list of genes to include in calculation of biomarker.</p>
</td></tr>
<tr><td><code id="get_biomarker_from_maf_+3A_biomarker_name">biomarker_name</code></td>
<td>
<p>(character)
Name of biomarker. Only needed if biomarker is not &quot;TMB&quot; or &quot;TIB&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with two columns, 'Tumor_Sample_Barcode' and values of the biomarker specified.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(head(get_biomarker_from_maf(example_maf_data$maf, sample_list = paste0("SAMPLE_", 1:100))))
</code></pre>

<hr>
<h2 id='get_biomarker_tables'>Get True Biomarker Values on Training, Validation and Test Sets</h2><span id='topic+get_biomarker_tables'></span>

<h3>Description</h3>

<p>A function, similar to get_mutation_tables(), but returning the true biomarker values for a training, validation and test sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biomarker_tables(
  maf,
  biomarker = "TIB",
  sample_list = NULL,
  gene_list = NULL,
  biomarker_name = NULL,
  tables = NULL,
  split = c(train = 0.7, val = 0.15, test = 0.15),
  seed_id = 1234
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_biomarker_tables_+3A_maf">maf</code></td>
<td>
<p>(dataframe)
A table of annotated mutations containing the columns 'Tumor_Sample_Barcode', 'Hugo_Symbol', and 'Variant_Classification'.</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_biomarker">biomarker</code></td>
<td>
<p>(character)
Which biomarker needs calculating? If &quot;TMB&quot; or &quot;TIB&quot;, then appropriate mutation types will be selected. Otherwise, will be interpreted as a vector of characters
denoting mutation types to include.</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_sample_list">sample_list</code></td>
<td>
<p>(character)
Vector of characters giving a list of values of Tumor_Sample_Barcode to include.</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
Vector of characters giving a list of genes to include in calculation of biomarker.</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_biomarker_name">biomarker_name</code></td>
<td>
<p>(character)
Name of biomarker. Only needed if biomarker is not &quot;TMB&quot; or &quot;TIB&quot;</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_tables">tables</code></td>
<td>
<p>(list)
Optional parameter, the output of a call to get_mutation_tables(), which already has a train/val/test split.</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_split">split</code></td>
<td>
<p>(numeric)
Optional parameter directly specifying the proportions of a train/test/val split.</p>
</td></tr>
<tr><td><code id="get_biomarker_tables_+3A_seed_id">seed_id</code></td>
<td>
<p>(numeric)
Input value for the function set.seed().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three objects: 'train', 'val' and 'test. Each comprises a dataframe with two columns, denoting sample ID and biomarker value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(head(get_biomarker_tables(example_maf_data$maf, sample_list = paste0("SAMPLE_", 1:100))))
</code></pre>

<hr>
<h2 id='get_gen_estimates'>Investigate Generative Model Comparisons</h2><span id='topic+get_gen_estimates'></span>

<h3>Description</h3>

<p>Given a generative model of the type we propose, and an alternate version (saturated &quot;S&quot;, sample-independent &quot;US&quot;, gene-independent &quot;UG&quot; or gene/variant interaction independent &quot;UI&quot;), either produces the estimated observations on the training dataset or calculates residual deviance between models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_gen_estimates(
  training_data,
  gen_model,
  alt_gen_model = NULL,
  alt_model_type = "S",
  gene_lengths = NULL,
  calculate_deviance = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_gen_estimates_+3A_training_data">training_data</code></td>
<td>
<p>(list) Likely the 'train' component of a call to get_mutation_tables().</p>
</td></tr>
<tr><td><code id="get_gen_estimates_+3A_gen_model">gen_model</code></td>
<td>
<p>(list) A generative model - result of a call to fit_gen_model*().</p>
</td></tr>
<tr><td><code id="get_gen_estimates_+3A_alt_gen_model">alt_gen_model</code></td>
<td>
<p>(list) An alternative generative model.</p>
</td></tr>
<tr><td><code id="get_gen_estimates_+3A_alt_model_type">alt_model_type</code></td>
<td>
<p>(character) One of &quot;S&quot; (saturated), &quot;US&quot; (sample-independent), &quot;UG&quot;, (gene-independent), &quot;UI&quot; (gene/variant-interaction independent).</p>
</td></tr>
<tr><td><code id="get_gen_estimates_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe) A gene lengths data frame.</p>
</td></tr>
<tr><td><code id="get_gen_estimates_+3A_calculate_deviance">calculate_deviance</code></td>
<td>
<p>(logical) If TRUE, returns residual deviance statistics. If FALSE, returns training data predictions.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If calculate_deviance = FALSE:
</p>
<p>A list with two entries, est_mut_vec and alt_est_mut_vec, each of length n_samples x n_genes x n_mut_types, giving expected mutation value for each combination of sample, gene and variant type in the training dataset under the two models being compared.
</p>
<p>If calculate_deviance = TRUE:
</p>
<p>A list with two entries, deviance and df, corresponding to the residual deviance and residual degrees of freedom between the two models on the training set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sat_dev &lt;- get_gen_estimates(training_data = example_tables$train,
                                       gen_model = example_gen_model,
                                       alt_model_type = "S",
                                       gene_lengths = example_maf_data$gene_lengths,
                                       calculate_deviance = TRUE)
</code></pre>

<hr>
<h2 id='get_K'>Construct Bias Penalisation</h2><span id='topic+get_K'></span>

<h3>Description</h3>

<p>An internal function, producing the correct bias penalisation for use in predictive model fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_K(
  gen_model,
  p_norm,
  training_matrix,
  marker_training_values = NULL,
  method = max
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_K_+3A_gen_model">gen_model</code></td>
<td>
<p>(list)
A generative mutation model, fitted by fit_gen_model().</p>
</td></tr>
<tr><td><code id="get_K_+3A_p_norm">p_norm</code></td>
<td>
<p>(numeric)
Scaling factor between coefficients of p and parameters of generative model (see paper for details).</p>
</td></tr>
<tr><td><code id="get_K_+3A_training_matrix">training_matrix</code></td>
<td>
<p>(sparse matrix)
A sparse matrix of mutations in the training dataset, produced by get_mutation_tables().</p>
</td></tr>
<tr><td><code id="get_K_+3A_marker_training_values">marker_training_values</code></td>
<td>
<p>(dataframe)
A dataframe containing training values for the biomarker in question.</p>
</td></tr>
<tr><td><code id="get_K_+3A_method">method</code></td>
<td>
<p>(function)
How to select a representative biomarker value from the training dataset. Defaults to max().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numerical value, to be used as a penalty weighting in the subsequent group lasso optimisation.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>K &lt;- get_K(example_gen_model, 1, example_tables$train$matrix)
print(K)
</code></pre>

<hr>
<h2 id='get_mutation_dictionary'>Group and Filter Mutation Types</h2><span id='topic+get_mutation_dictionary'></span>

<h3>Description</h3>

<p>A function to create a mutation dictionary to group and filter mutation types: this can be useful for computational practicality. It is often not practical to model
each distinct mutation type together, so for practicality one may group multiple classes together (e.g. all indel mutations, all nonsynonymous mutations).
Additionally, some mutation types may be excluded from modelling (for example, one may wish not to use synonymous mutations in the model fitting process).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mutation_dictionary(
  for_biomarker = "TIB",
  include_synonymous = TRUE,
  maf = NULL,
  dictionary = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mutation_dictionary_+3A_for_biomarker">for_biomarker</code></td>
<td>
<p>(string)
Specify some standard groupings of mutation types, corresponding the the coarsest groupings
of nonsynonymous mutations required to evaluate the biomarkers TMB and TIB. If &quot;TMB&quot;, groups all nonsynonymous
mutations together, if &quot;TIB&quot; groups indel mutations together and all other mutations together.</p>
</td></tr>
<tr><td><code id="get_mutation_dictionary_+3A_include_synonymous">include_synonymous</code></td>
<td>
<p>(logical)
Determine whether synonymous mutations should be included in the dictionary.</p>
</td></tr>
<tr><td><code id="get_mutation_dictionary_+3A_maf">maf</code></td>
<td>
<p>(dataframe)
An annotated mutation table containing the column 'Variant_Classification', only used to check if the dictionary specified does not contain all the variant types
in your dataset.</p>
</td></tr>
<tr><td><code id="get_mutation_dictionary_+3A_dictionary">dictionary</code></td>
<td>
<p>(character)
Directly specify the dictionary, in the form of a vector of grouping values. The names of the vector should correspond to the set
of variant classifications of interest in the mutation annotated file (MAF).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of characters, with values corresponding to the grouping labels for mutation types, and with names corresponding to the mutation types as they
will be referred to in a mutation annotated file (MAF). See examples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># To understand the dictionary format, note that the following code
dictionary &lt;- get_mutation_dictionary(for_biomarker = "TMB")
# is equivalent to
dictionary &lt;- c(rep("NS",9), rep("S", 8))
names(dictionary) &lt;- c('Missense_Mutation', 'Nonsense_Mutation',
'Splice_Site', 'Translation_Start_Site',
'Nonstop_Mutation', 'In_Frame_Ins',
'In_Frame_Del', 'Frame_Shift_Del',
'Frame_Shift_Ins', 'Silent',
'Splice_Region', '3\'Flank', '5\'Flank',
'Intron', 'RNA', '3\'UTR', '5\'UTR')
# where the grouping levels are chosen to be "NS" and "S" for
# nonsynonymous and synonymous mutations respectively.
# the code
dictionary &lt;- get_mutation_dictionary(for_biomarker = "TIB", include_synonymous = FALSE)
# is equivalent to
dictionary &lt;- dictionary &lt;- c(rep("NS",7), rep("I", 2))
names(dictionary) &lt;- c('Missense_Mutation', 'Nonsense_Mutation',
                       'Splice_Site', 'Translation_Start_Site',
                      'Nonstop_Mutation', 'In_Frame_Ins',
                      'In_Frame_Del', 'Frame_Shift_Del',
                      'Frame_Shift_Ins')
# where now "I" is used as a label to refer to indel mutations,
# and synonymous mutations are filtered out.
</code></pre>

<hr>
<h2 id='get_mutation_tables'>Produce Training, Validation and Test Matrices</h2><span id='topic+get_mutation_tables'></span>

<h3>Description</h3>

<p>This function allows for i) separation of a mutation dataset into training, validation and testing components, and ii) conversion from annotated mutation format
to sparse mutation matrices, as described in the function get_table_from_maf().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mutation_tables(
  maf,
  split = c(train = 0.7, val = 0.15, test = 0.15),
  sample_list = NULL,
  gene_list = NULL,
  acceptable_genes = NULL,
  for_biomarker = "TIB",
  include_synonymous = TRUE,
  dictionary = NULL,
  seed_id = 1234
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mutation_tables_+3A_maf">maf</code></td>
<td>
<p>(dataframe)
A table of annotated mutations containing the columns 'Tumor_Sample_Barcode', 'Hugo_Symbol', and 'Variant_Classification'.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_split">split</code></td>
<td>
<p>(double)
A vector of three positive values with names 'train', 'val' and 'test'. Specifies the proportions into which to split the dataset.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_sample_list">sample_list</code></td>
<td>
<p>sample_list (character)
Optional parameter specifying the set of samples to include in the mutation matrices.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
Optional parameter specifying the set of genes to include in the mutation matrices.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_acceptable_genes">acceptable_genes</code></td>
<td>
<p>(character)
Optional parameter specifying a set of acceptable genes, for example those which are in an ensembl databse.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_for_biomarker">for_biomarker</code></td>
<td>
<p>(character)
Used for defining a dictionary of mutations. See the function get_mutation_dictionary() for details.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_include_synonymous">include_synonymous</code></td>
<td>
<p>(logical)
Optional parameter specifying whether to include synonymous mutations in the mutation matrices.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_dictionary">dictionary</code></td>
<td>
<p>(character)
Optional parameter directly specifying the mutation dictionary to use. See the function get_mutation_dictionary() for details.</p>
</td></tr>
<tr><td><code id="get_mutation_tables_+3A_seed_id">seed_id</code></td>
<td>
<p>(numeric)
Input value for the function set.seed().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three items with names 'train', 'val' and 'test'. Each element will contain a sparse mutation matrix for the samples in that branch, alongside other information
as described as the output of the function get_table_from_maf().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>tables &lt;- get_mutation_tables(example_maf_data$maf, sample_list = paste0("SAMPLE_", 1:100))

print(names(tables))
print(names(tables$train))
</code></pre>

<hr>
<h2 id='get_p'>Construct Optimisation Parameters.</h2><span id='topic+get_p'></span>

<h3>Description</h3>

<p>An internal function. From the learned generative model and training data, produces a vector of weights p to be used in
the subsequent group lasso optimisation, alongside a biomarker-dependent normalisation quantity p_norm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_p(gen_model, training_matrix, marker_mut_types, gene_lengths)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_p_+3A_gen_model">gen_model</code></td>
<td>
<p>(list)
A generative mutation model, fitted by fit_gen_model().</p>
</td></tr>
<tr><td><code id="get_p_+3A_training_matrix">training_matrix</code></td>
<td>
<p>(sparse matrix)
A sparse matrix of mutations in the training dataset, produced by get_mutation_tables().</p>
</td></tr>
<tr><td><code id="get_p_+3A_marker_mut_types">marker_mut_types</code></td>
<td>
<p>(character)
A character vector listing which mutation types (of the set specified in the generative model
attribute 'names') constitute the biomarker in question.</p>
</td></tr>
<tr><td><code id="get_p_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A table with two columns: Hugo_Symbol and max_cds, providing the lengths of the genes to be modelled.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three entries:
</p>

<ul>
<li><p> A vector p, with an entry corresponding to each combination of gene and mutation type specified
in the generative model fitted. Each component is a non-negative value corresponding to a weighting
p to be supplied to a group lasso optimisation.
</p>
</li>
<li><p> A numeric p_norm, giving the factor between p_gs and phi_0gs (see paper for details).
</p>
</li>
<li><p> A vector biomarker_columns, detailing which of the elements of p correspond to gene/mutation type
combinations contributing to the biomarker in question.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- get_p(example_gen_model, example_tables$train$matrix,
           marker_mut_types = c("I"), gene_lengths = example_maf_data$gene_lengths)
print(p$p[1:5])
print(p$p_norm)
print(p$bc[1:5])
</code></pre>

<hr>
<h2 id='get_panels_from_fit'>Extract Panel Details from Group Lasso Fit</h2><span id='topic+get_panels_from_fit'></span>

<h3>Description</h3>

<p>An internal function for analysing a group Lasso fit as part of the predictive model learning procedure, which returns the sets of genes identified by different iterations of the group Lasso algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_panels_from_fit(gene_lengths, fit, gene_list, mut_types_list)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_panels_from_fit_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A table with two columns: Hugo_Symbol and max_cds, providing the lengths of the genes to be modelled.</p>
</td></tr>
<tr><td><code id="get_panels_from_fit_+3A_fit">fit</code></td>
<td>
<p>(list)
A fit from the group lasso algorithm, produced by the function gglasso (package: gglasso).</p>
</td></tr>
<tr><td><code id="get_panels_from_fit_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
A character vector of genes listing the genes (in order) included in the model pred_fit.</p>
</td></tr>
<tr><td><code id="get_panels_from_fit_+3A_mut_types_list">mut_types_list</code></td>
<td>
<p>(character)
A character vector listing the mutation type groupings (in order) included in the model pred_fit.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two elements:
</p>

<ul>
<li><p> panel_genes: A matrix where each row corresponds to a gene, each column to an iteration of the group
lasso with a different penalty factor, and the elements booleans specifying whether that gene was selected
to be included in that iteration.
</p>
</li>
<li><p> panel_lengths:
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>panels &lt;- get_panels_from_fit(example_maf_data$gene_lengths, example_first_pred_tmb$fit,
example_gen_model$names$gene_list, mut_types_list = example_gen_model$names$mut_types_list)

print(panels$fit)
</code></pre>

<hr>
<h2 id='get_predictions'>Produce Predictions on an Unseen Dataset</h2><span id='topic+get_predictions'></span>

<h3>Description</h3>

<p>A function taking a predictive model(s) and new observations, and applying the predictive model to them to return predicted biomarker values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_predictions(pred_model, new_data, s = NULL, max_panel_length = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_predictions_+3A_pred_model">pred_model</code></td>
<td>
<p>(list)
A predictive model as fitted by pred_first_fit(), pred_refit_panel() or
pred_refit_range().</p>
</td></tr>
<tr><td><code id="get_predictions_+3A_new_data">new_data</code></td>
<td>
<p>(list)
A new dataset, containing a matrix of observations and a list of sample IDs.
Likely comes from the 'train', 'val' or 'test' argument of a call to
get_mutation_tables().</p>
</td></tr>
<tr><td><code id="get_predictions_+3A_s">s</code></td>
<td>
<p>(numeric)
If producing predictions for a single panel, s chooses which panel
(column in a pred_fit object) to produce predictions for.</p>
</td></tr>
<tr><td><code id="get_predictions_+3A_max_panel_length">max_panel_length</code></td>
<td>
<p>(numeric)
If producing predictions for a single panel, maximum panel length to
specify that panel.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two elements:
</p>

<ul>
<li><p> predictions, a matrix containing a row for each sample and a column for each
panel.
</p>
</li>
<li><p> panel_lengths, a vector containing the length of each panel.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_predictions &lt;- get_predictions(example_refit_range, new_data =
example_tables$val)
</code></pre>

<hr>
<h2 id='get_r_squared'>R Squared Metrics for Predictions</h2><span id='topic+get_r_squared'></span>

<h3>Description</h3>

<p>A function to return R^2 metrics for predictions vs actual values. Works well when piped to straight from get_predictions().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_r_squared(predictions, biomarker_values, model = "", threshold = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_r_squared_+3A_predictions">predictions</code></td>
<td>
<p>(list)
A list with two elements, 'predictions' and 'panel_lengths',
as produced by the function get_predictions().</p>
</td></tr>
<tr><td><code id="get_r_squared_+3A_biomarker_values">biomarker_values</code></td>
<td>
<p>(dataframe)
A dataframe with two columns, 'Tumor_Sample_Barcode' and a
column with the name of the biomarker in question containing values.</p>
</td></tr>
<tr><td><code id="get_r_squared_+3A_model">model</code></td>
<td>
<p>(character)
The name of the model type producing these predictions.</p>
</td></tr>
<tr><td><code id="get_r_squared_+3A_threshold">threshold</code></td>
<td>
<p>(numeric)
Unusued in this function: present for calls to get_stats().</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with 5 columns:
</p>

<ul>
<li><p> panel_length: the length of each panel.
</p>
</li>
<li><p> model: the model that produced the predictions.
</p>
</li>
<li><p> biomarker: the name of the biomarker in question.
</p>
</li>
<li><p> stat: the R squared values for each panel.
</p>
</li>
<li><p> metric: a constant character &quot;R&quot; for R squared.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_r &lt;- get_r_squared(predictions = get_predictions(example_refit_panel, new_data =
  example_tables$val), biomarker_values = example_tmb_tables$val, model = "Refitted T")
</code></pre>

<hr>
<h2 id='get_stats'>Metrics for Predictive Performance</h2><span id='topic+get_stats'></span>

<h3>Description</h3>

<p>A function to return a variety metrics for predictions vs actual values. Works well when piped to straight from get_predictions().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_stats(
  predictions,
  biomarker_values,
  model = "",
  threshold = 300,
  metrics = c("R", "AUPRC")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_stats_+3A_predictions">predictions</code></td>
<td>
<p>(list)
A list with two elements, 'predictions' and 'panel_lengths',
as produced by the function get_predictions().</p>
</td></tr>
<tr><td><code id="get_stats_+3A_biomarker_values">biomarker_values</code></td>
<td>
<p>(dataframe)
A dataframe with two columns, 'Tumor_Sample_Barcode' and a
column with the name of the biomarker in question containing values.</p>
</td></tr>
<tr><td><code id="get_stats_+3A_model">model</code></td>
<td>
<p>(character)
The name of the model type producing these predictions.</p>
</td></tr>
<tr><td><code id="get_stats_+3A_threshold">threshold</code></td>
<td>
<p>(numeric)
The threshold for biomarker high/low categorisation.</p>
</td></tr>
<tr><td><code id="get_stats_+3A_metrics">metrics</code></td>
<td>
<p>(character)
A vector of the names of metrics to calculate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe with 5 columns:
</p>

<ul>
<li><p> panel_length: the length of each panel.
</p>
</li>
<li><p> model: the model that produced the predictions.
</p>
</li>
<li><p> biomarker: the name of the biomarker in question.
</p>
</li>
<li><p> stat: the metric values for each panel.
</p>
</li>
<li><p> metric: the name of the metric.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_stat &lt;- get_stats(predictions = get_predictions(example_refit_panel,
new_data = example_tables$val), biomarker_values = example_tmb_tables$val,
model = "Refitted T", threshold = 10)
</code></pre>

<hr>
<h2 id='get_table_from_maf'>Produce a Mutation Matrix from a MAF</h2><span id='topic+get_table_from_maf'></span>

<h3>Description</h3>

<p>A function to, given a mutation annotation dataset with columns for sample barcode, gene name and mutation type, to reformulate this as a mutation matrix,
with rows denoting samples, columns denoting gene/mutation type combinations, and the individual entries giving the number of mutations observed. This
will likely be very sparse, so we save it as a sparse matrix for efficiency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_table_from_maf(
  maf,
  sample_list = NULL,
  gene_list = NULL,
  acceptable_genes = NULL,
  for_biomarker = "TIB",
  include_synonymous = TRUE,
  dictionary = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_table_from_maf_+3A_maf">maf</code></td>
<td>
<p>(dataframe)
A table of annotated mutations containing the columns 'Tumor_Sample_Barcode', 'Hugo_Symbol', and 'Variant_Classification'.</p>
</td></tr>
<tr><td><code id="get_table_from_maf_+3A_sample_list">sample_list</code></td>
<td>
<p>(character)
Optional parameter specifying the set of samples to include in the mutation matrix.</p>
</td></tr>
<tr><td><code id="get_table_from_maf_+3A_gene_list">gene_list</code></td>
<td>
<p>(character)
Optional parameter specifying the set of genes to include in the mutation matrix.</p>
</td></tr>
<tr><td><code id="get_table_from_maf_+3A_acceptable_genes">acceptable_genes</code></td>
<td>
<p>(character)
Optional parameter specifying a set of acceptable genes, for example those which are in an ensembl databse.</p>
</td></tr>
<tr><td><code id="get_table_from_maf_+3A_for_biomarker">for_biomarker</code></td>
<td>
<p>(character)
Used for defining a dictionary of mutations. See the function get_mutation_dictionary() for details.</p>
</td></tr>
<tr><td><code id="get_table_from_maf_+3A_include_synonymous">include_synonymous</code></td>
<td>
<p>(logical)
Optional parameter specifying whether to include synonymous mutations in the mutation matrix.</p>
</td></tr>
<tr><td><code id="get_table_from_maf_+3A_dictionary">dictionary</code></td>
<td>
<p>(character)
Optional parameter directly specifying the mutation dictionary to use. See the function get_mutation_dictionary() for details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the following entries:
</p>

<ul>
<li><p> matrix: A mutation matrix, a sparse matrix showing the number of mutations present in each sample, gene and mutation type.
</p>
</li>
<li><p> sample_list: A vector of characters specifying the samples included in the matrix: the rows of the mutation matrix correspond to each of these.
</p>
</li>
<li><p> gene_list: A vector of characters specifying the the genes included in the matrix.
</p>
</li>
<li><p> mut_types_list: A vector of characters specifying the mutation types (as grouped into an appropriate dictionary) to be included in the matrix.
</p>
</li>
<li><p> col_names: A vector of characters identifying the columns of the mutation matrix. Each entry will be comprised of two parts separated by the character
'_', the first identifying the gene in question and the second identifying the mutation type. E.g. 'GENE1_NS&quot; where 'GENE1' is an element of gene_list,
and 'NS' is an element of the dictionary vector.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># We use the preloaded maf file example_maf_data
# Now we make a mutation matrix
table &lt;- get_table_from_maf(example_maf_data$maf, sample_list = paste0("SAMPLE_", 1:100))

print(names(table))
print(table$matrix[1:10,1:10])
print(table$col_names[1:10])
</code></pre>

<hr>
<h2 id='ICBioMark'>ICBioMark: A package for cost-effective design of gene panels to predict
exome-wide biomarkers.</h2><span id='topic+ICBioMark'></span>

<h3>Description</h3>

<p>This package implements the methodology proposed in
'Data-driven design of targeted gene panels for estimating immunotherapy biomarkers',
(Bradley and Cannings, 2021, preprint). It allows the user to fit generative models of
mutation from an annotated mutation dataset, and then further to produce tunable linear
estimators of exome-wide biomarkers. It also contains functions to simulate mutation
annotated format (MAF) data, as well as to analyse the output and performance of models.
</p>

<hr>
<h2 id='nsclc_maf'>Non-Small Cell Lung Cancer MAF Data</h2><span id='topic+nsclc_maf'></span>

<h3>Description</h3>

<p>A pre-loaded mutation dataset from Campbell et. al (2016), downloaded from The Cancer Genome Atlas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nsclc_maf
</code></pre>


<h3>Format</h3>

<p>An annotated mutation dataframe with 6 columns and 299855 rows:
</p>

<dl>
<dt>Tumor_Sample_Barcode</dt><dd><p>A sample id for each mutation.</p>
</dd>
<dt>Hugo_Symbol</dt><dd><p>The name of the gene location for each mutation.</p>
</dd>
<dt>Variant_Classification</dt><dd><p>The mutation type for each mutation.</p>
</dd>
<dt>Chromosome</dt><dd><p>Chromosome on which the mutation occurred.</p>
</dd>
<dt>Start_Position</dt><dd><p>Start nucleotide location for mutation.</p>
</dd>
<dt>End_Position</dt><dd><p>End nucleotide location for mutation.</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.cbioportal.org/study/summary?id=nsclc_tcga_broad_2016">https://www.cbioportal.org/study/summary?id=nsclc_tcga_broad_2016</a>
</p>

<hr>
<h2 id='nsclc_survival'>Non-Small Cell Lung Cancer Survival and Clinical Data</h2><span id='topic+nsclc_survival'></span>

<h3>Description</h3>

<p>A pre-loaded clinical dataset containing survival and clinical data from Campbell et. al (2016), downloaded from The Cancer Genome Atlas.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nsclc_survival
</code></pre>


<h3>Format</h3>

<p>An annotated mutation dataframe with 23 columns and 1144 rows. Each row
corresponds to a sample, and details clinical and survival information about the
patient from whom the sample was derived. Its columns are as follows:
</p>

<dl>
<dt>CASE_ID</dt><dd></dd>
<dt>AGE</dt><dd></dd>
<dt>AGE_AT_SURGERY</dt><dd></dd>
<dt>CANCER_TYPE</dt><dd></dd>
<dt>CANCER_TYPE_DETAILED</dt><dd></dd>
<dt>DAYS_TO_DEATH</dt><dd></dd>
<dt>DAYS_TO_LAST_FOLLOWUP</dt><dd></dd>
<dt>FRACTION_GENOME_ALTERED</dt><dd></dd>
<dt>HISTORY_NEOADJUVANT_TRTYN</dt><dd></dd>
<dt>HISTORY_OTHER_MALIGNANCY</dt><dd></dd>
<dt>MUTATION_COUNT</dt><dd></dd>
<dt>M_STAGE</dt><dd></dd>
<dt>N_STAGE</dt><dd></dd>
<dt>ONCOTREE_CODE</dt><dd></dd>
<dt>OS_MONTHS</dt><dd></dd>
<dt>OS_STATUS</dt><dd></dd>
<dt>SAMPLE_COUNT</dt><dd></dd>
<dt>SEX</dt><dd></dd>
<dt>SMOKING_HISTORY</dt><dd></dd>
<dt>SMOKING_PACK_YEARS</dt><dd></dd>
<dt>SOMATIC_STATUS</dt><dd></dd>
<dt>STAGE</dt><dd></dd>
<dt>T_STAGE</dt><dd></dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.cbioportal.org/study/clinicalData?id=nsclc_tcga_broad_2016">https://www.cbioportal.org/study/clinicalData?id=nsclc_tcga_broad_2016</a>
</p>

<hr>
<h2 id='pred_first_fit'>First-Fit Predicitve Model with Group Lasso</h2><span id='topic+pred_first_fit'></span>

<h3>Description</h3>

<p>This function implements the first-fit procedure described in Bradley and Cannings, 2021. It requires at least a generative model and a dataframe containing gene lengths as input.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_first_fit(
  gen_model,
  lambda = exp(seq(-16, -24, length.out = 100)),
  biomarker = "TMB",
  marker_mut_types = c("NS", "I"),
  training_matrix,
  gene_lengths,
  marker_training_values = NULL,
  K_method = max,
  free_genes = c()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_first_fit_+3A_gen_model">gen_model</code></td>
<td>
<p>(list)
A generative mutation model, fitted by fit_gen_model().</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_lambda">lambda</code></td>
<td>
<p>(numeric)
A vector of penalisation weights for input to the group lasso optimiser gglasso.</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_biomarker">biomarker</code></td>
<td>
<p>(character)
The biomarker in question. If &quot;TMB&quot; or &quot;TIB&quot;, then automatically defines the subsequent
variable marker_mut_types.</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_marker_mut_types">marker_mut_types</code></td>
<td>
<p>(character)
The set of mutation type groupings constituting the biomarker being estimated. Should be
a vector comprising of elements of the mut_types_list vector in the 'names' attribute of
gen_model.</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_training_matrix">training_matrix</code></td>
<td>
<p>(sparse matrix)
A sparse matrix of mutations in the training dataset, produced by get_mutation_tables().</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A table with two columns: Hugo_Symbol and max_cds, providing the lengths of the genes to be modelled.</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_marker_training_values">marker_training_values</code></td>
<td>
<p>(dataframe)
A dataframe containing two columns: 'Tumor_Sample_Barcode', containing the sample IDs for the training
dataset, and a second column containing training values for the biomarker in question.</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_k_method">K_method</code></td>
<td>
<p>(function)
How to select a representative biomarker value from the training dataset. Defaults to max().</p>
</td></tr>
<tr><td><code id="pred_first_fit_+3A_free_genes">free_genes</code></td>
<td>
<p>(character)
Which genes should escape penalisation (for example when augmenting a pre-existing panel).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of six elements:
</p>

<ul>
<li><p> fit: Output of call to gglasso.
</p>
</li>
<li><p> panel_genes: A matrix where each row corresponds to a gene, each column to an iteration of the group
lasso with a different penalty factor, and the elements booleans specifying whether that gene was selected
to be included in that iteration.
</p>
</li>
<li><p> panel_lengths: A vector giving total panel length for each gglasso iteration.
</p>
</li>
<li><p> p: The vector of weights used in the optimisation procedure.
</p>
</li>
<li><p> K: The bias penalty factor used in the optimisation procedure.
</p>
</li>
<li><p> names: Gene and mutation type information as used when fitting the generative model.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_first_fit &lt;- pred_first_fit(example_gen_model, lambda = exp(seq(-9, -14, length.out = 100)),
                                    training_matrix = example_tables$train$matrix,
                                    gene_lengths = example_maf_data$gene_lengths)

</code></pre>

<hr>
<h2 id='pred_intervals'>Produce Error Bounds for Predictions</h2><span id='topic+pred_intervals'></span>

<h3>Description</h3>

<p>A function to produce a confidence region for a linear predictor. In upcoming versions will (hopefully) be greatly simplified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_intervals(
  predictions,
  pred_model,
  gen_model,
  training_matrix,
  gene_lengths,
  biomarker_values,
  alpha = 0.1,
  range_factor = 1.1,
  s = NULL,
  max_panel_length = NULL,
  biomarker = "TMB",
  marker_mut_types = c("NS", "I"),
  model = "Refitted T"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_intervals_+3A_predictions">predictions</code></td>
<td>
<p>(list)
A predictions object, as produced by get_predictions().</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_pred_model">pred_model</code></td>
<td>
<p>(list)
A predictive model, as produced by pred_first_fit(), pred_refit_panel() or
pred_refit_range().</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_gen_model">gen_model</code></td>
<td>
<p>(list)
A generative model, as produce by fit_gen_model</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_training_matrix">training_matrix</code></td>
<td>
<p>(sparse matrix)
A training matrix, as produced by get_tables()$matrix or get_table_from_maf()$matrix.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(data frame)
A data frame with columns 'Hugo_Symbol' and 'max_cds'. See example_maf_data$gene_lengths, or
ensembl_gene_lengths for examples.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_biomarker_values">biomarker_values</code></td>
<td>
<p>(data frame)
A data frame containing the true values of the biomarker in question.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_alpha">alpha</code></td>
<td>
<p>(numeric)
Confidence level for error bounds.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_range_factor">range_factor</code></td>
<td>
<p>(numeric)
Value specifying how far beyond the range of max(biomarker) to plot confidence region.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_s">s</code></td>
<td>
<p>(numeric)
If input predictions are for a range of panels, s chooses which panel
(column in a pred_fit object) to produce predictions for.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_max_panel_length">max_panel_length</code></td>
<td>
<p>(numeric)
Select panel by maximum length.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_biomarker">biomarker</code></td>
<td>
<p>(character)
Which biomarker is being predicted.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_marker_mut_types">marker_mut_types</code></td>
<td>
<p>(character)
If biomarker is not one of &quot;TMB&quot; or &quot;TIB&quot;, then this is required to specify which mutation type
groups constitute the biomarker.</p>
</td></tr>
<tr><td><code id="pred_intervals_+3A_model">model</code></td>
<td>
<p>(character)
The model (must be based on a linear estimator) for which prediction intervals are being generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with two entries:
</p>

<ul>
<li><p> prediction_intervals:
</p>
</li>
<li><p> confidence_region:
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_intervals &lt;- pred_intervals(predictions = get_predictions(example_refit_range,
               new_data = example_tables$val),
               pred_model = example_refit_range, biomarker_values = example_tmb_tables$val,
               gen_model = example_gen_model, training_matrix = example_tables$train$matrix,
               max_panel_length = 15000, gene_lengths = example_maf_data$gene_lengths)

example_confidence_plot &lt;- ggplot2::ggplot() +
  ggplot2::geom_point(data = example_intervals$prediction_intervals,
             ggplot2::aes(x = true_value, y = estimated_value)) +
        ggplot2::geom_ribbon(data = example_intervals$confidence_region,
          ggplot2::aes(x = x, ymin = y_lower, ymax = y_upper),
                    fill = "red", alpha = 0.2) +
        ggplot2::geom_line(data = example_intervals$confidence_region,
          ggplot2::aes(x = x, y = y), linetype = 2) +
        ggplot2::scale_x_log10() + ggplot2::scale_y_log10()

plot(example_confidence_plot)
</code></pre>

<hr>
<h2 id='pred_refit_panel'>Refitted Predictive Model for a Given Panel</h2><span id='topic+pred_refit_panel'></span>

<h3>Description</h3>

<p>A function taking the output of a call to pred_first_fit(), as well as gene length information, and a specified panel (list of genes), and producing a refitted predictive model on that given panel.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_refit_panel(
  pred_first = NULL,
  gene_lengths = NULL,
  model = "T",
  genes,
  biomarker = "TMB",
  marker_mut_types = c("NS", "I"),
  training_data = NULL,
  training_values = NULL,
  mutation_vector = NULL,
  t_s = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_refit_panel_+3A_pred_first">pred_first</code></td>
<td>
<p>(list)
A first-fit predictive model as produced by pred_first_fit().</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A dataframe of gene lengths (see example_maf_data$gene_lengths for format).</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_model">model</code></td>
<td>
<p>(character)
A choice of &quot;T&quot;, &quot;OLM&quot; or &quot;Count&quot; specifying how predictions should be made.</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_genes">genes</code></td>
<td>
<p>(character)
A vector of gene names detailing the panel being used.</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_biomarker">biomarker</code></td>
<td>
<p>(character)
If &quot;TMB&quot; or &quot;TIB&quot;, automatically defines marker_mut_types, otherwise this will
need to be specified separately.</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_marker_mut_types">marker_mut_types</code></td>
<td>
<p>(character)
A vector specifying which mutation types groups determine the biomarker in question.</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_training_data">training_data</code></td>
<td>
<p>(list)
Training data, as produced by get_mutation_tables() (select train, val or test).</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_training_values">training_values</code></td>
<td>
<p>(dataframe)
Training true values, as produced by get_biomarker_tables() (select train, val or test).</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_mutation_vector">mutation_vector</code></td>
<td>
<p>(numeric)
Optional vector specifying the values of the training matrix (training_data$matrix) in
vector rather than matrix form.</p>
</td></tr>
<tr><td><code id="pred_refit_panel_+3A_t_s">t_s</code></td>
<td>
<p>(numeric)
Optional vector specifying the frequencies of different mutation types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three elements:
</p>

<ul>
<li><p> fit, a list including a sparse matrix 'beta' giving prediction weights.
</p>
</li>
<li><p> panel_genes, a sparse (logical) matrix giving the genes included in prediction.
</p>
</li>
<li><p> panel_lengths, a singleton vector giving the length of the panel used.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_refit_panel &lt;- pred_refit_panel(pred_first = example_first_pred_tmb,
  gene_lengths = example_maf_data$gene_lengths, genes = paste0("GENE_", 1:10))
</code></pre>

<hr>
<h2 id='pred_refit_range'>Get Refitted Predictive Models for a First-Fit Range of Panels</h2><span id='topic+pred_refit_range'></span>

<h3>Description</h3>

<p>A function producing a refitted predictive model for each panel produced by usage of the function pred_first_fit(), by repeatedly applying the function pred_refit_panel().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pred_refit_range(
  pred_first = NULL,
  gene_lengths = NULL,
  model = "T",
  biomarker = "TMB",
  marker_mut_types = c("NS", "I"),
  training_data = NULL,
  training_values = NULL,
  mutation_vector = NULL,
  t_s = NULL,
  max_panel_length = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pred_refit_range_+3A_pred_first">pred_first</code></td>
<td>
<p>(list)
A first-fit predictive model as produced by pred_first_fit().</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_gene_lengths">gene_lengths</code></td>
<td>
<p>(dataframe)
A dataframe of gene lengths (see example_maf_data$gene_lengths for format).</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_model">model</code></td>
<td>
<p>(character)
A choice of &quot;T&quot;, &quot;OLM&quot; or &quot;Count&quot; specifying how predictions should be made.</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_biomarker">biomarker</code></td>
<td>
<p>(character)
If &quot;TMB&quot; or &quot;TIB&quot;, automatically defines marker_mut_types, otherwise this will
need to be specified separately.</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_marker_mut_types">marker_mut_types</code></td>
<td>
<p>(character)
A vector specifying which mutation types groups determine the biomarker in question.</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_training_data">training_data</code></td>
<td>
<p>(sparse matrix)
Training matrix, as produced by get_mutation_tables() (select train, val or test).</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_training_values">training_values</code></td>
<td>
<p>(dataframe)
Training true values, as produced by get_biomarker_tables() (select train, val or test).</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_mutation_vector">mutation_vector</code></td>
<td>
<p>(numeric)
Optional vector specifying the values of the training matrix (training_data$matrix) in
vector rather than matrix form.</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_t_s">t_s</code></td>
<td>
<p>(numeric)
Optional vector specifying the frequencies of different mutation types.</p>
</td></tr>
<tr><td><code id="pred_refit_range_+3A_max_panel_length">max_panel_length</code></td>
<td>
<p>(numeric)
Upper bound for panels to fit refitted models to. Most useful for &quot;OLM&quot; and &quot;Count&quot;
model types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three elements:
</p>

<ul>
<li><p> fit, a list including a sparse matrix 'beta' giving prediction weights for each
first-fit panel (one panel per column).
</p>
</li>
<li><p> panel_genes, a sparse (logical) matrix giving the genes included in prediction
for each first-fit panel.
</p>
</li>
<li><p> panel_lengths, a vector giving the length of each first-fit panel.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>example_refit_range &lt;- pred_refit_range(pred_first = example_first_pred_tmb,
  gene_lengths = example_maf_data$gene_lengths)
</code></pre>

<hr>
<h2 id='vis_model_fit'>Visualise Generative Model Fit</h2><span id='topic+vis_model_fit'></span>

<h3>Description</h3>

<p>A function to visualise how well a general model has fitted to a mutation dataset across cross-validation folds. Designed to produce a similar output to glmnet's function plot.cv.glmnet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vis_model_fit(
  gen_model,
  x_sparsity = FALSE,
  y_sparsity = FALSE,
  mut_type = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vis_model_fit_+3A_gen_model">gen_model</code></td>
<td>
<p>(list)
A generative model fitted by fit_gen_model()</p>
</td></tr>
<tr><td><code id="vis_model_fit_+3A_x_sparsity">x_sparsity</code></td>
<td>
<p>Show model sparsity on x axis rather than lambda.</p>
</td></tr>
<tr><td><code id="vis_model_fit_+3A_y_sparsity">y_sparsity</code></td>
<td>
<p>Show model sparsity on y axis rather than deviance.</p>
</td></tr>
<tr><td><code id="vis_model_fit_+3A_mut_type">mut_type</code></td>
<td>
<p>Produce separate plots for each mutation type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Summary plot of the generative model fit across folds.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p &lt;- vis_model_fit(example_gen_model)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
