<!DOCTYPE html><html lang="en"><head><title>Help for package ICDS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ICDS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ICDS-package'><p>Identification of Cancer Dysfunctional Subpathway by integrating DNA methylation, copy number variation, and gene expression data</p></a></li>
<li><a href='#combinep_three'><p>combinep_three</p></a></li>
<li><a href='#combinep_two'><p>combinep_two</p></a></li>
<li><a href='#coverp2zscore'><p>coverp2zscore</p></a></li>
<li><a href='#envData'><p>The variables in the environment include an example expression profile,an methylation profile,an copy number variation data,amplified genes,deleted genes,A numeric vector of z_scores,p-values,A vector of 0/1s, indicating the class of samples,interested subpathways,Optimized subpathway,and the  statistical significance p value and FDR for these optimal subpathways</p></a></li>
<li><a href='#FindSubPath'><p>FindSubPath</p></a></li>
<li><a href='#getCnvp'><p>getCnvp</p></a></li>
<li><a href='#GetExampleData'><p>Get the example data</p></a></li>
<li><a href='#getExpp'><p>getExpp</p></a></li>
<li><a href='#getMethp'><p>getMethp</p></a></li>
<li><a href='#opt_subpath'><p>opt_subpath</p></a></li>
<li><a href='#Permutation'><p>Permutation</p></a></li>
<li><a href='#PlotSubpathway'><p>PlotSubpathway</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of Cancer Dysfunctional Subpathway with Omics
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identify Cancer Dysfunctional Sub-pathway by integrating gene expression, DNA methylation and copy number variation, and pathway topological information. 1)We firstly calculate the gene risk scores by integrating three kinds of data: DNA methylation, copy number variation, and gene expression.  2)Secondly, we perform a greedy search algorithm to identify the key dysfunctional sub-pathways within the pathways for which the discriminative scores were locally maximal. 3)Finally, the permutation test was used to calculate statistical significance level for these key dysfunctional sub-pathways.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>igraph, graphite, metap, methods, org.Hs.eg.db</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-01 06:56:52 UTC; btzheng</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [cre],
  Baotong Zheng [aut],
  Siyao Liu [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-01 09:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ICDS-package'>Identification of Cancer Dysfunctional Subpathway by integrating DNA methylation, copy number variation, and gene expression data</h2><span id='topic+ICDS-package'></span><span id='topic+ICDS'></span>

<h3>Description</h3>

<p>Identify Cancer Dysfunctional Subpathway by integrating gene expression, DNA methylation and copy number variation, and pathway topological information. 1)We firstly calculate the gene risk scores by integrating three kinds of data: DNA methylation, copy number variation, and gene expression.  2)Secondly, we perform a greedy search algorithm to identify the key dysfunctional subpathways within the pathways for which the discriminative scores were locally maximal. 3)Finally, the permutation test was used to calculate statistical significance level for these key dysfunctional subpathways.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Junwei Han <a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Baotong Zheng <a href="mailto:btzheng1116@163.com">btzheng1116@163.com</a>
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a> [contributor]
</p>
</li></ul>


<hr>
<h2 id='combinep_three'>combinep_three</h2><span id='topic+combinep_three'></span>

<h3>Description</h3>

<p>'combinep_three' combine three kinds of p-values,then,calculate z-score for them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combinep_three(p1, p2, p3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combinep_three_+3A_p1">p1</code></td>
<td>
<p>the p-values or corrected p-values</p>
</td></tr>
<tr><td><code id="combinep_three_+3A_p2">p2</code></td>
<td>
<p>the p-values or corrected p-values</p>
</td></tr>
<tr><td><code id="combinep_three_+3A_p3">p3</code></td>
<td>
<p>the p-values or corrected p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of z_scores
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.p&lt;-GetExampleData("exp.p")
meth.p&lt;-GetExampleData("meth.p")
cnv.p&lt;-GetExampleData("cnv.p")
combinep_three(exp.p,meth.p,cnv.p)
</code></pre>

<hr>
<h2 id='combinep_two'>combinep_two</h2><span id='topic+combinep_two'></span>

<h3>Description</h3>

<p>'combinep_two' combine two kinds of p-values,then,calculate z-score for them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combinep_two(p1, p2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combinep_two_+3A_p1">p1</code></td>
<td>
<p>A numeric vector of p-values or corrected p-values</p>
</td></tr>
<tr><td><code id="combinep_two_+3A_p2">p2</code></td>
<td>
<p>A numeric vector of p-values or corrected p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of z_scores
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.p&lt;-GetExampleData("exp.p")
meth.p&lt;-GetExampleData("meth.p")
combinep_two(exp.p,meth.p)
</code></pre>

<hr>
<h2 id='coverp2zscore'>coverp2zscore</h2><span id='topic+coverp2zscore'></span>

<h3>Description</h3>

<p>'coverp2zscore' calculate z-scores for p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coverp2zscore(pdata)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="coverp2zscore_+3A_pdata">pdata</code></td>
<td>
<p>A numeric vector of p-values or corrected p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of z_scores
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp.p&lt;-GetExampleData("exp.p")
meth.p&lt;-GetExampleData("meth.p")
cnv.p&lt;-GetExampleData("cnv.p")
coverp2zscore(exp.p)
coverp2zscore(meth.p)
coverp2zscore(cnv.p)
</code></pre>

<hr>
<h2 id='envData'>The variables in the environment include an example expression profile,an methylation profile,an copy number variation data,amplified genes,deleted genes,A numeric vector of z_scores,p-values,A vector of 0/1s, indicating the class of samples,interested subpathways,Optimized subpathway,and the  statistical significance p value and FDR for these optimal subpathways
</h2><span id='topic+envData'></span><span id='topic+exp_data'></span><span id='topic+meth_data'></span><span id='topic+cnv_data'></span><span id='topic+amp_gene'></span><span id='topic+del_gene'></span><span id='topic+zzz'></span><span id='topic+exp.p'></span><span id='topic+meth.p'></span><span id='topic+cnv.p'></span><span id='topic+label1'></span><span id='topic+label2'></span><span id='topic+subpathdata'></span><span id='topic+opt_subpathways'></span>

<h3>Description</h3>

<p>Identify Cancer Dysfunctional Subpathway by integrating gene expression, DNA methylation and copy number variation, and pathway topological information. 1)We firstly calculate the gene risk scores by integrating three kinds of data: DNA methylation, copy number variation, and gene expression.  2)Secondly, we perform a greedy search algorithm to identify the key dysfunctional subpathways within the pathways for which the discriminative scores were locally maximal. 3)Finally, the permutation test was used to calculate statistical significance level for these key dysfunctional subpathways.
</p>


<h3>Format</h3>

<p>An environment variable</p>


<h3>Details</h3>

<p>The environment variable includes the variable <code>exp_data</code>, <code>meth_data</code>,<code>cnv_data</code>,<code>amp_gene</code>,<code>del_gene</code>,<code>zzz</code>,<code>exp.p</code>,<code>meth.p</code>,<code>cnv.p</code>,<code>label1</code>,<code>label2</code>,<code>subpathdata</code>,<code>opt_subpathways</code>
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Baotong Zheng<a href="mailto:btzheng1116@163.com">btzheng1116@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>

<hr>
<h2 id='FindSubPath'>FindSubPath</h2><span id='topic+FindSubPath'></span>

<h3>Description</h3>

<p>'FindSubPath' uses a greedy search algorithm to search for key subpathways in each entire pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindSubPath(
  zz,
  Pathway = "kegg",
  delta = 0.05,
  seed_p = 0.05,
  min.size = 5,
  out.F = FALSE,
  out.file = "Subpath.txt"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FindSubPath_+3A_zz">zz</code></td>
<td>
<p>A numeric vector of z_scores.</p>
</td></tr>
<tr><td><code id="FindSubPath_+3A_pathway">Pathway</code></td>
<td>
<p>The name of the pathway database.</p>
</td></tr>
<tr><td><code id="FindSubPath_+3A_delta">delta</code></td>
<td>
<p>Diffusion coefficient in each step of searching subpath.</p>
</td></tr>
<tr><td><code id="FindSubPath_+3A_seed_p">seed_p</code></td>
<td>
<p>Define gene whose p-value smaller than seed_p as seed gene.</p>
</td></tr>
<tr><td><code id="FindSubPath_+3A_min.size">min.size</code></td>
<td>
<p>The smallest size of subpathways.</p>
</td></tr>
<tr><td><code id="FindSubPath_+3A_out.f">out.F</code></td>
<td>
<p>Logical,tell if output subpathways.</p>
</td></tr>
<tr><td><code id="FindSubPath_+3A_out.file">out.file</code></td>
<td>
<p>file name of subpathways.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Key dysfunctional subpathways in each pathway, in which the risk score of the genes were significantly higher.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(graphite)
zz&lt;-GetExampleData("zzz")
k&lt;-FindSubPath(zz)
</code></pre>

<hr>
<h2 id='getCnvp'>getCnvp</h2><span id='topic+getCnvp'></span>

<h3>Description</h3>

<p>'getCnvp' perform t-test on copy number variation data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCnvp(
  exp_data,
  cnv_data,
  amp_gene,
  del_gene,
  p.adjust = TRUE,
  method = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getCnvp_+3A_exp_data">exp_data</code></td>
<td>
<p>A data frame</p>
</td></tr>
<tr><td><code id="getCnvp_+3A_cnv_data">cnv_data</code></td>
<td>
<p>Copy number variation data</p>
</td></tr>
<tr><td><code id="getCnvp_+3A_amp_gene">amp_gene</code></td>
<td>
<p>A vector of strings, the IDs of amplified genes.</p>
</td></tr>
<tr><td><code id="getCnvp_+3A_del_gene">del_gene</code></td>
<td>
<p>A vector of strings, the IDs of deleted genes.</p>
</td></tr>
<tr><td><code id="getCnvp_+3A_p.adjust">p.adjust</code></td>
<td>
<p>Logical,tell if returns corrected p-values</p>
</td></tr>
<tr><td><code id="getCnvp_+3A_method">method</code></td>
<td>
<p>Correction method,which can be one of &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,</p>
</td></tr>
</table>


<h3>Details</h3>

<p>cnv_data is TCGA level4 data.if p.adjust=TRUE,return corrected p-values,if p.adjust=FALSE,return p-values
</p>


<h3>Value</h3>

<p>A numeric vector of p-values or corrected p-values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>exp_data&lt;-GetExampleData("exp_data")
meth_data&lt;-GetExampleData("meth_data")
cnv_data&lt;-GetExampleData("cnv_data")
amp_gene&lt;-GetExampleData("amp_gene")
del_gene&lt;-GetExampleData(("del_gene"))
getCnvp(exp_data,cnv_data,amp_gene,del_gene,p.adjust=FALSE,method="fdr")
</code></pre>

<hr>
<h2 id='GetExampleData'>Get the example data</h2><span id='topic+GetExampleData'></span>

<h3>Description</h3>

<p>Get the example data of test package for litte trials.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetExampleData(exampleData)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetExampleData_+3A_exampledata">exampleData</code></td>
<td>
<p>A character, should be one of &quot;exp_data&quot;, &quot;meth_data&quot;, &quot;cnv_data&quot;, &quot;amp_gene&quot;, &quot;del_gene&quot; ,&quot;label1&quot;,&quot;label2&quot;,&quot;zz&quot;,&quot;exp.p&quot;,&quot;meth.p&quot;,&quot;cnv.p&quot;and &quot;pathdata&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function getExampleData(ExampleData = &quot;exp.p)&quot;) obtains a vector of lncRNAs confirmed to be related with breast cancer. The function getExampleData(ExampleData = &quot;Profile&quot;) obtains the expression pr
</p>


<h3>References</h3>

<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>

<hr>
<h2 id='getExpp'>getExpp</h2><span id='topic+getExpp'></span>

<h3>Description</h3>

<p>'getExpp' perform t-test on Expression profile data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExpp(exp_data, label, p.adjust = TRUE, method = "fdr")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getExpp_+3A_exp_data">exp_data</code></td>
<td>
<p>A data frame, the expression profile to calculate p-value for each gene, the rownames should be the symbol of genes.</p>
</td></tr>
<tr><td><code id="getExpp_+3A_label">label</code></td>
<td>
<p>A vector of 0/1s, indicating the class of samples in the expression profile, 0 represents case, 1 represents control.</p>
</td></tr>
<tr><td><code id="getExpp_+3A_p.adjust">p.adjust</code></td>
<td>
<p>Logical,tell if returns corrected p-values</p>
</td></tr>
<tr><td><code id="getExpp_+3A_method">method</code></td>
<td>
<p>Correction method,which can be one of &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given expression profile of two conditions, ICDS package provide t-test method to calculate p-values or corrected p-values(if p.adjust=TRUE,return corrected p-values,if p.adjust=FALSE,return p-values.) for each genes. The row of the expression profile should be gene symbols and the column of the expression profile should be names of samples. Samples should be under two conditions and the label should be given as 0 and 1.
</p>


<h3>Value</h3>

<p>A numeric vector of p-values or corrected p-values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>profile&lt;-GetExampleData("exp_data")
label&lt;-GetExampleData("label1")
getExpp(profile,label,p.adjust=FALSE)
</code></pre>

<hr>
<h2 id='getMethp'>getMethp</h2><span id='topic+getMethp'></span>

<h3>Description</h3>

<p>'getMethp' perform t-test on Methylation profile data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getMethp(meth_data, label, p.adjust = TRUE, method = "fdr")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getMethp_+3A_meth_data">meth_data</code></td>
<td>
<p>A data frame, the Methylation profile to calculate p-value for each gene, the rownames should be the symbol of genes.</p>
</td></tr>
<tr><td><code id="getMethp_+3A_label">label</code></td>
<td>
<p>label A vector of 0/1s, indicating the class of samples in the Methylation profile, 0 represents case, 1 represents control.</p>
</td></tr>
<tr><td><code id="getMethp_+3A_p.adjust">p.adjust</code></td>
<td>
<p>Logical,tell if returns corrected p-values</p>
</td></tr>
<tr><td><code id="getMethp_+3A_method">method</code></td>
<td>
<p>Correction method,which can be one of &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;, &quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;,</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given Methylation profile of two conditions, ICDS package provide t-test method to calculate p-values or corrected p-values(if p.adjust=TRUE,return corrected p-values,if p.adjust=FALSE,return p-values.) for each genes. The row of the Methylation profile should be gene symbols and the column of the Methylation profile should be names of samples. Samples should be under two conditions and the label should be given as 0 and 1.
</p>


<h3>Value</h3>

<p>A numeric vector of p-values or corrected p-values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>profile&lt;-GetExampleData("meth_data")
label&lt;-GetExampleData("label2")
getMethp(profile,label,p.adjust=FALSE)
</code></pre>

<hr>
<h2 id='opt_subpath'>opt_subpath</h2><span id='topic+opt_subpath'></span>

<h3>Description</h3>

<p>'opt_subpath' Optimize interested subpathways.If the number of genes shared by the two pathways accounted for more than the Overlap ratio of each pathway genes,then combine two pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opt_subpath(subpathdata, zz, overlap = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="opt_subpath_+3A_subpathdata">subpathdata</code></td>
<td>
<p>interested subpathways</p>
</td></tr>
<tr><td><code id="opt_subpath_+3A_zz">zz</code></td>
<td>
<p>a vector of z-scores</p>
</td></tr>
<tr><td><code id="opt_subpath_+3A_overlap">overlap</code></td>
<td>
<p>Overlap ratio of each two pathway genes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Optimized subpathway:the number of genes shared by any two pathways accounted for less than the Overlap ratio of each pathway genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>zz&lt;-GetExampleData("zzz")
subpathdata&lt;-GetExampleData("subpathdata")
optsubpath&lt;-opt_subpath(subpathdata,zz,overlap=0.6)
</code></pre>

<hr>
<h2 id='Permutation'>Permutation</h2><span id='topic+Permutation'></span>

<h3>Description</h3>

<p>the permutation test method 1 and method 2 were used to calculate the statistical significance level for these optimal subpathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Permutation(
  subpathwayz,
  zz,
  nperm1 = 1000,
  method1 = TRUE,
  nperm2 = 1000,
  method2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Permutation_+3A_subpathwayz">subpathwayz</code></td>
<td>
<p>Optimize intersted subpathways</p>
</td></tr>
<tr><td><code id="Permutation_+3A_zz">zz</code></td>
<td>
<p>a vector of z-scores</p>
</td></tr>
<tr><td><code id="Permutation_+3A_nperm1">nperm1</code></td>
<td>
<p>times of permutation to perform use method1</p>
</td></tr>
<tr><td><code id="Permutation_+3A_method1">method1</code></td>
<td>
<p>permutation analysis method1</p>
</td></tr>
<tr><td><code id="Permutation_+3A_nperm2">nperm2</code></td>
<td>
<p>times of permutation to perform use method2</p>
</td></tr>
<tr><td><code id="Permutation_+3A_method2">method2</code></td>
<td>
<p>permutation analysis method2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the statistical significance p value and FDR for these optimal subpathways
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(graphite)
keysubpathways&lt;-GetExampleData("keysubpathways")
zzz&lt;-GetExampleData("zzz")
Permutation(keysubpathways,zzz,nperm1=10,method1=TRUE,nperm2=10,method2=FALSE)
</code></pre>

<hr>
<h2 id='PlotSubpathway'>PlotSubpathway</h2><span id='topic+PlotSubpathway'></span>

<h3>Description</h3>

<p>PlotSubpathway:plot a network graph when user input a list of gene
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotSubpathway(
  subpID,
  pathway.name,
  zz,
  Pathway = "kegg",
  layout = layout.fruchterman.reingold
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PlotSubpathway_+3A_subpid">subpID</code></td>
<td>
<p>gene list of a interested subpathway</p>
</td></tr>
<tr><td><code id="PlotSubpathway_+3A_pathway.name">pathway.name</code></td>
<td>
<p>name of the interested subpathway</p>
</td></tr>
<tr><td><code id="PlotSubpathway_+3A_zz">zz</code></td>
<td>
<p>z-score of each gene</p>
</td></tr>
<tr><td><code id="PlotSubpathway_+3A_pathway">Pathway</code></td>
<td>
<p>the name of the pathway database</p>
</td></tr>
<tr><td><code id="PlotSubpathway_+3A_layout">layout</code></td>
<td>
<p>The layout specification(<code><a href="igraph.html#topic+layout_">layout_</a></code>). It must be a call to a layout specification function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Network graph
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(graphite)

subpID&lt;-unlist(strsplit("ACSS1/ALDH3B2/ADH1B/ADH1A/ALDH2/DLAT/ACSS2","/"))
pathway.name="Glycolysis / Gluconeogenesis"
zzz&lt;- GetExampleData("zzz")
PlotSubpathway(subpID=subpID,pathway.name=pathway.name,zz=zzz)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
