<!DOCTYPE html><html lang="en"><head><title>Help for package ICGE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ICGE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#chowdary'>
<p>Chowdary Database</p></a></li>
<li><a href='#dbhatta'><p> Bhattacharyya Distance</p></a></li>
<li><a href='#dcor'><p> Correlation Distance</p></a></li>
<li><a href='#deltas'><p> Distance Between Groups</p></a></li>
<li><a href='#dermatology'><p>  Dermatology Database</p></a></li>
<li><a href='#dgower'><p> Gower Distance for Mixed Variables</p></a></li>
<li><a href='#dmahal'><p> Mahalanobis Distance</p></a></li>
<li><a href='#dproc2'><p> Modified Procrustes distance</p></a></li>
<li><a href='#estW'><p> INCA Statistic</p></a></li>
<li><a href='#INCAindex'><p> INCA index</p></a></li>
<li><a href='#INCAnumclu'><p>  Estimation of Number of Clusters in Data</p></a></li>
<li><a href='#INCAtest'><p> INCA Test</p></a></li>
<li><a href='#lympha'>
<p>Limphatic Database</p></a></li>
<li><a href='#proxi'><p> Proximity Function</p></a></li>
<li><a href='#SyntheticTimeCourse'>
<p>Synthetic Time Course data</p></a></li>
<li><a href='#vgeo'><p> Geometric Variability</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-10-14</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimation of Number of Clusters and Identification of Atypical
Units</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Itziar Irigoien &lt;itziar.irigoien@ehu.eus&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.0.1), utils, stats, MASS, cluster, fastcluster</td>
</tr>
<tr>
<td>Description:</td>
<td>It is a package that helps to estimate the number of real clusters in data as well as to identify atypical units. The underlying methods are based on distances rather than on unit x variables.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-17 08:36:47 UTC; itziar</td>
</tr>
<tr>
<td>Author:</td>
<td>Itziar Irigoien [aut, cre],
  Concepcion Arenas [aut]</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-17 09:25:23 UTC</td>
</tr>
</table>
<hr>
<h2 id='chowdary'>
Chowdary Database
</h2><span id='topic+chowdary'></span>

<h3>Description</h3>

<p>The original authors compared pairs of snap-frozen and RNAlater
preservative-suspended tissue from lymph node-negative breast tumors (B)
and Dukes' B colon tumors (C). The actual data set, by de Souto et. al
(2008), is build with purpose of separating B from C.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(chowdary)</code></pre>


<h3>Format</h3>

<p>Data frame with 183 rows and  104 columns.
</p>


<h3>Source</h3>

<p>Original source from &lsquo;National Center for Biotechnology Information&rsquo; from the United States of America, query GSE3726.
</p>


<h3>References</h3>

<p>de Souto MCP,  Costa IG, de  Araujo DSA, Ludermir TB, and  Schliep A
(2008). Clustering Cancer Gene Expression Data: a Comparative
Study. <em>BMC Bioinformatics</em>, <b>8</b>, 497&ndash;511.
</p>
<p>Chowdary D, Lathrop J, Skelton J, Curtin K, Briggs T, Zhang Y, Yu J,
Wang X, and  Mazumder A (2006). Prognostic gene expression signatures
can be measured in tissues collected in RNAlater
preservative. <em>Journal Molecular Diagnosis</em>, <b>8</b>, 31&ndash;39.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chowdary)

tumor &lt;- as.factor(as.matrix(chowdary[1,]))
x &lt;- as.matrix(chowdary[-1,])
mode(x) &lt;- "numeric"

s &lt;- sample(row.names(x),1)
boxplot( x[s,] ~ tumor , ylab=s)
</code></pre>

<hr>
<h2 id='dbhatta'> Bhattacharyya Distance </h2><span id='topic+dbhatta'></span>

<h3>Description</h3>

<p><code>dbhatta</code>  computes and returns the Bhattacharyya distance matrix between the rows of a data matrix. This distance is defined between two units <code class="reqn">i=(p_{i1},...,p_{im})</code> and <code class="reqn">j=(p_{j1},...,p_{jm})</code> being <code class="reqn">p_{kl}</code> frequencies with <code class="reqn">p_{kl}&gt;=0</code> and <code class="reqn">p_{k1}+...+p_{km}=1</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dbhatta(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dbhatta_+3A_x">x</code></td>
<td>
<p> a matrix containing, in its rows, the frequencies for each unit. Note: check that each row adds up to 1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object with distance information.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV-EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Bhattacharyya, A. (1946). On a measure of divergence of two multinomial populations.
<em>Sankhya: The Indian Journal of Statistics, Series A</em>. <b>14</b>, 177-136.</p>


<h3>See Also</h3>

  <p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+dmahal">dmahal</a></code>,
<code><a href="#topic+dgower">dgower</a></code>, <code><a href="#topic+dcor">dcor</a></code>, <code><a href="#topic+dproc2">dproc2</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#5 individuals represented by their relative frequencies of 4 characteristics (M1-M4):
f &lt;- matrix(c(0.36, 0.21, 0.23, 0.20,
              0.66, 0.18, 0.11, 0.05,
              0.01, 0.24, 0.62, 0.13,
              0.43, 0.38, 0.08, 0.11,
              0.16, 0.07, 0.09, 0.68), 
              byrow=TRUE, nrow=5, dimnames=list(1:5, paste("M", 1:4, sep="")))

# Bhattacharyya distances between pairs 
d &lt;- dbhatta(f)
</code></pre>

<hr>
<h2 id='dcor'> Correlation Distance </h2><span id='topic+dcor'></span>

<h3>Description</h3>

<p><code>dcor</code> computes and returns the Correlation distance matrix between the rows of a data matrix. This distance is defined by <code class="reqn">d=\sqrt{1-r}</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dcor(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dcor_+3A_x">x</code></td>
<td>
<p> a  numeric matrix. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object with distance information.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV-EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Gower, J.C. (1985). Measures of similarity, dissimilarity and distance. 
In: <em>Encyclopedia of Statistical Sciences</em>, volume <b>5</b>, 397&ndash;405. J. Wiley and Sons.</p>


<h3>See Also</h3>

  <p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+dmahal">dmahal</a></code>,
<code><a href="#topic+dgower">dgower</a></code>, <code><a href="#topic+dbhatta">dbhatta</a></code>, <code><a href="#topic+dproc2">dproc2</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 10 objects in dimension 8
n &lt;- 10
mu &lt;- sample(1:10, 8, replace=TRUE)
x &lt;- matrix(rnorm(n*8, mean=mu, sd=1), nrow=n, byrow=TRUE)


# Correlation distances between pairs 
d &lt;- dcor(x)
</code></pre>

<hr>
<h2 id='deltas'> Distance Between Groups</h2><span id='topic+deltas'></span>

<h3>Description</h3>

<p>Assume that n units are divided into k groups C1,...,Ck . Function <code>deltas</code>  computes and returns the distance between each pair of groups. It uses the distances between pairs of units. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deltas(d, pert = "onegroup")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="deltas_+3A_d">d</code></td>
<td>
<p> a distance matrix or a <code>dist</code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="deltas_+3A_pert">pert</code></td>
<td>
<p> an n-vector that indicates which group each unit belongs to. Note that the expected values of <code>pert</code> are numbers greater than or equal to 1 (for instance 1,2,3,4..., k). The default value indicates there is only one group in data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing the distances between each pair of groups.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>
<p>Cuadras, C.M., Fortiana, J. and Oliva, F. (1997). The proximity of an individual to a population with applications in
discriminant analysis.
<em> Journal of Classification</em>, <b>14</b>, 117&ndash;136.</p>


<h3>See Also</h3>

<p><code><a href="#topic+vgeo">vgeo</a></code>, <code><a href="#topic+proxi">proxi</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
d &lt;- dist(iris[,1:4])
deltas(d,iris[,5])
</code></pre>

<hr>
<h2 id='dermatology'>  Dermatology Database </h2><span id='topic+dermatology'></span>

<h3>Description</h3>

<p>Data from a dermatology study provided by H.A. Guvenir (Dpt. Computer Engineering and Information Science, Bilkent University, Turkey).The data set contains 366 instances presenting 34 different clinical attributes (12 clinical features as age or family history and 22 histopathological features obtained from a biopsy), and a class variable indicating the disease.
There are 8 missing values. 
This data set has been used extensively for classification tasks. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dermatology)</code></pre>


<h3>Format</h3>

<p>Matrix with 366 rows.
</p>


<h3>Details</h3>

<p>Attribute information obtained from the  UCI KDD data repository:
</p>
<p>Clinical Attributes: (they take values 0, 1, 2, 3, unless otherwise
indicated)
</p>
<p>1: erythema;
2: scaling;
3: definite borders;
4: itching;
5: koebner phenomenon;
6: polygonal papules;
7: follicular papules;
8: oral mucosal involvement;
9: knee and elbow involvement;
10: scalp involvement;
11: family history, (0 or 1);
34: Age.
</p>
<p>Histopathological Attributes: (they take values 0, 1, 2, 3)
</p>
<p>12: melanin incontinence;
13: eosinophils in the infiltrate;
14: PNL infiltrate;
15: fibrosis of the papillary dermis;
16: exocytosis;
17: acanthosis;
18: hyperkeratosis;
19: parakeratosis;
20: clubbing of the rete ridges;
21: elongation of the rete ridges;
22: thinning of the suprapapillary epidermis;
23: spongiform pustule;
24: munro microabcess;
25: focal hypergranulosis;
26: disappearance of the granular layer;
27: vacuolisation and damage of basal layer;
28: spongiosis;
29: saw-tooth appearance of retes;
30: follicular horn plug;
31: perifollicular parakeratosis;
32: inflammatory monoluclear inflitrate;
33: band-like infiltrate.
</p>
<p>The considered diseases are: 1 - psoriasis, 2 - seboreic dermatitis, 3- lichen planus, 4 - pityriasis rosea, 5 - chronic dermatitis, 6 - pityriasis rubra pilaris. 
</p>


<h3>Source</h3>

<p>The UCI KDD Archive.
</p>


<h3>References</h3>

<p>Guvenir H, Demiroz G, Ilter N (1998). Learning differential
diagnosis of erythemato-squamous diseases using voting feature
intervals. <em>Artificial Intelligence in Medicine</em>, <b>13</b>,
147&ndash;165.
</p>
<p>Irigoien I, Arenas C (2008). INCA: New statistic for estimating the
number of clusters and identifying atypical units. <em>Statistics
in Medicine</em>, <b>27</b>, 2948&ndash;2973.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dermatology)
x &lt;- dermatology[, 1:34]
group &lt;- as.factor(dermatology[,35])

plot(group)
</code></pre>

<hr>
<h2 id='dgower'> Gower Distance for Mixed Variables</h2><span id='topic+dgower'></span>

<h3>Description</h3>

<p><code>dgower</code> computes and returns the Gower distance matrix for mixed variables. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dgower(x, type = list())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dgower_+3A_x">x</code></td>
<td>
<p> data matrix.</p>
</td></tr>
<tr><td><code id="dgower_+3A_type">type</code></td>
<td>
<p> it is a list with components <code>cuant</code>, <code>bin</code>, <code>nom</code>. Each component
indicates the column position of the quantitative, binary or nominal variables, respectively.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distance  between two
pairs of objects i and j is obtained as
<code class="reqn">\sqrt{2(1-s_{ij})}</code> where <code class="reqn">s_{ij}</code> is the Gower's similarity coefficient for mixed data. This function allows
to include missing values (as <code>NA</code>) and therefore calculates distances based on  Gower's weighted similarity coefficient.
</p>


<h3>Value</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object with distance information.
</p>


<h3>Note</h3>

<p>There is the function <code>daisy()</code> in <code>cluster</code> package which can perform the Gower distance for mixed variables. The difference is that in <code>daisy()</code> the distance is calculated as <code class="reqn">d(i,j)=1-s_{ij}</code> and in  <code>dgower()</code> it is calculated as <code class="reqn">dij=sqrt(1-s_{ij})</code>. 
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Gower, J.C. (1971). A general coefficient of similarity and some of its properties. <em>Biometrics</em>, <b>27</b>, 857&ndash;871.</p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+dmahal">dmahal</a></code>,
<code><a href="#topic+dbhatta">dbhatta</a></code>, <code><a href="#topic+dcor">dcor</a></code>, <code><a href="#topic+dproc2">dproc2</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 10 objects in dimension 6
# Quantitative variables
mu &lt;- sample(1:10, 2, replace=TRUE)
xc &lt;- matrix(rnorm(10*2, mean = mu, sd = 1), ncol=2, byrow=TRUE)

# Binary variables
xb &lt;- cbind(rbinom(10, 1, 0.1), rbinom(10, 1, 0.5), rbinom(10, 1, 0.9))

# Nominal variables
xn &lt;- matrix(sample(1:3, 10, replace=TRUE), ncol=1)

x &lt;- cbind(xc, xb, xn)

# Distances
d &lt;- dgower(x, type=list(cuant=1:2, bin=3:5, nom=6))


</code></pre>

<hr>
<h2 id='dmahal'> Mahalanobis Distance</h2><span id='topic+dmahal'></span>

<h3>Description</h3>

<p><code>dmahal</code> computes and returns the Mahalanobis distance matrix between the rows of a data matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmahal(datos, S)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dmahal_+3A_datos">datos</code></td>
<td>
<p> data matrix.</p>
</td></tr>
<tr><td><code id="dmahal_+3A_s">S</code></td>
<td>
<p> covariance matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object with distance information.
</p>


<h3>Note</h3>

<p>There is a function <code>mahalanobis()</code> in stats package which can perform the Mahalanobis distance. While <code>mahalanobis()</code> calculates the Mahalanobis distance with respect  to given a center, function  <code>dmahal()</code> is designed to calculate the distance between each pair of units given a data matrix.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

 
<p>Everitt B. S. and Dunn G. (2001) <em> Applied Multivariate Data Analysis</em>. 2 edition, Edward Arnold, London.
</p>


<h3>See Also</h3>

 <p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+dbhatta">dbhatta</a></code>,
<code><a href="#topic+dgower">dgower</a></code>, <code><a href="#topic+dcor">dcor</a></code>, <code><a href="#topic+dproc2">dproc2</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#Generate 10 objects in dimension 2
mu &lt;- rep(0, 2)
Sigma &lt;- matrix(c(10,3,3,2),2,2)

x &lt;- mvrnorm(n=10, rep(0, 2), Sigma)

d &lt;- dmahal(x, Sigma)
</code></pre>

<hr>
<h2 id='dproc2'> Modified Procrustes distance</h2><span id='topic+dproc2'></span>

<h3>Description</h3>

<p><code>dproc2</code> computes and returns all the pairwise procrustes distances between genes in a time course experiment, using their expression profile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dproc2(x, timepoints = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dproc2_+3A_x">x</code></td>
<td>
<p> a  matrix containing, in its rows, the gene expression values at the T considered time points.</p>
</td></tr>
<tr><td><code id="dproc2_+3A_timepoints">timepoints</code></td>
<td>
<p>a T-vector with the T observed time points. If  <code>timepoints=NULL</code> (default), then timepoints=1:T.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each row i of matrix x is arranged in a two column matrix Xi. In Xi, the first column contains the time points and the second column the observed gene expression values (xi1...). 
</p>


<h3>Value</h3>

<p>A <code><a href="stats.html#topic+dist">dist</a></code> object with distance information.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p>Irigoien, I. , Vives, S. and Arenas, C. (2011). Microarray
Time Course Experiments: Finding Profiles. <em>IEEE/ACM Transactions
on Computational Biology and Bioinformatics</em>, <b>8</b>(2), 464&ndash;475.
</p>
<p>Gower, J. C. and Dijksterhuis, G. B. (2004)
<em>Procrustes Problems</em>. Oxford University Press.
</p>
<p>Sibson, R. (1978). Studies in the Robustness of Multidimensional
Scaling: Procrustes statistic. <em>Journal of the Royal Statistical
Society, Series B</em>, <b>40</b>, 234&ndash;238.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+dist">dist</a></code>, <code><a href="#topic+dmahal">dmahal</a></code>, <code><a href="#topic+dgower">dgower</a></code>, <code><a href="#topic+dcor">dcor</a></code> <code><a href="#topic+dbhatta">dbhatta</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'># Given  10  hypothetical time course profiles
# over 6 time points at 1, 2, ..., 6 hours.
x &lt;- matrix(c(0.38, 0.39, 0.38, 0.37, 0.385, 0.375,
              0.99, 1.19, 1.50, 1.83, 2.140, 2.770,
              0.38, 0.50, 0.71, 0.72, 0.980, 1.010,
              0.20, 0.40, 0.70, 1.06, 2.000, 2.500,
              0.90, 0.95, 0.97, 1.50, 2.500, 2.990,
              0.64, 2.61, 1.51, 1.34, 1.330 ,1.140,
              0.71, 1.82, 2.28, 1.72, 1.490, 1.060,
              0.71, 1.82, 2.28, 1.99, 1.975, 1.965,
              0.49, 0.78, 1.00, 1.27, 0.590, 0.340,
              0.71,1.00, 1.50, 1.75, 2.090, 1.380), nrow=10, byrow=TRUE)

# Graphical representation
matplot(t(x), type="b")

# Distance matrix between them
d &lt;- dproc2(x)

</code></pre>

<hr>
<h2 id='estW'> INCA Statistic</h2><span id='topic+estW'></span><span id='topic+print.incaest'></span>

<h3>Description</h3>

<p>Assume that n units are divided into k clusters C1,...,Ck, and consider a fixed unit x0. Function <code>estW</code> calculates the INCA statistic <code class="reqn">W(x0)</code> and the related <code class="reqn">U_i</code> statistics.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estW(d, dx0, pert = "onegroup")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estW_+3A_d">d</code></td>
<td>
<p> a distance matrix or a <code>dist</code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="estW_+3A_dx0">dx0</code></td>
<td>
<p> an n-vector containing the distances d0j between x0 and unit j.</p>
</td></tr>
<tr><td><code id="estW_+3A_pert">pert</code></td>
<td>
<p> an n-vector that indicates which group each unit belongs to. Note that the expected values of <code>pert</code> are consecutive integers bigger or equal than 1 (for instance 1,2,3,4..., k). The default value indicates the presence of only one group in data. </p>
</td></tr>
</table>


<h3>Value</h3>

<p> The function returns an object of class <code>incaest</code> which is a list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>Wvalue</code></td>
<td>
<p>is the INCA statistic <code class="reqn">W(x_0)</code>.</p>
</td></tr>
<tr><td><code>Uvalue</code></td>
<td>
<p>is a vector containing the statistics <code class="reqn">U_i</code>. </p>
</td></tr>
</table>


<h3>Note</h3>

<p> For a correct geometrical interpretation it is convenient to verify whether the distance matrix d is Euclidean.
</p>


<h3>Author(s)</h3>

<p>Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p>Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>
<p>Irigoien, I. and  Arenas, C. (2008). INCA: New statistic for estimating the number of clusters and identifying atypical units.
<em>Statistics in Medicine</em>, <b>27</b>(15), 2948&ndash;2973.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+vgeo">vgeo</a></code>, <code><a href="#topic+proxi">proxi</a></code> , <code><a href="#topic+deltas">deltas</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
d &lt;- dist(iris[,1:4])

# characteristics of a specific flower (likely group 1)
x0 &lt;- c(5.3, 3.6, 1.1, 0.1) 
# distances between  flower x0 and the rest of flowers in iris
dx0 &lt;- rep(0,150)
for (i in 1:150){
	dif &lt;-x0-iris[i,1:4]
	dx0[i] &lt;- sqrt(sum(dif*dif))
}
estW(d, dx0, iris[,5])

</code></pre>

<hr>
<h2 id='INCAindex'> INCA index</h2><span id='topic+INCAindex'></span><span id='topic+print.incaix'></span><span id='topic+summary.incaix'></span><span id='topic+plot.incaix'></span>

<h3>Description</h3>

<p><code>INCAindex</code> helps to estimate the number of clusters in a dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>INCAindex(d, pert_clus)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="INCAindex_+3A_d">d</code></td>
<td>
<p> a distance matrix or a  <code>dist</code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="INCAindex_+3A_pert_clus">pert_clus</code></td>
<td>
<p> an n-vector that indicates which group each unit belongs to. Note that the expected values of <code>pert</code> are numbers  greater than or equal to 1 (for instance 1,2,3,4..., k). The default value indicates the presence of only one group in data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>incaix</code> which is a list containing the following components:
</p>
<table role = "presentation">
<tr><td><code>well_class</code></td>
<td>
<p>a vector indicating the number of well classified units.</p>
</td></tr>
<tr><td><code>Ni_cluster</code></td>
<td>
<p>a vector indicating each cluster size.</p>
</td></tr>
<tr><td><code>Total</code></td>
<td>
<p>percentage of objects well classified in the partition defined by <code>pert_clus</code>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For a correct geometrical interpretation it is convenient to verify whether the distance matrix d is Euclidean.
It admits the associated methods summary and plot. The first simply returns the percentage of well-classified units and the second offers a barchart with the percentages of well classified units for each group in the given partition.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>
<p>Irigoien, I. and  Arenas, C. (2008). INCA: New statistic for estimating the number of clusters and identifying atypical units.
<em>Statistics in Medicine</em>, <b>27</b>(15), 2948&ndash;2973.</p>


<h3>See Also</h3>

 <p><code><a href="#topic+estW">estW</a></code>, <code><a href="#topic+INCAtest">INCAtest</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#generate 3 clusters, each of them with 20 objects in dimension 5.
mu1 &lt;- sample(1:10, 5, replace=TRUE)
x1 &lt;- matrix(rnorm(20*5, mean = mu1, sd = 1),ncol=5, byrow=TRUE)
mu2 &lt;- sample(1:10, 5, replace=TRUE)
x2 &lt;- matrix(rnorm(20*5, mean = mu2, sd = 1),ncol=5, byrow=TRUE)
mu3 &lt;- sample(1:10, 5, replace=TRUE)
x3 &lt;- matrix(rnorm(20*5, mean = mu3, sd = 1),ncol=5, byrow=TRUE)
x &lt;- rbind(x1,x2,x3)

# Euclidean distance between units.
d &lt;- dist(x)

# given the right partition, calculate the percentage of well classified objects.
partition &lt;- c(rep(1,20), rep(2,20), rep(3,20))
INCAindex(d, partition)


# In order to estimate the number of cluster in data, try several 
#  partitions and compare the results
library(cluster)
T &lt;- rep(NA, 5)
for (l in 2:5){
	part &lt;- pam(d,l)$clustering
	T[l] &lt;- INCAindex(d,part)$Total
}

plot(T, type="b",xlab="Number of clusters", ylab="INCA", xlim=c(1.5, 5.5))
</code></pre>

<hr>
<h2 id='INCAnumclu'>  Estimation of Number of Clusters in Data </h2><span id='topic+INCAnumclu'></span><span id='topic+plot.incanc'></span><span id='topic+print.incanc'></span>

<h3>Description</h3>

<p><code>INCAnumclu</code> helps to estimate the number of clusters in a
dataset. The INCA index associated to different partitions with
different number of clusters is calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>INCAnumclu(d, K, method = "pam", pert, L= NULL, noise=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="INCAnumclu_+3A_d">d</code></td>
<td>
<p> a distance matrix or a <code><a href="stats.html#topic+dist">dist</a></code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="INCAnumclu_+3A_k">K</code></td>
<td>
<p> the maximum number of cluster to be considered. For each k value ( k=2,..,K)  a partition with k clusters is calculated. </p>
</td></tr>
<tr><td><code id="INCAnumclu_+3A_method">method</code></td>
<td>
<p> character string defining the clustering method in
order to obtain the partitions. The hierarchical aglomerative clustering methods are perfomed via <code><a href="stats.html#topic+hclust">hclust</a></code> function in package <span class="pkg">fastcluster</span>. Other clustering methods are performed  via the functions in package <span class="pkg">cluster</span>, such as: <code><a href="cluster.html#topic+pam">pam</a></code>, 
<code><a href="cluster.html#topic+diana">diana</a></code> and  <code><a href="cluster.html#topic+fanny">fanny</a></code>. The available clustering methods are <code>pam</code> (default method), <code>average</code> (UPGMA), <code>single</code> (single linkage),
<code>complete</code> (complete linkage), <code>ward.D2</code> (Ward's method), <code>ward.D</code>, <code>centroid</code>, <code>median</code>, <code>diana</code> (hierarchical
divisive) and <code>fanny</code> (fuzzy clustering). Nevertheless, the user can introduce
particular or custom  partitions indicating <code>method="partition"</code> and specifying
the partitions in argument <code>pert</code>.</p>
</td></tr>
<tr><td><code id="INCAnumclu_+3A_pert">pert</code></td>
<td>
<p>only useful when parameter <code>method</code>=&quot;partition&quot;; it
is a matrix and each column contains a partition of the units. That means that each column is an n-vector that indicates which group each unit belongs to. Note that the expected values of each column of <code>pert</code> are numbers greater than or equal to 1 (for instance 1,2,3,4..., k). </p>
</td></tr>
<tr><td><code id="INCAnumclu_+3A_l">L</code></td>
<td>
<p> default value NULL, but when some units are considered by
the user as noise units, <code>L</code> must be specified as follows: (a)
<code>L</code> is greater than or equal to 1 and all units in clusters with a
cardinal &lt;= L  are considered noise units; (b) <code>L="custom"</code> when the user wants to specify which units are considered noise units. These units must be specified in argument <code>noise</code>.</p>
</td></tr>
<tr><td><code id="INCAnumclu_+3A_noise">noise</code></td>
<td>
<p>when <code>L="custom"</code>, it is a logical vector indicating
the units considered by the user as noise units.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an object of class <code>incanc</code> which is a numeric vector containing the INCA index associated to each of the k (k=2,...,K) partitions. When <code>noise</code> is no null, the function returns a list with the INCA index for each partition, which is calculated without noise units as well as with noise units. The associated <code>plot</code> returns INCA index plot, both, with and without noise.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Irigoien, I. and  Arenas, C. (2008). INCA: New statistic for estimating the number of clusters and identifying atypical units.
<em>Statistics in Medicine</em>, <b>27</b>(15), 2948&ndash;2973.
</p>
<p>Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>


<h3>See Also</h3>

  <p><code><a href="#topic+INCAindex">INCAindex</a></code>, <code><a href="#topic+estW">estW</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#------- Example 1 --------------------------------------
#generate 3 clusters, each of them with 20 objects in dimension 5.
mu1 &lt;- sample(1:10, 5, replace=TRUE)
x1 &lt;- matrix(rnorm(20*5, mean = mu1, sd = 1),ncol=5, byrow=TRUE)
mu2 &lt;- sample(1:10, 5, replace=TRUE)
x2 &lt;- matrix(rnorm(20*5, mean = mu2, sd = 1),ncol=5, byrow=TRUE)
mu3 &lt;- sample(1:10, 5, replace=TRUE)
x3 &lt;- matrix(rnorm(20*5, mean = mu3, sd = 1),ncol=5, byrow=TRUE)
x &lt;- rbind(x1,x2,x3)

# calculte euclidean distance between them
d &lt;- dist(x)

# calculate the INCA index associated to partitions with k=2, ..., k=5 clusters.
INCAnumclu(d, K=5)
out &lt;- INCAnumclu(d, K=5)
plot(out)

#------- Example 1 cont. --------------------------------
# With hypothetical noise elements
noiseunits &lt;- rep(FALSE, 60)
noiseunits[sample(1:60, 20)] &lt;- TRUE
out &lt;- INCAnumclu(d, K=5, L="custom", noise=noiseunits)
plot(out)
</code></pre>

<hr>
<h2 id='INCAtest'> INCA Test</h2><span id='topic+INCAtest'></span><span id='topic+print.incat'></span><span id='topic+summary.incat'></span>

<h3>Description</h3>

<p>Assume that n units are divided into k groups C1,...,Ck. Function <code>INCAtest</code> performs the typicality INCA test. Therein, the null hypothesis that a new unit x0 is a typical unit with respect to a previously fixed partition is tested versus the alternative hypothesis that the unit is atypical.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>INCAtest(d, pert, d_test, np = 1000, alpha = 0.05, P = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="INCAtest_+3A_d">d</code></td>
<td>
<p>a distance matrix or a <code>dist</code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="INCAtest_+3A_pert">pert</code></td>
<td>
<p> an n-vector that indicates which group each unit belongs to. Note that the expected values of <code>pert</code> are numbers greater than or equal to 1 (for instance 1,2,3,4..., k). The default value indicates there is only one group in data.</p>
</td></tr>
<tr><td><code id="INCAtest_+3A_d_test">d_test</code></td>
<td>
<p> an n-vector containing the distances from x0 to the other units.</p>
</td></tr>
<tr><td><code id="INCAtest_+3A_np">np</code></td>
<td>
<p> sample size for the bootstrap sample for the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="INCAtest_+3A_alpha">alpha</code></td>
<td>
<p> fixed level for the test.</p>
</td></tr>
<tr><td><code id="INCAtest_+3A_p">P</code></td>
<td>
<p> Number of times the bootstrap procedure is repeated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with class &quot;incat&quot; containing the following components:
</p>
<table role = "presentation">
<tr><td><code>StatisticW0</code></td>
<td>
<p>value of the INCA statistic.</p>
</td></tr>
<tr><td><code>ProjectionsU</code></td>
<td>
<p>values of statistics measuring the projection from the specific object to each considered group.</p>
</td></tr>
<tr><td><code>pvalues</code></td>
<td>
<p>p-values obtained in the <code>P</code> times repeated bootstrap procedure. Note: If <code>P</code>&gt;1, it is printed the number of times the p-values were smaller than <code>alpha</code>.</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>specified value of the level of the test.</p>
</td></tr>
</table>


<h3>Note</h3>

 
<p>To obtain the INCA statistic distribution, under the null hypothesis, the program can consume long time.
For a correct geometrical interpretation it is convenient to verify whether the distance matrix d is Euclidean.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.es">itziar.irigoien@ehu.es</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV-EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Irigoien, I. and  Arenas, C. (2008). INCA: New statistic for estimating the number of clusters and identifying atypical units.
<em>Statistics in Medicine</em>, <b>27</b>(15), 2948&ndash;2973.
</p>
<p>Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estW">estW</a></code>, <code><a href="#topic+INCAindex">INCAindex</a></code> </p>


<h3>Examples</h3>

<pre><code class='language-R'>#generate 3 clusters, each of them with 20 objects in dimension 5.
mu1 &lt;- sample(1:10, 5, replace=TRUE)
x1 &lt;- matrix(rnorm(20*5, mean = mu1, sd = 1),ncol=5, byrow=TRUE)
mu2 &lt;- sample(1:10, 5, replace=TRUE)
x2 &lt;- matrix(rnorm(20*5, mean = mu2, sd = 1),ncol=5, byrow=TRUE)
mu3 &lt;- sample(1:10, 5, replace=TRUE)
x3 &lt;- matrix(rnorm(20*5, mean = mu3, sd = 1),ncol=5, byrow=TRUE)
x &lt;- rbind(x1,x2,x3)

# Euclidean distance between units in matrix x.
d &lt;- dist(x)
# given the right partition
partition &lt;- c(rep(1,20), rep(2,20), rep(3,20))

# x0 contains a unit from one group, as for example group 1.
x0 &lt;-  matrix(rnorm(1*5, mean = mu1, sd = 1),ncol=5, byrow=TRUE)

# distances between x0 and the other units.
dx0 &lt;- rep(0,60)
for (i in 1:60){
	dif &lt;-x0-x[i,]
	dx0[i] &lt;- sqrt(sum(dif*dif))
}

INCAtest(d, partition, dx0, np=10)


# x0 contains a unit from a new group.
x0 &lt;-  matrix(rnorm(1*5, mean = sample(1:10, 5, replace=TRUE),
        sd = 1), ncol=5, byrow=TRUE)

# distances between x0 and the other units in matrix x.
dx0 &lt;- rep(0,60)
for (i in 1:60){
	dif &lt;-x0-x[i,]
	dx0[i] &lt;- sqrt(sum(dif*dif))
}

INCAtest(d, partition, dx0, np=10)

</code></pre>

<hr>
<h2 id='lympha'>
Limphatic Database
</h2><span id='topic+lympha'></span>

<h3>Description</h3>

<p>This lymphography domain was obtained from the University Medical Centre,
Institute of Oncology, Ljubljana, Yugoslavia.  Thanks go to M. Zwitter and 
M. Soklic for providing the data. The data are available at the  UCI
KDD data repository (Hettich S and Bay SD, 1999).
</p>
<p>The data set consists of 148 instances presenting 18 different
mixed attributes (1 cuantitative, 9 binaries and 9 nominals), and a class variable indicating the diagnostic.
There are not missing values.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lympha)</code></pre>


<h3>Format</h3>

<p>Data frame with 148 instances and 19 features. 
</p>


<h3>Details</h3>

<p>Attribute information:
</p>
<p>&mdash; NOTE: All attribute values in the database have been entered as
numeric values corresponding to their index in the list
of attribute values for that attribute domain as given
below.
</p>
<p>1. class: normal find, metastases, malign lymph, fibrosis
</p>
<p>2. lymphatics: normal, arched, deformed, displaced
</p>
<p>3. block of affere: no, yes
</p>
<p>4. bl. of lymph. c: no, yes
</p>
<p>5. bl. of lymph. s: no, yes
</p>
<p>6. by pass: no, yes
</p>
<p>7. extravasates: no, yes
</p>
<p>8. regeneration of: no, yes
</p>
<p>9. early uptake in: no, yes
</p>
<p>10. lym.nodes dimin: 0-3
</p>
<p>11. lym.nodes enlar: 1-4
</p>
<p>12. changes in lym.: bean, oval, round
</p>
<p>13. defect in node: no, lacunar, lac. marginal, lac. central
</p>
<p>14. changes in node: no, lacunar, lac. margin, lac. central
</p>
<p>15. changes in stru: no, grainy, drop-like, coarse, diluted, reticular, 
stripped, faint
</p>
<p>16. special forms: no, chalices, vesicles
</p>
<p>17. dislocation of: no, yes
</p>
<p>18. exclusion of no: no, yes
</p>
<p>19. no. of nodes in: 0-9, 10-19, 20-29, 30-39, 40-49, 50-59, 60-69, &gt;=70
</p>


<h3>Source</h3>

<p>The UCI KDD Archive.
</p>


<h3>References</h3>

<p>Hettich S and Bay SD (1999). The UCI KDD Archive. Department of
Information and Computer Science. University of California at Irvine,
Irvine, USA.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(lympha)
aux &lt;- table(lympha[,1])
barplot(aux, names.arg=c("normal", "metastases", "malign lymph", "fibrosis"))
</code></pre>

<hr>
<h2 id='proxi'> Proximity Function</h2><span id='topic+proxi'></span>

<h3>Description</h3>

<p>Assume that n units are divided into k groups C1,...,Ck. The function calculates the proximity function from a specific unit x0 to the groups Cj. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>proxi(d, dx0, pert = "onegroup")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="proxi_+3A_d">d</code></td>
<td>
<p> a distance matrix or a <code>dist</code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="proxi_+3A_dx0">dx0</code></td>
<td>
<p> an n-vector containing the distances from x0 to the other units.</p>
</td></tr>
<tr><td><code id="proxi_+3A_pert">pert</code></td>
<td>
<p> an n-vector that indicates which group each unit belongs to. Note that the expected values of <code>pert</code> are numbers greater than or equal to 1 (for instance 1,2,3,4..., k). The default value indicates there is only one group in data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>k-vector containing the proximity function value from x0 to each group.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>
<p>Cuadras, C.M., Fortiana, J. and Oliva, F. (1997). The proximity of an individual to a population with applications in
discriminant analysis.
<em> Journal of Classification</em>, <b>14</b>, 117&ndash;136. 
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vgeo">vgeo</a></code>, <code><a href="#topic+deltas">deltas</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
d &lt;- dist(iris[,1:4])

# xo contains a unit from one group, as for example group 1.
x0 &lt;- c(5.3, 3.6, 1.1, 0.1) 
# distances between x0 and the other units.
dx0 &lt;- rep(0,150)
for (i in 1:150){
	dif &lt;-x0-iris[i,1:4]
	dx0[i] &lt;- sqrt(sum(dif*dif))
}

proxi(d, dx0, iris[,5])


# xo contains a unit from one group, as for example group 2.
x0 &lt;- c(6.4, 3.0, 4.8, 1.3) 
# distances between x0 and the other units.
dx0 &lt;- rep(0,150)
for (i in 1:150){
	dif &lt;-x0-iris[i,1:4]
	dx0[i] &lt;- sqrt(sum(dif*dif))
}

proxi(d, dx0, iris[,5])
</code></pre>

<hr>
<h2 id='SyntheticTimeCourse'>
Synthetic Time Course data
</h2><span id='topic+SyntheticTimeCourse'></span>

<h3>Description</h3>

<p>Sythetic time course data  where 210 genes profiles along 6 time points
are reported and where the genes are drawn from 8 different populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SyntheticTimeCourse)</code></pre>


<h3>Format</h3>

<p>Data frame with 120 rows and  7 columns.
</p>


<h3>Details</h3>

<p>Attribute information:
Column cl: the class that the gen belongs to.
Columns t1 - t6: gene's expression along the t1, ..., t6 time
points considered.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SyntheticTimeCourse)
x &lt;- SyntheticTimeCourse[, 2:7]
cl &lt;- SyntheticTimeCourse[, 1]
par(mfrow=c(3,3))
for (g in 1:8){ 
   xx &lt;- t(x[cl==g,] )
   yy &lt;- matrix(c(1:6 ), nrow=6, ncol=15, byrow=FALSE)
   matplot(yy,xx,  pch=21, type="b", axes=FALSE,  
        ylim=c(0,3.5), xlim=c(0.5,6.5), xlab="", ylab="", col="black", main=paste("G",g)) 
   abline(h=0)  
   abline(v=0.5) 
   mtext("Time", side=1) 
   mtext("Expression", side=2)  
}
</code></pre>

<hr>
<h2 id='vgeo'> Geometric Variability</h2><span id='topic+vgeo'></span>

<h3>Description</h3>

<p>Assume that n units are divided into k groups C1,...,Ck. The function calculates the geometrical variability for each group in data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vgeo(d, pert = "onegroup")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vgeo_+3A_d">d</code></td>
<td>
<p> a distance matrix or a <code>dist</code> object with distance information between units.</p>
</td></tr>
<tr><td><code id="vgeo_+3A_pert">pert</code></td>
<td>
<p> an n-vector that indicates which group each unit belongs to. Note that the expected values of <code>pert</code> are numbers greater than or equal to 1  (for instance 1,2,3,4..., k). The default value indicates there is only one group in data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It is a matrix containing the geometric variability for each group.
</p>


<h3>Author(s)</h3>

<p>  Itziar Irigoien <a href="mailto:itziar.irigoien@ehu.eus">itziar.irigoien@ehu.eus</a>; Konputazio Zientziak eta Adimen Artifiziala, Euskal Herriko Unibertsitatea (UPV/EHU), Donostia, Spain.
</p>
<p>Conchita Arenas <a href="mailto:carenas@ub.edu">carenas@ub.edu</a>; Departament d'Estadistica, Universitat de Barcelona, Barcelona, Spain.
</p>


<h3>References</h3>

<p> Arenas, C. and Cuadras, C.M. (2002). Some recent statistical methods based on distances.<em> Contributions to Science</em>, <b>2</b>, 183&ndash;191.
</p>
<p>Cuadras, C.M. (1992). Some examples of distance based discrimination. <em>Biometrical Letters</em>, <b>29</b>(1), 3&ndash;20.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+deltas">deltas</a></code>, <code><a href="#topic+proxi">proxi</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(iris)
d &lt;- dist(iris[,1:4])
vgeo(d,iris[,5])


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
