<!DOCTYPE html><html lang="en"><head><title>Help for package idiffomix</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {idiffomix}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_transformation'><p>The function to filter, normalize and transform RNA-Seq and methylation data.</p></a></li>
<li><a href='#gene_chromosome_data'><p>Data containing chromosome information and the genes located on them.</p></a></li>
<li><a href='#gene_expression_data'><p>Gene expression data for patients suffering from breast cancer</p></a></li>
<li><a href='#idiffomix'><p>The idiffomix function</p></a></li>
<li><a href='#methylation_data'><p>Methylation array data for patients suffering from breast cancer</p></a></li>
<li><a href='#plot.idiffomix'><p>Plots for visualizing the idiffomix class object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrated Differential Analysis of Multi Omics Data using a
Joint Mixture Model</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Koyel Majumdar &lt;koyelmajumdar.phdresearch@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A joint mixture model has been developed by Majumdar et al. (2025) &lt;<a href="https://doi.org/10.48550%2FarXiv.2412.17511">doi:10.48550/arXiv.2412.17511</a>&gt; that integrates information from gene expression data and methylation data at the modelling stage to capture their inherent dependency structure, enabling simultaneous identification of differentially methylated cytosine-guanine dinucleotide (CpG) sites and differentially expressed genes. The model leverages a joint likelihood function that accounts for the nested structure in the data, with parameter estimation performed using an expectation-maximisation algorithm.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>foreach, doParallel, parallel, mclust, stats, utils, edgeR,
magrittr, ggplot2, scales, tidyr, dplyr, reshape2, gridExtra,
grid, tidyselect, cowplot</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-10 10:22:28 UTC; koyel</td>
</tr>
<tr>
<td>Author:</td>
<td>Koyel Majumdar [cre, aut],
  Isobel Claire Gorley [aut],
  Thomas Brendan Murphy [aut],
  Florence Jaffrezic [aut],
  Andrea Rau [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-13 17:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_transformation'>The function to filter, normalize and transform RNA-Seq and methylation data.</h2><span id='topic+data_transformation'></span>

<h3>Description</h3>

<p>The raw RNA-Seq and methylation data needs to be filtered, normalized and transformed before applying the idiffomix method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_transformation(seq_data, meth_data, gene_chr, N = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="data_transformation_+3A_seq_data">seq_data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">G \times (N+1)</code> containing raw RNA-Seq data for all G genes and N patients</p>
</td></tr>
<tr><td><code id="data_transformation_+3A_meth_data">meth_data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times (N+2)</code> containing beta methylation values for all $C$ CpG sites and $N$ patients along with the associated genes for each CpG site.</p>
</td></tr>
<tr><td><code id="data_transformation_+3A_gene_chr">gene_chr</code></td>
<td>
<p>A dataframe containing the genes and their corresponding chromosome number.</p>
</td></tr>
<tr><td><code id="data_transformation_+3A_n">N</code></td>
<td>
<p>Number of patients</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RNA-Seq data consisted of raw counts depicting the gene expression levels. To ensure data quality, only genes whose sum of expression counts across both biological conditions was &gt; 5 are retained. The data were normalized to account for differences in library sizes. The normalized count data were used to obtain CPM values which were further log-transformed to obtain log-CPM values. Given the paired design of the motivating setting, the log-fold changes between the tumour and benign samples were calculated for each gene in every patient and used in the subsequent analyses.
For the methylation array data, the beta values at the CpG sites are logit transformed to M-values. Similar to the RNA-Seq data, given the paired design, the difference in M-values between tumour and benign samples were calculated for each CpG site in every patient and used in the subsequent analyses.
</p>


<h3>Value</h3>

<p>The function returns a list with two dataframes containing the transformed gene expression and methylation array data:
</p>

<ul>
<li><p> seq_transformed - A dataframe containing the log-fold change for gene expression data.
</p>
</li>
<li><p> meth_transformed - A dataframe containing the differences in M-values for methylation data.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 4
data_output = data_transformation(seq_data=gene_expression_data,
                                  meth_data=methylation_data,
                                  gene_chr=gene_chromosome_data,
                                  N=N)

</code></pre>

<hr>
<h2 id='gene_chromosome_data'>Data containing chromosome information and the genes located on them.</h2><span id='topic+gene_chromosome_data'></span>

<h3>Description</h3>

<p>A dataset containing the chromosome information of the gene expression and methylation array data to be analysed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_chromosome_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows and 2 columns.
</p>

<ul>
<li><p>CHR: The chromosome containing the gene.
</p>
</li>
<li><p>Gene: The gene located on the chromosome.
</p>
</li></ul>



<h3>Details</h3>

<p>The dataset contains the information of chromosomes 1 AND 2 and the genes located on them.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gene_expression_data">gene_expression_data</a></code>
</p>
<p><code><a href="#topic+methylation_data">methylation_data</a></code>
</p>

<hr>
<h2 id='gene_expression_data'>Gene expression data for patients suffering from breast cancer</h2><span id='topic+gene_expression_data'></span>

<h3>Description</h3>

<p>A dataset containing simulated RNA-Seq data for <code class="reqn">G</code> genes located on chromosomes 1 and 2, from <code class="reqn">R=2</code> sample types, from <code class="reqn">N=4</code> patients. The sample types are assumed to be benign and tumour tissues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gene_expression_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 20 rows and 9 columns. The data contain no missing values.
</p>

<ul>
<li><p>Gene: The gene name.
</p>
</li>
<li><p>Patient1_GX1: Expression values from benign tissue from patient 1.
</p>
</li>
<li><p>Patient2_GX1: Expression values from benign tissue from patient 2.
</p>
</li>
<li><p>Patient3_GX1: Expression values from benign tissue from patient 3.
</p>
</li>
<li><p>Patient4_GX1: Expression values from benign tissue from patient 4.
</p>
</li>
<li><p>Patient1_GX2: Expression values from tumour tissue from patient 1.
</p>
</li>
<li><p>Patient2_GX2: Expression values from tumour tissue from patient 2.
</p>
</li>
<li><p>Patient3_GX2: Expression values from tumour tissue from patient 3.
</p>
</li>
<li><p>Patient4_GX2: Expression values from tumour tissue from patient 4.
</p>
</li></ul>



<h3>Details</h3>

<p>The simulated raw RNA-Seq data for genes located on the chromosomes 1 and 2 needs to be filtered, normalized and transformed before applying idiffomix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gene_chromosome_data">gene_chromosome_data</a></code>
</p>
<p><code><a href="#topic+methylation_data">methylation_data</a></code>
</p>

<hr>
<h2 id='idiffomix'>The idiffomix function</h2><span id='topic+idiffomix'></span>

<h3>Description</h3>

<p>Integrated differential analysis of multi-omics data using a joint mixture model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idiffomix(
  seq_data,
  meth_data,
  gene_chr,
  N,
  K = 3,
  L = 3,
  probs = c(0.1, 0.9),
  parallel_process = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="idiffomix_+3A_seq_data">seq_data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">G \times (N+1)</code> containing log-fold change values for all G genes and N patients</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_meth_data">meth_data</code></td>
<td>
<p>A dataframe of dimension <code class="reqn">C \times (N+2)</code> containing M-value differences between the two biological conditions for all $C$ CpG sites and $N$ patients along with the associated genes for each CpG site.</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_gene_chr">gene_chr</code></td>
<td>
<p>A dataframe containing the genes and their corresponding chromosome number.</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_n">N</code></td>
<td>
<p>Number of patients in the study.</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_k">K</code></td>
<td>
<p>Number of clusters for expression data (default = 3; E-,E0,E+).</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_l">L</code></td>
<td>
<p>Number of clusters for methylation data (default = 3; M-,M0,M+).</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_probs">probs</code></td>
<td>
<p>Quantile probabilities for initialization (default = c(0.1,0.9)).</p>
</td></tr>
<tr><td><code id="idiffomix_+3A_parallel_process">parallel_process</code></td>
<td>
<p>The &quot;TRUE&quot; option results in parallel processing of the models for increased computational efficiency. The default option has been set as &quot;FALSE&quot; due to package testing limitations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a function to fit the joint mixture model to the transformed and filtered gene expression and methylation data.
</p>


<h3>Value</h3>

<p>The function returns an object of the <code><a href="#topic+idiffomix">idiffomix</a></code> class which contains the following values:
</p>

<ul>
<li><p> tau - The proportion of genes belonging to K clusters.
</p>
</li>
<li><p> pi - A matrix containing the probability of a CpG site belonging to cluster <em>l</em>, given its associated associated gene belongs to cluster <em>k</em>.
</p>
</li>
<li><p> mu - The mean for each component of the gene expression data. If there is more than one component, this is a matrix whose <em>k</em>th column is the mean of the <em>k</em>th component of the mixture model.
</p>
</li>
<li><p> sigma - The variance for each component of the gene expression data.
</p>
</li>
<li><p> lambda - The mean for each component of the methylation data. If there is more than one component, this is a matrix whose <em>l</em>th column is the mean of the <em>l</em>th component of the mixture model.
</p>
</li>
<li><p> rho - The variance for each component of the methylation data.
</p>
</li>
<li><p> N - The number of patients analysed using the beta mixture models.
</p>
</li>
<li><p> R - The number of sample types analysed using the beta mixture models.
</p>
</li>
<li><p> U - A dataframe containing the posterior probabilities of genes belonging to the <em>K</em> clusters.
</p>
</li>
<li><p> V - A dataframe containing the posterior probabilities of CpG sites belonging to the <em>L</em> clusters.
</p>
</li>
<li><p> seq_classification - A dataframe containing the log-fold change for gene expression data and their classification corresponding to U.
</p>
</li>
<li><p> meth_classification - A dataframe containing the differences in M-values for methylation data and their classification corresponding to V.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 4
data_transformed = data_transformation(seq_data=gene_expression_data,
                                  meth_data=methylation_data,
                                  gene_chr=gene_chromosome_data,
                                  N=N)
data_output = idiffomix(seq_data=data_transformed$seq_transformed,
                        meth_data=data_transformed$meth_transformed,
                        gene_chr=gene_chromosome_data,
                        N=N, K=3, L=3, probs=c(0.25,0.75),
                        parallel_process = FALSE)

</code></pre>

<hr>
<h2 id='methylation_data'>Methylation array data for patients suffering from breast cancer</h2><span id='topic+methylation_data'></span>

<h3>Description</h3>

<p>A dataset containing simulated methylation array data for <code class="reqn">C</code> CpG sites associated to <code class="reqn">G</code> genes, from <code class="reqn">R=2</code> sample types, collected from <code class="reqn">N=4</code> patients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(methylation_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 205 rows and 10 columns.
</p>

<ul>
<li><p>Gene: The gene name.
</p>
</li>
<li><p>CpG: The CpG site associated to the gene.
</p>
</li>
<li><p>Patient1_M1: Methylation values from benign tissue from patient 1 for the corresponding CpG site.
</p>
</li>
<li><p>Patient2_M1: Methylation values from benign tissue from patient 2 for the corresponding CpG site.
</p>
</li>
<li><p>Patient3_M1: Methylation values from benign tissue from patient 3 for the corresponding CpG site.
</p>
</li>
<li><p>Patient4_M1: Methylation values from benign tissue from patient 4 for the corresponding CpG site.
</p>
</li>
<li><p>Patient1_M2: Methylation values from tumour tissue from patient 1 for the corresponding CpG site.
</p>
</li>
<li><p>Patient2_M2: Methylation values from tumour tissue from patient 2 for the corresponding CpG site.
</p>
</li>
<li><p>Patient3_M2: Methylation values from tumour tissue from patient 3 for the corresponding CpG site.
</p>
</li>
<li><p>Patient4_M2: Methylation values from tumour tissue from patient 4 for the corresponding CpG site.
</p>
</li></ul>



<h3>Details</h3>

<p>The methylation array data is assumed to be from benign and tumour tissues.
The methylation data comprised of beta-valued methylation levels. The data needs to be transformed before applying idiffomix.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gene_expression_data">gene_expression_data</a></code>
</p>
<p><code><a href="#topic+gene_chromosome_data">gene_chromosome_data</a></code>
</p>

<hr>
<h2 id='plot.idiffomix'>Plots for visualizing the idiffomix class object</h2><span id='topic+plot.idiffomix'></span>

<h3>Description</h3>

<p>Visualise a <code><a href="#topic+idiffomix">idiffomix</a></code> clustering solution by plotting the conditional probabilities estimated for genes and CpG sites (A) per chromosome and (B) for a gene and its corresponding CpG sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'idiffomix'
plot(
  x,
  what = "chromosome",
  CHR = 1,
  Gene = NULL,
  K = 3,
  L = 3,
  gene_cluster_name = c("E-", "E0", "E+"),
  cpg_cluster_name = c("M-", "M0", "M+"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.idiffomix_+3A_x">x</code></td>
<td>
<p>A <code><a href="#topic+idiffomix">idiffomix</a></code> object.</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_what">what</code></td>
<td>
<p>The different plots that can be obtained are either &quot;chromosome&quot; or &quot;gene&quot; (default = &quot;chromosome&quot;).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_chr">CHR</code></td>
<td>
<p>The chromosome number to be visualized (default = 1).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_gene">Gene</code></td>
<td>
<p>The name of the gene to be visualized (default = NULL).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_k">K</code></td>
<td>
<p>Number of clusters for expression data (default = 3; E-,E0,E+).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_l">L</code></td>
<td>
<p>Number of clusters for methylation data (default = 3; M-,M0,M+).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_gene_cluster_name">gene_cluster_name</code></td>
<td>
<p>The names to be given to the clusters for identification (default = c( &quot;E-&quot;,&quot;E0&quot;,&quot;E+&quot;)).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_cpg_cluster_name">cpg_cluster_name</code></td>
<td>
<p>The names to be given to the clusters for identification (default = c( &quot;M-&quot;,&quot;M0&quot;,&quot;M+&quot;)).</p>
</td></tr>
<tr><td><code id="plot.idiffomix_+3A_...">...</code></td>
<td>
<p>Other graphics parameters.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function displays two plots. The first plot displays the conditional probabilities estimated when the joint mixture model is applied to data from a chromosome.
Panel A displays the probability of a gene in the chromosome belonging to each of the K clusters.
Panel B details the estimated matrix pi of conditional probabilities of a CpG site's cluster membership, given its gene's cluster.
Panel C details the conditional probabilities of a gene belonging to cluster $k$ given a single CpG site associated with the gene belongs to cluster $l$, computed using Bayes' theorem,
given tau and pi. The second plot displays the log-fold changes and differences in M-values and the estimated posterior probability of the gene belonging to the K clusters.
Panel A shows the log-fold change and difference in M-values for the given gene and its associated CpG sites while Panel B shows the posterior probabilities of cluster membership for the gene under idiffomix.
</p>


<h3>Value</h3>

<p>This function displays the following plots as requested by the user:
</p>

<ul>
<li><p> chromosome plot - Plot showing the conditional probabilities estimated when the joint mixture model is applied to data from a chromosome.
</p>
</li>
<li><p> gene plot - Plot showing the log-fold changes and differences in M-values and the estimated posterior probability of the gene belonging to the K clusters.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+idiffomix">idiffomix</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>N &lt;- 4
data_transformed = data_transformation(seq_data=gene_expression_data,
                                  meth_data=methylation_data,
                                  gene_chr=gene_chromosome_data,
                                  N=N)
data_output = idiffomix(seq_data=data_transformed$seq_transformed,
                        meth_data=data_transformed$meth_transformed,
                        gene_chr=gene_chromosome_data,
                        N=N, K=3, L=3, probs=c(0.25,0.75),
                        parallel_process = FALSE)
plot(data_output,what="chromosome",CHR=1, Gene=NULL,K=3,L=3,
     gene_cluster_name=c( "E-","E0","E+"),
     cpg_cluster_name=c( "M-","M0","M+"),
     title=NULL)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
