<!DOCTYPE html><html lang="en"><head><title>Help for package iDINGO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iDINGO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#iDINGO-package'>
<p>iDINGO: Integrative Differential Network Analysis in Genomics</p></a></li>
<li><a href='#brca'>
<p>Modified TCGA Breast Cancer data</p></a></li>
<li><a href='#dingo'>
<p>Fit DINGO model</p></a></li>
<li><a href='#extendedBIC'><p>Extended bayesian information criteria for gaussian graphical models</p></a></li>
<li><a href='#gbm'>
<p>Modified TCGA Glioblastoma data</p></a></li>
<li><a href='#Greg.em'><p>Fitting precision regression models</p></a></li>
<li><a href='#idingo'>
<p>Fit iDINGO model</p></a></li>
<li><a href='#plotNetwork'>
<p>Plot differential network</p></a></li>
<li><a href='#scaledMat'>
<p>scale a square matrix</p></a></li>
<li><a href='#scoring.boot'>
<p>Calculating differential score</p></a></li>
<li><a href='#scoring.boot.parallel'>
<p>Calculating differential score with parallel bootstrap scoring</p></a></li>
<li><a href='#Sigmax'>
<p>group specific covariance matrices</p></a></li>
<li><a href='#single.boot'>
<p>Calculating differential score for a single bootstrap</p></a></li>
<li><a href='#trans.Fisher'>
<p>Fisher's Z-transformation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrative Differential Network Analysis in Genomics</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-07-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Caleb A. Class &lt;cclass@butler.edu&gt;, Min Jin Ha &lt;mjha@mdanderson.org&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Caleb A. Class &lt;cclass@butler.edu&gt;</td>
</tr>
<tr>
<td>Imports:</td>
<td>mvtnorm, glasso, parallel, GGMridge, visNetwork, scales</td>
</tr>
<tr>
<td>Depends:</td>
<td>igraph</td>
</tr>
<tr>
<td>Description:</td>
<td>Fits covariate dependent partial correlation matrices for integrative models to identify differential networks between two groups. The methods are described in Class et. al., (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx750">doi:10.1093/bioinformatics/btx750</a>&gt; and Ha et. al., (2015) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv406">doi:10.1093/bioinformatics/btv406</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-07-17 20:37:07 UTC; cclass</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-07-30 14:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='iDINGO-package'>
iDINGO: Integrative Differential Network Analysis in Genomics
</h2><span id='topic+iDINGO-package'></span><span id='topic+iDINGO'></span>

<h3>Description</h3>

<p>This packages jointly estimates group specific partial correlations in a multi-level/platform data set.
</p>


<h3>Details</h3>

<p>This packages jointly estimates group specific partial correlations in a multi-level/platform data set, considering the directionality of effects between platforms using a chain graph model.
</p>


<h3>Author(s)</h3>

<p>Min Jin Ha, Caleb Class, Veerabhadran Baladandayuthapani, and Kim-Anh Do
</p>

<hr>
<h2 id='brca'>
Modified TCGA Breast Cancer data
</h2><span id='topic+brca'></span>

<h3>Description</h3>

<p>Modified TCGA Breast Cancer data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(brca)</code></pre>


<h3>Format</h3>

<p>Three data frames, with columns as standardized miRNA, gene, and protein expressions. One vector with the two classes of the samples (tumor or normal tissue). Also one iDINGO fit object, obtained by running the example in the idingo manual entry.
</p>

<hr>
<h2 id='dingo'>
Fit DINGO model
</h2><span id='topic+dingo'></span>

<h3>Description</h3>

<p>This function fit DINGO model and calculates edge-wise differential scores for all pairwise edges among p variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dingo(dat,x,rhoarray=NULL,diff.score=T,B=100,verbose=T,cores=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dingo_+3A_dat">dat</code></td>
<td>

<p>nxp data with colnames as genename
</p>
</td></tr>
<tr><td><code id="dingo_+3A_x">x</code></td>
<td>
<p>a length n vector representing a binary covariate
</p>
</td></tr>
<tr><td><code id="dingo_+3A_rhoarray">rhoarray</code></td>
<td>
<p>a vector representing candidate tuning parameters of glasso for fitting global network model. If it is one value, then we use the value as the tuning parameter. It is set by NULL as default and we select 100 candidate values.
</p>
</td></tr>
<tr><td><code id="dingo_+3A_diff.score">diff.score</code></td>
<td>
<p>a logical value. If TRUE, edge-wise differential scores are calculated from bootstrap standard error. Otherwise, we fit Steps 1 and 2 of DINGO model to get group specific GGMs (partial correlations)  </p>
</td></tr>
<tr><td><code id="dingo_+3A_b">B</code></td>
<td>
<p>the number of bootstrap samples to calculate differential scores.
</p>
</td></tr>
<tr><td><code id="dingo_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, lists the procedure
</p>
</td></tr>
<tr><td><code id="dingo_+3A_cores">cores</code></td>
<td>
<p>the number of cores to run in parallel for bootstrapping, set to 1 as a default. If more cores are specified than the recommended maximum (the number of cores detected minus 1), this value will be replaced by the recommended value.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>genepair</code></td>
<td>
<p>a p(p-1)/2 x 2 matrix indicating all pairs of genes
</p>
</td></tr>
<tr><td><code>levels.x</code></td>
<td>
<p>a length 2 vector indicating levels of the binary covariate x, the first element is for group 1 and the second element is for group 2
</p>
</td></tr>
<tr><td><code>R1</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 1 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 2 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>boot.diff</code></td>
<td>
<p>a p(p-1)/2 x boot.B matrix indicating bootstrapped difference, Fisher's Z transformed R1 - R2. The rows are corresponding to the order of gene pair and the columns are corresponding to the bootstrap samples
</p>
</td></tr>
<tr><td><code>diff.score</code></td>
<td>
<p>a p(p-1)/2 vector of differential score corresponding to genepair
</p>
</td></tr>
<tr><td><code>p.val</code></td>
<td>
<p>a p(p-1)/2 vector of corrected p-values corresponding to genepair
</p>
</td></tr>
<tr><td><code>rho</code></td>
<td>
<p>selected tuning parameter of glasso fit
</p>
</td></tr>
<tr><td><code>P</code></td>
<td>
<p>p by p matrix of Global component of the DINGO model
</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>p by 2 matrix of the coefficient parameter of the local group specific component L(x) of the DINGO model.</p>
</td></tr>
<tr><td><code>Psi</code></td>
<td>
<p> p by p diagonal matrix of the noise covariance parameter of the local group specific component L(x) of the DINGO model.
</p>
</td></tr>
<tr><td><code>step.times</code></td>
<td>
<p> a length 3 vector containing the elapsed time for Step 1, Step 2, and Bootstrap Scoring, respectively.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Min Jin HA mjha@mdanderson.org, Caleb CLASS caclass@mdanderson.org
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gbm)
# Run DINGO (the first column, 'x', contains the group data).
# This may take 5-10 minutes.
## Not run: fit &lt;- dingo(gbm[,-1], gbm$x, diff.score = TRUE, B = 100, cores = 2)
</code></pre>

<hr>
<h2 id='extendedBIC'>Extended bayesian information criteria for gaussian graphical models
</h2><span id='topic+extendedBIC'></span>

<h3>Description</h3>

<p>Extended bayesian information criteria for gaussian graphical models
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extendedBIC(gamma,omegahat,S,n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extendedBIC_+3A_gamma">gamma</code></td>
<td>

<p>a tuning parameter taking a scalar in [0,1] and leading to stronger penalization of large graphs
</p>
</td></tr>
<tr><td><code id="extendedBIC_+3A_omegahat">omegahat</code></td>
<td>

<p>a p x p matrix indicating an estimates of precision (inverse covariance) matrix
</p>
</td></tr>
<tr><td><code id="extendedBIC_+3A_s">S</code></td>
<td>

<p>a p x p matrix indicating sample covariance matrix
</p>
</td></tr>
<tr><td><code id="extendedBIC_+3A_n">n</code></td>
<td>

<p>a scalar indicating sample size   
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Extended BIC penalized by the size of graphs
</p>


<h3>Author(s)</h3>

<p>Min Jin Ha &lt;mjha@mdanderson.org&gt;</p>


<h3>References</h3>

<p>Foygel, R. and Drton, M. (2010). Extended bayesian information criteria for gaussian graphical models. arXiv preprint arXiv:1011.6640 .
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(glasso)
data(gbm)
x = gbm[,1]
Y = gbm[,-1]

# Estimating inverse covariance matrix using GLasso #
S = cov(Y)

rhoarray = exp(seq(log(0.001),log(1),length=100))
BIC = rep(0,length(rhoarray))
for (rh in 1:length(rhoarray)) {
    fit.gl1 = glasso(S,rho=rhoarray[rh])
    BIC[rh] = extendedBIC(gamma=0,omegahat=fit.gl1$wi,S=S,n=nrow(Y))
}
rho = rhoarray[which.min(BIC)]
fit.gl2 = glasso(S,rho=rho)
Omega = fit.gl2$wi


</code></pre>

<hr>
<h2 id='gbm'>
Modified TCGA Glioblastoma data
</h2><span id='topic+gbm'></span>

<h3>Description</h3>

<p>Modified TCGA Glioblastoma data.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gbm)</code></pre>


<h3>Format</h3>

<p>A data frame with first column as a covariate and other columns as standardized gene expressions.
</p>

<hr>
<h2 id='Greg.em'>Fitting precision regression models
</h2><span id='topic+Greg.em'></span>

<h3>Description</h3>

<p>This function fits the covariance regression model by Hoff and Niu (2012) using EM algorithm with the restriction of diagonal matrix for the noise variance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Greg.em(formula, data = NULL, R = 1, tol = 1e-10, itmax = 1000, verbose = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Greg.em_+3A_formula">formula</code></td>
<td>

<p>an object of class &quot;formula&quot; used in model.frame function</p>
</td></tr>
<tr><td><code id="Greg.em_+3A_data">data</code></td>
<td>

<p>a data frame used in model.frame function
</p>
</td></tr>
<tr><td><code id="Greg.em_+3A_r">R</code></td>
<td>

<p>rank of the model 
</p>
</td></tr>
<tr><td><code id="Greg.em_+3A_tol">tol</code></td>
<td>

<p>a stopping criterion
</p>
</td></tr>
<tr><td><code id="Greg.em_+3A_itmax">itmax</code></td>
<td>

<p>maximum number of iteration
</p>
</td></tr>
<tr><td><code id="Greg.em_+3A_verbose">verbose</code></td>
<td>

<p>If true, estimation results for each iteration are printed
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>A</code></td>
<td>
<p>MLE of the baseline covariance matrix</p>
</td></tr>
<tr><td><code>B</code></td>
<td>
<p>MLE of the regression coefficients</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Min Jin Ha &lt;mjha@mdanderson.org&gt;</p>


<h3>References</h3>

<p>Hoff, P. D. and Niu, X. (2012) A covariance regression model. Statistica Sinica, 22, 729-753.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(glasso)
data(gbm)
x = gbm[,1]
Y = as.matrix(gbm[,-1])
p = ncol(Y)
# Estimating inverse covariance matrix using GLasso #
S = cov(Y)
w.upper = which(upper.tri(S))

rhoarray = exp(seq(log(0.001),log(1),length=100))
BIC = rep(0,length(rhoarray))
for (rh in 1:length(rhoarray)) {
    fit.gl1 = glasso(S,rho=rhoarray[rh])
    BIC[rh] = extendedBIC(gamma=0,omegahat=fit.gl1$wi,S=S,n=nrow(Y))
}
rho = rhoarray[which.min(BIC)]
fit.gl2 = glasso(S,rho=rho)
Omega = fit.gl2$wi

# Fitting (Covariance Regression on transformed data)
diag.Omega = diag(Omega)
P = -Omega/diag.Omega
diag(P) = 0

tY = Y 
mdat = apply(tY,2,mean)
sdat = apply(tY,2,sd)
std.tY = t((t(tY) - mdat)/sdat)
smat = diag(sdat)

## rank 1 covariance regression
fit.g = Greg.em(std.tY~x,R=1) 
</code></pre>

<hr>
<h2 id='idingo'>
Fit iDINGO model
</h2><span id='topic+idingo'></span>

<h3>Description</h3>

<p>This function fits the iDINGO model and calculates edge-wise differential scores for all pairwise edges among p variables between multiple platforms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>idingo(dat,dat2=NULL,dat3=NULL,x,plats=NULL,rhoarray=NULL,
        diff.score=T,B=100,verbose=T,cores=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="idingo_+3A_dat">dat</code></td>
<td>

<p>nxp dataframe/matrix with colnames as genename
</p>
</td></tr>
<tr><td><code id="idingo_+3A_dat2">dat2</code></td>
<td>

<p>Second nxp dataframe/matrix with colnames as genename (optional)
</p>
</td></tr>
<tr><td><code id="idingo_+3A_dat3">dat3</code></td>
<td>

<p>Third nxp dataframe/matrix with colnames as genename (optional)
</p>
</td></tr>
<tr><td><code id="idingo_+3A_x">x</code></td>
<td>
<p>a length n vector representing a binary covariate
</p>
</td></tr>
<tr><td><code id="idingo_+3A_plats">plats</code></td>
<td>
<p>a length 1-3 vector (corresponding to the number of data sets submitted, with names for the platforms/levels of the data, such as &quot;microRNA&quot; or &quot;RNAseq&quot;. This is optional, and default names &quot;platN&quot; will be used if names are not provided.
</p>
</td></tr>
<tr><td><code id="idingo_+3A_rhoarray">rhoarray</code></td>
<td>
<p>a vector representing candidate tuning parameters of glasso for fitting global network model. If it is one value, then we use the value as the tuning parameter. It is set by NULL as default and we select 100 candidate values.
</p>
</td></tr>
<tr><td><code id="idingo_+3A_diff.score">diff.score</code></td>
<td>
<p>a logical value. If TRUE, edge-wise differential scores are calculated from bootstrap standard error. Otherwise, we fit Steps 1 and 2 of DINGO model to get group specific GGMs (partial correlations)  </p>
</td></tr>
<tr><td><code id="idingo_+3A_b">B</code></td>
<td>
<p>the number of bootstrap samples to calculate differential scores.
</p>
</td></tr>
<tr><td><code id="idingo_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, lists the procedure
</p>
</td></tr>
<tr><td><code id="idingo_+3A_cores">cores</code></td>
<td>
<p>the number of cores to run in parallel for bootstrapping, set to 1 as a default. If more cores are specified than the recommended maximum (the number of cores detected minus 1), this value will be replaced by the recommended value.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>genepair</code></td>
<td>
<p>a p(p-1)/2 x 2 matrix indicating all pairs of genes
</p>
</td></tr>
<tr><td><code>levels.x</code></td>
<td>
<p>a length 2 vector indicating levels of the binary covariate x, the first element is for group 1 and the second element is for group 2
</p>
</td></tr>
<tr><td><code>R1</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 1 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 2 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>diff.score</code></td>
<td>
<p>a p(p-1)/2 vector of differential score corresponding to genepair
</p>
</td></tr>
<tr><td><code>p.val</code></td>
<td>
<p>a p(p-1)/2 vector of corrected p-values corresponding to genepair
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Caleb CLASS caclass@mdanderson.org, Min Jin HA mjha@mdanderson.org
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(brca)

# Run iDINGO with microRNA, RNA, and protein data.
# Generally, we recommend a minimum of 100 bootstraps.
## Not run: fit &lt;- idingo(brca$mirna, dat2 = brca$rna, dat3 = brca$prot,
    x = brca$class, plats = c("microRNA", "RNA", "Protein"),
    diff.score = TRUE, B = 20, cores = 2)
## End(Not run)
</code></pre>

<hr>
<h2 id='plotNetwork'>
Plot differential network
</h2><span id='topic+plotNetwork'></span>

<h3>Description</h3>

<p>This function plots the differential network from a completed DINGO or iDINGO model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNetwork(fit, threshold=0.05, thresh.type="p.val", layout="circular",
            legend.pos="left")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotNetwork_+3A_fit">fit</code></td>
<td>

<p>output from running dingo() or idingo()
</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_threshold">threshold</code></td>
<td>

<p>a numeric value containing the threshold for which edges will be included in the differential network plot. If 'thresh.type' is 'p.val', all edges with p-values below this threshold will be included in the plot. If 'thresh.type' is 'diff.score', all edges with absolute differential scores above this threshold will be included in the plot.
</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_thresh.type">thresh.type</code></td>
<td>
<p>either 'p.val' or 'diff.score', defining which variable is used as threshold for edge inclusion.
</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_layout">layout</code></td>
<td>
<p>either 'circular' or one of igraph's supported layouts. If 'circular', dingo networks will be plotted in a circle, and idingo networks will be plotted as a cylinder (with each platform/level as a separate circle).
</p>
</td></tr>
<tr><td><code id="plotNetwork_+3A_legend.pos">legend.pos</code></td>
<td>
<p>Legend position for multi-platform networks, in c(&quot;left&quot;, &quot;right&quot;). Legend is not included for single-platform networks.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>visNet</code></td>
<td>
<p>a network plot, using igraph and visNetwork
</p>
</td></tr>
</table>


<h3>Note</h3>

<p>To calculate differential scores and p-values for use in network plot thresholding, diff.score must be set to TRUE in dingo() or idingo().
</p>


<h3>Author(s)</h3>

<p>Caleb CLASS caclass@mdanderson.org, Min Jin HA mjha@mdanderson.org
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(brca)

# Plot the iDINGO result using a p-value threshold of 0.01.
plotNetwork(brca$fit, threshold = 0.01, thresh.type = "p.val")
</code></pre>

<hr>
<h2 id='scaledMat'>
scale a square matrix
</h2><span id='topic+scaledMat'></span>

<h3>Description</h3>

<p>scale a square matrix to have unit diagonal elements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scaledMat(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scaledMat_+3A_x">x</code></td>
<td>

<p>a square matrix with positive diagonal elements
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>scaled matrix of x
</p>


<h3>Author(s)</h3>

<p>Min Jin Ha mjha@mdanderson.org
</p>

<hr>
<h2 id='scoring.boot'>
Calculating differential score
</h2><span id='topic+scoring.boot'></span>

<h3>Description</h3>

<p>This function calculates standard errors for edge-wise partial correlation differences obtained from DINGO model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scoring.boot(stddat,z,Omega,A,B,boot.B=100,verbose=T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scoring.boot_+3A_stddat">stddat</code></td>
<td>

<p>standardized nxp data with colnames as genename
</p>
</td></tr>
<tr><td><code id="scoring.boot_+3A_z">z</code></td>
<td>
<p>a length n vector representing a binary covariate
</p>
</td></tr>
<tr><td><code id="scoring.boot_+3A_omega">Omega</code></td>
<td>
<p>a p x p precision matrix for std dat which implies the global network
</p>
</td></tr>
<tr><td><code id="scoring.boot_+3A_a">A</code></td>
<td>
<p>p x p matrix of the MLE for the baseline covariance matrix which is obtained from A value of the Greg.em function.
</p>
</td></tr>
<tr><td><code id="scoring.boot_+3A_b">B</code></td>
<td>
<p>p x 2 matrix of the MLE for the regression coefficient which is obtained from B value of the Greg.em function
</p>
</td></tr>
<tr><td><code id="scoring.boot_+3A_boot.b">boot.B</code></td>
<td>
<p>a scalar indicating the number of bootstraps
</p>
</td></tr>
<tr><td><code id="scoring.boot_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, lists the bootstrap replications
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>genepair</code></td>
<td>
<p>a p(p-1)/2 x 2 matrix indicating all pairs of genes
</p>
</td></tr>
<tr><td><code>levels.z</code></td>
<td>
<p>a length 2 vector indicating levels of the binary covariate z, the first element is for group 1 and the second element is for group 2
</p>
</td></tr>
<tr><td><code>R1</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 1 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 2 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>boot.diff</code></td>
<td>
<p>a p(p-1)/2 x boot.B matrix indicating bootstrapped difference, Fisher's Z transformed R1 - R2. The rows are corresponding to the order of gene pair and the columns are corresponding to the bootstrap samples
</p>
</td></tr>
<tr><td><code>diff.score</code></td>
<td>
<p>a p(p-1)/2 vector of differential score corresponding to genepair
</p>
</td></tr>
<tr><td><code>p.val</code></td>
<td>
<p>a p(p-1)/2 vector of corrected p-values corresponding to genepair
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Min Jin HA mjha@mdanderson.org
</p>

<hr>
<h2 id='scoring.boot.parallel'>
Calculating differential score with parallel bootstrap scoring
</h2><span id='topic+scoring.boot.parallel'></span>

<h3>Description</h3>

<p>This function calculates standard errors for edge-wise partial correlation differences obtained from DINGO model. Bootstrapping is done in parallel using parSapply from the &quot;parallel&quot; library.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scoring.boot.parallel(stddat,z,Omega,A,B,boot.B=100,verbose=T,cores=1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scoring.boot.parallel_+3A_stddat">stddat</code></td>
<td>

<p>standardized nxp data with colnames as genename
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_z">z</code></td>
<td>
<p>a length n vector representing a binary covariate
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_omega">Omega</code></td>
<td>
<p>a p x p precision matrix for std dat which implies the global network
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_a">A</code></td>
<td>
<p>p x p matrix of the MLE for the baseline covariance matrix which is obtained from A value of the Greg.em function.
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_b">B</code></td>
<td>
<p>p x 2 matrix of the MLE for the regression coefficient which is obtained from B value of the Greg.em function
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_boot.b">boot.B</code></td>
<td>
<p>a scalar indicating the number of bootstraps
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE, lists the bootstrap replications
</p>
</td></tr>
<tr><td><code id="scoring.boot.parallel_+3A_cores">cores</code></td>
<td>
<p>the number of cores to run in parallel for bootstrapping, set to 1 as a default.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>genepair</code></td>
<td>
<p>a p(p-1)/2 x 2 matrix indicating all pairs of genes
</p>
</td></tr>
<tr><td><code>levels.z</code></td>
<td>
<p>a length 2 vector indicating levels of the binary covariate z, the first element is for group 1 and the second element is for group 2
</p>
</td></tr>
<tr><td><code>R1</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 1 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>a length p(p-1)/2 vector indicating partial correlations for group 2 and the order is corresponding to the order of genepair
</p>
</td></tr>
<tr><td><code>boot.diff</code></td>
<td>
<p>a p(p-1)/2 x boot.B matrix indicating bootstrapped difference, Fisher's Z transformed R1 - R2. The rows are corresponding to the order of gene pair and the columns are corresponding to the bootstrap samples
</p>
</td></tr>
<tr><td><code>diff.score</code></td>
<td>
<p>a p(p-1)/2 vector of differential score corresponding to genepair
</p>
</td></tr>
<tr><td><code>p.val</code></td>
<td>
<p>a p(p-1)/2 vector of corrected p-values corresponding to genepair
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Min Jin HA mjha@mdanderson.org, Caleb CLASS caclass@mdanderson.org
</p>

<hr>
<h2 id='Sigmax'>
group specific covariance matrices
</h2><span id='topic+Sigmax'></span>

<h3>Description</h3>

<p>From parameters of DINGO model, group specific covariance matrices are obtained
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Sigmax(P = NULL, Q, Psi, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Sigmax_+3A_p">P</code></td>
<td>

<p>a p x p matrix specifying global component
</p>
</td></tr>
<tr><td><code id="Sigmax_+3A_q">Q</code></td>
<td>

<p>the coefficient parameter matrix of covariance regression model using Greg.em function
</p>
</td></tr>
<tr><td><code id="Sigmax_+3A_psi">Psi</code></td>
<td>

<p>the diagonal error variance matrix of covariance regression model using Greg.em function
</p>
</td></tr>
<tr><td><code id="Sigmax_+3A_x">x</code></td>
<td>

<p>a vector specifying group. This must be corresponding to the design matrix of Greg.em function
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>group specific precision matrix</p>


<h3>Author(s)</h3>

<p>Min Jin Ha &lt;mjha@mdanderson.org&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(glasso)
data(gbm)
x = gbm[,1]
Y = as.matrix(gbm[,-1])
p = ncol(Y)
# Estimating inverse covariance matrix using GLasso #
S = cov(Y)
w.upper = which(upper.tri(S))

rhoarray = exp(seq(log(0.001),log(1),length=100))
BIC = rep(0,length(rhoarray))
for (rh in 1:length(rhoarray)) {
    fit.gl1 = glasso(S,rho=rhoarray[rh])
    BIC[rh] = extendedBIC(gamma=0,omegahat=fit.gl1$wi,S=S,n=nrow(Y))
}
rho = rhoarray[which.min(BIC)]
fit.gl2 = glasso(S,rho=rho)
Omega = fit.gl2$wi

# Fitting (Covariance Regression on transformed data)
diag.Omega = diag(Omega)
P = -Omega/diag.Omega
diag(P) = 0

tY = Y 
mdat = apply(tY,2,mean)
sdat = apply(tY,2,sd)
std.tY = t((t(tY) - mdat)/sdat)
smat = diag(sdat)

## rank 1 covariance regression
fit.g = Greg.em(std.tY~x,R=1) 
## obtain covariance matrix of Y when x=1
sigmaX1 = Sigmax(Q=fit.g$B,P=P,Psi=fit.g$A,x=c(1,1)) 
</code></pre>

<hr>
<h2 id='single.boot'>
Calculating differential score for a single bootstrap
</h2><span id='topic+single.boot'></span>

<h3>Description</h3>

<p>This function calculates the edge-wise partial correlation difference for a single bootstrap.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>single.boot(i, z, n, tY.org, P, levels.z, w.upper)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="single.boot_+3A_i">i</code></td>
<td>
<p>iteration number. This is not used within this function, but necessary for parSapply within scoring.boot.parallel function.
</p>
</td></tr>
<tr><td><code id="single.boot_+3A_z">z</code></td>
<td>
<p>a length n vector representing a binary covariate
</p>
</td></tr>
<tr><td><code id="single.boot_+3A_n">n</code></td>
<td>
<p>the number of rows in data
</p>
</td></tr>
<tr><td><code id="single.boot_+3A_ty.org">tY.org</code></td>
<td>
<p>the transformed standardized data
</p>
</td></tr>
<tr><td><code id="single.boot_+3A_p">P</code></td>
<td>
<p>the global correlation component
</p>
</td></tr>
<tr><td><code id="single.boot_+3A_levels.z">levels.z</code></td>
<td>
<p>the levels of the covariates
</p>
</td></tr>
<tr><td><code id="single.boot_+3A_w.upper">w.upper</code></td>
<td>
<p>the upper triangular of Omega
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>boot.diff</code></td>
<td>
<p>the difference for this bootstrap
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Min Jin HA mjha@mdanderson.org, Caleb CLASS caclass@mdanderson.org
</p>

<hr>
<h2 id='trans.Fisher'>
Fisher's Z-transformation
</h2><span id='topic+trans.Fisher'></span>

<h3>Description</h3>

<p>Fisher's Z-transformation of (partial) correlation.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trans.Fisher_+3A_x">x</code></td>
<td>

<p>a vector having entries between -1 and 1
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Fisher's Z-transformed values
</p>


<h3>Author(s)</h3>

<p>Min Jin HA mjha@mdanderson.org	
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
