<!DOCTYPE html><html><head><title>Help for package idiogramFISH</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {idiogramFISH}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#idiogramFISH-package'><p>idiogramFISH: Shiny App. Idiograms with Marks and Karyotype Indices</p></a></li>
<li><a href='#addAttributesDfChrSize'><p>addAttributesDfChrSize</p></a></li>
<li><a href='#addNotes'><p>This is an internal function that adds notes</p></a></li>
<li><a href='#armRatioCI'><p>FUNCTIONS armRatioCI and swapChrRegionDfSizeAndMarks.</p></a></li>
<li><a href='#arrowPlotMark'><p>arrowPlotMark</p>
This is an internal function that plot marks</a></li>
<li><a href='#asymmetry'><p>FUNCTIONS asymmetry and asymmetryA2.</p></a></li>
<li><a href='#checkArmHolocenError'><p>checkArmHolocenError</p>
This is an internal function that makes list of d.fs</a></li>
<li><a href='#checkNameChrDfMarks'><p>Function to check names of OTUs (species) among data.frames</p></a></li>
<li><a href='#chrbasicdataHolo'><p>Chr. basic data Holo.</p></a></li>
<li><a href='#chrbasicdatamono'><p>Chr. basic data Monocen.</p></a></li>
<li><a href='#citrusSize'><p>FUNCTIONS: citrusSize, citrusMarkPos, markOverCMA</p></a></li>
<li><a href='#cMPlotMark'><p>This is an internal function that eliminates factors</p></a></li>
<li><a href='#customCeiling'><p>Function to do ceiling at different levels</p></a></li>
<li><a href='#dfMarkStyle'><p>Mark characteristics</p></a></li>
<li><a href='#dfToListColumn'><p>dfToListColumn</p>
This is an internal function that makes list of d.fs</a></li>
<li><a href='#filterExtraOTU'><p>filterExtraOTU</p>
This is an internal function that filters list of d.f. of marks</a></li>
<li><a href='#fixChrNameDupDF'><p>This function fixes duplicated names when no marks</p></a></li>
<li><a href='#genBankReadIF'><p>FUNCTION genBankReadIF</p></a></li>
<li><a href='#insertInList'><p>This is an internal function that inserts in vector or list</p></a></li>
<li><a href='#make.uniqueIF'><p>FUNCTION make.uniqueIF</p></a></li>
<li><a href='#makedfMarkColor'><p>Function to automatically make a data.frame of Marks' characterisitcs</p></a></li>
<li><a href='#makedfMarkColorMycolors'><p>FUNCTION to make a data.frame of Marks' characterisitcs</p></a></li>
<li><a href='#makeNumCols'><p>These are internal functions that eliminate factors</p></a></li>
<li><a href='#makeVectorNames'><p>makeVectorNames</p>
This is an internal function that makes a vector of names</a></li>
<li><a href='#mapGGChr'><p>FUNCTIONS mapGGChr and mapGGChrMark (for ggplot)</p></a></li>
<li><a href='#mapXY'><p>mapXY</p>
This is an internal function that creates coords for chr. or arms or marks</a></li>
<li><a href='#mapXYCenMimic'><p>mapXYCenMimic</p>
This is an internal function that creates coords for chr. or arms or marks</a></li>
<li><a href='#markdataholo'><p>Mark Positional data - Holocen.</p></a></li>
<li><a href='#markDistCenGISHfix'><p>markDistCenGISHfix</p>
This is an internal function that fixes markDistCen when GISH</a></li>
<li><a href='#markposDFs'><p>Mark Positional data - monocentrics</p></a></li>
<li><a href='#mimicCenPlotMark'><p>mimicCenPlotMark</p>
This is an internal function that plot marks</a></li>
<li><a href='#namesToColumns'><p>FUNCTION that modifies marks' names into columns</p></a></li>
<li><a href='#newOrderColumn'><p>newOrderColumn</p>
This is an internal function that adds neworder column to
listOfdfChromSizeCenType</a></li>
<li><a href='#perMark'><p>FUNCTION perMark</p></a></li>
<li><a href='#plotDotMarks'><p>plotDotMarks</p>
This is an internal function that plot dot marks</a></li>
<li><a href='#plotIdiograms'><p>Plot idiograms of karyotypes/genomes with and without centromere</p></a></li>
<li><a href='#plotlabelsright'><p>This is an internal function to plot labels to the right, when &quot;aside&quot; for</p>
its position</a></li>
<li><a href='#plotSqMarks'><p>sqPlotMarkCP</p>
This is an internal function that plot marks</a></li>
<li><a href='#posCalc'><p>FUNCTION posCalc and fillMarkInfo</p></a></li>
<li><a href='#removeNAFromList'><p>removeNAFromList</p>
This is an internal function that removes NA from list</a></li>
<li><a href='#robert'><p>FUNCTION to produce a Robertsonian translocation</p></a></li>
<li><a href='#rulerPlot'><p>rulerPlot</p>
This is an internal function that plots rulers</a></li>
<li><a href='#rulerTitle'><p>rulerTitle</p>
This is an internal function that plots rulers</a></li>
<li><a href='#runBoard'><p>FUNCTION runBoard</p></a></li>
<li><a href='#textLabel'><p>textLabel</p>
This is an internal function that plot labels of cM marks</a></li>
<li><a href='#textLabelDots'><p>textLabelDots</p>
This is an internal function that plot labels of cM marks</a></li>
<li><a href='#yVertoHor'><p>FUNCTIONS: yVertoHor, xHortoVer, xHortoVerRoundCen, xHortoVerDots,</p>
mapCircle, mapRadius,
transYList, mapOTUnames, addOTUnames, applyMapCircle, intercalate,
drawPlot, drawPlotMark,
drawPlotMarkLine, drawCen, circLabelMark, circPlotDots, plotChrNames,
addChrNameAttrMark,
addChrNameAttrMarkDots, xMarkMap, xMarkMapLeft, markMapPer, markMapPercM,
markMapPerDots,
radDotsPer, centerMarkMapPer, transyListMark, transyListCen, transRadDots,
transyListMarkDots,
oneDot, OTUlabelsright</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Shiny App. Idiograms with Marks and Karyotype Indices</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.13</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-08-21</td>
</tr>
<tr>
<td>Description:</td>
<td>Plot idiograms of karyotypes, plasmids, circular chr. having a set of data.frames for chromosome data and optionally mark data. Two styles of chromosomes can be used: without or with visible chromatids. Supports micrometers, cM and Mb or any unit. Three styles of centromeres are available: triangle, rounded and inProtein; and six styles of marks are available: square (squareLeft), dots, cM (cMLeft), cenStyle, upArrow (downArrow), exProtein (inProtein); its legend (label) can be drawn inline or to the right of karyotypes. Idiograms can also be plotted in concentric circles. It is possible to calculate chromosome indices by Levan et al. (1964) &lt;<a href="https://doi.org/10.1111%2Fj.1601-5223.1964.tb01953.x">doi:10.1111/j.1601-5223.1964.tb01953.x</a>&gt;, karyotype indices of Watanabe et al. (1999) &lt;<a href="https://doi.org/10.1007%2FPL00013869">doi:10.1007/PL00013869</a>&gt; and Romero-Zarco (1986) &lt;<a href="https://doi.org/10.2307%2F1221906">doi:10.2307/1221906</a>&gt; and classify chromosomes by morphology Guerra (1986) and Levan et al. (1964).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>rlang, crayon, plyr, dplyr, tidyr, scales</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rentrez, ggplot2, ggpubr, ggtree, phytools, treeio,
rmdformats, knitr, kableExtra, rvcheck, badger, rmarkdown,
RCurl, shiny, shinydashboard, rhandsontable, gtools,
rclipboard, clipr, shinyjs</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>rmdformats, knitr, kableExtra, rmarkdown, RCurl,
rvcheck, badger, rentrez</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>pandoc (&gt;= 2.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://ferroao.gitlab.io/manualidiogramfish/">https://ferroao.gitlab.io/manualidiogramfish/</a>,
<a href="https://ferroao.gitlab.io/idiogramfishhelppages">https://ferroao.gitlab.io/idiogramfishhelppages</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://gitlab.com/ferroao/idiogramFISH/-/issues">https://gitlab.com/ferroao/idiogramFISH/-/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-22 15:15:44 UTC; fernando</td>
</tr>
<tr>
<td>Author:</td>
<td>Fernando Roa [aut, cre],
  Mariana PC Telles [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fernando Roa &lt;froao@unal.edu.co&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-22 16:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='idiogramFISH-package'>idiogramFISH: Shiny App. Idiograms with Marks and Karyotype Indices</h2><span id='topic+idiogramFISH'></span><span id='topic+idiogramFISH-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Plot idiograms of karyotypes, plasmids, circular chr. having a set of data.frames for chromosome data and optionally mark data. Two styles of chromosomes can be used: without or with visible chromatids. Supports micrometers, cM and Mb or any unit. Three styles of centromeres are available: triangle, rounded and inProtein; and six styles of marks are available: square (squareLeft), dots, cM (cMLeft), cenStyle, upArrow (downArrow), exProtein (inProtein); its legend (label) can be drawn inline or to the right of karyotypes. Idiograms can also be plotted in concentric circles. It is possible to calculate chromosome indices by Levan et al. (1964) <a href="https://doi.org/10.1111/j.1601-5223.1964.tb01953.x">doi:10.1111/j.1601-5223.1964.tb01953.x</a>, karyotype indices of Watanabe et al. (1999) <a href="https://doi.org/10.1007/PL00013869">doi:10.1007/PL00013869</a> and Romero-Zarco (1986) <a href="https://doi.org/10.2307/1221906">doi:10.2307/1221906</a> and classify chromosomes by morphology Guerra (1986) and Levan et al. (1964).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Fernando Roa <a href="mailto:froao@unal.edu.co">froao@unal.edu.co</a>
</p>
<p>Other contributors:
</p>

<ul>
<li><p> Mariana PC Telles [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://ferroao.gitlab.io/manualidiogramfish/">https://ferroao.gitlab.io/manualidiogramfish/</a>
</p>
</li>
<li> <p><a href="https://ferroao.gitlab.io/idiogramfishhelppages">https://ferroao.gitlab.io/idiogramfishhelppages</a>
</p>
</li>
<li><p> Report bugs at <a href="https://gitlab.com/ferroao/idiogramFISH/-/issues">https://gitlab.com/ferroao/idiogramFISH/-/issues</a>
</p>
</li></ul>


<hr>
<h2 id='addAttributesDfChrSize'>addAttributesDfChrSize</h2><span id='topic+addAttributesDfChrSize'></span>

<h3>Description</h3>

<p>returns all cols. as character
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addAttributesDfChrSize(
  listOfdfChromSize,
  threshold,
  specialOTUNames,
  centromereSize,
  MbThreshold,
  cenFactor,
  chrWidth,
  specialChrWidth,
  squareness,
  modifyChr = TRUE,
  mymessage = TRUE
)
</code></pre>


<h3>Value</h3>

<p>list of data.frames
</p>

<hr>
<h2 id='addNotes'>This is an internal function that adds notes</h2><span id='topic+addNotes'></span>

<h3>Description</h3>

<p>This is an internal function that adds notes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addNotes(
  notes,
  listOfdfChromSizenoNA,
  groupName,
  indexCount,
  morphoCount,
  xmnoNA,
  ymnoNA,
  distTextChr,
  chrIdCount,
  notesPosX,
  notesPosY,
  notesTextSize,
  defaultFontFamily2,
  OTUasNote,
  OTUfont,
  OTUfamily,
  downNote,
  rightN,
  leftNoteFont,
  leftNoteFontUp,
  noteFont,
  parseTypes,
  parseStr2lang
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addNotes_+3A_notes">notes</code></td>
<td>
<p>x component of polygon</p>
</td></tr>
<tr><td><code id="addNotes_+3A_listofdfchromsizenona">listOfdfChromSizenoNA</code></td>
<td>
<p>y component of polygon</p>
</td></tr>
<tr><td><code id="addNotes_+3A_groupname">groupName</code></td>
<td>
<p>y of chr</p>
</td></tr>
<tr><td><code id="addNotes_+3A_indexcount">indexCount</code></td>
<td>
<p>colors for marks</p>
</td></tr>
<tr><td><code id="addNotes_+3A_morphocount">morphoCount</code></td>
<td>
<p>list of df. of mark pos.</p>
</td></tr>
<tr><td><code id="addNotes_+3A_xmnona">xmnoNA</code></td>
<td>
<p>thick of border line</p>
</td></tr>
<tr><td><code id="addNotes_+3A_ymnona">ymnoNA</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="addNotes_+3A_disttextchr">distTextChr</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="addNotes_+3A_chridcount">chrIdCount</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="addNotes_+3A_notesposx">notesPosX</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="addNotes_+3A_notesposy">notesPosY</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="addNotes_+3A_notestextsize">notesTextSize</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="addNotes_+3A_defaultfontfamily2">defaultFontFamily2</code></td>
<td>
<p>character, regex</p>
</td></tr>
<tr><td><code id="addNotes_+3A_downnote">downNote</code></td>
<td>
<p>numeric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='armRatioCI'>FUNCTIONS armRatioCI and swapChrRegionDfSizeAndMarks.</h2><span id='topic+armRatioCI'></span><span id='topic+swapChrRegionDfSizeAndMarks'></span>

<h3>Description</h3>

<p>armRatioCI: reads a data.frame and produces AR (arm ratio), CI
(centromeric index) , Guerra and Levan classifications.
</p>
<p>swapChrRegionDfSizeAndMarks: reads data.frames to swap arms
</p>


<h3>Usage</h3>

<pre><code class='language-R'>armRatioCI(dfChrSize, rnumeric = FALSE)

swapChrRegionDfSizeAndMarks(dfChrSize, dfMarkPos, chrNamesToSwap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="armRatioCI_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>name of data.frame with columns: shortArmSize, longArmSize</p>
</td></tr>
<tr><td><code id="armRatioCI_+3A_rnumeric">rnumeric</code></td>
<td>
<p>boolean, returns only numeric AR, CI</p>
</td></tr>
<tr><td><code id="armRatioCI_+3A_dfmarkpos">dfMarkPos</code></td>
<td>
<p>name of data.frame of marks</p>
</td></tr>
<tr><td><code id="armRatioCI_+3A_chrnamestoswap">chrNamesToSwap</code></td>
<td>
<p>name of chr. names to swap arms</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame (armRatioCI)
</p>
<p>list of data.frames (swapChrRegionDfSizeAndMarks)
</p>


<h3>References</h3>

<p>LEVAN A, FREDGA K, SANDBERG AA (1964) NOMENCLATURE FOR
CENTROMERIC POSITION ON CHROMOSOMES. Hereditas 52:201-220.
</p>
<p>Guerra. 1986. Reviewing the chromosome nomenclature of Levan et
al. Braz. Jour. Gen. Vol IX, 4, 741-743
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chrbasicdatamono">chrbasicdatamono</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>armRatioCI(dfOfChrSize)
armRatioCI(bigdfOfChrSize)

swapChrRegionDfSizeAndMarks(dfOfChrSize, dfOfMarks, "1")
</code></pre>

<hr>
<h2 id='arrowPlotMark'>arrowPlotMark
This is an internal function that plot marks</h2><span id='topic+arrowPlotMark'></span>

<h3>Description</h3>

<p>It returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>arrowPlotMark(
  squareness,
  xMark,
  yMark,
  dfMarkColorInt,
  listOfdfMarkPosArrow,
  chrWidth,
  n,
  lwd.chr,
  circularPlot,
  y,
  x,
  markLabelSize,
  separFactor,
  labelSpacing,
  circleCenter,
  circleCenterY,
  radius,
  ylistTransChr,
  rotation,
  arrowheadWidthShrink,
  alpha_val
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="arrowPlotMark_+3A_squareness">squareness</code></td>
<td>
<p>squareness of vertices &lt;</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_xmark">xMark</code></td>
<td>
<p>x component of polygon</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_ymark">yMark</code></td>
<td>
<p>yMark component of polygon</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_dfmarkcolorint">dfMarkColorInt</code></td>
<td>
<p>colors for marks</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_listofdfmarkposarrow">listOfdfMarkPosArrow</code></td>
<td>
<p>list of df. of mark pos.</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_chrwidth">chrWidth</code></td>
<td>
<p>numeric, width of chr.</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_n">n</code></td>
<td>
<p>numeric, to define vertices of rounded portions</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_lwd.chr">lwd.chr</code></td>
<td>
<p>thick of border line</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean TRUE for circ.</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_y">y</code></td>
<td>
<p>list, y coords.</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>numeric, font size</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_separfactor">separFactor</code></td>
<td>
<p>numeric, sep. kars.</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_labelspacing">labelSpacing</code></td>
<td>
<p>numeric, spacing</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_circlecenter">circleCenter</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_circlecentery">circleCenterY</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_radius">radius</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_ylisttranschr">ylistTransChr</code></td>
<td>
<p>list, transf. coords.</p>
</td></tr>
<tr><td><code id="arrowPlotMark_+3A_rotation">rotation</code></td>
<td>
<p>rotate*</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='asymmetry'>FUNCTIONS asymmetry and asymmetryA2.</h2><span id='topic+asymmetry'></span><span id='topic+asymmetryA2'></span>

<h3>Description</h3>

<p>asymmetry: calculates karyotype asymmetry A and A2.
</p>
<p>asymmetryA2: calculates karyotype asymmetry A2
</p>
<p>This functions read a data.frame  with columns:
<code>shortArmSize</code> and <code>longArmSize</code>
</p>
<p>If several species present, use
column <code>OTU</code>.
</p>
<p>It returns a list with the A and A2 indices
</p>
<p style="text-align: center;"><code class="reqn">A = \frac{\sum_{i=1}^{n} \frac{longArm_{i} - shortArm_{i}}{CL_{i}}
}{n} </code>
</p>

<p>A: Watanabe et al. (1999) asymmetry of karyotype ranging from 0
(symmetric) to 1 (asymmetric)
</p>
<p style="text-align: center;"><code class="reqn">A_{2} = \frac{sCL}{xCL}</code>
</p>

<p>(s = std dev, CL = chr. length, x = mean) (Romero-Zarco 1986)
</p>
<p>related to:
</p>
<p style="text-align: center;"><code class="reqn">CV_{CL} = A_{2} * 100</code>
</p>

<p>(CV = coeff. var.) (Paszko 2006)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>asymmetry(dfChrSize, asDf = FALSE)

asymmetryA2(dfChrSize)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="asymmetry_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>name of data.frame</p>
</td></tr>
<tr><td><code id="asymmetry_+3A_asdf">asDf</code></td>
<td>
<p>boolean, return d.f. instead of list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>References</h3>

<p>Watanabe K, Yahara T, Denda T, Kosuge K. 1999. Chromosomal
evolution in the genus Brachyscome (Asteraceae, Astereae): Statistical tests
regarding correlation between changes in karyotype and habit using
phylogenetic information. Journal of Plant Research 112: 145-161.
10.1007/PL00013869
</p>
<p>A2: Romero-Zarco. 1986. A New Method for Estimating Karyotype
Asymmetry. Taxon Vol. 35, No. 3  pp. 526-530
</p>
<p>Paszko B. 2006. A critical review and a new proposal of
karyotype asymmetry
indices. Plant Syst Evol 258:39-48.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chrbasicdatamono">chrbasicdatamono</a></code>
</p>
<p><code><a href="#topic+chrbasicdatamono">chrbasicdatamono</a></code>
</p>
<p><code><a href="#topic+chrbasicdataHolo">chrbasicdataHolo</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>asymmetry(dfOfChrSize)
myAlist &lt;- asymmetry(bigdfOfChrSize)
as.data.frame(myAlist)
asymmetryA2(dfOfChrSize)
as.data.frame(asymmetryA2(bigdfOfChrSize))
asymmetryA2(dfChrSizeHolo)
as.data.frame(asymmetryA2(bigdfChrSizeHolo))
</code></pre>

<hr>
<h2 id='checkArmHolocenError'>checkArmHolocenError
This is an internal function that makes list of d.fs</h2><span id='topic+checkArmHolocenError'></span>

<h3>Description</h3>

<p>checkArmHolocenError
This is an internal function that makes list of d.fs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkArmHolocenError(listOfdfArmGISHInternal, CenTypeNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkArmHolocenError_+3A_listofdfarmgishinternal">listOfdfArmGISHInternal</code></td>
<td>
<p>df list</p>
</td></tr>
<tr><td><code id="checkArmHolocenError_+3A_centypenames">CenTypeNames</code></td>
<td>
<p>character vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='checkNameChrDfMarks'>Function to check names of OTUs (species) among data.frames</h2><span id='topic+checkNameChrDfMarks'></span>

<h3>Description</h3>

<p>This function reads two data.frames the one with chromosome sizes, and one df
with data of marks (sites) both have to have the column <code>chrName</code> and if
several species, <code>OTU</code>. The functions returns a filtered list with 1.
data.frame of
chrSizes,
2. data.frame of marks, 3. mark names 4. max mark size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkNameChrDfMarks(listChrSize, listOfdfMarks)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkNameChrDfMarks_+3A_listchrsize">listChrSize</code></td>
<td>
<p>list of data.frames, with columns: OTU (optional),
chrName, for chr.
with cen.: shortArmSize, longArmSize, for holoc.: chrSize</p>
</td></tr>
<tr><td><code id="checkNameChrDfMarks_+3A_listofdfmarks">listOfdfMarks</code></td>
<td>
<p>list of data.frames of marks (sites): OTU (opt /
mandat. if in
dfChrSize), chrName markName markSize, for chr. with cen.:  chrRegion (p,q)
markDistCen, for holoc: markPos</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='chrbasicdataHolo'>Chr. basic data Holo.</h2><span id='topic+chrbasicdataHolo'></span><span id='topic+dfChrSizeHolo'></span><span id='topic+bigdfChrSizeHolo'></span><span id='topic+parentalAndHybHoloChrSize'></span><span id='topic+bigdfOfChrSize3Mb'></span>

<h3>Description</h3>

<p>dfChrSizeHolo: Example data for holocentrics for 1 species
</p>
<p>bigdfChrSizeHolo: Example data for holocentrics for several
species, OTU
</p>
<p>parentalAndHybHoloChrSize: Example data for holocentrics for
several species, OTU
</p>
<p>bigdfOfChrSize3Mb: Example data in Mb without chr. arms for
three species, OTU
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfChrSizeHolo

bigdfChrSizeHolo

parentalAndHybHoloChrSize

bigdfOfChrSize3Mb
</code></pre>


<h3>Format</h3>

<p>data.frame with columns:
</p>

<dl>
<dt>OTU</dt><dd><p>grouping OTU (species), optional if only one OTU</p>
</dd>
<dt>chrName</dt><dd><p>name of chromosome</p>
</dd>
<dt>chrSize</dt><dd><p>size of chromosome, micrometers or Mb</p>
</dd>
<dt>group</dt><dd><p>chromosome group, optional</p>
</dd>
<dt>chrNameUp</dt><dd><p>optional name over kar.</p>
</dd>
<dt>Mbp</dt><dd><p>optional name to show size in Mbp, use only when chrSize is not in Mbp</p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 12 rows and 3 columns.
</p>
<p>An object of class <code>data.frame</code> with 13 rows and 3 columns.
</p>
<p>An object of class <code>data.frame</code> with 14 rows and 3 columns.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+asymmetryA2">asymmetryA2</a></code>
</p>
<p><code><a href="#topic+plotIdiograms">plotIdiograms</a></code>
</p>
<p><code><a href="#topic+markdataholo">markdataholo</a></code>
</p>

<hr>
<h2 id='chrbasicdatamono'>Chr. basic data Monocen.</h2><span id='topic+chrbasicdatamono'></span><span id='topic+dfOfChrSize'></span><span id='topic+bigdfOfChrSize'></span><span id='topic+humChr'></span><span id='topic+allChrSizeSample'></span><span id='topic+parentalAndHybChrSize'></span><span id='topic+traspadf'></span>

<h3>Description</h3>

<p>dfOfChrSize: Example data for monocentrics
</p>
<p>bigdfOfChrSize: Example data for monocentrics for several
species, OTU
</p>
<p>humChr: Data for human karyotype, measured from Adler (1994)
</p>
<p>allChrSizeSample: Example data for monocentrics for several
species, OTU
</p>
<p>parentalAndHybChrSize: Example data for monocentrics for GISH
</p>
<p>traspadf: Example data for Tradescantia (Rhoeo) spathacea
(Golczyk et al. 2005)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfOfChrSize

bigdfOfChrSize

humChr

allChrSizeSample

parentalAndHybChrSize

traspadf
</code></pre>


<h3>Format</h3>

<p>data.frame with columns:
</p>

<dl>
<dt>OTU</dt><dd><p>OTU, species, optional if only one OTU (species)</p>
</dd>
<dt>chrName</dt><dd><p>name of chromosome</p>
</dd>
<dt>shortArmSize</dt><dd><p>size of short arm, micrometers </p>
</dd>
<dt>longArmSize</dt><dd><p>size of long arm, micrometers </p>
</dd>
<dt>group</dt><dd><p>chr group, optional</p>
</dd>
<dt>chrNameUp</dt><dd><p>optional name over kar.</p>
</dd>
<dt>Mbp</dt><dd><p>optional name to show size in Mbp, use only when shortArmSize is not in Mbp</p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 50 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 24 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 75 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 13 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 12 rows and 4 columns.
</p>


<h3>Source</h3>

<p><a href="http://www.pathology.washington.edu/research/cytopages/idiograms/human/">http://www.pathology.washington.edu/research/cytopages/idiograms/human/</a>
</p>


<h3>References</h3>

<p>Adler 1994. Idiogram Album.
<a href="http://www.pathology.washington.edu/research/cytopages/idiograms/human/">http://www.pathology.washington.edu/research/cytopages/idiograms/human/</a>
</p>
<p>Golczyk H, Hasterok R, Joachimiak AJ (2005) FISH-aimed
karyotyping and
characterization of Renner complexes in permanent heterozygote Rhoeo
spathacea. Genome
48:145-153.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotIdiograms">plotIdiograms</a></code>
</p>
<p><code><a href="#topic+armRatioCI">armRatioCI</a></code>
</p>
<p><code><a href="#topic+asymmetry">asymmetry</a></code>
</p>
<p><code><a href="#topic+markposDFs">markposDFs</a></code>
</p>

<hr>
<h2 id='citrusSize'>FUNCTIONS: citrusSize, citrusMarkPos, markOverCMA</h2><span id='topic+citrusSize'></span><span id='topic+citrusMarkPos'></span><span id='topic+markOverCMA'></span>

<h3>Description</h3>

<p>Helper function to create data.frames with
chr. size and mark size data for Citrus
based on categories in Carvalho et al. (2005)
</p>
<p>Special behaviour while plotting:
normally you will get chr. names as: B_1, B_2, etc.
to remove _*, use <code>chrIdPatternRem='_.*'</code> in
<code>plotIdiograms</code>. However, for FL+ and FL0,
this conversion is automatic. So, in plot you will
never see FL0_1, FL0_2, for example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'> citrusSize( A = 0, B = 0, C = 0, D = 0, E = 0, F = 0, FL = 0, FL0 = 0,
 G = 0, shortArm = 1.2, longArm = 1.7, shortArmFL = 1.3, longArmFL = 1.8,
 OTU = "OTU 1" 
)

citrusMarkPos( chrSizeDf, mSizePter = 0.25, mSizeQter = 0.35, mSizePprox = 0.35,
 mOther = 0.25, markName = "CMA" 
)

markOverCMA(
citrusMarkPosDF,
chrType = "B",
chrName,
chrRegion = "p",
markName = "45S",
shrinkMark = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="citrusSize_+3A_a">A</code></td>
<td>
<p>number of A to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_b">B</code></td>
<td>
<p>number of B to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_c">C</code></td>
<td>
<p>number of C to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_d">D</code></td>
<td>
<p>number of D to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_e">E</code></td>
<td>
<p>number of E to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_f">F</code></td>
<td>
<p>number of F to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_fl">FL</code></td>
<td>
<p>number of FL+ to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_fl0">FL0</code></td>
<td>
<p>number of FL0 to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_g">G</code></td>
<td>
<p>number of G to calculate (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_shortarm">shortArm</code></td>
<td>
<p>for A to G (not FL) (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_longarm">longArm</code></td>
<td>
<p>for A to G (not FL) (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_shortarmfl">shortArmFL</code></td>
<td>
<p>for FL (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_longarmfl">longArmFL</code></td>
<td>
<p>for FL (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_otu">OTU</code></td>
<td>
<p>name of species (citrusSize)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_chrsizedf">chrSizeDf</code></td>
<td>
<p>data.frame created with <code>citrusSize</code> function (citrusMarkPos)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_msizepter">mSizePter</code></td>
<td>
<p>numeric, default size for P(short) ter (terminal) bands. <code>0.25</code> (default) (citrusMarkPos)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_msizeqter">mSizeQter</code></td>
<td>
<p>numeric, default size for Q(long) ter (terminal) bands. <code>0.35</code> (default) (citrusMarkPos)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_msizepprox">mSizePprox</code></td>
<td>
<p>numeric, default size for P prox (proximal) bands. <code>0.35</code> (default) (citrusMarkPos)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_mother">mOther</code></td>
<td>
<p>numeric, default size for other bands. <code>0.25</code> (default) (citrusMarkPos)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_markname">markName</code></td>
<td>
<p>character, default name of mark <code>"CMA"</code>, or <code>"45S"</code>, respectively. (citrusMarkPos,markOverCMA)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_citrusmarkposdf">citrusMarkPosDF</code></td>
<td>
<p>data.frame, with CMA marks (markOverCMA)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_chrtype">chrType</code></td>
<td>
<p>character, defaults to &quot;B&quot;, chr. type to duplicate mark (markOverCMA)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_chrname">chrName</code></td>
<td>
<p>character, defaults to &quot;B&quot;, chr. name(s) to duplicate mark (markOverCMA)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_chrregion">chrRegion</code></td>
<td>
<p>character, arm, defaults to &quot;p&quot;. for mark duplication (markOverCMA)</p>
</td></tr>
<tr><td><code id="citrusSize_+3A_shrinkmark">shrinkMark</code></td>
<td>
<p>boolean, shrink new mark to be created (markOverCMA)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>
<p>data.frame
</p>
<p>data.frame
</p>


<h3>References</h3>

<p>Carvalho, R., Soares Filho, W. S., Brasileiro-Vidal, A. C., &amp; Guerra, M. (2005).
The relationships among lemons, limes and citron: A chromosomal comparison.
Cytogenetic and Genome Research, 109(1-3), 276-282. https://doi.org/10.1159/000082410
</p>


<h3>Examples</h3>

<pre><code class='language-R'>citrusSizeDF &lt;- citrusSize(B = 1, D = 11, F = 4, FL0 = 2, OTU = "C. jambhiri")
suppressMessages(
plotIdiograms(citrusSizeDF,
indexIdTextSize = .4, # font size
rulerNumberSize = .4, # font size
rulerTitleSize = .4, # font size
rulerPos = -.5, # ruler pos.
xPosRulerTitle = 1.5, # ruler title pos.
orderChr = "original" # order of chr. as in d.f.
)
)
citrusSizeDF2 &lt;- citrusSize(
B = 2, D = 10, F = 4, FL0 = 1,
FL = 1, # equivalent to FL+
OTU = "C. limettioides"
)


suppressMessages(
plotIdiograms(citrusSizeDF2, # FL^NA error corrected in 1.15.4
indexIdTextSize = .4, # font size
rulerNumberSize = .4, # font size
rulerTitleSize = .4, # font size
rulerPos = -.5, # ruler pos.
xPosRulerTitle = 1.5, # ruler title pos.
orderChr = "original" # order of chr. as in d.f.
)
)

citrusMarkPosDF &lt;- citrusMarkPos(citrusSizeDF)
suppressMessages(
plotIdiograms(
dfChrSize = citrusSizeDF, # chr. size data.frame
dfMarkPos = citrusMarkPosDF, # mark position data.frame (inc. cen.)
ruler = FALSE, # remove
chrIndex = FALSE, # remove
morpho = FALSE, # remove
karIndex = FALSE, # remove
indexIdTextSize = .4, # font size
xlimRightMod = 4, # xlim mod.
orderChr = "original", # order chr. as in d.f.
chrColor = "blue", # chr. color
legendHeight = 3 # legend item height
)
)
citrusMarkPosDF45S &lt;- markOverCMA(citrusMarkPosDF, chrType = "B", chrRegion = "p", markName = "45S")
suppressMessages(
plotIdiograms(
dfChrSize = citrusSizeDF, # chr. size data.frame
dfMarkPos = citrusMarkPosDF45S, # mark position data.frame (inc. cen.)
ruler = FALSE, # remove ruler
chrIndex = FALSE, # remove index
morpho = FALSE, # remove morphol.
karIndex = FALSE, # remove
indexIdTextSize = .4, # font size chr.
xlimRightMod = 4, # modify xlim
orderChr = "original", # as in d.f.
chrColor = "blue",
legendHeight = 5, # height of legend item
colorBorderMark = "black", # mark border color
OTUfont = 3 # italics
)
)
</code></pre>

<hr>
<h2 id='cMPlotMark'>This is an internal function that eliminates factors</h2><span id='topic+cMPlotMark'></span>

<h3>Description</h3>

<p>It returns a data.frames
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cMPlotMark(
  bannedMarkName,
  xMark,
  yMark,
  y,
  x,
  dfMarkColorInt,
  listOfdfMarkPoscM,
  lwd.cM,
  circularPlot,
  radius,
  circleCenter,
  circleCenterY,
  separFactor,
  markLabelSize,
  pattern,
  n,
  labelSpacing,
  chrWidth,
  ylistTransChr,
  rotation,
  labelOutwards,
  alpha_val
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cMPlotMark_+3A_xmark">xMark</code></td>
<td>
<p>x component of polygon</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_ymark">yMark</code></td>
<td>
<p>y component of polygon</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_y">y</code></td>
<td>
<p>y of chr</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_dfmarkcolorint">dfMarkColorInt</code></td>
<td>
<p>colors for marks</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_listofdfmarkposcm">listOfdfMarkPoscM</code></td>
<td>
<p>list of df. of mark pos.</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_lwd.cm">lwd.cM</code></td>
<td>
<p>thick of border line</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_radius">radius</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_circlecenter">circleCenter</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_circlecentery">circleCenterY</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_separfactor">separFactor</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_pattern">pattern</code></td>
<td>
<p>character, regex</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_n">n</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_labelspacing">labelSpacing</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_chrwidth">chrWidth</code></td>
<td>
<p>numeric, chr. width</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_ylisttranschr">ylistTransChr</code></td>
<td>
<p>list</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_rotation">rotation</code></td>
<td>
<p>rotation</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_labeloutwards">labelOutwards</code></td>
<td>
<p>srt</p>
</td></tr>
<tr><td><code id="cMPlotMark_+3A_alpha_val">alpha_val</code></td>
<td>
<p>numeric</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='customCeiling'>Function to do ceiling at different levels</h2><span id='topic+customCeiling'></span>

<h3>Description</h3>

<p>This function reads number to round
</p>


<h3>Usage</h3>

<pre><code class='language-R'>customCeiling(x, customDecimals = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="customCeiling_+3A_x">x</code></td>
<td>
<p>number to round</p>
</td></tr>
<tr><td><code id="customCeiling_+3A_customdecimals">customDecimals</code></td>
<td>
<p>desired decimal positions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric
</p>

<hr>
<h2 id='dfMarkStyle'>Mark characteristics</h2><span id='topic+dfMarkStyle'></span><span id='topic+dfMarkColor'></span><span id='topic+humMarkColor'></span><span id='topic+mydfMaColor'></span><span id='topic+dfMarkColor5S25S'></span><span id='topic+dfMarkColorIn'></span>

<h3>Description</h3>

<p>style column does not apply to cen. marks, only color.
</p>
<p>dfMarkColor: Example General data for marks NOT position
</p>
<p>humMarkColor: human bands' characteristics, from Adler (1994)
</p>
<p>mydfMaColor: mark characteristics used in vignette of phylogeny
</p>
<p>dfMarkColor5S25S: mark characteristics used in circular plot vignette
</p>
<p>dfMarkColorIn: mark characteristics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfMarkColor

humMarkColor

mydfMaColor

dfMarkColor5S25S

dfMarkColorIn
</code></pre>


<h3>Format</h3>

<p>dfMarkColor a data.frame with columns:
</p>

<dl>
<dt>markName</dt><dd><p>name of mark</p>
</dd>
<dt>markColor</dt><dd><p>use R colors</p>
</dd>
<dt>style</dt><dd><p>character, use square or dots, optional</p>
</dd>
<dt>protruding</dt><dd><p>numeric, modifies aspect of cM/cMLeft marks, see
parameter
<code>protruding</code> in <code><a href="#topic+plotIdiograms">plotIdiograms</a></code>, optional</p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 395 rows and 3 columns.
</p>
<p>An object of class <code>data.frame</code> with 6 rows and 3 columns.
</p>
<p>An object of class <code>data.frame</code> with 2 rows and 3 columns.
</p>
<p>An object of class <code>data.frame</code> with 5 rows and 3 columns.
</p>


<h3>Source</h3>

<p><a href="http://www.pathology.washington.edu/research/cytopages/idiograms/human/">http://www.pathology.washington.edu/research/cytopages/idiograms/human/</a>
</p>


<h3>References</h3>

<p>Adler 1994. Idiogram Album. URL:
<a href="http://www.pathology.washington.edu/research/cytopages/idiograms/human/">http://www.pathology.washington.edu/research/cytopages/idiograms/human/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plotIdiograms">plotIdiograms</a></code>
</p>
<p><code><a href="#topic+markposDFs">markposDFs</a></code>
</p>
<p><code><a href="#topic+markdataholo">markdataholo</a></code>
</p>

<hr>
<h2 id='dfToListColumn'>dfToListColumn
This is an internal function that makes list of d.fs</h2><span id='topic+dfToListColumn'></span>

<h3>Description</h3>

<p>dfToListColumn
This is an internal function that makes list of d.fs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dfToListColumn(dfLookColumn, columnToSplit = "OTU")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dfToListColumn_+3A_dflookcolumn">dfLookColumn</code></td>
<td>
<p>df with or without Column to list</p>
</td></tr>
<tr><td><code id="dfToListColumn_+3A_columntosplit">columnToSplit</code></td>
<td>
<p>character, column name to split</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='filterExtraOTU'>filterExtraOTU
This is an internal function that filters list of d.f. of marks</h2><span id='topic+filterExtraOTU'></span>

<h3>Description</h3>

<p>It returns rulers (axis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterExtraOTU(listOfdfChromSize, listOfdfMarkPosCenType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterExtraOTU_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>list of all d.f. of chr.</p>
</td></tr>
<tr><td><code id="filterExtraOTU_+3A_listofdfmarkposcentype">listOfdfMarkPosCenType</code></td>
<td>
<p>only d.f. of this cen. type</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='fixChrNameDupDF'>This function fixes duplicated names when no marks</h2><span id='topic+fixChrNameDupDF'></span>

<h3>Description</h3>

<p>It returns a list of data.frames
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fixChrNameDupDF(list1dfChromSize, mybooleanChrName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fixChrNameDupDF_+3A_list1dfchromsize">list1dfChromSize</code></td>
<td>
<p>list of 1 df of chr size</p>
</td></tr>
<tr><td><code id="fixChrNameDupDF_+3A_mybooleanchrname">mybooleanChrName</code></td>
<td>
<p>test for absence of marks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='genBankReadIF'>FUNCTION genBankReadIF</h2><span id='topic+genBankReadIF'></span>

<h3>Description</h3>

<p>genBankReadIF: creates a list of data.frames from a genbank
format file or a
rentrez object
</p>
<p>Requires tidyr
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genBankReadIF(filename.gb, forbiddenFields = "translation")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genBankReadIF_+3A_filename.gb">filename.gb</code></td>
<td>
<p>name of file to read, downloaded from genBank, or,
object from <code>rentrez::entrez_fetch( db="nuccore",</code> <code> id="theID",</code> <code> rettype="gbwithparts",</code> <code> retmode = "text" )</code></p>
</td></tr>
<tr><td><code id="genBankReadIF_+3A_forbiddenfields">forbiddenFields</code></td>
<td>
<p>names of field of feature (CDS, gene) to ignore, separated by |. Defaults to <code>"translation"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='insertInList'>This is an internal function that inserts in vector or list</h2><span id='topic+insertInList'></span>

<h3>Description</h3>

<p>It returns list or vector
</p>


<h3>Usage</h3>

<pre><code class='language-R'>insertInList(list, pos, elems)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="insertInList_+3A_list">list</code></td>
<td>
<p>vector or list</p>
</td></tr>
<tr><td><code id="insertInList_+3A_pos">pos</code></td>
<td>
<p>index to insert in</p>
</td></tr>
<tr><td><code id="insertInList_+3A_elems">elems</code></td>
<td>
<p>elements to be inserted</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list or vector
</p>

<hr>
<h2 id='make.uniqueIF'>FUNCTION make.uniqueIF</h2><span id='topic+make.uniqueIF'></span>

<h3>Description</h3>

<p>make.uniqueIF: produces unique strings from duplicated
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make.uniqueIF(string, sep = "_", letter = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make.uniqueIF_+3A_string">string</code></td>
<td>
<p>name of char. vector</p>
</td></tr>
<tr><td><code id="make.uniqueIF_+3A_sep">sep</code></td>
<td>
<p>separator</p>
</td></tr>
<tr><td><code id="make.uniqueIF_+3A_letter">letter</code></td>
<td>
<p>boolean, use numbers when <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>make.uniqueIF(c(1, 1, 2, 2))

</code></pre>

<hr>
<h2 id='makedfMarkColor'>Function to automatically make a data.frame of Marks' characterisitcs</h2><span id='topic+makedfMarkColor'></span>

<h3>Description</h3>

<p>This function reads character vector with marks names
</p>
<p>It returns a data.frame with color and style for each mark
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makedfMarkColor(
  dfMarkColor,
  markNames,
  colorstoremove = NA,
  defaultStyleMark = "square",
  reserveDF
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makedfMarkColor_+3A_dfmarkcolor">dfMarkColor</code></td>
<td>
<p>d.f of marks to use</p>
</td></tr>
<tr><td><code id="makedfMarkColor_+3A_marknames">markNames</code></td>
<td>
<p>names of marks</p>
</td></tr>
<tr><td><code id="makedfMarkColor_+3A_colorstoremove">colorstoremove</code></td>
<td>
<p>remove this colors from the result</p>
</td></tr>
<tr><td><code id="makedfMarkColor_+3A_defaultstylemark">defaultStyleMark</code></td>
<td>
<p>character vector with default style <code>"square"</code>
<code>"cM"</code>
<code>"dots"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='makedfMarkColorMycolors'>FUNCTION to make a data.frame of Marks' characterisitcs</h2><span id='topic+makedfMarkColorMycolors'></span>

<h3>Description</h3>

<p>This function reads character vector with marks names, a
character vector of
prohibited colors and a custom list of colors
to be associated with the mark names
</p>
<p>It returns a data.frame with color and style for each mark
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makedfMarkColorMycolors(
  markNames,
  mycolors,
  colorstoremove = NULL,
  defaultStyleMark = "square"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makedfMarkColorMycolors_+3A_marknames">markNames</code></td>
<td>
<p>names of marks</p>
</td></tr>
<tr><td><code id="makedfMarkColorMycolors_+3A_mycolors">mycolors</code></td>
<td>
<p>character vector of names of colors</p>
</td></tr>
<tr><td><code id="makedfMarkColorMycolors_+3A_colorstoremove">colorstoremove</code></td>
<td>
<p>character vector of colors to remove from mycolors. Default <code>NULL</code></p>
</td></tr>
<tr><td><code id="makedfMarkColorMycolors_+3A_defaultstylemark">defaultStyleMark</code></td>
<td>
<p>character vector with default style <code>"square"</code>. Other options:  <code>"squareLeft"</code>, <code>"cM"</code>
<code>"cMLeft"</code>, <code>"dots"</code>, <code>"upArrow"</code>, <code>"downArrow"</code>, <code>"cenStyle"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='makeNumCols'>These are internal functions that eliminate factors</h2><span id='topic+makeNumCols'></span>

<h3>Description</h3>

<p>It returns a data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeNumCols(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeNumCols_+3A_df">df</code></td>
<td>
<p>data.frame</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='makeVectorNames'>makeVectorNames
This is an internal function that makes a vector of names</h2><span id='topic+makeVectorNames'></span>

<h3>Description</h3>

<p>makeVectorNames
This is an internal function that makes a vector of names
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeVectorNames(namedList, attr, selector)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeVectorNames_+3A_namedlist">namedList</code></td>
<td>
<p>original list</p>
</td></tr>
<tr><td><code id="makeVectorNames_+3A_attr">attr</code></td>
<td>
<p>attribute</p>
</td></tr>
<tr><td><code id="makeVectorNames_+3A_selector">selector</code></td>
<td>
<p>value of attribute</p>
</td></tr>
</table>


<h3>Value</h3>

<p>character vector
</p>

<hr>
<h2 id='mapGGChr'>FUNCTIONS mapGGChr and mapGGChrMark (for ggplot)</h2><span id='topic+mapGGChr'></span><span id='topic+mapGGChrMark'></span>

<h3>Description</h3>

<p>Currently works for holocentrics when only one OTU. See vignette.
</p>
<p>mapGGChr: reads a data.frame and produces coordinates for ggplot of chr.
</p>
<p>mapGGChrMark: reads data.frames and produces data.frames for ggplot of chr. and marks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapGGChr(dfChrSize, chrSpacing = 0.5, squareness = 4, n = 50)

mapGGChrMark(dfChrSize, dfMarkPos, chrSpacing = 0.5, squareness = 4, n = 50)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapGGChr_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>character, name of data.frame with columns: chrSize</p>
</td></tr>
<tr><td><code id="mapGGChr_+3A_chrspacing">chrSpacing</code></td>
<td>
<p>numeric, <code>1 - chrSpacing</code> will be the width of chr.</p>
</td></tr>
<tr><td><code id="mapGGChr_+3A_squareness">squareness</code></td>
<td>
<p>numeric, squareness</p>
</td></tr>
<tr><td><code id="mapGGChr_+3A_n">n</code></td>
<td>
<p>numeric, vertices number for rounded portions</p>
</td></tr>
<tr><td><code id="mapGGChr_+3A_dfmarkpos">dfMarkPos</code></td>
<td>
<p>(<code>mapGGChrMark</code>) name of data.frame of marks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>
<p>list
</p>

<hr>
<h2 id='mapXY'>mapXY
This is an internal function that creates coords for chr. or arms or marks</h2><span id='topic+mapXY'></span>

<h3>Description</h3>

<p>mapXY
This is an internal function that creates coords for chr. or arms or marks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapXY(
  start,
  end,
  y,
  yMod,
  x,
  yfactor,
  r2,
  pts_1,
  pts_2,
  pts_3,
  pts_4,
  chrt = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapXY_+3A_start">start</code></td>
<td>
<p>1st index</p>
</td></tr>
<tr><td><code id="mapXY_+3A_end">end</code></td>
<td>
<p>last index</p>
</td></tr>
<tr><td><code id="mapXY_+3A_y">y</code></td>
<td>
<p>y coords</p>
</td></tr>
<tr><td><code id="mapXY_+3A_ymod">yMod</code></td>
<td>
<p>y coords modified because of squareness</p>
</td></tr>
<tr><td><code id="mapXY_+3A_x">x</code></td>
<td>
<p>x coords</p>
</td></tr>
<tr><td><code id="mapXY_+3A_yfactor">yfactor</code></td>
<td>
<p>y x factor</p>
</td></tr>
<tr><td><code id="mapXY_+3A_r2">r2</code></td>
<td>
<p>radius</p>
</td></tr>
<tr><td><code id="mapXY_+3A_pts_1">pts_1</code></td>
<td>
<p>points of squareness</p>
</td></tr>
<tr><td><code id="mapXY_+3A_pts_2">pts_2</code></td>
<td>
<p>points of squareness</p>
</td></tr>
<tr><td><code id="mapXY_+3A_pts_3">pts_3</code></td>
<td>
<p>points of squareness</p>
</td></tr>
<tr><td><code id="mapXY_+3A_pts_4">pts_4</code></td>
<td>
<p>points of squareness</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='mapXYCenMimic'>mapXYCenMimic
This is an internal function that creates coords for chr. or arms or marks</h2><span id='topic+mapXYCenMimic'></span>

<h3>Description</h3>

<p>mapXYCenMimic
This is an internal function that creates coords for chr. or arms or marks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mapXYCenMimic(start, end, y, x, yfactor, r2, pts, cenFormat = "triangle")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mapXYCenMimic_+3A_start">start</code></td>
<td>
<p>1st index</p>
</td></tr>
<tr><td><code id="mapXYCenMimic_+3A_end">end</code></td>
<td>
<p>last index</p>
</td></tr>
<tr><td><code id="mapXYCenMimic_+3A_y">y</code></td>
<td>
<p>y coords</p>
</td></tr>
<tr><td><code id="mapXYCenMimic_+3A_yfactor">yfactor</code></td>
<td>
<p>y x factor</p>
</td></tr>
<tr><td><code id="mapXYCenMimic_+3A_r2">r2</code></td>
<td>
<p>r2ius</p>
</td></tr>
<tr><td><code id="mapXYCenMimic_+3A_pts">pts</code></td>
<td>
<p>points of squareness</p>
</td></tr>
<tr><td><code id="mapXYCenMimic_+3A_cenformat">cenFormat</code></td>
<td>
<p>boolean</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='markdataholo'>Mark Positional data - Holocen.</h2><span id='topic+markdataholo'></span><span id='topic+bigdfMarkPosHolo'></span><span id='topic+dfMarkPosHolo'></span><span id='topic+dfAlloParentMarksHolo'></span><span id='topic+bigdfOfMarks3Mb'></span><span id='topic+bigdfMarkPosHolo2'></span>

<h3>Description</h3>

<p>When several OTUs, some can be monocen. and others holocen.
Marks distance for
monocen. are measured from cen. and for
holocen. from top or bottom depending on <code>param</code> <code>origin</code>. See
vignettes.
</p>
<p>bigdfMarkPosHolo: Example data for mark position of
holocentrics with column OTU
</p>
<p>dfMarkPosHolo: Example data for mark position of holocentrics
</p>
<p>dfAlloParentMarksHolo: Example data for mark position of GISH
</p>
<p>bigdfOfMarks3Mb: Example data for mark position in Mb
</p>
<p>bigdfMarkPosHolo2: Example data for mark position
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bigdfMarkPosHolo

dfMarkPosHolo

dfAlloParentMarksHolo

bigdfOfMarks3Mb

bigdfMarkPosHolo2
</code></pre>


<h3>Format</h3>

<p>data.frame with columns:
</p>

<dl>
<dt>OTU</dt><dd><p>OTU, species, optional</p>
</dd>
<dt>chrName</dt><dd><p>name of chromosome</p>
</dd>
<dt>markName</dt><dd><p>name of mark</p>
</dd>
<dt>markPos</dt><dd><p>position from bottom or top (see parameter <code>origin</code>
in plotIdiograms)</p>
</dd>
<dt>markSize</dt><dd><p>size of mark in micrometers or Mb</p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 6 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 13 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 12 rows and 5 columns.
</p>
<p>An object of class <code>data.frame</code> with 18 rows and 6 columns.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+markposDFs">markposDFs</a></code>
</p>
<p><code><a href="#topic+plotIdiograms">plotIdiograms</a></code>
</p>
<p><code><a href="#topic+chrbasicdataHolo">chrbasicdataHolo</a></code>
</p>

<hr>
<h2 id='markDistCenGISHfix'>markDistCenGISHfix
This is an internal function that fixes markDistCen when GISH</h2><span id='topic+markDistCenGISHfix'></span>

<h3>Description</h3>

<p>markDistCenGISHfix
This is an internal function that fixes markDistCen when GISH
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markDistCenGISHfix(
  dfArmGISHInternalMonocen,
  dfChrSizeInternal,
  columnArmSize,
  markDistType,
  listOfdfChromSize,
  addR2 = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="markDistCenGISHfix_+3A_dfarmgishinternalmonocen">dfArmGISHInternalMonocen</code></td>
<td>
<p>df of GISH OF ARM</p>
</td></tr>
<tr><td><code id="markDistCenGISHfix_+3A_dfchrsizeinternal">dfChrSizeInternal</code></td>
<td>
<p>d.f. of chr. size</p>
</td></tr>
<tr><td><code id="markDistCenGISHfix_+3A_columnarmsize">columnArmSize</code></td>
<td>
<p>column rel. to arm.</p>
</td></tr>
<tr><td><code id="markDistCenGISHfix_+3A_markdisttype">markDistType</code></td>
<td>
<p>character</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='markposDFs'>Mark Positional data - monocentrics</h2><span id='topic+markposDFs'></span><span id='topic+bigdfOfMarks'></span><span id='topic+dfOfMarks'></span><span id='topic+dfOfMarks2'></span><span id='topic+humMarkPos'></span><span id='topic+allMarksSample'></span><span id='topic+dfAlloParentMarks'></span><span id='topic+traspaMarks'></span>

<h3>Description</h3>

<p>When several OTUs, some can be monocen. and others holocen.
Marks distance for
monocen. are measured from cen. and for
holocen. from top or bottom depending on <code>param</code> <code>origin</code>. See
vignettes.
</p>
<p>bigdfOfMarks: Example data for mark position with column OTU
</p>
<p>dfOfMarks: Example data for marks' position
</p>
<p>dfOfMarks2: Marks' position including cen. marks
</p>
<p>humMarkPos: human karyotype bands' (marks) positions, measured
from Adler (1994)
</p>
<p>allMarksSample: Example data for marks' position
</p>
<p>dfAlloParentMarks: Example data for mark position of GISH of
monocen.
</p>
<p>traspaMarks: T. spathacea (Rhoeo) marks' positions, from
Golczyk et al. (2005)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bigdfOfMarks

dfOfMarks

dfOfMarks2

humMarkPos

allMarksSample

dfAlloParentMarks

traspaMarks
</code></pre>


<h3>Format</h3>

<p>bigdfOfMarks a data.frame with columns:
</p>

<dl>
<dt>OTU</dt><dd><p>OTU, species, mandatory if in dfChrSize</p>
</dd>
<dt>chrName</dt><dd><p>name of chromosome</p>
</dd>
<dt>markName</dt><dd><p>name of mark</p>
</dd>
<dt>chrRegion</dt><dd><p>use p for short arm, q for long arm, and cen for
centromeric</p>
</dd>
<dt>markDistCen</dt><dd><p>distance of mark to centromere (not for cen)</p>
</dd>
<dt>markSize</dt><dd><p>size of mark (not for cen)</p>
</dd>
</dl>

<p>An object of class <code>data.frame</code> with 4 rows and 5 columns.
</p>
<p>An object of class <code>data.frame</code> with 6 rows and 5 columns.
</p>
<p>An object of class <code>data.frame</code> with 395 rows and 5 columns.
</p>
<p>An object of class <code>data.frame</code> with 43 rows and 6 columns.
</p>
<p>An object of class <code>data.frame</code> with 16 rows and 4 columns.
</p>
<p>An object of class <code>data.frame</code> with 18 rows and 7 columns.
</p>


<h3>Source</h3>

<p><a href="http://www.pathology.washington.edu/research/cytopages/idiograms/human/">Washington U</a>
</p>


<h3>References</h3>

<p>Adler 1994. Idiogram Album. URL:
<a href="http://www.pathology.washington.edu/research/cytopages/idiograms/human/">Washington U.</a>
</p>
<p>Golczyk H, Hasterok R, Joachimiak AJ (2005) FISH-aimed
karyotyping and
characterization of Renner complexes in permanent heterozygote Rhoeo
spathacea. Genome
48:145-153.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+markdataholo">markdataholo</a></code>
</p>
<p><code><a href="#topic+plotIdiograms">plotIdiograms</a></code>
</p>
<p><code><a href="#topic+chrbasicdatamono">chrbasicdatamono</a></code>
</p>
<p><code><a href="#topic+dfMarkColor">dfMarkColor</a></code>
</p>

<hr>
<h2 id='mimicCenPlotMark'>mimicCenPlotMark
This is an internal function that plot marks</h2><span id='topic+mimicCenPlotMark'></span>

<h3>Description</h3>

<p>It returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mimicCenPlotMark(
  squareness,
  xMark,
  yMark,
  defCenStyleCol,
  listOfdfMarkPosCenStyle,
  chrWidth,
  specialChrWidth,
  yfactor,
  n,
  lwd.mimicCen,
  listOfdfChromSize,
  circularPlot,
  y,
  markLabelSize,
  separFactor,
  labelSpacing,
  circleCenter,
  circleCenterY,
  radius,
  ylistTransChr,
  rotation,
  labelOutwards,
  yMarkLine,
  xMarkRightLine,
  xMarkLeftLine,
  x,
  cenFormat = "triangle",
  pts
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mimicCenPlotMark_+3A_squareness">squareness</code></td>
<td>
<p>squareness of vertices &lt;</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_xmark">xMark</code></td>
<td>
<p>x component of polygon</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_ymark">yMark</code></td>
<td>
<p>yMark component of polygon</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_listofdfmarkposcenstyle">listOfdfMarkPosCenStyle</code></td>
<td>
<p>list of df. of mark pos.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_chrwidth">chrWidth</code></td>
<td>
<p>numeric, width of chr.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_specialchrwidth">specialChrWidth</code></td>
<td>
<p>numeric, width of chr.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_yfactor">yfactor</code></td>
<td>
<p>yMark distortion based on canvas proportion</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_n">n</code></td>
<td>
<p>numeric, to define vertices of rounded portions</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>chr size df list</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean TRUE for circ.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_y">y</code></td>
<td>
<p>list, y coords.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>numeric, font size</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_separfactor">separFactor</code></td>
<td>
<p>numeric, sep. kars.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_labelspacing">labelSpacing</code></td>
<td>
<p>numeric, spacing</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_circlecenter">circleCenter</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_circlecentery">circleCenterY</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_radius">radius</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_ylisttranschr">ylistTransChr</code></td>
<td>
<p>list, transf. coords.</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_rotation">rotation</code></td>
<td>
<p>rotate</p>
</td></tr>
<tr><td><code id="mimicCenPlotMark_+3A_labeloutwards">labelOutwards</code></td>
<td>
<p>srt</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='namesToColumns'>FUNCTION that modifies marks' names into columns</h2><span id='topic+namesToColumns'></span>

<h3>Description</h3>

<p>Reads a data.frame with marks' of styles
<code>downArrow,upArrow,cM,cMLeft</code>
positions. It separates names in columns, avoiding overlap when multiple
close names
</p>
<p>Exceptionally this function requires the column style in the
data.frame of marks'
positions.
</p>
<p>Returns a data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'> namesToColumns( marksDf, dfChrSize, markType = c("downArrow", "upArrow",
 "cMLeft", "cM"), amountofSpaces = 13, colNumber = 2, protruding = 0.2,
 protrudingInt = 0.5, circularPlot = TRUE, rotation = 0.5,
 defaultStyleMark = "square", orderChr = "size", halfModDown = NA,
 halfModUp = NA, rotatMod = 0 
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="namesToColumns_+3A_marksdf">marksDf</code></td>
<td>
<p>data.frame with columns: <code>markName,style,markPos</code></p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>data.frame, size of chr. Same of plot.</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_marktype">markType</code></td>
<td>
<p>character, use
<code>c("downArrow","upArrow","cM","cMLeft")</code> or a subset</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_amountofspaces">amountofSpaces</code></td>
<td>
<p>numeric, number of spaces for each column</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_colnumber">colNumber</code></td>
<td>
<p>numeric, number of columns</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_protruding">protruding</code></td>
<td>
<p>numeric, same as plot, minimal protruding for arrow
marks, equivalent to cM
protruding</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_protrudingint">protrudingInt</code></td>
<td>
<p>numeric, spacing of columns in terms of width of chr.
percent 1 = 100%.
Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean, use <code>TRUE</code> for circular plots. Use
<code>FALSE</code> otherwise</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_rotation">rotation</code></td>
<td>
<p>numeric, same as plot, anti-clockwise rotation, defaults to
<code>0.5</code> which
rotates chr. from top to -90 degrees. (-0.5*<code class="reqn">\pi</code> )</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_defaultstylemark">defaultStyleMark</code></td>
<td>
<p>character, if some data in column style missing
fill with this one.
Defaults to <code>"square"</code></p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_orderchr">orderChr</code></td>
<td>
<p>character, replaces <code>orderBySize - deprecated</code> when <code>"size"</code>, sorts chromosomes by total
length from the largest to the smallest. <code>"original"</code>: preserves d.f. order. <code>"name"</code>:
sorts alphabetically; <code>"group"</code>: sorts by group name</p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_halfmoddown">halfModDown</code></td>
<td>
<p>numeric, for circ. plots, when plotting several
chromosomes in a circular
plot, using a small value <code>0.05</code> corrects for alignment problems of
<code>downArrows,
cMLeft</code> labels. Defaults to <code>NA</code></p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_halfmodup">halfModUp</code></td>
<td>
<p>numeric, for circ. plots, when plotting several
chromosomes in a circular
plot, using a small value <code>0.05</code> corrects for alignment problems of
<code>upArrows, cM</code>
labels. Defaults to <code>NA</code></p>
</td></tr>
<tr><td><code id="namesToColumns_+3A_rotatmod">rotatMod</code></td>
<td>
<p>numeric, for circ. plots, when rotation != 0 (diff.),
corrects alignment of
labels. Defaults to <code>0</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='newOrderColumn'>newOrderColumn
This is an internal function that adds neworder column to
listOfdfChromSizeCenType</h2><span id='topic+newOrderColumn'></span>

<h3>Description</h3>

<p>It returns rulers (axis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>newOrderColumn(listOfdfChromSize, listOfdfChromSizeCenType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="newOrderColumn_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>list of all d.f. of chr.</p>
</td></tr>
<tr><td><code id="newOrderColumn_+3A_listofdfchromsizecentype">listOfdfChromSizeCenType</code></td>
<td>
<p>only d.f. of this cen. type</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>

<hr>
<h2 id='perMark'>FUNCTION perMark</h2><span id='topic+perMark'></span>

<h3>Description</h3>

<p>calculates fraction (%) of chromosome for each mark
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perMark(dfMarkPos, listOfdfChromSize, result = "list", bToRemove = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perMark_+3A_dfmarkpos">dfMarkPos</code></td>
<td>
<p>data.frame, of marks' position</p>
</td></tr>
<tr><td><code id="perMark_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>list of data.frames or data.frame of chr. sizes.</p>
</td></tr>
<tr><td><code id="perMark_+3A_result">result</code></td>
<td>
<p>character, type of return, <code>"data.frame"</code> or <code>"list"</code></p>
</td></tr>
<tr><td><code id="perMark_+3A_btoremove">bToRemove</code></td>
<td>
<p>character vector, bands to ignore</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("shinyApps", "iBoard/www/rda/monoholoCS.rda", package = "idiogramFISH"))
load(system.file("shinyApps", "iBoard/www/rda/monoholoMarks.rda", package = "idiogramFISH"))
monoholoMarks2 &lt;- fillMarkInfo(monoholoMarks, monoholoCS)
perMark(monoholoMarks2, monoholoCS, result = "data.frame")

</code></pre>

<hr>
<h2 id='plotDotMarks'>plotDotMarks
This is an internal function that plot dot marks</h2><span id='topic+plotDotMarks'></span>

<h3>Description</h3>

<p>It returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotDotMarks(
  bannedMarkName,
  xMarkCr,
  yMarkCr,
  rad,
  radX,
  colCr,
  n,
  xfactor,
  colBorderCr,
  circularPlot,
  y,
  x,
  radius,
  circleCenter,
  circleCenterY,
  separFactor,
  chrWidth,
  listOfdfMarkPosCr,
  markLabelSize,
  pattern,
  labelSpacing,
  useOneDot,
  legend,
  ylistTransChr,
  rotation,
  labelOutwards,
  dotsAsOval,
  alpha_val = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotDotMarks_+3A_xmarkcr">xMarkCr</code></td>
<td>
<p>x coords</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_ymarkcr">yMarkCr</code></td>
<td>
<p>y coords</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_rad">rad</code></td>
<td>
<p>radius</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_colcr">colCr</code></td>
<td>
<p>color</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_n">n</code></td>
<td>
<p>vertices number</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_xfactor">xfactor</code></td>
<td>
<p>aspect</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_colbordercr">colBorderCr</code></td>
<td>
<p>border color</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_y">y</code></td>
<td>
<p>y of chr.</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_x">x</code></td>
<td>
<p>x of chr.</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_radius">radius</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_circlecenter">circleCenter</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_circlecentery">circleCenterY</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_separfactor">separFactor</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_chrwidth">chrWidth</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_listofdfmarkposcr">listOfdfMarkPosCr</code></td>
<td>
<p>list of d.f.</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_pattern">pattern</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_labelspacing">labelSpacing</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_useonedot">useOneDot</code></td>
<td>
<p>boolean</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_legend">legend</code></td>
<td>
<p>character</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_ylisttranschr">ylistTransChr</code></td>
<td>
<p>list of trans. coords</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_rotation">rotation</code></td>
<td>
<p>rotation</p>
</td></tr>
<tr><td><code id="plotDotMarks_+3A_labeloutwards">labelOutwards</code></td>
<td>
<p>srt</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='plotIdiograms'>Plot idiograms of karyotypes/genomes with and without centromere</h2><span id='topic+plotIdiograms'></span>

<h3>Description</h3>

<p>This function reads a data.frame passed to <code>dfChrSize</code> with columns: <code>chrName</code> (mono/holo) and
<code>shortArmSize</code> and <code>longArmSize</code> for monocentrics or a column <code>chrSize</code> for holocentrics and produces a plot of
idiograms. If more
than one species, a column named <code>OTU</code> is needed.
</p>
<p>Optionally, it reads another data.frame passed to <code>dfMarkPos</code> with the position of
marks (sites). Examples: <code><a href="#topic+markposDFs">markposDFs</a></code>. Another data.frame for mark characteristics
can be used <code><a href="#topic+dfMarkColor">dfMarkColor</a></code> or a character vector passed to <code>mycolors</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'> plotIdiograms( dfChrSize, defaultFontFamily, revOTUs = FALSE,
 karHeight = 2, karHeiSpace = 2.5, karSepar = TRUE, amoSepar = 10,
 addMissingOTUAfter = NA, addMissingOTUBefore = NA, missOTUspacings = 0,
 moveKarHor = "", moveAllKarValueHor = 0, moveAllKarValueY = 0,
 karAnchorLeft = "", karAnchorRight = "", anchor = FALSE, anchorLineLty = 1,
 anchorText = "", anchorTextMParental, anchorTextMoveX = 0.5,
 anchorTextMoveY = 1, anchorTextMoveParenX = 0, anchorTextMoveParenY = 0,
 anchorVsizeF = 0.5, anchorHsizeF = 1, pchAnchor = 23, moveAnchorV = 0,
 moveAnchorH = 0, mkhValue = 0.5, n = 50, markN = 25, notes, leftNotes,
 leftNotesUp, notesTextSize = 1, leftNotesTextSize = 1, leftNotesUpTextSize = 1,
 notesLeft, notesPosX = 0.5, notesPosY = 0, leftNotesPosX = 0.5,
 leftNotesPosY = 0, leftNotesUpPosX = 0.5, leftNotesUpPosY = 0, noteFont = 1,
 leftNoteFont = 1, leftNoteFontUp = 1, parseTypes = TRUE, parseStr2lang = FALSE,
 propWidth = FALSE, MbThreshold = 10000, threshold = 35, MbUnit = "Mb",
 yTitle = "µm", specialyTitle = "cM", specialOTUNames = "", addOTUName = TRUE,
 OTUTextSize = 1, OTUfont, OTUfamily = "", OTUasNote = FALSE,
 OTUasLeftNote = FALSE, orderChr = "size", chrId = "original",
 classMbName = "Pm.", classcMName = "L.G.", classChrName = "Chr.",
 classChrNameUp = "Type", classGroupName = "", chrNameUp = FALSE,
 chrIdPatternRem, indexIdTextSize = 1, distTextChr = 1, groupUp = FALSE,
 groupName = TRUE, groupSepar = 0.5, chromatids = TRUE, arrowsBothChrt = TRUE,
 holocenNotAsChromatids = FALSE, excHoloFrArrToSide = FALSE, xModifier = 12,
 xModMonoHoloRate = 1, chrWidth = 0.5, chrSpacing = 0.5, specialChrWidth = 0.3,
 specialChrSpacing = 0.7, chrColor = "gray", chrBorderColor, centromereSize = 0,
 autoCenSize = TRUE, cenColor, fixCenBorder = NULL, gishCenBorder = FALSE,
 hideCenLines = 1.75, roundedCen, cenFormat = "rounded", cenFactor = 1,
 squareness = 4, lwd.chr = 0.5, lwd.cM, lwd.marks = 99, dfMarkPos,
 defaultStyleMark = "square", markDistType = "beg", protruding = 0.2,
 startPos = 0, pMarkFac = 0.25, origin = "b", efZero = 1e-05,
 cMBeginCenter = FALSE, arrowhead = 0.3, shrinkArrow = 0.3333,
 arrowheadWidthShrink = 0.1, arrowsToSide = TRUE, useOneDot = FALSE,
 dotsAsOval = FALSE, dfMarkColor, mycolors, alpha_val = 1,
 borderOfWhiteMarks = TRUE, colorBorderMark = "", lwd.mimicCen, defCenStyleCol,
 pattern = "", legend = "aside", remSimiMarkLeg = TRUE, bannedMarkName,
 bMarkNameAside = FALSE, forbiddenMark, legendWidth = 1.7, legendHeight = NA,
 markLabelSize = 1, markLabelSpacer = 1, legendYcoord = 0, markNewLine = NA,
 mylheight = 0.7, chrSize = FALSE, nsmall = 1, chrSizeMbp = FALSE, markPer = "",
 showMarkPos = FALSE, bToRemove = "", perAsFraction = FALSE, chrIndex = "both",
 morpho = "both", nameChrIndexPos = 2, karIndex = TRUE, karIndexPos = 0.5,
 ruler = TRUE, useMinorTicks = FALSE, miniTickFactor = 10, rulerPos = 0,
 ruler.tck = -0.02, rulerNumberPos = 0.5, rulerNumberSize = 1,
 collapseCen = TRUE, rulerInterval = 0, rulerIntervalcM = 0,
 rulerIntervalMb = 0, ceilingFactor = 0, xPosRulerTitle = 2.6,
 yPosRulerTitle = 0, rulerTitleSize = 1, xlimLeftMod = 1, xlimRightMod = 2,
 ylimBotMod = 0.2, ylimTopMod = 0.2, callPlot = TRUE, asp = 1,
 circularPlot = FALSE, verticalPlot = TRUE, karSpaceHor = 0, shrinkFactor = 0.9,
 separFactor = 1.5, labelSpacing = 0.7, labelOutwards = FALSE,
 chrLabelSpacing = 0.5, radius = 0.5, rotation = 0.5, circleCenter = 1,
 circleCenterY = 1, OTUlabelSpacing = 0.3, OTUsrt = 0, OTUplacing = "first",
 OTULabelSpacerx = 0, OTULabelSpacery = 0, OTUcentered = TRUE, OTUjustif = 0,
 OTUlegendHeight = NA, roundness, ... 
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotIdiograms_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>mandatory data.frame, with columns: <code>OTU</code> (optional), <code>chrName</code> (mandatory),
<code>shortArmSize</code>, <code>longArmSize</code> for monocen. or <code>chrSize</code> for holocen.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_defaultfontfamily">defaultFontFamily</code></td>
<td>
<p>character. use this as the font family. No default value.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_revotus">revOTUs</code></td>
<td>
<p>boolean, The order of species is the one in the main
data.frame, use <code>TRUE</code> to reverse</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karheight">karHeight</code></td>
<td>
<p>numeric, vertical size of karyotypes. See also  <code>karHeiSpace</code>. Defaults to <code>2</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karheispace">karHeiSpace</code></td>
<td>
<p>numeric, vertical size of karyotypes including spacing. Use with <code>karSepar=FALSE</code>. Proportional to
<code>karHeight</code>, if overlap, increase. Defautl value <code>2.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karsepar">karSepar</code></td>
<td>
<p>boolean, reduce distance among karyotypes <code>FALSE</code> = equally
sized karyotypes or <code>TRUE</code> = equally spaced karyotypes. Incompatible with <code>addMissingOTUAfter</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_amosepar">amoSepar</code></td>
<td>
<p>numeric, depends on <code>karSepar=TRUE</code>, if zero your
karyotypes will have no distance among them, if overlap,
increase this and <code>karHeiSpace</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_addmissingotuafter">addMissingOTUAfter</code></td>
<td>
<p>character, when you want to add space (ghost OTUs) after one or several OTUs, pass the names of OTUs
preceding the desired space in a character vector i.e. <code>c("species one",</code>  <code>"species five")</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_addmissingotubefore">addMissingOTUBefore</code></td>
<td>
<p>character, when you want to add space (ghost OTUs) before one or several OTUs, pass the names of OTUs after
the desired space in a character vector i.e. <code>c("species one",</code>  <code>"species five")</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_missotuspacings">missOTUspacings</code></td>
<td>
<p>numeric, when you use <code>addMissingOTUAfter</code> this numeric vector should have the same length and corresponds
to the number of free spaces (ghost OTUs) to add after each OTU respectively</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_movekarhor">moveKarHor</code></td>
<td>
<p>character, OTUs' names of karyotypes that should be moved horizontally. See <code>mkhValue</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_moveallkarvaluehor">moveAllKarValueHor</code></td>
<td>
<p>numeric, similar to <code>mkhValue</code>, but affects all karyotypes.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_moveallkarvaluey">moveAllKarValueY</code></td>
<td>
<p>numeric, similar to <code>moveAllKarValueHor</code>, but affects y axis.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karanchorleft">karAnchorLeft</code></td>
<td>
<p>character, OTUs' add anchor to the left of this OTU names of karyotypes. For <code>verticalPlot=FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karanchorright">karAnchorRight</code></td>
<td>
<p>character, OTUs' add anchor to the right of this OTU names of karyotypes. For <code>verticalPlot=FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchor">anchor</code></td>
<td>
<p>boolean, when <code>TRUE</code>, plots a parent progeny structure in karyotypes in <code>moveKarHor</code>. Or a horizontal anchor
to the left/right
of <code>karAnchorLeft, karAnchorRight</code> when <code>verticalPlot=FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchorlinelty">anchorLineLty</code></td>
<td>
<p>numeric, type of line in <code>anchor</code>, corresponds to <code>lty</code>. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchortext">anchorText</code></td>
<td>
<p>character, text to add to <code>anchor</code> structure near symbol. See <code>anchor</code>. Defaults to <code>""</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchortextmparental">anchorTextMParental</code></td>
<td>
<p>character, designed to fill with a character object the space left of a missing parental in the
<code>anchor</code> structure.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchortextmovex">anchorTextMoveX</code></td>
<td>
<p>numeric, for vertical plots with <code>anchorText</code> move text in X axis. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchortextmovey">anchorTextMoveY</code></td>
<td>
<p>numeric, for horizontal plots with <code>anchorText</code> move text in Y axis. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchortextmoveparenx">anchorTextMoveParenX</code></td>
<td>
<p>numeric, for plots with <code>anchorTextMParental</code> move text in X axis. Defaults to <code>0</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchortextmovepareny">anchorTextMoveParenY</code></td>
<td>
<p>numeric, for plots with <code>anchorTextMParental</code> move text in Y axis. Defaults to <code>0</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchorvsizef">anchorVsizeF</code></td>
<td>
<p>numeric, factor to modify vertical size of anchor <code>0.5</code> (default). Size itself is equal to <code>karHeiSpace</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_anchorhsizef">anchorHsizeF</code></td>
<td>
<p>numeric, factor to modify horizontal size of anchor <code>1</code> (default).</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_pchanchor">pchAnchor</code></td>
<td>
<p>numeric, symbol for anchor, see <code>?points</code> and <code>anchor</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_moveanchorv">moveAnchorV</code></td>
<td>
<p>numeric, displace anchor vertical portion to right or left. See <code>anchor</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_moveanchorh">moveAnchorH</code></td>
<td>
<p>numeric, displace anchor horizontal portion to right or left. See <code>anchor</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_mkhvalue">mkhValue</code></td>
<td>
<p>numeric, value to move kar. hor. See <code>moveKarHor</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_n">n</code></td>
<td>
<p>numeric vertices number for round corners</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_markn">markN</code></td>
<td>
<p>numeric vertices number for round corners of marks</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_notes">notes</code></td>
<td>
<p>data.frame, or csv file name in quotes, (shown to the right of kar.), with columns <code>OTU</code> and <code>note</code> for adding
notes to each OTU, they appear to the right of chromosomes</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotes">leftNotes</code></td>
<td>
<p>data.frame, or csv file name in quotes (shown to the left), with columns <code>OTU</code> and <code>note</code> for adding notes
to each OTU, they appear to the left of chromosomes</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotesup">leftNotesUp</code></td>
<td>
<p>data.frame, or csv file name in quotes, (shown to the left-up), similar to <code>leftNotes</code>, but intended for
placement over chr.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_notestextsize">notesTextSize</code></td>
<td>
<p>numeric, font size of notes, see <code>notes</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotestextsize">leftNotesTextSize</code></td>
<td>
<p>numeric, font size of notes, see <code>leftNotes</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotesuptextsize">leftNotesUpTextSize</code></td>
<td>
<p>numeric, font size of notes, see <code>leftNotesUp</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_notesleft">notesLeft</code></td>
<td>
<p>deprecated, use a data.frame for <code>leftNotes</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_notesposx">notesPosX</code></td>
<td>
<p>numeric, move right notes to the right or left (x axis)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_notesposy">notesPosY</code></td>
<td>
<p>numeric, move right notes down or up (y axis)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotesposx">leftNotesPosX</code></td>
<td>
<p>numeric, move left notes to the right or left (x axis)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotesposy">leftNotesPosY</code></td>
<td>
<p>numeric, move left notes (<code>leftNotes</code>) down or up (y axis)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotesupposx">leftNotesUpPosX</code></td>
<td>
<p>numeric, move up left notes to the right or left (x axis)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotesupposy">leftNotesUpPosY</code></td>
<td>
<p>numeric, move up left notes (<code>leftNotesUp</code>) down or up (y axis)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_notefont">noteFont</code></td>
<td>
<p>numeric  <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics. See <code>notes</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotefont">leftNoteFont</code></td>
<td>
<p>numeric  <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics. See
<code>leftNotes</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_leftnotefontup">leftNoteFontUp</code></td>
<td>
<p>numeric  <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics. See
<code>leftNotesUp</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_parsetypes">parseTypes</code></td>
<td>
<p>boolean, parse in <code>notes</code> the <em>Citrus</em> chr. types names. Creates subindex pos. for FL. Defaults to
<code>TRUE</code>. Incompatible with <code>parseStr2lang</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_parsestr2lang">parseStr2lang</code></td>
<td>
<p>bolean, parse string in <code>notes</code> with function <code>str2lang(paste0("paste(",note,")") )</code> for ex:
<code>"italic('C. sinensis'), ' Author'"</code>. See <code>notes</code>, <code>leftNotes</code>,<code>leftNotesUp</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_propwidth">propWidth</code></td>
<td>
<p>boolean, defaults to <code>FALSE</code>. Diminishes chr. width with increasing number of OTUs</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_mbthreshold">MbThreshold</code></td>
<td>
<p>numeric, if greater than this number (defaults to <code>10000</code>), <code>MbUnit</code> will apply and <code>specialyTitle</code>
will not.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_threshold">threshold</code></td>
<td>
<p>this is the max. value allowed for the main two significative digits, otherwise scale will shrink. For example,
after 35 <code class="reqn">\mu m</code> (Default), apparent size will be 3.5 and scale interval will change. See also <code>ceilingFactor</code>, you may have to
use <code>-1</code>. Introduced in 1.13</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_mbunit">MbUnit</code></td>
<td>
<p>character, text of units of title when <code>MbThreshold</code> met and OTU not in <code>specialOTUNames</code>. See
<code>specialyTitle</code>
Defaults to <code>"Mb"</code>, but anything can be used. Introduced in 1.13. See <code>specialyTitle</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_ytitle">yTitle</code></td>
<td>
<p>character, units for common title. Defaults to <code class="reqn">\mu m</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_specialytitle">specialyTitle</code></td>
<td>
<p>character, title of ruler if OTU is in <code>specialOTUNames</code>. Will not apply if <code>MbThreshold</code> met. In that
case use <code>MbUnit</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_specialotunames">specialOTUNames</code></td>
<td>
<p>character vector, normally title of ruler is micrometer or Mb (big numbers). Use this param. to be able to put
a different unit in ruler title. See <code>"specialyTitle"</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_addotuname">addOTUName</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds OTU (species) name to karyotype</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otutextsize">OTUTextSize</code></td>
<td>
<p>numeric, font size of OTU name (species). Defaults to <code>1</code>. When <code>OTUasNote</code> is <code>TRUE</code>, use
<code>notesTextSize</code> instead</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otufont">OTUfont</code></td>
<td>
<p>numeric, <code>1</code> for normal,  <code>2</code> for bold,  <code>3</code> for italics,  <code>4</code> for bold-italics</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otufamily">OTUfamily</code></td>
<td>
<p>character, font family for OTU name.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otuasnote">OTUasNote</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds OTU (species) name to the right, see <code>notes</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otuasleftnote">OTUasLeftNote</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds OTU (species) name to the left-up, see <code>leftNotesUp</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_orderchr">orderChr</code></td>
<td>
<p>character, when <code>"size"</code>, sorts chromosomes by total
length from the largest to the smallest. <code>"original"</code>: preserves d.f. order. <code>"name"</code>: sorts alphabetically;
<code>"group"</code>: sorts by group name; <code>"chrNameUp"</code>: sorts according to column <code>chrNameUp</code>. See <code>chrNameUp</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrid">chrId</code></td>
<td>
<p>character, print name of chromosome, <code>"original"</code> uses the original
name in OTU column of dfChrSize, <code>"simple"</code> (just 1 to ...) or <code>"none"</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_classmbname">classMbName</code></td>
<td>
<p>character, name of &quot;chromosome&quot; when in Mbp. Defaults to <code>"Pm"</code>. See <code>MbUnit</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_classcmname">classcMName</code></td>
<td>
<p>character, name of &quot;chromosome&quot; when OTU in <code>specialOTUNames</code>. Defaults to <code>"L.G."</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_classchrname">classChrName</code></td>
<td>
<p>character, name of &quot;chromosome&quot; when in micrometers (apparently). Defaults to <code>"Chr."</code>. See
<code>specialOTUnames</code>, <code>classMbName</code>, <code>classcMName</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_classchrnameup">classChrNameUp</code></td>
<td>
<p>character, name of &quot;chromosome&quot; for col. <code>"chrNameUp"</code>. Defaults to <code>"Type"</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_classgroupname">classGroupName</code></td>
<td>
<p>character, name of groups. Defaults to <code>""</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrnameup">chrNameUp</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds secondary chromosome name from col. <code>chrNameUp</code> over chrs. Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chridpatternrem">chrIdPatternRem</code></td>
<td>
<p>character, regex pattern to remove from chr. names</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_indexidtextsize">indexIdTextSize</code></td>
<td>
<p>numeric, font size of chr. and kar. indices and
chromosome name. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_disttextchr">distTextChr</code></td>
<td>
<p>numeric, distance from name of chromosome to chromosome,
also affects vertical separation of indices. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_groupup">groupUp</code></td>
<td>
<p>boolean, when <code>TRUE</code> when groups present, they appear over the chr. name. Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_groupname">groupName</code></td>
<td>
<p>boolean, when <code>TRUE</code> (default), shows group names. When <code>FALSE</code> only line</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_groupsepar">groupSepar</code></td>
<td>
<p>numeric, factor for affecting chr. spacing <code>chrSpacing</code> among groups. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chromatids">chromatids</code></td>
<td>
<p>boolean, when <code>TRUE</code> shows separated chromatids. Defaults to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_arrowsbothchrt">arrowsBothChrt</code></td>
<td>
<p>boolean, when <code>TRUE</code> (default) (for <code>chromatids=TRUE</code>) shows <code>upArrow,</code>  <code> downArrow</code>
styles of marks in both chromatids when <code>arrowsToSide=TRUE</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_holocennotaschromatids">holocenNotAsChromatids</code></td>
<td>
<p>boolean, when <code>TRUE</code> and <code>chromatids=TRUE</code> does not plot holocen kar. with chromatids.
Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_excholofrarrtoside">excHoloFrArrToSide</code></td>
<td>
<p>boolean, when <code>arrowsToSide=TRUE</code>, excludes holocen. from this behaviour, plotting a centered arrow only.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_xmodifier">xModifier</code></td>
<td>
<p>numeric, for <code>chromatids=TRUE</code>, separation among chromatids. Quotient for <code>chrWidth</code>. Defaults to <code>12
: chrWidth/12</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_xmodmonoholorate">xModMonoHoloRate</code></td>
<td>
<p>numeric, factor to shrink chromatid separ. for holocen. 5 means 5 times smaller (quotient).</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrwidth">chrWidth</code></td>
<td>
<p>numeric, relative chromosome width. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrspacing">chrSpacing</code></td>
<td>
<p>numeric, horizontal spacing among chromosomes, see also  <code>chrWidth</code>. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_specialchrwidth">specialChrWidth</code></td>
<td>
<p>numeric, relative chromosome width. Defaults to <code>0.5</code> for OTUs in <code>specialOTUNames</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_specialchrspacing">specialChrSpacing</code></td>
<td>
<p>numeric, horizontal spacing among chromosomes for OTUs in <code>specialOTUNames</code>, see also  <code>chrWidth</code>.
Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrcolor">chrColor</code></td>
<td>
<p>character, main color for chromosomes. Defaults to <code>"gray"</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrbordercolor">chrBorderColor</code></td>
<td>
<p>character, color for border of chromosomes, defaults to <code>chrColor</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_centromeresize">centromereSize</code></td>
<td>
<p>numeric, optional, this establishes the apparent size of cen. in the plot in <code class="reqn">\mu</code>m. See
<code>autoCenSize=TRUE</code>. Default: <code>0</code>. Use with <code>autoCenSize=FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_autocensize">autoCenSize</code></td>
<td>
<p>boolean, when <code>TRUE</code> ignores <code>centromereSize</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_cencolor">cenColor</code></td>
<td>
<p>character, color for centromeres, if GISH use <code>NULL</code> or <code>NA</code>. Defaults to <code>chrColor</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_fixcenborder">fixCenBorder</code></td>
<td>
<p>boolean, when <code>TRUE</code> uses <code>chrColor</code> as centromere (and cen. mark) border color. See also <code>cenColor</code>,
<code>chrColor</code>, <code>colorBorderMark</code>, <code>borderOfWhiteMarks</code>. No default value. When <code>chrColor</code> is <code>"white"</code> this
turns into <code>"black"</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_gishcenborder">gishCenBorder</code></td>
<td>
<p>boolean, when <code>TRUE</code>, cen. mark border color is the same as mark color, ignoring <code>colorBorderMark</code>. No
default.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_hidecenlines">hideCenLines</code></td>
<td>
<p>numeric, factor to multiply line width (lwd) used for covering cen. border, when <code>chrColor</code> is <code>white</code>
or when <code>gishCenBorder=TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_roundedcen">roundedCen</code></td>
<td>
<p>deprecated, see cenFormat</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_cenformat">cenFormat</code></td>
<td>
<p>boolean, when <code>"triangle"</code>, cen. has triangular aspect. When <code>"rounded"</code>, it has rounded aspect (Default).
<code>"inProtein"</code> for using the mark with style of same name.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_cenfactor">cenFactor</code></td>
<td>
<p>numeric, modifies any cen. mark and cen. size. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_squareness">squareness</code></td>
<td>
<p>numeric, shape of vertices of chromosomes and square marks,
higher values more squared. Defaults to <code>4</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_lwd.chr">lwd.chr</code></td>
<td>
<p>thickness of border of chr., some marks and ruler. Thick of <code>cM</code> marks when <code>lwd.cM</code> absent and other marks
when <code>lwd.marks</code> absent. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_lwd.cm">lwd.cM</code></td>
<td>
<p>thickness of cM marks. Defaults to <code>lwd.chr</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_lwd.marks">lwd.marks</code></td>
<td>
<p>thickness of most marks. Except <code>cM</code> marks and centr. related marks. See <code>lwd.chr</code>, <code>lwd.cM</code>. Defaults
to <code>lwd.chr</code> value when <code>99</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_dfmarkpos">dfMarkPos</code></td>
<td>
<p>data.frame of marks (sites): columns: <code>OTU</code> (opt), <code>chrName</code>,
<code>markName</code> (name of site), <code>chrRegion</code> (for monocen. and opt for whole arm (w) in holocen.), <code>markDistCen</code> (for
monocen.),
<code>markPos</code> (for holocen.), <code>markSize</code>; column <code>chrRegion</code>:
use <code>p</code> for short arm, <code>q</code> for long arm, <code>cen</code> for centromeric mark and <code>w</code> for whole chr. mark; column
<code>markDistCen</code>: use distance from
centromere to mark, not necessary for cen. marks (cen), w, p, q (when whole arm). See also param. <code>markDistType</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_defaultstylemark">defaultStyleMark</code></td>
<td>
<p>character, default style of mark, only used when <code>style</code> column of <code>dfMarkColor</code> data.frame is
missing or in absence of this data.frame. Use <code>"square"</code> (default), <code>"squareLeft"</code>, <code>"dots"</code>, <code>"cM"</code>,
<code>"cMLeft"</code>,<code>"cenStyle"</code>, <code>"upArrow"</code>, <code>"downArrow"</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_markdisttype">markDistType</code></td>
<td>
<p>character, if <code>"cen"</code> = the distance you provided in data.frame (<code>dfMarkPos</code>) column <code>markDistCen</code>
or <code>markPos</code>  is to the center of the mark, if <code>"beg"</code> = the distance you provided is to the
beginning of the mark (Default)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_protruding">protruding</code></td>
<td>
<p>numeric, when style of mark is <code>"cM"</code>, fraction of chrWidth to stretch marker. Defaults to <code>0.2</code>.
Introduced in 1.13</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_startpos">startPos</code></td>
<td>
<p>numeric, factor to increase separation of <code>exProtein</code> marks to chromosome. Defaults to <code>0</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_pmarkfac">pMarkFac</code></td>
<td>
<p>numeric, fraction of chr. size for <code>exProtein</code> style marks. Defaults to <code>0.25</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_origin">origin</code></td>
<td>
<p>For non-monocentric chr. (for holocentrics only) Use <code>"b"</code> (default) if distance to mark in (<code>"markPos"</code>
column in <code>"dfMarkPos"</code>) data.frame measured from bottom of chromosome, use <code>"t"</code> for distance to mark from top of chr.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_efzero">efZero</code></td>
<td>
<p>numeric, numbers below this one will be considered as zero, for comparison purposes. Defaults to <code>1e-5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_cmbegincenter">cMBeginCenter</code></td>
<td>
<p>boolean, start position of <code>cM</code> and <code>cMLeft</code> marks. If <code>TRUE</code>, starts in the center (width) of
chr. . Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_arrowhead">arrowhead</code></td>
<td>
<p>numeric, proportion of head of arrow (mark styles: <code>upArrow,</code>  <code>downArrow</code>). Defaults to <code>0.3</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_shrinkarrow">shrinkArrow</code></td>
<td>
<p>numeric, proportion, shrinks body of arrow. Defaults to <code>0.3333</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_arrowheadwidthshrink">arrowheadWidthShrink</code></td>
<td>
<p>numeric, proportion, shrinks head of arrow. Defaults to <code>0.1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_arrowstoside">arrowsToSide</code></td>
<td>
<p>boolean, when <code>FALSE</code> use a centered arrow, instead of an arrow next to chr. margins (<code>TRUE</code>, default).
See <code>arrowsBothChrt</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_useonedot">useOneDot</code></td>
<td>
<p>boolean, use one dot instead of two in style of marks <code>dots</code>. Defaults to <code>FALSE</code>. Not useful for
<code>chromatids=TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_dotsasoval">dotsAsOval</code></td>
<td>
<p>boolean, use oval instead of two dots in style of marks <code>dots</code>. Defaults to <code>FALSE</code>. See <code>useOneDot</code>.
Not useful for <code>chromatids=TRUE</code> or <code>circularPlot=TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_dfmarkcolor">dfMarkColor</code></td>
<td>
<p>data.frame, optional, specifying colors and style for marks (sites);
columns: <code>markName</code>, <code>markColor</code>, <code>style</code>. <code>style</code> accepts: <code>square</code>, <code>squareLeft</code>, <code>dots</code>,
<code>cM</code>,
<code>"cMLeft"</code>, <code>"cenStyle"</code>, <code>"upArrow"</code>, <code>"downArrow"</code>, <code>"exProtein"</code>.
(if column <code>style</code> missing all (except 5S) are plotted as in param. <code>defaultStyleMark</code>).</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_mycolors">mycolors</code></td>
<td>
<p>character vector, optional, i.e. <code>c("blue",</code> <code>"red",</code> <code>"green")</code> for specifying color of marks in order
of appearance. if diverges with number of marks will be recycled if <code>dfMarkColor</code> present, mycolors will be ignored. To know the
order of your marks use something like: <code>unique(dfMarkPos$markName)</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_alpha_val">alpha_val</code></td>
<td>
<p>numeric vector, make marks transparent, accepts values from 0 to 1, @see <code>scales::alpha</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_borderofwhitemarks">borderOfWhiteMarks</code></td>
<td>
<p>boolean, if <code>TRUE</code> (Default) uses black border for white marks. See <code>dfMarkColor</code>. Does not apply
to marks with style <code>cenStyle</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_colorbordermark">colorBorderMark</code></td>
<td>
<p>character, without default, pass a name of a color to use as border of marks. See <code>borderOfWhiteMarks</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_lwd.mimiccen">lwd.mimicCen</code></td>
<td>
<p>thickness of lines of <code>cenStyle</code> marks; affects only lateral borders. Defaults to <code>lwd.chr</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_defcenstylecol">defCenStyleCol</code></td>
<td>
<p>character, color of outer part of <code>cenStyle</code> marks. Defaults to <code>white</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_pattern">pattern</code></td>
<td>
<p>REGEX pattern to remove from names of marks</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_legend">legend</code></td>
<td>
<p>character, <code>"none"</code> for no legend; <code>"inline"</code> prints labels near
chromosomes; <code>"aside"</code> prints legend to the right of karyotypes (default). See <code>markLabelSpacer</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_remsimimarkleg">remSimiMarkLeg</code></td>
<td>
<p>boolean, when <code>legend="aside"</code>, if you use <code>pattern</code>, you can have several marks with same name. When
<code>TRUE</code> this remove this pseudoduplicates from legend. Be sure that this pseudoduplicates have the same color, otherwise you should
use <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_bannedmarkname">bannedMarkName</code></td>
<td>
<p>character, character string or vector with mark names to be removed from plot. Not the marks but the labels.
Except when <code>bMarkNameAside</code> is used.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_bmarknameaside">bMarkNameAside</code></td>
<td>
<p>boolean, when <code>TRUE</code> and <code>legend="inline"</code>, shows marks in <code>bannedMarkName</code> as
<code>legend="aside"</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_forbiddenmark">forbiddenMark</code></td>
<td>
<p>character, character string or vector with mark names to be removed from plot. Not the marks but the labels.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_legendwidth">legendWidth</code></td>
<td>
<p>numeric, factor to increase width of squares and of legend. Defaults to <code>1.7</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_legendheight">legendHeight</code></td>
<td>
<p>numeric, factor to increase height of squares and dots of legend. Automatic.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>numeric, only if legend != (not) &quot;&quot;, size of the font of
labels of marks (legend). Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_marklabelspacer">markLabelSpacer</code></td>
<td>
<p>numeric, only if <code>legend="aside"</code>, space from the
rightmost chr. to legend. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_legendycoord">legendYcoord</code></td>
<td>
<p>numeric, modify Y position of legend when <code>legend="aside"</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_marknewline">markNewLine</code></td>
<td>
<p>character, character to split mark Names into different lines. Applies to <code>square</code> marks. Defaults to <code>NA</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_mylheight">mylheight</code></td>
<td>
<p>numeric, for <code>markNewLine!=NA</code>; is equivalent to <code>lheight</code> of <code>par</code>: &quot;The line height multiplier. The
height of a line of text (used to vertically space multi-line text) is found by multiplying the character height both by the current
character expansion and by the line height multiplier.&quot; Defaults to <code>0.7</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrsize">chrSize</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds total chr size under each chr. Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_nsmall">nsmall</code></td>
<td>
<p>numeric, rounding decimals for <code>chrSize</code> parameter. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrsizembp">chrSizeMbp</code></td>
<td>
<p>boolean, when <code>TRUE</code> adds total Mbp chr. size to each chr. provided, there is a <code>Mbp</code> column in
<code>dfChrSize</code> data.frame. Defaults to <code>FALSE</code>. If data in columns <code>shortArmSize</code>, or col. <code>chrSize</code> is in millions
(&quot;Mbp&quot;). Use <code>chrSize=TRUE</code> not this one (not column <code>Mbp</code>, you don't need this).</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_markper">markPer</code></td>
<td>
<p>character vector, name of mark(s) to calculate % of mark in chr. and add it to plot. See <code>perAsFraction</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_showmarkpos">showMarkPos</code></td>
<td>
<p>boolean, adds position of marks under karyotype (fraction 0-1) when <code>TRUE</code>. Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_btoremove">bToRemove</code></td>
<td>
<p>character vector, bands to remove from calc. of pos., when <code>showMarkPos = TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_perasfraction">perAsFraction</code></td>
<td>
<p>boolean, when <code>TRUE</code> % is shown as fraction. Defaults to <code>FALSE</code>. See <code>markPer</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrindex">chrIndex</code></td>
<td>
<p>character, add arm ratio with <code>"AR"</code> and centromeric index with <code>"CI"</code>, or <code>"both"</code> (Default), or
<code>"none"</code> for none</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_morpho">morpho</code></td>
<td>
<p>character, when <code>"both"</code> (default) prints the Guerra and Levan classif of cen. position, use also <code>"Guerra"</code>
or  <code>"Levan"</code> or <code>"none"</code> for none. See also <code>?armRatioCI</code>.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_namechrindexpos">nameChrIndexPos</code></td>
<td>
<p>numeric, modify position of name of chr. indices</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karindex">karIndex</code></td>
<td>
<p>logical, add karyotype indices A (intrachromosomal -
centromere pos.) and A2 (interchromosomal asymmetry, variation among
chromosome sizes)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karindexpos">karIndexPos</code></td>
<td>
<p>numeric, move karyotype index. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_ruler">ruler</code></td>
<td>
<p>boolean, display ruler to the left of karyotype, when <code>FALSE</code> no ruler</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_useminorticks">useMinorTicks</code></td>
<td>
<p>boolean, display minor ticks between labeled ticks in ruler. See <code>miniTickFactor</code>. Defaults to <code>FALSE</code>.
(ticks without label)</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_minitickfactor">miniTickFactor</code></td>
<td>
<p>numeric, number of minor ticks for each labeled tick. See <code>useMinorTicks</code>. Defaults to <code>10</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulerpos">rulerPos</code></td>
<td>
<p>numeric, absolute position of ruler, corresponds to <code>pos</code>
argument of <code>axis</code> R plot</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_ruler.tck">ruler.tck</code></td>
<td>
<p>numeric, tick size of ruler, corresponds to <code>tck</code> argument of
<code>axis</code> R plot. Defaults to <code>-0.02</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulernumberpos">rulerNumberPos</code></td>
<td>
<p>numeric, modify position of numbers of ruler. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulernumbersize">rulerNumberSize</code></td>
<td>
<p>numeric, size of number's font in ruler. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_collapsecen">collapseCen</code></td>
<td>
<p>boolean, avoid spacing in ruler between short arm and long arm.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulerinterval">rulerInterval</code></td>
<td>
<p>numeric, intervals in ruler. No default, automatic.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulerintervalcm">rulerIntervalcM</code></td>
<td>
<p>numeric, intervals in ruler of OTU in <code>specialOTUNames</code>. No default. Introduced in 1.13</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulerintervalmb">rulerIntervalMb</code></td>
<td>
<p>numeric, intervals in ruler of OTU with data in Mb (&gt;<code>MbThreshold</code>) and absent from
<code>specialOTUNames</code>. No default. Usa data in millions</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_ceilingfactor">ceilingFactor</code></td>
<td>
<p>numeric, affects number of decimals for ceiling. Affects max. value of ruler. Defaults to <code>0</code>. When
<code>threshold</code> is greater than <code>35</code> this may have to be negative. Introduced in 1.13</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_xposrulertitle">xPosRulerTitle</code></td>
<td>
<p>numeric, modify position of ruler title. See <code>yTitle,</code>  <code> specialyTitle,</code>  <code>
MbUnit</code>. Defaults to <code>2.6</code>. A value of <code>2.6</code> means <code>2.6</code> times the value of <code>chrSpacing</code> to the left, from the first
chr.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_yposrulertitle">yPosRulerTitle</code></td>
<td>
<p>numeric, affects vertical position of ruler title. Defaults to <code>0</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rulertitlesize">rulerTitleSize</code></td>
<td>
<p>numeric font size of units of ruler. See also <code>xPosRulerTitle</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_xlimleftmod">xlimLeftMod</code></td>
<td>
<p>numeric, modifies <code>xlim</code> left argument of plot</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_xlimrightmod">xlimRightMod</code></td>
<td>
<p>numeric, <code>xlim</code> right side modification by adding space to the right
of idiograms. Defaults to <code>2</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_ylimbotmod">ylimBotMod</code></td>
<td>
<p>numeric, modify <code>ylim</code> bottom argument of plot</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_ylimtopmod">ylimTopMod</code></td>
<td>
<p>numeric, modify <code>ylim</code> top argument of plot</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_callplot">callPlot</code></td>
<td>
<p>boolean, create new plot in your device. Defaults to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_asp">asp</code></td>
<td>
<p>numeric, y x aspect of plot. Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean, if <code>TRUE</code> chromosomes/karyotypes are plotted in concentric circles. Defaults to <code>FALSE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_verticalplot">verticalPlot</code></td>
<td>
<p>boolean, when <code>TRUE</code> karyotypes are plotted vertically, otherwise, horizontally Defaults to <code>TRUE</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_karspacehor">karSpaceHor</code></td>
<td>
<p>numeric, separation among horizontal karyotypes. When <code>verticalPlot=FALSE</code>. Defaults to <code>0</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_shrinkfactor">shrinkFactor</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> percentage of usage of circle. Defaults to <code>0.9</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_separfactor">separFactor</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> modify separation of concentric karyotypes. Defaults to <code>1.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_labelspacing">labelSpacing</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Spacing of mark labels. Defaults to <code>0.7</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_labeloutwards">labelOutwards</code></td>
<td>
<p>boolean, inline labels projected outwards</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_chrlabelspacing">chrLabelSpacing</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Spacing of chr. labels. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_radius">radius</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Affects radius of karyotypes. Defaults to <code>0.5</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_rotation">rotation</code></td>
<td>
<p>numeric, anti-clockwise rotation, defaults to <code>0.5</code> which rotates chr. from top to -90 degrees. (-0.5*<code class="reqn">\pi</code> )</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_circlecenter">circleCenter</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Coordinate X of center of circles. Affects <code>legend="aside"</code> position.
Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_circlecentery">circleCenterY</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Coordinate Y of center of circles. Affects <code>legend="aside"</code> position.
Defaults to <code>1</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otulabelspacing">OTUlabelSpacing</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Spacing for OTU names. Defaults to <code>0.3</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otusrt">OTUsrt</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code>. Angle to use for OTU names. Defaults to <code>0</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otuplacing">OTUplacing</code></td>
<td>
<p>character, for <code>circularPlot=TRUE</code>. location of OTU name. Defaults to <code>"first"</code> plots name near
first chr. <code>"number"</code> places number near 1st chr. and index and name to the right or center.
<code>"simple"</code> place name to the right or center without numbering. See also <code>OTUcentered</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otulabelspacerx">OTULabelSpacerx</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Modifies x names position</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otulabelspacery">OTULabelSpacery</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Modifies y names position</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otucentered">OTUcentered</code></td>
<td>
<p>boolean, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. OTU name in center of circle
when <code>TRUE</code>, otherwise, to the right.</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otujustif">OTUjustif</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Justification of OTU name. <code>0</code>
= left
(Default); use <code>0.5</code> for centered. See <code>?text</code> -&gt; <code>adj</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_otulegendheight">OTUlegendHeight</code></td>
<td>
<p>numeric, for <code>circularPlot=TRUE</code> and <code>OTUplacing="number" or "simple"</code>. Modifies y names separation</p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_roundness">roundness</code></td>
<td>
<p>deprecated, use <code>squareness</code></p>
</td></tr>
<tr><td><code id="plotIdiograms_+3A_...">...</code></td>
<td>
<p>accepts other arguments for the plot, see, <code>?plot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+asymmetry">asymmetry</a></code>
</p>
<p><code><a href="#topic+armRatioCI">armRatioCI</a></code>
</p>
<p><code><a href="#topic+chrbasicdatamono">chrbasicdatamono</a></code>
</p>
<p><code><a href="#topic+chrbasicdataHolo">chrbasicdataHolo</a></code>
</p>
<p><code><a href="#topic+markposDFs">markposDFs</a></code>
</p>
<p><code><a href="#topic+markdataholo">markdataholo</a></code>
</p>
<p><code><a href="#topic+dfMarkColor">dfMarkColor</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dfOfChrSize)
plotIdiograms(dfOfChrSize, ylimBotMod = .75, rulerPos = -.5)
plotIdiograms(dfOfChrSize, circularPlot = TRUE, chrLabelSpacing = 1)
plotIdiograms(dfChrSizeHolo, rulerPos = -.5)
</code></pre>

<hr>
<h2 id='plotlabelsright'>This is an internal function to plot labels to the right, when &quot;aside&quot; for
its position</h2><span id='topic+plotlabelsright'></span>

<h3>Description</h3>

<p>It returns a graphic element with the legends
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotlabelsright(
  maxx,
  y,
  markLabelSpacer,
  chrWidth,
  dfMarkColorInt,
  allMarkMaxSize,
  normalizeToOne,
  markLabelSize,
  xfactor,
  legendWidth,
  legendHeight,
  n,
  pattern,
  legendYcoord,
  useOneDot,
  dotsAsOval,
  circularPlot
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotlabelsright_+3A_y">y</code></td>
<td>
<p>The y axis coordinates of chromosomes</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_marklabelspacer">markLabelSpacer</code></td>
<td>
<p>distance from right chr to legend</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_chrwidth">chrWidth</code></td>
<td>
<p>chr widht</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_dfmarkcolorint">dfMarkColorInt</code></td>
<td>
<p>data.frame of mark characteristics</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_allmarkmaxsize">allMarkMaxSize</code></td>
<td>
<p>maximum size of marks</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_normalizetoone">normalizeToOne</code></td>
<td>
<p>transformation value of karyotype height</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>font size of legends</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_xfactor">xfactor</code></td>
<td>
<p>relative proportion of x vs y axes</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_legendwidth">legendWidth</code></td>
<td>
<p>factor to increase width of legend squares and dots</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_legendheight">legendHeight</code></td>
<td>
<p>factor to increase height of legend squares and dots</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_n">n</code></td>
<td>
<p>numeric, vertices for round parts</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_pattern">pattern</code></td>
<td>
<p>character, regex to remove from markNames</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_legendycoord">legendYcoord</code></td>
<td>
<p>numeric modify Y position of legend</p>
</td></tr>
<tr><td><code id="plotlabelsright_+3A_useonedot">useOneDot</code></td>
<td>
<p>boolean, when <code>TRUE</code> plots only one dot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a graphics element
</p>

<hr>
<h2 id='plotSqMarks'>sqPlotMarkCP
This is an internal function that plot marks</h2><span id='topic+plotSqMarks'></span>

<h3>Description</h3>

<p>It returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotSqMarks(
  chromatidsCheck_1,
  chromatidsCheck_2,
  circularPlot,
  squareness,
  xMarkSq,
  yMarkSq,
  dfMarkColorInt,
  lwd.marks2,
  listMarkPosSq,
  listChrSize,
  specialChrWidth,
  chrWidth,
  yfactor,
  markN,
  ptsl,
  bannedMarkName3,
  y,
  markLabelSize,
  pattern,
  separFactor,
  labelSpacing,
  circleCenter,
  circleCenterY,
  radius,
  legend,
  ylistTransChrSimple,
  rotation,
  labelOutwards,
  x,
  xModifierMH,
  pts,
  alpha_val = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotSqMarks_+3A_circularplot">circularPlot</code></td>
<td>
<p>boolean TRUE for circ.</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_squareness">squareness</code></td>
<td>
<p>squareness of vertices &lt;</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_dfmarkcolorint">dfMarkColorInt</code></td>
<td>
<p>colors for marks</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_listmarkpossq">listMarkPosSq</code></td>
<td>
<p>list of df. of mark pos.</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_listchrsize">listChrSize</code></td>
<td>
<p>chr size df list</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_specialchrwidth">specialChrWidth</code></td>
<td>
<p>numeric, width of chr.</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_chrwidth">chrWidth</code></td>
<td>
<p>numeric, width of chr.</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_yfactor">yfactor</code></td>
<td>
<p>yMark distortion based on canvas proportion</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_y">y</code></td>
<td>
<p>list, y coords.</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>numeric, font size</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_pattern">pattern</code></td>
<td>
<p>character, regex</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_separfactor">separFactor</code></td>
<td>
<p>numeric, sep. kars.</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_labelspacing">labelSpacing</code></td>
<td>
<p>numeric, spacing</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_circlecenter">circleCenter</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_circlecentery">circleCenterY</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_radius">radius</code></td>
<td>
<p>numeric</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_legend">legend</code></td>
<td>
<p>character</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_rotation">rotation</code></td>
<td>
<p>rotate</p>
</td></tr>
<tr><td><code id="plotSqMarks_+3A_labeloutwards">labelOutwards</code></td>
<td>
<p>srt</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='posCalc'>FUNCTION posCalc and fillMarkInfo</h2><span id='topic+posCalc'></span><span id='topic+fillMarkInfo'></span><span id='topic+fillMarkInfo2'></span>

<h3>Description</h3>

<p>calculates position of marks in fraction of (%) chromosome units (0-1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posCalc(
  dfMarkPos,
  listOfdfChromSize,
  bToRemove = "",
  markDistType = "beg",
  origin = "b",
  showBandList = FALSE,
  result = "tibble"
)

fillMarkInfo2(dfMarkPos, dfChrSize)

fillMarkInfo(dfMarkPos, dfChrSize, markDistType = "beg", origin = "b")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="posCalc_+3A_dfmarkpos">dfMarkPos</code></td>
<td>
<p>data.frame of marks' position</p>
</td></tr>
<tr><td><code id="posCalc_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>list (for <code>posCalc</code>) or data.frames of chr. sizes.</p>
</td></tr>
<tr><td><code id="posCalc_+3A_btoremove">bToRemove</code></td>
<td>
<p>character, bands to remove from calc. of pos.</p>
</td></tr>
<tr><td><code id="posCalc_+3A_markdisttype">markDistType</code></td>
<td>
<p>markDistType character, if <code>"cen"</code> = the distance you provided in data.frame (<code>dfMarkPos</code>)
column <code>markDistCen</code>
or <code>markPos</code>  is to the center of the mark, if <code>"beg"</code> = the distance you provided is to the
beginning of the mark (Default)</p>
</td></tr>
<tr><td><code id="posCalc_+3A_origin">origin</code></td>
<td>
<p>character, For non-monocentric chr. (for holocentrics only) Use <code>"b"</code> (default) if distance to mark
in (<code>"markPos"</code> column in <code>"dfMarkPos"</code>) data.frame measured from bottom of chromosome, use <code>"t"</code>
for distance to mark from top of chr.</p>
</td></tr>
<tr><td><code id="posCalc_+3A_showbandlist">showBandList</code></td>
<td>
<p>boolean, show row of all bands in tibble, see <code>"result"</code></p>
</td></tr>
<tr><td><code id="posCalc_+3A_result">result</code></td>
<td>
<p>character, use <code>"tibble"</code> to get results in tibble, <code>"data.frame"</code>, or other string results in a <code>list</code></p>
</td></tr>
<tr><td><code id="posCalc_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>data.frame of chr. sizes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list, tibble
</p>
<p>data.frame of marks
</p>
<p>data.frame of marks
</p>


<h3>Examples</h3>

<pre><code class='language-R'>load(system.file("shinyApps", "iBoard/www/rda/monoholoCS.rda", package = "idiogramFISH"))
load(system.file("shinyApps", "iBoard/www/rda/monoholoMarks.rda", package = "idiogramFISH"))
monoholoMarks2 &lt;- fillMarkInfo(monoholoMarks, monoholoCS)
posCalc(monoholoMarks2, monoholoCS, result = "data.frame")

</code></pre>

<hr>
<h2 id='removeNAFromList'>removeNAFromList
This is an internal function that removes NA from list</h2><span id='topic+removeNAFromList'></span>

<h3>Description</h3>

<p>It returns rulers (axis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>removeNAFromList(noNA, areNA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="removeNAFromList_+3A_nona">noNA</code></td>
<td>
<p>original list</p>
</td></tr>
<tr><td><code id="removeNAFromList_+3A_arena">areNA</code></td>
<td>
<p>which are NA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>

<hr>
<h2 id='robert'>FUNCTION to produce a Robertsonian translocation</h2><span id='topic+robert'></span>

<h3>Description</h3>

<p>This function reads a data.frame with chr. sizes
<code><a href="#topic+chrbasicdatamono">chrbasicdatamono</a></code>
and another with marks' positions,
<code><a href="#topic+markposDFs">markposDFs</a></code> and gets as arguments two chr. names and two arms,
respectively.
</p>
<p>It returns a list with two data.frames. One with the chr. size
of the resulting
translocation
and another with the marks' positions for the derivative chr.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>robert(dfChrSize, dfMarkPos, chr1, chr2, arm1, arm2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="robert_+3A_dfchrsize">dfChrSize</code></td>
<td>
<p>name of data.frame of chr. sizes</p>
</td></tr>
<tr><td><code id="robert_+3A_dfmarkpos">dfMarkPos</code></td>
<td>
<p>name of data.frame of chr marks' positions</p>
</td></tr>
<tr><td><code id="robert_+3A_chr1">chr1</code></td>
<td>
<p>name of chr.</p>
</td></tr>
<tr><td><code id="robert_+3A_chr2">chr2</code></td>
<td>
<p>name of chr.</p>
</td></tr>
<tr><td><code id="robert_+3A_arm1">arm1</code></td>
<td>
<p>arm of <code>chr1</code> to be included</p>
</td></tr>
<tr><td><code id="robert_+3A_arm2">arm2</code></td>
<td>
<p>arm of <code>chr2</code> to be included</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list
</p>


<h3>References</h3>

<p>Robertson, W. R. B. (1916). Chromosome studies. I. Taxonomic
relationships shown in
the chromosomes of Tettigidae and Acrididae: V-shaped chromosomes and their
significance in
Acrididae, Locustidae, and Gryllidae: chromosomes and variation. Journal of
Morphology, 27(2),
179-331.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(humChr)
data(humMarkPos)
chrt13q14q &lt;- robert(humChr, humMarkPos, 13, 14, "q", "q")

</code></pre>

<hr>
<h2 id='rulerPlot'>rulerPlot
This is an internal function that plots rulers</h2><span id='topic+rulerPlot'></span>

<h3>Description</h3>

<p>It returns rulers (axis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rulerPlot(
  ycoord,
  listOfdfChromSize,
  listOfdfChromSizeCenType,
  fromZerotoMax,
  rulerNumberSize,
  rulerPos,
  ruler.tck,
  lwd.chr,
  moveKarHor2,
  mkhValue,
  useMinorTicks,
  miniTickFactor,
  verticalPlot,
  moveAllKarValueHor
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rulerPlot_+3A_ycoord">ycoord</code></td>
<td>
<p>y coords</p>
</td></tr>
<tr><td><code id="rulerPlot_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>list of all d.f chr size</p>
</td></tr>
<tr><td><code id="rulerPlot_+3A_listofdfchromsizecentype">listOfdfChromSizeCenType</code></td>
<td>
<p>only d.f. of this cen. type</p>
</td></tr>
<tr><td><code id="rulerPlot_+3A_fromzerotomax">fromZerotoMax</code></td>
<td>
<p>intervals</p>
</td></tr>
<tr><td><code id="rulerPlot_+3A_rulernumbersize">rulerNumberSize</code></td>
<td>
<p>font</p>
</td></tr>
<tr><td><code id="rulerPlot_+3A_rulerpos">rulerPos</code></td>
<td>
<p>pos. of ruler</p>
</td></tr>
<tr><td><code id="rulerPlot_+3A_ruler.tck">ruler.tck</code></td>
<td>
<p>tick size and orient.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>axis
</p>

<hr>
<h2 id='rulerTitle'>rulerTitle
This is an internal function that plots rulers</h2><span id='topic+rulerTitle'></span>

<h3>Description</h3>

<p>It returns rulers (axis)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rulerTitle(
  xmnoNA,
  ymnoNA,
  chrSpacing,
  yPosRulerTitle,
  listOfdfChromSizeCenType,
  MbUnit,
  specialyTitle,
  yTitle,
  xPosRulerTitle,
  rulerTitleSize,
  verticalPlot
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rulerTitle_+3A_xmnona">xmnoNA</code></td>
<td>
<p>x coords</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_ymnona">ymnoNA</code></td>
<td>
<p>y coords</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_chrspacing">chrSpacing</code></td>
<td>
<p>chr. spacing</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_listofdfchromsizecentype">listOfdfChromSizeCenType</code></td>
<td>
<p>only d.f. of this cen. type</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_mbunit">MbUnit</code></td>
<td>
<p>intervals</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_specialytitle">specialyTitle</code></td>
<td>
<p>modifier of pos.</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_ytitle">yTitle</code></td>
<td>
<p>pos. of ruler</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_xposrulertitle">xPosRulerTitle</code></td>
<td>
<p>font</p>
</td></tr>
<tr><td><code id="rulerTitle_+3A_rulertitlesize">rulerTitleSize</code></td>
<td>
<p>font size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='runBoard'>FUNCTION runBoard</h2><span id='topic+runBoard'></span>

<h3>Description</h3>

<p>runBoard: run shinyApp
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runBoard(installAll = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="runBoard_+3A_installall">installAll</code></td>
<td>
<p>boolean, when <code>TRUE</code> dependences are installed without asking. Defaults to <code>FALSE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>shiny
</p>

<hr>
<h2 id='textLabel'>textLabel
This is an internal function that plot labels of cM marks</h2><span id='topic+textLabel'></span>

<h3>Description</h3>

<p>It returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>textLabel(
  xMark,
  yMark,
  listChrSizeText1,
  listOfdfMarkPos,
  specialChrSpacing,
  chrSpacing,
  markLabelSize,
  pattern,
  bannedMarkName,
  isCentromeric = FALSE,
  markNewLine2,
  mylheight2,
  xsize = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="textLabel_+3A_xmark">xMark</code></td>
<td>
<p>x coords</p>
</td></tr>
<tr><td><code id="textLabel_+3A_ymark">yMark</code></td>
<td>
<p>y coords</p>
</td></tr>
<tr><td><code id="textLabel_+3A_listofdfmarkpos">listOfdfMarkPos</code></td>
<td>
<p>color</p>
</td></tr>
<tr><td><code id="textLabel_+3A_specialchrspacing">specialChrSpacing</code></td>
<td>
<p>vertices number</p>
</td></tr>
<tr><td><code id="textLabel_+3A_chrspacing">chrSpacing</code></td>
<td>
<p>aspect</p>
</td></tr>
<tr><td><code id="textLabel_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>size of mark label font</p>
</td></tr>
<tr><td><code id="textLabel_+3A_pattern">pattern</code></td>
<td>
<p>pattern to remove from mark name</p>
</td></tr>
<tr><td><code id="textLabel_+3A_iscentromeric">isCentromeric</code></td>
<td>
<p>use TRUE for cen. marks</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plotted text
</p>

<hr>
<h2 id='textLabelDots'>textLabelDots
This is an internal function that plot labels of cM marks</h2><span id='topic+textLabelDots'></span>

<h3>Description</h3>

<p>It returns a plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>textLabelDots(
  xMarkCr,
  yMarkCr,
  listOfdfChromSize,
  listOfdfMarkPosCr,
  specialChrSpacing,
  chrSpacing,
  markLabelSize,
  pattern,
  bannedMarkName,
  xBoundariesQuar
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="textLabelDots_+3A_listofdfchromsize">listOfdfChromSize</code></td>
<td>
<p>radius</p>
</td></tr>
<tr><td><code id="textLabelDots_+3A_listofdfmarkposcr">listOfdfMarkPosCr</code></td>
<td>
<p>color</p>
</td></tr>
<tr><td><code id="textLabelDots_+3A_specialchrspacing">specialChrSpacing</code></td>
<td>
<p>vertices number</p>
</td></tr>
<tr><td><code id="textLabelDots_+3A_chrspacing">chrSpacing</code></td>
<td>
<p>aspect</p>
</td></tr>
<tr><td><code id="textLabelDots_+3A_marklabelsize">markLabelSize</code></td>
<td>
<p>size of mark label font</p>
</td></tr>
<tr><td><code id="textLabelDots_+3A_pattern">pattern</code></td>
<td>
<p>pattern to remove from mark name</p>
</td></tr>
<tr><td><code id="textLabelDots_+3A_xboundariesquar">xBoundariesQuar</code></td>
<td>
<p>x size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plotted text
</p>

<hr>
<h2 id='yVertoHor'>FUNCTIONS: yVertoHor, xHortoVer, xHortoVerRoundCen, xHortoVerDots,
mapCircle, mapRadius,
transYList, mapOTUnames, addOTUnames, applyMapCircle, intercalate,
drawPlot, drawPlotMark,
drawPlotMarkLine, drawCen, circLabelMark, circPlotDots, plotChrNames,
addChrNameAttrMark,
addChrNameAttrMarkDots, xMarkMap, xMarkMapLeft, markMapPer, markMapPercM,
markMapPerDots,
radDotsPer, centerMarkMapPer, transyListMark, transyListCen, transRadDots,
transyListMarkDots,
oneDot, OTUlabelsright</h2><span id='topic+yVertoHor'></span>

<h3>Description</h3>

<p>FUNCTIONS: yVertoHor, xHortoVer, xHortoVerRoundCen, xHortoVerDots,
mapCircle, mapRadius,
transYList, mapOTUnames, addOTUnames, applyMapCircle, intercalate,
drawPlot, drawPlotMark,
drawPlotMarkLine, drawCen, circLabelMark, circPlotDots, plotChrNames,
addChrNameAttrMark,
addChrNameAttrMarkDots, xMarkMap, xMarkMapLeft, markMapPer, markMapPercM,
markMapPerDots,
radDotsPer, centerMarkMapPer, transyListMark, transyListCen, transRadDots,
transyListMarkDots,
oneDot, OTUlabelsright
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yVertoHor(y, monocenNames)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="yVertoHor_+3A_y">y</code></td>
<td>
<p>list, chr coords.</p>
</td></tr>
<tr><td><code id="yVertoHor_+3A_monocennames">monocenNames</code></td>
<td>
<p>character, vector with names of OTUs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
