<!DOCTYPE html><html><head><title>Help for package IDMIR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IDMIR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GetData_Mirna'><p>GetData_Mirna</p></a></li>
<li><a href='#GetGDEscore'><p>GetGDEscore</p></a></li>
<li><a href='#IdentifyMiRNA'><p>IdentifyMiRNA</p></a></li>
<li><a href='#MirnaData'><p>An environment variable that includes some example data</p></a></li>
<li><a href='#MutiMiRNA_CRModel'><p>MutiMiRNA_CRModel</p></a></li>
<li><a href='#PlotForest'><p>PlotForest</p></a></li>
<li><a href='#PlotHeatmap'><p>PlotHeatmap</p></a></li>
<li><a href='#PlotScatter'><p>PlotScatter</p></a></li>
<li><a href='#PlotSurvival'><p>PlotSurvival</p></a></li>
<li><a href='#SingleMiRNA_CRModel'><p>SingleMiRNA_CRModel</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identification of Dysregulated MiRNAs Based on MiRNA-MiRNA
Interaction Network</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A systematic biology tool was developed to identify dysregulated miRNAs via a miRNA-miRNA interaction network. 'IDMIR' first constructed a weighted miRNA interaction network through integrating miRNA-target interaction information, molecular function data from Gene Ontology (GO) database and gene transcriptomic data in specific-disease context, and then, it used a network propagation algorithm on the network to identify significantly dysregulated miRNAs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>egg, fastmatch, forestplot, ggplot2, grDevices, grid, igraph,
pheatmap, survival, survminer</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-09 11:34:00 UTC; 89800</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han [aut, cre, cph],
  Xilong Zhao [aut],
  Jiashuo Wu [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-09 15:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='GetData_Mirna'>GetData_Mirna</h2><span id='topic+GetData_Mirna'></span>

<h3>Description</h3>

<p>Get the example data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetData_Mirna(Data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetData_Mirna_+3A_data">Data</code></td>
<td>
<p>A character should be one of&quot;survival&quot;, &quot;GEP&quot;, &quot;MF_Target&quot;, &quot;MiRNA_Target&quot;, &quot;matrix_mirna_go_inter&quot;, &quot;matrix_mirna_go_jaccard&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data
</p>

<hr>
<h2 id='GetGDEscore'>GetGDEscore</h2><span id='topic+GetGDEscore'></span>

<h3>Description</h3>

<p>Function &quot;GetGDEscore&quot; is used to calculate gene differential expression levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetGDEscore(ExpData,Label)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetGDEscore_+3A_expdata">ExpData</code></td>
<td>
<p>A gene expression profile of interest (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="GetGDEscore_+3A_label">Label</code></td>
<td>
<p>A character vector consists of &quot;0&quot; and &quot;1&quot; which represent sample class in the gene expression profile. &quot;0&quot; means normal sample and &quot;1&quot; means disease sample.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with one column of GDEscore.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
label&lt;-GetData_Mirna("label")
# Run the function
GDEscore&lt;-GetGDEscore(GEP,label)
</code></pre>

<hr>
<h2 id='IdentifyMiRNA'>IdentifyMiRNA</h2><span id='topic+IdentifyMiRNA'></span>

<h3>Description</h3>

<p>The function &quot;IdentifyMiRNA&quot; is used to identify significantly dysregulated miRNAs by calculating the eigenvector centrality of miRNAs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IdentifyMiRNA(GDEscore.table,nperm=1000,damping=0.90)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IdentifyMiRNA_+3A_gdescore.table">GDEscore.table</code></td>
<td>
<p>A matrix with one column of GDEscore.</p>
</td></tr>
<tr><td><code id="IdentifyMiRNA_+3A_nperm">nperm</code></td>
<td>
<p>The Number of random permutations (default: 100).</p>
</td></tr>
<tr><td><code id="IdentifyMiRNA_+3A_damping">damping</code></td>
<td>
<p>Restart the probability of the random-walk algorithm (default: 0.9).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with seven columns those are &quot;MiRNA&quot;, &quot;Target&quot;, &quot;Number&quot; (number of targets), &quot;Score&quot; (Centrality score), &quot;P-value&quot;, and &quot;FDR&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
label&lt;-GetData_Mirna("label")
# Run the function
GDEscore&lt;-GetGDEscore(GEP,label)
MiRNAScore&lt;-IdentifyMiRNA(GDEscore,nperm=5, damping=0.90)
</code></pre>

<hr>
<h2 id='MirnaData'>An environment variable that includes some example data</h2><span id='topic+MirnaData'></span>

<h3>Description</h3>

<p>An environment variable that includes some example data.
matirx_mirna_go_inter:A matrix of Jaccard score between MiRNAs and GOMF.
matirx_mirna_go_jaccard:A matrix consisting of genes shared by MiRNAs targets and GOMF.
MiRNAScore:a ranked list of strong and weak associations with cancer.
MF_Target:GOMF and corresponding targets.
MiRNA_Target:MiRNAs and corresponding targets.
zscore_BRCA:An example gene expression profile.
label:A vector representing the label of the sample of BRCA, where &quot;1&quot; is the disease sample and &quot;0&quot; is the normal sample.
survival:A dataframe including three columns which are sample, status, and time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MirnaData
</code></pre>


<h3>Format</h3>

<p>An environment variable
</p>

<hr>
<h2 id='MutiMiRNA_CRModel'>MutiMiRNA_CRModel</h2><span id='topic+MutiMiRNA_CRModel'></span>

<h3>Description</h3>

<p>Function &quot;MutiMiRNA_CRModel&quot; can build a multivariate Cox model through integrating the models constructed separately based on different mirna targets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MutiMiRNA_CRModel(ExpData, MiRNAs,SurvivalData,cutoff.point=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MutiMiRNA_CRModel_+3A_expdata">ExpData</code></td>
<td>
<p>A gene expression profile of interest (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="MutiMiRNA_CRModel_+3A_mirnas">MiRNAs</code></td>
<td>
<p>An interest miRNA vector.</p>
</td></tr>
<tr><td><code id="MutiMiRNA_CRModel_+3A_survivaldata">SurvivalData</code></td>
<td>
<p>Survival data (the column names are: &quot;sample&quot;, &quot;status&quot;, and &quot;time&quot;) corresponding to the samples in gene expression profile of interest.</p>
</td></tr>
<tr><td><code id="MutiMiRNA_CRModel_+3A_cutoff.point">cutoff.point</code></td>
<td>
<p>A numeric value used to divide high-risk and low-risk groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list includes a data frame with seven parts those are &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;, &quot;target gene expression&quot;, &quot;risk score&quot;, &quot;group&quot;, and a data frame with five columns those are &quot;Gene&quot;, &quot;HR&quot;, &quot;HR.95L&quot;, &quot;HR.95H&quot;, &quot;beta&quot;, and &quot;P-value&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
survival&lt;-GetData_Mirna("survival")
MiRNAs&lt;-c("hsa-miR-21-5p","hsa-miR-26a-5p","hsa-miR-369-5p","hsa-miR-1238-3p","hsa-miR-10b-5p")
# Run the function
MutiMiRNA_CRData&lt;-MutiMiRNA_CRModel(GEP,
MiRNAs,survival,cutoff.point=NULL)
</code></pre>

<hr>
<h2 id='PlotForest'>PlotForest</h2><span id='topic+PlotForest'></span>

<h3>Description</h3>

<p>Function &quot;PlotForest&quot; can visualize the result of Cox regression analysis through forest plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotForest(MiRNA_CRData,g.pos = 2,b.size = 3,col = c("#FE0101", "#1C61B6", "#A4A4A4"),
lwd.zero = 2,lwd.ci = 3,x.lab = "Hazard Ratio Plot")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotForest_+3A_mirna_crdata">MiRNA_CRData</code></td>
<td>
<p>A list includes a data frame with seven parts those are &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;, &quot;target genes expression&quot;, &quot;risk score&quot;, &quot;group&quot;, and a data frame with five columns those are &quot;Gene&quot;, &quot;HR&quot;, &quot;HR.95L&quot;, &quot;HR.95H&quot;, &quot;beta&quot;, and &quot;P-value&quot;.</p>
</td></tr>
<tr><td><code id="PlotForest_+3A_g.pos">g.pos</code></td>
<td>
<p>The position of the graph element within the table of text. The position can be 1-(ncol(labeltext) + 1). You can also choose set the position to &quot;left&quot; or &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="PlotForest_+3A_b.size">b.size</code></td>
<td>
<p>Override the default box size based on precision.</p>
</td></tr>
<tr><td><code id="PlotForest_+3A_col">col</code></td>
<td>
<p>Set the colors for all the elements in the plot.</p>
</td></tr>
<tr><td><code id="PlotForest_+3A_lwd.zero">lwd.zero</code></td>
<td>
<p>lwd for the vertical line that gives the no-effect line, see gpar.</p>
</td></tr>
<tr><td><code id="PlotForest_+3A_lwd.ci">lwd.ci</code></td>
<td>
<p>lwd for the confidence bands, see gpar.</p>
</td></tr>
<tr><td><code id="PlotForest_+3A_x.lab">x.lab</code></td>
<td>
<p>x-axis label.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Forest maps associated with the Cox risk model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
survival&lt;-GetData_Mirna("survival")
MiRNAs&lt;-c("hsa-miR-21-5p","hsa-miR-26a-5p","hsa-miR-369-5p","hsa-miR-1238-3p","hsa-miR-10b-5p")
# Run the function
SingleMiRNA_CRData&lt;-SingleMiRNA_CRModel(GEP,
"hsa-miR-21-5p",survival,cutoff.point=NULL)
PlotForest(SingleMiRNA_CRData)
MutiMiRNA_CRData&lt;-MutiMiRNA_CRModel(GEP,
MiRNAs,survival,cutoff.point=NULL)
PlotForest(MutiMiRNA_CRData)
</code></pre>

<hr>
<h2 id='PlotHeatmap'>PlotHeatmap</h2><span id='topic+PlotHeatmap'></span>

<h3>Description</h3>

<p>The function &quot;PlotHeatmap&quot; is used to plot a heat map of miRNA targets expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotHeatmap(MiRNA_CRData,yaxis=c(-2,2),scale="row",
cluster.rows=FALSE,cluster.cols=FALSE,show.colnames=FALSE,
ann_colors=c("#ef6d6d","#5470c6"),col=c("#ef6d6d","#5470c6"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotHeatmap_+3A_mirna_crdata">MiRNA_CRData</code></td>
<td>
<p>A list includes a data frame with seven parts those are &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;, &quot;target genes expression&quot;, &quot;risk score&quot;, &quot;group&quot;, and a data frame with five columns those are &quot;Gene&quot;, &quot;HR&quot;, &quot;HR.95L&quot;, &quot;HR.95H&quot;, &quot;beta&quot;, and &quot;P-value&quot;.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_yaxis">yaxis</code></td>
<td>
<p>The upper and lower limits of this heat map.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_scale">scale</code></td>
<td>
<p>character indicating if the values should be centered and scaled in either the row direction or the column direction, or none. Corresponding values are &quot;row&quot;, &quot;column&quot; and &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_cluster.rows">cluster.rows</code></td>
<td>
<p>A logical value that represents whether row clustering is used.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_cluster.cols">cluster.cols</code></td>
<td>
<p>A logical value that represents whether col clustering is used.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_show.colnames">show.colnames</code></td>
<td>
<p>This parameter controls whether column names are displayed.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_ann_colors">ann_colors</code></td>
<td>
<p>Vector of colors used to define groups.</p>
</td></tr>
<tr><td><code id="PlotHeatmap_+3A_col">col</code></td>
<td>
<p>Vector of colors used in the heatmap.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A heat map of miRNA targets expression.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
survival&lt;-GetData_Mirna("survival")
MiRNAs&lt;-c("hsa-miR-21-5p","hsa-miR-26a-5p","hsa-miR-369-5p","hsa-miR-1238-3p","hsa-miR-10b-5p")
# Run the function
SingleMiRNA_CRData&lt;-SingleMiRNA_CRModel(GEP,
"hsa-miR-21-5p",survival,cutoff.point=NULL)
PlotHeatmap(SingleMiRNA_CRData)
MutiMiRNA_CRData&lt;-MutiMiRNA_CRModel(GEP,
MiRNAs,survival,cutoff.point=NULL)
PlotHeatmap(MutiMiRNA_CRData)
</code></pre>

<hr>
<h2 id='PlotScatter'>PlotScatter</h2><span id='topic+PlotScatter'></span>

<h3>Description</h3>

<p>Function &quot;PlotScatter&quot; is used to plot a scatter diagram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotScatter(MiRNA_CRData,status.0='Alive',status.1='Dead',
TitleYlab_A='Risk Score',TitleYlab_B='Survival Time',TitleXlab='Rank',
TitleLegend_A='Risk Group',TitleLegend_B='Status',
color.A=c(low='blue',high='red'),color.B=c(status.0='blue',status.1='red'))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotScatter_+3A_mirna_crdata">MiRNA_CRData</code></td>
<td>
<p>A list includes a data frame with seven parts those are &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;, &quot;target genes expression&quot;, &quot;risk score&quot;, &quot;group&quot;, and a data frame with five columns those are &quot;Gene&quot;, &quot;HR&quot;, &quot;HR.95L&quot;, &quot;HR.95H&quot;, &quot;beta&quot;, and &quot;P-value&quot;.</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_status.0">status.0</code></td>
<td>
<p>string. Code for event 0. Default is 'Alive'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_status.1">status.1</code></td>
<td>
<p>string. Code for event 1. Default is 'Dead'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_titleylab_a">TitleYlab_A</code></td>
<td>
<p>string, y-lab title for figure A. Default is 'Riskscore'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_titleylab_b">TitleYlab_B</code></td>
<td>
<p>string, y-lab title for figure B. Default is 'Survival Time'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_titlexlab">TitleXlab</code></td>
<td>
<p>string, x-lab title for figure B. Default is 'Rank'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_titlelegend_a">TitleLegend_A</code></td>
<td>
<p>string, legend title for figure A. Default is 'Risk Group'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_titlelegend_b">TitleLegend_B</code></td>
<td>
<p>string, legend title for figure B. Default is 'Status'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_color.a">color.A</code></td>
<td>
<p>color for figure A. Default is low = 'blue', high = 'red'</p>
</td></tr>
<tr><td><code id="PlotScatter_+3A_color.b">color.B</code></td>
<td>
<p>color for figure B. Default is status.0 = 'blue', status.1 = 'red'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A riskscore picture
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
survival&lt;-GetData_Mirna("survival")
MiRNAs&lt;-c("hsa-miR-21-5p","hsa-miR-26a-5p","hsa-miR-369-5p","hsa-miR-1238-3p","hsa-miR-10b-5p")
# Run the function
SingleMiRNA_CRData&lt;-SingleMiRNA_CRModel(GEP,
"hsa-miR-21-5p",survival,cutoff.point=NULL)
PlotScatter(SingleMiRNA_CRData)
MutiMiRNA_CRData&lt;-MutiMiRNA_CRModel(GEP,
MiRNAs,survival,cutoff.point=NULL)
PlotScatter(MutiMiRNA_CRData)
</code></pre>

<hr>
<h2 id='PlotSurvival'>PlotSurvival</h2><span id='topic+PlotSurvival'></span>

<h3>Description</h3>

<p>Function &quot;PlotSurvival&quot; is used to draw a Kaplan-Meier curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotSurvival(MiRNA_CRData,colors=c("#ef6d6d","#5470c6"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotSurvival_+3A_mirna_crdata">MiRNA_CRData</code></td>
<td>
<p>A list includes a data frame with seven parts those are &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;, &quot;target genes expression&quot;, &quot;risk score&quot;, &quot;group&quot;, and a data frame with five columns those are &quot;Gene&quot;, &quot;HR&quot;, &quot;HR.95L&quot;, &quot;HR.95H&quot;, &quot;beta&quot;, and &quot;P-value&quot;.</p>
</td></tr>
<tr><td><code id="PlotSurvival_+3A_colors">colors</code></td>
<td>
<p>Vector of colors used to define groups.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A survival curve of a data set.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
survival&lt;-GetData_Mirna("survival")
MiRNAs&lt;-c("hsa-miR-21-5p","hsa-miR-26a-5p","hsa-miR-369-5p","hsa-miR-1238-3p","hsa-miR-10b-5p")
# Run the function
SingleMiRNA_CRData&lt;-SingleMiRNA_CRModel(GEP,
"hsa-miR-21-5p",survival,cutoff.point=NULL)
PlotSurvival(SingleMiRNA_CRData)
MutiMiRNA_CRData&lt;-MutiMiRNA_CRModel(GEP,
MiRNAs,survival,cutoff.point=NULL)
PlotSurvival(MutiMiRNA_CRData)
</code></pre>

<hr>
<h2 id='SingleMiRNA_CRModel'>SingleMiRNA_CRModel</h2><span id='topic+SingleMiRNA_CRModel'></span>

<h3>Description</h3>

<p>Function &quot;SingleMiRNA_CRModel&quot; uses survival data to build a multivariate Cox model using the targets of a single miRNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SingleMiRNA_CRModel(ExpData,MiRNA,cutoff.point=NULL,SurvivalData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SingleMiRNA_CRModel_+3A_expdata">ExpData</code></td>
<td>
<p>A gene expression profile of interest (rows are genes, columns are samples).</p>
</td></tr>
<tr><td><code id="SingleMiRNA_CRModel_+3A_mirna">MiRNA</code></td>
<td>
<p>A miRNA ID.</p>
</td></tr>
<tr><td><code id="SingleMiRNA_CRModel_+3A_cutoff.point">cutoff.point</code></td>
<td>
<p>A numeric value is used to divide high-risk and low-risk groups.</p>
</td></tr>
<tr><td><code id="SingleMiRNA_CRModel_+3A_survivaldata">SurvivalData</code></td>
<td>
<p>Survival data (the column names are: &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;) corresponding to samples in the gene expression profile of interest.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list includes a data frame with seven parts those are &quot;sample&quot;, &quot;status&quot;, &quot;time&quot;, &quot;target genes expression&quot;, &quot;risk score&quot;, &quot;group&quot;, and a dataframe with five columns those are &quot;Gene&quot;, &quot;HR&quot;, &quot;HR.95L&quot;, &quot;HR.95H&quot;, &quot;beta&quot;, and &quot;P-value&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Obtain the example data
GEP&lt;-GetData_Mirna("GEP")
survival&lt;-GetData_Mirna("survival")
# Run the function
SingleMiRNA_CRData&lt;-SingleMiRNA_CRModel(GEP,
"hsa-miR-21-5p",cutoff.point=NULL,survival)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
