<!DOCTYPE html><html lang="en"><head><title>Help for package idopNetwork</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {idopNetwork}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bifun_clu'><p>main function for bifunctional clustering</p></a></li>
<li><a href='#bifun_clu_convert'><p>convert result of bifunctional clustering result</p></a></li>
<li><a href='#bifun_clu_parallel'><p>parallel version for functional clustering</p></a></li>
<li><a href='#bifun_clu_plot'><p>bifunctional clustering plot</p></a></li>
<li><a href='#biget_par_int'><p>acquire initial parameters for functional clustering</p></a></li>
<li><a href='#bipower_equation_plot'><p>plot power equation fitting results for bi-variate model</p></a></li>
<li><a href='#biQ_function'><p>Q-function to replace log-likelihood function</p></a></li>
<li><a href='#biqdODE_plot_all'><p>plot all decompose plot for two data</p></a></li>
<li><a href='#biqdODE_plot_base'><p>plot single decompose plot for two data</p></a></li>
<li><a href='#darken'><p>make color more dark</p></a></li>
<li><a href='#data_cleaning'><p>remove observation with too many 0 values</p></a></li>
<li><a href='#data_match'><p>match power_equation fit result for bi-variate model</p></a></li>
<li><a href='#fun_clu'><p>main function for functional clustering</p></a></li>
<li><a href='#fun_clu_BIC'><p>plot BIC results for functional clustering</p></a></li>
<li><a href='#fun_clu_convert'><p>convert result of functional clustering result</p></a></li>
<li><a href='#fun_clu_parallel'><p>parallel version for functional clustering</p></a></li>
<li><a href='#fun_clu_plot'><p>functional clustering plot</p></a></li>
<li><a href='#fun_clu_select'><p>select result of functional clustering result</p></a></li>
<li><a href='#get_biSAD1'><p>generate biSAD1 covariance matrix</p></a></li>
<li><a href='#get_interaction'><p>Lasso-based variable selection</p></a></li>
<li><a href='#get_legendre_matrix'><p>generate legendre matrix</p></a></li>
<li><a href='#get_legendre_par'><p>use legendre polynomials to fit a given data</p></a></li>
<li><a href='#get_mu'><p>curve fit with modified logistic function</p></a></li>
<li><a href='#get_mu2'><p>generate mean vectors with ck and stress condition</p></a></li>
<li><a href='#get_par_int'><p>acquire initial parameters for functional clustering</p></a></li>
<li><a href='#get_SAD1_covmatrix'><p>generate standard SAD1 covariance matrix</p></a></li>
<li><a href='#gut_microbe'><p>gut microbe OTU data (species level)</p></a></li>
<li><a href='#legendre_fit'><p>generate curve based on legendre polynomials</p></a></li>
<li><a href='#logsumexp'><p>calculate log-sum-exp values</p></a></li>
<li><a href='#mustard_microbe'><p>mustard microbe OTU data</p></a></li>
<li><a href='#network_conversion'><p>convert ODE results(ODE_solving2) to basic network plot table</p></a></li>
<li><a href='#network_maxeffect'><p>convert ODE results(ODE_solving2) to basic network plot table</p></a></li>
<li><a href='#network_plot'><p>generate network plot</p></a></li>
<li><a href='#normalization'><p>min-max normalization</p></a></li>
<li><a href='#power_equation'><p>use power equation parameters to generate y values</p></a></li>
<li><a href='#power_equation_all'><p>use power equation to fit observed values</p></a></li>
<li><a href='#power_equation_base'><p>use power equation to fit observed values</p></a></li>
<li><a href='#power_equation_fit'><p>use power equation to fit given dataset</p></a></li>
<li><a href='#power_equation_plot'><p>plot power equation fitting results</p></a></li>
<li><a href='#Q_function'><p>Q-function to replace log-likelihood function</p></a></li>
<li><a href='#qdODE_all'><p>wrapper for qdODE model</p></a></li>
<li><a href='#qdODE_fit'><p>legendre polynomials fit to qdODE model</p></a></li>
<li><a href='#qdODE_ls'><p>least-square fit for qdODE model</p></a></li>
<li><a href='#qdODE_parallel'><p>wrapper for qdODE_all in parallel version</p></a></li>
<li><a href='#qdODE_plot_all'><p>plot all decompose plot</p></a></li>
<li><a href='#qdODE_plot_base'><p>plot single decompose plot</p></a></li>
<li><a href='#qdODEmod'><p>quasi-dynamic lotka volterra model</p></a></li>
<li><a href='#qdODEplot_convert'><p>convert qdODE results to plot data</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>A Network Tool to Dissect Spatial Community Ecology</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Ang Dong</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ang Dong &lt;fantasys05227@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Most existing approaches for network reconstruction can only infer an overall network 
	and, also, fail to capture a complete set of network properties. To address these issues, 
	a new model has been developed, which converts static data into their 'dynamic' form. 
	'idopNetwork' is an 'R' interface to this model, it can inferring informative, dynamic, 
	omnidirectional and personalized networks. For more information on functional 
	clustering part, see Kim et al. (2008) &lt;<a href="https://doi.org/10.1534%2Fgenetics.108.093690">doi:10.1534/genetics.108.093690</a>&gt;, 
	Wang et al. (2011) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbr032">doi:10.1093/bib/bbr032</a>&gt;. For more information on our model, 
	see Chen et al. (2019) &lt;<a href="https://doi.org/10.1038%2Fs41540-019-0116-1">doi:10.1038/s41540-019-0116-1</a>&gt;, and Cao et al. (2022) 
	&lt;<a href="https://doi.org/10.1080%2F19490976.2022.2106103">doi:10.1080/19490976.2022.2106103</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, stats, mvtnorm, orthopolynom, parallel, deSolve,
ggplot2, reshape2, glmnet, igraph, scales, patchwork</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.50)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>covr, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cxzdsa2332/idopNetwork">https://github.com/cxzdsa2332/idopNetwork</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cxzdsa2332/idopNetwork/issues">https://github.com/cxzdsa2332/idopNetwork/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-18 06:29:49 UTC; ddaa</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-18 06:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bifun_clu'>main function for bifunctional clustering</h2><span id='topic+bifun_clu'></span>

<h3>Description</h3>

<p>main function for bifunctional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bifun_clu(
  data1,
  data2,
  k,
  Time1 = NULL,
  Time2 = NULL,
  trans = log10,
  inv.cov = NULL,
  initial.pars = NULL,
  iter.max = 100,
  parscale = 0.001
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bifun_clu_+3A_data1">data1</code></td>
<td>
<p>matrix or data for cluster</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_data2">data2</code></td>
<td>
<p>matrix or data for cluster</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_k">k</code></td>
<td>
<p>vector for the cluster number</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_time1">Time1</code></td>
<td>
<p>vector for the time point</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_time2">Time2</code></td>
<td>
<p>vector for the time point</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_trans">trans</code></td>
<td>
<p>indicate log/log2/log10 transform dataset</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_inv.cov">inv.cov</code></td>
<td>
<p>matrix for directly solve cov matrix, default not given(currently not available)</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_initial.pars">initial.pars</code></td>
<td>
<p>vector for manual give initial parameters, default not given</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_iter.max">iter.max</code></td>
<td>
<p>scales control iteration for EM algorithm</p>
</td></tr>
<tr><td><code id="bifun_clu_+3A_parscale">parscale</code></td>
<td>
<p>scales control parameters scales for cov pars</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the initial parameters for functional clustering
</p>

<hr>
<h2 id='bifun_clu_convert'>convert result of bifunctional clustering result</h2><span id='topic+bifun_clu_convert'></span>

<h3>Description</h3>

<p>convert result of bifunctional clustering result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bifun_clu_convert(result, best.k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bifun_clu_convert_+3A_result">result</code></td>
<td>
<p>list directly from bifun_clu_parallel function</p>
</td></tr>
<tr><td><code id="bifun_clu_convert_+3A_best.k">best.k</code></td>
<td>
<p>scale of BIC-determined cluster number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain module data and fitted data
</p>

<hr>
<h2 id='bifun_clu_parallel'>parallel version for functional clustering</h2><span id='topic+bifun_clu_parallel'></span>

<h3>Description</h3>

<p>parallel version for functional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bifun_clu_parallel(
  data1,
  data2,
  Time1 = NULL,
  Time2 = NULL,
  trans = log10,
  start,
  end,
  iter.max = 100,
  thread = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bifun_clu_parallel_+3A_data1">data1</code></td>
<td>
<p>data for cluster</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_data2">data2</code></td>
<td>
<p>data for cluster</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_time1">Time1</code></td>
<td>
<p>vector for the time point</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_time2">Time2</code></td>
<td>
<p>vector for the time point</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_trans">trans</code></td>
<td>
<p>indicate log/log2/log10 transform dataset</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_start">start</code></td>
<td>
<p>vector for the minimum cluster number</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_end">end</code></td>
<td>
<p>vector for the maximum cluster number</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_iter.max">iter.max</code></td>
<td>
<p>scales control iteration for EM algorithm</p>
</td></tr>
<tr><td><code id="bifun_clu_parallel_+3A_thread">thread</code></td>
<td>
<p>scales for how many thread used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the initial parameters for functional clustering
</p>

<hr>
<h2 id='bifun_clu_plot'>bifunctional clustering plot</h2><span id='topic+bifun_clu_plot'></span>

<h3>Description</h3>

<p>bifunctional clustering plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bifun_clu_plot(
  result,
  best.k,
  label = 10,
  degree = 1/4,
  show.legend = FALSE,
  color1 = "#38E54D",
  color2 = "#FF8787"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bifun_clu_plot_+3A_result">result</code></td>
<td>
<p>list directly from bifun_clu_parallel function</p>
</td></tr>
<tr><td><code id="bifun_clu_plot_+3A_best.k">best.k</code></td>
<td>
<p>scale of BIC-determined cluster number</p>
</td></tr>
<tr><td><code id="bifun_clu_plot_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="bifun_clu_plot_+3A_degree">degree</code></td>
<td>
<p>scalar control transparency degree</p>
</td></tr>
<tr><td><code id="bifun_clu_plot_+3A_show.legend">show.legend</code></td>
<td>
<p>show legend or not</p>
</td></tr>
<tr><td><code id="bifun_clu_plot_+3A_color1">color1</code></td>
<td>
<p>Hex Color Codes for first data</p>
</td></tr>
<tr><td><code id="bifun_clu_plot_+3A_color2">color2</code></td>
<td>
<p>Hex Color Codes for second data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>functional clustering plot
</p>

<hr>
<h2 id='biget_par_int'>acquire initial parameters for functional clustering</h2><span id='topic+biget_par_int'></span>

<h3>Description</h3>

<p>acquire initial parameters for functional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biget_par_int(X, k, times1, times2, n1, n2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="biget_par_int_+3A_x">X</code></td>
<td>
<p>matrix for cluster</p>
</td></tr>
<tr><td><code id="biget_par_int_+3A_k">k</code></td>
<td>
<p>vector for the cluster number</p>
</td></tr>
<tr><td><code id="biget_par_int_+3A_times1">times1</code></td>
<td>
<p>vector for the x values or time points</p>
</td></tr>
<tr><td><code id="biget_par_int_+3A_times2">times2</code></td>
<td>
<p>vector for the x values or time points</p>
</td></tr>
<tr><td><code id="biget_par_int_+3A_n1">n1</code></td>
<td>
<p>scalar for number of column contain first trait/location etc</p>
</td></tr>
<tr><td><code id="biget_par_int_+3A_n2">n2</code></td>
<td>
<p>scalar for number of column contain second trait/location etc</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the initial parameters for functional clustering
</p>

<hr>
<h2 id='bipower_equation_plot'>plot power equation fitting results for bi-variate model</h2><span id='topic+bipower_equation_plot'></span>

<h3>Description</h3>

<p>plot power equation fitting results for bi-variate model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bipower_equation_plot(
  result,
  label = 10,
  n = 9,
  show.legend = FALSE,
  color1 = "#38E54D",
  color2 = "#FF8787"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bipower_equation_plot_+3A_result">result</code></td>
<td>
<p>list object returned from data_match</p>
</td></tr>
<tr><td><code id="bipower_equation_plot_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="bipower_equation_plot_+3A_n">n</code></td>
<td>
<p>scales for how many subplots needed</p>
</td></tr>
<tr><td><code id="bipower_equation_plot_+3A_show.legend">show.legend</code></td>
<td>
<p>show legend or not</p>
</td></tr>
<tr><td><code id="bipower_equation_plot_+3A_color1">color1</code></td>
<td>
<p>Hex Color Codes for first data</p>
</td></tr>
<tr><td><code id="bipower_equation_plot_+3A_color2">color2</code></td>
<td>
<p>Hex Color Codes for second data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot show power curve fitting result
</p>

<hr>
<h2 id='biQ_function'>Q-function to replace log-likelihood function</h2><span id='topic+biQ_function'></span>

<h3>Description</h3>

<p>Q-function to replace log-likelihood function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biQ_function(par, prob_log, omega_log, X, k, n1, n2, times1, times2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="biQ_function_+3A_par">par</code></td>
<td>
<p>numeric vector for parameters need to be estimated</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_prob_log">prob_log</code></td>
<td>
<p>mixture component weights(log)</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_omega_log">omega_log</code></td>
<td>
<p>latent variables(log)</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_x">X</code></td>
<td>
<p>matrix for cluster</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_k">k</code></td>
<td>
<p>vector for the cluster number</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_n1">n1</code></td>
<td>
<p>scalar for number of column contain first trait/location etc</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_n2">n2</code></td>
<td>
<p>scalar for number of column contain second trait/location etc</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_times1">times1</code></td>
<td>
<p>vector for the x values or time points</p>
</td></tr>
<tr><td><code id="biQ_function_+3A_times2">times2</code></td>
<td>
<p>vector for the x values or time points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the Loglikelihood value
</p>

<hr>
<h2 id='biqdODE_plot_all'>plot all decompose plot for two data</h2><span id='topic+biqdODE_plot_all'></span>

<h3>Description</h3>

<p>plot all decompose plot for two data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biqdODE_plot_all(
  result1,
  result2,
  label = 10,
  show.legend = FALSE,
  remove.label = TRUE,
  nrow = NULL,
  ncol = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="biqdODE_plot_all_+3A_result1">result1</code></td>
<td>
<p>list of qdODE all for first data</p>
</td></tr>
<tr><td><code id="biqdODE_plot_all_+3A_result2">result2</code></td>
<td>
<p>list of qdODE all for second data</p>
</td></tr>
<tr><td><code id="biqdODE_plot_all_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="biqdODE_plot_all_+3A_show.legend">show.legend</code></td>
<td>
<p>to show legend</p>
</td></tr>
<tr><td><code id="biqdODE_plot_all_+3A_remove.label">remove.label</code></td>
<td>
<p>to remove x and y label</p>
</td></tr>
<tr><td><code id="biqdODE_plot_all_+3A_nrow">nrow</code></td>
<td>
<p>scalar for subplot row number</p>
</td></tr>
<tr><td><code id="biqdODE_plot_all_+3A_ncol">ncol</code></td>
<td>
<p>scalar for subplot column number</p>
</td></tr>
</table>

<hr>
<h2 id='biqdODE_plot_base'>plot single decompose plot for two data</h2><span id='topic+biqdODE_plot_base'></span>

<h3>Description</h3>

<p>plot single decompose plot for two data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biqdODE_plot_base(
  result1,
  result2,
  label = 10,
  show.legend = FALSE,
  remove.label = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="biqdODE_plot_base_+3A_result1">result1</code></td>
<td>
<p>list of qdODE all for first data</p>
</td></tr>
<tr><td><code id="biqdODE_plot_base_+3A_result2">result2</code></td>
<td>
<p>list of qdODE all for second data</p>
</td></tr>
<tr><td><code id="biqdODE_plot_base_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="biqdODE_plot_base_+3A_show.legend">show.legend</code></td>
<td>
<p>to show legend</p>
</td></tr>
<tr><td><code id="biqdODE_plot_base_+3A_remove.label">remove.label</code></td>
<td>
<p>to remove x and y label</p>
</td></tr>
</table>

<hr>
<h2 id='darken'>make color more dark</h2><span id='topic+darken'></span>

<h3>Description</h3>

<p>make color more dark
</p>


<h3>Usage</h3>

<pre><code class='language-R'>darken(color, factor = 1.2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="darken_+3A_color">color</code></td>
<td>
<p>hex color code</p>
</td></tr>
<tr><td><code id="darken_+3A_factor">factor</code></td>
<td>
<p>scalar for darken level</p>
</td></tr>
</table>


<h3>Value</h3>

<p>darkened hex color code
</p>


<h3>Examples</h3>

<pre><code class='language-R'>darken("#FF0000")
</code></pre>

<hr>
<h2 id='data_cleaning'>remove observation with too many 0 values</h2><span id='topic+data_cleaning'></span>

<h3>Description</h3>

<p>remove observation with too many 0 values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_cleaning(data, x = round(ncol(data) * 0.3))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="data_cleaning_+3A_data">data</code></td>
<td>
<p>dataframe of imported dataset, must have first column as ID</p>
</td></tr>
<tr><td><code id="data_cleaning_+3A_x">x</code></td>
<td>
<p>scales indicate how many 0 to remove</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe without too many 0 observations
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_cleaning(matrix(c(c(0,1,1,0,0,1,1), c(2,1,0,3,5,2,2), c(1,1,3,2,4,5,1)), 3, 7), 2)
</code></pre>

<hr>
<h2 id='data_match'>match power_equation fit result for bi-variate model</h2><span id='topic+data_match'></span>

<h3>Description</h3>

<p>match power_equation fit result for bi-variate model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_match(result1, result2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="data_match_+3A_result1">result1</code></td>
<td>
<p>list object from power_equation fit</p>
</td></tr>
<tr><td><code id="data_match_+3A_result2">result2</code></td>
<td>
<p>list object from power_equation fit</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a id match list for input dataset
</p>

<hr>
<h2 id='fun_clu'>main function for functional clustering</h2><span id='topic+fun_clu'></span>

<h3>Description</h3>

<p>main function for functional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fun_clu(
  data,
  k,
  Time = NULL,
  trans = log10,
  inv.cov = NULL,
  initial.pars = NULL,
  iter.max = 100,
  parscale = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fun_clu_+3A_data">data</code></td>
<td>
<p>matrix or data for cluster</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_k">k</code></td>
<td>
<p>vector for the cluster number</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_time">Time</code></td>
<td>
<p>vector for the time point</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_trans">trans</code></td>
<td>
<p>indicate log/log2/log10 transform dataset</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_inv.cov">inv.cov</code></td>
<td>
<p>matrix for directly solve cov matrix, default not given(currently not available)</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_initial.pars">initial.pars</code></td>
<td>
<p>vector for manual give initial parameters, default not given</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_iter.max">iter.max</code></td>
<td>
<p>scales control iteration for EM algorithm</p>
</td></tr>
<tr><td><code id="fun_clu_+3A_parscale">parscale</code></td>
<td>
<p>scales control parameters scales for cov pars</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the initial parameters for functional clustering
</p>

<hr>
<h2 id='fun_clu_BIC'>plot BIC results for functional clustering</h2><span id='topic+fun_clu_BIC'></span>

<h3>Description</h3>

<p>plot BIC results for functional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fun_clu_BIC(result, crit = "BIC", title = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fun_clu_BIC_+3A_result">result</code></td>
<td>
<p>list directly from fun_clu_parallel function</p>
</td></tr>
<tr><td><code id="fun_clu_BIC_+3A_crit">crit</code></td>
<td>
<p>either BIC or AIC for module selection</p>
</td></tr>
<tr><td><code id="fun_clu_BIC_+3A_title">title</code></td>
<td>
<p>title for the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the BIC plot
</p>

<hr>
<h2 id='fun_clu_convert'>convert result of functional clustering result</h2><span id='topic+fun_clu_convert'></span>

<h3>Description</h3>

<p>convert result of functional clustering result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fun_clu_convert(result, best.k)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fun_clu_convert_+3A_result">result</code></td>
<td>
<p>list directly from fun_clu_parallel function</p>
</td></tr>
<tr><td><code id="fun_clu_convert_+3A_best.k">best.k</code></td>
<td>
<p>scale of BIC-determined cluster number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain module data and fitted data
</p>

<hr>
<h2 id='fun_clu_parallel'>parallel version for functional clustering</h2><span id='topic+fun_clu_parallel'></span>

<h3>Description</h3>

<p>parallel version for functional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fun_clu_parallel(
  data,
  Time = NULL,
  trans = log10,
  start,
  end,
  iter.max = 100,
  thread = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fun_clu_parallel_+3A_data">data</code></td>
<td>
<p>data for cluster</p>
</td></tr>
<tr><td><code id="fun_clu_parallel_+3A_time">Time</code></td>
<td>
<p>vector for the time point</p>
</td></tr>
<tr><td><code id="fun_clu_parallel_+3A_trans">trans</code></td>
<td>
<p>indicate log/log2/log10 transform dataset</p>
</td></tr>
<tr><td><code id="fun_clu_parallel_+3A_start">start</code></td>
<td>
<p>vector for the minimum cluster number</p>
</td></tr>
<tr><td><code id="fun_clu_parallel_+3A_end">end</code></td>
<td>
<p>vector for the maximum cluster number</p>
</td></tr>
<tr><td><code id="fun_clu_parallel_+3A_iter.max">iter.max</code></td>
<td>
<p>scales control iteration for EM algorithm</p>
</td></tr>
<tr><td><code id="fun_clu_parallel_+3A_thread">thread</code></td>
<td>
<p>scales for how many threads used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the initial parameters for functional clustering
</p>

<hr>
<h2 id='fun_clu_plot'>functional clustering plot</h2><span id='topic+fun_clu_plot'></span>

<h3>Description</h3>

<p>functional clustering plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fun_clu_plot(result, best.k, label = 10, degree = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fun_clu_plot_+3A_result">result</code></td>
<td>
<p>list directly from fun_clu_parallel function</p>
</td></tr>
<tr><td><code id="fun_clu_plot_+3A_best.k">best.k</code></td>
<td>
<p>scalar of BIC-determined cluster number</p>
</td></tr>
<tr><td><code id="fun_clu_plot_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="fun_clu_plot_+3A_degree">degree</code></td>
<td>
<p>scalar control transparency degree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>functional clustering plot
</p>

<hr>
<h2 id='fun_clu_select'>select result of functional clustering result</h2><span id='topic+fun_clu_select'></span>

<h3>Description</h3>

<p>select result of functional clustering result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fun_clu_select(result_fit, result_funclu, i)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fun_clu_select_+3A_result_fit">result_fit</code></td>
<td>
<p>list directly from power_equation_fit</p>
</td></tr>
<tr><td><code id="fun_clu_select_+3A_result_funclu">result_funclu</code></td>
<td>
<p>list from fun_clu_convert</p>
</td></tr>
<tr><td><code id="fun_clu_select_+3A_i">i</code></td>
<td>
<p>scale of which cluster selected</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain microbe data and fitted data
</p>

<hr>
<h2 id='get_biSAD1'>generate biSAD1 covariance matrix</h2><span id='topic+get_biSAD1'></span>

<h3>Description</h3>

<p>generate biSAD1 covariance matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_biSAD1(par, n1, n2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_biSAD1_+3A_par">par</code></td>
<td>
<p>vector with four number, first two for ck and the rest for stress</p>
</td></tr>
<tr><td><code id="get_biSAD1_+3A_n1">n1</code></td>
<td>
<p>scalar indicate length of time1</p>
</td></tr>
<tr><td><code id="get_biSAD1_+3A_n2">n2</code></td>
<td>
<p>scalar indicate length of time2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>biSAD1 covariance matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_biSAD1(par=c(2,0.5,2,0.1),n1=4, n2 = 5)
</code></pre>

<hr>
<h2 id='get_interaction'>Lasso-based variable selection</h2><span id='topic+get_interaction'></span>

<h3>Description</h3>

<p>Lasso-based variable selection
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_interaction(data, col, reduction = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_interaction_+3A_data">data</code></td>
<td>
<p>data of clustered results, do not contain cluster column</p>
</td></tr>
<tr><td><code id="get_interaction_+3A_col">col</code></td>
<td>
<p>scalar of which row number selected</p>
</td></tr>
<tr><td><code id="get_interaction_+3A_reduction">reduction</code></td>
<td>
<p>use n/log(n) dimension reduction</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain relationship of each row
</p>

<hr>
<h2 id='get_legendre_matrix'>generate legendre matrix</h2><span id='topic+get_legendre_matrix'></span>

<h3>Description</h3>

<p>generate legendre matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_legendre_matrix(x, legendre_order)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_legendre_matrix_+3A_x">x</code></td>
<td>
<p>vector equal to the x value for legendre polynomials(in this case times)</p>
</td></tr>
<tr><td><code id="get_legendre_matrix_+3A_legendre_order">legendre_order</code></td>
<td>
<p>the order of legendre polynomials</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the polynomials value of each order
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_legendre_matrix(1:14,4)
</code></pre>

<hr>
<h2 id='get_legendre_par'>use legendre polynomials to fit a given data</h2><span id='topic+get_legendre_par'></span>

<h3>Description</h3>

<p>use legendre polynomials to fit a given data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_legendre_par(y, legendre_order, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_legendre_par_+3A_y">y</code></td>
<td>
<p>vector equal to the y observed data(in this case generic effect)</p>
</td></tr>
<tr><td><code id="get_legendre_par_+3A_legendre_order">legendre_order</code></td>
<td>
<p>scalar of legendre polynomials</p>
</td></tr>
<tr><td><code id="get_legendre_par_+3A_x">x</code></td>
<td>
<p>vector equal to the x value for legendre polynomials(in this case times)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the polynomials coefficients
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_legendre_par(14:1,4,1:14)
</code></pre>

<hr>
<h2 id='get_mu'>curve fit with modified logistic function</h2><span id='topic+get_mu'></span>

<h3>Description</h3>

<p>curve fit with modified logistic function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mu(mu_par, times)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mu_+3A_mu_par">mu_par</code></td>
<td>
<p>vector with five number</p>
</td></tr>
<tr><td><code id="get_mu_+3A_times">times</code></td>
<td>
<p>vector of time point</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector with the same length to times
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_mu(mu_par = 1:5, times = 1:14)
</code></pre>

<hr>
<h2 id='get_mu2'>generate mean vectors with ck and stress condition</h2><span id='topic+get_mu2'></span>

<h3>Description</h3>

<p>generate mean vectors with ck and stress condition
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mu2(par, times)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mu2_+3A_par">par</code></td>
<td>
<p>vector with ten number, first five for ck and the rest for stress</p>
</td></tr>
<tr><td><code id="get_mu2_+3A_times">times</code></td>
<td>
<p>vector of time point</p>
</td></tr>
</table>


<h3>Value</h3>

<p>numeric vector with the double length to times
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_mu2(par = 1:10, times = 1:14)
</code></pre>

<hr>
<h2 id='get_par_int'>acquire initial parameters for functional clustering</h2><span id='topic+get_par_int'></span>

<h3>Description</h3>

<p>acquire initial parameters for functional clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_par_int(X, k, times)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_par_int_+3A_x">X</code></td>
<td>
<p>matrix for cluster</p>
</td></tr>
<tr><td><code id="get_par_int_+3A_k">k</code></td>
<td>
<p>vector for the cluster number</p>
</td></tr>
<tr><td><code id="get_par_int_+3A_times">times</code></td>
<td>
<p>vector for the x values or time points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the initial parameters for functional clustering
</p>

<hr>
<h2 id='get_SAD1_covmatrix'>generate standard SAD1 covariance matrix</h2><span id='topic+get_SAD1_covmatrix'></span>

<h3>Description</h3>

<p>generate standard SAD1 covariance matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_SAD1_covmatrix(par, n)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_SAD1_covmatrix_+3A_par">par</code></td>
<td>
<p>vector with two number for SAD1 covariance matrix</p>
</td></tr>
<tr><td><code id="get_SAD1_covmatrix_+3A_n">n</code></td>
<td>
<p>scalar indicate length of time d</p>
</td></tr>
</table>


<h3>Value</h3>

<p>SAD1 covariance matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_SAD1_covmatrix(par = c(2,0.5), n = 14)
</code></pre>

<hr>
<h2 id='gut_microbe'>gut microbe OTU data (species level)</h2><span id='topic+gut_microbe'></span>

<h3>Description</h3>

<p>The original nucleotide sequences of this study were deposited to the NCBI
Sequence Read Archive under accession number SRP128619.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gut_microbe)
</code></pre>


<h3>Format</h3>

<p>A data frame with 65 rows and 21 column, contain first column as microbe ID:
</p>

<hr>
<h2 id='legendre_fit'>generate curve based on legendre polynomials</h2><span id='topic+legendre_fit'></span>

<h3>Description</h3>

<p>generate curve based on legendre polynomials
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legendre_fit(par, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="legendre_fit_+3A_par">par</code></td>
<td>
<p>vector of legendre polynomials coefficients</p>
</td></tr>
<tr><td><code id="legendre_fit_+3A_x">x</code></td>
<td>
<p>vector equal to the x value for legendre polynomials(in this case times)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the polynomials value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>legendre_fit(rep(1,5),1:14)
</code></pre>

<hr>
<h2 id='logsumexp'>calculate log-sum-exp values</h2><span id='topic+logsumexp'></span>

<h3>Description</h3>

<p>calculate log-sum-exp values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logsumexp(v)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="logsumexp_+3A_v">v</code></td>
<td>
<p>numeric vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log-sum-exp values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logsumexp(c(100, 1000, 10000))
</code></pre>

<hr>
<h2 id='mustard_microbe'>mustard microbe OTU data</h2><span id='topic+mustard_microbe'></span>

<h3>Description</h3>

<p>Wagner, M. R. et al. Host genotype and age shape the leaf and root microbiomes of a wild perennial plant.
Nat. Commun. 7:12151 doi: 10.1038/ncomms12151 (2016)
This dataset is a subset of otuTable97, we select location = JAM, keep samples with both root
and leaf data, and then run data_cleaning first (set x = 50) to reduce size of this data.
Moreover, sample 8_1382 is removed for the outlier reason.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mustard_microbe)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1557 rows and 176 column, contain first column as OTU ID:
</p>

<hr>
<h2 id='network_conversion'>convert ODE results(ODE_solving2) to basic network plot table</h2><span id='topic+network_conversion'></span>

<h3>Description</h3>

<p>convert ODE results(ODE_solving2) to basic network plot table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_conversion(result)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_conversion_+3A_result">result</code></td>
<td>
<p>list result from qsODE_parallel</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with basic information to plot network
</p>

<hr>
<h2 id='network_maxeffect'>convert ODE results(ODE_solving2) to basic network plot table</h2><span id='topic+network_maxeffect'></span>

<h3>Description</h3>

<p>convert ODE results(ODE_solving2) to basic network plot table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_maxeffect(result)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_maxeffect_+3A_result">result</code></td>
<td>
<p>list result from qsODE_parallel</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with basic information to plot network
</p>

<hr>
<h2 id='network_plot'>generate network plot</h2><span id='topic+network_plot'></span>

<h3>Description</h3>

<p>generate network plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_plot(result, title = NULL, maxeffect = NULL, type = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_plot_+3A_result">result</code></td>
<td>
<p>list result from network_conversion</p>
</td></tr>
<tr><td><code id="network_plot_+3A_title">title</code></td>
<td>
<p>text for plot title</p>
</td></tr>
<tr><td><code id="network_plot_+3A_maxeffect">maxeffect</code></td>
<td>
<p>control edge size when compare networks</p>
</td></tr>
<tr><td><code id="network_plot_+3A_type">type</code></td>
<td>
<p>select module effect or microbe effect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>network plot
</p>

<hr>
<h2 id='normalization'>min-max normalization</h2><span id='topic+normalization'></span>

<h3>Description</h3>

<p>min-max normalization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalization(x, z = 0.2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="normalization_+3A_x">x</code></td>
<td>
<p>numeric vector</p>
</td></tr>
<tr><td><code id="normalization_+3A_z">z</code></td>
<td>
<p>scalar add minimum value to avoid 0</p>
</td></tr>
</table>


<h3>Value</h3>

<p>normalized vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>normalization(runif(100,min = -100, max = 100))
</code></pre>

<hr>
<h2 id='power_equation'>use power equation parameters to generate y values</h2><span id='topic+power_equation'></span>

<h3>Description</h3>

<p>use power equation parameters to generate y values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_equation(x, power_par)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="power_equation_+3A_x">x</code></td>
<td>
<p>vector for x values</p>
</td></tr>
<tr><td><code id="power_equation_+3A_power_par">power_par</code></td>
<td>
<p>matrix contain parameters for power equation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>y values for given power equation parameters
</p>


<h3>Examples</h3>

<pre><code class='language-R'>power_equation(c(1,2,3,5,7), matrix(c(2,1,1,2),2,2))
</code></pre>

<hr>
<h2 id='power_equation_all'>use power equation to fit observed values</h2><span id='topic+power_equation_all'></span>

<h3>Description</h3>

<p>use power equation to fit observed values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_equation_all(x, y, maxit = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="power_equation_all_+3A_x">x</code></td>
<td>
<p>vector for x values</p>
</td></tr>
<tr><td><code id="power_equation_all_+3A_y">y</code></td>
<td>
<p>vector for y values</p>
</td></tr>
<tr><td><code id="power_equation_all_+3A_maxit">maxit</code></td>
<td>
<p>numeric value for maximum initial pars try</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nls model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>power_equation_all(c(1,2,3,5,7), c(5,10,15,17,20))
</code></pre>

<hr>
<h2 id='power_equation_base'>use power equation to fit observed values</h2><span id='topic+power_equation_base'></span>

<h3>Description</h3>

<p>use power equation to fit observed values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_equation_base(x, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="power_equation_base_+3A_x">x</code></td>
<td>
<p>vector for x values</p>
</td></tr>
<tr><td><code id="power_equation_base_+3A_y">y</code></td>
<td>
<p>vector for y valyes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nls model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>power_equation_base(c(1,2,3,5,7), c(5,10,15,17,20))
</code></pre>

<hr>
<h2 id='power_equation_fit'>use power equation to fit given dataset</h2><span id='topic+power_equation_fit'></span>

<h3>Description</h3>

<p>use power equation to fit given dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_equation_fit(data, n = 30, trans = log10, thread = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="power_equation_fit_+3A_data">data</code></td>
<td>
<p>cleaned dataframe</p>
</td></tr>
<tr><td><code id="power_equation_fit_+3A_n">n</code></td>
<td>
<p>scales for how many interpolation needed</p>
</td></tr>
<tr><td><code id="power_equation_fit_+3A_trans">trans</code></td>
<td>
<p>indicate log/log2/log10 transform dataset</p>
</td></tr>
<tr><td><code id="power_equation_fit_+3A_thread">thread</code></td>
<td>
<p>scales for how many thread used</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain power equation parameters and fitted data
</p>

<hr>
<h2 id='power_equation_plot'>plot power equation fitting results</h2><span id='topic+power_equation_plot'></span>

<h3>Description</h3>

<p>plot power equation fitting results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>power_equation_plot(result, label = 10, n = 9)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="power_equation_plot_+3A_result">result</code></td>
<td>
<p>list object returned from power_equation_fit</p>
</td></tr>
<tr><td><code id="power_equation_plot_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="power_equation_plot_+3A_n">n</code></td>
<td>
<p>scales for how many subplots needed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot show power curve fitting result
</p>

<hr>
<h2 id='Q_function'>Q-function to replace log-likelihood function</h2><span id='topic+Q_function'></span>

<h3>Description</h3>

<p>Q-function to replace log-likelihood function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Q_function(par, prob_log, omega_log, X, k, times)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Q_function_+3A_par">par</code></td>
<td>
<p>numeric vector for parameters need to be estimated</p>
</td></tr>
<tr><td><code id="Q_function_+3A_prob_log">prob_log</code></td>
<td>
<p>mixture component weights(log)</p>
</td></tr>
<tr><td><code id="Q_function_+3A_omega_log">omega_log</code></td>
<td>
<p>latent variables(log)</p>
</td></tr>
<tr><td><code id="Q_function_+3A_x">X</code></td>
<td>
<p>matrix for cluster</p>
</td></tr>
<tr><td><code id="Q_function_+3A_k">k</code></td>
<td>
<p>vector for the cluster number</p>
</td></tr>
<tr><td><code id="Q_function_+3A_times">times</code></td>
<td>
<p>vector for the x values or time points</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the Loglikelihood value
</p>

<hr>
<h2 id='qdODE_all'>wrapper for qdODE model</h2><span id='topic+qdODE_all'></span>

<h3>Description</h3>

<p>wrapper for qdODE model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODE_all(
  result,
  relationship,
  i,
  init_pars = 1,
  LOP_order = 6,
  method = "ls",
  new_time = NULL,
  n_expand = 100,
  maxit = 1000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODE_all_+3A_result">result</code></td>
<td>
<p>result from power_equation_fit</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_relationship">relationship</code></td>
<td>
<p>list contain variable selection results</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_i">i</code></td>
<td>
<p>scalar for which id used for qdODE solving, must &lt;= nrow</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_init_pars">init_pars</code></td>
<td>
<p>scalar for initial parameters</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_lop_order">LOP_order</code></td>
<td>
<p>scalar of LOP order</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_method">method</code></td>
<td>
<p>scalar of qdODE solving methodm, cuurent only support least square</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_new_time">new_time</code></td>
<td>
<p>vector produce new defined time point</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_n_expand">n_expand</code></td>
<td>
<p>scalar for how many interpolation needed</p>
</td></tr>
<tr><td><code id="qdODE_all_+3A_maxit">maxit</code></td>
<td>
<p>scalar of Optim iteration setting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain variable selection results and LOP parameters for every row
</p>

<hr>
<h2 id='qdODE_fit'>legendre polynomials fit to qdODE model</h2><span id='topic+qdODE_fit'></span>

<h3>Description</h3>

<p>legendre polynomials fit to qdODE model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODE_fit(
  pars,
  data,
  Time,
  power_par,
  LOP_order = 6,
  new_time = NULL,
  n_expand = 100
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODE_fit_+3A_pars">pars</code></td>
<td>
<p>vector of qdODE parameters</p>
</td></tr>
<tr><td><code id="qdODE_fit_+3A_data">data</code></td>
<td>
<p>dataframe of observed data</p>
</td></tr>
<tr><td><code id="qdODE_fit_+3A_time">Time</code></td>
<td>
<p>vector of time point</p>
</td></tr>
<tr><td><code id="qdODE_fit_+3A_power_par">power_par</code></td>
<td>
<p>matrix of power equation parameters for dependent effect</p>
</td></tr>
<tr><td><code id="qdODE_fit_+3A_lop_order">LOP_order</code></td>
<td>
<p>scalar of LOP order</p>
</td></tr>
<tr><td><code id="qdODE_fit_+3A_new_time">new_time</code></td>
<td>
<p>vector produce new defined time point</p>
</td></tr>
<tr><td><code id="qdODE_fit_+3A_n_expand">n_expand</code></td>
<td>
<p>scalar for how many interpolation needed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain legendre polynomials parameters, qdODE values and LOP fitted values
</p>

<hr>
<h2 id='qdODE_ls'>least-square fit for qdODE model</h2><span id='topic+qdODE_ls'></span>

<h3>Description</h3>

<p>least-square fit for qdODE model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODE_ls(pars, data, Time, power_par)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODE_ls_+3A_pars">pars</code></td>
<td>
<p>vector for unknown ODE parameters</p>
</td></tr>
<tr><td><code id="qdODE_ls_+3A_data">data</code></td>
<td>
<p>data contain independent effect as first row and dependent effect</p>
</td></tr>
<tr><td><code id="qdODE_ls_+3A_time">Time</code></td>
<td>
<p>vector of time point</p>
</td></tr>
<tr><td><code id="qdODE_ls_+3A_power_par">power_par</code></td>
<td>
<p>matrix of power equation parameters for dependent effect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>mean-square error
</p>

<hr>
<h2 id='qdODE_parallel'>wrapper for qdODE_all in parallel version</h2><span id='topic+qdODE_parallel'></span>

<h3>Description</h3>

<p>wrapper for qdODE_all in parallel version
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODE_parallel(result, reduction = FALSE, thread = 2, maxit = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODE_parallel_+3A_result">result</code></td>
<td>
<p>result from power_equation_fit</p>
</td></tr>
<tr><td><code id="qdODE_parallel_+3A_reduction">reduction</code></td>
<td>
<p>use n/log(n) dimension reduction</p>
</td></tr>
<tr><td><code id="qdODE_parallel_+3A_thread">thread</code></td>
<td>
<p>scales for how many threads used</p>
</td></tr>
<tr><td><code id="qdODE_parallel_+3A_maxit">maxit</code></td>
<td>
<p>scalar of Optim iteration setting</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contain variable selection results and LOP parameters for every row
</p>

<hr>
<h2 id='qdODE_plot_all'>plot all decompose plot</h2><span id='topic+qdODE_plot_all'></span>

<h3>Description</h3>

<p>plot all decompose plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODE_plot_all(
  result,
  label = 10,
  show.legend = TRUE,
  nrow = NULL,
  ncol = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODE_plot_all_+3A_result">result</code></td>
<td>
<p>list of qdODE parallel</p>
</td></tr>
<tr><td><code id="qdODE_plot_all_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="qdODE_plot_all_+3A_show.legend">show.legend</code></td>
<td>
<p>to show legend</p>
</td></tr>
<tr><td><code id="qdODE_plot_all_+3A_nrow">nrow</code></td>
<td>
<p>scalar for subplot row number</p>
</td></tr>
<tr><td><code id="qdODE_plot_all_+3A_ncol">ncol</code></td>
<td>
<p>scalar for subplot column number</p>
</td></tr>
</table>


<h3>Value</h3>

<p>all effect curve decompose plot
</p>

<hr>
<h2 id='qdODE_plot_base'>plot single decompose plot</h2><span id='topic+qdODE_plot_base'></span>

<h3>Description</h3>

<p>plot single decompose plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODE_plot_base(result, label = 10, show.legend = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODE_plot_base_+3A_result">result</code></td>
<td>
<p>list of qdODE all</p>
</td></tr>
<tr><td><code id="qdODE_plot_base_+3A_label">label</code></td>
<td>
<p>relabel x and y label due to log-transform, set 10 as default</p>
</td></tr>
<tr><td><code id="qdODE_plot_base_+3A_show.legend">show.legend</code></td>
<td>
<p>to show legend</p>
</td></tr>
</table>

<hr>
<h2 id='qdODEmod'>quasi-dynamic lotka volterra model</h2><span id='topic+qdODEmod'></span>

<h3>Description</h3>

<p>quasi-dynamic lotka volterra model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODEmod(Time, State, Pars, power_par)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODEmod_+3A_time">Time</code></td>
<td>
<p>vector of time point</p>
</td></tr>
<tr><td><code id="qdODEmod_+3A_state">State</code></td>
<td>
<p>vector of ODE initial state</p>
</td></tr>
<tr><td><code id="qdODEmod_+3A_pars">Pars</code></td>
<td>
<p>vector for unknown ODE parameters</p>
</td></tr>
<tr><td><code id="qdODEmod_+3A_power_par">power_par</code></td>
<td>
<p>matrix of power equation parameters for dependent effect</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list used in ode function
</p>

<hr>
<h2 id='qdODEplot_convert'>convert qdODE results to plot data</h2><span id='topic+qdODEplot_convert'></span>

<h3>Description</h3>

<p>convert qdODE results to plot data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qdODEplot_convert(result)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qdODEplot_convert_+3A_result">result</code></td>
<td>
<p>list of qdODE all</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
