<!DOCTYPE html><html><head><title>Help for package iDOS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iDOS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#create.counts.table'><p>create.counts.table</p></a></li>
<li><a href='#create.training.validation.split'><p>create.training.validation.split</p></a></li>
<li><a href='#estimate.expression.cna.correlation'><p>estimate.expression.cna.correlation</p></a></li>
<li><a href='#estimate.null.distribution.correlation'><p>estimate.null.distribution.correlation</p></a></li>
<li><a href='#find.DE.features'><p>find.DE.features</p></a></li>
<li><a href='#get.program.defaults'><p>get.program.defaults</p></a></li>
<li><a href='#get.test.data'><p>get.test.data</p></a></li>
<li><a href='#get.top.features'><p>get.top.features</p></a></li>
<li><a href='#load.datasets'><p>load.datasets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrated Discovery of Oncogenic Signatures</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Syed Haider [aut, cre],
  Francesca Buffa [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Syed Haider &lt;Syed.Haider@icr.ac.uk&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0), VennDiagram (&ge; 1.6.5)</td>
</tr>
<tr>
<td>Description:</td>
<td>A method to integrate molecular profiles of cancer patients (gene copy number and mRNA abundance) to 
	identify candidate gain of function alterations. These candidate alterations can be subsequently further tested  
	to discover cancer driver alterations. Briefly, this method tests of genomic correlates of mRNA dysregulation and prioritise 
	those where DNA gains/amplifications are associated with elevated mRNA expression of the same gene. For details see,
	Haider S et al. (2016) "Genomic alterations underlie a pan-cancer metabolic shift associated with tumour hypoxia", Genome Biology, <a href="https://pubmed.ncbi.nlm.nih.gov/27358048/">https://pubmed.ncbi.nlm.nih.gov/27358048/</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-09 20:20:57 UTC; shaider</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-11 18:50:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='create.counts.table'>create.counts.table</h2><span id='topic+create.counts.table'></span>

<h3>Description</h3>

<p>Summary function to collapse the counts of selected (e.g. correlated) features per cancer type into counts table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.counts.table(corr.summary = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create.counts.table_+3A_corr.summary">corr.summary</code></td>
<td>
<p>A list object containing subtype specific selected (e.g. correlated) features. This is the list object returned by <code>estimate.expression.cna.correlation</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of cancer type specific counts
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estimate.expression.cna.correlation">estimate.expression.cna.correlation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load test data
x &lt;- get.test.data(data.types = c("mRNA.T", "CNA"));

# temporary output directory
tmp.output.dir &lt;- tempdir();

# go through each cancer type iteratively and perform mRNA-CNA correlation analysis
correlated.features &lt;- list();
for (cancer.type in names(x$mRNA.T)) {

  # estimate mRNA and CNA correlation for each cancer/disease type
  correlated.features[[cancer.type]] &lt;- estimate.expression.cna.correlation(
    exp.data = x$mRNA.T[[cancer.type]], 
    cna.data.log2 = x$CNA.log2[[cancer.type]], 
    corr.threshold = 0.3, 
    corr.direction = "two.sided", 
    subtypes.metadata = list(
      "subtype.samples.list" = list("All" = colnames(x$mRNA.T[[cancer.type]]))
      ), 
    feature.ids = rownames(x$mRNA.T[[cancer.type]]), 
    cancer.type = cancer.type, 
    data.dir = paste(tmp.output.dir, "/data/", cancer.type, sep = ""),
    graphs.dir = paste(tmp.output.dir, "/graphs/", cancer.type, sep = "")
    );
  }

# create counts table across cancer types
counts.table &lt;- create.counts.table(corr.summary = correlated.features);

</code></pre>

<hr>
<h2 id='create.training.validation.split'>create.training.validation.split</h2><span id='topic+create.training.validation.split'></span>

<h3>Description</h3>

<p>Utility function to create random partitions of a dataset into training and validation sets. If samples are &lt; 200, 66:34; otherwise 50:50 partitions are generated between training and validation sets respectively
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.training.validation.split(
  exp.data = NULL, ann.data = NULL, seed.number = 51214
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create.training.validation.split_+3A_exp.data">exp.data</code></td>
<td>
<p>Feature by sample mRNA abundance matrix</p>
</td></tr>
<tr><td><code id="create.training.validation.split_+3A_ann.data">ann.data</code></td>
<td>
<p>Sample by clinical attribute matrix</p>
</td></tr>
<tr><td><code id="create.training.validation.split_+3A_seed.number">seed.number</code></td>
<td>
<p>Random seed for sampling</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of four matrices expression and two associated clinical matrices (exp.T, ann.T, exp.V and ann.V). One set for training and one for validation
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load test data
x &lt;- get.test.data(data.types = c("mRNA.T", "ann"));

# create training and validation sets
partitioned.datasets &lt;- create.training.validation.split(
  exp.data = x$mRNA.T$BLCA, 
  ann.data = x$ann$BLCA, 
  seed.number = 51214
  );

</code></pre>

<hr>
<h2 id='estimate.expression.cna.correlation'>estimate.expression.cna.correlation</h2><span id='topic+estimate.expression.cna.correlation'></span>

<h3>Description</h3>

<p>Estimate subtype specific correlation between mRNA and CNA profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.expression.cna.correlation(
  exp.data = NULL, 
  cna.data.log2 = NULL, 
  corr.threshold = 0.3, 
  corr.direction = "two.sided", 
  subtypes.metadata = NULL, 
  feature.ids = NULL, 
  cancer.type = NULL, 
  data.dir = NULL, 
  graphs.dir = NULL
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate.expression.cna.correlation_+3A_exp.data">exp.data</code></td>
<td>
<p>Feature by sample mRNA abundance matrix</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_cna.data.log2">cna.data.log2</code></td>
<td>
<p>Feature by sample CNA log ratio matrix</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_corr.threshold">corr.threshold</code></td>
<td>
<p>Threshold for Spearman's Rho to consider a feature as candidate driver</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_corr.direction">corr.direction</code></td>
<td>
<p>Whether to include positively (greater), negatively (less) or both (two.sided) correlated features. Defaults to <code>two.sided</code></p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_subtypes.metadata">subtypes.metadata</code></td>
<td>
<p>Subtypes metadata list of lists. Must contain at least one subtype specific samples using list <code>subtype.samples.list</code>. If no subtypes are present, specify list element &quot;All&quot; with all samples</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_feature.ids">feature.ids</code></td>
<td>
<p>Vector of features to be used to estimate correlation</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_cancer.type">cancer.type</code></td>
<td>
<p>Name of the cancer type or dataset</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_data.dir">data.dir</code></td>
<td>
<p>Path to output directory where mRNA and CNA correlation statistics will be stored</p>
</td></tr>
<tr><td><code id="estimate.expression.cna.correlation_+3A_graphs.dir">graphs.dir</code></td>
<td>
<p>Path to graphs directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of lists containing correlated features per cancer subtype
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load test data
x &lt;- get.test.data(data.types = c("mRNA.T", "CNA"));

# temporary output directory
tmp.output.dir &lt;- tempdir();

# estimate mRNA and CNA correlation
correlated.features &lt;- estimate.expression.cna.correlation(
  exp.data = x$mRNA.T$BLCA, 
  cna.data.log2 = x$CNA.log2$BLCA, 
  corr.threshold = 0.3, 
  corr.direction = "two.sided", 
  subtypes.metadata = list(
    "subtype.samples.list" = list("All" = colnames(x$mRNA.T$BLCA))
    ), 
  feature.ids = rownames(x$mRNA.T$BLCA), 
  cancer.type = "BLCA", 
  data.dir = paste(tmp.output.dir, "/data/BLCA/", sep = ""),
  graphs.dir = paste(tmp.output.dir, "/graphs/BLCA/", sep = "")
  );

</code></pre>

<hr>
<h2 id='estimate.null.distribution.correlation'>estimate.null.distribution.correlation</h2><span id='topic+estimate.null.distribution.correlation'></span>

<h3>Description</h3>

<p>Function to estimate probability of observing correlations as high as observed using a feature list of interest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.null.distribution.correlation(
  exp.data = NULL, 
  cna.data.log2 = NULL, 
  corr.threshold = 0.3, 
  corr.direction = "two.sided", 
  subtypes.metadata = NULL, 
  feature.ids = NULL, 
  observed.correlated.features = NULL, 
  iterations = 50, 
  cancer.type = NULL, 
  data.dir = NULL
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate.null.distribution.correlation_+3A_exp.data">exp.data</code></td>
<td>
<p>Feature by sample mRNA abundance matrix</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_cna.data.log2">cna.data.log2</code></td>
<td>
<p>Feature by sample CNA log ratio matrix</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_corr.threshold">corr.threshold</code></td>
<td>
<p>Threshold for Spearman's Rho to consider a feature as candidate driver</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_corr.direction">corr.direction</code></td>
<td>
<p>Whether to include positively (greater), negatively (less) or both (two.sided) correlated features. Defaults to <code>two.sided</code></p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_subtypes.metadata">subtypes.metadata</code></td>
<td>
<p>Subtypes metadata list. Contains at least subtype specific samples</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_feature.ids">feature.ids</code></td>
<td>
<p>Vector of features to be used to estimate correlation</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_observed.correlated.features">observed.correlated.features</code></td>
<td>
<p>List of features that were found to be correlated for subtypes of a given cancer type</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_iterations">iterations</code></td>
<td>
<p>Number of random permutations for estimating p value</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_cancer.type">cancer.type</code></td>
<td>
<p>Name of the cancer type or dataset</p>
</td></tr>
<tr><td><code id="estimate.null.distribution.correlation_+3A_data.dir">data.dir</code></td>
<td>
<p>Path to output directory where the randomisation results will be stored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>1 if successful
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>See Also</h3>

<p><code><a href="#topic+estimate.expression.cna.correlation">estimate.expression.cna.correlation</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load test data
x &lt;- get.test.data(data.types = c("mRNA.T", "CNA"));

# temporary output directory
tmp.output.dir &lt;- tempdir();

# estimate mRNA and CNA correlation for each cancer/disease type
correlated.features &lt;- estimate.expression.cna.correlation(
  exp.data = x$mRNA.T$BLCA, 
  cna.data.log2 = x$CNA.log2$BLCA, 
  corr.threshold = 0.3, 
  corr.direction = "two.sided", 
  subtypes.metadata = list(
    "subtype.samples.list" = list("All" = colnames(x$mRNA.T$BLCA))
    ), 
  feature.ids = rownames(x$mRNA.T$BLCA), 
  cancer.type = "BLCA", 
  data.dir = paste(tmp.output.dir, "/data/BLCA/", sep = ""),
  graphs.dir = paste(tmp.output.dir, "/graphs/BLCA/", sep = "")
  );

# estimate NULL distribution
estimate.null.distribution.correlation(
  exp.data = x$mRNA.T$BLCA,
  cna.data.log2 = x$CNA.log2$BLCA, 
  corr.threshold = 0.3, 
  corr.direction = "two.sided", 
  subtypes.metadata = list(
    "subtype.samples.list" = list("All" = colnames(x$mRNA.T$BLCA))
    ), 
  feature.ids = rownames(x$mRNA.T$BLCA), 
  observed.correlated.features = correlated.features$correlated.genes.subtypes, 
  iterations = 50, 
  cancer.type = "BLCA", 
  data.dir = paste(tmp.output.dir, "/data/BLCA/", sep = "")
  );

</code></pre>

<hr>
<h2 id='find.DE.features'>find.DE.features</h2><span id='topic+find.DE.features'></span>

<h3>Description</h3>

<p>Funtion to identify differentially expressed/variable features between Tumour (T) and Normal (N) profiles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find.DE.features(
  exp.data.T = NULL, 
  exp.data.N = NULL, 
  feature.ids = NULL,
  test.name = "t.test"
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="find.DE.features_+3A_exp.data.t">exp.data.T</code></td>
<td>
<p>Feature by sample mRNA abundance matrix; tumour samples</p>
</td></tr>
<tr><td><code id="find.DE.features_+3A_exp.data.n">exp.data.N</code></td>
<td>
<p>Feature by sample mRNA abundance matrix; normal/baseline samples</p>
</td></tr>
<tr><td><code id="find.DE.features_+3A_feature.ids">feature.ids</code></td>
<td>
<p>Vector of features to be used to estimate correlation</p>
</td></tr>
<tr><td><code id="find.DE.features_+3A_test.name">test.name</code></td>
<td>
<p>Specify the statistical test name (exactly as it appears in R). Supported tests are <code>t.test</code>, <code>wilcox.test</code>, <code>var.test</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Feature by cancer type matrix of log2 fold change (T vs N) and adjusted P values. P values are estimated through <code>test.name</code>
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+t.test">t.test</a></code>, 
<code><a href="stats.html#topic+wilcox.test">wilcox.test</a></code>, 
<code><a href="stats.html#topic+var.test">var.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load test data
x &lt;- get.test.data(data.types = c("mRNA.T", "mRNA.N"));

# list of features to be assessed for differential expression
feature.ids &lt;- rownames(x$mRNA.T$BLCA);

DE.results &lt;- find.DE.features(
  exp.data.T = x$mRNA.T, 
  exp.data.N = x$mRNA.N, 
  feature.ids = feature.ids,
  test.name = "t.test"
  );

</code></pre>

<hr>
<h2 id='get.program.defaults'>get.program.defaults</h2><span id='topic+get.program.defaults'></span>

<h3>Description</h3>

<p>Get default datasets bundled with package for test runs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.program.defaults()
</code></pre>


<h3>Value</h3>

<p>A list with <code>program.data.dir</code> containing path to example program directory and <code>test.data.dir</code> containing path to example datasets directory
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- get.program.defaults();
</code></pre>

<hr>
<h2 id='get.test.data'>get.test.data</h2><span id='topic+get.test.data'></span>

<h3>Description</h3>

<p>Function to load test data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.test.data(data.types = c("mRNA.T", "ann"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.test.data_+3A_data.types">data.types</code></td>
<td>
<p>Datatypes to be read Valid datatypes are: mRNA.T, mRNA.N, CNA (includes: log2, calls and fractions), annotations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of lists containing datasets and respective molecular profiles as matrices
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
x &lt;- get.test.data(data.types = c("mRNA.T", "mRNA.N", "ann"));

</code></pre>

<hr>
<h2 id='get.top.features'>get.top.features</h2><span id='topic+get.top.features'></span>

<h3>Description</h3>

<p>Prioritise top features satisfying the criteria specified by various parameters described below
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.top.features(
  DE.features = NULL, 
  cna.data.fractions = NULL, 
  mRNA.FC.up = 0, 
  mRNA.FC.down = 0, 
  mRNA.p = 0.05, 
  mRNA.top.n = NULL, 
  cna.fractions.gain = 0.2, 
  cna.fractions.loss = 0.2
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.top.features_+3A_de.features">DE.features</code></td>
<td>
<p>Matrix containing differentially expressed features with two columns: FC and P. P may contain adjusted P or raw</p>
</td></tr>
<tr><td><code id="get.top.features_+3A_cna.data.fractions">cna.data.fractions</code></td>
<td>
<p>Feature by cancer type matrix with CNA fractions</p>
</td></tr>
<tr><td><code id="get.top.features_+3A_mrna.fc.up">mRNA.FC.up</code></td>
<td>
<p>Log2 fold change threshold for selecting over-expressed features</p>
</td></tr>
<tr><td><code id="get.top.features_+3A_mrna.fc.down">mRNA.FC.down</code></td>
<td>
<p>Log2 fold change threshold for selecting under-expressed features</p>
</td></tr>
<tr><td><code id="get.top.features_+3A_mrna.p">mRNA.p</code></td>
<td>
<p>P value threshold for selecting significantly differentially expressed features. Mutually exclusive to <code>mRNA.top.n</code></p>
</td></tr>
<tr><td><code id="get.top.features_+3A_mrna.top.n">mRNA.top.n</code></td>
<td>
<p>Top n differentially expressed features satisfying each of the fold change criteria. Mutually exclusive to <code>mRNA.p</code></p>
</td></tr>
<tr><td><code id="get.top.features_+3A_cna.fractions.gain">cna.fractions.gain</code></td>
<td>
<p>Threshold for selecting copy number gain/amplifications</p>
</td></tr>
<tr><td><code id="get.top.features_+3A_cna.fractions.loss">cna.fractions.loss</code></td>
<td>
<p>Threshold for selecting copy number losses</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of top features
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# load test data
x &lt;- get.test.data(data.types = c("mRNA.T", "mRNA.N", "CNA"));

# list of features to be assessed for differential expression
feature.ids &lt;- rownames(x$mRNA.T$BLCA);

# get differentially expressed features
DE.results &lt;- find.DE.features(
  exp.data.T = x$mRNA.T, 
  exp.data.N = x$mRNA.N, 
  feature.ids = feature.ids,
  test.name = "t.test"
  );

# get top features
top.features &lt;- get.top.features(
  DE.features = cbind("FC" = DE.results[, 1], "P" = DE.results[, 2]),
  cna.data.fractions = x$CNA.fractions$BLCA, 
  mRNA.FC.up = 0.25, 
  mRNA.FC.down = 0.25, 
  mRNA.p = 0.05, 
  mRNA.top.n = NULL, 
  cna.fractions.gain = 0.2, 
  cna.fractions.loss = 0.2
  );

</code></pre>

<hr>
<h2 id='load.datasets'>load.datasets</h2><span id='topic+load.datasets'></span>

<h3>Description</h3>

<p>Function to load and systemise molecular datasets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load.datasets(
  data.dir = "./", 
  metadata = NULL, 
  data.types = c("mRNA.T", "ann")
  )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load.datasets_+3A_data.dir">data.dir</code></td>
<td>
<p>Path to base data directory or directory containing molecular profiles</p>
</td></tr>
<tr><td><code id="load.datasets_+3A_metadata">metadata</code></td>
<td>
<p>Dataset by profile metadata matrix containing file names of the molecular profiles for different datasets</p>
</td></tr>
<tr><td><code id="load.datasets_+3A_data.types">data.types</code></td>
<td>
<p>Datatypes to be read Valid datatypes are: mRNA.T, mRNA.N, CNA (includes: log2, calls and fractions), annotations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of lists containing datasets and respective molecular profiles as matrices
</p>


<h3>Author(s)</h3>

<p>Syed Haider
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# locate test data directory which comes with the package
data.dir &lt;- paste(system.file("programdata/testdata/", package = "iDOS"), "/", sep = "");

# read meta data file
metadata &lt;- read.table(
  file = paste(data.dir, "metadata.txt", sep = ""), 
  row.names = 1, 
  header = TRUE, 
  sep = "\t",
  stringsAsFactors = FALSE
  );

x &lt;- load.datasets(
  data.dir = data.dir,
  metadata = metadata,
  data.types = c("mRNA.T", "mRNA.N", "ann")
  );

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
