<!DOCTYPE html><html><head><title>Help for package IDSL.IPA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IDSL.IPA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alignedPeakPropertyTableCorrelationListCalculator'>
<p>Aligned Peak Property Table Correlation List Calculator</p></a></li>
<li><a href='#analyteRetentionTimeCorrector'>
<p>analyte retention time corrector</p></a></li>
<li><a href='#chromatogramMatrix'>
<p>chromatogram builder for m/z = 263.1678 in 003.d from cord blood sample</p></a></li>
<li><a href='#chromatographicPeakAnalysis'>
<p>Chromatography analysis</p></a></li>
<li><a href='#chromatographicPeakDetector'>
<p>peak detection</p></a></li>
<li><a href='#derivative5pointsStencil'>
<p>Numerical differentiation by five-point stencil method</p></a></li>
<li><a href='#gapFillingCore'>
<p>Gap-Filling Core Function</p></a></li>
<li><a href='#IPA_aggregate'>
<p>aggregation method for the IDSL.IPA modules</p></a></li>
<li><a href='#IPA_baselineDeveloper'>
<p>Develop a baseline for the chromatogram using local minima</p></a></li>
<li><a href='#IPA_CompoundsAnnotation'>
<p>Compound-centric peak annotation</p></a></li>
<li><a href='#IPA_GapFiller'>
<p>IPA GapFiller</p></a></li>
<li><a href='#IPA_IonPairing'>
<p>IPA Ion Pairing</p></a></li>
<li><a href='#IPA_logRecorder'>
<p>IPA_logRecorder</p></a></li>
<li><a href='#IPA_message'>
<p>IPA message</p></a></li>
<li><a href='#IPA_MSdeconvoluter'>
<p>MS deconvoluter</p></a></li>
<li><a href='#IPA_peak_alignment_folder_xlsxAnalyzer'>
<p>IPA peak alignment folder xlsxAnalyzer</p></a></li>
<li><a href='#IPA_PeakAlignment'>
<p>IPA peak alignment</p></a></li>
<li><a href='#IPA_PeakAnalyzer'>
<p>IPA Peak Analyzer</p></a></li>
<li><a href='#IPA_PeaklistAnnotation'>
<p>IPA Peaklist Annotation</p></a></li>
<li><a href='#IPA_spectraListAggregator'>
<p>spectraList filtering</p></a></li>
<li><a href='#IPA_targeted'>
<p>IPA Targeted Analysis</p></a></li>
<li><a href='#IPA_targeted_xlsxAnalyzer'>
<p>IPA Targeted xlsxAnalyzer</p></a></li>
<li><a href='#IPA_workflow'>
<p>IPA Workflow</p></a></li>
<li><a href='#IPA_xlsxAnalyzer'>
<p>IPA xlsx Analyzer</p></a></li>
<li><a href='#islocalminimum'>
<p>islocalminimum</p></a></li>
<li><a href='#islocaloptimum'>
<p>islocaloptimum</p></a></li>
<li><a href='#loadRdata'>
<p>loadRdata</p></a></li>
<li><a href='#mzClusteringRawXIC'>
<p>m/z clustering raw XIC</p></a></li>
<li><a href='#mzRTindexer'>
<p>m/z - RT Indexer</p></a></li>
<li><a href='#peak_spline'>
<p>peak spline</p></a></li>
<li><a href='#peakAlignmentCore'>
<p>Peak Alignment Core</p></a></li>
<li><a href='#peakAreaCalculator'>
<p>peak area</p></a></li>
<li><a href='#peakAsymmetryFactorCalculator'>
<p>Asymmetry factor for a chromatographic peak</p></a></li>
<li><a href='#peakDerivativeSkewnessCalculator'>
<p>Peak Derivative Skewness Calculator</p></a></li>
<li><a href='#peakFrontingTailingResolver'>
<p>Fronting and tailing peaks resolver</p></a></li>
<li><a href='#peakGaussianityCalculator'>
<p>Peak Gaussianity Calculator</p></a></li>
<li><a href='#peakPropertyTableFreqCalculator'>
<p>Peak Property Table Frequency Calculator</p></a></li>
<li><a href='#peakPropertyTableMedianCalculator'>
<p>Peak Property Table Median Calculator</p></a></li>
<li><a href='#peakPseudomomentsSymmetryCalculator'>
<p>Peak Pseudomoments Symmetry Calculator</p></a></li>
<li><a href='#peakSharpnessCalculator'>
<p>Peak Sharpness Calculator</p></a></li>
<li><a href='#peakUSPtailingFactorCalculator'>
<p>Peak USP Tailing Factor Calculator</p></a></li>
<li><a href='#peakWidthCalculator'>
<p>peak width measuement</p></a></li>
<li><a href='#peakXcolFiller'>
<p>Peak table producer</p></a></li>
<li><a href='#peakXcolFlagger'>
<p>PeakXcol Flagger</p></a></li>
<li><a href='#plot_mz_eic'>
<p>plot_mz_eic</p></a></li>
<li><a href='#plot_simple_tic'>
<p>plot_simple_tic</p></a></li>
<li><a href='#primaryXICdeconvoluter'>
<p>Primary peak analyzer</p></a></li>
<li><a href='#recursiveMZpeaklistCorrector'>
<p>recursive mass correction</p></a></li>
<li><a href='#referenceRetentionTimeDetector'>
<p>Reference retention time detector</p></a></li>
<li><a href='#segment'>
<p>segment</p></a></li>
<li><a href='#SNRbaseline'>
<p>SNR baseline</p></a></li>
<li><a href='#SNRrms'>
<p>SNR RMS</p></a></li>
<li><a href='#SNRxcms'>
<p>SNR xcms</p></a></li>
<li><a href='#targetedIonPairing'>
<p>Targeted Ion Pairing</p></a></li>
<li><a href='#XIC'>
<p>XIC</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Intrinsic Peak Analysis (IPA) for HRMS Data</td>
</tr>
<tr>
<td>Version:</td>
<td>2.9</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>IDSL.MXP, readxl</td>
</tr>
<tr>
<td>Author:</td>
<td>Sadjad Fakouri-Baygi
    <a href="https://orcid.org/0000-0002-6864-6911"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Dinesh Barupal <a href="https://orcid.org/0000-0002-9954-8628"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Dinesh Barupal &lt;dinesh.barupal@mssm.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A multi-layered untargeted pipeline for high-throughput LC/HRMS data processing to extract signals of organic small molecules. The package performs ion pairing, peak detection, peak table alignment, retention time correction, aligned peak table gap filling, peak annotation and visualization of extracted ion chromatograms (EICs) and total ion chromatograms (TICs). The 'IDSL.IPA' package was introduced in &lt;<a href="https://doi.org/10.1021%2Facs.jproteome.2c00120">doi:10.1021/acs.jproteome.2c00120</a>&gt; .</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/idslme/idsl.ipa">https://github.com/idslme/idsl.ipa</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/idslme/idsl.ipa/issues">https://github.com/idslme/idsl.ipa/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Archs:</td>
<td>i386, x64</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-30 21:21:43 UTC; sfbaygi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-31 23:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='alignedPeakPropertyTableCorrelationListCalculator'>
Aligned Peak Property Table Correlation List Calculator
</h2><span id='topic+alignedPeakPropertyTableCorrelationListCalculator'></span>

<h3>Description</h3>

<p>Aligned Peak Property Table Correlation List Calculator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignedPeakPropertyTableCorrelationListCalculator(peakPropertyTable,
RTtolerance = 0.05, minFreqDetection = 3, minRatioDetection = 0.01,
method = "pearson", minThresholdCorrelation = 0, number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_peakpropertytable">peakPropertyTable</code></td>
<td>

<p>peak property table such as &lsquo;peak_height', 'peak_area&rsquo; and  &lsquo;peak_R13C&rsquo;
</p>
</td></tr>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_rttolerance">RTtolerance</code></td>
<td>

<p>retention time tolerance (min)
</p>
</td></tr>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_minfreqdetection">minFreqDetection</code></td>
<td>

<p>minimum frequency of detection for a (m/z-RT) peak across the peak property table
</p>
</td></tr>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_minratiodetection">minRatioDetection</code></td>
<td>

<p>minimum ratio of detection for a (m/z-RT) peak across the peak property table. This value should be between (0 - 1).
</p>
</td></tr>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_method">method</code></td>
<td>

<p>a character string indicating which correlation coefficient (or covariance) is to be computed. One of &quot;pearson&quot; (default), or &quot;spearman&quot;: can be abbreviated. (from 'cor' function of the 'stats' package)
</p>
</td></tr>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_minthresholdcorrelation">minThresholdCorrelation</code></td>
<td>

<p>minimum threshold for the correlation method
</p>
</td></tr>
<tr><td><code id="alignedPeakPropertyTableCorrelationListCalculator_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of related peak IDs for each individual (m/z-RT) pair on the peak property table
</p>

<hr>
<h2 id='analyteRetentionTimeCorrector'>
analyte retention time corrector
</h2><span id='topic+analyteRetentionTimeCorrector'></span>

<h3>Description</h3>

<p>This function calculates corrected retention times for the peaklists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyteRetentionTimeCorrector(referenceMZRTpeaks, inputPathPeaklist, peaklistFileName,
massAccuracy, RTcorrectionMethod, refPeakTolerance = 1, degreePolynomial = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_referencemzrtpeaks">referenceMZRTpeaks</code></td>
<td>

<p>a matrix of reference peaks for retention time correction.
</p>
</td></tr>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_inputpathpeaklist">inputPathPeaklist</code></td>
<td>

<p>input path to peaklist
</p>
</td></tr>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_peaklistfilename">peaklistFileName</code></td>
<td>

<p>file name peaklist
</p>
</td></tr>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>mass error to detect common reference peaks.
</p>
</td></tr>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_rtcorrectionmethod">RTcorrectionMethod</code></td>
<td>

<p>c('RetentionIndex','Polynomial')
</p>
</td></tr>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_refpeaktolerance">refPeakTolerance</code></td>
<td>

<p>number of reference peaks for retention time correction using the 'RetentionIndex' method.
</p>
</td></tr>
<tr><td><code id="analyteRetentionTimeCorrector_+3A_degreepolynomial">degreePolynomial</code></td>
<td>

<p>polynomial degree for retention time correction using the 'Polynomial' method.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of corrected retention times for each peaklist.
</p>

<hr>
<h2 id='chromatogramMatrix'>
chromatogram builder for m/z = 263.1678 in 003.d from cord blood sample
</h2><span id='topic+chromatogramMatrix'></span>

<h3>Description</h3>

<p>This data illustrates a chromatogram and baseline vectors to indicate chromatogram development.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("chromatogramMatrix")</code></pre>


<h3>Format</h3>

<p>A data frame with 219 observations on the following 6 variables.
</p>

<dl>
<dt><code>scanNumber</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>retentionTime</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>smoothChromatogram</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>rawChromatogram</code></dt><dd><p>a numeric vector</p>
</dd>
<dt>&lsquo;<span class="samp">&#8288;12C/13C Isotopologue Pairs&#8288;</span>&rsquo;</dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Baseline</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(chromatogramMatrix)
</code></pre>

<hr>
<h2 id='chromatographicPeakAnalysis'>
Chromatography analysis
</h2><span id='topic+chromatographicPeakAnalysis'></span>

<h3>Description</h3>

<p>This function detects individual chromatographic peaks and measures their peak qualification metrics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chromatographicPeakAnalysis(spectraScanXIC, aggregatedSpectraList, retentionTime,
LretentionTime, massAccuracy, mzTarget, rtTarget = NULL, scanNumberStart,
scanNumberEnd, smoothingWindow, peakResolvingPower, minNIonPair, minPeakHeight,
minRatioIonPair, maxRPW, minSNRbaseline, maxR13CcumulatedIntensity,
maxPercentageMissingScans, nSpline, exportEICparameters  = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chromatographicPeakAnalysis_+3A_spectrascanxic">spectraScanXIC</code></td>
<td>

<p>a matrix consists of 5 columns. The column contents are the m/z of 12C isotopologues, intensity of 12C isotopologues, scan number (t), m/z of 13C isotopologues, and intensity of 13C isotopologues, respectively. Redundant scan numbers are not allowed for this module.
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_aggregatedspectralist">aggregatedSpectraList</code></td>
<td>

<p>aggregated spectraList and spectra matrix from the 'IPA_spectraListAggregator' module
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_retentiontime">retentionTime</code></td>
<td>

<p>a vector of retention times vs. corresponding scan numbers
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_lretentiontime">LretentionTime</code></td>
<td>

<p>length of the retention time vector
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>mass error to perform chromatography analysis
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_mztarget">mzTarget</code></td>
<td>

<p>m/z value to perform chromatography analysis
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_rttarget">rtTarget</code></td>
<td>

<p>retention time value for a targeted peak to calculate the ancillary chromatography parameters. When this parameter set at 0, the ancillary chromatography parameters are calculated for the entire detected peaks.
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_scannumberstart">scanNumberStart</code></td>
<td>

<p>the first scan number.
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_scannumberend">scanNumberEnd</code></td>
<td>

<p>the last scan number.
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_smoothingwindow">smoothingWindow</code></td>
<td>

<p>number of scans for peak smoothing
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_peakresolvingpower">peakResolvingPower</code></td>
<td>

<p>a value to represent peak resolving power
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_minnionpair">minNIonPair</code></td>
<td>

<p>minimum number of nIsoPair for an individual peak
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_minpeakheight">minPeakHeight</code></td>
<td>

<p>minimum peak height for an individual peak
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_minratioionpair">minRatioIonPair</code></td>
<td>

<p>minimum ratio of nIsoPair per number of available scans within an individual peak
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_maxrpw">maxRPW</code></td>
<td>

<p>maximum allowed value of ratio of peak width at half-height to baseline (RPW) for an individual peak
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_minsnrbaseline">minSNRbaseline</code></td>
<td>

<p>minimum S/N baseline for an individual peak
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_maxr13ccumulatedintensity">maxR13CcumulatedIntensity</code></td>
<td>

<p>maximum allowed value of average R13C for an individual peak
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_maxpercentagemissingscans">maxPercentageMissingScans</code></td>
<td>

<p>maximum allowed value of percentage missing scans on the raw chromatogram for an individual peak.
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_nspline">nSpline</code></td>
<td>

<p>number of points for further smoothing using a cubic spline smoothing method to calculate ancillary chromatographic parameters
</p>
</td></tr>
<tr><td><code id="chromatographicPeakAnalysis_+3A_exporteicparameters">exportEICparameters</code></td>
<td>

<p>When &lsquo;NULL', EICs are not plotted. 'exportEICparameters' should contain three variables of 1) an address to save IPA EICs figures, 2) &rsquo;HRMS' file name, and 3) a valid string of characters.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame consisting of 24 columns representing chromatography and mass spectrometry parameters. Each row represents an individual separated chromatographic peak.
</p>

<hr>
<h2 id='chromatographicPeakDetector'>
peak detection
</h2><span id='topic+chromatographicPeakDetector'></span>

<h3>Description</h3>

<p>This function detects separated chromatographic peaks on the chromatogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chromatographicPeakDetector(int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chromatographicPeakDetector_+3A_int">int</code></td>
<td>

<p>a vector of intensities of the chromatogram.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of 2 columns. Each row indicates peak boundary indices on the 'int' vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chromatogramMatrix)
int &lt;- chromatogramMatrix$smoothChromatogram
chromatographicPeakDetector(int)
</code></pre>

<hr>
<h2 id='derivative5pointsStencil'>
Numerical differentiation by five-point stencil method
</h2><span id='topic+derivative5pointsStencil'></span>

<h3>Description</h3>

<p>This module performs numerical differentiation using the five-point stencil method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>derivative5pointsStencil(x, y, n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="derivative5pointsStencil_+3A_x">x</code></td>
<td>

<p>a vector of values for x.
</p>
</td></tr>
<tr><td><code id="derivative5pointsStencil_+3A_y">y</code></td>
<td>

<p>a vector of values for y.
</p>
</td></tr>
<tr><td><code id="derivative5pointsStencil_+3A_n">n</code></td>
<td>

<p>order of numerical differentiation (n=1-4).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of 2 columns. The first column represents x and the second column represents numerical differentiation values. This matrix has four rows (two rows from the beginning and 2 rows from the end) less than length of x or y.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peak_spline)
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2]
n &lt;- 2 # second order derivative
derivative5pointsStencil(rt, int, n)
</code></pre>

<hr>
<h2 id='gapFillingCore'>
Gap-Filling Core Function
</h2><span id='topic+gapFillingCore'></span>

<h3>Description</h3>

<p>Gap-Filling Core Function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gapFillingCore(input_path_hrms, peakXcol, massAccuracy, RTtolerance, scanTolerance,
retentionTimeCorrectionCheck = FALSE, listCorrectedRTpeaklists = NULL,
inputPathPeaklist = NULL, ionMassDifference = 1.003354835336,
number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gapFillingCore_+3A_input_path_hrms">input_path_hrms</code></td>
<td>

<p>input_path_hrms
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_peakxcol">peakXcol</code></td>
<td>

<p>peakXcol
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>massAccuracy
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_rttolerance">RTtolerance</code></td>
<td>

<p>RTtolerance
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_scantolerance">scanTolerance</code></td>
<td>

<p>a scan tolerance to extend the chromatogram for better calculations.
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_retentiontimecorrectioncheck">retentionTimeCorrectionCheck</code></td>
<td>

<p>retentionTimeCorrectionCheck
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_listcorrectedrtpeaklists">listCorrectedRTpeaklists</code></td>
<td>

<p>listCorrectedRTpeaklists
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_inputpathpeaklist">inputPathPeaklist</code></td>
<td>

<p>inputPathPeaklist
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_ionmassdifference">ionMassDifference</code></td>
<td>

<p>ionMassDifference
</p>
</td></tr>
<tr><td><code id="gapFillingCore_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of gap-filled data
</p>

<hr>
<h2 id='IPA_aggregate'>
aggregation method for the IDSL.IPA modules
</h2><span id='topic+IPA_aggregate'></span>

<h3>Description</h3>

<p>This module is to optimize the 'indexVec' variable by removing elements that have redundant 'idVec' numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_aggregate(idVec, variableVec, indexVec, targetVar)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_aggregate_+3A_idvec">idVec</code></td>
<td>

<p>a vector of id numbers. Repeated id numbers are allowed.
</p>
</td></tr>
<tr><td><code id="IPA_aggregate_+3A_variablevec">variableVec</code></td>
<td>

<p>a vector of variable of the interest such as RT, m/z, etc.
</p>
</td></tr>
<tr><td><code id="IPA_aggregate_+3A_indexvec">indexVec</code></td>
<td>

<p>a vector of indices
</p>
</td></tr>
<tr><td><code id="IPA_aggregate_+3A_targetvar">targetVar</code></td>
<td>

<p>the targeted value in 'variableVec'
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a clean indexVec after removing repeated 'idVec'.
</p>

<hr>
<h2 id='IPA_baselineDeveloper'>
Develop a baseline for the chromatogram using local minima
</h2><span id='topic+IPA_baselineDeveloper'></span>

<h3>Description</h3>

<p>This function generates a vector of baselines for the chromatogram using local
minima. It also is capable of excluding outlier local minima to generate a
realistic baseline including true baseline regions. This baseline may represent
the local noise levels for the chromatogram.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_baselineDeveloper(segment, int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_baselineDeveloper_+3A_segment">segment</code></td>
<td>

<p>a matrix or a vecotr of adjusted scan number of local minima w/ or w/o redundant
local minima. Adjusted scan numbers are the scan numbers but adjusted to start
at 1.
</p>
</td></tr>
<tr><td><code id="IPA_baselineDeveloper_+3A_int">int</code></td>
<td>

<p>a vector of intensities of the chromatogram.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of baselines in the same size of the &quot;int&quot; vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(segment)
data(chromatogramMatrix)
int &lt;- chromatogramMatrix$smoothChromatogram
IPA_baselineDeveloper(segment, int)
</code></pre>

<hr>
<h2 id='IPA_CompoundsAnnotation'>
Compound-centric peak annotation
</h2><span id='topic+IPA_CompoundsAnnotation'></span>

<h3>Description</h3>

<p>This function performs compound-centric peak annotation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_CompoundsAnnotation(PARAM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_CompoundsAnnotation_+3A_param">PARAM</code></td>
<td>

<p>a data frame from IPA_xlsxAnalyzer function containing the IPA parameters.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function saves individual .csv files for each compound in the &quot;compound_centric_annotation&quot; folder.
</p>

<hr>
<h2 id='IPA_GapFiller'>
IPA GapFiller
</h2><span id='topic+IPA_GapFiller'></span>

<h3>Description</h3>

<p>This function fills the gaps on the peak table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_GapFiller(PARAM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_GapFiller_+3A_param">PARAM</code></td>
<td>

<p>a data frame from the 'IPA_xlsxAnalyzer' function containing the IPA parameters.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function saves individual .csv and .Rdata files for the gap-filled peak tables for peak height, area, and R13C properties in the &quot;peak_alignment&quot; folder.
</p>

<hr>
<h2 id='IPA_IonPairing'>
IPA Ion Pairing
</h2><span id='topic+IPA_IonPairing'></span>

<h3>Description</h3>

<p>This function pairs two ions with a fixed distance in high-resolution mass spectral datasets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_IonPairing(spectraList, minSpectraNoiseLevel, massAccuracyIonPair = 0.015,
ionMassDifference = 1.003354835336, number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_IonPairing_+3A_spectralist">spectraList</code></td>
<td>

<p>list of mass spectra in each chromatogram scan
</p>
</td></tr>
<tr><td><code id="IPA_IonPairing_+3A_minspectranoiselevel">minSpectraNoiseLevel</code></td>
<td>

<p>intensity threshold at each chromatogram scan
</p>
</td></tr>
<tr><td><code id="IPA_IonPairing_+3A_massaccuracyionpair">massAccuracyIonPair</code></td>
<td>

<p>mass error to detect pair ions
</p>
</td></tr>
<tr><td><code id="IPA_IonPairing_+3A_ionmassdifference">ionMassDifference</code></td>
<td>

<p>mass difference to pair ions. (Default = DeltaC = 13C - 12C = 1.003354835336), or DeltaS = 34S - 32S = 1.9957958356, or any numerical value.
</p>
</td></tr>
<tr><td><code id="IPA_IonPairing_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix consists of 5 columns. The column contents are the m/z of 12C isotopologues, intensity of 12C isotopologues, scan number (t), m/z of 13C isotopologues, and intensity of 13C isotopologues, respectively.
</p>

<hr>
<h2 id='IPA_logRecorder'>
IPA_logRecorder
</h2><span id='topic+IPA_logRecorder'></span>

<h3>Description</h3>

<p>IPA_logRecorder
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_logRecorder(messageQuote, allowedPrinting = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_logRecorder_+3A_messagequote">messageQuote</code></td>
<td>

<p>messageQuote
</p>
</td></tr>
<tr><td><code id="IPA_logRecorder_+3A_allowedprinting">allowedPrinting</code></td>
<td>

<p>allowedPrinting
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a line of communication messages is exported to the console and the log .txt file.
</p>

<hr>
<h2 id='IPA_message'>
IPA message
</h2><span id='topic+IPA_message'></span>

<h3>Description</h3>

<p>IPA_message
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_message(messageQuote, failedMessage= TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_message_+3A_messagequote">messageQuote</code></td>
<td>

<p>messageQuote
</p>
</td></tr>
<tr><td><code id="IPA_message_+3A_failedmessage">failedMessage</code></td>
<td>

<p>failedMessage
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a line of communication messages is exported to the console.
</p>

<hr>
<h2 id='IPA_MSdeconvoluter'>
MS deconvoluter
</h2><span id='topic+IPA_MSdeconvoluter'></span>

<h3>Description</h3>

<p>This function deconvolutes mass spectrometry files into a list of mass spectra and a vector of retention times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_MSdeconvoluter(inputHRMSfolderPath, MSfileName, MSlevel = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_MSdeconvoluter_+3A_inputhrmsfolderpath">inputHRMSfolderPath</code></td>
<td>

<p>address of the mass spectrometry file
</p>
</td></tr>
<tr><td><code id="IPA_MSdeconvoluter_+3A_msfilename">MSfileName</code></td>
<td>

<p>mass spectrometry file.
</p>
</td></tr>
<tr><td><code id="IPA_MSdeconvoluter_+3A_mslevel">MSlevel</code></td>
<td>

<p>MS level to extract information.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>spectraList</code></td>
<td>

<p>a list of mass spectra.
</p>
</td></tr>
<tr><td><code>retentionTime</code></td>
<td>

<p>a vector of retention times for scan numbers.
</p>
</td></tr>
<tr><td><code>MS_polarity</code></td>
<td>

<p>mass spectrometry ionization mode (+/-)
</p>
</td></tr>
</table>

<hr>
<h2 id='IPA_peak_alignment_folder_xlsxAnalyzer'>
IPA peak alignment folder xlsxAnalyzer
</h2><span id='topic+IPA_peak_alignment_folder_xlsxAnalyzer'></span>

<h3>Description</h3>

<p>IPA peak alignment folder xlsxAnalyzer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_peak_alignment_folder_xlsxAnalyzer(PARAM, PARAM_ID, checkpoint_parameter,
correctedRTcheck = FALSE, CSAcheck = FALSE, allowedVerbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_peak_alignment_folder_xlsxAnalyzer_+3A_param">PARAM</code></td>
<td>

<p>PARAM
</p>
</td></tr>
<tr><td><code id="IPA_peak_alignment_folder_xlsxAnalyzer_+3A_param_id">PARAM_ID</code></td>
<td>

<p>PARAM_ID
</p>
</td></tr>
<tr><td><code id="IPA_peak_alignment_folder_xlsxAnalyzer_+3A_checkpoint_parameter">checkpoint_parameter</code></td>
<td>

<p>checkpoint_parameter
</p>
</td></tr>
<tr><td><code id="IPA_peak_alignment_folder_xlsxAnalyzer_+3A_correctedrtcheck">correctedRTcheck</code></td>
<td>

<p>correctedRTcheck
</p>
</td></tr>
<tr><td><code id="IPA_peak_alignment_folder_xlsxAnalyzer_+3A_csacheck">CSAcheck</code></td>
<td>

<p>CSAcheck
</p>
</td></tr>
<tr><td><code id="IPA_peak_alignment_folder_xlsxAnalyzer_+3A_allowedverbose">allowedVerbose</code></td>
<td>

<p>c(TRUE, FALSE). A 'TRUE' allowedVerbose provides messages about the flow of the function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>PARAM</code></td>
<td>

<p>PARAM
</p>
</td></tr>
<tr><td><code>checkpoint_parameter</code></td>
<td>

<p>checkpoint_parameter
</p>
</td></tr>
</table>

<hr>
<h2 id='IPA_PeakAlignment'>
IPA peak alignment
</h2><span id='topic+IPA_PeakAlignment'></span>

<h3>Description</h3>

<p>This function produces an aligned peak table from individual peaklists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_PeakAlignment(PARAM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_PeakAlignment_+3A_param">PARAM</code></td>
<td>

<p>a data frame from the 'IPA_xlsxAnalyzer' function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function saves individual .csv and .Rdata files for the aligned peak tables for peak height, area, and R13C properties in the &quot;peak_alignment&quot; folder.
</p>

<hr>
<h2 id='IPA_PeakAnalyzer'>
IPA Peak Analyzer
</h2><span id='topic+IPA_PeakAnalyzer'></span>

<h3>Description</h3>

<p>This function performs the IPA peak detection module.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_PeakAnalyzer(PARAM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_PeakAnalyzer_+3A_param">PARAM</code></td>
<td>

<p>is a data frame from IPA_xlsxAnalyzer function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function saves individual peaklist files in '.csv' and '.Rdata' formats for HRMS files in the 'peaklists' folder.
</p>

<hr>
<h2 id='IPA_PeaklistAnnotation'>
IPA Peaklist Annotation
</h2><span id='topic+IPA_PeaklistAnnotation'></span>

<h3>Description</h3>

<p>This function performs sample-centric peak annotation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_PeaklistAnnotation(PARAM)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_PeaklistAnnotation_+3A_param">PARAM</code></td>
<td>

<p>a data frame from IPA_xlsxAnalyzer function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function saves individual .csv files for peak height, area, and R13C properties in the &quot;sample_centric_annotation&quot; folder.
</p>

<hr>
<h2 id='IPA_spectraListAggregator'>
spectraList filtering
</h2><span id='topic+IPA_spectraListAggregator'></span>

<h3>Description</h3>

<p>This module stacks the spectraList object and creates a list of ions for a rapid spectra query.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_spectraListAggregator(spectraList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_spectraListAggregator_+3A_spectralist">spectraList</code></td>
<td>

<p>a list of mass spectra in each chromatogram scan.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>aggregatedSpectraList</code></td>
<td>

<p>aggregated spectraList
</p>
</td></tr>
<tr><td><code>spectraListMatrix</code></td>
<td>

<p>matrix of row bounded spectraList
</p>
</td></tr>
</table>

<hr>
<h2 id='IPA_targeted'>
IPA Targeted Analysis
</h2><span id='topic+IPA_targeted'></span>

<h3>Description</h3>

<p>This function plots extracted ion chromatogram (EIC) figures in the targeted mode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_targeted(PARAM_targeted, allowedVerbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_targeted_+3A_param_targeted">PARAM_targeted</code></td>
<td>

<p>IPA parameters to feed the 'IPA_targeted' module. This variable can be produced using the 'IPA_targeted_xlsxAnalyzer' module.
</p>
</td></tr>
<tr><td><code id="IPA_targeted_+3A_allowedverbose">allowedVerbose</code></td>
<td>

<p>c(TRUE, FALSE). A 'TRUE' allowedVerbose provides messages about the flow of the function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This module saves extracted ion chromatograms (EICs) in .png format in the &quot;Targeted_EICs&quot; folder and saves a table of peak properties.
</p>

<hr>
<h2 id='IPA_targeted_xlsxAnalyzer'>
IPA Targeted xlsxAnalyzer
</h2><span id='topic+IPA_targeted_xlsxAnalyzer'></span>

<h3>Description</h3>

<p>This function processes the spreadsheet of the IPA parameters to ensure the parameter inputs are in agreement with the 'IPA_targeted' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_targeted_xlsxAnalyzer(spreadsheet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_targeted_xlsxAnalyzer_+3A_spreadsheet">spreadsheet</code></td>
<td>

<p>contains the IPA parameters.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'PARAM_targeted' which is the IPA parameters to feed the 'IPA_targeted' function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## To generate the IPA spreadsheet parameters
s_path &lt;- system.file("extdata", package = "IDSL.IPA")
SSh1 &lt;- paste0(s_path,"/IPA_parameters.xlsx")
spreadsheet &lt;- readxl::read_xlsx(SSh1, sheet = 'IPA_targeted')
PARAM_targeted = cbind(spreadsheet[, 2], spreadsheet[, 4])
temp_wd &lt;- tempdir()
temp_wd_zip &lt;- paste0(temp_wd,"/idsl_ipa_test_files.zip")
tryCatch({download.file(paste0("https://github.com/idslme/IDSL.IPA/blob/main/",
                               "IPA_educational_files/idsl_ipa_test_files.zip?raw=true"),
                        destfile = temp_wd_zip)
  unzip(temp_wd_zip, exdir = temp_wd)
  pass_download &lt;- TRUE},
  error = function(e) {pass_download &lt;- FALSE},
  warning = function(w) {pass_download &lt;- FALSE})
if (pass_download) {
  PARAM_targeted[3, 2] &lt;- temp_wd
  PARAM_targeted[7, 2] &lt;- temp_wd
  PARAM_targeted[8, 2] &lt;- "53.01853, 61.00759"
  PARAM_targeted[9, 2] &lt;- "0.951, 0.961"
  ##
  PARAM_targeted &lt;- IPA_targeted_xlsxAnalyzer(PARAM_targeted)
}

</code></pre>

<hr>
<h2 id='IPA_workflow'>
IPA Workflow
</h2><span id='topic+IPA_workflow'></span><span id='topic+IPA_Workflow'></span>

<h3>Description</h3>

<p>This function executes the IPA workflow in order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_workflow(spreadsheet)
IPA_Workflow(spreadsheet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_workflow_+3A_spreadsheet">spreadsheet</code></td>
<td>

<p>IPA spreadsheet
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function organizes the IPA file processing for a better performance using the template spreadsheet.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
s_path &lt;- system.file("extdata", package = "IDSL.IPA")
SSh1 &lt;- paste0(s_path,"/IPA_parameters.xlsx")
## To see the results, use a known folder instead of the `tempdir()` command
temp_wd &lt;- tempdir()
temp_wd_zip &lt;- paste0(temp_wd, "/idsl_ipa_test_files.zip")
spreadsheet &lt;- readxl::read_xlsx(SSh1)
PARAM = cbind(spreadsheet[, 2], spreadsheet[, 4])
tryCatch({download.file(paste0("https://github.com/idslme/IDSL.IPA/blob/main/",
                               "IPA_educational_files/idsl_ipa_test_files.zip?raw=true"),
                        destfile = temp_wd_zip)
  unzip(temp_wd_zip, exdir = temp_wd)
  pass_download &lt;- TRUE},
  error = function(e) {pass_download &lt;- FALSE},
  warning = function(w) {pass_download &lt;- FALSE})
if (pass_download) {
  PARAM[7, 2] &lt;- temp_wd
  PARAM[44, 2] &lt;- s_path
  PARAM[10, 2] &lt;- temp_wd
  IPA_workflow(PARAM)
}

</code></pre>

<hr>
<h2 id='IPA_xlsxAnalyzer'>
IPA xlsx Analyzer
</h2><span id='topic+IPA_xlsxAnalyzer'></span>

<h3>Description</h3>

<p>This function processes the spreadsheet of the IPA parameters to ensure the parameter inputs are in agreement with the IPA requirements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IPA_xlsxAnalyzer(spreadsheet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IPA_xlsxAnalyzer_+3A_spreadsheet">spreadsheet</code></td>
<td>

<p>IPA spreadsheet
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns the IPA parameters to feed the IPA_Workflow, IPA_CompoundsAnnotation, IPA_GapFiller, IPA_PeakAlignment, IPA_PeakAnalyzer, and IPA_PeaklistAnnotation functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
s_path &lt;- system.file("extdata", package = "IDSL.IPA")
SSh1 &lt;- paste0(s_path, "/IPA_parameters.xlsx")
temp_wd &lt;- tempdir()
temp_wd_zip &lt;- paste0(temp_wd,"/idsl_ipa_test_files.zip")
spreadsheet &lt;- readxl::read_xlsx(SSh1)
PARAM = cbind(spreadsheet[, 2], spreadsheet[, 4])
tryCatch({download.file(paste0("https://github.com/idslme/IDSL.IPA/blob/main/",
                     "IPA_educational_files/idsl_ipa_test_files.zip?raw=true"),
              destfile = temp_wd_zip)
unzip(temp_wd_zip, exdir = temp_wd)
pass_download &lt;- TRUE},
error = function(e) {pass_download &lt;- FALSE},
warning = function(w) {pass_download &lt;- FALSE})
if (pass_download) {
  PARAM[7, 2] &lt;- temp_wd
  PARAM[10, 2] &lt;- temp_wd # output data location
  PARAM[44, 2] &lt;- s_path # reference file location
  PARAM &lt;- IDSL.IPA::IPA_xlsxAnalyzer(PARAM)
}

</code></pre>

<hr>
<h2 id='islocalminimum'>
islocalminimum
</h2><span id='topic+islocalminimum'></span>

<h3>Description</h3>

<p>This function returns indices of local minimum points on a curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>islocalminimum(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="islocalminimum_+3A_y">y</code></td>
<td>

<p>is a vector of y values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector in the same size of the vector 'y'. Local minimum arrays represented by -1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chromatogramMatrix)
int &lt;- chromatogramMatrix$smoothChromatogram
islocalminimum(int)
</code></pre>

<hr>
<h2 id='islocaloptimum'>
islocaloptimum
</h2><span id='topic+islocaloptimum'></span>

<h3>Description</h3>

<p>This function returns indices of local minimum and maximum points on a curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>islocaloptimum(y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="islocaloptimum_+3A_y">y</code></td>
<td>

<p>is a vector of y values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector in the same size of the vector 'y'. Local minimum and maximum arrays represented by -1 and +1, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(chromatogramMatrix)
int &lt;- chromatogramMatrix$smoothChromatogram
islocaloptimum(int)
</code></pre>

<hr>
<h2 id='loadRdata'>
loadRdata
</h2><span id='topic+loadRdata'></span>

<h3>Description</h3>

<p>This function loads .Rdata files into a variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadRdata(fileName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadRdata_+3A_filename">fileName</code></td>
<td>

<p>is an '.Rdata' file.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The called variable into the new assigned variable name.
</p>

<hr>
<h2 id='mzClusteringRawXIC'>
m/z clustering raw XIC
</h2><span id='topic+mzClusteringRawXIC'></span>

<h3>Description</h3>

<p>This function clusters related 12C m/z values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mzClusteringRawXIC(spectraScan123, massAccuracy, minNIonPair, minPeakHeightXIC)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mzClusteringRawXIC_+3A_spectrascan123">spectraScan123</code></td>
<td>

<p>a matrix consists of 3 columns. The column contents are the m/z of 12C isotopologues, intensity of 12C isotopologues, and scan number (t).
</p>
</td></tr>
<tr><td><code id="mzClusteringRawXIC_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>mass accuracy to detect related 12C m/z values.
</p>
</td></tr>
<tr><td><code id="mzClusteringRawXIC_+3A_minnionpair">minNIonPair</code></td>
<td>

<p>minimum number of nIsoPair for an individual peak.
</p>
</td></tr>
<tr><td><code id="mzClusteringRawXIC_+3A_minpeakheightxic">minPeakHeightXIC</code></td>
<td>

<p>minimum peak height for an individual raw EIC
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns an list on index numbers of EICs for the &quot;spectraScan&quot; variable.
</p>

<hr>
<h2 id='mzRTindexer'>
m/z - RT Indexer
</h2><span id='topic+mzRTindexer'></span>

<h3>Description</h3>

<p>This function locate the closest pair of a reference (m/z - RT) pair in a 2-D grid of 'm/z' and 'RT' vectors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mzRTindexer(MZvec, RTvec, MZref, RTref, massAccuracy, RTtolerance)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mzRTindexer_+3A_mzvec">MZvec</code></td>
<td>

<p>m/z vector
</p>
</td></tr>
<tr><td><code id="mzRTindexer_+3A_rtvec">RTvec</code></td>
<td>

<p>RT vector
</p>
</td></tr>
<tr><td><code id="mzRTindexer_+3A_mzref">MZref</code></td>
<td>

<p>a reference m/z
</p>
</td></tr>
<tr><td><code id="mzRTindexer_+3A_rtref">RTref</code></td>
<td>

<p>a reference RT
</p>
</td></tr>
<tr><td><code id="mzRTindexer_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>m/z tolerance
</p>
</td></tr>
<tr><td><code id="mzRTindexer_+3A_rttolerance">RTtolerance</code></td>
<td>

<p>RT tolerance
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>index of closest pair to the reference (m/z - RT) pair
</p>


<h3>Note</h3>

<p>This function returns NULL in case no match is detected.
</p>

<hr>
<h2 id='peak_spline'>
peak spline
</h2><span id='topic+peak_spline'></span>

<h3>Description</h3>

<p>illusterates a smoothe peak using cubic spline smoothing method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("peak_spline")</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on the following 3 variables.
</p>

<dl>
<dt><code>rt_spline</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>int_spline</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>bl_approx</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(peak_spline)
</code></pre>

<hr>
<h2 id='peakAlignmentCore'>
Peak Alignment Core
</h2><span id='topic+peakAlignmentCore'></span>

<h3>Description</h3>

<p>This function aligns peaks from multiple peaklists and produces an aligned table of common peaks among multiple samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakAlignmentCore(peaklistInputFolderPath, peaklistFileNames, listCorrectedRTpeaklists,
massAccuracy, RTtolerance, number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakAlignmentCore_+3A_peaklistinputfolderpath">peaklistInputFolderPath</code></td>
<td>

<p>path to directory of peaklists.
</p>
</td></tr>
<tr><td><code id="peakAlignmentCore_+3A_peaklistfilenames">peaklistFileNames</code></td>
<td>

<p>name of peaklists for peak table production.
</p>
</td></tr>
<tr><td><code id="peakAlignmentCore_+3A_listcorrectedrtpeaklists">listCorrectedRTpeaklists</code></td>
<td>

<p>a list of corrected or uncorrected retention times for each peaklist.
</p>
</td></tr>
<tr><td><code id="peakAlignmentCore_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>mass error to detect common peaks.
</p>
</td></tr>
<tr><td><code id="peakAlignmentCore_+3A_rttolerance">RTtolerance</code></td>
<td>

<p>retention time tolerance to detect common peaks.
</p>
</td></tr>
<tr><td><code id="peakAlignmentCore_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns an aligned peak table with index numbers from individual peaklists for each peak.
</p>

<hr>
<h2 id='peakAreaCalculator'>
peak area
</h2><span id='topic+peakAreaCalculator'></span>

<h3>Description</h3>

<p>This function calculates area under the curve using a trapezoid method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakAreaCalculator(x, y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakAreaCalculator_+3A_x">x</code></td>
<td>

<p>is a vector of x values.
</p>
</td></tr>
<tr><td><code id="peakAreaCalculator_+3A_y">y</code></td>
<td>

<p>is a vector of y values.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A number for the integrated peak area.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2]
peakAreaCalculator(rt, int)
</code></pre>

<hr>
<h2 id='peakAsymmetryFactorCalculator'>
Asymmetry factor for a chromatographic peak
</h2><span id='topic+peakAsymmetryFactorCalculator'></span>

<h3>Description</h3>

<p>This function calculates an asymmetry factor for a chromatographic peak.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakAsymmetryFactorCalculator(rt, int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakAsymmetryFactorCalculator_+3A_rt">rt</code></td>
<td>

<p>a vector of retention times for the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakAsymmetryFactorCalculator_+3A_int">int</code></td>
<td>

<p>a vector of intensities corresponding to the vector of retention times for the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>asymmetry of the chromatographic peak. 1 is for very symmetric peak.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peak_spline)
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2] - peak_spline[, 3]
peakAsymmetryFactorCalculator(rt, int)
</code></pre>

<hr>
<h2 id='peakDerivativeSkewnessCalculator'>
Peak Derivative Skewness Calculator 
</h2><span id='topic+peakDerivativeSkewnessCalculator'></span>

<h3>Description</h3>

<p>This function calculates skewness of a chromatographic peak using first order degree of numerical differentiation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakDerivativeSkewnessCalculator(rt, int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakDerivativeSkewnessCalculator_+3A_rt">rt</code></td>
<td>

<p>a vector representing retention times of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakDerivativeSkewnessCalculator_+3A_int">int</code></td>
<td>

<p>a vector representing intensities of the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Skewness of a chromatographic peak. 1 is for very symmetric peak. Minimum is 0 from this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peak_spline)
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2]
peakDerivativeSkewnessCalculator(rt, int)
</code></pre>

<hr>
<h2 id='peakFrontingTailingResolver'>
Fronting and tailing peaks resolver
</h2><span id='topic+peakFrontingTailingResolver'></span>

<h3>Description</h3>

<p>This function attempts to resolve peak tailings or frontings into the main peak in case they were detected as seperate peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakFrontingTailingResolver(segment, int, maxScanDifference, peakResolvingPower = 0.025)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakFrontingTailingResolver_+3A_segment">segment</code></td>
<td>

<p>a matrix or a vector of peak boundaries.
</p>
</td></tr>
<tr><td><code id="peakFrontingTailingResolver_+3A_int">int</code></td>
<td>

<p>a vector of intensities of the entire chromatogram.
</p>
</td></tr>
<tr><td><code id="peakFrontingTailingResolver_+3A_maxscandifference">maxScanDifference</code></td>
<td>

<p>maximum scan number difference between peak tailing or fronting and the main peak.
</p>
</td></tr>
<tr><td><code id="peakFrontingTailingResolver_+3A_peakresolvingpower">peakResolvingPower</code></td>
<td>

<p>power of peak resolving tool.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of 2 columns. Each row indicates peak boundary indices on the 'int' vector after resolving fronting and tailing peaks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(segment)
data(chromatogramMatrix)
int &lt;- chromatogramMatrix$smoothChromatogram
maxScanDifference &lt;- 7
peakResolvingPower &lt;- 0.2
peakFrontingTailingResolver(segment, int, maxScanDifference, peakResolvingPower)
</code></pre>

<hr>
<h2 id='peakGaussianityCalculator'>
Peak Gaussianity Calculator
</h2><span id='topic+peakGaussianityCalculator'></span>

<h3>Description</h3>

<p>This module measures gaussianity of chromatographic peak using Pearson correlation coefficients (<code class="reqn">\rho</code>) at top 80 percent of peak.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakGaussianityCalculator(RT, Int, BL, gauge = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakGaussianityCalculator_+3A_rt">RT</code></td>
<td>

<p>a vector of retention times of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakGaussianityCalculator_+3A_int">Int</code></td>
<td>

<p>a vector of intensities of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakGaussianityCalculator_+3A_bl">BL</code></td>
<td>

<p>a vector of baseline of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakGaussianityCalculator_+3A_gauge">gauge</code></td>
<td>

<p>represents the gauge height of peak for Gaussianity measurement.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Gaussianity of the chromatographic peak.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
RT &lt;- peak_spline[, 1]
Int &lt;- peak_spline[, 2]
BL &lt;- peak_spline[, 3]
peakGaussianityCalculator(RT, Int, BL, gauge = 0.8)
</code></pre>

<hr>
<h2 id='peakPropertyTableFreqCalculator'>
Peak Property Table Frequency Calculator
</h2><span id='topic+peakPropertyTableFreqCalculator'></span>

<h3>Description</h3>

<p>Peak Property Table Frequency Calculator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakPropertyTableFreqCalculator(peakPropertyTable, startColumnIndex = 3,
number_processing_threads = 1, allowedVerbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakPropertyTableFreqCalculator_+3A_peakpropertytable">peakPropertyTable</code></td>
<td>

<p>peakPropertyTable
</p>
</td></tr>
<tr><td><code id="peakPropertyTableFreqCalculator_+3A_startcolumnindex">startColumnIndex</code></td>
<td>

<p>startColumnIndex
</p>
</td></tr>
<tr><td><code id="peakPropertyTableFreqCalculator_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number_processing_threads
</p>
</td></tr>
<tr><td><code id="peakPropertyTableFreqCalculator_+3A_allowedverbose">allowedVerbose</code></td>
<td>

<p>c(TRUE, FALSE). A 'TRUE' allowedVerbose provides messages about the flow of the function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of frequency of detection.
</p>

<hr>
<h2 id='peakPropertyTableMedianCalculator'>
Peak Property Table Median Calculator
</h2><span id='topic+peakPropertyTableMedianCalculator'></span>

<h3>Description</h3>

<p>Peak Property Table Median Calculator
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakPropertyTableMedianCalculator(peakPropertyTable, falggingVector = NULL,
number_processing_threads = 1, allowedVerbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakPropertyTableMedianCalculator_+3A_peakpropertytable">peakPropertyTable</code></td>
<td>

<p>peakPropertyTable
</p>
</td></tr>
<tr><td><code id="peakPropertyTableMedianCalculator_+3A_falggingvector">falggingVector</code></td>
<td>

<p>falggingVector
</p>
</td></tr>
<tr><td><code id="peakPropertyTableMedianCalculator_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number_processing_threads
</p>
</td></tr>
<tr><td><code id="peakPropertyTableMedianCalculator_+3A_allowedverbose">allowedVerbose</code></td>
<td>

<p>c(TRUE, FALSE). A 'TRUE' allowedVerbose provides messages about the flow of the function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>updated peak property table
</p>

<hr>
<h2 id='peakPseudomomentsSymmetryCalculator'>
Peak Pseudomoments Symmetry Calculator
</h2><span id='topic+peakPseudomomentsSymmetryCalculator'></span>

<h3>Description</h3>

<p>This function measures peak symmetry and skewness using the inflection points
of the peak on both sides.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakPseudomomentsSymmetryCalculator(rt, int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakPseudomomentsSymmetryCalculator_+3A_rt">rt</code></td>
<td>

<p>a vector of retention times for the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakPseudomomentsSymmetryCalculator_+3A_int">int</code></td>
<td>

<p>a vector of intensities corresponding to the vector of retention times for
the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>PeakSymmetry</code></td>
<td>

<p>peak symmetry for the chromatographic peak.
</p>
</td></tr>
<tr><td><code>Skewness</code></td>
<td>

<p>skewness for the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2] - peak_spline[, 3]
peakPseudomomentsSymmetryCalculator(rt, int)
</code></pre>

<hr>
<h2 id='peakSharpnessCalculator'>
Peak Sharpness Calculator
</h2><span id='topic+peakSharpnessCalculator'></span>

<h3>Description</h3>

<p>This function measures sharpness of a chromatographic peak
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakSharpnessCalculator(int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakSharpnessCalculator_+3A_int">int</code></td>
<td>

<p>a vector of intensities of the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A number representing peak sharpness. The higher values indicate higher
sharpness.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
int &lt;- peak_spline[, 2]
peakSharpnessCalculator(int)
</code></pre>

<hr>
<h2 id='peakUSPtailingFactorCalculator'>
Peak USP Tailing Factor Calculator
</h2><span id='topic+peakUSPtailingFactorCalculator'></span>

<h3>Description</h3>

<p>This function calculates USP tailing factor at above 10 percent of the height.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakUSPtailingFactorCalculator(rt, int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakUSPtailingFactorCalculator_+3A_rt">rt</code></td>
<td>

<p>a vector of retention times for the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakUSPtailingFactorCalculator_+3A_int">int</code></td>
<td>

<p>a vector of intensities corresponding to the vector of retention times for the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>USP tailing factor for the chromatographic peak.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peak_spline)
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2] - peak_spline[, 3]
peakUSPtailingFactorCalculator(rt, int)
</code></pre>

<hr>
<h2 id='peakWidthCalculator'>
peak width measuement
</h2><span id='topic+peakWidthCalculator'></span>

<h3>Description</h3>

<p>This function measures peak width at different peak heights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakWidthCalculator(rt, int, gauge)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakWidthCalculator_+3A_rt">rt</code></td>
<td>

<p>a vector of retention times of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakWidthCalculator_+3A_int">int</code></td>
<td>

<p>a vector of intensities of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="peakWidthCalculator_+3A_gauge">gauge</code></td>
<td>

<p>a height gauge to measure the peak width. This parameter should be between 0-1.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A peak width at the guaged height.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
rt &lt;- peak_spline[, 1]
int &lt;- peak_spline[, 2] - peak_spline[, 3]
gauge &lt;- 0.5
peakWidthCalculator(rt, int, gauge)
</code></pre>

<hr>
<h2 id='peakXcolFiller'>
Peak table producer
</h2><span id='topic+peakXcolFiller'></span>

<h3>Description</h3>

<p>This function fills the peak table from individual peaklists.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakXcolFiller(peakXcol, inputPathPeaklist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakXcolFiller_+3A_peakxcol">peakXcol</code></td>
<td>

<p>a matrix of index numbers in individual peaklists for each peak (m/z-RT).
</p>
</td></tr>
<tr><td><code id="peakXcolFiller_+3A_inputpathpeaklist">inputPathPeaklist</code></td>
<td>

<p>address of the peaklists.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>peak_height</code></td>
<td>

<p>peak table for height values
</p>
</td></tr>
<tr><td><code>peak_area</code></td>
<td>

<p>peak table for area values
</p>
</td></tr>
<tr><td><code>peak_R13C</code></td>
<td>

<p>peak table for R13C values
</p>
</td></tr>
</table>

<hr>
<h2 id='peakXcolFlagger'>
PeakXcol Flagger
</h2><span id='topic+peakXcolFlagger'></span>

<h3>Description</h3>

<p>PeakXcol Flagger
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakXcolFlagger(mzPeakXcol, rtPeakXcol, freqPeakXcol, massAccuracy, RTtolerance,
maxRedundantPeakFlagging)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakXcolFlagger_+3A_mzpeakxcol">mzPeakXcol</code></td>
<td>

<p>mzPeakXcol
</p>
</td></tr>
<tr><td><code id="peakXcolFlagger_+3A_rtpeakxcol">rtPeakXcol</code></td>
<td>

<p>rtPeakXcol
</p>
</td></tr>
<tr><td><code id="peakXcolFlagger_+3A_freqpeakxcol">freqPeakXcol</code></td>
<td>

<p>freqPeakXcol
</p>
</td></tr>
<tr><td><code id="peakXcolFlagger_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>massAccuracy
</p>
</td></tr>
<tr><td><code id="peakXcolFlagger_+3A_rttolerance">RTtolerance</code></td>
<td>

<p>RTtolerance
</p>
</td></tr>
<tr><td><code id="peakXcolFlagger_+3A_maxredundantpeakflagging">maxRedundantPeakFlagging</code></td>
<td>

<p>maxRedundantPeakFlagging
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector with flagged numbers
</p>

<hr>
<h2 id='plot_mz_eic'>
plot_mz_eic
</h2><span id='topic+plot_mz_eic'></span>

<h3>Description</h3>

<p>plot_mz_eic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_mz_eic(filelist, filelocation, mzTarget, massAccuracy,
number_processing_threads = 1, rtstart = 0, rtend = 0, plotTitle = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_mz_eic_+3A_filelist">filelist</code></td>
<td>

<p>filelist
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_filelocation">filelocation</code></td>
<td>

<p>filelocation
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_mztarget">mzTarget</code></td>
<td>

<p>mzTarget
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>massAccuracy
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_rtstart">rtstart</code></td>
<td>

<p>rtstart
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_rtend">rtend</code></td>
<td>

<p>rtend
</p>
</td></tr>
<tr><td><code id="plot_mz_eic_+3A_plottitle">plotTitle</code></td>
<td>

<p>plotTitle
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot_mz_eic
</p>

<hr>
<h2 id='plot_simple_tic'>
plot_simple_tic
</h2><span id='topic+plot_simple_tic'></span>

<h3>Description</h3>

<p>plot_simple_tic
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_simple_tic(filelist, filelocation, number_processing_threads = 1,
plotTitle = "Total Ion Chromatogram")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_simple_tic_+3A_filelist">filelist</code></td>
<td>

<p>filelist
</p>
</td></tr>
<tr><td><code id="plot_simple_tic_+3A_filelocation">filelocation</code></td>
<td>

<p>filelocation
</p>
</td></tr>
<tr><td><code id="plot_simple_tic_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
<tr><td><code id="plot_simple_tic_+3A_plottitle">plotTitle</code></td>
<td>

<p>plotTitle
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot_simple_tic
</p>

<hr>
<h2 id='primaryXICdeconvoluter'>
Primary peak analyzer
</h2><span id='topic+primaryXICdeconvoluter'></span>

<h3>Description</h3>

<p>This function performs the first round of the chromatography analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>primaryXICdeconvoluter(spectraScan, scanTolerance, indexXIC, aggregatedSpectraList,
retentionTime, massAccuracy, smoothingWindow, peakResolvingPower, minNIonPair,
minPeakHeight, minRatioIonPair, maxRPW, minSNRbaseline, maxR13CcumulatedIntensity,
maxPercentageMissingScans, nSpline, exportEICparameters = NULL,
number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="primaryXICdeconvoluter_+3A_spectrascan">spectraScan</code></td>
<td>

<p>a matrix consists of 5 columns. The column contents are the m/z of 12C isotopologues, intensity of 12C isotopologues, scan number (t), m/z of 13C isotopologues, and intensity of 13C isotopologues.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_scantolerance">scanTolerance</code></td>
<td>

<p>a scan tolerance to extend the chromatogram for better calculations.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_indexxic">indexXIC</code></td>
<td>

<p>a list of indices of candidate 12C m/z values from spectraScan matrix.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_aggregatedspectralist">aggregatedSpectraList</code></td>
<td>

<p>aggregated spectraList and spectra matrix from the 'IPA_spectraListAggregator' module
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_retentiontime">retentionTime</code></td>
<td>

<p>a vector of retention times vs. corresponding scan numbers.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>a m/z value to perform chromatography analysis.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_smoothingwindow">smoothingWindow</code></td>
<td>

<p>number of scans for peak smoothing.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_peakresolvingpower">peakResolvingPower</code></td>
<td>

<p>a value to represent peak resolving power.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_minnionpair">minNIonPair</code></td>
<td>

<p>minimum number of nIsoPair for an individual peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_minpeakheight">minPeakHeight</code></td>
<td>

<p>minimum peak height for an individual peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_minratioionpair">minRatioIonPair</code></td>
<td>

<p>minimum ratio of nIsoPair per number of available scans within an individual
peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_maxrpw">maxRPW</code></td>
<td>

<p>maximum allowed value of ratio of peak width at half-height to baseline (RPW) for an individual peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_minsnrbaseline">minSNRbaseline</code></td>
<td>

<p>minimum S/N baseline for an individual peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_maxr13ccumulatedintensity">maxR13CcumulatedIntensity</code></td>
<td>

<p>maximum allowed value of average R13C for an individual peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_maxpercentagemissingscans">maxPercentageMissingScans</code></td>
<td>

<p>maximum allowed value of percentage missing scans on the raw chromatogram for an individual peak.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_nspline">nSpline</code></td>
<td>

<p>number of points for further smoothing using a cubic spline smoothing method.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_exporteicparameters">exportEICparameters</code></td>
<td>

<p>When &lsquo;NULL', EICs are not plotted. 'exportEICparameters' should contain three variables of 1) an address to save IPA EICs figures, 2) &rsquo;HRMS' file name, and 3) a valid string of characters.
</p>
</td></tr>
<tr><td><code id="primaryXICdeconvoluter_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame consisting of 24 columns representing chromatography and mass spectrometry parameters. Each row represents an individual separated chromatographic peak.
</p>

<hr>
<h2 id='recursiveMZpeaklistCorrector'>
recursive mass correction
</h2><span id='topic+recursiveMZpeaklistCorrector'></span>

<h3>Description</h3>

<p>This function performs recursive mass correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recursiveMZpeaklistCorrector(peaklist, spectraScan, scanTolerance,
aggregatedSpectraList, retentionTime, massAccuracy, smoothingWindow,
peakResolvingPower, minNIonPair, minPeakHeight, minRatioIonPair, maxRPW,
minSNRbaseline, maxR13CcumulatedIntensity, maxPercentageMissingScans, nSpline,
exportEICparameters = NULL, number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_peaklist">peaklist</code></td>
<td>

<p>an IPA peaklist from 'primaryXICdeconvoluter' function.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_spectrascan">spectraScan</code></td>
<td>

<p>a matrix consists of 5 columns. The column contents are the m/z of 12C isotopologues, intensity of 12C isotopologues, scan number (t), m/z of 13C isotopologues, and intensity of 13C isotopologues.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_scantolerance">scanTolerance</code></td>
<td>

<p>a scan tolerance to extend the chromatogram for better calculations.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_aggregatedspectralist">aggregatedSpectraList</code></td>
<td>

<p>aggregated spectraList and spectra matrix from the 'IPA_spectraListAggregator' module
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_retentiontime">retentionTime</code></td>
<td>

<p>a vector of retention times for corresponding scan numbers.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>an m/z value to perform chromatography analysis.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_smoothingwindow">smoothingWindow</code></td>
<td>

<p>a number of scans for peak smoothing.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_peakresolvingpower">peakResolvingPower</code></td>
<td>

<p>a value to represent peak resolving power.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_minnionpair">minNIonPair</code></td>
<td>

<p>minimum number of nIsoPair for an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_minpeakheight">minPeakHeight</code></td>
<td>

<p>minimum peak height for an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_minratioionpair">minRatioIonPair</code></td>
<td>

<p>minimum ratio of nIsoPair per number of available scans within an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_maxrpw">maxRPW</code></td>
<td>

<p>maximum allowed value of ratio of peak width at half-height to baseline (RPW) for an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_minsnrbaseline">minSNRbaseline</code></td>
<td>

<p>minimum S/N baseline for an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_maxr13ccumulatedintensity">maxR13CcumulatedIntensity</code></td>
<td>

<p>maximum allowed value of average R13C for an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_maxpercentagemissingscans">maxPercentageMissingScans</code></td>
<td>

<p>maximum allowed value of percentage missing scans on the raw chromatogram for an individual peak.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_nspline">nSpline</code></td>
<td>

<p>number of points for further smoothing using a cubic spline smoothing method to calculate ancillary chromatographic parameters.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_exporteicparameters">exportEICparameters</code></td>
<td>

<p>When &lsquo;NULL', EICs are not plotted. 'exportEICparameters' should contain three variables of 1) an address to save IPA EICs figures, 2) &rsquo;HRMS' file name, and 3) a valid string of characters.
</p>
</td></tr>
<tr><td><code id="recursiveMZpeaklistCorrector_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe consisting of 24 columns representing chromatography and mass spectrometry parameters. Each row represents an individual separated chromatographic peak.
</p>

<hr>
<h2 id='referenceRetentionTimeDetector'>
Reference retention time detector
</h2><span id='topic+referenceRetentionTimeDetector'></span>

<h3>Description</h3>

<p>This module detects recurring reference peaks (m/z-RT) for retention time correction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>referenceRetentionTimeDetector(inputPathPeaklist, refPeaklistFileNames,
minFrequencyRefPeaks, massAccuracy, RTtolerance, number_processing_threads = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="referenceRetentionTimeDetector_+3A_inputpathpeaklist">inputPathPeaklist</code></td>
<td>

<p>path to directory of peaklists.
</p>
</td></tr>
<tr><td><code id="referenceRetentionTimeDetector_+3A_refpeaklistfilenames">refPeaklistFileNames</code></td>
<td>

<p>name of peaklists files to detect recurring reference peaks (m/z-RT).
</p>
</td></tr>
<tr><td><code id="referenceRetentionTimeDetector_+3A_minfrequencyrefpeaks">minFrequencyRefPeaks</code></td>
<td>

<p>minimum frequency of the recurring reference peaks (m/z-RT) in the reference
files.
</p>
</td></tr>
<tr><td><code id="referenceRetentionTimeDetector_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>mass error to detect common peaks.
</p>
</td></tr>
<tr><td><code id="referenceRetentionTimeDetector_+3A_rttolerance">RTtolerance</code></td>
<td>

<p>retention time tolerance to detect common peaks.
</p>
</td></tr>
<tr><td><code id="referenceRetentionTimeDetector_+3A_number_processing_threads">number_processing_threads</code></td>
<td>

<p>number of processing threads
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>referenceMZRTpeaks</code></td>
<td>

<p>a matrix of two columns of m/z and RT of common peaks in the reference samples.
</p>
</td></tr>
<tr><td><code>listRefRT</code></td>
<td>

<p>a list of corrected or uncorrected retention times for each peaklist.
</p>
</td></tr>
</table>

<hr>
<h2 id='segment'>
segment
</h2><span id='topic+segment'></span>

<h3>Description</h3>

<p>This data illustrates an output matrix of chromatogram peak detection module from the &quot;chromatogramMatrix.rda&quot; object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("segment")</code></pre>


<h3>Format</h3>

<p>The format is:
num [1:16, 1:2] 7 15 23 33 38 46 67 86 102 118 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(segment)
</code></pre>

<hr>
<h2 id='SNRbaseline'>
SNR baseline
</h2><span id='topic+SNRbaseline'></span>

<h3>Description</h3>

<p>This function calculates S/N using local noise levels from baseline,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNRbaseline(int, baseline)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNRbaseline_+3A_int">int</code></td>
<td>

<p>a vector of intensities corresponding to the vector of retention times for the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="SNRbaseline_+3A_baseline">baseline</code></td>
<td>

<p>a vector of baseline of the chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S/N value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
int &lt;- peak_spline[, 2]
baseline &lt;- peak_spline[, 3]
SNRbaseline(int, baseline)
</code></pre>

<hr>
<h2 id='SNRrms'>
SNR RMS
</h2><span id='topic+SNRrms'></span>

<h3>Description</h3>

<p>This function calculates signal-to-noise ratio using root mean square.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNRrms(int, baseline, gauge = 0.80)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNRrms_+3A_int">int</code></td>
<td>

<p>is the vector of intensities corresponding to the vector of retention times for the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="SNRrms_+3A_baseline">baseline</code></td>
<td>

<p>is a vector of baseline of the chromatographic peak.
</p>
</td></tr>
<tr><td><code id="SNRrms_+3A_gauge">gauge</code></td>
<td>

<p>represents the gauge height of peak for gaussianity measurement.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S/N value
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("peak_spline")
int &lt;- peak_spline[, 2]
baseline &lt;- peak_spline[, 3]
SNRrms(int, baseline)
</code></pre>

<hr>
<h2 id='SNRxcms'>
SNR xcms
</h2><span id='topic+SNRxcms'></span>

<h3>Description</h3>

<p>This function calculates S/N values using a method suggested in the xcms paper (Tautenhahn, 2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNRxcms(int)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNRxcms_+3A_int">int</code></td>
<td>

<p>a vector of intensities corresponding to the vector of retention times for the
chromatographic peak.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>S/N value
</p>


<h3>References</h3>

<p>Tautenhahn, R., Bttcher, C. and Neumann, S. (2008). Highly sensitive feature detection for high resolution LC/MS. <em>BMC bioinformatics</em>, 9(1), 1-16, <a href="https://doi.org/10.1186/1471-2105-9-504">doi:10.1186/1471-2105-9-504</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(peak_spline)
int &lt;- peak_spline[, 2]
SNRxcms(int)
</code></pre>

<hr>
<h2 id='targetedIonPairing'>
Targeted Ion Pairing
</h2><span id='topic+targetedIonPairing'></span>

<h3>Description</h3>

<p>This module only pairs 'mzTarget' values across 'scanNumberStart' through 'scanNumberEnd' scan numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>targetedIonPairing(spectraList, scanNumberStart, scanNumberEnd, mzTarget,
massAccuracy, ionMassDifference = 1.003354835336, massAccuracyIonPair = massAccuracy*1.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="targetedIonPairing_+3A_spectralist">spectraList</code></td>
<td>

<p>spectraList which is a list of mass spectra
</p>
</td></tr>
<tr><td><code id="targetedIonPairing_+3A_scannumberstart">scanNumberStart</code></td>
<td>

<p>the first scan number.
</p>
</td></tr>
<tr><td><code id="targetedIonPairing_+3A_scannumberend">scanNumberEnd</code></td>
<td>

<p>the last scan number.
</p>
</td></tr>
<tr><td><code id="targetedIonPairing_+3A_mztarget">mzTarget</code></td>
<td>

<p>m/z value to perform chromatography analysis
</p>
</td></tr>
<tr><td><code id="targetedIonPairing_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>mass accuracy to select the dominant ion
</p>
</td></tr>
<tr><td><code id="targetedIonPairing_+3A_ionmassdifference">ionMassDifference</code></td>
<td>

<p>mass difference to pair ions. (Default = DeltaC = 13C - 12C = 1.003354835336), or DeltaS = 34S - 32S = 1.9957958356, or any numerical value.
</p>
</td></tr>
<tr><td><code id="targetedIonPairing_+3A_massaccuracyionpair">massAccuracyIonPair</code></td>
<td>

<p>mass accuracy to select the second ion
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A targeted ion paired spectra and their scan numbers
</p>

<hr>
<h2 id='XIC'>
XIC
</h2><span id='topic+XIC'></span>

<h3>Description</h3>

<p>XIC
</p>


<h3>Usage</h3>

<pre><code class='language-R'>XIC(aggregatedSpectraList, scanNumberStart, scanNumberEnd, mzTarget, massAccuracy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="XIC_+3A_aggregatedspectralist">aggregatedSpectraList</code></td>
<td>

<p>aggregated spectraList and spectra matrix from the 'IPA_spectraListAggregator' module
</p>
</td></tr>
<tr><td><code id="XIC_+3A_scannumberstart">scanNumberStart</code></td>
<td>

<p>the first scan number.
</p>
</td></tr>
<tr><td><code id="XIC_+3A_scannumberend">scanNumberEnd</code></td>
<td>

<p>the last scan number.
</p>
</td></tr>
<tr><td><code id="XIC_+3A_mztarget">mzTarget</code></td>
<td>

<p>an m/z value to perform XIC analysis.
</p>
</td></tr>
<tr><td><code id="XIC_+3A_massaccuracy">massAccuracy</code></td>
<td>

<p>a mass error to perform XIC analysis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix of three columns representing scan number, m/z, and intensity.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
