<!DOCTYPE html><html><head><title>Help for package ieugwasr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ieugwasr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.data'><p>rlang .data</p></a></li>
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#afl2_chrpos'><p>Look up allele frequencies and LD scores for 1000 genomes populations by chrpos</p></a></li>
<li><a href='#afl2_list'><p>Retrieve a allele frequency and LD scores for pre-defined lists of variants</p></a></li>
<li><a href='#afl2_rsid'><p>Look up allele frequencies and LD scores for 1000 genomes populations by rsid</p></a></li>
<li><a href='#api_query'><p>Wrapper for sending queries and payloads to API</p></a></li>
<li><a href='#api_status'><p>MR-Base server status</p></a></li>
<li><a href='#associations'><p>Query specific variants from specific GWAS</p></a></li>
<li><a href='#batch_from_id'><p>Extract batch name from study ID</p></a></li>
<li><a href='#batches'><p>Get list of data batches in IEU GWAS database</p></a></li>
<li><a href='#check_access_token'><p>Check if authentication has been made</p></a></li>
<li><a href='#editcheck'><p>Check datasets that are in process of being uploaded</p></a></li>
<li><a href='#fill_n'><p>Look up sample sizes when meta data is missing from associations</p></a></li>
<li><a href='#get_access_token'><p>Get access token for OAuth2 access to MR Base</p></a></li>
<li><a href='#get_query_content'><p>Parse out json response from httr object</p></a></li>
<li><a href='#gwasinfo'><p>Get list of studies with available GWAS summary statistics through API</p></a></li>
<li><a href='#infer_ancestry'><p>Infer ancestry of GWAS dataset by matching against 1000 genomes allele frequencies</p></a></li>
<li><a href='#ld_clump'><p>Perform LD clumping on SNP data</p></a></li>
<li><a href='#ld_clump_api'><p>Perform clumping on the chosen variants using through API</p></a></li>
<li><a href='#ld_clump_local'><p>Wrapper for clump function using local plink binary and ld reference dataset</p></a></li>
<li><a href='#ld_matrix'><p>Get LD matrix for list of SNPs</p></a></li>
<li><a href='#ld_matrix_local'><p>Get LD matrix using local plink binary and reference dataset</p></a></li>
<li><a href='#ld_reflookup'><p>Check which rsids are present in a remote LD reference panel</p></a></li>
<li><a href='#legacy_ids'><p>Convert current IDs to legacy IDs</p></a></li>
<li><a href='#logging_info'><p>Details of how access token logs are used</p></a></li>
<li><a href='#phewas'><p>Perform fast phewas of a specific variants against all available GWAS datasets</p></a></li>
<li><a href='#print.ApiStatus'><p>Print API status</p></a></li>
<li><a href='#print.GwasInfo'><p>Print GWAS information</p></a></li>
<li><a href='#revoke_access_token'><p>Revoke access token for MR Base</p></a></li>
<li><a href='#select_api'><p>Toggle API address between development and release</p></a></li>
<li><a href='#tophits'><p>Obtain top hits from a GWAS dataset</p></a></li>
<li><a href='#variants_chrpos'><p>Obtain information about chr pos and surrounding region</p></a></li>
<li><a href='#variants_gene'><p>Obtain variants around a gene</p></a></li>
<li><a href='#variants_rsid'><p>Obtain information about rsid</p></a></li>
<li><a href='#variants_to_rsid'><p>Convert mixed array of rsid and chrpos to list of rsid</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Interface to the 'OpenGWAS' Database API</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Interface to the 'OpenGWAS' database API <a href="https://gwas-api.mrcieu.ac.uk/">https://gwas-api.mrcieu.ac.uk/</a>. Includes a wrapper
    to make generic calls to the API, plus convenience functions for
    specific queries.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/MRCIEU/ieugwasr">https://github.com/MRCIEU/ieugwasr</a>,
<a href="https://mrcieu.github.io/ieugwasr/">https://mrcieu.github.io/ieugwasr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/MRCIEU/ieugwasr/issues">https://github.com/MRCIEU/ieugwasr/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, googleAuthR, httr, jsonlite, magrittr, rlang, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, microbenchmark, pander, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-27 19:09:19 UTC; gh13047</td>
</tr>
<tr>
<td>Author:</td>
<td>Gibran Hemani <a href="https://orcid.org/0000-0003-0920-1055"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Ben Elsworth <a href="https://orcid.org/0000-0001-7328-4233"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Tom Palmer <a href="https://orcid.org/0000-0003-4655-4511"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Rita Rasteiro <a href="https://orcid.org/0000-0002-4217-3060"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gibran Hemani &lt;g.hemani@bristol.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-28 16:10:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='.data'>rlang .data</h2><span id='topic+.data'></span>

<h3>Description</h3>

<p>rlang .data
</p>

<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='afl2_chrpos'>Look up allele frequencies and LD scores for 1000 genomes populations by chrpos</h2><span id='topic+afl2_chrpos'></span>

<h3>Description</h3>

<p>Look up allele frequencies and LD scores for 1000 genomes populations by chrpos
</p>


<h3>Usage</h3>

<pre><code class='language-R'>afl2_chrpos(chrpos, reference = "1000g")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="afl2_chrpos_+3A_chrpos">chrpos</code></td>
<td>
<p>list of <code style="white-space: pre;">&#8288;&lt;chr&gt;:&lt;pos&gt;&#8288;</code> in build 37, e.g. <code>c("3:46414943", "3:122991235")</code>.
Also allows ranges e.g <code>"7:105561135-105563135"</code></p>
</td></tr>
<tr><td><code id="afl2_chrpos_+3A_reference">reference</code></td>
<td>
<p>Default=<code>"1000g"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing ancestry specific LD scores and allele frequencies for each variant
</p>

<hr>
<h2 id='afl2_list'>Retrieve a allele frequency and LD scores for pre-defined lists of variants</h2><span id='topic+afl2_list'></span>

<h3>Description</h3>

<p>Data frame includes 1000 genomes metadata including sample sizes,
allele frequency and LD score, separated by 5 super populations
(EUR = European, AFR = African, EAS = East Asian, AMR = Admixed American,
SAS = South Asian)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>afl2_list(variantlist = c("reduced", "hapmap3")[1])
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="afl2_list_+3A_variantlist">variantlist</code></td>
<td>
<p>Choose pre-defined list. reduced = ~20k SNPs that are
common in all super populations (default). hapmap3 = ~1.3 million hm3 SNPs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing ancestry specific LD scores and allele frequencies for each variant
</p>

<hr>
<h2 id='afl2_rsid'>Look up allele frequencies and LD scores for 1000 genomes populations by rsid</h2><span id='topic+afl2_rsid'></span>

<h3>Description</h3>

<p>Look up allele frequencies and LD scores for 1000 genomes populations by rsid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>afl2_rsid(rsid, reference = "1000g")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="afl2_rsid_+3A_rsid">rsid</code></td>
<td>
<p>Vector of rsids</p>
</td></tr>
<tr><td><code id="afl2_rsid_+3A_reference">reference</code></td>
<td>
<p>Default=<code>"1000g"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing ancestry specific LD scores and allele frequencies for each variant
</p>

<hr>
<h2 id='api_query'>Wrapper for sending queries and payloads to API</h2><span id='topic+api_query'></span>

<h3>Description</h3>

<p>There are a number of different GET and POST endpoints in the GWAS database API.
This is a generic way to access them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>api_query(
  path,
  query = NULL,
  access_token = check_access_token(),
  method = "GET",
  silent = TRUE,
  encode = "json",
  timeout = 300
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="api_query_+3A_path">path</code></td>
<td>
<p>Either a full query path (e.g. for get) or an endpoint (e.g. for post) queries</p>
</td></tr>
<tr><td><code id="api_query_+3A_query">query</code></td>
<td>
<p>If post query, provide a list of arguments as the payload. <code>NULL</code> by default</p>
</td></tr>
<tr><td><code id="api_query_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token.
Used to authenticate level of access to data. By default, checks if already
authenticated through <code><a href="#topic+get_access_token">get_access_token</a></code> and if not then does not perform authentication</p>
</td></tr>
<tr><td><code id="api_query_+3A_method">method</code></td>
<td>
<p><code>"GET"</code> (default) or <code>"POST"</code>, <code>"DELETE"</code> etc</p>
</td></tr>
<tr><td><code id="api_query_+3A_silent">silent</code></td>
<td>
<p><code>TRUE</code>/<code>FALSE</code> to be passed to httr call. <code>TRUE</code> by default</p>
</td></tr>
<tr><td><code id="api_query_+3A_encode">encode</code></td>
<td>
<p>Default = <code>"json"</code>, see <code><a href="httr.html#topic+POST">httr::POST</a></code> for options</p>
</td></tr>
<tr><td><code id="api_query_+3A_timeout">timeout</code></td>
<td>
<p>Default = <code>300</code>, avoid increasing this, preferentially
simplify the query first.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>httr response object
</p>

<hr>
<h2 id='api_status'>MR-Base server status</h2><span id='topic+api_status'></span>

<h3>Description</h3>

<p>MR-Base server status
</p>


<h3>Usage</h3>

<pre><code class='language-R'>api_status()
</code></pre>


<h3>Value</h3>

<p>list of values regarding status
</p>

<hr>
<h2 id='associations'>Query specific variants from specific GWAS</h2><span id='topic+associations'></span>

<h3>Description</h3>

<p>Every rsid is searched for against each requested GWAS id. To get a list of
available GWAS ids, or to find their meta data, use <code><a href="#topic+gwasinfo">gwasinfo</a></code>.
Can request LD proxies for instances when the requested rsid is not present
in a particular GWAS dataset. This currently only uses an LD reference panel
composed of Europeans in 1000 genomes version 3.
It is also restricted to biallelic single nucleotide polymorphisms (no indels),
with European MAF &gt; 0.01.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>associations(
  variants,
  id,
  proxies = 1,
  r2 = 0.8,
  align_alleles = 1,
  palindromes = 1,
  maf_threshold = 0.3,
  access_token = check_access_token()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="associations_+3A_variants">variants</code></td>
<td>
<p>Array of variants e.g. <code>c("rs234", "7:105561135-105563135")</code></p>
</td></tr>
<tr><td><code id="associations_+3A_id">id</code></td>
<td>
<p>Array of GWAS studies to query. See <code><a href="#topic+gwasinfo">gwasinfo</a></code> for available studies</p>
</td></tr>
<tr><td><code id="associations_+3A_proxies">proxies</code></td>
<td>
<p><code>0</code> or (default) <code>1</code> - indicating whether to look for proxies</p>
</td></tr>
<tr><td><code id="associations_+3A_r2">r2</code></td>
<td>
<p>Minimum proxy LD rsq value. Default=<code>0.8</code></p>
</td></tr>
<tr><td><code id="associations_+3A_align_alleles">align_alleles</code></td>
<td>
<p>Try to align tag alleles to target alleles (if <code>proxies = 1</code>).
<code>1</code> = yes (default), <code>0</code> = no</p>
</td></tr>
<tr><td><code id="associations_+3A_palindromes">palindromes</code></td>
<td>
<p>Allow palindromic SNPs (if <code>proxies = 1</code>). <code>1</code> = yes (default), <code>0</code> = no</p>
</td></tr>
<tr><td><code id="associations_+3A_maf_threshold">maf_threshold</code></td>
<td>
<p>MAF threshold to try to infer palindromic SNPs. Default = <code>0.3</code>.</p>
</td></tr>
<tr><td><code id="associations_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token.
Used to authenticate level of access to data.
By default, checks if already authenticated through <code><a href="#topic+get_access_token">get_access_token</a></code> and
if not then does not perform authentication</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe
</p>

<hr>
<h2 id='batch_from_id'>Extract batch name from study ID</h2><span id='topic+batch_from_id'></span>

<h3>Description</h3>

<p>Extract batch name from study ID
</p>


<h3>Usage</h3>

<pre><code class='language-R'>batch_from_id(id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="batch_from_id_+3A_id">id</code></td>
<td>
<p>Array of study IDs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Array of batch names
</p>

<hr>
<h2 id='batches'>Get list of data batches in IEU GWAS database</h2><span id='topic+batches'></span>

<h3>Description</h3>

<p>Get list of data batches in IEU GWAS database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>batches(access_token = check_access_token())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="batches_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token. Used to authenticate level of access to data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>

<hr>
<h2 id='check_access_token'>Check if authentication has been made</h2><span id='topic+check_access_token'></span>

<h3>Description</h3>

<p>If a call to <code><a href="#topic+get_access_token">get_access_token()</a></code> has been made then it will have generated <code>mrbase.oauth</code>.
Pass the token if it is present, if not, return <code>NULL</code> and do not authenticate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_access_token()
</code></pre>


<h3>Value</h3>

<p>NULL or access_token depending on current authentication state
</p>

<hr>
<h2 id='editcheck'>Check datasets that are in process of being uploaded</h2><span id='topic+editcheck'></span>

<h3>Description</h3>

<p>Check datasets that are in process of being uploaded
</p>


<h3>Usage</h3>

<pre><code class='language-R'>editcheck(id, access_token = check_access_token())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="editcheck_+3A_id">id</code></td>
<td>
<p>ID</p>
</td></tr>
<tr><td><code id="editcheck_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token.
Used to authenticate level of access to data.
By default, checks if already authenticated through <code><a href="#topic+get_access_token">get_access_token</a></code>
and if not then does not perform authentication</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe
</p>

<hr>
<h2 id='fill_n'>Look up sample sizes when meta data is missing from associations</h2><span id='topic+fill_n'></span>

<h3>Description</h3>

<p>Look up sample sizes when meta data is missing from associations
</p>
<p>Look up sample sizes when meta data is missing from associations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fill_n(d)

fill_n(d)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fill_n_+3A_d">d</code></td>
<td>
<p>Output from <code><a href="#topic+associations">associations</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame which is an updated version of the input but with sample sizes inferred where missing
</p>
<p>Updated version of d
</p>

<hr>
<h2 id='get_access_token'>Get access token for OAuth2 access to MR Base</h2><span id='topic+get_access_token'></span>

<h3>Description</h3>

<p>Get access token for OAuth2 access to MR Base
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_access_token()
</code></pre>


<h3>Value</h3>

<p>access token string
</p>

<hr>
<h2 id='get_query_content'>Parse out json response from httr object</h2><span id='topic+get_query_content'></span>

<h3>Description</h3>

<p>Parse out json response from httr object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_query_content(response)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_query_content_+3A_response">response</code></td>
<td>
<p>Output from httr</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Parsed json output from query, often in form of data frame.
If status code is not successful then return the actual response
</p>

<hr>
<h2 id='gwasinfo'>Get list of studies with available GWAS summary statistics through API</h2><span id='topic+gwasinfo'></span>

<h3>Description</h3>

<p>Get list of studies with available GWAS summary statistics through API
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gwasinfo(id = NULL, access_token = check_access_token())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gwasinfo_+3A_id">id</code></td>
<td>
<p>List of MR-Base IDs to retrieve. If <code>NULL</code> (default) retrieves all
available datasets</p>
</td></tr>
<tr><td><code id="gwasinfo_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token. Used to authenticate level of access to data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of details for all available studies
</p>

<hr>
<h2 id='infer_ancestry'>Infer ancestry of GWAS dataset by matching against 1000 genomes allele frequencies</h2><span id='topic+infer_ancestry'></span>

<h3>Description</h3>

<p>Uses ~20k SNPs selected for common frequency across 5 major super populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_ancestry(d, snpinfo = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infer_ancestry_+3A_d">d</code></td>
<td>
<p>Data frame containing at least <code>rsid</code> and <code>eaf</code> columns.
e.g. output from associations</p>
</td></tr>
<tr><td><code id="infer_ancestry_+3A_snpinfo">snpinfo</code></td>
<td>
<p>Output from <code><a href="#topic+afl2_list">afl2_list</a></code>, <code><a href="#topic+afl2_rsid">afl2_rsid</a></code> or <code><a href="#topic+afl2_chrpos">afl2_chrpos</a></code>.
If <code>NULL</code> then <code><a href="#topic+afl2_list">afl2_list()</a></code> is used by default</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame ordered by most likely ancestry, each row represents a super population and cor column represents the correlation between the GWAS dataset and the 1000 genomes super population allele frequencies
</p>

<hr>
<h2 id='ld_clump'>Perform LD clumping on SNP data</h2><span id='topic+ld_clump'></span>

<h3>Description</h3>

<p>Uses PLINK clumping method, where SNPs in LD within a particular window will be pruned.
The SNP with the lowest p-value is retained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_clump(
  dat = NULL,
  clump_kb = 10000,
  clump_r2 = 0.001,
  clump_p = 0.99,
  pop = "EUR",
  access_token = NULL,
  bfile = NULL,
  plink_bin = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld_clump_+3A_dat">dat</code></td>
<td>
<p>Dataframe. Must have a variant name column (<code>rsid</code>) and pval column called <code>pval</code>.
If <code>id</code> is present then clumping will be done per unique id.</p>
</td></tr>
<tr><td><code id="ld_clump_+3A_clump_kb">clump_kb</code></td>
<td>
<p>Clumping kb window. Default is very strict, <code>10000</code></p>
</td></tr>
<tr><td><code id="ld_clump_+3A_clump_r2">clump_r2</code></td>
<td>
<p>Clumping r2 threshold. Default is very strict, <code>0.001</code></p>
</td></tr>
<tr><td><code id="ld_clump_+3A_clump_p">clump_p</code></td>
<td>
<p>Clumping sig level for index variants. Default = <code>1</code> (i.e. no threshold)</p>
</td></tr>
<tr><td><code id="ld_clump_+3A_pop">pop</code></td>
<td>
<p>Super-population to use as reference panel. Default = <code>"EUR"</code>.
Options are <code>"EUR"</code>, <code>"SAS"</code>, <code>"EAS"</code>, <code>"AFR"</code>, <code>"AMR"</code>.
<code>'legacy'</code> also available - which is a previously used verison of the EUR
panel with a slightly different set of markers</p>
</td></tr>
<tr><td><code id="ld_clump_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token. Used to authenticate level of access to data</p>
</td></tr>
<tr><td><code id="ld_clump_+3A_bfile">bfile</code></td>
<td>
<p>If this is provided then will use the API. Default = <code>NULL</code></p>
</td></tr>
<tr><td><code id="ld_clump_+3A_plink_bin">plink_bin</code></td>
<td>
<p>If <code>NULL</code> and <code>bfile</code> is not <code>NULL</code> then will detect
packaged plink binary for specific OS. Otherwise specify path to plink binary.
Default = <code>NULL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function interacts with the OpenGWAS API, which houses LD reference panels
for the 5 super-populations in the 1000 genomes reference panel.
It includes only bi-allelic SNPs with MAF &gt; 0.01, so it's quite possible that
a variant you want to include in the clumping process will be absent.
If it is absent, it will be automatically excluded from the results.
</p>
<p>You can check if your variants are present in the LD reference panel using
<code><a href="#topic+ld_reflookup">ld_reflookup()</a></code>.
</p>
<p>This function does put load on the OpenGWAS servers, which makes life more
difficult for other users. We have implemented a method and made available
the LD reference panels to perform clumping locally, see
<code><a href="#topic+ld_clump">ld_clump()</a></code> and related vignettes for details.
</p>


<h3>Value</h3>

<p>Data frame
</p>

<hr>
<h2 id='ld_clump_api'>Perform clumping on the chosen variants using through API</h2><span id='topic+ld_clump_api'></span>

<h3>Description</h3>

<p>Perform clumping on the chosen variants using through API
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_clump_api(
  dat,
  clump_kb = 10000,
  clump_r2 = 0.1,
  clump_p,
  pop = "EUR",
  access_token = check_access_token()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld_clump_api_+3A_dat">dat</code></td>
<td>
<p>Dataframe. Must have a variant name column (<code>variant</code>) and pval column called <code>pval</code>.
If <code>id</code> is present then clumping will be done per unique id.</p>
</td></tr>
<tr><td><code id="ld_clump_api_+3A_clump_kb">clump_kb</code></td>
<td>
<p>Clumping kb window. Default is very strict, <code>10000</code></p>
</td></tr>
<tr><td><code id="ld_clump_api_+3A_clump_r2">clump_r2</code></td>
<td>
<p>Clumping r2 threshold. Default is very strict, <code>0.001</code></p>
</td></tr>
<tr><td><code id="ld_clump_api_+3A_clump_p">clump_p</code></td>
<td>
<p>Clumping sig level for index variants. Default = <code>1</code> (i.e. no threshold)</p>
</td></tr>
<tr><td><code id="ld_clump_api_+3A_pop">pop</code></td>
<td>
<p>Super-population to use as reference panel. Default = <code>"EUR"</code>.
Options are <code>"EUR"</code>, <code>"SAS"</code>, <code>"EAS"</code>, <code>"AFR"</code>, <code>"AMR"</code></p>
</td></tr>
<tr><td><code id="ld_clump_api_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token. Used to authenticate level of access to data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame of only independent variants
</p>

<hr>
<h2 id='ld_clump_local'>Wrapper for clump function using local plink binary and ld reference dataset</h2><span id='topic+ld_clump_local'></span>

<h3>Description</h3>

<p>Wrapper for clump function using local plink binary and ld reference dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_clump_local(dat, clump_kb, clump_r2, clump_p, bfile, plink_bin)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld_clump_local_+3A_dat">dat</code></td>
<td>
<p>Dataframe. Must have a variant name column (<code>variant</code>) and pval column called <code>pval</code>.
If <code>id</code> is present then clumping will be done per unique id.</p>
</td></tr>
<tr><td><code id="ld_clump_local_+3A_clump_kb">clump_kb</code></td>
<td>
<p>Clumping kb window. Default is very strict, <code>10000</code></p>
</td></tr>
<tr><td><code id="ld_clump_local_+3A_clump_r2">clump_r2</code></td>
<td>
<p>Clumping r2 threshold. Default is very strict, <code>0.001</code></p>
</td></tr>
<tr><td><code id="ld_clump_local_+3A_clump_p">clump_p</code></td>
<td>
<p>Clumping sig level for index variants. Default = <code>1</code> (i.e. no threshold)</p>
</td></tr>
<tr><td><code id="ld_clump_local_+3A_bfile">bfile</code></td>
<td>
<p>If this is provided then will use the API. Default = <code>NULL</code></p>
</td></tr>
<tr><td><code id="ld_clump_local_+3A_plink_bin">plink_bin</code></td>
<td>
<p>Specify path to plink binary. Default = <code>NULL</code>.
See <a href="https://github.com/explodecomputer/plinkbinr">https://github.com/explodecomputer/plinkbinr</a> for convenient access to plink binaries</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame of clumped variants
</p>

<hr>
<h2 id='ld_matrix'>Get LD matrix for list of SNPs</h2><span id='topic+ld_matrix'></span>

<h3>Description</h3>

<p>This function takes a list of SNPs and searches for them in a specified
super-population in the 1000 Genomes phase 3 reference panel.
It then creates an LD matrix of r values (signed, and not squared).
All LD values are with respect to the major alleles in the 1000G dataset.
You can specify whether the allele names are displayed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_matrix(
  variants,
  with_alleles = TRUE,
  pop = "EUR",
  bfile = NULL,
  plink_bin = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld_matrix_+3A_variants">variants</code></td>
<td>
<p>List of variants (rsids)</p>
</td></tr>
<tr><td><code id="ld_matrix_+3A_with_alleles">with_alleles</code></td>
<td>
<p>Whether to append the allele names to the SNP names. Default: <code>TRUE</code></p>
</td></tr>
<tr><td><code id="ld_matrix_+3A_pop">pop</code></td>
<td>
<p>Super-population to use as reference panel. Default = <code>"EUR"</code>.
Options are <code>"EUR"</code>, <code>"SAS"</code>, <code>"EAS"</code>, <code>"AFR"</code>, <code>"AMR"</code>.
<code>'legacy'</code> also available - which is a previously used verison of the EUR
panel with a slightly different set of markers</p>
</td></tr>
<tr><td><code id="ld_matrix_+3A_bfile">bfile</code></td>
<td>
<p>If this is provided then will use the API. Default = <code>NULL</code></p>
</td></tr>
<tr><td><code id="ld_matrix_+3A_plink_bin">plink_bin</code></td>
<td>
<p>If <code>NULL</code> and bfile is not <code>NULL</code> then will detect packaged
plink binary for specific OS. Otherwise specify path to plink binary. Default = <code>NULL</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The data used for generating the LD matrix includes only bi-allelic SNPs
with MAF &gt; 0.01, so it's quite possible that a variant you want to include
will be absent. If it is absent, it will be automatically excluded from the results.
</p>
<p>You can check if your variants are present in the LD reference panel using
<code><a href="#topic+ld_reflookup">ld_reflookup()</a></code>
</p>
<p>This function does put load on the OpenGWAS servers, which makes life more
difficult for other users, and has been limited to analyse only up to 500
variants at a time. We have implemented a method and made available the LD
reference panels to perform the operation locally, see <code><a href="#topic+ld_matrix">ld_matrix()</a></code> and
related vignettes for details.
</p>


<h3>Value</h3>

<p>Matrix of LD r values
</p>

<hr>
<h2 id='ld_matrix_local'>Get LD matrix using local plink binary and reference dataset</h2><span id='topic+ld_matrix_local'></span>

<h3>Description</h3>

<p>Get LD matrix using local plink binary and reference dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_matrix_local(variants, bfile, plink_bin, with_alleles = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld_matrix_local_+3A_variants">variants</code></td>
<td>
<p>List of variants (rsids)</p>
</td></tr>
<tr><td><code id="ld_matrix_local_+3A_bfile">bfile</code></td>
<td>
<p>Path to bed/bim/fam ld reference panel</p>
</td></tr>
<tr><td><code id="ld_matrix_local_+3A_plink_bin">plink_bin</code></td>
<td>
<p>Specify path to plink binary. Default = <code>NULL</code>.
See <a href="https://github.com/explodecomputer/plinkbinr">https://github.com/explodecomputer/plinkbinr</a> for convenient access to plink binaries</p>
</td></tr>
<tr><td><code id="ld_matrix_local_+3A_with_alleles">with_alleles</code></td>
<td>
<p>Whether to append the allele names to the SNP names.
Default: <code>TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>

<hr>
<h2 id='ld_reflookup'>Check which rsids are present in a remote LD reference panel</h2><span id='topic+ld_reflookup'></span>

<h3>Description</h3>

<p>Provide a list of rsids that you may want to perform LD operations on to
check if they are present in the LD reference panel. If they are not then
some functions e.g. <code><a href="#topic+ld_clump">ld_clump</a></code> will exclude them from the analysis,
so you may want to consider how to handle those variants in your data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ld_reflookup(rsid, pop = "EUR")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ld_reflookup_+3A_rsid">rsid</code></td>
<td>
<p>Array of rsids to check</p>
</td></tr>
<tr><td><code id="ld_reflookup_+3A_pop">pop</code></td>
<td>
<p>Super-population to use as reference panel. Default = <code>"EUR"</code>.
Options are <code>"EUR"</code>, <code>"SAS"</code>, <code>"EAS"</code>, <code>"AFR"</code>, <code>"AMR"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Array of rsids that are present in the LD reference panel
</p>

<hr>
<h2 id='legacy_ids'>Convert current IDs to legacy IDs</h2><span id='topic+legacy_ids'></span>

<h3>Description</h3>

<p>Convert current IDs to legacy IDs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>legacy_ids(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="legacy_ids_+3A_x">x</code></td>
<td>
<p>Vector of ids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of back compatible ids
</p>

<hr>
<h2 id='logging_info'>Details of how access token logs are used</h2><span id='topic+logging_info'></span>

<h3>Description</h3>

<p>Details of how access token logs are used
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logging_info()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='phewas'>Perform fast phewas of a specific variants against all available GWAS datasets</h2><span id='topic+phewas'></span>

<h3>Description</h3>

<p>This is faster than doing it manually through <code><a href="#topic+associations">associations</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phewas(
  variants,
  pval = 1e-05,
  batch = c(),
  access_token = check_access_token()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phewas_+3A_variants">variants</code></td>
<td>
<p>Array of variants e.g. <code>c("rs234", "7:105561135-105563135")</code></p>
</td></tr>
<tr><td><code id="phewas_+3A_pval">pval</code></td>
<td>
<p>p-value threshold. Default = <code>0.00001</code></p>
</td></tr>
<tr><td><code id="phewas_+3A_batch">batch</code></td>
<td>
<p>Vector of batch IDs to search across. If <code>c()</code> (default) then returns all batches</p>
</td></tr>
<tr><td><code id="phewas_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token.
Used to authenticate level of access to data.
By default, checks if already authenticated through <code><a href="#topic+get_access_token">get_access_token</a></code> and
if not then does not perform authentication</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe
</p>

<hr>
<h2 id='print.ApiStatus'>Print API status</h2><span id='topic+print.ApiStatus'></span>

<h3>Description</h3>

<p>Print API status
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ApiStatus'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ApiStatus_+3A_x">x</code></td>
<td>
<p>Output from <code><a href="#topic+api_status">api_status</a></code></p>
</td></tr>
<tr><td><code id="print.ApiStatus_+3A_...">...</code></td>
<td>
<p>Unused, for extensibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Print out of API status
</p>

<hr>
<h2 id='print.GwasInfo'>Print GWAS information</h2><span id='topic+print.GwasInfo'></span>

<h3>Description</h3>

<p>Print GWAS information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GwasInfo'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.GwasInfo_+3A_x">x</code></td>
<td>
<p>Output from <code><a href="#topic+gwasinfo">gwasinfo</a></code></p>
</td></tr>
<tr><td><code id="print.GwasInfo_+3A_...">...</code></td>
<td>
<p>Unused, for extensibility</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Print out of GWAS information
</p>

<hr>
<h2 id='revoke_access_token'>Revoke access token for MR Base</h2><span id='topic+revoke_access_token'></span>

<h3>Description</h3>

<p>Revoke access token for MR Base
</p>


<h3>Usage</h3>

<pre><code class='language-R'>revoke_access_token()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='select_api'>Toggle API address between development and release</h2><span id='topic+select_api'></span>

<h3>Description</h3>

<p>Toggle API address between development and release
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_api(where = "public", silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_api_+3A_where">where</code></td>
<td>
<p>Which API to use. Choice between <code>"public"</code>, <code>"private"</code>, <code>"dev1"</code>, <code>"dev2"</code>.
Default = <code>"public"</code>.</p>
</td></tr>
<tr><td><code id="select_api_+3A_silent">silent</code></td>
<td>
<p>Silent? Default = FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>

<hr>
<h2 id='tophits'>Obtain top hits from a GWAS dataset</h2><span id='topic+tophits'></span>

<h3>Description</h3>

<p>By default performs clumping on the server side.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tophits(
  id,
  pval = 5e-08,
  clump = 1,
  r2 = 0.001,
  kb = 10000,
  pop = "EUR",
  force_server = FALSE,
  access_token = check_access_token()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tophits_+3A_id">id</code></td>
<td>
<p>Array of GWAS studies to query. See <code><a href="#topic+gwasinfo">gwasinfo</a></code> for available studies</p>
</td></tr>
<tr><td><code id="tophits_+3A_pval">pval</code></td>
<td>
<p>use this p-value threshold. Default = <code>5e-8</code></p>
</td></tr>
<tr><td><code id="tophits_+3A_clump">clump</code></td>
<td>
<p>whether to clump (<code>1</code>) or not (<code>0</code>). Default = <code>1</code></p>
</td></tr>
<tr><td><code id="tophits_+3A_r2">r2</code></td>
<td>
<p>use this clumping r2 threshold. Default is very strict, <code>0.001</code></p>
</td></tr>
<tr><td><code id="tophits_+3A_kb">kb</code></td>
<td>
<p>use this clumping kb window. Default is very strict, <code>10000</code></p>
</td></tr>
<tr><td><code id="tophits_+3A_pop">pop</code></td>
<td>
<p>Super-population to use as reference panel. Default = <code>"EUR"</code>.
Options are <code>"EUR"</code>, <code>"SAS"</code>, <code>"EAS"</code>, <code>"AFR"</code>, <code>"AMR"</code></p>
</td></tr>
<tr><td><code id="tophits_+3A_force_server">force_server</code></td>
<td>
<p>Logical. By default will return preclumped hits.
p-value threshold 5e-8, with r2 threshold 0.001 and kb threshold 10000,
using only SNPs with MAF &gt; 0.01 in the European samples in 1000 genomes.
If force_server = <code>TRUE</code> then will recompute using server side LD reference panel.</p>
</td></tr>
<tr><td><code id="tophits_+3A_access_token">access_token</code></td>
<td>
<p>Google OAuth2 access token.
Used to authenticate level of access to data.
By default, checks if already authenticated through <code><a href="#topic+get_access_token">get_access_token</a></code>
and if not then does not perform authentication.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe. If <code>gwasglue = TRUE</code> then returns a gwasglue2 object.
</p>

<hr>
<h2 id='variants_chrpos'>Obtain information about chr pos and surrounding region</h2><span id='topic+variants_chrpos'></span>

<h3>Description</h3>

<p>For a list of chromosome and positions, finds all variants within a given radius
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants_chrpos(chrpos, radius = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_chrpos_+3A_chrpos">chrpos</code></td>
<td>
<p>list of <code style="white-space: pre;">&#8288;&lt;chr&gt;:&lt;pos&gt;&#8288;</code> in build 37,
e.g. <code>c("3:46414943", "3:122991235")</code>. Also allows ranges e.g. <code>"7:105561135-105563135"</code></p>
</td></tr>
<tr><td><code id="variants_chrpos_+3A_radius">radius</code></td>
<td>
<p>Radius around each chrpos, default = <code>0</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame
</p>

<hr>
<h2 id='variants_gene'>Obtain variants around a gene</h2><span id='topic+variants_gene'></span>

<h3>Description</h3>

<p>Provide a gene identified, either Ensembl or Entrez
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants_gene(gene, radius = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_gene_+3A_gene">gene</code></td>
<td>
<p>Vector of genes, either Ensembl or Entrez,
e.g. <code>c("ENSG00000123374", "ENSG00000160791")</code> or <code>1017</code></p>
</td></tr>
<tr><td><code id="variants_gene_+3A_radius">radius</code></td>
<td>
<p>Radius around the gene region to include. Default = <code>0</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with the following columns
</p>

<hr>
<h2 id='variants_rsid'>Obtain information about rsid</h2><span id='topic+variants_rsid'></span>

<h3>Description</h3>

<p>Obtain information about rsid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants_rsid(rsid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_rsid_+3A_rsid">rsid</code></td>
<td>
<p>Vector of rsids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame
</p>

<hr>
<h2 id='variants_to_rsid'>Convert mixed array of rsid and chrpos to list of rsid</h2><span id='topic+variants_to_rsid'></span>

<h3>Description</h3>

<p>Convert mixed array of rsid and chrpos to list of rsid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>variants_to_rsid(variants)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="variants_to_rsid_+3A_variants">variants</code></td>
<td>
<p>Array of variants e.g. <code>c("rs234", "7:105561135-105563135")</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of rsids
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
