<!DOCTYPE html><html><head><title>Help for package ifCNVR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ifCNVR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abSamples'><p>abSamples</p></a></li>
<li><a href='#abTargets'><p>abTargets</p></a></li>
<li><a href='#calculateRatio'><p>calculate Ratio</p></a></li>
<li><a href='#calculateScore'><p>calculate Score</p></a></li>
<li><a href='#CreateReadsMatrix'><p>CreateReadsMatrix</p></a></li>
<li><a href='#generateReport'><p>generateReport</p></a></li>
<li><a href='#normalizeReads'><p>normalizeReads</p></a></li>
<li><a href='#readsMatrixExample'><p>Example dataset</p></a></li>
<li><a href='#scoring'><p>scoring</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Isolation-Forest Based 'CNV' Detection from 'NGS' Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SimCab-CHU/ifCNVR">https://github.com/SimCab-CHU/ifCNVR</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Automatically detects Copy Number Variations (CNV) from Next Generation Sequencing data using a machine learning algorithm, Isolation forest. More details about the method can be found in the paper by Cabello-Aguilar (2022) &lt;<a href="https://doi.org/10.1101%2F2022.01.03.474771">doi:10.1101/2022.01.03.474771</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, rmarkdown, isotree</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-15 06:11:14 UTC; admin</td>
</tr>
<tr>
<td>Author:</td>
<td>Simon Cabello-Aguilar
    <a href="https://orcid.org/0000-0002-2904-7430"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Simon Cabello-Aguilar &lt;s-cabelloaguilar@chu-montpellier.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-15 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='abSamples'>abSamples</h2><span id='topic+abSamples'></span>

<h3>Description</h3>

<p>abSamples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abSamples(readsMatrix, conta = "auto", q = 0.99, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abSamples_+3A_readsmatrix">readsMatrix</code></td>
<td>
<p>a matrix of the number of reads per target</p>
</td></tr>
<tr><td><code id="abSamples_+3A_conta">conta</code></td>
<td>
<p>a parameter for the isotree function</p>
</td></tr>
<tr><td><code id="abSamples_+3A_q">q</code></td>
<td>
<p>quantile</p>
</td></tr>
<tr><td><code id="abSamples_+3A_verbose">verbose</code></td>
<td>
<p>a boolean</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the aberrant and normal samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>readsMatrix = data.frame(targets=paste0("target_",seq(1,50)), matrix(runif(500),nrow=50,ncol=10))
abSamples(readsMatrix)

</code></pre>

<hr>
<h2 id='abTargets'>abTargets</h2><span id='topic+abTargets'></span>

<h3>Description</h3>

<p>abTargets
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abTargets(readsMatrix, abSamples, opt = "regular", pred = 0.6)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="abTargets_+3A_readsmatrix">readsMatrix</code></td>
<td>
<p>the reads matrix</p>
</td></tr>
<tr><td><code id="abTargets_+3A_absamples">abSamples</code></td>
<td>
<p>the abSamples list result of the abSamples function</p>
</td></tr>
<tr><td><code id="abTargets_+3A_opt">opt</code></td>
<td>
<p>&quot;regular&quot; or &quot;extensive&quot; a parameter</p>
</td></tr>
<tr><td><code id="abTargets_+3A_pred">pred</code></td>
<td>
<p>a threshold on the isolation forest outlier prediction (range=[0,1[)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of dataframes of the targets tagged as outliers
</p>


<h3>Examples</h3>

<pre><code class='language-R'>abTargets(readsMatrixExample,abSamples(readsMatrixExample))
</code></pre>

<hr>
<h2 id='calculateRatio'>calculate Ratio</h2><span id='topic+calculateRatio'></span>

<h3>Description</h3>

<p>calculate Ratio
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateRatio(readsMatrix, abSamples, roi, soi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateRatio_+3A_readsmatrix">readsMatrix</code></td>
<td>
<p>the reads matrix</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_absamples">abSamples</code></td>
<td>
<p>the abSamples list result of the abSamples function</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_roi">roi</code></td>
<td>
<p>the region of interest</p>
</td></tr>
<tr><td><code id="calculateRatio_+3A_soi">soi</code></td>
<td>
<p>the sample of interest</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the ratio associated with the abTargets
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calculateRatio(readsMatrixExample, abSamples(readsMatrixExample), "EGFR-Ex20", "sample_2")
</code></pre>

<hr>
<h2 id='calculateScore'>calculate Score</h2><span id='topic+calculateScore'></span>

<h3>Description</h3>

<p>calculate Score
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculateScore(
  readsMatrix,
  abSamples,
  abTargets,
  roi = "Gene",
  sep = "-",
  thrScore = 7
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculateScore_+3A_readsmatrix">readsMatrix</code></td>
<td>
<p>a reads matrix with samples in columns and targets in lines (the first column are the targets)</p>
</td></tr>
<tr><td><code id="calculateScore_+3A_absamples">abSamples</code></td>
<td>
<p>the abSamples list result of the abSamples function</p>
</td></tr>
<tr><td><code id="calculateScore_+3A_abtargets">abTargets</code></td>
<td>
<p>a list of dataframes of the targets tagged as outliers result of the abTargets() function</p>
</td></tr>
<tr><td><code id="calculateScore_+3A_roi">roi</code></td>
<td>
<p>the region of interest (Gene or Gene-Exon)</p>
</td></tr>
<tr><td><code id="calculateScore_+3A_sep">sep</code></td>
<td>
<p>a character the separator between roi in the bed file</p>
</td></tr>
<tr><td><code id="calculateScore_+3A_thrscore">thrScore</code></td>
<td>
<p>(default 0) a threshold on the localization score</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the score associated with the abSamples in the desired roi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>abS &lt;- abSamples(readsMatrixExample)
abT &lt;- abTargets(readsMatrixExample,abSamples(readsMatrixExample))
calculateScore(readsMatrixExample, abS, abT, sep="-")
</code></pre>

<hr>
<h2 id='CreateReadsMatrix'>CreateReadsMatrix</h2><span id='topic+CreateReadsMatrix'></span>

<h3>Description</h3>

<p>CreateReadsMatrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateReadsMatrix(
  bamPath,
  bedFile,
  bedtoolsPath,
  outputFile = "n",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateReadsMatrix_+3A_bampath">bamPath</code></td>
<td>
<p>a path leading to the .bam and .bai files</p>
</td></tr>
<tr><td><code id="CreateReadsMatrix_+3A_bedfile">bedFile</code></td>
<td>
<p>a path leading to the .bed file (Warning replace chrX by X in the position)</p>
</td></tr>
<tr><td><code id="CreateReadsMatrix_+3A_bedtoolspath">bedtoolsPath</code></td>
<td>
<p>the path leading to bedtools</p>
</td></tr>
<tr><td><code id="CreateReadsMatrix_+3A_outputfile">outputFile</code></td>
<td>
<p>(optional) a path leading to a text file</p>
</td></tr>
<tr><td><code id="CreateReadsMatrix_+3A_verbose">verbose</code></td>
<td>
<p>a boolean</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a reads matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>bamPath &lt;- system.file("extdata/",package = "ifCNVR")
bed &lt;- system.file("bedFile.bed",package = "ifCNVR")
bedtools &lt;- 'n'
readsMatrix &lt;- CreateReadsMatrix(bamPath, bed, bedtools)

</code></pre>

<hr>
<h2 id='generateReport'>generateReport</h2><span id='topic+generateReport'></span>

<h3>Description</h3>

<p>generateReport
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateReport(outputFile = "n", readsMatrix, resTable, CNVpos)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateReport_+3A_outputfile">outputFile</code></td>
<td>
<p>a path to the html output file</p>
</td></tr>
<tr><td><code id="generateReport_+3A_readsmatrix">readsMatrix</code></td>
<td>
<p>the reads matrix</p>
</td></tr>
<tr><td><code id="generateReport_+3A_restable">resTable</code></td>
<td>
<p>the table result of the CalculateScore() function</p>
</td></tr>
<tr><td><code id="generateReport_+3A_cnvpos">CNVpos</code></td>
<td>
<p>the CNVpos list result of the abSamples function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a html report
</p>


<h3>Examples</h3>

<pre><code class='language-R'>generateReport()
</code></pre>

<hr>
<h2 id='normalizeReads'>normalizeReads</h2><span id='topic+normalizeReads'></span>

<h3>Description</h3>

<p>normalizeReads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalizeReads(readsMatrix)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalizeReads_+3A_readsmatrix">readsMatrix</code></td>
<td>
<p>a reads matrix with samples in columns and targets in lines (the first column are the targets)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a normalized reads matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>normReads &lt;- normalizeReads(readsMatrixExample)
</code></pre>

<hr>
<h2 id='readsMatrixExample'>Example dataset</h2><span id='topic+readsMatrixExample'></span>

<h3>Description</h3>

<p>A dataset for the examples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readsMatrixExample
</code></pre>


<h3>Format</h3>

<p>A data frame with 11 rows and 6 variables:
</p>

<dl>
<dt>targets</dt><dd><p>the targets</p>
</dd>
<dt>sample_1</dt><dd><p>the number of reads in sample 1</p>
</dd>
<dt>sample_2</dt><dd><p>the number of reads in sample 2</p>
</dd>
<dt>sample_3</dt><dd><p>the number of reads in sample 3</p>
</dd>
<dt>sample_4</dt><dd><p>the number of reads in sample 4</p>
</dd>
<dt>sample_5</dt><dd><p>the number of reads in sample 5</p>
</dd>
</dl>
<p>...

</p>

<hr>
<h2 id='scoring'>scoring</h2><span id='topic+scoring'></span>

<h3>Description</h3>

<p>scoring
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scoring(k, n, N)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scoring_+3A_k">k</code></td>
<td>
<p>number of modified targets on the region</p>
</td></tr>
<tr><td><code id="scoring_+3A_n">n</code></td>
<td>
<p>number of targets on the region</p>
</td></tr>
<tr><td><code id="scoring_+3A_n">N</code></td>
<td>
<p>number of targets in the panel</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the confidence score
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scoring(10,20,150)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
