<!DOCTYPE html><html><head><title>Help for package IFP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IFP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#allele.freq'><p>Allele Frequency Computation from Genotype Data</p></a></li>
<li><a href='#allele.freq.G'><p>Allele Frequency Computation from the sequencing data with a vcf type of the 1000 Genomes Project</p></a></li>
<li><a href='#apoe'><p>Genetic data of APOE gene region</p></a></li>
<li><a href='#apoeG'><p>Sequencing data of APOE gene region from the 1000 Genomes Project</p></a></li>
<li><a href='#drgegggne'><p>causal models with all possible causal factors: G, G*G, G*E and E</p></a></li>
<li><a href='#drgegne'><p>causal models with three possible causal factors: G, G*E and E</p></a></li>
<li><a href='#drgen'><p>causal models with G*E</p></a></li>
<li><a href='#drgene'><p>causal models with G*E and E</p></a></li>
<li><a href='#drggn'><p>causal models with G*G</p></a></li>
<li><a href='#drgn'><p>causal models with G</p></a></li>
<li><a href='#error.rates'><p>Error Rates Estimation for Likelihood Ratio Tests Designed for Identifying Number of Functional Polymorphisms</p></a></li>
<li><a href='#geno.freq'><p>Genotype Frequency Computation from the sequencing data with a vcf type of the 1000 Genomes Project</p></a></li>
<li><a href='#genotype'><p>Conversion to Genotypes from Alleles using the sequencing data with a vcf type of the 1000 Genomes Project</p></a></li>
<li><a href='#hap.freq'><p>Estimation of Haplotype Frequencies with Two SNPs</p></a></li>
<li><a href='#iter.mcmc'><p>mcmc inference of causal models with all possible causal factors: G, G*G, G*E and E</p></a></li>
<li><a href='#lrt'><p>Likelihood Ratio Tests for Identifying Number of Functional Polymorphisms</p></a></li>
<li><a href='#lrtG'><p>Likelihood Ratio Tests for Identifying Disease Polymorphisms with Same Effects</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.2.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-10-22</td>
</tr>
<tr>
<td>Title:</td>
<td>Identifying Functional Polymorphisms</td>
</tr>
<tr>
<td>Author:</td>
<td>Park L</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Leeyoung Park &lt;lypark@yonsei.ac.kr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A suite for identifying causal models using relative concordances and identifying causal polymorphisms in case-control genetic association data, especially with large controls re-sequenced data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.11.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>haplo.stats,coda</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-25 13:35:07 UTC; L</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-26 00:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='allele.freq'>Allele Frequency Computation from Genotype Data</h2><span id='topic+allele.freq'></span>

<h3>Description</h3>

<p>Computes allele frequencies from genotype data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> allele.freq(geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allele.freq_+3A_geno">geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>array of allele frequencies of each SNP. The computed allele is targeted as an order of alleles, &quot;A&quot;, &quot;C&quot;, &quot;G&quot;, and &quot;T&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(apoe)
 allele.freq(apoe7)
 allele.freq(apoe)
</code></pre>

<hr>
<h2 id='allele.freq.G'>Allele Frequency Computation from the sequencing data with a vcf type of the 1000 Genomes Project</h2><span id='topic+allele.freq.G'></span>

<h3>Description</h3>

<p>Computes allele frequencies from the sequencing data with a vcf type of the 1000 Genomes Project. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> allele.freq.G(genoG)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="allele.freq.G_+3A_genog">genoG</code></td>
<td>

<p>matrix of haplotypes. Each row indicates a variant, and each column ind
icates a haplotype of an individual. Two alleles of 0 and 1 are available.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>array of allele frequencies of each variant.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(apoeG)
 allele.freq.G(apoeG)
</code></pre>

<hr>
<h2 id='apoe'>Genetic data of APOE gene region</h2><span id='topic+apoe'></span><span id='topic+apoe7'></span>

<h3>Description</h3>

<p>This data set came from a re-sequenced data of APOE gene region in the Molecular Diversity and Epidemiology of Common Disease (MDECODE) database. Sixteen polymorphic sites were included. &quot;apoe7&quot; data contains the genetic data of seven single nucleotide polymorphisms with allele frequencies higher than 0.1 from the apoe data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(apoe)</code></pre>


<h3>Format</h3>

<p>A matrix with 48 rows and 32 columns</p>


<h3>Source</h3>

<p>http://droog.gs.washington.edu/mdecode/</p>


<h3>References</h3>

<p>Nickerson, D. A., S. L. Taylor, S. M. Fullerton, K. M. Weiss, A. G. Clark et al. (2000) Sequence diversity and large-scale typing of SNPs in the human apolipoprotein E gene. <em>Genome Res</em> 10: 1532-1545.
</p>

<hr>
<h2 id='apoeG'>Sequencing data of APOE gene region from the 1000 Genomes Project</h2><span id='topic+apoeG'></span><span id='topic+genoT'></span><span id='topic+genoC'></span>

<h3>Description</h3>

<p>This data set came from a re-sequenced data of APOE gene region from the 1000 Genomes Project. Thirty three polymorphic sites with allele frequencies higher than 0.001 were included for the original data set, apoeG. The test data sets, apoeT and apoeC, indicate the data of 100 controls and 100 cases respectively when the dominant variant is 15th variant with the odds ratio of 3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(apoeG)</code></pre>


<h3>Format</h3>

<p>A matrix with 33 rows and 2184 columns</p>


<h3>Source</h3>

<p>ftp://ftp-trace.ncbi.nih.gov/1000genomes/ftp/release/20110521/</p>


<h3>References</h3>

<p>Abecasis, G. R. et al. (2010) A map of human genome variation from population-scale sequencing. <em>Nature</em> 467, 1061-1073.
</p>

<hr>
<h2 id='drgegggne'>causal models with all possible causal factors: G, G*G, G*E and E</h2><span id='topic+drgegggne'></span>

<h3>Description</h3>

<p>provides concordance probabilities of relative pairs for a causal model with G, G*G, G*E and E components
</p>


<h3>Usage</h3>

<pre><code class='language-R'> drgegggne(fdg,frg,fdgg,frgg,fdge,frge,eg,e)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drgegggne_+3A_fdg">fdg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G component</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_frg">frg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G component</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_fdgg">fdgg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G*G component</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_frgg">frgg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G*G component</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_fdge">fdge</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G*E component</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_frge">frge</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G*E component</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_eg">eg</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause of G*E interactiong the genetic cause of G*E during their entire life</p>
</td></tr>
<tr><td><code id="drgegggne_+3A_e">e</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause during their entire life</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of NN, ND, and DD probabilities of 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>


<h3>See Also</h3>

<p>drggn
drgegne
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01



### for a model without one or more missing causal factors, 
### set the relevant parameters as zero.

pg&lt;-0.002  # the proportion of G component in total populations
pgg&lt;-0.002  # the proportion of G*G component in total populations
pge&lt;-0.003  # the proportion of G*E component in total populations
e&lt;-1-(1-ppt)/(1-pg)/(1-pgg)/(1-pge)   
   # the proportion of E component in total populations

fd&lt;-0.001  # one dominant gene
tt&lt;-3      # the number of recessive genes

temp&lt;-sqrt(1-((1-pg)/(1-fd)^2)^(1/tt))
fr&lt;-c(array(0,length(fd)),array(temp,tt))
fd&lt;-c(fd,array(0,tt))

ppd&lt;-sqrt(pgg)
fdg&lt;-array(1-sqrt(1-ppd^(1/2)),2)
ttg&lt;-1
temp&lt;-(pgg/ppd)^(1/2/ttg)
frg&lt;-c(array(0,length(fdg)),array(temp,ttg))
fdg&lt;-c(fdg,array(0,ttg))

ppe&lt;-0.5
ppg&lt;-pge/ppe

fdge&lt;-0.002
ttge&lt;-2      # the number of recessive genes

temp&lt;-sqrt(1-((1-ppg)/(1-fdge)^2)^(1/ttge))
frge&lt;-c(array(0,length(fdge)),array(temp,ttge))
fdge&lt;-c(fdge,array(0,ttge))


drgegggne(fd,fr,fdg,frg,fdge,frge,ppe,e)


</code></pre>

<hr>
<h2 id='drgegne'>causal models with three possible causal factors: G, G*E and E</h2><span id='topic+drgegne'></span>

<h3>Description</h3>

<p>provides concordance probabilities of relative pairs for a causal model with G, G*E and E components
</p>


<h3>Usage</h3>

<pre><code class='language-R'> drgegne(fdg,frg,fdge,frge,eg,e)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drgegne_+3A_fdg">fdg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G component</p>
</td></tr>
<tr><td><code id="drgegne_+3A_frg">frg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G component</p>
</td></tr>
<tr><td><code id="drgegne_+3A_fdge">fdge</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G*E component</p>
</td></tr>
<tr><td><code id="drgegne_+3A_frge">frge</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G*E component</p>
</td></tr>
<tr><td><code id="drgegne_+3A_eg">eg</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause of G*E interactiong the genetic cause of G*E during their entire life</p>
</td></tr>
<tr><td><code id="drgegne_+3A_e">e</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause during their entire life</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of NN, ND, and DD probabilities of 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>


<h3>See Also</h3>

<p>drgn
drgene
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01



pg&lt;-0.002  # the proportion of G component in total populations
pge&lt;-0.005  # the proportion of G*E component in total populations
e&lt;-1-(1-ppt)/(1-pg)/(1-pge)   
  # the proportion of E component in total populations

fd&lt;-0.001  # one dominant gene
tt&lt;-2      # the number of recessive genes

temp&lt;-sqrt(1-((1-pg)/(1-fd)^2)^(1/tt))
fr&lt;-c(array(0,length(fd)),array(temp,tt))
fd&lt;-c(fd,array(0,tt))

ppe&lt;-0.5
ppg&lt;-pge/ppe

fdge&lt;-0.002
ttge&lt;-2      # the number of recessive genes

temp&lt;-sqrt(1-((1-ppg)/(1-fdge)^2)^(1/ttge))
frge&lt;-c(array(0,length(fdge)),array(temp,ttge))
fdge&lt;-c(fdge,array(0,ttge))


drgegne(fd,fr,fdge,frge,ppe,e)


</code></pre>

<hr>
<h2 id='drgen'>causal models with G*E</h2><span id='topic+drgen'></span>

<h3>Description</h3>

<p>provides concordance probabilities of relative pairs for a causal model with G*E component
</p>


<h3>Usage</h3>

<pre><code class='language-R'> drgen(fd,fr,e)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drgen_+3A_fd">fd</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G component of G*E interacting with E of G*E</p>
</td></tr>
<tr><td><code id="drgen_+3A_fr">fr</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G component of G*E interacting with E of G*E</p>
</td></tr>
<tr><td><code id="drgen_+3A_e">e</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause of G*E interacting with genetic cause of G*E during their entire life</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of the g*e proportion in population and a matrix of NN, ND, and DD probabilities of 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>


<h3>See Also</h3>

<p>drgene.gm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01



### g*e model

pge&lt;-ppt  # the proportion of G*E component in total populations

ppe&lt;-0.5
ppg&lt;-pge/ppe

fd&lt;-0.0005  # one dominant gene
tt&lt;-3      # the number of recessive genes

temp&lt;-sqrt(1-((1-ppg)/(1-fd)^2)^(1/tt))
fr&lt;-c(array(0,length(fd)),array(temp,tt))
fd&lt;-c(fd,array(0,tt))

drgen(fd,fr,ppe)


</code></pre>

<hr>
<h2 id='drgene'>causal models with G*E and E</h2><span id='topic+drgene'></span>

<h3>Description</h3>

<p>provides concordance probabilities of relative pairs for a causal model with G*E and E components
</p>


<h3>Usage</h3>

<pre><code class='language-R'> drgene(fdg,frg,eg,e)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drgene_+3A_fdg">fdg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G component of G*E interacting with E of G*E</p>
</td></tr>
<tr><td><code id="drgene_+3A_frg">frg</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G component of G*E interacting with E of G*E</p>
</td></tr>
<tr><td><code id="drgene_+3A_eg">eg</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause of G*E interacting with genetic cause of G*E during their entire life</p>
</td></tr>
<tr><td><code id="drgene_+3A_e">e</code></td>
<td>
<p>a proportion of population who are exposed to environmental cause during their entire life</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of NN, ND, and DD probabilities of 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>


<h3>See Also</h3>

<p>drgen.gm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01



### g*e+e model

pge&lt;-0.007  # the proportion of G*E component in total populations
e&lt;-1-(1-ppt)/(1-pge)   # the proportion of E component in total populations

ppe&lt;-0.5
ppg&lt;-pge/ppe

fd&lt;-0.0005  # one dominant gene
tt&lt;-3      # the number of recessive genes

temp&lt;-sqrt(1-((1-ppg)/(1-fd)^2)^(1/tt))
fr&lt;-c(array(0,length(fd)),array(temp,tt))
fd&lt;-c(fd,array(0,tt))

drgene(fd,fr,ppe,e)


</code></pre>

<hr>
<h2 id='drggn'>causal models with G*G</h2><span id='topic+drggn'></span>

<h3>Description</h3>

<p>provides concordance probabilities of relative pairs for a causal model with G*G component
</p>


<h3>Usage</h3>

<pre><code class='language-R'> drggn(fd,fr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drggn_+3A_fd">fd</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G*G component</p>
</td></tr>
<tr><td><code id="drggn_+3A_fr">fr</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G*G component</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of PLI and a matrix of NN, ND, and DD probabilities of 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>


<h3>See Also</h3>

<p>drgegggne
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01



### g*g model

pp&lt;-ppt  # the proportion of G*G component in total populations

gd&lt;-sqrt(pp) # dominant gene proportion = recessive gene proportion
fd&lt;-array(1-sqrt(1-gd^(1/2)),2)  # two dominant genes
tt&lt;-2      # the number of recessive genes: 2

temp&lt;-(pp/gd)^(1/2/tt)
fr&lt;-c(array(0,length(fd)),array(temp,tt))
fd&lt;-c(fd,array(0,tt))

drggn(fd,fr)


</code></pre>

<hr>
<h2 id='drgn'>causal models with G</h2><span id='topic+drgn'></span>

<h3>Description</h3>

<p>provides concordance probabilities of relative pairs for a causal model with G component
</p>


<h3>Usage</h3>

<pre><code class='language-R'> drgn(fd,fr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drgn_+3A_fd">fd</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of dominant gene frequencies including 0 values of recessive genes of G component</p>
</td></tr>
<tr><td><code id="drgn_+3A_fr">fr</code></td>
<td>
<p>an array (size=number of dominant genes+recessive genes) of recessive gene frequencies including 0 values of dominant genes of G component</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of the value of PLI and the matrix of NN, ND, and DD probabilities of 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>


<h3>See Also</h3>

<p>drgegne.gm
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01



### g model

pp&lt;-ppt  # the proportion of G component in total populations

fdt&lt;-0.001 # one dominant gene with frequency of 0.001
tt&lt;-5      # the number of recessive genes: 5

fd&lt;-c(fdt,array(0,tt))
temp&lt;-sqrt(1-((1-pp)/(1-fdt)^2)^(1/tt))
fr&lt;-c(0,array(temp,tt))

drgn(fd,fr)


</code></pre>

<hr>
<h2 id='error.rates'>Error Rates Estimation for Likelihood Ratio Tests Designed for Identifying Number of Functional Polymorphisms</h2><span id='topic+error.rates'></span>

<h3>Description</h3>

<p>Compute error rates for a given model. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> error.rates(H0,Z, pMc, geno, no.ca, no.con=nrow(geno), sim.no = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="error.rates_+3A_h0">H0</code></td>
<td>
<p>the index number for a given model for functional SNPs</p>
</td></tr>
<tr><td><code id="error.rates_+3A_z">Z</code></td>
<td>
<p>number of functional SNPs for the given model</p>
</td></tr>
<tr><td><code id="error.rates_+3A_pmc">pMc</code></td>
<td>
<p>array of allele frequencies of case samples</p>
</td></tr>
<tr><td><code id="error.rates_+3A_geno">geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td></tr>
<tr><td><code id="error.rates_+3A_no.ca">no.ca</code></td>
<td>
<p>number of case chromosomes</p>
</td></tr>
<tr><td><code id="error.rates_+3A_no.con">no.con</code></td>
<td>
<p>number of control chromosomes</p>
</td></tr>
<tr><td><code id="error.rates_+3A_sim.no">sim.no</code></td>
<td>
<p>number of simulations for error rates estimation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>array of results consisted of Type I error rate (alpha=0.05), Type I error rate (alpha=0.01), Type II error rate (beta=0.05), Type II error rate (beta=0.01), percent when the target model has the lowest corrected -2 log likelihood ratio. 
</p>


<h3>See Also</h3>

<p>allele.freq
hap.freq
lrtB
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## LRT tests when SNP1 &amp; SNP6 are the functional polymorphisms.


data(apoe)

n&lt;-c(2000, 2000, 2000, 2000, 2000, 2000, 2000) #case sample size = 1000
x&lt;-c(1707, 281,1341, 435, 772, 416, 1797) #allele numbers in case samples 

Z&lt;-2 	#number of functional SNPs for tests
n.poly&lt;-ncol(apoe7)/2 	#total number of SNPs

#index number for the model in this case is 5 for SNP1 and 6. 
#apoe7 is considered to represent the true control allele and haplotype frequencies.
#Control sample size = 1000.

error.rates(5, 2, x/n, apoe7, 2000, 2000, sim.no=2)

# to obtain valid rates, use sim.no=1000.

</code></pre>

<hr>
<h2 id='geno.freq'>Genotype Frequency Computation from the sequencing data with a vcf type of the 1000 Genomes Project</h2><span id='topic+geno.freq'></span>

<h3>Description</h3>

<p>Computes genotype frequencies from the sequencing data with a vcf type of the 1000 Genomes Project. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> geno.freq(genoG)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geno.freq_+3A_genog">genoG</code></td>
<td>

<p>matrix of haplotypes. Each row indicates a variant, and each column ind
icates a haplotype of an individual. Two alleles of 0 and 1 are available.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of genotype frequencies of each variant.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(apoeG)
 geno.freq(apoeG)
</code></pre>

<hr>
<h2 id='genotype'>Conversion to Genotypes from Alleles using the sequencing data with a vcf type of the 1000 Genomes Project</h2><span id='topic+genotype'></span>

<h3>Description</h3>

<p>Convert sequencing data to genotypes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> genotype(genoG)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotype_+3A_genog">genoG</code></td>
<td>

<p>matrix of haplotypes. Each row indicates a variant, and each column ind
icates a haplotype of an individual. Two alleles of 0 and 1 are available.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of genotypes with rows of variants and with columns of individuals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(apoeG)
 genotype(apoeG)
</code></pre>

<hr>
<h2 id='hap.freq'>Estimation of Haplotype Frequencies with Two SNPs</h2><span id='topic+hap.freq'></span>

<h3>Description</h3>

<p>EM computation of haplotype frequencies with two SNPs. The computation is relied on the package&quot;haplo.stats&quot;. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> hap.freq(geno)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hap.freq_+3A_geno">geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of haplotype frequencies consisted of two alleles from each SNP. These alleles are the same ones computed for frequency using the function &quot;allele.freq&quot;.
</p>


<h3>See Also</h3>

<p>allele.freq
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(apoe)
 hap.freq(apoe7)
 hap.freq(apoe)
</code></pre>

<hr>
<h2 id='iter.mcmc'>mcmc inference of causal models with all possible causal factors: G, G*G, G*E and E</h2><span id='topic+iter.mcmc'></span>

<h3>Description</h3>

<p>provides proportions of each causal factor of G, G*G, G*E and E based on relative concordance data
</p>


<h3>Usage</h3>

<pre><code class='language-R'> iter.mcmc(ppt,aj=2,n.iter,n.chains,thinning=5,init.cut,darray,x,n,model,mcmcrg=0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iter.mcmc_+3A_ppt">ppt</code></td>
<td>
<p>population lifetime incidence</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_aj">aj</code></td>
<td>
<p>a constant for the stage of data collection</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_n.iter">n.iter</code></td>
<td>
<p>number of mcmc iterations</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_n.chains">n.chains</code></td>
<td>
<p>number of mcmc chain</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_thinning">thinning</code></td>
<td>
<p>mcmc thinning parameter (default=5)</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_init.cut">init.cut</code></td>
<td>
<p>mcmc data cut</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_darray">darray</code></td>
<td>
<p>indicating the array positions of available data among 9 relative pairs: 1:mzt,2:parent-offspring,3:dzt,4:sibling,5:2-direct(grandparent-grandchild),6:3rd(uncle-niece),7:3-direct(great-grandparent-great-grandchild),8:4th (causin),9:4d(great-great-grandparent-great-great-grandchild)
</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_x">x</code></td>
<td>
<p>number of disease concordance of relative pairs</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_n">n</code></td>
<td>
<p>total number of relative pairs</p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_model">model</code></td>
<td>
<p>an array, size of 4 (1: E component; 2: G component; 3: G*E component; 4: G*G component), indicating the existance of the causal component: 0: excluded; 1: included. </p>
</td></tr>
<tr><td><code id="iter.mcmc_+3A_mcmcrg">mcmcrg</code></td>
<td>
<p> parameter of the data collection stage (default=0.01)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of rejectionRate, result summary, Gelman-Rubin diagnostics (point est. &amp; upper C.I.) for output variables:
e[1]: proportion of environmental factor (E)
g[2]: proportion of genetic factor (G)
ge[3]: proportion of gene-environment interaction (G*E)
gg[4]: proportion of gene interactions (G*G)
gn[5]: number of recessive genes in G
ppe[6]: population proportion of interacting environment in G*E
ppg[7]: population proportion of interacting genetic factor in G*E
fd[8]: frequency of dominant genes in G
fdge[9]: frequency of dominant genes in G*E
gnge[10]: number of recessive genes in G*E
ppd[11]: population proportion of dominant genes in G*G
ppr[12]: population proportion of recessive genes in G*G
kd[13]: number of dominant genes in G*G
kr[14]: number of recessive genes in G*G
</p>


<h3>References</h3>

<p>L. Park, J. Kim, A novel approach for identifying causal models of complex disease from family data, Genetics, 2015 Apr; 199, 1007-1016.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### PLI=0.01.
ppt&lt;-0.01

### a simple causal model with G and E components

pg&lt;-0.007  # the proportion of G component in total populations
pgg&lt;-0  # the proportion of G*G component in total populations
pge&lt;-0  # the proportion of G*E component in total populations
e&lt;-1-(1-ppt)/(1-pg)   # the proportion of E component in total populations

fd&lt;-0.001  # one dominant gene
tt&lt;-3      # the number of recessive genes

temp&lt;-sqrt(1-((1-pg)/(1-fd)^2)^(1/tt))
fr&lt;-c(array(0,length(fd)),array(temp,tt))
fd&lt;-c(fd,array(0,tt))

rp&lt;-drgegggne(fd,fr,c(0,0),c(0,0),c(0,0),c(0,0),0,e)

sdata&lt;-rp[,3]/(rp[,2]+rp[,3])
#sdata&lt;-round(sdata*500)

darray&lt;-c(1:2,4:6)  
  ## available data= MZT, P-O, sibs, grandparent-grandchild, avuncular pair
n&lt;-array(1000,length(darray))
x&lt;-array()
for(i in 1:length(darray)){
x[i]&lt;-rbinom(1,n[i],sdata[darray[i]])
}
model&lt;-c(1,1,0,0)

## remove # from the following lines to test examples.
#iter.mcmc(ppt,2,15,2,1,1,darray,x,n,model) # provide a running test
#iter.mcmc(ppt,2,2000,2,10,500,darray,x,n,model) # provide a proper result

</code></pre>

<hr>
<h2 id='lrt'>Likelihood Ratio Tests for Identifying Number of Functional Polymorphisms</h2><span id='topic+lrt'></span>

<h3>Description</h3>

<p>Compute p-values and likelihoods of all possible models for a given number of functional SNP(s). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> lrt(n.fp, n, x, geno, no.con=nrow(geno))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lrt_+3A_n.fp">n.fp</code></td>
<td>
<p>number of functional SNPs for tests.</p>
</td></tr>
<tr><td><code id="lrt_+3A_n">n</code></td>
<td>
<p>array of each total number of case sample chromosomes for SNPs</p>
</td></tr>
<tr><td><code id="lrt_+3A_x">x</code></td>
<td>
<p>array of each total allele number in case samples</p>
</td></tr>
<tr><td><code id="lrt_+3A_geno">geno</code></td>
<td>

<p>matrix of alleles, such that each locus has a pair of adjacent columns of alleles, and the order of columns corresponds to the order of loci on a chromosome.  If there are K loci, then ncol(geno) = 2*K. Rows represent the alleles for each subject. Each allele shoud be represented as numbers (A=1,C=2,G=3,T=4).
</p>
</td></tr>
<tr><td><code id="lrt_+3A_no.con">no.con</code></td>
<td>
<p>number of control chromosomes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of likelihood ratio test results. First n.fp rows indicate the model for each set of disease polymorphisms, and followed by p-values, -2 log(likelihood ratio) with corrections for variances, maximum likelihood ratio estimates, and likelihood.
</p>


<h3>References</h3>

<p>L. Park, Identifying disease polymorphisms from case-control genetic association data, Genetica, 2010 138 (11-12), 1147-1159.
</p>


<h3>See Also</h3>

<p>allele.freq
hap.freq
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## LRT tests when SNP1 &amp; SNP6 are the functional polymorphisms.

data(apoe)

n&lt;-c(2000, 2000, 2000, 2000, 2000, 2000, 2000) #case sample size = 1000
x&lt;-c(1707, 281,1341, 435, 772, 416, 1797) #allele numbers in case samples 


Z&lt;-2 	#number of functional SNPs for tests
n.poly&lt;-ncol(apoe7)/2 	#total number of SNPs

#control sample generation( sample size = 1000 )
con.samp&lt;-sample(nrow(apoe7),1000,replace=TRUE)
con.data&lt;-array()
for (i in con.samp){
con.data&lt;-rbind(con.data,apoe7[i,])
}
con.data&lt;-con.data[2:1001,]

lrt(1,n,x,con.data)
lrt(2,n,x,con.data)
</code></pre>

<hr>
<h2 id='lrtG'>Likelihood Ratio Tests for Identifying Disease Polymorphisms with Same Effects</h2><span id='topic+lrtG'></span>

<h3>Description</h3>

<p>Compute p-values and likelihoods of all possible models for a given number of disease SNP(s). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'> lrtG(n.fp, genoT, genoC)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lrtG_+3A_n.fp">n.fp</code></td>
<td>
<p>number of disease SNPs for tests.</p>
</td></tr>
<tr><td><code id="lrtG_+3A_genot">genoT</code></td>
<td>

<p>matrix of control genotypes. Each row indicates a variant, and each column indicates a haplotype of an individual. Two alleles of 0 and 1 are allowed.
</p>
</td></tr>
<tr><td><code id="lrtG_+3A_genoc">genoC</code></td>
<td>

<p>matrix of case genotypes. Each row indicates a variant, and each column indicates a haplotype of an individual. Two alleles of 0 and 1 are allowed.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix of likelihood ratio test results. First row indicates the index, and following n.fp rows indicate the model for each set of disease polymorphisms, and followed by p-values, -2 log(likelihood ratio) with corrections for variances, and the degree of freedom.
</p>


<h3>References</h3>

<p>L. Park, J. Kim, Rare high-impact disease variants: properties and identification, Genetics Research, 2016 Mar; 98, e6.
</p>


<h3>See Also</h3>

<p>allele.freq.G
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## LRT tests for a dominant variant (15th variant)
## the odds ratio: 3, control: 100, case: 100.

data(apoeG)
lrtG(1,genoT[,1:20],genoC[,1:20])

# use "lrtG(1,genoT,genoC)" for the actual test.

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
