<!DOCTYPE html><html><head><title>Help for package iGasso</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iGasso}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#iGasso-package'>
<p>Statistical Tests and utilities for Genetic Association</p></a></li>
<li><a href='#genome.plot'><p>Genome-wide Plot of a Variable</p></a></li>
<li><a href='#h2_snp'><p>An exact method for SNP-heritability estimation using GWAS summary statistics</p></a></li>
<li><a href='#KAT.coin'><p>Conditional Inference for the Kernel Association Test (KAT)</p></a></li>
<li><a href='#MFree.test'><p>Model-free Association Tests</p></a></li>
<li><a href='#MR_het_test'><p>Test of Heterogeneity in MR using Principal Components</p></a></li>
<li><a href='#SKATplus'><p>A Gene- or Pathway-Based Test of Association</p>
</p>
<p><code>SKATlus</code> provides enhanced power over SKAT by properly estimating the null distribution of SKAT.</p></a></li>
<li><a href='#SMR_interval'><p>Interval Estimates for Summary Data Mendelian Randomization Analysis in the Presence of Winner's Curse</p></a></li>
<li><a href='#tarq'><p>An Accurate Normalization Method for High Throughput Sequencing Data</p></a></li>
<li><a href='#VSTF.test'><p>Association Tests for Rare Variants Based on Variance-Stabilizing Transformation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Tests and Utilities for Genetic Association</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-3-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Dr. Kai Wang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kai Wang &lt;kai-wang@uiowa.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>lattice, CompQuadForm, MASS, MBESS</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of statistical tests for genetic association studies and summary data based Mendelian randomization.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-11 19:03:55 UTC; kaiwang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-12 00:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='iGasso-package'>
Statistical Tests and utilities for Genetic Association
</h2><span id='topic+iGasso-package'></span><span id='topic+iGasso'></span>

<h3>Description</h3>

<p><code>iGasso</code> is a collection of statistical tests developed by our group for genetic association studies. So far it contains functions for rare variants association, for association with multiple phenotypes, for linear mixed model analysis, and for model-free association analysis. There is also a function for genome plot. It will keep growing as more tests are developed. Use <code>?iGasso</code> to see an introduction.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> iGasso</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.6</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-09-16</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Anscombe F.J. (1948) The transformation of Poisson, binomial and negative-binomial data. <em>Biometrika</em> <b>35(3/4)</b>, 246&ndash;254.
</p>
<p>Chanter, D. O. (1975). Modifications of the angular transformation. <em>Journal of the Royal Statistical Society. Series B (Applied Statistics)</em>, <b>24 (3)</b>, 354&ndash;359.
</p>
<p>Freeman, M. F., Tukey, J. W. (1950) Transformations related to the angular and the square root. <em>The Annals of Mathematical Statistics</em> <b>21(4)</b>, 607&ndash;611.
</p>
<p>Wang, K. (2012) An application of the proportional odds model to genetic association studies. Submitted.
</p>
<p>Wang K. (2012) Statistical tests of genetic association for case-control study designs. <em>Biostatistics</em>. Accepted. PMID: 22389176
</p>
<p>Wang, K., Fingert, J. (2012) Statistical tests for detecting rare variants using variance-stabilizing transformations. <em>Annals of Human Genetics</em>. Accepted. 
</p>
<p>Zar, J. H. (1999) <em>Biostatistical Analysis, 4th ed.</em>, New Jersey:Prentice-Hall, Inc.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>y = rnorm(100)
chr = c(rep(1, 20), rep(3, 20), rep(10, 20), rep(19, 30), rep("X", 10))
pos = c(1:20, 1:20, 1:20, 1:30, 1:10)
mydata = data.frame(y=y, chr=chr, pos=pos)
genome.plot(mydata, sig.line=c(1, -1), ylab="T Statistic")

G = rbind(c(14, 999), c(3, 1081))
VSTF.test(G)

G = rbind(c(161, 474, 489), c(231, 444, 380))
MFree.test(G)

G = matrix(sample(c(0,1,2), 200, replace=TRUE), ncol=10)
y = rnorm(10)
X = matrix(rnorm(10), ncol=1)
</code></pre>

<hr>
<h2 id='genome.plot'>Genome-wide Plot of a Variable</h2><span id='topic+genome.plot'></span>

<h3>Description</h3>

<p><code>genome.plot</code> plots the value of a variable across the genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genome.plot(
  mydata,
  style = 1,
  type = "h",
  sig.line = c(4, -4),
  sig.color = c("red", "red"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genome.plot_+3A_mydata">mydata</code></td>
<td>
<p>a data frame containing three variables: <code>y</code> (numeric, the value of the variable to be plotted), <code>chr</code> (character, chromosome label), and <code>pos</code> (numeric, position, for instance, in base pair or centi-Morgan). Examples of <code>y</code> include -log10 of p-values and test statistic values.</p>
</td></tr>
<tr><td><code id="genome.plot_+3A_style">style</code></td>
<td>
<p>either <code>1</code> (default) or <code>2</code>.</p>
</td></tr>
<tr><td><code id="genome.plot_+3A_type">type</code></td>
<td>
<p>a generic graphic parameter. Recommended values are <code>"h"</code> (default) and <code>"b"</code>.</p>
</td></tr>
<tr><td><code id="genome.plot_+3A_sig.line">sig.line</code></td>
<td>
<p>vertical locations of significance lines.</p>
</td></tr>
<tr><td><code id="genome.plot_+3A_sig.color">sig.color</code></td>
<td>
<p>colors of significance lines.</p>
</td></tr>
<tr><td><code id="genome.plot_+3A_...">...</code></td>
<td>
<p>other parameters to be passed to function <code>xyplot</code> in the <code>lattice</code> package.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function makes use of the function <code>xyplot</code> from package <code>lattice</code>.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  y = rnorm(100)
  chr = c(rep(1, 20), rep(3, 20), rep(10, 20), rep(19, 30), rep("X", 10))
  pos = c(1:20, 1:20, 1:20, 1:30, 1:10)
  mydata = data.frame(y=y, chr=chr, pos=pos)
  mydata2 = data.frame(y=y^2, chr=chr, pos=pos)
  
  genome.plot(mydata, sig.line=c(1, -1), ylab="T Statistic")
  genome.plot(mydata, sig.line=c(1, -1), ylab="T Statistic", type="b")
  genome.plot(mydata2, sig.line=c(2), ylab="y squared")
  genome.plot(mydata, style=2, sig.line=c(1, -1), ylab="T Statistic")
  genome.plot(mydata, style=2, sig.line=c(1, -1), ylab="T Statistic", type="b")

</code></pre>

<hr>
<h2 id='h2_snp'>An exact method for SNP-heritability estimation using GWAS summary statistics</h2><span id='topic+h2_snp'></span>

<h3>Description</h3>

<p><code>h2_snp</code> calculates heritability explained by a set of SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>h2_snp(beta, SE, N, R, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="h2_snp_+3A_beta">beta</code></td>
<td>
<p>a vector of beta coefficients for a set of SNPs. These coefficients are from a GWAS.</p>
</td></tr>
<tr><td><code id="h2_snp_+3A_se">SE</code></td>
<td>
<p>a vector of the standard errors of the beta coefficients.</p>
</td></tr>
<tr><td><code id="h2_snp_+3A_n">N</code></td>
<td>
<p>a vector of sample sizes used by the GWAS at these SNPs.</p>
</td></tr>
<tr><td><code id="h2_snp_+3A_r">R</code></td>
<td>
<p>LD matrix for these SNPs.</p>
</td></tr>
<tr><td><code id="h2_snp_+3A_alpha">alpha</code></td>
<td>
<p><code class="reqn">1-\alpha</code> is the confidence level of the confidence interval.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<p>* MLE of the heritability.
</p>
<p>* umvu (uniformly minimum variance unbiased) estimator of the heritability.
</p>
<p>* interval estimate for the heritability.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K. (2023) An exact method for SNP-heritability estimation using GWAS summary statistics without heritability modeling. <em>submitted</em>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>beta = c(0.225269, 0.221270, 0.162635, 0.261669, 0.150887, 
         0.214515, 0.170296, 0.204454, 0.254811, 0.213803)
SE = c(0.033244, 0.032551, 0.032171, 0.031042, 0.032815, 
       0.031908, 0.031717, 0.032023, 0.031907, 0.032291)
N = 10000
R = diag(1, 10)
alpha = 0.05
h2_snp(beta, SE, N, R, alpha)
</code></pre>

<hr>
<h2 id='KAT.coin'>Conditional Inference for the Kernel Association Test (KAT)</h2><span id='topic+KAT.coin'></span>

<h3>Description</h3>

<p>Computes the asymptotic and the approximate conditional p-values for the kernel association test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KAT.coin(y, G, X = NULL, out_type = "D", distribution = "asymptotic", B = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KAT.coin_+3A_y">y</code></td>
<td>
<p>a vector of phenotype on <code class="reqn">n</code> subjects.</p>
</td></tr>
<tr><td><code id="KAT.coin_+3A_g">G</code></td>
<td>
<p>an <code class="reqn">n \times m</code> matrix of SNP genotypes of <code class="reqn">n</code> study subjects at <code class="reqn">m</code> loci.</p>
</td></tr>
<tr><td><code id="KAT.coin_+3A_x">X</code></td>
<td>
<p>a matrix of covariates. It has <code class="reqn">n</code> rows.</p>
</td></tr>
<tr><td><code id="KAT.coin_+3A_out_type">out_type</code></td>
<td>
<p>an indicator of the outcome type. <code>"C"</code> for the continuous outcome and <code>"D"</code> for the dichotomous outcome.</p>
</td></tr>
<tr><td><code id="KAT.coin_+3A_distribution">distribution</code></td>
<td>
<p>a character, the conditional null distribution of the test statistic can be approximated by its asymptotic distribution (&quot;asymptotic&quot;, default) or via Monte Carlo resampling (&quot;approximate&quot;), as in package <code>coin</code>.</p>
</td></tr>
<tr><td><code id="KAT.coin_+3A_b">B</code></td>
<td>
<p>the number of permutations if <code>distribution = "approximate"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The asymptotic conditional null distribution is obtained using results in Strasser and Weber (1999). 
The p-value based on this distribution is computed using Davies' method.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
</p>
<p>* statistic the value of the kernel association test statistic.
</p>
<p>* parameter sample size and the number of SNPs.
</p>
<p>* p.value the p-value based on the asymptotic or the approximate conditional null distribution.
</p>
<p>* method a character string indicting the test performed.
</p>
<p>* data.name a character string giving the name of the data.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Strasser, H. and Weber, C. (1999) On the asymptotic theory of permutation statistics. <em>Mathematical Methods of Statistics</em>. 8(2):220-250.
</p>
<p>Wang, K. (2017) Conditional Inference for the Kernel Association Test. Bioinformatics 33 (23), 3733-3739.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>n=1000
y = c(rep(1, n/2), rep(0, n/2))
maf = seq(0.05, 0.5, 0.05)
g = NULL
for (j in 1:10){
   geno.freq = c(maf[j]^2, 2*maf[j]*(1-maf[j]), (1-maf[j])^2)
   g = cbind(g, sample(c(0,1,2), n, replace=TRUE, prob=geno.freq))
   }
KAT.coin(y, g, X=NULL, out_type="D", B=1000)

</code></pre>

<hr>
<h2 id='MFree.test'>Model-free Association Tests</h2><span id='topic+MFree.test'></span>

<h3>Description</h3>

<p><code>MFree.test</code> performs tests on association between an SNP and case-control status. It tests whether the frequencies of an allele are the same between cases and controls. It does not require specification of an inheritance model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MFree.test(G, method = "score")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MFree.test_+3A_g">G</code></td>
<td>
<p>a <code>2x3</code> two-dimensional contingency table in matrix form. The first row is for cases and the second one for controls. In each row, the entries are the number of subjects carrying 0, 1, and 2 copies of the reference allele, respective.</p>
</td></tr>
<tr><td><code id="MFree.test_+3A_method">method</code></td>
<td>
<p>a character string indicating the test statistic to use. One of <code>"score"</code> (default), <code>"Wald"</code>, and <code>"LRT"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each test is named after the author(s) of the corresponding publication.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>test</code>&quot; containing the following components:
* statistic the value of the test statistic.
* p.value the p-value for the test computed from a chi-square distribution with 1 df.
* method a character string indicting the test performed.
* data.name a character string giving the name of the data.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang K. (2012) Statistical tests of genetic association for case-control study designs. <em>Biostatistics</em>. 13(4):724-33. PMID: 22389176
</p>


<h3>Examples</h3>

<pre><code class='language-R'>G = rbind(c(161, 474, 489), c(231, 444, 380))
MFree.test(G)
MFree.test(G, method = "Wald")
MFree.test(G, method = "LRT")

</code></pre>

<hr>
<h2 id='MR_het_test'>Test of Heterogeneity in MR using Principal Components</h2><span id='topic+MR_het_test'></span>

<h3>Description</h3>

<p><code>MR_het_test</code> performs tests of heterogeneity in MR. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MR_het_test(x.b, y.b, x.se, y.se, b0, k = NULL, cum.prop = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MR_het_test_+3A_x.b">x.b</code></td>
<td>
<p>a vector of the estimated regression coefficients from the SNP-exposure GWAS</p>
</td></tr>
<tr><td><code id="MR_het_test_+3A_y.b">y.b</code></td>
<td>
<p>a vector of the estimated regression coefficients from the SNP-outcome GWAS</p>
</td></tr>
<tr><td><code id="MR_het_test_+3A_x.se">x.se</code></td>
<td>
<p>a vector of SEs for <code>x.b</code></p>
</td></tr>
<tr><td><code id="MR_het_test_+3A_y.se">y.se</code></td>
<td>
<p>a vector of SEs for <code>y.b</code></p>
</td></tr>
<tr><td><code id="MR_het_test_+3A_b0">b0</code></td>
<td>
<p>a value used for the common effect size. It is used for the weighting matrix</p>
</td></tr>
<tr><td><code id="MR_het_test_+3A_k">k</code></td>
<td>
<p>the number of principal components used. It is used by the <code class="reqn">\tilde Q(b_0)</code> statistic. The default is NULL</p>
</td></tr>
<tr><td><code id="MR_het_test_+3A_cum.prop">cum.prop</code></td>
<td>
<p>threshold for selecting <code>k</code>. It is void if <code>k</code> is specified. The default is 0.8, i.e., the proportion of variance explained by the top <code>k</code> principal components is 0.8</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>
<p>* <code class="reqn">P_\text{min}(b_0)</code> statistic and its <code class="reqn">P</code>-value.
</p>
<p>* <code class="reqn">\tilde Q_\text{min}(b_0)</code> statistic, its degrees of freedom, and its <code class="reqn">P</code>-value.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K, Alberding, Steven Y. (2024) Powerful test of heterogeneity in two-sample summary-data Mendelian randomization. Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = 10
b = 0.5
gamma.true = runif(p, 0.34, 1.1)
x.se = runif(p, 0.06, 0.1)
y.se = runif(p, 0.015, 0.11)
x.b = rnorm(p, gamma.true, x.se)
y.b = rnorm(p, b*gamma.true, y.se)
b0 = 0.4

MR_het_test(x.b, y.b, x.se, y.se, b0)

</code></pre>

<hr>
<h2 id='SKATplus'>A Gene- or Pathway-Based Test of Association
<code>SKATlus</code> provides enhanced power over SKAT by properly estimating the null distribution of SKAT.</h2><span id='topic+SKATplus'></span>

<h3>Description</h3>

<p>This version uses only subjects with lower phenotypic values for estimating the null distribution. That is, the &quot;controls&quot; are those of lower phenotypic values. When &quot;controls&quot; are of higher phenotypic values, change the sign of the phenotypic values in order to use this function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SKATplus(
  y,
  G,
  X = NULL,
  out_type = "D",
  tau = NULL,
  permutation = FALSE,
  B = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SKATplus_+3A_y">y</code></td>
<td>
<p>a vector of phenotype on <code class="reqn">n</code> subjects.</p>
</td></tr>
<tr><td><code id="SKATplus_+3A_g">G</code></td>
<td>
<p>an <code class="reqn">n \times m</code> matrix of SNP genotypes of <code class="reqn">n</code> study subjects at <code class="reqn">m</code> loci.</p>
</td></tr>
<tr><td><code id="SKATplus_+3A_x">X</code></td>
<td>
<p>a matrix of covariates. It has <code class="reqn">n</code> rows.</p>
</td></tr>
<tr><td><code id="SKATplus_+3A_out_type">out_type</code></td>
<td>
<p>an indicator of the outcome type. <code>"C"</code> for the continuous outcome and <code>"D"</code> for the dichotomous outcome.</p>
</td></tr>
<tr><td><code id="SKATplus_+3A_tau">tau</code></td>
<td>
<p>proportion of selected subjects used for SKATplus.</p>
</td></tr>
<tr><td><code id="SKATplus_+3A_permutation">permutation</code></td>
<td>
<p>an indicator. Use permutation for p-value or not.</p>
</td></tr>
<tr><td><code id="SKATplus_+3A_b">B</code></td>
<td>
<p>the number of permutations if <code>permutation=TRUE</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with class &quot;<code>htest</code>&quot; containing the following components:
* statistic the value of the test statistic, which is the same as SKAT statistic.
* parameter sample size and the number of SNPs
* p.value the p-value for SKATplus computed using Davies' method.
* method a character string indicting the test performed.
* data.name a character string giving the name of the data.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K. (2016) Boosting the power of the sequence kernel association test (SKAT) almost surely by properly estimating its null distribution. <em>A J Hum Genet</em>. 99 (1), 104-114.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    n=1000
    y = c(rep(1, n/2), rep(0, n/2))
    maf = seq(0.05, 0.5, 0.05)
    g = NULL
    for (j in 1:10){
       geno.freq = c(maf[j]^2, 2*maf[j]*(1-maf[j]), (1-maf[j])^2)
       g = cbind(g, sample(c(0,1,2), n, replace=TRUE, prob=geno.freq))
       }
       SKATplus(y, g, X=NULL, out_type="D", tau=NULL, permutation=FALSE, B=1000)

</code></pre>

<hr>
<h2 id='SMR_interval'>Interval Estimates for Summary Data Mendelian Randomization Analysis in the Presence of Winner's Curse</h2><span id='topic+SMR_interval'></span>

<h3>Description</h3>

<p><code>SMR_interval</code> calculates conservative box method interval, k-unit support interval, and Wald confidence interval for the causal effect. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SMR_interval(
  summary.data,
  sig.level = 5e-08,
  k = 2,
  alpha = 0.05,
  method = "box"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SMR_interval_+3A_summary.data">summary.data</code></td>
<td>
<p>a vector (<code class="reqn">\hat b_{gx}</code>, se(<code class="reqn">\hat b_{gx}</code>), <code class="reqn">\hat b_{gy}</code>, se(<code class="reqn">\hat b_{gy}</code>)) of summary data on the exposure <code class="reqn">X</code> and the outcome <code class="reqn">Y</code>. Due to winner's curse, the association <code class="reqn">p</code>-value between the SNP and the exposure is less than <code>sig.level</code>.</p>
</td></tr>
<tr><td><code id="SMR_interval_+3A_sig.level">sig.level</code></td>
<td>
<p>the threshold <code class="reqn">p</code>-value used to select the instrument SNP. The default is <code>5e-8</code>.</p>
</td></tr>
<tr><td><code id="SMR_interval_+3A_k">k</code></td>
<td>
<p>the unit used for the k-unit support. The default value is 2.</p>
</td></tr>
<tr><td><code id="SMR_interval_+3A_alpha">alpha</code></td>
<td>
<p><code class="reqn">(1-\alpha)</code> is the conservative coverage level for the box method interval or the SMR Wald interval. the default value is 0.05</p>
</td></tr>
<tr><td><code id="SMR_interval_+3A_method">method</code></td>
<td>
<p>method to construct the interval. It is either &quot;<code>support</code>&quot;, &quot;<code>box</code>&quot; or &quot;<code>wald</code>&quot;. The default is &quot;<code>box</code>&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is method-dependent.
</p>
<p>For <code>method == "box"</code>: A list containing the following components:
</p>
<p>* an interval estimate.
</p>
<p>* type of the interval: completely bounded, exclusive bounded, or bounded.
</p>
<p>For <code>method == "support"</code>: A list containing the following components:
</p>
<p>* Estimate The likelihood estimate of <code class="reqn">b</code>.
</p>
<p>* an interval estimate.
</p>
<p>For <code>method == "wald"</code>: an interval estimate.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K. (2023) Support interval for two-sample summary data-based mendelia randomization. <em>Genes</em>, 14(1):211.
</p>
<p>Wang, K. (2023) Interval estimate of causal effect in summary data based Mendelian randomization in the presence of winnerâ€™s curse. <em>Genetic Epidemiology</em>, 14(1):211.
</p>
<p>Zhu, Z. et al. (2016) Integration of summary data from GWAS and eQTL studies predicts complex trait gene targets. <em>Nature Genetics</em>, 48(5):481.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>summary.data = c(0.13707, 0.0235162, -0.0637257, 0.013774)
SMR_interval(summary.data)
SMR_interval(summary.data, method = "support")
SMR_interval(summary.data, method = "wald")

</code></pre>

<hr>
<h2 id='tarq'>An Accurate Normalization Method for High Throughput Sequencing Data</h2><span id='topic+tarq'></span>

<h3>Description</h3>

<p>Estimates scaling factors using the trimmed average of ratios of quantiles (TARQ) method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tarq(X, tau = 0.3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tarq_+3A_x">X</code></td>
<td>
<p>a matrix of raw counts. Rows are for taxa (genes, transcripts) and columns for samples</p>
</td></tr>
<tr><td><code id="tarq_+3A_tau">tau</code></td>
<td>
<p>a numerical value in (0, 0.5). The upper <code class="reqn">\tau/2 \times 100</code>% and the lower <code class="reqn">\tau/2 \times 100</code>% of the 
ratios of quantiles are trimmed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Estimation of scaling factors for NGS read counts data is challenging. TARQ provides a quantile-based method for estimating 
scaling factors. It starts by ordering the raw counts 
sample by sample and constructs a reference sample from these ordered counts. To compute the scaling factor for a sample, ratios
of its quantiles to those of the reference sample are formed. Zero ratios are removed. Then extreme ratios (too large or too 
small) are trimmed before taking average over the remaining ratios.
</p>


<h3>Value</h3>

<p>a vector of scaling factors. Normalized counts can be obtained by <code>sweep(X, 2, scale.factors, FUN="/")</code>
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Wang, K. (2018) An Accurate Normalization Method for Next-Generation Sequencing Data.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#data(throat.otu.tab)
#data(throat.meta)
#otu.tab = t(throat.otu.tab)
#tarq(otu.tab, 0.3)

##### Use TARQ with DESeq2 
#dds &lt;- DESeqDataSetFromMatrix(countData = otu.tab,
 #                             colData = throat.meta,
 #                             design= ~ SmokingStatus)
#sizeFactors(dds) &lt;- tarq(otu.tab, 0.3)
#dds &lt;- DESeq(dds)                 
#results(dds)
#
###### Use TARQ with edgeR
#cs &lt;- colSums(otu.tab)
#scale.factors &lt;- tarq(otu.tab, 0.3)
#tmp &lt;- scale.factors/cs
#norm.factors &lt;- tmp/exp(mean(log(tmp)))
#dgList &lt;- DGEList(counts = otu.tab, genes=rownames(otu.tab), norm.factors = norm.factors)
#designMat &lt;- model.matrix(~ throat.meta$SmokingStatus)
#dgList &lt;- estimateGLMCommonDisp(dgList, design=designMat)
#fit &lt;- glmFit(dgList, designMat)
#glmLRT(fit, coef=2)

</code></pre>

<hr>
<h2 id='VSTF.test'>Association Tests for Rare Variants Based on Variance-Stabilizing Transformation</h2><span id='topic+VSTF.test'></span>

<h3>Description</h3>

<p><code>VSTF.test</code> performs tests on association between a rare variant and case-control status using a variance-stabilizing transformation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>VSTF.test(G, method = "Anscombe")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="VSTF.test_+3A_g">G</code></td>
<td>
<p>a <code>2x2</code> matrix. The first row is for cases and the second one for controls. In each row, the first element is the number of non-carriers and the second one is the number of carriers with at least 1 copy of the variant.</p>
</td></tr>
<tr><td><code id="VSTF.test_+3A_method">method</code></td>
<td>
<p>a character string indicating which transformation to use. One of <code>"Anscombe"</code> (default), <code>"arcsine"</code>, <code>"Freeman-Tukey"</code>, and <code>"Chanter"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each test is named after the author(s) of the corresponding publication.
</p>


<h3>Value</h3>

<p>A list with class &quot;<code>test</code>&quot; containing the following components:
* statistic the value of the test statistic.
* p.value the p-value for the test computed from a chi-square distribution with 1 df.
* method a character string indicting the test performed.
* data.name a character string giving the name of the data.
</p>


<h3>Author(s)</h3>

<p>Kai Wang <code>&lt;kai-wang@uiowa.edu&gt;</code>
</p>


<h3>References</h3>

<p>Anscombe, F. J. (1948) The transformation of Poisson, binomial and negative-binomial data. <em>Biometrika</em> <b>35(3/4)</b>, 246&ndash;254.
</p>
<p>Chanter, D. O. (1975). Modifications of the angular transformation. <em>Journal of the Royal Statistical Society. Series B (Applied Statistics)</em> <b>24(3)</b>, 354&ndash;359.
</p>
<p>Freeman, M. F., Tukey, J. W. (1950) Transformations related to the angular and the square root. <em>The Annals of Mathematical Statistics</em> <b>21(4)</b>, 607&ndash;611.
</p>
<p>Wang, K., Fingert, J. (2012) Statistical tests for detecting rare variants using variance-stabilizing transformations. <em>Annals of Human Genetics</em>. 76(5):402-409.
</p>
<p>Zar, J.H. (1999) <em>Biostatistical Analysis, 4th ed.</em>, New Jersey:Prentice-Hall, Inc.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 of Li et al. (2010)
G = rbind(c(14, 999), c(3, 1081))
VSTF.test(G)
VSTF.test(G, method = "arcsine")
VSTF.test(G, method = "Freeman-Tukey")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
