<!DOCTYPE html><html><head><title>Help for package IgAScores</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IgAScores}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#igaprobability'><p>IgA Probability</p></a></li>
<li><a href='#igaprobabilityratio'><p>IgA Probability Ratio</p></a></li>
<li><a href='#igascores'><p>Generate various scores for IgA binding in IgA-Seq experiments</p></a></li>
<li><a href='#kauindex'><p>Kau Index</p></a></li>
<li><a href='#oligoMeta'><p>Metadata for the OligoMM12-Colitis experiment used as an example in the IgAScores package</p></a></li>
<li><a href='#oligoSpecies'><p>Species level counts for the OligoMM12-Colitis experiment used as an example in the IgAScores package</p></a></li>
<li><a href='#palmindex'><p>Palm Index</p></a></li>
<li><a href='#relabund'><p>Relative Abundance from Counts</p></a></li>
<li><a href='#simulateigaseq'><p>Simulate an IgA-Seq dataset from a pre-defined set of IgA-binding probabilities</p></a></li>
<li><a href='#taxnamesplit'><p>Extract name at a given taxonomic level from a full name in the &quot;p__;c__;o__;f__;g__;s__&quot; format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Score Taxon-Level IgA Binding in IgA-Seq Experiments</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions to calculate indices used to score immunoglobulin A (IgA) binding of bacteria in IgA sequencing (IgA-Seq) experiments.
  This includes the original Kau and Palm indices and more recent methods as described in Jackson et al. (2020) &lt;<a href="https://doi.org/10.1101%2F2020.08.19.257501">doi:10.1101/2020.08.19.257501</a>&gt;.
  Additionally the package contains a function to simulate IgA-Seq data and an example experimental data set for method testing. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, markdown</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://doi.org/10.1101/2020.08.19.257501">https://doi.org/10.1101/2020.08.19.257501</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-26 13:27:05 UTC; majackson</td>
</tr>
<tr>
<td>Author:</td>
<td>Matthew Jackson [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matthew Jackson &lt;mattjackson4@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-01 08:10:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='igaprobability'>IgA Probability</h2><span id='topic+igaprobability'></span>

<h3>Description</h3>

<p>Calculate the IgA Postive/Negative Probability as described in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>igaprobability(withinabund, gatesize, presortabund, nazeros = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="igaprobability_+3A_withinabund">withinabund</code></td>
<td>
<p>Abundance of the bacteria in the IgA gate under investigation (can be calculated for either the pos/high or neg/low gating) (abundances should sum to 1 not as a %).</p>
</td></tr>
<tr><td><code id="igaprobability_+3A_gatesize">gatesize</code></td>
<td>
<p>The fraction of events in the flow cytometer within the gate under investigation (as a decimal fraction not a %).</p>
</td></tr>
<tr><td><code id="igaprobability_+3A_presortabund">presortabund</code></td>
<td>
<p>Abundance of the bacteria in whole sample before sorting by IgA (abundances should sum to 1 not as a %).</p>
</td></tr>
<tr><td><code id="igaprobability_+3A_nazeros">nazeros</code></td>
<td>
<p>Return NA if the within and tot abundances are both zero. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the conditional probability that at bacteria will be sufficiently bound/not bound to immunoglobulin A (IgA) to end up in a given IgA gate based on its taxonomy. Calculated on one taxa for one sample.
</p>
<p>This uses Bayes' theorem assuming:
</p>

<ul>
<li><p> That the relative abundance of a given taxon in the IgA gate under question represents the probability of being that taxa given that it is within the IgA gate (either high or low).
</p>
</li>
<li><p> That the percentage of flow cytometery events binned into the IgA gate represents the probability of any bacteria being within the gate.
</p>
</li>
<li><p> That the abundance of the given taxon in the input sample (or whole fraction) represent the probability that any bacteria is assigned to the taxon.
If there is insufficient levels of a taxa in the whole fraction to account for its abundance in the IgA gate, the function assumes all of the taxa fall within this gate (i.e. a probability of 1).
</p>
</li></ul>

<p>Further details can be found in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Value</h3>

<p>A numeric value for the IgA Positive/Negative Probability (depending on the data used for 'withinabund' and 'gatesize') as defined in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>igaprobability(withinabund=0.5,gatesize=0.05,presortabund=0.5)
</code></pre>

<hr>
<h2 id='igaprobabilityratio'>IgA Probability Ratio</h2><span id='topic+igaprobabilityratio'></span>

<h3>Description</h3>

<p>Calculate the IgA Probability Ratio score as described in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>igaprobabilityratio(
  posabund,
  negabund,
  possize,
  negsize,
  pseudo = 1e-05,
  scaleratio = TRUE,
  nazeros = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="igaprobabilityratio_+3A_posabund">posabund</code></td>
<td>
<p>Abundance of the bacteria in the IgA positive/high fraction (abundances should sum to 1 not as a %).</p>
</td></tr>
<tr><td><code id="igaprobabilityratio_+3A_negabund">negabund</code></td>
<td>
<p>Abundance of the bacteria in the IgA negative/low fraction (abundances should sum to 1 not as a %).</p>
</td></tr>
<tr><td><code id="igaprobabilityratio_+3A_possize">possize</code></td>
<td>
<p>The fraction of events in the flow cytometer classed as IgA positive/high (as a decimal fraction not a %).</p>
</td></tr>
<tr><td><code id="igaprobabilityratio_+3A_negsize">negsize</code></td>
<td>
<p>The fraction of events in the flow cytometer classed as IgA negative/low (as a decimal fraction not a %).</p>
</td></tr>
<tr><td><code id="igaprobabilityratio_+3A_pseudo">pseudo</code></td>
<td>
<p>Pseudo count added to both the IgA positive and negative abundance values prior to calculation. Defaults to 1e-5. Recommend setting to minimum observed abundance in whole dataset.</p>
</td></tr>
<tr><td><code id="igaprobabilityratio_+3A_scaleratio">scaleratio</code></td>
<td>
<p>Should probratio scores be scaled to the pseudo count. Default is TRUE.</p>
</td></tr>
<tr><td><code id="igaprobabilityratio_+3A_nazeros">nazeros</code></td>
<td>
<p>Return NA if the pos and neg abundances are both zero. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the ratio of the immunoglobulin A (IgA) positive fraction probability relative to the IgA negative fraction probability for a single taxa in a single sample.
These probabilities can individually be calculated using the igaprobability() function. As both calculations have the whole fraction taxon abundance as a denominator it cancels.
This means the IgA probability ratio can be calculated without this information.
Further details can be found in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Value</h3>

<p>A numeric value for the IgA Probability Ratio as defined in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>igaprobabilityratio(posabund=0.2,negabund=0.05,possize=0.05,negsize=0.6,pseudo=0.0002)
</code></pre>

<hr>
<h2 id='igascores'>Generate various scores for IgA binding in IgA-Seq experiments</h2><span id='topic+igascores'></span>

<h3>Description</h3>

<p>Calculate various different IgA-Seq scores across all the taxa and samples in an experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>igascores(
  posabunds = NULL,
  negabunds = NULL,
  possizes = NULL,
  negsizes = NULL,
  pseudo = NULL,
  presortabunds = NULL,
  method = "probratio",
  scaleratio = TRUE,
  nazeros = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="igascores_+3A_posabunds">posabunds</code></td>
<td>
<p>A dataframe of taxa abundances in the positive/high IgA gate samples. Samples as columns and taxa as rows, column and row names must match across abundance tables.</p>
</td></tr>
<tr><td><code id="igascores_+3A_negabunds">negabunds</code></td>
<td>
<p>A dataframe of taxa abundances in the negative/low IgA gate samples. Samples as columns and taxa as rows, column and row names must match across abundance tables.</p>
</td></tr>
<tr><td><code id="igascores_+3A_possizes">possizes</code></td>
<td>
<p>A named vector containing the fraction of events in the IgA positive gate for each sample, with sample names matching abundance dataframes.</p>
</td></tr>
<tr><td><code id="igascores_+3A_negsizes">negsizes</code></td>
<td>
<p>A named vector containing the fraction of events in the IgA negative gate for each sample, with sample names matching abundance dataframes.</p>
</td></tr>
<tr><td><code id="igascores_+3A_pseudo">pseudo</code></td>
<td>
<p>The pseudo count to be used in scores. Default is 1e-5. Recommend setting to minimum observed abundance.</p>
</td></tr>
<tr><td><code id="igascores_+3A_presortabunds">presortabunds</code></td>
<td>
<p>A dataframe of taxa abundances in the whole/initial samples. Samples as columns and taxa as rows, column and row names must match across abundance tables.</p>
</td></tr>
<tr><td><code id="igascores_+3A_method">method</code></td>
<td>
<p>Method to use to score IgA binding. One of: &quot;probratio&quot;,&quot;prob&quot;,&quot;kau&quot;,&quot;palm&quot;. Default is &quot;probratio&quot;.</p>
</td></tr>
<tr><td><code id="igascores_+3A_scaleratio">scaleratio</code></td>
<td>
<p>Should probratio scores be scaled to the pseudo count. Default is TRUE.</p>
</td></tr>
<tr><td><code id="igascores_+3A_nazeros">nazeros</code></td>
<td>
<p>Should taxa with zero abundance in both the posabunds and negabunds (posabunds and presortabunds for prob method)  be scored as NA. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function enables calculation of a variety of different indices for scoring immunoglobulin A (IgA) binding to taxa in IgA sequencing (IgA-Seq) experiments.
It is designed to be called on dataframes of abundance values, allowing easy calculation of scores across multiple taxa and samples.
The igaprobabilityratio(), igaprobability(), kauindex() and palmindex() functions can be used to calculate scores for one taxa and one sample.
</p>
<p>Scoring method can be chosen by specifying the method parameter as one of: &quot;probratio&quot;, &quot;prob&quot;, &quot;kau&quot;, &quot;palm&quot; (Defaults to &quot;probratio&quot;).
Each method requires different inputs as detailed below:
</p>

<ul>
<li><p> probratio - equivalent to igaprobabilityratio() - requires two separate dataframes with iga positive abundances and iga negative abundances, two vectors with the sizes of the iga positive and negative gates per sample, and a pseudo count
</p>
</li>
<li><p> prob - equivalent to igaprobability() - requires a dataframe with iga pos or neg fraction abundances, a vector of iga pos or neg gate size per sample, and a dataframe of taxa abundances in the presort samples
</p>
</li>
<li><p> kau - equivalent to kauindex() - requires two separate dataframes with iga positive abundances and iga negative abundances, and a pseudo count
</p>
</li>
<li><p> palm - equivalent to palmindex() - requires two separate dataframes with iga positive abundances and iga negative abundances, and a pseudo count
</p>
</li></ul>



<h3>Value</h3>

<p>A data frame of IgA binding scores for all taxa and samples in the input data frame, generated using the scoring appraoch specified in 'method'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pab &lt;- data.frame(Samp1=c(0.01,0.02,0.03),Samp2=c(0.05,0.02,0.04))
rownames(pab) &lt;- c("Taxon1","Taxon2","Taxon3")
nab &lt;- data.frame(Samp1=c(0.08,0.2,0.11),Samp2=c(0.05,0.0,0.07))
rownames(nab) &lt;- c("Taxon1","Taxon2","Taxon3")
ps &lt;- c(0.04,0.1)
ns &lt;- c(0.08,0.4)
preab &lt;- data.frame(Samp1=c(0.1,0.3,0.2),Samp2=c(0.15,0.05,0.2))
rownames(preab) &lt;- c("Taxon1","Taxon2","Taxon3")
igascores(posabunds=pab,negabunds=nab, possizes=ps, negsizes=ns,pseudo=0.009)
igascores(posabunds=pab, possizes=ps, presortabunds=preab, method="prob")
igascores(posabunds=pab, negabunds=nab, pseudo=0.009, method="palm")
igascores(posabunds=pab, negabunds=nab, pseudo=0.009, method="kau")
</code></pre>

<hr>
<h2 id='kauindex'>Kau Index</h2><span id='topic+kauindex'></span>

<h3>Description</h3>

<p>This function calculates the immunoglobulin A (IgA) Index as defined in Kau et al. (2015, doi: <a href="https://doi.org/10.1126/scitranslmed.aaa4877">10.1126/scitranslmed.aaa4877</a>) for a single taxon in a single sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kauindex(posabund, negabund, pseudo = 1e-05, nazeros = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kauindex_+3A_posabund">posabund</code></td>
<td>
<p>The abundance of the bacteria in the IgA positive/high fraction (abundances should sum to 1 not as a %).</p>
</td></tr>
<tr><td><code id="kauindex_+3A_negabund">negabund</code></td>
<td>
<p>The abundance of the bacteria in the IgA negative/low fraction (abundances should sum to 1 not as a %).</p>
</td></tr>
<tr><td><code id="kauindex_+3A_pseudo">pseudo</code></td>
<td>
<p>Pseudo count added to both the IgA positive and negative fraction values prior to calculation. Defaults to 1e-5. Recommend setting to minimum observed abundance in whole dataset.</p>
</td></tr>
<tr><td><code id="kauindex_+3A_nazeros">nazeros</code></td>
<td>
<p>Return NA if the pos and neg abundances are both zero. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value for the Kau index as defined in Kau et al. (2015, doi: <a href="https://doi.org/10.1126/scitranslmed.aaa4877">10.1126/scitranslmed.aaa4877</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kauindex(posabund=0.1,negabund=0.2,pseudo=0.0002)
</code></pre>

<hr>
<h2 id='oligoMeta'>Metadata for the OligoMM12-Colitis experiment used as an example in the IgAScores package</h2><span id='topic+oligoMeta'></span>

<h3>Description</h3>

<p>Metadata associated with the oligoSpecies data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(oligoMeta)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"tibble"</code>.
</p>


<h3>Details</h3>

<p>Metadata for an experiment where mice with a defined gut microbiota (OligoMM12) were either given *Helicobacter hepaticus* and IL10R antibody or the antibody alone (the first developing colitis).
These data accompany the species level counts in oligoSpecies and also include a negative extraction control. A subset of the combined condition were not properly colonised by *H.hepaticus* and excluded from later analyses.
Further details can be found in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>References</h3>

<p>To come...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(oligoMeta)
</code></pre>

<hr>
<h2 id='oligoSpecies'>Species level counts for the OligoMM12-Colitis experiment used as an example in the IgAScores package</h2><span id='topic+oligoSpecies'></span>

<h3>Description</h3>

<p>Data from the colitis model described in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(oligoSpecies)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>"tibble"</code>.
</p>


<h3>Details</h3>

<p>Species level counts for an experiment where mice with a defined gut microbiota (OligoMM12) were either given *Helicobacter hepaticus* and IL10R antibody or the antibody alone (the first developing colitis).
Metadata for this experiment can be found in oligoMeta. Counts were generated from ASVs from the V4 region of the 16S rRNA gene processed using DADA2 and aligned to the *&quot;RefSeq-RDP16S_v2_May2018.fa.gz&quot;* database.
Further details can be found in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>


<h3>References</h3>

<p>To come...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(oligoSpecies)
</code></pre>

<hr>
<h2 id='palmindex'>Palm Index</h2><span id='topic+palmindex'></span>

<h3>Description</h3>

<p>This function calculates the immunoglobulin A (IgA) Index as defined in Palm et al. (2014, doi: <a href="https://doi.org/10.1016/j.cell.2014.08.006">10.1016/j.cell.2014.08.006</a>) for a single taxon in a single sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>palmindex(posabund, negabund, pseudo = 1e-05, nazeros = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="palmindex_+3A_posabund">posabund</code></td>
<td>
<p>Abundance of the bacteria in the IgA positive/high fraction.</p>
</td></tr>
<tr><td><code id="palmindex_+3A_negabund">negabund</code></td>
<td>
<p>Abundance of the bacteria in the IgA negative/low fraction.</p>
</td></tr>
<tr><td><code id="palmindex_+3A_pseudo">pseudo</code></td>
<td>
<p>Pseudo count added to the abundance of the IgA negative fraction if the bacteria is not in that fraction. Defaults to 1e-5. Recommend setting to minimum observed abundance in whole dataset.</p>
</td></tr>
<tr><td><code id="palmindex_+3A_nazeros">nazeros</code></td>
<td>
<p>Return NA if the pos and neg abundances are both zero. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric value for the Palm index as defined in Palm et al. (2014, doi: <a href="https://doi.org/10.1016/j.cell.2014.08.006">10.1016/j.cell.2014.08.006</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>palmindex(posabund=0.1,negabund=0.2,pseudo=0.0002)
</code></pre>

<hr>
<h2 id='relabund'>Relative Abundance from Counts</h2><span id='topic+relabund'></span>

<h3>Description</h3>

<p>This function converts values in a dataframe to a fraction/percentage of the sum of their column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relabund(counttable, percentage = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relabund_+3A_counttable">counttable</code></td>
<td>
<p>Data frame of numeric values with rows as observations and columns as samples.</p>
</td></tr>
<tr><td><code id="relabund_+3A_percentage">percentage</code></td>
<td>
<p>Should values be returned as a percentage? i.e multiplied by 100. Default is FALSE (as required for most IgA scoring approaches).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the input data normalised by column (to sum to either 1 or 100).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>taxcounts &lt;- data.frame(Sample1=c(1,2,10,10),Sample2=c(3,10,5,1))
rownames(taxcounts) &lt;- c("Taxon1","Taxon2","Taxon3","Taxon4")
relabund(taxcounts)
</code></pre>

<hr>
<h2 id='simulateigaseq'>Simulate an IgA-Seq dataset from a pre-defined set of IgA-binding probabilities</h2><span id='topic+simulateigaseq'></span>

<h3>Description</h3>

<p>Simulates IgA-Seq to create datasets with a defined binding distribution that can be used to test scoring method performance
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateigaseq(
  igavalmeans = NULL,
  igavalsds = NULL,
  nosamples = 10,
  samplingdepth = 1e+05,
  posthresh = 4,
  negthresh = 2,
  seed = 66,
  betweengroups = FALSE,
  betweenper = 10,
  betweensp = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulateigaseq_+3A_igavalmeans">igavalmeans</code></td>
<td>
<p>A vector of mean IgA values for as many species as you wish to simulate. Will default to an exponentially distributed vector of 10 species.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_igavalsds">igavalsds</code></td>
<td>
<p>A vector of standard deviations that will be used to generate IgA value distributions alongside the means. Defaults to 1 for all values.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_nosamples">nosamples</code></td>
<td>
<p>The number of samples to generate simulated data from. Defaults to 10.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_samplingdepth">samplingdepth</code></td>
<td>
<p>The number of bacteria to simulate in each sample. Defaults to 100000.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_posthresh">posthresh</code></td>
<td>
<p>The IgA value threshold above which a bacteria will be considered IgA positive. Defaults to 4 (which is reasonable with the other defaults). It is recommended to run a simulation twice to determine reasonable thresholds on the first go.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_negthresh">negthresh</code></td>
<td>
<p>The IgA value threshold below which a bacteria will be considered IgA negative. Defaults to 2 (which is reasonable with the other defaults). It is recommended to run a simulation twice to determine reasonable thresholds on the first go.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_seed">seed</code></td>
<td>
<p>Seed for random number generation. Has a default so must be changed to rerun simulations.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_betweengroups">betweengroups</code></td>
<td>
<p>If TRUE this will modify starting abundances of half of the samples similarly (by adding betweenper% of total counts to a single species) to simulate the case where there is an abundance shift without a change in IgA binding affinity. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_betweenper">betweenper</code></td>
<td>
<p>Percentage of total counts to add to a species in the second group in the betweengroups mode.</p>
</td></tr>
<tr><td><code id="simulateigaseq_+3A_betweensp">betweensp</code></td>
<td>
<p>Species (by index) to increased in between groups simulation. Chosen at random if NULL (default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will generate a simulated immunoglobulin A sequencing (IgA-Seq) data set starting from a list containing the mean (and standard deviations) of IgA binding values expected for each species and cut-offs for defining the IgA positive and negative gates.
The input is a vector giving the average IgA value of each species (any arbitrary value that will represent the relative level of IgA binding between the species, ensure standard deviation and cut-offs are in the same magnitude).
These values are treated as the means of a normal distribution of IgA binding values for each species.
Species counts are generated on a log distribution for a given number of samples at an even depth.
For each bacteria in each sample, an IgA binding value is then assigned by sampling from its species IgA value distribution.
The value thresholds defining the positive and negative gates are then used to generate positive and negative counts tables of the bacteria whose values fall into these groups.
A second mode can also be used (by toggling betweengroups) that will introduce a consistent abundance change in half the samples by increasing one species in them. This can be used to simulate case-control experiments where, as an example,  one taxa has bloomed.
Further details can be found in Jackson et al. (2020, doi: <a href="https://doi.org/10.1101/2020.08.19.257501">10.1101/2020.08.19.257501</a>).
</p>
<p>Note: IgA values are simulated for each bacteria in each sample, setting the combination of the samplingdepth, number of species, and number of samples too high will slow the data generation.
</p>


<h3>Value</h3>

<p>A list containing the simulated data set and relevant input parameters.
</p>

<ul>
<li><p> presortcounts - A data frame containing simulated species counts for each sample in the pre-sort sample.
</p>
</li>
<li><p> presortabunds - presortcounts as relative abundances.
</p>
</li>
<li><p> poscounts - A data frame containing simulated species counts for each sample in the IgA positive fraction.
</p>
</li>
<li><p> posabunds - poscounts as relative abundances.
</p>
</li>
<li><p> negcounts - A data frame containing simulated species counts for each sample in the IgA negative fraction.
</p>
</li>
<li><p> negabunds - negcounts as relative abundances.
</p>
</li>
<li><p> possizes - A vector of the IgA positive fraction sizes for each sample.
</p>
</li>
<li><p> negsizes - A vector of the IgA negative fraction sizes for each sample.
</p>
</li>
<li><p> igabinding - A long format data frame containing the simulated IgA binding values for all simulated bacteria used to generate the count tables.
</p>
</li>
<li><p> igavalmeans - A vector of the mean IgA values for each species used in the simulation.
</p>
</li>
<li><p> igavalsds - A vector of the standard deviations of the IgA values for each species used in the simulation.
</p>
</li>
<li><p> posthresh - Numeric, the lower threshold used to determine a bacteria is IgA postive in the simulation.
</p>
</li>
<li><p> negthresh - Numeric, the upper threshold used to determine a bacteria is IgA negative in the simulation.
</p>
</li>
<li><p> expgroup - A vector showing class labels for the experimental group of each sample in the experiment. Will be uniform unless doing between group simulations.
</p>
</li>
<li><p> expspecies - Numeric, showing which species was modelled as differentially abundant between experimental groups when carryingout between group simulations.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>dat &lt;- simulateigaseq(c(0.1,1,10,15),rep(1,4),posthresh=8,negthresh=4,samplingdepth=100)
</code></pre>

<hr>
<h2 id='taxnamesplit'>Extract name at a given taxonomic level from a full name in the &quot;p__;c__;o__;f__;g__;s__&quot; format</h2><span id='topic+taxnamesplit'></span>

<h3>Description</h3>

<p>This function splits a full taxonomic lineage as a given level and returns the latter half.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxnamesplit(names, level = "genus")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxnamesplit_+3A_names">names</code></td>
<td>
<p>Name string/ vector of name strings</p>
</td></tr>
<tr><td><code id="taxnamesplit_+3A_level">level</code></td>
<td>
<p>taxonomic level to split at must be in range phylum to species (default is genus).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string (or vector of strings if input is a vector) containing the second part of the input string split at the given taxonomic level.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>taxnamesplit("p__Bacteroidetes;c__Bacteroidia","class")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
