<!DOCTYPE html><html><head><title>Help for package IGST</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IGST}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#IGST.bootmrmrsvm.pval.cutoff'>
<p>Gene selection using SVM and MRMR feature selection techniques with bootstrapping procedure</p></a></li>
<li><a href='#IGST.bootmrmrsvm.weight.cutoff'>
<p>Identification of informative gene set based on weights obtained from SVM and MRMR feature selection technique with bootstrapping procedure</p></a></li>
<li><a href='#IGST.pval.bootmrmrsvm'>
<p>Computation of statistical significance values for genes using SVM and MRMR feature selection technique with bootstrapping procedure</p></a></li>
<li><a href='#IGST.weight.bootmrmrsvm'>
<p>Computation of weights for informative genes or gene set selection using SVM and MRMR feature selection technique with bootstrapping procedure</p></a></li>
<li><a href='#rice_cold'>
<p>A gene expression dataset of rice under cold stress</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Informative Gene Selection Tool</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Nitesh Kumar Sharma, Dwijesh Chandra Mishra, Neeraj Budhlakoti and Md. Samir Farooqi</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nitesh Kumar Sharma &lt;sharmanitesh.iasri@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Mining informative genes with certain biological meanings are important for clinical diagnosis of disease and discovery of disease mechanisms in plants and animals. This process involves identification of relevant genes and removal of redundant genes as much as possible from a whole gene set. This package selects the informative genes related to a specific trait using gene expression dataset. These trait specific genes are considered as informative genes. This package returns the informative gene set from the high dimensional gene expression data using a combination of methods SVM and MRMR (for feature selection) with bootstrapping procedure. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>e1071, BootMRMR</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-21 10:19:01 UTC; Nitesh-PC</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-31 16:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='IGST.bootmrmrsvm.pval.cutoff'>
Gene selection using SVM and MRMR feature selection techniques with bootstrapping procedure
</h2><span id='topic+IGST.bootmrmrsvm.pval.cutoff'></span>

<h3>Description</h3>

<p>The informative gene set which have maximum relevance with target class or trait and minimum redundancy among genes based on statistical significance values computed from the combination of SVM and MRMR feature selection techniques with bootstrapping procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IGST.bootmrmrsvm.pval.cutoff (x, y, s, Q, v, re)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IGST.bootmrmrsvm.pval.cutoff_+3A_x">x</code></td>
<td>
<p> x is a n by p data frame of gene expression values where rows represent genes and columns represent samples. Each cell entry represents the expression level of a gene in a sample or subject (row names of x as gene names or gene ids).</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.pval.cutoff_+3A_y">y</code></td>
<td>
<p> y is a p by 1 numeric vector with entries 1 or -1 representing sample labels, where, 1\/-1 represents the sample label of subjects orsamples for stress/control condition(for two class problems).</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.pval.cutoff_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the size of the informative gene set to be obtained.</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.pval.cutoff_+3A_q">Q</code></td>
<td>
<p>Q is a scalar representing the quartile value of the rank scores of genes (lies within 		1\/N to 1), usually the second quartile, i.e. 0.5 or third quartile i.e. 0.75 may be taken.</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.pval.cutoff_+3A_v">v</code></td>
<td>
<p>v is a scalar representing the weightage of a method and must be within 0 and 1. </p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.pval.cutoff_+3A_re">re</code></td>
<td>
<p>re is a scalar representing the number of bootstrap generated, re must be sufficiently large (i.e. number of times bootstrap samples are generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list of the genes\/informative gene set which are highly relevant to the particular trait or condition under investigation and minimal redundant among themselves without any spurious association among the genes.</p>


<h3>Author(s)</h3>

<p>Nitesh Kumar Sharma, Dwijesh Chandra Mishra, Neeraj Budhlakoti and Md. Samir Farooqi
</p>


<h3>References</h3>

<p>Das, S., Rai, A., Mishra, D. C., &amp; Rai, S. N. (2018). Statistical approach for selection of biologically informative genes. Gene, 655, 71-83.
</p>
<p>Guyon, I., Weston, J., Barnhill, S., &amp; Vapnik, V. (2002). Gene selection for cancer classification using support vector machines. Machine Learning, 46(1-3), 389-422.
</p>
<p>Wang, J., Chen, L., Wang, Y., Zhang, J., Liang, Y., &amp; Xu, D. (2013). A computational systems biology study for understanding salt tolerance mechanism in rice. PLoS One, 8(6), e64929.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#################################
library(IGST)
data(rice_cold)
x&lt;-rice_cold[-1,]
y&lt;-rice_cold[1,]
y&lt;-as.matrix(y)
y&lt;-as.vector(y)
s&lt;-10
Q&lt;-0.5
v&lt;-0.5
re&lt;-10
IGST.bootmrmrsvm.pval.cutoff (x, y, s, Q, v, re)
</code></pre>

<hr>
<h2 id='IGST.bootmrmrsvm.weight.cutoff'>
Identification of informative gene set based on weights obtained from SVM and MRMR feature selection technique with bootstrapping procedure
</h2><span id='topic+IGST.bootmrmrsvm.weight.cutoff'></span>

<h3>Description</h3>

<p>The function enables to find set of informative genes based on weights which are obtained by maximizing the relevancy of genes with classes or condition or trait and minimizing the redundancy among genes from the combination of SVM and MRMR feature selection techniques with bootstrapping procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IGST.bootmrmrsvm.weight.cutoff (x, y, s, v, re)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IGST.bootmrmrsvm.weight.cutoff_+3A_x">x</code></td>
<td>
<p> x is a n by p data frame of gene expression values where rows represent genes and columns represent samples. Each cell entry represents the expression level of a gene in a sample or subject (row names of x as gene names or gene ids).</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.weight.cutoff_+3A_y">y</code></td>
<td>
<p> y is a p by 1 numeric vector with entries 1 or -1 representing sample labels, where, 1 or -1 represents the sample label of subjects or samples for stress or control condition(for two class problems).</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.weight.cutoff_+3A_s">s</code></td>
<td>
<p>s is a scalar representing the size of the informative gene set to be obtained.</p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.weight.cutoff_+3A_v">v</code></td>
<td>
<p>v is a scalar representing the weightage of a method and must be within 0 and 1. </p>
</td></tr>
<tr><td><code id="IGST.bootmrmrsvm.weight.cutoff_+3A_re">re</code></td>
<td>
<p>re is a scalar representing the number of bootstrap generated, re must be sufficiently large (i.e. number of times bootstrap samples are generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a set of genes, which are highly informative to the trait or condition under consideration based on weights given by the combination of SVM and MRMR feature selection techniques with bootstrapping procedure. </p>


<h3>Author(s)</h3>

<p>Nitesh Kumar Sharma, Dwijesh Chandra Mishra, Neeraj Budhlakoti and Md. Samir Farooqi
</p>


<h3>References</h3>

<p>Ding, C., &amp; Peng, H. (2005). Minimum redundancy feature selection from microarray gene expression data. Journal of Bioinformatics and Computational Biology, 3(02), 185-205.
</p>
<p>Mishra DC, Kumar S, Lal SB, Saha A, Chaturvedi KK, Budhlakoti N, et al.( 2018) TAGPT: A Web Server for Prediction of Trait Associated Genes using Gene Expression Data. Annals of Genetics and Genetic Disorder. 1(1): 1003.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#################################
library(IGST)
data(rice_cold)
x&lt;-rice_cold[-1,]
y&lt;-rice_cold[1,]
y&lt;-as.matrix(y)
y&lt;-as.vector(y)
s&lt;-10
#Q&lt;-0.5
v&lt;-0.5
re&lt;-10
IGST.bootmrmrsvm.weight.cutoff (x, y, s, v, re)
</code></pre>

<hr>
<h2 id='IGST.pval.bootmrmrsvm'>
Computation of statistical significance values for genes using SVM and MRMR feature selection technique with bootstrapping procedure
</h2><span id='topic+IGST.pval.bootmrmrsvm'></span>

<h3>Description</h3>

<p>The function computes the statistical significance values for the genes from the non-parametric test &quot;H0: i-th gene is not informative against H1: i-th gene is informative&quot; for selection of informative genes using SVM and MRMR feature selection technique with bootstrapping procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IGST.pval.bootmrmrsvm(x, y, re, Q, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IGST.pval.bootmrmrsvm_+3A_x">x</code></td>
<td>
<p> x is a n by p data frame of gene expression values where rows represent genes and columns represent samples. Each cell entry represents the expression level of a gene in a sample or subject (row names of x as gene names or gene ids).</p>
</td></tr>
<tr><td><code id="IGST.pval.bootmrmrsvm_+3A_y">y</code></td>
<td>
<p> y is a p by 1 numeric vector with entries 1 or -1 representing sample labels, where, 1 or -1 represents the sample label of subjects or samples for stress or control condition(for two class problems).</p>
</td></tr>
<tr><td><code id="IGST.pval.bootmrmrsvm_+3A_q">Q</code></td>
<td>
<p>Q is a scalar representing the quartile value of the rank scores of genes (lies within 1\/N to 1), usually the second quartile, i.e. 0.5 or third quartile i.e. 0.75 may be taken.</p>
</td></tr>
<tr><td><code id="IGST.pval.bootmrmrsvm_+3A_v">v</code></td>
<td>
<p>v is a scalar representing the weightage of a method and must be within 0 and 1. </p>
</td></tr>
<tr><td><code id="IGST.pval.bootmrmrsvm_+3A_re">re</code></td>
<td>
<p>re is a scalar representing the number of bootstrap generated, re must be sufficiently large (i.e. number of times bootstrap samples are generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a vector of p-values for all the genes from the given statistical test in the dataset using SVM and MRMR feature selection technique with bootstrapping procedure.
</p>


<h3>Author(s)</h3>

<p>Nitesh Kumar Sharma, Dwijesh Chandra Mishra, Neeraj Budhlakoti and Md. Samir Farooqi
</p>


<h3>References</h3>

<p>Das, S., Rai, A., Mishra, D. C., &amp; Rai, S. N. (2018). Statistical approach for selection of biologically informative genes. Gene, 655, 71-83.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#################################
library(IGST)
data(rice_cold)
x&lt;-rice_cold[-1,]
y&lt;-rice_cold[1,]
y&lt;-as.matrix(y)
y&lt;-as.vector(y)
#s&lt;-10
Q&lt;-0.5
v&lt;-0.5
re&lt;-10
IGST.pval.bootmrmrsvm (x, y, re, Q, v)
</code></pre>

<hr>
<h2 id='IGST.weight.bootmrmrsvm'>
Computation of weights for informative genes or gene set selection using SVM and MRMR feature selection technique with bootstrapping procedure
</h2><span id='topic+IGST.weight.bootmrmrsvm'></span>

<h3>Description</h3>

<p>The function computes the weights associated with each genes for a given dataset using SVM and MRMR feature selection technique with bootstrapping procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IGST.weight.bootmrmrsvm (x, y, re, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IGST.weight.bootmrmrsvm_+3A_x">x</code></td>
<td>
<p> x is a n by p data frame of gene expression values where rows represent genes and columns represent samples. Each cell entry represents the expression level of a gene in a sample or subject (row names of x as gene names or gene ids).</p>
</td></tr>
<tr><td><code id="IGST.weight.bootmrmrsvm_+3A_y">y</code></td>
<td>
<p> y is a p by 1 numeric vector with entries 1 or -1 representing sample labels, where, 1\/-1 represents the sample label of subjects or samples for stress or control condition(for two class problems).</p>
</td></tr>
<tr><td><code id="IGST.weight.bootmrmrsvm_+3A_v">v</code></td>
<td>
<p>v is a scalar representing the weightage of a method and must be within 0 and 1. </p>
</td></tr>
<tr><td><code id="IGST.weight.bootmrmrsvm_+3A_re">re</code></td>
<td>
<p>re is a scalar representing the number of bootstrap generated, re must be sufficiently large (i.e. number of times bootstrap samples are generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a vector of weights associated with each genes computed from SVM and MRMR feature selection technique with bootstrapping procedure for a given dataset. </p>


<h3>Author(s)</h3>

<p>Nitesh Kumar Sharma, Dwijesh Chandra Mishra, Neeraj Budhlakoti and Md. Samir Farooqi
</p>


<h3>References</h3>

<p>Wang, J., Chen, L., Wang, Y., Zhang, J., Liang, Y., &amp; Xu, D. (2013). A computational systems biology study for understanding salt tolerance mechanism in rice. PLoS One, 8(6), e64929.
</p>
<p>Ding, C., &amp; Peng, H. (2005). Minimum redundancy feature selection from microarray gene expression data. Journal of Bioinformatics and Computational Biology, 3(02), 185-205.
</p>
<p>Mishra DC, Kumar S, Lal SB, Saha A, Chaturvedi KK, Budhlakoti N, et al.( 2018) TAGPT: A Web Server for Prediction of Trait Associated Genes using Gene Expression Data. Annals of Genetics and Genetic Disorder. 1(1): 1003.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#################################
library(IGST)
data(rice_cold)
x&lt;-rice_cold[-1,]
y&lt;-rice_cold[1,]
y&lt;-as.matrix(y)
y&lt;-as.vector(y)
#s&lt;-10
#Q&lt;-0.5
v&lt;-0.5
re&lt;-10
IGST.weight.bootmrmrsvm (x, y, re, v)
</code></pre>

<hr>
<h2 id='rice_cold'>
A gene expression dataset of rice under cold stress
</h2><span id='topic+rice_cold'></span>

<h3>Description</h3>

<p>This data has gene expression values of 250 genes over 36 samples or subjects for a cold vs. control study in rice. These 36 samples belong to either of cold stress or control condition (two class problem). This gene expression data is balanced type as the first 18 samples are under cold stress and the later 18 samples are under control condition. The first row of the data contains the samples or subjects labels with entries are 1 and -1, where the label ’1’ and ’-1’ represent samples generated under cold stress and control condition respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("rice_cold")</code></pre>


<h3>Format</h3>

<p>A data frame with 250 rows as genes with 36 columns as samples or subjects. Each column (sample) represent the gene expression values of genes. Each column as microarray samples with labels -1 or 1 represents control or cold stress respectively.
</p>


<h3>Details</h3>

<p>The data is created by taking 250 genes from the large number of genes from NCBI GEO database. The rows are the genes and columns are the samples or subjects. The first half of the samples or subjects are generated under cold stress condition and other half under control condition. The first row of the data contains the samples/subjects labels with entries are 1 and -1, where the label ’1’ and ’-1’ represent samples generated under cold stress and control condition respectively.
</p>


<h3>Source</h3>

<p>Gene Expression Omnibus: NCBI gene expression and hybridization array data repository.ncbi.nlm.nih.gov/geo/.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>####################################
library(IGST)
data(rice_cold)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
