<!DOCTYPE html><html lang="en"><head><title>Help for package ihclust</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ihclust}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ihclust'><p>Iterative Hierarchical Clustering (IHC)</p></a></li>
<li><a href='#opioidData'><p>Opioid Dispensing Rates</p></a></li>
<li><a href='#simcurve'><p>simcurve</p></a></li>
<li><a href='#testchange'><p>testchange</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Iterative Hierarchical Clustering (IHC)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a set of tools to
  i) identify geographic areas with significant change over time in drug utilization, and 
  ii) characterize common change over time patterns among the time series for multiple geographic areas.
  For reference, see below:
    1. Song, J., Carey, M., Zhu, H., Miao, H., Ram´ırez, J. C., &amp; Wu, H. (2018) &lt;<a href="https://doi.org/10.1504%2FIJCBDD.2018.10011910">doi:10.1504/IJCBDD.2018.10011910</a>&gt;
    2. Wu, S., Wu, H. (2013) &lt;<a href="https://doi.org/10.1186%2F1471-2105-14-6">doi:10.1186/1471-2105-14-6</a>&gt;
    3. Carey, M., Wu, S., Gan, G. &amp; Wu, H. (2016) &lt;<a href="https://doi.org/10.1016%2Fj.idm.2016.07.001">doi:10.1016/j.idm.2016.07.001</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GNU General Public License (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>factoextra, splines, ggplot2, foreach, doParallel, parallel</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-22 17:19:37 UTC; elincho</td>
</tr>
<tr>
<td>Author:</td>
<td>Elin Cho [aut, cre],
  Yuting Xu [aut],
  Jaejoon Song [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Elin Cho &lt;elincho524@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-27 07:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ihclust'>Iterative Hierarchical Clustering (IHC)</h2><span id='topic+ihclust'></span>

<h3>Description</h3>

<p>This function identifies inhomogeneous clusters using iterative hierarchical clustering (IHC) method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ihclust(
  data,
  smooth = TRUE,
  cor_criteria = 0.75,
  max_iteration = 100,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ihclust_+3A_data">data</code></td>
<td>
<p>a numeric matrix, each row representing a time-series
and each column representing a time point</p>
</td></tr>
<tr><td><code id="ihclust_+3A_smooth">smooth</code></td>
<td>
<p>if smooth = 'TRUE', a smooth function is applied before clustering</p>
</td></tr>
<tr><td><code id="ihclust_+3A_cor_criteria">cor_criteria</code></td>
<td>
<p>pre-specified correlation criteria</p>
</td></tr>
<tr><td><code id="ihclust_+3A_max_iteration">max_iteration</code></td>
<td>
<p>maximum number of iterations</p>
</td></tr>
<tr><td><code id="ihclust_+3A_verbose">verbose</code></td>
<td>
<p>if verbose = 'TRUE', the result of a progress is printed</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ihclust
</p>
<p>The IHC algorithm implements the three steps as outlined below.
First, the Initialization step clusters the data using hierarchical clustering.
Second, cluster centers are obtained as an average of all the data points in the cluster.
The Merging step considers each of the cluster centers (exemplars) as ‘new data point’,
and use the same procedure described in the Initialization step to merge the exemplars into a new set of clusters.
Third, the Pruning step streamlines the clusters and removes inconsistencies by
reassessing the cluster membership by each data point.
</p>


<h3>Value</h3>

<p>Output from the function is a list of three items:
</p>

<ul>
<li><p> Cluster_Label - the cluster label for each data point
</p>
</li>
<li><p> Num_Iterations - total number of iterations
</p>
</li>
<li><p> Unique_Clusters_in_Iteration - unique clusters in each iteration</p>
</li></ul>



<h3>References</h3>

<p>1. Song, J., Carey, M., Zhu, H., Miao, H., Ram´ırez, J. C., &amp; Wu, H. (2018). Identifying the dynamic gene regulatory network during latent HIV-1 reactivation using high-dimensional ordinary differential equations. International Journal of Computational Biology and Drug Design, 11,135-153. doi: 10.1504/IJCBDD.2018.10011910.
2. Wu, S., &amp; Wu, H. (2013). More powerful significant testing for time course gene expression data using functional principal component analysis approaches. BMC Bioinformatics, 14:6.
3. Carey, M., Wu, S., Gan, G. &amp; Wu, H. (2016). Correlation-based iterative clustering methods for time course data: The identification of temporal gene response modules for influenza infection in humans. Infectious Disease Modeling, 1, 28-39.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This is an example not using the permutation approach

opioid_data_noNA &lt;- opioidData[complete.cases(opioidData), ] #remove NAs

mydata &lt;- as.matrix(opioid_data_noNA[1:500,4:18])

testchange_results &lt;- testchange(data=mydata,perm=FALSE,time=seq(1,15,1))

data_change &lt;- testchange_results$sig.change

clustering_results &lt;- ihclust(data=data_change, smooth = TRUE,

cor_criteria = 0.75, max_iteration = 100, verbose = TRUE)
</code></pre>

<hr>
<h2 id='opioidData'>Opioid Dispensing Rates</h2><span id='topic+opioidData'></span>

<h3>Description</h3>

<p>A dataset containing estimated opioid dispensing rate per 100 persons in United States, 2006-2020.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(opioidData)
</code></pre>


<h3>Format</h3>

<p>data.frame; columns: fips = FIPS county code, State = State, County = County,
X2006-X2020 = estimated opioid dispensing rate per 100 persons in each year
</p>


<h3>Source</h3>

<p><a href="https://www.cdc.gov/drugoverdose/rxrate-maps/index.html">https://www.cdc.gov/drugoverdose/rxrate-maps/index.html</a>
</p>

<hr>
<h2 id='simcurve'>simcurve</h2><span id='topic+simcurve'></span>

<h3>Description</h3>

<p>This function generates two kinds of datasets.
1. Randomly generates curves with change/no change.
2. Generates true curves assumed from fixed coeffecients with some random noise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simcurve(numareas = c(300, 300, 300), p = 0.05, type, normerr = 0.1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simcurve_+3A_numareas">numareas</code></td>
<td>
<p>number of areas to generate</p>
</td></tr>
<tr><td><code id="simcurve_+3A_p">p</code></td>
<td>
<p>proportion of the areas that have significant change</p>
</td></tr>
<tr><td><code id="simcurve_+3A_type">type</code></td>
<td>
<p>type of curves generated</p>
</td></tr>
<tr><td><code id="simcurve_+3A_normerr">normerr</code></td>
<td>
<p>standard deviation of the Normal distribution (with mean zero) of which the coefficients are generated</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If type = &quot;random&quot;, the function generates curves with change/no change.
If type = &quot;fixed&quot;, the function generates true curves assumed from fixed coefficients with some random noise.
If numareas is not specified, it is assumed as a vector of c(300,300,300).
If normerr is not specified, it is assumed as a value of 0.01. It is ignored when type= &quot;random&quot;.
</p>


<h3>Value</h3>

<p>Output from the function is a list of two items:
</p>

<ul>
<li><p> data - simulated data
</p>
</li>
<li><p> parameters - parameters used to generate the data</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>mydata_ran &lt;- simcurve(numareas = c(300, 300, 300), p=0.01, type="random")

mydata_fixed &lt;- simcurve(numareas = c(300, 300, 300), p=0.01, type="fixed", normerr = 0.1)
</code></pre>

<hr>
<h2 id='testchange'>testchange</h2><span id='topic+testchange'></span>

<h3>Description</h3>

<p>This function identifies geographic areas with significant change over time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testchange(data, time, perm = FALSE, nperm = 100, numclust = 4, topF = 500)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="testchange_+3A_data">data</code></td>
<td>
<p>a numeric matrix, each row representing a time-series
and each column representing a time point</p>
</td></tr>
<tr><td><code id="testchange_+3A_time">time</code></td>
<td>
<p>defines the time sequence</p>
</td></tr>
<tr><td><code id="testchange_+3A_perm">perm</code></td>
<td>
<p>if perm = 'TRUE', a permutation is performed</p>
</td></tr>
<tr><td><code id="testchange_+3A_nperm">nperm</code></td>
<td>
<p>number of permuations</p>
</td></tr>
<tr><td><code id="testchange_+3A_numclust">numclust</code></td>
<td>
<p>defines the number of clusters for the parallel processing</p>
</td></tr>
<tr><td><code id="testchange_+3A_topf">topF</code></td>
<td>
<p>number of top F values to be selected when perm = 'FALSE'</p>
</td></tr>
</table>


<h3>Details</h3>

<p>number of permutations of &gt;=10,000 is ideal
</p>


<h3>Value</h3>

<p>Output if perm = 'TRUE' is a list of three items:
</p>

<ul>
<li><p> perm.F - F values obtained from permutation tests
</p>
</li>
<li><p> p.values - p-values obtained from permutation tests
</p>
</li>
<li><p> p.adjusted - p-values adjusted by Benjamini-Hochberg method
</p>
</li></ul>

<p>Output if perm = 'False' is a list of three items:
</p>

<ul>
<li><p> obs.F - conventional F-statistic values
</p>
</li>
<li><p> sig.change - areas with significant change over time pattern selected by top F-statistic values
</p>
</li>
<li><p> sel.F - top F-statistic values selected
</p>
</li></ul>



<h3>References</h3>

<p>1. Song, J., Carey, M., Zhu, H., Miao, H., Ram´ırez, J. C., &amp; Wu, H. (2018). Identifying the dynamic gene regulatory network during latent HIV-1 reactivation using high-dimensional ordinary differential equations. International Journal of Computational Biology and Drug Design, 11,135-153. doi: 10.1504/IJCBDD.2018.10011910.
2. Wu, S., &amp; Wu, H. (2013). More powerful significant testing for time course gene expression data using functional principal component analysis approaches. BMC Bioinformatics, 14:6.
3. Carey, M., Wu, S., Gan, G. &amp; Wu, H. (2016). Correlation-based iterative clustering methods for time course data: The identification of temporal gene response modules for influenza infection in humans. Infectious Disease Modeling, 1, 28-39.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># This is an example not using the permutation approach

opioid_data_noNA &lt;- opioidData[complete.cases(opioidData), ] #remove NAs

mydata &lt;- as.matrix(opioid_data_noNA[,4:18])

testchange_results &lt;- testchange(data=mydata,perm=FALSE,time=seq(1,15,1))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
