<!DOCTYPE html><html><head><title>Help for package iimi</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iimi}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#convert_bam_to_rle'><p>convert_bam_to_rle</p></a></li>
<li><a href='#convert_rle_to_df'><p>convert_rle_to_df</p></a></li>
<li><a href='#create_high_nucleotide_content'><p>create_high_nucleotide_content</p></a></li>
<li><a href='#create_mappability_profile'><p>create_mappability_profile</p></a></li>
<li><a href='#example_cov'><p>Coverage profiles of three plant samples.</p></a></li>
<li><a href='#example_diag'><p>Known diagnostics result of virus segments</p></a></li>
<li><a href='#nucleotide_info'><p>Nucleotide information of virus segments</p></a></li>
<li><a href='#plot_cov'><p>plot_cov()</p></a></li>
<li><a href='#predict_iimi'><p>predict_iimi()</p></a></li>
<li><a href='#train_iimi'><p>train_iimi()</p></a></li>
<li><a href='#trained_en'><p>A trained model using the default Elastic Net settings</p></a></li>
<li><a href='#trained_rf'><p>A trained model using the default Random Forest settings</p></a></li>
<li><a href='#trained_xgb'><p>A trained model using the default XGBoost settings</p></a></li>
<li><a href='#unreliable_regions'><p>The unreliable regions of the virus segments</p></a></li>
<li><a href='#virus_segments'><p>A DNAStringSet of virus segments from the Virtool virus data base (version 1.4.0)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Identifying Infection with Machine Intelligence</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Description:</td>
<td>A novel machine learning method for plant viruses diagnostic using 
    genome sequencing data. This package includes three different machine 
    learning models, random forest, XGBoost, and elastic net, to train and 
    predict mapped genome samples. Mappability profile and unreliable regions 
    are introduced to the algorithm, and users can build a mappability profile 
    from scratch with functions included in the package. Plotting mapped sample 
    coverage information is provided.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>GenomicAlignments, IRanges, Rsamtools, data.table, mltools,
randomForest, xgboost, Biostrings, stats, MTPS, R.utils, caret,
stringr, dplyr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-05 19:46:15 UTC; dsain</td>
</tr>
<tr>
<td>Author:</td>
<td>Haochen Ning [aut],
  Ian Boyes [aut],
  Ibrahim NumanagiÄ‡ <a href="https://orcid.org/0000-0002-2970-7937"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Michael Rott [aut],
  Li Xing <a href="https://orcid.org/0000-0002-4186-7909"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Xuekui Zhang <a href="https://orcid.org/0000-0003-4728-2343"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xuekui Zhang &lt;xuekui@uvic.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-07 10:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='convert_bam_to_rle'>convert_bam_to_rle</h2><span id='topic+convert_bam_to_rle'></span>

<h3>Description</h3>

<p>Converts one or more indexed and sorted BAM files into a run-
length encodings (RLEs) list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_bam_to_rle(bam_file, paired = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_bam_to_rle_+3A_bam_file">bam_file</code></td>
<td>
<p>path to BAM file(s).</p>
</td></tr>
<tr><td><code id="convert_bam_to_rle_+3A_paired">paired</code></td>
<td>
<p>Indicate if the sequencing paired is single-end or paired-end
reads. <code>TRUE</code> if paired-end. <code>FALSE</code> if single-end.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of coverage profile(s) in RLE format with one or more samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Please change the path to your folder where you
## store sorted and indexed BAM files of mapped samples

rles &lt;- convert_bam_to_rle("path/to/bam/file")

## End(Not run)

</code></pre>

<hr>
<h2 id='convert_rle_to_df'>convert_rle_to_df</h2><span id='topic+convert_rle_to_df'></span>

<h3>Description</h3>

<p>Converts a list of run-length encodings (RLEs) into a data frame
with 16 features after mappability profiling and nucleotide filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_rle_to_df(
  covs,
  unreliable_region_df = unreliable_regions,
  unreliable_region_enabled = TRUE,
  additional_nucleotide_info = data.frame()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_rle_to_df_+3A_covs">covs</code></td>
<td>
<p>A list of Coverage profile(s) in RLE format. Can be one or more
samples.</p>
</td></tr>
<tr><td><code id="convert_rle_to_df_+3A_unreliable_region_df">unreliable_region_df</code></td>
<td>
<p>The unreliable regions of the virus segments.
Default is <code>unreliable_regions</code>. It includes the mappability profile from
a host genome (we only have Arabidopsis thaliana right now) and virus
references, and the regions that have CG% and A% over 60% and 45%
respectively.</p>
</td></tr>
<tr><td><code id="convert_rle_to_df_+3A_unreliable_region_enabled">unreliable_region_enabled</code></td>
<td>
<p>Default is <code>TRUE</code>. If <code>TRUE</code>, the input will be
checked against <code>unreliable_region_df</code>. If <code>FALSE</code>, this step will be
skipped.</p>
</td></tr>
<tr><td><code id="convert_rle_to_df_+3A_additional_nucleotide_info">additional_nucleotide_info</code></td>
<td>
<p>Additional nucleotide information for virus
segments that are not included in <code>nucleotide_info</code>. The information
provided must be a data frame that follows the format of
<code>nucleotide_info</code>. Default is an empty data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame object that contains the mapping result for each virus
segment that the plant sample reads are aligned to and a RLE list of
coverage information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
df &lt;- convert_rle_to_df(example_cov)

## End(Not run)

</code></pre>

<hr>
<h2 id='create_high_nucleotide_content'>create_high_nucleotide_content</h2><span id='topic+create_high_nucleotide_content'></span>

<h3>Description</h3>

<p>Creates a data frame of the start and end positions of the
regions_a that are considered high in A% and GC%.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_high_nucleotide_content(gc = 0.6, a = 0.45, window = 75)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_high_nucleotide_content_+3A_gc">gc</code></td>
<td>
<p>The threshold for GC content. It is the proportion of G and C
nucleotides in a sliding window. Default is 0.6.</p>
</td></tr>
<tr><td><code id="create_high_nucleotide_content_+3A_a">a</code></td>
<td>
<p>The threshold for A nucleotide. It is the proportion of A nucleotide
in a sliding window. Default is 0.45.</p>
</td></tr>
<tr><td><code id="create_high_nucleotide_content_+3A_window">window</code></td>
<td>
<p>The sliding window size of your choice. Default is 75.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of the start and end positions of the regions_a that are
considered high in A% and GC%.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: high_nucleotides_regions &lt;- create_high_nucleotide_content()

</code></pre>

<hr>
<h2 id='create_mappability_profile'>create_mappability_profile</h2><span id='topic+create_mappability_profile'></span>

<h3>Description</h3>

<p>Creates a data frame of start and end positions of the regions
that are considered unmappable. Unmappable areas indicate that they can
be mapped to another virus segment or a host genome. Note that we only
have Arabidopsis Thaliana as a host.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mappability_profile(path_to_bam_files, category, window = 75)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_mappability_profile_+3A_path_to_bam_files">path_to_bam_files</code></td>
<td>
<p>Path to the folder that stores the indexed and
sorted BAM file(s).</p>
</td></tr>
<tr><td><code id="create_mappability_profile_+3A_category">category</code></td>
<td>
<p>Type of unreliable region you are creating. You can use
categories in the provided <code>unreliable_regions</code> data frame or customize
in your own way.</p>
</td></tr>
<tr><td><code id="create_mappability_profile_+3A_window">window</code></td>
<td>
<p>The sliding window size of your choice. Default is 75.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of start and end positions of the regions that are
considered unmappable.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## Please change the path to your folder where you store the mapped viruses
mappability_profile &lt;- create_mappability_profile("path/to/folder",
  category = "Unmappable regions")

## End(Not run)


</code></pre>

<hr>
<h2 id='example_cov'>Coverage profiles of three plant samples.</h2><span id='topic+example_cov'></span>

<h3>Description</h3>

<p>A list of coverage profiles for three plant samples. This is only a toy sample.
You can use it for running the examples in the vignette. We recommend using
more data to train the model, the more the better.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_cov
</code></pre>


<h3>Format</h3>

<p>A list of 3 run length encoding (RLE) lists for 3 plant samples.
Each RLE list has the RLE vector of a virus segment
</p>

<hr>
<h2 id='example_diag'>Known diagnostics result of virus segments</h2><span id='topic+example_diag'></span>

<h3>Description</h3>

<p>A matrix containing the known truth about the diagnostics result for each
plant sample for the example data. It records whether the sample is infected
with a virus segment. Each column is a sample, and each row is a virus
segment's diagnostics status for three samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_diag
</code></pre>


<h3>Format</h3>

<p>A matrix with 3 columns:
</p>

<dl>
<dt>S1</dt><dd><p>Sample one</p>
</dd>
<dt>S2</dt><dd><p>Sample two</p>
</dd>
<dt>S3</dt><dd><p>Sample three</p>
</dd>
</dl>


<hr>
<h2 id='nucleotide_info'>Nucleotide information of virus segments</h2><span id='topic+nucleotide_info'></span>

<h3>Description</h3>

<p>A data set containing the GC content and other information about the virus
segments from the official Virtool virus data base (version 1.4.0). The
variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nucleotide_info
</code></pre>


<h3>Format</h3>

<p>A data frame with 7 variables:
</p>

<dl>
<dt>virus_name</dt><dd><p>The virus name</p>
</dd>
<dt>iso_id</dt><dd><p>The virus isolate ID</p>
</dd>
<dt>seg_id</dt><dd><p>The virus segment ID</p>
</dd>
<dt>A_percent</dt><dd><p>The percentage of A nucleotides in the virus segment</p>
</dd>
<dt>C_percent</dt><dd><p>The percentage of C nucleotides in the virus segment</p>
</dd>
<dt>T_percent</dt><dd><p>The percentage of T nucleotides in the virus segment</p>
</dd>
<dt>GC_percent</dt><dd><p>The percentage of G and C nucleotides in the virus
segment (GC content)</p>
</dd>
<dt>seg_len</dt><dd><p>The length of the virus segment</p>
</dd>
</dl>


<hr>
<h2 id='plot_cov'>plot_cov()</h2><span id='topic+plot_cov'></span>

<h3>Description</h3>

<p>Plots the coverage profile of the mapped plant sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cov(
  covs,
  legend_status = TRUE,
  nucleotide_status = TRUE,
  window = 75,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_cov_+3A_covs">covs</code></td>
<td>
<p>An RLE list of coverage information of one or more plant samples.</p>
</td></tr>
<tr><td><code id="plot_cov_+3A_legend_status">legend_status</code></td>
<td>
<p>Whether display legend. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_cov_+3A_nucleotide_status">nucleotide_status</code></td>
<td>
<p>Whether display a sliding window of A percentage and
CG content. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_cov_+3A_window">window</code></td>
<td>
<p>The sliding window size. Default is 75.</p>
</td></tr>
<tr><td><code id="plot_cov_+3A_...">...</code></td>
<td>
<p>Other arguments that can be passed to <code>plot</code>,
<code>lines</code>, or <code>legend</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The coverage profile of the mapped plant sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_cov(example_cov$S1)



</code></pre>

<hr>
<h2 id='predict_iimi'>predict_iimi()</h2><span id='topic+predict_iimi'></span>

<h3>Description</h3>

<p>Uses a machine learning model to predict the infection status
for the plant sample(s). User can use their own model if needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_iimi(newdata, method = "xgb", trained_model, report_virus_level = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_iimi_+3A_newdata">newdata</code></td>
<td>
<p>A matrix or data frame that contains the features extracted
from the coverage profile using <code>convert_bam_to_cov()</code>.</p>
</td></tr>
<tr><td><code id="predict_iimi_+3A_method">method</code></td>
<td>
<p>The machine learning method of choice, <code>rf</code>,
<code>xgb</code>, or <code>en</code>. <code>rf</code> stands for random forest model; <code>xgb</code> stands for
XGBoost model; and <code>en</code> stands for elastic net model.</p>
</td></tr>
<tr><td><code id="predict_iimi_+3A_trained_model">trained_model</code></td>
<td>
<p>The trained model. If not provided, default model is
used.</p>
</td></tr>
<tr><td><code id="predict_iimi_+3A_report_virus_level">report_virus_level</code></td>
<td>
<p>If <code>TRUE</code>, the function returns the aggregated
results based on the virus. If <code>FALSE</code>, the function returns the
unaggregated results based on segment level with each decision's
probability decided by the model. We do not recommended to set this to
<code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of diagnostics result for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: df &lt;- convert_rle_to_df(example_cov)
predictions &lt;- predict_iimi(df)

## End(Not run)



</code></pre>

<hr>
<h2 id='train_iimi'>train_iimi()</h2><span id='topic+train_iimi'></span>

<h3>Description</h3>

<p>Trains a <code>XGBoost</code> (default), <code style="white-space: pre;">&#8288;Random Forest&#8288;</code>, or <code style="white-space: pre;">&#8288;Elastic Net&#8288;</code>
model using user-provided data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>train_iimi(
  train_x,
  train_y,
  method = "xgb",
  nrounds = 100,
  max_depth = 10,
  gamma = 6,
  ntree = 100,
  k = 5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="train_iimi_+3A_train_x">train_x</code></td>
<td>
<p>A data frame or a matrix of predictors.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_train_y">train_y</code></td>
<td>
<p>A response vector of labels (needs to be a factor).</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_method">method</code></td>
<td>
<p>The machine learning method of choice, <code style="white-space: pre;">&#8288;Random Forest&#8288;</code> or
<code>XGBoost</code>, or <code style="white-space: pre;">&#8288;Elastic Net&#8288;</code> model. Default is <code>XGBoost</code> model.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_nrounds">nrounds</code></td>
<td>
<p>Max number of boosting iterations for <code>XGBoost</code> model. Default
is 100.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_max_depth">max_depth</code></td>
<td>
<p>Maximum depth of a tree in <code>XGBoost</code> model. Default is 10.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_gamma">gamma</code></td>
<td>
<p>Minimum loss reduction required in <code>XGBoost</code> model. Default is 6.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_ntree">ntree</code></td>
<td>
<p>Number of trees in <code style="white-space: pre;">&#8288;Random Forest&#8288;</code> model. Default is 100.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_k">k</code></td>
<td>
<p>Number of folds. Default is 5.</p>
</td></tr>
<tr><td><code id="train_iimi_+3A_...">...</code></td>
<td>
<p>Other arguments that can be passed to <code>randomForest</code>,
<code>xgboost</code>, or <code>glmnet</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code style="white-space: pre;">&#8288;Random Forest&#8288;</code>, <code>XGBoost</code>, <code style="white-space: pre;">&#8288;Elastic Net&#8288;</code> model
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
df &lt;- convert_rle_to_df(example_cov)
train_x &lt;- df[,-c(1:4)]
train_y = c()
for (ii in 1:nrow(df)) {
  seg_id = df$seg_id[ii]
  sample_id = df$sample_id[ii]
  train_y = c(train_y, example_diag[seg_id, sample_id])
}
trained_model &lt;- train_iimi(train_x = train_x, train_y = train_y)

## End(Not run)




</code></pre>

<hr>
<h2 id='trained_en'>A trained model using the default Elastic Net settings</h2><span id='topic+trained_en'></span>

<h3>Description</h3>

<p>A trained model using the default Elastic Net settings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trained_en
</code></pre>


<h3>Format</h3>

<p>An object of class <code>cv.glmnet</code> of length 13.
</p>

<hr>
<h2 id='trained_rf'>A trained model using the default Random Forest settings</h2><span id='topic+trained_rf'></span>

<h3>Description</h3>

<p>A trained model using the default Random Forest settings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trained_rf
</code></pre>


<h3>Format</h3>

<p>An object of class <code>randomForest.formula</code> (inherits from <code>randomForest</code>) of length 15.
</p>

<hr>
<h2 id='trained_xgb'>A trained model using the default XGBoost settings</h2><span id='topic+trained_xgb'></span>

<h3>Description</h3>

<p>A trained model using the default XGBoost settings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trained_xgb
</code></pre>


<h3>Format</h3>

<p>An object of class <code>xgb.Booster</code> of length 9.
</p>

<hr>
<h2 id='unreliable_regions'>The unreliable regions of the virus segments</h2><span id='topic+unreliable_regions'></span>

<h3>Description</h3>

<p>A data frame of unmappable regions and regions of CG% and A% over 60% and 45%
respectively for the virus segments. It is worth to note that if a
virus segment does not have any unreliable regions, that virus segment is not
shown in this data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unreliable_regions
</code></pre>


<h3>Format</h3>

<p>A data frame of unreliable regions in the run-length encoding format
for virus segments.
</p>

<dl>
<dt>Start</dt><dd><p>The start position of the region that is considered unreliable</p>
</dd>
<dt>End</dt><dd><p>The end position of the region that is considered unreliable</p>
</dd>
<dt>Virus segment</dt><dd><p>The virus segment ID</p>
</dd>
<dt>Categories</dt><dd><p>The category that this unrelible region belong to, which
are Unmappable regions (host), Unmappable regions (virus), CG% &gt; 60%, A% &gt; 45%.</p>
</dd>
</dl>


<hr>
<h2 id='virus_segments'>A DNAStringSet of virus segments from the Virtool virus data base (version 1.4.0)</h2><span id='topic+virus_segments'></span>

<h3>Description</h3>

<p>A DNAStringSet of virus segments from the Virtool virus data base (version 1.4.0)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>virus_segments
</code></pre>


<h3>Format</h3>

<p>An object of class <code>DNAStringSet</code> of length 3138.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
