<!DOCTYPE html><html><head><title>Help for package IMIX</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IMIX}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#data_p'><p>P value matrix of two data types</p></a></li>
<li><a href='#FDR_control_adaptive'><p>The Adaptive Procedure for Across-Data-Type FDR Control</p></a></li>
<li><a href='#FDR_control_adaptive_imix'><p>The Adaptive Procedure for Across-Data-Type FDR Control for IMIX Output</p></a></li>
<li><a href='#IMIX'><p>IMIX</p></a></li>
<li><a href='#IMIX_cor'><p>IMIX-Cor</p></a></li>
<li><a href='#IMIX_cor_restrict'><p>IMIX-Cor-Restrict</p></a></li>
<li><a href='#IMIX_cor_twostep'><p>IMIX-Cor-Twostep</p></a></li>
<li><a href='#IMIX_ind'><p>IMIX-ind</p></a></li>
<li><a href='#model_selection'><p>Model Selection</p></a></li>
<li><a href='#model_selection_component'><p>Component Selection</p></a></li>
<li><a href='#plot_component'><p>Plot the AIC or BIC Values for Model Selection</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.5</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-07-13</td>
</tr>
<tr>
<td>Title:</td>
<td>Gaussian Mixture Model for Multi-Omics Data Integration</td>
</tr>
<tr>
<td>Description:</td>
<td>A multivariate Gaussian mixture model framework to integrate multiple types of genomic data and allow modeling of inter-data-type correlations for association analysis. 'IMIX' can be implemented to test whether a disease is associated with genes in multiple genomic data types, such as DNA methylation, copy number variation, gene expression, etc. It can also study the integration of multiple pathways. 'IMIX' uses the summary statistics of association test outputs and conduct integration analysis for two or three types of genomics data. 'IMIX' features statistically-principled model selection, global FDR control and computational efficiency. Details are described in Ziqiao Wang and Peng Wei (2020) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtaa1001">doi:10.1093/bioinformatics/btaa1001</a>&gt;. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>crayon,mvtnorm,mixtools,mclust,ggplot2,stats,utils,MASS</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ziqiaow/IMIX">https://github.com/ziqiaow/IMIX</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ziqiaow/IMIX/issues">https://github.com/ziqiaow/IMIX/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Ziqiao Wang <a href="https://orcid.org/0000-0003-3383-8670"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Peng Wei <a href="https://orcid.org/0000-0001-7758-6116"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ziqiao Wang &lt;wzqjanet@gmail.com&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-07-13 21:56:56 UTC; l014195</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-07-13 22:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='data_p'>P value matrix of two data types</h2><span id='topic+data_p'></span>

<h3>Description</h3>

<p>A dataset with summary statistics p values of 1000 genes for RNAseq and CNV data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_p)
</code></pre>


<h3>Format</h3>

<p>A data matrix with 1000 rows and 2 variables:
</p>

<dl>
<dt>p.rnaseq</dt><dd><p>P values of data type 1 for all genes</p>
</dd>
<dt>p.cnv</dt><dd><p>P values of data type 2 for all genes</p>
</dd>
</dl>


<hr>
<h2 id='FDR_control_adaptive'>The Adaptive Procedure for Across-Data-Type FDR Control</h2><span id='topic+FDR_control_adaptive'></span>

<h3>Description</h3>

<p>The adaptive procedure for across-data-type FDR control based on the output from IMIX models, this can be directly performed by IMIX function, however, if the user is interested in other mixture models, alpha level or combinations of components, this function would be useful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FDR_control_adaptive(lfdr, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FDR_control_adaptive_+3A_lfdr">lfdr</code></td>
<td>
<p>Local FDR for each gene of the mixture model results for one component or a combination of components</p>
</td></tr>
<tr><td><code id="FDR_control_adaptive_+3A_alpha">alpha</code></td>
<td>
<p>Prespecified FDR control level</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimated mFDR for the target component or component combinaitons and whether the genes is classified in this component/combination after FDR control at alpha level, 1 is yes, 0 is no.
</p>
<table>
<tr><td><code>significant_genes_with_FDRcontrol</code></td>
<td>
<p>The output of each gene ordered by the components based on FDR control and within each component ordered by the local FDR, &quot;localFDR&quot; is 1-posterior probability of each gene in the component based on the maximum posterior probability, &quot;class_withoutFDRcontrol&quot; is the classified component based on maximum posterior probability, &quot;class_FDRcontrol&quot; is the classified component based on the across-data-type FDR control at alpha level</p>
</td></tr>
<tr><td><code>estimatedFDR</code></td>
<td>
<p>The estimated marginal FDR value for each component starting from component 2 (component 1 is the global null)</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Prespecified nominal level for the across-data-type FDR control</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# First load the data
data("data_p")

# Specify inititial values (this step could be omitted)
mu_input &lt;- c(0,3,0,3)
sigma_input &lt;- rep(1,4)
p_input &lt;- rep(0.5,4)
test1 &lt;- IMIX(data_input = data_p,data_type = "p",mu_ini = mu_input,sigma_ini = sigma_input,
p_ini = p_input,alpha = 0.1,model_selection_method = "AIC")

# Check the selected model based on AIC value
test1$`Selected Model`

# Below is an example for data example 1 in controlling 
# the FDR at 0.2 for component 2 &amp; component 4. 
# First calculate the local FDR for component 2 &amp; component 4:
lfdr_ge_combined &lt;- 1 - (test1$IMIX_cor_twostep$`posterior prob`[,2] + 
test1$IMIX_cor_twostep$`posterior prob`[,4])  # Class 2: (ge+,cnv-); class 4: (ge+,cnv+)
names(lfdr_ge_combined) &lt;- rownames(test1$IMIX_cor_twostep$`posterior prob`)

# Perform the across-data-type FDR control for component 2 &amp; component 4 at alpha level 0.2
fdr_control1 &lt;- FDR_control_adaptive(lfdr = lfdr_ge_combined, alpha = 0.2)

</code></pre>

<hr>
<h2 id='FDR_control_adaptive_imix'>The Adaptive Procedure for Across-Data-Type FDR Control for IMIX Output</h2><span id='topic+FDR_control_adaptive_imix'></span>

<h3>Description</h3>

<p>The adaptive procedure for across-data-type FDR control based on the output from IMIX models, this can be directly performed by IMIX function, however, if the user is interested in other alpha levels, this function would be useful to avoid rerun the IMIX().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FDR_control_adaptive_imix(
  imix_output,
  model = c("IMIX_ind", "IMIX_cor_twostep", "IMIX_cor_restrict", "IMIX_cor"),
  alpha
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FDR_control_adaptive_imix_+3A_imix_output">imix_output</code></td>
<td>
<p>The result output from IMIX() function, result controlled at alpha level only for one component each time.</p>
</td></tr>
<tr><td><code id="FDR_control_adaptive_imix_+3A_model">model</code></td>
<td>
<p>The target model among &quot;IMIX_ind&quot;,&quot;IMIX_cor_twostep&quot;,&quot;IMIX_cor_restrict&quot;, and &quot;IMIX_cor&quot;. Default is IMIX_ind.</p>
</td></tr>
<tr><td><code id="FDR_control_adaptive_imix_+3A_alpha">alpha</code></td>
<td>
<p>Prespecified FDR control level.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimated mFDR for the target component and classify the genes in each component after FDR control at alpha level.
</p>
<table>
<tr><td><code>significant_genes_with_FDRcontrol</code></td>
<td>
<p>The output of each gene ordered by the components based on FDR control and within each component ordered by the local FDR, &quot;localFDR&quot; is 1-posterior probability of each gene in the component based on the maximum posterior probability, &quot;class_withoutFDRcontrol&quot; is the classified component based on maximum posterior probability, &quot;class_FDRcontrol&quot; is the classified component based on the across-data-type FDR control at alpha level</p>
</td></tr>
<tr><td><code>estimatedFDR</code></td>
<td>
<p>The estimated marginal FDR value for each component starting from component 2 (component 1 is the global null)</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Prespecified nominal level for the across-data-type FDR control</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# First generate the data
library(MASS)
N &lt;- 1000
truelabel &lt;- sample(1:8,prob = rep(0.125, 8),size = N,replace = TRUE)
mu1 &lt;- c(0, 5);mu2 &lt;- c(0, 5);mu3 &lt;- c(0, 5)
mu1_mv &lt;- c(mu1[1], mu2[1], mu3[1]);mu2_mv &lt;- c(mu1[2], mu2[1], mu3[1]);
mu3_mv &lt;- c(mu1[1], mu2[2], mu3[1]);mu4_mv &lt;- c(mu1[1], mu2[1], mu3[2]);
mu5_mv &lt;- c(mu1[2], mu2[2], mu3[1]);mu6_mv &lt;- c(mu1[2], mu2[1], mu3[2])
mu7_mv &lt;- c(mu1[1], mu2[2], mu3[2]);mu8_mv &lt;- c(mu1[2], mu2[2], mu3[2])
cov_sim &lt;- list()
for (i in 1:8) {
  cov_sim[[i]] &lt;- diag(3)
  }
data_z &lt;- array(0, c(N, 3))
data_z[which(truelabel == 1),] &lt;- mvrnorm(n = length(which(truelabel == 1)),
mu = mu1_mv,Sigma = cov_sim[[1]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 2),] &lt;- mvrnorm(n = length(which(truelabel == 2)),
mu = mu2_mv,Sigma = cov_sim[[2]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 3),] &lt;- mvrnorm(n = length(which(truelabel == 3)),
mu = mu3_mv,Sigma = cov_sim[[3]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 4),] &lt;- mvrnorm(n = length(which(truelabel == 4)),
mu = mu4_mv,Sigma = cov_sim[[4]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 5),] &lt;- mvrnorm(n = length(which(truelabel == 5)),
mu = mu5_mv,Sigma = cov_sim[[5]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 6),] &lt;- mvrnorm(n = length(which(truelabel == 6)),
mu = mu6_mv,Sigma = cov_sim[[6]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 7),] &lt;- mvrnorm(n = length(which(truelabel == 7)),
mu = mu7_mv,Sigma = cov_sim[[7]],tol = 1e-6,empirical = FALSE)
data_z[which(truelabel == 8),] &lt;- mvrnorm(n = length(which(truelabel == 8)),
mu = mu8_mv,Sigma = cov_sim[[8]],tol = 1e-6,empirical = FALSE)
rownames(data_z) &lt;- paste0("gene", 1:N)
colnames(data_z) &lt;- c("z.methy", "z.ge", "z.cnv")
dim(data_z)

# Fit the model
test2 &lt;- IMIX(data_input = data_z,data_type = "z",alpha = 0.05,verbose = TRUE)

# Adaptive FDR control at alpha 0.2 for IMIX_cor model
fdr_control2 &lt;- FDR_control_adaptive_imix(imix_output = test2, model = "IMIX_cor", 
alpha = 0.2) 

</code></pre>

<hr>
<h2 id='IMIX'>IMIX</h2><span id='topic+IMIX'></span>

<h3>Description</h3>

<p>Fitting a multivariate mixture model framework, model selection for the best model, and adaptive procedure for FDR control. Input of summary statistics z scores or p values of two or three data types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IMIX(
  data_input,
  data_type = c("p", "z"),
  mu_ini = NULL,
  sigma_ini = NULL,
  p_ini = NULL,
  tol = 1e-06,
  maxiter = 1000,
  seed = 10,
  ini.ind = TRUE,
  model = c("all", "IMIX_ind", "IMIX_cor_twostep", "IMIX_cor_restrict", "IMIX_cor"),
  model_selection_method = c("BIC", "AIC"),
  alpha = 0.2,
  verbose = FALSE,
  sort_label = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IMIX_+3A_data_input">data_input</code></td>
<td>
<p>An n x d data frame or matrix of the summary statistics z score or p value, n is the nubmer of genes, d is the number of data types. Each row is a gene, each column is a data type.</p>
</td></tr>
<tr><td><code id="IMIX_+3A_data_type">data_type</code></td>
<td>
<p>Whether the input data is the p values or z scores, default is p value</p>
</td></tr>
<tr><td><code id="IMIX_+3A_mu_ini">mu_ini</code></td>
<td>
<p>Initial values for the mean of the independent mixture model distribution. A vector of length 2*d, d is number of data types. Needs to be in a special format: for example, if d=3, needs to be in the format of (null_1,alternative_1,null_2,alternative_2,null_3,alternative_3).</p>
</td></tr>
<tr><td><code id="IMIX_+3A_sigma_ini">sigma_ini</code></td>
<td>
<p>Initial values for the standard deviations of the two components in each data type. A vector of length 2*d, d is number of data types. Needs to be in a special format: for example, if d=3, needs to be in the format of (null_1,alternative_1,null_2,alternative_2,null_3,alternative_3).</p>
</td></tr>
<tr><td><code id="IMIX_+3A_p_ini">p_ini</code></td>
<td>
<p>Initial values for the proportion of the distribution of the two components in each data type. A vector of length 2*d, d is number of data types. Needs to be in a special format: for example, if d=3, needs to be in the format of (null_1,alternative_1,null_2,alternative_2,null_3,alternative_3).</p>
</td></tr>
<tr><td><code id="IMIX_+3A_tol">tol</code></td>
<td>
<p>The convergence criterion. Convergence is declared when the change in the observed data log-likelihood increases by less than epsilon.</p>
</td></tr>
<tr><td><code id="IMIX_+3A_maxiter">maxiter</code></td>
<td>
<p>The maximum number of iteration, default is 1000</p>
</td></tr>
<tr><td><code id="IMIX_+3A_seed">seed</code></td>
<td>
<p>Set.seed, default is 10</p>
</td></tr>
<tr><td><code id="IMIX_+3A_ini.ind">ini.ind</code></td>
<td>
<p>Use the parameters estimated from IMIX-ind for initial values of other IMIX models, default is TRUE</p>
</td></tr>
<tr><td><code id="IMIX_+3A_model">model</code></td>
<td>
<p>Which model to use to compute the data, default is all</p>
</td></tr>
<tr><td><code id="IMIX_+3A_model_selection_method">model_selection_method</code></td>
<td>
<p>Model selection information criteria, based on AIC or BIC, default is BIC</p>
</td></tr>
<tr><td><code id="IMIX_+3A_alpha">alpha</code></td>
<td>
<p>Prespecified nominal level for global FDR control, default is 0.2</p>
</td></tr>
<tr><td><code id="IMIX_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the full log-likelihood for each iteration, default is FALSE</p>
</td></tr>
<tr><td><code id="IMIX_+3A_sort_label">sort_label</code></td>
<td>
<p>Whether to sort the component labels in case component labels switched after convergence of the initial values, default is TRUE, notice that if the users chooose not to, they might need to check the interested IMIX model for the converged mean for the true component labels and perform the adaptive FDR control separately for an acurate result</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of results of IMIX
</p>
<table>
<tr><td><code>IMIX_ind</code></td>
<td>
<p>Results of IMIX_ind, assuming all data types are independent</p>
</td></tr>
<tr><td><code>IMIX_cor_twostep</code></td>
<td>
<p>Results of IMIX_cor_twostep, by default the mean is the estimated value of IMIX_ind. If the users are interested to use another mean input, they could directly use function IMIX_cor_twostep and specify the mean</p>
</td></tr>
<tr><td><code>IMIX_cor</code></td>
<td>
<p>Results of IMIX_cor</p>
</td></tr>
<tr><td><code>IMIX_cor_restrict</code></td>
<td>
<p>Results of IMIX_cor_restrict</p>
</td></tr>
<tr><td><code>AIC/BIC</code></td>
<td>
<p>The AIC and BIC values of all fitted models</p>
</td></tr>
<tr><td><code>Selected Model</code></td>
<td>
<p>The model with the smallest AIC or BIC value, this is determined by user specifications in the function input &quot;model_selection_method&quot;, by default is BIC</p>
</td></tr>
<tr><td><code>significant_genes_with_FDRcontrol</code></td>
<td>
<p>The output of each gene ordered by the components based on FDR control and within each component ordered by the local FDR, &quot;localFDR&quot; is 1-posterior probability of each gene in the component based on the maximum posterior probability, &quot;class_withoutFDRcontrol&quot; is the classified component based on maximum posterior probability, &quot;class_FDRcontrol&quot; is the classified component based on the across-data-type FDR control at alpha level</p>
</td></tr>
<tr><td><code>estimatedFDR</code></td>
<td>
<p>The estimated marginal FDR value for each component starting from component 2 (component 1 is the global null)</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>Prespecified nominal level for the across-data-type FDR control</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>
<p>Tatiana Benaglia, Didier Chauveau, David R. Hunter, and Derek Young. 2009. “mixtools: An R Package for Analyzing Finite Mixture Models.” Journal of Statistical Software 32 (6): 1–29. <a href="https://www.jstatsoft.org/v32/i06/">https://www.jstatsoft.org/v32/i06/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A toy example
data("data_p")
set.seed(10)
data &lt;- data_p[sample(1:1000,200,replace = FALSE),]
mu_input &lt;- c(0,3,0,3)
sigma_input &lt;- rep(1,4)
p_input &lt;- rep(0.5,4)
test &lt;- IMIX(data_input = data,data_type = "p",mu_ini = mu_input,sigma_ini = sigma_input,
             p_ini = p_input,alpha = 0.1,model_selection_method = "BIC",
             sort_label = FALSE,model = "IMIX_ind")


# The details of this example can be found in Github vignette
# First load the data
data("data_p")

# Specify initial values (this step could be omitted)
mu_input &lt;- c(0,3,0,3)
sigma_input &lt;- rep(1,4)
p_input &lt;- rep(0.5,4)

# Fit IMIX model
test1 &lt;- IMIX(data_input = data_p,data_type = "p",mu_ini = mu_input,sigma_ini = sigma_input,
p_ini = p_input,alpha = 0.1,model_selection_method = "AIC")

#Results
# Print the estimated across-data-type FDR for each component
test1$estimatedFDR

# The AIC and BIC values for each model
test1$`AIC/BIC` 

# The best fitted model selected by AIC
test1$`Selected Model` 

# The output of IMIX_cor_twostep
str(test1$IMIX_cor_twostep) 

# The output of genes with local FDR values and classified components
dim(test1$significant_genes_with_FDRcontrol)
head(test1$significant_genes_with_FDRcontrol)

</code></pre>

<hr>
<h2 id='IMIX_cor'>IMIX-Cor</h2><span id='topic+IMIX_cor'></span>

<h3>Description</h3>

<p>Fitting a correlated multivariate mixture model. Input of summary statistics z scores or p values of two or three data types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IMIX_cor(
  data_input,
  data_type = c("p", "z"),
  g = 8,
  mu_vec,
  cov,
  p,
  tol = 1e-06,
  maxiter = 1000,
  seed = 10,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IMIX_cor_+3A_data_input">data_input</code></td>
<td>
<p>An n x d data frame or matrix of the summary statistics z score or p value, n is the nubmer of genes, d is the number of data types. Each row is a gene, each column is a data type.</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_data_type">data_type</code></td>
<td>
<p>Whether the input data is the p values or z scores, default is p value</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_g">g</code></td>
<td>
<p>The number of components, default is 8 for three data types</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_mu_vec">mu_vec</code></td>
<td>
<p>A list of initial values for the mean vectors for each component. If there are three data types and 8 components, then the initial is a list of 8 mean vectors, each vector is of length 3.</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_cov">cov</code></td>
<td>
<p>A list of initial values for the covariance matrices. If there are three data types and 8 components, then the initial is a list of 8 covariance matrices, each matix is 3*3.</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_p">p</code></td>
<td>
<p>Initial value for the proportion of the distribution in the Gaussian mixture model</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_tol">tol</code></td>
<td>
<p>The convergence criterion. Convergence is declared when the change in the observed data log-likelihood increases by less than epsilon.</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_maxiter">maxiter</code></td>
<td>
<p>The maximum number of iteration, default is 1000</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_seed">seed</code></td>
<td>
<p>set.seed, default is 10</p>
</td></tr>
<tr><td><code id="IMIX_cor_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the full log-likelihood for each iteration, default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the results of IMIX-cor
</p>
<table>
<tr><td><code>posterior prob</code></td>
<td>
<p>Posterior probability matrix of each gene for each component</p>
</td></tr>
<tr><td><code>Full LogLik all</code></td>
<td>
<p>Full log-likelihood of each iteration</p>
</td></tr>
<tr><td><code>Full MaxLogLik final</code></td>
<td>
<p>The final log-likelihood of the converged model</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>Number of iterations run</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>Estimated proportion of each component, sum to 1</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A list of estimated mean vectors of each component for each data type, each list corresponds to one component</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>A list of estimated variance-covariance matrix of each component</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>Number of components</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>

<hr>
<h2 id='IMIX_cor_restrict'>IMIX-Cor-Restrict</h2><span id='topic+IMIX_cor_restrict'></span>

<h3>Description</h3>

<p>Fitting a correlated multivariate mixture model with restrictions on the mean. Input of summary statistics z scores or p values of two or three data types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IMIX_cor_restrict(
  data_input,
  data_type = c("p", "z"),
  mu,
  cov,
  p,
  tol = 1e-06,
  maxiter = 1000,
  seed = 10,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IMIX_cor_restrict_+3A_data_input">data_input</code></td>
<td>
<p>An n x d data frame or matrix of the summary statistics z score or p value, n is the nubmer of genes, d is the number of data types. Each row is a gene, each column is a data type.</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_data_type">data_type</code></td>
<td>
<p>Whether the input data is the p values or z scores, default is p value</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_mu">mu</code></td>
<td>
<p>Initial value for the mean of the independent mixture model distribution. A vector of length 2*d, d is number of data types. Needs to be in a special format that corresponds to the initial value of mu, for example, if d=3, needs to be in the format of (null_1,alternative_1,null_2,alternative_2,null_3,alternative_3).</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_cov">cov</code></td>
<td>
<p>A list of initial values for the covariance matrices. If there are three data types and 8 components, then the initial is a list of 8 covariance matrices, each matix is 3*3.</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_p">p</code></td>
<td>
<p>Initial value for the proportion of the distribution in the Gaussian mixture model</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_tol">tol</code></td>
<td>
<p>The convergence criterion. Convergence is declared when the change in the observed data log-likelihood increases by less than epsilon.</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_maxiter">maxiter</code></td>
<td>
<p>The maximum number of iteration, default is 1000</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_seed">seed</code></td>
<td>
<p>set.seed, default is 10</p>
</td></tr>
<tr><td><code id="IMIX_cor_restrict_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the full log-likelihood for each iteration, default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the results of IMIX-cor-restrict
</p>
<table>
<tr><td><code>posterior prob</code></td>
<td>
<p>Posterior probability of each gene for each component</p>
</td></tr>
<tr><td><code>Full LogLik all</code></td>
<td>
<p>Full log-likelihood of each iteration</p>
</td></tr>
<tr><td><code>Full MaxLogLik final</code></td>
<td>
<p>The final log-likelihood of the converged model</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>Number of iterations run</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>Estimated proportion of each component, sum to 1</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>Estimated mean for the null and alternative of each data type: for two data types (mu10,mu11,mu20,mu21), three data types (mu10,mu11,mu20,mu21,mu30,mu31), mui0 is the null for data type i, mui1 is the alternative for data type i.</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>A list of estimated variance-covariance matrix of each component</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>

<hr>
<h2 id='IMIX_cor_twostep'>IMIX-Cor-Twostep</h2><span id='topic+IMIX_cor_twostep'></span>

<h3>Description</h3>

<p>Fitting a correlated multivariate mixture model with fixed mean from estimated parameters of IMIX-ind. Input of summary statistics z scores or p values of two or three data types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IMIX_cor_twostep(
  data_input,
  data_type = c("p", "z"),
  g = 8,
  mu_vec,
  cov,
  p,
  tol = 1e-06,
  maxiter = 1000,
  seed = 10,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IMIX_cor_twostep_+3A_data_input">data_input</code></td>
<td>
<p>An n x d data frame or matrix of the summary statistics z score or p value, n is the nubmer of genes, d is the number of data types. Each row is a gene, each column is a data type.</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_data_type">data_type</code></td>
<td>
<p>Whether the input data is the p values or z scores, default is p value</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_g">g</code></td>
<td>
<p>The number of components, default is 8 for three data types</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_mu_vec">mu_vec</code></td>
<td>
<p>Input of the mean value output from IMIX-Ind result, a list of the mean vectors for each component.</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_cov">cov</code></td>
<td>
<p>A list of initial values for the covariance matrices. If there are three data types and 8 components, then the initial is a list of 8 covariance matrices, each matix is 3*3.</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_p">p</code></td>
<td>
<p>Initial value for the proportion of the distribution in the Gaussian mixture model</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_tol">tol</code></td>
<td>
<p>The convergence criterion. Convergence is declared when the change in the observed data log-likelihood increases by less than epsilon.</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_maxiter">maxiter</code></td>
<td>
<p>The maximum number of iteration, default is 1000</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_seed">seed</code></td>
<td>
<p>set.seed, default is 10</p>
</td></tr>
<tr><td><code id="IMIX_cor_twostep_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the full log-likelihood for each iteration, default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the results of IMIX-cor-twostep
</p>
<table>
<tr><td><code>posterior prob</code></td>
<td>
<p>Posterior probability matrix of each gene for each component</p>
</td></tr>
<tr><td><code>Full LogLik all</code></td>
<td>
<p>Full log-likelihood of each iteration</p>
</td></tr>
<tr><td><code>Full MaxLogLik final</code></td>
<td>
<p>The final log-likelihood of the converged model</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>Number of iterations run</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>Estimated proportion of each component, sum to 1</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>A list of mean vectors of each component for each data type, this is the prespecified mean</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>A list of estimated variance-covariance matrix of each component</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>Number of components</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>

<hr>
<h2 id='IMIX_ind'>IMIX-ind</h2><span id='topic+IMIX_ind'></span>

<h3>Description</h3>

<p>Fitting an independent mixture model with restrictions on mean and variance. Input of summary statistics z scores or p values of two or three data types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IMIX_ind(
  data_input,
  data_type = c("p", "z"),
  mu,
  sigma,
  p,
  tol = 1e-06,
  maxiter = 1000,
  seed = 10,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IMIX_ind_+3A_data_input">data_input</code></td>
<td>
<p>An n x d data frame or matrix of the summary statistics z score or p value, n is the nubmer of genes, d is the number of data types. Each row is a gene, each column is a data type.</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_data_type">data_type</code></td>
<td>
<p>Whether the input data is the p values or z scores, default is p value</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_mu">mu</code></td>
<td>
<p>Initial value for the mean of each component of the independent mixture model distribution. A vector of length 2*d, d is number of data types. Needs to be in a special format that corresponds to the initial value of mu, for example, if d=3, needs to be in the format of (null_1,alternative_1,null_2,alternative_2,null_3,alternative_3).</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_sigma">sigma</code></td>
<td>
<p>Initial value for the standard deviation of each component of the independent mixture model distribution. A vector of length 2*d, d is number of data types. Needs to be in a special format that corresponds to the initial value of mu, for example, if d=3, needs to be in the format of (null_1,alternative_1,null_2,alternative_2,null_3,alternative_3).</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_p">p</code></td>
<td>
<p>Initial value for the proportion of the distribution in the Gaussian mixture model</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_tol">tol</code></td>
<td>
<p>The convergence criterion. Convergence is declared when the change in the observed data log-likelihood increases by less than epsilon.</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_maxiter">maxiter</code></td>
<td>
<p>The maximum number of iteration, default is 1000</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_seed">seed</code></td>
<td>
<p>set.seed, default is 10</p>
</td></tr>
<tr><td><code id="IMIX_ind_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the full log-likelihood for each iteration, default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of the results of IMIX-ind
</p>
<table>
<tr><td><code>posterior prob</code></td>
<td>
<p>Posterior probability matrix of each gene for each component</p>
</td></tr>
<tr><td><code>Full LogLik all</code></td>
<td>
<p>Full log-likelihood of each iteration</p>
</td></tr>
<tr><td><code>Full MaxLogLik final</code></td>
<td>
<p>The final log-likelihood of the converged model</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>Number of iterations run</p>
</td></tr>
<tr><td><code>pi</code></td>
<td>
<p>Estimated proportion of each component, sum to 1</p>
</td></tr>
<tr><td><code>mu</code></td>
<td>
<p>Estimated mean for the null and alternative of each data type: for two data types (mu10,mu11,mu20,mu21), three data types (mu10,mu11,mu20,mu21,mu30,mu31), mui0 is the null for data type i, mui1 is the alternative for data type i.</p>
</td></tr>
<tr><td><code>sigma</code></td>
<td>
<p>Estimated standard deviation for the null and alternative of each data type: for two data types (sigma10,sigma11,sigma20,sigma21), three data types (sigma10,sigma11,sigma20,sigma21,sigma30,sigma31), sigmai0 is the null for data type i, sigmai1 is the alternative for data type i.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>

<hr>
<h2 id='model_selection'>Model Selection</h2><span id='topic+model_selection'></span>

<h3>Description</h3>

<p>Model selection for sub-model outputs in IMIX, this step is to calculate the AIC or BIC values for one model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model_selection(
  loglik,
  n,
  g = 4,
  d = 2,
  modelname = c("IMIX_ind", "IMIX_ind_unrestrict", "IMIX_cor_twostep", "IMIX_cor",
    "IMIX_cor_restrict")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model_selection_+3A_loglik">loglik</code></td>
<td>
<p>Full log likelihood, result output from IMIX or a sub model in IMIX: 'Full MaxLogLik final'</p>
</td></tr>
<tr><td><code id="model_selection_+3A_n">n</code></td>
<td>
<p>Total number of genes</p>
</td></tr>
<tr><td><code id="model_selection_+3A_g">g</code></td>
<td>
<p>Number of components</p>
</td></tr>
<tr><td><code id="model_selection_+3A_d">d</code></td>
<td>
<p>Number of data types</p>
</td></tr>
<tr><td><code id="model_selection_+3A_modelname">modelname</code></td>
<td>
<p>The model name, default is IMIX_ind</p>
</td></tr>
</table>


<h3>Value</h3>

<p>AIC/BIC values of the target model
</p>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# First load the data
data("data_p")

# Specify the initial values
mu_input &lt;- c(0,3,0,3)
sigma_input &lt;- rep(1,4)
p_input &lt;- rep(0.5,4)

# Fit the IMIX model
test1 &lt;- IMIX(data_input = data_p,data_type = "p",mu_ini = mu_input,sigma_ini = sigma_input,
p_ini = p_input,alpha = 0.1,model_selection_method = "AIC")

# Calculate the AIC and BIC values for IMIX_ind with two data types and four components
model_selection(test1$IMIX_ind$`Full MaxLogLik final`,
n=dim(test1$IMIX_ind$`posterior prob`)[1],g=4,d=2, "IMIX_ind")

</code></pre>

<hr>
<h2 id='model_selection_component'>Component Selection</h2><span id='topic+model_selection_component'></span>

<h3>Description</h3>

<p>Model selection for components based on AIC and BIC values for models in IMIX
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model_selection_component(
  data_input,
  data_type = c("p", "z"),
  tol = 1e-06,
  maxiter = 1000,
  seed = 10,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model_selection_component_+3A_data_input">data_input</code></td>
<td>
<p>An n x d data frame or matrix of the summary statistics z score or p value, n is the nubmer of genes, d is the number of data types. Each row is a gene, each column is a data type.</p>
</td></tr>
<tr><td><code id="model_selection_component_+3A_data_type">data_type</code></td>
<td>
<p>Whether the input data is the p values or z scores, default is p value</p>
</td></tr>
<tr><td><code id="model_selection_component_+3A_tol">tol</code></td>
<td>
<p>The convergence criterion. Convergence is declared when the change in the observed data log-likelihood increases by less than epsilon.</p>
</td></tr>
<tr><td><code id="model_selection_component_+3A_maxiter">maxiter</code></td>
<td>
<p>The maximum number of iteration, default is 1000</p>
</td></tr>
<tr><td><code id="model_selection_component_+3A_seed">seed</code></td>
<td>
<p>set.seed, default is 10</p>
</td></tr>
<tr><td><code id="model_selection_component_+3A_verbose">verbose</code></td>
<td>
<p>Whether to print the full log-likelihood for each iteration, default is FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Selected number of components based on AIC and BIC
</p>
<table>
<tr><td><code>Component_Selected_AIC</code></td>
<td>
<p>Selected number of components by AIC with the smallest AIC value among all components and models</p>
</td></tr>
<tr><td><code>Component_Selected_BIC</code></td>
<td>
<p>Selected number of components by BIC with the smallest BIC value among all components and models</p>
</td></tr>
<tr><td><code>AIC/BIC</code></td>
<td>
<p>The AIC and BIC values for all components for IMIX_ind_unrestrict, IMIX_cor_twostep, and IMIX_cor</p>
</td></tr>
<tr><td><code>IMIX_ind_unrestrict</code></td>
<td>
<p>A list of the IMIX_ind_unrestrict for all components 1,2,...2^d, this step was fitted using R package &quot;Mclust&quot;, more details of the output can be found there</p>
</td></tr>
<tr><td><code>IMIX_cor_twostep</code></td>
<td>
<p>A list of the IMIX_cor_twostep for all components 1,2,...2^d, here, the mean is the estimated value of IMIX_ind_unrestrict</p>
</td></tr>
<tr><td><code>IMIX_cor</code></td>
<td>
<p>A list of the IMIX_cor_twostep for all components 1,2,...2^d</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
Luca Scrucca, Michael Fop, T. Brendan Murphy, and Adrian E. Raftery. 2016. “mclust 5: Clustering, Classification and Density Estimation Using Gaussian Finite Mixture Models.” The R Journal 8 (1): 289–317. &lt;doi:10.32614/RJ-2016-021&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A toy example
data("data_p")
set.seed(10)
data &lt;- data_p[sample(1:1000,20,replace = FALSE),]
select_comp0 &lt;- model_selection_component(data, data_type = "p", seed = 20)


# First load the data
data("data_p")

# Perform model selections on the data
select_comp1 = model_selection_component(data_p, data_type = "p", seed = 20)

</code></pre>

<hr>
<h2 id='plot_component'>Plot the AIC or BIC Values for Model Selection</h2><span id='topic+plot_component'></span>

<h3>Description</h3>

<p>Plot the result output of model selection for components based on AIC and BIC values in IMIX
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_component(res_select, type = c("AIC", "BIC"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_component_+3A_res_select">res_select</code></td>
<td>
<p>Result output from function model_selection_component()</p>
</td></tr>
<tr><td><code id="plot_component_+3A_type">type</code></td>
<td>
<p>Which information criteria to use for plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot for the model selection of components
</p>


<h3>References</h3>

<p>Ziqiao Wang and Peng Wei. 2020. “IMIX: a multivariate mixture model approach to association analysis through multi-omics data integration.” Bioinformatics. &lt;doi:10.1093/bioinformatics/btaa1001&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# First load the data
data("data_p")

# Perform model selections on the data
select_comp1 &lt;- model_selection_component(data_p, data_type = "p", seed = 20)

# Make a plot for BIC values
plot_component(select_comp1, type = "BIC")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
