<!DOCTYPE html><html><head><title>Help for package immuneSIM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {immuneSIM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#codon_replacement'><p>Replaces codons with synonymous codons</p></a></li>
<li><a href='#codon_replacement_reconstruction'><p>Decodes immuneSIM repertoire codon replacements events.</p></a></li>
<li><a href='#combine_into_paired'><p>Generates a dataframe from separate heavy and light or beta and alpha chain dataframes</p></a></li>
<li><a href='#gen_code'><p>Translation dictionary amino acid &lt;-&gt; nucleotide codon</p></a></li>
<li><a href='#hotspot_df'><p>Hotspot dataframe for SHM</p></a></li>
<li><a href='#hub_seqs_exclusion'><p>Deletes top hub sequences from repertoire, changing the network architecture.</p></a></li>
<li><a href='#immuneSIM'><p>Simulates an immune repertoire based on user-defined parameters</p></a></li>
<li><a href='#insertions_and_deletion_lengths_df'><p>Dataframe containing insertion sequences and deletion lengths</p></a></li>
<li><a href='#length_dist_simulation'><p>Vector containing VDJ length distributions</p></a></li>
<li><a href='#list_example_repertoires'><p>Example repertoires</p></a></li>
<li><a href='#list_germline_genes_allele_01'><p>Collection of germline genes and frequencies</p></a></li>
<li><a href='#load_insdel_data'><p>Loads full insertion/deletion data from GitHub</p></a></li>
<li><a href='#motif_implantation'><p>Implant random or predefined motifs into CDR3</p></a></li>
<li><a href='#one_spot_df'><p>One Spot</p></a></li>
<li><a href='#plot_repertoire_A_vs_B'><p>Comparative plots of main repertoire features of two input repertoires (length distribution, amino acid frequency, VDJ usage, kmer occurrence)</p></a></li>
<li><a href='#plot_report_repertoire'><p>Plots main repertoire features (length distribution,amino acid frequencies and VDJ usage)</p></a></li>
<li><a href='#shm_event_reconstruction'><p>Decodes immuneSIM repertoire shm_events column.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tunable Simulation of B- And T-Cell Receptor Repertoires</td>
</tr>
<tr>
<td>Version:</td>
<td>0.8.7</td>
</tr>
<tr>
<td>Author:</td>
<td>Cédric R. Weber [aut, cre],
  Victor Greiff [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Cédric R. Weber &lt;cedric.weber@bsse.ethz.ch&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Simulate full B-cell and T-cell receptor repertoires using an in silico 
    recombination process that includes a wide variety of tunable parameters to introduce noise and biases. 
    Additional post-simulation modification functions allow the user to implant motifs or codon biases as 
    well as remodeling sequence similarity architecture. The output repertoires contain records of all 
    relevant repertoire dimensions and can be analyzed using provided repertoire analysis functions.
    Preprint is available at bioRxiv (Weber et al., 2019 &lt;<a href="https://doi.org/10.1101%2F759795">doi:10.1101/759795</a>&gt;).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>poweRlaw, stringdist, Biostrings, igraph, stringr, data.table,
plyr, reshape2, ggplot2, grid, ggthemes, RColorBrewer, Metrics,
repmis</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://immuneSIM.readthedocs.io">https://immuneSIM.readthedocs.io</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/GreiffLab/immuneSIM/issues">https://github.com/GreiffLab/immuneSIM/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-09-20 08:43:33 UTC; ceweber</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-09-27 10:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='codon_replacement'>Replaces codons with synonymous codons</h2><span id='topic+codon_replacement'></span>

<h3>Description</h3>

<p>Replaces codons with synonymous codons
</p>


<h3>Usage</h3>

<pre><code class='language-R'>codon_replacement(repertoire, mode = "both", codon_replacement_list,
  skip_probability = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="codon_replacement_+3A_repertoire">repertoire</code></td>
<td>
<p>An annotated AIRR compliant immuneSIM repertoire.
</p>
<p>(http://docs.airr-community.org/en/latest/)</p>
</td></tr>
<tr><td><code id="codon_replacement_+3A_mode">mode</code></td>
<td>
<p>Defines whether codons should be replaced in the nt or AA sequence or in both (&quot;nt&quot;,&quot;AA&quot;,&quot;both&quot;)</p>
</td></tr>
<tr><td><code id="codon_replacement_+3A_codon_replacement_list">codon_replacement_list</code></td>
<td>
<p>List containing instructions for which codons should be replaced and how</p>
</td></tr>
<tr><td><code id="codon_replacement_+3A_skip_probability">skip_probability</code></td>
<td>
<p>Probability with which a sequence gets skipped in the codon replacement process between 0,1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>immuneSIM repertoire with replaced codons
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repertoire &lt;- list_example_repertoires[["example_repertoire_A"]]
rep_codon_repl &lt;- codon_replacement(repertoire, "both",
list(tat = "tac", agt = "agc", gtt = "gtg"), 0)
</code></pre>

<hr>
<h2 id='codon_replacement_reconstruction'>Decodes immuneSIM repertoire codon replacements events.</h2><span id='topic+codon_replacement_reconstruction'></span>

<h3>Description</h3>

<p>Decodes immuneSIM repertoire codon replacements events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>codon_replacement_reconstruction(codon_replacement_vec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="codon_replacement_reconstruction_+3A_codon_replacement_vec">codon_replacement_vec</code></td>
<td>
<p>An vector containing strings describing codon replacement events as generated
by codon_replacement() function. The string contains information on every replacement event in the form:
</p>
<p>&quot;initial_codon:replacement_codon:number_of_occurrences&quot;
</p>
<p>which is combined into: &quot;Replacement1|Replacement2|Replacement3&quot;.
</p>
<p>(For example: &quot;tac,tat:3|agc,agt:1|gtg,gtt:0&quot;.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of dataframes. Each entry contains replacement info including count of occurrences for each simulated sequence.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>codon_replacement_example &lt;- c("tat,tac:3|agt,agc:3|gtt,gtg:0", "tat,tac:1|agt,agc:1|gtt,gtg:1")
codon_replacement_list &lt;- codon_replacement_reconstruction(codon_replacement_example)
</code></pre>

<hr>
<h2 id='combine_into_paired'>Generates a dataframe from separate heavy and light or beta and alpha chain dataframes</h2><span id='topic+combine_into_paired'></span>

<h3>Description</h3>

<p>Generates a dataframe from separate heavy and light or beta and alpha chain dataframes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_into_paired(repertoire_heavy, repertoire_light)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="combine_into_paired_+3A_repertoire_heavy">repertoire_heavy</code></td>
<td>
<p>A repertoire containing heavy/beta chain data</p>
</td></tr>
<tr><td><code id="combine_into_paired_+3A_repertoire_light">repertoire_light</code></td>
<td>
<p>A repertoire containing light/alpha chain data</p>
</td></tr>
</table>


<h3>Value</h3>

<p>immuneSIM repertoire containing heavy/beta and light/alpha chain data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repertoire_heavy &lt;- immuneSIM(number_of_seqs = 5,species = "mm",receptor = "ig", chain = "h")
repertoire_light &lt;- immuneSIM(number_of_seqs = 5,species = "mm",receptor = "ig", chain = "kl")
paired_repertoire &lt;- combine_into_paired(repertoire_heavy,repertoire_light)
</code></pre>

<hr>
<h2 id='gen_code'>Translation dictionary amino acid &lt;-&gt; nucleotide codon</h2><span id='topic+gen_code'></span>

<h3>Description</h3>

<p>A dataframe containing a mapping from each of 64 codons
to amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gen_code
</code></pre>


<h3>Format</h3>

<p>A data frame with 64 rows and  variables:
</p>

<dl>
<dt>aa</dt><dd><p>amino acid</p>
</dd>
<dt>codon</dt><dd><p>nucleotide codon</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://www.genscript.com/tools/codon-table">https://www.genscript.com/tools/codon-table</a>
</p>

<hr>
<h2 id='hotspot_df'>Hotspot dataframe for SHM</h2><span id='topic+hotspot_df'></span>

<h3>Description</h3>

<p>A dataframe containing mutation probabilities for every possible 5mer pattern
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hotspot_df
</code></pre>


<h3>Format</h3>

<p>A data frame with 1024 rows and  variables:
</p>

<dl>
<dt>pattern</dt><dd><p>amino acid</p>
</dd>
<dt>toA</dt><dd><p>probability of mutation to adenine</p>
</dd>
<dt>toC</dt><dd><p>probability of mutation to cytosine</p>
</dd>
<dt>toG</dt><dd><p>probability of mutation to guanine</p>
</dd>
<dt>toT</dt><dd><p>probability of mutation to thymine</p>
</dd>
<dt>Source</dt><dd><p>source of probability</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://cran.r-project.org/package=AbSim">https://cran.r-project.org/package=AbSim</a>
</p>

<hr>
<h2 id='hub_seqs_exclusion'>Deletes top hub sequences from repertoire, changing the network architecture.</h2><span id='topic+hub_seqs_exclusion'></span>

<h3>Description</h3>

<p>Deletes top hub sequences from repertoire, changing the network architecture.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hub_seqs_exclusion(repertoire, top_x = 0.005, report = FALSE,
  output_dir = "", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hub_seqs_exclusion_+3A_repertoire">repertoire</code></td>
<td>
<p>An annotated AIRR compliant repertoire.
</p>
<p>(http://docs.airr-community.org/en/latest/)</p>
</td></tr>
<tr><td><code id="hub_seqs_exclusion_+3A_top_x">top_x</code></td>
<td>
<p>Determines what percentage of hub sequences get excluded
</p>
<p>(Default: 0.005, i.e. Top 0.5 percent)</p>
</td></tr>
<tr><td><code id="hub_seqs_exclusion_+3A_report">report</code></td>
<td>
<p>The user can choose to output a report csv file containing the excluded
sequences. (Default: FALSE)</p>
</td></tr>
<tr><td><code id="hub_seqs_exclusion_+3A_output_dir">output_dir</code></td>
<td>
<p>If user specifies and output directory a csv file containing
the excluded sequences is saved at that path, otherwise it will be saved in tempdir().</p>
</td></tr>
<tr><td><code id="hub_seqs_exclusion_+3A_verbose">verbose</code></td>
<td>
<p>Determines whether messages on plot locations are output to user. (Default: TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Repertoire reduced by hub sequence (new network architecture)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repertoire &lt;- list_example_repertoires[["example_repertoire_A"]]
rep_excluded_hubs &lt;- hub_seqs_exclusion(repertoire, top_x = 0.005, output_dir = "")
</code></pre>

<hr>
<h2 id='immuneSIM'>Simulates an immune repertoire based on user-defined parameters</h2><span id='topic+immuneSIM'></span>

<h3>Description</h3>

<p>Simulates an immune repertoire based on user-defined parameters
</p>


<h3>Usage</h3>

<pre><code class='language-R'>immuneSIM(number_of_seqs = 1000,
  vdj_list = list_germline_genes_allele_01, species = "mm",
  receptor = "ig", chain = "h",
  insertions_and_deletion_lengths = insertions_and_deletion_lengths_df,
  user_defined_alpha = 2, name_repertoire = "sim_rep",
  length_distribution_rand = length_dist_simulation, random = FALSE,
  shm.mode = "none", shm.prob = 15/350, vdj_noise = 0,
  vdj_dropout = c(V = 0, D = 0, J = 0), ins_del_dropout = c(""),
  equal_cc = FALSE, freq_update_time = round(0.5 * number_of_seqs),
  max_cdr3_length = 100, min_cdr3_length = 6, verbose = TRUE,
  airr_compliant = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="immuneSIM_+3A_number_of_seqs">number_of_seqs</code></td>
<td>
<p>Integer defining the number of sequences that should be simulated</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_vdj_list">vdj_list</code></td>
<td>
<p>List containing germline genes and their frequencies</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_species">species</code></td>
<td>
<p>String defining species for which repertoire should be simulated (&quot;mm&quot;: mouse, &quot;hs&quot;: human. Default: &quot;mm&quot;).</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_receptor">receptor</code></td>
<td>
<p>String defining receptor type (&quot;ig&quot; or &quot;tr&quot;. Default: &quot;ig&quot;)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_chain">chain</code></td>
<td>
<p>String defining chain (for ig: &quot;h&quot;,&quot;k&quot;,&quot;l&quot;, for tr: &quot;b&quot; or &quot;a&quot;. Default: &quot;h&quot;)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_insertions_and_deletion_lengths">insertions_and_deletion_lengths</code></td>
<td>
<p>Data.frame containing np1, np2 sequences as well as deletion lengths.
(Pooled from murine repertoire data, Greiff,2017)
Note: This is a subset of 500000 observations of the dataframe used in the paper. The full dataframe which can
be introduced here can be found on: (Git-Link)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_user_defined_alpha">user_defined_alpha</code></td>
<td>
<p>Numeric. Scaling parameter used for the simulation of powerlaw distribution
(recommended range 2-5. Default: 2, https://en.wikipedia.org/wiki/Power_law)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_name_repertoire">name_repertoire</code></td>
<td>
<p>String defining chosen repertoire name recorded in the name_repertoire column of the output for identification.</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_length_distribution_rand">length_distribution_rand</code></td>
<td>
<p>Vector containing lengths of immune receptor sequences based on immune repertoire data (Greiff, 2017).</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_random">random</code></td>
<td>
<p>Boolean. If TRUE repertoire will consist of fully random sequences, independent of germline genes.</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_shm.mode">shm.mode</code></td>
<td>
<p>String defining mode of somatic hypermutation simulation based on AbSim
(options: 'none', 'data','poisson', 'naive', 'motif', 'wrc'. Default: 'none'). See AbSim documentation.</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_shm.prob">shm.prob</code></td>
<td>
<p>Numeric defining probability of a SHM (somatic hypermutation) occurring at each position.</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_vdj_noise">vdj_noise</code></td>
<td>
<p>Numeric between 0,1, setting noise level to be introduced in provided V,D,J germline frequencies. 0 denotes no noise. (Default: 0)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_vdj_dropout">vdj_dropout</code></td>
<td>
<p>Named vector containing entries V,D,J setting the number of germline genes to be dropped out. (Default: c(&quot;V&quot;=0,&quot;D&quot;=0,&quot;J&quot;=0))</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_ins_del_dropout">ins_del_dropout</code></td>
<td>
<p>String determining whether insertions and deletions should occur.
Options: &quot;&quot;, &quot;no_insertions&quot;, &quot;no_insertions_n1&quot;, &quot;no_insertions_n2&quot;, &quot;no_deletions_v&quot;, &quot;no_deletions_d_5&quot;,
&quot;no_deletions_d_3&quot;, &quot;no_deletions_j&quot;, &quot;no_deletions_vd&quot;, &quot;no_deletions&quot;. Default: &quot;&quot;)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_equal_cc">equal_cc</code></td>
<td>
<p>Boolean that if set TRUE will override user_defined_alpha and generate a clone count distribution that is equal for all sequences.
Default: FALSE.</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_freq_update_time">freq_update_time</code></td>
<td>
<p>Numeric determining whether simulated VDJ frequencies agree with input after set amount of sequences to correct for VDJ bias.
Default: Update after 50 percent of sequences.</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_max_cdr3_length">max_cdr3_length</code></td>
<td>
<p>Numeric defining maximal length of cdr3. (Default: 100)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_min_cdr3_length">min_cdr3_length</code></td>
<td>
<p>Numeric defining minimal length of cdr3. (Default: 6)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_verbose">verbose</code></td>
<td>
<p>Boolean toggling printing of progress on and off (Default: FALSE)</p>
</td></tr>
<tr><td><code id="immuneSIM_+3A_airr_compliant">airr_compliant</code></td>
<td>
<p>Boolean determining whether output repertoire should be named in an AIRR compliant manner
(Default: TRUE). (http://docs.airr-community.org/en/latest/)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An annotated AIRR-compliant immuneSIM repertoire. (http://docs.airr-community.org/en/latest/)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim_rep &lt;- immuneSIM(number_of_seqs = 10, vdj_list = list_germline_genes_allele_01,
species = "mm", receptor = "ig", chain = "h",
insertions_and_deletion_lengths = insertions_and_deletion_lengths_df,
user_defined_alpha = 2,name_repertoire = "mm_igh_sim",
shm.mode = "data",shm.prob=15/350,vdj_noise = 0, vdj_dropout = c(V=0,D=0,J=0),
ins_del_dropout = "",min_cdr3_length = 6)
</code></pre>

<hr>
<h2 id='insertions_and_deletion_lengths_df'>Dataframe containing insertion sequences and deletion lengths</h2><span id='topic+insertions_and_deletion_lengths_df'></span>

<h3>Description</h3>

<p>A dataframe containing all insertions and deletions
observed in experimental data (pooled across all samples, Greiff, 2017)
This dataframe is a subset of the dataframe used in the application note.
The original dataframe which contains 11363603 rows can be downloaded
from:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>insertions_and_deletion_lengths_df
</code></pre>


<h3>Format</h3>

<p>A data frame with 500000 rows and variables:
</p>

<dl>
<dt>n1</dt><dd><p>np1 insertions</p>
</dd>
<dt>n2</dt><dd><p>np2 insertions</p>
</dd>
<dt>del_v</dt><dd><p>lengths of V gene deletions</p>
</dd>
<dt>del_d_5</dt><dd><p>lengths of 5' end D gene deletions</p>
</dd>
<dt>del_d_3</dt><dd><p>lengths of 3' end D gene deletions</p>
</dd>
<dt>del_j</dt><dd><p>lengths of J gene deletions</p>
</dd>
</dl>


<h3>Details</h3>

<p>https://github.com/GreiffLab/immuneSIM or using the provided
function: load_insdel_data()
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1016/j.celrep.2017.04.054">https://doi.org/10.1016/j.celrep.2017.04.054</a>
</p>

<hr>
<h2 id='length_dist_simulation'>Vector containing VDJ length distributions</h2><span id='topic+length_dist_simulation'></span>

<h3>Description</h3>

<p>A vector containing 10000 VDJ lengths for simulating
of fully random sequences (independent of germline genes)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>length_dist_simulation
</code></pre>


<h3>Format</h3>

<p>A vector with 10000 entries:
</p>

<dl>
<dt>length</dt><dd><p>VDJ nucleotide lengths sampled from murine naive
follicular B-cell data, Greiff 2017</p>
</dd>
</dl>


<h3>Source</h3>

<p><a href="https://doi.org/10.1016/j.celrep.2017.04.054">https://doi.org/10.1016/j.celrep.2017.04.054</a>
</p>

<hr>
<h2 id='list_example_repertoires'>Example repertoires</h2><span id='topic+list_example_repertoires'></span>

<h3>Description</h3>

<p>A list containing two example repertoires (100 sequences each)
simulated with immuneSIM using default parameters.
These repertoires are used in the examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_example_repertoires
</code></pre>


<h3>Format</h3>

<p>A list with 2 entries:
</p>

<dl>
<dt>example_repertoire_A</dt><dd><p>Repertoire simulated using standard parameters (A)</p>
</dd>
<dt>example_repertoire_A</dt><dd><p>Repertoire simulated using standard parameters (B)</p>
</dd>
</dl>


<h3>Source</h3>


<p><a href="https://immunesim.readthedocs.io">https://immunesim.readthedocs.io</a>

</p>

<hr>
<h2 id='list_germline_genes_allele_01'>Collection of germline genes and frequencies</h2><span id='topic+list_germline_genes_allele_01'></span>

<h3>Description</h3>

<p>A list containing sublists for species (&quot;hs&quot;,&quot;mm&quot;) which in turn
contain sublists for receptors (&quot;ig&quot;,&quot;tr&quot;) which are subset in
chains (&quot;h&quot;, &quot;k&quot;, &quot;l&quot; and &quot;b&quot;, &quot;a&quot;, respectively). Each entry
contains a list of three dataframes (&quot;V&quot;,&quot;D&quot; and &quot;J&quot;) with the major IMGT
annotated germline genes including name, sequence based on IMGT and frequencies based on
experimental data from DeWitt(2017), Emerson (2017), Greiff (2017) and Madi (2017)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_germline_genes_allele_01
</code></pre>


<h3>Format</h3>

<p>A list of lists containing dataframes with up to 126 entries:
</p>

<dl>
<dt>gene</dt><dd><p>name of germline gene</p>
</dd>
<dt>allele</dt><dd><p>allele number (presently restricted to allele 01)</p>
</dd>
<dt>sequence</dt><dd><p>nucleotide sequence of germline gene</p>
</dd>
<dt>species</dt><dd><p>name of species</p>
</dd>
<dt>frequency</dt><dd><p>Frequencies of germline genes based on experimental data</p>
</dd>
</dl>


<h3>Source</h3>


<p><a href="http://www.imgt.org/vquest/refseqh.html">http://www.imgt.org/vquest/refseqh.html</a>
</p>
<p><a href="https://doi.org/10.1371/journal.pone.0160853">https://doi.org/10.1371/journal.pone.0160853</a>
</p>
<p><a href="https://doi.org/10.1038/ng.3822">https://doi.org/10.1038/ng.3822</a>
</p>
<p><a href="https://doi.org/10.1016/j.celrep.2017.04.054">https://doi.org/10.1016/j.celrep.2017.04.054</a>
</p>
<p><a href="https://doi.org/10.7554/eLife.22057">https://doi.org/10.7554/eLife.22057</a>

</p>

<hr>
<h2 id='load_insdel_data'>Loads full insertion/deletion data from GitHub</h2><span id='topic+load_insdel_data'></span>

<h3>Description</h3>

<p>Loads full insertion/deletion data from GitHub
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_insdel_data()
</code></pre>


<h3>Value</h3>

<p>Dataframe containing insertions and deletions (11363603 rows, 6 columns)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
full_insertions_and_deletion_df &lt;- load_insdel_data()

</code></pre>

<hr>
<h2 id='motif_implantation'>Implant random or predefined motifs into CDR3</h2><span id='topic+motif_implantation'></span>

<h3>Description</h3>

<p>Implant random or predefined motifs into CDR3
</p>


<h3>Usage</h3>

<pre><code class='language-R'>motif_implantation(sim_repertoire, motif, fixed_pos = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="motif_implantation_+3A_sim_repertoire">sim_repertoire</code></td>
<td>
<p>An annotated AIRR compliant immuneSIM repertoire.</p>
</td></tr>
<tr><td><code id="motif_implantation_+3A_motif">motif</code></td>
<td>
<p>Either a list that contains number, length and frequencies of motifs or dataframe that contains predefined motifs and their frequencies</p>
</td></tr>
<tr><td><code id="motif_implantation_+3A_fixed_pos">fixed_pos</code></td>
<td>
<p>defines position at which motif is to be introduced. if 0 motif will be introduced at random position</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Repertoire with modified sequences containing implanted motifs in CDR3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim_repertoire &lt;- list_example_repertoires[["example_repertoire_A"]]
sim_rep_motifs &lt;- motif_implantation(sim_repertoire,list("n"=2,"k"=3,"freq"=c(0.1,0.1)),0)
</code></pre>

<hr>
<h2 id='one_spot_df'>One Spot</h2><span id='topic+one_spot_df'></span>

<h3>Description</h3>

<p>A dataframe containing a mutation probabilities to base per 5mer
(inherited from AbSim package)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>one_spot_df
</code></pre>


<h3>Format</h3>

<p>A dataframe with 32 entries:
</p>

<dl>
<dt>pattern</dt><dd><p>amino acid</p>
</dd>
<dt>toA</dt><dd><p>probability of mutation to adenine</p>
</dd>
<dt>toC</dt><dd><p>probability of mutation to cytosine</p>
</dd>
<dt>toG</dt><dd><p>probability of mutation to guanine</p>
</dd>
<dt>toT</dt><dd><p>probability of mutation to thymine</p>
</dd>
<dt>Source</dt><dd><p>source of probability</p>
</dd>
</dl>


<h3>Source</h3>


<p><a href="https://cran.r-project.org/package=AbSim">https://cran.r-project.org/package=AbSim</a>
</p>
<p><a href="https://doi.org/10.1093/bioinformatics/btx533">https://doi.org/10.1093/bioinformatics/btx533</a>

</p>

<hr>
<h2 id='plot_repertoire_A_vs_B'>Comparative plots of main repertoire features of two input repertoires (length distribution, amino acid frequency, VDJ usage, kmer occurrence)</h2><span id='topic+plot_repertoire_A_vs_B'></span>

<h3>Description</h3>

<p>Comparative plots of main repertoire features of two input repertoires (length distribution, amino acid frequency, VDJ usage, kmer occurrence)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_repertoire_A_vs_B(repertoire_A, repertoire_B,
  names_repertoires = c("Repertoire_A", "Repertoire_B"),
  length_aa_plot = 14, output_dir = "", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_repertoire_A_vs_B_+3A_repertoire_a">repertoire_A</code></td>
<td>
<p>An annotated AIRR-compliant immuneSIM repertoire.
</p>
<p>(http://docs.airr-community.org/en/latest/)</p>
</td></tr>
<tr><td><code id="plot_repertoire_A_vs_B_+3A_repertoire_b">repertoire_B</code></td>
<td>
<p>An annotated AIRR-compliant immuneSIM repertoire.</p>
</td></tr>
<tr><td><code id="plot_repertoire_A_vs_B_+3A_names_repertoires">names_repertoires</code></td>
<td>
<p>A vector containing two strings denoting the names of the repertoires / repertoire descriptions.</p>
</td></tr>
<tr><td><code id="plot_repertoire_A_vs_B_+3A_length_aa_plot">length_aa_plot</code></td>
<td>
<p>Defines sequence length for which the amino acid frequency plot will be made.</p>
</td></tr>
<tr><td><code id="plot_repertoire_A_vs_B_+3A_output_dir">output_dir</code></td>
<td>
<p>String containing full path of desired output folder. If empty, figures will be output in tempdir().</p>
</td></tr>
<tr><td><code id="plot_repertoire_A_vs_B_+3A_verbose">verbose</code></td>
<td>
<p>Determines whether messages on plot locations are output to user. (Default: TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE (plots saved as pdfs into subfolder 'figures')
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repertoire_A &lt;- list_example_repertoires[["example_repertoire_A"]]
repertoire_B &lt;- list_example_repertoires[["example_repertoire_B"]]
plot_repertoire_A_vs_B(
repertoire_A,
repertoire_B,
c("Sim_repertoire_1","Sim_repertoire_2"),
length_aa_plot = 14,
output_dir="")
</code></pre>

<hr>
<h2 id='plot_report_repertoire'>Plots main repertoire features (length distribution,amino acid frequencies and VDJ usage)</h2><span id='topic+plot_report_repertoire'></span>

<h3>Description</h3>

<p>Plots main repertoire features (length distribution,amino acid frequencies and VDJ usage)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_report_repertoire(repertoire, output_dir = "", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_report_repertoire_+3A_repertoire">repertoire</code></td>
<td>
<p>An annotated AIRR-compliant immuneSIM repertoire.
</p>
<p>(http://docs.airr-community.org/en/latest/)</p>
</td></tr>
<tr><td><code id="plot_report_repertoire_+3A_output_dir">output_dir</code></td>
<td>
<p>String containing full path of desired output folder. If empty figures will be output in tempdir().</p>
</td></tr>
<tr><td><code id="plot_report_repertoire_+3A_verbose">verbose</code></td>
<td>
<p>Determines whether messages on plot locations are output to user. (Default: TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE (plots saved as pdfs into subfolder 'figures')
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repertoire &lt;- list_example_repertoires[["example_repertoire_A"]]
plot_report_repertoire(repertoire,output_dir="")
</code></pre>

<hr>
<h2 id='shm_event_reconstruction'>Decodes immuneSIM repertoire shm_events column.</h2><span id='topic+shm_event_reconstruction'></span>

<h3>Description</h3>

<p>Decodes immuneSIM repertoire shm_events column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>shm_event_reconstruction(shm_event_vec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="shm_event_reconstruction_+3A_shm_event_vec">shm_event_vec</code></td>
<td>
<p>An vector containing strings describing SHM events as output in shm_events column of immuneSIM repertoires.
The string contains information on every mutation event in the form:
</p>
<p>&quot;Position:pre_mutation_nucleotide,post_mutation_nucleotide&quot;
combined as: &quot;Mutation1|Mutation2|Mutation3&quot;. For example: &quot;171:t,a|186:g,a&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of dataframes. Each entry contains location and shm mutation info for a simulated sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>shm_events_example&lt;-c("171:t,a|186:g,a|287:g,a|310:t,c","","294:c,g|316:t,c|330:c,t")
shm_list&lt;-shm_event_reconstruction(shm_events_example)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
