<!DOCTYPE html><html lang="en"><head><title>Help for package imsig</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {imsig}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#corr_matrix'><p>Correlation matrix</p></a></li>
<li><a href='#example_cli'><p>Example clinical data file for survival analysis with ImSig</p></a></li>
<li><a href='#example_data'><p>Example transcriptomics data</p></a></li>
<li><a href='#feature_select'><p>Feature selection of signature genes</p></a></li>
<li><a href='#gene_stat'><p>General stastitics of ImSig analysis</p></a></li>
<li><a href='#imsig'><p>Estimate the relative abundance of tissue-infiltrating immune subpopulations abundances using gene expression data</p></a></li>
<li><a href='#imsig_survival'><p>Survival analysis based on relative abundance of immune infiltration estimated by ImSig</p></a></li>
<li><a href='#plot_abundance'><p>Plot relative abundance of immune cells</p></a></li>
<li><a href='#plot_network'><p>Network graph of ImSig genes</p></a></li>
<li><a href='#plot_survival'><p>Forest plot of survial analysis by ImSig</p></a></li>
<li><a href='#pp_exp'><p>Pre-processing expression matrix</p></a></li>
<li><a href='#pp_sig'><p>Pre-processing ImSig file</p></a></li>
<li><a href='#sig'><p>ImSig genes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Immune Cell Gene Signatures for Profiling the Microenvironment
of Solid Tumours</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Ajit Johnson Nirmal</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ajit Johnson Nirmal &lt;ajitjohnson.n@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimate the relative abundance of tissue-infiltrating immune subpopulations abundances using gene expression data. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ajitjohnson/imsig/">https://github.com/ajitjohnson/imsig/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ajitjohnson/imsig/issues">https://github.com/ajitjohnson/imsig/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>HiClimR (&ge; 1.2), RColorBrewer (&ge; 1.1), igraph (&ge; 1.2),
ggplot2 (&ge; 2.2), gridExtra (&ge; 2.3), survival (&ge; 2.4)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-09 03:04:00 UTC; aj</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-10 01:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='corr_matrix'>Correlation matrix</h2><span id='topic+corr_matrix'></span>

<h3>Description</h3>

<p>Creates a correlation matrix of ImSig signature genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corr_matrix(exp, r)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="corr_matrix_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="corr_matrix_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Gene-gene correlation matrix of ImSig genes.
</p>

<hr>
<h2 id='example_cli'>Example clinical data file for survival analysis with ImSig</h2><span id='topic+example_cli'></span>

<h3>Description</h3>

<p>An example clinical data file. Minimum required informations are the sample name (same as that of the expression matrix), event (dead or alive) and time to event (days, months or years).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_cli
</code></pre>


<h3>Format</h3>

<p>dataframe
</p>

<hr>
<h2 id='example_data'>Example transcriptomics data</h2><span id='topic+example_data'></span>

<h3>Description</h3>

<p>Example expression data matrix. The data is preffered to be in natural scale with genes as rows and samples as columns.Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>example_data
</code></pre>


<h3>Format</h3>

<p>dataframe
</p>

<hr>
<h2 id='feature_select'>Feature selection of signature genes</h2><span id='topic+feature_select'></span>

<h3>Description</h3>

<p>ImSig genes were designed to be co-expressed in tissue transcriptomic data. However, depending on the dataset some of the genes may not co-express with the dominant module. In order to remove such deviant genes, a feature selection can be carried out based on correlation. This function removes genes that exhibit a poor correlation (less than the defined r value) with the dominant ImSig module. This step of feature selection is recommended to enrich the prediction of relative abundance of immune cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature_select(exp, r = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="feature_select_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="feature_select_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of 'feature selected' genes based on the set r value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>feature_select (exp = example_data, r = 0.7)
</code></pre>

<hr>
<h2 id='gene_stat'>General stastitics of ImSig analysis</h2><span id='topic+gene_stat'></span>

<h3>Description</h3>

<p>[Total genes in ImSig]: The total number of genes in ImSig list. [No. of ImSig genes in user dataset]: The number of ImSig genes found in user's dataset. Like all signatures, ImSig works best when this overlap is high, preferably over 75
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_stat(exp, r = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gene_stat_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="gene_stat_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of general statistics of ImSig analysis.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_select">feature_select</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>gene_stat (exp = example_data, r = 0.7)
</code></pre>

<hr>
<h2 id='imsig'>Estimate the relative abundance of tissue-infiltrating immune subpopulations abundances using gene expression data</h2><span id='topic+imsig'></span>

<h3>Description</h3>

<p>Estimates the relative abundance of immune cells across patients/samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imsig(exp, r = 0.6, sort = TRUE, sort_by = "T cells")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="imsig_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="imsig_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
<tr><td><code id="imsig_+3A_sort">sort</code></td>
<td>
<p>Sort the samples based on abundance of a particular cell type. 'Set sort = FALSE' if you wish not to apply sorting. By default the function sorts by abundance of T cells. The cell type of interest for sorting can be controlled by the 'sort_by' parameter.</p>
</td></tr>
<tr><td><code id="imsig_+3A_sort_by">sort_by</code></td>
<td>
<p>Can be used to sort the samples by predicted abundance of a particular cell type. All other cell types follow this sorting. By default it is sorted by 'T cells'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Relative abundance of immune cells across samples. Returns a dataframe.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_select">feature_select</a></code>, <code><a href="#topic+example_data">example_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cell_abundance = imsig (exp = example_data, r = 0.7, sort=TRUE, sort_by='T cells')
head(cell_abundance)
</code></pre>

<hr>
<h2 id='imsig_survival'>Survival analysis based on relative abundance of immune infiltration estimated by ImSig</h2><span id='topic+imsig_survival'></span>

<h3>Description</h3>

<p>Patients are split into two groups based on their immune cell abundance (median aundance value) and a regular survival analyis is carried out.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>imsig_survival(exp, cli, time = "time", status = "status", r = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="imsig_survival_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="imsig_survival_+3A_cli">cli</code></td>
<td>
<p>Clinical metadata containting the event data (dead or alive) and time to event data. Samples names should be in rownames and same as that in the expression file. Check head() of <code><a href="#topic+example_cli">example_cli</a></code> for an example clinical data.</p>
</td></tr>
<tr><td><code id="imsig_survival_+3A_time">time</code></td>
<td>
<p>Column name of time-to-event parameter.</p>
</td></tr>
<tr><td><code id="imsig_survival_+3A_status">status</code></td>
<td>
<p>Column name of event (dead or alive) parameter.</p>
</td></tr>
<tr><td><code id="imsig_survival_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Hazard Ratio
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_select">feature_select</a></code>, <code><a href="#topic+example_data">example_data</a></code>, <code><a href="#topic+example_cli">example_cli</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
survival = imsig_survival (exp = example_data, cli = example_cli)
head(survival)

</code></pre>

<hr>
<h2 id='plot_abundance'>Plot relative abundance of immune cells</h2><span id='topic+plot_abundance'></span>

<h3>Description</h3>

<p>Barplots of relative abundance of immune cells across samples.The order of the samples are the same as that of <code><a href="#topic+imsig">imsig</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_abundance(exp, r = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_abundance_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="plot_abundance_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_select">feature_select</a></code>, <code><a href="#topic+example_data">example_data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_abundance (exp = example_data, r = 0.7)
</code></pre>

<hr>
<h2 id='plot_network'>Network graph of ImSig genes</h2><span id='topic+plot_network'></span>

<h3>Description</h3>

<p>A Network visualization displays undirected graph structures and highlights the relationships between entities. The nodes are ImSig genes and the edges represent the correlation between them. The nodes are coloured based on cell type. Try using a correlation cut-off of '0' to get a complete picture.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_network(
  exp,
  r = 0.6,
  pt.cex = 2,
  cex = 1,
  inset = 0,
  x.intersp = 2,
  vertex.size = 3,
  vertex.label = NA,
  layout = layout_with_fr
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_network_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_pt.cex">pt.cex</code></td>
<td>
<p>expansion factor(s) for the points.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_cex">cex</code></td>
<td>
<p>character expansion factor relative to current par(&quot;cex&quot;). Used for text, and provides the default for pt.cex.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_inset">inset</code></td>
<td>
<p>inset distance(s) from the margins as a fraction of the plot region when legend is placed by keyword.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_x.intersp">x.intersp</code></td>
<td>
<p>character interspacing factor for horizontal (x) spacing.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_vertex.size">vertex.size</code></td>
<td>
<p>Node size of network graph</p>
</td></tr>
<tr><td><code id="plot_network_+3A_vertex.label">vertex.label</code></td>
<td>
<p>Add gene names to the network graph. Default set to NA.</p>
</td></tr>
<tr><td><code id="plot_network_+3A_layout">layout</code></td>
<td>
<p>Layout algorithm to be used for building network. Default set to force-directed layout algorithm by Fruchterman and Reingold. Read documentation of 'igraph' for other available algorithms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Network graph
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_select">feature_select</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_network (exp = example_data, r = 0.7)
</code></pre>

<hr>
<h2 id='plot_survival'>Forest plot of survial analysis by ImSig</h2><span id='topic+plot_survival'></span>

<h3>Description</h3>

<p>Patients are split into two groups based on their immune cell abundance (median aundance value) and a regular survival analyis is carried out. Raw values can be obtained from <code><a href="#topic+imsig_survival">imsig_survival</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_survival(exp, cli, time = "time", status = "status", r = 0.6)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_survival_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
<tr><td><code id="plot_survival_+3A_cli">cli</code></td>
<td>
<p>Clinical metadata containting the event data (dead or alive) and time to event data. Samples names should be in rownames and same as that in the expression file. Check head() of <code><a href="#topic+example_cli">example_cli</a></code> for an example clinical data.</p>
</td></tr>
<tr><td><code id="plot_survival_+3A_time">time</code></td>
<td>
<p>Column name of time-to-event parameter.</p>
</td></tr>
<tr><td><code id="plot_survival_+3A_status">status</code></td>
<td>
<p>Column name of event (dead or alive) parameter.</p>
</td></tr>
<tr><td><code id="plot_survival_+3A_r">r</code></td>
<td>
<p>Use a value between 0 and 1. Default is 0.6. This is a user defined correlation cut-off to perform feature selection (<code><a href="#topic+feature_select">feature_select</a></code>). Feature selection aids to enrich the prediction of relative abundance of immune cells by filtering off poorly correlated ImSig genes. To get an idea of what cut-off to use check the results of (<code><a href="#topic+gene_stat">gene_stat</a></code>) and choose a cut-off that displays high median correlation and maintains a high proportion of genes after feature selection.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Forest plot
</p>


<h3>See Also</h3>

<p><code><a href="#topic+feature_select">feature_select</a></code>, <code><a href="#topic+example_data">example_data</a></code>, <code><a href="#topic+example_cli">example_cli</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_survival (exp = example_data, r = 0.7, cli = example_cli, time = 'time', status= 'status')
</code></pre>

<hr>
<h2 id='pp_exp'>Pre-processing expression matrix</h2><span id='topic+pp_exp'></span>

<h3>Description</h3>

<p>Subsets the user's dataset based on the genes that are common to the users dataset and ImSig.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pp_exp(exp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pp_exp_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Expression dataframe
</p>

<hr>
<h2 id='pp_sig'>Pre-processing ImSig file</h2><span id='topic+pp_sig'></span>

<h3>Description</h3>

<p>Subsets ImSig genes based on the genes that are common to the users dataset and ImSig
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pp_sig(exp)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pp_sig_+3A_exp">exp</code></td>
<td>
<p>Dataframe of transcriptomic data (natural scale) containing genes as rows and samples as columns. Note: Gene names should be set as row names and duplicates are not allowed. Missing values are not allowed within the expression matrix. Check example- head(example_data): <code><a href="#topic+example_data">example_data</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ImSig dataframe
</p>

<hr>
<h2 id='sig'>ImSig genes</h2><span id='topic+sig'></span>

<h3>Description</h3>

<p>ImSig signature genes and the cell type they represent
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sig
</code></pre>


<h3>Format</h3>

<p>dataframe
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
