<!DOCTYPE html><html><head><title>Help for package inbreedR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {inbreedR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#inbreedR'><p>inbreedR: Workflows for analysing variance in inbreeding and HFCs based on SNP or microsatellite markers.</p></a></li>
<li><a href='#bodyweight'><p>Oldfield mouse bodyweight data</p></a></li>
<li><a href='#check_data'><p>Checks the data for consistency with the inbreedR working format.</p></a></li>
<li><a href='#convert_raw'><p>Genotype format converter</p></a></li>
<li><a href='#g2_microsats'><p>Estimating g2 from microsatellite data</p></a></li>
<li><a href='#g2_snps'><p>Estimating g2 from larger datasets, such as SNPs</p></a></li>
<li><a href='#HHC'><p>Calculates heterzygosity-heterozygosity correlations with</p>
standardized multilocus heterozygosities (sMLH)</a></li>
<li><a href='#MLH'><p>Calculate multilocus heterozygosity (MLH)</p></a></li>
<li><a href='#mouse_msats'><p>Oldfield mouse microsatellite data</p></a></li>
<li><a href='#mouse_snps'><p>Oldfield mouse SNP data</p></a></li>
<li><a href='#plot.inbreed'><p>Plot an inbreed object</p></a></li>
<li><a href='#print.inbreed'><p>Print an inbreed object</p></a></li>
<li><a href='#r2_hf'><p>Expected r2 between standardized multilocus heterozygosity (h) and inbreeding level (f)</p></a></li>
<li><a href='#r2_Wf'><p>Expected r2 between inbreeding level (f) and fitness (W)</p></a></li>
<li><a href='#simulate_g2'><p>Simulate <code class="reqn">g2</code></p></a></li>
<li><a href='#simulate_r2_hf'><p>Calculates the expected squared correlation between heteorzygosity and inbreeding for simulated marker sets</p></a></li>
<li><a href='#sMLH'><p>Calculate multilocus heterozygosity (MLH)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analysing Inbreeding Based on Genetic Markers</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.3</td>
</tr>
<tr>
<td>Description:</td>
<td>A framework for analysing inbreeding and heterozygosity-fitness
    correlations (HFCs) based on microsatellite and SNP markers.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.1)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table (&ge; 1.9.6), parallel, stats, graphics</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 0.10.0), knitr, rmarkdown, covr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-02 09:49:33 UTC; mstoffel</td>
</tr>
<tr>
<td>Author:</td>
<td>Martin A. Stoffel [aut, cre],
  Mareike Esser [aut],
  Joseph Hoffman [aut],
  Marty Kardos [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Martin A. Stoffel &lt;martin.adam.stoffel@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-02 11:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='inbreedR'>inbreedR: Workflows for analysing variance in inbreeding and HFCs based on SNP or microsatellite markers.</h2><span id='topic+inbreedR'></span><span id='topic+inbreedR-package'></span>

<h3>Description</h3>

<p><code>inbreedR</code> contains the following functions:
</p>
<p><a href="#topic+g2_microsats">g2_microsats</a>
<a href="#topic+g2_snps">g2_snps</a>
<a href="#topic+convert_raw">convert_raw</a>
<a href="#topic+check_data">check_data</a>
<a href="#topic+r2_hf">r2_hf</a>
<a href="#topic+r2_Wf">r2_Wf</a>
<a href="#topic+HHC">HHC</a>
<a href="#topic+sMLH">sMLH</a>
<a href="#topic+MLH">MLH</a>
<a href="#topic+simulate_g2">simulate_g2</a>
<a href="#topic+simulate_r2_hf">simulate_r2_hf</a>
<a href="#topic+plot.inbreed">plot.inbreed</a>
<a href="#topic+print.inbreed">print.inbreed</a>
</p>


<h3>Details</h3>

<p>A correlation between heterozygosity (h) and fitness (W) requires a simultaneous effect of 
inbreeding level (f) on both of them.
A heterozygosity-fitness correlation (HFC) thus is the product of two correlations, 
which can be summarized in the following equation:
</p>
<p style="text-align: center;"><code class="reqn">r(W, h) = r(W, f)r(h, f)</code>
</p>

<p>Estimating these parameters and their sensitivity towards the number and type of genetic markers used is the central framework
of the inbreedR package. At the heart of measuring inbreeding based on genetic markers is the g2 statistic, which estimates 
the correlation of heterozygosity across markers, called identity disequilibrium (ID). ID is a proxy for inbreeding.
</p>
<p>The package has three main goals:
</p>

<ul>
<li><p> Assessing identity disequilibria and the potential to detect heterozygosity-fitness correlations 
</p>
</li>
<li><p> Providing insights on the sensitivity of these measures based on the number/type of molecular markers used
</p>
</li>
<li><p> Implementing computationally efficient functions in a flexible environment for analysing 
inbreeding and HFC's with both small and large datasets.
</p>
</li></ul>

<p>For a short introduction to inbreedR start with the vignette:
<code>browseVignettes(package = "inbreedR")</code>
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com), Mareike Esser (messer@uni-bielefeld.de)
</p>


<h3>References</h3>

<p>Slate, J., David, P., Dodds, K. G., Veenvliet, B. A., Glass, B. C., Broad, T. E., &amp; McEwan, J. C. (2004). 
Understanding the relationship between the inbreeding coefficient 
and multilocus heterozygosity: theoretical expectations and empirical data. Heredity, 93(3), 255-265.
</p>
<p>Szulkin, M., Bierne, N., &amp; David, P. (2010). HETEROZYGOSITY-FITNESS CORRELATIONS: A TIME FOR REAPPRAISAL. 
Evolution, 64(5), 1202-1217.
</p>
<p>David, P., Pujol, B., Viard, F., Castella, V. and Goudet, J. (2007),
Reliable selfing rate estimates from imperfect population genetic data. Molecular Ecology,
16: 2474
</p>
<p>Hoffman, J.I., Simpson, F., David, P., Rijks, J.M., Kuiken, T., Thorne, M.A.S., Lacey, R.C. &amp; Dasmahapatra, K.K. (2014) High-throughput sequencing reveals inbreeding depression in a natural population.
Proceedings of the National Academy of Sciences of the United States of America, 111: 3775-3780.
</p>

<hr>
<h2 id='bodyweight'>Oldfield mouse bodyweight data</h2><span id='topic+bodyweight'></span>

<h3>Description</h3>

<p>Bodyweight data for 36 oldfield mice.
</p>


<h3>Format</h3>

<p>A vector with 36 elements.</p>


<h3>References</h3>

<p>Hoffman, J.I., Simpson, F., David, P., Rijks, J.M., Kuiken, T., Thorne, M.A.S., Lacey, R.C. &amp; Dasmahapatra, K.K. (2014) High-throughput sequencing reveals inbreeding depression in a natural population.
Proceedings of the National Academy of Sciences of the United States of America, 111: 3775-3780. Doi: 10.1073/pnas.1318945111
</p>

<hr>
<h2 id='check_data'>Checks the data for consistency with the inbreedR working format.</h2><span id='topic+check_data'></span>

<h3>Description</h3>

<p>The <code>inbreedR</code> working format is an i * l genotype matrix, whereby each individual is a row 
and each column is a locus.
Heterozygosity at a given locus should be coded as <code>1</code>, homozygosity as <code>0</code> and missing values
should be coded as <code>NA</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_data(genotypes, num_ind = NULL, num_loci = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_data_+3A_genotypes">genotypes</code></td>
<td>
<p><code>data.frame</code> (or <code>matrix</code>) with individuals in rows and loci in columns,
containing genotypes coded as <code>0</code> (homozygote), <code>1</code> (heterozygote) and <code>NA</code> (missing)</p>
</td></tr>
<tr><td><code id="check_data_+3A_num_ind">num_ind</code></td>
<td>
<p>Number of individuals</p>
</td></tr>
<tr><td><code id="check_data_+3A_num_loci">num_loci</code></td>
<td>
<p>Number of loci / markers</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Checks that (1) the genotype data just contains 3 elements, which is <code>0</code> for homozygote,
<code>1</code> for heterozygote and <code>NA</code> for missing data, (2) the number 
of individuals corresponds to the number of rows and the number of loci corresponds to the
number of columns, (3) the data type is numeric.
.
</p>


<h3>Value</h3>

<p>TRUE if the data format is correct, error message if any test failed
</p>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
# tranform raw genotypes into 0/1 format
genotypes &lt;- convert_raw(mouse_msats)
# check data
check_data(genotypes, num_ind = 36, num_loci = 12)


</code></pre>

<hr>
<h2 id='convert_raw'>Genotype format converter</h2><span id='topic+convert_raw'></span>

<h3>Description</h3>

<p>Turns raw genotype data into 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing), which is the working format for 
the <code>inbreedR</code> functions.
A raw genotype matrix has individuals in rows and each locus in two adjacent columns. Individual ID's can be rownames.
Type data(mouse_msats) for an example raw genotype data frame.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_raw(genotypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="convert_raw_+3A_genotypes">genotypes</code></td>
<td>
<p>Raw genotype <code>data.frame</code> or <code>matrix</code>. Rows represent individuals and each locus has two adjacent columns. 
Alleles within loci can be coded as numbers (e.g. microsatellite length) or characters (e.g. &quot;A&quot;, &quot;T&quot;)
See data(mouse_msat) for an example. Missing values should be coded as <code>NA</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> object with 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing data).
Each locus is a column and each individual is a row.
</p>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Mouse microsatellite data with missing values coded as NA
data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
head(genotypes)

</code></pre>

<hr>
<h2 id='g2_microsats'>Estimating g2 from microsatellite data</h2><span id='topic+g2_microsats'></span>

<h3>Description</h3>

<p>Estimating g2 from microsatellite data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g2_microsats(genotypes, nperm = 0, nboot = 0, boot_over = "inds",
  CI = 0.95, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="g2_microsats_+3A_genotypes">genotypes</code></td>
<td>
<p><code>data.frame</code> with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing)</p>
</td></tr>
<tr><td><code id="g2_microsats_+3A_nperm">nperm</code></td>
<td>
<p>Number of permutations for testing the hypothesis that the empirical g2-value is higher than the g2 for random associations between 
individuals and genotypes.</p>
</td></tr>
<tr><td><code id="g2_microsats_+3A_nboot">nboot</code></td>
<td>
<p>Number of bootstraps for estimating a confidence interval</p>
</td></tr>
<tr><td><code id="g2_microsats_+3A_boot_over">boot_over</code></td>
<td>
<p>Bootstrap over individuals by specifying &quot;inds&quot; and over loci with &quot;loci&quot;. Defaults to &quot;ind&quot;.</p>
</td></tr>
<tr><td><code id="g2_microsats_+3A_ci">CI</code></td>
<td>
<p>Confidence interval (default to 0.95)</p>
</td></tr>
<tr><td><code id="g2_microsats_+3A_verbose">verbose</code></td>
<td>
<p>If FALSE, nothing will be printed to show the status of bootstraps and permutations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates g2 from smaller datasets. The underlying formula is compationally expensive 
due to double summations over all paits of loci (see David et al. 2007). 
Use convert_raw to convert raw genotypes (with 2 columns per locus) into
the required format.
</p>


<h3>Value</h3>

<p>g2_microsats returns an object of class &quot;inbreed&quot;.
The functions 'print' and 'plot' are used to print a summary and to plot the distribution of bootstrapped g2 values and CI.
</p>
<p>An 'inbreed' object from <code>g2_microsats</code> is a list containing the following components:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>g2</code></td>
<td>
<p>g2 value</p>
</td></tr>
<tr><td><code>p_val</code></td>
<td>
<p>p value from permutation test</p>
</td></tr>
<tr><td><code>g2_permut</code></td>
<td>
<p>g2 values from permuted genotypes</p>
</td></tr>
<tr><td><code>g2_boot</code></td>
<td>
<p>g2 values from bootstrap samples</p>
</td></tr>
<tr><td><code>CI_boot</code></td>
<td>
<p>confidence interval from bootstraps</p>
</td></tr>
<tr><td><code>se_boot</code></td>
<td>
<p>standard error of g2 from bootstraps</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code>nloc</code></td>
<td>
<p>number of markers</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com) &amp;
Mareike Esser (messer@techfak.uni-bielefeld.de)
</p>


<h3>References</h3>

<p>David, P., Pujol, B., Viard, F., Castella, V. and Goudet, J. (2007),
Reliable selfing rate estimates from imperfect population genetic data. Molecular Ecology,
16: 2474
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
# tranform raw genotypes into 0/1 format
genotypes &lt;- convert_raw(mouse_msats)
(g2_mouse &lt;- g2_microsats(genotypes, nperm = 1000, nboot = 100, boot_over = "inds", CI = 0.95))

</code></pre>

<hr>
<h2 id='g2_snps'>Estimating g2 from larger datasets, such as SNPs</h2><span id='topic+g2_snps'></span>

<h3>Description</h3>

<p>Estimating g2 from larger datasets, such as SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g2_snps(genotypes, nperm = 0, nboot = 0, boot_over = "inds", CI = 0.95,
  parallel = FALSE, ncores = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="g2_snps_+3A_genotypes">genotypes</code></td>
<td>
<p><code>data.frame</code> with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and NA (missing)</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_nperm">nperm</code></td>
<td>
<p>number or permutations for to estimate a p-value</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_nboot">nboot</code></td>
<td>
<p>number of bootstraps to estimate a confidence interval</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_boot_over">boot_over</code></td>
<td>
<p>Bootstrap over individuals by specifying &quot;inds&quot; and over loci with &quot;loci&quot;. Defaults to &quot;ind&quot;.</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_ci">CI</code></td>
<td>
<p>confidence interval (default to 0.95)</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_parallel">parallel</code></td>
<td>
<p>Default is FALSE. If TRUE, bootstrapping and permutation tests are parallelized</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_ncores">ncores</code></td>
<td>
<p>Specify number of cores to use for parallelization. By default,
all available cores are used.</p>
</td></tr>
<tr><td><code id="g2_snps_+3A_verbose">verbose</code></td>
<td>
<p>If FALSE, nothing will be printed to show the status of bootstraps and permutations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculates g2 from SNP datasets. Use convert_raw to convert raw genotypes (with 2 columns per locus) into
the required format
</p>


<h3>Value</h3>

<p>g2_snps returns an object of class &quot;inbreed&quot;.
The functions 'print' and 'plot' are used to print a summary and to plot the distribution of bootstrapped g2 values and CI.
</p>
<p>An 'inbreed' object from <code>g2_snps</code> is a list containing the following components:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>g2</code></td>
<td>
<p>g2 value</p>
</td></tr>
<tr><td><code>p_val</code></td>
<td>
<p>p value from permutation test</p>
</td></tr>
<tr><td><code>g2_permut</code></td>
<td>
<p>g2 values from permuted genotypes</p>
</td></tr>
<tr><td><code>g2_boot</code></td>
<td>
<p>g2 values from bootstrap samples</p>
</td></tr>
<tr><td><code>CI_boot</code></td>
<td>
<p>confidence interval from bootstrap distribution</p>
</td></tr>
<tr><td><code>se_boot</code></td>
<td>
<p>standard error of g2 from bootstraps</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code>nloc</code></td>
<td>
<p>number of markers</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com) &amp;
Mareike Esser (messer@techfak.uni-bielefeld.de)
</p>


<h3>References</h3>

<p>Hoffman, J.I., Simpson, F., David, P., Rijks, J.M., Kuiken, T., Thorne, M.A.S., Lacey, R.C. &amp; Dasmahapatra, K.K. (2014) High-throughput sequencing reveals inbreeding depression in a natural population.
Proceedings of the National Academy of Sciences of the United States of America, 111: 3775-3780. Doi: 10.1073/pnas.1318945111
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load SNP genotypes in 0 (homozygous), 1 (heterozygous), NA (missing) format.
# low number of bootstraps and permutations for computational reasons.
data(mouse_snps)
(g2_mouse &lt;- g2_snps(mouse_snps, nperm = 10, nboot = 10, CI = 0.95, boot_over = "loci"))

# parallelized version for more bootstraps or permutations
## Not run: 
(g2_mouse &lt;- g2_snps(mouse_snps, nperm = 1000, nboot = 1000, 
                     CI = 0.95, parallel = TRUE, ncores = 4))

## End(Not run)

</code></pre>

<hr>
<h2 id='HHC'>Calculates heterzygosity-heterozygosity correlations with 
standardized multilocus heterozygosities (sMLH)</h2><span id='topic+HHC'></span>

<h3>Description</h3>

<p>Loci are randomly devided into two equal groups and the correlation coefficient
between the resulting sMLH values is calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HHC(genotypes, reps = 100, CI = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HHC_+3A_genotypes">genotypes</code></td>
<td>
<p><code>data.frame</code> with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing)</p>
</td></tr>
<tr><td><code id="HHC_+3A_reps">reps</code></td>
<td>
<p>number of repetitions, i.e. splittings of the dataset</p>
</td></tr>
<tr><td><code id="HHC_+3A_ci">CI</code></td>
<td>
<p>size of the confidence interval around the mean het-het correlation (default is 0.95)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>HHC_vals</code></td>
<td>
<p>vector of HHC's obtained by randomly splitting the dataset</p>
</td></tr>
<tr><td><code>summary_exp_r2</code></td>
<td>
<p>r2 mean and sd for each number of subsetted loci</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code>nloc</code></td>
<td>
<p>number of markers</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>References</h3>

<p>Balloux, F., Amos, W., &amp; Coulson, T. (2004). Does heterozygosity estimate inbreeding
in real populations?. Molecular Ecology, 13(10), 3021-3031.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
(out &lt;- HHC(genotypes, reps = 100, CI = 0.95))

</code></pre>

<hr>
<h2 id='MLH'>Calculate multilocus heterozygosity (MLH)</h2><span id='topic+MLH'></span>

<h3>Description</h3>

<p>MLH is defined as the total number of heterozygous loci in an individual divided 
by the number of loci typed in the focal individual. An MLH of 0.5 thus means
that 50 percent of an indiviudals loci are heterozygous.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MLH(genotypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MLH_+3A_genotypes">genotypes</code></td>
<td>
<p>data.frame with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and NA (missing)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of individual multilocus heterozygosities
</p>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>References</h3>

<p>Coltman, D. W., Pilkington, J. G., Smith, J. A., &amp; Pemberton, J. M. (1999). 
Parasite-mediated selection against inbred Soay sheep in a free-living, 
island population. Evolution, 1259-1267.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
het &lt;- MLH(genotypes)

</code></pre>

<hr>
<h2 id='mouse_msats'>Oldfield mouse microsatellite data</h2><span id='topic+mouse_msats'></span>

<h3>Description</h3>

<p>Dataset with each microsatellite locus in two adjecent columns (one per allel). 
Missing values are coded as <code>NA</code>.
</p>


<h3>Format</h3>

<p>A data frame with 36 observations at 13198 loci.</p>


<h3>References</h3>

<p>Hoffman, J.I., Simpson, F., David, P., Rijks, J.M., Kuiken, T., Thorne, M.A.S., Lacey, R.C. &amp; Dasmahapatra, K.K. (2014) High-throughput sequencing reveals inbreeding depression in a natural population.
Proceedings of the National Academy of Sciences of the United States of America, 111: 3775-3780. Doi: 10.1073/pnas.1318945111
</p>
<p>Dasmahapatra KK, Lacy RC, Amos W (2007) Estimating levels of inbreeding using AFLP markers. Heredity 100:286-295.
</p>

<hr>
<h2 id='mouse_snps'>Oldfield mouse SNP data</h2><span id='topic+mouse_snps'></span>

<h3>Description</h3>

<p>Mouse snp data in 0 (homozygous), 1(heterzygous) and <code>NA</code> (missing) format.
Each row represents an individual and each column is a locus.
</p>


<h3>Format</h3>

<p>A <code>data.frame</code> with 36 observations at 13198 loci.</p>


<h3>References</h3>

<p>Hoffman, J.I., Simpson, F., David, P., Rijks, J.M., Kuiken, T., Thorne, M.A.S., Lacey, R.C. &amp; Dasmahapatra, K.K. (2014) High-throughput sequencing reveals inbreeding depression in a natural population.
Proceedings of the National Academy of Sciences of the United States of America, 111: 3775-3780. Doi: 10.1073/pnas.1318945111
</p>
<p>Dasmahapatra KK, Lacy RC, Amos W (2007) Estimating levels of inbreeding using AFLP markers. Heredity 100:286-295.
</p>

<hr>
<h2 id='plot.inbreed'>Plot an inbreed object</h2><span id='topic+plot.inbreed'></span>

<h3>Description</h3>

<p>Plot an inbreed object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'inbreed'
plot(x, true_g2 = FALSE, plottype = c("boxplot",
  "histogram"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.inbreed_+3A_x">x</code></td>
<td>
<p>An <code>inbreed</code> object.</p>
</td></tr>
<tr><td><code id="plot.inbreed_+3A_true_g2">true_g2</code></td>
<td>
<p>For plotting a <code>simulate_g2</code> output. If TRUE, plots the real g2 (based on realized f)
as a reference line.</p>
</td></tr>
<tr><td><code id="plot.inbreed_+3A_plottype">plottype</code></td>
<td>
<p>deprecated. &quot;boxplot&quot; or &quot;histogram&quot; to plot the output of r2_hf() and to show either
the boxplots through resampling of loci or the histogram from the bootstrapping of 
r2 over individuals.</p>
</td></tr>
<tr><td><code id="plot.inbreed_+3A_...">...</code></td>
<td>
<p>Additional arguments to the hist() function for the g2 and HHC functions. 
Additional arguments to the boxplot() function for plotting the result of the r2_hf() function.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>See Also</h3>

<p><a href="#topic+g2_snps">g2_snps</a>, <a href="#topic+g2_microsats">g2_microsats</a>
</p>

<hr>
<h2 id='print.inbreed'>Print an inbreed object</h2><span id='topic+print.inbreed'></span>

<h3>Description</h3>

<p>Displays the results a <code>inbreed</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'inbreed'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.inbreed_+3A_x">x</code></td>
<td>
<p>An <code>inbreed</code> object from one of the <code>inbreedR</code> functions.</p>
</td></tr>
<tr><td><code id="print.inbreed_+3A_...">...</code></td>
<td>
<p>Additional arguments; none are used in this method.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>See Also</h3>

<p><a href="#topic+g2_snps">g2_snps</a>, <a href="#topic+g2_microsats">g2_microsats</a>, <a href="graphics.html#topic+plot">plot</a>
</p>

<hr>
<h2 id='r2_hf'>Expected r2 between standardized multilocus heterozygosity (h) and inbreeding level (f)</h2><span id='topic+r2_hf'></span>

<h3>Description</h3>

<p>Expected r2 between standardized multilocus heterozygosity (h) and inbreeding level (f)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r2_hf(genotypes, type = c("msats", "snps"), nboot = NULL,
  parallel = FALSE, ncores = NULL, CI = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="r2_hf_+3A_genotypes">genotypes</code></td>
<td>
<p><code>data.frame</code> with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing)</p>
</td></tr>
<tr><td><code id="r2_hf_+3A_type">type</code></td>
<td>
<p>specifies g2 formula to take. Type &quot;snps&quot; for large datasets and &quot;msats&quot; for smaller datasets.</p>
</td></tr>
<tr><td><code id="r2_hf_+3A_nboot">nboot</code></td>
<td>
<p>number of bootstraps over individuals to estimate a confidence interval
around r2(h, f)</p>
</td></tr>
<tr><td><code id="r2_hf_+3A_parallel">parallel</code></td>
<td>
<p>Default is FALSE. If TRUE, bootstrapping and permutation tests are parallelized</p>
</td></tr>
<tr><td><code id="r2_hf_+3A_ncores">ncores</code></td>
<td>
<p>Specify number of cores to use for parallelization. By default,
all available cores but one are used.</p>
</td></tr>
<tr><td><code id="r2_hf_+3A_ci">CI</code></td>
<td>
<p>confidence interval (default to 0.95)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>r2_hf_full</code></td>
<td>
<p>expected r2 between inbreeding and sMLH for the full dataset</p>
</td></tr>
<tr><td><code>r2_hf_boot</code></td>
<td>
<p>expected r2 values from bootstrapping over individuals</p>
</td></tr>
<tr><td><code>CI_boot</code></td>
<td>
<p>confidence interval around the expected r2</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code>nloc</code></td>
<td>
<p>number of markers</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>References</h3>

<p>Slate, J., David, P., Dodds, K. G., Veenvliet, B. A., Glass, B. C., Broad, T. E., &amp; McEwan, J. C. (2004). 
Understanding the relationship between the inbreeding coefficient 
and multilocus heterozygosity: theoretical expectations and empirical data. Heredity, 93(3), 255-265.
</p>
<p>Szulkin, M., Bierne, N., &amp; David, P. (2010). HETEROZYGOSITY-FITNESS CORRELATIONS: A TIME FOR REAPPRAISAL. 
Evolution, 64(5), 1202-1217.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
(out &lt;- r2_hf(genotypes, nboot = 100, type = "msats", parallel = FALSE))
plot(out)
</code></pre>

<hr>
<h2 id='r2_Wf'>Expected r2 between inbreeding level (f) and fitness (W)</h2><span id='topic+r2_Wf'></span>

<h3>Description</h3>

<p>Expected r2 between inbreeding level (f) and fitness (W)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r2_Wf(genotypes, trait, family = "gaussian", type = c("msats", "snps"),
  nboot = NULL, parallel = FALSE, ncores = NULL, CI = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="r2_Wf_+3A_genotypes">genotypes</code></td>
<td>
<p>A <code>data.frame</code> with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing).</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_trait">trait</code></td>
<td>
<p>vector of any type which can be specified in R's glm() function. Sequence of individuals has to
match sequence of individuals in the rows of the <code>genotypes</code> <code>data.frame</code>.</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_family">family</code></td>
<td>
<p>distribution of the trait. Default is gaussian. For other distributions, just naming the distribution
(e.g. binomial) will use the default link function (see ?family). Specifying another
link function can be done in the same way as in the glm() function. A binomial distribution with 
probit instead of logit link would be specified with family = binomial(link = &quot;probit&quot;)</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_type">type</code></td>
<td>
<p>specifies g2 formula to take. Type &quot;snps&quot; for large datasets and &quot;msats&quot; for smaller datasets.</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_nboot">nboot</code></td>
<td>
<p>number of bootstraps over individuals to estimate a confidence interval
around r2(W, f).</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_parallel">parallel</code></td>
<td>
<p>Default is FALSE. If TRUE, bootstrapping and permutation tests are parallelized.</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_ncores">ncores</code></td>
<td>
<p>Specify number of cores to use for parallelization. By default,
all available cores but one are used.</p>
</td></tr>
<tr><td><code id="r2_Wf_+3A_ci">CI</code></td>
<td>
<p>confidence interval (default to 0.95)</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>exp_r2_full</code></td>
<td>
<p>expected r2 between inbreeding and sMLH for the full dataset</p>
</td></tr>
<tr><td><code>r2_Wf_boot</code></td>
<td>
<p>expected r2 values from bootstrapping over individuals</p>
</td></tr>
<tr><td><code>CI_boot</code></td>
<td>
<p>confidence interval around the expected r2</p>
</td></tr>
<tr><td><code>nobs</code></td>
<td>
<p>number of observations</p>
</td></tr>
<tr><td><code>nloc</code></td>
<td>
<p>number of markers</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>References</h3>

<p>Slate, J., David, P., Dodds, K. G., Veenvliet, B. A., Glass, B. C., Broad, T. E., &amp; McEwan, J. C. (2004). 
Understanding the relationship between the inbreeding coefficient 
and multilocus heterozygosity: theoretical expectations and empirical data. Heredity, 93(3), 255-265.
</p>
<p>Szulkin, M., Bierne, N., &amp; David, P. (2010). HETEROZYGOSITY-FITNESS CORRELATIONS: A TIME FOR REAPPRAISAL. 
Evolution, 64(5), 1202-1217.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
data(bodyweight)
genotypes &lt;- convert_raw(mouse_msats)

(out &lt;- r2_Wf(genotypes = genotypes, trait = bodyweight, family = "gaussian", type = "msats",
              nboot = 100, parallel = FALSE, ncores = NULL, CI = 0.95))


</code></pre>

<hr>
<h2 id='simulate_g2'>Simulate <code class="reqn">g2</code></h2><span id='topic+simulate_g2'></span>

<h3>Description</h3>

<p>This function can be used to simulate genotype data, draw subsets of loci and calculate the
respective <code class="reqn">g2</code> values. Every subset of markers is drawn independently to give insights
into the variation and precision of <code class="reqn">g2</code> calculated from a given number of markers and individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_g2(n_ind = NULL, H_nonInb = 0.5, meanF = 0.2, varF = 0.03,
  subsets = NULL, reps = 100, type = c("msats", "snps"), CI = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_g2_+3A_n_ind">n_ind</code></td>
<td>
<p>number of individuals to sample from the population</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_h_noninb">H_nonInb</code></td>
<td>
<p>true genome-wide heteorzygosity of a non-inbred individual</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_meanf">meanF</code></td>
<td>
<p>mean realized inbreeding f</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_varf">varF</code></td>
<td>
<p>variance in realized inbreeding f</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_subsets">subsets</code></td>
<td>
<p>a vector specifying the sizes of marker-subsets to draw. Specifying  <code>subsets = c(2, 5, 10, 15, 20)</code> 
would draw marker sets of 2 to 20 markers. The minimum number of markers to calculate g2 is 2.</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_reps">reps</code></td>
<td>
<p>number of resampling repetitions</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_type">type</code></td>
<td>
<p>specifies g2 formula. Type &quot;snps&quot; for large datasets and &quot;msats&quot; for smaller datasets.</p>
</td></tr>
<tr><td><code id="simulate_g2_+3A_ci">CI</code></td>
<td>
<p>Confidence intervals to calculate (default to 0.95)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>simulate_g2</code> function simulates genotypes from which subsets of loci can be sampled independently. 
These simulations can be used to evaluate the effects of the number of individuals 
and loci on the precision and magnitude of <code class="reqn">g2</code>. The user specifies the number of simulated individuals (<code>n_ind</code>), the subsets of 
loci (<code>subsets</code>) to be drawn, the heterozygosity of non-inbred individuals (<code>H_nonInb</code>) and the 
distribution of <em>f</em> among the simulated individuals. The <em>f</em> values of the simulated individuals are sampled
randomly from a beta distribution with mean (<code>meanF</code>) and variance (<code>varF</code>) specified by the user 
(e.g. as in wang2011). This enables the simulation to mimic populations with known inbreeding 
characteristics, or to simulate hypothetical scenarios of interest. For computational simplicity, allele 
frequencies are assumed to be constant across all loci and the simulated loci are unlinked. Genotypes
(i.e. the heterozygosity/homozygosity status at each locus) are assigned stochastically based on the <em>f</em>
values of the simulated individuals. Specifically, the probability of an individual being heterozygous at
any given locus (<code class="reqn">H</code>) is expressed as <code class="reqn">H = H0(1-f)</code> , where <code class="reqn">H0</code> is the user-specified heterozygosity of a 
non-inbred individual and <em>f</em> is an individual's inbreeding coefficient drawn from the beta distribution.
</p>


<h3>Value</h3>

<p><code>simulate_g2</code> returns an object of class &quot;inbreed&quot;.
The functions 'print' and 'plot' are used to print a summary and to plot the g2 values with means and confidence intervals
</p>
<p>An 'inbreed' object from  <code>simulate_g2</code> is a list containing the following components:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>estMat</code></td>
<td>
<p>matrix with all r2(h,f) estimates. Each row contains the values for a given subset of markers</p>
</td></tr>
<tr><td><code>true_g2</code></td>
<td>
<p>&quot;true&quot; g2 value based on the assigned realized inbreeding values</p>
</td></tr>
<tr><td><code>n_ind</code></td>
<td>
<p>specified number of individuals</p>
</td></tr>
<tr><td><code>subsets</code></td>
<td>
<p>vector specifying the marker sets</p>
</td></tr>
<tr><td><code>reps</code></td>
<td>
<p>repetitions per subset</p>
</td></tr>
<tr><td><code>H_nonInb</code></td>
<td>
<p>true genome-wide heteorzygosity of a non-inbred individual</p>
</td></tr>
<tr><td><code>meanF</code></td>
<td>
<p>mean realized inbreeding f</p>
</td></tr>
<tr><td><code>varF</code></td>
<td>
<p>variance in realized inbreeding f</p>
</td></tr>
<tr><td><code>min_val</code></td>
<td>
<p>minimum g2 value</p>
</td></tr>
<tr><td><code>max_val</code></td>
<td>
<p>maximum g2 value</p>
</td></tr>
<tr><td><code>all_CI</code></td>
<td>
<p>confidence intervals for all subsets</p>
</td></tr>
<tr><td><code>all_sd</code></td>
<td>
<p>standard deviations for all subsets</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Marty Kardos (marty.kardos@ebc.uu.se) &amp;
Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
sim_g2 &lt;- simulate_g2(n_ind = 10, H_nonInb = 0.5, meanF = 0.2, varF = 0.03,
                      subsets = c(4,6,8,10), reps = 100, 
                      type = "msats")
plot(sim_g2)
</code></pre>

<hr>
<h2 id='simulate_r2_hf'>Calculates the expected squared correlation between heteorzygosity and inbreeding for simulated marker sets</h2><span id='topic+simulate_r2_hf'></span>

<h3>Description</h3>

<p>This function can be used to simulate genotype data, draw random subsamples and calculate the
expected squared correlations between heterozygosity and fitness (<code class="reqn">r2(h, f)</code>).
Every subset of markers is drawn independently to give insights
into the variation and precision of <code class="reqn">r2(h, f)</code> calculated from a given number of markers and individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_r2_hf(n_ind = NULL, H_nonInb = 0.5, meanF = 0.2, varF = 0.03,
  subsets = NULL, reps = 100, type = c("msats", "snps"), CI = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_r2_hf_+3A_n_ind">n_ind</code></td>
<td>
<p>number of individuals to sample from the population</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_h_noninb">H_nonInb</code></td>
<td>
<p>true genome-wide heteorzygosity of a non-inbred individual</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_meanf">meanF</code></td>
<td>
<p>mean realized inbreeding f</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_varf">varF</code></td>
<td>
<p>variance in realized inbreeding f</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_subsets">subsets</code></td>
<td>
<p>a vector specifying the sizes of marker-subsets to draw. Specifying  <code>subsets = c(2, 5, 10, 15, 20)</code> 
would draw marker sets of 2 to 20 markers. The minimum number of markers is 2.</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_reps">reps</code></td>
<td>
<p>number of resampling repetitions</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_type">type</code></td>
<td>
<p>specifies g2 formula. Type &quot;snps&quot; for large datasets and &quot;msats&quot; for smaller datasets.</p>
</td></tr>
<tr><td><code id="simulate_r2_hf_+3A_ci">CI</code></td>
<td>
<p>Confidence intervals to calculate (default to 0.95)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>simulate_r2_hf</code> function simulates genotypes from which subsets of loci can be sampled independently. 
These simulations can be used to evaluate the effects of the number of individuals 
and loci on the precision and magnitude of the expected squared correlation between heterozygosity and inbreeding 
(<code class="reqn">r2(h, f)</code>). The user specifies the number of simulated individuals (<code>n_ind</code>), the subsets of 
loci (<code>subsets</code>) to be drawn, the heterozygosity of non-inbred individuals (<code>H_nonInb</code>) and the 
distribution of <em>f</em> among the simulated individuals. The <em>f</em> values of the simulated individuals are sampled
randomly from a beta distribution with mean (<code>meanF</code>) and variance (<code>varF</code>) specified by the user 
(e.g. as in wang2011). This enables the simulation to mimic populations with known inbreeding 
characteristics, or to simulate hypothetical scenarios of interest. For computational simplicity, allele 
frequencies are assumed to be constant across all loci and the simulated loci are unlinked. Genotypes
(i.e. the heterozygosity/homozygosity status at each locus) are assigned stochastically based on the <em>f</em>
values of the simulated individuals. Specifically, the probability of an individual being heterozygous at
any given locus (<code class="reqn">H</code>) is expressed as <code class="reqn">H = H0(1-f)</code> , where <code class="reqn">H0</code> is the user-specified heterozygosity of a 
non-inbred individual and <em>f</em> is an individual's inbreeding coefficient drawn from the beta distribution.
</p>


<h3>Value</h3>

<p><code>simulate_r2_hf</code> returns an object of class &quot;inbreed&quot;.
The functions 'print' and 'plot' are used to print a summary and to plot the r2(h, f) values with means and confidence intervals
</p>
<p>An 'inbreed' object from  <code>simulate_g2</code> is a list containing the following components:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>function call.</p>
</td></tr>
<tr><td><code>estMat</code></td>
<td>
<p>matrix with all r2(h,f) estimates. Each row contains the values for a given subset of markers</p>
</td></tr>
<tr><td><code>n_ind</code></td>
<td>
<p>specified number of individuals</p>
</td></tr>
<tr><td><code>subsets</code></td>
<td>
<p>vector specifying the marker sets</p>
</td></tr>
<tr><td><code>reps</code></td>
<td>
<p>repetitions per subset</p>
</td></tr>
<tr><td><code>H_nonInb</code></td>
<td>
<p>true genome-wide heteorzygosity of a non-inbred individual</p>
</td></tr>
<tr><td><code>meanF</code></td>
<td>
<p>mean realized inbreeding f</p>
</td></tr>
<tr><td><code>varF</code></td>
<td>
<p>variance in realized inbreeding f</p>
</td></tr>
<tr><td><code>min_val</code></td>
<td>
<p>minimum g2 value</p>
</td></tr>
<tr><td><code>max_val</code></td>
<td>
<p>maximum g2 value</p>
</td></tr>
<tr><td><code>all_CI</code></td>
<td>
<p>confidence intervals for all subsets</p>
</td></tr>
<tr><td><code>all_sd</code></td>
<td>
<p>standard deviations for all subsets</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Marty Kardos (marty.kardos@ebc.uu.se) &amp;
Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
sim_r2 &lt;- simulate_r2_hf(n_ind = 10, H_nonInb = 0.5, meanF = 0.2, varF = 0.03,
                      subsets = c(4,6,8,10), reps = 100, 
                      type = "msats")
plot(sim_r2)
</code></pre>

<hr>
<h2 id='sMLH'>Calculate multilocus heterozygosity (MLH)</h2><span id='topic+sMLH'></span>

<h3>Description</h3>

<p>sMLH is defined as the total number of heterozygous loci in an individual divided 
by the sum of average observed heterozygosities in the population over the 
subset of loci successfully typed in the focal individual.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sMLH(genotypes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sMLH_+3A_genotypes">genotypes</code></td>
<td>
<p>data.frame with individuals in rows and loci in columns,
containing genotypes coded as 0 (homozygote), 1 (heterozygote) and <code>NA</code> (missing)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of individual standardized multilocus heterozygosities
</p>


<h3>Author(s)</h3>

<p>Martin A. Stoffel (martin.adam.stoffel@gmail.com)
</p>


<h3>References</h3>

<p>Coltman, D. W., Pilkington, J. G., Smith, J. A., &amp; Pemberton, J. M. (1999). 
Parasite-mediated selection against inbred Soay sheep in a free-living, 
island population. Evolution, 1259-1267.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mouse_msats)
genotypes &lt;- convert_raw(mouse_msats)
het &lt;- sMLH(genotypes)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
