<!DOCTYPE html><html lang="en"><head><title>Help for package indelmiss</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {indelmiss}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#indelmiss-package'>
<p>Insertion deletion analysis while accounting for possible missing data.</p></a></li>
<li><a href='#gardnerelladata'>
<p>Gardnerella vaginalis phyletic data (gene family memberships) and phylogenetic tree</p></a></li>
<li><a href='#indelrates'>
<p>Function for estimating indel rates while accounting for missing data.</p></a></li>
<li><a href='#mycobacteriumdata1'>
<p>Mycobacterium data phyletic data (gene family memberships) and tree</p></a></li>
<li><a href='#mycobacteriumdata2'>
<p>Alternate Mycobacterium tree.</p></a></li>
<li><a href='#plot.indelmiss'>
<p>Plot parameter estimates from the model fit</p></a></li>
<li><a href='#plotp'>
<p>Plot estimates for the parameter that accounts for possible missing data</p></a></li>
<li><a href='#plotrates'>
<p>Plot estimates for insertion and deletion rates</p></a></li>
<li><a href='#plottree'>
<p>Plot the tree used the branches colored according to the different specified branch groupings (or clades) following unique rates.</p></a></li>
<li><a href='#print.indelmiss'>
<p>Print summary information from fit</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Insertion Deletion Analysis While Accounting for Possible
Missing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.10</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Utkarsh J. Dang and G. Brian Golding</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Utkarsh J. Dang &lt;utkarshdang@cunet.carleton.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Genome-wide gene insertion and deletion rates can be modelled in a maximum 
    likelihood framework with the additional flexibility of modelling potential missing 
    data using the models included within. These models simultaneously estimate insertion 
    and deletion (indel) rates of gene families and proportions of "missing" data for 
    (multiple) taxa of interest. The likelihood framework is utilized for parameter 
    estimation. A phylogenetic tree of the taxa and gene presence/absence patterns 
    (with data ordered by the tips of the tree) are required. See Dang et al.
    (2016) &lt;<a href="https://doi.org/10.1534%2Fgenetics.116.191973">doi:10.1534/genetics.116.191973</a>&gt; for more details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.2), ape (&ge; 3.2), numDeriv (&ge; 2012.9.1),
phangorn (&ge; 1.99.13)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-15 13:40:42 UTC; udang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-15 16:52:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='indelmiss-package'>
Insertion deletion analysis while accounting for possible missing data.
</h2><span id='topic+indelmiss-package'></span><span id='topic+indelmiss'></span>

<h3>Description</h3>

<p>Genome-wide gene insertion and deletion rates can be modelled in a maximum 
likelihood framework with the additional flexibility of modelling potential missing 
data using the models included within. These models simultaneously estimate insertion 
and deletion (indel) rates of gene families and proportions of &quot;missing&quot; data for 
(multiple) taxa of interest. The likelihood framework is utilized for parameter 
estimation. A phylogenetic tree of the taxa and gene presence/absence patterns 
(with data ordered by the tips of the tree) are required. For more details, see 
Utkarsh J. Dang, Alison M. Devault, Tatum D. Mortimer, Caitlin S. Pepperell, 
Hendrik N. Poinar, G. Brian Golding (2016). Gene insertion deletion analysis 
while accounting for possible missing data. Genetics (accepted).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> indelmiss</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.7</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-08-21</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a>
</p>


<h3>References</h3>

  
<p>Eddelbuettel, Dirk and Romain Francois (2011). Rcpp: Seamless R and C++ Integration. Journal of Statistical Software, 40(8), 1&ndash;18.
</p>
<p>Felsenstein, Joseph. Inferring phylogenies. Vol. 2. Sunderland: Sinauer Associates, 2004.
</p>
<p>Gilbert, Paul and Ravi Varadhan (2012). numDeriv: Accurate Numerical Derivatives. R package version 2012.9-1.
</p>
<p>Hao, Weilong, and G. Brian Golding. &quot;The fate of laterally transferred genes: life in the fast lane to adaptation or death.&quot; Genome Research 16.5 (2006): 636&ndash;643.
</p>
<p>Paradis E., Claude J. &amp; Strimmer K. 2004. APE: analyses of phylogenetics and evolution in R language. Bioinformatics 20: 289-290. R package version 3.2.
</p>
<p>Schliep K.P. 2011. phangorn: phylogenetic analysis in R. Bioinformatics, 27(4) 592-593. R package version 1.99.11.
</p>
<p>Wickham, Hadley (2012). stringr: Make it easier to work with strings. R package version 0.6.2.
</p>

<hr>
<h2 id='gardnerelladata'>
Gardnerella vaginalis phyletic data (gene family memberships) and phylogenetic tree
</h2><span id='topic+gardnerelladata'></span>

<h3>Description</h3>

<p>Gene family memberships for thirty five species from the genus <em>Gardnerella vaginalis</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gardnerelladata")</code></pre>


<h3>Format</h3>

<p>Contains a list that comprises a tree (called &quot;tree&quot;) and phyletic gene family membership data (called &quot;phyl&quot;) as its components. The tree is in the ape package phylo format. The data component consists of a matrix of 0/1 patterns with the different patterns as the rows and the 35 taxa as the columns. An entry of 1 (0) describe gene presence (absence) in the taxa.
</p>


<h3>References</h3>

  
<p>Devault, A. (2014). Genomics of Ancient Pathogenic Bacteria: Novel Techniques &amp; Extraordinary Substrates. Ph. D. thesis, McMaster University, Hamilton.
</p>
<p>Devault, A. M., T. D. Mortimer, H. Kiesewetter, T. Smith, M. Kuch, A. Hussain, J. M. Enk, G. B. Golding, W. Aylward, H. N. Poinar, and C. S. Pepperell (2016). Ancient DNA analysis of a unique calcified urogenital infection from medieval Troy reveals insights into past female health. Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gardnerelladata)
</code></pre>

<hr>
<h2 id='indelrates'>
Function for estimating indel rates while accounting for missing data.
</h2><span id='topic+indelrates'></span>

<h3>Description</h3>

<p>This is the function used for running the four gene gain/loss rate models. The four models being run are &quot;M1&quot;, &quot;M2&quot;, &quot;M3&quot;, and &quot;M4&quot;. The first model only estimates indel rates where both the insertion and deletion rates are the same. The second model tries to account for possible missing data while estimating indel rates. The third model estimates insertion and deletion rates separately. The fourth model tries to account for possible missing data while estimating insertion and deletion rates separately. See modelnames in the arguments below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indelrates(verbose = FALSE, usertree = NULL, userphyl = NULL, 
          matchtipstodata = FALSE, datasource = "user", seed = 1, taxa = 5, 
          brlensh = c(1, 4), mu = 1, nu = 1, phyl = 5000, 
          pmiss = 0, toi = 1, bgtype = "listofnodes", bg = NULL,  
          zerocorrection = TRUE, rootprob = "stationary", rpvec = NULL,
          optmethod = "nlminb", init = 0.9, lowlim = 0.001, uplim = 100,
          numhessian = TRUE, modelnames = c("M1", "M2", "M3", "M4"),...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="indelrates_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, will print out estimates and standard errors from each model. Note that the order of printing is the deletion rate followed by the insertion rate. Default value is FALSE.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_usertree">usertree</code></td>
<td>
<p>Used with datasource = &quot;user&quot;. Rooted binary tree of class &quot;phylo&quot;. Read in Newick tree using read.tree() from package ape before passing this argument. The branch lengths must be in expected substitutions per site. Popular tree estimation programs like MrBayes and BEAST, for instance, yield branch lengths on that scale.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_userphyl">userphyl</code></td>
<td>
<p>Used with datasource = &quot;user&quot;. Matrix of gene phyletic patterns. Matrix entries denote presence (1) or absence (0) of a specific gene (rows) for an individual taxon (columns).</p>
</td></tr>
<tr><td><code id="indelrates_+3A_matchtipstodata">matchtipstodata</code></td>
<td>

<p>The default is FALSE, which means that the user must ensure that the ordering of the taxa in the data matrix must match the internal ordering of tip labels of the tree. Set to TRUE, if the column names of the data matrix, i.e., the taxa names, are all present, with the same spelling and notation in the tip labels of the tree provided, in which case, the restriction on ordering is not necessary.
</p>
</td></tr>
<tr><td><code id="indelrates_+3A_datasource">datasource</code></td>
<td>
<p>&quot;simulation&quot; or &quot;user&quot;. Default is &quot;user&quot;.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_seed">seed</code></td>
<td>
<p>seed number for replication (default is 1).</p>
</td></tr>
<tr><td><code id="indelrates_+3A_taxa">taxa</code></td>
<td>
<p>Used with datasource = &quot;simulation&quot;. Total number of taxa (default is 5). Tree is generated using ape package.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_brlensh">brlensh</code></td>
<td>
<p>Used with datasource = &quot;simulation&quot;. Branch lengths are simulated from a beta distribution. The shape parameters can be provided as a vector (default = c(1, 4)). For simulating closely related sequences (i.e., with smaller branch lengths more important in trees with more taxa), use something like c(1, 8).</p>
</td></tr>
<tr><td><code id="indelrates_+3A_mu">mu</code></td>
<td>
<p>(unstandardized) deletion rate (default is 1). Used with datasource = &quot;simulation&quot;. Sequences given the trees are generated after the rate matrix is standardized such that one unit of time is what we expect to see one change per gene site in. (cf. Equation 13.14 in Felsenstein, 2004.)</p>
</td></tr>
<tr><td><code id="indelrates_+3A_nu">nu</code></td>
<td>
<p>(unstandardized) insertion rate (default is 1). Used with datasource = &quot;simulation&quot;. An example of the conversion that phangorn implements is simulating with mu = 0.6 and nu = 3 when given mu = 1 and nu = 5. Similarly, phangorn simulates with mu = 0.67 (0.75) and nu = 2 (1.5) when given mu = 1 (1) and nu = 3 (2).</p>
</td></tr>
<tr><td><code id="indelrates_+3A_phyl">phyl</code></td>
<td>
<p>Used with datasource = &quot;simulation&quot;. Total number of gene phyletic patterns (default is 5000).</p>
</td></tr>
<tr><td><code id="indelrates_+3A_pmiss">pmiss</code></td>
<td>
<p>Proportion of present genes to remove in the taxa of interest for the purpose of simulating &quot;missingness&quot;. Primarily provided for used with datasource = &quot;simulation&quot; but can be used with datasource = &quot;user&quot; as well.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_toi">toi</code></td>
<td>
<p>A vector of tiplabels for taxa of interest as determined by using ape::tiplabels() function while plotting a tree of class &quot;phylo&quot;. Models &quot;M2&quot; and &quot;M4&quot; will account for possible missing data for these taxa. Using toi = &quot;all&quot; specifies all tips (although this often leads to over-parameterization). Default value is tip 1.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_bgtype">bgtype</code></td>
<td>
<p>If clade-specific insertion and deletion rates are required to be estimated, specify as &quot;ancestornodes&quot;. If, on the other hand, a group of branches (not in a clade) are hypothesized to follow the same rates, use argument option &quot;listofnodes&quot;.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_bg">bg</code></td>
<td>
<p>A vector of nodes can be given here if the &quot;ancestornodes&quot; option was chosen for argument &quot;bgtype&quot;. If &quot;listofnodes&quot; was chosen, a list should be provided with each element of the list being a vector of nodes that limit the branches that follow the same rates. See examples.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_zerocorrection">zerocorrection</code></td>
<td>
<p>Felsenstein's correction for unobserved data. The results are conditional on observing the gene present in at least one taxa. This accounts for the sampling bias. Default is TRUE.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_rootprob">rootprob</code></td>
<td>
<p>Four options are available for the prior probability of character state frequencies at the root. The &quot;equal&quot; option makes gene absence and presence contribute equally to the likelihood calculation at the root of the tree. The &quot;stationary&quot; option will weight the contributions by the averaged equilibrium frequencies of all different branch groupings. The &quot;maxlik&quot; option estimates the probability of gene family absence at the root. Default is &quot;stationary&quot;. If &quot;user&quot; is supplied here (e.g., for empirical frequencies), a vector of root frequency parameters can be provided to argument &quot;rpvec&quot;.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_rpvec">rpvec</code></td>
<td>
	
<p>If option &quot;user&quot; is specified for argument &quot;rootprob&quot;, supply a vector of length two, representing the root frequency parameters.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_optmethod">optmethod</code></td>
<td>
<p>&quot;nlminb&quot; (default) or &quot;optim&quot;. The &quot;indelrate&quot; model being a one-dimensional problem always uses &quot;nlminb&quot;. The &quot;optim&quot; option is provided purely as a fall-back; in most cases, &quot;nlminb&quot; will converge to the correct solution more stably and quickly as compared to &quot;optim&quot;.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_init">init</code></td>
<td>
	
<p>Initial value for the rates. The default value is 0.9.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_lowlim">lowlim</code></td>
<td>
	
<p>For finer control of the boundaries of the optimization problem. The default value is 0.001. This usually suffices if the branch lengths are in expected substitutions per site. However, if branch lengths are in different units, this should be changed accordingly.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_uplim">uplim</code></td>
<td>

<p>For finer control of the boundaries of the optimization problem. The default value is 100. This usually suffices if the branch lengths are in expected substitutions per site. However, if branch lengths are in different units, this should be changed accordingly.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_numhessian">numhessian</code></td>
<td>
<p>Set to FALSE if standard errors are not required (or standard errors are being calculated using bootstrapping). This speeds up the algorithm. Default is TRUE.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_modelnames">modelnames</code></td>
<td>
<p>Default is all models. The four options are &quot;M1&quot;, &quot;M2&quot;, &quot;M3&quot;, and &quot;M4&quot;. A vector of these values can be given. The first model only estimates indel rates where both the insertion and deletion rates are the same. The second model tries to account for possible missing data while estimating indel rates. The third model estimates insertion and deletion rates separately. The fourth model tries to account for possible missing data while estimating insertion and deletion rates separately.</p>
</td></tr>
<tr><td><code id="indelrates_+3A_...">...</code></td>
<td>
<p>Passing other arguments of the optimization algorithms used. See &quot;nlminb&quot; or &quot;optim&quot; documentation. For example, control = list(trace = 5) with method = &quot;nlminb&quot; will print progress at every 5th iteration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Gene presence/absence should be coded as 1/0. By default, any datapoints (in the data supplied) greater than 1 are changed to 1 and any rows consisting of zeros only are removed. Gene presence/ absence patterns should be ordered by the tips of the tree.
</p>


<h3>Value</h3>

<p>All arguments used while calling the indelrates function are attached in a list. Moreover, the following components are also returned:
</p>
<table role = "presentation">
<tr><td><code>call</code></td>
<td>
<p>Function call used.</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>A vector of convergence indicators for each model run. 0 denotes successful convergence.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Time taken in seconds.</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>The phylogenetic tree used.</p>
</td></tr>
<tr><td><code>bg</code></td>
<td>
<p>List of group of nodes modelled with individual insertion and deletion rates.</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>List of results including modelname, parameter estimates: insertion (&quot;nu&quot;) and deletion (&quot;mu&quot;) rates and proportion of missing data (&quot;p&quot;), standard errors, number of parameters fit, and AIC and BIC values. For the models with missing data proportions, an estimate (rounded off) of the number of genes missing for the taxa of interest specified is also provided. Furthermore, details from the optimization routine applied are also available.</p>
</td></tr>
<tr><td><code>data_red</code></td>
<td>
<p>Gene phyletic patterns observed.</p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>Number of times each gene phyletic pattern was observed.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+print.indelmiss">print.indelmiss</a></code>, <code><a href="#topic+plot.indelmiss">plot.indelmiss</a></code>, and <code><a href="#topic+plottree">plottree</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###User supplied tree and data###
#Simulate data
library(phangorn)
set.seed(1)
usertree &lt;- rtree(n = 7, br = rbeta(n = 7, shape1 = 1, shape2 = 7))
data &lt;- simSeq(usertree, l = 5000, type = "USER", levels = c(0, 1), 
bf = c(1/(1 + 5), 5/(1 + 5)), Q = 1) #1 and 5 correspond to 
#unstandardized rates. See item help descriptions on mu and nu.
datab &lt;- matrix(as.numeric(as.character(data)), nrow = 7)
userphyl &lt;- t(datab)
#Run the models.
indel_user &lt;- indelrates(datasource = "user", usertree = usertree, 
userphyl = userphyl, toi = 1, zerocorrection = TRUE, rootprob = "stationary", 
  modelnames = c("M3", "M4"), optmethod = "nlminb", 
control = list(trace = 10))
print(indel_user)

#####Simulation#####
#Simulate a dataset with default options and run algorithm.
indel1 &lt;- indelrates(verbose = TRUE, datasource = "simulation", 
control = list(trace = 5))
print(indel1)

#Estimate insertion/ deletion rates from gene presence/absence 
#data simulated on a simulated five taxon tree.
indel2 &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5, 
brlensh = c(1, 8), mu = 1, nu = 5, phyl = 5000, pmiss = 0, toi = 1, 
zerocorrection = TRUE, rootprob = "stationary", 
modelnames = c("M1", "M2", "M3", "M4"), optmethod = "nlminb", 
control = list(trace = 5))#1 and 5 correspond to unstandardized rates. 
#See item help descriptions on mu and nu.
print(indel2)

#With toi="all"
indel3 &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5, 
brlensh = c(1, 8), mu = 1, nu = 5, phyl = 5000, pmiss = c(0, 0.15, 0.25, 0, 0), toi = "all", 
zerocorrection = TRUE, rootprob = "maxlik", modelnames = c("M3", "M4"),
optmethod = "nlminb")
print(indel3)
#Compare with
indel3 &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5, 
brlensh = c(1, 8), mu = 1, nu = 5, phyl = 5000, pmiss = c(0.15, 0.25), toi = c(2, 3), 
zerocorrection = TRUE, rootprob = "maxlik", modelnames = c("M3", "M4"),
optmethod = "nlminb")
print(indel3)

#Here, a vector of ancestor nodes specify the nodes which 
#along with all their descendants have unique indel rates.

indel4 &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 10, 
brlensh = c(1, 8), mu = 1, nu = 5, phyl = 5000, pmiss = 0, toi = 1, 
bgtype = "ancestornodes", bg = c(15), zerocorrection = TRUE, rootprob = 
"maxlik", modelnames = c("M3", "M4"), optmethod = "nlminb")
print(indel4)
plot(indel4, model = "M4")

#Above command prints two plots that can be obtained individually.
#These are confidence intervals based on asymptotic normality 
#of the maximum likelihood estimators. 
#Different confidence interval levels can be specified with the cil option.
plotrates(indel4, model = "M4", ci = TRUE, cil = 95)
plotp(indel4, model = "M4", ci = TRUE, cil = 95)

#This is an alternate (more flexible but potentially less user-friendly) 
#way to specify groups of nodes which have unique indel rates. 
#A list of nodes is used here.

indel5 &lt;- indelrates(verbose = TRUE, datasource = "simulation", seed = 1, 
taxa = 5, brlensh = c(1, 8), mu = 1, nu = 3, phyl = 5000, pmiss = 0, 
toi = 1, bgtype = "listofnodes", bg = list(c(7, 1, 2), 
c(6, 8, 3, 7, 9, 5, 4, 9)), zerocorrection = TRUE, rootprob = "maxlik",
modelnames = c("M1", "M2", "M3", "M4"), optmethod = "nlminb")

#Mycobacterium data example
 data(mycobacteriumdata1)
 indel_myco &lt;- indelrates(verbose = TRUE, usertree = mycobacteriumdata1$tree, modelnames = "M4",
    userphyl = mycobacteriumdata1$phyl, matchtipstodata = TRUE, 
    datasource = "user", toi = c(3:4, 6:10), bgtype = "listofnodes", 
    zerocorrection = TRUE, rootprob = "stationary", optmethod = "nlminb", 
    numhessian = TRUE, control = list(eval.max = 50000, iter.max = 50000))

</code></pre>

<hr>
<h2 id='mycobacteriumdata1'>
Mycobacterium data phyletic data (gene family memberships) and tree
</h2><span id='topic+mycobacteriumdata1'></span>

<h3>Description</h3>

<p>Gene family memberships and a phylogenetic tree for ten species from the genus <em>Mycobacterium</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mycobacteriumdata1")</code></pre>


<h3>Format</h3>

<p>Contains a list that comprises a tree (called &quot;tree&quot;) and phyletic gene family membership data (called &quot;phyl&quot;) as its components. The tree is in the ape package phylo format. The data component consists of a data frame of 0/1 patterns with the different patterns as the rows and the 10 taxa as the columns. An entry of 1 (0) describe gene presence (absence) in the taxa.
</p>


<h3>References</h3>

  
<p>O'Neill, M. B., T. D. Mortimer, and C. S. Pepperell (2015). Diversity of mycobacterium tuberculosis across evolutionary scales. PLoS Pathogens 11(11), e1005257.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mycobacteriumdata1)
</code></pre>

<hr>
<h2 id='mycobacteriumdata2'>
Alternate Mycobacterium tree.
</h2><span id='topic+mycobacteriumdata2'></span>

<h3>Description</h3>

<p>Gene family memberships and a phylogenetic tree for ten species from the genus <em>Mycobacterium</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("mycobacteriumdata2")</code></pre>


<h3>Format</h3>

<p>Contains a list that comprises a tree (called &quot;tree&quot;) and phyletic gene family membership data (called &quot;phyl&quot;) as its components. The tree is in the ape package phylo format. The data component consists of a data frame of 0/1 patterns with the different patterns as the rows and the 10 taxa as the columns. An entry of 1 (0) describe gene presence (absence) in the taxa.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mycobacteriumdata2)
</code></pre>

<hr>
<h2 id='plot.indelmiss'>
Plot parameter estimates from the model fit
</h2><span id='topic+plot.indelmiss'></span>

<h3>Description</h3>

<p>Plotting command for use on an object of class &quot;indelmiss&quot;. Will draw two plots: one with the estimates for the rates and the other for the &quot;missingness&quot; parameter. plot.indelmiss() calls plotp and plotrates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'indelmiss'
plot(x, model = NULL, ci = TRUE, cil = 95, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.indelmiss_+3A_x">x</code></td>
<td>
<p>An object of class &quot;indelmiss&quot;.</p>
</td></tr>
<tr><td><code id="plot.indelmiss_+3A_model">model</code></td>
<td>
<p>One of &quot;M1&quot;, &quot;M2&quot;, &quot;M3&quot;, or &quot;M4&quot;. </p>
</td></tr>
<tr><td><code id="plot.indelmiss_+3A_ci">ci</code></td>
<td>
<p>TRUE plots confidence intervals around the estimates.</p>
</td></tr>
<tr><td><code id="plot.indelmiss_+3A_cil">cil</code></td>
<td>
<p>Confidence interval level.</p>
</td></tr>
<tr><td><code id="plot.indelmiss_+3A_...">...</code></td>
<td>
<p>Any further commands to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> Draws two plots: one with the estimates for the rates and the other for the &quot;missingness&quot; parameter. </p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+indelrates">indelrates</a></code>, <code><a href="#topic+plotrates">plotrates</a></code> and <code><a href="#topic+plotp">plotp</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indel &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5,
             mu = 1, nu = 5, phyl = 5000,  pmiss = 0, toi = 1, 
             bgtype="ancestornodes", bg = c(7, 9), 
             zerocorrection = TRUE, rootprob="maxlik",  
             modelnames = c("M1", "M2", "M3", "M4"),
             optmethod = "nlminb")
print(indel)
plot(indel, model="M4") 
</code></pre>

<hr>
<h2 id='plotp'>
Plot estimates for the parameter that accounts for possible missing data
</h2><span id='topic+plotp'></span>

<h3>Description</h3>

<p>Plotting command for use on an object of class &quot;indelmiss&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotp(x, model, ci = TRUE, cil = 95, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotp_+3A_x">x</code></td>
<td>
<p>An object of class &quot;indelmiss&quot;.</p>
</td></tr>
<tr><td><code id="plotp_+3A_model">model</code></td>
<td>
<p>One of &quot;M2&quot; or &quot;M4&quot;. </p>
</td></tr>
<tr><td><code id="plotp_+3A_ci">ci</code></td>
<td>
<p>TRUE plots confidence intervals around the estimates.</p>
</td></tr>
<tr><td><code id="plotp_+3A_cil">cil</code></td>
<td>
<p>Confidence interval level.</p>
</td></tr>
<tr><td><code id="plotp_+3A_...">...</code></td>
<td>
<p>Any further commands to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> Plot estimates for the parameter that accounts for possible missing data. Called by plot.indelmiss. </p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+plot.indelmiss">plot.indelmiss</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indel &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5,
             mu = 1, nu = 5, phyl = 5000,  pmiss = 0, toi = 1, 
             bgtype="ancestornodes", bg = c(7, 9), 
             zerocorrection = TRUE, rootprob = "stationary", 
             modelnames = c("M1", "M2", "M3", "M4"),
             optmethod = "nlminb")
print(indel)
plotp(indel, model="M4") 
</code></pre>

<hr>
<h2 id='plotrates'>
Plot estimates for insertion and deletion rates
</h2><span id='topic+plotrates'></span>

<h3>Description</h3>

<p>Plotting command for use on an object of class &quot;indelmiss&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotrates(x, model, ci = TRUE, cil = 95, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plotrates_+3A_x">x</code></td>
<td>
<p>An object of class &quot;indelmiss&quot;.</p>
</td></tr>
<tr><td><code id="plotrates_+3A_model">model</code></td>
<td>
<p>One of &quot;M1&quot;, &quot;M2&quot;, &quot;M3&quot;, or &quot;M4&quot;. </p>
</td></tr>
<tr><td><code id="plotrates_+3A_ci">ci</code></td>
<td>
<p>TRUE plots confidence intervals around the estimates.</p>
</td></tr>
<tr><td><code id="plotrates_+3A_cil">cil</code></td>
<td>
<p>Confidence interval level.</p>
</td></tr>
<tr><td><code id="plotrates_+3A_...">...</code></td>
<td>
<p>Any further commands to plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> Plot estimates for insertion and deletion rates. Called by plot.indelmiss. </p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+plot.indelmiss">plot.indelmiss</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indel &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5,
             mu = 1, nu = 5, phyl = 5000,  pmiss = 0, toi = 1, 
             bgtype="ancestornodes", bg = c(7, 9), 
             zerocorrection = TRUE, 
             modelnames = c("M1", "M2", "M3", "M4"),
             optmethod = "nlminb")
print(indel)
plotrates(indel, model="M4") 
</code></pre>

<hr>
<h2 id='plottree'>
Plot the tree used the branches colored according to the different specified branch groupings (or clades) following unique rates.
</h2><span id='topic+plottree'></span>

<h3>Description</h3>

<p>Plotting command for use on an object of class &quot;indelmiss&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plottree(x, toilabel = TRUE, colors = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plottree_+3A_x">x</code></td>
<td>
<p>An object of class &quot;indelmiss&quot;.</p>
</td></tr>
<tr><td><code id="plottree_+3A_toilabel">toilabel</code></td>
<td>
<p>If this is TRUE, a plus sign is printed next to the taxa of interest for which a missing data proportion was estimated. Note that the taxa labels being referred to can be seen by using ape::tiplabels().</p>
</td></tr>
<tr><td><code id="plottree_+3A_colors">colors</code></td>
<td>
<p>Vector of colours the same length as length(x$bg). Note that these colours are used to colour the different branch groupings associated with unique insertion and/or deletion rates.</p>
</td></tr>
<tr><td><code id="plottree_+3A_...">...</code></td>
<td>
<p>Any further commands to ape::plot.phylo.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> Plot the tree used the branches colored according to the different specified branch groupings (or clades) following unique rates. </p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+plot.indelmiss">plot.indelmiss</a></code> and <code><a href="ape.html#topic+plot.phylo">plot.phylo</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indel &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5,
             mu = 1, nu = 5, phyl = 5000,  pmiss = 0, toi = 1, 
             bgtype="ancestornodes", bg = 7, 
             zerocorrection = TRUE, 
             modelnames = c("M1", "M2", "M3", "M4"),
             optmethod = "nlminb")
print(indel)
plottree(indel,colors=c("blue","red"))
ape::tiplabels()
</code></pre>

<hr>
<h2 id='print.indelmiss'>
Print summary information from fit
</h2><span id='topic+print.indelmiss'></span>

<h3>Description</h3>

<p>Summary command for use on an object of class &quot;indelmiss&quot;. Depending on the model, the rates (mu: deletion; nu: insertion), missing data proportion (p), and prior probability of gene family absence at the root are printed. If branch groupings (or clades) were specified, then the rates (and corresponding standard errors) are displayed in a matrix with the columns representing the different branch groupings (ordered by the subsets of x$bg where x is an object of class &quot;indelmiss&quot;). The rows represent the gene deletion and insertion rate, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'indelmiss'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.indelmiss_+3A_x">x</code></td>
<td>
<p>An object of class &quot;indelmiss&quot;.</p>
</td></tr>
<tr><td><code id="print.indelmiss_+3A_...">...</code></td>
<td>
<p>Ignore this.</p>
</td></tr>
</table>


<h3>Value</h3>

<p> Output summary information from the fit. </p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+indelrates">indelrates</a></code> and <code><a href="#topic+plot.indelmiss">plot.indelmiss</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>indel &lt;- indelrates(datasource = "simulation", seed = 1, taxa = 5,
             mu = 1, nu = 3, phyl = 5000,  pmiss = c(0.2, 0.5), toi = c(1, 3), 
             zerocorrection = TRUE, 
             modelnames = c("M1", "M2", "M3", "M4"),
             optmethod = "nlminb")
print(indel)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
