<!DOCTYPE html><html><head><title>Help for package iNEXT.3D</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iNEXT.3D}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#iNEXT.3D-package'>
<p>Interpolation and Extrapolation for three dimensions of biodiversity</p></a></li>
<li><a href='#Brazil_rainforest_abun_data'>
<p>Abundance data (datatype = &quot;abundance&quot;)</p></a></li>
<li><a href='#Brazil_rainforest_distance_matrix'>
<p>Species pairwise distance matrix for species in the dataset <code>Brazil_rainforest_abun_data</code></p></a></li>
<li><a href='#Brazil_rainforest_phylo_tree'>
<p>Phylogenetic tree for <code>Brazil_rainforest_abun_data</code></p></a></li>
<li><a href='#DataInfo3D'><p>Data information for reference samples</p></a></li>
<li><a href='#estimate3D'><p>Compute 3D diversity estimates with a particular set of sample sizes/coverages</p></a></li>
<li><a href='#Fish_distance_matrix'>
<p>Species pairwise distance matrix for species in the dataset <code>Fish_incidence_data</code></p></a></li>
<li><a href='#Fish_incidence_data'>
<p>Incidence data (datatype = &quot;incidence_raw&quot;)</p></a></li>
<li><a href='#Fish_phylo_tree'>
<p>Phylogenetic tree for <code>Fish_incidence_data</code></p></a></li>
<li><a href='#ggiNEXT3D'><p>ggplot2 extension for an iNEXT3D object</p></a></li>
<li><a href='#ggObsAsy3D'><p>ggplot2 extension for plotting q-profile, time-profile, and tau-profile</p></a></li>
<li><a href='#iNEXT3D'><p>iNterpolation and EXTrapolation with three dimensions of biodiversity</p></a></li>
<li><a href='#ObsAsy3D'><p>Asymptotic diversity and observed diversity of order q</p></a></li>
<li><a href='#print.iNEXT3D'><p>Printing iNEXT3D object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interpolation and Extrapolation for Three Dimensions of
Biodiversity</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Author:</td>
<td>Anne Chao [aut, cre],
  KaiHsiang Hu [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anne Chao &lt;chao@stat.nthu.edu.tw&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://sites.google.com/view/chao-lab-website/software/inext-3d">https://sites.google.com/view/chao-lab-website/software/inext-3d</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Biodiversity is a multifaceted concept covering different levels of organization from
            genes to ecosystems. 'iNEXT.3D' extends 'iNEXT' to include three dimensions (3D) 
            of biodiversity, i.e., taxonomic diversity (TD), phylogenetic diversity (PD) and functional 
            diversity (FD). This package provides functions to compute standardized 3D diversity estimates 
            with a common sample size or sample coverage. A unified framework based on Hill numbers 
            and their generalizations (Hill-Chao numbers) are used to quantify 3D. All 3D estimates 
            are in the same units of species/lineage equivalents and can be meaningfully compared. 
            The package features size- and coverage-based rarefaction and extrapolation sampling 
            curves to facilitate rigorous comparison of 3D diversity across individual assemblages. 
            Asymptotic 3D diversity estimates are also provided. See Chao et al. (2021) 
            &lt;<a href="https://doi.org/10.1111%2F2041-210X.13682">doi:10.1111/2041-210X.13682</a>&gt; for more details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, tibble, ggplot2, reshape2, tidytree, phyclust, dplyr,
ape, Rcpp</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, gridExtra, ggthemes</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/AnneChao/iNEXT.3D/issues">https://github.com/AnneChao/iNEXT.3D/issues</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-28 06:13:58 UTC; stat-pc</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-28 16:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='iNEXT.3D-package'>
Interpolation and Extrapolation for three dimensions of biodiversity
</h2><span id='topic+iNEXT.3D-package'></span><span id='topic+iNEXT.3D'></span>

<h3>Description</h3>

<p><code>iNEXT.3D</code> (iNterpolation and EXTrapolation for three dimensions of biodiversity) is a sequel to <code>iNEXT</code> (Hsieh et al., 2016).
Here the three dimensions (3D) of diversity refer to taxonomic diversity (TD), phylogenetic diversity (PD) and functional diversity (FD). An online version &quot;iNEXT.3D Online&quot;
(https://chao.shinyapps.io/iNEXT_3D/) is also available for users without an R background.
</p>
<p>A unified framework based on Hill numbers (for TD) and their generalizations (Hill-Chao numbers, for PD and FD) 
is adopted to quantify 3D. In this framework, TD quantifies the effective number of species, PD quantifies the effective total branch length, mean-PD (PD divided by 
tree depth) quantifies the effective number of lineages, and FD quantifies the effective number of virtual functional 
groups (or functional &quot;species&quot;). Thus, TD, mean-PD, and FD are all in the same units of species/lineage equivalents 
and can be meaningfully compared; see Chao et al. (2014) for the basic standardization theory for TD, and Chao et al. (2021) for a review of the unified theory for 3D.
</p>
<p>For each of the three dimensions of biodiversity, <code>iNEXT.3D</code> features two statistical analyses (non-asymptotic and asymptotic):
</p>
<p>(1) A non-asymptotic approach based on interpolation and extrapolation for 3D diversity (i.e., Hill-Chao numbers)
</p>
<p><code>iNEXT.3D</code> computes the estimated 3D diversity for standardized samples with a common sample size or sample completeness. This approach aims to compare diversity estimates for equally-large (with a common sample size) or equally-complete (with a common sample coverage) samples; it is based on the seamless rarefaction and extrapolation (R/E) sampling curves of Hill-Chao numbers for q = 0, 1 and 2. For each dimension of biodiversity, <code>iNEXT.3D</code> offers three types of R/E sampling curves:
</p>
<p>* Sample-size-based (or size-based) R/E sampling curves: This type of sampling curve plots the diversity estimates with respect to sample size.
</p>
<p>* Coverage-based R/E sampling curves: This type of sampling curve plots the diversity estimates with respect to sample coverage.
</p>
<p>* Sample completeness curve: This curve depicts how sample coverage varies with sample size. The sample completeness curve provides a bridge between the size- and coverage-based R/E sampling curves.
</p>
<p>(2) An asymptotic approach to infer asymptotic 3D diversity (i.e., Hill-Chao numbers)
</p>
<p><code>iNEXT.3D</code> computes the estimated asymptotic 3D diversity and also plots 3D diversity profiles (q-profiles) for q between 0 and 2, in comparison with the observed diversity. Typically, the asymptotic estimates for q <code class="reqn">\geq</code> 1 are reliable, but for q &lt; 1 (especially for q = 0, species richness), the asymptotic estimates represent only lower bounds. <code>iNEXT.3D</code> also features a time-profile (which depicts the observed and asymptotic estimate of PD or mean PD with respect to reference times), and a tau-profile (which depicts the observed and asymptotic estimate of FD with respect to threshold level tau). <br />
</p>
<p>There are six main functions in this package: <br />
</p>
<p>1. <code>iNEXT3D</code> computes standardized 3D diversity of order q = 0, 1 and 2 for rarefied and extrapolated samples at specified sample coverage values and sample sizes. <br />
</p>
<p>2. <code>ggiNEXT3D</code> visualizes the output from the function <code>iNEXT3D</code>. <br />
</p>
<p>3. <code>ObsAsy3D</code> computes observed and asymptotic diversity of order q between 0 and 2 (in increments of 0.2) for 3D diversity; it also computes observed and asymptotic PD for specified reference times, and observed and asymptotic FD for specified threshold levels. <br />
</p>
<p>4. <code>ggObsAsy3D</code> visualizes the output from the function <code>ObsAsy3D</code>. <br />
</p>
<p>5. <code>estimate3D</code> computes 3D diversity of order q = 0, 1 and 2 with a particular set of user-specified level of sample sizes or sample coverage values. <br />
</p>
<p>6. <code>DataInfo3D</code> provides basic data information based on the observed data. <br />
</p>


<h3>Author(s)</h3>

<p>Anne Chao, Kai-Hsiang Hu <br />
</p>
<p>Maintainer: Anne Chao &lt;chao@stat.nthu.edu.tw&gt;
</p>


<h3>References</h3>

<p>Chao, A., Henderson, P. A., Chiu, C.-H., Moyes, F., Hu, K.-H., Dornelas, M. and Magurran, A. E. (2021). Measuring temporal change in alpha diversity: a framework integrating taxonomic, phylogenetic and functional diversity and the iNEXT.3D standardization. Methods in Ecology and Evolution, 12, 1926-1940. <br /><br />
Hsieh, T. C., Ma, K. H. and Chao, A. (2016). iNEXT: An R package for rarefaction and extrapolation of species diversity (Hill numbers). Methods in Ecology and Evolution, 7, 1451-1456. <br /><br />
</p>

<hr>
<h2 id='Brazil_rainforest_abun_data'>
Abundance data (datatype = &quot;abundance&quot;)
</h2><span id='topic+Brazil_rainforest_abun_data'></span>

<h3>Description</h3>

<p>This dataset includes tree species abundances in two assemblages/habitats (Edge and Interior) collected from rainforests in Brazil by Magnago et al. (2014, 2015, 2017). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Brazil_rainforest_abun_data)</code></pre>


<h3>Format</h3>

<p><code>Brazil_rainforest_abun_data</code> is a species-by-assemblages data.frame (in addition to species names and assemblage names) with 425 tree species and two assemblages (Edge and Interior). <br />
$ Edge: int 11 110 36 12 2 7 11 1 1 0 ... <br />
$ Interior: int 21 11 7 1 0 1 11 13 1 2 ... <br />
</p>


<h3>References</h3>

<p>Magnago, L. F. S., Edwards, D. P., Edwards, F. A., Magrach, A., Martins, S. V., &amp; Laurance, W. F. (2014). Functional attributes change but functional richness is unchanged after fragmentation of Brazilian Atlantic forests. Journal of Ecology, 102, 475-485. <br /><br />
Magnago, L. F. S., Magrach, A., Barlow, J., Schaefer, C. E. G. R., Laurance, W. F., Martins, S. V., &amp; Edwards, D. P. (2017). Do fragment size and edge effects predict carbon stocks in trees and lianas in tropical forests? Functional Ecology, 31, 542-552. <br /><br />
Magnago, L. F. S., Rocha, M. F., Meyer, L., Martins, S. V., &amp; Meira-Neto, J. A. A. (2015). Microclimatic conditions at forest edges have significant impacts on vegetation structure in large Atlantic forest fragments. Biodiversity and Conservation, 24, 2305-2318. <br /><br />
</p>

<hr>
<h2 id='Brazil_rainforest_distance_matrix'>
Species pairwise distance matrix for species in the dataset <code>Brazil_rainforest_abun_data</code>
</h2><span id='topic+Brazil_rainforest_distance_matrix'></span>

<h3>Description</h3>

<p>This dataset includes a tree species pairwise symmetric distance matrix for 425 species listed in the dataset <code>Brazil_rainforest_abun_data</code>. Each element in the matrix represents a Gower distance between two species computed from species traits. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Brazil_rainforest_distance_matrix)</code></pre>


<h3>Format</h3>

<p><code>Brazil_rainforest_distance_matrix</code> is a 425 x 425 (species by species) symmetric data.frame. Each element of the data.frame is between zero (for species pairs with identical traits) and one. <br />
$ Carpotroche_brasiliensis: num 0 0.522 0.522 0.253 0.263 ... <br />
$ Astronium_concinnum: num 0.522 0 0 0.525 0.509 ... <br />
$ Astronium_graveolens: num 0.522 0 0 0.525 0.509 ... <br />
....... <br />
</p>

<hr>
<h2 id='Brazil_rainforest_phylo_tree'>
Phylogenetic tree for <code>Brazil_rainforest_abun_data</code>
</h2><span id='topic+Brazil_rainforest_phylo_tree'></span>

<h3>Description</h3>

<p>This dataset includes a phylogenetic tree spanned by 425 species listed in the dataset <code>Brazil_rainforest_abun_data</code>. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Brazil_rainforest_phylo_tree)</code></pre>


<h3>Format</h3>

<p><code>Brazil_rainforest_phylo_tree</code> is a list with the following phylogenetic information: <br />
A list of 5: <br />
$ edge: int [1:629, 1:2] 426 427 428 429 430 431 427 428 429 430 ... <br />
$ edge.length: num [1:629] 75 146 8 6 4 14 10 10 127 6 ... <br />
$ Nnode: int 205 <br />
$ node.label: chr [1:205] &quot;magnoliales_to_asterales&quot; &quot;poales_to_asterales&quot; ... <br />
$ tip.label: chr [1:425] &quot;Carpotroche_brasiliensis&quot; &quot;Casearia_ulmifolia&quot; &quot;Casearia_sp4&quot; ... <br />
</p>

<hr>
<h2 id='DataInfo3D'>Data information for reference samples</h2><span id='topic+DataInfo3D'></span>

<h3>Description</h3>

<p><code>DataInfo3D</code> provides basic data information for 3D diversity based on a reference sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DataInfo3D(
  data,
  diversity = "TD",
  datatype = "abundance",
  nT = NULL,
  PDtree,
  PDreftime = NULL,
  FDdistM,
  FDtype = "AUC",
  FDtau = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DataInfo3D_+3A_data">data</code></td>
<td>
<p>(a) For <code>datatype = "abundance"</code>, data can be input as a vector of species abundances (for a single assemblage), matrix/data.frame (species by assemblages), or a list of species abundance vectors. <br />
(b) For <code>datatype = "incidence_raw"</code>, data can be input as a list of matrices/data.frames (species by sampling units); data can also be input as a single matrix/data.frame by merging all sampling units across assemblages based on species identity; in this case, the number of sampling units (<code>nT</code>, see below) must be specified.</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_diversity">diversity</code></td>
<td>
<p>selection of diversity type: <code>'TD'</code> = Taxonomic diversity, <code>'PD'</code> = Phylogenetic diversity, and <code>'FD'</code> = Functional diversity.</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_datatype">datatype</code></td>
<td>
<p>data type of input data: individual-based abundance data (<code>datatype = "abundance"</code>) or species by sampling-units incidence/occurrence matrix (<code>datatype = "incidence_raw"</code>) with all entries being 0 (non-detection) or 1 (detection).</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_nt">nT</code></td>
<td>
<p>(required only when <code>datatype = "incidence_raw"</code> and input data in a single matrix/data.frame) a vector of positive integers specifying the number of sampling units in each assemblage. If assemblage names are not specified (i.e., <code>names(nT) = NULL</code>), then assemblages are automatically named as &quot;assemblage1&quot;, &quot;assemblage2&quot;,..., etc.</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_pdtree">PDtree</code></td>
<td>
<p>(required argument for <code>diversity = "PD"</code>), a phylogenetic tree in Newick format for all observed species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_pdreftime">PDreftime</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), a vector of numerical values specifying reference times for PD. Default is <code>NULL</code> (i.e., the age of the root of <code>PDtree</code>).</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_fddistm">FDdistM</code></td>
<td>
<p>(required argument for <code>diversity = "FD"</code>), a species pairwise distance matrix for all species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_fdtype">FDtype</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code>), select FD type: <code>FDtype = "tau_values"</code> for FD under specified threshold values, or <code>FDtype = "AUC"</code> (area under the curve of tau-profile) for an overall FD which integrates all threshold values between zero and one. Default is <code>"AUC"</code>.</p>
</td></tr>
<tr><td><code id="DataInfo3D_+3A_fdtau">FDtau</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "tau_values"</code>), a numerical vector between 0 and 1 specifying tau values (threshold levels). If <code>NULL</code> (default), then threshold is set to be the mean distance between any two individuals randomly selected from the pooled assemblage (i.e., quadratic entropy).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame including basic data information.<br /><br /> 
For abundance data, basic information shared by TD, mean-PD and FD includes assemblage name (<code>Assemblage</code>),
sample size (<code>n</code>), observed species richness (<code>S.obs</code>), sample coverage estimates of the reference sample (<code>SC(n)</code>), 
sample coverage estimate for twice the reference sample size (<code>SC(2n)</code>). Other additional information is given below.<br /><br />
(1) TD: the first five species abundance counts (<code>f1</code>&ndash;<code>f5</code>).<br /><br />
(2) Mean-PD: the observed total branch length in the phylogenetic tree (<code>PD.obs</code>), 
the number of singletons (<code>f1*</code>) and doubletons (<code>f2*</code>) in the node/branch abundance set, as well as the total branch length 
of those singletons (<code>g1</code>) and of those doubletons (<code>g2</code>), and the reference time (<code>Reftime</code>).<br /><br />
(3) FD (<code>FDtype = "AUC"</code>): the minimum distance among all non-diagonal elements in the distance matrix (<code>dmin</code>), the mean distance
(<code>dmean</code>), and the maximum distance (<code>dmax</code>) in the distance matrix.<br /> <br />
(4) FD (<code>FDtype = "tau_values"</code>): the number of singletons (<code>a1*</code>) and of doubletons (<code>a2*</code>) among the functionally indistinct
set at the specified threshold level <code>'Tau'</code>, as well as the total attribute contribution of singletons (<code>h1</code>) and of doubletons (<code>h2</code>)
at the specified threshold level <code>'Tau'</code>.<br /><br />
</p>
<p>For incidence data, the basic information for TD includes assemblage name (<code>Assemblage</code>), number of sampling units (<code>T</code>), 
total number of incidences (<code>U</code>), observed species richness (<code>S.obs</code>), 
sample coverage estimates of the reference sample (<code>SC(T)</code>), sample coverage estimate for twice the reference sample size
(<code>SC(2T)</code>), as well as the first five species incidence frequency counts (<code>Q1</code>&ndash;<code>Q5</code>). For mean-PD and FD, output is similar to that
for abundance data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Taxonomic diversity for abundance data
data(Brazil_rainforest_abun_data)
DataInfo3D(Brazil_rainforest_abun_data, diversity = 'TD', datatype = "abundance")


# Phylogenetic diversity for abundance data
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_phylo_tree)
data &lt;- Brazil_rainforest_abun_data
tree &lt;- Brazil_rainforest_phylo_tree
DataInfo3D(data, diversity = 'PD', datatype = "abundance", PDtree = tree)


# Functional diversity for abundance data with FDtype = 'AUC'
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
DataInfo3D(data, diversity = 'FD', datatype = "abundance", 
           FDdistM = distM, FDtype = 'AUC')
           
# Functional diversity for abundance data with FDtype = 'tau_values'
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
DataInfo3D(data, diversity = 'FD', datatype = "abundance", 
           FDdistM = distM, FDtype = 'tau_values')


# Taxonomic diversity for incidence data
data(Fish_incidence_data)
DataInfo3D(Fish_incidence_data, diversity = 'TD', datatype = "incidence_raw")


# Phylogenetic diversity for incidence data
data(Fish_incidence_data)
data(Fish_phylo_tree)
data &lt;- Fish_incidence_data
tree &lt;- Fish_phylo_tree
DataInfo3D(data, diversity = 'PD', datatype = "incidence_raw", PDtree = tree)


# Functional diversity for incidence data with FDtype = 'AUC'
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
DataInfo3D(data, diversity = 'FD', datatype = "incidence_raw", 
           FDdistM = distM, FDtype = 'AUC')
           
# Functional diversity for incidence data with FDtype = 'tau_values'
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
DataInfo3D(data, diversity = 'FD', datatype = "incidence_raw", 
           FDdistM = distM, FDtype = 'tau_values')


</code></pre>

<hr>
<h2 id='estimate3D'>Compute 3D diversity estimates with a particular set of sample sizes/coverages</h2><span id='topic+estimate3D'></span>

<h3>Description</h3>

<p><code>estimate3D</code> computes 3D diversity (Hill-Chao number with q = 0, 1 and 2) with a particular set of user-specified levels of sample sizes or sample coverages. If no sample sizes or coverages are specified, this function by default computes 3D diversity estimates for the minimum sample coverage or minimum sample size among all samples extrapolated to double reference sizes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate3D(
  data,
  diversity = "TD",
  q = c(0, 1, 2),
  datatype = "abundance",
  base = "coverage",
  level = NULL,
  nboot = 50,
  conf = 0.95,
  nT = NULL,
  PDtree,
  PDreftime = NULL,
  PDtype = "meanPD",
  FDdistM,
  FDtype = "AUC",
  FDtau = NULL,
  FDcut_number = 50
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate3D_+3A_data">data</code></td>
<td>
<p>(a) For <code>datatype = "abundance"</code>, data can be input as a vector of species abundances (for a single assemblage), matrix/data.frame (species by assemblages), or a list of species abundance vectors. <br />
(b) For <code>datatype = "incidence_raw"</code>, data can be input as a list of matrices/data.frames (species by sampling units); data can also be input as a single matrix/data.frame by merging all sampling units across assemblages based on species identity; in this case, the number of sampling units (<code>nT</code>, see below) must be specified.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_diversity">diversity</code></td>
<td>
<p>selection of diversity type: <code>'TD'</code> = Taxonomic diversity, <code>'PD'</code> = Phylogenetic diversity, and <code>'FD'</code> = Functional diversity.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_q">q</code></td>
<td>
<p>a numerical vector specifying the diversity orders. Default is <code>c(0, 1, 2)</code>.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_datatype">datatype</code></td>
<td>
<p>data type of input data: individual-based abundance data (<code>datatype = "abundance"</code>) or species by sampling-units incidence/occurrence matrix (<code>datatype = "incidence_raw"</code>) with all entries being 0 (non-detection) or 1 (detection).</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_base">base</code></td>
<td>
<p>selection of sample-size-based (<code>base = "size"</code>) or coverage-based (<code>base = "coverage"</code>) rarefaction and extrapolation.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_level">level</code></td>
<td>
<p>A numerical vector specifying the particular sample sizes or sample coverages (between 0 and 1) for which 3D diversity estimates (q =0, 1 and 2) will be computed. <br />
If <code>base = "coverage"</code> (default) and <code>level = NULL</code>, then this function computes the diversity estimates for the minimum sample coverage among all samples extrapolated to double reference sizes. <br />
If <code>base = "size"</code> and <code>level = NULL</code>, then this function computes the diversity estimates for the minimum sample size among all samples extrapolated to double reference sizes.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_nboot">nboot</code></td>
<td>
<p>a positive integer specifying the number of bootstrap replications when assessing sampling uncertainty and constructing confidence intervals. Enter 0 to skip the bootstrap procedures. Default is 50.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_conf">conf</code></td>
<td>
<p>a positive number &lt; 1 specifying the level of confidence interval. Default is 0.95.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_nt">nT</code></td>
<td>
<p>(required only when <code>datatype = "incidence_raw"</code> and input data in a single matrix/data.frame) a vector of positive integers specifying the number of sampling units in each assemblage. If assemblage names are not specified (i.e., <code>names(nT) = NULL</code>), then assemblages are automatically named as &quot;assemblage1&quot;, &quot;assemblage2&quot;,..., etc.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_pdtree">PDtree</code></td>
<td>
<p>(required argument for <code>diversity = "PD"</code>), a phylogenetic tree in Newick format for all observed species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_pdreftime">PDreftime</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), a vector of numerical values specifying reference times for PD. Default is <code>NULL</code> (i.e., the age of the root of <code>PDtree</code>).</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_pdtype">PDtype</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), select PD type: <code>PDtype = "PD"</code> (effective total branch length) or <code>PDtype = "meanPD"</code> (effective number of equally divergent lineages). Default is <code>"meanPD"</code>, where <code>meanPD = PD/tree depth</code>.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_fddistm">FDdistM</code></td>
<td>
<p>(required argument for <code>diversity = "FD"</code>), a species pairwise distance matrix for all species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_fdtype">FDtype</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code>), select FD type: <code>FDtype = "tau_values"</code> for FD under specified threshold values, or <code>FDtype = "AUC"</code> (area under the curve of tau-profile) for an overall FD which integrates all threshold values between zero and one. Default is <code>"AUC"</code>.</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_fdtau">FDtau</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "tau_values"</code>), a numerical vector between 0 and 1 specifying tau values (threshold levels). If <code>NULL</code> (default), then threshold is set to be the mean distance between any two individuals randomly selected from the pooled assemblage (i.e., quadratic entropy).</p>
</td></tr>
<tr><td><code id="estimate3D_+3A_fdcut_number">FDcut_number</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "AUC"</code>), a numeric number to cut [0, 1] interval into equal-spaced sub-intervals to obtain the AUC value by integrating the tau-profile. Equivalently, the number of tau values that will be considered to compute the integrated AUC value. Default is <code>FDcut_number = 50</code>. A larger value can be set to obtain more accurate AUC value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame of diversity table including the following arguments: (when <code>base = "coverage"</code>)
</p>
<table>
<tr><td><code>Assemblage</code></td>
<td>
<p>the name of assemblage.</p>
</td></tr>
<tr><td><code>Order.q</code></td>
<td>
<p>the diversity order of q.</p>
</td></tr>
<tr><td><code>SC</code></td>
<td>
<p>the target standardized coverage value.</p>
</td></tr>
<tr><td><code>m</code>, <code>mT</code></td>
<td>
<p>the corresponding sample size (or number of sampling units) for the standardized coverage value.</p>
</td></tr>
<tr><td><code>qTD</code>, <code>qPD</code>, <code>qFD</code></td>
<td>
<p>the estimated diversity of order q for the target coverage value. The estimate for complete coverage (when <code>base = "coverage"</code> and <code>level = 1</code>, or <code>base = "size"</code> and <code>level = Inf</code>) represents the estimated asymptotic diversity.</p>
</td></tr>
<tr><td><code>Method</code></td>
<td>
<p>Rarefaction, Observed, or Extrapolation, depending on whether the target coverage is less than, equal to, or greater than the coverage of the reference sample.</p>
</td></tr>
<tr><td><code>s.e.</code></td>
<td>
<p>standard error of diversity estimate.</p>
</td></tr>
<tr><td><code>qTD.LCL</code>, <code>qPD.LCL</code>, <code>qFD.LCL and qTD.UCL</code>, <code>qPD.UCL</code>, <code>qFD.UCL</code></td>
<td>
<p>the bootstrap lower and upper confidence limits for the diversity of order q at the specified level (with a default value of 0.95).</p>
</td></tr>
<tr><td><code>Reftime</code></td>
<td>
<p>the reference times for PD.</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p><code>"PD"</code> (effective total branch length) or <code>"meanPD"</code> (effective number of equally divergent lineages) for PD.</p>
</td></tr>
<tr><td><code>Tau</code></td>
<td>
<p>the threshold of functional distinctiveness between any two species for FD (under <code>FDtype = tau_values</code>).</p>
</td></tr>
</table>
<p>Similar output is obtained for <code>base = "size"</code>. <br /><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Taxonomic diversity for abundance data with two target coverages (93% and 97%)
data(Brazil_rainforest_abun_data)
output_est_TD_abun &lt;- estimate3D(Brazil_rainforest_abun_data, diversity = 'TD', q = c(0, 1, 2), 
                                 datatype = "abundance", base = "coverage", level = c(0.93, 0.97))
output_est_TD_abun


# Phylogenetic diversity for abundance data with two target sizes (1500 and 3500)
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_phylo_tree)
data &lt;- Brazil_rainforest_abun_data
tree &lt;- Brazil_rainforest_phylo_tree
output_est_PD_abun &lt;- estimate3D(data, diversity = 'PD', datatype = "abundance", 
                                 base = "size", level = c(1500, 3500), PDtree = tree)
output_est_PD_abun


# Functional diversity for abundance data with two target coverages (93% and 97%)
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_est_FD_abun &lt;- estimate3D(data, diversity = 'FD', datatype = "abundance", 
                                 base = "coverage", level = c(0.93, 0.97), nboot = 10, 
                                 FDdistM = distM, FDtype = 'AUC')
output_est_FD_abun


# Taxonomic diversity for incidence data with two target coverages (97.5% and 99%)
data(Fish_incidence_data)
output_est_TD_inci &lt;- estimate3D(Fish_incidence_data, diversity = 'TD', q = c(0, 1, 2), 
                                 datatype = "incidence_raw", base = "coverage", 
                                 level = c(0.975, 0.99))
output_est_TD_inci


# Phylogenetic diversity for incidence data with two target coverages (97.5% and 99%)
data(Fish_incidence_data)
data(Fish_phylo_tree)
data &lt;- Fish_incidence_data
tree &lt;- Fish_phylo_tree
output_est_PD_inci &lt;- estimate3D(data, diversity = 'PD', datatype = "incidence_raw", 
                                 base = "coverage", level = c(0.975, 0.99), PDtree = tree)
output_est_PD_inci


# Functional diversity for incidence data with two target number of sampling units (30 and 70)
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
output_est_FD_inci &lt;- estimate3D(data, diversity = 'FD', datatype = "incidence_raw", 
                                 base = "size", level = c(30, 70), nboot = 10, 
                                 FDdistM = distM, FDtype = 'AUC')
output_est_FD_inci



</code></pre>

<hr>
<h2 id='Fish_distance_matrix'>
Species pairwise distance matrix for species in the dataset <code>Fish_incidence_data</code>
</h2><span id='topic+Fish_distance_matrix'></span>

<h3>Description</h3>

<p>This dataset includes a species pairwise symmetric distance matrix for 88 species listed in the <code>Fish_incidence_data</code> dataset. Each element in the matrix represents a Gower distance between two species computed from species traits. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fish_distance_matrix)</code></pre>


<h3>Format</h3>

<p><code>Fish_distance_matrix</code> is a 88 x 88 (species by species) symmetric data.frame. Each element of the data.frame is between zero (for species pairs with identical traits) and one. <br />
$ Agonus_cataphractus: num 0 0.326 0.280 0.055 0.093 ... <br />
$ Alosa_fallax: num 0.326 0 0.380 0.336 0.311 ... <br />
$ Ammodytes_marinus: num 0.280 0.380 0 0.243 0.364 ... <br />
....... <br />
</p>

<hr>
<h2 id='Fish_incidence_data'>
Incidence data (datatype = &quot;incidence_raw&quot;)
</h2><span id='topic+Fish_incidence_data'></span>

<h3>Description</h3>

<p>This dataset includes incidence-based occurrence records of estuarine fishes collected at Bridgwater Bay in UK's Bristol Channel (Chao et al. 2021). There are two time periods (&quot;2013-2015&quot; and &quot;2016-2018&quot;). Each time period is designated as an assemblage. The purpose was to compare 3D diversity of the two time periods. In each time period, species incidence was recorded in 36 sampling units in each assemblage; each sampling unit represents a sampling date. This dataset is a list of two data frames with each representing species-by-sampling-unit raw incidence (detection/non-detection) matrix. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fish_incidence_data)</code></pre>


<h3>Format</h3>

<p><code>Fish_incidence_data</code> is a list of two species-by-sampling-unit data frames for 88 species and 36 sampling units. Each element in the data frame is 1 for a detection, and 0 for a non-detection. <br />
A list of 2: <br />
$ 2013-2015: int [1:88, 1:36] 0 0 0 0 0 0 0 0 0 0 ... <br />
$ 2016-2018: int [1:88, 1:36] 1 0 0 0 0 0 0 0 0 0 ... <br />
</p>


<h3>References</h3>

<p>Chao, A., Henderson, P. A., Chiu, C.-H., Moyes, F., Hu, K.-H., Dornelas, M. and Magurran, A.
E. (2021). Measuring temporal change in alpha diversity: a framework integrating taxonomic, phylogenetic and functional diversity and the iNEXT.3D standardization. Methods in Ecology and Evolution, 12, 1926-1940. <br /><br />
</p>

<hr>
<h2 id='Fish_phylo_tree'>
Phylogenetic tree for <code>Fish_incidence_data</code>
</h2><span id='topic+Fish_phylo_tree'></span>

<h3>Description</h3>

<p>This dataset includes a phylogenetic tree spanned by 88 species listed in the <code>Fish_incidence_data</code> dataset. <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Fish_phylo_tree)</code></pre>


<h3>Format</h3>

<p><code>Fish_phylo_tree</code> is a list with the following phylogenetic information: <br />
A list of 5: <br />
$ edge: int [1:172, 1:2] 89 90 91 92 93 94 95 95 94 96 ... <br />
$ edge.length: num [1:172] 0.023 0.057 0.023 0.851 0.011 0.023 0.011 0.011 0.023 0.011 ... <br />
$ Nnode: int 85 <br />
$ node.label: chr [1:85] &quot;Vertebrata_-subphylum_in_Deuterostomia&quot; ... <br />
$ tip.label: chr [1:88] &quot;Alosa_fallax&quot; &quot;Sardina_pilchardus&quot; &quot;Clupea_harengus&quot; ... <br />
</p>

<hr>
<h2 id='ggiNEXT3D'>ggplot2 extension for an iNEXT3D object</h2><span id='topic+ggiNEXT3D'></span>

<h3>Description</h3>

<p><code>ggiNEXT3D</code> is a <code>ggplot</code> extension for an <code>iNEXT3D</code> object to plot sample-size- and coverage-based rarefaction/extrapolation sampling curves along with a bridging sample completeness curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggiNEXT3D(output, type = 1:3, facet.var = "Assemblage", color.var = "Order.q")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggiNEXT3D_+3A_output">output</code></td>
<td>
<p>an <code>iNEXT3D</code> object computed by <code>iNEXT3D</code>.</p>
</td></tr>
<tr><td><code id="ggiNEXT3D_+3A_type">type</code></td>
<td>
<p>three types of plots: sample-size-based rarefaction/extrapolation curve (<code>type = 1</code>); 
sample completeness curve (<code>type = 2</code>); coverage-based rarefaction/extrapolation curve (<code>type = 3</code>).</p>
</td></tr>
<tr><td><code id="ggiNEXT3D_+3A_facet.var">facet.var</code></td>
<td>
<p>create a separate plot for each value of a specified variable: 
no separation (<code>facet.var = "None"</code>); 
a separate plot for each diversity order (<code>facet.var = "Order.q"</code>); 
a separate plot for each assemblage (<code>facet.var = "Assemblage"</code>); 
a separate plot for each combination of diversity order and assemblage (<code>facet.var = "Both"</code>).</p>
</td></tr>
<tr><td><code id="ggiNEXT3D_+3A_color.var">color.var</code></td>
<td>
<p>create curves in different colors for values of a specified variable:
all curves are in the same color (<code>color.var = "None"</code>); 
use different colors for diversity orders (<code>color.var = "Order.q"</code>); 
use different colors for assemblages/sites (<code>color.var = "Assemblage"</code>); 
use different colors for combinations of diversity order and assemblage (<code>color.var = "Both"</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code>ggplot2</code> object for sample-size-based rarefaction/extrapolation curve (<code>type = 1</code>), sample completeness curve (<code>type = 2</code>), and coverage-based rarefaction/extrapolation curve (<code>type = 3</code>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Plot three types of curves of taxonomic diversity with facet.var = "Assemblage"
# for abundance data with order q = 0, 1, 2
data(Brazil_rainforest_abun_data)
output_TD_abun &lt;- iNEXT3D(Brazil_rainforest_abun_data, diversity = 'TD', q = c(0, 1, 2), 
                          datatype = "abundance")
ggiNEXT3D(output_TD_abun, facet.var = "Assemblage")


# Plot two types (1 and 3) of curves of phylogenetic diversity 
# for abundance data with order q = 0, 1, 2
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_phylo_tree)
data &lt;- Brazil_rainforest_abun_data
tree &lt;- Brazil_rainforest_phylo_tree
output_PD_abun &lt;- iNEXT3D(data, diversity = 'PD', q = c(0, 1, 2), datatype = "abundance", 
                          nboot = 20, PDtree = tree)
ggiNEXT3D(output_PD_abun, type = c(1, 3))


# Plot three types of curves of functional diversity for abundance data
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_FD_abun &lt;- iNEXT3D(data, diversity = 'FD', datatype = "abundance", nboot = 0, 
                          FDdistM = distM, FDtype = 'AUC')
ggiNEXT3D(output_FD_abun)


# Plot three types of curves of taxonomic diversity for incidence data with order q = 0, 1, 2
data(Fish_incidence_data)
output_TD_inci &lt;- iNEXT3D(Fish_incidence_data, diversity = 'TD', q = c(0, 1, 2), 
                          datatype = "incidence_raw")
ggiNEXT3D(output_TD_inci)


# Plot three types of curves of phylogenetic diversity with facet.var = "Order.q"
#  and color.var = "Assemblage" for incidence data with order q = 0, 1, 2
data(Fish_incidence_data)
data(Fish_phylo_tree)
data &lt;- Fish_incidence_data
tree &lt;- Fish_phylo_tree
output_PD_inci &lt;- iNEXT3D(data, diversity = 'PD', q = c(0, 1, 2), datatype = "incidence_raw", 
                          nboot = 20, PDtree = tree)
ggiNEXT3D(output_PD_inci, facet.var = "Order.q", color.var = "Assemblage")


# Plot three types of curves of functional diversity for incidence data
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
output_FD_inci &lt;- iNEXT3D(data, diversity = 'FD', datatype = "incidence_raw", nboot = 20, 
                          FDdistM = distM, FDtype = 'AUC')
ggiNEXT3D(output_FD_inci)



</code></pre>

<hr>
<h2 id='ggObsAsy3D'>ggplot2 extension for plotting q-profile, time-profile, and tau-profile</h2><span id='topic+ggObsAsy3D'></span>

<h3>Description</h3>

<p><code>ggObsAsy3D</code> is a <code>ggplot2</code> extension for an <code>ObsAsy3D</code> object to plot 3D q-profile (which depicts the observed diversity and asymptotic diversity estimate with respect to order q) for q between 0 and 2 (in increments of 0.2).<br /><br /> 
It also plots time-profile (which depicts the observed and asymptotic estimate of PD or mean PD with respect to reference times when <code>diversity = "PD"</code> specified in the <code>ObsAsy3D</code> function), and tau-profile (which depicts the observed and asymptotic estimate of FD with respect to threshold level tau when <code>diversity = "FD"</code> and <code>FDtype = "tau_values"</code> specified in the <code>ObsAsy3D</code> function) based on the output of <code>ObsAsy3D</code>.<br /><br /> 
In the plot of profiles, only confidence intervals of the asymptotic diversity will be shown when both the observed and asymptotic diversity estimates are computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggObsAsy3D(output, profile = "q")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggObsAsy3D_+3A_output">output</code></td>
<td>
<p>the output of the function <code>ObsAsy3D</code>.<br /></p>
</td></tr>
<tr><td><code id="ggObsAsy3D_+3A_profile">profile</code></td>
<td>
<p>a selection of profile. User can choose <code>'q'</code>, <code>'time'</code>, and <code>'tau'</code>. Default is <code>'q'</code> profile.<br /></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a q-profile, time-profile, or tau-profile based on the observed diversity and the asymptotic diversity estimate.<br /><br />
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Plot q-profile of taxonomic diversity for abundance data
# with order q between 0 and 2 (in increments of 0.2 by default).
data(Brazil_rainforest_abun_data)
output_ObsAsy_TD_abun &lt;- ObsAsy3D(Brazil_rainforest_abun_data, diversity = 'TD', 
                                  datatype = "abundance")
ggObsAsy3D(output_ObsAsy_TD_abun)


# Plot time-profile of phylogenetic diversity for abundance data
# with order q = 0, 1, 2 under reference times from 0.01 to 400 (tree height).
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_phylo_tree)
data &lt;- Brazil_rainforest_abun_data
tree &lt;- Brazil_rainforest_phylo_tree
output_ObsAsy_PD_abun &lt;- ObsAsy3D(data, diversity = 'PD', q = c(0, 1, 2), 
                                  PDreftime = seq(0.01, 400, length.out = 20),
                                  datatype = "abundance", nboot = 20, PDtree = tree)
ggObsAsy3D(output_ObsAsy_PD_abun, profile = "time")


# Plot tau-profile of functional diversity for abundance data
# with order q = 0, 1, 2 under tau values from 0 to 1.
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_ObsAsy_FD_abun_tau &lt;- ObsAsy3D(data, diversity = 'FD', q = c(0, 1, 2), 
                                      datatype = "abundance", nboot = 10, FDdistM = distM, 
                                      FDtype = 'tau_values', FDtau = seq(0, 1, 0.05))
ggObsAsy3D(output_ObsAsy_FD_abun_tau, profile = "tau")


# Plot q-profile of functional diversity for abundance data
# with order q between 0 and 2 (in increments of 0.5).
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_ObsAsy_FD_abun &lt;- ObsAsy3D(data, diversity = 'FD', q = seq(0, 2, 0.5), 
                                  datatype = "abundance", nboot = 10, 
                                  FDdistM = distM, FDtype = 'AUC')
ggObsAsy3D(output_ObsAsy_FD_abun, profile = "q")


# Plot q-profile of taxonomic diversity for incidence data
# with order q between 0 and 2 (in increments of 0.2 by default)
data(Fish_incidence_data)
output_ObsAsy_TD_inci &lt;- ObsAsy3D(Fish_incidence_data, diversity = 'TD', 
                                  datatype = "incidence_raw")
ggObsAsy3D(output_ObsAsy_TD_inci)


# Plot q-profile of phylogenetic diversity for incidence data 
# with order q between 0 and 2 (in increments of 0.2 by default), 
# for the default reference time = 0.977 (the tree depth).
data(Fish_incidence_data)
data(Fish_phylo_tree)
data &lt;- Fish_incidence_data
tree &lt;- Fish_phylo_tree
output_ObsAsy_PD_inci &lt;- ObsAsy3D(data, diversity = 'PD', q = seq(0, 2, 0.2), 
                                  datatype = "incidence_raw", nboot = 20, PDtree = tree, 
                                  PDreftime = NULL)
ggObsAsy3D(output_ObsAsy_PD_inci, profile = "q")


# Plot q-profile of functional diversity for incidence data
# with order q between 0 and 2 (in increments of 0.2 by default)
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
output_ObsAsy_FD_inci &lt;- ObsAsy3D(data, diversity = 'FD', datatype = "incidence_raw", 
                                  nboot = 20, FDdistM = distM, FDtype = 'AUC')
ggObsAsy3D(output_ObsAsy_FD_inci, profile = "q")



</code></pre>

<hr>
<h2 id='iNEXT3D'>iNterpolation and EXTrapolation with three dimensions of biodiversity</h2><span id='topic+iNEXT3D'></span>

<h3>Description</h3>

<p><code>iNEXT3D</code> mainly computes standardized 3D estimates with a common sample size or sample coverage for orders q = 0, 1 and 2. It also computes relevant information/statistics.<br /><br /> 
For <code>diversity = "TD"</code>, relevant data information is summarized in the output <code>$TDInfo</code>. Diversity estimates for rarefied and extrapolated samples are provided in the output <code>$TDiNextEst</code>, which includes two data frames (<code>"$size_based"</code> and <code>"$coverage_based"</code>) based on two different standardizations; in the size-based standardization, all samples are standardized to a common target sample size, whereas the in the latter standardization, all samples are standardized to a common target level of sample coverage. The asymptotic diversity estimates for q = 0, 1 and 2 are provided in the list <code>$TDAsyEst</code>.<br /><br /> 
For <code>diversity = "PD"</code>, the corresponding three lists are <code>$PDInfo</code>, <code>$PDiNextEst</code> and <code>$PDAsyEst</code>.<br /> 
For <code>diversity = "FD"</code>, the corresponding three lists are <code>$FDInfo</code>, <code>$FDiNextEst</code> and <code>$FDAsyEst</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iNEXT3D(
  data,
  diversity = "TD",
  q = c(0, 1, 2),
  datatype = "abundance",
  size = NULL,
  endpoint = NULL,
  knots = 40,
  nboot = 50,
  conf = 0.95,
  nT = NULL,
  PDtree = NULL,
  PDreftime = NULL,
  PDtype = "meanPD",
  FDdistM,
  FDtype = "AUC",
  FDtau = NULL,
  FDcut_number = 50
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iNEXT3D_+3A_data">data</code></td>
<td>
<p>(a) For <code>datatype = "abundance"</code>, data can be input as a vector of species abundances (for a single assemblage), matrix/data.frame (species by assemblages), or a list of species abundance vectors. <br />
(b) For <code>datatype = "incidence_raw"</code>, data can be input as a list of matrices/data.frames (species by sampling units); data can also be input as a single matrix/data.frame by merging all sampling units across assemblages based on species identity; in this case, the number of sampling units (<code>nT</code>, see below) must be specified.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_diversity">diversity</code></td>
<td>
<p>selection of diversity type: <code>'TD'</code> = Taxonomic diversity, <code>'PD'</code> = Phylogenetic diversity, and <code>'FD'</code> = Functional diversity.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_q">q</code></td>
<td>
<p>a numerical vector specifying the diversity orders. Default is <code>c(0, 1, 2)</code>.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_datatype">datatype</code></td>
<td>
<p>data type of input data: individual-based abundance data (<code>datatype = "abundance"</code>) or species by sampling-units incidence/occurrence matrix (<code>datatype = "incidence_raw"</code>) with all entries being 0 (non-detection) or 1 (detection).</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_size">size</code></td>
<td>
<p>an integer vector of sample sizes (number of individuals or sampling units) for which diversity estimates will be computed. 
If <code>NULL</code>, then diversity estimates will be computed for those sample sizes determined by the specified/default <code>endpoint</code> and <code>knots</code>.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_endpoint">endpoint</code></td>
<td>
<p>an integer specifying the sample size that is the <code>endpoint</code> for rarefaction/extrapolation. 
If <code>NULL</code>, then <code>endpoint</code> <code>=</code> double reference sample size.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_knots">knots</code></td>
<td>
<p>an integer specifying the number of equally-spaced <code>knots</code> (say K, default is 40) between size 1 and the <code>endpoint</code>;
each knot represents a particular sample size for which diversity estimate will be calculated.  
If the <code>endpoint</code> is smaller than the reference sample size, then <code>iNEXT3D()</code> computes only the rarefaction esimates for approximately K evenly spaced <code>knots</code>. 
If the <code>endpoint</code> is larger than the reference sample size, then <code>iNEXT3D()</code> computes rarefaction estimates for approximately K/2 evenly spaced <code>knots</code> between sample size 1 and the reference sample size, and computes extrapolation estimates for approximately K/2 evenly spaced <code>knots</code> between the reference sample size and the <code>endpoint</code>.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_nboot">nboot</code></td>
<td>
<p>a positive integer specifying the number of bootstrap replications when assessing sampling uncertainty and constructing confidence intervals. Enter 0 to skip the bootstrap procedures. Default is 50.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_conf">conf</code></td>
<td>
<p>a positive number &lt; 1 specifying the level of confidence interval. Default is 0.95.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_nt">nT</code></td>
<td>
<p>(required only when <code>datatype = "incidence_raw"</code> and input data in a single matrix/data.frame) a vector of positive integers specifying the number of sampling units in each assemblage. If assemblage names are not specified (i.e., <code>names(nT) = NULL</code>), then assemblages are automatically named as &quot;assemblage1&quot;, &quot;assemblage2&quot;,..., etc.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_pdtree">PDtree</code></td>
<td>
<p>(required argument for <code>diversity = "PD"</code>), a phylogenetic tree in Newick format for all observed species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_pdreftime">PDreftime</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), a vector of numerical values specifying reference times for PD. Default is <code>NULL</code> (i.e., the age of the root of <code>PDtree</code>).</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_pdtype">PDtype</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), select PD type: <code>PDtype = "PD"</code> (effective total branch length) or <code>PDtype = "meanPD"</code> (effective number of equally divergent lineages). Default is <code>"meanPD"</code>, where <code>meanPD = PD/tree depth</code>.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_fddistm">FDdistM</code></td>
<td>
<p>(required argument for <code>diversity = "FD"</code>), a species pairwise distance matrix for all species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_fdtype">FDtype</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code>), select FD type: <code>FDtype = "tau_values"</code> for FD under specified threshold values, or <code>FDtype = "AUC"</code> (area under the curve of tau-profile) for an overall FD which integrates all threshold values between zero and one. Default is <code>"AUC"</code>.</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_fdtau">FDtau</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "tau_values"</code>), a numerical vector between 0 and 1 specifying tau values (threshold levels). If <code>NULL</code> (default), then threshold is set to be the mean distance between any two individuals randomly selected from the pooled assemblage (i.e., quadratic entropy).</p>
</td></tr>
<tr><td><code id="iNEXT3D_+3A_fdcut_number">FDcut_number</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "AUC"</code>), a numeric number to cut [0, 1] interval into equal-spaced sub-intervals to obtain the AUC value by integrating the tau-profile. Equivalently, the number of tau values that will be considered to compute the integrated AUC value. Default is <code>FDcut_number = 50</code>. A larger value can be set to obtain more accurate AUC value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of three objects: <br /><br />
(1) <code>$TDInfo</code> (<code>$PDInfo</code>, or <code>$FDInfo</code>) for summarizing data information for q = 0, 1 and 2. Refer to the output of <code>DataInfo3D</code> for details. <br /><br />
(2) <code>$TDiNextEst</code> (<code>$PDiNextEst</code>, or <code>$FDiNextEst</code>) for showing diversity estimates for rarefied and extrapolated samples along with related statistics. There are two data frames: <code>"$size_based"</code> and <code>"$coverage_based"</code>. <br /><br />
In <code>"$size_based"</code>, the output includes:
</p>
<table>
<tr><td><code>Assemblage</code></td>
<td>
<p>the name of assemblage.</p>
</td></tr> 
<tr><td><code>Order.q</code></td>
<td>
<p>the diversity order of q.</p>
</td></tr>
<tr><td><code>m</code>, <code>mT</code></td>
<td>
<p>the target sample size (or number of sampling units for incidence data).</p>
</td></tr>
<tr><td><code>Method</code></td>
<td>
<p>Rarefaction, Observed, or Extrapolation, depending on whether the target sample size is less than, equal to, or greater than the size of the reference sample.</p>
</td></tr>
<tr><td><code>qTD</code>, <code>qPD</code>, <code>qFD</code></td>
<td>
<p>the estimated diversity estimate.</p>
</td></tr>
<tr><td><code>qTD.LCL</code>, <code>qPD.LCL</code>, <code>qFD.LCL and qTD.UCL</code>, <code>qPD.UCL</code>, <code>qFD.UCL</code></td>
<td>
<p>the bootstrap lower and upper confidence limits for the diversity of order q at the specified level (with a default value of 0.95).</p>
</td></tr>
<tr><td><code>SC</code></td>
<td>
<p>the standardized coverage value.</p>
</td></tr>
<tr><td><code>SC.LCL</code>, <code>SC.UCL</code></td>
<td>
<p>the bootstrap lower and upper confidence limits for coverage at the specified level (with a default value of 0.95).</p>
</td></tr>
<tr><td><code>Reftime</code></td>
<td>
<p>the reference times for PD.</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p><code>"PD"</code> (effective total branch length) or <code>"meanPD"</code> (effective number of equally divergent lineages) for PD.</p>
</td></tr>
<tr><td><code>Tau</code></td>
<td>
<p>the threshold of functional distinctiveness between any two species for FD (under <code>FDtype = tau_values</code>).</p>
</td></tr>
</table>
<p>Similar output is obtained for <code>"$coverage_based"</code>. <br /><br />
(3) <code>$TDAsyEst</code> (<code>$PDAsyEst</code>, or <code>$FDAsyEst</code>) for showing asymptotic diversity estimates along with related statistics: 
</p>
<table>
<tr><td><code>Assemblage</code></td>
<td>
<p>the name of assemblage.</p>
</td></tr> 
<tr><td><code>qTD</code>, <code>qPD</code>, <code>qFD</code></td>
<td>
<p>the diversity order of q.</p>
</td></tr>
<tr><td><code>TD_obs</code>, <code>PD_obs</code>, <code>FD_obs</code></td>
<td>
<p>the observed diversity.</p>
</td></tr>
<tr><td><code>TD_asy</code>, <code>PD_asy</code>, <code>FD_asy</code></td>
<td>
<p>the asymptotic diversity estimate.</p>
</td></tr>
<tr><td><code>s.e.</code></td>
<td>
<p>standard error of asymptotic diversity.</p>
</td></tr>
<tr><td><code>qTD.LCL</code>, <code>qPD.LCL</code>, <code>qFD.LCL and qTD.UCL</code>, <code>qPD.UCL</code>, <code>qFD.UCL</code></td>
<td>
<p>the bootstrap lower and upper confidence limits for asymptotic diversity at the specified level (with a default value of 0.95).</p>
</td></tr>
<tr><td><code>Reftime</code></td>
<td>
<p>the reference times for PD.</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p><code>"PD"</code> (effective total branch length) or <code>"meanPD"</code> (effective number of equally divergent lineages) for PD.</p>
</td></tr>
<tr><td><code>Tau</code></td>
<td>
<p>the threshold of functional distinctiveness between any two species for FD (under <code>FDtype = tau_values</code>).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# Compute standardized estimates of taxonomic diversity for abundance data with order q = 0, 1, 2
data(Brazil_rainforest_abun_data)
output_TD_abun &lt;- iNEXT3D(Brazil_rainforest_abun_data, diversity = 'TD', q = c(0, 1, 2), 
                          datatype = "abundance")
output_TD_abun


# Compute standardized estimates of phylogenetic diversity for abundance data with order q = 0, 1, 2
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_phylo_tree)
data &lt;- Brazil_rainforest_abun_data
tree &lt;- Brazil_rainforest_phylo_tree
output_PD_abun &lt;- iNEXT3D(data, diversity = 'PD', q = c(0, 1, 2), datatype = "abundance", 
                          nboot = 20, PDtree = tree)
output_PD_abun


# Compute standardized estimates of functional diversity for abundance data
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_FD_abun &lt;- iNEXT3D(data, diversity = 'FD', datatype = "abundance", nboot = 0, 
                          FDdistM = distM, FDtype = 'AUC')
output_FD_abun


# Compute standardized estimates of taxonomic diversity for incidence data with order q = 0, 1, 2
data(Fish_incidence_data)
output_TD_inci &lt;- iNEXT3D(Fish_incidence_data, diversity = 'TD', q = c(0, 1, 2), 
                          datatype = "incidence_raw")
output_TD_inci


# Compute standardized estimates of phylogenetic diversity for incidence data with order q = 0, 1, 2
data(Fish_incidence_data)
data(Fish_phylo_tree)
data &lt;- Fish_incidence_data
tree &lt;- Fish_phylo_tree
output_PD_inci &lt;- iNEXT3D(data, diversity = 'PD', q = c(0, 1, 2), 
                          datatype = "incidence_raw", nboot = 20, PDtree = tree)
output_PD_inci


# Compute estimates of functional diversity for incidence data
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
output_FD_inci &lt;- iNEXT3D(data, diversity = 'FD', datatype = "incidence_raw", nboot = 20, 
                          FDdistM = distM, FDtype = 'AUC')
output_FD_inci



</code></pre>

<hr>
<h2 id='ObsAsy3D'>Asymptotic diversity and observed diversity of order q</h2><span id='topic+ObsAsy3D'></span>

<h3>Description</h3>

<p><code>ObsAsy3D</code> computes observed and asymptotic diversity of order q between 0 and 2 (in increments of 0.2) for 3D diversity; these 3D values with different order q can be used to depict a q-profile in the <code>ggObsAsy3D</code> function.<br /><br /> 
It also computes observed and asymptotic PD for various reference times by specifying the argument <code>PDreftime</code>; these PD values with different reference times can be used to depict a time-profile in the <code>ggObsAsy3D</code> function.<br /><br />
It also computes observed and asymptotic FD for various threshold tau levels by specifying the argument <code>FDtau</code>; these FD values with different threshold levels can be used to depict a tau-profile in the <code>ggObsAsy3D</code> function.<br /><br />
For each dimension, by default, both the observed and asymptotic diversity estimates will be computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ObsAsy3D(
  data,
  diversity = "TD",
  q = seq(0, 2, 0.2),
  datatype = "abundance",
  nboot = 50,
  conf = 0.95,
  nT = NULL,
  method = c("Asymptotic", "Observed"),
  PDtree,
  PDreftime = NULL,
  PDtype = "meanPD",
  FDdistM,
  FDtype = "AUC",
  FDtau = NULL,
  FDcut_number = 50
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ObsAsy3D_+3A_data">data</code></td>
<td>
<p>(a) For <code>datatype = "abundance"</code>, data can be input as a vector of species abundances (for a single assemblage), matrix/data.frame (species by assemblages), or a list of species abundance vectors. <br />
(b) For <code>datatype = "incidence_raw"</code>, data can be input as a list of matrices/data.frames (species by sampling units); data can also be input as a single matrix/data.frame by merging all sampling units across assemblages based on species identity; in this case, the number of sampling units (<code>nT</code>, see below) must be specified.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_diversity">diversity</code></td>
<td>
<p>selection of diversity type: <code>'TD'</code> = Taxonomic diversity, <code>'PD'</code> = Phylogenetic diversity, and <code>'FD'</code> = Functional diversity.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_q">q</code></td>
<td>
<p>a numerical vector specifying the diversity orders. Default is <code>seq(0, 2, by = 0.2)</code>.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_datatype">datatype</code></td>
<td>
<p>data type of input data: individual-based abundance data (<code>datatype = "abundance"</code>) or species by sampling-units incidence/occurrence matrix (<code>datatype = "incidence_raw"</code>) with all entries being 0 (non-detection) or 1 (detection).</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_nboot">nboot</code></td>
<td>
<p>a positive integer specifying the number of bootstrap replications when assessing sampling uncertainty and constructing confidence intervals. Enter 0 to skip the bootstrap procedures. Default is 50.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_conf">conf</code></td>
<td>
<p>a positive number &lt; 1 specifying the level of confidence interval. Default is 0.95.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_nt">nT</code></td>
<td>
<p>(required only when <code>datatype = "incidence_raw"</code> and input data in a single matrix/data.frame) a vector of positive integers specifying the number of sampling units in each assemblage. If assemblage names are not specified (i.e., <code>names(nT) = NULL</code>), then assemblages are automatically named as &quot;assemblage1&quot;, &quot;assemblage2&quot;,..., etc.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_method">method</code></td>
<td>
<p>Select <code>'Asymptotic'</code> or <code>'Observed'</code>.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_pdtree">PDtree</code></td>
<td>
<p>(required argument for <code>diversity = "PD"</code>), a phylogenetic tree in Newick format for all observed species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_pdreftime">PDreftime</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), a vector of numerical values specifying reference times for PD. Default is <code>NULL</code> (i.e., the age of the root of <code>PDtree</code>).</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_pdtype">PDtype</code></td>
<td>
<p>(argument only for <code>diversity = "PD"</code>), select PD type: <code>PDtype = "PD"</code> (effective total branch length) or <code>PDtype = "meanPD"</code> (effective number of equally divergent lineages). Default is <code>"meanPD"</code>, where <code>meanPD = PD/tree depth</code>.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_fddistm">FDdistM</code></td>
<td>
<p>(required argument for <code>diversity = "FD"</code>), a species pairwise distance matrix for all species in the pooled assemblage.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_fdtype">FDtype</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code>), select FD type: <code>FDtype = "tau_values"</code> for FD under specified threshold values, or <code>FDtype = "AUC"</code> (area under the curve of tau-profile) for an overall FD which integrates all threshold values between zero and one. Default is <code>"AUC"</code>.</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_fdtau">FDtau</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "tau_values"</code>), a numerical vector between 0 and 1 specifying tau values (threshold levels). If <code>NULL</code> (default), then threshold is set to be the mean distance between any two individuals randomly selected from the pooled assemblage (i.e., quadratic entropy).</p>
</td></tr>
<tr><td><code id="ObsAsy3D_+3A_fdcut_number">FDcut_number</code></td>
<td>
<p>(argument only for <code>diversity = "FD"</code> and <code>FDtype = "AUC"</code>), a numeric number to cut [0, 1] interval into equal-spaced sub-intervals to obtain the AUC value by integrating the tau-profile. Equivalently, the number of tau values that will be considered to compute the integrated AUC value. Default is <code>FDcut_number = 50</code>. A larger value can be set to obtain more accurate AUC value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame including the following information/statistics: 
</p>
<table>
<tr><td><code>Assemblage</code></td>
<td>
<p>the name of assemblage.</p>
</td></tr>
<tr><td><code>Order.q</code></td>
<td>
<p>the diversity order of q.</p>
</td></tr>
<tr><td><code>qTD</code>, <code>qPD</code>, <code>qFD</code></td>
<td>
<p>the estimated asymptotic diversity or observed diversity of order q.</p>
</td></tr> 
<tr><td><code>s.e.</code></td>
<td>
<p>standard error of diversity.</p>
</td></tr>
<tr><td><code>qTD.LCL</code>, <code>qPD.LCL</code>, <code>qFD.LCL and qTD.UCL</code>, <code>qPD.UCL</code>, <code>qFD.UCL</code></td>
<td>
<p>the bootstrap lower and upper confidence limits for the diversity of order q at the specified level (with a default value of 0.95).</p>
</td></tr>
<tr><td><code>Method</code></td>
<td>
<p><code>"Asymptotic"</code> means asymptotic diversity and <code>"Observed"</code> means observed diversity.</p>
</td></tr>
<tr><td><code>Reftime</code></td>
<td>
<p>the reference times for PD.</p>
</td></tr>
<tr><td><code>Type</code></td>
<td>
<p><code>"PD"</code> (effective total branch length) or <code>"meanPD"</code> (effective number of equally divergent lineages) for PD.</p>
</td></tr>
<tr><td><code>Tau</code></td>
<td>
<p>the threshold of functional distinctiveness between any two species for FD (under <code>FDtype = tau_values</code>).</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# Compute the observed and asymptotic taxonomic diversity for abundance data
# with order q between 0 and 2 (in increments of 0.2 by default)
data(Brazil_rainforest_abun_data)
output_ObsAsy_TD_abun &lt;- ObsAsy3D(Brazil_rainforest_abun_data, diversity = 'TD', 
                                  datatype = "abundance")
output_ObsAsy_TD_abun


# Compute the observed and asymptotic phylogenetic diversity for abundance data
# with order q = 0, 1, 2 under reference times from 0.01 to 400 (tree height).
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_phylo_tree)
data &lt;- Brazil_rainforest_abun_data
tree &lt;- Brazil_rainforest_phylo_tree
output_ObsAsy_PD_abun &lt;- ObsAsy3D(data, diversity = 'PD', q = c(0, 1, 2), 
                                  PDreftime = seq(0.01, 400, length.out = 20),
                                  datatype = "abundance", nboot = 20, PDtree = tree)
output_ObsAsy_PD_abun


# Compute the observed and asymptotic functional diversity for abundance data
# with order q = 0, 1, 2 under tau values from 0 to 1.
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_ObsAsy_FD_abun_tau &lt;- ObsAsy3D(data, diversity = 'FD', q = c(0, 1, 2), 
                                      datatype = "abundance", nboot = 10, FDdistM = distM, 
                                      FDtype = 'tau_values', FDtau = seq(0, 1, 0.05))
output_ObsAsy_FD_abun_tau


# Compute the observed and asymptotic functional diversity for abundance data
# with order q between 0 and 2 (in increments of 0.5).
data(Brazil_rainforest_abun_data)
data(Brazil_rainforest_distance_matrix)
data &lt;- Brazil_rainforest_abun_data
distM &lt;- Brazil_rainforest_distance_matrix
output_ObsAsy_FD_abun &lt;- ObsAsy3D(data, diversity = 'FD', q = seq(0, 2, 0.5), 
                                  datatype = "abundance", nboot = 10, 
                                  FDdistM = distM, FDtype = 'AUC')
output_ObsAsy_FD_abun


# Compute the observed and asymptotic taxonomic diversity for incidence data
# with order q between 0 and 2 (in increments of 0.2 by default).
data(Fish_incidence_data)
output_ObsAsy_TD_inci &lt;- ObsAsy3D(Fish_incidence_data, diversity = 'TD', 
                                  datatype = "incidence_raw")
output_ObsAsy_TD_inci


# Compute the observed and asymptotic phylogenetic diversity for incidence data
# with order q between 0 and 2 (in increments of 0.2 by default), 
# for the default reference time = 0.977 (the tree depth).
data(Fish_incidence_data)
data(Fish_phylo_tree)
data &lt;- Fish_incidence_data
tree &lt;- Fish_phylo_tree
output_ObsAsy_PD_inci &lt;- ObsAsy3D(data, diversity = 'PD', q = seq(0, 2, 0.2), 
                                  datatype = "incidence_raw", nboot = 20, PDtree = tree, 
                                  PDreftime = NULL)
output_ObsAsy_PD_inci


# Compute the observed and asymptotic functional diversity for incidence data
# with order q between 0 and 2 (in increments of 0.2 by default).
data(Fish_incidence_data)
data(Fish_distance_matrix)
data &lt;- Fish_incidence_data
distM &lt;- Fish_distance_matrix
output_ObsAsy_FD_inci &lt;- ObsAsy3D(data, diversity = 'FD', datatype = "incidence_raw", 
                                  nboot = 20, FDdistM = distM, FDtype = 'AUC')
output_ObsAsy_FD_inci



</code></pre>

<hr>
<h2 id='print.iNEXT3D'>Printing iNEXT3D object</h2><span id='topic+print.iNEXT3D'></span>

<h3>Description</h3>

<p><code>print.iNEXT3D</code>: Print method for objects inheriting from class &quot;iNEXT3D&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'iNEXT3D'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.iNEXT3D_+3A_x">x</code></td>
<td>
<p>an <code>iNEXT3D</code> object computed by <code>iNEXT3D</code>.</p>
</td></tr>
<tr><td><code id="print.iNEXT3D_+3A_...">...</code></td>
<td>
<p>additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of three objects (see <code>iNEXT3D</code> for more details) with simplified outputs and notes.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
