<!DOCTYPE html><html><head><title>Help for package InfiniumPurify</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {InfiniumPurify}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#abbr'><p>abbr</p></a></li>
<li><a href='#beta.emp'><p>beta.emp</p></a></li>
<li><a href='#CancerTypeAbbr'>
<p>Print abbreviations of cancer types with known iDMCs.</p></a></li>
<li><a href='#getPurity'>
<p>Estimate the tumor purity for 450K methylation data</p></a></li>
<li><a href='#iDMC'><p>iDMC</p></a></li>
<li><a href='#InfiniumClust'>
<p>Tumor sample clustering from Infinium 450k array data</p></a></li>
<li><a href='#InfiniumDMC'>
<p>Differentially Methylation Calling accounting for tumor purity</p></a></li>
<li><a href='#InfiniumPurify'>
<p>Purify tumor methylomes caused by normal cell contamination.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimate and Account for Tumor Purity in Cancer Methylation Data
Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-1-8</td>
</tr>
<tr>
<td>Author:</td>
<td>Yufang Qin</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yufang Qin &lt;yfqin@shou.edu.cn&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>matrixStats</td>
</tr>
<tr>
<td>Description:</td>
<td>The proportion of cancer cells in solid tumor sample, known as the tumor purity, has adverse impact on a variety of data analyses if not properly accounted for. We develop 'InfiniumPurify', which is a comprehensive R package for estimating and accounting for tumor purity based on DNA methylation Infinium 450k array data. 'InfiniumPurify' provides functionalities for tumor purity estimation. In addition, it can perform differential methylation detection and tumor sample clustering with the consideration of tumor purities. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-01-11 13:10:18 UTC; zhengxq</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-01-14 12:12:25</td>
</tr>
</table>
<hr>
<h2 id='abbr'>abbr</h2><span id='topic+abbr'></span>

<h3>Description</h3>

<p>This data set lists abbreviations for all TCGA cancer types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abbr
</code></pre>


<h3>Format</h3>

<p>A dataframe containing names and abbreviations for all TCGA cancer types.</p>


<h3>Source</h3>

<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, accepted. 
</p>

<hr>
<h2 id='beta.emp'>beta.emp</h2><span id='topic+beta.emp'></span>

<h3>Description</h3>

<p>An example data set for InfiniumClust and InfiniumPurify.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>beta.emp
</code></pre>


<h3>Format</h3>

<p>A dataframe containing methylaton beta values for 62 tumor and normal samples.</p>


<h3>Source</h3>

<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, accepted. 
</p>

<hr>
<h2 id='CancerTypeAbbr'>
Print abbreviations of cancer types with known iDMCs. 
</h2><span id='topic+CancerTypeAbbr'></span>

<h3>Description</h3>

<p>Print tumor types and their abbreviations with known informative DMCs.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CancerTypeAbbr()
</code></pre>


<h3>Arguments</h3>

<p>None.
</p>


<h3>Author(s)</h3>

<p>Xiaoqi Zheng <a href="mailto:xqzheng@shnu.edu.cn">xqzheng@shnu.edu.cn</a>.
</p>


<h3>References</h3>

<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, in revision. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(abbr)
CancerTypeAbbr()
</code></pre>

<hr>
<h2 id='getPurity'>
Estimate the tumor purity for 450K methylation data 
</h2><span id='topic+getPurity'></span>

<h3>Description</h3>

<p>Estimate the percentage of tumor cells in cancer samples which are mixtures of cancer and normal cells.added a sentence
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPurity(tumor.data,normal.data = NULL,tumor.type = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPurity_+3A_tumor.data">tumor.data</code></td>
<td>
<p>numeric vector/matrix of beta values for tumor samlpes. The names/rownames of tumor.data
should be probe names of Infinium 450k array, and colnames should be names of tumor samples.</p>
</td></tr>
<tr><td><code id="getPurity_+3A_normal.data">normal.data</code></td>
<td>
<p>numeric matrix of beta values for normal samlpes. The rownames of normal.data
should be probe names of Infinium 450k array, and colnames should be names of normal samples.</p>
</td></tr>
<tr><td><code id="getPurity_+3A_tumor.type">tumor.type</code></td>
<td>
<p>cancer type (in abbreviation) of tumor and normal samlpes. Options are &quot;LUAD&quot;, &quot;BRCA&quot; and so
on. See <code>CancerTypeAbbr</code> for detail.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Arguments normal.data and tumor.type could be null. If either the number of tumor samples or number of normal smaples is less than 20, the tumor.type argument should be specified according to <code>CancerTypeAbbr</code>. If the numbers of tumor and normal samples are both more than 20, tumor.type could be null. In such case, <a href="#topic+getPurity">getPurity</a> first identify 1000 iDMCs by Wilcox rank-sum test, then tumor purity for each sample is estimated as the density mode of adjusted methylation levels of iDMCs. 
</p>


<h3>Value</h3>

<p>A vector of tumor purities for each tumor sample. 
</p>


<h3>Author(s)</h3>

<p>Xiaoqi Zheng <a href="mailto:xqzheng@shnu.edu.cn">xqzheng@shnu.edu.cn</a>.
</p>


<h3>References</h3>

<p>N. Zhang, H.J. Wu, W. Zhang, J. Wang, H. Wu and X. Zheng (2015) Predicting tumor purity from methylation
microarray data. <em>Bioinformatics</em> <b>31(21)</b>, 3401-3405. 
</p>
<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, accepted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(beta.emp)

normal.data &lt;- beta.emp[,1:21]
tumor.data &lt;- beta.emp[,22:61]

## call purity for single tumor sample 
purity &lt;- getPurity(tumor.data = tumor.data[,1],normal.data = NULL,tumor.type= "LUAD")

## call purity for less than 20 tumor samples
purity &lt;- getPurity(tumor.data = tumor.data[,1:10],normal.data = NULL,tumor.type= "LUAD")

## call purity for more than 20 tumor samples with matched normal samples
purity &lt;- getPurity(tumor.data = tumor.data[,1:40],normal.data = normal.data)
</code></pre>

<hr>
<h2 id='iDMC'>iDMC</h2><span id='topic+iDMC'></span>

<h3>Description</h3>

<p>This data set lists pre-selected iDMCs for all TCGA cancer types.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iDMC
</code></pre>


<h3>Format</h3>

<p>A list containing informative Differential methylation CpG sites (iDMC) and their average methylation levels in tumor and normal samples.</p>


<h3>Source</h3>

<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, accepted. 
</p>

<hr>
<h2 id='InfiniumClust'>
Tumor sample clustering from Infinium 450k array data  
</h2><span id='topic+InfiniumClust'></span>

<h3>Description</h3>

<p>Clustering of tumor samples into subtypes accounting for tumor purity. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InfiniumClust(tumor.data, purity, K, maxiter = 100, tol = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InfiniumClust_+3A_tumor.data">tumor.data</code></td>
<td>
<p>numeric matrix of beta values for tumor samlpes. The rownames of tumor.data
should be probe names of Infinium 450k array, and colnames should be names of tumor samples.</p>
</td></tr>
<tr><td><code id="InfiniumClust_+3A_purity">purity</code></td>
<td>
<p>purities for tumor samples. Could be estimated by <code>getPurity</code>, or user specified 
purities from other tools.</p>
</td></tr>
<tr><td><code id="InfiniumClust_+3A_k">K</code></td>
<td>
<p>the number of clusters.</p>
</td></tr>
<tr><td><code id="InfiniumClust_+3A_maxiter">maxiter</code></td>
<td>
<p>the maximum number of iterations allowed. Default is 100.</p>
</td></tr>
<tr><td><code id="InfiniumClust_+3A_tol">tol</code></td>
<td>
<p>tolerance for convergence of EM iterations. Default is 0.001.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An EM based statistical method for subtype classification based on DNA methylation data, while adjusting for tumor purity. 
</p>


<h3>Value</h3>

<p>InfiniumClust returns a list consisting oflikelihood <code>tol.ll</code> and membership matrix <code>Z</code>.
</p>
<table>
<tr><td><code>tol.ll</code></td>
<td>
<p>total log-likelihood of converged EM algorithm.</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>the membership matrix, where row corresponds to tumor samples and column corresponds to K clusters.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xiaoqi Zheng <a href="mailto:xqzheng@shnu.edu.cn">xqzheng@shnu.edu.cn</a> and Hao Wu <a href="mailto:hao.wu@emory.edu">hao.wu@emory.edu</a>
</p>


<h3>References</h3>

<p>W. Zhang, H. Feng, H. Wu and X. Zheng (2016). Tumor purity improves cancer subtype classification from DNA methylation data. Submitted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load example data
data(beta.emp)
normal.data &lt;- beta.emp[,1:21]
tumor.data &lt;- beta.emp[,22:31]

## estimate tumor purity
purity &lt;- getPurity(tumor.data = tumor.data,tumor.type= "LUAD")

## cluster tumor samples accounting for tumor purity
out &lt;- InfiniumClust(tumor.data,purity,K=3, maxiter=5, tol=0.001)
</code></pre>

<hr>
<h2 id='InfiniumDMC'>
Differentially Methylation Calling accounting for tumor purity
</h2><span id='topic+InfiniumDMC'></span>

<h3>Description</h3>

<p>Infer differentially methylated CpG sites with the consideration of tumor purities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InfiniumDMC(tumor.data,normal.data,purity,threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InfiniumDMC_+3A_tumor.data">tumor.data</code></td>
<td>
<p>numeric matrix of beta values for tumor samlpes. The rownames of tumor.data
should be probe names of Infinium 450k array, and colnames should be names of tumor samples.</p>
</td></tr>
<tr><td><code id="InfiniumDMC_+3A_normal.data">normal.data</code></td>
<td>
<p>numeric matrix of beta values for normal samlpes. The rownames of normal.data
should be probe names of Infinium 450k array, and colnames should be names of normal samples.</p>
</td></tr>
<tr><td><code id="InfiniumDMC_+3A_purity">purity</code></td>
<td>
<p>purities for tumor samples. Could be estimated by <a href="#topic+getPurity">getPurity</a>, or user specified 
purities from other tools.</p>
</td></tr>
<tr><td><code id="InfiniumDMC_+3A_threshold">threshold</code></td>
<td>
<p>probability threshold in control-free DM calling. Default is 0.1.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If normal.data is provided, the function tests each CpG site for differential methylation between tumor and normal samples with the consideration of tumor purities by a generalized linear regression. If normal.data is not provided, the function computes posterior probability to rank CpG sites. 
</p>


<h3>Value</h3>

<p>A data frame of statistics, p-values and q-values for all CpG sites.
</p>


<h3>Author(s)</h3>

<p>Xiaoqi Zheng <a href="mailto:xqzheng@shnu.edu.cn">xqzheng@shnu.edu.cn</a>.
</p>


<h3>References</h3>

<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, in revision. 
</p>


<h3>See Also</h3>

<p><code><a href="minfi.html#topic+dmpFinder">dmpFinder</a></code> in the <span class="pkg">minfi</span> package. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(beta.emp)

normal.data &lt;- beta.emp[,1:21]
tumor.data &lt;- beta.emp[,22:61]

## estimate tumor purity
purity &lt;- getPurity(tumor.data = tumor.data,normal.data = normal.data)

## DM calling with normal controls
DMC = InfiniumDMC(tumor.data = tumor.data,normal.data = normal.data,purity = purity)

## DM calling without normal control
DMC_ctlFree = InfiniumDMC(tumor.data = tumor.data,purity = purity)

</code></pre>

<hr>
<h2 id='InfiniumPurify'>
Purify tumor methylomes caused by normal cell contamination.
</h2><span id='topic+InfiniumPurify'></span>

<h3>Description</h3>

<p>Deconvolute purified tumor methylomes accounting for tumor purity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InfiniumPurify(tumor.data,normal.data,purity)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InfiniumPurify_+3A_tumor.data">tumor.data</code></td>
<td>
<p>numeric matrix of beta values for tumor samlpes. The rownames of tumor.data
should be probe names of Infinium 450k array, and colnames should be names of tumor samples.</p>
</td></tr>
<tr><td><code id="InfiniumPurify_+3A_normal.data">normal.data</code></td>
<td>
<p>numeric matrix of beta values for normal samlpes. The rownames of normal.data
should be probe names of Infinium 450k array, and colnames should be names of normal samples.</p>
</td></tr>
<tr><td><code id="InfiniumPurify_+3A_purity">purity</code></td>
<td>
<p>purities for tumor samples. Could be estimated by <a href="#topic+getPurity">getPurity</a>, or user specified 
purities from other tools.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function deconvolutes purified tumor methylomes by a linear regression model. 
</p>


<h3>Value</h3>

<p>A matrix of purified beta values for all CpG sites (row) and tumor samples (column).
</p>


<h3>Author(s)</h3>

<p>Xiaoqi Zheng <a href="mailto:xqzheng@shnu.edu.cn">xqzheng@shnu.edu.cn</a>.
</p>


<h3>References</h3>

<p>X. Zheng, N. Zhang, H.J. Wu and H. Wu, Estimating and accounting for tumor purity in the analysis of DNA methylation data from cancer studies. <em>Genome biology</em>, accepted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(beta.emp)

normal.data &lt;- beta.emp[,1:21]
tumor.data &lt;- beta.emp[,22:61]

## estimate tumor purity
purity &lt;- getPurity(tumor.data = tumor.data,normal.data = NULL,tumor.type= "LUAD")

## correct tumor methylome by tumor purity
tumor.purified = InfiniumPurify(tumor.data = tumor.data[1:100,],
                                normal.data = normal.data[1:100,],
                                purity = purity)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
