<!DOCTYPE html><html><head><title>Help for package InteRD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {InteRD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#evaluate'><p>Evaluation for estimated cell type proportions</p></a></li>
<li><a href='#generateBulk'><p>Pseudo bulk data generation function</p></a></li>
<li><a href='#InteRD.predict.prop'><p>Extract the estimated proportions from InteRD</p></a></li>
<li><a href='#InteRD1'><p>The InteRD1 estimate from reference ensemble</p></a></li>
<li><a href='#InteRD2'><p>The InteRD2 estimate</p></a></li>
<li><a href='#pop.ct.prop.scRNA'><p>Calculate the population-level cell type proportions from a single-cell data.</p></a></li>
<li><a href='#Ref_free'><p>A reference-free deconvolution estimate</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The Integrated and Robust Deconvolution</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>We developed the Integrated and Robust Deconvolution algorithm to infer cell-type proportions from target bulk RNA-seq data. This package is able to effectively integrate deconvolution results from multiple scRNA-seq datasets and calibrates estimates from reference-based deconvolution by taking into account extra biological information as priors. Moreover, the proposed algorithm is robust to inaccurate external information imposed in the deconvolution system.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/chencxxy28/InteRD">https://github.com/chencxxy28/InteRD</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/chencxxy28/InteRD/issues">https://github.com/chencxxy28/InteRD/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.0), limSolve, cowplot, ggplot2, pheatmap,
stats,DescTools, mgcv, reshape2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), Biobase</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-11 17:55:05 UTC; chixiang.chen</td>
</tr>
<tr>
<td>Author:</td>
<td>Chixiang Chen [cre, aut],
  Yuk Yee Leung [aut],
  Matei Lonita [aut],
  Li-San Wang [aut],
  Mingyao Li [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chixiang Chen &lt;chencxxy@hotmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-12 07:20:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='evaluate'>Evaluation for estimated cell type proportions</h2><span id='topic+evaluate'></span>

<h3>Description</h3>

<p>Several evaluation metrics are provided, such as mean absolute deviance ('MAD'), Kendall-tau correlation coefficient ('Ken'), Pearson correlation coefficient ('Cor'), given true cell type proportions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evaluate(est.prop,true.prop)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="evaluate_+3A_est.prop">est.prop</code></td>
<td>
<p>The estimated cell type proportions.</p>
</td></tr>
<tr><td><code id="evaluate_+3A_true.prop">true.prop</code></td>
<td>
<p>The True cell type proportions</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cell-type level evaluations based on MAD, Ken, and Pearson ('cell.type.eva'), and overall evaluations based on averaged MAD, Ken, and Pearson ('all.eva').
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
pseudo.seger&lt;-readRDSFromWeb(paste0(urlremote,"pseudo.seger.rds"))
true_p&lt;-readRDSFromWeb(paste0(urlremote,"true_p.rds"))
SCDC_ENSEMBLE_MAD&lt;-readRDSFromWeb(paste0(urlremote,"SCDC_ENSEMBLE_MAD_seger.rds"))
evaluate(SCDC_ENSEMBLE_MAD,true_p)$all.eva
</code></pre>

<hr>
<h2 id='generateBulk'>Pseudo bulk data generation function</h2><span id='topic+generateBulk'></span>

<h3>Description</h3>

<p>This function generates a pseudo bulk samples by random sampled number of cells per subject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generateBulk(eset,ct.varname,sample,disease = NULL,ct.sub,prop_mat = NULL,
nbulk = 50,samplewithRep = FALSE,low_s = 0.3,upp_s = 0.7)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generateBulk_+3A_eset">eset</code></td>
<td>
<p>The 'ExpressionSet' object for single cells.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_ct.varname">ct.varname</code></td>
<td>
<p>Variable name for 'cell types'.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_sample">sample</code></td>
<td>
<p>Variable name for subject/samples.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_disease">disease</code></td>
<td>
<p>Indicate the health condition of subjects.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_ct.sub">ct.sub</code></td>
<td>
<p>A subset of cell types that are selected to construct pseudo bulk samples. If NULL, then all cell types are used.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_prop_mat">prop_mat</code></td>
<td>
<p>Manually input the cell-type proportion for pseudo bulk samples.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_nbulk">nbulk</code></td>
<td>
<p>The number of pseudo bulk samples to be constructed.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_samplewithrep">samplewithRep</code></td>
<td>
<p>Logical, randomly sample single cells with replacement. Default is F.</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_low_s">low_s</code></td>
<td>
<p>Lower support a for uniform distribution U[a,b].</p>
</td></tr>
<tr><td><code id="generateBulk_+3A_upp_s">upp_s</code></td>
<td>
<p>Upper support b for uniform distribution U[a,b].</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Pseudo bulk samples in the format of 'ExpressionSet',
and the true cell-type proportions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
seger&lt;-readRDSFromWeb(paste0(urlremote,"segerstolpe.rds"))

##generate a pseudo bulk data with two samples
set.seed(1234567)
pseudo.seger&lt;-generateBulk(seger[["sc.eset.qc"]], ct.varname = "cluster",
sample = "sample", ct.sub = c("alpha","beta","delta","gamma"),
nbulk = 2, low_s = 0.3, upp_s = 0.7)

</code></pre>

<hr>
<h2 id='InteRD.predict.prop'>Extract the estimated proportions from InteRD</h2><span id='topic+InteRD.predict.prop'></span>

<h3>Description</h3>

<p>This function extract estimated cell type proportions via InteRD1 and InteRD2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InteRD.predict.prop(InteRD.output)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InteRD.predict.prop_+3A_interd.output">InteRD.output</code></td>
<td>
<p>An object from InteRD1 or InteRD2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated cell type proportions from InteRD.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
InteRD1.output&lt;-readRDSFromWeb(paste0(urlremote,"InteRD1.output.rds"))
lambda_option&lt;-0
cell_type_unique&lt;-c("alpha","beta","delta","gamma")
InteRD1&lt;-InteRD.predict.prop(InteRD.output=InteRD1.output)

</code></pre>

<hr>
<h2 id='InteRD1'>The InteRD1 estimate from reference ensemble</h2><span id='topic+InteRD1'></span>

<h3>Description</h3>

<p>This function provides a reference-based deconvolution by resembling all estimated cell-type proportions based on each reference set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InteRD1(bulk.data,list_marker,cell_type_unique,comb_used,
lambda_option,tol=1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InteRD1_+3A_bulk.data">bulk.data</code></td>
<td>
<p>The 'ExpressionSet' object for a target bulk data.</p>
</td></tr>
<tr><td><code id="InteRD1_+3A_list_marker">list_marker</code></td>
<td>
<p>A list of pre-specified marker genes corresponding to each cell type.</p>
</td></tr>
<tr><td><code id="InteRD1_+3A_cell_type_unique">cell_type_unique</code></td>
<td>
<p>A list of cell types. It should match the order in list.marker.</p>
</td></tr>
<tr><td><code id="InteRD1_+3A_comb_used">comb_used</code></td>
<td>
<p>A list of pre-estimated cell type proportions based on different references.</p>
</td></tr>
<tr><td><code id="InteRD1_+3A_lambda_option">lambda_option</code></td>
<td>
<p>A sequence of values for the tuning parameter.</p>
</td></tr>
<tr><td><code id="InteRD1_+3A_tol">tol</code></td>
<td>
<p>A tolerance value for convergence. The default is 1e-06</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing estimated cell type proportions corresponding to each tuning value, named 'est';
and a sequence of goodness-of-fit values corresponding to each tuning value, named 'metrics'.
The smaller the better; and a list of weights corresponding to each tuning value, named 'weights_list'.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
pseudo.seger&lt;-readRDSFromWeb(paste0(urlremote,"pseudo.seger.rds"))
comb&lt;-readRDSFromWeb(paste0(urlremote,"comb_seger.rds"))
list_marker&lt;-readRDSFromWeb(paste0(urlremote,"list_markerbaron20.rds"))
lambda_option&lt;-0
cell_type_unique&lt;-c("alpha","beta","delta","gamma")
InteRD1.output&lt;-InteRD1(bulk.data =pseudo.seger,list_marker,
cell_type_unique,comb_used=comb,lambda_option,tol=1e-02)
InteRD1&lt;-InteRD.predict.prop(InteRD.output=InteRD1.output)

</code></pre>

<hr>
<h2 id='InteRD2'>The InteRD2 estimate</h2><span id='topic+InteRD2'></span>

<h3>Description</h3>

<p>This function provides a robust deconvolution framework to integrate information from scRNA-seq references,
marker genes, and prior biological knowledge.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InteRD2(bulk.data,list_marker,cell_type_unique,comb_sampled,ave_est,ave_sd,
lambda_option,tol=0.0005)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InteRD2_+3A_bulk.data">bulk.data</code></td>
<td>
<p>The 'ExpressionSet' object for a target bulk data.</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_list_marker">list_marker</code></td>
<td>
<p>A list of pre-specified marker genes corresponding to each cell type.</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_cell_type_unique">cell_type_unique</code></td>
<td>
<p>A list of cell types. It should match the order in list.marker.</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_comb_sampled">comb_sampled</code></td>
<td>
<p>A pre-specified cell type proportions for the target bulk data, which could be obtained from reference-based deconvolution approach.</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_ave_est">ave_est</code></td>
<td>
<p>A pre-specified population-level cell type proportions, which could be obtained from single-cell RNA-seq and external expression data from different studies, species, or data types</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_ave_sd">ave_sd</code></td>
<td>
<p>A pre-specified standard deviation for cell-type proportion estimation. The default is 1 for each cell type.</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_lambda_option">lambda_option</code></td>
<td>
<p>A sequence of values for the tuning parameter.</p>
</td></tr>
<tr><td><code id="InteRD2_+3A_tol">tol</code></td>
<td>
<p>A tolerance value for convergence. The default is 0.0005.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing estimated cell type proportions corresponding to each tuning value, named 'est'; and a sequence of goodness-of-fit values corresponding to each tuning value, named 'metrics'. The smaller the better.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
pseudo.seger&lt;-readRDSFromWeb(paste0(urlremote,"pseudo.seger.rds"))
InteRD1&lt;-readRDSFromWeb(paste0(urlremote,"InteRD1.rds"))
ave_est&lt;-readRDSFromWeb(paste0(urlremote,"ave_est.rds"))
ave_sd&lt;-readRDSFromWeb(paste0(urlremote,"ave_sd.rds"))
list_marker&lt;-readRDSFromWeb(paste0(urlremote,"list_markerbaron20.rds"))
lambda_option&lt;-0
cell_type_unique&lt;-c("alpha","beta","delta","gamma")
lambda_option&lt;-10e+05
InteRD2.output&lt;-InteRD2(bulk.data=pseudo.seger,list_marker,cell_type_unique,
comb_sampled=InteRD1,ave_est,ave_sd,lambda_option=lambda_option,tol=0.01)
InteRD2&lt;-InteRD.predict.prop(InteRD.output=InteRD2.output)

</code></pre>

<hr>
<h2 id='pop.ct.prop.scRNA'>Calculate the population-level cell type proportions from a single-cell data.</h2><span id='topic+pop.ct.prop.scRNA'></span>

<h3>Description</h3>

<p>Calculate population-level cell type proportions from single-cell data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pop.ct.prop.scRNA(scRNA,cluster="cluster",sample="sample",cell_type_unique)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pop.ct.prop.scRNA_+3A_scrna">scRNA</code></td>
<td>
<p>The 'ExpressionSet' object for single-cell data.</p>
</td></tr>
<tr><td><code id="pop.ct.prop.scRNA_+3A_cluster">cluster</code></td>
<td>
<p>The character string specifying the variable name for cell types. The default is &quot;cluster&quot;.</p>
</td></tr>
<tr><td><code id="pop.ct.prop.scRNA_+3A_sample">sample</code></td>
<td>
<p>The character string specifying the variable name for subject/samples. The default is &quot;sample&quot;.</p>
</td></tr>
<tr><td><code id="pop.ct.prop.scRNA_+3A_cell_type_unique">cell_type_unique</code></td>
<td>
<p>A vector of cell types. It should match the order in list.marker.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The population-level cell type proportions ('pop.ct.prop') and corresponding standard deviations ('pop.ct.sd').
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
seger&lt;-readRDSFromWeb(paste0(urlremote,"segerstolpe.rds"))
cell_type_unique&lt;-c("alpha","beta","delta","gamma")
ave_est&lt;-pop.ct.prop.scRNA(scRNA=seger[["sc.eset.qc"]],
cell_type_unique=cell_type_unique)$pop.ct.prop
ave_est

</code></pre>

<hr>
<h2 id='Ref_free'>A reference-free deconvolution estimate</h2><span id='topic+Ref_free'></span>

<h3>Description</h3>

<p>This function provides a reference-free deconvolution estimate, given a list of marker genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Ref_free(bulk.data,list_marker,cell_type_unique,tol=0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Ref_free_+3A_bulk.data">bulk.data</code></td>
<td>
<p>The 'ExpressionSet' object for a target bulk data.</p>
</td></tr>
<tr><td><code id="Ref_free_+3A_list_marker">list_marker</code></td>
<td>
<p>A list of pre-specified marker genes corresponding to each cell type.</p>
</td></tr>
<tr><td><code id="Ref_free_+3A_cell_type_unique">cell_type_unique</code></td>
<td>
<p>A list of cell types. It should match the order in 'list.marker'.</p>
</td></tr>
<tr><td><code id="Ref_free_+3A_tol">tol</code></td>
<td>
<p>A tolerance value for convergence. The default is 0.001.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The estimated cell type proportions, named 'est'; and a goodness-of-fit value, named 'metrics'. The smaller the better.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##read data
library(InteRD)
readRDSFromWeb&lt;-function(ref) {readRDS(gzcon(url(ref)))}
urlremote&lt;-"https://github.com/chencxxy28/Data/raw/main/data_InteRD/"
pseudo.seger&lt;-readRDSFromWeb(paste0(urlremote,"pseudo.seger.rds"))
list_marker&lt;-readRDSFromWeb(paste0(urlremote,"list_markerbaron20.rds"))
cell_type_unique&lt;-c("alpha","beta","delta","gamma")
ref_free.output&lt;-Ref_free(bulk.data=pseudo.seger,list_marker=list_marker,
cell_type_unique=cell_type_unique,tol=0.01) #make tol=0.001
reffree&lt;-InteRD.predict.prop(InteRD.output=ref_free.output)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
