<!DOCTYPE html><html><head><title>Help for package IntegratedJM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IntegratedJM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#IntegratedJM'><p>Joint Modelling of Gene Expression and Bio-activity data, taking care of the effect due to Fingerprint Feature.</p></a></li>
<li><a href='#fitJM'><p>fitJM</p></a></li>
<li><a href='#getCorrUnad'><p>getCorrUnad</p></a></li>
<li><a href='#multiplot'><p>multiplot</p></a></li>
<li><a href='#non_nullcov'><p>non_nullcov</p></a></li>
<li><a href='#nullcov'><p>nullcov</p></a></li>
<li><a href='#plot1gene'><p>plot1gene</p></a></li>
<li><a href='#plotAsso'><p>plotAsso</p></a></li>
<li><a href='#plotEff'><p>plotEff</p></a></li>
<li><a href='#sampleData'><p>Sample Data Example</p></a></li>
<li><a href='#topkGenes'><p>topkGenes</p></a></li>
<li><a href='#volcano'><p>volcano</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Joint Modeling of the Gene-Expression and Bioassay Data, Taking
Care of the Effect Due to a Fingerprint Feature</td>
</tr>
<tr>
<td>Version:</td>
<td>1.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-08-03</td>
</tr>
<tr>
<td>Author:</td>
<td>Rudradev Sengupta, Nolen Joy Perualila</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Rudradev Sengupta &lt;rudradev.sengupta@uhasselt.be&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Offers modeling the association between gene-expression and bioassay data, taking care of the effect due to a fingerprint feature and helps with several plots to better understand the analysis.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0), grid</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, nlme, Biobase</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Repository/R-Forge/Project:</td>
<td>integratedjm</td>
</tr>
<tr>
<td>Repository/R-Forge/Revision:</td>
<td>14</td>
</tr>
<tr>
<td>Repository/R-Forge/DateTimeStamp:</td>
<td>2017-08-03 19:03:01</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-03 22:37:38 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-03 19:25:12 UTC; rforge</td>
</tr>
</table>
<hr>
<h2 id='IntegratedJM'>Joint Modelling of Gene Expression and Bio-activity data, taking care of the effect due to Fingerprint Feature.</h2><span id='topic+IntegratedJM'></span><span id='topic+IntegratedJM-package'></span>

<h3>Description</h3>

<p>The IntegratedJM package contains the functions to fit the Joint Model, to classify the genes based on different criteria and necessary plot functions.
</p>

<hr>
<h2 id='fitJM'>fitJM</h2><span id='topic+fitJM'></span>

<h3>Description</h3>

<p>The fitJM function fits the model for all the genes for a specific bio-activity vector and a particular fingerprint feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fitJM(dat, responseVector, covariate = NULL, methodMultTest)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fitJM_+3A_dat">dat</code></td>
<td>
<p>Contains the gene expression data matrix for all the genes - can be a matrix or an expression set.</p>
</td></tr>
<tr><td><code id="fitJM_+3A_responsevector">responseVector</code></td>
<td>
<p>Vector containing the bio-activity data.</p>
</td></tr>
<tr><td><code id="fitJM_+3A_covariate">covariate</code></td>
<td>
<p>Vector of 0's and 1's, containing data about the fingerprint feature.</p>
</td></tr>
<tr><td><code id="fitJM_+3A_methodmulttest">methodMultTest</code></td>
<td>
<p>Character string to specify the multiple testing method. Default is the BH-FDR method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default for the covariate parameter is NULL and if no covariate is specified it returns a data frame containing 5 variables, named as &quot;Pearson&quot;,&quot;Spearman&quot;,&quot;p&quot;, &quot;adj-p&quot;,&quot;logratio&quot;
and the data frame is ordered based on the column &quot;p&quot; which is the p-value obtained from the Log-Ratio Test. If there is a covariate, then the output is a dataframe containing 13 variables for all the genes,named as
&quot;adjPearson&quot;,&quot;adjSpearman&quot;,&quot;pPearson&quot;,&quot;Pearson&quot;, &quot;Spearman&quot;, &quot;pAdjR&quot;, &quot;CovEffect1&quot;, &quot;adjPeffect1&quot;, &quot;CovEffect2&quot;, &quot;adjPeffect2&quot;, &quot;rawP1&quot;, &quot;rawP2&quot;,&quot;logratio&quot; and sorted based on &quot;rawP1&quot;  and &quot;pPearson&quot; which are
p-value corresponding to the effect of the fingerprint feature on the gene expression data as obtained from the t-table after fitting the model using gls and the p-value obtained from the Log-Ratio Test, respectively.
In the first case without any covariate it calls the nullcov function inside it, otherwise the non_nullcov function is called to do the analysis.
</p>


<h3>Value</h3>

<p>A data frame, containing the results of the model, to be used later for plots or to identify the top genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
jmRes &lt;- fitJM(dat=gene_eset,responseVector=activity,methodMultTest='fdr')
jmRes &lt;- fitJM(dat=gene_eset,responseVector=activity,covariate = fp,methodMultTest='fdr')

## End(Not run)
</code></pre>

<hr>
<h2 id='getCorrUnad'>getCorrUnad</h2><span id='topic+getCorrUnad'></span>

<h3>Description</h3>

<p>The getCorrUnad function is a support function for the function plot1gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCorrUnad(geneName, fp, fpName, responseVector, dat, resPlot)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCorrUnad_+3A_genename">geneName</code></td>
<td>
<p>Character string, specifying the name of the gene.</p>
</td></tr>
<tr><td><code id="getCorrUnad_+3A_fp">fp</code></td>
<td>
<p>Vector containing 0's and 1's - the data about the fingerprint feature.</p>
</td></tr>
<tr><td><code id="getCorrUnad_+3A_fpname">fpName</code></td>
<td>
<p>Character string, used to make the title of the plots. If not specified, the plot title will be blank.</p>
</td></tr>
<tr><td><code id="getCorrUnad_+3A_responsevector">responseVector</code></td>
<td>
<p>Vector containing the bio-activity data.</p>
</td></tr>
<tr><td><code id="getCorrUnad_+3A_dat">dat</code></td>
<td>
<p>Contains the gene expression data matrix for all the genes - can be a matrix or an expression set.</p>
</td></tr>
<tr><td><code id="getCorrUnad_+3A_resplot">resPlot</code></td>
<td>
<p>Logical. If TRUE, creates the plot data for the residual plot</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Works as a support function for plot1gene.
</p>


<h3>Value</h3>

<p>A list containing the data to create the respective plots and the unadjusted association between the gene expression and bio-activity data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
getCorrUnad(geneName="Gene21",fp=fp,fpName="Fingerprint",
responseVector=activity,dat=gene_eset,resPlot=TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='multiplot'>multiplot</h2><span id='topic+multiplot'></span>

<h3>Description</h3>

<p>The multiplot function plots multiple ggplots in the same window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplot(..., cols = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplot_+3A_...">...</code></td>
<td>
<p>ggplot2 objects, separated by comma.</p>
</td></tr>
<tr><td><code id="multiplot_+3A_cols">cols</code></td>
<td>
<p>Integer, specifying the number of plots in one row in the layout.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots multiple ggplots in the same window - multiplot(p1,p2,p3,p4, cols=2) is similar to the standard R notation par(mfrow=c(2,2)).
</p>


<h3>Value</h3>

<p>Creates multiple ggplots in same window
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
multiplot(p1,p2,p3,cols=3)

## End(Not run)
</code></pre>

<hr>
<h2 id='non_nullcov'>non_nullcov</h2><span id='topic+non_nullcov'></span>

<h3>Description</h3>

<p>The non_nullcov function is called while fitting the model when the covariate is specified in the fitJM function. It returns a data.frame containing the results after fitting the model. The output of this function is also the output of the fitJM function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>non_nullcov(dat, responseVector, covariate, methodMultTest, data_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="non_nullcov_+3A_dat">dat</code></td>
<td>
<p>Contains the gene expression data matrix for all the genes - can be a matrix or an expression set.</p>
</td></tr>
<tr><td><code id="non_nullcov_+3A_responsevector">responseVector</code></td>
<td>
<p>Vector containing the bio-activity data.</p>
</td></tr>
<tr><td><code id="non_nullcov_+3A_covariate">covariate</code></td>
<td>
<p>Vector of 0's and 1's, containing data about the fingerprint feature.</p>
</td></tr>
<tr><td><code id="non_nullcov_+3A_methodmulttest">methodMultTest</code></td>
<td>
<p>Character string to specify the multiple testing method.</p>
</td></tr>
<tr><td><code id="non_nullcov_+3A_data_type">data_type</code></td>
<td>
<p>Binary, specifying the type of the parameter dat: 0 - expressionSet, 1 - matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits the model, adjusting for the covariate effect, using gls, calculates the correlation, p-values, adjusted p-values (based on the multiple testing method) and logratio from LRT and returns the required results.
</p>


<h3>Value</h3>

<p>A data frame, containing the results of the model - same as the output of the fitJM function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
non_nullcov(dat=gene_eset,responseVector=activity,covariate=fp,methodMultTest='fdr',data_type=0)

## End(Not run)
</code></pre>

<hr>
<h2 id='nullcov'>nullcov</h2><span id='topic+nullcov'></span>

<h3>Description</h3>

<p>The nullcov function is called while fitting the model when there no covariate is specified in the fitJM function. It returns a data.frame containing the results after fitting the model. The output of this function is also the output of the fitJM function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nullcov(dat, responseVector, methodMultTest, data_type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nullcov_+3A_dat">dat</code></td>
<td>
<p>Contains the gene expression data matrix for all the genes - can be a matrix or an expression set.</p>
</td></tr>
<tr><td><code id="nullcov_+3A_responsevector">responseVector</code></td>
<td>
<p>Vector containing the bio-activity data.</p>
</td></tr>
<tr><td><code id="nullcov_+3A_methodmulttest">methodMultTest</code></td>
<td>
<p>Character string to specify the multiple testing method. Default is the BH-FDR method.</p>
</td></tr>
<tr><td><code id="nullcov_+3A_data_type">data_type</code></td>
<td>
<p>Binary, specifying the type of the parameter dat: 0 - expressionSet, 1 - matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Fits the model using gls, calculates the correlation, p-values, adjusted p-values (based on the multiple testing method) and logratio from LRT and returns the required results.
</p>


<h3>Value</h3>

<p>A data frame, containing the results of the model - same as the output of the fitJM function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
nullcov(dat=gene_eset,responseVector=activity,methodMultTest='fdr',data_type=0)

## End(Not run)
</code></pre>

<hr>
<h2 id='plot1gene'>plot1gene</h2><span id='topic+plot1gene'></span>

<h3>Description</h3>

<p>The plot1gene function plots the data for a single gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot1gene(geneName, fp, fpName = "", responseVector, dat, resPlot = TRUE,
  colP = "blue", colA = "white")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot1gene_+3A_genename">geneName</code></td>
<td>
<p>Character string, specifying the name of the gene.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_fp">fp</code></td>
<td>
<p>Vector containing 0's and 1's - the data about the fingerprint feature.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_fpname">fpName</code></td>
<td>
<p>Character string, used to make the title of the plots. If not specified, the plot title will be blank.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_responsevector">responseVector</code></td>
<td>
<p>Vector containing the bio-activity data.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_dat">dat</code></td>
<td>
<p>Contains the gene expression data matrix for all the genes - can be a matrix or an expression set.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_resplot">resPlot</code></td>
<td>
<p>Logical. If TRUE, also plots the residual from the gls fit. Default is TRUE.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_colp">colP</code></td>
<td>
<p>Character string, specifying the colour for the 1's in the fp parameter. Default is blue.</p>
</td></tr>
<tr><td><code id="plot1gene_+3A_cola">colA</code></td>
<td>
<p>Character string, specifying the colour for the 0's in the fp parameter. Default is white.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calls the getCorrUnad function and creates the plot(s) accordingly.
</p>


<h3>Value</h3>

<p>Creates a plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plot1gene(geneName="Gene21",fp=fp,fpName="Fingerprint",responseVector=activity,dat=gene_eset)

## End(Not run)
</code></pre>

<hr>
<h2 id='plotAsso'>plotAsso</h2><span id='topic+plotAsso'></span>

<h3>Description</h3>

<p>The plotAsso function is used to plot the unadjusted association vs the adjusted association for all the genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAsso(jointModelResult, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAsso_+3A_jointmodelresult">jointModelResult</code></td>
<td>
<p>Data frame, containing the results from the fitJM function.</p>
</td></tr>
<tr><td><code id="plotAsso_+3A_type">type</code></td>
<td>
<p>Character string, specifying the type of association - Pearson or Spearman.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the unadjusted association vs the adjusted association for all the genes.
</p>


<h3>Value</h3>

<p>Creates a plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plotAsso(jointModelResult=jmRes,type="Pearson")

## End(Not run)
</code></pre>

<hr>
<h2 id='plotEff'>plotEff</h2><span id='topic+plotEff'></span>

<h3>Description</h3>

<p>The plotEff function is used to plot the fingerprint effect on gene expression vs the adjusted association for all the genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotEff(jointModelResult, type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotEff_+3A_jointmodelresult">jointModelResult</code></td>
<td>
<p>Data frame, containing the results from the fitJM function.</p>
</td></tr>
<tr><td><code id="plotEff_+3A_type">type</code></td>
<td>
<p>Character string, specifying the type of association - Pearson or Spearman.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots the fingerprint effect on gene expression vs the specified type of adjusted association for all the genes.
</p>


<h3>Value</h3>

<p>Creates a plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
plotEff(jointModelResult=jmRes,type="Pearson")

## End(Not run)
</code></pre>

<hr>
<h2 id='sampleData'>Sample Data Example</h2><span id='topic+sample_data'></span>

<h3>Description</h3>

<p>sample_data; Sample data included in the package. Gene expression data for 500 genes and 20 compounds and data on bio-activity and fingerprint feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(sampleData)</code></pre>


<h3>Format</h3>

<p>The format is:
List containing one gene expression data matrix, one vector each on bio-activity and fingerprint feature data, respectively. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sampleData)
gene_mx &lt;- sample_data[[1]]
activity &lt;- sample_data[[2]]
fp &lt;- sample_data[[3]]

</code></pre>

<hr>
<h2 id='topkGenes'>topkGenes</h2><span id='topic+topkGenes'></span>

<h3>Description</h3>

<p>The topkGenes function is to identify the top genes based on different criteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>topkGenes(jointModelResult, subset_type, ranking, k = 10, sigLevel = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="topkGenes_+3A_jointmodelresult">jointModelResult</code></td>
<td>
<p>Data frame, containing the results from the fitJM function.</p>
</td></tr>
<tr><td><code id="topkGenes_+3A_subset_type">subset_type</code></td>
<td>
<p>Character string to specify the set of genes. It can have four values: &quot;Effect&quot; for only differentially expressed genes, &quot;Correlation&quot; for only correlated genes, &quot;Effect and Correlation&quot; for genes which are both differentially expressed &amp; correlated and &quot;Other&quot; for the genes which are neither differentially expressed nor correlated.</p>
</td></tr>
<tr><td><code id="topkGenes_+3A_ranking">ranking</code></td>
<td>
<p>Character string, specifying one of the columns of the jointModelResult data frame, based on the genes will be ranked within the selected subset.</p>
</td></tr>
<tr><td><code id="topkGenes_+3A_k">k</code></td>
<td>
<p>Integer, specifying the number of genes, to be returned from the list of top genes. Default is 10.</p>
</td></tr>
<tr><td><code id="topkGenes_+3A_siglevel">sigLevel</code></td>
<td>
<p>Numeric between 0 and 1, specifying the level of significance, used to select the subset of genes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returned data frame contains 6 columns, named as &quot;Genes&quot;,&quot;FP-Effect&quot;, &quot;p-adj(Effect)&quot;, &quot;Unadj.Asso.&quot;,&quot;Adj.Asso.&quot;, &quot;p-adj(Adj.Asso.)&quot;.
</p>


<h3>Value</h3>

<p>A data frame containing top k genes according to the specified criteria from the specified set of genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
topkGenes(jointModelResult=jmRes,subset_type="Effect",ranking="Pearson",k=10,sigLevel = 0.05)

## End(Not run)
</code></pre>

<hr>
<h2 id='volcano'>volcano</h2><span id='topic+volcano'></span>

<h3>Description</h3>

<p>The volcano function produces the volcano plot for logratio / fp-effect vs corresponding p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcano(x, pValue, pointLabels, topPValues = 10, topXvalues = 10,
  smoothScatter = TRUE, xlab = NULL, ylab = NULL, main = NULL,
  newpage = TRUE, additionalPointsToLabel = NULL,
  additionalLabelColor = "red", dir = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="volcano_+3A_x">x</code></td>
<td>
<p>Numeric vector of logratios or covariate effect values to be plotted.</p>
</td></tr>
<tr><td><code id="volcano_+3A_pvalue">pValue</code></td>
<td>
<p>Numeric vector of corresponding p-values obtained from some statistical test.</p>
</td></tr>
<tr><td><code id="volcano_+3A_pointlabels">pointLabels</code></td>
<td>
<p>Character vector providing the texts for the points to be labelled in the plot.</p>
</td></tr>
<tr><td><code id="volcano_+3A_toppvalues">topPValues</code></td>
<td>
<p>Number of top p-values to be labelled. Default value is 10.</p>
</td></tr>
<tr><td><code id="volcano_+3A_topxvalues">topXvalues</code></td>
<td>
<p>Number of top logratios or covariate effect values to be labelled. Default value is 10.</p>
</td></tr>
<tr><td><code id="volcano_+3A_smoothscatter">smoothScatter</code></td>
<td>
<p>Logical parameter to decide if a smooth plot is expected or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="volcano_+3A_xlab">xlab</code></td>
<td>
<p>Text for the x-axis of the plot. Default is NULL.</p>
</td></tr>
<tr><td><code id="volcano_+3A_ylab">ylab</code></td>
<td>
<p>Text for the y-axis of the plot. Default is NULL.</p>
</td></tr>
<tr><td><code id="volcano_+3A_main">main</code></td>
<td>
<p>Text for the main title of the plot. Default is NULL.</p>
</td></tr>
<tr><td><code id="volcano_+3A_newpage">newpage</code></td>
<td>
<p>Logical parameter</p>
</td></tr>
<tr><td><code id="volcano_+3A_additionalpointstolabel">additionalPointsToLabel</code></td>
<td>
<p>Set of points other than the top values to be labelled in the plot. Default is NULL.</p>
</td></tr>
<tr><td><code id="volcano_+3A_additionallabelcolor">additionalLabelColor</code></td>
<td>
<p>Colour of the additionally labelled points. Default colour is red.</p>
</td></tr>
<tr><td><code id="volcano_+3A_dir">dir</code></td>
<td>
<p>Logical parameter deciding if the top values should be in decreasing (= TRUE) or increasing (= FALSE) order. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Creates a plot which looks like a volcano with the interesting points labelled within the plot.
</p>


<h3>Value</h3>

<p>A plot which looks like a volcano.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
volcano(x=jmRes$CovEffect1,pValue=jmRes$rawP1,pointLabels=rownames(jmRes),
topPValues = 10, topXvalues = 10,xlab="FP Effect (alpha)",ylab="-log(p-values)")

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
