<!DOCTYPE html><html><head><title>Help for package InterSIM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {InterSIM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#InterSIM'>
<p>InterSIM</p></a></li>
<li><a href='#InterSIM-package'>
<p>Simulation of inter-related genomic datasets</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Simulation of Inter-Related Genomic Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>2.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-07-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Prabhakar Chalise, Rama Raghavan, Brooke Fridley</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Prabhakar Chalise &lt;pchalise@kumc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Generates three inter-related genomic datasets : methylation, gene expression and protein expression.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), MASS, NMF, tools</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-13 22:20:16 UTC; pchalise</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-16 17:40:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='InterSIM'>
InterSIM
</h2><span id='topic+InterSIM'></span><span id='topic+CpG.gene.map.for.DEG'></span><span id='topic+cov.M'></span><span id='topic+cov.expr'></span><span id='topic+cov.protein'></span><span id='topic+mean.M'></span><span id='topic+mean.expr'></span><span id='topic+mean.expr.with.mapped.protein'></span><span id='topic+mean.protein'></span><span id='topic+methyl.gene.level.mean'></span><span id='topic+protein.gene.map.for.DEP'></span><span id='topic+rho.expr.protein'></span><span id='topic+rho.methyl.expr'></span><span id='topic+rev.logit'></span><span id='topic+logit'></span>

<h3>Description</h3>

<p>This function simulates three inter-related genomic datasets : DNA methylation, gene expression and protein expression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>InterSIM(n.sample=500, cluster.sample.prop=c(0.30,0.30,0.40), delta.methyl=2.0,
delta.expr=2.0, delta.protein=2.0, p.DMP=0.2,
p.DEG=NULL, p.DEP=NULL, sigma.methyl=NULL, sigma.expr=NULL, sigma.protein=NULL,
cor.methyl.expr=NULL, cor.expr.protein=NULL, do.plot=FALSE, sample.cluster=TRUE,
feature.cluster=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="InterSIM_+3A_n.sample">n.sample</code></td>
<td>

<p>Number of subjects to simulate
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_cluster.sample.prop">cluster.sample.prop</code></td>
<td>

<p>Proportion of samples in the clusters. The number of proportions entered is used to determine the number of clusters in the
simulated data. e.g. if (0.3,0.4,0.3) is entered then the number of clusters will be 3.
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_delta.methyl">delta.methyl</code></td>
<td>

<p>Cluster mean shift for methylation data
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_delta.expr">delta.expr</code></td>
<td>

<p>Cluster mean shift for expression data
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_delta.protein">delta.protein</code></td>
<td>

<p>Cluster mean shift for protein data
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_p.dmp">p.DMP</code></td>
<td>

<p>proportion of DE CpGs (DE = Differentially Expressed)
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_p.deg">p.DEG</code></td>
<td>

<p>proportion of DE mRNA, if NULL (default) mRNAs mapped by DE CpGs will be selected
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_p.dep">p.DEP</code></td>
<td>

<p>proportion of DE protein, if NULL (default) proteins mapped by DE mRNAs will be selected
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_sigma.methyl">sigma.methyl</code></td>
<td>

<p>Covariance structure methylation data, if NULL (default) precomputed values will be used.
&quot;indep&quot; gives covariance structure with diagonal elements only (Independent features)
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_sigma.expr">sigma.expr</code></td>
<td>

<p>Covariance structure mRNA data, if NULL (default) precomputed values will be used.
&quot;indep&quot; gives covariance structure with diagonal elements only (Independent features)
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_sigma.protein">sigma.protein</code></td>
<td>

<p>Covariance structure Protein data, if NULL (default) precomputed values will be used.
&quot;indep&quot; gives covariance structure with diagonal elements only (Independent features)
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_do.plot">do.plot</code></td>
<td>

<p>TRUE to generate heatmap, default is FALSE
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_sample.cluster">sample.cluster</code></td>
<td>

<p>TRUE (default), if clustering should be done on samples for heatmap. This option will be applicable only if do.plot=TRUE.
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_feature.cluster">feature.cluster</code></td>
<td>

<p>TRUE (default), if clustering should be done on genomic features for heatmap.This option will be applicable only if do.plot=TRUE.
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_cor.methyl.expr">cor.methyl.expr</code></td>
<td>

<p>Correlation between methylation and mRNA, if NULL (default) precomputed values will be used
</p>
</td></tr>
<tr><td><code id="InterSIM_+3A_cor.expr.protein">cor.expr.protein</code></td>
<td>

<p>Correlation between mRNA and protein, if NULL (default) precomputed values will be used
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns three datasets as matrices - DNA methylation, gene expression and protein expression. It also returns a vector that has true cluster assignment for each subject in the generated data.
</p>


<h3>Author(s)</h3>

<p>Prabhakar Chalise &lt;pchalise@kumc.edu&gt;, Rama Raghavan &lt;rraghavan@kumc.edu&gt;, Brooke Fridley &lt;bfridley@kumc.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#
prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 5
sim.data &lt;- InterSIM(n.sample=500, cluster.sample.prop = prop,
delta.methyl=effect, delta.expr=effect, delta.protein=effect,
p.DMP=0.2, p.DEG=NULL, p.DEP=NULL,
sigma.methyl=NULL, sigma.expr=NULL, sigma.protein=NULL,
cor.methyl.expr=NULL, cor.expr.protein=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)
sim.methyl &lt;- sim.data$dat.methyl
sim.expr &lt;- sim.data$dat.expr
sim.protein &lt;- sim.data$dat.protein

</code></pre>

<hr>
<h2 id='InterSIM-package'>
Simulation of inter-related genomic datasets
</h2><span id='topic+InterSIM-package'></span>

<h3>Description</h3>

<p>InterSIM is an R package that generates three inter-related data set with realistic inter- and intra- relationships based on the DNA methylation, mRNA expression and protein expression from the TCGA ovarian cancer study.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> InterSIM</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.2.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-07-13</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL&gt;=2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley;
Maintainer: Prabhakar Chalise
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
