<!DOCTYPE html><html lang="en"><head><title>Help for package interep</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {interep}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cv.interep'><p>k-folds cross-validation for interep</p></a></li>
<li><a href='#dat'><p>simulated data for demonstrating the features of interep</p></a></li>
<li><a href='#dmcp'><p>This function obtains the first derivative function of MCP (Minimax Concave Penalty)</p></a></li>
<li><a href='#interep'><p>fit generalized estimaitng equations with given tuning parameters</p></a></li>
<li><a href='#penalty'><p>This function gives the penalty functions</p></a></li>
<li><a href='#reformat'><p>This function changes the format of the longitudinal data from wide format to long format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interaction Analysis of Repeated Measure Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.4.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Fei Zhou, Jie Ren, Yuwen Liu, Xiaoxi Li, Cen Wu, Yu Jiang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Fei Zhou &lt;fei.zhou@outlook.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Extensive penalized variable selection methods have been developed in the past two decades for analyzing high dimensional omics data, such as gene expressions, single nucleotide polymorphisms (SNPs), copy number variations (CNVs) and others. However, lipidomics data have been rarely investigated by using high dimensional variable selection methods. This package incorporates our recently developed penalization procedures to conduct interaction analysis for high dimensional lipidomics data with repeated measurements. The core module of this package is developed in C++. The development of this software package and the associated statistical methods have been partially supported by an Innovative Research Award from Johnson Cancer Research Center, Kansas State University.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, MASS</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/feizhoustat/interep">https://github.com/feizhoustat/interep</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/feizhoustat/interep/issues">https://github.com/feizhoustat/interep/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-30 03:52:28 UTC; feizh</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-30 04:10:02 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
</table>
<hr>
<h2 id='cv.interep'>k-folds cross-validation for interep</h2><span id='topic+cv.interep'></span>

<h3>Description</h3>

<p>This function does k-fold cross-validation for interep and returns the optimal value of lambda.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cv.interep(e, g, y, beta0, lambda1, lambda2, nfolds, corre, pmethod, maxits)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cv.interep_+3A_e">e</code></td>
<td>
<p>matrix of environment factors.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_g">g</code></td>
<td>
<p>matrix of omics factors. In the case study, the omics measurements are lipidomics data.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_y">y</code></td>
<td>
<p>the longitudinal response.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_beta0">beta0</code></td>
<td>
<p>the intial value for the coefficient vector.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_lambda1">lambda1</code></td>
<td>
<p>a user-supplied sequence of <code class="reqn">\lambda_{1}</code> values, which serves as a tuning parameter for individual predictors.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_lambda2">lambda2</code></td>
<td>
<p>a user-supplied sequence of <code class="reqn">\lambda_{2}</code> values, which serves as a tuning parameter for interactions.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_nfolds">nfolds</code></td>
<td>
<p>the number of folds for cross-validation.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_corre">corre</code></td>
<td>
<p>the working correlation structure that is used in the estimation algorithm. interep provides three choices for the
working correlation structure: &quot;a&quot; as AR-1&quot;, &quot;i&quot; as &quot;independence&quot; and &quot;e&quot; as &quot;exchangeable&quot;.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_pmethod">pmethod</code></td>
<td>
<p>the penalization method. &quot;mixed&quot; refers to MCP penalty to individual main effects and group MCP penalty to interactions; &quot;individual&quot; means MCP penalty to all effects.</p>
</td></tr>
<tr><td><code id="cv.interep_+3A_maxits">maxits</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When dealing with predictors with both main effects and interactions, this function returns two optimal tuning parameters,
<code class="reqn">\lambda_{1}</code> and <code class="reqn">\lambda_{2}</code>; when there are only main effects in the predictors, this function returns <code class="reqn">\lambda_{1}</code>,
which is the optimal tuning parameter for individual predictors containing main effects.
</p>


<h3>Value</h3>

<p>an object of class &quot;cv.interep&quot; is returned, which is a list with components:
</p>
<table role = "presentation">
<tr><td><code>lam1</code></td>
<td>
<p>the optimal <code class="reqn">\lambda_{1}</code>.</p>
</td></tr>
<tr><td><code>lam2</code></td>
<td>
<p>the optimal <code class="reqn">\lambda_{2}</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhou, F., Ren, J., Li, G., Jiang, Y., Li, X., Wang, W.and Wu, C. (2019). Penalized variable selection for Lipid&ndash;environment interactions in a longitudinal lipidomics study.
<a href="https://www.mdpi.com/2073-4425/10/12/1002"><em>Genes</em>, 10(12), 1002</a>
</p>
<p>Zhou, F., Ren, J., Liu, Y., Li, X., Wang, W.and Wu, C. (2022). Interep: An r package for high-dimensional interaction analysis of the repeated measurement data.
<em>Genes</em>, 13(3): 554 <a href="https://doi.org/10.3390/genes13030544">doi:10.3390/genes13030544</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021) Gene–Environment Interaction: a Variable Selection Perspective.
<a href="https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13"><em>Epistasis: Methods and Protocols</em>, 191-223</a> 
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang,Y. and Wu, C. (2020). Semi-parametric Bayesian variable selection for Gene-Environment interactions.
<em>Statistics in Medicine</em>, 39(5): 617-638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Wu, C., Zhou, F., Ren, J., Li, X., Jiang, Y., Ma, S. (2019). A Selective Review of Multi-Level Omics Data Integration Using Variable Selection.
<em>High-Throughput</em>, 8(1) <a href="https://doi.org/10.3390/ht8010004">doi:10.3390/ht8010004</a>
</p>
<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang, Y. and Wu, C. (2019). Robust network-based regularization and variable selection for high-dimensional genomic data in cancer prognosis.
<em>Genetic epidemiology</em>, 43(3), 276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Ren, J., Jung, L., Du, Y., Wu, C., Jiang, Y. and Liu, J. (2019). regnet: Network-Based Regularization for Generalized Linear Models.
<a href="https://cran.r-project.org/package=regnet"><em>R package</em>, version 0.4.0</a>
</p>
<p>Wu, C., Zhang, Q., Jiang, Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi) genetic measurements.
<em>Journal of multivariate analysis</em>, 168, 119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Wu, C., Zhong, P.-S., and Cui, Y. (2018). Additive varying-coefficient model for nonlinear gene-environment interactions.
<em> Statistical Applications in Genetics and Molecular Biology</em>, 17(2) <a href="https://doi.org/10.1515/sagmb-2017-0008">doi:10.1515/sagmb-2017-0008</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y., Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37: 437–456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Ren, J., He, T., Li, Y., Liu, S., Du, Y., Jiang, Y. and Wu, C. (2017). Network-based regularization for high dimensional SNP data in the case–control study of Type 2 diabetes.
<em>BMC genetics</em>, 18(1), 44 <a href="https://doi.org/10.1186/s12863-017-0495-5">doi:10.1186/s12863-017-0495-5</a>
</p>
<p>Jiang, Y., Huang, Y., Du, Y., Zhao, Y., Ren, J., Ma, S., &amp; Wu, C. (2017). Identification of prognostic genes and pathways in lung adenocarcinoma using a Bayesian approach.
<em>Cancer Inform</em>, 1(7) <a href="https://doi.org/10.1177/1176935116684825">doi:10.1177/1176935116684825</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>
</p>
<p>Wu, C., Shi, X., Cui, Y. and Ma, S. (2015). A penalized robust semiparametric approach for gene-environment interactions.
<em>Statistics in Medicine</em>, 34 (30): 4016–4030 <a href="https://doi.org/10.1002/sim.6609">doi:10.1002/sim.6609</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C. and Cui, Y. (2013). A novel method for identifying nonlinear gene–environment interactions in case–control association studies.
<em>Human Genetics</em>, 132(12):1413–1425 <a href="https://doi.org/10.1007/s00439-013-1350-z">doi:10.1007/s00439-013-1350-z</a>
</p>
<p>Wu, C. and Cui, Y. (2013). Boosting signals in gene–based association studies via efficient SNP selection.
<em>Briefings in Bioinformatics</em>, 15(2):279–291 <a href="https://doi.org/10.1093/bib/bbs087">doi:10.1093/bib/bbs087</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<a href="https://www.stt.msu.edu/Links/Research_Memoranda/RM/RM_701.pdf"><em>Technical Report</em>, Michigan State University.</a>
</p>
<p>Wu, C., Li, S., and Cui, Y. (2012). Genetic Association Studies: An Information Content Perspective.
<em>Current Genomics</em>, 13(7),  566–573 <a href="https://doi.org/10.2174/138920212803251382">doi:10.2174/138920212803251382</a>
</p>

<hr>
<h2 id='dat'>simulated data for demonstrating the features of interep</h2><span id='topic+dat'></span>

<h3>Description</h3>

<p>Simulated data for demonstrating the features of interep.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dat")
</code></pre>


<h3>Format</h3>

<p>Each data consists of six components: e, z, x, y, coef and index; index shows the location of the true coefficients used to generate y.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dat")
</code></pre>

<hr>
<h2 id='dmcp'>This function obtains the first derivative function of MCP (Minimax Concave Penalty)</h2><span id='topic+dmcp'></span>

<h3>Description</h3>

<p>This function obtains the first derivative function of MCP (Minimax Concave Penalty)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmcp(theta, lambda, gamma = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dmcp_+3A_theta">theta</code></td>
<td>
<p>a coefficient vector.</p>
</td></tr>
<tr><td><code id="dmcp_+3A_lambda">lambda</code></td>
<td>
<p>the tuning parameter.</p>
</td></tr>
<tr><td><code id="dmcp_+3A_gamma">gamma</code></td>
<td>
<p>the regularization parameter in MCP (Minimax Concave Penalty).
It balances between the unbiasedness and concavity of MCP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Rigorously speaking, the regularization parametre <code class="reqn">\gamma</code> needs to be obtained via a data-driven approach.
Published studies suggest experimenting with a few values, such as 1.8, 3, 4.5, 6, and 10, then fixing its value. In our numerical
study, we have examined this sequence and found that the results are not sensitive to the choice of value of <code class="reqn">\gamma</code>,
and set the value at 3. In practice, to be prudent, values other than 3 should also be investigated. Similar discussions can be found
in the references below.
</p>


<h3>Value</h3>

<p>the first derivative of MCP function.
</p>


<h3>References</h3>

<p>Ren, J., Du, Y., Li, S., Ma, S., Jiang, Y. and Wu, C. (2019). Robust network-based regularization and variable selection for high-dimensional genomic data in cancer prognosis.
<em>Genetic epidemiology</em>, 43(3), 276-291 <a href="https://doi.org/10.1002/gepi.22194">doi:10.1002/gepi.22194</a>
</p>
<p>Ren, J., Jung, L., Du, Y., Wu, C., Jiang, Y. and Liu, J. (2019). regnet: Network-Based Regularization for Generalized Linear Models.
<a href="https://cran.r-project.org/package=regnet"><em>R package</em>, version 0.4.0</a>
</p>
<p>Wu, C., Zhang, Q., Jiang, Y. and Ma, S. (2018). Robust network-based analysis of the associations between (epi) genetic measurements.
<em>Journal of multivariate analysis</em>, 168, 119-130 <a href="https://doi.org/10.1016/j.jmva.2018.06.009">doi:10.1016/j.jmva.2018.06.009</a>
</p>
<p>Ren, J., He, T., Li, Y., Liu, S., Du, Y., Jiang, Y. and Wu, C. (2017). Network-based regularization for high dimensional SNP data in the case–control study of Type 2 diabetes.
<em>BMC genetics</em>, 18(1), 44 <a href="https://doi.org/10.1186/s12863-017-0495-5">doi:10.1186/s12863-017-0495-5</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>theta=runif(20,-5,5)
lambda=1
dmcp(theta,lambda,gamma=3)
</code></pre>

<hr>
<h2 id='interep'>fit generalized estimaitng equations with given tuning parameters</h2><span id='topic+interep'></span>

<h3>Description</h3>

<p>This function makes predictions for generalized estimating equation with a given value of lambda.
Typical usage is to have the cv.interep function compute the optimal lambda, then provide it to
the interep function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interep(e, g, y, beta0, corre, pmethod, lam1, lam2, maxits)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="interep_+3A_e">e</code></td>
<td>
<p>matrix of environment factors.</p>
</td></tr>
<tr><td><code id="interep_+3A_g">g</code></td>
<td>
<p>matrix of omics factors. In the case study, the omics measurements are lipidomics data.</p>
</td></tr>
<tr><td><code id="interep_+3A_y">y</code></td>
<td>
<p>the longitudinal response.</p>
</td></tr>
<tr><td><code id="interep_+3A_beta0">beta0</code></td>
<td>
<p>the inital coefficient vector.</p>
</td></tr>
<tr><td><code id="interep_+3A_corre">corre</code></td>
<td>
<p>the working correlation structure that is used in the estimation algorithm. interep provides three choices for the
working correlation structure: &quot;a&quot; as AR-1&quot;, &quot;i&quot; as &quot;independence&quot; and &quot;e&quot; as &quot;exchangeable&quot;.</p>
</td></tr>
<tr><td><code id="interep_+3A_pmethod">pmethod</code></td>
<td>
<p>the penalization method. &quot;mixed&quot; refers to MCP penalty to individual main effects and group MCP penalty to interactions; &quot;individual&quot; means MCP penalty to all effects.</p>
</td></tr>
<tr><td><code id="interep_+3A_lam1">lam1</code></td>
<td>
<p>the tuning parameter lambda1 for individual predictors.</p>
</td></tr>
<tr><td><code id="interep_+3A_lam2">lam2</code></td>
<td>
<p>the tuning parameter lambda2 for interactions.</p>
</td></tr>
<tr><td><code id="interep_+3A_maxits">maxits</code></td>
<td>
<p>the maximum number of iterations that is used in the estimation algorithm.  The default value is 30</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>coef</code></td>
<td>
<p>the coefficient vector.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhou, F., Ren, J., Li, G., Jiang, Y., Li, X., Wang, W.and Wu, C. (2019). Penalized variable selection for Lipid&ndash;environment interactions in a longitudinal lipidomics study.
<a href="https://www.mdpi.com/2073-4425/10/12/1002"><em>Genes</em>, 10(12), 1002</a>
</p>
<p>Zhou, F., Ren, J., Liu, Y., Li, X., Wang, W.and Wu, C. (2022). Interep: An r package for high-dimensional interaction analysis of the repeated measurement data.
<em>Genes</em>, 13(3): 554 <a href="https://doi.org/10.3390/genes13030544">doi:10.3390/genes13030544</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021) Gene–Environment Interaction: a Variable Selection Perspective.
<a href="https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13"><em>Epistasis: Methods and Protocols</em>, 191-223</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dat")
e=dat$e
g=dat$z
y=dat$y
beta0=dat$coef
index=dat$index
b = interep(e, g, y,beta0,corre="e",pmethod="mixed",lam1=dat$lam1, lam2=dat$lam2,maxits=30)
b[abs(b)&lt;0.05]=0
pos = which(b != 0)
tp = length(intersect(index, pos))
fp = length(pos) - tp
list(tp=tp, fp=fp)

</code></pre>

<hr>
<h2 id='penalty'>This function gives the penalty functions</h2><span id='topic+penalty'></span>

<h3>Description</h3>

<p>This function gives the penalty functions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penalty(x, n, p, q, beta, lam1, pmethod, p1, lam2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="penalty_+3A_x">x</code></td>
<td>
<p>matrix of covariates.</p>
</td></tr>
<tr><td><code id="penalty_+3A_n">n</code></td>
<td>
<p>the sample size.</p>
</td></tr>
<tr><td><code id="penalty_+3A_p">p</code></td>
<td>
<p>the number of predictors.</p>
</td></tr>
<tr><td><code id="penalty_+3A_q">q</code></td>
<td>
<p>the number of environment factors.</p>
</td></tr>
<tr><td><code id="penalty_+3A_beta">beta</code></td>
<td>
<p>the coefficient vector.</p>
</td></tr>
<tr><td><code id="penalty_+3A_lam1">lam1</code></td>
<td>
<p>the tuning parameter lambda1 for individual penalty.</p>
</td></tr>
<tr><td><code id="penalty_+3A_pmethod">pmethod</code></td>
<td>
<p>the penalization method. &quot;mixed&quot; refers to MCP penalty to individual main effects and group MCP penalty to interactions; &quot;individual&quot; means MCP penalty to all effects.</p>
</td></tr>
<tr><td><code id="penalty_+3A_p1">p1</code></td>
<td>
<p>the number of gene factors.</p>
</td></tr>
<tr><td><code id="penalty_+3A_lam2">lam2</code></td>
<td>
<p>the tuning parameter lambda2 for group penalty.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>E</code></td>
<td>
<p>the penalty function.</p>
</td></tr>
</table>

<hr>
<h2 id='reformat'>This function changes the format of the longitudinal data from wide format to long format</h2><span id='topic+reformat'></span>

<h3>Description</h3>

<p>This function changes the format of the longitudinal data from wide format to long format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reformat(k, y, x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="reformat_+3A_k">k</code></td>
<td>
<p>the number of repeated measurement.</p>
</td></tr>
<tr><td><code id="reformat_+3A_y">y</code></td>
<td>
<p>the longitudinal response.</p>
</td></tr>
<tr><td><code id="reformat_+3A_x">x</code></td>
<td>
<p>a matrix of predictors, consisting of omics and environment factors, as well as their interactions. In
the case study, the omics measurements are lipidomics data.</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
