<!DOCTYPE html><html><head><title>Help for package IntNMF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {IntNMF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ClusterEntropy'>
<p>A function to measure cluster entropy</p></a></li>
<li><a href='#ClusterPurity'>
<p>A function to measure cluster purity</p></a></li>
<li><a href='#ConsensusMatPlot'>
<p>A function to create image plot of the consensus matrix</p></a></li>
<li><a href='#nmf.mnnals'>
<p>Nonnegative Matrix Factorization of Multiple data using Nonnegative Alternating Least Square</p></a></li>
<li><a href='#nmf.opt.k'>
<p>Selection of optimum number of clusters (k)</p></a></li>
<li><a href='#SilhouettePlot'>
<p>A function to plot silhouette width</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrative Clustering of Multiple Genomic Dataset</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-07-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Prabhakar Chalise, Rama Raghavan, Brooke Fridley</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Prabhakar Chalise &lt;pchalise@kumc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Carries out integrative clustering analysis using multiple types of genomic dataset using integrative Non-negative Matrix factorization. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), MASS, NMF, mclust, cluster, InterSIM</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-18 15:39:55 UTC; pchalise</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-18 22:40:19 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
</table>
<hr>
<h2 id='ClusterEntropy'>
A function to measure cluster entropy
</h2><span id='topic+ClusterEntropy'></span>

<h3>Description</h3>

<p>Given the true clustering assignment for the subjects, this function calculates cluster entropy index comparing with clustering assignment determined by integrative NMF algorithm. Smaller value of cluster entropy indicates better cluster predictive discrimination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClusterEntropy(ComputedClusters, TrueClasses)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClusterEntropy_+3A_computedclusters">ComputedClusters</code></td>
<td>

<p>Clustering assignment determined by integrative NMF algorithm
</p>
</td></tr>
<tr><td><code id="ClusterEntropy_+3A_trueclasses">TrueClasses</code></td>
<td>

<p>True clustering assignment of the subjects
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cluster entropy index value
</p>


<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley
</p>


<h3>References</h3>

<p>Kim Hyunsoo and Park Haesun (2007). Sparse non-negative matrix factorization via alternating non-negativity constrained least squares for microarray data analysis. Bioinformatics, 23: 1495-1502.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 2.5
sim.D &lt;- InterSIM(n.sample=100,cluster.sample.prop=prop,delta.methyl=effect,
delta.expr=effect,delta.protein=effect,p.DMP=0.25,p.DEG=NULL,p.DEP=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)
dat1 &lt;- sim.D$dat.methyl
dat2 &lt;- sim.D$dat.expr
dat3 &lt;- sim.D$dat.protein
true.cluster.assignment &lt;- sim.D$clustering.assignment

## Make all data positive by shifting to positive direction.
## Also rescale the datasets so that they are comparable.
if (!all(dat1&gt;=0)) dat1 &lt;- pmax(dat1 + abs(min(dat1)), .Machine$double.eps)
dat1 &lt;- dat1/max(dat1)
if (!all(dat2&gt;=0)) dat2 &lt;- pmax(dat2 + abs(min(dat2)), .Machine$double.eps)
dat2 &lt;- dat2/max(dat2)
if (!all(dat3&gt;=0)) dat3 &lt;- pmax(dat3 + abs(min(dat3)), .Machine$double.eps)
dat3 &lt;- dat3/max(dat3)

# The function nmf.mnnals requires the samples to be on rows and variables on columns.
dat &lt;- list(dat1,dat2,dat3)
fit &lt;- nmf.mnnals(dat=dat,k=length(prop),maxiter=200,st.count=20,n.ini=15,ini.nndsvd=TRUE,
seed=TRUE)
ClusterEntropy(ComputedClusters=fit$clusters, TrueClasses=true.cluster.assignment$cluster.id)

</code></pre>

<hr>
<h2 id='ClusterPurity'>
A function to measure cluster purity
</h2><span id='topic+ClusterPurity'></span>

<h3>Description</h3>

<p>Given the true clustering assignment for the subjects, this function calculates cluster purity index comparing with clustering assignment determined by integrative NMF algorithm. Higher value of cluster purity indicates better cluster predictive discrimination.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClusterPurity(ComputedClusters, TrueClasses)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ClusterPurity_+3A_computedclusters">ComputedClusters</code></td>
<td>

<p>Clustering assignment determined by integrative NMF algorithm
</p>
</td></tr>
<tr><td><code id="ClusterPurity_+3A_trueclasses">TrueClasses</code></td>
<td>

<p>True clustering assignment of the subjects
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cluster purity index value
</p>


<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley
</p>


<h3>References</h3>

<p>Kim Hyunsoo and Park Haesun (2007). Sparse non-negative matrix factorization via alternating non-negativity constrained least squares for microarray data analysis. Bioinformatics, 23: 1495-1502.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 2.5
sim.D &lt;- InterSIM(n.sample=100,cluster.sample.prop=prop,delta.methyl=effect,
delta.expr=effect,delta.protein=effect,p.DMP=0.25,p.DEG=NULL,p.DEP=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)

dat1 &lt;- sim.D$dat.methyl
dat2 &lt;- sim.D$dat.expr
dat3 &lt;- sim.D$dat.protein
true.cluster.assignment &lt;- sim.D$clustering.assignment

## Make all data positive by shifting to positive direction.
## Also rescale the datasets so that they are comparable.
if (!all(dat1&gt;=0)) dat1 &lt;- pmax(dat1 + abs(min(dat1)), .Machine$double.eps)
dat1 &lt;- dat1/max(dat1)
if (!all(dat2&gt;=0)) dat2 &lt;- pmax(dat2 + abs(min(dat2)), .Machine$double.eps)
dat2 &lt;- dat2/max(dat2)
if (!all(dat3&gt;=0)) dat3 &lt;- pmax(dat3 + abs(min(dat3)), .Machine$double.eps)
dat3 &lt;- dat3/max(dat3)

# The function nmf.mnnals requires the samples to be on rows and variables on columns.
dat &lt;- list(dat1,dat2,dat3)
fit &lt;- nmf.mnnals(dat=dat,k=length(prop),maxiter=200,st.count=20,n.ini=15,ini.nndsvd=TRUE,
seed=TRUE)
ClusterPurity(ComputedClusters=fit$clusters, TrueClasses=true.cluster.assignment$cluster.id)
</code></pre>

<hr>
<h2 id='ConsensusMatPlot'>
A function to create image plot of the consensus matrix
</h2><span id='topic+ConsensusMatPlot'></span>

<h3>Description</h3>

<p>Given the integrative NMF fit object, the function creates image plot of the consensus matrix ordered according to clusters groups. Cleaner block structure indicates stronger clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConsensusMatPlot(fit, rowLab = TRUE, colLab = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConsensusMatPlot_+3A_fit">fit</code></td>
<td>

<p>A nmf.mnnals fit object
</p>
</td></tr>
<tr><td><code id="ConsensusMatPlot_+3A_rowlab">rowLab</code></td>
<td>

<p>If true row label is displayed. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="ConsensusMatPlot_+3A_collab">colLab</code></td>
<td>

<p>If true column label is displayed. Default is TRUE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Image plot of the consensus matrix ordered according to cluster groups is returned.
</p>


<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley
</p>


<h3>References</h3>

<p>Brunnet J, Tamayo P Golub, T and Mesirov J (2004) Metagene and molecular pattern discovery using matrix factorization. PNAS, 101, 4164-4169
</p>
<p>Monti S, Tamayo P, Mesirov J and Golup T (2003). Consesus Clustering: A resampling based method for class discovery and visualization of gene expression microarray data. Machine Learning J, 52:91-118.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 2.5
sim.D &lt;- InterSIM(n.sample=100,cluster.sample.prop=prop,delta.methyl=effect,
delta.expr=effect,delta.protein=effect,p.DMP=0.25,p.DEG=NULL,p.DEP=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)
dat1 &lt;- sim.D$dat.methyl
dat2 &lt;- sim.D$dat.expr
dat3 &lt;- sim.D$dat.protein
true.cluster.assignment &lt;- sim.D$clustering.assignment

## Make all data positive by shifting to positive direction.
## Also rescale the datasets so that they are comparable.
if (!all(dat1&gt;=0)) dat1 &lt;- pmax(dat1 + abs(min(dat1)), .Machine$double.eps)
dat1 &lt;- dat1/max(dat1)
if (!all(dat2&gt;=0)) dat2 &lt;- pmax(dat2 + abs(min(dat2)), .Machine$double.eps)
dat2 &lt;- dat2/max(dat2)
if (!all(dat3&gt;=0)) dat3 &lt;- pmax(dat3 + abs(min(dat3)), .Machine$double.eps)
dat3 &lt;- dat3/max(dat3)

# The function nmf.mnnals requires the samples to be on rows and variables on columns.
dat &lt;- list(dat1,dat2,dat3)
fit &lt;- nmf.mnnals(dat=dat,k=length(prop),maxiter=200,st.count=20,n.ini=15,ini.nndsvd=TRUE,
seed=TRUE)
ConsensusMatPlot(fit,rowLab=TRUE,colLab=TRUE)
</code></pre>

<hr>
<h2 id='nmf.mnnals'>
Nonnegative Matrix Factorization of Multiple data using Nonnegative Alternating Least Square
</h2><span id='topic+nmf.mnnals'></span>

<h3>Description</h3>

<p>Given a single or multiple types of datasets (e.g. DNA methylation, mRNA expression, protein expression, DNA copy number) measured on same set of samples and pre-defined number of clusters, the function carries out clustering of the samples together with cluster membership assignment to the samples utilizing all the data set in a single comprehensive step.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nmf.mnnals(dat = dat, k = k, maxiter = 200, st.count = 20, n.ini = 30, ini.nndsvd = TRUE,
seed = TRUE,wt=if(is.list(dat)) rep(1,length(dat)) else 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nmf.mnnals_+3A_dat">dat</code></td>
<td>

<p>A single data or a list of multiple data matrices measured on same set of samples. For each data matrix in the list, samples should be on rows and genomic features should be on columns.
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_k">k</code></td>
<td>

<p>Number of clusters
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iteration, default is 200.
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_st.count">st.count</code></td>
<td>

<p>Count for stability in connectivity matrix, default is 20.
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_n.ini">n.ini</code></td>
<td>

<p>Number of initializations of the random matrices, default is 30.
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_ini.nndsvd">ini.nndsvd</code></td>
<td>

<p>Initialization of the Hi matrices using non negative double singular value decomposition (NNDSVD). If true, one of the initializations of algorithm will use NNDSVD. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_seed">seed</code></td>
<td>

<p>Random seed for initialization of algorithm, default is TRUE
</p>
</td></tr>
<tr><td><code id="nmf.mnnals_+3A_wt">wt</code></td>
<td>

<p>Weight, default is 1 for each data.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>consensus</code></td>
<td>
<p>Consensus matrix</p>
</td></tr>
<tr><td><code>W</code></td>
<td>
<p>Common basis matrix across the multiple data sets</p>
</td></tr>
<tr><td><code>H</code></td>
<td>
<p>List of data specific coefficient matrices.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>Matrix with five columns and number of rows equal to number of iterative steps required to converge the algorithm or number of maximum iteration. The five columns represent number of iterations, count for stability in connectivity matrix, stability indicator (1/0), absolute difference in reconstruction error between ith and (i-1)th iteration and value of the objective function respectively.</p>
</td></tr>
<tr><td><code>min.f.WH</code></td>
<td>
<p>Collection of values of objective function at convergence for each initialization of the algorithm.</p>
</td></tr>
<tr><td><code>clusters</code></td>
<td>
<p>Cluster membership assignment to samples.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley
</p>


<h3>References</h3>

<p>Chalise P and Fridley B (2017). Integrative clustering of multi-level 'omic data based on non-negative matrix factorization algorithm. PLOS ONE, 12(5), e0176278.
</p>
<p>Chalise P, Raghavan R and Fridley B (2016). InterSIM: Simulation tool for multiple integrative 'omic datasets. Computer Methods and Programs in Biomedicine, 128:69-74.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### Simulation of three interrelated dataset
#prop &lt;- c(0.65,0.35)
#prop &lt;- c(0.30,0.40,0.30)
prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 2.5

library(InterSIM)
sim.D &lt;- InterSIM(n.sample=100, cluster.sample.prop=prop, delta.methyl=effect,
delta.expr=effect, delta.protein=effect, p.DMP=0.25, p.DEG=NULL, p.DEP=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)
dat1 &lt;- sim.D$dat.methyl
dat2 &lt;- sim.D$dat.expr
dat3 &lt;- sim.D$dat.protein
true.cluster.assignment &lt;- sim.D$clustering.assignment

## Make all data positive by shifting to positive direction.
## Also rescale the datasets so that they are comparable.
if (!all(dat1&gt;=0)) dat1 &lt;- pmax(dat1 + abs(min(dat1)), .Machine$double.eps)
dat1 &lt;- dat1/max(dat1)
if (!all(dat2&gt;=0)) dat2 &lt;- pmax(dat2 + abs(min(dat2)), .Machine$double.eps)
dat2 &lt;- dat2/max(dat2)
if (!all(dat3&gt;=0)) dat3 &lt;- pmax(dat3 + abs(min(dat3)), .Machine$double.eps)
dat3 &lt;- dat3/max(dat3)
# The function nmf.mnnals requires the samples to be on rows and variables on columns.
dat1[1:5,1:5]
dat2[1:5,1:5]
dat3[1:5,1:5]
dat &lt;- list(dat1,dat2,dat3)

# Find optimum number of clusters for the data
#opt.k &lt;- nmf.opt.k(dat=dat, n.runs=5, n.fold=5, k.range=2:7, result=TRUE,
#make.plot=TRUE, progress=TRUE)
# Find clustering assignment for the samples
fit &lt;- nmf.mnnals(dat=dat, k=length(prop), maxiter=200, st.count=20, n.ini=15,
ini.nndsvd=TRUE, seed=TRUE)
table(fit$clusters)
fit$clusters[1:10]
</code></pre>

<hr>
<h2 id='nmf.opt.k'>
Selection of optimum number of clusters (k)
</h2><span id='topic+nmf.opt.k'></span>

<h3>Description</h3>

<p>Given a single or multiple types of datasets (e.g. DNA methylation, mRNA expression, protein expression, DNA copy number) measured on same set of samples, the function finds optimum number of clusters for the data or dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nmf.opt.k(dat = dat, n.runs = 30, n.fold = 5, k.range = 2:8, result = TRUE,
make.plot = TRUE, progress = TRUE, st.count = 10, maxiter = 100,
wt=if(is.list(dat)) rep(1,length(dat)) else 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nmf.opt.k_+3A_dat">dat</code></td>
<td>

<p>A single data or list of multiple types of data set measured on same set of samples. For each data matrix in the list, samples should be on rows and genomic features should be on columns.
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_n.runs">n.runs</code></td>
<td>

<p>Number of runs of algorithm in order to find optimum number of clusters, default is 30.
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_n.fold">n.fold</code></td>
<td>

<p>Number of folds for k-fold cross-validation, default is 5.
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_k.range">k.range</code></td>
<td>

<p>Search range for optimum number of clusters, default is 2:8
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_result">result</code></td>
<td>

<p>Logical, to display the result-matrix, default is TRUE.
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_make.plot">make.plot</code></td>
<td>

<p>Logical, to display the plot of cluster prediction index vs search range of clusters, default is TRUE
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_progress">progress</code></td>
<td>

<p>Logical, to display the progress (in percentage) of the algorithm, default is TRUE
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_st.count">st.count</code></td>
<td>

<p>Count for stability in connectivity matrix, default is 10.
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_maxiter">maxiter</code></td>
<td>

<p>Maximum number of iteration, default is 100.
</p>
</td></tr>
<tr><td><code id="nmf.opt.k_+3A_wt">wt</code></td>
<td>

<p>Weight, default is 1 for each data.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a matrix of cluster prediction index (CPI) values for each run (columns) over the search range of number of clusters (rows). The function also generates plot of CPI over the search range of number of clusters.
</p>


<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley
</p>


<h3>References</h3>

<p>Chalise P and Fridley B (2017). Integrative clustering of multi-level 'omic data based on non-negative matrix factorization algorithm. PLOS ONE, 12(5), e0176278.
</p>
<p>Chalise P, Raghavan R and Fridley B (2016). InterSIM: Simulation tool for multiple integrative 'omic datasets. Computer Methods and Programs in Biomedicine, 128:69-74.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### Simulation of three interrelated dataset
#prop &lt;- c(0.65,0.35)
#prop &lt;- c(0.30,0.40,0.30)
prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 2.5

library(InterSIM)
sim.D &lt;- InterSIM(n.sample=100, cluster.sample.prop=prop, delta.methyl=effect,
delta.expr=effect, delta.protein=effect, p.DMP=0.25, p.DEG=NULL, p.DEP=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)
dat1 &lt;- sim.D$dat.methyl
dat2 &lt;- sim.D$dat.expr
dat3 &lt;- sim.D$dat.protein
true.cluster.assignment &lt;- sim.D$clustering.assignment

## Make all data positive by shifting to positive direction.
## Also rescale the datasets so that they are comparable.
if (!all(dat1&gt;=0)) dat1 &lt;- pmax(dat1 + abs(min(dat1)), .Machine$double.eps)
dat1 &lt;- dat1/max(dat1)
if (!all(dat2&gt;=0)) dat2 &lt;- pmax(dat2 + abs(min(dat2)), .Machine$double.eps)
dat2 &lt;- dat2/max(dat2)
if (!all(dat3&gt;=0)) dat3 &lt;- pmax(dat3 + abs(min(dat3)), .Machine$double.eps)
dat3 &lt;- dat3/max(dat3)
# The function nmf.mnnals requires the samples to be on rows and variables on columns.
dat1[1:5,1:5]
dat2[1:5,1:5]
dat3[1:5,1:5]
dat &lt;- list(dat1,dat2,dat3)

# Find optimum number of clusters for the data
#opt.k &lt;- nmf.opt.k(dat=dat, n.runs=5, n.fold=5, k.range=2:7, result=TRUE,
#make.plot=TRUE,progress=TRUE)
</code></pre>

<hr>
<h2 id='SilhouettePlot'>
A function to plot silhouette width
</h2><span id='topic+SilhouettePlot'></span>

<h3>Description</h3>

<p>Given the integrative NMF fit object, the function creates silhouette width plot with different colors for different cluster groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SilhouettePlot(fit, cluster.col = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SilhouettePlot_+3A_fit">fit</code></td>
<td>

<p>A nmf.mnnals fit object
</p>
</td></tr>
<tr><td><code id="SilhouettePlot_+3A_cluster.col">cluster.col</code></td>
<td>

<p>Colors for the cluster groups. If NULL, default colors are used.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Silhouette width plot is returned together with mean silhouette width for each group, overall silhouette width and summary statistics.
</p>


<h3>Author(s)</h3>

<p>Prabhakar Chalise, Rama Raghavan, Brooke Fridley
</p>


<h3>References</h3>

<p>Rousseeue PJ (1987). Silhouettes: a graphical aid to the interpretation and validation of cluster analysis. Journal of Computational and Applied Mathematics, 20:53-65
</p>


<h3>Examples</h3>

<pre><code class='language-R'>prop &lt;- c(0.20,0.30,0.27,0.23)
effect &lt;- 2.5
sim.D &lt;- InterSIM(n.sample=100,cluster.sample.prop=prop,delta.methyl=effect,
delta.expr=effect,delta.protein=effect,p.DMP=0.25,p.DEG=NULL,p.DEP=NULL,
do.plot=FALSE, sample.cluster=TRUE, feature.cluster=TRUE)

dat1 &lt;- sim.D$dat.methyl
dat2 &lt;- sim.D$dat.expr
dat3 &lt;- sim.D$dat.protein
true.cluster.assignment &lt;- sim.D$clustering.assignment

## Make all data positive by shifting to positive direction.
## Also rescale the datasets so that they are comparable.
if (!all(dat1&gt;=0)) dat1 &lt;- pmax(dat1 + abs(min(dat1)), .Machine$double.eps)
dat1 &lt;- dat1/max(dat1)
if (!all(dat2&gt;=0)) dat2 &lt;- pmax(dat2 + abs(min(dat2)), .Machine$double.eps)
dat2 &lt;- dat2/max(dat2)
if (!all(dat3&gt;=0)) dat3 &lt;- pmax(dat3 + abs(min(dat3)), .Machine$double.eps)
dat3 &lt;- dat3/max(dat3)

# The function nmf.mnnals requires the samples to be on rows and variables on columns.
dat &lt;- list(dat1,dat2,dat3)
fit &lt;- nmf.mnnals(dat=dat,k=length(prop),maxiter=200,st.count=20,n.ini=15,ini.nndsvd=TRUE,
seed=TRUE)
SilhouettePlot(fit,cluster.col=NULL)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
