<!DOCTYPE html><html><head><title>Help for package ipADMIXTURE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ipADMIXTURE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#biclustFunc'><p>biclustFunc function</p></a></li>
<li><a href='#getPhyloTree'><p>getPhyloTree</p></a></li>
<li><a href='#human27pop_labels'><p>Labels of 27 human populations</p></a></li>
<li><a href='#human27pop_Qmat'><p>A list of Q matrices of 27 human populations</p></a></li>
<li><a href='#ipADMIXTURE'><p>Iterative Pruning Population Admixture Inference Framework (ipADMIXTURE)</p></a></li>
<li><a href='#plotAdmixClusters'><p>plotAdmixClusters</p></a></li>
<li><a href='#plotClusterLeaves'><p>plotClusterLeaves</p></a></li>
<li><a href='#printClustersFromLabels'><p>printClustersFromLabels</p></a></li>
<li><a href='#UD1_Qmat'><p>A list of Q matrices of simulation of 20 populations</p></a></li>
<li><a href='#UD1labels'><p>Labels of 20 simulation populations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Iterative Pruning Population Admixture Inference Framework</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A data clustering package based on admixture ratios (Q matrix) of population structure. The framework is based on iterative Pruning procedure that performs data clustering by splitting a given population into subclusters until meeting the condition of stopping criteria the same as ipPCA, iNJclust, and IPCAPS frameworks. The package also provides a function to retrieve phylogeny tree that construct a neighbor-joining tree based on a similar matrix between clusters. By given multiple Q matrices with varying a number of ancestors (K), the framework define a similar value between clusters i,j as a minimum number K* that makes majority of members of two clusters are in the different clusters. This K* reflexes a minimum number of ancestors we need to splitting cluster i,j into different clusters if we assign K* clusters based on maximum admixture ratio of individuals. The publication of this package is at Chainarong Amornbunchornvej, Pongsakorn Wangkumhang, and Sissades Tongsima (2020) &lt;<a href="https://doi.org/10.1101%2F2020.03.21.001206">doi:10.1101/2020.03.21.001206</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats,treemap,ape</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/DarkEyes/ipADMIXTURE">https://github.com/DarkEyes/ipADMIXTURE</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/DarkEyes/ipADMIXTURE/issues">https://github.com/DarkEyes/ipADMIXTURE/issues</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-03-24 07:22:41 UTC; Chainarong</td>
</tr>
<tr>
<td>Author:</td>
<td>Chainarong Amornbunchornvej
    <a href="https://orcid.org/0000-0003-3131-0370"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chainarong Amornbunchornvej &lt;grandca@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-03-26 11:00:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='biclustFunc'>biclustFunc function</h2><span id='topic+biclustFunc'></span>

<h3>Description</h3>

<p>biclustFunc is a binary clustering function using hierarchical clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biclustFunc(Qmat, admixRatioThs = 0.5, method = "average")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="biclustFunc_+3A_qmat">Qmat</code></td>
<td>
<p>is a Q matrix that contains admixture ratios of all individuals where the <code>Qmat[i,j]</code> represents the admixture ratio of ancestor j for individual i.</p>
</td></tr>
<tr><td><code id="biclustFunc_+3A_admixratioths">admixRatioThs</code></td>
<td>
<p>is a threshold to determine that if a cluster has <code>maxDiffAdmixRatio</code> lower than threshold, then the cluster is a homogeneous cluster.</p>
</td></tr>
<tr><td><code id="biclustFunc_+3A_method">method</code></td>
<td>
<p>is a method parameter of <code>hclust</code> object for hierarchical clustering analysis. The default is &quot;average&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns binary clustering results.
</p>
<table>
<tr><td><code>heteroFlag</code></td>
<td>
<p> is a flag that represents a status whether a given cluster is heterogeneous (having sub-clusters). It is TRUE if <code>maxDiffAdmixRatio &gt;= admixRatioThs</code>.</p>
</td></tr>
<tr><td><code>clusterInx</code></td>
<td>
<p>is a vector of clustering assignment where <code>indexClsVec[i]</code> is a cluster number of individual i.</p>
</td></tr>
<tr><td><code>meanDiffAdmixRatio</code></td>
<td>
<p>is a vector of magnitude-difference of admixture ratios.
It is calculated by splitting a given cluster into two sub-clusters. Then, we take the absolute on the difference between mean admixture ratios of sub-clusters. </p>
</td></tr>
<tr><td><code>Qmat1</code></td>
<td>
<p>is a Q matrix of sub-cluster #1 after splitting a given cluster into two sub-clusters that contains admixture ratios of all individuals where the <code>Qmat[i,j]</code> represents the admixture ratio of ancestor j for individual i.</p>
</td></tr>
<tr><td><code>Qmat2</code></td>
<td>
<p>is a Q matrix of sub-cluster #2 after splitting a given cluster into two sub-clusters that contains admixture ratios of all individuals where the <code>Qmat[i,j]</code> represents the admixture ratio of ancestor j for individual i.</p>
</td></tr>
<tr><td><code>maxDiffAdmixRatio</code></td>
<td>
<p>is a maximum of magnitude-difference of admixture ratios for a given cluster before splitting into two sub-clusters. </p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Running biclustFunc on Q matrix of 27 human population dataset where K = 12
obj&lt;-biclustFunc(Qmat=ipADMIXTURE::human27pop_Qmat[[11]], admixRatioThs =0.15)

</code></pre>

<hr>
<h2 id='getPhyloTree'>getPhyloTree</h2><span id='topic+getPhyloTree'></span>

<h3>Description</h3>

<p>getPhyloTree is function that reports a phylogenetic tree of clusters based on admixture analysis.
The phylogeny tree that construct a neighbor-joining tree based on a similar matrix between clusters.
By given multiple Q matrices with varying a number of ancestors (K), the framework define a similar value between clusters i,j as a minimum number <code>K</code> that makes majority of members of two clusters are in the different clusters.
This <code>K</code> reflexes a minimum number of ancestors we need to splitting cluster i,j into different clusters if we assign <code>K</code> clusters based on maximum admixture ratio of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPhyloTree(QmatList, indexClsVec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPhyloTree_+3A_qmatlist">QmatList</code></td>
<td>
<p>is list of Q matrix where <code>QmatList[[k]]</code> is a Q matrix with <code>k+1</code> ancestors.</p>
</td></tr>
<tr><td><code id="getPhyloTree_+3A_indexclsvec">indexClsVec</code></td>
<td>
<p>is a vector of clustering assignment where <code>indexClsVec[i]</code> is a cluster number of individual i.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns an object of nj tree as well as a matrix <code>minDiffAncestorClsMat</code> that is used as a similarity matrix.
</p>
<table>
<tr><td><code>tree</code></td>
<td>
<p>is an object of nj tree calculated by ape::nj() function on a dissimilarity version of <code>minDiffAncestorClsMat</code>. </p>
</td></tr>
<tr><td><code>minDiffAncestorClsMat</code></td>
<td>
<p>is a minimum-ancestor-number matrix where <code>minDiffAncestorClsMat[i,j]</code> is a minimum number of ancestors that make i and j to be different clusters while <code>minDiffAncestorClsMat[i,j]-1</code> makes majority of members from i and j belong to the same cluster.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># Running ipADMIXTURE on Q matrices (K=2-12) of 27 human population dataset.
h27pop_obj&lt;-ipADMIXTURE(Qmat=ipADMIXTURE::human27pop_Qmat[[11]], admixRatioThs =0.15)
out&lt;-ipADMIXTURE::getPhyloTree(ipADMIXTURE::human27pop_Qmat,h27pop_obj$indexClsVec)
plot(out$tree)

</code></pre>

<hr>
<h2 id='human27pop_labels'>Labels of 27 human populations</h2><span id='topic+human27pop_labels'></span>

<h3>Description</h3>

<p>Labels of 27 human populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>human27pop_labels
</code></pre>


<h3>Format</h3>

<p>Labels of 27 human populations. :
</p>

<dl>
<dt>human27pop_labels</dt><dd><p>It is a vector of labels of 544 individuals. There are 27 populations.</p>
</dd>
</dl>
<p>...
</p>

<hr>
<h2 id='human27pop_Qmat'>A list of Q matrices of 27 human populations</h2><span id='topic+human27pop_Qmat'></span>

<h3>Description</h3>

<p>A dataset containing admixture ratios of 544 individuals from 27 human populations where the number of ancestors ranges from 2 to 12.
This dataset was the result of running ADMIXTURE software developed by Zhou, H., et al. (2011). A quasi-Newton acceleration for high-dimensional optimization algorithms. Statistics and computing, 21(2), 261-273.
on the 27-human-population dataset published by Xing, J., Watkins, W. S. et al. (2009). Fine-scaled human genetic structure revealed by SNP microarrays. Genome research, 19(5), 815-825.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>human27pop_Qmat
</code></pre>


<h3>Format</h3>

<p>A list of Q matrices of 544 individuals from 27 human populations. There are 2-12 ancestors in the list.
</p>

<dl>
<dt>human27pop_Qmat</dt><dd><p> It is list of Q matrices that contains admixture ratios of 544 individuals from the 27 population human dataset.
<code>human27pop_Qmat[[k]][i,j]</code> is the admixture ratio of jth ancestor for ith individual in the (k+1)-ancestor Q matrix.</p>
</dd>
</dl>
<p>...
</p>

<hr>
<h2 id='ipADMIXTURE'>Iterative Pruning Population Admixture Inference Framework (ipADMIXTURE)</h2><span id='topic+ipADMIXTURE'></span>

<h3>Description</h3>

<p>A data clustering package based on admixture ratios (Q matrix) of population structure.
</p>
<p>The framework is based on iterative Pruning procedure that performs data clustering by splitting a given population into subclusters until meeting the condition of stopping criteria the same as ipPCA, iNJclust, and IPCAPS frameworks.
The package also provides a function to retrieve phylogeny tree that construct a neighbor-joining tree based on a similar matrix between clusters. By given multiple Q matrices with varying a number of ancestors (K),
the framework define a similar value between clusters i,j as a minimum number K that makes majority of members of two clusters are in the different clusters.
This K reflexes a minimum number of ancestors we need to splitting cluster i,j into different clusters if we assign K clusters based on maximum admixture ratio of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ipADMIXTURE(Qmat, admixRatioThs, method = "average")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ipADMIXTURE_+3A_qmat">Qmat</code></td>
<td>
<p>is a Q matrix that contains admixture ratios of all individuals where the <code>Qmat[i,j]</code> represents the admixture ratio of ancestor j for individual i.</p>
</td></tr>
<tr><td><code id="ipADMIXTURE_+3A_admixratioths">admixRatioThs</code></td>
<td>
<p>is a threshold to determine that if a cluster has <code>maxDiffAdmixRatio</code> lower than threshold, then the cluster is a homogeneous cluster.</p>
</td></tr>
<tr><td><code id="ipADMIXTURE_+3A_method">method</code></td>
<td>
<p>is a method parameter of <code>hclust</code> object for hierarchical clustering analysis. The default is &quot;average&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns clustering results in a form of an object of ipADMIXTURE class.
The object contains the following items.
</p>
<table>
<tr><td><code>indexClsVec</code></td>
<td>
<p>is a vector of clustering assignment where <code>indexClsVec[i]</code> is a cluster number of individual i.</p>
</td></tr>
<tr><td><code>homoClusters</code></td>
<td>
<p>is a list of cluster objects where each object contains member indices, cluster's <code>maxDiffAdmixRatio</code>, ID, etc. </p>
</td></tr>
<tr><td><code>maxDiffAdmixRatioVec</code></td>
<td>
<p>is a vector of <code>maxDiffAdmixRatio</code>s for all clusters.</p>
</td></tr>
<tr><td><code>Qmat</code></td>
<td>
<p>is a Q matrix that contains admixture ratios of all individuals where the <code>Qmat[i,j]</code> represents the admixture ratio of ancestor j for individual i.</p>
</td></tr>
<tr><td><code>admixRatioThs</code></td>
<td>
<p>is a threshold to determine that if a cluster has <code>maxDiffAdmixRatio</code> lower than threshold, then the cluster is a homogeneous cluster.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Chainarong Amornbunchornvej, <a href="mailto:chai@ieee.org">chai@ieee.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Running ipADMIXTURE on Q matrix of 27 human population dataset where K = 12
h27pop_obj&lt;-ipADMIXTURE(Qmat=ipADMIXTURE::human27pop_Qmat[[11]], admixRatioThs =0.15)

</code></pre>

<hr>
<h2 id='plotAdmixClusters'>plotAdmixClusters</h2><span id='topic+plotAdmixClusters'></span>

<h3>Description</h3>

<p>plotAdmixClusters is function that plots admixture ratios
where the x axis represents individuals with cluster labels and y axis represents admixture ratios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotAdmixClusters(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotAdmixClusters_+3A_obj">obj</code></td>
<td>
<p>is an object of ipADMIXTURE class.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>h27pop_obj&lt;-ipADMIXTURE(Qmat=ipADMIXTURE::human27pop_Qmat[[11]], admixRatioThs =0.15)
ipADMIXTURE::plotAdmixClusters(h27pop_obj)

</code></pre>

<hr>
<h2 id='plotClusterLeaves'>plotClusterLeaves</h2><span id='topic+plotClusterLeaves'></span>

<h3>Description</h3>

<p>plotClusterLeaves is function that plots clusters in a form of treemap plot.
Subsquares represent clusters. Each subsquare contains cluster label (ID),
number of members (N), and a maximum of manitude-difference of admixture ratios (md).
A size of each subsquare represents a ratio of member numbers compared to other clusters.
A color represents an md value of cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotClusterLeaves(obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotClusterLeaves_+3A_obj">obj</code></td>
<td>
<p>is an object of ipADMIXTURE class.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>h27pop_obj&lt;-ipADMIXTURE(Qmat=ipADMIXTURE::human27pop_Qmat[[11]], admixRatioThs =0.15)
ipADMIXTURE::plotClusterLeaves(h27pop_obj)

</code></pre>

<hr>
<h2 id='printClustersFromLabels'>printClustersFromLabels</h2><span id='topic+printClustersFromLabels'></span>

<h3>Description</h3>

<p>printClustersFromLabels is function that reports that clustering results in text mode.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printClustersFromLabels(obj, labels)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printClustersFromLabels_+3A_obj">obj</code></td>
<td>
<p>is an object of ipADMIXTURE class.</p>
</td></tr>
<tr><td><code id="printClustersFromLabels_+3A_labels">labels</code></td>
<td>
<p>is a vector of labels of all individuals.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>h27pop_obj&lt;-ipADMIXTURE(Qmat=ipADMIXTURE::human27pop_Qmat[[11]], admixRatioThs =0.15)
ipADMIXTURE::printClustersFromLabels(h27pop_obj,ipADMIXTURE::human27pop_labels)

</code></pre>

<hr>
<h2 id='UD1_Qmat'>A list of Q matrices of simulation of 20 populations</h2><span id='topic+UD1_Qmat'></span>

<h3>Description</h3>

<p>A dataset containing admixture ratios of 1200 individuals from 20 simulation populations where the number of ancestors ranges from 2 to 18.
This dataset was the result of running LEA library developed by Frichot, E., &amp; Fran√ßois, O. (2015). LEA: An R package for landscape and ecological association studies. Methods in Ecology and Evolution, 6(8), 925-929.
on the 20-simulation-population dataset published by Limpiti, T., et al. (2014). iNJclust: iterative neighbor-joining tree clustering framework for inferring population structure. IEEE/ACM transactions on computational biology and bioinformatics, 11(5), 903-914.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UD1_Qmat
</code></pre>


<h3>Format</h3>

<p>A list of Q matrices of 1200 individuals from 20 populations. There are Q matrices that have the number of ancestors ranges from from 2 to 18.
</p>

<dl>
<dt>UD1_Qmat</dt><dd><p> It is list of Q matrices that contains admixture ratios of 1200 individuals from the 20-population dataset.
<code>UD1_Qmat[[k]][i,j]</code> is the admixture ratio of jth ancestor for ith individual in the (k+1)-ancestor Q matrix.</p>
</dd>
</dl>
<p>...
</p>

<hr>
<h2 id='UD1labels'>Labels of 20 simulation populations</h2><span id='topic+UD1labels'></span>

<h3>Description</h3>

<p>Labels of 20 simulation populations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UD1labels
</code></pre>


<h3>Format</h3>

<p>Labels of 20 populations. :
</p>

<dl>
<dt>UD1labels</dt><dd><p>It is a vector of labels of 1200 individuals. There are 20 populations.</p>
</dd>
</dl>
<p>...
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
