<!DOCTYPE html><html><head><title>Help for package ips</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ips}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ips-package'><p>Interfaces to Phylogenetic Software</p></a></li>
<li><a href='#aliscore'><p>Masking of Sequence Alignments with ALISCORE</p></a></li>
<li><a href='#code.simple.gaps'><p>Simple Gap/Indel Coding</p></a></li>
<li><a href='#collapseUnsupportedEdges'><p>Collapse Unsupported Edges/Branches in a Phylogeny</p></a></li>
<li><a href='#del.miss'><p>Delete Missing Data from DNA Sequences</p></a></li>
<li><a href='#deleteEmptyCells'><p>Delete Spurious Rows and Columns from DNA Alignments</p></a></li>
<li><a href='#deleteGaps'><p>Remove Gap Positions From DNA Sequences</p></a></li>
<li><a href='#descendants'><p>Descendants of an Internal Node in a Phylogeny</p></a></li>
<li><a href='#DNAbin2index'><p>Conversion of DNAbin to Index</p></a></li>
<li><a href='#eoi'><p>Identification of Stem-Lineage-Edges and MRCAs</p></a></li>
<li><a href='#fixNodes'>
<p>Standard Node Numbering in Phylo Objects</p></a></li>
<li><a href='#forceEqualTipHeights'><p>Equal Tip Heights</p></a></li>
<li><a href='#gblocks'><p>Masking of Sequence Alignments with GBLOCKS</p></a></li>
<li><a href='#index2DNAbin'><p>Conversion of Index to DNAbin</p></a></li>
<li><a href='#ips-internal'><p>Internal IPS Functions</p></a></li>
<li><a href='#ips.16S'><p>Bark Beetle 16S Sequences</p></a></li>
<li><a href='#ips.28S'>
<p>Bark Beetle 28S Sequences</p></a></li>
<li><a href='#ips.cox1'>
<p>Bark Beetle COX1 Sequences</p></a></li>
<li><a href='#ips.tree'><p>Ips Phylogeny</p></a></li>
<li><a href='#mafft'><p>Sequence Alignment with MAFFT</p></a></li>
<li><a href='#mafft.merge'><p>Profile Alignment with MAFFT</p></a></li>
<li><a href='#mrbayes'><p>Bayesian MCMC Tree Search with MrBayes</p></a></li>
<li><a href='#mrbayes.lset'><p>Model Settings for MrBayes</p></a></li>
<li><a href='#mrbayes.mcmc'><p>MCMC Settings for MrBayes</p></a></li>
<li><a href='#mrbayes.prset'><p>Set Priors for MrBayes</p></a></li>
<li><a href='#multistate'><p>MULTISTATE</p></a></li>
<li><a href='#neighboringPairs'><p>Neighboring Nodes in a Minimum Spanning Tree</p></a></li>
<li><a href='#ntip'><p>Numbers of Tips of (Sub)trees</p></a></li>
<li><a href='#partitionfinder'><p>PartitionFinder</p></a></li>
<li><a href='#pathd8'><p>PATHd8</p></a></li>
<li><a href='#phylo2mafft'><p>Convert Trees for MAFFT</p></a></li>
<li><a href='#phylo2mst'><p>Conversion from PHYLO to MST Object</p></a></li>
<li><a href='#pis'><p>Number of Potentially-Informative Sites</p></a></li>
<li><a href='#prank'><p>PRANK</p></a></li>
<li><a href='#raxml'><p>Maximum Likelihood Tree Estimation with RAxML</p></a></li>
<li><a href='#raxml.partitions'><p>Partition scheme for RAxML</p></a></li>
<li><a href='#rbeauti'><p>XML Input Files for BEAST</p></a></li>
<li><a href='#read'><p>Reading Sequence Files</p></a></li>
<li><a href='#read.beast'><p>Read Bayesian Trees</p></a></li>
<li><a href='#read.beast.table'>
<p>Extract node data from BEAST chronogram</p></a></li>
<li><a href='#sister'><p>Identification of Sister Nodes and Clades</p></a></li>
<li><a href='#splitIntoClades'><p>Find Monophyletic Subsets in Species Lists</p></a></li>
<li><a href='#terminalSisters'><p>Find Pairs of Sister Species</p></a></li>
<li><a href='#tipHeights'>
<p>Tip Heights in a Phylogenetic Tree</p></a></li>
<li><a href='#traitRate'><p>Trait-Dependent Shifts in Molecular Rate</p></a></li>
<li><a href='#trimEnds'><p>Trim Alignment Ends</p></a></li>
<li><a href='#unlistFirstLevel'><p>Unlist To First Level Only</p></a></li>
<li><a href='#write.fas'><p>Write DNA Sequences to File</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.0.11</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-06-20</td>
</tr>
<tr>
<td>Title:</td>
<td>Interfaces to Phylogenetic Software in R</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0), ape</td>
</tr>
<tr>
<td>Imports:</td>
<td>phangorn, plyr, seqinr, XML</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions that wrap popular phylogenetic software for sequence
    alignment, masking of sequence alignments, and estimation of phylogenies and
    ancestral character states.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-07-04 15:51:55 UTC; heibl</td>
</tr>
<tr>
<td>Author:</td>
<td>Christoph Heibl [aut, cre],
  Natalie Cusimano [aut],
  Franz-Sebastian Krah [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Christoph Heibl &lt;christoph.heibl@gmx.net&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-07-04 16:30:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ips-package'>Interfaces to Phylogenetic Software</h2><span id='topic+ips-package'></span><span id='topic+ips'></span>

<h3>Description</h3>

<p>This package presents a set of functions that were formerly included in the <em>phyloch</em> package and which wrap popular phylogenetic software for sequence alignment, masking of sequence alignments, and estimation of phylogenies and ancestral character states.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
 Package: </td><td style="text-align: left;"> ips</td>
</tr>
<tr>
 <td style="text-align: left;">
 Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 0.0.11</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-06-20</td>
</tr>
<tr>
 <td style="text-align: left;">
 License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>There are several functions for reading and writing DNA sequences in FASTA, PHYLIP, and NEXUS format: <code><a href="#topic+read.fas">read.fas</a></code>, <code><a href="#topic+read.phy">read.phy</a></code>, <code><a href="#topic+read.nex">read.nex</a></code>, <code><a href="#topic+write.fas">write.fas</a></code>, <code><a href="#topic+write.phy">write.phy</a></code>, and <code><a href="#topic+write.nex">write.nex</a></code>.
Some functions are available for integrating BEAST with R. XML input files for BEAST can be generated with <code><a href="#topic+rbeauti">rbeauti</a></code>. Two functions are designed to read TreeAnnotator output: <code><a href="#topic+read.beast">read.beast</a></code> will render an object of class <code>phylo</code> with additional node statistics appended as list elements. These additional node statistics will be lost be the subsequent use of <code><a href="ape.html#topic+ladderize">ladderize</a></code> or <code><a href="ape.html#topic+rotate">rotate</a></code> (or similar functions that change the ordering of internal nodes).<code><a href="#topic+read.beast.table">read.beast.table</a></code> also parses the TreeAnnotator output, but returns a matrix of node statistics. 
This package itself does not implement techniques for phylogenetic analyses, but provides a series of wrappers for commonly used software packages. Sequence alignment can be done with the <code><a href="#topic+mafft">mafft</a></code> and <code><a href="#topic+prank">prank</a></code>; cleaning of sequences with <code><a href="#topic+gblocks">gblocks</a></code> and <code><a href="#topic+aliscore">aliscore</a></code>. The function <code><a href="#topic+raxml">raxml</a></code> and <code><a href="#topic+mrbayes">mrbayes</a></code> are intended for phylogenetic tree search. Running <code><a href="#topic+mrbayes">mrbayes</a></code> with argument <code>run = FALSE</code> can be used to create MrBayes-executable NEXUS files. Finally,  wrappers is provided for <code>Multistate</code> in the <code>BayesTraits</code> package (see <code><a href="#topic+multistateML">multistateML</a></code> and <code><a href="#topic+multistateMCMC">multistateMCMC</a></code>).
Several plotting functions (<code>HPDbars</code>, <code>clade.bars</code>, <code>box.clades</code>, <code>box.tips</code>, <code>tip.color</code>, <code>edge.color</code> have been moved to the <b>viper</b> package.
</p>


<h3>Author(s)</h3>

<p>Natalie Cusimano, Christoph Heibl, Franz-Sebastian Krah,
Maintainer: Christoph Heibl (<a href="mailto:christoph.heibl@gmx.net">christoph.heibl@gmx.net</a>)
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ape-package">ape</a></code>
</p>

<hr>
<h2 id='aliscore'>Masking of Sequence Alignments with ALISCORE</h2><span id='topic+aliscore'></span>

<h3>Description</h3>

<p>Provides a interface to <b>Aliscore</b>, in order to remove
problematic regions of a DNA sequence alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>aliscore(x, gaps = "5state", w = 6, r, t, l, s, o, exec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="aliscore_+3A_x">x</code></td>
<td>
<p>DNA sequences of class <code>DNAbin</code>.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_gaps">gaps</code></td>
<td>
<p>A vector of mode <code>"character"</code> indicating how gaps shall be
treated: as <code>"5state"</code> or as <code>"ambiguous"</code>.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_w">w</code></td>
<td>
<p>An integer giving the size of the sliding window.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_r">r</code></td>
<td>
<p>An integer giving the number of random pairwise sequence
comparisons; defaults to <code>4 * N</code>.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_t">t</code></td>
<td>
<p><em>Not yet implemented</em>.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_l">l</code></td>
<td>
<p><em>Not yet implemented</em>.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_s">s</code></td>
<td>
<p><em>Not yet implemented</em>.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_o">o</code></td>
<td>
<p>A vector of mode <code>"character"</code> containing outgroup taxon names.</p>
</td></tr>
<tr><td><code id="aliscore_+3A_exec">exec</code></td>
<td>
<p>A character string, giving the path to the Aliscore script.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>matrix</code> of class <code>"DNAbin"</code>.
</p>


<h3>Note</h3>

<p>This function was developed with ALISCORE version 2.
</p>


<h3>References</h3>

<p>Misof, B. and K. Misof. 2009. A Monte Carlo approach successfully identifies
randomness in multiple sequence alignments: a more objective means of data
exclusion. <em>Syst. Biol.</em> <b>58</b>: 21&ndash;34.
</p>
<p>Kueck, P., K. Meusemann, J. Dambach, B. Thormann, B.M. von Reumont, J.W.
Waegele and B. Misof. 2010. Parametric and non-parametric masking of
randomness in sequence alignments can be improved and leads to better
resolved trees. <em>Frontiers in Zoology</em> <b>7</b>: 10.
</p>
<p><b>Aliscore website</b>: <a href="https://www.zfmk.de/en/research/research-centres-and-groups/aliscore">https://www.zfmk.de/en/research/research-centres-and-groups/aliscore</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mafft">mafft</a></code> and <code><a href="#topic+prank">prank</a></code> for multiple sequence
alignment; <code><a href="#topic+gblocks">gblocks</a></code> for another alignment masking algorithm.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.28S)
## Not run: aliscore(ips.28S)
</code></pre>

<hr>
<h2 id='code.simple.gaps'>Simple Gap/Indel Coding</h2><span id='topic+code.simple.gaps'></span>

<h3>Description</h3>

<p><code>code.simple.gaps</code> takes an aligned DNA sequence matrix and codes the simple gaps, i.e. gaps that do not overlap with other gaps. The gapped positions are excluded from the matrix and the coded gap characters are appended to the matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>code.simple.gaps(x, append = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="code.simple.gaps_+3A_x">x</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="code.simple.gaps_+3A_append">append</code></td>
<td>
<p>Logical.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a>.</code>
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>Simmons, M.P. &amp; H. Ochoterena. 2000. Gaps as characters in sequence-based phylogenetic analyses. <em>Systematic Biology</em> <b>49(2)</b>: 369&ndash;381.	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+deleteGaps">deleteGaps</a></code>, <code><a href="#topic+deleteEmptyCells">deleteEmptyCells</a></code>, <code><a href="#topic+trimEnds">trimEnds</a></code>
</p>

<hr>
<h2 id='collapseUnsupportedEdges'>Collapse Unsupported Edges/Branches in a Phylogeny</h2><span id='topic+collapseUnsupportedEdges'></span>

<h3>Description</h3>

<p>Given a set of node support values (e.g., bootstrap 
proportions, posterior probabilities) and a certain threshold, 
all edges receiving less support than the threshold will be collapsed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>collapseUnsupportedEdges(phy, value = "node.label", cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="collapseUnsupportedEdges_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="collapseUnsupportedEdges_+3A_value">value</code></td>
<td>
<p>A character string giving the name of the list element 
that contains the support values; default is <code>"node.label"</code>.</p>
</td></tr>
<tr><td><code id="collapseUnsupportedEdges_+3A_cutoff">cutoff</code></td>
<td>
<p>A numeric value giving the threshold below which edges 
will be collapsed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## phylogeny of bark beetles
data(ips.tree)
## non-parametric bootstrap proportions (BP)
ips.tree$node.label
## collapse clades with &lt; 70 BP
tr &lt;- collapseUnsupportedEdges(ips.tree, "node.label", 70)
## show new topology
par(mfrow = c(1, 2))
plot(ips.tree, no.margin = TRUE)
nodelabels(ips.tree$node.label, cex = .5, frame = "n", adj = c(0, .5))
plot(tr, no.margin = TRUE)
</code></pre>

<hr>
<h2 id='del.miss'>Delete Missing Data from DNA Sequences</h2><span id='topic+del.miss'></span>

<h3>Description</h3>

<p>Remove gaps (&quot;-&quot;) and/or missing and ambiguous data (&quot;N&quot;, &quot;?&quot;) 
from a sample of DNA sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>del.miss(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="del.miss_+3A_x">x</code></td>
<td>
<p>A matrix, a list, or a vector of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>
containing the DNA sequences.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list or a vector of class <code>DNAbin</code>.
</p>

<hr>
<h2 id='deleteEmptyCells'>Delete Spurious Rows and Columns from DNA Alignments</h2><span id='topic+deleteEmptyCells'></span>

<h3>Description</h3>

<p>After subsetting (see e.g. <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>), DNA sequence
alignments can contain rows and columns that consist entirely of missing
and/or ambiguous character states. <code>deleteEmptyCells</code> will delete all
such rows (taxa) and columns (characters) from a DNA sequence alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deleteEmptyCells(DNAbin, margin = c(1, 2), nset = c("-", "n", "?"),
  quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deleteEmptyCells_+3A_dnabin">DNAbin</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="deleteEmptyCells_+3A_margin">margin</code></td>
<td>
<p>A vector giving the subscripts the function will be applied
over: <code>1</code> indicates rows, <code>2</code> indicates columns, and
<code>c(1, 2)</code> indicates rows and columns.</p>
</td></tr>
<tr><td><code id="deleteEmptyCells_+3A_nset">nset</code></td>
<td>
<p>A vector of mode character; rows or columns that consist 
<strong>only</strong> of the characters given in <code>nset</code> will be deleted from 
the alignment. Allowed are <code>"-"</code>, <code>"?"</code>,<code>"n"</code>, <code>"b"</code>, 
<code>"d"</code>,<code>"h"</code>, <code>"v"</code>, <code>"r"</code>,<code>"y"</code>, <code>"s"</code>, 
<code>"w"</code>,<code>"k"</code>, and <code>"m"</code>.</p>
</td></tr>
<tr><td><code id="deleteEmptyCells_+3A_quiet">quiet</code></td>
<td>
<p>Logical: if set to <code>TRUE</code>, screen output will be suppressed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For faster execution, <code>deleteEmptyCells</code> handles sequences in 
<span class="pkg">ape</span>'s bit-level coding scheme.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.
</p>


<h3>References</h3>

<p>Cornish-Bowden, A. 1984. Nomenclature for incompletely 
specified bases in nucleic acid sequences: recommendations 1984. 
<em>Nucleic Acids Res.</em> <strong>13</strong>: 3021â€“3030.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trimEnds">trimEnds</a></code>, <code><a href="#topic+deleteGaps">deleteGaps</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  # COX1 sequences of bark beetles
  data(ips.cox1)
  # introduce completely ambiguous rows and colums
  x &lt;- as.character(ips.cox1[1:6, 1:60])
  x[3, ] &lt;- rep("n", 60)
  x[, 20:24] &lt;- rep("-", 6)
  x &lt;- as.DNAbin(x)
  image(x)
  # delete those rows and colums
  x &lt;- deleteEmptyCells(x)
  image(x)
</code></pre>

<hr>
<h2 id='deleteGaps'>Remove Gap Positions From DNA Sequences</h2><span id='topic+deleteGaps'></span>

<h3>Description</h3>

<p>Remove indel positions (or gaps) from a DNA sequence 
alignment. For faster execution, <code>deleteGaps</code> handles sequences 
in <span class="pkg">ape</span>'s bit-level coding scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deleteGaps(x, gap.max = nrow(x) - 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deleteGaps_+3A_x">x</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="deleteGaps_+3A_gap.max">gap.max</code></td>
<td>
<p>An integer, which gives the maximum number of
gap characters (&quot;-&quot;) that will be tolerated at any given alignment 
position (column). Only values between <code>0</code> and <code>nrow(x) - 4</code>
make sense phylogenetically.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default, <code>nmax = nrow(x) - 4</code>, removes all those 
positions from the alignment, which contain at least four non-gap 
characters, which is the minimum number of sequences needed to 
produce a non-trivial unrooted topology. All gaps will be excluded 
by selecting <code>nmax = 0</code> and half of all gaps with <code>nmax = 
nrow(x) / 2</code>.  
</p>
<p>In contrast, <code><a href="ape.html#topic+del.gaps">del.gaps</a></code> removes all gap characters 
from the alignment, so most probably the result will not be a set of 
sequences of equal length and the matrix will be coerced to a list.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a>.</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+code.simple.gaps">code.simple.gaps</a></code> for coding of simple gaps, 
<code><a href="ape.html#topic+del.gaps">del.gaps</a></code> for removal of all gap symbols from an 
alignment, <code><a href="#topic+gblocks">gblocks</a></code> and <code><a href="#topic+aliscore">aliscore</a></code> for more 
sophisticated  methods of cleaning/masking alignments.
</p>

<hr>
<h2 id='descendants'>Descendants of an Internal Node in a Phylogeny</h2><span id='topic+descendants'></span>

<h3>Description</h3>

<p>For any given internal node of a phylogeny, the function returns a vector containing the node numbers descending from that node.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>descendants(phy, node, type = "t", ignore.tip = TRUE, labels = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="descendants_+3A_phy">phy</code></td>
<td>

<p>an object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.
</p>
</td></tr>
<tr><td><code id="descendants_+3A_node">node</code></td>
<td>

<p>an integer giving the number of the internal node.
</p>
</td></tr>
<tr><td><code id="descendants_+3A_type">type</code></td>
<td>

<p>a character string, may be <code>"daughter"</code>, <code>"internal"</code>, <code>"terminal"</code>, <code>"all"</code>, or any unambiguous abbreviation of these.
</p>
</td></tr>
<tr><td><code id="descendants_+3A_ignore.tip">ignore.tip</code></td>
<td>

<p>logical, if <code>ignore.tip = FALSE</code>, the function will issue an error when <code>node</code> is not internal, otherwise the number of the corresponding terminal node will be returned.
</p>
</td></tr>
<tr><td><code id="descendants_+3A_labels">labels</code></td>
<td>

<p>logical, determines if node labels are returned instead of node number, currently ignored unless <code>type = "t"</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing terminal node numbers or tip labels.
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>See Also</h3>

<p><code><a href="#topic+sister">sister</a></code>, <code><a href="#topic+noi">noi</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'> # generate a random tree with 12 terminal and 11 internal nodes:
 tree &lt;- rtree(12)
 
 # get the descendants of internal node 15:
 x &lt;- descendants(tree, 15)
</code></pre>

<hr>
<h2 id='DNAbin2index'>Conversion of DNAbin to Index</h2><span id='topic+DNAbin2index'></span>

<h3>Description</h3>

<p>Extract the indices of non-empty positions in a sample of DNA
sequences to
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DNAbin2index(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DNAbin2index_+3A_x">x</code></td>
<td>
<p>A matrix of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+index2DNAbin">index2DNAbin</a></code>
</p>

<hr>
<h2 id='eoi'>Identification of Stem-Lineage-Edges and MRCAs</h2><span id='topic+eoi'></span><span id='topic+noi'></span><span id='topic+oi'></span>

<h3>Description</h3>

<p><code>noi</code> (<strong>n</strong>ode <strong>o</strong>f <strong>i</strong>nterest)
identifies the most recent common ancestor (MRCA) and <code>eoi</code>
(<strong>e</strong>dge <strong>o</strong>f <strong>i</strong>nterest) its subtending stem-lineage
edge of one or more sets of taxa/tips.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eoi(phy, node, group, regex = FALSE, stem = FALSE,
  monophyletic = FALSE)

noi(phy, group, regex = FALSE, stem = FALSE, monophyletic = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eoi_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="eoi_+3A_node">node</code></td>
<td>
<p>A vector of mode <code>"numeric"</code> giving the nodes numbers of the nodes whose subtending stem-lineages will be identified.</p>
</td></tr>
<tr><td><code id="eoi_+3A_group">group</code></td>
<td>
<p>A vector or list of vectors of mode <code>character</code> specifying
the taxon set(s). Will be ignored if <code>node</code> is given.</p>
</td></tr>
<tr><td><code id="eoi_+3A_regex">regex</code></td>
<td>
<p>A logical, if <code>regex = TRUE</code>, taxon sets are matched to the
tip labels as regular expressions of the form <code>"taxon1|taxon2"</code>;
otherwise strings will be matched exactly (see <code><a href="base.html#topic+which">which</a></code>).</p>
</td></tr>
<tr><td><code id="eoi_+3A_stem">stem</code></td>
<td>
<p>Logical, ...</p>
</td></tr>
<tr><td><code id="eoi_+3A_monophyletic">monophyletic</code></td>
<td>
<p>Logical, ...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of mode <code>"numeric"</code> containing node numbers.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+mrca">mrca</a></code>; <code><a href="#topic+descendants">descendants</a></code> for the contrary
operation to <code>noi</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># molecular phylogeny of Ips bark beetles
# ---------------------------------------
data(ips.tree)
ips.tree &lt;- ladderize(ips.tree)
ips.tree &lt;- fixNodes(ips.tree)
clade1 &lt;- descendants(ips.tree, 44, labels = TRUE)
mrca &lt;- noi(ips.tree, clade1)
stem_lineage &lt;- eoi(ips.tree, mrca)
ecol &lt;- rep("black", Nedge(ips.tree))
ecol[stem_lineage] &lt;- "red"
plot(ips.tree, no.margin = TRUE, edge.color = ecol)
nodelabels(node = mrca, pch = 22, col = "blue")
#gen &lt;- sapply(viperidae$tip.label, function(x) unlist(strsplit(x, "_"))[1])
#tax &lt;- data.frame(genus = gen, species = viperidae$tip.label, row.names = NULL)

# group can be a list
# -------------------
#myclades &lt;- split(tax$species, tax$genus)
#nds &lt;- noi(viperidae, myclades)
#plot(viperidae)
#nodeInfo(nds)

# group might contain tip numbers
# -------------------------------
#group &lt;- list(c(17, 22), c(13, 1))
#plot(viperidae)
#append2tips(phy, tips = unlist(group), pch = 19)
#nds &lt;- noi(viperidae, myclades)
#nodeInfo(nds)

# the 'group' argument can also take regular expressions
# ------------------------------------------------------
#rex &lt;- "aspis"
#node &lt;- noi(viperidae, rex, regex = TRUE)
#plot.phylo(viperidae, tip.color = 0, edge.color = 0)
#box.clades(viperidae, nodes = node, col = "#D2A6A7", align = "all")
#plot.phylo.upon(viperidae)
#nodelabels(node = node, pch = 21, cex = 1.2, col = "red", bg = "#D2A6A7")

# if the 'group' argument is a list of elements of length 2,
# n = length(group) nodes of interest will be returned
# ----------------------------------------------------
#group &lt;- list(
#    c("Vipera_berus", "Vipera_ursinii"),
#    c("Vipera_aspis_ssp._aspis", "Vipera_latastei"),
#    c("Vipera_ammodytes_ssp._ammodytes",
#        "Vipera_ammodytes_ssp._montandoni"),
#    c("Macrovipera_lebetina", "Vipera_wagneri")
#)
#clades &lt;- noi(viperidae, group)
#plot.phylo(viperidae, tip.color = 0, edge.color = 0)
#box.clades(viperidae, nodes = clades, col = c("#FFFFA5", "#D2A6A7",
#    "#A7D2A5", "#A5A6D2"), align = "all")
#plot.phylo.upon(viperidae)
</code></pre>

<hr>
<h2 id='fixNodes'>
Standard Node Numbering in Phylo Objects
</h2><span id='topic+fixNodes'></span>

<h3>Description</h3>

<p>The function (re-)establishes the standard numbering of terminal and internal nodes in phylogenies represented as objects of class <code><a href="ape.html#topic+phylo">phylo</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fixNodes(phy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fixNodes_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>. 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When reading phylogenetic trees from a NEXUS file that contains a <code>translate</code> section, it can happen that the terminal nodes (tips, leaves) of the corresponding <code>phylo</code> object are not numbered consecutively, which can be a problem in some downstream applications. You can use <code>fixNodes</code> to get the correct order of terminal node numbers.
</p>
<p><code>fixNodes</code> is also intended to re-establish the standard numbering of internal nodes and reorder all node value elements (e.g. node.label, posterior, ...) if a <code><a href="ape.html#topic+phylo">phylo</a></code> object has been modified by either <code><a href="ape.html#topic+root">root</a></code>, <code><a href="ape.html#topic+ladderize">ladderize</a></code>, or <code><a href="ape.html#topic+rotate">rotate</a></code>.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>


<h3>Note</h3>

<p><code>fixNodes</code> has been completely rewritten for <b>ips</b> version 1.0-0. It should now run absolutely stable and is much quicker. Nevertheless, I recommend checking carefully the results of <code>fixNodes</code>, until the function has been tested by a number of users. Then this comment will be removed.
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+read.tree">read.tree</a></code>, 
<code><a href="ape.html#topic+read.nexus">read.nexus</a></code>, <code><a href="#topic+read.beast">read.beast</a></code> 
for reading trees in NEWICK and NEXUS format; <code><a href="ape.html#topic+ladderize">ladderize</a></code> and <code><a href="ape.html#topic+rotate">rotate</a></code> for tree manipulation; <code>node.support</code> for plotting node support values has been moved to package <b>viper</b>. 
</p>

<hr>
<h2 id='forceEqualTipHeights'>Equal Tip Heights</h2><span id='topic+forceEqualTipHeights'></span>

<h3>Description</h3>

<p>Modify terminal edge lengths to create &quot;exactly&quot; (see Details)
equal tip heights (sum of edge lengths from root to tip)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>forceEqualTipHeights(phy, baseline = "mean")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forceEqualTipHeights_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="forceEqualTipHeights_+3A_baseline">baseline</code></td>
<td>
<p>A character string giving a function to calculate the
baseline tip height, e.g. <code>"min"</code>, <code>"max"</code> or <code>"mean"</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>What is &quot;exactly&quot; equal depends on the precision of the system
(<code><a href="base.html#topic+.Machine">.Machine</a></code>); in any case the resulting phylogeny will pass
<code><a href="ape.html#topic+is.ultrametric">is.ultrametric</a></code> with default arguments.
</p>


<h3>Value</h3>

<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code> with changed terminal
edge lengths.
</p>


<h3>Note</h3>

<p><code>forceEqualTipHeights</code> is only intended to correct small rounding
errors in edge lengths, not to make an additive phylogeny ultrametric. For
the latter, see e.g. <code><a href="ape.html#topic+chronos">chronos</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tipHeights">tipHeights</a></code>
</p>

<hr>
<h2 id='gblocks'>Masking of Sequence Alignments with GBLOCKS</h2><span id='topic+gblocks'></span>

<h3>Description</h3>

<p>Provides a wrapper to Gblocks, a computer program written in
ANSI C language that eliminates poorly aligned positions and divergent
regions of an alignment of DNA or protein sequences. Gblocks selects
conserved blocks from a multiple alignment according to a set of features
of the alignment positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gblocks(x, b1 = 0.5, b2 = b1, b3 = ncol(x), b4 = 2, b5 = "a",
  target = "alignment", exec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gblocks_+3A_x">x</code></td>
<td>
<p>A matrix of DNA sequences of classes <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="gblocks_+3A_b1">b1</code></td>
<td>
<p>A real number, the <b>minimum number of sequences for a
conserved position</b> given as a fraction. Values between 0.5 and 1.0 are
allowed. <em>Larger</em> values will <em>decrease</em> the number of selected
positions, i.e. are more <em>conservative</em>. Defaults to 0.5</p>
</td></tr>
<tr><td><code id="gblocks_+3A_b2">b2</code></td>
<td>
<p>A real number, the <b>minimum number of sequences for a flank
position</b> given as a fraction. Values must be equal or larger than
<code>b1</code>. <em>Larger</em> values will <em>decrease</em> the number of selected
positions, i.e. are <em>more conservative</em>. Defaults to 0.5</p>
</td></tr>
<tr><td><code id="gblocks_+3A_b3">b3</code></td>
<td>
<p>An integer, the <b>maximum number of contiguous nonconserved
positions</b>; any integer is allowed. <em>Larger</em> values will
<em>increase</em> the number of selected position, i.e. are <em>less
conservative</em>. Defaults to the number of positions in the alignment.</p>
</td></tr>
<tr><td><code id="gblocks_+3A_b4">b4</code></td>
<td>
<p>An integer, the <b>minimum length of a block</b>, any integer equal
to or bigger than 2 is allowed. <em>Larger</em> values will <em>decrease</em>
the number of selected positions, i.e. are more conservative. Defaults to
2.</p>
</td></tr>
<tr><td><code id="gblocks_+3A_b5">b5</code></td>
<td>
<p>A character string indicating the <b>treatment of gap
positions</b>. Three choices are possible. 1. <code>"n"</code>: <em>No</em> gap
positions are allowed in the final alignment. All positions with a single
gap or more are treated as a gap position for the block selection
procedure, and they and the adjacent nonconserved positions are eliminated.
2. <code>"h"</code>: Only positions where <em>50% or more</em> of the sequences
have a gap are treated as a gap position. Thus, positions with a gap in
less than 50% of the sequences can be selected in the final alignment if
they are within an appropriate block. 3. <code>"a"</code>: <em>All</em> gap
positions can be selected. Positions with gaps are not treated differently
from other positions (default).</p>
</td></tr>
<tr><td><code id="gblocks_+3A_target">target</code></td>
<td>
<p>A vector of mode <code>"character"</code> giving the output format:
<code>"alignment"</code> will return the alignment with only the selected
positions, <code>"index"</code> will return the indices of the selected position,
and <code>"score"</code> will provide a score for every position in the original
alignment (0 for excluded, 1 for included).</p>
</td></tr>
<tr><td><code id="gblocks_+3A_exec">exec</code></td>
<td>
<p>A character string indicating the path to the GBLOCKS executable.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Explanation of the routine taken from the Online Documentation:
First, the degree of conservation of every positions of the multiple
alignment is evaluated and classified as <em>nonconserved</em>,
<em>conserved</em>, or <em>highly conserved</em>. All stretches of contiguous
nonconserved positions bigger than a certain value (<b>b3</b>) are
rejected. In such stretches, alignments are normally ambiguous and, even
when in some cases a unique alignment could be given, multiple hidden
substitutions make them inadequate for phylogenetic analysis. In the
remaining blocks, flanks are examined and positions are removed until
blocks are surrounded by highly conserved positions at both flanks. This
way, selected blocks are anchored by positions that can be aligned with
high confidence. Then, all gap positions -that can be defined in three
different ways (<b>b5</b>)- are removed. Furthermore, nonconserved
positions adjacent to a gap position are also eliminated until a conserved
position is reached, because regions adjacent to a gap are the most
difficult to align. Finally, small blocks (falling below the limit of
<b>b4</b>) remaining after gap cleaning are also removed.
</p>


<h3>Value</h3>

<p>A <code>matrix</code> of class <code>"DNAbin"</code>
</p>


<h3>Note</h3>

<p><code>gblocks</code> was last updated and tested to work with Gblocks 0.91b.
If you have problems getting the function to work with a newer version of
Gblocks, please contact the package maintainer.
</p>


<h3>References</h3>

<p>Castresana, J. 2000. Selection of conserved blocks from multiple
alignments for their use in phylogenetic analysis. <em>Molecular Biology
and Evolution</em> <b>17</b>, 540-552.
</p>
<p>Talavera, G., and J. Castresana. 2007. Improvement of phylogenies
after removing divergent and ambiguously aligned blocks from protein
sequence alignments. <em>Systematic Biology</em> <b>56</b>, 564-577.
</p>
<p><b>Gblocks website</b>: 
<a href="http://molevol.cmima.csic.es/castresana/Gblocks.html">http://molevol.cmima.csic.es/castresana/Gblocks.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mafft">mafft</a></code> and <code><a href="#topic+prank">prank</a></code> for multiple sequence
alignment; <code><a href="#topic+aliscore">aliscore</a></code> for another alignment masking algorithm.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.28S)
## Not run: gblocks(ips.28S)  
</code></pre>

<hr>
<h2 id='index2DNAbin'>Conversion of Index to DNAbin</h2><span id='topic+index2DNAbin'></span>

<h3>Description</h3>

<p>Use indices of non-empty positions to convert a list of 
DNA sequences into a matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>index2DNAbin(DNAbin, index)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="index2DNAbin_+3A_dnabin">DNAbin</code></td>
<td>
<p>A list of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="index2DNAbin_+3A_index">index</code></td>
<td>
<p>A list of integers containing the indices of base positions.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+DNAbin2index">DNAbin2index</a></code>
</p>

<hr>
<h2 id='ips-internal'>Internal IPS Functions</h2><span id='topic+are.tips.consecutive'></span><span id='topic+assembleDataNode'></span><span id='topic+assembleDistributionNode'></span><span id='topic+assembleInitNode'></span><span id='topic+assembleLoggers'></span><span id='topic+assembleOperators'></span><span id='topic+assembleStateNode'></span><span id='topic+extractBEASTstats'></span><span id='topic+extractMRBAYESstats'></span><span id='topic+formatSet'></span><span id='topic+matrixBlock'></span><span id='topic+rbeauti.taxonset'></span><span id='topic+rc'></span><span id='topic+rc.core'></span><span id='topic+read.xml'></span><span id='topic+setClock'></span><span id='topic+setMCMC'></span><span id='topic+str2cha'></span><span id='topic+terminal.clades'></span><span id='topic+traitRateOutput'></span><span id='topic+traitRateParams'></span><span id='topic+unresolve.clade'></span>

<h3>Description</h3>

<p>Internal <span class="pkg">ips</span> functions.</p>


<h3>Note</h3>

<p>These are internal functions in <b>ips</b> and are not intended to be called by the user.
</p>

<hr>
<h2 id='ips.16S'>Bark Beetle 16S Sequences</h2><span id='topic+ips.16S'></span>

<h3>Description</h3>

<p>This DNA alignment contains 376 positions of 42 sequences of 16S ribosomal DNA of the bark beetle genera <em>Ips</em>, <em>Orthotomicus</em>, and <em>Pityogenes</em> (Scolytinae, Curculionidae, Coleoptera).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ips.16S)</code></pre>


<h3>Format</h3>

<p>The sequences are stored in binary format (see <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>).
</p>


<h3>Source</h3>

<p>The sequences were downloaded and assembled from the Nucleotide repository at GenBank on February 8, 2014.  
</p>


<h3>References</h3>

<p>The nucleotide database on the NCBI website: <a href="http://www.ncbi.nlm.nih.gov/nuccore">http://www.ncbi.nlm.nih.gov/nuccore</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.16S)
</code></pre>

<hr>
<h2 id='ips.28S'>
Bark Beetle 28S Sequences
</h2><span id='topic+ips.28S'></span>

<h3>Description</h3>

<p>This DNA alignment contains 562 positions of 28 sequences of 28S ribosomal DNA of the bark beetle genus <em>Ips</em> (Scolytinae, Curculionidae, Coleoptera).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ips.28S)</code></pre>


<h3>Format</h3>

<p>The sequences are stored in binary format (see <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>).
</p>


<h3>Source</h3>

<p>The sequences were downloaded and assembled from the Nucleotide repository at GenBank on February 8, 2014. 
</p>


<h3>References</h3>

<p>The nucleotide database on the NCBI website: <a href="http://www.ncbi.nlm.nih.gov/nuccore">http://www.ncbi.nlm.nih.gov/nuccore</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.28S)
</code></pre>

<hr>
<h2 id='ips.cox1'>
Bark Beetle COX1 Sequences
</h2><span id='topic+ips.cox1'></span>

<h3>Description</h3>

<p>This DNA alignment contains 770 positions of 26 sequences of cox1 of the bark beetle genera <em>Ips</em>, <em>Orthotomicus</em>, and <em>Pityogenes</em> (Scolytinae, Curculionidae, Coleoptera).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ips.cox1)</code></pre>


<h3>Format</h3>

<p>The sequences are stored in binary format (see <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>).
</p>


<h3>Source</h3>

<p>The sequences were downloaded and assembled from the Nucleotide repository at GenBank on February 8, 2014. 
</p>


<h3>References</h3>

<p>The nucleotide database on the NCBI website: <a href="http://www.ncbi.nlm.nih.gov/nuccore">http://www.ncbi.nlm.nih.gov/nuccore</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.cox1)
</code></pre>

<hr>
<h2 id='ips.tree'>Ips Phylogeny</h2><span id='topic+ips.tree'></span>

<h3>Description</h3>

<p>Phylogentic tree of bark beetles (genus <em>Ips</em>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ips.tree)</code></pre>


<h3>Format</h3>

<p>The format is:
List of 5
$ edge       : int [1:72, 1:2] 38 39 39 40 41 42 42 43 44 45 ...
$ Nnode      : int 36
$ tip.label  : chr [1:37] &quot;Ips_acuminatus&quot; &quot;Ips_duplicatus&quot; &quot;Ips_integer&quot; &quot;Ips_plastographus&quot; ...
$ edge.length: num [1:72] 0.2806 0.0727 0.0295 0.0097 0.021 ...
$ node.label : chr [1:36] &quot;&quot; &quot;100&quot; &quot;21&quot; &quot;12&quot; ...
- attr(*, &quot;class&quot;)= chr &quot;phylo&quot;
- attr(*, &quot;order&quot;)= chr &quot;cladewise&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.tree)
plot(ips.tree)
</code></pre>

<hr>
<h2 id='mafft'>Sequence Alignment with MAFFT</h2><span id='topic+mafft'></span>

<h3>Description</h3>

<p>This function is a wrapper for MAFFT and can be used for 
(profile) aligning of DNA and amino acid sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mafft(x, y, add, method = "auto", maxiterate = 0, op = 1.53,
  ep = 0, gt, options, thread = -1, exec, quiet, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mafft_+3A_x">x</code></td>
<td>
<p>An object of class <code>DNAbin</code> or <code>AAbin</code>.</p>
</td></tr>
<tr><td><code id="mafft_+3A_y">y</code></td>
<td>
<p>An object of class <code>DNAbin</code> or <code>AAbin</code>, if given both
<code>x</code> and <code>y</code> are preserved and aligned to each other (&quot;profile
alignment&quot;).</p>
</td></tr>
<tr><td><code id="mafft_+3A_add">add</code></td>
<td>
<p>A character string giving the method used for adding <code>y</code> to 
<code>x</code>: <code>"add"</code>, <code>"addprofile"</code> (default), or any unambiguous 
abbreviation of these.</p>
</td></tr>
<tr><td><code id="mafft_+3A_method">method</code></td>
<td>
<p>A character string giving the alignment method. Available 
accuracy-oriented methods for less than 200 sequences are 
<code>"localpair"</code>, <code>"globalpair"</code>, and <code>"genafpair"</code>; 
<code>"retree 1"</code> and <code>"retree 2"</code> are for speed-oriented alignment. 
The default is <code>"auto"</code>, which lets MAFFT choose an appropriate 
alignment method.</p>
</td></tr>
<tr><td><code id="mafft_+3A_maxiterate">maxiterate</code></td>
<td>
<p>An integer giving the number of cycles of iterative 
refinement to perform. Possible choices are <code>0</code>: progressive method, 
no iterative refinement (default); <code>2</code>: two cycles of iterative 
refinement; <code>1000</code>: at most 1000 cycles of iterative refinement.</p>
</td></tr>
<tr><td><code id="mafft_+3A_op">op</code></td>
<td>
<p>A numeric giving the <code>gap opening penalty</code> at group-to-group 
alignment; default 1.53.</p>
</td></tr>
<tr><td><code id="mafft_+3A_ep">ep</code></td>
<td>
<p>A numeric giving the offset value, which works like <code>gap 
extension penalty</code>, for group-to-group alignment; default 0.0, but 0.123 is
recommended if no long indels are expected.</p>
</td></tr>
<tr><td><code id="mafft_+3A_gt">gt</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code> that is to be used as a 
guide tree during alignment.</p>
</td></tr>
<tr><td><code id="mafft_+3A_options">options</code></td>
<td>
<p>A vector of mode character specifying additional arguments to 
MAFFT, that are not included in <code>mafft</code> such as, e.g., 
<code>--adjustdirection</code>.</p>
</td></tr>
<tr><td><code id="mafft_+3A_thread">thread</code></td>
<td>
<p>Integer giving the number of physical cores MAFFT should use; 
with <code>thread = -1</code> the number of cores is determined automatically.</p>
</td></tr>
<tr><td><code id="mafft_+3A_exec">exec</code></td>
<td>
<p>A character string giving the path to the MAFFT executable 
including its name, e.g. something like <code>/user/local/bin/mafft</code> under 
UNIX-alikes.</p>
</td></tr>
<tr><td><code id="mafft_+3A_quiet">quiet</code></td>
<td>
<p>Logical, if set to <code>TRUE</code>, mafft progress is printed out on
the screen.</p>
</td></tr>
<tr><td><code id="mafft_+3A_file">file</code></td>
<td>
<p>A character string indicating the filename of the output FASTA
file; if this is missing the the alignment will be returned as  matrix of
class <code>DNAbin</code> or <code>AAbin</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>"localpair"</code> selects the <b>L-INS-i</b> algorithm, probably 
most accurate; recommended for &lt;200 sequences; iterative refinement method 
incorporating local pairwise alignment information.
</p>
<p><code>"globalpair"</code> selects the <b>G-INS-i</b> algorithm suitable for 
sequences of similar lengths; recommended for &lt;200 sequences; iterative 
refinement method incorporating global pairwise alignment information.
</p>
<p><code>"genafpair"</code> selects the <b>E-INS-i</b> algorithm suitable for 
sequences containing large unalignable regions; recommended for &lt;200 
sequences.
</p>
<p><code>"retree 1"</code> selects the <b>FFT-NS-1</b> algorithm, the simplest 
progressive option in MAFFT; recommended for &gt;200 sequences.
</p>
<p><code>"retree 2"</code> selects the <b>FFT-NS-2</b> algorithm that uses a second 
iteration of alignment based on a guide tree computed from an FFT-NS-1 
alignment; this is the default in MAFFT; recommended for &gt;200 sequences.
</p>


<h3>Value</h3>

<p>A <code>matrix</code> of class <code>"DNAbin"</code> or <code>"AAbin"</code>.
</p>


<h3>Note</h3>

<p><code>mafft</code> was last updated and tested to work with MAFFT 7.205. If 
you have problems getting the function to work with a newer version of 
MAFFT, please contact the package maintainer.
</p>


<h3>References</h3>

<p>Katoh, K. and H. Toh. 2008. Recent developments in the MAFFT 
multiple sequence alignment program. <em>Briefings in Bioinformatics</em> 
<b>9</b>: 286-298.
</p>
<p>Katoh, K., K.-i. Kuma, H. Toh, and T. Miyata. 2005. Mafft version 5: 
improvement in accuracy of multiple sequence alignment. <em>Nucleic Acids
Research</em> <b>33</b>: 511&ndash;518.
</p>
<p>Katoh, K., K. Misawa, K.-i. Kuma, and T. Miyata. 2002. Mafft: a novel 
method for rapid multiple sequence alignment based on fast Fourier 
transform. <em>Nucleid Acids Research</em> <b>30</b>: 3059&ndash;3066.
</p>
<p><a href="http://mafft.cbrc.jp/alignment/software/index.html">http://mafft.cbrc.jp/alignment/software/index.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fas">read.fas</a></code> to import DNA sequences; <code><a href="#topic+prank">prank</a></code> 
for another alignment algorithm; <code><a href="#topic+gblocks">gblocks</a></code> and 
<code><a href="#topic+aliscore">aliscore</a></code> for alignment cleaning.
</p>

<hr>
<h2 id='mafft.merge'>Profile Alignment with MAFFT</h2><span id='topic+mafft.merge'></span>

<h3>Description</h3>

<p>Merge two or more DNA or amino acid sequence alignments by
profile alignment with MAFFT.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mafft.merge(subMSA, method = "auto", gt, thread = -1, exec,
  quiet = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mafft.merge_+3A_submsa">subMSA</code></td>
<td>
<p>A list of objects of class <code>"<a href="ape.html#topic+DNAbin">DNAbin</a>"</code> or
<code>"<a href="ape.html#topic+AAbin">AAbin</a>"</code>.</p>
</td></tr>
<tr><td><code id="mafft.merge_+3A_method">method</code></td>
<td>
<p>A character string giving the alignment method. Available
accuracy-oriented methods for less than 200 sequences are
<code>"localpair"</code>, <code>"globalpair"</code>, and <code>"genafpair"</code>;
<code>"retree 1"</code> and <code>"retree 2"</code> are for speed-oriented alignment.
The default is <code>"auto"</code>, which lets MAFFT choose an appropriate
alignment method.</p>
</td></tr>
<tr><td><code id="mafft.merge_+3A_gt">gt</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code> that is to be used as a
guide tree during alignment.</p>
</td></tr>
<tr><td><code id="mafft.merge_+3A_thread">thread</code></td>
<td>
<p>Integer giving the number of physical cores MAFFT should use;
with <code>thread = -1</code> the number of cores is determined automatically.</p>
</td></tr>
<tr><td><code id="mafft.merge_+3A_exec">exec</code></td>
<td>
<p>A character string giving the path to the MAFFT executable
including its name, e.g. something like <code>/user/local/bin/mafft</code> under
UNIX-alikes.</p>
</td></tr>
<tr><td><code id="mafft.merge_+3A_quiet">quiet</code></td>
<td>
<p>Logical, if set to <code>TRUE</code>, mafft progress is printed out on
the screen.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>"<a href="ape.html#topic+DNAbin">DNAbin</a>"</code> or <code>"<a href="ape.html#topic+AAbin">AAbin</a>"</code>.
</p>

<hr>
<h2 id='mrbayes'>Bayesian MCMC Tree Search with MrBayes</h2><span id='topic+mrbayes'></span>

<h3>Description</h3>

<p>Provides a wrapper for Bayesian phylogenetic tree search through MrBayes (Huelsenbeck &amp; Ronquist, 2001; Ronquist &amp; Huelsenbeck, 2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrbayes(x, file = "", lset, prset, mcmc, unlink, constraint,
        burnin = 10, contype = "allcompat", exec, run = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrbayes_+3A_x">x</code></td>
<td>

<p>An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code> in the case of <code>mrbayes</code>.
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_file">file</code></td>
<td>

<p>A character string, giving the name of the MrBayes input file.
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_lset">lset</code></td>
<td>

<p>A list as returned by <code><a href="#topic+mrbayes.lset">mrbayes.lset</a></code> containing the parameter settings of the model of molecular evolution.  
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_prset">prset</code></td>
<td>

<p>A list as returned by <code><a href="#topic+mrbayes.prset">mrbayes.prset</a></code> containing the parameter setting for the prior distributions. 
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_mcmc">mcmc</code></td>
<td>

<p>A list as returned by <code><a href="#topic+mrbayes.mcmc">mrbayes.mcmc</a></code> containing the parameter setting for the Markov chain Monte Carlo (MCMC).  
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_unlink">unlink</code></td>
<td>

</td></tr>
<tr><td><code id="mrbayes_+3A_constraint">constraint</code></td>
<td>

</td></tr>
<tr><td><code id="mrbayes_+3A_burnin">burnin</code></td>
<td>

<p>An integer; the number of samples from the MCMC to be discarded prior to further analysis.
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_contype">contype</code></td>
<td>

<p>A character string; the type of consensus tree calculated from the posterior distribution of trees: either <code>"halfcompat"</code> (majority-rule consensus tree) or <code>"allcombat"</code> (strict consensus tree).
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_exec">exec</code></td>
<td>

<p>A character string giving the full path of the MrBayes program.
</p>
</td></tr>
<tr><td><code id="mrbayes_+3A_run">run</code></td>
<td>

<p>Logical; <code>run = FALSE</code> will only print the NEXUS file, <code>run = TRUE</code> will also start the MCMC runs, if <code>exec</code> is correctly specified.
</p>
</td></tr>
</table>


<h3>Details</h3>

 	
<p><code>mrbayes</code> was last updated and tested with MrBayes <b>v3.2.2</b> under R 3.1.0 on a x86_64-apple-darwin10.8.0 (64-bit) platform. It is intended to offer a simply parameterized building block for larger scripts.	
</p>


<h3>Value</h3>

<p>None; a NEXUS file with MrBayes block is written to a file and, if <code>run = TRUE</code>, the MCMC runs in MrBayes are started.
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>J. P. Huelsenbeck &amp; Ronquist F. 2001. MrBayes: Bayesian inference of phylogenetic trees. <em>Bioinformatics</em> <b>17</b>: 754-755.
</p>
<p>Ronquist F. &amp; J. P. Huelsenbeck. 2003. MrBayes 3: Bayesian phylogenetic inference under mixed models. <em>Biometrics</em> <b>19</b>: 1572-1574.
</p>
<p>MrBayes website: <a href="http://mrbayes.sourceforge.net/">http://mrbayes.sourceforge.net/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mafft">mafft</a></code> and <code><a href="#topic+prank">prank</a></code> for sequence alignment; <code><a href="#topic+raxml">raxml</a> for maximum likelihood tree search.</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
data(ips.cox1)
x &lt;- ips.cox1[, 100:140] # tiny alignment
mrbayes(x, file = "", mcmc = mrbayes.mcmc(ngen = 100), run = FALSE)

## Not run: 

library(phangorn)
tree &lt;- rtree(10)
Y1 &lt;- simSeq(tree, l = 20)
Y2 &lt;- simSeq(tree, l = 20, type = "USER", levels=c("0", "1"))
Y &lt;- cbind(as.character(Y1), as.character(Y2))
mrbayes(Y, file = "", run = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='mrbayes.lset'>Model Settings for MrBayes</h2><span id='topic+mrbayes.lset'></span>

<h3>Description</h3>

<p>Set model parameters for <code><a href="#topic+mrbayes">mrbayes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrbayes.lset(..., partition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrbayes.lset_+3A_...">...</code></td>
<td>

<p>arguments in <code>tag = value</code> form, or a list of tagged values. The tags must come from the names of model parameters described in the â€˜Model Parametersâ€™ section.
</p>
</td></tr>
<tr><td><code id="mrbayes.lset_+3A_partition">partition</code></td>
<td>

</td></tr>
</table>


<h3>Value</h3>

<p>a list containing a subset (including the empty and the full set) of model parameters.
</p>


<h3>Model Parameters</h3>


<dl>
<dt>nucmodel</dt><dd>
<p><code>"4by4"</code>, <code>"doublet"</code>, <code>"codon"</code>, or <code>"protein"</code>.
</p>
</dd>
<dt>nst</dt><dd>
<p><code>1</code>, <code>2</code>, <code>6</code>, or <code>"mixed"</code>.
</p>
</dd>
<dt>code</dt><dd>
<p><code>"universal"</code>, <code>"vertmt"</code>, <code>"mycoplasma"</code>, <code>"yeast"</code>, <code>"ciliates"</code>, or <code>"metmt"</code>. 
</p>
</dd>
<dt>ploidy</dt><dd>
<p><code>"haploid"</code>, <code>"diploid"</code>,  or <code>"zlinked"</code>.
</p>
</dd>
<dt>rates</dt><dd>
<p><code>"equal"</code>, <code>"gamma"</code>, <code>"propinv"</code>,<code>"invgamma"</code>, or <code>"adgamma"</code>.
</p>
</dd>
<dt>ngammacat</dt><dd>
<p><code>1-24</code> 
</p>
</dd>
<dt>nbetacat</dt><dd>
<p><code>1-24</code>
</p>
</dd>
<dt>omegavar</dt><dd>
<p><code>"equal"</code>,  <code>"ny98"</code>,  or <code>"m3"</code>.
</p>
</dd>
<dt>covarion</dt><dd>
<p><code>"no"</code> or <code>"yes"</code>. 
</p>
</dd>
<dt>coding</dt><dd>
<p><code>"all"</code>, <code>"variable"</code>, <code>"noabsencesites"</code>, or <code>"nopresencesites"</code>. 
</p>
</dd>
<dt>parsmodel</dt><dd>
<p><code>"no"</code> or <code>"yes"</code>.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>J.P. Huelsenbeck &amp; Ronquist F. 2001. MrBayes: Bayesian inference of phylogenetic trees. <em>Bioinformatics</em> <b>17</b>: 754-755.
</p>
<p>Ronquist F. &amp; J.P. Huelsenbeck. 2003. MrBayes 3: Bayesian phylogenetic inference under mixed models. <em>Biometrics</em> <b>19</b>: 1572-1574.
</p>
<p>MrBayes website: <a href="http://mrbayes.sourceforge.net/">http://mrbayes.sourceforge.net/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mrbayes.prset">mrbayes.prset</a></code> to set prior distributions, <code><a href="#topic+mrbayes.mcmc">mrbayes.mcmc</a></code> to set parameters of the Markov chain Monte Carlo (MCMC), and <code><a href="#topic+mrbayes">mrbayes</a></code> to run MrBayes locally or prepare input files for a computer cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## F81
mrbayes.lset(nst = 2)

## GTR + Gamma
mrbayes.lset(nst = 6, rates = "gamma")

## GTR + Gamma + I
mrbayes.lset(nst = 6, rates = "invgamma")
</code></pre>

<hr>
<h2 id='mrbayes.mcmc'>MCMC Settings for MrBayes</h2><span id='topic+mrbayes.mcmc'></span>

<h3>Description</h3>

<p>Set Markov chain Monte Carlo (MCMC) parameters for <code><a href="#topic+mrbayes">mrbayes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrbayes.mcmc(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrbayes.mcmc_+3A_...">...</code></td>
<td>

<p>arguments in <code>tag = value</code> form, or a list of tagged values. The tags must come from the names of MCMC parameters described in the â€˜MCMC Parametersâ€™ section.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing a subset (including the empty and the full set) of model parameters.
</p>


<h3>MCMC Parameters</h3>


<dl>
<dt>ngen</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>nruns</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>nchains</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>temp</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>swapfreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>nswaps</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>samplefreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>printfreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>printall</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>printmax</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>mcmcdiagn</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>diagnfreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>diagnstat</dt><dd> 
<p><code>"avgstddev"</code> or <code>"maxstddev"</code>
</p>
</dd>
<dt>minpartfreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>allchains</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>allcomps</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>relburnin</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>burnin</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>burninfrac</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>stoprule</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>stopval</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>savetrees</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>checkpoint</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>checkfreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>startparams</dt><dd> 
<p><code>"current"</code> or <code>"reset"</code>
</p>
</dd>
<dt>starttree</dt><dd> 
<p><code>"current"</code>, <code>"random"</code>, or <code>"parsimony"</code>
</p>
</dd>
<dt>nperts</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
<dt>data</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>ordertaxa</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>append</dt><dd> 
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>autotune</dt><dd>
<p><code>"yes"</code> or <code>"no"</code>
</p>
</dd>
<dt>tunefreq</dt><dd> 
<p>&quot;NUMERIC&quot;
</p>
</dd>
</dl>



<h3>Note</h3>

<p>The parameters <b>reweight</b> and <b>filename</b> cannot be set via <code>mrbayes.mcmc</code>.</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>J.P. Huelsenbeck &amp; Ronquist F. 2001. MrBayes: Bayesian inference of phylogenetic trees. <em>Bioinformatics</em> <b>17</b>: 754-755.
</p>
<p>Ronquist F. &amp; J.P. Huelsenbeck. 2003. MrBayes 3: Bayesian phylogenetic inference under mixed models. <em>Biometrics</em> <b>19</b>: 1572-1574.
</p>
<p>MrBayes website: <a href="http://mrbayes.sourceforge.net/">http://mrbayes.sourceforge.net/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mrbayes.lset">mrbayes.lset</a></code> to set model parameters, <code><a href="#topic+mrbayes.prset">mrbayes.prset</a></code> to set prior distributions, and <code><a href="#topic+mrbayes">mrbayes</a></code> to run MrBayes locally or prepare input files for a computer cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mrbayes.mcmc()
</code></pre>

<hr>
<h2 id='mrbayes.prset'>Set Priors for MrBayes</h2><span id='topic+mrbayes.prset'></span>

<h3>Description</h3>

<p>Set prior distributions for <code><a href="#topic+mrbayes">mrbayes</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mrbayes.prset(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mrbayes.prset_+3A_...">...</code></td>
<td>

<p>arguments in <code>tag = value</code> form, or a list of tagged values. The tags must come from the names of prior distribution parameters described in the â€˜Prior Distribution Parametersâ€™ section.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of length zero (see 'Note')
</p>


<h3>Prior Distribution Parameters</h3>


<dl>
<dt>traitiopr</dt><dd></dd>
<dt>revmatpr</dt><dd></dd>
<dt>aamodelpr</dt><dd></dd>
<dt>aarevmatpr</dt><dd></dd>
<dt>omegapr</dt><dd></dd>
<dt>ny98omega1pr</dt><dd></dd>
<dt>ny98omega3pr</dt><dd></dd>
<dt>m3omegapr</dt><dd></dd>
<dt>codoncatfreqs</dt><dd></dd>
<dt>statefreqpr</dt><dd></dd>
<dt>shapepr</dt><dd></dd>
<dt>ratecorrpr</dt><dd></dd>
<dt>pinvarpr</dt><dd></dd>
<dt>covswitchpr</dt><dd></dd>
<dt>symdirihyperpr</dt><dd></dd>
<dt>topologypr</dt><dd></dd>
<dt>brlenspr</dt><dd></dd>
<dt>clockvarpr</dt><dd></dd>
<dt>igrvarpr</dt><dd></dd>
</dl>



<h3>Note</h3>

<p>This function currently returns an empty set of prior distribution parameters, i.e., you cannot change the MrBayes default parameters. </p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>J.P. Huelsenbeck &amp; Ronquist F. 2001. MrBayes: Bayesian inference of phylogenetic trees. <em>Bioinformatics</em> <b>17</b>: 754-755.
</p>
<p>Ronquist F. &amp; J.P. Huelsenbeck. 2003. MrBayes 3: Bayesian phylogenetic inference under mixed models. <em>Biometrics</em> <b>19</b>: 1572-1574.
</p>
<p>MrBayes website: <a href="http://mrbayes.sourceforge.net/">http://mrbayes.sourceforge.net/</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mrbayes.lset">mrbayes.lset</a></code> to set model parameters, <code><a href="#topic+mrbayes.mcmc">mrbayes.mcmc</a></code> to set parameters of the Markov chain Monte Carlo (MCMC), and <code><a href="#topic+mrbayes">mrbayes</a></code> to run MrBayes locally or prepare input files for a computer cluster.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mrbayes.prset()
</code></pre>

<hr>
<h2 id='multistate'>MULTISTATE</h2><span id='topic+multistate'></span><span id='topic+multistateMCMC'></span><span id='topic+multistateML'></span>

<h3>Description</h3>

<p>These functions provide wrappers to <code>BayesMultiState</code> in the BayesTraits package written by Mark Pagel and Andrew Meade.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multistateML(phy, traits, model = "ARD", anc.states = TRUE, 
  path = "/Applications/BayesTraits", dir = NULL)

multistateMCMC(phy, traits, model = "ARD", anc.states = TRUE, 
    rd = 2, rjhp = NULL, fixNodes = NULL, it = 1e+05, bi = 10000,
    sa = 1000, path = "/Applications/BayesTraits", dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multistate_+3A_phy">phy</code></td>
<td>
<p>an object of class <code>phylo</code>.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_traits">traits</code></td>
<td>
<p>a <code>data.frame</code> with two columns. The first column contains the taxon labels, the second column contains the character states.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_model">model</code></td>
<td>

</td></tr>
<tr><td><code id="multistate_+3A_anc.states">anc.states</code></td>
<td>
<p>either <code>logical</code> or a <code>list</code>, the latter containing the tip labels of those internal nodes, for which the likelihood of ancestral character states should be estimated.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_rd">rd</code></td>
<td>
<p>a real number, giving the <code>RateDev</code> parameter, i.e., the deviation of the normal distribution, that changes to the rates are drawn from. Should be set such that acceptance of the rate parameters is about 0.2.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_rjhp">rjhp</code></td>
<td>
<p>a character string giving the details of priors and hyperpriors for the reversible jump MCMC (rjMCMC). If left <code>NULL</code>, a conventional MCMC is used. In order to use the rjMCMC, you must specify the distribution of the prior and the interval of the uniform hyperprior distribution that seeds it. For example, <code>exp 0 30</code> specifies an exponential distribution seeded from a uniform distribution on the interval 0 to 30, and <code>gamma 0 10 0 10</code> specifies a gamma prior with its mean and standard deviation seeded from uniform distributions on the interval 0 to 10.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_fixnodes">fixNodes</code></td>
<td>
<p>a list giving fixed character states of certain internal nodes. This argument corresponds to the <code>fossil</code> command in the MultiState manual.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_it">it</code></td>
<td>
<p>numeric, sets the number of iterations to run the MCMC for.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_bi">bi</code></td>
<td>
<p>numeric, sets the number of iterations of the MCMC that will be discarded as burn-in.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_sa">sa</code></td>
<td>
<p>numeric, sets the the sample period in the MCMC.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_path">path</code></td>
<td>
<p>a character string giving the path to executables in the BayesTraits package.
</p>
</td></tr>
<tr><td><code id="multistate_+3A_dir">dir</code></td>
<td>
<p>a character string giving a directory name where the input and output files will be stored. The directory will be created by <code>multistateML</code> and must not exist already. If <code>dir = NULL</code> (default) input and output is written to the working directory (thereby overwriting existing output).
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>The BayesTraits manual: <a href="http://www.evolution.rdg.ac.uk/Files/BayesTraits-V1.0-Manual.pdf">http://www.evolution.rdg.ac.uk/Files/BayesTraits-V1.0-Manual.pdf</a>.
</p>
<p>Pagel, M., A. Meade, and D. Barker. 2004. Bayesian estiamation of ancestral character states on phylogenies. <em>Syst. Biol.</em> <b>53</b>: 673-684.
</p>
<p>Pagel, M. and A. Meade. 2006. Bayesian analysis of correlated evolution of discrete characters by reversible-jump Markov chain Monte Carlo. <em>Am. Nat.</em> <b>167</b>: 808-825.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+ace">ace</a></code>
</p>

<hr>
<h2 id='neighboringPairs'>Neighboring Nodes in a Minimum Spanning Tree</h2><span id='topic+neighboringPairs'></span>

<h3>Description</h3>

<p>Finds all pairs of adjacent nodes, i.e. 
nodes separated by only one edge, in a minimum spanning 
tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>neighboringPairs(mst)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="neighboringPairs_+3A_mst">mst</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+mst">mst</a></code>.</p>
</td></tr>
</table>

<hr>
<h2 id='ntip'>Numbers of Tips of (Sub)trees</h2><span id='topic+ntip'></span>

<h3>Description</h3>

<p>Counts the number of tips of a given clade of a 
phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ntip(phy, node)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ntip_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="ntip_+3A_node">node</code></td>
<td>
<p>An integer given the number of an internal node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An integer giving the number of tips.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
tr &lt;- rtree(12)
plot(tr); nodelabels()
ntip(tr, 16)
</code></pre>

<hr>
<h2 id='partitionfinder'>PartitionFinder</h2><span id='topic+partitionfinder'></span>

<h3>Description</h3>

<p>Provides a wrapper to the PartitionFinder software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>partitionfinder(alignment, user.tree, branchlengths = "linked",
  models = "all", model.selection = "BIC", search = "greedy",
  exec = "/Applications/PartitionFinderV1.1.1_Mac/PartitionFinder.py")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="partitionfinder_+3A_alignment">alignment</code></td>
<td>
<p>A</p>
</td></tr>
<tr><td><code id="partitionfinder_+3A_user.tree">user.tree</code></td>
<td>
<p>A</p>
</td></tr>
<tr><td><code id="partitionfinder_+3A_branchlengths">branchlengths</code></td>
<td>
<p>A</p>
</td></tr>
<tr><td><code id="partitionfinder_+3A_models">models</code></td>
<td>
<p>A</p>
</td></tr>
<tr><td><code id="partitionfinder_+3A_model.selection">model.selection</code></td>
<td>
<p>A</p>
</td></tr>
<tr><td><code id="partitionfinder_+3A_search">search</code></td>
<td>
<p>A</p>
</td></tr>
<tr><td><code id="partitionfinder_+3A_exec">exec</code></td>
<td>
<p>A character string giving the path to the executable (python 
script).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lanfear, R., B. Calcott, S.Y.W. Ho, and S. Guindon. 2012. 
PartitionFinder: combined selection of partitioning schemes and
substitution models for phylogenetic analyses. <em>Molecular 
Biology and Evolution</em> <strong>29</strong>: 1695-1701.  
</p>
<p>Lanfear, R., B. Calcott, K. David, C. Mayer, and A. Stamatakis. 2014. 
Selecting optimal partitioning schemes for phylogenomic datasets.
<em>BMC Evolutionary Biology</em> <strong>14</strong>: 82.
</p>

<hr>
<h2 id='pathd8'>PATHd8</h2><span id='topic+pathd8'></span>

<h3>Description</h3>

<p>This function is a wrapper for PATHd8 and can be used for
phylogenetic dating, especially of large trees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathd8(phy, exec = "/Applications/PATHd8/PATHd8", seql, calibration)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathd8_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="pathd8_+3A_exec">exec</code></td>
<td>
<p>A character string giving the path to the PATHd8 program.</p>
</td></tr>
<tr><td><code id="pathd8_+3A_seql">seql</code></td>
<td>
<p>sequence length of alignment</p>
</td></tr>
<tr><td><code id="pathd8_+3A_calibration">calibration</code></td>
<td>
<p>A data frame with 4 columns and as many rows as 
calibration points. Columns are: taxon 1; taxon 2; one of c(&quot;minage&quot;, 
&quot;maxage&quot;, &quot;fixage&quot;); age.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tree list of ultrametric trees returned from PATHd8 of class 
<code><a href="ape.html#topic+phylo">phylo</a></code>. First tree is PATHd8 chronogram, which is a calibrated
ultrametric tree. Second is a PATH tree, which is a ultrametric tree 
without calibration.
</p>


<h3>Author(s)</h3>

<p>Franz-Sebastian Krah
</p>


<h3>References</h3>

<p>Britton et al (2006). PATHd8&mdash;a new method for estimating 
divergence times in large phylogenetic trees without a molecular clock. 
Available from the authors (www.math.su.se/PATHd8)
</p>
<p>Britton et al. (2007). Estimating divergence times in large 
phylogenetic trees. Systematic biology. 56:741&ndash;752
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## This example is taken from the PATHD8 manual
cal &lt;- rbind(cal1 = c("Rat", "Ostrich", "minage", 260), 
cal2 = c("Human", "Platypus", "fixage", 125),
cal3 = c("Alligator", "Ostrich", "minage", 150))
colnames(cal) = c("tax1", "tax2", "age_type", "age")
phy &lt;- read.tree(text = paste0("((((Rat:0.007148,Human:0.001808):0.024345,",
                               "Platypus:0.016588):0.012920,(Ostrich:0.018119,",
                               "Alligator:0. 006232):0.004708):0.028037,Frog:0);")
seql &lt;- 1823
pathd8(phy, exec = "/Applications/PATHd8/PATHd8", seql = seql, calibration = cal)

## End(Not run)
</code></pre>

<hr>
<h2 id='phylo2mafft'>Convert Trees for MAFFT</h2><span id='topic+phylo2mafft'></span>

<h3>Description</h3>

<p>Converts a phylogenetic tree of class <code>"phylo"</code> to a format
usable as a guide tree by MAFFT. This function is called internally by 
<code><a href="#topic+mafft">mafft</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo2mafft(phy, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo2mafft_+3A_phy">phy</code></td>
<td>
<p>A phylogenetic tree of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="phylo2mafft_+3A_file">file</code></td>
<td>
<p>A character string giving a filename. May be missing, in which
case the results are only printed on the screen.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix coding the MAFFT-formatted tree, as a side effect the same 
matrix is written to <code>file</code>.
</p>


<h3>References</h3>

<p>The MAFFT website:
<a href="http://mafft.cbrc.jp/alignment/software/index.html">http://mafft.cbrc.jp/alignment/software/index.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mafft">mafft</a></code> for an interface to MAFFT.
</p>

<hr>
<h2 id='phylo2mst'>Conversion from PHYLO to MST Object</h2><span id='topic+phylo2mst'></span>

<h3>Description</h3>

<p>Converts a phylogenetic tree (class <code>phylo</code>) into a
minimum spanning tree (class <code>mst</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phylo2mst(phy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phylo2mst_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The current version of <code>phylo2mst</code> does not handle polytomies
and does not incorporate branch length information. Note that topological 
information is lost during the conversion.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phy &lt;- rtree(12)
plot(phy)
mst &lt;- phylo2mst(phy)
plot(mst)
</code></pre>

<hr>
<h2 id='pis'>Number of Potentially-Informative Sites</h2><span id='topic+pis'></span>

<h3>Description</h3>

<p>This function returns the number or positions of potentially-informative (parsimony-informative, phylogenetically-informative) sites in DNA sequence alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pis(x, what = "fraction", use.ambiguities = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pis_+3A_x">x</code></td>
<td>
<p> An object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="pis_+3A_what">what</code></td>
<td>
<p>Either of <code>"absolute"</code>, <code>"fraction"</code>, or <code>"index"</code>, which will return the absolute number, the relative number or the indeces of the potentially-informative sites.</p>
</td></tr>
<tr><td><code id="pis_+3A_use.ambiguities">use.ambiguities</code></td>
<td>
<p><em>Not yet available</em>.

</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric (depending on <code>what</code>, the number, fraction, or indices of potentially-informative nucleotide sites).
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>Examples</h3>

<pre><code class='language-R'># example data:
# -------------
data(ips.16S)	

# number of potentially-informative sites:
# ----------------------------------------
pis(ips.16S, what = "abs")
	
# proportion of potentially-informative sites:
# --------------------------------------------
pis(ips.16S, what = "frac")

# indeces of potentially-informative sites:
# --------------------------------------------
pis(ips.16S, what = "ind")
</code></pre>

<hr>
<h2 id='prank'>PRANK</h2><span id='topic+prank'></span>

<h3>Description</h3>

<p>DNA sequence Alignment Using the program PRANK.</p>


<h3>Usage</h3>

<pre><code class='language-R'>prank(x, outfile, guidetree = NULL, gaprate = 0.025, 
    gapext = 0.75, path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prank_+3A_x">x</code></td>
<td>
<p>an object of class <code>DNAbin</code>.</p>
</td></tr>
<tr><td><code id="prank_+3A_outfile">outfile</code></td>
<td>
<p>a character string giving a name for the output file.</p>
</td></tr>
<tr><td><code id="prank_+3A_guidetree">guidetree</code></td>
<td>
<p>an object of class <code>phylo</code> to be used as guidetree in alignment.</p>
</td></tr>
<tr><td><code id="prank_+3A_gaprate">gaprate</code></td>
<td>
<p>numeric giving the gap opening rate; defaults to 0.025.</p>
</td></tr>
<tr><td><code id="prank_+3A_gapext">gapext</code></td>
<td>
<p>numeric giving the gap extension penalty; defaults to 0.75.</p>
</td></tr>
<tr><td><code id="prank_+3A_path">path</code></td>
<td>
<p>a character string indicating the path to the PRANK executable.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>matrix</code> of class <code>"DNAbin"</code>
</p>


<h3>Note</h3>

<p><code>prank</code> was last updated and tested to work with PRANK v. 120814 on Windows XP. If you have problems getting the function to work with a newer version of PRANK, contact the package maintainer.</p>


<h3>References</h3>

<p><a href="https://www.ebi.ac.uk/research/goldman/software/prank">https://www.ebi.ac.uk/research/goldman/software/prank</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fas">read.fas</a></code> to import DNA sequences;
<code><a href="#topic+mafft">mafft</a></code> for another alignment algorithm;
<code><a href="#topic+gblocks">gblocks</a></code> and <code><a href="#topic+aliscore">aliscore</a></code> for alignment cleaning. 
</p>

<hr>
<h2 id='raxml'>Maximum Likelihood Tree Estimation with RAxML</h2><span id='topic+raxml'></span>

<h3>Description</h3>

<p>Provides an interface to the C program <b>RAxML</b> (see
Reference section) for maximum likelihood estimation of tree topology and/or
branch lengths, rapid and conventional non-parametric bootstrapping, mapping
splits onto individual topologies, and a lot more. See the RAxML manual for
details, especially if you are a new user of RAxML.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>raxml(DNAbin, m = "GTRCAT", f, N, p, b, x, k, weights, partitions,
  outgroup, backbone = NULL, file = paste0("fromR_", Sys.Date()), exec,
  threads)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="raxml_+3A_dnabin">DNAbin</code></td>
<td>
<p>A matrix of DNA sequences of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="raxml_+3A_m">m</code></td>
<td>
<p>A vector of mode <code>"character"</code> defining a model of molecular
evolution; currently only GTR model available.</p>
</td></tr>
<tr><td><code id="raxml_+3A_f">f</code></td>
<td>
<p>A vector of mode <code>"character"</code> selecting an RAxML algorithm
analogous to the <code>-f</code> flag (see Detail section and RAxML manual).</p>
</td></tr>
<tr><td><code id="raxml_+3A_n">N</code></td>
<td>
<p>Either of mode <code>"integer"</code> or <code>"character"</code>. Integers give
the number of independent searches on different starting tree or replicates
in bootstrapping. Alternatively, one of four bootstopping criteria can be
chosen: <code>"autoFC"</code>, <code>"autoMR"</code>, <code>"autoMRE"</code>, or
<code>"autoMRE_IGN"</code>.</p>
</td></tr>
<tr><td><code id="raxml_+3A_p">p</code></td>
<td>
<p>Integer, setting a random seed for the parsimony starting trees.</p>
</td></tr>
<tr><td><code id="raxml_+3A_b">b</code></td>
<td>
<p>Integer, setting a random seed for bootstrapping.</p>
</td></tr>
<tr><td><code id="raxml_+3A_x">x</code></td>
<td>
<p>Integer, setting a random seed for rapid bootstrapping.</p>
</td></tr>
<tr><td><code id="raxml_+3A_k">k</code></td>
<td>
<p>Logical, if <code>TRUE</code>, the branch lengths of bootstrapped trees are
recorded.</p>
</td></tr>
<tr><td><code id="raxml_+3A_weights">weights</code></td>
<td>
<p>A vector of mode <code>"numeric"</code> giving integers to assign
individual weights to each column of the alignment. (-a)</p>
</td></tr>
<tr><td><code id="raxml_+3A_partitions">partitions</code></td>
<td>
<p>A data frame giving the partitions of the alignment.</p>
</td></tr>
<tr><td><code id="raxml_+3A_outgroup">outgroup</code></td>
<td>
<p>A vector of mode <code>"character"</code> containing the names of
the outgroup taxa.</p>
</td></tr>
<tr><td><code id="raxml_+3A_backbone">backbone</code></td>
<td>
<p>A <code><a href="ape.html#topic+phylo">phylo</a></code> object representing a backbone tree.</p>
</td></tr>
<tr><td><code id="raxml_+3A_file">file</code></td>
<td>
<p>A vector of mode <code>"character"</code> giving a name for the output
files.</p>
</td></tr>
<tr><td><code id="raxml_+3A_exec">exec</code></td>
<td>
<p>A vector of mode <code>"character"</code> giving the path to the
directory containing the RAxML executable. The default value will work on
Mac OS X if the folder containing the executable is renamed to
<code>"RAxML-8.0.3"</code>.</p>
</td></tr>
<tr><td><code id="raxml_+3A_threads">threads</code></td>
<td>
<p>Integer, giving the number of parallel threads to use (PTHREADS
only).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>There are some limitations of this wrapper compared to RAxML run
directly from the command line. </p>
 <ol>
<li><p> Only DNA is allowed as
data type. </p>
</li>
<li><p> Option <code>f</code> can only take a limited number of values
(<code>d</code>, <code>a</code>). </p>
</li></ol>
 
<p>RAxML needs the specification of random seeds for parsimony estimation of
starting trees and for bootstrap resampling. The corresponding argument
names in <code>raxml</code> are identical to the flags used by RAxML (<code>-p</code>,
<code>-b</code>, and <code>-x</code>). If you choose not to give any values,
<code>raxml</code> will generate a (different) value for each required random
seed every time it is called. Be aware that <code><a href="base.html#topic+set.seed">set.seed</a></code> will work
only for <code>p</code>, but not for <code>b</code> or <code>x</code>.
</p>


<h3>Value</h3>

<p>A list with a variable number of elements, depending on the analysis
chosen: </p>

<table>
<tr>
 <td style="text-align: left;"> <code>"info"</code> </td><td style="text-align: left;"> RAxML log file as character
 string</td>
</tr>
<tr>
 <td style="text-align: left;"> <code>"bestTree"</code> </td><td style="text-align: left;"> MLE of tree</td>
</tr>
<tr>
 <td style="text-align: left;"> <code>"bipartitions"</code> </td><td style="text-align: left;">
 MLE of tree annotated with bootstrap proportions</td>
</tr>
<tr>
 <td style="text-align: left;"> <code>"bootstrap"</code> </td><td style="text-align: left;">
 bootstrapped trees</td>
</tr>
<tr>
 <td style="text-align: left;"> </td>
</tr>

</table>



<h3>Note</h3>

<p>RAxML is a C program and the source code is not contained in this
package. This means that in order to run this function you will need to
install RAxML yourself. See
<a href="http://sco.h-its.org/exelixis/web/software/raxml/">http://sco.h-its.org/exelixis/web/software/raxml/</a> for the most recent
documentation and source code of RAxML. Depending on where you chose to
install RAxML, you need to adjust the <code>exec</code> argument.
</p>
<p><code>raxml</code> was last tested and running fine on Mac OS X with RAxML 8.0.29.
Please be aware that calling third-party software from within R is a
platform-specific process and I cannot guarantee that <code>raxml</code> will
behave properly on any system.
</p>


<h3>References</h3>

<p>(in chronolocigal order)
</p>
<p>Stamatakis, A., T. Ludwig and H. Meier. 2004. RAxML-III: A fast program for
maximum likelihood-based inference of large phylogenetic trees.
<em>Bioinformatics</em> <b>1</b>: 1&ndash;8.
</p>
<p>Stamatakis, A. 2006. RAxML-VI-HPC: Maximum likelihood-based phylogenetic
analyses with thousands of taxa and mixed models. <em>Bioinformatics</em>
<b>22</b>: 2688&ndash;2690.
</p>
<p>Stamatakis, A., P. Hoover, and J. Rougemont. 2008. A rapid bootstrap
algorithm for the RAxML web-servers. <em>Syst. Biol.</em> <b>75</b>: 758&ndash;771.
</p>
<p>Pattengale, N. D., M. Alipour, O. R. P. Bininda-Emonds, B. M. E. Moret, and
A. Stamatakis. 2010. How many bootstrap replicates are necessary?
<em>Journal of Computational Biology</em> <b>17</b>: 337-354.
</p>
<p>Stamatakis, A. 2014. RAxML Version 8: A tool for phylogenetic analysis and
post-analysis of large phylogenies. <em>Bioinformatics</em> Advance Access.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+raxml.partitions">raxml.partitions</a></code> to store partitioning information in a
data frame suitable for input as <code>partitions</code> argument in <code>raxml</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## bark beetle sequences
data(ips.cox1)
data(ips.16S)
data(ips.28S)

ips &lt;- cbind(ips.cox1, ips.16S, ips.28S,
            fill.with.gaps = TRUE)

exec &lt;- "/Applications/RAxML-code/standard-RAxML/raxmlHPC-PTHREADS-AVX"
w &lt;- sample(1:5, ncol(ips.cox1), replace = TRUE)

## Not run: 

# Simple tree search with GTRCAT and GTRGAMMA
tr &lt;- raxml(ips.cox1, f = "d", N = 2, p = 1234,
           exec = exec) # -1743.528461
tr &lt;- raxml(ips.cox1, m = "GTRGAMMA", f = "d", N = 2, p = 1234,
           exec = exec)
   
# Applying weights to columns                   
tr &lt;- raxml(ips.cox1, f = "d", N = 2, p = 1234,
           weights = w, exec = exec) # -1743.528461

# Rapid bootstrap
tr &lt;- raxml(ips.cox1, m = "GTRGAMMA",
           f = "a", N = 10, p = 1234, x = 1234,
           exec = exec)

# Rapid bootstrap with automatic halt
tr &lt;- raxml(ips.cox1, m = "GTRGAMMA",
           f = "a", N = "autoMRE", p = 1234, x = 1234,
           exec = exec)

## End(Not run)
</code></pre>

<hr>
<h2 id='raxml.partitions'>Partition scheme for RAxML
</h2><span id='topic+raxml.partitions'></span>

<h3>Description</h3>

<p>Given a set of DNA sequence alignments, <code>raxml.partitions</code> creates a data frame with partition bounderies that can be input into <code><a href="#topic+raxml">raxml</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>raxml.partitions(...)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="raxml.partitions_+3A_...">...</code></td>
<td>
<p>Two or more DNA sequence alignments of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>raxml.partitions</code> to make sense, the DNA sequence alignments must be given exactly in the same order in which they are concatenated into a supermatrix (see Examples section). Without any testing, the type of sequences is supposed to be DNA.
</p>


<h3>Value</h3>

<p>A data frame with four columns (<code>type</code>, <code>locus</code>, <code>begin</code>, and <code>end</code>) and number of rows corresponding to the number of partitions.
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+cbind.DNAbin">cbind.DNAbin</a></code> to concatenate multiple alignments; <code><a href="#topic+raxml">raxml</a></code> for an interface to RAxML.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## bark beetle sequences
data(ips.cox1)
data(ips.16S)
data(ips.28S)

## Note the same order of individual
## alignments in both functions:
## ----------------------------
raxml.partitions(cox1 = ips.cox1, 
                 r16S = ips.16S, 
                 r28S = ips.28S)

cbind(ips.cox1, ips.16S, ips.28S,
  fill.with.gaps = TRUE)
</code></pre>

<hr>
<h2 id='rbeauti'>XML Input Files for BEAST</h2><span id='topic+rbeauti'></span>

<h3>Description</h3>

<p>This function is intended to prepare XML files for BEAST with R. BEAST uses an MCMC approach to estimate rooted phylogenies from molecular data (Drummond &amp; Rambaut, 2007). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rbeauti(..., file, template = "standard", taxonset)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rbeauti_+3A_...">...</code></td>
<td>
<p>one or more object(s) of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.</p>
</td></tr>
<tr><td><code id="rbeauti_+3A_file">file</code></td>
<td>
<p>A connection, or a character string naming the file to write to. If left empty the XML tree will be printed to the screen (see Examples).</p>
</td></tr>
<tr><td><code id="rbeauti_+3A_template">template</code></td>
<td>
<p><em>Currently unused.</em></p>
</td></tr>
<tr><td><code id="rbeauti_+3A_taxonset">taxonset</code></td>
<td>
<p>A list containing one or more taxon sets.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>rbeauti</code> has been completely rewritten to work with <b>BEAST 2</b>. Currently <code>rbeauti</code> offers few options, because the idea is not to create ready-to-use XML file. That can be done conveniently with <b>BEAUti</b> (the BEAST package's genuine XML generator). Instead, <code>rbeauti</code> is intended to make the definition of large numbers of taxon sets easy. The creation of taxon sets can be done via R scripts and the resulting XML files can be further modified with BEAUti.
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>The BEAST 2 website: <a href="http://beast.bio.ed.ac.uk/BEAST_v1.5.x_XML_Reference">http://beast.bio.ed.ac.uk/BEAST_v1.5.x_XML_Reference</a>	
</p>
<p>Drummond, A.J. &amp; A. Rambaut. 2007. BEAST: Bayesian evolutionary analysis by sampling trees. <em>BMC Evolutionary Biology</em> <b>7</b>: 240.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.beast">read.beast</a></code>, <code><a href="#topic+read.beast.table">read.beast.table</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.16S)

## define taxon sets
spec &lt;- rownames(ips.16S)
ingroup &lt;- spec[grep("Ips|Orthomotomicus", spec)]
outgroup &lt;- spec[grep("Pityogenes", spec)]
ts &lt;- list(list(id = "ingroup", taxon = ingroup),
           list(id = "outgroup", taxon = outgroup))

## print XML file to screen
rbeauti(ips.16S, taxonset = ts)
</code></pre>

<hr>
<h2 id='read'>Reading Sequence Files</h2><span id='topic+read'></span><span id='topic+read.fas'></span><span id='topic+read.nex'></span><span id='topic+read.phy'></span>

<h3>Description</h3>

<p>Read DNA and amino acid sequences from FASTA, PHILIP, and NEXUS 
formatted files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.fas(x, text)

read.nex(x)

read.phy(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_+3A_x">x</code></td>
<td>
<p>A character string, giving the file name.</p>
</td></tr>
<tr><td><code id="read_+3A_text">text</code></td>
<td>
<p>A character string in FASTA format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An matrix (aligned sequences)  or list (unaligned sequences) of class
<code>DNAbin</code> or <code>AAbin</code>.
</p>


<h3>References</h3>

<p>Maddison, D.R., D.L. Swofford, and W.P. Maddison. 1997. NEXUS: an
extensible file format for systematic information. <em>Syst. Biol.</em> 
<b>46</b>: 590-621.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mafft">mafft</a></code> and <code><a href="#topic+prank">prank</a></code> for sequence alignment,
<code><a href="#topic+gblocks">gblocks</a></code> and <code><a href="#topic+aliscore">aliscore</a></code> for quality check and
cleaning of sequence alignments, <code><a href="ape.html#topic+cbind.DNAbin">cbind.DNAbin</a></code> for
concatenation of sequence alignments.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## bark beetle COX1 sequences
data(ips.cox1)
## create temporary file names
format &lt;- c(".fas", ".phy", ".nex")
fn &lt;- sapply(format, tempfile, 
             pattern = "ips", tmpdir = tempdir())
## write sequences files
write.fas(ips.cox1, fn[".fas"])
write.phy(ips.cox1, fn[".phy"])
write.nex(ips.cox1, fn[".nex"])
## read sequence files
fas &lt;- read.fas(fn[".fas"])
phy &lt;- read.phy(fn[".phy"])
nex &lt;- read.nex(fn[".nex"])
## remove sequence files
unlink(fn)
</code></pre>

<hr>
<h2 id='read.beast'>Read Bayesian Trees</h2><span id='topic+read.mrbayes'></span><span id='topic+read.beast'></span><span id='topic+read.starbeast'></span>

<h3>Description</h3>

<p>Thess functions parse chronograms in NEXUS format as produced by TreeAnnotator or output by MrBayes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.mrbayes(file, digits = NULL)

read.beast(file, digits = NULL)

read.starbeast(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.beast_+3A_file">file</code></td>
<td>
<p>A character string giving the input file, which must be a TreeAnnotator-generated chronogram in NEXUS format.</p>
</td></tr>
<tr><td><code id="read.beast_+3A_digits">digits</code></td>
<td>
<p>NULL or integer, if <code>!is.null(digits)</code> values are rounded to the given integer.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>phylo</code>
</p>


<h3>Note</h3>

<p><code>read.starbeast</code> currently parses only skalars and ranges; node statistics with more than two values will be deleted and a warning message will be issued. Future version of <code>read.starbeast</code> will hopefully be able to append list or data frames to <code>phylo</code> objects. If you have any opinion or wishes regarding the question of how this exactly should be managed, send me a message.</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>TreeAnnotator: <a href="http://beast.bio.ed.ac.uk/TreeAnnotator">http://beast.bio.ed.ac.uk/TreeAnnotator</a>
</p>
<p>Metacomments in NEXUS: <a href="http://code.google.com/p/beast-mcmc/wiki/NexusMetacommentFormat">http://code.google.com/p/beast-mcmc/wiki/NexusMetacommentFormat</a>	
</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.beast.table">read.beast.table</a></code> to extract internal node data from NEXUS file, <code><a href="#topic+rbeauti">rbeauti</a></code> to create XML input for BEAST. <code>HPDbars</code> for plotting highest posterior densities on phylogenies has been moved to package <b>viper</b>.
</p>

<hr>
<h2 id='read.beast.table'>
Extract node data from BEAST chronogram
</h2><span id='topic+read.beast.table'></span>

<h3>Description</h3>

<p>This function reads a BEAST chronogram such as produced by TreeAnnotator and extracts time, rate, and support values for internal and external nodes. Nodes in the resulting data frame are ordered exactly like in the NEXUS file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.beast.table(file, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.beast.table_+3A_file">file</code></td>
<td>
<p>character string giving the input file, which must be a TreeAnnotaror-generated chronogram in NEXUS format</p>
</td></tr>
<tr><td><code id="read.beast.table_+3A_digits">digits</code></td>
<td>
<p>NULL or integer, if <code>!is.null(digits)</code> values are rounded to the given integer</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix; each row corresponds to an internal node, the (ape!)number of which is given in the first column; the remaining columns list the node values extracted from the chronogram.
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.beast">read.beast</a></code> to parse TreeAnnotator output, <code><a href="#topic+rbeauti">rbeauti</a></code> to create XML input for BEAST. <code>HPDbars</code> for plotting highest posterior densities on phylogenies has been moved to package <b>viper</b>.
</p>

<hr>
<h2 id='sister'>Identification of Sister Nodes and Clades</h2><span id='topic+sister'></span>

<h3>Description</h3>

<p>For any given internal node in a phylogeny, this function 
returns the sister clade.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sister(phy, node, type = "terminal", label = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sister_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="sister_+3A_node">node</code></td>
<td>
<p>A vector of mode <code>"numeric"</code> or <code>"character"</code> giving 
the number(s) or name(s) of the tiplabel(s); these must be monophyletic.</p>
</td></tr>
<tr><td><code id="sister_+3A_type">type</code></td>
<td>
<p>A character string, may be <code>"terminal"</code>, <code>"internal"</code>, 
<code>"daughter"</code>, <code>"all"</code>, or any unambiguous abbreviation of these; 
<code>"daughter"</code> will return the MRCA of the sister clade of 
<code>"node"</code>.</p>
</td></tr>
<tr><td><code id="sister_+3A_label">label</code></td>
<td>
<p>Logical, determining if tip number or tip labels will be 
returned.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of mode <code>"numeric"</code> or <code>"character"</code>, containing 
either tip numbers or labels, respectively.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+descendants">descendants</a></code>, <code><a href="#topic+noi">noi</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># A phylogeny of bark beetles ...
data(ips.tree)
tcol &lt;- rep("black", Ntip(ips.tree))
tcol[ips.tree$tip.label %in% c("Ips_typographus", "Ips_nitidus")] &lt;- "blue"
tcol[ips.tree$tip.label %in% c("Ips_duplicatus")] &lt;- "red"
plot(ips.tree, no.margin = TRUE, tip.color = tcol)
# What is the sister species of Ips typographus?
sister(ips.tree, "Ips_typographus", label = TRUE)
# Return the MRCA of the sister clade of Ips duplicatus
x &lt;- sister(ips.tree, "Ips_duplicatus", "daughter")
nodelabels(node = x, pch = 21, bg = "red")
</code></pre>

<hr>
<h2 id='splitIntoClades'>Find Monophyletic Subsets in Species Lists</h2><span id='topic+splitIntoClades'></span>

<h3>Description</h3>

<p>Takes a phylogeny and a subset of its tiplabels
and splits the list of tiplabels into monophyletic groups 
(clades).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitIntoClades(phy, tips)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="splitIntoClades_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="splitIntoClades_+3A_tips">tips</code></td>
<td>
<p>A vector of mode <code>"character"</code> containing any 
subset of the tiplabels in <code>phy</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list.
</p>

<hr>
<h2 id='terminalSisters'>Find Pairs of Sister Species</h2><span id='topic+terminalSisters'></span>

<h3>Description</h3>

<p>Finds pairs of sister species in
a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>terminalSisters(phy, labels = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="terminalSisters_+3A_phy">phy</code></td>
<td>
<p>An object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
<tr><td><code id="terminalSisters_+3A_labels">labels</code></td>
<td>
<p>Logical, indicating whether to return tip labels or tip numbers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of which each element contains the tip labels
of a sister species pair.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
tr &lt;- rtree(12)
plot(tr)
terminalSisters(tr)
</code></pre>

<hr>
<h2 id='tipHeights'>
Tip Heights in a Phylogenetic Tree
</h2><span id='topic+tipHeights'></span>

<h3>Description</h3>

<p>For each tip (leave, terminal node) in the phylogenetic tree the edge lengths (branch lengths) from root to tip, be it units of time or divergence, is summed up.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tipHeights(phy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tipHeights_+3A_phy">phy</code></td>
<td>
<p>an object of class <code><a href="ape.html#topic+phylo">phylo</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a numeric vector with distances from root to tip for each tip in the phylogenetic tree.
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+branching.times">branching.times</a></code>
</p>

<hr>
<h2 id='traitRate'>Trait-Dependent Shifts in Molecular Rate</h2><span id='topic+traitRate'></span>

<h3>Description</h3>

<p>Detection of trait-dependent shifts in the rate of molecular evolution with <b>traitRate</b> (Mayrose &amp; Otto, 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traitRate(phy, seq, x, mainType = "Optimize_Model", 
   n, charModelParam1 = 0.5, charModelParam2 = 1, 
   gammaParam = 0.5, seqModelParam1 = 2,
   exec = "/Applications/traitRate-1.1/programs/traitRate")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traitRate_+3A_phy">phy</code></td>
<td>
<p>a ultrametric phylogenetic tree of class <code><a href="ape.html#topic+phylo">phylo</a>.</code>
</p>
</td></tr>
<tr><td><code id="traitRate_+3A_seq">seq</code></td>
<td>
<p>a multiple sequence alignment of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code>.
</p>
</td></tr>
<tr><td><code id="traitRate_+3A_x">x</code></td>
<td>
<p>data frame containing a binary character in the first column.
</p>
</td></tr>
<tr><td><code id="traitRate_+3A_maintype">mainType</code></td>
<td>
<p>character string giving the type of analysis; two choices are possible: <code>"Optimize_Model"</code> will produce MLE of parameters and <code>"runTraitBootstrap"</code> will perform a parametric bootstrap analysis.</p>
</td></tr>
<tr><td><code id="traitRate_+3A_n">n</code></td>
<td>
<p>numeric, the number of bootstrap replicates. Will be ignored if <code>mainType = "Optimize_Model"</code>.</p>
</td></tr>
<tr><td><code id="traitRate_+3A_charmodelparam1">charModelParam1</code></td>
<td>
<p>numeric, giving an initial value for the rate of transitions of character state 0 to 1.</p>
</td></tr>
<tr><td><code id="traitRate_+3A_charmodelparam2">charModelParam2</code></td>
<td>
<p>numeric, giving an initial value for the rate of transitions of character state 1 to 0.</p>
</td></tr>
<tr><td><code id="traitRate_+3A_gammaparam">gammaParam</code></td>
<td>
<p>numeric, giving an initial value for the <b>alpha</b> parameter of the model of sequence evolution.</p>
</td></tr>
<tr><td><code id="traitRate_+3A_seqmodelparam1">seqModelParam1</code></td>
<td>
<p>numeric, giving an initial value for the <b>kappa</b> parameter of the model of sequence evolution.</p>
</td></tr>
<tr><td><code id="traitRate_+3A_exec">exec</code></td>
<td>
<p>character string giving the path to the program directory.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Currently none, but look for the output files in the <code>'RESULTS'</code> subdirectory in the current working directory.
</p>


<h3>Note</h3>

<p>This function is under development!
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl
</p>


<h3>References</h3>

<p>Mayrose, I. &amp; S.P. Otto. 2011. A likelihood method for detecting trait-dependent shifts in the rate of molecular evolution. <em>Mol. Biol. Evol.</em> <b>28</b>: 759-770
</p>


<h3>See Also</h3>

<p><code><a href="ape.html#topic+read.tree">read.tree</a></code> for reading phylogenetic trees, <code><a href="#topic+read.fas">read.fas</a></code> for reading multiple sequence alignments in FASTA format.
</p>

<hr>
<h2 id='trimEnds'>Trim Alignment Ends</h2><span id='topic+trimEnds'></span>

<h3>Description</h3>

<p>Trims both ends of a DNA sequence alignment to the first and
last alignment positions that contain a minimum number of IUPAC base
characters (<code>"a"</code>, <code>"c"</code>, <code>"g"</code>, <code>"t"</code>, <code>"r"</code>,
<code>"y"</code>, <code>"s"</code>, <code>"w"</code>, <code>"k"</code>, <code>"m"</code>, <code>"b"</code>,
<code>"d"</code>, <code>"h"</code>, <code>"v"</code>). In addition, all gap characters
(<code>"-"</code>) beyond the first and last base characters of each sequence are
replaced by the  character <code>"n"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trimEnds(x, min.n.seq = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trimEnds_+3A_x">x</code></td>
<td>
<p>An object of class <code>DNAbin</code>.</p>
</td></tr>
<tr><td><code id="trimEnds_+3A_min.n.seq">min.n.seq</code></td>
<td>
<p>A <code>numeric</code> giving the required minimum number of
sequences having an non-ambiguous base character (a, c, g, t) in the first
and last position of the alignment; defaults to <code>4</code>, which is the
minimum number of sequences needed to produce a non-trivial unrooted
topology. Can also be given as a fraction.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>DNAbin</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+deleteEmptyCells">deleteEmptyCells</a></code>, <code><a href="#topic+deleteGaps">deleteGaps</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simple example alignment:
x &lt;- structure(list(nb = 5, seq = c("acaaggtaca", "-caaggtac-",
"acaaggtaca", "aca--gtaca", "-ccaggta--"), nam = LETTERS[1:5]),
.Names = c("nb", "seq", "nam"), class = "alignment")
# convert to DNAbin:
x &lt;- as.DNAbin(x)
# fill missing nucleotides:
x &lt;- trimEnds(x)
# show results:
as.character(x[2, ])
</code></pre>

<hr>
<h2 id='unlistFirstLevel'>Unlist To First Level Only</h2><span id='topic+unlistFirstLevel'></span>

<h3>Description</h3>

<p>Does the same as <code>unlist</code>, but recurses only
one level.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unlistFirstLevel(z, use.names = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unlistFirstLevel_+3A_z">z</code></td>
<td>
<p>A list of lists.</p>
</td></tr>
<tr><td><code id="unlistFirstLevel_+3A_use.names">use.names</code></td>
<td>
<p>Logical, indicating if element names from the
element should be preserved.</p>
</td></tr>
</table>

<hr>
<h2 id='write.fas'>Write DNA Sequences to File</h2><span id='topic+write.fas'></span><span id='topic+write.phy'></span><span id='topic+write.nex'></span>

<h3>Description</h3>

<p>Write DNA sequences and morphological data to FASTA, PHYLIP, or NEXUS formatted files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	
write.fas(x, file, block.width = FALSE, 
    truncate = FALSE, append = FALSE)

write.phy(x, file, block.width = FALSE, 
    strict = FALSE)

write.nex(x, file, block.width = 60, 
    taxblock = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.fas_+3A_x">x</code></td>
<td>

<p>an object of class <code><a href="ape.html#topic+DNAbin">DNAbin</a></code> (usually as matrix, but <code>write.fas</code> also accepts lists) or a list of objects of class <code>DNAbin</code> (only <code>write.nex</code>) or a data frame containing standard (morphological, etc.) data (only <code>write.nex</code>).
</p>
</td></tr>
<tr><td><code id="write.fas_+3A_file">file</code></td>
<td>

<p>a character string giving the filename; a special case is <code>file = ""</code>, which causes the file content to be written on the standard output connection (i.e. the console). If <code>file</code> is left unspecified (default), the file content is returned as a vector of mode <code>"character"</code> and can be used as a building block for more complex data files.
</p>
</td></tr>
<tr><td><code id="write.fas_+3A_block.width">block.width</code></td>
<td>

<p>an integer, giving the number of characters per line.
</p>
</td></tr>
<tr><td><code id="write.fas_+3A_truncate">truncate</code></td>
<td>

<p>truncation of taxon names to the number of characters given as a integer, otherwise (default) taxon names will not be changed.
</p>
</td></tr>
<tr><td><code id="write.fas_+3A_append">append</code></td>
<td>

<p>logical, if <code>TRUE</code> the sequences will be appended to <code>file</code> (if it exists).
</p>
</td></tr>
<tr><td><code id="write.fas_+3A_strict">strict</code></td>
<td>

<p>logical, if <code>TRUE</code> the names of the sequences will be truncated to 10 strings.
</p>
</td></tr>
<tr><td><code id="write.fas_+3A_taxblock">taxblock</code></td>
<td>

<p>logical, if <code>TRUE</code>, a tax block will be added to the NEXUS file.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>write.nex</code> can handle multiple DNA sequence alignments, which are handed over as a list of objects of class <code>DNAbin</code>. Correct matching of the rows in the alignments is cared for automatically, hence the individual alignments can contain different numbers of samples and samples need not be in the same order.
</p>


<h3>Value</h3>

<p>None, except when called with <code>file</code> left unspecified, which causes the file content to be returned as a vector of mode <code>"character"</code>. This is particularly useful for constructing special types of input files, e.g. for MrBayes (<code><a href="#topic+mrbayes">mrbayes</a></code>).
</p>


<h3>Author(s)</h3>

<p>Christoph Heibl</p>


<h3>References</h3>

<p>Maddison, D.R., D.L. Swofford, and W.P. Maddison. 1997. NEXUS: an extensible file format for systematic information. <em>Syst. Biol.</em> <b>46</b>: 590-621.</p>


<h3>See Also</h3>

<p><code><a href="#topic+read.fas">read.fas</a></code>, <code><a href="#topic+read.phy">read.phy</a></code>, and <code><a href="#topic+read.nex">read.nex</a></code> for reading of DNA sequence files.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ips.cox1)
data(ips.28S)

## Examples for FASTA files
## ------------------------
write.fas(ips.cox1[1:5, 1:120], block.width = 60)

## Examples for PHYLIP files
## ------------------------
write.phy(ips.cox1[1:5, 1:20], block.width = 40)

## Examples for NEXUS files
## ------------------------
x &lt;- list(cox1 = ips.cox1[1:5, 1:10], 
          rna28S = ips.28S[1:5, 1:30])
write.nex(x, block.width = 20)

# Truncation of taxonnames:
# -------------------------
rownames(ips.cox1)[1] &lt;- "AVeeeeeeeeeeeeeeeeeryLongName"
write.fas(ips.cox1, truncate = 10)

# If truncation leads to identical taxonnames,
# a warning will be issued:
# -------------------------
rownames(ips.cox1)[1:2] &lt;- "AVeeeeeeeeeeeeeeeeeryLongName"
write.fas(ips.cox1, truncate = 10)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
