<!DOCTYPE html><html><head><title>Help for package iRafNet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iRafNet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#iRafNet'><p>Integrative random forest for gene regulatory network inference</p></a></li>
<li><a href='#iRafNet_network'><p>Compute permutation-based FDR of importance scores and return estimated regulations.</p></a></li>
<li><a href='#iRafNet_permutation'><p>Derive importance scores for one permuted data.</p></a></li>
<li><a href='#roc_curve'><p>Plot receiver operating characteristic (ROC) curve for weighted network generated by iRafNet</p></a></li>
<li><a href='#Run_permutation'><p>Derive importance scores for M permuted data sets.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.1-1</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-10-26</td>
</tr>
<tr>
<td>Title:</td>
<td>Integrative Random Forest for Gene Regulatory Network Inference</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ROCR</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a flexible integrative algorithm that allows information from prior data, such as protein protein interactions and gene knock-down, to be jointly considered for gene regulatory network inference.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-10-26 00:46:40 UTC; petraf01</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Francesca Petralia [aut, cre],
  Pei Wang [aut],
  Zhidong Tu [aut],
  Jialiang Yang [aut],
  Adele Cutler [ctb],
  Leo Breiman [ctb],
  Andy Liaw [ctb],
  Matthew Wiener [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francesca Petralia &lt;francesca.petralia@mssm.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-10-26 10:37:49</td>
</tr>
</table>
<hr>
<h2 id='iRafNet'>Integrative random forest for gene regulatory network inference</h2><span id='topic+iRafNet'></span>

<h3>Description</h3>

<p>This function fits iRafNet, a flexible unified integrative algorithm that allows information from prior data, such as protein-protein interactions and gene knock-down, to be jointly considered for gene regulatory network inference. This function takes as input only one set of sampling scores, computed considering one prior data such as protein-protein interactions or gene expression from knock-out experiments. Note that some of the functions utilized are a modified version of functions contained in the R package randomForest (A. Liaw and M. Wiener, 2002). </p>


<h3>Usage</h3>

<pre><code class='language-R'>iRafNet(X, W, ntree, mtry,genes.name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iRafNet_+3A_x">X</code></td>
<td>
<p><code>(n x p)</code> Matrix containing expression levels for <code>n</code> samples and <code>p</code> genes.</p>
</td></tr>
<tr><td><code id="iRafNet_+3A_w">W</code></td>
<td>
<p><code>(p x p)</code> Matrix containing iRafNet sampling scores. Element <code>(i,j)</code> contains score for regulatory relationship <code>(i -&gt; j)</code>. Scores must be non-negative. Larger value of sampling score corresponds to higher likelihood of gene <code>i</code> regulating gene <code>j</code>. Columns and rows of <code>W</code> must be in the same order as the columns of <code>X</code>. Sampling scores <code>W</code> are computed considering one prior data such as protein-protein interactions or gene expression from knock-out experiments.</p>
</td></tr>
<tr><td><code id="iRafNet_+3A_ntree">ntree</code></td>
<td>
<p>Numeric value: number of trees.</p>
</td></tr>
<tr><td><code id="iRafNet_+3A_mtry">mtry</code></td>
<td>
<p>Numeric value: number of potential regulators to be sampled at each tree node.</p>
</td></tr>
<tr><td><code id="iRafNet_+3A_genes.name">genes.name</code></td>
<td>
<p>Vector containing gene names. The order needs to match the columns of <code>X</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Importance score for each regulatory relationship. The first column contains gene name of regulators, the second column contains gene name of targets, and third column contains corresponding importance scores.
</p>


<h3>References</h3>

<p>Petralia, F., Wang, P., Yang, J., Tu, Z. (2015) Integrative random forest for gene regulatory network inference, <em>Bioinformatics</em>, <b>31</b>, i197-i205.
</p>
<p>A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. <em>R News</em> <b>2</b>, 18&ndash;22.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # --- Generate data sets
  n&lt;-20                  # sample size
  p&lt;-5                   # number of genes
  genes.name&lt;-paste("G",seq(1,p),sep="")   # genes name
  data&lt;-matrix(rnorm(p*n),n,p)      # generate expression matrix
  W&lt;-abs(matrix(rnorm(p*p),p,p))    # generate weights for regulatory relationships
 
  # --- Standardize variables to mean 0 and variance 1
  data &lt;- (apply(data, 2, function(x) { (x - mean(x)) / sd(x) } ))

  # --- Run iRafNet and obtain importance score of regulatory relationships
  out&lt;-iRafNet(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name)

</code></pre>

<hr>
<h2 id='iRafNet_network'>Compute permutation-based FDR of importance scores and return estimated regulations.</h2><span id='topic+iRafNet_network'></span>

<h3>Description</h3>

<p>This function computes permutation-based FDR of importance scores and returns gene-gene regulations.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iRafNet_network(out.iRafNet,out.perm,TH)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iRafNet_network_+3A_out.irafnet">out.iRafNet</code></td>
<td>
<p>Output object from function <code>iRafNet</code>.</p>
</td></tr>
<tr><td><code id="iRafNet_network_+3A_out.perm">out.perm</code></td>
<td>
<p>Output object from function <code>Run_permutation.</code></p>
</td></tr>
<tr><td><code id="iRafNet_network_+3A_th">TH</code></td>
<td>
<p>Threshold for FDR.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of estimated regulations. 
</p>


<h3>References</h3>

<p>Petralia, F., Song, W.M., Tu, Z. and Wang, P. (2016). New method for joint network analysis reveals common and different coexpression patterns among genes and proteins in breast cancer. <em>Journal of proteome research</em>, <b>15</b>(3), pp.743-754.
</p>
<p>A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. <em>R News</em> <b>2</b>, 18&ndash;22.
</p>
<p>Xie, Y., Pan, W. and Khodursky, A.B., 2005. A note on using permutation-based false discovery rate estimates to compare different analysis methods for microarray data. <em>Bioinformatics</em>, <b>21</b>(23), pp.4280-4288.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # --- Generate data sets
  n&lt;-20           # sample size 
  p&lt;-5            # number of genes
  genes.name&lt;-paste("G",seq(1,p),sep="")   # genes name
  M=5;            # number of permutations
  data&lt;-matrix(rnorm(p*n),n,p)       # generate gene expression matrix
  data[,1]&lt;-data[,2]                 # var 1 and var 2 interact
  W&lt;-abs(matrix(rnorm(p*p),p,p))     # generate weights for regulatory relationships
  
  # --- Standardize variables to mean 0 and variance 1
  data &lt;- (apply(data, 2, function(x) { (x - mean(x)) / sd(x) } ))

  # --- Run iRafNet and obtain importance score of regulatory relationships
  out.iRafNet&lt;-iRafNet(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name)

  # --- Run iRafNet for M permuted data sets
  out.perm&lt;-Run_permutation(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name,M)

  # --- Derive final networks
  final.net&lt;-iRafNet_network(out.iRafNet,out.perm,0.001)

</code></pre>

<hr>
<h2 id='iRafNet_permutation'>Derive importance scores for one permuted data.</h2><span id='topic+iRafNet_permutation'></span>

<h3>Description</h3>

<p>This function computes importance score for one permuted data set. Sample labels of target genes are randomly permuted and iRafNet is implemented. Resulting importance scores can be used to derive an estimate of FDR.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iRafNet_permutation(X, W, ntree, mtry,genes.name,perm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iRafNet_permutation_+3A_x">X</code></td>
<td>
<p><code>(n x p)</code> Matrix containing expression levels for <code>n</code> samples and <code>p</code> genes.</p>
</td></tr>
<tr><td><code id="iRafNet_permutation_+3A_w">W</code></td>
<td>
<p><code>(p x p)</code> Matrix containing iRafNet sampling scores. Element <code>(i,j)</code> contains score for regulatory relationship <code>(i -&gt; j)</code>. Scores must be non-negative. Larger value of sampling score corresponds to higher likelihood of gene <code>i</code> regulating gene <code>j</code>. Columns and rows of <code>W</code> must be in the same order as the columns of <code>X</code>. Sampling scores <code>W</code> are computed considering one prior data such as protein-protein interactions or gene expression from knock-out experiments.</p>
</td></tr>
<tr><td><code id="iRafNet_permutation_+3A_ntree">ntree</code></td>
<td>
<p>Numeric value: number of trees.</p>
</td></tr>
<tr><td><code id="iRafNet_permutation_+3A_mtry">mtry</code></td>
<td>
<p>Numeric value: number of predictors to be sampled at each node.</p>
</td></tr>
<tr><td><code id="iRafNet_permutation_+3A_genes.name">genes.name</code></td>
<td>
<p>Vector containing genes name. The order needs to match the rows of <code>x_j</code>.</p>
</td></tr>
<tr><td><code id="iRafNet_permutation_+3A_perm">perm</code></td>
<td>
<p>Integer: seed for permutation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing importance score for permuted data. 
</p>


<h3>References</h3>

<p>Petralia, F., Wang, P., Yang, J., Tu, Z. (2015) Integrative random forest for gene regulatory network inference, <em>Bioinformatics</em>, <b>31</b>, i197-i205.
</p>
<p>Petralia, F., Song, W.M., Tu, Z. and Wang, P. (2016). New method for joint network analysis reveals common and different coexpression patterns among genes and proteins in breast cancer. <em>Journal of proteome research</em>, <b>15</b>(3), pp.743-754.
</p>
<p>A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. <em>R News</em> <b>2</b>, 18&ndash;22.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # --- Generate data sets
  n&lt;-20                  # sample size
  p&lt;-5                   # number of genes
  genes.name&lt;-paste("G",seq(1,p),sep="")   # genes name
  data&lt;-matrix(rnorm(p*n),n,p)       # generate expression matrix
  W&lt;-abs(matrix(rnorm(p*p),p,p))     # generate weights for regulatory relationships
 
  # --- Standardize variables to mean 0 and variance 1
  data &lt;- (apply(data, 2, function(x) { (x - mean(x)) / sd(x) } ))

  # --- Run iRafNet and obtain importance score of regulatory relationships
  out.iRafNet&lt;-iRafNet(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name)

  # --- Run iRafNet for one permuted data set and obtain importance scores
  out.perm&lt;-iRafNet_permutation(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name,perm=1)

</code></pre>

<hr>
<h2 id='roc_curve'>Plot receiver operating characteristic (ROC) curve for weighted network generated by iRafNet</h2><span id='topic+roc_curve'></span>

<h3>Description</h3>

<p>This function uses R package ROCR to plot ROC curves for iRafNet object. </p>


<h3>Usage</h3>

<pre><code class='language-R'>roc_curve(out, truth)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="roc_curve_+3A_out">out</code></td>
<td>
<p>Output from iRafNet.</p>
</td></tr>
<tr><td><code id="roc_curve_+3A_truth">truth</code></td>
<td>
<p>Matrix of true regulations. Rows correspond to different regulations and match rows of <code>out</code>. First column contains name of regulators, second column contains name of targets and third column contains a binary variable equal 1 in case of regulation and 0 otherwise.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plot ROC curve and return area under ROC curve.
</p>


<h3>References</h3>

<p>Petralia, F., Wang, P., Yang, J., Tu, Z. (2015) Integrative random forest for gene regulatory network inference, <em>Bioinformatics</em>, <b>31</b>, i197-i205.
</p>
<p>Sing, Tobias, et al. (2005) ROCR: visualizing classifier performance in R, <em>Bioinformatics</em>, <b>21</b>, 3940-3941.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # --- Generate data sets
  n&lt;-20                  # sample size
  p&lt;-5                   # number of genes
  genes.name&lt;-paste("G",seq(1,p),sep="")   # genes name
  data&lt;-matrix(rnorm(p*n),n,p)    # generate expression matrix
  data[,1]&lt;-data[,2]              # var 1 and 2 interact
  W&lt;-abs(matrix(rnorm(p*p),p,p))  # generate score for regulatory relationships
 
  # --- Standardize variables to mean 0 and variance 1
  data &lt;- (apply(data, 2, function(x) { (x - mean(x)) / sd(x) } ))

  # --- Run iRafNet and obtain importance score of regulatory relationships
  out&lt;-iRafNet(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name)
  
  # --- Matrix of true regulations
  truth&lt;-out[,seq(1,2)]
  truth&lt;-cbind(as.character(truth[,1]),as.character(truth[,2])
  ,as.data.frame(rep(0,,dim(out)[1])));
  truth[(truth[,1]=="G2" &amp; truth[,2]=="G1") | (truth[,1]=="G1" &amp; truth[,2]=="G2"),3]&lt;-1 

  # --- Plot ROC curve and compute AUC
  auc&lt;-roc_curve(out,truth)

</code></pre>

<hr>
<h2 id='Run_permutation'>Derive importance scores for M permuted data sets.</h2><span id='topic+Run_permutation'></span>

<h3>Description</h3>

<p>This function computes importance score for <code>M</code> permuted data sets. Sample labels of target genes are randomly permuted and iRafNet is implemented. Resulting importance scores can be used to derive an estimate of FDR.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Run_permutation(X, W, ntree, mtry,genes.name,M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Run_permutation_+3A_x">X</code></td>
<td>
<p><code>(n x p)</code> Matrix containing expression levels for <code>n</code> samples and <code>p</code> genes.</p>
</td></tr>
<tr><td><code id="Run_permutation_+3A_w">W</code></td>
<td>
<p><code>(p x p)</code> Matrix containing iRafNet sampling scores. Element <code>(i,j)</code> contains score for regulatory relationship <code>(i -&gt; j)</code>. Scores must be non-negative. Larger value of sampling score corresponds to higher likelihood of gene <code>i</code> regulating gene <code>j</code>. Columns and rows of <code>W</code> must be in the same order as the columns of <code>X</code>. Sampling scores <code>W</code> are computed considering one prior data such as protein-protein interactions or gene expression from knock-out experiments.</p>
</td></tr>
<tr><td><code id="Run_permutation_+3A_ntree">ntree</code></td>
<td>
<p>Numeric value: number of trees.</p>
</td></tr>
<tr><td><code id="Run_permutation_+3A_mtry">mtry</code></td>
<td>
<p>Numeric value: number of predictors to be sampled at each node.</p>
</td></tr>
<tr><td><code id="Run_permutation_+3A_genes.name">genes.name</code></td>
<td>
<p>Vector containing genes name. The order needs to match the rows of <code>x_j</code>.</p>
</td></tr>
<tr><td><code id="Run_permutation_+3A_m">M</code></td>
<td>
<p>Integer: total number of permutations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with <code>I</code> rows and <code>M</code> columns with <code>I</code> being the total number of regulations and <code>M</code> the number of permutations. Element <code>(i,j)</code> corresponds to the importance score of interaction <code>i</code> for permuted data <code>j</code>. 
</p>


<h3>References</h3>

<p>Petralia, F., Wang, P., Yang, J., Tu, Z. (2015) Integrative random forest for gene regulatory network inference, <em>Bioinformatics</em>, <b>31</b>, i197-i205.
</p>
<p>Petralia, F., Song, W.M., Tu, Z. and Wang, P. (2016). New method for joint network analysis reveals common and different coexpression patterns among genes and proteins in breast cancer. <em>Journal of proteome research</em>, <b>15</b>(3), pp.743-754.
</p>
<p>A. Liaw and M. Wiener (2002). Classification and Regression by randomForest. <em>R News</em> <b>2</b>, 18&ndash;22.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  # --- Generate data sets
  n&lt;-20                  # sample size 
  p&lt;-5                   # number of genes
  genes.name&lt;-paste("G",seq(1,p),sep="")   # genes name
  M=5;            # number of permutations
 
  data&lt;-matrix(rnorm(p*n),n,p)       # generate expression matrix
  W&lt;-abs(matrix(rnorm(p*p),p,p))          # generate score for regulatory relationships
 
  # --- Standardize variables to mean 0 and variance 1
  data &lt;- (apply(data, 2, function(x) { (x - mean(x)) / sd(x) } ))

  # --- Run iRafNet and obtain importance score of regulatory relationships
  out.iRafNet&lt;-iRafNet(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name)

  # --- Run iRafNet for M permuted data sets
  out.perm&lt;-Run_permutation(data,W,mtry=round(sqrt(p-1)),ntree=1000,genes.name,M)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
