<!DOCTYPE html><html lang="en"><head><title>Help for package ISOpureR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {ISOpureR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ISOpure.calculate.tac'><p>Perform calculation for Tumour Adjacent Cell (TAC) profiles</p></a></li>
<li><a href='#ISOpure.model_optimize.cg_code.rminimize'><p>Minimize a differentiable multivariate function</p></a></li>
<li><a href='#ISOpure.model_optimize.vv.vv_deriv_loglikelihood'><p>Compute the derivative of the loglikelihood relevant to vv for step 1</p></a></li>
<li><a href='#ISOpure.model_optimize.vv.vv_loglikelihood'><p>Compute the loglikelihood relevant to vv for step 1</p></a></li>
<li><a href='#ISOpure.step1.CPE'><p>Perform first step of ISOpure purification algorithm</p></a></li>
<li><a href='#ISOpure.step2.PPE'><p>Perform second step of ISOpure purification algorithm</p></a></li>
<li><a href='#ISOpure.util.logsum'><p>Log-sum-exp</p></a></li>
<li><a href='#ISOpure.util.matlab_greater_than'><p>Greater than operator</p></a></li>
<li><a href='#ISOpure.util.matlab_less_than'><p>Less than operator</p></a></li>
<li><a href='#ISOpure.util.matlab_log'><p>Modified logarithm function</p></a></li>
<li><a href='#ISOpure.util.repmat'><p>Tiles matrix horizontally or vertically</p></a></li>
<li><a href='#ISOpureS1.model_core.compute_loglikelihood'><p>Compute loglikelihood given all model parameters for step 1</p></a></li>
<li><a href='#ISOpureS1.model_core.new_model'><p>Initialize a model list to hold all the parameters</p></a></li>
<li><a href='#ISOpureS1.model_core.optmodel'><p>Optimizes the ISOpure parameters for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood'><p>Compute loglikelihood relevant to kappa for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood'><p>Compute derivative of loglikelihood with respect to kappa for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.kappa.kappa_loglikelihood'><p>Compute loglikelihood relevant to kappa for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood'><p>Compute the derivative of the loglikelihood relevant to mm for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.mm.mm_loglikelihood'><p>Compute the loglikelihood relevant to mm for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.omega.omega_compute_loglikelihood'><p>Compute loglikelihood relevant to omega for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood'><p>Compute the derivative of loglikelihood relevant to omega for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.omega.omega_loglikelihood'><p>Compute the loglikelihood relevant to omega for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.opt_kappa'><p>Optimize kappa in step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.opt_mm'><p>Optimize the reference cancer profile, m, in step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.opt_omega'><p>Optimize omega in step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.opt_theta'><p>Optimize theta in step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.opt_vv'><p>Optimize vv in step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood'><p>Compute the derivative of loglikelihood relevant to theta for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.theta.theta_loglikelihood'><p>Compute the loglikelihood relevant to theta for step 1</p></a></li>
<li><a href='#ISOpureS1.model_optimize.vv.vv_compute_loglikelihood'><p>Compute loglikelihood relevant to vv for step 1</p></a></li>
<li><a href='#ISOpureS2.model_core.compute_loglikelihood'><p>Compute loglikelihood given all model parameters for step 2</p></a></li>
<li><a href='#ISOpureS2.model_core.new_model'><p>Compute loglikelihood given all model parameters for step 2</p></a></li>
<li><a href='#ISOpureS2.model_core.optmodel'><p>Optimizes the ISOpure parameters for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood'><p>Compute the derivative of loglikelihood relevant to the patient cancer profiles, cc, for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.cc.cc_loglikelihood'><p>Compute the loglikelihood relevant to the patient cancer profiles, cc, for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.kappa.kappa_compute_loglikelihood'><p>Compute loglikelihood relevant to kappa for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.kappa.kappa_deriv_loglikelihood'><p>Compute derivative of loglikelihood with respect to kappa for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.kappa.kappa_loglikelihood'><p>Compute loglikelihood relevant to kappa for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.opt_cc'><p>Optimize the tumor-specific cancer profiles in step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.opt_kappa'><p>Optimize kappa in step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.opt_theta'><p>Optimize theta in step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.opt_vv'><p>Optimize vv in step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood'><p>Compute the derivative of loglikelihood relevant to theta for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.theta.theta_loglikelihood'><p>Compute the loglikelihood relevant to theta for step 2</p></a></li>
<li><a href='#ISOpureS2.model_optimize.vv.vv_compute_loglikelihood'><p>Compute loglikelihood relevant to vv for step 2</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-10</td>
</tr>
<tr>
<td>Title:</td>
<td>Deconvolution of Tumour Profiles</td>
</tr>
<tr>
<td>Author:</td>
<td>Gerald Quon [aut],
  Catalina V Anghel [aut, trl],
  Syed Haider [aut],
  Francis Nguyen [aut],
  Amit G Deshwar [aut],
  Quaid D Morris [aut],
  Paul C Boutros [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Paul C Boutros &lt;pboutros@mednet.ucla.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.3), stats, futile.logger</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen (&ge; 0.3.2.2.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Description:</td>
<td>Deconvolution of mixed tumour profiles into normal and cancer for each patient, using 
	the ISOpure algorithm in Quon et al. Genome Medicine, 2013 5:29. Deconvolution requires 
	mixed tumour profiles and a set of unmatched "basis" normal profiles.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-10 21:14:29 UTC; jmacdonald</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-11 00:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='ISOpure.calculate.tac'>Perform calculation for Tumour Adjacent Cell (TAC) profiles</h2><span id='topic+ISOpure.calculate.tac'></span>

<h3>Description</h3>

<p>Performs the mathematical calculations taking bulk tumor data and deconvolved profiles
and returning deconvolved tumour adjacent cell profiles.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.calculate.tac(tumor.profiles, deconvolved.profiles, purity.estimates)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.calculate.tac_+3A_tumor.profiles">tumor.profiles</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of heterogeneous (mixed) 
tumor samples, where G is the number of genes, D is the number of tumor samples.</p>
</td></tr>
<tr><td><code id="ISOpure.calculate.tac_+3A_deconvolved.profiles">deconvolved.profiles</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of purified (ISOpure output)
tumor samples, where G is the number of genes, D is the number of tumor samples.</p>
</td></tr>
<tr><td><code id="ISOpure.calculate.tac_+3A_purity.estimates">purity.estimates</code></td>
<td>
<p>a vector D representing the purity estimates (output from ISOpure)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a GxD matrix representing gene expression profiles of purified (ISOpure output)
tumor adjacent cell signal, where G is the number of genes, D is the number of tumor samples. 
</p>


<h3>Author(s)</h3>

<p>Natalie Fox</p>

<hr>
<h2 id='ISOpure.model_optimize.cg_code.rminimize'>Minimize a differentiable multivariate function</h2><span id='topic+ISOpure.model_optimize.cg_code.rminimize'></span>

<h3>Description</h3>

<p>This function is a conjugate-gradient search with interpolation/extrapolation by Carl 
Edward Rasmussen.  A description of the Matlab code can be found at 
http://learning.eng.cam.ac.uk/carl/code/minimize/ (accessed Jan. 21, 2014). This is a 
implementation in R.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.model_optimize.cg_code.rminimize(X, f, df, run_length, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.model_optimize.cg_code.rminimize_+3A_x">X</code></td>
<td>
<p>The starting point is given by X which must be either a scalar or a column vector or
matrix, not a row matrix</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.cg_code.rminimize_+3A_f">f</code></td>
<td>
<p>The name of the function to be minimized, returning a scalar</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.cg_code.rminimize_+3A_df">df</code></td>
<td>
<p>The name of the function which returns the vector of partial derivatives of f wrt X,
where again the partial derivatives must be in scalar or column vector/matrix form</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.cg_code.rminimize_+3A_run_length">run_length</code></td>
<td>
<p>Gives the length of the run: if it is positive, it gives the maximum number 
of line searches, if negative its absolute gives the maximum allowed number of function 
evaluations. Note, for ISOpureR, used only positive run_length.</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.cg_code.rminimize_+3A_...">...</code></td>
<td>
<p>Parameters to be passed on to the function f.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function returns when either its length is up, or if no further progress can be made (ie,
we are at a (local) minimum, or so close that due to numerical problems, we cannot get any 
closer). NOTE: If the function terminates within a few iterations, it could be an indication 
that the function values and derivatives are not consistent (ie, there may be a bug in the 
implementation of your &quot;f&quot; function).
</p>
<p>The Polack-Ribiere flavour of conjugate gradients is used to compute search directions, and a 
line search using quadratic and cubic polynomial approximations and the Wolfe-Powell stopping 
criteria is used together with the slope ratio method for guessing initial step sizes. 
Additionally a bunch of checks are made to make sure that exploration is taking place and that
extrapolation will not be unboundedly large.</p>


<h3>Value</h3>

<p>A list with three components:
</p>
<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>The found solution X</p>
</td></tr>
<tr><td><code>fX</code></td>
<td>
<p>A vector of function values fX indicating the progress made</p>
</td></tr>
<tr><td><code>i</code></td>
<td>
<p>The number of iterations</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Catalina Anghel, Francis Nguyen, Carl Edward Rasmussen</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example from Carl E. Rasmussen's webpage

rosenbrock &lt;- function(x){
	D &lt;- length(x);
 	y &lt;- sum(100*(x[2:D] - x[1:(D-1)]^2)^2 + (1-x[1:(D-1)])^2);
 	return(y);
 	};
drosenbrock &lt;- function(x){
	D &lt;- length(x);
	df &lt;- numeric(D);
	df[1:D-1] &lt;- -400*x[1:(D-1)]*(x[2:D]-x[1:(D-1)]^2) - 2*(1-x[1:(D-1)]);
  	df[2:D] &lt;- df[2:D] + 200*(x[2:D]-x[1:(D-1)]^2);
  	return(df);
	};

ISOpure.model_optimize.cg_code.rminimize(c(0,0), rosenbrock, drosenbrock, 25)
#
# [[1]]
# [1] 1 1
#
# [[2]]
#  [1] 1.000000e+00 7.716094e-01 5.822402e-01 4.049274e-01 3.246633e-01
#  [6] 2.896041e-01 7.623420e-02 6.786212e-02 3.378424e-02 1.089908e-03
# [11] 1.087952e-03 8.974308e-05 1.218382e-07 6.756019e-09 3.870791e-15
# [16] 1.035408e-21 6.248025e-27 5.719242e-30 4.930381e-32
#
# [[3]]
# [1] 20
</code></pre>

<hr>
<h2 id='ISOpure.model_optimize.vv.vv_deriv_loglikelihood'>Compute the derivative of the loglikelihood relevant to vv for step 1</h2><span id='topic+ISOpure.model_optimize.vv.vv_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the loglikelihood function relevant to optimizing vv for 
step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.model_optimize.vv.vv_deriv_loglikelihood(ww, sum_log_theta, DD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.model_optimize.vv.vv_deriv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>log(vv-1), a Kx1 matrix</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.vv.vv_deriv_loglikelihood_+3A_sum_log_theta">sum_log_theta</code></td>
<td>
<p>the column sums of log(theta), a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.vv.vv_deriv_loglikelihood_+3A_dd">DD</code></td>
<td>
<p>the number of patients (a scalar)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the part of the loglikelihood function relevant to vv with 
respect to (log) vv</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpure.model_optimize.vv.vv_loglikelihood'>Compute the loglikelihood relevant to vv for step 1</h2><span id='topic+ISOpure.model_optimize.vv.vv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing vv for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.model_optimize.vv.vv_loglikelihood(ww, sum_log_theta, DD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.model_optimize.vv.vv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>log(vv-1), a Kx1 matrix</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.vv.vv_loglikelihood_+3A_sum_log_theta">sum_log_theta</code></td>
<td>
<p>the column sums of log(theta), a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpure.model_optimize.vv.vv_loglikelihood_+3A_dd">DD</code></td>
<td>
<p>the number of patients (a scalar)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to vv</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpure.step1.CPE'>Perform first step of ISOpure purification algorithm</h2><span id='topic+ISOpure.step1.CPE'></span>

<h3>Description</h3>

<p>Performs the first step of the ISOpure purification algorithm, taking tumor data 
normal profiles and returning the a list, ISOpureS1model, with all the updated parameters.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.step1.CPE(tumordata, BB, PP, MIN_KAPPA, logging.level) 
</code></pre>


<h3>Arguments</h3>


<table role = "presentation">
<tr><td><code id="ISOpure.step1.CPE_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of heterogeneous (mixed) 
tumor samples, where G is the number of genes, D is the number of tumor samples.</p>
</td></tr>
<tr><td><code id="ISOpure.step1.CPE_+3A_bb">BB</code></td>
<td>
<p>represents B = [b_1 ... b_(K-1)] matrix (from Genome Medicine paper) a Gx(K-1) 
matrix, where (K-1) is the number of normal profiles <code class="reqn">(\beta_1,...,\beta_(K-1))</code>, 
G is the number of genes.  These are the normal profiles representing normal cells that 
contaminate the tumor samples  (i.e. normal samples from the same tissue location as the 
tumor).  The minimum element of BB must be greater than 0 &ndash; i.e. every gene/transcript 
must be observed on some level in each normal sample.</p>
</td></tr>
<tr><td><code id="ISOpure.step1.CPE_+3A_pp">PP</code></td>
<td>
<p>a GxM matrix, representing the expression profiles whose convex combination form 
the prior over the purified cancer profile learned. </p>
</td></tr> 






<tr><td><code id="ISOpure.step1.CPE_+3A_min_kappa">MIN_KAPPA</code></td>
<td>
<p>(optional) The minimum value allowed for the strength parameter kappa placed
over the reference cancer profile m (see Quon et al, 2013).  By default, this is set to 
1/min(BB), such that the log likelihood of the model is always finite.  However, when the 
min(BB) is very small, this forces MIN_KAPPA to be very large, and can sometimes cause the 
reference profile m to look too much like a 'normal profile' (and therefore you may observe 
the tumor samples having low % cancer content estimates).  If this is the case, you can try 
setting MIN_KAPPA=1, or some other small value.  For reference, for the data presented in 
Quon et al., 2013, MIN_KAPPA is on the order of 10^5.</p>
</td></tr>
<tr><td><code id="ISOpure.step1.CPE_+3A_logging.level">logging.level</code></td>
<td>
<p>(optional) A string that gives the logging threshold for futile.logger. The
possible options are 'TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'. Currently the 
messages in ISOpureR are only in the categories 'INFO', 'WARN', and 'FATAL', and the default 
setting is 'INFO'.  Setting a setting for the entire package will over-ride the setting for
a particular function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ISOpureS1model, a list with the following important fields:
</p>
<table role = "presentation">
<tr><td><code>theta</code></td>
<td>
<p>a DxK matrix, giving the fractional composition of each tumor sample.  Each row 
represents a tumor sample that was part of the input, and the first K-1 columns correspond 
to the fractional composition with respect to the Source Panel contaminants.  The last 
column represents the fractional composition of the pure cancer cells.  In other words,	
each row sums to 1, and element (i,j) of the matrix denotes the fraction of tumor i 
attributable to component j (where the last column refers to cancer cells, and the first 
K-1 columns refer to different 'normal cell' components).  The 'cancer', or tumor purity, 
estimate of each tumor is simply the last column of theta.</p>
</td></tr>
<tr><td><code>alphapurities</code></td>
<td>
<p>tumor purities (alpha_i in paper), same as the last column of the theta 
variable, pulled out for user convenience.</p>
</td></tr>
<tr><td><code>mm</code></td>
<td>
<p>reference cancer profile, in the form of parameters of a multinomial or discrete 
distribution (sum of elements is 1).  This is the same as the purified cancer profile that 
ISOLATE was designed to learn.</p>
</td></tr>
<tr><td><code>omega</code></td>
<td>
<p>a Mx1 vector describing the convex combination weights learned by ISOpure step 1 
over the PPtranspose matrix, that when applied to the Site of Origin Panel, forms the prior 
over the reference cancer profile. When ISOpure step 1 is used in a similar fashion to the 
ISOLATE algorithm, entry i indicates the &quot;probability&quot; that the normal profile in the 
i-th column of PP is the site of origin of the secondary tumors stored in tumordata.</p>
</td></tr>
<tr><td><code>total_loglikelihood</code></td>
<td>
<p>log likelihood of the model</p>
</td></tr>
<tr><td><code>vv</code></td>
<td>
<p>(internal parameter) hyper-parameters from Dirichlet distribution, representing both
mean and strength of a Dirichlet distribution over theta</p>
</td></tr>
<tr><td><code>kappa</code></td>
<td>
<p>(internal parameter) the strength parameter over the Dirichlet distribution over 
the reference cancer parameter, mm</p>
</td></tr>
<tr><td><code>mm_weights</code>, <code>theta_weights</code>, <code>omega_weights</code></td>
<td>
<p>(internal parameters) used in the optimization
of mm, theta, and omega (instead of performing constrained optimization on these 
positively constrained variables directly, we optimize their logs in an unconstrained 
fashion.)</p>
</td></tr>
<tr><td><code>log_BBtranspose</code>, <code>PPtranspose</code>, <code>log_all_rates:</code></td>
<td>
<p> (internal parameters) used in the 
calculations of loglikelihood</p>
</td></tr>
<tr><td><code>MIN_KAPPA</code></td>
<td>
<p>(internal parameter) as described in the Arguments section</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>


<h3>References</h3>

<p>G Quon, S Haider, AG Deshwar, A Cui, PC Boutros, QD Morris. <em>Computational purification of 
individual tumor gene expression profiles</em>. Genome Medicine (2013) 5:29, 
<a href="http://genomemedicine.com/content/5/3/29">http://genomemedicine.com/content/5/3/29</a>.
</p>
<p>G Quon, QD Morris. <em>ISOLATE: a computational strategy for identifying the primary origin of
cancers using high-thoroughput sequencing</em>. Bioinformatics 2009, 25:2882-2889 
<a href="http://bioinformatics.oxfordjournals.org/content/25/21/2882">http://bioinformatics.oxfordjournals.org/content/25/21/2882</a>.
</p>

<hr>
<h2 id='ISOpure.step2.PPE'>Perform second step of ISOpure purification algorithm</h2><span id='topic+ISOpure.step2.PPE'></span>

<h3>Description</h3>

<p>Performs the second step of the ISOpure purification algorithm, taking tumor data and
normal profiles and returning the a list, ISOpureS2model, with all the updated parameters.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.step2.PPE(tumordata, BB, ISOpureS1model, MIN_KAPPA, logging.level) 
</code></pre>


<h3>Arguments</h3>


<table role = "presentation">
<tr><td><code id="ISOpure.step2.PPE_+3A_tumordata">tumordata</code></td>
<td>
<p>(same as for ISOpureS1) a GxD matrix representing gene expression profiles of 
heterogeneous (mixed) tumor samples, where G is the number of genes, D is the number of tumor
samples.</p>
</td></tr>
<tr><td><code id="ISOpure.step2.PPE_+3A_bb">BB</code></td>
<td>
<p>(same as for ISOpureS1) represents B = [b_1 ... b_(K-1)] matrix (from Genome Medicine 
paper) a Gx(K-1) matrix, where (K-1) is the number of normal profiles 
<code class="reqn">(\beta_1,...,\beta_(K-1))</code>, G is the number of genes.  These are the normal profiles 
representing normal cells that contaminate the tumor samples  (i.e. normal samples from the 
same tissue location as the tumor).  The minimum element of BB must be greater than 0 &ndash; 
i.e. every gene/transcript must be observed on some level in each normal sample.</p>
</td></tr>
<tr><td><code id="ISOpure.step2.PPE_+3A_isopures1model">ISOpureS1model</code></td>
<td>
<p>output model list from ISOpureS1 code</p>
</td></tr> 






<tr><td><code id="ISOpure.step2.PPE_+3A_min_kappa">MIN_KAPPA</code></td>
<td>
<p>(optional) The minimum value allowed for the strength parameter kappa placed
over the reference cancer profile m (see Quon et al, 2013).  By default, this is set to 
1/min(BB), such that the log likelihood of the model is always finite.  However, when the 
min(BB) is very small, this forces MIN_KAPPA to be very large, and can sometimes cause the 
reference profile m to look too much like a 'normal profile' (and therefore you may observe 
the tumor samples having low % cancer content estimates).  If this is the case, you can try 
setting MIN_KAPPA=1, or some other small value.  For reference, for the data presented in 
Quon et al., 2013, MIN_KAPPA is on the order of 10^5.</p>
</td></tr>
<tr><td><code id="ISOpure.step2.PPE_+3A_logging.level">logging.level</code></td>
<td>
<p>(optional) A string that gives the logging threshold for futile.logger. The
possible options are 'TRACE', 'DEBUG', 'INFO', 'WARN', 'ERROR', 'FATAL'. Currently the 
messages in ISOpureR are only in the categories 'INFO', 'WARN', and 'FATAL', and the default 
setting is 'INFO'.  Setting a setting for the entire package will over-ride the setting for
a particular function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ISOpureS2model, a list with the following important fields:
</p>
<table role = "presentation">
<tr><td><code>theta</code></td>
<td>
<p>a DxK matrix, giving the fractional composition of each tumor sample.  Each row 
represents a tumor sample that was part of the input, and the first K-1 columns correspond 
to the fractional composition with respect to the Source Panel contaminants.  The last 
column represents the fractional composition of the pure cancer cells.  In other words,	
each row sums to 1, and element (i,j) of the matrix denotes the fraction of tumor i 
attributable to component j (where the last column refers to cancer cells, and the first 
K-1 columns refer to different 'normal cell' components).  The 'cancer', or tumor purity, 
estimate of each tumor is simply the last column of theta.</p>
</td></tr>
<tr><td><code>alphapurities</code></td>
<td>
<p>(same as ISOpureS1) tumor purities (alpha_i in paper), same as the last 
column of the theta variable, pulled out for user convenience - not changed in step 2</p>
</td></tr>
<tr><td><code>cc_cancerprofiles</code></td>
<td>
<p>purified cancer profiles.  This matrix is of the same dimensionality as 
tumordata, and is also on the same scale (i.e. although ISOpureS2 treats purified cancer 
profiles as parameters of a multinomial distribution, we re-scale them to be on the same 
scale as the input tumor profiles &ndash; see Genome Medicine paper). Column i of
cc_cancerprofiles corresponds to column i of tumordata.</p>
</td></tr>
<tr><td><code>total_loglikelihood</code></td>
<td>
<p>log likelihood of the model</p>
</td></tr>
<tr><td><code>omega</code></td>
<td>
<p>(internal parameter, same as ISOpureS1) prior over the reference cancer profile -
not changed in step 2</p>
</td></tr>
<tr><td><code>vv</code></td>
<td>
<p>(internal parameter) hyper-parameters from Dirichlet distribution, representing both
mean and strength of a Dirichlet distribution over theta</p>
</td></tr>
<tr><td><code>kappa</code></td>
<td>
<p>(internal parameter) the strength parameter over the Dirichlet distribution over 
cc, given the reference cancer parameter, mm</p>
</td></tr>
<tr><td><code>mm_weights</code>, <code>theta_weights</code>, <code>omega_weights</code></td>
<td>
<p>(internal parameters) used in the optimization
of mm, theta, and omega (instead of performing constrained optimization on these 
positively constrained variables directly, we optimize their logs in an unconstrained 
fashion.)</p>
</td></tr>
<tr><td><code>log_BBtranspose</code>, <code>PPtranspose</code>, <code>log_all_rates:</code></td>
<td>
<p> (internal parameters) used in the 
calculations of loglikelihood</p>
</td></tr>
<tr><td><code>MIN_KAPPA</code></td>
<td>
<p>(internal parameter) as described in the Arguments section</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>


<h3>References</h3>

<p>G Quon, S Haider, AG Deshwar, A Cui, PC Boutros, QD Morris. <em>Computational purification of 
individual tumor gene expression profiles</em>. Genome Medicine (2013) 5:29, 
<a href="http://genomemedicine.com/content/5/3/29">http://genomemedicine.com/content/5/3/29</a>.
</p>
<p>G Quon, QD Morris. <em>ISOLATE: a computational strategy for identifying the primary origin of
cancers using high-thoroughput sequencing</em>. Bioinformatics 2009, 25:2882-2889 
<a href="http://bioinformatics.oxfordjournals.org/content/25/21/2882">http://bioinformatics.oxfordjournals.org/content/25/21/2882</a>.
</p>

<hr>
<h2 id='ISOpure.util.logsum'>Log-sum-exp</h2><span id='topic+ISOpure.util.logsum'></span>

<h3>Description</h3>

<p>Prevents underflow/overflow using the log-sum-exp trick </p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.util.logsum(xx, dimen);
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.util.logsum_+3A_xx">xx</code></td>
<td>
<p>A matrix of numerical values</p>
</td></tr>
<tr><td><code id="ISOpure.util.logsum_+3A_dimen">dimen</code></td>
<td>
<p>The dimension along which the long sum is taken (1 for row, 2 for column)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns log(sum(exp(x),dimen)), the log sum of exps, summing over dimension dimen
but in a way that tries to avoid underflow/overflow.</p>


<h3>Author(s)</h3>

<p>Gerald Quon and Catalina Anghel</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;-  c(1, 1e20, 1e40, -1e40, -1e20, -1);
x &lt;- as.matrix(x);

# compute log sum exp without the function
log(sum(exp(x)))
#[1] Inf

# compute log sum exp with the function
ISOpure.util.logsum(x, 1)
#[1] 1e+40 
</code></pre>

<hr>
<h2 id='ISOpure.util.matlab_greater_than'>Greater than operator</h2><span id='topic+ISOpure.util.matlab_greater_than'></span>

<h3>Description</h3>

<p>Greater than function that matches Matlab behaviour when one of the arguments is NA (i.e. returns FALSE instead of NA)</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.util.matlab_greater_than(a, b)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.util.matlab_greater_than_+3A_a">a</code></td>
<td>
<p>A numeric value (including Inf) or NA</p>
</td></tr>
<tr><td><code id="ISOpure.util.matlab_greater_than_+3A_b">b</code></td>
<td>
<p>A numeric value or NA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical: TRUE if a &gt; b, FALSE if a &lt;= b OR if one of a, b is NA or NaN</p>


<h3>Author(s)</h3>

<p>Catalina Anghel</p>


<h3>Examples</h3>

<pre><code class='language-R'>ISOpure.util.matlab_greater_than(5,3)
#[1] TRUE
ISOpure.util.matlab_greater_than(3,5)
#[1] FALSE
ISOpure.util.matlab_greater_than(5,NA)
#[1] FALSE
ISOpure.util.matlab_greater_than(NA,5)
#[1] FALSE
ISOpure.util.matlab_greater_than(5,Inf)
#[1] FALSE
ISOpure.util.matlab_greater_than(Inf,5)
#[1] TRUE
</code></pre>

<hr>
<h2 id='ISOpure.util.matlab_less_than'>Less than operator</h2><span id='topic+ISOpure.util.matlab_less_than'></span>

<h3>Description</h3>

<p>Less than function that matches Matlab behaviour when one of the arguments is NA (i.e. returns FALSE instead of NA)</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.util.matlab_less_than(a, b)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.util.matlab_less_than_+3A_a">a</code></td>
<td>
<p>A numeric value (including Inf) or NA</p>
</td></tr>
<tr><td><code id="ISOpure.util.matlab_less_than_+3A_b">b</code></td>
<td>
<p>A numeric value (including Inf) or NA</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Logical: TRUE if a &lt; b, FALSE if a &gt;= b OR if one of a, b is NA or NaN</p>


<h3>Author(s)</h3>

<p>Catalina Anghel</p>


<h3>Examples</h3>

<pre><code class='language-R'>ISOpure.util.matlab_less_than(5,3)
#[1] FALSE
ISOpure.util.matlab_less_than(3,5)
#[1] TRUE
ISOpure.util.matlab_less_than(5,NA)
#[1] FALSE
ISOpure.util.matlab_less_than(NA,5)
#[1] FALSE
ISOpure.util.matlab_less_than(5,Inf)
#[1] TRUE
ISOpure.util.matlab_less_than(Inf,5)
#[1] FALSE
</code></pre>

<hr>
<h2 id='ISOpure.util.matlab_log'>Modified logarithm function</h2><span id='topic+ISOpure.util.matlab_log'></span>

<h3>Description</h3>

<p>Logarithm function that matches Matlab behaviour on negative entries (i.e. returns a 
complex number)</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.util.matlab_log(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.util.matlab_log_+3A_x">x</code></td>
<td>
<p>A numeric or complex value, vector, or matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns log(x) if all entries of x &gt; 0. For complex or negative input, x, where x = a + bi, 
the function returns log(z) = log(abs(z)) + 1i*atan2(b,a) where atan(b,a) is on the half-closed
interval, (-pi, pi], as for the Matlab log function.</p>


<h3>Author(s)</h3>

<p>Catalina Anghel</p>


<h3>Examples</h3>

<pre><code class='language-R'>ISOpure.util.matlab_log(5)
#[1] 1.609438
ISOpure.util.matlab_log(-5)
#[1] 1.609438+3.141593i
ISOpure.util.matlab_log(complex(real=3, imaginary=4))
#[1] 1.609438+0.927295i
ISOpure.util.matlab_log(c(2,3,4,-7,1))
#[1] 0.6931472+0.000000i 1.0986123+0.000000i 1.3862944+0.000000i
#[4] 1.9459101+3.141593i 0.0000000+0.000000i
</code></pre>

<hr>
<h2 id='ISOpure.util.repmat'>Tiles matrix horizontally or vertically</h2><span id='topic+ISOpure.util.repmat'></span>

<h3>Description</h3>

<p>Tiles matrix horizontally or vertically in the same way as the Matlab repmat command</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpure.util.repmat(a, n, m)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpure.util.repmat_+3A_a">a</code></td>
<td>
<p>A matrix</p>
</td></tr>
<tr><td><code id="ISOpure.util.repmat_+3A_n">n</code></td>
<td>
<p>Number of times the matrix should be tiled horizontally</p>
</td></tr>
<tr><td><code id="ISOpure.util.repmat_+3A_m">m</code></td>
<td>
<p>number of times the matrix should be tiled vertically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix which has replicated and tiled the input matrix a by n rows and m columns</p>


<h3>Author(s)</h3>

<p>Catalina Anghel, Ohloh (now Black Duck Open Hub)</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- matrix(runif(6), 3, 2)
x
#           [,1]      [,2]
# [1,] 0.5167029 0.7543404
# [2,] 0.9064936 0.4316977
# [3,] 0.3256870 0.5310625
ISOpure.util.repmat(x, 1, 2)
#           [,1]      [,2]      [,3]      [,4]
# [1,] 0.5167029 0.7543404 0.5167029 0.7543404
# [2,] 0.9064936 0.4316977 0.9064936 0.4316977
# [3,] 0.3256870 0.5310625 0.3256870 0.5310625
ISOpure.util.repmat(x, 2, 1)
#           [,1]      [,2]
# [1,] 0.5167029 0.7543404
# [2,] 0.9064936 0.4316977
# [3,] 0.3256870 0.5310625
# [4,] 0.5167029 0.7543404
# [5,] 0.9064936 0.4316977
# [6,] 0.3256870 0.5310625
ISOpure.util.repmat(x, 2, 3)
#           [,1]      [,2]      [,3]      [,4]      [,5]      [,6]
# [1,] 0.5167029 0.7543404 0.5167029 0.7543404 0.5167029 0.7543404
# [2,] 0.9064936 0.4316977 0.9064936 0.4316977 0.9064936 0.4316977
# [3,] 0.3256870 0.5310625 0.3256870 0.5310625 0.3256870 0.5310625
# [4,] 0.5167029 0.7543404 0.5167029 0.7543404 0.5167029 0.7543404
# [5,] 0.9064936 0.4316977 0.9064936 0.4316977 0.9064936 0.4316977
# [6,] 0.3256870 0.5310625 0.3256870 0.5310625 0.3256870 0.5310625
</code></pre>

<hr>
<h2 id='ISOpureS1.model_core.compute_loglikelihood'>Compute loglikelihood given all model parameters for step 1</h2><span id='topic+ISOpureS1.model_core.compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes complete loglikelihood given all model parameters for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_core.compute_loglikelihood(tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_core.compute_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.compute_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters updated in ISOpure step one iterations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The scalar value of the complete loglikelihood obtained given the model parameters</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_core.new_model'>Initialize a model list to hold all the parameters</h2><span id='topic+ISOpureS1.model_core.new_model'></span>

<h3>Description</h3>

<p>Produces a list (the model) which initializes the parameters vv, log_BBtranspose, 
PPtranspose, kappa, theta, omega, log_all_rates for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_core.new_model(tumordata, kappa, INITIAL_VV, PPtranspose, BBtranspose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_core.new_model_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.new_model_+3A_kappa">kappa</code></td>
<td>
<p>scalar strength parameter kappa placed over the reference cancer profile mm</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.new_model_+3A_initial_vv">INITIAL_VV</code></td>
<td>
<p>a vector with K components, the prior over mixing proportions, theta, with 
last entry weighed more heavily </p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.new_model_+3A_pptranspose">PPtranspose</code></td>
<td>
<p>a (K-1)xG matrix, standardized so that all entries sum to 1, see 
ISOpure.step1.CPE.R</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.new_model_+3A_bbtranspose">BBtranspose</code></td>
<td>
<p>a (K-1)xG matrix of the standardized normal profiles, so that they sum to 1</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>a newly generated model list to hold all the parameters vv, log_BBtranspose, 
PPtranspose, kappa, theta, omega, log_all_rates</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_core.optmodel'>Optimizes the ISOpure parameters for step 1</h2><span id='topic+ISOpureS1.model_core.optmodel'></span>

<h3>Description</h3>

<p>Optimizes the ISOpure parameters for step 1 cyclically until convergence</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_core.optmodel(tumordata, model, NUM_ITERATIONS=35)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_core.optmodel_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.optmodel_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_core.optmodel_+3A_num_iterations">NUM_ITERATIONS</code></td>
<td>
<p>(optional) minimum number of iterations of optimization algorithm, 
default is 35</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>updated model list containing all the parameters</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood'>Compute loglikelihood relevant to kappa for step 1</h2><span id='topic+ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing kappa for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood(kappa, tumordata, model) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood_+3A_kappa">kappa</code></td>
<td>
<p>a scalar kappa, the strength parameter in the prior over the reference cancer profile</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_compute_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The part of the loglikelihood function relevant to optimizing kappa</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood'>Compute derivative of loglikelihood with respect to kappa for step 1</h2><span id='topic+ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the part of the loglikelihood function relevant to 
optimizing kappa for step 1. Instead of performing constrained optimization on kappa directly, 
we optimize the log of kappa in an unconstrained fashion.  Thus, if y=log(kappa) and L is the 
loglikelihood function w.r.t. y, to optimize L w.r.t. y,   dL/dy = dL/dkappa * dkappa/dy, 
where dkappa/dy = exp(y)= exp( log(kappa)). The input into the derivative function is 
log(kappa - model\$MIN_KAPPA). </p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood(log_kappa, tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood_+3A_log_kappa">log_kappa</code></td>
<td>
<p>the scalar log(kappa - model\$MIN_KAPPA)</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the part of the loglikelihood function relevant to kappa
with respect to log kappa (a scalar given that for step 1 of ISOpure kappa is a scalar)</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.kappa.kappa_loglikelihood'>Compute loglikelihood relevant to kappa for step 1</h2><span id='topic+ISOpureS1.model_optimize.kappa.kappa_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing kappa for step 1. 
Instead of performing constrained optimization on kappa directly, we optimize the log of kappa
in an unconstrained fashion.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.kappa.kappa_loglikelihood(log_kappa, tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_loglikelihood_+3A_log_kappa">log_kappa</code></td>
<td>
<p>the scalar log(kappa - model\$MIN_KAPPA)</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.kappa.kappa_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to optimizing kappa</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood'>Compute the derivative of the loglikelihood relevant to mm for step 1</h2><span id='topic+ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the loglikelihood function relevant to optimizing the 
reference cancer profile, mm, for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood(ww, tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the mm_weights, with G entries</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.mm.mm_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative the likelihood function relevant to optimizing mm.  The derivative 
is taken not with respect to mm but with respect to unconstrained variables via a change of 
variables.</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.mm.mm_loglikelihood'>Compute the loglikelihood relevant to mm for step 1</h2><span id='topic+ISOpureS1.model_optimize.mm.mm_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the loglikelihood function relevant to optimizing the reference cancer 
profile, mm, for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.mm.mm_loglikelihood(ww, tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.mm.mm_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the mm_weights, with G entries</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.mm.mm_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.mm.mm_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the likelihood function relevant to optimizing mm.</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.omega.omega_compute_loglikelihood'>Compute loglikelihood relevant to omega for step 1</h2><span id='topic+ISOpureS1.model_optimize.omega.omega_compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing omega for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.omega.omega_compute_loglikelihood(omega, tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_compute_loglikelihood_+3A_omega">omega</code></td>
<td>
<p>(K-1)x1 matrix representing the weights of the normal profiles B_i used to
make the weighted combination that forms the mean parameter vector for the 
Dirichlet distribution over m</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_compute_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_compute_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The part of the loglikelihood function relevant to optimizing omega</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood'>Compute the derivative of loglikelihood relevant to omega for step 1</h2><span id='topic+ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Compute the derivative of the part of the loglikelihood function relevant to omega 
with respect to (log) omega, in step 1. Instead of performing constrained optimization on omega 
directly, we optimize the log of omega in an unconstrained fashion.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood(ww, tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>(K-1)x1 matrix, log(omega), where the entries in omega are constrained to add to 1
where K-1 is the number of normal samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the part of the loglikelihood function relevant to omega   
with respect to (log) omega</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.omega.omega_loglikelihood'>Compute the loglikelihood relevant to omega for step 1</h2><span id='topic+ISOpureS1.model_optimize.omega.omega_loglikelihood'></span>

<h3>Description</h3>

<p>Compute the the part of the loglikelihood function relevant to omega in step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.omega.omega_loglikelihood(ww, tumordata, model) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>(K-1)x1 matrix, log(omega), where the entries in omega are constrained to add to 1
where K-1 is the number of normal samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.omega.omega_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood function relevant to omega</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.opt_kappa'>Optimize kappa in step 1</h2><span id='topic+ISOpureS1.model_optimize.opt_kappa'></span>

<h3>Description</h3>

<p>This function optimizes kappa, the strength parameter in the prior over the reference
cancer profile.  Note that we don't directly optimize kappa because it has constraints (must be 
greater than the minimum determined in ISOpure.step1.CPE.)</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.opt_kappa(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.opt_kappa_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_kappa_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_kappa_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_kappa_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_kappa_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the kappa parameter updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.opt_mm'>Optimize the reference cancer profile, m, in step 1</h2><span id='topic+ISOpureS1.model_optimize.opt_mm'></span>

<h3>Description</h3>

<p>The goal of this function is to optimize the reference cancer profile mm.
Because mm is constrained (must be parameters of multinomial/discrete
distribution), we don't directly optimize the likelihood function w.r.t.
mm, but we perform change of variables to do unconstrained
optimization.  We therefore store these unconstrained variables in the
field &quot;mm_weights&quot;, and update these variables.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.opt_mm(
	tumordata, model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.opt_mm_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_mm_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_mm_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_mm_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_mm_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with mm_weights updated (and log_all_rates)</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.opt_omega'>Optimize omega in step 1</h2><span id='topic+ISOpureS1.model_optimize.opt_omega'></span>

<h3>Description</h3>

<p>This function optimizes omega, in fact the convex mixing weights that govern prior 
over the reference cancer profile.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.opt_omega(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.opt_omega_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_omega_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_omega_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_omega_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_omega_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the omega_weights and omega parameters updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.opt_theta'>Optimize theta in step 1</h2><span id='topic+ISOpureS1.model_optimize.opt_theta'></span>

<h3>Description</h3>

<p>This function optimizes theta, in fact theta_weights.  Since thetas are constrained 
(must be parameters of multinomial/discrete distribution), we don't directly optimize the 
likelihood function w.r.t. theta, but we perform change of variables to do unconstrained
optimization.  We therefore store these unconstrained variables in the field &quot;theta_weights&quot;,
and update these variables.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.opt_theta(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.opt_theta_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_theta_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_theta_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_theta_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_theta_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the theta parameter updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.opt_vv'>Optimize vv in step 1</h2><span id='topic+ISOpureS1.model_optimize.opt_vv'></span>

<h3>Description</h3>

<p>This function optimizes vv, the strength parameter in the prior over the reference
cancer profile.  Note that we don't directly optimize vv because it has constraints (must be &gt;=1
to guarantee real-valued likelihoods).</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.opt_vv(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.opt_vv_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_vv_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_vv_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_vv_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.opt_vv_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the vv parameter updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood'>Compute the derivative of loglikelihood relevant to theta for step 1</h2><span id='topic+ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the loglikelihood function relevant to optimizing theta, not 
with respect to theta but with respect to unconstrained variables</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood(ww, tumordata, dd, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the theta weights corresponding to patient dd, a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood_+3A_dd">dd</code></td>
<td>
<p>the patient number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the loglikelihood function relevant to optimizing theta, not 
with respect to theta but with respect to unconstrained variables.</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.theta.theta_loglikelihood'>Compute the loglikelihood relevant to theta for step 1</h2><span id='topic+ISOpureS1.model_optimize.theta.theta_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing theta for step 1</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.theta.theta_loglikelihood(ww, tumordata, dd, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the theta weights corresponding to patient dd, a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_loglikelihood_+3A_dd">dd</code></td>
<td>
<p>the patient number</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.theta.theta_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to theta</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS1.model_optimize.vv.vv_compute_loglikelihood'>Compute loglikelihood relevant to vv for step 1</h2><span id='topic+ISOpureS1.model_optimize.vv.vv_compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing vv for step 1.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS1.model_optimize.vv.vv_compute_loglikelihood(vv, sum_log_theta, DD)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS1.model_optimize.vv.vv_compute_loglikelihood_+3A_vv">vv</code></td>
<td>
<p>Kx1 matrix representing the weights of the normal profiles B_i used to make the 
weighted combination that forms the mean parameter vector for the Dirichlet distribution
over m</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.vv.vv_compute_loglikelihood_+3A_sum_log_theta">sum_log_theta</code></td>
<td>
<p>the column sums of log(theta), a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpureS1.model_optimize.vv.vv_compute_loglikelihood_+3A_dd">DD</code></td>
<td>
<p>the number of patients (a scalar)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to optimizing vv</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_core.compute_loglikelihood'>Compute loglikelihood given all model parameters for step 2</h2><span id='topic+ISOpureS2.model_core.compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes complete loglikelihood given all model parameters for step 2</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_core.compute_loglikelihood(tumordata, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_core.compute_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.compute_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters updated in ISOpure step two iterations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The scalar value of the complete loglikelihood obtained given the model parameters</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_core.new_model'>Compute loglikelihood given all model parameters for step 2</h2><span id='topic+ISOpureS2.model_core.new_model'></span>

<h3>Description</h3>

<p>Produces a list (the model) which initializes the parameters vv, log_BBtranspose, 
PPtranspose, kappa, theta, omega, log_all_rates for step 2</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_core.new_model(tumordata, kappa, INITIAL_VV, PPtranspose, BBtranspose)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_core.new_model_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.new_model_+3A_kappa">kappa</code></td>
<td>
<p>a 1xD matrix which represents strength parameter kappa over cc, given the reference 
profile mm</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.new_model_+3A_initial_vv">INITIAL_VV</code></td>
<td>
<p>a vector with K components, the prior over mixing proportions, theta, with last 
entry weighed more heavily</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.new_model_+3A_pptranspose">PPtranspose</code></td>
<td>
<p>the prior on the tumor-specific cancer profiles is just the reference cancer
profile (1xG matrix) learned in ISOpureS1, standardized so that all entries sum to 1 </p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.new_model_+3A_bbtranspose">BBtranspose</code></td>
<td>
<p>a (K-1)xG matrix of the standardized normal profiles, so that they sum to 1</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>a newly generated model list to hold all the parameters</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_core.optmodel'>Optimizes the ISOpure parameters for step 2</h2><span id='topic+ISOpureS2.model_core.optmodel'></span>

<h3>Description</h3>

<p>Optimizes the ISOpure parameters for step 2 cyclically until convergence</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_core.optmodel(tumordata, model, NUM_ITERATIONS=35)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_core.optmodel_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.optmodel_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_core.optmodel_+3A_num_iterations">NUM_ITERATIONS</code></td>
<td>
<p>(optional) minimum number of iterations of optimization algorithm, 
default is 35</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>model</code></td>
<td>
<p>updated model list containing all the parameters</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood'>Compute the derivative of loglikelihood relevant to the patient cancer profiles, cc, for step 2</h2><span id='topic+ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the part of the likelihood function relevant to optimizing 
cc.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood(ww, tumordata, dd, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the cc_weights for patient dd, with G entries</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood_+3A_dd">dd</code></td>
<td>
<p>the patient number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the loglikelihood function relevant to optimizing cc for patient 
dd, the cancer profile for that patient. The derivative is taken not with respect to vv but 
with respect to unconstrained variables via a change of variables</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.cc.cc_loglikelihood'>Compute the loglikelihood relevant to the patient cancer profiles, cc, for step 2</h2><span id='topic+ISOpureS2.model_optimize.cc.cc_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing cc for patient 
dd, the cancer profile for that patient</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.cc.cc_loglikelihood(ww, tumordata, dd, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the cc_weights for patient dd, with G entries</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_loglikelihood_+3A_dd">dd</code></td>
<td>
<p>the patient number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.cc.cc_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative the part of the loglikelihood function relevant to optimizing cc for patient dd, 
the cancer profile for that patient.</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.kappa.kappa_compute_loglikelihood'>Compute loglikelihood relevant to kappa for step 2</h2><span id='topic+ISOpureS2.model_optimize.kappa.kappa_compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing kappa for step 2</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.kappa.kappa_compute_loglikelihood(kappa, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.kappa.kappa_compute_loglikelihood_+3A_kappa">kappa</code></td>
<td>
<p>a 1xK vector strength parameter in the prior over cc given the cancer profile mm</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.kappa.kappa_compute_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The part of the loglikelihood function relevant to optimizing kappa</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.kappa.kappa_deriv_loglikelihood'>Compute derivative of loglikelihood with respect to kappa for step 2</h2><span id='topic+ISOpureS2.model_optimize.kappa.kappa_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the part of the loglikelihood function relevant to 
optimizing kappa for step 2. Instead of performing constrained optimization on kappa 
directly, we optimize the log of kappa in an unconstrained fashion.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.kappa.kappa_deriv_loglikelihood(log_kappa, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.kappa.kappa_deriv_loglikelihood_+3A_log_kappa">log_kappa</code></td>
<td>
<p>the 1xD matrix log(kappa - model\$MIN_KAPPA)</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.kappa.kappa_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the part of the loglikelihood function relevant to kappa
with respect to log kappa (a Dx1 matrix).</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.kappa.kappa_loglikelihood'>Compute loglikelihood relevant to kappa for step 2</h2><span id='topic+ISOpureS2.model_optimize.kappa.kappa_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing kappa for step 2. 
Instead of performing constrained optimization on kappa directly, we optimize the log of kappa
in an unconstrained fashion.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.kappa.kappa_loglikelihood(log_kappa, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.kappa.kappa_loglikelihood_+3A_log_kappa">log_kappa</code></td>
<td>
<p>the 1xD matrix log(kappa - model\$MIN_KAPPA)</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.kappa.kappa_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to optimizing kappa</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.opt_cc'>Optimize the tumor-specific cancer profiles in step 2</h2><span id='topic+ISOpureS2.model_optimize.opt_cc'></span>

<h3>Description</h3>

<p>Optimize the tumor-specific cancer profiles.
Because cc is constrained (each cc_i are parameters of multinomial/discrete distribution), we 
don't directly optimize the likelihood function w.r.t. cc, but we perform change of variables 
to do unconstrained optimization.  We therefore store these unconstrained variables in the
field &quot;cc_weights&quot;, and update these variables.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.opt_cc(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.opt_cc_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_cc_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_cc_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_cc_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_cc_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with cc_weights and log_cc updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.opt_kappa'>Optimize kappa in step 2</h2><span id='topic+ISOpureS2.model_optimize.opt_kappa'></span>

<h3>Description</h3>

<p>This function optimizes kappa, the strength parameter in the prior over the reference
cancer profile.  Note that we don't directly optimize kappa because it has constraints (must be 
greater than the minimum determined in ISOpure.step2.PPE.)</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.opt_kappa(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.opt_kappa_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_kappa_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_kappa_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_kappa_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_kappa_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the kappa parameter (which is a 1xD vector) updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.opt_theta'>Optimize theta in step 2</h2><span id='topic+ISOpureS2.model_optimize.opt_theta'></span>

<h3>Description</h3>

<p>This function optimizes theta, in fact theta_weights.  Since thetas are constrained 
(must be parameters of multinomial/discrete distribution), we don't directly optimize the 
likelihood function w.r.t. theta, but we perform change of variables to do unconstrained
optimization.  We therefore store these unconstrained variables in the field &quot;theta_weights&quot;,
and update these variables.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.opt_theta(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.opt_theta_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_theta_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_theta_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_theta_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_theta_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the theta parameter updated (the first K-1 columns) corresponding to the 
normal sample contributions</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.opt_vv'>Optimize vv in step 2</h2><span id='topic+ISOpureS2.model_optimize.opt_vv'></span>

<h3>Description</h3>

<p>This function optimizes vv, the strength parameter in the prior over the reference
cancer profile.  Note that we don't directly optimize vv because it has constraints (must be &gt;=1
to guarantee real-valued likelihoods).</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.opt_vv(
	tumordata, 
	model, 
	NUM_ITERATIONS_RMINIMIZE, 
	iter, 
	NUM_GRID_SEARCH_ITERATIONS
	) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.opt_vv_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumour samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_vv_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_vv_+3A_num_iterations_rminimize">NUM_ITERATIONS_RMINIMIZE</code></td>
<td>
<p>minimum number of iteration that the minimization algorithm runs</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_vv_+3A_iter">iter</code></td>
<td>
<p>the iteration number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.opt_vv_+3A_num_grid_search_iterations">NUM_GRID_SEARCH_ITERATIONS</code></td>
<td>
<p>number of times to try restarting with different initial values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model with the vv parameter updated</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood'>Compute the derivative of loglikelihood relevant to theta for step 2</h2><span id='topic+ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the derivative of the loglikelihood function relevant to optimizing theta, not 
with respect to theta but with respect to unconstrained variables</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood(ww, tumordata, dd, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the theta weights corresponding to patient dd, a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood_+3A_dd">dd</code></td>
<td>
<p>the patient number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_deriv_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative derivative of the loglikelihood function relevant to optimizing theta, not 
with respect to theta but with respect to unconstrained variables.</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.theta.theta_loglikelihood'>Compute the loglikelihood relevant to theta for step 2</h2><span id='topic+ISOpureS2.model_optimize.theta.theta_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing theta for step 2</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.theta.theta_loglikelihood(ww, tumordata, dd, model)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>the theta weights corresponding to patient dd, a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_loglikelihood_+3A_tumordata">tumordata</code></td>
<td>
<p>a GxD matrix representing gene expression profiles of tumor samples</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_loglikelihood_+3A_dd">dd</code></td>
<td>
<p>the patient number</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.theta.theta_loglikelihood_+3A_model">model</code></td>
<td>
<p>list containing all the parameters to be optimized</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to theta</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

<hr>
<h2 id='ISOpureS2.model_optimize.vv.vv_compute_loglikelihood'>Compute loglikelihood relevant to vv for step 2</h2><span id='topic+ISOpureS2.model_optimize.vv.vv_compute_loglikelihood'></span>

<h3>Description</h3>

<p>Computes the part of the loglikelihood function relevant to optimizing vv for step 2.</p>


<h3>Usage</h3>

<pre><code class='language-R'>ISOpureS2.model_optimize.vv.vv_compute_loglikelihood(ww, sum_log_theta, D)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ISOpureS2.model_optimize.vv.vv_compute_loglikelihood_+3A_ww">ww</code></td>
<td>
<p>log(vv-1), a Kx1 matrix</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.vv.vv_compute_loglikelihood_+3A_sum_log_theta">sum_log_theta</code></td>
<td>
<p>the column sums of log(theta), a 1xK matrix</p>
</td></tr>
<tr><td><code id="ISOpureS2.model_optimize.vv.vv_compute_loglikelihood_+3A_d">D</code></td>
<td>
<p>the number of patients (a scalar)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The negative of the loglikelihood relevant to optimizing vv</p>


<h3>Author(s)</h3>

<p>Gerald Quon, Catalina Anghel, Francis Nguyen</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
