<!DOCTYPE html><html lang="en"><head><title>Help for package iZID</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {iZID}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bb.mle'><p>Maximum likelihood estimate for beta binomial distributions</p></a></li>
<li><a href='#bb.zihmle'><p>Maximum likelihood estimate for zero-inflated or hurdle beta binomial distributions.</p></a></li>
<li><a href='#dis.kstest'><p>The Monte Carlo estimate for the p-value of a discrete KS Test</p></a></li>
<li><a href='#model.lrt'><p>likelihood ratio test for two models</p></a></li>
<li><a href='#OTU'><p>Bacterial OTUs.</p></a></li>
<li><a href='#sample.h'><p>Generate random deviates from zero-inflated or hurdle models</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Identify Zero-Inflated Distributions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Computes bootstrapped Monte Carlo estimate of p value of Kolmogorov-Smirnov (KS) test and 
    likelihood ratio test for zero-inflated count data, based on the work of Aldirawi et al. (2019) 
    &lt;<a href="https://doi.org/10.1109%2FBHI.2019.8834661">doi:10.1109/BHI.2019.8834661</a>&gt;. With the package, user can also find tools to simulate 
    random deviates from zero inflated or hurdle models and obtain  maximum likelihood estimate
    of unknown parameters in these models.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>extraDistr (&ge; 1.8.11), methods (&ge; 3.1.0), rootSolve (&ge;
1.7), foreach (&ge; 1.4.7), doParallel (&ge; 1.0.15), stats (&ge;
3.1.0)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-06 08:49:55 UTC; Lenovo</td>
</tr>
<tr>
<td>Author:</td>
<td>Lei Wang [aut, cre, cph],
  Hani Aldirawi [aut, cph],
  Jie Yang [aut, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lei Wang &lt;slimewanglei@163.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-06 12:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bb.mle'>Maximum likelihood estimate for beta binomial distributions</h2><span id='topic+bb.mle'></span><span id='topic+bnb.mle'></span><span id='topic+nb.mle'></span><span id='topic+poisson.mle'></span>

<h3>Description</h3>

<p>calculate maximum likelihood estimate and the corresponding log likelihood value for beta binomial,
beta negative binomial, negative binomial and Poisson distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bb.mle(x, n, alpha1, alpha2, lowerbound = 0.01, upperbound = 10000)

bnb.mle(x, r, alpha1, alpha2, lowerbound = 0.01, upperbound = 10000)

nb.mle(x, r, p, lowerbound = 0.01, upperbound = 10000)

poisson.mle(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bb.mle_+3A_x">x</code></td>
<td>
<p>A vector of count data. Should be non-negative integers.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_n">n</code></td>
<td>
<p>An initial value of the number of trials. Must be a positive number, but not required to be an integer.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_alpha1">alpha1</code></td>
<td>
<p>An initial value for the first shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_alpha2">alpha2</code></td>
<td>
<p>An initial value for the second shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_lowerbound">lowerbound</code></td>
<td>
<p>A lower searching bound used in the optimization of likelihood function. Should be a small positive number.
The default is 1e-2.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_upperbound">upperbound</code></td>
<td>
<p>An upper searching bound used in the optimization of likelihood function. Should be a large positive number.
The default is 1e4.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_r">r</code></td>
<td>
<p>An initial value of the number of success before which m failures are observed, where m is the element of x.
Must be a positive number, but not required to be an integer.</p>
</td></tr>
<tr><td><code id="bb.mle_+3A_p">p</code></td>
<td>
<p>An initial value of the probability of success, should be a positive value within (0,1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>bb.mle</code>, <code>bnb.mle</code>, <code>nb.mle</code> and <code>poisson.mle</code> calculate the maximum likelihood estimate of beta binomial,
beta negative binomial, negative binomial and Poisson distributions, respectively.
</p>
<p>Please NOTE that the arguments in the four functions are NOT CHECKED AT ALL! The user must be aware of their inputs to avoid
getting suspicious results.
</p>
<p>Suppose that <code class="reqn">X</code> is a random count variable that only takes non-negative values. If <code class="reqn">p</code> has a prior distribution
<code class="reqn">beta(alpha1,alpha2)</code> and <code class="reqn">X</code> follows a binomial distribution <code class="reqn">b(n,p)</code>, then <code class="reqn">X</code> follows the beta binomial
distribution with
</p>
<p><code class="reqn">P(X=k)=C(n,k)Beta(k+alpha1,n-k+alpha2)/Beta(alpha1,alpha2)</code>,
</p>
<p>where <code class="reqn">C(,)</code> is the combination function, <code class="reqn">Beta(,)</code> is the beta function and <code class="reqn">beta(,)</code> stands for the beta distribution.
</p>
<p>If <code class="reqn">X</code> stands for the number of failures observed before the <code class="reqn">r</code>th success, the probability of <code class="reqn">X</code> taking the
value <code class="reqn">k</code> under the negative binomial distribution equals
</p>
<p><code class="reqn">P(X=k)=C(k+r-1,k)p^r(1-p)^k</code>,
</p>
<p>As in beta binomial distribution, assume the prior distribution of <code class="reqn">p</code> is <code class="reqn">beta(alpha1,alpha2)</code>. <code class="reqn">X</code> follows
a beta negative binomial distribution if <code class="reqn">X</code> follows a negative binomial distribution with parameters <code class="reqn">r</code> and <code class="reqn">p</code>.
The probability density function of a beta negative binomial distribution is defined as:
</p>
<p><code class="reqn">P(X=k)=\Gamma(r+k)Beta(r+alpha1,k+alpha2)/Beta(alpha1,alpha2)/\Gamma(r)/k!</code>,
</p>
<p>where <code class="reqn">\Gamma</code> represents the Gamma function.
</p>
<p>With the only parameter <code class="reqn">lambda</code>, the probability density function of a Poisson distribution is
</p>
<p><code class="reqn">P(X=k)=lambda^k exp(-lambda)/k!</code>
</p>
<p>The maximum likelihood estimate of all four distributions can be derived by minimizing the corresponding negative log likelihood
function. It is easy to deduce the sample estimate of <code class="reqn">lambda</code> which is equal to the sample mean. However, it is not so
straightforward to solve the optimization problems of the other three distributions. Thus, we adopt the optimization
algorithm &quot;L-BFGS-B&quot; by calling R basic
function <code>optim</code>. Lower and upper bounds on the unknown parameters are required for the algorithm &quot;L-BFGS-B&quot;, which are
determined by the arguments <code>lowerbound</code> and <code>upperbound</code>. But note that for the estimate of <code class="reqn">p</code>, the upper bound
for searching is essentially <code>1-lowerbound</code>.
</p>


<h3>Value</h3>

<p>A row vector containing the maximum likelihood estimate of unknown parameters and the corresponding value of log likelihood.
</p>
<p>With <code class="reqn">bb.mle</code>, the following values are returned:
</p>

<ul>
<li><p>n: the maximum likelihood estimate of n.
</p>
</li>
<li><p>alpha1: the maximum likelihood estimate of alpha1.
</p>
</li>
<li><p>alpha2: the maximum likelihood estimate of alpha2.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimates plugged-in.
</p>
</li></ul>

<p>With <code class="reqn">bnb.mle</code>, the following values are returned:
</p>

<ul>
<li><p>r: the maximum likelihood estimate of r.
</p>
</li>
<li><p>alpha1: the maximum likelihood estimate of alpha1.
</p>
</li>
<li><p>alpha2: the maximum likelihood estimate of alpha2.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimates plugged-in.
</p>
</li></ul>

<p>With <code class="reqn">nb.mle</code>, the following values are returned:
</p>

<ul>
<li><p>r: the maximum likelihood estimate of r.
</p>
</li>
<li><p>p: the maximum likelihood estimate of p.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimates plugged-in.
</p>
</li></ul>

<p>With <code class="reqn">poisson.mle</code>, the following values are returned:
</p>

<ul>
<li><p>lambda: the maximum likelihood estimate of lambda.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimate plugged-in.
</p>
</li></ul>



<h3>Reference</h3>


<ul>
<li><p>H. Aldirawi, J. Yang, A. A. Metwally (2019). Identifying Appropriate Probabilistic Models for Sparse Discrete Omics Data,
accepted for publication in 2019 IEEE EMBS International Conference on Biomedical &amp; Health Informatics (BHI).
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>x=extraDistr::rbbinom(2000,12,2,4)
bb.mle(x,3,1,1)
x=extraDistr::rbnbinom(2000,8,3,5)
bnb.mle(x, 3.3, 1, 1)
x=stats::rnbinom(2000,size=5,prob=0.3)
nb.mle(x, 7, 0.5)
x=stats::rpois(2000,7)
poisson.mle(x)
</code></pre>

<hr>
<h2 id='bb.zihmle'>Maximum likelihood estimate for zero-inflated or hurdle beta binomial distributions.</h2><span id='topic+bb.zihmle'></span><span id='topic+bnb.zihmle'></span><span id='topic+nb.zihmle'></span><span id='topic+poisson.zihmle'></span>

<h3>Description</h3>

<p>Calculate maximum likelihood estimate and the corresponding log likelihood value for zero-inflated or hurdle
beta binomial, beta negative binomial, negative binomial and Poisson distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bb.zihmle(x, n, alpha1, alpha2, type = c("zi", "h"), lowerbound = 0.01,
  upperbound = 10000)

bnb.zihmle(x, r, alpha1, alpha2, type = c("zi", "h"),
  lowerbound = 0.01, upperbound = 10000)

nb.zihmle(x, r, p, type = c("zi", "h"), lowerbound = 0.01,
  upperbound = 10000)

poisson.zihmle(x, type = c("zi", "h"), lowerbound = 0.01,
  upperbound = 10000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bb.zihmle_+3A_x">x</code></td>
<td>
<p>A vector of count data. Should be non-negative integers.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_n">n</code></td>
<td>
<p>An initial value of the number of trials. Must be a positive number, but not required to be an integer.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_alpha1">alpha1</code></td>
<td>
<p>An initial value for the first shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_alpha2">alpha2</code></td>
<td>
<p>An initial value for the second shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_type">type</code></td>
<td>
<p>The type of distribution used to calculate the sample estimate, where 'zi' and 'h' stand for zero-inflated
and hurdle distributions respectively.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_lowerbound">lowerbound</code></td>
<td>
<p>A lower searching bound used in the optimization of likelihood function. Should be a small positive number.
The default is 1e-2.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_upperbound">upperbound</code></td>
<td>
<p>An upper searching bound used in the optimization of likelihood function. Should be a large positive number.
The default is 1e4.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_r">r</code></td>
<td>
<p>An initial value of the number of success before which m failures are observed, where m is the element of x.
Must be a positive number, but not required to be an integer.</p>
</td></tr>
<tr><td><code id="bb.zihmle_+3A_p">p</code></td>
<td>
<p>An initial value of the probability of success, should be a positive value within (0,1).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By setting <code>type='zi'</code>, <code>bb.zihmle</code>, <code>bnb.zihmle</code>, <code>nb.zihmle</code> and <code>poisson.zihmle</code> calculate the
maximum likelihood estimate of zero-inflated beta binomial, beta negative binomial, negative binomial and Poisson
distributions, respectively.
</p>
<p>By setting <code>type='h'</code>, <code>bb.zihmle</code>, <code>bnb.zihmle</code>, <code>nb.zihmle</code> and <code>poisson.zihmle</code> calculate the
maximum likelihood estimate of hurdle beta binomial, beta negative binomial, negative binomial and Poisson
distributions, respectively.
</p>
<p>Please NOTE that the arguments in the four functions are NOT CHECKED AT ALL! The user must be aware of their inputs to avoid
getting suspicious results.
</p>
<p>For zero-inflated models, zeros occurred by either sampling process or specific structure of data with the structural parameter
<code class="reqn">0&lt;\phi&lt;1</code>. The density function for a zero-inflated model is
</p>
<p><code class="reqn">P_{zi}(X=k)=\phi 1_{k=0}+(1-\phi)P(X=k)</code>,
</p>
<p>where <code class="reqn">P(X=k)</code> is the probability under standard distributions.
</p>
<p>Aldirawi et al. (2019) proposed an estimating procedure for zero-inflated models by optimizing over a reparametrization of
the likelihood function where <code class="reqn">\phi</code> and the rest unknown parameters are separable. When <code class="reqn">X</code> comes from a
zero-inflated distribution, the maximum likelihood estimate of parameters except for <code class="reqn">\phi</code> are obtained by minimizing
the truncated version of negative log likelihood function. However, in the zero-deflated case, <code class="reqn">\phi=0</code> and the sample
estimate of other parameters are identical to those for its corresponding standard distributions. Meanwhile, an warning
message is shown on the screen such that 'cannot obtain mle with the current model type, the output estimate is derived from
general ... distribution'.
</p>
<p>For hurdle models, all zeros occurred purely by the structure of data with the structural parameter <code class="reqn">0&lt;\phi&lt;1</code>.
The density function for a hurdle model is
</p>
<p><code class="reqn">P_{h}(X=k)=\phi 1_{k=0}+(1-\phi)P_{tr}(X=k)</code>,
</p>
<p>where <code class="reqn">P_{tr}(X=k)</code> is the truncated probability under standard distributions, where <code class="reqn">P_{tr}(X=0)=0</code> and
<code class="reqn">P_{tr}(X=k)=P(X=k)/(1-P(X=0))</code>. Since <code class="reqn">\phi</code> and other unknown parameters are separable in the joint likelihood
function, <code class="reqn">\phi</code> can be estimated by a value with respect to the number of positive samples. The sample estimate of
other parameters can be obtained by the same procedure for zero-inflated model.
</p>
<p>A warning message may also occur when the algorithm of <code>optim</code> does not converge and the resulting estimates are not
valid. In this case, the results from the corresponding general distribution are output instead.
</p>


<h3>Value</h3>

<p>A row vector containing the maximum likelihood estimate of unknown parameters and the corresponding value of log likelihood.
</p>
<p>With <code>bb.zihmle</code>, the following values are returned:
</p>

<ul>
<li><p>n: the maximum likelihood estimate of n.
</p>
</li>
<li><p>alpha1: the maximum likelihood estimate of alpha1.
</p>
</li>
<li><p>alpha2: the maximum likelihood estimate of alpha2.
</p>
</li>
<li><p>phi: the maximum likelihood estimate of <code class="reqn">\phi</code>.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimates plugged-in.
</p>
</li></ul>

<p>With <code>bnb.zihmle</code>, the following values are returned:
</p>

<ul>
<li><p>r: the maximum likelihood estimate of r.
</p>
</li>
<li><p>alpha1: the maximum likelihood estimate of alpha1.
</p>
</li>
<li><p>alpha2: the maximum likelihood estimate of alpha2.
</p>
</li>
<li><p>phi: the maximum likelihood estimate of <code class="reqn">\phi</code>.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimates plugged-in.
</p>
</li></ul>

<p>With <code>nb.zihmle</code>, the following values are returned:
</p>

<ul>
<li><p>r: the maximum likelihood estimate of r.
</p>
</li>
<li><p>p: the maximum likelihood estimate of p.
</p>
</li>
<li><p>phi: the maximum likelihood estimate of <code class="reqn">\phi</code>.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimates plugged-in.
</p>
</li></ul>

<p>With <code>poisson.zihmle</code>, the following values are returned:
</p>

<ul>
<li><p>lambda: the maximum likelihood estimate of lambda.
</p>
</li>
<li><p>phi: the maximum likelihood estimate of <code class="reqn">\phi</code>.
</p>
</li>
<li><p>loglik: the value of log likelihood with maximum likelihood estimate plugged-in.
</p>
</li></ul>



<h3>Reference</h3>


<ul>
<li><p>H. Aldirawi, J. Yang, A. A. Metwally (2019). Identifying Appropriate Probabilistic Models for Sparse Discrete Omics Data,
accepted for publication in 2019 IEEE EMBS International Conference on Biomedical &amp; Health Informatics (BHI).
</p>
</li>
<li><p>H. Aldirawi, J. Yang (2019). Model Selection and Regression Analysis for Zero-altered or Zero-inflated Data, Statistical
Laboratory Technical Report, no.2019-01, University of Illinois at Chicago.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>t1=sample.h(N=2000,phi=0.2,distri='Poisson',lambda=5)  ##hurdle poisson random values
t2=sample.h(N=2000,phi=0.2,distri='nb',r=10,p=0.6)   ##hurdle negative binomial
t3=sample.zi(N=2000,phi=0.2,distri='bb',alpha1=8,alpha2=9,n=10)   ##zero-inflated beta binomial
##zero-inflated beta negative binomial.
t4=sample.zi(N=2000,phi=0.2,distri='bnb',r=10,alpha1=8,alpha2=9)
bb.zihmle(t3,3,1,1,type='h')
bnb.zihmle(t4, 3.3, 1, 1,type='h')
nb.zihmle(t2, 7, 0.5,type='zi')
poisson.zihmle(t1,type='zi')
</code></pre>

<hr>
<h2 id='dis.kstest'>The Monte Carlo estimate for the p-value of a discrete KS Test</h2><span id='topic+dis.kstest'></span>

<h3>Description</h3>

<p>Computes the Monte Carlo estimate for the p-value of a discrete one-sample Kolmogorov-Smirnov (KS) Test for Poisson, negative
binomial, beta binomial, beta negative binomial distributions and their zero-inflated as well as hurdle versions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dis.kstest(x, nsim = 100, bootstrap = TRUE, distri = "Poisson",
  r = NULL, p = NULL, alpha1 = NULL, alpha2 = NULL, n = NULL,
  lowerbound = 0.01, upperbound = 10000, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dis.kstest_+3A_x">x</code></td>
<td>
<p>A vector of count data. Should be non-negative integers. If elements of x are not integers, they will be
automatically rounded up to the smallest integers that are no less than themselves.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_nsim">nsim</code></td>
<td>
<p>The number of bootstrapped samples or simulated samples generated to compute p-value. If it is not an integer,
nsim will be automatically rounded up to the smallest integer that is no less than nsim. Should be greater than 30. Default is
100.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_bootstrap">bootstrap</code></td>
<td>
<p>Whether to generate bootstrapped samples or not. See Details. 'TRUE' or any numeric non-zero value indicates
the generation of bootstrapped samples. The default is 'TRUE'.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_distri">distri</code></td>
<td>
<p>The distribution used as the null hypothesis. Can be one of {'poisson','nb','bb',
'bnb','zip','zinb','zibb', zibnb','ph','nbh','bbh','bnbh'}, which corresponds to Poisson, negative binomial, beta binomial
and beta negative binomial distributions and their zero-inflated as well as hurdle versions, respectively. Default is 'Poisson'.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_r">r</code></td>
<td>
<p>An initial value of the number of success before which m failures are observed, where m is the element of x.
Must be a positive number, but not required to be an integer.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_p">p</code></td>
<td>
<p>An initial value of the probability of success, should be a positive value within (0,1).</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_alpha1">alpha1</code></td>
<td>
<p>An initial value for the first shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_alpha2">alpha2</code></td>
<td>
<p>An initial value for the second shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_n">n</code></td>
<td>
<p>An initial value of the number of trials. Must be a positive number, but not required to be an integer.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_lowerbound">lowerbound</code></td>
<td>
<p>A lower searching bound used in the optimization of likelihood function. Should be a small positive number.
The default is 1e-2.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_upperbound">upperbound</code></td>
<td>
<p>An upper searching bound used in the optimization of likelihood function. Should be a large positive number.
The default is 1e4.</p>
</td></tr>
<tr><td><code id="dis.kstest_+3A_parallel">parallel</code></td>
<td>
<p>whether to use multiple threads to parallelize computation. Default is FALSE. Please aware that it may take
longer time to execute the program with <code>parallel=FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For arguments <code>nsim</code>, <code>bootstrap</code>, <code>distri</code>, if the length is larger than 1, only the first element will be used.
For other arguments except for <code>x</code>, the first valid value will be used if the input is not <code>NULL</code>, otherwise some
naive sample estimates will be fed into the algorithm. Note that only the initial values that are occurred in the null
distribution <code>distri</code> are needed. For example, with <code>distri=poisson</code>, user may provide a value for <code>lambda</code> but
not for <code>r</code> or <code>p</code>, though it won't disturb the algorithm.
</p>
<p>With an output p-value less than some user-specified significance level, <code>x</code> is very likely from a distribution other
than the <code>distri</code>, given the current data. If p-values of more than one distributions are greater than the pre-specified
significance level, user may consider a following likelihood ratio test to select a 'better' distribution.
</p>
<p>The methodology of computing Monte Carlo p-value is taken from Aldirawi et al. (2019). With <code>bootstrap=TRUE</code>, <code>nsim</code>
bootstrapped samples will be generated by resampling <code>x</code> without replacement. Otherwise, <code>nsim</code> samples are
simulated from the null distribution with the maximum likelihood estimate of original data <code>x</code>. Then compute the maximum
likelihood estimates of <code>nsim</code> bootstrapped or simulated samples, based on which <code>nsim</code> new samples are generated
under the null distribution. <code>nsim</code> KS statistics are calculated for the <code>nsim</code> new samples, then the Monte Carlo
p-value is resulted from comparing the <code>nsim</code> KS statistics and the statistic of original data <code>x</code>.
</p>
<p>During the process of computing maximum likelihood estimates, the negative log likelihood function is minimized via basic R
function <code>optim</code> with the searching interval decided by <code>lowerbound</code> and <code>upperbound</code>, except that the optimization
of <code>p</code> takes <code>1-lowerbound</code> as the upper searching bound.
</p>
<p>To accelerate the whole process, the algorithm uses the parallel strategy via the packages <code>foreach</code> and
<code>doParallel</code>.
</p>


<h3>Value</h3>

<p>An object of class 'dis.kstest' including the following elements:
</p>

<ul>
<li><p>x: <code>x</code> used in computation.
</p>
</li>
<li><p>nsim: nsim used in computation.
</p>
</li>
<li><p>bootstrap: bootstrap used in computation.
</p>
</li>
<li><p>distri: distri used in computation..
</p>
</li>
<li><p>lowerbound: lowerbound used in computation.
</p>
</li>
<li><p>upperbound: upperboound used in computation.
</p>
</li>
<li><p>mle_new: A matrix of the maximum likelihood estimates of unknown parameters under the null distribution, using
<code class="reqn">nsim</code> bootstrapped or simulated samples.
</p>
</li>
<li><p>mle_ori: A row vector of the maximum likelihood estimates of unknown parameters under the null distribution, using the
original data <code>x</code>.
</p>
</li>
<li><p>pvalue: Monte Carlo p-value of the one-sample KS test.
</p>
</li>
<li><p>N: length of <code>x</code>.
</p>
</li>
<li><p>r: initial value of r used in computation.
</p>
</li>
<li><p>p: initial value of p used in computation.
</p>
</li>
<li><p>alpha1: initial value of alpha1 used in computation.
</p>
</li>
<li><p>alpha2: initial value of alpha2 used in computation.
</p>
</li>
<li><p>n: initial value of n used in computation.
</p>
</li></ul>



<h3>Reference</h3>


<ul>
<li><p>H. Aldirawi, J. Yang, A. A. Metwally (2019). Identifying Appropriate Probabilistic Models for Sparse Discrete Omics Data,
accepted for publication in 2019 IEEE EMBS International Conference on Biomedical &amp; Health Informatics (BHI).
</p>
</li>
<li><p>T. Wolodzko (2019). extraDistr: Additional Univariate and Multivariate Distributions, R package version 1.8.11,
https://CRAN.R-project.org/package=extraDistr.
</p>
</li>
<li><p>R. Calaway, Microsoft Corporation, S. Weston, D. Tenenbaum (2017). doParallel: Foreach Parallel Adaptor for the 'parallel'
Package, R package version 1.0.11, https://CRAN.R-project.org/package=doParallel.
</p>
</li>
<li><p>R. Calaway, Microsoft, S. Weston (2017). foreach: Provides Foreach Looping Construct for R, R package version 1.4.4,
https://CRAN.R-project.org/package=foreach.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+model.lrt">model.lrt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(2001)
temp1=sample.zi(N=300,phi=0.3,distri='poisson',lambda=5)
dis.kstest(temp1,nsim=100,bootstrap=TRUE,distri='Poisson')$pvalue
dis.kstest(temp1,nsim=100,bootstrap=TRUE,distri='nb')$pvalue
dis.kstest(temp1,nsim=100,bootstrap=TRUE,distri='zip')$pvalue
dis.kstest(temp1,nsim=100,bootstrap=TRUE,distri='zinb')$pvalue

</code></pre>

<hr>
<h2 id='model.lrt'>likelihood ratio test for two models</h2><span id='topic+model.lrt'></span>

<h3>Description</h3>

<p>Conduct likelihood ratio test for comparing two different models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model.lrt(d1, d2, parallel = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="model.lrt_+3A_d1">d1</code></td>
<td>
<p>An object of class 'dis.kstest'.</p>
</td></tr>
<tr><td><code id="model.lrt_+3A_d2">d2</code></td>
<td>
<p>An object of class 'dis.kstest'.</p>
</td></tr>
<tr><td><code id="model.lrt_+3A_parallel">parallel</code></td>
<td>
<p>Whether to use multiple threads to parallelize computation. Default is FALSE. Please aware that it may take
longer time to execute the program with <code>parallel=FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the <code>pvalue</code> of <code>d1</code> and <code>d2</code> are greater than the user-specified significance level, which indicates that
the original data <code>x</code> may come from the two distributions in <code>d1</code> and <code>d2</code>, a likelihood ratio test is
desired to choose a more 'possible' distribution based on the current data. NOTE that the <code>x</code> in <code>d1</code> and <code>d2</code>
must be IDENTICAL! Besides, NOTE that the <code>distri</code> in <code>d1</code> and <code>d2</code> must be DIFFERENT!
</p>
<p>The <code>distri</code> inherited from <code>d1</code> is the null distribution and that from <code>d2</code> is used as the alternative
distribution. Following Aldirawi et al. (2019), <code>nsim</code> bootstrapped or simulated samples will be generated according to
<code>bootstrap</code> of <code>d1</code>, based on which <code>nsim</code> maximum likelihood estimates of the parameters in null distribution
will be calculated. Remember that we have obtained <code>nsim</code> such maximum likelihood estimates while calling function
<code>dis.kstest</code>. Thus, the algorithm just adopts the <code>mle_new</code> from <code>d1</code> to avoid repetitive work. Using the
<code>nsim</code> maximum likelihood estimates to generate <code>nsim</code> new samples and calculate <code>nsim</code> corresponding new
likelihood ratio test statistics. The output p-value is the proportion of new samples that have statistics greater than the
test statistic of the original data <code>x</code>.
</p>
<p>As in <code><a href="#topic+dis.kstest">dis.kstest</a></code>, the computation is parallelized with the help of packages <code>foreach</code> and
<code>doParallel</code>.
</p>
<p>With the output p-value smaller than the user-specified significance level, the <code>distri</code> of <code>d2</code> is more appropriate
for modelling <code>x</code>. Otherwise, There is no significant difference between <code>distri</code> of <code>d1</code> and <code>distri</code> of <code>d2</code>,
given the current data.
</p>


<h3>Value</h3>

<p>The p-value of the likelihood ratio test.
</p>


<h3>Reference</h3>


<ul>
<li><p>H. Aldirawi, J. Yang, A. A. Metwally (2019). Identifying Appropriate Probabilistic Models for Sparse Discrete Omics Data,
accepted for publication in 2019 IEEE EMBS International Conference on Biomedical &amp; Health Informatics (BHI).
</p>
</li>
<li><p>T. Wolodzko (2019). extraDistr: Additional Univariate and Multivariate Distributions, R package version 1.8.11,
https://CRAN.R-project.org/package=extraDistr.
</p>
</li>
<li><p>R. Calaway, Microsoft Corporation, S. Weston, D. Tenenbaum (2017). doParallel: Foreach Parallel Adaptor for the 'parallel'
Package, R package version 1.0.11, https://CRAN.R-project.org/package=doParallel.
</p>
</li>
<li><p>R. Calaway, Microsoft, S. Weston (2017). foreach: Provides Foreach Looping Construct for R, R package version 1.4.4,
https://CRAN.R-project.org/package=foreach.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+dis.kstest">dis.kstest</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2001)
temp1=sample.zi(N=300,phi=0.3,distri='poisson',lambda=5)
d1=dis.kstest(temp1,nsim=100,bootstrap=TRUE,distri='zip')
d2=dis.kstest(temp1,nsim=100,bootstrap=TRUE,distri='zinb')
model.lrt(d1,d2)
</code></pre>

<hr>
<h2 id='OTU'>Bacterial OTUs.</h2><span id='topic+OTU'></span>

<h3>Description</h3>

<p>A dataset containing the OTUs (Operational Taxonomic Units) of 229 bacteria.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OTU
</code></pre>


<h3>Format</h3>

<p>The original data set is a skin microbiome from the National Human Genome Research Institute [1,2], which contains 229 bacterial and
fungal OTUs. The OTU data attached to this package is a data frame with 229 rows and 354 columns, where each row means one kind of bacteria and
each column represents a feature.
The first column <code>x</code> is the name of the bacteria. The rest of columns are the total number of reads for different features.
For example, the second column is the total number of reads for feature 'Center.Back.01', and so forth.</p>


<h3>Source</h3>


<ul>
<li><p>1. K. Findley, J. Oh, J. Yang, S. Conlan, C. Deming, J.A. Meyer, et al (2013). Human skin fungal diversity, Nature, 498:367&ndash;70. 
</p>
</li>
<li><p>2. E.A. Grice, H.H. Kong, S. Conlan, C.B. Deming, J. Davis, A.C. Young, et al (2009). Topographical and temporal diversity of the human skin
microbiome, Science, 324:1190&ndash;2.</p>
</li></ul>


<hr>
<h2 id='sample.h'>Generate random deviates from zero-inflated or hurdle models</h2><span id='topic+sample.h'></span><span id='topic+sample.zi'></span>

<h3>Description</h3>

<p>Generate random deviates from zero-inflated or hurdle Poisson, negative binomial, beta binomial and beta negative binomial
models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample.h(N, phi, distri = "poisson", lambda = NA, r = NA, p = NA,
  alpha1 = NA, alpha2 = NA, n = NA)

sample.zi(N, phi, distri = "poisson", lambda = NA, r = NA, p = NA,
  alpha1 = NA, alpha2 = NA, n = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sample.h_+3A_n">N</code></td>
<td>
<p>The sample size. Should be a positive number. If it is not an integer, N will be automatically rounded up to the
smallest integer that no less than N.</p>
</td></tr>
<tr><td><code id="sample.h_+3A_phi">phi</code></td>
<td>
<p>The structural parameter <code class="reqn">\phi</code>, should be a positive value within (0,1).</p>
</td></tr>
<tr><td><code id="sample.h_+3A_distri">distri</code></td>
<td>
<p>The corresponding standard distribution. Can be one of {'poisson','nb','bb', 'bnb'}, which corresponds to Poisson,
negative binomial, beta binomial and beta negative binomial distributions respectively.</p>
</td></tr>
<tr><td><code id="sample.h_+3A_lambda">lambda</code></td>
<td>
<p>A value for the parameter of Poisson distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="sample.h_+3A_r">r</code></td>
<td>
<p>the number of success before which m failures are observed, where m is a random variable from negative binomial
or beta negative binomial distribution. Must be a positive number. If it is not an integer, r will be automatically rounded up to the
smallest integer that no less than r.</p>
</td></tr>
<tr><td><code id="sample.h_+3A_p">p</code></td>
<td>
<p>The probability of success, should be a positive value within (0,1).</p>
</td></tr>
<tr><td><code id="sample.h_+3A_alpha1">alpha1</code></td>
<td>
<p>The first shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="sample.h_+3A_alpha2">alpha2</code></td>
<td>
<p>The second shape parameter of beta distribution. Should be a positive number.</p>
</td></tr>
<tr><td><code id="sample.h_+3A_n">n</code></td>
<td>
<p>The number of trials. Must be a positive number. If it is not an integer, n will be automatically rounded up to the
smallest integer that no less than n.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By setting <code>distri=poisson</code>, <code>sample.h</code> and <code>sample.zi</code> simulates <code>N</code> random deviates from hurdle and
zero-inflated Poisson distribution, respectively, and so on forth. For arguments with length larger than 1, only the first
element will be used.
</p>
<p>Arguments <code>r</code> and <code>p</code> are for the use of zero-inflated and hurdle negative binomial distributions. <code>alpha1</code>,
<code>alpha2</code> and <code>n</code> are for
zero-inflated and hurdle beta binomial distributions. <code>r</code>, <code>alpha1</code> and <code>alpha2</code> are used in zero-inflated and hurdle beta
negative binomial distributions.
</p>
<p>The procedure of generating random deviates follows the work of Aldirawi et al. (2019). The algorithm calls functions for
standard distributions to simulate the non-zero samples. Random deviates from standard Poisson and negative binomial
distributions can be generated by basic R function <code>rpois</code> and <code>rnbinom</code>. Functions rbbinom and rbnbinom are
available for standard beta binomial and beta negative binomial distributions in R package <code>extraDistr</code>.
</p>


<h3>Value</h3>

<p>A vector of length <code class="reqn">N</code> containing non-negative integers from the zero-inflated or hurdle version of distribution
determined by <code class="reqn">distri</code>.
</p>


<h3>Reference</h3>


<ul>
<li><p>H. Aldirawi, J. Yang, A. A. Metwally, Identifying Appropriate Probabilistic Models for Sparse Discrete Omics Data,
accepted for publication in 2019 IEEE EMBS International Conference on Biomedical &amp; Health Informatics (BHI) (2019).
</p>
</li>
<li><p>T. Wolodzko, extraDistr: Additional Univariate and Multivariate Distributions, R package version 1.8.11 (2019),
https://CRAN.R-project.org/package=extraDistr.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>t1=sample.h(N=2000,phi=0.2,distri='Poisson',lambda=5)  ##hurdle poisson random values
t2=sample.h(N=2000,phi=0.2,distri='nb',r=10,p=0.6)   ##hurdle negative binomial
t3=sample.h(N=2000,phi=0.2,distri='bb',alpha1=8,alpha2=9,n=10)   ##hurdle beta binomial
##hurdle beta negative binomial.
t4=sample.h(N=2000,phi=0.2,distri='bnb',r=10,alpha1=8,alpha2=9)

t1=sample.zi(N=2000,phi=0.2,distri='Poisson',lambda=5)  ##zero-inflated poisson random values
t2=sample.zi(N=2000,phi=0.2,distri='nb',r=10,p=0.6)   ##zero-inflated negative binomial
t3=sample.zi(N=2000,phi=0.2,distri='bb',alpha1=8,alpha2=9,n=10)   ##zero-inflated beta binomial
##zero-inflated beta negative binomial
t4=sample.zi(N=2000,phi=0.2,distri='bnb',r=10,alpha1=8,alpha2=9)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
