<!DOCTYPE html><html><head><title>Help for package jackstraw</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {jackstraw}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#find_k'><p>Find a number of clusters or principal components</p></a></li>
<li><a href='#jackstraw'><p>jackstraw: Statistical Inference for Unsupervised Learning</p></a></li>
<li><a href='#jackstraw_cluster'><p>Jackstraw for the User-Defined Clustering Algorithm</p></a></li>
<li><a href='#jackstraw_irlba'><p>Non-Parametric Jackstraw for Principal Component Analysis (PCA) using the augmented implicitly restarted Lanczos bidiagonalization algorithm (IRLBA)</p></a></li>
<li><a href='#jackstraw_kmeans'><p>Non-Parametric Jackstraw for K-means Clustering</p></a></li>
<li><a href='#jackstraw_kmeanspp'><p>Non-Parametric Jackstraw for K-means Clustering using RcppArmadillo</p></a></li>
<li><a href='#jackstraw_MiniBatchKmeans'><p>Non-Parametric Jackstraw for Mini Batch K-means Clustering</p></a></li>
<li><a href='#jackstraw_pam'><p>Non-Parametric Jackstraw for Partitioning Around Medoids (PAM)</p></a></li>
<li><a href='#jackstraw_pca'><p>Non-Parametric Jackstraw for Principal Component Analysis (PCA)</p></a></li>
<li><a href='#jackstraw_rpca'><p>Non-Parametric Jackstraw for Principal Component Analysis (PCA) using Randomized Singular Value Decomposition</p></a></li>
<li><a href='#jackstraw_subspace'><p>Jackstraw for the User-Defined Dimension Reduction Methods</p></a></li>
<li><a href='#Jurkat293T'><p>A Jurkat:293T equal mixture dataset from Zheng et al. (2017)</p></a></li>
<li><a href='#permutationPA'><p>Permutation Parallel Analysis</p></a></li>
<li><a href='#pip'><p>Compute posterior inclusion probabilities (PIPs)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Inference for Unsupervised Learning</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.9</td>
</tr>
<tr>
<td>Author:</td>
<td>Neo Christopher Chung &lt;nchchung@gmail.com&gt;, John D. Storey
    &lt;jstorey@princeton.edu&gt;, Wei Hao &lt;whao@princeton.edu&gt;, Alejandro Ochoa &lt;alejandro.ochoa@duke.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Neo Christopher Chung &lt;nchchung@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Test for association between the observed data and their estimated latent variables. The jackstraw package provides a resampling strategy and testing scheme to estimate statistical significance of association between the observed data and their latent variables. Depending on the data type and the analysis aim, the latent variables may be estimated by principal component analysis (PCA), factor analysis (FA), K-means clustering, and related algorithms. The jackstraw methods learn over-fitting characteristics inherent in this circular analysis, where the observed data are used to estimate the latent variables and used again to test against that estimated latent variables. When latent variables are estimated by PCA, the jackstraw enables statistical testing for association between observed variables and latent variables, as estimated by low-dimensional principal components (PCs). This essentially leads to identifying variables that are significantly associated with PCs. Similarly, unsupervised clustering, such as K-means clustering, partition around medoids (PAM), and others, finds coherent groups in high-dimensional data. The jackstraw estimates statistical significance of cluster membership, by testing association between data and cluster centers. Clustering membership can be improved by using the resulting jackstraw p-values and posterior inclusion probabilities (PIPs), with an application to unsupervised evaluation of cell identities in single cell RNA-seq.</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, stats, qvalue, corpcor, irlba, rsvd, ClusterR,
cluster</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-06 11:00:56 UTC; ncc</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-07 12:50:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='find_k'>Find a number of clusters or principal components</h2><span id='topic+find_k'></span>

<h3>Description</h3>

<p>There are a wide range of algorithms and visual techniques to identify
a number of clusters or principal components embeded in the observed data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>find_k()
</code></pre>


<h3>Details</h3>

<p>It is critical to explore the eigenvalues, cluster stability, and visualization.
See R packages <code>bootcluster</code>, <code>EMCluster</code>, and <code>nFactors</code>.
</p>
<p>Please see the R package <code>SC3</code>, which provides <code>estkTW()</code> function to
find the number of significant eigenvalues according to the Tracy-Widom test.
</p>
<p><code>ADPclust</code> package includes <code>adpclust()</code> function that runs the algorithm
on a range of K values. It helps you to identify the most suitable number of clusters.
</p>
<p>This package also provides an alternative methods in <code>permutationPA</code>.
Through a resampling-based Parallel Analysis, it finds a number of significant components.
</p>

<hr>
<h2 id='jackstraw'>jackstraw: Statistical Inference for Unsupervised Learning</h2><span id='topic+jackstraw'></span><span id='topic+jackstraw-package'></span>

<h3>Description</h3>

<p>Test for association between the observed data and their estimated latent variables. The jackstraw package provides a resampling strategy and testing scheme to estimate statistical significance of association between the observed data and their latent variables. Depending on the data type and the analysis aim, the latent variables may be estimated by principal component analysis (PCA), factor analysis (FA), K-means clustering, and related algorithms. The jackstraw methods learn over-fitting characteristics inherent in this circular analysis, where the observed data are used to estimate the latent variables and used again to test against that estimated latent variables. When latent variables are estimated by PCA, the jackstraw enables statistical testing for association between observed variables and latent variables, as estimated by low-dimensional principal components (PCs). This essentially leads to identifying variables that are significantly associated with PCs. Similarly, unsupervised clustering, such as K-means clustering, partition around medoids (PAM), and others, finds coherent groups in high-dimensional data. The jackstraw estimates statistical significance of cluster membership, by testing association between data and cluster centers. Clustering membership can be improved by using the resulting jackstraw p-values and posterior inclusion probabilities (PIPs), with an application to unsupervised evaluation of cell identities in single cell RNA-seq.
</p>


<h3>Details</h3>

<p>The jackstraw package provides a resampling strategy and testing scheme to estimate statistical significance
of association between the observed data and their latent variables. Depending on the data type and the analysis aim,
the latent variables may be estimated by principal component analysis (PCA), K-means clustering, and related algorithms.
The jackstraw methods learn over-fitting characteristics inherent in this circular analysis, where the observed data are used
to estimate the latent variables and used again to test against that estimated latent variables.
</p>
<p>The jackstraw tests enable us to identify the data features (i.e., variables or observations) that are
driving systematic variation, in a completely unsupervised manner. Using <a href="#topic+jackstraw_pca">jackstraw_pca</a>, we can
find statistically significant features with regard to the top <code>r</code> principal components.
Alternatively, <a href="#topic+jackstraw_kmeans">jackstraw_kmeans</a> can identify the data features that are statistically significant
members of the data-dependent clusters. Furthermore, this package includes more general algorithms such as
<a href="#topic+jackstraw_subspace">jackstraw_subspace</a> for the dimension reduction techniques and <a href="#topic+jackstraw_cluster">jackstraw_cluster</a> for the clustering algorithms.
</p>
<p>Overall, it computes <code>m</code> p-values of association between the <code>m</code> data features and their corresponding latent variables.
From <code>m</code> p-values, <a href="#topic+pip">pip</a> computes posterior inclusion probabilities, that are useful for feature selection and visualization.
</p>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung and Storey (2015) Statistical significance of variables driving systematic variation in high-dimensional data. Bioinformatics, 31(4): 545-554 <a href="https://academic.oup.com/bioinformatics/article/31/4/545/2748186">https://academic.oup.com/bioinformatics/article/31/4/545/2748186</a>
</p>
<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+jackstraw_pca">jackstraw_pca</a> <a href="#topic+jackstraw_subspace">jackstraw_subspace</a> <a href="#topic+jackstraw_kmeans">jackstraw_kmeans</a> <a href="#topic+jackstraw_cluster">jackstraw_cluster</a>
</p>

<hr>
<h2 id='jackstraw_cluster'>Jackstraw for the User-Defined Clustering Algorithm</h2><span id='topic+jackstraw_cluster'></span>

<h3>Description</h3>

<p>Test the cluster membership using a user-defined clustering algorithm
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_cluster(
  dat,
  k,
  cluster,
  centers,
  algorithm = function(x, centers, ...) stats::kmeans(x, centers, ...),
  s = 1,
  B = 1000,
  center = TRUE,
  noise = NULL,
  covariate = NULL,
  pool = TRUE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_cluster_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_k">k</code></td>
<td>
<p>a number of clusters.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_cluster">cluster</code></td>
<td>
<p>a vector of cluster assignments.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_centers">centers</code></td>
<td>
<p>a matrix of all cluster centers.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_algorithm">algorithm</code></td>
<td>
<p>a clustering algorithm to use, where an output must include 'cluster' and 'centers'. For exact specification, see <code><a href="stats.html#topic+kmeans">kmeans</a></code>.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_s">s</code></td>
<td>
<p>a number of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_b">B</code></td>
<td>
<p>a number of resampling iterations.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_center">center</code></td>
<td>
<p>a logical specifying to center the rows. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_noise">noise</code></td>
<td>
<p>specify a parametric distribution to generate a noise term. If <code>NULL</code>, a non-parametric jackstraw test is performed.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_covariate">covariate</code></td>
<td>
<p>a model matrix of covariates with <code>n</code> observations. Must include an intercept in the first column.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_pool">pool</code></td>
<td>
<p>a logical specifying to pool the null statistics across all clusters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress. By default, <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_cluster_+3A_...">...</code></td>
<td>
<p>additional, optional arguments to 'algorithm'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The clustering algorithms assign <code>m</code> rows into <code>K</code> clusters. This function enable statistical
evaluation if the cluster membership is correctly assigned. Each of <code>m</code> p-values refers to
the statistical test of that row with regard to its assigned cluster.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of clusters from the observed data
and protects against an anti-conservative bias.
</p>
<p>The user is expected to explore the data with a given clustering algorithm and
determine the number of clusters <code>k</code>.
Furthermore, provide <code>cluster</code> and <code>centers</code> as given by applying <code>algorithm</code> onto <code>dat</code>.
The rows of <code>centers</code> correspond to <code>k</code> clusters, as well as available levels in <code>cluster</code>.
This function allows you to specify a parametric distribution of a noise term. It is an experimental feature.
</p>


<h3>Value</h3>

<p><code>jackstraw_cluster</code> returns a list consisting of
</p>
<table>
<tr><td><code>F.obs</code></td>
<td>
<p><code>m</code> observed F statistics between variables and cluster centers.</p>
</td></tr>
<tr><td><code>F.null</code></td>
<td>
<p>F null statistics between null variables and cluster centers, from the jackstraw method.</p>
</td></tr>
<tr><td><code>p.F</code></td>
<td>
<p><code>m</code> p-values of membership.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>

<hr>
<h2 id='jackstraw_irlba'>Non-Parametric Jackstraw for Principal Component Analysis (PCA) using the augmented implicitly restarted Lanczos bidiagonalization algorithm (IRLBA)</h2><span id='topic+jackstraw_irlba'></span>

<h3>Description</h3>

<p>Test association between the observed variables and their latent variables captured by principal components (PCs). PCs are computed using the augmented implicitly restarted Lanczos bidiagonalization algorithm (IRLBA; see <code><a href="irlba.html#topic+irlba">irlba</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_irlba(
  dat,
  r = NULL,
  r1 = NULL,
  s = NULL,
  B = NULL,
  covariate = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_irlba_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_r">r</code></td>
<td>
<p>a number (a positive integer) of significant principal components. See <a href="#topic+permutationPA">permutationPA</a> and other methods.</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_r1">r1</code></td>
<td>
<p>a numeric vector of principal components of interest. Choose a subset of <code>r</code> significant PCs to be used.</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_s">s</code></td>
<td>
<p>a number (a positive integer) of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_b">B</code></td>
<td>
<p>a number (a positive integer) of resampling iterations. There will be a total of <code>s*B</code> null statistics.</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_covariate">covariate</code></td>
<td>
<p>a data matrix of covariates with corresponding <code>n</code> observations (do not include an intercept term).</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress.</p>
</td></tr>
<tr><td><code id="jackstraw_irlba_+3A_...">...</code></td>
<td>
<p>additional arguments to <code><a href="irlba.html#topic+irlba">irlba</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes <code>m</code> p-values of linear association between <code>m</code> variables and their PCs.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of PCs from the observed data
and protects against an anti-conservative bias.
</p>
<p>Provide the data matrix, with <code>m</code> variables as rows and <code>n</code> observations as columns.
Given that there are <code>r</code> significant PCs, this function tests for linear association between
<code>m</code> variables and their <code>r</code> PCs.
</p>
<p>You could specify a subset of significant PCs that you are interested in (<code>r1</code>). If <code>r1</code> is given,
then this function computes statistical significance of association between <code>m</code> variables and <code>r1</code>,
while adjusting for other PCs (i.e., significant PCs that are not your interest).
For example, if you want to identify variables associated with 1st and 2nd PCs,
when your data contains three significant PCs, set <code>r=3</code> and <code>r1=c(1,2)</code>.
</p>
<p>Please take a careful look at your data and use appropriate graphical and statistical criteria
to determine a number of significant PCs, <code>r</code>. The number of significant PCs depends on the data structure and the context.
In a case when you fail to specify <code>r</code>, it will be estimated from a permutation test (Buja and Eyuboglu, 1992)
using a function <a href="#topic+permutationPA">permutationPA</a>.
</p>
<p>If <code>s</code> is not supplied, <code>s</code> is set to about 10% of <code>m</code> variables.
If <code>B</code> is not supplied, <code>B</code> is set to <code>m*10/s</code>.
</p>


<h3>Value</h3>

<p><code>jackstraw_irlba</code> returns a list consisting of
</p>
<table>
<tr><td><code>p.value</code></td>
<td>
<p><code>m</code> p-values of association tests between variables and their principal components</p>
</td></tr>
<tr><td><code>obs.stat</code></td>
<td>
<p><code>m</code> observed F-test statistics</p>
</td></tr>
<tr><td><code>null.stat</code></td>
<td>
<p><code>s*B</code> null F-test statistics</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung and Storey (2015) Statistical significance of variables driving systematic variation in high-dimensional data. Bioinformatics, 31(4): 545-554 <a href="https://academic.oup.com/bioinformatics/article/31/4/545/2748186">https://academic.oup.com/bioinformatics/article/31/4/545/2748186</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+jackstraw">jackstraw</a> <a href="#topic+jackstraw_subspace">jackstraw_subspace</a> <a href="#topic+permutationPA">permutationPA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate data from a latent variable model: Y = BL + E
B = c(rep(1,10),rep(-1,10), rep(0,180))
L = rnorm(20)
E = matrix(rnorm(200*20), nrow=200)
dat = B %*% t(L) + E
dat = t(scale(t(dat), center=TRUE, scale=TRUE))

## apply the jackstraw
out = jackstraw_irlba(dat, r=1)

## Use optional arguments
## For example, set s and B for a balance between speed of the algorithm and accuracy of p-values
## Not run: 
## out = jackstraw_irlba(dat, r=1, s=10, B=200)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_kmeans'>Non-Parametric Jackstraw for K-means Clustering</h2><span id='topic+jackstraw_kmeans'></span>

<h3>Description</h3>

<p>Test the cluster membership for K-means clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_kmeans(
  dat,
  kmeans.dat,
  s = NULL,
  B = NULL,
  center = FALSE,
  covariate = NULL,
  match = TRUE,
  pool = TRUE,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_kmeans_+3A_dat">dat</code></td>
<td>
<p>a matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_kmeans.dat">kmeans.dat</code></td>
<td>
<p>an output from applying <code>kmeans()</code> onto <code>dat</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_s">s</code></td>
<td>
<p>a number of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_b">B</code></td>
<td>
<p>a number of resampling iterations.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_center">center</code></td>
<td>
<p>a logical specifying to center the rows of the null samples. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_covariate">covariate</code></td>
<td>
<p>a model matrix of covariates with <code>n</code> observations. Must include an intercept in the first column.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_match">match</code></td>
<td>
<p>a logical specifying to match the observed clusters and jackstraw clusters using minimum Euclidean distances.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_pool">pool</code></td>
<td>
<p>a logical specifying to pool the null statistics across all clusters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress. By default, <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeans_+3A_...">...</code></td>
<td>
<p>optional arguments to control the k-means clustering algorithm (refers to <code>kmeans</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>K-means clustering assign <code>m</code> rows into <code>K</code> clusters. This function enable statistical
evaluation if the cluster membership is correctly assigned. Each of <code>m</code> p-values refers to
the statistical test of that row with regard to its assigned cluster.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of clusters from the observed data
and protects against an anti-conservative bias.
</p>
<p>The input data (<code>dat</code>) must be of a class 'matrix'.
</p>


<h3>Value</h3>

<p><code>jackstraw_kmeans</code> returns a list consisting of
</p>
<table>
<tr><td><code>F.obs</code></td>
<td>
<p><code>m</code> observed F statistics between variables and cluster centers.</p>
</td></tr>
<tr><td><code>F.null</code></td>
<td>
<p>F null statistics between null variables and cluster centers, from the jackstraw method.</p>
</td></tr>
<tr><td><code>p.F</code></td>
<td>
<p><code>m</code> p-values of membership.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dat = t(scale(t(Jurkat293T), center=TRUE, scale=FALSE))
kmeans.dat &lt;- kmeans(dat, centers=2, nstart = 10, iter.max = 100)
jackstraw.out &lt;- jackstraw_kmeans(dat, kmeans.dat)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_kmeanspp'>Non-Parametric Jackstraw for K-means Clustering using RcppArmadillo</h2><span id='topic+jackstraw_kmeanspp'></span>

<h3>Description</h3>

<p>Test the cluster membership for K-means clustering, using K-means++ initialization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_kmeanspp(
  dat,
  kmeans.dat,
  s = NULL,
  B = NULL,
  center = TRUE,
  covariate = NULL,
  verbose = FALSE,
  pool = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_kmeanspp_+3A_dat">dat</code></td>
<td>
<p>a matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_kmeans.dat">kmeans.dat</code></td>
<td>
<p>an output from applying <code>ClusterR::KMeans_rcpp</code> onto <code>dat</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_s">s</code></td>
<td>
<p>a number of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_b">B</code></td>
<td>
<p>a number of resampling iterations.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_center">center</code></td>
<td>
<p>a logical specifying to center the rows. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_covariate">covariate</code></td>
<td>
<p>a model matrix of covariates with <code>n</code> observations. Must include an intercept in the first column.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress. By default, <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_pool">pool</code></td>
<td>
<p>a logical specifying to pool the null statistics across all clusters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_kmeanspp_+3A_...">...</code></td>
<td>
<p>optional arguments to control the k-means clustering algorithm (refers to <code>ClusterR::KMeans_rcpp</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>K-means clustering assign <code>m</code> rows into <code>K</code> clusters. This function enable statistical
evaluation if the cluster membership is correctly assigned. Each of <code>m</code> p-values refers to
the statistical test of that row with regard to its assigned cluster.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of clusters from the observed data
and protects against an anti-conservative bias.
</p>
<p>Generally, it functions identical to <code>jackstraw_kmeans</code>, but this uses <code>ClusterR::KMeans_rcpp</code> instead of <code>stats::kmeans</code>.
A speed improvement is gained by K-means++ initialization and <code>RcppArmadillo</code>. If the input data is still too large,
consider using <code>jackstraw_MiniBatchKmeans</code>.
</p>
<p>The input data (<code>dat</code>) must be of a class 'matrix'.
</p>


<h3>Value</h3>

<p><code>jackstraw_kmeanspp</code> returns a list consisting of
</p>
<table>
<tr><td><code>F.obs</code></td>
<td>
<p><code>m</code> observed F statistics between variables and cluster centers.</p>
</td></tr>
<tr><td><code>F.null</code></td>
<td>
<p>F null statistics between null variables and cluster centers, from the jackstraw method.</p>
</td></tr>
<tr><td><code>p.F</code></td>
<td>
<p><code>m</code> p-values of membership.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ClusterR)
dat = t(scale(t(Jurkat293T), center=TRUE, scale=FALSE))
kmeans.dat &lt;- KMeans_rcpp(dat,  clusters = 10, num_init = 1,
max_iters = 100, initializer = 'kmeans++')
jackstraw.out &lt;- jackstraw_kmeanspp(dat, kmeans.dat)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_MiniBatchKmeans'>Non-Parametric Jackstraw for Mini Batch K-means Clustering</h2><span id='topic+jackstraw_MiniBatchKmeans'></span>

<h3>Description</h3>

<p>Test the cluster membership for K-means clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_MiniBatchKmeans(
  dat,
  MiniBatchKmeans.output = NULL,
  s = NULL,
  B = NULL,
  center = TRUE,
  covariate = NULL,
  verbose = FALSE,
  batch_size = floor(nrow(dat)/100),
  initializer = "kmeans++",
  pool = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_minibatchkmeans.output">MiniBatchKmeans.output</code></td>
<td>
<p>an output from applying <code>ClusterR::MiniBatchKmeans()</code> onto <code>dat</code>. This provides more controls over the algorithm and subsequently the initial centroids used.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_s">s</code></td>
<td>
<p>a number of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_b">B</code></td>
<td>
<p>a number of resampling iterations.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_center">center</code></td>
<td>
<p>a logical specifying to center the rows. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_covariate">covariate</code></td>
<td>
<p>a model matrix of covariates with <code>n</code> observations. Must include an intercept in the first column.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress. By default, <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_batch_size">batch_size</code></td>
<td>
<p>the size of the mini batches.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_initializer">initializer</code></td>
<td>
<p>the method of initialization. By default, <code>kmeans++</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_pool">pool</code></td>
<td>
<p>a logical specifying to pool the null statistics across all clusters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_MiniBatchKmeans_+3A_...">...</code></td>
<td>
<p>optional arguments to control the Mini Batch K-means clustering algorithm (refers to <code>ClusterR::MiniBatchKmeans</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>K-means clustering assign <code>m</code> rows into <code>K</code> clusters. This function enable statistical
evaluation if the cluster membership is correctly assigned. Each of <code>m</code> p-values refers to
the statistical test of that row with regard to its assigned cluster.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of clusters from the observed data
and protects against an anti-conservative bias.
</p>


<h3>Value</h3>

<p><code>jackstraw_MiniBatchKmeans</code> returns a list consisting of
</p>
<table>
<tr><td><code>F.obs</code></td>
<td>
<p><code>m</code> observed F statistics between variables and cluster centers.</p>
</td></tr>
<tr><td><code>F.null</code></td>
<td>
<p>F null statistics between null variables and cluster centers, from the jackstraw method.</p>
</td></tr>
<tr><td><code>p.F</code></td>
<td>
<p><code>m</code> p-values of membership.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(ClusterR)
dat = t(scale(t(Jurkat293T), center=TRUE, scale=FALSE))
MiniBatchKmeans.output &lt;- MiniBatchKmeans(data=dat, clusters = 2, batch_size = 300,
initializer = "kmeans++")
jackstraw.output &lt;- jackstraw_MiniBatchKmeans(dat,
MiniBatchKmeans.output = MiniBatchKmeans.output)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_pam'>Non-Parametric Jackstraw for Partitioning Around Medoids (PAM)</h2><span id='topic+jackstraw_pam'></span>

<h3>Description</h3>

<p>Test the cluster membership for Partitioning Around Medoids (PAM)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_pam(
  dat,
  pam.dat,
  s = NULL,
  B = NULL,
  center = TRUE,
  covariate = NULL,
  verbose = FALSE,
  pool = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_pam_+3A_dat">dat</code></td>
<td>
<p>a matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_pam.dat">pam.dat</code></td>
<td>
<p>an output from applying <code>cluster::pam()</code> on <code>dat</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_s">s</code></td>
<td>
<p>a number of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_b">B</code></td>
<td>
<p>a number of resampling iterations.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_center">center</code></td>
<td>
<p>a logical specifying to center the rows. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_covariate">covariate</code></td>
<td>
<p>a model matrix of covariates with <code>n</code> observations. Must include an intercept in the first column.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress. By default, <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_pool">pool</code></td>
<td>
<p>a logical specifying to pool the null statistics across all clusters. By default, <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="jackstraw_pam_+3A_...">...</code></td>
<td>
<p>optional arguments to control the k-means clustering algorithm (refers to <code>kmeans</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>PAM assigns <code>m</code> rows into <code>K</code> clusters. This function enable statistical
evaluation if the cluster membership is correctly assigned. Each of <code>m</code> p-values refers to
the statistical test of that row with regard to its assigned cluster.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of clusters from the observed data
and protects against an anti-conservative bias.
</p>
<p>For a large dataset, PAM could be too slow. Consider using <code>cluster::clara</code> and <code>jackstraw::jackstraw_clara</code>.
</p>
<p>The input data (<code>dat</code>) must be of a class 'matrix'.
</p>


<h3>Value</h3>

<p><code>jackstraw_pam</code> returns a list consisting of
</p>
<table>
<tr><td><code>F.obs</code></td>
<td>
<p><code>m</code> observed F statistics between variables and cluster medoids.</p>
</td></tr>
<tr><td><code>F.null</code></td>
<td>
<p>F null statistics between null variables and cluster medoids, from the jackstraw method.</p>
</td></tr>
<tr><td><code>p.F</code></td>
<td>
<p><code>m</code> p-values of membership.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(cluster)
dat = t(scale(t(Jurkat293T), center=TRUE, scale=FALSE))
pam.dat &lt;- pam(dat, k=2)
jackstraw.out &lt;- jackstraw_pam(dat, pam.dat = pam.dat)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_pca'>Non-Parametric Jackstraw for Principal Component Analysis (PCA)</h2><span id='topic+jackstraw_pca'></span>

<h3>Description</h3>

<p>Test association between the observed variables and their latent variables captured by principal components (PCs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_pca(
  dat,
  r = NULL,
  r1 = NULL,
  s = NULL,
  B = NULL,
  covariate = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_pca_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_pca_+3A_r">r</code></td>
<td>
<p>a number (a positive integer) of significant principal components. See <a href="#topic+permutationPA">permutationPA</a> and other methods.</p>
</td></tr>
<tr><td><code id="jackstraw_pca_+3A_r1">r1</code></td>
<td>
<p>a numeric vector of principal components of interest. Choose a subset of <code>r</code> significant PCs to be used.</p>
</td></tr>
<tr><td><code id="jackstraw_pca_+3A_s">s</code></td>
<td>
<p>a number (a positive integer) of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_pca_+3A_b">B</code></td>
<td>
<p>a number (a positive integer) of resampling iterations. There will be a total of <code>s*B</code> null statistics.</p>
</td></tr>
<tr><td><code id="jackstraw_pca_+3A_covariate">covariate</code></td>
<td>
<p>a data matrix of covariates with corresponding <code>n</code> observations (do not include an intercept term).</p>
</td></tr>
<tr><td><code id="jackstraw_pca_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes <code>m</code> p-values of linear association between <code>m</code> variables and their PCs.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of PCs from the observed data
and protects against an anti-conservative bias.
</p>
<p>Provide the data matrix, with <code>m</code> variables as rows and <code>n</code> observations as columns.
Given that there are <code>r</code> significant PCs, this function tests for linear association between
<code>m</code> variables and their <code>r</code> PCs.
</p>
<p>You could specify a subset of significant PCs that you are interested in (<code>r1</code>). If <code>r1</code> is given,
then this function computes statistical significance of association between <code>m</code> variables and <code>r1</code>,
while adjusting for other PCs (i.e., significant PCs that are not your interest).
For example, if you want to identify variables associated with 1st and 2nd PCs,
when your data contains three significant PCs, set <code>r=3</code> and <code>r1=c(1,2)</code>.
</p>
<p>Please take a careful look at your data and use appropriate graphical and statistical criteria
to determine a number of significant PCs, <code>r</code>. The number of significant PCs depends on the data structure and the context.
In a case when you fail to specify <code>r</code>, it will be estimated from a permutation test (Buja and Eyuboglu, 1992)
using a function <a href="#topic+permutationPA">permutationPA</a>.
</p>
<p>If <code>s</code> is not supplied, <code>s</code> is set to about 10% of <code>m</code> variables.
If <code>B</code> is not supplied, <code>B</code> is set to <code>m*10/s</code>.
</p>


<h3>Value</h3>

<p><code>jackstraw_pca</code> returns a list consisting of
</p>
<table>
<tr><td><code>p.value</code></td>
<td>
<p><code>m</code> p-values of association tests between variables and their principal components</p>
</td></tr>
<tr><td><code>obs.stat</code></td>
<td>
<p><code>m</code> observed F-test statistics</p>
</td></tr>
<tr><td><code>null.stat</code></td>
<td>
<p><code>s*B</code> null F-test statistics</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung and Storey (2015) Statistical significance of variables driving systematic variation in high-dimensional data. Bioinformatics, 31(4): 545-554 <a href="https://academic.oup.com/bioinformatics/article/31/4/545/2748186">https://academic.oup.com/bioinformatics/article/31/4/545/2748186</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+jackstraw">jackstraw</a> <a href="#topic+jackstraw_subspace">jackstraw_subspace</a> <a href="#topic+permutationPA">permutationPA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
## simulate data from a latent variable model: Y = BL + E
B = c(rep(1,50),rep(-1,50), rep(0,900))
L = rnorm(20)
E = matrix(rnorm(1000*20), nrow=1000)
dat = B %*% t(L) + E
dat = t(scale(t(dat), center=TRUE, scale=TRUE))

## apply the jackstraw
out = jackstraw_pca(dat, r=1)

## Use optional arguments
## For example, set s and B for a balance between speed of the algorithm and accuracy of p-values
## out = jackstraw_pca(dat, r=1, s=10, B=1000)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_rpca'>Non-Parametric Jackstraw for Principal Component Analysis (PCA) using Randomized Singular Value Decomposition</h2><span id='topic+jackstraw_rpca'></span>

<h3>Description</h3>

<p>Test association between the observed variables and their latent variables captured by principal components (PCs). PCs are computed by randomized Singular Value Decomposition (see <code><a href="rsvd.html#topic+rsvd">rsvd</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_rpca(
  dat,
  r = NULL,
  r1 = NULL,
  s = NULL,
  B = NULL,
  covariate = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_rpca_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_r">r</code></td>
<td>
<p>a number (a positive integer) of significant principal components. See <a href="#topic+permutationPA">permutationPA</a> and other methods.</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_r1">r1</code></td>
<td>
<p>a numeric vector of principal components of interest. Choose a subset of <code>r</code> significant PCs to be used.</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_s">s</code></td>
<td>
<p>a number (a positive integer) of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_b">B</code></td>
<td>
<p>a number (a positive integer) of resampling iterations. There will be a total of <code>s*B</code> null statistics.</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_covariate">covariate</code></td>
<td>
<p>a data matrix of covariates with corresponding <code>n</code> observations (do not include an intercept term).</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress.</p>
</td></tr>
<tr><td><code id="jackstraw_rpca_+3A_...">...</code></td>
<td>
<p>additional arguments to <code>rpca</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes <code>m</code> p-values of linear association between <code>m</code> variables and their PCs.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of PCs from the observed data
and protects against an anti-conservative bias.
</p>
<p>Provide the data matrix, with <code>m</code> variables as rows and <code>n</code> observations as columns.
Given that there are <code>r</code> significant PCs, this function tests for linear association between
<code>m</code> variables and their <code>r</code> PCs.
</p>
<p>You could specify a subset of significant PCs that you are interested in (<code>r1</code>). If <code>r1</code> is given,
then this function computes statistical significance of association between <code>m</code> variables and <code>r1</code>,
while adjusting for other PCs (i.e., significant PCs that are not your interest).
For example, if you want to identify variables associated with 1st and 2nd PCs,
when your data contains three significant PCs, set <code>r=3</code> and <code>r1=c(1,2)</code>.
</p>
<p>Please take a careful look at your data and use appropriate graphical and statistical criteria
to determine a number of significant PCs, <code>r</code>. The number of significant PCs depends on the data structure and the context.
In a case when you fail to specify <code>r</code>, it will be estimated from a permutation test (Buja and Eyuboglu, 1992)
using a function <a href="#topic+permutationPA">permutationPA</a>.
</p>
<p>If <code>s</code> is not supplied, <code>s</code> is set to about 10% of <code>m</code> variables.
If <code>B</code> is not supplied, <code>B</code> is set to <code>m*10/s</code>.
</p>


<h3>Value</h3>

<p><code>jackstraw_rpca</code> returns a list consisting of
</p>
<table>
<tr><td><code>p.value</code></td>
<td>
<p><code>m</code> p-values of association tests between variables and their principal components</p>
</td></tr>
<tr><td><code>obs.stat</code></td>
<td>
<p><code>m</code> observed F-test statistics</p>
</td></tr>
<tr><td><code>null.stat</code></td>
<td>
<p><code>s*B</code> null F-test statistics</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung and Storey (2015) Statistical significance of variables driving systematic variation in high-dimensional data. Bioinformatics, 31(4): 545-554 <a href="https://academic.oup.com/bioinformatics/article/31/4/545/2748186">https://academic.oup.com/bioinformatics/article/31/4/545/2748186</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+jackstraw">jackstraw</a> <a href="#topic+jackstraw_subspace">jackstraw_subspace</a> <a href="#topic+permutationPA">permutationPA</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate data from a latent variable model: Y = BL + E
B = c(rep(1,10),rep(-1,10), rep(0,180))
L = rnorm(20)
E = matrix(rnorm(200*20), nrow=200)
dat = B %*% t(L) + E
dat = t(scale(t(dat), center=TRUE, scale=TRUE))

## apply the jackstraw
out = jackstraw_rpca(dat, r=1)

## Use optional arguments
## For example, set s and B for a balance between speed of the algorithm and accuracy of p-values
## Not run: 
## out = jackstraw_rpca(dat, r=1, s=10, B=200)

## End(Not run)

</code></pre>

<hr>
<h2 id='jackstraw_subspace'>Jackstraw for the User-Defined Dimension Reduction Methods</h2><span id='topic+jackstraw_subspace'></span>

<h3>Description</h3>

<p>Test association between the observed variables and their latent variables, captured by a user-defined dimension reduction method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jackstraw_subspace(
  dat,
  r,
  FUN,
  r1 = NULL,
  s = NULL,
  B = NULL,
  covariate = NULL,
  noise = NULL,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jackstraw_subspace_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_r">r</code></td>
<td>
<p>a number of significant latent variables.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_fun">FUN</code></td>
<td>
<p>Provide a specific function to estimate LVs. Must output <code>r</code> estimated LVs in a <code>n*r</code> matrix.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_r1">r1</code></td>
<td>
<p>a numeric vector of latent variables of interest.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_s">s</code></td>
<td>
<p>a number of &ldquo;synthetic&rdquo; null variables. Out of <code>m</code> variables, <code>s</code> variables are independently permuted.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_b">B</code></td>
<td>
<p>a number of resampling iterations.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_covariate">covariate</code></td>
<td>
<p>a model matrix of covariates with <code>n</code> observations. Must include an intercept in the first column.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_noise">noise</code></td>
<td>
<p>specify a parametric distribution to generate a noise term. If <code>NULL</code>, a non-parametric jackstraw test is performed.</p>
</td></tr>
<tr><td><code id="jackstraw_subspace_+3A_verbose">verbose</code></td>
<td>
<p>a logical specifying to print the computational progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes <code>m</code> p-values of linear association between <code>m</code> variables and their latent variables,
captured by a user-defined dimension reduction method.
Its resampling strategy accounts for the over-fitting characteristics due to direct computation of PCs from the observed data
and protects against an anti-conservative bias.
</p>
<p>This function allows you to specify a parametric distribution of a noise term. It is an experimental feature. Then, a small number <code>s</code> of observed variables
are replaced by synthetic null variables generated from a specified distribution.
</p>


<h3>Value</h3>

<p><code>jackstraw_subspace</code> returns a list consisting of
</p>
<table>
<tr><td><code>p.value</code></td>
<td>
<p><code>m</code> p-values of association tests between variables and their principal components</p>
</td></tr>
<tr><td><code>obs.stat</code></td>
<td>
<p><code>m</code> observed statistics</p>
</td></tr>
<tr><td><code>null.stat</code></td>
<td>
<p><code>s*B</code> null statistics</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>


<h3>References</h3>

<p>Chung and Storey (2015) Statistical significance of variables driving systematic variation in high-dimensional data. Bioinformatics, 31(4): 545-554 <a href="https://academic.oup.com/bioinformatics/article/31/4/545/2748186">https://academic.oup.com/bioinformatics/article/31/4/545/2748186</a>
</p>
<p>Chung (2020) Statistical significance of cluster membership for unsupervised evaluation of cell identities. Bioinformatics, 36(10): 3107–3114 <a href="https://academic.oup.com/bioinformatics/article/36/10/3107/5788523">https://academic.oup.com/bioinformatics/article/36/10/3107/5788523</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+jackstraw_pca">jackstraw_pca</a> <a href="#topic+jackstraw">jackstraw</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simulate data from a latent variable model: Y = BL + E
B = c(rep(1,50),rep(-1,50), rep(0,900))
L = rnorm(20)
E = matrix(rnorm(1000*20), nrow=1000)
dat = B %*% t(L) + E
dat = t(scale(t(dat), center=TRUE, scale=TRUE))

## apply the jackstraw with the svd as a function
out = jackstraw_subspace(dat, FUN = function(x) svd(x)$v[,1,drop=FALSE], r=1, s=100, B=50)

</code></pre>

<hr>
<h2 id='Jurkat293T'>A Jurkat:293T equal mixture dataset from Zheng et al. (2017)</h2><span id='topic+Jurkat293T'></span>

<h3>Description</h3>

<p>50
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Jurkat293T
</code></pre>


<h3>Format</h3>

<p>A data frame with 3381 rows corresponding to single cells and 10 columns corresponding to the top 10 principal components
</p>


<h3>Source</h3>

<p>Supplementary Data 1 from Zheng et al. (2017) <a href="https://static-content.springer.com/esm/art%3A10.1038%2Fncomms14049/MediaObjects/41467_2017_BFncomms14049_MOESM829_ESM.xlsx">https://static-content.springer.com/esm/art%3A10.1038%2Fncomms14049/MediaObjects/41467_2017_BFncomms14049_MOESM829_ESM.xlsx</a>
</p>


<h3>References</h3>

<p>Zheng et al. (2017) Massively parallel digital transcriptional profiling of single cells. Nature Communications. 8:14049. DOI: 10.1038/ncomms14049
</p>

<hr>
<h2 id='permutationPA'>Permutation Parallel Analysis</h2><span id='topic+permutationPA'></span>

<h3>Description</h3>

<p>Estimate a number of significant principal components from a permutation test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permutationPA(dat, B = 100, threshold = 0.05, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permutationPA_+3A_dat">dat</code></td>
<td>
<p>a data matrix with <code>m</code> rows as variables and <code>n</code> columns as observations.</p>
</td></tr>
<tr><td><code id="permutationPA_+3A_b">B</code></td>
<td>
<p>a number (a positive integer) of resampling iterations.</p>
</td></tr>
<tr><td><code id="permutationPA_+3A_threshold">threshold</code></td>
<td>
<p>a numeric value between 0 and 1 to threshold p-values.</p>
</td></tr>
<tr><td><code id="permutationPA_+3A_verbose">verbose</code></td>
<td>
<p>a logical indicator as to whether to print the progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Adopted from sva::num.sv, and based on Buja and Eyuboglu (1992)
</p>


<h3>Value</h3>

<p><code>permutationPA</code> returns
</p>
<table>
<tr><td><code>r</code></td>
<td>
<p>an estimated number of significant principal components based on thresholding p-values at <code>threshold</code></p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>a list of p-values for significance of principal components</p>
</td></tr>
</table>


<h3>References</h3>

<p>Buja A and Eyuboglu N. (1992) Remarks on parallel analysis. Multivariate Behavioral Research, 27(4), 509-540
</p>

<hr>
<h2 id='pip'>Compute posterior inclusion probabilities (PIPs)</h2><span id='topic+pip'></span>

<h3>Description</h3>

<p>From a set of p-values, computes posterior probabilities that a feature should be truly included.
For example, membership inclusion in a given cluster can be improved by filtering low quality members.
In using PCA and related methods, it helps select variables that are truly associated with given latent variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pip(pvalue, group = NULL, pi0 = NULL, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pip_+3A_pvalue">pvalue</code></td>
<td>
<p>a vector of p-values.</p>
</td></tr>
<tr><td><code id="pip_+3A_group">group</code></td>
<td>
<p>a vector of group indicators (optional).
If provided, PIP analysis is stratified.
Assumes groups are in 1:k where k is the number of unique groups.</p>
</td></tr>
<tr><td><code id="pip_+3A_pi0">pi0</code></td>
<td>
<p>a vector of pi0 values (optional).
Its length has to be either 1 or equal the number of groups.</p>
</td></tr>
<tr><td><code id="pip_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE, reports information.</p>
</td></tr>
<tr><td><code id="pip_+3A_...">...</code></td>
<td>
<p>optional arguments for <code><a href="qvalue.html#topic+lfdr">lfdr</a></code> to control a local FDR estimation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>pip</code> returns a vector of posterior inclusion probabilities
</p>


<h3>Author(s)</h3>

<p>Neo Christopher Chung <a href="mailto:nchchung@gmail.com">nchchung@gmail.com</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
