<!DOCTYPE html><html lang="en"><head><title>Help for package Jacquard</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Jacquard}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BoxplotDelta'>
<p>Boxplots of Jacquard Coefficients</p></a></li>
<li><a href='#BoxplotTheta'>
<p>Boxplots of Relateness parameters</p></a></li>
<li><a href='#CalculateMom'>
<p>Calculate Moment Estimators for Relatedness Parameters</p></a></li>
<li><a href='#CalculateTheta'>
<p>Calculate Relatedness Paramenters from Jacquard Coefficients tA Capitalized Title (ideally limited to 65 characters)</p></a></li>
<li><a href='#DeltaPair'><p>Extract estimates of Jacquard coefficients for a given pair</p></a></li>
<li><a href='#DeltaSimulatedPedigree'>
<p>Estimates of Jacquard Coefficients for Simulated SNP data according to a Pedigree</p></a></li>
<li><a href='#GTC'>
<p>Joint Genotype Counts according to a Pedigree</p></a></li>
<li><a href='#Jacquard.cls'>
<p>Estimation of Jacquard coefficients by Constrained Least Squares</p></a></li>
<li><a href='#JointGenotypeCounts'>
<p>Calculate joint pairwise Genotype Counts</p></a></li>
<li><a href='#mafvector'>
<p>Calculate the Minor Allele Frequency (MAF) for a matrix with (0,1,2) genotype data</p></a></li>
<li><a href='#MakeM'><p>Create matrix of conditional joint genotype probabilities</p></a></li>
<li><a href='#PairwiseList'>
<p>Create a Pairwise List of Jacquard Coefficients</p></a></li>
<li><a href='#SimulatedPedigree'>
<p>Simulated SNP data according to a Pedigree</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimation of Jacquard's Genetic Identity Coefficients</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-09-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Jan Graffelman [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jan Graffelman &lt;jan.graffelman@upc.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), Rsolnp</td>
</tr>
<tr>
<td>Description:</td>
<td>Contains procedures to estimate the nine condensed Jacquard genetic identity coefficients
              (Jacquard, 1974) &lt;<a href="https://doi.org/10.1007%2F978-3-642-88415-3">doi:10.1007/978-3-642-88415-3</a>&gt; by constrained least squares
              (Graffelman et al., 2024) &lt;<a href="https://doi.org/10.1101%2F2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>&gt; and by the method of moments
	      (Csuros, 2014) &lt;<a href="https://doi.org/10.1016%2Fj.tpb.2013.11.001">doi:10.1016/j.tpb.2013.11.001</a>&gt;. These procedures require previous estimation of the
	      allele frequencies. Functions are supplied that estimate relationship parameters that derive from the Jacquard
	      coefficients, such as individual inbreeding coefficients and kinship coefficients.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.r-project.org">https://www.r-project.org</a>, <a href="http://www-eio.upc.edu/~jan/">http://www-eio.upc.edu/~jan/</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-17 10:20:55 UTC; TanitSunShine</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-17 16:20:50 UTC</td>
</tr>
</table>
<hr>
<h2 id='BoxplotDelta'>
Boxplots of Jacquard Coefficients 
</h2><span id='topic+BoxplotDelta'></span>

<h3>Description</h3>

<p>Function <code>BoxplotDelta</code> produces boxplots of the Jacquard coefficients from
a list structure containing nine matrices with the pairwise coefficients. The
diagonals of J1 and J7 are plotted in separate boxplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxplotDelta(J, ind.sub = 1:nrow(J[[1]]), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BoxplotDelta_+3A_j">J</code></td>
<td>
<p>The list structure with nine fields consisting of nine matrices</p>
</td></tr>
<tr><td><code id="BoxplotDelta_+3A_ind.sub">ind.sub</code></td>
<td>
<p>Index for subsetting the individuals</p>
</td></tr>
<tr><td><code id="BoxplotDelta_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass on to <code>boxplot</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(DeltaSimulatedPedigree)
  BoxplotDelta(DeltaSimulatedPedigree)
</code></pre>

<hr>
<h2 id='BoxplotTheta'>
Boxplots of Relateness parameters
</h2><span id='topic+BoxplotTheta'></span>

<h3>Description</h3>

<p>Function <code>BoxplotTheta</code> makes boxplots of relatedess parameters from a
list object containing estimates of pairwise relatedness parameters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxplotTheta(KS, ind.sub = 1:nrow(KS[[1]]), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BoxplotTheta_+3A_ks">KS</code></td>
<td>
<p>A list object with four fields of matrices: kinship, inbreeding, T3 and T4.</p>
</td></tr>
<tr><td><code id="BoxplotTheta_+3A_ind.sub">ind.sub</code></td>
<td>
<p>Index for subsetting the individuals</p>
</td></tr>
<tr><td><code id="BoxplotTheta_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to the function <code>boxplot</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For individual inbreeding coefficients, two boxplots are made, the first based on the diagonal of T2,
the second on the row means of T2. The remaining boxplots (kinship, T3 and T4) are all pairwise, and
exclude self-pairs.
</p>


<h3>Value</h3>

<p>NULL</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Csuros, M. (2014) Non-identifiability of identity coefficients at biallelic loci. Theoretical Population
Biology 92, pp. 22-29. <a href="https://doi.org/10.1016/j.tpb.2013.11.001">doi:10.1016/j.tpb.2013.11.001</a>.
</p>
<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CalculateTheta">CalculateTheta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DeltaSimulatedPedigree)
Theta &lt;- CalculateTheta(DeltaSimulatedPedigree)
BoxplotTheta(Theta)
</code></pre>

<hr>
<h2 id='CalculateMom'>
Calculate Moment Estimators for Relatedness Parameters
</h2><span id='topic+CalculateMom'></span>

<h3>Description</h3>

<p>Function <code>CalculateMom</code> computes moment estimators for a set of
relatedness parameters (kinship, inbreeding, least one IBD out of three
and T4) using the genotype data and the allele frequencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalculateMom(Xgen, mafvec, ind.sub = 1:nrow(Xgen), verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CalculateMom_+3A_xgen">Xgen</code></td>
<td>
<p>Genotype data coded in (0,1,2) format</p>
</td></tr>
<tr><td><code id="CalculateMom_+3A_mafvec">mafvec</code></td>
<td>
<p>A vector of minor allele frequencies</p>
</td></tr>
<tr><td><code id="CalculateMom_+3A_ind.sub">ind.sub</code></td>
<td>
<p>Index for subsetting individuals</p>
</td></tr>
<tr><td><code id="CalculateMom_+3A_verbose">verbose</code></td>
<td>
<p>Print output on the progress of the algorithm if <code>verbose=TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with four fields:
</p>
<table role = "presentation">
<tr><td><code>T1</code></td>
<td>
<p>The pairwise coancestry or kinship coefficients (symmetric)</p>
</td></tr>
<tr><td><code>T2</code></td>
<td>
<p>The pairwise inbreeding coefficients (non-symmetric)</p>
</td></tr>
<tr><td><code>T3</code></td>
<td>
<p>The pairwise Least one IBD out of three (symmetric)</p>
</td></tr>
<tr><td><code>T4</code></td>
<td>
<p>T4 (skew-symmetric)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Csuros, M. (2014) Non-identifiability of identity coefficients at biallelic loci. Theoretical Population
Biology 92, pp. 22-29. <a href="https://doi.org/10.1016/j.tpb.2013.11.001">doi:10.1016/j.tpb.2013.11.001</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BoxplotTheta">BoxplotTheta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SimulatedPedigree)
Xgen &lt;- as.matrix(SimulatedPedigree[,6:ncol(SimulatedPedigree)])
mafvec &lt;- mafvector(Xgen)
Theta.mom &lt;- CalculateMom(Xgen[1:10,],mafvec)
</code></pre>

<hr>
<h2 id='CalculateTheta'>
Calculate Relatedness Paramenters from Jacquard Coefficients tA Capitalized Title (ideally limited to 65 characters)
</h2><span id='topic+CalculateTheta'></span>

<h3>Description</h3>

<p>Function <code>CalculateTheta</code> calculates five identifiable relatedness parameters from a
list structure containing nine matrices of Jacquard coefficients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalculateTheta(J, ind.sub = 1:nrow(J[[1]]))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CalculateTheta_+3A_j">J</code></td>
<td>
<p>A list structure with nine matrices of pairwise Jacquard coefficients</p>
</td></tr>
<tr><td><code id="CalculateTheta_+3A_ind.sub">ind.sub</code></td>
<td>
<p>Index for subsetting the individuals</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>CalculateTheta</code> produces four matrices according to the expressions in
Graffelman et al. (2024).
</p>


<h3>Value</h3>

<p>A list object with four fields:
</p>
<table role = "presentation">
<tr><td><code>T1</code></td>
<td>
<p>The pairwise coancestry or kinship coefficients (symmetric)</p>
</td></tr>
<tr><td><code>T2</code></td>
<td>
<p>The pairwise inbreeding coefficients (non-symmetric)</p>
</td></tr>
<tr><td><code>T3</code></td>
<td>
<p>The pairwise Least one IBD out of three (symmetric)</p>
</td></tr>
<tr><td><code>T4</code></td>
<td>
<p>T4 (skew-symmetric)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Csuros, M. (2014) Non-identifiability of identity coefficients at biallelic loci. Theoretical Population
Biology 92, pp. 22-29. <a href="https://doi.org/10.1016/j.tpb.2013.11.001">doi:10.1016/j.tpb.2013.11.001</a>.
</p>
<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Jacquard.cls">Jacquard.cls</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DeltaSimulatedPedigree)
Theta &lt;- CalculateTheta(DeltaSimulatedPedigree)
</code></pre>

<hr>
<h2 id='DeltaPair'>Extract estimates of Jacquard coefficients for a given pair</h2><span id='topic+DeltaPair'></span>

<h3>Description</h3>

<p>Function <code>DeltaPair</code> extracts from the list object of all pairwise Jacquard coefficients the
set of coefficients of a given pair (i,j).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DeltaPair(Delta, i, j, digits = 7)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DeltaPair_+3A_delta">Delta</code></td>
<td>
<p>A list with nine matrices of pairswise coefficients.</p>
</td></tr>
<tr><td><code id="DeltaPair_+3A_i">i</code></td>
<td>
<p>Index of the first individual.</p>
</td></tr>
<tr><td><code id="DeltaPair_+3A_j">j</code></td>
<td>
<p>Index of the second individual.</p>
</td></tr>
<tr><td><code id="DeltaPair_+3A_digits">digits</code></td>
<td>
<p>Number of digits to which the coefficients are rounded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector with nine elements</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>See Also</h3>

<p><code><a href="#topic+PairwiseList">PairwiseList</a></code>,<code><a href="#topic+Jacquard.cls">Jacquard.cls</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'> data(DeltaSimulatedPedigree)
 DeltaPair(DeltaSimulatedPedigree,1,2)
</code></pre>

<hr>
<h2 id='DeltaSimulatedPedigree'>
Estimates of Jacquard Coefficients for Simulated SNP data according to a Pedigree
</h2><span id='topic+DeltaSimulatedPedigree'></span>

<h3>Description</h3>

<p>A list object containing nine matrices with pairwise Jacquard coefficients
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("DeltaSimulatedPedigree")</code></pre>


<h3>Format</h3>

<p>A list with nine fields.</p>


<h3>Details</h3>

<p><code>DeltaSimulatedPedigree</code> can be generated by applying function <code>Jacquard.cls</code> to the
data in <code>SimulatedPedigree</code>.
</p>


<h3>References</h3>

<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DeltaSimulatedPedigree)
</code></pre>

<hr>
<h2 id='GTC'>
Joint Genotype Counts according to a Pedigree
</h2><span id='topic+GTC'></span>

<h3>Description</h3>

<p>Contains a list object with nine fields, consisting of nine lower triangular matrices with the
joint genotype counts for 111 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("GTC")</code></pre>


<h3>Format</h3>

<p>A list with nine fields.</p>


<h3>References</h3>

<p>Goudet, J. (2022) JGTeach: JG Teaching material. R package version 0.1.9. https://github.com/jgx65
</p>
<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GTC)
</code></pre>

<hr>
<h2 id='Jacquard.cls'>
Estimation of Jacquard coefficients by Constrained Least Squares
</h2><span id='topic+Jacquard.cls'></span>

<h3>Description</h3>

<p>Function <code>Jacquard.cls</code> estimates the nine condensed Jacquard coefficients of a pair of individuals
using their joint genotype probabilities and the allele frequencies using constrained least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Jacquard.cls(Xlist, mafvec = NULL, eps = 1e-06, delta.init = runif(9), Mavg = NULL,
             inner.iter = 1000, outer.iter = 1000, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Jacquard.cls_+3A_xlist">Xlist</code></td>
<td>
<p>A list object with nine fields containing the matrices with joint genotype counts.</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_mafvec">mafvec</code></td>
<td>
<p>A vector with the minor allele frequencies for all genetic variants.</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_eps">eps</code></td>
<td>
<p>Tolerance criterion for the solver (<code>eps=1e-06</code> by default).</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_delta.init">delta.init</code></td>
<td>
<p>Initial vector of estimates for the nine condensed Jacquard coefficients.</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_mavg">Mavg</code></td>
<td>
<p>A nine by nine matrix of conditional probabilities, allele frequency dependent. This matrix
is calculated by <code>Jacquard.cls</code> if it is not specified.</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_inner.iter">inner.iter</code></td>
<td>
<p>Maximum number of inner iterations for the solver (1000 by default).</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_outer.iter">outer.iter</code></td>
<td>
<p>Maximum number of outer iterations for the solver (1000 by default).</p>
</td></tr>
<tr><td><code id="Jacquard.cls_+3A_verbose">verbose</code></td>
<td>
<p>Print output on the progress of the algorithm if <code>verbose=TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function <code>Jacquard.cls</code> relies on the solver <code>solnp</code> from the Rsolnp package.
</p>


<h3>Value</h3>

<p>A list object with fields:
</p>
<table role = "presentation">
<tr><td><code>delta</code></td>
<td>
<p>A list with nine matrices of estimates of pairwise Jacquard coefficients.</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>A matrix with the convergence status for each pair (0 = converged; 1 = not converged).</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>
<p>Ghalanos, A. and Theussl, S. (2015) Rsolnp: General Non-linear Optimization Using Augmented Lagrange Multiplier Method.
R package version 1.16. <a href="https://cran.r-project.org/package=Rsolnp">https://cran.r-project.org/package=Rsolnp</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+JointGenotypeCounts">JointGenotypeCounts</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SimulatedPedigree)
Xgen &lt;- as.matrix(SimulatedPedigree[,6:ncol(SimulatedPedigree)])

data(GTC)

mafvec &lt;- mafvector(Xgen)

ii &lt;- 1:3

GTCsubset &lt;- list(length = 9)
for (k in 1:9) {
  GTCsubset[[k]] &lt;- matrix(numeric(3^2), ncol = 3)
  GTCsubset[[k]] &lt;- GTC[[k]][ii,ii]
}

output &lt;- Jacquard.cls(GTCsubset,mafvec=mafvec,
                       eps=1e-06)
Delta.cls &lt;- output$delta
print(Delta.cls)
print(output$convergence)

# A particular estimate of a Jacquard coefficient for a particular pair can
# be extracted from Delta.cls
#
# E.g., Delta\_9 of the first pair of individuals (1,2) can be extracted by
#

D9_12 &lt;- Delta.cls[[9]][1,2]

</code></pre>

<hr>
<h2 id='JointGenotypeCounts'>
Calculate joint pairwise Genotype Counts
</h2><span id='topic+JointGenotypeCounts'></span>

<h3>Description</h3>

<p>Function <code>JointGenotypeCounts</code> counts for each pair of individuals in the database their
nine joint genotype counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JointGenotypeCounts(X.gen, one.is.minor = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="JointGenotypeCounts_+3A_x.gen">X.gen</code></td>
<td>
<p>A matrix with genotype data coded in (0,1,2) format</p>
</td></tr>
<tr><td><code id="JointGenotypeCounts_+3A_one.is.minor">one.is.minor</code></td>
<td>
<p>If <code>TRUE</code> the genotype data represent the count
of the minor allele for each marker, if <code>FALSE</code> the major allele.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with nine fields containing:
</p>
<table role = "presentation">
<tr><td><code>f0000</code></td>
<td>
<p>Matrix of (0/0,0/0) counts for all pairs</p>
</td></tr>
<tr><td><code>f1111</code></td>
<td>
<p>Matrix of (1/1,1/1) counts for all pairs</p>
</td></tr>
<tr><td><code>f1101</code></td>
<td>
<p>Matrix of (1/1,0/1) counts for all pairs</p>
</td></tr>
<tr><td><code>f0111</code></td>
<td>
<p>Matrix of (0/1,1/1) counts for all pairs</p>
</td></tr>
<tr><td><code>f0101</code></td>
<td>
<p>Matrix of (0/1,0/1) counts for all pairs</p>
</td></tr>
<tr><td><code>f1100</code></td>
<td>
<p>Matrix of (1/1,0/0) counts for all pairs</p>
</td></tr>
<tr><td><code>f0011</code></td>
<td>
<p>Matrix of (0/0,1/1) counts for all pairs</p>
</td></tr>
<tr><td><code>f0100</code></td>
<td>
<p>Matrix of (0/1,0/0) counts for all pairs</p>
</td></tr>
<tr><td><code>f0001</code></td>
<td>
<p>Matrix of (0/0,0/1) counts for all pairs</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients
with bi-allelic variants by constrained least-squares. Preprint at bioRxiv. <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SimulatedPedigree)
JC &lt;- JointGenotypeCounts(SimulatedPedigree[1:3,1:100])
print(JC)
</code></pre>

<hr>
<h2 id='mafvector'>
Calculate the Minor Allele Frequency (MAF) for a matrix with (0,1,2) genotype data
</h2><span id='topic+mafvector'></span>

<h3>Description</h3>

<p>Function <code>mafvector</code> calculates genotype counts columnwise and determines
the minor allele frequency for each column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mafvector(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mafvector_+3A_x">X</code></td>
<td>
<p>A matrix of with (0,1,2) genotype data, individuals in rows, markers in columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>mafvector</code> calculates the frequency of the minor allele irrespective of the coding; i.e.,
irrespective of whether the genotype data represent major or minor allele counts. Missing values
are discarded for the calculation of the MAF.
</p>


<h3>Value</h3>

<p>a vector</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(SimulatedPedigree)
  p &lt;- mafvector(SimulatedPedigree[,1:10])
  print(p)
</code></pre>

<hr>
<h2 id='MakeM'>Create matrix of conditional joint genotype probabilities</h2><span id='topic+MakeM'></span>

<h3>Description</h3>

<p>Function <code>MakeM</code> creates the matrix of conditional joint genotype probabilities for biallelic markers for a given allele frequency.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MakeM(p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MakeM_+3A_p">p</code></td>
<td>
<p>the allele frequency</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 9 by 9 matrix</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>References</h3>

<p>Csuros, M. (2014) Non-identifiability of identity coefficients at biallelic loci. Theoretical Population Biology 92, pp. 22&ndash;29.
doi: 10.1016/j.tpb.2013.11.001
</p>
<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with bi-allelic variants by constrained least-squares. Under review.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(123)
p &lt;- runif(1)
M &lt;- MakeM(p)

</code></pre>

<hr>
<h2 id='PairwiseList'>
Create a Pairwise List of Jacquard Coefficients
</h2><span id='topic+PairwiseList'></span>

<h3>Description</h3>

<p>Function <code>PairwiseList</code> processes the list structure of Jacquard coefficients
and converts it into to table of pairs of individuauls with their Jacquard coefficients.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PairwiseList(X, digits = 3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PairwiseList_+3A_x">X</code></td>
<td>
<p>A list structure of nine matrices of Jacquard coefficients.</p>
</td></tr>
<tr><td><code id="PairwiseList_+3A_digits">digits</code></td>
<td>
<p>Jacquard coefficients are rounded to the given number of digits.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix</p>


<h3>Author(s)</h3>

<p>Jan Graffelman (jan.graffelman@upc.edu)</p>


<h3>See Also</h3>

<p><code><a href="#topic+Jacquard.cls">Jacquard.cls</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(DeltaSimulatedPedigree)

ii &lt;- 1:3

SubSet &lt;- list(length = 9)
for (k in 1:9) {
  SubSet[[k]] &lt;- matrix(numeric(3^2), ncol = 3)
  SubSet[[k]] &lt;- DeltaSimulatedPedigree[[k]][ii,ii]
}

List &lt;- PairwiseList(SubSet)

print(List)
</code></pre>

<hr>
<h2 id='SimulatedPedigree'>
Simulated SNP data according to a Pedigree
</h2><span id='topic+SimulatedPedigree'></span>

<h3>Description</h3>

<p>A matrix containing 111 individuals (rows), pedigree information (first five columns) and 20.000 single
nucleotide polymorphisms (remaining columns) coded in (0,1,2) format and simulated according to a
pedigree of 20 unrelated founders with six posterior generations. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("SimulatedPedigree")</code></pre>


<h3>Format</h3>

<p>A data frame containing 111 rows and 20.005 columns.</p>


<h3>Details</h3>

<p>The SNP data was generated using the JGTeach package. The genotype counts in the object SimulatedPedigree
represent the counts of the minor allele.
</p>


<h3>References</h3>

<p>Goudet, J. (2022) JGTeach: JG Teaching material. R package version 0.1.9. https://github.com/jgx65
</p>
<p>Graffelman, J., Weir, B.S. and Goudet, J. (2024) Estimation of Jacquard's genetic identity coefficients with
bi-allelic variants by constrained least-squares. Preprint at bioRxiv <a href="https://doi.org/10.1101/2024.03.25.586682">doi:10.1101/2024.03.25.586682</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(SimulatedPedigree)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
