<!DOCTYPE html><html><head><title>Help for package JBrowseR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {JBrowseR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#assembly'><p>Create an assembly for a custom JBrowse view</p></a></li>
<li><a href='#default_session'><p>Create a default session for a custom JBrowse view</p></a></li>
<li><a href='#JBrowseR'><p>R interface to JBrowse 2 genome browser</p></a></li>
<li><a href='#JBrowseR-shiny'><p>Shiny bindings for JBrowseR</p></a></li>
<li><a href='#json_config'><p>Read in a JBrowse 2 JSON configuration file</p></a></li>
<li><a href='#serve_data'><p>Serve a local data directory for use with a browser</p></a></li>
<li><a href='#text_index'><p>Create configuration for a JBrowse 2 text index</p></a></li>
<li><a href='#theme'><p>Create a theme for a custom JBrowse 2 view</p></a></li>
<li><a href='#track_alignments'><p>Create an AlignmentsTrack for a custom JBrowse 2 view</p></a></li>
<li><a href='#track_data_frame'><p>Create a track from an R data frame for a custom JBrowse 2 view</p></a></li>
<li><a href='#track_feature'><p>Create a FeatureTrack for a custom JBrowse 2 view</p></a></li>
<li><a href='#track_variant'><p>Create a VariantTrack for a custom JBrowse 2 view</p></a></li>
<li><a href='#track_wiggle'><p>Create a WiggleTrack for a custom JBrowse 2 view</p></a></li>
<li><a href='#tracks'><p>Create a set of tracks for a custom JBrowse 2 view</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>An R Interface to the JBrowse 2 Genome Browser</td>
</tr>
<tr>
<td>Version:</td>
<td>0.10.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides an R interface to the JBrowse 2 genome browser.
    Enables embedding a JB2 genome browser in a Shiny app or R Markdown
    document. The browser can also be launched from an interactive R console.
    The browser can be loaded with a variety of common genomics data types,
    and can be used with a custom theme.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://gmod.github.io/JBrowseR/">https://gmod.github.io/JBrowseR/</a> <a href="https://github.com/GMOD/JBrowseR">https://github.com/GMOD/JBrowseR</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>htmltools, htmlwidgets, reactR, stringr, magrittr, readr,
jsonlite, httpuv, mime, cli, ids, dplyr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-20 16:45:54 UTC; cdiesh</td>
</tr>
<tr>
<td>Author:</td>
<td>Elliot Hershberg <a href="https://orcid.org/0000-0003-2068-3366"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Colin Diesh <a href="https://orcid.org/0000-0003-2629-7570"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  the JBrowse 2 Team [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Colin Diesh &lt;colin.diesh@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-20 17:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='assembly'>Create an assembly for a custom JBrowse view</h2><span id='topic+assembly'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for an
indexed fasta or bgzip fasta so that it can be used
as the assembly in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>assembly(assembly_data, bgzip = FALSE, aliases = NULL, refname_aliases = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="assembly_+3A_assembly_data">assembly_data</code></td>
<td>
<p>the URL to your fasta file</p>
</td></tr>
<tr><td><code id="assembly_+3A_bgzip">bgzip</code></td>
<td>
<p>whether or not your fasta is bgzip compressed</p>
</td></tr>
<tr><td><code id="assembly_+3A_aliases">aliases</code></td>
<td>
<p>a vector of strings of the aliases for the assembly</p>
</td></tr>
<tr><td><code id="assembly_+3A_refname_aliases">refname_aliases</code></td>
<td>
<p>the URL to a file containing reference
name aliases. For more info see
<a href="https://jbrowse.org/jb2/docs/config_guide#configuring-reference-name-aliasing">https://jbrowse.org/jb2/docs/config_guide#configuring-reference-name-aliasing</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The string returned by <code>assembly</code> is stringified JSON.
JBrowseR is an interface to JBrowse 2, which receives its
configuration in JSON format. The stringified JSON returned
by <code>assembly</code> is parsed into a JavaScript object in the
browser, and is used to configure the genome browser.
</p>
<p>It is important to note that while only the fasta file is
passed as an argument, <code>assembly</code> assumes that a fasta
index of the same name is located with the fasta file (as
well as a gzi file in the case of a bgzip fasta).
</p>
<p>For example:
</p>
<p><code>assembly("data/hg38.fa")</code>
</p>
<p>Assumes that <code>data/hg38.fa.fai</code> also exists.
</p>
<p><code>assembly("data/hg38.fa", bgzip = TRUE)</code>
</p>
<p>Assumes that <code>data/hg38.fa.fai</code> and <code>data/hg38.fa.gzi</code> both exist.
</p>
<p>This is a JBrowse 2 convention, and the default naming output of samtools
and bgzip.
</p>
<p>For more information on creating these files, visit
<a href="https://jbrowse.org/jb2/docs/quickstart_web#adding-a-genome-assembly">https://jbrowse.org/jb2/docs/quickstart_web#adding-a-genome-assembly</a>
</p>


<h3>Value</h3>

<p>a character vector of JBrowseR assembly configuration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)
</code></pre>

<hr>
<h2 id='default_session'>Create a default session for a custom JBrowse view</h2><span id='topic+default_session'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for a default session for your
browser. A default session is the set of tracks that are displayed when your
browser is first displayed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_session(assembly, displayed_tracks, display_assembly = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="default_session_+3A_assembly">assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td></tr>
<tr><td><code id="default_session_+3A_displayed_tracks">displayed_tracks</code></td>
<td>
<p>a vector of tracks generated by a <code>track_*</code>
command.</p>
</td></tr>
<tr><td><code id="default_session_+3A_display_assembly">display_assembly</code></td>
<td>
<p>a boolean determining whether the reference sequence
is visible or not. TRUE by default.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of stringified JSON configuration for the
defaultSession to be used by the browser when first loaded
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create the assembly configuration
assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)

# create variant and wiggle tracks
variant &lt;- track_variant(
 "clinvar.vcf.gz",
 assembly
)
wiggle &lt;- track_wiggle(
 "read-cov.bw",
 assembly
)

# create a default session with those tracks open by default
default_session &lt;- default_session(
 assembly,
 c(variant, wiggle)
)
</code></pre>

<hr>
<h2 id='JBrowseR'>R interface to JBrowse 2 genome browser</h2><span id='topic+JBrowseR'></span>

<h3>Description</h3>

<p>Embed a JBrowse 2 linear genome view in your Shiny app,
Rmd document, or interactive R console.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JBrowseR(view, ..., width = NULL, height = NULL, elementId = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JBrowseR_+3A_view">view</code></td>
<td>
<p>Which JBrowse 2 view to use. View, JsonView, ViewHg19, ViewHg38</p>
</td></tr>
<tr><td><code id="JBrowseR_+3A_...">...</code></td>
<td>
<p>The parameters passed on to the view</p>
</td></tr>
<tr><td><code id="JBrowseR_+3A_width">width</code></td>
<td>
<p>The width of the htmlwidget</p>
</td></tr>
<tr><td><code id="JBrowseR_+3A_height">height</code></td>
<td>
<p>The height of the htmlwidget</p>
</td></tr>
<tr><td><code id="JBrowseR_+3A_elementid">elementId</code></td>
<td>
<p>The elementId of the htmlwidget</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an htmlwidget of the JBrowse 2 linear genome view.
</p>

<hr>
<h2 id='JBrowseR-shiny'>Shiny bindings for JBrowseR</h2><span id='topic+JBrowseR-shiny'></span><span id='topic+JBrowseROutput'></span><span id='topic+renderJBrowseR'></span><span id='topic+JBrowseR_html'></span>

<h3>Description</h3>

<p>Output and render functions for using JBrowseR within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JBrowseROutput(outputId, width = "100%", height = "400px")

renderJBrowseR(expr, env = parent.frame(), quoted = FALSE)

JBrowseR_html(id, style, class, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JBrowseR-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_width">width</code></td>
<td>
<p>Must be a valid CSS unit or a number, which will be coerced to a string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit or a number, which will be coerced to a string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a JBrowseR</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_id">id</code></td>
<td>
<p>htmltools id</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_style">style</code></td>
<td>
<p>htmltools style</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_class">class</code></td>
<td>
<p>htmltools class</p>
</td></tr>
<tr><td><code id="JBrowseR-shiny_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the Shiny UI bindings for a JBrowseR htmlwidget
</p>
<p>the Shiny server bindings for a JBrowseR htmlwidget
</p>
<p>the root HTML element to render the React component in
</p>

<hr>
<h2 id='json_config'>Read in a JBrowse 2 JSON configuration file</h2><span id='topic+json_config'></span>

<h3>Description</h3>

<p>Reads in a JSON file with values for configuring
your browser. Looks for assembly, tracks, defaultSession,
and theme. Only assembly is explicitly required for a
working browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>json_config(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="json_config_+3A_file">file</code></td>
<td>
<p>the file path or URL to a JBrowse 2 configuration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: this is the most advanced API. It offers full control
to do anything possible in JavaScript with an embedded JBrowse 2
React component, but comes with a steeper learning curve. For
more details on JBrowse 2 configuration, visit:
<a href="https://jbrowse.org/jb2/docs/config_guide">https://jbrowse.org/jb2/docs/config_guide</a>
</p>
<p>An example JSON config is provided with this package
</p>


<h3>Value</h3>

<p>a character vector of JSON configuration from a JBrowse 2
configuration file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: json_config("./config.json")
</code></pre>

<hr>
<h2 id='serve_data'>Serve a local data directory for use with a browser</h2><span id='topic+serve_data'></span>

<h3>Description</h3>

<p>This is a utility function that can be used to server
a local directory with data so that it can be used
in the genome browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>serve_data(path, port = 5000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="serve_data_+3A_path">path</code></td>
<td>
<p>The path to the directory with data to serve</p>
</td></tr>
<tr><td><code id="serve_data_+3A_port">port</code></td>
<td>
<p>The port to serve the directory on</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: This is intended for local development and use.
For a production deployment, refer to the vignette
on creating URLs for more robust options.
</p>


<h3>Value</h3>

<p>a list containing information about the newly
created HTTP server including the host, port, interval,
and URL. The list also contains the stop_server()
function which can be used to stop the server
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
server &lt;- serve_data("~/path/to/my-data")
# use server$stop_server() to stop

## End(Not run)
</code></pre>

<hr>
<h2 id='text_index'>Create configuration for a JBrowse 2 text index</h2><span id='topic+text_index'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for an adapter to a text index for
gene name search in the browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>text_index(ix_uri, ixx_uri, meta_uri, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="text_index_+3A_ix_uri">ix_uri</code></td>
<td>
<p>the URI for the ix file</p>
</td></tr>
<tr><td><code id="text_index_+3A_ixx_uri">ixx_uri</code></td>
<td>
<p>the URI for the ixx file</p>
</td></tr>
<tr><td><code id="text_index_+3A_meta_uri">meta_uri</code></td>
<td>
<p>the URI for the JSON metadata file</p>
</td></tr>
<tr><td><code id="text_index_+3A_assembly">assembly</code></td>
<td>
<p>the assembly associated with the text index</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: this function currently only supports aggregate indices.
</p>
<p>For more information on JBrowse 2 text indices, visit:
<a href="https://jbrowse.org/jb2/docs/config_guide/#text-searching">https://jbrowse.org/jb2/docs/config_guide/#text-searching</a>
</p>


<h3>Value</h3>

<p>a character vector with the JSON text index adapter.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>text_index(
"https://jbrowse.org/genomes/hg19/trix/hg19.ix",
"https://jbrowse.org/genomes/hg19/trix/hg19.ixx",
"https://jbrowse.org/genomes/hg19/trix/meta.json",
"hg19"
)
</code></pre>

<hr>
<h2 id='theme'>Create a theme for a custom JBrowse 2 view</h2><span id='topic+theme'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string
for a custom theme palette for your browser.
Accepts up to four hexadecimal colors. For more
information on how JBrowse 2 custom themes work,
visit <a href="https://jbrowse.org/jb2/docs/config_guide#configuring-the-theme">https://jbrowse.org/jb2/docs/config_guide#configuring-the-theme</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme(primary, secondary = NULL, tertiary = NULL, quaternary = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="theme_+3A_primary">primary</code></td>
<td>
<p>the primary color of your custom palette</p>
</td></tr>
<tr><td><code id="theme_+3A_secondary">secondary</code></td>
<td>
<p>the secondary color of your custom palette</p>
</td></tr>
<tr><td><code id="theme_+3A_tertiary">tertiary</code></td>
<td>
<p>the tertiary color of your custom palette</p>
</td></tr>
<tr><td><code id="theme_+3A_quaternary">quaternary</code></td>
<td>
<p>the quaternary color of your custom palette</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of stringified theme JSON configuration
to configure a custom color palette for the browser
</p>


<h3>Examples</h3>

<pre><code class='language-R'>theme("#311b92")
theme("#311b92", "#0097a7")
theme("#311b92", "#0097a7", "#f57c00")
theme("#311b92", "#0097a7", "#f57c00", "#d50000")
</code></pre>

<hr>
<h2 id='track_alignments'>Create an AlignmentsTrack for a custom JBrowse 2 view</h2><span id='topic+track_alignments'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for an
indexed BAM or CRAM alignment so that it can be used
in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_alignments(track_data, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_alignments_+3A_track_data">track_data</code></td>
<td>
<p>the URL to the BAM/CRAM alignments</p>
</td></tr>
<tr><td><code id="track_alignments_+3A_assembly">assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is important to note that while only the BAM/CRAM file is
passed as an argument, <code>tracks_alignment</code> assumes that a BAM/CRAM
index of the same name is located with the file
</p>
<p>For example:
</p>
<p><code>track_alignments("data/alignments.bam")</code>
</p>
<p>Assumes that <code>data/alignments.bam.bai</code> also exists.
</p>
<p>This is a JBrowse 2 convention, and the default naming output of samtools
</p>
<p>For more information on creating an index with samtools, visit
<a href="https://www.htslib.org/">https://www.htslib.org/</a>
</p>


<h3>Value</h3>

<p>a character vector of stringified AlignmentsTrack JSON configuration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)

track_alignments("alignments.bam", assembly)
track_alignments("alignments.cram", assembly)
</code></pre>

<hr>
<h2 id='track_data_frame'>Create a track from an R data frame for a custom JBrowse 2 view</h2><span id='topic+track_data_frame'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for an R data frame so that it can
be viewed as a track in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_data_frame(track_data, track_name, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_data_frame_+3A_track_data">track_data</code></td>
<td>
<p>the data frame with track data. Must have cols:
<code>chrom</code>, <code>start</code>, <code>end</code>, <code>name</code>. The column
<code>additional</code> can optionally be include with more feature information.
If a <code>score</code> column is present, it will be used and the track will
be rendered to display quantitative features.</p>
</td></tr>
<tr><td><code id="track_data_frame_+3A_track_name">track_name</code></td>
<td>
<p>the name to use for the track</p>
</td></tr>
<tr><td><code id="track_data_frame_+3A_assembly">assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of stringified track JSON configuration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)

df &lt;- data.frame(
        chrom = c(1, 2),
        start = c(123, 456),
        end = c(789, 101112),
        name = c('feature1', 'feature2')
      )

track_data_frame(df, "my_features", assembly)
</code></pre>

<hr>
<h2 id='track_feature'>Create a FeatureTrack for a custom JBrowse 2 view</h2><span id='topic+track_feature'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for an
indexed GFF3 file so that it can be used
in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_feature(track_data, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_feature_+3A_track_data">track_data</code></td>
<td>
<p>the URL to the GFF3 file</p>
</td></tr>
<tr><td><code id="track_feature_+3A_assembly">assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is important to note that while only the GFF3 file is
passed as an argument, <code>tracks_variant</code> assumes that a GFF3
index of the same name is located with the file
</p>
<p>For example:
</p>
<p><code>track_feature("data/features.gff")</code>
</p>
<p>Assumes that <code>data/features.gff.tbi</code> also exists.
</p>
<p>This is a JBrowse 2 convention, and the default naming output of tabix
</p>
<p>For more information on creating an index with tabix, visit
<a href="https://www.htslib.org/">https://www.htslib.org/</a>
</p>


<h3>Value</h3>

<p>a character vector of stringified FeatureTrack JSON configuration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)

track_feature("features.gff", assembly)
</code></pre>

<hr>
<h2 id='track_variant'>Create a VariantTrack for a custom JBrowse 2 view</h2><span id='topic+track_variant'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for an
indexed VCF file so that it can be used
in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_variant(track_data, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_variant_+3A_track_data">track_data</code></td>
<td>
<p>the URL to the VCF file</p>
</td></tr>
<tr><td><code id="track_variant_+3A_assembly">assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>It is important to note that while only the VCF file is
passed as an argument, <code>tracks_variant</code> assumes that a VCF
index of the same name is located with the file
</p>
<p>For example:
</p>
<p><code>track_alignments("data/variants.vcf")</code>
</p>
<p>Assumes that <code>data/variants.vcf.tbi</code> also exists.
</p>
<p>This is a JBrowse 2 convention, and the default naming output of tabix
</p>
<p>For more information on creating an index with tabix, visit
<a href="https://www.htslib.org/">https://www.htslib.org/</a>
</p>


<h3>Value</h3>

<p>a character vector of stringified VariantTrack JSON configuration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)

track_variant("variants.vcf", assembly)
</code></pre>

<hr>
<h2 id='track_wiggle'>Create a WiggleTrack for a custom JBrowse 2 view</h2><span id='topic+track_wiggle'></span>

<h3>Description</h3>

<p>Creates the necessary configuration string for a
bigWig file so that it can be used
in a JBrowse custom linear genome view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_wiggle(track_data, assembly)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_wiggle_+3A_track_data">track_data</code></td>
<td>
<p>the URL to the bigWig file</p>
</td></tr>
<tr><td><code id="track_wiggle_+3A_assembly">assembly</code></td>
<td>
<p>the config string generated by <code>assembly</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of stringified WiggleTrack JSON configuration
</p>


<h3>Examples</h3>

<pre><code class='language-R'>track_wiggle(
  "https://jbrowse.org/genomes/hg19/COLO829/colo_normal.bw",
  assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)
)
</code></pre>

<hr>
<h2 id='tracks'>Create a set of tracks for a custom JBrowse 2 view</h2><span id='topic+tracks'></span>

<h3>Description</h3>

<p>Accepts any number of tracks, returns the configuration
string necessary to load these tracks into your JBrowse
view.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tracks(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tracks_+3A_...">...</code></td>
<td>
<p>The tracks to be added to the JBrowse 2 view</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector of stringified JSON configuration for
all tracks to add to the browser
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create an assembly configuration and alignments track
assembly &lt;- assembly("https://jbrowse.org/genomes/hg19/fasta/hg19.fa.gz", bgzip = TRUE)
alignments &lt;- track_alignments("alignments.bam", assembly)

# create a tracks configuration with the alignments track
tracks(alignments)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
