<!DOCTYPE html><html lang="en"><head><title>Help for package jellyfisher</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {jellyfisher}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#extract_tables_from_clonevol'><p>Extract Data for Jellyfisher from ClonEvol Results</p></a></li>
<li><a href='#jellyfisher'><p>Creates a Jellyfish plot</p></a></li>
<li><a href='#jellyfisher_example_tables'><p>Jellyfisher example data</p></a></li>
<li><a href='#jellyfisher-shiny'><p>Shiny bindings for jellyfisher</p></a></li>
<li><a href='#select_patients'><p>Filter Jellyfish tables by patient</p></a></li>
<li><a href='#set_parents'><p>Set parents for samples</p></a></li>
<li><a href='#set_ranks'><p>Set ranks for samples</p></a></li>
<li><a href='#validate_tables'><p>Validate Jellyfish tables</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Visualize Spatiotemporal Tumor Evolution with Jellyfish Plots</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Description:</td>
<td>
    Generates interactive Jellyfish plots to visualize spatiotemporal tumor
    evolution by integrating sample and phylogenetic trees into a unified plot.
    This approach provides an intuitive way to analyze tumor heterogeneity and
    evolution over time and across anatomical locations. The Jellyfish plot
    visualization design was first introduced by Lahtinen, Lavikka, et al.
    (2023, &lt;<a href="https://doi.org/10.1016%2Fj.ccell.2023.04.017">doi:10.1016/j.ccell.2023.04.017</a>&gt;).
    This package also supports visualizing ClonEvol results, a tool developed
    by Dang, et al. (2017, &lt;<a href="https://doi.org/10.1093%2Fannonc%2Fmdx517">doi:10.1093/annonc/mdx517</a>&gt;), for analyzing clonal
    evolution from multi-sample sequencing data. The 'clonevol' package is not
    available on CRAN but can be installed from its GitHub
    repository (<a href="https://github.com/hdng/clonevol">https://github.com/hdng/clonevol</a>).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/HautaniemiLab/jellyfisher">https://github.com/HautaniemiLab/jellyfisher</a>,
<a href="https://hautaniemilab.github.io/jellyfisher/">https://hautaniemilab.github.io/jellyfisher/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/HautaniemiLab/jellyfisher/issues">https://github.com/HautaniemiLab/jellyfisher/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>htmlwidgets, dplyr, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, clonevol</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Visualization, Phylogenetics, Software, Spatial</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-02-05 12:40:36 UTC; klavikka</td>
</tr>
<tr>
<td>Author:</td>
<td>Kari Lavikka <a href="https://orcid.org/0000-0002-4163-4945"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cph, aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kari Lavikka &lt;kari.lavikka@helsinki.fi&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-02-06 19:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='extract_tables_from_clonevol'>Extract Data for Jellyfisher from ClonEvol Results</h2><span id='topic+extract_tables_from_clonevol'></span>

<h3>Description</h3>

<p>ClonEvol infers clonal evolution from multi-sample cancer sequencing data
and generates phylogenetic models of tumor evolution.
This function extracts data frames from a ClonEvol object that can be
used to create a Jellyfish plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_tables_from_clonevol(y, model = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_tables_from_clonevol_+3A_y">y</code></td>
<td>
<p>A ClonEvol object</p>
</td></tr>
<tr><td><code id="extract_tables_from_clonevol_+3A_model">model</code></td>
<td>
<p>The model to extract. Defaults to 1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: ClonEvol reports clonal prevalences as confidence intervals.
This function extracts the mean values and uses them as the prevalence
values.
</p>
<p>For more details about ClonEvol, including the installation instructions,
visit its GitHub <a href="https://github.com/hdng/clonevol">repository</a> or read
the publication by Dang et al. (2017, <a href="https://doi.org/10.1093/annonc/mdx517">doi:10.1093/annonc/mdx517</a>).
</p>


<h3>Value</h3>

<p>A named list with three data frames: samples, phylogeny,
and compositions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("clonevol", quietly = TRUE)) {
  # Run ClonEvol with its example data
  # (refer to ClonEvol documentation for details)
  data &lt;- clonevol::aml1
  y &lt;- clonevol::infer.clonal.models(
    variants = data$variants,
    cluster.col.name = "cluster",
    vaf.col.names = data$params$vaf.col.names,
    subclonal.test = "bootstrap",
    subclonal.test.model = "non-parametric",
    num.boots = 1000,
    founding.cluster = 1,
    cluster.center = "mean",
    ignore.clusters = NULL,
    min.cluster.vaf = 0.01,
    sum.p = 0.05,
    alpha = 0.05
  )
  # Make branch lengths available
  y &lt;- clonevol::convert.consensus.tree.clone.to.branch(y)

  # Extract data and plot the results
  extract_tables_from_clonevol(y, model = 1) |&gt;
    jellyfisher()
} else {
  message(
    "Please install the clonevol package from GitHub: devtools::install_github('hdng/clonevol')"
  )
}

</code></pre>

<hr>
<h2 id='jellyfisher'>Creates a Jellyfish plot</h2><span id='topic+jellyfisher'></span>

<h3>Description</h3>

<p>Creates a Jellyfish plot from samples, a phylogeny, and subclonal compositions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jellyfisher(
  tables,
  options = list(),
  controls = "closed",
  width = NULL,
  height = NULL,
  elementId = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jellyfisher_+3A_tables">tables</code></td>
<td>
<p>A named list of data frames. The list must contain the following elements:
</p>

<dl>
<dt><code>samples</code></dt><dd><p>A data frame with sample data. The expected columns are:
</p>

<dl>
<dt>sample</dt><dd><p>specifies the unique identifier for each sample. (string)</p>
</dd>
<dt>displayName</dt><dd><p>allows for specifying a custom name for each sample. If the column is omitted, the <code>sample</code> column is used as the display name. (string, optional)</p>
</dd>
<dt>rank</dt><dd><p>specifies the position of each sample in the Jellyfish plot. For example, different stages of a disease can be ranked in chronological order: diagnosis (1), interval (2), and relapse (3). The zeroth rank is reserved for the root of the sample tree. Ranks can be any integer, and unused ranks are automatically excluded from the plot. If the <code>rank</code> column is (integer)</p>
</dd>
<dt>parent</dt><dd><p>identifies the parent sample for each entry. Samples without a specified parent are treated as children of an imaginary root sample. (string)</p>
</dd>
</dl>

</dd>
<dt><code>phylogeny</code></dt><dd><p>A data frame with phylogeny data. The expected columns are:
</p>

<dl>
<dt>subclone</dt><dd><p>specifies subclone IDs, which can be any string. (string)</p>
</dd>
<dt>parent</dt><dd><p>designates the parent subclone. The subclone without a parent is considered the root of the phylogeny. (string)</p>
</dd>
<dt>color</dt><dd><p>specifies the color for the subclone. If the column is omitted, colors will be generated automatically. (string, optional)</p>
</dd>
<dt>branchLength</dt><dd><p>specifies the length of the branch leading to the subclone. The length may be based on, for example, the number of unique mutations in the subclone. The branch length is shown in the Jellyfish plot's legend as a bar chart. It is also used when generating a phylogeny-aware color scheme. (number)</p>
</dd>
</dl>

</dd>
<dt><code>compositions</code></dt><dd><p>A data frame with subclonal compositions. The expected columns are:
</p>

<dl>
<dt>sample</dt><dd><p>specifies the sample ID. (string)</p>
</dd>
<dt>subclone</dt><dd><p>specifies the subclone ID. (string)</p>
</dd>
<dt>clonalPrevalence</dt><dd><p>specifies the clonal prevalence of the subclone in the sample. The clonal prevalence is the proportion of the subclone in the sample. The clonal prevalences in a sample must sum to 1. (number)</p>
</dd>
</dl>

</dd>
<dt><code>ranks</code></dt><dd><p>An optional data frame with ranks. The expected columns are:
</p>

<dl>
<dt>rank</dt><dd><p>specifies the rank number. The zeroth rank is reserved for the inferred root of the sample tree. However, you are free to define a title for it. (integer)</p>
</dd>
<dt>title</dt><dd><p>specifies the title for the rank. (string)</p>
</dd>
</dl>

</dd>
</dl>
</td></tr>
<tr><td><code id="jellyfisher_+3A_options">options</code></td>
<td>
<p>A named list of options to configure the plot. Available options:
</p>

<dl>
<dt>crossingWeight</dt><dd><p>Weight for tentacle bundles between two pairs of samples crossing each other. Defaults to <code>10</code>.</p>
</dd>
<dt>pathLengthWeight</dt><dd><p>Weight for the total length of the paths (tentacle bundles) connecting samples. Defaults to <code>2</code>.</p>
</dd>
<dt>orderMismatchWeight</dt><dd><p>Weight for the mismatch in the order of samples. The order is based on the &quot;phylogenetic center of mass&quot; computed from the subclonal compositions. Defaults to <code>2</code>.</p>
</dd>
<dt>bundleMismatchWeight</dt><dd><p>Weight for the mismatch in the placement of bundles. The &quot;optimal&quot; placement is based on the subclonal compositions, but such placement may produce excessively long tentacle bundles. Defaults to <code>3</code>.</p>
</dd>
<dt>divergenceWeight</dt><dd><p>Weight for the sum of divergences between adjacent samples. Defaults to <code>4</code>.</p>
</dd>
<dt>bellTipShape</dt><dd><p>The shape of the bell tip. 0 is a sharp tip, 1 is a blunt tip. Defaults to <code>0.1</code>.</p>
</dd>
<dt>bellTipSpread</dt><dd><p>How much to spread nested bell tips. 0 is no spread, 1 is full spread. Defaults to <code>0.5</code>.</p>
</dd>
<dt>bellStrokeWidth</dt><dd><p>The width of strokes in the bell. Defaults to <code>1</code>.</p>
</dd>
<dt>bellStrokeDarkening</dt><dd><p>How much the stroke color of the bells is darkened. Defaults to <code>0.6</code>.</p>
</dd>
<dt>bellPlateauPos</dt><dd><p>Where the bell has fully appeared and the plateau starts. Defaults to <code>0.75</code>.</p>
</dd>
<dt>sampleHeight</dt><dd><p>Height of real sample nodes Defaults to <code>110</code>.</p>
</dd>
<dt>sampleWidth</dt><dd><p>Width of sample nodes Defaults to <code>90</code>.</p>
</dd>
<dt>inferredSampleHeight</dt><dd><p>Height of inferred sample nodes Defaults to <code>120</code>.</p>
</dd>
<dt>gapHeight</dt><dd><p>Height of gaps between samples. Gaps are routes for tentacle bundles. Defaults to <code>60</code>.</p>
</dd>
<dt>sampleSpacing</dt><dd><p>Vertical space between samples Defaults to <code>60</code>.</p>
</dd>
<dt>columnSpacing</dt><dd><p>Horizontal space between columns Defaults to <code>90</code>.</p>
</dd>
<dt>tentacleWidth</dt><dd><p>Width of tentacles in pixels Defaults to <code>2</code>.</p>
</dd>
<dt>tentacleSpacing</dt><dd><p>Space between tentacles in a bundle, in pixels Defaults to <code>5</code>.</p>
</dd>
<dt>inOutCPDistance</dt><dd><p>Relative distance of tentacle control points from the edge of the sample node Defaults to <code>0.3</code>.</p>
</dd>
<dt>bundleCPDistance</dt><dd><p>Relative distance of tentacle bundle's control points. The higher the value, the longer the individual tentacles stay together before diverging. Defaults to <code>0.6</code>.</p>
</dd>
<dt>sampleFontSize</dt><dd><p>Font size for sample labels Defaults to <code>12</code>.</p>
</dd>
<dt>showLegend</dt><dd><p>Whether to show the legend Defaults to <code>TRUE</code>.</p>
</dd>
<dt>phylogenyColorScheme</dt><dd><p>Whether to use a color scheme based on phylogeny Defaults to <code>TRUE</code>.</p>
</dd>
<dt>phylogenyHueOffset</dt><dd><p>Offset for the hue of the phylogeny color scheme. If the automatically generated hues are not to your liking, you can adjust the hue offset to get a different color scheme. Defaults to <code>0</code>.</p>
</dd>
<dt>sampleTakenGuide</dt><dd><p>Type of the &quot;sample taken&quot; guide. <code>"none"</code> for no guides, <code>"line"</code> for a faint dashed line in all samples, <code>"text"</code> same as line, but with a text label in one of the samples. <code>"text-all"</code> same as text, but with a text label in all samples. Defaults to <code>"text"</code>.</p>
</dd>
<dt>showRankTitles</dt><dd><p>Whether to show rank titles above the samples (if provided). Defaults to <code>TRUE</code>.</p>
</dd>
<dt>normalsAtPhylogenyRoot</dt><dd><p>Whether the root of the phylogenetic tree contains normal cells. If true, no tentacles will be drawn for the root clone and its color will be white if phylogenyColorScheme is used. Defaults to <code>FALSE</code>.</p>
</dd>
</dl>
</td></tr>
<tr><td><code id="jellyfisher_+3A_controls">controls</code></td>
<td>
<p>An optional parameter to set the initial state of the controls. Can be &quot;open&quot;, &quot;closed&quot;, or &quot;hidden&quot;.</p>
</td></tr>
<tr><td><code id="jellyfisher_+3A_width">width</code></td>
<td>
<p>The width of the widget</p>
</td></tr>
<tr><td><code id="jellyfisher_+3A_height">height</code></td>
<td>
<p>The height of the widget</p>
</td></tr>
<tr><td><code id="jellyfisher_+3A_elementid">elementId</code></td>
<td>
<p>An optional element ID for the widget</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The format of the data frames is described with examples in Jellyfish documentation:
https://github.com/HautaniemiLab/jellyfish?tab=readme-ov-file#input-data
</p>


<h3>Value</h3>

<p>A Jellyfish plot HTML widget
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Plot the bundled example data
jellyfisher(jellyfisher_example_tables,
            options = list(
              sampleHeight = 70,
              sampleTakenGuide = "none",
              showLegend = FALSE
            ))

</code></pre>

<hr>
<h2 id='jellyfisher_example_tables'>Jellyfisher example data</h2><span id='topic+jellyfisher_example_tables'></span>

<h3>Description</h3>

<p>Example data based on the following publication:
Lahtinen A, Lavikka K, Virtanen A, et al.
Evolutionary states and trajectories characterized by distinct pathways
stratify patients with ovarian high grade serous carcinoma.
<em>Cancer Cell</em>. 2023;41(6):1103-1117.e12.
<a href="https://doi.org/10.1016/j.ccell.2023.04.017">doi:10.1016/j.ccell.2023.04.017</a>
</p>


<h3>Format</h3>



<h4><code>jellyfisher_example_tables</code></h4>

<p>A named list of data frames containing the following tables:
</p>

<dl>
<dt><code>samples</code></dt><dd><p>A data frame with sample data. Columns are:
</p>

<dl>
<dt>sample</dt><dd><p>specifies the unique identifier for each sample. (string)</p>
</dd>
<dt>displayName</dt><dd><p>allows for specifying a custom name for each sample. If the column is omitted, the <code>sample</code> column is used as the display name. (string, optional)</p>
</dd>
<dt>rank</dt><dd><p>specifies the position of each sample in the Jellyfish plot. For example, different stages of a disease can be ranked in chronological order: diagnosis (1), interval (2), and relapse (3). The zeroth rank is reserved for the root of the sample tree. Ranks can be any integer, and unused ranks are automatically excluded from the plot. If the <code>rank</code> column is (integer)</p>
</dd>
<dt>parent</dt><dd><p>identifies the parent sample for each entry. Samples without a specified parent are treated as children of an imaginary root sample. (string)</p>
</dd>
</dl>

</dd>
<dt><code>phylogeny</code></dt><dd><p>A data frame with phylogeny data. Columns are:
</p>

<dl>
<dt>subclone</dt><dd><p>specifies subclone IDs, which can be any string. (string)</p>
</dd>
<dt>parent</dt><dd><p>designates the parent subclone. The subclone without a parent is considered the root of the phylogeny. (string)</p>
</dd>
<dt>color</dt><dd><p>specifies the color for the subclone. If the column is omitted, colors will be generated automatically. (string, optional)</p>
</dd>
<dt>branchLength</dt><dd><p>specifies the length of the branch leading to the subclone. The length may be based on, for example, the number of unique mutations in the subclone. The branch length is shown in the Jellyfish plot's legend as a bar chart. It is also used when generating a phylogeny-aware color scheme. (number)</p>
</dd>
</dl>

</dd>
<dt><code>compositions</code></dt><dd><p>A data frame with subclonal compositions. Columns are:
</p>

<dl>
<dt>sample</dt><dd><p>specifies the sample ID. (string)</p>
</dd>
<dt>subclone</dt><dd><p>specifies the subclone ID. (string)</p>
</dd>
<dt>clonalPrevalence</dt><dd><p>specifies the clonal prevalence of the subclone in the sample. The clonal prevalence is the proportion of the subclone in the sample. The clonal prevalences in a sample must sum to 1. (number)</p>
</dd>
</dl>

</dd>
<dt><code>ranks</code></dt><dd><p>An optional data frame with ranks. Columns are:
</p>

<dl>
<dt>rank</dt><dd><p>specifies the rank number. The zeroth rank is reserved for the inferred root of the sample tree. However, you are free to define a title for it. (integer)</p>
</dd>
<dt>title</dt><dd><p>specifies the title for the rank. (string)</p>
</dd>
</dl>

</dd>
</dl>




<h3>Source</h3>

<p><a href="https://github.com/HautaniemiLab/jellyfish/tree/main/data">https://github.com/HautaniemiLab/jellyfish/tree/main/data</a>
</p>

<hr>
<h2 id='jellyfisher-shiny'>Shiny bindings for jellyfisher</h2><span id='topic+jellyfisher-shiny'></span><span id='topic+jellyfisherOutput'></span><span id='topic+renderJellyfisher'></span>

<h3>Description</h3>

<p>Output and render functions for using jellyfisher within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jellyfisherOutput(outputId, width = "100%", height = "400px")

renderJellyfisher(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jellyfisher-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="jellyfisher-shiny_+3A_width">width</code>, <code id="jellyfisher-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="jellyfisher-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a jellyfisher</p>
</td></tr>
<tr><td><code id="jellyfisher-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="jellyfisher-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>


<h3>Value</h3>


<dl>
<dt><code>jellyfisherOutput()</code></dt><dd>
<p>A <code>shiny.tag</code> object that can be included in a Shiny UI definition.</p>
</dd>
<dt><code>renderJellyfisher()</code></dt><dd>
<p>A <code>shiny.render.function</code> that returns an <code>htmlwidget</code> object for use in
a Shiny server function.</p>
</dd>
</dl>


<hr>
<h2 id='select_patients'>Filter Jellyfish tables by patient</h2><span id='topic+select_patients'></span>

<h3>Description</h3>

<p>Given a list of tables, filter them by patient.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_patients(tables, patient)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="select_patients_+3A_tables">tables</code></td>
<td>
<p>A list of tables (samples, phylogeny, compositions, ranks)</p>
</td></tr>
<tr><td><code id="select_patients_+3A_patient">patient</code></td>
<td>
<p>The patient or patients to filter by</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables filtered by patient
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jellyfisher_example_tables |&gt;
  select_patients("EOC809")

</code></pre>

<hr>
<h2 id='set_parents'>Set parents for samples</h2><span id='topic+set_parents'></span>

<h3>Description</h3>

<p>Given a list of jellyfish input tables and a named list of parents for each
sample, set the parent for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_parents(tables, parents, unset_missing = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_parents_+3A_tables">tables</code></td>
<td>
<p>A list of tables (samples, phylogeny, compositions, ranks)</p>
</td></tr>
<tr><td><code id="set_parents_+3A_parents">parents</code></td>
<td>
<p>A named list of parents for each sample. Keys are the samples
and values are their new parents</p>
</td></tr>
<tr><td><code id="set_parents_+3A_unset_missing">unset_missing</code></td>
<td>
<p>If TRUE, unset the parent for samples that are not in
the parent list</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By default, all samples that have no explicit parent are children of the
<em>inferred root</em> sample. You can customize the parent-child relationships by
modifying the <code>parent</code> column in the <code>samples</code> data frame before plotting.
</p>
<p>You can also modify the relationships using the <code>set_parents</code> function.
</p>
<p>For example, if you have three samples, A, B, and C, they will have the
following relationships by default:
</p>
<div class="sourceCode"><pre>    Root
   / | \
  A  B  C
</pre></div>
<p>With the explicit parents, you can customize the relationships:
</p>
<div class="sourceCode R"><pre>tables |&gt;
  set_parents(list(
    # The parent of C is B
    C = "B"
  ) |&gt;
  jellyfisher()
</pre></div>
<div class="sourceCode"><pre>    Root
   /  \
  A    B
        \
         C
</pre></div>


<h3>Value</h3>

<p>A list of tables with parents set for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jellyfisher_example_tables |&gt;
  select_patients("EOC809") |&gt;
  set_parents(list("EOC809_r1Bow1_DNA1" = "EOC809_p2Per1_cO_DNA2")) |&gt;
  jellyfisher()

</code></pre>

<hr>
<h2 id='set_ranks'>Set ranks for samples</h2><span id='topic+set_ranks'></span>

<h3>Description</h3>

<p>Given a list of jellyfish input tables and a named list of ranks for each
sample, set the rank for each sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_ranks(tables, ranks, default = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_ranks_+3A_tables">tables</code></td>
<td>
<p>A list of tables (samples, phylogeny, compositions, ranks)</p>
</td></tr>
<tr><td><code id="set_ranks_+3A_ranks">ranks</code></td>
<td>
<p>A named list of ranks for each sample</p>
</td></tr>
<tr><td><code id="set_ranks_+3A_default">default</code></td>
<td>
<p>The default rank to use when a sample is not in the rank list
(default: 1)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of tables with ranks set for each sample
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jellyfisher_example_tables |&gt;
  select_patients("EOC809") |&gt;
  set_ranks(list("EOC809_r1Bow1_DNA1" = 2, "EOC809_p2Per1_cO_DNA2" = 2),
    default = 1
  ) |&gt;
  jellyfisher()

</code></pre>

<hr>
<h2 id='validate_tables'>Validate Jellyfish tables</h2><span id='topic+validate_tables'></span>

<h3>Description</h3>

<p>Superficially validate that the tables are in the correct format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_tables(tables)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="validate_tables_+3A_tables">tables</code></td>
<td>
<p>A list of tables (samples, phylogeny, compositions, ranks)</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
