<!DOCTYPE html><html><head><title>Help for package jetset</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {jetset}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#jmap'><p>Retrieve jetset mapped probe sets</p></a></li>
<li><a href='#jscores'><p>Retrieve jetset scores for probe sets</p></a></li>
<li><a href='#scores'><p>Data: Probe set quality scores</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>3.4.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-05</td>
</tr>
<tr>
<td>Title:</td>
<td>One-to-One Gene-Probeset Mapping for Affymetrix Human
Microarrays</td>
</tr>
<tr>
<td>Author:</td>
<td>Qiyuan Li, Aron Eklund</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aron Eklund &lt;eklund@cbs.dtu.dk&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>org.Hs.eg.db, AnnotationDbi</td>
</tr>
<tr>
<td>Description:</td>
<td>On Affymetrix gene expression microarrays, a single gene may be measured by multiple probe sets. This can present a mild conundrum when attempting to evaluate a gene "signature" that is defined by gene names rather than by specific probe sets. This package provides a one-to-one mapping from gene to "best" probe set for four Affymetrix human gene expression microarrays: hgu95av2, hgu133a, hgu133plus2, and u133x3p. This package also includes the pre-calculated probe set quality scores that were used to define the mapping.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://www.cbs.dtu.dk/biotools/jetset/">http://www.cbs.dtu.dk/biotools/jetset/</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-05 08:11:37 UTC; aron</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-05 11:15:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='jmap'>Retrieve jetset mapped probe sets</h2><span id='topic+jmap'></span>

<h3>Description</h3>

<p>This function retrieves probe sets corresponding to the queried genes</p>


<h3>Usage</h3>

<pre><code class='language-R'>jmap(chip, eg, symbol, alias, ensembl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jmap_+3A_chip">chip</code></td>
<td>
<p> Chip name </p>
</td></tr>
<tr><td><code id="jmap_+3A_eg">eg</code></td>
<td>
<p> A vector of Entrez GeneIDs (optional) </p>
</td></tr>
<tr><td><code id="jmap_+3A_symbol">symbol</code></td>
<td>
<p> A vector of gene symbols (optional) </p>
</td></tr>
<tr><td><code id="jmap_+3A_alias">alias</code></td>
<td>
<p> A vector of gene aliases (optional) </p>
</td></tr>
<tr><td><code id="jmap_+3A_ensembl">ensembl</code></td>
<td>
<p> A vector of Ensembl IDs (optional) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently, <code>chip</code> can be <code>"hgu95av2"</code>, <code>"hgu133a"</code>, <code>"hgu133plus2"</code>, or <code>"u133x3p"</code>.  
</p>
<p>Queried genes must be specified by either <code>eg</code>, <code>symbol</code>, <code>alias</code>, or <code>ensembl</code>.
</p>
<p>If the query is not recognized, or is ambiguous, or corresponds to a gene
that is not detected by the array, <code>NA</code> will be returned.
</p>
<p>Details about the jetset algorithm are available in the vignette.
</p>


<h3>Value</h3>

<p>A character vector of probe set IDs</p>


<h3>References</h3>

<p>Qiyuan Li, Nicolai J. Birkbak, Balazs Gyorffy, Zoltan Szallasi and Aron C. Eklund. (2011) Jetset: selecting the optimal microarray probe set to represent a gene. BMC Bioinformatics. 12:474.
</p>


<h3>See Also</h3>

<p>The underlying Entrez ID to probeset data is available in (e.g.) <code><a href="#topic+scores.hgu95av2">scores.hgu95av2</a></code>.
Symbol, alias, and ensembl lookups are generated from e.g.
<code><a href="org.Hs.eg.db.html#topic+org.Hs.egSYMBOL2EG">org.Hs.egSYMBOL2EG</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  genes &lt;- c('MKI67', 'CHD5', 'ESR1', 'FGF19', 'ERBB2', 'NoSuchGene')

  # This generates several informative warnings
  jmap('hgu133a', symbol = genes)

</code></pre>

<hr>
<h2 id='jscores'>Retrieve jetset scores for probe sets</h2><span id='topic+jscores'></span>

<h3>Description</h3>

<p>This function retrieves jetset scores, which indicate the predicted quality of individual probe sets on selected Affymetrix microarrays.</p>


<h3>Usage</h3>

<pre><code class='language-R'>jscores(chip, probeset, eg, symbol, alias, ensembl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="jscores_+3A_chip">chip</code></td>
<td>
<p> Chip name </p>
</td></tr>
<tr><td><code id="jscores_+3A_probeset">probeset</code></td>
<td>
<p> A vector of probe set IDs (optional) </p>
</td></tr>
<tr><td><code id="jscores_+3A_eg">eg</code></td>
<td>
<p> A vector of Entrez GeneIDs (optional) </p>
</td></tr>
<tr><td><code id="jscores_+3A_symbol">symbol</code></td>
<td>
<p> A vector of gene symbols (optional) </p>
</td></tr>
<tr><td><code id="jscores_+3A_alias">alias</code></td>
<td>
<p> A vector of gene aliases (optional) </p>
</td></tr>
<tr><td><code id="jscores_+3A_ensembl">ensembl</code></td>
<td>
<p> A vector of Ensembl IDs (optional) </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Currently, <code>chip</code> can be <code>"hgu95av2"</code>, <code>"hgu133a"</code>, <code>"hgu133plus2"</code>, or <code>"u133x3p"</code>.  If no further arguments are specified, the scores for all probe sets on the chip are returned.
</p>
<p>If any of  <code>probeset</code>, <code>eg</code>, <code>symbol</code>, <code>alias</code>, or <code>ensembl</code> are specified, these are used to filter the resulting data frame in a logical OR sense.
</p>
<p>Details about the jetset algorithm are available in the vignette.
</p>


<h3>Value</h3>

<p>A data frame in which each row corresponds to a probe set, with 8 columns:
</p>
<table>
<tr><td><code>EntrezID</code></td>
<td>
<p>Entrez GeneID of the targeted gene (character).</p>
</td></tr>
<tr><td><code>nProbes</code></td>
<td>
<p>Number of probes in the probe set  (integer).</p>
</td></tr>
<tr><td><code>process</code></td>
<td>
<p>Processivity requirement  (integer).</p>
</td></tr>
<tr><td><code>specificity</code></td>
<td>
<p>Specificity score (numeric).</p>
</td></tr>
<tr><td><code>coverage</code></td>
<td>
<p>Coverage score (numeric).</p>
</td></tr>
<tr><td><code>robust</code></td>
<td>
<p>Robustness score (numeric).</p>
</td></tr>
<tr><td><code>overall</code></td>
<td>
<p>Overall score (numeric).</p>
</td></tr>
<tr><td><code>symbol</code></td>
<td>
<p>HUGO gene symbol (character).</p>
</td></tr>
</table>
<p>The rows are sorted by decreasing overall score.
</p>


<h3>References</h3>

<p>Qiyuan Li, Nicolai J. Birkbak, Balazs Gyorffy, Zoltan Szallasi and Aron C. Eklund. (2011) Jetset: selecting the optimal microarray probe set to represent a gene. BMC Bioinformatics. 12:474.
</p>


<h3>See Also</h3>

<p>The underlying data comes from (e.g.) <code><a href="#topic+scores.hgu95av2">scores.hgu95av2</a></code>,
with gene symbol lookups coming from <code><a href="org.Hs.eg.db.html#topic+org.Hs.egSYMBOL">org.Hs.egSYMBOL</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  genes &lt;- c('MKI67', 'CHD5', 'ESR1', 'FGF19', 'ERBB2', 'NoSuchGene')

  # This generates several informative warnings
  jscores('hgu133a', symbol = genes)

</code></pre>

<hr>
<h2 id='scores'>Data: Probe set quality scores</h2><span id='topic+scores.hgu95av2'></span><span id='topic+scores.hgu133a'></span><span id='topic+scores.hgu133plus2'></span><span id='topic+scores.u133x3p'></span>

<h3>Description</h3>

<p>This data set provides gene target and quality scores for each probe set on the 
corresponding Affymetrix gene expression microarrays. </p>


<h3>Usage</h3>

<pre><code class='language-R'>scores.hgu95av2
scores.hgu133a
scores.hgu133plus2
scores.u133x3p
</code></pre>


<h3>Format</h3>

<p>A data frame with each row corresponding to a probe set, with 4 columns:
</p>

<dl>
<dt>EntrezID</dt><dd><p>Entrez GeneID of the targeted gene (character).</p>
</dd>
<dt>process</dt><dd><p>Processivity requirement  (integer).</p>
</dd>
<dt>specificity</dt><dd><p>Specificity score (numeric).</p>
</dd>
<dt>coverage</dt><dd><p>Coverage score (numeric).</p>
</dd>
</dl>



<h3>Details</h3>

<p>If there is a relative majority (plurality) of the probes in a probe set that are specific 
for a single gene, this is defined as the <em>targeted gene</em>. If no such majority exists, 
the targeted gene is defined as <code>NA</code>, as are the following scores.
</p>
<p>The <em>processivity requirement</em> is the number of consecutive bases that
must be synthesized to generate a target that can be detected by the probe set.
</p>
<p>The <em>specificity score</em> is the fraction of the probes in a probe set 
that are likely to detect the targeted gene and unlikely to detect other genes.
</p>
<p>The <em>coverage score</em> is the fraction of the splice isoforms
belonging to the targeted gene that are detected by the probe set.
</p>
<p>The following two scores are not contained in this data, but are calculated from the above scores; to see them use <code><a href="#topic+jscores">jscores</a></code>.
</p>
<p>The <em>robustness score</em> quantifies robustness against transcript
degradation.  The robustness score uses the processivity requirement to estimate the 
signal intensity of a probe set, relative to the ideal case of perfect processivity.
</p>
<p>The <em>overall score</em> is the product of the specificity score, 
coverage score, and robustness score.
</p>
<p>All scores can range from 0 to 1.  A higher score indicates better (predicted) performance. 
</p>
<p>Details about the jetset algorithm are available in the vignette.
</p>


<h3>Note</h3>

<p>This data is also available in CSV format from
<a href="http://www.cbs.dtu.dk/biotools/jetset/">http://www.cbs.dtu.dk/biotools/jetset/</a>
</p>


<h3>Source</h3>

<p>Scores are calculated from BLASTN alignments between probe sequences
and Refseq transcript sequences, as described in the vignette and in the reference below.
</p>
<p>The Refseq human RNA was downloaded from NCBI on 2017-04-04. The lookups were based on <span class="pkg">org.Hs.eg.db</span> version 3.4.0.
</p>


<h3>References</h3>

<p>Qiyuan Li, Nicolai J. Birkbak, Balazs Gyorffy, Zoltan Szallasi and Aron C. Eklund. (2011) Jetset: selecting the optimal microarray probe set to represent a gene. BMC Bioinformatics. 12:474.
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+jscores">jscores</a></code> for a more convenient way to access this data</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ## Here is the EntrezID for the ESR1 gene
  id &lt;- "2099"
  
  ## Extract the scores for all probe sets detecting ESR1
  scores.hgu95av2[which(scores.hgu95av2$EntrezID == id), ]

  ## Compare to the recommended function 'jscores'
  jscores("hgu95av2", eg = "2099")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
