<!DOCTYPE html><html lang="en"><head><title>Help for package joint.Cox</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {joint.Cox}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#joint.Cox-package'>
<p>Joint Frailty-Copula Models for Tumour Progression and Death in Meta-Analysis</p></a></li>
<li><a href='#cmprskCox.reg'>
<p>The Competing Risks Version of Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression</p>
and Death for Meta-Analysis</a></li>
<li><a href='#condCox.reg'>
<p>Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression</p>
and Death for Meta-Analysis; A Conditional Copula Approach</a></li>
<li><a href='#dataOvarian'>
<p>Survival data of 1003 ovarian cancer patients from 4 independent studies.</p></a></li>
<li><a href='#dataOvarian1'>
<p>Data on time-to-recurrence and 158 gene expressions for 912 ovarian cancer patients from 4 independent studies.</p></a></li>
<li><a href='#dataOvarian2'>
<p>Data on time-to-death and 128 gene expressions for 912 ovarian cancer patients from 4 independent studies.</p></a></li>
<li><a href='#F.KM'><p>Prediction of death using the Kaplan-Meier estimator</p></a></li>
<li><a href='#F.prediction'><p>Dynamic prediction of death</p></a></li>
<li><a href='#F.window'><p>Dynamic prediction of death under the joint frailty-copula model</p></a></li>
<li><a href='#F.window.Weibull'><p>Dynamic prediction of death under the joint frailty-copula model (the Weibull baseline hazard functions)</p></a></li>
<li><a href='#F.windows'><p>Dynamic prediction of death under the joint frailty-copula model</p></a></li>
<li><a href='#F.windows.Weibull'><p>Dynamic prediction of death under the joint frailty-copula model (the Weibull baseline hazard functions)</p></a></li>
<li><a href='#I.spline'><p> I-spline basis function</p></a></li>
<li><a href='#jointCox.indep.reg'>
<p>Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression</p>
and Death for Meta-Analysis</a></li>
<li><a href='#jointCox.reg'>
<p>Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression</p>
and Death for Meta-Analysis</a></li>
<li><a href='#jointCox.Weibull.reg'>
<p>Weibull-based Likelihood Estimation under the Joint Cox Models Between Tumour Progression</p>
and Death for Meta-Analysis</a></li>
<li><a href='#M.spline'><p> M-spline basis function</p></a></li>
<li><a href='#splineCox.reg'><p>Fitting the Cox model for survival data using a penalized spline model</p></a></li>
<li><a href='#Weibull.simu'><p>Simulating data from the Weibull joint frailty-copula model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Joint Frailty-Copula Models for Tumour Progression and Death in
Meta-Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>3.16</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-2-4</td>
</tr>
<tr>
<td>Author:</td>
<td>Takeshi Emura</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Takeshi Emura &lt;takeshiemura@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fit survival data and perform dynamic prediction under joint frailty-copula models for tumour progression and death.
 Likelihood-based methods are employed for estimating model parameters, where the baseline hazard functions are modeled by the cubic M-spline or the Weibull model.
 The methods are applicable for meta-analytic data containing individual-patient information from several studies.
 Survival outcomes need information on both terminal event time (e.g., time-to-death) and non-terminal event time (e.g., time-to-tumour progression).
 Methodologies were published in
 Emura et al. (2017) &lt;<a href="https://doi.org/10.1177%2F0962280215604510">doi:10.1177/0962280215604510</a>&gt;, Emura et al. (2018) &lt;<a href="https://doi.org/10.1177%2F0962280216688032">doi:10.1177/0962280216688032</a>&gt;,
 Emura et al. (2020) &lt;<a href="https://doi.org/10.1177%2F0962280219892295">doi:10.1177/0962280219892295</a>&gt;, Shinohara et al. (2020) &lt;<a href="https://doi.org/10.1080%2F03610918.2020.1855449">doi:10.1080/03610918.2020.1855449</a>&gt;,
 Wu et al. (2020) &lt;<a href="https://doi.org/10.1007%2Fs00180-020-00977-1">doi:10.1007/s00180-020-00977-1</a>&gt;, and Emura et al. (2021) &lt;<a href="https://doi.org/10.1177%2F09622802211046390">doi:10.1177/09622802211046390</a>&gt;.
 See also the book of Emura et al. (2019) &lt;<a href="https://doi.org/10.1007%2F978-981-13-3516-7">doi:10.1007/978-981-13-3516-7</a>&gt;.
 Survival data from ovarian cancer patients are also available.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>survival</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-04 07:04:30 UTC; biouser</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-02-04 10:10:10 UTC</td>
</tr>
</table>
<hr>
<h2 id='joint.Cox-package'>
Joint Frailty-Copula Models for Tumour Progression and Death in Meta-Analysis
</h2><span id='topic+joint.Cox-package'></span><span id='topic+joint.Cox'></span>

<h3>Description</h3>

<p>Fit survival data and perform dynamic prediction under joint frailty-copula models for tumour progression and death.
Likelihood-based methods are employed for estimating model parameters, where the baseline hazard functions are modeled by the cubic M-spline or the Weibull model.
The methods are applicable for meta-analytic data containing individual-patient information from several studies.
Survival outcomes need information on both terminal event time (e.g., time-to-death) and non-terminal event time (e.g., time-to-tumour progression).
Methodologies were published in Emura et al. (2017), Emura et al. (2018), Emura et al. (2020), Wu et al. (2020), Shinohara et al. (2020),
and Emura et al. (2021).
See also the book of Emura et al. (2019).
Survival data from ovarian cancer patients are also available.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> joint.Cox</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 3.16</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2022-2-4</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Takeshi Emura
Maintainer: Takeshi Emura &lt;takeshiemura@gmail.com&gt;
</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6):2649-66
</p>
<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>
<p>Emura T, Matsui S, Rondeau V (2019), Survival Analysis with Correlated Endpoints, Joint Frailty-Copula Models, JSS Research Series in Statistics, Springer
</p>
<p>Emura T, Shih JH, Ha ID, Wilke RA (2020), Comparison of the marginal hazard model and the sub-distribution hazard model for competing risks under an assumed Copula,
Stat Methods Med Res 29(8):2307-27
</p>
<p>Emura T, Sofeu C, Rondeau V (2021), Conditional copula models for correlated survival endpoints:
individual patient data meta-analysis of randomized controlled trials,
Stat Methods Med Res 30(12):2634-50
</p>
<p>Shinohara S, Lin YH, Michimae H, Emura T (2020), Dynamic lifetime prediction using a Weibull-based bivariate failure time model:
a meta-analysis of individual-patient data, Comm Stat-Simul, DOI:10.1080/03610918.2020.1855449
</p>
<p>Wu BH, Michimae H, Emura T (2020), Meta-analysis of individual patient data with semi-competing risks under
the Weibull joint frailty-copula model, Comp Stat 35(4):1525-52
</p>

<hr>
<h2 id='cmprskCox.reg'>
The Competing Risks Version of Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression
and Death for Meta-Analysis
</h2><span id='topic+cmprskCox.reg'></span>

<h3>Description</h3>

<p>Perform regression analyses under a copula-based joint Cox proportional hazards model between tumour progression and death for meta-analysis,
which is proposed in Section 6.2 of Emura et al. (2017) and Section 5.1 of Emura et al. (2020).
This is the competing risks version of &quot;jointCox.reg&quot;.
To avoid the indentifiability problem, the copula parameter (theta) should be given by user, e.g., theta=2.
The method is applicable for meta-analysis combining several studies or for cluster survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cmprskCox.reg(t.event, event1, event2, Z1, Z2, group, theta, alpha = 1,
 kappa1 = c(seq(10, 1e+17, length = 30)), kappa2 = c(seq(10, 1e+17, length = 30)),
 LCV.plot = TRUE, Randomize_num = 10, Adj = 500,convergence.par=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cmprskCox.reg_+3A_t.event">t.event</code></td>
<td>
<p>a vector for event tims</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_event1">event1</code></td>
<td>
<p>a vector for event-type 1 indicators (=1 with event; =0 without event)</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_event2">event2</code></td>
<td>
<p>a vector for event-type 2 indicators (=1 with event; =0 without event)</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_z1">Z1</code></td>
<td>
<p>a matrix for covariates associated with event-type 1; ncol(Z1)=the number of covariates</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_z2">Z2</code></td>
<td>
<p>a matrix for covariates associated with event-type 2; ncol(Z2)=the number of covariates</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_group">group</code></td>
<td>
<p>a vector for a group identification number, like 1,2,3....</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_theta">theta</code></td>
<td>
<p>A copula parameter under the Clayton copula (theta &gt; 0)</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_alpha">alpha</code></td>
<td>
<p>A value related to the frailty (e.g., alpha=0 or =1); alpha=1 is default</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_kappa1">kappa1</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_kappa2">kappa2</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_lcv.plot">LCV.plot</code></td>
<td>
<p>Plot the LCV curves if &quot;TRUE&quot;</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_randomize_num">Randomize_num</code></td>
<td>
<p>The number of randomizations for the ititial p0</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_adj">Adj</code></td>
<td>
<p>Numerical adjustment to prevent overflow; Adj=500 is recommended</p>
</td></tr>
<tr><td><code id="cmprskCox.reg_+3A_convergence.par">convergence.par</code></td>
<td>
<p>If TRUE, the converged estimate, gradient, and Hessian matrix are given (log-transformed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We employ &quot;nlm&quot; routine to maximize the penalized likelihood function with the initial value described in Emura et al. (2015).
If &quot;nlm&quot; does not converge, then we randomize the initial value by adding uniform random variables (Hu and Emura, 2015).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>count</code></td>
<td>
<p>Count for event occurences</p>
</td></tr>
<tr><td><code>beta1</code></td>
<td>
<p>Regression coefficient for Z1</p>
</td></tr>
<tr><td><code>beta2</code></td>
<td>
<p>Regression coefficient for Z2</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>Frailty parameter (variance)</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>Copula parameter under the Clayton copula (fixed by user)</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau corresponding to the copula parameter</p>
</td></tr>
<tr><td><code>LCV1</code></td>
<td>
<p>Likelihood cross-validation for event-type 1</p>
</td></tr>
<tr><td><code>LCV2</code></td>
<td>
<p>Likelihood cross-validation for event-type 2</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>M-spline coefficients for event-type 1</p>
</td></tr>
<tr><td><code>h</code></td>
<td>
<p>M-spline coefficients for event-type 2</p>
</td></tr>
<tr><td><code>g_var</code></td>
<td>
<p>Variance of M-spline coefficients for event-type 1</p>
</td></tr>
<tr><td><code>h_var</code></td>
<td>
<p>Variance of M-spline coefficients for event-type 2</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence results for maximizing penalized likelihood</p>
</td></tr>
<tr><td><code>convergence.parameters</code></td>
<td>
<p>converged estimate, gradient, and Hessian matrix (log-transformed)</p>
</td></tr>
</table>


<h3>Error </h3>

<p>&quot;Error in integrate(func1, 0.001, 10, stop.on.error = FALSE):non-finite function value&quot;,
an error occurring when the penalized likelihood is maximizad by &quot;nlm&quot;. The error may frequently
occur during the iterations for maximizing the penalized likelihood, but is not crucial (can simply be ignored).</p>


<h3>Warning </h3>

<p>&quot;NA/Inf replaced by maximum positive value&quot;, an error occurring when the penalized likelihood
is maximizad by &quot;nlm&quot;. The error frequently occurs during the iterations for maximizing the penalized likelihood,
but is not crucial (can simply be ignored).</p>


<h3>Author(s)</h3>

<p>Takeshi Emura, Shih JH</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66
</p>
<p>Emura T, Shih JH, Ha ID, Wilke RA (2020), Comparison of the marginal hazard model and the sub-distribution hazard model for competing risks under an assumed copula,
Stat Methods Med Res, 29(8): 2307-27
</p>
<p>Hu YH, Emura T (2015), Maximum likelihood estimation for a special exponential family under random double-truncation,
Compu Stat 30 (4): 1199-1229
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataOvarian)
t.event=dataOvarian$t.event
t.death=dataOvarian$t.death
event=dataOvarian$event
death=dataOvarian$death
non.event=which(event==1 &amp; death==1 &amp; t.event==t.death)
non.death=which(event==1 &amp; death==1 &amp; t.event&lt;t.death)
event[non.event]=0 ## relapse before death
death[non.death]=0 ## death before relapse (tie is counted as death)
Z=as.matrix(dataOvarian$CXCL12)
group=dataOvarian$group
alpha_given=0
theta=2.35
kappa_grid=seq(10,1e+17,length = 30)

#set.seed(1)
#cmprskCox.reg(t.event=t.event,event1=event,event2=death,
#              Z1=Z,Z2=Z,group=group,theta=theta,alpha=alpha_given,
#              kappa1=kappa_grid,kappa2=kappa_grid,LCV.plot=TRUE,Adj=500)
</code></pre>

<hr>
<h2 id='condCox.reg'>
Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression 
and Death for Meta-Analysis; A Conditional Copula Approach
</h2><span id='topic+condCox.reg'></span>

<h3>Description</h3>

<p>An extension of the function &quot;joint.Cox(.)&quot; by regression on a conditional copula.
Perform joint regression analyses under a copula-based joint Cox proportional hazards model between tumour progression and death for meta-analysis,
which is proposed by Emura et al. (2021).
The method extends the joint frailty copula model of Emura et al. (2017) by adding a regression function
on a copula parameter.
The method is applicable for meta-analysis combining several studies or for cluster survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>condCox.reg(t.event, event, t.death, death, Z1, Z2, Z12, group, alpha = 1,
 kappa1 = c(seq(10, 1e+17, length = 30)),kappa2 = c(seq(10, 1e+17, length = 30)),
 LCV.plot = TRUE, Randomize_num = 10, u.min = 0.001, u.max = 10,
 Adj = 500,convergence.par=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="condCox.reg_+3A_t.event">t.event</code></td>
<td>
<p>a vector for time-to-tumour progression (TTP)</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_event">event</code></td>
<td>
<p>a vector for progression indicator (=1 if progression; =0 if not progression)</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_t.death">t.death</code></td>
<td>
<p>a vector for overall survival (OS), i.e., time-to-death</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_death">death</code></td>
<td>
<p>a vector for death indicator(=1 if death; =0 if not death)</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_z1">Z1</code></td>
<td>
<p>a matrix for covariates associated with TTP; ncol(Z1)=the number of covariates</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_z2">Z2</code></td>
<td>
<p>a matrix for covariates associated with OS; ncol(Z2)=the number of covariates</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_z12">Z12</code></td>
<td>
<p>a matrix for covariates associated with copula; ncol(Z12)=the number of covariates</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_group">group</code></td>
<td>
<p>a vector for group identification numbers, like 1,2,3....</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_alpha">alpha</code></td>
<td>
<p>A value related to the frailty (e.g., alpha=0 or =1); alpha=1 is default</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_kappa1">kappa1</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_kappa2">kappa2</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_lcv.plot">LCV.plot</code></td>
<td>
<p>Plot the LCV curves if &quot;TRUE&quot;</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_randomize_num">Randomize_num</code></td>
<td>
<p>The number of randomizations for the ititial p0</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_u.min">u.min</code></td>
<td>
<p>the lower bound of the numerical integration for the frailty term</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_u.max">u.max</code></td>
<td>
<p>the upper bound of the numerical integration for the frailty term</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_adj">Adj</code></td>
<td>
<p>Numerical adjustment to prevent overflow; Adj=500 is recommended</p>
</td></tr>
<tr><td><code id="condCox.reg_+3A_convergence.par">convergence.par</code></td>
<td>
<p>If TRUE, the converged estimate, gradient, and Hessian matrix are given (log-transformed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We employ &quot;nlm&quot; routine to maximize the penalized likelihood function with the initial value described in Emura et al. (2017).
If &quot;nlm&quot; does not converge, then we randomize the initial value by adding uniform random variables (Hu and Emura, 2015).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>count</code></td>
<td>
<p>Count for event occurences</p>
</td></tr>
<tr><td><code>beta1</code></td>
<td>
<p>Regression coefficient for Z1</p>
</td></tr>
<tr><td><code>beta2</code></td>
<td>
<p>Regression coefficient for Z2</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>Frailty parameter (variance)</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>Baseline copula parameter under the Clayton copula</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau corresponding to the baseline copula parameter</p>
</td></tr>
<tr><td><code>beta12</code></td>
<td>
<p>Regression coefficient for a copula parameter</p>
</td></tr>
<tr><td><code>LCV1</code></td>
<td>
<p>Likelihood cross-validation for TTP</p>
</td></tr>
<tr><td><code>LCV2</code></td>
<td>
<p>Likelihood cross-validation for OS</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>M-spline coefficients for TTP</p>
</td></tr>
<tr><td><code>h</code></td>
<td>
<p>M-spline coefficients for OS</p>
</td></tr>
<tr><td><code>g_var</code></td>
<td>
<p>Variance of M-spline coefficients for TTP</p>
</td></tr>
<tr><td><code>h_var</code></td>
<td>
<p>Variance of M-spline coefficients for OS</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence results for maximizing penalized likelihood</p>
</td></tr>
<tr><td><code>convergence.parameters</code></td>
<td>
<p>converged estimate, gradient, and Hessian matrix (log-transformed)</p>
</td></tr>
</table>


<h3>Error </h3>

<p>&quot;Error in integrate(func1, 0.001, 10, stop.on.error = FALSE):non-finite function value&quot;,
an error occurring when the penalized likelihood is maximizad by &quot;nlm&quot;. The error may frequently
occur during the iterations for maximizing the penalized likelihood, but is not crucial (can simply be ignored).</p>


<h3>Warning </h3>

<p>&quot;NA/Inf replaced by maximum positive value&quot;, an error occurring when the penalized likelihood
is maximizad by &quot;nlm&quot;. The error frequently occurs during the iterations for maximizing the penalized likelihood,
but is not crucial (can simply be ignored).</p>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Sofeu C, Rondeau V (2021), Conditional copula models for correlated survival endpoints:
individual patient data meta-analysis of randomized controlled trials,
Stat Methods Med Res 30(12):2634-50
</p>
<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6):2649-66
</p>
<p>Hu YH, Emura T (2015), Maximum likelihood estimation for a special exponential family under random double-truncation,
Computational Stat 30(4):1199-1229
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data=Weibull.simu(G=10,N=50,scale1=1.5,scale2=1,beta1=-0.2,beta2=-0.2,beta12=0.5,
                    eta=0.5,copula="Clayton",theta=2,alpha=1,
                    C.max=5,Z.dist=rbinom,size=1,prob=0.5)
t.event=data$t.event
event=data$event
t.death=data$t.death
death=data$death
group=data$group
Z1=as.matrix(data$Z)
Z2=Z12=Z1
kappa=seq(1,10000,length=50)

#condCox.reg(t.event=t.event,event=event,t.death=t.death,death=death,
#             Z1=Z1,Z2=Z2,Z12=Z12,group=group,alpha=1,
#             kappa1=kappa,kappa2=kappa,Randomize_num=1,LCV.plot=FALSE,u.max=20)

</code></pre>

<hr>
<h2 id='dataOvarian'>
Survival data of 1003 ovarian cancer patients from 4 independent studies.
</h2><span id='topic+dataOvarian'></span>

<h3>Description</h3>

<p>The data consist of 1003 surgically treated ovarian cancer patients from four studies (N1=110, N2=58, N3=278, N4=557).
Survival outcomes are given to study if the CXCL12 gene expression is a prognostic factor in ovarian cancer.
The dataset was used in Emura et al. (2017), which is a subset of the curated ovarian data of Ganzfried et al (2013).
We prepared the dataset by using &quot;patientselection.config&quot; in &quot;Curated ovarian data&quot; around May 2015 in the analysis of Emura et al. (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dataOvarian")</code></pre>


<h3>Format</h3>

<p>A data frame with 1003 observations on the following 6 variables.
</p>

<dl>
<dt><code>t.event</code></dt><dd><p>: time to event (days from surgery to tumour recurrence)</p>
</dd>
<dt><code>event</code></dt><dd><p>: event indicator (1=recurrence, 0=no recurrence)</p>
</dd>
<dt><code>t.death</code></dt><dd><p>: time to death (days from surgery to death due to any cause)</p>
</dd>
<dt><code>death</code></dt><dd><p>: death indicator (1=death, 0=alive)</p>
</dd>
<dt><code>group</code></dt><dd><p>: study ID; group=4, 8, 11, or 14; see the details below</p>
</dd>
<dt><code>CXCL12</code></dt><dd><p>: CXCL12 gene expression</p>
</dd>
</dl>



<h3>Details</h3>

<p>The data include individual-patient information on 1003 patients from 4 studies (group=4, 8, 11, and 14). The numbers 4, 8, 11 and 14 corresponds to the study IDs from the original data of Ganzfried et al. (2013).
&quot;group=4&quot; corresponds to 110 Japanese patients from the study of Yoshihara et al. (2010) (GEO accession number: GSE17260). Other groupds are the studies of GSE30161 (58 patients), GSE9891 (278 patients), and TCGA (557 patients).
</p>


<h3>Source</h3>

<p>Ganzfried BF et al. (2013), Curated ovarian data: clinically annotated data for the ovarian cancer transcriptome, Database, Article ID bat013, doi:10.1093/database/bat013.
</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66
</p>
<p>Ganzfried BF et al. (2013), Curated ovarian data: clinically annotated data for the ovarian cancer transcriptome, Database, Article ID bat013, doi:10.1093/database/bat013.
</p>
<p>Yoshihara K et al. (2010) Gene expression profile for predicting survival in advanced-stage serous ovarian cancer across two independent datasets. PLoS One 12;5(3):e9615
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataOvarian)
study4=dataOvarian[dataOvarian$group==4,] # extract one study
study4
</code></pre>

<hr>
<h2 id='dataOvarian1'>
Data on time-to-recurrence and 158 gene expressions for 912 ovarian cancer patients from 4 independent studies. 
</h2><span id='topic+dataOvarian1'></span>

<h3>Description</h3>

<p>Meta-analytic data containing 158 gene expressions and time-to-relapse information for ovarian cancer patients.
The data include time-to-recurrence, residual tumour size (&gt;=1cm&gt; vs. &lt;1cm), and associated 158 gene expressions.
The dataset is a subset of the curated ovarian data of Ganzfried et al (2013).
We prepared the dataset by using &quot;patientselection.config&quot; in &quot;Curated ovarian data&quot; around October 2016. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dataOvarian1")</code></pre>


<h3>Format</h3>

<p>A data frame with 912 observations on the following 162 variables.
</p>

<dl>
<dt><code>t.event</code></dt><dd><p>: time-to-recurrence in days</p>
</dd>
<dt><code>event</code></dt><dd><p>: event indicator (1=recurrence, 0=no recurrence)</p>
</dd>
<dt><code>group</code></dt><dd><p>: study ID; group=4, 9, 12, or 16</p>
</dd>
<dt><code>debulk</code></dt><dd><p>: residual tumour size (&gt;=1cm&gt; vs. &lt;1cm)</p>
</dd>
<dt><code>ABI3BP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ADAM12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ADORA3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ANKRD27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>AP2M1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>AP3S1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ARHGAP28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ARHGAP29</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ARTN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ASAP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B4GALT5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BCAP31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BRD4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>C1QTNF3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CALD1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCNE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCNL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CDC42</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CDV3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CEBPB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLIC4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL10A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL11A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL16A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL3A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL5A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL5A2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COMP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CRISPLD2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CRYAB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CSE1L</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CTSK</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CXCL12</code></dt><dd><p>a numeric vector of gene expressions.
The CXCL12 gene expression is a predictive biomarker of survival in ovarian cancer (Popple et al. 2012).
It has been known that CXCL12 promotes tumour growth, participates in tumour metastasis,
and suppresses tumour immunity (Kryczek et al. 2007).
The statistical significance of the CXCL12 expression on survival is first examined by Popple et al. (2012), and
is further confirmed by Ganzfried et al. (2013) based on the meta-analysis of 14 independent studies.
A meta-analysis using a joint model further confirmed that the expression of CXCL12 gene is predictive of
both cancer relapse and death (Emura et al. 2017; 2018).</p>
</dd>
<dt><code>CYR61</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DCUN1D1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DDX27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DIAPH3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DNAJB4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DNAJC13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DNAJC8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DPYSL3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DVL3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EFNB2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EIF3K</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ELK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ENPP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EPYC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FABP4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FAM69A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FAP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FERMT2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FGF1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FN1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FOSL2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FSTL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GABRG3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GAS1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GFRA1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GFRA3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GJC1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GLIPR1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GPATCH1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HLTF</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HP1BP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HSD17B6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>INHBA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ITGB1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>JUN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KIAA0226</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KIAA0355</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KIAA1598</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KIN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KLHL25</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KPNA6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KRT7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KRTAP5.8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>L2HGDH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LGALS1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LOX</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LPP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LUM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LUZP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MAP7D1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MAPRE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MCL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MEOX2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>METTL9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MFN1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MICAL2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MMP12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MRPS22</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MXD1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MXRA8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>N4BP2L2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NCOA3</code></dt><dd><p>a numeric vector of gene expressions.
The NCOA3 gene encodes a nuclear receptor coactivator, and amplification of the gene
occurs in breast and ovarian cancers (Anzick et al. 1997).
The overexpression of NCOA3 is associated with tumor size (Spears et al. 2012) and
tamoxifen resistance (Osborne et al. 2003), which are involved in the progression.
Yoshida et al. (2005) reported that NCOA3 could contribute to ovarian cancer progression by promoting cell migration.
In Emura et al. (2018), the overexpression of the gene was highly associated with time-to-relapse
(Coefficient=0.194, P-value&lt;0.00001) and time-to-death (Coefficient=0.237, P-value&lt;0.00001).
This result is consistent with the function of these reports.</p>
</dd>
<dt><code>NDRG3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NINJ1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NNMT</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NOTCH2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NPY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NTM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NUAK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>OAT</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>OLFML2B</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PARD3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PCYT1A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PDE1A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PDGFD</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PDPN</code></dt><dd><p>a numeric vector of gene expressions. The PDPN gene encodes the podoplanin protein.
It is reported that cancer cells with higher PDPN expression have higher malignant potential due to enhanced platelet aggregation,
which promotes alteration of metastasis, cell motility, and epithelial-mesenchymal transition (Shindo et al. 2013).
Zhang et al. (2011) reported that overexpression of PDPN in fibroblasts is significantly associated with a poor prognosis in ovarian carcinoma.
In Emura et al. (2018), the overexpression of the gene was highly associated with time-to-relapse
(Coefficient=0.222, P-value&lt;0.00001) and time-to-death (Coefficient=0.161, P-value&lt;0.0001).</p>
</dd>
<dt><code>PGRMC1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PLAU</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PLOD2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PLSCR4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>POSTN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PPIC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PRDM2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PSMC4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RAB22A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RAB31</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RAB32</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RARRES1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RPS16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SERPINE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SGK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SH3PXD2A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SKIL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SLC12A8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SPARC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SPHK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STAU1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SULF1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SUPT5H</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TAGLN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TBCB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TEAD1</code></dt><dd><p>a numeric vector of gene expressions.
TEAD1 encodes a ubiquitous transcriptional enhancer factor that is a member of the TEA/ATTS domain family.
It is reported that the protein level of TEAD1 was associated with poor prognosis in prostate cancer patients (Knight et al. 2008).
In Emura et al. (2018), the overexpression of the gene was highly associated with
time-to-relapse (Coefficient=0.195, P-value&lt;0.00001) and time-to-death (Coefficient=0.223, P-value&lt;0.00001).</p>
</dd>
<dt><code>TESK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TGM5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>THEMIS2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TIMP2</code></dt><dd><p>a numeric vector of gene expressions. TIMP2 is a member of the TIMP gene family.
The proteins encoded by this gene family are natural inhibitors of the matrix metalloproteinases (MMPs).
MMPs and their inhibitors (TIMP gene family) play an important regulatory role in the homeostasis of
the extracellular matrix (Halon et al. 2012). In addition to inhibitors of MMPs,
TIMP2 has additional functions that are associated with cell proliferation and survival (Bourboulia et al., 2011).
In Emura et al. (2018), the overexpression of the gene was highly associated with time-to-relapse (Coefficient=0.235, P-value&lt;0.00001).</p>
</dd>
<dt><code>TIMP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TJP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TP73.AS1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TPM2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TPM4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TSC22D2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TUBB2A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TUBB6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TUFT1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>URI1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>USP48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>VCAN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>VSIG4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>YWHAB</code></dt><dd><p>a numeric vector of gene expressions.
YWHAB encodes a protein belonging to the 14-3-3 family of proteins,
members of which mediate signal transduction by binding to phosphoserine-containing proteins.
It is reported that the protein of YWHAB can regulate cell survival, proliferation, and motility (Tzivion 2006).
Actually, it is reported that overexpression of this gene promotes tumor progression and was associated with
extrahepatic metastasis and worse survival in hepatocellular carcinoma (Liu et al. 2011).
In Emura et al. (2018), the overexpression of the gene was highly associated with
time-to-relapse (Coefficient=0.169, P-value&lt;0.0001) and time-to-death (Coefficient=0.263, P-value&lt;0.00001)</p>
</dd>
<dt><code>ZFP36</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ZFP36L2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ZMYM1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ZNF148</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ZNF79</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>4 studies are combined (group=4, 9, 12, and 16). The numbers 4, 9, 12 and 16 corresponds to the IDs 
from the original data of Ganzfried et al. (2013).</p>


<h3>Source</h3>

<p>Ganzfried BF et al. (2013), Curated ovarian data: clinically annotated data for the ovarian cancer transcriptome, Database, Article ID bat013.
</p>


<h3>References</h3>

<p>Bourboulia D, et al. (2011), Endogenous angiogenesis inhibitor blocks tumor growth via direct and indirect effects on tumor microenvironment. Am J Pathol 179:2589-600
</p>
<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6):2649-66
</p>
<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>
<p>Ganzfried BF et al. (2013), Curated ovarian data: clinically annotated data for the ovarian cancer transcriptome, Database, Article ID bat013.
</p>
<p>Halon A, et al. (2012), Enhanced immunoreactivity of TIMP-2 in the stromal compartment of tumor as a marker of favorable prognosis in ovarian cancer patients.
J Histochem Cytochem 60:491-501
</p>
<p>Knight JF, et al. (2008), TEAD1 and c-Cbl are novel prostate basal cell markers that correlate with poor clinical outcome in prostate cancer. Br J Cancer 99:1849-58
</p>
<p>Kryczek I, et al. (2007), Stroma-derived factor (SDF-1/CXCL12) and human tumor pathogenesis. Am J Physiol 292:987-95
</p>
<p>Liu TA, et al. (2011), Increased expression of 14-3-3beta promotes tumor progression and predicts extrahepatic metastasis and worse survival
in hepatocellular carcinoma. Am J Pathol 179:2698-708
</p>
<p>Osborne CK, et al. (2003), Role of the estrogen receptor coactivator AIB1 (SRC-3) and HER-2/neu in tamoxifen resistance in breast cancer. J Natl Cancer Inst 95:353-61
</p>
<p>Popple A, et al. (2012), The chemokine, CXCL12, is an independent predictor of poor survival in ovarian cancer. Br J Cancer 106:1306-13
</p>
<p>Shindo K, et al. (2013), Podoplanin expression in cancer-associated fibroblasts enhances tumor progression of invasive ductal carcinoma of the pancreas. Mol Cancer 12:168
</p>
<p>Tzivion G, et al. (2006), 14-3-3 proteins as potential oncogenes. Semin Cancer Biol 16:203-13
</p>
<p>Yoshida H, et al. (2005), Steroid receptor coactivator-3, a homolog of Taiman that controls cell migration in the Drosophila ovary,
regulates migration of human ovarian cancer cells. Mol Cell Endocrinol 245:77-85
</p>
<p>Zhang Y, et al. (2011), Ovarian cancer-associated fibroblasts contribute to epithelial ovarian carcinoma metastasis by promoting angiogenesis,
lymphangiogenesis and tumor cell invasion. Cancer Lett 303:47-55
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataOvarian1)
######## univariate Cox ##########
t.event=dataOvarian1$t.event
event=dataOvarian1$event
X.mat=dataOvarian1[,-c(1,2,3,4)] ## gene expression
Symbol=colnames(dataOvarian1)[-c(1,2,3,4)] ## gene symbol

p=ncol(X.mat)
P_value=coef=NULL
for(j in 1:p){
  res=summary(coxph(Surv(t.event,event)~X.mat[,j]))$coefficients
  P_value=c(P_value,res[5])
  coef=c(coef,res[1])
}
data.frame( gene=Symbol[order(P_value)], P=P_value[order(P_value)], 
coef=round(coef[order(P_value)],3) )
</code></pre>

<hr>
<h2 id='dataOvarian2'>
Data on time-to-death and 128 gene expressions for 912 ovarian cancer patients from 4 independent studies.
</h2><span id='topic+dataOvarian2'></span>

<h3>Description</h3>

<p>Meta-analytic data containing 128 gene expressions and time-to-death information for ovarian cancer patients.
The data include time-to-death, residual tumour size (&gt;=1cm&gt; vs. &lt;1cm), and associated 128 gene expressions.
The dataset is a subset of the curated ovarian data of Ganzfried et al (2013).
We prepared the dataset by using &quot;patientselection.config&quot; in &quot;Curated ovarian data&quot; around October 2016. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dataOvarian2")</code></pre>


<h3>Format</h3>

<p>A data frame with 912 observations on the following 132 variables.
</p>

<dl>
<dt><code>t.death</code></dt><dd><p>: time to death in days</p>
</dd>
<dt><code>death</code></dt><dd><p>: death indicator (1=death, 0=alive)</p>
</dd>
<dt><code>group</code></dt><dd><p>: study ID; group=4, 9, 12, or 16</p>
</dd>
<dt><code>debulk</code></dt><dd><p>: residual tumour size (&gt;=1cm&gt; vs. &lt;1cm)</p>
</dd>
<dt><code>ANKRD27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>AP3S1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>APMAP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ARHGAP28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ASAP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ASAP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ASB7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>B4GALT5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>BYSL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>C1QTNF3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CASP8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CCL18</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CD79A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CDK19</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CLIC4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL11A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL16A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL3A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL5A1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COL5A2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COMP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>COX7A2P2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CPNE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CRISPLD2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CRYAB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CTNNBL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CXCL12</code></dt><dd><p>a numeric vector of gene expressions.
The CXCL12 gene expression is a predictive biomarker of survival in ovarian cancer (Popple et al. 2012).
It has been known that CXCL12 promotes tumour growth, participates in tumour metastasis,
and suppresses tumour immunity (Kryczek et al. 2007).
The statistical significance of the CXCL12 expression on survival is first examined by Popple et al. (2012), and
is further confirmed by Ganzfried et al. (2013) based on the meta-analysis of 14 independent studies.
A meta-analysis using a joint model further confirmed that the expression of CXCL12 gene is predictive of
both cancer relapse and death (Emura et al. 2017; 2018)</p>
</dd>
<dt><code>CXCL9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CYBRD1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CYR61</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>CYTH3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DDX27</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DLGAP4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DNAJC13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>DYNLRB1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EFNB2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EIF3K</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ELN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>EMP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ENPP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FABP4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FAP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FBL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FGF1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FOXN3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>FSTL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GABRG3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GAS1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GFRA1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GJC1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GPATCH1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>GZMB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HLA.DOB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HOXA5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HP1BP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>HSD17B6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>IL2RG</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>INHBA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ITGB1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ITPKC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>JAM2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>JUN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KCNH4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KDELC1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KIAA0355</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>KIN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LEP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LOX</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LPL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LSM14A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LUM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>LUZP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MAPRE1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MCL1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MEOX2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>MMP12</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>N4BP2L2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NCOA3</code></dt><dd><p>a numeric vector of gene expressions.
The NCOA3 gene encodes a nuclear receptor coactivator, and amplification of the gene
occurs in breast and ovarian cancers (Anzick et al. 1997).
The overexpression of NCOA3 is associated with tumor size (Spears et al. 2012) and
tamoxifen resistance (Osborne et al. 2003), which are involved in the progression.
Yoshida et al. (2005) reported that NCOA3 could contribute to ovarian cancer progression by promoting cell migration.
In Emura et al. (2018), the overexpression of the gene was highly associated with time-to-relapse
(Coefficient=0.194, P-value&lt;0.00001) and time-to-death (Coefficient=0.237, P-value&lt;0.00001).
This result is consistent with the function of these reports.</p>
</dd>
<dt><code>NCOA6</code></dt><dd><p>a numeric vector of gene expressions</p>
</dd>
<dt><code>NOTCH2NL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NR1H3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>NUAK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>OAT</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>OMD</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PAK4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PCDH9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PDP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PDPN</code></dt><dd><p>a numeric vector of gene expressions. The PDPN gene encodes the podoplanin protein.
It is reported that cancer cells with higher PDPN expression have higher malignant potential due to enhanced platelet aggregation,
which promotes alteration of metastasis, cell motility, and epithelial-mesenchymal transition (Shindo et al. 2013).
Zhang et al. (2011) reported that overexpression of PDPN in fibroblasts is significantly associated with a poor prognosis in ovarian carcinoma.
In Emura et al. (2018), the overexpression of the gene was highly associated with time-to-relapse
(Coefficient=0.222, P-value&lt;0.00001) and time-to-death (Coefficient=0.161, P-value&lt;0.0001).</p>
</dd>
<dt><code>PHF20</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PLXNA1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PSMC4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>PSMD8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RAB13</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RAI14</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RARRES1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RBM39</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RECQL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RIN2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RND3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RPS16</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SACS</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SH3PXD2A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SKI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SLAMF7</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SLC37A4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SMG5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SOCS5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SPARC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SSR4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>STAU1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>SUPT5H</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TBCB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TBCC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TEAD1</code></dt><dd><p>a numeric vector of gene expressions.
TEAD1 encodes a ubiquitous transcriptional enhancer factor that is a member of the TEA/ATTS domain family.
It is reported that the protein level of TEAD1 was associated with poor prognosis in prostate cancer patients (Knight et al. 2008).
In Emura et al. (2018), the overexpression of the gene was highly associated with
time-to-relapse (Coefficient=0.195, P-value&lt;0.00001) and time-to-death (Coefficient=0.223, P-value&lt;0.00001).</p>
</dd>
<dt><code>TESK1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TIMP3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TJP1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TP53BP2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TSPAN9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TTI1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TUBB2A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>TUBB6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>URI1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>USP48</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>YWHAB</code></dt><dd><p>a numeric vector of gene expressions.
YWHAB encodes a protein belonging to the 14-3-3 family of proteins,
members of which mediate signal transduction by binding to phosphoserine-containing proteins.
It is reported that the protein of YWHAB can regulate cell survival, proliferation, and motility (Tzivion 2006).
Actually, it is reported that overexpression of this gene promotes tumor progression and was associated with
extrahepatic metastasis and worse survival in hepatocellular carcinoma (Liu et al. 2011).
In Emura et al. (2018), the overexpression of the gene was highly associated with
time-to-relapse (Coefficient=0.169, P-value&lt;0.0001) and time-to-death (Coefficient=0.263, P-value&lt;0.00001). </p>
</dd>
<dt><code>ZFP36</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ZFP36L2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>ZNF148</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>4 studies are combined (group=4, 9, 12, and 16). The numbers 4, 9, 12 and 16 corresponds to the IDs 
from the original data of Ganzfried et al. (2013).</p>


<h3>Source</h3>

<p>Ganzfried BF et al. (2013), Curated ovarian data: clinically annotated data for the ovarian cancer transcriptome, Database, Article ID bat013
</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6):2649-66
</p>
<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>
<p>Ganzfried BF, et al. (2013), Curated ovarian data: clinically annotated data for the ovarian cancer transcriptome, Database, Article ID bat013.
</p>
<p>Knight JF, et al. (2008), TEAD1 and c-Cbl are novel prostate basal cell markers that correlate with poor clinical outcome in prostate cancer. Br J Cancer 99:1849-58
</p>
<p>Kryczek I, et al. (2007), Stroma-derived factor (SDF-1/CXCL12) and human tumor pathogenesis. Am J Physiol 292:987-95
</p>
<p>Liu TA, et al. (2011), Increased expression of 14-3-3beta promotes tumor progression and predicts extrahepatic metastasis and worse survival
in hepatocellular carcinoma. Am J Pathol 179:2698-708
</p>
<p>Osborne CK, et al. (2003), Role of the estrogen receptor coactivator AIB1 (SRC-3) and HER-2/neu in tamoxifen resistance in breast cancer. J Natl Cancer Inst 95:353-61
</p>
<p>Popple A, et al. (2012), The chemokine, CXCL12, is an independent predictor of poor survival in ovarian cancer. Br J Cancer 106:1306-13
</p>
<p>Shindo K, et al. (2013), Podoplanin expression in cancer-associated fibroblasts enhances tumor progression of invasive ductal carcinoma of the pancreas. Mol Cancer 12:168
</p>
<p>Tzivion G, et al. (2006), 14-3-3 proteins as potential oncogenes. Semin Cancer Biol 16:203-13
</p>
<p>Yoshida H, et al. (2005), Steroid receptor coactivator-3, a homolog of Taiman that controls cell migration in the Drosophila ovary,
regulates migration of human ovarian cancer cells. Mol Cell Endocrinol 245:77-85
</p>
<p>Zhang Y, et al. (2011), Ovarian cancer-associated fibroblasts contribute to epithelial ovarian carcinoma metastasis by promoting angiogenesis,
lymphangiogenesis and tumor cell invasion. Cancer Lett 303:47-55
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataOvarian2)
######## univariate Cox ##########
t.death=dataOvarian2$t.death
death=dataOvarian2$death
X.mat=dataOvarian2[,-c(1,2,3,4)] ## gene expression
Symbol=colnames(dataOvarian2)[-c(1,2,3,4)] ## gene symbol

p=ncol(X.mat)
P_value=coef=NULL
for(j in 1:p){
  res=summary(coxph(Surv(t.death,death)~X.mat[,j]))$coefficients
  P_value=c(P_value,res[5])
  coef=c(coef,res[1])
}
data.frame( gene=Symbol[order(P_value)], P=P_value[order(P_value)], 
coef=round(coef[order(P_value)],3) )
</code></pre>

<hr>
<h2 id='F.KM'>Prediction of death using the Kaplan-Meier estimator</h2><span id='topic+F.KM'></span>

<h3>Description</h3>

<p>Dynamic prediction of death using using the Kaplan-Meier estimator. 
Probability of death between t and t+w is calculated. The prediction probability is F(t,t+w)=1-S(t+w)/S(t), where S is the Kaplan-Meier estimator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F.KM(time, widths, t.death, death)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="F.KM_+3A_time">time</code></td>
<td>
<p>prediction time (=t)</p>
</td></tr>
<tr><td><code id="F.KM_+3A_widths">widths</code></td>
<td>
<p>length of window (=w)</p>
</td></tr>
<tr><td><code id="F.KM_+3A_t.death">t.death</code></td>
<td>
<p>a vector object for overall survival (OS), i.e., time-to-death</p>
</td></tr>
<tr><td><code id="F.KM_+3A_death">death</code></td>
<td>
<p>a vector object for death indicator(=1 if death; =0 if not death)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Prediction probability of death is calculated without covariates.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>time</code></td>
<td>
<p>t</p>
</td></tr>
<tr><td><code>widths</code></td>
<td>
<p>w</p>
</td></tr>
<tr><td><code>F</code></td>
<td>
<p>F(t,t+w)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
time=1
widths=c(0,0.5,1,1.5,2)
t.death=c(0.5,1,1.5,2,2.5,3)
death=c(1,1,1,1,1,1)
F.KM(time=time,width=widths,t.death=t.death,death=death)
</code></pre>

<hr>
<h2 id='F.prediction'>Dynamic prediction of death</h2><span id='topic+F.prediction'></span>

<h3>Description</h3>

<p>Dynamic prediction of death using a joint frailty-copula model. 
Probability of death between t and t+w is calculated given a tumour progression time X and covariates Z1 and Z2. 
If X&lt;=t, the prediction probability is F(t,t+w|X=x, Z1, Z2). If X&gt;t, the prediction probability is F(t,t+w|X&gt;t, Z1, Z2). 
This function is a simpler version of F.windows. The guide for using this function shall be explained by Emura et al. (2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F.prediction(time, widths, X, Z1, Z2, beta1, beta2, eta, theta, alpha,
 g, h, xi1, xi3, Fplot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="F.prediction_+3A_time">time</code></td>
<td>
<p>prediction time (=t)</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_widths">widths</code></td>
<td>
<p>length of window (=w)</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_x">X</code></td>
<td>
<p>time of tumour progression; if tumour progression does not occur before time t, one can set an arbitrary value X greater than t</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_z1">Z1</code></td>
<td>
<p>a vector of covariates for progression</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_z2">Z2</code></td>
<td>
<p>a vector of covariates for death</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_beta1">beta1</code></td>
<td>
<p>a vector of regression coefficients for progression</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_beta2">beta2</code></td>
<td>
<p>a vector of regression coefficients for death</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_eta">eta</code></td>
<td>
<p>frailty variance</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_theta">theta</code></td>
<td>
<p>copula parameter</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_alpha">alpha</code></td>
<td>
<p>parameter related to frailty; usually alpha=1</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_g">g</code></td>
<td>
<p>parameters related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_h">h</code></td>
<td>
<p>parameters related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_xi1">xi1</code></td>
<td>
<p>lower bound for time-to-event</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_xi3">xi3</code></td>
<td>
<p>upper bound for time-to-death</p>
</td></tr>
<tr><td><code id="F.prediction_+3A_fplot">Fplot</code></td>
<td>
<p>if FALSE, the plot is not shown</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted probability of death is calculated given the event status (X&lt;=t or X&gt;t) and covariates (Z1 and Z2).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>time</code></td>
<td>
<p>t</p>
</td></tr>
<tr><td><code>widths</code></td>
<td>
<p>w</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X</p>
</td></tr>
<tr><td><code>F</code></td>
<td>
<p>F(t,t+w|X=x, Z1, Z2) or F(t,t+w|X&gt;t, Z1, Z2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>
<p>Emura T, Michimae H, Matsui S (2019-), A clinician's guide for dynamic risk prediction of death using an R package joint.Cox, submitted for publication.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w=c(0,0.5,1,1.5,2)
par(mfrow=c(1,2))
F.prediction(time=1,X=0.8,widths=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
         alpha=1,g=rep(1,5),h=rep(1,5),xi1=0,xi3=3)
F.prediction(time=1,X=1.5,widths=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
         alpha=1,g=rep(1,5),h=rep(1,5),xi1=0,xi3=3)
</code></pre>

<hr>
<h2 id='F.window'>Dynamic prediction of death under the joint frailty-copula model</h2><span id='topic+F.window'></span>

<h3>Description</h3>

<p>Dynamic prediction of death using a joint frailty-copula model. 
Probability of death between t and t+w is calculated given a tumour progression time X and covariates Z1 and Z2. 
If X&lt;=t, the prediction probability is F(t,t+w|X=x, Z1, Z2). If X&gt;t, the prediction probability is F(t,t+w|X&gt;t, Z1, Z2). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F.window(time, width, X, Z1, Z2, beta1, beta2, eta, theta, alpha,
 g, h, xi1, xi3, Fplot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="F.window_+3A_time">time</code></td>
<td>
<p>prediction time (=t)</p>
</td></tr>
<tr><td><code id="F.window_+3A_width">width</code></td>
<td>
<p>length of window (=w)</p>
</td></tr>
<tr><td><code id="F.window_+3A_x">X</code></td>
<td>
<p>time of tumour progression &lt; time</p>
</td></tr>
<tr><td><code id="F.window_+3A_z1">Z1</code></td>
<td>
<p>a vector of covariates for progression</p>
</td></tr>
<tr><td><code id="F.window_+3A_z2">Z2</code></td>
<td>
<p>a vector of covariates for death</p>
</td></tr>
<tr><td><code id="F.window_+3A_beta1">beta1</code></td>
<td>
<p>a vector of regression coefficients for progression</p>
</td></tr>
<tr><td><code id="F.window_+3A_beta2">beta2</code></td>
<td>
<p>a vector of regression coefficients for death</p>
</td></tr>
<tr><td><code id="F.window_+3A_eta">eta</code></td>
<td>
<p>frailty variance</p>
</td></tr>
<tr><td><code id="F.window_+3A_theta">theta</code></td>
<td>
<p>copula parameter</p>
</td></tr>
<tr><td><code id="F.window_+3A_alpha">alpha</code></td>
<td>
<p>parameter related to frailty; usually alpha=1</p>
</td></tr>
<tr><td><code id="F.window_+3A_g">g</code></td>
<td>
<p>parameters related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.window_+3A_h">h</code></td>
<td>
<p>parameters related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.window_+3A_xi1">xi1</code></td>
<td>
<p>lower bound for time to event</p>
</td></tr>
<tr><td><code id="F.window_+3A_xi3">xi3</code></td>
<td>
<p>upper bound for time to death</p>
</td></tr>
<tr><td><code id="F.window_+3A_fplot">Fplot</code></td>
<td>
<p>if FALSE, the plot is not shown</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted probability of death is calculated given the event status (X&lt;=t or X&gt;t) and covariates (Z1 and Z2).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>time</code></td>
<td>
<p>t</p>
</td></tr>
<tr><td><code>width</code></td>
<td>
<p>w</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X</p>
</td></tr>
<tr><td><code>F_event_at_X</code></td>
<td>
<p>F(t,t+w|X=x, Z1, Z2)</p>
</td></tr>
<tr><td><code>F_noevent</code></td>
<td>
<p>F(t,t+w|X&gt;t, Z1, Z2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w=1
par(mfrow=c(1,2))
F.window(time=1,X=0.2,width=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
         alpha=1,g=rep(1,5),h=rep(1,5),xi1=0,xi3=3)
F.window(time=1,X=0.8,width=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
         alpha=1,g=rep(1,5),h=rep(1,5),xi1=0,xi3=3)
</code></pre>

<hr>
<h2 id='F.window.Weibull'>Dynamic prediction of death under the joint frailty-copula model (the Weibull baseline hazard functions) </h2><span id='topic+F.window.Weibull'></span>

<h3>Description</h3>

<p>Dynamic prediction of death using a joint frailty-copula model. 
Probability of death between t and t+w is calculated given a tumour progression time X and covariates Z1 and Z2. 
If X&lt;=t, the prediction probability is F(t,t+w|X=x, Z1, Z2). If X&gt;t, the prediction probability is F(t,t+w|X&gt;t, Z1, Z2). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F.window.Weibull(time, width, X, Z1, Z2, beta1, beta2, eta, theta, alpha, 
                  scale1, shape1,scale2,shape2, xi1, xi3, Fplot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="F.window.Weibull_+3A_time">time</code></td>
<td>
<p>prediction time (=t)</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_width">width</code></td>
<td>
<p>length of window (=w)</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_x">X</code></td>
<td>
<p>time of tumour progression &lt; time</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_z1">Z1</code></td>
<td>
<p>a vector of covariates for progression</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_z2">Z2</code></td>
<td>
<p>a vector of covariates for death</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_beta1">beta1</code></td>
<td>
<p>a vector of regression coefficients for progression</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_beta2">beta2</code></td>
<td>
<p>a vector of regression coefficients for death</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_eta">eta</code></td>
<td>
<p>frailty variance</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_theta">theta</code></td>
<td>
<p>copula parameter</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_alpha">alpha</code></td>
<td>
<p>parameter related to frailty; usually alpha=1</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_scale1">scale1</code></td>
<td>
<p>scale parameter related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_shape1">shape1</code></td>
<td>
<p>shape parameter related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_scale2">scale2</code></td>
<td>
<p>scale parameter related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_shape2">shape2</code></td>
<td>
<p>shape parameter related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_xi1">xi1</code></td>
<td>
<p>lower bound for time to event</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_xi3">xi3</code></td>
<td>
<p>upper bound for time to death</p>
</td></tr>
<tr><td><code id="F.window.Weibull_+3A_fplot">Fplot</code></td>
<td>
<p>if FALSE, the plot is not shown</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted probability of death is calculated given the event status (X&lt;=t or X&gt;t) and covariates (Z1 and Z2).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>time</code></td>
<td>
<p>t</p>
</td></tr>
<tr><td><code>width</code></td>
<td>
<p>w</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X</p>
</td></tr>
<tr><td><code>F_event_at_X</code></td>
<td>
<p>F(t,t+w|X=x, Z1, Z2)</p>
</td></tr>
<tr><td><code>F_noevent</code></td>
<td>
<p>F(t,t+w|X&gt;t, Z1, Z2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayaka Shinohara, Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>
<p>Shinohara S, Lin YH, Michimae H, Emura T (2020), Dynamic lifetime prediction using a Weibull-based bivariate failure time model:
a meta-analysis of individual-patient data, Comm Stat Simul, DOI:10.1080/03610918.2020.1855449
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w=1
par(mfrow=c(1,2))
F.window.Weibull(time=1,X=0.2,width=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
                 alpha=1,scale1=1,shape1=1,scale2=1,shape2=1,xi1=0,xi3=3)
F.window.Weibull(time=1,X=0.8,width=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
                 alpha=1,scale1=1,shape1=1,scale2=1,shape2=1,xi1=0,xi3=3)
</code></pre>

<hr>
<h2 id='F.windows'>Dynamic prediction of death under the joint frailty-copula model</h2><span id='topic+F.windows'></span>

<h3>Description</h3>

<p>Dynamic prediction of death using a joint frailty-copula model. 
Probability of death between t and t+w is calculated given a tumour progression time X and covariates Z1 and Z2. 
If X&lt;=t, the prediction probability is F(t,t+w|X=x, Z1, Z2). If X&gt;t, the prediction probability is F(t,t+w|X&gt;t, Z1, Z2). 
This is a vector version of F.window.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F.windows(time, widths, X, Z1, Z2, beta1, beta2, eta, theta, alpha,
 g, h, xi1, xi3, Fplot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="F.windows_+3A_time">time</code></td>
<td>
<p>prediction time (=t)</p>
</td></tr>
<tr><td><code id="F.windows_+3A_widths">widths</code></td>
<td>
<p>length of window (=w)</p>
</td></tr>
<tr><td><code id="F.windows_+3A_x">X</code></td>
<td>
<p>time of tumour progression &lt; time</p>
</td></tr>
<tr><td><code id="F.windows_+3A_z1">Z1</code></td>
<td>
<p>a vector of covariates for progression</p>
</td></tr>
<tr><td><code id="F.windows_+3A_z2">Z2</code></td>
<td>
<p>a vector of covariates for death</p>
</td></tr>
<tr><td><code id="F.windows_+3A_beta1">beta1</code></td>
<td>
<p>regression coefficients for progression</p>
</td></tr>
<tr><td><code id="F.windows_+3A_beta2">beta2</code></td>
<td>
<p>regression coefficients for death</p>
</td></tr>
<tr><td><code id="F.windows_+3A_eta">eta</code></td>
<td>
<p>frailty variance</p>
</td></tr>
<tr><td><code id="F.windows_+3A_theta">theta</code></td>
<td>
<p>copula parameter</p>
</td></tr>
<tr><td><code id="F.windows_+3A_alpha">alpha</code></td>
<td>
<p>parameter related to frailty; usually alpha=1</p>
</td></tr>
<tr><td><code id="F.windows_+3A_g">g</code></td>
<td>
<p>parameters related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.windows_+3A_h">h</code></td>
<td>
<p>parameters related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.windows_+3A_xi1">xi1</code></td>
<td>
<p>lower bound for time to event</p>
</td></tr>
<tr><td><code id="F.windows_+3A_xi3">xi3</code></td>
<td>
<p>upper bound for time to death</p>
</td></tr>
<tr><td><code id="F.windows_+3A_fplot">Fplot</code></td>
<td>
<p>if FALSE, the plot is not shown</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted probability of death is calculated given the event status (X&lt;=t or X&gt;t) and covariates (Z1 and Z2).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>time</code></td>
<td>
<p>t</p>
</td></tr>
<tr><td><code>widths</code></td>
<td>
<p>w</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X</p>
</td></tr>
<tr><td><code>F_event_at_X</code></td>
<td>
<p>F(t,t+w|X=x, Z1, Z2)</p>
</td></tr>
<tr><td><code>F_noevent</code></td>
<td>
<p>F(t,t+w|X&gt;t, Z1, Z2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w=c(0,0.5,1,1.5,2)
par(mfrow=c(1,2))
F.windows(time=1,X=0.2,widths=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
         alpha=1,g=rep(1,5),h=rep(1,5),xi1=0,xi3=3)
F.windows(time=1,X=0.8,widths=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
         alpha=1,g=rep(1,5),h=rep(1,5),xi1=0,xi3=3)
</code></pre>

<hr>
<h2 id='F.windows.Weibull'>Dynamic prediction of death under the joint frailty-copula model (the Weibull baseline hazard functions) </h2><span id='topic+F.windows.Weibull'></span>

<h3>Description</h3>

<p>Dynamic prediction of death using a joint frailty-copula model. 
Probability of death between t and t+w is calculated given a tumour progression time X and covariates Z1 and Z2. 
If X&lt;=t, the prediction probability is F(t,t+w|X=x, Z1, Z2). If X&gt;t, the prediction probability is F(t,t+w|X&gt;t, Z1, Z2). 
This is a vector version of F.window.Weibull.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>F.windows.Weibull(time, widths, X, Z1, Z2, beta1, beta2, eta, theta, alpha, 
                  scale1, shape1,scale2,shape2, xi1, xi3, Fplot = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="F.windows.Weibull_+3A_time">time</code></td>
<td>
<p>prediction time (=t)</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_widths">widths</code></td>
<td>
<p>length of window (=w)</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_x">X</code></td>
<td>
<p>time of tumour progression &lt; time</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_z1">Z1</code></td>
<td>
<p>a vector of covariates for progression</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_z2">Z2</code></td>
<td>
<p>a vector of covariates for death</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_beta1">beta1</code></td>
<td>
<p>a vector of regression coefficients for progression</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_beta2">beta2</code></td>
<td>
<p>a vector of regression coefficients for death</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_eta">eta</code></td>
<td>
<p>frailty variance</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_theta">theta</code></td>
<td>
<p>copula parameter</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_alpha">alpha</code></td>
<td>
<p>parameter related to frailty; usually alpha=1</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_scale1">scale1</code></td>
<td>
<p>scale parameter related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_shape1">shape1</code></td>
<td>
<p>shape parameter related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_scale2">scale2</code></td>
<td>
<p>scale parameter related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_shape2">shape2</code></td>
<td>
<p>shape parameter related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_xi1">xi1</code></td>
<td>
<p>lower bound for time to event</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_xi3">xi3</code></td>
<td>
<p>upper bound for time to death</p>
</td></tr>
<tr><td><code id="F.windows.Weibull_+3A_fplot">Fplot</code></td>
<td>
<p>if FALSE, the plot is not shown</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Predicted probability of death is calculated given the event status (X&lt;=t or X&gt;t) and covariates (Z1 and Z2).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>time</code></td>
<td>
<p>t</p>
</td></tr>
<tr><td><code>widths</code></td>
<td>
<p>w</p>
</td></tr>
<tr><td><code>X</code></td>
<td>
<p>X</p>
</td></tr>
<tr><td><code>F_event_at_X</code></td>
<td>
<p>F(t,t+w|X=x, Z1, Z2)</p>
</td></tr>
<tr><td><code>F_noevent</code></td>
<td>
<p>F(t,t+w|X&gt;t, Z1, Z2)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Sayaka Shinohara, Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Matsui S, Michimae H, Rondeau V (2018), Personalized dynamic prediction of death according to tumour progression and high-dimensional
genetic factors: meta-analysis with a joint model, Stat Methods Med Res 27(9):2842-58
</p>
<p>Shinohara S, Lin YH, Michimae H, Emura T (2020), Dynamic lifetime prediction using a Weibull-based bivariate failure time model:
a meta-analysis of individual-patient data, Comm Stat Simul, DOI:10.1080/03610918.2020.1855449
</p>


<h3>Examples</h3>

<pre><code class='language-R'>w=c(0,0.5,1,1.5,2)
par(mfrow=c(1,2))
F.windows.Weibull(time=1,X=0.2,widths=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
                  alpha=1,scale1=1,shape1=1,scale2=1,shape2=1,xi1=0,xi3=3)
F.windows.Weibull(time=1,X=0.8,widths=w,Z1=1,Z2=1,beta1=1,beta2=1,eta=0.5,theta=8,
                  alpha=1,scale1=1,shape1=1,scale2=1,shape2=1,xi1=0,xi3=3)
</code></pre>

<hr>
<h2 id='I.spline'> I-spline basis function </h2><span id='topic+I.spline'></span>

<h3>Description</h3>

<p> Calculate the I-spline basis functions (the integrals of the M-spline basis functions). </p>


<h3>Usage</h3>

<pre><code class='language-R'>I.spline(time, xi1, xi3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="I.spline_+3A_time">time</code></td>
<td>
<p> a vector of time points </p>
</td></tr>
<tr><td><code id="I.spline_+3A_xi1">xi1</code></td>
<td>
<p> lower bound of time points </p>
</td></tr>
<tr><td><code id="I.spline_+3A_xi3">xi3</code></td>
<td>
<p> upper bound of time points </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output shows the values of the 5 basis functions at &quot;time&quot;,
giving a matrix with nrow=length(time) and ncol=5.
The five basis functions were originally given in the Supplementary Material of Emura et al. (2017).
More details can be found in Emura and Chen (2018), Emura et al. (2019), and Shih and Emura (2021). 
The &quot;time&quot; argument should be a vector satisfying the contraints xi1&lt;=time&lt;=xi3.
If &quot;time&quot; does not meet the constraints, error messages are shown.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>NULL</code></td>
<td>
<p>A matrix with nrow=length(time) and ncol=5, containing the values of the 5 I-spline basis functions at &quot;time&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Chen YH (2018). Analysis of Survival Data with Dependent Censoring, 
Copula-Based Approaches, JSS Research Series in Statistics, Springer, Singapore.
</p>
<p>Emura T, Matsui S, Rondeau V (2019), Survival Analysis with Correlated Endpoints; Joint Frailty-Copula Models,
JSS Research Series in Statistics, Springer
</p>
<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66: Supplementary Material. 
</p>
<p>Shih JH, Emura T (2021) Penalized Cox regression with a five-parameter spline model, Commun Stat-Theor 50(16):3749-68
</p>


<h3>Examples</h3>

<pre><code class='language-R'>I.spline(time=c(1,2,3),xi1=1,xi3=3)
</code></pre>

<hr>
<h2 id='jointCox.indep.reg'>
Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression 
and Death for Meta-Analysis
</h2><span id='topic+jointCox.indep.reg'></span>

<h3>Description</h3>

<p>Perform regression analyses under a joint Cox proportional hazards model between tumour progression and death for meta-analysis,
which is proposed by Rondeau et al. (2015). 
The method is applicable for meta-analysis combining several studies or for cluster survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jointCox.indep.reg(t.event, event, t.death, death, Z1, Z2, group, alpha = 1,
 kappa1 = c(seq(10, 1e+17, length = 30)), kappa2 = c(seq(10, 1e+17, length = 30)),
 LCV.plot = TRUE,Randomize_num = 10, Adj = 500,convergence.par=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jointCox.indep.reg_+3A_t.event">t.event</code></td>
<td>
<p>a vector for time-to-tumour progression (TTP)</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_event">event</code></td>
<td>
<p>a vector for progression indicator (=1 if progression; =0 if not progression)</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_t.death">t.death</code></td>
<td>
<p>a vector for overall survival (OS), i.e., time-to-death</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_death">death</code></td>
<td>
<p>a vector for death indicator(=1 if death; =0 if not death)</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_z1">Z1</code></td>
<td>
<p>a matrix for covariates associated with TTP; ncol(Z1)=the number of covariates</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_z2">Z2</code></td>
<td>
<p>a matrix for covariates associated with OS; ncol(Z2)=the number of covariates</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_group">group</code></td>
<td>
<p>a vector for group identification numbers, like 1,2,3....</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_alpha">alpha</code></td>
<td>
<p>A value related to the frailty (e.g., alpha=0 or =1); alpha=1 is default</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_kappa1">kappa1</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_kappa2">kappa2</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_lcv.plot">LCV.plot</code></td>
<td>
<p>Plot the LCV curves if &quot;TRUE&quot;</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_randomize_num">Randomize_num</code></td>
<td>
<p>The number of randomizations for the ititial p0</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_adj">Adj</code></td>
<td>
<p>Numerical adjustment to prevent overflow; Adj=500 is recommended</p>
</td></tr>
<tr><td><code id="jointCox.indep.reg_+3A_convergence.par">convergence.par</code></td>
<td>
<p>If TRUE, the converged estimate, gradient, and Hessian matrix are given (log-transformed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We employ &quot;nlm&quot; routine to maximize the penalized likelihood function with the initial value described in Emura et al. (2015).
If &quot;nlm&quot; does not converge, then we randomize the initial value by adding uniform random variables (Hu and Emura, 2015).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>count</code></td>
<td>
<p>Count for event occurences</p>
</td></tr>
<tr><td><code>beta1</code></td>
<td>
<p>Regression coefficient for Z1</p>
</td></tr>
<tr><td><code>beta2</code></td>
<td>
<p>Regression coefficient for Z2</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>Frailty parameter (variance)</p>
</td></tr>
<tr><td><code>LCV1</code></td>
<td>
<p>Likelihood cross-validation for TTP</p>
</td></tr>
<tr><td><code>LCV2</code></td>
<td>
<p>Likelihood cross-validation for OS</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>M-spline coefficients for TTP</p>
</td></tr>
<tr><td><code>h</code></td>
<td>
<p>M-spline coefficients for OS</p>
</td></tr>
<tr><td><code>g_var</code></td>
<td>
<p>Variance of M-spline coefficients for TTP</p>
</td></tr>
<tr><td><code>h_var</code></td>
<td>
<p>Variance of M-spline coefficients for OS</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence results for maximizing penalized likelihood</p>
</td></tr>
<tr><td><code>convergence.parameters</code></td>
<td>
<p>converged estimate, gradient, and Hessian matrix (log-transformed)</p>
</td></tr>
</table>


<h3>Error </h3>

<p>&quot;Error in integrate(func1, 0.001, 10, stop.on.error = FALSE):non-finite function value&quot;,
an error occurring when the penalized likelihood is maximizad by &quot;nlm&quot;. The error may frequently
occur during the iterations for maximizing the penalized likelihood, but is not crucial (can simply be ignored).</p>


<h3>Warning </h3>

<p>&quot;NA/Inf replaced by maximum positive value&quot;, an error occurring when the penalized likelihood
is maximizad by &quot;nlm&quot;. The error frequently occurs during the iterations for maximizing the penalized likelihood,
but is not crucial (can simply be ignored).</p>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Rondeau V, Pignon JP, Michiels S (2015). A joint model for dependence between clustered times to tumour progression and deaths: A meta-analysis of chemotherapy in head and neck cancer.
Stat Methods Med Res 24(6):711-729.
</p>
<p>Hu YH, Emura T (2015), Maximum likelihood estimation for a special exponential family under random double-truncation,
Computational Statist 30(4): 1199-1229
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############## Reproduce the results of Emura et al. (2015) #############
data(dataOvarian)
t.event=dataOvarian$t.event
event=dataOvarian$event
t.death=dataOvarian$t.death
death=dataOvarian$death
Z1=dataOvarian$CXCL12
group=dataOvarian$group
alpha_given=0
kappa_grid=seq(10,1e+17,length=30)
set.seed(1)
#jointCox.indep.reg(t.event=t.event,event=event,t.death=t.death,death=death,
#                 Z1=Z1,Z2=Z1,group=group,alpha=alpha_given,
#                 kappa1=kappa_grid,kappa2=kappa_grid,LCV.plot=TRUE,Adj=500)

</code></pre>

<hr>
<h2 id='jointCox.reg'>
Penalized Likelihood Estimation under the Joint Cox Models Between Tumour Progression 
and Death for Meta-Analysis
</h2><span id='topic+jointCox.reg'></span>

<h3>Description</h3>

<p>Perform regression analyses under a copula-based joint Cox proportional hazards model between tumour progression and death for meta-analysis,
which is proposed by Emura et al. (2017). The methodological details can be found in Emura et al. (2019).
The method is applicable for meta-analysis combining several studies or for cluster survival data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jointCox.reg(t.event, event, t.death, death, Z1, Z2, group, alpha = 1,
 kappa1 = c(seq(10, 1e+17, length = 30)),kappa2 = c(seq(10, 1e+17, length = 30)),
 LCV.plot = TRUE, Randomize_num = 10, u.min = 0.001, u.max = 10,
 Adj = 500,convergence.par=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jointCox.reg_+3A_t.event">t.event</code></td>
<td>
<p>a vector for time-to-tumour progression (TTP)</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_event">event</code></td>
<td>
<p>a vector for progression indicator (=1 if progression; =0 if not progression)</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_t.death">t.death</code></td>
<td>
<p>a vector for overall survival (OS), i.e., time-to-death</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_death">death</code></td>
<td>
<p>a vector for death indicator(=1 if death; =0 if not death)</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_z1">Z1</code></td>
<td>
<p>a matrix for covariates associated with TTP; ncol(Z1)=the number of covariates</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_z2">Z2</code></td>
<td>
<p>a matrix for covariates associated with OS; ncol(Z2)=the number of covariates</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_group">group</code></td>
<td>
<p>a vector for group identification numbers, like 1,2,3....</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_alpha">alpha</code></td>
<td>
<p>A value related to the frailty (e.g., alpha=0 or =1); alpha=1 is default</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_kappa1">kappa1</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_kappa2">kappa2</code></td>
<td>
<p>a vector for candidate smoothing parameters</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_lcv.plot">LCV.plot</code></td>
<td>
<p>Plot the LCV curves if &quot;TRUE&quot;</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_randomize_num">Randomize_num</code></td>
<td>
<p>The number of randomizations for the ititial p0</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_u.min">u.min</code></td>
<td>
<p>the lower bound of the numerical integration for the frailty term</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_u.max">u.max</code></td>
<td>
<p>the upper bound of the numerical integration for the frailty term</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_adj">Adj</code></td>
<td>
<p>Numerical adjustment to prevent overflow; Adj=500 is recommended</p>
</td></tr>
<tr><td><code id="jointCox.reg_+3A_convergence.par">convergence.par</code></td>
<td>
<p>If TRUE, the converged estimate, gradient, and Hessian matrix are given (log-transformed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We employ &quot;nlm&quot; routine to maximize the penalized likelihood function with the initial value described in Emura et al. (2017).
If &quot;nlm&quot; does not converge, then we randomize the initial value by adding uniform random variables (Hu and Emura, 2015).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>count</code></td>
<td>
<p>Count for event occurences</p>
</td></tr>
<tr><td><code>beta1</code></td>
<td>
<p>Regression coefficient for Z1</p>
</td></tr>
<tr><td><code>beta2</code></td>
<td>
<p>Regression coefficient for Z2</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>Frailty parameter (variance)</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>Copula parameter under the Clayton copula</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau corresponding to the copula parameter</p>
</td></tr>
<tr><td><code>LCV1</code></td>
<td>
<p>Likelihood cross-validation for TTP</p>
</td></tr>
<tr><td><code>LCV2</code></td>
<td>
<p>Likelihood cross-validation for OS</p>
</td></tr>
<tr><td><code>g</code></td>
<td>
<p>M-spline coefficients for TTP</p>
</td></tr>
<tr><td><code>h</code></td>
<td>
<p>M-spline coefficients for OS</p>
</td></tr>
<tr><td><code>g_var</code></td>
<td>
<p>Variance of M-spline coefficients for TTP</p>
</td></tr>
<tr><td><code>h_var</code></td>
<td>
<p>Variance of M-spline coefficients for OS</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence results for maximizing penalized likelihood</p>
</td></tr>
<tr><td><code>convergence.parameters</code></td>
<td>
<p>converged estimate, gradient, and Hessian matrix (log-transformed)</p>
</td></tr>
</table>


<h3>Error </h3>

<p>&quot;Error in integrate(func1, 0.001, 10, stop.on.error = FALSE):non-finite function value&quot;,
an error occurring when the penalized likelihood is maximizad by &quot;nlm&quot;. The error may frequently
occur during the iterations for maximizing the penalized likelihood, but is not crucial (can simply be ignored).</p>


<h3>Warning </h3>

<p>&quot;NA/Inf replaced by maximum positive value&quot;, an error occurring when the penalized likelihood
is maximizad by &quot;nlm&quot;. The error frequently occurs during the iterations for maximizing the penalized likelihood,
but is not crucial (can simply be ignored).</p>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66
</p>
<p>Emura T, Matsui S, Rondeau V (2019), Survival Analysis with Correlated Endpoints; Joint Frailty-Copula Models,
JSS Research Series in Statistics, Springer
</p>
<p>Hu YH, Emura T (2015), Maximum likelihood estimation for a special exponential family under random double-truncation,
Computational Stat 30 (4): 1199-1229
</p>


<h3>Examples</h3>

<pre><code class='language-R'>############## Reproduce the results of Emura et al. (2017) #############
data(dataOvarian)
t.event=dataOvarian$t.event
event=dataOvarian$event
t.death=dataOvarian$t.death
death=dataOvarian$death
Z1=dataOvarian$CXCL12
group=dataOvarian$group
alpha_given=0
kappa_grid=seq(10,1e+17,length=30)
set.seed(1)
#jointCox.reg(t.event=t.event,event=event,t.death=t.death,death=death,
#                 Z1=Z1,Z2=Z1,group=group,alpha=alpha_given,
#                 kappa1=kappa_grid,kappa2=kappa_grid,LCV.plot=TRUE,Adj=500)

</code></pre>

<hr>
<h2 id='jointCox.Weibull.reg'>
Weibull-based Likelihood Estimation under the Joint Cox Models Between Tumour Progression 
and Death for Meta-Analysis
</h2><span id='topic+jointCox.Weibull.reg'></span>

<h3>Description</h3>

<p>Perform Weibull regression analyses under a copula-based joint Cox proportional hazards model between tumour progression and death for meta-analysis,
which is proposed by Wu et al. (2020).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jointCox.Weibull.reg(t.event, event, t.death, death, Z1, Z2, group, alpha = 1,
 Randomize_num = 10, u.min = 0.001, u.max = 10, Adj = 500,convergence.par=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jointCox.Weibull.reg_+3A_t.event">t.event</code></td>
<td>
<p>a vector for time-to-tumour progression (TTP)</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_event">event</code></td>
<td>
<p>a vector for progression indicator (=1 if progression; =0 if not progression)</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_t.death">t.death</code></td>
<td>
<p>a vector for overall survival (OS), i.e., time-to-death</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_death">death</code></td>
<td>
<p>a vector for death indicator(=1 if death; =0 if not death)</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_z1">Z1</code></td>
<td>
<p>a matrix for covariates associated with TTP; ncol(Z1)=the number of covariates</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_z2">Z2</code></td>
<td>
<p>a matrix for covariates associated with OS; ncol(Z2)=the number of covariates</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_group">group</code></td>
<td>
<p>a vector for group identification numbers, like 1,2,3....</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_alpha">alpha</code></td>
<td>
<p>A value related to the frailty (e.g., alpha=0 or =1); alpha=1 is default</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_randomize_num">Randomize_num</code></td>
<td>
<p>The number of randomizations for the ititial p0</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_u.min">u.min</code></td>
<td>
<p>the lower bound of the numerical integration for the frailty term</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_u.max">u.max</code></td>
<td>
<p>the upper bound of the numerical integration for the frailty term</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_adj">Adj</code></td>
<td>
<p>Numerical adjustment to prevent overflow; Adj=500 is recommended</p>
</td></tr>
<tr><td><code id="jointCox.Weibull.reg_+3A_convergence.par">convergence.par</code></td>
<td>
<p>If TRUE, the converged estimate, gradient, and Hessian matrix are given (log-transformed)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We employ &quot;nlm&quot; routine to maximize the penalized likelihood function with the initial value described in Wu et al. (2020).
If &quot;nlm&quot; does not converge, then we randomize the initial value by adding uniform random variables (Hu and Emura, 2015).
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>count</code></td>
<td>
<p>Count for event occurences</p>
</td></tr>
<tr><td><code>beta1</code></td>
<td>
<p>Regression coefficient for Z1</p>
</td></tr>
<tr><td><code>beta2</code></td>
<td>
<p>Regression coefficient for Z2</p>
</td></tr>
<tr><td><code>eta</code></td>
<td>
<p>Frailty parameter (variance)</p>
</td></tr>
<tr><td><code>theta</code></td>
<td>
<p>Copula parameter under the Clayton copula</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>Kendall's tau corresponding to the copula parameter</p>
</td></tr>
<tr><td><code>scale1</code></td>
<td>
<p>Scale parameter for the Weibull model of TTP</p>
</td></tr>
<tr><td><code>shape1</code></td>
<td>
<p>Shape parameter for the Weibull model of TTP</p>
</td></tr>
<tr><td><code>scale2</code></td>
<td>
<p>Scale parameter for the Weibull model of OS</p>
</td></tr>
<tr><td><code>shape2</code></td>
<td>
<p>Shape parameter for the Weibull model of OS</p>
</td></tr>
<tr><td><code>convergence</code></td>
<td>
<p>convergence results for maximizing penalized likelihood</p>
</td></tr>
<tr><td><code>convergence.parameters</code></td>
<td>
<p>converged estimate, gradient, and Hessian matrix (log-transformed)</p>
</td></tr>
</table>


<h3>Error </h3>

<p>&quot;Error in integrate(func1, 0.001, 10, stop.on.error = FALSE):non-finite function value&quot;,
an error occurring when the penalized likelihood is maximizad by &quot;nlm&quot;. The error may frequently
occur during the iterations for maximizing the penalized likelihood, but is not crucial (can simply be ignored).</p>


<h3>Warning </h3>

<p>&quot;NA/Inf replaced by maximum positive value&quot;, an error occurring when the penalized likelihood
is maximizad by &quot;nlm&quot;. The error frequently occurs during the iterations for maximizing the penalized likelihood,
but is not crucial (can simply be ignored).</p>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Wu BH, Michimae H, Emura T (2020), Meta-analysis of individual patient data with semi-competing risks under
the Weibull joint frailty-copula model. Comp Stat 35(4):1525-52
</p>
<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66
</p>
<p>Hu YH, Emura T (2015), Maximum likelihood estimation for a special exponential family under random double-truncation,
Comp Stat 30 (4): 1199-1229
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataOvarian)
t.event=dataOvarian$t.event
event=dataOvarian$event
t.death=dataOvarian$t.death
death=dataOvarian$death
Z1=dataOvarian$CXCL12
group=dataOvarian$group
alpha_given=0

#set.seed(1)
#jointCox.Weibull.reg(t.event=t.event,event=event,t.death=t.death,death=death,
#                 Z1=Z1,Z2=Z1,group=group,alpha=alpha_given,Adj=500)
</code></pre>

<hr>
<h2 id='M.spline'> M-spline basis function </h2><span id='topic+M.spline'></span>

<h3>Description</h3>

<p> Calculate the M-spline basis functions (a M-spline basis is a B-spline basis normalized so that the integral is 1).</p>


<h3>Usage</h3>

<pre><code class='language-R'>M.spline(time, xi1, xi3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="M.spline_+3A_time">time</code></td>
<td>
<p> a vector of time points </p>
</td></tr>
<tr><td><code id="M.spline_+3A_xi1">xi1</code></td>
<td>
<p> lower bound of time points </p>
</td></tr>
<tr><td><code id="M.spline_+3A_xi3">xi3</code></td>
<td>
<p> upper bound of time points </p>
</td></tr>
</table>


<h3>Details</h3>

<p>The output shows the values of the 5 basis functions at &quot;time&quot;,
giving a matrix with nrow=length(time) and ncol=5.
The five basis functions were originally given in the Supplementary Material of Emura et al. (2017).
More details can be found in Emura and Chen (2018), Emura et al. (2019), and Shih and Emura (2021). 
The &quot;time&quot; argument should be a vector satisfying the contraints xi1&lt;=time&lt;=xi3.
If &quot;time&quot; does not meet the constraints, error messages are shown.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>NULL</code></td>
<td>
<p>A matrix with nrow=length(time) and ncol=5, containing the values of the 5 spline basis functions at &quot;time&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Emura T, Chen YH (2018). Analysis of Survival Data with Dependent Censoring, 
Copula-Based Approaches, JSS Research Series in Statistics, Springer, Singapore.
</p>
<p>Emura T, Matsui S, Rondeau V (2019), Survival Analysis with Correlated Endpoints; Joint Frailty-Copula Models,
JSS Research Series in Statistics, Springer
</p>
<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66: Supplementary Material. 
</p>
<p>Shih JH, Emura T (2021) Penalized Cox regression with a five-parameter spline model, Commun Stat-Theor 50(16):3749-68
</p>


<h3>Examples</h3>

<pre><code class='language-R'>M.spline(time=c(1,2,3),xi1=1,xi3=3)
</code></pre>

<hr>
<h2 id='splineCox.reg'>Fitting the Cox model for survival data using a penalized spline model</h2><span id='topic+splineCox.reg'></span>

<h3>Description</h3>

<p>Fitting the Cox proportional hazards model when the baseline hazard function is specified by a five-parameter spline model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splineCox.reg(t.event, event, Z, xi1 = min(t.event), xi3 = max(t.event), 
kappa = c(seq(10, 1e+17, length = 30)), LCV.plot = TRUE,p0=rep(0,5+p))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="splineCox.reg_+3A_t.event">t.event</code></td>
<td>
<p>a vector for time-to-event</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_event">event</code></td>
<td>
<p>a vector for event indicator (=1 event; =0 censoring)</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_z">Z</code></td>
<td>
<p>a matrix for covariates; nrow(Z)=sample size, ncol(Z)=the number of covariates</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_xi1">xi1</code></td>
<td>
<p>lower bound for the hazard function; the default is min(t.event)</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_xi3">xi3</code></td>
<td>
<p>upper bound for the hazard function; the default is max(t.event)</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_kappa">kappa</code></td>
<td>
<p>a vector for candidate smoothing parameters. Only positive values are allowed. Values too close to zero may yeild errors (see below).</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_lcv.plot">LCV.plot</code></td>
<td>
<p>Plot the LCV curves if &quot;TRUE&quot;. This plot is used to find the optimal value from the candidate smoothing parameters given by &quot;kappa&quot;.</p>
</td></tr>
<tr><td><code id="splineCox.reg_+3A_p0">p0</code></td>
<td>
<p>Initial values to maximize the penalized likelihood (5+p parameters; five M-spline coefficients and p regression coefficients)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>One can perform Cox-type regression for censored survival data with covariates. The method is essentially the same as
as Cox regression (Cox 1972) expect for the models of the baseline hazard function.
Unlike the nonparametric model of Cox (1972), the method applies a five-parameter spline model as originally proposed by Emura et al. (2017).
The method is detailed in Section 2.4 of Emura et al. (2019). See also Shih and Emura (2021) for more details.
This method is also used as a subroutine for computing the optimal smoothing parameter (kappa1 and kappa2) for
many advanced functions, such as &quot;jointCox.reg&quot;, &quot;cmprskCox.reg&quot;, and &quot;condCox.reg&quot;.
The definition of LCV is given in Section 3.7 of Emura et al. (2019). See also Shih and Emura (2021).
The error message &quot;Error in nlm(l.func, p = rep(0, 5 + p), hessian = TRUE):non-finite value supplied by 'nlm'&quot;
may imply that some candidate parameters for kappa are too close to zero; please exclude such values from kappa.
The output values are usually similar to those given by &quot;coxph(Surv(t.event,event)~Z)&quot;. 
Unreasonable output values are usually caused by a wrong choice of &quot;kappa&quot; and occasionary caused by a wrong choice of p0.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>beta</code></td>
<td>
<p>Regression coefficient for Z</p>
</td></tr>
<tr><td><code>h</code></td>
<td>
<p>M-spline coefficients</p>
</td></tr>
<tr><td><code>h_var</code></td>
<td>
<p>Variance of M-spline coefficients</p>
</td></tr>
<tr><td><code>kappa</code></td>
<td>
<p>smoothing parameter at the optimal LCV</p>
</td></tr>
<tr><td><code>DF</code></td>
<td>
<p>degree of freedom at the optimal LCV</p>
</td></tr>
<tr><td><code>LCV</code></td>
<td>
<p>the optimal LCV(=logL-DF)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Cox DR (1972), Regression models and life-tables, JRSS(B) 34(2):187-202.
</p>
<p>Emura T, Matsui S, Rondeau V (2019), Survival Analysis with Correlated Endpoints; Joint Frailty-Copula Models,
JSS Research Series in Statistics, Springer
</p>
<p>Emura T, Nakatochi M, Murotani K, Rondeau V (2017), A joint frailty-copula model between tumour progression and death for meta-analysis,
Stat Methods Med Res 26(6): 2649-66: Supplementary Material. 
</p>
<p>Shih JH, Emura T (2021) Penalized Cox regression with a five-parameter spline model, Commun Stat-Theor 50(16):3749-68
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dataOvarian)
t.event=dataOvarian$t.event
event=dataOvarian$event
t.death=dataOvarian$t.death
death=dataOvarian$death
Z=dataOvarian$CXCL12
#splineCox.reg(t.event,event,Z,kappa=c(seq(10,1e+17,length=30)))
</code></pre>

<hr>
<h2 id='Weibull.simu'>Simulating data from the Weibull joint frailty-copula model</h2><span id='topic+Weibull.simu'></span>

<h3>Description</h3>

<p>This function generate clustered (grouped) bivariate event times from the joint frailty-copula model
with the Weibull baseline hazard functions.
Simulating (X_ij,D_ij,C_ij), i=1,2,...,G, and j=1,2,...,N, where G is the number of studies (groups),
and N is the number of individuals (patients) within each study.
X_ij is time-to-event, D_ij is time-to-death, and C_ij is time-to-censoring.
(X_ij, D_ij) and C_ij are independent.
Dependence structure on (X_ij, D_ij) is modeled by a copula, which can be the Clayton (default), Frank, Gumbel, or BB1.
Covariate effects are specified by the Cox models given a frailty term. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Weibull.simu(G,N,scale1,scale2,shape1,shape2,beta1,beta2,
 eta,copula="Clayton",theta,d=0,alpha,beta12=0,C.max,
 cmprsk=FALSE,tau=FALSE,Z.dist=runif,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Weibull.simu_+3A_g">G</code></td>
<td>
<p>The number of studies or groups</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_n">N</code></td>
<td>
<p>The number of patients within each study</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_scale1">scale1</code></td>
<td>
<p>scale parameter related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_scale2">scale2</code></td>
<td>
<p>scale parameter related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_shape1">shape1</code></td>
<td>
<p>shape parameter related to the baseline hazard for progression</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_shape2">shape2</code></td>
<td>
<p>shape parameter related to the baseline hazard for death</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_beta1">beta1</code></td>
<td>
<p>regression coefficients for progression</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_beta2">beta2</code></td>
<td>
<p>regression coefficients for death</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_eta">eta</code></td>
<td>
<p>frailty variance</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_copula">copula</code></td>
<td>
<p>copula function; &quot;Clayton&quot; (default), &quot;Gumbel&quot;, &quot;Frank&quot;, or &quot;BB1&quot;</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_theta">theta</code></td>
<td>
<p>copula parameter</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_d">d</code></td>
<td>
<p>BB1 copula's departure parameter from the Clayton (d=0 is the default)</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_alpha">alpha</code></td>
<td>
<p>parameter related to frailty, e.g., alpha=1</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_beta12">beta12</code></td>
<td>
<p>regression coefficients for copula</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_c.max">C.max</code></td>
<td>
<p>the upper bound for the censoring distribution</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_cmprsk">cmprsk</code></td>
<td>
<p>if TRUE, simulated data follow the competing risks setting</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_tau">tau</code></td>
<td>
<p>if TRUE, conditional Kendall's tau given Z is shown</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_z.dist">Z.dist</code></td>
<td>
<p>the distribution of a covariate Z</p>
</td></tr>
<tr><td><code id="Weibull.simu_+3A_...">...</code></td>
<td>
<p>parameters for Z.dist</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See Wu et al. (2020) for the algorithms for the Clayton copula.
The method was later extended by including covariate effects
on a copula (beta12) via the conditional copula model of Emura et al. (2021),
The available copulas are the Frank, Gumbel, and BB1 copulas.
For the BB1 copula, please see Supplementary Material:Additional simulation studies under
the copula misspecification in Emura et al. (2021),
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X</code></td>
<td>
<p>: time to event</p>
</td></tr>
<tr><td><code>D</code></td>
<td>
<p>: time to death</p>
</td></tr>
<tr><td><code>C</code></td>
<td>
<p>: time to independent censoring</p>
</td></tr>
<tr><td><code>t.event</code></td>
<td>
<p>: time to event (=min(X,D,C))</p>
</td></tr>
<tr><td><code>event</code></td>
<td>
<p>: event indicator (=I(X&lt;=D,X&lt;=C))</p>
</td></tr>
<tr><td><code>event1</code></td>
<td>
<p>: indicator for Event 1 (=I(X&lt;=D,X&lt;=C))</p>
</td></tr>
<tr><td><code>t.death</code></td>
<td>
<p>: time to death (=min(D,C))</p>
</td></tr>
<tr><td><code>death</code></td>
<td>
<p>: death indicator (=I(D&lt;=C))</p>
</td></tr>
<tr><td><code>event2</code></td>
<td>
<p>: indicator for Event 2 (=I(D&lt;X,D&lt;=C))</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>: study ID (=1,2,...,G)</p>
</td></tr>
<tr><td><code>Z</code></td>
<td>
<p>: covariate</p>
</td></tr>
<tr><td><code>tau</code></td>
<td>
<p>: Conditional Kendall's tau given Z</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Takeshi Emura</p>


<h3>References</h3>

<p>Wu BH, Michimae H, Emura T (2020), Meta-analysis of individual patient data with semi-competing risks under
the Weibull joint frailty-copula model. Comp Stat 35(4):1525-52
</p>
<p>Emura T, Shih JH, Ha ID, Wilke RA (2020), Comparison of the marginal hazard model and the sub-distribution hazard model for competing risks under an assumed copula,
Stat Methods Med Res 29(8):2307-27
</p>
<p>Emura T, Sofeu C, Rondeau V (2021), Conditional copula models for correlated survival endpoints:
individual patient data meta-analysis of randomized controlled trials,
Stat Methods Med Res 30(12):2634-50
</p>
<p>Supplementary Material:Additional simulation studies under the copula misspecification
in &quot;Emura T, Sofeu C, Rondeau V (2021), Conditional copula models for correlated survival endpoints:
individual patient data meta-analysis of randomized controlled trials,
Stat Methods Med Res 30(12):2634-50&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Weibull.simu(G=5,N=2,scale1=1,scale2=1,shape1=1,shape2=1,
        beta1=1,beta2=1,eta=0.5,theta=2,alpha=1,C.max=5)

Weibull.simu(G=5,N=2,scale1=1,scale2=1,shape1=1,shape2=1,
        beta1=1,beta2=1,eta=0.5,copula="Gumbel",theta=2,alpha=1,C.max=5)

Weibull.simu(G=5,N=2,scale1=1,scale2=1,shape1=1,shape2=1,
        beta1=1,beta2=1,eta=0.5,theta=2,alpha=1,C.max=5,Z.dist=rbinom,size=1,prob=0.5)

## simulated data follow the competing risks setting
Weibull.simu(G=5,N=2,scale1=1,scale2=1,shape1=1,shape2=1,
        beta1=1,beta2=1,eta=0.5,theta=2,alpha=1,C.max=5,cmprsk=TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
