<!DOCTYPE html><html><head><title>Help for package JSparO</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {JSparO}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#demo_JSparO'><p>demo_JSparO - The demo of JSparO package</p></a></li>
<li><a href='#L1HalfThr'><p>L1HalfThr - Iterative Half Thresholding Algorithm based on <code class="reqn">l_{1,1/2}</code> norm</p></a></li>
<li><a href='#L1HardThr'><p>L1HardThr - Iterative Hard Thresholding Algorithm based on <code class="reqn">l_{1,0}</code> norm</p></a></li>
<li><a href='#L1normFun'><p>L1normFun</p></a></li>
<li><a href='#L1SoftThr'><p>L1SoftThr - Iterative Soft Thresholding Algorithm based on <code class="reqn">l_{1,1}</code> norm</p></a></li>
<li><a href='#L1twothirdsThr'><p>L1twothirdsThr - Iterative Thresholding Algorithm based on <code class="reqn">l_{1,2/3}</code> norm</p></a></li>
<li><a href='#L2HalfThr'><p>L2HalfThr - Iterative Half Thresholding Algorithm based on <code class="reqn">l_{2,1/2}</code> norm</p></a></li>
<li><a href='#L2HardThr'><p>L2HardThr - Iterative Hard Thresholding Algorithm based on <code class="reqn">l_{2,0}</code> norm</p></a></li>
<li><a href='#L2NewtonThr'><p>L2NewtonThr - Iterative Thresholding Algorithm based on <code class="reqn">l_{2,q}</code> norm with Newton method</p></a></li>
<li><a href='#L2SoftThr'><p>L2SoftThr - Iterative Soft Thresholding Algorithm based on <code class="reqn">l_{2,1}</code> norm</p></a></li>
<li><a href='#L2twothirdsThr'><p>L2twothirdsThr - Iterative Thresholding Algorithm based on <code class="reqn">l_{2,2/3}</code> norm</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Joint Sparse Optimization via Proximal Gradient Method for Cell
Fate Conversion</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Xinlin Hu [aut, cre],
  Yaohua Hu [cph, aut]</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of joint sparse optimization (JSparO) to infer the gene regulatory network for cell fate conversion. The proximal gradient method is implemented to solve different low-order regularization models for JSparO.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xinlin Hu &lt;thompson-xinlin.hu@connect.polyu.hk&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>pracma</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-08-17 11:09:41 UTC; Thompson</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-08-18 08:40:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='demo_JSparO'>demo_JSparO - The demo of JSparO package</h2><span id='topic+demo_JSparO'></span>

<h3>Description</h3>

<p>This is the main function of JSparO aimed to solve the low-order regularization models with <code class="reqn">l_{p,q}</code> norm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>demo_JSparO(A, B, X, s, p, q, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="demo_JSparO_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="demo_JSparO_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="demo_JSparO_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="demo_JSparO_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="demo_JSparO_+3A_p">p</code></td>
<td>
<p>value for <code class="reqn">l_{p,q}</code> norm (i.e., p = 1 or 2)</p>
</td></tr>
<tr><td><code id="demo_JSparO_+3A_q">q</code></td>
<td>
<p>value for <code class="reqn">l_{p,q}</code> norm (i.e., 0 &lt;= q &lt;= 1)</p>
</td></tr>
<tr><td><code id="demo_JSparO_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The demo_JSparO function is used to solve joint sparse optimization problem via different algorithms.
Based on <code class="reqn">l_{p,q}</code> norm, functions with different p and q are implemented to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{p,q}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{p,q}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
res_JSparO &lt;- demo_JSparO(A0, B0, X0, s = 10, p = 2, q = 'half', maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L1HalfThr'>L1HalfThr - Iterative Half Thresholding Algorithm based on <code class="reqn">l_{1,1/2}</code> norm</h2><span id='topic+L1HalfThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{1,1/2}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L1HalfThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L1HalfThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L1HalfThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L1HalfThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L1HalfThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L1HalfThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L1HalfThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{1,1/2}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{1,1/2}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L1half &lt;- L1HalfThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L1HardThr'>L1HardThr - Iterative Hard Thresholding Algorithm based on <code class="reqn">l_{1,0}</code> norm</h2><span id='topic+L1HardThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{1,0}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L1HardThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L1HardThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L1HardThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L1HardThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L1HardThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L1HardThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L1HardThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{1,0}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{1,0}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L10 &lt;- L1HardThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L1normFun'>L1normFun</h2><span id='topic+L1normFun'></span>

<h3>Description</h3>

<p>The function aims to compute the <code class="reqn">l_1</code> norm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L1normFun(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L1normFun_+3A_x">x</code></td>
<td>
<p>vector</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L1normFun aims to compute the <code class="reqn">l_1</code> norm: <code class="reqn">\sum_i^n |x_i|</code>
</p>


<h3>Value</h3>

<p>The <code class="reqn">l_1</code> norm of vector x
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>

<hr>
<h2 id='L1SoftThr'>L1SoftThr - Iterative Soft Thresholding Algorithm based on <code class="reqn">l_{1,1}</code> norm</h2><span id='topic+L1SoftThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{1,1}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L1SoftThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L1SoftThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L1SoftThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L1SoftThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L1SoftThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L1SoftThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L1SoftThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{1,1}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{1,1}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L11 &lt;- L1SoftThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L1twothirdsThr'>L1twothirdsThr - Iterative Thresholding Algorithm based on <code class="reqn">l_{1,2/3}</code> norm</h2><span id='topic+L1twothirdsThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{1,2/3}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L1twothirdsThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L1twothirdsThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L1twothirdsThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L1twothirdsThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L1twothirdsThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L1twothirdsThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L1twothirdsThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{1,2/3}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{1,2/3}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L1twothirds &lt;- L1twothirdsThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L2HalfThr'>L2HalfThr - Iterative Half Thresholding Algorithm based on <code class="reqn">l_{2,1/2}</code> norm</h2><span id='topic+L2HalfThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{2,1/2}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L2HalfThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L2HalfThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L2HalfThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L2HalfThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L2HalfThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L2HalfThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L2HalfThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{2,1/2}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{2,1/2}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L2half &lt;- L2HalfThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L2HardThr'>L2HardThr - Iterative Hard Thresholding Algorithm based on <code class="reqn">l_{2,0}</code> norm</h2><span id='topic+L2HardThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{2,0}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L2HardThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L2HardThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L2HardThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L2HardThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L2HardThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L2HardThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L2HardThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{2,0}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{2,0}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L20 &lt;- L2HardThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L2NewtonThr'>L2NewtonThr - Iterative Thresholding Algorithm based on <code class="reqn">l_{2,q}</code> norm with Newton method</h2><span id='topic+L2NewtonThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{2,q}</code> regularized least squares, where the proximal optimization subproblems will be solved by Newton method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L2NewtonThr(A, B, X, s, q, maxIter = 200, innMaxIter = 30, innEps = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L2NewtonThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_q">q</code></td>
<td>
<p>value for <code class="reqn">l_{2,q}</code> norm (i.e., 0 &lt; q &lt; 1)</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_innmaxiter">innMaxIter</code></td>
<td>
<p>maximum iteration in Newton step</p>
</td></tr>
<tr><td><code id="L2NewtonThr_+3A_inneps">innEps</code></td>
<td>
<p>criterion to stop inner iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L2NewtonThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{2,q}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{2,q}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L2q &lt;- L2NewtonThr(A0, B0, X0, s = 10, q = 0.2, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L2SoftThr'>L2SoftThr - Iterative Soft Thresholding Algorithm based on <code class="reqn">l_{2,1}</code> norm</h2><span id='topic+L2SoftThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{2,1}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L2SoftThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L2SoftThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L2SoftThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L2SoftThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L2SoftThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L2SoftThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L2SoftThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{2,1}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{2,1}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L21 &lt;- L2SoftThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

<hr>
<h2 id='L2twothirdsThr'>L2twothirdsThr - Iterative Thresholding Algorithm based on <code class="reqn">l_{2,2/3}</code> norm</h2><span id='topic+L2twothirdsThr'></span>

<h3>Description</h3>

<p>The function aims to solve <code class="reqn">l_{2,2/3}</code> regularized least squares.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>L2twothirdsThr(A, B, X, s, maxIter = 200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="L2twothirdsThr_+3A_a">A</code></td>
<td>
<p>Gene expression data of transcriptome factors (i.e. feature matrix in machine learning).
The dimension of A is m * n.</p>
</td></tr>
<tr><td><code id="L2twothirdsThr_+3A_b">B</code></td>
<td>
<p>Gene expression data of target genes (i.e. observation matrix in machine learning).
The dimension of B is m * t.</p>
</td></tr>
<tr><td><code id="L2twothirdsThr_+3A_x">X</code></td>
<td>
<p>Gene expression data of Chromatin immunoprecipitation or other matrix
(i.e. initial iterative point in machine learning). The dimension of X is n * t.</p>
</td></tr>
<tr><td><code id="L2twothirdsThr_+3A_s">s</code></td>
<td>
<p>joint sparsity level</p>
</td></tr>
<tr><td><code id="L2twothirdsThr_+3A_maxiter">maxIter</code></td>
<td>
<p>maximum iteration</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The L2twothirdsThr function aims to solve the problem:
</p>
<p style="text-align: center;"><code class="reqn">\min \|AX-B\|_F^2 + \lambda \|X\|_{2,2/3}</code>
</p>

<p>to obtain s-joint sparse solution.
</p>


<h3>Value</h3>

<p>The solution of proximal gradient method with <code class="reqn">l_{2,2/3}</code> regularizer.
</p>


<h3>Author(s)</h3>

<p>Xinlin Hu <a href="mailto:thompson-xinlin.hu@connect.polyu.hk">thompson-xinlin.hu@connect.polyu.hk</a>
</p>
<p>Yaohua Hu <a href="mailto:mayhhu@szu.edu.cn">mayhhu@szu.edu.cn</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m &lt;- 256; n &lt;- 1024; t &lt;- 5; maxIter0 &lt;- 50
A0 &lt;- matrix(rnorm(m * n), nrow = m, ncol = n)
B0 &lt;- matrix(rnorm(m * t), nrow = m, ncol = t)
X0 &lt;- matrix(0, nrow = n, ncol = t)
NoA &lt;- norm(A0, '2'); A0 &lt;- A0/NoA; B0 &lt;- B0/NoA
res_L2twothirds &lt;- L2twothirdsThr(A0, B0, X0, s = 10, maxIter = maxIter0)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
