<!DOCTYPE html><html><head><title>Help for package junctions</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {junctions}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calc_k'><p>Calculate the limit of the number of junctions</p></a></li>
<li><a href='#calculate_mat'><p>Function to calculate the maximum accurate time</p></a></li>
<li><a href='#estimate_time'><p>Estimate the time since the onset of hybridization, using the number of</p>
junctions</a></li>
<li><a href='#estimate_time_diploid'><p>estimates the time since admixture, given diploid ancestry data.</p></a></li>
<li><a href='#estimate_time_haploid'><p>estimate time using likelihood for a single chromosome</p></a></li>
<li><a href='#estimate_time_one_chrom'><p>Estimate the time since the onset of hybridization, using the observed</p>
number of junctions, taking into account the distribution of markers on
a single chromosome</a></li>
<li><a href='#junctions-package'>
<p>Extending The Theory of Junctions</p></a></li>
<li><a href='#log_likelihood_diploid'><p>calculate the log likelihood of observing diploid ancestry data.</p></a></li>
<li><a href='#log_likelihood_haploid'><p>log likelihood of the time since admixture for a haploid genome</p></a></li>
<li><a href='#number_of_junctions'><p>Calculate the average number of junctions</p></a></li>
<li><a href='#number_of_junctions_backcross'><p>Calculate the average number of junctions during backcrossing</p></a></li>
<li><a href='#number_of_junctions_di'><p>Calculate the expected number of junctions between two markers separated by</p>
a given amount of recombination</a></li>
<li><a href='#number_of_junctions_markers'><p>Calculate the expected total number of junctions in a chromosome, given the</p>
distribution of markers</a></li>
<li><a href='#sim_backcrossing'><p>Function to simulate data using a back crossing scheme</p></a></li>
<li><a href='#sim_fin_chrom'><p>Individual Based Simulation of the accumulation of junctions</p></a></li>
<li><a href='#sim_inf_chrom'><p>Individual Based Simulation of the accumulation of junctions</p></a></li>
<li><a href='#sim_phased_unphased'><p>Individual Based Simulation of the accumulation of junctions</p></a></li>
<li><a href='#time_error'><p>Estimate the error in the time estimate</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The Breakdown of Genomic Ancestry Blocks in Hybrid Lineages</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Individual based simulations of hybridizing populations,
    where the accumulation of junctions is tracked. Furthermore,
    mathematical equations are provided to verify simulation outcomes.
    Both simulations and mathematical equations are based on Janzen
    (2018, &lt;<a href="https://doi.org/10.1101%2F058107">doi:10.1101/058107</a>&gt;) and Janzen (2020,
    &lt;<a href="https://doi.org/10.1101%2F2020.09.10.292441">doi:10.1101/2020.09.10.292441</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td>https//github.com/thijsjanzen/junctions</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/thijsjanzen/junctions/issues">https://github.com/thijsjanzen/junctions/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>RcppParallel (&ge; 5.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>nloptr, Rcpp, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>dplyr, ggplot2, knitr, magrittr, rmarkdown, testthat, tidyr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>nloptr, Rcpp, RcppParallel</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-13 14:48:31 UTC; thijsjanzen</td>
</tr>
<tr>
<td>Author:</td>
<td>Thijs Janzen [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thijs Janzen &lt;thijsjanzen@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-13 15:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='calc_k'>Calculate the limit of the number of junctions</h2><span id='topic+calc_k'></span>

<h3>Description</h3>

<p>Calculate the average number of junctions after an infinite
number of generations, provided information on the initial heterozygosity,
population size and the number of generations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_k(N = Inf, R = Inf, H_0 = 0.5, C = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_k_+3A_n">N</code></td>
<td>
<p>population size</p>
</td></tr>
<tr><td><code id="calc_k_+3A_r">R</code></td>
<td>
<p>number of markers</p>
</td></tr>
<tr><td><code id="calc_k_+3A_h_0">H_0</code></td>
<td>
<p>initial heterozygosity (at the time of admixture)</p>
</td></tr>
<tr><td><code id="calc_k_+3A_c">C</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of
the chromosome)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of junctions for at time = infinity
</p>


<h3>Examples</h3>

<pre><code class='language-R'>k &lt;-  calc_k(N = 100, R = 1000, H_0 = 0.5, C = 1)
</code></pre>

<hr>
<h2 id='calculate_mat'>Function to calculate the maximum accurate time</h2><span id='topic+calculate_mat'></span>

<h3>Description</h3>

<p>Function that calculates the maximum time after hybridization
after which the number of junctions can still be reliably used to estimate
the onset of hybridization. This is following equation 15 in
Janzen et al. 2018.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_mat(N = Inf, R = Inf, H_0 = 0.5, C = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_mat_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="calculate_mat_+3A_r">R</code></td>
<td>
<p>Number of genetic markers</p>
</td></tr>
<tr><td><code id="calculate_mat_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="calculate_mat_+3A_c">C</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of
the chromosome)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The maximum accurate time
</p>


<h3>Examples</h3>

<pre><code class='language-R'>calculate_mat(N = Inf, R = 1000, H_0 = 0.5, C = 1)
</code></pre>

<hr>
<h2 id='estimate_time'>Estimate the time since the onset of hybridization, using the number of
junctions</h2><span id='topic+estimate_time'></span>

<h3>Description</h3>

<p>Estimate the time since the onset of hybridization, following
equation 14 in Janzen et al. 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_time(J = NA, N = Inf, R = Inf, H_0 = 0.5, C = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_time_+3A_j">J</code></td>
<td>
<p>The observed number of junctions</p>
</td></tr>
<tr><td><code id="estimate_time_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="estimate_time_+3A_r">R</code></td>
<td>
<p>Number of genetic markers</p>
</td></tr>
<tr><td><code id="estimate_time_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="estimate_time_+3A_c">C</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of the
chromosome)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of generations passed since the onset of hybridization
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat("example calculate time")
J &lt;- number_of_junctions(N = 100, R = 1000, H_0 = 0.5, C = 1, t = 200)
estimate_time(J = J, N = 100, R = 1000, H_0 = 0.5, C = 1)
# should be 200 again
</code></pre>

<hr>
<h2 id='estimate_time_diploid'>estimates the time since admixture, given diploid ancestry data.</h2><span id='topic+estimate_time_diploid'></span>

<h3>Description</h3>

<p>Calculates the time since admixture, given unphased
ancestry data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_time_diploid(
  ancestry_information,
  analysis_type = "individuals",
  phased = FALSE,
  pop_size = 1000,
  freq_ancestor_1 = 0.5,
  lower_lim = 2,
  upper_lim = 2000,
  num_threads = 1,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_time_diploid_+3A_ancestry_information">ancestry_information</code></td>
<td>
<p>a matrix with five columns: column 1) indicator
of individual, column 2) indicator of chromosome, 3) location of marker in
Morgan, 4) ancestry at chromosome 5) ancestry at chromosome 2.</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_analysis_type">analysis_type</code></td>
<td>
<p>how should the data be broken down? there are multiple
options: &quot;individuals&quot; - time is inferred for each individual separately,
grouping all chromosomes together that belong to the same individual.
&quot;chromosomes&quot; - time is inferred for each chromosome separately, grouping
chromosomes together belonging from separate individuals. &quot;separate&quot; - time
is inferred for each chromosome from each individual separately, &quot;all&quot; - time
is inferred jointly for all chromosomes and individuals, grouping all
chromosomes and individuals together.</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_phased">phased</code></td>
<td>
<p>is the data phased?</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_pop_size">pop_size</code></td>
<td>
<p>population size</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_lower_lim">lower_lim</code></td>
<td>
<p>lower limit of the optimization algorithm. Increase
if the expected admixture time is relatively ancient</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_upper_lim">upper_lim</code></td>
<td>
<p>upper limit of hte optimization algorithm. If set too
large, recent admixture events can be overlooked - best to set as low
as possible.</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_num_threads">num_threads</code></td>
<td>
<p>num_threads, default is all threads. 5 threads is
recommended.</p>
</td></tr>
<tr><td><code id="estimate_time_diploid_+3A_verbose">verbose</code></td>
<td>
<p>display intermediate output? Default = FALSE</p>
</td></tr>
</table>

<hr>
<h2 id='estimate_time_haploid'>estimate time using likelihood for a single chromosome</h2><span id='topic+estimate_time_haploid'></span>

<h3>Description</h3>

<p>Estimate the time since the onset of hybridization, for a
haploid genome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_time_haploid(
  ancestry_matrix,
  N = 1000,
  freq_ancestor_1 = 0.5,
  lower_lim = 2,
  upper_lim = 1000,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_time_haploid_+3A_ancestry_matrix">ancestry_matrix</code></td>
<td>
<p>matrix with 3 columns, column 1 = chromosome,
column 2 = location in Morgan, column 3 = ancestry.</p>
</td></tr>
<tr><td><code id="estimate_time_haploid_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="estimate_time_haploid_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="estimate_time_haploid_+3A_lower_lim">lower_lim</code></td>
<td>
<p>lower limit of the optimization algorithm. Increase if the
expected admixture time is relatively ancient</p>
</td></tr>
<tr><td><code id="estimate_time_haploid_+3A_upper_lim">upper_lim</code></td>
<td>
<p>upper limit of the optimization algorithm. If set too large,
recent admixture events can be overlooked - best to set as low as possible.</p>
</td></tr>
<tr><td><code id="estimate_time_haploid_+3A_verbose">verbose</code></td>
<td>
<p>return verbose output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of generations passed since the onset of hybridization
</p>

<hr>
<h2 id='estimate_time_one_chrom'>Estimate the time since the onset of hybridization, using the observed
number of junctions, taking into account the distribution of markers on
a single chromosome</h2><span id='topic+estimate_time_one_chrom'></span>

<h3>Description</h3>

<p>Estimate the time since the onset of hybridization, following
equation 1 in Janzen et al. unpublished
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_time_one_chrom(
  J = NA,
  N = Inf,
  H_0 = 0.5,
  marker_distribution = NA,
  lower_lim = 2,
  upper_lim = 1000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_time_one_chrom_+3A_j">J</code></td>
<td>
<p>The observed number of junctions</p>
</td></tr>
<tr><td><code id="estimate_time_one_chrom_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="estimate_time_one_chrom_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="estimate_time_one_chrom_+3A_marker_distribution">marker_distribution</code></td>
<td>
<p>A vector containing the position of all markers
in Morgan.</p>
</td></tr>
<tr><td><code id="estimate_time_one_chrom_+3A_lower_lim">lower_lim</code></td>
<td>
<p>lower limit of the optimization algorithm. Increase if the
expected admixture time is relatively ancient</p>
</td></tr>
<tr><td><code id="estimate_time_one_chrom_+3A_upper_lim">upper_lim</code></td>
<td>
<p>upper limit of the optimization algorithm. If set too large,
recent admixture events can be overlooked - best to set as low as possible.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of generations passed since the onset of hybridization
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat("example estimate time one chrom")
markers &lt;- seq(from = 0, to = 1, length.out = 100)
J &lt;- number_of_junctions_markers(N = 100, H_0 = 0.5, t = 200,
marker_distribution = markers)
estimate_time_one_chrom(J = J,
                        N = 100,
                        H_0 = 0.5,
                        marker_distribution = markers) #should be 200 again
</code></pre>

<hr>
<h2 id='junctions-package'>
Extending The Theory of Junctions
</h2><span id='topic+junctions-package'></span><span id='topic+junctions'></span>

<h3>Description</h3>

<p>The theory of junctions is extended by this package by including the effect of a finite number of recombination sites along the chromosome. The package provides functions to calculate the estimated number of junctions, depending on the time since the onset of hybridization, population size, number of recombination sites, initial heterozygosity and the number of crossovers per meiosis.
</p>


<h3>Details</h3>

<p>This package provides individual based simulations in order to simulate the accumulation of junctions over time, both for chromosomes with a finite and an infinite number of recombination sites. Furthermore, the package provides mathematical tools to verify the outcomes of the individual based simulations.
Update version 2.1.0 : updated tbb::task_scheduler_init to tbb::global_control <br />
Update version 2.0.2 : simplified some tests <br />
Update version 2.0 : merged many functions with similar functionality, added vignette that provides overview of all functionality. <br />
</p>
<p>Update version 1.9 : added c++ versions of the unphased and phased likelihoods. <br />
</p>
<p>Update version 1.8 : added multithreading using the TBB library. <br />
</p>
<p>Update version 1.7 : further improved the recombination function following Hanno Hildenbrandt's suggestions <br />
</p>
<p>Update version 1.6  : improved the recombination function to run twice as fast <br />
</p>
<p>Update version 1.5.1: added option to track the true number of junctions <br />
</p>
<p>Update version 1.5: added support for inferring the time since admixture based on phased and unphased data. Also included are simulation functions to simulate appropriate data (e.g. phased and unphased). <br />
</p>
<p>Update version 1.4: added support for estimating the number of junctions, and simulating the number of junctions, under a backcrossing scheme, using the code supplied in Lavretsky et al. 2019. <br />
</p>
<p>Update version 1.3: added support for estimating the time since admixture using unphased data. <br />
Update version 1.3: added individual based simulations returning phased and unphased data. <br />
Update version 1.3: Updated entire package to Roxygen. <br />
</p>
<p>Update version 1.2: added support for estimating the expected number of junctions for arbitrarily distributed markers. <br />
</p>
<p>Update version 1.1: updated underlying random number generator for picking recombination sites. The previous generator had limited precision, which could generate duplicate recombination sites. This update fixes that <br />
</p>


<h3>Author(s)</h3>

<p>Maintainer: Thijs Janzen &lt;thijsjanzen@gmail.com&gt;
</p>


<h3>References</h3>

<p>Janzen, T. , Nolte, A. W. and Traulsen, A. (2018), The breakdown of genomic ancestry blocks in hybrid lineages given a finite number of recombination sites. Evolution, 72: 735-750. doi:10.1111/evo.13436 <br />
</p>
<p>Lavretsky, P, Janzen, T. and McCracken, KG.  (2019) Identifying hybrids &amp; the genomics of hybridization: Mallards &amp; American black ducks of Eastern North America. Ecology and Evolution 9: 3470-3490. doi:10.1002/ece3.4981 <br />
</p>

<hr>
<h2 id='log_likelihood_diploid'>calculate the log likelihood of observing diploid ancestry data.</h2><span id='topic+log_likelihood_diploid'></span>

<h3>Description</h3>

<p>Calculates the log likelihood of observing the phased data,
given the population size, initial heterozygosity and time since admixture
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_likelihood_diploid(
  local_anc_matrix,
  pop_size,
  freq_ancestor_1 = 0.5,
  t,
  phased = FALSE,
  num_threads = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log_likelihood_diploid_+3A_local_anc_matrix">local_anc_matrix</code></td>
<td>
<p>a matrix with four columns: column 1) chromosome
indicator, 2) location of marker in Morgan on respective chromosome 3)
ancestry at chromosome 4) ancestry at chromosome 2.</p>
</td></tr>
<tr><td><code id="log_likelihood_diploid_+3A_pop_size">pop_size</code></td>
<td>
<p>population size</p>
</td></tr>
<tr><td><code id="log_likelihood_diploid_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="log_likelihood_diploid_+3A_t">t</code></td>
<td>
<p>time since admixture</p>
</td></tr>
<tr><td><code id="log_likelihood_diploid_+3A_phased">phased</code></td>
<td>
<p>is the data phased or not? default is false.</p>
</td></tr>
<tr><td><code id="log_likelihood_diploid_+3A_num_threads">num_threads</code></td>
<td>
<p>number of threads, default is one thread. Set to -1 to
use all available threads.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>log likelihood
</p>

<hr>
<h2 id='log_likelihood_haploid'>log likelihood of the time since admixture for a haploid genome</h2><span id='topic+log_likelihood_haploid'></span>

<h3>Description</h3>

<p>log likelihood of the time since admixture for a set of single
chromosomes (for ex. in Yeast).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>log_likelihood_haploid(ancestry_matrix, N = 1000, freq_ancestor_1 = 0.5, t = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="log_likelihood_haploid_+3A_ancestry_matrix">ancestry_matrix</code></td>
<td>
<p>matrix with 3 columns, column 1 = chromosome,
column 2 = location in Morgan, column 3 = ancestry.</p>
</td></tr>
<tr><td><code id="log_likelihood_haploid_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="log_likelihood_haploid_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="log_likelihood_haploid_+3A_t">t</code></td>
<td>
<p>time since admixture</p>
</td></tr>
</table>


<h3>Value</h3>

<p>loglikelihood
</p>

<hr>
<h2 id='number_of_junctions'>Calculate the average number of junctions</h2><span id='topic+number_of_junctions'></span>

<h3>Description</h3>

<p>Calculate the average number of junctions in a single chromosome
after t generations, provided information on the initial heterozygosity,
population size and the number of generations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>number_of_junctions(N = Inf, R = Inf, H_0 = 0.5, C = 1, t = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="number_of_junctions_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="number_of_junctions_+3A_r">R</code></td>
<td>
<p>Number of genetic markers</p>
</td></tr>
<tr><td><code id="number_of_junctions_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="number_of_junctions_+3A_c">C</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of the
chromosome)</p>
</td></tr>
<tr><td><code id="number_of_junctions_+3A_t">t</code></td>
<td>
<p>Time since admixture</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated number of junctions at time t
</p>


<h3>Examples</h3>

<pre><code class='language-R'>jt &lt;-  number_of_junctions(N = 100, R = 1000, H_0 = 0.5, C = 1, t = 1000)
jt2 &lt;- number_of_junctions(N = 100, R = 1000, H_0 = 0.5, C = 1, t = 0:1000)
</code></pre>

<hr>
<h2 id='number_of_junctions_backcross'>Calculate the average number of junctions during backcrossing</h2><span id='topic+number_of_junctions_backcross'></span>

<h3>Description</h3>

<p>Calculate the expected number of junctions after t
generations, in a backcrossing mating scheme.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>number_of_junctions_backcross(H_0 = 0.5, C = 1, t = 100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="number_of_junctions_backcross_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="number_of_junctions_backcross_+3A_c">C</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of the
chromosome)</p>
</td></tr>
<tr><td><code id="number_of_junctions_backcross_+3A_t">t</code></td>
<td>
<p>Time since admixture</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated number of junctions at time t
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cat("example number of junctions backcross")
jt &lt;-  number_of_junctions_backcross(H_0 = 0.1, C = 1, t = 5)
</code></pre>

<hr>
<h2 id='number_of_junctions_di'>Calculate the expected number of junctions between two markers separated by
a given amount of recombination</h2><span id='topic+number_of_junctions_di'></span>

<h3>Description</h3>

<p>Calculate the expected number of junctions after t generations,
provided information on the initial heterozygosity, population size, the
number of generations since the onset of admixture and the distance between
two markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>number_of_junctions_di(N = Inf, H_0 = 0.5, t = 100, di = 1e-06)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="number_of_junctions_di_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="number_of_junctions_di_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="number_of_junctions_di_+3A_t">t</code></td>
<td>
<p>Time since admixture</p>
</td></tr>
<tr><td><code id="number_of_junctions_di_+3A_di">di</code></td>
<td>
<p>Distance between two markers in Morgan</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated number of junctions at time t
</p>


<h3>Examples</h3>

<pre><code class='language-R'>number_of_junctions_di(N = 100, H_0 = 0.5, t = 1000, di = 0.01)+
</code></pre>

<hr>
<h2 id='number_of_junctions_markers'>Calculate the expected total number of junctions in a chromosome, given the
distribution of markers</h2><span id='topic+number_of_junctions_markers'></span>

<h3>Description</h3>

<p>Calculate the expected number of junctions after t generations,
provided information on the initial heterozygosity, population size, the
number of generations since the onset of admixture and the distribution of
markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>number_of_junctions_markers(
  N = Inf,
  H_0 = 0.5,
  t = 100,
  marker_distribution = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="number_of_junctions_markers_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="number_of_junctions_markers_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="number_of_junctions_markers_+3A_t">t</code></td>
<td>
<p>Time since admixture</p>
</td></tr>
<tr><td><code id="number_of_junctions_markers_+3A_marker_distribution">marker_distribution</code></td>
<td>
<p>A vector containing the position of all markers
in Morgan.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Estimated number of observed junctions at time t
</p>


<h3>Examples</h3>

<pre><code class='language-R'>markers &lt;- seq(from = 0, to = 1, length.out = 1000)
jt &lt;-  number_of_junctions_markers(N = 100,
                                  H_0 = 0.5,
                                  t = 1000,
                                  marker_distribution = markers)
random_markers &lt;- sort(runif(1000, 0, 1))
jt2 &lt;- number_of_junctions_markers(N = 100,
                                  H_0 = 0.5,
                                  t = 1000,
                                  marker_distribution = random_markers)
</code></pre>

<hr>
<h2 id='sim_backcrossing'>Function to simulate data using a back crossing scheme</h2><span id='topic+sim_backcrossing'></span>

<h3>Description</h3>

<p>Individual based simulation of the accumulation of junctions,
under a back crossing scheme
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_backcrossing(
  population_size = 100,
  freq_ancestor_1 = 0.5,
  total_runtime = 5,
  size_in_morgan = 1,
  number_of_markers = 100,
  seed = 6,
  time_points = -1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_backcrossing_+3A_population_size">population_size</code></td>
<td>
<p>Population size</p>
</td></tr>
<tr><td><code id="sim_backcrossing_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="sim_backcrossing_+3A_total_runtime">total_runtime</code></td>
<td>
<p>Number of generations to simulate</p>
</td></tr>
<tr><td><code id="sim_backcrossing_+3A_size_in_morgan">size_in_morgan</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in
Morgan of the chromosome)</p>
</td></tr>
<tr><td><code id="sim_backcrossing_+3A_number_of_markers">number_of_markers</code></td>
<td>
<p>number of molecular markers</p>
</td></tr>
<tr><td><code id="sim_backcrossing_+3A_seed">seed</code></td>
<td>
<p>Seed of the pseudo-random number generator</p>
</td></tr>
<tr><td><code id="sim_backcrossing_+3A_time_points">time_points</code></td>
<td>
<p>vector with time points at which local ancestry has to be
recorded to be returned at the end of the simulation. If left at -1,
ancestry is recorded at every generation (computationally heavy).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with five entries: average_junctions: average number of
junctions over time, detected_junctions: average number of detected
junctions, given the markers. markers: vector with the locations of the
molecular markers, junction_distribution: distribution of junctions per
time step average_heterozygosity: average heterozygosity.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim_backcrossing(population_size = 100,
                       total_runtime = 5,
                       size_in_morgan = 1, number_of_markers = 100, seed = 6,
                       time_points = 1:5)
</code></pre>

<hr>
<h2 id='sim_fin_chrom'>Individual Based Simulation of the accumulation of junctions</h2><span id='topic+sim_fin_chrom'></span>

<h3>Description</h3>

<p>Individual based simulation of the accumulation of junctions
for a chromosome with regularly distributed markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_fin_chrom(
  pop_size = 100,
  freq_ancestor_1 = 0.5,
  total_runtime = 100,
  morgan = 1,
  seed = 42,
  R = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_fin_chrom_+3A_pop_size">pop_size</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="sim_fin_chrom_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="sim_fin_chrom_+3A_total_runtime">total_runtime</code></td>
<td>
<p>Maximum time after which the simulation is to be stopped</p>
</td></tr>
<tr><td><code id="sim_fin_chrom_+3A_morgan">morgan</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of
the chromosome)</p>
</td></tr>
<tr><td><code id="sim_fin_chrom_+3A_seed">seed</code></td>
<td>
<p>Seed of the pseudo-random number generator</p>
</td></tr>
<tr><td><code id="sim_fin_chrom_+3A_r">R</code></td>
<td>
<p>Number of regularly distributed markers</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>avgJunctions</code></td>
<td>
<p>vector of the average number of junctions at
time = [0, total_runtime]</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>cat("example sim_fin_chrom")
sim_fin_chrom(pop_size = 100, freq_ancestor_1 = 0.5,
                   total_runtime = 10, morgan = 1, seed = 42,
                   R = 100)
</code></pre>

<hr>
<h2 id='sim_inf_chrom'>Individual Based Simulation of the accumulation of junctions</h2><span id='topic+sim_inf_chrom'></span>

<h3>Description</h3>

<p>Individual based simulation of the accumulation of junctions
for a chromosome with an infinite number of recombination sites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_inf_chrom(
  pop_size = 100,
  freq_ancestor_1 = 0.5,
  total_runtime = 100,
  morgan = 1,
  markers = -1,
  seed = 42
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_inf_chrom_+3A_pop_size">pop_size</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="sim_inf_chrom_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="sim_inf_chrom_+3A_total_runtime">total_runtime</code></td>
<td>
<p>Maximum time after which the simulation is to be stopped</p>
</td></tr>
<tr><td><code id="sim_inf_chrom_+3A_morgan">morgan</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of
the chromosome)</p>
</td></tr>
<tr><td><code id="sim_inf_chrom_+3A_markers">markers</code></td>
<td>
<p>The number of genetic markers superimposed on the chromosome.
If markers is set to -1, no markers are superimposed (faster simulation)</p>
</td></tr>
<tr><td><code id="sim_inf_chrom_+3A_seed">seed</code></td>
<td>
<p>Seed of the pseudo-random number generator</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>avgJunctions</code></td>
<td>
<p>vector of the average number of junctions at
time = [0, total_runtime]</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>cat("example sim inf chrom")
v &lt;- sim_inf_chrom(pop_size = 100, freq_ancestor_1 = 0.5,
                   total_runtime = 10, morgan = 1, markers = 100,
                   seed = 42)
plot(v$avgJunctions, type = "l", xlab = "Generations",
ylab = "Number of Junctions", main = "Example Infinite Chromosome")
lines(v$detectedJunctions, col = "blue")
legend("bottomright", c("Real number","Number detected"),
       lty = 1, col = c("black", "blue"))
</code></pre>

<hr>
<h2 id='sim_phased_unphased'>Individual Based Simulation of the accumulation of junctions</h2><span id='topic+sim_phased_unphased'></span>

<h3>Description</h3>

<p>Individual based simulation of the accumulation of junctions,
returning phased and unphased data. Ancestry on both chromosomes of 10
randomly sampled individuals per generations is returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim_phased_unphased(
  pop_size = 100,
  freq_ancestor_1 = 0.5,
  total_runtime = 100,
  size_in_morgan = 1,
  markers = 100,
  time_points = -1,
  num_threads = 1,
  verbose = FALSE,
  record_true_junctions = FALSE,
  num_indiv_sampled = 10,
  coverage = 1,
  error_rate = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim_phased_unphased_+3A_pop_size">pop_size</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_freq_ancestor_1">freq_ancestor_1</code></td>
<td>
<p>Frequency of ancestor 1 at t = 0</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_total_runtime">total_runtime</code></td>
<td>
<p>Maximum time after which the simulation is to be stopped</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_size_in_morgan">size_in_morgan</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in
Morgan of the chromosome)</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_markers">markers</code></td>
<td>
<p>If a single number is provided, the number is used as the
total number of markers generated either randomly, or using a regular
distribution (a regular distribution is chosen if the number is negative). If
a vector is provided, that vector is used.</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_time_points">time_points</code></td>
<td>
<p>vector with time points at which local ancestry has to be
recorded to be returned at the end of the simulation. If left at -1,
ancestry is recorded at every generation (computationally heavy).</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_num_threads">num_threads</code></td>
<td>
<p>default is 1. -1 takes all available threads.</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_verbose">verbose</code></td>
<td>
<p>displays a progress bar</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_record_true_junctions">record_true_junctions</code></td>
<td>
<p>keep track of the true number of junctions?</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_num_indiv_sampled">num_indiv_sampled</code></td>
<td>
<p>the number of individuals sampled at each time point
to be genotyped</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_coverage">coverage</code></td>
<td>
<p>fraction of markers that can be succesfully phased</p>
</td></tr>
<tr><td><code id="sim_phased_unphased_+3A_error_rate">error_rate</code></td>
<td>
<p>fraction of markers that are erroneously
phased (e.g. swapped)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble with five columns: [time, individual, marker location,
ancestry chromosome 1, ancestry chromosome 2]
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
sim_phased_unphased(pop_size = 100, freq_ancestor_1 = 0.5,
                    total_runtime = 10, size_in_morgan = 1,
                    markers = 10, time_points = c(0, 5, 10),
                    num_threads = 1)

## End(Not run)
</code></pre>

<hr>
<h2 id='time_error'>Estimate the error in the time estimate</h2><span id='topic+time_error'></span>

<h3>Description</h3>

<p>Calculate the error in the estimate of the onset of
hybridization, following Equations 3 &amp; 4 in the Supplementary information of
Janzen et al. 2018.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>time_error(t = NA, N = Inf, R = Inf, H_0 = 0.5, C = 1, relative = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="time_error_+3A_t">t</code></td>
<td>
<p>Inferred time</p>
</td></tr>
<tr><td><code id="time_error_+3A_n">N</code></td>
<td>
<p>Population Size</p>
</td></tr>
<tr><td><code id="time_error_+3A_r">R</code></td>
<td>
<p>Number of genetic markers</p>
</td></tr>
<tr><td><code id="time_error_+3A_h_0">H_0</code></td>
<td>
<p>Frequency of heterozygosity at t = 0</p>
</td></tr>
<tr><td><code id="time_error_+3A_c">C</code></td>
<td>
<p>Mean number of crossovers per meiosis (e.g. size in Morgan of
the chromosome)</p>
</td></tr>
<tr><td><code id="time_error_+3A_relative">relative</code></td>
<td>
<p>Boolean flag, if TRUE: return the relative error, if FALSE:
return error in generations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Expected error in the time estimate
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
