<!DOCTYPE html><html><head><title>Help for package kangar00</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kangar00}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#kangar00-package'>
<p>kangar00 package</p></a></li>
<li><a href='#anno'><p>Example annotation file for three pathways.</p></a></li>
<li><a href='#calc_kernel'><p>Calculate the kernel-matrix for a pathway</p></a></li>
<li><a href='#gene_name_number+2Ccharacter-method'><p>Function to get genes names and numbers from kegg (for internal use)</p></a></li>
<li><a href='#geno'><p>Example genotypes for 50 individuals.</p></a></li>
<li><a href='#get_ana+2Cdata.frame-method'><p>Produce middle part of network kernel (for internal use)</p></a></li>
<li><a href='#get_anno+2Csnp_info+2Cpathway_info-method'><p>Annotates SNPs via genes to pathways</p></a></li>
<li><a href='#get_network_matrix'><p>Function to calculate the network <code>matrix</code> for a <code>pathway</code> object</p></a></li>
<li><a href='#gwas'><p>Example <code>GWASdata</code> object.</p></a></li>
<li><a href='#GWASdata'><p>S4 class for an object representing a Genome-wide Assocaition Study.</p></a></li>
<li><a href='#hsa04020'><p>Example <code>pathway</code> object for pathway hsa04020.</p></a></li>
<li><a href='#hsa04022_info'><p>Example <code>pathway_info</code> object for <code>pathway</code> hsa04022.</p></a></li>
<li><a href='#kernel-class'><p>An S4 class representing a kernel matrix calculated for a pathway</p></a></li>
<li><a href='#lkmt_test'><p>A function to calculate the p-values for kernel matrices.</p></a></li>
<li><a href='#lkmt-class'><p>An S4 class to represent the variance component test.</p></a></li>
<li><a href='#lkmt.net.kernel.hsa04020'><p>Example test result for the network-based <code>kernel</code> for</p>
<code>pathway</code> hsa04020.</a></li>
<li><a href='#lowrank_kernel-class'><p>An S4 class to represent a low-rank kernel for a SNPset at specified knots</p></a></li>
<li><a href='#make_psd+2Cmatrix-method'><p>Adjust network <code>matrix</code> to be positive semi-definite</p></a></li>
<li><a href='#net.kernel.hsa04020'><p>Example network-based kernel matrix for pathway hsa04020.</p></a></li>
<li><a href='#pathway'><p>An S4 class to represent a gene-gene interaction network</p></a></li>
<li><a href='#pathway_info'><p>An S4 class for an object assigning genes to pathways</p></a></li>
<li><a href='#pheno'><p>Example phenotype file for 50 individuals.</p></a></li>
<li><a href='#read_geno+2Ccharacter-method'><p>read genotype data from file to one of several available objects, which</p>
can be passed to a GWASdata object <code>GWASdata</code>.</a></li>
<li><a href='#rewire_network'><p>Rewires interactions in a <code>pathway</code>, which go through a gene not</p>
represented by any SNPs in the considered <code>GWASdata</code> dataset.</a></li>
<li><a href='#rs10243170_info'><p>Example <code>snp_info</code> object for SNP rs10243170.</p></a></li>
<li><a href='#set_names+2Cmatrix-method'><p>Helper function to translate gene numbers to names</p></a></li>
<li><a href='#set_one+2Cmatrix-method'><p>Helper function to set <code>matrix</code> entries to 0/1/-1 only</p></a></li>
<li><a href='#snp_info'><p>An S4 class for an object assigning SNP positions to rs-numbers (for internal use)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Kernel Approaches for Nonlinear Genetic Association Regression</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-05-09</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://kangar00.manitz.org/">https://kangar00.manitz.org/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/jmanitz/kangar00/issues">https://github.com/jmanitz/kangar00/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Methods to extract information on pathways, genes and various single-nucleotid polymorphisms (SNPs) from online databases. It provides functions for data preparation and evaluation of genetic influence on a binary outcome using the logistic kernel machine test (LKMT). Three different kernel functions are offered to analyze genotype information in this variance component test: A linear kernel, a size-adjusted kernel and a network-based kernel).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Collate:</td>
<td>'pathway.r' 'GWASdata.r' 'data.R' 'get_network_matrix.r'
'kernel.r' 'lkmt.r'</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, bigmemory, sqldf, CompQuadForm, data.table, lattice,
igraph</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>biomaRt, KEGGgraph, testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-09 02:48:05 UTC; nkalluru</td>
</tr>
<tr>
<td>Author:</td>
<td>Juliane Manitz [aut, cre],
  Benjamin Hofner [aut],
  Stefanie Friedrichs [aut],
  Patricia Burger [aut],
  Ngoc Thuy Ha [aut],
  Saskia Freytag [ctb],
  Heike Bickeboeller [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juliane Manitz &lt;r@manitz.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-09 03:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='kangar00-package'>
kangar00 package
</h2><span id='topic+kangar00-package'></span><span id='topic+kangar00'></span>

<h3>Description</h3>

<p>This package includes methods to extract information on pathways, genes and SNPs from online databases and to evaluate these data using the logistic kernel machine test (LKMT) (Liu et al. 2008).
</p>
<p>We defined SNP sets representing genes and whole pathways using knowledge on
gene membership and interaction from the Kyoto Encyclopedia of Genes and 
Genomes (KEGG) database (Kanehisa et al. 2014). SNPs are mapped to genes via 
base pair positions of SNPs and transcript start and end points of genes as 
documented in the Ensemble database (Cunningham et al. 2015). 
</p>
<p>In the LKMT, we employed the linear kernel (Wu et al. 2010) as well as two more 
advanced kernels,  adjusting for size bias in the number of SNPs and genes in a
pathway (size-adjusted kernels), and incorporating the network structure of 
genes within the pathway (pathway kernels), respectively (Freytag et al. 2012, 2014). P-values are derived in a variance component test using a moment matching method (Schaid, 2010) or Davies' algorithm (Davies, 1980). 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> kangar00</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-08-07</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Juliane Manitz [aut], Stefanie Friedrichs [aut], Patricia Burger [aut], 
Benjamin Hofner [aut], Ngoc Thuy Ha [aut], Saskia Freytag [ctb],
Heike Bickeboeller [ctb]  <br />
Maintainer: Juliane Manitz &lt;r@manitz.de&gt;
</p>


<h3>References</h3>


<ul>
<li><p>	Cunningham F, Ridwan Amode M, Barrell D, et al. Ensembl 2015. Nucleic Acids Research 2015 43 Database issue:D662-D669  <br />
</p>
</li>
<li><p>	Davies R: Algorithm as 155: the distribution of a linear combination of chi-2 random variables. J R Stat Soc Ser C 1980, 29:323-333.  <br />
</p>
</li>
<li><p> Freytag S, Bickeboeller H, Amos CI, Kneib T, Schlather M: A Novel Kernel for Correcting Size Bias in the Logistic Kernel Machine Test with an Application to Rheumatoid Arthritis. Hum Hered. 2012, 74(2):97-108. <br />
</p>
</li>
<li><p> Freytag S, Manitz J, Schlather M, Kneib T, Amos CI, Risch A, Chang-Claude J, Heinrich J, Bickeboeller H: A network-based kernel machine test for the identification of risk pathways in genome-wide association studies. Hum Hered. 2013, 76(2):64-75. <br />
</p>
</li>
<li><p> Friedrichs S, Manitz J, Burger P, Amos CI, Risch A, Chang-Claude JC,   Wichmann HE, Kneib T, Bickeboeller H, Hofner B: Pathway-Based Kernel Boosting for the Analysis of Genome-Wide Association Studies. Computational and Mathematical Methods in Medicine. 2017(6742763), 1-17. doi:10.1155/2017/6742763.<br />
</p>
</li>
<li><p> Kanehisa, M., Goto, S., Sato, Y., Kawashima, M., Furumichi, M., and Tanabe, M.; Data, information, knowledge and principle: back to metabolism in KEGG. Nucleic Acids Res. 42, D199-D205 (2014). <br />
</p>
</li>
<li><p> Liu D, Ghosh D, Lin X. Estimation and testing for the effect of a genetic pathway on a disease outcome using logistic kernel machine regression via logistic mixed models. BMC Bioinformatics. 2008 9:292. <br /> 
</p>
</li>
<li><p> Schaid DJ: Genomic similarity and kernel methods I: advancements by building on mathematical and statistical foundations. Hum Hered 2010, 70:109-131.   <br />
</p>
</li>
<li><p> Wu MC, Kraft P, Epstein MP, Taylor DM, Chanock SJ, Hunter DJ, Lin X: Powerful SNP-Set Analysis for Case-Control Genome-Wide Association Studies. Am J Hum Genet 2010, 86:929-42
</p>
</li></ul>


<hr>
<h2 id='anno'>Example annotation file for three pathways.</h2><span id='topic+anno'></span>

<h3>Description</h3>

<p>A dataset containing an annotation example for 4056 SNPs in three different 
pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(anno)
</code></pre>


<h3>Format</h3>

<p>A <code>data frame</code> with 4056 rows and 5 variables:
</p>

<dl>
<dt>pathway</dt><dd><p>includes KEGG identifiers of three example pathways</p>
</dd>
<dt>gene</dt><dd><p>names of genes in the pathways</p>
</dd>
<dt>chr</dt><dd><p>specifies the chromosome</p>
</dd>
<dt>snp</dt><dd><p>includes rs-numbers of example SNPs</p>
</dd>
<dt>position</dt><dd><p>gives positions of example SNPs</p>
</dd>
</dl>



<h3>Source</h3>

<p>simulated data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(anno)
head(anno)
# create gwas object
data(pheno)
data(geno)
gwas &lt;- new('GWASdata', pheno=pheno, geno=geno, anno=anno, desc="some study") 
</code></pre>

<hr>
<h2 id='calc_kernel'>Calculate the kernel-matrix for a pathway</h2><span id='topic+calc_kernel'></span><span id='topic+calc_kernel+2CGWASdata-method'></span><span id='topic+lin_kernel+2CGWASdata-method'></span><span id='topic+lin_kernel'></span><span id='topic+sia_kernel+2CGWASdata-method'></span><span id='topic+sia_kernel'></span><span id='topic+net_kernel+2CGWASdata-method'></span><span id='topic+net_kernel'></span>

<h3>Description</h3>

<p>Uses individuals' genotypes to create a <code><a href="#topic+kernel">kernel</a></code> object including 
the calculated kernel <code>matrix</code> for a specific <code><a href="#topic+pathway">pathway</a></code>.
Each numeric value within this <code>matrix</code> is calculated
from two individuals' genotypevectors of the SNPs within 
the <code><a href="#topic+pathway">pathway</a></code> by a kernel function. It can be interpreted as the genetic 
similiarity of the individuals. Association between the <code><a href="#topic+pathway">pathway</a></code> and a 
binary phenotype (case-control status) can be evaluated
in the logistic kernel machine test, based on the <code><a href="#topic+kernel">kernel</a></code> object. 
Three kernel functions are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'GWASdata'
calc_kernel(
  object,
  pathway,
  knots = NULL,
  type = c("lin", "sia", "net"),
  calculation = c("cpu", "gpu"),
  ...
)

## S4 method for signature 'GWASdata'
lin_kernel(object, pathway, knots = NULL, calculation = c("cpu", "gpu"), ...)

## S4 method for signature 'GWASdata'
sia_kernel(object, pathway, knots = NULL, calculation = c("cpu", "gpu"), ...)

## S4 method for signature 'GWASdata'
net_kernel(object, pathway, knots = NULL, calculation = c("cpu", "gpu"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_kernel_+3A_object">object</code></td>
<td>
<p><code>GWASdata</code> object containing the genotypes of the 
individuals for which a <code><a href="#topic+kernel">kernel</a></code> will be calculated.</p>
</td></tr>
<tr><td><code id="calc_kernel_+3A_pathway">pathway</code></td>
<td>
<p>object of the class <code><a href="#topic+pathway">pathway</a></code> specifying the SNP set 
for which a <code><a href="#topic+kernel">kernel</a></code> will be calculated.</p>
</td></tr>
<tr><td><code id="calc_kernel_+3A_knots">knots</code></td>
<td>
<p><code>GWASdata</code> object, if specified a <code><a href="#topic+kernel">kernel</a></code>
will be computed.</p>
</td></tr>
<tr><td><code id="calc_kernel_+3A_type">type</code></td>
<td>
<p><code>character</code> indicating the <code><a href="#topic+kernel">kernel</a></code> type: 
Use <code>'lin'</code> to specify the linear kernel, <code>'sia'</code> for the 
size-adjusted or <code>'net'</code> for the network-based kernel.</p>
</td></tr>
<tr><td><code id="calc_kernel_+3A_calculation">calculation</code></td>
<td>
<p><code>character</code> specifying if the kernel <code>matrix</code> 
is computed on CPU or GPU.</p>
</td></tr>
<tr><td><code id="calc_kernel_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code><a href="#topic+kernel">kernel</a></code> computations.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Different types of kernels can be constructed:
</p>

<ul>
<li> <p><code>type='lin'</code> creates the linear kernel assuming additive SNP 
effects to be evaluated in the logistic kernel machine test.
</p>
</li>
<li> <p><code>type='sia'</code> calculates the size-adjusted kernel which takes 
into consideration the numbers of SNPs and genes in a <code><a href="#topic+pathway">pathway</a></code> 
to correct for size bias.
</p>
</li>
<li> <p><code>type='net'</code> calculates the network-based kernel. Here not only information on gene membership and gene/pathway size in number of SNPs is incorporated, but also the interaction structure of genes in the <code><a href="#topic+pathway">pathway</a></code>.
</p>
</li></ul>

<p>For more details, check the references.
</p>


<h3>Value</h3>

<p>Returns an object of class <code><a href="#topic+kernel">kernel</a></code>, including the similarity 
<code>matrix</code> of the <code><a href="#topic+pathway">pathway</a></code> for the considered individuals. <br />
If <code>knots</code> are specified low-rank kernel of class a <code>lowrank_kernel</code> 
will be returned, which is not necessarily quadratic and symmetric.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>lin_kernel(GWASdata)</code>: 
</p>
</li>
<li> <p><code>sia_kernel(GWASdata)</code>: 
</p>
</li>
<li> <p><code>net_kernel(GWASdata)</code>: 
</p>
</li></ul>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs, Juliane Manitz
</p>


<h3>References</h3>


<ul>
<li><p> Wu MC, Kraft P, Epstein MP, Taylor DM, Chanock SJ, Hunter DJ, Lin X Powerful SNP-Set Analysis for Case-Control Genome-Wide Association Studies. Am J Hum Genet 2010, 86:929-42
</p>
</li>
<li><p> Freytag S, Bickeboeller H, Amos CI, Kneib T, Schlather M: A Novel Kernel for Correcting Size Bias in the Logistic Kernel Machine Test with an Application to Rheumatoid Arthritis. Hum Hered. 2012, 74(2):97-108.
</p>
</li>
<li><p> Freytag S, Manitz J, Schlather M, Kneib T, Amos CI, Risch A, Chang-Claude J, Heinrich J, Bickeboeller H: A network-based kernel machine test for the identification of risk pathways in genome-wide association studies. Hum Hered. 2013, 76(2):64-75.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+kernel-class">kernel-class</a></code>,<code><a href="#topic+pathway">pathway</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gwas)
data(hsa04020)
lin_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='lin', calculation='cpu')
summary(lin_kernel)
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')
summary(net_kernel)

</code></pre>

<hr>
<h2 id='gene_name_number+2Ccharacter-method'>Function to get genes names and numbers from kegg (for internal use)</h2><span id='topic+gene_name_number+2Ccharacter-method'></span>

<h3>Description</h3>

<p>This function extracts for a particular <code><a href="#topic+pathway">pathway</a></code> all genes and 
the numbers they are represented with in the KEGG network from the 
corresponding KGML pathway file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'character'
gene_name_number(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gene_name_number+2B2Ccharacter-method_+3A_x">x</code></td>
<td>
<p>A <code>character</code> hsa identifier of the pathway for which gene 
infomation should be extracted as used in KEGG database ('hsa00100').</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> listing the genes included in the pathway with 
their names as well as numbers used in KEGG database.
</p>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs, Patricia Burger
</p>

<hr>
<h2 id='geno'>Example genotypes for 50 individuals.</h2><span id='topic+geno'></span>

<h3>Description</h3>

<p>A matrix containing example genotypes for 4056 SNPs of 50 individuals. 
Column names give the rs-numbers of 4056 example SNPs, row names the 
identifiers of 50 example individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(geno)
</code></pre>


<h3>Format</h3>

<p>A <code>matrix</code> with 5 rows and 4056 columns:
</p>

<p>each entry in the matrix represents a simulated minor allele count for the
corresponding SNP and individual.

</p>


<h3>Source</h3>

<p>simulated data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(geno)
head(geno)
# create gwas object
data(pheno)
data(anno)
gwas &lt;- new('GWASdata', pheno=pheno, geno=geno, anno=anno, desc="some study") 
</code></pre>

<hr>
<h2 id='get_ana+2Cdata.frame-method'>Produce middle part of network kernel (for internal use)</h2><span id='topic+get_ana+2Cdata.frame-method'></span>

<h3>Description</h3>

<p>Produce middle part of network kernel (for internal use)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'data.frame'
get_ana(x, SNPset, pathway)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_ana+2B2Cdata.frame-method_+3A_x">x</code></td>
<td>
<p><code>data.frame</code> with annotation information as returned from
<code>get_anno</code></p>
</td></tr>
<tr><td><code id="get_ana+2B2Cdata.frame-method_+3A_snpset">SNPset</code></td>
<td>
<p><code>vector</code> with SNPs to be analyzed</p>
</td></tr>
<tr><td><code id="get_ana+2B2Cdata.frame-method_+3A_pathway">pathway</code></td>
<td>
<p><code>pathway</code> object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrix ANA' for inner part of network kernel
</p>


<h3>Author(s)</h3>

<p>Juliane Manitz, Saskia Freytag, Stefanie Friedrichs
</p>

<hr>
<h2 id='get_anno+2Csnp_info+2Cpathway_info-method'>Annotates SNPs via genes to pathways</h2><span id='topic+get_anno+2Csnp_info+2Cpathway_info-method'></span><span id='topic+get_anno'></span>

<h3>Description</h3>

<p>A function to create the annotation for a <code><a href="#topic+GWASdata">GWASdata</a></code> object. 
It combines a <code><a href="#topic+snp_info">snp_info</a></code> and a <code><a href="#topic+pathway_info">pathway_info</a></code>
object into an annotation <code>data.frame</code> used for <code><a href="#topic+pathway">pathway</a></code> 
analysis on GWAS. SNPs are assigned to pathways via gene membership.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'snp_info,pathway_info'
get_anno(object1, object2, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_anno+2B2Csnp_info+2B2Cpathway_info-method_+3A_object1">object1</code></td>
<td>
<p>A <code><a href="#topic+snp_info">snp_info</a></code> object with SNP information as returned
by the <code><a href="#topic+snp_info">snp_info</a></code> function. The included <code>data frame</code> contains
the columns 'chr', 'position' and 'snp'.</p>
</td></tr>
<tr><td><code id="get_anno+2B2Csnp_info+2B2Cpathway_info-method_+3A_object2">object2</code></td>
<td>
<p>A <code><a href="#topic+pathway_info">pathway_info</a></code> object with information on genes
contained in pathways. It is created by the <code><a href="#topic+pathway_info">pathway_info</a></code> 
function and contains a <code>data frame</code> with columns 
'pathway', 'gene_start', 'gene_end', 'chr', 'gene'.</p>
</td></tr>
<tr><td><code id="get_anno+2B2Csnp_info+2B2Cpathway_info-method_+3A_...">...</code></td>
<td>
<p>further argdata(hsa04020)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> mapping SNPs to genes and genes to
pathways. It includes the columns 'pathway', 'gene', 'chr', 'snp' and 
'position'.
</p>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs, Saskia Freytag, Ngoc-Thuy Ha
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snp_info">snp_info</a></code>, <code><a href="#topic+pathway_info">pathway_info</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hsa04022_info)  # pathway_info('hsa04020')
data(rs10243170_info)# snp_info("rs10243170")
get_anno(rs10243170_info, hsa04022_info)

</code></pre>

<hr>
<h2 id='get_network_matrix'>Function to calculate the network <code>matrix</code> for a <code><a href="#topic+pathway">pathway</a></code> object</h2><span id='topic+get_network_matrix'></span><span id='topic+get_network_matrix+2Cpathway-method'></span>

<h3>Description</h3>

<p><code>get_network_matrix</code> creates the adjacency matrix representing the gene-gene interaction structure within a particular <code><a href="#topic+pathway">pathway</a></code>. Note that a 
KEGG kgml file is downloaded and saved in the working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'pathway'
get_network_matrix(object, directed = TRUE, method = "auto")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_network_matrix_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+pathway">pathway</a></code> object identifying the pathway for which gene interaction infomation should be extracted. Here, KEGG IDs of format 'hsa00100' are used and information is downloaded from the KEGG database.</p>
</td></tr>
<tr><td><code id="get_network_matrix_+3A_directed">directed</code></td>
<td>
<p>A <code>logical</code> argument, stating whether the network matrix 
should return directed (<code>TRUE</code>) or undirected (<code>FALSE</code>) links.</p>
</td></tr>
<tr><td><code id="get_network_matrix_+3A_method">method</code></td>
<td>
<p>Download method to be used for downloading files, passed to via <code>KEGGgraph::retrieveKGML</code> to <code>utils::download.file</code> function. Currently supports <code>'auto'</code> (default), <code>'internal'</code>, <code>'wininet'</code> (Windows only), <code>'libcurl'</code>, <code>'wget'</code> and <code>'curl'</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>get_network_matrix</code> returns the modified <code><a href="#topic+pathway">pathway</a></code> object, where the slots <code>adj</code> and <code>sign</code> are altered according to the downloaded information in the KEGG kgml file.
</p>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs, Patricia Burger, Juliane Manitz
</p>

<hr>
<h2 id='gwas'>Example <code><a href="#topic+GWASdata">GWASdata</a></code> object.</h2><span id='topic+gwas'></span>

<h3>Description</h3>

<p>An object of type GWASdata containing the example files for 
annotation, phenotypes and genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gwas)
</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+GWASdata">GWASdata</a></code>:
</p>

<dl>
<dt>geno</dt><dd><p>contains example genotypes</p>
</dd>
<dt>anno</dt><dd><p>example annotation for three pathways</p>
</dd>
<dt>pheno</dt><dd><p>exemplary phenotypes for all 'genotyped' individuals</p>
</dd>
<dt>desc</dt><dd><p>a description of the GWAS study, here 'example study'</p>
</dd>
</dl>



<h3>Source</h3>

<p>simulated data
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create gwas object
data(pheno)
data(geno)
data(anno)
gwas &lt;- new('GWASdata', pheno=pheno, geno=geno, anno=anno, desc="some study") 
</code></pre>

<hr>
<h2 id='GWASdata'>S4 class for an object representing a Genome-wide Assocaition Study.</h2><span id='topic+GWASdata'></span><span id='topic+GWASdata+2CANY-method'></span><span id='topic+show+2CGWASdata-method'></span><span id='topic+summary+2CGWASdata-method'></span><span id='topic+GeneSNPsize+2CGWASdata-method'></span><span id='topic+GeneSNPsize'></span>

<h3>Description</h3>

<p>S4 class for an object representing a Genome-wide Assocaition Study.
</p>
<p><code>'GWASdata'</code> is a GWASdata object constructor.
</p>
<p><code>show</code> displays basic information on <code><a href="#topic+GWASdata">GWASdata</a></code> object
</p>
<p><code>summary</code> summarizes the content of a <code><a href="#topic+GWASdata">GWASdata</a></code> object 
and gives an overview about the information included in a 
<code><a href="#topic+GWASdata">GWASdata</a></code> object. Summary statistics for phenotype and genotype 
data are calculated.
</p>
<p><code>GeneSNPsize</code> creates a <code>data.frame</code> of <code><a href="#topic+pathway">pathway</a></code>
names with numbers of snps and genes in each <code><a href="#topic+pathway">pathway</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GWASdata(object, ...)

## S4 method for signature 'ANY'
GWASdata(geno, anno, pheno = NULL, desc = "")

## S4 method for signature 'GWASdata'
show(object)

## S4 method for signature 'GWASdata'
summary(object)

## S4 method for signature 'GWASdata'
GeneSNPsize(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GWASdata_+3A_object">object</code></td>
<td>
<p>A <code><a href="#topic+GWASdata">GWASdata</a></code> object.</p>
</td></tr>
<tr><td><code id="GWASdata_+3A_...">...</code></td>
<td>
<p>Further arguments can be added to the function.</p>
</td></tr>
<tr><td><code id="GWASdata_+3A_geno">geno</code></td>
<td>
<p>An object of any type, including the genotype information.</p>
</td></tr>
<tr><td><code id="GWASdata_+3A_anno">anno</code></td>
<td>
<p>A <code>data.frame</code> containing the annotation file for the 
<code>GWASdata</code> object.</p>
</td></tr>
<tr><td><code id="GWASdata_+3A_pheno">pheno</code></td>
<td>
<p>A <code>data.frame</code> specifying individual IDs, phenotypes and
covariates to be included in the regression model.</p>
</td></tr>
<tr><td><code id="GWASdata_+3A_desc">desc</code></td>
<td>
<p>A <code>character</code> giving the GWAS description, e.g. name of study.</p>
</td></tr>
</table>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>GeneSNPsize(GWASdata)</code>: creates a <code>data.frame</code> of <code><a href="#topic+pathway">pathway</a></code> names with numbers 
of snps and genes in each pathway.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>geno</code></dt><dd><p>An object of any type, including genotype information. The format
needs to be one line per individual and on colum per SNP in minor-allele 
coding (0,1,2). Other values between 0 and 2, as from impute dosages, are 
allowed. Missing values must be imputed prior to creation of a <code>GWASdata</code> object.</p>
</dd>
<dt><code>anno</code></dt><dd><p>A <code>data.frame</code> mapping SNPs to genes and genes to
pathways. Needs to include the columns 'pathway' (pathway ID, e.g. hsa
number from KEGG database), 'gene' (gene name (hgnc_symbol)), 'chr'
(chromosome), 'snp' (rsnumber) and 'position' (base pair position of SNP).</p>
</dd>
<dt><code>pheno</code></dt><dd><p>A <code>data.frame</code> specifying individual IDs, phenotypes and
covariates to be included in the regression model e.g. ID, pheno, sex,
pack.years. Note: IDs have to be in the first column!</p>
</dd>
<dt><code>desc</code></dt><dd><p>A <code>character</code> giving the GWAS description, e.g. name of study.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Juliane Manitz, Stefanie Friedrichs
</p>


<h3>Examples</h3>

<pre><code class='language-R'># create gwas data object
data(pheno)
data(geno)
data(anno)
gwas &lt;- new('GWASdata', pheno=pheno, geno=geno, anno=anno, desc="some study") 
# show and summary methods
gwas
summary(gwas)
# SNPs and genes in pathway
GeneSNPsize(gwas)
</code></pre>

<hr>
<h2 id='hsa04020'>Example <code><a href="#topic+pathway">pathway</a></code> object for pathway hsa04020.</h2><span id='topic+hsa04020'></span>

<h3>Description</h3>

<p>An object of class <code><a href="#topic+pathway">pathway</a></code> for the pathway with KEGG 
identifier hsa04020.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hsa04020)
</code></pre>


<h3>Format</h3>

<p>A <code><a href="#topic+pathway">pathway</a></code> object including 180 genes.
</p>

<dl>
<dt>id</dt><dd><p>KEGG identifier of the example pathways</p>
</dd>
<dt>adj</dt><dd><p>gives the quadratic adjacency <code>matrix</code> for the pathway and with 
that the network topology. Matrix dimensions equal the number of genes in the 
pathway</p>
</dd>
<dt>sign</dt><dd><p>includes a <code>vector</code> of signs to distinguish activations and 
inhibitions in the adjacency <code>matrix</code> </p>
</dd>
</dl>



<h3>Source</h3>

<p>simulated data and Ensembl extract
</p>

<hr>
<h2 id='hsa04022_info'>Example <code><a href="#topic+pathway_info">pathway_info</a></code> object for <code><a href="#topic+pathway">pathway</a></code> hsa04022.</h2><span id='topic+hsa04022_info'></span>

<h3>Description</h3>

<p>An object of class <code><a href="#topic+pathway_info">pathway_info</a></code> for the <code><a href="#topic+pathway">pathway</a></code>
with KEGG identifier hsa04020.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(hsa04022_info)
</code></pre>


<h3>Format</h3>

<p>A <code><a href="#topic+pathway_info">pathway_info</a></code> object including information on 163 genes.
</p>

<dl>
<dt>info</dt><dd><p>a <code>data frame</code> including information on genes included in 
pathway. Has columns 'pathway', 'gene_start', 'gene_end', 'chr', and 'gene'</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ensembl extract
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
pathway_info('hsa04020')

## End(Not run)
</code></pre>

<hr>
<h2 id='kernel-class'>An S4 class representing a kernel matrix calculated for a pathway</h2><span id='topic+kernel-class'></span><span id='topic+kernel'></span><span id='topic+show+2Ckernel-method'></span><span id='topic+summary+2Ckernel-method'></span><span id='topic+summary+2Ckernel+2CANY-method'></span><span id='topic+plot+2Ckernel+2Cmissing-method'></span><span id='topic+plot+2Ckernel+2CANY-method'></span>

<h3>Description</h3>

<p>An S4 class representing a kernel matrix calculated for a pathway
</p>
<p><code>show</code> displays the <code>kernel</code> object briefly
</p>
<p><code>summary</code> generates a <code>kernel</code> object summary including the number of 
individuals and genes for the <code><a href="#topic+pathway">pathway</a></code>
</p>
<p><code>plot</code> creates an image plot of a <code>kernel</code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'kernel'
show(object)

## S4 method for signature 'kernel'
summary(object)

## S4 method for signature 'kernel,missing'
plot(x, y = NA, hclust = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kernel-class_+3A_object">object</code></td>
<td>
<p>An object of class <code>kernel</code></p>
</td></tr>
<tr><td><code id="kernel-class_+3A_x">x</code></td>
<td>
<p>the <code>kernel</code> object to be plotted.</p>
</td></tr>
<tr><td><code id="kernel-class_+3A_y">y</code></td>
<td>
<p>missing (placeholder).</p>
</td></tr>
<tr><td><code id="kernel-class_+3A_hclust">hclust</code></td>
<td>
<p><code>logical</code>, indicating whether a dendrogram should be added.</p>
</td></tr>
<tr><td><code id="kernel-class_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to the function.</p>
</td></tr>
</table>


<h3>Slots</h3>


<dl>
<dt><code>type</code></dt><dd><p>A <code>character</code> representing the kernel type: Use 
<code>'lin'</code> for linear kernel, <code>'sia'</code> for the size-adjusted 
or <code>'net'</code> for the network-based kernel.</p>
</dd>
<dt><code>kernel</code></dt><dd><p>A kernel <code>matrix</code> of dimension equal to the number of individuals</p>
</dd>
<dt><code>pathway</code></dt><dd><p>A <code><a href="#topic+pathway">pathway</a></code> object</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Juliane Manitz
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gwas)
data(hsa04020)
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')

show(net_kernel)
summary(net_kernel)
plot(net_kernel, hclust=TRUE)

</code></pre>

<hr>
<h2 id='lkmt_test'>A function to calculate the p-values for kernel matrices.</h2><span id='topic+lkmt_test'></span><span id='topic+score_test+2Cmatrix-method'></span><span id='topic+davies_test+2Cmatrix-method'></span>

<h3>Description</h3>

<p>For parameter <code>'satt'</code> a pathway's influence on the probability of 
beeing a case is evaluated in the logistic kernel machine test and p-values 
are determined using a Sattherthwaite approximation as described by Dan Schaid.
</p>
<p>For parameter <code>'davies'</code> a pathways influence on the probability
of beeing a case is evaluated using the p-value calculation method described 
by Davies. Here the function <code><a href="CompQuadForm.html#topic+davies">davies</a></code> from package 
<span class="pkg">CompQuadForm</span> is used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lkmt_test(formula, kernel, GWASdata, method = c("satt", "davies"), ...)

## S4 method for signature 'matrix'
score_test(x1, x2)

## S4 method for signature 'matrix'
davies_test(x1, x2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lkmt_test_+3A_formula">formula</code></td>
<td>
<p>The formula to be used for the regression nullmodel.</p>
</td></tr>
<tr><td><code id="lkmt_test_+3A_kernel">kernel</code></td>
<td>
<p>An object of class <code>kernel</code> including the pathway 
representing kernel-matrix based on which the test statistic will be calculated.</p>
</td></tr>
<tr><td><code id="lkmt_test_+3A_gwasdata">GWASdata</code></td>
<td>
<p>A <code>GWASdata</code> object stating the data used in analysis.</p>
</td></tr>
<tr><td><code id="lkmt_test_+3A_method">method</code></td>
<td>
<p>A <code>character</code> specifying which method will be used for 
p-value calculation. Available are <code>'satt'</code> for the Satterthwaite 
approximation and <code>'davies'</code> for Davies' algorithm. For more details 
see the references.</p>
</td></tr>
<tr><td><code id="lkmt_test_+3A_...">...</code></td>
<td>
<p>Further arguments can be given to the function.</p>
</td></tr>
<tr><td><code id="lkmt_test_+3A_x1">x1</code></td>
<td>
<p>A <code><a href="#topic+matrix">matrix</a></code> which is the
similarity matrix calculated for the pathway to be tested.</p>
</td></tr>
<tr><td><code id="lkmt_test_+3A_x2">x2</code></td>
<td>
<p>An <code>lm</code> or <code>glm</code> object of the nullmodel with fixed 
effects covariates included, but no genetic random effects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>lkmt</code> object including the following test results
</p>

<ul>
<li><p> The formula of the regression nullmodel used in the variance component test.
</p>
</li>
<li><p> An object of class <code><a href="#topic+kernel">kernel</a></code> including the similarity matrix of the individuals based on which the pathways influence is evaluated.
</p>
</li>
<li><p> An object of class <code><a href="#topic+GWASdata">GWASdata</a></code> stating the data on which the test was conducted.
</p>
</li>
<li><p> statistic A <code>vector</code> giving the value of the variance component test statistic.
</p>
</li>
<li><p> df A <code>vector</code> giving the number of degrees of freedom.
</p>
</li>
<li><p> p.value A <code>vector</code> giving the p-value calculated for the pathway in the variance component test.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Stefanie Friedrichs, Juliane Manitz
</p>


<h3>References</h3>

<p>For details on the variance component test 
</p>

<ul>
<li><p> Wu MC, Kraft P, Epstein MP, Taylor DM, Chanock SJ, Hunter DJ, Lin X: Powerful SNP-Set Analysis for Case-Control Genome-Wide Association Studies. Am J Hum Genet 2010, 86:929-42
</p>
</li>
<li><p>  Liu D, Lin X, Ghosh D: Semiparametric regression of multidimensional genetic pathway data: least-squares kernel machines and linear mixed models. Biometrics 2007, 63(4):1079-88.
</p>
</li></ul>

<p>For details on the p-value calculation see
</p>

<ul>
<li><p> Schaid DJ: Genomic Similarity and Kernel Methods I: Advancements by Building on Mathematical and Statistical Foundations. Hum Hered 2010, 70:109-31
</p>
</li></ul>


<ul>
<li><p> Davies R: Algorithm as 155: the distribution of a linear combination of chi-2 random variables. J R Stat Soc Ser C 1980, 29:323-333.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(hsa04020)
data(gwas)
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')
lkmt_test(pheno ~ sex + age, net_kernel, gwas, method='satt')
</code></pre>

<hr>
<h2 id='lkmt-class'>An S4 class to represent the variance component test.</h2><span id='topic+lkmt-class'></span><span id='topic+lkmt'></span><span id='topic+show+2Clkmt-method'></span><span id='topic+show+2CGWASdata+2CANY-method'></span><span id='topic+summary+2Clkmt-method'></span><span id='topic+summary+2CGWASdata+2CANY-method'></span>

<h3>Description</h3>

<p>An S4 class to represent the variance component test.
</p>
<p><code>show</code> displays basic information on <code>lkmt</code> object
</p>
<p><code>summary</code> generates a <code>lkmt</code> object summary including the used kernel, pathway and the test result
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'lkmt'
show(object)

## S4 method for signature 'lkmt'
summary(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lkmt-class_+3A_object">object</code></td>
<td>
<p>An object of class <code>lkmt</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>show</code> Basic information on <code>lkmt</code> object.
</p>
<p><code>summary</code> Summarized information on <code>lkmt</code> object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>formula</code></dt><dd><p>A formula stating the regression nullmodel that will be used in
the variance component test.</p>
</dd>
<dt><code>kernel</code></dt><dd><p>An object of class <code><a href="#topic+kernel">kernel</a></code> representing the similarity
matrix of the individuals based on which the pathways influence is evaluated.</p>
</dd>
<dt><code>GWASdata</code></dt><dd><p>An object of class <code><a href="#topic+GWASdata">GWASdata</a></code> including the data
on which the test is conducted.</p>
</dd>
<dt><code>statistic</code></dt><dd><p>A <code>vector</code> giving the value of the variance component
test statistic.</p>
</dd>
<dt><code>df</code></dt><dd><p>A <code>vector</code> containing the number of degrees of freedom.</p>
</dd>
<dt><code>p.value</code></dt><dd><p>A <code>vector</code> giving the p-value calculated for the <code><a href="#topic+pathway">pathway</a></code> object considered in the variance component test.
</p>
<p>For details on the variance component test see the references.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Juliane Manitz, Stefanie Friedrichs
</p>


<h3>References</h3>


<ul>
<li><p> Liu D, Lin X, Ghosh D: Semiparametric regression of multidimensional genetic pathway data: least-squares kernel machines and linear mixed models. Biometrics 2007, 63(4):1079-88.
</p>
</li>
<li><p> Wu MC, Kraft P, Epstein MP, Taylor DM, Chanock SJ, Hunter DJ, Lin X: Powerful SNP-Set Analysis for Case-Control Genome-Wide Association Studies. Am J Hum Genet 2010, 86:929-42
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(hsa04020)
data(gwas)
# compute kernel
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')
# perform LKMT 
res &lt;- lkmt_test(pheno ~ sex + age, net_kernel, gwas, method='satt')
# show and summary methods
show(res)
summary(res)
# summary method
summary(lkmt.net.kernel.hsa04020)
</code></pre>

<hr>
<h2 id='lkmt.net.kernel.hsa04020'>Example test result for the network-based <code><a href="#topic+kernel">kernel</a></code> for 
<code><a href="#topic+pathway">pathway</a></code> hsa04020.</h2><span id='topic+lkmt.net.kernel.hsa04020'></span>

<h3>Description</h3>

<p>An object of class <code><a href="#topic+lkmt">lkmt</a></code> containing exemplary test results for an 
application of the logistic kernel machine test, derived from the example data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(lkmt.net.kernel.hsa04020)
</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+lkmt">lkmt</a></code> for the network-based  
<code><a href="#topic+kernel">kernel</a></code> and the <code><a href="#topic+pathway">pathway</a></code> hsa04020.
</p>

<dl>
<dt>formular</dt><dd><p>gives a formular defining the nullmodel used in the 
logistic kernel machine test</p>
</dd>
<dt>kernel</dt><dd><p>includes the <code><a href="#topic+kernel">kernel</a></code> object of the 
<code><a href="#topic+pathway">pathway</a></code> to be evaluated</p>
</dd>
<dt>GWASdata</dt><dd><p>gives the <code><a href="#topic+GWASdata">GWASdata</a></code> object including the study data 
considered in testing</p>
</dd>
<dt>statistic</dt><dd><p>gives the value of the test statistic</p>
</dd>
<dt>df</dt><dd><p>specifies the degrees of freedom</p>
</dd>
<dt>p.value</dt><dd><p>includes teh p-value resulting from the test</p>
</dd>
</dl>



<h3>Source</h3>

<p>simulated data and Ensembl extract
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hsa04020)
data(gwas)
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')
lkmt_test(pheno ~ sex + age, net_kernel, gwas, method='satt')
</code></pre>

<hr>
<h2 id='lowrank_kernel-class'>An S4 class to represent a low-rank kernel for a SNPset at specified knots</h2><span id='topic+lowrank_kernel-class'></span><span id='topic+lowrank_kernel'></span>

<h3>Description</h3>

<p>An S4 class to represent a low-rank kernel for a SNPset at specified knots
</p>


<h3>Details</h3>

<p>This kernel is used for predictions. If observations and knots are 
equal, better construct a full-rank kernel of class <code><a href="#topic+kernel">kernel</a></code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>type</code></dt><dd><p>character, kernel type: Use <code>'lin'</code> for the linear kernel,
<code>'sia'</code> for the size-adjusted or <code>'net'</code> for the network-based kernel.</p>
</dd>
<dt><code>kernel</code></dt><dd><p>kernel <code>matrix</code> of dimension equal to individuals</p>
</dd>
<dt><code>pathway</code></dt><dd><p><code><a href="#topic+pathway">pathway</a></code> object</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Juliane Manitz
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gwas)
data(hsa04020)
square &lt;- calc_kernel(gwas, hsa04020, knots=gwas, type='lin', calculation='cpu')
dim(square@kernel)

gwas2 &lt;- new('GWASdata', pheno=pheno[1:10,], geno=geno[1:10,], anno=anno, desc="study 2")
low_rank &lt;- calc_kernel(gwas, hsa04020, knots = gwas2, type='net', calculation='cpu')
dim(low_rank@kernel)
</code></pre>

<hr>
<h2 id='make_psd+2Cmatrix-method'>Adjust network <code>matrix</code> to be positive semi-definite</h2><span id='topic+make_psd+2Cmatrix-method'></span><span id='topic+make_psd'></span><span id='topic+matrix'></span>

<h3>Description</h3>

<p>Adjust network <code>matrix</code> to be positive semi-definite
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'matrix'
make_psd(x, eps = sqrt(.Machine$double.eps))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_psd+2B2Cmatrix-method_+3A_x">x</code></td>
<td>
<p>A <code>matrix</code> specifying the network adjacency matrix.</p>
</td></tr>
<tr><td><code id="make_psd+2B2Cmatrix-method_+3A_eps">eps</code></td>
<td>
<p>A <code>numeric</code> value, setting the tolance for smallest 
eigenvalue adjustment</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a <code>matrix</code> N, the closest positive semi-definite <code>matrix</code> is 
calculated as N* = rho*N + (1+rho)*I, where I is the identity <code>matrix</code>
and rho = 1/(1 - lambda) with lambda the smallest eigenvalue of N. 
For more details check the references.
</p>


<h3>Value</h3>

<p>The <code>matrix</code> <code>x</code>, if it is positive definite and the closest
positive semi-definite <code>matrix</code> if <code>x</code> is not positive semi-definite.
</p>


<h3>Author(s)</h3>

<p>Juliane Manitz, Saskia Freytag, Stefanie Friedrichs
</p>


<h3>References</h3>


<ul>
<li><p> Freytag S, Manitz J, Schlather M, Kneib T, Amos CI, Risch A, Chang-Claude J, Heinrich J, Bickeboeller H: A network-based kernel machine test for the identification of risk pathways in genome-wide association studies. Hum Hered. 2013, 76(2):64-75.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>set.seed(2345)
m &lt;- matrix(data=sample(size=25, c(0,0,1), replace=TRUE),5,5)
m &lt;- m + t(m)
min(eigen(m, only.values = TRUE, symmetric = TRUE)$values)
round(make_psd(m),2)

</code></pre>

<hr>
<h2 id='net.kernel.hsa04020'>Example network-based kernel matrix for pathway hsa04020.</h2><span id='topic+net.kernel.hsa04020'></span>

<h3>Description</h3>

<p>An example of a kernel object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(net.kernel.hsa04020)
</code></pre>


<h3>Format</h3>

<p>An object of class <code><a href="#topic+kernel">kernel</a></code> and type 'network' for the pathway 
hsa04020.
</p>

<dl>
<dt>type</dt><dd><p>specifies which kernel function was used to calculate the kernel</p>
</dd>
<dt>kernel</dt><dd><p>includes the kernel matrix calculated for the pathway</p>
</dd>
<dt>pathway</dt><dd><p>includes the <code><a href="#topic+pathway">pathway</a></code> object of the pathway, for which  
the kernel matrix was calculated</p>
</dd>
</dl>



<h3>Source</h3>

<p>simulated data and Ensembl extract
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(net.kernel.hsa04020)
# derivation 
data(gwas)
data(hsa04020)
net_kernel &lt;- calc_kernel(gwas, hsa04020, knots=NULL, type='net', calculation='cpu')
# are the value differences smaller than machine epsilon?
all(abs(net.kernel.hsa04020@kernel - net_kernel@kernel) &lt; sqrt(.Machine$double.eps))

</code></pre>

<hr>
<h2 id='pathway'>An S4 class to represent a gene-gene interaction network</h2><span id='topic+pathway'></span><span id='topic+pathway+2CANY-method'></span><span id='topic+show+2Cpathway-method'></span><span id='topic+show+2Cpathway+2CANY-method'></span><span id='topic+summary+2Cpathway-method'></span><span id='topic+summary+2Cpathway+2CANY-method'></span><span id='topic+pathway2igraph+2Cpathway-method'></span><span id='topic+pathway2igraph'></span><span id='topic+ANY-method'></span><span id='topic+analyze+2Cpathway-method'></span><span id='topic+analyze'></span><span id='topic+get_genes+2Cpathway-method'></span><span id='topic+get_genes'></span><span id='topic+plot+2Cpathway+2Cmissing-method'></span><span id='topic+plot+2Cpathway+2CANY-method'></span><span id='topic+sample_genes+2Cpathway-method'></span><span id='topic+sample_genes'></span>

<h3>Description</h3>

<p>An S4 class to represent a gene-gene interaction network
</p>
<p><code>pathway</code> is the <code><a href="#topic+pathway">pathway</a></code> object constructor.
</p>
<p><code>show</code> displays the <code><a href="#topic+pathway">pathway</a></code> object briefly
</p>
<p><code>summary</code> generates a <code><a href="#topic+pathway">pathway</a></code> object summary including basic network properties.
</p>
<p><code>pathway2igraph</code> converts a <code><a href="#topic+pathway">pathway</a></code> object into an 
<code><a href="igraph.html#topic+igraph">igraph</a></code> object with edge attribute <code>sign</code>
</p>
<p>analyze <code><a href="#topic+pathway">pathway</a></code> network properties
</p>
<p><code>get_genes</code> is a helper function that extracts the gene names in a 
<code><a href="#topic+pathway">pathway</a></code> and returns a <code>vector</code> containing <code>character</code>
elements of gene names
</p>
<p><code>plot</code> visualizes the <code><a href="#topic+pathway">pathway</a></code> as <code><a href="igraph.html#topic+igraph">igraph</a></code> object
</p>
<p><code>sample_genes</code> randomly selects effect gene in a 
<code><a href="#topic+pathway">pathway</a></code> according the betweenness centrality and (no -1) neighors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway(object, ...)

## S4 method for signature 'ANY'
pathway(id, adj = matrix(0), sign = NULL)

## S4 method for signature 'pathway'
show(object)

## S4 method for signature 'pathway'
summary(object)

## S4 method for signature 'pathway'
pathway2igraph(object)

## S4 method for signature 'pathway'
analyze(object, ...)

## S4 method for signature 'pathway'
get_genes(object)

## S4 method for signature 'pathway,missing'
plot(
  x,
  y = NA,
  highlight.genes = NULL,
  gene.names = c(NULL, "legend", "nodes"),
  main = NULL,
  asp = 0.95,
  vertex.size = 11,
  vertex.color = "khaki1",
  vertex.label.cex = 0.8,
  edge.width = 2,
  edge.color = "olivedrab4",
  ...
)

## S4 method for signature 'pathway'
sample_genes(object, no = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_+3A_object">object</code></td>
<td>
<p>An object of class <code>pathway-class</code></p>
</td></tr>
<tr><td><code id="pathway_+3A_...">...</code></td>
<td>
<p>Further arguments can be added to the function.</p>
</td></tr>
<tr><td><code id="pathway_+3A_id">id</code></td>
<td>
<p>A <code>character</code> repesenting the <code><a href="#topic+pathway">pathway</a></code> id.</p>
</td></tr>
<tr><td><code id="pathway_+3A_adj">adj</code></td>
<td>
<p>A <code>matrix</code> respresenting the network adjacency matrix of dimension 
equaling the number of genes (1 interaction, 0 otherwise)</p>
</td></tr>
<tr><td><code id="pathway_+3A_sign">sign</code></td>
<td>
<p>A <code>numeric</code> <code>vector</code> indicating the interaction type for
each link (1 activation, -1 inhibition) in the interaction network for the <code><a href="#topic+pathway">pathway</a></code>.</p>
</td></tr>
<tr><td><code id="pathway_+3A_x">x</code></td>
<td>
<p><code><a href="#topic+pathway">pathway</a></code> object</p>
</td></tr>
<tr><td><code id="pathway_+3A_y">y</code></td>
<td>
<p>missing (placeholder)</p>
</td></tr>
<tr><td><code id="pathway_+3A_highlight.genes">highlight.genes</code></td>
<td>
<p>vector of gene names or node id's, which should be highlighted in a different color, default is <code>NULL</code> so that no genes are highlighted</p>
</td></tr>
<tr><td><code id="pathway_+3A_gene.names">gene.names</code></td>
<td>
<p>character indicating whether the genes names should appear in a legend (<code>'legend'</code>), as vertex label (<code>'nodes'</code>), or should be omitted (<code>NA</code>)</p>
</td></tr>
<tr><td><code id="pathway_+3A_main">main</code></td>
<td>
<p>optional overall main title, default is <code>NULL</code>, which uses the <code><a href="#topic+pathway">pathway</a></code> id</p>
</td></tr>
<tr><td><code id="pathway_+3A_asp">asp</code></td>
<td>
<p>a <code>numeric</code> constant, which gives the aspect ratio parameter for plot, default is 0.95</p>
</td></tr>
<tr><td><code id="pathway_+3A_vertex.size">vertex.size</code></td>
<td>
<p>a <code>numeric</code> constant specifying the vertex size, default is 11</p>
</td></tr>
<tr><td><code id="pathway_+3A_vertex.color">vertex.color</code></td>
<td>
<p>a <code>character</code> or <code>numeric</code> constant specifying the vertex color, default is 'khaki1'</p>
</td></tr>
<tr><td><code id="pathway_+3A_vertex.label.cex">vertex.label.cex</code></td>
<td>
<p>a <code>numeric</code> constant specifying the the vertex label size, default is 0.8,</p>
</td></tr>
<tr><td><code id="pathway_+3A_edge.width">edge.width</code></td>
<td>
<p>a <code>numeric</code> constant specifying the edge width, default is 2</p>
</td></tr>
<tr><td><code id="pathway_+3A_edge.color">edge.color</code></td>
<td>
<p>a <code>character</code> or <code>numeric</code> constant specifying the edge color, default is 'olivedrab4'</p>
</td></tr>
<tr><td><code id="pathway_+3A_no">no</code></td>
<td>
<p>a <code>numeric</code> constant specifying the number of genes to be sampled, default is 3</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>pathway2igraph</code> returns an unweighted <code><a href="igraph.html#topic+igraph">igraph</a></code> object with edge attribute <code>sign</code>
</p>
<p><code>analyze</code> returns a <code>data.frame</code> consisting of 
</p>

<dl>
<dt>id</dt><dd><p>pathway id,</p>
</dd> 
<dt>vcount</dt><dd><p>number of genes,</p>
</dd>
<dt>ecount</dt><dd><p>number of links,</p>
</dd>
<dt>inh_ecount</dt><dd><p>number of inhibition links,</p>
</dd>
<dt>density</dt><dd><p>network density,</p>
</dd>
<dt>av_deg</dt><dd><p>average degree,</p>
</dd>
<dt>inh_deg</dt><dd><p>average degree of inhibition links,</p>
</dd>
<dt>diam</dt><dd><p>network diamter,</p>
</dd>
<dt>trans</dt><dd><p>transitivity, and </p>
</dd>
<dt>s_trans</dt><dd><p>signed transitivity (Kunegis et al., 2009).</p>
</dd>
</dl>

<p><code>get_genes</code> returns a character vector of gene names extracted from adjacency matrix rownames.
</p>
<p><code>sample_genes</code> returns a <code>vector</code> of length <code>no</code> with 
vertex id's of sampled genes
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>analyze(pathway)</code>: 
</p>
</li>
<li> <p><code>get_genes(pathway)</code>: 
</p>
</li>
<li> <p><code>sample_genes(pathway)</code>: 
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>id</code></dt><dd><p>A <code>character</code> repesenting the <code><a href="#topic+pathway">pathway</a></code> id, 
e.g. hsa00100 as used in the KEGG database.</p>
</dd>
<dt><code>adj</code></dt><dd><p>A <code>matrix</code> respresenting the network adjacency matrix of dimension 
equaling the number of genes (1 interaction, 0 otherwise)</p>
</dd>
<dt><code>sign</code></dt><dd><p>A <code>numeric</code> <code>vector</code> indicating the interaction type for 
each link (1 activation, -1 inhibition) in the interaction network for the 
<code><a href="#topic+pathway">pathway</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Juliane Manitz, Stefanie Friedrichs, Patricia Burger
</p>


<h3>References</h3>

<p>Details to the computation and interpretation can be found in:
</p>

<ul>
<li><p> Kolaczyk, E. D. (2009). Statistical analysis of network data: methods and models. Springer series in statistics. Springer.
</p>
</li>
<li><p> Kunegis, J., A. Lommatzsch, and C. Bauckhage (2009). The slashdot zoo: Mining a social network with negative egdes. In Proceedings of the 18th international conference on World wide web, pp. 741-750. ACM Press.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># pathway object constructor
pathway(id="hsa04022")

# convert to igraph object
data(hsa04020)
str(hsa04020)
g &lt;- pathway2igraph(hsa04020)
str(g)

# analyze pathway network properties
data(hsa04020)
summary(hsa04020)
analyze(hsa04020)

# extract gene names from pathway object
get_genes(hsa04020)

# plot pathway as igraph object
plot(hsa04020)
sample3 &lt;- sample_genes(hsa04020, no = 3)
plot(hsa04020, highlight.genes = sample3)

# sample effect genes
sample3 &lt;- sample_genes(hsa04020, no = 3)
plot(hsa04020, highlight.genes = sample3)
sample5 &lt;- sample_genes(hsa04020, no = 5)
plot(hsa04020, highlight.genes = sample5)

</code></pre>

<hr>
<h2 id='pathway_info'>An S4 class for an object assigning genes to pathways</h2><span id='topic+pathway_info'></span><span id='topic+pathway_info+2Ccharacter-method'></span><span id='topic+show+2Cpathway_info-method'></span><span id='topic+show+2Cpathway_info+2CANY-method'></span><span id='topic+summary+2Cpathway_info-method'></span><span id='topic+summary+2Cpathway_info+2CANY-method'></span>

<h3>Description</h3>

<p>This function lists all genes formig a particular <code><a href="#topic+pathway">pathway</a></code>. Start and end  
positions of these genes are extracted from the Ensemble database. The 
database is accessed via the R-package <span class="pkg">biomaRt</span>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway_info(x)

## S4 method for signature 'character'
pathway_info(x)

## S4 method for signature 'pathway_info'
show(object)

## S4 method for signature 'pathway_info'
summary(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway_info_+3A_x">x</code></td>
<td>
<p>A <code>character</code> identifying the pathway for which gene infomation 
should be extracted. Here KEGG IDs (format: 'hsa00100') are used.</p>
</td></tr>
<tr><td><code id="pathway_info_+3A_object">object</code></td>
<td>
<p>An object of class <code><a href="#topic+pathway_info">pathway_info</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> including as many rows as genes appear in the 
<code><a href="#topic+pathway">pathway</a></code>. for each gene its name, the start and end point and the chromosome 
it lies on are given.
</p>
<p><code>show</code> Basic information on <code><a href="#topic+pathway_info">pathway_info</a></code> object.
</p>
<p><code>summary</code> Summarized information on <code><a href="#topic+pathway_info">pathway_info</a></code> object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>info</code></dt><dd><p>A <code>data.frame</code> including information on genes contained in 
pathways with columns 'pathway', 'gene_start', 'gene_end', 'chr' and 'gene'.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs, Juliane Manitz
</p>


<h3>See Also</h3>

<p><code><a href="#topic+snp_info">snp_info</a></code>, <code><a href="#topic+get_anno">get_anno</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(hsa04022_info) # pathway_info('hsa04020') 
show(hsa04022_info)
summary(hsa04022_info)

</code></pre>

<hr>
<h2 id='pheno'>Example phenotype file for 50 individuals.</h2><span id='topic+pheno'></span>

<h3>Description</h3>

<p>A dataset containing simulated example phenotypes for 50 individuals 
row names include the identifiers of 50 example individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(pheno)
</code></pre>


<h3>Format</h3>

<p>A <code>data frame</code> with 50 rows and 3 variables:
</p>

<dl>
<dt>pheno</dt><dd><p>includes the case-control status for each individual, coded as
1(case) or 0 (control)</p>
</dd>
<dt>sex</dt><dd><p>includes gender information for the 50 individuals, coded as 1 
(male) or 0 (female)</p>
</dd>
<dt>age</dt><dd><p>numerical value giving the persons age</p>
</dd>
</dl>



<h3>Source</h3>

<p>simulated data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(pheno)
head(pheno)
# create gwas object
data(geno)
data(anno)
gwas &lt;- new('GWASdata', pheno=pheno, geno=geno, anno=anno, desc="some study") 
</code></pre>

<hr>
<h2 id='read_geno+2Ccharacter-method'>read genotype data from file to one of several available objects, which 
can be passed to a GWASdata object <code><a href="#topic+GWASdata">GWASdata</a></code>.</h2><span id='topic+read_geno+2Ccharacter-method'></span><span id='topic+read_geno'></span><span id='topic+character'></span>

<h3>Description</h3>

<p>read genotype data from file to one of several available objects, which 
can be passed to a GWASdata object <code><a href="#topic+GWASdata">GWASdata</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'character'
read_geno(
  file.path,
  save.path = NULL,
  sep = " ",
  header = TRUE,
  use.fread = TRUE,
  use.big = FALSE,
  row.names = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_file.path">file.path</code></td>
<td>
<p><code>character</code> giving the path to the data file to be read</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_save.path">save.path</code></td>
<td>
<p><code>character</code> containing the path for the backingfile</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_sep">sep</code></td>
<td>
<p><code>character</code>. A field delimeter. See 
<code><a href="bigmemory.html#topic+read.big.matrix">read.big.matrix</a></code> for details.</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_header">header</code></td>
<td>
<p><code>logical</code>. Does the data set contain column names?</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_use.fread">use.fread</code></td>
<td>
<p><code>logical</code>. Should the dataset be read using the function 
<code><a href="data.table.html#topic+fread">fread</a></code> <code>fread</code> from package <span class="pkg">data.table</span>?</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_use.big">use.big</code></td>
<td>
<p><code>logical</code>. Should the dataset be read using the function 
<code><a href="bigmemory.html#topic+read.big.matrix">read.big.matrix</a></code> from package <span class="pkg">bigmemory</span>?</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_row.names">row.names</code></td>
<td>
<p><code>logical</code>. Does the dataset include rownames?</p>
</td></tr>
<tr><td><code id="read_geno+2B2Ccharacter-method_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <code>read_geno</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the data set contains rownames specified, set option <code>has.row.names = TRUE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
path &lt;- system.file("extdata", "geno.txt", package = "kangar00")
geno &lt;- read_geno(path, save.path = getwd(), sep = " ", use.fread = FALSE, row.names = FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='rewire_network'>Rewires interactions in a <code><a href="#topic+pathway">pathway</a></code>, which go through a gene not 
represented by any SNPs in the considered <code><a href="#topic+GWASdata">GWASdata</a></code> dataset.</h2><span id='topic+rewire_network'></span><span id='topic+rewire_network+2Cpathway-method'></span>

<h3>Description</h3>

<p>Rewires interactions in a <code><a href="#topic+pathway">pathway</a></code>, which go through a gene not 
represented by any SNPs in the considered <code><a href="#topic+GWASdata">GWASdata</a></code> dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'pathway'
rewire_network(object, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rewire_network_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+pathway">pathway</a></code> object which's network <code>matrix</code> will be rewired</p>
</td></tr>
<tr><td><code id="rewire_network_+3A_x">x</code></td>
<td>
<p>A <code>vector</code> of gene names, indicating which genes are not represented
by SNPs in the considered <code><a href="#topic+GWASdata">GWASdata</a></code> object and will be removed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+pathway">pathway</a></code> object including the rewired network <code>matrix</code>
</p>


<h3>Author(s)</h3>

<p>Juliane Manitz, Stefanie Friedrichs
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(hsa04020)
summary(hsa04020)
hsa04020_rewired &lt;- rewire_network(hsa04020, x=c('ADCY3', 'CALML3','GNAQ'))
summary(hsa04020_rewired)

## End(Not run)
</code></pre>

<hr>
<h2 id='rs10243170_info'>Example <code><a href="#topic+snp_info">snp_info</a></code> object for SNP rs10243170.</h2><span id='topic+rs10243170_info'></span>

<h3>Description</h3>

<p>An object of class <code><a href="#topic+snp_info">snp_info</a></code> for rs10243170.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rs10243170_info)
</code></pre>


<h3>Format</h3>

<p>A <code><a href="#topic+snp_info">snp_info</a></code> object including information on the SNP as 
extracted from the Ensembl database.
</p>

<dl>
<dt>info</dt><dd><p>a <code>data frame</code> including the extracted information on the 
SNP. Columns given are 'chr', 'position', and 'rsnumber'</p>
</dd>
</dl>



<h3>Source</h3>

<p>Ensembl extract
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
snp_info("rs10243170")

## End(Not run)
</code></pre>

<hr>
<h2 id='set_names+2Cmatrix-method'>Helper function to translate gene numbers to names</h2><span id='topic+set_names+2Cmatrix-method'></span>

<h3>Description</h3>

<p>This function exchanges the numbers used for genes in KEGG download KGML files
with the corresponding gene names. Names are set to be the column names and 
rownames of the pathway's network matrix. The function 
is called by <code><a href="#topic+get_network_matrix">get_network_matrix</a></code>. (For internal use)
</p>
<p>This function exchanges the numbers used for genes in KEGG download KGML files
with the corresponding gene names. Names are set to be the column names and 
rownames of the pathway's network matrix. The function 
is called by <code><a href="#topic+get_network_matrix">get_network_matrix</a></code>. (For internal use)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'matrix'
set_names(x, nodes, my_list)

## S4 method for signature 'matrix'
set_names(x, nodes, my_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_names+2B2Cmatrix-method_+3A_x">x</code></td>
<td>
<p>A <code>matrix</code> representing the network matrix.</p>
</td></tr>
<tr><td><code id="set_names+2B2Cmatrix-method_+3A_nodes">nodes</code></td>
<td>
<p>A <code>vector</code> of gene numbers to be replaced by names.</p>
</td></tr>
<tr><td><code id="set_names+2B2Cmatrix-method_+3A_my_list">my_list</code></td>
<td>
<p>A <code>data.frame</code> listing gene names and numbers. Output from 
<code>gene_name_number</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>matrix</code> representing the interaction network in the pathway
with gene names as rownames and columnnames.
</p>
<p>A <code>matrix</code> representing the interaction network in the pathway
with gene names as rownames and columnnames.
</p>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs
</p>

<hr>
<h2 id='set_one+2Cmatrix-method'>Helper function to set <code>matrix</code> entries to 0/1/-1 only</h2><span id='topic+set_one+2Cmatrix-method'></span>

<h3>Description</h3>

<p>This function sets all entries in a <code>matrix</code> bigger than 1 to 1 and all entries 
smaller than -1 to -1. It is called by <code><a href="#topic+get_network_matrix">get_network_matrix</a></code>.
(For internal use)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'matrix'
set_one(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_one+2B2Cmatrix-method_+3A_x">x</code></td>
<td>
<p>numeric <code>matrix</code> representing the network adjacency matrix.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>matrix</code> representing the interaction network in the 
<code><a href="#topic+pathway">pathway</a></code> object with entries equal to 1, -1 or 0.
</p>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_network_matrix">get_network_matrix</a></code>
</p>

<hr>
<h2 id='snp_info'>An S4 class for an object assigning SNP positions to rs-numbers (for internal use)</h2><span id='topic+snp_info'></span><span id='topic+snp_info+2Ccharacter-method'></span><span id='topic+show+2Csnp_info-method'></span><span id='topic+summary+2Csnp_info-method'></span>

<h3>Description</h3>

<p>An S4 class for an object assigning SNP positions to rs-numbers (for internal use)
</p>
<p>This function gives for a <code>vector</code> of SNP identifiers the position of each SNP 
as extracted from the Ensemble database. The database is accessed via the
R-package <span class="pkg">biomaRt</span>.
</p>
<p><code>show</code> Shows basic information on <code><a href="#topic+snp_info">snp_info</a></code> object
</p>
<p><code>summary</code> Summarizes information on <code><a href="#topic+snp_info">snp_info</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snp_info(x, ...)

## S4 method for signature 'character'
snp_info(x)

## S4 method for signature 'snp_info'
show(object)

## S4 method for signature 'snp_info'
summary(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snp_info_+3A_x">x</code></td>
<td>
<p>A <code>character</code> <code>vector</code> of SNP rsnumbers for which 
positions will be extracted.</p>
</td></tr>
<tr><td><code id="snp_info_+3A_...">...</code></td>
<td>
<p>further arguments can be added.</p>
</td></tr>
<tr><td><code id="snp_info_+3A_object">object</code></td>
<td>
<p>An <code>object</code> of class <code><a href="#topic+snp_info">snp_info</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> including the SNP positions with columns
'chromosome', 'position' and 'snp'. SNPs not found in the Ensemble database
will not be listed in the returned <code>snp_info</code> object, SNPs with multiple
positions  will appear several times.
</p>
<p><code>show</code> Basic information on <code><a href="#topic+snp_info">snp_info</a></code> object.
</p>
<p><code>summary</code> Summarized information on <code><a href="#topic+snp_info">snp_info</a></code> object.
</p>


<h3>Slots</h3>


<dl>
<dt><code>info</code></dt><dd><p>A <code>data.frame</code> including information on SNP positions</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Stefanie Friedrichs
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pathway_info">pathway_info</a></code>, <code><a href="#topic+get_anno">get_anno</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rs10243170_info) # snp_info("rs10243170")

rs10243170_info
summary(rs10243170_info)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
