<!DOCTYPE html><html lang="en"><head><title>Help for package kaos</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kaos}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cgr'><p>Chaos Game Representation Object</p></a></li>
<li><a href='#cgr.plot'><p>Plot from a CGR object</p></a></li>
<li><a href='#cgr.res'><p>Produces a new matrix with new resolution</p></a></li>
<li><a href='#HIV'><p>Complete HIV 1 genome</p></a></li>
<li><a href='#vectorize'><p>Vectorizes the FCGR</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Encoding of Sequences Based on Frequency Matrix Chaos Game
Representation</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Author:</td>
<td>Dominik Eger and Hannah Franziska Löchel</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hannah Franziska Löchel &lt;loechelh@mathematik.uni-marburg.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Sequences encoding by using the chaos game representation.
    Löchel et al. (2019) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtz493">doi:10.1093/bioinformatics/btz493</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, reshape2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-09-27 07:51:55 UTC; franzi</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-09-27 08:50:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='cgr'>Chaos Game Representation Object</h2><span id='topic+cgr'></span>

<h3>Description</h3>

<p>Chaos Game Representation Object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cgr(data, seq.base = row.names(table(data)), sf = F, res = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cgr_+3A_data">data</code></td>
<td>
<p>Input as list/vector of characters from fasta file or similar</p>
</td></tr>
<tr><td><code id="cgr_+3A_seq.base">seq.base</code></td>
<td>
<p>By default the included unique elements in data will be used
in alphabetical order. It is also possible to define the alphabet
explicitly.<br />
Predefined alphabets can be used as well:<br />
</p>

<ul>
<li><p> &quot;digits&quot;: numbers from 0 to 9
</p>
</li>
<li><p> &quot;AMINO&quot;: alphabetical order of the amino acids in capital letters
</p>
</li>
<li><p> &quot;amino&quot;: alphabetical order of the amino acids in lowercase letters
</p>
</li>
<li><p> &quot;DNA&quot;: the four bases of DNA (&quot;A&quot;,&quot;G&quot;,&quot;T&quot;,&quot;C&quot;) in capital letters
</p>
</li>
<li><p> &quot;dna&quot;: the four bases of DNA (&quot;a&quot;,&quot;g&quot;,&quot;t&quot;,&quot;c&quot;) in lowercase letters
</p>
</li>
<li><p> &quot;LETTERS&quot;: The alphabetical order of capital letters from A to Z
</p>
</li>
<li><p> &quot;letters&quot;: The alphabetical order of lowercase Letters from a to z
</p>
</li></ul>
</td></tr>
<tr><td><code id="cgr_+3A_sf">sf</code></td>
<td>
<p>By default, the scaling factor for fractal polygons is used; the
scaling factor can also explicitly set to values between 0 and 1.</p>
</td></tr>
<tr><td><code id="cgr_+3A_res">res</code></td>
<td>
<p>resolution of the frequency matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function produces a chaos game representation (CGR)
object from a sequence (data)
</p>


<h3>Value</h3>

<p>CGR object as list of:
</p>

<ul>
<li><p> matrix: frequency matrix with given resolution
</p>
</li>
<li><p> x: x-coordinates for the CGR
</p>
</li>
<li><p> y: y-coordinates for the CGR
</p>
</li>
<li><p> sf: applied scaling factor for the CGR
</p>
</li>
<li><p> res: applied resolution to calculate the FCGR
</p>
</li>
<li><p> base.seq: chars or letters to build the edges of the CGR
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>###HIV data
data("HIV")

### encoding the sequence
HIV.cgr = cgr(HIV, res = 100)

###plot the sequence
cgr.plot(HIV.cgr, mode = "points")

###plot the FCGR
cgr.plot(HIV.cgr, mode = "matrix")

###change the resolution of matrix from 100x100 to 200x200
cgr.res(HIV.cgr, 200)

### get the FCGR encoded vector
vectorize(HIV.cgr)


</code></pre>

<hr>
<h2 id='cgr.plot'>Plot from a CGR object</h2><span id='topic+cgr.plot'></span>

<h3>Description</h3>

<p>Plot from a CGR object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cgr.plot(data, mode, corners = F, labels = F)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cgr.plot_+3A_data">data</code></td>
<td>
<p>CGR object</p>
</td></tr>
<tr><td><code id="cgr.plot_+3A_mode">mode</code></td>
<td>
<p>character string
</p>

<ul>
<li><p> &quot;points&quot;: CGR plot
</p>
</li>
<li><p> &quot;matrix&quot;: FCGR plot
</p>
</li></ul>
</td></tr>
<tr><td><code id="cgr.plot_+3A_corners">corners</code></td>
<td>
<p>if true, the corners are added as red dots</p>
</td></tr>
<tr><td><code id="cgr.plot_+3A_labels">labels</code></td>
<td>
<p>if true, the symbol associated with the corner is added</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots the chaos game reprasentation as points or as
frequency matrix representation
</p>

<hr>
<h2 id='cgr.res'>Produces a new matrix with new resolution</h2><span id='topic+cgr.res'></span>

<h3>Description</h3>

<p>Produces a new matrix with new resolution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cgr.res(data, res)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="cgr.res_+3A_data">data</code></td>
<td>
<p>CGR object</p>
</td></tr>
<tr><td><code id="cgr.res_+3A_res">res</code></td>
<td>
<p>new resolution</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function adjust the resolution of the FCGR of a CGR object.
</p>

<ul>
<li><p> matrix: frequency matrix with new given resolution
</p>
</li>
<li><p> x: x-coordinates for the CGR
</p>
</li>
<li><p> y: y-coordinates for the CGR
</p>
</li>
<li><p> sf: applied scaling factor for the CGR
</p>
</li>
<li><p> res: applied resolution to calculate the FCGR
</p>
</li>
<li><p> base.seq: chars or letters to build the edges of the CGR
</p>
</li></ul>


<hr>
<h2 id='HIV'>Complete HIV 1 genome</h2><span id='topic+HIV'></span>

<h3>Description</h3>

<p>Complete HIV 1 genome
</p>


<h3>Format</h3>

<p>A vector containing the genome of HIV-1 as single characters</p>


<h3>Source</h3>

<p>NC_001802.1 Human immunodeficiency virus 1, complete genome
</p>

<hr>
<h2 id='vectorize'>Vectorizes the FCGR</h2><span id='topic+vectorize'></span>

<h3>Description</h3>

<p>Vectorizes the FCGR
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vectorize(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vectorize_+3A_data">data</code></td>
<td>
<p>CGR object from cgr.R</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function returns the FCGR as vector.
</p>


<h3>Value</h3>

<p>Vector with FCGR-encoding
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
