<!DOCTYPE html><html><head><title>Help for package kin.cohort</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kin.cohort}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#internal'><p> Internal functions for marginal method</p></a></li>
<li><a href='#kc.marginal'><p> Marginal Maximum Likelihood estimation of kin-cohort data</p></a></li>
<li><a href='#kc.moments'><p> Kin-cohort estimation of penetrance by the method of moments</p></a></li>
<li><a href='#kin.cohort'><p> Analysis of kin-cohort data</p></a></li>
<li><a href='#kin.data'><p> sample data for kin-cohort analysis</p></a></li>
<li><a href='#methods'><p> methods for print and plot</p></a></li>
<li><a href='#simulations'><p> simulation of kin cohort studies</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of Kin-Cohort Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-08-15</td>
</tr>
<tr>
<td>Author:</td>
<td>Victor Moreno, Nilanjan Chatterjee, Bhramar Mukherjee</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Victor Moreno &lt;v.moreno@iconcologia.net&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>survival</td>
</tr>
<tr>
<td>Description:</td>
<td>Analysis of kin-cohort studies. kin.cohort provides estimates of age-specific 
 cumulative risk of a disease for carriers and noncarriers of a mutation. The cohorts are
 retrospectively built from relatives of probands for whom the genotype is known. Currently 
 the method of moments and marginal maximum likelihood are implemented. Confidence intervals 
 are calculated from bootstrap samples.
 Most of the code is a translation from previous 'MATLAB' code by N. Chatterjee.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-08-28 10:33:32 UTC; h501uvma</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-08-28 16:36:59</td>
</tr>
</table>
<hr>
<h2 id='internal'> Internal functions for marginal method </h2><span id='topic+pyear'></span><span id='topic+pwexp'></span><span id='topic+mendelian'></span><span id='topic+mendelian.combine'></span>

<h3>Description</h3>

<p>Internal functions for marginal method</p>


<h3>Value</h3>

<table>
<tr><td><code>pyear</code></td>
<td>
<p> calculates number of events and person years</p>
</td></tr>
<tr><td><code>pwexp</code></td>
<td>
<p> estimates survival and hazard for piece-wise exponential model</p>
</td></tr>
<tr><td><code>mendelian</code></td>
<td>
<p> calculates the mendelian probabilities of carrying the mutation conditional on the proband genotype for 1 gene.</p>
</td></tr>
<tr><td><code>mendelian.combine</code></td>
<td>
<p> combines mendelian probabilities of carrying the mutation conditional on the proband genotype for 2 genes.</p>
</td></tr>
</table>


<h3>See Also</h3>

 <p><code><a href="#topic+kc.marginal">kc.marginal</a></code> </p>

<hr>
<h2 id='kc.marginal'> Marginal Maximum Likelihood estimation of kin-cohort data </h2><span id='topic+kc.marginal'></span>

<h3>Description</h3>

<p>This function estimates cumulative risk and hazard at given ages for carriers and noncarriers of a mutation based on the probands genotypes. It uses the Marginal Maximum Likelihood estimation method (Chatterjee and Wacholder, 2001). Piece-wise exponential distribution is assumed for the survival function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kc.marginal(t, delta, genes, r, knots, f, pw = rep(1,length(t)), 
            set = NULL, B = 1, maxit = 1000, tol = 1e-5, subset,
            logrank=TRUE, trace=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kc.marginal_+3A_t">t</code></td>
<td>
<p> time variable. Usually age at diagnosis or at last follow-up </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_delta">delta</code></td>
<td>
<p> disease status (1: event, 0: no event </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_genes">genes</code></td>
<td>
<p> factor or numeric vector (1 gene), matrix or dataframe (2 genes) with genotypes of proband numeric. factors and data.frame with factors are prefered in order to use user-defined labels. Otherwise use codes (1:noncarrier, 2: carrier, 3: homozygous carrier) </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_r">r</code></td>
<td>
<p> relationship with proband 1:parent, 2:sibling 3:offspring 0:proband. Probands will be excluded from analysis and offspring will be recoded 1 internally.</p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_knots">knots</code></td>
<td>
<p> time points (ages) for cumulative risk and hazard estimates </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_f">f</code></td>
<td>
<p> vector of mutation allele frequencies in the population </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_pw">pw</code></td>
<td>
<p> prior weights, if needed </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_set">set</code></td>
<td>
<p> family id (only needed for bootstrap)   </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_b">B</code></td>
<td>
<p> number of boostrap samples (only needed for bootstrap) </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_maxit">maxit</code></td>
<td>
<p> max number of iterations for the EM algorithm </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_tol">tol</code></td>
<td>
<p> convergence tolerance </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_subset">subset</code></td>
<td>
<p> logical condition to subset data </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_logrank">logrank</code></td>
<td>
<p> Perform a logrank test </p>
</td></tr>
<tr><td><code id="kc.marginal_+3A_trace">trace</code></td>
<td>
<p> Show iterations for bootstrap </p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of classes &quot;kin.cohort&quot; and &quot;chatterjee&quot;. 
</p>
<table>
<tr><td><code>cumrisk</code></td>
<td>
<p>matrix with cumulative risk estimates for noncarriers, carriers and the cumulative risk ratio. Estimates are given for the times indicated in the knot vector</p>
</td></tr>
<tr><td><code>hazard</code></td>
<td>
<p>matrix with hazard estimates for noncarriers, carriers and the hazard ratio. Estimates are given for the times indicated in the knot vector</p>
</td></tr>
<tr><td><code>knots</code></td>
<td>
<p>vector of knots</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>if the EM algorithm converged</p>
</td></tr>
<tr><td><code>niter</code></td>
<td>
<p>number of iterations needed for convergence</p>
</td></tr>
<tr><td><code>ngeno.rel</code></td>
<td>
<p>number of combinations of genotypes in the relatives</p>
</td></tr>
<tr><td><code>events</code></td>
<td>
<p>matrix with number of events and person years per each knot</p>
</td></tr>
<tr><td><code>logHR</code></td>
<td>
<p>mean log hazard ratio estimate (unweighted)</p>
</td></tr>
<tr><td><code>logrank</code></td>
<td>
<p>logrank test. If 2 genes, for the main effects, the cross-classification and the stratified tests</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>copy of call</p>
</td></tr>
</table>
<p>if bootstrap confidence intervals are requested (B&gt;1) then the returned object is of classes &quot;kin.cohort.boot&quot; and &quot;chatterjee&quot; with previous items packed in value estimate and each bootstrap sample packed in matrices. 
</p>


<h3>Note</h3>

 
<p>This function is best called by kin.cohort than directly
</p>


<h3>References</h3>

 
<p>Chatterjee N and Wacholder S. A Marginal Likelihood Approach for Estimating Penetrance from Kin-Cohort Designs. Biometrics. 2001; 57: 245-52. 
</p>


<h3>See Also</h3>

 
<p><code><a href="#topic+kin.cohort">kin.cohort</a></code>, <code><a href="#topic+print.kin.cohort">print.kin.cohort</a></code>, <code><a href="#topic+plot.kin.cohort">plot.kin.cohort</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(kin.data)
attach(kin.data)
res.mml&lt;- kc.marginal(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02)
res.mml

## End(Not run)
</code></pre>

<hr>
<h2 id='kc.moments'> Kin-cohort estimation of penetrance by the method of moments </h2><span id='topic+kc.moments'></span>

<h3>Description</h3>

<p>This function estimates cumulative risk and hazard at given ages for carriers and noncarriers of a mutation based on the probands genotypes. It uses the method of moments described by Wacholder et al (1998)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kc.moments(t, delta, genes, r, knots, f, pw = rep(1,length(t)), 
           set = NULL, B = 1, logrank = TRUE, subset, trace=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kc.moments_+3A_t">t</code></td>
<td>
<p> time variable. Usually age at diagnosis or at last follow-up </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_delta">delta</code></td>
<td>
<p> disease status (1: event, 0: no event </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_genes">genes</code></td>
<td>
<p> genotype of proband numeric. A factor is preferred, otherwise numeric code of genotypes (1: noncarrier, 2:carrier, [3: homozygous carrier]) </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_r">r</code></td>
<td>
<p> relationship with proband 1:parent, 2:sibling 3:offspring 0:proband. Probands will be excluded from analysis and offspring will be recoded 1 internally. </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_knots">knots</code></td>
<td>
<p> time points (ages) for cumulative risk and hazard estimates </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_f">f</code></td>
<td>
<p> mutation allele frequency in the population </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_pw">pw</code></td>
<td>
<p> prior weights, if needed </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_set">set</code></td>
<td>
<p> family id (only needed for bootstrap) </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_b">B</code></td>
<td>
<p> number of boostrap samples (only needed for bootstrap) </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_logrank">logrank</code></td>
<td>
<p> if logrank test is desired </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_subset">subset</code></td>
<td>
<p> logical condition to subset data </p>
</td></tr>
<tr><td><code id="kc.moments_+3A_trace">trace</code></td>
<td>
<p> Show iterations for bootstrap </p>
</td></tr>
</table>


<h3>Value</h3>

<p>object of classes &quot;kin.cohort&quot; and &quot;wacholder&quot;. 
</p>
<table>
<tr><td><code>cumrisk</code></td>
<td>
<p>matrix of dimension (number of knots x 3) with cumulative risk festimates or noncarriers, carriers and the cumulative risk ratio</p>
</td></tr>
<tr><td><code>knots</code></td>
<td>
<p>vector of knots</p>
</td></tr>
<tr><td><code>km</code></td>
<td>
<p>object class survfit (package survival) </p>
</td></tr>
<tr><td><code>logrank</code></td>
<td>
<p>p-value of the logrank test</p>
</td></tr>  
<tr><td><code>events</code></td>
<td>
<p>matrix with number of events and person years per each knot</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>copy of call</p>
</td></tr>
</table>
<p>if bootstrap confidence intervals are requested (B&gt;1) then the returned object is of classes &quot;kin.cohort.boot&quot; and &quot;wacholder&quot; with previous items packed in value estimate and each bootstrap sample packed in matrices. 
</p>


<h3>Note</h3>

 
<p>This function is best called by kin.cohort than directly
</p>


<h3>References</h3>

 
<p>Wacholder S, Hartge P, Struewing JP, Pee D, McAdams M, Lawrence B, Tucker MA. 
The kin-cohort study for estimating penetrance. American Journal of Epidemiology. 1998; 148: 623-9.
</p>


<h3>See Also</h3>

 
<p><code><a href="#topic+kin.cohort">kin.cohort</a></code>, <code><a href="#topic+print.kin.cohort">print.kin.cohort</a></code>, <code><a href="#topic+plot.kin.cohort">plot.kin.cohort</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(kin.data)
attach(kin.data)
res.km&lt;- kc.moments(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02)
res.km

## End(Not run)
</code></pre>

<hr>
<h2 id='kin.cohort'> Analysis of kin-cohort data </h2><span id='topic+kin.cohort'></span>

<h3>Description</h3>

<p>This function estimates cumulative risk at given ages for carriers and noncarriers
of a mutation based on the probands genotypes. It can use the Marginal Maximum
Likelihood estimation method (Chatterjee and Wacholder, 2001) or the method of
moments (Wacholder et al, 2001). Bootstrap confidence intervals can be requested.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kin.cohort(..., method = c("marginal", "mml", "chatterjee", 
                             "moments",  "km",  "watcholder"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kin.cohort_+3A_...">...</code></td>
<td>
<p> see <code><a href="#topic+kc.marginal">kc.marginal</a></code> and <code><a href="#topic+kc.moments">kc.moments</a> for details</code> </p>
</td></tr>
<tr><td><code id="kin.cohort_+3A_method">method</code></td>
<td>
<p> choose estimation method: Marginal Maximum Likelihood
(selected by &quot;marginal&quot;, &quot;mml&quot;, &quot;chatterjee&quot;) or method of moments
(selected by &quot;moments&quot;, &quot;km&quot;, &quot;watcholder&quot;)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a wrapper that will call <code><a href="#topic+kc.marginal">kc.marginal</a></code> or <code><a href="#topic+kc.moments">kc.moments</a></code> depending on the argument method.
</p>


<h3>Author(s)</h3>

<p> Victor Moreno, Nilanjan Chatterjee, Bhramar Mukherjee. 
</p>
<p>Maintainer: Victor Moreno &lt;v.moreno@iconcologia.net&gt;
</p>


<h3>References</h3>

<p>Wacholder S, Hartge P, Struewing JP, Pee D, McAdams M, Lawrence B, Tucker MA. The kin-cohort study for estimating penetrance. American Journal of Epidemiology. 1998; 148: 623-9.
</p>
<p>Chatterjee N and Wacholder S. A Marginal Likelihood Approach for Estimating Penetrance from Kin-Cohort Designs. Biometrics. 2001; 57: 245-52.
</p>


<h3>See Also</h3>

 
<p><code><a href="#topic+kc.marginal">kc.marginal</a></code>, <code><a href="#topic+kc.moments">kc.moments</a></code> 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(kin.data)
attach(kin.data)

res.k&lt;-   kin.cohort(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02, 
                     method="km")
res.k          
plot(res.k)
plot(res.k,what="crr")

set.seed(1)
res.k.b&lt;- kin.cohort(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02, 
                     set=family, method="km", B=10)
res.k.b
plot(res.k.b)
plot(res.k.b,what="crr")

res.m&lt;-   kin.cohort(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02, 
                    method="mml")
res.m
plot(res.m)
plot(res.m, what="hr")

res.m2&lt;-  kin.cohort(age, cancer, data.frame(gen1,gen2), rel, 
                     knots=c(30,40,50,60,70,80), f=c(0.02,0.01), method="mml")
res.m2
plot(res.m2)
plot(res.m2, what="hr")

set.seed(1)
res.m.b&lt;- kin.cohort(age, cancer, gen1, rel, knots=c(30,40,50,60,70,80), f=0.02, 
                     set=family, method="mml", B=10)
res.m.b
plot(res.m.b)
plot(res.m.b, what="hr")

## End(Not run)
</code></pre>

<hr>
<h2 id='kin.data'> sample data for kin-cohort analysis</h2><span id='topic+kin.data'></span>

<h3>Description</h3>

<p>Simulated data of a study on the penetrance of breast cancer for carriers 2 mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kin.data)</code></pre>


<h3>Format</h3>

<p>A data frame with 15341 observations on the following 5 variables.
</p>

<dl>
<dt><code>age</code></dt><dd><p>age at diagnosis or at last follow-up</p>
</dd>
<dt><code>cancer</code></dt><dd><p>disease status (1: breast cancer, 0: no breast cancer</p>
</dd>
<dt><code>gen1</code></dt><dd><p>gen1 genotypes of proband</p>
</dd>
<dt><code>gen2</code></dt><dd><p>gen2 genotypes of proband</p>
</dd>
<dt><code>rel</code></dt><dd><p>relationship with proband 1:parent or offspring, 2:sibling</p>
</dd>
<dt><code>family</code></dt><dd><p>family id</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(kin.data)
</code></pre>

<hr>
<h2 id='methods'> methods for print and plot </h2><span id='topic+print.kin.cohort'></span><span id='topic+print.kin.cohort.boot'></span><span id='topic+plot.kin.cohort'></span><span id='topic+plot.kin.cohort.boot'></span>

<h3>Description</h3>

<p>Functions to print a formatted output and produce plots
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'kin.cohort'
print(x, descriptive = TRUE, cumrisk = TRUE, hazard = FALSE, survival = FALSE, 
        logrank = TRUE, HR = TRUE, digits = 5, ...)

## S3 method for class 'kin.cohort.boot'
print(x, cumrisk = TRUE, hazard = FALSE, HR = TRUE, conf = 0.95,
        digits = 5, show = TRUE, logrank = TRUE, ...)

## S3 method for class 'kin.cohort'
plot(x, what = c("cr", "hr", "crr"), min.age = min(x$knots), 
      max.age = max(x$knots), max.y, type, add=FALSE, color, line,  ...)

## S3 method for class 'kin.cohort.boot'
plot(x, conf = 0.95, what = c("cr", "hr", "crr"), min.age = min(x$knots), 
      max.age = max(x$knots), age.start = 0, start.ref, max.y, type,
      median = FALSE, add = FALSE, color, line, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="methods_+3A_x">x</code></td>
<td>
<p> object to be printed or plotted </p>
</td></tr>
<tr><td><code id="methods_+3A_descriptive">descriptive</code></td>
<td>
<p> print table with number of events and person-years</p>
</td></tr>
<tr><td><code id="methods_+3A_cumrisk">cumrisk</code></td>
<td>
<p>print cumulative risk</p>
</td></tr>
<tr><td><code id="methods_+3A_hazard">hazard</code></td>
<td>
<p>print hazard</p>
</td></tr>
<tr><td><code id="methods_+3A_survival">survival</code></td>
<td>
<p>print survival</p>
</td></tr>
<tr><td><code id="methods_+3A_hr">HR</code></td>
<td>
<p>print harard ratios</p>
</td></tr>
<tr><td><code id="methods_+3A_logrank">logrank</code></td>
<td>
<p>print logrank p value</p>
</td></tr>
<tr><td><code id="methods_+3A_digits">digits</code></td>
<td>
<p>digits for rounding</p>
</td></tr>
<tr><td><code id="methods_+3A_show">show</code></td>
<td>
<p>do not print</p>
</td></tr>
<tr><td><code id="methods_+3A_conf">conf</code></td>
<td>
<p> coverage for confidence intervals </p>
</td></tr>
<tr><td><code id="methods_+3A_what">what</code></td>
<td>
<p> type of plot desired: cumulative risk (&quot;cr&quot;), hazard ratio (&quot;hr&quot;, for marginal method only), cumulative risk ratio (&quot;crr&quot;, for moments method only) </p>
</td></tr>
<tr><td><code id="methods_+3A_min.age">min.age</code></td>
<td>
<p> Minimal age for plots </p>
</td></tr>
<tr><td><code id="methods_+3A_max.age">max.age</code></td>
<td>
<p> Maximal age for plots </p>
</td></tr>
<tr><td><code id="methods_+3A_age.start">age.start</code></td>
<td>
<p> initial age value (x) for plots </p>
</td></tr>
<tr><td><code id="methods_+3A_start.ref">start.ref</code></td>
<td>
<p> initial risk value (y) for plots </p>
</td></tr>
<tr><td><code id="methods_+3A_max.y">max.y</code></td>
<td>
<p> Max value for y axis </p>
</td></tr>
<tr><td><code id="methods_+3A_type">type</code></td>
<td>
<p> type of line in plots </p>
</td></tr>
<tr><td><code id="methods_+3A_add">add</code></td>
<td>
<p> If TRUE, then lines are added to current plot. Useful to compare analyses.</p>
</td></tr>
<tr><td><code id="methods_+3A_color">color</code></td>
<td>
<p> change line colors using a vector of values </p>
</td></tr>
<tr><td><code id="methods_+3A_line">line</code></td>
<td>
<p> change line width using a vector of values </p>
</td></tr>
<tr><td><code id="methods_+3A_median">median</code></td>
<td>
<p> plot median of bootstrap samples instead of point estimates  </p>
</td></tr>
<tr><td><code id="methods_+3A_...">...</code></td>
<td>
<p> additional arguments for print or plot </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Specific output and plot types can be selected with arguments
</p>

<hr>
<h2 id='simulations'> simulation of kin cohort studies </h2><span id='topic+kc.simul'></span><span id='topic+sample.caco'></span><span id='topic+summary.kin.cohort.sample'></span>

<h3>Description</h3>

<p>Functions to simulate data for kin-cohort analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
kc.simul(nfam, f, hr, rand = 0, mean.sibs = 2, mean.desc = 1.5, 
         a.age = 8, b.age = 80, a.cancer = 3, b.cancer = 180 )

sample.caco(object, p.cases = 1, caco.ratio = 1, verbose = TRUE)

## S3 method for class 'kin.cohort.sample'
summary(object,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulations_+3A_nfam">nfam</code></td>
<td>
<p>number of families to be generated </p>
</td></tr>
<tr><td><code id="simulations_+3A_f">f</code></td>
<td>
<p>allele frequency</p>
</td></tr>
<tr><td><code id="simulations_+3A_hr">hr</code></td>
<td>
<p>hazard ratio for disease carriers relative noncarriers</p>
</td></tr>
<tr><td><code id="simulations_+3A_rand">rand</code></td>
<td>
<p>variance of random effect for cancer incidence (ratio of hr)</p>
</td></tr>
<tr><td><code id="simulations_+3A_mean.sibs">mean.sibs</code></td>
<td>
<p> mean number of sibllings and descendants (~Poisson)</p>
</td></tr>
<tr><td><code id="simulations_+3A_mean.desc">mean.desc</code></td>
<td>
<p> mean number of sibllings and descendants (~Poisson)</p>
</td></tr>
<tr><td><code id="simulations_+3A_a.age">a.age</code></td>
<td>
<p>shape parameter for age (~Weibull)</p>
</td></tr>
<tr><td><code id="simulations_+3A_b.age">b.age</code></td>
<td>
<p>scale parameter for age (~Weibull)</p>
</td></tr>
<tr><td><code id="simulations_+3A_a.cancer">a.cancer</code></td>
<td>
<p>shape parameter for cancer incidence (~Weibull)</p>
</td></tr>
<tr><td><code id="simulations_+3A_b.cancer">b.cancer</code></td>
<td>
<p>scale parameter for cancer incidence (~Weibull)</p>
</td></tr>
<tr><td><code id="simulations_+3A_object">object</code></td>
<td>
<p>object of class <code>kin.cohort.sample</code> and <code>data.frame</code> </p>
</td></tr>
<tr><td><code id="simulations_+3A_p.cases">p.cases</code></td>
<td>
<p>proportion of cases (affected) to include in sample. if more than 1, the exact number is assumed</p>
</td></tr>
<tr><td><code id="simulations_+3A_caco.ratio">caco.ratio</code></td>
<td>
<p> ratio of controls per case to include in sample </p>
</td></tr>
<tr><td><code id="simulations_+3A_verbose">verbose</code></td>
<td>
<p>show the number of cases and controls sampled</p>
</td></tr>
<tr><td><code id="simulations_+3A_...">...</code></td>
<td>
<p>additional arguments</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>kc.simul</code> will generate a cohort of probands of size <code>nfam</code>. Default parameters simulate a typical cancer study. Each of them will be assigned: a <code>carrier</code> status with probability <code class="reqn">f^2+2f(1-f)</code>; a current <code>age</code> 
drawn from a Weibull distribution with parameters <code>a.age</code> and <code>b.age</code>; an age at diagnosis (<code>agecancer</code>) drawn from 
a Weibull distribution with parameters <code>a.cancer</code> and <code>b.cancer</code>, if noncarrier. For carries, the scale (<code>b.cancer</code>) is 
shifted to get the desired hazard ratio (<code>hr</code>). If <code>rand</code>&gt;0, then a family specific random effect is also added, drawn from a normal distribution with mean 0 and sd <code>rand</code>.
If <code>agecancer</code>&lt; <code>age</code> then the disease status (<code>cancer</code>) will be 1, 0 otherwise. 
</p>
<p>First degree relatives are generated for each proband: two parents, a random number of sibblings (drawn from a Poisson withe mean <code>mean.sibs</code>), and 
a random number of descendants (drawn from a Poisson with mean <code>mean.desc</code>). Each of them is assiggned a <code>carrier</code> status with probability according 
to mendelian transmission conditional of the proband carrier status. 
Current <code>age</code> for relatives are generated conditional on the proband's age, with random draws from normal distribution. Age at diagnosis (<code>agecancer</code>) is assumed independent, except for the optional family random effect.
Gender is assigned at random with probability 0.5 for all individuals.
</p>
<p>Note that the simulation of residual familial correlation with a random effect (<code>rand$&gt;0</code>) does not mantain the desired hazard ratio (<code>hr</code>).
</p>
<p>The generic function <code>summary</code> will show the number and proportion of carriers and affected subjects in the sample.
</p>
<p><code>sample.caco</code> will sample (from a simulation generated by <code>kc.simul</code>) a subset of cases (afected probands) and controls (unaffected probands) and their relatives. Currently only random sampling of controls is implemented (no matching). Sampling fraction is controled by <code>caco.ratio</code>.
</p>
<p>Currently, only one gene and one disease are simulated.
</p>


<h3>Value</h3>

<p>object of class <code>kin.cohort.sample</code> and <code>data.frame</code> with fields
</p>
<table>
<tr><td><code>famid</code></td>
<td>
<p>family id</p>
</td></tr>
<tr><td><code>rel</code></td>
<td>
<p> relative type (0=proband, 1=parents, 2=sibblings, 3=descendants)</p>
</td></tr>
<tr><td><code>age</code></td>
<td>
<p> current age of each subject</p>
</td></tr>
<tr><td><code>gender</code></td>
<td>
<p>gender (0=male, 1=female)</p>
</td></tr>
<tr><td><code>carrier</code></td>
<td>
<p>carrier status of proband (0=noncarrier, 1=carrier), common for all family members </p>
</td></tr>
<tr><td><code>cancer</code></td>
<td>
<p>affected (0=no, 1=yes) </p>
</td></tr>
<tr><td><code>agecancer</code></td>
<td>
<p> age at diagnosis or current age if not affected</p>
</td></tr>
<tr><td><code>real.carrier</code></td>
<td>
<p> carrier status or relatives (0=noncarrier, 1=carrier ) </p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed(7)
## cohort 
s&lt;-kc.simul(4000, f=0.03, hr=5)
summary(s)

## exclude probands
m.coh&lt;- kc.marginal(s$agecancer, s$cancer, factor(s$carrier), s$rel,
                    knots=c(30,40,50,60,70,80,90), f=0.03)
m.coh

## relatives only
r.coh&lt;- coxph(Surv(agecancer,cancer)~real.carrier, data=s)
print(exp(coef(r.coh)))

## probands only
p.coh&lt;- coxph(Surv(agecancer,cancer)~carrier, data=s)
print(exp(coef(p.coh)))

## case-control
s.cc&lt;- sample.caco(s)
summary(s.cc)

## exclude probands
m.caco&lt;- kc.marginal(s.cc$agecancer, s.cc$cancer, factor(s.cc$carrier), 
                     s.cc$rel, knots=c(30,40,50,60,70,80,90), f=0.03)
m.caco

## relatives only
r.caco&lt;- glm(cancer~real.carrier, family=binomial, data=s.cc, subset=(s.cc$rel!=0))
print(exp(coef(r.caco)[2]))

## probands only
p.caco&lt;- glm(cancer~carrier, family=binomial, data=s.cc, subset=(s.cc$rel==0))
print(exp(coef(p.caco)[2]))

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
