<!DOCTYPE html><html><head><title>Help for package kindisperse</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kindisperse}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#kindisperse-package'><p>kindisperse: Simulate and Estimate Close-Kin Dispersal Kernels</p></a></li>
<li><a href='#access_sigmas'><p>Access or assign dispersal sigmas of <code>KinPairSimulation</code> objects</p></a></li>
<li><a href='#axials'><p>Estimate the axial dispersal distance of a kernel</p></a></li>
<li><a href='#axials_add'><p>Add axial distributions</p></a></li>
<li><a href='#axials_combine'><p>Combine axial distributions to produce a mixed distribution</p></a></li>
<li><a href='#axials_decompose'><p>Decompose an axial distribution into simple components</p></a></li>
<li><a href='#axials_standard'><p>Calculate the intergenerational (PO) dispersal kernel from the distributions of close kin</p></a></li>
<li><a href='#axials_subtract'><p>Subtract axial distributions</p></a></li>
<li><a href='#axpermute'><p>Estimate the axial dispersal distance of a kernel with confidence intervals</p></a></li>
<li><a href='#axpermute_standard'><p>Calculate the intergenerational (PO) dispersal kernel from the distributions of close kin (bootstrapped)</p></a></li>
<li><a href='#axpermute_subtract'><p>Subtract axial distributions with confidence intervals</p></a></li>
<li><a href='#breeding_cycle'><p>Access breeding cycle at sampling of <code>DispersalModel</code> object.</p></a></li>
<li><a href='#breeding_stage'><p>Access life stage at which breeding occurs of <code>DispersalModel</code> object</p></a></li>
<li><a href='#check_valid_kinship'><p>Check valid kinship</p></a></li>
<li><a href='#check_valid_lifestage'><p>Check valid lifestage</p></a></li>
<li><a href='#csv_to_kinpair'><p>Reads <code>.csv</code> and converts to <code>KinPairData</code> object</p></a></li>
<li><a href='#df_to_kinpair'><p>Convert dataframe or tibble to <code>KinPairData</code> class</p></a></li>
<li><a href='#dispersal_model'><p>Create Dispersal Model of an Organism</p></a></li>
<li><a href='#dispersal_vector'><p>Access dispersal vector of <code>DispersalModel</code> object.</p></a></li>
<li><a href='#DispersalModel-class'><p>DispersalModel Class</p></a></li>
<li><a href='#display_appdata'><p>Show printout of named items stored in appdata.</p></a></li>
<li><a href='#distances'><p>Access or assign <code>distances</code> category of <code>KinPairData</code> class objects</p></a></li>
<li><a href='#elongate'><p>Change the shape (aspect ratio) of a rectangle while preserving area</p></a></li>
<li><a href='#filter_methods'><p>Access or modify the <code>filter</code> parameters of <code>KinPairSimulation</code> objects</p></a></li>
<li><a href='#filtertype'><p>Access filtertype of <code>KinPairSimulation</code> object</p></a></li>
<li><a href='#fs'><p>Access FS phase split point of <code>DispersalModel</code> object.</p></a></li>
<li><a href='#get_dispersal_model'><p>Access dispersal model of <code>KinPairSimulation</code> object</p></a></li>
<li><a href='#hs'><p>Access HS phase split point of <code>DispersalModel</code> object.</p></a></li>
<li><a href='#is.DispersalModel'><p>Check if object is of class <code>DispersalModel</code></p></a></li>
<li><a href='#is.KinPairData'><p>Check if object is of class KinPairData</p></a></li>
<li><a href='#is.KinPairSimulation'><p>Check if object is of class KinPairSimulation</p></a></li>
<li><a href='#kernelshape'><p>Access kernel type of <code>KinPairSimulation</code> object</p></a></li>
<li><a href='#kerneltype'><p>Access or assign kerneltype of <code>KinPairSimulation</code> object</p></a></li>
<li><a href='#kin_pair_data'><p>Make new KinPairData object</p></a></li>
<li><a href='#kin_pair_simulation'><p>KinPairSimulation</p></a></li>
<li><a href='#kinpair_to_csv'><p>Write <code>KinPairData</code> object to <code>.csv</code> format</p></a></li>
<li><a href='#kinpair_to_tibble'><p>Extract KinPairData class object to tibble</p></a></li>
<li><a href='#kinpair_to_tsv'><p>Write <code>KinPairData</code> object to <code>.tsv</code> format</p></a></li>
<li><a href='#KinPairData-class'><p>Formal class KinPairData</p></a></li>
<li><a href='#KinPairSimulation_composite'><p>Constructor for KinPairSimulation Class (composite)</p></a></li>
<li><a href='#KinPairSimulation_custom'><p>Constructor for KinPairSimulation Class (custom)</p></a></li>
<li><a href='#KinPairSimulation_simple'><p>Constructor for KinPairSimulation Class (simple)</p></a></li>
<li><a href='#KinPairSimulation-class'><p>KinPairSimulation Class</p></a></li>
<li><a href='#kinship'><p>Access or assign <code>kinship</code> category of <code>KinPairData</code> class objects</p></a></li>
<li><a href='#lifestage'><p>Access or assign <code>lifestage</code> category of <code>KinPairData</code> class objects</p></a></li>
<li><a href='#mentari'><p>Position &amp; kinship information of <em>Aedes aegypti</em> from Mentari Court, Malaysia</p></a></li>
<li><a href='#mount_appdata'><p>Mount <code>KinPairData</code> Objects for use in kindisperse app</p></a></li>
<li><a href='#read_kindata'><p>Reads <code>.kindata</code> filetype back to <code>KinPairData</code> or <code>KinPairSimulation</code> object.</p></a></li>
<li><a href='#rebase_dims'><p>Change the dimensions of a KinPairSimulation Object and shift kinpairs so at least one individual is within the area</p></a></li>
<li><a href='#reset_appdata'><p>Reset kindisperse appdata</p></a></li>
<li><a href='#reset_tempdata'><p>Reset app tempdata (internal mem)</p></a></li>
<li><a href='#retrieve_appdata'><p>Retrieve <code>KinPairData</code> object from appdata (single)</p></a></li>
<li><a href='#retrieve_tempdata'><p>Retrieve all tempdata (internal mem) from app (as list)</p></a></li>
<li><a href='#retrieveall_appdata'><p>Retrieve all <code>KinPairData</code> objects from appdata (as list)</p></a></li>
<li><a href='#run_kindisperse'><p>Run kindisperse app</p></a></li>
<li><a href='#sample_kindist'><p>Subsample and filter a <code>KinPairSimulation</code> or <code>KinPairData</code> Object</p></a></li>
<li><a href='#sampling_stage'><p>Access sampling stage of <code>DispersalModel</code> or <code>KinPairSimulation</code> object.</p></a></li>
<li><a href='#simdims'><p>Access simulation dimensions of <code>KinPairSimulation</code> object</p></a></li>
<li><a href='#simgraph_data'><p>Simple kin dispersal simulation for graphical display. (returns the data side as a tibble).</p></a></li>
<li><a href='#simgraph_graph'><p>Simple kin dispersal simulation for graphical display. (graphs the pre-existing simulation).</p></a></li>
<li><a href='#simtype'><p>Access or assign simulation type of <code>KinPairSimulation</code> object</p></a></li>
<li><a href='#simulate_kindist_composite'><p>Simulate kin dispersal distance pairs with composite sigmas</p></a></li>
<li><a href='#simulate_kindist_custom'><p>Simulate kin dispersal distance pairs with custom species dispersal models.</p></a></li>
<li><a href='#simulate_kindist_simple'><p>Simulate kin dispersal distance pairs with simple sigma</p></a></li>
<li><a href='#stages'><p>Access breeding cycle stages of <code>DispersalModel</code> object.</p></a></li>
<li><a href='#tsv_to_kinpair'><p>Reads <code>.tsv</code> and converts to <code>KinPairData</code> object</p></a></li>
<li><a href='#unmount_appdata'><p>Unmount a <code>KinPairData</code> Object (clear slot from appdata environment)</p></a></li>
<li><a href='#vector_to_kinpair'><p>Convert vector of kin separation distances to <code>KinPairData</code> class</p></a></li>
<li><a href='#visible_stage'><p>Access life stage at which individual is first visible to sampling (i.e. from which breeding cycles are calculated)</p></a></li>
<li><a href='#write_kindata'><p>Write <code>KinPairData</code> or <code>KinPairSimulation</code> object in <code>.kindata</code> format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Simulate and Estimate Close-Kin Dispersal Kernels</td>
</tr>
<tr>
<td>Version:</td>
<td>0.10.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for simulating and estimating kinship-related dispersal. Based
    on the methods described in M. Jasper, T.L. Schmidt., N.W. Ahmad, S.P. Sinkins &amp; A.A. 
    Hoffmann (2019) &lt;<a href="https://doi.org/10.1111%2F1755-0998.13043">doi:10.1111/1755-0998.13043</a>&gt; "A genomic approach to inferring kinship 
    reveals limited intergenerational dispersal in the yellow fever mosquito".
    Assumes an additive variance model of dispersal in two dimensions, compatible with 
    Wright's neighbourhood area. Simple and composite dispersal simulations are supplied, 
    as well as the functions needed to estimate parent-offspring dispersal for simulated or 
    empirical data, and to undertake sampling design for future field studies of dispersal. 
    For ease of use an integrated Shiny app is also included. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/moshejasper/kindisperse">https://github.com/moshejasper/kindisperse</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, readr, shiny, shinythemes, ggrepel, fitdistrplus,
LaplacesDemon, here, stats, methods, tibble, grid, magrittr,
plotly, dplyr, rlang, tidyselect, stringr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Collate:</td>
<td>'DispersalModel.R' 'KinPairData.R' 'dispersal_model.R'
'KinPairSimulation.R' 'app.R' 'app_ports.R'
'axial_helper_functions.R' 'axials_standard.R' 'data.R'
'export_functions.R' 'import_kinpairs.R'
'kindisperse-package.R' 'sample_kindist.R' 'simgraph_data.R'
'simgraph_graph.R' 'simulate_kindist_composite.R'
'simulate_kindist_custom.R' 'simulate_kindist_simple.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-07-28 08:32:08 UTC; mossu</td>
</tr>
<tr>
<td>Author:</td>
<td>Moshe-Elijah Jasper
    <a href="https://orcid.org/0000-0003-4541-3223"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Moshe-Elijah Jasper &lt;moshe.jasper@unimelb.edu.au&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-07-28 09:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='kindisperse-package'>kindisperse: Simulate and Estimate Close-Kin Dispersal Kernels</h2><span id='topic+kindisperse'></span><span id='topic+kindisperse-package'></span>

<h3>Description</h3>

<p>Functions for simulating and estimating kinship-related dispersal. Based
on the methods described in M. Jasper, T.L. Schmidt., N.W. Ahmad, S.P. Sinkins &amp; A.A. 
Hoffmann (2019) &lt;doi:10.1111/1755-0998.13043&gt; &quot;A genomic approach to inferring kinship 
reveals limited intergenerational dispersal in the yellow fever mosquito&quot;.
Assumes an additive variance model of dispersal in two dimensions, compatible with 
Wright's neighbourhood area. Simple and composite dispersal simulations are supplied, 
as well as the functions needed to estimate parent-offspring dispersal for simulated or 
empirical data, and to undertake sampling design for future field studies of dispersal. 
For ease of use an integrated Shiny app is also included.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Moshe-Elijah Jasper <a href="mailto:moshe.jasper@unimelb.edu.au">moshe.jasper@unimelb.edu.au</a> (<a href="https://orcid.org/0000-0003-4541-3223">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/moshejasper/kindisperse">https://github.com/moshejasper/kindisperse</a>
</p>
</li></ul>


<hr>
<h2 id='access_sigmas'>Access or assign dispersal sigmas of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> objects</h2><span id='topic+access_sigmas'></span><span id='topic+posigma'></span><span id='topic+posigma+3C-'></span><span id='topic+initsigma'></span><span id='topic+initsigma+3C-'></span><span id='topic+breedsigma'></span><span id='topic+breedsigma+3C-'></span><span id='topic+gravsigma'></span><span id='topic+gravsigma+3C-'></span><span id='topic+ovisigma'></span><span id='topic+ovisigma+3C-'></span><span id='topic+posigma+2CKinPairSimulation-method'></span><span id='topic+initsigma+2CKinPairSimulation-method'></span><span id='topic+breedsigma+2CKinPairSimulation-method'></span><span id='topic+gravsigma+2CKinPairSimulation-method'></span><span id='topic+ovisigma+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>These generics &amp; methods work with <code>KinPairSimulation</code> objects to access &amp; modify information about the dispersal
sigma parameters that define the stored simulation. The <code>posigma()</code> method accesses the single dispersal parameter
stored in a simulation with <code>simtype == "simple"</code>. The remaining parameters access the dispersal parameters stored
in a simulation with <code>simtype == "composite"</code>. The dispersal kernel sigma parameters of <code>simtype == "custom"</code>
simulations are not yet implemented here. Assignment operations currently only exist as generics (they are not yet applied
to the <code>KinPairSimulation</code> class).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>posigma(x)

posigma(x) &lt;- value

initsigma(x)

initsigma(x) &lt;- value

breedsigma(x)

breedsigma(x) &lt;- value

gravsigma(x)

gravsigma(x) &lt;- value

ovisigma(x)

ovisigma(x) &lt;- value

## S4 method for signature 'KinPairSimulation'
posigma(x)

## S4 method for signature 'KinPairSimulation'
initsigma(x)

## S4 method for signature 'KinPairSimulation'
breedsigma(x)

## S4 method for signature 'KinPairSimulation'
gravsigma(x)

## S4 method for signature 'KinPairSimulation'
ovisigma(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="access_sigmas_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="access_sigmas_+3A_value">value</code></td>
<td>
<p>new value to assign</p>
</td></tr>
<tr><td><code id="access_sigmas_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> value of specified sigma parameter or modified <code>KinPairSimulation</code> object
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>posigma,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>initsigma,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>breedsigma,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>gravsigma,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>ovisigma,KinPairSimulation-method</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpsmethods: 
<code><a href="#topic+filter_methods">filter_methods</a></code>,
<code><a href="#topic+kernelshape">kernelshape</a>()</code>,
<code><a href="#topic+kerneltype">kerneltype</a>()</code>,
<code><a href="#topic+simtype">simtype</a>()</code>
</p>

<hr>
<h2 id='axials'>Estimate the axial dispersal distance of a kernel</h2><span id='topic+axials'></span>

<h3>Description</h3>

<p>This function performs a basic estimation of axial dispersal for a numeric vector of distances between close kin dyads. The axial
dispersal distance returned is interpretable as the standard deviation of one dimension of a symmetric bivariate random distribution
centred on zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axials(valvect, composite = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axials_+3A_valvect">valvect</code></td>
<td>
<p>A numeric vector of distances between close kin OR an object of class <code><a href="#topic+KinPairData">KinPairData</a></code></p>
</td></tr>
<tr><td><code id="axials_+3A_composite">composite</code></td>
<td>
<p>numeric. The number of separate 'draws' (dispersal events)
from the kernel required to produce the final positions of the measured individuals.
For example, the displacement of a child from parent at the same lifestage would involve 1 draw and thus be composite = 1.
Two full siblings would be two draws (composite = 2) from the FS kernel. Non-symmetric relationships (e.g. AV, 1C)
should not be decomposed using this method, nor should any assumptions be made about different kernels
(e.g. the 1C relationship would appropriately be given the value 2, but not 4)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the value of the estimated axial dispersal distance of the kernel producing the dispersal distances measured. (numeric)
</p>


<h3>See Also</h3>

<p>Other axial_helpers: 
<code><a href="#topic+axials_add">axials_add</a>()</code>,
<code><a href="#topic+axials_decompose">axials_decompose</a>()</code>,
<code><a href="#topic+axials_subtract">axials_subtract</a>()</code>,
<code><a href="#topic+axpermute_subtract">axpermute_subtract</a>()</code>,
<code><a href="#topic+axpermute">axpermute</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>po_dists &lt;- c(5, 6, 7.5)
axials(po_dists) # one 'draw' (dispersal event) goes into the parent offspring category
# so composite is left to its default of 1

fs_dists &lt;- c(2, 3, 3)
axials(fs_dists, composite = 2) # two 'draws' (symmetric dispersal events)
# go into the full sibling category so composite is set to 2
</code></pre>

<hr>
<h2 id='axials_add'>Add axial distributions</h2><span id='topic+axials_add'></span>

<h3>Description</h3>

<p>Add axial distributions. Useful to construct an overall distribution that results from multiple 'draws'
from smaller distributions. E.g. The pathway between first cousins which can be found by adding
each of the component distributions of their respective lifespans along with the relevant offspring producing
(e.g. oviposition) of the parent.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axials_add(axvals)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axials_add_+3A_axvals">axvals</code></td>
<td>
<p>numeric. vector of axial distribution values from different kernels that are to be added.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> Returns the axial value that results from adding the input axial values under an additive variance framework.
</p>


<h3>See Also</h3>

<p>Other axial_helpers: 
<code><a href="#topic+axials_decompose">axials_decompose</a>()</code>,
<code><a href="#topic+axials_subtract">axials_subtract</a>()</code>,
<code><a href="#topic+axials">axials</a>()</code>,
<code><a href="#topic+axpermute_subtract">axpermute_subtract</a>()</code>,
<code><a href="#topic+axpermute">axpermute</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fullsibs_ax &lt;- 5
parent_offspring_ax &lt;- 25
cousin_ax &lt;- axials_add(c(fullsibs_ax, parent_offspring_ax))
</code></pre>

<hr>
<h2 id='axials_combine'>Combine axial distributions to produce a mixed distribution</h2><span id='topic+axials_combine'></span>

<h3>Description</h3>

<p>combines axial distributions to produce a mixed distribution.
This is useful in settings where you have two separate distributions (e.g. FS &amp; HS) with their own axial values,
but you want to average them appropriately so that they can be compared to e.g.
a mixed distribution of full &amp; half cousins which cannot be distinguished via kinship determination methods
and thus are best treated as an even mixture of the two categories. Different to adding dispersal events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axials_combine(axvals)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axials_combine_+3A_axvals">axvals</code></td>
<td>
<p>numeric. vector of axial distribution values from different kernels that are to be combined</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> Returns the axial value that results from combining the input axial values under an additive variance framework.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fullax &lt;- axials(c(2, 4, 5), composite = 2)
halfax &lt;- axials(c(6, 5, 7), composite = 2)
sibax &lt;- axials_combine(c(fullax, halfax))
</code></pre>

<hr>
<h2 id='axials_decompose'>Decompose an axial distribution into simple components</h2><span id='topic+axials_decompose'></span>

<h3>Description</h3>

<p>Decomposes an axial distribution into simple components. Note that this should only be used in the simplest situations.
It assumes all composite dispersal events are of identical magnitude and have happened equivalently to both branches of a 'symmetric'
pedigree leading to the final kin dyad. (it can be used to derive e.g.full-sibling dispersal parameters from the distribution of
full-siblings, or equivalent for first cousins, but <strong>not</strong> to divide the 'avuncular' kernel into its
component parts (uncle/aunt &amp; niece/nephew have different dispersal paths from their common ancestor)).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axials_decompose(ax, n_composites = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axials_decompose_+3A_ax">ax</code></td>
<td>
<p>numeric. The axial value to be decomposed.</p>
</td></tr>
<tr><td><code id="axials_decompose_+3A_n_composites">n_composites</code></td>
<td>
<p>numeric. The number of separate 'draws' (dispersal events)
from the kernel required to produce the final positions of the measured individuals.
For example, the displacement of a child from parent at the same life stage would involve 1 draw and thus be composite = 1.
Two full siblings would be two draws (composite = 2) from the FS kernel. Non-symmetric relationships (e.g. AV, 1C)
should not be decomposed using this method, nor should any assumptions be made about different kernels
(e.g. the 1C relationship would appropriately be given the value 2, but not 4)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the (numeric) axial distribution value of the underlying dispersal kernel from which the composite kernel was (or could be) created.
</p>


<h3>See Also</h3>

<p>Other axial_helpers: 
<code><a href="#topic+axials_add">axials_add</a>()</code>,
<code><a href="#topic+axials_subtract">axials_subtract</a>()</code>,
<code><a href="#topic+axials">axials</a>()</code>,
<code><a href="#topic+axpermute_subtract">axpermute_subtract</a>()</code>,
<code><a href="#topic+axpermute">axpermute</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fs_vect &lt;- c(10, 11, 12)
fs_axial_raw &lt;- axials(fs_vect, composite = 1) # composite hasn't corrected for two dispersal events
# inherent to this kin category!
fs_axial_final &lt;- axials_decompose(fs_axial_raw, n_composites = 2)
</code></pre>

<hr>
<h2 id='axials_standard'>Calculate the intergenerational (PO) dispersal kernel from the distributions of close kin</h2><span id='topic+axials_standard'></span>

<h3>Description</h3>

<p>This function takes (at least) two vectors of kinship dispersal distances from defined kinship categories, and returns a resulting calculation of the parent-offspring
(intergenerational) kinship dispersal kernel. Dispersal distances can be inputted as numeric vectors, or alternatively as objects of classes <code><a href="#topic+KinPairData">KinPairData</a></code>
or <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axials_standard(
  avect,
  bvect,
  acat = NULL,
  bcat = NULL,
  amix = FALSE,
  bmix = FALSE,
  amixcat = NULL,
  bmixcat = NULL,
  acomp = FALSE,
  bcomp = FALSE,
  acompvect = NULL,
  bcompvect = NULL,
  acompcat = NULL,
  bcompcat = NULL,
  acycle = NULL,
  bcycle = NULL,
  amixcycle = NULL,
  bmixcycle = NULL,
  acompcycle = NULL,
  bcompcycle = NULL,
  override = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axials_standard_+3A_avect">avect</code></td>
<td>
<p>vector a of kin dispersal distances for the less closely related kinship category OR object of class <code>KinPairData</code>.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bvect">bvect</code></td>
<td>
<p>vector b of kin dispersal distances for the more closely related kinship category OR object of class <code>KinPairData</code>.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_acat">acat</code></td>
<td>
<p>kinship category of kin dispersal vector avect. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bcat">bcat</code></td>
<td>
<p>kinship category of kin dispersal vector bvect. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_amix">amix</code></td>
<td>
<p>logical describing whether vector a is a mixture of two kinship categories. Used with amixcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bmix">bmix</code></td>
<td>
<p>logical describing whether vector b is a mixture of two kinship categories. Used with bmixcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_amixcat">amixcat</code></td>
<td>
<p>mixture kinship category of vector a. Must be set if amix == TRUE. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bmixcat">bmixcat</code></td>
<td>
<p>mixture kinship category of vector b. Must be set if bmix == TRUE. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_acomp">acomp</code></td>
<td>
<p>logical denoting whether vector a should be composited with an additional kinship category vector. Used with acompvect and acompcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bcomp">bcomp</code></td>
<td>
<p>logical denoting whether vector b should be composited with an additional kinship category vector. Used with bcompvect and bcompcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_acompvect">acompvect</code></td>
<td>
<p>vector acomp of kin dispersal distances for compositing with vector a OR object of class KinPairData. Must be set if acomp == TRUE.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bcompvect">bcompvect</code></td>
<td>
<p>vector bcomp of kin dispersal distances for compositing with vector b OR object of class KinPairData. Must be set if bcomp == TRUE.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_acompcat">acompcat</code></td>
<td>
<p>kinship category of kin dispersal vector acompvect. Must be set if acomp == TRUE.  Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bcompcat">bcompcat</code></td>
<td>
<p>kinship category of kin dispersal vector bcompvect. Must be set if bcomp == TRUE. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_acycle">acycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector avect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bcycle">bcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector bvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_amixcycle">amixcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector amixvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bmixcycle">bmixcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector bmixvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_acompcycle">acompcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector acompvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_bcompcycle">bcompcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector bcompvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axials_standard_+3A_override">override</code></td>
<td>
<p>whether or not to override the default -1 cycle compatibility check (default FALSE) override in situations where you are confident e.g. a c(-1, -1) cycle FS or HS category is truly zeroed (&amp; thus separated from later stages by a complete lifespan)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This (with its paired function <code><a href="#topic+axpermute_standard">axpermute_standard</a></code>) are the core functions implemented in the <code>kindisperse</code> package. They enable the decomposition of the
pedigree &amp; dispersal information contained in the sampled distributions of close kin dyads (full siblings, first cousins, etc.) &amp; its leveraging within an additive dispersal
framework to estimate the key intergenerational (parent-offspring) dispersal parameter of a population. Four key ideas underpin the approach in this function: (a) tracing
dispersal pedigrees to determine the number of complete intergenerational (breeding-cycle-spanning) dispersal events separating the sampled close-kin dyads; (b) using kin categories
that share the same overarching kinship 'phase' to control for residual 'phased' (non-intergenerational) disperal events that occur at the pedigree branch point (e.g. ovipositional
dispersal for full sibling mosquitoes), and (c) using synced or equivalent sampling points to eliminate non-intergenerational dispersal at the branch-tips of the pedigrees, then
finally (d) decomposing the 'pure' pedigree-associated (intergenerational) dispersal into an estimae of the single-generation intergenerational dispersal parameter.
</p>
<p>At its most basic, this function requires information about two dispersal vectors, a &amp; b - both of a phased kinship category, &amp; <code>vector a</code> having a more dispersed pedigree
than <code>vector b</code>. In addition to this initial pair of dispersed kin categories, either one or another matched pair of kin categories can be added:
</p>

<ol>
<li><p> A mixture category. This redefines the vector it is paired with (either a or b) so that rather than being considered as a 'pure' pedigree variant, it is considered as mixed with a
different kin category, often of a differing pedigree phase. If used, the other initial vector must also be paired with a related mixture category or composite vector.
</p>
</li>
<li><p> A composite dispersal vector. This is defined exactly as the initial dispersal vectors. After calculation, the axial value found is composited with that of the matched initial vector, and
its kinship category redefined as a mixture category as above. If used, the other initial vector must also be paired with a related mixture category or composite vector. These can be paired so
that a mixture category (e.g. first &amp; half-first cousins where these could not be separated with available genetic data) can be counterbalanced with the composition of full sibling &amp; half-sibling
dyads, which (assuming equal mixture) approximately controls for the phasing of the mixed kin categories, enabling an estimate of intergenerational dispersal without exact knowledge of the
composition of the cousins distribution.
</p>
</li></ol>

<p>Each vector or <code>KinPairData</code> / <code>KinPairSimulation</code> object is paired with several other parameters: (1) a logical (e.g. <code>amix</code> delineating whether the category is being
used in the calculation, (2) a category parameter (.e.g <code>acat</code>) defining what kin relationship is being measured, (3) an optional breeding cycle number (e.g. <code>acycle</code>)
showing the number of breeding cycles each member of the kin pair has passed through before being sampled (the cycle vector <code>c(1, 0)</code> corresponds to an adult &amp; a juvenile
being sampled at the same point in the breeding cycle; <code>c(1, 1)</code> represents two adults (i.e. after their first breeding), etc.) . If a <code><a href="#topic+KinPairData">KinPairData</a></code> or
<code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object is inputted, all paired parameters that are not explicitly set will default to those contained in the objects (using KinPair objects
is the ideal way to deploy this function).
</p>
<p>For further information on this function, package &amp; the dispersal estimation method it represents, see the paper by Jasper et al. - &quot;A genomic approach to inferring kinship reveals
limited intergenerational dispersal in the yellow fever mosquito&quot;, doi: <a href="https://doi.org/10.1111/1755-0998.13043">10.1111/1755-0998.13043</a>.
</p>


<h3>Value</h3>

<p>Returns a numeric estimate of PO (intergenerational) dispersal kernel axial distribution.
</p>


<h3>See Also</h3>

<p>Other axstandard: 
<code><a href="#topic+axpermute_standard">axpermute_standard</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cous &lt;- rexp(100, 1 / 100)
fullsibs &lt;- rexp(50, 1 / 50)
axials_standard(cous, fullsibs, acat = "1C", bcat = "FS")
</code></pre>

<hr>
<h2 id='axials_subtract'>Subtract axial distributions</h2><span id='topic+axials_subtract'></span>

<h3>Description</h3>

<p>Subtract axial distributions, finding the difference (under an additive variance framework).
This is most useful when one distribution subsumes another and includes a unique dispersal event that needs to be extracted.
For example, the FS category is subsumed by the 1C category, which can be written 'FS + PO'.
In this circumstance, subtracting FS from 1C will yield an estimate of the PO kernel (the basic intergenerational dispersal kernel)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axials_subtract(abig, asmall)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axials_subtract_+3A_abig">abig</code></td>
<td>
<p>numeric. The axial dispersal distance of the larger (subsuming) distribution (e.g. 1C).</p>
</td></tr>
<tr><td><code id="axials_subtract_+3A_asmall">asmall</code></td>
<td>
<p>numeric. The axial dispersal distance of the smaller (subsumed) distribution (e.g. FS).</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric</code> Returns an estimate of the axial dispersal distance of those dispersal elements that are unique to the larger dispersal distribution (e.g. PO).
</p>


<h3>See Also</h3>

<p>Other axial_helpers: 
<code><a href="#topic+axials_add">axials_add</a>()</code>,
<code><a href="#topic+axials_decompose">axials_decompose</a>()</code>,
<code><a href="#topic+axials">axials</a>()</code>,
<code><a href="#topic+axpermute_subtract">axpermute_subtract</a>()</code>,
<code><a href="#topic+axpermute">axpermute</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>axials_subtract(100, 70)
</code></pre>

<hr>
<h2 id='axpermute'>Estimate the axial dispersal distance of a kernel with confidence intervals</h2><span id='topic+axpermute'></span>

<h3>Description</h3>

<p>This function performs an estimation of axial dispersal for a numeric vector of distances between close kin dyads with confidence
intervals. The axial dispersal distance returned is interpretable as the standard deviation of one dimension of a
symmetric bivariate random distribution centred on zero. Confidence intervals are assigned via bootstrapping, or optionally the
vector of all bootstrapped results can be outputted by setting <code>output</code> to <code>'vect'</code>, enabling its passing to other
functions or external statistical analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axpermute(vals, nreps = 1000, nsamp = "std", composite = 1, output = "confs")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axpermute_+3A_vals">vals</code></td>
<td>
<p>numeric. Vector of distances between close kin OR object of class KinPairData.</p>
</td></tr>
<tr><td><code id="axpermute_+3A_nreps">nreps</code></td>
<td>
<p>numeric. Number of permutations to run for confidence intervals (default 1000)</p>
</td></tr>
<tr><td><code id="axpermute_+3A_nsamp">nsamp</code></td>
<td>
<p>numeric. Number of kin pairs to subsample for each permutation.
Either &quot;std&quot; or an integer. If &quot;std&quot; will be computed as equal to the sample size. (default &quot;std&quot;)</p>
</td></tr>
<tr><td><code id="axpermute_+3A_composite">composite</code></td>
<td>
<p>numeric. The number of separate 'draws' (dispersal events)
from the kernel required to produce the final positions of the measured individuals.
For example, the displacement of a child from parent at the same lifestage would involve 1 draw and thus be composite = 1.
Two full siblings would be two draws (composite = 2) from the FS kernel. Non-symmetric relationships (e.g. AV, 1C)
should not be decomposed using this method, nor should any assumptions be made about different kernels
(e.g. the 1C relationship would appropriately be given the value 2, but not 4)</p>
</td></tr>
<tr><td><code id="axpermute_+3A_output">output</code></td>
<td>
<p>character. Denotes what kind of output to return.
If 'confs', a vector of 95% confidence intervals. if 'vect', a vector of all permuted axial value results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If ouput = 'confs', returns a <code>numeric vector</code> of 95% confidence intervals and mean axial value.
if output = 'vect', returns a <code>numeric vector</code> of all permuted axial value results
</p>


<h3>See Also</h3>

<p>Other axial_helpers: 
<code><a href="#topic+axials_add">axials_add</a>()</code>,
<code><a href="#topic+axials_decompose">axials_decompose</a>()</code>,
<code><a href="#topic+axials_subtract">axials_subtract</a>()</code>,
<code><a href="#topic+axials">axials</a>()</code>,
<code><a href="#topic+axpermute_subtract">axpermute_subtract</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>po_dists &lt;- rexp(100, 1 / 50)
axpermute(po_dists, composite = 1)
</code></pre>

<hr>
<h2 id='axpermute_standard'>Calculate the intergenerational (PO) dispersal kernel from the distributions of close kin (bootstrapped)</h2><span id='topic+axpermute_standard'></span>

<h3>Description</h3>

<p>This function takes (at least) two vectors of kinship dispersal distances from defined kinship categories, and returns a resulting calculation of the parent-offspring
(intergenerational) kinship dispersal kernel with bootstrapped confidence intervals. Dispersal distances can be inputted as numeric vectors, or alternatively as objects of
classes <code><a href="#topic+KinPairData">KinPairData</a></code> or <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axpermute_standard(
  avect = NULL,
  bvect = NULL,
  acat = NULL,
  bcat = NULL,
  nreps = 1000,
  nsamp = "std",
  amix = FALSE,
  bmix = FALSE,
  amixcat = NULL,
  bmixcat = NULL,
  acomp = FALSE,
  bcomp = FALSE,
  acompvect = NULL,
  bcompvect = NULL,
  acompcat = NULL,
  bcompcat = NULL,
  acycle = NULL,
  bcycle = NULL,
  amixcycle = NULL,
  bmixcycle = NULL,
  acompcycle = NULL,
  bcompcycle = NULL,
  output = "confs",
  override = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axpermute_standard_+3A_avect">avect</code></td>
<td>
<p>vector a of kin dispersal distances for the less closely related kinship category OR object of class KinPairData.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bvect">bvect</code></td>
<td>
<p>vector b of kin dispersal distances for the more closely related kinship category OR object of class KinPairData.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_acat">acat</code></td>
<td>
<p>kinship category of kin dispersal vector avect. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bcat">bcat</code></td>
<td>
<p>kinship category of kin dispersal vector bvect. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_nreps">nreps</code></td>
<td>
<p>number of permutations to run for confidence intervals (default 1000)</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_nsamp">nsamp</code></td>
<td>
<p>number of kin pairs to subsample for each permutation. Either &quot;std&quot; or an integer. If &quot;std&quot; will be computed as equal to the sample size. (default &quot;std&quot;)</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_amix">amix</code></td>
<td>
<p>logical describing whether vector a is a mixture of two kinship categories. Used with amixcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bmix">bmix</code></td>
<td>
<p>logical describing whether vector b is a mixture of two kinship categories. Used with bmixcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_amixcat">amixcat</code></td>
<td>
<p>mixture kinship category of vector a. Must be set if amix == TRUE. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bmixcat">bmixcat</code></td>
<td>
<p>mixture kinship category of vector b. Must be set if bmix == TRUE. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_acomp">acomp</code></td>
<td>
<p>logical denoting whether vector a should be composited with an additional kinship category vector. Used with acompvect and acompcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bcomp">bcomp</code></td>
<td>
<p>logical denoting whether vector b should be composited with an additional kinship category vector. Used with bcompvect and bcompcat. Default FALSE.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_acompvect">acompvect</code></td>
<td>
<p>vector acomp of kin dispersal distances for compositing with vector a OR object of class KinPairData. Must be set if acomp == TRUE.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bcompvect">bcompvect</code></td>
<td>
<p>vector bcomp of kin dispersal distances for compositing with vector b OR object of class KinPairData. Must be set if bcomp == TRUE.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_acompcat">acompcat</code></td>
<td>
<p>kinship category of kin dispersal vector acompvect. Must be set if acomp == TRUE.  Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bcompcat">bcompcat</code></td>
<td>
<p>kinship category of kin dispersal vector bcompvect. Must be set if bcomp == TRUE. Must be one of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot;, &quot;HGAV&quot;, &quot;H1C&quot;, &quot;H1C1&quot;, &quot;H2C&quot;</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_acycle">acycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector avect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bcycle">bcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector bvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_amixcycle">amixcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector amixvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bmixcycle">bmixcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector bmixvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_acompcycle">acompcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector acompvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_bcompcycle">bcompcycle</code></td>
<td>
<p>breeding cycle number of kin dispersal vector bcompvect. Must be a nonnegative integer. (0, 1, 2, ...). Represents the number of complete breeding cycles the sampled individual has undergone before the checkpoint, where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_output">output</code></td>
<td>
<p>string denoting what kind of output to return. If 'confs', a vector of 95% confidence intervals. if 'vect', a vector of all permutated axial value results</p>
</td></tr>
<tr><td><code id="axpermute_standard_+3A_override">override</code></td>
<td>
<p>whether or not to override the default -1 cycle compatibility check (default FALSE) override in situations where you are confident e.g. a c(-1, -1) cycle FS or HS category is truly zeroed (&amp; thus separated from later stages by a complete lifespan)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This (with its paired function <code><a href="#topic+axials_standard">axials_standard</a></code>) are the core functions implemented in the <code>kindisperse</code> package. They enable the decomposition of the
pedigree &amp; dispersal information contained in the sampled distributions of close kin dyads (full siblings, first cousins, etc.) &amp; its leveraging within an additive dispersal
framework to estimate the key intergenerational (parent-offspring) dispersal parameter of a population. Four key ideas underpin the approach in this function: (a) tracing
dispersal pedigrees to determine the number of complete intergenerational (breeding-cycle-spanning) dispersal events separating the sampled close-kin dyads; (b) using kin categories
that share the same overarching kinship 'phase' to control for residual 'phased' (non-intergenerational) disperal events that occur at the pedigree branch point (e.g. ovipositional
dispersal for full sibling mosquitoes), and (c) using synced or equivalent sampling points to eliminate non-intergenerational dispersal at the branch-tips of the pedigrees, then
finally (d) decomposing the 'pure' pedigree-associated (intergenerational) dispersal into an estimae of the single-generation intergenerational dispersal parameter.
</p>
<p>At its most basic, this function requires information about two dispersal vectors, a &amp; b - both of a phased kinship category, &amp; <code>vector a</code> having a more dispersed pedigree
than <code>vector b</code>. In addition to this initial pair of dispersed kin categories, either one or another matched pair of kin categories can be added:
</p>

<ol>
<li><p> A mixture category. This redefines the vector it is paired with (either a or b) so that rather than being considered as a 'pure' pedigree variant, it is considered as mixed with a
different kin category, often of a differing pedigree phase. If used, the other initial vector must also be paired with a related mixture category or composite vector.
</p>
</li>
<li><p> A composite dispersal vector. This is defined exactly as the initial dispersal vectors. After calculation, the axial value found is composited with that of the matched initial vector, and
its kinship category redefined as a mixture category as above. If used, the other initial vector must also be paired with a related mixture category or composite vector. These can be paired so
that a mixture category (e.g. first &amp; half-first cousins where these could not be separated with available genetic data) can be counterbalanced with the composition of full sibling &amp; half-sibling
dyads, which (assuming equal mixture) approximately controls for the phasing of the mixed kin categories, enabling an estimate of intergenerational dispersal without exact knowledge of the
composition of the cousins distribution.
</p>
</li></ol>

<p>Each vector or <code>KinPairData</code> / <code>KinPairSimulation</code> object is paired with several other parameters: (1) a logical (e.g. <code>amix</code> delineating whether the category is being
used in the calculation, (2) a category parameter (.e.g <code>acat</code>) defining what kin relationship is being measured, (3) an optional breeding cycle number (e.g. <code>acycle</code>)
showing the number of breeding cycles each member of the kin pair has passed through before being sampled (the cycle vector <code>c(1, 0)</code> corresponds to an adult &amp; a juvenile
being sampled at the same point in the breeding cycle; <code>c(1, 1)</code> represents two adults (i.e. after their first breeding), etc.) . If a <code><a href="#topic+KinPairData">KinPairData</a></code> or
<code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object is inputted, all paired parameters that are not explicitly set will default to those contained in the objects (using KinPair objects
is the ideal way to deploy this function).
</p>
<p>Confidence intervals are assigned via bootstrapping, or optionally the
vector of all bootstrapped results can be outputted by setting <code>output</code> to <code>'vect'</code>, enabling its passing to other
functions or external statistical analysis.
</p>
<p>For further information on this function, package &amp; the dispersal estimation method it represents, see the paper by Jasper et al. - &quot;A genomic approach to inferring kinship reveals
limited intergenerational dispersal in the yellow fever mosquito&quot;, doi: <a href="https://doi.org/10.1111/1755-0998.13043">10.1111/1755-0998.13043</a>.
</p>


<h3>Value</h3>

<p>If output = 'confs' returns vector of 95% confidence intervals (with mean).
If output = 'vect' returns vector of individual axial estimates from each permutation
</p>


<h3>See Also</h3>

<p>Other axstandard: 
<code><a href="#topic+axials_standard">axials_standard</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cous &lt;- rexp(100, 1 / 100)
fullsibs &lt;- rexp(50, 1 / 50)
axpermute_standard(cous, fullsibs, acat = "1C", bcat = "FS")
</code></pre>

<hr>
<h2 id='axpermute_subtract'>Subtract axial distributions with confidence intervals</h2><span id='topic+axpermute_subtract'></span>

<h3>Description</h3>

<p>Finds the difference between two different empirical axial distributions with confidence intervals.
This is most useful when one distribution subsumes another and includes a unique dispersal event that needs to be extracted.
For example, the FS category is subsumed by the 1C category, which can be written 'FS + PO'.
In this circumstance, subtracting FS from 1C will yield an estimate of the PO kernel (the basic intergenerational dispersal kernel).
Confidence intervals are assigned via bootstrapping, or optionally the
vector of all bootstrapped results can be outputted by setting <code>output</code> to <code>'vect'</code>, enabling its passing to other
functions or external statistical analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>axpermute_subtract(
  bigvals,
  smallvals,
  nreps = 1000,
  nsamp = "std",
  composite = 2,
  output = "confs"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="axpermute_subtract_+3A_bigvals">bigvals</code></td>
<td>
<p>numeric. Vector of distance distributions of the larger (subsuming) distribution (e.g. 1C)  OR object of class KinPairData.</p>
</td></tr>
<tr><td><code id="axpermute_subtract_+3A_smallvals">smallvals</code></td>
<td>
<p>numeric. Vector of distance distributions of the smaller (subsumed) distribution (e.g. FS)  OR object of class KinPairData.</p>
</td></tr>
<tr><td><code id="axpermute_subtract_+3A_nreps">nreps</code></td>
<td>
<p>numeric. Number of permutations to perform when generating confidence intervals.</p>
</td></tr>
<tr><td><code id="axpermute_subtract_+3A_nsamp">nsamp</code></td>
<td>
<p>numeric. number of kin pairs to subsample for each permutation. Either &quot;std&quot; or an integer.
If &quot;std&quot; will be computed as equal to the sample size. (default &quot;std&quot;)</p>
</td></tr>
<tr><td><code id="axpermute_subtract_+3A_composite">composite</code></td>
<td>
<p>numeric. The number of separate 'draws' (dispersal events)
from the kernel required to produce the final positions of the measured individuals.
For example, the displacement of a child from parent at the same lifestage would involve 1 draw and thus be composite = 1.
Two full siblings would be two draws (composite = 2) from the FS kernel. Non-symmetric relationships (e.g. AV, 1C)
should not be decomposed using this method, nor should any assumptions be made about different kernels
(e.g. the 1C relationship would appropriately be given the value 2, but not 4)</p>
</td></tr>
<tr><td><code id="axpermute_subtract_+3A_output">output</code></td>
<td>
<p>character. What kind of output to return.
Either 'confs' (default -&gt; confidence intervals) or 'vect -&gt; vector of axial distances</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If output = 'confs' returns <code>numeric vector</code> of 95% confidence intervals and mean axial value.
If output = 'vect' returns <code>numeric vector</code> of individual axial estimates from each permutation
</p>


<h3>See Also</h3>

<p>Other axial_helpers: 
<code><a href="#topic+axials_add">axials_add</a>()</code>,
<code><a href="#topic+axials_decompose">axials_decompose</a>()</code>,
<code><a href="#topic+axials_subtract">axials_subtract</a>()</code>,
<code><a href="#topic+axials">axials</a>()</code>,
<code><a href="#topic+axpermute">axpermute</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>firstcous &lt;- rexp(100, 1 / 80)
fullsibs &lt;- rexp(100, 1 / 50)
axpermute_subtract(firstcous, fullsibs)
</code></pre>

<hr>
<h2 id='breeding_cycle'>Access breeding cycle at sampling of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.</h2><span id='topic+breeding_cycle'></span><span id='topic+breeding_cycle+2CDispersalModel-method'></span><span id='topic+breeding_cycle+2CKinPairData-method'></span>

<h3>Description</h3>

<p>Access breeding cycle at sampling of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breeding_cycle(x)

## S4 method for signature 'DispersalModel'
breeding_cycle(x)

## S4 method for signature 'KinPairData'
breeding_cycle(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="breeding_cycle_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="breeding_cycle_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="breeding_cycle_+3A_kinpairdata">KinPairData</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>integer(s) &gt;= -1</code> Breeding cycle numbers of modeled dispersed kin. Represents the number of complete
breeding cycles each indivdiual has undergone before the sampling point, where the time between birth and first
reproduction is coded as <code>0</code>, that between first and second reproduction <code>1</code>, etc.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li>
<li> <p><code>KinPairData</code>: 
</p>
</li></ul>

<hr>
<h2 id='breeding_stage'>Access life stage at which breeding occurs of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object</h2><span id='topic+breeding_stage'></span><span id='topic+breeding_stage+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>Access life stage at which breeding occurs of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>breeding_stage(x)

## S4 method for signature 'DispersalModel'
breeding_stage(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="breeding_stage_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="breeding_stage_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> life stage at which breeding occurs for modeled dispersed kin.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li></ul>

<hr>
<h2 id='check_valid_kinship'>Check valid kinship</h2><span id='topic+check_valid_kinship'></span>

<h3>Description</h3>

<p>Checks if vector of kinship categories contains all valid entries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_valid_kinship(vect)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_valid_kinship_+3A_vect">vect</code></td>
<td>
<p>vector of kinship categories</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if valid. Error otherwise.
</p>

<hr>
<h2 id='check_valid_lifestage'>Check valid lifestage</h2><span id='topic+check_valid_lifestage'></span>

<h3>Description</h3>

<p>Checks if vector of lifestages contains all valid entries
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_valid_lifestage(vect)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_valid_lifestage_+3A_vect">vect</code></td>
<td>
<p>vector of lifestages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if valid. Error otherwise
</p>

<hr>
<h2 id='csv_to_kinpair'>Reads <code>.csv</code> and converts to <code>KinPairData</code> object</h2><span id='topic+csv_to_kinpair'></span>

<h3>Description</h3>

<p>This function is part of suite of functions handling file import/export for kinship dispersal objects.
</p>
<p><code>.csv</code> &amp; <code>.tsv</code> reading functions at minimum require the .delim file to contain a column titled 'distance' containing distances
between kin pairs. It can optionally contain a column of kinship values 'kinship' as well as a column of lifestage values 'lifestage'.
If the file contains more than one value in the kinship or lifestage columns (e.g. bot 'FS' and 'HS') - the corresponding function
parameter must be set to pick a corresponding subset of dispersed pairs. where parameters are set in the absence of file columns,
these values are assigned to the returned <code>KinPairData</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>csv_to_kinpair(file, kinship = NULL, lifestage = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="csv_to_kinpair_+3A_file">file</code></td>
<td>
<p>The file path to read from</p>
</td></tr>
<tr><td><code id="csv_to_kinpair_+3A_kinship">kinship</code></td>
<td>
<p>character. kin category to assign or extract from data. one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C , H1C1 or H2C</p>
</td></tr>
<tr><td><code id="csv_to_kinpair_+3A_lifestage">lifestage</code></td>
<td>
<p>character. lifestage to assign or extract from data. one of 'unknown', 'immature' or 'ovipositional'.</p>
</td></tr>
<tr><td><code id="csv_to_kinpair_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to <code>read_csv</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns an object of class <code>KinPairData</code>
</p>


<h3>See Also</h3>

<p>Other import_functions: 
<code><a href="#topic+df_to_kinpair">df_to_kinpair</a>()</code>,
<code><a href="#topic+read_kindata">read_kindata</a>()</code>,
<code><a href="#topic+tsv_to_kinpair">tsv_to_kinpair</a>()</code>,
<code><a href="#topic+vector_to_kinpair">vector_to_kinpair</a>()</code>
</p>

<hr>
<h2 id='df_to_kinpair'>Convert dataframe or tibble to <code><a href="#topic+KinPairData">KinPairData</a></code> class</h2><span id='topic+df_to_kinpair'></span>

<h3>Description</h3>

<p>This function at minimum requires the dataframe to contain a column titled 'distance' containing distances
between kin pairs. It can optionally contain a column of kinship values 'kinship' as well as a column of lifestage values 'lifestage'.
If the file contains more than one value in the kinship or lifestage columns (e.g. bot 'FS' and 'HS') - the corresponding function
parameter must be set to pick a corresponding subset of dispersed pairs. where parameters are set in the absence of file columns,
these values are assigned to the returned <code>KinPairData</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df_to_kinpair(data, kinship = NULL, lifestage = NULL, lifecheck = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df_to_kinpair_+3A_data">data</code></td>
<td>
<p>data.frame or tibble of kin distances - can contain $distance (kin distances), $kinship (kin cats) &amp; $lifestage columns</p>
</td></tr>
<tr><td><code id="df_to_kinpair_+3A_kinship">kinship</code></td>
<td>
<p>character. kin category to assign or extract from data. one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C , H1C1 or H2C</p>
</td></tr>
<tr><td><code id="df_to_kinpair_+3A_lifestage">lifestage</code></td>
<td>
<p>character. lifestage to assign or extract from data. one of 'unknown', 'immature' or 'ovipositional'.</p>
</td></tr>
<tr><td><code id="df_to_kinpair_+3A_lifecheck">lifecheck</code></td>
<td>
<p>logical. If TRUE (default) tests if lifestage is valid, if FALSE, ignores this test. Set to FALSE when using custom lifestages.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns valid <code>KinPairData</code> object
</p>


<h3>See Also</h3>

<p>Other import_functions: 
<code><a href="#topic+csv_to_kinpair">csv_to_kinpair</a>()</code>,
<code><a href="#topic+read_kindata">read_kindata</a>()</code>,
<code><a href="#topic+tsv_to_kinpair">tsv_to_kinpair</a>()</code>,
<code><a href="#topic+vector_to_kinpair">vector_to_kinpair</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mydata &lt;- tibble::tibble(
  distance = 1:10, lifestage = "immature",
  kinship = c("FS", "FS", "FS", "FS", "FS", "FS", "HS", "HS", "HS", "HS")
)
df_to_kinpair(mydata, kinship = "FS")
</code></pre>

<hr>
<h2 id='dispersal_model'>Create Dispersal Model of an Organism</h2><span id='topic+dispersal_model'></span>

<h3>Description</h3>

<p>The function creates an object of class <code>DispersalModel</code> carrying organism-specific information about dispersal stages (with axial
sigmas), FS &amp; HS branch points, and the dispersal stage at which sampling occurs.It is used with the
<code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code> function to enable the simulation of uniquely defined breeding &amp; dispersal cycles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dispersal_model(
  ...,
  .FS = 0,
  .HS = .FS,
  .sampling_stage = 0,
  .cycle = 0,
  .breeding_stage = .HS,
  .visible_stage = .FS
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dispersal_model_+3A_...">...</code></td>
<td>
<p>name, value (numeric) pairs pairing custom lifestages with their corresponding axial dispersal values. MUST
be in chronological order across the entire breeding cycle.</p>
</td></tr>
<tr><td><code id="dispersal_model_+3A_.fs">.FS</code></td>
<td>
<p>(character) - breeding cycle stage at which first substantial FS-phased dispersal occurs. Must correspond to a
previously described cycle stage name. Typically reflects the first dispersal of female gametes from the mother at (variously)
egg-laying, birth, weaning stages (species-dependent). Use care in adapting to situations where multiple breeding and/or dispersal
routes commonly lead to the FS phase</p>
</td></tr>
<tr><td><code id="dispersal_model_+3A_.hs">.HS</code></td>
<td>
<p>(character) - breeding cycle stage at which first substantial HS-phased dispersal occurs. Must correspond to a
previously described cycle stage name. Typically reflects the movement of male gametes at e.g. the breeding stage (use care in adapting
to situations where multiple dispersal routes commonly lead to the HS phase)</p>
</td></tr>
<tr><td><code id="dispersal_model_+3A_.sampling_stage">.sampling_stage</code></td>
<td>
<p>(character) - stage in the breeding cycle at which samples are to be collected for kin identification. Must correspond
to a previously described cycle stage name. (so collection of eggs corresponds to an egg-laying stage, as juveniles to a juvenile stage, etc.)</p>
</td></tr>
<tr><td><code id="dispersal_model_+3A_.cycle">.cycle</code></td>
<td>
<p>(integer &gt;= -1 or vector of two such integers) breeding cycle numbers of dispersed kin to be modeled. Represents
the number of complete breeding cycles each simulated individual has undergone before the sampling point, where the time between
first dispersal and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). If .cycle
is specially set to '-1' this constitutes the sampling of an individual before it has differentiated (via dispersal) from the parent. Only use in spp.
where there is likely to be a reasonable equivalence between breeding stages across a lifespan. As the rest of the model is compatible
with a variety of cycle points, this parameter will often be overridden by the 'cycle' parameter in the <code>simulate_kindist_custom</code> function.</p>
</td></tr>
<tr><td><code id="dispersal_model_+3A_.breeding_stage">.breeding_stage</code></td>
<td>
<p>(character) - stage in the cycle at which breeding occurs. Must correspond to a previously described cycle
stage name. By default, equated with the .HS stage. This stage corresponds to the <strong>generation</strong> of next-generation individuals;
the .FS &amp; .HS stages correspond to their separation. Needed for situations where individuals are sampled before they separate from the parent.
Modify if the modeled .HS gamete dispersal event does not correspond to the initial breeding event.</p>
</td></tr>
<tr><td><code id="dispersal_model_+3A_.visible_stage">.visible_stage</code></td>
<td>
<p>(character) - stage in the cycle at the <strong>beginning</strong> of which individuals are visible to the study for
sampling rather than their parents (i.e. the beginning point of <code>cycle 0</code>). By default, equated with the .FS stage. This parameter
determines how many dispersal stages individuals have gone through before they are sampled - if <code>.sampling_stage</code> occurs just
<strong>after</strong> <code>.visible_stage</code>, the sampled individuals will have dispersed through only a small amount of the breeding cycle. if
<code>.sampling_stage</code> occurs just <strong>before</strong> <code>.visible_stage</code>, the sampled individuals will have dispersed throughout most
of the breeding cycle before being sampled. If <code>.cycle</code> is set to <code>-1</code>, dispersal stages between breeding &amp; visibility
can be accessed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The original simulation functions in this package (<code>simulate_kindist_simple()</code> &amp; <code>simulate_kindist_composite</code>) were
designed for an organism with a specific (&amp; relatively simple) breeding &amp; dispersal cycle. 'simple' corresponded to a single
dispersal event across a lifespan, equivalency of all dispersal phases (FS, HS, PO) and no lifetime overlaps. 'composite'
corresponded to many insect dispersal situations, where breeding &amp; oviposition are the key 'phase-defining' events (i.e.,
they lead to the initial gamete dispersal of half siblings &amp; full siblings from each other), where field sampling typically
occurs via ovitraps
</p>
<p>More general dispersal scenarios (e.g in mammals) require the ability to uniquely specify a variety of distinct breeding
ecologies &amp; sampling schemes: the <code><a href="#topic+DispersalModel">DispersalModel</a></code> class paired with the <code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code>
function achieves this by defining a breeding cycle with an arbitrary number of dispersal phases (the <code>dispersal_vector</code>
slot, accessed by the <code><a href="#topic+dispersal_vector">dispersal_vector</a></code> method).
</p>
<p>The breeding structure of a species may also impact at which stage
FS and HS phase branches occur. In <em>Ae. aegypti</em>, males mate with multiple females in a (single) breeding season, and a female
typically carried the egg of only one male. In this context the FS (full-sibling) phase would be set to correspond to the female's
oviposition dispersal, while the HS (half-sibling) phase would be set to correspond to the male's breeding dispersal (as its gametes
will then be dispersed by multiple females across their gravid &amp; ovipositional phases). However, in e.g. some species of the marsupial
<em>Antechinus</em>, the FS branch point would be more appropriately associated with juveniles at the time that they leave the mother's
pouch. The <code>.FS</code> and <code>.HS</code> parameters enable the assignment of these phase branches to any
defined life phase. Similarly, the <code>.sampling_stage</code> parameter allow the sampling point to be set to correspond to any
phase of the defined breeding cycle (this is later accessed with the <code><a href="#topic+sampling_stage">sampling_stage</a></code> method).
</p>
<p>The final parameter stored in this object is the breeding cycle number <code>.cycle</code>, accessed later by the <code><a href="#topic+breeding_cycle">breeding_cycle</a></code> method.
This parameter enables the treatment of species that undergo multiple breeding cycles in one lifetime. This is defined as a length two
vector describing the number of breeding cycles undergone by the final descendant of branch 1 and branch 2 of the dispersal pedigree before
their sampling. (where branch one is the 'senior' and branch two the 'junior' member of the pedigree) (so uncle is branch one, nephew branch
two, grandmother branch one, granddaughter branch two, etc.). For each member of the resulting kin pair, the cycle number represents the
number of complete breeding cycles each individual has undergone before the sampling point, where the time between birth and first
reproduction is coded as '0', that between first and second reproduction '1', etc. This enables an application of the simulation
functions defined here to deal with populations with some amount of overlap between generations.
</p>
<p>Note that this 'breeding cycle' approach is only applicable in situations where there is an approximate equivalence between the dispersal which
occurs in the first 'juvenile' breeding cycle and that which occurs between later breeding cycles. This parameter is implemented here, but it
will often be more productive to implement it instead as a parameter of the <code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code> function (the cycle parameter
there if set overrides whatever was defined within this object)
</p>


<h3>Value</h3>

<p>Returns an object of class <code>DispersalModel</code> containing custom lifestages and dispersal, phase &amp; sampling parameters that
can be passed to simulation functions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>antechinus_model &lt;- dispersal_model(pouch = 25, nest = 25, free_living = 250, breeding = 40,
gestation = 25, .FS = "nest", .HS = "breeding", .sampling_stage = "nest")
antechinus_model
</code></pre>

<hr>
<h2 id='dispersal_vector'>Access dispersal vector of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.</h2><span id='topic+dispersal_vector'></span><span id='topic+dispersal_vector+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>Access dispersal vector of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dispersal_vector(x)

## S4 method for signature 'DispersalModel'
dispersal_vector(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dispersal_vector_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="dispersal_vector_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric vector</code> named vector of custom lifestages &amp; associated dispersal sigmas.
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li></ul>

<hr>
<h2 id='DispersalModel-class'>DispersalModel Class</h2><span id='topic+DispersalModel-class'></span><span id='topic+DispersalModel'></span><span id='topic+show+2CDispersalModel-method'></span><span id='topic+initialize+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>The class <code>DispersalModel</code> is an S4 Class supplying organism-specific information about dispersal stages (with axial
sigmas), FS &amp; HS branch points, and the dispersal stage at which sampling occurs.It is used with the
<code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code> function to enable the simulation of uniquely defined breeding &amp; dispersal cycles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'DispersalModel'
show(object)

## S4 method for signature 'DispersalModel'
initialize(
  .Object,
  stages = NULL,
  dispersal_vector = NULL,
  fs = NULL,
  hs = NULL,
  sampling_stage = NULL,
  cycle = NULL,
  breeding_stage = NULL,
  visible_stage = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DispersalModel-class_+3A_object">object</code></td>
<td>
<p>an object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_.object">.Object</code></td>
<td>
<p>object to be constructed into DispersalModel class</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_stages">stages</code></td>
<td>
<p>character.  Ordered vector of all dispersal stages across the breeding cycle of the modeled species</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_dispersal_vector">dispersal_vector</code></td>
<td>
<p>numeric. Named vector of custom breeding cycle stages and their corresponding axial dispersal values</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_fs">fs</code></td>
<td>
<p>character.  breeding cycle stage at which first substantial FS-phased dispersal occurs</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_hs">hs</code></td>
<td>
<p>character. breeding cycle stage at which first substantial HS-phased dispersal occurs</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_sampling_stage">sampling_stage</code></td>
<td>
<p>character. stage in the breeding cycle at which samples are to be collected for kin identification.</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_cycle">cycle</code></td>
<td>
<p>non-negative integer. Breeding cycle numbers of dispersed kin to be modeled.  Represents
the number of complete breeding cycles each simulated individual has undergone before the sampling point, where the time between
birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0)</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_breeding_stage">breeding_stage</code></td>
<td>
<p>(character) - stage in the cycle at which breeding occurs. Must correspond to a previously described cycle
stage name. By default, equated with the .HS stage. This stage corresponds to the <strong>generation</strong> of next-generation individuals;
the .FS &amp; .HS stages correspond to their separation. Needed for situations where individuals are sampled before they separate from the parent.
Modify if the modeled .HS gamete dispersal event does not correspond to the initial breeding event.</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_visible_stage">visible_stage</code></td>
<td>
<p>(character) - stage in the cycle at the <strong>beginning</strong> of which individuals are visible to the study for
sampling rather than their parents (i.e. the beginning point of <code>cycle 0</code>). By default, equated with the <code>fs</code> stage. This parameter
determines how many dispersal stages individuals have gone through before they are sampled - if <code>.sampling_stage</code> occurs just
<strong>after</strong> <code>.visible_stage</code>, the sampled individuals will have dispersed through only a small amount of the breeding cycle. if
<code>.sampling_stage</code> occurs just <strong>before</strong> <code>.visible_stage</code>, the sampled individuals will have dispersed throughout most
of the breeding cycle before being sampled. If <code>.cycle</code> is set to <code>-1</code>, dispersal stages between breeding &amp; visibility
can be accessed.</p>
</td></tr>
<tr><td><code id="DispersalModel-class_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>an object of class DispersalModel</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The original simulation functions in this package (<code>simulate_kindist_simple()</code> &amp; <code>simulate_kindist_composite</code>) were
designed for an organism with a specific (&amp; relatively simple) breeding &amp; dispersal cycle. 'simple' corresponded to a single
dispersal event across a lifespan, equivalency of all dispersal phases (FS, HS, PO) and no lifetime overlaps. 'composite'
corresponded to many insect dispersal situations, where breeding &amp; oviposition are the key 'phase-defining' events (i.e.,
they lead to the initial gamete dispersal of half siblings &amp; full siblings from each other), where field sampling typically
occurs via ovitraps
</p>
<p>More general dispersal scenarios (e.g in mammals) require the ability to uniquely specify a variety of distinct breeding
ecologies &amp; sampling schemes: the <code>DispersalModel</code> class paired with the <code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code>
function achieves this by defining a breeding cycle with an arbitrary number of dispersal phases (the <code>dispersal_vector</code>
slot, accessed by the <code><a href="#topic+dispersal_vector">dispersal_vector</a></code> method).
</p>
<p>The breeding structure of a species may also impact at which stage
FS and HS phase branches occur. In <em>Ae. aegypti</em>, males mate with multiple females in a (single) breeding season, and a female
typically carried the egg of only one male. In this context the FS (full-sibling) phase would be set to correspond to the female's
oviposition dispersal, while the HS (half-sibling) phase would be set to correspond to the male's breeding dispersal (as its gametes
will then be dispersed by multiple females across their gravid &amp; ovipositional phases). However, in e.g. some species of the marsupial
<em>Antechinus</em>, the FS branch point would be more appropriately associated with juveniles at the time that they leave the mother's
pouch. The <code><a href="#topic+fs">fs</a></code> and <code><a href="#topic+hs">hs</a></code> slots &amp; accessor functions enable the assignment of these phase branches to any
defined life phase. Similarly, the <code><a href="#topic+sampling_stage">sampling_stage</a></code> slot &amp; method allow the sampling point to be set to correspond to any
phase of the defined breeding cycle.
</p>
<p>The next parameter stored in this object is the breeding cycle number <code>cycle</code>, accessed by the <code><a href="#topic+breeding_cycle">breeding_cycle</a></code> method.
This parameter enables the treatment of species that undergo multiple breeding cycles in one lifetime. This is defined as a length two
vector describing the number of breeding cycles undergone by the final descendant of branch 1 and branch 2 of the dispersal pedigree before
their sampling (or after branching in the case of PO). (where branch one is the 'senior' and branch two the 'junior' member of the pedigree) (so uncle is branch one, nephew branch
two, grandmother branch one, granddaughter branch two, etc.). For each member of the resulting kin pair, the cycle number represents the
number of complete breeding cycles each individual has undergone before the sampling point, where the time between birth and first
reproduction is coded as '0', that between first and second reproduction '1', etc. This enables an application of the simulation
functions defined here to deal with populations with some amount of overlap between generations.
</p>
<p>Note that this 'breeding cycle' approach is only applicable in situations where there is an approximate equivalence between the dispersal which
occurs in the first 'juvenile' breeding cycle and that which occurs between later breeding cycles. This parameter is implemented here, but it
will often be more productive to implement it instead as a parameter of the <code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code> function (the cycle parameter
there if set overrides whatever was defined within this object)
</p>
<p>The final parameter stored in this object is the breeding stage, <code>breeding_stage</code>. This describes the stage at which the
descendant individuals are generated (as opposed to <code>fs</code> &amp; <code>hs</code>, which describe the point at which they are dispersed
from the parent)
</p>


<h3>Value</h3>

<p>returns object of class <code>DispersalModel</code>
</p>
<p>No return value. Called for side effects
</p>
<p>returns an object of class <code>DispersalModel</code>
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show</code>: print method
</p>
</li>
<li> <p><code>initialize</code>: initialization method
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>dispersal_vector</code></dt><dd><p>numeric. Named vector of custom breeding cycle stages and their corresponding axial dispersal values</p>
</dd>
<dt><code>stages</code></dt><dd><p>character.  Ordered vector of all dispersal stages across the breeding cycle of the modeled species</p>
</dd>
<dt><code>fs</code></dt><dd><p>character.  breeding cycle stage at which first substantial FS-phased dispersal occurs</p>
</dd>
<dt><code>hs</code></dt><dd><p>character. breeding cycle stage at which first substantial HS-phased dispersal occurs</p>
</dd>
<dt><code>sampling_stage</code></dt><dd><p>character. stage in the breeding cycle at which samples are to be collected for kin identification.</p>
</dd>
<dt><code>cycle</code></dt><dd><p>non-negative integer. Breeding cycle numbers of dispersed kin to be modeled.  Represents
the number of complete breeding cycles each individual has undergone before the sampling point, where the time between
birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0)</p>
</dd>
<dt><code>breeding_stage</code></dt><dd><p>(character) - stage in the cycle at which breeding occurs. Must correspond to a previously described cycle
stage name. By default, equated with the .HS stage. This stage corresponds to the <strong>generation</strong> of next-generation individuals;
the .FS &amp; .HS stages correspond to their separation. Needed for situations where individuals are sampled before they separate from the parent.
Modify if the modeled .HS gamete dispersal event does not correspond to the initial breeding event.</p>
</dd>
<dt><code>visible_stage</code></dt><dd><p>(character) - stage in the cycle at the <strong>beginning</strong> of which individuals are visible to the study for
sampling rather than their parents (i.e. the beginning point of <code>cycle 0</code>). By default, equated with the <code>fs</code> stage. This parameter
determines how many dispersal stages individuals have gone through before they are sampled - if <code>.sampling_stage</code> occurs just
<strong>after</strong> <code>.visible_stage</code>, the sampled individuals will have dispersed through only a small amount of the breeding cycle. if
<code>.sampling_stage</code> occurs just <strong>before</strong> <code>.visible_stage</code>, the sampled individuals will have dispersed throughout most
of the breeding cycle before being sampled. If <code>.cycle</code> is set to <code>-1</code>, dispersal stages between breeding &amp; visibility
can be accessed.</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Other kdclasses: 
<code><a href="#topic+KinPairData-class">KinPairData-class</a></code>,
<code><a href="#topic+KinPairSimulation-class">KinPairSimulation-class</a></code>
</p>

<hr>
<h2 id='display_appdata'>Show printout of named items stored in appdata.</h2><span id='topic+display_appdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>This function prints a summary of all objects currently stored within the app interface environment, by name and class
</p>


<h3>Usage</h3>

<pre><code class='language-R'>display_appdata()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mount_appdata(kin_pair_data(), "my_kindata")
mount_appdata(simulate_kindist_simple(nsims = 10), "my_simdata")

display_appdata()
</code></pre>

<hr>
<h2 id='distances'>Access or assign <code>distances</code> category of <code><a href="#topic+KinPairData">KinPairData</a></code> class objects</h2><span id='topic+distances'></span><span id='topic+distances+2CKinPairData-method'></span>

<h3>Description</h3>

<p>Access or assign <code>distances</code> category of <code><a href="#topic+KinPairData">KinPairData</a></code> class objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distances(x)

## S4 method for signature 'KinPairData'
distances(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distances_+3A_x">x</code></td>
<td>
<p>object of class KinPairData</p>
</td></tr>
<tr><td><code id="distances_+3A_kinpairdata">KinPairData</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector of kin separation distances
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairData</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpdmethods: 
<code><a href="#topic+kinship">kinship</a>()</code>,
<code><a href="#topic+lifestage">lifestage</a>()</code>
</p>

<hr>
<h2 id='elongate'>Change the shape (aspect ratio) of a rectangle while preserving area</h2><span id='topic+elongate'></span>

<h3>Description</h3>

<p>This function is used to manipulate the dimensions parameter in other package functions, which control site dimentions.
These geometries can be entered innto functions in a few ways: (a) a single numeric value, which will be interpreted as the length of the side of a square;
(b) a numeric vector of length two, which will be interpreted as the length &amp; width of the sample site; (c) either of the above passed
to this function, which takes the rectangular site dimensions and alters their <code>aspect ratio</code> (ratio of
length to width) while preserving the underlying area the study site covers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>elongate(dims, aspect = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="elongate_+3A_dims">dims</code></td>
<td>
<p>Original rectangle dimensions - either single number (length of side of square) or length 2 numeric vector (lengths of sides x and y of rectangle)</p>
</td></tr>
<tr><td><code id="elongate_+3A_aspect">aspect</code></td>
<td>
<p>Aspect ratio of side lengths x &amp; y (i.e. x/y) in the new rectangle</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a numeric vector containing the side lengths c(x, y) of a transformed rectangle with preserved area
</p>


<h3>Examples</h3>

<pre><code class='language-R'>elongate(10, 100)
elongate(c(5, 125), 4)
</code></pre>

<hr>
<h2 id='filter_methods'>Access or modify the <code>filter</code> parameters of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> objects</h2><span id='topic+filter_methods'></span><span id='topic+upper'></span><span id='topic+upper+3C-'></span><span id='topic+lower'></span><span id='topic+lower+3C-'></span><span id='topic+spacing'></span><span id='topic+spacing+3C-'></span><span id='topic+samplenum'></span><span id='topic+samplenum+3C-'></span><span id='topic+sampledims'></span><span id='topic+sampledims+3C-'></span><span id='topic+upper+2CKinPairSimulation-method'></span><span id='topic+lower+2CKinPairSimulation-method'></span><span id='topic+spacing+2CKinPairSimulation-method'></span><span id='topic+samplenum+2CKinPairSimulation-method'></span><span id='topic+sampledims+2CKinPairSimulation-method'></span><span id='topic+upper+3C-+2CKinPairSimulation-method'></span><span id='topic+lower+3C-+2CKinPairSimulation-method'></span><span id='topic+spacing+3C-+2CKinPairSimulation-method'></span><span id='topic+samplenum+3C-+2CKinPairSimulation-method'></span><span id='topic+sampledims+3C-+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>These generics &amp; methods work as an interface between <code>KinPairSimulation</code> objects and the
<code><a href="#topic+sample_kindist">sample_kindist</a></code> function. They either retrieve the value of pre-existing filter steps that have
been applied to the object (e.g. <code>upper(x)</code>) or assign such a filtering parameter to the <code>KinPairSimulation</code>
object (e.g. <code>sampledims(x) &lt;- value</code>). In this case, the method passes the <code>KinPairSimulation</code> object to the
<code>sample_kindist()</code> function for subsampling or filtering, then updates the sampling parameter before returning the
modified object. Note that while the <code>sample_kindist</code> function can take <code><a href="#topic+KinPairData">KinPairData</a></code> objects, the methods
described here are only applicable to objects of class <code>KinPairSimulation</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upper(x)

upper(x) &lt;- value

lower(x)

lower(x) &lt;- value

spacing(x)

spacing(x) &lt;- value

samplenum(x)

samplenum(x) &lt;- value

sampledims(x)

sampledims(x) &lt;- value

## S4 method for signature 'KinPairSimulation'
upper(x)

## S4 method for signature 'KinPairSimulation'
lower(x)

## S4 method for signature 'KinPairSimulation'
spacing(x)

## S4 method for signature 'KinPairSimulation'
samplenum(x)

## S4 method for signature 'KinPairSimulation'
sampledims(x)

## S4 replacement method for signature 'KinPairSimulation'
upper(x) &lt;- value

## S4 replacement method for signature 'KinPairSimulation'
lower(x) &lt;- value

## S4 replacement method for signature 'KinPairSimulation'
spacing(x) &lt;- value

## S4 replacement method for signature 'KinPairSimulation'
samplenum(x) &lt;- value

## S4 replacement method for signature 'KinPairSimulation'
sampledims(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_methods_+3A_x">x</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
<tr><td><code id="filter_methods_+3A_value">value</code></td>
<td>
<p>value for parameter to be adjusted to</p>
</td></tr>
<tr><td><code id="filter_methods_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>either the accessed <code>numeric</code> filter parameter or a filtered <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>upper,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>lower,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>spacing,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>samplenum,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>sampledims,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>upper&lt;-,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>lower&lt;-,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>spacing&lt;-,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>samplenum&lt;-,KinPairSimulation-method</code>: 
</p>
</li>
<li> <p><code>sampledims&lt;-,KinPairSimulation-method</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpsmethods: 
<code><a href="#topic+access_sigmas">access_sigmas</a></code>,
<code><a href="#topic+kernelshape">kernelshape</a>()</code>,
<code><a href="#topic+kerneltype">kerneltype</a>()</code>,
<code><a href="#topic+simtype">simtype</a>()</code>
</p>

<hr>
<h2 id='filtertype'>Access filtertype of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object</h2><span id='topic+filtertype'></span><span id='topic+filtertype+3C-'></span><span id='topic+filtertype+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>Access filtertype of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filtertype(x)

filtertype(x) &lt;- value

## S4 method for signature 'KinPairSimulation'
filtertype(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filtertype_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="filtertype_+3A_value">value</code></td>
<td>
<p>new value to assign</p>
</td></tr>
<tr><td><code id="filtertype_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> filter status of simulation
</p>
<p>returns a modified object of the relevant class
</p>
<p><code>character</code> filter status of <code>KinPairSimulation</code> object
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairSimulation</code>: 
</p>
</li></ul>

<hr>
<h2 id='fs'>Access FS phase split point of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.</h2><span id='topic+fs'></span><span id='topic+fs+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>Access FS phase split point of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fs(x)

## S4 method for signature 'DispersalModel'
fs(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fs_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="fs_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> FS phase split
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li></ul>

<hr>
<h2 id='get_dispersal_model'>Access dispersal model of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object</h2><span id='topic+get_dispersal_model'></span><span id='topic+get_dispersal_model+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>Access dispersal model of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_dispersal_model(x)

## S4 method for signature 'KinPairSimulation'
get_dispersal_model(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_dispersal_model_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="get_dispersal_model_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns an object of class <code><a href="#topic+DispersalModel">DispersalModel</a></code>
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairSimulation</code>: 
</p>
</li></ul>

<hr>
<h2 id='hs'>Access HS phase split point of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.</h2><span id='topic+hs'></span><span id='topic+hs+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>Access HS phase split point of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hs(x)

## S4 method for signature 'DispersalModel'
hs(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hs_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="hs_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> HS phase split
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li></ul>

<hr>
<h2 id='is.DispersalModel'>Check if object is of class <code>DispersalModel</code></h2><span id='topic+is.DispersalModel'></span>

<h3>Description</h3>

<p>Check if object is of class <code>DispersalModel</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.DispersalModel(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.DispersalModel_+3A_x">x</code></td>
<td>
<p>object to be checked</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns TRUE if of class <code>DispersalModel</code>, FALSE if not
</p>

<hr>
<h2 id='is.KinPairData'>Check if object is of class KinPairData</h2><span id='topic+is.KinPairData'></span>

<h3>Description</h3>

<p>Check if object is of class KinPairData
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.KinPairData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.KinPairData_+3A_x">x</code></td>
<td>
<p>object to be checked</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns TRUE if of class <code>KinPairData</code>, FALSE if not.
</p>

<hr>
<h2 id='is.KinPairSimulation'>Check if object is of class KinPairSimulation</h2><span id='topic+is.KinPairSimulation'></span>

<h3>Description</h3>

<p>Check if object is of class KinPairSimulation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.KinPairSimulation(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.KinPairSimulation_+3A_x">x</code></td>
<td>
<p>object to be checked</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns TRUE if of class KinPairSimulation, FALSE if not
</p>

<hr>
<h2 id='kernelshape'>Access kernel type of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object</h2><span id='topic+kernelshape'></span><span id='topic+kernelshape+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>Access kernel type of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kernelshape(x)

## S4 method for signature 'KinPairSimulation'
kernelshape(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kernelshape_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="kernelshape_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> the shape parameter used in kernel simulation (if <code>kerneltype</code> is vgamma)
</p>
<p><code>character</code> the shape parameter used in kernel simulation (if <code>kerneltype</code> is vgamma)
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairSimulation</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpsmethods: 
<code><a href="#topic+access_sigmas">access_sigmas</a></code>,
<code><a href="#topic+filter_methods">filter_methods</a></code>,
<code><a href="#topic+kerneltype">kerneltype</a>()</code>,
<code><a href="#topic+simtype">simtype</a>()</code>
</p>

<hr>
<h2 id='kerneltype'>Access or assign kerneltype of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object</h2><span id='topic+kerneltype'></span><span id='topic+kerneltype+3C-'></span><span id='topic+kerneltype+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>Access or assign kerneltype of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kerneltype(x)

kerneltype(x) &lt;- value

## S4 method for signature 'KinPairSimulation'
kerneltype(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kerneltype_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="kerneltype_+3A_value">value</code></td>
<td>
<p>new value to assign</p>
</td></tr>
<tr><td><code id="kerneltype_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> the type of statistical kernel used to run the simulation (Gaussian, Laplace, vgamma)
</p>
<p>returns a modified object of the relevant class with altered kerneltype parameter
</p>
<p><code>character</code> the type of statistical kernel used to run the simulation (Gaussian, Laplace, vgamma)
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairSimulation</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpsmethods: 
<code><a href="#topic+access_sigmas">access_sigmas</a></code>,
<code><a href="#topic+filter_methods">filter_methods</a></code>,
<code><a href="#topic+kernelshape">kernelshape</a>()</code>,
<code><a href="#topic+simtype">simtype</a>()</code>
</p>

<hr>
<h2 id='kin_pair_data'>Make new KinPairData object</h2><span id='topic+kin_pair_data'></span>

<h3>Description</h3>

<p>Make new KinPairData object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kin_pair_data(data = NULL, kinship = NULL, lifestage = NULL, cycle = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kin_pair_data_+3A_data">data</code></td>
<td>
<p>tlb_df. Tibble of kinpair distances</p>
</td></tr>
<tr><td><code id="kin_pair_data_+3A_kinship">kinship</code></td>
<td>
<p>character. - one of PO, FS, HS, AV, HAV, GG, 1C, H1C, GAV, HGAV, 1C1, H1C1, GGG, 2C, H2C &amp; UN.</p>
</td></tr>
<tr><td><code id="kin_pair_data_+3A_lifestage">lifestage</code></td>
<td>
<p>character. - one of 'unknown', 'immature' or 'ovipositional', or alternatively a custom
stage that corresponds to a dispersal stage contained in a <code>DispersalModel</code> object.</p>
</td></tr>
<tr><td><code id="kin_pair_data_+3A_cycle">cycle</code></td>
<td>
<p>non-negative integer of length one or two (here, 1 is equivalent to c(1, 1)). Represents
the number of complete breeding cycles each individual has undergone before the sampling point, where the time between
birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). If the first individual
was sampled as a juvenile &amp; the second as an adult of equivalent stage, the vector c(0, 1) would be used.
In most situations, the default will be appropriate</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns an object of class <code>KinPairData</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kin_pair_data()
</code></pre>

<hr>
<h2 id='kin_pair_simulation'>KinPairSimulation</h2><span id='topic+kin_pair_simulation'></span>

<h3>Description</h3>

<p>KinPairSimulation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kin_pair_simulation(
  data = NULL,
  kinship = NULL,
  lifestage = NULL,
  simtype = NULL,
  kerneltype = NULL,
  posigma = NULL,
  initsigma = NULL,
  breedsigma = NULL,
  gravsigma = NULL,
  ovisigma = NULL,
  customsigma = NULL,
  simdims = NULL,
  kernelshape = NULL,
  cycle = NULL,
  call = NULL,
  filtertype = NULL,
  upper = NULL,
  lower = NULL,
  spacing = NULL,
  samplenum = NULL,
  sampledims = NULL,
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kin_pair_simulation_+3A_data">data</code></td>
<td>
<p>tbl_df. tibble  of simulation values</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_kinship">kinship</code></td>
<td>
<p>character - one of PO, FS, HS, AV, HAV, GG, 1C, H1C, GAV, HGAV, 1C1, H1C1, GGG, 2C, and H2C.</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_lifestage">lifestage</code></td>
<td>
<p>character - one of 'unknown', 'immature' or 'ovipositional'</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_simtype">simtype</code></td>
<td>
<p>character - simulation type</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_kerneltype">kerneltype</code></td>
<td>
<p>character. - 'Gaussian', 'Laplace' or 'vgamma' (variance-gamma)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_posigma">posigma</code></td>
<td>
<p>numeric - overall value of dispersal sigma (for simple kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_initsigma">initsigma</code></td>
<td>
<p>numeric.    - value of pre-breeding dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_breedsigma">breedsigma</code></td>
<td>
<p>numeric.    - value of breeding dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_gravsigma">gravsigma</code></td>
<td>
<p>numeric.    - value of post-breeding dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_ovisigma">ovisigma</code></td>
<td>
<p>numeric.    - value of oviposition dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_customsigma">customsigma</code></td>
<td>
<p>numeric. - vector of named custom dispersal sigmas (for custom kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_simdims">simdims</code></td>
<td>
<p>numeric. - dimensions of sampling area (assumes one side of square)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_kernelshape">kernelshape</code></td>
<td>
<p>numeric. - value of kernel shape of simulation (if using kernel with shape parameter e.g. vgamma)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_cycle">cycle</code></td>
<td>
<p>integer - number of breeding cycles sampled individual has survived (for custom kernel)</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_call">call</code></td>
<td>
<p>call. Call to create object</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_filtertype">filtertype</code></td>
<td>
<p>character. whether the initial sim has been further filtered</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_upper">upper</code></td>
<td>
<p>numeric.       - FILTER: upper threshold used</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_lower">lower</code></td>
<td>
<p>numeric.       - FILTER: lower threshold used</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_spacing">spacing</code></td>
<td>
<p>numeric.       - FILTER: spacing used</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_samplenum">samplenum</code></td>
<td>
<p>numeric.       - FILTER: sample number used</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_sampledims">sampledims</code></td>
<td>
<p>numeric.       - FILTER: sample dimensions used</p>
</td></tr>
<tr><td><code id="kin_pair_simulation_+3A_model">model</code></td>
<td>
<p>list - model information if custom simulation used to generate object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns an object of class <code>KinPairSimulation</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kin_pair_simulation()
</code></pre>

<hr>
<h2 id='kinpair_to_csv'>Write <code>KinPairData</code> object to <code>.csv</code> format</h2><span id='topic+kinpair_to_csv'></span>

<h3>Description</h3>

<p>This function is part of suite of functions handling file import/export for kinship dispersal objects. Writing
to .csv or .tsv formats strips most <code>KinPairData</code> &amp; <code>KinPairSimulation</code> class metadata and leaves a delimited file
containing ids, kinship category, geographical distance, &amp; x &amp; y coordinates for each simulated pair. (removes class attributes)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinpair_to_csv(x, file, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinpair_to_csv_+3A_x">x</code></td>
<td>
<p>Object of class <code>KinPairData</code> or <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="kinpair_to_csv_+3A_file">file</code></td>
<td>
<p>The file path to write to</p>
</td></tr>
<tr><td><code id="kinpair_to_csv_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>write_csv</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the initial object
</p>


<h3>See Also</h3>

<p>Other export_functions: 
<code><a href="#topic+kinpair_to_tibble">kinpair_to_tibble</a>()</code>,
<code><a href="#topic+kinpair_to_tsv">kinpair_to_tsv</a>()</code>,
<code><a href="#topic+write_kindata">write_kindata</a>()</code>
</p>

<hr>
<h2 id='kinpair_to_tibble'>Extract KinPairData class object to tibble</h2><span id='topic+kinpair_to_tibble'></span>

<h3>Description</h3>

<p>Extract <code>KinPairData</code> class object to tibble. Strips out most class metadata leaving a
dataframe of disersal simulation data with a column added covering lifestage at sampling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinpair_to_tibble(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinpair_to_tibble_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>tibble (class <code>tbl_df</code>)
</p>


<h3>See Also</h3>

<p>Other export_functions: 
<code><a href="#topic+kinpair_to_csv">kinpair_to_csv</a>()</code>,
<code><a href="#topic+kinpair_to_tsv">kinpair_to_tsv</a>()</code>,
<code><a href="#topic+write_kindata">write_kindata</a>()</code>
</p>

<hr>
<h2 id='kinpair_to_tsv'>Write <code>KinPairData</code> object to <code>.tsv</code> format</h2><span id='topic+kinpair_to_tsv'></span>

<h3>Description</h3>

<p>This function is part of suite of functions handling file import/export for kinship dispersal objects. Writing
to .csv or .tsv formats strips most <code>KinPairData</code> &amp; <code>KinPairSimulation</code> class metadata and leaves a delimited file
containing ids, kinship category, geographical distance, &amp; x &amp; y coordinates for each simulated pair. (removes class attributes)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinpair_to_tsv(x, file, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinpair_to_tsv_+3A_x">x</code></td>
<td>
<p>Object of class <code>KinPairData</code> or <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="kinpair_to_tsv_+3A_file">file</code></td>
<td>
<p>The file path to write to</p>
</td></tr>
<tr><td><code id="kinpair_to_tsv_+3A_...">...</code></td>
<td>
<p>Additional arguments to pass to <code>write_tsv</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the initial object
</p>


<h3>See Also</h3>

<p>Other export_functions: 
<code><a href="#topic+kinpair_to_csv">kinpair_to_csv</a>()</code>,
<code><a href="#topic+kinpair_to_tibble">kinpair_to_tibble</a>()</code>,
<code><a href="#topic+write_kindata">write_kindata</a>()</code>
</p>

<hr>
<h2 id='KinPairData-class'>Formal class KinPairData</h2><span id='topic+KinPairData-class'></span><span id='topic+KinPairData'></span><span id='topic+show+2CKinPairData-method'></span><span id='topic+initialize+2CKinPairData-method'></span>

<h3>Description</h3>

<p>The class <code>KinPairData</code> is a formal (S4) class for storing kinship and lifespan dispersal
information concerning kin pairs. It is the base class on which the <code>KinPairSimulation</code> class is built.
The <code>KinPairData</code> class is used to store information about the spatial distribution of kin dyads for use
in calculating axial sigmas of intergenerational dispersal as initially implemented in Jasper et al. 2019
(doi: <a href="https://doi.org/10.1111/1755-0998.13043">10.1111/1755-0998.13043</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'KinPairData'
show(object)

## S4 method for signature 'KinPairData'
initialize(
  .Object,
  data = NULL,
  kinship = NULL,
  lifestage = NULL,
  cycle = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KinPairData-class_+3A_object">object</code></td>
<td>
<p>an object of class KinpairData</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_.object">.Object</code></td>
<td>
<p>the KinPairData object to be constructed</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_data">data</code></td>
<td>
<p>data about kinship to be used to construct object (tibble, data.frame, or numeric vector of distances)</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_kinship">kinship</code></td>
<td>
<p>character. Kinship category value for object. - one of PO, FS, HS, AV, HAV, GG, 1C, H1C, GAV, HGAV, 1C1, H1C1, GGG, 2C, and H2C.</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_lifestage">lifestage</code></td>
<td>
<p>character. Lifestage value for object. - one of 'immature', 'ovipositional' or 'unknown'</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_cycle">cycle</code></td>
<td>
<p>non-negative integer or vector of two such integers - Represents
the number of complete breeding cycles each simulated individual has undergone before the sampling point, where the time between
birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). If the first individual
was sampled as a juvenile &amp; the second as an adult of equivalent stage, the vector c(0, 1) would be used. In most situations, defualt will be appropriate</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_...">...</code></td>
<td>
<p>additional argument to pass to downstream functions in future</p>
</td></tr>
<tr><td><code id="KinPairData-class_+3A_kinpairdata">KinPairData</code></td>
<td>
<p>object of class KinPairData</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This class is essentially wrapped around the <code>tbl_df</code> class but with (a) expectations around certain columns
that must be present (<code>id1, id2, kinship</code>, &amp; <code>distance</code> - three 'character' &amp; one 'numeric' column), as
well as (b) additional attributes (<code>kinship</code>, <code>lifestage</code>, &amp; <code>cycle</code>) characterizing the close-kin
dyads being stored.These attributes, as well as the embedded vector of distances, can be accessed with the methods
<code><a href="#topic+kinship">kinship</a></code>, <code><a href="#topic+lifestage">lifestage</a></code>, <code><a href="#topic+breeding_cycle">breeding_cycle</a></code> and <code><a href="#topic+distances">distances</a></code>.
</p>
<p>Objects from this class are returned from the <code><a href="#topic+df_to_kinpair">df_to_kinpair</a></code> and <code><a href="#topic+csv_to_kinpair">csv_to_kinpair</a></code> functions
(&amp; related), and are directly constructed with the namesake <code>KinPairData()</code> function.
They can be passed to the <code><a href="#topic+sample_kindist">sample_kindist</a></code> function for filtering and subsampling, and to
axial functions (including <code><a href="#topic+axials_standard">axials_standard</a></code> and <code><a href="#topic+axpermute_standard">axpermute_standard</a></code>) for estimation of
axial dispersal.
</p>


<h3>Value</h3>

<p>returns object of class <code>KinPairData</code>
</p>
<p>No return value, called for side effects
</p>
<p>Returns an object of class <code>KinPairData</code>
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show</code>: standard print method
</p>
</li>
<li> <p><code>initialize</code>: initialize method
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>kinship</code></dt><dd><p>character - one of PO, FS, HS, AV, HAV, GG, 1C, H1C, GAV, HGAV, 1C1, H1C1, GGG, 2C, and H2C.</p>
</dd>
<dt><code>lifestage</code></dt><dd><p>character - lifestage at sampling - either 'immature', 'ovipositional' or a stage
corresponding to a <code>DispersalModel</code> custom stage</p>
</dd>
<dt><code>cycle</code></dt><dd><p>non-negative integer or vector of two such integers - Represents
the number of complete breeding cycles each  individual has undergone before the sampling point, where the time between
birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0). If the first individual
was sampled as a juvenile &amp; the second as an adult of equivalent stage, the vector c(0, 1) would be used. In most situations, the default will be appropriate</p>
</dd>
<dt><code>tab</code></dt><dd><p>tbl_df. - tibble of dispersal values</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Other kdclasses: 
<code><a href="#topic+DispersalModel-class">DispersalModel-class</a></code>,
<code><a href="#topic+KinPairSimulation-class">KinPairSimulation-class</a></code>
</p>

<hr>
<h2 id='KinPairSimulation_composite'>Constructor for KinPairSimulation Class (composite)</h2><span id='topic+KinPairSimulation_composite'></span>

<h3>Description</h3>

<p>Constructor for KinPairSimulation Class (composite)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KinPairSimulation_composite(
  data = NULL,
  kinship = NULL,
  kerneltype = NULL,
  initsigma = NULL,
  breedsigma = NULL,
  gravsigma = NULL,
  ovisigma = NULL,
  simdims = NULL,
  lifestage = NULL,
  kernelshape = NULL,
  call = NULL,
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KinPairSimulation_composite_+3A_data">data</code></td>
<td>
<p>tibble of pairwise kin classes &amp; distances. Ideally contains fields id1 &amp; id2 (chr) an distance (dbl) optionally includes coords (x1, y1, x2, y2), lifestage (ls1 &amp; ls2), kinship (chr) and sims (dbl)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_kinship">kinship</code></td>
<td>
<p>character. Code for kinship category of simulation. one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C or H2C</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_kerneltype">kerneltype</code></td>
<td>
<p>character. Statistical model for simulated dispersal kernel. Currently either &quot;Gaussian&quot;, &quot;Laplace&quot; or &quot;vgamma&quot; (variance-gamma).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_initsigma">initsigma</code></td>
<td>
<p>numeric. Axial sigma of prebreeding ('juvenile') dispersal kernel (axial standard deviation).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_breedsigma">breedsigma</code></td>
<td>
<p>numeric. Axial sigma of breeding dispersal kernel (axial standard deviation).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_gravsigma">gravsigma</code></td>
<td>
<p>numeric. Axial sigma of post-breeding ('gravid') dispersal kernel (axial standard deviation).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_ovisigma">ovisigma</code></td>
<td>
<p>numeric. Axial sigma of oviposition dispersal kernel (axial standard deviation).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_simdims">simdims</code></td>
<td>
<p>numeric. Length of side of simulated area square.</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_lifestage">lifestage</code></td>
<td>
<p>character. Simulated lifestage of sampling. Either &quot;immature&quot; (sampled at hatching) or &quot;ovipositional&quot;
(sampled as an adult during oviposition - essentially one lifespan later than 'immature')</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_kernelshape">kernelshape</code></td>
<td>
<p>numeric. Value of shape parameter for simulated kernel if kernel requires one (e.g. vgamma kernel).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_call">call</code></td>
<td>
<p>call object. Use to pass the system call that led to the generation of this class. (via sys.call)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_composite_+3A_model">model</code></td>
<td>
<p>DispersalModel - model information passed from simulation function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>KinPairSimulation</code> Class object with simtype set to 'composite' and relevant fields included.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kindata &lt;- tibble::tibble(
  id1 = c("a", "b", "c"), id2 = c("x", "y", "z"),
  distance = c(50, 45, 65), kinship = c("1C", "1C", "1C")
)
KinPairSimulation_composite(kindata,
  kinship = "1C", kerneltype = "Gaussian",
  initsigma = 15, breedsigma = 25, gravsigma = 20, ovisigma = 10, lifestage = "immature"
)
</code></pre>

<hr>
<h2 id='KinPairSimulation_custom'>Constructor for KinPairSimulation Class (custom)</h2><span id='topic+KinPairSimulation_custom'></span>

<h3>Description</h3>

<p>Constructor for KinPairSimulation Class (custom)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KinPairSimulation_custom(
  data = NULL,
  kinship = NULL,
  kerneltype = NULL,
  customsigma = NULL,
  simdims = NULL,
  lifestage = NULL,
  kernelshape = NULL,
  cycle = NULL,
  call = NULL,
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KinPairSimulation_custom_+3A_data">data</code></td>
<td>
<p>tibble of pairwise kin classes &amp; distances. Ideally contains fields id1 &amp; id2 (chr) an distance (dbl) optionally includes coords (x1, y1, x2, y2), lifestage (ls1 &amp; ls2), kinship (chr) and sims (dbl)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_kinship">kinship</code></td>
<td>
<p>character. Code for kinship category of simulation. one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C or H2C</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_kerneltype">kerneltype</code></td>
<td>
<p>character. Statistical model for simulated dispersal kernel. Currently either &quot;Gaussian&quot;, &quot;Laplace&quot; or &quot;vgamma&quot; (variance-gamma).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_customsigma">customsigma</code></td>
<td>
<p>numeric. Named vector of custom breeding cycle stages and their corresponding axial dispersal values</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_simdims">simdims</code></td>
<td>
<p>numeric. Length of side of simulated area square.</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_lifestage">lifestage</code></td>
<td>
<p>character. Simulated lifestage of sampling. Here, must correspond to a custom lifestage derived from 'customsigma'</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_kernelshape">kernelshape</code></td>
<td>
<p>numeric. Value of shape parameter for simulated kernel if kernel requires one (e.g. vgamma kernel).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_cycle">cycle</code></td>
<td>
<p>non-negative integer. Breeding cycle numbers of dispersed kin to be modeled.  Represents
the number of complete breeding cycles each simulated individual has undergone before the sampling point, where the time between
birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_call">call</code></td>
<td>
<p>call object. Use to pass the system call that led to the generation of this class. (via sys.call)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_custom_+3A_model">model</code></td>
<td>
<p>DispersalModel - model information passed from simulation function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>KinPairSimulation</code> Class object with simtype set to 'custom' and relevant fields included.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kindata &lt;- tibble::tibble(
  id1 = c("a", "b", "c"), id2 = c("x", "y", "z"),
  distance = c(50, 45, 65), kinship = c("1C", "1C", "1C")
)
KinPairSimulation_custom(kindata,
  kinship = "1C", kerneltype = "Gaussian",
  customsigma = c(initsigma = 15, breedsigma = 25, gravsigma = 20, ovisigma = 10),
  lifestage = "ovisigma", cycle = 0
)
</code></pre>

<hr>
<h2 id='KinPairSimulation_simple'>Constructor for KinPairSimulation Class (simple)</h2><span id='topic+KinPairSimulation_simple'></span>

<h3>Description</h3>

<p>Constructor for KinPairSimulation Class (simple)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KinPairSimulation_simple(
  data = NULL,
  kinship = NULL,
  kerneltype = NULL,
  posigma = NULL,
  simdims = NULL,
  lifestage = NULL,
  kernelshape = NULL,
  call = NULL,
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KinPairSimulation_simple_+3A_data">data</code></td>
<td>
<p>tibble of pairwise kin classes &amp; distances. Ideally contains fields id1 &amp; id2 (chr) an distance (dbl) optionally includes coords (x1, y1, x2, y2), lifestage (ls1 &amp; ls2), kinship (chr) and sims (dbl)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_kinship">kinship</code></td>
<td>
<p>character. Code for kinship category of simulation. one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C or H2C</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_kerneltype">kerneltype</code></td>
<td>
<p>character. Statistical model for simulated dispersal kernel. Currently either &quot;Gaussian&quot;, &quot;Laplace&quot; or &quot;vgamma&quot; (variance-gamma).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_posigma">posigma</code></td>
<td>
<p>numeric. Axial sigma of dispersal kernel (axial standard deviation).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_simdims">simdims</code></td>
<td>
<p>numeric. Length of side of simulated area square.</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_lifestage">lifestage</code></td>
<td>
<p>character. Simulated lifestage of sampling. Either &quot;immature&quot; (sampled at hatching) or &quot;ovipositional&quot;
(sampled as an adult during oviposition - essentially one lifespan later than 'immature')</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_kernelshape">kernelshape</code></td>
<td>
<p>numeric. Value of shape parameter for simulated kernel if kernel requires one (e.g. vgamma kernel).</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_call">call</code></td>
<td>
<p>call object. Use to pass the system call that led to the generation of this class. (via sys.call)</p>
</td></tr>
<tr><td><code id="KinPairSimulation_simple_+3A_model">model</code></td>
<td>
<p>DispersalModel - model information passed from simulation function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <code>KinPairSimulation</code> Class object with simtype set to 'simple' and relevant fields included.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>kindata &lt;- tibble::tibble(
  id1 = c("a", "b", "c"), id2 = c("x", "y", "z"),
  distance = c(50, 45, 65), kinship = c("1C", "1C", "1C")
)
KinPairSimulation_simple(kindata,
  kinship = "1C", kerneltype = "Gaussian",
  posigma = 38, lifestage = "immature"
)
</code></pre>

<hr>
<h2 id='KinPairSimulation-class'>KinPairSimulation Class</h2><span id='topic+KinPairSimulation-class'></span><span id='topic+KinPairSimulation'></span><span id='topic+show+2CKinPairSimulation-method'></span><span id='topic+initialize+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>The class <code>KinPairSimulation</code> is a formal (S4) class for storing kinship and dispersal
distribution information derived from simulations in the <code>kindisperse</code> package.
It is derived from the <code><a href="#topic+KinPairData">KinPairData</a></code> class.
The <code>KinPairSimulation</code> class is used to store information about the spatial distribution of kin dyads for use
in calculating axial sigmas of intergenerational dispersal as initially implemented in Jasper et al. 2019
(doi: <a href="https://doi.org/10.1111/1755-0998.13043">10.1111/1755-0998.13043</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'KinPairSimulation'
show(object)

## S4 method for signature 'KinPairSimulation'
initialize(
  .Object,
  data = NULL,
  kinship = NULL,
  lifestage = NULL,
  simtype = NULL,
  kerneltype = NULL,
  kernelshape = NULL,
  posigma = NULL,
  initsigma = NULL,
  breedsigma = NULL,
  gravsigma = NULL,
  ovisigma = NULL,
  customsigma = NULL,
  cycle = NULL,
  simdims = NULL,
  call = NULL,
  filtertype = NULL,
  upper = NULL,
  lower = NULL,
  spacing = NULL,
  samplenum = NULL,
  sampledims = NULL,
  model = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KinPairSimulation-class_+3A_object">object</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_.object">.Object</code></td>
<td>
<p>object to be constructed into KinPairSimulation class</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_data">data</code></td>
<td>
<p>tbl_df. tibble  of simulation values</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_kinship">kinship</code></td>
<td>
<p>character - one of PO, FS, HS, AV, HAV, GG, 1C, H1C, GAV, HGAV, 1C1, H1C1, GGG, 2C, and H2C.</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_lifestage">lifestage</code></td>
<td>
<p>character - one of 'unknown', 'immature' or 'ovipositional'</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_simtype">simtype</code></td>
<td>
<p>character - simulation type</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_kerneltype">kerneltype</code></td>
<td>
<p>character. - 'Gaussian', 'Laplace' or 'vgamma' (variance-gamma)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_kernelshape">kernelshape</code></td>
<td>
<p>numeric. - value of kernel shape of simulation (if using kernel with shape parameter e.g. vgamma)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_posigma">posigma</code></td>
<td>
<p>numeric - overall value of dispersal sigma (for simple kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_initsigma">initsigma</code></td>
<td>
<p>numeric.    - value of pre-breeding dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_breedsigma">breedsigma</code></td>
<td>
<p>numeric.    - value of breeding dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_gravsigma">gravsigma</code></td>
<td>
<p>numeric.    - value of post-breeding dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_ovisigma">ovisigma</code></td>
<td>
<p>numeric.    - value of oviposition dispersal sigma (for composite kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_customsigma">customsigma</code></td>
<td>
<p>numeric.  - vector of named custom dispersal sigmas (for custom kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_cycle">cycle</code></td>
<td>
<p>integer - number of breeding cycles sampled individual has survived (for custom kernel)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_simdims">simdims</code></td>
<td>
<p>numeric. - dimensions of sampling area (assumes one side of square)</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_call">call</code></td>
<td>
<p>call. Call to create object</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_filtertype">filtertype</code></td>
<td>
<p>character. whether the initial sim has been further filtered</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_upper">upper</code></td>
<td>
<p>numeric.       - FILTER: upper threshold used</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_lower">lower</code></td>
<td>
<p>numeric.       - FILTER: lower threshold used</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_spacing">spacing</code></td>
<td>
<p>numeric.       - FILTER: spacing used</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_samplenum">samplenum</code></td>
<td>
<p>numeric.       - FILTER: sample number used</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_sampledims">sampledims</code></td>
<td>
<p>numeric.       - FILTER: sample dimensions used</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_model">model</code></td>
<td>
<p>list - model information if custom simulation used to generate object</p>
</td></tr>
<tr><td><code id="KinPairSimulation-class_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>an object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This class is essentially wrapped around the <code>tbl_df</code> class but with (a) expectations around certain columns
that must be present (<code>id1, id2, kinship</code>, &amp; <code>distance</code> - three 'character' &amp; one 'numeric' column), as
well as (b) additional attributes (<code>kinship</code>, <code>lifestage</code>, &amp; <code>cycle</code>) characterizing the close-kin
dyads being stored.These attributes, as well as the embedded vector of distances, can be accessed with the methods
<code><a href="#topic+kinship">kinship</a></code>, <code><a href="#topic+lifestage">lifestage</a></code>, <code><a href="#topic+breeding_cycle">breeding_cycle</a></code> and <code><a href="#topic+distances">distances</a></code>.
In addition to the above attributes (derived from the <code>KinPairData</code> class), this class contains attributes
capturing the simulation type &amp; parameters used to generate the final distribution of kin dyads.
</p>
<p>Objects from this class are returned from the <code><a href="#topic+simulate_kindist_composite">simulate_kindist_composite</a></code>,
<code><a href="#topic+simulate_kindist_simple">simulate_kindist_simple</a></code> and <code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code> functions
(&amp; related), and are directly constructed with the namesake <code>KinPairSimulation()</code> function.
They can be passed to the <code><a href="#topic+sample_kindist">sample_kindist</a></code> function for filtering and subsampling, and to
axial functions (including <code><a href="#topic+axials_standard">axials_standard</a></code> and <code><a href="#topic+axpermute_standard">axpermute_standard</a></code>) for estimation of
axial dispersal.
</p>


<h3>Value</h3>

<p>returns object of class <code>KinPairSimulation</code>
</p>
<p>No return value, called for side effects
</p>
<p>Returns an object of class <code>KinPairSimulation</code>
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show</code>: print method
</p>
</li>
<li> <p><code>initialize</code>: initialisation method
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>kinship</code></dt><dd><p>character - one of PO, FS, HS, AV, HAV, GG, 1C, H1C, GAV, HGAV, 1C1, H1C1, GGG, 2C, and H2C.</p>
</dd>
<dt><code>simtype</code></dt><dd><p>character. - one of 'simple', 'composite' or 'custom'</p>
</dd>
<dt><code>kerneltype</code></dt><dd><p>character. - 'Gaussian', 'Laplace' or 'vgamma' (variance-gamma)</p>
</dd>
<dt><code>posigma</code></dt><dd><p>numeric.       - overall value of dispersal sigma (for simple kernel)</p>
</dd>
<dt><code>initsigma</code></dt><dd><p>numeric.    - value of pre-breeding dispersal sigma (for composite kernel)</p>
</dd>
<dt><code>breedsigma</code></dt><dd><p>numeric.  - value of breeding dispersal sigma (for composite kernel)</p>
</dd>
<dt><code>gravsigma</code></dt><dd><p>numeric.   - value of post-breeding dispersal sigma (for composite kernel)</p>
</dd>
<dt><code>ovisigma</code></dt><dd><p>numeric.    - value of oviposition dispersal sigma (for composite kernel)</p>
</dd>
<dt><code>customsigma</code></dt><dd><p>numeric  - vector of named custom dispersal sigmas (for custom kernel)</p>
</dd>
<dt><code>simdims</code></dt><dd><p>numeric.        - dimensions of sampling area (assumes 1 side of square)</p>
</dd>
<dt><code>lifestage</code></dt><dd><p>character. - lifestage at sampling - either 'immature' or 'ovipositional'</p>
</dd>
<dt><code>cycle</code></dt><dd><p>integer - number of breeding cycles sampled individuals have survived (for custom kernel)</p>
</dd>
<dt><code>kernelshape</code></dt><dd><p>numeric.   - shape parameter if vgamma kerneltype</p>
</dd>
<dt><code>call</code></dt><dd><p>call.           - call to create initial simulation</p>
</dd>
<dt><code>tab</code></dt><dd><p>tbl_df.          - tibble of simulation values</p>
</dd>
<dt><code>filtertype</code></dt><dd><p>character. - whether the initial sim has been further filtered</p>
</dd>
<dt><code>upper</code></dt><dd><p>numeric.       - FILTER: upper threshold used</p>
</dd>
<dt><code>lower</code></dt><dd><p>numeric.       - FILTER: lower threshold used</p>
</dd>
<dt><code>spacing</code></dt><dd><p>numeric.     - FILTER: spacing used</p>
</dd>
<dt><code>samplenum</code></dt><dd><p>numeric.   - FILTER: sample number used</p>
</dd>
<dt><code>sampledims</code></dt><dd><p>numeric.  - FILTER: dimensions used</p>
</dd>
<dt><code>model</code></dt><dd><p><code>DispersalModel</code> - model of dispersal used to create object (with custom type)</p>
</dd>
</dl>


<h3>See Also</h3>

<p>Other kdclasses: 
<code><a href="#topic+DispersalModel-class">DispersalModel-class</a></code>,
<code><a href="#topic+KinPairData-class">KinPairData-class</a></code>
</p>

<hr>
<h2 id='kinship'>Access or assign <code>kinship</code> category of <code><a href="#topic+KinPairData">KinPairData</a></code> class objects</h2><span id='topic+kinship'></span><span id='topic+kinship+3C-'></span><span id='topic+kinship+2CKinPairData-method'></span><span id='topic+kinship+3C-+2CKinPairData-method'></span>

<h3>Description</h3>

<p>Access or assign <code>kinship</code> category of <code><a href="#topic+KinPairData">KinPairData</a></code> class objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kinship(x)

kinship(x) &lt;- value

## S4 method for signature 'KinPairData'
kinship(x)

## S4 replacement method for signature 'KinPairData'
kinship(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kinship_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
<tr><td><code id="kinship_+3A_value">value</code></td>
<td>
<p>value to assign to slot</p>
</td></tr>
<tr><td><code id="kinship_+3A_kinpairdata">KinPairData</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns <code>character</code> kinship category of object or <code>KinPairData</code> object with modified kinship category
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairData</code>: 
</p>
</li>
<li> <p><code>KinPairData</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpdmethods: 
<code><a href="#topic+distances">distances</a>()</code>,
<code><a href="#topic+lifestage">lifestage</a>()</code>
</p>

<hr>
<h2 id='lifestage'>Access or assign <code>lifestage</code> category of <code><a href="#topic+KinPairData">KinPairData</a></code> class objects</h2><span id='topic+lifestage'></span><span id='topic+lifestage+3C-'></span><span id='topic+lifestage+2CKinPairData-method'></span><span id='topic+lifestage+3C-+2CKinPairData-method'></span>

<h3>Description</h3>

<p>Access or assign <code>lifestage</code> category of <code><a href="#topic+KinPairData">KinPairData</a></code> class objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lifestage(x)

lifestage(x) &lt;- value

## S4 method for signature 'KinPairData'
lifestage(x)

## S4 replacement method for signature 'KinPairData'
lifestage(x) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lifestage_+3A_x">x</code></td>
<td>
<p>object with relevant method</p>
</td></tr>
<tr><td><code id="lifestage_+3A_value">value</code></td>
<td>
<p>new value to assign</p>
</td></tr>
<tr><td><code id="lifestage_+3A_kinpairdata">KinPairData</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns <code>character</code> lifestage of object or <code>KinPairData</code> object with modified lifestage
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairData</code>: 
</p>
</li>
<li> <p><code>KinPairData</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpdmethods: 
<code><a href="#topic+distances">distances</a>()</code>,
<code><a href="#topic+kinship">kinship</a>()</code>
</p>

<hr>
<h2 id='mentari'>Position &amp; kinship information of <em>Aedes aegypti</em> from Mentari Court, Malaysia</h2><span id='topic+mentari'></span>

<h3>Description</h3>

<p>A data file containing the positions &amp; kinship values of 98 <em>Ae. aegypti</em> larval
kin pairs collected between September 19 &amp; October 10, 2017 in Mentari Court
(Petaling Jaya), Malaysia.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mentari
</code></pre>


<h3>Format</h3>

<p>A data frame with 98 rows and 10 variables
</p>

<dl>
<dt>id1</dt><dd><p>id of first individual of kinpair</p>
</dd>
<dt>id2</dt><dd><p>id of second individual of kinpair</p>
</dd>
<dt>kinship</dt><dd><p>kinship category of the pairing</p>
</dd>
<dt>distance</dt><dd><p>geographical distance between kinpair</p>
</dd>
<dt>x1</dt><dd><p>relative x coordinate of first individual in metres</p>
</dd>
<dt>y1</dt><dd><p>relative y coordinate of first individual in metres</p>
</dd>
<dt>x2</dt><dd><p>relative x coordinate of second individual in metres</p>
</dd>
<dt>y2</dt><dd><p>relative y coordinate of second individual in metres</p>
</dd>
<dt>lifestage</dt><dd><p>lifestage at time of sampling of kinpair</p>
</dd>
<dt>k_loiselle</dt><dd><p>calculated Loiselle's <em>k</em> value for kinpair</p>
</dd>
</dl>



<h3>Details</h3>

<p>162 individuals were sourced as larvae from ovitraps placed in eight apartment
buildings (in floors three or four for each), collected over three weeks. Entire
larval bodies were extracted and sequenced using the double-digest restriction-site-
associated DNA sequencing protocol for <em>Ae. aegypti</em>
(doi: <a href="https://doi.org/10.1186/1471-2164-15-275">10.1186/1471-2164-15-275</a>. After sequencing &amp; genotyping,
Loiselle's <em>k</em> was used as an initial estimate of genetic kinship. The program
ML-Relate (doi: <a href="https://doi.org/10.1111/j.1471-8286.2006.01256.x">10.1111/j.1471-8286.2006.01256.x</a>) was then used to
estimate the pedigree kinships for the FS and HS categories. Following simulation
work described in doi: <a href="https://doi.org/10.1111/1755-0998.13043">10.1111/1755-0998.13043</a> the
1C category was assigned to all remaining unassigned individuals with a Loiselle's
<em>k</em> of less than 0.06.
</p>


<h3>Value</h3>

<p>returns an object of class <code>tbl_df</code>
</p>


<h3>Source</h3>

<p>doi: <a href="https://doi.org/10.1111/1755-0998.13043">10.1111/1755-0998.13043</a>
</p>

<hr>
<h2 id='mount_appdata'>Mount <code>KinPairData</code> Objects for use in kindisperse app</h2><span id='topic+mount_appdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>This function takes an object of class <code>KinPairData</code> or <code>KinPairSimulation</code>, assigns it an identifying name,
and adds it to the app interface environment, making it accessible within the app. Once added, this object will be
accessible under its name from the <code>Load</code> menue of the app. (The app interface uses the same function
internally, enabling objects to be passed to the interface from the app also).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mount_appdata(x, nm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mount_appdata_+3A_x">x</code></td>
<td>
<p>An object of class <code>KinPairData</code> or KinPairSimulation</p>
</td></tr>
<tr><td><code id="mount_appdata_+3A_nm">nm</code></td>
<td>
<p>character. A name to store the object as</p>
</td></tr>
</table>


<h3>Value</h3>

<p>invisibly returns x.
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mount_appdata(kin_pair_data(), "mydata")
</code></pre>

<hr>
<h2 id='read_kindata'>Reads <code>.kindata</code> filetype back to <code>KinPairData</code> or <code>KinPairSimulation</code> object.</h2><span id='topic+read_kindata'></span>

<h3>Description</h3>

<p>This function is part of suite of functions handling file import/export for kinship dispersal objects.
</p>
<p>The custom <code>.kindata</code> format enables complete preservation of <code>KinPairData</code> &amp; <code>KinPairSimulation</code> formats
without any loss of class attributes or metadata - ideal for saving and retrieving simulation data that is intended for further in-package
processing with kindisperse. This function loads a previously stored object into its original class format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_kindata(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_kindata_+3A_file">file</code></td>
<td>
<p>Character giving path reference to file with extension <code>.kinpair</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns either <code>KinPairData</code> or <code>KinPairSimulation</code> object.
</p>


<h3>See Also</h3>

<p>Other import_functions: 
<code><a href="#topic+csv_to_kinpair">csv_to_kinpair</a>()</code>,
<code><a href="#topic+df_to_kinpair">df_to_kinpair</a>()</code>,
<code><a href="#topic+tsv_to_kinpair">tsv_to_kinpair</a>()</code>,
<code><a href="#topic+vector_to_kinpair">vector_to_kinpair</a>()</code>
</p>

<hr>
<h2 id='rebase_dims'>Change the dimensions of a KinPairSimulation Object and shift kinpairs so at least one individual is within the area</h2><span id='topic+rebase_dims'></span>

<h3>Description</h3>

<p>Change the dimensions of a KinPairSimulation Object and shift kinpairs so at least one individual is within the area
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rebase_dims(kindist, dims)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rebase_dims_+3A_kindist">kindist</code></td>
<td>
<p><code>KinPairSimulation</code> - <code>KinPairSimulation</code> Class Object</p>
</td></tr>
<tr><td><code id="rebase_dims_+3A_dims">dims</code></td>
<td>
<p>New site dimensions - either single number (length of side of square) or length 2 vector (lengths of sides x and y of rectangle)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a rebased object of class <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> with adjusted simulation dimensions
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simobject &lt;- simulate_kindist_simple()

rebase_dims(simobject, c(1, 100))
rebase_dims(simobject, 15)
</code></pre>

<hr>
<h2 id='reset_appdata'>Reset kindisperse appdata</h2><span id='topic+reset_appdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>When called, this function clears all attached objects from the app interface environment, keeping it from becoming
over-clutttered &amp; taking up space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reset_appdata()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>reset_appdata()
</code></pre>

<hr>
<h2 id='reset_tempdata'>Reset app tempdata (internal mem)</h2><span id='topic+reset_tempdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>This function resets the internal <code>tempdata</code> environment used by the kindisperse app, keeping it from becoming
over-cluttered &amp; freeing up space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reset_tempdata()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>reset_tempdata()
</code></pre>

<hr>
<h2 id='retrieve_appdata'>Retrieve <code>KinPairData</code> object from appdata (single)</h2><span id='topic+retrieve_appdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>This function accesses the app interface environment and retrieves an object (typically of class <code>KinPairData</code> or
<code>KinPairSimulation</code>) with the name <code>nm</code>, making it accessible from within our outside the app. This can be
used to load simulation objects that were saved from the interface while using the app into the regular R environment
(after closing the app). (The app uses this function internally to load objects from the interface into its own internal
environment for display &amp; processing.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieve_appdata(nm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="retrieve_appdata_+3A_nm">nm</code></td>
<td>
<p>character. Name of item as stored in appdata</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns KinPairData object accessible by name nm
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mount_appdata(kin_pair_data(), "mydata")

retrieve_appdata("mydata")
</code></pre>

<hr>
<h2 id='retrieve_tempdata'>Retrieve all tempdata (internal mem) from app (as list)</h2><span id='topic+retrieve_tempdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>This function accesses the app internal environment and retrieves a named list of all objects (typically of classes
<code>KinPairData</code> or <code>KinPairSimulation</code> contained within it, making them accessible outside of the app). This
is used to quickly retrieve all objects stored in the app's internal memory. Ordinarily, these would be passed to the
interface environment, but this function is useful if the app crashed and important results were only present in the
app's internal environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieve_tempdata()
</code></pre>


<h3>Value</h3>

<p>A list of all KinPairData objects in kindisperse app's tempdata
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>retrieve_tempdata()
</code></pre>

<hr>
<h2 id='retrieveall_appdata'>Retrieve all <code>KinPairData</code> objects from appdata (as list)</h2><span id='topic+retrieveall_appdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>This function accesses the app interface environment and retrieves a named list of all objects (typically of classes
<code>KinPairData</code> or <code>KinPairSimulation</code> contained within it, making them accessible outside of the app). This
is used to quickly pass all simulation objects that were saved to this interface environment while using the app to
the regular R environment (after closing the app).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>retrieveall_appdata()
</code></pre>


<h3>Value</h3>

<p>Returns a list of objects stored in the appdata environment
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+unmount_appdata">unmount_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mount_appdata(kin_pair_data(), "k1")
mount_appdata(kin_pair_simulation(), "s1")
retrieveall_appdata()
</code></pre>

<hr>
<h2 id='run_kindisperse'>Run kindisperse app</h2><span id='topic+run_kindisperse'></span>

<h3>Description</h3>

<p>Run kindisperse app
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run_kindisperse()
</code></pre>


<h3>Value</h3>

<p>returns a shiny app instance of kindisperse
</p>

<hr>
<h2 id='sample_kindist'>Subsample and filter a <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> or <code><a href="#topic+KinPairData">KinPairData</a></code> Object</h2><span id='topic+sample_kindist'></span>

<h3>Description</h3>

<p>This function takes a pre-existing <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> or <code><a href="#topic+KinPairData">KinPairData</a></code> Object with distance and coordinate data and filters
it to simulate various in-field sampling schemes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_kindist(
  kindist,
  upper = NULL,
  lower = NULL,
  spacing = NULL,
  n = NULL,
  dims = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_kindist_+3A_kindist">kindist</code></td>
<td>
<p><code>KinPairSimulation</code> Class Object</p>
</td></tr>
<tr><td><code id="sample_kindist_+3A_upper">upper</code></td>
<td>
<p><code>numeric</code> - upper cutoff for kin pair distances</p>
</td></tr>
<tr><td><code id="sample_kindist_+3A_lower">lower</code></td>
<td>
<p><code>numeric</code> - lower cutoff for kin pair distances</p>
</td></tr>
<tr><td><code id="sample_kindist_+3A_spacing">spacing</code></td>
<td>
<p><code>numeric</code> - spacing between traps (location-independent)</p>
</td></tr>
<tr><td><code id="sample_kindist_+3A_n">n</code></td>
<td>
<p><code>numeric</code> - number of individuals to keep after filtering (if possible)</p>
</td></tr>
<tr><td><code id="sample_kindist_+3A_dims">dims</code></td>
<td>
<p>dimensions to sample within (works with the <code>KinPairSimulation</code> spatial &amp; dimension information).
Either <code>num</code> (defining a square) or <code>c(num1, num2)</code> (defining a rectangle).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function enables the testing of the impact of some basic sampling constraints that might be encountered in study design or
implementation on the effectiveness of the <code>kindisperse</code> estimation of intergenerational dispersal. It is typically paired with
a simulation function such as <code><a href="#topic+simulate_kindist_composite">simulate_kindist_composite</a></code> to generate a 'pure' dataset, then an estimation function
such as <code><a href="#topic+axpermute">axpermute</a></code> to examine the impact of filter settings on the 'detected' value of dispersal sigma. The filter
parameters <code>upper</code>, <code>lower</code>, &amp; <code>spacing</code> all work on the vector of (direction-independent) distances, &amp; the parameter
<code>n</code> enables the random subsampling of n kin dyads. The parameter <code>dims</code> requires 2D location information for each individual,
meaning it can ordinarily only be used with the <code>KinPairSimulation</code> object (not <code>KinPairData</code>). All filter parameters are
stackable.
</p>
<p>The <code>upper</code> parameter implements a cutoff for the <strong>maximum</strong> distance allowable in the dataset. If set to e.g. 100m, all kin dyads
separated by a distance greater than 100m will be excluded from the filtered dataset. Note that this is a geometry-independent metric;
it is naive to the edge effects of an actual sample site. The <code>lower</code> parameter implements a cutoff for the <strong>minimum</strong>
distance allowable in the dataset.It operates in the same manner as the previous parameter (in this case, removing results smaller
than a distance threshold)
</p>
<p>The <code>spacing</code> parameter as currently implemented takes all distances &amp; alters them to lie at the midpoint of a bin with width set
by this parameter. So if <code>spacing</code> is set to 10 meters, all kin pairs with distances between 0 and 10m will have their distances
rest to 5m, all between 10 &amp; 20 will be set to 15 m, etc. (quantizing the data). Note that once again this is a geometry-independent
action: These binwiths &amp; 'trap spacing' are not spatially related to each other like they would be in a sample site, and there is no
simulated dropout of kinpairs too far from a trap. There is also no geometry-dependent profiling of possible frequency of recaptures
across each distance category (will be implemented in a future version). (this parameter leaves 2D spatial information intact)
</p>
<p>The <code>dims</code> parameter defines the dimensions of a rectangle within which <strong>both</strong> individuals of a kin dyad will need to lie
to be included in the filtered dataset. This measure (which excludes e.g. long-distance dispersal into &amp; out of the study site) is
<em>geometry-dependent</em>, unlike the <code>upper</code> parameter. This enables the testing of (rectangular) site geometries potentially
corresponding to an actual site (two-dimensional estimates of dispersal such as <code>kindisperse</code> become unreliable as edge effects
significantly reduce the size of either one or both dimensions with respect to the real underlying dispersal sigma). These site
geometries can be entered in a few ways: (a) a single numeric value, which will be interpreted as the length of the side of a square;
(b) a numeric vector of length two, which will be interpreted as the length &amp; width of the sample site; (c) either of the above passed
to the <code><a href="#topic+elongate">elongate</a></code> function, which takes the rectangular site dimensions and alters their <code>aspect ratio</code> (ratio of
length to width) while preserving the underlying area the study site covers. The implementation of this filtering step permutes the
absolute positions of all dyads so that at least one member of the dyad is in the inial site rectangle, while preserving their relative
positions (and angles) with respect to each other. This means that following this step, the xy coordinate positions of each
individual will not match those contained in the previous round. It also means that the repeated calling of this function will
result in a steady reduction in retained kin dyads due to edge effects.
</p>
<p>The <code>n</code> parameter randomly samples n pairs from the dataset. It is implemented after all other filtering has taken place, so
will only sample surviving individuals A typical strategy for the use of this functions in simulations would be to simulate an
extremely large (e.g. one million pairs) dataset, then pass it repeatedly to this filter function, with a final sub-sampling step of
1,000 included. This enables comparisons across sampling conditions (in most cases) regardless of the amount of data filtered prior to
this step.
</p>
<p>As this function returns a <code>KinPairData</code> or <code>KinPairSimulation</code> object, the returned object can be passed back for filtering
an arbitrary number of times, or alternatively passed to an estimation strategy.
</p>
<p>This function can be used to test for bias in the results of a close-kin dispersal study that has been conducted. After the field
sampling, kin identification, &amp; sigma calculation steps, use the estimated sigmas as inputs into simulation functions that are
then filtered for size &amp; geometry of the actual study site (via the <code>dims</code> method). Then pass this filtered dataset back to the
sigma-determining functions. If filtering has resulted in a substantial drop in sigma, the estimate of sigma from the study site has
likely been biased by the site geometry (note that the impact of this is dependent on the shape of the dispersal kernel - the more
leptokurtic (dominated by long-distance dispersal), the more severe bias will be for a particular sigma and site geometry.
</p>


<h3>Value</h3>

<p>returns an object of class <code>KinPairData</code> or <code>KinPairSimulation</code> containing simulation and filtering
details and a filtered dataset of dispersed individuals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simobject &lt;- simulate_kindist_simple(nsims = 100000, sigma = 100, kinship = "PO")

sample_kindist(simobject, upper = 200, lower = 50, spacing = 15, n = 100)
</code></pre>

<hr>
<h2 id='sampling_stage'>Access sampling stage of <code><a href="#topic+DispersalModel">DispersalModel</a></code> or <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object.</h2><span id='topic+sampling_stage'></span><span id='topic+sampling_stage+3C-'></span><span id='topic+sampling_stage+2CDispersalModel-method'></span><span id='topic+sampling_stage+3C-+2CDispersalModel-method'></span><span id='topic+sampling_stage+2CKinPairData-method'></span>

<h3>Description</h3>

<p>Access sampling stage of <code><a href="#topic+DispersalModel">DispersalModel</a></code> or <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sampling_stage(x)

sampling_stage(x) &lt;- value

## S4 method for signature 'DispersalModel'
sampling_stage(x)

## S4 replacement method for signature 'DispersalModel'
sampling_stage(x) &lt;- value

## S4 method for signature 'KinPairData'
sampling_stage(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sampling_stage_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="sampling_stage_+3A_value">value</code></td>
<td>
<p><code>character</code> new sampling stage to assign model</p>
</td></tr>
<tr><td><code id="sampling_stage_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="sampling_stage_+3A_kinpairdata">KinPairData</code></td>
<td>
<p>object of class <code>KinPairData</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> sampling stage
</p>
<p>returns a modified object of class <code>DispersalModel</code>
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li>
<li> <p><code>KinPairData</code>: 
</p>
</li></ul>

<hr>
<h2 id='simdims'>Access simulation dimensions of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object</h2><span id='topic+simdims'></span><span id='topic+simdims+3C-'></span><span id='topic+simdims+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>Access simulation dimensions of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simdims(x)

simdims(x) &lt;- value

## S4 method for signature 'KinPairSimulation'
simdims(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simdims_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="simdims_+3A_value">value</code></td>
<td>
<p>new value to assign</p>
</td></tr>
<tr><td><code id="simdims_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>numeric vector</code> dimensions of simulated object
</p>
<p>returns a modified object of the relevant class
</p>
<p><code>numeric vector</code> simulation dimensions of <code>KinPairSimulation</code> object
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairSimulation</code>: 
</p>
</li></ul>

<hr>
<h2 id='simgraph_data'>Simple kin dispersal simulation for graphical display. (returns the data side as a tibble).</h2><span id='topic+simgraph_data'></span>

<h3>Description</h3>

<p>Simple kin dispersal simulation for graphical display. (returns the data side as a tibble).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simgraph_data(nsims = 1000, posigma = 50, dims = 250, kinship = "2C")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simgraph_data_+3A_nsims">nsims</code></td>
<td>
<p>Integer. The number of kin dispersal families to simulate.</p>
</td></tr>
<tr><td><code id="simgraph_data_+3A_posigma">posigma</code></td>
<td>
<p>Integer. The axial deviation of the (simple) parent-offspring dispersal kernel governing this simulation.</p>
</td></tr>
<tr><td><code id="simgraph_data_+3A_dims">dims</code></td>
<td>
<p>Integer. Lays out the length of the sides of a square within which parent individuals are seeded.</p>
</td></tr>
<tr><td><code id="simgraph_data_+3A_kinship">kinship</code></td>
<td>
<p>Character. Lists the kin category the simulation is reconstructing. One of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot; (no half-categories included)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a tibble containing the coordinates of the f0 to f2 generations, as well as coordinates and distances relative to the 'focus' kinship categories. (kindist, kinmid, k1 &amp; k2)
</p>


<h3>See Also</h3>

<p>Other simgraph: 
<code><a href="#topic+simgraph_graph">simgraph_graph</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simgraph_data(nsims = 100, dims = 1000, kinship = "GAV")
</code></pre>

<hr>
<h2 id='simgraph_graph'>Simple kin dispersal simulation for graphical display. (graphs the pre-existing simulation).</h2><span id='topic+simgraph_graph'></span>

<h3>Description</h3>

<p>Simple kin dispersal simulation for graphical display. (graphs the pre-existing simulation).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simgraph_graph(
  result,
  nsims = 10,
  labls = TRUE,
  steps = TRUE,
  moves = TRUE,
  shadows = TRUE,
  kinship = NULL,
  show_area = TRUE,
  centred = FALSE,
  pinwheel = FALSE,
  scattered = FALSE,
  lengths = TRUE,
  lengthlabs = TRUE,
  histogram = FALSE,
  binwidth = posigma/5,
  freqpoly = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simgraph_graph_+3A_result">result</code></td>
<td>
<p>simulation supplied from simgraph_data() function (tibble)</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_nsims">nsims</code></td>
<td>
<p>number of families to graph</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_labls">labls</code></td>
<td>
<p>Logical. Displays labels.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_steps">steps</code></td>
<td>
<p>Logical. Whether or not to show any details of dispersal movement</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_moves">moves</code></td>
<td>
<p>Logical. Whether or not to show (curved) lines denoting dispersal movement</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_shadows">shadows</code></td>
<td>
<p>Logical. Whether or not to show (dashed) shadows tracing dispersal movement.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_kinship">kinship</code></td>
<td>
<p>Character. Lists the kin category the simulation is reconstructing. One of &quot;PO&quot;, &quot;FS&quot;, &quot;HS&quot;, &quot;AV&quot;, &quot;GG&quot;, &quot;HAV&quot;, &quot;GGG&quot;, &quot;1C&quot;, &quot;1C1&quot;, &quot;2C&quot;, &quot;GAV&quot; (no half-categoris included)</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_show_area">show_area</code></td>
<td>
<p>Logical. Whether or not to show the parental seed area as defined in data$dims</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_centred">centred</code></td>
<td>
<p>Logical. Whether or not to centre the coordinates on one individual.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_pinwheel">pinwheel</code></td>
<td>
<p>Logical. Whether the final graph should be of the pinwheel form.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_scattered">scattered</code></td>
<td>
<p>Logical. Whether the final graph should be of the scatter form.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_lengths">lengths</code></td>
<td>
<p>Logical. Whether or not to show a dashed line connecting the 'focus' kin to illustrate overall distance of dispersal.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_lengthlabs">lengthlabs</code></td>
<td>
<p>Logical. Whether to show labels denoting distance of dispersal between focus kin.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_histogram">histogram</code></td>
<td>
<p>Logical. Whether the final graph should be of the histogram form.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_binwidth">binwidth</code></td>
<td>
<p>Numeric. Binwidth for histogram or freqpoly.</p>
</td></tr>
<tr><td><code id="simgraph_graph_+3A_freqpoly">freqpoly</code></td>
<td>
<p>Logical. Whether the final graph should be of the freqpoly form.</p>
</td></tr>
</table>


<h3>Value</h3>

<pre> Returns a ggplot object for graphing.
</pre>


<h3>See Also</h3>

<p>Other simgraph: 
<code><a href="#topic+simgraph_data">simgraph_data</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simdata &lt;- simgraph_data()
simgraph_graph(simdata)
</code></pre>

<hr>
<h2 id='simtype'>Access or assign simulation type of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object</h2><span id='topic+simtype'></span><span id='topic+simtype+3C-'></span><span id='topic+simtype+2CKinPairSimulation-method'></span>

<h3>Description</h3>

<p>Access or assign simulation type of <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code> object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simtype(x)

simtype(x) &lt;- value

## S4 method for signature 'KinPairSimulation'
simtype(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simtype_+3A_x">x</code></td>
<td>
<p>object of class <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="simtype_+3A_value">value</code></td>
<td>
<p>new value to assign</p>
</td></tr>
<tr><td><code id="simtype_+3A_kinpairsimulation">KinPairSimulation</code></td>
<td>
<p>object of class KinPairSimulation</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> the kind of simulation stored in the object (simple or composite)
</p>
<p>returns a modified object of relevant class
</p>
<p><code>character</code> the kind of simulation stored in the object (simple or composite)
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>KinPairSimulation</code>: 
</p>
</li></ul>


<h3>See Also</h3>

<p>Other kpsmethods: 
<code><a href="#topic+access_sigmas">access_sigmas</a></code>,
<code><a href="#topic+filter_methods">filter_methods</a></code>,
<code><a href="#topic+kernelshape">kernelshape</a>()</code>,
<code><a href="#topic+kerneltype">kerneltype</a>()</code>
</p>

<hr>
<h2 id='simulate_kindist_composite'>Simulate kin dispersal distance pairs with composite sigmas</h2><span id='topic+simulate_kindist_composite'></span>

<h3>Description</h3>

<p>Simulates intergenerational dispersal made up of composite dispersal stages in a species with a defined breeding and dispersal
structure similar to that of <em>Ae. aegypti</em> - i.e. with initial, breeding, gravid &amp; ovipositional dispersal phases,
approximately non-overlapping life cycles, and defined sampling points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_kindist_composite(
  nsims = 100,
  initsigma = 100,
  breedsigma = 50,
  gravsigma = 50,
  ovisigma = 25,
  dims = 100,
  method = "Gaussian",
  kinship = "FS",
  lifestage = "immature",
  shape = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_kindist_composite_+3A_nsims">nsims</code></td>
<td>
<p>(integer) -   number of pairs to simulate</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_initsigma">initsigma</code></td>
<td>
<p>(numeric) -   size of pre-breeding (axial) sigma</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_breedsigma">breedsigma</code></td>
<td>
<p>(numeric) -   size of breeding (axial) sigma</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_gravsigma">gravsigma</code></td>
<td>
<p>(numeric) -   size of post-breeding (axial) sigma</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_ovisigma">ovisigma</code></td>
<td>
<p>(numeric) -   size of oviposition (axial) sigma</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_dims">dims</code></td>
<td>
<p>(numeric) -   length of sides of (square) simulated site area</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_method">method</code></td>
<td>
<p>(character) - kernel shape to use: either 'Gaussian', 'Laplace' or 'vgamma' (variance-gamma)</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_kinship">kinship</code></td>
<td>
<p>(character)- kin category to simulate: one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C H1C1 or H2C</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_lifestage">lifestage</code></td>
<td>
<p>(character) lifestage at sample collection: either 'immature' or 'ovipositional'</p>
</td></tr>
<tr><td><code id="simulate_kindist_composite_+3A_shape">shape</code></td>
<td>
<p>(numeric) - value of shape parameter to use with 'vgamma' method. Default 0.5. Must be &gt; 0. Increment towards zero for increasingly heavy-tailed (leptokurtic) dispersal</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is one of a family of functions that implement the core intergenerational dispersal simulations
contained in the <code>kindisperse</code> package. Each of these functions proceeds by the following steps:
</p>

<ol>
<li><p> identify the pedigree
relationship, dispersal phase (FS, HS &amp; PO) and sampling stage that must be generated;
</p>
</li>
<li><p> randomly assign a coordinate position
to the 'root' individual within the pedigree (i.e. last common ancestor of the dyad, inclusive);
</p>
</li>
<li><p> 'disperse' both pathways from
this root position via the appropriately defined phase dispersal (additively via random draws from the underlying statistical
model, defined by an axial standard deviation - sigma);
</p>
</li>
<li><p> further disperse both phased descendant branches according to the
number of realised breeding dispersal cycles contained in the defining pedigree (additively via random draws from the chosen
underlying statistical model);
</p>
</li>
<li><p> add displacement caused by dispersal before the sampling point in a similar manner to above,
defining the final positions of the sampled dispersed kin dyads;
</p>
</li>
<li><p> calculating geographical distances between the
resulting dyads.
</p>
</li></ol>

<p>These simulation functions operate under an additive variance framework: all individual dispersal events are modeled as random
draws from a bivariate probability distribution defined by an axial standard deviation <code>sigma</code> and (sometimes) a shape
parameter. At present, three such distributions are included as options accessible with the <code>method</code> parameter: the
bivariate normal distribution '<code>Gaussian</code>', the bivariate Laplace distribution '<code>Laplace</code>', and the bivariate
variance-gamma distribution '<code>vgamma</code>'. The <code>Gaussian</code> (normal) distribution enables easy compatibility with the
framework under which much population genetic &amp; dispersal theory (isolation by distance, neighbourhoods, etc.) have been
developed. The <code>Laplace</code> distribution is a multivariate adaptation of the (positive) exponential distribution, and
represents a more 'fat-tailed' (leptokurtic) disperal situation than Gaussian. The <code>vgamma</code> distribution is a mixture
distribution formed by mixing the gamma distribution with the bivariate normal distribution. The flexibility of this
distribution's <code>shape</code> parameter enables us to model arbitrarily leptokurtic dispesal kernels, providing a helpful way
to examine the impacts of (e.g.) long distance dispersal on the overall disperal distribution and sampling decisions. A
<code>vgamma</code> distribution with shape parameter equal to 1 reduces to the bivariate Laplace distribution. As shape approaches
infinity, the <code>vgamma</code> distribution approaches the bivariate normal distribution. As shape approaches zero, the distribution
becomes increasingly leptokurtic.
</p>
<p>The <code>simulate_kindist_composite()</code> function is designed to enable modeling of the composite dispersal events that occur
<strong>within</strong> the breeding cycle of an organism, and enables the separate treatment of the <code>PO, FS</code>, and <code>HS</code>
phases (where, for example, the final distributions of full and half siblings are different in contexts where males mate
with multiple females but females primarily carry the offspring of one male). This function has been designed primarily in the
context of modelling dispersal in the mosquito <em>Ae. aegypti</em>; parameter names and the structure of kinship phases
reflect a single-generational breeding organism with an initial dispersal phase, a mating phase (where HS individuals branch),
a gravid phase, and an oviposition phase (where FS individuals branch). The sampling options ('immature' &amp; 'ovipositional')
also reflect common mosquito trapping methods (i.e. ovitraps &amp; gravitraps) which both target individuals dispersing in the
defined oviposition phase. This function should be easily adaptable to a vast number of other animals, especially insects, where
breeding occurs in one generation and parameters such as this hold. For slightly more complex scenarios (multiple breeding cycles,
differing sample points, more or less dispersal components making up a lifespan, different FS/HS branchpoints, etc.), the
enhanced capabilities of the <code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a></code> function may be required.
</p>
<p>Following simulation, the results are returned as an object of the specially defined package class <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code>,
which stores the simulation results along with information about all simulation parameters, and can be further passed to
sample filtering &amp; dispersal estimation functions.
</p>


<h3>Value</h3>

<p>returns an object of class <code>KinPairSimulation</code> containing simulation details and a tibble (tab) of simulation values
</p>


<h3>See Also</h3>

<p>Other simulate_kindist: 
<code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a>()</code>,
<code><a href="#topic+simulate_kindist_simple">simulate_kindist_simple</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simulate_kindist_composite(nsims = 100)
simulate_kindist_composite(
  nsims = 10000, initsigma = 20, breedsigma = 30, gravsigma = 30,
  ovisigma = 12, dims = 500, method = "Laplace", kinship = "1C", lifestage = "immature"
)
</code></pre>

<hr>
<h2 id='simulate_kindist_custom'>Simulate kin dispersal distance pairs with custom species dispersal models.</h2><span id='topic+simulate_kindist_custom'></span>

<h3>Description</h3>

<p>Simulates intergenerational dispersal in a species defined by multiple dispersal components across the breeding cycle, with
dispersal, breeding &amp; sampling &amp; basic generational structure custom-defined by a <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_kindist_custom(
  nsims = 100,
  model = dispersal_model(init = 100, breed = 50, grav = 50, ovi = 25, .FS = "ovi", .HS
    = "breed"),
  dims = 100,
  method = "Gaussian",
  kinship = "FS",
  cycle = 0,
  shape = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_kindist_custom_+3A_nsims">nsims</code></td>
<td>
<p>(integer) -   number of pairs to simulate</p>
</td></tr>
<tr><td><code id="simulate_kindist_custom_+3A_model">model</code></td>
<td>
<p>(object of class <code>DispersalModel</code>) - species-specific model of dispersal containing lifestage,  phase  &amp; sampling parameters</p>
</td></tr>
<tr><td><code id="simulate_kindist_custom_+3A_dims">dims</code></td>
<td>
<p>(numeric) -   length of sides of (square) simulated site area</p>
</td></tr>
<tr><td><code id="simulate_kindist_custom_+3A_method">method</code></td>
<td>
<p>(character) - kernel shape to use: either 'Gaussian', 'Laplace' or 'vgamma' (variance-gamma)</p>
</td></tr>
<tr><td><code id="simulate_kindist_custom_+3A_kinship">kinship</code></td>
<td>
<p>(character)- kin category to simulate: one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C H1C1 or H2C</p>
</td></tr>
<tr><td><code id="simulate_kindist_custom_+3A_cycle">cycle</code></td>
<td>
<p>(numeric) - breeding cycle number(s) of dispersed kin to be modeled. Must be a integer equal to or greater than -1, (-1, 0, 1, 2, ...)
or vector of two such integers. Represents the number of complete breeding cycles each simulated individual has undergone before the sampling point,
where the time between birth and first reproduction is coded as '0', that between first and second reproduction '1', etc. (default 0).
If <code>cycle</code> is specially set to '-1' this constitutes the sampling of an individual before it has differentiated (via dispersal) from the parent.
Only use in spp. where there is likely to be a reasonable equivalence between breeding stages across a lifespan.</p>
</td></tr>
<tr><td><code id="simulate_kindist_custom_+3A_shape">shape</code></td>
<td>
<p>(numeric) - value of shape parameter to use with 'vgamma' method. Default 0.5. Must be &gt; 0. Increment towards zero for increasingly heavy-tailed (leptokurtic) dispersal</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is one of a family of functions that implement the core intergenerational dispersal simulations
contained in the <code>kindisperse</code> package. Each of these functions proceeds by the following steps:
</p>

<ol>
<li><p> identify the pedigree
relationship, dispersal phase (FS, HS &amp; PO) and sampling stage that must be generated;
</p>
</li>
<li><p> randomly assign a coordinate position
to the 'root' individual within the pedigree (i.e. last common ancestor of the dyad, inclusive);
</p>
</li>
<li><p> 'disperse' both pathways from
this root position via the appropriately defined phase dispersal (additively via random draws from the underlying statistical
model, defined by an axial standard deviation - sigma);
</p>
</li>
<li><p> further disperse both phased descendant branches according to the
number of realised breeding dispersal cycles contained in the defining pedigree (additively via random draws from the chosen
underlying statistical model);
</p>
</li>
<li><p> add displacement caused by dispersal before the sampling point in a similar manner to above,
defining the final positions of the sampled dispersed kin dyads;
</p>
</li>
<li><p> calculating geographical distances between the
resulting dyads.
</p>
</li></ol>

<p>These simulation functions operate under an additive variance framework: all individual dispersal events are modeled as random
draws from a bivariate probability distribution defined by an axial standard deviation <code>sigma</code> and (sometimes) a shape
parameter. At present, three such distributions are included as options accessible with the <code>method</code> parameter: the
bivariate normal distribution '<code>Gaussian</code>', the bivariate Laplace distribution '<code>Laplace</code>', and the bivariate
variance-gamma distribution '<code>vgamma</code>'. The <code>Gaussian</code> (normal) distribution enables easy compatibility with the
framework under which much population genetic &amp; dispersal theory (isolation by distance, neighbourhoods, etc.) have been
developed. The <code>Laplace</code> distribution is a multivariate adaptation of the (positive) exponential distribution, and
represents a more 'fat-tailed' (leptokurtic) disperal situation than Gaussian. The <code>vgamma</code> distribution is a mixture
distribution formed by mixing the gamma distribution with the bivariate normal distribution. The flexibility of this
distribution's <code>shape</code> parameter enables us to model arbitrarily leptokurtic dispesal kernels, providing a helpful way
to examine the impacts of (e.g.) long distance dispersal on the overall disperal distribution and sampling decisions. A
<code>vgamma</code> distribution with shape parameter equal to 1 reduces to the bivariate Laplace distribution. As shape approaches
infinity, the <code>vgamma</code> distribution approaches the bivariate normal distribution. As shape approaches zero, the distribution
becomes increasingly leptokurtic.
</p>
<p>The <code>simulate_kindist_custom()</code> function is designed to enable modeling of the composite dispersal events that occur
<strong>within</strong> the breeding cycle of an organism, and enables the separate treatment of the <code>PO, FS</code>, and <code>HS</code>
phases in situations where the breeding and dispersal cycle of an organism is (somewhat more complex that that
encountered in organisms such as mosquitoes (i.e. single-generational breeding organisms with defined sampling
points). This function relies on a custom dispersal model of class <code><a href="#topic+DispersalModel">DispersalModel</a></code> defined via parameter
<code>model</code> to supply organism-specific information about dispersal stages (with axial sigmas), FS &amp; HS branch points, and
the dispersal stage at which sampling occurs. Via this model object (or overridden by the <code>cycle</code> parameter) you can
also define the number of breeding cycles each final individual within the close-kin dyad has passed through before sampling.
This is defined as a length one or two non-negative integer (where a length-one integer of value a is converted to a length
two integer of value c(a, a)), where the first integer defines the number of life cycles passed through by the 'senior'
pedigree member of the dyad, and the second the number passed through by the 'junior' member (so the GG phase has a grandparent
as senior, the grandchild as junior, etc. (in practice this distinction is unimportant for dyads). A cycle number of 0
references an individual that hasn't lived through an entire breeding cycle (sampling phase to sampling phase) before being
sampled. A value of 1 references an individual that has lived through one such cycle (e.g. a female entering her second breeding
season, an ovipositing mosquito (where the oviposition dispersal stage overlaps with the larval dispersal stage)). A value of
2 references two such cycles, etc. As all cycles are considered equivalent in the current formulation of this model (whether
an individual enters the cycle as a juvenile or as an adult) care must be taken in applying this system to species where the
dispersal behaviour of a second cycle individual (i.e. adult) is likely to be substantially different to that of a first cycle
individual (often an immature individual).
</p>
<p>This function can only handle one kinship pairing &amp; dispersal mode in the one simulation: where multiple dispersal pathways lead
to the same kinship outcome, each pathway should be simulated separately, and the resulting distributions combined subsequently.
</p>
<p>Following simulation, the results are returned as an object of the specially defined package class <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code>,
which stores the simulation results along with information about all simulation parameters, and can be further passed to
sample filtering &amp; dispersal estimation functions.
</p>


<h3>Value</h3>

<p>returns an object of class <code>KinPairSimulation</code> containing simulation details and a tibble (tab) of simulation values
</p>


<h3>See Also</h3>

<p>Other simulate_kindist: 
<code><a href="#topic+simulate_kindist_composite">simulate_kindist_composite</a>()</code>,
<code><a href="#topic+simulate_kindist_simple">simulate_kindist_simple</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>custom_dispersal_model &lt;- dispersal_model(a = 10, b = 25, .FS = "b",
.HS = "a", .sampling_stage = "b")
simulate_kindist_custom(nsims = 100, model = custom_dispersal_model,
cycle = c(0, 1), kinship = "FS")
</code></pre>

<hr>
<h2 id='simulate_kindist_simple'>Simulate kin dispersal distance pairs with simple sigma</h2><span id='topic+simulate_kindist_simple'></span>

<h3>Description</h3>

<p>Simulates intergenerational dispersal defined by a simple dispersal sigma (covering the entire lifecycle) and ignoring phase
differences between full &amp; half sibling dispersal categories. Returns an object of class <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulate_kindist_simple(
  nsims = 100,
  sigma = 125,
  dims = 100,
  method = "Gaussian",
  kinship = "PO",
  lifestage = "immature",
  shape = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simulate_kindist_simple_+3A_nsims">nsims</code></td>
<td>
<p>(integer) -   number of pairs to simulate</p>
</td></tr>
<tr><td><code id="simulate_kindist_simple_+3A_sigma">sigma</code></td>
<td>
<p>(numeric) -   size of simple (axial) sigma</p>
</td></tr>
<tr><td><code id="simulate_kindist_simple_+3A_dims">dims</code></td>
<td>
<p>(numeric) -   length of sides of (square) simulated site area</p>
</td></tr>
<tr><td><code id="simulate_kindist_simple_+3A_method">method</code></td>
<td>
<p>(character) - kernel shape to use: either 'Gaussian', 'Laplace' or 'vgamma' (variance-gamma)</p>
</td></tr>
<tr><td><code id="simulate_kindist_simple_+3A_kinship">kinship</code></td>
<td>
<p>(character)- kin category to simulate: one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C or H2C</p>
</td></tr>
<tr><td><code id="simulate_kindist_simple_+3A_lifestage">lifestage</code></td>
<td>
<p>(lifestage) lifestage at sample collection: either 'immature' or 'ovipositional'</p>
</td></tr>
<tr><td><code id="simulate_kindist_simple_+3A_shape">shape</code></td>
<td>
<p>(numeric) - value of shape parameter to use with 'vgamma' method. Default 0.5. Must be &gt; 0. Increment towards zero for increasingly heavy-tailed (leptokurtic) dispersal</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is one of a family of functions that implement the core intergenerational dispersal simulations
contained in the <code>kindisperse</code> package. Each of these functions proceeds by the following steps:
</p>

<ol>
<li><p> identify the pedigree
relationship, dispersal phase (FS, HS &amp; PO) and sampling stage that must be generated;
</p>
</li>
<li><p> randomly assign a coordinate position
to the 'root' individual within the pedigree (i.e. last common ancestor of the dyad, inclusive);
</p>
</li>
<li><p> 'disperse' both pathways from
this root position via the appropriately defined phase dispersal (additively via random draws from the underlying statistical
model, defined by an axial standard deviation - sigma);
</p>
</li>
<li><p> further disperse both phased descendant branches according to the
number of realised breeding dispersal cycles contained in the defining pedigree (additively via random draws from the chosen
underlying statistical model);
</p>
</li>
<li><p> add displacement caused by dispersal before the sampling point in a similar manner to above,
defining the final positions of the sampled dispersed kin dyads;
</p>
</li>
<li><p> calculating geographical distances between the
resulting dyads.
</p>
</li></ol>

<p>These simulation functions operate under an additive variance framework: all individual dispersal events are modeled as random
draws from a bivariate probability distribution defined by an axial standard deviation <code>sigma</code> and (sometimes) a shape
parameter. At present, three such distributions are included as options accessible with the <code>method</code> parameter: the
bivariate normal distribution '<code>Gaussian</code>', the bivariate Laplace distribution '<code>Laplace</code>', and the bivariate
variance-gamma distribution '<code>vgamma</code>'. The <code>Gaussian</code> (normal) distribution enables easy compatibility with the
framework under which much population genetic &amp; dispersal theory (isolation by distance, neighbourhoods, etc.) have been
developed. The <code>Laplace</code> distribution is a multivariate adaptation of the (positive) exponential distribution, and
represents a more 'fat-tailed' (leptokurtic) disperal situation than Gaussian. The <code>vgamma</code> distribution is a mixture
distribution formed by mixing the gamma distribution with the bivariate normal distribution. The flexibility of this
distribution's <code>shape</code> parameter enables us to model arbitrarily leptokurtic dispesal kernels, providing a helpful way
to examine the impacts of (e.g.) long distance dispersal on the overall disperal distribution and sampling decisions. A
<code>vgamma</code> distribution with shape parameter equal to 1 reduces to the bivariate Laplace distribution. As shape approaches
infinity, the <code>vgamma</code> distribution approaches the bivariate normal distribution. As shape approaches zero, the distribution
becomes increasingly leptokurtic.
</p>
<p>The <code>simulate_kindist_simple()</code> function is the most basic of the simulation functions, ignoring all information about
dispersal phase and treating dispersal with a single sigma corresponding to the entire lifecycle to breeding of the
dispersed individuals. It is useful for exploring simple intergenerational dispersal in a stripped back context; for many
typical contexts involving complex dispersal across different phases of the breeding cycle, the other dispersal simulation
functions would be more suitable.
</p>
<p>Following simulation, the results are returned as an object of the specially defined package class <code><a href="#topic+KinPairSimulation">KinPairSimulation</a></code>,
which stores the simulation results along with information about all simulation parameters, and can be further passed to
sample filtering &amp; dispersal estimation functions.
</p>


<h3>Value</h3>

<pre> returns an object of class \code{\link{KinPairSimulation}} containing simulation details and a \code{tibble} (tab) of simulation values
</pre>


<h3>See Also</h3>

<p>Other simulate_kindist: 
<code><a href="#topic+simulate_kindist_composite">simulate_kindist_composite</a>()</code>,
<code><a href="#topic+simulate_kindist_custom">simulate_kindist_custom</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test &lt;- simulate_kindist_simple(nsims = 10, sigma = 50, dims = 1000, method = "Laplace")
simulate_kindist_simple(nsims = 10000, sigma = 75, kinship = "PO", lifestage = "ovipositional")
</code></pre>

<hr>
<h2 id='stages'>Access breeding cycle stages of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.</h2><span id='topic+stages'></span><span id='topic+stages+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>Access breeding cycle stages of <code><a href="#topic+DispersalModel">DispersalModel</a></code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stages(x)

## S4 method for signature 'DispersalModel'
stages(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="stages_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
<tr><td><code id="stages_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> ordered vector of custom lifestages contained in the object
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li></ul>

<hr>
<h2 id='tsv_to_kinpair'>Reads <code>.tsv</code> and converts to <code>KinPairData</code> object</h2><span id='topic+tsv_to_kinpair'></span>

<h3>Description</h3>

<p>This function is part of suite of functions handling file import/export for kinship dispersal objects.
</p>
<p><code>.csv</code> &amp; <code>.tsv</code> reading functions at minimum require the .delim file to contain a column titled 'distance' containing distances
between kin pairs. It can optionally contain a column of kinship values 'kinship' as well as a column of lifestage values 'lifestage'.
If the file contains more than one value in the kinship or lifestage columns (e.g. bot 'FS' and 'HS') - the corresponding function
parameter must be set to pick a corresponding subset of dispersed pairs. where parameters are set in the absence of file columns,
these values are assigned to the returned <code>KinPairData</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsv_to_kinpair(file, kinship = NULL, lifestage = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsv_to_kinpair_+3A_file">file</code></td>
<td>
<p>The file path to read from</p>
</td></tr>
<tr><td><code id="tsv_to_kinpair_+3A_kinship">kinship</code></td>
<td>
<p>character. kin category to assign or extract from data. one of PO, FS, HS, AV, GG, HAV, GGG, 1C, 1C1, 2C, GAV, HGAV, H1C , H1C1 or H2C</p>
</td></tr>
<tr><td><code id="tsv_to_kinpair_+3A_lifestage">lifestage</code></td>
<td>
<p>character. lifestage to assign or extract from data. one of 'unknown', 'immature' or 'ovipositional'.</p>
</td></tr>
<tr><td><code id="tsv_to_kinpair_+3A_...">...</code></td>
<td>
<p>additional arguments to pass to <code>read_tsv</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns object of class <code>KinPairData</code>
</p>


<h3>See Also</h3>

<p>Other import_functions: 
<code><a href="#topic+csv_to_kinpair">csv_to_kinpair</a>()</code>,
<code><a href="#topic+df_to_kinpair">df_to_kinpair</a>()</code>,
<code><a href="#topic+read_kindata">read_kindata</a>()</code>,
<code><a href="#topic+vector_to_kinpair">vector_to_kinpair</a>()</code>
</p>

<hr>
<h2 id='unmount_appdata'>Unmount a <code>KinPairData</code> Object (clear slot from appdata environment)</h2><span id='topic+unmount_appdata'></span>

<h3>Description</h3>

<p>This function is part of a suite of functions handling the interface between the kindisperse app &amp; R
package. Due to how shiny's interactive programming works, ordinary objects are not visible to the reactive functions
embedded in the app. The solution implemented here is to construct a custom environment, <code>env_appdata</code>, that is
accessible within the app and outside of it.
</p>
<p>When called, this function clears any objects with names found in the vector <code>nms</code> from the app interface
environment, keeping it from becoming over-cluttered &amp; taking up space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unmount_appdata(nms)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unmount_appdata_+3A_nms">nms</code></td>
<td>
<p>A character vector of names of objects to unmount from the appdata environment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>See Also</h3>

<p>Other app_ports: 
<code><a href="#topic+display_appdata">display_appdata</a>()</code>,
<code><a href="#topic+mount_appdata">mount_appdata</a>()</code>,
<code><a href="#topic+reset_appdata">reset_appdata</a>()</code>,
<code><a href="#topic+reset_tempdata">reset_tempdata</a>()</code>,
<code><a href="#topic+retrieve_appdata">retrieve_appdata</a>()</code>,
<code><a href="#topic+retrieve_tempdata">retrieve_tempdata</a>()</code>,
<code><a href="#topic+retrieveall_appdata">retrieveall_appdata</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mount_appdata(kin_pair_data(), "mydata")

unmount_appdata("mydata")
</code></pre>

<hr>
<h2 id='vector_to_kinpair'>Convert vector of kin separation distances to <code><a href="#topic+KinPairData">KinPairData</a></code> class</h2><span id='topic+vector_to_kinpair'></span>

<h3>Description</h3>

<p>Function takes at minimum a (numeric) vector of distances between related kinpairs, and
returns a <code>KinPairData</code> object. Optional parameters can assign kinship and lifestage values to the
returned object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vector_to_kinpair(vect, kinship = NULL, lifestage = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vector_to_kinpair_+3A_vect">vect</code></td>
<td>
<p>vector of kinpair distances</p>
</td></tr>
<tr><td><code id="vector_to_kinpair_+3A_kinship">kinship</code></td>
<td>
<p>character or character vector containing kinship categories of kinpairs</p>
</td></tr>
<tr><td><code id="vector_to_kinpair_+3A_lifestage">lifestage</code></td>
<td>
<p>character or character vector containing lifestages of kinpairs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns valid <code>KinPairData</code> object.
</p>


<h3>See Also</h3>

<p>Other import_functions: 
<code><a href="#topic+csv_to_kinpair">csv_to_kinpair</a>()</code>,
<code><a href="#topic+df_to_kinpair">df_to_kinpair</a>()</code>,
<code><a href="#topic+read_kindata">read_kindata</a>()</code>,
<code><a href="#topic+tsv_to_kinpair">tsv_to_kinpair</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vector_to_kinpair(1:10, "FS", "immature")
</code></pre>

<hr>
<h2 id='visible_stage'>Access life stage at which individual is first visible to sampling (i.e. from which breeding cycles are calculated)</h2><span id='topic+visible_stage'></span><span id='topic+visible_stage+2CDispersalModel-method'></span>

<h3>Description</h3>

<p>Access life stage at which individual is first visible to sampling (i.e. from which breeding cycles are calculated)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visible_stage(x)

## S4 method for signature 'DispersalModel'
visible_stage(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visible_stage_+3A_x">x</code></td>
<td>
<p>object of class <code>DispersalModel</code> or </p>
</td></tr>
<tr><td><code id="visible_stage_+3A_dispersalmodel">DispersalModel</code></td>
<td>
<p>object of class <code>DispersalModel</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>character</code> stage in life cycle at which an individual is assumed to be sampled by default rather than its parent
(anchors the breeding cycle system)
</p>


<h3>Methods (by class)</h3>


<ul>
<li> <p><code>DispersalModel</code>: 
</p>
</li></ul>

<hr>
<h2 id='write_kindata'>Write <code>KinPairData</code> or <code>KinPairSimulation</code> object in <code>.kindata</code> format</h2><span id='topic+write_kindata'></span>

<h3>Description</h3>

<p>This function is part of suite of functions handling file import/export for kinship dispersal objects.
Writing to the custom <code>.kindata</code> format enables complete preservation of <code>KinPairData</code> &amp; <code>KinPairSimulation</code> formats
without any loss of class attributes or metadata - ideal for saving simulation data that is intended for further in-package
processing with kindisperse.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_kindata(x, file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_kindata_+3A_x">x</code></td>
<td>
<p>Object of class <code>KinPairData</code> or <code>KinPairSimulation</code></p>
</td></tr>
<tr><td><code id="write_kindata_+3A_file">file</code></td>
<td>
<p>The file path to write to. If is doesn't end it '<code>.kindata</code>', this will be added.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Invisibly returns the initial object
</p>


<h3>See Also</h3>

<p>Other export_functions: 
<code><a href="#topic+kinpair_to_csv">kinpair_to_csv</a>()</code>,
<code><a href="#topic+kinpair_to_tibble">kinpair_to_tibble</a>()</code>,
<code><a href="#topic+kinpair_to_tsv">kinpair_to_tsv</a>()</code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
