<!DOCTYPE html><html><head><title>Help for package KLINK</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {KLINK}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#KLINK-package'><p>KLINK: Kinship Analysis with Linked Markers</p></a></li>
<li><a href='#getLinkedPairs'><p>Identify linked marker pairs</p></a></li>
<li><a href='#launchApp'><p>Launch KLINK</p></a></li>
<li><a href='#LINKAGEMAP'><p>Built-in linkage map</p></a></li>
<li><a href='#linkedLR'><p>LR with pairwise linked markers</p></a></li>
<li><a href='#loadFamFile'><p>Load .fam file</p></a></li>
<li><a href='#markerSummary'><p>Generate table of marker data</p></a></li>
<li><a href='#parseXML'><p>Parse XML file associated with .fam file</p></a></li>
<li><a href='#paternity'><p>Dataset for a paternity case</p></a></li>
<li><a href='#writeResult'><p>Write data and results to Excel</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Kinship Analysis with Linked Markers</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>A 'shiny' application for forensic kinship testing, based on
    the 'pedsuite' R packages. 'KLINK' is closely aligned with the (non-R)
    software 'Familias' and 'FamLink', but offers several unique features,
    including visualisations and automated report generation. The
    calculation of likelihood ratios supports pairs of linked markers, and
    all common mutation models.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/magnusdv/KLINK">https://github.com/magnusdv/KLINK</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/magnusdv/KLINK/issues">https://github.com/magnusdv/KLINK/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>forrel (&ge; 1.6.0), gt (&ge; 0.10.0), openxlsx, pedFamilias,
pedmut (&ge; 0.7.1), pedprobr (&ge; 0.9.3), pedtools (&ge; 2.6.0),
shiny (&ge; 1.8.0), shinydashboard, shinyjs, verbalisr (&ge;
0.6.0), xml2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-02 13:26:49 UTC; magnu</td>
</tr>
<tr>
<td>Author:</td>
<td>Magnus Dehli Vigeland
    <a href="https://orcid.org/0000-0002-9134-4962"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Magnus Dehli Vigeland &lt;m.d.vigeland@medisin.uio.no&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-02 17:20:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='KLINK-package'>KLINK: Kinship Analysis with Linked Markers</h2><span id='topic+KLINK'></span><span id='topic+KLINK-package'></span>

<h3>Description</h3>

<p>A 'shiny' application for forensic kinship testing, based on the 'pedsuite' R packages. 'KLINK' is closely aligned with the (non-R) software 'Familias' and 'FamLink', but offers several unique features, including visualisations and automated report generation. The calculation of likelihood ratios supports pairs of linked markers, and all common mutation models.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Magnus Dehli Vigeland <a href="mailto:m.d.vigeland@medisin.uio.no">m.d.vigeland@medisin.uio.no</a> (<a href="https://orcid.org/0000-0002-9134-4962">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/magnusdv/KLINK">https://github.com/magnusdv/KLINK</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/magnusdv/KLINK/issues">https://github.com/magnusdv/KLINK/issues</a>
</p>
</li></ul>


<hr>
<h2 id='getLinkedPairs'>Identify linked marker pairs</h2><span id='topic+getLinkedPairs'></span>

<h3>Description</h3>

<p>This function returns a &quot;maximal&quot; set of disjoint pairs of linked markers,
given a genetic marker map and a subset of the markers included in the map.
The pairs are identified in a greedy manner, successively choosing the
closest markers on each chromosome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getLinkedPairs(markers, linkageMap, maxdist = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getLinkedPairs_+3A_markers">markers</code></td>
<td>
<p>A character vector containing marker names.</p>
</td></tr>
<tr><td><code id="getLinkedPairs_+3A_linkagemap">linkageMap</code></td>
<td>
<p>A data frame with columns including <code>Marker</code>, <code>Chrom</code> and
<code>PosCM</code>.</p>
</td></tr>
<tr><td><code id="getLinkedPairs_+3A_maxdist">maxdist</code></td>
<td>
<p>A positive number indicating the maximum linkage distance (in
cM). Markers further apart than this are considered unlinked.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of character vectors, each containing two marker names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example using the built-in map of 50 STR markers
map = KLINK::LINKAGEMAP

getLinkedPairs(map$Marker, map, maxdist = 25)
</code></pre>

<hr>
<h2 id='launchApp'>Launch KLINK</h2><span id='topic+launchApp'></span><span id='topic+runKLINK'></span>

<h3>Description</h3>

<p>This launches the KLINK app. <code>runKLINK()</code> is a synonym for <code>launchApp()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>launchApp()

runKLINK()
</code></pre>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
launchApp()

## End(Not run)

</code></pre>

<hr>
<h2 id='LINKAGEMAP'>Built-in linkage map</h2><span id='topic+LINKAGEMAP'></span>

<h3>Description</h3>

<p>A genetic map including 50 autosomal STR markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LINKAGEMAP
</code></pre>


<h3>Format</h3>

<p>A data frame with 50 rows and 4 columns: <code>Marker</code>, <code>Kit</code>, <code>Chrom</code>,
<code>PosCM</code>.
</p>

<hr>
<h2 id='linkedLR'>LR with pairwise linked markers</h2><span id='topic+linkedLR'></span>

<h3>Description</h3>

<p>This function does the main LR calculations of the KLINK app.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linkedLR(
  pedigrees,
  linkageMap,
  linkedPairs = NULL,
  maxdist = Inf,
  markerData = NULL,
  mapfun = "Kosambi",
  lumpSpecial = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linkedLR_+3A_pedigrees">pedigrees</code></td>
<td>
<p>A list of two pedigrees.</p>
</td></tr>
<tr><td><code id="linkedLR_+3A_linkagemap">linkageMap</code></td>
<td>
<p>A data frame with columns including <code>Marker</code>, <code>Chrom</code> and
<code>PosCM</code>.</p>
</td></tr>
<tr><td><code id="linkedLR_+3A_linkedpairs">linkedPairs</code></td>
<td>
<p>A list of marker pairs. If not supplied, calculated as
<code>getLinkedPairs(markerData$Marker, linkageMap, maxdist = maxdist)</code>.</p>
</td></tr>
<tr><td><code id="linkedLR_+3A_maxdist">maxdist</code></td>
<td>
<p>A number, passed onto <code>getLinkedMarkers()</code> if <code>linkedPairs</code> is
NULL.</p>
</td></tr>
<tr><td><code id="linkedLR_+3A_markerdata">markerData</code></td>
<td>
<p>A data frame with marker data, typically the output of
<code>markerSummary(pedigrees)</code>.</p>
</td></tr>
<tr><td><code id="linkedLR_+3A_mapfun">mapfun</code></td>
<td>
<p>Name of the map function to be used; either &quot;Haldane&quot; or
&quot;Kosambi&quot; (default).</p>
</td></tr>
<tr><td><code id="linkedLR_+3A_lumpspecial">lumpSpecial</code></td>
<td>
<p>A logical, by default TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with detailed LR results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(forrel)

ped1 = nuclearPed(fa = "AF", child = "CH") |&gt;
  profileSim(markers = NorwegianFrequencies)

ped2 = singletons(c("AF", "CH")) |&gt;
  transferMarkers(from = ped1, to = _)

pedigrees = list(ped1, ped2)

linkedLR(pedigrees, KLINK::LINKAGEMAP)

# For testing
# .linkedLR(pedigrees, markerpair = c("SE33", "D6S474"), linkageMap = LINKAGEMAP)

</code></pre>

<hr>
<h2 id='loadFamFile'>Load .fam file</h2><span id='topic+loadFamFile'></span>

<h3>Description</h3>

<p>Load .fam file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loadFamFile(path, fallbackModel = "equal", withParams = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loadFamFile_+3A_path">path</code></td>
<td>
<p>The path to a .fam file.</p>
</td></tr>
<tr><td><code id="loadFamFile_+3A_fallbackmodel">fallbackModel</code></td>
<td>
<p>The name of a mutation model; passed on to
<code><a href="pedFamilias.html#topic+readFam">pedFamilias::readFam()</a></code>.</p>
</td></tr>
<tr><td><code id="loadFamFile_+3A_withparams">withParams</code></td>
<td>
<p>A logical indicating if the Familias parameters should be
included in the output. (See <code><a href="pedFamilias.html#topic+readFam">pedFamilias::readFam()</a></code>.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two <code>ped</code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fam = system.file("extdata/halfsib-test.fam", package = "KLINK")
peds = loadFamFile(fam)
pedtools::plotPedList(peds)

</code></pre>

<hr>
<h2 id='markerSummary'>Generate table of marker data</h2><span id='topic+markerSummary'></span>

<h3>Description</h3>

<p>Generate table of marker data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>markerSummary(pedigrees, replaceNames = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="markerSummary_+3A_pedigrees">pedigrees</code></td>
<td>
<p>A list of 2 pedigrees.</p>
</td></tr>
<tr><td><code id="markerSummary_+3A_replacenames">replaceNames</code></td>
<td>
<p>A logical, indicating if IDs should be changed to
Person1, Person2, ...</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>markerSummary(paternity)

</code></pre>

<hr>
<h2 id='parseXML'>Parse XML file associated with .fam file</h2><span id='topic+parseXML'></span>

<h3>Description</h3>

<p>Parse XML file associated with .fam file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parseXML(xml)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="parseXML_+3A_xml">xml</code></td>
<td>
<p>Path to a file with extension .xml.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'># (No example included)

</code></pre>

<hr>
<h2 id='paternity'>Dataset for a paternity case</h2><span id='topic+paternity'></span>

<h3>Description</h3>

<p>A list of two pedigrees forming the hypotheses in a paternity case: H1 (<code>AF</code> is
the father of <code>CH</code>) and H2 (unrelated). <code>AF</code> and <code>CH</code> are genotyped with 11
markers, with allele frequencies from <code>forrel::NorwegianFrequencies</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>paternity
</code></pre>


<h3>Format</h3>

<p>A list of two pedigrees, named H1 and H2.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pedtools::plotPedList(paternity, marker = "SE33")
markerSummary(paternity)

forrel::kinshipLR(paternity)
</code></pre>

<hr>
<h2 id='writeResult'>Write data and results to Excel</h2><span id='topic+writeResult'></span>

<h3>Description</h3>

<p>This function produces an Excel document containing the genotype data and
various LR tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeResult(
  resultTable,
  pedigrees,
  linkageMap,
  markerData,
  outfile,
  notes = NULL,
  famname = NULL,
  settings = NULL,
  XML = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeResult_+3A_resulttable">resultTable</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_pedigrees">pedigrees</code></td>
<td>
<p>A list of two <code>ped</code> objects.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_linkagemap">linkageMap</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_markerdata">markerData</code></td>
<td>
<p>A data frame.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_outfile">outfile</code></td>
<td>
<p>The output file name.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_notes">notes</code></td>
<td>
<p>A character vector.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_famname">famname</code></td>
<td>
<p>The name of the input <code>.fam</code> file.</p>
</td></tr>
<tr><td><code id="writeResult_+3A_settings">settings</code></td>
<td>
<p>A list of KLINK settings to be included in the output</p>
</td></tr>
<tr><td><code id="writeResult_+3A_xml">XML</code></td>
<td>
<p>Optional data from .xml file.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# Built-in dataset `paternity`
peds = paternity
map = LINKAGEMAP
mdata = markerSummary(peds)

# Result table
LRtab = linkedLR(pedigrees = peds, linkageMap = map, markerData = mdata)

# Write to excel
tmp = paste0(tempfile(), ".xlsx")
writeResult(LRtab,
            pedigrees = peds,
            linkageMap = map,
            markerData = mdata,
            outfile = tmp)

# openxlsx::openXL(tmp)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
