<!DOCTYPE html><html lang="en"><head><title>Help for package KnockoffScreen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {KnockoffScreen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#create.MK'>
<p>Sequential knockoff generator for genetic data</p></a></li>
<li><a href='#KS.chr'>
<p>Scan a .vcf/.bgen file to identify the putative causal loci in whole-genome sequencing data</p></a></li>
<li><a href='#KS.prelim'>
<p>Preliminary data management for KnockoffScreen</p></a></li>
<li><a href='#KS.summary'>
<p>Summarize KnockoffScreen results</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Whole-Genome Sequencing Data Analysis via Knockoff Statistics</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-11-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Zihuai He &lt;zihuai@stanford.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zihuai He &lt;zihuai@stanford.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for identification of putative causal loci in whole-genome sequencing data. The functions allow genome-wide association scan. It also includes an efficient knockoff generator for genetic data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.5.0), Matrix, seqminer, bigmemory</td>
</tr>
<tr>
<td>Imports:</td>
<td>CompQuadForm, data.table, SPAtest, irlba</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-11-19 22:32:04 UTC; zhe</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-11-19 22:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='create.MK'>
Sequential knockoff generator for genetic data
</h2><span id='topic+create.MK'></span>

<h3>Description</h3>

<p>Generate single/multiple knockoffs for genetic variants for customized analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create.MK(X,pos,M=5,corr_max=0.75,maxN.neighbor=Inf,maxBP.neighbor=100000,
n.AL=floor(10*nrow(X)^(1/3)*log(nrow(X))),thres.ultrarare=25,
R2.thres=1,method='shrinkage',bigmemory=T)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create.MK_+3A_x">X</code></td>
<td>

<p>A n*p genotype matrix, where n is the sample size and p is the number of genetic variants.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_pos">pos</code></td>
<td>

<p>A vector of length p. Location of the p genetic variants.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_m">M</code></td>
<td>

<p>Number of knockoffs per variant. The default is 5.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_corr_max">corr_max</code></td>
<td>

<p>The correlation threshold for hierarchical clustering, such that variants from two different clusters do not have a correlation greater than corr_max. The hierarchical clustering step is a practical strategy to improve the power for tightly linked variants. The default is 0.75.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_maxn.neighbor">maxN.neighbor</code></td>
<td>

<p>The maximum number of neighoring variables used to generate knockoffs. The default is Inf. Smaller number will inprove the computational efficiency, but the knockoffs will be less accurate.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_maxbp.neighbor">maxBP.neighbor</code></td>
<td>

<p>The size of neighboring region (in base pairs) used to generate knockoffs. The default is 100000.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_n.al">n.AL</code></td>
<td>

<p>The sample size for the algorithmic leveraging. The default is 10*n^(1/3)*log(n)).
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_thres.ultrarare">thres.ultrarare</code></td>
<td>

<p>The minor allele count threshold that defines ultrarare variants. The knockoff generation for variants with minor allele counts below the threshold will be based on permutaton. The default is 25.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_r2.thres">R2.thres</code></td>
<td>

<p>The maximum R2 allowed in the auto-regressive model. More liberal values (&lt;1) lead to higher power for tightly linked variants, but the knockoffs will be less accurate.  The default is 1.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_method">method</code></td>
<td>

<p>The method for subsampling. The default is &quot;shrinkage&quot;, corresponding to &quot;shrinkage algorithmic leveraging&quot;.
</p>
</td></tr>
<tr><td><code id="create.MK_+3A_bigmemory">bigmemory</code></td>
<td>

<p>Whether &quot;bigmemory&quot; operation is applied. Default is TRUE.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>X_k</code></td>
<td>

<p>An M dimentions list, where each dimention is an n*p matrix as a knockoff copy of original data.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(KnockoffScreen)

# load example vcf file from package "seqminer"
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])

# filter out constant variants
s&lt;-apply(example.G,2,sd)
example.G&lt;-example.G[,s!=0]
pos&lt;-as.numeric(gsub("^.*:","",colnames(example.G)))

# generate multiple knockoffs
example.G_k&lt;-create.MK(example.G,pos,M=5,corr_max=0.75)

</code></pre>

<hr>
<h2 id='KS.chr'>
Scan a .vcf/.bgen file to identify the putative causal loci in whole-genome sequencing data
</h2><span id='topic+KS.chr'></span>

<h3>Description</h3>

<p>Once the preliminary work is done by &quot;KS.prelim()&quot;, this function scan a chromosome given a set of pre-defined windows. It also evalautes invidual variants within those windows.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KS.chr(result.prelim,seq.filename,window.bed,region.pos=NULL,
tested.pos=NULL,excluded.pos=NULL,M=5,thres.single=0.01,
thres.ultrarare=25,thres.missing=0.10,midout.dir=NULL,temp.dir=NULL,
jobtitle=NULL,Gsub.id=NULL,impute.method='fixed',
bigmemory=T,leveraging=T,LD.filter=NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KS.chr_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;KS.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_seq.filename">seq.filename</code></td>
<td>

<p>A character specifying the directory (including the file name) of the vcf/bgen file.The algorithm will recognize the file extension and analyze the file accordingly.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_window.bed">window.bed</code></td>
<td>

<p>A matrix specifying the windows being tested. Each row presents a window (chr, start, end), similar to a .bed file. We recommand to define window.bed with window sizes 1000,5000,10000 (base pairs) for sample size ~5000. For studies with smaller sample size, we recommand to increase the window size for stable inference of ultra rare variants.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_region.pos">region.pos</code></td>
<td>

<p>A vector specifying the start and end of each region being scanned.This provides better control for memory use. Usually we define region.pos such that number of variants in each region is bounded by 5000. For example: region.pos=c(1,100,300) corresponds to two regions (1,100) and (100,300). Default is defined by window.bed.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_tested.pos">tested.pos</code></td>
<td>

<p>A vector specifying position being tested. The default is to test all variants.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_excluded.pos">excluded.pos</code></td>
<td>

<p>A vector specifying position being excluded (due to QC or other filters). The default is to not exclude any variant.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_m">M</code></td>
<td>

<p>Number of knockoffs per variant. The default is 5.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_thres.single">thres.single</code></td>
<td>

<p>The minor allele frequency threshold to define single genetic variants being tested. Variants with minor allele frequencies above the threshold will be tested individually. The default is 0.01 (for sample size ~5000). Smaller threshold requires a larger sample size.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_thres.ultrarare">thres.ultrarare</code></td>
<td>

<p>The minor allele count threshold to filter out ultrarare variants. Variants with minor allele counts below the threshold will be excluded. The default is 25.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_thres.missing">thres.missing</code></td>
<td>

<p>The missing rate threshold to filter out variants with a low genotyping rate. Variants with missing rate above the threshold will be excluded. The default is 0.1.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_midout.dir">midout.dir</code></td>
<td>

<p>A character specifying the directory to save intermediate results. It is recommended when large scale data is being analyzed.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_temp.dir">temp.dir</code></td>
<td>

<p>A character specifying the directory to save temporary data. It is required when CADD or GenoNet scores are used.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_jobtitle">jobtitle</code></td>
<td>

<p>A character specifying the job title.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an n dimensional vector. This is used to match phenotype with genotype. The default is NULL, where the subject id in the vcf file is used.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;bestguess&quot; uses the genotype with highest probability.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_bigmemory">bigmemory</code></td>
<td>

<p>Whether &quot;bigmemory&quot; operation is applied. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_leveraging">leveraging</code></td>
<td>

<p>Whether &quot;algorithmic leveraging&quot; is applied. Default is TRUE.
</p>
</td></tr>
<tr><td><code id="KS.chr_+3A_ld.filter">LD.filter</code></td>
<td>

<p>Choose the LD filter for tightly linked variants. Default is 0.75. This applied a hierarchical clustering of genetic variants prior to the analysis, where variants in the same cluster have a pair-wise correlation &gt;=0.75. Then the analysis is restricted to one randomly selected representative variant in each cluster.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>result.window</code></td>
<td>

<p>Results for all windows. Each row presents a window.
</p>
</td></tr>
<tr><td><code>result.single</code></td>
<td>

<p>Results for all individual variants with minor allele frequency above the specified threshold. Each row presents a variant.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(KnockoffScreen)

# load example vcf file from package "seqminer"
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])

# simulated outcomes, covariates and inidividual id.
Y&lt;-as.matrix(rnorm(nrow(example.G),0,1))
X&lt;-as.matrix(rnorm(nrow(example.G),0,1))
id&lt;-rownames(example.G)

# fit null model
result.prelim&lt;-KS.prelim(Y,X=X,id=id,out_type="C")

# Define the window.bed file
chr&lt;-1
pos.min&lt;-196621007;pos.max&lt;-196716634
window.size=c(2000)

window.bed&lt;-c();
for(size in window.size){
  pos.tag&lt;-seq(pos.min,pos.max,by=size*1/2)
  window.bed&lt;-rbind(window.bed,cbind(chr,pos.tag,pos.tag+size))
}
window.bed&lt;-window.bed[order(as.numeric(window.bed[,2])),]

# scan the vcf file
midout.dir&lt;-NULL # or '/YourProjectDir/MidResults/'
temp.dir&lt;-NULL # or '/YourProjectDir/Temp_out/' #this is a folder to save temporary results
jobtitle&lt;-'YourProjectTitle'

# we set thres.single=0.1,thres.ultrarare=0 for a proof of concept.
# note that the default for real data analysis is thres.single=0.01, thres.ultrarare=25
fit &lt;- KS.chr(result.prelim,vcf.filename,window.bed,M=5,thres.single=0.1,thres.ultrarare=0,
midout.dir=midout.dir,temp.dir=temp.dir,jobtitle=jobtitle)

# summarize the results
result.summary&lt;-KS.summary(fit$result.window,fit$result.single,M=5)

</code></pre>

<hr>
<h2 id='KS.prelim'>
Preliminary data management for KnockoffScreen
</h2><span id='topic+KS.prelim'></span>

<h3>Description</h3>

<p>This function does the preliminary data management and fit the model under null hypothesis. The output will be passed to the other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KS.prelim(Y, X=NULL, id=NULL, out_type="C")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KS.prelim_+3A_y">Y</code></td>
<td>

<p>The outcome variable, an n*1 matrix where n is the total number of observations
</p>
</td></tr>
<tr><td><code id="KS.prelim_+3A_x">X</code></td>
<td>

<p>An n*d covariates matrix where d is the total number of covariates.
</p>
</td></tr>
<tr><td><code id="KS.prelim_+3A_id">id</code></td>
<td>

<p>The subject id. This is used to match phenotype with genotype. The default is NULL, where the matched phenotype and genotype matrices are assumed.
</p>
</td></tr>
<tr><td><code id="KS.prelim_+3A_out_type">out_type</code></td>
<td>

<p>Type of outcome variable. Can be either &quot;C&quot; for continuous or &quot;D&quot; for dichotomous. The default is &quot;C&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list that will be passed to function KS.chr().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(KnockoffScreen)

# load example vcf file from package "seqminer"
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])

# simulated outcomes, covariates and inidividual id.
Y&lt;-as.matrix(rnorm(nrow(example.G),0,1))
X&lt;-as.matrix(rnorm(nrow(example.G),0,1))
id&lt;-rownames(example.G)

# fit null model
result.prelim&lt;-KS.prelim(Y,X=X,id=id,out_type="C")

</code></pre>

<hr>
<h2 id='KS.summary'>
Summarize KnockoffScreen results
</h2><span id='topic+KS.summary'></span>

<h3>Description</h3>

<p>Summarize results generated by function KS.VCF.chr(). Calculate q-values for each window/variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KS.summary(result.window,result.single,M)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KS.summary_+3A_result.window">result.window</code></td>
<td>

<p>A result matrix generated by KS.VCF.chr() for all windows. Each row present a tested window. If the genome is partitioned into smaller regions and parellel computing is applied (e.g. each chromosome can be partitioned into 50 contiguous regions), the result matrices should be combined and then processed by KS.summary() jointly for genome-wide FDR control.
</p>
</td></tr>
<tr><td><code id="KS.summary_+3A_result.single">result.single</code></td>
<td>

<p>A result matrix generated by KS.VCF.chr() for individual variants with minor allele frequency above the specified threshold. Each row present a tested variant. If the genome is partitioned into smaller segments and parellel computing is applied (e.g. each chromosome can be partitioned into 50 contiguous segments), the result matrices should be combined and then processed by KS.summary() jointly for genome-wide FDR control.
</p>
</td></tr>
<tr><td><code id="KS.summary_+3A_m">M</code></td>
<td>

<p>Number of knockoffs per variant. Should be same as M used in KS.VCF.chr().
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>result.summary</code></td>
<td>

<p>A matrix summarizing the KnockoffScreen results.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(KnockoffScreen)

# load example vcf file from package "seqminer"
vcf.filename = system.file("vcf/1000g.phase1.20110521.CFH.var.anno.vcf.gz", package = "seqminer")

## this is how the actual genotype matrix from package "seqminer" looks like
example.G &lt;- t(readVCFToMatrixByRange(vcf.filename, "1:196621007-196716634",annoType='')[[1]])

# simulated outcomes, covariates and inidividual id.
Y&lt;-as.matrix(rnorm(nrow(example.G),0,1))
X&lt;-as.matrix(rnorm(nrow(example.G),0,1))
id&lt;-rownames(example.G)

# fit null model
result.prelim&lt;-KS.prelim(Y,X=X,id=id,out_type="C")

# Define the window.bed file
chr&lt;-1
pos.min&lt;-196621007;pos.max&lt;-196716634
window.size=c(2000)

window.bed&lt;-c();
for(size in window.size){
  pos.tag&lt;-seq(pos.min,pos.max,by=size*1/2)
  window.bed&lt;-rbind(window.bed,cbind(chr,pos.tag,pos.tag+size))
}
window.bed&lt;-window.bed[order(as.numeric(window.bed[,2])),]

# scan the vcf file
midout.dir&lt;-NULL # or '/YourProjectDir/MidResults/'
temp.dir&lt;-NULL # or '/YourProjectDir/Temp_out/' #this is a folder to save temporary results
jobtitle&lt;-'YourProjectTitle'

# we set thres.single=0.1,thres.ultrarare=0 for a proof of concept.
# note that the default for real data analysis is thres.single=0.01, thres.ultrarare=25
fit &lt;- KS.chr(result.prelim,vcf.filename,window.bed,M=5,thres.single=0.1,thres.ultrarare=0,
midout.dir=midout.dir,temp.dir=temp.dir,jobtitle=jobtitle)

# summarize the results
result.summary&lt;-KS.summary(fit$result.window,fit$result.single,M=5)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
