<!DOCTYPE html><html lang="en"><head><title>Help for package KnockoffTrio</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {KnockoffTrio}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#causal_loci'><p>Identification of putative causal loci</p></a></li>
<li><a href='#create_knockoff'><p>Create knockoff genotype data</p></a></li>
<li><a href='#KnockoffTrio'><p>Calculate KnockoffTrio's feature statistics</p></a></li>
<li><a href='#KnockoffTrio.example'><p>Example data for KnockoffTrio</p></a></li>
<li><a href='#meta_analysis'><p>Meta-analysis for KnockoffTrio</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>GWAS with Trio and Duo Data using Knockoff Statistics for FDR
Control</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification of putative causal variants in genome-wide association studies with trio and duo families. The package calculates the W feature statistics from KnockoffTrio and p-values from the family-based association test (FBAT) using trio and/or duo data. Compared to previous versions, a significant improvement has been made in Version 1.1.0 to allow the package to be applied not only to trio families but also to duo families. The package implements the methods in the paper: "Yang, Y., Wang, C., Liu, L., Buxbaum, J., He, Z., &amp; Ionita-Laza, I. (2022). KnockoffTrio: A knockoff framework for the identification of putative causal variants in genome-wide association studies with trio design. The American Journal of Human Genetics, 109(10), 1761-1776." </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-12-26 07:44:47 UTC; yiyang</td>
</tr>
<tr>
<td>Author:</td>
<td>Yi Yang <a href="https://orcid.org/0000-0002-1471-4026"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yi Yang &lt;yi.yang@cityu.edu.hk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-23 09:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='causal_loci'>Identification of putative causal loci</h2><span id='topic+causal_loci'></span>

<h3>Description</h3>

<p>Identification of putative causal loci using KnockoffTrio's feature statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>causal_loci(window, M = 10, fdr = 0.1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="causal_loci_+3A_window">window</code></td>
<td>
<p>The result window from function KnockoffTrio. If there are multiple result windows (e.g., when you analyze multiple regions in the genome), please use rbind to combine all the windows before running causal_loci.</p>
</td></tr>
<tr><td><code id="causal_loci_+3A_m">M</code></td>
<td>
<p>A positive integer for the number of knockoffs. The default is 10.</p>
</td></tr>
<tr><td><code id="causal_loci_+3A_fdr">fdr</code></td>
<td>
<p>A real number in a range of (0,1) indicating the target FDR level. The default is 0.1. Use 0.2 for a more lenient FDR control.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains the following elements for claiming significance using knockoff statistics. The result window also contains FBAT p-values and ACAT-combined p-values, which can be used for claiming significance in addition to knockoff statistics. If p-values are used, Bonferroni correction is usually necessary to adjust for multiple testing for controlling the family-wise error rate - see examples below. 
</p>

<dl>
<dt>window</dt><dd><p>A data frame for an updated result window that includes an extra column for KnockoffTrio's Q-values. A locus with a Q-value &lt;= the target FDR level, i.e., window$q&lt;=fdr, is considered as putative causal at the target FDR.</p>
</dd>
<dt>thr.w</dt><dd><p>A positive real number indicating the significance threshold for KnockoffTrio's feature statistics. A locus with a feature statistic &gt;= thr.w, i.e., window$w&gt;=thr.w is considered as putative causal at the target FDR. The loci selected by window$w&gt;=thr.w are equivalent to those by window$q&lt;=fdr. No loci are selected at the target FDR level if thr.w=Inf.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffTrio.example)
knockoff&lt;-create_knockoff(trio.hap=KnockoffTrio.example$trio.hap,
          duo.hap=KnockoffTrio.example$duo.hap, pos=KnockoffTrio.example$pos, M=10)
window&lt;-KnockoffTrio(trio=KnockoffTrio.example$trio, trio.ko=knockoff$trio.ko,
        duo=knockoff$duo, duo.ko=knockoff$duo.ko, pos=KnockoffTrio.example$pos)

#Identification of significant loci using KnockoffTrio's feature statistics (W or Q) 
#at a target FDR
target_fdr&lt;-0.1
result&lt;-causal_loci(window,M=10,fdr=target_fdr)
sig_loci_by_w_index&lt;-which(result$window$w&gt;=result$thr.w)
sig_loci_by_q_index&lt;-which(result$window$q&lt;=target_fdr)

#Identification of significant loci using FBAT p-values with Bonferroni correction
#for controlling the family-wise error rate at 0.05
sig_loci_by_p_fbat_index&lt;-which(window$p.burden&lt;0.05/nrow(window))

#Identification of significant loci using ACAT p-values with Bonferroni correction
#for controlling the family-wise error rate at 0.05
sig_loci_by_p_acat_index&lt;-which(window$p&lt;0.05/nrow(window))
</code></pre>

<hr>
<h2 id='create_knockoff'>Create knockoff genotype data</h2><span id='topic+create_knockoff'></span>

<h3>Description</h3>

<p>Create knockoff genotype data using phased haplotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_knockoff(
  trio.hap = NULL,
  duo.hap = NULL,
  pos,
  M = 10,
  maxcor = 0.7,
  maxbp = 80000,
  phasing.dad = NULL,
  phasing.mom = NULL,
  seed = 100
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_knockoff_+3A_trio.hap">trio.hap</code></td>
<td>
<p>A 6n*p matrix for trio haplotype data, in which n is the number of trios and p is the number of variants. Each trio must consist of father, mother, and offspring (in this order). The haplotypes must be coded as 0 or 1. Missing haplotypes are not allowed.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_duo.hap">duo.hap</code></td>
<td>
<p>A 4m*p matrix for duo haplotype data, in which m is the number of duos and p is the number of variants. Each duo must consist of a single parent and offspring (in this order). The haplotypes must be coded as 0 or 1. Missing haplotypes are not allowed.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of length p for the position of p variants.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_m">M</code></td>
<td>
<p>A positive integer for the number of knockoffs. The default is 10.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_maxcor">maxcor</code></td>
<td>
<p>A real number in a range of [0,1] for hierarchical clustering of neighboring variants used to generate knockoff parents. The default is 0.7.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_maxbp">maxbp</code></td>
<td>
<p>A positive integer for the size of neighboring base pairs used to generate knockoff parents. The default is 80000.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_phasing.dad">phasing.dad</code></td>
<td>
<p>A numeric vector of length n that contains 1 or 2 to indicate which paternal haplotype was transmitted to offspring in each trio. If NULL, the function will calculate the phasing information based on the input trio haplotype matrix.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_phasing.mom">phasing.mom</code></td>
<td>
<p>A numeric vector of length n that contains 1 or 2 to indicate which maternal haplotype was transmitted to offspring in each trio. If NULL, the function will calculate the phasing information based on the input trio haplotype matrix.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_seed">seed</code></td>
<td>
<p>An integer for the random seed used for knockoff generation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains:
</p>

<dl>
<dt>trio.ko</dt><dd><p>A 3n*p*M array for knockoff trio genotype data if trio.hap is provided.</p>
</dd>
<dt>duo.ko</dt><dd><p>A 3m*p*M array for knockoff duo genotype data if duo.hap is provided.</p>
</dd>
<dt>duo</dt><dd><p>A 3m*p matrix for duo genotype data if duo.hap is provided.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffTrio.example)
knockoff&lt;-create_knockoff(trio.hap=KnockoffTrio.example$trio.hap,
          duo.hap=KnockoffTrio.example$duo.hap, pos=KnockoffTrio.example$pos, M=10)
</code></pre>

<hr>
<h2 id='KnockoffTrio'>Calculate KnockoffTrio's feature statistics</h2><span id='topic+KnockoffTrio'></span>

<h3>Description</h3>

<p>Calculate KnockoffTrio's feature statistics and FBAT statistics using original and knockoff genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KnockoffTrio(
  trio,
  trio.ko = NULL,
  duo = NULL,
  duo.ko = NULL,
  pos,
  start = NULL,
  end = NULL,
  size = c(1, 1000, 5000, 10000, 20000, 50000),
  p_value_only = FALSE,
  adjust_for_cov = FALSE,
  y = NULL,
  chr = "1",
  xchr = FALSE,
  sex = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KnockoffTrio_+3A_trio">trio</code></td>
<td>
<p>A 3n*p matrix for the trio genotype data, in which n is the number of trios and p is the number of variants. Each trio must consist of father, mother, and offspring (in this order). The genotypes must be coded as 0, 1, or 2. Missing genotypes are not allowed.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_trio.ko">trio.ko</code></td>
<td>
<p>A 3n*p*M array for the knockoff trio genotype data created by function create_knockoff. M is the number of knockoffs.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_duo">duo</code></td>
<td>
<p>A 3m*p matrix for the duo genotype data created by function create_knockoff, in which m is the number of duos and p is the number of variants. Please do not use the original 2m*p duo genotype matrix.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_duo.ko">duo.ko</code></td>
<td>
<p>A 3m*p*M array for the knockoff duo genotype data created by function create_knockoff. M is the number of knockoffs.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of length p for the position of p variants.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_start">start</code></td>
<td>
<p>An integer for the first position of sliding windows. If NULL, start=min(pos). Only used if you would like to use the same starting position for different cohorts/analyses.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_end">end</code></td>
<td>
<p>An integer for the last position of sliding windows. If NULL, end=max(pos). Only used if you would like to use the same ending position for different cohorts/analyses.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_size">size</code></td>
<td>
<p>A numeric vector for the size(s) of sliding windows when scanning the genome.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_p_value_only">p_value_only</code></td>
<td>
<p>A logical value indicating whether to perform the knockoff analysis. When p_value_only is TRUE, only the ACAT-combined p-values are to be calculated for each window. When p_value_only is FALSE, trio.ko or duo.ko is required and KnockoffTrio's feature statistics are to be calculated for each window in addition to the p-values.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_adjust_for_cov">adjust_for_cov</code></td>
<td>
<p>A logical value indicating whether to adjust for covariates. When adjust_for_cov is TRUE, y is required.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_y">y</code></td>
<td>
<p>A numeric vector of length n for the residual Y-Y_hat. Y_hat is the predicted value from the regression model in which the quantitative trait Y is regressed on the covariates. If Y is dichotomous, you may treat Y as quantitative when applying the regression model.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_chr">chr</code></td>
<td>
<p>A character for the name of the chromosome, e.g., &quot;1&quot;, &quot;2&quot;, ..., &quot;22&quot;, and &quot;X&quot;.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_xchr">xchr</code></td>
<td>
<p>A logical value indicating whether the analysis is for the X chromosome. When xchr is TRUE, the analysis is for the X chromosome and sex is required. When xchr is FALSE, the analysis is for the autosomes. The default if FALSE.</p>
</td></tr>
<tr><td><code id="KnockoffTrio_+3A_sex">sex</code></td>
<td>
<p>A numeric vector of length n for the sex of offspring. 0s indicate females and 1s indicate males. Sex is required when xchr is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame for analysis results from KnockoffTrio and FBAT. The data frame contains the following columns if p_value_only is FALSE:
</p>

<dl>
<dt>chr</dt><dd><p>The chromosome number.</p>
</dd>
<dt>start, end</dt><dd><p>The start and end position of a window.</p>
</dd>
<dt>actual_start, actual_end</dt><dd><p>The position of the first and last variant in a window.</p>
</dd>
<dt>n</dt><dd><p>The number of variants in a window.</p>
</dd>
<dt>dir</dt><dd><p>The direction of effect of the most significant variant in a window.</p>
</dd>
<dt>w</dt><dd><p>The W knockoff feature statistic for a window. Please use function causal_loci to obtain the significance threshold for w at target FDRs.</p>
</dd>
<dt>p</dt><dd><p>The ACAT-combined p-value for a window. If a window contains multiple variants (i.e., n&gt;1), ACAT combines FBAT p-values for each variant and a burden FBAT p-value for all variants in the window. If a window contains only one variant (i.e., n=1), the ACAT-combined p-value is equivalent to the FBAT p-value for this variant.</p>
</dd>
<dt>z</dt><dd><p>The FBAT z-score for a window. If a window contains multiple variants (i.e., n&gt;1), z is the burden FBAT z-score for all variants in the window. If a window contains only one variant (i.e., n=1), z is the FBAT z-score for this variant.</p>
</dd>
<dt>p.burden</dt><dd><p>The FBAT p-value for a window. If a window contains multiple variants (i.e., n&gt;1), p.burden is the burden FBAT p-value for all variants in the window. If a window contains only one variant (i.e., n=1), p.burden is the FBAT p-value for this variant.</p>
</dd>
<dt>kappa, tau</dt><dd><p>The two columns are used by function causal_loci for knockoff inference.</p>
</dd>
<dt>p_1, ..., p_M</dt><dd><p>The ACAT-combined p-values for M knockoffs.</p>
</dd>
<dt>z_1, ..., z_M</dt><dd><p>The FBAT z-scores for M knockoffs.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffTrio.example)
knockoff&lt;-create_knockoff(trio.hap=KnockoffTrio.example$trio.hap,
          duo.hap=KnockoffTrio.example$duo.hap, pos=KnockoffTrio.example$pos, M=10)

#Analysis for both trios and duos
window&lt;-KnockoffTrio(trio=KnockoffTrio.example$trio, trio.ko=knockoff$trio.ko,
        duo=knockoff$duo, duo.ko=knockoff$duo.ko, pos=KnockoffTrio.example$pos)

#Analysis for trios only
window&lt;-KnockoffTrio(trio=KnockoffTrio.example$trio, trio.ko=knockoff$trio.ko,
        duo=NULL, duo.ko=NULL, pos=KnockoffTrio.example$pos)

#Analysis for duos only
window&lt;-KnockoffTrio(trio=NULL, trio.ko=NULL,
        duo=knockoff$duo, duo.ko=knockoff$duo.ko, pos=KnockoffTrio.example$pos)
</code></pre>

<hr>
<h2 id='KnockoffTrio.example'>Example data for KnockoffTrio</h2><span id='topic+KnockoffTrio.example'></span>

<h3>Description</h3>

<p>A toy example of haplotype and genotype data for trios and duos.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KnockoffTrio.example
</code></pre>


<h3>Format</h3>

<p>KnockoffTrio.example contains the following items:
</p>

<dl>
<dt>trio</dt><dd><p>A 9*5 numeric genotype matrix of 3 trios and 5 variants. Each trio contains 3 rows in the order of father, mother and offspring. Each column represents a variant.</p>
</dd>
<dt>trio.hap</dt><dd><p>A 18*5 numeric haplotype matrix of 3 trios and 5 variants. Each trio contains 6 rows in the order of father, mother and offspring. Each column represents a variant.</p>
</dd>
<dt>duo.hap</dt><dd><p>A 12*5 numeric haplotype matrix of 3 duos and 5 variants. Each duo contains 4 rows in the order of a single parent and offspring. Each column represents a variant.</p>
</dd>
<dt>pos</dt><dd><p>A numeric vector of length 5 for the position of 5 variants.</p>
</dd>
</dl>


<hr>
<h2 id='meta_analysis'>Meta-analysis for KnockoffTrio</h2><span id='topic+meta_analysis'></span>

<h3>Description</h3>

<p>Meta-analysis for KnockoffTrio
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta_analysis(window, n = NA, M = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meta_analysis_+3A_window">window</code></td>
<td>
<p>A list of windows for the analysis results from different cohorts/studies.</p>
</td></tr>
<tr><td><code id="meta_analysis_+3A_n">n</code></td>
<td>
<p>A positive integer vector for the number of families in each cohort/study. For weighted meta-analysis, a study's weight is based on the number of families. The default is NA for unweighted meta-analysis.</p>
</td></tr>
<tr><td><code id="meta_analysis_+3A_m">M</code></td>
<td>
<p>A positive integer for the number of knockoffs. The default is 10.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame for the meta-analysis results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffTrio.example)
knockoff&lt;-create_knockoff(trio.hap=KnockoffTrio.example$trio.hap,
          duo.hap=KnockoffTrio.example$duo.hap, pos=KnockoffTrio.example$pos, M=10)
window&lt;-KnockoffTrio(trio=KnockoffTrio.example$trio, trio.ko=knockoff$trio.ko,
        duo=knockoff$duo, duo.ko=knockoff$duo.ko, pos=KnockoffTrio.example$pos)
window.list&lt;-list(window,window)
window.meta&lt;-meta_analysis(window.list,M=10)
result&lt;-causal_loci(window.meta,M=10,fdr=0.1)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
