<!DOCTYPE html><html><head><title>Help for package KnockoffHybrid</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {KnockoffHybrid}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_weight'><p>Calculate weight from population designs</p></a></li>
<li><a href='#causal_loci'><p>Identification of causal loci</p></a></li>
<li><a href='#create_knockoff'><p>Create knockoff genotype data in trio designs</p></a></li>
<li><a href='#KnockoffHybrid'><p>Calculate KnockoffHybrid's feature statistics</p></a></li>
<li><a href='#KnockoffHybrid.example'><p>Example data for KnockoffHybrid</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Hybrid Analysis of Population and Trio Data with Knockoff
Statistics for FDR Control</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Identification of putative causal variants in genome-wide association studies using hybrid analysis of both the trio and population designs. </td>
</tr>
<tr>
<td>Imports:</td>
<td>glmnet, SPAtest</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-05 04:31:56 UTC; yiyang</td>
</tr>
<tr>
<td>Author:</td>
<td>Yi Yang <a href="https://orcid.org/0000-0002-1471-4026"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yi Yang &lt;yi.yang@cityu.edu.hk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-05 16:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_weight'>Calculate weight from population designs</h2><span id='topic+calculate_weight'></span>

<h3>Description</h3>

<p>Calculate weight using population genotype or summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_weight(
  pval = NULL,
  beta = NULL,
  method = "score",
  geno = NULL,
  y = NULL,
  phetype = "D",
  PCs = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_weight_+3A_pval">pval</code></td>
<td>
<p>A numeric vector of length p for p-values. P-values must be between 0 and 1. If not NULL, weight will be calculated as -log10(p-value).</p>
</td></tr>
<tr><td><code id="calculate_weight_+3A_beta">beta</code></td>
<td>
<p>A numeric vector of length p for beta coefficients. If not NULL, weight will be calculated as the absolute value of beta coefficients.</p>
</td></tr>
<tr><td><code id="calculate_weight_+3A_method">method</code></td>
<td>
<p>A character string for the name of the weight estimation method. Must not be NULL if population genotype is used to calculate weight. Weight can be calculated using &quot;score&quot; (i.e., single variant score test) or &quot;lasso&quot; (i.e., least absolute shrinkage and selection operator). The default is &quot;score&quot;.</p>
</td></tr>
<tr><td><code id="calculate_weight_+3A_geno">geno</code></td>
<td>
<p>A n*p matrix for the population genotype data, in which n is the number of subjects and p is the number of variants. The genotypes must be coded as 0, 1, or 2.</p>
</td></tr>
<tr><td><code id="calculate_weight_+3A_y">y</code></td>
<td>
<p>A numeric vector of length n for the phenotype data for the n subjects.</p>
</td></tr>
<tr><td><code id="calculate_weight_+3A_phetype">phetype</code></td>
<td>
<p>A character for the variable type of the phenotype. The type can be &quot;C&quot; (i.e., continuous) or &quot;D&quot; (i.e., dichotomous). The default is &quot;D&quot;.</p>
</td></tr>
<tr><td><code id="calculate_weight_+3A_pcs">PCs</code></td>
<td>
<p>A n*k matrix for the principal components of population structure, in which n is the number of subjects and k is the number of (top) principal components. If not NULL, principal components will be included as covariates when calculating weight from population genotype.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A numeric vector of length p for the weight.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffHybrid.example)
weight&lt;-calculate_weight(geno=KnockoffHybrid.example$dat.pop,y=KnockoffHybrid.example$y.pop)
</code></pre>

<hr>
<h2 id='causal_loci'>Identification of causal loci</h2><span id='topic+causal_loci'></span>

<h3>Description</h3>

<p>Identification of causal loci using KnockoffHybrid's feature statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>causal_loci(window, M = 10, fdr = 0.2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="causal_loci_+3A_window">window</code></td>
<td>
<p>The result window from KnockoffHybrid. If there are multiple windows, please use rbind to combine the windows.</p>
</td></tr>
<tr><td><code id="causal_loci_+3A_m">M</code></td>
<td>
<p>A positive integer for the number of knockoffs. The default is 10.</p>
</td></tr>
<tr><td><code id="causal_loci_+3A_fdr">fdr</code></td>
<td>
<p>A real number in a range of (0,1) indicating the target FDR level. The default is 0.2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that contains:
</p>

<dl>
<dt>window</dt><dd><p>A data frame for an updated window that includes an extra column for KnockoffHybrid's Q-values. A locus with a Q-value &lt;= the target FDR level, i.e., window$q&lt;=fdr, is considered as causal.</p>
</dd>
<dt>thr.w</dt><dd><p>A positive real number indicating the significance threshold for KnockoffTrio's feature statistics. A locus with a feature statistic &gt;= thr.w, i.e., window$w&gt;=thr.w is considered as causal. The loci selected by window$w&gt;=thr.w are equivalent to those by window$q&lt;=fdr. No loci are selected at the target FDR level if thr.w=Inf.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffHybrid.example)
dat.ko&lt;-create_knockoff(KnockoffHybrid.example$dat.hap,KnockoffHybrid.example$pos,M=10)
weight&lt;-calculate_weight(geno=KnockoffHybrid.example$dat.pop,y=KnockoffHybrid.example$y.pop)
window&lt;-KnockoffHybrid(dat=KnockoffHybrid.example$dat,dat.ko=dat.ko,
        pos=KnockoffHybrid.example$pos,weight=weight)
result&lt;-causal_loci(window,M=10,fdr=0.2)
</code></pre>

<hr>
<h2 id='create_knockoff'>Create knockoff genotype data in trio designs</h2><span id='topic+create_knockoff'></span>

<h3>Description</h3>

<p>Create knockoff genotype data using phased haplotype data in trio designs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_knockoff(
  dat.hap,
  pos,
  M = 10,
  maxcor = 0.7,
  maxbp = 90000,
  phasing.dad = NULL,
  phasing.mom = NULL,
  seed = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_knockoff_+3A_dat.hap">dat.hap</code></td>
<td>
<p>A 6n*p matrix for the haplotype data, in which n is the number of trios and p is the number of variants. Each trio must consist of father, mother, and offspring (in this order). The haplotypes must be coded as 0 or 1. Missing haplotypes are not allowed.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of length p for the position of p variants.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_m">M</code></td>
<td>
<p>A positive integer for the number of knockoffs. The default is 10.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_maxcor">maxcor</code></td>
<td>
<p>A real number in a range of [0,1] for the correlation threshold in hierarchical clustering, such that variants from two different clusters do not have a correlation greater than maxcor when constructing knockoff parents. The default is 0.7.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_maxbp">maxbp</code></td>
<td>
<p>A positive integer for the size of neighboring base pairs used to generate knockoff parents. The default is 90000.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_phasing.dad">phasing.dad</code></td>
<td>
<p>A numeric vector of length n that contains 1 or 2 to indicate which paternal haplotype was transmitted to offspring in each trio. If NA, the function will calculate the phasing information based on the input haplotype matrix.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_phasing.mom">phasing.mom</code></td>
<td>
<p>A numeric vector of length n that contains 1 or 2 to indicate which maternal haplotype was transmitted to offspring in each trio. If NA, the function will calculate the phasing information based on the input haplotype matrix.</p>
</td></tr>
<tr><td><code id="create_knockoff_+3A_seed">seed</code></td>
<td>
<p>An integer for the random seed in the knockoff generation. The default is 100.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A 3n*p*M array for the knockoff genotype data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffHybrid.example)
dat.ko&lt;-create_knockoff(KnockoffHybrid.example$dat.hap,KnockoffHybrid.example$pos,M=10)
</code></pre>

<hr>
<h2 id='KnockoffHybrid'>Calculate KnockoffHybrid's feature statistics</h2><span id='topic+KnockoffHybrid'></span>

<h3>Description</h3>

<p>Calculate KnockoffHybrid's feature statistics using original and knockoff genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KnockoffHybrid(
  dat,
  dat.ko = NA,
  pos,
  allele = NA,
  start = NA,
  end = NA,
  size = c(1, 1000, 5000, 10000, 20000, 50000),
  p_value_only = FALSE,
  adjust_for_cov = FALSE,
  y = NA,
  chr = "1",
  sex = NA,
  weight = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KnockoffHybrid_+3A_dat">dat</code></td>
<td>
<p>A 3n*p matrix for the original trio genotype data, in which n is the number of trios and p is the number of variants. Each trio must consist of father, mother, and offspring (in this order). The genotypes must be coded as 0, 1, or 2. Missing genotypes are not allowed.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_dat.ko">dat.ko</code></td>
<td>
<p>A 3n*p*M array for the knockoff trio genotype data created by function create_knockoff. M is the number of knockoffs.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_pos">pos</code></td>
<td>
<p>A numeric vector of length p for the position of p variants.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_allele">allele</code></td>
<td>
<p>A vector of length p for the minor allele at each position. Minor alleles for windows with multiple variants will be shown as &quot;W&quot; in the output.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_start">start</code></td>
<td>
<p>An integer for the first position of sliding windows. If NA, start=min(pos). Only used if you would like to use the same starting position for different cohorts/analyses.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_end">end</code></td>
<td>
<p>An integer for the last position of sliding windows. If NA, end=max(pos). Only used if you would like to use the same ending position for different cohorts/analyses.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_size">size</code></td>
<td>
<p>A numeric vector for the size(s) of sliding windows when scanning the genome</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_p_value_only">p_value_only</code></td>
<td>
<p>A logical value indicating whether to perform the knockoff analysis. When p_value_only is TRUE, only the ACAT-combined p-values are to be calculated for each window. When p_value_only is FALSE, dat.ko is required and KnockoffHybrid's feature statistics are to be calculated for each window in addition to the p-values.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_adjust_for_cov">adjust_for_cov</code></td>
<td>
<p>A logical value indicating whether to adjust for covariates. When adjust_for_cov is TRUE, y is required.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_y">y</code></td>
<td>
<p>A numeric vector of length n for the residual Y-Y_hat. Y_hat is the predicted value from the regression model in which the quantitative trait Y is regressed on the covariates. If Y is dichotomous, you may treat Y as quantitative when applying the regression model.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_chr">chr</code></td>
<td>
<p>A character for the name of the chromosome, e.g., &quot;1&quot;, &quot;2&quot;, ..., &quot;22&quot;, and &quot;X&quot;.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_sex">sex</code></td>
<td>
<p>A numeric vector of length n for the sex of offspring. 0s indicate females and 1s indicate males.</p>
</td></tr>
<tr><td><code id="KnockoffHybrid_+3A_weight">weight</code></td>
<td>
<p>A numeric vector of length p for the weight of p variants. The weight can be obtained via the function &quot;calculate_weight&quot; using population genotype or summary statistics. If NULL, the weight will be calculated based on minor allele frequencies.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame for the hybrid analysis results. Each row contains the p-values and, if p_value_only is FALSE, KnockoffHybrid's feature statistics for a window.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(KnockoffHybrid.example)
dat.ko&lt;-create_knockoff(KnockoffHybrid.example$dat.hap,KnockoffHybrid.example$pos,M=10)
weight&lt;-calculate_weight(geno=KnockoffHybrid.example$dat.pop,y=KnockoffHybrid.example$y.pop)
window&lt;-KnockoffHybrid(dat=KnockoffHybrid.example$dat,dat.ko=dat.ko,
        pos=KnockoffHybrid.example$pos,weight=weight)
</code></pre>

<hr>
<h2 id='KnockoffHybrid.example'>Example data for KnockoffHybrid</h2><span id='topic+KnockoffHybrid.example'></span>

<h3>Description</h3>

<p>A toy example of the haplotype and genotype data for trio designs and the genotype and phenotype data for population designs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KnockoffHybrid.example
</code></pre>


<h3>Format</h3>

<p>KnockoffHybrid.example contains the following items:
</p>

<dl>
<dt>dat</dt><dd><p>A numeric genotype matrix of 3 trios and 5 variants. Each trio contains 3 rows in the order of father, mother and offspring. Each column represents a variant.</p>
</dd>
<dt>dat.hap</dt><dd><p>A numeric haplotype matrix of 3 trios and 5 variants. Each trio contains 6 rows in the order of father, mother and offspring. Each column represents a variant.</p>
</dd>
<dt>dat.pop</dt><dd><p>A numeric genotype matrix of 8 subjects and 5 variants. Each row represents a subject and each column represents a variant.</p>
</dd>
<dt>y.pop</dt><dd><p>A numeric vector of length 8 for the dichotomous phenotypes of 8 subjects.</p>
</dd>
<dt>pos</dt><dd><p>A numeric vector of length 5 for the position of 5 variants.</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
