<!DOCTYPE html><html><head><title>Help for package KOGMWU</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {KOGMWU}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#KOGMWU-package'>
<p>Functional summary and meta-analysis of gene expression data</p></a></li>
<li><a href='#adults.3dHeat.logFoldChange'>
<p>Heat stress response of adult coral</p></a></li>
<li><a href='#corrPlot'>
<p>Plots a pairwise correlation with linear regression line</p></a></li>
<li><a href='#gene2kog'>
<p>KOG class annotations</p></a></li>
<li><a href='#kog.ft'>
<p>One-tailed Fisher's exact test for KOG enrichment.</p></a></li>
<li><a href='#kog.mwu'>
<p>Tests for KOG class enrichment.</p></a></li>
<li><a href='#kog.mwut'>
<p>Mann-Whitney U test for KOG enrichment.</p></a></li>
<li><a href='#larvae.longTerm'>
<p>Long-term heat stress response of coral larvae</p></a></li>
<li><a href='#larvae.shortTerm'>
<p>Short-term heat stress response of coral larvae</p></a></li>
<li><a href='#makeDeltaRanksTable'>
<p>Make a combined delta-ranks table from several kog.mwu() results.</p></a></li>
<li><a href='#panel.cor'>
<p>accessory function for pairs() to display Pearson correlations</p></a></li>
<li><a href='#panel.cor.pval'>
<p>accessory function for pairs() to display pvalue of the Pearson correlation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Functional Summary and Meta-Analysis of Gene Expression Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-02-19</td>
</tr>
<tr>
<td>Author:</td>
<td>Mikhail V. Matz</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mikhail V. Matz &lt;matz@utexas.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Rank-based tests for enrichment of KOG (euKaryotic Orthologous Groups) classes with up- or down-regulated genes based on a continuous measure. The meta-analysis is based on correlation of KOG delta-ranks across datasets (delta-rank is the difference between mean rank of genes belonging to a KOG class and mean rank of all other genes). With binary measure (1 or 0 to indicate significant and non-significant genes), one-tailed Fisher's exact test for over-representation of each KOG class among significant genes will be performed. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10), pheatmap</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-02-19 18:30:06 UTC; c-monstr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-02-19 19:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='KOGMWU-package'>
Functional summary and meta-analysis of gene expression data
</h2><span id='topic+KOGMWU-package'></span><span id='topic+KOGMWU'></span>

<h3>Description</h3>

<p>Rank-based tests for enrichment of KOG (euKaryotic Orthologous Groups) classes with up- or down-regulated genes based on a continuous measure. The meta-analysis is based on correlation of KOG delta-ranks across datasets (delta-rank is the difference between mean rank of genes belonging to a KOG class and mean rank of all other genes). With binary measure (1 or 0 to indicate significant and non-significant genes), one-tailed Fisher's exact test for over-representation of each KOG class among significant genes will be performed. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> KOGMWU</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-02-19</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The most important function is kog.mwu, which performs a series of Mann-Whitney U tests when given two data tables: one, containing measures of interest for each gene (for example, log fold-change), and another, listing the association of each gene with a KOG class. The KOG class annotations for a collection of genes can be obtained using eggNOG-mapper: http://eggnogdb.embl.de/#/app/emapper. To extract KOG annotations understood by this package out of the eggNOG-mapper output, see here: https://github.com/z0on/emapper_to_GOMWU_KOGMWU  
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz
</p>
<p>Maintainer: Mikhail V. Matz &lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Dixon, G. B., Davies, S. W., Aglyamova, G. V., Meyer, E., Bay, L. K. and Matz, M. V. Genomic determinants of coral heat tolerance across latitudes. Science 2015, 348:1460-1462. 
eggNOG-mapper to obtain KOG annotations: http://eggnogdb.embl.de/#/app/emapper
To extract KOG annotations from eggNOG-mapper output: https://github.com/z0on/emapper_to_GOMWU_KOGMWU
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(adults.3dHeat.logFoldChange)
data(larvae.longTerm)
data(larvae.shortTerm)
data(gene2kog)

# Analyzing adult coral response to 3-day heat stress:
alfc.lth=kog.mwu(adults.3dHeat.logFoldChange,gene2kog) 
alfc.lth 

# coral larvae response to 5-day heat stress:
l.lth=kog.mwu(larvae.longTerm,gene2kog)
l.lth

# coral larvae response to 4-hour heat stress 
l.sth=kog.mwu(larvae.shortTerm,gene2kog)
l.sth

# compiling a table of delta-ranks to compare these results:
ktable=makeDeltaRanksTable(list("adults.long"=alfc.lth,"larvae.long"=l.lth,"larvae.short"=l.sth))

# Making a heatmap with hierarchical clustering trees: 
pheatmap(as.matrix(ktable),clustering_distance_cols="correlation") 

# exploring correlations between datasets
pairs(ktable, lower.panel = panel.smooth, upper.panel = panel.cor)
# p-values of these correlations in the upper panel:
pairs(ktable, lower.panel = panel.smooth, upper.panel = panel.cor.pval)

# plotting individual delta-rank correlations:
corrPlot(x="adults.long",y="larvae.long",ktable)
corrPlot(x="larvae.short",y="larvae.long",ktable)

## End(Not run)
</code></pre>

<hr>
<h2 id='adults.3dHeat.logFoldChange'>
Heat stress response of adult coral
</h2><span id='topic+adults.3dHeat.logFoldChange'></span>

<h3>Description</h3>

<p>Acropora millepora (adult) response to three days of heat stress (31.5oC)
log-fold-changes inferred using DESeq package from tag-based RNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("adults.3dHeat.logFoldChange")</code></pre>


<h3>Format</h3>

<p>A data frame with 44363 observations on the following 2 variables.
</p>

<dl>
<dt><code>gene</code></dt><dd><p>gene id, a factor with 44363 levels </p>
</dd>
<dt><code>lfc</code></dt><dd><p>log fold-change, a numeric vector</p>
</dd>
</dl>


<hr>
<h2 id='corrPlot'>
Plots a pairwise correlation with linear regression line
</h2><span id='topic+corrPlot'></span>

<h3>Description</h3>

<p>Plots Pearson's correlation between two columns in a dataframe, identified by column names. Also plots linear regression line and lists the correlation coefficient (r) and cor.test p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corrPlot(x, y, data, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="corrPlot_+3A_x">x</code></td>
<td>

<p>Name of the column to form X axis
</p>
</td></tr>
<tr><td><code id="corrPlot_+3A_y">y</code></td>
<td>

<p>Name of the column to form Y axis
</p>
</td></tr>
<tr><td><code id="corrPlot_+3A_data">data</code></td>
<td>

<p>The dataframe containing the two columns
</p>
</td></tr>
<tr><td><code id="corrPlot_+3A_...">...</code></td>
<td>

<p>Additional options for plot()
</p>
</td></tr>
</table>


<h3>References</h3>

<p>Dixon GB, Davies SW, Aglyamova GA, Meyer E, Bay LK and Matz MV (2015) Genomic determinants of coral heat tolerance across latitudes. 
</p>

<hr>
<h2 id='gene2kog'>
KOG class annotations
</h2><span id='topic+gene2kog'></span>

<h3>Description</h3>

<p>KOG class annotations for Acropora millerpoa transcriptome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gene2kog")</code></pre>


<h3>Format</h3>

<p>A data frame with 16175 observations on the following 2 variables.
</p>

<dl>
<dt><code>V1</code></dt><dd><p>a factor with 16175 levels </p>
</dd>
<dt><code>V2</code></dt><dd><p>a factor with 23 levels </p>
</dd>
</dl>



<h3>Source</h3>

<p>https://dl.dropboxusercontent.com/u/37523721/amillepora_transcriptome_july2014.zip
</p>


<h3>References</h3>

<p>Transcriptome assembly: Moya et al (2012), Mol Ecol 21:2440-2454.
Transcriptome annotation: Dixon GB, Davies SW, Aglyamova GA, Meyer E, Bay LK and Matz MV (2015) Genomic determinants of coral heat tolerance across latitudes. 
</p>

<hr>
<h2 id='kog.ft'>
One-tailed Fisher's exact test for KOG enrichment.
</h2><span id='topic+kog.ft'></span>

<h3>Description</h3>

<p>Accessory function to kog.mwu()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kog.ft(gos)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kog.ft_+3A_gos">gos</code></td>
<td>

<p>A dataframe with three columns, 'seq' (gene id),'term' (KOG class) and 'value' (either 0 or 1, indicating significance).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with three columns: 'term', 'nseqs', 'pval' and 'padj'
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz
</p>


<h3>References</h3>

<p>Dixon GB, Davies SW, Aglyamova GA, Meyer E, Bay LK and Matz MV (2015) Genomic determinants of coral heat tolerance across latitudes. 
</p>

<hr>
<h2 id='kog.mwu'>
Tests for KOG class enrichment.
</h2><span id='topic+kog.mwu'></span>

<h3>Description</h3>

<p>Determines whether some KOG classes are significantly enriched with up- or down-regulated genes (Mann-Whitney U test for continuous measure), or whether some KOG classes are significantly over-represented among &quot;significant&quot; genes (one-tailed Fisher's exact test for binary measure, 0 or 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kog.mwu(data, gene2kog, Alternative = "t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kog.mwu_+3A_data">data</code></td>
<td>

<p>Two-column dataframe: gene id, measure of significance. 
</p>
</td></tr>
<tr><td><code id="kog.mwu_+3A_gene2kog">gene2kog</code></td>
<td>

<p>Two-column dataframe of gene annotations: gene id, KOG class. The gene list can be longer or shorter than the first column in the 'data' item.
</p>
</td></tr>
<tr><td><code id="kog.mwu_+3A_alternative">Alternative</code></td>
<td>

<p>Tailedness of the Mann-Whitney U test: two-tailed (&quot;t&quot;), greater (&quot;g&quot;), or less (&quot;l&quot;)
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The measure can be continuous (such as log fold change), in which case Mann-Whitney U test will be performed, or binary (1 or 0: significant or not), in which case Fisher's exact test will be performed. The KOG class annotations for a collection of genes can be obtained using Weizhng Li's lab KOG BLAST server.
</p>


<h3>Value</h3>

<p>For continuous measure, a dataframe with three columns: 
term : KOG class
nseqs : Number of genes in this class
delta.rank : Difference between the mean rank of genes belonging to this KOG class and all other genes
pval : p-value of the Mann-Whitney U test
padj : p-value adjusted using Benjamini-Hochberg 1995 &quot;fdr&quot; procedure
</p>
<p>For binary measure, the output is similar but does not contain the delta.rank column.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz &lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Dixon GB, Davies SW, Aglyamova GA, Meyer E, Bay LK and Matz MV (2015) Genomic determinants of coral heat tolerance across latitudes. 
Weizhong Li's KOG BLAST server: http://weizhong-lab.ucsd.edu/metagenomic-analysis/server/kog/ 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(adults.3dHeat.logFoldChange)
data(larvae.longTerm)
data(larvae.shortTerm)
data(gene2kog)

# Analyzing adult coral response to 3-day heat stress:
alfc.lth=kog.mwu(adults.3dHeat.logFoldChange,gene2kog) 
alfc.lth 

# coral larvae response to 5-day heat stress:
l.lth=kog.mwu(larvae.longTerm,gene2kog)
l.lth

# coral larvae response to 4-hour heat stress 
l.sth=kog.mwu(larvae.shortTerm,gene2kog)
l.sth

# compiling a table of delta-ranks to compare these results:
ktable=makeDeltaRanksTable(list("adults.long"=alfc.lth,"larvae.long"=l.lth,"larvae.short"=l.sth))

# Making a heatmap with hierarchical clustering trees: 
pheatmap(as.matrix(ktable),clustering_distance_cols="correlation") 

# exploring correlations between datasets
pairs(ktable, lower.panel = panel.smooth, upper.panel = panel.cor)
# p-values of these correlations in the upper panel:
pairs(ktable, lower.panel = panel.smooth, upper.panel = panel.cor.pval)

# plotting individual delta-rank correlations:
corrPlot(x="adults.long",y="larvae.long",ktable)
corrPlot(x="larvae.short",y="larvae.long",ktable)

## End(Not run)
</code></pre>

<hr>
<h2 id='kog.mwut'>
Mann-Whitney U test for KOG enrichment.
</h2><span id='topic+kog.mwut'></span>

<h3>Description</h3>

<p>Accessory function to kog.mwu()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kog.mwut(gos, Alternative = "t")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kog.mwut_+3A_gos">gos</code></td>
<td>

<p>A dataframe with three columns, 'seq' (gene id),'term' (KOG class) and 'value' (continuous measure, such as log fold-change).
</p>
</td></tr>
<tr><td><code id="kog.mwut_+3A_alternative">Alternative</code></td>
<td>

<p>Tailedness of the MWU test: two-tailed (&quot;t&quot;), greater-than (&quot;g&quot;), or less-than (&quot;l&quot;)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with three columns: 'term', 'nseqs', 'delta.rank', 'pval' and 'padj'
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz
</p>


<h3>References</h3>

<p>Dixon GB, Davies SW, Aglyamova GA, Meyer E, Bay LK and Matz MV (2015) Genomic determinants of coral heat tolerance across latitudes. 
</p>

<hr>
<h2 id='larvae.longTerm'>
Long-term heat stress response of coral larvae
</h2><span id='topic+larvae.longTerm'></span>

<h3>Description</h3>

<p>Acropora millepora (larvae) response to five days of heat stress (31.5oC)
log-fold-changes inferred using DESeq package from tag-based RNA-seq data 
from Meyer et al Mol Ecol 2011,17:3599-3616
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("larvae.longTerm")</code></pre>


<h3>Format</h3>

<p>A data frame with 31844 observations on the following 2 variables.
</p>

<dl>
<dt><code>gene</code></dt><dd><p>gene id, a factor with 31844 levels </p>
</dd>
<dt><code>lfc</code></dt><dd><p>log fold-change, a numeric vector</p>
</dd>
</dl>


<hr>
<h2 id='larvae.shortTerm'>
Short-term heat stress response of coral larvae
</h2><span id='topic+larvae.shortTerm'></span>

<h3>Description</h3>

<p>Acropora millepora (larvae) response to four hours of heat stress (31.5oC)
log-fold-changes inferred using DESeq package from tag-based RNA-seq data 
from Meyer et al Mol Ecol 2011,17:3599-3616.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("larvae.shortTerm")</code></pre>


<h3>Format</h3>

<p>A data frame with 32307 observations on the following 2 variables.
</p>

<dl>
<dt><code>gene</code></dt><dd><p>gene id, a factor with 32307 levels</p>
</dd>
<dt><code>lfc</code></dt><dd><p>log fold-change, a numeric vector</p>
</dd>
</dl>


<hr>
<h2 id='makeDeltaRanksTable'>
Make a combined delta-ranks table from several kog.mwu() results. 
</h2><span id='topic+makeDeltaRanksTable'></span>

<h3>Description</h3>

<p>Extracts delta ranks from several kog.mwu() result tables and combines them into a single dataframe for heat map plotting and correlation analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>makeDeltaRanksTable(ll)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="makeDeltaRanksTable_+3A_ll">ll</code></td>
<td>

<p>A list of dataframes output by kog.mwu() function.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe of delta-ranks (rows - KOG classes, columns - delta-ranks in different datasets).
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz
</p>


<h3>References</h3>

<p>Dixon GB, Davies SW, Aglyamova GA, Meyer E, Bay LK and Matz MV (2015) Genomic determinants of coral heat tolerance across latitudes. 
</p>

<hr>
<h2 id='panel.cor'>
accessory function for pairs() to display Pearson correlations
</h2><span id='topic+panel.cor'></span>

<h3>Description</h3>

<p>works as upper.panel or lower.panel argument of pairs() (package graphics).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel.cor(x, y, digits=2, cex.cor) 
  </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panel.cor_+3A_x">x</code></td>
<td>

<p>x element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor_+3A_y">y</code></td>
<td>

<p>y element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor_+3A_digits">digits</code></td>
<td>

<p>number of non-zero digits to leave at the end
</p>
</td></tr>
<tr><td><code id="panel.cor_+3A_cex.cor">cex.cor</code></td>
<td>

<p>scaling factor for displayed text
</p>
</td></tr>
</table>


<h3>References</h3>

<p>cannibalized from an example in ?pairs (package graphics)
</p>

<hr>
<h2 id='panel.cor.pval'>
accessory function for pairs() to display pvalue of the Pearson correlation
</h2><span id='topic+panel.cor.pval'></span>

<h3>Description</h3>

<p>works as upper.panel or lower.panel argument of pairs() (package graphics). Displays pvalues better than 0.1.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel.cor.pval(x, y, digits = 2, cex.cor, p.cut=0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panel.cor.pval_+3A_x">x</code></td>
<td>

<p>x element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_y">y</code></td>
<td>

<p>y element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_digits">digits</code></td>
<td>

<p>number of non-zero digits to leave at the end
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_cex.cor">cex.cor</code></td>
<td>

<p>scaling factor for displayed text
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_p.cut">p.cut</code></td>
<td>

<p>p-value cutoff
</p>
</td></tr>
</table>


<h3>References</h3>

<p>cannibalized from an example in ?pairs (package graphics)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
