<!DOCTYPE html><html lang="en"><head><title>Help for package kpcaIG</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kpcaIG}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#kernelpca'><p>Kernel Principal Components Analysis</p></a></li>
<li><a href='#kpca_igrad'><p>KPCA-IG: Variables Interpretability</p>
in Kernel PCA</a></li>
<li><a href='#plot_kpca2D'><p>2D Kernel PCA Plot with Variables Representation</p></a></li>
<li><a href='#plot_kpca3D'><p>3D Kernel PCA Plot with Variables Representation</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Variables Interpretability with Kernel PCA</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mitja Briscik &lt;mitja.briscik@math.univ-toulouse.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The kernelized version of principal component analysis (KPCA) has proven to be a valid nonlinear alternative for tackling the nonlinearity of biological sample spaces. However, it poses new challenges in terms of the interpretability of the original variables. 'kpcaIG' aims to provide a tool to select the most relevant variables based on the kernel PCA representation of the data as in Briscik et al. (2023) &lt;<a href="https://doi.org/10.1186%2Fs12859-023-05404-y">doi:10.1186/s12859-023-05404-y</a>&gt;. It also includes functions for 2D and 3D visualization of the original variables (as arrows) into the kernel principal components axes, highlighting the contribution of the most important ones.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, rgl, kernlab, ggplot2, stats, progress, viridis,
WallomicsData</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-17 15:29:03 UTC; mbriscik</td>
</tr>
<tr>
<td>Author:</td>
<td>Mitja Briscik [aut, cre],
  Mohamed Heimida [aut],
  Sébastien Déjean [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-21 08:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='kernelpca'>Kernel Principal Components Analysis</h2><span id='topic+kernelpca'></span>

<h3>Description</h3>

<p>Kernel Principal Components Analysis,  a nonlinear version of principal component analysis obtrained through the so-called kernel trick.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kernelpca(data, kernel = "vanilladot", kpar = list(), features = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kernelpca_+3A_data">data</code></td>
<td>
<p>The data matrix organized by rows. Users should scale the data appropriately before applying this function, if relevant.</p>
</td></tr>
<tr><td><code id="kernelpca_+3A_kernel">kernel</code></td>
<td>
<p>The kernel function used for the analysis. It can be chosen from the following strings:
</p>

<ul>
<li><p><code>'rbfdot'</code>: Radial Basis kernel function &quot;Gaussian&quot;
</p>
</li>
<li><p><code>'polydot'</code>: Polynomial kernel function
</p>
</li>
<li><p><code>'vanilladot'</code>: Linear kernel function
</p>
</li>
<li><p><code>'tanhdot'</code>: Hyperbolic tangent kernel function
</p>
</li></ul>

</td></tr>
<tr><td><code id="kernelpca_+3A_kpar">kpar</code></td>
<td>
<p>The list of hyper-parameters (kernel parameters) used with the kernel function. The valid parameters for each kernel type are as follows:
</p>

<ul>
<li><p><code>sigma</code>: inverse kernel width for the Radial Basis kernel function <code>"rbfdot"</code>.
</p>
</li>
<li><p><code>degree</code>, <code>scale</code>, <code>offset</code> for the Polynomial kernel function <code>"polydot"</code>.
</p>
</li>
<li><p><code>scale</code>, <code>offset</code> for the Hyperbolic tangent kernel function <code>"tanhdot"</code>.
</p>
</li></ul>

</td></tr>
<tr><td><code id="kernelpca_+3A_features">features</code></td>
<td>
<p>The number of features (kernel principal components) to use for the analysis. Default: 0 , (all)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>kernelpca</code> returns an S4 object of formal class kpca as in library(kernlab) containing the principal component vectors along with the corresponding eigenvalues.
</p>
<table role = "presentation">
<tr><td><code>pcv</code></td>
<td>
<p>pcv	a matrix containing the principal component vectors (column wise)</p>
</td></tr>
<tr><td><code>eig</code></td>
<td>
<p>The corresponding eigenvalues</p>
</td></tr>
<tr><td><code>rotated</code></td>
<td>
<p>The original data projected (rotated) on the principal components</p>
</td></tr>
<tr><td><code>xmatrix</code></td>
<td>
<p>The original data matrix</p>
</td></tr>
</table>


<h3>References</h3>

<p>Scholkopf B., Smola A. and Muller K.R. (1998) Nonlinear component analysis as a kernel eigenvalue problem. Neural Computation, 10, 1299-1319.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example
library(WallomicsData)
library(kpcaIG)
library(ggplot2)
library(kernlab)
Transcriptomics_Stems_s &lt;- scale(Transcriptomics_Stems)

kpca_tan &lt;-  kernelpca(as.matrix(Transcriptomics_Stems_s),
                          kernel = "tanhdot",
                          kpar = list(scale = 0.0001, offset = 0.01))


ggplot(data = data.frame(rotated(kpca_tan), Genetic_Cluster), 
       aes(x = X1, y = X2, shape = Genetic_Cluster)) +
  geom_point(size = 2, aes(color = Genetic_Cluster)) +
  xlab("1st kernel PC") +
  ylab("2nd kernel PC") +
  labs(color = "Genetic_Cluster", shape = "Genetic_Cluster") +
  theme_minimal()


ggplot(data = data.frame(rotated(kpca_tan), Ecotype), 
       aes(x = X1, y = X2, shape =  Ecotype)) +
  geom_point(size = 2, aes(color =  Ecotype)) +
  xlab("1st kernel PC") +
  ylab("2nd kernel PC") +
  labs(color = " Ecotype", shape = " Ecotype") +
  theme_minimal()


</code></pre>

<hr>
<h2 id='kpca_igrad'>KPCA-IG: Variables Interpretability
in Kernel PCA</h2><span id='topic+kpca_igrad'></span>

<h3>Description</h3>

<p>KPCA-IG, kernel pca interpetable gradient. It is the fuction that gives the feature ranking,
from the most to the least relevant variable. The ranking is
obtained through the kernel’s partial derivatives computation. A score, which corresponds to the score mean among the sample points, is assigned to each input feature.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kpca_igrad(kpca_result, dim, mean_type = "arithmetic", trim_ratio = 0.1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kpca_igrad_+3A_kpca_result">kpca_result</code></td>
<td>
<p>The result of the previously obtained kernel PCA analysis.</p>
</td></tr>
<tr><td><code id="kpca_igrad_+3A_dim">dim</code></td>
<td>
<p>Number of kernel principal component to use for the computation of the scores. It should be less or equal to the number of component of the kPCA.</p>
</td></tr>
<tr><td><code id="kpca_igrad_+3A_mean_type">mean_type</code></td>
<td>
<p>Type of mean. Possible values are &quot;arithmetic&quot;, &quot;geometric&quot;, &quot;harmonic&quot;, &quot;median&quot;, or &quot;trimmed&quot;. Default = &quot;arithmetic&quot;</p>
</td></tr>
<tr><td><code id="kpca_igrad_+3A_trim_ratio">trim_ratio</code></td>
<td>
<p>For mean_type == &quot;trimmed&quot;, it is the fraction (0 to 0.5) of scores to be trimmed from each end before the mean is computed for a more robust to outliers arithmetic mean computation. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing the sorted variables and their scores sorted in decreasing order.
</p>


<h3>References</h3>

<p>Briscik, M., Dillies, MA. &amp; Déjean, S. Improvement of variables interpretability in kernel PCA. BMC Bioinformatics 24, 282 (2023). DOI: <a href="https://doi.org/10.1186/s12859-023-05404-y">doi:10.1186/s12859-023-05404-y</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(WallomicsData)
library(kpcaIG)

Transcriptomics_Stems_s &lt;- scale(Transcriptomics_Stems)

kpca_tan &lt;-  kernelpca(as.matrix(Transcriptomics_Stems_s),
                          kernel = "tanhdot",
                          kpar = list(scale = 0.0001, offset = 0.01))

#Compute the most relevant genes based on the first two components of kpca_tan

kpca_ig_tan &lt;- kpca_igrad(kpca_tan, dim = c(1,2))
head(kpca_ig_tan)

</code></pre>

<hr>
<h2 id='plot_kpca2D'>2D Kernel PCA Plot with Variables Representation</h2><span id='topic+plot_kpca2D'></span>

<h3>Description</h3>

<p><code>plot_kpca2D</code> allows to visualize an original variable of interest in the specified principal components. The variable is displayed as an arrow, showing its relevance in the relative position of each sample point in the kernel component space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_kpca2D(kpca_result, target_variable, groups = NULL, scale = 100, 
  components = c(1, 2), arrow_col = "#D3D3D3", 
  main_title = "Kernel principal component analysis")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_kpca2D_+3A_kpca_result">kpca_result</code></td>
<td>
<p>The result of the previously obtained kernel PCA analysis</p>
</td></tr>
<tr><td><code id="plot_kpca2D_+3A_target_variable">target_variable</code></td>
<td>
<p>A string indicating the name of the variable of interest to visualize as arrows on the kernel PCA plot.</p>
</td></tr>
<tr><td><code id="plot_kpca2D_+3A_groups">groups</code></td>
<td>
<p>A vector indicating the grouping of data points, if applicable. Default: NULL</p>
</td></tr>
<tr><td><code id="plot_kpca2D_+3A_scale">scale</code></td>
<td>
<p>Coefficient to adjust the lengths of the arrows. Default: 100</p>
</td></tr>
<tr><td><code id="plot_kpca2D_+3A_components">components</code></td>
<td>
<p>A numeric vector of length 2 specifying the indices of the components to plot. Default: c(1, 2)</p>
</td></tr>
<tr><td><code id="plot_kpca2D_+3A_arrow_col">arrow_col</code></td>
<td>
<p>Colour of the arrows. Default: '#D3D3D3'</p>
</td></tr>
<tr><td><code id="plot_kpca2D_+3A_main_title">main_title</code></td>
<td>
<p>Graph title. Default: &quot;Kernel principal component analysis&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Provides a 2D plot of class ggplot that displays the sample points projected onto the specified kernel principal component axes, with the variables of interest represented as arrows.
</p>


<h3>References</h3>

<p>Briscik, M., Dillies, MA. &amp; Déjean, S. Improvement of variables interpretability in kernel PCA. BMC Bioinformatics 24, 282 (2023). DOI: <a href="https://doi.org/10.1186/s12859-023-05404-y">doi:10.1186/s12859-023-05404-y</a>.
Variables representation as in Reverter, F., Vegas, E. &amp; Oller, J.M. Kernel-PCA data integration with enhanced interpretability. BMC Syst Biol 8 (Suppl 2), S6 (2014). DOI: <a href="doi:10.1186/1752-0509-8-S2-S6">doi:10.1186/1752-0509-8-S2-S6</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(WallomicsData)
library(kpcaIG)

Transcriptomics_Stems_s &lt;- scale(Transcriptomics_Stems)

kpca_tan &lt;-  kernelpca(as.matrix(Transcriptomics_Stems_s),
                          kernel = "tanhdot",
                          kpar = list(scale = 0.0001, offset = 0.01))

# Compute the most relevant genes based on the first two components of kpca_tan

kpca_ig_tan &lt;- kpca_igrad(kpca_tan, dim = c(1,2))
head(kpca_ig_tan)

# Visualize the most relevant variable (gene) according to kpca_igrad, "AT4G12060".

plot_kpca2D(kpca_tan, "AT4G12060", groups = Ecotype, scale = 1000, components = c(1, 2))

# Visualize using the second and third components

plot_kpca2D(kpca_tan, "AT4G12060", groups = Ecotype, scale = 1000, components = c(2, 3))

#The selected gene shows upper expression in the samples with genotype type Col.

</code></pre>

<hr>
<h2 id='plot_kpca3D'>3D Kernel PCA Plot with Variables Representation</h2><span id='topic+plot_kpca3D'></span>

<h3>Description</h3>

<p><code>plot_kpca3D</code> allows to visualize an original variable of interest in the first three principal components. The variable is displayed as an arrow, showing its relevance in the relative position of each sample point in the kernel component space.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_kpca3D(kpca_result, target_variable, groups, scale=1, 
type = "s", size = 3/4, arrow_col = "#999999", 
angles = 12, main = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_kpca3D_+3A_kpca_result">kpca_result</code></td>
<td>
<p>The result of the previously obtained kernel PCA analysis.</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_target_variable">target_variable</code></td>
<td>
<p>A string indicating the name of the variable to visualize as arrows on the kernel PCA plot.</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_groups">groups</code></td>
<td>
<p>A vector indicating the grouping of data points, if applicable. Default: NULL</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_scale">scale</code></td>
<td>
<p>Coefficient to adjust the lengths of the arrows. Default 1</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_type">type</code></td>
<td>
<p>A character indicating the type of point for the observations.
Supported types are: 'p' for points, 's' for spheres. Default: 's'</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_size">size</code></td>
<td>
<p>The size of the plotted points. Default: 3/4</p>
</td></tr> 
<tr><td><code id="plot_kpca3D_+3A_arrow_col">arrow_col</code></td>
<td>
<p>Colour of the arrows. Default: '#999999</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_angles">angles</code></td>
<td>
<p>Number of barbs of the arrows. Default: 12</p>
</td></tr>
<tr><td><code id="plot_kpca3D_+3A_main">main</code></td>
<td>
<p>Graph title. Default: NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Provides an interactive 3D plot that displays the sample points projected onto the first three kernel principal component axes, with the variables of interest represented as arrows.
</p>


<h3>References</h3>

<p>Briscik, M., Dillies, MA. &amp; Déjean, S. Improvement of variables interpretability in kernel PCA. BMC Bioinformatics 24, 282 (2023). DOI: <a href="https://doi.org/10.1186/s12859-023-05404-y">doi:10.1186/s12859-023-05404-y</a>.
Variables representation as in Reverter, F., Vegas, E. &amp; Oller, J.M. Kernel-PCA data integration with enhanced interpretability. BMC Syst Biol 8 (Suppl 2), S6 (2014). DOI: <a href="doi:10.1186/1752-0509-8-S2-S6">doi:10.1186/1752-0509-8-S2-S6</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(WallomicsData)
library(kpcaIG)

Transcriptomics_Stems_s &lt;- scale(Transcriptomics_Stems)

kpca_tan &lt;-  kernelpca(as.matrix(Transcriptomics_Stems_s),
                          kernel = "tanhdot",
                          kpar = list(scale = 0.0001, offset = 0.01))

#Compute the most relevant genes based on the first two components of kpca_tan

kpca_ig_tan &lt;- kpca_igrad(kpca_tan, dim = c(1,2))
head(kpca_ig_tan)

#Visualize the most relevant variable (gene) according to kpca_igrad, "AT4G12060". 

plot_kpca3D(kpca_tan, "AT4G12060", groups = Ecotype, scale = 1000)

#The selected gene shows upper expression in the samples with genotype type Col.

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
