<!DOCTYPE html><html><head><title>Help for package kpmt</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kpmt}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#genefeat'><p>Transcript features of the protein coding genes of the human genome</p></a></li>
<li><a href='#GO0007186'><p>The genes pertaining to GO term GO:0007186 (G-protein coupled receptor signaling pathway)</p></a></li>
<li><a href='#kpmt'><p>Known Population Median Test</p></a></li>
<li><a href='#logStirling'><p>Log Stirling</p></a></li>
<li><a href='#minMedian'><p>Minimal median</p></a></li>
<li><a href='#validate_accuracy'><p>Validate computational accuracy of Known Population Median Test algorithm</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-07-28</td>
</tr>
<tr>
<td>Title:</td>
<td>Known Population Median Test</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions that implement the known population median test.</td>
</tr>
<tr>
<td>Author:</td>
<td>Matthew M Parks &lt;map2085@med.cornell.edu&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matthew M Parks &lt;map2085@med.cornell.edu&gt;</td>
</tr>
<tr>
<td>Imports:</td>
<td>matrixStats, stats, methods, utils</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-07-31 17:04:17 UTC; map2085</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-01 04:51:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='genefeat'>Transcript features of the protein coding genes of the human genome</h2><span id='topic+genefeat'></span>

<h3>Description</h3>

<p>Various quantitative features of the transcripts of the set of all 19,962 protein coding genes
in the annotated human genome, according to Ensembl 84.  Features include length of the Coding Sequence (CDS),
lengths of the 5' and 3' UTRs, and the relative usage for each amino acid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genefeat
</code></pre>


<h3>Format</h3>

<p>matrix.</p>


<h3>Source</h3>

<p><a href="http://useast.ensembl.org/index.html">Ensembl</a>
</p>


<h3>References</h3>

<p>Yates, et. al. (2016) Nucleic Acids Res. 2016 44 Database issue:D710-6. &quot;Ensembl 2016&quot;.
(<a href="dx.doi.org/10.1093/nar/gkv1157">doi</a>)
</p>

<hr>
<h2 id='GO0007186'>The genes pertaining to GO term GO:0007186 (G-protein coupled receptor signaling pathway)</h2><span id='topic+GO0007186'></span>

<h3>Description</h3>

<p>The names of all 868 protein coding genes annotated as members of Gene Ontology (GO) term GO:0007186 (G-protein coupled receptor signaling pathway),
according to the Gene Ontology consortium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GO0007186
</code></pre>


<h3>Format</h3>

<p>character vector.</p>


<h3>Source</h3>

<p><a href="http://geneontology.org/">Gene Ontology</a>
</p>


<h3>References</h3>

<p>Ashburner et al. Gene ontology: tool for the unification of biology (2000) Nat Genet 25(1):25-9.
(doi: <a href="http://doi.org/10.1038/75556">10.1038/75556</a>)
The Gene Ontology Consortium. Gene Ontology Consortium: going forward. (2015) Nucl Acids Res 43 Database issue
(<a href="https://doi.org/10.1093/nar/gku1179">doi</a>)
</p>

<hr>
<h2 id='kpmt'>Known Population Median Test</h2><span id='topic+kpmt'></span>

<h3>Description</h3>

<p>Performs the known population median test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kpmt(pop, obs = NULL, med = NULL, size = NULL, tail = "two-sided",
  verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kpmt_+3A_pop">pop</code></td>
<td>
<p>[data frame, matrix, vector] numeric values for the whole population.
If a data frame or matrix is given, it should have format:
</p>

<ul>
<li><p>rownames  = population member names (e.g. gene names)
</p>
</li>
<li><p>colnames  = features to test (e.g. relative codon usage, UTR length, MFE, etc.)
</p>
</li></ul>

<p>For a data frame or matrix, the test will be performed on each column, separately.
If a named vector is given, it should have format:
</p>

<ul>
<li><p>names = population member names.
</p>
</li>
<li><p>values = numeric values of the feature.
</p>
</li></ul>
</td></tr>
<tr><td><code id="kpmt_+3A_obs">obs</code></td>
<td>
<p>[character vector or named list of character vectors]
a character vector of population member names, or a named list of character vectors of population member names.
</p>

<ol>
<li><p>  If obs is a list, then each list element name should correspond to a feature name of pop.
</p>
</li>
<li><p>  If obs is a vector, then it is considered to be the same sample for each population feature.
</p>
</li>
<li><p>  if size = NULL, then obs is considered to be a sample of population member names
</p>
</li>
<li><p>  if size is non-NULL, then obs is considered to be the observed
median values per column of pop. (length obs must be equal to number of features in pop)
</p>
</li></ol>
</td></tr>
<tr><td><code id="kpmt_+3A_med">med</code></td>
<td>
<p>[number or vector] pre-computed minimal medians of pop.</p>
</td></tr>
<tr><td><code id="kpmt_+3A_size">size</code></td>
<td>
<p>[integer]  size of the set which generated the observed median.  If obs is a sample, i.e. contains
population member names, then size must be NULL.</p>
</td></tr>
<tr><td><code id="kpmt_+3A_tail">tail</code></td>
<td>
<p>[&quot;two-sided&quot;, &quot;lower&quot;, &quot;upper&quot;]  if NULL, then the minimum of lower and upper will be reported.</p>
</td></tr>
<tr><td><code id="kpmt_+3A_verbose">verbose</code></td>
<td>
<p>display extra messages for tracking execution.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with columns:
</p>
 
<ul>
<li><p>&quot;name&quot;  			a column from pop
</p>
</li>
<li><p>&quot;median.sample&quot;		min median of the sample
</p>
</li>
<li><p>&quot;median.all&quot;		min median of the whole population
</p>
</li>
<li><p>&quot;median.background&quot;	min median of the non-sampled members.
</p>
</li>
<li><p>&quot;logp&quot;			log of p.value if sample median is different from all
</p>
</li>
<li><p>&quot;p.value&quot;		p.value if sample median is different from all
</p>
</li>
<li><p>&quot;FDR&quot;			only if &gt; 30 features, i.e columns of pop
</p>
</li></ul>

<p>each row is a different population feature, i.e. column of pop.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(genefeat)
data(GO0007186)
res  &lt;-  kpmt( pop = genefeat , obs = GO0007186 )
</code></pre>

<hr>
<h2 id='logStirling'>Log Stirling</h2><span id='topic+logStirling'></span>

<h3>Description</h3>

<p>Computes the log of the Stirling approximation of n!.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logStirling(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logStirling_+3A_n">n</code></td>
<td>
<p>integer or vector of integers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Stirling approximation of log(n!).  If n &lt;= 14, then computes log(n!) directly, i.e. no Stirling approximation.
</p>

<hr>
<h2 id='minMedian'>Minimal median</h2><span id='topic+minMedian'></span>

<h3>Description</h3>

<p>Computes the minimal median of a vector or matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>minMedian(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="minMedian_+3A_x">x</code></td>
<td>
<p>a vector or matrix of real numbers.  if a matrix, then the minimal median will be computed for each column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the minimal median of x.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>minMedian(1:6) # returns 3

</code></pre>

<hr>
<h2 id='validate_accuracy'>Validate computational accuracy of Known Population Median Test algorithm</h2><span id='topic+validate_accuracy'></span>

<h3>Description</h3>

<p>The implementation of the exact analytical solution of the Known Population Median Test
involves approximations using the Stirling series and is therefore suspect for computational error.
This function creates a population and empirically computes p-values via resampling.  It then
compares these empirical p-values to those calculated by the Known Population Median Test, and returns
the error in log(p).  WARNING!  Takes a long time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>validate_accuracy(N = 50, n = 10, nrep = 1e+08)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="validate_accuracy_+3A_n">N</code></td>
<td>
<p>population size</p>
</td></tr>
<tr><td><code id="validate_accuracy_+3A_n">n</code></td>
<td>
<p>sample size</p>
</td></tr>
<tr><td><code id="validate_accuracy_+3A_nrep">nrep</code></td>
<td>
<p>number of resampling samples for empirically estimating p-values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with log(p) error and other information.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
