<!DOCTYPE html><html><head><title>Help for package krm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {krm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#aa.prop.list'>
<p>Amino Acid Properties</p></a></li>
<li><a href='#calcPairwiseIdentity'>
<p>Functions Related to Sequence Alignment</p></a></li>
<li><a href='#chi.norm'>
<p>A Transformation of Chi-squared Random Variable</p></a></li>
<li><a href='#cloud9'>
<p>9-Component Mixture Dirichlet Prior for Protein Sequences</p></a></li>
<li><a href='#dmdirichlet'>
<p>Functions related to mixture Dirichlet distribution</p></a></li>
<li><a href='#getSeqKernel'>
<p>Protein Sequence Kernels</p></a></li>
<li><a href='#hmmMargLlik'>
<p>Functions related to profile HMM</p></a></li>
<li><a href='#krm package'><p>Kernel-based Regression Models</p></a></li>
<li><a href='#krm.most'>
<p>Kernel-based Regression Model Maximum of adjusted Score Test</p></a></li>
<li><a href='#krm.score.test'>
<p>Adjusted Score Test</p></a></li>
<li><a href='#readFastaFile'>
<p>Read a Fasta Sequence File</p></a></li>
<li><a href='#sim.liu.2008'>
<p>Simulate sDataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>LazyData:</td>
<td>yes</td>
</tr>
<tr>
<td>Version:</td>
<td>2022.10-17</td>
</tr>
<tr>
<td>Title:</td>
<td>Kernel Based Regression Models</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0), kyotil</td>
</tr>
<tr>
<td>Suggests:</td>
<td>RUnit, MASS</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements several methods for testing the variance component parameter in regression models that contain kernel-based random effects, including a maximum of adjusted scores test. Several kernels are supported, including a profile hidden Markov model mutual information kernel for protein sequence. This package is described in Fong et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fbiostatistics%2Fkxu056">doi:10.1093/biostatistics/kxu056</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-17 17:48:50 UTC; Youyi</td>
</tr>
<tr>
<td>Author:</td>
<td>Youyi Fong [cre],
  Saheli Datta [aut],
  Krisztian Sebestyen [aut],
  Steve Park [ctb],
  Dave Geyer [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Youyi Fong &lt;youyifong@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-18 07:40:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='aa.prop.list'>
Amino Acid Properties
</h2><span id='topic+aa.prop.list'></span>

<h3>Description</h3>

<p>Amino Acid Properties
</p>


<h3>Format</h3>

<p>A data frame with 20 observations on the following 13 variables.
</p>

<dl>
<dt><code>Symbol</code></dt><dd><p>a chacracter vector with 20 values: A through Y</p>
</dd>
<dt><code>AA_Name</code></dt><dd><p>a chacracter vector with 20 values: Alanine through Tyrosine</p>
</dd>
<dt><code>AA_Symbol</code></dt><dd><p>a chacracter vector with 20 values: Ala through Tyr</p>
</dd>
<dt><code>Surface_Area_Chothia</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Residue_Volume_Zamayatin</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Bulkiness_Jones</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Polarity_Jones</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Refractivity_Jones</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hydrophobicity_Engleman</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hydrophobicity_Prabha</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hydrophilicity_Hopp</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>Hydrophilicity_Levitt</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>RelMutability_Jones</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>


<hr>
<h2 id='calcPairwiseIdentity'>
Functions Related to Sequence Alignment
</h2><span id='topic+calcPairwiseIdentity'></span><span id='topic+alignment2count'></span><span id='topic+alignment2trancount'></span><span id='topic+removeGap'></span>

<h3>Description</h3>

<p>Functions related to sequence alignment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcPairwiseIdentity(alignment, dissimilarity, removeGap)
alignment2count (alignment, level=20, weight=rep(1,nrow(alignment)))
alignment2trancount (alignment, weight=rep(1,nrow(alignment)))
removeGap (seq) 


</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcPairwiseIdentity_+3A_alignment">alignment</code></td>
<td>
<p>matrix of arabic representation of sequences (1 based)</p>
</td></tr>
<tr><td><code id="calcPairwiseIdentity_+3A_dissimilarity">dissimilarity</code></td>
<td>
<p>Boolean.</p>
</td></tr>
<tr><td><code id="calcPairwiseIdentity_+3A_removegap">removeGap</code></td>
<td>
<p>Boolean</p>
</td></tr>
<tr><td><code id="calcPairwiseIdentity_+3A_level">level</code></td>
<td>
<p>integer. Size of alphabet</p>
</td></tr>
<tr><td><code id="calcPairwiseIdentity_+3A_weight">weight</code></td>
<td>
<p>numeric vector. Weights given to each sequence</p>
</td></tr>
<tr><td><code id="calcPairwiseIdentity_+3A_seq">seq</code></td>
<td>
<p>string. A string of amino acids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>alignment2count return T by 20 matrix, where T is the number of column in the alignment.
alignment2trancount return a T by 4 matrix, each row is the count of MM, MD, DM, DD for each position.
</p>

<hr>
<h2 id='chi.norm'>
A Transformation of Chi-squared Random Variable
</h2><span id='topic+chi.norm'></span>

<h3>Description</h3>

<p>A transformation of Chi-squared random variable to make it normal like.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chi.norm(q, v)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chi.norm_+3A_q">q</code></td>
<td>

<p>numeric. A random variable following chi-squared distribution
</p>
</td></tr>
<tr><td><code id="chi.norm_+3A_v">v</code></td>
<td>

<p>numeric. A random variable following normal distribution
</p>
</td></tr>
</table>

<hr>
<h2 id='cloud9'>
9-Component Mixture Dirichlet Prior for Protein Sequences
</h2><span id='topic+cloud9'></span>

<h3>Description</h3>

<p>9-Component Mixture Dirichlet Prior for Protein Sequences
</p>


<h3>Format</h3>

<p>List of 2. The alpha element is a 9 by 20 matrix, where each row represents one Dirichlet distribution of 20 dimensions. 
The mix.coef element contains the mixing probability, a vector 0f 9 numbers that add up to 1.
</p>

<hr>
<h2 id='dmdirichlet'>
Functions related to mixture Dirichlet distribution
</h2><span id='topic+dmdirichlet'></span><span id='topic+ddirichlet'></span><span id='topic+rdirichlet'></span><span id='topic+rmdirichlet'></span><span id='topic+modifyDirichlet'></span><span id='topic+logIntegrateMixDirichlet'></span><span id='topic+logIntegrateDirichlet'></span>

<h3>Description</h3>

<p>Functions related to mixture Dirichlet distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmdirichlet(x, mAlpha, mixtureCoef)
ddirichlet (x, alpha)
rdirichlet (n, alpha) 
rmdirichlet (mAlpha, mixtureCoef) 
modifyDirichlet (prior, y) 
logIntegrateMixDirichlet(y, prior, tau=1) 
logIntegrateDirichlet (y, alpha) 


</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dmdirichlet_+3A_x">x</code></td>
<td>
<p>A vector containing a single deviate or matrix containing one random deviate per row.</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_malpha">mAlpha</code></td>
<td>
<p>matrix. Each row is a parameter of Dirichlet</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_alpha">alpha</code></td>
<td>
<p>numeric vector. Parameter for a Dirichlet distribution</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_mixturecoef">mixtureCoef</code></td>
<td>
<p>numeric vector</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_n">n</code></td>
<td>
<p>integer</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_prior">prior</code></td>
<td>
<p>list of two components: alpha and mix.coef</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_y">y</code></td>
<td>
<p>numeric vector of counts</p>
</td></tr>
<tr><td><code id="dmdirichlet_+3A_tau">tau</code></td>
<td>
<p>numeric</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ddirichlet andn rdirichlet are identically copied from MCMCpack
</p>

<hr>
<h2 id='getSeqKernel'>
Protein Sequence Kernels
</h2><span id='topic+getSeqKernel'></span>

<h3>Description</h3>

<p>Get mutual information and other kernels for protein sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getSeqKernel (sequences, kern.type=c("mm", "prop", "mi"), tau, call.C=TRUE
    , seq.start=NULL, seq.end=NULL) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getSeqKernel_+3A_sequences">sequences</code></td>
<td>

<p>String or list. If string, the name of a fasta file containing aligned sequences. If list, a list of strings, each string is a protein sequence. If list, call.C will be set to FALSE internally because C/C++ function needs sequence file name as input
</p>
</td></tr>
<tr><td><code id="getSeqKernel_+3A_kern.type">kern.type</code></td>
<td>

<p>string. Type of kernel. mm: match-mismatch, prop: physicochemical properties, mi: mutual information.
</p>
</td></tr>
<tr><td><code id="getSeqKernel_+3A_tau">tau</code></td>
<td>

<p>Numeric. It is the same as rho^-2.
</p>
</td></tr>
<tr><td><code id="getSeqKernel_+3A_call.c">call.C</code></td>
<td>

<p>Boolean. If TRUE, do a .C call. If FALSE, the implementation is in R. The .C call is 50 times faster.
</p>
</td></tr>
<tr><td><code id="getSeqKernel_+3A_seq.start">seq.start</code></td>
<td>

<p>integer. Start position of subsequence to be used in computing kernel.
</p>
</td></tr>
<tr><td><code id="getSeqKernel_+3A_seq.end">seq.end</code></td>
<td>

<p>integer. End position of subsequence to be used in computing kernel.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>call.C option is to allow comparison of R and C implementation. The two should give the same results and C implementation is 50 times faster. 
</p>
<p>when kern.type is mi and call.C is TRUE and when running on linux, this function will print messages like &quot;read ...&quot;. This message is generated from U::openRead
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
fileName=paste(system.file(package="krm")[1],'/misc/SETpfamseed_aligned_for_testing.fasta',
     sep="")
K=getSeqKernel (fileName, kern.type="mi", tau=1, call.C=TRUE)
K


</code></pre>

<hr>
<h2 id='hmmMargLlik'>
Functions related to profile HMM</h2><span id='topic+hmmMargLlik'></span><span id='topic+readPriorFromFile'></span>

<h3>Description</h3>

<p>Functions related to profile HMM</p>


<h3>Usage</h3>

<pre><code class='language-R'>hmmMargLlik(dat, aaPrior, tau)
readPriorFromFile(priorFileName) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hmmMargLlik_+3A_dat">dat</code></td>
<td>

<p>a matrix representation of a multiple sequence alignment, each row is a sequence, each column is a position
</p>
</td></tr>
<tr><td><code id="hmmMargLlik_+3A_aaprior">aaPrior</code></td>
<td>

<p>a list of two elements, &quot;alpha&quot;    &quot;mix.coef&quot;, representing mixture Dirichlet prior
</p>
</td></tr>
<tr><td><code id="hmmMargLlik_+3A_tau">tau</code></td>
<td>

<p>numeric
</p>
</td></tr>
<tr><td><code id="hmmMargLlik_+3A_priorfilename">priorFileName</code></td>
<td>

<p>string
</p>
</td></tr>
</table>

<hr>
<h2 id='krm+20package'>Kernel-based Regression Models</h2><span id='topic+krm'></span>

<h3>Description</h3>

<p>Implements tests for kernel-based regression model. The main function is krm.most(). Both Euclidean and protein sequence covariates can be used to form kernels.
</p>

<hr>
<h2 id='krm.most'>
Kernel-based Regression Model Maximum of adjusted Score Test
</h2><span id='topic+krm.most'></span>

<h3>Description</h3>

<p>Performs maximum of adjusted score test for kernel-based regression models. Both Euclidean and protein sequence covariates can be used to form kernels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>krm.most (formula, data, regression.type=c("logistic","linear"), 
    kern.type=c("rbf","mi","mm","prop"), n.rho=10, range.rho=0.99, 
    n.mc=2000, 
    seq.file.name=NULL, formula.kern=NULL, seq.start=NULL, seq.end=NULL,
    inference.method=c("parametric.bootstrap", "perturbation", "Davies"),
    verbose=FALSE) 

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="krm.most_+3A_formula">formula</code></td>
<td>

<p>a formula object describing the null model
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_data">data</code></td>
<td>

<p>data frame
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_regression.type">regression.type</code></td>
<td>

<p>logistic regression or linear regression
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_kern.type">kern.type</code></td>
<td>

<p>rbf: radial basis function kernel, a kernel type for Euclidean covariates. 
The other three kernels are for protein sequence covariates (Fong et al. 2014). mm: match-mismatch, prop: physicochemical properties, mi: mutual information, 
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_n.rho">n.rho</code></td>
<td>

<p>integer. Number of rhos to maximize over
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_range.rho">range.rho</code></td>
<td>

<p>numeric. A number between 0 and 1. It controls the range of rhos to use to compute kernel
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_seq.file.name">seq.file.name</code></td>
<td>

<p>There are two ways to provide protein sequence information. One is to supply a sequence file named 'seq.file.name', which contains sequences in fasta format.
Two is to supply a formula through formula.kern, and the variable name should be part of data.
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_formula.kern">formula.kern</code></td>
<td>

<p>The formula for the covariates used to form the kernel. It may specify Euclidean covariates or a string covariate that contains protein sequences.
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_seq.start">seq.start</code></td>
<td>

<p>integer. Start position of subsequence to be used in computing kernel. Only supported when the sequence is specified through formula.kern.
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_seq.end">seq.end</code></td>
<td>

<p>integer. End position of subsequence to be used in computing kernel. Only supported when the sequence is specified through formula.kern.
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_n.mc">n.mc</code></td>
<td>

<p>integer. Number of bootstrap samples used to compute p-values.
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_inference.method">inference.method</code></td>
<td>

<p>parametric.bootstrap implements methods from Fong et al. (2014). 
perturbation uses methods from Wu et a. (2013)
Davies uses the upper bound method from Davies (1987) and Liu et al. (2008). 
</p>
</td></tr>
<tr><td><code id="krm.most_+3A_verbose">verbose</code></td>
<td>

<p>boolean
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class krm
</p>
<table>
<tr><td><code>p.values</code></td>
<td>
<p>If inference.method==&quot;Davies&quot;, a single p-value. If inference.method==&quot;perturbation&quot; or &quot;parametric.bootstrap&quot;, 
a vector of four p-values, named chiI, chiII, normI, normII. For perturbation, chiII and normII are NA.
chiI/chiII p-values are based on chi-squared approximation and normI/normII are based on normal approximations.
chiI/normI p-values are based on plugin estimator of  mean and variance of score statistic, chiII/normII are based on modified estimator of mean and variance of score statistic. 
chiII or normII are more powerful than chiI and normI. For more details, see Fong et al. (2014)</p>
</td></tr>
</table>


<h3>References</h3>

<p>Fong, Y. and Datta, S. and Georgiev, I. and Kwong, P. and Tomaras, G. (2014) Kernel-based
Logistic Regression Model for Protein Sequence Without Vectorialization, Biostatistics.
</p>
<p>Wu et al. (2013) Kernel Machine SNP-Set Testing Under Multiple Candidate Kernels, Genetic epidemiology.
</p>
<p>Davies, R. (1987) Hypothesis testing when a nuisance parameter is present only under the
alternative, Biometrika, 74, 33-43.
</p>
<p>Liu, D., Ghosh, D. and Lin, X. (2008) Estimation and testing for the effect of a genetic
pathway on a disease outcome using logistic kernel machine regression via logistic mixed
models, BMC Bioinformatics, 9, 292.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# in addition to the examples listed here, there are more examples 
# under folder R/library/krm/unitTests

## Not run: 
# the examples are not run during package build because it takes a little too long to run

# an Euclidean kernel example from Liu et al. (2008)
data=sim.liu.2008 (n=100, a=.1, seed=1) 
test = krm.most(y~x, data, formula.kern=~z.1+z.2+z.3+z.4+z.5, kern.type="rbf")


# a protein sequence kernel example
dat.file.name=paste(system.file(package="krm")[1],'/misc/y1.txt', sep="") 
seq.file.name=paste(system.file(package="krm")[1],'/misc/sim1.fasta', sep="")
dat=read.table(dat.file.name); names(dat)="y"
test = krm.most (y~1, dat, seq.file.name=seq.file.name, kern.type="mi")



## End(Not run) 


</code></pre>

<hr>
<h2 id='krm.score.test'>
Adjusted Score Test
</h2><span id='topic+krm.score.test'></span>

<h3>Description</h3>

<p>Adjusted score test for kernel-based regression models. This function is typically not used directly, but is called within krm.most().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>krm.score.test(formula, data, K, regression.type=c("logistic","linear"), verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="krm.score.test_+3A_formula">formula</code></td>
<td>

<p>a formula object. Model under null.
</p>
</td></tr>
<tr><td><code id="krm.score.test_+3A_data">data</code></td>
<td>

<p>a data frame
</p>
</td></tr>
<tr><td><code id="krm.score.test_+3A_k">K</code></td>
<td>

<p>a n by n kernel/correlation matrix
</p>
</td></tr>
<tr><td><code id="krm.score.test_+3A_regression.type">regression.type</code></td>
<td>

<p>a string
</p>
</td></tr>
<tr><td><code id="krm.score.test_+3A_verbose">verbose</code></td>
<td>

<p>Boolean
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
dat=sim.liu.2008(n=100, a=0, seed=1)
z=as.matrix(subset(dat, select=c(z.1,z.2,z.3,z.4,z.5)))
rho=1
K=kyotil::getK(z,kernel="rbf",para=rho^-2)
krm.score.test (y~x, dat, K, regression.type="logistic") 


</code></pre>

<hr>
<h2 id='readFastaFile'>
Read a Fasta Sequence File
</h2><span id='topic+readFastaFile'></span><span id='topic+writeFastaFile'></span><span id='topic+aa2arabic'></span><span id='topic+string2arabic'></span><span id='topic+fastaFile2arabicFile'></span><span id='topic+selexFile2arabicFile'></span><span id='topic+stringList2arabicFile'></span><span id='topic+arabic2arabicFile'></span><span id='topic+readSelexFile'></span><span id='topic+readSelexAsMatrix'></span><span id='topic+arabic2fastaFile'></span><span id='topic+readArabicFile'></span><span id='topic+readBlockFile'></span>

<h3>Description</h3>

<p>Read a Fasta Sequence File
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readFastaFile(fileName, sep = " ")
writeFastaFile (seqList, fileName)
aa2arabic (seq1) 
string2arabic (seqList) 
fastaFile2arabicFile (fastaFile, arabicFile, removeGapMajor=FALSE)
selexFile2arabicFile (selexFile, arabicFile, removeGapMajor=FALSE)
stringList2arabicFile (seqList, arabicFile, removeGapMajor=FALSE)
arabic2arabicFile (alignment, arabicFile) 
readSelexFile (fileName)
readSelexAsMatrix (fileName) 
arabic2fastaFile (alignment, fileName)
readArabicFile (fileName) 
readBlockFile (fileName) 


</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="readFastaFile_+3A_filename">fileName</code></td>
<td>
<p>string</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_fastafile">fastaFile</code></td>
<td>
<p>string</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_arabicfile">arabicFile</code></td>
<td>
<p>string</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_selexfile">selexFile</code></td>
<td>
<p>string</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_sep">sep</code></td>
<td>
<p>string</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_seq1">seq1</code></td>
<td>
<p>string. A string of amino acids</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_seqlist">seqList</code></td>
<td>
<p>list of string.</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_removegapmajor">removeGapMajor</code></td>
<td>
<p>Boolean</p>
</td></tr>
<tr><td><code id="readFastaFile_+3A_alignment">alignment</code></td>
<td>
<p>matrix of arabic representation of sequences (1 based)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>string2arabic returns a matrix of arabic numbers representing aa.
readSelexFile return a list of strings.
readArabicFile return a matrix of n by p alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(RUnit)
fileName=paste(system.file(package="krm")[1],'/misc/SETpfamseed_aligned_for_testing.fasta', sep="")
seqs = readFastaFile (fileName, sep=" ")
checkEquals(length(seqs),11)

</code></pre>

<hr>
<h2 id='sim.liu.2008'>
Simulate sDataset
</h2><span id='topic+sim.liu.2008'></span><span id='topic+sim.liu.2007'></span>

<h3>Description</h3>

<p>Per Liu et al (2008) and Liu et al (2007)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.liu.2008(n, a, seed = NULL)
sim.liu.2007(n, a, seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sim.liu.2008_+3A_n">n</code></td>
<td>

<p>sample size
</p>
</td></tr>
<tr><td><code id="sim.liu.2008_+3A_a">a</code></td>
<td>

<p>numeric. If a is 0, then the data is used to study size, otherwise power
</p>
</td></tr>
<tr><td><code id="sim.liu.2008_+3A_seed">seed</code></td>
<td>

<p>optional random number generator seed
</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
