<!DOCTYPE html><html><head><title>Help for package kronos</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {kronos}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bigdata'><p>Snippet of example data to demonstrate the functionality of 'kronos' between and among three different groups</p></a></li>
<li><a href='#bigmeta'><p>Descriptional metadata for the 'bigdata' object, for the purpose of demonstration.</p></a></li>
<li><a href='#build_kronos_formula'><p>Update 'kronos' formula in light of sine and cosine components</p></a></li>
<li><a href='#delistKronos_groupwise'><p>Get Results from list of KronosOut Objects</p></a></li>
<li><a href='#delistKronos_pairwise'><p>Get Results from list of <code>KronosOut</code> Objects</p></a></li>
<li><a href='#fit_cosinor_model'><p>Fit cosinor model</p></a></li>
<li><a href='#fit_groupwise_model'><p>Fit cosinor model</p></a></li>
<li><a href='#fw_kronos'><p>Fit a cosinor model and extract relevant parameters on a feature table.</p></a></li>
<li><a href='#get_cos_sine'><p>Get sine and cosine components</p></a></li>
<li><a href='#get_vars'><p>Figure out what variable represents time. Called by main 'kronos' function.</p></a></li>
<li><a href='#getKronos'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_fit'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_groupwise'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_input'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_pairwise'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_pairwise_p'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_params'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#getKronos_trace'><p>Get Results from <code>KronosOut</code> Object</p></a></li>
<li><a href='#gg_kronos_acrogram'><p>A plotting method for acrophase circleplots using 'ggplot2'.</p></a></li>
<li><a href='#gg_kronos_circle'><p>A plotting method for circadian plots using 'ggplot2'.</p></a></li>
<li><a href='#gg_kronos_sinusoid'><p>A plotting method for circadian plots using 'ggplot2'.</p></a></li>
<li><a href='#groupdata'><p>Snippet of example data to demonstrate the functionality of 'kronos' between and among three different groups</p></a></li>
<li><a href='#kronos'><p>Fit a cosinor model and extract relevant parameters</p></a></li>
<li><a href='#kronos_anova'><p>Extract p-value from full fit</p></a></li>
<li><a href='#kronos_predict'><p>Give tracing information for plotting purposes</p></a></li>
<li><a href='#kronosListToTable'><p>Wrangle results from list of <code>KronosOut</code> Objects to publication ready table.</p></a></li>
<li><a href='#kronosOut-class'><p>An S4 class to contain all output parameters and plotting information from the 'kronos' main pipeline</p></a></li>
<li><a href='#onevariable'><p>Snippet of example data to demonstrate the functionality of 'kronos' in the most simple scenario.</p></a></li>
<li><a href='#pairwise_cosinor_model'><p>Fit pairwise cosinor models as some sort of TukeyHSD.</p></a></li>
<li><a href='#show,kronosOut-method'><p>Show method for <code>kronosOut</code> object.</p></a></li>
<li><a href='#twowaydata'><p>Snippet of example data to demonstrate the functionality of 'kronos' in the two-factor design scenario.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Microbiome Oriented Circadian Rhythm Analysis Toolkit</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/thomazbastiaanssen/kronos">https://github.com/thomazbastiaanssen/kronos</a></td>
</tr>
<tr>
<td>Description:</td>
<td>The goal of 'kronos' is to provide an easy-to-use framework to analyse circadian or otherwise rhythmic data using the familiar R linear modelling syntax, while taking care of the trigonometry under the hood. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, methods, ggplot2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-25 21:26:38 UTC; thomaz</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomaz Bastiaanssen [aut, cre],
  Sarah-Jane Leigh [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomaz Bastiaanssen &lt;thomazbastiaanssen@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-26 06:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bigdata'>Snippet of example data to demonstrate the functionality of 'kronos' between and among three different groups</h2><span id='topic+bigdata'></span>

<h3>Description</h3>

<p>Snippet of example data to demonstrate the functionality of 'kronos' between and among three different groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bigdata
</code></pre>


<h3>Format</h3>

<p>A long format data.frame object with three 113 columns, features, and 94 rows, samples.
</p>


<h3>Source</h3>

<p>Dummy data modified from untargeted hippocampal metabolomics over the day to demonstrate rhythmicity between multiple groups in a large dataset.
</p>

<hr>
<h2 id='bigmeta'>Descriptional metadata for the 'bigdata' object, for the purpose of demonstration.</h2><span id='topic+bigmeta'></span>

<h3>Description</h3>

<p>Descriptional metadata for the 'bigdata' object, for the purpose of demonstration.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bigmeta
</code></pre>


<h3>Format</h3>

<p>A long format data.frame object with three 113 columns and 94 rows.
</p>


<h3>Source</h3>

<p>Dummy metadata modified from untargeted hippocampal metabolomics over the day
</p>

<hr>
<h2 id='build_kronos_formula'>Update 'kronos' formula in light of sine and cosine components</h2><span id='topic+build_kronos_formula'></span>

<h3>Description</h3>

<p>Update 'kronos' formula.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_kronos_formula(formula, time, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_kronos_formula_+3A_formula">formula</code></td>
<td>
<p>A formula. Use the <code>time</code> function to designate which variable represents time.</p>
</td></tr>
<tr><td><code id="build_kronos_formula_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="build_kronos_formula_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large data sets.</p>
</td></tr>
</table>

<hr>
<h2 id='delistKronos_groupwise'>Get Results from list of KronosOut Objects</h2><span id='topic+delistKronos_groupwise'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a list of <code>kronosOut</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delistKronos_groupwise(kronos_list, padjust = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delistKronos_groupwise_+3A_kronos_list">kronos_list</code></td>
<td>
<p>a list of preferrably named kronosOut objects.</p>
</td></tr>
<tr><td><code id="delistKronos_groupwise_+3A_padjust">padjust</code></td>
<td>
<p>a boolean. Toggles FDR using Benjamini Hochbergs procedure.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table with circadian output stats per group per feature.
</p>

<hr>
<h2 id='delistKronos_pairwise'>Get Results from list of <code>KronosOut</code> Objects</h2><span id='topic+delistKronos_pairwise'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a list of <code>kronosOut</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delistKronos_pairwise(kronos_list, padjust = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delistKronos_pairwise_+3A_kronos_list">kronos_list</code></td>
<td>
<p>a list of preferrably named <code>kronosOut</code> objects.</p>
</td></tr>
<tr><td><code id="delistKronos_pairwise_+3A_padjust">padjust</code></td>
<td>
<p>a boolean. Toggles FDR using Benjamini Hochbergs procedure.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ANOVA-like adjusted p-values for how each factor interacts with time.
</p>

<hr>
<h2 id='fit_cosinor_model'>Fit cosinor model</h2><span id='topic+fit_cosinor_model'></span>

<h3>Description</h3>

<p>Fit cosinor model for totality of data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_cosinor_model(
  formula,
  data,
  time = NULL,
  verbose = verbose,
  for_pw = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_cosinor_model_+3A_formula">formula</code></td>
<td>
<p>A formula. Use the <code>time</code> function to designate which variable represents time.</p>
</td></tr>
<tr><td><code id="fit_cosinor_model_+3A_data">data</code></td>
<td>
<p>Input data</p>
</td></tr>
<tr><td><code id="fit_cosinor_model_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="fit_cosinor_model_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large datasets.</p>
</td></tr>
<tr><td><code id="fit_cosinor_model_+3A_for_pw">for_pw</code></td>
<td>
<p>A boolean. Toggles whether to perform pairwise ANOVAs as a TukeyHSD-like post-hoc.</p>
</td></tr>
</table>

<hr>
<h2 id='fit_groupwise_model'>Fit cosinor model</h2><span id='topic+fit_groupwise_model'></span>

<h3>Description</h3>

<p>Fit cosinor model for one aspect of data. Called by main 'kronos' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_groupwise_model(data, group, time, period, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_groupwise_model_+3A_data">data</code></td>
<td>
<p>input data</p>
</td></tr>
<tr><td><code id="fit_groupwise_model_+3A_group">group</code></td>
<td>
<p>A character string. Signifies which group will be assessed.</p>
</td></tr>
<tr><td><code id="fit_groupwise_model_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="fit_groupwise_model_+3A_period">period</code></td>
<td>
<p>A numeric. The length of a period, in the same format as the <code>time</code> parameter.</p>
</td></tr>
<tr><td><code id="fit_groupwise_model_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large data sets.</p>
</td></tr>
</table>

<hr>
<h2 id='fw_kronos'>Fit a cosinor model and extract relevant parameters on a feature table.</h2><span id='topic+fw_kronos'></span>

<h3>Description</h3>

<p>This wrapper applies kronos(), the main workhorse function in the 'kronos' package. It manages the individual functionalities of 'kronos', including rhythmicity analysis and differential rhythmicity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fw_kronos(
  x,
  formula,
  metadata,
  time = NULL,
  period = 24,
  verbose = FALSE,
  pairwise = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fw_kronos_+3A_x">x</code></td>
<td>
<p>Input data. A table with rows being features and columns being samples</p>
</td></tr>
<tr><td><code id="fw_kronos_+3A_formula">formula</code></td>
<td>
<p>A formula. Use the <code>time</code> function to designate which variable represents time. Leave the left-hand side of the formula empty as it will be sequentially replaced by every feature in the table.</p>
</td></tr>
<tr><td><code id="fw_kronos_+3A_metadata">metadata</code></td>
<td>
<p>A metadata table, with rows being samples and columns being metadata entries</p>
</td></tr>
<tr><td><code id="fw_kronos_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="fw_kronos_+3A_period">period</code></td>
<td>
<p>A numeric. The length of a period, in the same format as the <code>time</code> parameter.</p>
</td></tr>
<tr><td><code id="fw_kronos_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large datasets.</p>
</td></tr>
<tr><td><code id="fw_kronos_+3A_pairwise">pairwise</code></td>
<td>
<p>A boolean. Toggles whether to perform pairwise ANOVAs as a TukeyHSD-like post-hoc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of <code>kronosOut</code> S4 objects containing coefficients and all operations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)
</code></pre>

<hr>
<h2 id='get_cos_sine'>Get sine and cosine components</h2><span id='topic+get_cos_sine'></span>

<h3>Description</h3>

<p>Based on 'cosinor' and 'limorhyde' packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cos_sine(data, period, colnamePrefix = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cos_sine_+3A_data">data</code></td>
<td>
<p>input data</p>
</td></tr>
<tr><td><code id="get_cos_sine_+3A_period">period</code></td>
<td>
<p>A numeric. The length of a period, in the same format as the <code>time</code> parameter.</p>
</td></tr>
<tr><td><code id="get_cos_sine_+3A_colnameprefix">colnamePrefix</code></td>
<td>
<p>A character string. Typically the name of the &quot;Time&quot; variable.</p>
</td></tr>
</table>

<hr>
<h2 id='get_vars'>Figure out what variable represents time. Called by main 'kronos' function.</h2><span id='topic+get_vars'></span>

<h3>Description</h3>

<p>Extracts time from the formula and from the time argument. Also handles inconsistencies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_vars(formula, time, data, verbose = verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_vars_+3A_formula">formula</code></td>
<td>
<p>A formula. Use the <code>time</code> function to designate which variable represents time.</p>
</td></tr>
<tr><td><code id="get_vars_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="get_vars_+3A_data">data</code></td>
<td>
<p>input data</p>
</td></tr>
<tr><td><code id="get_vars_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large data sets.</p>
</td></tr>
</table>

<hr>
<h2 id='getKronos'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos(kronosOut, target)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
<tr><td><code id="getKronos_+3A_target">target</code></td>
<td>
<p>the specific entry of the <code>kronosOut</code> object to be extracted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.frame</code> of results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='getKronos_fit'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_fit'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_fit(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_fit_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The model fit used.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)
</code></pre>

<hr>
<h2 id='getKronos_groupwise'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_groupwise'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_groupwise(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_groupwise_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Rhythmicity parameters per group.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='getKronos_input'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_input'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_input(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_input_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The data used as input for the model.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='getKronos_pairwise'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_pairwise'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_pairwise(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_pairwise_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Pairwise comparisons between groups.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='getKronos_pairwise_p'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_pairwise_p'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_pairwise_p(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_pairwise_p_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ANOVA-like adjusted p-values for how each factor interacts with time.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='getKronos_params'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_params'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_params(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_params_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The names and values of additional circadian model parameters, mostly for plotting purposes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='getKronos_trace'>Get Results from <code>KronosOut</code> Object</h2><span id='topic+getKronos_trace'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a <code>kronosOut</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getKronos_trace(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getKronos_trace_+3A_kronosout">kronosOut</code></td>
<td>
<p>a <code>kronosOut</code> output object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The traces per group for plotting.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='gg_kronos_acrogram'>A plotting method for acrophase circleplots using 'ggplot2'.</h2><span id='topic+gg_kronos_acrogram'></span>

<h3>Description</h3>

<p>Wrapper around 'ggplot2' to make circadian circleplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_kronos_acrogram(kronosOutList)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_kronos_acrogram_+3A_kronosoutlist">kronosOutList</code></td>
<td>
<p>A list of KronosOut output objects from the main <code>kronos</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'ggplot2' compatible object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in Kronos library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)
</code></pre>

<hr>
<h2 id='gg_kronos_circle'>A plotting method for circadian plots using 'ggplot2'.</h2><span id='topic+gg_kronos_circle'></span>

<h3>Description</h3>

<p>Wrapper around 'ggplot2' to make circadian circleplots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_kronos_circle(kronosOut)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_kronos_circle_+3A_kronosout">kronosOut</code></td>
<td>
<p>an output object from the main <code>kronos</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'ggplot2' compatible object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in Kronos library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)
</code></pre>

<hr>
<h2 id='gg_kronos_sinusoid'>A plotting method for circadian plots using 'ggplot2'.</h2><span id='topic+gg_kronos_sinusoid'></span>

<h3>Description</h3>

<p>Wrapper around 'ggplot2' to make figures with a sinusoid trace.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_kronos_sinusoid(kronosOut, fill = "unique_group")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_kronos_sinusoid_+3A_kronosout">kronosOut</code></td>
<td>
<p>an output object from the main <code>kronos</code> function.</p>
</td></tr>
<tr><td><code id="gg_kronos_sinusoid_+3A_fill">fill</code></td>
<td>
<p>The name of the variable that should be used to mark different groups. In the case of a single group, leave empty.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 'ggplot2' compatible object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in Kronos library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='groupdata'>Snippet of example data to demonstrate the functionality of 'kronos' between and among three different groups</h2><span id='topic+groupdata'></span>

<h3>Description</h3>

<p>Snippet of example data to demonstrate the functionality of 'kronos' between and among three different groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>groupdata
</code></pre>


<h3>Format</h3>

<p>A long format data.frame object with three columns, and 94 rows, samples.
</p>


<h3>Source</h3>

<p>Dummy data modified from PCR analysis of mouse ileum over the day to demonstrate rhythmicity between multiple groups.
</p>

<hr>
<h2 id='kronos'>Fit a cosinor model and extract relevant parameters</h2><span id='topic+kronos'></span>

<h3>Description</h3>

<p>This is the main workhorse function in the 'kronos' package. It manages the individual functionalities of 'kronos', including rhythmicity analysis and differential rhythmicity.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kronos(
  formula,
  data,
  time = NULL,
  period = 24,
  verbose = TRUE,
  pairwise = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kronos_+3A_formula">formula</code></td>
<td>
<p>A formula. Use the <code>time</code> function to designate which variable represents time.</p>
</td></tr>
<tr><td><code id="kronos_+3A_data">data</code></td>
<td>
<p>input data</p>
</td></tr>
<tr><td><code id="kronos_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="kronos_+3A_period">period</code></td>
<td>
<p>A numeric. The length of a period, in the same format as the <code>time</code> variable</p>
</td></tr>
<tr><td><code id="kronos_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large data sets.</p>
</td></tr>
<tr><td><code id="kronos_+3A_pairwise">pairwise</code></td>
<td>
<p>A boolean. Toggles whether to perform pairwise ANOVAs as a TukeyHSD-like post-hoc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>kronosOut</code> S4 object containing coefficients and all operations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in Kronos library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)


#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='kronos_anova'>Extract p-value from full fit</h2><span id='topic+kronos_anova'></span>

<h3>Description</h3>

<p>Compute p-values from full fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kronos_anova(fit, time)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kronos_anova_+3A_fit">fit</code></td>
<td>
<p>A lm model fit.</p>
</td></tr>
<tr><td><code id="kronos_anova_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
</table>

<hr>
<h2 id='kronos_predict'>Give tracing information for plotting purposes</h2><span id='topic+kronos_predict'></span>

<h3>Description</h3>

<p>Generate data needed to plot cosinor trace line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kronos_predict(fit, period, time, factors, verbose = verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kronos_predict_+3A_fit">fit</code></td>
<td>
<p>A model fit</p>
</td></tr>
<tr><td><code id="kronos_predict_+3A_period">period</code></td>
<td>
<p>A numeric. The length of a period, in the same format as the <code>time</code> parameter.</p>
</td></tr>
<tr><td><code id="kronos_predict_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="kronos_predict_+3A_factors">factors</code></td>
<td>
<p>A vector. The names of the independent variables.</p>
</td></tr>
<tr><td><code id="kronos_predict_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large data sets.</p>
</td></tr>
</table>

<hr>
<h2 id='kronosListToTable'>Wrangle results from list of <code>KronosOut</code> Objects to publication ready table.</h2><span id='topic+kronosListToTable'></span>

<h3>Description</h3>

<p>These functions provides a unified wrapper to retrieve results
from a list of <code>kronosOut</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kronosListToTable(kronos_list, padjust = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="kronosListToTable_+3A_kronos_list">kronos_list</code></td>
<td>
<p>a list of preferrably named <code>kronosOut</code> objects.</p>
</td></tr>
<tr><td><code id="kronosListToTable_+3A_padjust">padjust</code></td>
<td>
<p>a boolean. Toggles FDR using Benjamini and Hochbergs procedure.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table with circadian output stats per group per feature.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load prepared data stored in 'kronos' library
data("kronos_demo")
output &lt;- kronos(formula = Variable_1 ~ time(Timepoint), 
data = onevariable, period = 24, verbose = TRUE, pairwise = FALSE)

#Extracting data from the output object:
getKronos_fit(output)
getKronos_trace(output)
getKronos_groupwise(output)

#Plotting:
gg_kronos_circle(output)
gg_kronos_sinusoid(output)

#For high-dimensional data, use fw_kronos:
out_list = fw_kronos(x = bigdata[1:50,], formula = ~ Group + time(Timepoint), 
metadata = bigmeta, period = 24, verbose = FALSE, pairwise = TRUE) 

#Extracting data from the output object: 
kronosListToTable(out_list)

#Plotting:
gg_kronos_acrogram(out_list)

</code></pre>

<hr>
<h2 id='kronosOut-class'>An S4 class to contain all output parameters and plotting information from the 'kronos' main pipeline</h2><span id='topic+kronosOut-class'></span>

<h3>Description</h3>

<p>kronosOut is the main output container of the main 'kronos' functions.
</p>


<h3>Slots</h3>


<dl>
<dt><code>input</code></dt><dd><p>A data.frame with the data that was fed to the main workhorse function as 'x'</p>
</dd>
<dt><code>fit</code></dt><dd><p>An lm fit for the entire model for the purpose of assessing differential rhytmicity.</p>
</dd>
<dt><code>to_plot</code></dt><dd><p>A data.frame with the traces required to plot individual sinusoid curves</p>
</dd>
<dt><code>ind_fit</code></dt><dd><p>A data.frame with the parameters from individual rhythmic model fits.</p>
</dd>
<dt><code>pairwise_t</code></dt><dd><p>A data.frame with the p.values for pairwise comparisons, if applicable.</p>
</dd>
</dl>

<hr>
<h2 id='onevariable'>Snippet of example data to demonstrate the functionality of 'kronos' in the most simple scenario.</h2><span id='topic+onevariable'></span>

<h3>Description</h3>

<p>Snippet of example data to demonstrate the functionality of 'kronos' in the most simple scenario.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onevariable
</code></pre>


<h3>Format</h3>

<p>A long format data.frame object with three columns, and 31 rows, samples.
</p>


<h3>Source</h3>

<p>Dummy data modified from metagenomic analysis of mouse caecal contents over the day to demonstrate rhythmicity within one group.
</p>

<hr>
<h2 id='pairwise_cosinor_model'>Fit pairwise cosinor models as some sort of TukeyHSD.</h2><span id='topic+pairwise_cosinor_model'></span>

<h3>Description</h3>

<p>Fit cosinor model for subset of data. Called by main 'kronos' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwise_cosinor_model(data, formula, time, verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pairwise_cosinor_model_+3A_data">data</code></td>
<td>
<p>input data</p>
</td></tr>
<tr><td><code id="pairwise_cosinor_model_+3A_formula">formula</code></td>
<td>
<p>A formula. Use the <code>time</code> function to designate which variable represents time.</p>
</td></tr>
<tr><td><code id="pairwise_cosinor_model_+3A_time">time</code></td>
<td>
<p>A string. Should be the column name containing the time values.</p>
</td></tr>
<tr><td><code id="pairwise_cosinor_model_+3A_verbose">verbose</code></td>
<td>
<p>A boolean. Toggles whether to print diagnostic information while running. Useful for debugging errors on large data sets.</p>
</td></tr>
</table>

<hr>
<h2 id='show+2CkronosOut-method'>Show method for <code>kronosOut</code> object.</h2><span id='topic+show+2CkronosOut-method'></span>

<h3>Description</h3>

<p>method to print <code>kronosOut</code> object by calling <code>show</code>.
Since kronosOut objects are typically unwieldy, also gives some tips on how to handle it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'kronosOut'
show(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="show+2B2CkronosOut-method_+3A_object">object</code></td>
<td>
<p>An <code>kronosOut</code> object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Does not return anything, for efficiency reasons. The obvious side effect is output to the terminal.
</p>

<hr>
<h2 id='twowaydata'>Snippet of example data to demonstrate the functionality of 'kronos' in the two-factor design scenario.</h2><span id='topic+twowaydata'></span>

<h3>Description</h3>

<p>Snippet of example data to demonstrate the functionality of 'kronos' in the two-factor design scenario.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twowaydata
</code></pre>


<h3>Format</h3>

<p>A wide format data.frame object with nine columns, and 150 rows, samples.
</p>


<h3>Source</h3>

<p>Dummy data modified from metagenomic analysis of mouse caecal contents over the day to demonstrate rhythmicity within one group.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
