<!DOCTYPE html><html><head><title>Help for package KSEAapp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {KSEAapp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#KSData'><p>Kinase-Substrate (K-S) Relationship Dataset</p></a></li>
<li><a href='#KSEA.Barplot'><p>The KSEA App Analysis (KSEA Bar Plot Only)</p></a></li>
<li><a href='#KSEA.Complete'><p>The Complete KSEA App Analysis</p></a></li>
<li><a href='#KSEA.Heatmap'><p>The KSEA App Analysis (KSEA Heatmap Only)</p></a></li>
<li><a href='#KSEA.KS_table'><p>The KSEA App Analysis (K-S Dataset Only)</p></a></li>
<li><a href='#KSEA.Scores'><p>The KSEA App Analysis (KSEA Kinase Scores Only)</p></a></li>
<li><a href='#KSEA.Scores.1'><p>One of the 3 datasets for heatmap plotting</p></a></li>
<li><a href='#KSEA.Scores.2'><p>One of the 3 datasets for heatmap plotting</p></a></li>
<li><a href='#KSEA.Scores.3'><p>One of the 3 datasets for heatmap plotting</p></a></li>
<li><a href='#PX'><p>PX dataset for KSEA calculations</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.99.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-04-25</td>
</tr>
<tr>
<td>Title:</td>
<td>Kinase-Substrate Enrichment Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Infers relative kinase activity from phosphoproteomics data
    using the method described by Casado et al. (2013) &lt;<a href="https://doi.org/10.1126%2Fscisignal.2003573">doi:10.1126/scisignal.2003573</a>&gt;.</td>
</tr>
<tr>
<td>Author:</td>
<td>Danica D. Wiredja </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Danica D. Wiredja &lt;dwiredja@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gplots, graphics, stats, grDevices, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-05-02 13:25:34 UTC; Danica</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-05-02 16:11:49 UTC</td>
</tr>
</table>
<hr>
<h2 id='KSData'>Kinase-Substrate (K-S) Relationship Dataset</h2><span id='topic+KSData'></span>

<h3>Description</h3>

<p>K-S annotations from PhosphoSitePlus and NetworKIN predictions;
This is an abbreviated version of the full dataset used purely for demonstration;
please go to the GitHub page for access to the complete file: github.com/casecpb/KSEA/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(KSData)
</code></pre>


<h3>Format</h3>

<p>abbreviated dataframe containing the kinase-substrate annotations and source</p>


<h3>References</h3>

<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>

<hr>
<h2 id='KSEA.Barplot'>The KSEA App Analysis (KSEA Bar Plot Only)</h2><span id='topic+KSEA.Barplot'></span>

<h3>Description</h3>

<p>Takes a formatted phoshoproteomics data input and returns just the summary bar plot of kinase scores
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KSEA.Barplot(KSData, PX, NetworKIN, NetworKIN.cutoff, m.cutoff, p.cutoff,
  export)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KSEA.Barplot_+3A_ksdata">KSData</code></td>
<td>
<p>the Kinase-Substrate dataset uploaded from the file 
prefaced with &quot;PSP&amp;NetworKIN_&quot;
available from github.com/casecpb/KSEA/</p>
</td></tr>
<tr><td><code id="KSEA.Barplot_+3A_px">PX</code></td>
<td>
<p>the experimental data file formatted as described in the KSEA.Complete() documentation</p>
</td></tr>
<tr><td><code id="KSEA.Barplot_+3A_networkin">NetworKIN</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not to include NetworKIN predictions; 
NetworKIN = TRUE means inclusion of NetworKIN predictions</p>
</td></tr>
<tr><td><code id="KSEA.Barplot_+3A_networkin.cutoff">NetworKIN.cutoff</code></td>
<td>
<p>a numeric value between 1 and infinity setting the 
minimum NetworKIN score (can be left out if NetworKIN = FALSE)</p>
</td></tr>
<tr><td><code id="KSEA.Barplot_+3A_m.cutoff">m.cutoff</code></td>
<td>
<p>a numeric value between 0 and infinity indicating the min. # of substrates 
a kinase must have to be included in the bar plot output</p>
</td></tr>
<tr><td><code id="KSEA.Barplot_+3A_p.cutoff">p.cutoff</code></td>
<td>
<p>a numeric value between 0 and 1 indicating the p-value cutoff 
for indicating significant kinases in the bar plot</p>
</td></tr>
<tr><td><code id="KSEA.Barplot_+3A_export">export</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not 
to export the bar plot as a .tiff image into the working directory</p>
</td></tr>
</table>


<h3>Value</h3>

<p>creates the bar plot output highlighting key kinase results
</p>


<h3>References</h3>

<p>Casado et al. (2013) Sci Signal. 6(268):rs6
</p>
<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>KSEA.Barplot(KSData, PX, NetworKIN=TRUE, NetworKIN.cutoff=5, 
             m.cutoff=5, p.cutoff=0.01, export=FALSE)
KSEA.Barplot(KSData, PX, NetworKIN=TRUE, NetworKIN.cutoff=5, 
             m.cutoff=8, p.cutoff=0.05, export=TRUE)
KSEA.Barplot(KSData, PX, NetworKIN=FALSE, m.cutoff=2, p.cutoff=0.05, export=TRUE)

</code></pre>

<hr>
<h2 id='KSEA.Complete'>The Complete KSEA App Analysis</h2><span id='topic+KSEA.Complete'></span>

<h3>Description</h3>

<p>Takes a formatted phoshoproteomics data input and performs KSEA calculations to infer relative kinase activities
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KSEA.Complete(KSData, PX, NetworKIN, NetworKIN.cutoff, m.cutoff, p.cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KSEA.Complete_+3A_ksdata">KSData</code></td>
<td>
<p>the Kinase-Substrate dataset uploaded from the file 
prefaced with &quot;PSP&amp;NetworKIN_&quot;
available from github.com/casecpb/KSEA/</p>
</td></tr>
<tr><td><code id="KSEA.Complete_+3A_px">PX</code></td>
<td>
<p>the experimental data file formatted exactly as described below;
must have 6 columns in the exact order: Protein, Gene, Peptide, Residue.Both, p, FC;
cannot have NA values, or else the entire peptide row is deleted;
Description of each column in PX: 
</p>

<ul>
<li><p>&quot;Protein&quot; the Uniprot ID for the parent protein 
</p>
</li>
<li><p>&quot;Gene&quot; the HUGO gene name for the parent protein 
</p>
</li>
<li><p>&quot;Peptide&quot; the peptide sequence
</p>
</li>
<li><p>&quot;Residue.Both&quot; all phosphosites from that peptide, separated by semicolons if applicable;
must be formatted as the single amino acid abbrev. with the residue position (e.g. S102)
</p>
</li>
<li><p>&quot;p&quot; the p-value of that peptide (if none calculated, please write &quot;NULL&quot;, cannot be NA)
</p>
</li>
<li><p>&quot;FC&quot; the fold change (not log-transformed); usually the control sample is the denominator
</p>
</li></ul>
</td></tr>
<tr><td><code id="KSEA.Complete_+3A_networkin">NetworKIN</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not to include NetworKIN predictions; 
NetworKIN = TRUE means inclusion of NetworKIN predictions</p>
</td></tr>
<tr><td><code id="KSEA.Complete_+3A_networkin.cutoff">NetworKIN.cutoff</code></td>
<td>
<p>a numeric value between 1 and infinity setting the minimum NetworKIN score 
(can be left out if NetworKIN = FALSE)</p>
</td></tr>
<tr><td><code id="KSEA.Complete_+3A_m.cutoff">m.cutoff</code></td>
<td>
<p>a numeric value between 0 and infinity indicating the min. # of substrates 
a kinase must have to be included in the bar plot output</p>
</td></tr>
<tr><td><code id="KSEA.Complete_+3A_p.cutoff">p.cutoff</code></td>
<td>
<p>a numeric value between 0 and 1 indicating the p-value cutoff for
indicating significant kinases in the bar plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>creates the following outputs that are deposited into your working directory: 
a bar plot highlighting key kinase results, a .csv file of all KSEA kinase scores, 
and a .csv file listing all kinase-substrate relationships used for the calculations
</p>


<h3>References</h3>

<p>Casado et al. (2013) Sci Signal. 6(268):rs6
</p>
<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>KSEA.Complete(KSData, PX, NetworKIN=TRUE, NetworKIN.cutoff=5, m.cutoff=5, p.cutoff=0.01)
KSEA.Complete(KSData, PX, NetworKIN=FALSE, m.cutoff=2, p.cutoff=0.05)

</code></pre>

<hr>
<h2 id='KSEA.Heatmap'>The KSEA App Analysis (KSEA Heatmap Only)</h2><span id='topic+KSEA.Heatmap'></span>

<h3>Description</h3>

<p>Takes a list of the KSEA kinase score outputs from KSEA.Scores() 
and creates a merged heatmap (only applicable for multi-treatment studies)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KSEA.Heatmap(score.list, sample.labels, stats, m.cutoff, p.cutoff,
  sample.cluster)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KSEA.Heatmap_+3A_score.list">score.list</code></td>
<td>
<p>the data frame outputs from the KSEA.Scores() function, compiled in a list format</p>
</td></tr>
<tr><td><code id="KSEA.Heatmap_+3A_sample.labels">sample.labels</code></td>
<td>
<p>a character vector of all the sample names for heatmap annotation; 
the names must be in the same order as the data in score.list;
please avoid long names, as they may get cropped in the final image</p>
</td></tr>
<tr><td><code id="KSEA.Heatmap_+3A_stats">stats</code></td>
<td>
<p>character string of either &quot;p.value&quot; or &quot;FDR&quot; indicating 
the data column to use for marking statistically significant scores</p>
</td></tr>
<tr><td><code id="KSEA.Heatmap_+3A_m.cutoff">m.cutoff</code></td>
<td>
<p>a numeric value between 0 and infinity indicating the min. # 
of substrates a kinase must have to be included in the heatmap</p>
</td></tr>
<tr><td><code id="KSEA.Heatmap_+3A_p.cutoff">p.cutoff</code></td>
<td>
<p>a numeric value between 0 and 1 indicating the p-value/FDR cutoff 
for indicating significant kinases in the heatmap</p>
</td></tr>
<tr><td><code id="KSEA.Heatmap_+3A_sample.cluster">sample.cluster</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not 
to perform hierarchical clustering of the sample columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>exports a .png heatmap image highlighting the merged datasets; 
heatmap was generated using the heatmap.2() function (gplots package);
asterisks mark scores that met the statistical cutoff, as defined by p.cutoff;
blue color indicates negative kinase score, and red indicates positive kinase score
</p>


<h3>References</h3>

<p>Casado et al. (2013) Sci Signal. 6(268):rs6
</p>
<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The score.list input must be a list of the data frame outputs from KSEA.Scores() function
#KSEA.Scores.1, KSEA.Scores.2, and KSEA.Scores.3 are all 
#sample datasets provided within this package

KSEA.Heatmap(score.list=list(KSEA.Scores.1, KSEA.Scores.2, KSEA.Scores.3), 
             sample.labels=c("Tumor.A", "Tumor.B", "Tumor.C"), 
             stats="p.value", m.cutoff=3, p.cutoff=0.05, sample.cluster=TRUE)

</code></pre>

<hr>
<h2 id='KSEA.KS_table'>The KSEA App Analysis (K-S Dataset Only)</h2><span id='topic+KSEA.KS_table'></span>

<h3>Description</h3>

<p>Takes a formatted phoshoproteomics data input and returns 
just the kinase-subtrate (K-S) annotations used for KSEA calculations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KSEA.KS_table(KSData, PX, NetworKIN, NetworKIN.cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KSEA.KS_table_+3A_ksdata">KSData</code></td>
<td>
<p>the Kinase-Substrate dataset uploaded from the file 
prefaced with &quot;PSP&amp;NetworKIN_&quot;
available from github.com/casecpb/KSEA/</p>
</td></tr>
<tr><td><code id="KSEA.KS_table_+3A_px">PX</code></td>
<td>
<p>the experimental data file formatted as described in the KSEA.Complete() documentation</p>
</td></tr>
<tr><td><code id="KSEA.KS_table_+3A_networkin">NetworKIN</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not to include NetworKIN predictions; 
NetworKIN = TRUE means inclusion of NetworKIN predictions</p>
</td></tr>
<tr><td><code id="KSEA.KS_table_+3A_networkin.cutoff">NetworKIN.cutoff</code></td>
<td>
<p>a numeric value between 1 and infinity setting 
the minimum NetworKIN score (can be left out if NetworKIN = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>creates a new data frame in R with all kinase-substrate relationships used for the KSEA calculations
</p>


<h3>References</h3>

<p>Casado et al. (2013) Sci Signal. 6(268):rs6
</p>
<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>KSData.dataset = KSEA.KS_table(KSData, PX, NetworKIN=TRUE, NetworKIN.cutoff=3)
KSData.dataset = KSEA.KS_table(KSData, PX, NetworKIN=FALSE)


</code></pre>

<hr>
<h2 id='KSEA.Scores'>The KSEA App Analysis (KSEA Kinase Scores Only)</h2><span id='topic+KSEA.Scores'></span>

<h3>Description</h3>

<p>Takes a formatted phoshoproteomics data input and returns just the KSEA kinase scores and statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KSEA.Scores(KSData, PX, NetworKIN, NetworKIN.cutoff)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KSEA.Scores_+3A_ksdata">KSData</code></td>
<td>
<p>the Kinase-Substrate dataset uploaded from the file 
prefaced with &quot;PSP&amp;NetworKIN_&quot;
available from github.com/casecpb/KSEA/</p>
</td></tr>
<tr><td><code id="KSEA.Scores_+3A_px">PX</code></td>
<td>
<p>the experimental data file formatted as described in the KSEA.Complete() documentation</p>
</td></tr>
<tr><td><code id="KSEA.Scores_+3A_networkin">NetworKIN</code></td>
<td>
<p>a binary input of TRUE or FALSE, indicating whether or not to include NetworKIN predictions; 
NetworKIN = TRUE means inclusion of NetworKIN predictions</p>
</td></tr>
<tr><td><code id="KSEA.Scores_+3A_networkin.cutoff">NetworKIN.cutoff</code></td>
<td>
<p>a numeric value between 1 and infinity setting 
the minimum NetworKIN score (can be left out if NetworKIN = FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>creates a new data frame in R with all the KSEA kinase scores, 
along with each one's statistical assessment
</p>


<h3>References</h3>

<p>Casado et al. (2013) Sci Signal. 6(268):rs6
</p>
<p>Hornbeck et al. (2015) Nucleic Acids Res. 43:D512-20
</p>
<p>Horn et al. (2014) Nature Methods 11(6):603-4
</p>


<h3>Examples</h3>

<pre><code class='language-R'>scores = KSEA.Scores(KSData, PX, NetworKIN=TRUE, NetworKIN.cutoff=3)
scores = KSEA.Scores(KSData, PX, NetworKIN=FALSE)

</code></pre>

<hr>
<h2 id='KSEA.Scores.1'>One of the 3 datasets for heatmap plotting</h2><span id='topic+KSEA.Scores.1'></span>

<h3>Description</h3>

<p>A sample KSEA.Scores output generated from the KSEA.Scores() function 
(or alternatively, the &quot;KSEA Kinase Scores.csv&quot; output from the KSEA.Complete() function, loaded into R)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(KSEA.Scores.1)
</code></pre>


<h3>Format</h3>

<p>dataframe containing 7 columns in the exact order as listed below.
</p>

<ul>
<li><p>&quot;KinaseGene&quot; the HUGO gene name of the kinase
</p>
</li>
<li><p>&quot;mS&quot; the mean log2FC of all the kinase's identified substrates
</p>
</li>
<li><p>&quot;Enrichment&quot; the enrichment score (refer to Casado et al. (2013) Sci. Signal., 6, rs6-rs6)
</p>
</li>
<li><p>&quot;m&quot; the number of experimentally-identifed substrates annotating to that kinase
</p>
</li>
<li><p>&quot;z.score&quot; the normalized kinase score
</p>
</li>
<li><p>&quot;p.value&quot; the statistical assessment of the kinase score
</p>
</li>
<li><p>&quot;FDR&quot; the p-value adjusted for multiple hypothesis testing by the Benjamin-Hochberg method
</p>
</li></ul>


<h3>References</h3>

<p>unpublished data
</p>

<hr>
<h2 id='KSEA.Scores.2'>One of the 3 datasets for heatmap plotting</h2><span id='topic+KSEA.Scores.2'></span>

<h3>Description</h3>

<p>A sample KSEA.Scores output generated from the KSEA.Scores() function 
(or alternatively, the &quot;KSEA Kinase Scores.csv&quot; output from the KSEA.Complete() function, loaded into R)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(KSEA.Scores.2)
</code></pre>


<h3>Format</h3>

<p>dataframe containing 7 columns in the exact order as listed below.
</p>

<ul>
<li><p>&quot;KinaseGene&quot; the HUGO gene name of the kinase
</p>
</li>
<li><p>&quot;mS&quot; the mean log2FC of all the kinase's identified substrates
</p>
</li>
<li><p>&quot;Enrichment&quot; the enrichment score (refer to Casado et al. (2013) Sci. Signal., 6, rs6-rs6)
</p>
</li>
<li><p>&quot;m&quot; the number of experimentally-identifed substrates annotating to that kinase
</p>
</li>
<li><p>&quot;z.score&quot; the normalized kinase score
</p>
</li>
<li><p>&quot;p.value&quot; the statistical assessment of the kinase score
</p>
</li>
<li><p>&quot;FDR&quot; the p-value adjusted for multiple hypothesis testing by the Benjamin-Hochberg method
</p>
</li></ul>


<h3>References</h3>

<p>unpublished data
</p>

<hr>
<h2 id='KSEA.Scores.3'>One of the 3 datasets for heatmap plotting</h2><span id='topic+KSEA.Scores.3'></span>

<h3>Description</h3>

<p>A sample KSEA.Scores output generated from the KSEA.Scores() function 
(or alternatively, the &quot;KSEA Kinase Scores.csv&quot; output from the KSEA.Complete() function, loaded into R)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(KSEA.Scores.3)
</code></pre>


<h3>Format</h3>

<p>dataframe containing 7 columns in the exact order as listed below.
</p>

<ul>
<li><p>&quot;KinaseGene&quot; the HUGO gene name of the kinase
</p>
</li>
<li><p>&quot;mS&quot; the mean log2FC of all the kinase's identified substrates
</p>
</li>
<li><p>&quot;Enrichment&quot; the enrichment score (refer to Casado et al. (2013) Sci. Signal., 6, rs6-rs6)
</p>
</li>
<li><p>&quot;m&quot; the number of experimentally-identifed substrates annotating to that kinase
</p>
</li>
<li><p>&quot;z.score&quot; the normalized kinase score
</p>
</li>
<li><p>&quot;p.value&quot; the statistical assessment of the kinase score
</p>
</li>
<li><p>&quot;FDR&quot; the p-value adjusted for multiple hypothesis testing by the Benjamin-Hochberg method
</p>
</li></ul>


<h3>References</h3>

<p>unpublished data
</p>

<hr>
<h2 id='PX'>PX dataset for KSEA calculations</h2><span id='topic+PX'></span>

<h3>Description</h3>

<p>A sample PX dataset of the experimental phosphoproteomics input
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(PX)
</code></pre>


<h3>Format</h3>

<p>the experimental data file must be formatted exactly as described below;
must have 6 columns in the exact order: Protein, Gene, Peptide, Residue.Both, p, FC;
cannot have NA values, or else the entire peptide row is deleted;
Description of each column in PX: 
</p>

<ul>
<li><p>&quot;Protein&quot; the Uniprot ID for the parent protein 
</p>
</li>
<li><p>&quot;Gene&quot; the HUGO gene name for the parent protein 
</p>
</li>
<li><p>&quot;Peptide&quot; the peptide sequence
</p>
</li>
<li><p>&quot;Residue.Both&quot; all phosphosites from that peptide, separated by semicolons if applicable;
must be formatted as the single amino acid abbrev. with the residue position (e.g. S102)
</p>
</li>
<li><p>&quot;p&quot; the p-value of that peptide (if none calculated, please write &quot;NULL&quot;, cannot be NA)
</p>
</li>
<li><p>&quot;FC&quot; the fold change (not log-transformed); usually the control sample is the denominator
</p>
</li></ul>


<h3>References</h3>

<p>unpublished data
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
