<!DOCTYPE html><html><head><title>Help for package langevitour</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {langevitour}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#knnDenoise'><p>k-nearest neighbor denoising of a set of points</p></a></li>
<li><a href='#langevitour'><p>Langevin Tour</p></a></li>
<li><a href='#langevitour-shiny'><p>Shiny bindings for langevitour</p></a></li>
<li><a href='#zeiselPC'><p>Principal components of scRNA-Seq of mouse brain cells</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Langevin Tour</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7</td>
</tr>
<tr>
<td>Description:</td>
<td>
    An HTML widget that randomly tours 2D projections of numerical data. A random walk through projections of the data is shown. The user can manipulate the plot to use specified axes, or turn on Guided Tour mode to find an informative projection of the data. Groups within the data can be hidden or shown, as can particular axes. Points can be brushed, and the selection can be linked to other widgets using crosstalk. The underlying method to produce the random walk and projection pursuit uses Langevin dynamics. The widget can be used from within R, or included in a self-contained R Markdown or Quarto document or presentation, or used in a Shiny app.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://logarithmic.net/langevitour/">https://logarithmic.net/langevitour/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/pfh/langevitour/issues/">https://github.com/pfh/langevitour/issues/</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>htmlwidgets, crosstalk, RANN, assertthat</td>
</tr>
<tr>
<td>Suggests:</td>
<td>shiny, palmerpenguins, knitr, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-24 04:00:47 UTC; pfh</td>
</tr>
<tr>
<td>Author:</td>
<td>Paul Harrison <a href="https://orcid.org/0000-0002-3980-268X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Paul Harrison &lt;pfh@logarithmic.net&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-24 04:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='knnDenoise'>k-nearest neighbor denoising of a set of points</h2><span id='topic+knnDenoise'></span>

<h3>Description</h3>

<p>Reduce noise in a high-dimensional dataset by averaging each point with its nearby neighbors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>knnDenoise(X, block = rep(1, nrow(X)), k = 30, steps = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="knnDenoise_+3A_x">X</code></td>
<td>
<p>A matrix of numeric data, or something that can be cast to a matrix. Each row represents a point.</p>
</td></tr>
<tr><td><code id="knnDenoise_+3A_block">block</code></td>
<td>
<p>Optional. A block for each row in X. A factor, or something that can be cast to a factor. Denoising will be performed independently within each block.</p>
</td></tr>
<tr><td><code id="knnDenoise_+3A_k">k</code></td>
<td>
<p>Number of nearest neighbors to find around each point (including itself).</p>
</td></tr>
<tr><td><code id="knnDenoise_+3A_steps">steps</code></td>
<td>
<p>Number of steps to take along the directed k-nearest neighbor graph. <code>steps=1</code> uses the k-nearest neighbors, <code>steps=2</code> uses the k-nearest neighbors and their k-nearest neighbors, etc.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>knnDenoise</code> first finds the <code>k</code>-nearest neighbors to each point (including the point itself). Then, for each point, the average is found of the points reachable in <code>steps</code> steps along the directed k-nearest neighbor graph.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(palmerpenguins)

completePenguins &lt;- na.omit(penguins[,c(1,3,4,5,6)])

# Dimensions need to be on comparable scales to apply knnDenoise
scaled &lt;- scale(completePenguins[,-1])

denoised &lt;- knnDenoise(scaled)

langevitour(denoised, completePenguins$species, pointSize=2)

</code></pre>

<hr>
<h2 id='langevitour'>Langevin Tour</h2><span id='topic+langevitour'></span>

<h3>Description</h3>

<p>Make a Langevin Tour HTML widget, which can be used to explore high-dimensional numerical datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>langevitour(
  X,
  group = NULL,
  name = NULL,
  center = NULL,
  scale = NULL,
  extraAxes = NULL,
  lineFrom = NULL,
  lineTo = NULL,
  lineColors = NULL,
  axisColors = NULL,
  levelColors = NULL,
  colorVariation = 0.1,
  pointSize = 1,
  subsample = NULL,
  state = NULL,
  width = NULL,
  height = NULL,
  elementId = NULL,
  link = NULL,
  link_filter = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="langevitour_+3A_x">X</code></td>
<td>
<p>The data to plot. A matrix of numeric data, or something that can be cast to a matrix. Rows will be shown as points in the widget. Columns are the variables of your data.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_group">group</code></td>
<td>
<p>A group for each row in X, will be used to color points. A factor, or something that can be cast to a factor.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_name">name</code></td>
<td>
<p>A name for each row in X.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_center">center</code></td>
<td>
<p>Center for each variable. If omitted, the column means will be used.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_scale">scale</code></td>
<td>
<p>Scale for each variable. Scale +/- center will be the range of guaranteed visible data. If omitted, a reasonable default will be chosen, equal for all variables. (The default is the largest singular value of the centered X times 2.5.)</p>
</td></tr>
<tr><td><code id="langevitour_+3A_extraaxes">extraAxes</code></td>
<td>
<p>A matrix with each column defining a projection of interest. The columns of <code>X %*% extraAxes</code> will be presented as extra &quot;variables&quot;.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_linefrom">lineFrom</code></td>
<td>
<p>A vector of row numbers. Draw lines starting at these rows.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_lineto">lineTo</code></td>
<td>
<p>A vector of row numbers. Draw lines ending at these rows.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_linecolors">lineColors</code></td>
<td>
<p>Character vector. A CSS color for each line.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_axiscolors">axisColors</code></td>
<td>
<p>Character vector. CSS colors for each variable and then each extra axis.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_levelcolors">levelColors</code></td>
<td>
<p>Character vector. CSS colors for each level of <code>group</code>.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_colorvariation">colorVariation</code></td>
<td>
<p>Number between 0 and 1. Individual points are given slightly different brightnesses. How strong should this effect be?</p>
</td></tr>
<tr><td><code id="langevitour_+3A_pointsize">pointSize</code></td>
<td>
<p>Point radius in pixels. A single number, or a number for each row in X.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_subsample">subsample</code></td>
<td>
<p>For speed, randomly subsample down to this many rows.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_state">state</code></td>
<td>
<p>A JSON string, or an object that htmlwidgets will convert to the correct JSON. Initial widget state settings. The state of a widget can be obtained from its &quot;further controls and information&quot; pane. I am not going to guarantee that states will be compatible between versions of langevitour. Hint: Since JSON uses double quotes, surround the string in single quotes.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_width">width</code></td>
<td>
<p>Width of widget in CSS units, for example &quot;700px&quot; or &quot;100%&quot;.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_height">height</code></td>
<td>
<p>Height of widget in CSS units, for example &quot;600px&quot; or &quot;75vh&quot;.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_elementid">elementId</code></td>
<td>
<p>An element ID for the widget, see <code>htmlwidgets::createWidget</code>.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_link">link</code></td>
<td>
<p>A SharedData object from the crosstalk package to share selections and filters with other htmlwidgets. The data in this object is not used, just the keys and group name. The rows of <code>link$origData()</code> should correspond to the rows of X.</p>
</td></tr>
<tr><td><code id="langevitour_+3A_link_filter">link_filter</code></td>
<td>
<p>TRUE or FALSE. If using crosstalk, should hiding groups in langevitour also cause them to be filtered in linked widgets?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The only required argument is <code>X</code>, the high-dimensional collection of points. The <code>group</code> argument is also commonly used so that groups of points can be distinguished by color. Further arguments adjust the appearance or provide advanced features.
</p>
<p>langevitour will by default not scale variables individually. If you want variables to be individually scaled, use something like <code>scale=apply(X,2,sd)*4</code>. Using the <code>scale</code> argument rather than modifying <code>X</code> directly ensures the plot axes within the widgets retain the original units.
</p>
<p>In Javascript, the langevitour object can be obtained using <code>document.getElementById(elementId).langevitour</code>. For example you could have a button that sets the state of a widget using <code>document.getElementById(elementId).langevitour.setState(desiredState)</code>.
</p>


<h3>Value</h3>

<p>An htmlwidget object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(palmerpenguins)

completePenguins &lt;- na.omit(penguins[,c(1,3,4,5,6)])
scale &lt;- apply(completePenguins[,-1], 2, sd)*4

langevitour(
    completePenguins[,-1], 
    completePenguins$species, 
    scale=scale, pointSize=2)


# An example setting the widget's initial state
 
langevitour(
    completePenguins[,-1], 
    completePenguins$species, 
    scale=scale, pointSize=2,
    state='{"guideType":"pca","labelInactive":["bill_length_mm"]}')

</code></pre>

<hr>
<h2 id='langevitour-shiny'>Shiny bindings for langevitour</h2><span id='topic+langevitour-shiny'></span><span id='topic+langevitourOutput'></span><span id='topic+renderLangevitour'></span>

<h3>Description</h3>

<p>Output and render functions for using langevitour within Shiny
applications and interactive Rmd documents.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>langevitourOutput(outputId, width = "100%", height = "600px")

renderLangevitour(expr, env = parent.frame(), quoted = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="langevitour-shiny_+3A_outputid">outputId</code></td>
<td>
<p>output variable to read from</p>
</td></tr>
<tr><td><code id="langevitour-shiny_+3A_width">width</code>, <code id="langevitour-shiny_+3A_height">height</code></td>
<td>
<p>Must be a valid CSS unit (like <code>'100%'</code>,
<code>'400px'</code>, <code>'auto'</code>) or a number, which will be coerced to a
string and have <code>'px'</code> appended.</p>
</td></tr>
<tr><td><code id="langevitour-shiny_+3A_expr">expr</code></td>
<td>
<p>An expression that generates a langevitour, usually a block of code ending with a call to <code>langevitour()</code></p>
</td></tr>
<tr><td><code id="langevitour-shiny_+3A_env">env</code></td>
<td>
<p>The environment in which to evaluate <code>expr</code>.</p>
</td></tr>
<tr><td><code id="langevitour-shiny_+3A_quoted">quoted</code></td>
<td>
<p>Is <code>expr</code> a quoted expression (with <code>quote()</code>)? This
is useful if you want to save an expression in a variable.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
library(shiny)
library(palmerpenguins)

completePenguins &lt;- na.omit(penguins[,c(1,3,4,5,6)])
scale &lt;- apply(completePenguins[,-1], 2, sd)*4

ui &lt;- fluidPage(
    sliderInput('zoom', 'Zoom', 0, min=-1, max=1, step=0.1),
    langevitourOutput('widget')
)

server &lt;- function(input,output) { 
    output$widget &lt;- renderLangevitour({
        langevitour(
            completePenguins[,-1], 
            completePenguins$species, 
            scale=scale * 10^input$zoom, pointSize=2)
    })
}

app &lt;- shinyApp(ui, server)

# Use runApp(app) or runGadget(app) to run app.

</code></pre>

<hr>
<h2 id='zeiselPC'>Principal components of scRNA-Seq of mouse brain cells</h2><span id='topic+zeiselPC'></span>

<h3>Description</h3>

<p>Single-cell RNA-Seq gene expression of 2,816 mouse brain cells (Zeisel, 2015). The top 10 principal components were produced using the steps in the <a href="https://bioconductor.org/books/3.13/OSCA.workflows/zeisel-mouse-brain-strt-seq.html">Bioconductor OSCA workflow</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(zeiselPC)
</code></pre>


<h3>Format</h3>

<p>A data frame with 2,816 rows representing brain cells and 11 columns:
</p>

<dl>
<dt>type</dt><dd><p>Cell type.</p>
</dd>
<dt>PC1</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC2</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC3</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC4</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC5</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC6</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC7</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC8</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC9</dt><dd><p>Principal component score.</p>
</dd>
<dt>PC10</dt><dd><p>Principal component score.</p>
</dd>
</dl>



<h3>References</h3>

<p>Zeisel, A., Muñoz-Manchado, A. B., Codeluppi, S., Lönnerberg, P., La Manno, G., Juréus, A., Marques, S., Munguba, H., He, L., Betsholtz, C., Rolny, C., Castelo-Branco, G., Hjerling-Leffler, J., &amp; Linnarsson, S. (2015). Cell types in the mouse cortex and hippocampus revealed by single-cell RNA-seq. <em>Science</em>, 347(6226), 1138–1142. <a href="https://doi.org/10.1126/science.aaa1934">doi:10.1126/science.aaa1934</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(zeiselPC)
langevitour(zeiselPC[,-1], zeiselPC$type)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
