<!DOCTYPE html><html><head><title>Help for package LDAandLDAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LDAandLDAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LDAandLDAS-package'><p>Linkage Disequilibrium of Ancestry (LDA) and LDA Score (LDAS)</p></a></li>
<li><a href='#cal_lda'><p>LDA of a pair of SNPs</p></a></li>
<li><a href='#example_map'>
<p>Example genetic maps</p></a></li>
<li><a href='#example_painting_p1'>
<p>Example painting data for population 1.</p></a></li>
<li><a href='#example_painting_p2'>
<p>Example painting data for population 2.</p></a></li>
<li><a href='#LDA'><p>LDA of all pairs of SNPs</p></a></li>
<li><a href='#LDAS'><p>LDA Score</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Linkage Disequilibrium of Ancestry (LDA) and LDA Score (LDAS)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yaoling Yang &lt;yaoling.yang@bristol.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Computation of linkage disequilibrium of ancestry (LDA) and linkage disequilibrium of ancestry score (LDAS). LDA calculates the pairwise linkage disequilibrium of ancestry between single nucleotide polymorphisms (SNPs). LDAS calculates the LDA score of SNPs. The methods are described in Barrie W, Yang Y, Irving-Pease E.K, et al (2024) &lt;<a href="https://doi.org/10.1038%2Fs41586-023-06618-z">doi:10.1038/s41586-023-06618-z</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.8), parallel</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-09 00:48:07 UTC; yyl19</td>
</tr>
<tr>
<td>Author:</td>
<td>Yaoling Yang <a href="https://orcid.org/0000-0003-4905-8097"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Daniel Lawson <a href="https://orcid.org/0000-0002-5311-6213"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-09 08:00:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='LDAandLDAS-package'>Linkage Disequilibrium of Ancestry (LDA) and LDA Score (LDAS)</h2><span id='topic+LDAandLDAS'></span><span id='topic+LDAandLDAS-package'></span>

<h3>Description</h3>

<p>This is the software for Linkage disequilibrium of ancestry (LDA) and LDA score (LDAS) which is proposed by
the papar Genetic risk for Multiple Sclerosis originated in Pastoralist Steppe populations,
Barrie W, Yang Y, Attfield K E, et al (2022).
</p>
<p>LDA quantifies the correlations between the ancestry of two SNPs,
measuring the proportion of individuals who have experienced a recombination
leading to a change in ancestry, relative to the genome-wide baseline.
</p>
<p>LDA score is the total amount of genome in LDA with each SNP
(measured in recombination map distance), which is useful for detecting the
signal of “recombinant favouring selection”.
</p>
<p>The codes for LDA and LDAS are hosted at <a href="https://github.com/YaolingYang/LDAandLDAscore">https://github.com/YaolingYang/LDAandLDAscore</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Yaoling Yang <a href="mailto:yaoling.yang@bristol.ac.uk">yaoling.yang@bristol.ac.uk</a> (<a href="https://orcid.org/0000-0003-4905-8097">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Daniel Lawson <a href="mailto:Dan.Lawson@bristol.ac.uk">Dan.Lawson@bristol.ac.uk</a> (<a href="https://orcid.org/0000-0002-5311-6213">ORCID</a>)
</p>
</li></ul>



<h3>References</h3>

<p>Barrie, W., Yang, Y., Irving-Pease, E.K. et al. Elevated genetic risk for multiple sclerosis emerged in steppe pastoralist populations. Nature 625, 321–328 (2024).
</p>

<hr>
<h2 id='cal_lda'>LDA of a pair of SNPs</h2><span id='topic+cal_lda'></span>

<h3>Description</h3>

<p>Computation of the pairwise Linkage Disequilibrium of Ancestry (LDA) between a pair of single nucleotide polymorphisms (SNPs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cal_lda(data_resample, data_base, data_experiment, n_ancestry)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cal_lda_+3A_data_resample">data_resample</code></td>
<td>
<p>a data frame of the first SNP's ancestry probabilities after resampling.
Different ancestry probabilities are in different columns.</p>
</td></tr>
<tr><td><code id="cal_lda_+3A_data_base">data_base</code></td>
<td>
<p>a data frame of the first SNP's ancestry probabilities.
Different ancestry probabilities are in different columns.</p>
</td></tr>
<tr><td><code id="cal_lda_+3A_data_experiment">data_experiment</code></td>
<td>
<p>a data frame of the second SNP's ancestry probabilities.
Different ancestry probabilities are in different columns.</p>
</td></tr>
<tr><td><code id="cal_lda_+3A_n_ancestry">n_ancestry</code></td>
<td>
<p>a positive integer representing the number of different ancestries.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the LDA between two SNPs.
Resampling of one of the SNPs' painting data is required prior to implementing this function.
To compute pairwise LDA between multiple pairs of SNPs, please use <code><a href="#topic+LDA">LDA</a></code>.
</p>


<h3>Value</h3>

<p>a numeric number representing the pairwise LDA of the two SNPs.
</p>


<h3>References</h3>

<p>Barrie, W., Yang, Y., Irving-Pease, E.K. et al. Elevated genetic risk for multiple sclerosis emerged in steppe pastoralist populations. Nature 625, 321–328 (2024).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# compute the LDA between the 50th SNP and the 55th SNP
# painting data for the 50th SNP (2 ancestries)
data_base &lt;- cbind(LDAandLDAS::example_painting_p1[,50],
                   LDAandLDAS::example_painting_p2[,50])

# painting data for the 55th SNP (2 ancestries)
data_experiment &lt;- cbind(LDAandLDAS::example_painting_p1[,55],
                         LDAandLDAS::example_painting_p2[,55])

# resample painting data for the 50th SNP
data_resample &lt;- data_base[sample(1:nrow(data_base)),]

#compute their pairwise LDA
LDA_value &lt;- cal_lda(data_resample,data_base,data_experiment,2)

</code></pre>

<hr>
<h2 id='example_map'>
Example genetic maps
</h2><span id='topic+example_map'></span>

<h3>Description</h3>

<p>Example genetic maps including the physical position and the genetic distance of single nucleotide polymorphisms (SNPs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("example_map")</code></pre>


<h3>Format</h3>

<p>A data frame with 2,000 haploid genomes of the physical position, named <code>SNP</code>, and the genetic distance of these SNP, named <code>gd</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_map)
</code></pre>

<hr>
<h2 id='example_painting_p1'>
Example painting data for population 1.
</h2><span id='topic+example_painting_p1'></span>

<h3>Description</h3>

<p>Example painting data (average ancestry probabilities) of a chromosome for population 1, including 2000 genomes (observations) and 200 single nucleotide polymorphisms (SNPs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("example_painting_p1")</code></pre>


<h3>Format</h3>

<p>A data frame with 1500 haploid genomes (observations) on 200 SNPs (variables).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_painting_p1)
</code></pre>

<hr>
<h2 id='example_painting_p2'>
Example painting data for population 2.
</h2><span id='topic+example_painting_p2'></span>

<h3>Description</h3>

<p>Example painting data (average ancestry probabilities) of a chromosome for population 2, including 2000 genomes (observations) and 200 single nucleotide polymorphisms (SNPs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("example_painting_p2")</code></pre>


<h3>Format</h3>

<p>A data frame with 1500 haploid genomes (observations) on 200 SNPs (variables).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_painting_p2)
</code></pre>

<hr>
<h2 id='LDA'>LDA of all pairs of SNPs</h2><span id='topic+LDA'></span>

<h3>Description</h3>

<p>Computation of the pairwise Linkage Disequilibrium of Ancestry (LDA) between all pairs of single nucleotide polymorphisms (SNPs).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDA(paintings, SNPidx = NULL, SNPlimit = NULL, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDA_+3A_paintings">paintings</code></td>
<td>
<p>a list of data frames of the N*k painting data
(average ancestry probabilities) from different populations
(N is the number of genomes, k is the number of SNPs).</p>
</td></tr>
<tr><td><code id="LDA_+3A_snpidx">SNPidx</code></td>
<td>
<p>a numeric vector representing the LDA of which SNPs (indices) are computed.
By default, SNPidx=NULL which specifies the LDA of all the SNPs will be computed.</p>
</td></tr>
<tr><td><code id="LDA_+3A_snplimit">SNPlimit</code></td>
<td>
<p>a positive integer representing the maximum number of SNPs at each side of
a SNP that is used to calculate the pairwise LDA for the SNP.
The value shouldn't be larger than the total number of SNPs.
We may set a limit if the LDAs between SNPs far in distance are not to be investigated.</p>
</td></tr>
<tr><td><code id="LDA_+3A_verbose">verbose</code></td>
<td>
<p>logical. If verbose=TRUE, print the process of calculating the pairwise LDA for the i-th SNP.
By default, verbose=FALSE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Linkage Disequilibrium of Ancestry (LDA) quantifies the correlations
between the ancestry of two SNPs, measuring the proportion of individuals who
have experienced a recombination leading to a change in ancestry,
relative to the genome-wide baseline.
</p>


<h3>Value</h3>

<p>a data frame of the pairwise LDA, with SNPs in the decreasing
order of physical position on a chromosome.
</p>


<h3>References</h3>

<p>Barrie, W., Yang, Y., Irving-Pease, E.K. et al. Elevated genetic risk for multiple sclerosis emerged in steppe pastoralist populations. Nature 625, 321–328 (2024).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# visualize the painting data
# Painting data are the average probabilities of different populations
head(LDAandLDAS::example_painting_p1[1:5,],10)

# combine the painting data for two populations as a list
# to make to input data for function 'LDA'.
paintings=list(LDAandLDAS::example_painting_p1,
               LDAandLDAS::example_painting_p2)

# calculate the pairwise LDA of SNPs
LDA_result &lt;- LDA(paintings)

# if we only want to calculate the LDA of the 76th-85th SNP in the map
# based on the 31st-130th SNP, which aims at saving the memory
paintings2=list(LDAandLDAS::example_painting_p1[,31:130],
                LDAandLDAS::example_painting_p2[,31:130])
# note that the 76th-85th SNP in the original dataset is only the
# (76-30)th-(85-30)th SNP in the new dataset (paintings2)
LDA_result2 &lt;- LDA(paintings2,SNPidx=76:85-30)


</code></pre>

<hr>
<h2 id='LDAS'>LDA Score</h2><span id='topic+LDAS'></span>

<h3>Description</h3>

<p>Computation of the Linkage Disequilibrium of Ancestry Score (LDAS) of each single nucleotide polymorphism (SNP).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDAS(
  LDA_data,
  SNPidx = NULL,
  map,
  window = 5,
  runparallel = FALSE,
  mc.cores = 8,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDAS_+3A_lda_data">LDA_data</code></td>
<td>
<p>a data frame of LDA between all pairs of SNPs that are within the 'window'.
SNPs should be in the decreasing order of physical position on a chromosome for both rows and columns.
This is the output from <code><a href="#topic+LDA">LDA</a></code>.</p>
</td></tr>
<tr><td><code id="LDAS_+3A_snpidx">SNPidx</code></td>
<td>
<p>a numeric vector denoting the LDAS of which SNPs
(located in which rows of the map) are computed.
All the SNPs' indices in the LDA_data should be included in SNPidx.
By default, SNPidx=NULL which specifies the LDAS of all the SNPs in the map will be computed.</p>
</td></tr>
<tr><td><code id="LDAS_+3A_map">map</code></td>
<td>
<p>a data frame of the physical position and genetic distance of
all the SNPs contained in 'LDA_data'.
'map' contains two columns.
The first column is the physical distance (unit: b) of SNPs in the decreasing order.
The second column is the genetic distance (unit: cM) of SNPs.</p>
</td></tr>
<tr><td><code id="LDAS_+3A_window">window</code></td>
<td>
<p>a positive number specifying the genetic distance that
the LDA score of each SNP is computed within. By default, window=5.</p>
</td></tr>
<tr><td><code id="LDAS_+3A_runparallel">runparallel</code></td>
<td>
<p>logical. Parallel programming or not (note: unavailable for Windows system).</p>
</td></tr>
<tr><td><code id="LDAS_+3A_mc.cores">mc.cores</code></td>
<td>
<p>a positive number specifying the number of cores used for parallel programming.
By default, mc.cores=8.</p>
</td></tr>
<tr><td><code id="LDAS_+3A_verbose">verbose</code></td>
<td>
<p>logical. Print the process of calculating the LDA score for the i-th SNP.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>LDA score is the total amount of genome in LDA with each SNP
(measured in recombination map distance).
A low LDA score is the signal of “recombinant favouring selection”.
</p>


<h3>Value</h3>

<p>a data frame of the LDA score and its upper and lower bound
at the physical position of each SNP.
</p>


<h3>References</h3>

<p>Barrie, W., Yang, Y., Irving-Pease, E.K. et al. Elevated genetic risk for multiple sclerosis emerged in steppe pastoralist populations. Nature 625, 321–328 (2024).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# visualize the painting data
# Painting data are the average probabilities of different populations
head(LDAandLDAS::example_painting_p1[1:5,],10)

# combine the painting data for two ancestries as a list
# to make to input data for function 'LDA'.
paintings=list(LDAandLDAS::example_painting_p1,
          LDAandLDAS::example_painting_p2)

# calculate the pairwise LDA of SNPs
LDA_result &lt;- LDA(paintings)

# map is the data containing two columns
# The first column is the physical position (unit: b) (decreasing order)
# The second column is the recombination distance (unit: cM) of the SNPs
head(LDAandLDAS::example_map,10)

# calculate the LDA score for the SNPs
LDA_score &lt;- LDAS(LDA_result,map=LDAandLDAS::example_map,window=10)

#visualize the LDA scores
plot(x=LDA_score$SNP,y=LDA_score$LDAS)

#' # if we only want to calculate the LDA of the 76th-85th SNP in the map
# based on the 31st-130th SNP, which aims at saving the memory
paintings2=list(LDAandLDAS::example_painting_p1[,31:130],
                LDAandLDAS::example_painting_p2[,31:130])
# note that the 76th-85th SNP in the original dataset is only the
# (76-30)th-(85-30)th SNP in the new dataset (paintings2)
LDA_result2 &lt;- LDA(paintings2,SNPidx=76:85-30)

# calculate the LDA score for the SNPs
LDA_score2 &lt;- LDAS(LDA_result2,SNPidx=76:85-30,
                   map=LDAandLDAS::example_map[31:130,],window=5)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
