<!DOCTYPE html><html><head><title>Help for package LDcorSV</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LDcorSV}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LDcorSV-package'><p>LDcorSV</p></a></li>
<li><a href='#data.test'><p>data.test</p></a></li>
<li><a href='#Info.Locus'><p>Information on loci</p></a></li>
<li><a href='#Inv.proj.matrix.sdp'><p>Inv.proj.matrix.sdp</p></a></li>
<li><a href='#LD.Measures'><p>LD Measures</p></a></li>
<li><a href='#Measure.R2'><p>r^2 measure</p></a></li>
<li><a href='#Measure.R2S'><p>r^2_S measure</p></a></li>
<li><a href='#Measure.R2V'><p>r^2_V measure</p></a></li>
<li><a href='#Measure.R2VS'><p>r^2_VS measure</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Linkage Disequilibrium Corrected by the Structure and the
Relatedness</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel</td>
</tr>
<tr>
<td>Author:</td>
<td>David Desrousseaux, Florian Sandron, Aurélie Siberchicot,
        Christine Cierco-Ayrolles and Brigitte Mangin</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aurélie Siberchicot &lt;aurelie.siberchicot@univ-lyon1.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Four measures of linkage disequilibrium are provided: the usual r^2
        measure, the r^2_S measure (r^2 corrected by the structure
        sample), the r^2_V (r^2 corrected by the relatedness of
        genotyped individuals), the r^2_VS measure (r^2 corrected by
        both the relatedness of genotyped individuals and the structure
        of the sample).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-25 09:46:01 UTC; aurelie</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-26 09:00:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='LDcorSV-package'>LDcorSV</h2><span id='topic+LDcorSV-package'></span><span id='topic+LDcorSV'></span>

<h3>Description</h3>

<p>The package provides a set of functions which aim is to propose four measures of linkage disequilibrium:
- <code>Measure.R2</code>: the usual <code class="reqn">r^2</code> measure.
</p>
<p>- <code>Measure.R2S</code>:  <code class="reqn">r^2</code> corrected by the structure of the sample (<code class="reqn">r^2_S</code>).
</p>
<p>- <code>Measure.R2V</code>: <code class="reqn">r^2</code> corrected by the relatedness of genotyped individuals (<code class="reqn">r^2_V</code>).
</p>
<p>- <code>Measure.R2VS</code>: <code class="reqn">r^2</code> corrected by both the relatedness of genotyped individuals  and the structure of the sample (<code class="reqn">r^2_VS</code>).
</p>
<p>- <code>LD.Measures</code>: this function computes the four measures of linkage disequilibrium (<code class="reqn">r^2</code>, <code class="reqn">r^2_V</code>, 
<code class="reqn">r^2_S</code> and <code class="reqn">r^2_{VS}</code>) for a set of loci and gives extra information about them.
</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin 
</p>
<p>Maintainer: aurelie.siberchicot@univ-lyon1.fr
</p>


<h3>References</h3>

<p>Mangin, B., Siberchicot, A., Nicolas, S., Doligez, A., This, P., Cierco-Ayrolles, C. (2012). Novel measures of linkage disequilibrium that correct the bias due to population structure and relatedness. Heredity, 108 (3), 285-291. DOI: 10.1038/hdy.2011.73</p>

<hr>
<h2 id='data.test'>data.test</h2><span id='topic+data.test'></span>

<h3>Description</h3>

<p><code>data.test</code> is a list of 3 elements: 
</p>
<p>- <code>Geno</code>: allelic doses of 20 markers on a chromosome of 91 Vitis vinifera plants.
</p>
<p>- <code>V.WAIS</code>: kinship matrix of 91 plants of Vitis vinifera.
</p>
<p>- <code>S.2POP</code>: structure population matrix of 91 plants of Vitis vinifera in two sub-populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data.test)</code></pre>


<h3>Format</h3>

<p>A list containing the following components:
</p>
<p>- <code>Geno</code>: matrix (91 x 20) of numerical values
</p>
<p>- <code>V.WAIS</code>: matrix (91 x 91) of numerical values
</p>
<p>- <code>S.2POP</code>: matrix (91 x 1) of numerical values
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)

# Allelic doses of 20 markers on a chromosome of 91 Vitis vinifera plants
Geno &lt;- data.test[[1]]
Geno

# Kinship matrix of 91 plants of Vitis vinifera
V.WAIS &lt;- data.test[[2]]
V.WAIS

# Structure population matrix of 91 plants of Vitis vinifera 
# in two sub-populations
S.2POP &lt;- data.test[[3]]
S.2POP
</code></pre>

<hr>
<h2 id='Info.Locus'>Information on loci</h2><span id='topic+Info.Locus'></span>

<h3>Description</h3>

<p>For a locus, this function computes the minor allelic frequency, the frequency of heterozygous genotypes and the missing value frequency.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Info.Locus(locus,data="G")</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Info.Locus_+3A_locus">locus</code></td>
<td>
<p>Numeric vector of allelic doses.</p>
</td></tr>
<tr><td><code id="Info.Locus_+3A_data">data</code></td>
<td>
<p>Value equal to &quot;G&quot; or &quot;H&quot; depending on the type of data (Genotype or Haplotype).
Default value is &quot;G&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is a numeric vector of three values which are respectively 
the minor allelic frequency, the  frequency of heterozygous genotypes 
(NA if haplotype data) and the missing value frequency.
</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
Geno &lt;- data.test[[1]]
info &lt;- apply(Geno, 2, Info.Locus)
info
</code></pre>

<hr>
<h2 id='Inv.proj.matrix.sdp'>Inv.proj.matrix.sdp</h2><span id='topic+Inv.proj.matrix.sdp'></span>

<h3>Description</h3>

<p>This function computes the Moore-Penrose pseudo-inverse of a symetric matrix. 
A single value decomposition is performed, the non positive eigen values are set to zero, then the pseudo-inverse is computed.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Inv.proj.matrix.sdp(matrix)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Inv.proj.matrix.sdp_+3A_matrix">matrix</code></td>
<td>
<p>symmetric matrix</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is the pseudo-inverse matrix.</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
V.WAIS &lt;- data.test[[2]]
Inv.V.WAIS &lt;- Inv.proj.matrix.sdp(V.WAIS)
Inv.V.WAIS
</code></pre>

<hr>
<h2 id='LD.Measures'>LD Measures</h2><span id='topic+LD.Measures'></span>

<h3>Description</h3>

<p>This function estimates for a set of loci:
</p>
<p>- the usual measure of linkage disequilibrium (<code class="reqn">r^2</code>)
</p>
<p>- the measure of linkage disequilibrium corrected by the structure of the sample (<code class="reqn">r^2_S</code>).
</p>
<p>- the measure of linkage disequilibrium corrected by the relatedness of genotyped individuals (<code class="reqn">r^2_V</code>).
</p>
<p>- the measure of linkage disequilibrium corrected by both, the relatedness of genotyped individuals and the structure of the sample  (<code class="reqn">r^2_{VS}</code>).
</p>
<p>This function gives extra informations on the studied loci.</p>


<h3>Usage</h3>

<pre><code class='language-R'>LD.Measures(donnees, V = NA, S = NA, data = "G", supinfo = FALSE, na.presence=TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LD.Measures_+3A_donnees">donnees</code></td>
<td>
<p>Numeric matrix (N x M), where N is the number of genotypes (or haplotypes) and M is the number of markers.
Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Row names correspond to the ID of individuals.
Column names correspond to the ID of markers.
Missing values are allowed.</p>
</td></tr>
<tr><td><code id="LD.Measures_+3A_v">V</code></td>
<td>
<p>Numeric matrix (N x N), where N is the number of genotypes (or haplotypes).
Matrix values are coefficients of genetic variance-covariance between every pair of individuals.
Row and column names must correspond to the ID of individuals.
No missing value.</p>
</td></tr>
<tr><td><code id="LD.Measures_+3A_s">S</code></td>
<td>
<p>Numeric matrix (N x (1-P)), where N is the number of genotypes (or haplotypes) and P the number of sub-populations.
Matrix values are the probabilities (between 0 and 1) for each genotype (or haplotype) to belong to each sub-populations.
Row names must correspond to the ID of individuals.
Column names correspond to the ID of sub-populations.
The matrix must be inversible, if the structure is with P sub-populations, only P-1 columns are expected.
No missing value.</p>
</td></tr>
<tr><td><code id="LD.Measures_+3A_data">data</code></td>
<td>
<p>Value equal to &quot;G&quot; or &quot;H&quot; depending on the type of data (Genotype or Haplotype).
Default value is &quot;G&quot;.</p>
</td></tr>
<tr><td><code id="LD.Measures_+3A_supinfo">supinfo</code></td>
<td>
<p>Boolean indicating whether you wish to get information about the loci.
If supinfo=TRUE, for each locus, the Minor Allelic Frequency (MAF), the frequency of heterozygous genotypes (only if the data are genotypes) and the missing value frequency are computed.
By default, supinfo=FALSE.</p>
</td></tr>
<tr><td><code id="LD.Measures_+3A_na.presence">na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
By default, na.presence=TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is a dataframe of size (M(M-1))/2 rows and C columns, where M is the number of markers and C is a number between 3 and 12 depending on options chosen by user.
The first three columns contain respectively the name of the first marker, the name of the second marker and the estimated value of the usual measure of linkage disequilibrium (<code class="reqn">r^2</code>) between these two markers.
</p>
<p>If only V is different from NA, the fourth column contains the estimated value of the measure of linkage disequilibrium corrected by the relatedness of genotyped individuals (<code class="reqn">r^2_V</code>).
</p>
<p>If only S is different from NA, the fourth column contains the estimated value of the measure of linkage disequilibrium corrected by relatedness corrected by the structure of the sample (<code class="reqn">r^2_S</code>).
</p>
<p>If V and S are simultaneously different from NA, the fourth, fifth and sixth columns respectively contain the estimated values of <code class="reqn">r^2_V</code>, <code class="reqn">r^2_S</code> and <code class="reqn">r^2_{VS}</code> (<code class="reqn">r^2</code> corrected by both the relatedness of genotyped individuals and the structure of the sample).
</p>
<p>If Supinfo=TRUE, then the last six columns contain information for both loci : the MAF, the frequency of heterozygous genotype (NA if haplotype data) and the missing value frequency.
</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Mangin, B., Siberchicot, A., Nicolas, S., Doligez, A., This, P., Cierco-Ayrolles, C. (2012). Novel measures of linkage disequilibrium that correct the bias due to population structure and relatedness. Heredity, 108 (3), 285-291. DOI: 10.1038/hdy.2011.73</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
Geno &lt;- data.test[[1]]
V.WAIS &lt;- data.test[[2]]
S.2POP &lt;- data.test[[3]]

LD &lt;- LD.Measures(Geno, V = V.WAIS, S = S.2POP, data = "G", supinfo = TRUE, na.presence = TRUE)
head(LD)
</code></pre>

<hr>
<h2 id='Measure.R2'>r^2 measure</h2><span id='topic+Measure.R2'></span>

<h3>Description</h3>

<p>This function estimates the usual measure of linkage disequilibrium (<code class="reqn">r^2</code>)  between two loci.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Measure.R2(biloci, na.presence = TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Measure.R2_+3A_biloci">biloci</code></td>
<td>
<p>Numeric matrix (N x 2), where N is the number of genotypes (or haplotypes)
</p>
<p>Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Rows names correspond to the ID of individuals.
</p>
<p>Columns names correspond to the ID of markers.</p>
</td></tr>
<tr><td><code id="Measure.R2_+3A_na.presence">na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
</p>
<p>If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
</p>
<p>By default, na.presence=TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is the estimated value of the usual measure of linkage disequilibrium (<code class="reqn">r^2</code>) 
or NA if less than 5 individuals have non-missing data at both loci
</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Hill, W.G, Robertson, A. (1968). Linkage diseqilibrium in finite populations. Theoretical and Applied Genetics, 38, 226-231. DOI: 10.1007/BF01245622</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
Geno &lt;- data.test[[1]]
Measure.R2(Geno)
</code></pre>

<hr>
<h2 id='Measure.R2S'>r^2_S measure</h2><span id='topic+Measure.R2S'></span>

<h3>Description</h3>

<p>This function estimates the novel measure of linkage disequilibrium which is corrected by the structure of the sample.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Measure.R2S(biloci, struc, na.presence=TRUE)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Measure.R2S_+3A_biloci">biloci</code></td>
<td>
<p>Numeric matrix (N x 2), where N is the number of genotypes (or haplotypes)
</p>
<p>Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Row names correspond to the  ID of individuals.
</p>
<p>Column names correspond to the ID of markers.</p>
</td></tr>
<tr><td><code id="Measure.R2S_+3A_struc">struc</code></td>
<td>
<p>Numeric matrix (N x (P-1)), where N is the number of genotypes (or haplotypes) and P the number of sub-populations.
</p>
<p>Matrix values are the probabilities for each genotypes (or haplotypes) to belong to each sub-populations.
</p>
<p>Row names must correspond to the ID of individuals and must be ranged as in the biloci matrix.
</p>
<p>Column names correspond to the ID of sub-populations.
</p>
<p>The matrix must be inversible, if the structure is with P sub-populations, only P-1 columns are expected.
</p>
<p>No missing value.</p>
</td></tr>
<tr><td><code id="Measure.R2S_+3A_na.presence">na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
</p>
<p>If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
</p>
<p>By default, na.presence=TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is the estimated value of the measure of linkage disequilibrium  corrected by  the structure of the sample or NA if less than 5 individuals have non-missing data at both loci.</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Mangin, B., Siberchicot, A., Nicolas, S., Doligez, A., This, P., Cierco-Ayrolles, C. (2012). Novel measures of linkage disequilibrium that correct the bias due to population structure and relatedness. Heredity, 108 (3), 285-291. DOI: 10.1038/hdy.2011.73</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
Geno &lt;- data.test[[1]]
S.2POP &lt;- data.test[[3]]
Measure.R2S(Geno, S.2POP)
</code></pre>

<hr>
<h2 id='Measure.R2V'>r^2_V measure</h2><span id='topic+Measure.R2V'></span>

<h3>Description</h3>

<p>This function estimates the novel measure of linkage disequilibrium which is corrected by the relatedness of genotyped individuals.</p>


<h3>Usage</h3>

<pre><code class='language-R'>Measure.R2V(biloci, V, na.presence=TRUE, V_inv=NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Measure.R2V_+3A_biloci">biloci</code></td>
<td>
<p>Numeric matrix (N x 2), where N is the number of genotypes (or haplotypes).
</p>
<p>Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Row names correspond to the ID of individuals.
</p>
<p>Column names correspond to the ID of markers.</p>
</td></tr>
<tr><td><code id="Measure.R2V_+3A_v">V</code></td>
<td>
<p>Numeric matrix (N x N), where N is the number of genotypes (or haplotypes).
</p>
<p>Matrix values are coefficients of genetic covariance for each pair of individuals.
</p>
<p>Rows and columns names must correspond to the ID of individuals and must be ranged in the same order as in the biloci matrix.
</p>
<p>No missing value.</p>
</td></tr>
<tr><td><code id="Measure.R2V_+3A_na.presence">na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
</p>
<p>If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
</p>
<p>By default, na.presence=TRUE.</p>
</td></tr>
<tr><td><code id="Measure.R2V_+3A_v_inv">V_inv</code></td>
<td>
<p>Should stay NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is the estimated value of the measure of linkage disequilibrium corrected by the relatedness of genotyped individuals or NA if less than 5 individuals have non-missing data at both loci.</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Mangin, B., Siberchicot, A., Nicolas, S., Doligez, A., This, P., Cierco-Ayrolles, C. (2012). Novel measures of linkage disequilibrium that correct the bias due to population structure and relatedness. Heredity, 108 (3), 285-291. DOI: 10.1038/hdy.2011.73</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
Geno &lt;- data.test[[1]]
V.WAIS &lt;- data.test[[2]]
Measure.R2V(Geno, V.WAIS)
</code></pre>

<hr>
<h2 id='Measure.R2VS'>r^2_VS measure</h2><span id='topic+Measure.R2VS'></span>

<h3>Description</h3>

<p>This function estimates the novel measure of linkage disequilibrium which is corrected by  both the relatedness of genotyped individuals and  the structure of the sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Measure.R2VS(biloci, V, struc, na.presence = TRUE, V_inv = NULL)</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Measure.R2VS_+3A_biloci">biloci</code></td>
<td>
<p>Numeric matrix (N x 2), where N is the number of genotypes (or haplotypes)
</p>
<p>Matrix values are the allelic doses:
</p>
<p>- (0,1,2) for genotypes.
</p>
<p>- (0,1) for haplotypes.
</p>
<p>Row names correspond to the ID of individuals.
</p>
<p>Column names correspond to the ID of markers.
</p>
</td></tr>
<tr><td><code id="Measure.R2VS_+3A_v">V</code></td>
<td>
<p>Numeric matrix (N x N), where N is the number of genotypes (or haplotypes).
</p>
<p>Matrix values are coefficients of genetic variance-covariance for every pair of individuals.
Row and column names must correspond to the ID of individuals and must be ranged as in the biloci matrix.
</p>
<p>No missing value.
</p>
</td></tr>
<tr><td><code id="Measure.R2VS_+3A_struc">struc</code></td>
<td>
<p>Numeric matrix (N x (P-1)), where N is the number of genotypes (or haplotypes) and P the number of sub-populations.
</p>
<p>Matrix values are the probabilities (between 0 and 1)
for each genotypes (or haplotypes) to belong to each sub-populations.
</p>
<p>Row names must correspond to the ID of individuals and must be ranged as in the biloci matrix.
</p>
<p>Column names correspond to the ID of sub-populations.
</p>
<p>The matrix must be inversible, if the structure is with P sub-populations, only P-1 columns are expected.
</p>
<p>No missing value.</p>
</td></tr>
<tr><td><code id="Measure.R2VS_+3A_na.presence">na.presence</code></td>
<td>
<p>Boolean indicating the presence of missing values in data.
</p>
<p>If na.presence=FALSE (no missing data), computation of <code class="reqn">r^2_V</code> and <code class="reqn">r^2_{VS}</code> is largely optimized.
</p>
<p>By default, na.presence=TRUE.</p>
</td></tr>
<tr><td><code id="Measure.R2VS_+3A_v_inv">V_inv</code></td>
<td>
<p>Should stay NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The returned value is the estimated value of the linkage disequilibrium measure corrected by both the relatedness of genotyped individuals and  the structure of the sample or NA if less than 5 individuals have non-missing data at both loci.
</p>


<h3>Author(s)</h3>

<p>David Desrousseaux, Florian Sandron, Aurélie Siberchicot, Christine Cierco-Ayrolles and Brigitte Mangin</p>


<h3>References</h3>

<p>Mangin, B., Siberchicot, A., Nicolas, S., Doligez, A., This, P., Cierco-Ayrolles, C. (2012). Novel measures of linkage disequilibrium that correct the bias due to population structure and relatedness. Heredity, 108 (3), 285-291. DOI: 10.1038/hdy.2011.73</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.test)
Geno &lt;- data.test[[1]]
V.WAIS &lt;- data.test[[2]]
S.2POP &lt;- data.test[[3]]
Measure.R2VS(Geno, V.WAIS, S.2POP)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
