<!DOCTYPE html><html><head><title>Help for package LDlinkR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LDlinkR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LDexpress'><p>Determine if genomic variants are associated with gene expression.</p></a></li>
<li><a href='#LDhap'><p>Calculates population specific haplotype frequencies of all</p>
haplotypes observed for a list of query variants.</a></li>
<li><a href='#LDlinkR-package'><p>LDlinkR: Calculating Linkage Disequilibrium (LD) in Human Population Groups of Interest</p></a></li>
<li><a href='#LDmatrix'><p>Generates a data frame of pairwise linkage disequilibrium</p>
statistics.</a></li>
<li><a href='#LDpair'><p>Investigates potentially correlated alleles for a</p>
pair of variants.</a></li>
<li><a href='#LDpop'><p>Investigates allele frequencies and linkage disequilibrium</p>
patterns across 1000 Genomes Project populations.</a></li>
<li><a href='#LDproxy'><p>Explore proxy and putative functional variants for a</p>
single query variant.</a></li>
<li><a href='#LDproxy_batch'><p>Query LDproxy using a list of query variants, one per line.</p></a></li>
<li><a href='#LDtrait'><p>Determine if genomic variants are associated with a trait or disease.</p></a></li>
<li><a href='#list_chips'><p>Provides a data frame listing the names and abbreviation codes</p>
for available commercial SNP Chip Arrays from Illumina and Affymetrix.</a></li>
<li><a href='#list_gtex_tissues'><p>Provides a data frame listing the GTEx full names, 'LDexpress' full names</p>
(without spaces) and acceptable abbreviation codes of the 54 non-diseased
tissue sites collected for the GTEx Portal and used as input for the
'LDexpress' function.</a></li>
<li><a href='#list_pop'><p>Provides a data frame listing the available reference populations from</p>
the 1000 Genomes Project.</a></li>
<li><a href='#SNPchip'><p>Find commercial genotyping chip arrays for variants of interest.</p></a></li>
<li><a href='#SNPclip'><p>Prune a list of variants by linkage disequilibrium.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculating Linkage Disequilibrium (LD) in Human Population
Groups of Interest</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Timothy A. Myers &lt;myersta@mail.nih.gov&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides access to the 'LDlink' API (<a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a>)
    using the R console.  This programmatic access facilitates researchers who are 
    interested in performing batch queries in 1000 Genomes Project (2015) &lt;<a href="https://doi.org/10.1038%2Fnature15393">doi:10.1038/nature15393</a>&gt; 
    data using 'LDlink'. 'LDlink' is an interactive and powerful suite of web-based tools for querying 
    germline variants in human population groups of interest. For more details, please see 
    Machiela et al. (2015) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv402">doi:10.1093/bioinformatics/btv402</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://ldlink.nih.gov">https://ldlink.nih.gov</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/CBIIT/LDlinkR/issues">https://github.com/CBIIT/LDlinkR/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>httr (&ge; 1.4.0), utils (&ge; 3.4.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, spelling</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-31 17:56:57 UTC; myersta</td>
</tr>
<tr>
<td>Author:</td>
<td>Timothy A. Myers <a href="https://orcid.org/0000-0001-8127-3446"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Stephen J. Chanock [aut],
  Mitchell J. Machiela
    <a href="https://orcid.org/0000-0001-6538-9705"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-31 18:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='LDexpress'>Determine if genomic variants are associated with gene expression.</h2><span id='topic+LDexpress'></span>

<h3>Description</h3>

<p>Search if a list of genomic variants (or variants in LD with those variants) is associated with gene
expression in tissues of interest. Quantitative trait loci data is downloaded from the
GTEx Portal (<a href="https://gtexportal.org/home/">https://gtexportal.org/home/</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDexpress(
  snps,
  pop = "CEU",
  tissue = "ALL",
  r2d = "r2",
  r2d_threshold = 0.1,
  p_threshold = 0.1,
  win_size = 5e+05,
  genome_build = "grch37",
  token = NULL,
  file = FALSE,
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDexpress_+3A_snps">snps</code></td>
<td>
<p>between 1 - 10 variants, using an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;.
Use the 'list_pop' function to see a list of available human reference populations.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_tissue">tissue</code></td>
<td>
<p>select from 1 - 54 non-diseased tissue sites collected for the GTEx project, multiple
allowed.  Acceptable user input is taken either from &quot;tissue_name_ldexpress&quot; or &quot;tissue_abbrev_ldexpress&quot;
(tissue abbreviation) code listed in available GTEx tissue sites using the
<code>list_getex_tissues()</code> function (e.g. &quot;ADI_SUB&quot; for Adipose Subcutaneous). Input is case sensitive.
Default = &quot;ALL&quot; for all available tissue types.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_r2d">r2d</code></td>
<td>
<p>either &quot;r2&quot; for LD R2 or &quot;d&quot; for LD D', default = &quot;r2&quot;.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_r2d_threshold">r2d_threshold</code></td>
<td>
<p>R2 or D' (depends on 'r2d' user input parameter) threshold for LD filtering. Any variants
within -/+ of the specified genomic window and R^2 or D' less than the threshold will be removed. Value needs
to be in the range 0 to 1. Default value is 0.1.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_p_threshold">p_threshold</code></td>
<td>
<p>define the eQTL significance threshold used for returning query results. Default value
is 0.1 which returns all GTEx eQTL associations with P-value less than 0.1.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_win_size">win_size</code></td>
<td>
<p>set genomic window size for LD calculation. Specify a value greater than or equal to zero and less than or
equal to 1,000,000 basepairs (bp). Default value is -/+ 500,000bp.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDexpress_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDexpress_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of all query variant RS numbers, respective QTL which are in LD with query variant,
and associated gene expression.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDexpress(snps = c("rs345", "rs456"),
                   pop = c("YRI", "CEU"),
                   tissue = c("ADI_SUB", "ADI_VIS_OME"),
                   r2d = "r2",
                   r2d_threshold = "0.1",
                   p_threshold = "0.1",
                   win_size = "500000",
                   genome_build = "grch37",
                   token = Sys.getenv("LDLINK_TOKEN")
                  )
         
## End(Not run)

</code></pre>

<hr>
<h2 id='LDhap'>Calculates population specific haplotype frequencies of all
haplotypes observed for a list of query variants.</h2><span id='topic+LDhap'></span>

<h3>Description</h3>

<p>Calculates population specific haplotype frequencies of all
haplotypes observed for a list of query variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDhap(
  snps,
  pop = "CEU",
  token = NULL,
  file = FALSE,
  table_type = "haplotype",
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDhap_+3A_snps">snps</code></td>
<td>
<p>list of between 1 - 30 variants, using an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDhap_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="LDhap_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDhap_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDhap_+3A_table_type">table_type</code></td>
<td>
<p>Choose from one of four options available to determine output
format type...'haplotype', 'variant', 'both' and 'merged'. Default = &quot;haplotype&quot;.</p>
</td></tr>
<tr><td><code id="LDhap_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDhap_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame or list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDhap(c("rs3", "rs4", "rs148890987"), "CEU", token = Sys.getenv("LDLINK_TOKEN"))
## Not run: LDhap("rs148890987", c("YRI", "CEU"), token = Sys.getenv("LDLINK_TOKEN"))

</code></pre>

<hr>
<h2 id='LDlinkR-package'>LDlinkR: Calculating Linkage Disequilibrium (LD) in Human Population Groups of Interest</h2><span id='topic+LDlinkR'></span><span id='topic+LDlinkR-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Provides access to the 'LDlink' API (<a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a>) using the R console. This programmatic access facilitates researchers who are interested in performing batch queries in 1000 Genomes Project (2015) <a href="https://doi.org/10.1038/nature15393">doi:10.1038/nature15393</a> data using 'LDlink'. 'LDlink' is an interactive and powerful suite of web-based tools for querying germline variants in human population groups of interest. For more details, please see Machiela et al. (2015) <a href="https://doi.org/10.1093/bioinformatics/btv402">doi:10.1093/bioinformatics/btv402</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Timothy A. Myers <a href="mailto:myersta@mail.nih.gov">myersta@mail.nih.gov</a> (<a href="https://orcid.org/0000-0001-8127-3446">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Stephen J. Chanock <a href="mailto:chanocks@mail.nih.gov">chanocks@mail.nih.gov</a>
</p>
</li>
<li><p> Mitchell J. Machiela <a href="mailto:mitchell.machiela@nih.gov">mitchell.machiela@nih.gov</a> (<a href="https://orcid.org/0000-0001-6538-9705">ORCID</a>)
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://ldlink.nih.gov">https://ldlink.nih.gov</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/CBIIT/LDlinkR/issues">https://github.com/CBIIT/LDlinkR/issues</a>
</p>
</li></ul>


<hr>
<h2 id='LDmatrix'>Generates a data frame of pairwise linkage disequilibrium
statistics.</h2><span id='topic+LDmatrix'></span>

<h3>Description</h3>

<p>Generates a data frame of pairwise linkage disequilibrium
statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDmatrix(
  snps,
  pop = "CEU",
  r2d = "r2",
  token = NULL,
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDmatrix_+3A_snps">snps</code></td>
<td>
<p>list of between 2 - 1,000 variants, using an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDmatrix_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="LDmatrix_+3A_r2d">r2d</code></td>
<td>
<p>r2d, either &quot;r2&quot; for LD R2 or &quot;d&quot; for LD D', default = &quot;r2&quot;</p>
</td></tr>
<tr><td><code id="LDmatrix_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDmatrix_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDmatrix_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDmatrix_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDmatrix(c("rs3", "rs4", "rs148890987"),
                  "YRI", "r2",
                  token = Sys.getenv("LDLINK_TOKEN"))
                 
## End(Not run)

</code></pre>

<hr>
<h2 id='LDpair'>Investigates potentially correlated alleles for a
pair of variants.</h2><span id='topic+LDpair'></span>

<h3>Description</h3>

<p>Investigates potentially correlated alleles for a
pair of variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDpair(
  var1,
  var2,
  pop = "CEU",
  token = NULL,
  output = "table",
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDpair_+3A_var1">var1</code></td>
<td>
<p>the first RS number or genomic coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDpair_+3A_var2">var2</code></td>
<td>
<p>the second RS number or genomic coordinate (e.g. &quot;ch7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDpair_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population(s), (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="LDpair_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDpair_+3A_output">output</code></td>
<td>
<p>two output options available, &quot;text&quot;, which displays a two-by-two matrix displaying haplotype counts and allele frequencies along with other statistics, or &quot;table&quot;, which displays the same data in rows and columns, default = &quot;table&quot;</p>
</td></tr>
<tr><td><code id="LDpair_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDpair_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDpair_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>text or data frame, depending on the output option
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDpair(var1 = "rs3", var2 = "rs4", pop = "YRI", token = Sys.getenv("LDLINK_TOKEN"))
## Not run: LDpair("rs3", "rs4", "YRI", token = Sys.getenv("LDLINK_TOKEN"), "text")

</code></pre>

<hr>
<h2 id='LDpop'>Investigates allele frequencies and linkage disequilibrium
patterns across 1000 Genomes Project populations.</h2><span id='topic+LDpop'></span>

<h3>Description</h3>

<p>Investigates allele frequencies and linkage disequilibrium
patterns across 1000 Genomes Project populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDpop(
  var1,
  var2,
  pop = "CEU",
  r2d = "r2",
  token = NULL,
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDpop_+3A_var1">var1</code></td>
<td>
<p>the first RS number or genomic coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDpop_+3A_var2">var2</code></td>
<td>
<p>the second RS number or genomic coordinate (e.g. &quot;ch7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="LDpop_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population(s), (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="LDpop_+3A_r2d">r2d</code></td>
<td>
<p>either &quot;r2&quot; for LD R2 or &quot;d&quot; for LD D', default = &quot;r2&quot;</p>
</td></tr>
<tr><td><code id="LDpop_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDpop_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDpop_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDpop_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDpop(var1 = "rs3", var2 = "rs4",
               pop = "YRI", r2d = "r2",
               token = Sys.getenv("LDLINK_TOKEN"))
             
## End(Not run)

</code></pre>

<hr>
<h2 id='LDproxy'>Explore proxy and putative functional variants for a
single query variant.</h2><span id='topic+LDproxy'></span>

<h3>Description</h3>

<p>Explore proxy and putative functional variants for a
single query variant.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDproxy(
  snp,
  pop = "CEU",
  r2d = "r2",
  token = NULL,
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDproxy_+3A_snp">snp</code></td>
<td>
<p>an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;), one per query</p>
</td></tr>
<tr><td><code id="LDproxy_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="LDproxy_+3A_r2d">r2d</code></td>
<td>
<p>either &quot;r2&quot; for LD R2 or &quot;d&quot; for LD D', default = &quot;r2&quot;</p>
</td></tr>
<tr><td><code id="LDproxy_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDproxy_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDproxy_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDproxy_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDproxy("rs456", "YRI", "r2", token = Sys.getenv("LDLINK_TOKEN"))

</code></pre>

<hr>
<h2 id='LDproxy_batch'>Query LDproxy using a list of query variants, one per line.</h2><span id='topic+LDproxy_batch'></span>

<h3>Description</h3>

<p>Query LDproxy using a list of query variants, one per line.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDproxy_batch(
  snp,
  pop = "CEU",
  r2d = "r2",
  token = NULL,
  append = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDproxy_batch_+3A_snp">snp</code></td>
<td>
<p>a character string or data frame listing rsID's or chromosome coordinates (e.g. &quot;chr7:24966446&quot;), one per line</p>
</td></tr>
<tr><td><code id="LDproxy_batch_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="LDproxy_batch_+3A_r2d">r2d</code></td>
<td>
<p>either &quot;r2&quot; for LD R2 or &quot;d&quot; for LD D', default = &quot;r2&quot;</p>
</td></tr>
<tr><td><code id="LDproxy_batch_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDproxy_batch_+3A_append">append</code></td>
<td>
<p>A logical. If TRUE, output for each query variant is appended to a text file. If FALSE, output of each query variant is saved in its own text file.
Default is FALSE.</p>
</td></tr>
<tr><td><code id="LDproxy_batch_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDproxy_batch_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>text file(s) are saved to the current working directory.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: snps_to_upload &lt;- c("rs3", "rs4")
## Not run: LDproxy_batch(snp = snps_to_upload, token = Sys.getenv("LDLINK_TOKEN"), append = FALSE)

</code></pre>

<hr>
<h2 id='LDtrait'>Determine if genomic variants are associated with a trait or disease.</h2><span id='topic+LDtrait'></span>

<h3>Description</h3>

<p>Search if a list of variants (or variants in LD with those variants) have been
previously associated with a trait or disease. Trait and disease data is updated
nightly from the GWAS Catalog (<a href="https://www.ebi.ac.uk/gwas/docs/file-downloads">https://www.ebi.ac.uk/gwas/docs/file-downloads</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LDtrait(
  snps,
  pop = "CEU",
  r2d = "r2",
  r2d_threshold = 0.1,
  win_size = 5e+05,
  token = NULL,
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LDtrait_+3A_snps">snps</code></td>
<td>
<p>between 1 - 50 variants, using an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;).
All input variants must match a bi-allelic variant.</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;.
Use the 'list_pop' function to see a list of available human reference populations.</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_r2d">r2d</code></td>
<td>
<p>use &quot;r2&quot; to filter desired output from a threshold based on estimated
LD R2 (R squared) or &quot;d&quot; for LD D' (D-prime), default = &quot;r2&quot;.</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_r2d_threshold">r2d_threshold</code></td>
<td>
<p>R2 or D' (depends on 'r2d' user input parameter) threshold for LD filtering. Any variants
within -/+ of the specified genomic window and R^2 or D' less than the threshold will be removed. Value needs
to be in the range 0 to 1. Default value is 0.1.</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_win_size">win_size</code></td>
<td>
<p>set genomic window size for LD calculation. Specify a value greater than or equal to zero and less than or
equal to 1,000,000bp. Default value is -/+ 500,000 bp.</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="LDtrait_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="LDtrait_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of all query variant RS numbers with a list of queried variants
in LD with a variant reported in the GWAS Catalog (<a href="https://www.ebi.ac.uk/gwas/docs/file-downloads">https://www.ebi.ac.uk/gwas/docs/file-downloads</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: LDtrait(snps = "rs456",
                 pop = c("YRI", "CEU"),
                 r2d = "r2",
                 r2d_threshold = "0.1",
                 win_size = "500000",
                 token = Sys.getenv("LDLINK_TOKEN")
                )
         
## End(Not run)

</code></pre>

<hr>
<h2 id='list_chips'>Provides a data frame listing the names and abbreviation codes
for available commercial SNP Chip Arrays from Illumina and Affymetrix.</h2><span id='topic+list_chips'></span>

<h3>Description</h3>

<p>Provides a data frame listing the names and abbreviation codes
for available commercial SNP Chip Arrays from Illumina and Affymetrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_chips()
</code></pre>


<h3>Value</h3>

<p>a data frame listing the names and abbreviation codes for available SNP Chip Arrays from Illumina and Affymetrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>list_chips()
</code></pre>

<hr>
<h2 id='list_gtex_tissues'>Provides a data frame listing the GTEx full names, 'LDexpress' full names
(without spaces) and acceptable abbreviation codes of the 54 non-diseased
tissue sites collected for the GTEx Portal and used as input for the
'LDexpress' function.</h2><span id='topic+list_gtex_tissues'></span>

<h3>Description</h3>

<p>Provides a data frame listing the GTEx full names, 'LDexpress' full names
(without spaces) and acceptable abbreviation codes of the 54 non-diseased
tissue sites collected for the GTEx Portal and used as input for the
'LDexpress' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_gtex_tissues()
</code></pre>


<h3>Value</h3>

<p>a data frame listing the GTEx tissues, their names and abbreviation codes
used as input for LDexpress.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>list_gtex_tissues()
</code></pre>

<hr>
<h2 id='list_pop'>Provides a data frame listing the available reference populations from
the 1000 Genomes Project.</h2><span id='topic+list_pop'></span>

<h3>Description</h3>

<p>Provides a data frame listing the available reference populations from
the 1000 Genomes Project.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_pop()
</code></pre>


<h3>Value</h3>

<p>a data frame listing the available reference populations, continental (ex: European, African, and Admixed American) and sub-populations (ex: Finnish, Gambian, and Peruvian)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>list_pop()

</code></pre>

<hr>
<h2 id='SNPchip'>Find commercial genotyping chip arrays for variants of interest.</h2><span id='topic+SNPchip'></span>

<h3>Description</h3>

<p>Find commercial genotyping chip arrays for variants of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPchip(
  snps,
  chip = "ALL",
  token = NULL,
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPchip_+3A_snps">snps</code></td>
<td>
<p>between 1 - 5,000 variants, using an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="SNPchip_+3A_chip">chip</code></td>
<td>
<p>chip or arrays, platform code(s) for a SNP chip array, ALL_Illumina, ALL_Affy or ALL, default=ALL</p>
</td></tr>
<tr><td><code id="SNPchip_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at  <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="SNPchip_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="SNPchip_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="SNPchip_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: SNPchip(c("rs3", "rs4", "rs148890987"), "ALL",
                 token = Sys.getenv("LDLINK_TOKEN"))
               
## End(Not run)
## Not run: SNPchip(c("rs3", "rs4", "rs148890987"),
                 c("A_CHB2", "A_SNP5.0"),
                 token = Sys.getenv("LDLINK_TOKEN"))
                 
## End(Not run)
## Not run: SNPchip("rs148890987", "ALL_Affy", token = Sys.getenv("LDLINK_TOKEN"))

</code></pre>

<hr>
<h2 id='SNPclip'>Prune a list of variants by linkage disequilibrium.</h2><span id='topic+SNPclip'></span>

<h3>Description</h3>

<p>Prune a list of variants by linkage disequilibrium.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNPclip(
  snps,
  pop = "CEU",
  r2_threshold = "0.1",
  maf_threshold = "0.01",
  token = NULL,
  file = FALSE,
  genome_build = "grch37",
  api_root = "https://ldlink.nih.gov/LDlinkRest"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNPclip_+3A_snps">snps</code></td>
<td>
<p>a list of between 1 - 5,000 variants, using an rsID or chromosome coordinate (e.g. &quot;chr7:24966446&quot;)</p>
</td></tr>
<tr><td><code id="SNPclip_+3A_pop">pop</code></td>
<td>
<p>a 1000 Genomes Project population, (e.g. YRI or CEU), multiple allowed, default = &quot;CEU&quot;</p>
</td></tr>
<tr><td><code id="SNPclip_+3A_r2_threshold">r2_threshold</code></td>
<td>
<p>LD R2 threshold between 0-1, default = 0.1</p>
</td></tr>
<tr><td><code id="SNPclip_+3A_maf_threshold">maf_threshold</code></td>
<td>
<p>minor allele frequency threshold between 0-1, default = 0.01</p>
</td></tr>
<tr><td><code id="SNPclip_+3A_token">token</code></td>
<td>
<p>LDlink provided user token, default = NULL, register for token at <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a></p>
</td></tr>
<tr><td><code id="SNPclip_+3A_file">file</code></td>
<td>
<p>Optional character string naming a path and file for saving results.  If file = FALSE, no file will be generated, default = FALSE.</p>
</td></tr>
<tr><td><code id="SNPclip_+3A_genome_build">genome_build</code></td>
<td>
<p>Choose between one of the three options...'grch37' for genome build GRCh37 (hg19),
'grch38' for GRCh38 (hg38), or 'grch38_high_coverage' for GRCh38 High Coverage (hg38) 1000 Genome Project
data sets.  Default is GRCh37 (hg19).</p>
</td></tr>
<tr><td><code id="SNPclip_+3A_api_root">api_root</code></td>
<td>
<p>Optional alternative root url for API.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: SNPclip(c("rs3", "rs4", "rs148890987"), "YRI", "0.1", "0.01",
                    token = Sys.getenv("LDLINK_TOKEN"))
                 
## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
