<!DOCTYPE html><html lang="en"><head><title>Help for package LDM</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LDM}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adjust.data.by.covariates'><p>Adjusting data (distance matrix and OTU table) by covariates</p></a></li>
<li><a href='#avgdist.squared'><p>Averaging the squared distance matrices each calculated from a rarefied OTU table</p></a></li>
<li><a href='#jaccard.mean'><p>Expected value of the Jaccard distance matrix</p></a></li>
<li><a href='#ldm'><p>Testing hypotheses about the microbiome using a linear decomposition model (LDM)</p></a></li>
<li><a href='#permanovaFL'><p>PERMANOVA test of association based on the Freedman-Lane permutation scheme</p></a></li>
<li><a href='#sim.meta'><p>Metadata of the simulated microbiome samples</p></a></li>
<li><a href='#sim.otu.tab'><p>OTU count table of the simulated microbiome samples</p></a></li>
<li><a href='#sim.otu.tab5'><p>Filtered OTU count table of the simulated microbiome samples</p></a></li>
<li><a href='#throat.meta'><p>Metadata of the throat microbiome samples</p></a></li>
<li><a href='#throat.otu.tab'><p>OTU count table from 16S sequencing of the throat microbiome samples</p></a></li>
<li><a href='#throat.otu.tab5'><p>Filtered OTU count table from 16S sequencing of the throat microbiome samples</p></a></li>
<li><a href='#throat.tree'><p>UPGMA tree of the OTUs from 16S sequencing of the throat microbiome samples</p></a></li>
<li><a href='#unifrac.mean'><p>Expected value of the unweighted UniFrac distance matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Testing Hypotheses About the Microbiome using the Linear
Decomposition Model</td>
</tr>
<tr>
<td>Version:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-07</td>
</tr>
<tr>
<td>Description:</td>
<td>A single analysis path that includes distance-based ordination, global tests of any effect of the microbiome, and tests of the effects of individual taxa with false-discovery-rate (FDR) control. It accommodates both continuous and discrete covariates as well as interaction terms to be tested either singly or in combination, allows for adjustment of confounding covariates, and uses permutation-based p-values that can control for sample correlations. It can be applied to transformed data, and an omnibus test can combine results from analyses conducted on different transformation scales. It can also be used for testing presence-absence associations based on infinite number of rarefaction replicates, testing mediation effects of the microbiome, analyzing censored time-to-event outcomes, and for compositional analysis by fitting linear models to centered-log-ratio taxa count data. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>GUniFrac, vegan, permute, parallel, BiocParallel, matrixStats,
castor, phangorn, modeest</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.rsp, testthat, survival</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/yijuanhu/LDM">https://github.com/yijuanhu/LDM</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/yijuanhu/LDM/issues">https://github.com/yijuanhu/LDM/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-07 08:15:26 UTC; yhu30</td>
</tr>
<tr>
<td>Author:</td>
<td>Yi-Juan Hu [aut, cre],
  Glen A Satten [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-07 08:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='adjust.data.by.covariates'>Adjusting data (distance matrix and OTU table) by covariates</h2><span id='topic+adjust.data.by.covariates'></span>

<h3>Description</h3>

<p>This function produces adjusted distance matrix and OTU table (if provided) 
after removing the effects of covariates (e.g., confounders). 
Observations with any missing data are removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adjust.data.by.covariates(
  formula = NULL,
  data = .GlobalEnv,
  otu.table = NULL,
  tree = NULL,
  dist.method = "bray",
  binary = FALSE,
  dist = NULL,
  square.dist = TRUE,
  center.dist = TRUE,
  scale.otu.table = TRUE,
  center.otu.table = TRUE,
  freq.scale.only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="adjust.data.by.covariates_+3A_formula">formula</code></td>
<td>
<p>a symbolic description of the covariate model in the form <code> ~ model</code>, 
where <code>model</code> is specified in the same way as for <code>lm</code> or <code>glm</code>. For example, 
<code>~ a + b</code> specifies a model with the main effects of covariates <code>a</code> and <code>b</code>, and 
<code>~ a*b</code>, equivalently <code>~ a + b + a:b</code>, specifies a model with the main effects of 
<code>a</code> and <code>b</code> as well as their interaction.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible 
by as.data.frame to a data frame) containing the covariates. 
If not found in <code>data</code>, the covariates are taken from environment (formula), 
typically the environment from which <code>adjust.data.by.covariates</code> is called. 
The default is .GlobalEnv.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_otu.table">otu.table</code></td>
<td>
<p>the <code>n.obs</code> by <code>n.otu</code> matrix of read counts. 
If provided, an adjusted (and column-centered) OTU table at the frequency (i.e., relative abundance) scale 
and an adjusted (and columnn-centered) OTU table at the arcsin-root-transformed frequency scale are output. If provided, 
it is also used for calculating the distance matrix unless the distance matrix is directly 
imported through <code>dist</code>.
The default is NULL.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree. Only used for calculating a
phylogenetic-tree-based distance matrix. Not needed if the calculation of
requested distance does not require a phylogenetic tree, or if the distance
matrix is directly imported through <code>dist</code>. The default is NULL.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_dist.method">dist.method</code></td>
<td>
<p>method for calculating the distance measure, partial
match to all methods supported by <code>vegdist</code> in the <code>vegan</code> package
(i.e., &quot;manhattan&quot;, &quot;euclidean&quot;, &quot;canberra&quot;, &quot;bray&quot;, &quot;kulczynski&quot;, &quot;jaccard&quot;, &quot;gower&quot;, 
&quot;altGower&quot;, &quot;morisita&quot;, &quot;horn&quot;, &quot;mountford&quot;, &quot;raup&quot; , &quot;binomial&quot;, &quot;chao&quot;, &quot;cao&quot;, &quot;mahalanobis&quot;)
as well as &quot;hellinger&quot; and &quot;wt-unifrac&quot;. 
The default is &quot;bray&quot;. 
For more details, see the <code>dist.method</code> argument in the <code>ldm</code> function.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_binary">binary</code></td>
<td>
<p>the &quot;binary&quot; parameter in <code>vegdist</code>. The default is FALSE.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_dist">dist</code></td>
<td>
<p>a distance matrix. Can be either an object of class &quot;dist&quot; or &quot;matrix&quot;.
The elements of the distance matrix will be squared and then the matrix will be centered if the default choices 
<code>square.dist=TRUE</code> and <code>center.dist=TRUE</code> are used. If <code>dist=NULL</code>, the distance matrix is 
calculated from the <code>otu.table</code>, using the value of <code>dist.method</code> (and <code>tree</code> if required). 
The default is NULL.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_square.dist">square.dist</code></td>
<td>
<p>a logical variable indicating whether to square the 
distance matrix. The default is TRUE.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_center.dist">center.dist</code></td>
<td>
<p>a logical variable indicating whether to center the 
distance matrix as described by Gower (1966). The default is TRUE.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_scale.otu.table">scale.otu.table</code></td>
<td>
<p>a logical variable indicating whether to scale the rows of the OTU table 
for the frequency scale.  For count data, this corresponds to dividing by the library size to give 
relative frequencies. The default is TRUE.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_center.otu.table">center.otu.table</code></td>
<td>
<p>a logical variable indicating whether to center the 
columns of the OTU table. The OTU table should be centered if the distance 
matrix has been centered. Applied to both OTU tables at frequency and transformed scales. The default is TRUE.</p>
</td></tr>
<tr><td><code id="adjust.data.by.covariates_+3A_freq.scale.only">freq.scale.only</code></td>
<td>
<p>a logical variable indicating whether to provide adjusted frequency-scale OTU table only 
(not adjusted OTU table at the arcsin-root transformed frequency scale). The default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list consisting of 
</p>
<table role = "presentation">
<tr><td><code>adj.dist</code></td>
<td>
<p>the (squared/centered) distance matrix
after adjustment of covariates.</p>
</td></tr>
<tr><td><code>y.freq</code></td>
<td>
<p>the (column-centered) frequency-scale OTU table after adjustment of covariates.</p>
</td></tr> 
<tr><td><code>y.tran</code></td>
<td>
<p>the (column-centered) arcsin-root-transformed 
OTU table after adjustment of covariates.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;, Glen A. Satten &lt;gsatten@emory.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>adj.data &lt;- adjust.data.by.covariates(formula= ~ Sex + AntibioticUse, data=throat.meta,
                                      otu.table=throat.otu.tab5, dist.method="bray")
</code></pre>

<hr>
<h2 id='avgdist.squared'>Averaging the squared distance matrices each calculated from a rarefied OTU table</h2><span id='topic+avgdist.squared'></span>

<h3>Description</h3>

<p>This function computes a distance matrix for each rarefied OTU table, square the distance matrix (in an element-wise manner), 
and then average the squared distance matrices.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>avgdist.squared(
  otu.table,
  dist.method = "jaccard",
  tree = NULL,
  scale.otu.table = FALSE,
  n.rarefy = 100,
  binary = TRUE,
  seed = 123
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="avgdist.squared_+3A_otu.table">otu.table</code></td>
<td>
<p>the <code>n.obs</code> by <code>n.otu</code> matrix of read counts.</p>
</td></tr>
<tr><td><code id="avgdist.squared_+3A_dist.method">dist.method</code></td>
<td>
<p>method for calculating the distance measure, partial
match to all methods supported by <code>vegdist</code> in the <code>vegan</code> package. The default is &quot;jaccard&quot;. 
For more details, see the <code>dist.method</code> argument in the <code>ldm</code> function.</p>
</td></tr>
<tr><td><code id="avgdist.squared_+3A_tree">tree</code></td>
<td>
<p>the phylogeneic tree. The default is NULL.</p>
</td></tr>
<tr><td><code id="avgdist.squared_+3A_scale.otu.table">scale.otu.table</code></td>
<td>
<p>a logical variable indicating whether to scale the rows of the OTU table. 
For count data, this corresponds to dividing by the library size to give relative frequencies. 
The default is FALSE.</p>
</td></tr>
<tr><td><code id="avgdist.squared_+3A_n.rarefy">n.rarefy</code></td>
<td>
<p>number of rarefactions. The default is 100.</p>
</td></tr>
<tr><td><code id="avgdist.squared_+3A_binary">binary</code></td>
<td>
<p>the &quot;binary&quot; parameter in <code>vegdist</code>. The default is TRUE.</p>
</td></tr>
<tr><td><code id="avgdist.squared_+3A_seed">seed</code></td>
<td>
<p>a single-value integer seed for the random process of drawing rarefaction replicates. 
The seed is user supplied or internally generated. The default is 123.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single matrix object
</p>
<table role = "presentation">
<tr><td><code>D2.avg</code></td>
<td>
<p>The average of the squared distance matrices.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;, Glen A. Satten &lt;gsatten@emory.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>dist.avg.D2 &lt;- avgdist.squared(throat.otu.tab5, dist.method="jaccard", n.rarefy=100)
</code></pre>

<hr>
<h2 id='jaccard.mean'>Expected value of the Jaccard distance matrix</h2><span id='topic+jaccard.mean'></span>

<h3>Description</h3>

<p>This function computes the expected value of the Jaccard distance matrix over rarefaction replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jaccard.mean(
  otu.table,
  rarefy.depth = min(rowSums(otu.table)),
  first.order.approx.only = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jaccard.mean_+3A_otu.table">otu.table</code></td>
<td>
<p>the <code>n.obs</code> by <code>n.otu</code> matrix of read counts.</p>
</td></tr>
<tr><td><code id="jaccard.mean_+3A_rarefy.depth">rarefy.depth</code></td>
<td>
<p>rarefaction depth. The default is the minimum library size observed in the OTU table.</p>
</td></tr>
<tr><td><code id="jaccard.mean_+3A_first.order.approx.only">first.order.approx.only</code></td>
<td>
<p>a logical value indicating whether to calculate the expected value 
using the first order approixmation by the delta method. 
The default is FALSE, using the second order approixmation.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list consisting of 
</p>
<table role = "presentation">
<tr><td><code>jac.mean.o1</code></td>
<td>
<p>Expected Jaccard distance matrix by the first order approixmation.</p>
</td></tr>
<tr><td><code>jac.mean.o2</code></td>
<td>
<p>Expected Jaccard distance matrix by the second order approixmation.</p>
</td></tr> 
<tr><td><code>jac.mean.sq.o1</code></td>
<td>
<p>Expected squared Jaccard distance matrix by the first order approixmation.</p>
</td></tr> 
<tr><td><code>jac.mean.sq.o2</code></td>
<td>
<p>Expected squared Jaccard distance matrix by the second order approixmation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;, Glen A. Satten &lt;gsatten@emory.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>res.jaccard &lt;- jaccard.mean( throat.otu.tab5 )
</code></pre>

<hr>
<h2 id='ldm'>Testing hypotheses about the microbiome using a linear decomposition model (LDM)</h2><span id='topic+ldm'></span>

<h3>Description</h3>

<p>This function allows you to 
1. simultaneously test the global association with the overall  
microbiome composition and individual OTU associations to give coherent 
results;
2. test hypotheses based on data at both the frequency (i.e., relative abundance) and arcsine-root-transformed frequency scales, 
and perform an &ldquo;omnibus&quot; test that combines results from analyses conducted on the two scales;
3. test presence-absence associations based on infinite number of rarefaction replicates;
4. handle complex design features such as 
confounders, interactions, and clustered data (with between- and within-cluster covariates);
5. test associations with a survival outcome (i.e., censored survival times);
6. perform mediation analysis of the microbiome;
7. perform the omnibus test LDM-omni3 that combines results from analyses conducted on the frequency, arcsine-root-transformed, and presence-absence scales.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldm(
  formula,
  other.surv.resid = NULL,
  data = .GlobalEnv,
  tree = NULL,
  dist.method = "bray",
  dist = NULL,
  cluster.id = NULL,
  strata = NULL,
  how = NULL,
  perm.within.type = "free",
  perm.between.type = "none",
  perm.within.ncol = 0,
  perm.within.nrow = 0,
  n.perm.max = NULL,
  n.rej.stop = 100,
  seed = NULL,
  fdr.nominal = 0.1,
  square.dist = TRUE,
  scale.otu.table = TRUE,
  center.otu.table = TRUE,
  freq.scale.only = FALSE,
  binary = FALSE,
  n.rarefy = 0,
  test.mediation = FALSE,
  test.omni3 = FALSE,
  comp.anal = FALSE,
  comp.anal.adjust = "median",
  n.cores = 4,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ldm_+3A_formula">formula</code></td>
<td>
<p>a symbolic description of the 
model to be fitted. The details of model specification are given under 
&quot;Details&quot;.</p>
</td></tr>
<tr><td><code id="ldm_+3A_other.surv.resid">other.surv.resid</code></td>
<td>
<p>a vector of data, usually the Martingale or deviance residuals from fitting the Cox model to the survival outcome (if it is the outcome of interest) and other covariates.</p>
</td></tr>
<tr><td><code id="ldm_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible 
by as.data.frame to a data frame) containing the covariates of interest and 
confounding covariates. 
If not found in <code>data</code>, the covariates are taken from environment(formula), 
typically the environment from which <code>ldm</code> is called. The default is .GlobalEnv.</p>
</td></tr>
<tr><td><code id="ldm_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree. Only used for calculating a 
phylogenetic-tree-based distance matrix. Not needed if the calculation of 
the requested distance does not involve a phylogenetic tree, or if the 
distance matrix is directly imported through <code>dist</code>.</p>
</td></tr>
<tr><td><code id="ldm_+3A_dist.method">dist.method</code></td>
<td>
<p>method for calculating the distance measure, partial
match to all methods supported by <code>vegdist</code> in the <code>vegan</code> package
(i.e., &quot;manhattan&quot;, &quot;euclidean&quot;, &quot;canberra&quot;, &quot;bray&quot;, &quot;kulczynski&quot;, &quot;jaccard&quot;, 
&quot;gower&quot;, &quot;altGower&quot;, &quot;morisita&quot;, &quot;horn&quot;, &quot;mountford&quot;, &quot;raup&quot; , &quot;binomial&quot;, 
&quot;chao&quot;, &quot;cao&quot;, &quot;mahalanobis&quot;) as well as &quot;hellinger&quot; and &quot;wt-unifrac&quot;. 
The Hellinger distance measure (<code>dist.method="hellinger"</code>) takes the form
<code>0.5*E</code>, where E is the Euclidean distance between the square-root-transformed 
frequency data. The weighted UniFrac distance (<code>dist.method="wt-unifrac"</code>) 
is calculated by interally calling <code>GUniFrac</code> in the <code>GUniFrac</code> package.
Not used when anything other than <code>dist=NULL</code> is specified for <code>dist</code>.
The default is &quot;bray&quot;.</p>
</td></tr>
<tr><td><code id="ldm_+3A_dist">dist</code></td>
<td>
<p>a distance matrix. Can be an object of class either &quot;dist&quot; or &quot;matrix&quot;.
The elements of the distance matrix will be squared and then the matrix will be centered if the default choices 
<code>square.dist=TRUE</code> and <code>center.otu.table=TRUE</code> are used. If <code>dist=NULL</code>, the distance matrix is 
calculated from the <code>otu.table</code>, using the value of <code>dist.method</code> (and <code>tree</code> if required). 
The default is NULL.</p>
</td></tr>
<tr><td><code id="ldm_+3A_cluster.id">cluster.id</code></td>
<td>
<p>character or factor variable that identifies clusters. The default value
cluster.id=NULL if the observations are not clustered (i.e., are independent).</p>
</td></tr>
<tr><td><code id="ldm_+3A_strata">strata</code></td>
<td>
<p>a character or factor variable that defines strata (groups), within which to constrain permutations. 
The default is NULL.</p>
</td></tr>
<tr><td><code id="ldm_+3A_how">how</code></td>
<td>
<p>a permutation control list, for users who want to specify their own call to the <code>how</code> function from the <code>permute</code> package.
The default is NULL.</p>
</td></tr>
<tr><td><code id="ldm_+3A_perm.within.type">perm.within.type</code></td>
<td>
<p>a character string that takes values &quot;free&quot;, &quot;none&quot;, &quot;series&quot;, or &quot;grid&quot;.
The default is &quot;free&quot; (for random permutations).</p>
</td></tr>
<tr><td><code id="ldm_+3A_perm.between.type">perm.between.type</code></td>
<td>
<p>a character string that takes values &quot;free&quot;, &quot;none&quot;, or &quot;series&quot;.
The default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="ldm_+3A_perm.within.ncol">perm.within.ncol</code></td>
<td>
<p>a positive integer, only used if perm.within.type=&quot;grid&quot;. 
The default is 0.  See documentation for permute package for additional details.</p>
</td></tr>
<tr><td><code id="ldm_+3A_perm.within.nrow">perm.within.nrow</code></td>
<td>
<p>a positive integer, only used if perm.within.type=&quot;grid&quot;. 
The default is 0.  See documentation for permute package for additional details.</p>
</td></tr>
<tr><td><code id="ldm_+3A_n.perm.max">n.perm.max</code></td>
<td>
<p>the maximum number of permutations. The default is NULL, in which case a maximum of
5000 permutations are used for the global test and a maximum of <code>n.otu</code> * <code>n.rej.stop</code> * (1/<code>fdr.nominal</code>) 
are used for the OTU test, where <code>n.otu</code> is the number of OTUs.  If a numeric value for <code>n.perm.max</code> is specified, 
this value is used for both global and OTU-level tests.</p>
</td></tr>
<tr><td><code id="ldm_+3A_n.rej.stop">n.rej.stop</code></td>
<td>
<p>the minimum number of rejections (i.e., the permutation 
statistic exceeds the observed statistic) to obtain before stopping. The 
default is 100.</p>
</td></tr>
<tr><td><code id="ldm_+3A_seed">seed</code></td>
<td>
<p>a user-supplied integer seed for the random number generator in the 
permutation procedure. The default is NULL; with the default value, an integer seed will be 
generated internally and randomly. In either case, the integer seed will be stored
in the output object in case 
the user wants to reproduce the permutation replicates.</p>
</td></tr>
<tr><td><code id="ldm_+3A_fdr.nominal">fdr.nominal</code></td>
<td>
<p>the nominal FDR value. The default is 0.1.</p>
</td></tr>
<tr><td><code id="ldm_+3A_square.dist">square.dist</code></td>
<td>
<p>a logical variable indicating whether to square the 
distance matrix. The default is TRUE.</p>
</td></tr>
<tr><td><code id="ldm_+3A_scale.otu.table">scale.otu.table</code></td>
<td>
<p>a logical variable indicating whether to scale the rows of the OTU table.  For count 
data, this corresponds to dividing by the library size to give frequencies (i.e., relative abundances).  Does not affect the tran scale.  
The default is TRUE.</p>
</td></tr>
<tr><td><code id="ldm_+3A_center.otu.table">center.otu.table</code></td>
<td>
<p>a logical variable indicating whether to center the 
columns of the OTU table. The OTU table should be centered if the distance 
matrix has been centered. Applied to both the frequency and transformed scales.  The default is TRUE.</p>
</td></tr>
<tr><td><code id="ldm_+3A_freq.scale.only">freq.scale.only</code></td>
<td>
<p>a logical variable indicating whether to perform analysis of the frequency-scale data only 
(not the arcsin-root transformed frequency data and the omnibus test). The default is FALSE.</p>
</td></tr>
<tr><td><code id="ldm_+3A_binary">binary</code></td>
<td>
<p>a logical value indicating whether to perform presence-absence
analysis. The default is FALSE (analyzing relative abundance data).</p>
</td></tr>
<tr><td><code id="ldm_+3A_n.rarefy">n.rarefy</code></td>
<td>
<p>an integer-valued number of rarefactions. The value &quot;all&quot; is also allowed, 
and requests the LDM-A method that essentially aggregate information from all rarefactions.
The default is 0 (no rarefaction).</p>
</td></tr>
<tr><td><code id="ldm_+3A_test.mediation">test.mediation</code></td>
<td>
<p>a logical value indicating whether to perform the mediation analysis. The default is FALSE. 
If TRUE, the formula takes the specific form <code>otu.table ~ exposure + outcome</code> or most generally
<code>otu.table | (set of confounders) ~ (set of exposures) + (set of outcomes)</code>.</p>
</td></tr>
<tr><td><code id="ldm_+3A_test.omni3">test.omni3</code></td>
<td>
<p>a logical value indicating whether to perform the new omnibus test (LDM-omni3). The default is FALSE.</p>
</td></tr>
<tr><td><code id="ldm_+3A_comp.anal">comp.anal</code></td>
<td>
<p>a logical value indicating whether the centered-log-ratio taxa count data are used (LDM-clr). The default is FALSE.</p>
</td></tr>
<tr><td><code id="ldm_+3A_comp.anal.adjust">comp.anal.adjust</code></td>
<td>
<p>a character string that takes value &quot;median&quot; or &quot;mode&quot; to choose the estimator for the beta mean (Hu and Satten, 2023). The default is &quot;median&quot;.</p>
</td></tr>
<tr><td><code id="ldm_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use in parallel computing, i.e., at most how many child processes will be run simultaneously. 
The default is 4.</p>
</td></tr>
<tr><td><code id="ldm_+3A_verbose">verbose</code></td>
<td>
<p>a logical value indicating whether to generate verbose output during the permutation process. Default is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The formula has the form 
</p>
<p><code>otu.table ~ (first set of covariates) + (second set of covariates)
... + (last set of covariates)</code> 
</p>
<p>or 
</p>
<p><code>otu.table | confounders ~ (first set of covariates) + (second set of covariates)
... + (last set of covariates)</code> 
</p>
<p>where <code>otu.table</code> is
the OTU table with rows for samples and columns for OTUs and each set of 
covariates are enclosed in parentheses. The covariates in each submodel (set of covariates) are tested jointly,
after projecting off terms in submodels that appear earlier in the model.
</p>
<p>For example, given OTU table <code>y</code> and a data frame <code>metadata</code> that contains 4 covariates, 
<code>a</code>, <code>b</code>, <code>c</code> and <code>d</code>,  
some valid formulas would be:
</p>
<p><code>y ~ a + b + c + d</code> ### no confounders, 4 submodels (i.e., sets of covariates)
</p>
<p><code>y ~ (a+b) + (c+d)</code> ### no confounders, 2 submodels each having 
2 covariates
</p>
<p><code>y | b ~ (a+c) + d</code> ### <code>b</code> is a confounder, submodel 1 is 
<code>(a+c)</code>, and submodel 2 is <code>d</code>
</p>
<p><code>y | b+c ~ a*d</code>     ### there are 2 confounders <code>b</code> 
and <code>c</code>; there is 1 submodel consisting of the three terms <code>a</code>, <code>d</code>, and <code>a:d</code> (interaction). 
This example is equivalent to <code>y | b+c ~ (a+d+a:d)</code>
</p>
<p><code>y | as.factor(b) ~ (a+d) + a:d</code>  ### the confounder 
<code>b</code> will be treated as a factor variable, submodel 1 will have the main 
effects <code>a</code> and <code>d</code>, and submodel 2 will have only the interaction 
between <code>a</code> and <code>d</code>
</p>
<p><code>y | as.factor(b) ~ (a) + (d) + (a:d)</code> ### there are 3 submodels <code>a</code>, <code>d</code>, and <code>a:d</code>.
Putting paratheses around a single variable is allowed but not necessary.
</p>
<p>Submodels that combine character and numeric values are allowed; character-valued variables are coerced into factor 
variables.  Confounders are distinguished from other covariates as test statistics are not calculated for confounders
(which are included for scientific reasons, not by virtue of significance test results); 
consequently they also do not contribute to stopping criteria.  If tests of confounders are desired, confounders should
put on the right hand side of the formula as the first submodel.
</p>
<p>For testing mediation effects of the microbiome that mediate the effect of the exposure(s) on the outcome(s), 
the formula takes the specific form:
</p>
<p><code>otu.table ~ exposure + outcome</code>
</p>
<p>or most generally
</p>
<p><code>otu.table | (set of confounders) ~ (set of exposures) + (set of outcomes)</code>
</p>
<p>in which there should be exactly two terms on the right hand side of the regression, 
corresponding to the exposure(s) and the outcome(s), the outcome(s) must appear after the exposure(s), 
and the covariates or confounders must appear after <code>|</code>.
</p>
<p>LDM uses two sequential stopping criteria. For the global test, LDM uses the 
stopping rule of Besag and Clifford (1991), which stops permutation when a 
pre-specified minimum number (default=100) of rejections (i.e., the permutation 
statistic exceeded the observed test statistic) has been reached. For the 
OTU-specific tests, LDM uses the stopping rule of Sandve et al. (2011), 
which stops permutation when every OTU test has either reached the pre-specified 
number (default=100) of rejections or yielded a q-value that is below the 
nominal FDR level (default=0.1). As a convention, we call a test &quot;stopped&quot;
if the corresponding stopping criterion has been satisfied. Although all tests 
are always terminated if a pre-specified maximum number (see description of <code>n.perm.max</code> in Arguments list) of 
permutations have been generated, some tests may not have &quot;stopped&quot;.  This typically occurs when
the relevant p-value is small or near the cutoff for inclusion in a list of significant findings; 
for global tests meeting the stopping criterion is not critical, but 
caution is advised when interpreting OTU-level tests that have not stopped as additional OTUs may be found 
with a larger number of permutations.
</p>


<h3>Value</h3>

<p>a list consisting of 
</p>
<table role = "presentation">
<tr><td><code>x</code></td>
<td>
<p>the (orthonormal) design matrix X as defined in Hu and Satten (2020)</p>
</td></tr> 
<tr><td><code>dist</code></td>
<td>
<p>the (squared/centered) distance matrix</p>
</td></tr> 
<tr><td><code>mean.freq</code></td>
<td>
<p>the mean relative abundance of OTUs (the column means of the frequency-scale OTU table)</p>
</td></tr> 
<tr><td><code>y.freq</code></td>
<td>
<p>the frequency-scale OTU table, scaled and centered if so specified</p>
</td></tr> 
<tr><td><code>d.freq</code></td>
<td>
<p>a vector of the non-negative diagonal elements of <code>D</code> that satisfies
<code>x^T y.freq = D v^T</code></p>
</td></tr>
<tr><td><code>v.freq</code></td>
<td>
<p>the v matrix with unit columns that satisfies
<code>x^T y.freq = D v^T</code></p>
</td></tr>
<tr><td><code>y.tran</code></td>
<td>
<p>the (column-centered) arcsin-root-transformed 
OTU table</p>
</td></tr> 
<tr><td><code>d.tran</code></td>
<td>
<p>a vector of the non-negative diagonal elements of <code>D</code> that satisfies
<code>x^T y.tran = D v^T</code></p>
</td></tr>
<tr><td><code>v.tran</code></td>
<td>
<p>the v matrix with unit columns that satisfies
<code>x^T y.tran = D v^T</code></p>
</td></tr>
<tr><td><code>low</code></td>
<td>
<p>a vector of lower indices for confounders (if there is any) and submodels</p>
</td></tr>
<tr><td><code>up</code></td>
<td>
<p>a vector of upper indices for confounders (if there is any) and submodels</p>
</td></tr>
<tr><td><code>beta</code></td>
<td>
<p>a matrix of effect sizes of every trait on every OTU</p>
</td></tr>
<tr><td><code>phi</code></td>
<td>
<p>a matrix of probabilities that the rarefied count of an OTU in a sample is non-zero</p>
</td></tr>
<tr><td><code>VE.global.freq.confounders</code></td>
<td>
<p>Variance explained (VE) by confounders, based on the frequency-scale data</p>
</td></tr>
<tr><td><code>VE.global.freq.submodels</code></td>
<td>
<p>VE by each submodel, based on the frequency-scale data</p>
</td></tr>
<tr><td><code>VE.global.freq.residuals</code></td>
<td>
<p>VE by each component in the residual distance, based on the frequency-scale data</p>
</td></tr>
<tr><td><code>VE.otu.freq.confounders</code></td>
<td>
<p>Contribution of each OTU to VE by confounders, based on the frequency-scale data</p>
</td></tr>
<tr><td><code>VE.otu.freq.submodel</code></td>
<td>
<p>Contribution of each OTU to VE by each submodel, based on the frequency-scale data</p>
</td></tr>
<tr><td><code>VE.global.tran.confounders</code></td>
<td>
<p>VE by confounders, based on 
the arcsin-root-transformed frequency data</p>
</td></tr>
<tr><td><code>VE.global.tran.submodels</code></td>
<td>
<p>VE by each submodel, based on 
the arcsin-root-transformed frequency data</p>
</td></tr>
<tr><td><code>VE.global.tran.residuals</code></td>
<td>
<p>VE by each component in the residual distance, based on 
the arcsin-root-transformed frequency data</p>
</td></tr>
<tr><td><code>VE.otu.tran.confounders</code></td>
<td>
<p>Contribution of each OTU to VE by confounders, based on 
the arcsin-root-transformed frequency data</p>
</td></tr>
<tr><td><code>VE.otu.tran.submodels</code></td>
<td>
<p>Contribution of each OTU to VE by each submodel, based on 
the arcsin-root-transformed frequency data</p>
</td></tr>
<tr><td><code>VE.df.confounders</code></td>
<td>
<p>Degree of freedom (i.e., number of components) associated with the VE for confounders</p>
</td></tr>
<tr><td><code>VE.df.submodels</code></td>
<td>
<p>Degree of freedom (i.e., number of components) associated with the VE for each submodel</p>
</td></tr>
<tr><td><code>F.global.freq</code></td>
<td>
<p>F statistics for testing each submodel, based on
the frequency-scale data</p>
</td></tr> 
<tr><td><code>F.global.tran</code></td>
<td>
<p>F statistics for testing each submodel, based on 
the arcsin-root-transformed frequency data</p>
</td></tr> 
<tr><td><code>F.otu.freq</code></td>
<td>
<p>F statistics for testing each OTU for each submodel, based on the frequency-scale data</p>
</td></tr> 
<tr><td><code>F.otu.tran</code></td>
<td>
<p>F statistics for testing each OTU for each submodel, based on the arcsin-root-transformed data</p>
</td></tr> 
<tr><td><code>p.global.freq</code></td>
<td>
<p>p-values for the global test of each set of covariates
based on the frequency-scale data</p>
</td></tr> 
<tr><td><code>p.global.tran</code></td>
<td>
<p>p-values for the global test of each set of covariates
based on the arcsin-root-transformed frequency data</p>
</td></tr> 
<tr><td><code>p.global.pa</code></td>
<td>
<p>p-values for the global test of each set of covariates
based on the presence-absence data</p>
</td></tr> 
<tr><td><code>p.global.omni</code></td>
<td>
<p>p-values for the global test of each set of covariates 
based on the omnibus statistics in LDM-omni, which are the minima of the p-values obtained 
from the frequency scale and the arcsin-root-transformed frequency data 
as the final test statistics, and use the corresponding minima from the 
permuted data to simulate the null distributions</p>
</td></tr> 
<tr><td><code>p.global.harmonic</code></td>
<td>
<p>p-values for the global test of each set of covariates
based on the Harmonic-mean p-value combination method applied to the OTU-level omnibus p-values</p>
</td></tr> 
<tr><td><code>p.global.fisher</code></td>
<td>
<p>p-values for the global test of each set of covariates
based on the Fisher p-value combination method applied to the OTU-level omnibus p-values</p>
</td></tr> 
<tr><td><code>p.global.omni3</code></td>
<td>
<p>p-values for the global test of each set of covariates 
based on the omnibus test LDM-omni3</p>
</td></tr> 
<tr><td><code>p.global.freq.OR</code>, <code>p.global.tran.OR</code>, <code>p.global.pa.OR</code>, <code>p.global.omni.OR</code>, <code>p.global.harmonic.OR</code>, <code>p.global.fisher.OR</code>, <code>p.global.omni3.OR</code></td>
<td>
<p>global p-values for testing <code>other.surv.resid</code></p>
</td></tr>
<tr><td><code>p.global.freq.com</code>, <code>p.global.tran.com</code>, <code>p.global.pa.com</code>, <code>p.global.omni.com</code>, <code>p.global.harmonic.com</code>, <code>p.global.fisher.com</code>, <code>p.global.omni3.com</code></td>
<td>
<p>global p-values from the combination test that combines the results from analyzing both the Martingale and deviance residuals from a Cox model (one of them is supplied by <code>other.surv.resid</code>)</p>
</td></tr>
<tr><td><code>p.otu.freq</code></td>
<td>
<p>p-values for the OTU-specific tests based on the 
frequency scale data</p>
</td></tr> 
<tr><td><code>p.otu.tran</code></td>
<td>
<p>p-values for the OTU-specific tests based on the 
arcsin-root-transformed frequency data</p>
</td></tr>
<tr><td><code>p.otu.pa</code></td>
<td>
<p>p-values for the OTU-specific tests based on the 
presence-absence data</p>
</td></tr> 
<tr><td><code>p.otu.omni</code></td>
<td>
<p>p-values for the OTU-specific tests based on the 
omnibus test LDM-omni</p>
</td></tr> 
<tr><td><code>p.otu.omni3</code></td>
<td>
<p>p-values for the OTU-specific tests based on the 
omnibus test LDM-omni3</p>
</td></tr> 
<tr><td><code>q.otu.freq</code></td>
<td>
<p>q-values (i.e., FDR-adjusted p-values) 
for the OTU-specific tests based on the frequency scale data</p>
</td></tr> 
<tr><td><code>q.otu.tran</code></td>
<td>
<p>q-values for the OTU-specific tests based on 
the arcsin-root-transformed frequency data</p>
</td></tr> 
<tr><td><code>q.otu.pa</code></td>
<td>
<p>q-values (i.e., FDR-adjusted p-values) 
for the OTU-specific tests based on the presence-absence data</p>
</td></tr> 
<tr><td><code>q.otu.omni</code></td>
<td>
<p>q-values for the OTU-specific tests based on the 
omnibus test LDM-omni</p>
</td></tr>
<tr><td><code>q.otu.omni3</code></td>
<td>
<p>q-values for the OTU-specific tests based on the 
omnibus test LDM-omni3</p>
</td></tr> 
<tr><td><code>p.otu.freq.OR</code>, <code>p.otu.tran.OR</code>, <code>p.otu.pa.OR</code>, <code>p.otu.omni.OR</code>, <code>p.otu.omni3.OR</code>, <code>q.otu.freq.OR</code>, <code>q.otu.tran.OR</code>, <code>q.otu.pa.OR</code>, <code>q.otu.omni.OR</code>, <code>q.otu.omni3.OR</code></td>
<td>
<p>OTU-level p-values and q-values for testing <code>other.surv.resid</code></p>
</td></tr>
<tr><td><code>p.otu.freq.com</code>, <code>p.otu.tran.com</code>, <code>p.otu.pa.com</code>, <code>p.otu.omni.com</code>, <code>p.otu.omni3.com</code>, <code>q.otu.freq.com</code>, <code>q.otu.tran.com</code>, <code>q.otu.pa.com</code>, <code>q.otu.omni.com</code>, <code>q.otu.omni3.com</code></td>
<td>
<p>OTU-level p-values and q-values from the combination tests that combine the results from analyzing both the Martingale and deviance residuals from a Cox model (one of them is supplied by <code>other.surv.resid</code>)</p>
</td></tr>
<tr><td><code>detected.otu.freq</code></td>
<td>
<p>detected OTUs (whose names are found in the column names of the OTU table) at the nominal FDR, based on the frequency scale data</p>
</td></tr> 
<tr><td><code>detected.otu.tran</code></td>
<td>
<p>detected OTUs based on the arcsin-root-transformed frequency data</p>
</td></tr> 
<tr><td><code>detected.otu.pa</code></td>
<td>
<p>detected OTUs based on the presence-absence data</p>
</td></tr> 
<tr><td><code>detected.otu.omni</code></td>
<td>
<p>detected OTU based on the 
omnibus test LDM-omni</p>
</td></tr> 
<tr><td><code>detected.otu.omni3</code></td>
<td>
<p>detected OTU based on the 
omnibus test LDM-omni3</p>
</td></tr> 
<tr><td><code>detected.otu.freq.OR</code>, <code>detected.otu.tran.OR</code>, <code>detected.otu.pa.OR</code>, <code>detected.otu.omni.OR</code>, <code>detected.otu.omni3.OR</code></td>
<td>
<p>detected OTUs for <code>other.surv.resid</code></p>
</td></tr>
<tr><td><code>detected.otu.freq.com</code>, <code>detected.otu.tran.com</code>, <code>detected.otu.pa.com</code>, <code>detected.otu.omni.com</code>, <code>detected.otu.omni3.com</code></td>
<td>
<p>detected OTUs by the combination tests that combines the Martingale and deviance residuals from a Cox model (one of them is supplied by <code>other.surv.resid</code>)</p>
</td></tr>
<tr><td><code>med.p.global.freq</code>, <code>med.p.global.tran</code>, <code>med.p.global.omni</code>, <code>med.p.global.pa</code>, <code>med.p.global.harmonic</code>, <code>med.p.global.fisher</code>, <code>med.p.global.omni3</code></td>
<td>
<p>p-values for the global tests of the overall mediation effect by the microbiome</p>
</td></tr>
<tr><td><code>med.p.global.freq.OR</code>, <code>med.p.global.tran.OR</code>, <code>med.p.global.omni.OR</code>, <code>med.p.global.pa.OR</code>, <code>med.p.global.harmonic.OR</code>, <code>med.p.global.fisher.OR</code>, <code>med.p.global.omni3.OR</code></td>
<td>
<p>p-values for the global tests of the overall mediation effect by the microbiome, when the outcome is <code>other.surv.resid</code></p>
</td></tr>
<tr><td><code>med.p.global.freq.com</code>, <code>med.p.global.tran.com</code>, <code>med.p.global.omni.com</code>, <code>med.p.global.pa.com</code>, <code>med.p.global.harmonic.com</code>, <code>med.p.global.fisher.com</code>, <code>med.p.global.omni3.com</code></td>
<td>
<p>p-values for the global tests of the overall mediation effect by the microbiome, combining the results from analyzing both the Martingale and deviance residuals as outcomes</p>
</td></tr>
<tr><td><code>med.detected.otu.freq</code>, <code>med.detected.otu.tran</code>, <code>med.detected.otu.omni</code>, <code>med.detected.otu.pa</code>, <code>med.detected.otu.omni3</code></td>
<td>
<p>detected mediating OTUs</p>
</td></tr>
<tr><td><code>med.detected.otu.freq.OR</code>, <code>med.detected.otu.tran.OR</code>, <code>med.detected.otu.omni.OR</code>, <code>med.detected.otu.pa.OR</code>, <code>med.detected.otu.omni3.OR</code></td>
<td>
<p>detected mediating OTUs for the outcome <code>other.surv.resid</code></p>
</td></tr>
<tr><td><code>med.detected.otu.freq.com</code>, <code>med.detected.otu.tran.com</code>, <code>med.detected.otu.omni.com</code>, <code>med.detected.otu.pa.com</code>, <code>med.detected.otu.omni3.com</code></td>
<td>
<p>detected mediating OTUs, combining the results from analyzing both the Martingale and deviance residuals as outcomes</p>
</td></tr>
<tr><td><code>n.perm.completed</code></td>
<td>
<p>number of permutations completed</p>
</td></tr> 
<tr><td><code>global.tests.stopped</code></td>
<td>
<p>a logical value indicating whether the 
stopping criterion has been met by all global tests</p>
</td></tr> 
<tr><td><code>otu.tests.stopped</code></td>
<td>
<p>a logical value indicating whether the 
stopping criterion has been met by all OTU-specific tests</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>the seed that is user supplied or internally generated, stored in case 
the user wants to reproduce the permutation replicates</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;, Glen A. Satten &lt;gsatten@emory.edu&gt;
</p>


<h3>References</h3>

<p>Hu YJ, Satten GA (2020). Testing hypotheses about the microbiome using the linear decomposition model (LDM) 
Bioinformatics, 36(14), 4106-4115.
</p>
<p>Hu YJ, Lane A, and Satten GA (2021). A rarefaction-based extension of the LDM for testing presence-absence associations in the microbiome. Bioinformatics, 37(12):1652-1657.
</p>
<p>Zhu Z, Satten GA, Caroline M, and Hu YJ (2020). Analyzing matched sets of microbiome data using the LDM and PERMANOVA. Microbiome, 9(133), https://doi.org/10.1186/s40168-021-01034-9.
</p>
<p>Zhu Z, Satten GA, and Hu YJ (2022). Integrative analysis of relative abundance data and presence-absence data of the microbiome using the LDM. Bioinformatics, doi.org/10.1093/bioinformatics/btac181.
</p>
<p>Yue Y and Hu YJ (2021) A new approach to testing mediation of the microbiome using the LDM. bioRxiv, https://doi.org/10.1101/2021.11.12.468449.
</p>
<p>Hu Y, Li Y, Satten GA, and Hu YJ (2022) Testing microbiome associations with censored survival outcomes at both the community and individual taxon levels. bioRxiv, doi.org/10.1101/2022.03.11.483858.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>res.ldm &lt;- ldm(formula=throat.otu.tab5 | (Sex+AntibioticUse) ~ SmokingStatus+PackYears, 
              data=throat.meta, seed=67817, fdr.nominal=0.1, n.perm.max=1000, n.cores=1, 
              verbose=FALSE) 
</code></pre>

<hr>
<h2 id='permanovaFL'>PERMANOVA test of association based on the Freedman-Lane permutation scheme</h2><span id='topic+permanovaFL'></span>

<h3>Description</h3>

<p>This function performs the PERMANOVA test that can allow adjustment of
confounders and control of clustered data. It can also be used for testing 
presence-absence associations based on infinite number of rarefaction replicates. 
As in <code>ldm</code>, 
<code>permanovaFL</code> allows multiple sets of covariates to be tested, 
in the way that the sets are entered sequentially and the variance 
explained by each set is that part that remains after the previous 
sets have been fit. It allows testing of a survival outcome, by using the Martingale or deviance residual (from fitting a Cox model to the survival outcome and other covariates) as a covariate in the regression. 
It allows multiple distance matrices and provides an omnibus test in such cases. 
It also allows testing of the mediation effect of the microbiome in the pathway between the exposure(s) and the outcome(s), 
where the exposure(s) and outcomes(s) are specified as the first and second (sets of) covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permanovaFL(
  formula,
  other.surv.resid = NULL,
  data = .GlobalEnv,
  tree = NULL,
  dist.method = c("bray"),
  dist.list = NULL,
  cluster.id = NULL,
  strata = NULL,
  how = NULL,
  perm.within.type = "free",
  perm.between.type = "none",
  perm.within.ncol = 0,
  perm.within.nrow = 0,
  n.perm.max = 5000,
  n.rej.stop = 100,
  seed = NULL,
  square.dist = TRUE,
  center.dist = TRUE,
  scale.otu.table = c(TRUE),
  binary = c(FALSE),
  n.rarefy = 0,
  test.mediation = FALSE,
  n.cores = 4,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="permanovaFL_+3A_formula">formula</code></td>
<td>
<p>a symbolic description of the model to be fitted in the form
of <code>data.matrix ~ sets of covariates</code> or <code>data.matrix |
confounders ~ sets of covariates</code>. The details of model specification are
given in &quot;Details&quot; of <code>ldm</code>. Additionally, in <code>permanovaFL</code>, the <code>data.matrix</code>
can be either an OTU table or a distance matrix. If it is an OTU table,
the distance matrix will be calculated internally using the OTU table, <code>tree</code> (if required), and 
<code>dist.method</code>. If <code>data.matrix</code> is a distance
matrix (having class <code>dist</code> or <code>matrix</code>), it can be squared and//or centered by
specifying <code>square.dist</code> and <code>center.dist</code> (described below).  Distance matrices are distinguished
from OTU tables by checking for symmetry of <code>as.matrix(data.matrix)</code>.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_other.surv.resid">other.surv.resid</code></td>
<td>
<p>a vector of data, usually the Martingale or deviance residuals from fitting the Cox model to the survival outcome (if it is the outcome of interest) and other covariates.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_data">data</code></td>
<td>
<p>an optional data frame, list or environment (or object coercible 
to a dataframe) containing the covariates of interest and confounding covariates. 
If not found in <code>data</code>, the covariates are taken from environment(formula), 
typically the environment from which <code>permanovaFL</code> is called. The default is .GlobalEnv.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_tree">tree</code></td>
<td>
<p>a phylogenetic tree. Only used for calculating a 
phylogenetic-tree-based distance matrix. Not needed if the calculation of 
the requested distance does not involve a phylogenetic tree, or if a 
distance matrix is directly imported through <code>formula</code>.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_dist.method">dist.method</code></td>
<td>
<p>a vector of methods for calculating the distance measure, partial
match to all methods supported by <code>vegdist</code> in the <code>vegan</code> package
(i.e., &quot;manhattan&quot;, &quot;euclidean&quot;, &quot;canberra&quot;, &quot;bray&quot;, &quot;kulczynski&quot;, &quot;jaccard&quot;, &quot;gower&quot;, 
&quot;altGower&quot;, &quot;morisita&quot;, &quot;horn&quot;, &quot;mountford&quot;, &quot;raup&quot; , &quot;binomial&quot;, &quot;chao&quot;, &quot;cao&quot;, &quot;mahalanobis&quot;)
as well as &quot;hellinger&quot; and &quot;wt-unifrac&quot;. 
Not used if a distance matrix is specified in <code>formula</code> or <code>dist.list</code>. 
The default is c(&quot;bray&quot;). 
For more details, see the <code>dist.method</code> argument in the <code>ldm</code> function.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_dist.list">dist.list</code></td>
<td>
<p>a list of pre-calculated distance matrices.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_cluster.id">cluster.id</code></td>
<td>
<p>cluster identifiers. The default is value of NULL should be used if the observations are 
not in clusters (i.e., independent).</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_strata">strata</code></td>
<td>
<p>a factor variable (or, character variable converted into a factor) to define strata (groups), within which to constrain permutations. 
The default is NULL.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_how">how</code></td>
<td>
<p>a permutation control list, for users who want to specify their permutation control list using the <code>how</code> function 
from the <code>permute</code> R package.  The default is NULL.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_perm.within.type">perm.within.type</code></td>
<td>
<p>a character string that takes values &quot;free&quot;, &quot;none&quot;, &quot;series&quot;, or &quot;grid&quot;.
The default is &quot;free&quot; (for random permutations).</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_perm.between.type">perm.between.type</code></td>
<td>
<p>a character string that takes values &quot;free&quot;, &quot;none&quot;, or &quot;series&quot;.
The default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_perm.within.ncol">perm.within.ncol</code></td>
<td>
<p>a positive integer, only used if perm.within.type=&quot;grid&quot;. 
The default is 0.  See the documentation for the R package <code>permute</code> for further details.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_perm.within.nrow">perm.within.nrow</code></td>
<td>
<p>a positive integer, only used if perm.within.type=&quot;grid&quot;. 
The default is 0.  See the documentation for the R package <code>permute</code> for further details.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_n.perm.max">n.perm.max</code></td>
<td>
<p>the maximum number of permutations.
The default is 5000.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_n.rej.stop">n.rej.stop</code></td>
<td>
<p>the minimum number of rejections (i.e., the permutation 
statistic exceeds the observed statistic) to obtain before stopping. 
The default is 100.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_seed">seed</code></td>
<td>
<p>a user-supplied integer seed for the random number generator in the 
permutation procedure. The default is NULL; with the default value, an integer seed will be 
generated internally and randomly. In either case, the integer seed will be stored
in the output object in case 
the user wants to reproduce the permutation replicates.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_square.dist">square.dist</code></td>
<td>
<p>a logical variable indicating whether to square the
distance matrix. The default is TRUE.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_center.dist">center.dist</code></td>
<td>
<p>a logical variable indicating whether to center the 
distance matrix as described by Gower (1966). The default is TRUE.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_scale.otu.table">scale.otu.table</code></td>
<td>
<p>a vector of logical variables indicating whether to scale the OTU table in calculating the distance matrices in <code>dist.method</code>.
For count data, this corresponds to dividing by the library size to give
relative abundances. The default is TRUE.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_binary">binary</code></td>
<td>
<p>a vector of logical values indicating whether to base the calculation of the distance matrices in <code>dist.method</code> on presence-absence (binary) data. The default is c(FALSE) (analyzing relative abundance data).</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_n.rarefy">n.rarefy</code></td>
<td>
<p>number of rarefactions. The default is 0 (no rarefaction).</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_test.mediation">test.mediation</code></td>
<td>
<p>a logical value indicating whether to perform the mediation analysis. The default is FALSE. 
If TRUE, the formula takes the specific form <code>otu.table ~ exposure + outcome</code> or most generally
<code>otu.table or distance matrix | (set of confounders) ~ (set of exposures) + (set of outcomes)</code>.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_n.cores">n.cores</code></td>
<td>
<p>The number of cores to use in parallel computing, i.e., at most how many child processes will be run simultaneously. 
The default is 4.</p>
</td></tr>
<tr><td><code id="permanovaFL_+3A_verbose">verbose</code></td>
<td>
<p>a logical value indicating whether to generate verbose output during the permutation process. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list consisting of 
</p>
<table role = "presentation">
<tr><td><code>F.statistics</code></td>
<td>
<p>F statistics for testing each set of covariates</p>
</td></tr>
<tr><td><code>R.squared</code></td>
<td>
<p>R-squared statistic for each set of covariates</p>
</td></tr>
<tr><td><code>F.statistics.OR</code>, <code>R.squared.OR</code></td>
<td>
<p>F statistics and R-squared statistic when the last covariate is <code>other.surv.resid</code></p>
</td></tr>
<tr><td><code>p.permanova</code></td>
<td>
<p>p-values for testing each set of covariates</p>
</td></tr> 
<tr><td><code>p.permanova.omni</code></td>
<td>
<p>the omnibus p-values (that combines information from multiple distance matrices) for testing each set of covariates</p>
</td></tr> 
<tr><td><code>med.p.permanova</code></td>
<td>
<p>p-values for testing mediation</p>
</td></tr>
<tr><td><code>med.p.permanova.omni</code></td>
<td>
<p>the omnibus p-values for testing mediation</p>
</td></tr>
<tr><td><code>p.permanova.OR</code>, <code>p.permanova.omni.OR</code></td>
<td>
<p>when using <code>other.surv.resid</code> as the last covariate</p>
</td></tr>
<tr><td><code>med.p.permanova.OR</code>, <code>med.p.permanova.omni.OR</code></td>
<td>
<p>when using <code>other.surv.resid</code> as the outcome in the mediation analysis</p>
</td></tr>
<tr><td><code>p.permanova.com</code>, <code>p.permanova.omni.com</code></td>
<td>
<p>the combination test that combines the results from analyzing the Martingale residual and the Deviance residual (one specified in the formula and one specified in <code>other.surv.resid</code>)</p>
</td></tr>
<tr><td><code>med.p.permanova.com</code>, <code>med.p.permanova.omni.com</code></td>
<td>
<p>the combination test for the mediation effect</p>
</td></tr>
<tr><td><code>n.perm.completed</code></td>
<td>
<p>number of permutations completed</p>
</td></tr>
<tr><td><code>permanova.stopped</code></td>
<td>
<p>a logical value indicating whether the 
stopping criterion has been met by all tests of covariates</p>
</td></tr>
<tr><td><code>seed</code></td>
<td>
<p>the seed that is user supplied or internally generated, stored in case 
the user wants to reproduce the permutation replicates</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;, Glen A. Satten &lt;gsatten@emory.edu&gt;
</p>


<h3>References</h3>

<p>Hu YJ, Satten GA (2020). Testing hypotheses about the microbiome using the linear decomposition model (LDM) 
Bioinformatics, 36(14), 4106-4115.
</p>
<p>Hu YJ and Satten GA (2021). A rarefaction-without-resampling extension of PERMANOVA for testing presence-absence associations in the microbiome. bioRxiv, https://doi.org/10.1101/2021.04.06.438671.
</p>
<p>Zhu Z, Satten GA, Caroline M, and Hu YJ (2020). Analyzing matched sets of microbiome data using the LDM and PERMANOVA. Microbiome, 9(133), https://doi.org/10.1186/s40168-021-01034-9.
</p>
<p>Hu Y, Li Y, Satten GA, and Hu YJ (2022) Testing microbiome associations with censored survival outcomes at both the community and individual taxon levels. bioRxiv, doi.org/10.1101/2022.03.11.483858.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>res.perm &lt;- permanovaFL(throat.otu.tab5 | (Sex+AntibioticUse) ~ SmokingStatus+PackYears, 
                       data=throat.meta, dist.method="bray", seed=82955, n.perm.max=1000, n.cores=1, 
                       verbose=FALSE)
</code></pre>

<hr>
<h2 id='sim.meta'>Metadata of the simulated microbiome samples</h2><span id='topic+sim.meta'></span>

<h3>Description</h3>

<p>This data set includes 100 simulated samples from 50 subjects, with two samples from each subject. 
The metadata contain subject IDs and data from four variables, two (X.between and Y.between) varying between subjects 
and two (X.within and Y.within) varying within subjects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("sim.meta")
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on 5 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim.meta)
</code></pre>

<hr>
<h2 id='sim.otu.tab'>OTU count table of the simulated microbiome samples</h2><span id='topic+sim.otu.tab'></span>

<h3>Description</h3>

<p>This is the OTU table corresponding to the 100 simulated samples in sim.meta.
The samples in the rows of the OTU table match with the samples in the rows of sim.meta.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("sim.otu.tab")
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on 813 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim.otu.tab)
</code></pre>

<hr>
<h2 id='sim.otu.tab5'>Filtered OTU count table of the simulated microbiome samples</h2><span id='topic+sim.otu.tab5'></span>

<h3>Description</h3>

<p>This table was derived from sim.otu.tab, after filtering out OTUs that are present (having non-zero counts) 
in less than five samples. This filter reduced the number of OTUs from 813 to 593.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("sim.otu.tab5")
</code></pre>


<h3>Format</h3>

<p>A data frame with 100 observations on 593 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(sim.otu.tab5)
</code></pre>

<hr>
<h2 id='throat.meta'>Metadata of the throat microbiome samples</h2><span id='topic+throat.meta'></span>

<h3>Description</h3>

<p>This data set includes samples from the microbiome of the nasopharynx and oropharynx 
on each side of the body. It were generated to study the effect of smoking on the 
microbiota of the upper respiratory tract in 60 individuals, 28 smokers and 32 nonsmokers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("throat.meta")
</code></pre>


<h3>Format</h3>

<p>A data frame with 60 observations on 16 variables.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of Cigarette Smokers. PLoS ONE 5(12): e15216.
</p>


<h3>References</h3>

<p>R package &quot;GUniFrac&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.meta)
</code></pre>

<hr>
<h2 id='throat.otu.tab'>OTU count table from 16S sequencing of the throat microbiome samples</h2><span id='topic+throat.otu.tab'></span>

<h3>Description</h3>

<p>This data set contains 60 subjects with 28 smokers and 32 nonsmokers. 
Microbiome data were collected from right and left nasopharynx and oropharynx 
region to form an OTU table with 856 OTUs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("throat.otu.tab")
</code></pre>


<h3>Format</h3>

<p>A data frame with 60 observations on 856 variables.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of Cigarette Smokers. PLoS ONE 5(12): e15216.
</p>


<h3>References</h3>

<p>R package &quot;GUniFrac&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
</code></pre>

<hr>
<h2 id='throat.otu.tab5'>Filtered OTU count table from 16S sequencing of the throat microbiome samples</h2><span id='topic+throat.otu.tab5'></span>

<h3>Description</h3>

<p>This table was derived from throat.otu.tab, after filtering out OTUs that are present (having non-zero counts) 
in less than five samples. This filter reduced the number of OTUs from 856 to 233.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("throat.otu.tab5")
</code></pre>


<h3>Format</h3>

<p>A data frame with 60 observations on 233 variables.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab5)
</code></pre>

<hr>
<h2 id='throat.tree'>UPGMA tree of the OTUs from 16S sequencing of the throat microbiome samples</h2><span id='topic+throat.tree'></span>

<h3>Description</h3>

<p>The OTU tree is constructed using UPGMA on the K80 distance matrix of the OTUs. 
It is a rooted tree of class &quot;phylo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("throat.tree")
</code></pre>


<h3>Format</h3>

<p>List of 4 data frames.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of Cigarette Smokers. PLoS ONE 5(12): e15216.
</p>


<h3>References</h3>

<p>R package &quot;GUniFrac&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.tree)
</code></pre>

<hr>
<h2 id='unifrac.mean'>Expected value of the unweighted UniFrac distance matrix</h2><span id='topic+unifrac.mean'></span>

<h3>Description</h3>

<p>This function computes the expected value of the unweighted UniFrac distance matrix over rarefaction replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unifrac.mean(
  otu.table,
  tree,
  rarefy.depth = min(rowSums(otu.table)),
  first.order.approx.only = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="unifrac.mean_+3A_otu.table">otu.table</code></td>
<td>
<p>the <code>n.obs</code> by <code>n.otu</code> matrix of read counts.</p>
</td></tr>
<tr><td><code id="unifrac.mean_+3A_tree">tree</code></td>
<td>
<p>the phylogeneic tree.</p>
</td></tr>
<tr><td><code id="unifrac.mean_+3A_rarefy.depth">rarefy.depth</code></td>
<td>
<p>rarefaction depth. The default is the minimum library size observed in the OTU table.</p>
</td></tr>
<tr><td><code id="unifrac.mean_+3A_first.order.approx.only">first.order.approx.only</code></td>
<td>
<p>a logical value indicating whether to calculate the expected value 
using the first order approixmation by the delta method. The default is FALSE, 
using the second order approixmation.</p>
</td></tr>
<tr><td><code id="unifrac.mean_+3A_verbose">verbose</code></td>
<td>
<p>a logical value indicating whether to generate verbose output. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list consisting of 
</p>
<table role = "presentation">
<tr><td><code>unifrac.mean.o1</code></td>
<td>
<p>Expected unweighted UniFrac distance matrix by the first order approixmation.</p>
</td></tr>
<tr><td><code>unifrac.mean.o2</code></td>
<td>
<p>Expected unweighted UniFrac distance matrix by the second order approixmation.</p>
</td></tr> 
<tr><td><code>unifrac.mean.sq.o1</code></td>
<td>
<p>Expected squared unweighted UniFrac distance matrix by the first order approixmation.</p>
</td></tr> 
<tr><td><code>unifrac.mean.sq.o2</code></td>
<td>
<p>Expected squared unweighted UniFrac distance matrix by the second order approixmation.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Yi-Juan Hu &lt;yijuan.hu@emory.edu&gt;, Glen A. Satten &lt;gsatten@emory.edu&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab5)
data(throat.tree)
res.unifrac &lt;- unifrac.mean( throat.otu.tab5[1:20,], throat.tree)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
