<!DOCTYPE html><html><head><title>Help for package LEANR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LEANR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LEANR-package'>
<p>Finds &quot;local subnetworks&quot; within an interaction network which show enrichment for differentially expressed genes</p></a></li>
<li><a href='#CCM.pvals'>
<p>Gene p-value list derived from knock-out experiments of the three CCM genes</p></a></li>
<li><a href='#g_red'>
<p>igraph graph object used in unit tests</p></a></li>
<li><a href='#g2'>
<p>igraph graph object used in examples for function run.lean.fromdata</p></a></li>
<li><a href='#gene.annots'>
<p>Annotation for STRING protein Ids</p></a></li>
<li><a href='#gene.list.scores'>
<p>Gene p-value list used in examples for function run.lean.fromdata</p></a></li>
<li><a href='#get.ls.info'>
<p>Extract the genes of a &quot;local subnetwork&quot;&quot; around a given protein</p></a></li>
<li><a href='#pvals_red'>
<p>Gene p-value list used in unit tests</p></a></li>
<li><a href='#run.lean'>
<p>Run the LEAN approach</p></a></li>
<li><a href='#subnet.simulation'>
<p>Simulate subnetworks</p></a></li>
<li><a href='#write.ls.to.sif'>
<p>Extract the &quot;local subnetwork&quot; around a given protein</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Finds "Local Subnetworks" Within an Interaction Network which
Show Enrichment for Differentially Expressed Genes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-11-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Frederik Gwinner</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Frederik Gwinner &lt;frederik.gwinner@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements the method described in "Network-based analysis of omics data: The LEAN method" [Gwinner Boulday (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw676">doi:10.1093/bioinformatics/btw676</a>&gt;]
 Given a protein interaction network and a list of p-values describing a measure of interest (as e.g. differential gene expression) this method
 computes an enrichment p-value for the protein neighborhood of each gene and compares it to a background distribution of randomly drawn p-values.
 The resulting scores are corrected for multiple testing and significant hits are returned in tabular format.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-11-11 23:37:29 UTC; fgwinner</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.14), igraph(&ge; 0.7.1), foreach(&ge; 1.4.2)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, doMC, rmarkdown, ROCR, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-11-12 15:47:01</td>
</tr>
</table>
<hr>
<h2 id='LEANR-package'>
Finds &quot;local subnetworks&quot; within an interaction network which show enrichment for differentially expressed genes
</h2><span id='topic+LEAN-package'></span><span id='topic+LEANR'></span><span id='topic+LEAN'></span><span id='topic+Local+20enrichement+20analysis'></span>

<h3>Description</h3>

<p>Implements the method described in &quot;Network-based analysis of omics data: The LEAN method&quot;.
Given a protein interaction network and a list of p-values describing a measure of interest (as e.g. differential gene expression) 
this method computes an enrichment p-value for the protein neighborhood of each gene and compares it to a background distribution 
of randomly drawn p-values. The resulting scores are corrected for multiple testing and significant hits are returned in tabular format.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> LEANR</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.4.8</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-11-11</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>See help page of run.lean for a more detailed description of how to use this package. 
Type vignette(&quot;CCM-data&quot;) for an example showing the application of LEAN to the CCM knockout data set discussed in the paper.
Type vignette(&quot;subnet-sim&quot;) for an example showing the application of LEAN to simulated subnetwork data discussed in the paper.
</p>


<h3>Author(s)</h3>

<p>Frederik Gwinner
</p>
<p>Maintainer: Frederik Gwinner &lt;frederik.gwinner@gmail.com&gt;
</p>


<h3>References</h3>

<p>Gwinner et al., Network-based analysis of omics data: The LEAN method, Bioinformatics 2016
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run.lean">run.lean</a></code>
vignette(&quot;CCM-data&quot;)
vignette(&quot;subnet-sim&quot;)
</p>

<hr>
<h2 id='CCM.pvals'>
Gene p-value list derived from knock-out experiments of the three CCM genes
</h2><span id='topic+CCM.pvals'></span>

<h3>Description</h3>

<p>Gene p-value list derived from knock-out experiments of the three CCM genes CCM1, CCM2 and CCM3.
Contains p-values obtained from a limma differential expression analysis of knock-out samples versus control samples (each done in triplicate).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("CCM.pvals")</code></pre>


<h3>Format</h3>

<p>Named list (CCM1,2,3) of named numericals (names = gene ids, values = limma p-values) 
</p>

<hr>
<h2 id='g_red'>
igraph graph object used in unit tests
</h2><span id='topic+g_red'></span>

<h3>Description</h3>

<p>igraph graph object used in unit tests.
Obtained by restricting the graph &lt;g2&gt; to the graph induced by randomly selecting 1500 genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("g_red")</code></pre>


<h3>Format</h3>

<p>The format is:
IGRAPH UNW- 1500 2818 &ndash; 
</p>


<h3>Source</h3>

<p>STRING v9.1 Mouse filtered for confidence scores &gt;= 0.9; radnomly reduced to 1500 genes and all interactions between them
</p>

<hr>
<h2 id='g2'>
igraph graph object used in examples for function run.lean.fromdata
</h2><span id='topic+g2'></span>

<h3>Description</h3>

<p>igraph graph object used in examples for function run.lean.fromdata.
Obtained by parsing the STRING v.91 murine interaction network and restricting 
it to proteins mappable to genes contained on the Affymetrix MouseGene v1.0 ST chip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("g2")</code></pre>


<h3>Format</h3>

<p>The format is:
IGRAPH UNW- 7342 63617 &ndash; 
</p>


<h3>Source</h3>

<p>STRING v9.1 Mouse filtered for confidence scores &gt;= 0.9
</p>

<hr>
<h2 id='gene.annots'>
Annotation for STRING protein Ids
</h2><span id='topic+gene.annots'></span>

<h3>Description</h3>

<p>Annotation table giving gene names and descriptions for each protein contained in the STRING network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gene.annots")</code></pre>


<h3>Format</h3>

<p>A data frame with 7342 observations on the following 4 variables.
</p>

<dl>
<dt><code>ensembl_gene_id</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>mgi_symbol</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>entrezgene</code></dt><dd><p>a character vector</p>
</dd>
<dt><code>description</code></dt><dd><p>a character vector</p>
</dd>
</dl>



<h3>Details</h3>

<p>Row.names of the data.frame are STRING protein Ids
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gene.annots)
str(gene.annots)
</code></pre>

<hr>
<h2 id='gene.list.scores'>
Gene p-value list used in examples for function run.lean.fromdata
</h2><span id='topic+gene.list.scores'></span>

<h3>Description</h3>

<p>Gene p-value list used in examples for function run.lean.fromdata
Contains p-values obtained from a limma differential expression analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("gene.list.scores")</code></pre>


<h3>Format</h3>

<p>The format is:
Named num [1:7342] 0.772 0.813 0.979 0.841 0.607 ...
- attr(*, &quot;names&quot;)= chr [1:7342] &quot;10090.ENSMUSP00000000001&quot; &quot;10090.ENSMUSP00000010205&quot; &quot;10090.ENSMUSP00000053818&quot; &quot;10090.ENSMUSP00000000153&quot; ...
</p>

<hr>
<h2 id='get.ls.info'>
Extract the genes of a &quot;local subnetwork&quot;&quot; around a given protein 
</h2><span id='topic+get.ls.info'></span><span id='topic+local.subnetwork.info'></span>

<h3>Description</h3>

<p>Extract the genes of a &quot;local subnetwork&quot;&quot; around a given protein and present in tabular format
</p>


<h3>Usage</h3>

<pre><code class='language-R'> get.ls.info(prot_id, LEANres)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get.ls.info_+3A_prot_id">prot_id</code></td>
<td>

<p>Protein id compatible with node names used in graph.
</p>
</td></tr>
<tr><td><code id="get.ls.info_+3A_leanres">LEANres</code></td>
<td>

<p>LEAN result object (list) returned by &lt;run.lean&gt; or &lt;run.lean.fromdata&gt;
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Frederik Gwinner
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run.lean">run.lean</a></code>
</p>

<hr>
<h2 id='pvals_red'>
Gene p-value list used in unit tests
</h2><span id='topic+pvals_red'></span>

<h3>Description</h3>

<p>Gene p-value list used in unit tests
Contains p-values obtained on the CCM2 data for a radnom subselection of 1500 genes. 
To be used in conjunction with the network contained in &lt;g_red&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("pvals_red")</code></pre>


<h3>Format</h3>

<p>The format is:
Named num [1:1500] 0.5091 0.4833 0.0454 0.0814 0.0324 ...
- attr(*, &quot;names&quot;)= chr [1:1500] &quot;10090.ENSMUSP00000079341&quot; &quot;10090.ENSMUSP00000106951&quot; &quot;10090.ENSMUSP00000045284&quot; &quot;10090.ENSMUSP00000077744&quot; ...
</p>

<hr>
<h2 id='run.lean'>
Run the LEAN approach
</h2><span id='topic+lean'></span><span id='topic+run.lean'></span>

<h3>Description</h3>

<p>Apply the LEAN approach to a given network and a list of pvalues
</p>


<h3>Usage</h3>

<pre><code class='language-R'>run.lean(ranking, network, ranked = F, 
    add.scored.genes = F, keep.nodes.without.scores = F, 
    verbose = F, n_reps = 10000, bootstrap = F, ncores = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run.lean_+3A_ranking">ranking</code></td>
<td>

<p>Either a file containing gene p-values or a named numerical vector of p-values with names matching node names used in the network
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_network">network</code></td>
<td>

<p>Either a file containing the network in sif format or an igraph graph object representing the network
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_ranked">ranked</code></td>
<td>

<p>whether to transform input p-values into a uniformly distributed list of p-values based on the genes' rank before p* calculation
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_add.scored.genes">add.scored.genes</code></td>
<td>

<p>whether to create one singleton node for each gene with a score but not occurring in the graph
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_keep.nodes.without.scores">keep.nodes.without.scores</code></td>
<td>

<p>whether to keep nodes of the graph that have no recorded score. For those nodes it is still possible to compute enrichment scores if at least one of their network neighbors has a recorded score.
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_verbose">verbose</code></td>
<td>

<p>whether to print additional status messages
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_n_reps">n_reps</code></td>
<td>

<p>the number of samples each background distribution should consist of. Largely influences the run-time, but higher values needed for meaningful empirical pvalues !
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_bootstrap">bootstrap</code></td>
<td>

<p>whether to draw the pvalues of the background distributions with or without replacement
</p>
</td></tr>
<tr><td><code id="run.lean_+3A_ncores">ncores</code></td>
<td>

<p>number of cores to be used in parallel computation. Default (NULL) leads to automatic guessing of max number of cores to be used (depending on operating system).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing the results of the LEAN run. The list encompasses the following elements:
</p>
<table>
<tr><td><code>restab</code></td>
<td>
<p>Result table of applying LEAN to the real data</p>
</td></tr>
<tr><td><code>randtab</code></td>
<td>
<p>Result table of applying LEAN to a permuted p-value list</p>
</td></tr>
<tr><td><code>indGraph</code></td>
<td>
<p>igraph graph representing the input network after adapting it according to parameters &lt;add.scored.genes&gt;, &lt;keep.nodes.without.scores&gt; and the presence of gene scores in the input scores</p>
</td></tr>
<tr><td><code>nhs</code></td>
<td>
<p>The extracted local subnetworks. Encoded as a named (by protein/gene ids) list of igraph node indices detailing each evaluated local subnetwork</p>
</td></tr>
<tr><td><code>gene.scores</code></td>
<td>
<p>The gene p-values extracted from the input scores. Encoded as a numeric vector named with protein/gene ids</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Frederik Gwinner
</p>


<h3>References</h3>

<p>Gwinner et al., Network-based analysis of omics data: The LEAN method, MS submitted to Bioinformatics
</p>


<h3>See Also</h3>

<p><code><a href="#topic+LEANR-package">LEANR-package</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simple use case starting from a test network and p-value list
## Not run:  
# compute LEAN p-values starting from a p-value file and a network file
rank_file&lt;-system.file('extdata/pvals_red.txt.gz', package='LEANR')
net_file&lt;-system.file('extdata/g_red.sif.gz', package='LEANR')
system.time(res&lt;-run.lean(ranking=rank_file, network=net_file, 
  add.scored.genes=T, verbose=T, n_reps=1000, ncores=3))

# compute LEAN p-values starting from a list of gene scores and a graph
data(pvals_red)
data(g_red)
system.time(res2&lt;-run.lean(ranking=pvals_red, network=g_red, verbose=T, 
   n_reps=1000, ncores=3))


## End(Not run) 
</code></pre>

<hr>
<h2 id='subnet.simulation'>
Simulate subnetworks
</h2><span id='topic+subnet.simulation'></span>

<h3>Description</h3>

<p>Simulate subnetworks (also called modules) and gene p-values to be then used in a ROC performance evaluation study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subnet.simulation(g, nmods=10, mod_lims=c(10,50), pval_scaling=0.1, mod_enrich_perc=0.5, 
         spec='', prob_function=function(degs){degs/sum(degs)}, create.files=T)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subnet.simulation_+3A_g">g</code></td>
<td>

<p>igraph graph representing the network in which subnetworks are supposed to be simulated
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_nmods">nmods</code></td>
<td>

<p>number of subnetworks/modules to simulate
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_mod_lims">mod_lims</code></td>
<td>

<p>minimum and maximum size (number of genes) of each module
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_pval_scaling">pval_scaling</code></td>
<td>

<p>parameter value for &lt;p_scale&gt;
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_mod_enrich_perc">mod_enrich_perc</code></td>
<td>

<p>parameter value for &lt;p_enrich&gt;
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_spec">spec</code></td>
<td>

<p>string, specifier appended to the created pvalue files (if create.files=T)
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_prob_function">prob_function</code></td>
<td>

<p>probability function used for picking attachment point in iterative construction of subnetworks. defaults to preferential attachment based on node degree. To disable preferential attachment, use prob_function=function(degs)rep(1/length(degs),length(degs))
</p>
</td></tr>
<tr><td><code id="subnet.simulation_+3A_create.files">create.files</code></td>
<td>

<p>whether to write subnetwork simulation results to file so external approaches can be run and evaluated on them
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object containing the simulated subnetworks. The list encompasses the following elements:
</p>
<table>
<tr><td><code>mods</code></td>
<td>
<p>List of simulated modules/subnetworks. Each module is given by the igraph indices of the nodes contained in it.</p>
</td></tr>
<tr><td><code>pvals</code></td>
<td>
<p>Result table containing for each gene in the graph its simulated pvalue (column P.Value) and its association to subnetworks or background (column NodeType)</p>
</td></tr>
<tr><td><code>pvalfile</code></td>
<td>
<p>String containing the name of the file containing the equivalent information to &lt;pvals&gt; created in this run if create.files=T</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Frederik Gwinner
</p>


<h3>References</h3>

<p>Gwinner et al., Network-based analysis of omics data: The LEAN method, MS submitted to Bioinformatics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### See vignette("subnet-sim") for a use case.
</code></pre>

<hr>
<h2 id='write.ls.to.sif'>
Extract the &quot;local subnetwork&quot; around a given protein
</h2><span id='topic+write.ls.to.sif'></span><span id='topic+local.subnetwork.export'></span>

<h3>Description</h3>

<p>Extract the &quot;local subnetwork&quot; around a given protein and write it to a Cytoscape-readable .sif file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write.ls.to.sif(prot_id, LEANres, outfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write.ls.to.sif_+3A_prot_id">prot_id</code></td>
<td>

<p>protein id compatible with node names used in graph <code>g</code>
</p>
</td></tr>
<tr><td><code id="write.ls.to.sif_+3A_leanres">LEANres</code></td>
<td>

<p>LEAN result object (list) returned by &lt;run.lean&gt; or &lt;run.lean.fromdata&gt;
</p>
</td></tr>
<tr><td><code id="write.ls.to.sif_+3A_outfile">outfile</code></td>
<td>

<p>character string describing the location of an output file. Should end in .sif to be able to load it in Cytoscape.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Frederik Gwinner
</p>


<h3>See Also</h3>

<p><code><a href="#topic+run.lean">run.lean</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
