<!DOCTYPE html><html><head><title>Help for package learnPopGen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {learnPopGen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#clt'><p>Illustrates the concept of the Central Limit Theorem</p></a></li>
<li><a href='#coalescent.plot'><p>Creates a (usually animated) simulation of gene coalescence within a population</p></a></li>
<li><a href='#drift.selection'><p>Simulation of genetic drift &amp; natural selection at a biallelic locus</p></a></li>
<li><a href='#founder.event'><p>Simulation of a founder event or population bottleneck</p></a></li>
<li><a href='#freqdep'><p>Numerical analysis of a frequency dependent seletion model</p></a></li>
<li><a href='#genetic.drift'><p>Genetic drift simulation</p></a></li>
<li><a href='#hardy.weinberg'><p>Computes Hardy-Weinberg frequencies for a multiallelic locus or across multiple loci</p></a></li>
<li><a href='#hawk.dove'><p>Analysis of hawk-dove game theoretic model</p></a></li>
<li><a href='#msd'><p>Migration, drift, and selection</p></a></li>
<li><a href='#mutation.selection'><p>Gene frequencies over time under mutation-selection balance</p></a></li>
<li><a href='#phenotype.freq'><p>Computes phenotypic distribution and its change through time due to natural selection on a polygenic trait</p></a></li>
<li><a href='#rcd'><p>Simulation of reproductive character displacement in an ecological community</p></a></li>
<li><a href='#selection'><p>Numerical analysis of biallelic locus frequency independent selection</p></a></li>
<li><a href='#sexratio'><p>Hypothetical analysis of frequency dependent selection on a sex determining genetic locus</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-05-20</td>
</tr>
<tr>
<td>Title:</td>
<td>Population Genetic Simulations &amp; Numerical Analysis</td>
</tr>
<tr>
<td>Author:</td>
<td>Liam J. Revell</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Liam J. Revell &lt;liam.revell@umb.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>grDevices, graphics, gtools, methods, phytools, stats</td>
</tr>
<tr>
<td>ZipData:</td>
<td>no</td>
</tr>
<tr>
<td>Description:</td>
<td>Conducts various numerical analyses and simulations in population genetics and evolutionary theory, primarily for the purpose of teaching (and learning about) key concepts in population &amp; quantitative genetics, and evolutionary theory.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://github.com/liamrevell/learnPopGen">http://github.com/liamrevell/learnPopGen</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-05-20 20:17:27 UTC; liamj</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-05-20 20:40:03 UTC</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='clt'>Illustrates the concept of the Central Limit Theorem</h2><span id='topic+clt'></span><span id='topic+print.clt'></span><span id='topic+plot.clt'></span>

<h3>Description</h3>

<p>The <em>Central Limit Theorem</em> tells us that when independent random variables are added together, the distribution of their sum tends towards a normal distribution, regardless of the shape of their individual distributions. This function attempts to illustrate this concept by allowing the user to visualize the sum of an arbitrary number of different independent random variables with different underlying distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	clt(nvar=1, nobs=1000, df=c("normal","uniform","exponential","binomial"),
		theta=NULL,	breaks="Sturges", show=c("sum","mean"))
	## S3 method for class 'clt'
print(x, ...)
	## S3 method for class 'clt'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clt_+3A_nvar">nvar</code></td>
<td>
<p>number of random variables to sum (1 or more).</p>
</td></tr>
<tr><td><code id="clt_+3A_nobs">nobs</code></td>
<td>
<p>total number of observations (per random variable).</p>
</td></tr>
<tr><td><code id="clt_+3A_df">df</code></td>
<td>
<p>distribution functions of individual random variables to sum. These can be <code>"normal"</code>, <code>"uniform"</code>, <code>"exponential"</code>, or <code>"binomial"</code>.</p>
</td></tr>
<tr><td><code id="clt_+3A_theta">theta</code></td>
<td>
<p>parameter of the distribution functions: variance in the case of <code>df = "normal"</code>, maximum value in the case of <code>"uniform"</code> (the minimum value will be assumed to be 0), rate in the case of <code>"exponential"</code>.</p>
</td></tr>
<tr><td><code id="clt_+3A_breaks">breaks</code></td>
<td>
<p>breaks (see <code><a href="graphics.html#topic+hist">hist</a></code>).</p>
</td></tr>
<tr><td><code id="clt_+3A_show">show</code></td>
<td>
<p>whether to show the row-wise <em>sum</em> of the independent random variables (<code>show="sum"</code>), or their mean (<code>show="mean"</code>).</p>
</td></tr>
<tr><td><code id="clt_+3A_x">x</code></td>
<td>
<p>object of class <code>"clt"</code> for <code>print</code> and <code>plot</code> methods.</p>
</td></tr>
<tr><td><code id="clt_+3A_...">...</code></td>
<td>
<p>optional arguments for <code>print</code> and <code>plot</code> methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The central limit theorem (CLT) establishes that when independent random variables are added together their (normalized) sum will tend towards a normal distribution, regardless of the distribution of the original random variables. That is to say if we were to generate a set of <code>nvar</code> (say) independent, uniform, random variables, normalize each one to have the same variance, and then sum or average the variables by observation, this sum or average will tend towards a normal distribution as the number of random variables (<code>nvar</code> in this function) is increased.
</p>


<h3>Value</h3>

<p>Creates a plot showing the observation-wise distribution of the sum or average of the independent random variables.
</p>
<p>The distribution of the observation-wise sum or average and the underlying data are also returned invisibly to the user in the form of an object of class <code>"clt"</code>. This object can in turn be printed or re-plotted using custom <code>print</code> and <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+phenotype.freq">phenotype.freq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	clt(nvar=1,df="exponential")
	clt(nvar=10,df="exponential")
	object&lt;-clt(nvar=40,df="exponential")
	print(object)
	plot(object)
</code></pre>

<hr>
<h2 id='coalescent.plot'>Creates a (usually animated) simulation of gene coalescence within a population</h2><span id='topic+coalescent.plot'></span><span id='topic+print.coalescent.plot'></span><span id='topic+plot.coalescent.plot'></span>

<h3>Description</h3>

<p>Coalescence or coalescent theory is a model for genetic drift within a population in which we envision gene copies merging or &quot;coalescing&quot; into ancestors in the past. This function generates a(n) (optionally animated) visualization of this process of coalescence within a population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	coalescent.plot(n=10, ngen=20, colors=NULL, ...)
	## S3 method for class 'coalescent.plot'
print(x, ...)
	## S3 method for class 'coalescent.plot'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coalescent.plot_+3A_n">n</code></td>
<td>
<p>number of haploid individuals or gene copies.</p>
</td></tr>
<tr><td><code id="coalescent.plot_+3A_ngen">ngen</code></td>
<td>
<p>number of generations.</p>
</td></tr>
<tr><td><code id="coalescent.plot_+3A_colors">colors</code></td>
<td>
<p>colors to use for plotting individuals and lines. By default, the function tries to use a contrasting color scheme such that adjacent allele copies are dissimilar (to facilitate visualization of the coalescent process.)</p>
</td></tr>
<tr><td><code id="coalescent.plot_+3A_x">x</code></td>
<td>
<p>object of class <code>"coalescent.plot"</code> for <code>print</code> and <code>plot</code> methods.</p>
</td></tr>
<tr><td><code id="coalescent.plot_+3A_...">...</code></td>
<td>
<p>optional arguments. For <code>coalescent.plot</code> optional arguments include: <code>sleep</code>, the time to pause between generations (set to 0.2s by default); <code>lwd</code>, the line width for parent-offspring lines in the coalescent geneology; and <code>col.order</code>, if <code>colors=NULL</code>, whether to use 'sequential' (<code>col.order="sequential"</code>, the default) or 'alternating' (<code>col.order="alternating"</code>) colors for adjacent alleles.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot or animation.
</p>
<p>Invisibly returns an object of class <code>"coalescent.plot"</code> containing the alleles (coded numerically) and the parent-offspring relationships from the coalescent simulation. This object can be printed or re-plotted using <code>print</code> and <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+drift.selection">drift.selection</a></code>, <code><a href="#topic+genetic.drift">genetic.drift</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	coalescent.plot()
	## Not run: 
		coalescent.plot(n=20,ngen=30,col.order="alternating")
		object&lt;-coalescent.plot()
		print(object)
		plot(object)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='drift.selection'>Simulation of genetic drift &amp; natural selection at a biallelic locus</h2><span id='topic+drift.selection'></span>

<h3>Description</h3>

<p>Simulates drift and natural selection at a single biallelic locus within one or various populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	drift.selection(p0=0.5, Ne=100, w=c(1,1,1), ngen=400, nrep=10, colors=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drift.selection_+3A_p0">p0</code></td>
<td>
<p>starting frequency for the <em>A</em> allele.</p>
</td></tr>
<tr><td><code id="drift.selection_+3A_ne">Ne</code></td>
<td>
<p>effective population size.</p>
</td></tr>
<tr><td><code id="drift.selection_+3A_w">w</code></td>
<td>
<p>fitnesses of the three genotypes: <em>AA</em>, <em>Aa</em>, and <em>aa</em>.</p>
</td></tr>
<tr><td><code id="drift.selection_+3A_nrep">nrep</code></td>
<td>
<p>number of replicate simulations.</p>
</td></tr>
<tr><td><code id="drift.selection_+3A_ngen">ngen</code></td>
<td>
<p>total time, in number of generations, for the simulation.</p>
</td></tr>
<tr><td><code id="drift.selection_+3A_colors">colors</code></td>
<td>
<p>colors to use for plotting.</p>
</td></tr>
<tr><td><code id="drift.selection_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently the only arguments are <code>type</code> (e.g., <code>"l"</code>, <code>"s"</code>) and <code>lwd</code> in the <code>plot</code> method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a plot and returns an object of class <code>"drift.selelction"</code> consisting of list containing the allele frequency through time for each simulation. This object can be printed or plotted using corresponding methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+genetic.drift">genetic.drift</a></code>, <code><a href="#topic+selection">selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	drift.selection()
	p&lt;-drift.selection(p0=0.01,Ne=100,w=c(1,0.9,0.8),ngen=200,nrep=5)
	print(p)
	plot(p)
</code></pre>

<hr>
<h2 id='founder.event'>Simulation of a founder event or population bottleneck</h2><span id='topic+founder.event'></span>

<h3>Description</h3>

<p>This function simulates genetic drift with a founding event / population bottleneck at time <code>etime</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	founder.event(p0=0.5, Ne=1000, Nf=10, ttime=100, etime=50, show="p", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="founder.event_+3A_p0">p0</code></td>
<td>
<p>Starting frequency for the <em>A</em> allele.</p>
</td></tr>
<tr><td><code id="founder.event_+3A_ne">Ne</code></td>
<td>
<p>Effective population size at the start of the simulation and after the founding event.</p>
</td></tr>
<tr><td><code id="founder.event_+3A_nf">Nf</code></td>
<td>
<p>Size of the founding population.</p>
</td></tr>
<tr><td><code id="founder.event_+3A_ttime">ttime</code></td>
<td>
<p>Total time of the simulation.</p>
</td></tr>
<tr><td><code id="founder.event_+3A_etime">etime</code></td>
<td>
<p>Time for the founding event. Can either be a single generation, or a sequence of generations (e.g., <code>etime=40:50</code>) for a prolonged founder event. Note that to simulate a prolonged bottleneck the user <em>must</em> supply the sequence of generations during which the population is to be bottlenecked and <em>not</em> merely the start and end times of the bottleneck.</p>
</td></tr>
<tr><td><code id="founder.event_+3A_show">show</code></td>
<td>
<p>Two different options for plotting. <code>"p"</code> shows the frequency of the A allele through time; <code>"var"</code> shows the genetic variation in the population, calculated as p*(1-p). The default is <code>show="p"</code>.</p>
</td></tr>
<tr><td><code id="founder.event_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently, the only optional argument in <code>founder.event</code> is <code>ltype</code> which specifies the line type and defaults to <code>"s"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates one of two different plots, depending on the value of <code>show</code>.
</p>
<p>The function also invisibly returns an object of class <code>"founder.event"</code> which can be printed or plotted using corresponding <code>print</code> and <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+drift.selection">drift.selection</a></code>, <code><a href="#topic+genetic.drift">genetic.drift</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	founder.event()
	p&lt;-founder.event(show="variation")
	print(p)
	plot(p,show="p",ltype="l")
</code></pre>

<hr>
<h2 id='freqdep'>Numerical analysis of a frequency dependent seletion model</h2><span id='topic+freqdep'></span>

<h3>Description</h3>

<p>This function performs numerical analysis of a frequency dependent selection model based on Rice (2004; <em>Evolutionary Theory: Mathematical &amp; Conceptual Foundations</em>). The fitnesses of the three genotypes in the model are as follows, where f(<em>XX</em>) denotes the frequency of the <em>XX</em> genotype: w(<em>AA</em>)=1-3*f(<em>Aa</em>)+3*f(<em>aa</em>); w(<em>Aa</em>)=1-s*f(<em>Aa</em>); and w(<em>aa</em>)=1-3*f(<em>Aa</em>)+3*f(<em>AA</em>). As shown in Rice (2004), though entirely deterministic, the model can exhibit chaotic behavior under some values for <code>s</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	freqdep(p0=0.01, s=0, time=100, show="p", pause=0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="freqdep_+3A_p0">p0</code></td>
<td>
<p>Starting frequency for the <em>A</em> allele.</p>
</td></tr>
<tr><td><code id="freqdep_+3A_s">s</code></td>
<td>
<p>Parameter that determines the strength of selection against heterozygotes when they are common.</p>
</td></tr>
<tr><td><code id="freqdep_+3A_time">time</code></td>
<td>
<p>Number of generations to run the analysis.</p>
</td></tr>
<tr><td><code id="freqdep_+3A_show">show</code></td>
<td>
<p>Various options for plotting. <code>"p"</code> shows the frequency of A through time; <code>"q"</code> gives the frequency of the <em>a</em> allele; <code>"fitness"</code> gives the mean population fitness through time; <code>"surface"</code> plots the mean fitness as a function of p; <code>"deltap"</code> shows the change in p as a function of p; <code>"cobweb"</code> creates a cobweb plot showing p(t) by p(t+1). The default is <code>show="p"</code>.</p>
</td></tr>
<tr><td><code id="freqdep_+3A_pause">pause</code></td>
<td>
<p>Pause between generations. <code>pause=0.01</code> (for instance) might smooth animation.</p>
</td></tr>
<tr><td><code id="freqdep_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently, the only optional argument in <code>freqdep</code> is <code>color</code>, which can be used to change the color of the lines of the plot. The <code>plot</code> method can also accept the optional arguments <code>type</code> (e.g., <code>"l"</code> or <code>"s"</code>) and <code>lwd</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates one of several possible plots, depending on the value of <code>show</code>.
</p>
<p>The use of cobweb plots follows <code><a href="#topic+selection">selection</a></code>.
</p>
<p>The function also invisibly returns an object of class <code>"freqdep"</code> containing the frequency of the allele A through time, if this was calculated by the selected method. This can be printed or plotted using the corresponding methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+sexratio">sexratio</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	freqdep(time=100)
	freqdep(s=1.5,time=100)
	p&lt;-freqdep(s=2,show="cobweb",time=100)
	plot(p)
</code></pre>

<hr>
<h2 id='genetic.drift'>Genetic drift simulation</h2><span id='topic+genetic.drift'></span>

<h3>Description</h3>

<p>This function simulates genetic drift at a biallelic genetic locus with no selection and no mutation in a sexually reproducing diploid population or set of populations. It is essentially redundant with <code><a href="#topic+drift.selection">drift.selection</a></code>, but in which there is no difference in relative fitness among genotypes; however, it also allows the user to visualize heterozygosity or genetic variation through time - options that are not yet implemented in <code><a href="#topic+drift.selection">drift.selection</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	genetic.drift(p0=0.5, Ne=20, nrep=10, time=100, show="p", pause=0.1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetic.drift_+3A_p0">p0</code></td>
<td>
<p>Starting frequency for the A allele.</p>
</td></tr>
<tr><td><code id="genetic.drift_+3A_ne">Ne</code></td>
<td>
<p>Effective population size.</p>
</td></tr>
<tr><td><code id="genetic.drift_+3A_nrep">nrep</code></td>
<td>
<p>Number of replicate simulations.</p>
</td></tr>
<tr><td><code id="genetic.drift_+3A_time">time</code></td>
<td>
<p>Total time, in number of generations, for the simulation.</p>
</td></tr>
<tr><td><code id="genetic.drift_+3A_show">show</code></td>
<td>
<p>Various options for plotting. <code>"p"</code> shows the frequency of the A allele through time; <code>"genotypes"</code> creates an animated histogram with the frequencies of each of the three genotypes through time; <code>"fixed"</code> shows the fraction of populations that have fixed for each allele, a or A; <code>"heterozygosity"</code> plots the mean heterozygosity and the expected heterozygosity through time. The default is <code>show="p"</code>.</p>
</td></tr>
<tr><td><code id="genetic.drift_+3A_pause">pause</code></td>
<td>
<p>Pause between generations. <code>pause=0.01</code> (for instance) might smooth animation.</p>
</td></tr>
<tr><td><code id="genetic.drift_+3A_...">...</code></td>
<td>
<p>optional arguments. In <code>genetic.drift</code> the optional arguments are presently: <code>colors</code> (a vector giving the colors to be used to graph the various simulations); and <code>lwd</code>. The <code>plot</code> method of the object class adds the optional argument <code>type</code> (e.g., <code>"l"</code> or <code>"s"</code>.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates one of several possible plots, depending on the value of <code>show</code>.
</p>
<p>The function also invisibly returns an object of class <code>"genetic.drift"</code> that can be printed or re-plotted by the user using corresponding <code>print</code> and <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+drift.selection">drift.selection</a></code>, <code><a href="#topic+founder.event">founder.event</a></code>, <code><a href="#topic+selection">selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
		genetic.drift()
		object&lt;-genetic.drift(p0=0.7,show="heterozygosity")
		plot(object,show="genotypes")
	
## End(Not run)
</code></pre>

<hr>
<h2 id='hardy.weinberg'>Computes Hardy-Weinberg frequencies for a multiallelic locus or across multiple loci</h2><span id='topic+hardy.weinberg'></span><span id='topic+multilocus.hw'></span>

<h3>Description</h3>

<p><code>hardy.weinberg</code> computes Hardy-Weinberg frequencies for a multiallelic locus using arbitrary allele frequencies.
</p>
<p><code>multilocus.hw</code> computes multilocus Hardy-Weinberg frequencies for a set of biallelic loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	hardy.weinberg(p=c(0.5,0.5), alleles=c("A","a"), as.matrix=FALSE)
	multilocus.hw(nloci=2, p=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hardy.weinberg_+3A_p">p</code></td>
<td>
<p>allele frequencies. In the case of <code>multilocus.hw</code> the frequencies of the <em>dominant</em> (in this case, merely uppercase) allele at each locus.</p>
</td></tr>
<tr><td><code id="hardy.weinberg_+3A_alleles">alleles</code></td>
<td>
<p>names of the alleles.</p>
</td></tr>
<tr><td><code id="hardy.weinberg_+3A_as.matrix">as.matrix</code></td>
<td>
<p>logical argument indicating whether to return the result in the form of a matrix (if <code>TRUE</code>) or a vector.</p>
</td></tr>
<tr><td><code id="hardy.weinberg_+3A_nloci">nloci</code></td>
<td>
<p>for <code>multilocus.hw</code> the number of loci.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a matrix or vector.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+phenotype.freq">phenotype.freq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	hardy.weinberg()
	hardy.weinberg(p=c(0.4,0.3,0.2,0.1),alleles=letters[1:4])
</code></pre>

<hr>
<h2 id='hawk.dove'>Analysis of hawk-dove game theoretic model</h2><span id='topic+hawk.dove'></span>

<h3>Description</h3>

<p>This function performs numerical analysis of a discrete-time hawk-dove model in which &quot;payoff&quot; determines relative fitness in the population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	hawk.dove(p=c(0.01,0.99), M=NULL, time=100)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hawk.dove_+3A_p">p</code></td>
<td>
<p>Starting frequency of hawk &amp; dove phenotypes, respectively. Should correspond with the rows of <code>M</code>. If a single value is given then <code>p</code> will automatically be set to <code>p=c(p,1-p)</code>.</p>
</td></tr>
<tr><td><code id="hawk.dove_+3A_m">M</code></td>
<td>
<p>Payoff matrix. <code>M[i,j]</code> should contain the fitness of <code>i</code> when interacting with <code>j</code>.</p>
</td></tr>
<tr><td><code id="hawk.dove_+3A_time">time</code></td>
<td>
<p>Number of generations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a two panel plot. The upper panel shows the relative frequencies of each of the two interacting phenotypes. The lower panel shows mean fitness of the population and of each morph through time.
</p>
<p>The function also invisibly returns an object of class <code>"hawk.dove"</code> containing the frequencies of each strategy through time and their fitnesses. This object can be printed or re-plotted using corresponding <code>print</code> and <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+freqdep">freqdep</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	hawk.dove(time=60)
	Payoff&lt;-matrix(c(0.5,0.6,1.5,1.0),2,2)
	object&lt;-hawk.dove(M=Payoff,time=60)
	print(object)
	plot(object)
</code></pre>

<hr>
<h2 id='msd'>Migration, drift, and selection</h2><span id='topic+msd'></span>

<h3>Description</h3>

<p>Simulates migration, natural selection, and genetic drift. Selection can be in opposite directions in the two populations experiencing gene flow.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	msd(p0=c(0.5,0.5), Ne=c(100,100), w=list(c(1,1,1),c(1,1,1)), 
		m=c(0.01,0.01), ngen=400, colors=c("red","blue"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="msd_+3A_p0">p0</code></td>
<td>
<p>starting frequency for the <em>A</em> allele in each of two populations.</p>
</td></tr>
<tr><td><code id="msd_+3A_ne">Ne</code></td>
<td>
<p>effective population size for each of two populations.</p>
</td></tr>
<tr><td><code id="msd_+3A_w">w</code></td>
<td>
<p>fitnesses of the three genotypes (<em>AA</em>, <em>Aa</em>, and <em>aa</em>, in that order) in each of the two populations. <code>w</code> should take the form of a list of two vectors.</p>
</td></tr>
<tr><td><code id="msd_+3A_m">m</code></td>
<td>
<p>rates of migration <em>from</em> the first population to the second, and from the second population to the first, in that order. This value is best interpreted as the <em>probability</em> that an individual born in population 1 will migrate to population 2 before reproduction, and <em>vice versa</em>.</p>
</td></tr>
<tr><td><code id="msd_+3A_ngen">ngen</code></td>
<td>
<p>total time, in number of generations, for the simulation.</p>
</td></tr>
<tr><td><code id="msd_+3A_colors">colors</code></td>
<td>
<p>colors to use for plotting.</p>
</td></tr>
<tr><td><code id="msd_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently, the only optional argument for <code>msd</code> is <code>show.legend</code> (which defaults to <code>TRUE</code>). The <code>plot</code> method adds the additional optional arguments of <code>colors</code> (a vector of colors for the two simulated populations), <code>lwd</code>, and <code>type</code> (e.g., <code>"l"</code> or <code>"s"</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates a plot and invisibly returns a list containing the allele frequency through time for each of the two simulated populations.
</p>
<p>The returned object is of class <code>"msd"</code> and can be printed or re-plotted using corresponding <code>print</code> or <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+drift.selection">drift.selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	msd()
	msd(p0=c(0.25,0.75),w=list(c(1,0.9,0.8),c(0.8,0.9,1)))
	object&lt;-msd(p0=c(0.75,0.25),w=list(c(1,0.9,0.8),
		c(0.8,0.9,1)),m=c(0.1,0.1),ngen=100)
	print(object)
	plot(object,colors=c("black","grey"),lwd=4,type="s")
</code></pre>

<hr>
<h2 id='mutation.selection'>Gene frequencies over time under mutation-selection balance</h2><span id='topic+mutation.selection'></span>

<h3>Description</h3>

<p>This function performs numerical analysis of mutation-selection balance with mutation from <em>A</em> to <em>a</em> and selection against (either or both of) <em>Aa</em> and <em>aa</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	mutation.selection(p0=1.0, w=c(1,0), u=0.001, time=100, show="q", pause=0, 
		ylim=c(0,1))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mutation.selection_+3A_p0">p0</code></td>
<td>
<p>Starting frequency for the <em>A</em> allele.</p>
</td></tr>
<tr><td><code id="mutation.selection_+3A_w">w</code></td>
<td>
<p>Fitnesses of the heterozygote (<em>Aa</em>) and homozygote deleterious (<em>aa</em>) genotypes. The fitness of genotype <em>AA</em> is assumed to be 1.0.</p>
</td></tr>
<tr><td><code id="mutation.selection_+3A_u">u</code></td>
<td>
<p>Rate at which <em>A</em> alleles are converted to <em>a</em> alleles by mutation.</p>
</td></tr>
<tr><td><code id="mutation.selection_+3A_time">time</code></td>
<td>
<p>Number of generations to run the analysis.</p>
</td></tr>
<tr><td><code id="mutation.selection_+3A_show">show</code></td>
<td>
<p>Two options for plotting. <code>"q"</code> shows the frequency of a through time; <code>"fitness"</code> plots the mean fitness over time. The default is <code>show="q"</code>.</p>
</td></tr>
<tr><td><code id="mutation.selection_+3A_pause">pause</code></td>
<td>
<p>Pause between generations. <code>pause=0.01</code> (for instance) might smooth animation.</p>
</td></tr>
<tr><td><code id="mutation.selection_+3A_ylim">ylim</code></td>
<td>
<p>Limits on the <em>y</em>-axis for plotting.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates one of three possible plots, depending on the value of <code>show</code>.
</p>
<p>The function also invisibly returns the frequency of the A allele through time and the mean population fitness as an object of class <code>"mutation.selection"</code> that can be printed or re-plotted with associated <code>print</code> and <code>plot</code> methods, respectively. The <code>plot</code> method also permits user control over various attributes of the appearance of the plot, such as the color of the plotted lines (<code>color</code>), the line widths (<code>lwd</code>), the limits of the <em>y</em>-axis (<code>ylim</code>), and the type of line (e.g., <code>"l"</code> vs. <code>"s"</code>, via the argument <code>type</code>).
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+genetic.drift">genetic.drift</a></code>, <code><a href="#topic+selection">selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	mutation.selection(w=c(1,0),time=100,ylim=c(0,0.1))
</code></pre>

<hr>
<h2 id='phenotype.freq'>Computes phenotypic distribution and its change through time due to natural selection on a polygenic trait</h2><span id='topic+phenotype.freq'></span><span id='topic+phenotype.selection'></span>

<h3>Description</h3>

<p><code>phenotype.freq</code> computes the phenotypic trait distribution for a polygenic trait. Can be used to demonstrate that the phenotypic distribution of a polygenic trait will tend to normality as the number of loci is increased, regardless of the allele frequencies at each locus.
</p>
<p><code>phenotype.selection</code> computes the change in the phenotypic trait distribution through time under natural selection. Can be used to show that natural selection on a polygenic trait can move the value of the trait well beyond its original distribution in the population.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	phenotype.freq(nloci=6, p=NULL, effect=1/nloci)
	phenotype.selection(nloci=6, p=NULL, effect=1/nloci, beta=0.1, ngen=20, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phenotype.freq_+3A_nloci">nloci</code></td>
<td>
<p>number of loci. For simplicity all loci are assumed to be biallelic.</p>
</td></tr>
<tr><td><code id="phenotype.freq_+3A_p">p</code></td>
<td>
<p>allele frequency, <em>p</em>, for each locus, in a vector. If not supplied, initially frequencies will be assumed to be 0.5 at all loci.</p>
</td></tr>
<tr><td><code id="phenotype.freq_+3A_effect">effect</code></td>
<td>
<p>additive effect of an allele substitution. For simplicity, this is assumed to be the same at all loci.</p>
</td></tr>
<tr><td><code id="phenotype.freq_+3A_beta">beta</code></td>
<td>
<p>selection gradient.</p>
</td></tr>
<tr><td><code id="phenotype.freq_+3A_ngen">ngen</code></td>
<td>
<p>number of generations to analyze.</p>
</td></tr>
<tr><td><code id="phenotype.freq_+3A_...">...</code></td>
<td>
<p>optional arguments. Presently the only optional argument in the function <code>phenotype.selection</code> is <code>sleep</code>, which can be used to specify the time delay in seconds between generations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Creates a plot or animation.
</p>
<p><code>phenotype.freq</code> also invisibly returns an object of class <code>"phenotype.freq"</code> that can be printed or re-plotted using <code>print</code> and <code>plot</code> methods corresponding to the object type. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+clt">clt</a></code>, <code><a href="#topic+selection">selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
		phenotype.freq(n=4)
		object&lt;-phenotype.freq(nloci=6,p=runif(n=6),effect=1/6)
		print(object)
		plot(object)
		object&lt;-phenotype.freq(nloci=10,p=runif(n=10),effect=rexp(n=10))
		print(object)
		plot(object)
		phenotype.selection(ngen=100)
	
## End(Not run)
</code></pre>

<hr>
<h2 id='rcd'>Simulation of reproductive character displacement in an ecological community</h2><span id='topic+rcd'></span>

<h3>Description</h3>

<p>This function conducts individual-based, genetically explicit numerical simulation of reproductive character displacement in an ecological community. The model is one of multiple species (with fixed relative abundance) competing to utilize the same signal space. There is both stabilizing selection on the signal trait for detectability, as well as (in multi-species simulations) countervailing selection for divergence due to the costs of erroneous mismating attempts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	rcd(nsp=3, nindivs=c(700,400,100), w_t=10, gen=c(500,500), figs="on", pf=100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rcd_+3A_nsp">nsp</code></td>
<td>
<p>Number of species in the simulation. If <code>figs="on"</code>, <code>nsp</code> must be 1, 2, or 3.</p>
</td></tr>
<tr><td><code id="rcd_+3A_nindivs">nindivs</code></td>
<td>
<p>A vector of length <code>nsp</code> containing the integer number of individuals in each species of the simulation.</p>
</td></tr>
<tr><td><code id="rcd_+3A_w_t">w_t</code></td>
<td>
<p>Shape parameter of the Gaussian selection surface for the male signalling trait.</p>
</td></tr>
<tr><td><code id="rcd_+3A_gen">gen</code></td>
<td>
<p>Vector containing the number of allopatric generations followed by the number of sympatric generations for simulation.</p>
</td></tr>
<tr><td><code id="rcd_+3A_figs">figs</code></td>
<td>
<p>Either <code>"on"</code> if plotting is turned on, or <code>"off"</code> to suppress plotting.</p>
</td></tr>
<tr><td><code id="rcd_+3A_pf">pf</code></td>
<td>
<p>Print frequency for the simulation status to screen.</p>
</td></tr>
<tr><td><code id="rcd_+3A_...">...</code></td>
<td>
<p>Optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list containing the mean male signal trait and the mean female preference over time. It also (optionally) plots these.
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+drift.selection">drift.selection</a></code>, <code><a href="#topic+genetic.drift">genetic.drift</a></code>, <code><a href="#topic+freqdep">freqdep</a></code>, <code><a href="#topic+selection">selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	## Not run: 
		obj&lt;-rcd(nsp=2,nindivs=c(500,500))
	
## End(Not run)
</code></pre>

<hr>
<h2 id='selection'>Numerical analysis of biallelic locus frequency independent selection</h2><span id='topic+selection'></span>

<h3>Description</h3>

<p>This function performs numerical analysis of a simple biallelic selection model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	selection(p0=0.01, w=c(1.0,0.9,0.8), time=100, show="p", pause=0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selection_+3A_p0">p0</code></td>
<td>
<p>Starting frequency for the <em>A</em> allele.</p>
</td></tr>
<tr><td><code id="selection_+3A_w">w</code></td>
<td>
<p>Fitnesses for the three genotypes in the following order: <em>AA</em>, <em>Aa</em>, <em>aa</em>.</p>
</td></tr>
<tr><td><code id="selection_+3A_time">time</code></td>
<td>
<p>Number of generations to run the analysis.</p>
</td></tr>
<tr><td><code id="selection_+3A_show">show</code></td>
<td>
<p>Various options for plotting. <code>"p"</code> shows the frequency of <em>A</em> through time; <code>"surface"</code> plots the mean fitness as a function of <em>p</em>; <code>"deltap"</code> shows the change in <em>p</em> as a function of <em>p</em>; <code>"cobweb"</code> creates a cobweb plot showing <em>p</em>(<em>t</em>) by <em>p</em>(<em>t</em>+1). The default is <code>show="p"</code>.</p>
</td></tr>
<tr><td><code id="selection_+3A_pause">pause</code></td>
<td>
<p>Pause between generations. <code>pause=0.01</code> (for instance) might smooth animation.</p>
</td></tr>
<tr><td><code id="selection_+3A_...">...</code></td>
<td>
<p>Optional arguments, including: <code>add</code>, a logical value indicating whether or not to add to the current plot (applies only to <code>show="p"</code>); <code>color</code>, change the color of the plotted line (works nicely with <code>add</code>, for obvious reasons); and <code>equil</code>, a logical value indicating whether or not to show the equilibrium value of <em>p</em> using vertical (or horizontal) lines on the graph (defaults to <code>FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates one of several possible plots, depending on the value of <code>show</code>.
</p>
<p>The cobweb plot shows <em>p</em>(<em>t+1</em>) as a function of <em>p</em>(<em>t</em>), with stairsteps giving the changes across generations given the initial value of <em>p</em> (<code>p0</code>) and total time (<code>time</code>) that are specified by the user.
</p>
<p>The function invisibly returns an object of class <code>"selection"</code> which can be printed or re-plotted using associated <code>print</code> and <code>plot</code> methods. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+drift.selection">drift.selection</a></code>, <code><a href="#topic+freqdep">freqdep</a></code>, <code><a href="#topic+msd">msd</a></code>, <code><a href="#topic+mutation.selection">mutation.selection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	selection(w=c(1.0,0.8,0.8),time=500)
	selection(w=c(1.0,1.0,0.0),show="surface")
	object&lt;-selection(w=c(0.8,1.1,0.7))
	print(object)
	plot(object,show="cobweb")
</code></pre>

<hr>
<h2 id='sexratio'>Hypothetical analysis of frequency dependent selection on a sex determining genetic locus</h2><span id='topic+sexratio'></span>

<h3>Description</h3>

<p>This function performs numerical analysis of a frequency dependent selection model of a hypothetical diploid sexually reproducing population in which sex is determined by the genotype at a biallelic genetic locus. Genotype <em>AA</em> are male, genotype <em>aa</em> are female, and genotype <em>Aa</em> might be male or female with probabilities that can be specified by the user. (Users may find, for instance, that setting <code>sex.Aa=c(1,0)</code> will result in evolution towards an <em>XY</em> sex determination system; whereas <code>sex.Aa=c(0,1)</code> will evolve towards a <em>ZW</em> system.)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>	sexratio(p0=0.01, time=40, show="p", pause=0, sex.Aa=c(0.5,0.5))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sexratio_+3A_p0">p0</code></td>
<td>
<p>Starting frequency for the <em>A</em> allele. Individuals with <em>AA</em> genotypes are male, while individuals with <em>Aa</em> genotypes are male or female with probability given by <code>sex.Aa</code>.</p>
</td></tr>
<tr><td><code id="sexratio_+3A_time">time</code></td>
<td>
<p>Number of generations to run the analysis.</p>
</td></tr>
<tr><td><code id="sexratio_+3A_show">show</code></td>
<td>
<p>Two different options for plotting. <code>"p"</code> shows the frequency of <em>A</em> through time; <code>"fitness"</code> plots the mean fitness through time; <code>"sex-ratio"</code> plots the relative frequency of each sex; and <code>"genotypes"</code> plots the frequencies of the three genotypes in the population. The default is <code>show="p"</code>.</p>
</td></tr>
<tr><td><code id="sexratio_+3A_pause">pause</code></td>
<td>
<p>Pause between generations. <code>pause=0.01</code> (for instance) might smooth animation.</p>
</td></tr>
<tr><td><code id="sexratio_+3A_sex.aa">sex.Aa</code></td>
<td>
<p>Probability that individuals with genotype <em>Aa</em> are male or female, respectively.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function creates one of four possible plots, depending on the value of <code>show</code>. Numerical analysis of this model shows how frequency dependent selection should favor alleles that tend to produce the rarer sex in the population.
</p>
<p>The function invisibly returns an object of class <code>"sexratio"</code> that can be printed or re-plotted by the user. (See examples.)
</p>


<h3>Author(s)</h3>

<p>Liam Revell <a href="mailto:liam.revell@umb.edu">liam.revell@umb.edu</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+freqdep">freqdep</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	sexratio()
	sexratio(p0=0.001,show="sex-ratio")
	sexratio(p0=0.001,show="fitness")
	object&lt;-sexratio(p0=0.001,sex.Aa=c(0.9,0.1),
		time=20)
	print(object)
	par(mfrow=c(2,1))
	plot(object,lwd=4,type="s",show="sex-ratio")
	plot(object,lwd=4,type="s",show="genotypes")
	par(mfrow=c(1,1))
	
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
