<!DOCTYPE html><html><head><title>Help for package lfc</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lfc}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#CenterMedian'><p>Subtract the median of the given vector (for normalizing log2 fold changes).</p></a></li>
<li><a href='#dlfc'><p>The log2 fold change distribution</p></a></li>
<li><a href='#EmpiricalBayesPrior'><p>Computes the prior parameters (i.e. pseudocounts incremented by 1) for</p>
the log2 fold change distribution</a></li>
<li><a href='#ltop'><p>Inverse logit transformation to obtain proportion representation from the</p>
log fold change representation.</a></li>
<li><a href='#NormLFC'><p>Standard LFC effect size estimator</p></a></li>
<li><a href='#PsiLFC'><p>Psi LFC effect size estimator</p></a></li>
<li><a href='#PsiLFC.se'><p>Psi LFC effect size estimator</p></a></li>
<li><a href='#ptol'><p>Logit transformation to obtain the log fold change representation from the</p>
proportion representation.</a></li>
<li><a href='#results'><p>Psi LFC effect size estimator for DESeq2</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Log Fold Change Distribution Tools for Working with Ratios of
Counts</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Florian Erhard</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Florian Erhard &lt;florian.erhard@uni-wuerzburg.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Ratios of count data such as obtained from RNA-seq are modelled
    using Bayesian statistics to derive posteriors for effects sizes. This
    approach is described in Erhard &amp; Zimmer (2015) &lt;<a href="https://doi.org/10.1093%2Fnar%2Fgkv696">doi:10.1093/nar/gkv696</a>&gt; 
    and Erhard (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbty471">doi:10.1093/bioinformatics/bty471</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/erhard-lab/lfc">https://github.com/erhard-lab/lfc</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/erhard-lab/lfc/issues">https://github.com/erhard-lab/lfc/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, DESeq2, SummarizedExperiment</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bayesian, Transcriptomics, DifferentialExpression</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-19 17:48:08 UTC; erhard</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-19 18:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='CenterMedian'>Subtract the median of the given vector (for normalizing log2 fold changes).</h2><span id='topic+CenterMedian'></span>

<h3>Description</h3>

<p>Subtract the median of the given vector (for normalizing log2 fold changes).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CenterMedian(l)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CenterMedian_+3A_l">l</code></td>
<td>
<p>Vector of effect sizes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of length 2 containing the two parameters
</p>


<h3>See Also</h3>

<p>PsiLFC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  CenterMedian(rnorm(1000,200))
</code></pre>

<hr>
<h2 id='dlfc'>The log2 fold change distribution</h2><span id='topic+dlfc'></span><span id='topic+plfc'></span><span id='topic+qlfc'></span><span id='topic+rlfc'></span>

<h3>Description</h3>

<p>Density, distribution function, quantile function and random
generation for the log2 fold change  distribution with parameters
‘a’ and ‘b’ (corresponding to (pseudo-)counts incremented by 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dlfc(x, a, b, log = FALSE)

plfc(q, a, b, lower.tail = TRUE, log.p = FALSE)

qlfc(p, a, b, lower.tail = TRUE, log.p = FALSE)

rlfc(n, a, b)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dlfc_+3A_x">x</code>, <code id="dlfc_+3A_q">q</code></td>
<td>
<p>vector of quantiles</p>
</td></tr>
<tr><td><code id="dlfc_+3A_a">a</code></td>
<td>
<p>non-negative parameter</p>
</td></tr>
<tr><td><code id="dlfc_+3A_b">b</code></td>
<td>
<p>non-negative parameter</p>
</td></tr>
<tr><td><code id="dlfc_+3A_log">log</code>, <code id="dlfc_+3A_log.p">log.p</code></td>
<td>
<p>if TRUE, probabilities p are given as log(p)</p>
</td></tr>
<tr><td><code id="dlfc_+3A_lower.tail">lower.tail</code></td>
<td>
<p>if TRUE (default), probabilities are P[X &lt;= x],
otherwise, P[X &gt; x].</p>
</td></tr>
<tr><td><code id="dlfc_+3A_p">p</code></td>
<td>
<p>vector of probabilities</p>
</td></tr>
<tr><td><code id="dlfc_+3A_n">n</code></td>
<td>
<p>number of observations</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The density
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>dlfc()</code>: Density function
</p>
</li>
<li> <p><code>plfc()</code>: Distribution function
</p>
</li>
<li> <p><code>qlfc()</code>: Quantile function
</p>
</li>
<li> <p><code>rlfc()</code>: random generation
</p>
</li></ul>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- seq (-5,5,by=0.01)
plot(x,dlfc(x,1,1))
</code></pre>

<hr>
<h2 id='EmpiricalBayesPrior'>Computes the prior parameters (i.e. pseudocounts incremented by 1) for
the log2 fold change distribution</h2><span id='topic+EmpiricalBayesPrior'></span>

<h3>Description</h3>

<p>Computes the prior parameters (i.e. pseudocounts incremented by 1) for
the log2 fold change distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EmpiricalBayesPrior(A, B, min.sd = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EmpiricalBayesPrior_+3A_a">A</code></td>
<td>
<p>Vector of counts from condition A</p>
</td></tr>
<tr><td><code id="EmpiricalBayesPrior_+3A_b">B</code></td>
<td>
<p>Vector of counts from condition B</p>
</td></tr>
<tr><td><code id="EmpiricalBayesPrior_+3A_min.sd">min.sd</code></td>
<td>
<p>minimal standard deviation of the prior</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of length 2 containing the two parameters
</p>


<h3>See Also</h3>

<p>PsiLFC
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  EmpiricalBayesPrior(rnorm(1000,200),rnorm(1000,100))
</code></pre>

<hr>
<h2 id='ltop'>Inverse logit transformation to obtain proportion representation from the
log fold change representation.</h2><span id='topic+ltop'></span>

<h3>Description</h3>

<p>Inverse logit transformation to obtain proportion representation from the
log fold change representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ltop(l)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ltop_+3A_l">l</code></td>
<td>
<p>Effect size in log2 fold change representation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The proportion representation of the effect size
</p>


<h3>See Also</h3>

<p>ptol
</p>
<p>Other Effect size representations: 
<code><a href="#topic+ptol">ptol</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ptol(0)
ptol(1)
</code></pre>

<hr>
<h2 id='NormLFC'>Standard LFC effect size estimator</h2><span id='topic+NormLFC'></span>

<h3>Description</h3>

<p>Computes the standard, normalized log2 fold change with given pseudocounts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NormLFC(A, B, pseudo = c(1, 1), normalizeFun = CenterMedian)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NormLFC_+3A_a">A</code></td>
<td>
<p>Vector of counts from condition A</p>
</td></tr>
<tr><td><code id="NormLFC_+3A_b">B</code></td>
<td>
<p>Vector of counts from condition B</p>
</td></tr>
<tr><td><code id="NormLFC_+3A_pseudo">pseudo</code></td>
<td>
<p>Vector of length 2 of the pseudo counts</p>
</td></tr>
<tr><td><code id="NormLFC_+3A_normalizefun">normalizeFun</code></td>
<td>
<p>Function to normalize the obtained effect sizes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The vector containing the estimates
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  NormLFC(rnorm(1000,200),rnorm(1000,100))
</code></pre>

<hr>
<h2 id='PsiLFC'>Psi LFC effect size estimator</h2><span id='topic+PsiLFC'></span>

<h3>Description</h3>

<p>Computes the optimal effect size estimate and credible intervals if needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PsiLFC(
  A,
  B,
  prior = EmpiricalBayesPrior(A, B),
  normalizeFun = CenterMedian,
  cre = FALSE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PsiLFC_+3A_a">A</code></td>
<td>
<p>Vector of counts from condition A</p>
</td></tr>
<tr><td><code id="PsiLFC_+3A_b">B</code></td>
<td>
<p>Vector of counts from condition B</p>
</td></tr>
<tr><td><code id="PsiLFC_+3A_prior">prior</code></td>
<td>
<p>Vector of length 2 of the prior parameters</p>
</td></tr>
<tr><td><code id="PsiLFC_+3A_normalizefun">normalizeFun</code></td>
<td>
<p>Function to normalize the obtained effect sizes</p>
</td></tr>
<tr><td><code id="PsiLFC_+3A_cre">cre</code></td>
<td>
<p>Compute credible intervals as well? (can also be a vector of quantiles)</p>
</td></tr>
<tr><td><code id="PsiLFC_+3A_verbose">verbose</code></td>
<td>
<p>verbose status updates?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a vector containing the estimates, or a data frame containing
the credible interval as well
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  PsiLFC(rnorm(1000,200),rnorm(1000,100))
</code></pre>

<hr>
<h2 id='PsiLFC.se'>Psi LFC effect size estimator</h2><span id='topic+PsiLFC.se'></span>

<h3>Description</h3>

<p>Computes the optimal effect size estimate and credible intervals if needed
for a Bioconductor SummarizedExperiment object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PsiLFC.se(se, contrast, cre = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PsiLFC.se_+3A_se">se</code></td>
<td>
<p>SummarizedExperiment object</p>
</td></tr>
<tr><td><code id="PsiLFC.se_+3A_contrast">contrast</code></td>
<td>
<p>Vector of length 3 (&lt;name&gt;,&lt;A&gt;,&lt;B&gt;)</p>
</td></tr>
<tr><td><code id="PsiLFC.se_+3A_cre">cre</code></td>
<td>
<p>Compute credible intervals as well? (can also be a vector of quantiles)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a vector containing the estimates, or a data frame containing
the credible interval as well
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
   data(airway, package="airway")
   head(PsiLFC.se(airway,contrast=c("dex","untrt","trt")))

## End(Not run)
</code></pre>

<hr>
<h2 id='ptol'>Logit transformation to obtain the log fold change representation from the
proportion representation.</h2><span id='topic+ptol'></span>

<h3>Description</h3>

<p>Logit transformation to obtain the log fold change representation from the
proportion representation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ptol(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ptol_+3A_p">p</code></td>
<td>
<p>Effect size in proportion representation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log2 fold change representation of the effect size
</p>


<h3>See Also</h3>

<p>ltop
</p>
<p>Other Effect size representations: 
<code><a href="#topic+ltop">ltop</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ptol(0.5)
ptol(2/3)
</code></pre>

<hr>
<h2 id='results'>Psi LFC effect size estimator for DESeq2</h2><span id='topic+results'></span>

<h3>Description</h3>

<p>Drop-in replacement for DESeq2's results function for simple settings involving
a single variable. Appends the PsiLFC estimate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>results(object, contrast, cre = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="results_+3A_object">object</code></td>
<td>
<p>the DESeq2DataSet object</p>
</td></tr>
<tr><td><code id="results_+3A_contrast">contrast</code></td>
<td>
<p>Vector of length 3, specifying the variable and the two levels
to compute effect sizes for (&lt;name&gt;,&lt;A&gt;,&lt;B&gt;)</p>
</td></tr>
<tr><td><code id="results_+3A_cre">cre</code></td>
<td>
<p>Compute credible intervals as well? (can also be a vector of quantiles)</p>
</td></tr>
<tr><td><code id="results_+3A_...">...</code></td>
<td>
<p>Handed over to DESeq2's results function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a vector containing the estimates, or a data frame containing
the credible interval as well
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
