<!DOCTYPE html><html><head><title>Help for package LFDREmpiricalBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LFDREmpiricalBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LFDREmpiricalBayes-package'>
<p>Estimating Local False Discovery Rates Using Empirical Bayes Methods</p></a></li>
<li><a href='#caution.parameter.actions'><p>Based on a Decision-Theoretic Approach, Performs a Multiple Hypothesis Testing Problem under a Zero-One Loss Function</p></a></li>
<li><a href='#ME.log'><p>Provides Reliable LFDR Estimates by Selecting an Appropriate Reference Class</p></a></li>
<li><a href='#PRGM.action'><p>Based on the Robust Bayes Approach, Performs a Multiple Hyothesis Testing Problem under an Squared Error Loss Function</p></a></li>
<li><a href='#SEL.caution.parameter'><p>Based on a Decision-Theoretic Approach, Performs a Multiple Hyothesis Testing Problem under an Squared Error Loss Function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimating Local False Discovery Rates Using Empirical Bayes
Methods</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-09-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Ali Karimnezhad, Johnary Kim, Anna Akpawu, Justin Chitpin and David R Bickel</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ali Karimnezhad &lt;ali_karimnezhad@yahoo.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>New empirical Bayes methods aiming at analyzing the association of single nucleotide polymorphisms (SNPs) to some particular disease are implemented in this package. The package uses local false discovery rate (LFDR) estimates of SNPs within a sample population defined as a  "reference class" and discovers if SNPs are associated with the corresponding disease. Although SNPs are used throughout this document, other biological data such as protein data and other gene data can be used. Karimnezhad, Ali and Bickel, D. R. (2016) <a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a>.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.14.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>matrixStats, stats, R6</td>
</tr>
<tr>
<td>Suggests:</td>
<td>LFDR.MLE, testthat</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bayesian, MathematicalBiology, MultipleComparison</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://davidbickel.com">https://davidbickel.com</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-09-27 01:34:32 UTC; a.karimnezhad</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-09-27 09:08:46 UTC</td>
</tr>
</table>
<hr>
<h2 id='LFDREmpiricalBayes-package'>
Estimating Local False Discovery Rates Using Empirical Bayes Methods
</h2><span id='topic+LFDREmpiricalBayes-package'></span><span id='topic+LFDREmpiricalBayes'></span>

<h3>Description</h3>

<p>New empirical Bayes methods aiming at analyzing the association of single nucleotide polymorphisms (SNPs) to some particular disease are implemented in this package. The package uses local false discovery rate (LFDR) estimates of SNPs within a sample population defined as a  &quot;reference class&quot; and discovers if SNPs are associated with the corresponding disease. Although SNPs are used throughout this document, other biological data such as protein data and other gene data can be used. Karimnezhad, Ali and Bickel, D. R. (2016) &lt;http://hdl.handle.net/10393/34889&gt;.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> LFDREmpiricalBayes</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-09-26</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R(&gt;= 2.14.2) </td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> matrixStats, stats</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> LFDR.MLE </td>
</tr>
<tr>
 <td style="text-align: left;">
URL: </td><td style="text-align: left;"> https://davidbickel.com</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Ali Karimnezhad, Johnary Kim, Anna Akpawu, Justin Chitpin and David R Bickel
</p>
<p>Maintainer: Ali Karimnezhad &lt;ali_karimnezhad@yahoo.com&gt;
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Bickel, D. R. (2016). Incorporating prior knowledge about
genetic variants into the analysis of genetic association data: An empirical
Bayes approach. Working paper. Retrieved from
<a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a></p>


<h3>See Also</h3>

<p>For more information on how to interpret the outputs, look at the
supplementary file in the vignette directory,
&quot;Using the LFDREmpiricalBayes Package.&quot;
</p>

<hr>
<h2 id='caution.parameter.actions'>Based on a Decision-Theoretic Approach, Performs a Multiple Hypothesis Testing Problem under a Zero-One Loss Function</h2><span id='topic+caution.parameter.actions'></span>

<h3>Description</h3>

<p>Assuming a zero-onr loss function, it provides three caution-type actions using estimated LFDRs computed based on both separate and combined reference classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>caution.parameter.actions(x1,x2,l1=4,l2=1) # default values l1=4 and l2=1
# to obtain a threshold of 20%.
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="caution.parameter.actions_+3A_x1">x1</code></td>
<td>
<p>A vector of LFDRs in the combined reference class.</p>
</td></tr>
<tr><td><code id="caution.parameter.actions_+3A_x2">x2</code></td>
<td>
<p>A vector of LFDRs in the separate reference class.</p>
</td></tr>
<tr><td><code id="caution.parameter.actions_+3A_l1">l1</code></td>
<td>
<p>Loss value (Type-I error) for deriving the threshold of the Bayes
action.</p>
</td></tr>
<tr><td><code id="caution.parameter.actions_+3A_l2">l2</code></td>
<td>
<p>Loss value (Type-II error) for deriving the threshold of the Bayes
action.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Accepts previously obtained LFDR estimates of SNPs falling inside the intersection of the separate and combined reference classes. The LFDR estimates of some biological feature (SNP or gene) within a sample population that we will refer to as &lsquo;reference class&rsquo;. If a reference class, containing LFDR estimates
is a subset of the other, it is referred to as &lsquo;separate class&rsquo;.
The entire set of LFDR estimates is called a &lsquo;combined&rsquo; reference class. Then,
a multiple hypothesis problem is conducted using three caution-type estimators.
The threshold set for rejecting the null hypothesis is derived from
pre-specified <code>l1</code> and <code>l2</code> values. Since having a type-I error is
worse than a type-II error, <code>l1</code> is recommende to be greater than
<code>l2</code>.
</p>
<p>In generating the output, there are two potential outputs for each index of the
three caution-type actions. Check the <b>Value</b> section for the
corresponding caution-type actions.
</p>
<p>For each index of the output, one of two potential outputs based on Bayes action
are shown:
</p>

<table>
<tr>
 <td style="text-align: right;">
 <code>0</code> </td><td style="text-align: left;"> Do not reject the null hypothesis </td>
</tr>
<tr>
 <td style="text-align: right;">
 <code>1</code> </td><td style="text-align: left;"> Reject the null hypothesis</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>For each corresponding index in the output, the decision on whether to reject or
not reject the null hypothesis for biological feature can be based on
<code>CGM1</code>, <code>CGM0</code>, and <code>CGM0.5</code> decisions. Check <b>See Also</b> for
more details on how to better interpret the outputs.
</p>


<h3>Value</h3>

<p>Outputs three vectors of equal size as seen below:
</p>
<table>
<tr><td><code>CGM1</code></td>
<td>
<p>Decision values for the Conditional Gamma Minimax (CGMinimax).</p>
</td></tr>
<tr><td><code>CGM0</code></td>
<td>
<p>Decision values for the Conditional Gamma Minimin (CGMinimin).</p>
</td></tr>
<tr><td><code>CGM0.5</code></td>
<td>
<p>Decision values for the CG0.5 caution case (a balance between
CGMinimax and CGMinimin.</p>
</td></tr>
</table>
<p>Note that the length of the input vectors <code>x1</code> and <code>x2</code> determines the
number of null hypothesis values seen in the output.
</p>


<h3>Note</h3>

<p>A limitation to the code is that both reference classes: <code>x1</code> and <code>x2</code>
must be of the same vector length.
</p>


<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br />
Documentation: Justin Chitpin, Anna Akpawu and Johnary Kim. <br />
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Bickel, D. R. (2016). Incorporating prior knowledge about
genetic variants into the analysis of genetic association data: An empirical
Bayes approach. Working paper. Retrieved from
<a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a>
</p>


<h3>See Also</h3>

<p>For more information on how to interpret the outputs, look at the vignette,
&ldquo;Using <code>LFDREmpiricalBayes</code>&rdquo;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#LFDR reference class values generated

#First reference class (separate class)
LFDR.Separate &lt;- c(.14,.8,.251,.30)
#Second reference class (combined class)
LFDR.Combined &lt;- c(.21,.61,.0888,.10)

# Default threshold at (20%).
output &lt;- caution.parameter.actions(x1=LFDR.Separate, x2=LFDR.Combined)

# Three caution cases
output

</code></pre>

<hr>
<h2 id='ME.log'>Provides Reliable LFDR Estimates by Selecting an Appropriate Reference Class</h2><span id='topic+ME.log'></span>

<h3>Description</h3>

<p>Selects an appropriate reference class given two reference classes. Considers two vecotr of LFDR estimates computed based on the two alternative reference classes and provides a vector of more reliable LFDR estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ME.log(stat,lfdr.C,p0.C,ncp.C,p0.S,ncp.S,a=3,lower.p0=0,upper.p0=1,
lower.ncp=0.1,upper.ncp=50,length.p0=200,length.ncp=200)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ME.log_+3A_stat">stat</code></td>
<td>
<p>A vector of test statistics for SNPs falling inside the
intersection of the separate and combined reference classes.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_lfdr.c">lfdr.C</code></td>
<td>
<p>A data frame of local false discovery rates of features falling
inside the intersection of the separate and combined reference classes,
computed based on all features belonging to the combined reference class.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_p0.c">p0.C</code></td>
<td>
<p>An estimate of the proportion of the non-associated features
applied to the combined reference class. </p>
</td></tr>
<tr><td><code id="ME.log_+3A_ncp.c">ncp.C</code></td>
<td>
<p>A non-centrality parameter applied to the combined reference
class.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_p0.s">p0.S</code></td>
<td>
<p>An estimate of the proportion of the non-associated features
applied to the separate reference class. </p>
</td></tr>
<tr><td><code id="ME.log_+3A_ncp.s">ncp.S</code></td>
<td>
<p>A non-centrality parameter applied to the separate reference
class.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_a">a</code></td>
<td>
<p>Parameter used to define the grade of evidence that alternative
reference class should be favoured instead of the separate reference class.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_lower.p0">lower.p0</code></td>
<td>
<p>The lower bound for the proportion of unassociated features.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_upper.p0">upper.p0</code></td>
<td>
<p>The upper bound for the proportion of unassociated features.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_lower.ncp">lower.ncp</code></td>
<td>
<p>The lower bound for the non-centrality parameter.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_upper.ncp">upper.ncp</code></td>
<td>
<p>The lower bound for the non-centrality parameter.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_length.p0">length.p0</code></td>
<td>
<p>Desired length of a sequence vector containing the proportion
of non-associated features. The sequences starts from <code>lower.p0</code> and ends
at <code>upper.p0</code>.</p>
</td></tr>
<tr><td><code id="ME.log_+3A_length.ncp">length.ncp</code></td>
<td>
<p>Desired length of a sequence vector containing
non-centrality parameters. The sequences starts from <code>lower.ncp</code> and ends
up at <code>upper.ncp</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The terms &lsquo;separate&rsquo; and &lsquo;combined&rsquo; reference classes are used when one
sample population (reference class) is a subset of the other. Detailed
explanations can be found in the vignette
&quot;Using the LFDREmpiricalBayes Package&quot;.
</p>


<h3>Value</h3>

<p>Returns the following values:
</p>
<table>
<tr><td><code>p0.hat</code></td>
<td>
<p>estimate of the proportion of non-associated SNPs</p>
</td></tr>
<tr><td><code>ncp.hat</code></td>
<td>
<p>estimate of the non-centrality parameter</p>
</td></tr>
<tr><td><code>LFDR.hat</code></td>
<td>
<p>A vector of LFDR estimates for features falling inside the
intersection of the separate and combined reference classes, obtained by the
Maximum Entropy method.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The vector of test statistics: <code>stat</code>, need to be positive values in order
for the function <code>ME.log</code> to work.
</p>


<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br />
Documentation: Johnary Kim and Anna Akpawu. <br />
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Bickel, D. R. (2016). Incorporating prior knowledge about
genetic variants into the analysis of genetic association data: An empirical
Bayes approach. Working paper. Retrieved from
<a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#import the function ``lfdr.mle'' from package``LFDR.MLE''
library(LFDR.MLE)

#Consider a separate reference class and a combined reference class below:

n.SNPs.S&lt;-3  # number of SNPs in the separate reference class
n.SNPs.Sc&lt;-2 # number of SNPs in the complement of the separate reference class.

#Create a series of test statistics for SNPs in the separate reference class.
stat.Small&lt;-rchisq(n.SNPs.S,df=1,ncp=0)
ncp.Sc&lt;-10

#Create a series of test statistics for SNPs in the combined reference class.
stat.Big&lt;-c(stat.Small,rchisq(n.SNPs.Sc,df=1,ncp=ncp.Sc))

#Using lfdr.mle, a series of arguments are used.
dFUN=dchisq; lower.ncp = .1; upper.ncp = 50;
lower.p0 = 0; upper.p0 = 1;

#Maximum Likelihood estimates for the LFDRs of SNPs in the created
# separate reference class.

#Separate reference class.
estimates.S&lt;-lfdr.mle(x=stat.Small,dFUN=dchisq,df=1,lower.ncp = lower.ncp,
upper.ncp = upper.ncp)
LFDR.Small&lt;-estimates.S$LFDR
p0.Small&lt;-estimates.S$p0.hat
ncp.Small&lt;-estimates.S$ncp.hat

# Maximum Likelihood estimates for the LFDRs of SNPs in the created combined
# reference class.
estimates.C&lt;-lfdr.mle(x=stat.Big,dFUN=dchisq,df=1,lower.ncp = lower.ncp,
upper.ncp = upper.ncp)
LFDR.Big&lt;-estimates.C$LFDR
p0.Big&lt;-estimates.C$p0.hat
ncp.Big&lt;-estimates.C$ncp.hat


#The first three values of the combined reference class correspond to the
#separate reference class in this example
LFDR.SBig&lt;-LFDR.Big[1:3]

LFDR.ME&lt;-ME.log(stat=stat.Small,lfdr.C=LFDR.SBig,p0.C=p0.Big,ncp.C=ncp.Big,
p0.S=p0.Small,ncp.S=ncp.Small)

LFDR.ME
</code></pre>

<hr>
<h2 id='PRGM.action'>Based on the Robust Bayes Approach, Performs a Multiple Hyothesis Testing Problem under an Squared Error Loss Function</h2><span id='topic+PRGM.action'></span>

<h3>Description</h3>

<p>Assuming a squared error loss function, it provides Robust Bayes estimates of the LFDR estimates giving credit to both separate and combined reference classes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PRGM.action(x1,x2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PRGM.action_+3A_x1">x1</code></td>
<td>
<p>Input numeric vector of LFDR estimates of the separate reference
class.</p>
</td></tr>
<tr><td><code id="PRGM.action_+3A_x2">x2</code></td>
<td>
<p>Input numeric vector of LFDR estimated of the combined reference
class.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The output is a vector of the LFDR estimates based on the two reference classes.
</p>


<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br />
Documentation: Johnary Kim and Anna Akpawu. <br />
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Bickel, D. R. (2016). Incorporating prior knowledge about
genetic variants into the analysis of genetic association data: An empirical
Bayes approach. Working paper. Retrieved from
<a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#LFDR reference class values generated

#First reference class
LFDR.Separate &lt;- c(0.14, 0.8, 0.16, 0.30)
#Second reference class
LFDR.Combined &lt;- c(0.21, 0.61, 0.12, 0.10)

output &lt;- PRGM.action(LFDR.Separate, LFDR.Combined)

# Vector of the LFDR estimates
output

</code></pre>

<hr>
<h2 id='SEL.caution.parameter'>Based on a Decision-Theoretic Approach, Performs a Multiple Hyothesis Testing Problem under an Squared Error Loss Function</h2><span id='topic+SEL.caution.parameter'></span>

<h3>Description</h3>

<p>Assuming a squared error loss function, it provides three caution-type actions using estimated LFDRs computed based on both separate and combined reference classes. </p>


<h3>Usage</h3>

<pre><code class='language-R'>SEL.caution.parameter(x1,x2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SEL.caution.parameter_+3A_x1">x1</code></td>
<td>
<p>Input numeric vector of LFDR estimates in the separate reference
class.</p>
</td></tr>
<tr><td><code id="SEL.caution.parameter_+3A_x2">x2</code></td>
<td>
<p>Input numeric vector of LFDR estimates in the combined reference
class.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Much like <code>caution.parameter.actions</code>, this function returns three vectors
of equal size as seen below:
</p>
<table>
<tr><td><code>CGM1</code></td>
<td>
<p>Squared error loss value for the Conditional Gamma Minimax
(CGMinimax).</p>
</td></tr>
<tr><td><code>CGM0</code></td>
<td>
<p>Squared error loss value for the Conditional Gamma Minimin
(CGMinimin).</p>
</td></tr>
<tr><td><code>CGM0.5</code></td>
<td>
<p>Squared error loss value for the Action/Decision estimate
(a balance between CGMinimax and CGMinimin.</p>
</td></tr>
</table>
<p>For each index of the vectors, the squared error loss values are given.
</p>


<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br />
Documentation: Johnary Kim and Anna Akpawu. <br />
</p>


<h3>References</h3>

<p>Karimnezhad, A. and Bickel, D. R. (2016). Incorporating prior knowledge about
genetic variants into the analysis of genetic association data: An empirical
Bayes approach. Working paper. Retrieved from
<a href="http://hdl.handle.net/10393/34889">http://hdl.handle.net/10393/34889</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#Similar to caution.parameter actions we have the following classes

#First reference class
LFDR.Separate &lt;- c(0.14, 0.8, 0.16, 0.30)
#Second reference class
LFDR.Combined &lt;- c(0.21, 0.61, 0.12, 0.10)

output &lt;- SEL.caution.parameter(LFDR.Separate, LFDR.Combined)

# Three caution cases with SEL values.
output
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
