<!DOCTYPE html><html><head><title>Help for package LFDR.MME</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LFDR.MME}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LFDR.MM'><p>Performs a Multiple Hypothesis Testing Using the Method of Moments</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimating Local False Discovery Rates Using the Method of
Moments</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-11-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Ali Karimnezhad</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ali Karimnezhad &lt;ali.karimnezhad@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimation of the local false discovery rate using the method of moments.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.14.2)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bayesian, MathematicalBiology, MultipleComparison</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-11-18 02:52:23 UTC; alikarimnezhad</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-20 09:10:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='LFDR.MM'>Performs a Multiple Hypothesis Testing Using the Method of Moments</h2><span id='topic+LFDR.MM'></span>

<h3>Description</h3>

<p>Based on a given vector of chi-square test statistics, provides estimates of local false discoveries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LFDR.MM(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LFDR.MM_+3A_x">x</code></td>
<td>
<p>A vector of chi-square test statistics with one degree of freedom.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code class="reqn">N</code> given features (genes, proteins, SNPs, etc.), the function tests the null hypothesis <code class="reqn">H_{0i}</code>, <code class="reqn">i=1,\ldots,N</code>, indicating that there is no association between feature <code class="reqn">i</code> and a specific disease, versus its alternative hypothesis <code class="reqn">H_{1i}</code>. For each unassociated feature <code class="reqn">i</code>, it is suppoed that the corresponding test stiatistic <code class="reqn">x_i</code> follows a central chi-square distribution with one degree of freedom.  For each associated feature <code class="reqn">i</code>, it is assumed that the corresponding test stiatistic <code class="reqn">x_i</code> follows a non-central chi-square distribution with one degree of freedom and non-centrality parameter <code class="reqn">\lambda</code>. In this packag, association is measured by estimating the local false discovery rate (LFDR), the posterior probability that the null hypothesis <code class="reqn">H_{0i}</code> given the test statistic <code class="reqn">x_i</code> is true.
This package returns three components as mentioned in the <b>Value</b> section.</p>


<h3>Value</h3>

<p>Outputs three elements as seen below:
</p>
<table>
<tr><td><code>pi0.hat</code></td>
<td>
<p>estimate of proportion of unassocaited features <code class="reqn">\pi_0</code>.</p>
</td></tr>
<tr><td><code>ncp.hat</code></td>
<td>
<p>estimate of the non-centrality parameter <code class="reqn">\lambda</code> of the chi-square model for associated features.</p>
</td></tr>
<tr><td><code>lfdr.hat</code></td>
<td>
<p>estimates of local false discovery rates.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Code: Ali Karimnezhad. <br />
Documentation: Ali Karimnezhad. <br />
</p>


<h3>References</h3>

<p>Karimnezhad, A. (2020). A Simple Yet Efficient Parametric Method of Local False Discovery Rate Estimation Designed for Genome-Wide Association Data Analysis. Retrieved from
<a href="https://arxiv.org/abs/1909.13307">https://arxiv.org/abs/1909.13307</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># vector of test statistics for assocaited features
stat.assoc&lt;- rchisq(n=1000,df=1, ncp = 3)

# vector of test statistics for unassocaited features
stat.unassoc&lt;- rchisq(n=9000,df=1, ncp = 0)

# vector of test statistics
stat&lt;- c(stat.assoc,stat.unassoc)

output &lt;- LFDR.MM(x=stat)

# Estimated pi0
output$p0.hat

# Estimated non-centrality parameter
output$ncp.hat

# Estimated LFDRs
output$lfdr.hat
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
