<!DOCTYPE html><html><head><title>Help for package lfmm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lfmm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#effect_size'><p>Direct effect sizes estimated from latent factor models</p></a></li>
<li><a href='#example.data'><p>Genetic and phenotypic data for Arabidopsis thaliana</p></a></li>
<li><a href='#glm_test'><p>GLM tests with latent factor mixed models</p></a></li>
<li><a href='#lfmm'><p>R package : Fast and Accurate statistical methods for adjusting confounding factors</p>
in association studies.</a></li>
<li><a href='#lfmm_lasso'><p>LFMM least-squares estimates with lasso penalty (Sparse LFMM)</p></a></li>
<li><a href='#lfmm_ridge'><p>LFMM least-squares estimates with ridge penalty</p></a></li>
<li><a href='#lfmm_sampler'><p>LFMM generative data sampler</p></a></li>
<li><a href='#lfmm_test'><p>Statistical tests with latent factor mixed models (linear models)</p></a></li>
<li><a href='#predict_lfmm'><p>Predict polygenic scores from latent factor models</p></a></li>
<li><a href='#skin.exposure'><p>Simulated (and real) methylation levels for sun exposed patient patients</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Latent Factor Mixed Models</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-06-29</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast and accurate inference of 
             gene-environment associations (GEA) in genome-wide studies 
             (Caye et al., 2019, &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmsz008">doi:10.1093/molbev/msz008</a>&gt;). 
             We developed a least-squares estimation approach for confounder and effect sizes 
             estimation that provides a unique framework for several categories of genomic data, 
             not restricted to genotypes. 
             The speed of the new algorithm is several times faster than the existing GEA approaches, 
             then our previous version of the 'LFMM' program present in the 'LEA' package 
             (Frichot and Francois, 2015, &lt;<a href="https://doi.org/10.1111%2F2041-210X.12382">doi:10.1111/2041-210X.12382</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>Imports:</td>
<td>foreach, rmarkdown, knitr, MASS, RSpectra, stats, ggplot2,
readr, methods, purrr, Rcpp</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>RcppEigen, Rcpp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/bcm-uga/lfmm/issues">https://github.com/bcm-uga/lfmm/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-06-30 11:25:43 UTC; jumentib</td>
</tr>
<tr>
<td>Author:</td>
<td>Basile Jumentier [aut, cre],
  Kevin Caye [ctb],
  Olivier François [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Basile Jumentier &lt;basile.jumentier@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-06-30 13:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='effect_size'>Direct effect sizes estimated from latent factor models</h2><span id='topic+effect_size'></span>

<h3>Description</h3>

<p>This function returns 'direct' effect sizes for the regression of X (of dimension 1) on the matrix Y,
as usually computed in genome-wide association studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effect_size(Y, X, lfmm.object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="effect_size_+3A_y">Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns.
Each column is a response variable (numeric).</p>
</td></tr>
<tr><td><code id="effect_size_+3A_x">X</code></td>
<td>
<p>an explanatory variable with n rows and d = 1 column (numeric).</p>
</td></tr>
<tr><td><code id="effect_size_+3A_lfmm.object">lfmm.object</code></td>
<td>
<p>an object of class <code>lfmm</code> returned by the <a href="#topic+lfmm_lasso">lfmm_lasso</a>
or <a href="#topic+lfmm_ridge">lfmm_ridge</a> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>a vector of length p containing all effect sizes for the regression
of X on the matrix Y
</p>


<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(lfmm)

## Simulation of 1000 genotypes for 100 individuals (y)
u &lt;- matrix(rnorm(300, sd = 1), nrow = 100, ncol = 3) 
v &lt;- matrix(rnorm(3000, sd = 3), nrow = 3, ncol = 1000)
w &lt;- u %*% v
y &lt;- matrix(rbinom(100000, size = 2, 
                  prob = 1/(1 + exp(-0.3*(w
                  + rnorm(100000, sd = 2))))),
                  nrow = 100,
                  ncol = 1000)

#PCA of genotypes, 3 main axes of variation (K = 2) 
plot(prcomp(y))
  
## Simulation of 1000 phenotypes (x)
## Only the last 10 genotypes have significant effect sizes (b)
b &lt;- matrix(c(rep(0, 990), rep(6000, 10)))
x &lt;- y%*%b + rnorm(100, sd = 100)

## Compute effect sizes using lfmm_ridge
## Note that centering is important (scale = F).
mod.lfmm &lt;- lfmm_ridge(Y = y, 
                  X = x,
                  K = 2)
              
## Compute direct effect sizes using lfmm_ridge estimates
b.estimates &lt;- effect_size(y, x, mod.lfmm)

## plot the last 30 effect sizes (true values are 0 and 6000)
plot(b.estimates[971:1000])
abline(0, 0)
abline(6000, 0, col = 2)

## Prediction of phenotypes
candidates &lt;- 991:1000 #set of causal loci 
x.pred &lt;- scale(y[,candidates], scale = FALSE) %*% matrix(b.estimates[candidates])

## Check predictions
plot(x - mean(x), x.pred, 
     pch = 19, col = "grey", 
     xlab = "Observed phenotypes (centered)", 
     ylab = "Predicted from PRS")
     abline(0,1)
     abline(lm(x.pred ~ scale(x, scale = FALSE)), col = 2)
</code></pre>

<hr>
<h2 id='example.data'>Genetic and phenotypic data for Arabidopsis thaliana</h2><span id='topic+example.data'></span>

<h3>Description</h3>

<p>A dataset containing SNP frequency and simulated phenotypic data for 170 plant accessions.
The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(example.data)
</code></pre>


<h3>Format</h3>

<p>A list with 4 arguments: genotype, phenotype, causal.set, chrpos
</p>


<h3>Details</h3>


<ul>
<li><p> genotype: binary (0 or 1) SNP frequency for 170 individuals (26943 SNPs).
</p>
</li>
<li><p> phenotype: simulated phenotypic data for 170 individuals.
</p>
</li>
<li><p> causal.set: set of indices for causal SNPs.
</p>
</li>
<li><p> chrpos: genetic map including chromosome position of each SNP.
</p>
</li></ul>

<p>Reference: Atwell et al (2010).
Genome-wide association study of 107 phenotypes in Arabidopsis thaliana inbred lines.
Nature 465, 627–631.
</p>

<hr>
<h2 id='glm_test'>GLM tests with latent factor mixed models</h2><span id='topic+glm_test'></span>

<h3>Description</h3>

<p>This function returns significance values for the association between each column of the
response matrix, Y, and the explanatory variables, X, including correction for unobserved confounders
(latent factors). The test is based on an LFMM fitted with a ridge or lasso penalty and a generalized linear
model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glm_test(Y, X, lfmm.obj, calibrate = "gif", family = binomial(link = "logit"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glm_test_+3A_y">Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns.
Each column is a response variable (numeric).</p>
</td></tr>
<tr><td><code id="glm_test_+3A_x">X</code></td>
<td>
<p>an explanatory variable matrix with n rows and d columns.
Each column corresponds to an explanatory variable (numeric).</p>
</td></tr>
<tr><td><code id="glm_test_+3A_lfmm.obj">lfmm.obj</code></td>
<td>
<p>an object of class <code>lfmm</code> returned by the <a href="#topic+lfmm_lasso">lfmm_lasso</a>
or <a href="#topic+lfmm_ridge">lfmm_ridge</a> function</p>
</td></tr>
<tr><td><code id="glm_test_+3A_calibrate">calibrate</code></td>
<td>
<p>a character string, &quot;gif&quot;. If the &quot;gif&quot; option is set (default),
significance values are calibrated by using the genomic control method. Genomic control
uses a robust estimate of the variance of z-scores called &quot;genomic inflation factor&quot;.</p>
</td></tr>
<tr><td><code id="glm_test_+3A_family">family</code></td>
<td>
<p>a description of the error distribution and link function to be used
in the model. For glm this can be a character string naming a family function,
a family function or the result of a call to a family function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>a list with the following attributes:
</p>

<ul>
<li><p> B the effect size matrix with dimensions p x d.
</p>
</li>
<li><p> score a p x d matrix which contains z-scores for each explanatory variable (columns of X),
</p>
</li>
<li><p> pvalue a p x d matrix which contains p-values for each explanatory variable,
</p>
</li>
<li><p> calibrated.pvalue a p x d matrix which contains calibrated p-values for each explanatory variable,
</p>
</li>
<li><p> gif a numeric value for the genomic inflation factor.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>See Also</h3>

<p><a href="#topic+lfmm_test">lfmm_test</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(lfmm)


## An EWAS example with Y = methylation data 
## and X = "exposure" 
## Simulate the data

dat &lt;- lfmm_sampler(n = 100, 
                    p = 500,
                    K = 3,
                    outlier.prop = 0.01,
                    cs = 0.1,
                    sigma = 0.2,
                    B.sd = 5,
                    B.mean = 0,
                    U.sd = 1.0,
                    V.sd = 1.0)
                    
Y &lt;- pnorm(dat$Y)
X &lt;- dat$X

## Fit an LFMM with 2 latent factors
mod.lfmm &lt;- lfmm_ridge(Y = Y,
                       X = X, 
                       K = 3)
                       
## Perform association testing using the fitted model:
pv &lt;- glm_test(Y = pnorm(Y), 
                X = X,
                lfmm.obj = mod.lfmm, 
                family = binomial(link = "probit"),
                calibrate = "gif")
                
## Manhattan plot with true associations shown
causal &lt;- dat$outlier
pvalues &lt;- pv$calibrated.pvalue
plot(-log10(pvalues), 
     pch = 19, 
     cex = .3,
     xlab = "Probe",
     col = "grey")
     
points(causal, 
      -log10(pvalues)[causal], 
       col = "blue")
</code></pre>

<hr>
<h2 id='lfmm'>R package : Fast and Accurate statistical methods for adjusting confounding factors
in association studies.</h2><span id='topic+lfmm'></span>

<h3>Description</h3>

<p>Implements statistical methods for adjusting confounding factors
in association studies.
</p>


<h3>References</h3>

<p>Caye, K., B. Jumentier, J. Lepeule, and O. François, 2019  LFMM 2: fast and accurate inference of gene-environment associations in genome-widestudies. Mol. Biol. Evol. 36: 852–860.https://doi.org/10.1093/molbev/msz008
</p>
<p>B. Jumentier, Caye, K., J. Lepeule, and O. François, 2019 Sparse latent factor regression models for genome-wide and epigenome-wide association studies (in prep)
</p>

<hr>
<h2 id='lfmm_lasso'>LFMM least-squares estimates with lasso penalty (Sparse LFMM)</h2><span id='topic+lfmm_lasso'></span>

<h3>Description</h3>

<p>This function computes regularized least squares estimates
for latent factor mixed models using a lasso penalty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lfmm_lasso(
  Y,
  X,
  K,
  nozero.prop = 0.01,
  mu.num = 100,
  mu.min.ratio = 0.01,
  mu = NULL,
  it.max = 100,
  relative.err.epsilon = 1e-06
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lfmm_lasso_+3A_y">Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns.
Each column is a response variable (e.g., SNP genotype,
gene expression level, beta-normalized methylation profile, etc).
Response variables must be encoded as numeric.</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_x">X</code></td>
<td>
<p>an explanatory variable matrix with n rows and d columns.
Each column corresponds to a distinct explanatory variable (eg. phenotype, exposure, outcome).
Explanatory variables must be encoded as numeric.</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_k">K</code></td>
<td>
<p>an integer for the number of latent factors in the regression model.</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_nozero.prop">nozero.prop</code></td>
<td>
<p>a numeric value for the expected proportion of non-zero effect sizes.</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_mu.num">mu.num</code></td>
<td>
<p>a numeric value for the number of 'mu' values (advance parameter).</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_mu.min.ratio">mu.min.ratio</code></td>
<td>
<p>(advance parameter) A fraction of <code>mu.max</code>, the data derived entry value
(i.e. the smallest value for which all coefficients are zero).</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_mu">mu</code></td>
<td>
<p>(advance parameter) Smallest value of <code>mu</code>. Null value by default.</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_it.max">it.max</code></td>
<td>
<p>an integer value for the number of iterations of the algorithm.</p>
</td></tr>
<tr><td><code id="lfmm_lasso_+3A_relative.err.epsilon">relative.err.epsilon</code></td>
<td>
<p>a numeric value for a relative convergence error. Determine
whether the algorithm converges or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The algorithm minimizes the following penalized least-squares criterion
</p>
<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>an object of class <code>lfmm</code> with the following attributes:
</p>

<ul>
<li><p> U the latent variable score matrix with dimensions n x K,
</p>
</li>
<li><p> V the latent variable axes matrix with dimensions p x K,
</p>
</li>
<li><p> B the effect size matrix with dimensions p x d.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>References</h3>

<p>B. Jumentier, Caye, K., J. Lepeule, and O. François, 2019 Sparse latent factor regression models for genome-wide and epigenome-wide association studies (in prep)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(lfmm)

## An EWAS example with Y = methylation data 
## and X = exposure

## Simulate the data 

dat &lt;- lfmm_sampler(n = 100, 
                    p = 1000,
                    K = 3,
                    outlier.prop = 0.02,
                    cs = 0.1,
                    sigma = 0.2,
                    B.sd = 5,
                    B.mean = 0,
                    U.sd = 1.0,
                    V.sd = 1.0)

Y &lt;- scale(dat$Y)
X &lt;- scale(dat$X)

## Fit an LFMM with 2 latent factors
mod.lfmm &lt;- lfmm_lasso(Y = Y,
                       X = X, 
                       K = 3,
                       nozero.prop = 0.02)
                       
## Manhattan plot of sparse effect sizes
effect &lt;- mod.lfmm$B
causal &lt;- dat$outlier

plot(effect, 
     pch = 19, 
     cex = .3,
     xlab = "Probe",
     col = "grey")
     
points(causal, 
       effect[causal], 
       col = "blue")
</code></pre>

<hr>
<h2 id='lfmm_ridge'>LFMM least-squares estimates with ridge penalty</h2><span id='topic+lfmm_ridge'></span>

<h3>Description</h3>

<p>This function computes regularized least squares estimates
for latent factor mixed models using a ridge penalty.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lfmm_ridge(
  Y,
  X,
  K,
  lambda = 1e-05,
  algorithm = "analytical",
  it.max = 100,
  relative.err.min = 1e-06
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lfmm_ridge_+3A_y">Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns.
Each column corresponds to a distinct response variable (e.g., SNP genotype,
gene expression level, beta-normalized methylation profile, etc).
Response variables must be encoded as numeric.</p>
</td></tr>
<tr><td><code id="lfmm_ridge_+3A_x">X</code></td>
<td>
<p>an explanatory variable matrix with n rows and d columns.
Each column corresponds to a distinct explanatory variable (eg. phenotype, exposure, outcome).
Explanatory variables must be encoded as numeric variables.</p>
</td></tr>
<tr><td><code id="lfmm_ridge_+3A_k">K</code></td>
<td>
<p>an integer for the number of latent factors in the regression model.</p>
</td></tr>
<tr><td><code id="lfmm_ridge_+3A_lambda">lambda</code></td>
<td>
<p>a numeric value for the regularization parameter.</p>
</td></tr>
<tr><td><code id="lfmm_ridge_+3A_algorithm">algorithm</code></td>
<td>
<p>exact (analytical) algorithm or numerical algorithm.
The exact algorithm is based on the global minimum of the loss function and
computation is very fast. The numerical algorithm converges toward a local
minimum of the loss function. The exact method should preferred. The numerical method is
for very large n.</p>
</td></tr>
<tr><td><code id="lfmm_ridge_+3A_it.max">it.max</code></td>
<td>
<p>an integer value for the number of iterations for the
numerical algorithm.</p>
</td></tr>
<tr><td><code id="lfmm_ridge_+3A_relative.err.min">relative.err.min</code></td>
<td>
<p>a numeric value for a relative convergence error. Test
whether the numerical algorithm converges or not (numerical algorithm only).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The algorithm minimizes the following penalized least-squares criterion
</p>
<p style="text-align: center;"><code class="reqn"> L(U, V, B) = \frac{1}{2} ||Y - U V^{T} - X B^T||_{F}^2 
+ \frac{\lambda}{2} ||B||^{2}_{2} ,</code>
</p>

<p>where Y is a response data matrix, X contains all explanatory variables,
U denotes the score matrix, V is the loading matrix, B is the (direct) effect
size matrix, and lambda is a regularization parameter.
</p>
<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>an object of class <code>lfmm</code> with the following attributes:
</p>

<ul>
<li><p> U the latent variable score matrix with dimensions n x K,
</p>
</li>
<li><p> V the latent variable axis matrix with dimensions p x K,
</p>
</li>
<li><p> B the effect size matrix with dimensions p x d.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>References</h3>

<p>Caye, K., B. Jumentier, J. Lepeule, and O. François, 2019  LFMM 2: fast and accurate inference of gene-environment associations in genome-widestudies. Mol. Biol. Evol. 36: 852–860.https://doi.org/10.1093/molbev/msz008
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(lfmm)

## a GWAS example with Y = SNPs and X = phenotype
data(example.data)
Y &lt;- example.data$genotype[, 1:10000]
X &lt;- example.data$phenotype

## Fit an LFMM with K = 6 factors
mod.lfmm &lt;- lfmm_ridge(Y = Y, 
                       X = X, 
                       K = 6)

## Perform association testing using the fitted model:
pv &lt;- lfmm_test(Y = Y, 
                X = X, 
                lfmm = mod.lfmm, 
                calibrate = "gif")

## Manhattan plot with causal loci shown

pvalues &lt;- pv$calibrated.pvalue
plot(-log10(pvalues), pch = 19, 
     cex = .2, col = "grey", xlab = "SNP")
points(example.data$causal.set[1:5], 
      -log10(pvalues)[example.data$causal.set[1:5]], 
       type = "h", col = "blue")


## An EWAS example with Y = methylation data and X = exposure
Y &lt;- skin.exposure$beta.value
X &lt;- as.numeric(skin.exposure$exposure)

## Fit an LFMM with 2 latent factors
mod.lfmm &lt;- lfmm_ridge(Y = Y,
                       X = X, 
                       K = 2)
                       
## Perform association testing using the fitted model:
pv &lt;- lfmm_test(Y = Y, 
                X = X,
                lfmm = mod.lfmm, 
                calibrate = "gif")
                
## Manhattan plot with true associations shown
pvalues &lt;- pv$calibrated.pvalue
plot(-log10(pvalues), 
     pch = 19, 
     cex = .3,
     xlab = "Probe",
     col = "grey")
     
causal.set &lt;- seq(11, 1496, by = 80)
points(causal.set, 
      -log10(pvalues)[causal.set], 
       col = "blue")
</code></pre>

<hr>
<h2 id='lfmm_sampler'>LFMM generative data sampler</h2><span id='topic+lfmm_sampler'></span>

<h3>Description</h3>

<p>Simulate data from the latent factor model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lfmm_sampler(
  n,
  p,
  K,
  outlier.prop,
  cs,
  sigma = 0.2,
  B.sd = 1,
  B.mean = 0,
  U.sd = 1,
  V.sd = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lfmm_sampler_+3A_n">n</code></td>
<td>
<p>number of observations.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_p">p</code></td>
<td>
<p>number of response variables.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_k">K</code></td>
<td>
<p>number of latent variables (factors).</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_outlier.prop">outlier.prop</code></td>
<td>
<p>proportion of outlier.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_cs">cs</code></td>
<td>
<p>correlation between X and U.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_sigma">sigma</code></td>
<td>
<p>standard deviation of residual errors.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_b.sd">B.sd</code></td>
<td>
<p>standard deviation for the effect size (B).</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_b.mean">B.mean</code></td>
<td>
<p>mean of B.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_u.sd">U.sd</code></td>
<td>
<p>standard deviations for K factors.</p>
</td></tr>
<tr><td><code id="lfmm_sampler_+3A_v.sd">V.sd</code></td>
<td>
<p>standard deviations for loadings.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>lfmm_sampler()</code> sample a response matrix Y and a primary variable X such that</p>
<pre>     Y = U t(V) + X t(B) + Epsilon.
</pre>
<p>U,V, B and Epsilon are simulated according to normal multivariate distributions.
Moreover U and X are such that <code>cor(U[,i], X) = cs[i]</code>.
</p>


<h3>Value</h3>

<p>A list with simulated data.
</p>


<h3>Author(s)</h3>

<p>kevin caye, olivier francois
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dat &lt;- lfmm_sampler(n = 100, 
                    p = 1000, 
                    K = 3,
                    outlier.prop = 0.1,
                    cs = c(0.8),
                    sigma = 0.2,
                    B.sd = 1.0, 
                    B.mean = 0.0,
                    U.sd = 1.0, 
                    V.sd = 1.0)
</code></pre>

<hr>
<h2 id='lfmm_test'>Statistical tests with latent factor mixed models (linear models)</h2><span id='topic+lfmm_test'></span>

<h3>Description</h3>

<p>This function returns significance values for the association between each column of the
response matrix, Y, and the explanatory variables, X, including correction for unobserved confounders
(latent factors). The test is based on an LFMM fitted with a ridge or lasso penalty (linear model).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lfmm_test(Y, X, lfmm, calibrate = "gif")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lfmm_test_+3A_y">Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns.
Each column is a response variable (numeric).</p>
</td></tr>
<tr><td><code id="lfmm_test_+3A_x">X</code></td>
<td>
<p>an explanatory variable matrix with n rows and d columns.
Each column corresponds to an explanatory variable (numeric).</p>
</td></tr>
<tr><td><code id="lfmm_test_+3A_lfmm">lfmm</code></td>
<td>
<p>an object of class <code>lfmm</code> returned by the <a href="#topic+lfmm_lasso">lfmm_lasso</a>
or <a href="#topic+lfmm_ridge">lfmm_ridge</a> function</p>
</td></tr>
<tr><td><code id="lfmm_test_+3A_calibrate">calibrate</code></td>
<td>
<p>a character string, &quot;gif&quot; or &quot;median+MAD&quot;. If the &quot;gif&quot; option is set (default),
significance values are calibrated by using the genomic control method. Genomic control
uses a robust estimate of the variance of z-scores called &quot;genomic inflation factor&quot;.
If the &quot;median+MAD&quot; option is set, the pvalues are calibrated by computing the median and MAD of the zscores. If <code>NULL</code>, the
pvalues are not calibrated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>a list with the following attributes:
</p>

<ul>
<li><p> B the effect size matrix with dimensions p x d.
</p>
</li>
<li><p> score a p x d matrix which contains z-scores for each explanatory variable (columns of X),
</p>
</li>
<li><p> pvalue a p x d matrix which contains p-values for each explanatory variable,
</p>
</li>
<li><p> calibrated.pvalue a p x d matrix which contains calibrated p-values for each explanatory variable,
</p>
</li>
<li><p> gif a numeric value for the genomic inflation factor.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>See Also</h3>

<p><a href="#topic+glm_test">glm_test</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(lfmm)

## a GWAS example with Y = SNPs and X = phenotype
data(example.data)
Y &lt;- example.data$genotype[, 1:10000]
X &lt;- example.data$phenotype

## Fit an LFMM with K = 6 factors
mod.lfmm &lt;- lfmm_ridge(Y = Y, 
                       X = X, 
                       K = 6)

## Perform association testing using the fitted model:
pv &lt;- lfmm_test(Y = Y, 
                X = X, 
                lfmm = mod.lfmm, 
                calibrate = "gif")

## Manhattan plot with causal loci shown

pvalues &lt;- pv$calibrated.pvalue
plot(-log10(pvalues), pch = 19, 
     cex = .2, col = "grey", xlab = "SNP")
points(example.data$causal.set[1:5], 
      -log10(pvalues)[example.data$causal.set[1:5]], 
       type = "h", col = "blue")


## An EWAS example with Y = methylation data and X = exposure
data("skin.exposure")
Y &lt;- scale(skin.exposure$beta.value)
X &lt;- scale(as.numeric(skin.exposure$exposure))

## Fit an LFMM with 2 latent factors
mod.lfmm &lt;- lfmm_ridge(Y = Y,
                       X = X, 
                       K = 2)
                       
## Perform association testing using the fitted model:
pv &lt;- lfmm_test(Y = Y, 
                X = X,
                lfmm = mod.lfmm, 
                calibrate = "gif")
                
## Manhattan plot with true associations shown
pvalues &lt;- pv$calibrated.pvalue
plot(-log10(pvalues), 
     pch = 19, 
     cex = .3,
     xlab = "Probe",
     col = "grey")
     
causal.set &lt;- seq(11, 1496, by = 80)
points(causal.set, 
      -log10(pvalues)[causal.set], 
       col = "blue")
</code></pre>

<hr>
<h2 id='predict_lfmm'>Predict polygenic scores from latent factor models</h2><span id='topic+predict_lfmm'></span>

<h3>Description</h3>

<p>This function computes polygenic risk scores from the estimates of latent factor models.
It uses the indirect' effect sizes for the regression of X (a single phenotype) on the matrix Y,
for predicting phenotypic values for new genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_lfmm(Y, X, lfmm.object, fdr.level = 0.1, newdata = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predict_lfmm_+3A_y">Y</code></td>
<td>
<p>a response variable matrix with n rows and p columns, typically containing genotypes.
Each column is a response variable (numeric).</p>
</td></tr>
<tr><td><code id="predict_lfmm_+3A_x">X</code></td>
<td>
<p>an explanatory variable with n rows and d = 1 column (numeric) representing a phenotype
with zero mean across the sample.</p>
</td></tr>
<tr><td><code id="predict_lfmm_+3A_lfmm.object">lfmm.object</code></td>
<td>
<p>an object of class <code>lfmm</code> returned by the <a href="#topic+lfmm_lasso">lfmm_lasso</a>
or <a href="#topic+lfmm_ridge">lfmm_ridge</a> function, computed for X and Y.</p>
</td></tr>
<tr><td><code id="predict_lfmm_+3A_fdr.level">fdr.level</code></td>
<td>
<p>a numeric value for the FDR level in the lfmm test used to define
candidate variables for predicting new phenotypes.</p>
</td></tr>
<tr><td><code id="predict_lfmm_+3A_newdata">newdata</code></td>
<td>
<p>a matrix with n rows and p' columns, and similar to Y, on which
predictions of X will be based. If NULL, Y is used as new data.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The response variable matrix Y and the explanatory variable are centered.
</p>


<h3>Value</h3>

<p>a list with the following attributes:
</p>

<ul>
<li><p> prediction: a vector of length n containing the predicted values for X. If newdata
= NULL, the fitted values are returned.
</p>
</li>
<li><p> candidates: a vector of candidate columns of Y on which the predictions are built.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Kevin Caye, Basile Jumentier, Olivier Francois
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(lfmm)

## Simulation of 1000 genotypes for 100 individuals (y)
u &lt;- matrix(rnorm(300, sd = 1), nrow = 100, ncol = 3) 
v &lt;- matrix(rnorm(3000, sd = 3), nrow = 3, ncol = 1000)
w &lt;- u %*% v
y &lt;- matrix(rbinom(100000, size = 2, 
                  prob = 1/(1 + exp(-0.3 * (w 
                  + rnorm(100000, sd = 2))))),
                  nrow = 100,
                  ncol = 1000)

#PCA of genotypes, 2 main axes of variation (K = 2) 
plot(prcomp(y))
  
## Simulation of 1000 phenotypes (x)
## Only the last 10 genotypes have significant effect sizes (b)
b &lt;- matrix(c(rep(0, 990), rep(6000, 10)))
x &lt;- y%*%b + rnorm(100, sd = 100)

## Compute effect sizes using lfmm_ridge
mod &lt;- lfmm_ridge(Y = y, 
                  X = x,
                  K = 2)
              
x.pred &lt;- predict_lfmm(Y = y, 
                       X = x,
                       fdr.level = 0.25, 
                       mod)
                    
x.pred$candidates

##Compare simulated and predicted/fitted phenotypes
plot(x - mean(x), x.pred$pred, 
     pch = 19, col = "grey", 
     xlab = "Observed phenotypes (centered)", 
     ylab = "Predicted from PRS")
abline(0,1)
abline(lm(x.pred$pred ~ scale(x, scale = FALSE)), col = 2)
</code></pre>

<hr>
<h2 id='skin.exposure'>Simulated (and real) methylation levels for sun exposed patient patients</h2><span id='topic+skin.exposure'></span>

<h3>Description</h3>

<p>A data set containing normalized beta values, and sun exposure and simulated
phenotypic data for 78 tissue samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("skin.exposure")
</code></pre>


<h3>Format</h3>

<p>A list with 6 arguments: beta.value, phenotype, causal.set, chrpos
</p>


<h3>Details</h3>

<p>The variables are:
</p>

<ul>
<li><p> beta.value: 1496 filtered normalized beta values (methyation probabilities)
for 78 tissue samples.
</p>
</li>
<li><p> exposure: Sun exposure levels for 78 tissue samples.
</p>
</li>
<li><p> phenotype: Simulated binary phenotypic data for 78 tissue samples.
</p>
</li>
<li><p>  age: age of patients.
</p>
</li>
<li><p> gender: sex of patients.
</p>
</li>
<li><p> tissue: category for tissue samples.
</p>
</li></ul>

<p>Reference: Vandiver et al (2015).
Age and sun exposure-related widespread genomic blocks of hypomethylation in nonmalignant skin.
Genome Biol. 16.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
