<!DOCTYPE html><html lang="en"><head><title>Help for package lfproQC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lfproQC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#lfproQC-package'><p>lfproQC: Quality Control for Label-Free Proteomics Expression Data</p></a></li>
<li><a href='#best_combination'><p>Best combination of normalization and imputation method</p></a></li>
<li><a href='#Boxplot_data'><p>Creating Boxplot for a dataset</p></a></li>
<li><a href='#Corrplot_data'><p>Creating Correlation matrix plot for a dataset</p></a></li>
<li><a href='#Densityplot_data'><p>Creating Density plot for a dataset</p></a></li>
<li><a href='#MAplot_DE_fn'><p>Find out the Up and Down regulated proteins from MA plot</p></a></li>
<li><a href='#MDSplot_data'><p>Creating MDS plot for a dataset</p></a></li>
<li><a href='#QQplot_data'><p>Creating QQ-Plot for a dataset</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
<li><a href='#rlr_knn_yeast_data'><p>Normalized and imputed complete yeast lysate - UPS1 benchmark dataset</p></a></li>
<li><a href='#top_table_fn'><p>Creating the top table</p></a></li>
<li><a href='#volcanoplot_DE_fn'><p>Find out the Up and Down regulated proteins from volcano plot</p></a></li>
<li><a href='#yeast_data'><p>Yeast lysate - UPS1 benchmark dataset</p></a></li>
<li><a href='#yeast_groups'><p>Group information of Yeast lysate - UPS1 benchmark dataset</p></a></li>
<li><a href='#yeast_top_table'><p>Top table for Yeast Lysate - UPS1 dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quality Control for Label-Free Proteomics Expression Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kabilan S &lt;kabilan151414@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Label-free bottom-up proteomics expression data is often affected by data heterogeneity and missing values. Normalization and missing value imputation are commonly used techniques to address these issues and make the dataset suitable for further downstream analysis. This package provides an optimal combination of normalization and imputation methods for the dataset. The package utilizes three normalization methods and three imputation methods.The statistical evaluation measures named pooled co-efficient of variance, pooled estimate of variance and pooled median absolute deviation are used for selecting the best combination of normalization and imputation method for the given dataset. The user can also visualize the results by using various plots available in this package. The user can also perform the differential expression  analysis between two sample groups with the function included in this package. The chosen three normalization methods, three imputation methods and three evaluation measures were chosen for this study based on the research papers published by	Välikangas et al. (2016) &lt;<a href="https://doi.org/10.1093%2Fbib%2Fbbw095">doi:10.1093/bib/bbw095</a>&gt;, Jin et al. (2021) &lt;<a href="https://doi.org/10.1038%2Fs41598-021-81279-4">doi:10.1038/s41598-021-81279-4</a>&gt; and Srivastava et al. (2023) &lt;<a href="https://doi.org/10.2174%2F1574893618666230223150253">doi:10.2174/1574893618666230223150253</a>&gt;.</td>
</tr>
<tr>
<td>Imports:</td>
<td>VIM, dplyr, limma, matrixStats, pcaMethods, vsn, reshape2,
laeken, ggplot2, Hmisc, reshape, stats, tidyr, magrittr,
plotly, MASS, tidyselect</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, plyr, rmarkdown, testthat (&ge; 3.0.0), tibble</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kabilansbio/lfproQC">https://github.com/kabilansbio/lfproQC</a>,
<a href="https://dabiniasri.shinyapps.io/lfproQC">https://dabiniasri.shinyapps.io/lfproQC</a>,
<a href="http://omics.icar.gov.in/lfproQC">http://omics.icar.gov.in/lfproQC</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kabilansbio/lfproQC/issues">https://github.com/kabilansbio/lfproQC/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-10 12:49:22 UTC; kabil</td>
</tr>
<tr>
<td>Author:</td>
<td>Kabilan S [aut, cre],
  Dr Shashi Bhushan Lal [aut, ths],
  Dr Sudhir Srivastava [aut, ths],
  Dr Krishna Kumar Chaturvedi [ths],
  Dr Yasin Jeshima K [ths],
  Dr Ramasubramanian V [ths],
  Dr Girish Kumar Jha [ctb]</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-10 13:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='lfproQC-package'>lfproQC: Quality Control for Label-Free Proteomics Expression Data</h2><span id='topic+lfproQC'></span><span id='topic+lfproQC-package'></span>

<h3>Description</h3>

<p>Label-free bottom-up proteomics expression data is often affected by data heterogeneity and missing values. Normalization and missing value imputation are commonly used techniques to address these issues and make the dataset suitable for further downstream analysis. This package provides an optimal combination of normalization and imputation methods for the dataset. The package utilizes three normalization methods and three imputation methods.The statistical evaluation measures named pooled co-efficient of variance, pooled estimate of variance and pooled median absolute deviation are used for selecting the best combination of normalization and imputation method for the given dataset. The user can also visualize the results by using various plots available in this package. The user can also perform the differential expression analysis between two sample groups with the function included in this package. The chosen three normalization methods, three imputation methods and three evaluation measures were chosen for this study based on the research papers published by Välikangas et al. (2016) <a href="https://doi.org/10.1093/bib/bbw095">doi:10.1093/bib/bbw095</a>, Jin et al. (2021) <a href="https://doi.org/10.1038/s41598-021-81279-4">doi:10.1038/s41598-021-81279-4</a> and Srivastava et al. (2023) <a href="https://doi.org/10.2174/1574893618666230223150253">doi:10.2174/1574893618666230223150253</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Kabilan S <a href="mailto:kabilan151414@gmail.com">kabilan151414@gmail.com</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Dr Shashi Bhushan Lal <a href="mailto:sb.lal@icar.gov.in">sb.lal@icar.gov.in</a> [thesis advisor]
</p>
</li>
<li><p> Dr Sudhir Srivastava <a href="mailto:Sudhir.Srivastava@icar.gov.in">Sudhir.Srivastava@icar.gov.in</a> [thesis advisor]
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Dr Krishna Kumar Chaturvedi <a href="mailto:kk.chaturvedi@icar.gov.in">kk.chaturvedi@icar.gov.in</a> [thesis advisor]
</p>
</li>
<li><p> Dr Yasin Jeshima K <a href="mailto:yasinlab1.icar@gmail.com">yasinlab1.icar@gmail.com</a> [thesis advisor]
</p>
</li>
<li><p> Dr Ramasubramanian V <a href="mailto:R.Subramanian@icar.gov.in">R.Subramanian@icar.gov.in</a> [thesis advisor]
</p>
</li>
<li><p> Dr Girish Kumar Jha <a href="mailto:girish.jha@icar.gov.in">girish.jha@icar.gov.in</a> [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/kabilansbio/lfproQC">https://github.com/kabilansbio/lfproQC</a>
</p>
</li>
<li> <p><a href="https://dabiniasri.shinyapps.io/lfproQC">https://dabiniasri.shinyapps.io/lfproQC</a>
</p>
</li>
<li> <p><a href="http://omics.icar.gov.in/lfproQC">http://omics.icar.gov.in/lfproQC</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/kabilansbio/lfproQC/issues">https://github.com/kabilansbio/lfproQC/issues</a>
</p>
</li></ul>


<hr>
<h2 id='best_combination'>Best combination of normalization and imputation method</h2><span id='topic+best_combination'></span>

<h3>Description</h3>

<p>This function will provide the best combinations of normalization
and imputation methods for the user given dataset based on the intragroup
variation evaluation parameters called PCV, PEV, PMAD and NRMSE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>best_combination(data_input, groups, data_type, aggr_method)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="best_combination_+3A_data_input">data_input</code></td>
<td>
<p>Label-free proteomics expression data as a dataframe</p>
</td></tr>
<tr><td><code id="best_combination_+3A_groups">groups</code></td>
<td>
<p>Group information about the input data</p>
</td></tr>
<tr><td><code id="best_combination_+3A_data_type">data_type</code></td>
<td>
<p>A character string specifying the type of data being used. Use &quot;Peptide&quot; if your dataset contains peptide information, where the first column represents peptide IDs and the second column represents protein IDs. Use &quot;Protein&quot; if your dataset consists of protein data, where the first column represents protein IDs. The function will handle the data accordingly based on this parameter.</p>
</td></tr>
<tr><td><code id="best_combination_+3A_aggr_method">aggr_method</code></td>
<td>
<p>A character string specifying the method for aggregating peptide data to corresponding protein values. Use &quot;sum&quot; to aggregate by the sum of peptide intensities, &quot;mean&quot; to aggregate by the mean of peptide intensities, or &quot;median&quot; to aggregate by the median of peptide intensities. This parameter is only applicable when &quot;data_type&quot; is set to &quot;Peptide&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Label-free LC-MS proteomics expression data is often affected by heterogeneity and missing values. 
Normalization and missing value imputation are the commonly used techniques to solve these issues and make the dataset suitable for further downstream analysis. 
This function provides the best combination of normalization and imputation methods for the dataset, choosing from the three normalization methods (vsn, loess, and rlr) and three imputation methods (knn, lls, svd). 
The intragroup variation evaluation measures named pooled co-efficient of variance (PCV), pooled estimate of variance (PEV) and pooled median absolute deviation (PMAD) are used for selecting the best combination of normalization and imputation method for the given dataset.
It will return the best combinations based on each evaluation parameters of PCV, PEV, and PMAD. 
</p>
<p>Along with this, the user can get all three normalized datasets, nine combinations of normalized and missing values imputed datasets, and the PCV, PEV, and PMAD result values. The user can also obtain the Normalized Root Mean Square Error (NRMSE) values, 
calculated by comparing the normalized and imputed dataset with the original dataset across all nine combinations.
These NRMSE values provide insight into the accuracy of the imputation and normalization processes.
</p>


<h3>Value</h3>

<p>This function gives the list  which consist of following results.
</p>
<p>'Best Combinations'  The best combinations based on each PCV, PEV and PMAD
for the given dataset.
</p>
<p>'PCV Result' Values of groupwise PCV, overall PCV, PCV mean, PCV median and
PCV standard deviation for all combinations.
</p>
<p>'PEV Result' Values of groupwise PEV, overall PEV, PEV mean, PEV median and
PEV standard deviation for all combinations.
</p>
<p>'PMAD Result' Values of groupwise PMAD, overall PMAD, PMAD mean, PMAD median
and PMAD standard deviation for all combinations.
</p>
<p>'NRMSE Result' NRMSE values calculated for the normalized and imputed dataset 
to the original dataset. 
</p>
<p>'rollup_protein' The aggregated protein values for the peptide dataset are 
based on either the sum, mean, or median.
</p>
<p>'vsn_data' The 'vsn' normalized dataset
</p>
<p>'loess_data' The 'loess' normalized dataset
</p>
<p>'rlr_data' The 'rlr' normalized dataset
</p>
<p>'vsn_knn_data' The dataset normalized by 'vsn' method and missing values imputed
by 'knn' method.
</p>
<p>'vsn_lls_data' The dataset normalized by 'vsn' method and missing values imputed
by 'lls' method.
</p>
<p>'vsn_svd_data' The dataset normalized by 'vsn' method and missing values imputed
by 'svd' method.
</p>
<p>'loess_knn_data' The dataset normalized by 'loess' method and missing values imputed
by 'knn' method.
</p>
<p>'loess_lls_data' The dataset normalized by 'loess' method and missing values imputed
by 'lls' method.
</p>
<p>'loess_svd_data' The dataset normalized by 'loess' method and missing values imputed
by 'svd' method.
</p>
<p>'rlr_knn_data' The dataset normalized by 'rlr' method and missing values imputed
by 'knn' method.
</p>
<p>'rlr_lls_data' The dataset normalized by 'rlr' method and missing values imputed
by 'lls' method.
</p>
<p>'rlr_svd_data' The dataset normalized by 'rlr' method and missing values imputed
by 'svd' method.
</p>


<h3>Author(s)</h3>

<p>Dr Sudhir Srivastava (&quot;Sudhir.Srivastava@icar.gov.in&quot;)
</p>
<p>Kabilan S (&quot;kabilan151414@gmail.com&quot;)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
result &lt;- best_combination(yeast_data, yeast_groups, data_type = "Protein")
result$`Best combinations`
result$`PCV Result`
result$`PMAD Result`
result$`rlr_knn_data`

</code></pre>

<hr>
<h2 id='Boxplot_data'>Creating Boxplot for a dataset</h2><span id='topic+Boxplot_data'></span>

<h3>Description</h3>

<p>The box and whiskers plot displays the distribution of a continuous variable. 
It visualises five summary statistics (the median, two hinges and two whiskers), and all &quot;outlying&quot; points individually. 
The 'ggplot2' package is used here for creating the boxplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Boxplot_data(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Boxplot_data_+3A_data">data</code></td>
<td>
<p>Proteomics expression dataset (original or normalized dataset)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can also be used for comparing the original dataset with the normalized dataset.
</p>


<h3>Value</h3>

<p>Interactive box and whiskers plot
</p>


<h3>See Also</h3>

<p>'geom_boxplot()'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Boxplot_data(yeast_data)
Boxplot_data(rlr_knn_yeast_data)
</code></pre>

<hr>
<h2 id='Corrplot_data'>Creating Correlation matrix plot for a dataset</h2><span id='topic+Corrplot_data'></span>

<h3>Description</h3>

<p>A graphical display of a correlation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Corrplot_data(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Corrplot_data_+3A_data">data</code></td>
<td>
<p>Proteomics expression dataset (original or normalized dataset) along with the protein information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can also be used for comparing the original dataset with the normalized dataset.
</p>


<h3>Value</h3>

<p>Interactive corrleation matrix plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Corrplot_data(yeast_data)
Corrplot_data(rlr_knn_yeast_data)
</code></pre>

<hr>
<h2 id='Densityplot_data'>Creating Density plot for a dataset</h2><span id='topic+Densityplot_data'></span>

<h3>Description</h3>

<p>Computes and draws kernel density estimate, which is a smoothed version of the histogram. 
This is a useful alternative to the histogram for continuous data that comes from an underlying smooth distribution. 
The 'ggplot2' package is used here for creating the boxplot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Densityplot_data(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Densityplot_data_+3A_data">data</code></td>
<td>
<p>Proteomics expression dataset (original or normalized dataset) along with the protein information</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can also be used for comparing the original dataset with the normalized dataset.
</p>


<h3>Value</h3>

<p>Interactive column-wise density plot
</p>


<h3>See Also</h3>

<p>'geom_density()'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Densityplot_data(yeast_data)
Densityplot_data(rlr_knn_yeast_data)
</code></pre>

<hr>
<h2 id='MAplot_DE_fn'>Find out the Up and Down regulated proteins from MA plot</h2><span id='topic+MAplot_DE_fn'></span>

<h3>Description</h3>

<p>MA plot is used for visualizing the differentially expressed proteins by plotting the log mean intensity data in x axis and log fold change values in y axis.
</p>
<p>This function can be used for visualizing the up regulated, down regulated, and non-significant proteins along with their information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAplot_DE_fn(top_table, x1 = NULL, x2 = NULL, p = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MAplot_DE_fn_+3A_top_table">top_table</code></td>
<td>
<p>Top table information</p>
</td></tr>
<tr><td><code id="MAplot_DE_fn_+3A_x1">x1</code></td>
<td>
<p>Cut-off limit for down-regulated proteins</p>
</td></tr>
<tr><td><code id="MAplot_DE_fn_+3A_x2">x2</code></td>
<td>
<p>Cut-off limit for up-regulated proteins</p>
</td></tr>
<tr><td><code id="MAplot_DE_fn_+3A_p">p</code></td>
<td>
<p>Cut-off limit for p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'Result' Top table along with up, down, significant and non-significant protein information.
</p>
<p>'MA plot' Interacive MA plot with the details of up and down regulated proteins
</p>
<p>'Up-regulated' Up-regulated protein information
</p>
<p>'Down-regulated' Down-regulated protein information
</p>
<p>'Non-significant' Non-significant protein information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- MAplot_DE_fn(yeast_top_table, -1, 1, 0.05)
result$`MA Plot`
result$`Result`
result$`Up-regulated`
result$`Down-regulated`
result$`Non-significant`
</code></pre>

<hr>
<h2 id='MDSplot_data'>Creating MDS plot for a dataset</h2><span id='topic+MDSplot_data'></span>

<h3>Description</h3>

<p>Multi-dimensional scaling (MDS) plots showing a 2-dimensional projection of distances between the dataset samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MDSplot_data(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MDSplot_data_+3A_data">data</code></td>
<td>
<p>Normalized and imputed Proteomics expression dataset along with the protein information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>MDS plot
</p>


<h3>See Also</h3>

<p>'mdsPlot'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MDSplot_data(rlr_knn_yeast_data)
</code></pre>

<hr>
<h2 id='QQplot_data'>Creating QQ-Plot for a dataset</h2><span id='topic+QQplot_data'></span>

<h3>Description</h3>

<p>A Q–Q plot (quantile-quantile plot) is a plot of the quantiles of two distributions against each other, or a plot based on estimates of the quantiles. 
The normality of the data can be understand by this plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QQplot_data(data)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QQplot_data_+3A_data">data</code></td>
<td>
<p>Proteomics expression dataset (original or normalized dataset)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can be used for comparing the original dataset with the
normalized dataset.
</p>


<h3>Value</h3>

<p>Interactive column-wise QQ-plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>qqplot &lt;- QQplot_data(rlr_knn_yeast_data)
</code></pre>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>magrittr</dt><dd><p><code><a href="magrittr.html#topic+pipe">%&gt;%</a></code></p>
</dd>
</dl>

<hr>
<h2 id='rlr_knn_yeast_data'>Normalized and imputed complete yeast lysate - UPS1 benchmark dataset</h2><span id='topic+rlr_knn_yeast_data'></span>

<h3>Description</h3>

<p>This is the groupwise normalized and missing values imputed dataset of the complete yeast lysate - UPS1 benchmark dataset. 
Normalization has been done by RLR normalization method and missing values imputation has been done by KNN imputation method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rlr_knn_yeast_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 835 rows and 7 variables:
</p>

<dl>
<dt>Majority protein IDs</dt><dd><p>Protein ID information</p>
</dd>
<dt>A1</dt><dd><p>1st sample group, 1st technical replicate</p>
</dd>
<dt>A2</dt><dd><p>1st sample group, 2nd technical replicate</p>
</dd>
<dt>A3</dt><dd><p>1st sample group, 3rd technical replicate</p>
</dd>
<dt>B1</dt><dd><p>2nd sample group, 1st technical replicate</p>
</dd>
<dt>B2</dt><dd><p>2nd sample group, 2nd technical replicate</p>
</dd>
<dt>B3</dt><dd><p>2nd sample group, 3rd technical replicate</p>
</dd>
</dl>


<hr>
<h2 id='top_table_fn'>Creating the top table</h2><span id='topic+top_table_fn'></span>

<h3>Description</h3>

<p>Top table can be used for identifying the pairwise differential abundance analysis of proteins in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>top_table_fn(data, groups, ch_gr1, ch_gr2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="top_table_fn_+3A_data">data</code></td>
<td>
<p>Normalized and missing values imputed expression dataset containing protein information</p>
</td></tr>
<tr><td><code id="top_table_fn_+3A_groups">groups</code></td>
<td>
<p>Group information about the input data</p>
</td></tr>
<tr><td><code id="top_table_fn_+3A_ch_gr1">ch_gr1</code></td>
<td>
<p>Group number of the dataset for pairwise comparison with the another group</p>
</td></tr>
<tr><td><code id="top_table_fn_+3A_ch_gr2">ch_gr2</code></td>
<td>
<p>Group number of the dataset to be compared with the chosen group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Top table consists of following values
</p>
<p>'logFC' -  Log fold change values,
</p>
<p>'AveExpr' - Average intensity values,
</p>
<p>'t'- t-statistic values,
</p>
<p>'P.Value' - P-values,
</p>
<p>'adj.P.Val' - Adjusted P-values,
</p>
<p>'B'- B-statistic values
</p>


<h3>See Also</h3>

<p>'limma::topTable'
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
top_table &lt;- top_table_fn(rlr_knn_yeast_data, yeast_groups, 2, 1)
top_table

</code></pre>

<hr>
<h2 id='volcanoplot_DE_fn'>Find out the Up and Down regulated proteins from volcano plot</h2><span id='topic+volcanoplot_DE_fn'></span>

<h3>Description</h3>

<p>Volcano plot is used for visualizing the differentially expressed proteins by plotting the log fold change values in x axis and (-log10 p-values) in y axis.
</p>
<p>This function can be used for visualizing the up regulated, down regulated, and non-significant proteins along with their information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcanoplot_DE_fn(top_table, x1 = NULL, x2 = NULL, p = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="volcanoplot_DE_fn_+3A_top_table">top_table</code></td>
<td>
<p>Top table information</p>
</td></tr>
<tr><td><code id="volcanoplot_DE_fn_+3A_x1">x1</code></td>
<td>
<p>Cut-off limit for down-regulated proteins</p>
</td></tr>
<tr><td><code id="volcanoplot_DE_fn_+3A_x2">x2</code></td>
<td>
<p>Cut-off limit for up-regulated proteins</p>
</td></tr>
<tr><td><code id="volcanoplot_DE_fn_+3A_p">p</code></td>
<td>
<p>Cut-off limit for p-values</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'Result' Top table along with up, down, significant and non-significant protein information.
</p>
<p>'Volcano plot' Interactive MA plot with the details of up and down regulated proteins
</p>
<p>'Up-regulated' Up-regulated protein information
</p>
<p>'Down-regulated' Down-regulated protein information
</p>
<p>'Non-significant' Non-significant protein information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>result &lt;- volcanoplot_DE_fn(yeast_top_table, -1, 1, 0.05)
result$`Volcano Plot`
result$`Result`
result$`Up-regulated`
result$`Down-regulated`
result$`Non-significant`
</code></pre>

<hr>
<h2 id='yeast_data'>Yeast lysate - UPS1 benchmark dataset</h2><span id='topic+yeast_data'></span>

<h3>Description</h3>

<p>This dataset was given by Ramus et al., (2016). It is based on a highly complex sample (yeast lysate) spiked with different spiked amounts of the UPS1 standard mixture of 48 recombinant proteins. 
The original dataset contains 2644 rows of proteins and 2 groups of samples with three replicates each. 
From the original dataset, only the pairwise comparison of 50vs0.5 has taken here. This dataset
contains 874 rows of proteins which includes 48 spiked-in ups proteins in the yeast.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 874 rows and 7 variables:
</p>

<dl>
<dt>Majority protein IDs</dt><dd><p>Protein ID information</p>
</dd>
<dt>A1</dt><dd><p>1st condition, 1st technical replicate</p>
</dd>
<dt>A2</dt><dd><p>1st condition, 2nd technical replicate</p>
</dd>
<dt>A3</dt><dd><p>1st condition, 3rd technical replicate</p>
</dd>
<dt>B1</dt><dd><p>2nd condition, 1st technical replicate</p>
</dd>
<dt>B2</dt><dd><p>2nd condition, 2nd technical replicate</p>
</dd>
<dt>B3</dt><dd><p>2nd condition, 3rd technical replicate</p>
</dd>
</dl>



<h3>Details</h3>

<p>This standard proteomic dataset is suitable for benchmarking and comparing software for label-free quantification. 
And can also be applied to the evaluation of post-processing steps such as normalization, imputation of missing values, and statistical methods.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1016/j.dib.2015.11.063">doi:10.1016/j.dib.2015.11.063</a>
</p>

<hr>
<h2 id='yeast_groups'>Group information of Yeast lysate - UPS1 benchmark dataset</h2><span id='topic+yeast_groups'></span>

<h3>Description</h3>

<p>The standard benchmark yeast lysate - UPS1 dataset contains the variables of
6 columns, 2 conditions and 3 technical replicates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_groups
</code></pre>


<h3>Format</h3>

<p>A data frame with 6 rows and 2 variables:
</p>

<dl>
<dt>Samples</dt><dd><p>Technical replicate details - Two kinds of technical replicates
(A1, A2, A3) and (B1, B2, B3)</p>
</dd>
<dt>Groups</dt><dd><p>Condition details - Two conditions (GrA) and (GrB)</p>
</dd>
</dl>



<h3>Details</h3>

<p>The above information is required for further analysis of the dataset.
</p>


<h3>Source</h3>

<p><a href="https://doi.org/10.1016/j.dib.2015.11.063">doi:10.1016/j.dib.2015.11.063</a>
</p>

<hr>
<h2 id='yeast_top_table'>Top table for Yeast Lysate - UPS1 dataset</h2><span id='topic+yeast_top_table'></span>

<h3>Description</h3>

<p>Top table can be used for identifying the differential abundance analysis of proteins in the dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>yeast_top_table
</code></pre>


<h3>Format</h3>

<p>A data frame with 835 rows and 6 variables along with 'Fasta headers':
</p>

<dl>
<dt>logFC</dt><dd><p>Log fold change values</p>
</dd>
<dt>AveExpr</dt><dd><p>Average intensity values</p>
</dd>
<dt>t</dt><dd><p>t-statistic values</p>
</dd>
<dt>P.Value</dt><dd><p>P-values</p>
</dd>
<dt>adj.P.Val</dt><dd><p>Adjusted P-values</p>
</dd>
<dt>B</dt><dd><p>B-statistic values</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
