<!DOCTYPE html><html lang="en"><head><title>Help for package LGEWIS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LGEWIS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#GA.prelim'>
<p>The preliminary data management for GA (tests for genetic association)</p></a></li>
<li><a href='#GA.SSD.All'>
<p>Genetic association tests for multiple regions/genes using SSD format files</p></a></li>
<li><a href='#GA.SSD.OneSet_SetIndex'>
<p>Genetic association tests for a single region/gene using SSD format files</p></a></li>
<li><a href='#GA.test'>
<p>Test the association between an quantitative outcome variable and a region/gene by a generalized score type test.</p></a></li>
<li><a href='#GEI.prelim'>
<p>The preliminary data management for GEI (tests for gene-environment interaction)</p></a></li>
<li><a href='#GEI.SSD.All'>
<p>Gene-environment interaction tests for multiple regions/genes using SSD format files</p></a></li>
<li><a href='#GEI.SSD.OneSet_SetIndex'>
<p>Gene-environment interaction tests for a single region/gene using SSD format files</p></a></li>
<li><a href='#GEI.test'>
<p>Test the interaction between an environemental exposure and a region/gene by a generalized score type test.</p></a></li>
<li><a href='#LGEWIS.example'>
<p>Data example for LGEWIS (tests for genetic association or gene-environment interaction)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Tests for Genetic Association/Gene-Environment Interaction in
Longitudinal Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-8-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Zihuai He, Seunggeun Lee, Bhramar Mukherjee, Min Zhang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zihuai He &lt;zihuai@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for genome-wide association studies (GWAS)/gene-environment-wide interaction studies (GEWIS) with longitudinal outcomes and exposures. He et al. (2017) "Set-Based Tests for Gene-Environment Interaction in Longitudinal Studies" and He et al. (2017) "Rare-variant association tests in longitudinal studies, with an application to the Multi-Ethnic Study of Atherosclerosis (MESA)".</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>CompQuadForm, SKAT, geeM, splines, mvtnorm</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-09-01 13:22:34 UTC; statzihuai</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-09-01 13:30:52 UTC</td>
</tr>
</table>
<hr>
<h2 id='GA.prelim'>
The preliminary data management for GA (tests for genetic association)
</h2><span id='topic+GA.prelim'></span>

<h3>Description</h3>

<p>Before testing a specific region using a generalized score type test, this function does the preliminary data management, such as fitting the model under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GA.prelim(Y,time,X=NULL,corstr="exchangeable")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GA.prelim_+3A_y">Y</code></td>
<td>

<p>The outcome variable, an n*1 matrix where n is the total number of observations
</p>
</td></tr>
<tr><td><code id="GA.prelim_+3A_time">time</code></td>
<td>

<p>An n*2 matrix describing how the observations are measured. The first column is the subject id. The second column is the measured exam (1,2,3,etc.).
</p>
</td></tr>
<tr><td><code id="GA.prelim_+3A_x">X</code></td>
<td>

<p>An n*p covariates matrix where p is the total number of covariates.
</p>
</td></tr>
<tr><td><code id="GA.prelim_+3A_corstr">corstr</code></td>
<td>

<p>The working correlation as specified in 'geeglm'. The following are permitted: &quot;independence&quot;, &quot;exchangeable&quot;, &quot;ar1&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list used for function GA.test().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(LGEWIS)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# G: genotype matrix, m by q matrix where m is the total number of subjects

data(LGEWIS.example)
Y&lt;-LGEWIS.example$Y
X&lt;-LGEWIS.example$X
time&lt;-LGEWIS.example$time
G&lt;-LGEWIS.example$G

# Preliminary data management
result.prelim&lt;-GA.prelim(Y,time,X=X)
</code></pre>

<hr>
<h2 id='GA.SSD.All'>
Genetic association tests for multiple regions/genes using SSD format files
</h2><span id='topic+GA.SSD.All'></span>

<h3>Description</h3>

<p>Test the association between an quantitative outcome and multiple region/genes using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GA.SSD.All(SSD.INFO, result.prelim, Gsub.id=NULL, B=5000, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GA.SSD.All_+3A_ssd.info">SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function &ldquo;Open_SSD&quot;. The sets are defined by this file.
</p>
</td></tr>
<tr><td><code id="GA.SSD.All_+3A_result.prelim">result.prelim</code></td>
<td>

<p>Output of function &quot;GA.prelim()&quot;.
</p>
</td></tr>
<tr><td><code id="GA.SSD.All_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X, E and time.
</p>
</td></tr>
<tr><td><code id="GA.SSD.All_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td></tr>
<tr><td><code id="GA.SSD.All_+3A_...">...</code></td>
<td>

<p>Other options of the generalized score type test. Defined same as in function &quot;GA.test()&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please see SKAT vignettes for using SSD files.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>results</code></td>
<td>

<p>Results of the set based analysis. First column contains the set ID; Second column (second and third columns when the MinP test is compared) contains the p-values; Last column contains the number of tested SNPs.
</p>
</td></tr>
<tr><td><code>results.single</code></td>
<td>

<p>Results of the single variant analysis for all variants in the sets. First column contains the regions' names; Second column is the variants' names; Third column contains the minor allele frequencies; Last column contains the p.values.
</p>
</td></tr>
</table>

<hr>
<h2 id='GA.SSD.OneSet_SetIndex'>
Genetic association tests for a single region/gene using SSD format files
</h2><span id='topic+GA.SSD.OneSet_SetIndex'></span>

<h3>Description</h3>

<p>Test the genetic association between an quantitative outcome and one region/gene using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GA.SSD.OneSet_SetIndex(SSD.INFO, SetIndex, result.prelim, Gsub.id=NULL,
B=5000, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GA.SSD.OneSet_SetIndex_+3A_ssd.info">SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function &ldquo;Open_SSD&quot;. The sets are defined by this file.
</p>
</td></tr>
<tr><td><code id="GA.SSD.OneSet_SetIndex_+3A_setindex">SetIndex</code></td>
<td>

<p>Set index. From 1 to the total number of sets.
</p>
</td></tr>
<tr><td><code id="GA.SSD.OneSet_SetIndex_+3A_result.prelim">result.prelim</code></td>
<td>

<p>Output of function &quot;GA.prelim()&quot;.
</p>
</td></tr>
<tr><td><code id="GA.SSD.OneSet_SetIndex_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X, E and time.
</p>
</td></tr>
<tr><td><code id="GA.SSD.OneSet_SetIndex_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td></tr>
<tr><td><code id="GA.SSD.OneSet_SetIndex_+3A_...">...</code></td>
<td>

<p>Other options of the generalized score type test. Defined same as in function &quot;GA.test()&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please see SKAT vignettes for using SSD files.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
<tr><td><code>p.single</code></td>
<td>

<p>P-values of the incorporated single SNP analyses
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>
</table>

<hr>
<h2 id='GA.test'>
Test the association between an quantitative outcome variable and a region/gene by a generalized score type test.
</h2><span id='topic+GA.test'></span>

<h3>Description</h3>

<p>Once the preliminary work is done using &quot;GA.prelim()&quot;, this function tests a specifc region/gene. Single SNP analyses are also incorporated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GA.test(result.prelim,G,Gsub.id=NULL,weights='beta',B=5000, B.coef=NULL,
impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GA.test_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;GEI.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="GA.test_+3A_g">G</code></td>
<td>

<p>Genetic variants in the target region/gene, an m*q matrix where m is the subject ID and q is the total number of genetic variables. Note that the number of rows in Z should be same as the number of subjects.
</p>
</td></tr>
<tr><td><code id="GA.test_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="GA.test_+3A_weights">weights</code></td>
<td>

<p>Can be a numeric vector of weights for genetic variants (The length should be same as the number of genetic variants in the set.), or pre-determined weights: &quot;beta&quot; (beta weights as in SKAT paper), &quot;rare&quot; (restrited to MAF&lt;0.01), &quot;common&quot; (restrited to MAF&gt;0.01). The default is NULL, where the flat weights are applied.
</p>
</td></tr>
<tr><td><code id="GA.test_+3A_b">B</code></td>
<td>

<p>Number of Bootstrap replicates. The default is 5000.
</p>
</td></tr>
<tr><td><code id="GA.test_+3A_b.coef">B.coef</code></td>
<td>

<p>Direct import of Bootstrap coefficients, an m by B matrix. This is in order to efficiently implement the Bootstrap step. The default is NULL.
</p>
</td></tr>
<tr><td><code id="GA.test_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
<tr><td><code>p.single</code></td>
<td>

<p>P-values of the incorporated single SNP analyses
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of heterozygous SNPs in the SNP set.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## GA.prelim does the preliminary data management.
# Input: Y, time, X (covariates)
## GA.test tests a region.
# Input: G (genetic variants) and result of GEI.prelim

library(LGEWIS)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# G: genotype matrix, m by q matrix where m is the total number of subjects

data(LGEWIS.example)
Y&lt;-LGEWIS.example$Y
X&lt;-LGEWIS.example$X
time&lt;-LGEWIS.example$time
G&lt;-LGEWIS.example$G

# Preliminary data management
result.prelim&lt;-GA.prelim(Y,time,X=X)

# test with 5000 bootstrap replicates
result&lt;-GA.test(result.prelim,G,B=5000)
</code></pre>

<hr>
<h2 id='GEI.prelim'>
The preliminary data management for GEI (tests for gene-environment interaction)
</h2><span id='topic+GEI.prelim'></span>

<h3>Description</h3>

<p>Before testing a specific region using a generalized score type test, this function does the preliminary data management, such as pareparing spline basis functions for E etc..
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEI.prelim(Y,time,E,X=NULL,E.method='ns',E.df=floor(sqrt(length(unique(time[,1])))),
corstr="exchangeable")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GEI.prelim_+3A_y">Y</code></td>
<td>

<p>The outcome variable, an n*1 matrix where n is the total number of observations
</p>
</td></tr>
<tr><td><code id="GEI.prelim_+3A_time">time</code></td>
<td>

<p>An n*2 matrix describing how the observations are measured. The first column is the subject id. The second column is the measured exam (1,2,3,etc.).
</p>
</td></tr>
<tr><td><code id="GEI.prelim_+3A_e">E</code></td>
<td>

<p>An n*1 environmental exposure.
</p>
</td></tr>
<tr><td><code id="GEI.prelim_+3A_x">X</code></td>
<td>

<p>An n*p covariates matrix where p is the total number of covariates.
</p>
</td></tr>
<tr><td><code id="GEI.prelim_+3A_e.method">E.method</code></td>
<td>

<p>The method of sieves for the main effect of E. It can be &quot;ns&quot; for natural cubic spline sieves; &quot;bs&quot; for B-spline sieves; &quot;ps&quot; for polynomial sieves. The default is &quot;ns&quot;.
</p>
</td></tr>
<tr><td><code id="GEI.prelim_+3A_e.df">E.df</code></td>
<td>

<p>Model complexity for the method of sieves, i.e., number of basis functions. The default is sqrt(m).
</p>
</td></tr>
<tr><td><code id="GEI.prelim_+3A_corstr">corstr</code></td>
<td>

<p>The working correlation as specified in 'geeglm'. The following are permitted: &quot;independence&quot;, &quot;exchangeable&quot;, &quot;ar1&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list used for function GEI.test().
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(LGEWIS)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# E: environmental exposure, n by 1 matrix
# time: describe longitudinal structure, n by 2 matrix
# G: genotype matrix, m by q matrix where m is the total number of subjects

data(LGEWIS.example)
Y&lt;-LGEWIS.example$Y
X&lt;-LGEWIS.example$X
E&lt;-LGEWIS.example$E
time&lt;-LGEWIS.example$time
G&lt;-LGEWIS.example$G

# Preliminary data management
result.prelim&lt;-GEI.prelim(Y,time,E,X=X)
</code></pre>

<hr>
<h2 id='GEI.SSD.All'>
Gene-environment interaction tests for multiple regions/genes using SSD format files
</h2><span id='topic+GEI.SSD.All'></span>

<h3>Description</h3>

<p>Test the interaction between an environmental exposure and multiple region/genes on a quantitative outcome using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEI.SSD.All(SSD.INFO, result.prelim, Gsub.id=NULL, MinP.adjust=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GEI.SSD.All_+3A_ssd.info">SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function &ldquo;Open_SSD&quot;. The sets are defined by this file.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.All_+3A_result.prelim">result.prelim</code></td>
<td>

<p>Output of function &quot;GEI.prelim()&quot;.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.All_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X, E and time.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.All_+3A_minp.adjust">MinP.adjust</code></td>
<td>

<p>If the users would like to compare with the MinP test, this parameter specify the adjustment thereshold as in Gao, et al. (2008) &quot;A multiple testing correction method for genetic association studies using correlated single nucleotide polymorphisms&quot;. Values from 0 to 1 are permitted. The default is NULL, i.e., no comparison. The value suggested by Gao, et al. (2008) is 0.95.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.All_+3A_...">...</code></td>
<td>

<p>Other options of the generalized score type test. Defined same as in function &quot;GEI.test()&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please see SKAT vignettes for using SSD files.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>results</code></td>
<td>

<p>Results of the set based analysis. First column contains the set ID; Second column (second and third columns when the MinP test is compared) contains the p-values; Last column contains the number of tested SNPs.
</p>
</td></tr>
<tr><td><code>results.single</code></td>
<td>

<p>Results of the single variant analysis for all variants in the sets. First column contains the regions' names; Second column is the variants' names; Third column contains the minor allele frequencies; Last column contains the p.values.
</p>
</td></tr>
</table>

<hr>
<h2 id='GEI.SSD.OneSet_SetIndex'>
Gene-environment interaction tests for a single region/gene using SSD format files
</h2><span id='topic+GEI.SSD.OneSet_SetIndex'></span>

<h3>Description</h3>

<p>Test the interaction between an environmental exposure and one region/gene on a quantitative outcome using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEI.SSD.OneSet_SetIndex(SSD.INFO, SetIndex, result.prelim, Gsub.id=NULL,
MinP.adjust=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GEI.SSD.OneSet_SetIndex_+3A_ssd.info">SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function &ldquo;Open_SSD&quot;. The sets are defined by this file.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.OneSet_SetIndex_+3A_setindex">SetIndex</code></td>
<td>

<p>Set index. From 1 to the total number of sets.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.OneSet_SetIndex_+3A_result.prelim">result.prelim</code></td>
<td>

<p>Output of function &quot;GEI.prelim()&quot;.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.OneSet_SetIndex_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X, E and time.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.OneSet_SetIndex_+3A_minp.adjust">MinP.adjust</code></td>
<td>

<p>If the users would like to compare with the MinP test, this parameter specify the adjustment thereshold as in Gao, et al. (2008) &quot;A multiple testing correction method for genetic association studies using correlated single nucleotide polymorphisms&quot;. Values from 0 to 1 are permitted. The default is NULL, i.e., no comparison. The value suggested by Gao, et al. (2008) is 0.95.
</p>
</td></tr>
<tr><td><code id="GEI.SSD.OneSet_SetIndex_+3A_...">...</code></td>
<td>

<p>Other options of the generalized score type test. Defined same as in function &quot;GEI.test()&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please see SKAT vignettes for using SSD files.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
<tr><td><code>p.single</code></td>
<td>

<p>P-values of the incorporated single SNP analyses
</p>
</td></tr>
<tr><td><code>p.MinP</code></td>
<td>

<p>P-value of the MinP test.
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>
<tr><td><code>E.df</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>
<tr><td><code>G.df</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>
</table>

<hr>
<h2 id='GEI.test'>
Test the interaction between an environemental exposure and a region/gene by a generalized score type test.
</h2><span id='topic+GEI.test'></span>

<h3>Description</h3>

<p>Once the preliminary work is done using &quot;GEI.prelim()&quot;, this function tests a specifc region/gene. Single SNP analyses are also incorporated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GEI.test(result.prelim,G,Gsub.id=NULL,G.method='wPCA',G.df=floor(sqrt(nrow(G))),
bootstrap=NULL,MinP.adjust=NULL,impute.method='fixed')
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GEI.test_+3A_result.prelim">result.prelim</code></td>
<td>

<p>The output of function &quot;GEI.prelim()&quot;
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_g">G</code></td>
<td>

<p>Genetic variants in the target region/gene, an m*q matrix where m is the subject ID and q is the total number of genetic variables. Note that the number of rows in Z should be same as the number of subjects.
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_g.method">G.method</code></td>
<td>

<p>The dimension reduction method for main effect adjustment of G. The following are permitted: &quot;wPCA&quot; for weighted principal component analysis; &quot;PCA&quot; for principal component analysis; &quot;R2&quot; for ordering the principal components by their R-squares. The dimension reduction method is in order to analyze large regions, i.e., the number of variants is close to or larger than the number of subjects. The default is &quot;wPCA&quot;.
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_g.df">G.df</code></td>
<td>

<p>Number of components selected by the dimension reduction method. The default is sqrt(m).
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_bootstrap">bootstrap</code></td>
<td>

<p>Whether to use bootstrap for small sample size adjustement. This is recommended when the number of subjects is small, or the set contains rare variants. The default is NULL, but a suggested number is 10000 when it is needed.
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_minp.adjust">MinP.adjust</code></td>
<td>

<p>If the users would like to compare with the MinP test, this parameter specify the adjustment thereshold as in Gao, et al. (2008) &quot;A multiple testing correction method for genetic association studies using correlated single nucleotide polymorphisms&quot;. Values from 0 to 1 are permitted. The default is NULL, i.e., no comparison. The value suggested by Gao, et al. (2008) is 0.95.
</p>
</td></tr>
<tr><td><code id="GEI.test_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>

<p>P-value of the set based generalized score type test.
</p>
</td></tr>
<tr><td><code>p.single</code></td>
<td>

<p>P-values of the incorporated single SNP analyses
</p>
</td></tr>
<tr><td><code>p.MinP</code></td>
<td>

<p>P-value of the MinP test.
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of heterozygous SNPs in the SNP set.
</p>
</td></tr>
<tr><td><code>E.df</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>
<tr><td><code>G.df</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## GEI.prelim does the preliminary data management.
# Input: Y, time, E, X (covariates)
## GEI.test tests a region.
# Input: G (genetic variants) and result of GEI.prelim

library(LGEWIS)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# E: environmental exposure, n by 1 matrix
# time: describe longitudinal structure, n by 2 matrix
# G: genotype matrix, m by q matrix where m is the total number of subjects

data(LGEWIS.example)
Y&lt;-LGEWIS.example$Y
X&lt;-LGEWIS.example$X
E&lt;-LGEWIS.example$E
time&lt;-LGEWIS.example$time
G&lt;-LGEWIS.example$G

# Preliminary data management
result.prelim&lt;-GEI.prelim(Y,time,E,X=X)

# test without the MinP test
result&lt;-GEI.test(result.prelim,G,MinP.adjust=NULL)

# test with the MinP test
result&lt;-GEI.test(result.prelim,G,MinP.adjust=0.95)

# test with the MinP test and the small sample adjustment
result&lt;-GEI.test(result.prelim,G,MinP.adjust=0.95,bootstrap=1000)
</code></pre>

<hr>
<h2 id='LGEWIS.example'>
Data example for LGEWIS (tests for genetic association or gene-environment interaction)
</h2><span id='topic+LGEWIS.example'></span>

<h3>Description</h3>

<p>Example data for LGEWIS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LGEWIS.example)</code></pre>


<h3>Format</h3>

<p>LGEWIS.example contains the following objects:
</p>

<dl>
<dt>G</dt><dd><p>a numeric genotype matrix of 10 individuals and 20 SNPs.
Each row represents a different individual, and each column represents a different SNP marker.</p>
</dd>
<dt>Y</dt><dd><p>a numeric vector of continuous phenotypes of 10 individuals with 4 repeated measurements.</p>
</dd>
<dt>time</dt><dd><p>a numeric matrix. The first column is the subject ID and the second column is the measured exam.</p>
</dd>
<dt>X</dt><dd><p>a numeric matrix of 1 covariates.</p>
</dd>
<dt>E</dt><dd><p>a numeric vector of environmental exposure.</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
