<!DOCTYPE html><html><head><title>Help for package LGRF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LGRF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#IBS_pseudo'>
<p>Generate IBS pseudo variables</p></a></li>
<li><a href='#LGRF.example'>
<p>Data example for LGRF</p></a></li>
<li><a href='#LGRF.SSD.All'>
<p>LGRF tests for multiple regions/genes using SSD format files</p></a></li>
<li><a href='#LGRF.SSD.OneSet_SetIndex'>
<p>LGRF tests for a single region/gene using SSD format files</p></a></li>
<li><a href='#null.LGRF'>
<p>Fit the null model for longitudinal genetic random field model</p></a></li>
<li><a href='#test.LGRF'>
<p>Test the association between an outcome variable and a region/gene by LGRF</p></a></li>
<li><a href='#test.MinP'>
<p>Test the association between an outcome variable and a region/gene by MinP</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Set-Based Tests for Genetic Association in Longitudinal Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-08-20</td>
</tr>
<tr>
<td>Author:</td>
<td>Zihuai He</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zihuai He &lt;zihuai@umich.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for the longitudinal genetic random field method (He et al., 2015, &lt;<a href="https://doi.org/10.1111%2Fbiom.12310">doi:10.1111/biom.12310</a>&gt;) to test the association between a longitudinally measured quantitative outcome and a set of genetic variants in a gene/region.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>CompQuadForm, SKAT, geepack</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-09-12 17:07:13 UTC; statzihuai</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-09-13 09:22:48</td>
</tr>
</table>
<hr>
<h2 id='IBS_pseudo'>
Generate IBS pseudo variables
</h2><span id='topic+IBS_pseudo'></span>

<h3>Description</h3>

<p>If users want to calculate the IBS similarity, this function creates the IBS pseudo variables. This is in order to calculate the IBS similarity in an efficient way.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IBS_pseudo(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IBS_pseudo_+3A_x">x</code></td>
<td>
<p>An n by q matrix of genetic variants.

</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns an n by 3p matrix of pseudo variables for efficiently calculating IBS similarity.





</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(LGRF)

# Load data example
# Z: genotype matrix, n by q matrix

data(LGRF.example)
Z&lt;-LGRF.example$Z
A&lt;-IBS_pseudo(Z)

# Then the IBS matrix can be calculated by K.IBS&lt;-AA^T.

</code></pre>

<hr>
<h2 id='LGRF.example'>
Data example for LGRF
</h2><span id='topic+LGRF.example'></span>

<h3>Description</h3>

<p>The dataset contains outcome variable Y, covariate X, time and genotype data Z. The first column in time is the subject ID and the second column is the measured exam. Y, X and time are all in long form. Z is a genotype matrix where each row corresponds to one subject.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LGRF.example)</code></pre>


<h3>Examples</h3>

<pre><code class='language-R'>data(LGRF.example)

</code></pre>

<hr>
<h2 id='LGRF.SSD.All'>
LGRF tests for multiple regions/genes using SSD format files
</h2><span id='topic+LGRF.SSD.All'></span>

<h3>Description</h3>

<p>Test the association between an outcome variable and multiple regions/genes using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LGRF.SSD.All(SSD.INFO, result.null, Gsub.id=NULL, interGXT=FALSE, similarity='GR', 
impute.method='fixed', MinP.compare=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LGRF.SSD.All_+3A_ssd.info">SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function &ldquo;Open_SSD&quot;. The sets are defined by this file.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_result.null">result.null</code></td>
<td>

<p>Output of function &ldquo;null.LGRF&quot;.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_intergxt">interGXT</code></td>
<td>

<p>Whether to incorperate the gene-time interaction effect. Incorperating this effect can improve power if there is any gene-time interaction, but has slight power loss otherwise. The default is FALSE. *Please note that the second column of time should be included as a covairate when interGXT is TRUE. 
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_similarity">similarity</code></td>
<td>

<p>Choose the similarity measurement for the genetic variants. Can be either &quot;GR&quot; for genetic relationship or &quot;IBS&quot; for identity by state. The default is &quot;GR&quot; for better computational efficiency.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_minp.compare">MinP.compare</code></td>
<td>

<p>Whether to compare with the GEE based minimum p-value (MinP) test. The default is FALSE. Please note that implementing the GEE based MinP test is time consuming.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.All_+3A_...">...</code></td>
<td>

<p>Other options of the GEE based MinP test. Defined same as in function &ldquo;test.MinP&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>results</code></td>
<td>

<p>First column contains the set ID; Second column contains the p-values; Third column contains the number of tested SNPs.
</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# * Since the Plink data files used here are hard to be included in a R package, 
# The usage is marked by "#" to pass the package check.

#library(LGRF)

##############################################

# Plink data files: File.Bed, File.Bim, File.Fam
# Files defining the sets: File.SetID, File.SSD, File.Info
# For longitudinal data, outcome and covariates are saved in a separate file: Y, time, X. 
# Null model was fitted using function null.LGRF.

# Create the MW File
# File.Bed&lt;-"./example.bed"
# File.Bim&lt;-"./example.bim"
# File.Fam&lt;-"./example.fam"
# File.SetID&lt;-"./example.SetID"
# File.SSD&lt;-"./example.SSD"
# File.Info&lt;-"./example.SSD.info"

# Generate SSD file
# To use binary ped files, you have to generate SSD file first.
# If you already have a SSD file, you do not need to call this function.
# Generate_SSD_SetID(File.Bed, File.Bim, File.Fam, File.SetID, File.SSD, File.Info)

# SSD.INFO&lt;-Open_SSD(File.SSD, File.Info)
# Number of samples
# SSD.INFO$nSample
# Number of Sets
# SSD.INFO$nSets

## Fit the null model
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# result.null&lt;-null.LGRF(Y,time,X=cbind(X,time[,2]))

# *Please note that the second column of time should be included as a covairate if
# the gene by time interaction effect will be incorperated.  

## Test all regions
# out_all&lt;-LGRF.SSD.All(SSD.INFO, result.null)

# Example result
# out.all$results
#      SetID   P.value N.Marker
# 1  GENE_01 0.6568851       94
# 2  GENE_02 0.1822183       84
# 3  GENE_03 0.3836986      108
# 4  GENE_04 0.1265337      101
# 5  GENE_05 0.3236089      103
# 6  GENE_06 0.9401741       94
# 7  GENE_07 0.1043820      104
# 8  GENE_08 0.6093275       96
# 9  GENE_09 0.6351147      100
# 10 GENE_10 0.5631549      100

## Test all regions, and compare with GEE based MinP test
# out_all&lt;-LGRF.SSD.All(SSD.INFO, result.null,MinP.compare=T)

# Example result
# out.all$results
#      SetID P.value P.value.MinP N.Marker
# 1  GENE_01 0.62842       1.0000       94
# 2  GENE_02 0.06558       0.2718       84
# 3  GENE_03 0.61795       1.0000      108
# 4  GENE_04 0.39667       0.7052      101
# 5  GENE_05 0.17371       0.5214      103
# 6  GENE_06 0.90104       1.0000       94
# 7  GENE_07 0.10143       0.1188      104
# 8  GENE_08 0.78082       0.3835       96
# 9  GENE_09 0.21966       0.5364      100
# 10 GENE_10 0.25468       0.3527      100

</code></pre>

<hr>
<h2 id='LGRF.SSD.OneSet_SetIndex'>
LGRF tests for a single region/gene using SSD format files
</h2><span id='topic+LGRF.SSD.OneSet_SetIndex'></span>

<h3>Description</h3>

<p>Test the association between an outcome variable and one region/gene using SSD format files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LGRF.SSD.OneSet_SetIndex(SSD.INFO, SetIndex, result.null, Gsub.id=NULL, interGXT=FALSE, 
similarity='GR', impute.method='fixed', MinP.compare=FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_ssd.info">SSD.INFO</code></td>
<td>

<p>SSD format information file, output of function &ldquo;Open_SSD&quot;. The sets are defined by this file.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_setindex">SetIndex</code></td>
<td>

<p>Set index. From 1 to the total number of sets.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_result.null">result.null</code></td>
<td>

<p>Output of function &ldquo;null.LGRF&quot;.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_intergxt">interGXT</code></td>
<td>

<p>Whether to incorperate the gene-time interaction effect. Incorperating this effect can improve power if there is any gene-time interaction, but has slight power loss otherwise. The default is FALSE. *Please note that the second column of time should be included as a covairate when interGXT is TRUE. 
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_similarity">similarity</code></td>
<td>

<p>Choose the similarity measurement for the genetic variants. Can be either &quot;GR&quot; for genetic relationship or &quot;IBS&quot; for identity by state. The default is &quot;GR&quot; for better computational efficiency.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_minp.compare">MinP.compare</code></td>
<td>

<p>Whether to compare with the GEE based minimum p-value (MinP) test. The default is FALSE. Please note that implementing the GEE based MinP test is time consuming.
</p>
</td></tr>
<tr><td><code id="LGRF.SSD.OneSet_SetIndex_+3A_...">...</code></td>
<td>

<p>Other options of the GEE based MinP test. Defined same as in function &ldquo;test.MinP&quot;.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>

<p>p-value of the LGRF test.
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>





</table>


<h3>Examples</h3>

<pre><code class='language-R'>
# * Since the Plink data files used here are hard to be included in a R package, 
# The usage is marked by "#" to pass the package check.

#library(LGRF)

##############################################

# Plink data files: File.Bed, File.Bim, File.Fam
# Files defining the sets: File.SetID, File.SSD, File.Info
# For longitudinal data, outcome and covariates are saved in a separate file: Y, time, X. 
# Null model was fitted using function null.LGRF.

# Create the MW File
# File.Bed&lt;-"./example.bed"
# File.Bim&lt;-"./example.bim"
# File.Fam&lt;-"./example.fam"
# File.SetID&lt;-"./example.SetID"
# File.SSD&lt;-"./example.SSD"
# File.Info&lt;-"./example.SSD.info"

# Generate SSD file
# To use binary ped files, you have to generate SSD file first.
# If you already have a SSD file, you do not need to call this function.
# Generate_SSD_SetID(File.Bed, File.Bim, File.Fam, File.SetID, File.SSD, File.Info)

# SSD.INFO&lt;-Open_SSD(File.SSD, File.Info)
# Number of samples
# SSD.INFO$nSample
# Number of Sets
# SSD.INFO$nSets

## Fit the null model
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# result.null&lt;-null.LGRF(Y,time,X=cbind(X,time[,2]))

# *Please note that the second column of time should be included as a covairate if
# the gene by time interaction effect will be incorperated. 

## Test a single region
# out_single&lt;-LGRF.SSD.OneSet_SetIndex(SSD.INFO=SSD.INFO, SetIndex=1, 
# result.null=result.null, MinP.compare=F)

# Example result
# $p.value
# [1] 0.6284

# $n.marker
# [1] 94

## Test a single region, and compare with GEE based MinP test
# out_single&lt;-LGRF.SSD.OneSet_SetIndex(SSD.INFO=SSD.INFO, SetIndex=1, 
# result.null=result.null,MinP.compare=T)

# $p.value
#       LGRF MinP
# [1,] 0.6284    1

# $n.marker
# [1] 94

</code></pre>

<hr>
<h2 id='null.LGRF'>
Fit the null model for longitudinal genetic random field model
</h2><span id='topic+null.LGRF'></span>

<h3>Description</h3>

<p>Before testing a specific region using a score test, this function fits the longitudinal genetic random field model under the null hypothesis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>null.LGRF(Y, time, X = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="null.LGRF_+3A_y">Y</code></td>
<td>

<p>The outcome variable, an n*1 matrix where n is the total number of observations
</p>
</td></tr>
<tr><td><code id="null.LGRF_+3A_time">time</code></td>
<td>

<p>An n*2 matrix describing how the observations are measured. The first column is the subject id. The second column is the measured exam (1,2,3,etc.). 
</p>
</td></tr>
<tr><td><code id="null.LGRF_+3A_x">X</code></td>
<td>

<p>An n*p covariates matrix where p is the total number of covariates. 
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list used for function test.LGRF().





</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(LGRF)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# Z: genotype matrix, m by q matrix where m is the total number of subjects

data(LGRF.example)
Y&lt;-LGRF.example$Y;time&lt;-LGRF.example$time;X&lt;-LGRF.example$X;Z&lt;-LGRF.example$Z

# Fit the null model
result.null&lt;-null.LGRF(Y,time,X=cbind(X,time[,2]))

# *Please note that the second column of time should be included as a covairate if
# the gene by time interaction effect will be incorperated.  
</code></pre>

<hr>
<h2 id='test.LGRF'>
Test the association between an outcome variable and a region/gene by LGRF
</h2><span id='topic+test.LGRF'></span>

<h3>Description</h3>

<p>Once the model under the null model is fitted using &quot;null.LGRF()&quot;, this function tests a specifc region/gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test.LGRF(Z, result.null, Gsub.id=NULL, interGXT = FALSE, similarity = "GR",
impute.method="fixed")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test.LGRF_+3A_z">Z</code></td>
<td>

<p>Genetic variants in the target region/gene, an m*q matrix where m is the subject ID and q is the total number of genetic variables. Note that the number of rows in Z should be same as the number of subjects.
</p>
</td></tr>
<tr><td><code id="test.LGRF_+3A_result.null">result.null</code></td>
<td>

<p>The output of function &quot;null.LGRF()&quot;
</p>
</td></tr>
<tr><td><code id="test.LGRF_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="test.LGRF_+3A_intergxt">interGXT</code></td>
<td>

<p>Whether to incorperate the gene-time interaction effect. Incorperating this effect can improve power if there is any gene-time interaction, but has slight power loss otherwise. The default is FALSE. *Please note that the second column of time should be included as a covairate when interGXT is TRUE. 
</p>
</td></tr>
<tr><td><code id="test.LGRF_+3A_similarity">similarity</code></td>
<td>

<p>Choose the similarity measurement for the genetic variants. Can be either &quot;GR&quot; for genetic relationship or &quot;IBS&quot; for identity by state. The default is &quot;GR&quot; for better computational efficiency.
</p>
</td></tr>
<tr><td><code id="test.LGRF_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>

<p>p-value of the LGRF test.
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>





</table>


<h3>Examples</h3>

<pre><code class='language-R'>## null.LGRF fits the null model. 
# Input: Y, time, X (covariates)
## test.LGRF tests a region and give p-value. 
# Input: Z (genetic variants) and result of null.longGRF

library(LGRF)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# Z: genotype matrix, m by q matrix where m is the total number of subjects

data(LGRF.example)
Y&lt;-LGRF.example$Y;time&lt;-LGRF.example$time;X&lt;-LGRF.example$X;Z&lt;-LGRF.example$Z

# Fit the null model
result.null&lt;-null.LGRF(Y,time,X=cbind(X,time[,2]))

# *Please note that the second column of time should be included as a covairate if
# the gene by time interaction effect will be incorperated.  

# The LGRF-G test
pLGRF_G&lt;-test.LGRF(Z,result.null)

# The LGRF-GT test
pLGRF_GT&lt;-test.LGRF(Z,result.null,interGXT=TRUE)

# The LGRF-G test using the IBS similarity
pLGRF_G_IBS&lt;-test.LGRF(Z,result.null,similarity="IBS")

# The LGRF-GT test, main effect is modeled using the IBS similarity
pLGRF_GT_IBS&lt;-test.LGRF(Z,result.null,interGXT=TRUE,similarity="IBS")
</code></pre>

<hr>
<h2 id='test.MinP'>
Test the association between an outcome variable and a region/gene by MinP
</h2><span id='topic+test.MinP'></span>

<h3>Description</h3>

<p>If users want to compare LGRF with the minimum p-value (MinP) test, this function tests a specifc region/gene by a GEE based minimum p-value test after fitting &quot;null.LGRF()&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test.MinP(Z, result.null, Gsub.id=NULL, corstr="exchangeable", MinP.adjust=0.95, 
impute.method="fixed")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="test.MinP_+3A_z">Z</code></td>
<td>

<p>Genetic variants in the target region/gene, an m*q matrix where m is the subject ID and q is the total number of genetic variables. Note that the number of rows in Z should be same as the number of subject.
</p>
</td></tr>
<tr><td><code id="test.MinP_+3A_result.null">result.null</code></td>
<td>

<p>The output of function &quot;null.LGRF()&quot;.
</p>
</td></tr>
<tr><td><code id="test.MinP_+3A_gsub.id">Gsub.id</code></td>
<td>

<p>The subject id corresponding to the genotype matrix, an m dimensional vector. This is in order to match the phenotype and genotype matrix. The default is NULL, where the order is assumed to be matched with Y, X and time.
</p>
</td></tr>
<tr><td><code id="test.MinP_+3A_corstr">corstr</code></td>
<td>

<p>The working correlation as specified in 'geeglm'. The following are permitted: '&quot;independence&quot;', '&quot;exchangeable&quot;', '&quot;ar1&quot;', '&quot;unstructured&quot;' and '&quot;userdefined&quot;.
</p>
</td></tr>
<tr><td><code id="test.MinP_+3A_minp.adjust">MinP.adjust</code></td>
<td>

<p>The minimum p-value is adjusted by the number of independent tests. Choose the adjustment thereshold as specified in Gao, et al. (2008) &quot;A multiple testing correction method for genetic association studies using correlated single nucleotide polymorphisms&quot;. Values from 0 to 1 are permitted.
</p>
</td></tr>
<tr><td><code id="test.MinP_+3A_impute.method">impute.method</code></td>
<td>

<p>Choose the imputation method when there is missing genotype. Can be &quot;random&quot;, &quot;fixed&quot; or &quot;bestguess&quot;. Given the estimated allele frequency, &quot;random&quot; simulates the genotype from binomial distribution; &quot;fixed&quot; uses the genotype expectation; &quot;Best guess&quot; uses the genotype with highest probability.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>p.value</code></td>
<td>

<p>p-value of the MinP test.
</p>
</td></tr>
<tr><td><code>n.marker</code></td>
<td>

<p>number of tested SNPs in the SNP set.
</p>
</td></tr>





</table>


<h3>Examples</h3>

<pre><code class='language-R'>## null.LGRF fits the null model. 
# Input: Y, time, X (covariates)
## test.MinP tests a region and give p-value. 
# Input: Z (genetic variants) and result of null.longGRF

library(LGRF)

# Load data example
# Y: outcomes, n by 1 matrix where n is the total number of observations
# X: covariates, n by p matrix
# time: describe longitudinal structure, n by 2 matrix
# Z: genotype matrix, m by q matrix where m is the total number of subjects

data(LGRF.example)
Y&lt;-LGRF.example$Y;time&lt;-LGRF.example$time;X&lt;-LGRF.example$X;Z&lt;-LGRF.example$Z

# Fit the null model
result.null&lt;-null.LGRF(Y,time,X=X)

# The minimum p-value test based on GEE
pMinP&lt;-test.MinP(Z,result.null,corstr="exchangeable",MinP.adjust=0.95)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
