<!DOCTYPE html><html><head><title>Help for package LifemapR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LifemapR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#+.lifemap_obj'><p>Add a graphical element to a tree visualisation.</p></a></li>
<li><a href='#add_lm_branches'><p>Compute the aesthetics for a subtree visualisation.</p></a></li>
<li><a href='#add_lm_markers'><p>Compute the aesthetics for markers visualisation.</p></a></li>
<li><a href='#add_lm_piecharts'><p>Compute the aesthetics for discret values visualisation.</p></a></li>
<li><a href='#build_Lifemap'><p>A function to construct a LifemapR object, usable by the other functions of the package.</p></a></li>
<li><a href='#create_matrix'><p>Create a dataframe for the ancestry.</p></a></li>
<li><a href='#create_value_range'><p>Compute a new scale for a value</p></a></li>
<li><a href='#display_map'><p>Create a Lifemap base.</p></a></li>
<li><a href='#display_option'><p>Compute the different display options.</p></a></li>
<li><a href='#draw_Lifemap'><p>Represent data on a Lifemap basemap.</p></a></li>
<li><a href='#eukaryotes_1000'><p>NCBI information for 1000 eukaryotes</p></a></li>
<li><a href='#eukaryotes_80'><p>NCBI information for 80 eukaryotes</p></a></li>
<li><a href='#gen_res'><p>Genomic results</p></a></li>
<li><a href='#is.lifemap_obj'><p>Reports whether x is a lifemap_obj object.</p></a></li>
<li><a href='#is.lm_branches'><p>Reports whether x is a lm_branches object.</p></a></li>
<li><a href='#is.lm_markers'><p>Reports whether x is a lm_markers object.</p></a></li>
<li><a href='#is.lm_piecharts'><p>Reports whether x is a lm_branches object.</p></a></li>
<li><a href='#kraken_res'><p>Kraken results</p></a></li>
<li><a href='#lifemap'><p>Initialise a new Lifemap visualisation.</p></a></li>
<li><a href='#lm_branches'><p>Add a &quot;branches&quot; layer to a lifemap_obj object.</p></a></li>
<li><a href='#LM_eukaryotes'><p>Transformation in a LifemapR format of NCBI information for 1000 eukaryotes</p></a></li>
<li><a href='#lm_markers'><p>add a &quot;markers&quot; layer to a lifemap_obj object.</p></a></li>
<li><a href='#lm_piecharts'><p>Add a &quot;piecharts&quot; layer to a lifemap_obj object.</p></a></li>
<li><a href='#make_newick'><p>Create a newick for the given dataset.</p></a></li>
<li><a href='#pass_infos'><p>Infer numerical values to nodes.</p></a></li>
<li><a href='#pass_infos_discret'><p>Infer discret values to nodes for lm_piecharts function.</p></a></li>
<li><a href='#print.lifemap_obj'><p>Method to print lifemap_obj objects.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Data Visualisation on 'Lifemap' Tree</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>leaflet, shiny, jsonlite, dplyr, tidyr, leaflet.minicharts,
htmltools, rlang, RCurl, fastmatch</td>
</tr>
<tr>
<td>Description:</td>
<td>Allow to visualise data on the NCBI phylogenetic tree as presented in Lifemap '<a href="http://lifemap.univ-lyon1.fr/">http://lifemap.univ-lyon1.fr/</a>'. It takes as input a dataframe with at least a "taxid" column containing NCBI format TaxIds and allows to draw multiple layers with different visualisation tools.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-20 15:26:42 UTC; siberchicot</td>
</tr>
<tr>
<td>Author:</td>
<td>Cassandra Bompard [aut],
  Damien M. de Vienne [aut],
  Aurélie Siberchicot [ctb, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aurélie Siberchicot &lt;aurelie.siberchicot@univ-lyon1.fr&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-20 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+2B.lifemap_obj'>Add a graphical element to a tree visualisation.</h2><span id='topic++2B.lifemap_obj'></span>

<h3>Description</h3>

<p>Add a graphical element to a tree visualisation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lifemap_obj'
e1 + e2
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B2B.lifemap_obj_+3A_e1">e1</code></td>
<td>
<p>An object of class lifemap_obj that contains at least $df, a dataframe, and $basemap, the map used to get the coordinates.</p>
</td></tr>
<tr><td><code id="+2B2B.lifemap_obj_+3A_e2">e2</code></td>
<td>
<p>A description of the graphical features wanted for a set of points (eg. markers, subtree, piecharts, ...).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A lifemap_obj object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Only run examples in interactive R sessions
if (interactive()) {
data(LM_eukaryotes)
LM_obj &lt;- lifemap(LM_eukaryotes) + lm_markers() + lm_branches()
}

</code></pre>

<hr>
<h2 id='add_lm_branches'>Compute the aesthetics for a subtree visualisation.</h2><span id='topic+add_lm_branches'></span>

<h3>Description</h3>

<p>Compute the aesthetics for a subtree visualisation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_lm_branches(
  proxy,
  aes,
  df,
  df_visible,
  df_descendants,
  group_info,
  all_taxids
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_lm_branches_+3A_proxy">proxy</code></td>
<td>
<p>The map to be modified.</p>
</td></tr>
<tr><td><code id="add_lm_branches_+3A_aes">aes</code></td>
<td>
<p>The dataframe containing the aesthetics details (must be of lm_branches class).</p>
</td></tr>
<tr><td><code id="add_lm_branches_+3A_df">df</code></td>
<td>
<p>The full dataframe.</p>
</td></tr>
<tr><td><code id="add_lm_branches_+3A_df_visible">df_visible</code></td>
<td>
<p>The dataframe containing visible taxa.</p>
</td></tr>
<tr><td><code id="add_lm_branches_+3A_df_descendants">df_descendants</code></td>
<td>
<p>The dataframe containing all the information on the descendants of visible taxa.</p>
</td></tr>
<tr><td><code id="add_lm_branches_+3A_group_info">group_info</code></td>
<td>
<p>the ID of this group of lines.</p>
</td></tr>
<tr><td><code id="add_lm_branches_+3A_all_taxids">all_taxids</code></td>
<td>
<p>A vector containing all the visible taxids and their direct descendants.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated map with the new layer added.
</p>

<hr>
<h2 id='add_lm_markers'>Compute the aesthetics for markers visualisation.</h2><span id='topic+add_lm_markers'></span>

<h3>Description</h3>

<p>Compute the aesthetics for markers visualisation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_lm_markers(proxy, aes, df, df_visible, group_info)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_lm_markers_+3A_proxy">proxy</code></td>
<td>
<p>The map to be modified.</p>
</td></tr>
<tr><td><code id="add_lm_markers_+3A_aes">aes</code></td>
<td>
<p>The dataframe containing the aesthetics information (must be of lm_markers class).</p>
</td></tr>
<tr><td><code id="add_lm_markers_+3A_df">df</code></td>
<td>
<p>The full dataframe.</p>
</td></tr>
<tr><td><code id="add_lm_markers_+3A_df_visible">df_visible</code></td>
<td>
<p>The dataframe containing visible taxa.</p>
</td></tr>
<tr><td><code id="add_lm_markers_+3A_group_info">group_info</code></td>
<td>
<p>The ID of this group of markers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated map with the new layer added.
</p>

<hr>
<h2 id='add_lm_piecharts'>Compute the aesthetics for discret values visualisation.</h2><span id='topic+add_lm_piecharts'></span>

<h3>Description</h3>

<p>Compute the aesthetics for discret values visualisation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_lm_piecharts(proxy, aes, df, df_visible, layer)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_lm_piecharts_+3A_proxy">proxy</code></td>
<td>
<p>The map to be modified.</p>
</td></tr>
<tr><td><code id="add_lm_piecharts_+3A_aes">aes</code></td>
<td>
<p>The dataframe containing the aesthetics details (must be of lm_piecharts class).</p>
</td></tr>
<tr><td><code id="add_lm_piecharts_+3A_df">df</code></td>
<td>
<p>The full dataframe.</p>
</td></tr>
<tr><td><code id="add_lm_piecharts_+3A_df_visible">df_visible</code></td>
<td>
<p>The dataframe containing visible taxa.</p>
</td></tr>
<tr><td><code id="add_lm_piecharts_+3A_layer">layer</code></td>
<td>
<p>The ID of this group of charts</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated map with the new layer added.
</p>

<hr>
<h2 id='build_Lifemap'>A function to construct a LifemapR object, usable by the other functions of the package.</h2><span id='topic+build_Lifemap'></span>

<h3>Description</h3>

<p>A function to construct a LifemapR object, usable by the other functions of the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_Lifemap(df, basemap = c("ncbi", "base", "fr", "virus"), verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_Lifemap_+3A_df">df</code></td>
<td>
<p>A dataframe containing at least one column named &quot;taxid&quot; that contains NCBI Taxonomy Identifiers (taxid).
The dataframe can contain any number of additional columns defining traits/characters/values associated to
each taxid.</p>
</td></tr>
<tr><td><code id="build_Lifemap_+3A_basemap">basemap</code></td>
<td>
<p>The chosen basemap for Lifemap (&quot;fr&quot;, &quot;ncbi&quot;, &quot;base&quot; or &quot;virus&quot;).</p>
</td></tr>
<tr><td><code id="build_Lifemap_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE (the default), the function will print detailed information to the console.
If FALSE, it will run silently.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of class lifemap_obj containing:
</p>

<ul>
<li><p> df : a dataframe containing at least for each taxid :
</p>

<ul>
<li><p> The x coordinate (lon)
</p>
</li>
<li><p> The y coordonate (lat)
</p>
</li>
<li><p> The scientific name (sci_name)
</p>
</li>
<li><p> The zoom level at which the taxa is visible (zoom)
</p>
</li>
<li><p> A list of its ascendants (ascend)
</p>
</li>
<li><p> Its type (&quot;requested&quot; or &quot;ancestor&quot;)
</p>
</li>
<li><p> Its direct ancestor
</p>
</li>
<li><p> Its type (type), i.e. whether the taxid was
requested by the user (&quot;requested&quot;) or if it is the anecestor of a requested taxid (&quot;ancestor&quot;)
</p>
</li></ul>

</li>
<li><p> basemap : the basemap used to get taxa's details
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data(eukaryotes_80)
## Not run: 
# make sure you have a good internet connection to load these very large files
LM &lt;- build_Lifemap(eukaryotes_80, "fr")

## End(Not run)
</code></pre>

<hr>
<h2 id='create_matrix'>Create a dataframe for the ancestry.</h2><span id='topic+create_matrix'></span>

<h3>Description</h3>

<p>Create a dataframe with pairs of taxids, each taxid and their corresponding values are associated with every one of its ancestors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_matrix(df, cols)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_matrix_+3A_df">df</code></td>
<td>
<p>A dataframe containing taxids and values.</p>
</td></tr>
<tr><td><code id="create_matrix_+3A_cols">cols</code></td>
<td>
<p>The columns containing the values which needs to be inferred.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LM_eukaryotes)

create_matrix(LM_eukaryotes$df, c("GC.", "Genes"))
</code></pre>

<hr>
<h2 id='create_value_range'>Compute a new scale for a value</h2><span id='topic+create_value_range'></span>

<h3>Description</h3>

<p>Compute a new scale for a value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_value_range(value, df, df2, min, max)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_value_range_+3A_value">value</code></td>
<td>
<p>A vector of values.</p>
</td></tr>
<tr><td><code id="create_value_range_+3A_df">df</code></td>
<td>
<p>The full dataframe.</p>
</td></tr>
<tr><td><code id="create_value_range_+3A_df2">df2</code></td>
<td>
<p>The dataframe containing visibles taxa.</p>
</td></tr>
<tr><td><code id="create_value_range_+3A_min">min</code></td>
<td>
<p>The new minimum of the range.</p>
</td></tr>
<tr><td><code id="create_value_range_+3A_max">max</code></td>
<td>
<p>The new maximum of the range.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of values.
</p>

<hr>
<h2 id='display_map'>Create a Lifemap base.</h2><span id='topic+display_map'></span>

<h3>Description</h3>

<p>This function create a blank Leaflet map using the leaflet function from the leaflet package.
If a dataframe is provided, it will be used for the creation of the map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>display_map(df = NULL, basemap = c("fr", "ncbi", "base", "virus"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="display_map_+3A_df">df</code></td>
<td>
<p>A dataframe. If given, its columns can be easily accessed with &quot;~&quot; (eg. ~GC.).</p>
</td></tr>
<tr><td><code id="display_map_+3A_basemap">basemap</code></td>
<td>
<p>The basemap choosen to be displayed, it can be either :
</p>

<ul>
<li><p> 'fr' for <a href="https://lifemap-fr.univ-lyon1.fr/">https://lifemap-fr.univ-lyon1.fr/</a>
</p>
</li>
<li><p> 'ncbi' for <a href="https://lifemap-ncbi.univ-lyon1.fr/">https://lifemap-ncbi.univ-lyon1.fr/</a>
</p>
</li>
<li><p> 'base' for <a href="https://lifemap.univ-lyon1.fr/">https://lifemap.univ-lyon1.fr/</a>
</p>
</li>
<li><p> 'virus' for <a href="https://virusmap.univ-lyon1.fr/">https://virusmap.univ-lyon1.fr/</a>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>An HTML widget object  with graphics layers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>display_map()
</code></pre>

<hr>
<h2 id='display_option'>Compute the different display options.</h2><span id='topic+display_option'></span>

<h3>Description</h3>

<p>Compute the different display options.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>display_option(m, aes, df, type, leaves, i)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="display_option_+3A_m">m</code></td>
<td>
<p>The map to be modified.</p>
</td></tr>
<tr><td><code id="display_option_+3A_aes">aes</code></td>
<td>
<p>The dataframe containing the aesthetics details</p>
</td></tr>
<tr><td><code id="display_option_+3A_df">df</code></td>
<td>
<p>The full dataframe.</p>
</td></tr>
<tr><td><code id="display_option_+3A_type">type</code></td>
<td>
<p>A string indicating the type of representation, either &quot;markers&quot; or &quot;discret&quot;</p>
</td></tr>
<tr><td><code id="display_option_+3A_leaves">leaves</code></td>
<td>
<p>The Vector of all the terminal taxids.</p>
</td></tr>
<tr><td><code id="display_option_+3A_i">i</code></td>
<td>
<p>The index of the aesthetics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An updated map.
</p>

<hr>
<h2 id='draw_Lifemap'>Represent data on a Lifemap basemap.</h2><span id='topic+draw_Lifemap'></span>

<h3>Description</h3>

<p>Draw a map and all the aesthetics in the order you put them in, the last one will be on top of the others.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>draw_Lifemap(lm_obj)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="draw_Lifemap_+3A_lm_obj">lm_obj</code></td>
<td>
<p>A Lifemap object filled with aesthetics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A shiny application
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Only run examples in interactive R sessions
if (interactive()) {
data(LM_eukaryotes)
lifemap(LM_eukaryotes) + lm_markers() + lm_branches()
}
</code></pre>

<hr>
<h2 id='eukaryotes_1000'>NCBI information for 1000 eukaryotes</h2><span id='topic+eukaryotes_1000'></span>

<h3>Description</h3>

<p>A dataset containing NCBI information of 1000 eukaryotes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eukaryotes_1000)
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows and 19 variables:
</p>

<dl>
<dt>X.Organism.Name</dt><dd><p>Organism name at the species level</p>
</dd>
<dt>taxid</dt><dd><p>NCBI taxid</p>
</dd>
<dt>BioProject.Accession</dt><dd><p>BioProject Accession number (from BioProject database)</p>
</dd>
<dt>BioProject.ID</dt><dd><p>BioProject ID</p>
</dd>
<dt>Group</dt><dd><p>Commonly used organism groups:  Animals, Fungi, Plants, Protists</p>
</dd>
<dt>SubGroup</dt><dd><p>NCBI Taxonomy level below group: Mammals, Birds, Fishes, Flatworms, Insects, Amphibians, Reptiles, Roundworms, Ascomycetes, Basidiomycetes, Land Plants, Green Algae, Apicomplexans, Kinetoplasts</p>
</dd>
<dt>Size..Mb.</dt><dd><p>Total length of DNA submitted for the project</p>
</dd>
<dt>GC.</dt><dd><p>Percent of nitrogenous bases (guanine or cytosine) in DNA submitted for the project</p>
</dd>
<dt>Assembly.Accession</dt><dd><p>Name of the genome assembly (from NCBI Assembly database)</p>
</dd>
<dt>Replicons</dt><dd><p>Number of replicons in the assembly</p>
</dd>
<dt>WGS</dt><dd><p>Four-letter Accession prefix followed by version as defined in WGS division of GenBank/INSDC</p>
</dd>
<dt>Scaffolds</dt><dd><p>Number of scaffolds in the assembly</p>
</dd>
<dt>Genes</dt><dd><p>Number of Genes annotated in the assembly</p>
</dd>
<dt>Proteins</dt><dd><p>Number of Proteins annotated in the assembly</p>
</dd>
<dt>Release.Date</dt><dd><p>First public sequence release for the project</p>
</dd>
<dt>Modify.Date</dt><dd><p>Sequence modification date for the project</p>
</dd>
<dt>Status</dt><dd><p>Highest level of assembly: <br> Chromosomes: one or more chromosomes are assembled<br> Scaffolds or contigs: sequence assembled but no chromosomes</p>
</dd>
<dt>Center</dt><dd><p>Origin of the sample</p>
</dd>
<dt>BioSample.Accession</dt><dd><p>BioSample Accession number</p>
</dd>
</dl>


<hr>
<h2 id='eukaryotes_80'>NCBI information for 80 eukaryotes</h2><span id='topic+eukaryotes_80'></span>

<h3>Description</h3>

<p>A dataset containing NCBI information of 80 eukaryotes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(eukaryotes_80)
</code></pre>


<h3>Format</h3>

<p>A data frame with 80 rows and 19 variables:
</p>

<dl>
<dt>X.Organism.Name</dt><dd><p>Organism name at the species level</p>
</dd>
<dt>taxid</dt><dd><p>NCBI taxid</p>
</dd>
<dt>BioProject.Accession</dt><dd><p>BioProject Accession number (from BioProject database)</p>
</dd>
<dt>BioProject.ID</dt><dd><p>BioProject ID</p>
</dd>
<dt>Group</dt><dd><p>Commonly used organism groups:  Animals, Fungi, Plants, Protists</p>
</dd>
<dt>SubGroup</dt><dd><p>NCBI Taxonomy level below group: Mammals, Birds, Fishes, Flatworms, Insects, Amphibians, Reptiles, Roundworms, Ascomycetes, Basidiomycetes, Land Plants, Green Algae, Apicomplexans, Kinetoplasts</p>
</dd>
<dt>Size..Mb.</dt><dd><p>Total length of DNA submitted for the project</p>
</dd>
<dt>GC.</dt><dd><p>Percent of nitrogenous bases (guanine or cytosine) in DNA submitted for the project</p>
</dd>
<dt>Assembly.Accession</dt><dd><p>Name of the genome assembly (from NCBI Assembly database)</p>
</dd>
<dt>Replicons</dt><dd><p>Number of replicons in the assembly</p>
</dd>
<dt>WGS</dt><dd><p>Four-letter Accession prefix followed by version as defined in WGS division of GenBank/INSDC</p>
</dd>
<dt>Scaffolds</dt><dd><p>Number of scaffolds in the assembly</p>
</dd>
<dt>Genes</dt><dd><p>Number of Genes annotated in the assembly</p>
</dd>
<dt>Proteins</dt><dd><p>Number of Proteins annotated in the assembly</p>
</dd>
<dt>Release.Date</dt><dd><p>First public sequence release for the project</p>
</dd>
<dt>Modify.Date</dt><dd><p>Sequence modification date for the project</p>
</dd>
<dt>Status</dt><dd><p>Highest level of assembly: <br> Chromosomes: one or more chromosomes are assembled<br> Scaffolds or contigs: sequence assembled but no chromosomes</p>
</dd>
<dt>Center</dt><dd><p>Origin of the sample</p>
</dd>
<dt>BioSample.Accession</dt><dd><p>BioSample Accession number</p>
</dd>
</dl>


<hr>
<h2 id='gen_res'>Genomic results</h2><span id='topic+gen_res'></span>

<h3>Description</h3>

<p>A dataset containing information on Genome size and TE content for 808 taxids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gen_res)
</code></pre>


<h3>Format</h3>

<p>A data frame with 808 rows and 3 variables:
</p>

<dl>
<dt>taxid</dt><dd><p>NCBI taxid</p>
</dd>
<dt>Genome_size</dt><dd><p>the Genome size in pb</p>
</dd>
<dt>TEcontent_bp</dt><dd><p>the transposable element content in pb</p>
</dd>
</dl>


<hr>
<h2 id='is.lifemap_obj'>Reports whether x is a lifemap_obj object.</h2><span id='topic+is.lifemap_obj'></span>

<h3>Description</h3>

<p>Reports whether x is a lifemap_obj object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.lifemap_obj(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.lifemap_obj_+3A_x">x</code></td>
<td>
<p>The object to test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean indicating whether or not the object is of lifemap_obj type.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Only run examples in interactive R sessions
if (interactive()) {
data(LM_eukaryotes)
is.lifemap_obj(LM_eukaryotes)
}

</code></pre>

<hr>
<h2 id='is.lm_branches'>Reports whether x is a lm_branches object.</h2><span id='topic+is.lm_branches'></span>

<h3>Description</h3>

<p>Reports whether x is a lm_branches object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.lm_branches(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.lm_branches_+3A_x">x</code></td>
<td>
<p>The object to test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean indicating whether or not the object is of lm_branches type.
</p>

<hr>
<h2 id='is.lm_markers'>Reports whether x is a lm_markers object.</h2><span id='topic+is.lm_markers'></span>

<h3>Description</h3>

<p>Reports whether x is a lm_markers object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.lm_markers(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.lm_markers_+3A_x">x</code></td>
<td>
<p>The object to test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean indicating whether or not the object is of lm_markers type.
</p>

<hr>
<h2 id='is.lm_piecharts'>Reports whether x is a lm_branches object.</h2><span id='topic+is.lm_piecharts'></span>

<h3>Description</h3>

<p>Reports whether x is a lm_branches object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is.lm_piecharts(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is.lm_piecharts_+3A_x">x</code></td>
<td>
<p>The object to test.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A boolean indicating whether or not the object is of lm_piecharts type.
</p>

<hr>
<h2 id='kraken_res'>Kraken results</h2><span id='topic+kraken_res'></span>

<h3>Description</h3>

<p>A dataset containing NCBI information of 1000 eukaryotes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kraken_res)
</code></pre>


<h3>Format</h3>

<p>A data frame with 4427 rows and 6 variables:
</p>

<dl>
<dt>coverage_percent</dt><dd><p>Percentage of fragments covered by the clade rooted at this taxon</p>
</dd>
<dt>coverage_number</dt><dd><p>Number of fragments covered by the clade rooted at this taxon</p>
</dd>
<dt>fragment_number</dt><dd><p>Number of fragments assigned directly to this taxon</p>
</dd>
<dt>rank</dt><dd><p>A rank code, indicating (U)nclassified, (R)oot, (D)omain, (K)ingdom, (P)hylum, (C)lass, (O)rder, (F)amily, (G)enus, or (S)pecies. Taxa that are not at any of these 10 ranks have a rank code that is formed by using the rank code of the closest ancestor rank with a number indicating the distance from that rank. E.g., &quot;G2&quot; is a rank code indicating a taxon is between genus and species and the grandparent taxon is at the genus rank.</p>
</dd>
<dt>taxid</dt><dd><p>NCBI taxonomic ID number</p>
</dd>
<dt>name</dt><dd><p>Indented scientific name</p>
</dd>
</dl>


<hr>
<h2 id='lifemap'>Initialise a new Lifemap visualisation.</h2><span id='topic+lifemap'></span>

<h3>Description</h3>

<p>Initialise a lifemap_obj object.
It can be used to describe aesthetics like the zoom level at which taxids becomes visible...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lifemap(lm_obj, zoom = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lifemap_+3A_lm_obj">lm_obj</code></td>
<td>
<p>lifemap_obj object used for data visualisation.</p>
</td></tr>
<tr><td><code id="lifemap_+3A_zoom">zoom</code></td>
<td>
<p>The level of zoom for which values are displayed (if zoom = 0 only the nodes at the current zoom level or lower can have their values displayed).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A lifemap_obj object.
</p>

<hr>
<h2 id='lm_branches'>Add a &quot;branches&quot; layer to a lifemap_obj object.</h2><span id='topic+lm_branches'></span>

<h3>Description</h3>

<p>Add a &quot;branches&quot; layer to a lifemap_obj object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lm_branches(
  data = NULL,
  color = NULL,
  var_color = NULL,
  size = 5,
  min = 2,
  max = 20,
  opacity = 0.5,
  FUN = NULL,
  legend = TRUE,
  legendPosition = c("topright", "bottomright", "bottomleft", "topleft")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lm_branches_+3A_data">data</code></td>
<td>
<p>A sub dataset to use, if NULL then all of the taxids from the lifemap object given to lifemap() will be used.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_color">color</code></td>
<td>
<p>Either a color for the branches or a palette if a variable is used to represent branches' color.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_var_color">var_color</code></td>
<td>
<p>A column name of the original dataframe to represent this variable by the color of branches.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_size">size</code></td>
<td>
<p>Either a numeric for the branche's thickness or a variable to be represented by the branche's thickness.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_min">min</code></td>
<td>
<p>An integer indicating the minimal thickness of the branches if the size is a column name.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_max">max</code></td>
<td>
<p>An integer indicating the maximal thickness of the branches if the size is a column name.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_opacity">opacity</code></td>
<td>
<p>An integer indicating branche's opacity.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_fun">FUN</code></td>
<td>
<p>The function to be applied to infer values. If NULL values won't be inferred</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_legend">legend</code></td>
<td>
<p>A logical indicating whether or not to display the legend.</p>
</td></tr>
<tr><td><code id="lm_branches_+3A_legendposition">legendPosition</code></td>
<td>
<p>c(&quot;topright&quot;, &quot;bottomright&quot;, &quot;bottomleft&quot;, &quot;topleft&quot;), the position of the legend.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An lm_branches object containing all aesthetics details for one layer of branches
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LM_eukaryotes)

lm_branches(var_color = "GC.", color = "Accent")

lm_branches(data = LM_eukaryotes$df[LM_eukaryotes$df$Group %in% "Plants",])

</code></pre>

<hr>
<h2 id='LM_eukaryotes'>Transformation in a LifemapR format of NCBI information for 1000 eukaryotes</h2><span id='topic+LM_eukaryotes'></span>

<h3>Description</h3>

<p>A dataset containing NCBI information of 1000 eukaryotes. The variables are as follows:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(LM_eukaryotes)
</code></pre>


<h3>Format</h3>

<p>A lifemap object - a list containing the basemap used to fetch data and df, a data frame with 2760 rows and 26 variables:
</p>

<dl>
<dt>X.Organism.Name</dt><dd><p>Organism name at the species level</p>
</dd>
<dt>taxid</dt><dd><p>NCBI taxid</p>
</dd>
<dt>BioProject.Accession</dt><dd><p>BioProject Accession number (from BioProject database)</p>
</dd>
<dt>BioProject.ID</dt><dd><p>BioProject ID</p>
</dd>
<dt>Group</dt><dd><p>Commonly used organism groups:  Animals, Fungi, Plants, Protists</p>
</dd>
<dt>SubGroup</dt><dd><p>NCBI Taxonomy level below group: Mammals, Birds, Fishes, Flatworms, Insects, Amphibians, Reptiles, Roundworms, Ascomycetes, Basidiomycetes, Land Plants, Green Algae, Apicomplexans, Kinetoplasts</p>
</dd>
<dt>Size..Mb.</dt><dd><p>Total length of DNA submitted for the project</p>
</dd>
<dt>GC.</dt><dd><p>Percent of nitrogenous bases (guanine or cytosine) in DNA submitted for the project</p>
</dd>
<dt>Assembly.Accession</dt><dd><p>Name of the genome assembly (from NCBI Assembly database)</p>
</dd>
<dt>Replicons</dt><dd><p>Number of replicons in the assembly</p>
</dd>
<dt>WGS</dt><dd><p>Four-letter Accession prefix followed by version as defined in WGS division of GenBank/INSDC</p>
</dd>
<dt>Scaffolds</dt><dd><p>Number of scaffolds in the assembly</p>
</dd>
<dt>Genes</dt><dd><p>Number of Genes annotated in the assembly</p>
</dd>
<dt>Proteins</dt><dd><p>Number of Proteins annotated in the assembly</p>
</dd>
<dt>Release.Date</dt><dd><p>First public sequence release for the project</p>
</dd>
<dt>Modify.Date</dt><dd><p>Sequence modification date for the project</p>
</dd>
<dt>Status</dt><dd><p>Highest level of assembly: <br> Chromosomes: one or more chromosomes are assembled<br> Scaffolds or contigs: sequence assembled but no chromosomes</p>
</dd>
<dt>Center</dt><dd><p>Origin of the sample</p>
</dd>
<dt>BioSample.Accession</dt><dd><p>BioSample Accession number</p>
</dd>
<dt>lon</dt><dd><p>longitude of taxids on a specific basemap</p>
</dd>
<dt>lat</dt><dd><p>latitude of taxids on a specific basemap</p>
</dd>
<dt>sci_name</dt><dd><p>scientific name of taxids</p>
</dd>
<dt>zoom</dt><dd><p>zoom of taxids on a specific basemap</p>
</dd>
<dt>ascend</dt><dd><p>the list of all ancestors of taxids on a specific basemap</p>
</dd>
<dt>type</dt><dd><p>either &quot;requested&quot; if the taxid was given, &quot;ancestor&quot; if gotten from the database</p>
</dd>
<dt>ancestor</dt><dd><p>the direct ancestor oftaxids on a specific basemap</p>
</dd>
</dl>


<hr>
<h2 id='lm_markers'>add a &quot;markers&quot; layer to a lifemap_obj object.</h2><span id='topic+lm_markers'></span>

<h3>Description</h3>

<p>Adds a layer with circles that can represent data. The main parameters that can be used to represent data are the following:
</p>

<ul>
<li><p> radius
</p>
</li>
<li><p> var_fillColor
</p>
</li>
<li><p> var_color (stroke's color)
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>lm_markers(
  data = NULL,
  radius = 10,
  min = 10,
  max = 40,
  fillColor = NULL,
  var_fillColor = NULL,
  fillOpacity = 0.8,
  stroke = FALSE,
  color = NULL,
  var_color = NULL,
  weight = 1,
  opacity = 0.8,
  legend = TRUE,
  legendPosition = c("topright", "bottomright", "bottomleft", "topleft"),
  legendOrientation = c("vertical", "horizontal"),
  legendOpacity = 0.5,
  FUN = NULL,
  display = c("auto", "requested", "all", "leaves"),
  popup = NULL,
  label = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lm_markers_+3A_data">data</code></td>
<td>
<p>A sub dataset to use, if NULL then all of the taxids from the lifemap object given to lifemap() will be used.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_radius">radius</code></td>
<td>
<p>Either a numerical value or a column name of the original dataframe to represent this variable by the size of markers.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_min">min</code></td>
<td>
<p>An integer indicating the minimal size of the markers if radius is a column name.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_max">max</code></td>
<td>
<p>An integer indicating the maximal size of the markers if radius is a column name.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_fillcolor">fillColor</code></td>
<td>
<p>Either a color or a palette if the fillColor is represented by a variable.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_var_fillcolor">var_fillColor</code></td>
<td>
<p>A column name of the original dataframe to represent this variable by the fillColor of markers.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_fillopacity">fillOpacity</code></td>
<td>
<p>A numeric indicating the fill opacity.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_stroke">stroke</code></td>
<td>
<p>A logical indicating whether to draw a border for markers.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_color">color</code></td>
<td>
<p>The stroke color. Either a color or a palette if the stroke color is represented by a variable.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_var_color">var_color</code></td>
<td>
<p>A column name of the original dataframe to represent this variable by the stroke color.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_weight">weight</code></td>
<td>
<p>The stroke width in pixels.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_opacity">opacity</code></td>
<td>
<p>The stroke opacity.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_legend">legend</code></td>
<td>
<p>Whether to display the legend or not.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_legendposition">legendPosition</code></td>
<td>
<p>c(&quot;topright&quot;, &quot;bottomright&quot;, &quot;bottomleft&quot;, &quot;topleft&quot;).</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_legendorientation">legendOrientation</code></td>
<td>
<p>c(&quot;vertical&quot;, &quot;horizontal&quot;).</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_legendopacity">legendOpacity</code></td>
<td>
<p>Legend opacity (applies on the shapes in the legend, not the background itself).</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_fun">FUN</code></td>
<td>
<p>The function to be applied to infer values. If NULL then the information missing from the parent nodes won't be inferred.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_display">display</code></td>
<td>
<p>c(&quot;auto&quot;, &quot;requested&quot;, &quot;all&quot;, &quot;leaves&quot;), a string indicating how to display markers :
</p>

<ul>
<li><p> &quot;auto&quot; : the markers are displayed depending on the zoom level, by default, allow to have a lot of markers
</p>
</li>
<li><p> &quot;requested&quot; : only displays the requested taxa, but all at the same time
</p>
</li>
<li><p> &quot;all&quot; : displays all the taxa including all the ancestors to the root
</p>
</li>
<li><p> &quot;leaves&quot; : displays only the latest (most recent) taxa
</p>
</li></ul>

<p>(WARNING : &quot;requested&quot;, &quot;leaves&quot; and &quot;auto&quot; shouldn't be used to display more than 2000 markers as it may result in long computing time).</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_popup">popup</code></td>
<td>
<p>A column name indicating what to display when clicking on a node.</p>
</td></tr>
<tr><td><code id="lm_markers_+3A_label">label</code></td>
<td>
<p>A column name indicating what to display when hovering on a node.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An lm_markers object containing all aesthetics details for one layer of markers.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LM_eukaryotes)

lm_markers(data = LM_eukaryotes$df[LM_eukaryotes$df$Group %in% "Plants", ])

lm_markers(radius = "GC.", var_fillColor = "Genes")

</code></pre>

<hr>
<h2 id='lm_piecharts'>Add a &quot;piecharts&quot; layer to a lifemap_obj object.</h2><span id='topic+lm_piecharts'></span>

<h3>Description</h3>

<p>Add a &quot;piecharts&quot; layer to a lifemap_obj object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lm_piecharts(
  data = NULL,
  param,
  type = c("pie", "bar", "polar-area", "polar-radius", "auto"),
  width = 30,
  height = 30,
  opacity = 1,
  showLabels = FALSE,
  pal = "Accent",
  legend = TRUE,
  legendPosition = c("topright", "bottomright", "bottomleft", "topleft"),
  display = c("auto", "requested", "all", "leaves")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lm_piecharts_+3A_data">data</code></td>
<td>
<p>A sub dataset to use, if NULL then all of the taxids from the lifemap object given to lifemap() will be used.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_param">param</code></td>
<td>
<p>A column name indicating the discret variable to be represented.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_type">type</code></td>
<td>
<p>The type of chart to draw.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_width">width</code></td>
<td>
<p>The maximal width of the charts.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_height">height</code></td>
<td>
<p>The maximal height of the charts.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_opacity">opacity</code></td>
<td>
<p>The chart's opacity.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_showlabels">showLabels</code></td>
<td>
<p>A boolean indicating whether to display the values directly on the chart or not.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_pal">pal</code></td>
<td>
<p>The palette to be used for the charts.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_legend">legend</code></td>
<td>
<p>A boolean indiacting whether to draw the legend or not.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_legendposition">legendPosition</code></td>
<td>
<p>c(&quot;topright&quot;, &quot;bottomright&quot;, &quot;bottomleft&quot;, &quot;topleft&quot;). Where should the legend be placed.</p>
</td></tr>
<tr><td><code id="lm_piecharts_+3A_display">display</code></td>
<td>
<p>c(&quot;auto&quot;, &quot;requested&quot;, &quot;all&quot;, &quot;leaves&quot;), a string indicating how to display charts :
</p>

<ul>
<li><p> &quot;auto&quot; : the markers are displayed depending on the zoom level, by default, allow to have a lot of charts
</p>
</li>
<li><p> &quot;requested&quot; : only displays the requested taxa, but all at the same time
</p>
</li>
<li><p> &quot;all&quot; : displays all the taxa including all the ancestors to the root
</p>
</li>
<li><p> &quot;leaves&quot; : displays only the latest (most recent) taxa
</p>
</li></ul>

<p>(WARNING : &quot;requested&quot;, &quot;leaves&quot; and &quot;auto&quot; shouldn't be used to display more than 2000 charts as it may result in long computing time)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An lm_piecharts object containing all aesthetics details for one layer of charts
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LM_eukaryotes)

lm_piecharts(param = "Status")

lm_piecharts(data = LM_eukaryotes$df[LM_eukaryotes$df$Group %in% "Plants", ], param = "Status")

</code></pre>

<hr>
<h2 id='make_newick'>Create a newick for the given dataset.</h2><span id='topic+make_newick'></span>

<h3>Description</h3>

<p>Create a newick for the given dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>make_newick(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="make_newick_+3A_df">df</code></td>
<td>
<p>A dataframe either enriched with by the build_Lifemap function or containing at least two columns :
</p>

<ul>
<li><p> &quot;taxid&quot; containing NCBI format taxids
</p>
</li>
<li><p> &quot;ancestor&quot; containing for each taxid, it's direct ancestor in the phylogeny
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A character string.
</p>

<hr>
<h2 id='pass_infos'>Infer numerical values to nodes.</h2><span id='topic+pass_infos'></span>

<h3>Description</h3>

<p>Infer numerical values to nodes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pass_infos(M, FUN, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pass_infos_+3A_m">M</code></td>
<td>
<p>The dataframe returned by create_matrix.</p>
</td></tr>
<tr><td><code id="pass_infos_+3A_fun">FUN</code></td>
<td>
<p>The function to be applied when inferring the values.</p>
</td></tr>
<tr><td><code id="pass_infos_+3A_value">value</code></td>
<td>
<p>The column name to which the function applies.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An array of values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LM_eukaryotes)

infos &lt;- create_matrix(LM_eukaryotes$df, c("GC.", "Genes"))

inferred_values &lt;- pass_infos(M = infos, FUN = mean, value = "GC.")
</code></pre>

<hr>
<h2 id='pass_infos_discret'>Infer discret values to nodes for lm_piecharts function.</h2><span id='topic+pass_infos_discret'></span>

<h3>Description</h3>

<p>Infer discret values to nodes for lm_piecharts function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pass_infos_discret(M, value)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pass_infos_discret_+3A_m">M</code></td>
<td>
<p>The dataframe returned by create_matrix.</p>
</td></tr>
<tr><td><code id="pass_infos_discret_+3A_value">value</code></td>
<td>
<p>The column name represented as piecharts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe containing the TaxIDs and as many columns as there are distinct values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(LM_eukaryotes)

infos &lt;- create_matrix(LM_eukaryotes$df, "Status")

inferred_values &lt;- pass_infos_discret(M = infos, value = "Status")
</code></pre>

<hr>
<h2 id='print.lifemap_obj'>Method to print lifemap_obj objects.</h2><span id='topic+print.lifemap_obj'></span>

<h3>Description</h3>

<p>Method to print lifemap_obj objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lifemap_obj'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.lifemap_obj_+3A_x">x</code></td>
<td>
<p>An lifemap_obj.</p>
</td></tr>
<tr><td><code id="print.lifemap_obj_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a description of the dataframe and basemap used for the lm_obj object, or a shiny application if aesthetics are furnished
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Only run examples in interactive R sessions
if (interactive()) {
data(LM_eukaryotes)
print(LM_eukaryotes)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
