<!DOCTYPE html><html><head><title>Help for package limorhyde2</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {limorhyde2}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getDiffRhythmStats'><p>Compute differential rhythm statistics from fitted models</p></a></li>
<li><a href='#getExpectedMeas'><p>Compute expected measurements from fitted models</p></a></li>
<li><a href='#getExpectedMeasIntervals'><p>Compute credible intervals for expected measurements</p></a></li>
<li><a href='#getModelFit'><p>Fit linear models for rhythmicity in one or more conditions</p></a></li>
<li><a href='#getPosteriorFit'><p>Compute posterior fit for linear models for rhythmicity</p></a></li>
<li><a href='#getPosteriorSamples'><p>Draw samples from posterior distributions of fitted models</p></a></li>
<li><a href='#getRhythmStats'><p>Compute rhythm statistics from fitted models</p></a></li>
<li><a href='#getStatsIntervals'><p>Compute credible intervals for rhythm or differential rhythm statistics</p></a></li>
<li><a href='#GSE34018'><p>Gene expression data for GSE34018</p></a></li>
<li><a href='#GSE54650'><p>Gene expression data for GSE54650</p></a></li>
<li><a href='#mergeMeasMeta'><p>Merge measurements and metadata</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quantify Rhythmicity and Differential Rhythmicity in Genomic
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Fit linear models based on periodic splines, moderate model
  coefficients using multivariate adaptive shrinkage, then compute properties of
  the moderated curves.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://limorhyde2.hugheylab.org">https://limorhyde2.hugheylab.org</a>,
<a href="https://github.com/hugheylab/limorhyde2">https://github.com/hugheylab/limorhyde2</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6)</td>
</tr>
<tr>
<td>Imports:</td>
<td>abind (&ge; 1.4-5), ashr (&ge; 2.2-54), checkmate (&ge; 2.0.0),
data.table (&ge; 1.12.8), DESeq2 (&ge; 1.30.0), foreach (&ge; 1.5.0),
HDInterval (&ge; 0.2.2), iterators (&ge; 1.0.12), limma (&ge;
3.42.2), mashr (&ge; 0.2.50), pbs (&ge; 1.1), zeallot (&ge; 0.1.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>cowplot (&ge; 1.1.1), knitr, doParallel (&ge; 1.0.15), ggplot2
(&ge; 3.3.5), glue (&ge; 1.6.1), qs (&ge; 0.24.1), rmarkdown,
testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-26 20:15:15 UTC; jakehughey</td>
</tr>
<tr>
<td>Author:</td>
<td>Jake Hughey [aut, cre],
  Dora Obodo [aut],
  Elliot Outland [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jake Hughey &lt;jakejhughey@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-26 20:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='getDiffRhythmStats'>Compute differential rhythm statistics from fitted models</h2><span id='topic+getDiffRhythmStats'></span>

<h3>Description</h3>

<p>This function computes differences in rhythmicity between fitted curves for a
given pair of conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDiffRhythmStats(fit, rhyStats, conds = fit$conds, dopar = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDiffRhythmStats_+3A_fit">fit</code></td>
<td>
<p>A <code>limorhyde2</code> object containing data from multiple conditions.</p>
</td></tr>
<tr><td><code id="getDiffRhythmStats_+3A_rhystats">rhyStats</code></td>
<td>
<p>A <code>data.table</code> of rhythmic statistics, as returned by
<code><a href="#topic+getRhythmStats">getRhythmStats()</a></code>, for fitted models in <code>fit</code>.</p>
</td></tr>
<tr><td><code id="getDiffRhythmStats_+3A_conds">conds</code></td>
<td>
<p>A character vector indicating the conditions to compare
pairwise, by default all conditions in <code>fit</code>.</p>
</td></tr>
<tr><td><code id="getDiffRhythmStats_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to run calculations in parallel if
a parallel backend is already set up, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>. Recommended to minimize runtime.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code> containing the following differential rhythm
statistics:
</p>

<ul>
<li> <p><code>mean_mesor</code>
</p>
</li>
<li> <p><code>mean_peak_trough_amp</code>
</p>
</li>
<li> <p><code>mean_rms_amp</code> (only calculated if <code>rms</code> to <code><a href="#topic+getRhythmStats">getRhythmStats()</a></code> was <code>TRUE</code>)
</p>
</li>
<li> <p><code>diff_mesor</code>
</p>
</li>
<li> <p><code>diff_peak_trough_amp</code>
</p>
</li>
<li> <p><code>diff_rms_amp</code> (only calculated if <code>rms</code> to <code><a href="#topic+getRhythmStats">getRhythmStats()</a></code> was <code>TRUE</code>)
</p>
</li>
<li> <p><code>diff_peak_phase</code>: circular difference between <code>-fit$period/2</code> and
<code>fit$period/2</code>
</p>
</li>
<li> <p><code>diff_trough_phase</code>: circular difference between <code>-fit$period/2</code> and
<code>fit$period/2</code>
</p>
</li>
<li> <p><code>diff_rhy_dist</code>: Euclidean distance between polar coordinates
(<code>peak_trough_amp</code>, <code>peak_phase</code>)
</p>
</li>
<li> <p><code>rms_diff_rhy</code>: root mean square difference in mean-centered fitted curves
(only calculated if <code>rms</code> to <code><a href="#topic+getRhythmStats">getRhythmStats()</a></code> was <code>TRUE</code>)
</p>
</li></ul>

<p>The stats will be based on the value for <code>cond2</code> minus the value for <code>cond1</code>.
The rows of the <code>data.table</code> depend on the 'fitType' attribute of <code>rhyStats</code>:
</p>

<ul>
<li><p> 'fitType' is 'posterior_mean' or 'raw': one row per feature per pair of
conditions.
</p>
</li>
<li><p> 'fitType' is 'posterior_samples': one row per feature per posterior sample
per pair of conditions.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getRhythmStats">getRhythmStats()</a></code>, <code><a href="#topic+getStatsIntervals">getStatsIntervals()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# rhythmicity in one condition
y = GSE54650$y
metadata = GSE54650$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '13869'))

# rhythmicity and differential rhythmicity in multiple conditions
y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata, nKnots = 3L, condColname = 'cond')
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '12686'))
diffRhyStats = getDiffRhythmStats(fit, rhyStats)
</code></pre>

<hr>
<h2 id='getExpectedMeas'>Compute expected measurements from fitted models</h2><span id='topic+getExpectedMeas'></span>

<h3>Description</h3>

<p>This function computes expected measurements (corresponding to the fitted
curves) for the specified times and features in all combinations of
conditions and covariates (if they exist).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExpectedMeas(
  fit,
  times,
  fitType = c("posterior_mean", "posterior_samples", "raw"),
  features = NULL,
  dopar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExpectedMeas_+3A_fit">fit</code></td>
<td>
<p>A 'limorhyde2' object.</p>
</td></tr>
<tr><td><code id="getExpectedMeas_+3A_times">times</code></td>
<td>
<p>Numeric vector of times, in units of
<code>fit$metadata[[fit$timeColname]]</code>.</p>
</td></tr>
<tr><td><code id="getExpectedMeas_+3A_fittype">fitType</code></td>
<td>
<p>String indicating which fitted models to use to compute the
expected measurements. A typical analysis using <code>limorhyde2</code> will be based
on 'posterior_mean', the default.</p>
</td></tr>
<tr><td><code id="getExpectedMeas_+3A_features">features</code></td>
<td>
<p>Vector of names, row numbers, or logical values for
subsetting the features. <code>NULL</code> indicates all features.</p>
</td></tr>
<tr><td><code id="getExpectedMeas_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to run calculations in parallel if
a parallel backend is already set up, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>. Recommended to minimize runtime.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getModelFit">getModelFit()</a></code>, <code><a href="#topic+getPosteriorFit">getPosteriorFit()</a></code>, <code><a href="#topic+getPosteriorSamples">getPosteriorSamples()</a></code>,
<code><a href="#topic+getExpectedMeasIntervals">getExpectedMeasIntervals()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)

measObs = mergeMeasMeta(y, metadata, features = c('13170', '12686'))
measFitMean = getExpectedMeas(
  fit, times = seq(0, 24, 0.5), features = c('13170', '12686'))
</code></pre>

<hr>
<h2 id='getExpectedMeasIntervals'>Compute credible intervals for expected measurements</h2><span id='topic+getExpectedMeasIntervals'></span>

<h3>Description</h3>

<p>This functions uses posterior samples to quantify uncertainty in the
expected measurements from fitted models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExpectedMeasIntervals(expectedMeas, mass = 0.9, method = c("eti", "hdi"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExpectedMeasIntervals_+3A_expectedmeas">expectedMeas</code></td>
<td>
<p>A <code>data.table</code> of expected measurements for posterior
samples, as returned by <code><a href="#topic+getExpectedMeas">getExpectedMeas()</a></code>.</p>
</td></tr>
<tr><td><code id="getExpectedMeasIntervals_+3A_mass">mass</code></td>
<td>
<p>Number between 0 and 1 indicating the probability mass for which
to calculate the intervals.</p>
</td></tr>
<tr><td><code id="getExpectedMeasIntervals_+3A_method">method</code></td>
<td>
<p>String indicating the type of interval: 'eti' for equal-tailed
using <code><a href="stats.html#topic+quantile">stats::quantile()</a></code>, or 'hdi' for highest density using
<code><a href="HDInterval.html#topic+hdi">HDInterval::hdi()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code> containing lower and upper bounds of the expected
measurement for each combination of feature, time, and possibly condition
and covariate.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getExpectedMeas">getExpectedMeas()</a></code>, <code><a href="#topic+getStatsIntervals">getStatsIntervals()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
fit = getPosteriorSamples(fit, nPosteriorSamples = 10L)

measFitSamps = getExpectedMeas(
  fit, times = seq(0, 24, 0.5), fitType = 'posterior_samples',
  features = c('13170', '12686'))
measFitInts = getExpectedMeasIntervals(measFitSamps)
</code></pre>

<hr>
<h2 id='getModelFit'>Fit linear models for rhythmicity in one or more conditions</h2><span id='topic+getModelFit'></span>

<h3>Description</h3>

<p>This is the first step in an analysis using <code>limorhyde2</code>, the second is to
moderate the fits using <code><a href="#topic+getPosteriorFit">getPosteriorFit()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getModelFit(
  y,
  metadata,
  period = 24,
  nKnots = 3L,
  degree = if (nKnots &gt; 2) 3L else 2L,
  sinusoid = FALSE,
  timeColname = "time",
  condColname = NULL,
  covarColnames = NULL,
  sampleColname = "sample",
  nShifts = 3L,
  method = c("trend", "voom", "deseq2"),
  lmFitArgs = list(),
  eBayesArgs = if (method == "trend") list(trend = TRUE) else list(),
  DESeqArgs = list(),
  keepLmFits = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getModelFit_+3A_y">y</code></td>
<td>
<p>Matrix-like object of measurements, with rows corresponding to
features and columns to samples.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_metadata">metadata</code></td>
<td>
<p>data.frame containing experimental design information for
each sample. Rows of <code>metadata</code> must correspond to columns of <code>y</code>. Row
names are ignored.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_period">period</code></td>
<td>
<p>Number specifying the period for the time variable, in the same
units as the values in the <code>timeColname</code> column.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_nknots">nKnots</code></td>
<td>
<p>Number of internal knots for the periodic spline for the time
variable.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_degree">degree</code></td>
<td>
<p>Integer indicating degree of the piecewise polynomial for the
spline.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_sinusoid">sinusoid</code></td>
<td>
<p>Logical indicating whether to fit a cosinor-based model
instead of a spline-based model.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_timecolname">timeColname</code></td>
<td>
<p>String indicating the column in <code>metadata</code> containing the
time at which each sample was acquired.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_condcolname">condColname</code></td>
<td>
<p>String indicating the column in <code>metadata</code> containing the
condition in which each sample was acquired. <code>NULL</code> indicates all samples
came from the same condition. If not <code>NULL</code>, the model will include main
effects and interactions with the terms for time.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_covarcolnames">covarColnames</code></td>
<td>
<p>Character vector indicating the columns in <code>metadata</code>
containing covariates to include in the model. <code>NULL</code> indicates no
covariates.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_samplecolname">sampleColname</code></td>
<td>
<p>String indicating the column in <code>metadata</code> containing
the name of each sample, which must correspond to the column names of <code>y</code>.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_nshifts">nShifts</code></td>
<td>
<p>Number of shifted models to fit. Only used for periodic
splines, not for cosinor. Do not change from the default unless you know
what you're doing.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_method">method</code></td>
<td>
<p>String indicating method to estimate model coefficients. For
microarray data, use 'trend'. For RNA-seq count data, use 'voom' or
'deseq2'.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_lmfitargs">lmFitArgs</code></td>
<td>
<p>List of arguments passed to <code><a href="limma.html#topic+lmFit">limma::lmFit()</a></code>.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_ebayesargs">eBayesArgs</code></td>
<td>
<p>List of arguments passed to <code><a href="limma.html#topic+ebayes">limma::eBayes()</a></code>.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_deseqargs">DESeqArgs</code></td>
<td>
<p>List of arguments passed to <code><a href="DESeq2.html#topic+DESeq">DESeq2::DESeq()</a></code>.</p>
</td></tr>
<tr><td><code id="getModelFit_+3A_keeplmfits">keepLmFits</code></td>
<td>
<p>Logical indicating whether to keep the complete fit objects
from <code>limma</code> or <code>DESeq2</code>. Not needed by any functions in <code>limorhyde2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>limorhyde2</code> object with elements:
</p>

<ul>
<li> <p><code>metadata</code>: As supplied above, converted to a <code>data.table</code>.
</p>
</li>
<li> <p><code>timeColname</code>: As supplied above.
</p>
</li>
<li> <p><code>condColname</code>: As supplied above.
</p>
</li>
<li> <p><code>covarColnames</code>: As supplied above.
</p>
</li>
<li> <p><code>coefficients</code>: Matrix with rows corresponding to features and columns to
model terms, including all shifted models.
</p>
</li>
<li> <p><code>shifts</code>: Numeric vector indicating amount by which timepoints were shifted
for each shifted model.
</p>
</li>
<li> <p><code>period</code>: As supplied above.
</p>
</li>
<li> <p><code>conds</code>: If <code>condColname</code> is not <code>NULL</code>, a vector of unique values of
the condition variable.
</p>
</li>
<li> <p><code>nKnots</code>: Number of knots.
</p>
</li>
<li> <p><code>degree</code>: As supplied above.
</p>
</li>
<li> <p><code>sinusoid</code>: As supplied above.
</p>
</li>
<li> <p><code>nConds</code>: Number of conditions.
</p>
</li>
<li> <p><code>nCovs</code>: Number of covariates.
</p>
</li>
<li> <p><code>lmFits</code>: If <code>keepLmFits</code> is <code>TRUE</code>, a list of objects from <code>limma</code> or
<code>DESeq2</code>, with length equal to length of the <code>shifts</code> element.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getPosteriorFit">getPosteriorFit()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# rhythmicity in one condition
y = GSE54650$y
metadata = GSE54650$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '13869'))

# rhythmicity and differential rhythmicity in multiple conditions
y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata, nKnots = 3L, condColname = 'cond')
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '12686'))
diffRhyStats = getDiffRhythmStats(fit, rhyStats)
</code></pre>

<hr>
<h2 id='getPosteriorFit'>Compute posterior fit for linear models for rhythmicity</h2><span id='topic+getPosteriorFit'></span>

<h3>Description</h3>

<p>This is the second step in an analysis using <code>limorhyde2</code>, the first is to
fit linear models using <code><a href="#topic+getModelFit">getModelFit()</a></code>. This function obtains posterior
estimates of coefficients using multivariate adaptive shrinkage (mash), which
learns patterns in the data and accounts for noise in the original fits. The
defaults for arguments should work well in most cases, so only change them if
you know what you're doing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPosteriorFit(
  fit,
  covMethod = c("data-driven", "canonical", "both"),
  getSigResArgs = list(),
  npc = fit$nKnots,
  covEdArgs = list(),
  overwrite = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPosteriorFit_+3A_fit">fit</code></td>
<td>
<p>A <code>limorhyde2</code> object.</p>
</td></tr>
<tr><td><code id="getPosteriorFit_+3A_covmethod">covMethod</code></td>
<td>
<p>String indicating the type(s) of covariance matrices to use
for the mash fit.</p>
</td></tr>
<tr><td><code id="getPosteriorFit_+3A_getsigresargs">getSigResArgs</code></td>
<td>
<p>List of arguments passed to
<code><a href="mashr.html#topic+get_significant_results">mashr::get_significant_results()</a></code>. Only used if <code>covMethod</code> is
'data-driven' or 'both'.</p>
</td></tr>
<tr><td><code id="getPosteriorFit_+3A_npc">npc</code></td>
<td>
<p>Number of principal components passed to <code><a href="mashr.html#topic+cov_pca">mashr::cov_pca()</a></code>. Only
used if <code>covMethod</code> is 'data-driven' or 'both'.</p>
</td></tr>
<tr><td><code id="getPosteriorFit_+3A_covedargs">covEdArgs</code></td>
<td>
<p>List of arguments passed to <code><a href="mashr.html#topic+cov_ed">mashr::cov_ed()</a></code>. Only used if
<code>covMethod</code> is 'data-driven' or 'both'.</p>
</td></tr>
<tr><td><code id="getPosteriorFit_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical for whether to recompute the mash fit if it already
exists.</p>
</td></tr>
<tr><td><code id="getPosteriorFit_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="mashr.html#topic+mash">mashr::mash()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>limorhyde2</code> object containing everything in <code>fit</code> with added or
updated elements:
</p>

<ul>
<li> <p><code>mashData</code>: list of <code>mash</code> data objects
</p>
</li>
<li> <p><code>mashFits</code>: list of <code>mash</code> fit objects
</p>
</li>
<li> <p><code>mashCoefficients</code>: Matrix of posterior mean coefficients, with rows
corresponding to features and columns to model terms.
</p>
</li>
<li> <p><code>mashIdx</code>: Vector indicating which model terms were included in the mash
fit.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getModelFit">getModelFit()</a></code>, <code><a href="#topic+getRhythmStats">getRhythmStats()</a></code>, <code><a href="#topic+getExpectedMeas">getExpectedMeas()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# rhythmicity in one condition
y = GSE54650$y
metadata = GSE54650$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '13869'))

# rhythmicity and differential rhythmicity in multiple conditions
y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata, nKnots = 3L, condColname = 'cond')
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '12686'))
diffRhyStats = getDiffRhythmStats(fit, rhyStats)
</code></pre>

<hr>
<h2 id='getPosteriorSamples'>Draw samples from posterior distributions of fitted models</h2><span id='topic+getPosteriorSamples'></span>

<h3>Description</h3>

<p>This is an optional step in an analysis using <code>limorhyde2</code>, and is useful for
quantifying uncertainty in posterior estimates of fitted curves and rhythmic
statistics. The function calls <code><a href="mashr.html#topic+mash_compute_posterior_matrices">mashr::mash_compute_posterior_matrices()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getPosteriorSamples(fit, nPosteriorSamples = 200L, overwrite = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getPosteriorSamples_+3A_fit">fit</code></td>
<td>
<p>A &lsquo;limorhyde2&rsquo; object containing posterior fits.</p>
</td></tr>
<tr><td><code id="getPosteriorSamples_+3A_nposteriorsamples">nPosteriorSamples</code></td>
<td>
<p>Number of samples to draw from each posterior
distribution.</p>
</td></tr>
<tr><td><code id="getPosteriorSamples_+3A_overwrite">overwrite</code></td>
<td>
<p>Logical indicating whether to recompute posterior samples if
they already exist.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>limorhyde2</code> object containing everything in <code>fit</code> with added or
updated element:
</p>

<ul>
<li> <p><code>mashPosteriorSamples</code>: a three-dimensional array of coefficients, with dim
1 corresponding to features, dim 2 to model terms, and dim 3 to posterior
samples.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getPosteriorFit">getPosteriorFit()</a></code>, <code><a href="#topic+getRhythmStats">getRhythmStats()</a></code>, <code><a href="#topic+getExpectedMeas">getExpectedMeas()</a></code>,
<code><a href="#topic+getStatsIntervals">getStatsIntervals()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

y = GSE54650$y
metadata = GSE54650$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
fit = getPosteriorSamples(fit, nPosteriorSamples = 10L)

rhyStatsSamps = getRhythmStats(
  fit, features = c('13170', '13869'), fitType = 'posterior_samples')
rhyStatsInts = getStatsIntervals(rhyStatsSamps)
</code></pre>

<hr>
<h2 id='getRhythmStats'>Compute rhythm statistics from fitted models</h2><span id='topic+getRhythmStats'></span>

<h3>Description</h3>

<p>This function uses <code><a href="stats.html#topic+optim">stats::optim()</a></code> to compute various properties of
fitted curves with respect to time, potentially in each condition and for
each posterior sample, and adjusting for any covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getRhythmStats(
  fit,
  fitType = c("posterior_mean", "posterior_samples", "raw"),
  features = NULL,
  dopar = TRUE,
  rms = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getRhythmStats_+3A_fit">fit</code></td>
<td>
<p>A <code>limorhyde2</code> object.</p>
</td></tr>
<tr><td><code id="getRhythmStats_+3A_fittype">fitType</code></td>
<td>
<p>String indicating which fitted models to use to compute the
rhythmic statistics. A typical analysis using <code>limorhyde2</code> will be based on
'posterior_mean', the default.</p>
</td></tr>
<tr><td><code id="getRhythmStats_+3A_features">features</code></td>
<td>
<p>Vector of names, row numbers, or logical values for
subsetting the features. <code>NULL</code> indicates all features.</p>
</td></tr>
<tr><td><code id="getRhythmStats_+3A_dopar">dopar</code></td>
<td>
<p>Logical indicating whether to run calculations in parallel if
a parallel backend is already set up, e.g., using
<code><a href="doParallel.html#topic+registerDoParallel">doParallel::registerDoParallel()</a></code>. Recommended to minimize runtime.</p>
</td></tr>
<tr><td><code id="getRhythmStats_+3A_rms">rms</code></td>
<td>
<p>Logical indicating whether to calculate <code>rms_amp</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code> containing the following rhythm statistics:
</p>

<ul>
<li> <p><code>peak_phase</code>: time between 0 and <code>fit$period</code> at which the peak or maximum
value occurs
</p>
</li>
<li> <p><code>peak_value</code>
</p>
</li>
<li> <p><code>trough_phase</code>: time between 0 and <code>fit$period</code> at which the trough or
minimum value occurs
</p>
</li>
<li> <p><code>trough_value</code>
</p>
</li>
<li> <p><code>peak_trough_amp</code>: <code>peak_value - trough_value</code>
</p>
</li>
<li> <p><code>rms_amp</code>: root mean square difference between fitted curve and mean value
between time 0 and <code>fit$period</code> (only calculated if <code>rms</code> is <code>TRUE</code>)
</p>
</li>
<li> <p><code>mesor</code>: mean value between time 0 and <code>fit$period</code>
</p>
</li></ul>

<p>The rows of the <code>data.table</code> depend on the <code>fit</code> object and <code>fitType</code>:
</p>

<ul>
<li> <p><code>fit</code> contains data from one condition and <code>fitType</code> is posterior_mean' or
'raw': one row per feature.
</p>
</li>
<li> <p><code>fit</code> contains data from one condition and <code>fitType</code> is
'posterior_samples': one row per feature per posterior sample.
</p>
</li>
<li> <p><code>fit</code> contains data from multiple conditions and <code>fitType</code> is
'posterior_mean' or 'raw': one row per feature per condition.
</p>
</li>
<li> <p><code>fit</code> contains data from multiple conditions and <code>fitType</code> is
'posterior_samples': one row per feature per condition per posterior
sample.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+getModelFit">getModelFit()</a></code>, <code><a href="#topic+getPosteriorFit">getPosteriorFit()</a></code>,  <code><a href="#topic+getPosteriorSamples">getPosteriorSamples()</a></code>,
<code><a href="#topic+getDiffRhythmStats">getDiffRhythmStats()</a></code>, <code><a href="#topic+getStatsIntervals">getStatsIntervals()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

# rhythmicity in one condition
y = GSE54650$y
metadata = GSE54650$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '13869'))

# rhythmicity and differential rhythmicity in multiple conditions
y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata, nKnots = 3L, condColname = 'cond')
fit = getPosteriorFit(fit)
rhyStats = getRhythmStats(fit, features = c('13170', '12686'))
diffRhyStats = getDiffRhythmStats(fit, rhyStats)
</code></pre>

<hr>
<h2 id='getStatsIntervals'>Compute credible intervals for rhythm or differential rhythm statistics</h2><span id='topic+getStatsIntervals'></span>

<h3>Description</h3>

<p>This function uses posterior samples to quantify uncertainty in the
properties of fitted curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getStatsIntervals(posteriorStats, mass = 0.9, method = c("eti", "hdi"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getStatsIntervals_+3A_posteriorstats">posteriorStats</code></td>
<td>
<p>A <code>data.table</code> of statistics for posterior samples, as
returned by <code><a href="#topic+getRhythmStats">getRhythmStats()</a></code> or <code><a href="#topic+getDiffRhythmStats">getDiffRhythmStats()</a></code>.</p>
</td></tr>
<tr><td><code id="getStatsIntervals_+3A_mass">mass</code></td>
<td>
<p>Number between 0 and 1 indicating the probability mass for which
to calculate the intervals.</p>
</td></tr>
<tr><td><code id="getStatsIntervals_+3A_method">method</code></td>
<td>
<p>String indicating the type of interval: 'eti' for equal-tailed
using <code><a href="stats.html#topic+quantile">stats::quantile()</a></code>, or 'hdi' for highest density using
<code><a href="HDInterval.html#topic+hdi">HDInterval::hdi()</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code> containing lower and upper bounds of various
statistics for each feature or each feature-condition pair. For
<code>peak_trough_amp</code> and <code>rms_amp</code>, a negative lower bound indicates a rhythm
of the opposite phase.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getRhythmStats">getRhythmStats()</a></code>, <code><a href="#topic+getDiffRhythmStats">getDiffRhythmStats()</a></code>,
<code><a href="#topic+getExpectedMeasIntervals">getExpectedMeasIntervals()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

y = GSE54650$y
metadata = GSE54650$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)
fit = getPosteriorSamples(fit, nPosteriorSamples = 10L)

rhyStatsSamps = getRhythmStats(
  fit, features = c('13170', '13869'), fitType = 'posterior_samples')
rhyStatsInts = getStatsIntervals(rhyStatsSamps)
</code></pre>

<hr>
<h2 id='GSE34018'>Gene expression data for GSE34018</h2><span id='topic+GSE34018'></span>

<h3>Description</h3>

<p>Data are based on total RNA, measured by microarray, obtained from livers of
wild-type and liver-specific Reverb alpha/beta double knockout mice at
various times in a 12h:12h light:dark cycle. To save space and time, the data
include only a subset of genes, and so are mainly useful for examples of how
to use <code>limorhyde2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSE34018
</code></pre>


<h3>Format</h3>

<p>A list with two elements:
</p>

<ul>
<li> <p><code>y</code>: Matrix of normalized, log-transformed expression values. Rows
correspond to genes (rownames are Entrez Gene IDs) and columns to samples.
</p>
</li>
<li> <p><code>metadata</code>: <code>data.table</code> with one row per sample. <code>time</code> is in hours.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE34018">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE34018</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+GSE54650">GSE54650</a>, <code><a href="#topic+getModelFit">getModelFit()</a></code>
</p>

<hr>
<h2 id='GSE54650'>Gene expression data for GSE54650</h2><span id='topic+GSE54650'></span>

<h3>Description</h3>

<p>Data are based on total RNA, measured by microarray, obtained from livers of
wild-type mice at various times after transfer to constant darkness. To save
space and time, the data include only a subset of genes, and so are mainly
useful for examples of how to use <code>limorhyde2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GSE54650
</code></pre>


<h3>Format</h3>

<p>A list with two elements:
</p>

<ul>
<li> <p><code>y</code>: Matrix of normalized, log-transformed expression values. Rows
correspond to genes (rownames are Entrez Gene IDs) and columns to samples.
</p>
</li>
<li> <p><code>metadata</code>: <code>data.table</code> with one row per sample. <code>time</code> is in hours.
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE54650">https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE54650</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+GSE34018">GSE34018</a>, <code><a href="#topic+getModelFit">getModelFit()</a></code>
</p>

<hr>
<h2 id='mergeMeasMeta'>Merge measurements and metadata</h2><span id='topic+mergeMeasMeta'></span>

<h3>Description</h3>

<p>This function is useful for plotting time-courses for individual features.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mergeMeasMeta(y, metadata, features = NULL, sampleColname = "sample")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mergeMeasMeta_+3A_y">y</code></td>
<td>
<p>Matrix-like object of measurements, with rows corresponding to
features and columns to samples.</p>
</td></tr>
<tr><td><code id="mergeMeasMeta_+3A_metadata">metadata</code></td>
<td>
<p>data.frame containing experimental design information for
each sample. Rows of <code>metadata</code> must correspond to columns of <code>y</code>. Row
names are ignored.</p>
</td></tr>
<tr><td><code id="mergeMeasMeta_+3A_features">features</code></td>
<td>
<p>Vector of names, row numbers, or logical values for
subsetting the features. <code>NULL</code> indicates all features.</p>
</td></tr>
<tr><td><code id="mergeMeasMeta_+3A_samplecolname">sampleColname</code></td>
<td>
<p>String indicating the column in <code>metadata</code> containing
the name of each sample, which must correspond to the column names of <code>y</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>data.table</code> with one row for each sample-feature pair.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+getExpectedMeas">getExpectedMeas()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('data.table')

y = GSE34018$y
metadata = GSE34018$metadata

fit = getModelFit(y, metadata)
fit = getPosteriorFit(fit)

measObs = mergeMeasMeta(y, metadata, features = c('13170', '12686'))
measFitMean = getExpectedMeas(
  fit, times = seq(0, 24, 0.5), features = c('13170', '12686'))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
