<!DOCTYPE html><html lang="en"><head><title>Help for package LinkageMapView</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LinkageMapView}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LinkageMapView'><p>LinkageMapView: A package for plotting linkage group maps and QTLs</p></a></li>
<li><a href='#carrot'><p>a carrot comparative linkage map data frame</p>
kindly provided by Massimo Iorizzo:
Cavagnaro et al. BMC Genomics 2014, 15:1118</a></li>
<li><a href='#lmv.linkage.plot'><p>LinkageMapView plotting function</p></a></li>
<li><a href='#lmvdencolor'><p>LinkageMapView density color function</p></a></li>
<li><a href='#oat'><p>oat consensus map data frame</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Plot Linkage Group Maps with Quantitative Trait Loci</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Produces high resolution, publication ready linkage maps
    and quantitative trait loci maps. Input can be output from 'R/qtl',
    simple text or comma delimited files. Output is currently
    a portable document file.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 2.10)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/louellette/LinkageMapView">https://github.com/louellette/LinkageMapView</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/louellette/LinkageMapView/issues">https://github.com/louellette/LinkageMapView/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>qtl (&ge; 1.39-5), plotrix (&ge; 3.6-3), grDevices, graphics,
utils, RColorBrewer</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, testthat, knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-01-21 14:30:53 UTC; loue5</td>
</tr>
<tr>
<td>Author:</td>
<td>Lisa Ouellette [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lisa Ouellette &lt;louellet@uncc.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-01-21 15:00:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='LinkageMapView'>LinkageMapView: A package for plotting linkage group maps and QTLs</h2><span id='topic+LinkageMapView'></span><span id='topic+LinkageMapView-package'></span>

<h3>Description</h3>

<p>LinkageMapView produces high resolution, publication ready linkage maps and QTL maps.
</p>


<h3>Details</h3>

<p>There are many optional parameters to format the output pdf.
Please see the help for function lmv.linkage.plot for a full description of
each parameter and examples.
</p>

<hr>
<h2 id='carrot'>a carrot comparative linkage map data frame
kindly provided by Massimo Iorizzo:
Cavagnaro et al. BMC Genomics 2014, 15:1118</h2><span id='topic+carrot'></span>

<h3>Description</h3>

<p>Contains the following columns:
</p>

<ol>
<li><p> group - This will be the title for the linkage
group unless overridden.
</p>
</li>
<li><p> position - must be in numerical order ascending within
linkage group name.
</p>
</li>
<li><p> locus - marker name at this position.
</p>
</li></ol>



<h3>Usage</h3>

<pre><code class='language-R'>carrot
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 126 rows and 3 columns.</p>

<hr>
<h2 id='lmv.linkage.plot'>LinkageMapView plotting function</h2><span id='topic+lmv.linkage.plot'></span>

<h3>Description</h3>

<p>lmv.linkage.plot is the main function to produce linkage group maps and has many
parameters to customize the pdf output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmv.linkage.plot(mapthis, outfile, mapthese = NULL, at.axis = NULL,
  autoconnadj = TRUE, cex.axis = par("cex.axis"),
  cex.lgtitle = par("cex.main"), cex.main = par("cex.main"),
  col.axis = par("col.axis"), col.lgtitle = par("col.main"),
  col.main = par("col.main"), conndf = NULL, denmap = FALSE,
  dupnbr = FALSE, font.axis = par("font.axis"),
  font.lgtitle = par("font.main"), font.main = par("font.main"),
  header = TRUE, labdist = 0.3, labels.axis = TRUE, lcex = par("cex"),
  lcol = par("col"), lfont = par("font"), lgperrow = NULL,
  lgtitles = NULL, lgw = 0.25, lg.col = NULL, lg.lwd = par("lwd"),
  lty.axis = "solid", lwd.axis = 1, lwd.ticks.axis = lwd.axis,
  main = NULL, markerformatlist = NULL, maxnbrcolsfordups = 3,
  pdf.bg = "transparent", pdf.family = "Helvetica", pdf.fg = "black",
  pdf.width = NULL, pdf.height = NULL, pdf.pointsize = 12,
  pdf.title = "LinkageMapView R output", posonleft = NULL,
  prtlgtitles = TRUE, qtldf = NULL, revthese = NULL, rcex = par("cex"),
  rcol = par("col"), rfont = par("font"), roundpos = 1, rsegcol = TRUE,
  ruler = FALSE, sectcoldf = NULL, segcol = NULL, qtlscanone = NULL,
  showonly = NULL, units = "cM", ylab = units)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lmv.linkage.plot_+3A_mapthis">mapthis</code></td>
<td>
<p>Required, either a 'cross' object from r/qtl, a csv or txt
file or a data frame with the following 3 columns in this order:
</p>

<ol>
<li><p> Required, linkage group name. This will be the title for the linkage
group unless overridden - see lgtitles.
</p>
</li>
<li><p> Required, position - must be in numerical order ascending within
linkage group name.
</p>
</li>
<li><p> Required, locus - marker name at this position.
</p>
</li>
<li><p> Optional, segcol - color for the line across the chromosome
at this marker.  See also segcol parameter.
</p>
</li></ol>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_outfile">outfile</code></td>
<td>
<p>Required, name for the output pdf file.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_mapthese">mapthese</code></td>
<td>
<p>Optional vector of linkage group names to print.
The default, NULL, will print all linkage groups in mapthis.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_at.axis">at.axis</code></td>
<td>
<p>Optional. The points at which tick-marks are to be drawn on the ruler.
Non-finite (infinite, NaN or NA) values are omitted.
By default (when NULL) tickmark locations are computed.
#' @seealso <code><a href="graphics.html#topic+axis">axis</a></code></p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_autoconnadj">autoconnadj</code></td>
<td>
<p>If TRUE (the default), locus with the same name
(homologs) on adjacent linkage groups will be connected with a line.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>The magnification to be used for axis (ruler) text.
The default is par(&quot;cex.axis&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_cex.lgtitle">cex.lgtitle</code></td>
<td>
<p>The magnification to be used for linkage group titles.
The default is par(&quot;cex.main&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_cex.main">cex.main</code></td>
<td>
<p>The magnification to be used for main title.
The default is par(&quot;cex.main&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_col.axis">col.axis</code></td>
<td>
<p>The color to be used for axis (ruler) text.
Defaults to par(&quot;col.axis&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_col.lgtitle">col.lgtitle</code></td>
<td>
<p>The color to be used for linkage group titles.
Defaults to par(&quot;col.main&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_col.main">col.main</code></td>
<td>
<p>The color to be used for the main title.
Defaults to par(&quot;col.main&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_conndf">conndf</code></td>
<td>
<p>An optional data frame containing markers to be connected
with lines (homologs).  If autoconnadj = TRUE, these lines will
appear as well as those with the same name in adjacent linkage
groups.  Required columns:
</p>

<ul>
<li><p> fromchr Linkage group for the line start.
</p>
</li>
<li><p> fromlocus Locus name for the line start.
</p>
</li>
<li><p> tochr Linkage group for the line end.
</p>
</li>
<li><p> tolocus Locus name for the line end.
</p>
</li></ul>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_denmap">denmap</code></td>
<td>
<p>If TRUE, you are requesting a density map which means no locus
or position labels will be printed and the following parameters are
set:
ruler = TRUE
autoconndf = FALSE
conndf = NULL
See also sectcoldf parameter</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_dupnbr">dupnbr</code></td>
<td>
<p>If TRUE, only the first marker name at a position will print
with (## more) afterwards indicating the number of duplicate markers
at that position.  dupnbr should be left to the default, FALSE,
if showonly provided.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_font.axis">font.axis</code></td>
<td>
<p>An integer which specifies which font to use for the
axis (ruler) text.
The default is par(&quot;font.axis&quot;). 1 is plain text.  2 is bold.
3 is italic. 4 is bold italic.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_font.lgtitle">font.lgtitle</code></td>
<td>
<p>An integer which specifies which font to use for the
linkage group titles text.
The default is par(&quot;font.main&quot;). 1 is plain text.  2 is bold.
3 is italic. 4 is bold italic.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_font.main">font.main</code></td>
<td>
<p>An integer which specifies which font to use for title text.
The default is par(&quot;font.main&quot;). 1 is plain text.  2 is bold.
3 is italic. 4 is bold italic.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_header">header</code></td>
<td>
<p>A boolean indicating if the input file has a header row.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_labdist">labdist</code></td>
<td>
<p>Distance in inches from the chromosome to the position
and locus labels. The default is 0.3 inches.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_labels.axis">labels.axis</code></td>
<td>
<p>Optional. This can either be a logical value specifying
whether (numerical) annotations are to be made at the tickmarks on the ruler,
or a character or expression vector of labels to be placed at the tickpoints.
If this is not logical, at should also be supplied and of the same length.
The default is TRUE.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lcex">lcex</code></td>
<td>
<p>A numerical value giving the amount by which position labels
should be magnified.  The default is par(&quot;cex&quot;).
See also rcex for locus labels.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lcol">lcol</code></td>
<td>
<p>The color for the position labels.  The default is par(&quot;col&quot;).
See also rcol for locus labels.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lfont">lfont</code></td>
<td>
<p>An integer which specifies which font to use for the position
labels.  The default is par(&quot;font&quot;).  See also rfont for locus labels.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lgperrow">lgperrow</code></td>
<td>
<p>An integer specifying how many linkage groups to plot in
one row.  As many rows as needed to plot all requested linkage
groups will be plotted.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lgtitles">lgtitles</code></td>
<td>
<p>Optional vector of titles for the linkage groups.  These
will override the default, which is that the linkage group names
in the input print as titles.  This may be useful if in mapthese you
have indicated to print the same linkage group more than once for the
purpose of showing homologous markers without having lines cross.
See also cex.lgtitle, col.lgtitle, font.lgtitle</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lgw">lgw</code></td>
<td>
<p>Width of chromosome in inches.  Default is 0.25 inches.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lg.col">lg.col</code></td>
<td>
<p>Linkage group color.  The color of the chromosomes.
The default is the background color (pdf.bg).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lg.lwd">lg.lwd</code></td>
<td>
<p>Linkage group linewidth. The width of the line around
the chromosome.  Defaults to par(&quot;lwd&quot;).</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lty.axis">lty.axis</code></td>
<td>
<p>Optional. Line type for both the axis line and the tick marks.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lwd.axis">lwd.axis</code></td>
<td>
<p>Optional.  Line width for the axis line.  The default is 1.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_lwd.ticks.axis">lwd.ticks.axis</code></td>
<td>
<p>Optional.  Line width for the axis tick marks.
Default is lwd.axis</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_main">main</code></td>
<td>
<p>An optional title for the linkage group map.  See also
cex.main, col.main, and font.main.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_markerformatlist">markerformatlist</code></td>
<td>
<p>An optional list containing the following vectors:
</p>

<ul>
<li><p> locus Required.  A vector of loci for which the following
should be applied.
</p>
</li>
<li><p> col Optional.  The color for these locus labels.  This color
will override rcol.  See also rsegcol.
</p>
</li>
<li><p> cex Optional.  A numerical vlue giving the amount by which
these locus labels should be magnified.  This value will override
rcex.
</p>
</li>
<li><p> font Optional. An integer which specifies which font to use
for these locus labels. This value will override rfont.
</p>
</li></ul>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_maxnbrcolsfordups">maxnbrcolsfordups</code></td>
<td>
<p>Indicates the number of columns across the page for
locus labels appearing at duplicate positions.  The default is 3.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.bg">pdf.bg</code></td>
<td>
<p>Background color for the pdf.  Default is &quot;transparent&quot;.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.family">pdf.family</code></td>
<td>
<p>Font family for all text.  Default is &quot;Helvetica&quot;.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.fg">pdf.fg</code></td>
<td>
<p>Foreground color for the pdf.  Default is black.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.width">pdf.width</code></td>
<td>
<p>Width of the output file in inches.  Defaults to the size
necessary to fit all linkage groups with other options specified.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.height">pdf.height</code></td>
<td>
<p>Height of the output file in inches.  Defaults to the size
necessary to fit all linkage groups with other options specified.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.pointsize">pdf.pointsize</code></td>
<td>
<p>The default point size to be used.  Defaults to 12.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_pdf.title">pdf.title</code></td>
<td>
<p>Title to be passed to pdf as metadata.  This title does not
appear except in the pdf metadata.  Defaults to
&quot;LinkageMapView R output&quot;.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_posonleft">posonleft</code></td>
<td>
<p>A vector of boolean (TRUE or FALSE) the length of the
number of linkage groups to be plotted. If FALSE, print positions on
right hand side of linkage group and locus names on left hand side
of linkage group.  Default is TRUE.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_prtlgtitles">prtlgtitles</code></td>
<td>
<p>If FALSE do not print linkage group titles.
Default is TRUE.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_qtldf">qtldf</code></td>
<td>
<p>An optional data frame containing QTL information for plotting.
The data frame, if provided, must contain:
</p>

<ul>
<li><p> chr Linkage group name for QTL.
</p>
</li>
<li><p> qtl Name (label) for QTL.
</p>
</li>
<li><p> so Start of outer interval. Numeric.
</p>
</li>
<li><p> si Start of inner interval. Numeric.
</p>
</li>
<li><p> ei End of inner interval. Numeric.
</p>
</li>
<li><p> eo End of outer interval. Numeric.
</p>
</li>
<li><p> col Color for QTL.
</p>
</li></ul>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_revthese">revthese</code></td>
<td>
<p>Optional vector of linkage group names to reverse.  The end
position becomes position 0 and position 0 becomes the end position.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_rcex">rcex</code></td>
<td>
<p>A numerical value giving the amount by which locus labels
should be magnified.  The default is par(&quot;cex&quot;).
See also lcex for position labels.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_rcol">rcol</code></td>
<td>
<p>The color for the locus labels.  The default is par(&quot;col&quot;).
See also lcol for position labels.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_rfont">rfont</code></td>
<td>
<p>An integer which specifies which font to use for the locus
labels.  The default is par(&quot;font&quot;).
See also lfont for position labels.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_roundpos">roundpos</code></td>
<td>
<p>Number of positions after the decimal point to print for
positions.  Default is 1</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_rsegcol">rsegcol</code></td>
<td>
<p>Color of the segments across the chromosome and to the label.
TRUE, the default, indicates the color should be the
same as the label.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_ruler">ruler</code></td>
<td>
<p>A single boolean (TRUE OR FALSE).  If TRUE, an axis is
drawn on the left hand side of the page and the position labels
are not printed on any linkage group.  The default is FALSE.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_sectcoldf">sectcoldf</code></td>
<td>
<p>Optional data frame containing the following named columns
indicating sections of the chromosome to be colored:
</p>

<ul>
<li><p> Required, chr - matches from input file or cross object
linkage group name
</p>
</li>
<li><p> Required, s - start position in cM
</p>
</li>
<li><p> Required, e - end position in cM
</p>
</li>
<li><p> Required, col - color for section
</p>
</li>
<li><p> Optional, dens - a numeric cm / marker value used to print
the density map legend.
</p>
</li></ul>

<p>For a density map, use the lmvdencolor function to populate sectcoldf.
When denmap = TRUE and no sectcoldf parameter is supplied, lmvdencolor
is called with defaults fully populating the sectcoldf data frame.
See also the denmap parameter.
</p>
<p>@seealso <code><a href="#topic+lmvdencolor">lmvdencolor</a></code></p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_segcol">segcol</code></td>
<td>
<p>Optional. Name of the column in mapthis that contains colors for the line
segments across the chromosome.  If specified, this overrides rsegcol.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_qtlscanone">qtlscanone</code></td>
<td>
<p>Optional scanone data frame from package r/qtl.  If provided,
all QTLs in the dataframe will be drawn by calculating their
start and end with the r/qtl function bayesint with defaults.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_showonly">showonly</code></td>
<td>
<p>Optional vector of marker names.  If provided, only these
marker names will be printed.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_units">units</code></td>
<td>
<p>Units of the position values supplied in mapthis.  The default
value is cM (centimorgan) but any value can be provided.  The
value provided is only used for a ruler (y axis) title and the density
map legend text.</p>
</td></tr>
<tr><td><code id="lmv.linkage.plot_+3A_ylab">ylab</code></td>
<td>
<p>Optional.  Title for the y-axis (ruler).  The default value is
units.  See units parameter.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
## take a cross object from r/qtl and produce linkage map
## on chr 1,4,6,15

library(qtl)
data(hyper)
outfile = file.path(tempdir(), "hyper.pdf")
lmv.linkage.plot(hyper,outfile,mapthese=c(1,4,6,15))

## color some of the markers for emphasis

library(qtl)
data(hyper)

# make a list to pass label options
flist &lt;- list()
locus &lt;- c("D1Mit123","D1Mit105","D6Mit273","D15Mit56","D15Mit156")
col   &lt;- c("red")
flist[[1]] &lt;- list(locus=locus,col=col)

outfile = file.path(tempdir(), "hyperred.pdf")
lmv.linkage.plot(hyper,outfile,mapthese=c(1,4,6,15),markerformatlist=flist)

## change some of the pdf options and chromosome color
## changing linkage group title color (col.lgtitle) to same as
## foreground pdf color

library(qtl)
data(hyper)

outfile = file.path(tempdir(), "hyperlg.pdf")
lmv.linkage.plot(hyper,outfile,
mapthese=c(1,4,6,15),
pdf.bg="black",pdf.fg="white",col.lgtitle="white",
pdf.height=8,pdf.title="myhyper",lg.col="tan")

## change all label colors and fonts

library(qtl)
data(hyper)

outfile = file.path(tempdir(), "hypercol.pdf")
lmv.linkage.plot(hyper,outfile,mapthese=c(1,4,6,15),
lcol="blue",lfont=2,lcex=1.2,rcol="red",rfont=3,rcex=2)

## make a dataframe to pass sections of chr to col
## use a ruler instead of printing positions as labels
## only allow one column for duplicate markers at same position
## (default is 3)

library(qtl)
data(hyper)

chr = c(1, 4, 6, 15)
s = c(82,35,9.8,7.7)
e = c(94,47,21.9,13.1)
col = c("pink","blue","blue","green")
sectcoldf &lt;-  data.frame(chr, s, e, col,stringsAsFactors = FALSE)

outfile = file.path(tempdir(), "hyperruler.pdf")
lmv.linkage.plot(hyper,outfile,mapthese=c(1,4,6,15),
ruler=TRUE,maxnbrcolsfordups = 1, sectcoldf=sectcoldf)

## plot qtls also out of a r/qtl scanone object
## plot marker names on left (instead of right) of chr 4 and 7

library(qtl)
data(hyper)

# create scanone df for testing
hyper &lt;-
 calc.genoprob(hyper,
               step = 2.0,
               map.function = "haldane",
               stepwidth = "fixed")
hyper.scanone &lt;- scanone(hyper)

outfile = file.path(tempdir(), "testrqtlhyper2.pdf")
lmv.linkage.plot(hyper,
   outfile, mapthese=c(1,4,6,7,15),
   qtlscanone = hyper.scanone,
   posonleft = c(TRUE,FALSE,TRUE,FALSE,TRUE))

## Not run: 
## plot a carrot comparative linkage map
## kindly provided by Massimo Iorizzo:
## Cavagnaro et al. BMC Genomics 2014, 15:1118

# make a df to pass qtl info
qtldf &lt;- data.frame(
  chr = character(),
  qtl = character(),
  so = numeric(),
  si = numeric(),
  ei = numeric(),
  eo = numeric(),
  col = character(),
  stringsAsFactors = FALSE
)
qtldf &lt;- rbind(qtldf,
               data.frame(
                 chr = "70349LG3",
                 qtl = "RTPE-Q1",
                 so = 36.6,
                 si = 37,
                 ei = 37,
                 eo = 38,
                 col="red"
               ))
# make a list to pass label options
flist &lt;- list()
locus &lt;- c("BSSR-094", "K0149", "K0627", "K2161", "ESSR-087", "ESSR-057")
font  &lt;- c(2)   #bold
flist[[1]] &lt;- list(locus = locus, font = font)
locus &lt;- c("F3H", "FLS1")
font  &lt;- c(4)   #bold italic
flist[[2]] &lt;- list(locus = locus, font = font)
locus &lt;- c("P3", "P1", "Raa1")
font  &lt;- c(3)   #italic
col &lt;- c("red")
flist[[3]] &lt;- list(locus = locus, font = font, col = col)
filename &lt;- system.file("extdata", "Carrot.csv", package="LinkageMapView")
outfile = file.path(tempdir(), "carrot.pdf")
lmv.linkage.plot(
  mapthis = filename,
  outfile = outfile,
  ruler = TRUE,
  lgtitle = c("2170", "70349", "10117"),
  maxnbrcolsfordups = 1,
  markerformatlist = flist,
  lg.col = "lightblue1",
  pdf.width =10,
  revthese = c("70349LG3"),
  qtldf=qtldf
)

## End(Not run)

## do a density map with default colors
data(oat)

outfile = file.path(tempdir(), "oat_Mrg01.pdf")
lmv.linkage.plot(oat,outfile,mapthese=c("Mrg01","Mrg02"),denmap=TRUE)

## Not run: 
## do a density map and provide your own colors with lmvdencolor helper
data(oat)
##
outfile = file.path(tempdir(), "oat_Mrg01_YlGn.pdf")

sectcoldf &lt;- lmvdencolor(oat,colorin =
colorRampPalette(RColorBrewer::brewer.pal(8, "YlGn"))(5))

lmv.linkage.plot(oat,outfile,denmap=TRUE,sectcoldf=sectcoldf)

## End(Not run)
</code></pre>

<hr>
<h2 id='lmvdencolor'>LinkageMapView density color function</h2><span id='topic+lmvdencolor'></span>

<h3>Description</h3>

<p>lmvdencolor is a helper function which you can use to create a data frame
of colors to be used as the sectcoldf input parameter on the lmv.linkage.plot command.
The colors will be used to color the linkage group based on the density
of position/marker.This function is called with default values when the
denmap = TRUE parameter is specified for lmv.linkage.plot and no sectcoldf parameter
is found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lmvdencolor(df, wsize = 30, bias = 5,
  colorin = colorRampPalette(RColorBrewer::brewer.pal(8, "Spectral"))(25))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lmvdencolor_+3A_df">df</code></td>
<td>
<p>Required, a data frame with the first two columns in this order:
</p>

<ol>
<li><p> Linkage group name.
</p>
</li>
<li><p> Position - must be in numerical order ascending within
linkage group name.  If the maximum position in any linkage
group is &lt; 1000, the density will be calculated
for each position.  Otherwise the number of positions included
for each density calculation will be:
ceiling(maximum position of an linkage group/1000)
</p>
</li></ol>
</td></tr>
<tr><td><code id="lmvdencolor_+3A_wsize">wsize</code></td>
<td>
<p>Optional, default = 30.  # of postions in the sliding window
for calculating postitions/marker. If the maximum position in
any linkage group is &gt;= 1000, the default sliding window size
will be adjusted by the same ratio as the number of positions
included for each density calculate.</p>
</td></tr>
<tr><td><code id="lmvdencolor_+3A_bias">bias</code></td>
<td>
<p>Optional, default = 5. a positive number. Higher values give
more widely spaced colors at the high end.</p>
</td></tr>
<tr><td><code id="lmvdencolor_+3A_colorin">colorin</code></td>
<td>
<p>Optional, a vector of colors to use where the first value is
the color for the lowest density and the last value is the color
for the highest density.  Default is:
rev(colorRampPalette(RColorBrewer::brewer.pal(8, &quot;Spectral&quot;))(25))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame that can be used as sectcoldf input on the lmv.linkage.plot
function to color the chromosome for a density map.
</p>


<h3>See Also</h3>

<p><code><a href="grDevices.html#topic+colorRamp">colorRamp</a></code>
</p>
<p><code><a href="#topic+lmv.linkage.plot">lmv.linkage.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# add a column to a linkage group data frame to specify colors for
# line segments in lmv.linkage.plot using default colors from RColorBrewer
# Spectral palette.  Then just plot the returned colors out to see how
# they look.

data(oat)

sectcoldf &lt;- lmvdencolor(oat)

# see colors produced

image(seq_along(oat[,2]), 1, as.matrix(seq_along(oat[,2])),
 col=sectcoldf$col, axes=FALSE, xlab="", ylab="")
</code></pre>

<hr>
<h2 id='oat'>oat consensus map data frame</h2><span id='topic+oat'></span>

<h3>Description</h3>

<p>Chaffin, A. S., Y. Huang, S. Smith, W. A. Bekele, E. Babiker, B. N. Gnanesh,
B. J. Foresman, S. G. Blanchard, J. J. Jay, R. W. Reid, C. P. Wight, S. Chao,
R. Oliver, E. Islamovic, F. L. Kolb, C. McCartney, J. W. Mitchell Fetch,
A. D. Beattie, ?. Bjornstad, J. M. Bonman, T. Langdon, C. J. Howarth,
C. R. Brouwer, E. N. Jellen, K. E. Klos, J. A. Poland, T. Hsieh, R. Brown,
E. Jackson, J. A. Schlueter, and N. A. Tinker.
2016. A Consensus Map in Cultivated Hexaploid Oat Reveals Conserved Grass
Synteny with Substantial Subgenome Rearrangement.
Plant Genome 9. doi:10.3835/plantgenome2015.10.0102
</p>


<h3>Usage</h3>

<pre><code class='language-R'>oat
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 16668 rows and 3 columns.</p>


<h3>Details</h3>

<p>Contains the following columns:
</p>

<ol>
<li><p> Group - This will be the title for the linkage
group unless overridden.
</p>
</li>
<li><p> Position - must be in numerical order ascending within
linkage group name.
</p>
</li>
<li><p> Locus - marker name at this position.
</p>
</li></ol>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
