<!DOCTYPE html><html><head><title>Help for package lit</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lit}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gamut_plink'><p>GAMuT</p></a></li>
<li><a href='#lit'><p>Latent Interaction Testing</p></a></li>
<li><a href='#lit_h'><p>LIT correcting for dominance effects</p></a></li>
<li><a href='#lit_plink'><p>Latent Interaction Testing</p></a></li>
<li><a href='#marginal'><p>Marginal (SQ/CP) approach</p></a></li>
<li><a href='#marginal_plink'><p>Marginal (SQ/CP) approach</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Latent Interaction Testing for Genome-Wide Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Andrew Bass &lt;ajbass@emory.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identifying latent genetic interactions in genome-wide association studies
    using the Latent Interaction Testing (LIT) framework.
    LIT is a flexible kernel-based approach that leverages information across
    multiple traits to detect latent genetic interactions without specifying or
    observing the interacting variable (e.g., environment). LIT accepts
    standard PLINK files as inputs to analyze large genome-wide association studies.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ajbass/lit">https://github.com/ajbass/lit</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-2">LGPL-2</a> | <a href="https://www.r-project.org/Licenses/LGPL-2.1">LGPL-2.1</a> | <a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a> [expanded from: LGPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppEigen</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.11), genio, CompQuadForm</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-08-12 18:45:11 UTC; ajbass</td>
</tr>
<tr>
<td>Author:</td>
<td>Andrew Bass [aut, cre],
  Michael Epstein [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-08-15 10:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gamut_plink'>GAMuT</h2><span id='topic+gamut_plink'></span>

<h3>Description</h3>

<p>The <code>GAMuT</code> function is a kernel-based multivariate association test.
Note that our code to process plink files builds from the
<code>genio</code> R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gamut_plink(y, file, adjustment = NULL, pop_struct = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gamut_plink_+3A_y">y</code></td>
<td>
<p>matrix of traits (n observations by k traits)</p>
</td></tr>
<tr><td><code id="gamut_plink_+3A_file">file</code></td>
<td>
<p>path to plink files</p>
</td></tr>
<tr><td><code id="gamut_plink_+3A_adjustment">adjustment</code></td>
<td>
<p>matrix of covariates to adjust traits</p>
</td></tr>
<tr><td><code id="gamut_plink_+3A_pop_struct">pop_struct</code></td>
<td>
<p>matrix of PCs that captures population structure</p>
</td></tr>
<tr><td><code id="gamut_plink_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE (default) print progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of p-values where the columns are the cross products/squared residuals
and the rows are SNPs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lit_plink">lit_plink</a></code>, <code><a href="#topic+marginal_plink">marginal_plink</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set seed
set.seed(123)

# path to plink files
file &lt;- system.file("extdata", 'sample.bed', package = "genio", mustWork = TRUE)

# Generate trait expression
Y &lt;- matrix(rnorm(10*4), ncol = 4)

out &lt;- gamut_plink(Y, file = file)

</code></pre>

<hr>
<h2 id='lit'>Latent Interaction Testing</h2><span id='topic+lit'></span>

<h3>Description</h3>

<p><code>lit</code> performs a kernel-based testing procedure, Latent Interaction Testing (LIT), using a set of traits and SNPs.
LIT tests whether the squared residuals (SQ) and cross products (CP) are statistically independent of the genotypes.
In particular, we construct a kernel matrix for the SQ/CP terms to measure the pairwise
similarity between individuals, and also construct an analogous one for the genotypes.
We then test whether these two matrices are independent.
Currently, we implement the linear and projection kernel functions to measure pairwise similarity between individuals.
We then combine the p-values of these implementations using a Cauchy combination test to maximize the number of discoveries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lit(y, x, adjustment = NULL, pop_struct = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lit_+3A_y">y</code></td>
<td>
<p>matrix of traits (n observations by k traits)</p>
</td></tr>
<tr><td><code id="lit_+3A_x">x</code></td>
<td>
<p>matrix of SNPs (n observations by m SNPs)</p>
</td></tr>
<tr><td><code id="lit_+3A_adjustment">adjustment</code></td>
<td>
<p>matrix of covariates to adjust traits</p>
</td></tr>
<tr><td><code id="lit_+3A_pop_struct">pop_struct</code></td>
<td>
<p>matrix of PCs that captures population structure</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of p-values where the columns are
</p>

<ul>
<li><p><code>wlit</code>: LIT using a linear kernel
</p>
</li>
<li><p><code>ulit</code>: LIT using a projection kernel
</p>
</li>
<li><p><code>alit</code>: Cauchy combination test of the above two LIT implementations.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+lit_plink">lit_plink</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set seed
set.seed(123)

# Generate SNPs and traits
X &lt;- matrix(rbinom(10*2, size = 2, prob = 0.25), ncol = 2)
Y &lt;- matrix(rnorm(10*4), ncol = 4)

out &lt;- lit(Y, X)

</code></pre>

<hr>
<h2 id='lit_h'>LIT correcting for dominance effects</h2><span id='topic+lit_h'></span>

<h3>Description</h3>

<p>Internal use for now
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lit_h(y, x, adjustment = NULL, pop_struct = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lit_h_+3A_y">y</code></td>
<td>
<p>matrix of traits (n observations by k traits)</p>
</td></tr>
<tr><td><code id="lit_h_+3A_x">x</code></td>
<td>
<p>matrix of SNPs (n observations by m SNPs)</p>
</td></tr>
<tr><td><code id="lit_h_+3A_adjustment">adjustment</code></td>
<td>
<p>matrix of covariates to adjust traits</p>
</td></tr>
<tr><td><code id="lit_h_+3A_pop_struct">pop_struct</code></td>
<td>
<p>matrix of PCs that captures population structure</p>
</td></tr>
</table>

<hr>
<h2 id='lit_plink'>Latent Interaction Testing</h2><span id='topic+lit_plink'></span>

<h3>Description</h3>

<p><code>lit_plink</code> performs a kernel-based testing procedure, Latent Interaction Testing (LIT), using a set of traits and SNPs.
LIT tests whether the squared residuals (SQ) and cross products (CP) are statistically independent of the genotypes.
In particular, we construct a kernel matrix for the SQ/CP terms to measure the pairwise
similarity between individuals, and also construct an analogous one for the genotypes.
We then test whether these two matrices are independent.
Currently, we implement the linear and projection kernel functions to measure pairwise similarity between individuals.
We then combine the p-values of these implementations using a Cauchy combination test to maximize the number of discoveries.
This function is suitable for large  datasets (e.g., UK Biobank) in plink format.
Note that our code to process plink files builds from the
<code>genio</code> R package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lit_plink(y, file, adjustment = NULL, pop_struct = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lit_plink_+3A_y">y</code></td>
<td>
<p>matrix of traits (n observations by k traits)</p>
</td></tr>
<tr><td><code id="lit_plink_+3A_file">file</code></td>
<td>
<p>path to plink files</p>
</td></tr>
<tr><td><code id="lit_plink_+3A_adjustment">adjustment</code></td>
<td>
<p>matrix of covariates to adjust traits</p>
</td></tr>
<tr><td><code id="lit_plink_+3A_pop_struct">pop_struct</code></td>
<td>
<p>matrix of PCs that captures population structure</p>
</td></tr>
<tr><td><code id="lit_plink_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE (default) print progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of p-values where the columns are
</p>

<ul>
<li><p><code>wlit</code>: LIT using a linear kernel
</p>
</li>
<li><p><code>ulit</code>: LIT using a projection kernel
</p>
</li>
<li><p><code>alit</code>: Cauchy combination test of the above two LIT implementations.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+lit">lit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set seed
set.seed(123)

# path to plink files
file &lt;- system.file("extdata", 'sample.bed', package = "genio", mustWork = TRUE)

# Generate trait expression
Y &lt;- matrix(rnorm(10*4), ncol = 4)

out &lt;- lit_plink(Y, file = file)

</code></pre>

<hr>
<h2 id='marginal'>Marginal (SQ/CP) approach</h2><span id='topic+marginal'></span>

<h3>Description</h3>

<p>The <code>marginal</code> function performs a trait-by-trait univariate test for latent interactions
using the squared residuals and cross products.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marginal(y, x, adjustment = NULL, pop_struct = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="marginal_+3A_y">y</code></td>
<td>
<p>matrix of traits (n observations by k traits)</p>
</td></tr>
<tr><td><code id="marginal_+3A_x">x</code></td>
<td>
<p>matrix of SNPs (n observations by m SNPs)</p>
</td></tr>
<tr><td><code id="marginal_+3A_adjustment">adjustment</code></td>
<td>
<p>matrix of covariates to adjust traits</p>
</td></tr>
<tr><td><code id="marginal_+3A_pop_struct">pop_struct</code></td>
<td>
<p>matrix of PCs that captures population structure</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of p-values where the columns are the cross products/squared residuals
and the rows are SNPs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+marginal_plink">marginal_plink</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set seed
set.seed(123)

# Generate SNPs and traits
X &lt;- matrix(rbinom(10*2, size = 2, prob = 0.25), ncol = 2)
Y &lt;- matrix(rnorm(10*4), ncol = 4)

out &lt;- marginal(Y, X)

</code></pre>

<hr>
<h2 id='marginal_plink'>Marginal (SQ/CP) approach</h2><span id='topic+marginal_plink'></span>

<h3>Description</h3>

<p>The <code>marginal_plink</code> function performs a trait-by-trait univariate test for latent interactions
using the squared residuals and cross products. This function is suitable for large
datasets (e.g., UK Biobank) in plink format. Note that our code to process plink files builds from the
<code>genio</code> R package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marginal_plink(y, file, adjustment = NULL, pop_struct = NULL, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="marginal_plink_+3A_y">y</code></td>
<td>
<p>matrix of traits (n observations by k traits)</p>
</td></tr>
<tr><td><code id="marginal_plink_+3A_file">file</code></td>
<td>
<p>path to plink files</p>
</td></tr>
<tr><td><code id="marginal_plink_+3A_adjustment">adjustment</code></td>
<td>
<p>matrix of covariates to adjust traits</p>
</td></tr>
<tr><td><code id="marginal_plink_+3A_pop_struct">pop_struct</code></td>
<td>
<p>matrix of PCs that captures population structure</p>
</td></tr>
<tr><td><code id="marginal_plink_+3A_verbose">verbose</code></td>
<td>
<p>If TRUE (default) print progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of p-values where the columns are the cross products/squared residuals
and the rows are SNPs.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+marginal_plink">marginal_plink</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># set seed
set.seed(123)

# Path to plink files
file &lt;- system.file("extdata", 'sample.bed', package = "genio", mustWork = TRUE)

# Generate trait expression
Y &lt;- matrix(rnorm(10*4), ncol = 4)

out &lt;- marginal_plink(Y, file = file)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
