<!DOCTYPE html><html><head><title>Help for package lncDIFF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lncDIFF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cov'><p>Batch information  for samples in hnsc.edata.</p></a></li>
<li><a href='#design'><p>Design matrix for samples in hnsc.edata.</p></a></li>
<li><a href='#hnsc.edata'><p>lncRNA Fragments Per Killobase per Million (FPKM) in a head and neck squamous cell carcinomas (hnsc) study.</p></a></li>
<li><a href='#lncDIFF'><p>lncRNA Differential Expression (DE) analysis</p></a></li>
<li><a href='#LRT'><p>Likelihood ratio test based on ZIQML.fit()</p></a></li>
<li><a href='#tissue'><p>Tissue type for samples in hnsc.edata.</p></a></li>
<li><a href='#ZIQML.fit'><p>Group and covariate effects on lncRNA counts by Generalized Linear Model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Long Non-Coding RNA Differential Expression Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Description:</td>
<td>We developed an approach to detect differential expression features in long non-coding RNA low counts, using generalized linear model with zero-inflated exponential quasi likelihood ratio test. Methods implemented in this package are described in Li (2019) &lt;<a href="https://doi.org/10.1186%2Fs12864-019-5926-4">doi:10.1186/s12864-019-5926-4</a>&gt;. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-10 03:36:50 UTC; liq</td>
</tr>
<tr>
<td>Author:</td>
<td>Qian Li [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Qian Li &lt;qian.li10000@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-13 22:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='cov'>Batch information  for samples in hnsc.edata.</h2><span id='topic+cov'></span>

<h3>Description</h3>

<p>Batch information  for samples in hnsc.edata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cov
</code></pre>


<h3>Format</h3>

<p>A matrix of covariate(s) in columns.</p>

<hr>
<h2 id='design'>Design matrix for samples in hnsc.edata.</h2><span id='topic+design'></span>

<h3>Description</h3>

<p>Design matrix for samples in hnsc.edata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>design
</code></pre>


<h3>Format</h3>

<p>A model matrix with 80 rows (i.e. samples) and 3 columns of tissue type and batch.</p>

<hr>
<h2 id='hnsc.edata'>lncRNA Fragments Per Killobase per Million (FPKM) in a head and neck squamous cell carcinomas (hnsc) study.</h2><span id='topic+hnsc.edata'></span>

<h3>Description</h3>

<p>lncRNA Fragments Per Killobase per Million (FPKM) in a head and neck squamous cell carcinomas (hnsc) study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hnsc.edata
</code></pre>


<h3>Format</h3>

<p>A data frame of lncRNA FPKM with 1000 rows (i.e. genes) and 80 columns (i.e. samples ).</p>

<hr>
<h2 id='lncDIFF'>lncRNA Differential Expression (DE) analysis</h2><span id='topic+lncDIFF'></span>

<h3>Description</h3>

<p>lncDIFF returns DE analysis results based on lncRNA counts and grouping variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lncDIFF(
  edata,
  group,
  covariate = NULL,
  link.function = "log",
  CompareGroups = NULL,
  simulated.pvalue = FALSE,
  permutation = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lncDIFF_+3A_edata">edata</code></td>
<td>
<p>Normalized counts matrix with genes in rows and samples in columns.</p>
</td></tr>
<tr><td><code id="lncDIFF_+3A_group">group</code></td>
<td>
<p>Primary factor of interest in DE analysis, e.g., treatment groups, tissue types, other phenotypes.</p>
</td></tr>
<tr><td><code id="lncDIFF_+3A_covariate">covariate</code></td>
<td>
<p>Other variables (or covariates) associated with expression level. Input must be a matrix or data frame with each column being a covariate matching to <code>group</code></p>
</td></tr>
<tr><td><code id="lncDIFF_+3A_link.function">link.function</code></td>
<td>
<p>Link function for the generalized linear model, either 'log' or 'identity', default as 'log'.</p>
</td></tr>
<tr><td><code id="lncDIFF_+3A_comparegroups">CompareGroups</code></td>
<td>
<p>Labels of treatment groups or phenotypes of interest to be compared in DE analysis. Input must be a vector of <code>group</code> labels without duplicates.</p>
</td></tr>
<tr><td><code id="lncDIFF_+3A_simulated.pvalue">simulated.pvalue</code></td>
<td>
<p>If empirical p-values are computed, simulated.pvalue=TRUE. The default is FALSE.</p>
</td></tr>
<tr><td><code id="lncDIFF_+3A_permutation">permutation</code></td>
<td>
<p>The number of permutations used in simulating pvalues. The default value is 100.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>DE.results</code></td>
<td>
<p>Likelihood ratio test results with test statistics, p-value, FDR, DE genes, groupwise mean expression, fold change (if two groups are compared). 
If simulated.pvalue=TRUE, test.results also includes simulated p-value and FDR.</p>
</td></tr>  
<tr><td><code>full.model.fit</code></td>
<td>
<p>Generalized linear model with zero-inflated Exponential likelihood function, estimating group effect compared to a reference group.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Li, Q., Yu, X., Chaudhary, R. et al.'lncDIFF: a novel quasi-likelihood method for differential expression analysis of non-coding RNA'. BMC Genomics (2019) 20: 539.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data('hnsc.edata','tissue','cov')  

# DE analysis comparing two groups (normal vs tumor) for 100 genes
result=lncDIFF(edata=hnsc.edata[1:100,],group=tissue,covariate=cov) 

# Recommend at least 50 permutations if simulated.pvalue=TRUE




</code></pre>

<hr>
<h2 id='LRT'>Likelihood ratio test based on ZIQML.fit()</h2><span id='topic+LRT'></span>

<h3>Description</h3>

<p>ZIQML.LRT returns the likelihood ratio test statistics and p-value based on the object returned by ZIQML.fit().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LRT(ZIQML.fit, coef = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LRT_+3A_ziqml.fit">ZIQML.fit</code></td>
<td>
<p>Object returned by ZIQML.fit()</p>
</td></tr>
<tr><td><code id="LRT_+3A_coef">coef</code></td>
<td>
<p>An integer or vector indicating the coefficient(s) in design matrix to be tested. coef=1 is the intercept (i.e. baseline group effect),
and should not be tested.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>LRT.stat</code></td>
<td>
<p>Likelihood ratio test statistics.</p>
</td></tr>  
<tr><td><code>LRT.pvalue</code></td>
<td>
<p>Likelihood ratio test p-value.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data('hnsc.edata','design')  
# 'hnsc.edata' contains FPKM of 1132 lncRNA genes and 80 samples. 
# 'design' is the design matrix of tissue type (tumor vs normal). 

# Fit GLM by ZIQML.fit for the first 100 genes 
fit.log=ZIQML.fit(edata=hnsc.edata[1:100,],design.matrix=design) 


# Likelihood ratio test to compare tumor vs normal in gene expression level. 
LRT.results=LRT(fit.log,coef=2)  

</code></pre>

<hr>
<h2 id='tissue'>Tissue type for samples in hnsc.edata.</h2><span id='topic+tissue'></span>

<h3>Description</h3>

<p>Tissue type for samples in hnsc.edata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tissue
</code></pre>


<h3>Format</h3>

<p>A character vector of tissue type.</p>

<hr>
<h2 id='ZIQML.fit'>Group and covariate effects on lncRNA counts by Generalized Linear Model</h2><span id='topic+ZIQML.fit'></span>

<h3>Description</h3>

<p>ZIQML.fit estimates the group effect on gene expression using zero-inflated exponential quasi likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZIQML.fit(edata, design.matrix, link = "log")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ZIQML.fit_+3A_edata">edata</code></td>
<td>
<p>Normalized counts matrix with genes in rows and samples in columns.</p>
</td></tr>
<tr><td><code id="ZIQML.fit_+3A_design.matrix">design.matrix</code></td>
<td>
<p>Design matrix for groups and covariates, generated by model.matrix().</p>
</td></tr>
<tr><td><code id="ZIQML.fit_+3A_link">link</code></td>
<td>
<p>Link function for the generalized linear model and likelihood function,either 'log' or 'identity'. The default is 'log'.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>Estimates</code></td>
<td>
<p>Estimated group effect on gene expression by zero-inflated exponential quasi maximum likelihood (ZIQML) estimator.</p>
</td></tr>
<tr><td><code>logLikelihood</code></td>
<td>
<p>The value of zero-inflated quasi likelihood.</p>
</td></tr> 
<tr><td><code>edata</code></td>
<td>
<p>lncRNA counts or expression matrix.</p>
</td></tr> 
<tr><td><code>design.matrix</code></td>
<td>
<p>The design matrix of groups and covariates.</p>
</td></tr> 
<tr><td><code>link</code></td>
<td>
<p>The specified link function.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
data('hnsc.edata','design') 
# 'hnsc.edata' contains FPKM of 1000 lncRNA genes and 80 samples 
# 'design' is the design matrix for tissue and batch.

# For the first 100 genes  
# Fit GLM by ZIQML with logarithmic link function                                      
fit.log=ZIQML.fit(edata=hnsc.edata[1:100,],design.matrix=design,link='log') 

# Fit GLM by ZIQML with identity link function
fit.identity=ZIQML.fit(edata=hnsc.edata[1:100,],design.matrix=design,link='identity') 

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
