<!DOCTYPE html><html><head><title>Help for package LncPath</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LncPath}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#drawAHeatMap'>

<p>Draw a heatmap for the genes of a pathway</p></a></li>
<li><a href='#findSigGenes'>

<p>Find genes significantly differentially expressed between two conditions.</p></a></li>
<li><a href='#geneSetDetail'>

<p>Gain insight into the detail of the genes in a certain pathway</p></a></li>
<li><a href='#getExampleData'>

<p>Get the example data</p></a></li>
<li><a href='#getNet'>

<p>Get the background lncRNA-mRNA interaction network</p></a></li>
<li><a href='#lncPath'>
<p>Identify pathways synergisticly regulated by lncRNA sets.</p></a></li>
<li><a href='#LncPath-internal'><p>LncPath internal functions</p></a></li>
<li><a href='#lncPath2Table'>

<p>Simplify the lncPath object into table</p></a></li>
<li><a href='#LncPathEnvir'><p>The variables in the environment variable <code>LncPathEnvir</code> of the system</p></a></li>
<li><a href='#plotRunningES'>

<p>Visualize the Kolmogorov-Smirnov running score of pathway</p></a></li>
<li><a href='#printSignifResult'>

<p>Output the details of significant pathways</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Identifying the Pathways Regulated by LncRNA Sets of Interest</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-09-26</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han, Zeguo Sun</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Identifies pathways synergisticly regulated by the interested lncRNA(long non-coding RNA) sets based on a lncRNA-mRNA(messenger RNA) interaction network. 1) The lncRNA-mRNA interaction network was built from the protein-protein interactions and the lncRNA-mRNA co-expression relationships in 28 RNA-Seq data sets. 2) The interested lncRNAs can be mapped into networks as seed nodes and a random walk strategy will be performed to evaluate the rate of each coding genes influenced by the seed lncRNAs. 3) Pathways regulated by the lncRNA set will be evaluated by a weighted Kolmogorov-Smirnov statistic as an ES Score. 4) The p value and false discovery rate value will also be calculated through a permutation analysis. 5) The running score of each pathway can be plotted and the heat map of each pathway can also be plotted if an expression profile is provided. 6) The rank and scores of the gene list of each pathway can be printed.</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, graphics, utils, grDevices</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.1), igraph</td>
</tr>
<tr>
<td>Suggests:</td>
<td>Matrix,graph</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-09-26 06:47:17 UTC; liu</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-09-26 14:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='drawAHeatMap'>

Draw a heatmap for the genes of a pathway
</h2><span id='topic+drawAHeatMap'></span>

<h3>Description</h3>


<p>Draw a heatmap for the genes of a certain pathway based on the expression profile user specified.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drawAHeatMap(Result, Name, PCExpr, Labels)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drawAHeatMap_+3A_result">Result</code></td>
<td>


<p>A lncPath object come from the lncPath function.
</p>
</td></tr>
<tr><td><code id="drawAHeatMap_+3A_name">Name</code></td>
<td>


<p>A string, the name of the pathway to be plot.
</p>
</td></tr>
<tr><td><code id="drawAHeatMap_+3A_pcexpr">PCExpr</code></td>
<td>


<p>A data frame, the expression profile to be plotted.
</p>
</td></tr>
<tr><td><code id="drawAHeatMap_+3A_labels">Labels</code></td>
<td>


<p>A vector of 0 and 1, 0 indicates control and 1 indicates case.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>Draw a heatmap of the genes of a pathway based on the expression profile. The rows of heatmap are genes ranked by their weights and the columns of heatmap are samples ordered the same as the expression profile. 
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

Result &lt;- getExampleData("Result")
Profile &lt;- getExampleData("Profile")
Labels &lt;- getExampleData("Labels")
drawAHeatMap(Result, "KEGG_RIBOSOME", Profile, Labels)

</code></pre>

<hr>
<h2 id='findSigGenes'>

Find genes significantly differentially expressed between two conditions.
</h2><span id='topic+findSigGenes'></span>

<h3>Description</h3>


<p>For a given expression profile of two conditions, find the genes differencially expressed using T-test, fold change or SAM algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findSigGenes(Expr, Label, Method = "tTest", Directed = TRUE, 
	FdrCut = 0.01, FDCut = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findSigGenes_+3A_expr">Expr</code></td>
<td>


<p>A data frame, the expression profile to find differentially expressed genes, the rownames should be the ID of genes.
</p>
</td></tr>
<tr><td><code id="findSigGenes_+3A_label">Label</code></td>
<td>


<p>A vector of 0/1s, indicating the class of samples in the expression profile, 0 represents case, 1 represents control.
</p>
</td></tr>
<tr><td><code id="findSigGenes_+3A_method">Method</code></td>
<td>


<p>A string, specifying the method to calculate the differentially expressed genes, should be one of the &quot;tTest&quot;or&quot;foldChange&quot;.
</p>
</td></tr>
<tr><td><code id="findSigGenes_+3A_directed">Directed</code></td>
<td>


<p>Logical, if the the up or down regulated set should be distinguished.
</p>
</td></tr>
<tr><td><code id="findSigGenes_+3A_fdrcut">FdrCut</code></td>
<td>


<p>Numeric, the fdr cutoff for T test, can be ignored if not using t-test.
</p>
</td></tr>
<tr><td><code id="findSigGenes_+3A_fdcut">FDCut</code></td>
<td>


<p>Numeric, the cutoff for fold change, can be ignored if not using fold change.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>For a given expression profile of two conditions, lncPath package provide two method to find differentially expressed genes: t-text and fold change. The row of the expression profile should be gene IDs and the column of the expression profile should be names of samples. Samples should be under two conditions and the label should be given as 0 and 1. For t-test, fold change and SAM, different threshold can be set for significant differentially expressed genes.  
</p>


<h3>Value</h3>






<p>A vector of strings, the IDs of differentially expressed genes.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.
Profile &lt;- getExampleData("Profile")
Labels &lt;- getExampleData("Labels")

SigGenes &lt;- findSigGenes(Profile, Labels)
head(SigGenes)

</code></pre>

<hr>
<h2 id='geneSetDetail'>

Gain insight into the detail of the genes in a certain pathway
</h2><span id='topic+geneSetDetail'></span>

<h3>Description</h3>


<p>Gain insight into the detail of the genes in a certain pathway, inculding the ranks, weights and cummulative running scores of each gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geneSetDetail(Result, Name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="geneSetDetail_+3A_result">Result</code></td>
<td>


<p>A lncPath object come from the lncPath function.
</p>
</td></tr>
<tr><td><code id="geneSetDetail_+3A_name">Name</code></td>
<td>


<p>A string, the name of the pathway to be print.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>List all the genes of pathways ranked by the weights. The table also contains the gene name, the rank of genes in the whole gene list, the cumulative ES score and whether the gene is in the core gene sets which contribute to the score of the pathway.
</p>


<h3>Value</h3>






<p>A data frame, the rows are gene names and the columns are detail of genes including gene name, rank, weight, cumulative ES score and core erichment.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

Result &lt;- getExampleData("Result")
Detail &lt;- geneSetDetail(Result, "KEGG_RIBOSOME")
head(Detail)

</code></pre>

<hr>
<h2 id='getExampleData'>

Get the example data
</h2><span id='topic+getExampleData'></span>

<h3>Description</h3>


<p>Get the example data of LncPath package for litte trials.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getExampleData(ExampleData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getExampleData_+3A_exampledata">ExampleData</code></td>
<td>


<p>A character, should be one of &quot;SigLncs&quot;, &quot;ExampleNet&quot;, &quot;Labels&quot;, &quot;Profile&quot;, &quot;Result&quot; and &quot;Table&quot;.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>The function getExampleData(ExampleData = &quot;SigLncs&quot;) obtains a vector of lncRNAs confirmed to be related with breast cancer.
The function getExampleData(ExampleData = &quot;Profile&quot;) obtains the expression profile as a data frame.
The function getExampleData(ExampleData = &quot;Labels&quot;) obtains a vector of 0/1s describing the class of samples in the expression profile.
The function getExampleData(ExampleData = &quot;Result&quot;) obtains a lncPath object come from the lncPath function.
The function getExampleData(ExampleData = &quot;Table&quot;) obtains a data frame as the summary of lncPath object.
The function getExampleData(ExampleData = &quot;ExampleNet&quot;) obtains a data frame as the edges of lncRNA-mRNA interaction net.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>

<hr>
<h2 id='getNet'>

Get the background lncRNA-mRNA interaction network
</h2><span id='topic+getNet'></span>

<h3>Description</h3>


<p>Get the background lncRNA-mRNA interaction network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNet()
</code></pre>


<h3>Details</h3>


<p>Get the background lncRNA-mRNA interaction network, it was built by intergrating an lncRNA-mRNA co-expression network and the protein-protein interaction network.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.
LncPathNet &lt;- getNet();
</code></pre>

<hr>
<h2 id='lncPath'>
Identify pathways synergisticly regulated by lncRNA sets.
</h2><span id='topic+lncPath'></span>

<h3>Description</h3>

<p>Identify pathways synergisticly regulated by lncRNA sets by combining the random walk strategy and weighted Kolmogorov-Smirnov statistic based on a huge lncRNA-mRNA interaction network.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lncPath(LncRNAList, Network, Weighted = TRUE, PathwayDataSet = "KEGG", 
	minPathSize = 15, maxPathSize = 500, nperm = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lncPath_+3A_lncrnalist">LncRNAList</code></td>
<td>


<p>A character vector, contains the user interested lncRNAs, the ID of lncRNAs should be the Ensembl ID .
</p>
</td></tr>
<tr><td><code id="lncPath_+3A_network">Network</code></td>
<td>


<p>A dataframe with two columns, describing the edges of the network to perform the random walk.
</p>
</td></tr>
<tr><td><code id="lncPath_+3A_weighted">Weighted</code></td>
<td>


<p>Logical, tell if a weighted analysis to be performed, see detail.
</p>
</td></tr>
<tr><td><code id="lncPath_+3A_pathwaydataset">PathwayDataSet</code></td>
<td>


<p>A character, tells which pathway database is to be used, should be one of &quot;KEGG&quot;, &quot;Reactome&quot; and &quot;BioCarta&quot;.
</p>
</td></tr>
<tr><td><code id="lncPath_+3A_minpathsize">minPathSize</code></td>
<td>


<p>An integer, the lower limit of the mapped genes in pathway.
</p>
</td></tr>
<tr><td><code id="lncPath_+3A_maxpathsize">maxPathSize</code></td>
<td>


<p>An integer, the upper limit of the mapped genes in pathway.
</p>
</td></tr>
<tr><td><code id="lncPath_+3A_nperm">nperm</code></td>
<td>


<p>An integer, how manny times of perturbation to be performed in the perturbation analysis.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>lncPath is the main function of lncPath package, it takes a list of interested lncRNAs and a lncRNA-mRNA interaction network as input. Then it maps the lncRNAs into the lncRNA-mRNA interaction network as seed nodes and performs a random walk strategy to evaluate the rate of noedes effected by the seed nodes. A weighted Kolmogorov-Smirnov statistic was finnally used to evaluate the pathways related to the lncRNA sets.
If the Weighted parameter is set to TRUE, the scores of mRNAs generated from random walk will be treated as the weight in Kolmogorov-Smirnov statistic.If the Weighted parameter is set to FALSE, only the ranks of mRNAs will be taken into consideration.
Now three pathway data sets are surpported, includeing the KEGG, Reactome and BioCarta. And pathways with number of genes out of the limit will be filtered.
</p>


<h3>Value</h3>






<p>A lncPath object, containing the details of each pathways: pathway ID, pathway name, number of genes, gene names, score of genes etc.
It can be summarized by function by function lncPath2Table and can be visualized by function plotRunningES.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.
## get example data
SigLncs &lt;- getExampleData("SigLncs")
head(SigLncs)

ExampleNet &lt;- getExampleData("ExampleNet")
head(ExampleNet)

##run lncPath
Result &lt;- lncPath(SigLncs, ExampleNet,  Weighted = TRUE, PathwayDataSet = "KEGG", nperm = 100,
 minPathSize = 0, maxPathSize = 500)

## Print to table
Table &lt;- lncPath2Table(Result)
head(Table)

</code></pre>

<hr>
<h2 id='LncPath-internal'>LncPath internal functions</h2><span id='topic+initializeLncPathEnvir'></span><span id='topic+Getenvir'></span><span id='topic+rw'></span><span id='topic+RandomWalk2igraph'></span><span id='topic+lncPath.HeatMapPlot'></span><span id='topic+lncPath.EnrichmentScore'></span><span id='topic+lncPath.EnrichmentScore2'></span><span id='topic+ESNorm'></span><span id='topic+calT'></span><span id='topic+calFD'></span>

<h3>Description</h3>

<p>These functions internally called by other functions in the package <code>LncPath</code> and not meant to be called by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
    initializeLncPathEnvir()
    rw(W, p0, gamma)
    RandomWalk2igraph(igraphM, VertexWeight, EdgeWeight = TRUE, gamma = 0.7)
    lncPath.HeatMapPlot(V, row.names = F, col.labels, col.classes, 
	col.names = F, main = " ", xlab = " ", ylab = " ")
    lncPath.EnrichmentScore(gene.list, gene.set, weighted.score.type = 1, 
	correl.vector = NULL)
    lncPath.EnrichmentScore2(gene.list, gene.set, weighted.score.type = 1, 
	correl.vector = NULL)
    ESNorm(ESs)
    calT(inData, classLabel)
    calFD(inData, classLabel)

</code></pre>


<h3>Author(s)</h3>

<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;</p>

<hr>
<h2 id='lncPath2Table'>

Simplify the lncPath object into table 
</h2><span id='topic+lncPath2Table'></span>

<h3>Description</h3>


<p>Simplify the LncPath object into a data frame, which discribes the detail imformation of each pathway.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lncPath2Table(Result)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lncPath2Table_+3A_result">Result</code></td>
<td>


<p>The lncPath object come from the lncPath function.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>The lncPath object come from the lncPath function may be too complicated for user to view. This function can simplify it into a data frame. Each row of the data frame describe the detail of one pathway, including informations of pathway name, number of genes in the pathway, enrichment scores, normalized enrichment scores, p value and false discovery rate.
</p>


<h3>Value</h3>






<p>A data frame, rows are pathways and columns are details of each pathway.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

## The function is currently defined as
Result &lt;- getExampleData("Result")
Table &lt;- lncPath2Table(Result)
head(Table)

</code></pre>

<hr>
<h2 id='LncPathEnvir'>The variables in the environment variable <code>LncPathEnvir</code> of the system</h2><span id='topic+LncPathEnvir'></span>

<h3>Description</h3>

<p>The variables in the environment variable <code>LncPathEnvir</code> of the system.
</p>


<h3>Format</h3>

<p>An environment variable</p>


<h3>Author(s)</h3>

<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;</p>

<hr>
<h2 id='plotRunningES'>

Visualize the Kolmogorov-Smirnov running score of pathway
</h2><span id='topic+plotRunningES'></span>

<h3>Description</h3>


<p>Visualize the Kolmogorov-Smirnov running score of each gene of a certain pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotRunningES(Result, Name)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotRunningES_+3A_result">Result</code></td>
<td>


<p>A lncPath object come from the lncPath function.
</p>
</td></tr>
<tr><td><code id="plotRunningES_+3A_name">Name</code></td>
<td>


<p>A string, the name of the pathway to be plot.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>Plot the KS-statistic running score of certain pathway. The plot has three sections, the top section is a curve describes the cumulative ES score of pathway through all coding genes. The middle section contains signals telling which gene is in the pathway. The bottom section describes the weight distribution of genes. 
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.

Result &lt;- getExampleData("Result")
plotRunningES(Result, "KEGG_RIBOSOME")

</code></pre>

<hr>
<h2 id='printSignifResult'>

Output the details of significant pathways
</h2><span id='topic+printSignifResult'></span>

<h3>Description</h3>


<p>Export all of the significant pathways into a specified location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printSignifResult(Result, Threshold = 0.01, Path = ".", HeatPlot = FALSE, 
	PCExpr = "", Labels = "", Top = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="printSignifResult_+3A_result">Result</code></td>
<td>


<p>A lncPath object come from the lncPath function.
</p>
</td></tr>
<tr><td><code id="printSignifResult_+3A_threshold">Threshold</code></td>
<td>


<p>Numeric, the FDR threshold for selecting signifcant pathways.
</p>
</td></tr>
<tr><td><code id="printSignifResult_+3A_path">Path</code></td>
<td>


<p>String, the output directory.
</p>
</td></tr>
<tr><td><code id="printSignifResult_+3A_heatplot">HeatPlot</code></td>
<td>


<p>Logical, should the heatmaps be plotted.
</p>
</td></tr>
<tr><td><code id="printSignifResult_+3A_pcexpr">PCExpr</code></td>
<td>


<p>A data frame, represents the expression profile of genes, the rownames must be gene names, must be set if HeatPlot is TRUE.
</p>
</td></tr>
<tr><td><code id="printSignifResult_+3A_labels">Labels</code></td>
<td>


<p>A vector of 0 and 1, 0 indicates control and 1 indicates case.
</p>
</td></tr>
<tr><td><code id="printSignifResult_+3A_top">Top</code></td>
<td>


<p>An integer, indicates the number of the most significant pathways to be print, the Threshold will be ignored.
</p>
</td></tr>
</table>


<h3>Details</h3>


<p>For a result from the lncPath function, pritSignifResult will output all the details of significant pathways. Significant pathways can be defined by the threshold user submit or by ranks. The detail of pathways contains the running score plot , the gene sets detail and the heatmap of each pathway. For heatmap plot , the corresponding expression profile is needed. Considering a lot of files will be output, the output directory can be specified.
</p>


<h3>Author(s)</h3>


<p>Junwei Han &lt;hanjunwei1981@163.com&gt;, Zeguo Sun &lt;zeguo.sun@163.com&gt;
</p>


<h3>References</h3>


<p>Subramanian, A., Tamayo, P., Mootha, V.K., Mukherjee, S., Ebert, B.L., Gillette, M.A., Paulovich, A., Pomeroy, S.L., Golub, T.R., Lander, E.S. et al. (2005) Gene set enrichment analysis: a knowledgebased approach for interpreting genome-wide expression profiles. Proc Natl Acad Sci U S A, 102, 15545-15550.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##---- Should be DIRECTLY executable !! ----
##-- ==&gt;  Define data, use random,
##--	or do  help(data=index)  for the standard data sets.
## Not run: 
Result &lt;- getExampleData("Result")
Profile &lt;- getExampleData("Profile")
Labels &lt;- getExampleData("Labels")
dir.create("Signif")
SignifReport(Result, Threshold = 0.01, Path = "Signif", HeatPlot = TRUE, Profile, Labels, Top = 30)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
