<!DOCTYPE html><html><head><title>Help for package loadings</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {loadings}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#antibiotic'><p>Metagenome and metabolome data from antibiotic-treated mouse feces.</p></a></li>
<li><a href='#covid19'><p>Proteome and metabolome data of COVID-19 serum samples.</p></a></li>
<li><a href='#fasting'><p>Metabolome analysis of mouse liver samples in fasting conditions</p></a></li>
<li><a href='#greentea'><p>Metabolome analysis of Japanese green tea</p></a></li>
<li><a href='#multipls_geigen'><p>Multiset PLS : Multiset partial least squares</p></a></li>
<li><a href='#multipls_loading'><p>Multiset partial least squares loading</p></a></li>
<li><a href='#multipls_rog'><p>Multiset PLS-ROG : Multiset partial least squares with rank order of groups</p></a></li>
<li><a href='#multiplsrog_loading'><p>Multiset partial least squares rank order of groups loading</p></a></li>
<li><a href='#one_kpca'><p>One-sided kernel principal component analysis</p></a></li>
<li><a href='#onekpca_loading'><p>One-sided kernel principal component loading</p></a></li>
<li><a href='#os_pca'><p>Orthogonal smoothed principal component analysis</p></a></li>
<li><a href='#ospca_loading'><p>Orthogonal smoothed principal component loading</p></a></li>
<li><a href='#pca_loading'><p>Principal component loading</p></a></li>
<li><a href='#pls_da'><p>Partial least squares discriminant analysis (PLS-DA)</p></a></li>
<li><a href='#pls_loading'><p>Partial least squares loading</p></a></li>
<li><a href='#pls_rog'><p>PLS-ROG : Partial least squares with rank order of groups</p></a></li>
<li><a href='#pls_svd'><p>Partial least squares</p></a></li>
<li><a href='#plsda_loading'><p>Partial least squares discriminant analysis (PLS-DA) loading</p></a></li>
<li><a href='#plsrog_loading'><p>Partial least squares rank order of groups loading</p></a></li>
<li><a href='#rcca_da'><p>Regularized canonical correlation analysis for discriminant analysis (RCCA-DA)</p></a></li>
<li><a href='#rcca_loading'><p>Regularized canonical correlation analysis (RCCA-DA) loading</p></a></li>
<li><a href='#turnover'><p>Metabolomic Turnover analysis of the yeast culture mediums</p></a></li>
<li><a href='#unsv_multipls'><p>Unsupervised multiset partial least squares</p></a></li>
<li><a href='#unsvmultipls_loading'><p>Unsupervised multiset partial least squares loading</p></a></li>
<li><a href='#whhl'><p>Metabolome analysis of plasma, liver, heart and brain samples from rabbits with hyperlipidemia.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Loadings for Principal Component Analysis and Partial Least
Squares</td>
</tr>
<tr>
<td>Version:</td>
<td>0.5.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Hiroyuki Yamamoto</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hiroyuki Yamamoto &lt;h.yama2396@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Computing statistical hypothesis testing for loading in principal component analysis (PCA) (Yamamoto, H. et al. (2014) &lt;<a href="https://doi.org/10.1186%2F1471-2105-15-51">doi:10.1186/1471-2105-15-51</a>&gt;), orthogonal smoothed PCA (OS-PCA) (Yamamoto, H. et al. (2021) &lt;<a href="https://doi.org/10.3390%2Fmetabo11030149">doi:10.3390/metabo11030149</a>&gt;), one-sided kernel PCA (Yamamoto, H. (2023) &lt;<a href="https://doi.org/10.51094%2Fjxiv.262">doi:10.51094/jxiv.262</a>&gt;), partial least squares (PLS) and PLS discriminant analysis (PLS-DA) (Yamamoto, H. et al. (2009) &lt;<a href="https://doi.org/10.1016%2Fj.chemolab.2009.05.006">doi:10.1016/j.chemolab.2009.05.006</a>&gt;), PLS with rank order of groups (PLS-ROG) (Yamamoto, H. (2017) &lt;<a href="https://doi.org/10.1002%2Fcem.2883">doi:10.1002/cem.2883</a>&gt;), regularized canonical correlation analysis discriminant analysis (RCCA-DA) (Yamamoto, H. et al. (2008) &lt;<a href="https://doi.org/10.1016%2Fj.bej.2007.12.009">doi:10.1016/j.bej.2007.12.009</a>&gt;), multiset PLS and PLS-ROG (Yamamoto, H. (2022) &lt;<a href="https://doi.org/10.1101%2F2022.08.30.505949">doi:10.1101/2022.08.30.505949</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>geigen</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-01 03:45:03 UTC; hyama</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-01 04:30:11 UTC</td>
</tr>
</table>
<hr>
<h2 id='antibiotic'>Metagenome and metabolome data from antibiotic-treated mouse feces.</h2><span id='topic+antibiotic'></span>

<h3>Description</h3>

<p>This is metagenome and metabolome data of antibiotic-treated mouse fecal samples [Y. Wakita, et al, BMC Microbiology, 18:188 (2018)]. This dataset includes six groups, comprising three types (Clea, CR, and Slc) of control mice and their antibiotic-treated counterparts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(antibiotic)
</code></pre>


<h3>Format</h3>

<p>The list object &quot;antibiotic&quot; contains the following elements:
</p>
<p>data$X$microbiome$dist: Weighted UniFrac distance matrix for the metagenome.
</p>
<p>data$X$microbiome$comp: Composition data for the metagenome.
</p>
<p>data$X$metabolome : Metabolome data.
</p>
<p>data$Y: A dummy matrix that includes group information with 0 or 1 in each column.
</p>
<p>annotation$metabolome : Compound annotation for metabolome data.
</p>


<h3>Source</h3>

<p>Y. Wakita, et al, Taxonomic classification for microbiome analysis, which correlates well with the metabolite milieu of the gut, BMC microbiology, 18:188 (2018)
</p>

<hr>
<h2 id='covid19'>Proteome and metabolome data of COVID-19 serum samples.</h2><span id='topic+covid19'></span>

<h3>Description</h3>

<p>This is proteome and metabolome data of COVID-19 serum samples [B. Shen et al, Cell. 182 (2020) 59-72.e15.]. This dataset includes three groups of healthy control, mild and severe COVD-19.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(covid19)
</code></pre>


<h3>Format</h3>

<p>The list object &quot;covid19&quot; contains the following elements:
</p>
<p>X : List of data matrix that include samples in each rows and protein or metabolite in each columns for proteomics and metabolomics.
</p>
<p>Y : Dummy matrix that include group information 0,1 in each columns.
</p>
<p>D : Example of differential matrix used in PLS-ROG or multiset PLS-ROG.
</p>
<p>tau : Example of strength parameter matrix used in mutliset PLS or multiset PLS-ROG. All weights were set as the same value of 0.1.
</p>
<p>K : List of kernel matrix by hyperbolic tangent kernel (scale=1, offset=1).
</p>


<h3>Source</h3>

<p>B. Shen, et al, Proteomic and Metabolomic Characterization of COVID-19 Patient Sera, Cell. 182 (2020) 59-72.e15.
</p>

<hr>
<h2 id='fasting'>Metabolome analysis of mouse liver samples in fasting conditions</h2><span id='topic+fasting'></span>

<h3>Description</h3>

<p>This data includes metabolome data of liver samples from normal and 12 hour fasted mice measured by capillary electrophoresis time-of-flight mass spectrometry. The data matrix contains 10 samples (5 samples in each groups) and 282 metabolites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fasting)
</code></pre>


<h3>Format</h3>

<p>The list object &quot;fasting&quot; contains the following elements:
</p>
<p>X : 	Data matrix that include metabolites in each columns.
</p>
<p>Y :   Dummy matrix that include group information 0,1 in each columns.
</p>


<h3>Source</h3>

<p>Yamamoto H., Fujimori T., Sato H., Ishikawa G., Kami K., Ohashi Y. (2014). &quot;Statistical hypothesis testing of factor loading in principal component analysis and its application to metabolite set enrichment analysis&quot;. BMC Bioinformatics, (2014) 15(1):51.
</p>

<hr>
<h2 id='greentea'>Metabolome analysis of Japanese green tea</h2><span id='topic+greentea'></span>

<h3>Description</h3>

<p>This includes metabolome data of Japanese green tea measured by gas chromatgraphy mass spectrometry. The data matrix contains 30 samples and 225 metabolites.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(greentea)
</code></pre>


<h3>Format</h3>

<p>The list object &quot;greentea&quot; contains the following elements:
</p>
<p>X : Data matrix that include metabolites in each columns.
</p>
<p>Y : Dummy matrix that include group information 0,1 in each columns.
</p>
<p>D : Differential matrix (Second order).
</p>
<p>M : Averaging matrix for repeated data.
</p>
<p>y : Ranking of taste in competitive exhibition.
</p>


<h3>Source</h3>

<p>Statistics in Microsoft Excel ( http://prime.psc.riken.jp/compms/others/main.html#Statistics )
</p>
<p>Pongsuwan W., Fukusaki E., Bamba T., Yonetani T., Yamahara A.T., Kobayashi A. Prediction of Japanese Green Tea Ranking by Gas Chromatography/Mass Spectrometry-Based Hydrophilic Metabolite Fingerprinting. J. Agric. Food Chem. 2007;55:231-236.
</p>

<hr>
<h2 id='multipls_geigen'>Multiset PLS : Multiset partial least squares</h2><span id='topic+multipls_geigen'></span>

<h3>Description</h3>

<p>This function performs Multiset partial least squares (Multiset PLS).
In this function, data matrix is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multipls_geigen(X,Y,tau)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multipls_geigen_+3A_x">X</code></td>
<td>

<p>List of data matrix that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="multipls_geigen_+3A_y">Y</code></td>
<td>

<p>Dummy matrix that include group information 0,1 in each columns.
</p>
</td></tr>
<tr><td><code id="multipls_geigen_+3A_tau">tau</code></td>
<td>

<p>Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diagonal elements of matrix tau must be 0.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A list of matrix with Multiset PLS coefficients for the explanatory variables in each column for each dataset
</p>
<p>T : A list of matrix with Multiset PLS scores for the explanatory variables in each column for each dataset
</p>
<p>Q : A matrix with Multiset PLS coefficients for the response variable in each column
</p>
<p>U : A matrix with Multiset PLS scores for the response variable in each column
</p>
<p>tau : Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information (same as input value).
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2022) Multiset partial least squares with rank order of groups for integrating multi-omics data, bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X
Y &lt;- whhl$Y
tau &lt;- whhl$tau

multipls &lt;- multipls_geigen(X,Y,tau)

</code></pre>

<hr>
<h2 id='multipls_loading'>Multiset partial least squares loading</h2><span id='topic+multipls_loading'></span>

<h3>Description</h3>

<p>This function computes multiset partial least squares (PLS) loading from the result of the &quot;multipls_geigen&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multipls_loading(multipls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multipls_loading_+3A_multipls">multipls</code></td>
<td>

<p>The following variables (P,T,Q and U) are included in the multipls object.
</p>
<p>P : A list of matrix with Multiset PLS coefficients for the explanatory variables in each column for each dataset
</p>
<p>T : A list of matrix with Multiset PLS scores for the explanatory variables in each column for each dataset
</p>
<p>Q : A matrix with Multiset PLS coefficients for the response variable in each column
</p>
<p>U : A matrix with Multiset PLS scores for the response variable in each column
</p>
<p>tau : Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The multiset PLS loading and the p-value by statistical hypothesis testing is added to the multipls object returned by the &quot;multipls_geigen&quot; function in this package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>Score : A list of matrix with Multiset PLS score for auxiliary variable in each column for each dataset
</p>
<p>R : A list of matrix with Multiset PLS loadings in each column for each dataset
</p>
<p>p.value : A list of matrix with p-value of multiset PLS loading in each column for each dataset
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2022) Multiset partial least squares with rank order of groups for integrating multi-omics data, bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X
Y &lt;- whhl$Y
tau &lt;- whhl$tau

multipls &lt;- multipls_geigen(X,Y,tau)

multipls &lt;- multipls_loading(multipls)
multipls$loading$R
multipls$loading$p.value
</code></pre>

<hr>
<h2 id='multipls_rog'>Multiset PLS-ROG : Multiset partial least squares with rank order of groups</h2><span id='topic+multipls_rog'></span>

<h3>Description</h3>

<p>This function performs multiset partial least squares with rank order of groups (Multiset PLS-ROG).
In this function, data matrix is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multipls_rog(X,Y,tau,D,kappa)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multipls_rog_+3A_x">X</code></td>
<td>

<p>List of data matrix that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="multipls_rog_+3A_y">Y</code></td>
<td>

<p>Dummy matrix that include group information 0,1 in each columns.
</p>
</td></tr>
<tr><td><code id="multipls_rog_+3A_tau">tau</code></td>
<td>

<p>Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information.
</p>
</td></tr>
<tr><td><code id="multipls_rog_+3A_d">D</code></td>
<td>

<p>Differential matrix.
</p>
</td></tr>
<tr><td><code id="multipls_rog_+3A_kappa">kappa</code></td>
<td>

<p>The smoothing parameter (default : 0.999).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diagonal elements of matrix tau must be 0.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A list of matrix with Multiset PLS-ROG coefficients for the explanatory variables in each column for each dataset
</p>
<p>T : A list of matrix with Multiset PLS-ROG scores for the explanatory variables in each column for each dataset
</p>
<p>Q : A matrix with Multiset PLS-ROG coefficients for the response variable in each column
</p>
<p>U : A matrix with Multiset PLS-ROG scores for the response variable in each column
</p>
<p>tau : Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information (same as input value).
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2022) Multiset partial least squares with rank order of groups for integrating multi-omics data, bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X
Y &lt;- whhl$Y
D &lt;- whhl$D
tau &lt;- whhl$tau

multiplsrog &lt;- multipls_rog(X,Y,tau,D)
# multiplsrog &lt;- multipls_rog(X,Y,tau,D, kappa=0.999)

</code></pre>

<hr>
<h2 id='multiplsrog_loading'>Multiset partial least squares rank order of groups loading</h2><span id='topic+multiplsrog_loading'></span>

<h3>Description</h3>

<p>This function computes multiset partial least squares rank order of groups (PLS-ROG) loading from the result of the &quot;multipls_rog&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplsrog_loading(multiplsrog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplsrog_loading_+3A_multiplsrog">multiplsrog</code></td>
<td>

<p>The following variables (P,T,Q and U) are included in the multiplsrog object.
</p>
<p>P : A list of matrix with Multiset PLS-ROG coefficients for the explanatory variables in each column for each dataset
</p>
<p>T : A list of matrix with Multiset PLS-ROG scores for the explanatory variables in each column for each dataset
</p>
<p>Q : A matrix with Multiset PLS-ROG coefficients for the response variable in each column
</p>
<p>U : A matrix with Multiset PLS-ROG scores for the response variable in each column
</p>
<p>tau : Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The multiset PLS-ROG loading and the p-value by statistical hypothesis testing is added to the multiplsrog object returned by the &quot;multipls_rog&quot; function in this package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>Score : A list of matrix with Multiset PLS score for auxiliary variable in each column for each dataset
</p>
<p>R : A list of matrix with Multiset PLS-ROG loadings in each column for each dataset
</p>
<p>p.value : A list of matrix with p-value of multiset PLS-ROG loading in each column for each dataset
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2022) Multiset partial least squares with rank order of groups for integrating multi-omics data, bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X
Y &lt;- whhl$Y
D &lt;- whhl$D
tau &lt;- whhl$tau

multiplsrog &lt;- multipls_rog(X,Y,tau,D)
# multiplsrog &lt;- multipls_rog(X,Y,tau,D, kappa=0.999)

multiplsrog &lt;- multiplsrog_loading(multiplsrog)
multiplsrog$loading$R
multiplsrog$loading$p.value
</code></pre>

<hr>
<h2 id='one_kpca'>One-sided kernel principal component analysis</h2><span id='topic+one_kpca'></span>

<h3>Description</h3>

<p>This function performs one-sided kernel principal component analysis (one-sided KPCA).
In this function, data matrix is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>one_kpca(X,K)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="one_kpca_+3A_x">X</code></td>
<td>

<p>Data matrix that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="one_kpca_+3A_k">K</code></td>
<td>

<p>Kernel matrix computed from the data matrix X.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The kernel matrix K, which is the argument of the one_kpca function, must be centered.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A matrix with one-sided KPCA loading in each column
</p>
<p>T : A matrix with one-sided KPCA score for linear side in each column
</p>
<p>U : A matrix with one-sied KPCA score for nonlinear side in each column
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2023) One-sided Kernel Principal Component Analysis, Jxiv, &lt;doi:10.51094/jxiv.262&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COVID-19 metabolome data
data(covid19)

X &lt;- covid19$X$metabolomics
K &lt;- covid19$K$metabolomics

onekpca &lt;- one_kpca(X,K)

</code></pre>

<hr>
<h2 id='onekpca_loading'>One-sided kernel principal component loading</h2><span id='topic+onekpca_loading'></span>

<h3>Description</h3>

<p>This function computes one-sided kernel principal component (one-sided KPC) loading from the result of the &quot;one_kpca&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>onekpca_loading(onekpca)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="onekpca_loading_+3A_onekpca">onekpca</code></td>
<td>

<p>The following variables (P,T and U) are included in the onekpca object.
</p>
<p>P : A matrix with one-sided KPCA loading in each column
</p>
<p>T : A matrix with one-sided KPCA score for linear side in each column
</p>
<p>U : A matrix with one-sied KPCA score for nonlinear side in each column
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The one-sided KPC loading and the p-value by statistical hypothesis testing is added to the onekpca object returned by the &quot;one_kpca&quot; function.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : one-sided KPC loading (Correlation coefficient between one-sided KPC score of nonlinear variable (kernel matrix) and each variables in data matrix.)
</p>
<p>p.value : p-value of one-sided KPC loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2023) One-sided Kernel Principal Component Analysis, Jxiv, &lt;doi:10.51094/jxiv.262&gt;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># COVID-19 metabolome data
data(covid19)

X &lt;- covid19$X$metabolomics
K &lt;- covid19$K$metabolomics

onekpca &lt;- one_kpca(X,K)

onekpca &lt;- onekpca_loading(onekpca)

onekpca$loading$R
onekpca$loading$p.value

</code></pre>

<hr>
<h2 id='os_pca'>Orthogonal smoothed principal component analysis</h2><span id='topic+os_pca'></span>

<h3>Description</h3>

<p>This function performs orthogonal smoothed principal component analysis (OS-PCA).
In this function, data matrix is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>os_pca(X,D,kappa,M)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="os_pca_+3A_x">X</code></td>
<td>

<p>Data matrix that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="os_pca_+3A_d">D</code></td>
<td>

<p>Differential matrix.
</p>
</td></tr>
<tr><td><code id="os_pca_+3A_kappa">kappa</code></td>
<td>

<p>The smoothing parameter (default : 0.999).
</p>
</td></tr>
<tr><td><code id="os_pca_+3A_m">M</code></td>
<td>

<p>Averaging matrix for repeated data (default : Identity matrix).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The kappa represents the degree of smoothing. The value of kappa increases, the strength of the smoothing increases.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A matrix with OS-PC loading in each column
</p>
<p>T : A matrix with OS-PC score in each column
</p>
<p>MT : A matrix with averaging OS-PC score for repeated data in each column
(If not for repeated data, the matrix MT equals to the matrix T)
</p>
<p>Q : A matrix with OS-PC loading for auxiliary variable in each column
</p>
<p>U : A matrix with OS-PC score for auxiliary variable in each column
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H., Nakayama Y., Tsugawa H. (2021) OS-PCA: Orthogonal Smoothed Principal Component Analysis Applied to Metabolome Data, Metabolites, 11(3):149.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># metabolic turnover data
data(turnover)

X &lt;- turnover$X
D &lt;- turnover$D

ospca &lt;- os_pca(X,D,0.999)

# metabolome data
data(greentea)

X &lt;- greentea$X
D &lt;- greentea$D
M &lt;- greentea$M

ospca &lt;- os_pca(X,D,0.1,M)

</code></pre>

<hr>
<h2 id='ospca_loading'>Orthogonal smoothed principal component loading</h2><span id='topic+ospca_loading'></span>

<h3>Description</h3>

<p>This function computes orthogonals smoothed principal component (OS-PC) loading from the result of the &quot;os_pca&quot; function.
This is also wrapper function of pls_loading function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ospca_loading(ospca)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ospca_loading_+3A_ospca">ospca</code></td>
<td>

<p>The following variables (P,T,Q and U) are included in the ospca object.
</p>
<p>P : A matrix with OS-PC loading in each column
</p>
<p>T : A matrix with OS-PC score in each column
</p>
<p>MT : A matrix with averaging OS-PC score for repeated data in each column
(If not for repeated data, the matrix MT equals to the matrix T)
</p>
<p>Q : A matrix with OS-PC loading for auxiliary variable in each column
</p>
<p>U : A matrix with OS-PC score for auxiliary variable in each column
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The OS-PC loading and the p-value by statistical hypothesis testing is added to the ospca object returned by the &quot;os-pca&quot; function.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : OS-PCA loading (Correlation coefficient between OS-PC score of auxiliary variable and each variables in data matrix.)
</p>
<p>p.value : p-value of OS-PCA loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H., Nakayama Y., Tsugawa H. (2021) OS-PCA: Orthogonal Smoothed Principal Component Analysis Applied to Metabolome Data, Metabolites, 11(3):149.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># metabolic turnover data
data(turnover)
X &lt;- turnover$X
D &lt;- turnover$D

ospca &lt;- os_pca(X,D,0.999)

ospca &lt;- ospca_loading(ospca)
ospca$loading$R
ospca$loading$p.value

# metabolome data

data(greentea)
X &lt;- greentea$X
D &lt;- greentea$D
M &lt;- greentea$M

ospca &lt;- os_pca(X,D,0.1,M)

ospca &lt;- ospca_loading(ospca)
ospca$loading$R
ospca$loading$p.value
</code></pre>

<hr>
<h2 id='pca_loading'>Principal component loading</h2><span id='topic+pca_loading'></span>

<h3>Description</h3>

<p>This function computes principal component (PC) loading from the result of the &quot;prcomp&quot; function.
(The &quot;princomp&quot; function is not supported. For &quot;princomp&quot; function, the &quot;loadings&quot; function in stats package should be used.)
In this function, data matrix is should be scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_loading(pca)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_loading_+3A_pca">pca</code></td>
<td>

<p>The return object of the &quot;prcomp&quot; function.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PC loading and the p-value by statistical hypothesis testing is added to the pca object returned by the &quot;prcomp&quot; function.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : PC loading (Correlation coefficient between PC score and each variables in data matrix.)
</p>
<p>p.value : p-value of PC loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H., Fujimori T., Sato H., Ishikawa G., Kami K., Ohashi Y. (2014). &quot;Statistical hypothesis testing of factor loading in principal component analysis and its application to metabolite set enrichment analysis&quot;. BMC Bioinformatics, (2014) 15(1):51.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(fasting)
X &lt;- fasting$X

pca &lt;- prcomp(X, scale=TRUE)

pca &lt;- pca_loading(pca)
pca$loading$R # PC loading
pca$loading$p.value # p-value
</code></pre>

<hr>
<h2 id='pls_da'>Partial least squares discriminant analysis (PLS-DA)</h2><span id='topic+pls_da'></span>

<h3>Description</h3>

<p>This function performs partial least squares discriminant analysis (PLS-DA). In this function, data matrix for explanatory variable is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pls_da(X,Y,k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pls_da_+3A_x">X</code></td>
<td>

<p>Data matrix of explanatory variables that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="pls_da_+3A_y">Y</code></td>
<td>

<p>Dummy matrix that include group information 0,1 in each columns.
</p>
</td></tr>
<tr><td><code id="pls_da_+3A_k">k</code></td>
<td>

<p>Number of components.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates PLS-DA. For PLS, use the 'pls_svd' function for PLS.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P: A matrix containing the PLS-DA loadings for each explanatory variable in the columns, before transformation.
</p>
<p>T : A matrix with PLS-DA score for explanatory variable in each column
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. et al., Dimensionality reduction for metabolome data using PCA, PLS, OPLS, and RFDA with differential penalties to latent variables&quot;, Chemom. Intell. Lab. Syst., 98 (2009)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y

plsda &lt;- pls_da(X,Y,2)
</code></pre>

<hr>
<h2 id='pls_loading'>Partial least squares loading</h2><span id='topic+pls_loading'></span>

<h3>Description</h3>

<p>This function computes partial least squares (PLS) loading from the result of the &quot;pls_svd&quot; function.
PLS loading can also be computed from the result of the &quot;pls_eigen&quot; function in chemometrics package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pls_loading(pls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pls_loading_+3A_pls">pls</code></td>
<td>

<p>The following variables (P,T,Q and U) are included in the pls object.
</p>
<p>P : A matrix with PLS loading for explanatory variable in each column
</p>
<p>T : A matrix with PLS score for explanatory variable in each column
</p>
<p>Q : A matrix with PLS loading for response variable in each column
</p>
<p>U : A matrix with PLS score for response variable in each column
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PLS loading and the p-value by statistical hypothesis testing is added to the pls object returned by the &quot;pls_svd&quot; function in this package or &quot;pls_eigen&quot; function in chemometrics package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : PLS loading (Correlation coefficient between PLS score of response variable and each variables in data matrix.)
</p>
<p>p.value : p-value of PLS loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. (2017) PLS-ROG: Partial least squares with rank order of groups., Journal of Chemometrics, 31(3) (2017) e2883.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y

pls &lt;- pls_svd(X,Y)

pls &lt;- pls_loading(pls)
pls$loading$R
pls$loading$p.value

</code></pre>

<hr>
<h2 id='pls_rog'>PLS-ROG : Partial least squares with rank order of groups</h2><span id='topic+pls_rog'></span>

<h3>Description</h3>

<p>This function performs partial least squares with rank order of groups (PLS-ROG).
In this function, data matrix is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pls_rog(X,Y,D,kappa)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pls_rog_+3A_x">X</code></td>
<td>

<p>Data matrix of explanatory variables that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="pls_rog_+3A_y">Y</code></td>
<td>

<p>Dummy matrix that include group information 0,1 in each columns.
</p>
</td></tr>
<tr><td><code id="pls_rog_+3A_d">D</code></td>
<td>

<p>Differential matrix.
</p>
</td></tr>
<tr><td><code id="pls_rog_+3A_kappa">kappa</code></td>
<td>

<p>The smoothing parameter (default : kappa = 0.999).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The kappa represents the degree of smoothing. The value of kappa increases, the strength of the smoothing increases.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A matrix with PLS loading for explanatory variable in each column
</p>
<p>T : A matrix with PLS score for explanatory variable in each column
</p>
<p>Q : A matrix with PLS loading for response variable in each column
</p>
<p>U : A matrix with PLS score for response variable in each column
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. (2017) PLS-ROG: Partial least squares with rank order of groups., Journal of Chemometrics, 31(3) (2017) e2883.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y
D &lt;- whhl$D

plsrog &lt;- pls_rog(X,Y,D)
</code></pre>

<hr>
<h2 id='pls_svd'>Partial least squares</h2><span id='topic+pls_svd'></span>

<h3>Description</h3>

<p>This function  performs partial least squares. In this function, data matrix for explanatory variable is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pls_svd(X,Y)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pls_svd_+3A_x">X</code></td>
<td>

<p>Data matrix of explanatory variables that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="pls_svd_+3A_y">Y</code></td>
<td>

<p>Dummy matrix that include group information 0,1 in each columns.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is wrapper function of &quot;pls_rog&quot; fucntion that the smoothing parameter kappa=0.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A matrix with PLS loading for explanatory variable in each column
</p>
<p>T : A matrix with PLS score for explanatory variable in each column
</p>
<p>Q : A matrix with PLS loading for response variable in each column
</p>
<p>U : A matrix with PLS score for response variable in each column
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Barker, M. and Rayens, W. (2003) Partial Least Squares for Discrimination. Journal of Chemometrics, 17, 166-173.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y

pls &lt;- pls_svd(X,Y)
</code></pre>

<hr>
<h2 id='plsda_loading'>Partial least squares discriminant analysis (PLS-DA) loading</h2><span id='topic+plsda_loading'></span>

<h3>Description</h3>

<p>This function computes partial least squares discriminant analysis (PLS-DA) loading from the result of the 'pls_da' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plsda_loading(plsda)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plsda_loading_+3A_plsda">plsda</code></td>
<td>

<p>The following variables (P,T) are included in the plsda object.
</p>
<p>P: A matrix containing the PLS-DA loadings for each explanatory variable in the columns, before transformation.
</p>
<p>T : A matrix with PLS-DA score for explanatory variable in each column
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PLS-DA loading and the p-value by statistical hypothesis testing is added to the pls object returned by the 'pls_da' function in this package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : PLS-DA loading (Correlation coefficient between PLS-DA score of explanatory variable and each variables in data matrix.)
</p>
<p>p.value : p-value of PLS-DA loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. et al., Dimensionality reduction for metabolome data using PCA, PLS, OPLS, and RFDA with differential penalties to latent variables&quot;, Chemom. Intell. Lab. Syst., 98 (2009)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y
plsda &lt;- pls_da(X,Y,2)

plsda &lt;- plsda_loading(plsda)

plsda$loading$R
plsda$loading$p.value

</code></pre>

<hr>
<h2 id='plsrog_loading'>Partial least squares rank order of groups loading</h2><span id='topic+plsrog_loading'></span>

<h3>Description</h3>

<p>This function computes partial least squares rank rder of groups (PLS-ROG) loading from the result of the &quot;pls_rog&quot; function.
PLS-ROG loading can also be computed from the result of the &quot;pls_rog&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plsrog_loading(plsrog)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plsrog_loading_+3A_plsrog">plsrog</code></td>
<td>

<p>The following variables (P,T,Q and U) are included in the plsrog object.
</p>
<p>P : A matrix with PLS-ROG loading for explanatory variable in each column
</p>
<p>T : A matrix with PLS-ROG score for explanatory variable in each column
</p>
<p>Q : A matrix with PLS-ROG loading for response variable in each column
</p>
<p>U : A matrix with PLS-ROG score for response variable in each column
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PLS-ROG loading and the p-value by statistical hypothesis testing is added to the pls object returned by the &quot;pls_rog&quot; function in this package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : PLS-ROG loading (Correlation coefficient between PL-ROG score of response variable and each variables in data matrix.)
</p>
<p>p.value : p-value of PLS-ROG loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. (2017) PLS-ROG: Partial least squares with rank order of groups., Journal of Chemometrics, 31(3) (2017) e2883.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y
D &lt;- whhl$D

plsrog &lt;- pls_rog(X,Y,D)

plsrog &lt;- plsrog_loading(plsrog)
plsrog$loading$R
plsrog$loading$p.value

</code></pre>

<hr>
<h2 id='rcca_da'>Regularized canonical correlation analysis for discriminant analysis (RCCA-DA)</h2><span id='topic+rcca_da'></span>

<h3>Description</h3>

<p>This function performs regularized canonical correlation analysis for discriminant analysis (RCCA-DA). In this function, data matrix for explanatory variable is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rcca_da(X,Y,lambda,k)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rcca_da_+3A_x">X</code></td>
<td>

<p>Data matrix of explanatory variables that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="rcca_da_+3A_y">Y</code></td>
<td>

<p>Dummy matrix that include group information 0,1 in each columns.
</p>
</td></tr>
<tr><td><code id="rcca_da_+3A_lambda">lambda</code></td>
<td>

<p>The regularized parameter has a value in the range [0, 1), meaning it can be 0 but is less than 1.&quot;
</p>
</td></tr>
<tr><td><code id="rcca_da_+3A_k">k</code></td>
<td>

<p>Number of components.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>RCCA-DA is equivalent to Regularized Fisher discriminant analysis, theoretically.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P: A matrix containing the RCCA-DA loadings for each explanatory variable in the columns, before transformation.
</p>
<p>T : A matrix with RCCA-DA score for explanatory variable in each column
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. et al., Canonical correlation analysis for multivariate regression and its application to metabolic fingerprinting&quot;, Biochem. Eng. Journal, 40 (2008) 199-204.
</p>
<p>Yamamoto, H. et al., Dimensionality reduction for metabolome data using PCA, PLS, OPLS, and RFDA with differential penalties to latent variables&quot;, Chemom. Intell. Lab. Syst., 98 (2009)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y

rcca &lt;- rcca_da(X,Y,0.5,2)
</code></pre>

<hr>
<h2 id='rcca_loading'>Regularized canonical correlation analysis (RCCA-DA) loading</h2><span id='topic+rcca_loading'></span>

<h3>Description</h3>

<p>This function computes regularized canonical correlation analysis for discriminant analysis (RCCA-DA) loading from the result of the 'rcca_da' function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rcca_loading(rcca)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rcca_loading_+3A_rcca">rcca</code></td>
<td>

<p>The following variables (Wx,P,T,Wy and S) are included in the rcca object.
</p>
<p>Wx : A matrix with RCCA-DA weight for explanatory variable in each column
</p>
<p>P: A matrix containing the RCCA-DA loadings for each explanatory variable in the columns, before transformation.
</p>
<p>T : A matrix with RCCA-DA score for explanatory variable in each column
</p>
<p>Wy : A matrix with RCCA-DA weight for explanatory variable in each column
</p>
<p>S : A matrix with RCCA-DA score for response variable in each column
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The RCCA-DA loading and the p-value by statistical hypothesis testing is added to the pls object returned by the 'rcca_da' function in this package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>R : RCCA-DA loading (Correlation coefficient between PLS-DA score of explanatory variable and each variables in data matrix.)
</p>
<p>p.value : p-value of RCCA-DA loading
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto, H. et al., Canonical correlation analysis for multivariate regression and its application to metabolic fingerprinting&quot;, Biochem. Eng. Journal, 40 (2008) 199-204.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X$liver
Y &lt;- whhl$Y
rcca &lt;- rcca_da(X,Y,0.5,2)

rcca &lt;- rcca_loading(rcca)

rcca$loading$R
rcca$loading$p.value

</code></pre>

<hr>
<h2 id='turnover'>Metabolomic Turnover analysis of the yeast culture mediums</h2><span id='topic+turnover'></span>

<h3>Description</h3>

<p>This includes differential isotopomer ratio from metabolomic turnover analysis of yeast cultures by gas chromatgraphy mass spectrometry. The data matrix contains 11 samping time for three groups of different yeast strains and culture condition (33 samples total) and 60 peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(turnover)
</code></pre>


<h3>Format</h3>

<p>The list object &quot;turnover&quot; contains the following elements:
</p>
<p>X : Data matrix of explanatory variables that include variables in each columns.
</p>
<p>Y : Dummy matrix that include group information 0,1 in each columns.
</p>
<p>D : Differential matrix (Second order).
</p>
<p>annotation :  Compound annotation for metabolome data.
</p>


<h3>References</h3>

<p>Nakayama Y., Tamada Y., Tsugawa H., Bamba T., Fukusaki E. Novel Strategy for Non-Targeted Isotope-Assisted Metabolomics by Means of Metabolic Turnover and Multivariate Analysis. Metabolites. 2014;4:722-739.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(turnover)
</code></pre>

<hr>
<h2 id='unsv_multipls'>Unsupervised multiset partial least squares</h2><span id='topic+unsv_multipls'></span>

<h3>Description</h3>

<p>This function performs Unsupervised multiset partial least squares.
In this function, data matrix is automatically scaled to zero mean and unit variance (i.e. autoscaling) for each variables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unsv_multipls(X,tau)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unsv_multipls_+3A_x">X</code></td>
<td>

<p>List of data matrix that include variables in each columns.
</p>
</td></tr>
<tr><td><code id="unsv_multipls_+3A_tau">tau</code></td>
<td>

<p>Matrix for strength parameter of the connection between omics datasets or between omics dataset and group information.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Diagonal elements of matrix tau must be 0.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>P : A list of matrix with Multiset PLS coefficients in each column for each dataset
</p>
<p>T : A list of matrix with Multiset PLS scores in each column for each dataset
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2022) Multiset partial least squares with rank order of groups for integrating multi-omics data, bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X
tau &lt;- rbind(c(0,1/4,1/4,1/4),c(1/4,0,1/4,1/4),c(1/4,1/4,0,1/4),c(1/4,1/4,1/4,0))

unsvmpls &lt;- unsv_multipls(X,tau)

</code></pre>

<hr>
<h2 id='unsvmultipls_loading'>Unsupervised multiset partial least squares loading</h2><span id='topic+unsvmultipls_loading'></span>

<h3>Description</h3>

<p>This function computes unsupervised multiset partial least squares loading from the result of the &quot;unsv_multipls&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unsvmultipls_loading(unsvmpls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unsvmultipls_loading_+3A_unsvmpls">unsvmpls</code></td>
<td>

<p>The following variables (P,T) are included in the unsvmpls object.
</p>
<p>P : A list of matrix with Multiset PLS coefficients in each column for each dataset
</p>
<p>T : A list of matrix with Multiset PLS scores in each column for each dataset
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The unsupervised multiset PLS loading and the p-value by statistical hypothesis testing is added to the unsvmpls object returned by the &quot;unsv_multipls&quot; function in this package.
</p>


<h3>Value</h3>

<p>The return value is a list object that contains the following elements:
</p>
<p>Score : A list of matrix with Multiset PLS score for auxiliary variable in each column for each dataset
</p>
<p>R : A list of matrix with Unsupervised multiset PLS loadings in each column for each dataset
</p>
<p>p.value : A list of matrix with p-value of Unsupervised multiset PLS loading in each column for each dataset
</p>


<h3>Author(s)</h3>

<p>Hiroyuki Yamamoto
</p>


<h3>References</h3>

<p>Yamamoto H. (2022) Multiset partial least squares with rank order of groups for integrating multi-omics data, bioRxiv.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(whhl)
X &lt;- whhl$X
tau &lt;- rbind(c(0,1/4,1/4,1/4),c(1/4,0,1/4,1/4),c(1/4,1/4,0,1/4),c(1/4,1/4,1/4,0))

unsvmpls &lt;- unsv_multipls(X,tau)

unsvmpls &lt;- unsvmultipls_loading(unsvmpls)
unsvmpls$loading$R
unsvmpls$loading$p.value

</code></pre>

<hr>
<h2 id='whhl'>Metabolome analysis of plasma, liver, heart and brain samples from rabbits with hyperlipidemia.</h2><span id='topic+whhl'></span>

<h3>Description</h3>

<p>This is metabolome data from plasma, liver, heart and brain samples of normal, hyperlipidemic, and statin-treated rabbits measured by capillary electrophoresis time-of-flight mass spectrometry. The data matrix contains 9 samples (3 samples in each groups).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(whhl)
</code></pre>


<h3>Format</h3>

<p>The list object &quot;whhl&quot; contains the following elements:
</p>
<p>X : List of data matrix that include samples in each rows and metabolites in each columns for plasma, liver, heart and brain samples.
</p>
<p>Y : Dummy matrix that include group information 0,1 in each columns.
</p>
<p>D : Example of differential matrix used in PLS-ROG or multiset PLS-ROG.
</p>
<p>tau : Example of strength parameter matrix used in mutliset PLS or multiset PLS-ROG. All weights were set as the same value of 0.1.
</p>


<h3>Source</h3>

<p>Ooga, T., Sato, H., Nagashima, A., Sasaki, K., Tomita, M., Soga T. and Ohashi, Y. (2011) Metabolomic Anatomy of Animal Model Revealing Homeostatic Imbalances in Dyslipidemia. Y. Mol. BioSyst.,7, 1217-23.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
