<!DOCTYPE html><html><head><title>Help for package locuszoomr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {locuszoomr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eqtl_plot'><p>Locus eQTL plot</p></a></li>
<li><a href='#genetrack_ly'><p>Gene tracks using 'plotly'</p></a></li>
<li><a href='#genetracks'><p>Plot gene tracks</p></a></li>
<li><a href='#genetracks_grob'><p>Create gene tracks grob</p></a></li>
<li><a href='#gg_addgenes'><p>Add gene tracks to a ggplot2 plot</p></a></li>
<li><a href='#gg_genetracks'><p>Plot gene tracks</p></a></li>
<li><a href='#gg_scatter'><p>Locus scatter plot using ggplot2</p></a></li>
<li><a href='#line_plot'><p>Locus line plot</p></a></li>
<li><a href='#link_eqtl'><p>Obtain GTEx eQTL data via LDlinkR</p></a></li>
<li><a href='#link_LD'><p>Obtain LD at a locus from LDlink</p></a></li>
<li><a href='#link_recomb'><p>Query UCSC for Recombination data</p></a></li>
<li><a href='#locus'><p>Create locus object for plotting</p></a></li>
<li><a href='#locus_ggplot'><p>Locus plot using ggplot2</p></a></li>
<li><a href='#locus_plot'><p>Locus plot</p></a></li>
<li><a href='#locus_plotly'><p>Locus plotly</p></a></li>
<li><a href='#multi_layout'><p>Layout multiple locus plots</p></a></li>
<li><a href='#overlay_plot'><p>Plot overlaying eQTL and GWAS data</p></a></li>
<li><a href='#quick_peak'><p>Fast peak finder in GWAS data</p></a></li>
<li><a href='#scatter_plot'><p>Locus scatter plot</p></a></li>
<li><a href='#scatter_plotly'><p>Locus scatter plotly</p></a></li>
<li><a href='#set_layers'><p>Set up a column of multiple plots</p></a></li>
<li><a href='#SLE_gwas_sub'><p>SLE GWAS data subset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Gene Locus Plot with Gene Annotations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/myles-lewis/locuszoomr/issues">https://github.com/myles-lewis/locuszoomr/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/myles-lewis/locuszoomr">https://github.com/myles-lewis/locuszoomr</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Publication-ready regional gene locus plots similar to those produced by the web interface 'LocusZoom' <a href="https://my.locuszoom.org">https://my.locuszoom.org</a>, but running locally in R. Genetic or genomic data with gene annotation tracks are plotted via R base graphics, 'ggplot2' or 'plotly', allowing flexibility and easy customisation including laying out multiple locus plots on the same page. It uses the 'LDlink' API <a href="https://ldlink.nih.gov/?tab=apiaccess">https://ldlink.nih.gov/?tab=apiaccess</a> to query linkage disequilibrium data from the 1000 Genomes Project and can overlay this on plots.</td>
</tr>
<tr>
<td>Language:</td>
<td>en-gb</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5)</td>
</tr>
<tr>
<td>Imports:</td>
<td>AnnotationFilter, BiocGenerics, cowplot, dplyr, ensembldb,
GenomeInfoDb, GenomicRanges, gggrid, ggplot2, ggrepel,
graphics, grDevices, grid, IRanges, LDlinkR, memoise, plotly,
rlang, rtracklayer, zoo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationHub, EnsDb.Hsapiens.v75, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-17 13:57:44 UTC; myles</td>
</tr>
<tr>
<td>Author:</td>
<td>Myles Lewis <a href="https://orcid.org/0000-0001-9365-5345"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Myles Lewis &lt;myles.lewis@qmul.ac.uk&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-17 14:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='eqtl_plot'>Locus eQTL plot</h2><span id='topic+eqtl_plot'></span>

<h3>Description</h3>

<p>Produces a plot of eQTL data embedded in a 'locus' class object. Intended for
use with <code><a href="#topic+set_layers">set_layers()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eqtl_plot(
  loc,
  tissue = "Whole Blood",
  eqtl_gene = loc$gene,
  scheme = "RdYlBu",
  col = NA,
  pcutoff = NULL,
  xlab = NULL,
  ylab = expression("-log"[10] ~ "P"),
  cex.axis = 0.9,
  xticks = TRUE,
  border = FALSE,
  add = FALSE,
  align = TRUE,
  legend_pos = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eqtl_plot_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <a href="#topic+locus">locus</a>.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_tissue">tissue</code></td>
<td>
<p>GTex tissue in which eQTL has been measured</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_eqtl_gene">eqtl_gene</code></td>
<td>
<p>Gene showing eQTL effect</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_scheme">scheme</code></td>
<td>
<p>Character string specifying palette for effect size showing
up/downregulation eQTL using <a href="grDevices.html#topic+palettes">grDevices::hcl.colors</a>. Alternatively a
vector of 6 colours.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_col">col</code></td>
<td>
<p>Outline point colour. <code>NA</code> for no outlines.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_xlab">xlab</code></td>
<td>
<p>x axis title.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_ylab">ylab</code></td>
<td>
<p>y axis title.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis numbers and axis title are plotted.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted around upper and
lower plots.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_add">add</code></td>
<td>
<p>Logical whether to add points to an existing plot or generate a
new plot.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_align">align</code></td>
<td>
<p>Logical whether set <code><a href="graphics.html#topic+par">par()</a></code> to align the plot.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character value specifying legend position. See <code><a href="graphics.html#topic+legend">legend()</a></code>.</p>
</td></tr>
<tr><td><code id="eqtl_plot_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="graphics.html#topic+plot">plot()</a></code> for the scatter plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Produces a scatter plot using base graphics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+set_layers">set_layers()</a></code> <code><a href="#topic+scatter_plot">scatter_plot()</a></code>
</p>

<hr>
<h2 id='genetrack_ly'>Gene tracks using 'plotly'</h2><span id='topic+genetrack_ly'></span>

<h3>Description</h3>

<p>Plot gene annotation tracks from <code>ensembldb</code> data using <code>plotly</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genetrack_ly(
  locus,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  cex.text = 0.7,
  gene_col = "blue4",
  exon_col = "blue4",
  exon_border = "blue4",
  maxrows = 8,
  xlab = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetrack_ly_+3A_locus">locus</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons. Set to <code>NA</code> for no
border.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td></tr>
<tr><td><code id="genetrack_ly_+3A_xlab">xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified
in <code>locus</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can used to plot gene annotation tracks on their own.
</p>


<h3>Value</h3>

<p>A 'plotly' plotting object showing gene tracks.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
genetrack_ly(loc)
}
</code></pre>

<hr>
<h2 id='genetracks'>Plot gene tracks</h2><span id='topic+genetracks'></span>

<h3>Description</h3>

<p>Plot gene annotation tracks from <code>ensembldb</code> data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genetracks(
  locus,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  border = FALSE,
  cex.axis = 0.9,
  cex.lab = 1,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = NULL,
  text_pos = "top",
  xticks = TRUE,
  xlab = NULL,
  showRecomb = TRUE,
  align = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetracks_+3A_locus">locus</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="genetracks_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_showexons">showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_text_pos">text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis ticks and numbers are plotted.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_xlab">xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified
in <code>locus</code>.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_showrecomb">showRecomb</code></td>
<td>
<p>Logical controls alignment of right margin if
recombination data present.</p>
</td></tr>
<tr><td><code id="genetracks_+3A_align">align</code></td>
<td>
<p>Logical whether to set <code><a href="graphics.html#topic+par">par()</a></code> to align the plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is called by <code><a href="#topic+locus_plot">locus_plot()</a></code>. It can be used to plot
the gene annotation tracks on their own. It uses base graphics, so
<code><a href="graphics.html#topic+layout">layout()</a></code> can be used to position adjacent plots above or below.
</p>


<h3>Value</h3>

<p>No return value.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
genetracks(loc)

## Limit the number of tracks
genetracks(loc, maxrows = 4)

## Filter by gene biotype
genetracks(loc, filter_gene_biotype = 'protein_coding')

## Customise colours
genetracks(loc, gene_col = 'grey', exon_col = 'orange',
           exon_border = 'darkgrey')
}
</code></pre>

<hr>
<h2 id='genetracks_grob'>Create gene tracks grob</h2><span id='topic+genetracks_grob'></span>

<h3>Description</h3>

<p>Plot gene annotation tracks from <code>ensembldb</code> data using the grid package to
create a grob.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genetracks_grob(
  locus,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  border = FALSE,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = NULL,
  text_pos = "top"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genetracks_grob_+3A_locus">locus</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_showexons">showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td></tr>
<tr><td><code id="genetracks_grob_+3A_text_pos">text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is called by <code><a href="#topic+locus_ggplot">locus_ggplot()</a></code>. It can be used to plot
the gene annotation tracks on their own.
</p>


<h3>Value</h3>

<p>A grob object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
g &lt;- genetracks_grob(loc)
grid::grid.newpage()
grid::grid.draw(g)
}
</code></pre>

<hr>
<h2 id='gg_addgenes'>Add gene tracks to a ggplot2 plot</h2><span id='topic+gg_addgenes'></span>

<h3>Description</h3>

<p>Adds gene tracks to an existing ggplot2 plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_addgenes(p, loc, heights = c(3, 2), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_addgenes_+3A_p">p</code></td>
<td>
<p>ggplot2 plot object. This can be generated by <code><a href="#topic+gg_scatter">gg_scatter()</a></code> and
then modified.</p>
</td></tr>
<tr><td><code id="gg_addgenes_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="gg_addgenes_+3A_heights">heights</code></td>
<td>
<p>Vector specifying ratio of heights of upper plot and lower
gene track.</p>
</td></tr>
<tr><td><code id="gg_addgenes_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+gg_genetracks">gg_genetracks()</a></code> to control
colours of gene tracks etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 plotting object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gg_scatter">gg_scatter()</a></code> <code><a href="#topic+gg_genetracks">gg_genetracks()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
p &lt;- gg_scatter(loc)
gg_addgenes(p, loc)
}
</code></pre>

<hr>
<h2 id='gg_genetracks'>Plot gene tracks</h2><span id='topic+gg_genetracks'></span>

<h3>Description</h3>

<p>Plot gene annotation tracks from <code>ensembldb</code> data using ggplot2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_genetracks(
  loc,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  border = FALSE,
  cex.axis = 1,
  cex.lab = 1,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = NULL,
  text_pos = "top",
  xticks = TRUE,
  xlab = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_genetracks_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_showexons">showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene annotation
panel.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_text_pos">text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis ticks and numbers are plotted.</p>
</td></tr>
<tr><td><code id="gg_genetracks_+3A_xlab">xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified in
<code>locus</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is called by <code><a href="#topic+locus_ggplot">locus_ggplot()</a></code>. It can be used to plot
the gene annotation tracks on their own.
</p>


<h3>Value</h3>

<p>A ggplot2 object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
grid::grid.newpage()
gg_genetracks(loc)
}
</code></pre>

<hr>
<h2 id='gg_scatter'>Locus scatter plot using ggplot2</h2><span id='topic+gg_scatter'></span>

<h3>Description</h3>

<p>Produces a scatter plot from a 'locus' class object (without gene tracks).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gg_scatter(
  loc,
  index_snp = loc$index_snp,
  pcutoff = 5e-08,
  scheme = c("royalblue", "red", "purple"),
  size = 2,
  cex.axis = 1,
  cex.lab = 1,
  xlab = NULL,
  ylab = NULL,
  yzero = (loc$yvar == "logP"),
  xticks = TRUE,
  border = FALSE,
  showLD = TRUE,
  LD_scheme = c("grey", "royalblue", "cyan2", "green3", "orange", "red", "purple"),
  recomb_col = "blue",
  legend_pos = "topleft",
  labels = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gg_scatter_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <a href="#topic+locus">locus</a>.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_index_snp">index_snp</code></td>
<td>
<p>Specifies index SNP to be shown in a different colour and
symbol. Defaults to the SNP with the lowest p-value. Set to <code>NULL</code> to not
show this.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_scheme">scheme</code></td>
<td>
<p>Vector of 3 colours if LD is not shown: 1st = normal points,
2nd = colour for significant points, 3rd = index SNP.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_size">size</code></td>
<td>
<p>Specifies size for points.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_xlab">xlab</code></td>
<td>
<p>x axis title.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_ylab">ylab</code></td>
<td>
<p>y axis title.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_yzero">yzero</code></td>
<td>
<p>Logical whether to force y axis limit to include y=0.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis numbers and axis title are plotted.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted around the plot.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_showld">showLD</code></td>
<td>
<p>Logical whether to show LD with colours</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_ld_scheme">LD_scheme</code></td>
<td>
<p>Vector of colours for plotting LD. The first colour is for SNPs
which lack LD information. The next 5 colours are for r2 or D' LD results
ranging from 0 to 1 in intervals of 0.2. The final colour is for the index
SNP.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_recomb_col">recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to NA to hide the line. See <code><a href="#topic+link_recomb">link_recomb()</a></code> to add
recombination rate data.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Position of legend. Set to <code>NULL</code> to hide legend.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_labels">labels</code></td>
<td>
<p>Character vector of SNP or genomic feature IDs to label. The
value &quot;index&quot; selects the highest point or index SNP as defined when
<code><a href="#topic+locus">locus()</a></code> is called. Set to <code>NULL</code> to remove all labels.</p>
</td></tr>
<tr><td><code id="gg_scatter_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code>geom_text_repel()</code> to configure
label drawing.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If recombination rate data is included in the locus object following a call
to <code><a href="#topic+link_recomb">link_recomb()</a></code>, this is plotted as an additional line with a secondary y
axis. In the base graphics version the line is placed under the scatter
points, but this is not possible with ggplot2 as the secondary y axis data
must be plotted on top of the primary scatter point data.
</p>


<h3>Value</h3>

<p>Returns a ggplot2 plot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+gg_addgenes">gg_addgenes()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
gg_scatter(loc)
}
</code></pre>

<hr>
<h2 id='line_plot'>Locus line plot</h2><span id='topic+line_plot'></span>

<h3>Description</h3>

<p>Produces a line plot from a 'locus' class object. Intended for use with
<code><a href="#topic+set_layers">set_layers()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>line_plot(
  loc,
  pcutoff = 5e-08,
  xlab = NULL,
  ylab = expression("-log"[10] ~ "P"),
  cex.axis = 1,
  xticks = FALSE,
  border = FALSE,
  align = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="line_plot_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <a href="#topic+locus">locus</a>.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_xlab">xlab</code></td>
<td>
<p>x axis title.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_ylab">ylab</code></td>
<td>
<p>y axis title.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis numbers and axis title are plotted.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted around upper and
lower plots.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_align">align</code></td>
<td>
<p>Logical whether set <code><a href="graphics.html#topic+par">par()</a></code> to align the plot.</p>
</td></tr>
<tr><td><code id="line_plot_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="graphics.html#topic+plot">plot()</a></code> for the scatter plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Produces a scatter plot using base graphics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+set_layers">set_layers()</a></code> <code><a href="#topic+scatter_plot">scatter_plot()</a></code>
</p>

<hr>
<h2 id='link_eqtl'>Obtain GTEx eQTL data via LDlinkR</h2><span id='topic+link_eqtl'></span>

<h3>Description</h3>

<p>Adds eQTL (expression quantitative trait loci) information from GTEx
(https://gtexportal.org/) to a 'locus' class object. It queries LDlink
(https://ldlink.nci.nih.gov/) via the <code>LDlinkR</code> package to retrieve GTEx eQTL
information on a reference SNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>link_eqtl(loc, pop = "CEU", r2d = "r2", token = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="link_eqtl_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code></p>
</td></tr>
<tr><td><code id="link_eqtl_+3A_pop">pop</code></td>
<td>
<p>A 1000 Genomes Project population, (e.g. YRI or CEU), multiple
allowed, default = &quot;CEU&quot;. Passed to <code>LDlinkR::LDexpress()</code>.</p>
</td></tr>
<tr><td><code id="link_eqtl_+3A_r2d">r2d</code></td>
<td>
<p>Either &quot;r2&quot; for LD r^2 or &quot;d&quot; for LD D', default = &quot;r2&quot;. Passed
to <code>LDlinkR::LDexpress()</code>.</p>
</td></tr>
<tr><td><code id="link_eqtl_+3A_token">token</code></td>
<td>
<p>Personal access token for accessing 1000 Genomes LD data via
LDlink API. See <code>LDlinkR</code> package documentation.</p>
</td></tr>
<tr><td><code id="link_eqtl_+3A_...">...</code></td>
<td>
<p>Optional arguments such as <code>genome_build</code> which are passed on to
<code>LDlinkR::LDexpress()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The additional eQTL information obtained from LDlink web server can be
displayed using <code><a href="#topic+eqtl_plot">eqtl_plot()</a></code> which generates a scatter plot with gene tracks
similar to a locus plot, or with <code><a href="#topic+overlay_plot">overlay_plot()</a></code> which tries to overlay the
EQTL analysis over the original locus results (e.g. GWAS).
</p>


<h3>Value</h3>

<p>Returns an object of class 'locus' with an extra list element 'LDexp'
containing a dataframe of information obtained via <code>LDexpress()</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+eqtl_plot">eqtl_plot()</a></code> <code><a href="#topic+overlay_plot">overlay_plot()</a></code>
</p>

<hr>
<h2 id='link_LD'>Obtain LD at a locus from LDlink</h2><span id='topic+link_LD'></span>

<h3>Description</h3>

<p>Adds LD information to a 'locus' class object. It queries LDlink
(https://ldlink.nci.nih.gov/) via the <code>LDlinkR</code> package to retrieve linkage
disequilibrium (LD) information on a reference SNP.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>link_LD(loc, pop = "CEU", r2d = "r2", token = "", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="link_LD_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code></p>
</td></tr>
<tr><td><code id="link_LD_+3A_pop">pop</code></td>
<td>
<p>A 1000 Genomes Project population, (e.g. YRI or CEU), multiple
allowed, default = &quot;CEU&quot;. Passed to <code>LDlinkR::LDmatrix()</code>.</p>
</td></tr>
<tr><td><code id="link_LD_+3A_r2d">r2d</code></td>
<td>
<p>Either &quot;r2&quot; for LD r^2 or &quot;d&quot; for LD D', default = &quot;r2&quot;. Passed
to <code>LDlinkR::LDmatrix()</code>.</p>
</td></tr>
<tr><td><code id="link_LD_+3A_token">token</code></td>
<td>
<p>Personal access token for accessing 1000 Genomes LD data via
LDlink API. See <code>LDlinkR</code> package documentation.</p>
</td></tr>
<tr><td><code id="link_LD_+3A_...">...</code></td>
<td>
<p>Optional arguments such as <code>genome_build</code> which are passed on to
<code>LDlinkR::LDmatrix()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Results are cached using <code>memoise</code> to reduce API requests.
</p>


<h3>Value</h3>

<p>Returns a list object of class 'locus'. LD information is added as a
column <code>ld</code> in list element <code>data</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code>
</p>

<hr>
<h2 id='link_recomb'>Query UCSC for Recombination data</h2><span id='topic+link_recomb'></span>

<h3>Description</h3>

<p>Adds recombination data to a 'locus' object by querying UCSC genome browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>link_recomb(loc, genome = "hg38", table = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="link_recomb_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' generated by <code><a href="#topic+locus">locus()</a></code></p>
</td></tr>
<tr><td><code id="link_recomb_+3A_genome">genome</code></td>
<td>
<p>Either <code>"hg38"</code> or <code>"hg19"</code></p>
</td></tr>
<tr><td><code id="link_recomb_+3A_table">table</code></td>
<td>
<p>Optional character value specifying which recombination table to
use.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Uses the <code>rtracklayer</code> package to query UCSC genome browser for recombination
rate data.
</p>
<p>Possible options for <code>table</code> for hg19 are <code>"hapMapRelease24YRIRecombMap"</code>,
<code>"hapMapRelease24CEURecombMap"</code>, <code>"hapMapRelease24CombinedRecombMap"</code> (the
default).
</p>
<p>The only option for <code>table</code> for hg38 is <code>"recomb1000GAvg"</code> (the default).
</p>
<p>Sometimes <code>rtracklayer</code> generates an intermittent API error: try calling
<code>link_recomb()</code> again. Errors are handled gracefully using <code>try()</code> to allow
users to wrap <code>link_recomb()</code> in a loop without quitting halfway. Error
messages are still shown. Successful API calls are cached using <code>memoise</code> to
reduce API requests.
</p>


<h3>Value</h3>

<p>A list object of class 'locus'. Recombination data is added as list
element <code>recomb</code>.
</p>

<hr>
<h2 id='locus'>Create locus object for plotting</h2><span id='topic+locus'></span>

<h3>Description</h3>

<p>Creates object of class 'locus' for genomic locus plot similar to
<code>locuszoom</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus(
  gene = NULL,
  data = NULL,
  xrange = NULL,
  seqname = NULL,
  flank = NULL,
  fix_window = NULL,
  ens_db,
  chrom = NULL,
  pos = NULL,
  p = NULL,
  yvar = NULL,
  labs = NULL,
  index_snp = NULL,
  LD = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locus_+3A_gene">gene</code></td>
<td>
<p>Optional character value specifying which gene to view. Either
<code>gene</code>, or <code>xrange</code> plus <code>seqname</code>, or <code>index_snp</code> must be specified.</p>
</td></tr>
<tr><td><code id="locus_+3A_data">data</code></td>
<td>
<p>Dataset (data.frame or data.table) to use for plot. If
unspecified or <code>NULL</code>, gene track information alone is returned.</p>
</td></tr>
<tr><td><code id="locus_+3A_xrange">xrange</code></td>
<td>
<p>Optional vector of genomic position range for the x axis.</p>
</td></tr>
<tr><td><code id="locus_+3A_seqname">seqname</code></td>
<td>
<p>Optional, specifies which chromosome to plot.</p>
</td></tr>
<tr><td><code id="locus_+3A_flank">flank</code></td>
<td>
<p>Single value or vector with 2 values for how much flanking
region left and right of the gene to show. Defaults to 100kb.</p>
</td></tr>
<tr><td><code id="locus_+3A_fix_window">fix_window</code></td>
<td>
<p>Optional alternative to <code>flank</code>, which allows users to
specify a fixed genomic window centred on the specified gene. Both <code>flank</code>
and <code>fix_window</code> cannot be specified simultaneously.</p>
</td></tr>
<tr><td><code id="locus_+3A_ens_db">ens_db</code></td>
<td>
<p>Either a character string which specifies which Ensembl
database package (version 86 and earlier for Homo sapiens) to query for
gene and exon positions (see <code>ensembldb</code> Bioconductor package). Or an
<code>ensembldb</code> object which can be obtained from the AnnotationHub database.
See the vignette and the <code>AnnotationHub</code> Bioconductor package for how to
create this object.</p>
</td></tr>
<tr><td><code id="locus_+3A_chrom">chrom</code></td>
<td>
<p>Determines which column in <code>data</code> contains chromosome
information. If <code>NULL</code> tries to autodetect the column.</p>
</td></tr>
<tr><td><code id="locus_+3A_pos">pos</code></td>
<td>
<p>Determines which column in <code>data</code> contains position information.
If <code>NULL</code> tries to autodetect the column.</p>
</td></tr>
<tr><td><code id="locus_+3A_p">p</code></td>
<td>
<p>Determines which column in <code>data</code> contains SNP p-values.
If <code>NULL</code> tries to autodetect the column.</p>
</td></tr>
<tr><td><code id="locus_+3A_yvar">yvar</code></td>
<td>
<p>Specifies column in <code>data</code> for plotting on the y axis as an
alternative to specifying p-values. Both <code>p</code> and <code>yvar</code> cannot be specified
simultaneously.</p>
</td></tr>
<tr><td><code id="locus_+3A_labs">labs</code></td>
<td>
<p>Determines which column in <code>data</code> contains SNP rs IDs.
If <code>NULL</code> tries to autodetect the column.</p>
</td></tr>
<tr><td><code id="locus_+3A_index_snp">index_snp</code></td>
<td>
<p>Specifies the index SNP. If not specified, the SNP with the
lowest P value is selected. Can be used to specify locus region instead of
specifying <code>gene</code>, or <code>seqname</code> and <code>xrange</code>.</p>
</td></tr>
<tr><td><code id="locus_+3A_ld">LD</code></td>
<td>
<p>Optional character value to specify which column in <code>data</code> contains
LD information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is an R version of <code>locuszoom</code> (http://locuszoom.org) for generating
publication ready Manhattan plots of gene loci. It references Ensembl
databases using the <code>ensembldb</code> Bioconductor package framework for annotating
genes and exons in the locus.
</p>


<h3>Value</h3>

<p>Returns an object of class 'locus' ready for plotting, containing
the subset of GWAS data to be plotted,
chromosome and genomic position range,
Ensembl database version number,
column names for chromosome, position, SNP ID, p-value or variable for
plotting on y axis,
locus gene information from Ensembl and
locus exon information from Ensembl. If <code>data</code> is <code>NULL</code> then gene track
information alone is returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Bioconductor package EnsDb.Hsapiens.v75 is needed for these examples
if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
summary(loc)
locus_plot(loc)
loc2 &lt;- locus(SLE_gwas_sub, gene = 'STAT4', flank = 1e5,
              ens_db = "EnsDb.Hsapiens.v75")
locus_plot(loc2)
}
</code></pre>

<hr>
<h2 id='locus_ggplot'>Locus plot using ggplot2</h2><span id='topic+locus_ggplot'></span>

<h3>Description</h3>

<p>Genomic locus plot similar to locuszoom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_ggplot(
  loc,
  heights = c(3, 2),
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  border = FALSE,
  cex.axis = 1,
  cex.lab = 1,
  cex.text = 0.7,
  gene_col = ifelse(showExons, "blue4", "skyblue"),
  exon_col = "blue4",
  exon_border = "blue4",
  showExons = TRUE,
  maxrows = 12,
  text_pos = "top",
  xticks = "top",
  xlab = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locus_ggplot_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_heights">heights</code></td>
<td>
<p>Vector supplying the ratio of top to bottom plot.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_showexons">showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle. If <code>showExons = FALSE</code> colours are specified by <code>exon_border</code>
for rectangle border and <code>gene_col</code> for the fill colour.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene annotation
panel.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_text_pos">text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis ticks and numbers are plotted.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_xlab">xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified in
<code>locus</code>.</p>
</td></tr>
<tr><td><code id="locus_ggplot_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to <code><a href="#topic+gg_scatter">gg_scatter()</a></code> to control
the scatter plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Arguments to control plotting of the gene tracks are passed onto
<code><a href="#topic+gg_genetracks">gg_genetracks()</a></code> and for the scatter plot are passed via <code>...</code> to
<code><a href="#topic+gg_scatter">gg_scatter()</a></code>. See the documentation for each of these functions for
details.
</p>


<h3>Value</h3>

<p>Returns a ggplot2 plot containing a scatter plot with genetracks
underneath.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gg_scatter">gg_scatter()</a></code> <code><a href="#topic+gg_genetracks">gg_genetracks()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
locus_ggplot(loc)
}
</code></pre>

<hr>
<h2 id='locus_plot'>Locus plot</h2><span id='topic+locus_plot'></span>

<h3>Description</h3>

<p>Genomic locus plot similar to locuszoom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_plot(
  loc,
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  xlab = NULL,
  cex = 1,
  cex.axis = 0.9,
  cex.lab = 1,
  cex.text = 0.7,
  use_layout = TRUE,
  heights = c(3, 2),
  showExons = TRUE,
  maxrows = 7,
  xticks = "bottom",
  border = FALSE,
  gene_col = "blue4",
  exon_col = "blue4",
  exon_border = "blue4",
  text_pos = "top",
  recomb_col = "blue",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locus_plot_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="locus_plot_+3A_xlab">xlab</code></td>
<td>
<p>x axis title.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_cex">cex</code></td>
<td>
<p>Specifies size for points.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_use_layout">use_layout</code></td>
<td>
<p>Logical whether <code>graphics::layout</code> is called. Default
<code>TRUE</code> is for a standard single plot. Set to <code>FALSE</code> if a more complex
layout with multiple plots is required e.g. using <code><a href="#topic+multi_layout">multi_layout()</a></code>.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_heights">heights</code></td>
<td>
<p>Ratio of top to bottom plot. See <a href="graphics.html#topic+layout">layout</a>.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_showexons">showExons</code></td>
<td>
<p>Logical whether to show exons or simply show whole gene as a
rectangle</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_xticks">xticks</code></td>
<td>
<p>Character value of either 'top' or 'bottom' specifying
whether x axis ticks and numbers are plotted on top or bottom plot window.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted around upper and
lower plots.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons (or genes if
<code>showExons</code> is <code>FALSE</code>). Set to <code>NA</code> for no border.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_text_pos">text_pos</code></td>
<td>
<p>Character value of either 'top' or 'left' specifying
placement of gene name labels.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_recomb_col">recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to <code>NA</code> to hide the line. See <code><a href="#topic+link_recomb">link_recomb()</a></code> to add
recombination rate data.</p>
</td></tr>
<tr><td><code id="locus_plot_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+scatter_plot">scatter_plot()</a></code> and <code><a href="graphics.html#topic+plot">plot()</a></code> to control
the scatter plot, e.g. <code>ylab</code>, <code>main</code>, etc.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is an R version of locuszoom for generating publication ready Manhattan
plots of gene loci. It references Ensembl databases for annotating genes
and exons. Use <code><a href="#topic+locus">locus()</a></code> first to generate an object of class 'locus' for
plotting. LDlink web server can be queried using function <code><a href="#topic+link_LD">link_LD()</a></code> to
retrieve linkage disequilibrium (LD) information on the index SNP.
</p>
<p>Arguments to control plotting of the gene tracks are passed onto
<code><a href="#topic+genetracks">genetracks()</a></code> and for the scatter plot are passed via <code>...</code> to
<code><a href="#topic+scatter_plot">scatter_plot()</a></code>. See the documentation for each of these functions for
details.
</p>


<h3>Value</h3>

<p>No return value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+scatter_plot">scatter_plot()</a></code> <code><a href="#topic+genetracks">genetracks()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', flank = 1e5,
             ens_db = "EnsDb.Hsapiens.v75")
locus_plot(loc)

## Use embedded LD information in column `r2`
loc2 &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
              ens_db = "EnsDb.Hsapiens.v75")
## Add label for index SNP
locus_plot(loc2, labels = "index")
}
</code></pre>

<hr>
<h2 id='locus_plotly'>Locus plotly</h2><span id='topic+locus_plotly'></span>

<h3>Description</h3>

<p>Genomic locus plot similar to locuszoom, using plotly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>locus_plotly(
  loc,
  heights = c(0.6, 0.4),
  filter_gene_name = NULL,
  filter_gene_biotype = NULL,
  cex.text = 0.7,
  gene_col = "blue4",
  exon_col = "blue4",
  exon_border = "blue4",
  maxrows = 8,
  xlab = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="locus_plotly_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_heights">heights</code></td>
<td>
<p>Vector controlling relative height of each panel on 0-1 scale.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_filter_gene_name">filter_gene_name</code></td>
<td>
<p>Vector of gene names to display.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_filter_gene_biotype">filter_gene_biotype</code></td>
<td>
<p>Vector of gene biotypes to be filtered. Use
<code><a href="ensembldb.html#topic+EnsDb-class">ensembldb::listGenebiotypes()</a></code> to display possible biotypes. For example,
<code>ensembldb::listGenebiotypes(EnsDb.Hsapiens.v75)</code></p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_cex.text">cex.text</code></td>
<td>
<p>Font size for gene text.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_gene_col">gene_col</code></td>
<td>
<p>Colour for gene lines.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_exon_col">exon_col</code></td>
<td>
<p>Fill colour for exons.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_exon_border">exon_border</code></td>
<td>
<p>Border line colour outlining exons. Set to <code>NA</code> for no
border.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_maxrows">maxrows</code></td>
<td>
<p>Specifies maximum number of rows to display in gene
annotation panel.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_xlab">xlab</code></td>
<td>
<p>Title for x axis. Defaults to chromosome <code>seqname</code> specified
in <code>locus</code>.</p>
</td></tr>
<tr><td><code id="locus_plotly_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="#topic+scatter_plotly">scatter_plotly()</a></code> to control the
scatter plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is an R/plotly version of locuszoom for exploring regional Manhattan
plots of gene loci. Use <code><a href="#topic+locus">locus()</a></code> first to generate an object of class
'locus' for plotting. This references a selected Ensembl database for
annotating genes and exons. Hover over the points or gene tracks to reveal
more information.
</p>


<h3>Value</h3>

<p>A 'plotly' plotting object showing a scatter plot above gene tracks.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+genetrack_ly">genetrack_ly()</a></code> <code><a href="#topic+scatter_plotly">scatter_plotly()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {
data(SLE_gwas_sub)
loc &lt;- locus(SLE_gwas_sub, gene = "IRF5", flank = c(7e4, 2e5), LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
locus_plotly(loc)
}
</code></pre>

<hr>
<h2 id='multi_layout'>Layout multiple locus plots</h2><span id='topic+multi_layout'></span>

<h3>Description</h3>

<p>Produces pages with multiple locus plots on.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_layout(
  plots,
  nrow = 1,
  ncol = 1,
  heights = c(3, 2),
  legend_pos = "topleft",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_layout_+3A_plots">plots</code></td>
<td>
<p>Either an 'expression' to be evaluated which is a series of
calls to <code><a href="#topic+locus_plot">locus_plot()</a></code> or similar plotting functions, or a list of 'locus'
class objects which are plotted in sequence.</p>
</td></tr>
<tr><td><code id="multi_layout_+3A_nrow">nrow</code></td>
<td>
<p>Number of rows of plots</p>
</td></tr>
<tr><td><code id="multi_layout_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns of plots</p>
</td></tr>
<tr><td><code id="multi_layout_+3A_heights">heights</code></td>
<td>
<p>Vector of length 2 specifying height for plot and gene tracks</p>
</td></tr>
<tr><td><code id="multi_layout_+3A_legend_pos">legend_pos</code></td>
<td>
<p>A keyword either &quot;topleft&quot; or &quot;topright&quot; or <code>NULL</code> to hide
the legend. Not invoked if <code>plots</code> is an expression. The legend is only
shown on one plot on each page.</p>
</td></tr>
<tr><td><code id="multi_layout_+3A_...">...</code></td>
<td>
<p>Optional arguments passed to <code><a href="#topic+locus_plot">locus_plot()</a></code> if <code>plots</code> contains a
list</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus_plot">locus_plot()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(require(EnsDb.Hsapiens.v75)) {

data(SLE_gwas_sub)
genes &lt;- c("STAT4", "UBE2L3", "IRF5")
loclist &lt;- lapply(genes, locus,
                  data = SLE_gwas_sub,
                  ens_db = "EnsDb.Hsapiens.v75",
                  LD = "r2")
## produce 3 locus plots, one on each page
multi_layout(loclist)

## place 3 locus plots in a row on a single page
multi_layout(loclist, ncol = 3)

## full control
loc &lt;- locus(SLE_gwas_sub, gene = 'STAT4', flank = 1e5, LD = "r2",
             ens_db = "EnsDb.Hsapiens.v75")
loc2 &lt;- locus(SLE_gwas_sub, gene = 'IRF5', flank = c(7e4, 2e5), LD = "r2",
              ens_db = "EnsDb.Hsapiens.v75")
loc3 &lt;- locus(SLE_gwas_sub, gene = 'UBE2L3', LD = "r2",
              ens_db = "EnsDb.Hsapiens.v75")
multi_layout(ncol = 3,
             plots = {
               locus_plot(loc, use_layout = FALSE, legend_pos = 'topleft')
               locus_plot(loc2, use_layout = FALSE, legend_pos = NULL)
               locus_plot(loc3, use_layout = FALSE, legend_pos = NULL)
             })

}
</code></pre>

<hr>
<h2 id='overlay_plot'>Plot overlaying eQTL and GWAS data</h2><span id='topic+overlay_plot'></span>

<h3>Description</h3>

<p>Experimental plotting function for overlaying eQTL data from GTEx on top of
GWAS results. y axis shows the -log10 p-value for the GWAS result.
Significant eQTL for the specified gene are overlaid using colours and
symbols.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>overlay_plot(
  loc,
  base_col = "black",
  alpha = 0.5,
  scheme = "RdYlBu",
  tissue = "Whole Blood",
  eqtl_gene = loc$gene,
  legend_pos = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="overlay_plot_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <code><a href="#topic+locus">locus()</a></code>.</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_base_col">base_col</code></td>
<td>
<p>Colour of points for SNPs which do not have eQTLs.</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_alpha">alpha</code></td>
<td>
<p>Alpha opacity for non-eQTL points</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_scheme">scheme</code></td>
<td>
<p>Character string specifying palette for effect size showing
up/downregulation eQTL using <a href="grDevices.html#topic+palettes">grDevices::hcl.colors</a>. Alternatively a
vector of 6 colours.</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_tissue">tissue</code></td>
<td>
<p>GTex tissue in which eQTL has been measured</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_eqtl_gene">eqtl_gene</code></td>
<td>
<p>Gene showing eQTL effect</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Character value specifying legend position. See <code><a href="graphics.html#topic+legend">legend()</a></code>.</p>
</td></tr>
<tr><td><code id="overlay_plot_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="#topic+locus_plot">locus_plot()</a></code> for the locus plot.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Produces a plot using base graphics.
</p>

<hr>
<h2 id='quick_peak'>Fast peak finder in GWAS data</h2><span id='topic+quick_peak'></span>

<h3>Description</h3>

<p>Simple but fast function for finding peaks in genome-wide association study
(GWAS) data based on setting a minimum distance between peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>quick_peak(
  data,
  npeaks = NA,
  p_cutoff = 5e-08,
  span = 1e+06,
  min_points = 2,
  chrom = NULL,
  pos = NULL,
  p = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="quick_peak_+3A_data">data</code></td>
<td>
<p>GWAS dataset (data.frame or data.table)</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_npeaks">npeaks</code></td>
<td>
<p>Number of peaks to find. If set to <code>NA</code>, algorithm finds all
distinct peaks separated from one another by region size specified by
<code>span</code>.</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_p_cutoff">p_cutoff</code></td>
<td>
<p>Specifies cut-off for p-value significance above which
p-values are ignored.</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_span">span</code></td>
<td>
<p>Minimum genomic distance between peaks (default 1 Mb)</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_min_points">min_points</code></td>
<td>
<p>Minimum number of p-value significant points which must lie
within the span of a peak. This removes peaks with single or only a few low
p-value SNPs. To disable set <code>min_points</code> to 1 or less.</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_chrom">chrom</code></td>
<td>
<p>Determines which column in <code>data</code> contains chromosome
information. If <code>NULL</code> tries to autodetect the column.</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_pos">pos</code></td>
<td>
<p>Determines which column in <code>data</code> contains position information.
If <code>NULL</code> tries to autodetect the column.</p>
</td></tr>
<tr><td><code id="quick_peak_+3A_p">p</code></td>
<td>
<p>Determines which column in <code>data</code> contains SNP p-values. If <code>NULL</code>
tries to autodetect the column.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is designed for speed. SNP p-values are filtered to only those
which are significant as specified by <code>p_cutoff</code>. Each peak is identified as
the SNP with the lowest p-value and then SNPs in proximity to each peak
within the distance specified by <code>span</code> are removed. Regions such as the HLA
whose peaks may well be broader than <code>span</code> may produce multiple entries.
</p>


<h3>Value</h3>

<p>Vector of row indices
</p>

<hr>
<h2 id='scatter_plot'>Locus scatter plot</h2><span id='topic+scatter_plot'></span>

<h3>Description</h3>

<p>Produces a scatter plot from a 'locus' class object. Intended for use with
<code><a href="#topic+set_layers">set_layers()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatter_plot(
  loc,
  index_snp = loc$index_snp,
  pcutoff = 5e-08,
  scheme = c("royalblue", "red", "purple"),
  cex = 1,
  cex.axis = 0.9,
  cex.lab = 1,
  xlab = NULL,
  ylab = NULL,
  yzero = (loc$yvar == "logP"),
  xticks = TRUE,
  border = FALSE,
  showLD = TRUE,
  LD_scheme = c("grey", "royalblue", "cyan2", "green3", "orange", "red", "purple"),
  recomb_col = "blue",
  legend_pos = "topleft",
  labels = NULL,
  label_x = 4,
  label_y = 4,
  add = FALSE,
  align = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatter_plot_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <a href="#topic+locus">locus</a>.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_index_snp">index_snp</code></td>
<td>
<p>Specifies index SNP to be shown in a different colour and
symbol. Defaults to the SNP with the lowest p-value. Set to <code>NULL</code> to not
show this.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_scheme">scheme</code></td>
<td>
<p>Vector of 3 colours if LD is not shown: 1st = normal points,
2nd = colour for significant points, 3rd = index SNP.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_cex">cex</code></td>
<td>
<p>Specifies size for points.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_cex.axis">cex.axis</code></td>
<td>
<p>Specifies font size for axis numbering.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_cex.lab">cex.lab</code></td>
<td>
<p>Specifies font size for axis titles.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_xlab">xlab</code></td>
<td>
<p>x axis title.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_ylab">ylab</code></td>
<td>
<p>y axis title.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_yzero">yzero</code></td>
<td>
<p>Logical whether to force y axis limit to include y=0.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_xticks">xticks</code></td>
<td>
<p>Logical whether x axis numbers and axis title are plotted.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_border">border</code></td>
<td>
<p>Logical whether a bounding box is plotted around upper and
lower plots.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_showld">showLD</code></td>
<td>
<p>Logical whether to show LD with colours</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_ld_scheme">LD_scheme</code></td>
<td>
<p>Vector of colours for plotting LD. The first colour is for
SNPs which lack LD information. The next 5 colours are for r2 or D' LD
results ranging from 0 to 1 in intervals of 0.2. The final colour is for
the index SNP.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_recomb_col">recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to <code>NA</code> to hide the line. See <code><a href="#topic+link_recomb">link_recomb()</a></code> to add
recombination rate data.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_legend_pos">legend_pos</code></td>
<td>
<p>Position of legend. See <code><a href="graphics.html#topic+legend">legend()</a></code>. Set to <code>NULL</code> to hide
legend.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_labels">labels</code></td>
<td>
<p>Character vector of SNP or genomic feature IDs to label. The
value &quot;index&quot; selects the highest point or index SNP as defined when
<code><a href="#topic+locus">locus()</a></code> is called. Set to <code>NULL</code> to remove all labels.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_label_x">label_x</code></td>
<td>
<p>Value or vector for position of label as percentage of x axis
scale.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_label_y">label_y</code></td>
<td>
<p>Value or vector for position of label as percentage of y axis
scale.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_add">add</code></td>
<td>
<p>Logical whether to add points to an existing plot or generate a
new plot.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_align">align</code></td>
<td>
<p>Logical whether to set <code><a href="graphics.html#topic+par">par()</a></code> to align the plot.</p>
</td></tr>
<tr><td><code id="scatter_plot_+3A_...">...</code></td>
<td>
<p>Other arguments passed to <code><a href="graphics.html#topic+plot">plot()</a></code> to control the scatter plot
e.g. <code>main</code>, <code>ylim</code> etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value. Produces a scatter plot using base graphics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+set_layers">set_layers()</a></code>
</p>

<hr>
<h2 id='scatter_plotly'>Locus scatter plotly</h2><span id='topic+scatter_plotly'></span>

<h3>Description</h3>

<p>Produces a scatter plot from a 'locus' class object using plotly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scatter_plotly(
  loc,
  index_snp = loc$index_snp,
  pcutoff = 5e-08,
  chromCol = "royalblue",
  sigCol = "red",
  xlab = NULL,
  ylab = NULL,
  yzero = (loc$yvar == "logP"),
  showLD = TRUE,
  LD_scheme = c("grey", "royalblue", "cyan2", "green3", "orange", "red", "purple"),
  marker_outline = "black",
  marker_size = 7,
  recomb_col = "blue"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scatter_plotly_+3A_loc">loc</code></td>
<td>
<p>Object of class 'locus' to use for plot. See <a href="#topic+locus">locus</a>.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_index_snp">index_snp</code></td>
<td>
<p>Specifies index SNP to be shown in a different colour and
symbol. Defaults to the SNP with the lowest p-value. Set to <code>NULL</code> to not
show this.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_pcutoff">pcutoff</code></td>
<td>
<p>Cut-off for p value significance. Defaults to p = 5e-08. Set
to <code>NULL</code> to disable.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_chromcol">chromCol</code></td>
<td>
<p>Colour for normal points if <code>LD</code> is <code>FALSE</code> when the locus
object is made.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_sigcol">sigCol</code></td>
<td>
<p>Colour for significant points if <code>LD</code> is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_xlab">xlab</code></td>
<td>
<p>x axis title.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_ylab">ylab</code></td>
<td>
<p>y axis title.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_yzero">yzero</code></td>
<td>
<p>Logical whether to force y axis limit to include y=0.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_showld">showLD</code></td>
<td>
<p>Logical whether to show LD with colours</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_ld_scheme">LD_scheme</code></td>
<td>
<p>Vector of colours for plotting LD. The first colour is for
SNPs which lack LD information. The next 5 colours are for r^2 or D' LD
results ranging from 0 to 1 in intervals of 0.2. The final colour is for
the index SNP.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_marker_outline">marker_outline</code></td>
<td>
<p>Specifies colour for outlining points.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_marker_size">marker_size</code></td>
<td>
<p>Value for size of markers in plotly units.</p>
</td></tr>
<tr><td><code id="scatter_plotly_+3A_recomb_col">recomb_col</code></td>
<td>
<p>Colour for recombination rate line if recombination rate
data is present. Set to <code>NA</code> to hide the line. See <code><a href="#topic+link_recomb">link_recomb()</a></code> to add
recombination rate data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>plotly</code> scatter plot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+locus">locus()</a></code> <code><a href="#topic+locus_plotly">locus_plotly()</a></code>
</p>

<hr>
<h2 id='set_layers'>Set up a column of multiple plots</h2><span id='topic+set_layers'></span>

<h3>Description</h3>

<p>Uses <code><a href="graphics.html#topic+layout">layout()</a></code> to set up multiple locus plots aligned in a column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_layers(n = 1, heights = c(rep(3, n), 2), rev = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_layers_+3A_n">n</code></td>
<td>
<p>Number of plots (not including gene tracks on bottom)</p>
</td></tr>
<tr><td><code id="set_layers_+3A_heights">heights</code></td>
<td>
<p>Vector of length <code>nrow + 1</code> specifying height for plots with
a gene track on the bottom</p>
</td></tr>
<tr><td><code id="set_layers_+3A_rev">rev</code></td>
<td>
<p>Logical whether to reverse plotting order and plot from bottom to
top</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Sets <code><a href="graphics.html#topic+layout">layout()</a></code> to enable multiple plots aligned in a column. The
gene track is assumed to be positioned on the bottom. Returns <code>par()</code>
invisibly so that layout can be reset to default at the end of plotting.
</p>


<h3>See Also</h3>

<p><code><a href="graphics.html#topic+layout">layout()</a></code>
</p>

<hr>
<h2 id='SLE_gwas_sub'>SLE GWAS data subset</h2><span id='topic+SLE_gwas_sub'></span>

<h3>Description</h3>

<p>Dataset of SNPs at 3 gene loci (UBE2L3, STAT4, IRF5) from GWAS on SLE
(Bentham et al, 2015, Nature Genetics 47(12):1457-64, PMID: 26502338).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(SLE_gwas_sub)
</code></pre>


<h3>Format</h3>

<p>Data frame with 1990 rows and 11 variables
</p>


<h3>Source</h3>

<p><a href="https://www.ebi.ac.uk/gwas/studies/GCST003156">https://www.ebi.ac.uk/gwas/studies/GCST003156</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
