<!DOCTYPE html><html><head><title>Help for package lodGWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lodGWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#lodGWAS-package'>
<p>Genome-Wide Association Analysis of a Biomarker Accounting for</p>
Limit of Detection</a></li>
<li><a href='#lod_GWAS'>
<p>Genome Wide Association Analysis accounting for Limit of Detection</p></a></li>
<li><a href='#lod_QC'><p>Checking the coding of phenotype and outsideLOD values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genome-Wide Association Analysis of a Biomarker Accounting for
Limit of Detection</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0-7</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-11-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Ahmad Vaez, Ilja M. Nolte, Peter J. van der Most</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ilja M. Nolte &lt;i.m.nolte@umcg.nl&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>survival, rms, stats, utils</td>
</tr>
<tr>
<td>Description:</td>
<td>Genome-wide association (GWAS) analyses
  of a biomarker that account for the limit of detection.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2015-11-10 13:46:43 UTC; mostPJ</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2015-11-30 13:35:40</td>
</tr>
</table>
<hr>
<h2 id='lodGWAS-package'>
Genome-Wide Association Analysis of a Biomarker Accounting for
Limit of Detection
</h2><span id='topic+lodGWAS-package'></span><span id='topic+lodGWAS'></span>

<h3>Description</h3>

<p>Statistical analysis of a biomarker is often complicated
because the detection range of the assay of the biomarker is
restricted. The limits of detection (LOD) are the floor and/or
ceiling values of the biomarker that can be accurately measured
by a particular assay type. Any value of the biomarker beyond
the range of LOD, either smaller than the lower LOD or larger
than the upper LOD, cannot be determined accurately. Those
observations, so-called non-detects (NDs), cannot simply be
excluded from the analysis, because NDs are not 'missing at
random'. They can be considered as censored data, and can
therefore be best analyzed by using statistical methods for
survival analysis.
</p>
<p>lodGWAS is a flexible package for running genome-wide association
analysis of a biomarker that accounts for the problem of
limit of detection of the assay. It treats non-detected values
as censored, and performs a parametric survival analysis on
the phenotype of interest.
</p>
<p>The analysis itself is carried out by the function <code><a href="#topic+lod_GWAS">lod_GWAS</a></code>.
A second function, <code><a href="#topic+lod_QC">lod_QC</a></code>, can be used to check
if the coding of LOD values in the phenotype file is correct.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> lodGWAS</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0-7</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-11-10</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 3)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Ahmad Vaez, Ilja M. Nolte, Peter J. van der Most
</p>
<p>Maintainer: Ilja M. Nolte &lt;i.m.nolte@umcg.nl&gt;
</p>

<hr>
<h2 id='lod_GWAS'>
Genome Wide Association Analysis accounting for Limit of Detection
</h2><span id='topic+lod_GWAS'></span>

<h3>Description</h3>

<p><code>lod_GWAS</code> enables the user to perform a Genome Wide
Association Analysis (GWAS) of a biomarker accommodating the
problem of Limit of Detection (LOD). This function performs a
parametric survival analysis on the phenotype of interest that
includes both measured and censored data.
</p>
<p><code><a href="#topic+lod_QC">lod_QC</a></code> is automatically called within <code>lod_GWAS</code>,
and its quality report will be saved in a separate text file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lod_GWAS(phenofile, pheno_name,
        basic_model = NULL,
        dist = "gaussian",
        mapfile, genofile,
        outputfile,
        filedirectory = getwd(),
        outputheader = "QCGWAS", gzip_output = TRUE,
        lower_limit = NA, upper_limit = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lod_GWAS_+3A_phenofile">phenofile</code></td>
<td>

<p>Either a data frame containing the phenotype (and covariate)
values, or the filename (including the extension) of a data
file containing the same. See below for information on the
required format.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_pheno_name">pheno_name</code></td>
<td>

<p>The name of the column in phenofile that contains the phenotype
values.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_basic_model">basic_model</code></td>
<td>

<p>A formula (coded as a character string) describing the basic
model, not including the genetic component. The covariates
to be included into the analysis are mentioned within
quotation marks separated by plus signs: for example,
<code>basic_model="sex+age"</code>. Please note that covariate
names should exactly match the appropriate column names of
phenotype file. The default is <code>NULL</code>, in which case the
association is modelled without covariates.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_dist">dist</code></td>
<td>

<p>Assumed distribution for (raw or transformed)
phenotype. The options are <code>weibull</code>,
<code>exponential</code>, <code>gaussian</code>, <code>logistic</code>,
<code>lognormal</code> and <code>loglogistic</code>. Default is
<code>gaussian</code>. For more information, see the function
<code>psm</code> of package <code>rms</code>.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_mapfile">mapfile</code></td>
<td>

<p>The file name of the genotype map file (including the file
extension). See below for information on the
required format.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_genofile">genofile</code></td>
<td>

<p>The file name of the genotype dosage file (including the
file extension). See below for information on the
required format.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_outputfile">outputfile</code></td>
<td>

<p>The name for the output file.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_filedirectory">filedirectory</code></td>
<td>

<p>The directory that contains the phenotype and genotype files
and where the output file will be saved. Please note that R
uses <em>forward </em>slash (/) where Windows uses backslash
(\). The default setting is current R working directory.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_outputheader">outputheader</code></td>
<td>

<p>The output format of the analysis results file, to make it
compatible with different software packages. The options are
<code>"QCGWAS"</code>, <code>"GWAMA"</code>, <code>"PLINK"</code>,
<code>"META"</code>, and <code>"GenABEL"</code>. Default is
<code>"QCGWAS"</code>.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_gzip_output">gzip_output</code></td>
<td>

<p>Logical; determines whether the output file is compressed.
Default is <code>TRUE</code>.
</p>
</td></tr>
<tr><td><code id="lod_GWAS_+3A_lower_limit">lower_limit</code>, <code id="lod_GWAS_+3A_upper_limit">upper_limit</code></td>
<td>

<p>Arguments passed to <code><a href="#topic+lod_QC">lod_QC</a></code>. Specifying the limit
of detection allows <code><a href="#topic+lod_QC">lod_QC</a></code> to check if the
phenotypes and outsideLOD columns have been coded correctly.
Please note that these arguments are only used for a quality
check. Any errors will be reported but <em>not</em> corrected.
Default is <code>NA</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>lod_GWAS is the main function of the package, and is capable
of performing a genome-wide association study (GWAS)
accommodating the problem of LOD. It treats non-detects as
censored data, either left- or right-censored or both, and 
performs a parametric survival analysis on the phenotype of
interest that includes both measured and censored values. 
</p>


<h3>Value</h3>

<p><code>lod_GWAS</code> returns an invisible <code>NULL</code>. The real
output are the association results (saved as
<code>[output_file].txt</code>) and the log file generated by
<code><a href="#topic+lod_QC">lod_QC</a></code> (saved as <code>[output_file].txt.log</code>).
</p>


<h3>Input File Format</h3>

<p>An analysis with <code>lod_GWAS</code> requires two files for the
genotypes and one phenotype file. The files can be either
space or tab delimited. The package also accepts files
compressed in the gzip format (extension .gz).
</p>
<p><em>Genotype Files</em>
</p>
<p><code>lodGWAS</code> uses the PLINK dosage format for the genotype
data. This means that two files are needed: one with the
genotypes themselves (genotype dosage file), and one with the
locations of the genetic variants (map file).
</p>
<p><em>Genotype Dosage File</em>
</p>
<p>The genotype dosage file should contain a header line. The
header line (first line) should be:
</p>
<p><code>SNP A1 A2 FID1 IID1 FID2 IID2 ... FIDn IIDn</code>
</p>
<p>The first three columns must appear before the dosage data. The
following columns are the family identifier (FID) and the
individual identifier (IID) of individuals 1 to n. Thus, the
number of columns of the header line should be exactly 3 +
(2 x n_individuals).
</p>
<p>The next lines contain the actual genetic data per individual,
with each row corresponding to a genetic variant. The PLINK
dosage format can be any of three formats: dosage,
two-probabilities, or three-probabilities (see below). lodGWAS
accepts all three formats and will automatically recognize
whether there are one (dosage), two (two-probabilities), or
three (three-probabilities) columns per individual. In case
of any other format it will report that it cannot recognize
the format and will not run.
</p>
<p><em>Dosage format</em>
</p>
<p>A dosage is provided in one column per individual. Each dosage
is a number between 0 and 2. A dosage of 0, 1, or 2 means that
the individual is homozygous for the A2 allele, heterozygous,
or homozygous for the A1 allele, respectively. When the
genetic dataset is expanded using imputation, non-integer
values are also possible, and are defined as the weighted sum
of genotype probabilities (i.e. 0 x prob(A2/A2) + 1 x prob(A1/A2)
+ 2 x prob(A1/A1) ).
The number of columns of the (non-header) lines in a genotype
file in dosage format should be exactly 3 + n_individuals.
</p>
<p>Example of the dosage format:  
</p>
<p><code>      SNP  A1  A2   FID1 IID1 FID2 IID2 FID3 IID3</code>
</p>
<p><code>   rs0001  A   C    0.08      0.72      1.99</code>
</p>
<p><em>Two-probabilities format</em>
</p>
<p>Two numbers, representing the probabilities of the A1/A1 and
A1/A2 genotypes, respectively. The probability of A2/A2 equals
1 minus the sum of Prob(A1/A1) and Prob(A1/A2). Each
probability is a number between 0 and 1. The number of columns
of the (non-header) lines in a genotype file in two-probabilities
format should be exactly  3 + (2 x n_individuals).
</p>
<p>Example of the two-probabilities format:
</p>
<p><code>      SNP  A1  A2   FID1 IID1   FID2 IID2</code>
</p>
<p><code>   rs0001   A   C   0.97 0.02   0.88 0.10</code>
</p>
<p><em>Three-probabilities format</em>
</p>
<p>Three numbers, representing the probabilities of the A1/A1,
A1/A2, and A2/A2 genotypes, respectively. Each probability is
a number between 0 and 1, and the three probabilities per
genetic variant per individual should add up to 1. The number
of columns of the (non-header) lines in a genotype file in
three-probabilities format should be exactly  3 +
(3 x n_individuals).
</p>
<p>Example of the three-probabilities format:
</p>
<p><code>      SNP  A1  A2   FID1 IID1       FID2 IID2</code>
</p>
<p><code>   rs0001   A   C   0.97 0.02 0.01  0.88 0.10 0.02</code>
</p>
<p><em>Genotype Map File</em>
</p>
<p>The genotype map file contains the locations of the genetic
variants, with each row of the file corresponding to a variant.
It must contain four columns:
</p>

<ul>
<li><p> Chromosome (1-22, X, Y or 0 if unspecified)
</p>
</li>
<li><p> Marker ID (identifier of the genetic variant)
</p>
</li>
<li><p> Genetic distance (Morgan, this is not used by <code>lod_GWAS</code>,
so the actual value doesn't matter)
</p>
</li>
<li><p> Physical position (base-pair position)
</p>
</li></ul>

<p><em>Note</em>: unlike the other input files, the map file has <em>no</em>
header line.
</p>
<p><em>Phenotype File</em>
</p>
<p>The phenotype file is a text file containing the non-genetic
data, with each row of the file corresponding to an individual.
It must meet the following requirements:
</p>

<ul>
<li><p> The file must have a header line.
</p>
</li>
<li><p> it must contain the following variables: family
ID, individual ID, phenotype, and outsideLOD (which 
indicates whether the phenotype is measered, or left- or
right-censored). Other columns, e.g. for covariates, are
optional.
</p>
</li>
<li><p> The header (name) of columns for family ID, individual
ID, and outsideLOD must be <code>FID</code>, <code>IID</code>, and
<code>outsideLOD</code>, respectively (note that R is case sensitive).
The other columns (phenotype and cov1 to covN) can have any
arbitrary name.
</p>
</li></ul>

<p>The order of the rows (samples) or columns is not important.
</p>
<p><em>Column descriptions of phenotype file</em>
</p>

<ul>
<li><p> FID: family identifier of the individual. It must
match the FIDs in the genotype dosage file.
</p>
</li>
<li><p> IID: the unique identifier of the individual within
each family. It must match the IIDs in the genotype dosage
file.
</p>
</li>
<li><p> Phenotype: the phenotype or trait of interest, which
can be any numeric value. See below for a few considerations
regarding the phenotype.
</p>
</li>
<li><p> outsideLOD: The variable <code>outsideLOD</code> indicates
whether the phenotype value is within or beyond the range
of LOD. It must be to be coded as <code>0</code> if phenotype &gt;
upper LOD; <code>1</code> if phenotype is within the detection
interval; and <code>2</code> if phenotype &lt; lower LOD. Values
other than <code>0</code>, <code>1</code>, or <code>2</code> are not accepted.
</p>
</li>
<li><p> cov1 to covN: covariate 1 to covariate N. The phenotype
file can contain as many covariates as necessary. Some
examples are: age, sex, BMI, smoking status, medication,
population stratification parameters (principal components),
dosage data of a particular genetic variant (for conditional
analysis), study center, etc.
</p>
</li></ul>

<p><em>A few considerations regarding the phenotype</em>
</p>
<p>Please pay particular attention to these instructions, as
failing to heed them may cause invalid results.
</p>
<p>1) The user must carefully distinguish between two types of
missing phenotype: missing and censored values. Any mix-up
between these two types will yield incorrect results.
</p>
<p>2)	<em>Missing phenotype values</em> are those phenotypes that are
missing for any reason other than being beyond the LOD. They
are considered as real missing (at random). These values must
be coded as <code>NA</code> in both <code>Phenotype</code> and
<code>outsideLOD</code> columns.
</p>
<p>3) <em>Censored phenotype values</em> are NDs, i.e. measurements
that fall beyond the LOD of the assay. NDs are not real
missing values, since they do provide information about the
distribution of the phenotype. Any ND that is below the lower
LOD should be changed to the value of the lower LOD (and the
corresponding outsideLOD value should set to <code>2</code>). Any ND
that is above the upper LOD should be changed to the value of
the upper LOD (and the corresponding outsideLOD value should
be set to <code>0</code>). NDs should <em>NOT</em> be coded as missing
(<code>NA</code>). <code>lodGWAS</code> can handle multiple lower and upper
LOD levels (e.g. as a result from different assays used to
measure the biomarker) in a single file. In that case the
phenotype of an ND should be changed to the lower/upper LOD
level of the assay type used for that individual.
</p>
<p>4) The column phenotype can be either raw or transformed values
of the phenotype. Please take care that NDs (whose phenotype
value equals the LOD) must also be transformed appropriately.
</p>


<h3>Output File Format</h3>

<p>Column descriptions of the output file (as per default settings,
with <code>outputheader="QCGWAS"</code>) are as following: 
</p>

<ul>
<li><p> MARKER: marker ID (identifier of the genetic variant)
as specified in the genotype input files
</p>
</li>
<li><p> CHR: chromosome as specified in the genotype map file
</p>
</li>
<li><p> POSITION: physical position (base-pair position) as
specified in the genotype map file
</p>
</li>
<li><p> OTHER_ALL: non-effect allele (non-coded allele)
</p>
</li>
<li><p> EFFECT_ALL: effect allele (coded allele)
</p>
</li>
<li><p> N_TOTAL: total sample size, including all NDs as well
as valid measured values
</p>
</li>
<li><p> N_VALID: the sample size of valid measured values
(excluding all NDs). This is useful if the user wants to
know the percentage of NDs to the total sample size.
</p>
</li>
<li><p> EFF_ALL_FREQ: effect allele frequency 
</p>
</li>
<li><p> EFFECT: effect size (beta) of effect allele
</p>
</li>
<li><p> STDERR: standard error of effect allele
</p>
</li>
<li><p> PVALUE: p-value of association 
</p>
</li>
<li><p> IMP_QUALITY: imputation quality of the genetic variant
</p>
</li></ul>

<p>If another output format is chosen, the same columns will be
present in the output file, but with header names as required
by the specified software program.
</p>


<h3>Note</h3>

<p>GWAS analysis will not be performed: 1) on rare
genetic variants (with allele frequency &lt;0.001 or &gt;0.999),
and 2) on badly imputed genetic variants (with imputation
quality score &lt; 0.01). Those genetic variants will be
included in the output file, but the association results will
be <code>NA</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># For use in this example, the 3 Sample files in the
# extdata folder of the lodGWAS library will be copied
# to your current R working directory
  
## Not run: 

file.copy(from = file.path(system.file("extdata", package = "lodGWAS"), "Sample_geno.dose"),
               to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "lodGWAS"), "Sample_geno.map"),
               to = getwd(), overwrite = FALSE, recursive = FALSE)
file.copy(from = file.path(system.file("extdata", package = "lodGWAS"), "Sample_pheno.txt"),
               to = getwd(), overwrite = FALSE, recursive = FALSE)

lod_GWAS(phenofile = "Sample_pheno.txt", pheno_name = "outcome1",
         basic_model = "sex",
         mapfile = "Sample_geno.map", genofile = "Sample_geno.dose",
         outputfile = "Sample_output.txt", gzip_output = FALSE,
         lower_limit = 0.1, upper_limit = 2)

## End(Not run)
</code></pre>

<hr>
<h2 id='lod_QC'>Checking the coding of phenotype and outsideLOD values</h2><span id='topic+lod_QC'></span>

<h3>Description</h3>

<p>It is important that the phenotype and outsideLOD values in
the phenotype file are coded correctly. The function <code>lod_QC</code>
checks the quality of the phenotype file, and provides a
number of descriptive statistics about the phenotype, as well
as warnings about suspected problems. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lod_QC(phenofile,
      pheno_name,
      filedirectory = getwd(),
      outputfile = "lodQC",
      lower_limit = NA, upper_limit = NA,
      stop_if_error = FALSE,
      reprint_warnings = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lod_QC_+3A_phenofile">phenofile</code></td>
<td>

<p>either a data frame containing the phenotype (and covariate)
values, or the filename (including the extension) of a data
file containing the same. For details on the required format
of the phenotype file, see the section on File Format in the
description of <code><a href="#topic+lod_GWAS">lod_GWAS</a></code>.
</p>
</td></tr>
<tr><td><code id="lod_QC_+3A_pheno_name">pheno_name</code></td>
<td>

<p>the name of the column in phenofile contain the phenotype
values.
</p>
</td></tr>
<tr><td><code id="lod_QC_+3A_filedirectory">filedirectory</code></td>
<td>

<p>the directory that contains the phenotype file
and where the output file will be saved. Please note that R
uses <em>forward</em> slash (/) where Windows uses backslash
(\). The default setting is current R working directory.
</p>
</td></tr>
<tr><td><code id="lod_QC_+3A_outputfile">outputfile</code></td>
<td>

<p>the name for the output file.
</p>
</td></tr>
<tr><td><code id="lod_QC_+3A_lower_limit">lower_limit</code>, <code id="lod_QC_+3A_upper_limit">upper_limit</code></td>
<td>

<p>two numeric values specifying the lower and upper LOD of the
assay. Defaults are <code>NA</code>, in which case <code>lod_QC</code> only
checks for gaps between measured and censored values. 
</p>
</td></tr>
<tr><td><code id="lod_QC_+3A_stop_if_error">stop_if_error</code></td>
<td>

<p>logical; determines whether the function aborts or continues
when it encounters errors in the dataset. Default is <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="lod_QC_+3A_reprint_warnings">reprint_warnings</code></td>
<td>

<p>logical; this argument is used by <code><a href="#topic+lod_GWAS">lod_GWAS</a></code>
and isn't relevant for users. Setting it to <code>TRUE</code> will
cause certain errors to be reported via <code><a href="base.html#topic+warning">warning</a></code>
(in addition to the console and the log file). Default is
<code>FALSE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The user can specify values for the lower and upper LOD. The
function will then compare the values in the column outsideLOD
with the phenotype values, conditional on the user-specified
LOD. For details on the required format of the phenotype file,
see the section on File Format in the description of
<code><a href="#topic+lod_GWAS">lod_GWAS</a></code>.
</p>
<p>Also, <code>lod_QC</code> will check if there is a gap between the
smallest/largest values within the LOD (i.e. truly measured
values, where outsideLOD is <code>1</code>), and the censored values.
If the censored values are far smaller/larger than
smallest/largest measured value, <code>lod_QC</code> will print a
warning. An example
of such an erroneous coding of censored values is
when the phenotype values below the lower LOD have been set
to -9, while the measured values are &gt;0.
</p>


<h3>Value</h3>

<p>The function <code>lod_QC</code> returns an invisible <code>NULL</code>.
The real output is a log file containing information on the
distribution of the phenotype, and any problems it
encountered. These warnings are also displayed in the R console.
</p>


<h3>Note</h3>

<p>The function <code>lod_QC</code> only checks and reports. It does
<em>not</em> correct the phenotype or outsideLOD values.
</p>
<p>Also, the function assumes that there is a single lower and/or
upper LOD. If multiple LODs are used in the file, the function
may produce warnings when both censored and real values
do not match the specified limits.  In that case it is up to
the user to ensure the quality of the phenotype file.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lod_GWAS">lod_GWAS</a></code> for the complete GWAS function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phenos &lt;- data.frame(FID = c(1, 2, 3, 4, 5, 6, 7, 8, 9, 10),
                     IID = c("female1", "male2", "male3", "female4", "female5",
                             "male6", "female7", "male8", "male9", "male10"),
                     sex = c(0, 1, 1, 0, 0, 1, 0, 1, 1, 1),
                     outcome1 = c(0.3, 0.5, 0.9, 0.7, 2, 2, 0.1, 1.1, 2, 0.7),
                     outsideLOD = as.integer(c(1, 1, 1, 1, 1, 0, 2, 1, 0, 1)),
                     stringsAsFactors = FALSE)

lod_QC(phenofile = phenos, pheno_name = "outcome1",
       outputfile = "Sample_QC", lower_limit = 0.1, upper_limit = 2)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
