<!DOCTYPE html><html><head><title>Help for package lolliplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lolliplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#lolliplot'><p>Lolliplots</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Plot Variants and Somatic Mutations</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zhang Renliang &lt;zhang_renliang@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Draw lolliplot using GRanges objects. this package was designed 
  only for drawing lolliplot. So, it's faster than 'trackViewer', but un-related 
  functions has been derived.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, scales, grDevices, IRanges, grid, grImport,
GenomicRanges</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-01-26 12:25:54 UTC; zhang</td>
</tr>
<tr>
<td>Author:</td>
<td>Zhang Renliang [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-01-26 18:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='lolliplot'>Lolliplots</h2><span id='topic+lolliplot'></span>

<h3>Description</h3>

<p>Plot variants and somatic mutations.
Jianhong Ou and Lihua Julie Zhu (2019)&lt;doi:10.1038/s41592-019-0430-y&gt;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lolliplot(
  SNP.gr,
  features = NULL,
  ranges = NULL,
  type = "circle",
  newpage = TRUE,
  ylab = TRUE,
  ylab.gp = gpar(col = "black"),
  yaxis = TRUE,
  yaxis.gp = gpar(col = "black"),
  xaxis = TRUE,
  xaxis.gp = gpar(col = "black"),
  legend = NULL,
  cex = 1,
  dashline.col = "gray80",
  jitter = c("node", "label"),
  rescale = FALSE,
  label_on_feature = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lolliplot_+3A_snp.gr">SNP.gr</code></td>
<td>
<p>A object of <a href="GenomicRanges.html#topic+GRanges-class">GRanges</a>,
<a href="GenomicRanges.html#topic+GRangesList-class">GRangesList</a>
or a list of <a href="GenomicRanges.html#topic+GRanges-class">GRanges</a>.
All the width of GRanges must be 1.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_features">features</code></td>
<td>
<p>A object of <a href="GenomicRanges.html#topic+GRanges-class">GRanges</a>,
<a href="GenomicRanges.html#topic+GRangesList-class">GRangesList</a>
or a list of <a href="GenomicRanges.html#topic+GRanges-class">GRanges</a>.
The metadata 'featureLayerID' are used for drawing features in different layers.
See details in vignette.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_ranges">ranges</code></td>
<td>
<p>A object of <a href="GenomicRanges.html#topic+GRanges-class">GRanges</a> or
<a href="GenomicRanges.html#topic+GRangesList-class">GRangesList</a>.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_type">type</code></td>
<td>
<p>character. Could be circle, pie, pin, pie.stack or flag.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_newpage">newpage</code></td>
<td>
<p>Plot in the new page or not.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_ylab">ylab</code></td>
<td>
<p>Plot ylab or not. If it is a character vector,
the vector will be used as ylab.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_ylab.gp">ylab.gp</code>, <code id="lolliplot_+3A_xaxis.gp">xaxis.gp</code>, <code id="lolliplot_+3A_yaxis.gp">yaxis.gp</code></td>
<td>
<p>An object of class gpar for ylab, xaxis or yaxis.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_yaxis">yaxis</code></td>
<td>
<p>Plot yaxis or not.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_xaxis">xaxis</code></td>
<td>
<p>Plot xaxis or not. If it is a numeric vector with length greater than 1,
the vector will be used as the points at which tick-marks are to be drawn.
And the names of the vector will be used to as labels to be placed at the tick
points if it has names.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_legend">legend</code></td>
<td>
<p>If it is a list with named color vectors, a legend will be added.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_cex">cex</code></td>
<td>
<p>cex will control the size of circle.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_dashline.col">dashline.col</code></td>
<td>
<p>color for the dashed line.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_jitter">jitter</code></td>
<td>
<p>jitter the position of nodes or labels.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_rescale">rescale</code></td>
<td>
<p>logical(1), character(1), numeric vector,
or a dataframe with rescale from and to.
Rescalse the x-axis or not.
if dataframe is used, colnames must be from.start, from.end,
to.start, to.end. And the from scale must cover the whole plot region.
The rescale parameter can be set as &quot;exon&quot; or &quot;intron&quot; to
emphasize &quot;exon&quot; or &quot;intron&quot; region. The &quot;exon&quot; or &quot;intron&quot;
can be followed with an integer e.g. &quot;exon_80&quot;, or &quot;intron_99&quot;.
The integer indicates the total percentage of &quot;exon&quot; or &quot;intron&quot; region.
Here &quot;exon&quot; indicates all regions in features.
And &quot;intron&quot; indicates all flank regions of the features.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_label_on_feature">label_on_feature</code></td>
<td>
<p>Labels of the feature directly on them.
Default FALSE.</p>
</td></tr>
<tr><td><code id="lolliplot_+3A_...">...</code></td>
<td>
<p>not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In SNP.gr and features, metadata of the GRanges object will be used to control the
color, fill, border, alpha, shape, height, cex, dashline.col, data source of pie if the type is pie.
And also the controls for labels by name the metadata start as
label.parameter.&lt;properties&gt;
such as label.parameter.rot, label.parameter.gp. The parameter is used for
<a href="grid.html#topic+grid.text">grid.text</a>. The metadata 'featureLayerID' for features are used
for drawing features in different layers. The metadata 'SNPsideID' for SNP.gr
are used for determining the side of lollipops. And the 'SNPsideID' could only
be 'top' or 'bottom'.
</p>


<h3>Value</h3>

<p>No return value, called for side effects
</p>


<h3>Examples</h3>

<pre><code class='language-R'>SNP &lt;- c(10, 100, 105, 108, 400, 410, 420, 600, 700, 805, 840, 1400, 1402)
x &lt;- sample.int(100, length(SNP))
SNP.gr &lt;- GenomicRanges::GRanges("chr1", IRanges::IRanges(SNP, width=1, names=paste0("snp", SNP)),
                                 value1=x, value2=100-x)
SNP.gr$color &lt;- rep(list(c("red", 'blue')), length(SNP))
SNP.gr$border &lt;- sample.int(7, length(SNP), replace=TRUE)
features &lt;- GenomicRanges::GRanges(
  "chr1", IRanges::IRanges(
    c(1, 501, 1001),
    width=c(120, 500, 405),
    names=paste0("block", 1:3)),
  color="black",
  fill=c("#FF8833", "#51C6E6", "#DFA32D"),
  height=c(0.1, 0.05, 0.08),
  label.parameter.rot=45)
lolliplot(SNP.gr, features, type="pie")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
