<!DOCTYPE html><html lang="en"><head><title>Help for package longreadvqs</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {longreadvqs}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#longreadvqs-package'><p>longreadvqs: Viral Quasispecies Comparison from Long-Read Sequencing Data</p></a></li>
<li><a href='#+25+26gt+3B+25'><p>Pipe operator</p></a></li>
<li><a href='#AAcompare'><p>Comparing viral quasispecies diversity metrics at amino acid level</p></a></li>
<li><a href='#filtfast'><p>Filtering highly dissimilar reads/sequences out of the alignment</p></a></li>
<li><a href='#gapremove'><p>Removing gap-rich positions and/or reads/sequences</p></a></li>
<li><a href='#otucompare'><p>Comparing operational taxonomic unit (OTU) by k-means clustering between samples</p></a></li>
<li><a href='#pctopt'><p>Optimizing cut-off percentage for noise minimization</p></a></li>
<li><a href='#snvcompare'><p>Plotting single nucleotide variant (SNV) frequency in read alignment across different samples</p></a></li>
<li><a href='#vqsassess'><p>Sequencing error and noise minimization, read down-sampling, and data preparation for viral quasispecies comparison</p></a></li>
<li><a href='#vqscompare'><p>Comparing viral quasispecies profile and operational taxonomic unit (OTU) classified by k-means clustering between samples</p></a></li>
<li><a href='#vqscustompct'><p>Sequencing error and noise minimization with customized % cut-off at particular nucleotide region, read down-sampling, and data preparation for viral quasispecies comparison</p></a></li>
<li><a href='#vqsout'><p>Exporting viral quasispecies profile comparison results</p></a></li>
<li><a href='#vqsresub'><p>Computing viral quasispecies diversity metrics of noise-minimized repeatedly down-sampled read alignments</p></a></li>
<li><a href='#vqssub'><p>Computing viral quasispecies diversity metrics of noise-minimized down-sampled read alignment</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Viral Quasispecies Comparison from Long-Read Sequencing Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Performs variety of viral quasispecies diversity analyses [see Pamornchainavakul et al. (2024) &lt;<a href="https://doi.org/10.21203%2Frs.3.rs-4637890%2Fv1">doi:10.21203/rs.3.rs-4637890/v1</a>&gt;] based on long-read sequence alignment. Main functions include 1) sequencing error and other noise minimization and read sampling, 2) Single nucleotide variant (SNV) profiles comparison, and 3) viral quasispecies profiles comparison and visualization. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/NakarinP/longreadvqs">https://github.com/NakarinP/longreadvqs</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/NakarinP/longreadvqs/issues">https://github.com/NakarinP/longreadvqs/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, Biostrings, cowplot, dplyr, ggplot2, ggpubr, grDevices,
IRanges, magrittr, methods, plyr, purrr, QSutils, RColorBrewer,
reshape2, scales, seqinr, stats, stringdist, stringr, tibble,
tidyr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-08-26 18:21:05 UTC; nakarinpamornchainavakul</td>
</tr>
<tr>
<td>Author:</td>
<td>Nakarin Pamornchainavakul
    <a href="https://orcid.org/0000-0003-0378-0316"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nakarin Pamornchainavakul &lt;pamornakarin@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-26 19:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='longreadvqs-package'>longreadvqs: Viral Quasispecies Comparison from Long-Read Sequencing Data</h2><span id='topic+longreadvqs'></span><span id='topic+longreadvqs-package'></span>

<h3>Description</h3>

<p>Performs variety of viral quasispecies diversity analyses [see Pamornchainavakul et al. (2024) <a href="https://doi.org/10.21203/rs.3.rs-4637890/v1">doi:10.21203/rs.3.rs-4637890/v1</a>] based on long-read sequence alignment. Main functions include 1) sequencing error and other noise minimization and read sampling, 2) Single nucleotide variant (SNV) profiles comparison, and 3) viral quasispecies profiles comparison and visualization.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Nakarin Pamornchainavakul <a href="mailto:pamornakarin@gmail.com">pamornakarin@gmail.com</a> (<a href="https://orcid.org/0000-0003-0378-0316">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/NakarinP/longreadvqs">https://github.com/NakarinP/longreadvqs</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/NakarinP/longreadvqs/issues">https://github.com/NakarinP/longreadvqs/issues</a>
</p>
</li></ul>


<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='AAcompare'>Comparing viral quasispecies diversity metrics at amino acid level</h2><span id='topic+AAcompare'></span>

<h3>Description</h3>

<p>Pools noise-minimized down-sampled read samples and compares their diversity metrices based on protein haplotype and single amino acid variation (SAV) group that is classified by k-means clustering of SAV distance. This function is a subset of &quot;vqscompare&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AAcompare(
  samplelist = list(BC1, BC2, BC3),
  kmeans.n = 20,
  removestopcodon = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AAcompare_+3A_samplelist">samplelist</code></td>
<td>
<p>List of samples, i.e., name of resulting objects from &quot;vqsassess&quot; or &quot;vqscustompct&quot; functions, for example list(BC1, BC2, BC3).</p>
</td></tr>
<tr><td><code id="AAcompare_+3A_kmeans.n">kmeans.n</code></td>
<td>
<p>Number of single amino acid variation (SAV) groups needed from k-means clustering on multidimensional scale (MDS) of all samples' pairwise SAV distance.</p>
</td></tr>
<tr><td><code id="AAcompare_+3A_removestopcodon">removestopcodon</code></td>
<td>
<p>Remove the last amino acid (expected to be a stop codon) from translated amino acid sequences before further analysis (optional). If not specified or if removestopcodon = FALSE, the last amino acid will not be removed (default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of 1) &quot;aadiv&quot;: comparative table of viral quasispecies diversity metrics between listed samples based on translated reads calculated by QSutils package, and 2) &quot;savgrpdiv&quot;: comparative table of single amino acid (SAV) group diversity metrics between listed samples calculated from consensus amino acid sequence of each SAV group
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
set.seed(123)
sample1 &lt;- vqsassess(sample1filepath, pct = 5, samsize = 50, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 5, samsize = 50, label = "sample2")

## Compare protein haplotype and SAV group (4 clusters) diversity metrics between two samples-----
AAcompare(samplelist = list(sample1, sample2), kmeans.n = 4)

</code></pre>

<hr>
<h2 id='filtfast'>Filtering highly dissimilar reads/sequences out of the alignment</h2><span id='topic+filtfast'></span>

<h3>Description</h3>

<p>Removes reads/sequences of which Hamming similarity to the consensus of all reads/sequences in the alignment is less than the specified quantile (qt) of the similarity distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filtfast(fasta, qt = 0.25, fastaname = "filteredfast.fasta")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filtfast_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read or multiple sequence alignment in FASTA format</p>
</td></tr>
<tr><td><code id="filtfast_+3A_qt">qt</code></td>
<td>
<p>If Hamming similarity score of a read/sequence to the consensus of all reads/sequences is less than the specified quantile (qt) of the similarity distribution, that read/sequence will be removed.</p>
</td></tr>
<tr><td><code id="filtfast_+3A_fastaname">fastaname</code></td>
<td>
<p>Output file name in FASTA format</p>
</td></tr>
</table>


<h3>Value</h3>

<p>FASTA read or multiple sequence alignment written out to the input directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "dissimfast.fasta", package = "longreadvqs")

## Indicate output directory and file name---------------------------------------------------------
outfast &lt;- tempfile()

## Remove reads/sequences that the similarity &lt; 1st quartile (0.25 quantile)-----------------------
filtfast(fastafilepath, qt = 0.25, fastaname = outfast)

</code></pre>

<hr>
<h2 id='gapremove'>Removing gap-rich positions and/or reads/sequences</h2><span id='topic+gapremove'></span>

<h3>Description</h3>

<p>Removes nucleotide positions (vertical) and/or reads/sequences (horizontal) that contain gaps more than the specified cut-off percentage from the alignment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gapremove(fasta, vgappct = 70, hgappct = 70, fastaname = "filteredfast.fasta")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gapremove_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read or multiple sequence alignment in FASTA format</p>
</td></tr>
<tr><td><code id="gapremove_+3A_vgappct">vgappct</code></td>
<td>
<p>The percent cut-off of vertical gap (-), i.e., if a position in the alignment has %gap &gt;= vgappct, that position will be removed.</p>
</td></tr>
<tr><td><code id="gapremove_+3A_hgappct">hgappct</code></td>
<td>
<p>The percent cut-off of horizontal gap (-), i.e., if a sequence or read in the alignment has %gap &gt;= hgappct, that sequence or read will be removed.</p>
</td></tr>
<tr><td><code id="gapremove_+3A_fastaname">fastaname</code></td>
<td>
<p>Output file name in FASTA format</p>
</td></tr>
</table>


<h3>Value</h3>

<p>FASTA read or multiple sequence alignment written out to the input directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "gaprichfast.fasta", package = "longreadvqs")

## Indicate output directory and file name---------------------------------------------------------
outfast &lt;- tempfile()

## Remove positions with gap &gt;= 60% and reads/sequences with gap &gt;= 10%----------------------------
gapremove(fastafilepath, vgappct = 60, hgappct = 10, fastaname = outfast)

</code></pre>

<hr>
<h2 id='otucompare'>Comparing operational taxonomic unit (OTU) by k-means clustering between samples</h2><span id='topic+otucompare'></span>

<h3>Description</h3>

<p>Pools noise-minimized down-sampled read samples and compares their diversity based on operational taxonomic unit (OTU) classified by k-means clustering of single nucleotide variant (SNV) distance. This function is a subset of &quot;vqscompare&quot; function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otucompare(samplelist = list(BC1, BC2, BC3), kmeans.n = 20)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="otucompare_+3A_samplelist">samplelist</code></td>
<td>
<p>List of samples, i.e., name of resulting objects from &quot;vqsassess&quot; or &quot;vqscustompct&quot; functions, for example list(BC1, BC2, BC3).</p>
</td></tr>
<tr><td><code id="otucompare_+3A_kmeans.n">kmeans.n</code></td>
<td>
<p>Number of operational taxonomic units (OTUs) needed from k-means clustering on multidimensional scale (MDS) of all samples' pairwise genetic distance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Comparative table of OTU diversity metrics between listed samples calculated from consensus sequence of each OTU by QSutils package
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
sample1 &lt;- vqsassess(sample1filepath, pct = 10, samsize = 20, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 10, samsize = 20, label = "sample2")

## Compare OTU (4 clusters) diversity metrics between two samples---------------------------------
otucompare(samplelist = list(sample1, sample2), kmeans.n = 4)

</code></pre>

<hr>
<h2 id='pctopt'>Optimizing cut-off percentage for noise minimization</h2><span id='topic+pctopt'></span>

<h3>Description</h3>

<p>Finds an optimal cut-off percentage for noise minimization (in vqssub, vqsassess, and vqscustompct functions) that can decrease the number of singleton haplotypes to less than the desired percentage of the total reads.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pctopt_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read alignment in FASTA format</p>
</td></tr>
<tr><td><code id="pctopt_+3A_pctsing">pctsing</code></td>
<td>
<p>The desired percentage of singleton haplotypes relative to the total reads in the alignment.</p>
</td></tr>
<tr><td><code id="pctopt_+3A_method">method</code></td>
<td>
<p>Sequencing error and noise minimization methods that replace low frequency nucleotide base (less than the &quot;pct&quot; cut-off) with consensus base of that position (&quot;conbase&quot;: default) or with base of the dominant haplotype (&quot;domhapbase&quot;).</p>
</td></tr>
<tr><td><code id="pctopt_+3A_samplingfirst">samplingfirst</code></td>
<td>
<p>Downsampling before (TRUE) or after (FALSE: default) the noise minimization.</p>
</td></tr>
<tr><td><code id="pctopt_+3A_gappct">gappct</code></td>
<td>
<p>The percent cut-off particularly specified for gap (-). If it is not specified or less than &quot;pct&quot;, &quot;gappct&quot; will be equal to &quot;pct&quot; (default).</p>
</td></tr>
<tr><td><code id="pctopt_+3A_ignoregappositions">ignoregappositions</code></td>
<td>
<p>Replace all nucleotides in the positions in the alignment containing gap(s) with gap. This will make such positions no longer single nucleotide variant (SNV). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="pctopt_+3A_samsize">samsize</code></td>
<td>
<p>Sample size (number of reads) after down-sampling. If it is not specified or more than number of reads in the original alignment, down-sampling will not be performed (default).</p>
</td></tr>
<tr><td><code id="pctopt_+3A_label">label</code></td>
<td>
<p>String within quotation marks indicating name of read alignment (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An optimal cut-off percentage for noise minimization of an input sample and parameter settings. If label is specified, the output will be a data frame with percentage of singleton haplotypes at each cut-off percentage from zero to the optimal cut-off percentage.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")

## Find an cut-off percentage that creates singleton haplotypes less than 50% of the alignment.----
pctopt(fastafilepath, pctsing = 50, label = "s1")

</code></pre>

<hr>
<h2 id='snvcompare'>Plotting single nucleotide variant (SNV) frequency in read alignment across different samples</h2><span id='topic+snvcompare'></span>

<h3>Description</h3>

<p>Compares single nucleotide variant (SNV) profile between noise-minimized down-sampled read samples using cowplot's &quot;plot_grid&quot; function. The resulting plot may help evaluating the optimal cut-off percentage of low frequency nucleotide base used in &quot;vqsassess&quot;, &quot;vqscustompct&quot;, or &quot;vqssub&quot; functions.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="snvcompare_+3A_samplelist">samplelist</code></td>
<td>
<p>List of samples, i.e., name of resulting objects from &quot;vqsassess&quot; or &quot;vqscustompct&quot; functions, for example list(BC1, BC2, BC3).</p>
</td></tr>
<tr><td><code id="snvcompare_+3A_ncol">ncol</code></td>
<td>
<p>Number of columns for multiple plots (see cowplot's &quot;plot_grid&quot; function)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Comparative plot of SNV frequency in read alignment across different samples
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
sample1 &lt;- vqsassess(sample1filepath, pct = 10, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 10, label = "sample2")

## Compare SNV profile between two listed samples-------------------------------------------------
snvcompare(samplelist = list(sample1, sample2), ncol = 1)

</code></pre>

<hr>
<h2 id='vqsassess'>Sequencing error and noise minimization, read down-sampling, and data preparation for viral quasispecies comparison</h2><span id='topic+vqsassess'></span>

<h3>Description</h3>

<p>Minimizes potential long-read sequencing error and noise based on the specified cut-off percentage of low frequency nucleotide base and down-samples read for further comparison with other samples. The output of this function is a list of several objects representing diversity of each sample that must be used as an input for other functions such as &quot;snvcompare&quot; or &quot;vqscompare&quot;.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vqsassess_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read alignment in FASTA format</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_method">method</code></td>
<td>
<p>Sequencing error and noise minimization methods that replace low frequency nucleotide base (less than the &quot;pct&quot; cut-off) with consensus base of that position (&quot;conbase&quot;: default) or with base of the dominant haplotype (&quot;domhapbase&quot;).</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_samplingfirst">samplingfirst</code></td>
<td>
<p>Downsampling before (TRUE) or after (FALSE: default) the noise minimization.</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_pct">pct</code></td>
<td>
<p>Percent cut-off defining low frequency nucleotide base that will be replaced (must be specified).</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_gappct">gappct</code></td>
<td>
<p>The percent cut-off particularly specified for gap (-). If it is not specified or less than &quot;pct&quot;, &quot;gappct&quot; will be equal to &quot;pct&quot; (default).</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_ignoregappositions">ignoregappositions</code></td>
<td>
<p>Replace all nucleotides in the positions in the alignment containing gap(s) with gap. This will make such positions no longer single nucleotide variant (SNV). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_samsize">samsize</code></td>
<td>
<p>Sample size (number of reads) after down-sampling. If it is not specified or more than number of reads in the original alignment, down-sampling will not be performed (default).</p>
</td></tr>
<tr><td><code id="vqsassess_+3A_label">label</code></td>
<td>
<p>String within quotation marks indicating name of read alignment (optional). Please don't use underscore (_) in the label.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of 1) &quot;dat&quot;: viral quasispecies diversity metrics calculated by QSutils package (similar to &quot;vqssub&quot; function's output), 2) &quot;snvhap&quot;: SNV profile of each haplotype with frequency and new label for &quot;vqscompare&quot; function, 3) &quot;snv&quot;: plot of SNV frequency for &quot;snvcompare&quot; function, 4) &quot;hapre&quot;: DNAStringSet of read alignment of each haplotype for &quot;vqscompare&quot; function, 5) &quot;lab&quot;: name of sample or read alignment
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file------------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison ------------------------------------------------
sample1 &lt;- vqsassess(sample1filepath, pct = 10, samsize = 20, label = "sample1")

## For more examples on other choices of arguments, please see "vqssub" function's examples-------

</code></pre>

<hr>
<h2 id='vqscompare'>Comparing viral quasispecies profile and operational taxonomic unit (OTU) classified by k-means clustering between samples</h2><span id='topic+vqscompare'></span>

<h3>Description</h3>

<p>Pools noise-minimized down-sampled read samples and compares their diversity by 1) viral quasispecies profile (haplotype and metrics from QSutils package), 2) operational taxonomic unit (OTU) classified by k-means clustering of single nucleotide variant (SNV) distance, and 3) visualization of different comparative method, i.e., haplotype, OTU, phylogenetic tree, MDS plot. Such comparisons can also be performed at the amino acid level (protein haplotype and single amino acid variation (SAV) group).
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vqscompare_+3A_samplelist">samplelist</code></td>
<td>
<p>List of samples, i.e., name of resulting objects from &quot;vqsassess&quot; or &quot;vqscustompct&quot; functions, for example list(BC1, BC2, BC3).</p>
</td></tr>
<tr><td><code id="vqscompare_+3A_lab_name">lab_name</code></td>
<td>
<p>Name of variable or type of sample for instance &quot;barcode&quot;, &quot;sample&quot;, &quot;dpi&quot;, or &quot;isolate&quot; (optional).</p>
</td></tr>
<tr><td><code id="vqscompare_+3A_kmeans.n">kmeans.n</code></td>
<td>
<p>Number of operational taxonomic units (OTUs) and single amino acid variation (SAV) groups needed from k-means clustering on multidimensional scale (MDS) of all samples' pairwise SNV and SAV distances.</p>
</td></tr>
<tr><td><code id="vqscompare_+3A_showhap.n">showhap.n</code></td>
<td>
<p>Number of largest haplotypes (default = 30) labeled in the top five OTUs' MDS plot (optional).</p>
</td></tr>
<tr><td><code id="vqscompare_+3A_proteincoding">proteincoding</code></td>
<td>
<p>Translate gene or protein-coding reads into amino acid sequences and regroup them into protein haplotypes and single amino acid variation (SAV) groups, which are comparable to haplotypes and OTUs at the nucleotide level, respectively (optional). If not specified or if proteincoding = FALSE, gene translation and downstream analyses will not be performed (default).</p>
</td></tr>
<tr><td><code id="vqscompare_+3A_removestopcodon">removestopcodon</code></td>
<td>
<p>Remove the last amino acid (expected to be a stop codon) from translated amino acid sequences before further analysis (optional). If not specified or if removestopcodon = FALSE, the last amino acid will not be removed (default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of 1) &quot;hapdiv&quot;: comparative table of viral quasispecies diversity metrics between listed samples calculated by QSutils package, 2) &quot;otudiv&quot;: comparative table of operational taxonomic unit (OTU) diversity metrics between listed samples calculated from consensus sequence of each OTU (similar to &quot;otucompare&quot; function's output), 3) &quot;sumsnv_hap_otu&quot;: frequency and SNV profile (by position in the alignment) of all haplotypes and OTUs, 4) &quot;fullseq&quot;: complete read sequence of all haplotypes, 5) &quot;fulldata&quot;: complete read sequence of all haplotypes in every sample with frequency and OTU classification, 6) &quot;summaryplot&quot;: visualization of viral quasispecies comparison between samples including  6.1) &quot;happlot&quot;: proportion of haplotypes (top left), 6.2) &quot;otuplot&quot;: proportion of OTUs (bottom left), 6.3) multidimensional scale (MDS) plots (right) of k-means OTU (&quot;top5otumds&quot;: 5 largest groups with major haplotypes labeled and &quot;allotumds&quot;: all groups), 7) &quot;aadiv&quot;: comparative table of viral quasispecies diversity metrics between listed samples based on translated reads calculated by QSutils package (similar to one of &quot;AAcompare&quot; function's outputs), 8) &quot;savgrpdiv&quot;: comparative table of single amino acid (SAV) group diversity metrics between listed samples calculated from consensus amino acid sequence of each SAV group (similar to one of &quot;AAcompare&quot; function's outputs), 9) &quot;sumsav_phap_savgrp&quot;: frequency and SAV profile (by position in the alignment) of all protein haplotypes and SAV groups, 10) &quot;fullseq_aa&quot;: complete amino acid sequence of all protein haplotypes, 11) &quot;fulldata_aa&quot;: complete amino acid sequence of all protein haplotypes in every sample with frequency and SAV group classification, 12) &quot;summaryplot_aa&quot;: visualization of viral quasispecies comparison between samples based on translated reads including  12.1) &quot;phapplot&quot;: proportion of protein haplotypes (top left), 12.2) &quot;savgrpplot&quot;: proportion of SAV groups (bottom left), 12.3) multidimensional scale (MDS) plots (right) of k-means SAV group (&quot;top5savgrpmds&quot;: 5 largest SAV groups with major protein haplotypes labeled and &quot;allsavgrpmds&quot;: all SAV groups), 7) to 12) will be generated only when proteincoding = TRUE
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
set.seed(123)
sample1 &lt;- vqsassess(sample1filepath, pct = 5, samsize = 50, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 5, samsize = 50, label = "sample2")

## Compare viral quasispecies and OTU (4 clusters) diversity between two samples------------------
out &lt;- vqscompare(samplelist = list(sample1, sample2),
           lab_name = "Sample", kmeans.n = 4, showhap.n = 5)
out$summaryplot

</code></pre>

<hr>
<h2 id='vqscustompct'>Sequencing error and noise minimization with customized % cut-off at particular nucleotide region, read down-sampling, and data preparation for viral quasispecies comparison</h2><span id='topic+vqscustompct'></span>

<h3>Description</h3>

<p>Minimizes potential long-read sequencing error and noise based on the specified cut-off percentages of low frequency nucleotide base and down-samples read for further comparison with other samples. In this function, the cut-off percentage can be specifically adjusted for different ranges of nucleotide positions which is very useful when sequencing error heavily occurs in a particular part of reads. The output of this function is a list of several objects representing diversity of each sample that must be used as an input for other functions such as &quot;snvcompare&quot; or &quot;vqscompare&quot;.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vqscustompct_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read alignment in FASTA format</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_method">method</code></td>
<td>
<p>Sequencing error and noise minimization methods that replace low frequency nucleotide base (less than the &quot;pct&quot; cut-off) with consensus base of that position (&quot;conbase&quot;: default) or with base of the dominant haplotype (&quot;domhapbase&quot;).</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_samplingfirst">samplingfirst</code></td>
<td>
<p>Downsampling before (TRUE) or after (FALSE: default) the noise minimization.</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_pct">pct</code></td>
<td>
<p>Percent cut-off defining low frequency nucleotide base that will be replaced (must be specified).</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_brkpos">brkpos</code></td>
<td>
<p>Ranges of nucleotide positions with different % cut-off specified in &quot;lspct&quot; for example c(&quot;1:50&quot;,&quot;51:1112&quot;) meaning that the first and the second ranges are nucleotide positions 1 to 50 and 51 to 1112, respectively.</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_lspct">lspct</code></td>
<td>
<p>List of customized % cut-off applied to nucleotide ranges set in &quot;brkpos&quot; for example c(15,8) meaning that 15% and 8% cut-offs will be applied to the first and the second ranges, respectively.</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_gappct">gappct</code></td>
<td>
<p>The percent cut-off particularly specified for gap (-). If it is not specified or less than &quot;pct&quot;, &quot;gappct&quot; will be equal to &quot;pct&quot; (default).</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_ignoregappositions">ignoregappositions</code></td>
<td>
<p>Replace all nucleotides in the positions in the alignment containing gap(s) with gap. This will make such positions no longer single nucleotide variant (SNV). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_samsize">samsize</code></td>
<td>
<p>Sample size (number of reads) after down-sampling. If it is not specified or more than number of reads in the original alignment, down-sampling will not be performed (default).</p>
</td></tr>
<tr><td><code id="vqscustompct_+3A_label">label</code></td>
<td>
<p>String within quotation marks indicating name of read alignment (optional). Please don't use underscore (_) in the label.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of 1) &quot;dat&quot;: viral quasispecies diversity metrics calculated by QSutils package (similar to &quot;vqssub&quot; function's output), 2) &quot;snvhap&quot;: SNV profile of each haplotype with frequency and new label for &quot;vqscompare&quot; function, 3) &quot;snv&quot;: plot of SNV frequency for &quot;snvcompare&quot; function, 4) &quot;hapre&quot;: DNAStringSet of read alignment of each haplotype for &quot;vqscompare&quot; function, 5) &quot;lab&quot;: name of sample or read alignment
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "badend.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison using 10% cut-off across all positions----------
nocustom &lt;- vqsassess(fastafilepath, pct = 10, label = "nocustom")

## Prepare data using 10% cut-off for the first 74 positions and 30% cut-off for the rest---------
custom &lt;- vqscustompct(fastafilepath, pct = 10,
                       brkpos = c("1:74","75:84"), lspct = c(10,30), label = "custom")

## Use "snvcompare" function to check whether SNV profile looks better or not---------------------
snvcompare(samplelist = list(nocustom, custom), ncol = 1)

</code></pre>

<hr>
<h2 id='vqsout'>Exporting viral quasispecies profile comparison results</h2><span id='topic+vqsout'></span>

<h3>Description</h3>

<p>Writes out resulting objects from &quot;vqscompare&quot; function as tables (TSV files) and alignment (FASTA file) to the working directory.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vqsout(vqscompare.obj, directory = "path/to/directory")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vqsout_+3A_vqscompare.obj">vqscompare.obj</code></td>
<td>
<p>A resulting object from &quot;vqscompare&quot; function.</p>
</td></tr>
<tr><td><code id="vqsout_+3A_directory">directory</code></td>
<td>
<p>Path to desired directory (location) for output files. If it is not specified, the directory will be the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TSV files of viral quasispecies profile comparison results and FASTA file of unique haplotype alignment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA files-----------------------------------------------------------------------
sample1filepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")
sample2filepath &lt;- system.file("extdata", "s2.fasta", package = "longreadvqs")

## Prepare data for viral quasispecies comparison between two samples-----------------------------
set.seed(123)
sample1 &lt;- vqsassess(sample1filepath, pct = 0, samsize = 50, label = "sample1")
sample2 &lt;- vqsassess(sample2filepath, pct = 0, samsize = 50, label = "sample2")

## Compare viral quasispecies and OTU (4 clusters) diversity between two samples------------------
comp &lt;- vqscompare(samplelist = list(sample1, sample2),
           lab_name = "Sample", kmeans.n = 4, showhap.n = 5)

## Export Key outputs from "vqscompare" function--------------------------------------------------
notrun &lt;- vqsout(comp, directory = tempdir())

</code></pre>

<hr>
<h2 id='vqsresub'>Computing viral quasispecies diversity metrics of noise-minimized repeatedly down-sampled read alignments</h2><span id='topic+vqsresub'></span>

<h3>Description</h3>

<p>Minimizes potential long-read sequencing error and noise based on the specified cut-off percentage of low frequency nucleotide base and repeatedly down-samples read for sensitivity analysis of the diversity metrics varied by different sample sizes. The output of this function is a summary of viral quasispecies diversity metrics per each iteration of down-sampling calculated by QSutils package's functions. This function is an extension of &quot;vqssub&quot; function.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vqsresub_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read alignment in FASTA format</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_iter">iter</code></td>
<td>
<p>Number of iterations for downsampling after noise minimization.</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_method">method</code></td>
<td>
<p>Sequencing error and noise minimization methods that replace low frequency nucleotide base (less than the &quot;pct&quot; cut-off) with consensus base of that position (&quot;conbase&quot;: default) or with base of the dominant haplotype (&quot;domhapbase&quot;).</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_pct">pct</code></td>
<td>
<p>Percent cut-off defining low frequency nucleotide base that will be replaced (must be specified).</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_gappct">gappct</code></td>
<td>
<p>The percent cut-off particularly specified for gap (-). If it is not specified or less than &quot;pct&quot;, &quot;gappct&quot; will be equal to &quot;pct&quot; (default).</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_ignoregappositions">ignoregappositions</code></td>
<td>
<p>Replace all nucleotides in the positions in the alignment containing gap(s) with gap. This will make such positions no longer single nucleotide variant (SNV). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_samsize">samsize</code></td>
<td>
<p>Sample size (number of reads) after down-sampling. If it is not specified or more than number of reads in the original alignment, down-sampling will not be performed (default).</p>
</td></tr>
<tr><td><code id="vqsresub_+3A_label">label</code></td>
<td>
<p>String within quotation marks indicating name of read alignment (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing all viral quasispecies diversity metrics calculated by QSutils package, noise minimization, and down-sampling information per each downsampling iteration.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")

## Summarize viral quasispecies diversity metrics from five downsampling iterations.---------------
vqsresub(fastafilepath, iter = 5, pct = 10, samsize = 20, label = "sample1")

</code></pre>

<hr>
<h2 id='vqssub'>Computing viral quasispecies diversity metrics of noise-minimized down-sampled read alignment</h2><span id='topic+vqssub'></span>

<h3>Description</h3>

<p>Minimizes potential long-read sequencing error and noise based on the specified cut-off percentage of low frequency nucleotide base and down-samples read for further comparison with other samples. The output of this function is a summary of viral quasispecies diversity metrics calculated by QSutils package's functions. This function is a subset of &quot;vqsassess&quot; function.
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vqssub_+3A_fasta">fasta</code></td>
<td>
<p>Input as a read alignment in FASTA format</p>
</td></tr>
<tr><td><code id="vqssub_+3A_method">method</code></td>
<td>
<p>Sequencing error and noise minimization methods that replace low frequency nucleotide base (less than the &quot;pct&quot; cut-off) with consensus base of that position (&quot;conbase&quot;: default) or with base of the dominant haplotype (&quot;domhapbase&quot;).</p>
</td></tr>
<tr><td><code id="vqssub_+3A_samplingfirst">samplingfirst</code></td>
<td>
<p>Downsampling before (TRUE) or after (FALSE: default) the noise minimization.</p>
</td></tr>
<tr><td><code id="vqssub_+3A_pct">pct</code></td>
<td>
<p>Percent cut-off defining low frequency nucleotide base that will be replaced (must be specified).</p>
</td></tr>
<tr><td><code id="vqssub_+3A_gappct">gappct</code></td>
<td>
<p>The percent cut-off particularly specified for gap (-). If it is not specified or less than &quot;pct&quot;, &quot;gappct&quot; will be equal to &quot;pct&quot; (default).</p>
</td></tr>
<tr><td><code id="vqssub_+3A_ignoregappositions">ignoregappositions</code></td>
<td>
<p>Replace all nucleotides in the positions in the alignment containing gap(s) with gap. This will make such positions no longer single nucleotide variant (SNV). The default is &quot;FALSE&quot;.</p>
</td></tr>
<tr><td><code id="vqssub_+3A_samsize">samsize</code></td>
<td>
<p>Sample size (number of reads) after down-sampling. If it is not specified or more than number of reads in the original alignment, down-sampling will not be performed (default).</p>
</td></tr>
<tr><td><code id="vqssub_+3A_label">label</code></td>
<td>
<p>String within quotation marks indicating name of read alignment (optional).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing all viral quasispecies diversity metrics calculated by QSutils package, noise minimization, and down-sampling information.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Locate input FASTA file-------------------------------------------------------------------------
fastafilepath &lt;- system.file("extdata", "s1.fasta", package = "longreadvqs")

## Summarize viral quasispecies diversity metrics--------------------------------------------------
# From noise-minimized unsampled reads (10% cut-off):
vqssub(fastafilepath, pct = 10, label = "sample1")
# From noise-minimized sampled reads (n = 20):
vqssub(fastafilepath, pct = 10, samsize = 20, label = "sample1")
# From noise-minimized sampled reads with 50% cut-off for gap:
vqssub(fastafilepath, pct = 10, gappct = 50, samsize = 20, label = "sample1")
# From noise-minimized sampled reads but ignore positions with gap:
vqssub(fastafilepath, pct = 10, ignoregappositions = TRUE, samsize = 20, label = "sample1")
# From reads that were down-sampled before noise minimization:
vqssub(fastafilepath, pct = 10, samplingfirst = TRUE, samsize = 20, label = "sample1")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
