<!DOCTYPE html><html><head><title>Help for package loon.data</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {loon.data}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alaska_forest'><p>Cooperative Alaska Forest Inventory data</p></a></li>
<li><a href='#binaryalphadigits'><p>Binary Alphadigits</p></a></li>
<li><a href='#blocks'><p>A set of 100 plastic &quot;blocks&quot;</p></a></li>
<li><a href='#bone'><p>Relative Spinal Bone Mineral Density Data</p></a></li>
<li><a href='#bone_ext'><p>Spinal Bone Mineral Density (Extended) Data</p></a></li>
<li><a href='#covidNZ'><p>Covid 19 Case Data New Zealand</p></a></li>
<li><a href='#crabSpecies'><p>Colour and Sex of Rock Crabs</p></a></li>
<li><a href='#diabetes'><p>Diabetes Data from Andrews and Herzberg</p></a></li>
<li><a href='#digits'><p>USPS Handwritten Digits</p></a></li>
<li><a href='#elements'><p>The first 118 chemical elements</p></a></li>
<li><a href='#faces'><p>Olivetti Faces</p></a></li>
<li><a href='#frey'><p>Frey Faces</p></a></li>
<li><a href='#igg1'><p>Human immunoglobulin G1 antibody molecule</p></a></li>
<li><a href='#judgment'><p>Judgment samples of plastic blocks</p></a></li>
<li><a href='#lepto'><p>Rock Crabs (Leptograpsus variegatus) data</p></a></li>
<li><a href='#lightspeeds'><p>Historical Determinations of the Speed of Light</p></a></li>
<li><a href='#lizards'><p>A fictional data set on lizards perch choices</p></a></li>
<li><a href='#medicalRecords'><p>Mining medical records (fictional)</p></a></li>
<li><a href='#michelson_1879'><p>A.A. Michelson's 1879 Determinations of the Speed of Light</p></a></li>
<li><a href='#minority'><p>Canadian Visible Minority Data 2006</p></a></li>
<li><a href='#ordalphadigits'><p>Binary Alphadigits Isomap</p></a></li>
<li><a href='#ordfrey'><p>Frey Faces Isomap</p></a></li>
<li><a href='#pandemic'><p>Fictional pandemic data</p></a></li>
<li><a href='#pkg_data'><p>Simple summary of data available in named packages</p></a></li>
<li><a href='#placenamesCanada'><p>Canadian place names and their geo-locations.</p></a></li>
<li><a href='#SAheart'><p>South African Heart Disease Data</p></a></li>
<li><a href='#SCmolecule'><p>A protein/DNA complex molecule from Saccharomyces Cerevisiae</p></a></li>
<li><a href='#trtPan'><p>Fictional pandemic data (with treatment variable)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Used to Illustrate 'Loon' Functionality</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Data used as examples in the 'loon' package.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://great-northern-diver.github.io/loon.data/">https://great-northern-diver.github.io/loon.data/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-05-13 19:30:25 UTC; rwoldford</td>
</tr>
<tr>
<td>Author:</td>
<td>R. Wayne Oldford [aut, cre],
  Adrian Waddell [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>R. Wayne Oldford &lt;rwoldford@uwaterloo.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-05-13 20:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='alaska_forest'>Cooperative Alaska Forest Inventory data</h2><span id='topic+alaska_forest'></span>

<h3>Description</h3>

<p>Cooperative Alaska Forest Inventory data
</p>


<h3>Format</h3>

<p>A data frame with 4848 rows and 8 variables
</p>

<dl>
<dt>site</dt><dd><p>A number identifying the sample plot at which measurements were taken.</p>
</dd>
<dt>mass</dt><dd><p>The biomass recorded in metric tonnes per hectare (Mg/ha)</p>
</dd>
<dt>speciesCode</dt><dd><p>Code identifying the ranking of the species at the site. One of {<code>sp01</code>, <code>sp02</code>, ... , <code>sp05</code> }
with <code>sp01</code> identifying this record as the most prevalent species at the site when measured,  
<code>sp02</code> the second most prevalent, and so on.</p>
</dd>
<dt>year</dt><dd><p>The year in which the site was visited and the measurements taken.</p>
</dd>
<dt>scientific</dt><dd><p>The scientific name of the species measured</p>
</dd>
<dt>common</dt><dd><p>The common name of the species measured.</p>
</dd>
<dt>longitude</dt><dd><p>The longitude locating the site in degrees.</p>
</dd>
<dt>latitude</dt><dd><p>The latitude locating the site in degrees.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Data on the biomass (Mg/ha) of dominant forest species from several sample plots (sites)
in Alaska, each sampled over several years.  Original data taken from the Cooperative Alaska
Forest Inventory website.
</p>
<p>From website:
</p>
<p>&quot;The Cooperative Alaska Forest Inventory (CAFI) is a comprehensive database of boreal 
forest conditions and dynamics in Alaska. The CAFI consists of field-gathered information
from numerous permanent sample plots distributed across interior and south-central Alaska 
including the Kenai Peninsula. 
The CAFI currently has 570 permanent sample plots on 190 sites representing a wide variety 
of growing conditions. New plots are being added to the inventory annually. (...)
Repeated periodic inventories on CAFI permanent sample plots provide valuable long-term 
information for modeling of forest dynamics such as growth and yield. 
Periodic remeasurements can also be used to test and monitor large-scale environmental 
and climate change.&quot;
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford.
</p>


<h3>Source</h3>

<p>Raw data from the Forest Service of the U.S. Department of Agriculture 
fs.usda.gov publication (No. 32894) website and provided by
</p>
<p>Sol Cooperdock and Brendan Rogers,
Woods Hole Research Center, 149 Woods Hole Road, Falmouth MA 02540, USA and
</p>
<p>Scott Goetz, School of Informatics, Computing, and Cyber Systems, Northern Arizona University, Flagstaff AZ 86011, USA.
</p>
<p>Processed and provided in current form by  R.W. Oldford, Statistics and Actuarial Science, University of Waterloo, Waterloo ON N2L 3G1, Canada.
</p>


<h3>References</h3>

<p>Malone, Thomas; Liang, Jingjing; Packee, Edmond C. 2009. 
Cooperative Alaska Forest Inventory. 
Gen. Tech. Rep. PNW-GTR-785. 
Portland, OR: 
U.S. Department of Agriculture, Forest Service, Pacific Northwest Research Station. 42 p
</p>

<hr>
<h2 id='binaryalphadigits'>Binary Alphadigits</h2><span id='topic+binaryalphadigits'></span>

<h3>Description</h3>

<p>Binary 20x16 digits of &quot;0&quot; through &quot;9&quot; and capital &quot;A&quot; through &quot;Z&quot;. 39 examples of each class.
</p>


<h3>Format</h3>

<p>A data frame with 1,404 rows each representing an image (39 examples of 36 classes)
and 320 variables taking binary values 1 or 0 (black =1 and white = 0 for each pixel value
of a 20x16 alpha-numeric image).
</p>


<h3>Source</h3>

<p>Sam Roweis's data page at cs.nyu.edu.
</p>


<h3>References</h3>

<p>From Simon Lucas' (sml@essex.ac.uk), Algoval system.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ordalphadigits">ordalphadigits</a></code>
</p>

<hr>
<h2 id='blocks'>A set of 100 plastic &quot;blocks&quot;</h2><span id='topic+blocks'></span>

<h3>Description</h3>

<p>A population of 100 &quot;blocks&quot; of different shapes and sizes cut from a single sheet of opaque plastic a few millimetres thick.
</p>
<p>The blocks are of uniform thickness and density (all blocks were cut from the same opaque plastic sheet of
about 5 mm thickness), but have different shapes.  Each block has a number from 1 to 100 etched on one side.
Ambiguous block numbers (e.g. 6 and 9) were disambiguated by placing a decimal place at the bottom of the number (e.g. 6. and 9.).
</p>
<p>The blocks are treated as an entire population of 100 and presented in their entirety, spatially mixed, with the number side up.
Students are asked to visually judge the blocks and to select 10 blocks whose average weight matches the average weight of all 100.
Each student independently selects 10 block ids to serve as their judgment sample (see  <code><a href="#topic+judgment">judgment</a></code>).
</p>
<p>The group variable identifies two strata (&quot;A&quot; or &quot;B&quot;) roughly determined by the apparent area/volume of the blocks.
</p>
<p>The data can be used to illustrate different sampling methods (e.g. simple random sampling, stratified sampling, and regression estimates based on modelling weight as a function of perimeter).
Histograms of resulting estimates (over several samples) can be revealing.
</p>


<h3>Format</h3>

<p>A data frame with 100 rows and 4 variates
</p>

<dl>
<dt>id</dt><dd><p>The id number etched on the block.</p>
</dd>
<dt>weight</dt><dd><p>The block's weight in grams.</p>
</dd>
<dt>perimeter</dt><dd><p>The perimeter length of the block in centimetres.</p>
</dd>
<dt>group</dt><dd><p>Group identification for the block: A are smaller blocks, B are larger.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>R. Jock Mackay and R. Wayne Oldford
</p>


<h3>See Also</h3>

<p><code><a href="#topic+judgment">judgment</a></code>
</p>

<hr>
<h2 id='bone'>Relative Spinal Bone Mineral Density Data</h2><span id='topic+bone'></span>

<h3>Description</h3>

<p>From the web source: 
&quot;Relative spinal bone mineral density measurements on 261 North
American adolescents.  Each value is the difference in spnbmd
taken on two consecutive visits, divided by the average. The age is
the average age over the two visits.&quot;
</p>
<p>The data are a repackaging and extension of the data of the same name
from the now archived (in 2020) of the 2015 'ElemStatLearn' package of Kjetil B. Halvorsen.
</p>
<p>The principal changes here is the renaming of the variables <code>gender</code> and <code>spnbmd</code> AND in 
the addition of a new variable <code>ethnic</code> derived from the <code><a href="#topic+bone_ext">bone_ext</a></code> data set.
</p>
<p>The variables <code>gender</code> and <code>spnbmd</code> were renamed to <code>sex</code> and <code>rspnbmd</code>,
respectively, to better agree with names in the larger and more complete data set <code><a href="#topic+bone_ext">bone_ext</a></code> 
taken from the same study and source webpage.  The variable <code>ethnic</code> was extracted from
<code><a href="#topic+bone_ext">bone_ext</a></code> by matching 'idnum' in the two datasets.
</p>


<h3>Format</h3>

<p>A data frame with 485 rows and 5 variables
</p>

<dl>
<dt>idnum</dt><dd><p>Identifies the subject, and hence the repeat measurements</p>
</dd>
<dt>age</dt><dd><p>Age of subject averaged over the two times measurements of 
<code>spnbmd</code> were taken to determine the relative change <code>rspnbmd</code>.</p>
</dd>
<dt>sex</dt><dd><p>Sex of the subject.  A factor with levels &quot;female&quot; and &quot;male&quot;.</p>
</dd>
<dt>rspnbmd</dt><dd><p>Relative spinal bone mineral density measurement. 
Each value is the (unitless) difference in <code>spnbmd</code>' taken on two 
consecutive visits, divided by the average of the two measurements.</p>
</dd>
<dt>ethnic</dt><dd><p>The &quot;ethnicity/race&quot; of the subject.  A factor with levels 
&quot;Asian&quot;, &quot;Black&quot;, &quot;Hispanic&quot;, and &quot;White&quot;.</p>
</dd>
</dl>

<p>The row order of the values follow their order of appearence in the source webpage.
</p>


<h3>Details</h3>

<p>The purpose of the study was to examine ethnic and sex differences in bone mineral acquisition over
time for young (aged 9-25 years) healthy Asian, black, Hispanic, and white males and females.
The study recorded areal bone mineral density (BMD) in grams per square centimetre in the lumbar spine.
</p>
<p>These data are a subset of 261 subjects taken from a convenience sample of 
423 healthy young people of various &quot;ethnicities.&quot;   
</p>
<p>The source website does not describe how this subset was chosen.
</p>
<p>Rather than the <code>spnbmd</code> measurement at each visit of a subject (as with 'bone_ext'),
the response <code>rspnbmd</code> denotes the relative change in <code>spnbmd</code> between visits and is
calculated as the difference between the later and early visit values of <code>spnbmd</code> divided
by the average of these two values.  The 'age' variable is similarily taken to be
the average of the subject's ages at the two visits.
</p>
<p>On the subjects (Bachrach et al, 1999): 
</p>
<p>&quot;A convenience sample of healthy youth was recruited from the community
through advertisements and personal contact (21, 22). Individuals
with a history of medical conditions or use of medications affecting
bone mineral were excluded. Subjects were encouraged to return annually
for a total of four visits or until they had reached age 26 yr.
Recruitment occurred between May 1992 and February 1996; data collection
ended in February 1997. The cohort at entry included 103 non-Hispanic whites, 
103 Hispanics, 103 Asians, and 114 non-Hispanic blacks, aged 8.8 –25.9 yr; 
230 females and 193 males were enrolled as previously reported (22). 
For simplicity, ethnicity and race will be used as interchangeable terms, 
and the groups will be referred to as white, Hispanic, Asian, and black.
A total of 280 subjects completed 2 visits; 189 were studied 3 times, and 
113 were evaluated 4 times. Subjects who completed fewer than 4 visits included 
those who refused, relocated, or reached age 26 yr during the study period; 
in addition, subjects who were recruited late in the study did not complete 
all visits because funding had terminated.
</p>
<p>So-called &quot;ethnicity&quot; can be found in the data set 'bone_ext'.
</p>
<p>See references, particularly Bachrach et al (1999), for more details.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p>Trevor Hastie's &quot;Elements of Statistical Learning&quot; page at Stanford.
</p>


<h3>References</h3>

<p>Laura K. Bachrach, Trevor Hastie, May-Choo Wang, Balasubramanian Narasimhan, and Robert Marcus (1999) 
&quot;Bone Mineral Acquisition in Healthy Asian, Hispanic, Black and 
Caucasian Youth. A Longitudinal Study&quot;, J Clin Endocrinol Metab, 84, 4702-12.
</p>
<p>Trevor Hastie, Robert Tibshirani, and Jerome Friedman (2009) 
&quot;The Elements of Statistical Learning&quot;, 2nd Edition, Springer New York
&lt;doi:10.1007/978-0-387-84858-7&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bone_ext">bone_ext</a></code>
</p>

<hr>
<h2 id='bone_ext'>Spinal Bone Mineral Density (Extended) Data</h2><span id='topic+bone_ext'></span>

<h3>Description</h3>

<p>These are measures on spinal areal bone mineral density.  The data
are taken from the &quot;large&quot; bone density dataset on the source website (see source).
</p>


<h3>Format</h3>

<p>A data frame with 1003 rows and 5 variables
</p>

<dl>
<dt>idnum</dt><dd><p>Identifies the subject, and hence the repeat measurements</p>
</dd>
<dt>ethnic</dt><dd><p>The &quot;ethnicity/race&quot; of the subject.  A factor with levels 
&quot;Asian&quot;, &quot;Black&quot;, &quot;Hispanic&quot;, and &quot;White&quot;.</p>
</dd>
<dt>age</dt><dd><p>The age in years of the subject at the time the measurement <code>spnbmd</code> was taken.</p>
</dd>
<dt>sex</dt><dd><p>Sex of the subject.  A factor with levels &quot;female&quot; and &quot;male&quot;.</p>
</dd>
<dt>spnbmd</dt><dd><p>The spinal areal bone mineral density (BMD) measurement 
in grams per square centimetre.</p>
</dd>
</dl>

<p>The row order of the values follow their order of appearence in the source webpage.
</p>


<h3>Details</h3>

<p>The purpose of the study was to examine ethnic and sex differences in bone mineral acquisition over
time for young (aged 9-25 years) healthy Asian, Black, Hispanic, and White males and females.
The study recorded areal bone mineral density (BMD) in grams per square centimetre in the lumbar spine.
</p>
<p>The sample was a convenience sample of 423 healthy young people of various &quot;ethnicities.&quot; 
</p>
<p>On the subjects (Bachrach et al, 1999): 
</p>
<p>&quot;A convenience sample of healthy youth was recruited from the community
through advertisements and personal contact. Individuals
with a history of medical conditions or use of medications affecting
bone mineral were excluded. Subjects were encouraged to return annually
for a total of four visits or until they had reached age 26 yr.
Recruitment occurred between May 1992 and February 1996; data collection
ended in February 1997. The cohort at entry included 103 non-Hispanic whites, 
103 Hispanics, 103 Asians, and 114 non-Hispanic blacks, aged 8.8 –25.9 yr; 
230 females and 193 males were enrolled as previously reported (22). 
For simplicity, ethnicity and race will be used as interchangeable terms, 
and the groups will be referred to as white, Hispanic, Asian, and black.
A total of 280 subjects completed 2 visits; 189 were studied 3 times, and 
113 were evaluated 4 times. Subjects who completed fewer than 4 visits included 
those who refused, relocated, or reached age 26 yr during the study period; 
in addition, subjects who were recruited late in the study did not complete 
all visits because funding had terminated.&quot;
</p>
<p>See references, particularly Bachrach et al (1999), for more details.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p>Trevor Hastie's &quot;Elements of Statistical Learning&quot; page at Stanford.
</p>


<h3>References</h3>

<p>Laura K. Bachrach, Trevor Hastie, May-Choo Wang, Balasubramanian Narasimhan, and Robert Marcus (1999) 
&quot;Bone Mineral Acquisition in Healthy Asian, Hispanic, Black and 
Caucasian Youth. A Longitudinal Study&quot;, J Clin Endocrinol Metab, 84, 4702-12.
</p>
<p>Trevor Hastie, Robert Tibshirani, and Jerome Friedman (2009) 
&quot;The Elements of Statistical Learning&quot;, 2nd Edition, Springer New York
&lt;doi:10.1007/978-0-387-84858-7&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bone">bone</a></code>
</p>

<hr>
<h2 id='covidNZ'>Covid 19 Case Data New Zealand</h2><span id='topic+covidNZ'></span>

<h3>Description</h3>

<p>Case data published by the New Zealand Ministry of Health downloaded from the source on May 28, 2020.
</p>
<p>The data consist of 1,154 individuals having been confirmed cases of COVID-19.
Confirmed cases are people that have had a positive laboratory test.
</p>


<h3>Format</h3>

<p>A data frame with 1154 rows and 9 variables
</p>

<dl>
<dt>Case_date</dt><dd><p>The date notified of a potential case.  This variable is of class &quot;Date&quot; in day-month-year format. </p>
</dd>
<dt>Sex</dt><dd><p>The sex of the person, a factor with two levels: Male and Female</p>
</dd>
<dt>Age</dt><dd><p>The age group to which the person belongs.  This is a factor having levels marked in year groups:
&quot;&lt; 1&quot;, &quot;01 to 04&quot;, &quot;05 to 09&quot;, &quot;10 to 14&quot;, &quot;15 to 19&quot;, &quot;20 to 29&quot;, &quot;30 to 39&quot;, &quot;40 to 49&quot;, </p>
</dd>
<dt>District_Health_Board</dt><dd><p>A character vector giving the name of the DHB or District Health Board where the case occurred.</p>
</dd>
<dt>Overseas_travel</dt><dd><p>A character vector indicating whether the patient recently travelled overseas.</p>
</dd>
<dt>Last_country_visited</dt><dd><p>A character vector giving the name, if known, of the last country visited by the person. </p>
</dd>
<dt>Flight_number</dt><dd><p>A character vector recording the number of the flight flown from the last country.</p>
</dd>
<dt>Flight_departure_date</dt><dd><p>A vector of class &quot;Date&quot; giving the flight's departure date from the last country visited.</p>
</dd>
<dt>Arrival_date</dt><dd><p>A vector of class &quot;Date&quot; giving the flight's arrival date from the last country visited.</p>
</dd>
</dl>



<h3>Details</h3>

<p>From the source:  (May 28, 2020: health.gov.nz) 
&quot;The case definition has been amended to de-couple respiratory symptoms 
from a history of travel. Testing is available to people with respiratory 
symptoms suggestive of COVID-19 infection (including the acute onset of 
cough with or without fever). This is regardless of travel history or 
known contact with a confirmed or probable case of COVID-19. 
Priority groups for testing have been included in the case definitions.&quot;
</p>
<p>&quot;Confirmed case:
</p>
<p>&quot;A case that has laboratory definitive evidence. 
Laboratory definitive evidence requires at least one of the following:
* detection of SARS-CoV-2 from a clinical specimen using a validated NAAT (PCR)
* detection of coronavirus from a clinical specimen using pan-coronavirus NAAT (PCR) and confirmation as SARS-CoV-2 by sequencing
* significant rise in IgG antibody level to SARS-CoV-2 between paired sera (when serological testing becomes available).&quot;
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p>New Zealand government health website accessed May 28, 2020.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+igg1">igg1</a></code> <code><a href="#topic+medicalRecords">medicalRecords</a></code> <code><a href="#topic+pandemic">pandemic</a></code>
</p>

<hr>
<h2 id='crabSpecies'>Colour and Sex of Rock Crabs</h2><span id='topic+crabSpecies'></span>

<h3>Description</h3>

<p>A sample of 200 rock crabs (Leptograpsus variegatus) found in the southern subtropical Pacific Ocean.
These are small sea crabs that grow at most to about 50 millimetres of shell width.
</p>
<p>The data contain the sex and species of each crab as determined by the researchers (see references).
</p>


<h3>Format</h3>

<p>A data frame with 200 rows and 2 variables
</p>

<dl>
<dt>Species</dt><dd><p>A two level factor distinguishing the species by colour: either &quot;blue&quot; or &quot;orange&quot; </p>
</dd>
<dt>Sex</dt><dd><p>A two level factor identifying the crab's sex: either &quot;male&quot; or &quot;female&quot;.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The purpose of this data set is to identify which crabs in <code>lepto</code> belong to each of the four groups identified here.
</p>
<p>There are 50 of each combination of factor levels.
</p>
<p>Data are a subset of the <code>crabs</code> data set from the <code>MASS</code> package.  
Only the species and sex variables appear here and their row order here now match (row for row) the
order of the rows from the physical size measurements on the purple rock crab given in <code>lepto</code>.
</p>
<p>This separation now allows clustering methods to be explored on the data set <code>lepto</code> and
compared to the classification by colour (Species) and sex given here.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>N.A. Campbell and R.J. Mahon (1974). &quot;A multivariate study of variation in two species of rock crab of the genus Leptograpsus.&quot;
Australian Journal of Zoology 22, 417-425. &lt;doi:10.1071/ZO9740417&gt;
</p>
<p>Venables, W. N. and Ripley, B. D. (2002) Modern Applied Statistics with S. Fourth edition. Springer.  &lt;doi:10.1007/978-0-387-21706-2&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lepto">lepto</a></code>
</p>

<hr>
<h2 id='diabetes'>Diabetes Data from Andrews and Herzberg</h2><span id='topic+diabetes'></span>

<h3>Description</h3>

<p>Data  on 145 non-obese patients collected at the Stanford Clinical Reseach Center to investigate &quot;the relationship between
chemical subclinical and overt nonketotic diabetes.  The three primary variables used in the analysis and  presented [here] are glucose intolerance, insulin response
to oral glucose, and insulin resistance.&quot;
</p>
<p>From Andrews and Herzberg (1985) book called &quot;Data&quot;. See reference.
</p>


<h3>Format</h3>

<p>A data frame with 145 rows and 6 variables
</p>

<dl>
<dt>RelativeWeight</dt><dd><p>The relative weight of the patient.</p>
</dd>
<dt>FastingPlasmaGlucose</dt><dd><p>The fasting plasma glucose level.</p>
</dd>
<dt>GlucoseArea</dt><dd><p>This is a measurement of the glucose intolerance as measured by the area under the straightline
connecting glucose levels determined from blood samples drawn during a three hour
glucose tolerance test following an oral administration of a glucose load.</p>
</dd>
<dt>InsulinArea</dt><dd><p>This is a measurement of the insulin response
to oral glucose as measured by the area under the straightline
connecting insulin levels, again determined from blood samples drawn during a three hour
glucose tolerance test following an oral administration of a glucose load.</p>
</dd>
<dt>SSPG</dt><dd><p>The steady state plasma glucose (SSPG) determined  after chemical suppression 
of endogenous insulin secretion. This is a measure of insulin resistance.</p>
</dd>
<dt>ClinClass</dt><dd><p>Clinical classification of each patient, by the contemporary (1979) medical criteria,
into one of three groups: &quot;Overt&quot; diabetic,  &quot;Chemical&quot; diabetic, or  &quot;Normal&quot;. </p>
</dd>
</dl>

<p>The row order of the values match the &quot;Patient Number&quot; given in the source table.
</p>


<h3>Details</h3>

<p>This is a dataset from the &quot;Data&quot; book by Andrews and Herzberg (1985) Chapter 36, pp. 215-220  Table 36.1
</p>
<p>A more complete description can be found there. An extract from the source follows.
</p>
<p>The purpose of the data was to investigate
</p>
<p>&quot;the relationship between chemical subclinical
and overt nonketotic diabetes in 145 non-obese
adult subjects.
The three primary variables used in the analysis and
presented [here] are glucose intolerance, insulin response
to oral glucose, and insulin resistance.  The first two
[of these] variables are the areas under the straightline
connecting glucose and insulin levels, respectively,
determined from blood samples drawn during a three hour
glucose tolerance test following an oral administration
of a glucose load. [These are variables GlucoseArea and
InsulinArea, respectively.] Insulin resistance is measured
by the steady state plasma glucose (SSPG) determined
afterchemical suppression of endogenous insulin secretion.
In addition, the relative weight and fasting plasma
glucose were measured for each individual at the Stanford
Clinical Reseach Center and are included [here]&quot;
</p>
<p>Each row of diabetes is a patient, and the row number
is the &quot;Patient Number&quot; from Andrews and Herzberg.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>David F. Andrews and Agnes M. Herzberg (1985) 
&quot;Data: A Collection of Problems from Many Fields for the Student and Research Worker&quot;, 
Springer, New York. &lt;doi:10.1007/978-1-4612-5098-2&gt;
</p>

<hr>
<h2 id='digits'>USPS Handwritten Digits</h2><span id='topic+digits'></span>

<h3>Description</h3>

<p>U.S. Postal Service handwritten digit &quot;0&quot; through &quot;9&quot;.
</p>
<p>Each is an 8-bit 16x16 grayscale image of a single digit;
</p>
<p>1100 examples of each class.
</p>


<h3>Format</h3>

<p>A data frame with 256 rows and 11000 variables.
</p>
<p>Each row corresponds to an 8-bit value of one
of the 256 cells contained in a 16 x 16 image.
</p>
<p>Each column is the 16 x 16 image of a single handwritten digit.
There are 1100 different handwritten images of each digit
appearing in blocks of 1100 columns in the order
&quot;1&quot;, &quot;2&quot;, &quot;3&quot;, &quot;4&quot;, &quot;5&quot;, &quot;6&quot;, &quot;7&quot;, &quot;8&quot;, &quot;9&quot;, &quot;0&quot;
</p>


<h3>Source</h3>

<p>Sam Roweis's data page at cs.nyu.edu.
</p>


<h3>See Also</h3>

<p>binaryalphadigits
</p>

<hr>
<h2 id='elements'>The first 118 chemical elements</h2><span id='topic+elements'></span>

<h3>Description</h3>

<p>The first 118 elements, ordered by atomic number.
</p>
<p>Some values of density, melting point,and boiling point are predictions
rather than measurements. For those elements which do not have any stable
nuclides, the mass number given is that of the longest-lived isotope
(exceptions here are bismuth, thorium, protactinium, and uranium standard
atomic weights are available).  See the source for more detail.
</p>


<h3>Format</h3>

<p>A data frame with 118 rows and 17 variates:
</p>

<dl>
<dt>Number</dt><dd><p>The atomic number.  The number of protons found the nucleus of every atom of the element.
It is the (positive) charge number of the element's nucleus.</p>
</dd>
<dt>Symbol</dt><dd><p>One or two letter atomic symbol for the element.</p>
</dd>
<dt>Name</dt><dd><p>Name of the element.</p>
</dd>
<dt>Group</dt><dd><p>Identifies elements having similar chemical behaviours. 
For most elements, the column in the periodic table is identical to the group.</p>
</dd>
<dt>Period</dt><dd><p>Period identify a collection of elements of sequential mass typically from metals to nonmetals.
For most elements, period identifies its row in a periodic table.</p>
</dd>
<dt>Mass</dt><dd><p>Relative atomic mass or atomic weight. 
It is a dimensionless physical quantity defined 
as the ratio of the average mass of atoms of a chemical element
in a given sample to the atomic mass constant. 
Measurements are in unified atomic mass units or Daltons.
Expressed in these units, it is within 1 percent of the mass number.</p>
</dd>
<dt>Mass_number</dt><dd><p>Total number of protons and neutrons in the atomic nucleus.</p>
</dd>
<dt>Density</dt><dd><p>Element density in grams per cubic centimetre. </p>
</dd>
<dt>Melting_point</dt><dd><p>Temperature in degrees Kelvin at which the 
element changes state from solid to liquid.
Marked as approximate for Flerovium and Oganesson.</p>
</dd>
<dt>Boiling_point</dt><dd><p>Temperature in degrees Kelvin at which the 
element changes state from liquid to vapour.</p>
</dd>
<dt>Specific_heat_capacity</dt><dd><p>The amount of heat energy required 
to increase the temperature by one Kelvin degree measured in Joules
per gram and degree Kelvin.</p>
</dd>
<dt>Electro_negativity</dt><dd><p>The tendency of an atom to attract a shared pair of
electrons (or electron density) towards itself.  The higher the number
the greater the attraction.</p>
</dd>
<dt>Abundance</dt><dd><p>The estimated abundance of the element in the 
Earth's crust in millograns per kilogram.  
For Technetium and Francium the value is marked 
as approximate; for Neptunium and Plutonium the value 
is an upper bound.</p>
</dd>
<dt>Category</dt><dd><p>Identifies where elements lie on the
metal - metalloid - nonmetal categorization.</p>
</dd>
<dt>Subcategory</dt><dd><p>Identifies the subcategory of the element on the
metal - metalloid - nonmetal categorization.</p>
</dd>
<dt>x</dt><dd><p>The geometric horizontal position in the periodic table where this element appears.</p>
</dd>
<dt>y</dt><dd><p>The geometric vertical position in the periodic table where this element appears.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p>Data extracted from Wikipedia's &quot;List of chemical elements&quot; <a href="https://en.wikipedia.org/wiki/List_of_chemical_elements">https://en.wikipedia.org/wiki/List_of_chemical_elements</a> (April 17, 2020).
</p>

<hr>
<h2 id='faces'>Olivetti Faces</h2><span id='topic+faces'></span>

<h3>Description</h3>

<p>Grayscale faces 8 bit [0-255], a few images of several different people.
</p>


<h3>Format</h3>

<p>Data frame with 400 variables (one image per variable) and 4,096 rows
(the greyscale values of a 64x64 image).
</p>


<h3>Details</h3>

<p>400 total images, 64x64 size.
</p>
<p>From the Olivetti database at ATT.
</p>


<h3>Source</h3>

<p>Sam Roweis's data page at cs.nyu.edu.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+frey">frey</a></code>, <code><a href="#topic+ordfrey">ordfrey</a></code>
</p>

<hr>
<h2 id='frey'>Frey Faces</h2><span id='topic+frey'></span>

<h3>Description</h3>

<p>1,965 images of Brendan Frey's face, taken from
sequential frames of a small video. Image size: 20x28.
</p>


<h3>Format</h3>

<p>Data frame with 1,965 variables (one image per column) and 560 rows 
(the 560 greyscale values of a 20x28 image).
</p>


<h3>Source</h3>

<p>Sam Roweis's data page at cs.nyu.edu.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ordfrey">ordfrey</a></code>, <code><a href="#topic+faces">faces</a></code>
</p>

<hr>
<h2 id='igg1'>Human immunoglobulin G1 antibody molecule</h2><span id='topic+igg1'></span>

<h3>Description</h3>

<p>Human immunoglobulin G1 is an antibody molecule of
10,401 atoms.  The great bulk of these form residues attached
to 1,556 alpha carbons (alpha refers to the first carbon
that attaches to a functional group).
In this protein molecule, the functional groups called
residues are either amino acids or carbohydrates (sugar molecules).
</p>
<p>On the geometry of this protein, from Padlan (1994, p. 172):
&quot;An antibody molecule is composed of three major
fragments: the two Fabs, which are identical and each
of which contains the light chain and the first two
domains of the heavy chain, and the Fc, which contains
the C-terminal constant domains of the two heavy chains.
The Fabs are linked to the Fc by the hinge
region, which varies in length and flexibility in the
different antibody classes and isotypes. The antigen binding
sites (paratopes) are located at the tips of the Fabs.&quot;
</p>
<p>Full names for amino acid residues and group characteristics
were taken from the commercial website www.tocris.com.
</p>
<p>The data records all 1,556 alpha carbons, their residues,
which of five separate chains each carbon belongs to,
and the geometric location given by coordinates
x, y, and z as determined by X-ray crystallography and as available
to Padlan (1994) either from the Protein Data Bank or or
from original investigators at the time of publication.
</p>
<p>From the source website: &quot;It is a composite model built from F(ab)2
fragments from Brookhaven file 2IG2.PDB, and an Fc fragment from
Brookhaven file 1FC2.PDB. Part of the hinge region and other details
are theoretically modeled.&quot;
</p>


<h3>Format</h3>

<p>A data frame with 1556 rows and 10 variates:
</p>

<dl>
<dt>recordType</dt><dd><p>Either 'ATOM' or 'HETATM'.
Here 'ATOM' indicates an atom having a standard residue
of the protein; 'HETATM' (hetero atom) indicates one either
having a non-standard residue of protein, or one in a group of a different kind
such as carbohydrates, substrates, ligands, solvent, or metal ions. In
the igg1 molecule, these will be a part of some carbohydrate.</p>
</dd>
<dt>name</dt><dd><p>Name of the alpha carbon atom.</p>
</dd>
<dt>residue</dt><dd><p>The three letter abbreviated name of the residue.</p>
</dd>
<dt>chainID</dt><dd><p>Chains H and I (heavy), residues 1-452 each; Chains L, and M (light), residues 501-716; Chain C (carbohydrate), residues 1-9, 10-18.</p>
</dd>
<dt>residueSequenceNum</dt><dd><p>Order in which that carbon atom appears in its chain.</p>
</dd>
<dt>x, y, z</dt><dd><p>Coordinates of the carbon atom in three-dimensonal space.</p>
</dd>
<dt>residueName</dt><dd><p>Full name of the residue.</p>
</dd>
<dt>group</dt><dd><p>A group characteristic for that residue.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>Eduardo A. Padlan (1994) &quot;Anatomy of the Antibody Molecule&quot;,
Molecular Immunology, 31, Issue 3, pp. 169 - 217.
</p>
<p>Lisa J. Harris, Steven B. Larson, Karl W. Hasel,
John Day, Aaron Greenwood and Alexander McPherson (1992)
&quot;The three-dimensional structure of an intact monoclonal
antibody for canine lymphoma&quot;, Nature, 360, pp. 369-372.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+elements">elements</a></code> <code><a href="#topic+SCmolecule">SCmolecule</a></code>
</p>

<hr>
<h2 id='judgment'>Judgment samples of plastic blocks</h2><span id='topic+judgment'></span>

<h3>Description</h3>

<p>Thirty-three graduate students were shown a set of 100 &quot;physical blocks&quot; of different shapes and sizes, all cut from a single sheet of opaque plastic a few millimetres thick (see  <code><a href="#topic+blocks">blocks</a></code>).
</p>
<p>The blocks were numbered 1 to 100 and each student was asked to choose 10 blocks whose average weight would, in their judgment,  equal the average weight of all 100 blocks.
The only information available to each student to help them make their choice was the visual shape and size of all 100 blocks.
They had a few minutes each to make and record their choices.
</p>
<p>The task was presented as a competition with a prize to go to the student whose sample came closest to the population average weight.
</p>
<p>Actual block weights and other information are available in the dataset <code><a href="#topic+blocks">blocks</a></code>.
</p>


<h3>Format</h3>

<p>A data frame with 33 rows and 11 variates
</p>

<dl>
<dt>studentID</dt><dd><p>The last four digits of the student's ID number (leading zeros removed).</p>
</dd>
<dt>first</dt><dd><p>The block id number of the first block selected by the student.</p>
</dd>
<dt>second</dt><dd><p>The block id number of the second block selected by the student.</p>
</dd>
<dt>third</dt><dd><p>The block id number of the third block selected by the student.</p>
</dd>
<dt>fourth</dt><dd><p>The block id number of the fourth block selected by the student.</p>
</dd>
<dt>fifth</dt><dd><p>The block id number of the fifth block selected by the student.</p>
</dd>
<dt>sixth</dt><dd><p>The block id number of the sixth block selected by the student.</p>
</dd>
<dt>seventh</dt><dd><p>The block id number of the seventh block selected by the student.</p>
</dd>
<dt>eighth</dt><dd><p>The block id number of the eighth block selected by the student.</p>
</dd>
<dt>ninth</dt><dd><p>The block id number of the ninth block selected by the student.</p>
</dd>
<dt>tenth</dt><dd><p>The block id number of the tenth block selected by the student.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>See Also</h3>

<p><code><a href="#topic+blocks">blocks</a></code>
</p>

<hr>
<h2 id='lepto'>Rock Crabs (Leptograpsus variegatus) data</h2><span id='topic+lepto'></span>

<h3>Description</h3>

<p>Body measurements on a sample of 200 (purple) rock crabs (Leptograpsus variegatus) found in the southern subtropical Pacific Ocean.
These are small sea crabs that grow at most to about 50 millimetres of shell width.
</p>
<p>The objective is to determine whether the sample actually represents two different species of rock crab base only on their
physical measurements. See reference for details.
</p>


<h3>Format</h3>

<p>A data frame with 200 rows and 5 variables
</p>

<dl>
<dt>front</dt><dd><p>The length in millimetres of the crab's carapace frontal lobe region just before the &quot;frontal tubercles&quot;.</p>
</dd>
<dt>rear</dt><dd><p>The length in millimetres of the crab's carapace rear.</p>
</dd>
<dt>length</dt><dd><p>This is a measurement (in mm) of the length  of the crab's carapace from front to rear along its centre line.</p>
</dd>
<dt>width</dt><dd><p>This is a measurement (in mm) of the width of the carapace of the crab at its widest point.</p>
</dd>
<dt>depth</dt><dd><p>This is a measurement (in mm) of the depth of the crab body.</p>
</dd>
</dl>

<p>There are known to be four distinguishable subsets in this data set.
</p>


<h3>Details</h3>

<p>These are physical measurements on the purple rock crab (Leptograpsus variegatus), a large eyed
crab found in the southern subtropical Pacific ocean. These are small sea crabs that grow at most to
about 50 millimetres of shell width. 
</p>
<p>Data are a subset of the <code>crabs</code> data set from the <code>MASS</code> package.  
Only the five size measurements or the crabs are given here and their order has been randomized.
</p>
<p>A good data set for cluster analysis
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>N.A. Campbell and R.J. Mahon (1974). &quot;A multivariate study of variation in two species of rock crab of the genus Leptograpsus.&quot;
Australian Journal of Zoology 22, 417-425. &lt;doi:10.1071/ZO9740417&gt;
</p>
<p>Venables, W. N. and Ripley, B. D. (2002) Modern Applied Statistics with S. 
Fourth edition. Springer.  &lt;doi:10.1007/978-0-387-21706-2&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+crabSpecies">crabSpecies</a></code>
</p>

<hr>
<h2 id='lightspeeds'>Historical Determinations of the Speed of Light</h2><span id='topic+lightspeeds'></span>

<h3>Description</h3>

<p>Contains historical determinations of the speed of light from many studies from Fizeau's toothed wheel
in 1849, to determinations using stabilized lasers in 1983.  
Methods, year of study, authors, information
on mechanisms used, as well as other remarks are recorded.  The estimated speed of light is recorded
for each study as well as the authors' determination of the error of their measurement. 
</p>
<p>The data are of particular value since in 1974 the speed of light was defined to be 299,792.458 
kilometres per second (in vacuo).  The data therefore provide a rare case where the 'true value' 
is known. 
</p>
<p>Also, the values might be grouped by the different methods were used over time to 
estimate the speed of light.  In this way, the data provide a useful case study to discuss
methods of meta-analysis as well.
</p>


<h3>Format</h3>

<p>A data frame with 81 rows and 11 variables
</p>

<dl>
<dt>method</dt><dd><p>The general method used to determine the speed of light (see details for more on
this).</p>
</dd>
<dt>year</dt><dd><p>Year in which the determination was made.</p>
</dd>
<dt>researcher_1</dt><dd><p>First researcher named as conducting the study
(surname, or prefixed with initials if surname not unique).</p>
</dd>
<dt>researcher_2</dt><dd><p>Second researcher named, if any.</p>
</dd>
<dt>researcher_3</dt><dd><p>Third researcher named, if any.</p>
</dd>
<dt>researcher_4</dt><dd><p>Fourth researcher named, if any.</p>
</dd>
<dt>mechanism</dt><dd><p>Type of mechanism used (see details below).</p>
</dd>
<dt>mechanism_2</dt><dd><p>More detail on mechanism used (see details below).</p>
</dd>
<dt>remark</dt><dd><p>Remark on some detail of the study or method used.</p>
</dd>
<dt>speed</dt><dd><p>The determined speed of light in air in kilometres per second.</p>
</dd>
<dt>error</dt><dd><p>The error of the estimate (in km/second) as reported by the researchers
(see details below).</p>
</dd>
</dl>

<p>The row order of the values follow their order of appearence 
in the paper given as reference below.
</p>


<h3>Details</h3>

<p>See reference for details.
</p>
<p>On the meaning of the <code>error</code> variable, from the reference:
&quot;This error is rarely a standard deviation.  Nor is it based solely on measurements taken
in the study.  Instead, it is a number out together bt the researchers from a number
of possible sources and is very subjective.  It is not uncommon for subsequent researchers
to examine in detail the results of a given study and to arrive at a different value
of the error.   Finally, physicists are accustomed to reporting the probable error which
can be interpreted as approximately 0.6745 times the standard deviation of the estimate.&quot;
</p>
<p>On the meaning of variables related to <code>method</code>, also from the reference:
</p>
<p><code>Optical</code>: These methods are based on having a light beam leave a source, strike a <code>rotating
mirror</code> or pass through the spaces of a <code>toothed wheel</code>, travel some considerable distance to
be reflected back (again striking the rotating mirror or passing through the spaces of a
toothed wheel) to near the original soource.  The speed of rotation must be just right and
is used in the determination of the speed of light.  Precision of estimation could ve increased
by increasing the distance the light had to travel (from source to stationary mirror and back) or
by increasing the speed of rotation.
</p>
<p><code>Electrical</code>: This method is introduced after the electromagnetic theory of light was developed.
Light could now be thought of as electromagnetic radiation.  As such measures of the speed of any
<code>electromagnetic radiation in vacuo</code> would also be legitimate measures of the speed of light (in vacuo).
Moreover the <code>ratio of electrostatic to electromagnetic units</code> of measurement of electrical quantities could 
be taken to be measurements of the speed of light.
</p>
<p><code>Electro-optical</code>:  These are based largely on the same principle as the toothed wheel in optical
methods which effectively use a mechanical shutter to switch light on and off.
With the <code>electro-optical</code> methods, non-mechanical shutters are used to much more rapidly (and
in a more finely controlled way) alternate the light and so increase the precision.  
The <code>Kerr cell</code> consists of two electrodes immersed in a liquid like nitrobenzene.  When high
voltage is applied to the electrodes, the polarity of light passing through the cell changes
from planar to elliptical.  Switching between high and low voltage effects the shutter.
The <code>quartz modulator</code> passes sound waves through a crystal to change its refractive index.
An acoustic frequency can be found to produce a diffraction grating for light passing through the
crystal; double that frequency and the diffraction grating dissappears.  Switching between the two
frequencies produces the shutter effect.
</p>
<p><code>Radio wave</code>: Understanding light to be electromagetic radiation also means that radio waves can 
be used in place of visible light to make measurements of its speed.  
<code>Radar</code> sends a short pulse of high frequency radio waves from a source to a distant object and
measures the time taken to receive the reflected wave from the distant object.  Knowing the
actual distance allows a determination of the speed of the radio wave (or light).
The <code>cavity resonator</code> sends high frequency radio waves down a hollow cylinder sealed at both ends.
The cylinder resonates if its length is a whole number multiple of the half-wavelength of the
radio wave. The speed of light (in the medium within the cylinder or cavity) can be determined
from the dimensions of the cylinder (cavity).
</p>
<p><code>Geodetic</code>:  These are improvements on the <code>Kerr cell</code> technology to make it capable of measuring
geodetic distances.  The resulting (commercial) instrument was called a <code>Geodimeter</code>.  With known
distances these instruments could be turned around to be used to provide measures of the speed of
light. The <code>Tellurometer</code> was another instrument invented to determine geodetic distances.  
The principal difference between it and the Geodimeter is that it used microwave radiation to
carry the signal.
</p>
<p><code>Spectroscopy</code>: <code>Bombarding molecules with electromagnetic radiation</code> causes them to 
absorb enough energy to change various states.  
Bombarding molecules with microwave radiation changes their
rotational state,  with infra-red radiation their vibrational state.  Quantum theory
allows the measurement of these to changed states to be turned into a determination of the 
speed of light.  Different studies bombarded different molecules.
</p>
<p><code>Ultrasonic modulation</code>:  This method can be regarded as an improvement on the <code>quartz modulator</code>.
Instead of acoustic waves on a crystal, a diffraction grating is produced with ultrasonic waves
in a liquid. Turning the diffraction grating on and off produces the shutter.
</p>
<p><code>Interferometry</code>:  A single source light bean is split in two.  Each travels some distance, is 
reflected, and returns to the source.  The two are made to travel different distances and
the amount by which they are out of phase with one another upon return, together with the
difference in distances travelled, can be turned into a measure of the speed of light. 
Instead of visible light, <code>radio waves</code> or <code>micro waves</code> were used.
</p>
<p><code>Stabilized lasers</code>:  In <code>interferometry</code> there can be some uncertainty in the measure of the
wavelengths used.  With stabilized lasers using a technique called <code>sub-Doppler saturated
absorption spectroscopy</code> it became possible to fix the frequency (and hence the wavelength)
of some lasers within a very narrow range of the electromagnetic spectrum.  Such lasers are
called stabilized lasers and have nice short <code>wavelengths</code> (micrometres) that allow more precise
measurements of the speed of light.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p><a href="https://www.researchgate.net/publication/275521939_The_speed_of_light_A_case_study_in_empirical_problem_solving">https://www.researchgate.net/publication/275521939_The_speed_of_light_A_case_study_in_empirical_problem_solving</a>
</p>


<h3>References</h3>

<p>R.W. Oldford 1994, 
'The speed of light: A case study in empirical problem solving',
Unpublished manuscript. &lt;doi:10.13140&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+michelson_1879">michelson_1879</a></code>
</p>

<hr>
<h2 id='lizards'>A fictional data set on lizards perch choices</h2><span id='topic+lizards'></span>

<h3>Description</h3>

<p>An entirely artificially constructed data set and context designed for classroom discussion and
analysis. 
</p>
<p>The fictional context is that 384 different lizards have been observed in nature, from each of 4 different species (&quot;A&quot;, &quot;B&quot;, &quot;C&quot;, or &quot;D&quot;).
In addition to their species, the sex of each lizard was identified.   
At the time of observation, each lizard was perched on the branch of a bush.
The perch height from the ground in feet was measured and the diameter in inches of the branch where the lizard was perched was measured.
Lizards are collected and numbered (row.names) in the order they appear in the data set.
</p>
<p>We could imagine the study having proceeded as follows.
</p>
<p>Problem:
</p>
<p>Suppose that we are interested in the habitat of lizards on some other remote island archipelago. 
Here there are trees of all heights and many different species of lizards. 
Interest lies in understanding the perch height preference for all lizards.
</p>
<p>Plan:
</p>
<p>As with the source study, lizards will be observed during daylight hours for two separate
summer time expeditions. The height (in feet) at which the lizard is observed will be
recorded as is the diameter (in inches) of the branch where the lizard is perched.
The species and sex of each lizard will be recorded.
</p>
<p>Data:
A total of 384 lizards of several species were captured and measured together with the
height and diameter of their perch when first spotted.
</p>
<p>NOTE:  The data are inspired by the reference but are otherwise entirely fictional and constructed primarily for pedagogical purposes.
Instructors might choose to invent their own context.
</p>


<h3>Format</h3>

<p>A data frame with 384 rows and 4 variates
</p>

<dl>
<dt>perchHeight</dt><dd><p>The height in feet where the lizard was found to be perched.</p>
</dd>
<dt>perchDiameter</dt><dd><p>The diameter in inches of the branch where the lizard was perched.</p>
</dd>
<dt>sex</dt><dd><p>The sex of the lizard: a factor with levels &quot;male&quot; and &quot;female&quot;</p>
</dd>
<dt>species</dt><dd><p>The species of lizard: a factor with four levels &quot;A&quot;, &quot;B&quot;, &quot;C&quot;, and &quot;D&quot;.</p>
</dd>
</dl>

<p>'rownames(lizards)' labels the order in which the lizards were collected.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>T.W. Schoener (1968) “The Anolis lizards of Bimini: Resource partitioning in a complex fauna”, Ecology, Vol. 49, pp. 704-726.
</p>

<hr>
<h2 id='medicalRecords'>Mining medical records (fictional)</h2><span id='topic+medicalRecords'></span>

<h3>Description</h3>

<p>An entirely artificially constructed data set and context designed for classroom discussion and
analysis. 
</p>
<p>A medical data mining context is given in detail below.  
In light of the context, interesting scientific questions will arise as to the
data collection, and how the results should, or should not, be 
interpreted. 
It should also raise questions on what might be done in any follow up studies.
</p>
<p>Instructors might choose to invent their own context.
</p>


<h3>Format</h3>

<p>A data frame with 16 rows and 5 variables (providing the counts for a 2x2x2x2 contingency table).
</p>

<dl>
<dt>Age</dt><dd><p>A two level factor recording one of two age groups: &quot;20-39&quot; or &quot;40-59&quot;. </p>
</dd>
<dt>Sex</dt><dd><p>A two level factor recording sex: &quot;Male&quot; or &quot;Female&quot;. </p>
</dd>
<dt>Treatment</dt><dd><p>A two level factor recording the treatment received: &quot;A&quot; or &quot;B&quot;. </p>
</dd>
<dt>Outcome</dt><dd><p>A two level factor recording patient outcome after treatment: &quot;Recovered&quot; or &quot;Died&quot;. </p>
</dd>
<dt>Freq</dt><dd><p>The frequency count of patients having that combination of factors.</p>
</dd>
</dl>



<h3>Details</h3>

<p>One fictional context (constructed in March 2020) for this data set is given below (in the PPDAC
style of Mackay and Oldford (2000)).
</p>
<p>Problem:
</p>
<p>A disease epidemic has broken out in the population of some country.  
It is thought that adults under the age of 60 appear to be particularly vulnerable.  
Both men and women contract the disease and need to be treated.  
Those who go untreated die within 5 days of contracting the disease.
</p>
<p>The medical community has tried two quite different approaches to treat patients 
having the disease &ndash; call these 'Treatment A' and 'Treatment B'.  
For the health of the country, it is important to determine which of these two 
treatments is more effective.
</p>
<p>Plan:
</p>
<p>To investigate which is the better treatment, it was decided to mine the medical 
records from another country of those who had contracted the disease and had been 
treated with one of the two treatments.  
Patients treated with either A or B survive the disease and recover fully; some however still die.
</p>
<p>Electronic medical records available from several of the more populous districts are accessible.  
These can be searched to provide records from patients that have received treatment.  
It was decided that there should be the same number of records drawn for each treatment.
</p>
<p>Moreover, concern was raised that the investigation have gender balance 
(i.e. equal numbers of males and females).  
So, to make sure that both sexes were equally represented, it was also decided that the 
number of female patients would be the same as the number of male patients.  
</p>
<p>Finally, it was desirable to detect even small differences in success rates of the 
two treatments since small differences could mean many more lives being saved.  
A sample size of about n = 3,000 was decided on.  
</p>
<p>Records would be collected until 3,000 were found, 1500 of which were treated with 'A', 
1500 with 'B', and there were equal numbers of males and females in the study.
</p>
<p>Data:
</p>
<p>In this stage, the plan is executed.  Instead of 1500 records of treatment 'A' and 'B', 
1600 of each were found.  The number of males and females was kept equal (now 1600 of each sex).
</p>
<p>The process was to search the records in order, selecting those first encountered to 
get 1600 for each treatment and 1600 of each sex.  Many records might be discarded 
whenever one quota was met and the search continued to meet the other quotas.  
It was also noticed that the patient's age was available for each record, so that the 
effect of treatment on younger and older adults might also be considered.
</p>
<p>The counts which fell into the various categories were assembled into the data presented here.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>R.J. MacKay and R.W. Oldford 2000, 
'Scientific Method, Statistical Method, and the Speed of Light',
Statistical Science, Volume 15, No. 3, pp. 254-278.  &lt;doi:10.1214/ss/1009212817&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pandemic">pandemic</a></code>  <code><a href="#topic+covidNZ">covidNZ</a></code>
</p>

<hr>
<h2 id='michelson_1879'>A.A. Michelson's 1879 Determinations of the Speed of Light</h2><span id='topic+michelson_1879'></span>

<h3>Description</h3>

<p>In 1879, Albert Abraham Michelson conducted an experimental study to
determine the speed of light using a rotating mirror apparatus at the
U.S. Naval Academy in Annapolis, Maryland in 1879.  
</p>
<p>Details on the apparatus, the optical theory, and the conduct of the experiment are
given in the reference.  An abbreviated summary of these follows the variable descriptions.
</p>


<h3>Format</h3>

<p>A data frame with 100 rows and 15 variables
</p>

<dl>
<dt>Speed</dt><dd><p>The determined speed of light in air in kilometres per second.</p>
</dd>
<dt>Beat</dt><dd><p>Number of beats per second between tuning forks.</p>
</dd>
<dt>Correction</dt><dd><p>Correction for temperature to a standard fork in beats per second.</p>
</dd>
<dt>Day</dt><dd><p>Day of experiment in progress (June 5 is day 1) on which these measurements were taken.</p>
</dd>
<dt>Difference</dt><dd><p>Difference betweeen the greatest and least values of revolutions.</p>
</dd>
<dt>Quality</dt><dd><p>Subjective measure of the quality of the image 'I'; the more distinct
was the image the higher the quality (1 = poor, 3 = good). </p>
</dd>
<dt>Displacement</dt><dd><p>Displacement of image 'I' from slit 'S' in divisions of the micrometer.</p>
</dd>
<dt>Image</dt><dd><p>Micrometer position of the deflected image.</p>
</dd>
<dt>Radius</dt><dd><p>Radius of measurement in feet.</p>
</dd>
<dt>Revolutions</dt><dd><p>Number of times the mirror revolved per second.</p>
</dd>
<dt>Screw</dt><dd><p>Measure of one screw turn in millimetres.</p>
</dd>
<dt>Slit</dt><dd><p>Micrometer position of the slit providing the light source 'S'.</p>
</dd>
<dt>Temperature</dt><dd><p>Air temperature measured in degrees Fahrenheit.</p>
</dd>
<dt>Time.of.day</dt><dd><p>Time of day at which the observation was recorded.
'AM' means one hour after sunrise and 'PM' one hour before sunset.</p>
</dd>
<dt>Remarks</dt><dd><p>Unusual remarks recorded for that observation.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The experiment is conducted within a closed and darkened small building at the U.S. Naval Academy.
Light enters the building from one corner passing through a slit 'S' whose location is
precisely determined using a micrometer.  
</p>
<p>The light then proceeds to hit a rotating mirror
at the other end of the building's interior from whence it is reflected out of the
building through an opening in a corner different from that of the source.  
</p>
<p>The light 
beam travels outside to strike another (stationary) mirror which reflects it back into the
building through the same corner it exited whereupon it then strikes the rotating mirror.
</p>
<p>Depending on the position of the rotating mirror, the returning light will be reflected off
it to land at some position 'I' near the original source given by the slit 'S'.
</p>
<p>The speed of the rotating mirror is controlled using an adjustable pump to blow air across a surface to
rotate it.  
If the speed of rotation is just right, a crisp image 'I' of the reflected slit will appear near
the original source 'S'.  The speed is adjusted until this is the case.
</p>
<p>The speed of rotation is determined using an electric tuning fork connected to the
rotating mirror and whose frequency was measured by comparing it to a second standard tunig
fork of known frequency.  The electronic fork frequency was compared to the standard fork
by determining the number of beats per second difference the two (by counting over 60 seconds).
</p>
<p>With a speed of revolution and the displacement measured between 'S' and its returned image 'I', 
a measurement of the speed of light could be had.
</p>
<p>See reference for more details.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>References</h3>

<p>R.J. MacKay and R.W. Oldford 2000, 
'Scientific Method, Statistical Method, and the Speed of Light',
Statistical Science, Volume 15, No. 3, pp. 254-278. &lt;doi:10.1214/ss/1009212817&gt;
</p>


<h3>See Also</h3>

<p><code><a href="#topic+lightspeeds">lightspeeds</a></code>
</p>

<hr>
<h2 id='minority'>Canadian Visible Minority Data 2006</h2><span id='topic+minority'></span>

<h3>Description</h3>

<p>Population census count of various named
visible minority groups in each of 33 major census metropolitan
areas of Canada in 2006.
</p>
<p>These data are from the 2006 Canadian census, publicly available from
Statistics Canada.
</p>


<h3>Format</h3>

<p>A data frame with 33 rows and 18 variates
</p>

<dl>
<dt>Arab</dt><dd><p>Number identifying as 'Arab'.</p>
</dd>
<dt>Black</dt><dd><p>Number identifying as 'Black'.</p>
</dd>
<dt>Chinese</dt><dd><p>Number identifying as 'Chinese'.</p>
</dd>
<dt>Filipino</dt><dd><p>Number identifying as 'Filipino'.</p>
</dd>
<dt>Japanese</dt><dd><p>Number identifying as 'Japanese'.</p>
</dd>
<dt>Korean</dt><dd><p>Number identifying as 'Korean'.</p>
</dd>
<dt>Latin.American</dt><dd><p>Number identifying as 'Latin American'.</p>
</dd>
<dt>Multiple.visible.minority</dt><dd><p>Number identifying as being a member
of more than one visible minority.</p>
</dd>
<dt>South.Asian</dt><dd><p>Number identifying as 'South Asian'.</p>
</dd>
<dt>Southeast.Asian</dt><dd><p>Number identifying as 'Southeast Asian'.</p>
</dd>
<dt>Total.population</dt><dd><p>Total population of the metropolitan census area.</p>
</dd>
<dt>Visible.minority.not.included.elsewhere</dt><dd><p>Number identifying as a
member of a visible minority that was not included elsewhere.</p>
</dd>
<dt>Visible.minority.population</dt><dd><p>Total number identifying as a
member of some visible minority.</p>
</dd>
<dt>West.Asian</dt><dd><p>Number identifying as 'West Asian'.</p>
</dd>
<dt>lat, long</dt><dd><p>Latitude and longitude in degrees of the metropolitan census area.</p>
</dd>
<dt>googleLat, googleLong</dt><dd><p>Latitude and longitude in degrees determined
using the Google Maps Geocoding API.</p>
</dd>
</dl>

<p>'rownames(minority)' are the names of the metropolitan areas or cities.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>

<hr>
<h2 id='ordalphadigits'>Binary Alphadigits Isomap</h2><span id='topic+ordalphadigits'></span>

<h3>Description</h3>

<p>Binary Alphadigits Isomap
</p>


<h3>Format</h3>

<p>Object of class 'isomap'.
</p>


<h3>Details</h3>

<p>Dissimilarity object of class 'isomap'. Returned from:
</p>
<p><code>isomap(vegdist(binaryalphadigits), k=6)</code>.
</p>
<p>Introduced simply to cache the results of this step so as to speed up demos.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+binaryalphadigits">binaryalphadigits</a></code>
</p>

<hr>
<h2 id='ordfrey'>Frey Faces Isomap</h2><span id='topic+ordfrey'></span>

<h3>Description</h3>

<p>Frey Faces Isomap
</p>


<h3>Format</h3>

<p>Object of class 'isomap'.
</p>


<h3>Details</h3>

<p>Dissimilarity object of class 'isomap' for Frey Faces data, created with:
</p>
<p><code>isomap(vegdist(t(frey), method="euclidean"),k = 12, ndim=6, fragmentedOK = TRUE)</code>
</p>
<p>Introduced simply to cache the results of this step so as to speed up demos.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+frey">frey</a></code>
</p>

<hr>
<h2 id='pandemic'>Fictional pandemic data</h2><span id='topic+pandemic'></span>

<h3>Description</h3>

<p>An entirely artificially constructed data set and context designed for classroom discussion and
analysis. 
</p>
<p>The data (and the fictional narrative below) are identical to those of  <code><a href="#topic+trtPan">trtPan</a></code>;
the only difference is the organization of the data. 
Which organization the instructor might choose depends upon the modelling 
and/or data manipulation is intended for analysis.
</p>
<p>Should raise interesting scientific questions on how the results should, or should not, be 
interpreted.   It should also raise questions on what might be done next.
</p>
<p>A &quot;pandemic&quot; context is given in the details since the data were created during the first week of
March, 2020.
</p>
<p>Instructors might choose to invent their own context.
</p>


<h3>Format</h3>

<p>A data frame with 100 rows and 4 variables
</p>

<dl>
<dt>City</dt><dd><p>City for which the outcome data were recorded. </p>
</dd>
<dt>A</dt><dd><p>The percent survival rate for infected persons given medical treatment &quot;A&quot;.</p>
</dd>
<dt>B</dt><dd><p>The percent survival rate for infected persons given medical treatment &quot;B&quot;.</p>
</dd>
<dt>C</dt><dd><p>The percent survival rate for infected persons given medical treatment &quot;C&quot;.</p>
</dd>
</dl>



<h3>Details</h3>

<p>One fictional narrative for this data set is as follows.
</p>
<p>A virulent virus has led to a world wide pandemic and that the case fatality rate 
(proportion of those infected who die) is huge (say 6
</p>
<p>Suppose that through a concerted and collaborative effort of health scientists worldwide, 
three different treatments have been developed for this group.  
All three treatments have been used at one time or another on numerous patients in 
this group from 100 different cities worldwide.  
The data are observational, in that they were simply collected and the treatment given noted.
No information is available on why one treatment or another was given in any particular instance.
</p>
<p>The recovery rates (as a percent) for the patients treated by each of the three treatments
were simply recorded for each of the hundred cities and are available for analysis as the 
data frame <code>pandemic</code>.
</p>
<p>Some obvious questions of interest are the comparisons of treatments. For example, is treatment
A better than B? Than C?  Is B better than C?
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>See Also</h3>

<p><code><a href="#topic+trtPan">trtPan</a></code>
<code><a href="#topic+medicalRecords">medicalRecords</a></code>  <code><a href="#topic+covidNZ">covidNZ</a></code>
</p>

<hr>
<h2 id='pkg_data'>Simple summary of data available in named packages</h2><span id='topic+pkg_data'></span>

<h3>Description</h3>

<p>Simple summary of data available in named packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pkg_data(package = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pkg_data_+3A_package">package</code></td>
<td>
<p>A character vector giving the package(s) to look in for data sets, 
or NULL. By default, all packages in the search path are used, then the ‘data’ 
subdirectory (if present) of the current working directory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Syntactic sugar wrapping call to <code>utils::data(package = package)</code> 
to return basic information on datasets in package.
No data are loaded by the call.
</p>


<h3>See Also</h3>

<p>data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(pkg_data("loon.data"))
</code></pre>

<hr>
<h2 id='placenamesCanada'>Canadian place names and their geo-locations.</h2><span id='topic+placenamesCanada'></span>

<h3>Description</h3>

<p>The names (in English and in French) of 10,776 places in Canada together with their 
geographic locations as compiled by the Government of Canada.
</p>
<p>The data are part of the Canadian government's open data project.
</p>


<h3>Format</h3>

<p>A data frame with 10,676 rows and 6 variables
</p>

<dl>
<dt>ID</dt><dd><p>The identification number of the place name (called &quot;PNuid_NLidu&quot; in the source).</p>
</dd>
<dt>Name</dt><dd><p>A character vector containing the place name in English.</p>
</dd>
<dt>Nom</dt><dd><p>A character vector containing the place name in French.</p>
</dd>
<dt>Province</dt><dd><p>A factor with 13 levels giving the two-letter code for the Canadian province or territory.</p>
</dd>
<dt>Latitude</dt><dd><p>A numeric vector giving the latitude of the place.</p>
</dd>
<dt>Longitude</dt><dd><p>A numeric vector giving the longitude of the place.</p>
</dd>
</dl>



<h3>Details</h3>

<p>Note that English and French names rarely differ in this data set.
</p>
<p>Details from the source:  
</p>
<p>&quot;The collection of geolocated placenames in Canada represents a consistent and comprehensive 
distribution of named places across Canada. Named places include large and small cities, villages, 
First Nations Communities, Small Hamlets etc.
</p>
<p>&quot;This data draws from public information maintained by Natural Resources Canada as part of 
the Canadian Geographical Names Database and public information maintained by Crown-Indigenous 
Relations and Northern Affairs Canada.
</p>
<p>&quot;The set of geolocated placenames is currently used for the administration of rural broadband 
Internet contribution programs, but is equally applicable for other mapping or modelling purposes 
where a comprehensive set of geolocated placenames across Canada is required.&quot;
</p>
<p>(downloaded May 29, 2020 from open.canada.ca at the slow loading
open.canada.ca/data/en/dataset/fe945388-1dd9-4a4a-9a1e-5c552579a28c)
</p>

<hr>
<h2 id='SAheart'>South African Heart Disease Data</h2><span id='topic+SAheart'></span>

<h3>Description</h3>

<p>From the web source: 
&quot;A retrospective sample of males in a heart-disease high-risk region 
of the Western Cape, South Africa. There are roughly two controls per case of CHD. 
Many of the CHD positive men have undergone blood pressure reduction treatment 
and other programs to reduce their risk factors after their CHD event. 
In some cases the measurements were made after these treatments.&quot;
</p>
<p>The data are packaged here from the source (below).  With one significant change 
(making chd a factor) they are also a repackaging of the data of the same name
from the now archived (in 2020) of the 2015 'ElemStatLearn' package of Kjetil B. Halvorsen.
</p>


<h3>Format</h3>

<p>A data frame with 462 rows and 10 variables
</p>

<dl>
<dt>sbp</dt><dd><p>Systolic blood pressure in millimetres of mercury (mm Hg).</p>
</dd>
<dt>tobacco</dt><dd><p>Cumulative tobacco use in kilograms. 
Appears to be lifetime cumulative; not annual.</p>
</dd>
<dt>ldl</dt><dd><p>Low density lipoprotein cholesterol.</p>
</dd>
<dt>adiposity</dt><dd><p>Not recorded in source; 
presumably another measurement of obesity similar to BMI. 
Possibly a &quot;corrected&quot; version of obesity measure.</p>
</dd>
<dt>famhist</dt><dd><p>Factor indicating presence or absence of a family history 
of ischaemic heart disease.</p>
</dd>
<dt>typea</dt><dd><p>Type-A coronary prone personality behaviour as measured by
a self-administered Bortner Short Rating Scale.  Possible total
scores can range from 12 to 84.  Rossouw et al. (1983) &quot;arbitrarily&quot; 
classify those with scores of 55 or more &quot;as exhibiting type A behaviour.&quot;</p>
</dd>
<dt>obesity</dt><dd><p>A measure of obesity; body mass index (or BMI) is consistent with Rossouw et al. (1983).
Having BMI &gt;= 30 scored as &quot;obese&quot; by Rossouw et al. (1983).</p>
</dd>
<dt>alcohol</dt><dd><p>Current alcohol consumption.  Units of measurement (quantity/time) 
are unclear (e.g litres per annum, ounces per month?);
alcohol not mentioned in Rossouw et al. (1983).</p>
</dd>
<dt>age</dt><dd><p>Age in years at time of study (Source web page: &lsquo;Age at onset&rsquo;.)</p>
</dd>
<dt>chd</dt><dd><p>The response, a factor identifying whether the subject 
had been diagnosed as having coronary heart disease or not.</p>
</dd>
</dl>

<p>The row order of the values follow their order of appearence in the source webpage.
</p>


<h3>Details</h3>

<p>In the late 1970s, an unusually high incidence of ischaemic heart disease had been 
observed to exist amongst white Afrikaans-speaking segments of 
South African society (Wyndham, 1982). 
Using an intensive postal campaign in 1979, Rossouw et al. (1983) recruited about 82
known target population of inhabitants of three Afrikaner communities in the 
southwestern Cape Province (3,357 white males and 3,831 white females).
</p>
<p>For each subject, the binary response &quot;chd&quot; (originally appearing in the original file
as 1 if they had coronary heart disease and 0 otherwise; but now as &quot;Yes&quot; or &quot;No&quot;) 
was determined in the survey together with a variety of known risk factors for heart disease. 
</p>
<p>The goal was to explore the prevalence and intensity of chd risk factors 
in these high incidence communities with particular attention to those major risk factors 
(e.g. hypercholestrolaemia, hypertension, and smoking) which might be considered 
reversible (Rossouw et al., 1983).
</p>
<p>Hastie and Tibshirani (1987) selected a subset of 465 subjects from the 3,357 white males
(in these communities, male mortality rates were about two and a half times 
that of the females; see Rossouw et al., 1983).
The 465 subjects consisted of all 162 cases having had coronary heart disease as well as 
303 controls sampled from the remaining set of survey subjects. 
</p>
<p>The same (or similar) data seems to be used again for illustration in Hastie, Tibshirani, and 
Friedman (2009) and it is that which is now ported here from the book's accompanying website 
(see source).
Curiously, this data set (viz. that recorded here) contains values on only 462 subjects, 
of which now only 160 are cases and 302 are controls. 
</p>
<p>In the current data set, rows 1-261 have
row numbers matching the source &quot;row.name&quot;, thereafter the row number is one less than
the source &quot;row.name&quot;. It would appear that subject with &quot;row.name&quot; 262 is absent from
the source (below) and, speculatively, perhaps also those whose &quot;row.name&quot; could have 
been 464 and 465.
</p>
<p>See references, particularly Rossouw et al (1983), for more details.
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p>Trevor Hastie's &quot;Elements of Statistical Learning&quot; page at Stanford.
</p>


<h3>References</h3>

<p>Trevor Hastie and Robert Tibshirani (1987) 
&quot;Non-parametric logistic and proportional odds regression&quot;, 
JRSS-C (Applied Statistics), 36(3), 260–276.
</p>
<p>Trevor Hastie, Robert Tibshirani, and Jerome Friedman (2009) 
&quot;The Elements of Statistical Learning&quot;, 2nd Edition, Springer New York
&lt;doi:10.1007/978-0-387-84858-7&gt;
</p>
<p>J.E. Rossouw, J.P.D. Plessis, A.J.S. Benad\'e, P.C.J. Jordaan, 
J.P. Kotz\'e, P.L. Jooste, and J.J. Ferreira (1983)
&quot;Coronary risk factor screening in three rural communities: The CORIS baseline study&quot;. 
South African Medical Journal, 64, 430-436.
</p>
<p>C. Wyndham (1982) &quot;Trends with time of cardiovascular mortality rates in the 
populations of the RSA for the period 1968-1977&quot;, South African Medical Journal, 61, 987-993.
</p>

<hr>
<h2 id='SCmolecule'>A protein/DNA complex molecule from Saccharomyces Cerevisiae</h2><span id='topic+SCmolecule'></span>

<h3>Description</h3>

<p>Contains three dimensional structure of the 
GAL4 protein of Saccharomyces Cerevisiae (or Baker's Yeast) 
recognizing and binding to a Deoxyribonucleic acid (DNA) sequence.
</p>
<p>There are effectively two molecules here (a protein and DNA) binding together.
It is a transcription/DNA complex within Saccharomyces Cerevisiae.
</p>
<p>From Marmorstein et al (1992): 
</p>
<p>&quot;The yeast protein GAL4 activates transcription of genes
required for catabolism of galactose and melibiose.
The DNA sequences recognized by GAL4 are 17 base pairs (bp) 
in length and each site binds a dimer of the protein.&quot;
</p>
<p>and
</p>
<p>&quot;The protein fragment binds to its DNA site as a symmetrical dimer. 
Each subunit folds into three distinct modules: 
a compact, metal-binding domain (residues 8-40), 
an extended linker (41-49), 
and an a-helical dimerization element ( 50-64). 
Residues 1-7 and 65-66 are disordered. An overall view of the complex 
shows that a large part of the DNA major groove is not contacted by the protein. 
The DNA is relatively straight. A metal domain lies in the major
groove near each end of the DNA fragment. The paired parallel
helices of the dimerization element project away from the DNA
along the 2-fold axis of the complex. The metal-binding domain
contacts three DNA base pairs in the major groove, and we
therefore refer to it as a &lsquo;recognition module.&rsquo; ...
</p>
<p>The recognition module is held together by two metal ions, 
tetrahedrally coordinated by the six cysteines. 
Two of the cysteines (11 and 28) ligate both metals, 
creating a &lsquo;binuclear cluster&rsquo; ... &quot;
</p>
<p>See Marmorstein et al (1992) for more on the geometry.
</p>
<p>The source of most data here is Protein Data Bank (PDB) entry 1d66.
</p>
<p>All coordinates, chains, residues, backbone atom identities, and displacements are
taken from the PDB entry. Not included here are those entries from the PDB record
which simply identify the terminus of each of the chains D, E, A, and B.
Each of these 'TER' entries contain no coordinates since it simply marks the 
end of its chain.
</p>
<p>Values were determined by X-ray crystallography at 2.7 Angstrom resolution.
</p>
<p>These values have been supplemented with variable values from a variety of sources
so as to help in the identification of components of the molecular structure.
</p>


<h3>Format</h3>

<p>A data frame with 1762 rows and 14 variates:
</p>

<dl>
<dt>group</dt><dd><p>One of 'ATOM' or  'HETATM'.
Here 'ATOM' indicates an atom having a standard residue
of the protein; 'HETATM' (hetero atom) indicates one either
having a non-standard residue of protein, or one in a group of a different kind
such as carbohydrates, substrates, ligands, solvent, or metal ions. In
the 'SCmolecule', these will be either a water molecule 'HOH' or a Cadmium ion 'CD'. 
</p>
</dd>
<dt>id</dt><dd><p>Identification number of the backbone atom as given in the protein data bank (PDB).</p>
</dd>
<dt>label</dt><dd><p>Atom identifier.  These follow a standard used by the PDB.
The first character is the element abbreviation of the backbone atom.  
The remaining characters of the nomenclature identify which of the atoms of that type are being
referred to in the structure.</p>
</dd>
<dt>residue</dt><dd><p>A two or three letter abbreviation naming the residue attached to that atom.</p>
</dd>
<dt>chain</dt><dd><p>Identifies a chain of atoms. These are polypeptide or DNA chains.</p>
</dd>
<dt>sequence</dt><dd><p>Order in which that bakbone atom appears in its chain.</p>
</dd>
<dt>x, y, z</dt><dd><p>Coordinates of the bakbone atom in three-dimensonal space.</p>
</dd>
<dt>displacement</dt><dd><p>Equivalent isotropic displacement factor; also sometimes earlier called a temperature
factor.  It is a measure of the possible coordinate location displacement of an atom from any source.
Displacements could arise, for example, from atomic vibrations, 
such as (large) molecular motion or (smaller) internal vibrations, 
or any of a variety of sources of disorder.
This is recorded as a spherical Gaussian (isotropic) measure of the variability 
of the location by the average eigen-value of a variance-covariance matrix.</p>
</dd>
<dt>type</dt><dd><p>The element symbol of the backbone atom.</p>
</dd>
<dt>mass</dt><dd><p>Atomic mass of the backbone atom.</p>
</dd>
<dt>residueType</dt><dd><p>Type of the residue.</p>
</dd>
<dt>residueName</dt><dd><p>Full name of the residue.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>Source</h3>

<p><a href="https://www.rcsb.org/3d-view/1D66/">https://www.rcsb.org/3d-view/1D66/</a>. <a href="https://bioinformatics.org/firstglance/fgij/fg.htm?mol=1d66">https://bioinformatics.org/firstglance/fgij/fg.htm?mol=1d66</a>
</p>


<h3>References</h3>

<p>Ronen Marmorstein, Michael Carey, Mark Ptashne, and Stephen C. Harrison (1992)
&quot;DNA recognition by GAL4: structure of a protein-DNA complex&quot;, Nature, 356, pp. 408-414.
</p>
<p>John L. Markley, Ad Bax, Yoji Arata, C. W. Hilbers, Robert Kaptein, 
Brian D. Syke, Peter E. Wright, and Kurt Wuthrich (1998)
&quot;Recommendations for the presentation of NMR structures of proteins and nucleic acids&quot;, 
European Journal of Biochemistry, 256, pp. 1-15.
</p>
<p>Reinhard X. Fischer and Ekkehart Tilmanns (1988) 
&quot;The equivalent isotropic displacement factor&quot;,
Acta Crystallographica C44, pp. 775-776.
</p>
<p>K.N. Truebloof, H.-B. Burgi, H. Burzlaff, J.D. Dunitz, 
C.M. Gramaccioli, H.H. Schulz, U. Shmueli, and S.C. Abrahams (1996)
&quot;Atomic displacement parameter nomenclature&quot;
Acta Crystallographica A52, pp. 770-781.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+elements">elements</a></code> <code><a href="#topic+igg1">igg1</a></code>
</p>

<hr>
<h2 id='trtPan'>Fictional pandemic data (with treatment variable)</h2><span id='topic+trtPan'></span>

<h3>Description</h3>

<p>An entirely artificially constructed data set and context designed for classroom discussion and
analysis. 
</p>
<p>The data (and the fictional narrative below) are identical to those of  <code><a href="#topic+pandemic">pandemic</a></code>;
the only difference is the organization of the data. 
Which organization the instructor might choose depends upon the modelling 
and/or data manipulation is intended for analysis.
</p>
<p>Should raise interesting scientific questions on how the results should, or should not, be 
interpreted.   It should also raise questions on what might be done next.
</p>
<p>A &quot;pandemic&quot; context is given in the details since the data were created during the first week of
March, 2020.
</p>
<p>Instructors might choose to invent their own context.
</p>


<h3>Format</h3>

<p>A data frame with 300 rows and 3 variables
</p>

<dl>
<dt>City</dt><dd><p>City for which the outcome data were recorded. </p>
</dd>
<dt>Treatment</dt><dd><p>The treatment used (one of &quot;A&quot;, &quot;B&quot;, or &quot;C&quot;).</p>
</dd>
<dt>Recovery</dt><dd><p>The percent survival rate for infected persons in that city when given that medical treatment.</p>
</dd>
</dl>



<h3>Details</h3>

<p>One fictional narrative for this data set is as follows.
</p>
<p>A virulent virus has led to a world wide pandemic and that the case fatality rate 
(proportion of those infected who die) is huge (say 6
</p>
<p>Suppose that through a concerted and collaborative effort of health scientists worldwide, 
three different treatments have been developed for this group.  
All three treatments have been used at one time or another on numerous patients in 
this group from 100 different cities worldwide.  
The data are observational, in that they were simply collected and the treatment given noted.
No information is available on why one treatment or another was given in any particular instance.
</p>
<p>The recovery rates (as a percent) for the patients treated by each of the three treatments
were simply recorded for each of the hundred cities and are available for analysis as the 
data frame <code>pandemic</code>.
</p>
<p>Some obvious questions of interest are the comparisons of treatments. For example, is treatment
A better than B? Than C?  Is B better than C?
</p>


<h3>Author(s)</h3>

<p>R.W. Oldford
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pandemic">pandemic</a></code>
<code><a href="#topic+medicalRecords">medicalRecords</a></code>  <code><a href="#topic+covidNZ">covidNZ</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
