<!DOCTYPE html><html lang="en"><head><title>Help for package LorMe</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LorMe}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Alpha_diversity_calculator'><p>Calculate alpha diversity based on tax summary object</p></a></li>
<li><a href='#Alpha_diversity_calculator2'><p>Calculate alpha diversity based on tax summary object or dataframe table</p></a></li>
<li><a href='#anova_report'><p>Print Analysis of Variance report</p></a></li>
<li><a href='#auto_signif_test'><p>Automatic significance test</p></a></li>
<li><a href='#circulation_lm'><p>Circulation of fitting Linear Models</p></a></li>
<li><a href='#color_scheme'><p>Get color scheme</p></a></li>
<li><a href='#combine_and_translate'><p>Combine data for visualization</p></a></li>
<li><a href='#community_plot'><p>Generate Community Composition Plot Based on Tax_summary Object</p></a></li>
<li><a href='#compare_plot'><p>Comparison plot generator</p>
This function help generate comparsion plot including bar plot, box plot, and violin plot</a></li>
<li><a href='#Deseq_analysis'><p>Deseq Analysis Function</p></a></li>
<li><a href='#Deseq_analysis2'><p>Deseq analysis</p></a></li>
<li><a href='#differential_bar'><p>Generate Differential Bar Plot and Error bar Plot</p></a></li>
<li><a href='#Dimension_reduction'><p>Dimension_reduction: PCA, PCOA, and NMDS Analysis</p></a></li>
<li><a href='#Facet_group'><p>Tax summary object with Facet 2x2 Groups</p></a></li>
<li><a href='#Filter_function'><p>Filter OTU/ASV/metagenomic profile/gene profile by threshold</p></a></li>
<li><a href='#indicator_analysis'><p>Indicator Analysis</p></a></li>
<li><a href='#kruskal_report'><p>Print Kruskal-Wallis Rank Sum Test report</p></a></li>
<li><a href='#LorMe'><p>LorMe package: Lightening One-Code Resolving Microbial Ecology Program</p></a></li>
<li><a href='#manhattan'><p>Manhattan Plot Generator</p></a></li>
<li><a href='#Module_abundance'><p>Calculate network module abundance for each sample</p></a></li>
<li><a href='#Module_composition'><p>Pie chart for network module composition</p></a></li>
<li><a href='#nc'><p>Calculate Network Natural Connectivity</p></a></li>
<li><a href='#NC_remove'><p>Natural connectivity analysis</p></a></li>
<li><a href='#network_analysis'><p>Conduct Network analysis based on tax summary object</p></a></li>
<li><a href='#network_analysis2'><p>Conduct Network analysis</p></a></li>
<li><a href='#network_stat'><p>Igraph network statistics</p></a></li>
<li><a href='#network_visual'><p>Network Visualization</p></a></li>
<li><a href='#network_visual_re'><p>Re-visualize network plot from <code>network_visual</code> or <code>network_withdiff</code></p></a></li>
<li><a href='#network_withdiff'><p>Network Analysis with Differential Species</p></a></li>
<li><a href='#object_config'><p>Set taxonomy summary configuration</p></a></li>
<li><a href='#structure_plot'><p>Visualize microbial community composition structure based on tax summary object</p></a></li>
<li><a href='#sub_tax_summary'><p>Subsetting tax summary objects</p></a></li>
<li><a href='#t_test_report'><p>Print Student's t-Test report</p></a></li>
<li><a href='#tax_summary'><p>Encapsulate meta file, feature tables and taxonomy annotation into tax summary object</p></a></li>
<li><a href='#tbRDA_analysis'><p>tbRDA analysis</p></a></li>
<li><a href='#testotu'><p>test otudata</p></a></li>
<li><a href='#theme_zg'><p>A classic theme for ggplot</p></a></li>
<li><a href='#Three_group'><p>Tax summary object with three groups</p></a></li>
<li><a href='#Top_taxa'><p>Calculate top taxa and others</p></a></li>
<li><a href='#Two_group'><p>Tax summary object with two groups</p></a></li>
<li><a href='#volcano_plot'><p>Generate Volcano plot base on Deseq_analysis or indicator_analysis results</p></a></li>
<li><a href='#wilcox_test_report'><p>Print Wilcoxon Rank Sum and Signed Rank Tests reprot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Lightening One-Code Resolving Microbial Ecology Solution</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ningqi Wang &lt;2434066068@qq.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides a robust collection of functions tailored for microbial ecology analysis, encompassing both data analysis and visualization. It introduces an encapsulation feature that streamlines the process into a summary object. With the initial configuration of this summary object, users can execute a wide range of analyses with a single line of code, requiring only two essential parameters for setup. The package delivers comprehensive outputs including analysis objects, statistical outcomes, and visualization-ready data, enhancing the efficiency of research workflows. Designed with user-friendliness in mind, it caters to both novices and seasoned researchers, offering an intuitive interface coupled with adaptable customization options to meet diverse analytical needs.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, ggalluvial, ggplot2, ggpubr, grDevices, magrittr,
RColorBrewer, scales, stringr, tidyr, vegan, dplyr, igraph,
Hmisc, fdrtool, agricolae, multcompView, DescTools, HH, coin</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), ggrepel, readr, patchwork, grid,
permute, indicspecies, S4Vectors, tibble, DESeq2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-14 02:32:49 UTC; 24340</td>
</tr>
<tr>
<td>Author:</td>
<td>Ningqi Wang [aut, cre, cph],
  Yaozhong Zhang [aut],
  Gaofei Jiang [aut]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-14 09:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Alpha_diversity_calculator'>Calculate alpha diversity based on tax summary object</h2><span id='topic+Alpha_diversity_calculator'></span>

<h3>Description</h3>

<p>Calculate alpha diversity for each sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Alpha_diversity_calculator(taxobj, taxlevel, prefix = "")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Alpha_diversity_calculator_+3A_taxobj">taxobj</code></td>
<td>
<p>Configured tax summary objects.See in <code><a href="#topic+object_config">object_config</a></code>.</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator_+3A_taxlevel">taxlevel</code></td>
<td>
<p>taxonomy levels used for visualization.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator_+3A_prefix">prefix</code></td>
<td>
<p>A character string as prefix of diversity index. Default:&quot;&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>'Alpha_diversity_calculator' returns alpha-diversity of each sample in format of column table (dataframe) combined with group information in meta file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###data preparation####
data("Two_group")
require(ggplot2)

###analysis####
Alpha_results&lt;- Alpha_diversity_calculator(taxobj = Two_group,taxlevel = "Base")

#Check data frame contained alpha diversity
head(Alpha_results$alphaframe,5)

#Check contained statistics and plot list
names(Alpha_results$plotlist)

#Check statistics for Shannon
Alpha_results$plotlist$Plotobj_Shannon$Statistics

#Extract plot for Shannon
Alpha_results$plotlist$Plotobj_Shannon$Barplot
Alpha_results$plotlist$Plotobj_Shannon$Boxplot
Alpha_results$plotlist$Plotobj_Shannon$Violinplot


</code></pre>

<hr>
<h2 id='Alpha_diversity_calculator2'>Calculate alpha diversity based on tax summary object or dataframe table</h2><span id='topic+Alpha_diversity_calculator2'></span>

<h3>Description</h3>

<p>Calculate alpha diversity of each sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Alpha_diversity_calculator2(
  taxobj = NULL,
  taxlevel = NULL,
  prefix = "",
  input,
  inputformat,
  reads
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Alpha_diversity_calculator2_+3A_taxobj">taxobj</code></td>
<td>
<p>tax summary objects computed by <code><a href="#topic+tax_summary">tax_summary</a></code>. Default:NULL.</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator2_+3A_taxlevel">taxlevel</code></td>
<td>
<p>taxonomy levels used for visualization.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).Default:NULL.</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator2_+3A_prefix">prefix</code></td>
<td>
<p>A character string as prefix of diversity index. Default:&quot;&quot;</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator2_+3A_input">input</code></td>
<td>
<p>Reads or relative abundance of OTU/Taxa/gene data frame,see details in inputformat. (Useless when taxobj is set).</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator2_+3A_inputformat">inputformat</code></td>
<td>
<p>(Useless when taxobj is set)
1:data frame with first column of OTUID and last column of taxonomy
</p>
<p>2:data frame with first column of OTUID/taxonomy
</p>
<p>3:data frame of all numeric</p>
</td></tr>
<tr><td><code id="Alpha_diversity_calculator2_+3A_reads">reads</code></td>
<td>
<p>If the input data frame were from reads table or not(relative abundance table).(Useless when taxobj is set).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>when tax taxobj is set, returns column table with group information combined with for alpha-diversity of each sample,else returns data frame for alpha-diversity of each sample
</p>


<h3>Note</h3>

<p>1.When input data frame is in relative abundance table,Chao and ACE are not available
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Data preparation ####
data(testotu)
groupinformation &lt;- data.frame(
  group = c(rep("a", 10), rep("b", 10)),
  factor1 = rnorm(10),
  factor2 = rnorm(mean = 100, 10),
  subject = factor(c(1:10, 1:10))
)

# Summary OTU table into genus table and phylum table
testtax_summary &lt;- tax_summary(
  groupfile = groupinformation,
  inputtable = testotu[, 2:21],
  reads = TRUE,
  taxonomytable = testotu[, c(1, 22)]
)

### Use taxsummary object as input ###
Alpha &lt;- Alpha_diversity_calculator2(
  taxobj = testtax_summary,
  taxlevel = "Base"
)
head(Alpha)

# In genus level
Alpha &lt;- Alpha_diversity_calculator2(
  taxobj = testtax_summary,
  taxlevel = "Genus",
  prefix = "Genus"
)
head(Alpha)

### Input dataframe from reads table ###
Alpha &lt;- Alpha_diversity_calculator2(
  input = testotu,
  prefix = "Bacterial",
  inputformat = 1,
  reads = TRUE
)

### Input dataframe from relative abundance table ###
if (!require(magrittr)) install.packages("magrittr")
library(magrittr)
Alpha &lt;- Filter_function(
  input = testotu,
  threshold = 0,
  format = 1
) %&gt;%
  Alpha_diversity_calculator2(
    input = .,
    prefix = "Bacterial",
    inputformat = 1,
    reads = FALSE
  )
head(Alpha)
</code></pre>

<hr>
<h2 id='anova_report'>Print Analysis of Variance report</h2><span id='topic+anova_report'></span>

<h3>Description</h3>

<p>Print Analysis of Variance report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anova_report(
  data,
  treatment_col,
  value_col,
  prior = FALSE,
  comparison_method = "Auto",
  equally_rep = TRUE,
  report = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="anova_report_+3A_data">data</code></td>
<td>
<p>Data frame containing the treatment, value and other information.</p>
</td></tr>
<tr><td><code id="anova_report_+3A_treatment_col">treatment_col</code></td>
<td>
<p>Numeric indicating where treatment locates (column number) in data.</p>
</td></tr>
<tr><td><code id="anova_report_+3A_value_col">value_col</code></td>
<td>
<p>Numeric indicating where treatment value (column number) in data.</p>
</td></tr>
<tr><td><code id="anova_report_+3A_prior">prior</code></td>
<td>
<p>logical. Whether conducted prior comparisons.</p>
</td></tr>
<tr><td><code id="anova_report_+3A_comparison_method">comparison_method</code></td>
<td>
<p>Default would automaticly choose method. Method of multiple comparison,must be one of &quot;SNK&quot;, &quot;Tukey&quot;, &quot;bonferroni&quot;,&quot;LSD&quot; or &quot;Scheffe&quot;.</p>
</td></tr>
<tr><td><code id="anova_report_+3A_equally_rep">equally_rep</code></td>
<td>
<p>Logical. Whether all treatments have same number of replication.</p>
</td></tr>
<tr><td><code id="anova_report_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>anova_report returns list of:
</p>
<p>1)basic data description
</p>
<p>2)ANOVA model
</p>
<p>3)summary of ANOVA model
</p>
<p>4)model of multiple comparison
</p>
<p>5)difference of multiple comparison
</p>
<p>6)letters of multiple comparison, which could be use for visualization.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  #' Data loading from 'agricolae' package
  data("cotton", package = "agricolae")

  #' ANOVA report with default settings
  anova_results &lt;- anova_report(
    data = cotton,
    treatment_col = 3,
    value_col = 5
  )
  ## Here returns NULL because no significance among groups

  ## To conduct prior comparisons
  anova_results &lt;- anova_report(
    data = cotton,
    treatment_col = 3,
    value_col = 5,
    prior = TRUE
  )

  ## Here found no difference among groups, thus change to a more sensitive method
  ## (maybe illegal, but only as an example)
  anova_results &lt;- anova_report(
    data = cotton,
    treatment_col = 3,
    value_col = 5,
    prior = TRUE,
    comparison_method = "LSD"
  )

  #' Data loading 'iris' dataset
  data("iris")

  #' ANOVA report for 'iris' dataset
  anova_results &lt;- anova_report(
    data = iris,
    treatment_col = 5,
    value_col = 2
  )

  ### Extract return

  ### Basic data description
  print(anova_results$basicdata)

  ### ANOVA model
  print(anova_results$anova_model)

  ### Summary of ANOVA model
  print(anova_results$anova_summary)

  ### Model of multiple comparison
  print(anova_results$multiple_comparison_model)

  ### Difference of multiple comparison
  print(anova_results$comparison_results)

  ### Letters of multiple comparison, which could be used for visualization
  print(anova_results$comparison_letters)
}
</code></pre>

<hr>
<h2 id='auto_signif_test'>Automatic significance test</h2><span id='topic+auto_signif_test'></span>

<h3>Description</h3>

<p>Automatically conduct significance testing
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auto_signif_test(
  data,
  treatment_col,
  value_col,
  paired,
  subject_col,
  prior = FALSE,
  comparison_method = NULL,
  equally_rep = TRUE,
  output = "console",
  output_dir = "./",
  filename = "auto_signif_test",
  report = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="auto_signif_test_+3A_data">data</code></td>
<td>
<p>Data frame containing the treatment, value and other information.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_treatment_col">treatment_col</code></td>
<td>
<p>Numeric indicating where treatment locates (column number) in data.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_value_col">value_col</code></td>
<td>
<p>Numeric indicating where treatment value (column number) in data.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_paired">paired</code></td>
<td>
<p>Logical indicating whether you want a paired t-test.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_subject_col">subject_col</code></td>
<td>
<p>Only meaningful when Pair is ture. Numeric indicating where subject of treatment (column number) in data.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_prior">prior</code></td>
<td>
<p>logical. Whether conducted prior comparisons.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_comparison_method">comparison_method</code></td>
<td>
<p>Character string. Only use for more than 2 treatment. Default would automatically choose method. Method of multiple comparison,must be one of &quot;SNK&quot;, &quot;Tukey&quot;, &quot;bonferroni&quot;,&quot;LSD&quot; or &quot;Scheffe&quot;.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_equally_rep">equally_rep</code></td>
<td>
<p>Logical indicating Whether all treatments have same number of replication.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_output">output</code></td>
<td>
<p>A character string indicating output style. Default: &quot;console&quot;, which print the report in console. And &quot;file&quot; is available to output report into text-file.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_output_dir">output_dir</code></td>
<td>
<p>Default:&quot;./&quot;. Available only when output=&quot;file&quot;. The direction of output file.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_filename">filename</code></td>
<td>
<p>A character string indicating file name of output file. Only work when output set as 'file'.</p>
</td></tr>
<tr><td><code id="auto_signif_test_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>auto_signif_test returns results of significant test and print report in console or file. See details in example.
</p>
<p>See results return in <code><a href="#topic+t_test_report">t_test_report</a></code>, <code><a href="#topic+wilcox_test_report">wilcox_test_report</a></code>, <code><a href="#topic+anova_report">anova_report</a></code>, <code><a href="#topic+kruskal_report">kruskal_report</a></code>.
</p>


<h3>Note</h3>

<p>1.when choose output=&quot;file&quot;, once caused error that terminate the program, use 'sink()' to end the written of exist files.
</p>
<p>2.Please confirm your data is in format of dataframe, else may cause bug! (e.g. Do not use 'read.xlsx' to load data into tibble format)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Here shows different types of experimental design ###
data("cotton", package = "agricolae")

### Two randomly designed groups ###
sig_results &lt;- auto_signif_test(
  data = cotton,
  treatment_col = 1,
  value_col = 5
)

### Two paired design groups ###
sig_results &lt;- auto_signif_test(
  data = cotton,
  treatment_col = 1,
  value_col = 5,
  paired = TRUE,
  subject_col = 2
)

### More than two randomly designed groups ###
sig_results &lt;- auto_signif_test(
  data = cotton,
  treatment_col = 2,
  value_col = 5
)
head(sig_results)  # Check outputs

### Conduct prior comparisons ###
sig_results &lt;- auto_signif_test(
  data = cotton,
  treatment_col = 2,
  value_col = 5,
  prior = TRUE
)
head(sig_results)  # Check outputs
print(sig_results$basicdata)  # Check statistical summary
print(sig_results$anova_model)  # Extract ANOVA model
print(sig_results$anova_summary)  # Check ANOVA summary
print(sig_results$multiple_comparison_model)  # Extract multiple comparison model
print(sig_results$comparison_results)  # Check between-group comparison
print(sig_results$comparison_letters)  # Check letters (can be used in visualization)

## Change multiple comparison method (maybe not illegal!!)
sig_results &lt;- auto_signif_test(
  data = cotton,
  treatment_col = 2,
  value_col = 5,
  prior = TRUE,
  comparison_method = "LSD"
)
head(sig_results)  # Check outputs
print(sig_results$comparison_letters)  # Note that letters become different

</code></pre>

<hr>
<h2 id='circulation_lm'>Circulation of fitting Linear Models</h2><span id='topic+circulation_lm'></span>

<h3>Description</h3>

<p>Using circulation to fit linear models between one dependent variable and series of independent variable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circulation_lm(y, xframe, margin)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="circulation_lm_+3A_y">y</code></td>
<td>
<p>Dependent variable</p>
</td></tr>
<tr><td><code id="circulation_lm_+3A_xframe">xframe</code></td>
<td>
<p>Matrix or data frame of independent variable</p>
</td></tr>
<tr><td><code id="circulation_lm_+3A_margin">margin</code></td>
<td>
<p>A vector of 1 or 2 indicates arrangement of xframe. 1:by rows 2:by columns</p>
</td></tr>
</table>


<h3>Details</h3>

<p>if row names(for margin 1) and column names(for margin 2) are not given, ID column of return data frame will be row/column numbers.
</p>


<h3>Value</h3>

<p>Data frame contains lm statistics of all Independent Variable
</p>


<h3>Note</h3>

<p>Other arguments used in function lm were set as default. See in  <code><a href="stats.html#topic+lm">lm</a></code>.
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi<a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(testotu)

###using margin 1, arrange by rows##
dep=testotu[1,2:21]
in_dep=testotu[-1,2:21]
lm_stat&lt;-circulation_lm(y = dep,xframe = in_dep,margin = 1)
lm_stat

###using margin 2, arrange by column##
dep=testotu[,2]
in_dep=testotu[,3:21]
lm_stat&lt;-circulation_lm(y = dep,xframe = in_dep,margin = 2)
lm_stat

</code></pre>

<hr>
<h2 id='color_scheme'>Get color scheme</h2><span id='topic+color_scheme'></span>

<h3>Description</h3>

<p>color_scheme() can generate color scheme from nine color scheme database and expand into colorRamp
</p>


<h3>Usage</h3>

<pre><code class='language-R'>color_scheme(Plan, expand = NULL, names = NULL, show = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="color_scheme_+3A_plan">Plan</code></td>
<td>
<p>Character, 'Plan1' to 'Plan10' are optional.</p>
</td></tr>
<tr><td><code id="color_scheme_+3A_expand">expand</code></td>
<td>
<p>Numeric, default:NULL. Numeric indicating numbers to expand color scheme into colorRamp</p>
</td></tr>
<tr><td><code id="color_scheme_+3A_names">names</code></td>
<td>
<p>Character string. Names to assign for color scheme.</p>
</td></tr>
<tr><td><code id="color_scheme_+3A_show">show</code></td>
<td>
<p>Logical. If show assigned color in plot panel. Default:TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If parameter 'names' is not given, 'color_scheme' returns character string including color scheme.When 'names' is set,'color_scheme' returns named vector of color scheme.
</p>


<h3>Note</h3>

<p>1.Parameter 'names' is strongly recommended to assign for fixed color scheme, see details in ggplot::scale_color_manual
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Commonly used example ###
my_color &lt;- color_scheme(
  Plan = "Plan1",
  names = c("Treatment1", "Treatment2")
)

### Generate colorRamp still based on 'Plan1'
my_color &lt;- color_scheme(
  Plan = "Plan1",
  expand = 4,
  names = c("Treatment1", "Treatment2", "Treatment3", "Treatment4")
)

### View color scheme from plan1 to plan10 in 'Plots' interface ###
color_scheme(Plan = "Plan1")
color_scheme(Plan = "Plan2")
color_scheme(Plan = "Plan3")
color_scheme(Plan = "Plan4")
color_scheme(Plan = "Plan5")
color_scheme(Plan = "Plan6")
color_scheme(Plan = "Plan7")
color_scheme(Plan = "Plan8")
color_scheme(Plan = "Plan9")
color_scheme(Plan = "Plan10")
</code></pre>

<hr>
<h2 id='combine_and_translate'>Combine data for visualization</h2><span id='topic+combine_and_translate'></span>

<h3>Description</h3>

<p>Combine group information and index into data frame for visualization(scatter, bar plot, alluvial,box plot etc.).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_and_translate(inputframe, groupframe, itemname, indexname, inputtype)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine_and_translate_+3A_inputframe">inputframe</code></td>
<td>
<p>Data frame of index ,sample ID in column,requires all numeric(e.g. result from Alpha_diversity_calculator or Top_taxa function)</p>
</td></tr>
<tr><td><code id="combine_and_translate_+3A_groupframe">groupframe</code></td>
<td>
<p>Data frame of group information(and other abiotic/geographic factors)</p>
</td></tr>
<tr><td><code id="combine_and_translate_+3A_itemname">itemname</code></td>
<td>
<p>A character string of your inputframe itemname</p>
</td></tr>
<tr><td><code id="combine_and_translate_+3A_indexname">indexname</code></td>
<td>
<p>A character string of your inputframe indexname</p>
</td></tr>
<tr><td><code id="combine_and_translate_+3A_inputtype">inputtype</code></td>
<td>
<p>If sample ID were in row and index in column in inputframe.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>key-value pairs data frame
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi<a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  require(magrittr)
  data(testotu)

  ## Data preparation ##
  Alpha &lt;- Alpha_diversity_calculator2(
    input = testotu,
    prefix = "Bacterial",
    inputformat = 1,
    reads = TRUE
  )

  topotu &lt;- data.frame(
    Top_taxa(
      input = testotu,
      n = 10,
      inputformat = 1,
      outformat = 1
    )[, -1],
    row.names = paste0(rep("otu", 11), 1:11)
  )

  groupinformation1 &lt;- data.frame(
    group = c(rep("a", 10), rep("b", 10)),
    factor1 = rnorm(10),
    factor2 = rnorm(mean = 100, 10)
  )

  ### Use inputtype = FALSE ###
  head(Alpha)
  combine_and_translate(
    Alpha, groupinformation1,
    itemname = "Alpha", indexname = "index",
    inputtype = FALSE
  )

  ### Use inputtype = TRUE ###
  head(topotu)
  combine_and_translate(
    topotu, groupinformation1,
    itemname = "OTU", indexname = "reads",
    inputtype = TRUE
  )
}
</code></pre>

<hr>
<h2 id='community_plot'>Generate Community Composition Plot Based on Tax_summary Object</h2><span id='topic+community_plot'></span>

<h3>Description</h3>

<p>Microbial community composition visualization in format of barplot, areaplot and alluvialplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>community_plot(
  taxobj,
  taxlevel,
  n = 10,
  palette = "Spectral",
  nrow = NULL,
  rmprefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="community_plot_+3A_taxobj">taxobj</code></td>
<td>
<p>Configured tax summary objects.See in <code><a href="#topic+object_config">object_config</a></code>.</p>
</td></tr>
<tr><td><code id="community_plot_+3A_taxlevel">taxlevel</code></td>
<td>
<p>Character. taxonomy levels used for visualization.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).</p>
</td></tr>
<tr><td><code id="community_plot_+3A_n">n</code></td>
<td>
<p>Numeric. Top n taxa remained according to relative abundance. Default:10</p>
</td></tr>
<tr><td><code id="community_plot_+3A_palette">palette</code></td>
<td>
<p>Character. Palette for visualization,default:&quot;Spectral&quot;,recommended to use &quot;Paired&quot; for more than 15 tax.</p>
</td></tr>
<tr><td><code id="community_plot_+3A_nrow">nrow</code></td>
<td>
<p>Numeric. Number of rows when wrap panels,default:NULL.</p>
</td></tr>
<tr><td><code id="community_plot_+3A_rmprefix">rmprefix</code></td>
<td>
<p>Numeric. Removed prefix character in taxonomy annotation.Default:NULL. See details in example.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>community_plot2 returns three ggplot objects, two data frame used in visualization and one character of filled mapping colors
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi<a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  require(magrittr)
  ### Data preparation ###
  data("Two_group")

  ## Use taxonomy summary objects
  phylum10 &lt;- community_plot(
    taxobj = Two_group,
    taxlevel = "Phylum",
    n = 10,
    rmprefix = "p__"
  )

  phylum10$barplot  # Check bar plot
  phylum10$areaplot  # Check area plot
  phylum10$alluvialplot  # Check alluvial plot

  phylum10$Top10Phylum %&gt;% head(10)  # Check top taxa data frame
  phylum10$Grouped_Top10Phylum %&gt;% head(10)  # Check grouped top taxa data frame
  print(phylum10$filled_color)  # Check mapping colors

  # Double facet
  data("Facet_group")

  # Using palette by default
  phylum10 &lt;- community_plot(
    taxobj = Facet_group,
    taxlevel = "Phylum",
    n = 10,
    rmprefix = " p__"
  )
  phylum10$barplot
  phylum10$areaplot
  phylum10$alluvialplot

  # Another example
  genus20 &lt;- community_plot(
    taxobj = Facet_group,
    taxlevel = "Genus",
    n = 20,
    palette = "Paired",
    rmprefix = " g__"
  )
  genus20$alluvialplot
}
</code></pre>

<hr>
<h2 id='compare_plot'>Comparison plot generator
This function help generate comparsion plot including bar plot, box plot, and violin plot</h2><span id='topic+compare_plot'></span>

<h3>Description</h3>

<p>Comparison plot generator
This function help generate comparsion plot including bar plot, box plot, and violin plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_plot(
  inputframe,
  treat_location,
  value_location,
  aes_col = NULL,
  point = TRUE,
  facet_location = NULL,
  ylab_text = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="compare_plot_+3A_inputframe">inputframe</code></td>
<td>
<p>A data frame contain information for visualization.</p>
</td></tr>
<tr><td><code id="compare_plot_+3A_treat_location">treat_location</code></td>
<td>
<p>Numeric. Treatment column number in inputframe.</p>
</td></tr>
<tr><td><code id="compare_plot_+3A_value_location">value_location</code></td>
<td>
<p>Numeric. Value column number in inputframe.</p>
</td></tr>
<tr><td><code id="compare_plot_+3A_aes_col">aes_col</code></td>
<td>
<p>Named character string, default:NULL. A set of aesthetic character to map treatment to.</p>
</td></tr>
<tr><td><code id="compare_plot_+3A_point">point</code></td>
<td>
<p>Logical. If draw point on bar, box and violin plot. Default:TRUE.</p>
</td></tr>
<tr><td><code id="compare_plot_+3A_facet_location">facet_location</code></td>
<td>
<p>Numeric, default:NULL. Facet column number in inputframe.</p>
</td></tr>
<tr><td><code id="compare_plot_+3A_ylab_text">ylab_text</code></td>
<td>
<p>Character. Text for y axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contained plot and statistics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("iris")
results=compare_plot(inputframe=iris,treat_location=5,
                     value_location=1,ylab_text = "Sepal Length")

#Check statistics
results$Statistics
#Extract plot
results$Barplot
results$Boxplot
results$Violinplot


iris$Treat2=rep(c(rep("A",25),rep("B",25)),3)

results=compare_plot(inputframe=iris,treat_location=5,
                     value_location=1,facet_location = 6,
                     ylab_text = "Sepal Length")

#Check statistics
results$Statistics
#Extract plot
results$Barplot
results$Boxplot
results$Violinplot
#Extract combined plot
results$All_Barplot
results$All_Boxplot
results$All_Violinplot

</code></pre>

<hr>
<h2 id='Deseq_analysis'>Deseq Analysis Function</h2><span id='topic+Deseq_analysis'></span>

<h3>Description</h3>

<p>This function performs a differential expression analysis using the DESeq2 package.
It is designed to work with microbiome data and can handle paired or non-paired samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Deseq_analysis(
  taxobj,
  taxlevel,
  comparison = NULL,
  cutoff,
  control_name,
  paired = FALSE,
  subject = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Deseq_analysis_+3A_taxobj">taxobj</code></td>
<td>
<p>Configured tax summary objects.See in <code><a href="#topic+object_config">object_config</a></code>.</p>
</td></tr>
<tr><td><code id="Deseq_analysis_+3A_taxlevel">taxlevel</code></td>
<td>
<p>The taxonomic level for the analysis.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;)</p>
</td></tr>
<tr><td><code id="Deseq_analysis_+3A_comparison">comparison</code></td>
<td>
<p>A vector of conditions to compare. Default: NULL, all unique conditions are compared (only for Two groups).</p>
</td></tr>
<tr><td><code id="Deseq_analysis_+3A_cutoff">cutoff</code></td>
<td>
<p>The log2 fold change cutoff for considering as differential taxon.</p>
</td></tr>
<tr><td><code id="Deseq_analysis_+3A_control_name">control_name</code></td>
<td>
<p>Character. The name of the control group for the comparison.</p>
</td></tr>
<tr><td><code id="Deseq_analysis_+3A_paired">paired</code></td>
<td>
<p>Logical. Should the samples be treated as paired? Default: False</p>
</td></tr>
<tr><td><code id="Deseq_analysis_+3A_subject">subject</code></td>
<td>
<p>Optional. The subject identifier for paired samples. Default: Null</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the results of the differential expression analysis.
</p>


<h3>Note</h3>


<ol>
<li><p> Regulation is judged by cutoff of q-value(adjust p value).Detail see in <code><a href="DESeq2.html#topic+DESeq">DESeq</a></code>
</p>
</li>
<li><p> For more than two groups in taxobj, the 'comparison' must be assigned.
</p>
</li>
<li><p> The function requires the 'DESeq2', 'S4Vectors', and 'tibble' packages.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="DESeq2.html#topic+DESeqDataSetFromMatrix">DESeqDataSetFromMatrix</a></code>, <code><a href="DESeq2.html#topic+DESeq">DESeq</a></code>, <code><a href="S4Vectors.html#topic+DataFrame">DataFrame</a></code>, <code><a href="tibble.html#topic+as_tibble">as_tibble</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("DESeq2", quietly = TRUE) &amp;&amp;
    requireNamespace("S4Vectors", quietly = TRUE) &amp;&amp;
    requireNamespace("tibble", quietly = TRUE)) {

    ### Data preparation ###
    data("Two_group")

    ### Deseq analysis ###
    deseq_results &lt;- Deseq_analysis(
      taxobj = Two_group,
      taxlevel = "Genus",
      cutoff = 1,
      control_name = "Control"
    )

    # Visualization of volcano plot ##
    volcano_plot &lt;- volcano_plot(
      inputframe = deseq_results,
      cutoff = 1,
      aes_col = Two_group$configuration$treat_col
    )
    volcano_plot$FC_FDR
    volcano_plot$Mean_FC

    # Visualization of Manhattan plot ##
    manhattan_object &lt;- manhattan(
      inputframe = deseq_results,
      taxlevel = "Phylum",
      control_name = "Control",
      mode = "most",
      top_n = 10,
      rmprefix = "p__"
    )
    manhattan_object$manhattan  # Tradition manhattan plot
    manhattan_object$manhattan_circle  # Circular manhattan plot

    # For object with more than two groups
    ### Data preparation ###
    data("Three_group")

    # Specific comparison
    deseq_results_BFCF &lt;- Deseq_analysis(
      taxobj = Three_group,
      taxlevel = "Genus",
      comparison = c("BF", "CF"),
      cutoff = 1,
      control_name = "CF"
    )
    volcano_plot &lt;- volcano_plot(
      inputframe = deseq_results_BFCF,
      cutoff = 1,
      aes_col = Three_group$configuration$treat_col
    )
    volcano_plot$FC_FDR
  } else {
    message(
      "The 'DESeq2', 'S4Vectors', and/or 'tibble' package(s) are not installed. ",
      "Please install them to use all features of Deseq_analysis."
    )
  }

</code></pre>

<hr>
<h2 id='Deseq_analysis2'>Deseq analysis</h2><span id='topic+Deseq_analysis2'></span>

<h3>Description</h3>

<p>Deseq analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Deseq_analysis2(inputframe, condition, cutoff, control_name, paired, subject)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Deseq_analysis2_+3A_inputframe">inputframe</code></td>
<td>
<p>Otu/gene/taxa table with all integer numeric variables.Rownames must be  Otu/gene/taxa names,colnames must be sample names with control in front and treatment behind. Reads table is recommended.</p>
</td></tr>
<tr><td><code id="Deseq_analysis2_+3A_condition">condition</code></td>
<td>
<p>A character string which indicates group of samples</p>
</td></tr>
<tr><td><code id="Deseq_analysis2_+3A_cutoff">cutoff</code></td>
<td>
<p>threshold of log2(Foldchange).Detail see in <code><a href="DESeq2.html#topic+DESeq">DESeq</a></code></p>
</td></tr>
<tr><td><code id="Deseq_analysis2_+3A_control_name">control_name</code></td>
<td>
<p>A character indicating the control group name</p>
</td></tr>
<tr><td><code id="Deseq_analysis2_+3A_paired">paired</code></td>
<td>
<p>Logical to determine if paired comparision would be used. TRUE or FALSE.</p>
</td></tr>
<tr><td><code id="Deseq_analysis2_+3A_subject">subject</code></td>
<td>
<p>A character string which indicates paired design of samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Statistics dataframe of all otu/gene/taxa
</p>


<h3>Note</h3>


<ol>
<li><p> Inputframe must be all integer numeric variables without NA/NAN/inf! In case your data is not an integer one,a practical method is to multiply them in equal proportion(eg. x 1e6) then round them into integer
</p>
</li>
<li><p> Regulation is judged by cutoff of qvalue(adjust p value).Detail see in <code><a href="DESeq2.html#topic+DESeq">DESeq</a></code>
</p>
</li>
<li><p> Set cutoff as 1 is recommened.In case of too few taxa(eg. Phylum level deseq),cutoff can be set to 0.
</p>
</li>
<li><p> if control_name is not given, the control group will be set according to ASCII
</p>
</li>
<li><p> The function requires the 'DESeq2', 'S4Vectors', and 'tibble' packages.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>See Also</h3>

<p><code><a href="DESeq2.html#topic+DESeqDataSetFromMatrix">DESeqDataSetFromMatrix</a></code>, <code><a href="DESeq2.html#topic+DESeq">DESeq</a></code>, <code><a href="S4Vectors.html#topic+DataFrame">DataFrame</a></code>, <code><a href="tibble.html#topic+as_tibble">as_tibble</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
{
  ### Data preparation ###
  data(testotu)
  rownames(testotu) &lt;- testotu[, 1]
  inputotu &lt;- testotu[, -c(1, ncol(testotu))]
  head(inputotu)
  group &lt;- c(rep("a", 10), rep("b", 10))

  ### DESeq analysis ###
  if (requireNamespace("DESeq2", quietly = TRUE) &amp;&amp;
    requireNamespace("S4Vectors", quietly = TRUE) &amp;&amp;
    requireNamespace("tibble", quietly = TRUE)) {
    Deseqresult &lt;- Deseq_analysis2(
      inputframe = inputotu,
      condition = group,
      cutoff = 1,
      control_name = "b"
    )

    ### Paired DESeq analysis ###
    subject &lt;- factor(c(1:10, 1:10))
    Deseqresult &lt;- Deseq_analysis2(
      inputframe = inputotu,
      condition = group,
      cutoff = 1,
      control_name = "b",
      paired = TRUE,
      subject = subject
    )
  }
}

</code></pre>

<hr>
<h2 id='differential_bar'>Generate Differential Bar Plot and Error bar Plot</h2><span id='topic+differential_bar'></span>

<h3>Description</h3>

<p>Generate Differential Bar Plot and Error bar Plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>differential_bar(
  taxobj,
  taxlevel,
  comparison = NULL,
  rel_threshold = 0.005,
  anno_row = "taxonomy",
  aes_col = NULL,
  limit_num = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="differential_bar_+3A_taxobj">taxobj</code></td>
<td>
<p>Configured tax summary objects.See in <code><a href="#topic+object_config">object_config</a></code>.</p>
</td></tr>
<tr><td><code id="differential_bar_+3A_taxlevel">taxlevel</code></td>
<td>
<p>Taxonomy levels used for visualization. Must be one of
c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).</p>
</td></tr>
<tr><td><code id="differential_bar_+3A_comparison">comparison</code></td>
<td>
<p>A vector of conditions to compare. Default: NULL, all unique conditions are compared (only for Two groups).</p>
</td></tr>
<tr><td><code id="differential_bar_+3A_rel_threshold">rel_threshold</code></td>
<td>
<p>Threshold filtering taxa for differential analysis. Default:0.005</p>
</td></tr>
<tr><td><code id="differential_bar_+3A_anno_row">anno_row</code></td>
<td>
<p>Default: 'taxonomy'. Rownames for visualization. Options are 'taxonomy' for showing taxonomic information and 'ID' for showing taxonomic ID.</p>
</td></tr>
<tr><td><code id="differential_bar_+3A_aes_col">aes_col</code></td>
<td>
<p>A named vector of colors to be used in the plots.</p>
</td></tr>
<tr><td><code id="differential_bar_+3A_limit_num">limit_num</code></td>
<td>
<p>Numeric. The maximum number of significant results to display. Default: NULL, showing all differential taxa.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the bar plot, source data for the bar plot, difference plot, and source data for the difference plot.
</p>


<h3>Note</h3>

<p>The differential analysis is performed using two-sided Welch's t-test.
The p-values are adjusted using the 'BH' (i.e., FDR) method.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # Data preparation
  data("Two_group")

  # Simple mode
  diff_results &lt;- differential_bar(
    taxobj = Two_group,
    taxlevel = "Genus"
  )
  print(diff_results$Barplot)  # Print Barplot
  head(diff_results$Barplot_sourcedata)  # Show source data of barplot
  print(diff_results$Differenceplot)  # Print Differential errorbar plot
  head(diff_results$Differenceplot_sourcedata)  # Show source data of Differential errorbar plot

  require(patchwork)
  diff_results$Barplot|diff_results$Differenceplot
  # Displaying ID
  diff_results &lt;- differential_bar(
    taxobj = Two_group,
    taxlevel = "Base",
    anno_row = "ID"
  )
  print(diff_results$Barplot)

  # Threshold adjustment
  diff_results &lt;- differential_bar(
    taxobj = Two_group,
    taxlevel = "Base",
    rel_threshold = 0.001
  )
  print(diff_results$Barplot)

  # Limit the displaying number
  diff_results &lt;- differential_bar(
    taxobj = Two_group,
    taxlevel = "Base",
    rel_threshold = 0.001,
    limit_num = 10
  )
  print(diff_results$Barplot)

  # For object with more than two groups
  # Data preparation
  data("Three_group")

  # Specific comparison
  Three_group_col &lt;- Three_group$configuration$treat_col
  diff_results &lt;- differential_bar(
    taxobj = Three_group,
    taxlevel = "Genus",
    comparison = c("BF", "CF"),
    aes_col = Three_group_col
  )
  print(diff_results$Barplot)
}
</code></pre>

<hr>
<h2 id='Dimension_reduction'>Dimension_reduction: PCA, PCOA, and NMDS Analysis</h2><span id='topic+Dimension_reduction'></span>

<h3>Description</h3>

<p>Performs dimension reduction analysis using PCA, PCOA, or NMDS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dimension_reduction(inputframe, group, format)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Dimension_reduction_+3A_inputframe">inputframe</code></td>
<td>
<p>An OTU/gene/taxa table with all numeric variables and no NA/NAN/inf values.</p>
</td></tr>
<tr><td><code id="Dimension_reduction_+3A_group">group</code></td>
<td>
<p>Group information with the sample order the same as in inputframe.</p>
</td></tr>
<tr><td><code id="Dimension_reduction_+3A_format">format</code></td>
<td>
<p>The format of analysis: 1 for PCA, 2 for PCOA, 3 for NMDS.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing data frames and other statistics for dimension reduction analysis.
</p>


<h3>Note</h3>

<p>Inputframe should be a numeric matrix without NA/NAN/inf values.
</p>
<p>The row names of inputframe should be set as OTU/gene/taxa annotations for further analysis.
</p>
<p>The results are combined into a list for output. Use <code>as.data.frame(result[[1]])</code> to extract the data frame, and <code style="white-space: pre;">&#8288;$result$&#8288;</code> to extract other statistics. See examples for details.
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Data preparation ###
data(testotu)
rownames(testotu) &lt;- testotu[, 1]
inputotu &lt;- testotu[, -c(1, ncol(testotu))]
head(inputotu)

groupinformation1 &lt;- data.frame(
  group = c(rep("a", 10), rep("b", 10)),
  factor1 = rnorm(10),
  factor2 = rnorm(mean = 100, 10)
)

### PCA ###
PCAresult &lt;- Dimension_reduction(inputotu, groupinformation1, 1)
PCAframe &lt;- PCAresult$outframe   # Extract data for visualization
head(PCAresult$data.pca$rotation,5)  # OTU coordinates

### PCOA ###
PCOAresult &lt;- Dimension_reduction(inputotu, groupinformation1, 2)
PCOAframe &lt;- PCOAresult$outframe  # Extract data for visualization
head(PCOAresult$PCOA$values,2)  # Explanation of first two axis

### NMDS ###
NMDSresult &lt;- Dimension_reduction(inputotu, groupinformation1, 3)
NMDSframe &lt;- NMDSresult$outframe  # Extract data for visualization
# Here we got a warning of `stress is (nearly) zero: you may have insufficient data`,
# so make sure you have sufficient data for NMDS
print(NMDSresult$NMDSstat$stress)  # Extract stress of NMDS
</code></pre>

<hr>
<h2 id='Facet_group'>Tax summary object with Facet 2x2 Groups</h2><span id='topic+Facet_group'></span>

<h3>Description</h3>

<p>Enraptured summary object with facet 2x2 Groups.Configuration has been assigned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Facet_group
</code></pre>


<h3>Format</h3>

<p>Tax summary object with configuration
</p>

<hr>
<h2 id='Filter_function'>Filter OTU/ASV/metagenomic profile/gene profile by threshold</h2><span id='topic+Filter_function'></span>

<h3>Description</h3>

<p>Sequenced data of taxonomy&amp;gene still remains some sequencing error which we needed to be wiped off before analyzing.
Here we provide function including four formats to wipe them clean.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Filter_function(input, threshold, format, report = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Filter_function_+3A_input">input</code></td>
<td>
<p>Data frame of absolute abundance of standard OTU table,with the first column of OTUID and the final column of taxonomy annotation.
If your data frame is gene table or not a standard OTU table, please manually transformed into a standard input data frame.</p>
</td></tr>
<tr><td><code id="Filter_function_+3A_threshold">threshold</code></td>
<td>
<p>threshold of filter.Relative abundance for format 1 and 4, reads number for format 2, sample size for format 3</p>
</td></tr>
<tr><td><code id="Filter_function_+3A_format">format</code></td>
<td>
<p>1:filter OTU/gene below overall-sample relative abundance threshold(&lt;)
</p>
<p>2:filter OTU/gene below overall-sample reads threshold(&lt;)
</p>
<p>3:filter OTU/gene reads 0 over threshold sample size(&gt;)
</p>
<p>4:filter OTU/gene below relative abundance threshold in each sample(&lt;)</p>
</td></tr>
<tr><td><code id="Filter_function_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Dataframe of OTU/gene in format of absolute abundacne(reads) or relative abundance(%)
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Data frame with absolute abundance (reads)###
### And first column of OTUID and last column of taxonomy ###
data(testotu)

#### If your data frame does not contain the OTUID column or taxonomy column,
#### you can add a simulated column to fit the input format like testotu ##

### 1. Filter OTU with total relative abundance below 0.0001###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 0.0001,
  format = 1
)

### 2. Filter OTU with total reads below 20 ###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 20,
  format = 2
)

### 3. Filter OTU reads 0 over (&gt;=) 11 samples ###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 11,
  format = 3
)

### 4. Filter OTU with relative abundance below 0.0001 in each sample ###
filtered_otu &lt;- Filter_function(
  input = testotu,
  threshold = 0.0001,
  format = 4
)
</code></pre>

<hr>
<h2 id='indicator_analysis'>Indicator Analysis</h2><span id='topic+indicator_analysis'></span>

<h3>Description</h3>

<p>Performs the indicator analysis based on taxonomic summary object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indicator_analysis(taxobj, taxlevel, func = "r.g", reads = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="indicator_analysis_+3A_taxobj">taxobj</code></td>
<td>
<p>Configured tax summary objects.See in <code><a href="#topic+object_config">object_config</a></code>.</p>
</td></tr>
<tr><td><code id="indicator_analysis_+3A_taxlevel">taxlevel</code></td>
<td>
<p>taxonomy levels used for visualization.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).</p>
</td></tr>
<tr><td><code id="indicator_analysis_+3A_func">func</code></td>
<td>
<p>Default: &quot;r.g&quot;.The function to use for the indicator analysis, see in <code><a href="indicspecies.html#topic+multipatt">multipatt</a></code></p>
</td></tr>
<tr><td><code id="indicator_analysis_+3A_reads">reads</code></td>
<td>
<p>A logical value indicating whether the input data is in terms of raw reads (TRUE) or relative abundance (FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the results of the indicator analysis, including adjusted p-values, tags and taxonomic information.
</p>


<h3>Note</h3>

<p>This function depends on the following packages: indicspecies, permute.
These packages are not automatically loaded and should be installed before using this function.
</p>


<h3>See Also</h3>

<p><code><a href="indicspecies.html#topic+multipatt">multipatt</a></code>, <code><a href="permute.html#topic+how">how</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("Two_group")
if (requireNamespace("indicspecies", quietly = TRUE) &amp;&amp;
    requireNamespace("permute", quietly = TRUE)) {
  set.seed(999)
  indicator_results &lt;- indicator_analysis(
    taxobj = Two_group,
    taxlevel = "Genus"
  )
  head(indicator_results)
}
</code></pre>

<hr>
<h2 id='kruskal_report'>Print Kruskal-Wallis Rank Sum Test report</h2><span id='topic+kruskal_report'></span>

<h3>Description</h3>

<p>Print Kruskal-Wallis Rank Sum Test report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>kruskal_report(
  data,
  treatment_col,
  value_col,
  prior = FALSE,
  comparison_method = "Auto",
  equally_rep = TRUE,
  report = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="kruskal_report_+3A_data">data</code></td>
<td>
<p>Data frame containing the treatment, value and other information.</p>
</td></tr>
<tr><td><code id="kruskal_report_+3A_treatment_col">treatment_col</code></td>
<td>
<p>Numeric indicating where treatment locates (column number) in data.</p>
</td></tr>
<tr><td><code id="kruskal_report_+3A_value_col">value_col</code></td>
<td>
<p>Numeric indicating where treatment value (column number) in data.</p>
</td></tr>
<tr><td><code id="kruskal_report_+3A_prior">prior</code></td>
<td>
<p>logical. Whether conducted prior comparisons.</p>
</td></tr>
<tr><td><code id="kruskal_report_+3A_comparison_method">comparison_method</code></td>
<td>
<p>Default would automaticly choose method. Method of multiple comparison,must be one of &quot;SNK&quot; or &quot;Tukey&quot;.</p>
</td></tr>
<tr><td><code id="kruskal_report_+3A_equally_rep">equally_rep</code></td>
<td>
<p>Logical. Whether all treatments have same number of replication.</p>
</td></tr>
<tr><td><code id="kruskal_report_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>kruskal_report returns list with
</p>
<p>1)basic data description
</p>
<p>2)summary of Kruskal-Wallis Rank Sum Test
</p>
<p>3)model of multiple comparison
</p>
<p>4)difference of multiple comparison
</p>
<p>5)letters of multiple comparison, which could be use for visualization.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("cotton",package ="agricolae" )
kruskal_results=kruskal_report(data = cotton,treatment_col =3,value_col = 5)
##here returns NULL because no significance among groups

##to conduct prior comparisons.
kruskal_results=kruskal_report(data = cotton,treatment_col =3,value_col = 5,prior = TRUE)


data("iris")
kruskal_results=kruskal_report(data = iris,treatment_col = 5,value_col = 2)

###extract return##

###basic data description
kruskal_results$basicdata

###summry of Kruskal-Wallis Rank Sum Test
kruskal_results$Kruskal_Wallis_summary

###model of multiple comparision
kruskal_results$muiltiple_comparision_model

###difference of multiple comparision
kruskal_results$comparision_results

###letters of multiple comparision, which could be use for visualization.
kruskal_results$comparison_letters

</code></pre>

<hr>
<h2 id='LorMe'>LorMe package: Lightening One-Code Resolving Microbial Ecology Program</h2><span id='topic+LorMe'></span>

<h3>Description</h3>

<p>LorMe package summarizes a series of functions normally used in microbiome analysis
analysis.
</p>


<h3>Details</h3>

<p>_PACKAGE
</p>
<p>#Basic functions####
</p>
<p><code><a href="#topic+auto_signif_test">auto_signif_test</a></code> Automatically conduct significance testing
</p>
<p><code><a href="#topic+compare_plot">compare_plot</a></code> Comparison plot generator
</p>
<p><code><a href="#topic+Filter_function">Filter_function</a></code> Filter OTU/ASV/metagenomic profile/gene profile by threshold
</p>
<p><code><a href="#topic+tax_summary">tax_summary</a></code> Encapsulate meta file, feature tables and taxonomy annotation into tax summary object
</p>
<p><code><a href="#topic+sub_tax_summary">sub_tax_summary</a></code> subsets tax summary objects according to meta file
</p>
<p><code><a href="#topic+combine_and_translate">combine_and_translate</a></code> Combine feature table with meta file and transform into a recognizable data frame for visualization.
</p>
<p><code><a href="#topic+color_scheme">color_scheme</a></code> generate color scheme from nine color scheme database and expand into colorRamp
</p>
<p><code><a href="#topic+theme_zg">theme_zg</a></code> A classic theme for ggplot.
</p>
<p>#Community features####
</p>
<p><code><a href="#topic+Alpha_diversity_calculator">Alpha_diversity_calculator</a></code> Calculator for alpha diversity of each sample.
</p>
<p><code><a href="#topic+Dimension_reduction">Dimension_reduction</a></code> Dimension reduction analysis including PCA,PCOA and NMDS
</p>
<p><code><a href="#topic+structure_plot">structure_plot</a></code> A fast view of microbial structure with PCA plot,PCOA plot and NMDS plot.
</p>
<p><code><a href="#topic+Top_taxa">Top_taxa</a></code> Calculate most abundant taxon
</p>
<p><code><a href="#topic+community_plot">community_plot</a></code> A fast view of microbial community with bar plot,alluvial plot and area plot.
</p>
<p>#Differential analysis####
</p>
<p><code><a href="#topic+Deseq_analysis">Deseq_analysis</a></code> Performs a differential expression analysis
</p>
<p><code><a href="#topic+indicator_analysis">indicator_analysis</a></code> Performs the indicator analysis based on taxonomic summary object
</p>
<p><code><a href="#topic+differential_bar">differential_bar</a></code> Generate Differential Bar Plot and errorbar plot
</p>
<p><code><a href="#topic+volcano_plot">volcano_plot</a></code> Generate volcano plot base on Deseq_analysis or indicator_analysis results
</p>
<p><code><a href="#topic+manhattan">manhattan</a></code> Generate Manhattan Plot  base on Deseq_analysis or indicator_analysis results
</p>
<p>#Network analysis####
</p>
<p><code><a href="#topic+network_analysis">network_analysis</a></code> A convenient and fast network analysis function, with output results suitable for cytoscape and gephi
</p>
<p><code><a href="#topic+network_withdiff">network_withdiff</a></code> Meta network analysis integrating differential taxon into a network analysis
</p>
<p><code><a href="#topic+network_visual">network_visual</a></code> Visualizes a network based on network object from <code><a href="#topic+network_analysis">network_analysis</a></code>
</p>
<p><code><a href="#topic+network_visual_re">network_visual_re</a></code>  Re-visualize or adjust network plot from <code><a href="#topic+network_visual">network_visual</a></code> or <code><a href="#topic+network_withdiff">network_withdiff</a></code>
</p>
<p><code><a href="#topic+Module_composition">Module_composition</a></code> Pie chart for network module composition
</p>
<p><code><a href="#topic+Module_abundance">Module_abundance</a></code> Calculate network module abundance for each sample
</p>
<p><code><a href="#topic+nc">nc</a></code> Calculate network Natural Connectivity
</p>
<p><code><a href="#topic+NC_remove">NC_remove</a></code> Conduct natural connectivity analysis
</p>
<p>#Correlation analysis####
</p>
<p><code><a href="#topic+circulation_lm">circulation_lm</a></code> Quick test using circulation to fit linear models between one dependent variable and series of independent variable
</p>
<p><code><a href="#topic+tbRDA_analysis">tbRDA_analysis</a></code> RDA analysis including co-linearity diagnostics and necessary statistics.
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi
</p>

<hr>
<h2 id='manhattan'>Manhattan Plot Generator</h2><span id='topic+manhattan'></span>

<h3>Description</h3>

<p>Generate Manhattan Plot  base on Deseq_analysis or indicator_analysis results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan(
  inputframe,
  taxlevel = "Phylum",
  control_name,
  mode = "all",
  top_n = NULL,
  palette = "Set1",
  select_tax = NULL,
  rmprefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="manhattan_+3A_inputframe">inputframe</code></td>
<td>
<p>A data frame generated from <code><a href="#topic+Deseq_analysis">Deseq_analysis</a></code> or <code><a href="#topic+indicator_analysis">indicator_analysis</a></code></p>
</td></tr>
<tr><td><code id="manhattan_+3A_taxlevel">taxlevel</code></td>
<td>
<p>Taxonomy levels used for visualization.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).Default:NULL.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_control_name">control_name</code></td>
<td>
<p>Character. The name of the control group for the comparison.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_mode">mode</code></td>
<td>
<p>The mode for selecting which taxa to plot: &quot;all&quot; for all taxa, &quot;most&quot; for the top N taxa, and &quot;select&quot; for specific taxa selection</p>
</td></tr>
<tr><td><code id="manhattan_+3A_top_n">top_n</code></td>
<td>
<p>The number of top taxa to plot when mode is set to &quot;most&quot;</p>
</td></tr>
<tr><td><code id="manhattan_+3A_palette">palette</code></td>
<td>
<p>Character. Palette for visualization,default:&quot;Set1&quot;.Optional palette same as 'RColorBrewer'. &quot;Plan1&quot; to &quot;Plan10&quot; were also optional,see in <code><a href="#topic+color_scheme">color_scheme</a></code></p>
</td></tr>
<tr><td><code id="manhattan_+3A_select_tax">select_tax</code></td>
<td>
<p>A vector of taxa to be selected for plotting when mode is &quot;select&quot;.</p>
</td></tr>
<tr><td><code id="manhattan_+3A_rmprefix">rmprefix</code></td>
<td>
<p>A string prefix to be removed from the taxonomic annotation.Default:NULL.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing the Manhattan plot, circular Manhattan plot, source data, and color assignments
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
{
  # Data preparation
  data("Two_group")

  # DESeq analysis
  deseq_results &lt;- Deseq_analysis(
    taxobj = Two_group,
    taxlevel = "Base",
    cutoff = 1,
    control_name = "Control"
  )

  # Indicator analysis
  indicator_results &lt;- indicator_analysis(
    taxobj = Two_group,
    taxlevel = "Genus"
  )

  # Show all with Manhattan plot
    manhattan_object &lt;- manhattan(
      inputframe = deseq_results,
      taxlevel = "Phylum",
      control_name = "Control"
    )
    print(manhattan_object$manhattan)  # Tradition Manhattan plot
    print(manhattan_object$manhattan_circle)  # Circular Manhattan plot
    print(manhattan_object$sourcedata)  # Source data for plot
    print(manhattan_object$aes_color)  # Aesthetic color for plot

  # Top 8 Phyla with most taxon
    manhattan_object &lt;- manhattan(
      inputframe = indicator_results,
      taxlevel = "Phylum",
      control_name = "Control",
      mode = "most",
      top_n = 8,
      palette = "Set1"
    )
    print(manhattan_object$manhattan)

  # Specific phyla
  # Top nine dominant phyla
    community &lt;- community_plot(
      taxobj = Two_group,
      taxlevel = "Phylum",
      n = 9,
      palette = "Paired",
      rmprefix = "p__"
    )

    manhattan_object &lt;- manhattan(
      inputframe = indicator_results,
      taxlevel = "Phylum",
      control_name = "Control",
      mode = "select",
      palette = community$filled_color,
      select_tax = names(community$filled_color),
      rmprefix = "p__"
    )
    print(manhattan_object$manhattan)
    print(manhattan_object$manhattan_circle)
}

</code></pre>

<hr>
<h2 id='Module_abundance'>Calculate network module abundance for each sample</h2><span id='topic+Module_abundance'></span>

<h3>Description</h3>

<p>Calculate network module abundance for each sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Module_abundance(network_obj, No.module)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Module_abundance_+3A_network_obj">network_obj</code></td>
<td>
<p>Network analysis results generated from <code><a href="#topic+network_analysis">network_analysis</a></code></p>
</td></tr>
<tr><td><code id="Module_abundance_+3A_no.module">No.module</code></td>
<td>
<p>Numeric or numeric vector of No.module</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing module abundance in metafile and column table of corresponding data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data preparation
data("Two_group")
##network analysis
network_results&lt;- network_analysis(taxobj = Two_group,taxlevel = "Genus",n = 10,threshold = 0.8)
require(ggplot2)
#one module
moduleframe=Module_abundance(network_obj =network_results,No.module = 3 )
moduleframe$rowframe   #combine into metafile
moduleframe$columnframe #column table
#statistics
moduleframe$plotlist$Plotobj_Module3$Statistics
#extract plot
moduleframe$plotlist$Plotobj_Module3$Barplot
moduleframe$plotlist$Plotobj_Module3$Boxplot
moduleframe$plotlist$Plotobj_Module3$Violinplot


#multiple modules
moduleframe=Module_abundance(network_results,c(1,3,6))
moduleframe$rowframe
moduleframe$columnframe #column table can be used in ggplot visualization
#same as above to extract plots and statistics
moduleframe$plotlist$Plotobj_Module6$Barplot
</code></pre>

<hr>
<h2 id='Module_composition'>Pie chart for network module composition</h2><span id='topic+Module_composition'></span>

<h3>Description</h3>

<p>This function analyzes the composition of modules within a network object,
providing a visual and data summary based on taxonomic levels.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Module_composition(
  network_obj,
  No.module,
  taxlevel = "Phylum",
  mode = "all",
  top_n = NULL,
  palette = "Set1",
  select_tax = NULL,
  rmprefix = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Module_composition_+3A_network_obj">network_obj</code></td>
<td>
<p>Network analysis results generated from <code><a href="#topic+network_analysis">network_analysis</a></code></p>
</td></tr>
<tr><td><code id="Module_composition_+3A_no.module">No.module</code></td>
<td>
<p>Numeric or numeric vector of No.module</p>
</td></tr>
<tr><td><code id="Module_composition_+3A_taxlevel">taxlevel</code></td>
<td>
<p>Taxonomy levels used for visualization.Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).Default:&quot;Phylum&quot;.</p>
</td></tr>
<tr><td><code id="Module_composition_+3A_mode">mode</code></td>
<td>
<p>The mode for selecting which taxa to plot: &quot;all&quot; for all taxa, &quot;most&quot; for the top N taxa, and &quot;select&quot; for specific taxa selection</p>
</td></tr>
<tr><td><code id="Module_composition_+3A_top_n">top_n</code></td>
<td>
<p>The number of top taxa to plot when mode is set to &quot;most&quot;</p>
</td></tr>
<tr><td><code id="Module_composition_+3A_palette">palette</code></td>
<td>
<p>Character. Palette for visualization,default:&quot;Set1&quot;.See optional palette in same as 'RColorBrewer'. And &quot;Plan1&quot; to &quot;Plan10&quot; were also optional,see in <code><a href="#topic+color_scheme">color_scheme</a></code></p>
</td></tr>
<tr><td><code id="Module_composition_+3A_select_tax">select_tax</code></td>
<td>
<p>A vector of taxa to be selected for plotting when mode is &quot;select&quot;.</p>
</td></tr>
<tr><td><code id="Module_composition_+3A_rmprefix">rmprefix</code></td>
<td>
<p>A string prefix to be removed from the taxonomic annotation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list containing pie chart of specific module,corresponding source data and color assignments
</p>


<h3>Examples</h3>

<pre><code class='language-R'> #Data loading
 data("Two_group")

 # Network analysis
 network_Two_group &lt;- network_analysis(
    taxobj = Two_group,
    taxlevel = "Genus",
    reads = TRUE,
    n = 8,
    threshold = 0.7
  )

  # Show all taxa
  module_results &lt;- Module_composition(
    network_obj = network_Two_group,
    No.module = c(2, 5),
    taxlevel = "Phylum"
  )
  print(module_results$Module5$Pie)
  print(module_results$Module2$Pie)  # View pie chart
  head(module_results$Module2$source_data_Module2)  # View source data for pie chart
  print(module_results$aes_color)  # Check aesthetic color

  # Show taxa with top five frequency
  module_results &lt;- Module_composition(
    network_obj = network_Two_group,
    No.module = c(2, 5),
    taxlevel = "Phylum",
    mode = "most",
    top_n = 5
  )
  print(module_results$Module2$Pie_plot_Module2)

  # Show specific taxa
  community &lt;- community_plot(
    taxobj = Two_group,
    taxlevel = "Phylum",
    n = 5,
    palette = "Paired"
  )  # Get top 5 dominant phyla
  top5_phyla &lt;- names(community$filled_color)

  module_results &lt;- Module_composition(
    network_obj = network_Two_group,
    No.module = c(2, 5),
    taxlevel = "Phylum",
    mode = "select",
    palette = community$filled_color,
    select_tax = top5_phyla
  )
  print(module_results$Module2$Pie_plot_Module2)

  # Specific taxa with no prefix 'p__'
  module_results &lt;- Module_composition(
    network_obj = network_Two_group,
    No.module = 2,
    taxlevel = "Phylum",
    mode = "select",
    select_tax = c("Proteobacteria", "Actinobacteria")
  )
  print(module_results$Module2$Pie_plot_Module2)

  # Remove 'p__' prefix
  module_results &lt;- Module_composition(
    network_obj = network_Two_group,
    No.module = 2,
    taxlevel = "Phylum",
    mode = "most",
    top_n = 5,
    palette = "Set2",
    rmprefix = "p__"
  )
  print(module_results$Module2$Pie_plot_Module2)
</code></pre>

<hr>
<h2 id='nc'>Calculate Network Natural Connectivity</h2><span id='topic+nc'></span>

<h3>Description</h3>

<p>Calculate Network Natural Connectivity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nc(adj_matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="nc_+3A_adj_matrix">adj_matrix</code></td>
<td>
<p>Adjacency data frame or matrix. Can be calculated from <code><a href="#topic+network_analysis">network_analysis</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value of natural connectivity
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  ### Data preparation ###
  data("Two_group")

  ### One input network analysis ###
  network_results &lt;- network_analysis(
    taxobj = Two_group,
    taxlevel = "Base",
    reads = FALSE,
    n = 10,
    threshold = 0.6
  )

  # Convert network results to a data frame for the adjacency matrix
  network_matrix &lt;- as.data.frame(network_results[[3]])  # Complete adjacency matrix

  # Check initial natural connectivity
  nc_initial &lt;- nc(network_matrix)
  print(nc_initial)  # Print the initial natural connectivity
}
</code></pre>

<hr>
<h2 id='NC_remove'>Natural connectivity analysis</h2><span id='topic+NC_remove'></span>

<h3>Description</h3>

<p>Natural connectivity analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NC_remove(input, num, seed = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NC_remove_+3A_input">input</code></td>
<td>
<p>Network adjacency matrix. Can be generated by <code><a href="#topic+network_analysis">network_analysis</a></code></p>
</td></tr>
<tr><td><code id="NC_remove_+3A_num">num</code></td>
<td>
<p>Max number of removed nodes. Default: Automatically match max number that can be removed.</p>
</td></tr>
<tr><td><code id="NC_remove_+3A_seed">seed</code></td>
<td>
<p>Random seed Number to be set. Default: 1. See in <code><a href="base.html#topic+set.seed">set.seed</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>NC_remove returns data frame with removed nodes and corresponding natural connectivity
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi<a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  ### Data preparation ###
  data("Two_group")

  ### One input network analysis ###
  network_results &lt;- network_analysis(
    taxobj = Two_group,
    taxlevel = "Base",
    reads = FALSE,
    n = 10,
    threshold = 0.6
  )

  network_matrix &lt;- as.data.frame(network_results[[3]])  # Complete adjacency matrix

  # Check initial natural connectivity
  nc &lt;- nc(network_matrix)

  # Conduct natural connectivity analysis
  nc_remove &lt;- NC_remove(input = network_matrix)
  head(nc_remove)
  tail(nc_remove)

  # Set target number for natural connectivity analysis
  nc_remove &lt;- NC_remove(input = network_matrix, num = 400)
}
</code></pre>

<hr>
<h2 id='network_analysis'>Conduct Network analysis based on tax summary object</h2><span id='topic+network_analysis'></span>

<h3>Description</h3>

<p>Conduct Network analysis based on tax summary object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_analysis(
  taxobj,
  taxlevel,
  reads = FALSE,
  n,
  threshold,
  rel_threshold = 0,
  method = "spearman",
  display = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_analysis_+3A_taxobj">taxobj</code></td>
<td>
<p>tax summary objects computed by <code><a href="#topic+tax_summary">tax_summary</a></code>.</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_taxlevel">taxlevel</code></td>
<td>
<p>taxonomy levels used for analysis. Must be one of c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;)</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_reads">reads</code></td>
<td>
<p>Logical,default:FALSE. Taxonomy abundance type used in analysis.FALSE for relative abundance, TRUE for absolute abundance.</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_n">n</code></td>
<td>
<p>Numeric. Number of sample size indicating kept asv/otu/gene/taxa appearing. Recommended to set more than half of total sample size.</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_threshold">threshold</code></td>
<td>
<p>Numeric.Threshold of absolute correlation value (r value for pearson method and rho value for spearman method).</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_rel_threshold">rel_threshold</code></td>
<td>
<p>Numeric.Threshold of relative abundance included in the network analysis.Default:0</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_method">method</code></td>
<td>
<p>Character, default: &quot;spearman&quot;. A character indicating which correlation coefficient method to be computed. One of &quot;pearson&quot; or &quot;spearman&quot;</p>
</td></tr>
<tr><td><code id="network_analysis_+3A_display">display</code></td>
<td>
<p>Logical, default:TRUE. If display a preview plot of network based on igraph. FALSE for the first attempt is recommended in case of too many vertices and edges.</p>
</td></tr>
</table>


<h3>Details</h3>


<ol>
<li><p> We had optimized the correlation algorithm to achieve a faster running speed. It takes less than 2 minute to calculate dataframe correlation and p value which more than 400 samples and 10000 OTUs for computer with dual Core i5 processor.
However, too many vertices(&gt;2000) or links(&gt;10000) may slow the statistical process and visualization,so we recommend that in your first attempt,set <code>display</code> paramter as <code>F</code> to have a preview.
Then you can adjust your n/threshold/method paramter to generate a suitable visualization network
</p>
</li>
<li><p> We display a preview plot so as to adjusting your network. Generally a global figure (like we show in examples) with less than 1000 vertices and 5000 edges/links
is recommended. Further more,we recommend you to output the statistics and adjacency table and use software like cytoscape or gephi for better visualization.
</p>
</li></ol>



<h3>Value</h3>

<p>One list contains nodes information table, adjacency column table, adjacency matrix and 'igraph' object.
</p>


<h3>Note</h3>


<ol>
<li><p> Replicates should be at least 5,more than 8 is recommend.
</p>
</li>
<li><p> In case of too many edges/links or not a global network plot, you can stop the process immediately to provent wasting too much time.
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>{
  ### Data preparation ###
  data("Two_group")
  set.seed(999)
  ## Analysis
  network_results &lt;- network_analysis(
    taxobj = Two_group,
    taxlevel = "Genus",
    n = 10,
    threshold = 0.8
  )

  # Nodes information table
  network_nodes &lt;- network_results$Nodes_info
  head(network_nodes)

  # Adjacency table
  network_adjacency &lt;- network_results$Adjacency_column_table
  head(network_adjacency)

  # Complete adjacency matrix
  network_matrix &lt;- network_results$Adjacency_matrix
  print(network_matrix[1:10, 1:10])

  # igraph object
  igraph_object &lt;- network_results$Igraph_object
  network_stat(igraph_object)  # In case you want to see statistics again
  # or do other analysis based on igraph.
}
</code></pre>

<hr>
<h2 id='network_analysis2'>Conduct Network analysis</h2><span id='topic+network_analysis2'></span>

<h3>Description</h3>

<p>A convenient and fast network analysis function, with output results suitable for cytoscape and gephi
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_analysis2(
  input,
  inputtype,
  n,
  threshold,
  method = "spearman",
  display = TRUE,
  input2,
  input2type
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_analysis2_+3A_input">input</code></td>
<td>
<p>Input dataframe with otu/gene/taxa in row and sample ID in column,at least 5 replicates(more than 8 replicates are recommened).</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_inputtype">inputtype</code></td>
<td>
<p>Input dataframe type
</p>
<p>1:dataframe with first column of OTUID and last column of taxonomy
</p>
<p>2:dataframe with first column of OTUID/taxonomy
</p>
<p>3:dataframe of all numeric</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_n">n</code></td>
<td>
<p>Only keep otu/gene/taxa appearing in n sample size</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_threshold">threshold</code></td>
<td>
<p>Threshold of correlation r value</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_method">method</code></td>
<td>
<p>A character string indicating which correlation coefficient  is to be computed. One of &quot;pearson&quot; or &quot;spearman&quot;</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_display">display</code></td>
<td>
<p>If display a preview plot of network based on igraph. FALSE for the first attempt is recommended in case of too many vertices and edges.</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_input2">input2</code></td>
<td>
<p>A second input data frame with otu/gene/taxa in row and sample ID in column. Default:NULL</p>
</td></tr>
<tr><td><code id="network_analysis2_+3A_input2type">input2type</code></td>
<td>
<p>The second input data frame type. Details the same as above. Default:NULL</p>
</td></tr>
</table>


<h3>Details</h3>


<ol>
<li><p> We had optimized the correlation algorithm to achieve a faster running speed. It takes less than 2 minute to calculate dataframe correlation and p value which more than 400 samples and 10000 OTUs for computer with dual Core i5 processor.
However, too many vertices(&gt;2000) or links(&gt;10000) may slow the statistical process and visualization,so we recommend that in your first attempt,set <code>display</code> paramter as <code>F</code> to have a preview.
Then you can adjust your n/threshold/method paramter to generate a suitable visualization network
</p>
</li>
<li><p> We display a preview plot so as to adjusting your network. Generally a global figure (like we show in examples) with less than 1000 vertices and 5000 edges/links
is recommended. Further more,we recommend you to output the statistics and adjacency table and use software like cytoscape or gephi for better visualization.
</p>
</li></ol>



<h3>Value</h3>

<p>One list contains a statistics table of network vertices/nodes and an adjacency table. One preview plot of network in the plot interface and an igraph object(named <code>igraph1</code>) in global environment.
</p>


<h3>Note</h3>


<ol>
<li><p> Replicates should be at least 5,more than 8 is recommend.
</p>
</li>
<li><p> In case of too many edges/links or not a global network plot, you can stop the process immediately to provent wasting too much time.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
{
  ### Data preparation ###
  data(testotu)
  rownames(testotu) &lt;- testotu[, 1]
  inputotu &lt;- testotu[, -c(1, ncol(testotu))]
  head(inputotu)
  set.seed(999)
  ### One input network analysis ###
  network_result &lt;- network_analysis2(
    inputotu,
    3,
    10,
    0.9,
    "spearman",
    TRUE
  )

  # Nodes information table
  network_nodes &lt;- network_result$Nodes_info
  head(network_nodes)

  # Adjacency table
  network_adjacency &lt;- network_result$Adjacency_column_table
  head(network_adjacency)

  # Complete adjacency matrix
  network_matrix &lt;- network_result$Adjacency_matrix
  print(network_matrix[1:10, 1:10])

  # igraph object
  igraph_object &lt;- network_result$Igraph_object
  network_stat(igraph_object)  # In case you want to see statistics again
  # or do other analysis based on igraph.

  ### Two inputs network analysis ###
  inputotu1 &lt;- inputotu[1:456, ]
  inputotu2 &lt;- inputotu[524:975, ]
  network_result &lt;- network_analysis2(
    input = inputotu1,
    inputtype = 3,
    input2 = inputotu2,
    input2type = 3,
    n = 10,
    threshold = 0.85,
    method = "spearman",
    display = TRUE
  )

  #### Incorrect demonstration !! ###
  {
     network_result &lt;- network_analysis2(inputotu, 3, 3, 0.8, "spearman", TRUE)
  }
  # Total edges/links: 10199
  # Total vertices: 826
  # Too many edges and not a global network

}

</code></pre>

<hr>
<h2 id='network_stat'>Igraph network statistics</h2><span id='topic+network_stat'></span>

<h3>Description</h3>

<p>Igraph network statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_stat(input, report = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_stat_+3A_input">input</code></td>
<td>
<p>An igraph object.</p>
</td></tr>
<tr><td><code id="network_stat_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>network statistics
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>

<hr>
<h2 id='network_visual'>Network Visualization</h2><span id='topic+network_visual'></span>

<h3>Description</h3>

<p>Visualizes a network based on a network object from <code><a href="#topic+network_analysis">network_analysis</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_visual(
  network_obj,
  mode = "major_module",
  major_num = 5,
  taxlevel = NULL,
  select_tax = NULL,
  palette = "Set1",
  vertex.size = 6
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_visual_+3A_network_obj">network_obj</code></td>
<td>
<p>A network analysis results object generated from <code><a href="#topic+network_analysis">network_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="network_visual_+3A_mode">mode</code></td>
<td>
<p>The visualization mode, optionally &quot;major_module&quot; or &quot;major_tax&quot;.</p>
</td></tr>
<tr><td><code id="network_visual_+3A_major_num">major_num</code></td>
<td>
<p>The number of major modules to display in the network.</p>
</td></tr>
<tr><td><code id="network_visual_+3A_taxlevel">taxlevel</code></td>
<td>
<p>Taxonomy levels used for visualization when mode is &quot;major_tax&quot;.</p>
</td></tr>
<tr><td><code id="network_visual_+3A_select_tax">select_tax</code></td>
<td>
<p>A vector of taxa to be selected for displaying in &quot;major_tax&quot; mode.</p>
</td></tr>
<tr><td><code id="network_visual_+3A_palette">palette</code></td>
<td>
<p>Character. Palette for visualization.</p>
</td></tr>
<tr><td><code id="network_visual_+3A_vertex.size">vertex.size</code></td>
<td>
<p>Numeric. The size of the vertices.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the configured igraph object and the coordinates of the vertices, with network visualization displayed in the plots panel.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # Data preparation
  data("Two_group")
  set.seed(999)
  # Analysis
  network_results &lt;- network_analysis(
    taxobj = Two_group,
    taxlevel = "Species",
    n = 10,
    threshold = 0.6
  )

  # Default mode
  network_visual_obj &lt;- network_visual(network_obj = network_results)

  # View again
  network_visual_re(network_visual_obj)

  # More modules
  network_visual_obj &lt;- network_visual(
    network_obj = network_results,
    major_num = 10
  )

  # Specific tax
  # Generate top 5 phyla for displaying
  community &lt;- community_plot(
    taxobj = Two_group,
    taxlevel = "Phylum",
    n = 5,
    palette = "Paired"
  )
  display_phyla &lt;- names(community$filled_color)

  network_visual_obj &lt;- network_visual(
    network_obj = network_results,
    mode = "major_tax",
    taxlevel = "Phylum",
    select_tax = display_phyla,
    palette = community$filled_color
  )

  # Another sample for specific tax
  network_visual_obj &lt;- network_visual(
    network_obj = network_results,
    mode = "major_tax",
    taxlevel = "Phylum",
    select_tax = "p__Proteobacteria"
  )
}
</code></pre>

<hr>
<h2 id='network_visual_re'>Re-visualize network plot from <code><a href="#topic+network_visual">network_visual</a></code> or <code><a href="#topic+network_withdiff">network_withdiff</a></code></h2><span id='topic+network_visual_re'></span>

<h3>Description</h3>

<p>Re-visualize network plot from <code><a href="#topic+network_visual">network_visual</a></code> or <code><a href="#topic+network_withdiff">network_withdiff</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_visual_re(
  network_visual_obj,
  module_paint = FALSE,
  module_num = NULL,
  module_palette = c("aquamarine3", "antiquewhite2", "goldenrod2"),
  vertex.size = 6,
  vertex.shape = "circle"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_visual_re_+3A_network_visual_obj">network_visual_obj</code></td>
<td>
<p>Network object from <code><a href="#topic+network_visual">network_visual</a></code> or <code><a href="#topic+network_withdiff">network_withdiff</a></code></p>
</td></tr>
<tr><td><code id="network_visual_re_+3A_module_paint">module_paint</code></td>
<td>
<p>Logical. If network module should be painted. Only work for network object from <code><a href="#topic+network_withdiff">network_withdiff</a></code>.</p>
</td></tr>
<tr><td><code id="network_visual_re_+3A_module_num">module_num</code></td>
<td>
<p>Numeric indicating which module to be painted.</p>
</td></tr>
<tr><td><code id="network_visual_re_+3A_module_palette">module_palette</code></td>
<td>
<p>Character string with at least two elements. Palette for painting modules.</p>
</td></tr>
<tr><td><code id="network_visual_re_+3A_vertex.size">vertex.size</code></td>
<td>
<p>Numeric. The size of the vertices, default:6. Only for network object from <code><a href="#topic+network_visual">network_visual</a></code></p>
</td></tr>
<tr><td><code id="network_visual_re_+3A_vertex.shape">vertex.shape</code></td>
<td>
<p>Character. The shape of vertices, default: &quot;circle&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL but visualization in plot panel.
</p>

<hr>
<h2 id='network_withdiff'>Network Analysis with Differential Species</h2><span id='topic+network_withdiff'></span>

<h3>Description</h3>

<p>Meta network analysis integrating differential taxon into a network analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>network_withdiff(network_obj, diff_frame, aes_col = NULL, tag_threshold = 5)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="network_withdiff_+3A_network_obj">network_obj</code></td>
<td>
<p>Network analysis results generated from <code><a href="#topic+network_analysis">network_analysis</a></code></p>
</td></tr>
<tr><td><code id="network_withdiff_+3A_diff_frame">diff_frame</code></td>
<td>
<p>Differential analysis results generated from <code><a href="#topic+indicator_analysis">indicator_analysis</a></code> or <code><a href="#topic+Deseq_analysis">Deseq_analysis</a></code>.</p>
</td></tr>
<tr><td><code id="network_withdiff_+3A_aes_col">aes_col</code></td>
<td>
<p>A named vector of colors to be used to highlight differential taxon vertices</p>
</td></tr>
<tr><td><code id="network_withdiff_+3A_tag_threshold">tag_threshold</code></td>
<td>
<p>Numeric. A threshold for the minimum number of differential taxon to display.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the configured igraph object, vertices coordinates, parameters, and tag statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
{
  # Data preparation
  data("Two_group")
  set.seed(999)
  # Analysis
  network_results &lt;- network_analysis(
    taxobj = Two_group,
    taxlevel = "Genus",
    n = 10,
    threshold = 0.8
  )
  indicator_results &lt;- indicator_analysis(
    taxobj = Two_group,
    taxlevel = "Genus"
  )
  deseq_results &lt;- Deseq_analysis(
    taxobj = Two_group,
    taxlevel = "Genus",
    cutoff = 1,
    control_name = "Control"
  )

  # Visualize
  network_diff_obj &lt;- network_withdiff(
    network_obj = network_results,
    diff_frame = indicator_results
  )
  # Check contained tags for each model
  print(network_diff_obj$tag_statistics$sum_of_tags)
  # Check contained different tags for each model
  print(network_diff_obj$tag_statistics$detailed_tags)

  # Re-visualize
  network_visual_re(
    network_visual_obj = network_diff_obj,
    module_paint = TRUE,
    module_num = c(1, 4)
  )  # Show module with most Treatment indicators

  my_module_palette &lt;- color_scheme(
    c("#83BA9E", "#F49128"),
    5
  )
  network_visual_re(
    network_visual_obj = network_diff_obj,
    module_paint = TRUE,
    module_num = c(1, 4, 6, 3, 8),
    module_palette = my_module_palette
  )  # Show module with most Treatment indicators

  # Available also for DESeq analysis results
  network_diff_obj &lt;- network_withdiff(
    network_obj = network_results,
    diff_frame = deseq_results
  )

  # Parameter adjustment
  network_diff_obj &lt;- network_withdiff(
    network_obj = network_results,
    diff_frame = indicator_results,
    tag_threshold = 20
  )  # The 'tag_threshold' set too high

  network_diff_obj &lt;- network_withdiff(
    network_obj = network_results,
    diff_frame = indicator_results,
    tag_threshold = 10
  )  # Set lower
  # Check contained tags for each model
  print(network_diff_obj$tag_statistics$sum_of_tags)
  # Check contained different tags for each model
  print(network_diff_obj$tag_statistics$detailed_tags)

  network_diff_obj &lt;- network_withdiff(
    network_obj = network_results,
    diff_frame = indicator_results,
    tag_threshold = 1
  )  # Set too low

  # Another example
  data("Three_group")
  network_results &lt;- network_analysis(
    taxobj = Three_group,
    taxlevel = "Genus",
    n = 15,
    threshold = 0.9
  )
  indicator_results &lt;- indicator_analysis(
    taxobj = Three_group,
    taxlevel = "Genus"
  )

  tag_color &lt;- c(
    "CF" = "#F8766D",
    "CF_OF" = "#FFFF00",
    "OF" = "#00BA38",
    "OF_BF" = "#800080",
    "BF" = "#619CFF",
    "CF_BF" = "#00FFFF"
  )
  network_diff_obj &lt;- network_withdiff(
    network_obj = network_results,
    diff_frame = indicator_results,
    aes_col = tag_color,
    tag_threshold = 10
  )

  # Re-visualize
  print(network_diff_obj$tag_statistics$detailed_tags)
  network_visual_re(
    network_visual_obj = network_diff_obj,
    module_paint = TRUE,
    module_num = c(8, 10, 11)
  )  # Show module with most BF indicators
  network_visual_re(
    network_visual_obj = network_diff_obj,
    module_paint = TRUE,
    module_num = c(1, 6, 8)
  )  # Show module with most BF and OF_BF indicators
}

</code></pre>

<hr>
<h2 id='object_config'>Set taxonomy summary configuration</h2><span id='topic+object_config'></span>

<h3>Description</h3>

<p>This function set taxonomy summary configuration by assigning treatment column number, facet column number, replication column number, treatment mapping color, treatment order and facet order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>object_config(
  taxobj,
  treat_location,
  facet_location = NULL,
  rep_location,
  subject_location = NULL,
  treat_col = NULL,
  treat_order = NULL,
  facet_order = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="object_config_+3A_taxobj">taxobj</code></td>
<td>
<p>Taxonomic summary object generated by <code><a href="#topic+tax_summary">tax_summary</a></code></p>
</td></tr>
<tr><td><code id="object_config_+3A_treat_location">treat_location</code></td>
<td>
<p>Numeric. Treatment column number in metafile/groupinformation.</p>
</td></tr>
<tr><td><code id="object_config_+3A_facet_location">facet_location</code></td>
<td>
<p>Numeric, default:NULL. Facet column number in metafile/groupinformation.</p>
</td></tr>
<tr><td><code id="object_config_+3A_rep_location">rep_location</code></td>
<td>
<p>Numeric. Replication column number in metafile/groupinformation.</p>
</td></tr>
<tr><td><code id="object_config_+3A_subject_location">subject_location</code></td>
<td>
<p>Numeric, default:NULL.Subject column number in metafile/groupinformation (used for pairwise experiment).</p>
</td></tr>
<tr><td><code id="object_config_+3A_treat_col">treat_col</code></td>
<td>
<p>Named character string, default:NULL. A set of aesthetic character to map treatment to.</p>
</td></tr>
<tr><td><code id="object_config_+3A_treat_order">treat_order</code></td>
<td>
<p>Character string, default:NULL. The character string indicating treatment displaying order.</p>
</td></tr>
<tr><td><code id="object_config_+3A_facet_order">facet_order</code></td>
<td>
<p>Character string, default:NULL. The character string indicating facet displaying order.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>object_config returns taxonomy summary object with configuration.
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi<a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  ### Data preparation ###
  data(testotu)
  groupinformation &lt;- data.frame(
    group = c(rep("a", 10), rep("b", 10)),
    factor1 = rnorm(10),
    factor2 = rnorm(mean = 100, 10),
    subject = factor(c(1:10, 1:10)),
    group2 = c(rep("e", 5), rep("f", 5), rep("e", 5), rep("f", 5))
  )

  ### Packaging metafile, community data, and taxonomy table ###
  test_object &lt;- tax_summary(
    groupfile = groupinformation,
    inputtable = testotu[, 2:21],
    reads = TRUE,
    taxonomytable = testotu[, c(1, 22)]
  )

  ### Object configuration ###
  test_object_plan1 &lt;- object_config(
    taxobj = test_object,
    treat_location = 1,
    rep_location = 4
  )

  ### Facet configuration ###
  test_object_plan2 &lt;- object_config(
    taxobj = test_object,
    treat_location = 1,
    rep_location = 4,
    facet_location = 5
  )
}
</code></pre>

<hr>
<h2 id='structure_plot'>Visualize microbial community composition structure based on tax summary object</h2><span id='topic+structure_plot'></span>

<h3>Description</h3>

<p>Function for visualization of microbial structure with PCAplot, PCoAplot and NMDSplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>structure_plot(
  taxobj,
  taxlevel,
  ptsize = 2,
  diagram = NULL,
  ellipse.level = 0.85,
  facet_row = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="structure_plot_+3A_taxobj">taxobj</code></td>
<td>
<p>Configured tax summary objects.See in <code><a href="#topic+object_config">object_config</a></code>.</p>
</td></tr>
<tr><td><code id="structure_plot_+3A_taxlevel">taxlevel</code></td>
<td>
<p>taxonomy levels used for visualization.Must be one of
c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;,&quot;Base&quot;).</p>
</td></tr>
<tr><td><code id="structure_plot_+3A_ptsize">ptsize</code></td>
<td>
<p>Numeric, default: 2. Size of point in plot. See
<code><a href="ggplot2.html#topic+geom_point">geom_point</a></code> for details.</p>
</td></tr>
<tr><td><code id="structure_plot_+3A_diagram">diagram</code></td>
<td>
<p>Character, default: NULL. A character indicating group diagram,
should be in c(&quot;ellipse&quot;, &quot;stick&quot;, &quot;polygon&quot;).</p>
</td></tr>
<tr><td><code id="structure_plot_+3A_ellipse.level">ellipse.level</code></td>
<td>
<p>Numeric, default: 0.85. The level at which to draw an ellipse,
or, if type = &quot;euclid&quot;, the radius of the circle to be drawn. See
<code><a href="ggplot2.html#topic+stat_ellipse">stat_ellipse</a></code> for details.</p>
</td></tr>
<tr><td><code id="structure_plot_+3A_facet_row">facet_row</code></td>
<td>
<p>Numeric, default: NULL. Number of rows when wrap panels. See
<code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code> for details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Microbial structure analysis object.
</p>


<h3>Note</h3>


<ol>
<li><p> Do not use NMDS when warning: In metaMDS(t(inputframe)) :stress is (nearly) zero: you may have insufficient data
</p>
</li>
<li><p> Ellipse not available when replicates less than 3, please use 'stick' or 'polygon' instead
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>###data preparation####
data("Two_group")

###analysis####
set.seed(999)
community_structure&lt;- structure_plot(taxobj = Two_group,taxlevel = "Base")
 #check output list in console (not run)
 ######Output list##
 #####Plot#
 ####PCAplot:named as('PCA_Plot')(1/3)
 ####PCoAplot:named as('PCoA_Plot')(2/3)
 ####NMDSplot:named as('NMDS_Plot')(3/3)
 #####Analysis object#
 ####PCA object:named as('PCA_object')
 ####PCoA object:named as('PCoA_object')
 ####NMDS object:named as ('NMDS_object')
 #####Coordinates dataframe#
 ####PCA Coordinates dataframe:named as('PCA_coordinates')
 ####PCoA Coordinates dataframe:named as('PCoA_coordinates')
 ####NMDS Coordinates dataframe:named as('NMDS_coordinates')
 ######Done##
 #check PERMANOVA results
 community_structure$PERMANOVA_statistics

 #extract plot
 community_structure$PCA_Plot
 community_structure$PCoA_Plot
 community_structure$NMDS_Plot

 #extract object
 PCA_obj&lt;- community_structure$PCA_object
 print(PCA_obj)

 #extract coordinates frame
 PCA_coord&lt;- community_structure$PCA_coordinates
 head(PCA_coord)

 #stick plot
 set.seed(999)
 community_structure&lt;- structure_plot(taxobj = Two_group,taxlevel = "Base",diagram = "stick")
 community_structure$PCoA_Plot

 #faced form
 data("Facet_group")
 set.seed(999)
 community_structure&lt;- structure_plot(taxobj = Facet_group,taxlevel = "Genus",diagram = "stick")
 community_structure$PERMANOVA_statistics
 community_structure$PCA_Plot
 community_structure$PCoA_Plot
 community_structure$NMDS_Plot
</code></pre>

<hr>
<h2 id='sub_tax_summary'>Subsetting tax summary objects</h2><span id='topic+sub_tax_summary'></span>

<h3>Description</h3>

<p>Subsetting tax summary objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sub_tax_summary(taxobj, ..., specificnum = NULL, taxnum = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sub_tax_summary_+3A_taxobj">taxobj</code></td>
<td>
<p>tax summary objects computed by <code><a href="#topic+tax_summary">tax_summary</a></code>.</p>
</td></tr>
<tr><td><code id="sub_tax_summary_+3A_...">...</code></td>
<td>
<p>logical expression that are are defined in terms of the variables in Groupfile of tax summary objects. See details in <code><a href="base.html#topic+subset">subset</a></code>.</p>
</td></tr>
<tr><td><code id="sub_tax_summary_+3A_specificnum">specificnum</code></td>
<td>
<p>specific numbers indicating samples to keep based on Groupfile of tax summary objects.</p>
</td></tr>
<tr><td><code id="sub_tax_summary_+3A_taxnum">taxnum</code></td>
<td>
<p>specific numbers indicating taxonomy to keep based on Base file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Subset of tax summary objects.Same as <code><a href="#topic+tax_summary">tax_summary</a></code>.
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data("Three_group")

  # Check meta file
  print(Three_group$Groupfile)

  # Subsetting tax summary objects

  # Select BF and OF groups
  sub_testtax_summary &lt;- sub_tax_summary(Three_group, Group %in% c("BF", "OF"))
  print(sub_testtax_summary$Groupfile)

  # Subsetting according to taxonomy

  Proteo &lt;- sub_tax_summary(
    Three_group,
    taxnum = which(Three_group$Base_taxonomy$Phylum == "p__Proteobacteria")
  )
  print(Proteo$Phylum_percent)  # Check phylum table
  print(Proteo$Genus_percent)   # Check genus table
</code></pre>

<hr>
<h2 id='t_test_report'>Print Student's t-Test report</h2><span id='topic+t_test_report'></span>

<h3>Description</h3>

<p>Print Student's t-Test report
</p>


<h3>Usage</h3>

<pre><code class='language-R'>t_test_report(
  data,
  treatment_col,
  value_col,
  paired,
  subject_col,
  report = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="t_test_report_+3A_data">data</code></td>
<td>
<p>Data frame containing the treatment, value and other information.</p>
</td></tr>
<tr><td><code id="t_test_report_+3A_treatment_col">treatment_col</code></td>
<td>
<p>Numeric indicating where treatment locates (column number) in data.</p>
</td></tr>
<tr><td><code id="t_test_report_+3A_value_col">value_col</code></td>
<td>
<p>Numeric indicating where treatment value (column number) in data.</p>
</td></tr>
<tr><td><code id="t_test_report_+3A_paired">paired</code></td>
<td>
<p>Logical indicating whether you want a paired t-test.</p>
</td></tr>
<tr><td><code id="t_test_report_+3A_subject_col">subject_col</code></td>
<td>
<p>Only meaningful when Pair is ture. Numeric indicating where subject of treatment (column number) in data.</p>
</td></tr>
<tr><td><code id="t_test_report_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>t_test_report returns list containing:
</p>

<ol>
<li><p> data frame of basic data descrption
</p>
</li>
<li><p> results of student's t-Test
</p>
</li></ol>



<h3>Examples</h3>

<pre><code class='language-R'>{
  ### Data preparation ###
  testdata &lt;- data.frame(
    treatment = c(rep("A", 6), rep("B", 6)),
    subject = rep(c(1:6), 2),
    value = c(rnorm(6, 2), rnorm(6, 1))
  )

  # Perform t-test (unpaired)
  t_test_result &lt;- t_test_report(
    data = testdata,
    treatment_col = 1,
    value_col = 3
  )

  # Perform paired t-test
  t_test_result &lt;- t_test_report(
    data = testdata,
    treatment_col = 1,
    value_col = 3,
    paired = TRUE,
    subject_col = 2
  )

  ### Basic data description ###
  print(t_test_result[[1]])
  print(t_test_result$basicdata)

  ### T-test results ###
  print(t_test_result[[2]])
  print(t_test_result$t.test_results)
}
</code></pre>

<hr>
<h2 id='tax_summary'>Encapsulate meta file, feature tables and taxonomy annotation into tax summary object</h2><span id='topic+tax_summary'></span>

<h3>Description</h3>

<p>The function packages meta file, feature tables and taxonomy annotation into tax summary object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_summary(
  groupfile,
  inputtable,
  reads = TRUE,
  taxonomytable,
  into = "standard",
  sep = ";",
  outputtax = c("Phylum", "Genus")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tax_summary_+3A_groupfile">groupfile</code></td>
<td>
<p>A data frame containing treatment information</p>
</td></tr>
<tr><td><code id="tax_summary_+3A_inputtable">inputtable</code></td>
<td>
<p>OTU/ASV/species data frame with all numeric. Samples ID should be in column names.</p>
</td></tr>
<tr><td><code id="tax_summary_+3A_reads">reads</code></td>
<td>
<p>Logical.True for reads table and FALSE for percentage table. Default: TRUE</p>
</td></tr>
<tr><td><code id="tax_summary_+3A_taxonomytable">taxonomytable</code></td>
<td>
<p>Taxonomy annotation data frame,with first column OTU/ASV/TAX number ID and second column taxonomy annotation. See details in example.</p>
</td></tr>
<tr><td><code id="tax_summary_+3A_into">into</code></td>
<td>
<p>Names of separated taxonomy to create as character vector. Must select from c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;).
Shortcut input:1)By default.&quot;standard&quot;:c(&quot;Domain&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;). Used for standard taxonomy annotation to OTU/ASV table.
2)&quot;complete&quot;:c(&quot;Domain&quot;,&quot;Kingdom&quot;,&quot;Phylum&quot;,&quot;Class&quot;,&quot;Order&quot;,&quot;Family&quot;,&quot;Genus&quot;,&quot;Species&quot;). Used for complete taxonomy annotation to meta genomic table.</p>
</td></tr>
<tr><td><code id="tax_summary_+3A_sep">sep</code></td>
<td>
<p>Separator of taxonomy table.Default: &quot;;&quot;.</p>
</td></tr>
<tr><td><code id="tax_summary_+3A_outputtax">outputtax</code></td>
<td>
<p>Names of output taxonomy level table. Default:c(&quot;Phylum&quot;,&quot;Genus&quot;). Shortcut input is available with 'standard' and 'complete' same as above.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One list containing taxonomy table data frame,containing reads and percentage table for each specified output. Full taxonomy annotation data frame is output in global environment.
</p>


<h3>Note</h3>

<p>For taxonomy annotation with 'Kingdom' level, please set 'into' parameter as 'complete'!!!
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # Load data
  data(testotu)

  # Create group information data frame
  groupinformation &lt;- data.frame(
    group = c(rep("a", 10), rep("b", 10)),
    factor1 = rnorm(10),
    factor2 = rnorm(mean = 100, 10),
    subject = factor(c(1:10, 1:10))
  )

  # Packaging data into a taxonomy summary object
  test_object &lt;- tax_summary(
    groupfile = groupinformation,
    inputtable = testotu[, 2:21],
    reads = TRUE,
    taxonomytable = testotu[, c(1, 22)]
  )

  # Check integrated object
  print(test_object)

  # Extract genus relative abundance table
  test_Genus &lt;- test_object$Genus_percent
  head(test_Genus)

  # Check corresponding taxonomy information of genus table
  test_Genus_tax &lt;- test_object$Genus_taxonomy
  head(test_Genus_tax)

  # Summary base table into all taxonomy levels with standard output
  test_object &lt;- tax_summary(
    groupfile = groupinformation,
    inputtable = testotu[, 2:21],
    reads = TRUE,
    taxonomytable = testotu[, c(1, 22)],
    outputtax = "standard"
  )
  head(test_object$Species_percent)  # View first 10 rows of species percentage
  head(test_object$Genus)  # View first 10 rows of genus table
}
</code></pre>

<hr>
<h2 id='tbRDA_analysis'>tbRDA analysis</h2><span id='topic+tbRDA_analysis'></span>

<h3>Description</h3>

<p>RDA analysis including co-linearity diagnostics and necessary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tbRDA_analysis(otudata, envdata, collinearity, perm.test = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tbRDA_analysis_+3A_otudata">otudata</code></td>
<td>
<p>Feature table of all numeric variable, with annotation in row names</p>
</td></tr>
<tr><td><code id="tbRDA_analysis_+3A_envdata">envdata</code></td>
<td>
<p>Environmental factor of all numeric variable,with sample-ID in row names and environmental factor in column names</p>
</td></tr>
<tr><td><code id="tbRDA_analysis_+3A_collinearity">collinearity</code></td>
<td>
<p>If done collinearity diagnostics. Default,TRUE.</p>
</td></tr>
<tr><td><code id="tbRDA_analysis_+3A_perm.test">perm.test</code></td>
<td>
<p>Logical. If conduct permutation test. Default:TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Three permutation test result print ,one preview plot ,a RDA object(default name:otu.tab.1) and a summary of RDA object
</p>


<h3>Note</h3>


<ol>
<li><p> When Axis length in first axis more than 4, you should choose CCA instead of RDA.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  ### Data preparation ###
  library(vegan)
  data(varechem)
  head(varechem)
  data(testotu)
  require(tidyr); require(magrittr)  ## Or use pipe command in "dplyr"

  sep_testotu &lt;- Filter_function(
    input = testotu,
    threshold = 0.0001,
    format = 1
  ) %&gt;%
  separate(
    ., col = taxonomy,
    into = c("Domain", "Phylum", "Order", "Family", "Class", "Genus", "Species"),
    sep = ";"
  )

  top10phylum &lt;- aggregate(
    sep_testotu[, 2:21],
    by = list(sep_testotu$Phylum),
    FUN = sum
  ) %&gt;%
  Top_taxa(
    input = .,
    n = 10,
    inputformat = 2,
    outformat = 1
  )
  rownames(top10phylum) &lt;- top10phylum[, 1]
  top10phylum &lt;- top10phylum[, -1]

  group &lt;- data.frame(
    group = c(rep("a", 10), rep("b", 10)),
    factor1 = rnorm(10),
    factor2 = rnorm(mean = 100, 10)
  )

  ### RDA analysis ###
  set.seed(999)
  RDAresult &lt;- tbRDA_analysis(
    top10phylum,
    varechem[1:20, ],
    TRUE
  )

  # Environmental statistics
  print(RDAresult$factor_statistics)

  # Visualization using ggplot
  rda_object &lt;- RDAresult$rda_object
  rda_summary &lt;- RDAresult$rdasummary
  rda_scores &lt;- RDAresult$rdascores
  rda_env &lt;- as.data.frame(rda_scores$biplot)
  rda_sample &lt;- as.data.frame(rda_scores$sites)
  rda_otu &lt;- as.data.frame(rda_scores$species)

  xlab &lt;- paste0("RDA1:", round(RDAresult$rdasummary$concont$importance[2, 1], 4) * 100, "%")
  ylab &lt;- paste0("RDA2:", round(RDAresult$rdasummary$concont$importance[2, 2], 4) * 100, "%")

  library(ggrepel)
  # Create a sample RDA plot
  RDAplot &lt;- ggplot(data = rda_sample, aes(RDA1, RDA2)) +
    geom_point(aes(color = group$group), size = 2) +
    geom_point(data = rda_otu, pch = "+", color = "orange", size = 4) +
    geom_hline(yintercept = 0) +
    geom_vline(xintercept = 0) +
    geom_segment(data = rda_env, aes(x = 0, y = 0, xend = RDA1 * 0.8, yend = RDA2 * 0.8),
                 arrow = arrow(angle = 22.5, length = unit(0.35, "cm")),
                 linetype = 1, size = 0.6, colour = "red") +
    geom_text_repel(color = "red", data = rda_env,
                    aes(RDA1, RDA2, label = row.names(rda_env))) +
    labs(x = xlab, y = ylab, color = "Treatment",
         title = paste0("p = ", anova.cca(rda_object)["Model", "Pr(&gt;F)"])) +
    stat_ellipse(aes(color = group$group), level = 0.95) +
    geom_text_repel(size = 3, color = "orange",
                    data = subset(rda_otu, RDA1 &gt; 0.1 | RDA1 &lt; (-0.1)),
                    aes(RDA1, RDA2, label = rownames(subset(rda_otu, RDA1&gt;0.1|RDA1&lt;(-0.1))))) +
    theme_zg()

  # Print the RDA plot
  print(RDAplot)
</code></pre>

<hr>
<h2 id='testotu'>test otudata</h2><span id='topic+testotu'></span>

<h3>Description</h3>

<p>A dataset containing 20 samples and 1000 OTUs from soil to test,taxonomy information
is covered randomly(not actual)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>testotu
</code></pre>


<h3>Format</h3>

<p>A data frame with 1000 rows and 22 variables.
</p>

<hr>
<h2 id='theme_zg'>A classic theme for ggplot</h2><span id='topic+theme_zg'></span>

<h3>Description</h3>

<p>A classic theme for ggplot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theme_zg()
</code></pre>


<h3>Value</h3>

<p>ggplot theme
</p>


<h3>Note</h3>

<p>Build inside the LorMe package, Please use theme_zg() as a theme directly
</p>

<hr>
<h2 id='Three_group'>Tax summary object with three groups</h2><span id='topic+Three_group'></span>

<h3>Description</h3>

<p>Enraptured summary object with three groups.Configuration has been assigned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Three_group
</code></pre>


<h3>Format</h3>

<p>Tax summary object with configuration
</p>

<hr>
<h2 id='Top_taxa'>Calculate top taxa and others</h2><span id='topic+Top_taxa'></span>

<h3>Description</h3>

<p>Top taxa is widely used in data analysis,here we provide a simple function to calculate which simplify your R script.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Top_taxa(input, n, inputformat, outformat)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Top_taxa_+3A_input">input</code></td>
<td>
<p>Reads or relative abundance(recommended) of OTU/Taxa/gene data frame,see details in inputformat</p>
</td></tr>
<tr><td><code id="Top_taxa_+3A_n">n</code></td>
<td>
<p>Top n taxa remained according to relative abundance</p>
</td></tr>
<tr><td><code id="Top_taxa_+3A_inputformat">inputformat</code></td>
<td>
<p>1:data frame with first column of OTUID and last column of taxonomy
</p>
<p>2:data frame with first column of OTUID/taxonomy (recommended!!!)
</p>
<p>3:data frame of all numeric,with row names of OTUID/taxonomy</p>
</td></tr>
<tr><td><code id="Top_taxa_+3A_outformat">outformat</code></td>
<td>

<ol>
<li><p> return outformat the same as inputformat
</p>
</li>
<li><p> return data frame of all numeric with OTU/gene/taxa ID in row names(not available for inputformat 1).
</p>
</li></ol>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with top n taxa
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi<a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Data preparation ####
data(testotu)
require(tidyr); require(magrittr)  ## Or use pipe command in "dplyr"

testotu.pct &lt;- data.frame(
  OTU.ID = testotu[, 1],
  sweep(testotu[, -c(1, 22)], 2, colSums(testotu[, -c(1, 22)]), "/"),
  taxonomy = testotu[, 22]
)

sep_testotu &lt;- Filter_function(
  input = testotu,
  threshold = 0.0001,
  format = 1
) %&gt;%
  separate(
    ., col = taxonomy,
    into = c("Domain", "Phylum", "Order", "Family", "Class", "Genus", "Species"),
    sep = ";"
  )

phylum &lt;- aggregate(
  sep_testotu[, 2:21], by = list(sep_testotu$Phylum), FUN = sum
)

phylum1 &lt;- data.frame(row.names = phylum[, 1], phylum[, -1])

##### Input format 1, top 100 OTU #####
top100otu &lt;- Top_taxa(
  input = testotu.pct,
  n = 100,
  inputformat = 1,
  outformat = 1
)

##### Input format 2, top 15 phylum #####
head(phylum)
top15phylum &lt;- Top_taxa(
  input = phylum,
  n = 15,
  inputformat = 2,
  outformat = 1
)

##### Input format 3, top 15 phylum #####
head(phylum1)
top15phylum &lt;- Top_taxa(
  input = phylum1,
  n = 15,
  inputformat = 3,
  outformat = 1
)
</code></pre>

<hr>
<h2 id='Two_group'>Tax summary object with two groups</h2><span id='topic+Two_group'></span>

<h3>Description</h3>

<p>Enraptured summary object with two groups.Configuration has been assigned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Two_group
</code></pre>


<h3>Format</h3>

<p>Tax summary object with configuration
</p>

<hr>
<h2 id='volcano_plot'>Generate Volcano plot base on Deseq_analysis or indicator_analysis results</h2><span id='topic+volcano_plot'></span>

<h3>Description</h3>

<p>Generate Volcano plot base on Deseq_analysis or indicator_analysis results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcano_plot(inputframe, cutoff = NULL, aes_col = c("#FE5C5C", "#75ABDE"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="volcano_plot_+3A_inputframe">inputframe</code></td>
<td>
<p>A data frame containing the results based on <code><a href="#topic+Deseq_analysis">Deseq_analysis</a></code> or <code><a href="#topic+indicator_analysis">indicator_analysis</a></code> (only two group indicators)</p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_cutoff">cutoff</code></td>
<td>
<p>A numeric value specifying the fold change cutoff,should be the same as in <code><a href="#topic+Deseq_analysis">Deseq_analysis</a></code></p>
</td></tr>
<tr><td><code id="volcano_plot_+3A_aes_col">aes_col</code></td>
<td>
<p>A named vector of colors to be used in the plots</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two ggplot objects, one for the fold change versus adjusted p-value plot and
another for the mean abundance versus fold change or enrichment factor plot.
</p>


<h3>Author(s)</h3>

<p>Wang Ningqi <a href="mailto:2434066068@qq.com">2434066068@qq.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>###data prepration###

{
  # Load data
  data("Two_group")

  # Define color based on treatment column
  mycolor &lt;- Two_group$configuration$treat_col

  ### DESeq analysis ###
  deseq_results &lt;- Deseq_analysis(
    taxobj = Two_group,
    taxlevel = "Genus",
    cutoff = 1,
    control_name = "Control"
  )

  ### Or indicator analysis ###
  indicator_results &lt;- indicator_analysis(
    taxobj = Two_group,
    taxlevel = "Genus"
  )

  # Create volcano plot for DESeq results
  volcano_plot &lt;- volcano_plot(
    inputframe = deseq_results,
    cutoff = 1,
    aes_col = mycolor
  )
  print(volcano_plot$FC_FDR)  # Fold Change and FDR values
  print(volcano_plot$Mean_FC)  # Mean Fold Change values

  # Create volcano plot for indicator results
  volcano_plot &lt;- volcano_plot(
    inputframe = indicator_results,
    cutoff = 1,
    aes_col = mycolor
  )
  print(volcano_plot$FC_FDR)  # Fold Change and FDR values
  print(volcano_plot$Mean_FC)  # Mean Fold Change values
}

</code></pre>

<hr>
<h2 id='wilcox_test_report'>Print Wilcoxon Rank Sum and Signed Rank Tests reprot</h2><span id='topic+wilcox_test_report'></span>

<h3>Description</h3>

<p>Print Wilcoxon Rank Sum and Signed Rank Tests reprot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wilcox_test_report(
  data,
  treatment_col,
  value_col,
  paired = FALSE,
  subject_col = NULL,
  report = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wilcox_test_report_+3A_data">data</code></td>
<td>
<p>Data frame containing the treatment, value and other information.</p>
</td></tr>
<tr><td><code id="wilcox_test_report_+3A_treatment_col">treatment_col</code></td>
<td>
<p>Numeric indicating where treatment locates (column number) in data.</p>
</td></tr>
<tr><td><code id="wilcox_test_report_+3A_value_col">value_col</code></td>
<td>
<p>Numeric indicating where treatment value (column number) in data.</p>
</td></tr>
<tr><td><code id="wilcox_test_report_+3A_paired">paired</code></td>
<td>
<p>Logical indicating whether you want a paired test.</p>
</td></tr>
<tr><td><code id="wilcox_test_report_+3A_subject_col">subject_col</code></td>
<td>
<p>Only meaningful when Pair is ture. Numeric indicating where subject of treatment (column number) in data.</p>
</td></tr>
<tr><td><code id="wilcox_test_report_+3A_report">report</code></td>
<td>
<p>Logical. If print report to console. Default:TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>wilcox_test_report returns data frame of basic data description.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
  # Data preparation
  testdata &lt;- data.frame(
    treatment = c(rep("A", 6), rep("B", 6)),
    subject = rep(1:6, 2),
    value = c(rnorm(6, 2), rnorm(6, 1))
  )

  # Wilcoxon test (unpaired)
  wilcox_result &lt;- wilcox_test_report(
    data = testdata,
    treatment_col = 1,
    value_col = 3
  )

  # Wilcoxon signed rank test (paired)
  wilcox_result &lt;- wilcox_test_report(
    data = testdata,
    treatment_col = 1,
    value_col = 3,
    paired = TRUE,
    subject_col = 2
  )

  ### Basic data description ###
  print(wilcox_result)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
