<!DOCTYPE html><html lang="en"><head><title>Help for package lpda</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {lpda}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bestPC'><p>Choosing the best number of Principal Components (PCs) for lpda-pca model.</p></a></li>
<li><a href='#bestVariability'><p>Choosing the best explained variability for lpda-pca model.</p></a></li>
<li><a href='#CVktest'><p>CVktest evaluates the error rate classification with crossvalidation</p></a></li>
<li><a href='#CVloo'><p>CVloo evaluates the error rate classification with leave one out procedure</p></a></li>
<li><a href='#lpda'><p>Computing discriminating hyperplane for two groups</p></a></li>
<li><a href='#lpda.fit'><p>lpda.fit computes the discriminating hyperplane for two groups</p></a></li>
<li><a href='#lpda.pca'><p>lpda.pca computes a PCA to the original data and selects the desired PCs when Variability is supplied</p></a></li>
<li><a href='#lpdaCV'><p>lpdaCV evaluates the error rate classification with a crossvalidation procedure</p></a></li>
<li><a href='#palmdates'><p>Spectrometry and composition chemical of Spanish and Arabian palm dates</p></a></li>
<li><a href='#PCA'><p>Principal Component Analysis</p></a></li>
<li><a href='#plot.lpda'><p>Plot method for lpda classification</p></a></li>
<li><a href='#predict.lpda'><p>Predict method for lpda classification</p></a></li>
<li><a href='#RNAseq'><p>Simulated RNA-Seq dataset example</p></a></li>
<li><a href='#stand'><p>stand center and scale a data matrix</p></a></li>
<li><a href='#stand2'><p>stand2 center and scale a data matrix with the parameters of another one</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Linear Programming Discriminant Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Maria Jose Nueda &lt;mj.nueda@ua.es&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Maria Jose Nueda &lt;mj.nueda@ua.es&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Classification method obtained through linear programming.
    It is advantageous with respect to the classical developments when the distribution of the variables
    involved is unknown or when the number of variables is much greater than the number of individuals.
    LPDA method is published in Nueda, et al. (2022) "LPDA: A new classification method based on linear programming".
    &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0270403">doi:10.1371/journal.pone.0270403</a>&gt;.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rglpk</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-03-06 13:33:14 UTC; USUARIO</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-03-07 07:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bestPC'>Choosing the best number of Principal Components (PCs) for lpda-pca model. </h2><span id='topic+bestPC'></span>

<h3>Description</h3>

<p><code>bestPC</code> computes the classification error for lpda.pca models applied with the number of components specified in PCs argument. The result is the average classification error rate from the R models computed for each number of PCs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bestPC(data, group, ntest = 10, R = 10, PCs = c(10,15,20), f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bestPC_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="bestPC_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="bestPC_+3A_ntest">ntest</code></td>
<td>
<p>Number of samples to evaluate in the test-set.</p>
</td></tr>
<tr><td><code id="bestPC_+3A_r">R</code></td>
<td>
<p>Times the model is evaluated with each Variability indicated in Vars vector.</p>
</td></tr>
<tr><td><code id="bestPC_+3A_pcs">PCs</code></td>
<td>
<p>The PCs to check.</p>
</td></tr>
<tr><td><code id="bestPC_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group. If NULL they are equally weighted.</p>
</td></tr>
<tr><td><code id="bestPC_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group. If NULL they are equally weighted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>bestPC</code> returns a vector with the average prediction error rate obtained from the R models for each PC specified in PCs input.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(RNAseq)
  group = as.factor(rep(c("G1","G2"), each = 30))
  bestPC(RNAseq, group, ntest = 10, R = 5, PCs = c(2, 10))
  </code></pre>

<hr>
<h2 id='bestVariability'>Choosing the best explained variability for lpda-pca model. </h2><span id='topic+bestVariability'></span>

<h3>Description</h3>

<p><code>bestVariability</code> computes the classification error for lpda.pca models obtained with the number of components needed to reach the  explained variability specified in 'Vars' argument. The result is the average classification error rate from the R models computed for each explained variability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bestVariability(data, group, ntest = 10, R = 10, Vars = c(0.5,0.7), f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bestVariability_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="bestVariability_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="bestVariability_+3A_ntest">ntest</code></td>
<td>
<p>Number of samples to evaluate in the test-set.</p>
</td></tr>
<tr><td><code id="bestVariability_+3A_r">R</code></td>
<td>
<p>Times the model is evaluated with each Variability indicated in Vars vector.</p>
</td></tr>
<tr><td><code id="bestVariability_+3A_vars">Vars</code></td>
<td>
<p>The different variabilities to check from which the best variability parameter will be chosen for lpdapca model.</p>
</td></tr>
<tr><td><code id="bestVariability_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group. If NULL they are equally weighted.</p>
</td></tr>
<tr><td><code id="bestVariability_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group. If NULL they are equally weighted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>bestVar</code> returns a vector with the average prediction error rate obtained from the R models for each variability specified in Vars input.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(RNAseq)
  group = as.factor(rep(c("G1","G2"),each=30))
  bestVariability(RNAseq, group, ntest = 10, R = 5, Vars = c(0.1,0.9))
  </code></pre>

<hr>
<h2 id='CVktest'>CVktest evaluates the error rate classification with crossvalidation</h2><span id='topic+CVktest'></span>

<h3>Description</h3>

<p><code>CVktest</code> evaluates the error rate classification in k samples that do not participate in the model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVktest(data, group, scale = FALSE, pca = FALSE, PC = 2,
                    Variability = NULL, ntest = 10, R = 10, f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CVktest_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="CVktest_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="CVktest_+3A_scale">scale</code></td>
<td>
<p>Logical indicating if data is standarised.</p>
</td></tr>
<tr><td><code id="CVktest_+3A_pca">pca</code></td>
<td>
<p>Logical indicating if a reduction of dimension is required</p>
</td></tr>
<tr><td><code id="CVktest_+3A_pc">PC</code></td>
<td>
<p>Number of Principal Components (PC) for PCA. By default it is 2. When the number of PC is not decided, it can be determined choosing the desired proportion of explained variability (Variability parameter) or choosing the maximum number of errors allowed in the training set (Error.max).</p>
</td></tr>
<tr><td><code id="CVktest_+3A_variability">Variability</code></td>
<td>
<p>Parameter for Principal Components (PC) selection. This is the desired
proportion of variability explained for the PC of the variables.</p>
</td></tr>
<tr><td><code id="CVktest_+3A_ntest">ntest</code></td>
<td>
<p>Number of samples to evaluate in the test-set.</p>
</td></tr>
<tr><td><code id="CVktest_+3A_r">R</code></td>
<td>
<p>Number of times that the error is evaluated.</p>
</td></tr>
<tr><td><code id="CVktest_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group. If NULL they are equally weighted.</p>
</td></tr>
<tr><td><code id="CVktest_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group. If NULL they are equally weighted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>lpdaktest</code> The prediction error rate.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpdaCV">lpdaCV</a></code></p>

<hr>
<h2 id='CVloo'>CVloo evaluates the error rate classification with leave one out procedure</h2><span id='topic+CVloo'></span>

<h3>Description</h3>

<p><code>CVloo</code> evaluates the error rate classification with leave one out procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CVloo(data, group, scale = FALSE, pca = FALSE, PC = 2,
                  Variability = NULL, f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CVloo_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns.</p>
</td></tr>
<tr><td><code id="CVloo_+3A_group">group</code></td>
<td>
<p>Vector with the variable group.</p>
</td></tr>
<tr><td><code id="CVloo_+3A_scale">scale</code></td>
<td>
<p>Logical indicating if data is standarised.</p>
</td></tr>
<tr><td><code id="CVloo_+3A_pca">pca</code></td>
<td>
<p>Logical indicating if a reduction of dimension is required.</p>
</td></tr>
<tr><td><code id="CVloo_+3A_pc">PC</code></td>
<td>
<p>Number of Principal Components (PC) for PCA. By default it is 2. When the number of PC is not decided, it can be determined choosing the desired proportion of explained variability (Variability parameter) or choosing the maximum number of errors allowed in the training set (Error.max).</p>
</td></tr>
<tr><td><code id="CVloo_+3A_variability">Variability</code></td>
<td>
<p>Parameter for Principal Components (PC) selection. This is the desired
proportion of variability explained for the PC of the variables.</p>
</td></tr>
<tr><td><code id="CVloo_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group. If NULL they are equally weighted.</p>
</td></tr>
<tr><td><code id="CVloo_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group. If NULL they are equally weighted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>CVloo</code> The prediction error rate.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpdaCV">lpdaCV</a></code> </p>

<hr>
<h2 id='lpda'>Computing discriminating hyperplane for two groups</h2><span id='topic+lpda'></span>

<h3>Description</h3>

<p>This function computes a discriminating hyperplane for two groups with original data (calling lpda.fit) or with principal components (calling lpda.pca)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lpda(data, group, scale = FALSE, pca = FALSE, PC = 2, Variability = NULL,
                 f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lpda_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="lpda_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="lpda_+3A_scale">scale</code></td>
<td>
<p>Logical indicating if data is standarised. When pca=TRUE data is always scaled.</p>
</td></tr>
<tr><td><code id="lpda_+3A_pca">pca</code></td>
<td>
<p>Logical indicating if Principal Components Analysis is required</p>
</td></tr>
<tr><td><code id="lpda_+3A_pc">PC</code></td>
<td>
<p>Number of Principal Components (PC) for PCA. By default it is 2. When the number of PC is not decided, it can be determined choosing the desired proportion of explained variability (Variability parameter).</p>
</td></tr>
<tr><td><code id="lpda_+3A_variability">Variability</code></td>
<td>
<p>Parameter for Principal Components (PC) selection. This is the minimum desired proportion of variability explained for the PC of the variables. The analysis is always done with a minimum of 2 PCs. If it is NULL the PCA will be computed with PC parameter.</p>
</td></tr>
<tr><td><code id="lpda_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group. If NULL they are equally weighted.</p>
</td></tr>
<tr><td><code id="lpda_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group. If NULL they are equally weighted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>lpda</code> returns an object of class &quot;<code>lpda</code>&quot;.
</p>
<p>The functions<code>predict</code> and <code>plot</code> can be used to obtain the predicted classes and a plot in two dimensions with the distances to the computed hyperplane for the two classes.
</p>
<table role = "presentation">
<tr><td><code>coef</code></td>
<td>
<p>Hyperplane coefficients</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>Input data matrix when pca = FALSE and scores when pca = TRUE</p>
</td></tr>
<tr><td><code>group</code></td>
<td>
<p>Input group vector</p>
</td></tr>
<tr><td><code>scale</code></td>
<td>
<p>Input scale argument</p>
</td></tr>
<tr><td><code>pca</code></td>
<td>
<p>Input pca argument</p>
</td></tr>
<tr><td><code>loadings</code></td>
<td>
<p>Principal Components loadings. Showed when pca = TRUE</p>
</td></tr>
<tr><td><code>scores</code></td>
<td>
<p>Principal Components scores. Showed when pca = TRUE</p>
</td></tr>
<tr><td><code>var.exp</code></td>
<td>
<p>A matrix containing the explained variance for each component and the cumulative variance. Showed when pca = TRUE</p>
</td></tr>
<tr><td><code>PCs</code></td>
<td>
<p>Number of Principal Components in the analysis. Showed when pca = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>References</h3>

<p>Nueda MJ, Gandía C, Molina MD (2022) LPDA: A new classification method based on linear programming. PLoS ONE 17(7): e0270403. &lt;https://doi.org/10.1371/journal.pone.0270403&gt;
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda.pca">lpda.pca</a></code>, <code><a href="#topic+lpda.fit">lpda.fit</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
######### palmdates example in lpda package:
data(palmdates)
group = as.factor( c(rep("Spanish",11),rep("Foreign",10)) )

# with concentration data:
model = lpda(data = palmdates$conc, group = group )
pred = predict(model)
table(pred$fitted, group)
plot(model, main = "Palmdates example")

model.pca = lpda(data = palmdates$conc, group = group, pca=TRUE, PC = 2)
plot(model.pca, PCscores = TRUE, main = "Palmdates example")

# with spectra data
model.pca = lpda(data = palmdates$spectra, group = group, pca=TRUE, Variability = 0.9)
model.pca$PCs # 4 PCs to explain 90% of the variability
plot(model.pca, PCscores = TRUE, main = "Spectra palmdates")

</code></pre>

<hr>
<h2 id='lpda.fit'>lpda.fit computes the discriminating hyperplane for two groups</h2><span id='topic+lpda.fit'></span>

<h3>Description</h3>

<p><code>lpda.fit</code> computes the discriminating hyperplane for two groups, giving as a result the coefficients of the hyperplane.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lpda.fit(data, group, f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lpda.fit_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="lpda.fit_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="lpda.fit_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group</p>
</td></tr>
<tr><td><code id="lpda.fit_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>coef</code></td>
<td>
<p>Hyperplane coefficients</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>References</h3>

<p>Nueda MJ, Gandía C, Molina MD (2022) LPDA: A new classification method based on linear programming. PLoS ONE 17(7): e0270403. &lt;https://doi.org/10.1371/journal.pone.0270403&gt;
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>

<hr>
<h2 id='lpda.pca'>lpda.pca computes a PCA to the original data and selects the desired PCs when Variability is supplied</h2><span id='topic+lpda.pca'></span>

<h3>Description</h3>

<p><code>lpda.pca</code> computes the discriminating hyperplane for two groups with Principal Components (PC)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lpda.pca(data, group, PC = 2, Variability = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lpda.pca_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="lpda.pca_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="lpda.pca_+3A_pc">PC</code></td>
<td>
<p>Number of Principal Components (PC) for PCA. By default it is 2. When the number of PC is not decided, it can be determined choosing the desired proportion of explained variability (Variability parameter).</p>
</td></tr>
<tr><td><code id="lpda.pca_+3A_variability">Variability</code></td>
<td>
<p>Parameter for Principal Components (PC) selection. This is the minimum desired proportion of variability explained for the PC of the variables. The analysis is always done with a minimum of 2 PCs. If it is NULL the PCA will be computed with PC parameter.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>loadings</code></td>
<td>
<p>Principal Components loadings.</p>
</td></tr>
<tr><td><code>scores</code></td>
<td>
<p>Principal Components scores.</p>
</td></tr>
<tr><td><code>var.exp</code></td>
<td>
<p>A matrix containing the explained variance for each component and the cumulative variance.</p>
</td></tr>
<tr><td><code>PCs</code></td>
<td>
<p>Number of Principal Components in the analysis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>References</h3>

<p>Nueda MJ, Gandía C, Molina MD (2022) LPDA: A new classification method based on linear programming. PLoS ONE 17(7): e0270403. &lt;https://doi.org/10.1371/journal.pone.0270403&gt;
</p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>

<hr>
<h2 id='lpdaCV'>lpdaCV evaluates the error rate classification with a crossvalidation procedure</h2><span id='topic+lpdaCV'></span>

<h3>Description</h3>

<p>lpdaCV evaluates the error rate classification with a crossvalidation procedure
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lpdaCV(data, group, scale = FALSE, pca = FALSE, PC = 2, Variability = NULL,
                   CV = "loo", ntest = 10, R = 10, f1 = NULL, f2 = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="lpdaCV_+3A_data">data</code></td>
<td>
<p>Matrix containing data. Individuals in rows and variables in columns</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_group">group</code></td>
<td>
<p>Vector with the variable group</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_scale">scale</code></td>
<td>
<p>Logical indicating if data is standarised.</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_pca">pca</code></td>
<td>
<p>Logical indicating if a reduction of dimension is required</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_pc">PC</code></td>
<td>
<p>Number of Principal Components (PC) for PCA. By default it is 2. When the number of PC is not decided, it can be determined choosing the desired proportion of explained variability (Variability parameter) or choosing the maximum number of errors allowed in the training set (Error.max).</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_variability">Variability</code></td>
<td>
<p>Parameter for Principal Components (PC) selection. This is the desired
proportion of variability explained for the PC of the variables.</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_cv">CV</code></td>
<td>
<p>Crossvalidation mode: loo &quot;leave one out&quot; or ktest: that leaves k in the test set.</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_ntest">ntest</code></td>
<td>
<p>Number of samples to evaluate in the test-set.</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_r">R</code></td>
<td>
<p>Number of times that the error is evaluated.</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_f1">f1</code></td>
<td>
<p>Vector with weights for individuals of the first group. If NULL they are equally weighted.</p>
</td></tr>
<tr><td><code id="lpdaCV_+3A_f2">f2</code></td>
<td>
<p>Vector with weights for individuals of the second group. If NULL they are equally weighted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>lpdaCV</code> The prediction error rate.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>
  data(RNAseq)
  group = as.factor(rep(c("G1","G2"), each = 30))
  lpdaCV(RNAseq, group, pca = TRUE, CV = "ktest", ntest = 2)

  </code></pre>

<hr>
<h2 id='palmdates'>Spectrometry and composition chemical of Spanish and Arabian palm dates</h2><span id='topic+palmdates'></span>

<h3>Description</h3>

<p>A data set with scores of 21 dates on spectrometry and concentration measurements of the substances that better define the quality of the dates: fibre, sorbitol, fructose, glucose and myo-inositol. The first 11 dates are Spanish (from Elche, Alicante) and the last 10 are from other countries, mainly Arabian.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>palmdates</code></pre>


<h3>Format</h3>

<p>A data frame with 2 elements:
</p>

<dl>
<dt>conc</dt><dd><p>a data frame with 5 columns: fibre, sorbitol, fructose, glucose and myo-inositol.</p>
</dd>
<dt>spectra</dt><dd><p>a data frame with 2050 columns.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>References</h3>

<p>Abdrabo, S.S., Gras, L., Grindlay, G. and Mora, J. (2021) Evaluation of Fourier Transform-Raman Spectroscopy for palm dates characterization. Journal of food composition and analysis. Submitted.
</p>

<hr>
<h2 id='PCA'>Principal Component Analysis
</h2><span id='topic+PCA'></span>

<h3>Description</h3>

<p>Computes a Principal Component Analysis when both when p&gt;n and when p&lt;=n.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PCA(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PCA_+3A_x">X</code></td>
<td>
<p>Matrix or data.frame with variables in columns and observations in rows.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>eigen</code></td>
<td>
<p>A eigen class object with eigenvalues and eigenvectors of the analysis.</p>
</td></tr>
<tr><td><code>var.exp</code></td>
<td>
<p>A matrix containing the explained variance for each component and the cumulative variance.</p>
</td></tr>
<tr><td><code>scores</code></td>
<td>
<p>Scores of the PCA analysis.</p>
</td></tr>
<tr><td><code>loadings</code></td>
<td>
<p>Loadings of the PCA analysis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Simulate data matrix with 500 variables and 10 observations
datasim = matrix(sample(0:100, 5000, replace = TRUE), nrow = 10)
## PCA
myPCA = PCA(datasim)
## Extracting the variance explained by each principal component
myPCA$var.exp
</code></pre>

<hr>
<h2 id='plot.lpda'>Plot method for lpda classification</h2><span id='topic+plot.lpda'></span>

<h3>Description</h3>

<p><code>plot.lpda</code> is applied to an <code>lpda</code> class object. It shows a plot
in two dimensions with the distances to the computed hyperplane of each individual coloring each case with the real class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lpda'
plot(x, PCscores = FALSE, xlim = NULL, main = NULL,
legend.pos = "topright", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.lpda_+3A_x">x</code></td>
<td>
<p>Object of class inheriting from &quot;lpda&quot;</p>
</td></tr>
<tr><td><code id="plot.lpda_+3A_pcscores">PCscores</code></td>
<td>
<p>Logical to show the first 2 PCscores. Only possible when PCA is applied.</p>
</td></tr>
<tr><td><code id="plot.lpda_+3A_xlim">xlim</code></td>
<td>
<p>An optional vector with two values with the x-axis range. If omitted, it will be computed.</p>
</td></tr>
<tr><td><code id="plot.lpda_+3A_main">main</code></td>
<td>
<p>An optional title for the plot.</p>
</td></tr>
<tr><td><code id="plot.lpda_+3A_legend.pos">legend.pos</code></td>
<td>
<p>The position for the legend. By default it is topright. NULL when no legend is required.</p>
</td></tr>
<tr><td><code id="plot.lpda_+3A_...">...</code></td>
<td>
<p>Other arguments passed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Two dimensinal plot representing the distances to the computed hyperplane of each individual colored with the real class.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>

<hr>
<h2 id='predict.lpda'>Predict method for lpda classification</h2><span id='topic+predict.lpda'></span>

<h3>Description</h3>

<p>Predict method for lpda classification
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'lpda'
predict(object, datatest = object$data,...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict.lpda_+3A_object">object</code></td>
<td>
<p>Object of class inheriting from &quot;lpda&quot;</p>
</td></tr>
<tr><td><code id="predict.lpda_+3A_datatest">datatest</code></td>
<td>
<p>An optional data to predict their class. If omitted, the original data is used.</p>
</td></tr>
<tr><td><code id="predict.lpda_+3A_...">...</code></td>
<td>
<p>Other arguments passed.</p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>fitted</code></td>
<td>
<p>Predicted class</p>
</td></tr>
<tr><td><code>eval</code></td>
<td>
<p>Evaluation of each individual in the fitted model</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>


<h3>See Also</h3>

 <p><code><a href="#topic+lpda">lpda</a></code></p>

<hr>
<h2 id='RNAseq'>Simulated RNA-Seq dataset example</h2><span id='topic+RNAseq'></span>

<h3>Description</h3>

<p>A simulated RNA-Seq dataset example.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RNAseq</code></pre>


<h3>Format</h3>

<p>A data frame with 600 variables (in columns) and 60 samples (rows).
</p>


<h3>Details</h3>

<p>This dataset is a RNA-Seq simulated example.
It has been simulated as Negative Binomial distributed and transformed to rpkm  (Reads per kilo base per million mapped reads).
It contains  600 genes (in columns) and 60 samples (rows), 30 of each one of the experimental groups.
First 30 samples are from first group and the remaining samples from the second one.
</p>


<h3>Author(s)</h3>

<p>Maria Jose Nueda, <a href="mailto:mj.nueda@ua.es">mj.nueda@ua.es</a></p>

<hr>
<h2 id='stand'>stand center and scale a data matrix</h2><span id='topic+stand'></span>

<h3>Description</h3>

<p><code>stand</code> center and scale a data matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stand(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stand_+3A_x">X</code></td>
<td>
<p>a data matrix with individuals in rows and variables in columns</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scaled data matrix
</p>

<hr>
<h2 id='stand2'>stand2 center and scale a data matrix with the parameters of another one</h2><span id='topic+stand2'></span>

<h3>Description</h3>

<p><code>stand2</code> center and scale a data matrix with the parameters of another one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>stand2(X, X2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="stand2_+3A_x">X</code></td>
<td>
<p>the data matrix from which mean and standard deviation is computed</p>
</td></tr>
<tr><td><code id="stand2_+3A_x2">X2</code></td>
<td>
<p>the data matrix to center and scale</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Scaled X2 data matrix
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
