<!DOCTYPE html><html lang="en"><head><title>Help for package LSPFP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {LSPFP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#LSPFP-package'>
<p>Lysate and Secretome Peptide Feature Plotter</p></a></li>
<li><a href='#LSPFP-internal'>
<p>Lysate and Secretome Peptide Feature Plotter</p></a></li>
<li><a href='#printSelectedPeptides'>
<p>Only print peptides</p></a></li>
<li><a href='#wrapperLSPFP'>
<p>wrapperLSPFP</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Lysate and Secretome Peptide Feature Plotter</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-05-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Rafael Dellen, with contributions of Fabian Kruse</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gereon Poschmann &lt;gereon.poschmann@hhu.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Creates plots of peptides from shotgun proteomics analysis of secretome and lysate samples. These plots contain associated protein features and scores for potential secretion and truncation.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>seqinr, RCurl, data.table, bit64</td>
</tr>
<tr>
<td>Imports:</td>
<td>R.utils, graphics, grDevices, stats, utils</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-05-11 15:01:39 UTC; Poschmann</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-05-13 23:40:23 UTC</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
</table>
<hr>
<h2 id='LSPFP-package'>
Lysate and Secretome Peptide Feature Plotter
</h2><span id='topic+LSPFP-package'></span><span id='topic+LSPFP'></span>

<h3>Description</h3>

<p>Creates plots of peptides from shotgun proteomics analysis of secretome and lysate samples. These plots contain associated protein features and scores for potential secretion and truncation.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> LSPFP</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> Lysate and Secretome Peptide Feature Plotter</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.3</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-05-11</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Rafael Dellen, with contributions of Fabian Kruse</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Gereon Poschmann &lt;gereon.poschmann@hhu.de&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> Creates plots of peptides from shotgun proteomics analysis of secretome and lysate samples. These plots contain associated protein features and scores for potential secretion and truncation.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> seqinr, RCurl, data.table, bit64</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> R.utils, graphics, grDevices, stats, utils</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2016-05-19 10:46:48 UTC; dellen</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository: </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
Date/Publication: </td><td style="text-align: left;"> 2016-05-19 16:35:29</td>
</tr>
<tr>
 <td style="text-align: left;">
RoxygenNote: </td><td style="text-align: left;"> 6.1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
LSPFP-package           Lysate and Secretome Peptide Feature Plotter
printSelectedPeptides   Only print peptides
wrapperLSPFP            wrapperLSPFP
</pre>
<p>The package offers two functions. 
Please have a look at wrapperLSPFP and printSelectedPeptides for details.
</p>


<h3>Author(s)</h3>

<p>Rafael Dellen, with contributions of Fabian Kruse
</p>
<p>Maintainer: Gereon Poschmann &lt;gereon.poschmann@hhu.de&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Please have a look at wrapperLSPFP for an example.
</code></pre>

<hr>
<h2 id='LSPFP-internal'>
Lysate and Secretome Peptide Feature Plotter
</h2><span id='topic+bdown'></span><span id='topic+create_countintensematrix'></span><span id='topic+createFeatureTable'></span><span id='topic+createNewExp'></span><span id='topic+detectTerminal'></span><span id='topic+fasta2rds'></span><span id='topic+localfasta2rds'></span><span id='topic+getAttributeField'></span><span id='topic+getExperimentSourceFiles'></span><span id='topic+getPositions'></span><span id='topic+getTopofeatures'></span><span id='topic+gff2rds'></span><span id='topic+localgff2rds'></span><span id='topic+gffRead'></span><span id='topic+loadData'></span><span id='topic+LoadExp'></span><span id='topic+mylogisfunc'></span><span id='topic+openCSV'></span><span id='topic+openProgress'></span><span id='topic+openTXT'></span><span id='topic+peptide_filter'></span><span id='topic+print_peptides'></span><span id='topic+print_topoareas'></span><span id='topic+secscoring'></span><span id='topic+setDirStruc'></span><span id='topic+lspfp_getFastaInformation'></span><span id='topic+ss_openCSV'></span><span id='topic+ss_openTXT'></span><span id='topic+truncscoring'></span><span id='topic+wrapperPLM'></span><span id='topic+wrapperSecScore'></span>

<h3>Description</h3>

<p>Internal LSPFP functions.
</p>


<h3>Details</h3>

<p>These are not to be called by the user (or in some cases are just
waiting for proper documentation to be written :).
</p>

<hr>
<h2 id='printSelectedPeptides'>
Only print peptides
</h2><span id='topic+printSelectedPeptides'></span>

<h3>Description</h3>

<p>This function allows printing of different peptide plots after running wrapperLSPFP once.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>printSelectedPeptides(path, fname, mysorteddf = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="printSelectedPeptides_+3A_path">path</code></td>
<td>

<p>Character vector giving the path to a data folder in the analysis directory.
</p>
</td></tr>
<tr><td><code id="printSelectedPeptides_+3A_fname">fname</code></td>
<td>

<p>Character vector containing the name of the new file.
</p>
</td></tr>
<tr><td><code id="printSelectedPeptides_+3A_mysorteddf">mysorteddf</code></td>
<td>

<p>A data.frame that can be a sorted and/or shortened data.frame of the <code>feature_table</code> from <code>path</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function offers the possibility to print more selectively peptide plots after wrapperLSPFP was applied.
To get a smaller plot, it is necessary to commit an existing anlaysis directory from the AnalysisData directory.
The new file is named like <code>fname</code> and stored in the specified directory <code>path</code>. If a smaller dataset should be printed, a row wise sorted or shortened <code>feature_table</code> is to be used as basis for printing but no columns should be removed. This new data.frame (<code>mysorteddf</code>) will be processed for printing.
</p>


<h3>Value</h3>

<p>A PDF file named like fname in the analysis directory path. Returns TRUE if the printing was successful and FALSE if not.
</p>


<h3>Note</h3>

<p>A detailed vignette will follow soon.
</p>


<h3>Author(s)</h3>

<p>Rafael Dellen <br />
<a href="mailto:Rafael.Dellen@uni-duesseldorf.de">Rafael.Dellen@uni-duesseldorf.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
#To run this example you need to run the example from wrapperLSPFP

path &lt;- paste0(getwd(),"/AnalysisData/Test_Mouse")
test &lt;- read.csv(paste0(path, "/feature_table.csv"))


#Rows can be deleted or sorted but columns should not be removed
#Print the first 25 rows
myprint &lt;- test[1:25,]
#Sort by accession
myprint2 &lt;- test[sort(test$Accession), ]

printSelectedPeptides(path, "newfeaturetable", mysorteddf = myprint)


## End(Not run)

</code></pre>

<hr>
<h2 id='wrapperLSPFP'>
wrapperLSPFP
</h2><span id='topic+wrapperLSPFP'></span>

<h3>Description</h3>

<p>THIS IS AN ALPHA BUILD OF 1.0.1
This function plots the positions of peptides with associated proteins from shotgun proteomics data from MaxQuant or Progenesis as input. The plots contain informations about: intensity, position, protein structure, location of protein domains, genename, protein accession, secretion score and truncation score. The plots are written to a PDF file
and a data.frame containing protein feature information is saved as a rds-file and csv-file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wrapperLSPFP(globpath, expname, sourcefiles, org, grlocationdf,
            version = "actual", species = c("HUMAN", "MOUSE", "RAT", "PIG"),
            proteomeid = c("UP000005640", "UP000000589", "UP000002494", "UP000008227"),
            taxid = c("9606", "10090", "10116", "9823"), domain = rep("Eukaryota", 4),
            forcedl = FALSE, pepstack = 2, pepque = 2, sortprint = "fcsmall",
            unipep = TRUE, localfasta = "none")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="wrapperLSPFP_+3A_globpath">globpath</code></td>
<td>

<p>Character string indicating the path of the global directory.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_expname">expname</code></td>
<td>

<p>Character string indicating the name of the directory where the files for this run are saved.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_sourcefiles">sourcefiles</code></td>
<td>

<p>Character string indicating the path to the peptides-file.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_org">org</code></td>
<td>

<p>Character string specifying the organism the peptides are from.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_grlocationdf">grlocationdf</code></td>
<td>

<p>Data.frame including the following columns: Expname, Location, Treatment, Sample, Group.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_version">version</code></td>
<td>

<p>Character string indicating what version of the &lsquo;<span class="file">BasicData</span>&rsquo; file should be used.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_species">species</code></td>
<td>

<p>Character strings specifying the UniProt species names of the data sets for download from UniProt to &lsquo;<span class="file">BasicData</span>&rsquo;.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_proteomeid">proteomeid</code></td>
<td>

<p>Character strings specifying the UniProt proteome IDs of the organisms for download. Must be in the same order as <code>species</code>.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_taxid">taxid</code></td>
<td>

<p>Character strings containing the UniProt taxonomic IDs of the organisms for download. Must be in the same order as <code>species</code>.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_domain">domain</code></td>
<td>

<p>Character strings containing the UniProt domain description. Must be in the same order as <code>species</code>.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_forcedl">forcedl</code></td>
<td>

<p>Logical, TRUE: indicates that the actual <code>version</code> of &lsquo;<span class="file">BasicData</span>&rsquo; should be downloaded again, FALSE: if not.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_pepstack">pepstack</code></td>
<td>

<p>Numerical indicating the minimal number of runs per group where the same peptide should have been measured.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_pepque">pepque</code></td>
<td>

<p>Numerical indicating the minimal number of peptides that should have been measured in each groupe.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_sortprint">sortprint</code></td>
<td>

<p>Character string that indicates how the peptide plot will be sorted. &quot;fcsmall&quot;: decreasing <code>fcsmall</code> values  , &quot;lflf&quot;: decreasing <code>fclf</code> values, &quot;trunc&quot;: decreasing <code>tscore</code> values, &quot;acc&quot;: increasing <code>accessions</code>, &quot;ntts&quot;: decreasing values of <code>NTT</code>*<code>NTTcov</code>, &quot;ctts&quot;: decreasing values of <code>CTT</code>*<code>CTTcov</code>.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_unipep">unipep</code></td>
<td>

<p>Logical that indicates if only unique peptides should be used. TRUE: only uniques, FALSE: all peptides in the file.
</p>
</td></tr>
<tr><td><code id="wrapperLSPFP_+3A_localfasta">localfasta</code></td>
<td>

<p>Character string indicating that a local FASTA file should be used. Default: &quot;none&quot;, if the FASTA file should be downloaded from UniProt, &quot;...&quot;: any valid <code>path</code> to a FASTA file on the system.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><b>Worklfow</b><br />
1. Create a data structure.<br />
2. Download and prepare data from UniProt.<br />
3. Calculate features.<br />
4. Print plots to PDF.<br />
5. Save features as .rds and .csv.<br /><br />
</p>
<p><b>Input</b><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
peptide-files: the program uses the file extension to decide if the input was created by MaxQuant (.txt) or by Progenesis (.csv).<br /><br />
<b><code>grlocationdf</code>:</b> Should be a data.frame that contains the following columns:<br /><br />
<b><code>Expname</code>:</b> This column should contain all experiment names from the peptide file that should be used for the feature plotting. Put each experiment in a single row. It should be a character string. Spaces will be filled with underscores automatically.<br />
</p>
<p><b><code>Location</code>:</b> Filled with &quot;Secretome&quot; or &quot;Proteome&quot; character strings that indicates if the experiment is from a lysate or not .<br />
<b><code>Treatment</code>:</b> If the experiments are based on different treatments of the cell, should be marked here. The value must be a character string that matches [A-Z][A-Z].<br />
<b><code>Sample</code>:</b> Here should be marked from which sample each experiment is from. The values should be simple numbers of type numeric.<br />
<b><code>Group</code>:</b> These values are used to set experiment groups for the secretome scores. They have to be numbers of type numeric. Set at least two groups if there are to different Locations (e.g. &quot;Secretome&quot; = 1 and &quot;Proteome&quot; = 2). Choose more groups if there are dependencies between treatment, sample and location.<br />
</p>
<p><code>grlocationdf</code> is used to assign the experiments correctly during the different scoring and plotting functions.<br /><br />
</p>
<p><b>Datastructure</b><br />
At the beginning the following file-structure is created: <br />
&lsquo;<span class="file">globalpath/BasicData</span>&rsquo;<br />
&lsquo;<span class="file">globalpath/AnalysisData</span>&rsquo;<br />
If it already exists, nothing new will be created and the old one is used.
In BasicData all the different versions of UniProt download files are stored.
In AnalysisData the output-files are stored.
</p>
<p><b>UniProt</b><br />
For annotation and protein sequence information the organisem specific gff-files and fasta-files are downloaded from UniProt database <a href="http://www.uniprot.org/">http://www.uniprot.org/</a>. As default the actual data-sets are retrieved. If an already downloaded dataset should be used, set <code>version</code> to the foldername of the existing dataset. No archived version can be downloaded from UniProt.<br />
<br />
</p>
<p><b>Output</b><br />
The PDF output file will contain a plot for every unique protein a peptide was identified by MaxQuant or Progenesis. <br />
</p>
<p>A data.frame containing values that are used for the score calculation will be at: <br /><br /> &lsquo;<span class="file">globpath/AnalysisData/expname/feature_table.rds</span>&rsquo; and <br />&lsquo;<span class="file">globpath/AnalysisData/expname/feature_table.csv</span>&rsquo; <br />
It contains the following columns: Accession, NTT, NTTcov, CTT, CTTcov, TotalPep, ProtL, MeanSec, MeanProt, MeanSecLF, MeanProtLF, ntts, ctts, tscore, fcsmall, fclf <br />
A .csv-file of <code>grlocationdf</code> will be saved in &lsquo;<span class="file">AnalysisData</span>&rsquo;, too.<br />
Further more &lsquo;<span class="file">intenscount_table.csv</span>&rsquo; and &lsquo;<span class="file">namesdf_table.csv</span>&rsquo; are stored in the &lsquo;<span class="file">AnalysisData</span>&rsquo; directory. They
contain information that enables printing with <code>print_selected_peptides</code>.<br />
Further more there should be two tables (&lsquo;<span class="file">metaIntens.csv</span>&rsquo;, &lsquo;<span class="file">metaIntens.csv</span>&rsquo;) that contain different length and intensity values based on annotated extracellular and cytoplasmic regions.<br />
</p>


<h3>Value</h3>

<p>The return value is TRUE if no error occured and FALSE otherwise.
The plots and the feature table can be found in &lsquo;<span class="file">globpath/AnalysisData/expname</span>&rsquo;.
</p>


<h3>Note</h3>

<p>The download from UniProt may take a while, between 5 and 10 minutes per organism.
A detailed vignette will follow soon.
</p>


<h3>Author(s)</h3>

<p>Rafael Dellen <br />
<a href="mailto:Rafael.Dellen@uni-duesseldorf.de">Rafael.Dellen@uni-duesseldorf.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The download of gff-files and FASTA-sequences from UniProt
# might be time consuming.
# Please consider this before running the example.

## Not run: 
#please choose a path
globpath &lt;- getwd()

expname &lt;- "Test_Mouse"

sourcefiles &lt;- system.file("extdata", "Mouse.txt",
                        package="LSPFP")

org &lt;- "Mouse"


#prepare grlocationdf
expnames &lt;- c("Lysat_PB1a","Lysat_PB2a","Lysat_PB3a","Lysat_PB4a",
  "Lysat_PB5a","Lysat_PK1a", "Lysat_PK2a","Lysat_PK3a","Lysat_PK4a",
  "Lysat_PK5a","Sekretom_PB1a","Sekretom_PB2a","Sekretom_PB3a",
  "Sekretom_PB4a","Sekretom_PB5a","Sekretom_PK1a","Sekretom_PK2a",
  "Sekretom_PK3a","Sekretom_PK4a","Sekretom_PK5a")

# Are the values from the secretome or the proteome of the cells?
explocation &lt;- c(rep("Proteome",10),rep("Secretome",10))

# Are the cells from the same culture eg. patient?
expsample &lt;- c(rep(1:5,4))

# Are the samples differently treated?
#(different environments, chemicals, tissue extraction technique)?
exptreatment &lt;- c(rep("AA",5),rep("BB",5),rep("AB",5),rep("BC",5))

#Group specifies which experiments belong together
group &lt;- c(rep(1,10),rep(2,10))
grlocationdf &lt;- data.frame(Expname = expnames, Location = explocation,
                          Treatment = exptreatment, Sample = expsample,
                          Group = group, stringsAsFactors = FALSE)


species &lt;- "MOUSE"
proteomeid &lt;- "UP000000589"
taxid &lt;- "10090"
domain &lt;- "Eukaryota"


res &lt;- wrapperLSPFP(globpath, expname, sourcefiles, org, grlocationdf,
                    species= species, proteomeid = proteomeid,
                    taxid = taxid, domain = domain)


## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
