<!DOCTYPE html><html lang="en-US"><head><title>Help for package Luminescence</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Luminescence}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Luminescence-package'><p>Comprehensive Luminescence Dating Data Analysis<br /></p>

<img src="../help/figures/Luminescence_logo.png" width="75" alt="r-luminescence.org" /></a></li>
<li><a href='#add_metadata+26lt+3B-'><p>Safe manipulation of object metadata</p></a></li>
<li><a href='#analyse_Al2O3C_CrossTalk'><p>Al2O3:C Reader Cross Talk Analysis</p></a></li>
<li><a href='#analyse_Al2O3C_ITC'><p>Al2O3 Irradiation Time Correction Analysis</p></a></li>
<li><a href='#analyse_Al2O3C_Measurement'><p>Al2O3:C Passive Dosimeter Measurement Analysis</p></a></li>
<li><a href='#analyse_baSAR'><p>Bayesian models (baSAR) applied on luminescence data</p></a></li>
<li><a href='#analyse_FadingMeasurement'><p>Analyse fading measurements and returns the fading rate per decade (g-value)</p></a></li>
<li><a href='#analyse_IRSAR.RF'><p>Analyse IRSAR RF measurements</p></a></li>
<li><a href='#analyse_pIRIRSequence'><p>Analyse post-IR IRSL measurement sequences</p></a></li>
<li><a href='#analyse_portableOSL'><p>Analyse portable CW-OSL measurements</p></a></li>
<li><a href='#analyse_SAR.CWOSL'><p>Analyse SAR CW-OSL measurements</p></a></li>
<li><a href='#Analyse_SAR.OSLdata'><p>Analyse SAR CW-OSL measurements (deprecated)</p></a></li>
<li><a href='#analyse_SAR.TL'><p>Analyse SAR TL measurements</p></a></li>
<li><a href='#apply_CosmicRayRemoval'><p>Function to remove cosmic rays from an RLum.Data.Spectrum S4 class object</p></a></li>
<li><a href='#apply_Crosstalk'><p>Apply crosstalk</p></a></li>
<li><a href='#apply_EfficiencyCorrection'><p>Function to apply spectral efficiency correction to RLum.Data.Spectrum S4</p>
class objects</a></li>
<li><a href='#as'><p>as() - RLum-object coercion</p></a></li>
<li><a href='#BaseDataSet.ConversionFactors'><p>Base data set of dose-rate conversion factors</p></a></li>
<li><a href='#BaseDataSet.CosmicDoseRate'><p>Base data set for cosmic dose rate calculation</p></a></li>
<li><a href='#BaseDataSet.FractionalGammaDose'><p>Base data set of fractional gamma-dose values</p></a></li>
<li><a href='#BaseDataSet.GrainSizeAttenuation'><p>Base dataset for grain size attenuation data by Gu√©rin et al. (2012)</p></a></li>
<li><a href='#bin_RLum.Data'><p>Channel binning for RLum.Data S4 class objects.</p></a></li>
<li><a href='#calc_AliquotSize'><p>Estimate the amount of grains on an aliquot</p></a></li>
<li><a href='#calc_AverageDose'><p>Calculate the Average Dose and the dose rate dispersion</p></a></li>
<li><a href='#calc_CentralDose'><p>Apply the central age model (CAM) after Galbraith et al. (1999) to a given</p>
De distribution</a></li>
<li><a href='#calc_CobbleDoseRate'><p>Calculate dose rate of slices in a spherical cobble</p></a></li>
<li><a href='#calc_CommonDose'><p>Apply the (un-)logged common age model after Galbraith et al. (1999) to a</p>
given De distribution</a></li>
<li><a href='#calc_CosmicDoseRate'><p>Calculate the cosmic dose rate</p></a></li>
<li><a href='#calc_EED_Model'><p>Modelling Exponential Exposure Distribution</p></a></li>
<li><a href='#calc_FadingCorr'><p>Fading Correction after Huntley &amp; Lamothe (2001)</p></a></li>
<li><a href='#calc_FastRatio'><p>Calculate the Fast Ratio for CW-OSL curves</p></a></li>
<li><a href='#calc_FiniteMixture'><p>Apply the finite mixture model (FMM) after Galbraith (2005) to a given De</p>
distribution</a></li>
<li><a href='#calc_FuchsLang2001'><p>Apply the model after Fuchs &amp; Lang (2001) to a given De distribution</p></a></li>
<li><a href='#calc_gSGC'><p>Calculate De value based on the gSGC by Li et al., 2015</p></a></li>
<li><a href='#calc_gSGC_feldspar'><p>Calculate Global Standardised Growth Curve (gSGC) for Feldspar MET-pIRIR</p></a></li>
<li><a href='#calc_HomogeneityTest'><p>Apply a simple homogeneity test after Galbraith (2003)</p></a></li>
<li><a href='#calc_Huntley2006'><p>Apply the Huntley (2006) model</p></a></li>
<li><a href='#calc_IEU'><p>Apply the internal-external-uncertainty (IEU) model after Thomsen et al.</p>
(2007) to a given De distribution</a></li>
<li><a href='#calc_Lamothe2003'><p>Apply fading correction after Lamothe et al., 2003</p></a></li>
<li><a href='#calc_MaxDose'><p>Apply the maximum age model to a given De distribution</p></a></li>
<li><a href='#calc_MinDose'><p>Apply the (un-)logged minimum age model (MAM) after Galbraith et al. (1999)</p>
to a given De distribution</a></li>
<li><a href='#calc_MoransI'><p>Calculate Moran's I</p></a></li>
<li><a href='#calc_OSLLxTxDecomposed'><p>Calculate Lx/Tx ratio for decomposed CW-OSL signal components</p></a></li>
<li><a href='#calc_OSLLxTxRatio'><p>Calculate <code>Lx/Tx</code> ratio for CW-OSL curves</p></a></li>
<li><a href='#calc_SourceDoseRate'><p>Calculation of the source dose rate via the date of measurement</p></a></li>
<li><a href='#calc_Statistics'><p>Function to calculate statistic measures</p></a></li>
<li><a href='#calc_ThermalLifetime'><p>Calculates the Thermal Lifetime using the Arrhenius equation</p></a></li>
<li><a href='#calc_TLLxTxRatio'><p>Calculate the Lx/Tx ratio for a given set of TL curves -beta version-</p></a></li>
<li><a href='#calc_WodaFuchs2008'><p>Obtain the equivalent dose using the approach by Woda and Fuchs 2008</p></a></li>
<li><a href='#combine_De_Dr'><p>Combine Dose Rate and Equivalent Dose Distribution</p></a></li>
<li><a href='#convert_Activity2Concentration'><p>Convert Nuclide Activities to Abundance and Vice Versa</p></a></li>
<li><a href='#convert_BIN2CSV'><p>Export Risoe BIN-file(s) to CSV-files</p></a></li>
<li><a href='#convert_Concentration2DoseRate'><p>Dose-rate conversion function</p></a></li>
<li><a href='#convert_CW2pHMi'><p>Transform a CW-OSL curve into a pHM-OSL curve via interpolation under</p>
hyperbolic modulation conditions</a></li>
<li><a href='#convert_CW2pLM'><p>Transform a CW-OSL curve into a pLM-OSL curve</p></a></li>
<li><a href='#convert_CW2pLMi'><p>Transform a CW-OSL curve into a pLM-OSL curve via interpolation under linear</p>
modulation conditions</a></li>
<li><a href='#convert_CW2pPMi'><p>Transform a CW-OSL curve into a pPM-OSL curve via interpolation under</p>
parabolic modulation conditions</a></li>
<li><a href='#convert_Daybreak2CSV'><p>Export measurement data produced by a Daybreak luminescence reader to CSV-files</p></a></li>
<li><a href='#convert_PSL2CSV'><p>Export PSL-file(s) to CSV-files</p></a></li>
<li><a href='#convert_PSL2Risoe.BINfileData'><p>Convert portable OSL data to a Risoe.BINfileData object</p></a></li>
<li><a href='#convert_RLum2Risoe.BINfileData'><p>Converts RLum.Analysis and RLum.Data.Curve objects to RLum2Risoe.BINfileData objects</p></a></li>
<li><a href='#convert_Second2Gray'><p>Converting equivalent dose values from seconds (s) to Gray (Gy)</p></a></li>
<li><a href='#convert_SG2MG'><p>Converts Single-Grain Data to Multiple-Grain Data</p></a></li>
<li><a href='#convert_Wavelength2Energy'><p>Emission Spectra Conversion from Wavelength to Energy Scales (Jacobian Conversion)</p></a></li>
<li><a href='#convert_XSYG2CSV'><p>Export XSYG-file(s) to CSV-files</p></a></li>
<li><a href='#ExampleData.Al2O3C'><p>Example Al2O3:C Measurement Data</p></a></li>
<li><a href='#ExampleData.BINfileData'><p>Example data from a SAR OSL and SAR TL measurement for the package</p>
Luminescence</a></li>
<li><a href='#ExampleData.CobbleData'><p>Example data for calc_CobbleDoseRate()</p></a></li>
<li><a href='#ExampleData.CW_OSL_Curve'><p>Example CW-OSL curve data for the package Luminescence</p></a></li>
<li><a href='#ExampleData.DeValues'><p>Example De data sets for the package Luminescence</p></a></li>
<li><a href='#ExampleData.Fading'><p>Example data for feldspar fading measurements</p></a></li>
<li><a href='#ExampleData.FittingLM'><p>Example data for fit_LMCurve() in the package Luminescence</p></a></li>
<li><a href='#ExampleData.LxTxData'><p>Example Lx/Tx data from CW-OSL SAR measurement</p></a></li>
<li><a href='#ExampleData.LxTxOSLData'><p>Example Lx and Tx curve data from an artificial OSL measurement</p></a></li>
<li><a href='#ExampleData.MortarData'><p>Example equivalent dose data from mortar samples</p></a></li>
<li><a href='#ExampleData.portableOSL'><p>Example portable OSL curve data for the package Luminescence</p></a></li>
<li><a href='#ExampleData.RLum.Analysis'><p>Example data as RLum.Analysis objects</p></a></li>
<li><a href='#ExampleData.RLum.Data.Image'><p>Example data as RLum.Data.Image objects</p></a></li>
<li><a href='#ExampleData.ScaleGammaDose'><p>Example data for scale_GammaDose()</p></a></li>
<li><a href='#ExampleData.SurfaceExposure'><p>Example OSL surface exposure dating data</p></a></li>
<li><a href='#ExampleData.TR_OSL'><p>Example TR-OSL data</p></a></li>
<li><a href='#ExampleData.XSYG'><p>Example data for a SAR OSL measurement and a TL spectrum using a lexsyg</p>
reader</a></li>
<li><a href='#extdata'><p>Collection of External Data</p></a></li>
<li><a href='#extract_IrradiationTimes'><p>Extract Irradiation Times from an XSYG-file</p></a></li>
<li><a href='#extract_ROI'><p>Extract Pixel Values through Circular Region-of-Interests (ROI) from an Image</p></a></li>
<li><a href='#fit_CWCurve'><p>Nonlinear Least Squares Fit for CW-OSL curves -beta version-</p></a></li>
<li><a href='#fit_DoseResponseCurve'><p>Fit a dose-response curve for luminescence data (Lx/Tx against dose)</p></a></li>
<li><a href='#fit_EmissionSpectra'><p>Luminescence Emission Spectra Deconvolution</p></a></li>
<li><a href='#fit_LMCurve'><p>Non-linear Least Squares Fit for LM-OSL curves</p></a></li>
<li><a href='#fit_OSLLifeTimes'><p>Fitting and Deconvolution of OSL Lifetime Components</p></a></li>
<li><a href='#fit_SurfaceExposure'><p>Nonlinear Least Squares Fit for OSL surface exposure data</p></a></li>
<li><a href='#fit_ThermalQuenching'><p>Fitting Thermal Quenching Data</p></a></li>
<li><a href='#get_Layout'><p>Collection of layout definitions</p></a></li>
<li><a href='#get_Quote'><p>Function to return essential quotes</p></a></li>
<li><a href='#get_rightAnswer'><p>Function to get the right answer</p></a></li>
<li><a href='#get_Risoe.BINfileData'><p>General accessor function for Risoe.BINfileData objects</p></a></li>
<li><a href='#get_RLum'><p>General accessor function for RLum S4 class objects</p></a></li>
<li><a href='#GitHub-API'><p>GitHub API</p></a></li>
<li><a href='#import_Data'><p>Import Luminescence Data into R</p></a></li>
<li><a href='#install_DevelopmentVersion'><p>Attempts to install the development version of the 'Luminescence' package</p></a></li>
<li><a href='#length_RLum'><p>Length retrieval function for RLum S4 class objects</p></a></li>
<li><a href='#melt_RLum'><p>Melt RLum-class objects into a flat data.frame</p></a></li>
<li><a href='#merge_Risoe.BINfileData'><p>Merge Risoe.BINfileData objects or Risoe BIN-files</p></a></li>
<li><a href='#merge_RLum'><p>General merge function for RLum S4 class objects</p></a></li>
<li><a href='#merge_RLum.Analysis'><p>Merge function for RLum.Analysis S4 class objects</p></a></li>
<li><a href='#merge_RLum.Data.Curve'><p>Merge function for RLum.Data.Curve S4 class objects</p></a></li>
<li><a href='#merge_RLum.Data.Spectrum'><p>Merge function for RLum.Data.Spectrum S4 class objects</p></a></li>
<li><a href='#merge_RLum.Results'><p>Merge function for RLum.Results S4-class objects</p></a></li>
<li><a href='#methods_RLum'><p>methods_RLum</p></a></li>
<li><a href='#names_RLum'><p>Name retrieval function for RLum S4 class objects</p></a></li>
<li><a href='#plot_AbanicoPlot'><p>Function to create an Abanico Plot.</p></a></li>
<li><a href='#plot_DetPlot'><p>Create De(t) plot</p></a></li>
<li><a href='#plot_DoseResponseCurve'><p>Plot a dose-response curve for luminescence data (Lx/Tx against dose)</p></a></li>
<li><a href='#plot_DRCSummary'><p>Create a Dose-Response Curve Summary Plot</p></a></li>
<li><a href='#plot_DRTResults'><p>Visualise dose recovery test results</p></a></li>
<li><a href='#plot_FilterCombinations'><p>Plot filter combinations along with the (optional) net transmission window</p></a></li>
<li><a href='#plot_GrowthCurve'><p>Fit and plot a dose-response curve for luminescence data (Lx/Tx against dose)</p></a></li>
<li><a href='#plot_Histogram'><p>Plot a histogram with separate error plot</p></a></li>
<li><a href='#plot_KDE'><p>Plot kernel density estimate with statistics</p></a></li>
<li><a href='#plot_MoranScatterplot'><p>Moran Scatter Plot: Visualizing Spatial Dependency</p></a></li>
<li><a href='#plot_NRt'><p>Visualise natural/regenerated signal ratios</p></a></li>
<li><a href='#plot_OSLAgeSummary'><p>Plot Posterior OSL-Age Summary</p></a></li>
<li><a href='#plot_RadialPlot'><p>Function to create a Radial Plot</p></a></li>
<li><a href='#plot_Risoe.BINfileData'><p>Plot Single Luminescence Curves from a Risoe.BINfileData-class object</p></a></li>
<li><a href='#plot_RLum'><p>General plot function for RLum S4 class objects</p></a></li>
<li><a href='#plot_RLum.Analysis'><p>Plot function for an RLum.Analysis S4 class object</p></a></li>
<li><a href='#plot_RLum.Data.Curve'><p>Plot function for an RLum.Data.Curve S4 class object</p></a></li>
<li><a href='#plot_RLum.Data.Image'><p>Plot function for an <code>RLum.Data.Image</code> S4 class object</p></a></li>
<li><a href='#plot_RLum.Data.Spectrum'><p>Plot function for an RLum.Data.Spectrum S4 class object</p></a></li>
<li><a href='#plot_RLum.Results'><p>Plot function for an RLum.Results S4 class object</p></a></li>
<li><a href='#plot_ROI'><p>Create Regions of Interest (ROI) Graphic</p></a></li>
<li><a href='#plot_SingleGrainDisc'><p>Plot a disc with its values</p></a></li>
<li><a href='#plot_ViolinPlot'><p>Create a violin plot</p></a></li>
<li><a href='#PSL2Risoe.BINfileData'><p>Convert portable OSL data to a Risoe.BINfileData object</p></a></li>
<li><a href='#read_BIN2R'><p>Import Ris√∏ BIN/BINX-files into R</p></a></li>
<li><a href='#read_Daybreak2R'><p>Import measurement data produced by a Daybreak TL/OSL reader into R</p></a></li>
<li><a href='#read_HeliosOSL2R'><p>Import Luminescence Data from Helios Luminescence Reader</p></a></li>
<li><a href='#read_PSL2R'><p>Import PSL files to R</p></a></li>
<li><a href='#read_RF2R'><p>Import RF-files to R</p></a></li>
<li><a href='#read_SPE2R'><p>Import Princeton Instruments (TM) SPE-file into R</p></a></li>
<li><a href='#read_TIFF2R'><p>Import TIFF Image Data into R</p></a></li>
<li><a href='#read_XSYG2R'><p>Import XSYG files to R</p></a></li>
<li><a href='#replicate_RLum'><p>General replication function for RLum S4 class objects</p></a></li>
<li><a href='#report_RLum'><p>Create a HTML-report for (RLum) objects</p></a></li>
<li><a href='#Risoe.BINfileData-class'><p>Class <code>"Risoe.BINfileData"</code></p></a></li>
<li><a href='#Risoe.BINfileData2RLum.Analysis'><p>Convert Risoe.BINfileData object to an RLum.Analysis object</p></a></li>
<li><a href='#RLum-class'><p>Class <code>"RLum"</code></p></a></li>
<li><a href='#RLum.Analysis-class'><p>Class <code>"RLum.Analysis"</code></p></a></li>
<li><a href='#RLum.Data-class'><p>Class <code>"RLum.Data"</code></p></a></li>
<li><a href='#RLum.Data.Curve-class'><p>Class <code>"RLum.Data.Curve"</code></p></a></li>
<li><a href='#RLum.Data.Image-class'><p>Class <code>"RLum.Data.Image"</code></p></a></li>
<li><a href='#RLum.Data.Spectrum-class'><p>Class <code>"RLum.Data.Spectrum"</code></p></a></li>
<li><a href='#RLum.Results-class'><p>Class <code>"RLum.Results"</code></p></a></li>
<li><a href='#scale_GammaDose'><p>Calculate the gamma dose deposited within a sample taking layer-to-layer</p>
variations in radioactivity into account (according to Aitken, 1985)</a></li>
<li><a href='#Second2Gray'><p>Converting equivalent dose values from seconds (s) to Gray (Gy)</p></a></li>
<li><a href='#set_Risoe.BINfileData'><p>General setter function for Risoe.BINfileData objects</p></a></li>
<li><a href='#set_RLum'><p>General setter function for RLum S4 class objects</p></a></li>
<li><a href='#smooth_RLum'><p>Smoothing of data for RLum S4-class objects</p></a></li>
<li><a href='#sort_RLum'><p>Sort data for RLum S4-class objects</p></a></li>
<li><a href='#sTeve'><p>sTeve - sophisticated tool for efficient data validation and evaluation</p></a></li>
<li><a href='#structure_RLum'><p>General structure function for RLum S4 class objects</p></a></li>
<li><a href='#subset_SingleGrainData'><p>Simple Subsetting of Single Grain Data from Ris√∏ BIN/BINX files</p></a></li>
<li><a href='#template_DRAC'><p>Create a DRAC input data template (v1.2)</p></a></li>
<li><a href='#trim_RLum.Data'><p>Trim Channels of RLum.Data-class Objects</p></a></li>
<li><a href='#tune_Data'><p>Tune data for experimental purpose</p></a></li>
<li><a href='#use_DRAC'><p>Use DRAC to calculate dose rate data</p></a></li>
<li><a href='#verify_SingleGrainData'><p>Verify single grain data sets and check for invalid grains, i.e.</p>
zero-light level grains</a></li>
<li><a href='#view'><p>Convenience data visualisation function</p></a></li>
<li><a href='#write_R2BIN'><p>Export Risoe.BINfileData into Ris√∏ BIN/BINX-file</p></a></li>
<li><a href='#write_R2TIFF'><p>Export RLum.Data.Image and RLum.Data.Spectrum objects to TIFF Images</p></a></li>
<li><a href='#write_RLum2CSV'><p>Export RLum-objects to CSV</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Comprehensive Luminescence Dating Data Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2025-03-07</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sebastian Kreutzer &lt;maintainer_luminescence@r-luminescence.org&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of various R functions for the purpose of Luminescence
    dating data analysis. This includes, amongst others, data import, export,
    application of age models, curve deconvolution, sequence analysis and
    plotting of equivalent dose distributions.</td>
</tr>
<tr>
<td>Contact:</td>
<td>Package Developers &lt;developers@r-luminescence.org&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://r-lum.github.io/Luminescence/">https://r-lum.github.io/Luminescence/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/R-Lum/Luminescence/issues">https://github.com/R-Lum/Luminescence/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3), utils</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp (&ge; 1.0.12), RcppArmadillo (&ge; 0.12.8.4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>bbmle (&ge; 1.0.25.1), data.table (&ge; 1.15.4), DEoptim (&ge;
2.2-8), httr (&ge; 1.4.7), interp (&ge; 1.1-6), lamW (&ge; 2.2.3),
matrixStats (&ge; 1.3.0), methods, minpack.lm (&ge; 1.2-4), mclust
(&ge; 6.1), Rcpp (&ge; 1.0.12), shape (&ge; 1.4.6), parallel, XML (&ge;
3.99-0.16),</td>
</tr>
<tr>
<td>Suggests:</td>
<td>spelling (&ge; 2.3.0), plotly (&ge; 4.10.4), rmarkdown (&ge; 2.27),
rstudioapi (&ge; 0.16.0), rjags (&ge; 4-15), coda (&ge; 0.19-4),
knitr, pander (&ge; 0.6.5), testthat (&ge; 3.2.1), vdiffr (&ge;
1.0.0), tiff (&ge; 0.1-12), devtools (&ge; 2.4.5), R.rsp (&ge;
0.46.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp, knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-03-07 12:17:26 UTC; marco</td>
</tr>
<tr>
<td>Author:</td>
<td>Sebastian Kreutzer
    <a href="https://orcid.org/0000-0002-0734-2199"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, trl,
    cre, dtc],
  Christoph Burow <a href="https://orcid.org/0000-0002-5023-4046"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, trl, dtc],
  Michael Dietze <a href="https://orcid.org/0000-0001-6063-1726"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Margret C. Fuchs <a href="https://orcid.org/0000-0001-7210-1132"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Christoph Schmidt <a href="https://orcid.org/0000-0002-2309-3209"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Manfred Fischer [aut, trl],
  Johannes Friedrich
    <a href="https://orcid.org/0000-0002-0805-9547"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Norbert Mercier <a href="https://orcid.org/0000-0002-6375-9108"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Rachel K. Smedley <a href="https://orcid.org/0000-0001-7773-5193"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Claire Christophe [ctb],
  Antoine Zink <a href="https://orcid.org/0000-0001-7146-1101"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Julie Durcan <a href="https://orcid.org/0000-0001-8724-8022"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Georgina E. King <a href="https://orcid.org/0000-0003-1059-8192"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb, dtc],
  Anne Philippe <a href="https://orcid.org/0000-0002-5331-5087"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Guillaume Guerin <a href="https://orcid.org/0000-0001-6298-5579"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Svenja Riedesel <a href="https://orcid.org/0000-0003-2936-8776"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Martin Autzen <a href="https://orcid.org/0000-0001-6249-426X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Pierre Guibert <a href="https://orcid.org/0000-0001-8969-8684"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Dirk Mittelstrass <a href="https://orcid.org/0000-0002-9567-8791"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Harrison J. Gray <a href="https://orcid.org/0000-0002-4555-7473"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Jean-Michel Galharret
    <a href="https://orcid.org/0000-0003-2219-8727"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Marco Colombo <a href="https://orcid.org/0000-0001-6672-0623"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Luc Steinbuch <a href="https://orcid.org/0000-0001-6484-0920"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Anna-Maartje de Boer
    <a href="https://orcid.org/0000-0002-7359-6939"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Markus Fuchs <a href="https://orcid.org/0000-0003-4669-6528"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ths]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-03-07 16:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='Luminescence-package'>Comprehensive Luminescence Dating Data Analysis<br />

<img src="../help/figures/Luminescence_logo.png" width="75" alt="r-luminescence.org" />
</h2><span id='topic+Luminescence-package'></span><span id='topic+Luminescence'></span>

<h3>Description</h3>

<p>A collection of various R functions for the purpose of luminescence dating
data analysis. This includes, amongst others, data import, export,
application of age models, curve deconvolution, sequence analysis and
plotting of equivalent dose distributions.
</p>


<h3>Details</h3>

<p><strong>Supervisor of the initial version in 2012</strong>
</p>
<p>Markus Fuchs, Justus-Liebig-University Giessen, Germany
</p>
<p><strong>Support contact</strong>
</p>

<ul>
<li> <p><a href="mailto:developers@r-luminescence.org">developers@r-luminescence.org</a>
</p>
</li>
<li> <p><a href="https://github.com/R-Lum/Luminescence/discussions">https://github.com/R-Lum/Luminescence/discussions</a>
</p>
</li></ul>

<p><strong>Bug reporting</strong>
</p>

<ul>
<li> <p><a href="mailto:developers@r-luminescence.org">developers@r-luminescence.org</a> or
</p>
</li>
<li> <p><a href="https://github.com/R-Lum/Luminescence/issues">https://github.com/R-Lum/Luminescence/issues</a>
</p>
</li></ul>

<p><strong>Project website</strong>
</p>

<ul>
<li> <p><a href="https://r-luminescence.org">https://r-luminescence.org</a>
</p>
</li></ul>

<p><strong>Project source code repository</strong>
</p>

<ul>
<li> <p><a href="https://github.com/R-Lum/Luminescence">https://github.com/R-Lum/Luminescence</a>
</p>
</li></ul>

<p><strong>Related package projects</strong>
</p>

<ul>
<li> <p><a href="https://cran.r-project.org/package=RLumShiny">https://cran.r-project.org/package=RLumShiny</a>
</p>
</li>
<li> <p><a href="https://cran.r-project.org/package=RLumModel">https://cran.r-project.org/package=RLumModel</a>
</p>
</li>
<li> <p><a href="https://cran.r-project.org/package=RLumCarlo">https://cran.r-project.org/package=RLumCarlo</a>
</p>
</li>
<li> <p><a href="https://cran.r-project.org/package=RCarb">https://cran.r-project.org/package=RCarb</a>
</p>
</li></ul>

<p><strong>Funding</strong>
</p>

<ul>
<li><p> 2011-2013:  The initial version of the package was developed, while Sebastian Kreutzer
was funded through the DFG programme &quot;Rekonstruktion der Umweltbedingungen
des Sp√§tpleistoz√§ns in Mittelsachsen anhand von L√∂ss-Pal√§obodensequenzen&quot;
(DFG id: 46526743)
</p>
</li>
<li><p> 2014-2018: Cooperation and personal exchange between the developers is gratefully
funded by the DFG (SCHM 3051/3-1) in the framework of the program
&quot;Scientific Networks&quot;. Project title: &quot;RLum.Network: Ein
Wissenschaftsnetzwerk zur Analyse von Lumineszenzdaten mit R&quot; (2014-2018)
</p>
</li>
<li><p> 05/2014-12/2019: The work of Sebastian Kreutzer as maintainer of the package was supported
by LabEx LaScArBx (ANR - n. ANR-10-LABX-52).
</p>
</li>
<li><p> 01/2020-04/2022: Sebastian Kreutzer as maintainer of the package has received funding
from the European Union‚Äôs Horizon 2020 research and innovation programme under
the Marie Sk≈Çodowska-Curie grant agreement No 844457 (CREDit), and could continue
maintaining the package.
</p>
</li>
<li><p> since 03/2023: Sebastian Kreutzer as maintainer of the package receives funding from the
DFG Heisenberg programme No 505822867.
</p>
</li>
<li><p> All other authors gratefully received additional funding from various public funding bodies.
</p>
</li></ul>



<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Sebastian Kreutzer <a href="mailto:maintainer_luminescence@r-luminescence.org">maintainer_luminescence@r-luminescence.org</a> (<a href="https://orcid.org/0000-0002-0734-2199">ORCID</a>) [translator, data contributor]
</p>
<p>Authors:
</p>

<ul>
<li><p> Christoph Burow (<a href="https://orcid.org/0000-0002-5023-4046">ORCID</a>) [translator, data contributor]
</p>
</li>
<li><p> Michael Dietze (<a href="https://orcid.org/0000-0001-6063-1726">ORCID</a>)
</p>
</li>
<li><p> Margret C. Fuchs (<a href="https://orcid.org/0000-0001-7210-1132">ORCID</a>)
</p>
</li>
<li><p> Christoph Schmidt (<a href="https://orcid.org/0000-0002-2309-3209">ORCID</a>)
</p>
</li>
<li><p> Manfred Fischer [translator]
</p>
</li>
<li><p> Johannes Friedrich (<a href="https://orcid.org/0000-0002-0805-9547">ORCID</a>)
</p>
</li>
<li><p> Norbert Mercier (<a href="https://orcid.org/0000-0002-6375-9108">ORCID</a>)
</p>
</li>
<li><p> Anne Philippe (<a href="https://orcid.org/0000-0002-5331-5087">ORCID</a>)
</p>
</li>
<li><p> Svenja Riedesel (<a href="https://orcid.org/0000-0003-2936-8776">ORCID</a>)
</p>
</li>
<li><p> Martin Autzen (<a href="https://orcid.org/0000-0001-6249-426X">ORCID</a>)
</p>
</li>
<li><p> Dirk Mittelstrass (<a href="https://orcid.org/0000-0002-9567-8791">ORCID</a>)
</p>
</li>
<li><p> Harrison J. Gray (<a href="https://orcid.org/0000-0002-4555-7473">ORCID</a>)
</p>
</li>
<li><p> Jean-Michel Galharret (<a href="https://orcid.org/0000-0003-2219-8727">ORCID</a>)
</p>
</li>
<li><p> Marco Colombo (<a href="https://orcid.org/0000-0001-6672-0623">ORCID</a>)
</p>
</li>
<li><p> Luc Steinbuch (<a href="https://orcid.org/0000-0001-6484-0920">ORCID</a>)
</p>
</li>
<li><p> Anna-Maartje de Boer (<a href="https://orcid.org/0000-0002-7359-6939">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Rachel K. Smedley (<a href="https://orcid.org/0000-0001-7773-5193">ORCID</a>) [contributor]
</p>
</li>
<li><p> Claire Christophe [contributor]
</p>
</li>
<li><p> Antoine Zink (<a href="https://orcid.org/0000-0001-7146-1101">ORCID</a>) [contributor]
</p>
</li>
<li><p> Julie Durcan (<a href="https://orcid.org/0000-0001-8724-8022">ORCID</a>) [contributor]
</p>
</li>
<li><p> Georgina E. King (<a href="https://orcid.org/0000-0003-1059-8192">ORCID</a>) [contributor, data contributor]
</p>
</li>
<li><p> Guillaume Guerin (<a href="https://orcid.org/0000-0001-6298-5579">ORCID</a>) [contributor]
</p>
</li>
<li><p> Pierre Guibert (<a href="https://orcid.org/0000-0001-8969-8684">ORCID</a>) [contributor]
</p>
</li>
<li><p> Markus Fuchs (<a href="https://orcid.org/0000-0003-4669-6528">ORCID</a>) [thesis advisor]
</p>
</li></ul>



<h3>References</h3>

<p>Dietze, M., Kreutzer, S., Fuchs, M.C., Burow, C., Fischer, M.,
Schmidt, C., 2013. A practical guide to the R package Luminescence.
Ancient TL, 31 (1), 11-18.
</p>
<p>Dietze, M., Kreutzer, S., Burow, C., Fuchs, M.C., Fischer, M., Schmidt, C., 2016. The abanico plot:
visualising chronometric data with individual standard errors. Quaternary Geochronology 31, 1-7.
https://doi.org/10.1016/j.quageo.2015.09.003
</p>
<p>Fuchs, M.C., Kreutzer, S., Burow, C., Dietze, M., Fischer, M., Schmidt, C.,
Fuchs, M., 2015. Data processing in luminescence dating analysis: An
exemplary workflow using the R package 'Luminescence'. Quaternary
International, 362,8-13. https://doi.org/10.1016/j.quaint.2014.06.034
</p>
<p>Kreutzer, S., Schmidt, C., Fuchs, M.C., Dietze, M., Fischer, M., Fuchs, M.,
2012. Introducing an R package for luminescence dating analysis. Ancient TL,
30 (1), 1-8.
</p>
<p>Mercier, N., Kreutzer, S., Christophe, C., Gu√©rin, G., Guibert, P., Lahaye, C., Lanos, P., Philippe, A.,
Tribolo, C., 2016. Bayesian statistics in luminescence dating: The 'baSAR'-model and its
implementation in the R package ‚ÄôLuminescence‚Äô. Ancient TL 34 (2), 14-21.
</p>
<p>Mercier, N., Galharret, J.-M., Tribolo, C., Kreutzer, S., Philippe, A., 2022.
Luminescence age calculation through Bayesian convolution of equivalent dose
and dose-rate distributions: the De_Dr model.
Geochronology 4, 297‚Äì310. https://doi.org/10.5194/gchron-4-297-2022
</p>
<p>Smedley, R.K., 2015. A new R function for the Internal External Uncertainty (IEU) model.
Ancient TL, 33 (1), 16-21.
</p>
<p>King, E.G., Burow, C., Roberts, H., Pearce, N.J.G., 2018. Age determination
using feldspar: evaluating fading-correction model performance. Radiation Measurements 119, 58-73.
https://doi.org/10.1016/j.radmeas.2018.07.013
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://r-lum.github.io/Luminescence/">https://r-lum.github.io/Luminescence/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/R-Lum/Luminescence/issues">https://github.com/R-Lum/Luminescence/issues</a>
</p>
</li></ul>


<hr>
<h2 id='add_metadata+26lt+3B-'>Safe manipulation of object metadata</h2><span id='topic+add_metadata+3C-'></span><span id='topic+rename_metadata+3C-'></span><span id='topic+replace_metadata+3C-'></span>

<h3>Description</h3>

<p>Generic functions for manipulation of metadata in <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>,
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> and <a href="#topic+RLum.Data-class">RLum.Data</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_metadata(object, ...) &lt;- value

rename_metadata(object, ...) &lt;- value

replace_metadata(object, ...) &lt;- value
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_metadata+2B26lt+2B3B-_+3A_object">object</code></td>
<td>
<p>(<strong>required</strong>) object to manipulate</p>
</td></tr>
<tr><td><code id="add_metadata+2B26lt+2B3B-_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
<tr><td><code id="add_metadata+2B26lt+2B3B-_+3A_value">value</code></td>
<td>
<p>the value to be assigned</p>
</td></tr>
</table>


<h3>How to cite</h3>

<p>Colombo, M., 2025. add_metadata&lt;-(): Safe manipulation of object metadata. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.BINfileData, envir = environment())

## show data
CWOSL.SAR.Data

## add a new field
add_metadata(CWOSL.SAR.Data,
             info_element = "INSTITUTE") &lt;- "Heidelberg University"

## rename a field
rename_metadata(CWOSL.SAR.Data,
                info_element = "INSTITUTE") &lt;- "INSTITUTION"

## replace all LTYPE to RSL
## but only for the first position
replace_metadata(
 object = CWOSL.SAR.Data,
 info_element = "LTYPE",
 subset = (POSITION == 1)) &lt;- "RSL"

## replacing a field with NULL allows to remove that field
replace_metadata(CWOSL.SAR.Data,
                 info_element = "PREVIOUS") &lt;- NULL

## show the modified data
CWOSL.SAR.Data

</code></pre>

<hr>
<h2 id='analyse_Al2O3C_CrossTalk'>Al2O3:C Reader Cross Talk Analysis</h2><span id='topic+analyse_Al2O3C_CrossTalk'></span>

<h3>Description</h3>

<p>The function provides the analysis of cross-talk measurements on a
FI lexsyg SMART reader using Al2O3:C chips
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_Al2O3C_CrossTalk(
  object,
  signal_integral = NULL,
  dose_points = c(0, 4),
  recordType = c("OSL (UVVIS)"),
  irradiation_time_correction = NULL,
  method_control = NULL,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
measurement input</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_signal_integral">signal_integral</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
signal integral, used for the signal and the background.
If nothing is provided the full range is used</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_dose_points">dose_points</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
vector with dose points, if dose points are repeated, only the general
pattern needs to be provided. Default values follow the suggestions
made by Kreutzer et al., 2018</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_recordtype">recordType</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): input curve selection, which is passed to
function <a href="#topic+get_RLum">get_RLum</a>. To deactivate the automatic selection set the argument to <code>NULL</code></p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_irradiation_time_correction">irradiation_time_correction</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> (<em>optional</em>):
information on the used irradiation time correction obtained by another
experiment.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
optional parameters to control the calculation.
See details for further explanations</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_CrossTalk_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to the plot output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Function returns results numerically and graphically:
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> summed apparent dose table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data_full&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> full apparent dose table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit&#8288;</code> </td><td style="text-align: left;"> <code>lm</code> </td><td style="text-align: left;"> the linear model obtained from fitting </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$col.seq&#8288;</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the used colour vector </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>

<ul>
<li><p> An overview of the obtained apparent dose values
</p>
</li></ul>



<h3>Function version</h3>

<p>0.1.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_Al2O3C_CrossTalk(): Al2O3:C Reader Cross Talk Analysis. Function version 0.1.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Kreutzer, S., Martin, L., Gu√©rin, G., Tribolo, C., Selva, P., Mercier, N., 2018. Environmental Dose Rate
Determination Using a Passive Dosimeter: Techniques and Workflow for alpha-Al2O3:C Chips.
Geochronometria 45, 56-67. doi: 10.1515/geochr-2015-0086
</p>


<h3>See Also</h3>

<p><a href="#topic+analyse_Al2O3C_ITC">analyse_Al2O3C_ITC</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.Al2O3C, envir = environment())

##run analysis
analyse_Al2O3C_CrossTalk(data_CrossTalk)

</code></pre>

<hr>
<h2 id='analyse_Al2O3C_ITC'>Al2O3 Irradiation Time Correction Analysis</h2><span id='topic+analyse_Al2O3C_ITC'></span>

<h3>Description</h3>

<p>The function provides a very particular analysis to correct the irradiation
time while irradiating Al2O3:C chips in a luminescence reader.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_Al2O3C_ITC(
  object,
  signal_integral = NULL,
  dose_points = c(2, 4, 8, 12, 16),
  recordType = c("OSL (UVVIS)"),
  method_control = NULL,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_Al2O3C_ITC_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
results obtained from the measurement.
Alternatively a list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects can be provided to allow an automatic analysis</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_signal_integral">signal_integral</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
signal integral, used for the signal and the background.
If nothing is provided the full range is used. Argument can be provided as <a href="base.html#topic+list">list</a>.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_dose_points">dose_points</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
vector with dose points, if dose points are repeated, only the general
pattern needs to be provided. Default values follow the suggestions
made by Kreutzer et al., 2018. Argument can be provided as <a href="base.html#topic+list">list</a>.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_recordtype">recordType</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): input curve selection, which is passed to
function <a href="#topic+get_RLum">get_RLum</a>. To deactivate the automatic selection set the argument to <code>NULL</code></p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
optional parameters to control the calculation.
See details for further explanations</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_ITC_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to the plot output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Background: Due to their high dose sensitivity Al2O3:C chips are usually
irradiated for only a very short duration or under the closed beta-source
within a luminescence reader. However, due to its high dose sensitivity, during
the movement towards the beta-source, the pellet already receives and non-negligible
dose. Based on measurements following a protocol suggested by Kreutzer et al., 2018,
a dose response curve is constructed and the intersection (absolute value) with the time axis
is taken as real irradiation time.
</p>
<p><strong><code>method_control</code></strong>
</p>
<p>To keep the generic argument list as clear as possible, arguments to allow a
deeper control of the method are all preset with meaningful default parameters and can be
handled using the argument <code>method_control</code> only, e.g.,
<code>method_control = list(fit.method = "LIN")</code>. Supported arguments are:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>ARGUMENT</strong> </td><td style="text-align: left;"> <strong>FUNCTION</strong> </td><td style="text-align: left;"> <strong>DESCRIPTION</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mode</code> </td><td style="text-align: left;"> <code>fit_DoseResponseCurve</code> </td><td style="text-align: left;"> as in <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>; sets the mode used for fitting</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fit.method</code> </td><td style="text-align: left;"> <code>fit_DoseResponseCurve</code> </td><td style="text-align: left;"> as in <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>; sets the function applied for fitting</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>Function returns results numerically and graphically:
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> correction value and error </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$table&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> table used for plotting  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$table_mean&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> table used for fitting </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit&#8288;</code> </td><td style="text-align: left;"> <code>lm</code> or <code>nls</code> </td><td style="text-align: left;"> the fitting as returned by the function <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>

<ul>
<li><p> A dose response curve with the marked correction values
</p>
</li></ul>



<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_Al2O3C_ITC(): Al2O3 Irradiation Time Correction Analysis. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Kreutzer, S., Martin, L., Gu√©rin, G., Tribolo, C., Selva, P., Mercier, N., 2018. Environmental Dose Rate
Determination Using a Passive Dosimeter: Techniques and Workflow for alpha-Al2O3:C Chips.
Geochronometria 45, 56-67. doi: 10.1515/geochr-2015-0086
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.Al2O3C, envir = environment())

##run analysis
analyse_Al2O3C_ITC(data_ITC)

</code></pre>

<hr>
<h2 id='analyse_Al2O3C_Measurement'>Al2O3:C Passive Dosimeter Measurement Analysis</h2><span id='topic+analyse_Al2O3C_Measurement'></span>

<h3>Description</h3>

<p>The function provides the analysis routines for measurements on a
FI lexsyg SMART reader using Al2O3:C chips according to Kreutzer et al., 2018
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_Al2O3C_Measurement(
  object,
  signal_integral = NULL,
  dose_points = c(0, 4),
  recordType = c("OSL (UVVIS)", "TL (UVVIS)"),
  calculate_TL_dose = FALSE,
  irradiation_time_correction = NULL,
  cross_talk_correction = NULL,
  travel_dosimeter = NULL,
  test_parameters = NULL,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):  measurement input</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_signal_integral">signal_integral</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): signal integral, used for the signal
and the background. Example: <code>c(1:10)</code> for the first 10 channels.
If nothing is provided the full range is used</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_dose_points">dose_points</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
vector with dose points, if dose points are repeated, only the general
pattern needs to be provided. Default values follow the suggestions
made by Kreutzer et al., 2018</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_recordtype">recordType</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): input curve selection, which is passed to
function <a href="#topic+get_RLum">get_RLum</a>. To deactivate the automatic selection set the argument to <code>NULL</code></p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_calculate_tl_dose">calculate_TL_dose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enables/disable experimental dose estimation based on the TL curves.
It is computed as the ratio of the peak sums of each curves +/- 5 channels.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_irradiation_time_correction">irradiation_time_correction</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> (<em>optional</em>):
information on the used irradiation time correction obtained by another experiments.
If a <code>numeric</code> is provided it has to be of length two: mean, standard error</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_cross_talk_correction">cross_talk_correction</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> (<em>optional</em>):
information on the used irradiation time correction obtained by another experiments.
If a <code>numeric</code> vector is provided it has to be of length three:
mean, 2.5 % quantile, 97.5 % quantile.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_travel_dosimeter">travel_dosimeter</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): specify the position of
the travel dosimeter (so far measured at the same time). The dose of
travel dosimeter will be subtracted from all other values.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_test_parameters">test_parameters</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>with default</em>):
set test parameters. Supported parameters are: <code>TL_peak_shift</code> All input: <a href="base.html#topic+numeric">numeric</a>
values, <code>NA</code> and <code>NULL</code> (see details).</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.
If <code>object</code> is of type <a href="base.html#topic+list">list</a>, a numeric vector can be provided to limit
the plot output to certain aliquots.</p>
</td></tr>
<tr><td><code id="analyse_Al2O3C_Measurement_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to the plot output, supported are <code>norm</code>, <code>main</code>, <code>mtext</code>,
<code>title</code> (for self-call mode to specify, e.g., sample names)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Working with a travel dosimeter</strong>
</p>
<p>The function allows to define particular position numbers as travel
dosimeters. For example: <code>travel_dosimeter = c(1,3,5)</code> sets positions 1,
3 and 5 as travel dosimeters. These dose values #' of this dosimeters
are combined and automatically subtracted from the obtained dose values
of the other dosimeters.
</p>
<p><strong>Calculate TL dose</strong>
</p>
<p>The argument <code>calculate_TL_dose</code> provides the possibility to experimentally calculate a TL-dose,
i.e. an apparent dose value derived from the TL curve ratio. However, it should be noted that
this value is only a fall back in case something went wrong during the measurement of the optical
stimulation. The TL derived dose value is corrected for cross-talk and for the irradiation time,
but not considered if a travel dosimeter is defined.
</p>
<p>Calculating the palaeodose is possible without <strong>any TL</strong> curve in the sequence!
</p>
<p><strong>Test parameters</strong>
</p>
<p><code>TL_peak_shift</code> <a href="base.html#topic+numeric">numeric</a> (default: <code>15</code>):
</p>
<p>Checks whether the TL peak shift is bigger &gt; 15 K, indicating a problem with the
thermal contact of the chip.
</p>
<p><code>stimulation_power</code> <a href="base.html#topic+numeric">numeric</a> (default: <code>0.05</code>):
</p>
<p>So far available, information on the delivered optical stimulation are compared. Compared are
the information from the first curves with all others. If the ratio differs more from
unity than the defined by the threshold, a warning is returned.
</p>


<h3>Value</h3>

<p>Function returns results numerically and graphically:
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> the estimated equivalent dose </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data_table&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> full dose and signal table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test_parameters</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> results with test parameters </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data_TDcorrected</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> travel dosimeter corrected results (only if TD was provided)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><em>Note: If correction the irradiation time and the cross-talk correction method is used, the De
values in the table <code>data</code> table are already corrected, i.e. if you want to get an uncorrected value,
you can use the column <code>CT_CORRECTION</code> remove the correction</em>
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>

<ul>
<li><p> OSL and TL curves, combined on two plots.
</p>
</li></ul>



<h3>Function version</h3>

<p>0.2.6
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_Al2O3C_Measurement(): Al2O3:C Passive Dosimeter Measurement Analysis. Function version 0.2.6. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Kreutzer, S., Martin, L., Gu√©rin, G., Tribolo, C., Selva, P., Mercier, N., 2018. Environmental Dose Rate
Determination Using a Passive Dosimeter: Techniques and Workflow for alpha-Al2O3:C Chips.
Geochronometria 45, 56-67.
</p>


<h3>See Also</h3>

<p><a href="#topic+analyse_Al2O3C_ITC">analyse_Al2O3C_ITC</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load data
data(ExampleData.Al2O3C, envir = environment())

##run analysis
analyse_Al2O3C_Measurement(data_CrossTalk)

</code></pre>

<hr>
<h2 id='analyse_baSAR'>Bayesian models (baSAR) applied on luminescence data</h2><span id='topic+analyse_baSAR'></span>

<h3>Description</h3>

<p>This function allows the application of Bayesian models on luminescence data, measured
with the single-aliquot regenerative-dose (SAR, Murray and Wintle, 2000) protocol. In particular,
it follows the idea proposed by Comb√®s et al., 2015 of using an hierarchical model for estimating
a central equivalent dose from a set of luminescence measurements. This function is (I) the adoption
of this approach for the R environment and (II) an extension and a technical refinement of the
published code.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_baSAR(
  object,
  CSV_file = NULL,
  aliquot_range = NULL,
  source_doserate = NULL,
  signal.integral,
  signal.integral.Tx = NULL,
  background.integral,
  background.integral.Tx = NULL,
  irradiation_times = NULL,
  sigmab = 0,
  sig0 = 0.025,
  distribution = "cauchy",
  baSAR_model = NULL,
  n.MCMC = 1e+05,
  fit.method = "EXP",
  fit.force_through_origin = TRUE,
  fit.includingRepeatedRegPoints = TRUE,
  method_control = list(),
  digits = 3L,
  distribution_plot = "kde",
  plot = TRUE,
  plot_reduced = TRUE,
  plot_singlePanels = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_baSAR_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>,
<a href="base.html#topic+character">character</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
input object used for the Bayesian analysis. If a <code>character</code> is provided the function
assumes a file connection and tries to import a BIN/BINX-file using the provided path. If a <code>list</code> is
provided the list can only contain either <code>Risoe.BINfileData</code> objects or <code>character</code>s
providing a file connection. Mixing of both types is not allowed. If an <a href="#topic+RLum.Results-class">RLum.Results</a>
is provided the function directly starts with the Bayesian Analysis (see details)</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_csv_file">CSV_file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
if a <code>character</code>, it must be the path to a CSV file with data for the
analysis. Either way, data should contain 3 columns:
the name of the file, the disc position and the grain position
(the last being 0 for multi-grain measurements).<br /></p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_aliquot_range">aliquot_range</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
allows to limit the range of the aliquots used for the analysis.
This argument has only an effect if the argument <code>CSV_file</code> is used or
the input is the previous output (i.e. is <a href="#topic+RLum.Results-class">RLum.Results</a>). In this case the
new selection will add the aliquots to the removed aliquots table.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_source_doserate">source_doserate</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
source dose rate of beta-source used for the measurement and its uncertainty
in Gy/s, e.g., <code>source_doserate = c(0.12, 0.04)</code>. Parameter can be provided
as <code>list</code>, for the case that more than one BIN-file is provided, e.g.,
<code>source_doserate = list(c(0.04, 0.004), c(0.05, 0.004))</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_signal.integral">signal.integral</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<strong>required</strong>):
vector with the limits for the signal integral used for the calculation,
e.g., <code>signal.integral = c(1:5)</code>. Ignored if <code>object</code> is an <a href="#topic+RLum.Results-class">RLum.Results</a> object.
The parameter can be provided as <code>list</code>, see <code>source_doserate</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_signal.integral.tx">signal.integral.Tx</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
vector with the limits for the signal integral for the Tx curve. I
f nothing is provided the value from <code>signal.integral</code> is used and it is ignored
if <code>object</code> is an <a href="#topic+RLum.Results-class">RLum.Results</a> object.
The parameter can be provided as <code>list</code>, see <code>source_doserate</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_background.integral">background.integral</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<strong>required</strong>):
vector with the bounds for the background integral.
Ignored if <code>object</code> is an <a href="#topic+RLum.Results-class">RLum.Results</a> object.
The parameter can be provided as <code>list</code>, see <code>source_doserate</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_background.integral.tx">background.integral.Tx</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
vector with the limits for the background integral for the Tx curve.
If nothing is provided the value from <code>background.integral</code> is used.
Ignored if <code>object</code> is an <a href="#topic+RLum.Results-class">RLum.Results</a> object.
The parameter can be provided as <code>list</code>, see <code>source_doserate</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_irradiation_times">irradiation_times</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): if set this vector replaces all irradiation
times for one aliquot and one cycle (Lx and Tx curves) and recycles it for all others cycles and aliquots.
Please note that if this argument is used, for every(!) single curve
in the dataset an irradiation time needs to be set.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
option to set a manual value for the overdispersion (for <code>LnTx</code> and <code>TnTx</code>),
used for the <code>Lx</code>/<code>Tx</code> error calculation. The value should be provided as
absolute squared count values, cf. <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>.
The parameter can be provided as <code>list</code>, see <code>source_doserate</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_sig0">sig0</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
allow adding an extra component of error to the final Lx/Tx error value
(e.g., instrumental error, see details is <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>).
The parameter can be provided as <code>list</code>, see <code>source_doserate</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_distribution">distribution</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
type of distribution that is used during Bayesian calculations for
determining the Central dose and overdispersion values.
Allowed inputs are <code>"cauchy"</code>, <code>"normal"</code> and <code>"log_normal"</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_basar_model">baSAR_model</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
option to provide an own modified or new model for the Bayesian calculation
(see details). If an own model is provided the argument <code>distribution</code> is
ignored and set to <code>'user_defined'</code></p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_n.mcmc">n.MCMC</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number of iterations for the Markov chain Monte Carlo (MCMC) simulations</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_fit.method">fit.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
equation used for the fitting of the dose-response curve using the function
<a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> and then for the Bayesian modelling. Here supported methods: <code>EXP</code>, <code>EXP+LIN</code> and <code>LIN</code></p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_fit.force_through_origin">fit.force_through_origin</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
force fitting through origin</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_fit.includingrepeatedregpoints">fit.includingRepeatedRegPoints</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
includes the recycling point (assumed to be measured during the last cycle)</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
named list of control parameters that can be directly
passed to the Bayesian analysis, e.g., <code>method_control = list(n.chains = 4)</code>.
See details for further information</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_digits">digits</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
round output to the number of given digits</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_distribution_plot">distribution_plot</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): sets the final distribution plot that
shows equivalent doses obtained using the frequentist approach and sets in the central dose
as comparison obtained using baSAR. Allowed input is <code>'abanico'</code> or <code>'kde'</code>. If set to <code>NULL</code> nothing is plotted.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_plot_reduced">plot_reduced</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the advanced plot output.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable single plots or plots arranged by <code>analyse_baSAR</code>.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="analyse_baSAR_+3A_...">...</code></td>
<td>
<p>parameters that can be passed to the function <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>
(almost full support), <a href="data.table.html#topic+fread">data.table::fread</a> (<code>skip</code>), <a href="#topic+read_BIN2R">read_BIN2R</a> (<code>n.records</code>,
<code>position</code>, <code>duplicated.rm</code>), see details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Internally the function consists of two parts: (I) The Bayesian core for the Bayesian calculations
and applying the hierarchical model and (II) a data pre-processing part. The Bayesian core can be run
independently, if the input data are sufficient (see below). The data pre-processing part was
implemented to simplify the analysis for the user as all needed data pre-processing is done
by the function, i.e. in theory it is enough to provide a BIN/BINX-file with the SAR measurement
data. For the Bayesian analysis for each aliquot the following information are needed from the SAR analysis.
<code>LxTx</code>, the <code>LxTx</code> error and the dose values for all regeneration points.
</p>
<p><strong>How is the systematic error contribution calculated?</strong>
</p>
<p>Standard errors (so far) provided with the source dose rate are considered as systematic uncertainties
and added to final central dose by:
</p>
<p style="text-align: center;"><code class="reqn">systematic.error = 1/n \sum SE(source.doserate)</code>
</p>

<p style="text-align: center;"><code class="reqn">SE(central.dose.final) = \sqrt{SE(central.dose)^2 + systematic.error^2}</code>
</p>

<p>Please note that this approach is rather rough and can only be valid if the source dose rate
errors, in case different readers had been used, are similar. In cases where more than
one source dose rate is provided a warning is given.
</p>
<p><strong>Input / output scenarios</strong>
</p>
<p>Various inputs are allowed for this function. Unfortunately this makes the function handling rather
complex, but at the same time very powerful. Available scenarios:
</p>
<p><strong>(1) - <code>object</code> is BIN-file or link to a BIN-file</strong>
</p>
<p>Finally it does not matter how the information of the BIN/BINX file are provided. The function
supports <strong>(a)</strong> either a path to a file or directory or a <code>list</code> of file names or paths or
<strong>(b)</strong> a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object or a list of these objects. The latter one can
be produced by using the function <a href="#topic+read_BIN2R">read_BIN2R</a>, but this function is called automatically
if only a file name and/or a path is provided. In both cases it will become the data that can be
used for the analysis.
</p>
<p><code style="white-space: pre;">&#8288;[CSV_file = NULL]&#8288;</code>
</p>
<p>If no CSV file (or data frame with the same format) is provided, the
function runs an automatic process that consists of the following steps:
</p>

<ol>
<li><p> Select all valid aliquots using the function <a href="#topic+verify_SingleGrainData">verify_SingleGrainData</a>
</p>
</li>
<li><p> Calculate <code>Lx/Tx</code> values using the function <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>
</p>
</li>
<li><p> Calculate De values using the function <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>
</p>
</li></ol>

<p>These proceeded data are subsequently used in for the Bayesian analysis
</p>
<p><code style="white-space: pre;">&#8288;[CSV_file != NULL]&#8288;</code>
</p>
<p>If a CSV file is provided (or a <code>data.frame</code> containing similar information)
the pre-processing phase consists of the following steps:
</p>

<ol>
<li><p> Calculate <code>Lx/Tx</code> values using the function <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>
</p>
</li>
<li><p> Calculate De values using the function <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>
</p>
</li></ol>

<p>The CSV file should contain the BIN-file names and the aliquots selected
for the further analysis. This allows a manual selection of input data, as the automatic selection
by <a href="#topic+verify_SingleGrainData">verify_SingleGrainData</a> might be not totally sufficient.
</p>
<p><strong>(2) - <code>object</code> <code style="white-space: pre;">&#8288;RLum.Results object&#8288;</code></strong>
</p>
<p>If an <a href="#topic+RLum.Results-class">RLum.Results</a> object is provided as input and(!) this object was
previously created by the function <code>analyse_baSAR()</code> itself, the pre-processing part
is skipped and the function starts directly with the Bayesian analysis. This option is very powerful
as it allows to change parameters for the Bayesian analysis without the need to repeat
the data pre-processing. If furthermore the argument <code>aliquot_range</code> is set, aliquots
can be manually excluded based on previous runs.
</p>
<p><strong><code>method_control</code></strong>
</p>
<p>These are arguments that can be passed directly to the Bayesian calculation core, supported arguments
are:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Parameter</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower_centralD</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> sets the lower bound for the expected De range. Change it only if you know what you are doing!</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper_centralD</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> sets the upper bound for the expected De range. Change it only if you know what you are doing!</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.chains</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> </td><td style="text-align: left;"> sets number of parallel chains for the model (default = 3) (cf. <a href="rjags.html#topic+jags.model">rjags::jags.model</a>)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>inits</code> </td><td style="text-align: left;"> <a href="base.html#topic+list">list</a> </td><td style="text-align: left;"> option to set initialisation values (cf. <a href="rjags.html#topic+jags.model">rjags::jags.model</a>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>thin</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> thinning interval for monitoring the Bayesian process (cf. <a href="rjags.html#topic+jags.model">rjags::jags.model</a>)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>variable.names</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> set the variables to be monitored during the MCMC run, default:
<code>'central_D'</code>, <code>'sigma_D'</code>, <code>'D'</code>, <code>'Q'</code>, <code>'a'</code>, <code>'b'</code>, <code>'c'</code>, <code>'g'</code>.
Note: only variables present in the model can be monitored.
</td>
</tr>

</table>

<p><strong>User defined models</strong><br />
</p>
<p>The function provides the option to modify and to define own models that can be used for
the Bayesian calculation. In the case the user wants to modify a model, a new model
can be piped into the function via the argument <code>baSAR_model</code> as <code>character</code>.
The model has to be provided in the JAGS dialect of the BUGS language (cf. <a href="rjags.html#topic+jags.model">rjags::jags.model</a>)
and parameter names given with the pre-defined names have to be respected, otherwise the function
will break.
</p>
<p><strong>FAQ</strong>
</p>
<p>Q: How can I set the seed for the random number generator (RNG)?
</p>
<p>A: Use the argument <code>method_control</code>, e.g., for three MCMC chains
(as it is the default):
</p>
<div class="sourceCode"><pre>method_control = list(
inits = list(
 list(.RNG.name = "base::Wichmann-Hill", .RNG.seed = 1),
 list(.RNG.name = "base::Wichmann-Hill", .RNG.seed = 2),
 list(.RNG.name = "base::Wichmann-Hill", .RNG.seed = 3)
))
</pre></div>
<p>This sets a reproducible set for every chain separately.<br />
</p>
<p>Q: How can I modify the output plots?
</p>
<p>A: You can't, but you can use the function output to create own, modified plots.
</p>
<p>Q: Can I change the boundaries for the central_D?
</p>
<p>A: Yes, we made it possible, but we DO NOT recommend it, except you know what you are doing!<br />
Example: <code style="white-space: pre;">&#8288;method_control = list(lower_centralD = 10))&#8288;</code>
</p>
<p>Q: The lines in the baSAR-model appear to be in a wrong logical order?<br />
</p>
<p>A: This is correct and allowed (cf. JAGS manual)
</p>
<p><strong>Additional arguments support via the <code>...</code> argument</strong>
</p>
<p>This list summarizes the additional arguments that can be passed to the internally used
functions.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Supported argument</strong> </td><td style="text-align: left;"> <strong>Corresponding function</strong> </td><td style="text-align: left;"> <strong>Default</strong> </td><td style="text-align: left;"> **Short description **</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>threshold</code> </td><td style="text-align: left;"> <a href="#topic+verify_SingleGrainData">verify_SingleGrainData</a> </td><td style="text-align: left;"> <code>30</code> </td><td style="text-align: left;"> change rejection threshold for curve selection </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>skip</code> </td><td style="text-align: left;"> <a href="data.table.html#topic+fread">data.table::fread</a> </td><td style="text-align: left;"> <code>0</code> </td><td style="text-align: left;"> number of rows to be skipped during import</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.records</code> </td><td style="text-align: left;"> <a href="#topic+read_BIN2R">read_BIN2R</a> </td><td style="text-align: left;"> <code>NULL</code> </td><td style="text-align: left;"> limit records during BIN-file import</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>duplicated.rm</code> </td><td style="text-align: left;"> <a href="#topic+read_BIN2R">read_BIN2R</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> remove duplicated records in the BIN-file</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pattern</code> </td><td style="text-align: left;"> <a href="#topic+read_BIN2R">read_BIN2R</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> select BIN-file by name pattern</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>position</code> </td><td style="text-align: left;"> <a href="#topic+read_BIN2R">read_BIN2R</a> </td><td style="text-align: left;"> <code>NULL</code> </td><td style="text-align: left;"> limit import to a specific position</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>background.count.distribution</code> </td><td style="text-align: left;"> <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a> </td><td style="text-align: left;"> <code>"non-poisson"</code> </td><td style="text-align: left;"> set assumed count distribution</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fit.weights</code> </td><td style="text-align: left;"> <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> enable/disable fit weights</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fit.bounds</code> </td><td style="text-align: left;"> <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> enable/disable fit bounds</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.MC</code> </td><td style="text-align: left;"> <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> </td><td style="text-align: left;"> <code>100</code> </td><td style="text-align: left;"> number of MC runs for error calculation</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>output.plot</code> </td><td style="text-align: left;"> <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> enable/disable dose response curve plot</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>output.plotExtended</code> </td><td style="text-align: left;"> <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> enable/disable extended dose response curve plot</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>recordType</code> </td><td style="text-align: left;"> <a href="#topic+get_RLum">get_RLum</a> </td><td style="text-align: left;"> <code style="white-space: pre;">&#8288;c(OSL (UVVIS), irradiation (NA)&#8288;</code> </td><td style="text-align: left;"> helps for the curve selection</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>Function returns results numerically and graphically:
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$summary&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> statistical summary, including the central dose </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$mcmc&#8288;</code> </td><td style="text-align: left;"> <code>mcmc</code> </td><td style="text-align: left;"> <a href="coda.html#topic+mcmc.list">coda::mcmc.list</a> object including raw output </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$models&#8288;</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> implemented models used in the baSAR-model core </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$input_object&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> summarising table (same format as the XLS-file) including, e.g., Lx/Tx values</td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$removed_aliquots&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> table with removed aliquots (e.g., <code>NaN</code>, or <code>Inf</code> <code>Lx</code>/<code>Tx</code> values). If nothing was removed <code>NULL</code> is returned
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>

<ul>
<li><p> (A) Ln/Tn curves with set integration limits,
</p>
</li>
<li><p> (B) trace plots are returned by the baSAR-model, showing the convergence of the parameters (trace)
and the resulting kernel density plots. If <code>plot_reduced = FALSE</code> for every(!) dose a trace and
a density plot is returned (this may take a long time),
</p>
</li>
<li><p> (C) dose plots showing the dose for every aliquot as boxplots and the marked
HPD in within. If boxes are coloured 'orange' or 'red' the aliquot itself should be checked,
</p>
</li>
<li><p> (D) the dose response curve resulting from the monitoring of the Bayesian modelling are
provided along with the Lx/Tx values and the HPD. Note: The amount for curves displayed
is limited to 1000 (random choice) for performance reasons,
</p>
</li>
<li><p> (E) the final plot is the De distribution as calculated using the conventional (frequentist) approach
and the central dose with the HPDs marked within. This figure is only provided for a comparison,
no further statistical conclusion should be drawn from it.
</p>
</li></ul>

<p><strong>Please note: If distribution was set to <code>log_normal</code> the central dose is given as geometric mean!</strong>
</p>


<h3>Function version</h3>

<p>0.1.36
</p>


<h3>How to cite</h3>

<p>Mercier, N., Kreutzer, S., 2025. analyse_baSAR(): Bayesian models (baSAR) applied on luminescence data. Function version 0.1.36. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>If you provide more than one BIN-file</strong>, it is <strong>strongly</strong> recommended to provide
a <code>list</code> with the same number of elements for the following parameters:
</p>
<p><code>source_doserate</code>, <code>signal.integral</code>, <code>signal.integral.Tx</code>, <code>background.integral</code>,
<code>background.integral.Tx</code>, <code>sigmab</code>, <code>sig0</code>.
</p>
<p>Example for two BIN-files: <code>source_doserate = list(c(0.04, 0.006), c(0.05, 0.006))</code>
</p>
<p><strong>The function is currently limited to work with standard Risoe BIN-files only!</strong>
</p>


<h3>Author(s)</h3>

<p>Norbert Mercier, Archa√©sciences Bordeaux, CNRS-Universit√© Bordeaux Montaigne (France) <br />
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany) <br />
The underlying Bayesian model based on a contribution by Comb√®s et al., 2015.
, RLum Developer Team</p>


<h3>References</h3>

<p>Comb√®s, B., Philippe, A., Lanos, P., Mercier, N., Tribolo, C., Guerin, G., Guibert, P., Lahaye, C., 2015.
A Bayesian central equivalent dose model for optically stimulated luminescence dating.
Quaternary Geochronology 28, 62-70. doi:10.1016/j.quageo.2015.04.001
</p>
<p>Mercier, N., Kreutzer, S., Christophe, C., Guerin, G., Guibert, P., Lahaye, C., Lanos, P., Philippe, A.,
Tribolo, C., 2016. Bayesian statistics in luminescence dating: The 'baSAR'-model and its implementation
in the R package 'Luminescence'. Ancient TL 34, 14-21.
</p>
<p><strong>Further reading</strong>
</p>
<p>Gelman, A., Carlin, J.B., Stern, H.S., Dunson, D.B., Vehtari, A., Rubin, D.B., 2013.
Bayesian Data Analysis, Third Edition. CRC Press.
</p>
<p>Murray, A.S., Wintle, A.G., 2000. Luminescence dating of quartz using an improved single-aliquot
regenerative-dose protocol. Radiation Measurements 32, 57-73. doi:10.1016/S1350-4487(99)00253-X
</p>
<p>Plummer, M., 2017. JAGS Version 4.3.0 user manual. <code style="white-space: pre;">&#8288;https://sourceforge.net/projects/mcmc-jags/files/Manuals/4.x/jags_user_manual.pdf/download&#8288;</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>, <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>,
<a href="data.table.html#topic+fread">data.table::fread</a>, <a href="#topic+verify_SingleGrainData">verify_SingleGrainData</a>,
<a href="rjags.html#topic+jags.model">rjags::jags.model</a>, <a href="rjags.html#topic+coda.samples">rjags::coda.samples</a>, <a href="graphics.html#topic+boxplot.default">boxplot.default</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) load package test data set
data(ExampleData.BINfileData, envir = environment())

##(2) selecting relevant curves, and limit dataset
CWOSL.SAR.Data &lt;- subset(
  CWOSL.SAR.Data,
  subset = POSITION%in%c(1:3) &amp; LTYPE == "OSL")

## Not run: 
##(3) run analysis
##please not that the here selected parameters are
##choosen for performance, not for reliability
results &lt;- analyse_baSAR(
  object = CWOSL.SAR.Data,
  source_doserate = c(0.04, 0.001),
  signal.integral = c(1:2),
  background.integral = c(80:100),
  fit.method = "LIN",
  plot = FALSE,
  n.MCMC = 200

)

print(results)


##CSV_file template
##copy and paste this the code below in the terminal
##you can further use the function write.csv() to export the example

CSV_file &lt;-
structure(
list(
 BIN_FILE = NA_character_,
 DISC = NA_real_,
 GRAIN = NA_real_),
   .Names = c("BIN_FILE", "DISC", "GRAIN"),
   class = "data.frame",
   row.names = 1L
)


## End(Not run)

</code></pre>

<hr>
<h2 id='analyse_FadingMeasurement'>Analyse fading measurements and returns the fading rate per decade (g-value)</h2><span id='topic+analyse_FadingMeasurement'></span>

<h3>Description</h3>

<p>The function analyses fading measurements and returns a fading rate
including an error estimation. The function is not limited to standard
fading measurements, as can be seen, e.g., Huntley and Lamothe (2001).
Additionally, the density of recombination centres (rho') is estimated after
Kars et al. (2008).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_FadingMeasurement(
  object,
  structure = c("Lx", "Tx"),
  signal.integral,
  background.integral,
  t_star = "half",
  n.MC = 100,
  verbose = TRUE,
  plot = TRUE,
  plot_singlePanels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_FadingMeasurement_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
input object with the measurement data. Alternatively, a <a href="base.html#topic+list">list</a> containing <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
objects or a <a href="base.html#topic+data.frame">data.frame</a> with three columns
(x = LxTx, y = LxTx error, z = time since irradiation) can be provided.
Can also be a wide table, i.e. a <a href="base.html#topic+data.frame">data.frame</a> with a number of columns divisible by 3
and where each triplet has the before mentioned column structure.
</p>
<p><strong>Please note: The input object should solely consists of the curve needed
for the data analysis, i.e. only IRSL curves representing Lx (and Tx). If
the object originated from an XSYG file, also the irradiation steps must
be preserved in the input object.</strong>
</p>
<p>If data from multiple aliquots are provided please <strong>see the details below</strong> with regard to
Lx/Tx normalisation. <strong>The function assumes that all your measurements are related to
one (comparable) sample. If you have to treat independent samples, you have use this function
in a loop.</strong></p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_structure">structure</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
sets the structure of the measurement data. Allowed are <code>'Lx'</code> or <code>c('Lx','Tx')</code>.
Other input is ignored</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_signal.integral">signal.integral</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<strong>required</strong>): vector with channels for the signal integral
(e.g., <code>c(1:10)</code>). Not required if a <code>data.frame</code> with <code>LxTx</code> values is provided.</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_background.integral">background.integral</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<strong>required</strong>): vector with channels for the background integral
(e.g., <code>c(90:100)</code>). Not required if a <code>data.frame</code> with <code>LxTx</code> values is provided.</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_t_star">t_star</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
method for calculating the time elapsed since irradiation if input is <strong>not</strong> a <code>data.frame</code>.
Options are: <code>'half'</code> (the default), <code style="white-space: pre;">&#8288;'half_complex&#8288;</code>, which uses the long equation in Auclair et al. 2003, and
and <code>'end'</code>, which takes the time between irradiation and the measurement step.
Alternatively, <code>t_star</code> can be a function with one parameter which works on <code>t1</code>.
For more information see details. <br />
</p>
<p><em><code>t_star</code> has no effect if the input is a <a href="base.html#topic+data.frame">data.frame</a>, because this input comes
without irradiation times.</em></p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number for Monte Carlo runs for the error estimation.</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>) or <a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
enable/disable single plot mode, i.e. one plot window per plot.
Alternatively a vector specifying the plot to be drawn, e.g.,
<code>plot_singlePanels = c(3,4)</code> draws only the last two plots</p>
</td></tr>
<tr><td><code id="analyse_FadingMeasurement_+3A_...">...</code></td>
<td>
<p>(<em>optional</em>) further arguments that can be passed to internally used functions. Supported arguments:
<code>xlab</code>, <code>log</code>, <code>mtext</code>, <code>plot.trend</code> (enable/disable trend blue line), and <code>xlim</code> for the
two first curve plots, and <code>ylim</code> for the fading
curve plot. For further plot customization please use the numerical output of the functions for
own plots.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>All provided output corresponds to the <code class="reqn">tc</code> value obtained by this analysis. Additionally
in the output object the g-value normalised to 2-days is provided. The output of this function
can be passed to the function <a href="#topic+calc_FadingCorr">calc_FadingCorr</a>.
</p>
<p><strong>Fitting and error estimation</strong>
</p>
<p>For the fitting the function <a href="stats.html#topic+lm">stats::lm</a> is used without applying weights. For the
error estimation all input values, except <code>tc</code>, as the precision can be considered as sufficiently
high enough with regard to the underlying problem, are sampled assuming a normal distribution
for each value with the value as the mean and the provided uncertainty as standard deviation.
</p>
<p><strong>The options for <code>t_star</code></strong>
</p>

<ul>
<li> <p><code>t_star = "half"</code> (the default) The calculation follows the simplified
version in Auclair et al. (2003), which reads
</p>
<p style="text-align: center;"><code class="reqn">t_{star} := t_1 + (t_2 - t_1)/2</code>
</p>

</li>
<li> <p><code>t_star = "half_complex"</code> This option applies the complex function shown in Auclair et al. (2003),
which is derived from Aitken (1985) appendix F, equations 9 and 11.
It reads </p>
<p style="text-align: center;"><code class="reqn">t_{star} = t0 * 10^[(t_2 log(t_2/t_0) - t_1 log(t_1/t_0) - 0.43(t_2 - t_1))/(t_2 - t_1)]</code>
</p>

<p>where 0.43 = <code class="reqn">1/ln(10)</code>. t0, which is an arbitrary constant, is set to 1.
Please note that the equation in Auclair et al. (2003) is incorrect
insofar that it reads <code class="reqn">10exp(...)</code>, where the base should be 10 and not the Euler's number.
Here we use the correct version (base 10).
</p>
</li>
<li> <p><code>t_star = "end"</code> This option uses the simplest possible form for <code>t_star</code> which is the time since
irradiation without taking into account any addition parameter and it equals t1 in Auclair et al. (2003)
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;t_star = &lt;function&gt;&#8288;</code> This last option allows you to provide an R function object that works on t1 and
gives you all possible freedom. For instance, you may want to define the following
function <code>fun &lt;- function(x) {x^2}</code>, this would square all values of t1, because internally
it calls <code>fun(t1)</code>. The name of the function does not matter.
</p>
</li></ul>

<p><strong>Density of recombination centres</strong>
</p>
<p>The density of recombination centres, expressed by the dimensionless variable rho', is estimated
by fitting equation 5 in Kars et al. (2008) to the data. For the fitting the function
<a href="stats.html#topic+nls">stats::nls</a> is used without applying weights. For the error estimation the same
procedure as for the g-value is applied (see above).
</p>
<p><strong>Multiple aliquots &amp; Lx/Tx normalisation</strong>
</p>
<p>Be aware that this function will always normalise all <code class="reqn">\frac{L_x}{T_x}</code> values
by the <code class="reqn">\frac{L_x}{T_x}</code> value of the
prompt measurement of the first aliquot. This implicitly assumes that there are no systematic
inter-aliquot variations in the <code class="reqn">\frac{L_x}{T_x}</code> values.
If deemed necessary to normalise the <code class="reqn">\frac{L_x}{T_x}</code> values  of each
aliquot by its individual prompt measurement please do so <strong>before</strong> running
<a href="#topic+analyse_FadingMeasurement">analyse_FadingMeasurement</a> and provide the already normalised values for <code>object</code> instead.
</p>
<p><strong>Shine-down curve plots</strong>
Please note that the shine-down curve plots are for information only. As such
not all pause steps are plotted to avoid graphically overloaded plots.
However, <em>all</em> pause times are taken into consideration for the analysis.
</p>


<h3>Value</h3>

<p>An <a href="#topic+RLum.Results-class">RLum.Results</a> object is returned:
</p>
<p>Slot: <strong>@data</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>COMMENT</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fading_results</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> results of the fading measurement in a table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fit</code> </td><td style="text-align: left;"> <code>lm</code> </td><td style="text-align: left;"> object returned by the used linear fitting function <a href="stats.html#topic+lm">stats::lm</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>rho_prime</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> results of rho' estimation after Kars et al. (2008) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LxTx_table</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> Lx/Tx table, if curve data had been provided </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>irr.times</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> vector with the irradiation times in seconds </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Slot: <strong>@info</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>OBJECT</strong> </td><td style="text-align: left;"> <code>TYPE</code> </td><td style="text-align: left;"> <code>COMMENT</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>call</code> </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.1.23
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Burow, C., 2025. analyse_FadingMeasurement(): Analyse fading measurements and returns the fading rate per decade (g-value). Function version 0.1.23. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany) <br />
Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Aitken, M.J., 1985. Thermoluminescence dating, Studies in archaeological science.
Academic Press, London, Orlando.
</p>
<p>Auclair, M., Lamothe, M., Huot, S., 2003. Measurement of anomalous fading for feldspar IRSL using
SAR. Radiation Measurements 37, 487-492. <a href="https://doi.org/10.1016/S1350-4487%2803%2900018-0">doi:10.1016/S1350-4487(03)00018-0</a>
</p>
<p>Huntley, D.J., Lamothe, M., 2001. Ubiquity of anomalous fading in K-feldspars and the measurement
and correction for it in optical dating. Canadian Journal of Earth Sciences 38,
1093-1106. doi: <code>10.1139/cjes-38-7-1093</code>
</p>
<p>Kars, R.H., Wallinga, J., Cohen, K.M., 2008. A new approach towards anomalous
fading correction for feldspar  IRSL dating-tests on samples in field saturation.
Radiation Measurements 43, 786-790. <a href="https://doi.org/10.1016/j.radmeas.2008.01.021">doi:10.1016/j.radmeas.2008.01.021</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+read_XSYG2R">read_XSYG2R</a>,
<a href="#topic+extract_IrradiationTimes">extract_IrradiationTimes</a>, <a href="#topic+calc_FadingCorr">calc_FadingCorr</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data (sample UNIL/NB123, see ?ExampleData.Fading)
data("ExampleData.Fading", envir = environment())

##(1) get fading measurement data (here a three column data.frame)
fading_data &lt;- ExampleData.Fading$fading.data$IR50

##(2) run analysis
g_value &lt;- analyse_FadingMeasurement(
fading_data,
plot = TRUE,
verbose = TRUE,
n.MC = 10)

##(3) this can be further used in the function
## to correct the age according to Huntley &amp; Lamothe, 2001
results &lt;- calc_FadingCorr(
age.faded = c(100,2),
g_value = g_value,
n.MC = 10)

</code></pre>

<hr>
<h2 id='analyse_IRSAR.RF'>Analyse IRSAR RF measurements</h2><span id='topic+analyse_IRSAR.RF'></span>

<h3>Description</h3>

<p>Function to analyse IRSAR RF measurements on K-feldspar samples, performed
using the protocol according to Erfurt et al. (2003) and beyond.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_IRSAR.RF(
  object,
  sequence_structure = c("NATURAL", "REGENERATED"),
  RF_nat.lim = NULL,
  RF_reg.lim = NULL,
  method = "FIT",
  method_control = NULL,
  test_parameters = NULL,
  n.MC = 10,
  txtProgressBar = TRUE,
  plot = TRUE,
  plot_reduced = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_IRSAR.RF_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or a <a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>-objects (<strong>required</strong>):
input object containing data for protocol analysis. The function expects to
find at least two curves in the <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object: (1) <code>RF_nat</code>, (2) <code>RF_reg</code>.
If a <code>list</code> is provided as input all other parameters can be provided as
<code>list</code> as well to gain full control.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_sequence_structure">sequence_structure</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> <a href="base.html#topic+character">character</a> (<em>with default</em>):
specifies the general sequence structure. Allowed steps are <code>NATURAL</code> and
<code>REGENERATED</code>, and one of each must appear.
In addition any other character is allowed in the sequence structure;
such curves will be ignored during the analysis.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_rf_nat.lim">RF_nat.lim</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>with default</em>):
set minimum and maximum channel range for natural signal fitting and sliding.
If only one value is provided this will be treated as minimum value and the
maximum limit will be added automatically.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_rf_reg.lim">RF_reg.lim</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>with default</em>):
set minimum and maximum channel range for regenerated signal fitting and sliding.
If only one value is provided this will be treated as minimum value and the
maximum limit will be added automatically.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_method">method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): select method applied for the
data analysis. Possible options are <code>"FIT"</code>, <code>"SLIDE"</code>, <code>"VSLIDE"</code>;
<code>"NONE"</code> can be used to disable the analysis and plot the natural points
at their original position.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
parameters to control the method, that can be passed to the chosen method.
These are for (1) <code>method = "FIT"</code>: <code>'trace'</code>, <code>'maxiter'</code>, <code>'warnOnly'</code>, <code>'minFactor'</code> and for
(2) <code>method = "SLIDE"</code>: <code>'correct_onset'</code>, <code>'show_density'</code>,  <code>'show_fit'</code>, <code>'trace'</code>.
See details.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_test_parameters">test_parameters</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>with default</em>):
set test parameters. Supported parameters are: <code>curves_ratio</code>, <code>residuals_slope</code> (only for
<code>method = "SLIDE"</code>), <code>curves_bounds</code>, <code>dynamic_ratio</code>,
<code>lambda</code>, <code>beta</code> and <code>delta.phi</code>. All input: <a href="base.html#topic+numeric">numeric</a>
values, <code>NA</code> and <code>NULL</code> (see Details).
</p>
<p>(see Details for further information)</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
set number of Monte Carlo runs for start parameter estimation (<code>method = "FIT"</code>) or
error estimation (<code>method = "SLIDE"</code>). This value can be set to <code>NULL</code> to skip the
MC runs. Note: Large values will significantly increase the computation time</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar during MC runs.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_plot_reduced">plot_reduced</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
provides a reduced plot output if enabled to allow common R plot combinations,
e.g., <code>par(mfrow(...))</code>. If <code>TRUE</code> no residual plot
is returned; it has no effect if <code>plot = FALSE</code></p>
</td></tr>
<tr><td><code id="analyse_IRSAR.RF_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the plot output.
Currently supported arguments are <code>main</code>, <code>xlab</code>, <code>ylab</code>,
<code>xlim</code>, <code>ylim</code>, <code>log</code>, <code>legend</code> (<code>TRUE/FALSE</code>),
<code>legend.pos</code>, <code>legend.text</code> (passes argument to x,y in
<a href="graphics.html#topic+legend">graphics::legend</a>), <code>xaxt</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs an IRSAR analysis described for K-feldspar samples by
Erfurt et al. (2003) assuming a negligible sensitivity change of the RF
signal.
</p>
<p><strong>General Sequence Structure</strong> (according to Erfurt et al., 2003)
</p>

<ol>
<li><p> Measuring IR-RF intensity of the natural dose for a few seconds (<code class="reqn">RF_{nat}</code>)
</p>
</li>
<li><p> Bleach the samples under solar conditions for at least 30 min without changing the geometry
</p>
</li>
<li><p> Waiting for at least one hour
</p>
</li>
<li><p> Regeneration of the IR-RF signal to at least the natural level (measuring (<code class="reqn">RF_{reg}</code>)
</p>
</li>
<li><p> Fitting data with a stretched exponential function
</p>
</li>
<li><p> Calculate the palaeodose <code class="reqn">D_{e}</code> using the parameters from the fitting
</p>
</li></ol>

<p>Actually three methods are supported to obtain the <code class="reqn">D_{e}</code>:
<code>method = "FIT"</code>, <code>method = "SLIDE"</code> and <code>method = "VSLIDE"</code>:
</p>
<p><strong><code>method = "FIT"</code></strong>
</p>
<p>The principle is described above and follows the original suggestions by
Erfurt et al., 2003. For the fitting the mean count value of the <code>RF_nat</code> curve is used.
</p>
<p>Function used for the fitting (according to Erfurt et al. (2003)):
</p>
<p style="text-align: center;"><code class="reqn">\phi(D) = \phi_{0}-\Delta\phi(1-exp(-\lambda*D))^\beta</code>
</p>

<p>with
<code class="reqn">\phi(D)</code> the dose dependent IR-RF flux,
<code class="reqn">\phi_{0}</code> the initial IR-RF flux,
<code class="reqn">\Delta\phi</code> the dose dependent change of the IR-RF flux,
<code class="reqn">\lambda</code> the exponential parameter, <code class="reqn">D</code> the dose and
<code class="reqn">\beta</code> the dispersive factor.
</p>
<p>To obtain the palaeodose
<code class="reqn">D_{e}</code> the function is changed to:
</p>
<p style="text-align: center;"><code class="reqn">D_{e} = ln(-(\phi(D) - \phi_{0})/(-\lambda*\phi)^{1/\beta}+1)/-\lambda</code>
</p>

<p>The fitting is done using the <code>port</code> algorithm of the <a href="stats.html#topic+nls">nls</a> function.
</p>
<p><strong><code>method = "SLIDE"</code></strong>
</p>
<p>For this method, the natural curve is slid along the x-axis until
congruence with the regenerated curve is reached. As opposed to fitting, this
allows working with the original data without the need for any physical
model. This approach was introduced for RF curves by Buylaert et al., 2012
and Lapp et al., 2012.
</p>
<p>Here the sliding is done by searching for the minimum of the sum of
squared residuals.
For the mathematical details of the implementation see Frouin et al., 2017
</p>
<p><strong><code>method = "VSLIDE"</code></strong>
</p>
<p>Same as <code>"SLIDE"</code> but searching also vertically for the best match (i.e. in xy-direction.)
See Kreutzer et al. (2017) and Murari et al. (2021). By default the vertical sliding
range is set automatically, but can be set manually by changing the
<code>vslide_range</code> parameter (see <code>method_control</code>).
</p>
<p><strong><code>method_control</code></strong>
</p>
<p>To keep the generic argument list as clear as possible, parameters to control
the methods for De estimation are preset with meaningful default values,
which can however be modified using the <code>method_control</code> argument, e.g.,
<code>method_control = list(trace = TRUE)</code>. Supported parameters are:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>PARAMETER</strong> </td><td style="text-align: left;"> <strong>METHOD</strong> </td><td style="text-align: left;"> <strong>DESCRIPTION</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trace</code>   </td><td style="text-align: left;"> <code>FIT</code>, <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> as in <a href="stats.html#topic+nls">nls</a>; shows sum of squared residuals</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trace_vslide</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> argument to enable or disable the tracing of the vertical sliding</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>maxiter</code> </td><td style="text-align: left;"> <code>FIT</code> </td><td style="text-align: left;"> as in <a href="stats.html#topic+nls">nls</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>warnOnly</code> </td><td style="text-align: left;"> <code>FIT</code> </td><td style="text-align: left;"> as in <a href="stats.html#topic+nls">nls</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>minFactor</code> </td><td style="text-align: left;"> <code>FIT</code> </td><td style="text-align: left;"> as in <a href="stats.html#topic+nls">nls</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>correct_onset</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> The logical argument shifts the curves along the x-axis by the first channel,
as light is expected in the first channel. The default value is <code>TRUE</code>.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>show_density</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> (<em>with default</em>)
enable/disable KDE plots for MC run results. If the distribution is too narrow nothing is shown.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>show_fit</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> (<em>with default</em>)
enable/disable the plot of the fitted curve routinely obtained during the evaluation.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.MC</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> (<em>with default</em>):
This controls the number of MC runs within the sliding (assessing the possible minimum values).
The default <code>n.MC = 1000</code>. Note: This parameter is not the same as controlled by the
function argument <code>n.MC</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>vslide_range</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> or <a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
This argument sets the boundaries for a vertical curve
sliding. The argument expects a vector with an absolute minimum and a maximum (e.g., <code>c(-1000,1000)</code>).
Alternatively the values <code>NULL</code> and <code>'auto'</code> are allowed. The automatic mode detects the
reasonable vertical sliding range (<strong>recommended</strong>). <code>NULL</code> applies no vertical sliding.
The default is <code>NULL</code>.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>num_slide_windows</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> (<em>with default</em>):
This argument controls how many differently-sized windows are tested when
sliding: the higher the value (up to a maximum of 10), the more time is
spent in searching the global optimum. The default is 3, which attempts to
strike a balance between quality of the fit and computation speed.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>cores</code> </td><td style="text-align: left;"> <code>SLIDE</code> or <code>VSLIDE</code> </td><td style="text-align: left;"> <code>number</code> or <code>character</code> (<em>with default</em>): set number of cores to be allocated
for a parallel processing of the Monte-Carlo runs. The default value is <code>NULL</code> (single thread),
the recommended values is <code>'auto'</code>. An optional number (e.g., <code>cores</code> = 8) assigns a value manually.
</td>
</tr>

</table>

<p><strong>Error estimation</strong>
</p>
<p>For <strong><code>method = "FIT"</code></strong> the asymmetric error range is obtained by using the 2.5 % (lower) and
the 97.5 % (upper) quantiles of the <code class="reqn">RF_{nat}</code> curve for calculating the <code class="reqn">D_{e}</code> error range.
</p>
<p>For <strong><code>method = "SLIDE"</code></strong> and <strong><code>method = "VSLIDE"</code></strong> the error is obtained
by bootstrapping the residuals of the slid
curve to construct new natural curves for a Monte Carlo simulation. The error is returned in two
ways: (a) the standard deviation of the <code class="reqn">D_{e}</code> obtained from the MC
runs and (b) the confidence
interval using the  2.5 % (lower) and the 97.5 % (upper) quantiles. The results of the MC runs
are returned with the function output.
</p>
<p><strong>Test parameters</strong>
</p>
<p>The argument <code>test_parameters</code> allows to pass some thresholds for several test parameters,
which will be evaluated during the function run. If a threshold is set and
it is exceeded, the
test parameter status will be set to <code>"FAILED"</code>. Intentionally this parameter is not termed
<code>'rejection criteria'</code> as not all test parameters are evaluated for both methods and some parameters
are calculated but not evaluated by default. Common for all parameters are the allowed argument options
<code>NA</code> and <code>NULL</code>. If the parameter is set to <code>NA</code> the value is calculated but the
result will not be evaluated, therefore it will have no effect on the
status (<code>"OK"</code> or <code>"FAILED"</code>) of the parameter.
Setting the parameter to <code>NULL</code> disables the parameter entirely and the parameter will be
also removed from the function output. This might be useful in cases where a particular parameter
requires a long computation time. Currently supported parameters are:
</p>
<p><code>curves_ratio</code> <a href="base.html#topic+numeric">numeric</a> (default: <code>1.001</code>):
</p>
<p>The ratio of <code class="reqn">RF_{nat}</code> over <code class="reqn">RF_{reg}</code> in the range of<code class="reqn">RF_{nat}</code> of is calculated
and should not exceed the threshold value.
</p>
<p><code>intersection_ratio</code> <a href="base.html#topic+numeric">numeric</a> (default: <code>NA</code>):
</p>
<p>Calculated as absolute difference from 1 of the ratio of the integral of the normalised RF-curves,
This value indicates intersection of the RF-curves and should be close to 0 if the curves
have a similar shape. For this calculation first the corresponding time-count pair value on the RF_reg
curve is obtained using the maximum count value of the <code>RF_nat</code> curve and only this segment (fitting to
the <code>RF_nat</code> curve) on the RF_reg curve is taken for further calculating this ratio. If nothing is
found at all, <code>Inf</code> is returned.
</p>
<p><code>residuals_slope</code> <a href="base.html#topic+numeric">numeric</a> (default: <code>NA</code>; only for <code>method = "SLIDE"</code>):
</p>
<p>A linear function is fitted on the residuals after sliding.
The corresponding slope can be used to discard values as a high (positive, negative) slope
may indicate that both curves are fundamentally different and the method cannot be applied at all.
Per default the value of this parameter is calculated but not evaluated.
</p>
<p><code>curves_bounds</code> <a href="base.html#topic+numeric">numeric</a> (default: <code class="reqn">max(RF_{reg_counts})</code>:
</p>
<p>This measure uses the maximum time (x) value of the regenerated curve.
The maximum time (x) value of the natural curve cannot be larger than this value. However, although
this is not recommended the value can be changed or disabled.
</p>
<p><code>dynamic_ratio</code> <a href="base.html#topic+numeric">numeric</a> (default: <code>NA</code>):
</p>
<p>The dynamic ratio of the regenerated curve is calculated as ratio of the minimum and maximum count values.
</p>
<p><code>lambda</code>, <code>beta</code> and <code>delta.phi</code>
<a href="base.html#topic+numeric">numeric</a> (default: <code>NA</code>; <code>method = "SLIDE"</code>):
</p>
<p>The stretched exponential function suggested by Erfurt et al. (2003) describing the decay of
the RF signal, comprises several parameters that might be useful to evaluate the shape of the curves.
For <code>method = "FIT"</code> this parameter is obtained during the fitting, for <code>method = "SLIDE"</code> a
rather rough estimation is made using the function <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a> and the equation
given above. Note: As this procedure requests more computation time, setting of one of these three parameters
to <code>NULL</code> also prevents a calculation of the remaining two.
</p>


<h3>Value</h3>

<p>The function returns numerical output and an (<em>optional</em>) plot.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>
<p><code style="white-space: pre;">&#8288;[.. $data : data.frame]&#8288;</code>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Column</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the obtained equivalent dose</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DE.ERROR</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> (only <code>method = "SLIDE"</code>) standard deviation obtained from MC runs </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DE.LOWER</code> </td><td style="text-align: left;"> <code>numeric</code></td><td style="text-align: left;"> 2.5% quantile for De values obtained by MC runs </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DE.UPPER</code> </td><td style="text-align: left;"> <code>numeric</code></td><td style="text-align: left;"> 97.5% quantile for De values obtained by MC runs  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DE.STATUS</code>  </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> test parameter status</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RF_NAT.LIM</code>  </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> used <code>RF_nat</code> curve limits </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RF_REG.LIM</code> </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> used <code>RF_reg</code> curve limits</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POSITION</code> </td><td style="text-align: left;"> <code>integer</code></td><td style="text-align: left;"> (<em>optional</em>) position of the curves</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DATE</code> </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> (<em>optional</em>) measurement date</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SEQUENCE_NAME</code> </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> (<em>optional</em>) sequence name</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UID</code> </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> unique data set ID
</td>
</tr>

</table>

<p><code style="white-space: pre;">&#8288;[.. $De.MC : numeric]&#8288;</code>
</p>
<p>A <code>numeric</code> vector with all the De values obtained by the MC runs.
</p>
<p><code style="white-space: pre;">&#8288;[.. $test_parameters : data.frame]&#8288;</code>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Column</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>POSITION</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> aliquot position </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>PARAMETER</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> test parameter name </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>THRESHOLD</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> set test parameter threshold value </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VALUE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the calculated test parameter value (to be compared with the threshold)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>STATUS</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> test parameter status either <code>"OK"</code> or <code>"FAILED"</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>SEQUENCE_NAME</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> name of the sequence, so far available </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>UID</code> </td><td style="text-align: left;"> <code>character</code></td><td style="text-align: left;"> unique data set ID
</td>
</tr>

</table>

<p><code style="white-space: pre;">&#8288;[.. $fit : data.frame]&#8288;</code>
</p>
<p>An <a href="stats.html#topic+nls">nls</a> object produced by the fitting.
</p>
<p><code style="white-space: pre;">&#8288;[.. $slide : list]&#8288;</code>
</p>
<p>A <a href="base.html#topic+list">list</a> with data produced during the sliding. Some elements are previously
reported with the summary object data. List elements are:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>De</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the final De obtained with the sliding approach </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>De.MC</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> all De values obtained by the MC runs </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>residuals</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the obtained residuals for each channel of the curve </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trend.fit</code> </td><td style="text-align: left;"> <code>lm</code> </td><td style="text-align: left;"> fitting results produced by the fitting of the residuals </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RF_nat.slid</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the slid <code>RF_nat</code> curve </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>t_n.id</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the index of the t_n offset </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>I_n</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the vertical intensity offset if a vertical slide was applied </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>algorithm_error</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the vertical sliding suffers from a systematic effect induced by the used
algorithm. The returned value is the standard deviation of all obtained De values while expanding the
vertical sliding range. It can be added as systematic error to the final De error; so far wanted.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>vslide_range</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the range used for the vertical sliding </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>num_slide_windows</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> the number of windows used for the vertical sliding </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>squared_residuals</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the squared residuals (horizontal sliding)
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call (<a href="methods.html#topic+LanguageClasses">methods::language</a>-object)
</p>
<p>The output (<code>data</code>) should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>The slid IR-RF curves with the finally obtained De
</p>


<h3>Function version</h3>

<p>0.7.10
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_IRSAR.RF(): Analyse IRSAR RF measurements. Function version 0.7.10. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function assumes that there is no sensitivity change during the
measurements (natural vs. regenerated signal), which is in contrast to the
findings by Buylaert et al. (2012).
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Buylaert, J.P., Jain, M., Murray, A.S., Thomsen, K.J., Lapp, T.,
2012. IR-RF dating of sand-sized K-feldspar extracts: A test of accuracy.
Radiation Measurements 44 (5-6), 560-565. doi: 10.1016/j.radmeas.2012.06.021
</p>
<p>Erfurt, G., Krbetschek, M.R., 2003. IRSAR - A single-aliquot
regenerative-dose dating protocol applied to the infrared radiofluorescence
(IR-RF) of coarse- grain K-feldspar. Ancient TL 21, 35-42.
</p>
<p>Erfurt, G., 2003. Infrared luminescence of Pb+ centres in potassium-rich
feldspars. physica status solidi (a) 200, 429-438.
</p>
<p>Erfurt, G., Krbetschek, M.R., 2003. Studies on the physics of the infrared
radioluminescence of potassium feldspar and on the methodology of its
application to sediment dating. Radiation Measurements 37, 505-510.
</p>
<p>Erfurt, G., Krbetschek, M.R., Bortolot, V.J., Preusser, F., 2003. A fully
automated multi-spectral radioluminescence reading system for geochronometry
and dosimetry. Nuclear Instruments and Methods in Physics Research Section
B: Beam Interactions with Materials and Atoms 207, 487-499.
</p>
<p>Frouin, M., Huot, S., Kreutzer, S., Lahaye, C., Lamothe, M., Philippe, A., Mercier, N., 2017.
An improved radiofluorescence single-aliquot regenerative dose protocol for K-feldspars.
Quaternary Geochronology 38, 13-24. doi:10.1016/j.quageo.2016.11.004
</p>
<p>Kreutzer, S., Murari, M.K., Frouin, M., Fuchs, M., Mercier, N., 2017.
Always remain suspicious: a case study on tracking down a technical artefact while measuring IR-RF.
Ancient TL 35, 20‚Äì30.
</p>
<p>Murari, M.K., Kreutzer, S., Fuchs, M., 2018. Further investigations on IR-RF:
Dose recovery and correction. Radiation Measurements 120, 110‚Äì119.
doi: 10.1016/j.radmeas.2018.04.017
</p>
<p>Lapp, T., Jain, M., Thomsen, K.J., Murray, A.S., Buylaert, J.P., 2012. New
luminescence measurement facilities in retrospective dosimetry. Radiation
Measurements 47, 803-808. doi:10.1016/j.radmeas.2012.02.006
</p>
<p>Trautmann, T., 2000. A study of radioluminescence kinetics of natural
feldspar dosimeters: experiments and simulations. Journal of Physics D:
Applied Physics 33, 2304-2310.
</p>
<p>Trautmann, T., Krbetschek, M.R., Dietrich, A., Stolz, W., 1998.
Investigations of feldspar radioluminescence: potential for a new dating
technique. Radiation Measurements 29, 421-425.
</p>
<p>Trautmann, T., Krbetschek, M.R., Dietrich, A., Stolz, W., 1999. Feldspar
radioluminescence: a new dating method and its physical background. Journal
of Luminescence 85, 45-58.
</p>
<p>Trautmann, T., Krbetschek, M.R., Stolz, W., 2000. A systematic study of the
radioluminescence properties of single feldspar grains. Radiation
Measurements 32, 685-690.
</p>
<p>** Further reading**
</p>
<p>Murari, M.K., Kreutzer, S., King, G.E., Frouin, M., Tsukamoto, S., Schmidt, C., Lauer, T.,
Klasen, N., Richter, D., Friedrich, J., Mercier, N., Fuchs, M., 2021.
Infrared radiofluorescence (IR-RF) dating: A review. Quaternary Geochronology 64,
101155. doi: 10.1016/j.quageo.2021.101155
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>,
<a href="stats.html#topic+nls">nls</a>, <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>, <code>parallel::mclapply</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.RLum.Analysis, envir = environment())

##(1) perform analysis using the method 'FIT'
results &lt;- analyse_IRSAR.RF(object = IRSAR.RF.Data)

##show De results and test paramter results
get_RLum(results, data.object = "data")
get_RLum(results, data.object = "test_parameters")

##(2) perform analysis using the method 'SLIDE'
results &lt;- analyse_IRSAR.RF(object = IRSAR.RF.Data, method = "SLIDE", n.MC = 1)

## Not run: 
##(3) perform analysis using the method 'SLIDE' and method control option
## 'trace
results &lt;- analyse_IRSAR.RF(
 object = IRSAR.RF.Data,
 method = "SLIDE",
 method_control = list(trace = TRUE))

## End(Not run)

</code></pre>

<hr>
<h2 id='analyse_pIRIRSequence'>Analyse post-IR IRSL measurement sequences</h2><span id='topic+analyse_pIRIRSequence'></span>

<h3>Description</h3>

<p>The function performs an analysis of post-IR IRSL sequences
including curve
fitting on <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_pIRIRSequence(
  object,
  signal.integral.min,
  signal.integral.max,
  background.integral.min,
  background.integral.max,
  dose.points = NULL,
  sequence.structure = c("TL", "IR50", "pIRIR225"),
  plot = TRUE,
  plot_singlePanels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_pIRIRSequence_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or <a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects (<strong>required</strong>):
input object containing data for analysis.
If a <a href="base.html#topic+list">list</a> is provided the functions tries to iterate over each element
in the list.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_signal.integral.min">signal.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
lower bound of the signal integral. Provide this value as vector for different
integration limits for the different IRSL curves.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_signal.integral.max">signal.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
upper bound of the signal integral. Provide this value as vector for different
integration limits for the different IRSL curves.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_background.integral.min">background.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
lower bound of the background integral. Provide this value as vector for
different integration limits for the different IRSL curves.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_background.integral.max">background.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
upper bound of the background integral. Provide this value as vector for
different integration limits for the different IRSL curves.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_dose.points">dose.points</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
a numeric vector containing the dose points values. Using this argument overwrites dose point
values in the signal curves.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_sequence.structure">sequence.structure</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> <a href="base.html#topic+character">character</a> (<em>with default</em>):
specifies the general sequence structure. Allowed values are <code>"TL"</code> and
any <code>"IR"</code> combination (e.g., <code>"IR50"</code>,<code>"pIRIR225"</code>).
Additionally a parameter <code>"EXCLUDE"</code> is allowed to exclude curves from
the analysis (Note: If a preheat without PMT measurement is used, i.e.
preheat as none TL, remove the TL step.)</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable plotting of the results in a single windows for each plot.
Ignored if <code>plot = FALSE</code>.</p>
</td></tr>
<tr><td><code id="analyse_pIRIRSequence_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to
<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> and <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>. Furthermore, the
arguments <code>main</code> (headers), <code>log</code> (IRSL curves), <code>cex</code> (control
the size) and <code>mtext.outer</code> (additional text on the plot area) can be passed to influence the plotting. If the input
is a list, <code>main</code> can be passed as <a href="base.html#topic+vector">vector</a> or <a href="base.html#topic+list">list</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>To allow post-IR IRSL protocol (Thomsen et al., 2008) measurement
analyses, this function has been written as extended wrapper for function
<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>, thus facilitating an entire sequence analysis in
one run. With this, its functionality is strictly limited by the
functionality provided by  <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>.
</p>
<p><strong>Defining the sequence structure</strong>
</p>
<p>The argument <code>sequence.structure</code> expects a shortened pattern of your sequence structure and was
mainly introduced to ease the use of the function. For example: If your measurement data contains
the following curves: <code>TL</code>, <code>IRSL</code>, <code>IRSL</code>, <code>TL</code>, <code>IRSL</code>, <code>IRSL</code>, the sequence pattern in <code>sequence.structure</code>
becomes <code>c('TL', 'IRSL', 'IRSL')</code>. The second part of your sequence for one cycle should be
similar and can be discarded. If this is not the case (e.g., additional hotbleach) such curves
have to be removed before using the function.
</p>
<p><strong>If the input is a <code>list</code></strong>
</p>
<p>If the input is a list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects, every argument
can be provided as list to allow
for different sets of parameters for every single input element.
For further information see <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>.
</p>


<h3>Value</h3>

<p>Plots (<em>optional</em>) and an <a href="#topic+RLum.Results-class">RLum.Results</a> object is
returned containing the following elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>DATA.OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>DESCRIPTION</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$data</code> : </td><td style="text-align: left;">  <code>data.frame</code> </td><td style="text-align: left;"> Table with De values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$LnLxTnTx.table</code> : </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> with the <code>LnLxTnTx</code> values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$rejection.criteria</code> : </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> rejection criteria </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$Formula</code> : </td><td style="text-align: left;"> <a href="base.html#topic+list">list</a> </td><td style="text-align: left;"> Function used for fitting of the dose response curve </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$call</code> : </td><td style="text-align: left;"> <a href="base.html#topic+call">call</a> </td><td style="text-align: left;"> the original function call
</td>
</tr>

</table>

<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.2.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_pIRIRSequence(): Analyse post-IR IRSL measurement sequences. Function version 0.2.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Best graphical output can be achieved by using the function <code>pdf</code>
with the following options:
</p>
<p><code>pdf(file = "&lt;YOUR FILENAME&gt;", height = 18, width = 18)</code>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Murray, A.S., Wintle, A.G., 2000. Luminescence dating of quartz
using an improved single-aliquot regenerative-dose protocol. Radiation
Measurements 32, 57-73. <a href="https://doi.org/10.1016/S1350-4487%2899%2900253-X">doi:10.1016/S1350-4487(99)00253-X</a>
</p>
<p>Thomsen, K.J., Murray, A.S., Jain, M., Boetter-Jensen, L., 2008. Laboratory
fading rates of various luminescence signals from feldspar-rich sediment
extracts. Radiation Measurements 43, 1474-1486.
<a href="https://doi.org/10.1016/j.radmeas.2008.06.002">doi:10.1016/j.radmeas.2008.06.002</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>, <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>, <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>,
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a> <a href="#topic+get_RLum">get_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

### NOTE: For this example existing example data are used. These data are non pIRIR data.
###
##(1) Compile example data set based on existing example data (SAR quartz measurement)
##(a) Load example data
data(ExampleData.BINfileData, envir = environment())

##(b) Transform the values from the first position in a RLum.Analysis object
object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)

##(c) Grep curves and exclude the last two (one TL and one IRSL)
object &lt;- get_RLum(object, record.id = c(-29,-30))

##(d) Define new sequence structure and set new RLum.Analysis object
sequence.structure  &lt;- c(1,2,2,3,4,4)
sequence.structure &lt;- as.vector(sapply(seq(0,length(object)-1,by = 4),
                                       function(x){sequence.structure + x}))

object &lt;-  sapply(1:length(sequence.structure), function(x){
  object[[sequence.structure[x]]]
})

object &lt;- set_RLum(class = "RLum.Analysis", records = object, protocol = "pIRIR")

##(2) Perform pIRIR analysis (for this example with quartz OSL data!)
## Note: output as single plots to avoid problems with this example
results &lt;- analyse_pIRIRSequence(object,
     signal.integral.min = 1,
     signal.integral.max = 2,
     background.integral.min = 900,
     background.integral.max = 1000,
     fit.method = "EXP",
     sequence.structure = c("TL", "pseudoIRSL1", "pseudoIRSL2"),
     main = "Pseudo pIRIR data set based on quartz OSL",
     plot_singlePanels = TRUE)


##(3) Perform pIRIR analysis (for this example with quartz OSL data!)
## Alternative for PDF output, uncomment and complete for usage
## Not run: 
tempfile &lt;- tempfile(fileext = ".pdf")
pdf(file = tempfile, height = 18, width = 18)
  results &lt;- analyse_pIRIRSequence(object,
         signal.integral.min = 1,
         signal.integral.max = 2,
         background.integral.min = 900,
         background.integral.max = 1000,
         fit.method = "EXP",
         main = "Pseudo pIRIR data set based on quartz OSL")

  dev.off()

## End(Not run)

</code></pre>

<hr>
<h2 id='analyse_portableOSL'>Analyse portable CW-OSL measurements</h2><span id='topic+analyse_portableOSL'></span>

<h3>Description</h3>

<p>The function analyses CW-OSL curve data produced by a SUERC portable OSL reader and
produces a combined plot of OSL/IRSL signal intensities, OSL/IRSL depletion ratios
and the IRSL/OSL ratio.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_portableOSL(
  object,
  signal.integral = NULL,
  invert = FALSE,
  normalise = FALSE,
  mode = "profile",
  coord = NULL,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_portableOSL_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>): <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object produced by <a href="#topic+read_PSL2R">read_PSL2R</a>.
The input can be a <a href="base.html#topic+list">list</a> of such objects, in such case each input is treated as a separate sample
and the results are merged.</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_signal.integral">signal.integral</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): A vector of two values specifying the lower and upper channel used to calculate the OSL/IRSL signal. Can be provided in form of <code>c(1, 5)</code> or <code>1:5</code>.</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_invert">invert</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): <code>TRUE</code> flip the plot the data in reverse order.</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_normalise">normalise</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): <code>TRUE</code> to normalise the OSL/IRSL signals
to the <em>mean</em> of all corresponding data curves.</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_mode">mode</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): defines the analysis mode, allowed
are <code>"profile"</code> (the default) and <code>"surface"</code> for surface interpolation. If you select
something else, nothing will be plotted (similar to <code>plot = FALSE</code>).</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_coord">coord</code></td>
<td>
<p><a href="base.html#topic+list">list</a> <a href="base.html#topic+matrix">matrix</a> (<em>optional</em>): a list or matrix of the same length as
number of samples measured with coordinates for the sampling positions. Coordinates
are expected to be provided in meter (unit: m).
Expected are x and y coordinates, e.g.,
<code style="white-space: pre;">&#8288;coord = list(samp1 = c(0.1, 0.2)&#8288;</code>. If you have not measured x coordinates, please x should be 0.</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_portableOSL_+3A_...">...</code></td>
<td>
<p>other parameters to be passed to modify the plot output.
Supported are <code>run</code> to provide the run name ,
if the input is a <code>list</code>, this is set automatically. Further plot parameters are
<code>surface_values</code> (<a href="base.html#topic+character">character</a> with value to plot), <code>legend</code> (<code>TRUE</code>/<code>FALSE</code>), <code>col_ramp</code> (for
surface mode), <code>contour</code> (contour lines <code>TRUE</code>/<code>FALSE</code> in surface mode), <code>grid</code> (<code>TRUE</code>/<code>FALSE</code>), <code>col</code>, <code>pch</code> (for profile mode), <code>xlim</code> (a name <a href="base.html#topic+list">list</a> for profile mode), <code>ylim</code>,
<code>zlim</code> (surface mode only), <code>ylab</code>, <code>xlab</code>, <code>zlab</code> (here x-axis labelling), <code>main</code>, <code>bg_img</code> (for
profile mode background image, usually a profile photo; should be a raster object),
<code>bg_img_positions</code> (a vector with the four corner positions, cf. <a href="graphics.html#topic+rasterImage">graphics::rasterImage</a>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function only works with <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects produced by <a href="#topic+read_PSL2R">read_PSL2R</a>.
It further assumes (or rather requires) an equal amount of OSL and IRSL curves that
are pairwise combined for calculating the IRSL/OSL ratio. For calculating the depletion ratios
the cumulative signal of the last n channels (same number of channels as specified
by <code>signal.integral</code>) is divided by cumulative signal of the first n channels (<code>signal.integral</code>).
</p>
<p><strong>Note:  The function assumes the following sequence pattern: <code style="white-space: pre;">&#8288;DARK COUNT&#8288;</code>, <code>IRSL</code>, <code style="white-space: pre;">&#8288;DARK COUNT&#8288;</code>, <code>BSL</code>, <code style="white-space: pre;">&#8288;DARK COUNT&#8288;</code>. If you have written a different sequence, the analysis function will (likely) not work!</strong>.
</p>
<p><strong>Signal processing</strong>
The function processes the signals as follows: <code>BSL</code> and <code>IRSL</code> signals are extracted using the
chosen signal integral, dark counts are taken in full.
</p>
<p><strong>Working with coordinates</strong>
Usually samples are taken from a profile with a certain stratigraphy. In the past the function
calculated an index. With this newer version, you have two option of passing on xy-coordinates
to the function:
</p>

<ul>
<li><p> (1) Add coordinates to the sample name during measurement. The form is rather
strict and has to follow the scheme <code style="white-space: pre;">&#8288;_x:&lt;number&gt;|y:&lt;number&gt;&#8288;</code>. Example:
<code>sample_x:0.2|y:0.4</code>.
</p>
</li>
<li><p> (2) Alternatively, you can provide a <a href="base.html#topic+list">list</a> or <a href="base.html#topic+matrix">matrix</a> with the sample coordinates.
Example: <code>coord = list(c(0.2, 1), c(0.3,1.2))</code>
</p>
</li></ul>

<p>Please note that the unit is meter (m) and the function expects always xy-coordinates.
The latter one is useful for surface interpolations. If you have measured a profile where
the x-coordinates to not measure, x-coordinates should be 0.
</p>


<h3>Value</h3>

<p>Returns an S4 <a href="#topic+RLum.Results-class">RLum.Results</a> object with the following elements:
</p>
<p><code style="white-space: pre;">&#8288;$data&#8288;</code><br />
<code>.. $summary</code>: <a href="base.html#topic+data.frame">data.frame</a> with the results<br />
<code>.. $data</code>: <a href="base.html#topic+list">list</a> with the <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects<br />
<code>.. $args</code>: <a href="base.html#topic+list">list</a> the input arguments
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Burow, C., Kreutzer, S., 2025. analyse_portableOSL(): Analyse portable CW-OSL measurements. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany), Sebastian Kreutzer,
Institute of Geography, Ruprecht-Karl University of Heidelberg, Germany
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+read_PSL2R">read_PSL2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## example profile plot
# (1) load example data set
data("ExampleData.portableOSL", envir = environment())

# (2) merge and plot all RLum.Analysis objects
merged &lt;- merge_RLum(ExampleData.portableOSL)
plot_RLum(
 object = merged,
 combine = TRUE,
 records_max = 5,
 legend.pos = "outside")
merged

# (3) analyse and plot
results &lt;- analyse_portableOSL(
  merged,
  signal.integral = 1:5,
  invert = FALSE,
  normalise = TRUE)
get_RLum(results)

</code></pre>

<hr>
<h2 id='analyse_SAR.CWOSL'>Analyse SAR CW-OSL measurements</h2><span id='topic+analyse_SAR.CWOSL'></span>

<h3>Description</h3>

<p>The function performs a SAR CW-OSL analysis on an
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object including growth curve fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_SAR.CWOSL(
  object,
  signal.integral.min = NA,
  signal.integral.max = NA,
  background.integral.min = NA,
  background.integral.max = NA,
  OSL.component = NULL,
  rejection.criteria = list(),
  dose.points = NULL,
  trim_channels = FALSE,
  mtext.outer = "",
  plot = TRUE,
  plot_onePage = FALSE,
  plot_singlePanels = FALSE,
  onlyLxTxTable = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_SAR.CWOSL_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
input object containing data for analysis, alternatively a <a href="base.html#topic+list">list</a> of
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects can be provided. The object should contain <strong>only</strong> curves
considered part of the SAR protocol (see Details).</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_signal.integral.min">signal.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
lower bound of the signal integral. Can be a <a href="base.html#topic+list">list</a> of <a href="base.html#topic+integer">integer</a>s, if <code>object</code> is
of type <a href="base.html#topic+list">list</a>. If the input is vector (e.g., <code>c(1,2)</code>) the 2nd value will be interpreted
as the minimum signal integral for the <code>Tx</code> curve. Can be set to <code>NA</code>, in this
case no integrals are taken into account.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_signal.integral.max">signal.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
upper bound of the signal integral. Can be a <a href="base.html#topic+list">list</a> of <a href="base.html#topic+integer">integer</a>s, if <code>object</code> is
of type <a href="base.html#topic+list">list</a>. If the input is vector (e.g., <code>c(1,2)</code>) the 2nd value will be interpreted
as the maximum signal integral for the <code>Tx</code> curve. Can be set to <code>NA</code>, in this
case no integrals are taken into account.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_background.integral.min">background.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
lower bound of the background integral. Can be a <a href="base.html#topic+list">list</a> of <a href="base.html#topic+integer">integer</a>s, if <code>object</code> is
of type <a href="base.html#topic+list">list</a>. If the input is vector (e.g., <code>c(1,2)</code>) the 2nd value will be interpreted
as the minimum background integral for the <code>Tx</code> curve. Can be set to <code>NA</code>, in this
case no integrals are taken into account.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_background.integral.max">background.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
upper bound of the background integral. Can be a <a href="base.html#topic+list">list</a> of <a href="base.html#topic+integer">integer</a>s, if <code>object</code> is
of type <a href="base.html#topic+list">list</a>. If the input is vector (e.g., <code>c(1,2)</code>) the 2nd value will be interpreted
as the maximum background integral for the <code>Tx</code> curve. Can be set to <code>NA</code>, in this
case no integrals are taken into account.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_osl.component">OSL.component</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+integer">integer</a> (<em>optional</em>): s single index
or a <a href="base.html#topic+character">character</a> defining the signal component to be evaluated.
It requires that the object was processed by <code>OSLdecomposition::RLum.OSL_decomposition</code>.
This argument can either be the name of the OSL component assigned by
<code>OSLdecomposition::RLum.OSL_global_fitting</code> or the index in the descending
order of decay rates. Then <code>"1"</code> selects the fastest decaying component, <code>"2"</code>
the second fastest and so on. Can be a <a href="base.html#topic+list">list</a> of <a href="base.html#topic+integer">integer</a>s or strings (or mixed)
If object is a <a href="base.html#topic+list">list</a> and this parameter is provided as <a href="base.html#topic+list">list</a> it alternates over
the elements (aliquots) of the object list, e.g., <code>list(1,2)</code> processes the first
aliquot with component <code>1</code> and the second aliquot with component <code>2</code>.
<code>NULL</code> does not process any component.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_rejection.criteria">rejection.criteria</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>with default</em>):
provide a <em>named</em> list and set rejection criteria in <strong>percentage</strong>
for further calculation. Can be a <a href="base.html#topic+list">list</a> in a <a href="base.html#topic+list">list</a>, if <code>object</code> is of type <a href="base.html#topic+list">list</a>.
Note: If an <em>unnamed</em> <a href="base.html#topic+list">list</a> is provided the new settings are ignored!
</p>
<p>Allowed arguments are <code>recycling.ratio</code>, <code>recuperation.rate</code>,
<code>palaeodose.error</code>, <code>testdose.error</code>, <code>exceed.max.regpoint = TRUE/FALSE</code>,
<code>recuperation_reference = "Natural"</code> (or any other dose point, e.g., <code>"R1"</code>).
Example: <code>rejection.criteria = list(recycling.ratio = 10)</code>.
Per default all numerical values are set to 10, <code>exceed.max.regpoint = TRUE</code>.
Every criterion can be set to <code>NA</code>. In this value are calculated, but not considered, i.e.
the RC.Status becomes always <code>'OK'</code></p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_dose.points">dose.points</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
a numeric vector containing the dose points values. Using this argument
overwrites dose point values extracted from other data. Can be a <a href="base.html#topic+list">list</a> of
<a href="base.html#topic+numeric">numeric</a> vectors, if <code>object</code> is of type <a href="base.html#topic+list">list</a></p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_trim_channels">trim_channels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): trim channels per record category
to the lowest number of channels in the category by using <a href="#topic+trim_RLum.Data">trim_RLum.Data</a>.
Applies only to <code>OSL</code> and <code>IRSL</code> curves. For a more granular control use <a href="#topic+trim_RLum.Data">trim_RLum.Data</a>
before passing the input object.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_mtext.outer">mtext.outer</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
option to provide an outer margin <code>mtext</code>. Can be a <a href="base.html#topic+list">list</a> of <a href="base.html#topic+character">character</a>s,
if <code>object</code> is of type <a href="base.html#topic+list">list</a></p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_plot_onepage">plot_onePage</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable one page
plot output.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>) or <a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
single plot output (<code>TRUE/FALSE</code>) to allow for plotting the results in single plot windows.
If a <a href="base.html#topic+numeric">numeric</a> vector is provided the plots can be selected individually, i.e.
<code>plot_singlePanels = c(1,2,3,4)</code> will plot the TL and Lx, Tx curves but
not the legend (5) or the
growth curve (6), (7) and (8) belong to rejection criteria plots. Requires
<code>plot = TRUE</code>.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_onlylxtxtable">onlyLxTxTable</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): If <code>TRUE</code> the dose response
curve fitting and plotting is skipped.
This allows to get hands on the <code>Lx/Tx</code> table for large datasets
without the need for a curve fitting.</p>
</td></tr>
<tr><td><code id="analyse_SAR.CWOSL_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> or <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>
(supported: <code>background.count.distribution</code>, <code>sigmab</code>, <code>sig0</code>).
<strong>Please note</strong> that if you consider to use the early light subtraction
method you should provide your own <code>sigmab</code> value!</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function performs an analysis for a standard SAR protocol measurements
introduced by Murray and Wintle (2000) with CW-OSL curves. For the
calculation of the <code>Lx/Tx</code> value the function <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a> is
used. For <strong>changing the way the Lx/Tx error is calculated</strong> use the argument
<code>background.count.distribution</code> and <code>sigmab</code>, which will be passed to the function
<a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>.
</p>
<p><strong>What is part of a SAR sequence?</strong>
</p>
<p>The function is rather picky when it comes down to accepted curve input (OSL,IRSL,...) and structure.
A SAR sequence is basically a set of <code class="reqn">L_{x}/T_{x}</code> curves. Hence, every 2nd curve
is considered a shine-down curve related to the test dose. It also means that the number of
curves for <code class="reqn">L_{x}</code> has to be equal to the number of <code class="reqn">T_{x}</code> curves, and that
hot-bleach curves <strong>do not</strong> belong into a SAR sequence; at least not for the analysis.
Other curves allowed and processed are preheat curves, or preheat curves measured as TL, and
irradiation curves. The later one indicates the duration of the irradiation, the
dose and test dose points, e.g., as part of XSYG files.
</p>
<p><strong>Argument <code>object</code> is of type <code>list</code></strong>
</p>
<p>If the argument <code>object</code> is of type <a href="base.html#topic+list">list</a> containing <strong>only</strong>
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects, the function re-calls itself as often as elements
are in the list. This is useful if an entire measurement wanted to be analysed without
writing separate for-loops. To gain in full control of the parameters (e.g., <code>dose.points</code>) for
every aliquot (corresponding to one <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object in the list), in
this case the arguments can be provided as <a href="base.html#topic+list">list</a>. This <code>list</code> should
be of similar length as the <code>list</code> provided with the argument <code>object</code>,
otherwise the function will create an own list of the requested length.
Function output will be just one single <a href="#topic+RLum.Results-class">RLum.Results</a> object.
</p>
<p>Please be careful when using this option. It may allow a fast an efficient data analysis, but
the function may also break with an unclear error message, due to wrong input data.
</p>
<p><strong>Working with IRSL data</strong>
</p>
<p>The function was originally designed to work just for 'OSL' curves,
following the principles of the SAR protocol. An IRSL measurement protocol
may follow this procedure, e.g., post-IR IRSL protocol (Thomsen et al.,
2008). Therefore this functions has been enhanced to work with IRSL data,
however, the function is only capable of analysing curves that follow the
SAR protocol structure, i.e., to analyse a post-IR IRSL protocol, curve data
have to be pre-selected by the user to fit the standards of the SAR
protocol, i.e., Lx,Tx,Lx,Tx and so on.
</p>
<p>Example: Imagine the measurement contains <code>pIRIR50</code> and <code>pIRIR225</code> IRSL curves.
Only one curve type can be analysed at the same time: The <code>pIRIR50</code> curves or
the <code>pIRIR225</code> curves.
</p>
<p><strong>Supported rejection criteria</strong>
</p>
<p><code style="white-space: pre;">&#8288;[recycling.ratio]&#8288;</code>: calculated for every repeated regeneration dose point.
</p>
<p><code style="white-space: pre;">&#8288;[recuperation.rate]&#8288;</code>: recuperation rate calculated by comparing the
<code>Lx/Tx</code> values of the zero regeneration point with the <code>Ln/Tn</code> value (the <code>Lx/Tx</code>
ratio of the natural signal). For methodological background see Aitken and
Smith (1988). As a variant with the argument <code>recuperation_reference</code> another dose point can be
selected as reference instead of <code>Ln/Tn</code>.
</p>
<p><code style="white-space: pre;">&#8288;[testdose.error]&#8288;</code>: set the allowed error for the test dose, which per
default should not exceed 10%. The test dose error is calculated as <code>Tx_net.error/Tx_net</code>.
The calculation of the <code class="reqn">T_{n}</code> error is detailed in <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>.
</p>
<p><code style="white-space: pre;">&#8288;[palaeodose.error]&#8288;</code>: set the allowed error for the De value, which per
default should not exceed 10%.
</p>
<p><strong>Irradiation times</strong>
</p>
<p>The function makes two attempts to extra irradiation data (dose points)
automatically from the input object, if the argument <code>dose.points</code> was not
set (aka set to <code>NULL</code>).
</p>

<ol>
<li><p> It searches in every curve for an info object called <code>IRR_TIME</code>. If this was set, any value
set here is taken as dose point.
</p>
</li>
<li><p> If the object contains curves of type <code>irradiation</code>, the function tries to
use this information to assign these values to the curves. However, the function
does <strong>not</strong> overwrite values preset in <code>IRR_TIME</code>.
</p>
</li></ol>



<h3>Value</h3>

<p>A plot (<em>optional</em>) and an <a href="#topic+RLum.Results-class">RLum.Results</a> object is
returned containing the following elements:
</p>
<table role = "presentation">
<tr><td><code>data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> containing De-values, De-error and further parameters</p>
</td></tr>
<tr><td><code>LnLxTnTx.values</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> of all calculated Lx/Tx values including signal,
background counts and the dose points</p>
</td></tr>
<tr><td><code>rejection.criteria</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> with values that might by used as rejection criteria.
<code>NA</code> is produced if no R0 dose point exists.</p>
</td></tr>
<tr><td><code>Formula</code></td>
<td>
<p><a href="stats.html#topic+formula">formula</a> formula that have been used for the growth curve fitting</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.10.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_SAR.CWOSL(): Analyse SAR CW-OSL measurements. Function version 0.10.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function must not be mixed up with the function
<a href="#topic+Analyse_SAR.OSLdata">Analyse_SAR.OSLdata</a>, which works with
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects.
</p>
<p><strong>The function currently does support only 'OSL', 'IRSL' and 'POSL' data!</strong>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Aitken, M.J. and Smith, B.W., 1988. Optical dating: recuperation
after bleaching. Quaternary Science Reviews 7, 387-393.
</p>
<p>Duller, G., 2003. Distinguishing quartz and feldspar in single grain
luminescence measurements. Radiation Measurements, 37 (2), 161-165.
</p>
<p>Murray, A.S. and Wintle, A.G., 2000. Luminescence dating of quartz using an
improved single-aliquot regenerative-dose protocol. Radiation Measurements
32, 57-73.
</p>
<p>Thomsen, K.J., Murray, A.S., Jain, M., Boetter-Jensen, L., 2008. Laboratory
fading rates of various luminescence signals from feldspar-rich sediment
extracts. Radiation Measurements 43, 1474-1486.
doi:10.1016/j.radmeas.2008.06.002
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>, <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>,
<a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
##ExampleData.BINfileData contains two BINfileData objects
##CWOSL.SAR.Data and TL.SAR.Data
data(ExampleData.BINfileData, envir = environment())

##transform the values from the first position in a RLum.Analysis object
object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)

##perform SAR analysis and set rejection criteria
results &lt;- analyse_SAR.CWOSL(
object = object,
signal.integral.min = 1,
signal.integral.max = 2,
background.integral.min = 900,
background.integral.max = 1000,
log = "x",
fit.method = "EXP",
rejection.criteria = list(
  recycling.ratio = 10,
  recuperation.rate = 10,
  testdose.error = 10,
  palaeodose.error = 10,
  recuperation_reference = "Natural",
  exceed.max.regpoint = TRUE)
)

##show De results
get_RLum(results)

##show LnTnLxTx table
get_RLum(results, data.object = "LnLxTnTx.table")

</code></pre>

<hr>
<h2 id='Analyse_SAR.OSLdata'>Analyse SAR CW-OSL measurements (deprecated)</h2><span id='topic+Analyse_SAR.OSLdata'></span>

<h3>Description</h3>

<p>The function analyses SAR CW-OSL curve data and provides a summary of the
measured data for every position. The output of the function is optimised
for SAR OSL measurements on quartz.
</p>
<p>The function works only for standard SAR protocol measurements introduced by
Murray and Wintle (2000) with CW-OSL curves. For the calculation of the
Lx/Tx value the function <a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a> is used.
</p>
<p><strong>Provided rejection criteria</strong>
</p>
<p><code style="white-space: pre;">&#8288;[recyling ratio]&#8288;</code>: calculated for every repeated regeneration dose point.
</p>
<p><code style="white-space: pre;">&#8288;[recuperation]&#8288;</code>: recuperation rate calculated by comparing the <code>Lx/Tx</code> values of the zero
regeneration point with the <code>Ln/Tn</code> value (the <code>Lx/Tx</code> ratio of the natural
signal). For methodological background see Aitken and Smith (1988)
</p>
<p><code style="white-space: pre;">&#8288;[IRSL/BOSL]&#8288;</code>: the integrated counts (<code>signal.integral</code>) of an
IRSL curve are compared to the integrated counts of the first regenerated
dose point. It is assumed that IRSL curves got the same dose as the first
regenerated dose point. <strong>Note:</strong> This is not the IR depletion ratio
described by Duller (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Analyse_SAR.OSLdata(
  input.data,
  signal.integral,
  background.integral,
  position,
  run,
  set,
  dtype,
  keep.SEL = FALSE,
  info.measurement = "unknown measurement",
  output.plot = FALSE,
  plot_singlePanels = FALSE,
  cex.global = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Analyse_SAR.OSLdata_+3A_input.data">input.data</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
input data from a Ris√∏ BIN file, produced by the function <a href="#topic+read_BIN2R">read_BIN2R</a>.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_signal.integral">signal.integral</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<strong>required</strong>):
channels used for the signal integral, e.g. <code>signal.integral=c(1:2)</code></p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_background.integral">background.integral</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<strong>required</strong>):
channels used for the background integral, e.g. <code>background.integral=c(85:100)</code></p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_position">position</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
reader positions that want to be analysed (e.g. <code>position=c(1:48)</code>.
Empty positions are automatically omitted. If no value is given all
positions are analysed by default.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_run">run</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
range of runs used for the analysis. If no value is given the range of the
runs in the sequence is deduced from the <code>Risoe.BINfileData</code> object.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_set">set</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
range of sets used for the analysis. If no value is given the range of the
sets in the sequence is deduced from the <code>Risoe.BINfileData</code> object.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_dtype">dtype</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
allows to further limit the curves by their data type (<code>DTYPE</code>),
e.g., <code>dtype = c("Natural", "Dose")</code> limits the curves to this two data types.
By default all values are allowed.
See <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> for allowed data types.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_keep.sel">keep.SEL</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (default):
option allowing to use the <code>SEL</code> element of the <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> manually.
<strong>NOTE:</strong> In this case any limitation provided by <code>run</code>, <code>set</code> and <code>dtype</code>
are ignored!</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_info.measurement">info.measurement</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide information about the measurement on the plot
output (e.g. name of the BIN or BINX file).</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_output.plot">output.plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
single plot output (<code>TRUE/FALSE</code>) to allow for plotting the results in
single plot windows. Requires <code>output.plot = TRUE</code>.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_cex.global">cex.global</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
global scaling factor.</p>
</td></tr>
<tr><td><code id="Analyse_SAR.OSLdata_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a> (supported: <code>background.count.distribution</code>, <code>sigmab</code>,
<code>sig0</code>; e.g., for instrumental error) and can be used to adjust the plot.
Supported&quot; <code>mtext</code>, <code>log</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot (<em>optional</em>) and <a href="base.html#topic+list">list</a> is returned containing the
following elements:
</p>
<table role = "presentation">
<tr><td><code>LnLxTnTx</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> of all calculated Lx/Tx values including signal, background counts and the dose points.</p>
</td></tr>
<tr><td><code>RejectionCriteria</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> with values that might by used as rejection criteria. NA is produced if no R0 dose point exists.</p>
</td></tr>
<tr><td><code>SARParameters</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> of additional measurement parameters obtained from the BIN file, e.g. preheat or read temperature
(not valid for all types of measurements).</p>
</td></tr>
</table>


<h3>Function version</h3>

<p>0.2.18
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Fuchs, M.C., 2025. Analyse_SAR.OSLdata(): Analyse SAR CW-OSL measurements (deprecated). Function version 0.2.18. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Rejection criteria are calculated but not considered during the
analysis to discard values.
</p>
<p><strong>The analysis of IRSL data is not directly supported</strong>. You may want to
consider using the functions <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> or
<a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a> instead.
</p>
<p><strong>The development of this function will not be continued. We recommend to use the function <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> or instead.</strong>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Margret C. Fuchs, HZDR, Freiberg (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Aitken, M.J. and Smith, B.W., 1988. Optical dating: recuperation
after bleaching. Quaternary Science Reviews 7, 387-393.
</p>
<p>Duller, G., 2003. Distinguishing quartz and feldspar in single grain
luminescence measurements. Radiation Measurements, 37 (2), 161-165.
</p>
<p>Murray, A.S. and Wintle, A.G., 2000. Luminescence dating of quartz using an
improved single-aliquot regenerative-dose protocol. Radiation Measurements
32, 57-73.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_OSLLxTxRatio">calc_OSLLxTxRatio</a>, <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load data
data(ExampleData.BINfileData, envir = environment())

##analyse data
suppressWarnings( # silence the deprecation warning
output &lt;- Analyse_SAR.OSLdata(input.data = CWOSL.SAR.Data,
                              signal.integral = c(1:5),
                              background.integral = c(900:1000),
                              position = c(1:1),
                              output.plot = TRUE)
)

##combine results relevant for further analysis
output.SAR &lt;- data.frame(Dose = output$LnLxTnTx[[1]]$Dose,
                         LxTx = output$LnLxTnTx[[1]]$LxTx,
                         LxTx.Error = output$LnLxTnTx[[1]]$LxTx.Error)
output.SAR

</code></pre>

<hr>
<h2 id='analyse_SAR.TL'>Analyse SAR TL measurements</h2><span id='topic+analyse_SAR.TL'></span>

<h3>Description</h3>

<p>The function performs a SAR TL analysis on a
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object including growth curve fitting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_SAR.TL(
  object,
  object.background,
  signal.integral.min,
  signal.integral.max,
  integral_input = "channel",
  sequence.structure = c("PREHEAT", "SIGNAL", "BACKGROUND"),
  rejection.criteria = list(recycling.ratio = 10, recuperation.rate = 10),
  dose.points,
  log = "",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="analyse_SAR.TL_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or a <a href="base.html#topic+list">list</a> of such objects (<strong>required</strong>) :
input object containing data for analysis</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_object.background">object.background</code></td>
<td>
<p>currently not used</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_signal.integral.min">signal.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
requires the channel number for the lower signal integral bound
(e.g. <code>signal.integral.min = 100</code>)</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_signal.integral.max">signal.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
requires the channel number for the upper signal integral bound
(e.g. <code>signal.integral.max = 200</code>)</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_integral_input">integral_input</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
defines the input for the arguments <code>signal.integral.min</code> and
<code>signal.integral.max</code>. These limits can be either provided <code>'channel'</code>
number (the default) or <code>'temperature'</code>. If <code>'temperature'</code> is chosen, the
best matching channel is selected.</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_sequence.structure">sequence.structure</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> <a href="base.html#topic+character">character</a> (<em>with default</em>):
specifies the general sequence structure. Three steps are allowed
(<code>"PREHEAT"</code>, <code>"SIGNAL"</code>, <code>"BACKGROUND"</code>), in addition a
parameter <code>"EXCLUDE"</code>. This allows excluding TL curves which are not
relevant for the protocol analysis.  (<strong>Note:</strong> No TL are removed by default)</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_rejection.criteria">rejection.criteria</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>with default</em>):
list containing rejection criteria in percentage for the calculation.</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_dose.points">dose.points</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
option set dose points manually</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
a character string which contains <code>"x"</code> if the x-axis is to be logarithmic,
<code>"y"</code> if the y axis is to be logarithmic and <code>"xy"</code> or <code>"yx"</code> if both axes
are to be logarithmic. See
<a href="graphics.html#topic+plot.default">plot.default</a>).</p>
</td></tr>
<tr><td><code id="analyse_SAR.TL_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function performs a SAR TL analysis on a set of curves. The SAR
procedure in general is given by Murray and Wintle (2000). For the
calculation of the <code>Lx/Tx</code> value the function <a href="#topic+calc_TLLxTxRatio">calc_TLLxTxRatio</a> is
used.
</p>
<p><strong>Provided rejection criteria</strong>
</p>
<p><code style="white-space: pre;">&#8288;[recycling.ratio]&#8288;</code>: calculated for every repeated regeneration dose point.
</p>
<p><code style="white-space: pre;">&#8288;[recuperation.rate]&#8288;</code>: recuperation rate calculated by
comparing the <code>Lx/Tx</code> values of the zero regeneration point with the <code>Ln/Tn</code>
value (the <code>Lx/Tx</code> ratio of the natural signal).  For methodological
background see Aitken and Smith (1988)
</p>


<h3>Value</h3>

<p>A plot (<em>optional</em>) and an <a href="#topic+RLum.Results-class">RLum.Results</a> object is
returned containing the following elements:
</p>
<table role = "presentation">
<tr><td><code>De.values</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> containing De-values and further parameters</p>
</td></tr>
<tr><td><code>LnLxTnTx.values</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> of all calculated <code>Lx/Tx</code> values including signal, background counts and the dose points.</p>
</td></tr>
<tr><td><code>rejection.criteria</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> with values that might by used as rejection criteria. NA is produced if no R0 dose point exists.</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.3.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. analyse_SAR.TL(): Analyse SAR TL measurements. Function version 0.3.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>THIS IS A BETA VERSION</strong>
</p>
<p>None TL curves will be removed
from the input object without further warning.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Aitken, M.J. and Smith, B.W., 1988. Optical dating: recuperation
after bleaching.  Quaternary Science Reviews 7, 387-393.
</p>
<p>Murray, A.S. and Wintle, A.G., 2000. Luminescence dating of quartz using an
improved single-aliquot regenerative-dose protocol. Radiation Measurements
32, 57-73.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_TLLxTxRatio">calc_TLLxTxRatio</a>, <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>,
<a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.BINfileData, envir = environment())

##transform the values from the first position in a RLum.Analysis object
object &lt;- Risoe.BINfileData2RLum.Analysis(TL.SAR.Data, pos=3)

##perform analysis
analyse_SAR.TL(
 object = object,
 signal.integral.min = 210,
 signal.integral.max = 220,
 fit.method = "EXP OR LIN",
 sequence.structure = c("SIGNAL", "BACKGROUND"))

</code></pre>

<hr>
<h2 id='apply_CosmicRayRemoval'>Function to remove cosmic rays from an RLum.Data.Spectrum S4 class object</h2><span id='topic+apply_CosmicRayRemoval'></span>

<h3>Description</h3>

<p>The function provides several methods for cosmic-ray removal and spectrum
smoothing <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> objects and such objects embedded in <a href="base.html#topic+list">list</a> or
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_CosmicRayRemoval(
  object,
  method = "smooth",
  method.Pych.smoothing = 2,
  method.Pych.threshold_factor = 3,
  MARGIN = 2,
  verbose = FALSE,
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apply_CosmicRayRemoval_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> or <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>): input
object to be treated. This can be also provided as <a href="base.html#topic+list">list</a>. If an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object
is provided, only the <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> objects are treated. Please note: this mixing of
objects does not work for a list of <code>RLum.Data</code> objects.</p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_method">method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
Defines method that is applied for cosmic ray removal. Allowed methods are
<code>smooth</code>, the default, (<a href="stats.html#topic+smooth">stats::smooth</a>), <code>smooth.spline</code> (<a href="stats.html#topic+smooth.spline">stats::smooth.spline</a>),
<code>smooth_RLum</code> (<a href="#topic+smooth_RLum">smooth_RLum</a>) and <code>Pych</code>. See details for further information.</p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_method.pych.smoothing">method.Pych.smoothing</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
Smoothing parameter for cosmic ray removal according to Pych (2003).
The value defines how many neighbouring values in each frame are used for smoothing
(e.g., <code>2</code> means that the two previous and two following values are used).</p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_method.pych.threshold_factor">method.Pych.threshold_factor</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Threshold for zero-bins in the histogram. Small values mean that more peaks
are removed, but signal might be also affected by this removal.</p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_margin">MARGIN</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
on which part the function cosmic ray removal should be applied on:
</p>

<ul>
<li><p> 1 = along the time axis (line by line),
</p>
</li>
<li><p> 2 = along the wavelength axis (column by column).
</p>
</li></ul>

<p><strong>Note:</strong> This argument currently only affects the methods <code>smooth</code> and <code>smooth.spline</code></p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If <code>TRUE</code> the histograms used for the cosmic-ray removal are returned as plot
including the used threshold. Note: A separate plot is returned for each frame!
Currently only for <code>method = "Pych"</code> a graphical output is provided.</p>
</td></tr>
<tr><td><code id="apply_CosmicRayRemoval_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters that will be passed
to the <a href="stats.html#topic+smooth">stats::smooth</a>, <a href="stats.html#topic+smooth.spline">stats::smooth.spline</a> or <a href="#topic+smooth_RLum">smooth_RLum</a>. See details for more
information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong><code>method = "Pych"</code></strong>
</p>
<p>This method applies the cosmic-ray removal algorithm described by Pych
(2003). Some aspects that are different to the publication:
</p>

<ul>
<li><p> For interpolation between neighbouring values the median and not the mean is used.
</p>
</li>
<li><p> The number of breaks to construct the histogram is set to: <code>length(number.of.input.values)/2</code>
</p>
</li></ul>

<p>For further details see references below.
</p>
<p><strong>Other methods</strong>
Arguments supported through <code>...</code>
</p>

<table>
<tr>
 <td style="text-align: left;">
METHOD </td><td style="text-align: left;"> ARGUMENT </td><td style="text-align: left;"> TYPE </td><td style="text-align: left;"> REMARKS</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"smooth"</code> </td><td style="text-align: left;"> <code>kind</code> </td><td style="text-align: left;">  <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> see <a href="stats.html#topic+smooth">stats::smooth</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>twiceit</code> </td><td style="text-align: left;">  <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> see <a href="stats.html#topic+smooth">stats::smooth</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"smooth.spline"</code> </td><td style="text-align: left;"> <code>spar</code> </td><td style="text-align: left;">  <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> see <a href="stats.html#topic+smooth.spline">stats::smooth.spline</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>"smooth_RLum"</code> </td><td style="text-align: left;"> <code>k</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> see <a href="#topic+smooth_RLum">smooth_RLum</a></td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>fill</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> see <a href="#topic+smooth_RLum">smooth_RLum</a></td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>align</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> see <a href="#topic+smooth_RLum">smooth_RLum</a></td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> <code>method</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> see <a href="#topic+smooth_RLum">smooth_RLum</a></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>Best practice</strong>
</p>
<p>There is no single silver-bullet-strategy for the cosmic-ray removal, because it depends
on the characteristic of the detector and the chosen settings. For instance,
high values for pixel binning will improve the light output, but also causes
multiple pixels being affected a single cosmic-ray. The same is valid for
longer integration times. The best strategy is to combine methods and ensure
that the spectrum is not distorted on a case-to-case basis.
</p>
<p><strong>How to combine methods?</strong>
</p>
<p>Different methods can be combined by applying the method repeatedly to the
dataset (see example).
</p>


<h3>Value</h3>

<p>Returns same object as input.
</p>


<h3>Function version</h3>

<p>0.4.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. apply_CosmicRayRemoval(): Function to remove cosmic rays from an RLum.Data.Spectrum S4 class object. Function version 0.4.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Pych, W., 2004. A Fast Algorithm for Cosmic-Ray Removal from
Single Images. The Astronomical Society of the Pacific 116 (816), 148-153.
<a href="https://doi.org/10.1086/381786">doi:10.1086/381786</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="stats.html#topic+smooth">stats::smooth</a>, <a href="stats.html#topic+smooth.spline">stats::smooth.spline</a>,
<a href="#topic+smooth_RLum">smooth_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) - use with your own data and combine (uncomment for usage)
## run two times the default method and smooth with another method
## your.spectrum &lt;- apply_CosmicRayRemoval(your.spectrum, method = "Pych")
## your.spectrum &lt;- apply_CosmicRayRemoval(your.spectrum, method = "Pych")
## your.spectrum &lt;- apply_CosmicRayRemoval(your.spectrum, method = "smooth")

</code></pre>

<hr>
<h2 id='apply_Crosstalk'>Apply crosstalk</h2><span id='topic+apply_Crosstalk'></span>

<h3>Description</h3>

<p>Add crosstalk, evenly spread in rook (top-right-bottom-left)
directions, to all grain whole locations on one measurement discs
(=on position on a measurement wheel in a reader). An
added crosstalk value of as example 0.2 means that 0.2 of the value of
the central grain is added to each grain in the rook directions. This is an additive
action: the central grain itself is not affected by this operation (but will on its
turn increase because of crosstalk from its neighbours).
This function is used for simulations: can this added crosstalk be detected?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_Crosstalk(object, n_crosstalk = 0.2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apply_Crosstalk_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): containing
a numerical vector of length 100, representing one or more measurement
discs (&quot;positions&quot;) in a reader.
Each element in the vector represents one grain hole location on a disc.</p>
</td></tr>
<tr><td><code id="apply_Crosstalk_+3A_n_crosstalk">n_crosstalk</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): A single number quantifying the added
crosstalk. Defaults for testing purposes to 0.2. Can be any number, even negative,
but for realistic simulations we suggest something between 0 and 0.25.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If an element in <code>object</code> is <code>NA</code>, it is internally set to 0, so it will not
be added.
</p>


<h3>Value</h3>

<p>A vector of size 100, with the value at each grain hole location including simulated crosstalk.
</p>


<h3>How to cite</h3>

<p>Boer, A.d., Steinbuch, L., 2025. apply_Crosstalk(): Apply crosstalk. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Anna-Maartje de Boer, Luc Steinbuch, Wageningen University &amp; Research, 2025
, RLum Developer Team</p>


<h3>References</h3>

<p>de Boer, A-M., Steinbuch, L., Heuvelink, G.B.M., Wallinga, J., 2025.
A novel tool to assess crosstalk in single-grain luminescence detection.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Create artificial disc observation
observations &lt;- set_RLum(class = "RLum.Results",
                         data = list(vn_values = rep(x = c(1,2), each = 50))
                       )
hist(get_RLum(object = observations))

## Add crosstalk (with default set to 0.2), and visualize the difference
## in the resulting histogram.
observations_with_simulated_crosstalk &lt;- apply_Crosstalk(observations)
hist(observations_with_simulated_crosstalk)

</code></pre>

<hr>
<h2 id='apply_EfficiencyCorrection'>Function to apply spectral efficiency correction to RLum.Data.Spectrum S4
class objects</h2><span id='topic+apply_EfficiencyCorrection'></span>

<h3>Description</h3>

<p>The function allows spectral efficiency corrections for RLum.Data.Spectrum
S4 class objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>apply_EfficiencyCorrection(object, spectral.efficiency)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="apply_EfficiencyCorrection_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> or <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
S4 object of class <code>RLum.Data.Spectrum</code>,  <code>RLum.Analysis</code>or a <a href="base.html#topic+list">list</a> of such objects. Other objects in
the list are skipped.</p>
</td></tr>
<tr><td><code id="apply_EfficiencyCorrection_+3A_spectral.efficiency">spectral.efficiency</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
Data set containing wavelengths (x-column) and relative spectral response values
(y-column) (values between 0 and 1). The provided data will be used to correct all spectra if <code>object</code> is
a <a href="base.html#topic+list">list</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The efficiency correction is based on a spectral response dataset provided
by the user. Usually the data set for the quantum efficiency is of lower
resolution and values are interpolated for the required spectral resolution using
the function <a href="stats.html#topic+approxfun">stats::approx</a>
</p>
<p>If the energy calibration differs for both data set <code>NA</code> values are produces that
will be removed from the matrix.
</p>


<h3>Value</h3>

<p>Returns same object as provided as input
</p>


<h3>Function version</h3>

<p>0.2.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Friedrich, J., 2025. apply_EfficiencyCorrection(): Function to apply spectral efficiency correction to RLum.Data.Spectrum S4 class objects. Function version 0.2.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Please note that the spectral efficiency data from the camera alone may not
sufficiently correct for spectral efficiency of the entire optical system
(e.g., spectrometer, camera ...).
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, IRAMAT-CRP2A, UMR 5060, CNRS-Universit√© Bordeaux Montaigne (France)<br />
Johannes Friedrich, University of Bayreuth (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) - use with your own data (uncomment for usage)
## spectral.efficiency &lt;- read.csv("your data")
##
## your.spectrum &lt;- apply_EfficiencyCorrection(your.spectrum, )

</code></pre>

<hr>
<h2 id='as'>as() - RLum-object coercion</h2><span id='topic+as'></span>

<h3>Description</h3>

<p>for <code style="white-space: pre;">&#8288;[RLum.Analysis-class]&#8288;</code>
</p>
<p>for <code style="white-space: pre;">&#8288;[RLum.Data.Curve-class]&#8288;</code>
</p>
<p>for <code style="white-space: pre;">&#8288;[RLum.Data.Image-class]&#8288;</code>
</p>
<p>for <code style="white-space: pre;">&#8288;[RLum.Data.Spectrum-class]&#8288;</code>
</p>
<p>for <code style="white-space: pre;">&#8288;[RLum.Results-class]&#8288;</code>
</p>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as_+3A_from">from</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a>, <a href="base.html#topic+list">list</a>, <a href="base.html#topic+data.frame">data.frame</a>, <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>):
object to be coerced from</p>
</td></tr>
<tr><td><code id="as_+3A_to">to</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
class name to be coerced to</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong><a href="#topic+RLum.Analysis-class">RLum.Analysis</a></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>from</strong> </td><td style="text-align: left;"> <strong>to</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>list</code> </td><td style="text-align: left;"> <code>list</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Given that the <a href="base.html#topic+list">list</a> consists of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects.
</p>
<p><strong><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>from</strong> </td><td style="text-align: left;"> <strong>to</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>list</code> </td><td style="text-align: left;"> <code>list</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.frame</code> </td><td style="text-align: left;"> <code>data.frame</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>matrix</code> </td><td style="text-align: left;"> <code>matrix</code>
</td>
</tr>

</table>

<p><strong><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>from</strong> </td><td style="text-align: left;"> <strong>to</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.frame</code> </td><td style="text-align: left;"> <code>data.frame</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>matrix</code> </td><td style="text-align: left;"> <code>matrix</code>
</td>
</tr>

</table>

<p><strong><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>from</strong> </td><td style="text-align: left;"> <strong>to</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data.frame</code> </td><td style="text-align: left;"> <code>data.frame</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>matrix</code> </td><td style="text-align: left;"> <code>matrix</code>
<code>list</code> </td><td style="text-align: left;"> <code>list</code>
</td>
</tr>

</table>

<p><strong><a href="#topic+RLum.Results-class">RLum.Results</a></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>from</strong> </td><td style="text-align: left;"> <strong>to</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>list</code> </td><td style="text-align: left;"> <code>list</code></td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Given that the <a href="base.html#topic+list">list</a> consists of <a href="#topic+RLum.Results-class">RLum.Results</a> objects.
</p>


<h3>Note</h3>

<p>Due to the complex structure of the <code>RLum</code> objects itself a coercing to standard
R data structures will be always loosely!
</p>


<h3>See Also</h3>

<p><a href="methods.html#topic+as">methods::as</a>
</p>

<hr>
<h2 id='BaseDataSet.ConversionFactors'>Base data set of dose-rate conversion factors</h2><span id='topic+BaseDataSet.ConversionFactors'></span>

<h3>Description</h3>

<p>Collection of published dose-rate conversion factors to convert concentrations
of radioactive isotopes to dose rate values.
</p>


<h3>Format</h3>

<p>A <code><a href="base.html#topic+list">list</a></code> with three elements with dose-rate conversion factors
sorted by article and radiation type (alpha, beta, gamma):
</p>

<table>
<tr>
 <td style="text-align: left;">

<code>AdamiecAitken1998</code>: </td><td style="text-align: left;">
Conversion factors from Tables 5 and 6 </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>Cresswelletal2018</code>: </td><td style="text-align: left;">
Conversion factors from Tables 5 and 6 </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>Guerinetal2011</code>: </td><td style="text-align: left;">
Conversion factors from Tables 1, 2 and 3 </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>Liritzisetal2013</code>: </td><td style="text-align: left;">
Conversion factors from Tables 1, 2 and 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Version</h3>

<p>0.2.0
</p>


<h3>Source</h3>

<p>All gamma conversion factors were carefully read from the tables given in the
references above.
</p>


<h3>References</h3>

<p>Adamiec, G., Aitken, M.J., 1998. Dose-rate conversion factors: update.
Ancient TL 16, 37-46.
</p>
<p>Cresswell., A.J., Carter, J., Sanderson, D.C.W., 2018.
Dose rate conversion parameters: Assessment of nuclear data.
Radiation Measurements 120, 195-201.
</p>
<p>Guerin, G., Mercier, N., Adamiec, G., 2011. Dose-rate conversion
factors: update. Ancient TL, 29, 5-8.
</p>
<p>Liritzis, I., Stamoulis, K., Papachristodoulou, C., Ioannides, K., 2013.
A re-evaluation of radiation dose-rate conversion factors. Mediterranean
Archaeology and Archaeometry 13, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load data
data("BaseDataSet.ConversionFactors", envir = environment())

</code></pre>

<hr>
<h2 id='BaseDataSet.CosmicDoseRate'>Base data set for cosmic dose rate calculation</h2><span id='topic+BaseDataSet.CosmicDoseRate'></span><span id='topic+values.cosmic.Softcomp'></span><span id='topic+values.factor.Altitude'></span><span id='topic+values.par.FJH'></span>

<h3>Description</h3>

<p>Collection of data from various sources needed for cosmic dose rate
calculation
</p>


<h3>Format</h3>


<table>
<tr>
 <td style="text-align: left;">

<code>values.cosmic.Softcomp</code>: </td><td style="text-align: left;">
data frame containing cosmic dose rates
for shallow depths (&lt; 167 g cm^-2) obtained using the "AGE" program by
Rainer Gruen (cf. Gruen 2009). These data essentially reproduce the graph
shown in Fig. 1 of Prescott &amp; Hutton (1988). </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>values.factor.Altitude</code>: </td><td style="text-align: left;">
data frame containing altitude factors
for adjusting geomagnetic field-change factors. Values were read from Fig. 1
in Prescott &amp; Hutton (1994). </td>
</tr>
<tr>
 <td style="text-align: left;">

<code>values.par.FJH</code>: </td><td style="text-align: left;">
data frame containing values for parameters F, J
and H (read from Fig. 2 in Prescott &amp; Hutton 1994) used in the expression </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p style="text-align: center;"><code class="reqn">Dc = D0*(F+J*exp((altitude/1000)/H))</code>
</p>



<h3>Version</h3>

<p>0.1
</p>


<h3>Source</h3>

<p>The following data were carefully read from figures in mentioned
sources and used for fitting procedures. The derived expressions are used in
the function <code>calc_CosmicDoseRate</code>.
</p>
<p><strong>values.cosmic.Softcomp</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Program: </td><td style="text-align: left;"> "AGE"</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> Gruen (2009) </td>
</tr>
<tr>
 <td style="text-align: left;">
Fit: </td><td style="text-align: left;"> Polynomials in the form of
</td>
</tr>

</table>

<p>For depths between 40-167 g cm^-2:
</p>
<p style="text-align: center;"><code class="reqn">y = 2*10^-6*x^2-0.0008*x+0.2535</code>
</p>

<p>(For depths &lt;40 g cm^-2)
</p>
<p style="text-align: center;"><code class="reqn">y = -6*10^-8*x^3+2*10^-5*x^2-0.0025*x+0.2969</code>
</p>

<p><strong><code>values.factor.Altitude</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> Prescott &amp; Hutton (1994) </td>
</tr>
<tr>
 <td style="text-align: left;">
Page: </td><td style="text-align: left;"> 499 </td>
</tr>
<tr>
 <td style="text-align: left;">
Figure: </td><td style="text-align: left;"> 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
Fit: </td><td style="text-align: left;"> 2-degree polynomial in the form of
</td>
</tr>

</table>

<p style="text-align: center;"><code class="reqn">y = -0.026*x^2 + 0.6628*x + 1.0435</code>
</p>

<p><strong><code>values.par.FJH</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> Prescott &amp; Hutton (1994) </td>
</tr>
<tr>
 <td style="text-align: left;">
Page: </td><td style="text-align: left;"> 500 </td>
</tr>
<tr>
 <td style="text-align: left;">
Figure: </td><td style="text-align: left;"> 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
Fits: </td><td style="text-align: left;"> 3-degree polynomials and linear fits
</td>
</tr>

</table>

<p>F (non-linear part, <code class="reqn">\lambda</code> &lt; 36.5 deg.):
</p>
<p style="text-align: center;"><code class="reqn">y = -7*10^-7*x^3-8*10^-5*x^2-0.0009*x+0.3988</code>
</p>

<p>F (linear part, <code class="reqn">\lambda</code> &gt; 36.5 deg.):
</p>
<p style="text-align: center;"><code class="reqn">y = -0.0001*x + 0.2347</code>
</p>

<p>J (non-linear part, <code class="reqn">\lambda</code> &lt; 34 deg.):
</p>
<p style="text-align: center;"><code class="reqn">y = 5*10^-6*x^3-5*10^-5*x^2+0.0026*x+0.5177</code>
</p>

<p>J (linear part, <code class="reqn">\lambda</code> &gt; 34 deg.):
</p>
<p style="text-align: center;"><code class="reqn">y = 0.0005*x + 0.7388</code>
</p>

<p>H (non-linear part, <code class="reqn">\lambda</code> &lt; 36 deg.):
</p>
<p style="text-align: center;"><code class="reqn">y = -3*10^-6*x^3-5*10^-5*x^2-0.0031*x+4.398</code>
</p>

<p>H (linear part, <code class="reqn">\lambda</code> &gt; 36 deg.):
</p>
<p style="text-align: center;"><code class="reqn">y = 0.0002*x + 4.0914</code>
</p>



<h3>References</h3>

<p>Gruen, R., 2009. The &quot;AGE&quot; program for the calculation of luminescence age estimates.
Ancient TL, 27, pp. 45-46.
</p>
<p>Prescott, J.R., Hutton, J.T., 1988. Cosmic ray and gamma ray dosimetry for
TL and ESR. Nuclear Tracks and Radiation Measurements, 14, pp. 223-227.
</p>
<p>Prescott, J.R., Hutton, J.T., 1994. Cosmic ray contributions to dose rates
for luminescence and ESR dating: large depths and long-term time variations.
Radiation Measurements, 23, pp. 497-500.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(BaseDataSet.CosmicDoseRate)

</code></pre>

<hr>
<h2 id='BaseDataSet.FractionalGammaDose'>Base data set of fractional gamma-dose values</h2><span id='topic+BaseDataSet.FractionalGammaDose'></span>

<h3>Description</h3>

<p>Collection of (un-)published fractional gamma dose-rate values to scale the
gamma-dose rate considering layer-to-layer variations in soil radioactivity.
</p>


<h3>Format</h3>

<p>A <code><a href="base.html#topic+list">list</a></code> with fractional gamma dose-rate values
sorted by article:
</p>

<table>
<tr>
 <td style="text-align: left;">

<code>Aitken1985</code>: </td><td style="text-align: left;">
Fractional gamma-dose values from table H.1
</td>
</tr>

</table>



<h3>Version</h3>

<p>0.1
</p>


<h3>Source</h3>

<p>Fractional gamma dose values were carefully read from the tables given in the
references above.
</p>


<h3>References</h3>

<p>Aitken, M.J., 1985. Thermoluminescence Dating. Academic Press, London.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load data
data("BaseDataSet.FractionalGammaDose", envir = environment())

</code></pre>

<hr>
<h2 id='BaseDataSet.GrainSizeAttenuation'>Base dataset for grain size attenuation data by Gu√©rin et al. (2012)</h2><span id='topic+BaseDataSet.GrainSizeAttenuation'></span>

<h3>Description</h3>

<p>Grain size correction data for beta-dose rates
published by Gu√©rin et al. (2012).
</p>
<p>#' @format
</p>
<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> seven columns and sixteen rows. Column headers
are <code>GrainSize</code>, <code>Q_K</code>, <code>FS_K</code>, <code>Q_Th</code>, <code>FS_Th</code>, <code>Q_U</code>, <code>FS_U</code>.
Grain sizes are quoted in ¬µm (e.g., 20, 40, 60 etc.)
</p>


<h3>Version</h3>

<p>0.1.0
</p>


<h3>Source</h3>

<p>Gu√©rin, G., Mercier, N., Nathan, R., Adamiec, G., Lefrais, Y., 2012.
On the use of the infinite matrix assumption and associated concepts:
A critical review. Radiation Measurements, 47, 778-785.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load data
data("BaseDataSet.GrainSizeAttenuation", envir = environment())

</code></pre>

<hr>
<h2 id='bin_RLum.Data'>Channel binning for RLum.Data S4 class objects.</h2><span id='topic+bin_RLum.Data'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum.Data-class">RLum.Data</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum.Data-class">RLum.Data</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bin_RLum.Data(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bin_RLum.Data_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data-class">RLum.Data</a> (<strong>required</strong>):
S4 object of class <code>RLum.Data</code></p>
</td></tr>
<tr><td><code id="bin_RLum.Data_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same type as the input object provided.
</p>


<h3>Function version</h3>

<p>0.2.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. bin_RLum.Data(): Channel binning for RLum.Data S4 class objects.. Function version 0.2.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Currently only <code>RLum.Data</code> objects of class <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
and <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> are supported.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.CW_OSL_Curve, envir = environment())

## create RLum.Data.Curve object from this example
curve &lt;-
  set_RLum(
      class = "RLum.Data.Curve",
      recordType = "OSL",
      data = as.matrix(ExampleData.CW_OSL_Curve)
  )

## plot data without and with 2 and 4 channel binning
plot_RLum(curve)
plot_RLum(bin_RLum.Data(curve, bin_size = 2))
plot_RLum(bin_RLum.Data(curve, bin_size = 4))

</code></pre>

<hr>
<h2 id='calc_AliquotSize'>Estimate the amount of grains on an aliquot</h2><span id='topic+calc_AliquotSize'></span>

<h3>Description</h3>

<p>Estimate the number of grains on an aliquot. Alternatively, the packing
density of an aliquot is computed.
</p>
<p>This function can be used to either estimate the number of grains on an
aliquot or to compute the packing density depending on the arguments
provided.
</p>
<p>The following function is used to estimate the number of grains <code>n</code>:
</p>
<p style="text-align: center;"><code class="reqn">n = (\pi*x^2)/(\pi*y^2)*d</code>
</p>

<p>where <code>x</code> is the radius of the aliquot size (microns), <code>y</code> is the mean
radius of the mineral grains (mm) and <code>d</code> is the packing density
(value between 0 and 1).
</p>
<p><strong>Packing density</strong>
</p>
<p>The default value for <code>packing.density</code> is 0.65, which is the mean of
empirical values determined by Heer et al. (2012) and unpublished data from
the Cologne luminescence laboratory. If <code>packing.density = "Inf"</code> a maximum
density of  <code class="reqn">\pi/\sqrt12 = 0.9068\ldots</code> is used. However, note that
this value is not appropriate as the standard preparation procedure of
aliquots resembles a  PECC (<em>&quot;Packing Equal Circles in a Circle&quot;</em>) problem
where the maximum packing density is asymptotic to about 0.87.
</p>
<p><strong>Monte Carlo simulation</strong>
</p>
<p>The number of grains on an aliquot can be estimated by Monte Carlo simulation
when setting <code>MC = TRUE</code>. Each of the parameters necessary to calculate
<code>n</code> (<code>x</code>, <code>y</code>, <code>d</code>) are assumed to be normally distributed with means
<code class="reqn">\mu_x, \mu_y, \mu_d</code> and standard deviations <code class="reqn">\sigma_x, \sigma_y, \sigma_d</code>.
</p>
<p>For the mean grain size random samples are taken first from
<code class="reqn">N(\mu_y, \sigma_y)</code>, where <code class="reqn">\mu_y = mean.grain.size</code> and
<code class="reqn">\sigma_y = (max.grain.size-min.grain.size)/4</code> so that 95% of all
grains are within the provided the grain size range. This effectively takes
into account that after sieving the sample there is still a small chance of
having grains smaller or larger than the used mesh sizes. For each random
sample the mean grain size is calculated, from which random subsamples are
drawn for the Monte Carlo simulation.
</p>
<p>The packing density is assumed
to be normally distributed with an empirically determined <code class="reqn">\mu = 0.65</code>
(or provided value) and <code class="reqn">\sigma = 0.18</code>. The normal distribution is
truncated at <code>d = 0.87</code> as this is approximately the maximum packing
density that can be achieved in PECC problem.
</p>
<p>The sample diameter has
<code class="reqn">\mu = sample.diameter</code> and <code class="reqn">\sigma = 0.2</code> to take into account
variations in sample disc preparation (i.e. applying silicon spray to the
disc). A lower truncation point at <code>x = 0.5</code> is used, which assumes
that aliquots with smaller sample diameters of 0.5 mm are discarded.
Likewise, the normal distribution is truncated at 9.8 mm, which is the
diameter of the sample disc.
</p>
<p>For each random sample drawn from the
normal distributions the amount of grains on the aliquot is calculated. By
default, <code>10^5</code> iterations are used, but can be reduced/increased with
<code>MC.iter</code> (see <code>...</code>). The results are visualised in a bar- and
boxplot together with a statistical summary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_AliquotSize(
  grain.size,
  sample.diameter,
  packing.density = 0.65,
  MC = TRUE,
  grains.counted,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_AliquotSize_+3A_grain.size">grain.size</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
mean grain size (microns) or a range of grain sizes from which the
mean grain size is computed (e.g. <code>c(100,200)</code>).</p>
</td></tr>
<tr><td><code id="calc_AliquotSize_+3A_sample.diameter">sample.diameter</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
diameter (mm) of the targeted area on the sample carrier.</p>
</td></tr>
<tr><td><code id="calc_AliquotSize_+3A_packing.density">packing.density</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
empirical value for mean packing density. <br />
If <code>packing.density = "Inf"</code> a hexagonal structure on an infinite plane with
a packing density of <code class="reqn">0.906\ldots</code> is assumed.</p>
</td></tr>
<tr><td><code id="calc_AliquotSize_+3A_mc">MC</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
if <code>TRUE</code> the function performs a Monte Carlo simulation for estimating the
amount of grains on the sample carrier and assumes random errors in grain
size distribution and packing density. Requires a vector with min and max
grain size for <code>grain.size</code>. For more information see details.</p>
</td></tr>
<tr><td><code id="calc_AliquotSize_+3A_grains.counted">grains.counted</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
grains counted on a sample carrier. If a non-zero positive integer is provided this function
will calculate the packing density of the aliquot. If more than one value is
provided the mean packing density and its standard deviation is calculated.
Note that this overrides <code>packing.density</code>.</p>
</td></tr>
<tr><td><code id="calc_AliquotSize_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_AliquotSize_+3A_...">...</code></td>
<td>
<p>further arguments to pass (<code style="white-space: pre;">&#8288;main, xlab, MC.iter&#8288;</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following element:
</p>
<table role = "presentation">
<tr><td><code>.$summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant calculation results.</p>
</td></tr>
<tr><td><code>.$args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>.$call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>.$MC</code></td>
<td>
<p><a href="base.html#topic+list">list</a> results of the Monte Carlo simulation</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.31
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_AliquotSize(): Estimate the amount of grains on an aliquot. Function version 0.31. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Duller, G.A.T., 2008. Single-grain optical dating of Quaternary
sediments: why aliquot size matters in luminescence dating. Boreas 37,
589-612.
</p>
<p>Heer, A.J., Adamiec, G., Moska, P., 2012. How many grains
are there on a single aliquot?. Ancient TL 30, 9-16.
</p>
<p><strong>Further reading</strong>
</p>
<p>Chang, H.-C., Wang, L.-C., 2010. A simple proof of Thue's
Theorem on Circle Packing. <a href="https://arxiv.org/pdf/1009.4322v1">https://arxiv.org/pdf/1009.4322v1</a>,
2013-09-13.
</p>
<p>Graham, R.L., Lubachevsky, B.D., Nurmela, K.J.,
Oestergard, P.R.J., 1998.  Dense packings of congruent circles in a circle.
Discrete Mathematics 181, 139-154.
</p>
<p>Huang, W., Ye, T., 2011. Global
optimization method for finding dense packings of equal circles in a circle.
European Journal of Operational Research 210, 474-481.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Estimate the amount of grains on a small aliquot
calc_AliquotSize(grain.size = c(100,150), sample.diameter = 1, MC.iter = 100)

## Calculate the mean packing density of large aliquots
calc_AliquotSize(grain.size = c(100,200), sample.diameter = 8,
                 grains.counted = c(2525,2312,2880), MC.iter = 100)

</code></pre>

<hr>
<h2 id='calc_AverageDose'>Calculate the Average Dose and the dose rate dispersion</h2><span id='topic+calc_AverageDose'></span>

<h3>Description</h3>

<p>This functions calculates the Average Dose and its extrinsic dispersion,
estimating the standard errors by bootstrapping based on the Average
Dose Model by Guerin et al., 2017.
</p>
<p><strong><code>sigma_m</code></strong><br />
</p>
<p>The program requires the input of a known value of <code>sigma_m</code>,
which corresponds to the intrinsic overdispersion, as determined
by a dose recovery experiment. Then the dispersion in doses (<code>sigma_d</code>)
will be that over and above <code>sigma_m</code> (and individual uncertainties <code>sigma_wi</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_AverageDose(
  data,
  sigma_m,
  Nb_BE = 500,
  na.rm = TRUE,
  plot = TRUE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_AverageDose_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with <code>De</code> <code>(data[,1])</code> and <code style="white-space: pre;">&#8288;De error&#8288;</code> <code>(values[,2])</code></p>
</td></tr>
<tr><td><code id="calc_AverageDose_+3A_sigma_m">sigma_m</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
the overdispersion resulting from a dose recovery
experiment, i.e. when all grains have  received the same dose. Indeed in such a case, any
overdispersion (i.e. dispersion on top of analytical uncertainties) is, by definition, an
unrecognised measurement uncertainty.</p>
</td></tr>
<tr><td><code id="calc_AverageDose_+3A_nb_be">Nb_BE</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
sample size used for the bootstrapping</p>
</td></tr>
<tr><td><code id="calc_AverageDose_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
exclude NA values from the data set prior to any further operation.</p>
</td></tr>
<tr><td><code id="calc_AverageDose_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_AverageDose_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="calc_AverageDose_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to <a href="graphics.html#topic+hist">graphics::hist</a>. As three plots
are returned all arguments need to be provided as <a href="base.html#topic+list">list</a>,
e.g., <code>main = list("Plot 1", "Plot 2", "Plot 3")</code>.
Note: not all arguments of <code>hist</code> are
supported, but the output of <code>hist</code> is returned and can be used of own plots. <br />
</p>
<p>Further supported arguments: <code>mtext</code> (<a href="base.html#topic+character">character</a>), <code>rug</code> (<code>TRUE/FALSE</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns numerical output and an (<em>optional</em>) plot.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code> <br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<strong><code>RLum.Results</code></strong>-object<br />
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong> <br />
</p>
<p><code style="white-space: pre;">&#8288;[.. $summary : data.frame]&#8288;</code><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Column</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
AVERAGE_DOSE </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> the obtained average dose</td>
</tr>
<tr>
 <td style="text-align: left;">
AVERAGE_DOSE.SE </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> the average dose error </td>
</tr>
<tr>
 <td style="text-align: left;">
SIGMA_D </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a></td><td style="text-align: left;"> sigma </td>
</tr>
<tr>
 <td style="text-align: left;">
SIGMA_D.SE </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a></td><td style="text-align: left;"> standard error of the sigma  </td>
</tr>
<tr>
 <td style="text-align: left;">
IC_AVERAGE_DOSE.LEVEL  </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a></td><td style="text-align: left;"> confidence level average dose</td>
</tr>
<tr>
 <td style="text-align: left;">
IC_AVERAGE_DOSE.LOWER  </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a></td><td style="text-align: left;"> lower quantile of average dose </td>
</tr>
<tr>
 <td style="text-align: left;">
IC_AVERAGE_DOSE.UPPER </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a></td><td style="text-align: left;"> upper quantile of average dose</td>
</tr>
<tr>
 <td style="text-align: left;">
IC_SIGMA_D.LEVEL </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a></td><td style="text-align: left;"> confidence level sigma</td>
</tr>
<tr>
 <td style="text-align: left;">
IC_SIGMA_D.LOWER </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a></td><td style="text-align: left;"> lower sigma quantile</td>
</tr>
<tr>
 <td style="text-align: left;">
IC_SIGMA_D.UPPER </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a></td><td style="text-align: left;"> upper sigma quantile</td>
</tr>
<tr>
 <td style="text-align: left;">
L_MAX </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a></td><td style="text-align: left;"> maximum likelihood value
</td>
</tr>

</table>

<p><code style="white-space: pre;">&#8288;[.. $dstar : matrix]&#8288;</code> <br />
</p>
<p>Matrix with bootstrap values<br />
</p>
<p><code style="white-space: pre;">&#8288;[.. $hist : list]&#8288;</code><br />
</p>
<p>Object as produced by the function histogram
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>The function returns two different plot panels.
</p>
<p>(1) An abanico plot with the dose values
</p>
<p>(2) A histogram panel comprising 3 histograms with the equivalent dose and the bootstrapped average
dose and the sigma values.
</p>


<h3>Function version</h3>

<p>0.1.5
</p>


<h3>How to cite</h3>

<p>Christophe, C., Philippe, A., Kreutzer, S., 2025. calc_AverageDose(): Calculate the Average Dose and the dose rate dispersion. Function version 0.1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function has beta status!
</p>


<h3>Author(s)</h3>

<p>Claire Christophe, IRAMAT-CRP2A, Universit√© de Nantes (France),
Anne Philippe, Universit√© de Nantes, (France),
Guillaume Gu√©rin, IRAMAT-CRP2A, Universit√© Bordeaux Montaigne, (France),
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Guerin, G., Christophe, C., Philippe, A., Murray, A.S., Thomsen, K.J., Tribolo, C., Urbanova, P.,
Jain, M., Guibert, P., Mercier, N., Kreutzer, S., Lahaye, C., 2017. Absorbed dose, equivalent dose,
measured dose rates, and implications for OSL age estimates: Introducing the Average Dose Model.
Quaternary Geochronology 1-32. doi:10.1016/j.quageo.2017.04.002
</p>
<p><strong>Further reading</strong><br />
</p>
<p>Efron, B., Tibshirani, R., 1986. Bootstrap Methods for Standard Errors, Confidence Intervals,
and Other Measures of Statistical Accuracy. Statistical Science 1, 54-75.
</p>


<h3>See Also</h3>

<p><a href="utils.html#topic+read.table">read.table</a>, <a href="graphics.html#topic+hist">graphics::hist</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##Example 01 using package example data
##load example data
data(ExampleData.DeValues, envir = environment())

##calculate Average dose
##(use only the first 56 values here)
AD &lt;- calc_AverageDose(ExampleData.DeValues$CA1[1:56,], sigma_m = 0.1)

##plot De and set Average dose as central value
plot_AbanicoPlot(
 data = ExampleData.DeValues$CA1[1:56,],
 z.0 = AD$summary$AVERAGE_DOSE)

</code></pre>

<hr>
<h2 id='calc_CentralDose'>Apply the central age model (CAM) after Galbraith et al. (1999) to a given
De distribution</h2><span id='topic+calc_CentralDose'></span>

<h3>Description</h3>

<p>This function calculates the central dose and dispersion of the De
distribution, their standard errors and the profile log likelihood function
for <code>sigma</code>.
</p>
<p>This function uses the equations of Galbraith &amp; Roberts (2012). The
parameters <code>delta</code> and <code>sigma</code> are estimated by numerically solving
eq. 15 and 16. Their standard errors are approximated using eq. 17.
In addition, the profile log-likelihood function for <code>sigma</code> is
calculated using eq. 18 and presented as a plot. Numerical values of the
maximum likelihood approach are <strong>only</strong> presented in the plot and <strong>not</strong>
in the console. A detailed explanation on maximum likelihood estimation can
be found in the appendix of Galbraith &amp; Laslett (1993, 468-470) and
Galbraith &amp; Roberts (2012, 15)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_CentralDose(data, sigmab, log = TRUE, plot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_CentralDose_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[,1])</code> and De error <code>(data[,2])</code>.
Records containing missing values will be removed.</p>
</td></tr>
<tr><td><code id="calc_CentralDose_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
additional spread in De values.
This value represents the expected overdispersion in the data should the sample be
well-bleached (Cunningham &amp; Walling 2012, p. 100).
<strong>NOTE</strong>: For the logged model (<code>log = TRUE</code>) this value must be
a fraction, e.g. 0.2 (= 20 %). If the un-logged model is used (<code>log = FALSE</code>),
sigmab must be provided in the same absolute units of the De values (seconds or Gray).</p>
</td></tr>
<tr><td><code id="calc_CentralDose_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
fit the (un-)logged central age model to De data. Log transformation is
allowed only if the De values are positive.</p>
</td></tr>
<tr><td><code id="calc_CentralDose_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_CentralDose_+3A_...">...</code></td>
<td>
<p>further arguments (<code>trace</code>, <code>verbose</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot (<em>optional</em>) and terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the following elements:
</p>
<table role = "presentation">
<tr><td><code>.$summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>.$data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>.$args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>.$call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>.$profile</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> the log likelihood profile for sigma</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>1.5
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_CentralDose(): Apply the central age model (CAM) after Galbraith et al. (1999) to a given De distribution. Function version 1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany) <br />
Based on a rewritten S script of Rex Galbraith, 2010
, RLum Developer Team</p>


<h3>References</h3>

<p>Galbraith, R.F. &amp; Laslett, G.M., 1993. Statistical models for
mixed fission track ages. Nuclear Tracks Radiation Measurements 4, 459-470.
</p>
<p>Galbraith, R.F., Roberts, R.G., Laslett, G.M., Yoshida, H. &amp; Olley,
J.M., 1999. Optical dating of single grains of quartz from Jinmium rock
shelter, northern Australia. Part I: experimental design and statistical
models.  Archaeometry 41, 339-364.
</p>
<p>Galbraith, R.F. &amp; Roberts, R.G., 2012. Statistical aspects of equivalent dose and error calculation and
display in OSL dating: An overview and some recommendations. Quaternary
Geochronology 11, 1-27.
</p>
<p><strong>Further reading</strong>
</p>
<p>Arnold, L.J. &amp; Roberts, R.G., 2009. Stochastic modelling of multi-grain equivalent dose
(De) distributions: Implications for OSL dating of sediment mixtures.
Quaternary Geochronology 4, 204-230.
</p>
<p>Bailey, R.M. &amp; Arnold, L.J., 2006. Statistical modelling of single grain quartz De distributions and an
assessment of procedures for estimating burial dose. Quaternary Science
Reviews 25, 2475-2502.
</p>
<p>Cunningham, A.C. &amp; Wallinga, J., 2012. Realizing the potential of fluvial archives using robust OSL chronologies.
Quaternary Geochronology 12, 98-106.
</p>
<p>Rodnight, H., Duller, G.A.T., Wintle, A.G. &amp; Tooth, S., 2006. Assessing the reproducibility and accuracy
of optical dating of fluvial deposits.  Quaternary Geochronology, 1 109-120.
</p>
<p>Rodnight, H., 2008. How many equivalent dose values are needed to
obtain a reproducible distribution?. Ancient TL 26, 3-10.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+calc_CommonDose">calc_CommonDose</a>, <a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>,
<a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load example data
data(ExampleData.DeValues, envir = environment())

##apply the central dose model
calc_CentralDose(ExampleData.DeValues$CA1)

</code></pre>

<hr>
<h2 id='calc_CobbleDoseRate'>Calculate dose rate of slices in a spherical cobble</h2><span id='topic+calc_CobbleDoseRate'></span>

<h3>Description</h3>

<p>Calculates the dose rate profile through the cobble based on Riedesel and Autzen (2020).
</p>
<p>Corrects the beta dose rate in the cobble for the grain size following results
of Gu√©rin et al. (2012). Sediment beta and gamma dose rates are corrected
for the water content of the sediment using the correction factors of Aitken (1985).
Water content in the cobble is assumed to be 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_CobbleDoseRate(input, conversion = "Guerinetal2011")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_CobbleDoseRate_+3A_input">input</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): A table containing all relevant information
for each individual layer. For the table layout see details.</p>
</td></tr>
<tr><td><code id="calc_CobbleDoseRate_+3A_conversion">conversion</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): dose rate conversion factors
to use, see <a href="#topic+BaseDataSet.ConversionFactors">BaseDataSet.ConversionFactors</a> for the accepted values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>The input table layout</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
COLUMN </td><td style="text-align: left;"> DATA TYPE </td><td style="text-align: left;"> DESCRIPTION</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Distance</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> distance from the surface of the cobble to the top of each rock slice in mm. The distance for each slice will be listed in this column</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DistanceError</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Error on the distance in mm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Thickness</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Thickness of each slice in mm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TicknessError</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> uncertainty of the thickness in mm.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Mineral</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> <code>'FS'</code> for feldspar, <code>'Q'</code> for quartz, depending which mineral in the cobble is used for dating</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Cobble_K</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> K nuclide content in % of the bulk cobble</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Cobble_K_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on K nuclide content in % of the bulk cobble</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Cobble_Th</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Th nuclide content in ppm of the bulk cobble</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Cobble_Th_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on Th nuclide content in ppm of the bulk cobble</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Cobble_U</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> U nuclide content in ppm of the bulk cobble</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CobbleU_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on U nuclide content in ppm of the bulk cobble</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>GrainSize</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> average grain size in ¬µm of the grains used for dating</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Density</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Density of the cobble. Default is 2.7 g cm^-3</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CobbleDiameter</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Diameter of the cobble in cm.</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Sed_K</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> K nuclide content in % of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Sed_K_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on K nuclide content in % of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Sed_Th</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Th nuclide content in ppm of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Sed_Th_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on Th nuclide content in ppm of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Sed_U</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> U nuclide content in ppm of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Sed_U_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on U nuclide content in ppm of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>GrainSize</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> average grain size of the sediment matrix</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WaterContent</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> mean water content of the sediment matrix in %</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WaterContent_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> relative error on water content
</td>
</tr>

</table>

<p><strong>Water content</strong>
The water content provided by the user should be calculated according to:
</p>
<p style="text-align: center;"><code class="reqn">(Wet\_weight - Dry\_weight) / Dry\_weight * 100</code>
</p>



<h3>Value</h3>

<p>The function returns an <a href="#topic+RLum.Results-class">RLum.Results</a> object for which the first element
is a <a href="base.html#topic+matrix">matrix</a> (<code>DataIndividual</code>) that gives the dose rate results for each slice
for each decay chain individually, for both, the cobble dose rate and the sediment
dose rate. The second element is also a <a href="base.html#topic+matrix">matrix</a> (<code>DataComponent</code>) that gives
the total beta and gamma-dose rates for the cobble and the adjacent sediment
for each slice of the cobble.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Riedesel, S., Autzen, M., 2025. calc_CobbleDoseRate(): Calculate dose rate of slices in a spherical cobble. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Svenja Riedesel, Aberystwyth University (United Kingdom) <br />
Martin Autzen, DTU NUTECH Center for Nuclear Technologies (Denmark)
, RLum Developer Team</p>


<h3>References</h3>

<p>Riedesel, S., Autzen, M., 2020. Beta and gamma dose rate attenuation in rocks and sediment.
Radiation Measurements 133, 106295.
</p>


<h3>See Also</h3>

<p><a href="#topic+convert_Concentration2DoseRate">convert_Concentration2DoseRate</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load example data
data("ExampleData.CobbleData", envir = environment())

## run function
calc_CobbleDoseRate(ExampleData.CobbleData)

</code></pre>

<hr>
<h2 id='calc_CommonDose'>Apply the (un-)logged common age model after Galbraith et al. (1999) to a
given De distribution</h2><span id='topic+calc_CommonDose'></span>

<h3>Description</h3>

<p>Function to calculate the common dose of a De distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_CommonDose(data, sigmab, log = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_CommonDose_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[,1])</code> and De error <code>(data[,2])</code></p>
</td></tr>
<tr><td><code id="calc_CommonDose_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
additional spread in De values.
This value represents the expected overdispersion in the data should the sample be
well-bleached (Cunningham &amp; Walling 2012, p. 100).
<strong>NOTE</strong>: For the logged model (<code>log = TRUE</code>) this value must be
a fraction, e.g. 0.2 (= 20%). If the un-logged model is used (<code>log = FALSE</code>),
<code>sigmab</code> must be provided in the same absolute units of the De values
(seconds or Gray).</p>
</td></tr>
<tr><td><code id="calc_CommonDose_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
fit the (un-)logged central age model to De data</p>
</td></tr>
<tr><td><code id="calc_CommonDose_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>(Un-)logged model</strong>
</p>
<p>When <code>log = TRUE</code> this function
calculates the weighted mean of logarithmic De values. Each of the estimates
is weighted by the inverse square of its relative standard error. The
weighted mean is then transformed back to the dose scale (Galbraith &amp;
Roberts 2012, p. 14).
</p>
<p>The log transformation is not applicable if the
De estimates are close to zero or negative. In this case the un-logged model
can be applied instead (<code>log = FALSE</code>). The weighted mean is then
calculated using the un-logged estimates of De and their absolute standard
error (Galbraith &amp; Roberts 2012, p. 14).
</p>


<h3>Value</h3>

<p>Returns a terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following element:
</p>
<table role = "presentation">
<tr><td><code>$summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>$data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>$args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>$call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_CommonDose(): Apply the (un-)logged common age model after Galbraith et al. (1999) to a given De distribution. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Galbraith, R.F. &amp; Laslett, G.M., 1993. Statistical models for
mixed fission track ages. Nuclear Tracks Radiation Measurements 4, 459-470.
</p>
<p>Galbraith, R.F., Roberts, R.G., Laslett, G.M., Yoshida, H. &amp; Olley,
J.M., 1999. Optical dating of single grains of quartz from Jinmium rock
shelter, northern Australia. Part I: experimental design and statistical
models.  Archaeometry 41, 339-364.
</p>
<p>Galbraith, R.F. &amp; Roberts, R.G., 2012. Statistical aspects of equivalent dose and error calculation and
display in OSL dating: An overview and some recommendations. Quaternary
Geochronology 11, 1-27.
</p>
<p><strong>Further reading</strong>
</p>
<p>Arnold, L.J. &amp; Roberts, R.G., 2009. Stochastic modelling of multi-grain equivalent dose
(De) distributions: Implications for OSL dating of sediment mixtures.
Quaternary Geochronology 4, 204-230.
</p>
<p>Bailey, R.M. &amp; Arnold, L.J., 2006. Statistical modelling of single grain quartz De distributions and an
assessment of procedures for estimating burial dose. Quaternary Science
Reviews 25, 2475-2502.
</p>
<p>Cunningham, A.C. &amp; Wallinga, J., 2012. Realizing the potential of fluvial archives using robust OSL chronologies.
Quaternary Geochronology 12, 98-106.
</p>
<p>Rodnight, H., Duller, G.A.T., Wintle, A.G. &amp; Tooth, S., 2006. Assessing the reproducibility and accuracy
of optical dating of fluvial deposits.  Quaternary Geochronology, 1 109-120.
</p>
<p>Rodnight, H., 2008. How many equivalent dose values are needed to
obtain a reproducible distribution?. Ancient TL 26, 3-10.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_CentralDose">calc_CentralDose</a>, <a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>,
<a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.DeValues, envir = environment())

## apply the common dose model
calc_CommonDose(ExampleData.DeValues$CA1)

</code></pre>

<hr>
<h2 id='calc_CosmicDoseRate'>Calculate the cosmic dose rate</h2><span id='topic+calc_CosmicDoseRate'></span>

<h3>Description</h3>

<p>This function calculates the cosmic dose rate taking into account the soft-
and hard-component of the cosmic ray flux and allows corrections for
geomagnetic latitude, altitude above sea-level and geomagnetic field
changes.
</p>
<p>This function calculates the total cosmic dose rate considering both the
soft- and hard-component of the cosmic ray flux.
</p>
<p><strong>Internal calculation steps</strong>
</p>
<p>(1)
Calculate total depth of all absorber in hg/cm¬≤ (1 hg/cm¬≤ = 100 g/cm¬≤)
</p>
<p style="text-align: center;"><code class="reqn">absorber = depth_1*density_1 + depth_2*density_2 + ... + depth_n*density_n</code>
</p>

<p>(2)
If <code>half.depth = TRUE</code>
</p>
<p style="text-align: center;"><code class="reqn">absorber = absorber/2</code>
</p>

<p>(3)
Calculate cosmic dose rate at sea-level and 55 deg. latitude
</p>
<p>a) If absorber is &gt; 167 g/cm¬≤ (only hard-component; Allkofer et al.  1975):
apply equation given by Prescott &amp; Hutton (1994) (c.f. Barbouti &amp; Rastin
1983)
</p>
<p style="text-align: center;"><code class="reqn">D0 = C/(((absorber+d)^\alpha+a)*(absober+H))*exp(-B*absorber)</code>
</p>

<p>b) If absorber is &lt; 167 g/cm¬≤ (soft- and hard-component): derive D0 from
Fig. 1 in Prescott &amp; Hutton (1988).
</p>
<p>(4)
Calculate geomagnetic latitude (Prescott &amp; Stephan 1982, Prescott &amp;
Hutton 1994)
</p>
<p style="text-align: center;"><code class="reqn">\lambda = arcsin(0.203*cos(latitude)*cos(longitude-291)+0.979*
sin(latitude))</code>
</p>

<p>(5)
Apply correction for geomagnetic latitude and altitude above sea-level.
Values for F, J and H were read from Fig. 3 shown in Prescott &amp; Stephan
(1982) and fitted with 3-degree polynomials for lambda &lt; 35 degree and a
linear fit for lambda &gt; 35 degree.
</p>
<p style="text-align: center;"><code class="reqn">Dc = D0*(F+J*exp((altitude/1000)/H))</code>
</p>

<p>(6)
Optional: Apply correction for geomagnetic field changes in the last
0-80 ka (Prescott &amp; Hutton 1994). Correction and altitude factors are given
in Table 1 and Fig. 1 in Prescott &amp; Hutton (1994). Values for altitude
factor were fitted with a 2-degree polynomial. The altitude factor is
operated on the decimal part of the correction factor.
</p>
<p style="text-align: center;"><code class="reqn">Dc' = Dc*correctionFactor</code>
</p>

<p><strong>Usage of <code>depth</code> and <code>density</code></strong>
</p>
<p>(1) If only one value for depth and density is provided, the cosmic dose
rate is calculated for exactly one sample and one absorber as overburden
(i.e. <code>depth*density</code>).
</p>
<p>(2) In some cases it might be useful to calculate the cosmic dose rate for a
sample that is overlain by more than one absorber, e.g. in a profile with
soil layers of different thickness and a distinct difference in density.
This can be calculated by providing a matching number of values for
<code>depth</code> and <code>density</code> (e.g. <code style="white-space: pre;">&#8288;depth = c(1, 2), density = c(1.7, 2.4)&#8288;</code>)
</p>
<p>(3) Another possibility is to calculate the cosmic dose rate for more than
one sample of the same profile. This is done by providing more than one
values for <code>depth</code> and only one for <code>density</code>. For example,
<code>depth = c(1, 2, 3)</code> and <code>density = 1.7</code> will calculate the cosmic dose rate
for three samples in 1, 2 and 3 m depth in a sediment of density 1.7 g/cm¬≥.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_CosmicDoseRate(
  depth,
  density,
  latitude,
  longitude,
  altitude,
  corr.fieldChanges = FALSE,
  est.age = NA,
  half.depth = FALSE,
  error = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_CosmicDoseRate_+3A_depth">depth</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
depth of overburden (m). For more than one absorber use <br />
<code>c(depth_1, depth_2, ..., depth_n)</code></p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_density">density</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
average overburden density (g/cm¬≥). For more than one absorber use <br />
<code>c(density_1, density_2, ..., density_n)</code></p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_latitude">latitude</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
latitude (decimal degree), N positive</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_longitude">longitude</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
longitude (decimal degree), E positive</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_altitude">altitude</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
altitude (m above sea-level)</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_corr.fieldchanges">corr.fieldChanges</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
correct for geomagnetic field changes after Prescott &amp; Hutton (1994).
Apply only when justified by the data.</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_est.age">est.age</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
estimated age range (ka) for geomagnetic field change correction (0-80 ka allowed)</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_half.depth">half.depth</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
How to overcome with varying overburden thickness. If <code>TRUE</code> only half the
depth is used for calculation. Apply only when justified, i.e. when a constant
sedimentation rate can safely be assumed.</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_error">error</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
general error (percentage) to be implemented on corrected cosmic dose rate estimate</p>
</td></tr>
<tr><td><code id="calc_CosmicDoseRate_+3A_...">...</code></td>
<td>
<p>further arguments (<code>verbose</code> to disable/enable console output).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following element:
</p>
<table role = "presentation">
<tr><td><code>summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant calculation results.</p>
</td></tr>
<tr><td><code>args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.5.3
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_CosmicDoseRate(): Calculate the cosmic dose rate. Function version 0.5.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Despite its universal use, the equation to calculate the cosmic dose
rate provided by Prescott &amp; Hutton (1994) is falsely stated to be valid from
the surface to 10^4 hg/cm¬≤ of standard rock. The original expression by
Barbouti &amp; Rastin (1983) only considers the muon flux (i.e. hard-component)
and is, by their own definition, only valid for depths between 10-10^4
hg/cm¬≤.
</p>
<p>Thus, for near-surface samples (i.e. for depths &lt; 167 g/cm¬≤) the equation
of Prescott &amp; Hutton (1994) underestimates the total cosmic dose rate, as it
neglects the influence of the soft-component of the cosmic ray flux. For
samples at zero depth and at sea-level the underestimation can be as large
as ~0.1 Gy/ka. In a previous article, Prescott &amp; Hutton (1988) give another
approximation of Barbouti &amp; Rastin's equation in the form of
</p>
<p style="text-align: center;"><code class="reqn">D = 0.21*exp(-0.070*absorber+0.0005*absorber^2)</code>
</p>

<p>which is valid for depths between 150-5000 g/cm¬≤. For shallower depths (&lt;
150 g/cm¬≤) they provided a graph (Fig. 1) from which the dose rate can be
read.
</p>
<p>As a result, this function employs the equation of Prescott &amp; Hutton (1994)
only for depths &gt; 167 g/cm¬≤, i.e. only for the hard-component of the cosmic
ray flux. Cosmic dose rate values for depths &lt; 167 g/cm¬≤ were obtained from
the &quot;AGE&quot; program (Gruen 2009) and fitted with a 6-degree polynomial curve
(and hence reproduces the graph shown in Prescott &amp; Hutton 1988). However,
these values assume an average overburden density of 2 g/cm¬≥.
</p>
<p>It is currently not possible to obtain more precise cosmic dose rate values
for near-surface samples as there is no equation known to the author of this
function at the time of writing.
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Allkofer, O.C., Carstensen, K., Dau, W.D., Jokisch, H., 1975.
Letter to the editor. The absolute cosmic ray flux at sea level. Journal of
Physics G: Nuclear and Particle Physics 1, L51-L52.
</p>
<p>Barbouti, A.I., Rastin, B.C., 1983. A study of the absolute intensity of muons at sea level
and under various thicknesses of absorber. Journal of Physics G: Nuclear and
Particle Physics 9, 1577-1595.
</p>
<p>Crookes, J.N., Rastin, B.C., 1972. An
investigation of the absolute intensity of muons at sea-level. Nuclear
Physics B 39, 493-508.
</p>
<p>Gruen, R., 2009. The &quot;AGE&quot; program for the
calculation of luminescence age estimates. Ancient TL 27, 45-46.
</p>
<p>Prescott, J.R., Hutton, J.T., 1988. Cosmic ray and gamma ray dosimetry for
TL and ESR. Nuclear Tracks and Radiation Measurements 14, 223-227.
</p>
<p>Prescott, J.R., Hutton, J.T., 1994. Cosmic ray contributions to dose rates
for luminescence and ESR dating: large depths and long-term time variations.
Radiation Measurements 23, 497-500.
</p>
<p>Prescott, J.R., Stephan, L.G., 1982. The contribution of cosmic radiation to the environmental dose for
thermoluminescence dating. Latitude, altitude and depth dependences. PACT 6, 17-25.
</p>


<h3>See Also</h3>

<p><a href="#topic+BaseDataSet.CosmicDoseRate">BaseDataSet.CosmicDoseRate</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) calculate cosmic dose rate (one absorber)
calc_CosmicDoseRate(depth = 2.78, density = 1.7,
                    latitude = 38.06451, longitude = 1.49646,
                    altitude = 364, error = 10)

##(2a) calculate cosmic dose rate (two absorber)
calc_CosmicDoseRate(depth = c(5.0, 2.78), density = c(2.65, 1.7),
                    latitude = 38.06451, longitude = 1.49646,
                    altitude = 364, error = 10)

##(2b) calculate cosmic dose rate (two absorber) and
##correct for geomagnetic field changes
calc_CosmicDoseRate(depth = c(5.0, 2.78), density = c(2.65, 1.7),
                    latitude = 12.04332, longitude = 4.43243,
                    altitude = 364, corr.fieldChanges = TRUE,
                    est.age = 67, error = 15)


##(3) calculate cosmic dose rate and export results to .csv file
#calculate cosmic dose rate and save to variable
results&lt;- calc_CosmicDoseRate(depth = 2.78, density = 1.7,
                              latitude = 38.06451, longitude = 1.49646,
                              altitude = 364, error = 10)

# the results can be accessed by
get_RLum(results, "summary")

#export results to .csv file - uncomment for usage
#write.csv(results, file = "c:/users/public/results.csv")

##(4) calculate cosmic dose rate for 6 samples from the same profile
##    and save to .csv file
#calculate cosmic dose rate and save to variable
results&lt;- calc_CosmicDoseRate(depth = c(0.1, 0.5 , 2.1, 2.7, 4.2, 6.3),
                              density = 1.7, latitude = 38.06451,
                              longitude = 1.49646, altitude = 364,
                              error = 10)

#export results to .csv file - uncomment for usage
#write.csv(results, file = "c:/users/public/results_profile.csv")

</code></pre>

<hr>
<h2 id='calc_EED_Model'>Modelling Exponential Exposure Distribution</h2><span id='topic+calc_EED_Model'></span>

<h3>Description</h3>

<p>Modelling incomplete and heterogeneous bleaching of mobile grains partially
exposed to the light, an implementation of the EED model proposed by Guibert et al. (2019).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_EED_Model(
  data,
  D0 = 120L,
  expected_dose,
  MinIndivDose = NULL,
  MaxIndivDose = NULL,
  kappa = NULL,
  sigma_distr = NULL,
  n.simul = 5000L,
  n.minSimExp = 50L,
  sample_name = "",
  method_control = list(),
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_EED_Model_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): input data consisting of two columns, the De and the
SE(De). Values are expected in Gy</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_d0">D0</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): D0 value (in Gy), defining the
characterisation behaviour of the quartz.</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_expected_dose">expected_dose</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): expected equivalent dose</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_minindivdose">MinIndivDose</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): value specifying the minimum dose taken into
account for the plateau. <code>NULL</code> applies all values.</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_maxindivdose">MaxIndivDose</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): value specifying the maximum dose taken into
account for the plateau. <code>NULL</code> applies all values.</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_kappa">kappa</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): positive dimensionless exposure parameter
characterising the bleaching state of the grains. Low values (&lt; 10) indicate
poor bleaching</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_sigma_distr">sigma_distr</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): positive dose rate parameter, representing the
dose variability to which the grains were exposed ##TODO perhaps it should be renamed</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_n.simul">n.simul</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): number of simulations</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_n.minsimexp">n.minSimExp</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): number of MC runs for calculating the uncertainty
contribution from the sampling</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_sample_name">sample_name</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): name of the sample</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>with default</em>): additional deep control parameters, parameters
need to be provided as named list, see details</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_EED_Model_+3A_...">...</code></td>
<td>
<p>further parameters that can be passed to better control the plot output. Support arguments
are <code>xlab</code>, <code>xlim</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is an implementation and enhancement of the scripts used in
Guibert et al. (2019). The implementation supports a semi-automated estimation
of the parameters <code>kappa</code> and <code>sigma_distr</code>. If set to <code>NULL</code>, a surface interpolation
is used to estimated those values.
</p>
<p><strong>Method control parameters</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>ARGUMENT</strong> </td><td style="text-align: left;"> <strong>FUNCTION</strong> </td><td style="text-align: left;"> <strong>DEFAULT</strong> </td><td style="text-align: left;"> <strong>DESCRIPTION</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower</code>  </td><td style="text-align: left;">  - </td><td style="text-align: left;"> <code>c(0.1,0,0)</code> </td><td style="text-align: left;"> set lower bounds for kappa, sigma, and the expected De in auto mode </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper</code> </td><td style="text-align: left;">  - </td><td style="text-align: left;"> <code>c(1000,2)</code> </td><td style="text-align: left;"> set upper bounds for kappa, sigma, and the expected De in auto mode </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>iter_max</code> </td><td style="text-align: left;"> - </td><td style="text-align: left;"> <code>1000</code> </td><td style="text-align: left;"> maximum number for iterations for used to find kappa and sigma </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trace</code> </td><td style="text-align: left;"> - </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> enable/disable terminal trace mode; overwritten by global argument <code>verbose</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trace_plot</code> </td><td style="text-align: left;"> - </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> enable/disable additional trace plot output; overwritten by global argument <code>verbose</code> </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Guibert, P., Kreutzer, S., 2025. calc_EED_Model(): Modelling Exponential Exposure Distribution. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Pierre Guibert, IRAMAT-CRP2A, UMR 5060, Universit√© Bordeaux Montaigne (France),
Sebastian Kreutzer, Geography &amp; Earth Sciences, Aberystwyth University (United Kingdom)
, RLum Developer Team</p>


<h3>References</h3>

<p>Guibert, P., Christophe, C., Urbanova, P., Gu√©rin, G., Blain, S., 2017.
Modelling incomplete and heterogeneous bleaching of mobile grains partially exposed to the
light - Towards a new tool for single grain OSL dating of poorly bleached mortars.
Radiation Measurements 107, 48‚Äì57. <a href="https://doi.org/10.1016/j.radmeas.2017.10.003">doi:10.1016/j.radmeas.2017.10.003</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>, <a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_IEU">calc_IEU</a>,
<a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ExampleData.MortarData, envir = environment())
calc_EED_Model(
 data = MortarData,
 kappa = 14,
 sigma_distr = 0.37,
 expected_dose = 11.7)

## automated estimation of
## sigma_distribution and
## kappa
## Not run: 
 calc_EED_Model(
 data = MortarData,
 kappa = NULL,
 sigma_distr = NULL,
 expected_dose = 11.7)

## End(Not run)

</code></pre>

<hr>
<h2 id='calc_FadingCorr'>Fading Correction after Huntley &amp; Lamothe (2001)</h2><span id='topic+calc_FadingCorr'></span>

<h3>Description</h3>

<p>Apply a fading correction according to Huntley &amp; Lamothe (2001) for a given
<code class="reqn">g</code>-value and a given <code class="reqn">t_{c}</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_FadingCorr(
  age.faded,
  g_value,
  tc,
  tc.g_value = tc,
  n.MC = 10000,
  seed = NULL,
  interval = c(0.01, 500),
  txtProgressBar = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_FadingCorr_+3A_age.faded">age.faded</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> <a href="base.html#topic+vector">vector</a> (<strong>required</strong>):
vector of length 2 containing the uncorrected age and the error in ka
(see example).</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_g_value">g_value</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> (<strong>required</strong>):
either a vector of length 2 containing the g-value and error obtained from
separate fading measurements (see example), or an <a href="#topic+RLum.Results-class">RLum.Results</a>
object produced by <a href="#topic+analyse_FadingMeasurement">analyse_FadingMeasurement</a>. If the latter, the <code>tc</code>
argument is set automatically.</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_tc">tc</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
time in seconds between irradiation and the prompt measurement (cf. Huntley &amp; Lamothe 2001).
The argument is ignored when <code>g_value</code> is an <a href="#topic+RLum.Results-class">RLum.Results</a> object.</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_tc.g_value">tc.g_value</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
time in seconds between irradiation and the prompt measurement used in the
estimation of the g-value. If the g-value was normalised, the normalisation
time (in seconds) should be given, e.g., for a g-value normalised to 2 days,
the value 172800 should be used. If nothing is provided the time is set to
<code>tc</code>, which is usual case for g-values obtained using the
SAR method and <code class="reqn">g</code>-values that have been not normalised to 2 days.</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number of Monte Carlo simulation runs for error estimation.
If <code>n.MC = 'auto'</code> is used the function tries to find a 'stable' error for
the age. See details for further information.
<strong>Note:</strong> This may take a while!</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_seed">seed</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
sets the seed for the random number generator in R using <a href="base.html#topic+set.seed">set.seed</a></p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_interval">interval</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
a vector containing the end-points (age interval) of the interval to be searched for the root in 'ka'.
This argument is passed to the function <a href="stats.html#topic+uniroot">stats::uniroot</a> used for solving the equation.</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar.</p>
</td></tr>
<tr><td><code id="calc_FadingCorr_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function solves the equation used for correcting the fading affected age
including the error for a given <code class="reqn">g</code>-value according to Huntley &amp; Lamothe (2001):
</p>
<p style="text-align: center;"><code class="reqn">
\frac{A_{f}}{A} = 1 - \kappa * \Big[ln(\frac{A}{t_c}) - 1\Big]
</code>
</p>

<p>with <code class="reqn">\kappa</code> defined as
</p>
<p style="text-align: center;"><code class="reqn">
\kappa = \frac{\frac{\mathrm{g\_value}}{ln(10)}}{100}
</code>
</p>

<p><code class="reqn">A</code> and <code class="reqn">A_{f}</code> are given in ka. <code class="reqn">t_c</code> is given in s, however, it
is internally recalculated to ka.
</p>
<p>As the <code class="reqn">g</code>-value slightly depends on the time <code class="reqn">t_{c}</code> between
irradiation and the prompt measurement, a value for <code>tc</code> must always be
provided.
If the <code class="reqn">g</code>-value was normalised to a distinct
time or evaluated with a different tc value (e.g., external irradiation), also
the <code class="reqn">t_{c}</code> value for the <code class="reqn">g</code>-value needs to be provided (argument <code>tc.g_value</code>
and then the <code class="reqn">g</code>-value is recalculated
to <code class="reqn">t_{c}</code> of the measurement used for estimating the age applying the
following equation:
</p>
<p style="text-align: center;"><code class="reqn">\kappa_{tc} = \kappa_{tc.g} / (1 - \kappa_{tc.g} * ln(tc/tc.g))</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">\kappa_{tc.g} = g / 100 / ln(10)</code>
</p>

<p>The error of the fading-corrected age is determined using a Monte Carlo
simulation approach. Solving of the equation is performed using
<a href="stats.html#topic+uniroot">uniroot</a>. Large values for <code>n.MC</code> will significantly
increase the computation time.<br />
</p>
<p><strong><code>n.MC = 'auto'</code></strong>
</p>
<p>The error estimation based on a stochastic process, i.e. for a small number of
MC runs the calculated error varies considerably every time the function is called,
even with the same input values.
The argument option <code>n.MC = 'auto'</code> tries to find a stable value for the standard error, i.e.
the standard deviation of values calculated during the MC runs (<code>age.corr.MC</code>),
within a given precision (2 digits) by increasing the number of MC runs stepwise and
calculating the corresponding error.
</p>
<p>If the determined error does not differ from the 9 values calculated previously
within a precision of (here) 3 digits the calculation is stopped as it is assumed
that the error is stable. Please note that (a) the duration depends on the input
values as well as on the provided computation resources and it may take a while,
(b) the length (size) of the output
vector <code>age.corr.MC</code>, where all the single values produced during the MC runs
are stored, equals the number of MC runs (here termed observations).
</p>
<p>To avoid an endless loop the calculation is stopped if the number of observations
exceeds 10^7.
This limitation can be overwritten by setting the number of MC runs manually,
e.g. <code>n.MC = 10000001</code>. Note: For this case the function is not checking whether the calculated
error is stable.<br />
</p>
<p><strong><code>seed</code></strong>
</p>
<p>This option allows to recreate previously calculated results by setting the seed
for the R random number generator (see <a href="base.html#topic+set.seed">set.seed</a> for details). This option
should not be mixed up with the option <strong><code>n.MC = 'auto'</code></strong>. The results may
appear similar, but they are not comparable!<br />
</p>
<p><strong>FAQ</strong><br />
</p>
<p><strong>Q</strong>: Which <code class="reqn">t_{c}</code> value is expected?<br />
</p>
<p><strong>A</strong>: <code class="reqn">t_{c}</code> is the time in seconds between irradiation and the prompt measurement
applied during your <code class="reqn">D_{e}</code> measurement. However, this <code class="reqn">t_{c}</code> might
differ from the <code class="reqn">t_{c}</code> used for estimating the <code class="reqn">g</code>-value. In the
case of an SAR measurement <code class="reqn">t_{c}</code> should be similar, however,
if it differs, you have to provide this
<code class="reqn">t_{c}</code>  value (the one used for estimating the <code class="reqn">g</code>-value) using
the argument <code>tc.g_value</code>.<br />
</p>
<p><strong>Q</strong>: The function could not find a solution, what should I do?<br />
</p>
<p><strong>A</strong>: This usually happens for model parameters exceeding the boundaries of the
fading correction model (e.g., very high <code class="reqn">g</code>-value). Please check
whether another fading correction model might be more appropriate.
</p>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.<br />
</p>
<p>Slot: <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Comment</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>age.corr</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> Corrected age </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>age.corr.MC</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> MC simulation results with all possible ages from that simulation </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Slot: <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Comment</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>info</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> the original function call
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.4.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. calc_FadingCorr(): Fading Correction after Huntley &amp; Lamothe (2001). Function version 0.4.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Special thanks to S√©bastien Huot for his support and clarification via e-mail.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Huntley, D.J., Lamothe, M., 2001. Ubiquity of anomalous fading
in K-feldspars and the measurement and correction for it in optical dating.
Canadian Journal of Earth Sciences, 38, 1093-1106.
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+analyse_FadingMeasurement">analyse_FadingMeasurement</a>, <a href="#topic+get_RLum">get_RLum</a>, <a href="stats.html#topic+uniroot">uniroot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##run the examples given in the appendix of Huntley and Lamothe, 2001

##(1) faded age: 100 a
results &lt;- calc_FadingCorr(
   age.faded = c(0.1,0),
   g_value = c(5.0, 1.0),
   tc = 2592000,
   tc.g_value = 172800,
   n.MC = 100)

##(2) faded age: 1 ka
results &lt;- calc_FadingCorr(
   age.faded = c(1,0),
   g_value = c(5.0, 1.0),
   tc = 2592000,
   tc.g_value = 172800,
   n.MC = 100)

##(3) faded age: 10.0 ka
results &lt;- calc_FadingCorr(
   age.faded = c(10,0),
   g_value = c(5.0, 1.0),
   tc = 2592000,
   tc.g_value = 172800,
   n.MC = 100)

##access the last output
get_RLum(results)

</code></pre>

<hr>
<h2 id='calc_FastRatio'>Calculate the Fast Ratio for CW-OSL curves</h2><span id='topic+calc_FastRatio'></span>

<h3>Description</h3>

<p>Function to calculate the fast ratio of quartz CW-OSL single grain or single
aliquot curves after Durcan &amp; Duller (2011).
</p>
<p>This function follows the equations of Durcan &amp; Duller (2011). The energy
required to reduce the fast and medium quartz OSL components to <code>x</code> and
<code>x2</code> % respectively using eq. 3 to determine channels L2 and L3 (start
and end). The fast ratio is then calculated from: <code class="reqn">(L1-L3)/(L2-L3)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_FastRatio(
  object,
  stimulation.power = 30.6,
  wavelength = 470,
  sigmaF = 2.6e-17,
  sigmaM = 4.28e-18,
  Ch_L1 = 1,
  Ch_L2 = NULL,
  Ch_L3 = NULL,
  x = 1,
  x2 = 0.1,
  dead.channels = c(0, 0),
  fitCW.sigma = FALSE,
  fitCW.curve = FALSE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_FastRatio_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
x, y data of measured values (time and counts).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_stimulation.power">stimulation.power</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Stimulation power in mW/cm¬≤</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_wavelength">wavelength</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Stimulation wavelength in nm</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_sigmaf">sigmaF</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Photoionisation cross-section (cm¬≤) of the fast component.
Default value after Durcan &amp; Duller (2011).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_sigmam">sigmaM</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Photoionisation cross-section (cm¬≤) of the medium component.
Default value after Durcan &amp; Duller (2011).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_ch_l1">Ch_L1</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
An integer specifying the channel for L1.</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_ch_l2">Ch_L2</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
An integer specifying the channel for L2.</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_ch_l3">Ch_L3</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
A vector of length 2 with integer values specifying the start and end
channels for L3 (e.g., <code>c(40, 50)</code>), with the second component greater
than or equal to the first.</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_x">x</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Percentage of signal remaining from the fast component.
Used to define the location of L2 and L3 (start).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_x2">x2</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Percentage of signal remaining from the medium component.
Used to define the location of L3 (end).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_dead.channels">dead.channels</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Vector of length 2 in the form of <code>c(x, y)</code>.
Channels that do not contain OSL data, i.e. at the start or end of measurement.</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_fitcw.sigma">fitCW.sigma</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
fit CW-OSL curve using <a href="#topic+fit_CWCurve">fit_CWCurve</a> to calculate <code>sigmaF</code> and <code>sigmaM</code> (<strong>experimental</strong>).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_fitcw.curve">fitCW.curve</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
fit CW-OSL curve using <a href="#topic+fit_CWCurve">fit_CWCurve</a> and derive the counts of L2 and L3
from the fitted OSL curve (<strong>experimental</strong>).</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_FastRatio_+3A_...">...</code></td>
<td>
<p>available options: <code>verbose</code> (<a href="base.html#topic+logical">logical</a>).
Further arguments passed to <a href="#topic+fit_CWCurve">fit_CWCurve</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot (<em>optional</em>) and an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
The slot <code>data</code> contains a <a href="base.html#topic+list">list</a> with the following elements:
</p>
<table role = "presentation">
<tr><td><code>summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant results</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p>the original input data</p>
</td></tr>
<tr><td><code>fit</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> object if either <code>fitCW.sigma</code> or <code>fitCW.curve</code> is <code>TRUE</code></p>
</td></tr>
<tr><td><code>args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> of used arguments</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
</table>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>King, G.E., Durcan, J., Burow, C., 2025. calc_FastRatio(): Calculate the Fast Ratio for CW-OSL curves. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Georgina E. King, University of Bern (Switzerland) <br />
Julie A. Durcan, University of Oxford (United Kingdom) <br />
Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Durcan, J.A. &amp; Duller, G.A.T., 2011. The fast ratio: A rapid measure for testing
the dominance of the fast component in the initial OSL signal from quartz.
Radiation Measurements 46, 1065-1072.
</p>
<p>Madsen, A.T., Duller, G.A.T., Donnelly, J.P., Roberts, H.M. &amp; Wintle, A.G., 2009.
A chronology of hurricane landfalls at Little Sippewissett Marsh, Massachusetts, USA,
using optical dating. Geomorphology 109, 36-45.
</p>
<p><strong>Further reading</strong>
</p>
<p>Steffen, D., Preusser, F. &amp; Schlunegger, 2009. OSL quartz age underestimation
due to unstable signal components. Quaternary Geochronology 4, 353-362.
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_CWCurve">fit_CWCurve</a>, <a href="#topic+get_RLum">get_RLum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>,
<a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load example CW-OSL curve
data("ExampleData.CW_OSL_Curve")

# calculate the fast ratio w/o further adjustments
res &lt;- calc_FastRatio(ExampleData.CW_OSL_Curve)

# show the summary table
get_RLum(res)

</code></pre>

<hr>
<h2 id='calc_FiniteMixture'>Apply the finite mixture model (FMM) after Galbraith (2005) to a given De
distribution</h2><span id='topic+calc_FiniteMixture'></span>

<h3>Description</h3>

<p>This function fits a k-component mixture to a De distribution with differing
known standard errors. Parameters (doses and mixing proportions) are
estimated by maximum likelihood assuming that the log dose estimates are
from a mixture of normal distributions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_FiniteMixture(
  data,
  sigmab,
  n.components,
  grain.probability = FALSE,
  pdf.weight = TRUE,
  pdf.sigma = "sigmab",
  pdf.colors = "gray",
  plot.proportions = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_FiniteMixture_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[,1])</code> and De error <code>(values[,2])</code></p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
spread in De values given as a fraction (e.g. 0.2). This value represents the expected
overdispersion in the data should the sample be well-bleached
(Cunningham &amp; Wallinga 2012, p. 100).</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_n.components">n.components</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
number of components to be fitted. If a vector is provided (e.g. <code>c(2:8)</code>) the
finite mixtures for 2, 3 ... 8 components are calculated and a plot and a
statistical evaluation of the model performance (BIC score and maximum
log-likelihood) is provided.</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_grain.probability">grain.probability</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
prints the estimated probabilities of which component each grain is in</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_pdf.weight">pdf.weight</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
weight the probability density functions by the components proportion (applies only
when a vector is provided for <code>n.components</code>)</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_pdf.sigma">pdf.sigma</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
if <code>"sigmab"</code> the components normal distributions are plotted with a common standard
deviation (i.e. <code>sigmab</code>) as assumed by the FFM. Alternatively,
<code>"se"</code> takes the standard error of each component for the sigma
parameter of the normal distribution</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_pdf.colors">pdf.colors</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
colour coding of the components in the plot.
Possible options are <code>"gray"</code>, <code>"colors"</code> and <code>"none"</code></p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_plot.proportions">plot.proportions</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
plot <a href="graphics.html#topic+barplot">graphics::barplot</a> showing the proportions of components if
<code>n.components</code> a vector with a length &gt; 1 (e.g., <code>n.components = c(2:3)</code>)</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the  plot output.</p>
</td></tr>
<tr><td><code id="calc_FiniteMixture_+3A_...">...</code></td>
<td>
<p>further arguments to pass. See details for their usage.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This model uses the maximum likelihood and Bayesian Information Criterion
(BIC) approaches.
</p>
<p>Indications of overfitting are:
</p>

<ul>
<li><p> increasing BIC
</p>
</li>
<li><p> repeated dose estimates
</p>
</li>
<li><p> covariance matrix not positive definite
</p>
</li>
<li><p> covariance matrix produces <code>NaN</code>
</p>
</li>
<li><p> convergence problems
</p>
</li></ul>

<p><strong>Plot</strong>
</p>
<p>If a vector (<code>c(k.min:k.max)</code>) is provided
for <code>n.components</code> a plot is generated showing the k components
equivalent doses as normal distributions. By default <code>pdf.weight</code> is
set to <code>FALSE</code>, so that the area under each normal distribution is
always 1. If <code>TRUE</code>, the probability density functions are weighted by
the components proportion for each iteration of k components, so the sum of
areas of each component equals 1. While the density values are on the same
scale when no weights are used, the y-axis are individually scaled if the
probability density are weighted by the components proportion.<br />
The standard deviation (sigma) of the normal distributions is by default
determined by a common <code>sigmab</code> (see <code>pdf.sigma</code>). For
<code>pdf.sigma = "se"</code> the standard error of each component is taken
instead.<br />
The stacked <a href="graphics.html#topic+barplot">graphics::barplot</a> shows the proportion of each component (in
per cent) calculated by the FMM. The last plot shows the achieved BIC scores
and maximum log-likelihood estimates for each iteration of k.
</p>


<h3>Value</h3>

<p>Returns a plot (<em>optional</em>) and terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following elements:
</p>
<table role = "presentation">
<tr><td><code>.$summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>.$data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>.$args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>.$call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>.$mle</code></td>
<td>
<p> covariance matrices of the log likelihoods</p>
</td></tr>
<tr><td><code>.$BIC</code></td>
<td>
<p> BIC score</p>
</td></tr>
<tr><td><code>.$llik</code></td>
<td>
<p> maximum log likelihood</p>
</td></tr>
<tr><td><code>.$grain.probability</code></td>
<td>
<p> probabilities of a grain belonging to a component</p>
</td></tr>
<tr><td><code>.$components</code></td>
<td>
<p><a href="base.html#topic+matrix">matrix</a> estimates of the de, de error and proportion for each component</p>
</td></tr>
<tr><td><code>.$single.comp</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> single component FFM estimate</p>
</td></tr>
</table>
<p>If a vector for <code>n.components</code> is provided (e.g.  <code>c(2:8)</code>),
<code>mle</code> and <code>grain.probability</code> are lists containing matrices of the
results for each iteration of the model.
</p>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.4.2
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_FiniteMixture(): Apply the finite mixture model (FMM) after Galbraith (2005) to a given De distribution. Function version 0.4.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany) <br />
Based on a rewritten S script of Rex Galbraith, 2006.
, RLum Developer Team</p>


<h3>References</h3>

<p>Galbraith, R.F. &amp; Green, P.F., 1990. Estimating the component
ages in a finite mixture. Nuclear Tracks and Radiation Measurements 17,
197-206.
</p>
<p>Galbraith, R.F. &amp; Laslett, G.M., 1993. Statistical models
for mixed fission track ages. Nuclear Tracks Radiation Measurements 4,
459-470.
</p>
<p>Galbraith, R.F. &amp; Roberts, R.G., 2012. Statistical aspects of
equivalent dose and error calculation and display in OSL dating: An overview
and some recommendations. Quaternary Geochronology 11, 1-27.
</p>
<p>Roberts, R.G., Galbraith, R.F., Yoshida, H., Laslett, G.M. &amp; Olley, J.M., 2000.
Distinguishing dose populations in sediment mixtures: a test of single-grain
optical dating procedures using mixtures of laboratory-dosed quartz.
Radiation Measurements 32, 459-465.
</p>
<p>Galbraith, R.F., 2005. Statistics for Fission Track Analysis, Chapman &amp; Hall/CRC, Boca Raton.
</p>
<p><strong>Further reading</strong>
</p>
<p>Arnold, L.J. &amp; Roberts, R.G., 2009. Stochastic
modelling of multi-grain equivalent dose (De) distributions: Implications
for OSL dating of sediment mixtures. Quaternary Geochronology 4,
204-230.
</p>
<p>Cunningham, A.C. &amp; Wallinga, J., 2012. Realizing the
potential of fluvial archives using robust OSL chronologies. Quaternary
Geochronology 12, 98-106.
</p>
<p>Rodnight, H., Duller, G.A.T., Wintle, A.G. &amp;
Tooth, S., 2006. Assessing the reproducibility and accuracy of optical
dating of fluvial deposits.  Quaternary Geochronology 1, 109-120.
</p>
<p>Rodnight, H. 2008. How many equivalent dose values are needed to obtain a
reproducible distribution?. Ancient TL 26, 3-10.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_CentralDose">calc_CentralDose</a>, <a href="#topic+calc_CommonDose">calc_CommonDose</a>,
<a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.DeValues, envir = environment())

## (1) apply the finite mixture model
## NOTE: the data set is not suitable for the finite mixture model,
## which is why a very small sigmab is necessary
calc_FiniteMixture(ExampleData.DeValues$CA1,
                   sigmab = 0.2, n.components = 2,
                   grain.probability = TRUE)

## (2) repeat the finite mixture model for 2, 3 and 4 maximum number of fitted
## components and save results
## NOTE: The following example is computationally intensive. Please un-comment
## the following lines to make the example work.
FMM&lt;- calc_FiniteMixture(ExampleData.DeValues$CA1,
                         sigmab = 0.2, n.components = c(2:4),
                         pdf.weight = TRUE)

## show structure of the results
FMM

## show the results on equivalent dose, standard error and proportion of
## fitted components
get_RLum(object = FMM, data.object = "components")

</code></pre>

<hr>
<h2 id='calc_FuchsLang2001'>Apply the model after Fuchs &amp; Lang (2001) to a given De distribution</h2><span id='topic+calc_FuchsLang2001'></span>

<h3>Description</h3>

<p>This function applies the method according to Fuchs &amp; Lang (2001) for
heterogeneously bleached samples with a given coefficient of variation
threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_FuchsLang2001(data, cvThreshold = 5, startDeValue = 1, plot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_FuchsLang2001_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[,1])</code> and De error <code>(values[,2])</code></p>
</td></tr>
<tr><td><code id="calc_FuchsLang2001_+3A_cvthreshold">cvThreshold</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
coefficient of variation in percent, as threshold for the method,
e.g. <code>cvThreshold = 3</code>. See details
.</p>
</td></tr>
<tr><td><code id="calc_FuchsLang2001_+3A_startdevalue">startDeValue</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
number of the first aliquot that is used for the calculations</p>
</td></tr>
<tr><td><code id="calc_FuchsLang2001_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_FuchsLang2001_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to <a href="base.html#topic+plot">plot</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Used values</strong>
</p>
<p>If the coefficient of variation (<code>c[v]</code>) of the first
two values is larger than the threshold <code>c[v_threshold]</code>, the first value is
skipped.  Use the <code>startDeValue</code> argument to define a start value for
calculation (e.g. 2nd or 3rd value).
</p>
<p><strong>Basic steps of the approach</strong>
</p>

<ol>
<li><p> Estimate natural relative variation of the sample using a dose recovery test
</p>
</li>
<li><p> Sort the input values in ascending order
</p>
</li>
<li><p> Calculate a running mean, starting with the lowermost two values and add values iteratively.
</p>
</li>
<li><p> Stop if the calculated <code>c[v]</code> exceeds the specified <code>cvThreshold</code>
</p>
</li></ol>



<h3>Value</h3>

<p>Returns a plot (<em>optional</em>) and terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following elements:
</p>
<table role = "presentation">
<tr><td><code>summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>usedDeValues</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> containing the used values for the calculation</p>
</td></tr>
</table>


<h3>Function version</h3>

<p>0.4.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Burow, C., 2025. calc_FuchsLang2001(): Apply the model after Fuchs &amp; Lang (2001) to a given De distribution. Function version 0.4.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Please consider the requirements and the constraints of this method
(see Fuchs &amp; Lang, 2001)
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany) <br />
Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Fuchs, M. &amp; Lang, A., 2001. OSL dating of coarse-grain fluvial
quartz using single-aliquot protocols on sediments from NE Peloponnese,
Greece. In: Quaternary Science Reviews 20, 783-787.
</p>
<p>Fuchs, M. &amp; Wagner, G.A., 2003. Recognition of insufficient bleaching by
small aliquots of quartz for reconstructing soil erosion in Greece.
Quaternary Science Reviews 22, 1161-1167.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>, <a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>, <a href="#topic+calc_CentralDose">calc_CentralDose</a>,
<a href="#topic+calc_CommonDose">calc_CommonDose</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## load example data
data(ExampleData.DeValues, envir = environment())

## calculate De according to Fuchs &amp; Lang (2001)
temp&lt;- calc_FuchsLang2001(ExampleData.DeValues$BT998, cvThreshold = 5)

</code></pre>

<hr>
<h2 id='calc_gSGC'>Calculate De value based on the gSGC by Li et al., 2015</h2><span id='topic+calc_gSGC'></span>

<h3>Description</h3>

<p>Function returns De value and De value error using the global standardised growth
curve (gSGC) assumption proposed by Li et al., 2015 for OSL dating of sedimentary quartz
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_gSGC(
  data,
  gSGC.type = "0-250",
  gSGC.parameters,
  n.MC = 100,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_gSGC_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
input data of providing the following columns: <code>LnTn</code>, <code>LnTn.error</code>, <code>Lr1Tr1</code>, <code>Lr1Tr1.error</code>, <code>Dr1</code>
<strong>Note:</strong> column names are not required. The function expects the input data in the given order</p>
</td></tr>
<tr><td><code id="calc_gSGC_+3A_gsgc.type">gSGC.type</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
define the function parameters that
should be used for the iteration procedure: Li et al., 2015 (Table 2)
presented function parameters for two dose ranges: <code>"0-450"</code> and <code>"0-250"</code></p>
</td></tr>
<tr><td><code id="calc_gSGC_+3A_gsgc.parameters">gSGC.parameters</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
option to provide own function parameters used for fitting as named list.
Nomenclature follows Li et al., 2015, i.e. <code>list(A, A.error, D0, D0.error, c, c.error, Y0, Y0.error, range)</code>, where <code>range</code> is defines the interval
where the function is considered as valid, e.g. <code>range = c(0,250)</code>.<br />
Using this option overwrites the default parameter list of the gSGC, meaning the argument
<code>gSGC.type</code> will be without effect</p>
</td></tr>
<tr><td><code id="calc_gSGC_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number of Monte Carlo simulation runs for error estimation, see details.</p>
</td></tr>
<tr><td><code id="calc_gSGC_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="calc_gSGC_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_gSGC_+3A_...">...</code></td>
<td>
<p>parameters will be passed to the plot output</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The error of the De value is determined using a Monte Carlo simulation approach.
Solving of the equation is realised using <a href="stats.html#topic+uniroot">uniroot</a>.
Large values for <code>n.MC</code> will significantly increase the computation time.
</p>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
</p>
<p><strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong><br />
<code style="white-space: pre;">&#8288;$ De.value&#8288;</code> (<a href="base.html#topic+data.frame">data.frame</a>) <br />
<code>.. $ De</code>  <br />
<code>.. $ De.error</code> <br />
<code>.. $ Eta</code> <br />
<code style="white-space: pre;">&#8288;$ De.MC&#8288;</code> (<a href="base.html#topic+list">list</a>) contains the matrices from the error estimation.<br />
<code style="white-space: pre;">&#8288;$ uniroot&#8288;</code> (<a href="base.html#topic+list">list</a>) contains the <a href="stats.html#topic+uniroot">uniroot</a> outputs of the De estimations<br />
</p>
<p><strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong><br />
'$ call&ldquo; (<a href="base.html#topic+call">call</a>) the original function call
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. calc_gSGC(): Calculate De value based on the gSGC by Li et al., 2015. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Li, B., Roberts, R.G., Jacobs, Z., Li, S.-H., 2015. Potential of establishing
a 'global standardised growth curve' (gSGC) for optical dating of quartz from sediments.
Quaternary Geochronology 27, 94-104. doi:10.1016/j.quageo.2015.02.011
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>, <a href="stats.html#topic+uniroot">uniroot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
results &lt;- calc_gSGC(data = data.frame(
LnTn =  2.361, LnTn.error = 0.087,
Lr1Tr1 = 2.744, Lr1Tr1.error = 0.091,
Dr1 = 34.4))

get_RLum(results, data.object = "De")

</code></pre>

<hr>
<h2 id='calc_gSGC_feldspar'>Calculate Global Standardised Growth Curve (gSGC) for Feldspar MET-pIRIR</h2><span id='topic+calc_gSGC_feldspar'></span>

<h3>Description</h3>

<p>Implementation of the gSGC approach for feldspar MET-pIRIR by Li et al. (2015)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_gSGC_feldspar(
  data,
  gSGC.type = "50LxTx",
  gSGC.parameters,
  n.MC = 100,
  plot = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_gSGC_feldspar_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): data frame with five columns per sample
<code>c("LnTn", "LnTn.error", "Lr1Tr1", "Lr1Tr1.error","Dr1")</code></p>
</td></tr>
<tr><td><code id="calc_gSGC_feldspar_+3A_gsgc.type">gSGC.type</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): growth curve type to be selected
according to Table 3 in Li et al. (2015). Allowed options are
<code>"50LxTx"</code>, <code>"50Lx"</code>, <code>"50Tx"</code>, <code>"100LxTx"</code>, <code>"100Lx"</code>, <code>"100Tx"</code>, <code>"150LxTx"</code>,
<code>"150Lx"</code>, <code>"150Tx"</code>, <code>"200LxTx"</code>, <code>"200Lx"</code>, <code>"200Tx"</code>, <code>"250LxTx"</code>, <code>"250Lx"</code>,
<code>"250Tx"</code></p>
</td></tr>
<tr><td><code id="calc_gSGC_feldspar_+3A_gsgc.parameters">gSGC.parameters</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>): an own parameter set for the
gSGC with the following columns <code>y1</code>, <code>y1_err</code>, <code>D1</code>
<code>D1_err</code>, <code>y2</code>, <code>y2_err</code>, <code>y0</code>, <code>y0_err</code>.</p>
</td></tr>
<tr><td><code id="calc_gSGC_feldspar_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): number of Monte-Carlo runs for the
error calculation</p>
</td></tr>
<tr><td><code id="calc_gSGC_feldspar_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>##TODO
</p>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
</p>
<p><strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong><br />
<code style="white-space: pre;">&#8288;$ df&#8288;</code> (<a href="base.html#topic+data.frame">data.frame</a>) <br />
<code>.. $DE</code> the calculated equivalent dose<br />
<code>.. $DE.ERROR</code> error on the equivalent dose, which is the standard deviation of the MC runs<br />
<code>.. $HPD95_LOWER</code> lower boundary of the highest probability density (95%)<br />
<code>.. $HPD95_UPPER</code> upper boundary of the highest probability density (95%)<br />
<code style="white-space: pre;">&#8288;$ m.MC&#8288;</code> (<a href="base.html#topic+list">list</a>) numeric vector with results from the MC runs.<br />
</p>
<p><strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong><br />
'$ call&ldquo; (<a href="base.html#topic+call">call</a>) the original function call
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Gray, H.J., Kreutzer, S., 2025. calc_gSGC_feldspar(): Calculate Global Standardised Growth Curve (gSGC) for Feldspar MET-pIRIR. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Harrison Gray, USGS (United States),
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Li, B., Roberts, R.G., Jacobs, Z., Li, S.-H., Guo, Y.-J., 2015.
Construction of a ‚Äúglobal standardised growth curve‚Äù (gSGC) for infrared
stimulated luminescence dating of K-feldspar 27, 119‚Äì130. <a href="https://doi.org/10.1016/j.quageo.2015.02.010">doi:10.1016/j.quageo.2015.02.010</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>, <a href="stats.html#topic+uniroot">uniroot</a>, <a href="#topic+calc_gSGC">calc_gSGC</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##test on a generated random sample
n_samples &lt;- 10
data &lt;- data.frame(
  LnTn = rnorm(n=n_samples, mean=1.0, sd=0.02),
  LnTn.error = rnorm(n=n_samples, mean=0.05, sd=0.002),
  Lr1Tr1 = rnorm(n=n_samples, mean=1.0, sd=0.02),
  Lr1Tr1.error = rnorm(n=n_samples, mean=0.05, sd=0.002),
  Dr1 = rep(100,n_samples))

results &lt;- calc_gSGC_feldspar(
  data = data, gSGC.type = "50LxTx",
  plot = FALSE)

plot_AbanicoPlot(results)

</code></pre>

<hr>
<h2 id='calc_HomogeneityTest'>Apply a simple homogeneity test after Galbraith (2003)</h2><span id='topic+calc_HomogeneityTest'></span>

<h3>Description</h3>

<p>A simple homogeneity test for De estimates
For details see Galbraith (2003).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_HomogeneityTest(data, log = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_HomogeneityTest_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[,1])</code> and De error <code>(values[,2])</code></p>
</td></tr>
<tr><td><code id="calc_HomogeneityTest_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
perform the homogeneity test with (un-)logged data</p>
</td></tr>
<tr><td><code id="calc_HomogeneityTest_+3A_...">...</code></td>
<td>
<p>further arguments (for internal compatibility only).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a>-object is returned containing the
following elements:
</p>
<table role = "presentation">
<tr><td><code>summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.3.0
</p>


<h3>How to cite</h3>

<p>Burow, C., Kreutzer, S., 2025. calc_HomogeneityTest(): Apply a simple homogeneity test after Galbraith (2003). Function version 0.3.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany), Sebastian Kreutzer,
IRAMAT-CRP2A, Universit√© Bordeaux Montaigne (France)
, RLum Developer Team</p>


<h3>References</h3>

<p>Galbraith, R.F., 2003. A simple homogeneity test for estimates
of dose obtained using OSL. Ancient TL 21, 75-77.
</p>


<h3>See Also</h3>

<p><a href="stats.html#topic+Chisquare">stats::pchisq</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.DeValues, envir = environment())

## apply the homogeneity test
calc_HomogeneityTest(ExampleData.DeValues$BT998)

## using the data presented by Galbraith (2003)
df &lt;-
 data.frame(
   x = c(30.1, 53.8, 54.3, 29.0, 47.6, 44.2, 43.1),
   y = c(4.8, 7.1, 6.8, 4.3, 5.2, 5.9, 3.0))

calc_HomogeneityTest(df)


</code></pre>

<hr>
<h2 id='calc_Huntley2006'>Apply the Huntley (2006) model</h2><span id='topic+calc_Huntley2006'></span>

<h3>Description</h3>

<p>A function to calculate the expected sample specific fraction of saturation
based on the model of Huntley (2006) using the approach as implemented
in Kars et al. (2008) or Guralnik et al. (2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_Huntley2006(
  data,
  LnTn = NULL,
  rhop,
  ddot,
  readerDdot,
  normalise = TRUE,
  fit.method = c("EXP", "GOK"),
  lower.bounds = c(-Inf, -Inf, -Inf, -Inf),
  summary = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_Huntley2006_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
A <code>data.frame</code> with one of the following structures:
</p>

<ul>
<li><p> A <strong>three column</strong> data frame with numeric values on a) dose (s), b) <code>LxTx</code> and
c) <code>LxTx</code> error.
</p>
</li>
<li><p> If a <strong>two column</strong> data frame is provided it is automatically
assumed that errors on <code>LxTx</code> are missing. A third column will be attached
with an arbitrary 5 % error on the provided <code>LxTx</code> values.
</p>
</li>
<li><p> Can also be a <strong>wide table</strong>, i.e. a <a href="base.html#topic+data.frame">data.frame</a> with a number of columns divisible by 3
and where each triplet has the aforementioned column structure.
</p>
</li></ul>

<div class="sourceCode"><pre>                        (optional)
     | dose (s)| LxTx | LxTx error |
     |  [ ,1]  | [ ,2]|    [ ,3]   |
     |---------|------|------------|
[1, ]|  0      | LnTn | LnTn error | (optional, see arg 'LnTn')
[2, ]|  R1     | L1T1 | L1T1 error |
 ... |    ...  |  ... |     ...    |
[x, ]|  Rx     | LxTx | LxTx error |

</pre></div>
<p><strong>NOTE:</strong> The function assumes the first row of the function to be the
<code>Ln/Tn</code>-value. If you want to provide more than one <code>Ln/Tn</code>-value consider
using the argument <code>LnTn</code>.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_lntn">LnTn</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
This argument should <strong>only</strong> be used to provide more than one <code>Ln/Tn</code>-value.
It assumes a two column data frame with the following structure:
</p>
<div class="sourceCode"><pre>     |  LnTn  |  LnTn error  |
     |  [ ,1] |      [ ,2]   |
     |--------|--------------|
[1, ]| LnTn_1 | LnTn_1 error |
[2, ]| LnTn_2 | LnTn_2 error |
 ... |   ...  |      ...     |
[x, ]| LnTn_x | LnTn_x error |
</pre></div>
<p>The function will calculate a <strong>mean</strong> <code>Ln/Tn</code>-value and uses either the
standard deviation or the highest individual error, whichever is larger. If
another mean value (e.g. a weighted mean or median) or error is preferred,
this value must be calculated beforehand and used in the first row in the
data frame for argument <code>data</code>.
</p>
<p><strong>NOTE:</strong> If you provide <code>LnTn</code>-values with this argument the data frame
for the <code>data</code>-argument <strong>must not</strong> contain any <code>LnTn</code>-values!</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_rhop">rhop</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
The density of recombination centres (<code class="reqn">\rho</code>') and its error (see Huntley 2006),
given as numeric vector of length two. Note that <code class="reqn">\rho</code>' must <strong>not</strong> be
provided as the common logarithm. Example: <code>rhop = c(2.92e-06, 4.93e-07)</code>.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_ddot">ddot</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
Environmental dose rate and its error, given as a numeric vector of length two.
Expected unit: Gy/ka. Example: <code>ddot = c(3.7, 0.4)</code>.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_readerddot">readerDdot</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
Dose rate of the irradiation source of the OSL reader and its error,
given as a numeric vector of length two.
Expected unit: Gy/s. Example: <code>readerDdot = c(0.08, 0.01)</code>.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_normalise">normalise</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): If <code>TRUE</code> (the default) all measured and computed <code class="reqn">\frac{L_x}{T_x}</code> values are normalised by the pre-exponential factor <code>A</code> (see details).</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_fit.method">fit.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
Fit function of the dose response curve. Can either be <code>EXP</code> (the default)
or <code>GOK</code>. Note that <code>EXP</code> (single saturating exponential) is the original
function the model after Huntley (2006) and Kars et al. (2008) was
designed to use. The use of a general-order kinetics function (<code>GOK</code>)
is an experimental adaptation of the model and should be used
with great care.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_lower.bounds">lower.bounds</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
A vector of length 4 that contains the lower bound values to be applied
when fitting the models with <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>. In most cases, the
default values (<code>c(-Inf, -Inf, -Inf, -Inf)</code>) are appropriate for finding
a best fit, but sometimes it may be useful to restrict the lower bounds to
e.g. <code>c(0, 0, 0, 0)</code>. The values of the vectors are, respectively, for
parameters <code>a</code>, <code>D0</code>, <code>c</code> and <code>d</code> in that order (parameter <code>d</code> is ignored
when <code>fit.method = "EXP"</code>). More details can be found in
<a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If <code>TRUE</code> (the default) various parameters provided by the user
and calculated by the model are added as text on the right-hand side of the
plot.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_Huntley2006_+3A_...">...</code></td>
<td>
<p>Further parameters:
</p>

<ul>
<li> <p><code>verbose</code> <a href="base.html#topic+logical">logical</a>: Show or hide console output
</p>
</li>
<li> <p><code>n.MC</code> <a href="base.html#topic+numeric">numeric</a>: Number of Monte Carlo iterations (default = <code>10000</code>).
<strong>Note</strong> that it is generally advised to have a large number of Monte Carlo
iterations for the results to converge. Decreasing the number of iterations
will often result in unstable estimates.
</p>
</li></ul>

<p>All other arguments are passed to <a href="base.html#topic+plot">plot</a> and <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a> (in particular
<code>mode</code> for the fit mode and <code>fit.force_through_origin</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function applies the approach described in Kars et al. (2008) or Guralnik et al. (2015),
which are both developed from the model of Huntley (2006) to calculate the expected sample
specific fraction of saturation of a feldspar and also to calculate fading
corrected age using this model. <code class="reqn">\rho</code>' (<code>rhop</code>), the density of recombination
centres, is a crucial parameter of this model and must be determined
separately from a fading measurement. The function <a href="#topic+analyse_FadingMeasurement">analyse_FadingMeasurement</a>
can be used to calculate the sample specific <code class="reqn">\rho</code>' value.
</p>
<p><strong>Kars et al. (2008) - Single saturating exponential</strong>
</p>
<p>To apply the approach after Kars et al. (2008) use <code>fit.method = "EXP"</code>.
</p>
<p>Firstly, the unfaded <code class="reqn">D_0</code> value is determined through applying equation 5 of
Kars et al. (2008) to the measured <code class="reqn">\frac{L_x}{T_x}</code> data as a function of irradiation
time, and fitting the data with a single saturating exponential of the form:
</p>
<p style="text-align: center;"><code class="reqn">\frac{L_x}{T_x}(t^*) = A  \phi(t^*) \{1 - exp(-\frac{t^*}{D_0}))\}</code>
</p>

<p>where
</p>
<p style="text-align: center;"><code class="reqn">\phi(t^*) = exp(-\rho' ln(1.8  \tilde{s} t^*)^3)</code>
</p>

<p>after King et al. (2016) where <code class="reqn">A</code> is a pre-exponential factor,
<code class="reqn">t^*</code> (s) is the irradiation time, starting at the mid-point of
irradiation (Auclair et al. 2003) and <code class="reqn">\tilde{s}</code> (<code class="reqn">3\times10^{15}</code> s<code class="reqn">^{-1}</code>) is the athermal frequency factor after Huntley (2006). <br />
</p>
<p>Using fit parameters <code class="reqn">A</code> and <code class="reqn">D_0</code>, the function then computes a natural dose
response curve using the environmental dose rate, <code class="reqn">\dot{D}</code> (Gy/s) and equations
<code style="white-space: pre;">&#8288;[1]&#8288;</code> and <code style="white-space: pre;">&#8288;[2]&#8288;</code>. Computed <code class="reqn">\frac{L_x}{T_x}</code> values are then fitted using the
<a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a> function and the laboratory measured LnTn can then
be interpolated onto this curve to determine the fading corrected
<code class="reqn">D_e</code> value, from which the fading corrected age is calculated.
</p>
<p><strong>Guralnik et al. (2015) - General-order kinetics</strong>
</p>
<p>To apply the approach after Guralnik et al. (2015) use <code>fit.method = "GOK"</code>.
</p>
<p>The approach of Guralnik et al. (2015) is very similar to that of
Kars et al. (2008), but instead of using a single saturating exponential
the model fits a general-order kinetics function of the form:
</p>
<p style="text-align: center;"><code class="reqn">\frac{L_x}{T_x}(t^*) = A \phi (t^*)(1 - (1 + (\frac{1}{D_0}) t^* c)^{-1/c})</code>
</p>

<p>where <code class="reqn">A</code>, <code class="reqn">\phi</code>, <code class="reqn">t^*</code> and <code class="reqn">D_0</code> are the same as above and <code class="reqn">c</code> is a
dimensionless kinetic order modifier (cf. equation 10 in
Guralnik et al., 2015).
</p>
<p><strong>Level of saturation</strong>
</p>
<p>The <a href="#topic+calc_Huntley2006">calc_Huntley2006</a> function also calculates the level of saturation (<code class="reqn">\frac{n}{N}</code>)
and the field saturation (i.e. athermal steady state, (n/N)_SS) value for
the sample under investigation using the sample specific <code class="reqn">\rho</code>',
unfaded <code class="reqn">D_0</code> and <code class="reqn">\dot{D}</code> values, following the approach of Kars et al. (2008).
</p>
<p>The computation is done using 1000 equally-spaced points in the interval
[0.01, 3]. This can be controlled by setting option <code>rprime</code>, such as
in <code>rprime = seq(0.01, 3, length.out = 1000)</code> (the default).
</p>
<p><strong>Uncertainties</strong>
</p>
<p>Uncertainties are reported at <code class="reqn">1\sigma</code> and are assumed to be normally
distributed and are estimated using Monte-Carlo re-sampling (<code>n.MC = 1000</code>)
of <code class="reqn">\rho</code>' and <code class="reqn">\frac{L_x}{T_x}</code> during dose response curve fitting, and of <code class="reqn">\rho</code>'
in the derivation of (<code class="reqn">n/N</code>) and (n/N)_SS.
</p>
<p><strong>Age calculated from 2D0 of the simulated natural DRC</strong>
</p>
<p>In addition to the age calculated from the equivalent dose derived from
<code class="reqn">\frac{L_n}{T_n}</code> projected on the simulated natural dose response curve (DRC), this function
also calculates an age from twice the characteristic saturation dose (<code>D0</code>)
of the simulated natural DRC. This can be a useful information for
(over)saturated samples (i.e., no intersect of <code class="reqn">\frac{L_n}{T_n}</code> on the natural DRC)
to obtain at least a &quot;minimum age&quot; estimate of the sample. In the console
output this value is denoted by <em>&quot;Age @2D0 (ka):&quot;</em>.
</p>


<h3>Value</h3>

<p>An <a href="#topic+RLum.Results-class">RLum.Results</a> object is returned:
</p>
<p>Slot: <strong>@data</strong><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>COMMENT</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>results</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> results of the of Kars et al. 2008 model </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> original input data </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Ln</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> Ln and its error </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>LxTx_tables</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> A <code>list</code> of <code>data.frames</code> containing data on dose,
LxTx and LxTx error for each of the dose response curves.
Note that these <strong>do not</strong> contain the natural Ln signal, which is provided separately. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>fits</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> A <code>list</code> of <code>nls</code> objects produced by <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a> when fitting the dose response curves </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Slot: <strong>@info</strong><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>COMMENT</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>call</code> </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>args</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> arguments of the original function call </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.4.5
</p>


<h3>How to cite</h3>

<p>King, G.E., Burow, C., Kreutzer, S., 2025. calc_Huntley2006(): Apply the Huntley (2006) model. Function version 0.4.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function has BETA status, in particular for the GOK implementation. Please verify
your results carefully
</p>


<h3>Author(s)</h3>

<p>Georgina E. King, University of Lausanne (Switzerland) <br />
Christoph Burow, University of Cologne (Germany) <br />
Sebastian Kreutzer, Ruprecht-Karl University of Heidelberg (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Kars, R.H., Wallinga, J., Cohen, K.M., 2008. A new approach towards anomalous fading correction for feldspar
IRSL dating-tests on samples in field saturation. Radiation Measurements 43, 786-790. doi:10.1016/j.radmeas.2008.01.021
</p>
<p>Guralnik, B., Li, B., Jain, M., Chen, R., Paris, R.B., Murray, A.S., Li, S.-H., Pagonis, P.,
Herman, F., 2015. Radiation-induced growth and isothermal decay of infrared-stimulated luminescence
from feldspar. Radiation Measurements 81, 224-231.
</p>
<p>Huntley, D.J., 2006. An explanation of the power-law decay of luminescence.
Journal of Physics: Condensed Matter 18, 1359-1365. doi:10.1088/0953-8984/18/4/020
</p>
<p>King, G.E., Herman, F., Lambert, R., Valla, P.G., Guralnik, B., 2016.
Multi-OSL-thermochronometry of feldspar. Quaternary Geochronology 33, 76-87. doi:10.1016/j.quageo.2016.01.004
</p>
<p><strong>Further reading</strong>
</p>
<p>Morthekai, P., Jain, M., Cunha, P.P., Azevedo, J.M., Singhvi, A.K., 2011. An attempt to correct
for the fading in million year old basaltic rocks. Geochronometria 38(3), 223-230.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load example data (sample UNIL/NB123, see ?ExampleData.Fading)
data("ExampleData.Fading", envir = environment())

## (1) Set all relevant parameters
# a. fading measurement data (IR50)
fading_data &lt;- ExampleData.Fading$fading.data$IR50

# b. Dose response curve data
data &lt;- ExampleData.Fading$equivalentDose.data$IR50

## (2) Define required function parameters
ddot &lt;- c(7.00, 0.004)
readerDdot &lt;- c(0.134, 0.0067)

# Analyse fading measurement and get an estimate of rho'.
# Note that the RLum.Results object can be directly used for further processing.
# The number of MC runs is reduced for this example
rhop &lt;- analyse_FadingMeasurement(fading_data, plot = TRUE, verbose = FALSE, n.MC = 10)

## (3) Apply the Kars et al. (2008) model to the data
kars &lt;- calc_Huntley2006(
 data = data,
 rhop = rhop,
 ddot = ddot,
 readerDdot = readerDdot,
 n.MC = 25)


## Not run: 
# You can also provide LnTn values separately via the 'LnTn' argument.
# Note, however, that the data frame for 'data' must then NOT contain
# a LnTn value. See argument descriptions!
LnTn &lt;- data.frame(
 LnTn = c(1.84833, 2.24833),
 nTn.error = c(0.17, 0.22))

LxTx &lt;- data[2:nrow(data), ]

kars &lt;- calc_Huntley2006(
 data = LxTx,
 LnTn = LnTn,
 rhop = rhop,
 ddot = ddot,
 readerDdot = readerDdot,
 n.MC = 25)

## End(Not run)
</code></pre>

<hr>
<h2 id='calc_IEU'>Apply the internal-external-uncertainty (IEU) model after Thomsen et al.
(2007) to a given De distribution</h2><span id='topic+calc_IEU'></span>

<h3>Description</h3>

<p>Function to calculate the IEU De for a De data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_IEU(data, a, b, interval, decimal.point = 2, plot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_IEU_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[,1])</code> and
De error <code>(values[,2])</code></p>
</td></tr>
<tr><td><code id="calc_IEU_+3A_a">a</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
slope</p>
</td></tr>
<tr><td><code id="calc_IEU_+3A_b">b</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
intercept</p>
</td></tr>
<tr><td><code id="calc_IEU_+3A_interval">interval</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
fixed interval (e.g. 5 Gy) used for iteration of <code>Dbar</code>, from the mean to
Lowest.De used to create Graph.IEU <code style="white-space: pre;">&#8288;[Dbar.Fixed vs Z]&#8288;</code></p>
</td></tr>
<tr><td><code id="calc_IEU_+3A_decimal.point">decimal.point</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
number of decimal points for rounding calculations (e.g. 2)</p>
</td></tr>
<tr><td><code id="calc_IEU_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_IEU_+3A_...">...</code></td>
<td>
<p>further arguments (<code style="white-space: pre;">&#8288;trace, verbose&#8288;</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses the equations of Thomsen et al. (2007). The parameters a
and b are estimated from dose-recovery experiments.
</p>


<h3>Value</h3>

<p>Returns a plot (<em>optional</em>) and terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following elements:
</p>
<table role = "presentation">
<tr><td><code>.$summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>.$data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>.$args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>.$call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>.$tables</code></td>
<td>
<p><a href="base.html#topic+list">list</a> a list of data frames containing all calculation tables</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Smedley, R.K., Colombo, M., 2025. calc_IEU(): Apply the internal-external-uncertainty (IEU) model after Thomsen et al. (2007) to a given De distribution. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Rachel Smedley, Geography &amp; Earth Sciences, Aberystwyth University (United Kingdom) <br />
Based on an excel spreadsheet and accompanying macro written by Kristina Thomsen. <br />
Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Smedley, R.K., 2015. A new R function for the Internal External Uncertainty (IEU) model.
Ancient TL 33, 16-21.
</p>
<p>Thomsen, K.J., Murray, A.S., Boetter-Jensen, L. &amp; Kinahan, J.,
2007. Determination of burial dose in incompletely bleached fluvial samples
using single grains of quartz. Radiation Measurements 42, 370-379.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+calc_CommonDose">calc_CommonDose</a>, <a href="#topic+calc_CentralDose">calc_CentralDose</a>, <a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>,
<a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load data
data(ExampleData.DeValues, envir = environment())

## apply the IEU model
ieu &lt;- calc_IEU(ExampleData.DeValues$CA1, a = 0.2, b = 1.9, interval = 1)

</code></pre>

<hr>
<h2 id='calc_Lamothe2003'>Apply fading correction after Lamothe et al., 2003</h2><span id='topic+calc_Lamothe2003'></span>

<h3>Description</h3>

<p>This function applies the fading correction for the prediction of long-term fading as suggested
by Lamothe et al., 2003. The function basically adjusts the $L_n/T_n$ values and fits a new dose-response
curve using the function <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_Lamothe2003(
  object,
  dose_rate.envir,
  dose_rate.source,
  g_value,
  tc = NULL,
  tc.g_value = tc,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_Lamothe2003_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): Input data for applying the
fading correction. Allow are (1) <a href="base.html#topic+data.frame">data.frame</a> with three columns (<code>dose</code>, <code>LxTx</code>, <code style="white-space: pre;">&#8288;LxTx error&#8288;</code>; see details), (2)
<a href="#topic+RLum.Results-class">RLum.Results</a> object created by the function <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> or <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a></p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_dose_rate.envir">dose_rate.envir</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> vector of length 2 (<strong>required</strong>): Environmental dose rate in mGy/a</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_dose_rate.source">dose_rate.source</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> vector of length 2 (<strong>required</strong>): Irradiation source dose rate in Gy/s,
which is, according to Lamothe et al. (2003) De/t*.</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_g_value">g_value</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> vector of length 2 (<strong>required</strong>): g_value in
%/decade <em>recalculated at the moment</em> the equivalent dose was calculated,
i.e. <code>tc</code> is either similar for the <em>g</em>-value measurement <strong>and</strong> the
De measurement or needs be to recalculated (cf. <a href="#topic+calc_FadingCorr">calc_FadingCorr</a>).
Inserting a normalised g-value, e.g., normalised to 2-days , will
lead to wrong results.</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_tc">tc</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): time in seconds between the <strong>end</strong> of
the irradiation and the prompt measurement used in the equivalent dose
estimation (cf. Huntley &amp; Lamothe 2001).
If set to <code>NULL</code> it is assumed that <code>tc</code> is similar for the equivalent dose
estimation and the <em>g</em>-value estimation</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_tc.g_value">tc.g_value</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): the time in seconds between irradiation and the
prompt measurement estimating the <em>g</em>-value. If the <em>g</em>-value was normalised to, e.g., 2 days,
this time in seconds (i.e., <code>172800</code>) should be entered here along with the time used for the
equivalent dose estimation. If nothing is provided the time is set to <code>tc</code>, which is the
usual case for <em>g</em>-values obtained using the SAR method and <em>g</em>-values that had been not normalised to 2 days.
Note: If this value is not <code>NULL</code> the functions expects a <a href="base.html#topic+numeric">numeric</a> value for <code>tc</code>.</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_Lamothe2003_+3A_...">...</code></td>
<td>
<p>further arguments passed to function <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Format of <code>object</code> if <code>data.frame</code></strong>
</p>
<p>If <code>object</code> is of type <a href="base.html#topic+data.frame">data.frame</a>, all input values most be of type <a href="base.html#topic+numeric">numeric</a>.
Dose values are excepted in seconds (s) not Gray (Gy). No <code>NA</code> values are allowed and
the value for the natural dose (first row) should be <code>0</code>. Example for three dose points,
column names are arbitrary:
</p>
<div class="sourceCode"><pre> object &lt;- data.frame(
 dose = c(0,25,50),
 LxTx = c(4.2, 2.5, 5.0),
 LxTx_error = c(0.2, 0.1, 0.2))
</pre></div>
<p><strong>Note on the g-value and <code>tc</code></strong>
</p>
<p>Users new to R and fading measurements are often confused about what to
enter for <code>tc</code> and why it may differ from <code>tc.g_value</code>. The <code>tc</code> value
is, by convention (Huntley &amp; Lamothe 2001), the time elapsed between the end of the irradiation and the prompt
measurement. Usually there is no reason for having a <code>tc</code> value different for the equivalent dose measurement
and the <em>g</em>-value measurement, except if different equipment was used.
However, if, for instance, the <em>g</em>-value measurement sequence was analysed
with the <em>Analyst</em> (Duller 2015) and <code>Luminescence</code> is used to correct for fading,
there is a high chance that the value returned by the <em>Analyst</em> comes normalised to 2-days;
even the <code>tc</code> values of the measurement were identical.
In such cases, the fading correction cannot be correct until the <code>tc.g_value</code> was manually
set to 2-days (<code>172800</code> s) because the function will internally recalculate values
to an identical <code>tc</code> value.
</p>


<h3>Value</h3>

<p>The function returns are graphical output produced by the function <a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a> and
an <a href="#topic+RLum.Results-class">RLum.Results</a>.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> the fading corrected values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit&#8288;</code> </td><td style="text-align: left;"> <code>nls</code> </td><td style="text-align: left;"> the object returned by the dose response curve fitting </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>'<strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Mercier, N., 2025. calc_Lamothe2003(): Apply fading correction after Lamothe et al., 2003. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany), Norbert Mercier,
IRAMAT-CRP2A, Universit√© Bordeaux Montaigne (France)
, RLum Developer Team</p>


<h3>References</h3>

<p>Huntley, D.J., Lamothe, M., 2001. Ubiquity of anomalous fading in K-feldspars and the measurement
and correction for it in optical dating. Canadian Journal of Earth Sciences 38, 1093-1106.
</p>
<p>Duller, G.A.T., 2015. The Analyst software package for luminescence data: overview and recent improvements.
Ancient TL 33, 35‚Äì42.
</p>
<p>Lamothe, M., Auclair, M., Hamzaoui, C., Huot, S., 2003.
Towards a prediction of long-term anomalous fading of feldspar IRSL. Radiation Measurements 37,
493-498.
</p>


<h3>See Also</h3>

<p><a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>, <a href="#topic+calc_FadingCorr">calc_FadingCorr</a>, <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>, <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
##ExampleData.BINfileData contains two BINfileData objects
##CWOSL.SAR.Data and TL.SAR.Data
data(ExampleData.BINfileData, envir = environment())

##transform the values from the first position in a RLum.Analysis object
object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)

##perform SAR analysis and set rejection criteria
results &lt;- analyse_SAR.CWOSL(
object = object,
signal.integral.min = 1,
signal.integral.max = 2,
background.integral.min = 900,
background.integral.max = 1000,
verbose = FALSE,
plot = FALSE,
onlyLxTxTable = TRUE
)

##run fading correction
results_corr &lt;- calc_Lamothe2003(
  object = results,
  dose_rate.envir =  c(1.676 , 0.180),
  dose_rate.source = c(0.184, 0.003),
  g_value =  c(2.36, 0.6),
  plot = TRUE,
  fit.method = "EXP")


</code></pre>

<hr>
<h2 id='calc_MaxDose'>Apply the maximum age model to a given De distribution</h2><span id='topic+calc_MaxDose'></span>

<h3>Description</h3>

<p>Function to fit the maximum age model to De data. This is a wrapper function
that calls <a href="#topic+calc_MinDose">calc_MinDose</a> and applies a similar approach as described in
Olley et al. (2006).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_MaxDose(
  data,
  sigmab,
  log = TRUE,
  par = 3,
  bootstrap = FALSE,
  init.values,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_MaxDose_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[ ,1])</code> and De error <code>(data[ ,2])</code>.</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
additional spread in De values.
This value represents the expected overdispersion in the data should the sample be
well-bleached (Cunningham &amp; Walling 2012, p. 100).
<strong>NOTE</strong>: For the logged model (<code>log = TRUE</code>) this value must be
a fraction, e.g. 0.2 (= 20 %). If the un-logged model is used (<code>log = FALSE</code>),
sigmab must be provided in the same absolute units of the De values (seconds or Gray).
See details (<a href="#topic+calc_MinDose">calc_MinDose</a>.</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
fit the (un-)logged three parameter minimum dose model to De data</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_par">par</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
apply the 3- or 4-parameter minimum age model (<code>par=3</code> or <code>par=4</code>).</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_bootstrap">bootstrap</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
apply the recycled bootstrap approach of Cunningham &amp; Wallinga (2012).</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_init.values">init.values</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
starting values for gamma, sigma, p0 and mu. Custom values need to be provided in a vector of
length three in the form of <code>c(gamma, sigma, p0)</code>.</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_MaxDose_+3A_...">...</code></td>
<td>
<p>further arguments for bootstrapping (<code style="white-space: pre;">&#8288;bs.M, bs.N, bs.h, sigmab.sd&#8288;</code>).
See details for their usage.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Data transformation</strong>
</p>
<p>To estimate the maximum dose population
and its standard error, the three parameter minimum age model of Galbraith
et al. (1999) is adapted. The measured De values are transformed as follows:
</p>

<ol>
<li><p> convert De values to natural logs
</p>
</li>
<li><p> multiply the logged data to create a mirror image of the De distribution
</p>
</li>
<li><p> shift De values along x-axis by the smallest x-value found to obtain only positive values
</p>
</li>
<li><p> combine in quadrature the measurement error associated with each De value
with a relative error specified by <code>sigmab</code>
</p>
</li>
<li><p> apply the MAM to these data
</p>
</li></ol>

<p>When all calculations are done the results are then converted as follows
</p>

<ol>
<li><p> subtract the x-offset
</p>
</li>
<li><p> multiply the natural logs by -1
</p>
</li>
<li><p> take the exponent to obtain the maximum dose estimate in Gy
</p>
</li></ol>

<p><strong>Further documentation</strong>
</p>
<p>Please see <a href="#topic+calc_MinDose">calc_MinDose</a>.
</p>


<h3>Value</h3>

<p>Please see <a href="#topic+calc_MinDose">calc_MinDose</a>.
</p>


<h3>Function version</h3>

<p>0.3.1
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_MaxDose(): Apply the maximum age model to a given De distribution. Function version 0.3.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany) <br />
Based on a rewritten S script of Rex Galbraith, 2010
, RLum Developer Team</p>


<h3>References</h3>

<p>Arnold, L.J., Roberts, R.G., Galbraith, R.F. &amp; DeLong, S.B.,
2009. A revised burial dose estimation procedure for optical dating of young
and modern-age sediments. Quaternary Geochronology 4, 306-325.
</p>
<p>Galbraith, R.F. &amp; Laslett, G.M., 1993. Statistical models for mixed fission
track ages. Nuclear Tracks Radiation Measurements 4, 459-470.
</p>
<p>Galbraith, R.F., Roberts, R.G., Laslett, G.M., Yoshida, H. &amp; Olley, J.M.,
1999. Optical dating of single grains of quartz from Jinmium rock shelter,
northern Australia. Part I: experimental design and statistical models.
Archaeometry 41, 339-364.
</p>
<p>Galbraith, R.F., 2005. Statistics for
Fission Track Analysis, Chapman &amp; Hall/CRC, Boca Raton.
</p>
<p>Galbraith, R.F. &amp; Roberts, R.G., 2012. Statistical aspects of equivalent dose and error
calculation and display in OSL dating: An overview and some recommendations.
Quaternary Geochronology 11, 1-27.
</p>
<p>Olley, J.M., Roberts, R.G., Yoshida, H., Bowler, J.M., 2006. Single-grain optical dating of grave-infill
associated with human burials at Lake Mungo, Australia. Quaternary Science
Reviews 25, 2469-2474
</p>
<p><strong>Further reading</strong>
</p>
<p>Arnold, L.J. &amp; Roberts, R.G., 2009. Stochastic modelling of multi-grain equivalent dose
(De) distributions: Implications for OSL dating of sediment mixtures.
Quaternary Geochronology 4, 204-230.
</p>
<p>Bailey, R.M. &amp; Arnold, L.J., 2006. Statistical modelling of single grain quartz De distributions and an
assessment of procedures for estimating burial dose. Quaternary Science
Reviews 25, 2475-2502.
</p>
<p>Cunningham, A.C. &amp; Wallinga, J., 2012. Realizing the potential of fluvial archives using robust OSL chronologies.
Quaternary Geochronology 12, 98-106.
</p>
<p>Rodnight, H., Duller, G.A.T., Wintle, A.G. &amp; Tooth, S., 2006. Assessing the reproducibility and accuracy
of optical dating of fluvial deposits.  Quaternary Geochronology 1, 109-120.
</p>
<p>Rodnight, H., 2008. How many equivalent dose values are needed to
obtain a reproducible distribution?. Ancient TL 26, 3-10.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_CentralDose">calc_CentralDose</a>, <a href="#topic+calc_CommonDose">calc_CommonDose</a>, <a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>,
<a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_MinDose">calc_MinDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.DeValues, envir = environment())

# apply the maximum dose model
calc_MaxDose(ExampleData.DeValues$CA1, sigmab = 0.2, par = 3)

</code></pre>

<hr>
<h2 id='calc_MinDose'>Apply the (un-)logged minimum age model (MAM) after Galbraith et al. (1999)
to a given De distribution</h2><span id='topic+calc_MinDose'></span>

<h3>Description</h3>

<p>Function to fit the (un-)logged three or four parameter minimum dose model
(MAM-3/4) to De data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_MinDose(
  data,
  sigmab,
  log = TRUE,
  par = 3,
  bootstrap = FALSE,
  init.values,
  level = 0.95,
  log.output = FALSE,
  plot = TRUE,
  multicore = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_MinDose_+3A_data">data</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: two columns with De <code>(data[ ,1])</code> and De error <code>(data[ ,2])</code>.</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
additional spread in De values.
This value represents the expected overdispersion in the data should the sample be
well-bleached (Cunningham &amp; Walling 2012, p. 100).
<strong>NOTE</strong>: For the logged model (<code>log = TRUE</code>) this value must be
a fraction, e.g. 0.2 (= 20 %). If the un-logged model is used (<code>log = FALSE</code>),
<code>sigmab</code> must be provided in the same absolute units of the De values (seconds or Gray).
See details.</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
fit the (un-)logged minimum dose model to De data.</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_par">par</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
apply the 3- or 4-parameter minimum age model (<code>par=3</code> or <code>par=4</code>). The MAM-3 is
used by default.</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_bootstrap">bootstrap</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
apply the recycled bootstrap approach of Cunningham &amp; Wallinga (2012).</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_init.values">init.values</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
a named list with starting values for <code>gamma</code>, <code>sigma</code>, <code>p0</code> and <code>mu</code>
(e.g. <code>list(gamma=100, sigma=1.5, p0=0.1, mu=100)</code>). If no values are
provided, reasonable values will be estimated from the data.
<strong>NOTE</strong>: the initial values must always be given in the absolute units.
If a logged model is applied (<code>log = TRUE</code>), the provided <code>init.values</code>
are automatically log-transformed.</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_level">level</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
the confidence level required (defaults to 0.95).</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_log.output">log.output</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If <code>TRUE</code> the console output will also show the logged values of the final parameter estimates
and confidence intervals (only applicable if <code>log = TRUE</code>).</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_multicore">multicore</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable parallel computation of the bootstrap by creating a multicore SNOW cluster. Depending
on the number of available logical CPU cores this may drastically reduce
the computation time. Note that this option is highly experimental and may not
work on all machines. (<code>TRUE</code>/<code>FALSE</code>)</p>
</td></tr>
<tr><td><code id="calc_MinDose_+3A_...">...</code></td>
<td>
<p>(<em>optional</em>) further arguments for bootstrapping
(<code>bs.M</code>, <code>bs.N</code>, <code>bs.h</code>, <code>sigmab.sd</code>). See details for their usage.
Further arguments are
</p>

<ul>
<li> <p><code>verbose</code> to de-/activate console output (logical),
</p>
</li>
<li> <p><code>debug</code> for extended console output (logical) and
</p>
</li>
<li> <p><code>cores</code> (integer) to manually specify the number of cores to be used when <code>multicore=TRUE</code>.
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Parameters</strong>
</p>
<p>This model has four parameters:
</p>

<table>
<tr>
 <td style="text-align: right;">
<code>gamma</code>: </td><td style="text-align: left;"> minimum dose on the log scale </td>
</tr>
<tr>
 <td style="text-align: right;">
<code>mu</code>: </td><td style="text-align: left;"> mean of the non-truncated normal distribution </td>
</tr>
<tr>
 <td style="text-align: right;">
<code>sigma</code>: </td><td style="text-align: left;"> spread in ages above the minimum </td>
</tr>
<tr>
 <td style="text-align: right;">
<code>p0</code>: </td><td style="text-align: left;"> proportion of grains at gamma </td>
</tr>
<tr>
 <td style="text-align: right;"> </td>
</tr>

</table>

<p>If <code>par=3</code> (default) the 3-parameter minimum age model is applied,
where <code>gamma=mu</code>. For <code>par=4</code> the 4-parameter model is applied instead.
</p>
<p><strong>(Un-)logged model</strong>
</p>
<p>In the original version of the minimum dose model, the basic data are the natural
logarithms of the De estimates and relative standard errors of the De
estimates. The value for <code>sigmab</code> must be provided as a ratio
(e.g, 0.2 for 20 %). This model will be applied if <code>log = TRUE</code>.
</p>
<p>If <code>log=FALSE</code>, the modified un-logged model will be applied instead. This
has essentially the same form as the original version.  <code>gamma</code> and
<code>sigma</code> are in Gy and <code>gamma</code> becomes the minimum true dose in the
population.
<strong>Note</strong> that the un-logged model requires <code>sigmab</code> to be in the same
absolute unit as the provided De values (seconds or Gray).
</p>
<p>While the original (logged) version of the minimum dose
model may be appropriate for most samples (i.e. De distributions), the
modified (un-logged) version is specially designed for modern-age and young
samples containing negative, zero or near-zero De estimates (Arnold et al.
2009, p. 323).
</p>
<p><strong>Initial values &amp; boundaries</strong>
</p>
<p>The log likelihood calculations use the <a href="stats.html#topic+nlminb">nlminb</a> function for box-constrained
optimisation using PORT routines.  Accordingly, initial values for the four
parameters can be specified via <code>init.values</code>. If no values are
provided for <code>init.values</code>, reasonable starting values are estimated
from the input data.  If the final estimates of <em>gamma</em>, <em>mu</em>,
<em>sigma</em> and <em>p0</em> are totally off target, consider providing custom
starting values via <code>init.values</code>.
In contrast to previous versions of this function, the boundaries for the
individual model parameters are no longer required to be explicitly specified.
If you want to override the default boundary values use arguments
<code>gamma.lower</code>, <code>gamma.upper</code>, <code>sigma.lower</code>, <code>sigma.upper</code>, <code>p0.lower</code>, <code>p0.upper</code>,
<code>mu.lower</code> and <code>mu.upper</code>.
</p>
<p><strong>Bootstrap</strong>
</p>
<p>When <code>bootstrap=TRUE</code> the function applies the bootstrapping method as
described in Wallinga &amp; Cunningham (2012). By default, the minimum age model
produces 1000 first level and 3000 second level bootstrap replicates
(actually, the number of second level bootstrap replicates is three times
the number of first level replicates unless specified otherwise).  The
uncertainty on <code>sigmab</code> is 0.04 by default. These values can be changed by
using the arguments <code>bs.M</code> (first level replicates), <code>bs.N</code>
(second level replicates) and <code>sigmab.sd</code> (error on <code>sigmab</code>). With
<code>bs.h</code> the bandwidth of the kernel density estimate can be specified.
By default, <code>h</code> is calculated as
</p>
<p style="text-align: center;"><code class="reqn">h = (2*\sigma_{DE})/\sqrt{n}</code>
</p>

<p><strong>Multicore support</strong>
</p>
<p>This function supports parallel computing and can be activated by <code>multicore=TRUE</code>.
By default, the number of available logical CPU cores is determined
automatically, but can be changed with <code>cores</code>. The multicore support
is only available when <code>bootstrap=TRUE</code> and spawns <code>n</code> R instances
for each core to get MAM estimates for each of the N and M bootstrap
replicates. Note that this option is highly experimental and may or may not
work for your machine. The performance gain increases for larger number
of bootstrap replicates. However, note that with each additional core (and
hence R instance) the memory usage can significantly increase and depending
on the number of bootstrap replicates. When insufficient memory is available,
there will be a massive performance hit.
</p>
<p><strong>Likelihood profiles</strong>
</p>
<p>The likelihood profiles are generated and plotted by the <code>bbmle</code> package.
The profile likelihood plots look different to ordinary profile likelihood as
</p>
<p>&quot;<code style="white-space: pre;">&#8288;[...]&#8288;</code> the plot method for likelihood profiles displays the square root of
the deviance difference (twice the difference in negative log-likelihood from
the best fit), so it will be V-shaped for cases where the quadratic approximation
works well <code style="white-space: pre;">&#8288;[...]&#8288;</code>.&quot; (Bolker 2016).
</p>
<p>For more details on the profile likelihood
calculations and plots please see the vignettes of the <code>bbmle</code> package
(also available here: <a href="https://CRAN.R-project.org/package=bbmle">https://CRAN.R-project.org/package=bbmle</a>).
</p>


<h3>Value</h3>

<p>Returns a plot (<em>optional</em>) and terminal output. In addition an
<a href="#topic+RLum.Results-class">RLum.Results</a> object is returned containing the
following elements:
</p>
<table role = "presentation">
<tr><td><code>$summary</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> summary of all relevant model results.</p>
</td></tr>
<tr><td><code>$data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> original input data</p>
</td></tr>
<tr><td><code>$args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
<tr><td><code>$call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>$mle</code></td>
<td>
<p><a href="bbmle.html#topic+mle2">bbmle::mle2</a> object containing the maximum log likelihood functions for all parameters</p>
</td></tr>
<tr><td><code>$BIC</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> BIC score</p>
</td></tr>
<tr><td><code>$confint</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> confidence intervals for all parameters</p>
</td></tr>
<tr><td><code>$profile</code></td>
<td>
<p><a href="stats.html#topic+profile">stats::profile</a> the log likelihood profiles</p>
</td></tr>
<tr><td><code>$bootstrap</code></td>
<td>
<p><a href="base.html#topic+list">list</a> bootstrap results</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.4.4
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. calc_MinDose(): Apply the (un-)logged minimum age model (MAM) after Galbraith et al. (1999) to a given De distribution. Function version 0.4.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The default starting values for <em>gamma</em>, <em>mu</em>, <em>sigma</em>
and <em>p0</em> may only be appropriate for some De data sets and may need to
be changed for other data. This is especially true when the un-logged
version is applied. <br />
Also note that all R warning messages are suppressed
when running this function. If the results seem odd consider re-running the
model with <code>debug=TRUE</code> which provides extended console output and
forwards all internal warning messages.
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany) <br />
Based on a rewritten S script of Rex Galbraith, 2010 <br />
The bootstrap approach is based on a rewritten MATLAB script of Alastair Cunningham. <br />
Alastair Cunningham is thanked for his help in implementing and cross-checking the code.
, RLum Developer Team</p>


<h3>References</h3>

<p>Arnold, L.J., Roberts, R.G., Galbraith, R.F. &amp; DeLong, S.B.,
2009. A revised burial dose estimation procedure for optical dating of young
and modern-age sediments. Quaternary Geochronology 4, 306-325.
</p>
<p>Galbraith, R.F. &amp; Laslett, G.M., 1993. Statistical models for mixed fission
track ages. Nuclear Tracks Radiation Measurements 4, 459-470.
</p>
<p>Galbraith, R.F., Roberts, R.G., Laslett, G.M., Yoshida, H. &amp; Olley, J.M.,
1999. Optical dating of single grains of quartz from Jinmium rock shelter,
northern Australia. Part I: experimental design and statistical models.
Archaeometry 41, 339-364.
</p>
<p>Galbraith, R.F., 2005. Statistics for
Fission Track Analysis, Chapman &amp; Hall/CRC, Boca Raton.
</p>
<p>Galbraith, R.F. &amp; Roberts, R.G., 2012. Statistical aspects of equivalent dose and error
calculation and display in OSL dating: An overview and some recommendations.
Quaternary Geochronology 11, 1-27.
</p>
<p>Olley, J.M., Roberts, R.G., Yoshida, H., Bowler, J.M., 2006. Single-grain optical dating of grave-infill
associated with human burials at Lake Mungo, Australia. Quaternary Science
Reviews 25, 2469-2474.
</p>
<p><strong>Further reading</strong>
</p>
<p>Arnold, L.J. &amp; Roberts, R.G., 2009. Stochastic modelling of multi-grain equivalent dose
(De) distributions: Implications for OSL dating of sediment mixtures.
Quaternary Geochronology 4, 204-230.
</p>
<p>Bolker, B., 2016. Maximum likelihood estimation analysis with the bbmle package.
In: Bolker, B., R Development Core Team, 2016. bbmle: Tools for General Maximum Likelihood Estimation.
R package version 1.0.18. <a href="https://CRAN.R-project.org/package=bbmle">https://CRAN.R-project.org/package=bbmle</a>
</p>
<p>Bailey, R.M. &amp; Arnold, L.J., 2006. Statistical modelling of single grain quartz De distributions and an
assessment of procedures for estimating burial dose. Quaternary Science
Reviews 25, 2475-2502.
</p>
<p>Cunningham, A.C. &amp; Wallinga, J., 2012. Realizing the potential of fluvial archives using robust OSL chronologies.
Quaternary Geochronology 12, 98-106.
</p>
<p>Rodnight, H., Duller, G.A.T., Wintle, A.G. &amp; Tooth, S., 2006. Assessing the reproducibility and accuracy
of optical dating of fluvial deposits.  Quaternary Geochronology 1, 109-120.
</p>
<p>Rodnight, H., 2008. How many equivalent dose values are needed to
obtain a reproducible distribution?. Ancient TL 26, 3-10.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_CentralDose">calc_CentralDose</a>, <a href="#topic+calc_CommonDose">calc_CommonDose</a>, <a href="#topic+calc_FiniteMixture">calc_FiniteMixture</a>,
<a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_MaxDose">calc_MaxDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load example data
data(ExampleData.DeValues, envir = environment())

# (1) Apply the minimum age model with minimum required parameters.
# By default, this will apply the un-logged 3-parameter MAM.
calc_MinDose(data = ExampleData.DeValues$CA1, sigmab = 0.1)

## Not run: 
# (2) Re-run the model, but save results to a variable and turn
# plotting of the log-likelihood profiles off.
mam &lt;- calc_MinDose(
 data = ExampleData.DeValues$CA1,
 sigmab = 0.1,
 plot = FALSE)

# Show structure of the RLum.Results object
mam

# Show summary table that contains the most relevant results
res &lt;- get_RLum(mam, "summary")
res

# Plot the log likelihood profiles retroactively, because before
# we set plot = FALSE
plot_RLum(mam)

# Plot the dose distribution in an abanico plot and draw a line
# at the minimum dose estimate
plot_AbanicoPlot(data = ExampleData.DeValues$CA1,
                 main = "3-parameter Minimum Age Model",
                 line = mam,polygon.col = "none",
                 hist = TRUE,
                 rug = TRUE,
                 summary = c("n", "mean", "mean.weighted", "median", "in.ci"),
                 centrality = res$de,
                 line.col = "red",
                 grid.col = "none",
                 line.label = paste0(round(res$de, 1), "\U00B1",
                                     round(res$de_err, 1), " Gy"),
                 bw = 0.1,
                 ylim = c(-25, 18),
                 summary.pos = "topleft",
                 mtext = bquote("Parameters: " ~
                                  sigma[b] == .(get_RLum(mam, "args")$sigmab) ~ ", " ~
                                  gamma == .(round(log(res$de), 1)) ~ ", " ~
                                  sigma == .(round(res$sig, 1)) ~ ", " ~
                                  rho == .(round(res$p0, 2))))



# (3) Run the minimum age model with bootstrap
# NOTE: Bootstrapping is computationally intensive
# (3.1) run the minimum age model with default values for bootstrapping
calc_MinDose(data = ExampleData.DeValues$CA1,
             sigmab = 0.15,
             bootstrap = TRUE)

# (3.2) Bootstrap control parameters
mam &lt;- calc_MinDose(data = ExampleData.DeValues$CA1,
                    sigmab = 0.15,
                    bootstrap = TRUE,
                    bs.M = 300,
                    bs.N = 500,
                    bs.h = 4,
                    sigmab.sd = 0.06,
                    plot = FALSE)

# Plot the results
plot_RLum(mam)

# save bootstrap results in a separate variable
bs &lt;- get_RLum(mam, "bootstrap")

# show structure of the bootstrap results
str(bs, max.level = 2, give.attr = FALSE)

# print summary of minimum dose and likelihood pairs
summary(bs$pairs$gamma)

# Show polynomial fits of the bootstrap pairs
bs$poly.fits$poly.three

# Plot various statistics of the fit using the generic plot() function
par(mfcol=c(2,2))
plot(bs$poly.fits$poly.three, ask = FALSE)

# Show the fitted values of the polynomials
summary(bs$poly.fits$poly.three$fitted.values)

## End(Not run)

</code></pre>

<hr>
<h2 id='calc_MoransI'>Calculate Moran's I</h2><span id='topic+calc_MoransI'></span>

<h3>Description</h3>

<p>Calculate Moran's I
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_MoransI(
  object,
  df_neighbours = NULL,
  spatial_autocorrelation = TRUE,
  compute_pseudo_p = FALSE,
  tested_moransI = NULL,
  n_permutations = 999,
  ignore_borders = FALSE,
  return_intermediate_values = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_MoransI_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>) containing
the values of the grains of one. Should have length 100; can contain <code>NA</code>
values.</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_df_neighbours">df_neighbours</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>with default</em>): a data frame with 3
columns (<code>location</code>, <code>neighbour</code> and <code>weight</code>, respectively), with each row
indicating the index of one location and one of its neighbours (note that
the concept of &quot;location&quot; versus &quot;neighbour&quot; is symmetric, so each
neighbouring pair needs to be specified only once), alongside their relative
weight (generally set to 1). If <code>NULL</code> (default), this is constructed
automatically by the internal function <code>.get_Neighbours</code>.</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_spatial_autocorrelation">spatial_autocorrelation</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether spatial
autocorrelation should be considered in the computation of Moran's I
(<code>TRUE</code> by default). If <code>FALSE</code>, the function computes Moran's I expected
value in case of no autocorrelation (H_0). See details for further
information.</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_compute_pseudo_p">compute_pseudo_p</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether a pseudo p-value
should be computed (<code>FALSE</code> by default).</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_tested_moransi">tested_moransI</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): The value of Moran's I to
be tested against when computing the pseudo p-value. If <code>NULL</code> (default), the
value calculated by the function will be used. Ignored if <code>compute_pseudo_p</code>
is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_n_permutations">n_permutations</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): number of random
permutations tested to calculate the fraction which is the pseudo p
(defaults to 999). Influences the calculation speed, which will have
impact in case of large scale simulation loops. Ignored if <code>compute_pseudo_p</code>
is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_ignore_borders">ignore_borders</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether only grain
locations that do not lie on the border of the disc should be considered
(<code>FALSE</code> by default). Thus if <code>TRUE</code>, only the inner 8x8 grain locations
rather than the full 10x10 are considered. Ignored if <code>df_neighbours</code> is
not <code>NULL</code> or if <code>spatial_autocorrelation = FALSE</code>.</p>
</td></tr>
<tr><td><code id="calc_MoransI_+3A_return_intermediate_values">return_intermediate_values</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether the
function should return a list with several intermediate calculation results
(defaults to <code>FALSE</code>). Ignored if <code>spatial_autocorrelation</code> is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Case of no spatial autocorrelation</strong>
</p>
<p>Perhaps a bit counter-intuitive, the expected value of Moran's I under the
null hypothesis of no spatial correlation is a value slightly smaller than
zero. When setting <code>spatial_autocorrelation = FALSE</code>, this function
calculates the expected value based on the number of observations or the
length of the observation vector (while taking out <code>NA</code> values). Note that
the expected value only depends on the number of observed separate grain
values. This can be useful for plotting.
</p>
<p>The expected Moran's I for the null hypothesis of no spatial correlation
corresponds to <code>-1 / (n - 1)</code>, with <code>n</code> being the number of non-missing
observations.
</p>


<h3>Value</h3>

<p>By default one numerical value, roughly between -1 and 1, where
close to zero means no spatial correlation, and value close to 1 a positive
spatial correlation given the pattern we interested in (by default all rook
neighbours). A value closer to -1 has no meaning within the context of
luminescence crosstalk. If <code>compute_pseudo_p = TRUE</code>, then the computed
pseudo p-value is returned. If <code>return_intermediate_values</code> is set to <code>TRUE</code>,
a list with several values used for calculation is returned instead of a
single outcome.
</p>


<h3>How to cite</h3>

<p>Boer, A.d., Steinbuch, L., 2025. calc_MoransI(): Calculate Moran's I. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Anna-Maartje de Boer, Luc Steinbuch, Wageningen University &amp; Research, 2025
, RLum Developer Team</p>


<h3>References</h3>

<p>de Boer, A-M., Steinbuch, L., Heuvelink, G.B.M., Wallinga, J., 2025.
A novel tool to assess crosstalk in single-grain luminescence detection.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Test a fictional sample with spatial correlation
calc_MoransI(object = c(1:100))

## Test some fictional samples without spatial correlation;
## note the randomness with each repetition
calc_MoransI(object = rnorm(n = 100))
calc_MoransI(object = rnorm(n = 100))
calc_MoransI(object = rnorm(n = 100))

</code></pre>

<hr>
<h2 id='calc_OSLLxTxDecomposed'>Calculate Lx/Tx ratio for decomposed CW-OSL signal components</h2><span id='topic+calc_OSLLxTxDecomposed'></span>

<h3>Description</h3>

<p>Calculate <code>Lx/Tx</code> ratios from a given set of decomposed
CW-OSL curves decomposed by <code>OSLdecomposition::RLum.OSL_decomposition</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_OSLLxTxDecomposed(
  Lx.data,
  Tx.data = NULL,
  OSL.component = 1L,
  sig0 = 0,
  digits = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_OSLLxTxDecomposed_+3A_lx.data">Lx.data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): Component table created by
<code>OSLdecomposition::RLum.OSL_decomposition</code> and per default located
at <code>object@records[[...]]@info$COMPONENTS</code>.The value of <code style="white-space: pre;">&#8288;$n[OSL.component]&#8288;</code>
is set as <code>LnLx</code>. The value of <code style="white-space: pre;">&#8288;$n.error[OSL.component]&#8288;</code> is set as <code>LnLx.error</code></p>
</td></tr>
<tr><td><code id="calc_OSLLxTxDecomposed_+3A_tx.data">Tx.data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>): Component table created by
<code>OSLdecomposition::RLum.OSL_decomposition</code> and per default located at
<code>object@records[[...]]@info$COMPONENTS</code>. The value of <code style="white-space: pre;">&#8288;$n[OSL.component]&#8288;</code>
is set as <code>TnTx</code>. The value of <code style="white-space: pre;">&#8288;$n.error[OSL.component]&#8288;</code> is set as <code>TnTx.error</code></p>
</td></tr>
<tr><td><code id="calc_OSLLxTxDecomposed_+3A_osl.component">OSL.component</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> or <a href="base.html#topic+character">character</a> (<em>optional</em>):
a single index or a name describing which OSL signal component shall be evaluated.
This argument can either be the name of the OSL component assigned by
<code>OSLdecomposition::RLum.OSL_global_fitting</code> or the index of component.
Then <code>'1'</code> selects the fastest decaying component, <code>'2'</code> the
second fastest and so on. If not defined, the fastest decaying component is selected.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxDecomposed_+3A_sig0">sig0</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): allows adding an extra error component
to the final <code>Lx/Tx</code> error value (e.g., instrumental error).</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxDecomposed_+3A_digits">digits</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): round numbers to the specified digits.
If digits is set to <code>NULL</code> nothing is rounded.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
</p>
<p>Slot <code>data</code> contains a <a href="base.html#topic+list">list</a> with the following structure:
</p>
<p><strong>@data</strong>
</p>
<div class="sourceCode"><pre>$LxTx.table (data.frame)
.. $ LnLx
.. $ TnTx
.. $ Net_LnLx
.. $ Net_LnLx.Error
.. $ Net_TnTx
.. $ Net_TnTx.Error
.. $ LxTx
.. $ LxTx.relError
.. $ LxTx.Error
</pre></div>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Mittelstrass, D., 2025. calc_OSLLxTxDecomposed(): Calculate Lx/Tx ratio for decomposed CW-OSL signal components. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Dirk Mittelstrass
, RLum Developer Team</p>


<h3>References</h3>

<p>Mittelstrass D., Schmidt C., Beyer J., Straessner A., 2019.
Automated identification and separation of quartz CW-OSL signal components with R.
talk presented at DLED 2019, Bingen, Germany
<a href="http://luminescence.de/OSLdecomp_talk.pdf">http://luminescence.de/OSLdecomp_talk.pdf</a><br />
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>, <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>
</p>

<hr>
<h2 id='calc_OSLLxTxRatio'>Calculate <code>Lx/Tx</code> ratio for CW-OSL curves</h2><span id='topic+calc_OSLLxTxRatio'></span>

<h3>Description</h3>

<p>Calculate <code>Lx/Tx</code> ratios from a given set of CW-OSL curves assuming late light
background subtraction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_OSLLxTxRatio(
  Lx.data,
  Tx.data = NULL,
  signal.integral,
  signal.integral.Tx = NULL,
  background.integral,
  background.integral.Tx = NULL,
  background.count.distribution = "non-poisson",
  use_previousBG = FALSE,
  sigmab = NULL,
  sig0 = 0,
  digits = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_OSLLxTxRatio_+3A_lx.data">Lx.data</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
requires a CW-OSL shine down curve (x = time, y = counts)</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_tx.data">Tx.data</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
requires a CW-OSL shine down curve (x = time, y = counts). If no
input is given the <code>Tx.data</code> will be treated as <code>NA</code> and no <code>Lx/Tx</code> ratio
is calculated.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_signal.integral">signal.integral</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): vector with the limits for the signal integral.
Can be set to <code>NA</code> than now integrals are considered and all other integrals are set to <code>NA</code> as well.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_signal.integral.tx">signal.integral.Tx</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
vector with the limits for the signal integral for the <code>Tx</code>-curve. If
missing, the value from <code>signal.integral</code> is used.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_background.integral">background.integral</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
vector with the bounds for the background integral.
Can be set to <code>NA</code> than now integrals are considered and all other integrals are set to <code>NA</code> as well.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_background.integral.tx">background.integral.Tx</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
vector with the limits for the background integral for the <code>Tx</code> curve.
If missing, the value from <code>background.integral</code> is used.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_background.count.distribution">background.count.distribution</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
sets the count distribution assumed for the error calculation.
Possible arguments are <code>"poisson"</code> or <code>"non-poisson"</code> (default). See
details for further information.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_use_previousbg">use_previousBG</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If set to <code>TRUE</code> the background of the <code>Lx</code>-signal is subtracted also
from the <code>Tx</code>-signal. Please note that in this case separate
signal integral limits for the <code>Tx</code>-signal are not allowed and will be reset.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_sigmab">sigmab</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
option to set a manual value for the overdispersion (for <code>LnTx</code> and <code>TnTx</code>),
used for the <code>Lx/Tx</code> error calculation. The value should be provided as
absolute squared count values, e.g. <code>sigmab = c(300,300)</code>.
<strong>Note:</strong> If only one value is provided this value is taken for both (<code>LnTx</code> and <code>TnTx</code>) signals.</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_sig0">sig0</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
allow adding an extra component of error to the final <code>Lx/Tx</code> error value
(e.g., instrumental error, see details).</p>
</td></tr>
<tr><td><code id="calc_OSLLxTxRatio_+3A_digits">digits</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
round numbers to the specified digits. If set to <code>NULL</code> no rounding occurs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The integrity of the chosen values for the signal and background integral is
checked by the function; the signal integral limits have to be lower than
the background integral limits. If a <a href="base.html#topic+vector">vector</a> is given as input instead
of a <a href="base.html#topic+data.frame">data.frame</a>, an artificial <a href="base.html#topic+data.frame">data.frame</a> is produced. The
error calculation is done according to Galbraith (2002).
</p>
<p><strong>Please note:</strong> In cases where the calculation results in <code>NaN</code> values (for
example due to zero-signal, and therefore a division of 0 by 0), these <code>NaN</code>
values are replaced by 0.
</p>
<p><strong><code>sigmab</code></strong>
</p>
<p>The default value of <code>sigmab</code> is calculated assuming the background is
constant and <strong>would not</strong> applicable when the background varies as,
e.g., as observed for the early light subtraction method.
</p>
<p><strong>sig0</strong>
</p>
<p>This argument allows to add an extra component of error to the final <code>Lx/Tx</code>
error value. The input will be treated as factor that is multiplied with
the already calculated <code>LxTx</code> and the result is add up by:
</p>
<p style="text-align: center;"><code class="reqn">se(LxTx) = \sqrt(se(LxTx)^2 + (LxTx * sig0)^2)</code>
</p>

<p><strong>background.count.distribution</strong>
</p>
<p>This argument allows selecting the distribution assumption that is used for
the error calculation. According to Galbraith (2002, 2014) the background
counts may be overdispersed (i.e. do not follow a Poisson distribution,
which is assumed for the photomultiplier counts). In that case (might be the
normal case) it has to be accounted for the overdispersion by estimating
<code class="reqn">\sigma^2</code> (i.e. the overdispersion value). Therefore the relative
standard error is calculated as:
</p>

<ul>
<li> <p><code>poisson</code>
</p>
<p style="text-align: center;"><code class="reqn">rse(\mu_{S}) \approx \sqrt(Y_{0} + Y_{1}/k^2)/Y_{0} - Y_{1}/k</code>
</p>

</li>
<li> <p><code>non-poisson</code>
</p>
<p style="text-align: center;"><code class="reqn">rse(\mu_{S}) \approx \sqrt(Y_{0} + Y_{1}/k^2 + \sigma^2(1+1/k))/Y_{0} - Y_{1}/k</code>
</p>

</li></ul>

<p><strong>Please note</strong> that when using the early background subtraction method in
combination with the 'non-poisson' distribution argument, the corresponding <code>Lx/Tx</code> error
may considerably increase due to a high <code>sigmab</code> value.
Please check whether this is valid for your data set and  if necessary
consider to provide an own <code>sigmab</code> value using the corresponding argument <code>sigmab</code>.
</p>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
</p>
<p>Slot <code>data</code> contains a <a href="base.html#topic+list">list</a> with the following structure:
</p>
<p><strong>@data</strong>
</p>
<div class="sourceCode"><pre>$LxTx.table (data.frame)
.. $ LnLx
.. $ LnLx.BG
.. $ TnTx
.. $ TnTx.BG
.. $ Net_LnLx
.. $ Net_LnLx.Error
.. $ Net_TnTx
.. $ Net_TnTx.Error
.. $ LxTx
.. $ LxTx.Error
$ calc.parameters (list)
.. $ sigmab.LnTx
.. $ sigmab.TnTx
.. $ k
</pre></div>
<p><strong>@info</strong>
</p>
<div class="sourceCode"><pre>$ call (original function call)
</pre></div>


<h3>Function version</h3>

<p>0.8.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. calc_OSLLxTxRatio(): Calculate Lx/Tx ratio for CW-OSL curves. Function version 0.8.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The results of this function have been cross-checked with the Analyst
(version 3.24b). Access to the results object via <a href="#topic+get_RLum">get_RLum</a>.
</p>
<p><strong>Caution:</strong> If you are using early light subtraction (EBG), please either provide your
own <code>sigmab</code> value or use <code>background.count.distribution = "poisson"</code>.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Duller, G., 2018. Analyst v4.57 - User Manual.
<code style="white-space: pre;">&#8288;https://users.aber.ac.uk/ggd&#8288;</code><br />
</p>
<p>Galbraith, R.F., 2002. A note on the variance of a background-corrected OSL
count. Ancient TL, 20 (2), 49-51.
</p>
<p>Galbraith, R.F., 2014. A further note on the variance of a
background-corrected OSL count. Ancient TL, 31 (2), 1-3.
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+Analyse_SAR.OSLdata">Analyse_SAR.OSLdata</a>, <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>,
<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.LxTxOSLData, envir = environment())

##calculate Lx/Tx ratio
results &lt;- calc_OSLLxTxRatio(
 Lx.data = Lx.data,
 Tx.data = Tx.data,
 signal.integral = c(1:2),
 background.integral = c(85:100))

##get results object
get_RLum(results)

</code></pre>

<hr>
<h2 id='calc_SourceDoseRate'>Calculation of the source dose rate via the date of measurement</h2><span id='topic+calc_SourceDoseRate'></span>

<h3>Description</h3>

<p>Calculating the dose rate of the irradiation source via the date of
measurement based on: source calibration date, source dose rate, dose rate
error. The function returns a data.frame that provides the input argument
dose_rate for the function <a href="#topic+convert_Second2Gray">convert_Second2Gray</a>.
</p>
<p>Calculation of the source dose rate based on the time elapsed since the last
calibration of the irradiation source. Decay parameters assume a Sr-90 beta
source. </p>
<p style="text-align: center;"><code class="reqn">dose.rate = D0 * exp(-log(2) / T.1/2 * t)</code>
</p>
 <p><br /> with: D0 &lt;-
calibration dose rate T.1/2 &lt;- half-life of the source nuclide (here in
days) t &lt;- time since source calibration (in days) log(2) / T.1/2 equals the
decay constant lambda
</p>
<p>Information on the date of measurements may be taken from the data's
original .BIN file (using e.g., <code>BINfile &lt;- readBIN2R()</code> and the slot
<code>BINfile@METADATA$DATE</code>)
</p>
<p><strong>Allowed source types and related values</strong>
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>#</strong> </td><td style="text-align: left;"> <strong>Source type</strong> </td><td style="text-align: left;"> <strong>T.1/2</strong> </td><td style="text-align: left;"> <strong>Reference</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[1]&#8288;</code> </td><td style="text-align: left;"> Sr-90 </td><td style="text-align: left;"> 28.90 y </td><td style="text-align: left;"> NNDC, Brookhaven National Laboratory </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[2]&#8288;</code></td><td style="text-align: left;"> Am-214 </td><td style="text-align: left;"> 432.6 y </td><td style="text-align: left;"> NNDC, Brookhaven National Laboratory </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[3]&#8288;</code> </td><td style="text-align: left;"> Co-60 </td><td style="text-align: left;"> 5.274 y </td><td style="text-align: left;"> NNDC, Brookhaven National Laboratory </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[4&#8288;</code> </td><td style="text-align: left;"> Cs-137 </td><td style="text-align: left;"> 30.08 y </td><td style="text-align: left;"> NNDC, Brookhaven National Laboratory</td>
</tr>

</table>



<h3>Usage</h3>

<pre><code class='language-R'>calc_SourceDoseRate(
  measurement.date = Sys.Date(),
  calib.date,
  calib.dose.rate,
  calib.error,
  source.type = "Sr-90",
  dose.rate.unit = "Gy/s",
  predict = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_SourceDoseRate_+3A_measurement.date">measurement.date</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+Date">Date</a> (<em>with default</em>): Date of measurement in <code>"YYYY-MM-DD"</code>.
If no value is provided, the date will be set to today. The argument can be provided as vector.</p>
</td></tr>
<tr><td><code id="calc_SourceDoseRate_+3A_calib.date">calib.date</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+Date">Date</a> (<strong>required</strong>):
date of source calibration in <code>"YYYY-MM-DD"</code></p>
</td></tr>
<tr><td><code id="calc_SourceDoseRate_+3A_calib.dose.rate">calib.dose.rate</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
dose rate at date of calibration in Gy/s or Gy/min</p>
</td></tr>
<tr><td><code id="calc_SourceDoseRate_+3A_calib.error">calib.error</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
error of dose rate at date of calibration Gy/s or Gy/min</p>
</td></tr>
<tr><td><code id="calc_SourceDoseRate_+3A_source.type">source.type</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
specify irradiation source (<code>Sr-90</code>, <code>Co-60</code>, <code>Cs-137</code>, <code>Am-214</code>),
see details for further information</p>
</td></tr>
<tr><td><code id="calc_SourceDoseRate_+3A_dose.rate.unit">dose.rate.unit</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
specify dose rate unit for input (<code>Gy/min</code> or <code>Gy/s</code>), the output is given in
Gy/s as valid for the function <a href="#topic+convert_Second2Gray">convert_Second2Gray</a></p>
</td></tr>
<tr><td><code id="calc_SourceDoseRate_+3A_predict">predict</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
option allowing to predict the dose rate of the source over time in days
set by the provided value. Starting date is the value set with
<code>measurement.date</code>, e.g., <code>calc_SourceDoseRate(..., predict = 100)</code> calculates
the source dose rate for the next 100 days.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
Slot <code>data</code> contains a <a href="base.html#topic+list">list</a> with the following structure:
</p>
<div class="sourceCode"><pre>$ dose.rate (data.frame)
.. $ dose.rate
.. $ dose.rate.error
.. $ date (corresponding measurement date)
$ parameters (list)
.. $ source.type
.. $ halflife
.. $ dose.rate.unit
$ call (the original function call)
</pre></div>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
A plot method of the output is provided via <a href="#topic+plot_RLum">plot_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.3.2
</p>


<h3>How to cite</h3>

<p>Fuchs, M.C., Kreutzer, S., 2025. calc_SourceDoseRate(): Calculation of the source dose rate via the date of measurement. Function version 0.3.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Please be careful when using the option <code>predict</code>, especially when a
multiple set for <code>measurement.date</code> and <code>calib.date</code> is provided. For the
source dose rate prediction, the function takes the last <code>measurement.date</code>
value and predicts from that the source dose rate for the number of days
requested, that is: the (multiple) original input will be replaced.
However, the function does not change entries for the calibration dates,
but mixes them up. Therefore, it is not recommended to use this option
when multiple calibration dates (<code>calib.date</code>) are provided.
</p>


<h3>Author(s)</h3>

<p>Margret C. Fuchs, HZDR, Helmholtz-Institute Freiberg for Resource Technology (Germany) <br />
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>NNDC, Brookhaven National Laboratory <code style="white-space: pre;">&#8288;http://www.nndc.bnl.gov/&#8288;</code>
</p>


<h3>See Also</h3>

<p><a href="#topic+convert_Second2Gray">convert_Second2Gray</a>, <a href="#topic+get_RLum">get_RLum</a>, <a href="#topic+plot_RLum">plot_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) Simple function usage
##Basic calculation of the dose rate for a specific date
dose.rate &lt;-  calc_SourceDoseRate(measurement.date = "2012-01-27",
                                  calib.date = "2014-12-19",
                                  calib.dose.rate = 0.0438,
                                  calib.error = 0.0019)

##show results
get_RLum(dose.rate)

##(2) Usage in combination with another function (e.g., convert_Second2Gray() )
## load example data
data(ExampleData.DeValues, envir = environment())

## use the calculated variable dose.rate as input argument
## to convert De(s) to De(Gy)
convert_Second2Gray(ExampleData.DeValues$BT998, dose.rate)

##(3) source rate prediction and plotting
dose.rate &lt;-  calc_SourceDoseRate(measurement.date = "2012-01-27",
                                  calib.date = "2014-12-19",
                                  calib.dose.rate = 0.0438,
                                  calib.error = 0.0019,
                                  predict = 1000)
plot_RLum(dose.rate)


##(4) export output to a LaTeX table (example using the package 'xtable')
## Not run: 
xtable::xtable(get_RLum(dose.rate))


## End(Not run)

</code></pre>

<hr>
<h2 id='calc_Statistics'>Function to calculate statistic measures</h2><span id='topic+calc_Statistics'></span>

<h3>Description</h3>

<p>This function calculates a number of descriptive statistics for estimates
with a given standard error (SE), most fundamentally using error-weighted approaches.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_Statistics(
  data,
  weight.calc = "square",
  digits = NULL,
  n.MCM = NULL,
  na.rm = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_Statistics_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> object (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a> two columns: De (<code>data[, 1]</code>) and De error (<code>data[, 2]</code>).</p>
</td></tr>
<tr><td><code id="calc_Statistics_+3A_weight.calc">weight.calc</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
type of weight calculation. One out of <code>"reciprocal"</code> (weight is 1/error),
<code>"square"</code> (weight is 1/error^2). Default is <code>"square"</code>.</p>
</td></tr>
<tr><td><code id="calc_Statistics_+3A_digits">digits</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number of decimal places to be used when rounding numbers. If set to <code>NULL</code>
(default), no rounding occurs.</p>
</td></tr>
<tr><td><code id="calc_Statistics_+3A_n.mcm">n.MCM</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
number of samples drawn for Monte Carlo-based statistics.
<code>NULL</code> (the default) disables MC runs.</p>
</td></tr>
<tr><td><code id="calc_Statistics_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
indicating whether <code>NA</code> values should be stripped before the computation proceeds.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The option to use Monte Carlo Methods (<code>n.MCM</code>) allows calculating
all descriptive statistics based on random values. The distribution of these
random values is based on the Normal distribution with <code>De</code> values as
means and <code>De_error</code> values as one standard deviation. Increasing the
number of MCM-samples linearly increases computation time. On a Lenovo X230
machine evaluation of 25 Aliquots with n.MCM = 1000 takes 0.01 s, with
n = 100000, ca. 1.65 s. It might be useful to work with logarithms of these
values. See Dietze et al. (2016, Quaternary Geochronology) and the function
<a href="#topic+plot_AbanicoPlot">plot_AbanicoPlot</a> for details.
</p>


<h3>Value</h3>

<p>Returns a list with weighted and unweighted statistic measures.
</p>


<h3>Function version</h3>

<p>0.1.7
</p>


<h3>How to cite</h3>

<p>Dietze, M., 2025. calc_Statistics(): Function to calculate statistic measures. Function version 0.1.7. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)
, RLum Developer Team</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.DeValues, envir = environment())

## show a rough plot of the data to illustrate the non-normal distribution
plot_KDE(ExampleData.DeValues$BT998)

## calculate statistics and show output
str(calc_Statistics(ExampleData.DeValues$BT998))

## Not run: 
## now the same for 10000 normal distributed random numbers with equal errors
x &lt;- as.data.frame(cbind(rnorm(n = 10^5, mean = 0, sd = 1),
                         rep(0.001, 10^5)))

## note the congruent results for weighted and unweighted measures
str(calc_Statistics(x))

## End(Not run)

</code></pre>

<hr>
<h2 id='calc_ThermalLifetime'>Calculates the Thermal Lifetime using the Arrhenius equation</h2><span id='topic+calc_ThermalLifetime'></span>

<h3>Description</h3>

<p>The function calculates the thermal lifetime of charges for given E (in eV), s (in 1/s) and
T (in deg. C.) parameters. The function can be used in two operational modes:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_ThermalLifetime(
  E,
  s,
  T = 20,
  output_unit = "Ma",
  profiling = FALSE,
  profiling_config = NULL,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_ThermalLifetime_+3A_e">E</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
vector of trap depths in eV,
if <code>profiling = TRUE</code> only the first two elements are considered</p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_s">s</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
vector of frequency factor in 1/s,
if <code>profiling = TRUE</code> only the first two elements are considered</p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_t">T</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
temperature in deg. C for which the lifetime(s) will be calculated.
A vector can be provided.</p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_output_unit">output_unit</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
output unit of the calculated lifetimes, accepted
entries are: <code>"Ma"</code>, <code>"ka"</code>, <code>"a"</code>, <code>"d"</code>, <code>"h"</code>, <code>"min"</code>, <code>"s"</code></p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_profiling">profiling</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
this option allows to estimate uncertainties based on
given E and s parameters and their corresponding standard error
(cf. details and examples section)</p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_profiling_config">profiling_config</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
allows to set configuration parameters used for the profiling
(and only have an effect here). Supported parameters are:
</p>

<ul>
<li> <p><code>n</code> (number of MC runs),
</p>
</li>
<li> <p><code>E.distribution</code> (distribution used for the re-sampling for E) and
</p>
</li>
<li> <p><code>s.distribution</code> (distribution used for the re-sampling for s).
</p>
</li></ul>

<p>Currently only the normal distribution is supported
(e.g., <code>profiling_config = list(E.distribution = "norm")</code></p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output, currently only in combination with
<code>profiling = TRUE</code>.</p>
</td></tr>
<tr><td><code id="calc_ThermalLifetime_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed in combination with the plot output.
Standard plot parameters are supported (<a href="graphics.html#topic+plot.default">plot.default</a>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Mode 1 <code>(profiling = FALSE)</code></strong>
</p>
<p>An arbitrary set of input parameters (E, s, T) can be provided and the
function calculates the thermal lifetimes using the Arrhenius equation for
all possible combinations of these input parameters. An array with 3-dimensions
is returned that can be used for further analyses or graphical output (see example 1)
</p>
<p><strong>Mode 2 <code>(profiling = TRUE)</code></strong>
</p>
<p>This mode tries to profile the variation of the thermal lifetime for a chosen
temperature by accounting for the provided E and s parameters and their corresponding
standard errors, e.g., <code>E = c(1.600, 0.001)</code>
The calculation based on a Monte Carlo simulation, where values are sampled from a normal
distribution (for E and s).
</p>
<p><strong>Used equation (Arrhenius equation)</strong>
</p>
<p style="text-align: center;"><code class="reqn">\tau = 1/s exp(E/kT)</code>
</p>

<p>where:
<code class="reqn">\tau</code> in s as the mean time an electron spends in the trap for a given <code class="reqn">T</code>,
<code class="reqn">E</code> trap depth in eV,
<code class="reqn">s</code> the frequency factor in 1/s,
<code class="reqn">T</code> the temperature in K and <code class="reqn">k</code> the Boltzmann constant in eV/K (cf. Furetta, 2010).
</p>


<h3>Value</h3>

<p>A <a href="#topic+RLum.Results-class">RLum.Results</a> object is returned a along with a plot (for
<code>profiling = TRUE</code>). The output object contain the following slots:
</p>
<p><strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lifetimes</code> </td><td style="text-align: left;"> <a href="base.html#topic+array">array</a> or <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> calculated lifetimes </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>profiling_matrix</code> </td><td style="text-align: left;"> <a href="base.html#topic+matrix">matrix</a> </td><td style="text-align: left;"> profiling matrix used for the MC runs
</td>
</tr>

</table>

<p><strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>call</code> </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. calc_ThermalLifetime(): Calculates the Thermal Lifetime using the Arrhenius equation. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The profiling is currently based on re-sampling from a normal distribution, this
distribution assumption might be, however, not valid for given E and s parameters.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Furetta, C., 2010. Handbook of Thermoluminescence, Second Edition. World Scientific.
</p>


<h3>See Also</h3>

<p><a href="graphics.html#topic+matplot">graphics::matplot</a>, <a href="stats.html#topic+Normal">stats::rnorm</a>, <a href="#topic+get_RLum">get_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##EXAMPLE 1
##calculation for two trap-depths with similar frequency factor for different temperatures
E &lt;- c(1.66, 1.70)
s &lt;- 1e+13
T &lt;- 10:20
temp &lt;- calc_ThermalLifetime(
  E = E,
  s = s,
  T = T,
  output_unit = "Ma"
)
graphics::contour(x = E, y = T, z = temp$lifetimes[1,,],
        ylab = "Temperature [\u00B0C]",
        xlab = "Trap depth [eV]",
        main = "Thermal Lifetime Contour Plot"
)
mtext(side = 3, "(values quoted in Ma)")

##EXAMPLE 2
##profiling of thermal life time for E and s and their standard error
E &lt;- c(1.600, 0.003)
s &lt;- c(1e+13,1e+011)
T &lt;- 20
calc_ThermalLifetime(
  E = E,
  s = s,
  T = T,
  profiling = TRUE,
  output_unit = "Ma"
)

</code></pre>

<hr>
<h2 id='calc_TLLxTxRatio'>Calculate the Lx/Tx ratio for a given set of TL curves -beta version-</h2><span id='topic+calc_TLLxTxRatio'></span>

<h3>Description</h3>

<p>Calculate Lx/Tx ratio for a given set of TL curves.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_TLLxTxRatio(
  Lx.data.signal,
  Lx.data.background = NULL,
  Tx.data.signal,
  Tx.data.background = NULL,
  signal.integral.min,
  signal.integral.max
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_TLLxTxRatio_+3A_lx.data.signal">Lx.data.signal</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
TL data (x = temperature, y = counts) (TL signal)</p>
</td></tr>
<tr><td><code id="calc_TLLxTxRatio_+3A_lx.data.background">Lx.data.background</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
TL data (x = temperature, y = counts).
If no data are provided no background subtraction is performed.</p>
</td></tr>
<tr><td><code id="calc_TLLxTxRatio_+3A_tx.data.signal">Tx.data.signal</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
TL data (x = temperature, y = counts) (TL test signal)</p>
</td></tr>
<tr><td><code id="calc_TLLxTxRatio_+3A_tx.data.background">Tx.data.background</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
TL data (x = temperature, y = counts).
If no data are provided no background subtraction is performed.</p>
</td></tr>
<tr><td><code id="calc_TLLxTxRatio_+3A_signal.integral.min">signal.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
channel number for the lower signal integral bound
(e.g. <code>signal.integral.min = 100</code>)</p>
</td></tr>
<tr><td><code id="calc_TLLxTxRatio_+3A_signal.integral.max">signal.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
channel number for the upper signal integral bound
(e.g. <code>signal.integral.max = 200</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Uncertainty estimation</strong>
</p>
<p>The standard errors are calculated using the following generalised equation:
</p>
<p style="text-align: center;"><code class="reqn">SE_{signal} = abs(Signal_{net} * BG_f /BG_{signal})</code>
</p>

<p>where <code class="reqn">BG_f</code> is a term estimated by calculating the standard deviation of the sum of
the <code class="reqn">L_x</code> background counts and the sum of the <code class="reqn">T_x</code> background counts. However,
if both signals are similar the error becomes zero.
</p>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
Slot <code>data</code> contains a <a href="base.html#topic+list">list</a> with the following structure:
</p>
<div class="sourceCode"><pre>$ LxTx.table
.. $ LnLx
.. $ LnLx.BG
.. $ TnTx
.. $ TnTx.BG
.. $ Net_LnLx
.. $ Net_LnLx.Error
</pre></div>


<h3>Function version</h3>

<p>0.3.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Schmidt, C., 2025. calc_TLLxTxRatio(): Calculate the Lx/Tx ratio for a given set of TL curves -beta version-. Function version 0.3.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>This function has still BETA status!</strong> Please further note that a similar
background for both curves results in a zero error and is therefore set to <code>NA</code>.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany) <br />
Christoph Schmidt, University of Bayreuth (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+analyse_SAR.TL">analyse_SAR.TL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load package example data
data(ExampleData.BINfileData, envir = environment())

##convert Risoe.BINfileData into a curve object
temp &lt;- Risoe.BINfileData2RLum.Analysis(TL.SAR.Data, pos = 3)


Lx.data.signal &lt;- get_RLum(temp, record.id=1)
Lx.data.background &lt;- get_RLum(temp, record.id=2)
Tx.data.signal &lt;- get_RLum(temp, record.id=3)
Tx.data.background &lt;- get_RLum(temp, record.id=4)
signal.integral.min &lt;- 210
signal.integral.max &lt;- 230

output &lt;- calc_TLLxTxRatio(
 Lx.data.signal,
 Lx.data.background,
 Tx.data.signal,
 Tx.data.background,
 signal.integral.min,
 signal.integral.max)
get_RLum(output)

</code></pre>

<hr>
<h2 id='calc_WodaFuchs2008'>Obtain the equivalent dose using the approach by Woda and Fuchs 2008</h2><span id='topic+calc_WodaFuchs2008'></span>

<h3>Description</h3>

<p>The function generates a histogram-like reorganisation of the data, to
assess counts per bin. The log-transformed counts per bin are used to
calculate the second derivative of the data (i.e., the curvature of the
curve) and to find the central value of the bin hosting the distribution
maximum. A normal distribution model is fitted to the counts per bin
data to estimate the dose distribution parameters. The uncertainty of the
model is estimated based on all input equivalent doses smaller that of the
modelled central value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_WodaFuchs2008(data, breaks = NULL, plot = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_WodaFuchs2008_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> <a href="base.html#topic+vector">vector</a>, or <a href="#topic+RLum.Results-class">RLum.Results</a> object (<strong>required</strong>):
for <a href="base.html#topic+data.frame">data.frame</a>: either two columns: De (<code>values[,1]</code>) and De error
(<code>values[,2]</code>), or one: De (<code>values[,1]</code>). If a numeric vector or a
single-column data frame is provided, De error is set to <code>NA</code>.
For plotting multiple data sets, these must be provided as <code>list</code>
(e.g. <code>list(dataset1, dataset2)</code>).</p>
</td></tr>
<tr><td><code id="calc_WodaFuchs2008_+3A_breaks">breaks</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
Either number or locations of breaks. See <code style="white-space: pre;">&#8288;[hist]&#8288;</code> for details.
If missing, the number of breaks will be estimated based on the bin width
(as function of median error).</p>
</td></tr>
<tr><td><code id="calc_WodaFuchs2008_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="calc_WodaFuchs2008_+3A_...">...</code></td>
<td>
<p>Further plot arguments passed to the function.</p>
</td></tr>
</table>


<h3>Function version</h3>

<p>0.2.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., 2025. calc_WodaFuchs2008(): Obtain the equivalent dose using the approach by Woda and Fuchs 2008. Function version 0.2.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany),<br />
Michael Dietze, GFZ Potsdam (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Woda, C., Fuchs, M., 2008. On the applicability of the leading edge method to
obtain equivalent doses in OSL dating and dosimetry. Radiation Measurements 43, 26-37.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_FuchsLang2001">calc_FuchsLang2001</a>, <a href="#topic+calc_CentralDose">calc_CentralDose</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## read example data set
data(ExampleData.DeValues, envir = environment())

results &lt;- calc_WodaFuchs2008(
  data = ExampleData.DeValues$CA1,
   xlab = expression(paste(D[e], " [Gy]"))
 )

</code></pre>

<hr>
<h2 id='combine_De_Dr'>Combine Dose Rate and Equivalent Dose Distribution</h2><span id='topic+combine_De_Dr'></span>

<h3>Description</h3>

<p>A Bayesian statistical analysis of OSL age requiring dose rate sample.
Estimation contains a preliminary step for detecting outliers in the equivalent
dose sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>combine_De_Dr(
  De,
  s,
  Dr,
  int_OD,
  Age_range = c(1, 300),
  outlier_threshold = 0.05,
  outlier_method = "default",
  outlier_analysis_plot = FALSE,
  method_control = list(),
  par_local = TRUE,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="combine_De_Dr_+3A_de">De</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): a equivalent dose sample</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_s">s</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): a vector of measurement errors on the equivalent dose</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_dr">Dr</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): a dose rate sample</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_int_od">int_OD</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): the intrinsic overdispersion, typically the standard deviation
characterizing a dose-recovery test distribution</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_age_range">Age_range</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): the age range to be investigated by the algorithm, the larger
the value the more iterations are needed and the longer it takes. Should not be set too narrow, cut
the algorithm some slack.</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_outlier_threshold">outlier_threshold</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): the required significance level used
for the outlier detection. If set to <code>1</code>, no outliers are removed. If
<code>outlier_method = "RousseeuwCroux1993"</code>, the median distance is used as outlier threshold.
Please see details for further information.</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_outlier_method">outlier_method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): select the outlier detection
method, either <code>"default"</code> or <code>"RousseeuwCroux1993"</code>. See details for further information.</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_outlier_analysis_plot">outlier_analysis_plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the
outlier analysis plot. Note: the outlier analysis will happen independently
of the plot output.</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>with default</em>): named <a href="base.html#topic+list">list</a> of further parameters passed down
to the <a href="rjags.html#topic+rjags-package">rjags::rjags</a> modelling</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_par_local">par_local</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): if set to <code>TRUE</code> the function uses its
own <a href="graphics.html#topic+par">graphics::par</a> settings (which will end in two plots next to each other)</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="combine_De_Dr_+3A_...">...</code></td>
<td>
<p>a few further arguments to fine-tune the plot output such as
<code>cdf_ADr_quantiles</code> (<code>TRUE</code>/<code>FALSE</code>), <code>legend.pos</code>, <code>legend</code> (<code>TRUE</code>/<code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Outlier detection</strong>
</p>
<p>Two different outlier detection methods are implemented (full details are given
in the cited literature).
</p>

<ol>
<li><p> The <em>default</em> and recommend method, uses quantiles to compare prior and
posterior distributions of the individual variances of the equivalent doses.
If the corresponding quantile in the corresponding posterior distribution is larger
than the quantile in the prior distribution, the value is marked
as outlier (cf. Galharret et al., preprint)
</p>
</li>
<li><p> The alternative method employs the method suggested by Rousseeuw and Croux (1993)
using the absolute median distance.
</p>
</li></ol>

<p><strong>Parameters available for <code>method_control</code></strong>
</p>
<p>The parameters listed below are used to granular control Bayesian modelling using
<a href="rjags.html#topic+rjags-package">rjags::rjags</a>. Internally the functions <code>.calc_IndividualAgeModel()</code> and
<code>.calc_BayesianCentraAgelModel()</code>. The parameter settings affect both models.
Note: <code>method_control</code> expects a <strong>named</strong> list of parameters
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>PARAMETER</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>DEFAULT</strong> </td><td style="text-align: left;"> <strong>REMARKS</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>variable.names_IAM</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> <code>c('A', 'a', 'sig_a')</code> </td><td style="text-align: left;"> variables names to be monitored in the modelling process using the internal function <code>.calc_IndividualAgeModel()</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>variable.names_BCAM</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> <code>c('A', 'D_e')</code> </td><td style="text-align: left;"> variables names to be monitored in the modelling process using the internal function <code>.calc_BayesianCentraAgelModel()</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.chains</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> number of MCMC chains</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.adapt</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> </td><td style="text-align: left;"> <code>1000</code> </td><td style="text-align: left;"> number of iterations for the adaptation</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.iter</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> </td><td style="text-align: left;"> <code>5000</code> </td><td style="text-align: left;"> number of iterations to monitor cf. <a href="rjags.html#topic+coda.samples">rjags::coda.samples</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>thin</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> <code>1</code> </td><td style="text-align: left;"> thinning interval for the monitoring cf. <a href="rjags.html#topic+coda.samples">rjags::coda.samples</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>diag</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> additional terminal convergence diagnostic.
<code>FALSE</code> if <code>verbose = FALSE</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>progress.bar</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> enable/disable progress bar. <code>FALSE</code> if <code>verbose = FALSE</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>quiet</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> silence terminal output. Set to <code>TRUE</code> if <code>verbose = FALSE</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>return_mcmc</code></td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> return additional MCMC diagnostic information</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>The function returns a plot if <code>plot = TRUE</code> and an <a href="#topic+RLum.Results-class">RLum.Results</a>
object with the following slots:
</p>
<p><code style="white-space: pre;">&#8288;@data&#8288;</code><br />
<code>.. $Ages</code>: a <a href="base.html#topic+numeric">numeric</a> vector with the modelled ages to be further analysed or visualised<br />
<code>.. $Ages_stats</code>: a <a href="base.html#topic+data.frame">data.frame</a> with sum HPD, CI 68% and CI 95% for the ages <br />
<code>.. $outliers_index</code>: the index with the detected outliers<br />
<code>.. $cdf_ADr_mean</code> : empirical cumulative density distribution A * Dr (mean)<br />
<code>.. $cdf_ADr_quantiles</code> : empirical cumulative density distribution A * Dr (quantiles .025,.975)<br />
<code>.. $cdf_De_no_outlier</code> : empirical cumulative density distribution of the De with no outliers<br />
<code>.. $cdf_De_initial</code> : empirical cumulative density distribution of the initial De<br />
<code>.. $mcmc_IAM</code> : the MCMC list of the Individual Age Model, only of <code>method_control = list(return_mcmc = TRUE)</code> otherwise <code>NULL</code><br />
<code>.. $mcmc_BCAM</code> : the MCMC list of the Bayesian Central Age Model, only of <code>method_control = list(return_mcmc = TRUE)</code> otherwise <code>NULL</code><br />
</p>
<p><code style="white-space: pre;">&#8288;@info&#8288;</code><br />
<code>.. $call</code>: the original function call<br />
<code>.. $model_IAM</code>: the BUGS model used to derive the individual age<br />
<code>.. $model_BCAM</code>: the BUGS model used to calculate the Bayesian Central Age<br />
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Philippe, A., Galharret, J., Mercier, N., Kreutzer, S., 2025. combine_De_Dr(): Combine Dose Rate and Equivalent Dose Distribution. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Anne Philippe, Universit√© de Nantes (France),
Jean-Michel Galharret, Universit√© de Nantes (France),
Norbert Mercier, IRAMAT-CRP2A, Universit√© Bordeaux Montaigne (France),
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Mercier, N., Galharret, J.-M., Tribolo, C., Kreutzer, S., Philippe, A., preprint.
Luminescence age calculation through Bayesian convolution of equivalent dose and
dose-rate distributions: the De_Dr model. Geochronology, 1-22.
</p>
<p>Galharret, J-M., Philippe, A., Mercier, N., preprint. Detection of outliers with
a Bayesian hierarchical model: application to the single-grain luminescence dating method.
Electronic Journal of Applied Statistics
</p>
<p><strong>Further reading</strong>
</p>
<p>Rousseeuw, P.J., Croux, C., 1993. Alternatives to the median absolute deviation.
Journal of the American Statistical Association 88, 1273‚Äì1283. <a href="https://doi.org/10.2307/2291267">doi:10.2307/2291267</a>
</p>
<p>Rousseeuw, P.J., Debruyne, M., Engelen, S., Hubert, M., 2006. Robustness and outlier detection in chemometrics.
Critical Reviews in Analytical Chemistry 36, 221‚Äì242. <a href="https://doi.org/10.1080/10408340600969403">doi:10.1080/10408340600969403</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+plot_OSLAgeSummary">plot_OSLAgeSummary</a>, <a href="rjags.html#topic+rjags-package">rjags::rjags</a>, <a href="mclust.html#topic+mclust-package">mclust::mclust-package</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## set parameters
Dr &lt;- stats::rlnorm (1000, 0, 0.3)
De &lt;- 50*sample(Dr, 50, replace = TRUE)
s &lt;- stats::rnorm(50, 10, 2)

## run modelling
## note: modify parameters for more realistic results
## Not run: 
results &lt;- combine_De_Dr(
 Dr = Dr,
 int_OD = 0.1,
 De,
 s,
 Age_range = c(0,100),
 method_control = list(
  n.iter = 100,
  n.chains = 1))

## show models used
writeLines(results@info$model_IAM)
writeLines(results@info$model_BCAM)

## End(Not run)

</code></pre>

<hr>
<h2 id='convert_Activity2Concentration'>Convert Nuclide Activities to Abundance and Vice Versa</h2><span id='topic+convert_Activity2Concentration'></span>

<h3>Description</h3>

<p>The function performs the conversion of the specific activities into
mass abundance and vice versa for the radioelements U, Th, and K to
harmonise the measurement unit with the required data input unit of
potential analytical tools for, e.g. dose rate calculation or related
functions such as <a href="#topic+use_DRAC">use_DRAC</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_Activity2Concentration(data, input_unit = "activity", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_Activity2Concentration_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
provide dose rate data (activity or concentration) in three columns.
The first column indicates the nuclide, the 2nd column measured value and
in the 3rd column its error value. Allowed nuclide data are
<code>'U-238'</code>, <code>'Th-232'</code> and <code>'K-40'</code>. See examples for an example.</p>
</td></tr>
<tr><td><code id="convert_Activity2Concentration_+3A_input_unit">input_unit</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
specify unit of input data given in the dose rate data frame, choose between
<code>"activity"</code> (considered as given Bq/kg) and <code>"abundance"</code> (considered as given in mug/g or mass. %).
The default value is <code>"activity"</code></p>
</td></tr>
<tr><td><code id="convert_Activity2Concentration_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The conversion from nuclide activity of a sample to nuclide concentration
is performed using conversion factors that are based on the mass-related
specific activity of the respective nuclide.
</p>
<p>Constants used in this function were obtained from <code style="white-space: pre;">&#8288;https://physics.nist.gov/cuu/Constants/&#8288;</code>
all atomic weights and composition values from
<code style="white-space: pre;">&#8288;https://www.nist.gov/pml/atomic-weights-and-isotopic-compositions-relative-atomic-masses&#8288;</code>
and the nuclide data from <code style="white-space: pre;">&#8288;https://www.iaea.org/resources/databases/livechart-of-nuclides-advanced-version&#8288;</code>
</p>
<p>The factors can be calculated using the equation:
</p>
<p style="text-align: center;"><code class="reqn">
A = N_A \frac{N_{abund}}{N_{mol.mass}}  ln(2) / N.half.life
</code>
</p>

<p>to convert in ¬µg/g we further use:
</p>
<p style="text-align: center;"><code class="reqn">
f = A / 10^6
</code>
</p>

<p>where:
</p>

<ul>
<li> <p><code>N_A</code> - Avogadro constant in 1/mol
</p>
</li>
<li> <p><code>A</code> - specific activity of the nuclide in Bq/kg
</p>
</li>
<li> <p><code>N.abund</code> - relative natural abundance of the isotope
</p>
</li>
<li> <p><code>N.mol.mass</code> molar mass in kg/mol
</p>
</li>
<li> <p><code>N.half.life</code> half-life of the nuclide in s
</p>
</li></ul>

<p>example for calculating the activity of the radionuclide U-238:
</p>

<ul>
<li> <p><code>N_A</code> = 6.02214076e+23 (1/mol)
</p>
</li>
<li> <p><code>T_0.5</code> = 1.41e+17 (s)
</p>
</li>
<li> <p><code>m_U_238</code> = 0.23802891 (kg/mol)
</p>
</li>
<li> <p><code>U_abund</code> = 0.992745 (unitless)
</p>
</li></ul>

<p style="text-align: center;"><code class="reqn">A_{U} = N_{A} * U_{abund} / m_{U_238} * ln(2) / T_{1/2} = 2347046</code>
</p>
<p> (Bq/kg)
</p>
<p style="text-align: center;"><code class="reqn">f.U = A_{U} / 10^6</code>
</p>



<h3>Value</h3>

<p>Returns an <a href="#topic+RLum.Results-class">RLum.Results</a> object with a <a href="base.html#topic+data.frame">data.frame</a> containing
input and newly calculated values. Please not that in the column header ¬µg/g
is written as <code>mug/g</code> due to the R requirement to maintain packages portable using
ASCII characters only.
</p>


<h3>Function version</h3>

<p>0.1.2
</p>


<h3>How to cite</h3>

<p>Fuchs, M.C., 2025. convert_Activity2Concentration(): Convert Nuclide Activities to Abundance and Vice Versa. Function version 0.1.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Although written otherwise for historical reasons. Input values must be element values.
For instance, if a value is provided for U-238 the function assumes that this value
represents the sum (activity or abundance) of U-238, U-235 and U-234.
In other words, 1 ¬µg/g of U means that this is the composition of 0.992 parts of U-238,
0.000054 parts of U-234, and 0.00072 parts of U-235.
</p>


<h3>Author(s)</h3>

<p>Margret C. Fuchs, Helmholtz-Institute Freiberg for Resource Technology (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Debertin, K., Helmer, R.G., 1988. Gamma- and X-ray Spectrometry with
Semiconductor Detectors, Elsevier Science Publishers, p.283
</p>
<p>Wiechen, A., Ruehle, H., Vogl, K., 2013. Bestimmung der massebezogenen
Aktivitaet von Radionukliden. AEQUIVAL/MASSAKT, ISSN 1865-8725,
<a href="https://www.bmuv.de/fileadmin/Daten_BMU/Download_PDF/Strahlenschutz/aequival-massakt_v2013-07_bf.pdf">https://www.bmuv.de/fileadmin/Daten_BMU/Download_PDF/Strahlenschutz/aequival-massakt_v2013-07_bf.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##construct data.frame
data &lt;- data.frame(
 NUCLIDES = c("U-238", "Th-232", "K-40"),
 VALUE = c(40,80,100),
 VALUE_ERROR = c(4,8,10),
 stringsAsFactors = FALSE)

##perform analysis
convert_Activity2Concentration(data)

</code></pre>

<hr>
<h2 id='convert_BIN2CSV'>Export Risoe BIN-file(s) to CSV-files</h2><span id='topic+convert_BIN2CSV'></span>

<h3>Description</h3>

<p>This function is a wrapper function around the functions <a href="#topic+read_BIN2R">read_BIN2R</a> and
<a href="#topic+write_RLum2CSV">write_RLum2CSV</a> and it imports a Risoe BIN-file and directly exports its
content to CSV-files. If nothing is set for the argument <code>path</code>
(<a href="#topic+write_RLum2CSV">write_RLum2CSV</a>) the input folder will become the output folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_BIN2CSV(file, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_BIN2CSV_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the BIN-file to be converted to CSV-files</p>
</td></tr>
<tr><td><code id="convert_BIN2CSV_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+read_BIN2R">read_BIN2R</a> and <a href="#topic+write_RLum2CSV">write_RLum2CSV</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns either a CSV-file (or many of them) or for the
option <code>export == FALSE</code> a list comprising objects of type <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_BIN2CSV(): Export Risoe BIN-file(s) to CSV-files. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>,
<a href="utils.html#topic+write.table">utils::write.table</a>, <a href="#topic+write_RLum2CSV">write_RLum2CSV</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##transform Risoe.BINfileData values to a list
data(ExampleData.BINfileData, envir = environment())
convert_BIN2CSV(subset(CWOSL.SAR.Data, POSITION == 1), export = FALSE)

## Not run: 
##select your BIN-file
file &lt;- file.choose()

##convert
convert_BIN2CSV(file)


## End(Not run)

</code></pre>

<hr>
<h2 id='convert_Concentration2DoseRate'>Dose-rate conversion function</h2><span id='topic+convert_Concentration2DoseRate'></span>

<h3>Description</h3>

<p>This function converts radionuclide concentrations
(K in %, Th and U in ppm) into dose rates (Gy/ka). Beta-dose rates are also
attenuated for the grain size. Beta and gamma-dose rates are corrected
for the water content. This function converts concentrations into dose rates
(Gy/ka) and corrects for grain size attenuation and water content
</p>
<p>Dose rate conversion factors can be chosen from Adamiec and Aitken (1998),
Guerin et al. (2011), Liritzis et al. (201) and Cresswell et al. (2018).
Default is Guerin et al. (2011).
</p>
<p>Grain size correction for beta dose rates is achieved using the correction
factors published by Gu√©rin et al. (2012).
</p>
<p>Water content correction is based on factors provided by Aitken (1985),
with the factor for beta dose rate being 1.25 and for gamma 1.14.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_Concentration2DoseRate(input, conversion = "Guerinetal2011")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_Concentration2DoseRate_+3A_input">input</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>): a table containing all relevant information
for each individual layer if nothing is provided, the function returns a template <a href="base.html#topic+data.frame">data.frame</a>
Please note that until one dataset per input is supported!</p>
</td></tr>
<tr><td><code id="convert_Concentration2DoseRate_+3A_conversion">conversion</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): which dose rate conversion factors to use,
defaults uses Gu√©rin et al. (2011). For accepted values see <a href="#topic+BaseDataSet.ConversionFactors">BaseDataSet.ConversionFactors</a></p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>The input data</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
COLUMN </td><td style="text-align: left;"> DATA TYPE </td><td style="text-align: left;"> DESCRIPTION</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Mineral</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> <code>'FS'</code> for feldspar, <code>'Q'</code> for quartz</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>K</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> K nuclide content in %</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>K_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on K nuclide content in %</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Th</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Th nuclide content in ppm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Th_SE</code> </td><td style="text-align: left;"> <code>numeric</code> error on Th nuclide content in ppm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>U</code> </td><td style="text-align: left;"> <code>numeric</code> U nuclide content in ppm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>U_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> error on U nuclide content in ppm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>GrainSize</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> average grain size in ¬µm</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WaterContent</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> mean water content in %</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>WaterContent_SE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> relative error on water content
</td>
</tr>

</table>

<p><strong>Water content</strong>
The water content provided by the user should be calculated according to:
</p>
<p style="text-align: center;"><code class="reqn">(Wet_weight - Dry_weight) / Dry_weight * 100</code>
</p>

<p>The unit for the weight is gram (g).
</p>


<h3>Value</h3>

<p>The function returns an <a href="#topic+RLum.Results-class">RLum.Results</a> object for which the first
element is <a href="base.html#topic+matrix">matrix</a> with the converted values. If no input is provided, the
function returns a template <a href="base.html#topic+data.frame">data.frame</a> that can be used as input.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Riedesel, S., Autzen, M., 2025. convert_Concentration2DoseRate(): Dose-rate conversion function. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Svenja Riedesel, Aberystwyth University (United Kingdom) <br />
Martin Autzen, DTU NUTECH Center for Nuclear Technologies (Denmark)
, RLum Developer Team</p>


<h3>References</h3>

<p>Adamiec, G., Aitken, M.J., 1998. Dose-rate conversion factors: update. Ancient TL 16, 37-46.
</p>
<p>Cresswell., A.J., Carter, J., Sanderson, D.C.W., 2018. Dose rate conversion parameters:
Assessment of nuclear data. Radiation Measurements 120, 195-201.
</p>
<p>Guerin, G., Mercier, N., Adamiec, G., 2011. Dose-rate conversion factors: update.
Ancient TL, 29, 5-8.
</p>
<p>Guerin, G., Mercier, N., Nathan, R., Adamiec, G., Lefrais, Y., 2012. On the use
of the infinite matrix assumption and associated concepts: A critical review.
Radiation Measurements, 47, 778-785.
</p>
<p>Liritzis, I., Stamoulis, K., Papachristodoulou, C., Ioannides, K., 2013.
A re-evaluation of radiation dose-rate conversion factors. Mediterranean
Archaeology and Archaeometry 13, 1-15.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## create input template
input &lt;- convert_Concentration2DoseRate()

## fill input
input$Mineral &lt;- "FS"
input$K &lt;- 2.13
input$K_SE &lt;- 0.07
input$Th &lt;- 9.76
input$Th_SE &lt;- 0.32
input$U &lt;- 2.24
input$U_SE &lt;- 0.12
input$GrainSize &lt;- 200
input$WaterContent &lt;- 30
input$WaterContent_SE &lt;- 5

## convert
convert_Concentration2DoseRate(input)

</code></pre>

<hr>
<h2 id='convert_CW2pHMi'>Transform a CW-OSL curve into a pHM-OSL curve via interpolation under
hyperbolic modulation conditions</h2><span id='topic+convert_CW2pHMi'></span><span id='topic+CW2pHMi'></span>

<h3>Description</h3>

<p>This function transforms a conventionally measured continuous-wave (CW)
OSL-curve to a pseudo hyperbolic modulated (pHM) curve under hyperbolic
modulation conditions using the interpolation procedure described by Bos &amp;
Wallinga (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_CW2pHMi(values, delta)

CW2pHMi(values, delta)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_CW2pHMi_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> with measured curve data of type
stimulation time (t) (<code>values[,1]</code>) and measured counts (cts) (<code>values[,2]</code>).</p>
</td></tr>
<tr><td><code id="convert_CW2pHMi_+3A_delta">delta</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
stimulation rate parameter, if no value is given, the optimal value is
estimated automatically (see details). Smaller values of delta produce more
points in the rising tail of
the curve.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The complete procedure of the transformation is described in Bos &amp; Wallinga
(2012). The input <code>data.frame</code> consists of two columns: time (t) and
count values (CW(t))
</p>
<p><strong>Internal transformation steps</strong>
</p>
<p>(1) log(CW-OSL) values
</p>
<p>(2)
Calculate t' which is the transformed time:
</p>
<p style="text-align: center;"><code class="reqn">t' = t-(1/\delta)*log(1+\delta*t)</code>
</p>

<p>(3)
Interpolate CW(t'), i.e. use the log(CW(t)) to obtain the count values
for the transformed time (t'). Values beyond <code>min(t)</code> and <code>max(t)</code>
produce <code>NA</code> values.
</p>
<p>(4)
Select all values for t' &lt; <code>min(t)</code>, i.e. values beyond the time
resolution of t. Select the first two values of the transformed data set
which contain no <code>NA</code> values and use these values for a linear fit
using <a href="stats.html#topic+lm">lm</a>.
</p>
<p>(5)
Extrapolate values for t' &lt; <code>min(t)</code> based on the previously
obtained fit parameters.
</p>
<p>(6)
Transform values using
</p>
<p style="text-align: center;"><code class="reqn">pHM(t) = (\delta*t/(1+\delta*t))*c*CW(t')</code>
</p>

<p style="text-align: center;"><code class="reqn">c = (1+\delta*P)/\delta*P</code>
</p>

<p style="text-align: center;"><code class="reqn">P = length(stimulation~period)</code>
</p>

<p>(7) Combine all values and truncate all values for t' &gt; <code>max(t)</code>
</p>
<p><strong>NOTE:</strong>
The number of values for t' &lt; <code>min(t)</code> depends on the stimulation rate
parameter <code>delta</code>. To avoid the production of too many artificial data
at the raising tail of the determined pHM curve, it is recommended to use
the automatic estimation routine for <code>delta</code>, i.e. provide no value for
<code>delta</code>.
</p>


<h3>Value</h3>

<p>The function returns the same data type as the input data type with
the transformed curve values.
</p>
<p><strong><code>RLum.Data.Curve</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$CW2pHMi.x.t&#8288;</code> </td><td style="text-align: left;">: transformed time values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$CW2pHMi.method&#8288;</code> </td><td style="text-align: left;">: used method for the production of the new data points
</td>
</tr>

</table>

<p><strong><code>data.frame</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$x&#8288;</code> </td><td style="text-align: left;">: time</td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$y.t&#8288;</code> </td><td style="text-align: left;">: transformed count values</td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$x.t&#8288;</code> </td><td style="text-align: left;">: transformed time values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$method&#8288;</code> </td><td style="text-align: left;">: used method for the production of the new data points
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.2.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_CW2pHMi(): Transform a CW-OSL curve into a pHM-OSL curve via interpolation under hyperbolic modulation conditions. Function version 0.2.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>According to Bos &amp; Wallinga (2012), the number of extrapolated points
should be limited to avoid artificial intensity data. If <code>delta</code> is
provided manually and more than two points are extrapolated, a warning
message is returned.
</p>
<p>The function <a href="stats.html#topic+approx">approx</a> may produce some <code>Inf</code> and <code>NaN</code> data.
The function tries to manually interpolate these values by calculating
the <code>mean</code> using the adjacent channels. If two invalid values are succeeding,
the values are removed and no further interpolation is attempted.
In every case a warning message is shown.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Based on comments and suggestions from:<br />
Adrie J.J. Bos, Delft University of Technology, The Netherlands
, RLum Developer Team</p>


<h3>References</h3>

<p>Bos, A.J.J. &amp; Wallinga, J., 2012. How to visualize quartz OSL
signal components. Radiation Measurements, 47, 752-758.<br />
</p>
<p><strong>Further Reading</strong>
</p>
<p>Bulur, E., 1996. An Alternative Technique For
Optically Stimulated Luminescence (OSL) Experiment. Radiation Measurements,
26, 701-709.
</p>
<p>Bulur, E., 2000. A simple transformation for converting CW-OSL curves to
LM-OSL curves. Radiation Measurements, 32, 141-145.
</p>


<h3>See Also</h3>

<p><a href="#topic+convert_CW2pLM">convert_CW2pLM</a>, <a href="#topic+convert_CW2pLMi">convert_CW2pLMi</a>, <a href="#topic+convert_CW2pPMi">convert_CW2pPMi</a>,
<a href="#topic+fit_LMCurve">fit_LMCurve</a>, <a href="stats.html#topic+lm">lm</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) - simple transformation

##load CW-OSL curve data
data(ExampleData.CW_OSL_Curve, envir = environment())

##transform values
values.transformed &lt;- convert_CW2pHMi(ExampleData.CW_OSL_Curve)

##plot
plot(values.transformed$x, values.transformed$y.t, log = "x")

##(2) - load CW-OSL curve from BIN-file and plot transformed values

##load BINfile
#BINfileData&lt;-readBIN2R("[path to BIN-file]")
data(ExampleData.BINfileData, envir = environment())

##grep first CW-OSL curve from ALQ 1
curve.ID&lt;-CWOSL.SAR.Data@METADATA[CWOSL.SAR.Data@METADATA[,"LTYPE"]=="OSL" &amp;
                                    CWOSL.SAR.Data@METADATA[,"POSITION"]==1
                                  ,"ID"]

curve.HIGH&lt;-CWOSL.SAR.Data@METADATA[CWOSL.SAR.Data@METADATA[,"ID"]==curve.ID[1]
                                    ,"HIGH"]

curve.NPOINTS&lt;-CWOSL.SAR.Data@METADATA[CWOSL.SAR.Data@METADATA[,"ID"]==curve.ID[1]
                                       ,"NPOINTS"]

##combine curve to data set

curve&lt;-data.frame(x = seq(curve.HIGH/curve.NPOINTS,curve.HIGH,
                          by = curve.HIGH/curve.NPOINTS),
                  y=unlist(CWOSL.SAR.Data@DATA[curve.ID[1]]))


##transform values

curve.transformed &lt;- convert_CW2pHMi(curve)

##plot curve
plot(curve.transformed$x, curve.transformed$y.t, log = "x")


##(3) - produce Fig. 4 from Bos &amp; Wallinga (2012)

##load data
data(ExampleData.CW_OSL_Curve, envir = environment())
values &lt;- CW_Curve.BosWallinga2012

##open plot area
plot(NA, NA,
     xlim=c(0.001,10),
     ylim=c(0,8000),
     ylab="pseudo OSL (cts/0.01 s)",
     xlab="t [s]",
     log="x",
     main="Fig. 4 - Bos &amp; Wallinga (2012)")

values.t &lt;- convert_CW2pLMi(values, P = 1/20)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col="red" ,lwd=1.3)
text(0.03,4500,"LM", col="red" ,cex=.8)

values.t &lt;- convert_CW2pHMi(values, delta = 40)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col="black", lwd=1.3)
text(0.005,3000,"HM", cex=.8)

values.t &lt;- convert_CW2pPMi(values, P = 1/10)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col="blue", lwd=1.3)
text(0.5,6500,"PM", col="blue" ,cex=.8)

</code></pre>

<hr>
<h2 id='convert_CW2pLM'>Transform a CW-OSL curve into a pLM-OSL curve</h2><span id='topic+convert_CW2pLM'></span><span id='topic+CW2pLM'></span>

<h3>Description</h3>

<p>Transforms a conventionally measured continuous-wave (CW) curve into a
pseudo linearly modulated (pLM) curve using the equations given in Bulur
(2000).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_CW2pLM(values)

CW2pLM(values)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_CW2pLM_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
<code>RLum.Data.Curve</code> data object. Alternatively, a <code>data.frame</code> of the measured
curve data of type stimulation time (t) (<code>values[,1]</code>) and measured counts (cts)
(<code>values[,2]</code>) can be provided.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>According to Bulur (2000) the curve data are transformed by introducing two
new parameters <code>P</code> (stimulation period) and <code>u</code> (transformed time):
</p>
<p style="text-align: center;"><code class="reqn">P=2*max(t)</code>
</p>
 <p style="text-align: center;"><code class="reqn">u=\sqrt{(2*t*P)}</code>
</p>

<p>The new count values are then calculated by
</p>
<p style="text-align: center;"><code class="reqn">ctsNEW = cts(u/P)</code>
</p>

<p>and the returned <code>data.frame</code> is produced by: <code>data.frame(u,ctsNEW)</code>
</p>
<p>The output of the function can be further used for LM-OSL fitting.
</p>


<h3>Value</h3>

<p>The function returns the same data type as the input data type with
the transformed curve values (<a href="base.html#topic+data.frame">data.frame</a> or <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>).
</p>


<h3>Function version</h3>

<p>0.4.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_CW2pLM(): Transform a CW-OSL curve into a pLM-OSL curve. Function version 0.4.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The transformation is recommended for curves recorded with a channel
resolution of at least 0.05 s/channel.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Bulur, E., 2000. A simple transformation for converting CW-OSL
curves to LM-OSL curves. Radiation Measurements, 32, 141-145.
</p>
<p><strong>Further Reading</strong>
</p>
<p>Bulur, E., 1996. An Alternative Technique For Optically Stimulated
Luminescence (OSL) Experiment. Radiation Measurements, 26, 701-709.
</p>


<h3>See Also</h3>

<p><a href="#topic+CW2pHMi">CW2pHMi</a>, <a href="#topic+CW2pLMi">CW2pLMi</a>, <a href="#topic+CW2pPMi">CW2pPMi</a>, <a href="#topic+fit_LMCurve">fit_LMCurve</a>, <a href="stats.html#topic+lm">lm</a>,
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##read curve from CWOSL.SAR.Data transform curve and plot values
data(ExampleData.BINfileData, envir = environment())

##read id for the 1st OSL curve
id.OSL &lt;- CWOSL.SAR.Data@METADATA[CWOSL.SAR.Data@METADATA[,"LTYPE"] == "OSL","ID"]

##produce x and y (time and count data for the data set)
x&lt;-seq(CWOSL.SAR.Data@METADATA[id.OSL[1],"HIGH"]/CWOSL.SAR.Data@METADATA[id.OSL[1],"NPOINTS"],
       CWOSL.SAR.Data@METADATA[id.OSL[1],"HIGH"],
       by = CWOSL.SAR.Data@METADATA[id.OSL[1],"HIGH"]/CWOSL.SAR.Data@METADATA[id.OSL[1],"NPOINTS"])
y &lt;- unlist(CWOSL.SAR.Data@DATA[id.OSL[1]])
values &lt;- data.frame(x,y)

##transform values
values.transformed &lt;- convert_CW2pLM(values)

##plot
plot(values.transformed)

</code></pre>

<hr>
<h2 id='convert_CW2pLMi'>Transform a CW-OSL curve into a pLM-OSL curve via interpolation under linear
modulation conditions</h2><span id='topic+convert_CW2pLMi'></span><span id='topic+CW2pLMi'></span>

<h3>Description</h3>

<p>Transforms a conventionally measured continuous-wave (CW) OSL-curve into a
pseudo linearly modulated (pLM) curve under linear modulation conditions
using the interpolation procedure described by Bos &amp; Wallinga (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_CW2pLMi(values, P)

CW2pLMi(values, P)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_CW2pLMi_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <code>data.frame</code> with measured curve data of type
stimulation time (t) (<code>values[,1]</code>) and measured counts (cts) (<code>values[,2]</code>)</p>
</td></tr>
<tr><td><code id="convert_CW2pLMi_+3A_p">P</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
stimulation time in seconds. If no value is given the optimal value is
estimated automatically (see details). Greater values of P produce more
points in the rising tail of the curve.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The complete procedure of the transformation is given in Bos &amp; Wallinga
(2012). The input <code>data.frame</code> consists of two columns: time (t) and
count values (CW(t))
</p>
<p><strong>Nomenclature</strong>
</p>

<ul>
<li><p> P = stimulation time (s)
</p>
</li>
<li><p> 1/P = stimulation rate (1/s)
</p>
</li></ul>

<p><strong>Internal transformation steps</strong>
</p>
<p>(1)
log(CW-OSL) values
</p>
<p>(2)
Calculate t' which is the transformed time:
</p>
<p style="text-align: center;"><code class="reqn">t' = 1/2*1/P*t^2</code>
</p>

<p>(3)
Interpolate CW(t'), i.e. use the log(CW(t)) to obtain the count values
for the transformed time (t'). Values beyond <code>min(t)</code> and <code>max(t)</code>
produce <code>NA</code> values.
</p>
<p>(4)
Select all values for t' &lt; <code>min(t)</code>, i.e. values beyond the time resolution
of t. Select the first two values of the transformed data set which contain
no <code>NA</code> values and use these values for a linear fit using <a href="stats.html#topic+lm">lm</a>.
</p>
<p>(5)
Extrapolate values for t' &lt; <code>min(t)</code> based on the previously obtained
fit parameters.
</p>
<p>(6)
Transform values using
</p>
<p style="text-align: center;"><code class="reqn">pLM(t) = t/P*CW(t')</code>
</p>

<p>(7)
Combine values and truncate all values for t' &gt; <code>max(t)</code>
</p>
<p><strong>NOTE:</strong>
The number of values for t' &lt; <code>min(t)</code> depends on the stimulation
period (P) and therefore on the stimulation rate 1/P. To avoid the
production of too many artificial data at the raising tail of the determined
pLM curves it is recommended to use the automatic estimation routine for
<code>P</code>, i.e. provide no own value for <code>P</code>.
</p>


<h3>Value</h3>

<p>The function returns the same data type as the input data type with
the transformed curve values.
</p>
<p><strong><code>RLum.Data.Curve</code></strong>
</p>

<table>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$CW2pLMi.x.t&#8288;</code> </td><td style="text-align: left;">: transformed time values </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$CW2pLMi.method&#8288;</code> </td><td style="text-align: left;">: used method for the production of the new data points
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.3.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_CW2pLMi(): Transform a CW-OSL curve into a pLM-OSL curve via interpolation under linear modulation conditions. Function version 0.3.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>According to Bos &amp; Wallinga (2012) the number of extrapolated points
should be limited to avoid artificial intensity data. If <code>P</code> is
provided manually and more than two points are extrapolated, a warning
message is returned.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
</p>
<p>Based on comments and suggestions from:<br />
Adrie J.J. Bos, Delft University of Technology, The Netherlands
, RLum Developer Team</p>


<h3>References</h3>

<p>Bos, A.J.J. &amp; Wallinga, J., 2012. How to visualize quartz OSL
signal components. Radiation Measurements, 47, 752-758.
</p>
<p><strong>Further Reading</strong>
</p>
<p>Bulur, E., 1996. An Alternative Technique For
Optically Stimulated Luminescence (OSL) Experiment. Radiation Measurements,
26, 701-709.
</p>
<p>Bulur, E., 2000. A simple transformation for converting CW-OSL curves to
LM-OSL curves. Radiation Measurements, 32, 141-145.
</p>


<h3>See Also</h3>

<p><a href="#topic+convert_CW2pLM">convert_CW2pLM</a>, <a href="#topic+convert_CW2pHMi">convert_CW2pHMi</a>, <a href="#topic+convert_CW2pPMi">convert_CW2pPMi</a>,
<a href="#topic+fit_LMCurve">fit_LMCurve</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1)
##load CW-OSL curve data
data(ExampleData.CW_OSL_Curve, envir = environment())

##transform values
values.transformed &lt;- convert_CW2pLMi(ExampleData.CW_OSL_Curve)

##plot
plot(values.transformed$x, values.transformed$y.t, log = "x")

##(2) - produce Fig. 4 from Bos &amp; Wallinga (2012)
##load data
data(ExampleData.CW_OSL_Curve, envir = environment())
values &lt;- CW_Curve.BosWallinga2012

##open plot area
plot(NA, NA,
     xlim = c(0.001,10),
     ylim = c(0,8000),
     ylab = "pseudo OSL (cts/0.01 s)",
     xlab = "t [s]",
     log = "x",
     main = "Fig. 4 - Bos &amp; Wallinga (2012)")


values.t &lt;- convert_CW2pLMi(values, P = 1/20)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col = "red", lwd = 1.3)
text(0.03,4500,"LM", col = "red", cex = .8)

values.t &lt;- convert_CW2pHMi(values, delta = 40)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col = "black", lwd = 1.3)
text(0.005,3000,"HM", cex =.8)

values.t &lt;- convert_CW2pPMi(values, P = 1/10)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col = "blue", lwd = 1.3)
text(0.5,6500,"PM", col = "blue", cex = .8)

</code></pre>

<hr>
<h2 id='convert_CW2pPMi'>Transform a CW-OSL curve into a pPM-OSL curve via interpolation under
parabolic modulation conditions</h2><span id='topic+convert_CW2pPMi'></span><span id='topic+CW2pPMi'></span>

<h3>Description</h3>

<p>Transforms a conventionally measured continuous-wave (CW) OSL-curve into a
pseudo parabolic modulated (pPM) curve under parabolic modulation conditions
using the interpolation procedure described by Bos &amp; Wallinga (2012).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_CW2pPMi(values, P)

CW2pPMi(values, P)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_CW2pPMi_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <code>data.frame</code> with measured curve data of type
stimulation time (t) (<code>values[,1]</code>) and measured counts (cts) (<code>values[,2]</code>)</p>
</td></tr>
<tr><td><code id="convert_CW2pPMi_+3A_p">P</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
stimulation period in seconds. If no value is given, the optimal value is
estimated automatically (see details). Greater values of P produce more
points in the rising tail of the curve.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The complete procedure of the transformation is given in Bos &amp; Wallinga
(2012). The input <code>data.frame</code> consists of two columns: time (t) and
count values (CW(t))
</p>
<p><strong>Nomenclature</strong>
</p>

<ul>
<li><p> P = stimulation time (s)
</p>
</li>
<li><p> 1/P = stimulation rate (1/s)
</p>
</li></ul>

<p><strong>Internal transformation steps</strong>
</p>
<p>(1)
log(CW-OSL) values
</p>
<p>(2)
Calculate t' which is the transformed time:
</p>
<p style="text-align: center;"><code class="reqn">t' = (1/3)*(1/P^2)t^3</code>
</p>

<p>(3)
Interpolate CW(t'), i.e. use the log(CW(t)) to obtain the count values for
the transformed time (t'). Values beyond <code>min(t)</code> and <code>max(t)</code>
produce <code>NA</code> values.
</p>
<p>(4)
Select all values for t' &lt; <code>min(t)</code>, i.e. values beyond the time resolution
of t. Select the first two values of the transformed data set which contain
no <code>NA</code> values and use these values for a linear fit using <a href="stats.html#topic+lm">lm</a>.
</p>
<p>(5)
Extrapolate values for t' &lt; <code>min(t)</code> based on the previously obtained
fit parameters. The extrapolation is limited to two values. Other values at
the beginning of the transformed curve are set to 0.
</p>
<p>(6)
Transform values using
</p>
<p style="text-align: center;"><code class="reqn">pLM(t) = t^2/P^2*CW(t')</code>
</p>

<p>(7)
Combine all values and truncate all values for t' &gt; <code>max(t)</code>
</p>
<p><strong>NOTE:</strong>
The number of values for t' &lt; <code>min(t)</code> depends on the stimulation
period <code>P</code>. To avoid the production of too many artificial data at the
raising tail of the determined pPM curve, it is recommended to use the
automatic estimation routine for <code>P</code>, i.e. provide no value for
<code>P</code>.
</p>


<h3>Value</h3>

<p>The function returns the same data type as the input data type with
the transformed curve values.
</p>
<p><code>RLum.Data.Curve</code>
</p>

<table>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$CW2pPMi.x.t&#8288;</code> </td><td style="text-align: left;">: transformed time values </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$CW2pPMi.method&#8288;</code> </td><td style="text-align: left;">: used method for the production of the new data points
</td>
</tr>

</table>

<p><code>data.frame</code>
</p>

<table>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$x&#8288;</code> </td><td style="text-align: left;">: time</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$y.t&#8288;</code> </td><td style="text-align: left;">: transformed count values</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$x.t&#8288;</code> </td><td style="text-align: left;">: transformed time values </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;$method&#8288;</code> </td><td style="text-align: left;">: used method for the production of the new data points
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.2.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_CW2pPMi(): Transform a CW-OSL curve into a pPM-OSL curve via interpolation under parabolic modulation conditions. Function version 0.2.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>According to Bos &amp; Wallinga (2012), the number of extrapolated points
should be limited to avoid artificial intensity data. If <code>P</code> is
provided manually, not more than two points are extrapolated.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
</p>
<p>Based on comments and suggestions from:<br />
Adrie J.J. Bos, Delft University of Technology, The Netherlands
, RLum Developer Team</p>


<h3>References</h3>

<p>Bos, A.J.J. &amp; Wallinga, J., 2012. How to visualize quartz OSL
signal components. Radiation Measurements, 47, 752-758.
</p>
<p><strong>Further Reading</strong>
</p>
<p>Bulur, E., 1996. An Alternative Technique For
Optically Stimulated Luminescence (OSL) Experiment. Radiation Measurements,
26, 701-709.
</p>
<p>Bulur, E., 2000. A simple transformation for converting CW-OSL curves to
LM-OSL curves. Radiation Measurements, 32, 141-145.
</p>


<h3>See Also</h3>

<p><a href="#topic+convert_CW2pLM">convert_CW2pLM</a>, <a href="#topic+convert_CW2pLMi">convert_CW2pLMi</a>, <a href="#topic+convert_CW2pHMi">convert_CW2pHMi</a>,
<a href="#topic+fit_LMCurve">fit_LMCurve</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

##(1)
##load CW-OSL curve data
data(ExampleData.CW_OSL_Curve, envir = environment())

##transform values
values.transformed &lt;- convert_CW2pPMi(ExampleData.CW_OSL_Curve)

##plot
plot(values.transformed$x,values.transformed$y.t, log = "x")

##(2) - produce Fig. 4 from Bos &amp; Wallinga (2012)

##load data
data(ExampleData.CW_OSL_Curve, envir = environment())
values &lt;- CW_Curve.BosWallinga2012

##open plot area
plot(NA, NA,
     xlim = c(0.001,10),
     ylim = c(0,8000),
     ylab = "pseudo OSL (cts/0.01 s)",
     xlab = "t [s]",
     log = "x",
     main = "Fig. 4 - Bos &amp; Wallinga (2012)")

values.t &lt;- convert_CW2pLMi(values, P = 1/20)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col = "red",lwd = 1.3)
text(0.03,4500,"LM", col = "red", cex = .8)

values.t &lt;- convert_CW2pHMi(values, delta = 40)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col = "black", lwd = 1.3)
text(0.005,3000,"HM", cex = .8)

values.t &lt;- convert_CW2pPMi(values, P = 1/10)
lines(values[1:length(values.t[, 1]), 1], values.t[, 2],
      col = "blue", lwd = 1.3)
text(0.5,6500,"PM", col = "blue", cex = .8)

</code></pre>

<hr>
<h2 id='convert_Daybreak2CSV'>Export measurement data produced by a Daybreak luminescence reader to CSV-files</h2><span id='topic+convert_Daybreak2CSV'></span>

<h3>Description</h3>

<p>This function is a wrapper function around the functions <a href="#topic+read_Daybreak2R">read_Daybreak2R</a> and
<a href="#topic+write_RLum2CSV">write_RLum2CSV</a> and it imports a Daybreak-file (TXT-file, DAT-file)
and directly exports its content to CSV-files.  If nothing is set for the
argument <code>path</code> (<a href="#topic+write_RLum2CSV">write_RLum2CSV</a>) the input folder will become the output folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_Daybreak2CSV(file, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_Daybreak2CSV_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the Daybreak-file (TXT-file, DAT-file) to be converted to CSV-files</p>
</td></tr>
<tr><td><code id="convert_Daybreak2CSV_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+read_Daybreak2R">read_Daybreak2R</a> and <a href="#topic+write_RLum2CSV">write_RLum2CSV</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns either a CSV-file (or many of them) or for the option <code>export = FALSE</code>
a list comprising objects of type <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_Daybreak2CSV(): Export measurement data produced by a Daybreak luminescence reader to CSV-files. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>,
<a href="utils.html#topic+write.table">utils::write.table</a>, <a href="#topic+write_RLum2CSV">write_RLum2CSV</a>, <a href="#topic+read_Daybreak2R">read_Daybreak2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
##select your BIN-file
file &lt;- file.choose()

##convert
convert_Daybreak2CSV(file)


## End(Not run)

</code></pre>

<hr>
<h2 id='convert_PSL2CSV'>Export PSL-file(s) to CSV-files</h2><span id='topic+convert_PSL2CSV'></span>

<h3>Description</h3>

<p>This function is a wrapper function around the functions <a href="#topic+read_PSL2R">read_PSL2R</a> and
<a href="#topic+write_RLum2CSV">write_RLum2CSV</a> and it imports an PSL-file (SUERC portable OSL reader file format)
and directly exports its content to CSV-files.
If nothing is set for the argument <code>path</code> (<a href="#topic+write_RLum2CSV">write_RLum2CSV</a>) the input folder will
become the output folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_PSL2CSV(file, extract_raw_data = FALSE, single_table = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_PSL2CSV_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the PSL-file to be converted to CSV-files</p>
</td></tr>
<tr><td><code id="convert_PSL2CSV_+3A_extract_raw_data">extract_raw_data</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable raw data
extraction. The PSL files imported into R contain an element <code style="white-space: pre;">&#8288;$raw_data&#8288;</code>, which
provides a few more information (e.g., count errors), sometimes it makes
sense to use this data of the more compact standard values created by <a href="#topic+read_PSL2R">read_PSL2R</a></p>
</td></tr>
<tr><td><code id="convert_PSL2CSV_+3A_single_table">single_table</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the creation
of single table with n rows and n columns, instead of separate <a href="base.html#topic+data.frame">data.frame</a>
objects. Each curve will be represented by two columns for time and counts</p>
</td></tr>
<tr><td><code id="convert_PSL2CSV_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+read_PSL2R">read_PSL2R</a> and <a href="#topic+write_RLum2CSV">write_RLum2CSV</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns either a CSV-file (or many of them) or for the option
<code>export = FALSE</code> a list comprising objects of type <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
</p>


<h3>Function version</h3>

<p>0.1.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_PSL2CSV(): Export PSL-file(s) to CSV-files. Function version 0.1.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>,
<a href="utils.html#topic+write.table">utils::write.table</a>, <a href="#topic+write_RLum2CSV">write_RLum2CSV</a>, <a href="#topic+read_PSL2R">read_PSL2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## export into single data.frame
file &lt;- system.file("extdata/DorNie_0016.psl", package="Luminescence")
convert_PSL2CSV(file, export = FALSE, single_table = TRUE)


## Not run: 
##select your BIN-file
file &lt;- file.choose()

##convert
convert_PSL2CSV(file)


## End(Not run)

</code></pre>

<hr>
<h2 id='convert_PSL2Risoe.BINfileData'>Convert portable OSL data to a Risoe.BINfileData object</h2><span id='topic+convert_PSL2Risoe.BINfileData'></span>

<h3>Description</h3>

<p>Converts an <code>RLum.Analysis</code> object produced by the function <code>read_PSL2R()</code> to
a <code>Risoe.BINfileData</code> object <strong>(BETA)</strong>.
</p>
<p>This function converts an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object that was produced
by the <a href="#topic+read_PSL2R">read_PSL2R</a> function to a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>.
The <code>Risoe.BINfileData</code> can be used to write a Risoe BIN file via
<a href="#topic+write_R2BIN">write_R2BIN</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_PSL2Risoe.BINfileData(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_PSL2Risoe.BINfileData_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
<code>RLum.Analysis</code> object produced by <a href="#topic+read_PSL2R">read_PSL2R</a></p>
</td></tr>
<tr><td><code id="convert_PSL2Risoe.BINfileData_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S4 <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object that can be used to write a
BIN file using <a href="#topic+write_R2BIN">write_R2BIN</a>.
</p>


<h3>Function version</h3>

<p>0.0.1
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. convert_PSL2Risoe.BINfileData(): Convert portable OSL data to a Risoe.BINfileData object. Function version 0.0.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# (1) load and plot example data set
data("ExampleData.portableOSL", envir = environment())
plot_RLum(ExampleData.portableOSL)

# (2) merge all RLum.Analysis objects into one
merged &lt;- merge_RLum(ExampleData.portableOSL)
merged

# (3) convert to RisoeBINfile object
bin &lt;- convert_PSL2Risoe.BINfileData(merged)
bin

# (4) write Risoe BIN file
## Not run: 
write_R2BIN(bin, "~/portableOSL.binx")

## End(Not run)

</code></pre>

<hr>
<h2 id='convert_RLum2Risoe.BINfileData'>Converts RLum.Analysis and RLum.Data.Curve objects to RLum2Risoe.BINfileData objects</h2><span id='topic+convert_RLum2Risoe.BINfileData'></span>

<h3>Description</h3>

<p>The functions converts <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> and
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects (or a <a href="base.html#topic+list">list</a> of such objects) to
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects. The function intends to provide a
minimum of compatibility between both formats. The created
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object can be later exported to a BIN-file using
function <a href="#topic+write_R2BIN">write_R2BIN</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_RLum2Risoe.BINfileData(object, keep.position.number = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_RLum2Risoe.BINfileData_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> (<strong>required</strong>): input object to
be converted</p>
</td></tr>
<tr><td><code id="convert_RLum2Risoe.BINfileData_+3A_keep.position.number">keep.position.number</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): keeps the original
position number or re-calculate the numbers to avoid doubling</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object.
</p>


<h3>Function version</h3>

<p>0.1.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_RLum2Risoe.BINfileData(): Converts RLum.Analysis and RLum.Data.Curve objects to RLum2Risoe.BINfileData objects. Function version 0.1.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The conversion can be never perfect. The <code>RLum</code> objects may contain information which are
not part of the <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> definition.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+write_R2BIN">write_R2BIN</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##simple conversion using the example dataset
data(ExampleData.RLum.Analysis, envir = environment())
convert_RLum2Risoe.BINfileData(IRSAR.RF.Data)

</code></pre>

<hr>
<h2 id='convert_Second2Gray'>Converting equivalent dose values from seconds (s) to Gray (Gy)</h2><span id='topic+convert_Second2Gray'></span>

<h3>Description</h3>

<p>Conversion of absorbed radiation dose in seconds (s) to the SI unit Gray
(Gy) including error propagation. Normally used for equivalent dose data.
</p>
<p>Calculation of De values from seconds (s) to Gray (Gy)
</p>
<p style="text-align: center;"><code class="reqn">De [Gy] = De [s] * Dose Rate [Gy/s])</code>
</p>

<p>Provided calculation error propagation methods for error calculation
(with <code>'se'</code> as the standard error and <code>'DR'</code> of the dose rate of the beta-source):
</p>
<p><strong>(1) <code>omit</code></strong> (default)
</p>
<p style="text-align: center;"><code class="reqn">se(De) [Gy] = se(De) [s] * DR [Gy/s]</code>
</p>

<p>In this case the standard error of the dose rate of the beta-source is
treated as systematic (i.e. non-random), it error propagation is omitted.
However, the error must be considered during calculation of the final age.
(cf. Aitken, 1985, pp. 242). This approach can be seen as method (2) (gaussian)
for the case the (random) standard error of the beta-source calibration is
0. Which particular method is requested depends on the situation and cannot
be prescriptive.
</p>
<p><strong>(2) <code>gaussian</code></strong> error propagation
</p>
<p style="text-align: center;"><code class="reqn">se(De) [Gy] = \sqrt((DR [Gy/s] * se(De) [s])^2 + (De [s] * se(DR) [Gy/s])^2)</code>
</p>

<p>Applicable under the assumption that errors of <code>De</code> and <code>se</code> are uncorrelated.
</p>
<p><strong>(3) <code>absolute</code></strong> error propagation
</p>
<p style="text-align: center;"><code class="reqn">se(De) [Gy]= abs(DR [Gy/s] * se(De) [s]) + abs(De [s] * se(DR) [Gy/s])</code>
</p>

<p>Applicable under the assumption that errors of <code>De</code> and <code>se</code> are correlated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_Second2Gray(data, dose.rate, error.propagation = "omit")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_Second2Gray_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
input values, structure: data (<code>values[,1]</code>) and data error (<code>values [,2]</code>)
are required</p>
</td></tr>
<tr><td><code id="convert_Second2Gray_+3A_dose.rate">dose.rate</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="base.html#topic+data.frame">data.frame</a> or <a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
<code>RLum.Results</code> needs to be originated from the function <a href="#topic+calc_SourceDoseRate">calc_SourceDoseRate</a>,
for <code>vector</code> dose rate in Gy/s and dose rate error in Gy/s</p>
</td></tr>
<tr><td><code id="convert_Second2Gray_+3A_error.propagation">error.propagation</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
error propagation method used for error calculation (<code>omit</code>, <code>gaussian</code> or
<code>absolute</code>), see details for further information</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <a href="base.html#topic+data.frame">data.frame</a> with converted values.
</p>


<h3>Function version</h3>

<p>0.6.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., Fuchs, M.C., 2025. convert_Second2Gray(): Converting equivalent dose values from seconds (s) to Gray (Gy). Function version 0.6.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>If no or a wrong error propagation method is given, the execution of the function is
stopped. Furthermore, if a <code>data.frame</code> is provided for the dose rate values is has to
be of the same length as the data frame provided with the argument <code>data</code>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany)<br />
Margret C. Fuchs, HZDR, Helmholtz-Institute Freiberg for Resource Technology (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Aitken, M.J., 1985. Thermoluminescence dating. Academic Press.
</p>


<h3>See Also</h3>

<p><a href="#topic+calc_SourceDoseRate">calc_SourceDoseRate</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(A) for known source dose rate at date of measurement
## - load De data from the example data help file
data(ExampleData.DeValues, envir = environment())
## - convert De(s) to De(Gy)
convert_Second2Gray(ExampleData.DeValues$BT998, c(0.0438,0.0019))


##(B) for source dose rate calibration data
## - calculate source dose rate first
dose.rate &lt;-  calc_SourceDoseRate(measurement.date = "2012-01-27",
                                  calib.date = "2014-12-19",
                                  calib.dose.rate = 0.0438,
                                  calib.error = 0.0019)
# read example data
data(ExampleData.DeValues, envir = environment())

# apply dose.rate to convert De(s) to De(Gy)
convert_Second2Gray(ExampleData.DeValues$BT998, dose.rate)

</code></pre>

<hr>
<h2 id='convert_SG2MG'>Converts Single-Grain Data to Multiple-Grain Data</h2><span id='topic+convert_SG2MG'></span>

<h3>Description</h3>

<p>Conversion of single-grain data to multiple-grain data by adding signals
from grains belonging to one disc (unique pairs of position, set and run).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_SG2MG(object, write_file = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_SG2MG_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> <a href="base.html#topic+character">character</a> (<strong>required</strong>): <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
object or BIN/BINX-file name</p>
</td></tr>
<tr><td><code id="convert_SG2MG_+3A_write_file">write_file</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): if the input was a path to a file, the
output can be written to a file if <code>TRUE</code>. The multiple grain file will be written into the
same folder and with extension <code>-SG</code> to the file name.</p>
</td></tr>
<tr><td><code id="convert_SG2MG_+3A_...">...</code></td>
<td>
<p>further arguments passed down to <a href="#topic+read_BIN2R">read_BIN2R</a> if input is file path</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object and if <code>write_file = TRUE</code> and the input
was a file path, a file is written to origin folder.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Mercier, N., 2025. convert_SG2MG(): Converts Single-Grain Data to Multiple-Grain Data. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Norbert Mercier, IRAMAT-CRP2A, UMR 5060, CNRS-Universit√© Bordeaux Montaigne (France)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+write_R2BIN">write_R2BIN</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## simple run
## (please not that the example is not using SG data)
data(ExampleData.BINfileData, envir = environment())
convert_SG2MG(CWOSL.SAR.Data)

</code></pre>

<hr>
<h2 id='convert_Wavelength2Energy'>Emission Spectra Conversion from Wavelength to Energy Scales (Jacobian Conversion)</h2><span id='topic+convert_Wavelength2Energy'></span>

<h3>Description</h3>

<p>The function provides a convenient and fast way to convert emission spectra wavelength
to energy scales. The function works on <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a> and
a <a href="base.html#topic+list">list</a> of such objects. The function was written to smooth the workflow while analysing
emission spectra data. This is in particular useful if you want to further treat your data
and apply, e.g., a signal deconvolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_Wavelength2Energy(object, digits = 3L, order = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_Wavelength2Energy_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="base.html#topic+data.frame">data.frame</a>, <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>): input object to be converted.
If the input is not an <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, the first column is always treated as the wavelength
column. The function supports a list of allowed input objects.</p>
</td></tr>
<tr><td><code id="convert_Wavelength2Energy_+3A_digits">digits</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): set the number of digits on the returned energy axis</p>
</td></tr>
<tr><td><code id="convert_Wavelength2Energy_+3A_order">order</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable sorting of the
values in ascending energy order. After the conversion, the longest
wavelength has the lowest energy value and the shortest wavelength the
highest. While this is correct, some R functions expect increasing x-values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The intensity of the spectrum is re-calculated using the following approach to recalculate
wavelength and corresponding intensity values
(e.g., Appendix 4 in Blasse and Grabmaier, 1994; Mooney and Kambhampati, 2013):
</p>
<p style="text-align: center;"><code class="reqn">\phi_{E} = \phi_{\lambda} * \lambda^2 / (hc)</code>
</p>

<p>with
<code class="reqn">\phi_{E}</code> the intensity per interval of energy <code class="reqn">E</code> (1/eV),
<code class="reqn">\phi_{\lambda}</code> the intensity per interval of wavelength <code class="reqn">\lambda</code>
(1/nm) and
<code class="reqn">h</code> (eV * s) the Planck constant and <code class="reqn">c</code> (nm/s) the velocity of light.
</p>
<p>For transforming the wavelength axis (x-values) the equation as follow is used
</p>
<p style="text-align: center;"><code class="reqn">E = hc/\lambda</code>
</p>



<h3>Value</h3>

<p>The same object class as provided as input is returned.
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_Wavelength2Energy(): Emission Spectra Conversion from Wavelength to Energy Scales (Jacobian Conversion). Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This conversion works solely for emission spectra. In case of absorption spectra only
the x-axis has to be converted.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Blasse, G., Grabmaier, B.C., 1994. Luminescent Materials. Springer.
</p>
<p>Mooney, J., Kambhampati, P., 2013. Get the Basics Right: Jacobian Conversion of Wavelength and
Energy Scales for Quantitative Analysis of Emission Spectra. J. Phys. Chem. Lett. 4, 3316‚Äì3318.
<a href="https://doi.org/10.1021/jz401508t">doi:10.1021/jz401508t</a>
</p>
<p>Mooney, J., Kambhampati, P., 2013. Correction to ‚ÄúGet the Basics Right: Jacobian Conversion of
Wavelength and Energy Scales for Quantitative Analysis of Emission Spectra.‚Äù J. Phys. Chem. Lett. 4,
3316‚Äì3318. <a href="https://doi.org/10.1021/jz401508t">doi:10.1021/jz401508t</a>
</p>
<p><strong>Further reading</strong>
</p>
<p>Angulo, G., Grampp, G., Rosspeintner, A., 2006. Recalling the appropriate representation of
electronic spectra. Spectrochimica Acta Part A: Molecular and Biomolecular Spectroscopy 65,
727‚Äì731. <a href="https://doi.org/10.1016/j.saa.2006.01.007">doi:10.1016/j.saa.2006.01.007</a>
</p>
<p>Wang, Y., Townsend, P.D., 2013. Potential problems in collection and data processing of
luminescence signals. Journal of Luminescence 142, 202‚Äì211. <a href="https://doi.org/10.1016/j.jlumin.2013.03.052">doi:10.1016/j.jlumin.2013.03.052</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+plot_RLum">plot_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##=====================##
##(1) Literature example after Mooney et al. (2013)
##(1.1) create matrix
m &lt;- matrix(
  data = c(seq(400, 800, 50), rep(1, 9)), ncol = 2)

##(1.2) set plot function to reproduce the
##literature figure
p &lt;- function(m) {
 plot(x = m[, 1], y = m[, 2])
 polygon(
 x = c(m[, 1], rev(m[, 1])),
 y = c(m[, 2], rep(0, nrow(m))))
 for (i in 1:nrow(m)) {
  lines(x = rep(m[i, 1], 2), y = c(0, m[i, 2]))
 }
}

##(1.3) plot curves
par(mfrow = c(1,2))
p(m)
p(convert_Wavelength2Energy(m))

##=====================##
##(2) Another example using density curves
##create dataset
xy &lt;- density(
 c(rnorm(n = 100, mean = 500, sd = 20),
 rnorm(n = 100, mean = 800, sd = 20)))
xy &lt;- data.frame(xy$x, xy$y)

##plot
par(mfrow = c(1,2))
plot(
 xy,
 type = "l",
 xlim = c(150, 1000),
 xlab = "Wavelength [nm]",
 ylab = "Luminescence [a.u.]"
)
plot(
 convert_Wavelength2Energy(xy),
 xy$y,
 type = "l",
 xlim = c(1.23, 8.3),
 xlab = "Energy [eV]",
 ylab = "Luminescence [a.u.]"
)

</code></pre>

<hr>
<h2 id='convert_XSYG2CSV'>Export XSYG-file(s) to CSV-files</h2><span id='topic+convert_XSYG2CSV'></span>

<h3>Description</h3>

<p>This function is a wrapper function around the functions <a href="#topic+read_XSYG2R">read_XSYG2R</a> and
<a href="#topic+write_RLum2CSV">write_RLum2CSV</a> and it imports an XSYG-file and directly exports its content
to CSV-files. If nothing is set for the argument <code>path</code> (<a href="#topic+write_RLum2CSV">write_RLum2CSV</a>)
the input folder will become the output folder.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert_XSYG2CSV(file, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert_XSYG2CSV_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the XSYG-file to be converted to CSV-files</p>
</td></tr>
<tr><td><code id="convert_XSYG2CSV_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+read_XSYG2R">read_XSYG2R</a> and <a href="#topic+write_RLum2CSV">write_RLum2CSV</a></p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns either a CSV-file (or many of them) or for the option <code>export = FALSE</code>
a list comprising objects of type <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. convert_XSYG2CSV(): Export XSYG-file(s) to CSV-files. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>,
<a href="utils.html#topic+write.table">utils::write.table</a>, <a href="#topic+write_RLum2CSV">write_RLum2CSV</a>, <a href="#topic+read_XSYG2R">read_XSYG2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##transform XSYG-file values to a list
data(ExampleData.XSYG, envir = environment())
convert_XSYG2CSV(OSL.SARMeasurement$Sequence.Object[1:10], export = FALSE)

## Not run: 
##select your BIN-file
file &lt;- file.choose()

##convert
convert_XSYG2CSV(file)


## End(Not run)

</code></pre>

<hr>
<h2 id='ExampleData.Al2O3C'>Example Al2O3:C Measurement Data</h2><span id='topic+ExampleData.Al2O3C'></span><span id='topic+data_CrossTalk'></span><span id='topic+data_ITC'></span>

<h3>Description</h3>

<p>Measurement data obtained from measuring Al2O3:C chips at the IRAMAT-CRP2A, Universit√© Bordeaux
Montaigne in 2017 on a Freiberg Instruments lexsyg SMART reader.
The example data used in particular to allow test of the functions
developed in framework of the work by Kreutzer et al., 2018.
</p>


<h3>Format</h3>

<p>Two datasets comprising <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> data imported using the function <a href="#topic+read_XSYG2R">read_XSYG2R</a>
</p>

<p><code>data_ITC</code>: Measurement data to determine the irradiation time correction, the data can
be analysed with the function <a href="#topic+analyse_Al2O3C_ITC">analyse_Al2O3C_ITC</a>
</p>
<p><code>data_CrossTalk</code>: Measurement data obtained while estimating the irradiation cross-talk of the
reader used for the experiments. The data can be analysed either with the function
<a href="#topic+analyse_Al2O3C_CrossTalk">analyse_Al2O3C_CrossTalk</a> or <a href="#topic+analyse_Al2O3C_Measurement">analyse_Al2O3C_Measurement</a>

</p>


<h3>Note</h3>

<p>From both datasets unneeded curves have been removed and
the number of aliquots have been reduced to a required minimum to keep the file size small, but
still being able to run the corresponding functions.
</p>


<h3>References</h3>

<p>Kreutzer, S., Martin, L., Gu√©rin, G., Tribolo, C., Selva, P., Mercier, N., 2018.
Environmental Dose Rate Determination Using a Passive Dosimeter: Techniques and Workflow for alpha-Al2O3:C Chips.
Geochronometria 45, 56‚Äì67.
</p>


<h3>See Also</h3>

<p><a href="#topic+analyse_Al2O3C_ITC">analyse_Al2O3C_ITC</a>, <a href="#topic+analyse_Al2O3C_CrossTalk">analyse_Al2O3C_CrossTalk</a>, <a href="#topic+analyse_Al2O3C_Measurement">analyse_Al2O3C_Measurement</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) curves
data(ExampleData.Al2O3C, envir = environment())
plot_RLum(data_ITC[1:2])

</code></pre>

<hr>
<h2 id='ExampleData.BINfileData'>Example data from a SAR OSL and SAR TL measurement for the package
Luminescence</h2><span id='topic+ExampleData.BINfileData'></span><span id='topic+CWOSL.SAR.Data'></span><span id='topic+TL.SAR.Data'></span>

<h3>Description</h3>

<p>Example data from a SAR OSL and TL measurement for package Luminescence
directly extracted from a Risoe BIN-file and provided in an object of type
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>


<h3>Format</h3>

<p><code>CWOSL.SAR.Data</code>: SAR OSL measurement data
</p>
<p><code>TL.SAR.Data</code>: SAR TL measurement data
</p>
<p>Each class object contains two slots: (a) <code>METADATA</code> is a <a href="base.html#topic+data.frame">data.frame</a> with
all metadata stored in the BIN file of the measurements and (b) <code>DATA</code>
contains a list of vectors of the measured data (usually count values).
</p>


<h3>Version</h3>

<p>0.1
</p>


<h3>Note</h3>

<p>Please note that this example data cannot be exported to a BIN-file using the function
<code>writeR2BIN</code> as it was generated and implemented in the package long time ago. In the meantime
the BIN-file format changed.
</p>


<h3>Source</h3>

<p><strong>CWOSL.SAR.Data</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Bayreuth </td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> BT607 </td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Saxony/Germany </td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Middle grain quartz measured on aluminium cups on a Ris√∏ TL/OSL DA-15 reader</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> unpublished
</td>
</tr>

</table>

<p><strong>TL.SAR.Data</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory of Cologne</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> LP1_5</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Spain</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Flint </td>
</tr>
<tr>
 <td style="text-align: left;">
Setup: </td><td style="text-align: left;"> Risoe TL/OSL DA-20 reader (Filter: Semrock Brightline, HC475/50, N2, unpolished steel discs) </td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> unpublished </td>
</tr>
<tr>
 <td style="text-align: left;">
Remarks: </td><td style="text-align: left;"> dataset limited to one position
</td>
</tr>

</table>



<h3>References</h3>

<p><strong>CWOSL.SAR.Data</strong>: unpublished data
</p>
<p><strong>TL.SAR.Data</strong>: unpublished data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## show first 5 elements of the METADATA and DATA elements in the terminal
data(ExampleData.BINfileData, envir = environment())
CWOSL.SAR.Data@METADATA[1:5,]
CWOSL.SAR.Data@DATA[1:5]

</code></pre>

<hr>
<h2 id='ExampleData.CobbleData'>Example data for calc_CobbleDoseRate()</h2><span id='topic+ExampleData.CobbleData'></span>

<h3>Description</h3>

<p>An example data set for the function <a href="#topic+calc_CobbleDoseRate">calc_CobbleDoseRate</a> containing layer
specific information for the cobble to be used in the function.
</p>


<h3>Format</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code>. Please see <a href="#topic+calc_CobbleDoseRate">calc_CobbleDoseRate</a> for detailed information
on the structure of the <a href="base.html#topic+data.frame">data.frame</a>.
</p>


<h3>Version</h3>

<p>0.1.0
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load data
data("ExampleData.CobbleData", envir = environment())

</code></pre>

<hr>
<h2 id='ExampleData.CW_OSL_Curve'>Example CW-OSL curve data for the package Luminescence</h2><span id='topic+ExampleData.CW_OSL_Curve'></span><span id='topic+CW_Curve.BosWallinga2012'></span>

<h3>Description</h3>

<p><code>data.frame</code> containing CW-OSL curve data (time, counts)
</p>


<h3>Format</h3>

<p>Data frame with 1000 observations on the following 2 variables:
</p>

<ul>
<li> <p><code>list("x")</code>: a numeric vector, time
</p>
</li>
<li> <p><code>list("y")</code>: a numeric vector, counts
</p>
</li></ul>



<h3>Source</h3>

<p><strong>ExampleData.CW_OSL_Curve</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Bayreuth</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> BT607</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Saxony/Germany</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Middle grain quartz measured on aluminium cups on a Ris√∏ TL/OSL DA-15 reader.</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> unpublished data </td>
</tr>

</table>

<p><strong>CW_Curve.BosWallinga2012</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Netherlands Centre for Luminescence Dating (NCL)</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> NCL-2108077</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Guadalentin Basin, Spain</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Coarse grain quartz</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> Bos &amp; Wallinga (2012) and Baartman et al. (2011)
</td>
</tr>

</table>



<h3>References</h3>

<p>Baartman, J.E.M., Veldkamp, A., Schoorl, J.M., Wallinga, J.,
Cammeraat, L.H., 2011. Unravelling Late Pleistocene and Holocene landscape
dynamics: The Upper Guadalentin Basin, SE Spain. Geomorphology, 125,
172-185.
</p>
<p>Bos, A.J.J. &amp; Wallinga, J., 2012. How to visualize quartz OSL signal
components. Radiation Measurements, 47, 752-758.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(ExampleData.CW_OSL_Curve, envir = environment())
plot(ExampleData.CW_OSL_Curve)

</code></pre>

<hr>
<h2 id='ExampleData.DeValues'>Example De data sets for the package Luminescence</h2><span id='topic+ExampleData.DeValues'></span>

<h3>Description</h3>

<p>Equivalent dose (De) values measured for a fine grain quartz sample from a
loess section in Rottewitz (Saxony/Germany) and for a coarse grain quartz
sample from a fluvial deposit in the rock shelter of Cueva Anton
(Murcia/Spain).
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+list">list</a> with two elements, each containing a two column <a href="base.html#topic+data.frame">data.frame</a>:
</p>

<p><code style="white-space: pre;">&#8288;$BT998&#8288;</code>: De and De error values for a fine grain quartz
sample from a loess section in Rottewitz.
</p>
<p><code style="white-space: pre;">&#8288;$CA1&#8288;</code>: Single grain De
and De error values for a coarse grain quartz sample from a fluvial deposit
in the rock shelter of Cueva Anton

</p>


<h3>References</h3>

<p><strong>BT998</strong>
</p>
<p>Unpublished data
</p>
<p><strong>CA1</strong>
</p>
<p>Burow, C., Kehl, M., Hilgers, A., Weniger, G.-C., Angelucci, D., Villaverde,
V., Zapata, J. and Zilhao, J.  (2015). Luminescence dating of fluvial
deposits in the rock shelter of Cueva Anton, Spain. Geochronometria 52, 107-125.
</p>
<p><strong>BT998</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Bayreuth</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> BT998</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Rottewitz (Saxony/Germany)</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Fine grain quartz measured on aluminium discs on a Ris√∏ TL/OSL DA-15 reader</td>
</tr>
<tr>
 <td style="text-align: left;">
Units: </td><td style="text-align: left;"> Values are given in seconds </td>
</tr>
<tr>
 <td style="text-align: left;">
Dose Rate: </td><td style="text-align: left;"> Dose rate of the beta-source at measurement ca. 0.0438 Gy/s +/- 0.0019 Gy/s</td>
</tr>
<tr>
 <td style="text-align: left;">
Measurement Date: </td><td style="text-align: left;"> 2012-01-27
</td>
</tr>

</table>

<p><strong>CA1</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Cologne Luminescence Laboratory (CLL)</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> C-L2941</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Cueva Anton (Murcia/Spain)</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Coarse grain quartz (200-250 microns) measured on single grain discs on a Risoe TL/OSL DA-20 reader</td>
</tr>
<tr>
 <td style="text-align: left;">
Units: </td><td style="text-align: left;"> Values are given in Gray </td>
</tr>
<tr>
 <td style="text-align: left;">
Measurement Date: </td><td style="text-align: left;"> 2012
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
##(1) plot values as histogram
data(ExampleData.DeValues, envir = environment())
plot_Histogram(ExampleData.DeValues$BT998, xlab = "De [s]")

##(2) plot values as histogram (with second to gray conversion)
data(ExampleData.DeValues, envir = environment())

De.values &lt;- convert_Second2Gray(ExampleData.DeValues$BT998,
                         dose.rate = c(0.0438, 0.0019))


plot_Histogram(De.values, xlab = "De [Gy]")

</code></pre>

<hr>
<h2 id='ExampleData.Fading'>Example data for feldspar fading measurements</h2><span id='topic+ExampleData.Fading'></span>

<h3>Description</h3>

<p>Example data set for fading measurements of the IR50, IR100, IR150 and
IR225 feldspar signals of sample UNIL/NB123. It further contains regular equivalent dose
measurement data of the same sample, which can be used to apply a
fading correction to.
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+list">list</a> with two elements, each containing a further <a href="base.html#topic+list">list</a> of
<a href="base.html#topic+data.frame">data.frame</a>s containing the data on the fading and equivalent dose measurements:
</p>

<p><code style="white-space: pre;">&#8288;$fading.data&#8288;</code>: A named <a href="base.html#topic+list">list</a> of <a href="base.html#topic+data.frame">data.frame</a>s,
each having three named columns (<code style="white-space: pre;">&#8288;LxTx, LxTx.error, timeSinceIrradiation&#8288;</code>).<br />
<code>..$IR50</code>: Fading data of the IR50 signal.<br />
<code>..$IR100</code>: Fading data of the IR100 signal.<br />
<code>..$IR150</code>: Fading data of the IR150 signal.<br />
<code>..$IR225</code>: Fading data of the IR225 signal.<br />
</p>
<p><code style="white-space: pre;">&#8288;$equivalentDose.data&#8288;</code>: A named of <a href="base.html#topic+data.frame">data.frame</a>s,
each having three named columns (<code style="white-space: pre;">&#8288;dose, LxTx, LxTx.error&#8288;</code>).<br />
<code>..$IR50</code>: Equivalent dose measurement data of the IR50 signal.<br />
<code>..$IR100</code>: Equivalent dose measurement data of the IR100 signal.<br />
<code>..$IR150</code>: Equivalent dose measurement data of the IR150 signal.<br />
<code>..$IR225</code>: Equivalent dose measurement data of the IR225 signal.<br />

</p>


<h3>Source</h3>

<p>These data were kindly provided by Georgina E. King. Detailed information
on the sample UNIL/NB123 can be found in the reference given below. The raw
data can be found in the accompanying supplementary information.
</p>


<h3>References</h3>

<p>King, G.E., Herman, F., Lambert, R., Valla, P.G., Guralnik, B., 2016.
Multi-OSL-thermochronometry of feldspar. Quaternary Geochronology 33, 76-87.
doi:10.1016/j.quageo.2016.01.004
</p>
<p><strong>Details</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> University of Lausanne </td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> UNIL/NB123 </td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Namche Barwa (eastern Himalayas)</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Coarse grained (180-212 microns) potassium feldspar </td>
</tr>
<tr>
 <td style="text-align: left;">
Units: </td><td style="text-align: left;"> Values are given in seconds </td>
</tr>
<tr>
 <td style="text-align: left;">
Lab Dose Rate: </td><td style="text-align: left;"> Dose rate of the beta-source at measurement ca. 0.1335 +/- 0.004 Gy/s </td>
</tr>
<tr>
 <td style="text-align: left;">
Environmental Dose Rate: </td><td style="text-align: left;"> 7.00 +/- 0.92 Gy/ka (includes internal dose rate)
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
## Load example data
data("ExampleData.Fading", envir = environment())

## Get fading measurement data of the IR50 signal
IR50_fading &lt;- ExampleData.Fading$fading.data$IR50
head(IR50_fading)

## Determine g-value and rho' for the IR50 signal
IR50_fading.res &lt;- analyse_FadingMeasurement(IR50_fading)

## Show g-value and rho' results
gval &lt;- get_RLum(IR50_fading.res)
rhop &lt;- get_RLum(IR50_fading.res, "rho_prime")

gval
rhop

## Get LxTx values of the IR50 DE measurement
IR50_De.LxTx &lt;- ExampleData.Fading$equivalentDose.data$IR50

## Calculate the De of the IR50 signal
IR50_De &lt;- fit_DoseResponseCurve(IR50_De.LxTx,
                                mode = "interpolation",
                                fit.method = "EXP")

## Extract the calculated De and its error
IR50_De.res &lt;- get_RLum(IR50_De)
De &lt;- c(IR50_De.res$De, IR50_De.res$De.Error)

## Apply fading correction (age conversion greatly simplified)
IR50_Age &lt;- De / 7.00
IR50_Age.corr &lt;- calc_FadingCorr(IR50_Age, g_value = IR50_fading.res)


</code></pre>

<hr>
<h2 id='ExampleData.FittingLM'>Example data for fit_LMCurve() in the package Luminescence</h2><span id='topic+ExampleData.FittingLM'></span><span id='topic+values.curve'></span><span id='topic+values.curveBG'></span>

<h3>Description</h3>

<p>Linearly modulated (LM) measurement data from a quartz sample from Norway
including background measurement. Measurements carried out in the
luminescence laboratory at the University of Bayreuth.
</p>


<h3>Format</h3>

<p>Two objects (data.frames) with two columns (time and counts).
</p>


<h3>Source</h3>


<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Bayreuth</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> BT900</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Norway</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Beach deposit, coarse grain quartz measured on aluminium discs on a Ris√∏ TL/OSL DA-15 reader</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>References</h3>

<p>Fuchs, M., Kreutzer, S., Fischer, M., Sauer, D., Soerensen, R., 2012. OSL and IRSL
dating of raised beach sand deposits along the south-eastern coast of Norway.
Quaternary Geochronology, 10, 195-200.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##show LM data
data(ExampleData.FittingLM, envir = environment())
plot(values.curve,log="x")

</code></pre>

<hr>
<h2 id='ExampleData.LxTxData'>Example Lx/Tx data from CW-OSL SAR measurement</h2><span id='topic+ExampleData.LxTxData'></span><span id='topic+LxTxData'></span>

<h3>Description</h3>

<p>LxTx data from a SAR measurement for the package Luminescence.
</p>


<h3>Format</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> with 4 columns (Dose, LxTx, LxTx.Error, TnTx).
</p>


<h3>Source</h3>


<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Bayreuth</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> BT607</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Ostrau (Saxony-Anhalt/Germany)</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Middle grain (38-63 <code class="reqn">\mu</code>m) quartz measured on a Risoe TL/OSL DA-15 reader.
</td>
</tr>

</table>



<h3>References</h3>

<p>unpublished data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## plot Lx/Tx data vs dose [s]
data(ExampleData.LxTxData, envir = environment())
plot(LxTxData$Dose,LxTxData$LxTx)

</code></pre>

<hr>
<h2 id='ExampleData.LxTxOSLData'>Example Lx and Tx curve data from an artificial OSL measurement</h2><span id='topic+ExampleData.LxTxOSLData'></span><span id='topic+Lx.data'></span><span id='topic+Tx.data'></span>

<h3>Description</h3>

<p><code>Lx</code> and <code>Tx</code> data of continuous wave (CW-) OSL signal curves.
</p>


<h3>Format</h3>

<p>Two <code><a href="base.html#topic+data.frame">data.frame</a></code>s containing time and count values.
</p>


<h3>Source</h3>

<p>Arbitrary OSL measurement.
</p>


<h3>References</h3>

<p>unpublished data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.LxTxOSLData, envir = environment())

##plot data
plot(Lx.data)
plot(Tx.data)

</code></pre>

<hr>
<h2 id='ExampleData.MortarData'>Example equivalent dose data from mortar samples</h2><span id='topic+ExampleData.MortarData'></span><span id='topic+MortarData'></span>

<h3>Description</h3>

<p>Arbitrary data to test the function <code>calc_EED_Model</code>
</p>


<h3>Format</h3>

<p>Two <code><a href="base.html#topic+data.frame">data.frame</a></code>s containing De and De error
</p>


<h3>Source</h3>

<p>Arbitrary measurements.
</p>


<h3>References</h3>

<p>unpublished data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.MortarData, envir = environment())

##plot data
plot(MortarData)

</code></pre>

<hr>
<h2 id='ExampleData.portableOSL'>Example portable OSL curve data for the package Luminescence</h2><span id='topic+ExampleData.portableOSL'></span>

<h3>Description</h3>

<p>A <code>list</code> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects, each containing
the same number of <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects representing
individual OSL, IRSL and dark count measurements of a sample.
</p>


<h3>Source</h3>

<p><strong>ExampleData.portableOSL</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Cologne Luminescence Laboratory</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> <code>none</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Nievenheim/Germany</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Fine grain quartz </td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> unpublished data
</td>
</tr>

</table>



<h3>Examples</h3>

<pre><code class='language-R'>
data(ExampleData.portableOSL, envir = environment())
plot_RLum(ExampleData.portableOSL)

</code></pre>

<hr>
<h2 id='ExampleData.RLum.Analysis'>Example data as <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects</h2><span id='topic+ExampleData.RLum.Analysis'></span><span id='topic+IRSAR.RF.Data'></span>

<h3>Description</h3>

<p>Collection of different <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects for
protocol analysis.
</p>


<h3>Format</h3>

<p><code>IRSAR.RF.Data</code>: IRSAR.RF.Data on coarse grain feldspar
</p>
<p>Each object contains data needed for the given protocol analysis.
</p>


<h3>Version</h3>

<p>0.1
</p>


<h3>Source</h3>

<p><strong>IRSAR.RF.Data</strong>
</p>
<p>These data were kindly provided by Tobias Lauer and Matthias Krbetschek.
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory TU Bergakademie Freiberg</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> ZEU/SA1</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Zeuchfeld (Zeuchfeld Sandur; Saxony-Anhalt/Germany)</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> K-feldspar (130-200 <code class="reqn">\mu</code>m)</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> Kreutzer et al. (2014)
</td>
</tr>

</table>



<h3>References</h3>

<p><strong>IRSAR.RF.Data</strong>
</p>
<p>Kreutzer, S., Lauer, T., Meszner, S., Krbetschek, M.R., Faust, D., Fuchs,
M., 2014. Chronology of the Quaternary profile Zeuchfeld in Saxony-Anhalt /
Germany - a preliminary luminescence dating study. Zeitschrift fuer
Geomorphologie 58, 5-26. doi: 10.1127/0372-8854/2012/S-00112
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.RLum.Analysis, envir = environment())

##plot data
plot_RLum(IRSAR.RF.Data)

</code></pre>

<hr>
<h2 id='ExampleData.RLum.Data.Image'>Example data as <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> objects</h2><span id='topic+ExampleData.RLum.Data.Image'></span>

<h3>Description</h3>

<p>Measurement of Princton Instruments camera imported with the function
<a href="#topic+read_SPE2R">read_SPE2R</a> to R to produce an
<a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> object.
</p>


<h3>Format</h3>

<p>Object of class <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>
</p>


<h3>Version</h3>

<p>0.1
</p>


<h3>Source</h3>

<p><strong>ExampleData.RLum.Data.Image</strong>
</p>
<p>These data were kindly provided by Regina DeWitt.
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab.: </td><td style="text-align: left;"> Department of Physics, East-Carolina University, NC, USA</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> - </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Image data is a measurement of fluorescent ceiling lights with a cooled
Princeton Instruments (TM) camera fitted on Ris√∏ DA-20 TL/OSL reader.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.RLum.Data.Image, envir = environment())

##plot data
plot_RLum(ExampleData.RLum.Data.Image)

</code></pre>

<hr>
<h2 id='ExampleData.ScaleGammaDose'>Example data for scale_GammaDose()</h2><span id='topic+ExampleData.ScaleGammaDose'></span>

<h3>Description</h3>

<p>An example data set for the function <code>scale_GammaDose()</code> containing layer
specific information to scale the gamma dose rate considering variations in
soil radioactivity.
</p>


<h3>Format</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code>. Please see <code>?scale_GammaDose()</code> for a detailed description
of its structure.
</p>


<h3>Version</h3>

<p>0.1
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load data
data("ExampleData.ScaleGammaDose", envir = environment())

</code></pre>

<hr>
<h2 id='ExampleData.SurfaceExposure'>Example OSL surface exposure dating data</h2><span id='topic+ExampleData.SurfaceExposure'></span>

<h3>Description</h3>

<p>A set of synthetic OSL surface exposure dating data to demonstrate the
<a href="#topic+fit_SurfaceExposure">fit_SurfaceExposure</a> functionality. See examples to reproduce the data
interactively.
</p>


<h3>Format</h3>

<p>A <a href="base.html#topic+list">list</a> with 4 elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Content</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$sample_1&#8288;</code> </td><td style="text-align: left;"> A <a href="base.html#topic+data.frame">data.frame</a> with 3 columns (depth, intensity, error) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$sample_2&#8288;</code> </td><td style="text-align: left;"> A <a href="base.html#topic+data.frame">data.frame</a> with 3 columns (depth, intensity, error) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$set_1&#8288;</code> </td><td style="text-align: left;"> A <a href="base.html#topic+list">list</a> of 4 <a href="base.html#topic+data.frame">data.frame</a>s, each representing a sample with different ages </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$set_2&#8288;</code> </td><td style="text-align: left;"> A <a href="base.html#topic+list">list</a> of 5 <a href="base.html#topic+data.frame">data.frame</a>s, each representing a sample with different ages </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p><strong><code style="white-space: pre;">&#8288;$sample_1&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: center;">
<strong>mu</strong> </td><td style="text-align: center;"> <strong><code>sigmaphi</code></strong> </td><td style="text-align: center;"> <strong>age</strong> </td>
</tr>
<tr>
 <td style="text-align: center;">
0.9 </td><td style="text-align: center;"> 5e-10 </td><td style="text-align: center;"> 10000 </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>

<p><strong><code style="white-space: pre;">&#8288;$sample_2&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: center;">
<strong>mu</strong> </td><td style="text-align: center;"> <strong><code>sigmaphi</code></strong> </td><td style="text-align: center;"> <strong>age</strong> </td><td style="text-align: center;"> <strong>Dose rate</strong> </td><td style="text-align: center;"> <strong>D0</strong> </td>
</tr>
<tr>
 <td style="text-align: center;">
0.9 </td><td style="text-align: center;"> 5e-10 </td><td style="text-align: center;"> 10000 </td><td style="text-align: center;"> 2.5 </td><td style="text-align: center;"> 40 </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>

<p><strong><code style="white-space: pre;">&#8288;$set_1&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: center;">
<strong>mu</strong> </td><td style="text-align: center;"> <strong><code>sigmaphi</code></strong> </td><td style="text-align: center;"> <strong>ages</strong> </td>
</tr>
<tr>
 <td style="text-align: center;">
0.9 </td><td style="text-align: center;"> 5e-10 </td><td style="text-align: center;"> 1e3, 1e4, 1e5, 1e6 </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>

<p><strong><code style="white-space: pre;">&#8288;$set_2&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: center;">
<strong>mu</strong> </td><td style="text-align: center;"> <strong><code>sigmaphi</code></strong> </td><td style="text-align: center;"> <strong>ages</strong> </td><td style="text-align: center;"> <strong>Dose rate</strong> </td><td style="text-align: center;"> <strong>D0</strong> </td>
</tr>
<tr>
 <td style="text-align: center;">
0.9 </td><td style="text-align: center;"> 5e-10 </td><td style="text-align: center;"> 1e2, 1e3, 1e4, 1e5, 1e6 </td><td style="text-align: center;"> 1.0 </td><td style="text-align: center;"> 40 </td>
</tr>
<tr>
 <td style="text-align: center;">
</td>
</tr>

</table>



<h3>Source</h3>

<p>See examples for the code used to create the data sets.
</p>


<h3>References</h3>

<p>Unpublished synthetic data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## ExampleData.SurfaceExposure$sample_1
sigmaphi &lt;- 5e-10
age &lt;- 10000
mu &lt;- 0.9
x &lt;- seq(0, 10, 0.1)
fun &lt;- exp(-sigmaphi * age * 365.25*24*3600 * exp(-mu * x))

set.seed(666)
synth_1 &lt;- data.frame(depth = x,
                      intensity = jitter(fun, 1, 0.1),
                      error = runif(length(x), 0.01, 0.2))

## VALIDATE sample_1
fit_SurfaceExposure(synth_1, mu = mu, sigmaphi = sigmaphi)




## ExampleData.SurfaceExposure$sample_2
sigmaphi &lt;- 5e-10
age &lt;- 10000
mu &lt;- 0.9
x &lt;- seq(0, 10, 0.1)
Ddot &lt;- 2.5   / 1000 / 365.25 / 24 / 60 / 60 # 2.5 Gy/ka in Seconds
D0 &lt;- 40
fun &lt;- (sigmaphi * exp(-mu * x) *
          exp(-(age * 365.25*24*3600) *
                (sigmaphi * exp(-mu * x) + Ddot/D0)) + Ddot/D0) /
  (sigmaphi * exp(-mu * x) + Ddot/D0)

set.seed(666)
synth_2 &lt;- data.frame(depth = x,
                      intensity = jitter(fun, 1, 0.1),
                      error = runif(length(x), 0.01, 0.2))

## VALIDATE sample_2
fit_SurfaceExposure(synth_2, mu = mu, sigmaphi = sigmaphi, Ddot = 2.5, D0 = D0)



## ExampleData.SurfaceExposure$set_1
sigmaphi &lt;- 5e-10
mu &lt;- 0.9
x &lt;- seq(0, 15, 0.2)
age &lt;- c(1e3, 1e4, 1e5, 1e6)
set.seed(666)

synth_3 &lt;- vector("list", length = length(age))

for (i in 1:length(age)) {
  fun &lt;- exp(-sigmaphi * age[i] * 365.25*24*3600 * exp(-mu * x))
  synth_3[[i]] &lt;- data.frame(depth = x,
                             intensity = jitter(fun, 1, 0.05))
}


## VALIDATE set_1
fit_SurfaceExposure(synth_3, age = age, sigmaphi = sigmaphi)



## ExampleData.SurfaceExposure$set_2
sigmaphi &lt;- 5e-10
mu &lt;- 0.9
x &lt;- seq(0, 15, 0.2)
age &lt;- c(1e2, 1e3, 1e4, 1e5, 1e6)
Ddot &lt;- 1.0 / 1000 / 365.25 / 24 / 60 / 60 # 2.0 Gy/ka in Seconds
D0 &lt;- 40
set.seed(666)

synth_4 &lt;- vector("list", length = length(age))

for (i in 1:length(age)) {
  fun &lt;- (sigmaphi * exp(-mu * x) *
            exp(-(age[i] * 365.25*24*3600) *
                  (sigmaphi * exp(-mu * x) + Ddot/D0)) + Ddot/D0) /
    (sigmaphi * exp(-mu * x) + Ddot/D0)

  synth_4[[i]] &lt;- data.frame(depth = x,
                             intensity = jitter(fun, 1, 0.05))
}


## VALIDATE set_2
fit_SurfaceExposure(synth_4, age = age, sigmaphi = sigmaphi, D0 = D0, Ddot = 1.0)

## Not run: 
ExampleData.SurfaceExposure &lt;- list(
  sample_1 = synth_1,
  sample_2 = synth_2,
  set_1 = synth_3,
  set_2 = synth_4
)

## End(Not run)

</code></pre>

<hr>
<h2 id='ExampleData.TR_OSL'>Example TR-OSL data</h2><span id='topic+ExampleData.TR_OSL'></span>

<h3>Description</h3>

<p>Single TR-OSL curve obtained by Schmidt et al. (under review) for quartz sample BT729
(origin: Trebgast Valley, Germany, quartz, 90-200 ¬µm, unpublished data).
</p>


<h3>Format</h3>

<p>One <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> dataset imported using the function <a href="#topic+read_XSYG2R">read_XSYG2R</a>
</p>

<p><code>ExampleData.TR_OSL</code>: A single <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> object with the TR-OSL data
</p>



<h3>References</h3>

<p>Schmidt, C., Simmank, O., Kreutzer, S., under review.
Time-Resolved Optically Stimulated Luminescence of Quartz in the Nanosecond Time Domain. Journal
of Luminescence, 1-90
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_OSLLifeTimes">fit_OSLLifeTimes</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) curves
data(ExampleData.TR_OSL, envir = environment())
plot_RLum(ExampleData.TR_OSL)

</code></pre>

<hr>
<h2 id='ExampleData.XSYG'>Example data for a SAR OSL measurement and a TL spectrum using a lexsyg
reader</h2><span id='topic+ExampleData.XSYG'></span><span id='topic+OSL.SARMeasurement'></span><span id='topic+TL.Spectrum'></span>

<h3>Description</h3>

<p>Example data from a SAR OSL measurement and a TL spectrum for package
Luminescence imported from a Freiberg Instruments XSYG file using the
function <a href="#topic+read_XSYG2R">read_XSYG2R</a>.
</p>


<h3>Format</h3>

<p><code>OSL.SARMeasurement</code>: SAR OSL measurement data
</p>
<p>The data contain two elements: (a) <code style="white-space: pre;">&#8288;$Sequence.Header&#8288;</code> is a
<a href="base.html#topic+data.frame">data.frame</a> with metadata from the measurement,(b)
<code>Sequence.Object</code> contains an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object
for further analysis.
</p>
<p><code>TL.Spectrum</code>: TL spectrum data
</p>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> object for further analysis. The
spectrum was cleaned from cosmic-rays using the function
</p>
<p><code>apply_CosmicRayRemoval</code>. Note that no quantum efficiency calibration
was performed.
</p>


<h3>Version</h3>

<p>0.1
</p>


<h3>Source</h3>

<p><strong>OSL.SARMeasurement</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Giessen</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> no code</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> not specified</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Coarse grain quartz  on steel cups on lexsyg research reader</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> unpublished
</td>
</tr>

</table>

<p><strong>TL.Spectrum</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
Lab: </td><td style="text-align: left;"> Luminescence Laboratory Giessen</td>
</tr>
<tr>
 <td style="text-align: left;">
Lab-Code: </td><td style="text-align: left;"> BT753</td>
</tr>
<tr>
 <td style="text-align: left;">
Location: </td><td style="text-align: left;"> Dolni Vestonice/Czech Republic</td>
</tr>
<tr>
 <td style="text-align: left;">
Material: </td><td style="text-align: left;"> Fine grain polymineral on steel cups on lexsyg research reader</td>
</tr>
<tr>
 <td style="text-align: left;">
Reference: </td><td style="text-align: left;"> Fuchs et al., 2013 </td>
</tr>
<tr>
 <td style="text-align: left;">
Spectrum: </td><td style="text-align: left;"> Integration time 19 s, channel time 20 s</td>
</tr>
<tr>
 <td style="text-align: left;">
Heating: </td><td style="text-align: left;"> 1 K/s, up to 500 deg. C
</td>
</tr>

</table>



<h3>References</h3>

<p>Unpublished data measured to serve as example data for that
package. Location origin of sample BT753 is given here:
</p>
<p>Fuchs, M., Kreutzer, S., Rousseau, D.D., Antoine, P., Hatte, C., Lagroix,
F., Moine, O., Gauthier, C., Svoboda, J., Lisa, L., 2013. The loess sequence
of Dolni Vestonice, Czech Republic: A new OSL-based chronology of the Last
Climatic Cycle. Boreas, 42, 664&ndash;677.
</p>


<h3>See Also</h3>

<p><a href="#topic+read_XSYG2R">read_XSYG2R</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>,
<a href="#topic+plot_RLum">plot_RLum</a>, <a href="#topic+plot_RLum.Analysis">plot_RLum.Analysis</a>, <a href="#topic+plot_RLum.Data.Spectrum">plot_RLum.Data.Spectrum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##show data
data(ExampleData.XSYG, envir = environment())

## =========================================
##(1) OSL.SARMeasurement
OSL.SARMeasurement

##show $Sequence.Object
OSL.SARMeasurement$Sequence.Object

##grep OSL curves and plot the first curve
OSLcurve &lt;- get_RLum(OSL.SARMeasurement$Sequence.Object,
recordType="OSL")[[1]]
plot_RLum(OSLcurve)

## =========================================
##(2) TL.Spectrum
TL.Spectrum

##plot simple spectrum (2D)
plot_RLum.Data.Spectrum(TL.Spectrum,
                        plot.type="contour",
                        xlim = c(310,750),
                        ylim = c(0,300),
                        bin.rows=10,
                        bin.cols = 1)

##plot 3d spectrum (uncomment for usage)
# plot_RLum.Data.Spectrum(TL.Spectrum, plot.type="persp",
# xlim = c(310,750), ylim = c(0,300), bin.rows=10,
# bin.cols = 1)

</code></pre>

<hr>
<h2 id='extdata'>Collection of External Data</h2><span id='topic+extdata'></span>

<h3>Description</h3>

<p>Description and listing of data provided in the folder <code>data/extdata</code>
</p>


<h3>Details</h3>

<p>The <strong>R</strong> package <code>Luminescence</code> includes a number of raw data files, which are mostly used in
the example sections of appropriate functions. They are also used internally for testing corresponding
functions using the <code>testthat</code> package (see files in <code style="white-space: pre;">&#8288;tests/testthat/&#8288;</code>) to ensure their operational
reliability.
</p>
<p><strong>Accessibility</strong>
</p>
<p>If the <strong>R</strong> package <code>Luminescence</code> is installed correctly the preferred way to access and use these
data from within <strong>R</strong> is as follows:
</p>
<p><code>system.file("extdata/&lt;FILENAME&gt;", package = "Luminescence")</code>
</p>
<p><strong>Individual file descriptions</strong>
</p>
<p><em>&gt;&gt;Daybreak_TestFile.DAT/.txt&lt;&lt;</em>
</p>
<p><strong>Type:</strong> raw measurement data <br />
<strong>Device:</strong> Daybreak OSL/TL reader<br />
<strong>Measurement date:</strong> unknown<br />
<strong>Location:</strong> unknown<br />
<strong>Provided by:</strong> unknown<br />
<strong>Related R function(s):</strong> <code>read_Daybreak2R()</code><br />
<strong>Reference:</strong> unknown
</p>
<p><em>&gt;&gt;DorNie_0016.psl&lt;&lt;</em>
</p>
<p><strong>Type:</strong> raw measurement data <br />
<strong>Device:</strong> SUERC portable OSL reader <br />
<strong>Measurement date:</strong> 19/05/2016 <br />
<strong>Location:</strong> Dormagen-Nievenheim, Germany <br />
<strong>Provided by:</strong> Christoph Burow (University of Cologne) <br />
<strong>Related R function(s):</strong> <code>read_PSL2R()</code> <br />
<strong>Reference:</strong> unpublished <br />
<strong>Additional information:</strong> Sample measured at an archaeological site near <br />
Dormagen-Nievenheim (Germany) during a practical course on Luminescence dating in 2016. <br />
</p>
<p><em>&gt;&gt;QNL84_2_bleached.txt</em>, <em>QNL84_2_unbleached.txt&lt;&lt;</em>
</p>
<p><strong>Type:</strong> Test data for exponential fits <br />
<strong>Reference:</strong> Berger, G.W., Huntley, D.J., 1989. Test data for exponential fits. Ancient TL 7, 43-46. <br />
</p>
<p><em>&gt;&gt;STRB87_1_bleached.txt</em>, <em>STRB87_1_unbleached.txt&lt;&lt;</em>
</p>
<p><strong>Type:</strong> Test data for exponential fits <br />
<strong>Reference:</strong> Berger, G.W., Huntley, D.J., 1989. Test data for exponential fits. Ancient TL 7, 43-46.
</p>
<p><em>&gt;&gt;XSYG_file.xsyg</em>
</p>
<p><strong>Type:</strong> XSYG-file stump <br />
**Info: ** XSYG-file with some basic curves to test functions <br />
<strong>Reference:</strong> no reference available
</p>

<hr>
<h2 id='extract_IrradiationTimes'>Extract Irradiation Times from an XSYG-file</h2><span id='topic+extract_IrradiationTimes'></span>

<h3>Description</h3>

<p>Extracts irradiation times, dose and times since last irradiation, from a
Freiberg Instruments XSYG-file. These information can be further used to
update an existing BINX-file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_IrradiationTimes(
  object,
  file.BINX,
  recordType = c("irradiation (NA)", "IRSL (UVVIS)", "OSL (UVVIS)", "TL (UVVIS)"),
  compatibility.mode = TRUE,
  txtProgressBar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_IrradiationTimes_+3A_object">object</code></td>
<td>
<p><a href="base.html#topic+character">character</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
path and file name of the XSYG file or an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
produced by the function <a href="#topic+read_XSYG2R">read_XSYG2R</a>;
alternatively a <code>list</code> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> can be provided.
</p>
<p><strong>Note</strong>: If an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> is used, any input for
the arguments <code>file.BINX</code> and <code>recordType</code> will be ignored!</p>
</td></tr>
<tr><td><code id="extract_IrradiationTimes_+3A_file.binx">file.BINX</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
path and file name of an existing BINX-file. If a file name is provided the
file will be updated with the information from the XSYG file in the same
folder as the original BINX-file.
</p>
<p><strong>Note:</strong> The XSYG and the BINX-file must originate from the
same measurement!</p>
</td></tr>
<tr><td><code id="extract_IrradiationTimes_+3A_recordtype">recordType</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
select relevant curves types from the XSYG file or <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
object. As the XSYG-file format comprises much more information than usually
needed for routine data analysis and allowed in the BINX-file format, only
the relevant curves are selected by using the function
<a href="#topic+get_RLum">get_RLum</a>. The argument <code>recordType</code> works as
described for this function.
</p>
<p><strong>Note:</strong> A wrong selection will causes a function error. Please change this
argument only if you have reasons to do so.</p>
</td></tr>
<tr><td><code id="extract_IrradiationTimes_+3A_compatibility.mode">compatibility.mode</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
this option is parsed only if a BIN/BINX file is produced and it will reset all position
values to a max. value of 48, cf.<a href="#topic+write_R2BIN">write_R2BIN</a></p>
</td></tr>
<tr><td><code id="extract_IrradiationTimes_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar during import and export.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function was written to compensate missing information in the BINX-file
output of Freiberg Instruments lexsyg readers. As all information are
available within the XSYG-file anyway, these information can be extracted
and used for further analysis or/and to stored in a new BINX-file, which can
be further used by other software, e.g., Analyst (Geoff Duller).
</p>
<p>Typical application example: g-value estimation from fading measurements
using the Analyst or any other self-written script.
</p>
<p>Beside some simple data transformation steps, the function applies
functions <a href="#topic+read_XSYG2R">read_XSYG2R</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+write_R2BIN">write_R2BIN</a> for data import and export.
</p>


<h3>Value</h3>

<p>An <a href="#topic+RLum.Results-class">RLum.Results</a> object is returned with the
following structure:
</p>
<div class="sourceCode"><pre>.. $irr.times (data.frame)
</pre></div>
<p>If a BINX-file path and name is set, the output will be additionally
transferred into a new BINX-file with the function name as suffix. For the
output the path of the input BINX-file itself is used. Note that this will
not work if the input object is a file path to an XSYG-file, instead of a
link to only one file. In this case the argument input for <code>file.BINX</code> is ignored.
</p>
<p>In the self call mode (input is a <code>list</code> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
a list of <a href="#topic+RLum.Results-class">RLum.Results</a> is returned.
</p>


<h3>Function version</h3>

<p>0.3.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. extract_IrradiationTimes(): Extract Irradiation Times from an XSYG-file. Function version 0.3.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The function can be also used to extract irradiation times from <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
previously imported via <a href="#topic+read_BIN2R">read_BIN2R</a> (<code>fastForward = TRUE</code>) or in combination with <a href="#topic+Risoe.BINfileData2RLum.Analysis">Risoe.BINfileData2RLum.Analysis</a>.
Unfortunately the timestamp might not be very precise (or even invalid),
but it allows to essentially treat different formats in a similar manner.
</p>
<p>The produced output object contains still the irradiation steps to
keep the output transparent. However, for the BINX-file export this steps
are removed as the BINX-file format description does not allow irradiations
as separate sequences steps.
</p>
<p><strong>BINX-file 'Time Since Irradiation' value differs from the table output?</strong>
</p>
<p>The way the value 'Time Since Irradiation' is defined differs. In the BINX-file the
'Time Since Irradiation' is calculated as the 'Time Since Irradiation' plus the 'Irradiation
Time'. The table output returns only the real 'Time Since Irradiation', i.e. time between the
end of the irradiation and the next step.
</p>
<p><strong>Negative values for <code>TIMESINCELAST.STEP</code>?</strong>
</p>
<p>Yes, this is possible and no bug, as in the XSYG-file multiple curves are stored for one step.
Example: TL step may comprise three curves:
</p>

<ul>
<li><p> (a) counts vs. time,
</p>
</li>
<li><p> (b) measured temperature vs. time and
</p>
</li>
<li><p> (c) predefined temperature vs. time.
</p>
</li></ul>

<p>Three curves, but they are all belonging to one TL measurement step, but with regard to
the time stamps this could produce negative values as the important function
(<a href="#topic+read_XSYG2R">read_XSYG2R</a>) do not change the order of entries for one step
towards a correct time order.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Duller, G.A.T., 2015. The Analyst software package for luminescence data: overview and
recent improvements. Ancient TL 33, 35-42.
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>,
<a href="#topic+read_XSYG2R">read_XSYG2R</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+write_R2BIN">write_R2BIN</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## (1) - example for your own data
##
## set files and run function
#
#   file.XSYG &lt;- file.choose()
#   file.BINX &lt;- file.choose()
#
#     output &lt;- extract_IrradiationTimes(file.XSYG = file.XSYG, file.BINX = file.BINX)
#     get_RLum(output)
#
## export results additionally to a CSV-file in the same directory as the XSYG-file
#       write.table(x = get_RLum(output),
#                   file = paste0(file.BINX,"_extract_IrradiationTimes.csv"),
#                   sep = ";",
#                   row.names = FALSE)

</code></pre>

<hr>
<h2 id='extract_ROI'>Extract Pixel Values through Circular Region-of-Interests (ROI) from an Image</h2><span id='topic+extract_ROI'></span>

<h3>Description</h3>

<p>Light-weighted function to extract pixel values from pre-defined regions-of-interest (ROI) from
<a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>, <a href="base.html#topic+array">array</a> or <a href="base.html#topic+matrix">matrix</a> objects and provide simple image processing
capacity. The function is limited to circular ROIs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_ROI(object, roi, roi_summary = "mean", plot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="extract_ROI_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>, <a href="base.html#topic+array">array</a> or <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>): input image data</p>
</td></tr>
<tr><td><code id="extract_ROI_+3A_roi">roi</code></td>
<td>
<p><a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>): matrix with three columns containing the centre coordinates
of the ROI (first two columns) and the diameter of the circular ROI. All numbers must by of type <a href="base.html#topic+integer">integer</a>
and will forcefully coerced into such numbers using <code>as.integer()</code> regardless.</p>
</td></tr>
<tr><td><code id="extract_ROI_+3A_roi_summary">roi_summary</code></td>
<td>
<p>(<em>with default</em>): defines what is returned in the
<code>roi_summary</code> element; it can be <code>"mean"</code> (default), <code>"median"</code>, <code>"sd"</code> or
<code>"sum"</code>.
Pixel values are conveniently summarised using the above defined keyword.</p>
</td></tr>
<tr><td><code id="extract_ROI_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>): enable/disable the plot output. Only
the first image frame is shown.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses a cheap approach to decide whether a pixel lies within
a circle or not. It assumes that pixel coordinates are integer values and
that a pixel centring within the circle is satisfied by:
</p>
<p style="text-align: center;"><code class="reqn">x^2 + y^2 &lt;= (d/2)^2</code>
</p>

<p>where <code class="reqn">x</code> and <code class="reqn">y</code> are integer pixel coordinates and <code class="reqn">d</code> is the integer
diameter of the circle in pixel.
</p>


<h3>Value</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a> object with the following elements:
<code>..$roi_signals</code>: a named <a href="base.html#topic+list">list</a> with all ROI values and their coordinates
<code>..$roi_summary</code>: a <a href="base.html#topic+matrix">matrix</a> where rows are frames from the image, and columns are different ROI
The element has two attributes: <code>summary</code> (the method used to summarise pixels) and <code>area</code> (the pixel area)
<code>..$roi_coord</code>: a <a href="base.html#topic+matrix">matrix</a> that can be passed to <a href="#topic+plot_ROI">plot_ROI</a>
</p>
<p>If <code>plot = TRUE</code> a control plot is returned.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. extract_ROI(): Extract Pixel Values through Circular Region-of-Interests (ROI) from an Image. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
m &lt;- matrix(runif(100,0,255), ncol = 10, nrow = 10)
roi &lt;- matrix(c(2.,4,2,5,6,7,3,1,1), ncol = 3)
extract_ROI(object = m, roi = roi, plot = TRUE)

</code></pre>

<hr>
<h2 id='fit_CWCurve'>Nonlinear Least Squares Fit for CW-OSL curves -beta version-</h2><span id='topic+fit_CWCurve'></span>

<h3>Description</h3>

<p>The function determines the weighted least-squares estimates of the
component parameters of a CW-OSL signal for a given maximum number of
components and returns various component parameters. The fitting procedure
uses the <a href="stats.html#topic+nls">nls</a> function with the <code>port</code> algorithm.
</p>
<p><strong>Fitting function</strong>
</p>
<p>The function for the CW-OSL fitting has the general form:
</p>
<p style="text-align: center;"><code class="reqn">y = I0_{1}*\lambda_{1}*exp(-\lambda_1*x) + ,\ldots, + I0_{i}*\lambda_{i}*exp(-\lambda_i*x) </code>
</p>

<p>where <code class="reqn">0 &lt; i &lt; 8</code>
</p>
<p>and <code class="reqn">\lambda</code> is the decay constant <br />
and <code class="reqn">I0</code> the initial number of trapped electrons.
</p>
<p><em>(for the used equation cf. Boetter-Jensen et al., 2003, Eq. 2.31)</em>
</p>
<p><strong>Start values</strong>
</p>
<p>Start values are estimated automatically by fitting a linear function to the
logarithmized input data set. Currently, there is no option to manually
provide start parameters.
</p>
<p><strong>Goodness of fit</strong>
</p>
<p>The goodness of the fit is given as pseudoR¬≤ value (pseudo coefficient of
determination). According to Lave (1970), the value is calculated as:
</p>
<p style="text-align: center;"><code class="reqn">pseudoR^2 = 1 - RSS/TSS</code>
</p>

<p>where <code class="reqn">RSS = Residual~Sum~of~Squares</code> <br />
and <code class="reqn">TSS = Total~Sum~of~Squares</code>
</p>
<p><strong>Error of fitted component parameters</strong>
</p>
<p>The 1-sigma error for the
components is calculated using the function <a href="stats.html#topic+confint">stats::confint</a>. Due to
considerable calculation time, this option is deactivated by default. In
addition, the error for the components can be estimated by using internal R
functions like <a href="base.html#topic+summary">summary</a>. See the <a href="stats.html#topic+nls">nls</a> help page
for more information.
</p>
<p><em>For details on the nonlinear regression in R, see Ritz &amp; Streibig (2008).</em>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_CWCurve(
  values,
  n.components.max,
  fit.failure_threshold = 5,
  fit.method = "port",
  fit.trace = FALSE,
  fit.calcError = FALSE,
  LED.power = 36,
  LED.wavelength = 470,
  cex.global = 0.6,
  sample_code = "Default",
  verbose = TRUE,
  output.terminalAdvanced = TRUE,
  plot = TRUE,
  method_control = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_CWCurve_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
x, y data of measured values (time and counts). See examples.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_n.components.max">n.components.max</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
maximum number of components that are to be used for fitting.
The upper limit is 7.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_fit.failure_threshold">fit.failure_threshold</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
limits the failed fitting attempts.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_fit.method">fit.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
select fit method, allowed values: <code>'port'</code> and <code>'LM'</code>. <code>'port'</code> uses the 'port'
routine from the function <a href="stats.html#topic+nls">nls</a> <code>'LM'</code> utilises the function <code>nlsLM</code> from
the package <code>minpack.lm</code> and with that the Levenberg-Marquardt algorithm.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_fit.trace">fit.trace</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
traces the fitting process on the terminal.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_fit.calcerror">fit.calcError</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
calculate 1-sigma error range of components using <a href="stats.html#topic+confint">stats::confint</a></p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_led.power">LED.power</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
LED power (max.) used for intensity ramping in mW/cm¬≤.
<strong>Note:</strong> The value is used for the calculation of the absolute
photoionisation cross section.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_led.wavelength">LED.wavelength</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
LED wavelength used for stimulation in nm.
<strong>Note:</strong> The value is used for the calculation of the absolute
photoionisation cross section.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_cex.global">cex.global</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
global scaling factor.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_sample_code">sample_code</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
sample code used for the plot and the optional output table (<code>mtext</code>).</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_output.terminaladvanced">output.terminalAdvanced</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enhanced terminal output. Only valid if <code>verbose = TRUE</code>.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>): options to control the output
produced. Currently only the 'export.comp.contrib.matrix' (logical) option
is supported, to enable/disable export of the component contribution
matrix.</p>
</td></tr>
<tr><td><code id="fit_CWCurve_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to <a href="base.html#topic+plot">plot</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong>plot (<em>optional</em>)</strong>
</p>
<p>the fitted CW-OSL curves are returned as plot.
</p>
<p><strong>RLum.Results object</strong>
</p>
<p>Beside the plot and table output options, an <a href="#topic+RLum.Results-class">RLum.Results</a> object is
returned.
</p>
<p><code>fit</code>:
an <code>nls</code> object (<code style="white-space: pre;">&#8288;$fit&#8288;</code>) for which generic R functions are
provided, e.g. <a href="base.html#topic+summary">summary</a>, <a href="stats.html#topic+confint">stats::confint</a>, <a href="stats.html#topic+profile">profile</a>. For more
details, see <a href="stats.html#topic+nls">nls</a>.
</p>
<p><code>output.table</code>:
a <a href="base.html#topic+data.frame">data.frame</a> containing the summarised parameters including the error
</p>
<p><code>component.contribution.matrix</code>:
<a href="base.html#topic+matrix">matrix</a> containing the values for the component to sum contribution plot
(<code style="white-space: pre;">&#8288;$component.contribution.matrix&#8288;</code>).
Produced only if <code>method_control$export.comp.contrib.matrix = TRUE</code>).
</p>
<p>Matrix structure:<br />
Column 1 and 2: time and <code>rev(time)</code> values <br />
Additional columns are used for the components, two for each component,
containing I0 and n0. The last columns <code>cont.</code> provide information on
the relative component contribution for each time interval including the row
sum for this values.
</p>


<h3>Function version</h3>

<p>0.5.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. fit_CWCurve(): Nonlinear Least Squares Fit for CW-OSL curves -beta version-. Function version 0.5.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>Beta version - This function has not been properly tested yet and</strong>
<strong>should therefore not be used for publication purposes!</strong>
</p>
<p>The pseudo-R¬≤ may not be the best parameter to describe the goodness of the
fit. The trade off between the <code>n.components</code> and the pseudo-R¬≤ value
is currently not considered.
</p>
<p>The function <strong>does not</strong> ensure that the fitting procedure has reached a
global minimum rather than a local minimum!
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Boetter-Jensen, L., McKeever, S.W.S., Wintle, A.G., 2003.
Optically Stimulated Luminescence Dosimetry. Elsevier Science B.V.
</p>
<p>Lave, C.A.T., 1970. The Demand for Urban Mass Transportation. The Review of
Economics and Statistics, 52 (3), 320-323.
</p>
<p>Ritz, C. &amp; Streibig, J.C., 2008. Nonlinear Regression with R. In: R.
Gentleman, K. Hornik, G. Parmigiani, eds., Springer, p. 150.
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_LMCurve">fit_LMCurve</a>, <a href="base.html#topic+plot">plot</a>,<a href="stats.html#topic+nls">nls</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
<a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>, <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.CW_OSL_Curve, envir = environment())

##fit data
fit &lt;- fit_CWCurve(values = ExampleData.CW_OSL_Curve,
                   main = "CW Curve Fit",
                   n.components.max = 4,
                   log = "x")

</code></pre>

<hr>
<h2 id='fit_DoseResponseCurve'>Fit a dose-response curve for luminescence data (Lx/Tx against dose)</h2><span id='topic+fit_DoseResponseCurve'></span>

<h3>Description</h3>

<p>A dose-response curve is produced for luminescence measurements using a
regenerative or additive protocol. The function supports interpolation and
extrapolation to calculate the equivalent dose.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_DoseResponseCurve(
  object,
  mode = "interpolation",
  fit.method = "EXP",
  fit.force_through_origin = FALSE,
  fit.weights = TRUE,
  fit.includingRepeatedRegPoints = TRUE,
  fit.NumberRegPoints = NULL,
  fit.NumberRegPointsReal = NULL,
  fit.bounds = TRUE,
  n.MC = 100,
  txtProgressBar = TRUE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_DoseResponseCurve_+3A_object">object</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> or a <a href="base.html#topic+list">list</a> of such objects (<strong>required</strong>):
data frame with columns for <code>Dose</code>, <code>LxTx</code>, <code>LxTx.Error</code> and <code>TnTx</code>.
The column for the test dose response is optional, but requires <code>'TnTx'</code> as
column name if used. For exponential fits at least three dose points
(including the natural) should be provided. If <code>object</code> is a list,
the function is called on each of its elements.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_mode">mode</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
selects calculation mode of the function.
</p>

<ul>
<li> <p><code>"interpolation"</code> (default) calculates the De by interpolation,
</p>
</li>
<li> <p><code>"extrapolation"</code> calculates the equivalent dose by extrapolation (useful for MAAD measurements) and
</p>
</li>
<li> <p><code>"alternate"</code> calculates no equivalent dose and just fits the data points.
</p>
</li></ul>

<p>Please note that for option <code>"interpolation"</code> the first point is considered
as natural dose</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.method">fit.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
function used for fitting. Possible options are:
</p>

<ul>
<li> <p><code>LIN</code>,
</p>
</li>
<li> <p><code>QDR</code>,
</p>
</li>
<li> <p><code>EXP</code>,
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;EXP OR LIN&#8288;</code>,
</p>
</li>
<li> <p><code>EXP+LIN</code>,
</p>
</li>
<li> <p><code>EXP+EXP</code>,
</p>
</li>
<li> <p><code>GOK</code>,
</p>
</li>
<li> <p><code>LambertW</code>
</p>
</li></ul>

<p>See details.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.force_through_origin">fit.force_through_origin</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>)
allow to force the fitted function through the origin.
For <code>method = "EXP+EXP"</code> the function will be fixed through
the origin in either case, so this option will have no effect.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.weights">fit.weights</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
option whether the fitting is done with or without weights. See details.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.includingrepeatedregpoints">fit.includingRepeatedRegPoints</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
includes repeated points for fitting (<code>TRUE</code>/<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.numberregpoints">fit.NumberRegPoints</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
set number of regeneration points manually. By default the number of all (!)
regeneration points is used automatically.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.numberregpointsreal">fit.NumberRegPointsReal</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
if the number of regeneration points is provided manually, the value of the
real, regeneration points = all points (repeated points) including reg 0,
has to be inserted.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_fit.bounds">fit.bounds</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
set lower fit bounds for all fitting parameters to 0. Limited for the use
with the fit methods <code>EXP</code>, <code>EXP+LIN</code>, <code style="white-space: pre;">&#8288;EXP OR LIN&#8288;</code>, <code>GOK</code>, <code>LambertW</code>
Argument to be inserted for experimental application only!</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):  number of Monte Carlo simulations
for error estimation, see details.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar. If <code>verbose = FALSE</code> also no
<code>txtProgressBar</code> is shown.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="fit_DoseResponseCurve_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed (currently ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Fitting methods</strong>
</p>
<p>For all options (except for the <code>LIN</code>, <code>QDR</code> and the <code style="white-space: pre;">&#8288;EXP OR LIN&#8288;</code>),
the <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a> function with the <code>LM</code> (Levenberg-Marquardt algorithm)
algorithm is used. Note: For historical reasons for the Monte Carlo
simulations partly the function <a href="stats.html#topic+nls">nls</a> using the <code>port</code> algorithm.
</p>
<p>The solution is found by transforming the function or using <a href="stats.html#topic+uniroot">stats::uniroot</a>.
</p>
<p><code>LIN</code>: fits a linear function to the data using
<a href="stats.html#topic+lm">lm</a>: </p>
<p style="text-align: center;"><code class="reqn">y = mx + n</code>
</p>

<p><code>QDR</code>: fits a linear function with a quadratic term to the data using
<a href="stats.html#topic+lm">lm</a>: </p>
<p style="text-align: center;"><code class="reqn">y = a + bx + cx^2</code>
</p>

<p><code>EXP</code>: tries to fit a function of the form
</p>
<p style="text-align: center;"><code class="reqn">y = a(1 - exp(-\frac{(x+c)}{b}))</code>
</p>

<p>Parameters b and c are approximated by a linear fit using <a href="stats.html#topic+lm">lm</a>. Note: b = D0
</p>
<p><code style="white-space: pre;">&#8288;EXP OR LIN&#8288;</code>: works for some cases where an <code>EXP</code> fit fails.
If the <code>EXP</code> fit fails, a <code>LIN</code> fit is done instead.
</p>
<p><code>EXP+LIN</code>: tries to fit an exponential plus linear function of the
form:
</p>
<p style="text-align: center;"><code class="reqn">y = a(1-exp(-\frac{x+c}{b}) + (gx))</code>
</p>

<p>The <code class="reqn">D_e</code> is calculated by iteration.
</p>
<p><strong>Note:</strong> In the context of luminescence dating, this
function has no physical meaning. Therefore, no D0 value is returned.
</p>
<p><code>EXP+EXP</code>: tries to fit a double exponential function of the form
</p>
<p style="text-align: center;"><code class="reqn">y = (a_1 (1-exp(-\frac{x}{b_1}))) + (a_2 (1 - exp(-\frac{x}{b_2})))</code>
</p>

<p>This fitting procedure is not robust against wrong start parameters and
should be further improved.
</p>
<p><code>GOK</code>: tries to fit the general-order kinetics function after
Guralnik et al. (2015) of the form of
</p>
<p style="text-align: center;"><code class="reqn">y = a (d - (1 + (\frac{1}{b}) x c)^{(-1/c)})</code>
</p>

<p>where <strong>c &gt; 0</strong> is a kinetic order modifier
(not to be confused with <strong>c</strong> in <code>EXP</code> or <code>EXP+LIN</code>!).
</p>
<p><code>LambertW</code>: tries to fit a dose-response curve based on the Lambert W function
according to Pagonis et al. (2020). The function has the form
</p>
<p style="text-align: center;"><code class="reqn">y ~ (1 + (W((R - 1) * exp(R - 1 - ((x + D_{int}) / D_{c}))) / (1 - R))) * N</code>
</p>

<p>with <code class="reqn">W</code> the Lambert W function, calculated using the package <a href="lamW.html#topic+lamW">lamW::lambertW0</a>,
<code class="reqn">R</code> the dimensionless retrapping ratio, <code class="reqn">N</code> the total concentration
of trappings states in cm^-3 and <code class="reqn">D_{c} = N/R</code> a constant. <code class="reqn">D_{int}</code> is
the offset on the x-axis. Please note that finding the root in <code>mode = "extrapolation"</code>
is a non-easy task due to the shape of the function and the results might be
unexpected.
</p>
<p><strong>Fit weighting</strong>
</p>
<p>If the option <code>fit.weights =  TRUE</code> is chosen, weights are calculated using
provided signal errors (Lx/Tx error):
</p>
<p style="text-align: center;"><code class="reqn">fit.weights = \frac{\frac{1}{error}}{\Sigma{\frac{1}{error}}}</code>
</p>

<p><strong>Error estimation using Monte Carlo simulation</strong>
</p>
<p>Error estimation is done using a parametric bootstrapping approach. A set of
<code>Lx/Tx</code> values is constructed by randomly drawing curve data sampled from normal
distributions. The normal distribution is defined by the input values (<code>mean = value</code>, <code>sd = value.error</code>). Then, a dose-response curve fit is attempted for each
dataset resulting in a new distribution of single <code>De</code> values. The standard
deviation of this distribution becomes then the error of the <code>De</code>. With increasing
iterations, the error value becomes more stable. However, naturally the error
will not decrease with more MC runs.
</p>
<p>Alternatively, the function returns highest probability density interval
estimates as output, users may find more useful under certain circumstances.
</p>
<p><strong>Note:</strong> It may take some calculation time with increasing MC runs,
especially for the composed functions (<code>EXP+LIN</code> and <code>EXP+EXP</code>).<br />
Each error estimation is done with the function of the chosen fitting method.
</p>


<h3>Value</h3>

<p>An <code>RLum.Results</code> object is returned containing the slot <code>data</code> with the
following elements:
</p>
<p><strong>Overview elements</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>DATA.OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>DESCRIPTION</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$De</code> : </td><td style="text-align: left;">  <code>data.frame</code> </td><td style="text-align: left;"> Table with De values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$De.MC</code> : </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> Table with De values from MC runs </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$Fit</code> : </td><td style="text-align: left;"> <a href="stats.html#topic+nls">nls</a> or <a href="stats.html#topic+lm">lm</a> </td><td style="text-align: left;"> object from the fitting for <code>EXP</code>, <code>EXP+LIN</code> and <code>EXP+EXP</code>.
In case of a resulting  linear fit when using <code>LIN</code>, <code>QDR</code> or <code style="white-space: pre;">&#8288;EXP OR LIN&#8288;</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..Fit.Args</code> : </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> Arguments to the function </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$Formula</code> : </td><td style="text-align: left;"> <a href="base.html#topic+expression">expression</a> </td><td style="text-align: left;"> Fitting formula as R expression </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>..$call</code> : </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> The original function call</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>If <code>object</code> is a list, then the function returns a list of <code>RLum.Results</code>
objects as defined above.
</p>
<p><strong>Details - <code>DATA.OBJECT$De</code></strong>
This object is a <a href="base.html#topic+data.frame">data.frame</a> with the following columns
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>De</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> equivalent dose </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>De.Error</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> standard error the equivalent dose </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>D01</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> D-naught value, curvature parameter of the exponential </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>D01.ERROR</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> standard error of the D-naught value</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>D02</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> 2nd D-naught value, only for <code>EXP+EXP</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>D02.ERROR</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> standard error for 2nd D-naught; only for <code>EXP+EXP</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Dc</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> value indicating saturation level; only for <code>LambertW</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n_N</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> saturation level of dose-response curve derived via integration from the used function; it compares the full integral of the curves (<code>N</code>) to the integral until <code>De</code> (<code>n</code>) (e.g.,  Guralnik et al., 2015)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>De.MC</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> equivalent dose derived by Monte-Carlo simulation; ideally identical to <code>De</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>De.plot</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> equivalent dose use for plotting </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Fig</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> applied fit function </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HPDI68_L</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> highest probability density of approximated equivalent dose probability curve representing the lower boundary of 68% probability </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HPDI68_U</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> same as <code>HPDI68_L</code> for the upper bound </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HPDI95_L</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> same as <code>HPDI68_L</code> but for 95% probability </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>HPDI95_U</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> same as <code>HPDI95_L</code> but for the upper bound </td>
</tr>
<tr>
 <td style="text-align: left;">

</td>
</tr>

</table>



<h3>Function version</h3>

<p>1.2.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., Colombo, M., 2025. fit_DoseResponseCurve(): Fit a dose-response curve for luminescence data (Lx/Tx against dose). Function version 1.2.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany) <br />
Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Berger, G.W., Huntley, D.J., 1989. Test data for exponential fits. Ancient TL 7, 43-46.
</p>
<p>Guralnik, B., Li, B., Jain, M., Chen, R., Paris, R.B., Murray, A.S., Li, S.-H., Pagonis, P.,
Herman, F., 2015. Radiation-induced growth and isothermal decay of infrared-stimulated luminescence
from feldspar. Radiation Measurements 81, 224-231.
</p>
<p>Pagonis, V., Kitis, G., Chen, R., 2020. A new analytical equation for the dose response of dosimetric materials,
based on the Lambert W function. Journal of Luminescence 225, 117333. <a href="https://doi.org/10.1016/j.jlumin.2020.117333">doi:10.1016/j.jlumin.2020.117333</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+plot_GrowthCurve">plot_GrowthCurve</a>, <a href="stats.html#topic+nls">nls</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+get_RLum">get_RLum</a>,
<a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>, <a href="stats.html#topic+lm">lm</a>, <a href="stats.html#topic+uniroot">uniroot</a>, <a href="lamW.html#topic+lamW">lamW::lambertW0</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) fit growth curve for a dummy data.set and show De value
data(ExampleData.LxTxData, envir = environment())
temp &lt;- fit_DoseResponseCurve(LxTxData)
get_RLum(temp)

##(1b) to access the fitting value try
get_RLum(temp, data.object = "Fit")

##(2) fit using the 'extrapolation' mode
LxTxData[1,2:3] &lt;- c(0.5, 0.001)
print(fit_DoseResponseCurve(LxTxData, mode = "extrapolation"))

##(3) fit using the 'alternate' mode
LxTxData[1,2:3] &lt;- c(0.5, 0.001)
print(fit_DoseResponseCurve(LxTxData, mode = "alternate"))

##(4) import and fit test data set by Berger &amp; Huntley 1989
QNL84_2_unbleached &lt;-
read.table(system.file("extdata/QNL84_2_unbleached.txt", package = "Luminescence"))

results &lt;- fit_DoseResponseCurve(
 QNL84_2_unbleached,
 mode = "extrapolation",
 verbose = FALSE)

#calculate confidence interval for the parameters
#as alternative error estimation
confint(results$Fit, level = 0.68)


## Not run: 
QNL84_2_bleached &lt;-
read.table(system.file("extdata/QNL84_2_bleached.txt", package = "Luminescence"))
STRB87_1_unbleached &lt;-
read.table(system.file("extdata/STRB87_1_unbleached.txt", package = "Luminescence"))
STRB87_1_bleached &lt;-
read.table(system.file("extdata/STRB87_1_bleached.txt", package = "Luminescence"))

print(
 fit_DoseResponseCurve(
 QNL84_2_bleached,
 mode = "alternate",
 verbose = FALSE)$Fit)

print(
 fit_DoseResponseCurve(
 STRB87_1_unbleached,
 mode = "alternate",
 verbose = FALSE)$Fit)

print(
 fit_DoseResponseCurve(
 STRB87_1_bleached,
 mode = "alternate",
 verbose = FALSE)$Fit)
 
## End(Not run)

</code></pre>

<hr>
<h2 id='fit_EmissionSpectra'>Luminescence Emission Spectra Deconvolution</h2><span id='topic+fit_EmissionSpectra'></span>

<h3>Description</h3>

<p>Luminescence spectra deconvolution on <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> and <a href="base.html#topic+matrix">matrix</a> objects
on an <strong>energy scale</strong>. The function is optimised for emission spectra typically
obtained in the context of TL, OSL and RF measurements detected between 200 and 1000 nm.
The function is not prepared to deconvolve TL curves (counts against temperature;
no wavelength scale). If you are interested in such analysis, please check, e.g.,
the package <code>'tgcd'</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_EmissionSpectra(
  object,
  frame = NULL,
  n_components = NULL,
  start_parameters = NULL,
  sub_negative = 0,
  input_scale = NULL,
  method_control = list(),
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_EmissionSpectra_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>): input
object. Please note that an energy spectrum is expected</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_frame">frame</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>): number of the frame to be analysed. If
<code>NULL</code>, all available frames are analysed.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_n_components">n_components</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>): maximum number of components
desired: the number of component actually fitted may be smaller than this.
Can be combined with other parameters.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_start_parameters">start_parameters</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): allows to provide own start parameters for a
semi-automated procedure. Parameters need to be provided in eV. Every value provided replaces a
value from the automated peak finding algorithm (in ascending order).</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_sub_negative">sub_negative</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): substitute negative values in the input object
by the number provided here (default: <code>0</code>). Can be set to <code>NULL</code>, i.e. negative values are kept.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_input_scale">input_scale</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): defines whether your x-values
are expressed as wavelength or energy values. Allowed values are <code>"wavelength"</code>,
<code>"energy"</code> or <code>NULL</code>, in which case the function tries to guess the input
automatically.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>): options to control the fit method
and the output produced, see details.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="fit_EmissionSpectra_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to control the plot output
(supported: <code>main</code>, <code>xlab</code>, <code>ylab</code>, <code>xlim</code>, <code>ylim</code>, <code>log</code>, <code>mtext</code>, <code>legend</code> (<code>TRUE</code> or <code>FALSE</code>),
<code>legend.text</code>, <code>legend.pos</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Used equation</strong>
</p>
<p>The emission spectra (on an energy scale) can be best described as the sum of multiple
Gaussian components:
</p>
<p>'</p>
<p style="text-align: center;"><code class="reqn">
y = \Sigma  Ci * 1/(\sigma_{i} * \sqrt(2 * \pi)) * exp(-1/2 * ((x - \mu_{i})/\sigma_{i}))^2)
</code>
</p>

<p>with the parameters <code class="reqn">\sigma</code> (peak width) and <code class="reqn">\mu</code> (peak centre) and <code class="reqn">C</code>
(scaling factor).
</p>
<p><strong>Start parameter estimation and fitting algorithm</strong>
</p>
<p>The spectrum deconvolution consists of the following steps:
</p>

<ol>
<li><p> Peak finding <br />
</p>
</li>
<li><p> Start parameter estimation <br />
</p>
</li>
<li><p> Fitting via <a href="minpack.lm.html#topic+nls.lm">minpack.lm::nls.lm</a><br />
</p>
</li></ol>

<p>The peak finding is realised by an approach (re-)suggested by Petr Pikal via the R-help
mailing list (<code style="white-space: pre;">&#8288;https://stat.ethz.ch/pipermail/r-help/2005-November/thread.html&#8288;</code>) in November 2005.
This goes back to even earlier discussion in 2001 based on Prof Brian Ripley's idea.
It smartly uses the functions <a href="stats.html#topic+embed">stats::embed</a> and <a href="base.html#topic+max.col">max.col</a> to identify peaks positions.
For the use in this context, the algorithm has been further modified to scale on the
input data resolution (cf. source code).<br />
</p>
<p>The start parameter estimation uses random sampling from a range of meaningful parameters
and repeats the fitting until 1000 successful fits have been produced or the set <code>max.runs</code> value
is exceeded.
</p>
<p>Currently the best fit is the one with the lowest number for squared residuals, but
other parameters are returned as well. If a series of curves needs to be analysed,
it is recommended to make few trial runs, then fix the number of components and
run  at least 10,000 iterations (parameter <code>method_control = list(max.runs = 10000)</code>).
</p>
<p><strong>Supported <code>method_control</code> settings</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Parameter</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Default</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>max.runs</code> </td><td style="text-align: left;"> <a href="base.html#topic+integer">integer</a> </td><td style="text-align: left;"> <code>10000</code> </td><td style="text-align: left;"> maximum allowed search iterations, if exceed
the searching stops </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>graining</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> <code>15</code> </td><td style="text-align: left;"> control over how coarse or fine the spectrum is split into search intervals for the peak finding algorithm </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>norm</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>TRUE</code> </td><td style="text-align: left;"> normalise data to the highest count value before fitting </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>export.plot.data</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> enable/disable export
of the values of all curves in the plot for each frame analysed </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trace</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> <code>FALSE</code> </td><td style="text-align: left;"> enable/disable the tracing of the minimisation routine
</td>
</tr>

</table>



<h3>Value</h3>

<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the final fit matrix </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit&#8288;</code> </td><td style="text-align: left;"> <code>nls</code> </td><td style="text-align: left;"> the fit object returned by <a href="minpack.lm.html#topic+nls.lm">minpack.lm::nls.lm</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit_info&#8288;</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> a few additional parameters that can be used
to assess the quality of the fit </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$df_plot&#8288;</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> values of all curves in the plot for each
frame analysed (only if <code>method_control$export.plot.data = TRUE</code>)
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ TERMINAL OUTPUT ]&#8288;</code>   <br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>The terminal output provides brief information on the
deconvolution process and the obtained results.
Terminal output is only shown when <code>verbose = TRUE</code>.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code>      <br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>The function returns a plot showing the raw signal with the
detected components. If the fitting failed, a basic plot is returned
showing the raw data and indicating the peaks detected for the start
parameter estimation. The grey band in the residual plot indicates the
10% deviation from 0 (means no residual).
</p>


<h3>Function version</h3>

<p>0.1.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Colombo, M., 2025. fit_EmissionSpectra(): Luminescence Emission Spectra Deconvolution. Function version 0.1.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+plot_RLum">plot_RLum</a>,
<a href="#topic+convert_Wavelength2Energy">convert_Wavelength2Energy</a>, <a href="minpack.lm.html#topic+nls.lm">minpack.lm::nls.lm</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load example data
data(ExampleData.XSYG, envir = environment())

##subtract background
TL.Spectrum@data &lt;- TL.Spectrum@data[] - TL.Spectrum@data[,15]

results &lt;- fit_EmissionSpectra(
 object = TL.Spectrum,
 frame = 5,
 method_control = list(max.runs = 10)
)

##deconvolution of a TL spectrum
## Not run: 

##load example data

##replace 0 values
results &lt;- fit_EmissionSpectra(
 object = TL.Spectrum,
 frame = 5, main = "TL spectrum"
)


## End(Not run)

</code></pre>

<hr>
<h2 id='fit_LMCurve'>Non-linear Least Squares Fit for LM-OSL curves</h2><span id='topic+fit_LMCurve'></span>

<h3>Description</h3>

<p>The function determines weighted non-linear least-squares estimates of the
component parameters of an LM-OSL curve (Bulur 1996) for a given number of
components and returns various component parameters. The fitting procedure
uses the function <a href="stats.html#topic+nls">nls</a> with the <code>port</code> algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_LMCurve(
  values,
  values.bg,
  n.components = 3,
  start_values,
  input.dataType = "LM",
  fit.method = "port",
  sample_code = "",
  sample_ID = "",
  LED.power = 36,
  LED.wavelength = 470,
  fit.trace = FALSE,
  fit.advanced = FALSE,
  fit.calcError = FALSE,
  bg.subtraction = "polynomial",
  verbose = TRUE,
  plot = TRUE,
  plot.BG = FALSE,
  method_control = list(),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_LMCurve_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
x,y data of measured values (time and counts). See examples.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_values.bg">values.bg</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> or <a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
x,y data of measured values (time and counts) for background subtraction.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_n.components">n.components</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
fixed number of components that are to be recognised during fitting
(min = 1, max = 7).</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_start_values">start_values</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>optional</em>):
start parameters for <code>lm</code> and <code>xm</code> data for the fit. If no start values are given,
an automatic start value estimation is attempted (see details).</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_input.datatype">input.dataType</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
alter the plot output depending on the input data: <code>"LM"</code> or <code>"pLM"</code> (pseudo-LM).
See: <a href="#topic+convert_CW2pLM">convert_CW2pLM</a></p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_fit.method">fit.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
select fit method, allowed values: <code>'port'</code> and <code>'LM'</code>. <code>'port'</code> uses the 'port'
routine from the function <a href="stats.html#topic+nls">nls</a> <code>'LM'</code> utilises the function <code>nlsLM</code> from
the package <code>minpack.lm</code> and with that the Levenberg-Marquardt algorithm.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_sample_code">sample_code</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
sample code used for the plot and the optional output table (mtext).</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_sample_id">sample_ID</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
additional identifier used as column header for the table output.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_led.power">LED.power</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
LED power (max.) used for intensity ramping in mW/cm¬≤.
<strong>Note:</strong> This value is used for the calculation of the absolute
photoionisation cross section.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_led.wavelength">LED.wavelength</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
LED wavelength in nm used for stimulation.
<strong>Note:</strong> This value is used for the calculation of the absolute
photoionisation cross section.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_fit.trace">fit.trace</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
traces the fitting process on the terminal.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_fit.advanced">fit.advanced</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enables advanced fitting attempt for automatic start parameter recognition.
Works only if no start parameters are provided.
<strong>Note:</strong> It may take a while and it is not compatible with <code>fit.method = "LM"</code>.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_fit.calcerror">fit.calcError</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
calculate 1-sigma error range of components using <a href="stats.html#topic+confint">stats::confint</a>.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_bg.subtraction">bg.subtraction</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
specifies method for background subtraction (<code>polynomial</code>, <code>linear</code>, <code>channel</code>,
see Details). <strong>Note:</strong> requires input for <code>values.bg</code>.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_plot.bg">plot.BG</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
returns a plot of the background values with the fit used for the
background subtraction.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>): options to control the output
produced. Currently only the 'export.comp.contrib.matrix' (logical) option
is supported, to enable/disable export of the component contribution
matrix.</p>
</td></tr>
<tr><td><code id="fit_LMCurve_+3A_...">...</code></td>
<td>
<p>Further arguments that may be passed to the plot output, e.g.
<code>xlab</code>, <code>xlab</code>, <code>main</code>, <code>log</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Fitting function</strong>
</p>
<p>The function for the fitting has the general
form:
</p>
<p style="text-align: center;"><code class="reqn">y = (exp(0.5)*Im_1*x/xm_1)*exp(-x^2/(2*xm_1^2)) + ,\ldots, + exp(0.5)*Im_i*x/xm_i)*exp(-x^2/(2*xm_i^2))</code>
</p>

<p>where <code class="reqn">1 &lt; i &lt; 8</code>
</p>
<p>This function and the equations for the conversion to b (detrapping probability)
and n0 (proportional to initially trapped charge) have been taken from Kitis
et al. (2008):
</p>
<p style="text-align: center;"><code class="reqn">xm_i=\sqrt{max(t)/b_i}</code>
</p>

<p style="text-align: center;"><code class="reqn">Im_i=exp(-0.5)n0/xm_i</code>
</p>

<p><strong>Background subtraction</strong>
</p>
<p>Three methods for background subtraction
are provided for a given background signal (<code>values.bg</code>).
</p>

<ul>
<li> <p><code>polynomial</code>: default method. A polynomial function is fitted using <a href="stats.html#topic+glm">glm</a>
and the resulting function is used for background subtraction:
</p>
<p style="text-align: center;"><code class="reqn">y = a*x^4 + b*x^3 + c*x^2 + d*x + e</code>
</p>

</li>
<li> <p><code>linear</code>: a linear function is fitted using <a href="stats.html#topic+glm">glm</a> and the resulting function
is used for background subtraction:
</p>
<p style="text-align: center;"><code class="reqn">y = a*x + b</code>
</p>

</li>
<li> <p><code>channel</code>: the measured
background signal is subtracted channel wise from the measured signal.
</p>
</li></ul>

<p><strong>Start values</strong>
</p>
<p>The choice of the initial parameters for the <code>nls</code>-fitting is a crucial
point and the fitting procedure may mainly fail due to ill chosen start
parameters. Here, three options are provided:
</p>
<p><strong>(a)</strong>
If no start values (<code>start_values</code>) are provided by the user, a cheap guess is made
by using the detrapping values found by Jain et al. (2003) for quartz for a
maximum of 7 components. Based on these values, the pseudo start parameters
<code>xm</code> and <code>Im</code> are recalculated for the given data set. In all cases, the fitting
starts with the ultra-fast component and (depending on <code>n.components</code>)
steps through the following values. If no fit could be achieved, an error
plot (for <code>plot = TRUE</code>) with the pseudo curve (based on the
pseudo start parameters) is provided. This may give the opportunity to
identify appropriate start parameters visually.
</p>
<p><strong>(b)</strong>
If start values are provided, the function works like a simple <a href="stats.html#topic+nls">nls</a>
fitting approach.
</p>
<p><strong>(c)</strong>
If no start parameters are provided and
the option <code>fit.advanced = TRUE</code> is chosen, an advanced start parameter
estimation is applied using a stochastic attempt. Therefore, the
recalculated start parameters <strong>(a)</strong> are used to construct a normal
distribution. The start parameters are then sampled randomly from this
distribution. A maximum of 100 attempts will be made. <strong>Note:</strong> This
process may be time consuming.
</p>
<p><strong>Goodness of fit</strong>
</p>
<p>The goodness of the fit is given by a pseudo-R¬≤ value (pseudo coefficient of
determination). According to Lave (1970), the value is calculated as:
</p>
<p style="text-align: center;"><code class="reqn">pseudoR^2 = 1 - RSS/TSS</code>
</p>

<p>where <code class="reqn">RSS = Residual~Sum~of~Squares</code>
and <code class="reqn">TSS = Total~Sum~of~Squares</code>
</p>
<p><strong>Error of fitted component parameters</strong>
</p>
<p>The 1-sigma error for the components is calculated using
the function <a href="stats.html#topic+confint">stats::confint</a>. Due to considerable calculation time, this
option is deactivated by default. In addition, the error for the components
can be estimated by using internal R functions like <a href="base.html#topic+summary">summary</a>. See the
<a href="stats.html#topic+nls">nls</a> help page for more information.
</p>
<p><em>For more details on the nonlinear regression in R, see Ritz &amp; Streibig (2008).</em>
</p>


<h3>Value</h3>

<p>Various types of plots are returned. For details see above. Furthermore an
<code>RLum.Results</code> object is returned with the following structure:
</p>
<p><strong><code style="white-space: pre;">&#8288;@data:&#8288;</code></strong>
</p>
<p><code>.. $data</code> : <a href="base.html#topic+data.frame">data.frame</a> with fitting results<br />
<code>.. $fit</code> : nls (<a href="stats.html#topic+nls">nls</a> object)<br />
<code>.. $component_matrix</code> : <a href="base.html#topic+matrix">matrix</a> with numerical xy-values of the single fitted components with the resolution of the input data
<code>.. $component.contribution.matrix</code> : <a href="base.html#topic+list">list</a> component distribution matrix
(produced only if <code>method_control$export.comp.contrib.matrix = TRUE</code>)
</p>
<p><strong><code style="white-space: pre;">&#8288;info:&#8288;</code></strong>
</p>
<p><code>.. $call</code> : <a href="base.html#topic+call">call</a> the original function call
</p>
<p>Matrix structure for the distribution matrix:
</p>
<p>Column 1 and 2: time and <code>rev(time)</code> values<br />
Additional columns are used for the components, two for each component,
containing I0 and n0. The last columns <code>cont.</code> provide information on
the relative component contribution for each time interval including the row
sum for this values.
</p>


<h3>Function version</h3>

<p>0.3.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. fit_LMCurve(): Non-linear Least Squares Fit for LM-OSL curves. Function version 0.3.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The pseudo-R¬≤ may not be the best parameter to describe the goodness
of the fit. The trade off between the <code>n.components</code> and the pseudo-R¬≤
value currently remains unconsidered.
</p>
<p>The function <strong>does not</strong> ensure that the fitting procedure has reached a
global minimum rather than a local minimum! In any case of doubt, the use of
manual start values is highly recommended.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Bulur, E., 1996. An Alternative Technique For Optically
Stimulated Luminescence (OSL) Experiment. Radiation Measurements, 26, 5,
701-709.
</p>
<p>Jain, M., Murray, A.S., Boetter-Jensen, L., 2003. Characterisation of
blue-light stimulated luminescence components in different quartz samples:
implications for dose measurement. Radiation Measurements, 37 (4-5),
441-449.
</p>
<p>Kitis, G. &amp; Pagonis, V., 2008. Computerized curve deconvolution analysis for
LM-OSL. Radiation Measurements, 43, 737-741.
</p>
<p>Lave, C.A.T., 1970. The Demand for Urban Mass Transportation. The Review of
Economics and Statistics, 52 (3), 320-323.
</p>
<p>Ritz, C. &amp; Streibig, J.C., 2008. Nonlinear Regression with R. R. Gentleman,
K. Hornik, &amp; G. Parmigiani, eds., Springer, p. 150.
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_CWCurve">fit_CWCurve</a>, <a href="base.html#topic+plot">plot</a>, <a href="stats.html#topic+nls">nls</a>, <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>, <a href="#topic+get_RLum">get_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) fit LM data without background subtraction
data(ExampleData.FittingLM, envir = environment())
fit_LMCurve(values = values.curve, n.components = 3, log = "x")

##(2) fit LM data with background subtraction and export as JPEG
## -alter file path for your preferred system
##jpeg(file = "~/Desktop/Fit_Output\%03d.jpg", quality = 100,
## height = 3000, width = 3000, res = 300)
data(ExampleData.FittingLM, envir = environment())
fit_LMCurve(values = values.curve, values.bg = values.curveBG,
            n.components = 2, log = "x", plot.BG = TRUE)
##dev.off()

##(3) fit LM data with manual start parameters
data(ExampleData.FittingLM, envir = environment())
fit_LMCurve(values = values.curve,
            values.bg = values.curveBG,
            n.components = 3,
            log = "x",
            start_values = data.frame(Im = c(170,25,400), xm = c(56,200,1500)))

</code></pre>

<hr>
<h2 id='fit_OSLLifeTimes'>Fitting and Deconvolution of OSL Lifetime Components</h2><span id='topic+fit_OSLLifeTimes'></span>

<h3>Description</h3>

<p>Fitting and Deconvolution of OSL Lifetime Components
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_OSLLifeTimes(
  object,
  tp = 0,
  signal_range = NULL,
  n.components = NULL,
  method_control = list(),
  plot = TRUE,
  plot_simple = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_OSLLifeTimes_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="base.html#topic+data.frame">data.frame</a> or <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>):
Input object containing the data to be analysed. All objects can be provided also as list for an automated
processing. Please note: <code>NA</code> values are automatically removed and the dataset should comprise at least 5 data points (possibly more if <code>n.components</code> is
set to a value greater than 1)</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_tp">tp</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): option to account for the stimulation pulse width. For off-time measurements
the default value is 0. <code>tp</code> has the same unit as the measurement data, e.g., ¬µs. Please set this parameter
carefully, if it all, otherwise you may heavily bias your fit results.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_signal_range">signal_range</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): allows to set a channel range, by default all channels are used, e.g.
<code>signal_range = c(2,100)</code> considers only channels 2 to 100 and <code>signal_range = c(2)</code> considers only channels
from channel 2 onwards.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_n.components">n.components</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): Fix the number of components. If set the algorithm will try
to fit the number of predefined components. If nothing is set, the algorithm will try to find the best number
of components.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>): Named to allow a more fine control of the fitting process. See details
for allowed options.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_plot_simple">plot_simple</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Enable/disable the reduced
plot output. If <code>TRUE</code>, no
residual plot is shown, however, plot output can be combined using the standard R layout options,
such as <code>par(mfrow = c(2,2))</code>.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="fit_OSLLifeTimes_+3A_...">...</code></td>
<td>
<p>parameters passed to <a href="graphics.html#topic+plot.default">plot.default</a> to control the plot output, supported are:
<code>main</code>, <code>xlab</code>, <code>ylab</code>, <code>log</code>, <code>xlim</code>, <code>ylim</code>, <code>col</code>, <code>lty</code>, <code>legend.pos</code>, <code>legend.text</code>. If the input
object is of type <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> this arguments can be provided as a <a href="base.html#topic+list">list</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function intends to provide an easy access to pulsed optically stimulated luminescence (POSL) data,
in order determine signal lifetimes. The fitting is currently optimised to work with the off-time flank of POSL measurements
only. For the signal deconvolution, a differential evolution optimisation is combined with nonlinear least-square fitting
following the approach by Bluszcz &amp; Adamiec (2006).
</p>
<p><strong>Component deconvolution algorithm</strong>
</p>
<p>The component deconvolution consists of two steps:
</p>
<p>(1) Adaptation phase
</p>
<p>In the adaptation phase the function tries to figure out the optimal and statistically justified
number of signal components following roughly the approach suggested by Bluszcz &amp; Adamiec (2006). In
contrast to their work, for the optimisation by differential evolution here the package <code>'DEoptim'</code> is used.
</p>
<p>The function to be optimized has the form:
</p>
<p style="text-align: center;"><code class="reqn">\chi^2 = \sum(w * (n_i/c - \sum(A_i * exp(-x/(tau_i + t_p))))^2)</code>
</p>

<p>with <code class="reqn">w = 1</code> for unweighted regression analysis (<code>method_control = list(weights = FALSE)</code>) or
<code class="reqn">w = c^2/n_i</code> for weighted regression analysis. The default values is <code>TRUE</code>.
</p>
<p style="text-align: center;"><code class="reqn">F = (\Delta\chi^2 / 2) / (\chi^2/(N - 2*m - 2))</code>
</p>

<p>(2) Final fitting
</p>
<p><strong><code>method_control</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Parameter</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> controls the probability for the F statistic reference values. For a significance level of 5 % a value of 0.95 (the default) should be added, for 1 %, a value of 0.99 is sufficient: 1 &gt; p &gt; 0 (cf. <a href="stats.html#topic+Fdist">stats::FDist</a>)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seed</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> set the seed for the random number generator, provide a value here to get reproducible results </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DEoptim.trace</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> enable/disable the tracing of the differential evolution (cf. <a href="DEoptim.html#topic+DEoptim.control">DEoptim::DEoptim.control</a>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>DEoptim.itermax</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> control the number of the allowed generations (cf. <a href="DEoptim.html#topic+DEoptim.control">DEoptim::DEoptim.control</a>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>weights</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> enable/disable the weighting for the start parameter estimation and fitting (see equations above).
The default values is <code>TRUE</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nlsLM.trace</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> enable/disable trace mode for the nls fitting (<a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>), can be used to identify convergence problems, default is <code>FALSE</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nlsLM.upper</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> enable/disable upper parameter boundary, default is <code>TRUE</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nlsLM.lower</code> </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> enable/disable lower parameter boundary, default is <code>TRUE</code>
</td>
</tr>

</table>



<h3>Value</h3>

<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the final fit matrix </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$start_matrix&#8288;</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the start matrix used for the fitting </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$total_counts&#8288;</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> Photon count sum </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit&#8288;</code> </td><td style="text-align: left;"> <code>nls</code> </td><td style="text-align: left;"> the fit object returned by <a href="minpack.lm.html#topic+nls.lm">minpack.lm::nls.lm</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ TERMINAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>Terminal output is only shown of the argument <code>verbose = TRUE</code>.
</p>
<p><em>(1) Start parameter and component adaption</em><br />
Trave of the parameter adaptation process
</p>
<p><em>(2) Fitting results (sorted by ascending tau)</em><br />
The fitting results sorted by ascending tau value. Please note
that if you access the <code>nls</code> fitting object, the values are not sorted.
</p>
<p><em>(3) Further information</em><br />
</p>

<ul>
<li><p> The photon count sum
</p>
</li>
<li><p> Durbin-Watson residual statistic to assess whether the residuals are correlated, ideally
the residuals should be not correlated at all. Rough measures are: <br />
D = 0: the residuals are systematically correlated <br />
D = 2: the residuals are randomly distributed <br />
D = 4: the residuals are systematically anti-correlated<br />
</p>
</li></ul>

<p>You should be suspicious if D differs largely from 2.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>A plot showing the original data and the fit so far possible. The lower plot shows the
residuals of the fit.
</p>


<h3>Function version</h3>

<p>0.1.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Schmidt, C., 2025. fit_OSLLifeTimes(): Fitting and Deconvolution of OSL Lifetime Components. Function version 0.1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Geography &amp; Earth Sciences, Aberystwyth University,
Christoph Schmidt, University of Bayreuth (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Bluszcz, A., Adamiec, G., 2006. Application of differential evolution to fitting OSL decay curves.
Radiation Measurements 41, 886-891. <a href="https://doi.org/10.1016/j.radmeas.2006.05.016">doi:10.1016/j.radmeas.2006.05.016</a><br />
</p>
<p>Durbin, J., Watson, G.S., 1950. Testing for Serial Correlation in Least Squares Regression: I.
Biometrika 37, 409-21. doi:10.2307/2332391
</p>
<p><strong>Further reading</strong>
</p>
<p>Hughes, I., Hase, T., 2010. Measurements and Their Uncertainties. Oxford University Press.
</p>
<p>Storn, R., Price, K., 1997. Differential Evolution ‚Äì
A Simple and Efficient Heuristic for Global Optimization over Continuous Spaces.
Journal of Global Optimization 11, 341‚Äì359.
</p>


<h3>See Also</h3>

<p><a href="minpack.lm.html#topic+nls.lm">minpack.lm::nls.lm</a>, <a href="DEoptim.html#topic+DEoptim">DEoptim::DEoptim</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load example data
data(ExampleData.TR_OSL, envir = environment())

##fit lifetimes (short run)
fit_OSLLifeTimes(
 object = ExampleData.TR_OSL,
 n.components = 1)

##long example
## Not run: 
fit_OSLLifeTimes(
object = ExampleData.TR_OSL)

## End(Not run)

</code></pre>

<hr>
<h2 id='fit_SurfaceExposure'>Nonlinear Least Squares Fit for OSL surface exposure data</h2><span id='topic+fit_SurfaceExposure'></span>

<h3>Description</h3>

<p>This function determines the (weighted) least-squares estimates of the
parameters of either equation 1 in <em>Sohbati et al. (2012a)</em> or equation 12 in
<em>Sohbati et al. (2012b)</em> for a given OSL surface exposure data set (<strong>BETA</strong>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_SurfaceExposure(
  data,
  sigmaphi = NULL,
  mu = NULL,
  age = NULL,
  Ddot = NULL,
  D0 = NULL,
  weights = FALSE,
  plot = TRUE,
  legend = TRUE,
  error_bars = TRUE,
  coord_flip = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_SurfaceExposure_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
Measured OSL surface exposure data with the following structure:
</p>
<div class="sourceCode"><pre>                               (optional)
     | depth (a.u.)| intensity | error |
     |     [ ,1]   |    [ ,2]  | [ ,3] |
     |-------------|-----------|-------|
[1, ]|    ~~~~     |    ~~~~   | ~~~~  |
[2, ]|    ~~~~     |    ~~~~   | ~~~~  |
 ... |     ...     |     ...   |  ...  |
[x, ]|    ~~~~     |    ~~~~   | ~~~~  |

</pre></div>
<p>Alternatively, a <a href="base.html#topic+list">list</a> of <code>data.frames</code> can be provided, where each
<code>data.frame</code> has the same structure as shown above, with the exception that
they must <strong>not</strong> include the optional error column. Providing a <a href="base.html#topic+list">list</a> as
input automatically activates the global fitting procedure (see details).</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_sigmaphi">sigmaphi</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
A numeric value for <code>sigmaphi</code>, i.e. the charge detrapping rate.
Example: <code>sigmaphi = 5e-10</code></p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_mu">mu</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
A numeric value for mu, i.e. the light attenuation coefficient.
Example: <code>mu = 0.9</code></p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_age">age</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
The age (a) of the sample, if known. If <code>data</code> is a <a href="base.html#topic+list">list</a> of <em>x</em> samples,
then <code>age</code> must be a numeric vector of length <em>x</em>.
Example: <code>age = 10000</code>, or <code>age = c(1e4, 1e5, 1e6)</code>.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_ddot">Ddot</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
A numeric value for the environmental dose rate (Gy/ka). For this argument
to be considered a value for <code>D0</code> must also be provided; otherwise it will be
ignored.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_d0">D0</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
A numeric value for the characteristic saturation dose (Gy). For this argument
to be considered a value for <code>Ddot</code> must also be provided; otherwise it will be
ignored.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_weights">weights</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
If <code>TRUE</code> the fit will be weighted by the inverse square of the error.
Requires <code>data</code> to be a <a href="base.html#topic+data.frame">data.frame</a> with three columns.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_legend">legend</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
Show or hide the equation inside the plot.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_error_bars">error_bars</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
Show or hide error bars (only applies if errors were provided).</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_coord_flip">coord_flip</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
Flip the coordinate system.</p>
</td></tr>
<tr><td><code id="fit_SurfaceExposure_+3A_...">...</code></td>
<td>
<p>Further parameters passed to <a href="base.html#topic+plot">plot</a>.
Custom parameters include:
</p>

<ul>
<li> <p><code>verbose</code> (<a href="base.html#topic+logical">logical</a>): show or hide console output
</p>
</li>
<li> <p><code>line_col</code>: Colour of the fitted line
</p>
</li>
<li> <p><code>line_lty</code>: Type of the fitted line (see <code>lty</code> in <code>?par</code>)
</p>
</li>
<li> <p><code>line_lwd</code>: Line width of the fitted line (see <code>lwd</code> in <code>?par</code>)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Weighted fitting</strong>
</p>
<p>If <code>weights = TRUE</code> the function will use the inverse square of the error (<code class="reqn">1/\sigma^2</code>)
as weights during fitting using <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>. Naturally, for this to
take effect individual errors must be provided in the third column of the
<code>data.frame</code> for <code>data</code>. Weighted fitting is <strong>not</strong> supported if <code>data</code>
is a list of multiple <code>data.frame</code>s, i.e., it is not available for global
fitting.
</p>
<p><strong>Dose rate</strong>
If any of the arguments <code>Ddot</code> or <code>D0</code> is at its default value (<code>NULL</code>),
this function will fit equation 1 in Sohbati et al. (2012a) to the data. If
the effect of dose rate (i.e., signal saturation) needs to be considered,
numeric values for the dose rate (<code>Ddot</code>) (in Gy/ka) and the characteristic
saturation dose (<code>D0</code>) (in Gy) must be provided. The function will then fit
equation 12 in Sohbati et al. (2012b) to the data.
</p>
<p><strong>NOTE</strong>: Currently, this function does <strong>not</strong> consider the variability
of the dose rate with sample depth (<code>x</code>)! In the original equation the dose
rate <code>D</code> is an arbitrary function of <code>x</code> (term <code>D(x)</code>), but here <code>D</code> is assumed
constant.
</p>
<p><strong>Global fitting</strong>
If <code>data</code> is <a href="base.html#topic+list">list</a> of multiple <code>data.frame</code>s, each representing a separate
sample, the function automatically performs a global fit to the data. This
may be useful to better constrain the parameters <code>sigmaphi</code> or <code>mu</code> and
<strong>requires</strong> that known ages for each sample is provided
(e.g., <code>age = c(100, 1000)</code> if <code>data</code> is a list with two samples).
</p>


<h3>Value</h3>

<p>Function returns results numerically and graphically:
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$summary&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> summary of the fitting results </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> the original input data </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$fit&#8288;</code> </td><td style="text-align: left;"> <code>nls</code> </td><td style="text-align: left;"> the fitting object produced by <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$args&#8288;</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> arguments of the call </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$call&#8288;</code> </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>Currently unused.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>A scatter plot of the provided depth-intensity OSL surface exposure data
with the fitted model.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. fit_SurfaceExposure(): Nonlinear Least Squares Fit for OSL surface exposure data. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>This function has BETA status. If possible, results should be</strong>
<strong>cross-checked.</strong>
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Sohbati, R., Murray, A.S., Chapot, M.S., Jain, M., Pederson, J., 2012a.
Optically stimulated luminescence (OSL) as a chronometer for surface exposure
dating. Journal of Geophysical Research 117, B09202. doi:
<a href="https://doi.org/10.1029/2012JB009383">doi:10.1029/2012JB009383</a>
</p>
<p>Sohbati, R., Jain, M., Murray, A.S., 2012b. Surface exposure dating of
non-terrestrial bodies using optically stimulated luminescence: A new method.
Icarus 221, 160-166.
</p>


<h3>See Also</h3>

<p><a href="#topic+ExampleData.SurfaceExposure">ExampleData.SurfaceExposure</a>, <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Load example data
data("ExampleData.SurfaceExposure")

## Example 1 - Single sample
# Known parameters: 10000 a, mu = 0.9, sigmaphi = 5e-10
sample_1 &lt;- ExampleData.SurfaceExposure$sample_1
head(sample_1)
results &lt;- fit_SurfaceExposure(
 data = sample_1,
 mu = 0.9,
 sigmaphi = 5e-10)
get_RLum(results)


## Example 2 - Single sample and considering dose rate
# Known parameters: 10000 a, mu = 0.9, sigmaphi = 5e-10,
# dose rate = 2.5 Gy/ka, D0 = 40 Gy
sample_2 &lt;- ExampleData.SurfaceExposure$sample_2
head(sample_2)
results &lt;- fit_SurfaceExposure(
 data = sample_2,
 mu = 0.9,
 sigmaphi = 5e-10,
 Ddot = 2.5,
 D0 = 40)
get_RLum(results)

## Example 3 - Multiple samples (global fit) to better constrain 'mu'
# Known parameters: ages = 1e3, 1e4, 1e5, 1e6 a, mu = 0.9, sigmaphi = 5e-10
set_1 &lt;- ExampleData.SurfaceExposure$set_1
str(set_1, max.level = 2)
results &lt;- fit_SurfaceExposure(
  data = set_1,
  age = c(1e3, 1e4, 1e5, 1e6),
  sigmaphi = 5e-10)
get_RLum(results)


## Example 4 - Multiple samples (global fit) and considering dose rate
# Known parameters: ages = 1e2, 1e3, 1e4, 1e5, 1e6 a, mu = 0.9, sigmaphi = 5e-10,
# dose rate = 1.0 Ga/ka, D0 = 40 Gy
set_2 &lt;- ExampleData.SurfaceExposure$set_2
str(set_2, max.level = 2)
results &lt;- fit_SurfaceExposure(
 data = set_2,
 age = c(1e2, 1e3, 1e4, 1e5, 1e6),
 sigmaphi = 5e-10,
 Ddot = 1,
 D0 = 40)
get_RLum(results)

</code></pre>

<hr>
<h2 id='fit_ThermalQuenching'>Fitting Thermal Quenching Data</h2><span id='topic+fit_ThermalQuenching'></span>

<h3>Description</h3>

<p>Applying a nls-fitting to thermal quenching data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_ThermalQuenching(
  data,
  start_param = list(),
  method_control = list(),
  n.MC = 100,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fit_ThermalQuenching_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): input data with three columns, the first column contains
temperature values in deg. C, columns 2 and 3 the dependent values with its error</p>
</td></tr>
<tr><td><code id="fit_ThermalQuenching_+3A_start_param">start_param</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>): option to provide the start
parameters for the fitting, see details</p>
</td></tr>
<tr><td><code id="fit_ThermalQuenching_+3A_method_control">method_control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>): further options to fine tune
the fitting, see details for
further information</p>
</td></tr>
<tr><td><code id="fit_ThermalQuenching_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): number of Monte Carlo runs for the error estimation. If <code>n.MC</code> is
<code>NULL</code> or <code style="white-space: pre;">&#8288;&lt;=1&#8288;</code>, the error estimation is skipped</p>
</td></tr>
<tr><td><code id="fit_ThermalQuenching_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="fit_ThermalQuenching_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="fit_ThermalQuenching_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to control the plotting, support are <code>main</code>, <code>pch</code>,
<code>col_fit</code>, <code>col_points</code>, <code>lty</code>, <code>lwd</code>, <code>xlab</code>, <code>ylab</code>, <code>xlim</code>, <code>ylim</code>, <code>xaxt</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Used equation</strong><br />
</p>
<p>The equation used for the fitting is
</p>
<p style="text-align: center;"><code class="reqn">y = (A / (1 + C * (exp(-W / (k * x))))) + c</code>
</p>

<p><em>W</em> is the energy depth in eV and <em>C</em> is dimensionless constant. <em>A</em> and <em>c</em> are used to
adjust the curve for the given signal. <em>k</em> is the Boltzmann in eV/K and <em>x</em> is the absolute
temperature in K.
</p>
<p><strong>Error estimation</strong><br />
</p>
<p>The error estimation is done be varying the input parameters using the given uncertainties in
a Monte Carlo simulation. Errors are assumed to follow a normal distribution.
</p>
<p><strong><code>start_param</code></strong> <br />
</p>
<p>The function allows the injection of own start parameters via the argument <code>start_param</code>. The
parameters needs to be provided as names list. The names are the parameters to be optimised.
Examples: <code>start_param = list(A = 1, C = 1e+5, W = 0.5, c = 0)</code>
</p>
<p><strong><code>method_control</code></strong> <br />
</p>
<p>The following arguments can be provided via <code>method_control</code>. Please note that arguments provided
via <code>method_control</code> are not further tested, i.e., if the function crashes your input was probably
wrong.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>ARGUMENT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>DESCRIPTION</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper</code> </td><td style="text-align: left;"> named <a href="base.html#topic+vector">vector</a> </td><td style="text-align: left;"> sets upper fitting boundaries, if provided boundaries for all arguments
are required, e.g., <code>c(A = 0, C = 0, W = 0, c = 0)</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower</code> </td><td style="text-align: left;"> names <a href="base.html#topic+vector">vector</a> </td><td style="text-align: left;"> set lower fitting boundaries (see <code>upper</code> for details) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trace</code>   </td><td style="text-align: left;"> <a href="base.html#topic+logical">logical</a> </td><td style="text-align: left;"> enable/disable progression trace for <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>weights</code> </td><td style="text-align: left;"> <a href="base.html#topic+numeric">numeric</a> </td><td style="text-align: left;"> option to provide own weights for the fitting, the length of this
vector needs to be equal to the number for rows of the input <code>data.frame</code>. If set to <code>NULL</code> no weights
are applied. The weights are defined by the third column of the input <code>data.frame</code>.
</td>
</tr>

</table>



<h3>Value</h3>

<p>The function returns numerical output and an (<em>optional</em>) plot.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>
<p><code style="white-space: pre;">&#8288;[.. $data : data.frame]&#8288;</code><br />
</p>
<p>A table with all fitting parameters and the number of Monte Carlo runs used for the error estimation.
</p>
<p><code style="white-space: pre;">&#8288;[.. $fit : nls object]&#8288;</code> <br />
</p>
<p>The nls <a href="stats.html#topic+nls">stats::nls</a> object returned by the function <a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>. This object
can be further passed to other functions supporting an nls object (cf. details section
in <a href="stats.html#topic+nls">stats::nls</a>)
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p><code style="white-space: pre;">&#8288;[.. $call : call]&#8288;</code><br />
</p>
<p>The original function call.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ GAPHICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p>Plotted are temperature against the signal and their uncertainties.
The fit is shown as dashed-line (can be modified). Please note that for the fitting the absolute
temperature values are used but are re-calculated to deg. C for the plot.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. fit_ThermalQuenching(): Fitting Thermal Quenching Data. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Wintle, A.G., 1975. Thermal Quenching of Thermoluminescence in Quartz. Geophys. J. R. astr. Soc. 41, 107‚Äì113.
</p>


<h3>See Also</h3>

<p><a href="minpack.lm.html#topic+nlsLM">minpack.lm::nlsLM</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##create short example dataset
data &lt;- data.frame(
  T = c(25, 40, 50, 60, 70, 80, 90, 100, 110),
  V = c(0.06, 0.058, 0.052, 0.051, 0.041, 0.034, 0.035, 0.033, 0.032),
  V_X = c(0.012, 0.009, 0.008, 0.008, 0.007, 0.006, 0.005, 0.005, 0.004))

##fit
fit_ThermalQuenching(
 data = data,
 n.MC = NULL)

</code></pre>

<hr>
<h2 id='get_Layout'>Collection of layout definitions</h2><span id='topic+get_Layout'></span>

<h3>Description</h3>

<p>This helper function returns a list with layout definitions for homogeneous
plotting.
</p>
<p>The easiest way to create a user-specific layout definition is perhaps to
create either an empty or a default layout object and fill/modify the
definitions (<code>user.layout &lt;- get_Layout(data = "empty")</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Layout(layout)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_Layout_+3A_layout">layout</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+list">list</a> object (<strong>required</strong>):
name of the layout definition to be returned. If name is provided the
respective definition is returned. One of the following
supported layout definitions is possible: <code>"default"</code>,
<code>"journal.1"</code>, <code>"small"</code>, <code>"empty"</code>.
</p>
<p>User-specific layout definitions must be provided as a list object of
predefined structure, see details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object with layout definitions for plot functions.
</p>


<h3>Function version</h3>

<p>0.1
</p>


<h3>How to cite</h3>

<p>Dietze, M., 2025. get_Layout(): Collection of layout definitions. Function version 0.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)
, RLum Developer Team</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## read example data set
data(ExampleData.DeValues, envir = environment())

## show structure of the default layout definition
layout.default &lt;- get_Layout(layout = "default")
str(layout.default)

## show colour definitions for Abanico plot, only
layout.default$abanico$colour

## set Abanico plot title colour to orange
layout.default$abanico$colour$main &lt;- "orange"

## create Abanico plot with modofied layout definition
plot_AbanicoPlot(data = ExampleData.DeValues,
                 layout = layout.default)

## create Abanico plot with predefined layout "journal"
plot_AbanicoPlot(data = ExampleData.DeValues,
                 layout = "journal")

</code></pre>

<hr>
<h2 id='get_Quote'>Function to return essential quotes</h2><span id='topic+get_Quote'></span>

<h3>Description</h3>

<p>This function returns one of the collected essential quotes in the
growing library. If called without any parameters, a random quote is
returned.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Quote(ID, separated = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_Quote_+3A_id">ID</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): quote ID to be returned.</p>
</td></tr>
<tr><td><code id="get_Quote_+3A_separated">separated</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): return result in separated form.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a character with quote and respective (false) author.
</p>


<h3>Function version</h3>

<p>0.1.5
</p>


<h3>How to cite</h3>

<p>Dietze, M., Kreutzer, S., 2025. get_Quote(): Function to return essential quotes. Function version 0.1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Quote credits: Michael Dietze, GFZ Potsdam (Germany), Sebastian Kreutzer, Geography &amp; Earth Science, Aberystwyth University (United Kingdom), Dirk Mittelstra√ü, TU Dresden (Germany), Jakob Wallinga (Wageningen University, Netherlands)
, RLum Developer Team</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## ask for an arbitrary quote
get_Quote()

</code></pre>

<hr>
<h2 id='get_rightAnswer'>Function to get the right answer</h2><span id='topic+get_rightAnswer'></span>

<h3>Description</h3>

<p>This function returns just the right answer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_rightAnswer(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_rightAnswer_+3A_...">...</code></td>
<td>
<p>you can pass an infinite number of further arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the right answer
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>NA, NA, , , 2025. get_rightAnswer(): Function to get the right answer. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>inspired by R.G.
, RLum Developer Team</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## you really want to know?
get_rightAnswer()

</code></pre>

<hr>
<h2 id='get_Risoe.BINfileData'>General accessor function for Risoe.BINfileData objects</h2><span id='topic+get_Risoe.BINfileData'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects. Depending on the input object, the
corresponding function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Risoe.BINfileData(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_Risoe.BINfileData_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
S4 object of class <code>Risoe.BINfileData</code></p>
</td></tr>
<tr><td><code id="get_Risoe.BINfileData_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same type as the input object provided.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. get_Risoe.BINfileData(): General accessor function for Risoe.BINfileData objects. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>

<hr>
<h2 id='get_RLum'>General accessor function for RLum S4 class objects</h2><span id='topic+get_RLum'></span><span id='topic+get_RLum+2Clist-method'></span><span id='topic+get_RLum+2CNULL-method'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum-class">RLum</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_RLum(object, ...)

## S4 method for signature 'list'
get_RLum(object, class = NULL, null.rm = FALSE, ...)

## S4 method for signature 'NULL'
get_RLum(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code> or an object of type <a href="base.html#topic+list">list</a> containing only objects
of type <a href="#topic+RLum-class">RLum</a></p>
</td></tr>
<tr><td><code id="get_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method.</p>
</td></tr>
<tr><td><code id="get_RLum_+3A_class">class</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): allows to define the class that gets
selected if applied to a list, e.g., if a list consists of different type
of <a href="#topic+RLum-class">RLum</a> objects, this arguments allows to make selection. If nothing
is provided, all RLum-objects are treated.</p>
</td></tr>
<tr><td><code id="get_RLum_+3A_null.rm">null.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): remove empty and <code>NULL</code> objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same type as the input object provided.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>get_RLum(list)</code>: Returns a list of <a href="#topic+RLum-class">RLum</a> objects that had been passed to <a href="#topic+get_RLum">get_RLum</a>
</p>
</li>
<li> <p><code>get_RLum(`NULL`)</code>: Returns <code>NULL</code>.
</p>
</li></ul>


<h3>Function version</h3>

<p>0.3.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. get_RLum(): General accessor function for RLum S4 class objects. Function version 0.3.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Example based using data and from the calc_CentralDose() function

## load example data
data(ExampleData.DeValues, envir = environment())

## apply the central dose model 1st time
temp1 &lt;- calc_CentralDose(ExampleData.DeValues$CA1)

## get results and store them in a new object
temp.get &lt;- get_RLum(object = temp1)

</code></pre>

<hr>
<h2 id='GitHub-API'>GitHub API</h2><span id='topic+GitHub-API'></span><span id='topic+github_commits'></span><span id='topic+github_branches'></span><span id='topic+github_issues'></span>

<h3>Description</h3>

<p>R Interface to the GitHub API v3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>github_commits(user = "r-lum", repo = "luminescence", branch = "master", n = 5)

github_branches(user = "r-lum", repo = "luminescence")

github_issues(user = "r-lum", repo = "luminescence", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GitHub-API_+3A_user">user</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
GitHub user name (defaults to <code>'r-lum'</code>).</p>
</td></tr>
<tr><td><code id="GitHub-API_+3A_repo">repo</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
name of a GitHub repository (defaults to <code>'luminescence'</code>).</p>
</td></tr>
<tr><td><code id="GitHub-API_+3A_branch">branch</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
branch of a GitHub repository (defaults to <code>'master'</code>).</p>
</td></tr>
<tr><td><code id="GitHub-API_+3A_n">n</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number of commits returned (defaults to 5).</p>
</td></tr>
<tr><td><code id="GitHub-API_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions can be used to query a specific repository hosted on GitHub. <br />
</p>
<p><code>github_commits</code> lists the most recent <code>n</code> commits of a specific branch of a repository.
</p>
<p><code>github_branches</code> can be used to list all current branches of a
repository and returns the corresponding SHA hash as well as an installation
command to install the branch in R via the 'devtools' package.
</p>
<p><code>github_issues</code> lists all open issues for a repository in valid YAML.
</p>


<h3>Value</h3>

<p><code>github_commits</code>: <a href="base.html#topic+data.frame">data.frame</a> with columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,1]&#8288;</code> </td><td style="text-align: left;"> SHA </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,2]&#8288;</code> </td><td style="text-align: left;"> AUTHOR </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,3]&#8288;</code> </td><td style="text-align: left;"> DATE </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,4]&#8288;</code> </td><td style="text-align: left;"> MESSAGE </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>github_branches</code>: <a href="base.html#topic+data.frame">data.frame</a> with columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,1]&#8288;</code> </td><td style="text-align: left;"> BRANCH </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,2]&#8288;</code> </td><td style="text-align: left;"> SHA </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[ ,3]&#8288;</code> </td><td style="text-align: left;"> INSTALL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>github_commits</code>: Nested <a href="base.html#topic+list">list</a> with <code>n</code> elements.
Each commit element is a list with elements:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[1]]&#8288;</code> </td><td style="text-align: left;"> NUMBER </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[2]]&#8288;</code> </td><td style="text-align: left;"> TITLE </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[3]]&#8288;</code> </td><td style="text-align: left;"> BODY </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[4]]&#8288;</code> </td><td style="text-align: left;"> CREATED </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[5]]&#8288;</code> </td><td style="text-align: left;"> UPDATED </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[6]]&#8288;</code> </td><td style="text-align: left;"> CREATOR </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[7]]&#8288;</code> </td><td style="text-align: left;"> URL </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;[[8]]&#8288;</code> </td><td style="text-align: left;"> STATUS </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. GitHub-API(): GitHub API. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>GitHub Developer API v3. <a href="https://docs.github.com/v3/">https://docs.github.com/v3/</a>, last accessed: 10/01/2017.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
github_branches(user = "r-lum", repo = "luminescence")
github_issues(user = "r-lum", repo = "luminescence")
github_commits(user = "r-lum", repo = "luminescence", branch = "master", n = 10)

## End(Not run)

</code></pre>

<hr>
<h2 id='import_Data'>Import Luminescence Data into R</h2><span id='topic+import_Data'></span>

<h3>Description</h3>

<p>Convenience wrapper function to provide a quicker and more standardised way of
reading data into R by looping through all in the package available data import
functions starting with <code>read_</code>. Import data types can be mixed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_Data(file, ..., fastForward = TRUE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="import_Data_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>): file to be imported, can be a <a href="base.html#topic+list">list</a> or a <a href="base.html#topic+character">character</a> vector</p>
</td></tr>
<tr><td><code id="import_Data_+3A_...">...</code></td>
<td>
<p>arguments to be further passed down to supported functions (please check the functions
to determine the correct arguments)</p>
</td></tr>
<tr><td><code id="import_Data_+3A_fastforward">fastForward</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): option to create <a href="#topic+RLum-class">RLum</a> objects
during import or a <a href="base.html#topic+list">list</a> of such objects</p>
</td></tr>
<tr><td><code id="import_Data_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Always returns a <a href="base.html#topic+list">list</a>; empty or filled with <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
</p>


<h3>Function version</h3>

<p>0.1.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. import_Data(): Import Luminescence Data into R. Function version 0.1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+read_HeliosOSL2R">read_HeliosOSL2R</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+read_Daybreak2R">read_Daybreak2R</a>, <a href="#topic+read_RF2R">read_RF2R</a>, <a href="#topic+read_SPE2R">read_SPE2R</a>, <a href="#topic+read_PSL2R">read_PSL2R</a>, <a href="#topic+read_XSYG2R">read_XSYG2R</a>, <a href="#topic+read_TIFF2R">read_TIFF2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## import BINX/BIN
file &lt;- system.file("extdata/BINfile_V8.binx", package = "Luminescence")
temp &lt;- import_Data(file)

## RF data
file &lt;- system.file("extdata", "RF_file.rf", package = "Luminescence")
temp &lt;- import_Data(file)

</code></pre>

<hr>
<h2 id='install_DevelopmentVersion'>Attempts to install the development version of the 'Luminescence' package</h2><span id='topic+install_DevelopmentVersion'></span>

<h3>Description</h3>

<p>This function is a convenient method for installing the development
version of the R package 'Luminescence' directly from GitHub.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_DevelopmentVersion(force_install = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="install_DevelopmentVersion_+3A_force_install">force_install</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
If <code>FALSE</code> (the default) the function produces and prints the required
code to the console for the user to run manually afterwards. When <code>TRUE</code>
and all requirements are fulfilled (see details) this function attempts to install
the package itself.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function uses <a href="#topic+GitHub-API">Luminescence::github_branches</a> to check
which development branches of the R package 'Luminescence' are currently
available on GitHub. The user is then prompted to choose one of the branches
to be installed. It further checks whether the R package 'devtools' is
currently installed and available on the system. Finally, it prints R code
to the console that the user can copy and paste to the R console in order
to install the desired development version of the package.
</p>
<p>If <code>force_install = TRUE</code> the functions checks if 'devtools' is available
and then attempts to install the chosen development branch via
<a href="devtools.html#topic+remote-reexports">devtools::install_github</a>.
</p>


<h3>Value</h3>

<p>This function requires user input at the command prompt to choose the
desired development branch to be installed. The required R code to install
the package is then printed to the console.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
install_DevelopmentVersion()

## End(Not run)

</code></pre>

<hr>
<h2 id='length_RLum'>Length retrieval function for RLum S4 class objects</h2><span id='topic+length_RLum'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum-class">RLum</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>length_RLum(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="length_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="base.html#topic+integer">integer</a> indicating the length of the object.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. length_RLum(): Length retrieval function for RLum S4 class objects. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
<a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>,
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a>,
<a href="#topic+RLum.Results-class">RLum.Results</a>
</p>

<hr>
<h2 id='melt_RLum'>Melt RLum-class objects into a flat data.frame</h2><span id='topic+melt_RLum'></span><span id='topic+melt_RLum+2Clist-method'></span>

<h3>Description</h3>

<p>Melt RLum-class objects into a flat data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>melt_RLum(object, ...)

## S4 method for signature 'list'
melt_RLum(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="melt_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code></p>
</td></tr>
<tr><td><code id="melt_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A flat <a href="base.html#topic+data.frame">data.frame</a>.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>melt_RLum(list)</code>: Returns a list a single <a href="base.html#topic+data.frame">data.frame</a>
</p>
</li></ul>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. melt_RLum(): Melt RLum-class objects into a flat data.frame. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleData.XSYG, envir = environment())
melt_RLum(OSL.SARMeasurement[[2]][[1]])

</code></pre>

<hr>
<h2 id='merge_Risoe.BINfileData'>Merge Risoe.BINfileData objects or Risoe BIN-files</h2><span id='topic+merge_Risoe.BINfileData'></span>

<h3>Description</h3>

<p>Function allows merging Risoe BIN/BINX files or <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_Risoe.BINfileData(
  input.objects,
  output.file,
  keep.position.number = FALSE,
  position.number.append.gap = 0
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_Risoe.BINfileData_+3A_input.objects">input.objects</code></td>
<td>
<p><a href="base.html#topic+character">character</a> with <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects (<strong>required</strong>):
Character vector with path and files names
(e.g. <code>input.objects = c("path/file1.bin", "path/file2.bin")</code> or
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects (e.g. <code>input.objects = c(object1, object2)</code>).
Alternatively a <code>list</code> is supported.</p>
</td></tr>
<tr><td><code id="merge_Risoe.BINfileData_+3A_output.file">output.file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
File output path and name. If no value is given, a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> is
returned instead of a file.</p>
</td></tr>
<tr><td><code id="merge_Risoe.BINfileData_+3A_keep.position.number">keep.position.number</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Allows keeping the original position numbers of the input objects.
Otherwise the position numbers are recalculated.</p>
</td></tr>
<tr><td><code id="merge_Risoe.BINfileData_+3A_position.number.append.gap">position.number.append.gap</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
Set the position number gap between merged BIN-file sets, if the option
<code>keep.position.number = FALSE</code> is used. See details for further
information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function allows merging different measurements to one file or one
object. The record IDs are recalculated for the new object. Other values
are kept for each object. The number of input objects is not limited.
</p>
<p><code>position.number.append.gap</code> option
</p>
<p>If the option <code>keep.position.number = FALSE</code> is used, the position
numbers of the new data set are recalculated by adding the highest position
number of the previous data set to the each position number of the next data
set. For example: The highest position number is 48, then this number will
be added to all other position numbers of the next data set (e.g. 1 + 48 =
49)
</p>
<p>However, there might be cases where an additional addend (summand) is needed
before the next position starts. Example:
</p>

<ul>
<li><p> Position number set (A): <code style="white-space: pre;">&#8288;1,3,5,7&#8288;</code>
</p>
</li>
<li><p> Position number set (B): <code style="white-space: pre;">&#8288;1,3,5,7&#8288;</code>
</p>
</li></ul>

<p>With no additional summand the new position numbers would be:
<code style="white-space: pre;">&#8288;1,3,5,7,8,9,10,11&#8288;</code>. That might be unwanted. Using the argument
<code>position.number.append.gap = 1</code> it will become:
<code style="white-space: pre;">&#8288;1,3,5,7,9,11,13,15,17&#8288;</code>.
</p>


<h3>Value</h3>

<p>Returns a <code>file</code> or a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object.
</p>


<h3>Function version</h3>

<p>0.2.9
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. merge_Risoe.BINfileData(): Merge Risoe.BINfileData objects or Risoe BIN-files. Function version 0.2.9. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The validity of the output objects is not further checked.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Duller, G.A.T., 2007. Analyst (Version 3.24) (manual). Aberystwyth University, Aberystwyth.
</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+write_R2BIN">write_R2BIN</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##merge two objects
data(ExampleData.BINfileData, envir = environment())

object1 &lt;- CWOSL.SAR.Data
object2 &lt;- CWOSL.SAR.Data

object.new &lt;- merge_Risoe.BINfileData(c(object1, object2))

</code></pre>

<hr>
<h2 id='merge_RLum'>General merge function for RLum S4 class objects</h2><span id='topic+merge_RLum'></span>

<h3>Description</h3>

<p>Function calls object-specific merge functions for RLum S4 class objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_RLum(objects, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_RLum_+3A_objects">objects</code></td>
<td>
<p><a href="base.html#topic+list">list</a> of <a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
list of S4 object of class <code>RLum</code></p>
</td></tr>
<tr><td><code id="merge_RLum_+3A_...">...</code></td>
<td>
<p>further arguments that one might want to pass to the specific merge function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function provides a generalised access point for merging specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the
corresponding merge function will be selected.  Allowed arguments can be
found in the documentation of each merge function.
Empty list elements (<code>NULL</code>) are automatically removed from the input <code>list</code>.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>object</strong> </td><td style="text-align: left;"> </td><td style="text-align: left;"> <strong>corresponding merge function</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <code>merge_RLum.Data.Curve</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <code>merge_RLum.Analysis</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Results-class">RLum.Results</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <code>merge_RLum.Results</code>
</td>
</tr>

</table>



<h3>Value</h3>

<p>Return is the same as input objects as provided in the list.
</p>


<h3>Function version</h3>

<p>0.1.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. merge_RLum(): General merge function for RLum S4 class objects. Function version 0.1.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>So far not for every <code>RLum</code> object a merging function exists.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

##Example based using data and from the calc_CentralDose() function

##load example data
data(ExampleData.DeValues, envir = environment())

##apply the central dose model 1st time
temp1 &lt;- calc_CentralDose(ExampleData.DeValues$CA1)

##apply the central dose model 2nd time
temp2 &lt;- calc_CentralDose(ExampleData.DeValues$CA1)

##merge the results and store them in a new object
temp.merged &lt;- get_RLum(merge_RLum(objects = list(temp1, temp2)))

</code></pre>

<hr>
<h2 id='merge_RLum.Analysis'>Merge function for RLum.Analysis S4 class objects</h2><span id='topic+merge_RLum.Analysis'></span>

<h3>Description</h3>

<p>Function allows merging of RLum.Analysis objects and adding of allowed
objects to an RLum.Analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_RLum.Analysis(objects)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_RLum.Analysis_+3A_objects">objects</code></td>
<td>
<p><a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
list of S4 objects of class <code>RLum.Analysis</code>. Furthermore other objects of
class <a href="#topic+RLum-class">RLum</a> can be added, see details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simply allows to merge <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
objects. Moreover, other <a href="#topic+RLum-class">RLum</a> objects can be added
to an existing <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object. Supported objects
to be added are: <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> and
<a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>.
</p>
<p>The order in the new <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object is the object
order provided with the input list.
</p>


<h3>Value</h3>

<p>Return an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.
</p>


<h3>Function version</h3>

<p>0.2.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. merge_RLum.Analysis(): Merge function for RLum.Analysis S4 class objects. Function version 0.2.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The information for the slot 'protocol' is taken from the first
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object in the input list. Therefore at
least one object of type <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> has to be provided.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+merge_RLum">merge_RLum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>, <a href="#topic+RLum-class">RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

##merge different RLum objects from the example data
data(ExampleData.RLum.Analysis, envir = environment())
data(ExampleData.BINfileData, envir = environment())

object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)
curve &lt;- get_RLum(object)[[2]]

temp.merged &lt;- merge_RLum.Analysis(list(curve, IRSAR.RF.Data, IRSAR.RF.Data))

</code></pre>

<hr>
<h2 id='merge_RLum.Data.Curve'>Merge function for RLum.Data.Curve S4 class objects</h2><span id='topic+merge_RLum.Data.Curve'></span>

<h3>Description</h3>

<p>This function allows to merge <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects in different
ways without modifying the original objects. Merging is always applied on
the 2nd column of the object's data matrix.
</p>
<p><strong>Supported merge operations are:</strong>
</p>
<p><code>"mean"</code> (default)
</p>
<p>The mean over the count values is calculated using the function
<a href="base.html#topic+rowMeans">rowMeans</a>.
</p>
<p><code>"median"</code>
</p>
<p>The median over the count values is calculated using the function
<a href="matrixStats.html#topic+rowMedians">matrixStats::rowMedians</a>.
</p>
<p><code>"sum"</code>
</p>
<p>All count values will be summed up using the function <a href="base.html#topic+rowSums">rowSums</a>.
</p>
<p><code>"sd"</code>
</p>
<p>The standard deviation over the count values is calculated using the function
<a href="matrixStats.html#topic+rowSds">matrixStats::rowSds</a>.
</p>
<p><code>"var"</code>
</p>
<p>The variance over the count values is calculated using the function
<a href="matrixStats.html#topic+rowVars">matrixStats::rowVars</a>.
</p>
<p><code>"min"</code>
</p>
<p>The min values from the count values is chosen using the function
<a href="matrixStats.html#topic+rowRanges">matrixStats::rowMins</a>.
</p>
<p><code>"max"</code>
</p>
<p>The max values from the count values is chosen using the function
<a href="matrixStats.html#topic+rowRanges">matrixStats::rowMins</a>.
</p>
<p><code>"append"</code>
</p>
<p>Appends count values of all curves to one combined data curve. The channel width
is automatically re-calculated, but requires a constant channel width of the
original data.
</p>
<p><code>"-"</code>
</p>
<p>The row sums of the last objects are subtracted from the first object.
</p>
<p><code>"*"</code>
</p>
<p>The row sums of the last objects are multiplied with the first object.
</p>
<p><code>"/"</code>
</p>
<p>Values of the first object are divided by row sums of the last objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_RLum.Data.Curve(object, merge.method = "mean", method.info)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_RLum.Data.Curve_+3A_object">object</code></td>
<td>
<p><a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> (<strong>required</strong>):
list of objects to be merged.</p>
</td></tr>
<tr><td><code id="merge_RLum.Data.Curve_+3A_merge.method">merge.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
method for combining of the objects, e.g. <code>'mean'</code> (default), <code>'median'</code>,
<code>'sum'</code>, see details for
further information and allowed methods.  Note: Elements in slot info will
be taken from the first curve in the list.</p>
</td></tr>
<tr><td><code id="merge_RLum.Data.Curve_+3A_method.info">method.info</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
allows to specify how info elements of the input objects are combined,
e.g. <code>1</code> means that just the elements from the first object are kept,
<code>2</code> keeps only the info elements from the 2 object etc.
If nothing is provided all elements are combined.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> object.
</p>


<h3>S3-generic support</h3>

<p>This function is fully operational via S3-generics:
<code>+</code>, <code>-</code>, <code>/</code>, <code>*</code>, <code>merge</code>
</p>


<h3>Function version</h3>

<p>0.2.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. merge_RLum.Data.Curve(): Merge function for RLum.Data.Curve S4 class objects. Function version 0.2.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The information from the slot <code>recordType</code> is taken from the first
object in the input list. The slot
'curveType' is filled with the name <code>merged</code>.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+merge_RLum">merge_RLum</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load example data
data(ExampleData.XSYG, envir = environment())

##grep first and 3d TL curves
TL.curves  &lt;- get_RLum(OSL.SARMeasurement$Sequence.Object, recordType = "TL (UVVIS)")
TL.curve.1 &lt;- TL.curves[[1]]
TL.curve.3 &lt;- TL.curves[[3]]

##plot single curves
plot_RLum(TL.curve.1)
plot_RLum(TL.curve.3)

##subtract the 1st curve from the 2nd and plot
TL.curve.merged &lt;- merge_RLum.Data.Curve(list(TL.curve.3, TL.curve.1), merge.method = "/")
plot_RLum(TL.curve.merged)

</code></pre>

<hr>
<h2 id='merge_RLum.Data.Spectrum'>Merge function for RLum.Data.Spectrum S4 class objects</h2><span id='topic+merge_RLum.Data.Spectrum'></span>

<h3>Description</h3>

<p>This function allows to merge <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> objects in
different ways without modifying the original objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_RLum.Data.Spectrum(
  object,
  merge.method = "mean",
  method.info,
  max.temp.diff = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_RLum.Data.Spectrum_+3A_object">object</code></td>
<td>
<p><a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> (<strong>required</strong>):
list of objects to be merged.</p>
</td></tr>
<tr><td><code id="merge_RLum.Data.Spectrum_+3A_merge.method">merge.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
method for combining of the objects, e.g. <code>'mean'</code> (default), <code>'median'</code>,
<code>'sum'</code>, see details for
further information and allowed methods.  Note: Elements in slot info will
be taken from the first object in the list.</p>
</td></tr>
<tr><td><code id="merge_RLum.Data.Spectrum_+3A_method.info">method.info</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
allows to specify how info elements of the input objects are combined,
e.g. <code>1</code> means that just the elements from the first object are kept,
<code>2</code> keeps only the info elements from the 2 object etc.
If nothing is provided all elements are combined.</p>
</td></tr>
<tr><td><code id="merge_RLum.Data.Spectrum_+3A_max.temp.diff">max.temp.diff</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
maximum difference in the time/temperature values between the spectra to
be merged: when differences exceed this threshold value, the merging
occurs but a warning is raised.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Supported merge operations are:</strong>
</p>
<p><code>"mean"</code> (default)
</p>
<p>The mean over the cell values is calculated using the function
<a href="base.html#topic+rowMeans">rowMeans</a>.
</p>
<p><code>"median"</code>
</p>
<p>The median over the cell values is calculated using the function
<a href="matrixStats.html#topic+rowMedians">matrixStats::rowMedians</a>.
</p>
<p><code>"sum"</code>
</p>
<p>All cell values will be summed up using the function <a href="base.html#topic+rowSums">rowSums</a>.
</p>
<p><code>"sd"</code>
</p>
<p>The standard deviation over the cell values is calculated using the function
<a href="matrixStats.html#topic+rowSds">matrixStats::rowSds</a>.
</p>
<p><code>"var"</code>
</p>
<p>The variance over the cell values is calculated using the function
<a href="matrixStats.html#topic+rowVars">matrixStats::rowVars</a>.
</p>
<p><code>"min"</code>
</p>
<p>The min values from the cell values is chosen using the function
<a href="matrixStats.html#topic+rowRanges">matrixStats::rowMins</a>.
</p>
<p><code>"max"</code>
</p>
<p>The max values from the cell values is chosen using the function
<a href="matrixStats.html#topic+rowRanges">matrixStats::rowMins</a>.
</p>
<p><code>"append"</code> (only for <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>)
</p>
<p>Appends cell values of all curves to one combined data curve. The channel width
is automatically re-calculated, but requires a constant channel width of the
original data.
</p>
<p><code>"-"</code>
</p>
<p>The cell sums of the last objects are subtracted from the first object.
</p>
<p><code>"*"</code>
</p>
<p>The cell sums of the last objects are multiplied with the first object.
</p>
<p><code>"/"</code>
</p>
<p>Values of the first object are divided by cell sums of the last objects.
</p>


<h3>Value</h3>

<p>Returns an <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> object.
</p>


<h3>S3-generic support</h3>

<p>This function is fully operational via S3-generics:
<code>+</code>, <code>-</code>, <code>/</code>, <code>*</code>, <code>merge</code>
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Colombo, M., Kreutzer, S., 2025. merge_RLum.Data.Spectrum(): Merge function for RLum.Data.Spectrum S4 class objects. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The information from the slot <code>recordType</code> is taken from the first
object in the input list. The slot
'curveType' is filled with the name <code>merged</code>.
</p>


<h3>Author(s)</h3>

<p>Marco Colombo, Institute of Geography, Heidelberg University (Germany)
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+merge_RLum">merge_RLum</a>, <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.XSYG, envir = environment())

## plot single curve
plot_RLum(TL.Spectrum)

## sum two copies of the same curve
merged &lt;- merge_RLum.Data.Spectrum(list(TL.Spectrum, TL.Spectrum),
                                   merge.method = "sum")
plot_RLum(merged)

</code></pre>

<hr>
<h2 id='merge_RLum.Results'>Merge function for RLum.Results S4-class objects</h2><span id='topic+merge_RLum.Results'></span>

<h3>Description</h3>

<p>Function merges objects of class <a href="#topic+RLum.Results-class">RLum.Results</a>. The slots in the objects
are combined depending on the object type, e.g., for <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
rows are appended.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_RLum.Results(objects)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_RLum.Results_+3A_objects">objects</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<strong>required</strong>):
a list of <a href="#topic+RLum.Results-class">RLum.Results</a> objects</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Elements are appended where possible and attributes are preserved if
not of similar name as the default attributes of, e.g., a <a href="base.html#topic+data.frame">data.frame</a>
</p>


<h3>Function version</h3>

<p>0.2.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. merge_RLum.Results(): Merge function for RLum.Results S4-class objects. Function version 0.2.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The <code>originator</code> is taken from the first element and not reset to <code>merge_RLum</code>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>

<hr>
<h2 id='methods_RLum'>methods_RLum</h2><span id='topic+methods_RLum'></span><span id='topic+plot.list'></span><span id='topic+plot.RLum.Results'></span><span id='topic+plot.RLum.Analysis'></span><span id='topic+plot.RLum.Data.Curve'></span><span id='topic+plot.RLum.Data.Spectrum'></span><span id='topic+plot.RLum.Data.Image'></span><span id='topic+plot.Risoe.BINfileData'></span><span id='topic+hist.RLum.Results'></span><span id='topic+hist.RLum.Data.Image'></span><span id='topic+hist.RLum.Data.Curve'></span><span id='topic+hist.RLum.Analysis'></span><span id='topic+summary.RLum.Results'></span><span id='topic+summary.RLum.Analysis'></span><span id='topic+summary.RLum.Data.Image'></span><span id='topic+summary.RLum.Data.Curve'></span><span id='topic+subset.Risoe.BINfileData'></span><span id='topic+subset.RLum.Analysis'></span><span id='topic+bin'></span><span id='topic+bin.RLum.Data.Curve'></span><span id='topic+bin.RLum.Data.Spectrum'></span><span id='topic+length.RLum.Results'></span><span id='topic+length.RLum.Analysis'></span><span id='topic+length.RLum.Data.Curve'></span><span id='topic+length.Risoe.BINfileData'></span><span id='topic+dim.RLum.Data.Curve'></span><span id='topic+dim.RLum.Data.Spectrum'></span><span id='topic+rep.RLum'></span><span id='topic+names.RLum.Data.Curve'></span><span id='topic+names.RLum.Data.Spectrum'></span><span id='topic+names.RLum.Data.Image'></span><span id='topic+names.RLum.Analysis'></span><span id='topic+names.RLum.Results'></span><span id='topic+names.Risoe.BINfileData'></span><span id='topic+row.names.RLum.Data.Spectrum'></span><span id='topic+as.data.frame.RLum.Data.Curve'></span><span id='topic+as.data.frame.RLum.Data.Spectrum'></span><span id='topic+as.data.frame.Risoe.BINfileData'></span><span id='topic+as.list.RLum.Results'></span><span id='topic+as.list.RLum.Data.Curve'></span><span id='topic+as.list.RLum.Data.Image'></span><span id='topic+as.list.RLum.Data.Spectrum'></span><span id='topic+as.list.RLum.Analysis'></span><span id='topic+as.matrix.RLum.Data.Curve'></span><span id='topic+as.matrix.RLum.Data.Spectrum'></span><span id='topic+as.matrix.RLum.Data.Image'></span><span id='topic+is.RLum'></span><span id='topic+is.RLum.Data'></span><span id='topic+is.RLum.Data.Curve'></span><span id='topic+is.RLum.Data.Spectrum'></span><span id='topic+is.RLum.Data.Image'></span><span id='topic+is.RLum.Analysis'></span><span id='topic+is.RLum.Results'></span><span id='topic+merge.RLum'></span><span id='topic+unlist.RLum.Analysis'></span><span id='topic++2B.RLum.Data.Curve'></span><span id='topic++2B.RLum.Data.Spectrum'></span><span id='topic+-.RLum.Data.Curve'></span><span id='topic+-.RLum.Data.Spectrum'></span><span id='topic++2A.RLum.Data.Curve'></span><span id='topic++2A.RLum.Data.Spectrum'></span><span id='topic++2F.RLum.Data.Curve'></span><span id='topic++2F.RLum.Data.Spectrum'></span><span id='topic++5B.RLum.Data.Curve'></span><span id='topic++5B.RLum.Data.Spectrum'></span><span id='topic++5B.RLum.Data.Image'></span><span id='topic++5B.RLum.Analysis'></span><span id='topic++5B.RLum.Results'></span><span id='topic++5B+3C-.RLum.Data.Curve'></span><span id='topic++5B+5B.RLum.Analysis'></span><span id='topic++5B+5B.RLum.Results'></span><span id='topic++24.RLum.Data.Curve'></span><span id='topic++24.RLum.Analysis'></span><span id='topic++24.RLum.Results'></span>

<h3>Description</h3>

<p>Methods for S3-generics implemented for the package 'Luminescence'.
This document summarises all implemented S3-generics. The name of the function
is given before the first dot, after the dot the name of the object that is
supported by this method is given, e.g. <code>plot.RLum.Data.Curve</code> can be called
by <code>plot(object, ...)</code>, where <code>object</code> is the <code>RLum.Data.Curve</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'list'
plot(x, y, ...)

## S3 method for class 'RLum.Results'
plot(x, y, ...)

## S3 method for class 'RLum.Analysis'
plot(x, y, ...)

## S3 method for class 'RLum.Data.Curve'
plot(x, y, ...)

## S3 method for class 'RLum.Data.Spectrum'
plot(x, y, ...)

## S3 method for class 'RLum.Data.Image'
plot(x, y, ...)

## S3 method for class 'Risoe.BINfileData'
plot(x, y, ...)

## S3 method for class 'RLum.Results'
hist(x, ...)

## S3 method for class 'RLum.Data.Image'
hist(x, ...)

## S3 method for class 'RLum.Data.Curve'
hist(x, ...)

## S3 method for class 'RLum.Analysis'
hist(x, ...)

## S3 method for class 'RLum.Results'
summary(object, ...)

## S3 method for class 'RLum.Analysis'
summary(object, ...)

## S3 method for class 'RLum.Data.Image'
summary(object, ...)

## S3 method for class 'RLum.Data.Curve'
summary(object, ...)

## S3 method for class 'Risoe.BINfileData'
subset(x, subset, records.rm = TRUE, ...)

## S3 method for class 'RLum.Analysis'
subset(x, subset = NULL, ...)

bin(x, ...)

## S3 method for class 'RLum.Data.Curve'
bin(x, bin_size = 2, ...)

## S3 method for class 'RLum.Data.Spectrum'
bin(x, bin_size.row = 1, bin_size.col = 1, ...)

## S3 method for class 'RLum.Results'
length(x, ...)

## S3 method for class 'RLum.Analysis'
length(x, ...)

## S3 method for class 'RLum.Data.Curve'
length(x, ...)

## S3 method for class 'Risoe.BINfileData'
length(x, ...)

## S3 method for class 'RLum.Data.Curve'
dim(x)

## S3 method for class 'RLum.Data.Spectrum'
dim(x)

## S3 method for class 'RLum'
rep(x, ...)

## S3 method for class 'RLum.Data.Curve'
names(x, ...)

## S3 method for class 'RLum.Data.Spectrum'
names(x, ...)

## S3 method for class 'RLum.Data.Image'
names(x, ...)

## S3 method for class 'RLum.Analysis'
names(x, ...)

## S3 method for class 'RLum.Results'
names(x, ...)

## S3 method for class 'Risoe.BINfileData'
names(x)

## S3 method for class 'RLum.Data.Spectrum'
row.names(x, ...)

## S3 method for class 'RLum.Data.Curve'
as.data.frame(x, row.names = NULL, optional = FALSE, ...)

## S3 method for class 'RLum.Data.Spectrum'
as.data.frame(x, row.names = NULL, optional = FALSE, ...)

## S3 method for class 'Risoe.BINfileData'
as.data.frame(x, row.names = NULL, optional = FALSE, ...)

## S3 method for class 'RLum.Results'
as.list(x, ...)

## S3 method for class 'RLum.Data.Curve'
as.list(x, ...)

## S3 method for class 'RLum.Data.Image'
as.list(x, ...)

## S3 method for class 'RLum.Data.Spectrum'
as.list(x, ...)

## S3 method for class 'RLum.Analysis'
as.list(x, ...)

## S3 method for class 'RLum.Data.Curve'
as.matrix(x, ...)

## S3 method for class 'RLum.Data.Spectrum'
as.matrix(x, ...)

## S3 method for class 'RLum.Data.Image'
as.matrix(x, ...)

is.RLum(x, ...)

is.RLum.Data(x, ...)

is.RLum.Data.Curve(x, ...)

is.RLum.Data.Spectrum(x, ...)

is.RLum.Data.Image(x, ...)

is.RLum.Analysis(x, ...)

is.RLum.Results(x, ...)

## S3 method for class 'RLum'
merge(x, y, ...)

## S3 method for class 'RLum.Analysis'
unlist(x, recursive = TRUE, ...)

## S3 method for class 'RLum.Data.Curve'
x + y

## S3 method for class 'RLum.Data.Spectrum'
x + y

## S3 method for class 'RLum.Data.Curve'
x - y

## S3 method for class 'RLum.Data.Spectrum'
x - y

## S3 method for class 'RLum.Data.Curve'
x * y

## S3 method for class 'RLum.Data.Spectrum'
x * y

## S3 method for class 'RLum.Data.Curve'
x / y

## S3 method for class 'RLum.Data.Spectrum'
x / y

## S3 method for class 'RLum.Data.Curve'
x[y, z, drop = TRUE]

## S3 method for class 'RLum.Data.Spectrum'
x[y, z, drop = TRUE]

## S3 method for class 'RLum.Data.Image'
x[y, z, drop = TRUE]

## S3 method for class 'RLum.Analysis'
x[i, drop = FALSE]

## S3 method for class 'RLum.Results'
x[i, drop = TRUE]

## S3 replacement method for class 'RLum.Data.Curve'
x[i, j] &lt;- value

## S3 method for class 'RLum.Analysis'
x[[i]]

## S3 method for class 'RLum.Results'
x[[i]]

## S3 method for class 'RLum.Data.Curve'
x$i

## S3 method for class 'RLum.Analysis'
x$i

## S3 method for class 'RLum.Results'
x$i
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="methods_RLum_+3A_x">x</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> or <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
input object</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_y">y</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
the row index of the matrix, data.frame</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to the method</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
input object</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_subset">subset</code></td>
<td>
<p><code style="white-space: pre;">&#8288;[subset]&#8288;</code> <a href="base.html#topic+expression">expression</a> (<strong>required</strong>):
logical expression indicating elements or rows to keep, this function works
in <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects like <a href="base.html#topic+subset.data.frame">subset.data.frame</a>, but takes care
of the object structure. Works also on <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects.</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_records.rm">records.rm</code></td>
<td>
<p><a href="base.html#topic+subset">subset</a> <a href="base.html#topic+logical">logical</a> (<em>with default</em>):
remove records from data set, can be disabled, to just set the column <code>SET</code> to <code>TRUE</code> or <code>FALSE</code></p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_row.names">row.names</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable row names (<code>as.data.frame</code>).</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_optional">optional</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
logical. If TRUE, setting row names and converting column names
(to syntactic names: see make.names) is optional (see <a href="base.html#topic+as.data.frame">base::as.data.frame</a>)</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_recursive">recursive</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable further sub-setting (<code>unlist</code>).</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_z">z</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
the column index of the matrix, data.frame</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_drop">drop</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
keep object structure or drop it</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_i">i</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
name of the wanted record type or data object or row in the <code>RLum.Data.Curve</code> object</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_j">j</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
column of the data matrix in the <code>RLum.Data.Curve</code> object</p>
</td></tr>
<tr><td><code id="methods_RLum_+3A_value">value</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>):
numeric value which replace the value in the <code>RLum.Data.Curve</code> object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The term S3-generics sounds complicated, however, it just means that something
has been implemented in the package to increase the usability for users new
in R and who are not familiar with the underlying <code>RLum</code>-object structure of
the package. The practical outcome is that operations and functions presented
in standard books on R can be used without knowing the specifics of the R
package <code>'Luminescence'</code>. For examples see the example section.
</p>


<h3>Note</h3>

<p><code>methods_RLum</code> are not really new functions, everything given here are mostly just
surrogates for existing functions in the package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load example data
data(ExampleData.RLum.Analysis, envir = environment())


##combine curve is various ways
curve1 &lt;- IRSAR.RF.Data[[1]]
curve2 &lt;-  IRSAR.RF.Data[[1]]
curve1 + curve2
curve1 - curve2
curve1 / curve2
curve1 * curve2


##`$` access curves
IRSAR.RF.Data$RF

</code></pre>

<hr>
<h2 id='names_RLum'>Name retrieval function for RLum S4 class objects</h2><span id='topic+names_RLum'></span><span id='topic+names_RLum+2Clist-method'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum-class">RLum</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>names_RLum(object)

## S4 method for signature 'list'
names_RLum(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="names_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="base.html#topic+character">character</a> vector.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>names_RLum(list)</code>: Returns a list of <a href="#topic+RLum-class">RLum</a> objects that had been passed to <a href="#topic+names_RLum">names_RLum</a>
</p>
</li></ul>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. names_RLum(): Name retrieval function for RLum S4 class objects. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>

<hr>
<h2 id='plot_AbanicoPlot'>Function to create an Abanico Plot.</h2><span id='topic+plot_AbanicoPlot'></span>

<h3>Description</h3>

<p>A plot is produced which allows comprehensive presentation of data precision
and its dispersion around a central value as well as illustration of a
kernel density estimate, histogram and/or dot plot of the dose values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_AbanicoPlot(
  data,
  na.rm = TRUE,
  log.z = TRUE,
  z.0 = "mean.weighted",
  dispersion = "qr",
  plot.ratio = 0.75,
  rotate = FALSE,
  mtext,
  summary,
  summary.pos,
  summary.method = "MCM",
  legend,
  legend.pos,
  stats,
  rug = FALSE,
  kde = TRUE,
  hist = FALSE,
  dots = FALSE,
  boxplot = FALSE,
  y.axis = TRUE,
  error.bars = FALSE,
  bar,
  bar.col,
  polygon.col,
  line,
  line.col,
  line.lty,
  line.label,
  grid.col,
  frame = 1,
  bw = "SJ",
  interactive = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_AbanicoPlot_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> object (<strong>required</strong>):
for <code>data.frame</code> two columns: De (<code>data[,1]</code>) and De error (<code>data[,2]</code>).
To plot several data sets in one plot the data sets must be provided as
<code>list</code>, e.g. <code>list(data.1, data.2)</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
exclude NA values from the data set prior to any further operations.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_log.z">log.z</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to display the z-axis in logarithmic scale. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_z.0">z.0</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
User-defined central value, used for centring of data. One out of <code>"mean"</code>,
<code>"mean.weighted"</code> and <code>"median"</code> or a numeric value (not its logarithm).
Default is <code>"mean.weighted"</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_dispersion">dispersion</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
measure of dispersion, used for drawing the scatter polygon. One out of
</p>

<ul>
<li> <p><code>"qr"</code> (quartile range, default),
</p>
</li>
<li> <p><code>"pnn"</code> (symmetric percentile range with <code>nn</code> the lower percentile, e.g.
<code>"p05"</code> indicating the range between 5 and 95 %, or <code>"p10"</code> indicating
the range between 10 and 90 %), or
</p>
</li>
<li> <p><code>"sd"</code> (standard deviation) and
</p>
</li>
<li> <p><code>"2sd"</code> (2 standard deviations),
</p>
</li></ul>

<p>The default is <code>"qr"</code>. Note that <code>"sd"</code> and <code>"2sd"</code> are only meaningful in
combination with <code>"z.0 = 'mean'"</code> because the unweighted mean is used to
centre the polygon.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_plot.ratio">plot.ratio</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
Relative space, given to the radial versus the cartesian plot part,
default is <code>0.75</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_rotate">rotate</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to turn the plot by 90 degrees.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_mtext">mtext</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
additional text below the plot title.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
add statistic measures of centrality and dispersion to the plot.
Can be one or more of several keywords. See details for available keywords.
Results differ depending on the log-option for the z-scale (see details).</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_summary.pos">summary.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>) for the
statistical summary. Alternatively, the keyword <code>"sub"</code> may be
specified to place the summary below the plot header. However, this latter
option in only possible if <code>mtext</code> is not used.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_summary.method">summary.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
keyword indicating the method used to calculate the statistic summary.
One out of
</p>

<ul>
<li> <p><code>"unweighted"</code>,
</p>
</li>
<li> <p><code>"weighted"</code> and
</p>
</li>
<li> <p><code>"MCM"</code>.
</p>
</li></ul>

<p>See <a href="#topic+calc_Statistics">calc_Statistics</a> for details.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_legend">legend</code></td>
<td>
<p><a href="base.html#topic+character">character</a> vector (<em>optional</em>):
legend content to be added to the plot.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_legend.pos">legend.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>)
for the legend to be plotted.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_stats">stats</code></td>
<td>
<p><a href="base.html#topic+character">character</a>:
additional labels of statistically important values in the plot.
One or more out of the following:
</p>

<ul>
<li> <p><code>"min"</code>,
</p>
</li>
<li> <p><code>"max"</code>,
</p>
</li>
<li> <p><code>"median"</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_rug">rug</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to add a rug to the KDE part, to indicate the location of individual values.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_kde">kde</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to add a KDE plot to the dispersion part, default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_hist">hist</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to add a histogram to the dispersion part. Only meaningful when not
more than one data set is plotted.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_dots">dots</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to add a dot plot to the dispersion part. If number of dots exceeds
space in the dispersion part, a square indicates this.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_boxplot">boxplot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to add a boxplot to the dispersion part, default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_y.axis">y.axis</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Option to hide standard y-axis
labels and show 0 only.
Useful for data with small scatter. If you want to suppress the y-axis entirely
please use <code>yaxt == 'n'</code> (the standard <a href="graphics.html#topic+par">graphics::par</a> setting) instead.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_error.bars">error.bars</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to show De-errors as error bars on De-points. Useful in combination
with <code style="white-space: pre;">&#8288;y.axis = FALSE, bar.col = "none"&#8288;</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_bar">bar</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
option to add one or more dispersion bars (i.e., bar showing the 2-sigma range)
centred at the defined values. By default a bar is drawn according to <code>"z.0"</code>.
To omit the bar set <code>"bar = FALSE"</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_bar.col">bar.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
colour of the dispersion bar. Default is <code>"grey60"</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_polygon.col">polygon.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
colour of the polygon showing the data scatter. Sometimes this
polygon may be omitted for clarity. To disable it use <code>FALSE</code> or
<code>polygon = FALSE</code>. Default is <code>"grey80"</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_line">line</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
numeric values of the additional lines to be added.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_line.col">line.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a>:
colour of the additional lines.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_line.lty">line.lty</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a>:
line type of additional lines</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_line.label">line.label</code></td>
<td>
<p><a href="base.html#topic+character">character</a>:
labels for the additional lines.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_grid.col">grid.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
colour of the grid lines (originating at <code style="white-space: pre;">&#8288;[0,0]&#8288;</code> and stretching to
the z-scale). To disable grid lines use <code>FALSE</code>. Default is <code>"grey"</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_frame">frame</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
option to modify the plot frame type. Can be one out of
</p>

<ul>
<li> <p><code>0</code> (no frame),
</p>
</li>
<li> <p><code>1</code> (frame originates at 0,0 and runs along min/max isochrons),
</p>
</li>
<li> <p><code>2</code> (frame embraces the 2-sigma bar),
</p>
</li>
<li> <p><code>3</code> (frame embraces the entire plot as a rectangle).
</p>
</li></ul>

<p>Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_bw">bw</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
bin-width for KDE, choose a numeric value for manual setting.</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_interactive">interactive</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
create an interactive abanico plot (requires the <code>'plotly'</code> package)</p>
</td></tr>
<tr><td><code id="plot_AbanicoPlot_+3A_...">...</code></td>
<td>
<p>Further plot arguments to pass (see <a href="graphics.html#topic+plot.default">graphics::plot.default</a>).
Supported are: <code>main</code>, <code>sub</code>, <code>ylab</code>, <code>xlab</code>, <code>zlab</code>, <code>zlim</code>, <code>ylim</code>, <code>cex</code>,
<code>lty</code>, <code>lwd</code>, <code>pch</code>, <code>col</code>, <code>at</code>, <code>breaks</code>. <code>xlab</code> must be
a vector of length two, specifying the upper and lower x-axis labels.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Abanico Plot is a combination of the classic Radial Plot
(<code>plot_RadialPlot</code>) and a kernel density estimate plot (e.g
<code>plot_KDE</code>). It allows straightforward visualisation of data precision,
error scatter around a user-defined central value and the combined
distribution of the values, on the actual scale of the measured data (e.g.
seconds, equivalent dose, years). The principle of the plot is shown in
Galbraith &amp; Green (1990). The function authors are thankful for the
thought-provoking figure in this article.
</p>
<p>The semi circle (z-axis) of the classic Radial Plot is bent to a straight
line here, which actually is the basis for combining this polar (radial)
part of the plot with any other Cartesian visualisation method
(KDE, histogram, PDF and so on). Note that the plot allows displaying
two measures of distribution. One is the 2-sigma
bar, which illustrates the spread in value errors, and the other is the
polygon, which stretches over both parts of the Abanico Plot (polar and
Cartesian) and illustrates the actual spread in the values themselves.
</p>
<p>Since the 2-sigma-bar is a polygon, it can be (and is) filled with shaded
lines. To change density (lines per inch, default is 15) and angle (default
is 45 degrees) of the shading lines, specify these parameters. See
<code>?polygon()</code> for further help.
</p>
<p>The Abanico Plot supports other than the weighted mean as measure of
centrality. When it is obvious that the data
is not (log-)normally distributed, the mean (weighted or not) cannot be a
valid measure of centrality and hence central dose. Accordingly, the median
and the weighted median can be chosen as well to represent a proper measure
of centrality (e.g. <code>centrality = "median.weighted"</code>). Also
user-defined numeric values (e.g. from the central age model) can be used if
this appears appropriate.
</p>
<p>The proportion of the polar part and the cartesian part of the Abanico Plot
can be modified for display reasons (<code>plot.ratio = 0.75</code>). By default,
the polar part spreads over 75 % and leaves 25 % for the part that
shows the KDE graph.
</p>
<p>A statistic summary, i.e. a collection of statistic measures of
centrality and dispersion (and further measures) can be added by specifying
one or more of the following keywords:
</p>

<ul>
<li> <p><code>"n"</code> (number of samples)
</p>
</li>
<li> <p><code>"mean"</code> (mean De value)
</p>
</li>
<li> <p><code>"median"</code> (median of the De values)
</p>
</li>
<li> <p><code>"sd.rel"</code> (relative standard deviation in percent)
</p>
</li>
<li> <p><code>"sd.abs"</code> (absolute standard deviation)
</p>
</li>
<li> <p><code>"se.rel"</code> (relative standard error)
</p>
</li>
<li> <p><code>"se.abs"</code> (absolute standard error)
</p>
</li>
<li> <p><code>"in.2s"</code> (percent of samples in 2-sigma range)
</p>
</li>
<li> <p><code>"kurtosis"</code> (kurtosis)
</p>
</li>
<li> <p><code>"skewness"</code> (skewness)
</p>
</li></ul>

<p><strong>Note</strong> that the input data for the statistic summary is sent to the function
<code>calc_Statistics()</code> depending on the log-option for the z-scale. If
<code>"log.z = TRUE"</code>, the summary is based on the logarithms of the input
data. If <code>"log.z = FALSE"</code> the linearly scaled data is used.
</p>
<p><strong>Note</strong> as well, that <code>"calc_Statistics()"</code> calculates these statistic
measures in three different ways: <code>unweighted</code>, <code>weighted</code> and
<code>MCM-based</code> (i.e., based on Monte Carlo Methods). By default, the
MCM-based version is used. If you wish to use another method, indicate this
with the appropriate keyword using the argument <code>summary.method</code>.
</p>
<p>The optional parameter <code>layout</code> allows more sophisticated ways to modify
the entire plot. Each element of the plot can be addressed and its properties
can be defined. This includes font type, size and decoration, colours and
sizes of all plot items. To infer the definition of a specific layout style
cf. <code>get_Layout()</code> or type e.g., for the layout type <code>"journal"</code>
<code>get_Layout("journal")</code>. A layout type can be modified by the user by
assigning new values to the list object.
</p>
<p>It is possible for the z-scale to specify where ticks are to be drawn
by using the parameter <code>at</code>, e.g. <code>at = seq(80, 200, 20)</code>, cf. function
documentation of <code>axis</code>. Specifying tick positions manually overrides a
<code>zlim</code>-definition.
</p>


<h3>Value</h3>

<p>Returns a plot object and, optionally, a list with plot calculus data.
</p>


<h3>Function version</h3>

<p>0.1.18
</p>


<h3>How to cite</h3>

<p>Dietze, M., Kreutzer, S., 2025. plot_AbanicoPlot(): Function to create an Abanico Plot.. Function version 0.1.18. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)<br />
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Inspired by a plot introduced by Galbraith &amp; Green (1990)
, RLum Developer Team</p>


<h3>References</h3>

<p>Galbraith, R. &amp; Green, P., 1990. Estimating the component ages
in a finite mixture. International Journal of Radiation Applications and
Instrumentation. Part D. Nuclear Tracks and Radiation Measurements, 17 (3),
197-206.
</p>
<p>Dietze, M., Kreutzer, S., Burow, C., Fuchs, M.C., Fischer, M., Schmidt, C., 2015.
The abanico plot: visualising chronometric data with individual standard errors.
Quaternary Geochronology. doi:10.1016/j.quageo.2015.09.003
</p>


<h3>See Also</h3>

<p><a href="#topic+plot_RadialPlot">plot_RadialPlot</a>, <a href="#topic+plot_KDE">plot_KDE</a>, <a href="#topic+plot_Histogram">plot_Histogram</a>, <a href="#topic+plot_ViolinPlot">plot_ViolinPlot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data and recalculate to Gray
data(ExampleData.DeValues, envir = environment())
ExampleData.DeValues &lt;- ExampleData.DeValues$CA1

## plot the example data straightforward
plot_AbanicoPlot(data = ExampleData.DeValues)

## now with linear z-scale
plot_AbanicoPlot(data = ExampleData.DeValues,
                 log.z = FALSE)

## now with output of the plot parameters
plot1 &lt;- plot_AbanicoPlot(data = ExampleData.DeValues,
                          output = TRUE)
str(plot1)
plot1$zlim

## now with adjusted z-scale limits
plot_AbanicoPlot(data = ExampleData.DeValues,
                 zlim = c(10, 200))

## now with adjusted x-scale limits
plot_AbanicoPlot(data = ExampleData.DeValues,
                 xlim = c(0, 20))

## now with rug to indicate individual values in KDE part
plot_AbanicoPlot(data = ExampleData.DeValues,
                 rug = TRUE)

## now with a smaller bandwidth for the KDE plot
plot_AbanicoPlot(data = ExampleData.DeValues,
                 bw = 0.04)

## now with a histogram instead of the KDE plot
plot_AbanicoPlot(data = ExampleData.DeValues,
                 hist = TRUE,
                 kde = FALSE)

## now with a KDE plot and histogram with manual number of bins
plot_AbanicoPlot(data = ExampleData.DeValues,
                 hist = TRUE,
                 breaks = 20)

## now with a KDE plot and a dot plot
plot_AbanicoPlot(data = ExampleData.DeValues,
                 dots = TRUE)

## now with user-defined plot ratio
plot_AbanicoPlot(data = ExampleData.DeValues,
                 plot.ratio = 0.5)
## now with user-defined central value
plot_AbanicoPlot(data = ExampleData.DeValues,
                 z.0 = 70)

## now with median as central value
plot_AbanicoPlot(data = ExampleData.DeValues,
                 z.0 = "median")

## now with the 17-83 percentile range as definition of scatter
plot_AbanicoPlot(data = ExampleData.DeValues,
                 z.0 = "median",
                 dispersion = "p17")

## now with user-defined green line for minimum age model
CAM &lt;- calc_CentralDose(ExampleData.DeValues,
                        plot = FALSE)

plot_AbanicoPlot(data = ExampleData.DeValues,
                 line = CAM,
                 line.col = "darkgreen",
                 line.label = "CAM")

## now create plot with legend, colour, different points and smaller scale
plot_AbanicoPlot(data = ExampleData.DeValues,
                 legend = "Sample 1",
                 col = "tomato4",
                 bar.col = "peachpuff",
                 pch = "R",
                 cex = 0.8)

## now without 2-sigma bar, polygon, grid lines and central value line
plot_AbanicoPlot(data = ExampleData.DeValues,
                 bar.col = FALSE,
                 polygon.col = FALSE,
                 grid.col = FALSE,
                 y.axis = FALSE,
                 lwd = 0)

## now with direct display of De errors, without 2-sigma bar
plot_AbanicoPlot(data = ExampleData.DeValues,
                 bar.col = FALSE,
                 ylab = "",
                 y.axis = FALSE,
                 error.bars = TRUE)

## now with user-defined axes labels
plot_AbanicoPlot(data = ExampleData.DeValues,
                 xlab = c("Data error (%)",
                          "Data precision"),
                 ylab = "Scatter",
                 zlab = "Equivalent dose [Gy]")

## now with minimum, maximum and median value indicated
plot_AbanicoPlot(data = ExampleData.DeValues,
                 stats = c("min", "max", "median"))

## now with a brief statistical summary as subheader
plot_AbanicoPlot(data = ExampleData.DeValues,
                 summary = c("n", "in.2s"))

## now with another statistical summary
plot_AbanicoPlot(data = ExampleData.DeValues,
                 summary = c("mean.weighted", "median"),
                 summary.pos = "topleft")

## now a plot with two 2-sigma bars for one data set
plot_AbanicoPlot(data = ExampleData.DeValues,
                 bar = c(30, 100))

## now the data set is split into sub-groups, one is manipulated
data.1 &lt;- ExampleData.DeValues[1:30,]
data.2 &lt;- ExampleData.DeValues[31:62,] * 1.3

## now a common dataset is created from the two subgroups
data.3 &lt;- list(data.1, data.2)

## now the two data sets are plotted in one plot
plot_AbanicoPlot(data = data.3)

## now with some graphical modification
plot_AbanicoPlot(data = data.3,
                 z.0 = "median",
                 col = c("steelblue4", "orange4"),
                 bar.col = c("steelblue3", "orange3"),
                 polygon.col = c("steelblue1", "orange1"),
                 pch = c(2, 6),
                 angle = c(30, 50),
                 summary = c("n", "in.2s", "median"))

## create Abanico plot with predefined layout definition
plot_AbanicoPlot(data = ExampleData.DeValues,
                 layout = "journal")

## now with predefined layout definition and further modifications
plot_AbanicoPlot(
 data = data.3,
 z.0 = "median",
 layout = "journal",
 col = c("steelblue4", "orange4"),
 bar.col = adjustcolor(c("steelblue3", "orange3"),
                         alpha.f = 0.5),
 polygon.col = c("steelblue3", "orange3"))

## for further information on layout definitions see documentation
## of function get_Layout()

## now with manually added plot content
## create empty plot with numeric output
AP &lt;- plot_AbanicoPlot(data = ExampleData.DeValues,
                       pch = NA,
                       output = TRUE)

## identify data in 2 sigma range
in_2sigma &lt;- AP$data[[1]]$data.in.2s

## restore function-internal plot parameters
par(AP$par)

## add points inside 2-sigma range
points(x = AP$data[[1]]$precision[in_2sigma],
       y = AP$data[[1]]$std.estimate.plot[in_2sigma],
       pch = 16)

## add points outside 2-sigma range
points(x = AP$data[[1]]$precision[!in_2sigma],
       y = AP$data[[1]]$std.estimate.plot[!in_2sigma],
       pch = 1)

</code></pre>

<hr>
<h2 id='plot_DetPlot'>Create De(t) plot</h2><span id='topic+plot_DetPlot'></span>

<h3>Description</h3>

<p>Plots the equivalent dose (<code class="reqn">D_e</code>) in dependency of the chosen signal integral
(cf. Bailey et al., 2003). The function is simply passing several arguments
to the function <a href="base.html#topic+plot">plot</a> and the used analysis functions and runs it in a loop.
Example: <code>legend.pos</code> for legend position, <code>legend</code> for legend text.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_DetPlot(
  object,
  signal.integral.min,
  signal.integral.max,
  background.integral.min,
  background.integral.max,
  method = "shift",
  signal_integral.seq = NULL,
  analyse_function = "analyse_SAR.CWOSL",
  analyse_function.control = list(),
  n.channels = NULL,
  show_ShineDownCurve = TRUE,
  respect_RC.Status = FALSE,
  multicore = TRUE,
  verbose = TRUE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_DetPlot_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>): input object containing data for analysis
Can be provided as a <a href="base.html#topic+list">list</a> of such objects.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_signal.integral.min">signal.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
lower bound of the signal integral.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_signal.integral.max">signal.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
upper bound of the signal integral. Must be strictly greater than
<code>signal.integral.min</code>.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_background.integral.min">background.integral.min</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
lower bound of the background integral.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_background.integral.max">background.integral.max</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<strong>required</strong>):
upper bound of the background integral.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_method">method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
method applied for constructing the De(t) plot.
</p>

<ul>
<li> <p><code>shift</code> (<em>the default</em>): the chosen signal integral is shifted the shine down curve,
</p>
</li>
<li> <p><code>expansion</code>: the chosen signal integral is expanded each time by its length
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_signal_integral.seq">signal_integral.seq</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
argument to provide an own signal integral sequence for constructing the De(t) plot</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_analyse_function">analyse_function</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
name of the analyse function to be called. Supported functions are:
<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>, <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a></p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_analyse_function.control">analyse_function.control</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
selected arguments to be passed to the supported analyse functions
(<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>, <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a>). The arguments must be provided
as named <a href="base.html#topic+list">list</a>, e.g., <code style="white-space: pre;">&#8288;list(dose.points = c(0,10,20,30,0,10)&#8288;</code> will set the
regeneration dose points.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_n.channels">n.channels</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
number of channels used for the De(t) plot. If nothing is provided all
De-values are calculated and plotted until the start of the background
integral.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_show_shinedowncurve">show_ShineDownCurve</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable shine down curve in the plot output.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_respect_rc.status">respect_RC.Status</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
remove De values with 'FAILED' RC.Status from the plot (cf. <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>
and <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a>).</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_multicore">multicore</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>) : enable/disable multi core
calculation if <code>object</code> is a <a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects. Can be an
<a href="base.html#topic+integer">integer</a> specifying the number of cores</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.
Disabling the plot is useful in cases where the output need to be processed
differently.</p>
</td></tr>
<tr><td><code id="plot_DetPlot_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to
<a href="graphics.html#topic+plot.default">plot.default</a>, <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> and <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a> (see details for further information).
Plot control parameters are: <code>ylim</code>, <code>xlim</code>, <code>ylab</code>, <code>xlab</code>, <code>main</code>, <code>pch</code>, <code>mtext</code>, <code>cex</code>, <code>legend</code>,
<code>legend.text</code>, <code>legend.pos</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>method</strong>
</p>
<p>The original method presented by Bailey et al., 2003 shifted the signal integrals and slightly
extended them accounting for changes in the counting statistics. Example: <code>c(1:3, 3:5, 5:7)</code>.
However, here also another method is provided allowing to expand the signal integral by
consecutively expanding the integral by its chosen length. Example: <code>c(1:3, 1:5, 1:7)</code>
</p>
<p>Note that in both cases the integral limits are overlap. The finally applied limits are part
of the function output.
</p>
<p><strong>analyse_function.control</strong>
</p>
<p>The argument <code>analyse_function.control</code> currently supports the following arguments:
<code>sequence.structure</code>, <code>dose.points</code>, <code>mtext.outer</code>, <code>fit.method</code>,
<code>fit.force_through_origin</code>, <code>plot</code>, <code>plot_singlePanels</code>
</p>


<h3>Value</h3>

<p>A plot and an <a href="#topic+RLum.Results-class">RLum.Results</a> object with the produced <code class="reqn">D_e</code> values
</p>
<p><code style="white-space: pre;">&#8288;@data&#8288;</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>De.values</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> table with De values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>signal_integral.seq</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> integral sequence used for the calculation
</td>
</tr>

</table>

<p><code style="white-space: pre;">&#8288;@info&#8288;</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
call </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.1.8
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_DetPlot(): Create De(t) plot. Function version 0.1.8. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The entire analysis is based on the used analysis functions, namely
<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> and <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a>. However, the integrity
checks of this function are not that thoughtful as in these functions itself.
It means, that every sequence should be checked carefully before running long
calculations using several hundreds of channels.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Ruprecht-Karl University of Heidelberg (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Bailey, R.M., Singarayer, J.S., Ward, S., Stokes, S., 2003. Identification of partial resetting
using De as a function of illumination time. Radiation Measurements 37, 511-518.
doi:10.1016/S1350-4487(03)00063-5
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>, <a href="#topic+analyse_pIRIRSequence">analyse_pIRIRSequence</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
##load data
##ExampleData.BINfileData contains two BINfileData objects
##CWOSL.SAR.Data and TL.SAR.Data
data(ExampleData.BINfileData, envir = environment())

##transform the values from the first position in a RLum.Analysis object
object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)

plot_DetPlot(
  object,
  signal.integral.min = 1,
  signal.integral.max = 3,
  background.integral.min = 900,
  background.integral.max = 1000,
  n.channels = 5)

## End(Not run)

</code></pre>

<hr>
<h2 id='plot_DoseResponseCurve'>Plot a dose-response curve for luminescence data (Lx/Tx against dose)</h2><span id='topic+plot_DoseResponseCurve'></span>

<h3>Description</h3>

<p>A dose-response curve is produced for luminescence measurements using a
regenerative or additive protocol as implemented in <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_DoseResponseCurve(
  object,
  plot_extended = TRUE,
  plot_singlePanels = FALSE,
  cex.global = 1,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_DoseResponseCurve_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> (<strong>required</strong>):
An object produced by <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>.</p>
</td></tr>
<tr><td><code id="plot_DoseResponseCurve_+3A_plot_extended">plot_extended</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If <code>TRUE</code>, 3 plots on one plot area are provided:
</p>

<ol>
<li><p> growth curve,
</p>
</li>
<li><p> histogram from Monte Carlo error simulation and
</p>
</li>
<li><p> a test dose response plot.
</p>
</li></ol>

<p>If <code>FALSE</code>, just the growth curve will be plotted.</p>
</td></tr>
<tr><td><code id="plot_DoseResponseCurve_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
single plot output (<code>TRUE/FALSE</code>) to allow for plotting the results in
single plot windows. Ignored if <code>plot_extended = FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_DoseResponseCurve_+3A_cex.global">cex.global</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
global scaling factor.</p>
</td></tr>
<tr><td><code id="plot_DoseResponseCurve_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="plot_DoseResponseCurve_+3A_...">...</code></td>
<td>
<p>Further graphical parameters to be passed (supported:
<code>main</code>, <code>mtext</code>, <code>xlim</code>, <code>ylim</code>, <code>xlab</code>, <code>ylab</code>, <code>legend</code>, <code>reg_points_pch</code>,
<code>density_polygon</code> (<code>TRUE/FALSE</code>), <code>density_polygon_col</code>, <code>density_rug</code> (<code>TRUE</code>/<code>FALSE</code>)),
<code>box</code> (<code>TRUE</code>/<code>FALSE</code>)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot (or a series of plots) is produced.
</p>


<h3>Function version</h3>

<p>1.0.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., Colombo, M., 2025. plot_DoseResponseCurve(): Plot a dose-response curve for luminescence data (Lx/Tx against dose). Function version 1.0.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany) <br />
Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Berger, G.W., Huntley, D.J., 1989. Test data for exponential fits. Ancient TL 7, 43-46.
</p>
<p>Guralnik, B., Li, B., Jain, M., Chen, R., Paris, R.B., Murray, A.S., Li, S.-H., Pagonis, P.,
Herman, F., 2015. Radiation-induced growth and isothermal decay of infrared-stimulated luminescence
from feldspar. Radiation Measurements 81, 224-231.
</p>
<p>Pagonis, V., Kitis, G., Chen, R., 2020. A new analytical equation for the dose response of dosimetric materials,
based on the Lambert W function. Journal of Luminescence 225, 117333. <a href="https://doi.org/10.1016/j.jlumin.2020.117333">doi:10.1016/j.jlumin.2020.117333</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) plot dose-response curve for a dummy dataset
data(ExampleData.LxTxData, envir = environment())
fit &lt;- fit_DoseResponseCurve(LxTxData)
plot_DoseResponseCurve(fit)

##(1b) horizontal plot arrangement
layout(mat = matrix(c(1,1,2,3), ncol = 2))
plot_DoseResponseCurve(fit, plot_singlePanels = TRUE)

##(2) plot the dose-response curve with pdf output - uncomment to use
##pdf(file = "~/Dose_Response_Curve_Dummy.pdf", paper = "special")
plot_DoseResponseCurve(fit)
##dev.off()

##(3) plot the growth curve with pdf output - uncomment to use, single output
##pdf(file = "~/Dose_Response_Curve_Dummy.pdf", paper = "special")
plot_DoseResponseCurve(fit, plot_singlePanels = TRUE)
##dev.off()

##(4) plot resulting function for given interval x
x &lt;- seq(1,10000, by = 100)
plot(
 x = x,
 y = eval(fit$Formula),
 type = "l"
)

</code></pre>

<hr>
<h2 id='plot_DRCSummary'>Create a Dose-Response Curve Summary Plot</h2><span id='topic+plot_DRCSummary'></span>

<h3>Description</h3>

<p>While analysing OSL SAR or pIRIR-data the view on the data is usually
limited to one dose-response curve (DRC) at the time for one aliquot. This
function overcomes this limitation by plotting all DRCs from an
<a href="#topic+RLum.Results-class">RLum.Results</a> object created by <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a> in one single
plot.
</p>
<p>If you want plot your DRC on an energy scale (dose in Gy), you can either
use option <code>source_dose_rate</code> or perform your SAR analysis with the dose
points in Gy (better axis scaling).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_DRCSummary(
  object,
  source_dose_rate = NULL,
  sel_curves = NULL,
  show_dose_points = FALSE,
  show_natural = FALSE,
  n = 51L,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_DRCSummary_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> (<strong>required</strong>): input object created by
<a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>. The input object can be provided as <a href="base.html#topic+list">list</a>.</p>
</td></tr>
<tr><td><code id="plot_DRCSummary_+3A_source_dose_rate">source_dose_rate</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): allows to modify the axis
and show values in Gy, instead seconds. Only a single numerical value is
allowed.</p>
</td></tr>
<tr><td><code id="plot_DRCSummary_+3A_sel_curves">sel_curves</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): id of the curves to be plotted in
its occurring order. A sequence can be provided for selecting, e.g., only
every 2nd curve from the input object.</p>
</td></tr>
<tr><td><code id="plot_DRCSummary_+3A_show_dose_points">show_dose_points</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable plotting
of dose points in the graph.</p>
</td></tr>
<tr><td><code id="plot_DRCSummary_+3A_show_natural">show_natural</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot
of the natural <code>Lx/Tx</code> values.</p>
</td></tr>
<tr><td><code id="plot_DRCSummary_+3A_n">n</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): number of x-values used to evaluate
one curve object. Large numbers slow down the plotting process and are
usually not needed.</p>
</td></tr>
<tr><td><code id="plot_DRCSummary_+3A_...">...</code></td>
<td>
<p>Further arguments and graphical parameters to be passed. In particular: <code>main</code>, <code>xlab</code>, <code>ylab</code>, <code>xlim</code>, <code>ylim</code>, <code>lty</code>, <code>lwd</code>, <code>pch</code>, <code>col.pch</code>, <code>col.lty</code>, <code>mtext</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <a href="#topic+RLum.Results-class">RLum.Results</a> object is returned:
</p>
<p>Slot: <strong>@data</strong><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>COMMENT</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>results</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> with dose and LxTx values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data</code> </td><td style="text-align: left;"> <a href="#topic+RLum.Results-class">RLum.Results</a> </td><td style="text-align: left;"> original input data </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Slot: <strong>@info</strong><br />
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>OBJECT</strong> </td><td style="text-align: left;"> <strong>TYPE</strong> </td><td style="text-align: left;"> <strong>COMMENT</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>call</code> </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>args</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> arguments of the original function call </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><em>Note: If the input object is a <a href="base.html#topic+list">list</a> a list of <a href="#topic+RLum.Results-class">RLum.Results</a> objects is returned.</em>
</p>


<h3>Function version</h3>

<p>0.2.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Burow, C., 2025. plot_DRCSummary(): Create a Dose-Response Curve Summary Plot. Function version 0.2.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany) <br />
Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#load data example data
data(ExampleData.BINfileData, envir = environment())

#transform the values from the first position in a RLum.Analysis object
object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)

results &lt;- analyse_SAR.CWOSL(
  object = object,
  signal.integral.min = 1,
  signal.integral.max = 2,
   background.integral.min = 900,
   background.integral.max = 1000,
   plot = FALSE
 )

##plot only DRC
plot_DRCSummary(results)

</code></pre>

<hr>
<h2 id='plot_DRTResults'>Visualise dose recovery test results</h2><span id='topic+plot_DRTResults'></span>

<h3>Description</h3>

<p>The function provides a standardised plot output for dose recovery test
measurements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_DRTResults(
  values,
  given.dose = NULL,
  error.range = 10,
  preheat,
  boxplot = FALSE,
  mtext,
  summary,
  summary.pos,
  legend,
  legend.pos,
  par.local = TRUE,
  na.rm = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_DRTResults_+3A_values">values</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
input values containing at least De and De error. To plot
more than one data set in one figure, a <code>list</code> of the individual data
sets must be provided (e.g. <code>list(dataset.1, dataset.2)</code>).</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_given.dose">given.dose</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
given dose used for the dose recovery test to normalise data.
If only one given dose is provided this given dose is valid for all input
data sets (i.e., <code>values</code> is a list).  Otherwise a given dose for each input
data set has to be provided (e.g., <code>given.dose = c(100,200)</code>).
If <code>given.dose</code> in <code>NULL</code> the values are plotted without normalisation
(might be useful for preheat plateau tests).
<strong>Note:</strong> Unit has to be the same as from the input values (e.g., Seconds or
Gray).</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_error.range">error.range</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
symmetric error range in percent will be shown as dashed lines in the plot.
Set <code>error.range</code> to 0 to void plotting of error ranges.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_preheat">preheat</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
optional vector of preheat temperatures to be used for grouping the De values.
If specified, the temperatures are assigned to the x-axis.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_boxplot">boxplot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
plot values that are grouped by preheat temperature as boxplots.
Only possible when <code>preheat</code> vector is specified.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_mtext">mtext</code></td>
<td>
<p><a href="base.html#topic+character">character</a>:
additional text below the plot title.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
adds numerical output to the plot.  Can be one or more out of:
</p>

<ul>
<li> <p><code>"n"</code> (number of samples),
</p>
</li>
<li> <p><code>"mean"</code> (mean De value),
</p>
</li>
<li> <p><code>"weighted$mean"</code> (error-weighted mean),
</p>
</li>
<li> <p><code>"median"</code> (median of the De values),
</p>
</li>
<li> <p><code>"sd.rel"</code> (relative standard deviation in percent),
</p>
</li>
<li> <p><code>"sd.abs"</code> (absolute standard deviation),
</p>
</li>
<li> <p><code>"se.rel"</code> (relative standard error) and
</p>
</li>
<li> <p><code>"se.abs"</code> (absolute standard error)
</p>
</li></ul>

<p>and all other measures returned by the function <a href="#topic+calc_Statistics">calc_Statistics</a>.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_summary.pos">summary.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>)
for the statistical summary. Alternatively, the keyword <code>"sub"</code> may be
specified to place the summary below the plot header. However, this latter
option in only possible if <code>mtext</code> is not used.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_legend">legend</code></td>
<td>
<p><a href="base.html#topic+character">character</a> vector (<em>optional</em>):
legend content to be added to the plot.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_legend.pos">legend.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>) for the
legend to be plotted.</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_par.local">par.local</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
use local graphical parameters for plotting, e.g. the plot is shown in one
column and one row. If <code>par.local = FALSE</code>, global parameters are inherited,
i.e. parameters provided via <code>par()</code> work</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): indicating whether <code>NA</code> values are
removed before plotting from the input data set</p>
</td></tr>
<tr><td><code id="plot_DRTResults_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to <a href="base.html#topic+plot">plot</a>, supported are:
<code>xlab</code>, <code>ylab</code>, <code>xlim</code>, <code>ylim</code>, <code>main</code>, <code>cex</code>, <code>las</code> and 'pch&ldquo;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Procedure to test the accuracy of a measurement protocol to reliably
determine the dose of a specific sample. Here, the natural signal is erased
and a known laboratory dose administered which is treated as unknown. Then
the De measurement is carried out and the degree of congruence between
administered and recovered dose is a measure of the protocol's accuracy for
this sample.<br />
In the plot the normalised De is shown on the y-axis, i.e. obtained De/Given Dose.
</p>


<h3>Value</h3>

<p>A plot is returned.
</p>


<h3>Function version</h3>

<p>0.1.14
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., 2025. plot_DRTResults(): Visualise dose recovery test results. Function version 0.1.14. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Further data and plot arguments can be added by using the appropriate R
commands.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Wintle, A.G., Murray, A.S., 2006. A review of quartz optically
stimulated luminescence characteristics and their relevance in
single-aliquot regeneration dating protocols. Radiation Measurements, 41,
369-391.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## read example data set and misapply them for this plot type
data(ExampleData.DeValues, envir = environment())

## plot values
plot_DRTResults(
  values = ExampleData.DeValues$BT998[7:11,],
  given.dose = 2800,
  mtext = "Example data")

## plot values with legend
plot_DRTResults(
  values = ExampleData.DeValues$BT998[7:11,],
  given.dose = 2800,
  legend = "Test data set")

## create and plot two subsets with randomised values
x.1 &lt;- ExampleData.DeValues$BT998[7:11,]
x.2 &lt;- ExampleData.DeValues$BT998[7:11,] * c(runif(5, 0.9, 1.1), 1)

plot_DRTResults(
  values = list(x.1, x.2),
  given.dose = 2800)

## some more user-defined plot parameters
plot_DRTResults(
  values = list(x.1, x.2),
  given.dose = 2800,
  pch = c(2, 5),
  col = c("orange", "blue"),
  xlim = c(0, 8),
  ylim = c(0.85, 1.15),
  xlab = "Sample aliquot")

## plot the data with user-defined statistical measures as legend
plot_DRTResults(
  values = list(x.1, x.2),
  given.dose = 2800,
  summary = c("n", "weighted$mean", "sd.abs"))

## plot the data with user-defined statistical measures as sub-header
plot_DRTResults(
  values = list(x.1, x.2),
  given.dose = 2800,
  summary = c("n", "weighted$mean", "sd.abs"),
  summary.pos = "sub")

## plot the data grouped by preheat temperatures
plot_DRTResults(
  values = ExampleData.DeValues$BT998[7:11,],
  given.dose = 2800,
  preheat = c(200, 200, 200, 240, 240))

## read example data set and misapply them for this plot type
data(ExampleData.DeValues, envir = environment())

## plot values
plot_DRTResults(
  values = ExampleData.DeValues$BT998[7:11,],
  given.dose = 2800,
  mtext = "Example data")

## plot two data sets grouped by preheat temperatures
plot_DRTResults(
  values = list(x.1, x.2),
  given.dose = 2800,
  preheat = c(200, 200, 200, 240, 240))

## plot the data grouped by preheat temperatures as boxplots
plot_DRTResults(
  values = ExampleData.DeValues$BT998[7:11,],
  given.dose = 2800,
  preheat = c(200, 200, 200, 240, 240),
  boxplot = TRUE)

</code></pre>

<hr>
<h2 id='plot_FilterCombinations'>Plot filter combinations along with the (optional) net transmission window</h2><span id='topic+plot_FilterCombinations'></span>

<h3>Description</h3>

<p>The function allows to plot transmission windows for different filters. Missing data for specific
wavelengths are automatically interpolated for the given filter data using the function <a href="stats.html#topic+approx">approx</a>.
With that a standardised output is reached and a net transmission window can be shown.
</p>
<p><strong>Calculations</strong>
</p>
<p><strong>Net transmission window</strong>
</p>
<p>The net transmission window of two filters is approximated by
</p>
<p style="text-align: center;"><code class="reqn">T_{final} = T_{1} * T_{2}</code>
</p>

<p><strong>Optical density</strong>
</p>
<p style="text-align: center;"><code class="reqn">OD = -log10(T)</code>
</p>

<p><strong>Total optical density</strong>
</p>
<p style="text-align: center;"><code class="reqn">OD_{total} = OD_{1} +  OD_{2}</code>
</p>

<p>Please consider using own calculations for more precise values.
</p>
<p><strong>How to provide input data?</strong>
</p>
<p><em>CASE 1</em>
</p>
<p>The function expects that all filter values are either of type <code>matrix</code> or <code>data.frame</code>
with two columns. The first columns contains the wavelength, the second the relative transmission
(but not in percentage, i.e. the maximum transmission can be only become 1).
</p>
<p>In this case only the transmission window is show as provided. Changes in filter thickness and
reflection factor are not considered.
</p>
<p><em>CASE 2</em>
</p>
<p>The filter data itself are provided as list element containing a <code>matrix</code> or
<code>data.frame</code> and additional information on the thickness of the filter, e.g.,
<code>list(filter1 = list(filter_matrix, d = 2))</code>.
The given filter data are always considered as standard input and the filter thickness value
is taken into account by
</p>
<p style="text-align: center;"><code class="reqn">Transmission = Transmission^(d)</code>
</p>

<p>with d given in the same dimension as the original filter data.
</p>
<p><em>CASE 3</em>
</p>
<p>Same as CASE 2 but additionally a reflection factor P is provided, e.g.,
<code>list(filter1 = list(filter_matrix, d = 2, P = 0.9))</code>.
The final transmission becomes:
</p>
<p style="text-align: center;"><code class="reqn">Transmission = Transmission^(d) * P</code>
</p>

<p><strong>Advanced plotting parameters</strong>
</p>
<p>The following further non-common plotting parameters can be passed to the function:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong><code>Argument</code></strong> </td><td style="text-align: left;"> <strong><code>Datatype</code></strong> </td><td style="text-align: left;"> <strong><code>Description</code></strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>legend</code> </td><td style="text-align: left;"> <code>logical</code> </td><td style="text-align: left;"> enable/disable legend </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>legend.pos</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> change legend position (<a href="graphics.html#topic+legend">graphics::legend</a>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>legend.text</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> same as the argument <code>legend</code> in (<a href="graphics.html#topic+legend">graphics::legend</a>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>net_transmission.col</code> </td><td style="text-align: left;"> <code>col</code> </td><td style="text-align: left;"> colour of net transmission window polygon </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>net_transmission.col_lines</code> </td><td style="text-align: left;"> <code>col</code> </td><td style="text-align: left;"> colour of net transmission window polygon lines </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>net_transmission.density</code> </td><td style="text-align: left;">  <code>numeric</code> </td><td style="text-align: left;"> specify line density in the transmission polygon </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>grid</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> full list of arguments that can be passed to the function <a href="graphics.html#topic+grid">graphics::grid</a>
</td>
</tr>

</table>

<p>For further modifications standard additional R plot functions are recommend, e.g., the legend
can be fully customised by disabling the standard legend and use the function <a href="graphics.html#topic+legend">graphics::legend</a>
instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_FilterCombinations(
  filters,
  wavelength_range = 200:1000,
  show_net_transmission = TRUE,
  interactive = FALSE,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_FilterCombinations_+3A_filters">filters</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<strong>required</strong>):
a named list of filter data for each filter to be shown.
The filter data itself should be either provided as <a href="base.html#topic+data.frame">data.frame</a> or <a href="base.html#topic+matrix">matrix</a>
(see details for more options).</p>
</td></tr>
<tr><td><code id="plot_FilterCombinations_+3A_wavelength_range">wavelength_range</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
wavelength range used for the interpolation</p>
</td></tr>
<tr><td><code id="plot_FilterCombinations_+3A_show_net_transmission">show_net_transmission</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
show net transmission window as polygon.</p>
</td></tr>
<tr><td><code id="plot_FilterCombinations_+3A_interactive">interactive</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable interactive plots.</p>
</td></tr>
<tr><td><code id="plot_FilterCombinations_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="plot_FilterCombinations_+3A_...">...</code></td>
<td>
<p>further arguments that can be passed to control the plot output.
Supported are <code>main</code>, <code>xlab</code>, <code>ylab</code>, <code>xlim</code>, <code>ylim</code>, <code>type</code>, <code>lty</code>, <code>lwd</code>.
For non common plotting parameters see the details section.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an S4 object of type <a href="#topic+RLum.Results-class">RLum.Results</a>.
</p>
<p><strong>@data</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong><code>Object</code></strong> </td><td style="text-align: left;"> <strong><code>Type</code></strong> <strong><code>Description</code></strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>net_transmission_window</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the resulting net transmission window </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>OD_total</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the total optical density</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>filter_matrix</code> </td><td style="text-align: left;"> <code>matrix</code> </td><td style="text-align: left;"> the filter matrix used for plotting
</td>
</tr>

</table>

<p><strong>@info</strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Object</strong> </td><td style="text-align: left;"> <strong>Type</strong> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>call</code> </td><td style="text-align: left;"> <a href="base.html#topic+call">call</a> </td><td style="text-align: left;"> the original function call
</td>
</tr>

</table>



<h3>Function version</h3>

<p>0.3.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_FilterCombinations(): Plot filter combinations along with the (optional) net transmission window. Function version 0.3.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="stats.html#topic+approx">approx</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## (For legal reasons no real filter data are provided)

## Create filter sets
filter1 &lt;- density(rnorm(100, mean = 450, sd = 20))
filter1 &lt;- matrix(c(filter1$x, filter1$y/max(filter1$y)), ncol = 2)
filter2 &lt;- matrix(c(200:799,rep(c(0,0.8,0),each = 200)), ncol = 2)

## Example 1 (standard)
plot_FilterCombinations(filters = list(filter1, filter2))

## Example 2 (with d and P value and name for filter 2)
results &lt;- plot_FilterCombinations(
filters = list(filter_1 = filter1, Rectangle = list(filter2, d = 2, P = 0.6)))
results

## Example 3 show optical density
plot(results$OD_total)

## Not run: 
##Example 4
##show the filters using the interactive mode
plot_FilterCombinations(filters = list(filter1, filter2), interactive = TRUE)


## End(Not run)


</code></pre>

<hr>
<h2 id='plot_GrowthCurve'>Fit and plot a dose-response curve for luminescence data (Lx/Tx against dose)</h2><span id='topic+plot_GrowthCurve'></span>

<h3>Description</h3>

<p>A dose-response curve is produced for luminescence measurements using a
regenerative or additive protocol as implemented in <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>
and <a href="#topic+plot_DoseResponseCurve">plot_DoseResponseCurve</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_GrowthCurve(
  sample,
  mode = "interpolation",
  fit.method = "EXP",
  output.plot = TRUE,
  output.plotExtended = TRUE,
  plot_singlePanels = FALSE,
  cex.global = 1,
  verbose = TRUE,
  n.MC = 100,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_GrowthCurve_+3A_sample">sample</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
data frame with columns for <code>Dose</code>, <code>LxTx</code>, <code>LxTx.Error</code> and <code>TnTx</code>.
The column for the test dose response is optional, but requires <code>'TnTx'</code> as
column name if used. For exponential fits at least three dose points
(including the natural) should be provided.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_mode">mode</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
selects calculation mode of the function.
</p>

<ul>
<li> <p><code>"interpolation"</code> (default) calculates the De by interpolation,
</p>
</li>
<li> <p><code>"extrapolation"</code> calculates the equivalent dose by extrapolation (useful for MAAD measurements) and
</p>
</li>
<li> <p><code>"alternate"</code> calculates no equivalent dose and just fits the data points.
</p>
</li></ul>

<p>Please note that for option <code>"interpolation"</code> the first point is considered
as natural dose</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_fit.method">fit.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
function used for fitting. Possible options are:
</p>

<ul>
<li> <p><code>LIN</code>,
</p>
</li>
<li> <p><code>QDR</code>,
</p>
</li>
<li> <p><code>EXP</code>,
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;EXP OR LIN&#8288;</code>,
</p>
</li>
<li> <p><code>EXP+LIN</code>,
</p>
</li>
<li> <p><code>EXP+EXP</code>,
</p>
</li>
<li> <p><code>GOK</code>,
</p>
</li>
<li> <p><code>LambertW</code>
</p>
</li></ul>

<p>See details in <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_output.plot">output.plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_output.plotextended">output.plotExtended</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If' <code>TRUE</code>, 3 plots on one plot area are provided:
</p>

<ol>
<li><p> growth curve,
</p>
</li>
<li><p> histogram from Monte Carlo error simulation and
</p>
</li>
<li><p> a test dose response plot.
</p>
</li></ol>

<p>If <code>FALSE</code>, just the growth curve will be plotted.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
single plot output (<code>TRUE/FALSE</code>) to allow for plotting the results in
single plot windows. Requires <code>plotExtended = TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_cex.global">cex.global</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
global scaling factor.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_n.mc">n.MC</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
number of MC runs for error calculation.</p>
</td></tr>
<tr><td><code id="plot_GrowthCurve_+3A_...">...</code></td>
<td>
<p>Further arguments to <a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a> (<code>fit_weights</code>,
<code>fit_bounds</code>, <code>fit.force_through_origin</code>, <code>fit.includingRepeatedRegPoints</code>,
<code>fit.NumberRegPoints</code>, <code>fit.NumberRegPointsReal</code>, <code>n.MC</code>,
<code>txtProgressBar</code>) and graphical parameters to be passed (supported:
<code>xlim</code>, <code>ylim</code>, <code>main</code>, <code>xlab</code>, <code>ylab</code>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Along with a plot (if wanted) the <code>RLum.Results</code> object produced by
<a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a> is returned.
</p>


<h3>Function version</h3>

<p>1.2.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., Colombo, M., 2025. plot_GrowthCurve(): Fit and plot a dose-response curve for luminescence data (Lx/Tx against dose). Function version 1.2.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany) <br />
Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Berger, G.W., Huntley, D.J., 1989. Test data for exponential fits. Ancient TL 7, 43-46.
</p>
<p>Guralnik, B., Li, B., Jain, M., Chen, R., Paris, R.B., Murray, A.S., Li, S.-H., Pagonis, P.,
Herman, F., 2015. Radiation-induced growth and isothermal decay of infrared-stimulated luminescence
from feldspar. Radiation Measurements 81, 224-231.
</p>
<p>Pagonis, V., Kitis, G., Chen, R., 2020. A new analytical equation for the dose response of dosimetric materials,
based on the Lambert W function. Journal of Luminescence 225, 117333. <a href="https://doi.org/10.1016/j.jlumin.2020.117333">doi:10.1016/j.jlumin.2020.117333</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+fit_DoseResponseCurve">fit_DoseResponseCurve</a>, <a href="#topic+plot_DoseResponseCurve">plot_DoseResponseCurve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) plot growth curve for a dummy dataset
data(ExampleData.LxTxData, envir = environment())
plot_GrowthCurve(LxTxData)

##(1b) horizontal plot arrangement
layout(mat = matrix(c(1,1,2,3), ncol = 2))
plot_GrowthCurve(LxTxData, plot_singlePanels = TRUE)

##(2) plot the growth curve with pdf output - uncomment to use
##pdf(file = "~/Desktop/Growth_Curve_Dummy.pdf", paper = "special")
plot_GrowthCurve(LxTxData)
##dev.off()

##(3) plot the growth curve with pdf output - uncomment to use, single output
##pdf(file = "~/Desktop/Growth_Curve_Dummy.pdf", paper = "special")
temp &lt;- plot_GrowthCurve(LxTxData, plot_singlePanels = TRUE)
##dev.off()

##(4) plot resulting function for given interval x
x &lt;- seq(1,10000, by = 100)
plot(
 x = x,
 y = eval(temp$Formula),
 type = "l"
)

##(5) plot using the 'extrapolation' mode
LxTxData[1,2:3] &lt;- c(0.5, 0.001)
print(plot_GrowthCurve(LxTxData, mode = "extrapolation"))

##(6) plot using the 'alternate' mode
LxTxData[1,2:3] &lt;- c(0.5, 0.001)
print(plot_GrowthCurve(LxTxData, mode = "alternate"))

</code></pre>

<hr>
<h2 id='plot_Histogram'>Plot a histogram with separate error plot</h2><span id='topic+plot_Histogram'></span>

<h3>Description</h3>

<p>Function plots a predefined histogram with an accompanying error plot as
suggested by Rex Galbraith at the UK LED in Oxford 2010.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_Histogram(
  data,
  na.rm = TRUE,
  mtext,
  cex.global,
  se,
  rug,
  normal_curve,
  summary,
  summary.pos,
  colour,
  interactive = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_Histogram_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> object (<strong>required</strong>):
for <code>data.frame</code>: two columns: De (<code>data[,1]</code>) and De error (<code>data[,2]</code>)</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
excludes <code>NA</code> values from the data set prior to any further operations.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_mtext">mtext</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
further sample information (<a href="graphics.html#topic+mtext">mtext</a>).</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_cex.global">cex.global</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
global scaling factor.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_se">se</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
plots standard error points over the histogram, default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_rug">rug</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>optional</em>):
adds rugs to the histogram, default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_normal_curve">normal_curve</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
adds a normal curve to the histogram. Mean and standard deviation are calculated from the
input data. More see details section.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
add statistic measures of centrality and dispersion to the plot.
Can be one or more of several keywords. See details for available keywords.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_summary.pos">summary.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>)
for the statistical summary. Alternatively, the keyword <code>"sub"</code> may be
specified to place the summary below the plot header. However, this latter
option in only possible if <code>mtext</code> is not used. In case of coordinate
specification, y-coordinate refers to the right y-axis.</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_colour">colour</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional vector of length 4 which specifies the colours of the following
plot items in exactly this order: histogram bars, rug lines, normal
distribution curve and standard error points
(e.g., <code>c("grey", "black", "red", "grey")</code>).</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_interactive">interactive</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
create an interactive histogram plot (requires the 'plotly' package)</p>
</td></tr>
<tr><td><code id="plot_Histogram_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to <a href="base.html#topic+plot">plot</a> or
<a href="graphics.html#topic+hist">hist</a>. If y-axis labels are provided, these must be specified as a vector
of length 2 since the plot features two axes
(e.g. <code>ylab = c("axis label 1", "axis label 2")</code>). Y-axes limits
(<code>ylim</code>) must be provided as vector of length four, with the first two
elements specifying the left axes limits and the latter two elements giving
the right axis limits.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the normal curve is added, the y-axis in the histogram will show the
probability density.
</p>
<p>A statistic summary, i.e. a collection of statistic measures of
centrality and dispersion (and further measures) can be added by specifying
one or more of the following keywords:
</p>

<ul>
<li> <p><code>"n"</code> (number of samples),
</p>
</li>
<li> <p><code>"mean"</code> (mean De value),
</p>
</li>
<li> <p><code>"mean.weighted"</code> (error-weighted mean),
</p>
</li>
<li> <p><code>"median"</code> (median of the De values),
</p>
</li>
<li> <p><code>"sdrel"</code> (relative standard deviation in percent),
</p>
</li>
<li> <p><code>"sdrel.weighted"</code> (error-weighted relative standard deviation in percent),
</p>
</li>
<li> <p><code>"sdabs"</code> (absolute standard deviation),
</p>
</li>
<li> <p><code>"sdabs.weighted"</code> (error-weighted absolute standard deviation),
</p>
</li>
<li> <p><code>"serel"</code> (relative standard error),
</p>
</li>
<li> <p><code>"serel.weighted"</code> (error-weighted relative standard error),
</p>
</li>
<li> <p><code>"seabs"</code> (absolute standard error),
</p>
</li>
<li> <p><code>"seabs.weighted"</code> (error-weighted absolute standard error),
</p>
</li>
<li> <p><code>"kurtosis"</code> (kurtosis) and
</p>
</li>
<li> <p><code>"skewness"</code> (skewness).
</p>
</li></ul>



<h3>Function version</h3>

<p>0.4.5
</p>


<h3>How to cite</h3>

<p>Dietze, M., Kreutzer, S., 2025. plot_Histogram(): Plot a histogram with separate error plot. Function version 0.4.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The input data is not restricted to a special type.
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)<br />
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="graphics.html#topic+hist">hist</a>, <a href="base.html#topic+plot">plot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load data
data(ExampleData.DeValues, envir = environment())
ExampleData.DeValues &lt;- convert_Second2Gray(ExampleData.DeValues$BT998,
                                            dose.rate = c(0.0438,0.0019))

## plot histogram the easiest way
plot_Histogram(ExampleData.DeValues)

## plot histogram with some more modifications
plot_Histogram(ExampleData.DeValues,
               rug = TRUE,
               normal_curve = TRUE,
               cex.global = 0.9,
               pch = 2,
               colour = c("grey", "black", "blue", "green"),
               summary = c("n", "mean", "sdrel"),
               summary.pos = "topleft",
               main = "Histogram of De-values",
               mtext = "Example data set",
               ylab = c(expression(paste(D[e], " distribution")),
                        "Standard error"),
               xlim = c(100, 250),
               ylim = c(0, 0.1, 5, 20))


</code></pre>

<hr>
<h2 id='plot_KDE'>Plot kernel density estimate with statistics</h2><span id='topic+plot_KDE'></span>

<h3>Description</h3>

<p>Plot a kernel density estimate of measurement values in combination with the
actual values and associated error bars in ascending order. If enabled, the
boxplot will show the usual distribution parameters (median as
bold line, box delimited by the first and third quartile, whiskers defined
by the extremes and outliers shown as points) and also the mean and
standard deviation as pale bold line and pale polygon, respectively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_KDE(
  data,
  na.rm = TRUE,
  values.cumulative = TRUE,
  order = TRUE,
  boxplot = TRUE,
  rug = TRUE,
  summary,
  summary.pos,
  summary.method = "MCM",
  bw = "nrd0",
  output = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_KDE_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a>, <a href="base.html#topic+vector">vector</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> object (<strong>required</strong>):
for <code>data.frame</code>: either two columns: De (<code>values[,1]</code>) and De error
(<code>values[,2]</code>), or one: De (<code>values[,1]</code>). If a numeric vector or a
single-column data frame is provided, De error is assumed to be 10^-9
for all measurements and error bars are not drawn.
For plotting multiple data sets, these must be provided as
<code>list</code> (e.g. <code>list(dataset1, dataset2)</code>).</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
exclude NA values from the data set prior to any further operation.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_values.cumulative">values.cumulative</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
show cumulative individual data.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_order">order</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Order data in ascending order.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_boxplot">boxplot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
optionally show a boxplot (depicting median as thick central line,
first and third quartile as box limits, whiskers denoting +/- 1.5
interquartile ranges and dots further outliers).</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_rug">rug</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
optionally add rug.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
add statistic measures of centrality and dispersion to the plot. Can be one
or more of several keywords. See details for available keywords.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_summary.pos">summary.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>optional</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>)
for the statistical summary. Alternatively, the keyword <code>"sub"</code> may be
specified to place the summary below the plot header. However, this latter
option in only possible if <code>mtext</code> is not used. In case of coordinate
specification, y-coordinate refers to the right y-axis.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_summary.method">summary.method</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
keyword indicating the method used to calculate the statistic summary.
One out of <code>"unweighted"</code>, <code>"weighted"</code> and <code>"MCM"</code>.
See <a href="#topic+calc_Statistics">calc_Statistics</a> for details.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_bw">bw</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
bin-width, chose a numeric value for manual setting.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_output">output</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Optional output of numerical plot parameters. These can be useful to
reproduce similar plots. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_KDE_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to <a href="base.html#topic+plot">plot</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function allows passing several plot arguments, such as <code>main</code>,
<code>xlab</code>, <code>cex</code>. However, as the figure is an overlay of two
separate plots, <code>ylim</code> must be specified in the order: c(ymin_axis1,
ymax_axis1, ymin_axis2, ymax_axis2) when using the cumulative values plot
option. See examples for some further explanations. For details on the
calculation of the bin-width (parameter <code>bw</code>) see
<a href="stats.html#topic+density">density</a>.
</p>
<p>A statistic summary, i.e. a collection of statistic measures of
centrality and dispersion (and further measures) can be added by specifying
one or more of the following keywords:
</p>

<ul>
<li> <p><code>"n"</code> (number of samples)
</p>
</li>
<li> <p><code>"mean"</code> (mean De value)
</p>
</li>
<li> <p><code>"median"</code> (median of the De values)
</p>
</li>
<li> <p><code>"sd.rel"</code> (relative standard deviation in percent)
</p>
</li>
<li> <p><code>"sd.abs"</code> (absolute standard deviation)
</p>
</li>
<li> <p><code>"se.rel"</code> (relative standard error)
</p>
</li>
<li> <p><code>"se.abs"</code> (absolute standard error)
</p>
</li>
<li> <p><code>"in.2s"</code> (percent of samples in 2-sigma range)
</p>
</li>
<li> <p><code>"kurtosis"</code> (kurtosis)
</p>
</li>
<li> <p><code>"skewness"</code> (skewness)
</p>
</li></ul>

<p><strong>Note</strong> that the input data for the statistic summary is sent to the function
<code>calc_Statistics()</code> depending on the log-option for the z-scale. If
<code>"log.z = TRUE"</code>, the summary is based on the logarithms of the input
data. If <code>"log.z = FALSE"</code> the linearly scaled data is used.
</p>
<p><strong>Note</strong> as well, that <code>"calc_Statistics()"</code> calculates these statistic
measures in three different ways: <code>unweighted</code>, <code>weighted</code> and
<code>MCM-based</code> (i.e., based on Monte Carlo Methods). By default, the
MCM-based version is used. If you wish to use another method, indicate this
with the appropriate keyword using the argument <code>summary.method</code>.
</p>


<h3>Function version</h3>

<p>3.6.0
</p>


<h3>How to cite</h3>

<p>Dietze, M., 2025. plot_KDE(): Plot kernel density estimate with statistics. Function version 3.6.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The plot output is no 'probability density' plot (cf. the discussion
of Berger and Galbraith in Ancient TL; see references)!
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)<br />
Geography &amp; Earth Sciences, Aberystwyth University (United Kingdom)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="stats.html#topic+density">density</a>, <a href="base.html#topic+plot">plot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## read example data set
data(ExampleData.DeValues, envir = environment())
ExampleData.DeValues &lt;-
  convert_Second2Gray(ExampleData.DeValues$BT998, c(0.0438,0.0019))

## create plot straightforward
plot_KDE(data = ExampleData.DeValues)

## create plot with logarithmic x-axis
plot_KDE(data = ExampleData.DeValues,
         log = "x")

## create plot with user-defined labels and axes limits
plot_KDE(data = ExampleData.DeValues,
         main = "Dose distribution",
         xlab = "Dose (s)",
         ylab = c("KDE estimate", "Cumulative dose value"),
         xlim = c(100, 250),
         ylim = c(0, 0.08, 0, 30))

## create plot with boxplot option
plot_KDE(data = ExampleData.DeValues,
         boxplot = TRUE)

## create plot with statistical summary below header
plot_KDE(data = ExampleData.DeValues,
         summary = c("n", "median", "skewness", "in.2s"))

## create plot with statistical summary as legend
plot_KDE(data = ExampleData.DeValues,
         summary = c("n", "mean", "sd.rel", "se.abs"),
         summary.pos = "topleft")

## split data set into sub-groups, one is manipulated, and merge again
data.1 &lt;- ExampleData.DeValues[1:15,]
data.2 &lt;- ExampleData.DeValues[16:25,] * 1.3
data.3 &lt;- list(data.1, data.2)

## create plot with two subsets straightforward
plot_KDE(data = data.3)

## create plot with two subsets and summary legend at user coordinates
plot_KDE(data = data.3,
         summary = c("n", "median", "skewness"),
         summary.pos = c(110, 0.07),
         col = c("blue", "orange"))

## example of how to use the numerical output of the function
## return plot output to draw a thicker KDE line
KDE_out &lt;- plot_KDE(data = ExampleData.DeValues,
output = TRUE)

</code></pre>

<hr>
<h2 id='plot_MoranScatterplot'>Moran Scatter Plot: Visualizing Spatial Dependency</h2><span id='topic+plot_MoranScatterplot'></span>

<h3>Description</h3>

<p>Scatter plot, with on the x axis the original grain signal and on the y axis
the weighted mean of the neighbour grain signals. The plot area is divided into four quadrants,
and also a least square line (which slopes indicates, but not exactly represents, Moran's I) and
an 1:1 line (which indicates a Moran's I of around 1).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_MoranScatterplot(
  object,
  df_neighbours = NULL,
  str_y_def = "mean_neighbours",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_MoranScatterplot_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): containing a numerical vector of length 100,
representing one or more measurement discs (&quot;positions&quot;) in a reader.
Each element in the vector represents one grain hole location on a disc.</p>
</td></tr>
<tr><td><code id="plot_MoranScatterplot_+3A_df_neighbours">df_neighbours</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>with default</em>) Data frame indicating
which borders to consider, and their respective weights (see the description
provided for <a href="#topic+calc_MoransI">calc_MoransI</a>). If <code>NULL</code> (default), this is constructed
automatically by the internal function <code>.get_Neighbours</code>.</p>
</td></tr>
<tr><td><code id="plot_MoranScatterplot_+3A_str_y_def">str_y_def</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>) Calculation of y position. Defaults to <code>"mean_neighbours"</code>
which is the plain mean of all neighbour values and the most illustrative. The other option is <code>"weighted_sum"</code>,
which means the sum of the border weights times the neighbour values, which is actually closer
to the way Moran's I is by default calculated in this package.</p>
</td></tr>
<tr><td><code id="plot_MoranScatterplot_+3A_...">...</code></td>
<td>
<p>Other parameters to be forwarded to the base R plot functions.
<code>legend</code> (<code>TRUE/FALSE</code>) to enable/disable the legend.
Note that <code>xlab</code> (x axis label), <code>ylab</code> (y axis label) and <code>cex</code> (scaling
value) are given default values. Because of sometimes large value
differences, <code>log = "x"</code>, <code>log = "y"</code> and <code>log = "xy"</code>are supported.
In case of negative values and logarithmic plotting, values are increased
so the smallest value to plot is 1. Summary elements such as means, least
square line etc. will still be based on the linear case.
<code>pch</code> accepts options &quot;show_location_ids&quot;<code style="white-space: pre;">&#8288;(plots grain location id's),&#8288;</code>&quot;show_n_neighbours&quot;<code style="white-space: pre;">&#8288;(indicates numbers of neighbours) and the normal base plot&#8288;</code>pch' options.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that this function plots on the y-axis the mean of the neighbours, while the function
<a href="#topic+calc_MoransI">calc_MoransI</a> by default will for its global calculation weight every border the same. So, grain locations
with 1, 2 or 3 neighbours will appear higher on the y-axis than their influence on Moran's I justify &ndash; apart
from scaling, this explains a part of the differences of Moran's scatter plots between different packages.
Also note that island' grain locations (=those not bordering other grains) are left out of these plots but
might still influence Moran's I calculations.
</p>


<h3>Value</h3>

<p>Returns (invisibly) a data frame with plotting coordinates and
grain location id's, for creating user-defined plots.
</p>


<h3>How to cite</h3>

<p>Boer, A.d., Steinbuch, L., 2025. plot_MoranScatterplot(): Moran Scatter Plot: Visualizing Spatial Dependency. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Anna-Maartje de Boer, Luc Steinbuch, Wageningen University &amp; Research, 2025
, RLum Developer Team</p>


<h3>References</h3>

<p>de Boer, A-M., Steinbuch, L., Heuvelink, G.B.M., Wallinga, J., 2025.
A novel tool to assess crosstalk in single-grain luminescence detection.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_MoranScatterplot(1:100)

</code></pre>

<hr>
<h2 id='plot_NRt'>Visualise natural/regenerated signal ratios</h2><span id='topic+plot_NRt'></span>

<h3>Description</h3>

<p>This function creates a Natural/Regenerated signal vs. time (NR(t)) plot
as shown in Steffen et al. 2009.
</p>
<p>This function accepts the individual curve data in many different formats. If
<code>data</code> is a <code>list</code>, each element of the list must contain a two
column <code>data.frame</code> or <code>matrix</code> containing the <code>XY</code> data of the curves
(time and counts). Alternatively, the elements can be objects of class
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>.
</p>
<p>Input values can also be provided as a <code>data.frame</code> or <code>matrix</code> where
the first column contains the time values and each following column contains
the counts of each curve.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_NRt(
  data,
  log = FALSE,
  smooth = c("none", "spline", "rmean"),
  k = 3,
  legend = TRUE,
  legend.pos = "topright",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_NRt_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+list">list</a>, <a href="base.html#topic+data.frame">data.frame</a>, <a href="base.html#topic+matrix">matrix</a> or <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
X,Y data of measured values (time and counts). See details on individual data structure.</p>
</td></tr>
<tr><td><code id="plot_NRt_+3A_log">log</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
logarithmic axes (<code>c("x", "y", "xy")</code>).</p>
</td></tr>
<tr><td><code id="plot_NRt_+3A_smooth">smooth</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
apply data smoothing. If <code>"none"</code> (default), no data smoothing is applied.
Use <code>"rmean"</code> to calculate the rolling mean, where <code>k</code> determines the
width of the rolling window (see <a href="data.table.html#topic+froll">data.table::frollmean</a>). <code>"spline"</code>
applies a smoothing spline to each curve (see <a href="stats.html#topic+smooth.spline">stats::smooth.spline</a>)</p>
</td></tr>
<tr><td><code id="plot_NRt_+3A_k">k</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
integer width of the rolling window.</p>
</td></tr>
<tr><td><code id="plot_NRt_+3A_legend">legend</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot legend.</p>
</td></tr>
<tr><td><code id="plot_NRt_+3A_legend.pos">legend.pos</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
keyword specifying the position of the legend (see <a href="graphics.html#topic+legend">legend</a>).</p>
</td></tr>
<tr><td><code id="plot_NRt_+3A_...">...</code></td>
<td>
<p>further parameters passed to <a href="base.html#topic+plot">plot</a> (also see <a href="graphics.html#topic+par">par</a>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot and <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.
</p>


<h3>How to cite</h3>

<p>Burow, C., 2025. plot_NRt(): Visualise natural/regenerated signal ratios. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Steffen, D., Preusser, F., Schlunegger, F., 2009. OSL quartz underestimation due to
unstable signal components. Quaternary Geochronology, 4, 353-362.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data("ExampleData.BINfileData", envir = environment())

## EXAMPLE 1

## convert Risoe.BINfileData object to RLum.Analysis object
data &lt;- Risoe.BINfileData2RLum.Analysis(object = CWOSL.SAR.Data, pos = 8, ltype = "OSL")

## extract all OSL curves
allCurves &lt;- get_RLum(data)

## keep only the natural and regenerated signal curves
pos &lt;- seq(1, 9, 2)
curves &lt;- allCurves[pos]

## plot a standard NR(t) plot
plot_NRt(curves)

## re-plot with rolling mean data smoothing
plot_NRt(curves, smooth = "rmean", k = 10)

## re-plot with a logarithmic x-axis
plot_NRt(curves, log = "x", smooth = "rmean", k = 5)

## re-plot with custom axes ranges
plot_NRt(curves, smooth = "rmean", k = 5,
         xlim = c(0.1, 5), ylim = c(0.4, 1.6),
         legend.pos = "bottomleft")

## re-plot with smoothing spline on log scale
plot_NRt(curves, smooth = "spline", log = "x",
         legend.pos = "top")

## EXAMPLE 2

# you may also use this function to check whether all
# TD curves follow the same shape (making it a TnTx(t) plot).
posTD &lt;- seq(2, 14, 2)
curves &lt;- allCurves[posTD]

plot_NRt(curves, main = "TnTx(t) Plot",
         smooth = "rmean", k = 20,
         ylab = "TD natural / TD regenerated",
         xlim = c(0, 20), legend = FALSE)

## EXAMPLE 3

# extract data from all positions
data &lt;- lapply(1:24, FUN = function(pos) {
   Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos = pos, ltype = "OSL")
})

# get individual curve data from each aliquot
aliquot &lt;- lapply(data, get_RLum)

# set graphical parameters
par(mfrow = c(2, 2))

# create NR(t) plots for all aliquots
for (i in 1:length(aliquot)) {
   plot_NRt(aliquot[[i]][pos],
            main = paste0("Aliquot #", i),
            smooth = "rmean", k = 20,
            xlim = c(0, 10),
            cex = 0.6, legend.pos = "bottomleft")
}

# reset graphical parameters
par(mfrow = c(1, 1))


</code></pre>

<hr>
<h2 id='plot_OSLAgeSummary'>Plot Posterior OSL-Age Summary</h2><span id='topic+plot_OSLAgeSummary'></span>

<h3>Description</h3>

<p>A graphical summary of the statistical inference of an OSL age
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_OSLAgeSummary(object, level = 0.95, digits = 1L, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_OSLAgeSummary_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): an object produced
by <a href="#topic+combine_De_Dr">combine_De_Dr</a>. Alternatively, a <a href="base.html#topic+numeric">numeric</a> vector of a parameter from an MCMC process</p>
</td></tr>
<tr><td><code id="plot_OSLAgeSummary_+3A_level">level</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): probability of shown credible interval</p>
</td></tr>
<tr><td><code id="plot_OSLAgeSummary_+3A_digits">digits</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): number of digits considered for the calculation</p>
</td></tr>
<tr><td><code id="plot_OSLAgeSummary_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="plot_OSLAgeSummary_+3A_...">...</code></td>
<td>
<p>further arguments to modify the plot, supported: <code>xlim</code>, <code>ylim</code>, <code>xlab</code>, <code>ylab</code>,
<code>main</code>, <code>lwd</code>, <code>lty</code>, <code>col</code>, <code>polygon_col</code>, <code>polygon_density</code>, <code>rug</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is called automatically by <a href="#topic+combine_De_Dr">combine_De_Dr</a>
</p>


<h3>Value</h3>

<p>A posterior distribution plot and an <a href="#topic+RLum.Results-class">RLum.Results</a>
object with the credible interval.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Philippe, A., Galharret, J., Mercier, N., Kreutzer, S., 2025. plot_OSLAgeSummary(): Plot Posterior OSL-Age Summary. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Anne Philippe, Universit√© de Nantes (France),
Jean-Michel Galharret, Universit√© de Nantes (France),
Norbert Mercier, IRAMAT-CRP2A, Universit√© Bordeaux Montaigne (France),
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+combine_De_Dr">combine_De_Dr</a>, <a href="graphics.html#topic+plot.default">plot.default</a>, <a href="rjags.html#topic+rjags-package">rjags::rjags</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##generate random data
set.seed(1234)
object &lt;- rnorm(1000, 100, 10)
plot_OSLAgeSummary(object)

</code></pre>

<hr>
<h2 id='plot_RadialPlot'>Function to create a Radial Plot</h2><span id='topic+plot_RadialPlot'></span>

<h3>Description</h3>

<p>A Galbraith's radial plot is produced on a logarithmic or a linear scale.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RadialPlot(
  data,
  na.rm = TRUE,
  log.z = TRUE,
  central.value,
  centrality = "mean.weighted",
  mtext,
  summary,
  summary.pos,
  legend,
  legend.pos,
  stats,
  rug = FALSE,
  plot.ratio,
  bar.col,
  y.ticks = TRUE,
  grid.col,
  line,
  line.col,
  line.label,
  output = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RadialPlot_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> object (<strong>required</strong>):
for <code>data.frame</code>: either two columns: De (<code>data[,1]</code>) and De error
(<code>data[,2]</code>), or one: De (<code>values[,1]</code>). If a single-column data frame
is provided, De error is assumed to be 10^-9 for all measurements.
To plot several data sets in one plot, the data sets must be provided as
<code>list</code>, e.g. <code>list(data.1, data.2)</code>.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
excludes <code>NA</code> values from the data set prior to any further operations.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_log.z">log.z</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Option to display the z-axis in logarithmic scale. Default is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_central.value">central.value</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
User-defined central value, primarily used for horizontal centring
of the z-axis.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_centrality">centrality</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
measure of centrality, used for automatically centring the plot and drawing
the central line. Can either be one out of
</p>

<ul>
<li> <p><code>"mean"</code>,
</p>
</li>
<li> <p><code>"median"</code>,
</p>
</li>
<li> <p><code>"mean.weighted"</code> and
</p>
</li>
<li> <p><code>"median.weighted"</code> or a
</p>
</li>
<li><p> numeric value used for the standardisation.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_mtext">mtext</code></td>
<td>
<p><a href="base.html#topic+character">character</a>:
additional text below the plot title.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
add statistic measures of centrality and dispersion to the plot.
Can be one or more of several keywords. See details for available keywords.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_summary.pos">summary.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position coordinates or keyword (e.g. <code>"topright"</code>)
for the statistical summary. Alternatively, the keyword <code>"sub"</code> may be
specified to place the summary below the plot header. However, this latter
option is only possible if <code>mtext</code> is not used.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_legend">legend</code></td>
<td>
<p><a href="base.html#topic+character">character</a> vector (<em>optional</em>):
legend content to be added to the plot.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_legend.pos">legend.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (with
default): optional position coordinates or keyword (e.g. <code>"topright"</code>)
for the legend to be plotted.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_stats">stats</code></td>
<td>
<p><a href="base.html#topic+character">character</a>: additional labels of statistically
important values in the plot. One or more out of the following:
</p>

<ul>
<li> <p><code>"min"</code>,
</p>
</li>
<li> <p><code>"max"</code>,
</p>
</li>
<li> <p><code>"median"</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_rug">rug</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a>:
Option to add a rug to the z-scale, to indicate the location of individual values</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_plot.ratio">plot.ratio</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
User-defined plot area ratio (i.e. curvature of the z-axis). If omitted,
the default value (<code>4.5/5.5</code>) is used and modified automatically to optimise
the z-axis curvature. The parameter should be decreased when data points
are plotted outside the z-axis or when the z-axis gets too elliptic.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_bar.col">bar.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
colour of the bar showing the 2-sigma range around the central
value. To disable the bar, use <code>"none"</code>. Default is <code>"grey"</code>.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_y.ticks">y.ticks</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a>:
Option to hide y-axis labels. Useful for data with small scatter.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_grid.col">grid.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
colour of the grid lines (originating at <code style="white-space: pre;">&#8288;[0,0]&#8288;</code> and stretching to
the z-scale). To disable grid lines, use <code>"none"</code>. Default is <code>"grey"</code>.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_line">line</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
numeric values of the additional lines to be added.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_line.col">line.col</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a>:
colour of the additional lines.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_line.label">line.label</code></td>
<td>
<p><a href="base.html#topic+character">character</a>:
labels for the additional lines.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_output">output</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a>:
Optional output of numerical plot parameters. These can be useful to
reproduce similar plots. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_RadialPlot_+3A_...">...</code></td>
<td>
<p>Further plot arguments to pass. <code>xlab</code> must be a vector of
length 2, specifying the upper and lower x-axes labels.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Details and the theoretical background of the radial plot are given in the
cited literature. This function is based on an S script of Rex Galbraith. To
reduce the manual adjustments, the function has been rewritten. Thanks to
Rex Galbraith for useful comments on this function. <br />
Plotting can be disabled by adding the argument <code>plot = "FALSE"</code>, e.g.
to return only numeric plot output.
</p>
<p>Earlier versions of the Radial Plot in this package had the 2-sigma-bar
drawn onto the z-axis. However, this might have caused misunderstanding in
that the 2-sigma range may also refer to the z-scale, which it does not!
Rather it applies only to the x-y-coordinate system (standardised error vs.
precision). A spread in doses or ages must be drawn as lines originating at
zero precision (x0) and zero standardised estimate (y0). Such a range may be
drawn by adding lines to the radial plot ( <code>line</code>, <code>line.col</code>,
<code>line.label</code>, cf. examples).
</p>
<p>A statistic summary, i.e. a collection of statistic measures of
centrality and dispersion (and further measures) can be added by specifying
one or more of the following keywords:
</p>

<ul>
<li> <p><code>"n"</code> (number of samples),
</p>
</li>
<li> <p><code>"mean"</code> (mean De value),
</p>
</li>
<li> <p><code>"mean.weighted"</code> (error-weighted mean),
</p>
</li>
<li> <p><code>"median"</code> (median of the De values),
</p>
</li>
<li> <p><code>"sdrel"</code> (relative standard deviation in percent),
</p>
</li>
<li> <p><code>"sdrel.weighted"</code> (error-weighted relative standard deviation in percent),
</p>
</li>
<li> <p><code>"sdabs"</code> (absolute standard deviation),
</p>
</li>
<li> <p><code>"sdabs.weighted"</code> (error-weighted absolute standard deviation),
</p>
</li>
<li> <p><code>"serel"</code> (relative standard error),
</p>
</li>
<li> <p><code>"serel.weighted"</code> (error-weighted relative standard error),
</p>
</li>
<li> <p><code>"seabs"</code> (absolute standard error),
</p>
</li>
<li> <p><code>"seabs.weighted"</code> (error-weighted absolute standard error),
</p>
</li>
<li> <p><code>"in.2s"</code> (percent of samples in 2-sigma range),
</p>
</li>
<li> <p><code>"kurtosis"</code> (kurtosis) and
</p>
</li>
<li> <p><code>"skewness"</code> (skewness).
</p>
</li></ul>



<h3>Value</h3>

<p>Returns a plot object.
</p>


<h3>Function version</h3>

<p>0.5.9
</p>


<h3>How to cite</h3>

<p>Dietze, M., Kreutzer, S., 2025. plot_RadialPlot(): Function to create a Radial Plot. Function version 0.5.9. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)<br />
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Based on a rewritten S script of Rex Galbraith, 2010
, RLum Developer Team</p>


<h3>References</h3>

<p>Galbraith, R.F., 1988. Graphical Display of Estimates Having
Differing Standard Errors. Technometrics, 30 (3), 271-281.
</p>
<p>Galbraith, R.F., 1990. The radial plot: Graphical assessment of spread in
ages. International Journal of Radiation Applications and Instrumentation.
Part D. Nuclear Tracks and Radiation Measurements, 17 (3), 207-214.
</p>
<p>Galbraith, R. &amp; Green, P., 1990. Estimating the component ages in a finite
mixture. International Journal of Radiation Applications and
Instrumentation. Part D. Nuclear Tracks and Radiation Measurements, 17 (3)
197-206.
</p>
<p>Galbraith, R.F. &amp; Laslett, G.M., 1993. Statistical models for mixed fission
track ages. Nuclear Tracks And Radiation Measurements, 21 (4), 459-470.
</p>
<p>Galbraith, R.F., 1994. Some Applications of Radial Plots. Journal of the
American Statistical Association, 89 (428), 1232-1242.
</p>
<p>Galbraith, R.F., 2010. On plotting OSL equivalent doses. Ancient TL, 28 (1),
1-10.
</p>
<p>Galbraith, R.F. &amp; Roberts, R.G., 2012. Statistical aspects of equivalent
dose and error calculation and display in OSL dating: An overview and some
recommendations. Quaternary Geochronology, 11, 1-27.
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+plot_KDE">plot_KDE</a>, <a href="#topic+plot_Histogram">plot_Histogram</a>, <a href="#topic+plot_AbanicoPlot">plot_AbanicoPlot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.DeValues, envir = environment())
ExampleData.DeValues &lt;- convert_Second2Gray(
  ExampleData.DeValues$BT998, c(0.0438,0.0019))

## plot the example data straightforward
plot_RadialPlot(data = ExampleData.DeValues)

## now with linear z-scale
plot_RadialPlot(
  data = ExampleData.DeValues,
  log.z = FALSE)

## now with output of the plot parameters
plot1 &lt;- plot_RadialPlot(
  data = ExampleData.DeValues,
  log.z = FALSE,
  output = TRUE)
plot1
plot1$zlim

## now with adjusted z-scale limits
plot_RadialPlot(
  data = ExampleData.DeValues,
  log.z = FALSE,
  xlim = c(0, 5),
  zlim = c(100, 200))

## now the two plots with serious but seasonally changing fun
#plot_RadialPlot(data = data.3, fun = TRUE)

## now with user-defined central value, in log-scale again
plot_RadialPlot(
  data = ExampleData.DeValues,
  central.value = 150)

## now with a rug, indicating individual De values at the z-scale
plot_RadialPlot(
  data = ExampleData.DeValues,
  rug = TRUE)

## now with legend, colour, different points and smaller scale
plot_RadialPlot(
  data = ExampleData.DeValues,
  legend.text = "Sample 1",
  col = "tomato4",
  bar.col = "peachpuff",
  pch = "R",
  cex = 0.8)

## now without 2-sigma bar, y-axis, grid lines and central value line
plot_RadialPlot(
  data = ExampleData.DeValues,
  bar.col = "none",
  grid.col = "none",
  y.ticks = FALSE,
  lwd = 0)

## now with user-defined axes labels
plot_RadialPlot(
  data = ExampleData.DeValues,
  xlab = c("Data error (%)", "Data precision"),
  ylab = "Scatter",
  zlab = "Equivalent dose [Gy]")

## now with minimum, maximum and median value indicated
plot_RadialPlot(
  data = ExampleData.DeValues,
  central.value = 150,
  stats = c("min", "max", "median"))

## now with a brief statistical summary
plot_RadialPlot(
  data = ExampleData.DeValues,
  summary = c("n", "in.2s"))

## now with another statistical summary as subheader
plot_RadialPlot(
  data = ExampleData.DeValues,
  summary = c("mean.weighted", "median"),
  summary.pos = "sub")

## now the data set is split into sub-groups, one is manipulated
data.1 &lt;- ExampleData.DeValues[1:15,]
data.2 &lt;- ExampleData.DeValues[16:25,] * 1.3

## now a common dataset is created from the two subgroups
data.3 &lt;- list(data.1, data.2)

## now the two data sets are plotted in one plot
plot_RadialPlot(data = data.3)

## now with some graphical modification
plot_RadialPlot(
  data = data.3,
  col = c("darkblue", "darkgreen"),
  bar.col = c("lightblue", "lightgreen"),
  pch = c(2, 6),
  summary = c("n", "in.2s"),
  summary.pos = "sub",
  legend = c("Sample 1", "Sample 2"))

</code></pre>

<hr>
<h2 id='plot_Risoe.BINfileData'>Plot Single Luminescence Curves from a Risoe.BINfileData-class object</h2><span id='topic+plot_Risoe.BINfileData'></span>

<h3>Description</h3>

<p>Plots single luminescence curves from an object returned by the
<a href="#topic+read_BIN2R">read_BIN2R</a> function <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_Risoe.BINfileData(
  data,
  position,
  run,
  set,
  sorter = "POSITION",
  ltype = c("IRSL", "OSL", "TL", "RIR", "RBR", "RL"),
  curve.transformation = "None",
  dose_rate,
  temp.lab,
  cex.global = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_Risoe.BINfileData_+3A_data">data</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
requires an S4 object returned by the <a href="#topic+read_BIN2R">read_BIN2R</a> function.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_position">position</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
option to limit the plotted curves by position
(e.g. <code>position = 1</code>, <code>position = c(1,3,5)</code>).</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_run">run</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
option to limit the plotted curves by run
(e.g., <code>run = 1</code>, <code>run = c(1,3,5)</code>).</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_set">set</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
option to limit the plotted curves by set
(e.g., <code>set = 1</code>, <code>set = c(1,3,5)</code>).</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_sorter">sorter</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
the plot output can be ordered by &quot;POSITION&quot;,&quot;SET&quot; or &quot;RUN&quot;.
POSITION, SET and RUN are options defined in the Risoe Sequence Editor.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_ltype">ltype</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
option to limit the plotted curves by the type of luminescence stimulation.
Allowed values: <code>"IRSL"</code>, <code>"OSL"</code>,<code>"TL"</code>, <code>"RIR"</code>, <code>"RBR"</code>
(corresponds to LM-OSL), <code>"RL"</code>.  All type of curves are plotted by
default.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_curve.transformation">curve.transformation</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
allows transforming CW-OSL and CW-IRSL curves to pseudo-LM curves via
transformation functions. Allowed values are: <code>CW2pLM</code>, <code>CW2pLMi</code>,
<code>CW2pHMi</code> and <code>CW2pPMi</code>, see details. If set to <code>None</code> (default), no
transformation is applied.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_dose_rate">dose_rate</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
dose rate of the irradiation source at the measurement date.
If set, the given irradiation dose will be shown in Gy.  See details.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_temp.lab">temp.lab</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
option to allow for different temperature units. If no value is set deg. C is chosen.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_cex.global">cex.global</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
global scaling factor.</p>
</td></tr>
<tr><td><code id="plot_Risoe.BINfileData_+3A_...">...</code></td>
<td>
<p>further undocumented plot arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>Nomenclature</strong>
</p>
<p>See <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>
<p><strong>curve.transformation</strong>
</p>
<p>This argument allows transforming continuous wave (CW) curves to pseudo
(linear) modulated curves. For the transformation, the functions of the
package are used.  Currently, it is not possible to pass further arguments
to the transformation functions. The argument works only for <code>ltype</code>
<code>OSL</code> and <code>IRSL</code>.
</p>
<p><strong>Irradiation time</strong>
</p>
<p>Plotting the irradiation time (s) or the given dose (Gy) requires that the
variable <code>IRR_TIME</code> has been set within the BIN-file.  This is normally
done by using the 'Run Info' option within the Sequence Editor or by editing
in R.
</p>


<h3>Value</h3>

<p>Returns a plot.
</p>


<h3>Function version</h3>

<p>0.4.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., 2025. plot_Risoe.BINfileData(): Plot Single Luminescence Curves from a Risoe.BINfileData-class object. Function version 0.4.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The function has been successfully tested for the Sequence Editor file
output version 3 and 4.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Duller, G., 2007. Analyst. pp. 1-45.
</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+convert_CW2pLM">convert_CW2pLM</a>,
<a href="#topic+convert_CW2pLMi">convert_CW2pLMi</a>, <a href="#topic+convert_CW2pPMi">convert_CW2pPMi</a>, <a href="#topic+convert_CW2pHMi">convert_CW2pHMi</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.BINfileData, envir = environment())

##plot all curves from the first position to the desktop
#pdf(file = "~/Desktop/CurveOutput.pdf", paper = "a4", height = 11, onefile = TRUE)

##example - load from *.bin file
#BINfile&lt;- file.choose()
#BINfileData&lt;-read_BIN2R(BINfile)

#par(mfrow = c(4,3), oma = c(0.5,1,0.5,1))
#plot_Risoe.BINfileData(CWOSL.SAR.Data,position = 1)
#mtext(side = 4, BINfile, outer = TRUE, col = "blue", cex = .7)
#dev.off()

</code></pre>

<hr>
<h2 id='plot_RLum'>General plot function for RLum S4 class objects</h2><span id='topic+plot_RLum'></span>

<h3>Description</h3>

<p>Function calls object specific plot functions for RLum S4 class objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RLum(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code>. Optional a <a href="base.html#topic+list">list</a> containing objects of
class <a href="#topic+RLum-class">RLum</a> can be provided. In this case the function tries to plot
every object in this list according to its <code>RLum</code> class. Non-RLum objects are
removed.</p>
</td></tr>
<tr><td><code id="plot_RLum_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters that will be passed
to the specific plot functions. The only argument that is supported directly is <code>main</code>
(setting the plot title). In contrast to the normal behaviour <code>main</code> can be here provided as
<a href="base.html#topic+list">list</a> and the arguments in the list will dispatched to the plots if the <code>object</code>
is of type <code>list</code> as well.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function provides a generalised access point for plotting specific
<a href="#topic+RLum-class">RLum</a> objects.<br />
Depending on the input object, the
corresponding plot function will be selected.  Allowed arguments can be
found in the documentations of each plot function.
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>object</strong> </td><td style="text-align: left;"> </td><td style="text-align: left;"> <strong>corresponding plot function</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <a href="#topic+plot_RLum.Data.Curve">plot_RLum.Data.Curve</a> </td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <a href="#topic+plot_RLum.Data.Spectrum">plot_RLum.Data.Spectrum</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <a href="#topic+plot_RLum.Data.Image">plot_RLum.Data.Image</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <a href="#topic+plot_RLum.Analysis">plot_RLum.Analysis</a></td>
</tr>
<tr>
 <td style="text-align: left;">
<a href="#topic+RLum.Results-class">RLum.Results</a> </td><td style="text-align: left;"> : </td><td style="text-align: left;"> <a href="#topic+plot_RLum.Results">plot_RLum.Results</a>
</td>
</tr>

</table>



<h3>Value</h3>

<p>Returns a plot.
</p>


<h3>Function version</h3>

<p>0.4.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_RLum(): General plot function for RLum S4 class objects. Function version 0.4.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The provided plot output depends on the input object.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+plot_RLum.Data.Curve">plot_RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+plot_RLum.Data.Spectrum">plot_RLum.Data.Spectrum</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+plot_RLum.Data.Image">plot_RLum.Data.Image</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+plot_RLum.Analysis">plot_RLum.Analysis</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+plot_RLum.Results">plot_RLum.Results</a>,
<a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load Example data
data(ExampleData.CW_OSL_Curve, envir = environment())

#transform data.frame to RLum.Data.Curve object
temp &lt;- as(ExampleData.CW_OSL_Curve, "RLum.Data.Curve")

#plot RLum object
plot_RLum(temp)

</code></pre>

<hr>
<h2 id='plot_RLum.Analysis'>Plot function for an RLum.Analysis S4 class object</h2><span id='topic+plot_RLum.Analysis'></span>

<h3>Description</h3>

<p>The function provides a standardised plot output for curve data of an
RLum.Analysis S4 class object
</p>
<p>The function produces a multiple plot output. A file output is recommended
(e.g., <a href="grDevices.html#topic+pdf">pdf</a>).
</p>
<p><strong>curve.transformation</strong>
</p>
<p>This argument allows transforming continuous wave (CW) curves to pseudo
(linear) modulated curves. For the transformation, the functions of the
package are used. Currently, it is not possible to pass further arguments to
the transformation functions. The argument works only for <code>ltype</code>
<code>OSL</code> and <code>IRSL</code>.
</p>
<p>Please note: The curve transformation within this functions works roughly,
i.e. every IRSL or OSL curve is transformed, without considering whether it
is measured with the PMT or not! However, for a fast look it might be
helpful.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RLum.Analysis(
  object,
  subset = NULL,
  nrows,
  ncols,
  abline = NULL,
  combine = FALSE,
  records_max = NULL,
  curve.transformation = "None",
  plot_singlePanels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RLum.Analysis_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
S4 object of class <code>RLum.Analysis</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_subset">subset</code></td>
<td>
<p>named <a href="base.html#topic+list">list</a> (<em>optional</em>):
subsets elements for plotting. The arguments in the named <a href="base.html#topic+list">list</a> will be
directly passed to the function <a href="#topic+get_RLum">get_RLum</a>
(e.g., <code>subset = list(curveType = "measured")</code>)</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_nrows">nrows</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
sets number of rows for plot output, if nothing is set the function
tries to find a value.</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_ncols">ncols</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
sets number of columns for plot output, if nothing is set the function
tries to find a value.</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_abline">abline</code></td>
<td>
<p><a href="base.html#topic+list">list</a> (<em>optional</em>):
allows to add ab-lines to the plot. Argument are provided
in a list and will be forward to the function <a href="graphics.html#topic+abline">abline</a>,
e.g., <code>list(v = c(10, 100))</code> adds two vertical lines add 10 and 100 to all
plots. In contrast <code style="white-space: pre;">&#8288;list(v = c(10), v = c(100)&#8288;</code> adds a vertical at 10 to
the first and a vertical line at 100 to the 2nd plot.</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_combine">combine</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
allows to combine all <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects in one single plot.</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_records_max">records_max</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): limits number of records
shown if <code>combine = TRUE</code>. Shown are always the first and the last curve,
the other number of curves to be shown a distributed evenly, this may result
in fewer curves plotted as specified. This parameter has only
an effect for  n &gt; 2.</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_curve.transformation">curve.transformation</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
allows transforming CW-OSL and CW-IRSL curves to pseudo-LM curves via
transformation functions. Allowed values are: <code>CW2pLM</code>, <code>CW2pLMi</code>,
<code>CW2pHMi</code> and <code>CW2pPMi</code>, see details. If set to <code>None</code> (default), no
transformation is applied.</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
global par settings are considered, normally this should end in one plot per page</p>
</td></tr>
<tr><td><code id="plot_RLum.Analysis_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters will be passed to
the <code>plot</code> function.
</p>
<p>Supported arguments: <code>main</code>, <code>mtext</code>, <code>log</code>, <code>lwd</code>, <code>lty</code> <code>type</code>, <code>pch</code>, <code>col</code>,
<code>norm</code> (see <a href="#topic+plot_RLum.Data.Curve">plot_RLum.Data.Curve</a>), <code>xlim</code>,<code>ylim</code>, <code>xlab</code>, <code>ylab</code>, ...
</p>
<p>and for <code>combine = TRUE</code> also: <code>sub_title</code>, <code>legend</code>, <code>legend.text</code>, <code>legend.pos</code>
(typical plus 'outside'), <code>legend.col</code>, <code>smooth</code>.
</p>
<p>All arguments can be provided as <code>vector</code> or <code>list</code> to gain in full control
of all plot settings.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns multiple plots.
</p>


<h3>Function version</h3>

<p>0.3.16
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_RLum.Analysis(): Plot function for an RLum.Analysis S4 class object. Function version 0.3.16. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Not all arguments available for <a href="base.html#topic+plot">plot</a> will be passed and they partly do not behave in the
way you might expect them to work. This function was designed to serve as an overview
plot, if you want to have more control, extract the objects and plot them individually.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+plot_RLum">plot_RLum</a>, <a href="#topic+plot_RLum.Data.Curve">plot_RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.BINfileData, envir = environment())

##convert values for position 1
temp &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos=1)

##(1) plot (combine) TL curves in one plot
plot_RLum.Analysis(
temp,
subset = list(recordType = "TL"),
combine = TRUE,
norm = TRUE,
abline = list(v = c(110))
)

##(2) same as example (1) but using
## the argument smooth = TRUE
plot_RLum.Analysis(
temp,
subset = list(recordType = "TL"),
combine = TRUE,
norm = TRUE,
smooth = TRUE,
abline = list(v = c(110))
)

</code></pre>

<hr>
<h2 id='plot_RLum.Data.Curve'>Plot function for an RLum.Data.Curve S4 class object</h2><span id='topic+plot_RLum.Data.Curve'></span>

<h3>Description</h3>

<p>The function provides a standardised plot output for curve data of an
<code>RLum.Data.Curve</code> S4-class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RLum.Data.Curve(
  object,
  par.local = TRUE,
  norm = FALSE,
  smooth = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RLum.Data.Curve_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> (<strong>required</strong>):
S4 object of class <code>RLum.Data.Curve</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Curve_+3A_par.local">par.local</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
use local graphical parameters for plotting, e.g. the plot is shown in one
column and one row. If <code>par.local = FALSE</code>, global parameters are inherited.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Curve_+3A_norm">norm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> <a href="base.html#topic+character">character</a> (<em>with default</em>): whether curve
normalisation should occur (<code>FALSE</code> by default). Alternatively, the function
offers modes <code>"max"</code> (used with <code>TRUE</code>), <code>"last"</code> and <code>"huot"</code>, see details.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Curve_+3A_smooth">smooth</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
provides automatic curve smoothing based on the internal function <code>.smoothing</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Curve_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters that will be passed
to the <code>plot</code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Only single curve data can be plotted with this function. Arguments
according to <a href="base.html#topic+plot">plot</a>.
</p>
<p><strong>Curve normalisation</strong>
</p>
<p>The argument <code>norm</code> normalises all count values. To date the following
options are supported:
</p>
<p><code>norm = TRUE</code> or <code>norm = "max"</code>: Curve values are normalised to the highest
count value in the curve
</p>
<p><code>norm = "last"</code>: Curve values are normalised to the last count value
(this can be useful in particular for radiofluorescence curves)
</p>
<p><code>norm = "huot"</code>: Curve values are normalised as suggested by S√©bastien Huot
via GitHub:
</p>
<p style="text-align: center;"><code class="reqn">
y = (observed - median(background)) / (max(observed) - median(background))
</code>
</p>

<p>The background of the curve is defined as the last 20% of the count values
of a curve.
</p>


<h3>Value</h3>

<p>Returns a plot.
</p>


<h3>Function version</h3>

<p>0.2.6
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_RLum.Data.Curve(): Plot function for an RLum.Data.Curve S4 class object. Function version 0.2.6. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Not all arguments of <a href="base.html#topic+plot">plot</a> will be passed!
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+plot_RLum">plot_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##plot curve data

#load Example data
data(ExampleData.CW_OSL_Curve, envir = environment())

#transform data.frame to RLum.Data.Curve object
temp &lt;- as(ExampleData.CW_OSL_Curve, "RLum.Data.Curve")

#plot RLum.Data.Curve object
plot_RLum.Data.Curve(temp)


</code></pre>

<hr>
<h2 id='plot_RLum.Data.Image'>Plot function for an <code>RLum.Data.Image</code> S4 class object</h2><span id='topic+plot_RLum.Data.Image'></span>

<h3>Description</h3>

<p>The function provides very basic plot functionality for image data of an
<a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> object. For more sophisticated plotting it is recommended
to use other very powerful packages for image processing.
</p>
<p><strong>Details on the plot functions</strong>
</p>
<p>Supported plot types:
</p>
<p><strong><code>plot.type = "plot.raster"</code></strong>
</p>
<p>Uses the standard plot function of R <a href="graphics.html#topic+image">graphics::image</a>. If wanted, the image
is enhanced, using the argument <code>stretch</code>. Possible values are <code>hist</code>, <code>lin</code>, and
<code>NULL</code>. The latter does nothing. The argument <code>useRaster = TRUE</code> is used by default, but
can be set to <code>FALSE</code>.
</p>
<p><strong><code>plot.type = "contour"</code></strong>
</p>
<p>This uses the function <a href="graphics.html#topic+contour">graphics::contour</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RLum.Data.Image(
  object,
  frames = NULL,
  par.local = TRUE,
  plot.type = "plot.raster",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RLum.Data.Image_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> (<strong>required</strong>): S4
object of class <code>RLum.Data.Image</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Image_+3A_frames">frames</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): sets the frames to be set, by default all
frames are plotted. Can be sequence of numbers, as long as the frame number is valid.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Image_+3A_par.local">par.local</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): use local graphical
parameters for plotting, e.g. the plot is shown in one column and one row.
If <code>par.local = FALSE</code> global parameters are inherited.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Image_+3A_plot.type">plot.type</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): plot types.
Supported types are <code>plot.raster</code>, <code>contour</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Image_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters that will be passed
to the specific plot functions. Standard supported parameters are <code>xlim</code>, <code>ylim</code>, <code>zlim</code>,
<code>xlab</code>, <code>ylab</code>, <code>main</code>, <code>legend</code> (<code>TRUE</code> or <code>FALSE</code>), <code>col</code>, <code>cex</code>, <code>axes</code> (<code>TRUE</code> or <code>FALSE</code>),
<code>zlim_image</code> (adjust the z-scale over different images), <code>stretch</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot
</p>


<h3>Function version</h3>

<p>0.2.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_RLum.Data.Image(): Plot function for an RLum.Data.Image S4 class object. Function version 0.2.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The axes limitations (<code>xlim</code>, <code>zlim</code>, <code>zlim</code>) work directly on the object,
so that regardless of the chosen limits the image parameters can be adjusted for
best visibility. However, in particular for z-scale limitations this is not always
wanted, please use <code>zlim_image</code> to maintain a particular value range over a
series of images.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>, <a href="base.html#topic+plot">plot</a>, <a href="#topic+plot_RLum">plot_RLum</a>, <a href="graphics.html#topic+image">graphics::image</a>, <a href="graphics.html#topic+contour">graphics::contour</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.RLum.Data.Image, envir = environment())

##plot data
plot_RLum.Data.Image(ExampleData.RLum.Data.Image)

</code></pre>

<hr>
<h2 id='plot_RLum.Data.Spectrum'>Plot function for an RLum.Data.Spectrum S4 class object</h2><span id='topic+plot_RLum.Data.Spectrum'></span>

<h3>Description</h3>

<p>The function provides a standardised plot output for spectrum data of an
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> class object. The purpose of this function is to provide
easy and straight-forward spectra plotting, not provide a full customised access to
all plot parameters. If this is wanted, standard R plot functionality should be used
instead.
</p>
<p><strong>Matrix structure</strong> <br /> (cf. <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>)
</p>

<ul>
<li> <p><code>rows</code> (x-values): wavelengths/channels (<code>xlim</code>, <code>xlab</code>)
</p>
</li>
<li> <p><code>columns</code> (y-values): time/temperature (<code>ylim</code>, <code>ylab</code>)
</p>
</li>
<li> <p><code>cells</code> (z-values): count values (<code>zlim</code>, <code>zlab</code>)
</p>
</li></ul>

<p><em>Note: This nomenclature is valid for all plot types of this function!</em>
</p>
<p><strong>Nomenclature for value limiting</strong>
</p>

<ul>
<li> <p><code>xlim</code>: Limits values along the wavelength axis
</p>
</li>
<li> <p><code>ylim</code>: Limits values along the time/temperature axis
</p>
</li>
<li> <p><code>zlim</code>: Limits values along the count value axis
</p>
</li></ul>

<p><strong>Details on the plot functions</strong>
</p>
<p>Spectrum is visualised as 3D or 2D plot. Both plot types are based on
internal R plot functions.
</p>
<p><strong><code>plot.type = "persp"</code></strong>
</p>
<p>Arguments that will be passed to <a href="graphics.html#topic+persp">graphics::persp</a>:
</p>

<ul>
<li> <p><code>shade</code>: default is <code>0.4</code>
</p>
</li>
<li> <p><code>phi</code>: default is <code>15</code>
</p>
</li>
<li> <p><code>theta</code>: default is <code>-30</code>
</p>
</li>
<li> <p><code>lphi</code>: default is <code>15</code>
</p>
</li>
<li> <p><code>ltheta</code>: default is <code>-30</code>
</p>
</li>
<li> <p><code>expand</code>: default is <code>1</code>
</p>
</li>
<li> <p><code>axes</code>: default is <code>TRUE</code>
</p>
</li>
<li> <p><code>box</code>: default is <code>TRUE</code>; accepts <code>"alternate"</code> for a custom plot design
</p>
</li>
<li> <p><code>ticktype</code>: default is <code>detailed</code>, <code>r</code>: default is <code>10</code>
</p>
</li></ul>

<p><strong>Note:</strong> Further parameters can be adjusted via <code>par</code>. For example
to set the background transparent and reduce the thickness of the lines use:
<code>par(bg = NA, lwd = 0.7)</code> before the function call.
</p>
<p><strong><code>plot.type = "single"</code></strong>
</p>
<p>Per frame a single curve is returned. Frames are time or temperature
steps.
</p>
<p>-<code>frames</code>: pick the frames to be plotted (depends on the binning!). Check without
this setting before plotting.
</p>
<p><strong><code>plot.type = "multiple.lines"</code></strong>
</p>
<p>All frames plotted in one frame.
</p>
<p>-<code>frames</code>: pick the frames to be plotted (depends on the binning!). Check without
this setting before plotting.
</p>
<p><strong><code>plot.type = "image"</code> or <code>plot.type = "contour"</code></strong>
</p>
<p>These plot types use the R functions <a href="graphics.html#topic+image">graphics::image</a> or <a href="graphics.html#topic+contour">graphics::contour</a>.
The advantage is that many plots can be arranged conveniently using standard
R plot functionality. If <code>plot.type = "image"</code> a contour is added by default,
which can be disabled using the argument <code>contour = FALSE</code> to add own contour
lines of choice.
</p>
<p><strong><code>plot.type = "transect"</code></strong>
</p>
<p>Depending on the selected wavelength/channel range a transect over the
time/temperature (y-axis) will be plotted along the wavelength/channels
(x-axis). If the range contains more than one channel, values (z-values) are
summed up. To select a transect use the <code>xlim</code> argument, e.g.
<code>xlim = c(300,310)</code> plot along the summed up count values of channel
300 to 310.
</p>
<p><strong>Further arguments that will be passed (depending on the plot type)</strong>
</p>
<p><code>xlab</code>, <code>ylab</code>, <code>zlab</code>, <code>xlim</code>, <code>ylim</code>, <code>box</code>,
<code>zlim</code>, <code>main</code>, <code>mtext</code>, <code>pch</code>, <code>type</code> (<code>"single"</code>, <code>"multiple.lines"</code>, <code>"interactive"</code>),
<code>col</code>, <code>border</code>, <code>lwd</code>, <code>bty</code>, <code>showscale</code> (<code>"interactive"</code>, <code>"image"</code>)
<code>contour</code>, <code>contour.col</code> (<code>"image"</code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RLum.Data.Spectrum(
  object,
  par.local = TRUE,
  plot.type = "contour",
  optical.wavelength.colours = TRUE,
  bg.spectrum = NULL,
  bg.channels = NULL,
  bin.rows = 1,
  bin.cols = 1,
  norm = NULL,
  rug = TRUE,
  limit_counts = NULL,
  xaxis.energy = FALSE,
  legend.text,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> or <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>):
S4 object of class <code>RLum.Data.Spectrum</code> or a <code>matrix</code> containing count
values of the spectrum.<br />
Please note that in case of a matrix row names and col names are set
automatically if not provided.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_par.local">par.local</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
use local graphical parameters for plotting, e.g. the plot is shown in one column and one row.
If <code>par.local = FALSE</code> global parameters are inherited.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_plot.type">plot.type</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): for a 3D-plot use <code>persp</code>
or <code>interactive</code>; for a 2D-plot you can use <code>image</code>, <code>contour</code>,
<code>single</code> or <code>multiple.lines</code> (along the time or temperature axis)
or <code>transect</code> (along the wavelength axis)</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_optical.wavelength.colours">optical.wavelength.colours</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
use optical wavelength colour palette. Note: For this, the spectrum range is
limited: <code>c(350,750)</code>. Own colours can be set with the argument <code>col</code>. If you provide already
binned spectra, the colour assignment is likely to be wrong, since the colour gradients are calculated
using the bin number.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_bg.spectrum">bg.spectrum</code></td>
<td>
<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> or <a href="base.html#topic+matrix">matrix</a> (<em>optional</em>): Spectrum
used for the background subtraction. By definition, the background spectrum should have been
measured with the same setting as the signal spectrum. If a spectrum is provided, the
argument <code>bg.channels</code> works only on the provided background spectrum.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_bg.channels">bg.channels</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
defines the channels used for background subtraction. If a vector is
provided, the mean of the channels is used for subtraction. If a spectrum
is provided via <code>bg.spectrum</code>, this argument only works on the background
spectrum.
</p>
<p><strong>Note:</strong> Background subtraction is applied prior to channel binning</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_bin.rows">bin.rows</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
allow summing-up wavelength channels (horizontal binning),
e.g. <code>bin.rows = 2</code> two channels are summed up.
Binning is applied after the background subtraction.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_bin.cols">bin.cols</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
allow summing-up channel counts (vertical binning) for plotting,
e.g. <code>bin.cols = 2</code> two channels are summed up.
Binning is applied after the background subtraction.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_norm">norm</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): Normalise data to the maximum (<code>norm = "max"</code>) or
minimum (<code>norm = "min"</code>) count values. The normalisation is applied after the binning.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_rug">rug</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable colour rug. Currently only implemented for plot
type <code>multiple.lines</code> and <code>single</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_limit_counts">limit_counts</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
value to limit all count values to this value, i.e. all count values above
this threshold will be replaced by this threshold. This is helpful
especially in case of TL-spectra.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_xaxis.energy">xaxis.energy</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable using energy
instead of wavelength on the x-axis. Function <a href="#topic+convert_Wavelength2Energy">convert_Wavelength2Energy</a>
is used to perform the conversion.
</p>
<p><strong>Note:</strong> Besides being used in setting the axis, with this option the
the spectrum is recalculated in terms of intensity, see details.</p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_legend.text">legend.text</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
possibility to provide own legend text. This argument is only considered for
plot types providing a legend, e.g. <code>plot.type="transect"</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output. If
the plot output is disabled, the <a href="base.html#topic+matrix">matrix</a> used for the plotting and the
calculated colour values (as attributes) are returned. This way, the
(binned, transformed etc.) output can be used in other functions and
packages, such as plotting with the package <code>'plot3D'</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Data.Spectrum_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters that will be passed
to the <code>plot</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a plot and the transformed <code>matrix</code> used for plotting with some useful
attributes such as the <code>colour</code> and <code>pmat</code> (the transpose matrix from <a href="graphics.html#topic+persp">graphics::persp</a>)
</p>


<h3>Function version</h3>

<p>0.6.9
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_RLum.Data.Spectrum(): Plot function for an RLum.Data.Spectrum S4 class object. Function version 0.6.9. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Not all additional arguments (<code>...</code>) will be passed similarly!
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+convert_Wavelength2Energy">convert_Wavelength2Energy</a>, <a href="base.html#topic+plot">plot</a>, <a href="#topic+plot_RLum">plot_RLum</a>, <a href="graphics.html#topic+persp">graphics::persp</a>, <a href="plotly.html#topic+plot_ly">plotly::plot_ly</a>, <a href="graphics.html#topic+contour">graphics::contour</a>, <a href="graphics.html#topic+image">graphics::image</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load example data
data(ExampleData.XSYG, envir = environment())

##(1)plot simple spectrum (2D) - image
plot_RLum.Data.Spectrum(
 TL.Spectrum,
 plot.type="image",
 xlim = c(310,750),
 ylim = c(0,300),
 bin.rows=10,
 bin.cols = 1)

##(2) plot spectrum (3D)
plot_RLum.Data.Spectrum(
  TL.Spectrum,
  plot.type="persp",
  xlim = c(310,750),
  ylim = c(0,100),
  bin.rows=10,
  bin.cols = 1)

##(3) plot spectrum on energy axis
##please note the background subtraction
plot_RLum.Data.Spectrum(TL.Spectrum,
plot.type="persp",
ylim = c(0,200),
bin.rows=10,
bg.channels = 10,
bin.cols = 1,
xaxis.energy = TRUE)

##(4) plot multiple lines (2D) - multiple.lines (with ylim)
plot_RLum.Data.Spectrum(
 TL.Spectrum,
 plot.type="multiple.lines",
 xlim = c(310,750),
 ylim = c(0,100),
 bin.rows=10,
 bin.cols = 1)

## Not run: 
 ##(4) interactive plot using the package plotly ("surface")
 plot_RLum.Data.Spectrum(TL.Spectrum, plot.type="interactive",
 xlim = c(310,750), ylim = c(0,300), bin.rows=10,
 bin.cols = 1)

 ##(5) interactive plot using the package plotly ("contour")
 plot_RLum.Data.Spectrum(TL.Spectrum, plot.type="interactive",
 xlim = c(310,750), ylim = c(0,300), bin.rows=10,
 bin.cols = 1,
 type = "contour",
 showscale = TRUE)

 ##(6) interactive plot using the package plotly ("heatmap")
 plot_RLum.Data.Spectrum(TL.Spectrum, plot.type="interactive",
 xlim = c(310,750), ylim = c(0,300), bin.rows=10,
 bin.cols = 1,
 type = "heatmap",
 showscale = TRUE)


## End(Not run)

</code></pre>

<hr>
<h2 id='plot_RLum.Results'>Plot function for an RLum.Results S4 class object</h2><span id='topic+plot_RLum.Results'></span>

<h3>Description</h3>

<p>The function provides a standardised plot output for data of an RLum.Results
S4 class object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_RLum.Results(object, single = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_RLum.Results_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> (<strong>required</strong>):
S4 object of class <code>RLum.Results</code></p>
</td></tr>
<tr><td><code id="plot_RLum.Results_+3A_single">single</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
single plot output (<code>TRUE/FALSE</code>) to allow for plotting the results in as
few plot windows as possible.</p>
</td></tr>
<tr><td><code id="plot_RLum.Results_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters will be passed to
the <code>plot</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function produces a multiple plot output.  A file output is recommended
(e.g., <a href="grDevices.html#topic+pdf">pdf</a>).
</p>


<h3>Value</h3>

<p>Returns multiple plots.
</p>


<h3>Function version</h3>

<p>0.2.1
</p>


<h3>How to cite</h3>

<p>Burow, C., Kreutzer, S., 2025. plot_RLum.Results(): Plot function for an RLum.Results S4 class object. Function version 0.2.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Not all arguments available for <a href="base.html#topic+plot">plot</a> will be passed!
Only plotting of <code>RLum.Results</code> objects are supported.
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany) <br />
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="base.html#topic+plot">plot</a>, <a href="#topic+plot_RLum">plot_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

###load data
data(ExampleData.DeValues, envir = environment())

# apply the un-logged minimum age model
mam &lt;- calc_MinDose(data = ExampleData.DeValues$CA1, sigmab = 0.2, log = TRUE, plot = FALSE)

##plot
plot_RLum.Results(mam)

# estimate the number of grains on an aliquot
grains&lt;- calc_AliquotSize(grain.size = c(100,150), sample.diameter = 1, plot = FALSE, MC.iter = 100)

##plot
plot_RLum.Results(grains)


</code></pre>

<hr>
<h2 id='plot_ROI'>Create Regions of Interest (ROI) Graphic</h2><span id='topic+plot_ROI'></span>

<h3>Description</h3>

<p>Create ROI graphic with data extracted from the data imported
via <a href="#topic+read_RF2R">read_RF2R</a>. This function is used internally by <a href="#topic+analyse_IRSAR.RF">analyse_IRSAR.RF</a> but
might be of use to work with reduced data from spatially resolved measurements.
The plot dimensions mimic the original image dimensions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ROI(
  object,
  exclude_ROI = c(1),
  dist_thre = -Inf,
  dim.CCD = NULL,
  bg_image = NULL,
  plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ROI_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a> or a <a href="base.html#topic+list">list</a> of such objects (<strong>required</strong>):
data input. Please note that to avoid function errors, only input created
by the functions <a href="#topic+read_RF2R">read_RF2R</a> or <a href="#topic+extract_ROI">extract_ROI</a> is accepted</p>
</td></tr>
<tr><td><code id="plot_ROI_+3A_exclude_roi">exclude_ROI</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): option to remove particular ROIs from the
analysis. Those ROIs are plotted but not coloured and not taken into account
in distance analysis. <code>NULL</code> excludes nothing.</p>
</td></tr>
<tr><td><code id="plot_ROI_+3A_dist_thre">dist_thre</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): euclidean distance threshold in pixel
distance. All ROI for which the euclidean distance is smaller are marked. This
helps to identify ROIs that might be affected by signal cross-talk. Note:
the distance is calculated from the centre of an ROI, e.g., the threshold
should include consider the ROIs or grain radius.</p>
</td></tr>
<tr><td><code id="plot_ROI_+3A_dim.ccd">dim.CCD</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): metric x and y for the recorded (chip)
surface in ¬µm. For instance <code>c(8192,8192)</code>, if set additional x and y-axes are shown</p>
</td></tr>
<tr><td><code id="plot_ROI_+3A_bg_image">bg_image</code></td>
<td>
<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> (<em>optional</em>): background image object
(please note that dimensions are not checked).</p>
</td></tr>
<tr><td><code id="plot_ROI_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="plot_ROI_+3A_...">...</code></td>
<td>
<p>further parameters to manipulate the plot. On top of all arguments of
<a href="graphics.html#topic+plot.default">graphics::plot.default</a> the following arguments are supported: <code>lwd.ROI</code>, <code>lty.ROI</code>,
<code>col.ROI</code>, <code>col.pixel</code>, <code>text.labels</code>, <code>text.offset</code>, <code>grid</code> (<code>TRUE/FALSE</code>), <code>legend</code> (<code>TRUE/FALSE</code>),
<code>legend.text</code>, <code>legend.pos</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An ROI plot and an <a href="#topic+RLum.Results-class">RLum.Results</a> object with a matrix containing
the extracted ROI data and a object produced by <a href="stats.html#topic+dist">stats::dist</a> containing
the euclidean distance between the ROIs.
</p>


<h3>Function version</h3>

<p>0.2.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_ROI(): Create Regions of Interest (ROI) Graphic. Function version 0.2.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+read_RF2R">read_RF2R</a>, <a href="#topic+analyse_IRSAR.RF">analyse_IRSAR.RF</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## simple example
file &lt;- system.file("extdata", "RF_file.rf", package = "Luminescence")
temp &lt;- read_RF2R(file)
plot_ROI(temp)

## in combination with extract_ROI()
m &lt;- matrix(runif(100,0,255), ncol = 10, nrow = 10)
roi &lt;- matrix(c(2.,4,2,5,6,7,3,1,1), ncol = 3)
t &lt;- extract_ROI(object = m, roi = roi)
plot_ROI(t, bg_image = m)

</code></pre>

<hr>
<h2 id='plot_SingleGrainDisc'>Plot a disc with its values</h2><span id='topic+plot_SingleGrainDisc'></span>

<h3>Description</h3>

<p>Shows a schematic representation of the physical appearance
of one disc (one position in the reader)
and illustrates the measured or calculated values per grain location.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SingleGrainDisc(
  object,
  show_coordinates = FALSE,
  show_location_ids = FALSE,
  show_neighbours = FALSE,
  show_positioning_holes = TRUE,
  df_neighbours = NULL,
  ignore_borders = FALSE,
  str_transform = "sqrt",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_SingleGrainDisc_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Results-class">RLum.Results</a> or <a href="base.html#topic+numeric">numeric</a> (<strong>required</strong>): the values
to show, should have length 100.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_show_coordinates">show_coordinates</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Show coordinates (1..10)
in x and in y direction. Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_show_location_ids">show_location_ids</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Show id with every
grain location (1..100). Defaults to <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_show_neighbours">show_neighbours</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a>  (<em>with default</em>): Show which
neighbour connections are taken into account if calculating Moran's I.
This makes sense when there are <code>NA</code> observations, or when a non-standard
neighbour setting is defined.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_show_positioning_holes">show_positioning_holes</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): Show the 3
positioning holes for orientation. Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_df_neighbours">df_neighbours</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<em>with default</em>): only relevant if
<code>show_neighbours</code> is <code>TRUE</code>. Data frame indicating which borders to
consider, and their respective weights (see the description provided for
<a href="#topic+calc_MoransI">calc_MoransI</a>). If <code>NULL</code> (default), this is constructed automatically by
the internal function <code>.get_Neighbours</code>.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_ignore_borders">ignore_borders</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether only grain
locations that do not lie on the border of the disc should be considered
(<code>FALSE</code> by default). Thus if <code>TRUE</code>, only the inner 8x8 grain locations
rather than the full 10x10 are considered. Ignored if <code>df_neighbours</code> is
not <code>NULL</code> or if <code>show_neighbours = FALSE</code>.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_str_transform">str_transform</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>): The observed value of each individual grain is
reflected in the size of a triangle (or other dot-like element). To account for large value differences,
the transformation from value to triangle size can be <code>"lin"</code> (linear), <code>"log"</code> (logarithmic) and <code>"sqrt"</code>
(square root). Defaults to <code>"sqrt"</code>, so that the surface is linear to the value. Note that
the log and sqrt transformations can come with an addition to avoid negative values. When the legend
is shown, the actual lower, middle and upper values are printed.</p>
</td></tr>
<tr><td><code id="plot_SingleGrainDisc_+3A_...">...</code></td>
<td>
<p>other arguments to be given to the base R plot function, such
as <code>main</code>, <code>col</code> and <code>pch</code>. <code>legend</code> can be used to enable/disable the
legend (<code>FALSE</code> by default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Depending of the available plotting space, some optional elements might have not enough room
to be displayed. As this function is wrapped around the base plot function, one can also choose to add elements
manually.
</p>


<h3>How to cite</h3>

<p>Boer, A.d., Steinbuch, L., 2025. plot_SingleGrainDisc(): Plot a disc with its values. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Anna-Maartje de Boer, Luc Steinbuch, Wageningen University &amp; Research, 2025
, RLum Developer Team</p>


<h3>References</h3>

<p>de Boer, A-M., Steinbuch, L., Heuvelink, G.B.M., Wallinga, J., 2025.
A novel tool to assess crosstalk in single-grain luminescence detection.
Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
plot_SingleGrainDisc(1:100)

</code></pre>

<hr>
<h2 id='plot_ViolinPlot'>Create a violin plot</h2><span id='topic+plot_ViolinPlot'></span>

<h3>Description</h3>

<p>Draws a kernel density plot in combination with a boxplot in its middle. The shape of the violin
is constructed using a mirrored density curve. This plot is especially designed for cases
where the individual errors are zero or too small to be visualised. The idea for this plot is
based on the 'volcano plot' in the ggplot2 package by Hadley Wickham and Winston Chang.
The general idea for the violin plot seems to have been introduced by
Hintze and Nelson (1998).
</p>
<p>The function is passing several arguments to the functions <a href="base.html#topic+plot">plot</a>,
<a href="stats.html#topic+density">stats::density</a>, <a href="graphics.html#topic+boxplot">graphics::boxplot</a>:
</p>
<p>Supported arguments are:
<code>xlim</code>, <code>main</code>, <code>xlab</code>, <code>ylab</code>, <code>col.violin</code>, <code>col.boxplot</code>, <code>mtext</code>, <code>cex</code>, <code>mtext</code>
</p>
<p><strong><code style="white-space: pre;">&#8288;Valid summary keywords&#8288;</code></strong>
</p>
<p><code>'n'</code>, <code>'mean'</code>, <code>'median'</code>, <code>'sd.abs'</code>, <code>'sd.rel'</code>, <code>'se.abs'</code>, <code>'se.rel'</code>.
<code>'skewness'</code>, <code>'kurtosis'</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ViolinPlot(
  data,
  boxplot = TRUE,
  rug = TRUE,
  summary = NULL,
  summary.pos = "sub",
  na.rm = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_ViolinPlot_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="#topic+RLum.Results-class">RLum.Results</a> (<strong>required</strong>):
input data for plotting. Alternatively a <a href="base.html#topic+data.frame">data.frame</a> or a <a href="base.html#topic+matrix">matrix</a> can
be provided, but only the first column will be considered by the
function</p>
</td></tr>
<tr><td><code id="plot_ViolinPlot_+3A_boxplot">boxplot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable boxplot</p>
</td></tr>
<tr><td><code id="plot_ViolinPlot_+3A_rug">rug</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable rug</p>
</td></tr>
<tr><td><code id="plot_ViolinPlot_+3A_summary">summary</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
add statistic measures of centrality and dispersion to the plot.
Can be one or more of several keywords. See details for available keywords.</p>
</td></tr>
<tr><td><code id="plot_ViolinPlot_+3A_summary.pos">summary.pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+character">character</a> (<em>with default</em>):
optional position keywords (cf. <a href="graphics.html#topic+legend">legend</a>) for the statistical summary.
Alternatively, the keyword <code>"sub"</code> may be specified to place the summary
below the plot header. However, this latter option in only possible if
<code>mtext</code> is not used.</p>
</td></tr>
<tr><td><code id="plot_ViolinPlot_+3A_na.rm">na.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
exclude NA values from the data set prior to any further operations.</p>
</td></tr>
<tr><td><code id="plot_ViolinPlot_+3A_...">...</code></td>
<td>
<p>further arguments and graphical parameters passed to
<a href="graphics.html#topic+plot.default">plot.default</a>, <a href="stats.html#topic+density">stats::density</a> and <a href="graphics.html#topic+boxplot">boxplot</a>. See details for further
information.</p>
</td></tr>
</table>


<h3>Function version</h3>

<p>0.1.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. plot_ViolinPlot(): Create a violin plot. Function version 0.1.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Although the code for this function was developed independently and just the idea for the plot
was based on the 'ggplot2' package plot type 'volcano', it should be mentioned that, beyond this,
two other R packages exist providing a possibility to produces this kind of plot, namely:
<code>'vioplot'</code> and <code>'violinmplot'</code> (see references for details).
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Daniel Adler (2005). vioplot: A violin plot is a combination of a box plot and a kernel density plot.
R package version 0.2 http://CRAN.R-project.org/package=violplot
</p>
<p>Hintze, J.L., Nelson, R.D., 1998. A Box Plot-Density Trace Synergism. The American Statistician 52, 181-184.
</p>
<p>Raphael W. Majeed (2012). violinmplot: Combination of violin plot with mean and standard deviation.
R package version 0.2.1. http://CRAN.R-project.org/package=violinmplot
</p>
<p>Wickham. H (2009). ggplot2: elegant graphics for data analysis. Springer New York.
</p>


<h3>See Also</h3>

<p><a href="stats.html#topic+density">stats::density</a>, <a href="base.html#topic+plot">plot</a>, <a href="graphics.html#topic+boxplot">boxplot</a>, <a href="graphics.html#topic+rug">rug</a>, <a href="#topic+calc_Statistics">calc_Statistics</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## read example data set
data(ExampleData.DeValues, envir = environment())
ExampleData.DeValues &lt;- convert_Second2Gray(ExampleData.DeValues$BT998,
                                            c(0.0438,0.0019))

## create plot straightforward
plot_ViolinPlot(data = ExampleData.DeValues)

</code></pre>

<hr>
<h2 id='PSL2Risoe.BINfileData'>Convert portable OSL data to a Risoe.BINfileData object</h2><span id='topic+PSL2Risoe.BINfileData'></span>

<h3>Description</h3>

<p>This function is defunct, use <a href="#topic+convert_PSL2Risoe.BINfileData">convert_PSL2Risoe.BINfileData</a> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PSL2Risoe.BINfileData(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PSL2Risoe.BINfileData_+3A_...">...</code></td>
<td>
<p>Unused.</p>
</td></tr>
</table>

<hr>
<h2 id='read_BIN2R'>Import Ris√∏ BIN/BINX-files into R</h2><span id='topic+read_BIN2R'></span>

<h3>Description</h3>

<p>Import a <code style="white-space: pre;">&#8288;*.bin&#8288;</code> or a <code style="white-space: pre;">&#8288;*.binx&#8288;</code> file produced by a Ris√∏ DA15 and DA20 TL/OSL
reader into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_BIN2R(
  file,
  show.raw.values = FALSE,
  position = NULL,
  n.records = NULL,
  zero_data.rm = TRUE,
  duplicated.rm = FALSE,
  fastForward = FALSE,
  show.record.number = FALSE,
  txtProgressBar = TRUE,
  forced.VersionNumber = NULL,
  ignore.RECTYPE = FALSE,
  pattern = NULL,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_BIN2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>): path and file name of the
BIN/BINX file (URLs are supported). If input is a <code>list</code> it should comprise
only <code>character</code>s representing each valid path and BIN/BINX-file names.
Alternatively, the input character can be just a directory (path), in which
case the function tries to detect and import all BIN/BINX files found in
the directory.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_show.raw.values">show.raw.values</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
shows raw values from BIN-file for <code>LTYPE</code>, <code>DTYPE</code> and <code>LIGHTSOURCE</code> without
translation in characters. Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_position">position</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
imports only the selected position. Note: the import performance will not
benefit by any selection made here.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_n.records">n.records</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): limits the number of imported records
to the provided record id (e.g., <code>n.records = 1:10</code> imports the first ten records,
while <code>n.records = 3</code> imports only record number 3. Can be used in combination with
<code>show.record.number</code> for debugging purposes, e.g. corrupt BIN-files.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_zero_data.rm">zero_data.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
remove erroneous data with no count values. As such data are usually not
needed for the subsequent data analysis they will be removed by default.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_duplicated.rm">duplicated.rm</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
remove duplicated entries if <code>TRUE</code>. This may happen due to an erroneous
produced BIN/BINX-file. This option compares only predecessor and successor.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_fastforward">fastForward</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if <code>TRUE</code> for a more efficient data processing only a list of <code>RLum.Analysis</code>
objects is returned instead of a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_show.record.number">show.record.number</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
shows record number of the imported record, for debugging usage only.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.
Ignored if <code>verbose = FALSE</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar. Ignored if <code>verbose = FALSE</code>.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_forced.versionnumber">forced.VersionNumber</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
allows to cheat the version number check in the function by own values for
cases where the BIN-file version is not supported.
Can be provided as <code>list</code> if <code>file</code> is a <code>list</code>.
</p>
<p><strong>Note:</strong> The usage is at own risk, only supported BIN-file versions have been tested.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_ignore.rectype">ignore.RECTYPE</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> or <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
this argument allows to ignore values in the byte 'RECTYPE' (BIN-file version 08),
in case there are not documented or faulty set. In this case the corrupted records are skipped.
If the setting is <a href="base.html#topic+numeric">numeric</a> (e.g., <code>ignore.RECTYPE = 128</code>), records of those type are ignored
for import.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_pattern">pattern</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
argument that is used if only a path is provided. The argument will than be
passed to the function <a href="base.html#topic+list.files">list.files</a> used internally to construct a <code>list</code>
of wanted files</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="read_BIN2R_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="#topic+Risoe.BINfileData2RLum.Analysis">Risoe.BINfileData2RLum.Analysis</a>. Please note that any matching argument
automatically sets <code>fastForward = TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The binary data file is parsed byte by byte following the data structure
published in the Appendices of the Analyst manual p. 42.
</p>
<p>For the general BIN/BINX-file structure, the reader is referred to the
Ris√∏ website: <a href="https://www.fysik.dtu.dk">https://www.fysik.dtu.dk</a>
</p>


<h3>Value</h3>

<p>Returns an S4 <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object containing two
slots:
</p>
<table role = "presentation">
<tr><td><code>METADATA</code></td>
<td>
<p>A <a href="base.html#topic+data.frame">data.frame</a> containing all variables stored in the BIN-file.</p>
</td></tr>
<tr><td><code>DATA</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> containing a numeric <a href="base.html#topic+vector">vector</a> of the measured data.
The ID corresponds to the record ID in METADATA.</p>
</td></tr>
</table>
<p>If <code>fastForward = TRUE</code> a list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object is returned. The
internal coercing is done using the function <a href="#topic+Risoe.BINfileData2RLum.Analysis">Risoe.BINfileData2RLum.Analysis</a>
</p>


<h3>Function version</h3>

<p>0.18
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Fuchs, M.C., Colombo, M., 2025. read_BIN2R(): Import Ris√∏ BIN/BINX-files into R. Function version 0.18. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The function works for BIN/BINX-format versions 03, 04, 05, 06, 07 and 08. The
version number depends on the used Sequence Editor.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Margret C. Fuchs, HZDR Freiberg, (Germany) <br />
Marco Colombo, Institute of Geography, Heidelberg University (Germany)<br />
based on information provided by Torben Lapp and Karsten Bracht Nielsen (Ris√∏ DTU, Denmark)
, RLum Developer Team</p>


<h3>References</h3>

<p>DTU Nutech, 2016. The Sequence Editor, Users Manual, February, 2016.
<a href="https://www.fysik.dtu.dk">https://www.fysik.dtu.dk</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+write_R2BIN">write_R2BIN</a>, <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>,
<a href="base.html#topic+readBin">base::readBin</a>, <a href="#topic+merge_Risoe.BINfileData">merge_Risoe.BINfileData</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
<a href="utils.html#topic+txtProgressBar">utils::txtProgressBar</a>, <a href="base.html#topic+list.files">list.files</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file &lt;- system.file("extdata/BINfile_V8.binx", package = "Luminescence")
temp &lt;- read_BIN2R(file)
temp

</code></pre>

<hr>
<h2 id='read_Daybreak2R'>Import measurement data produced by a Daybreak TL/OSL reader into R</h2><span id='topic+read_Daybreak2R'></span>

<h3>Description</h3>

<p>Import a TXT-file (ASCII file) or a DAT-file (binary file) produced by a
Daybreak reader into R. The import of the DAT-files is limited to the file
format described for the software TLAPLLIC v.3.2 used for a Daybreak, model 1100.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_Daybreak2R(file, raw = FALSE, verbose = TRUE, txtProgressBar = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_Daybreak2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
path and file name of the file to be imported. Alternatively a list of file
names can be provided or just the path a folder containing measurement data.
Please note that the specific, common, file extension (txt) is likely
leading to function failures during import when just a path is provided.</p>
</td></tr>
<tr><td><code id="read_Daybreak2R_+3A_raw">raw</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if the input is a DAT-file (binary) a <a href="data.table.html#topic+data.table">data.table::data.table</a> instead of
the <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object can be returned for debugging purposes.</p>
</td></tr>
<tr><td><code id="read_Daybreak2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="read_Daybreak2R_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable <a href="utils.html#topic+txtProgressBar">txtProgressBar</a>.</p>
</td></tr>
<tr><td><code id="read_Daybreak2R_+3A_...">...</code></td>
<td>
<p>not in use, for compatibility reasons only</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects (each per position) is provided.
</p>


<h3>Function version</h3>

<p>0.3.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Zink, A., 2025. read_Daybreak2R(): Import measurement data produced by a Daybreak TL/OSL reader into R. Function version 0.3.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong><code style="white-space: pre;">&#8288;[BETA VERSION]&#8288;</code></strong>
This function still needs to be tested properly. In particular
the function has underwent only very rough tests using a few files.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Antoine Zink, C2RMF, Palais du Louvre, Paris (France)
</p>
<p>The ASCII-file import is based on a suggestion by Willian Amidon and Andrew Louis Gorin
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="data.table.html#topic+data.table">data.table::data.table</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
file &lt;- system.file("extdata/Daybreak_TestFile.txt", package = "Luminescence")
temp &lt;- read_Daybreak2R(file)

## End(Not run)

</code></pre>

<hr>
<h2 id='read_HeliosOSL2R'>Import Luminescence Data from Helios Luminescence Reader</h2><span id='topic+read_HeliosOSL2R'></span>

<h3>Description</h3>

<p>Straightforward import of files with the ending <code>.osl</code> produced
by the zero rad Helios luminescence reader and conversion to <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_HeliosOSL2R(file, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_HeliosOSL2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>): path to file to be imported. Can be a <a href="base.html#topic+list">list</a>
for further processing</p>
</td></tr>
<tr><td><code id="read_HeliosOSL2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="read_HeliosOSL2R_+3A_...">...</code></td>
<td>
<p>not in use, for compatibility reasons only</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. read_HeliosOSL2R(): Import Luminescence Data from Helios Luminescence Reader. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Thanks to Krzysztof Maternicki for providing example data.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file &lt;- system.file("extdata/HeliosOSL_Example.osl", package = "Luminescence")
read_HeliosOSL2R(file)

</code></pre>

<hr>
<h2 id='read_PSL2R'>Import PSL files to R</h2><span id='topic+read_PSL2R'></span>

<h3>Description</h3>

<p>Imports PSL files produced by a SUERC portable OSL reader into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_PSL2R(
  file,
  drop_bg = FALSE,
  as_decay_curve = TRUE,
  smooth = FALSE,
  merge = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_PSL2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
path and file name of the PSL file. If input is a <code>vector</code> it should comprise
only <code>character</code>s representing valid paths and PSL file names.
Alternatively the input character can be just a directory (path). In this case the
the function tries to detect and import all PSL files found in the directory.</p>
</td></tr>
<tr><td><code id="read_PSL2R_+3A_drop_bg">drop_bg</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to automatically remove all non-OSL/IRSL curves.</p>
</td></tr>
<tr><td><code id="read_PSL2R_+3A_as_decay_curve">as_decay_curve</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
Portable OSL Reader curves are often given as cumulative light sum curves.
Use <code>TRUE</code> (default) to convert the curves to the more usual decay form.</p>
</td></tr>
<tr><td><code id="read_PSL2R_+3A_smooth">smooth</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to apply Tukey's Running Median Smoothing for OSL and IRSL decay curves.
Smoothing is encouraged if you see random signal drops within the decay curves related
to hardware errors.</p>
</td></tr>
<tr><td><code id="read_PSL2R_+3A_merge">merge</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to merge all <code>RLum.Analysis</code> objects. Only applicable if multiple
files are imported.</p>
</td></tr>
<tr><td><code id="read_PSL2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="read_PSL2R_+3A_...">...</code></td>
<td>
<p>currently not used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides an import routine for the SUERC portable OSL Reader PSL
format (measurement data and sequence). PSL files are just plain text and can be
viewed with any text editor.  Due to the formatting of PSL files this import
function relies heavily on regular expression to find and extract all relevant information. See <strong>note</strong>.
</p>


<h3>Value</h3>

<p>Returns an S4 <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object containing
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects for each curve.
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Burow, C., Kreutzer, S., 2025. read_PSL2R(): Import PSL files to R. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Because this function relies heavily on regular expressions to parse
PSL files it is currently only in beta status. If the routine fails to import
a specific PSL file please report to <code style="white-space: pre;">&#8288;&lt;christoph.burow@gmx.net&gt;&#8288;</code> so the
function can be updated.
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany),
Sebastian Kreutzer, Institut of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# (1) Import PSL file to R

file &lt;- system.file("extdata", "DorNie_0016.psl", package = "Luminescence")
psl &lt;- read_PSL2R(file, drop_bg = FALSE, as_decay_curve = TRUE, smooth = TRUE, merge = FALSE)
print(str(psl, max.level = 3))
plot(psl, combine = TRUE)

</code></pre>

<hr>
<h2 id='read_RF2R'>Import RF-files to R</h2><span id='topic+read_RF2R'></span>

<h3>Description</h3>

<p>Import files produced by the IR-RF 'ImageJ' macro (<code>SR-RF.ijm</code>; Mittelstra√ü and Kreutzer, 2021) into R and create a list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_RF2R(file, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_RF2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>): path and file name of the RF file. Alternatively a list of file
names can be provided.</p>
</td></tr>
<tr><td><code id="read_RF2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="read_RF2R_+3A_...">...</code></td>
<td>
<p>not used, only for compatible reasons</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The results of spatially resolved IR-RF data are summarised in
so-called RF-files (Mittelstra√ü and Kreutzer, 2021).
This functions provides an easy import to process the data seamlessly with the R package 'Luminescence'.
The output of the function can be passed to function <a href="#topic+analyse_IRSAR.RF">analyse_IRSAR.RF</a>.
</p>


<h3>Value</h3>

<p>Returns an S4 <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object containing
<a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects for each curve.
</p>


<h3>Function version</h3>

<p>0.1.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. read_RF2R(): Import RF-files to R. Function version 0.1.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Geography &amp; Earth Science, Aberystwyth University (United Kingdom)
, RLum Developer Team</p>


<h3>References</h3>

<p>Mittelstra√ü, D., Kreutzer, S., 2021. Spatially resolved infrared radiofluorescence:
single-grain K-feldspar dating using CCD imaging. Geochronology 3, 299‚Äì319. <a href="https://doi.org/10.5194/gchron-3-299-2021">doi:10.5194/gchron-3-299-2021</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+analyse_IRSAR.RF">analyse_IRSAR.RF</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##Import
file &lt;- system.file("extdata", "RF_file.rf", package = "Luminescence")
temp &lt;- read_RF2R(file)

</code></pre>

<hr>
<h2 id='read_SPE2R'>Import Princeton Instruments (TM) SPE-file into R</h2><span id='topic+read_SPE2R'></span>

<h3>Description</h3>

<p>Function imports Princeton Instruments (TM) SPE-files into R environment and
provides <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> objects as output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_SPE2R(
  file,
  output.object = "RLum.Data.Image",
  frame.range,
  txtProgressBar = TRUE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_SPE2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
SPE-file name (including path), e.g.
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;[WIN]&#8288;</code>: <code>read_SPE2R("C:/Desktop/test.spe")</code>
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;[MAC/LINUX]&#8288;</code>: <code>read_SPE2R("/User/test/Desktop/test.spe")</code>.
Additionally, it can be a URL starting with http:// or https://.
</p>
</li></ul>
</td></tr>
<tr><td><code id="read_SPE2R_+3A_output.object">output.object</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
set <code>RLum</code> output object.  Allowed types are <code>"RLum.Data.Spectrum"</code>,
<code>"RLum.Data.Image"</code> or <code>"matrix"</code></p>
</td></tr>
<tr><td><code id="read_SPE2R_+3A_frame.range">frame.range</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a> (<em>optional</em>):
limit frame range, e.g. select first 100 frames by <code>frame.range = c(1,100)</code></p>
</td></tr>
<tr><td><code id="read_SPE2R_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar. Ignored if <code>verbose = FALSE</code>.</p>
</td></tr>
<tr><td><code id="read_SPE2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="read_SPE2R_+3A_...">...</code></td>
<td>
<p>not used, for compatibility reasons only</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function provides an R only import routine for the Princeton Instruments
SPE format. Import functionality is based on the file format description provided by
Princeton Instruments and a MatLab script written by Carl Hall (see
references).
</p>


<h3>Value</h3>

<p>Depending on the chosen option the functions returns three different
type of objects:
</p>
<p><code>output.object</code>
</p>
<p><code>RLum.Data.Spectrum</code>
</p>
<p>An object of type <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> is returned.  Row
sums are used to integrate all counts over one channel.
</p>
<p><code>RLum.Data.Image</code>
</p>
<p>An object of type <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> is returned.  Due to
performance reasons the import is aborted for files containing more than 100
frames. This limitation can be overwritten manually by using the argument
<code>frame.range</code>.
</p>
<p><code>matrix</code>
</p>
<p>Returns a matrix of the form: Rows = Channels, columns = Frames. For the
transformation the function <a href="#topic+get_RLum">get_RLum</a> is used,
meaning that the same results can be obtained by using the function
<a href="#topic+get_RLum">get_RLum</a> on an <code>RLum.Data.Spectrum</code> or <code>RLum.Data.Image</code> object.
</p>


<h3>Function version</h3>

<p>0.1.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. read_SPE2R(): Import Princeton Instruments (TM) SPE-file into R. Function version 0.1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>The function does not test whether the input data are spectra or pictures for spatial resolved analysis!</strong>
</p>
<p>The function has been successfully tested for SPE format versions 2.x.
</p>
<p><em>Currently not all information provided by the SPE format are supported.</em>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Princeton Instruments, 2014. Princeton Instruments SPE 3.0 File
Format Specification, Version 1.A (for document URL please use an internet search machine)
</p>
<p>Hall, C., 2012: readSPE.m.
<code style="white-space: pre;">&#8288;https://www.mathworks.com/matlabcentral/fileexchange/35940-readspe&#8288;</code>
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+readBin">readBin</a>, <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## to run examples uncomment lines and run the code

##(1) Import data as RLum.Data.Spectrum object
#file &lt;- file.choose()
#temp &lt;- read_SPE2R(file)
#temp

##(2) Import data as RLum.Data.Image object
#file &lt;- file.choose()
#temp &lt;- read_SPE2R(file, output.object = "RLum.Data.Image")
#temp

##(3) Import data as matrix object
#file &lt;- file.choose()
#temp &lt;- read_SPE2R(file, output.object = "matrix")
#temp

##(4) Export raw data to csv, if temp is a RLum.Data.Spectrum object
# write.table(x = get_RLum(temp),
#             file = "[your path and filename]",
#             sep = ";", row.names = FALSE)


</code></pre>

<hr>
<h2 id='read_TIFF2R'>Import TIFF Image Data into R</h2><span id='topic+read_TIFF2R'></span>

<h3>Description</h3>

<p>Simple wrapper around <a href="tiff.html#topic+readTIFF">tiff::readTIFF</a> to import TIFF images
and TIFF image stacks to be further processed within the package <code>'Luminescence'</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_TIFF2R(file, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_TIFF2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>): file name</p>
</td></tr>
<tr><td><code id="read_TIFF2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="read_TIFF2R_+3A_...">...</code></td>
<td>
<p>not in use, for compatibility reasons only</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> object
</p>


<h3>Function version</h3>

<p>0.1.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. read_TIFF2R(): Import TIFF Image Data into R. Function version 0.1.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="tiff.html#topic+readTIFF">tiff::readTIFF</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
file &lt;- file.choose()
image &lt;- read_TIFF2R(file)


## End(Not run)

</code></pre>

<hr>
<h2 id='read_XSYG2R'>Import XSYG files to R</h2><span id='topic+read_XSYG2R'></span>

<h3>Description</h3>

<p>Imports XSYG-files produced by a Freiberg Instruments lexsyg reader into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_XSYG2R(
  file,
  recalculate.TL.curves = TRUE,
  n_records = NULL,
  fastForward = FALSE,
  import = TRUE,
  pattern = ".xsyg",
  verbose = TRUE,
  txtProgressBar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_XSYG2R_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+list">list</a> (<strong>required</strong>):
path and file name of the XSYG file. If input is a <code>list</code> it should comprise
only <code>character</code>s representing each valid path and XSYG-file names.
Alternatively, the input character can be just a directory (path), in which
case the function tries to detect and import all XSYG-files found in the
directory.</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_recalculate.tl.curves">recalculate.TL.curves</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if set to <code>TRUE</code>, TL curves are returned as temperature against count values
(see details for more information) Note: The option overwrites the time vs.
count TL curve. Select <code>FALSE</code> to import the raw data delivered by the
lexsyg. Works for TL curves and spectra.</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_n_records">n_records</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): set the number of records to be imported; by default
the function attempts to import all records</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_fastforward">fastForward</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if <code>TRUE</code> for a more efficient data processing only a list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
objects is returned.</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_import">import</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if set to <code>FALSE</code>, only the XSYG file structure is shown.</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_pattern">pattern</code></td>
<td>
<p><a href="base.html#topic+regex">regex</a> (<em>with default</em>):
optional regular expression if <code>file</code> is a link to a folder, to select just
specific XSYG-files</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): enable/disable output to the
terminal. If verbose is <code>FALSE</code> the <code>txtProgressBar</code> is also switched off</p>
</td></tr>
<tr><td><code id="read_XSYG2R_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar during import. Ignored if <code>verbose = FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>How does the import function work?</strong>
</p>
<p>The function uses the <code>'XML'</code> package to parse the file structure. Each
sequence is subsequently translated into an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.
</p>
<p><strong>General structure XSYG format</strong>
</p>
<div class="sourceCode"><pre>&lt;?xml?&gt;
&lt;Sample&gt;
  &lt;Sequence&gt;
    &lt;Record&gt;
      &lt;Curve name="first curve" /&gt;
      &lt;Curve name="curve with data"&gt;x0 , y0 ; x1 , y1 ; x2 , y2 ; x3 , y3&lt;/Curve&gt;
    &lt;/Record&gt;
  &lt;/Sequence&gt;
&lt;/Sample&gt;
</pre></div>
<p>So far, each
XSYG file can only contain one <code style="white-space: pre;">&#8288;&lt;Sample&gt;&lt;/Sample&gt;&#8288;</code>, but multiple
sequences.
</p>
<p>Each record may comprise several curves.
</p>
<p><strong>TL curve recalculation</strong>
</p>
<p>On the FI lexsyg device TL curves are recorded as time against count values.
Temperature values are monitored on the heating plate and stored in a
separate curve (time vs. temperature). If the option
<code>recalculate.TL.curves = TRUE</code> is chosen, the time values for each TL
curve are replaced by temperature values.
</p>
<p>Practically, this means combining two matrices (Time vs. Counts and Time vs.
Temperature) with different row numbers by their time values. Three cases
are considered:
</p>

<ol>
<li><p> HE: Heating element
</p>
</li>
<li><p> PMT: Photomultiplier tube
</p>
</li>
<li><p> Interpolation is done using the function <a href="stats.html#topic+approx">approx</a>
</p>
</li></ol>

<p>CASE (1): <code>nrow(matrix(PMT))</code> &gt; <code>nrow(matrix(HE))</code>
</p>
<p>Missing temperature values from the heating element are calculated using
time values from the PMT measurement.
</p>
<p>CASE (2): <code>nrow(matrix(PMT))</code> &lt; <code>nrow(matrix(HE))</code>
</p>
<p>Missing count values from the PMT are calculated using time values from the
heating element measurement.
</p>
<p>CASE (3): <code>nrow(matrix(PMT))</code> == <code>nrow(matrix(HE))</code>
</p>
<p>A new matrix is produced using temperature values from the heating element
and count values from the PMT.
</p>
<p><strong>Note:</strong>
Please note that due to the recalculation of the temperature
values based on values delivered by the heating element, it may happen that
multiple count values exists for each temperature value and temperature
values may also decrease during heating, not only increase.
</p>
<p><strong>Advanced file import</strong>
</p>
<p>To allow for a more efficient usage of the function, instead of single path
to a file just a directory can be passed as input. In this particular case
the function tries to extract all XSYG-files found in the directory and import
them all. Using this option internally the function constructs as list of
the XSYG-files found in the directory. Please note no recursive detection
is supported as this may lead to endless loops.
</p>


<h3>Value</h3>

<p><strong>Using the option <code>import = FALSE</code></strong>
</p>
<p>A list consisting of two elements is shown:
</p>

<ul>
<li> <p><a href="base.html#topic+data.frame">data.frame</a> with information on file.
</p>
</li>
<li> <p><a href="base.html#topic+data.frame">data.frame</a> with information on the sequences stored in the XSYG file.
</p>
</li></ul>

<p><strong>Using the option <code>import = TRUE</code> (default)</strong>
</p>
<p>A list is provided, the list elements
contain: </p>
<table role = "presentation">
<tr><td><code>Sequence.Header</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> with information on the
sequence.</p>
</td></tr> <tr><td><code>Sequence.Object</code></td>
<td>
<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
containing the curves.</p>
</td></tr>
</table>


<h3>Function version</h3>

<p>0.6.14
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. read_XSYG2R(): Import XSYG files to R. Function version 0.6.14. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function is a beta version as the XSYG file format is not yet
fully specified. Thus, further file operations (merge, export, write) should
be done using the functions provided with the package <code>'XML'</code>.
</p>
<p><strong>So far, no image data import is provided!</strong> <br />
Corresponding values in the XSXG file are skipped.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Grehl, S., Kreutzer, S., Hoehne, M., 2013. Documentation of the
XSYG file format. Unpublished Technical Note. Freiberg, Germany
</p>
<p><strong>Further reading</strong>
</p>
<p>XML: <a href="https://en.wikipedia.org/wiki/XML">https://en.wikipedia.org/wiki/XML</a>
</p>


<h3>See Also</h3>

<p><code>'XML'</code>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="stats.html#topic+approx">approx</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##(1) import XSYG file to R (uncomment for usage)

#FILE &lt;- file.choose()
#temp &lt;- read_XSYG2R(FILE)

##(2) additional examples for pure XML import using the package XML
##    (uncomment for usage)

  ##import entire XML file
  #FILE &lt;- file.choose()
  #temp &lt;- XML::xmlRoot(XML::xmlTreeParse(FILE))

  ##search for specific subnodes with curves containing 'OSL'
  #getNodeSet(temp, "//Sample/Sequence/Record[@recordType = 'OSL']/Curve")

##(2) How to extract single curves ... after import
data(ExampleData.XSYG, envir = environment())

##grep one OSL curves and plot the first curve
OSLcurve &lt;- get_RLum(OSL.SARMeasurement$Sequence.Object, recordType="OSL")[[1]]

##(3) How to see the structure of an object?
structure_RLum(OSL.SARMeasurement$Sequence.Object)

</code></pre>

<hr>
<h2 id='replicate_RLum'>General replication function for RLum S4 class objects</h2><span id='topic+replicate_RLum'></span>

<h3>Description</h3>

<p>The function replicates RLum S4 class objects and returns a list of such
objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replicate_RLum(object, times = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replicate_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
an <a href="#topic+RLum-class">RLum</a> object</p>
</td></tr>
<tr><td><code id="replicate_RLum_+3A_times">times</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
number for times each element is repeated element</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="base.html#topic+list">list</a> with the object repeated.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. replicate_RLum(): General replication function for RLum S4 class objects. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum-class">RLum</a>
</p>

<hr>
<h2 id='report_RLum'>Create a HTML-report for (RLum) objects</h2><span id='topic+report_RLum'></span>

<h3>Description</h3>

<p>Create a HTML-report for (RLum) objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_RLum(
  object,
  file = tempfile(),
  title = "RLum.Report",
  compact = TRUE,
  timestamp = TRUE,
  show_report = TRUE,
  launch.browser = FALSE,
  css.file = NULL,
  quiet = TRUE,
  clean = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="report_RLum_+3A_object">object</code></td>
<td>
<p>(<strong>required</strong>):
The object to be reported on, preferably of any <code>RLum</code>-class.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
A character string naming the output file. If no filename is provided a
temporary file is created.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_title">title</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
A character string specifying the title of the document.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_compact">compact</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
When <code>TRUE</code> the following report components are hidden:
<code style="white-space: pre;">&#8288;@.pid&#8288;</code>, <code style="white-space: pre;">&#8288;@.uid&#8288;</code>, <code>'Object structure'</code>, <code>'Session Info'</code>
and only the first and last 5 rows of long matrices and data frames are shown.
See details.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_timestamp">timestamp</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to add a timestamp to the filename (suffix).</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_show_report">show_report</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): If set to <code>TRUE</code> the function tries to display
the report output in the local viewer, e.g., within <em>RStudio</em> after rendering.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_launch.browser">launch.browser</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to open the HTML file in the system's default web browser after
it has been rendered.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_css.file">css.file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
Path to a CSS file to change the default styling of the HTML document.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_quiet">quiet</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to suppress printing of the pandoc command line.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_clean">clean</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
<code>TRUE</code> to clean intermediate files created during rendering.</p>
</td></tr>
<tr><td><code id="report_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to or from other methods and to control
the document's structure (see details).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function creates a HTML-report for a given object, listing its complete
structure and content. The object itself is saved as a serialised .Rds file.
The report file serves both as a convenient way of browsing through objects with
complex data structures as well as a mean of properly documenting and saving
objects.
</p>
<p>The HTML report is created with <a href="rmarkdown.html#topic+render">rmarkdown::render</a> and has the
following structure:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Section</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Header</code> </td><td style="text-align: left;"> A summary of general characteristics of the object </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;Object content&#8288;</code> </td><td style="text-align: left;"> A comprehensive list of the complete structure and content of the provided object. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;Object structure&#8288;</code> </td><td style="text-align: left;"> Summary of the objects structure given as a table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>File</code> </td><td style="text-align: left;"> Information on the saved RDS file </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;Session Info&#8288;</code> </td><td style="text-align: left;"> Captured output from <code>sessionInfo()</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Plots</code> </td><td style="text-align: left;"> (<em>optional</em>) For <code>RLum-class</code> objects a variable number of plots </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The structure of the report can be controlled individually by providing one or more of the
following arguments (all <code>logical</code>):
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Argument</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>header</code> </td><td style="text-align: left;"> Hide or show general information on the object </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>main</code> </td><td style="text-align: left;"> Hide or show the object's content </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>structure</code> </td><td style="text-align: left;"> Hide or show object's structure </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>rds</code> </td><td style="text-align: left;"> Hide or show information on the saved RDS file </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>session</code> </td><td style="text-align: left;"> Hide or show the session info </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>plot</code> </td><td style="text-align: left;"> Hide or show the plots (depending on object) </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Note that these arguments have higher precedence than <code>compact</code>.
</p>
<p>Further options that can be provided via the <code>...</code> argument:
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Argument</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>short_table</code> </td><td style="text-align: left;"> If <code>TRUE</code> only show the first and last 5 rows of long tables. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>theme</code> </td><td style="text-align: left;"> Specifies the Bootstrap
theme to use for the report. Valid themes include <code>"default"</code>, <code>"cerulean"</code>, <code>"journal"</code>, <code>"flatly"</code>,
<code>"readable"</code>, <code>"spacelab"</code>, <code>"united"</code>, <code>"cosmo"</code>, <code>"lumen"</code>, <code>"paper"</code>, <code>"sandstone"</code>,
<code>"simplex"</code>, and <code>"yeti"</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>highlight</code> </td><td style="text-align: left;"> Specifies the syntax highlighting style.
Supported styles include <code>"default"</code>, <code>"tango"</code>, <code>"pygments"</code>, <code>"kate"</code>, <code>"monochrome"</code>,
<code>"espresso"</code>, <code>"zenburn"</code>, <code>"haddock"</code>, and <code>"textmate"</code>. </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>css</code> </td><td style="text-align: left;"> <code>TRUE</code> or <code>FALSE</code> to enable/disable custom CSS styling </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>The following arguments can be used to customise the report via CSS (Cascading Style Sheets):
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Argument</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>font_family</code> </td><td style="text-align: left;"> Define the font family of the HTML document (default: <code>"arial"</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>headings_size</code> </td><td style="text-align: left;"> Size of the <code style="white-space: pre;">&#8288;&lt;h1&gt;&#8288;</code> to <code style="white-space: pre;">&#8288;&lt;h6&gt;&#8288;</code> tags used to define HTML headings (default: 166%). </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>content_color</code> </td><td style="text-align: left;"> Colour of the object's content (default: #a72925). </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Note that these arguments must all be of class <a href="base.html#topic+character">character</a> and follow standard CSS syntax.
For exhaustive CSS styling you can provide a custom CSS file for argument <code>css.file</code>.
CSS styling can be turned of using <code>css = FALSE</code>.
</p>


<h3>Value</h3>

<p>Writes a HTML and .Rds file.
</p>


<h3>Function version</h3>

<p>0.1.5
</p>


<h3>How to cite</h3>

<p>Burow, C., Kreutzer, S., 2025. report_RLum(): Create a HTML-report for (RLum) objects. Function version 0.1.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function requires the R packages 'rmarkdown', 'pander' and 'rstudioapi'.
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany),
Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany) <br />
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="rmarkdown.html#topic+render">rmarkdown::render</a>, <a href="pander.html#topic+pander_return">pander::pander_return</a>,
<a href="pander.html#topic+openFileInOS">pander::openFileInOS</a>, <a href="rstudioapi.html#topic+viewer">rstudioapi::viewer</a>,
<a href="utils.html#topic+browseURL">browseURL</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
## Example: RLum.Results ----

# load example data
data("ExampleData.DeValues")

# apply the MAM-3 age model and save results
mam &lt;- calc_MinDose(ExampleData.DeValues$CA1, sigmab = 0.2)

# create the HTML report
report_RLum(object = mam, file = "~/CA1_MAM.Rmd",
            timestamp = FALSE,
            title = "MAM-3 for sample CA1")

# when creating a report the input file is automatically saved to a
# .Rds file (see saveRDS()).
mam_report &lt;- readRDS("~/CA1_MAM.Rds")
all.equal(mam, mam_report)


## Example: Temporary file &amp; Viewer/Browser ----

# (a)
# Specifying a filename is not necessarily required. If no filename is provided,
# the report is rendered in a temporary file. If you use the RStudio IDE, the
# temporary report is shown in the interactive Viewer pane.
report_RLum(object = mam)

# (b)
# Additionally, you can view the HTML report in your system's default web browser.
report_RLum(object = mam, launch.browser = TRUE)


## Example: RLum.Analysis ----

data("ExampleData.RLum.Analysis")

# create the HTML report (note that specifying a file
# extension is not necessary)
report_RLum(object = IRSAR.RF.Data, file = "~/IRSAR_RF")


## Example: RLum.Data.Curve ----

data.curve &lt;- get_RLum(IRSAR.RF.Data)[[1]]

# create the HTML report
report_RLum(object = data.curve, file = "~/Data_Curve")

## Example: Any other object ----
x &lt;- list(x = 1:10,
          y = runif(10, -5, 5),
          z = data.frame(a = LETTERS[1:20], b = dnorm(0:9)),
          NA)

report_RLum(object = x, file = "~/arbitray_list")

## End(Not run)

</code></pre>

<hr>
<h2 id='Risoe.BINfileData-class'>Class <code>"Risoe.BINfileData"</code></h2><span id='topic+Risoe.BINfileData-class'></span><span id='topic+show+2CRisoe.BINfileData-method'></span><span id='topic+set_Risoe.BINfileData+2CANY-method'></span><span id='topic+get_Risoe.BINfileData+2CRisoe.BINfileData-method'></span><span id='topic+add_metadata+3C-+2CRisoe.BINfileData-method'></span><span id='topic+rename_metadata+3C-+2CRisoe.BINfileData-method'></span><span id='topic+replace_metadata+3C-+2CRisoe.BINfileData-method'></span><span id='topic+sort_RLum+2CRisoe.BINfileData-method'></span><span id='topic+view+2CRisoe.BINfileData-method'></span>

<h3>Description</h3>

<p>S4 class object for luminescence data in R. The object is produced as output
of the function <a href="#topic+read_BIN2R">read_BIN2R</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'Risoe.BINfileData'
show(object)

## S4 method for signature 'ANY'
set_Risoe.BINfileData(
  METADATA = data.frame(),
  DATA = list(),
  .RESERVED = list()
)

## S4 method for signature 'Risoe.BINfileData'
get_Risoe.BINfileData(object, ...)

## S4 replacement method for signature 'Risoe.BINfileData'
add_metadata(object, info_element) &lt;- value

## S4 replacement method for signature 'Risoe.BINfileData'
rename_metadata(object, info_element) &lt;- value

## S4 replacement method for signature 'Risoe.BINfileData'
replace_metadata(object, info_element, subset = NULL) &lt;- value

## S4 method for signature 'Risoe.BINfileData'
sort_RLum(object, info_element, decreasing = FALSE, ...)

## S4 method for signature 'Risoe.BINfileData'
view(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Risoe.BINfileData-class_+3A_object">object</code></td>
<td>
<p>(<strong>required</strong>) an object of class <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a></p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_metadata">METADATA</code></td>
<td>
<p>Object of class &quot;data.frame&quot; containing the meta information
for each curve.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_data">DATA</code></td>
<td>
<p>Object of class &quot;list&quot; containing numeric vector with count data.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_.reserved">.RESERVED</code></td>
<td>
<p>Object of class &quot;list&quot; containing list of undocumented raw
values for internal use only.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_...">...</code></td>
<td>
<p>other arguments that might be passed</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_info_element">info_element</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>): name of the metadata field
to use in sorting.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_value">value</code></td>
<td>
<p>(<strong>required</strong>) The value assigned to the selected elements
of the metadata field. If <code>NULL</code> the elements named in <code>info_element</code>
will be removed.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_subset">subset</code></td>
<td>
<p><a href="base.html#topic+expression">expression</a> (<em>optional</em>) logical expression to limit the
substitution only to the selected subset of elements</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData-class_+3A_decreasing">decreasing</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether the sort order should
be decreasing (<code>FALSE</code> by default).</p>
</td></tr>
</table>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show(Risoe.BINfileData)</code>: Show structure of RLum and Risoe.BINfile class objects
</p>
</li>
<li> <p><code>set_Risoe.BINfileData(ANY)</code>: The Risoe.BINfileData is normally produced as output of the function read_BIN2R.
This construction method is intended for internal usage only.
</p>
</li>
<li> <p><code>get_Risoe.BINfileData(Risoe.BINfileData)</code>: Formal get-method for Risoe.BINfileData object. It does not allow accessing
the object directly, it is just showing a terminal message.
</p>
</li>
<li> <p><code>add_metadata(Risoe.BINfileData) &lt;- value</code>: Adds metadata to <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects
</p>
</li>
<li> <p><code>rename_metadata(Risoe.BINfileData) &lt;- value</code>: Renames a metadata entry of <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects
</p>
</li>
<li> <p><code>replace_metadata(Risoe.BINfileData) &lt;- value</code>: Replaces or removes metadata of <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects
</p>
</li>
<li> <p><code>sort_RLum(Risoe.BINfileData)</code>: Sort method for <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects
</p>
</li>
<li> <p><code>view(Risoe.BINfileData)</code>: View method for <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>METADATA</code></dt><dd><p>Object of class &quot;data.frame&quot; containing the meta information for each curve.</p>
</dd>
<dt><code>DATA</code></dt><dd><p>Object of class &quot;list&quot; containing numeric vector with count data.</p>
</dd>
<dt><code>.RESERVED</code></dt><dd><p>Object of class &quot;list&quot; containing list of undocumented raw values for internal use only.</p>
</dd>
</dl>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form
<code>new("Risoe.BINfileData", ...)</code>.
</p>


<h3>Function version</h3>

<p>0.4.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. Risoe.BINfileData-class(): Class 'Risoe.BINfileData'. Function version 0.4.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>Internal METADATA - object structure</strong>
</p>
<p>This structure is compatible with BIN/BINX-files version 03-08, however, it does not follow (in its
sequential arrangement) the manual provided by the manufacturer,
but an own structure accounting for the different versions.
</p>

<table>
<tr>
 <td style="text-align: right;">
<strong>#</strong> </td><td style="text-align: left;"> <strong>Name</strong> </td><td style="text-align: left;"> <strong>Data Type</strong> </td><td style="text-align: left;"> <strong>V</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,1]&#8288;</code> </td><td style="text-align: left;"> <code>ID</code>  </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> RLum </td><td style="text-align: left;"> Unique record ID (same ID as in slot <code>DATA</code>)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,2]&#8288;</code> </td><td style="text-align: left;"> <code>SEL</code> </td><td style="text-align: left;"> <code>logic</code> </td><td style="text-align: left;"> RLum </td><td style="text-align: left;"> Record selection, not part official BIN-format, triggered by TAG</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,3]&#8288;</code> </td><td style="text-align: left;"> <code>VERSION</code> </td><td style="text-align: left;"> <code>raw</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> BIN-file version number </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,4]&#8288;</code> </td><td style="text-align: left;"> <code>LENGTH</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Length of this record</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,5]&#8288;</code> </td><td style="text-align: left;"> <code>PREVIOUS</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Length of previous record</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,6]&#8288;</code> </td><td style="text-align: left;"> <code>NPOINTS</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Number of data points in the record</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,7]&#8288;</code> </td><td style="text-align: left;"> <code>RECTYPE</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Record type </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,8]&#8288;</code> </td><td style="text-align: left;"> <code>RUN</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Run number</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,9]&#8288;</code> </td><td style="text-align: left;"> <code>SET</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Set number</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,10]&#8288;</code>  </td><td style="text-align: left;"> <code>POSITION</code> </td><td style="text-align: left;">  <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Position number</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,11]&#8288;</code> </td><td style="text-align: left;"> <code>GRAIN</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-04 </td><td style="text-align: left;"> Grain number</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,12]&#8288;</code> </td><td style="text-align: left;"> <code>GRAINNUMBER</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Grain number</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,13]&#8288;</code> </td><td style="text-align: left;"> <code>CURVENO</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Curve number</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,14]&#8288;</code> </td><td style="text-align: left;"> <code>XCOORD</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> X position of a single grain</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,15]&#8288;</code> </td><td style="text-align: left;"> <code>YCOORD</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Y position of a single grain</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,16]&#8288;</code> </td><td style="text-align: left;"> <code>SAMPLE</code> </td><td style="text-align: left;"> <code>factor</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Sample name</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,17]&#8288;</code> </td><td style="text-align: left;"> <code>COMMENT</code> </td><td style="text-align: left;"> <code>factor</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Comment name</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,18]&#8288;</code> </td><td style="text-align: left;"> <code>SYSTEMID</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Ris√∏ system id</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,19]&#8288;</code> </td><td style="text-align: left;"> <code>FNAME</code> </td><td style="text-align: left;"> <code>factor</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> File name (<em>.bin/</em>.binx)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,20]&#8288;</code> </td><td style="text-align: left;"> <code>USER</code> </td><td style="text-align: left;"> <code>factor</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> User name</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,21]&#8288;</code> </td><td style="text-align: left;"> <code>TIME</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Data collection time (<code>hh-mm-ss</code>)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,22]&#8288;</code> </td><td style="text-align: left;"> <code>DATE</code> </td><td style="text-align: left;"> <code>factor</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Data collection date (<code>ddmmyy</code>)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,23]&#8288;</code> </td><td style="text-align: left;"> <code>DTYPE</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Data type</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,24]&#8288;</code> </td><td style="text-align: left;"> <code>BL_TIME</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Bleaching time</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,25]&#8288;</code> </td><td style="text-align: left;"> <code>BL_UNIT</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Bleaching unit (mJ, J, s, min, h)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,26]&#8288;</code> </td><td style="text-align: left;"> <code>NORM1</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Normalisation factor (1)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,27]&#8288;</code> </td><td style="text-align: left;"> <code>NORM2</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Normalisation factor (2)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,28]&#8288;</code> </td><td style="text-align: left;"> <code>NORM3</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Normalisation factor (3)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,29]&#8288;</code> </td><td style="text-align: left;"> <code>BG</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Background level</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,30]&#8288;</code> </td><td style="text-align: left;"> <code>SHIFT</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Number of channels to shift data</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,31]&#8288;</code> </td><td style="text-align: left;"> <code>TAG</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Tag, triggers <code>SEL</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,32]&#8288;</code> </td><td style="text-align: left;"> <code>LTYPE</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Luminescence type</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,33]&#8288;</code> </td><td style="text-align: left;"> <code>LIGHTSOURCE</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Light source</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,34]&#8288;</code> </td><td style="text-align: left;"> <code>LPOWER</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Optical stimulation power</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,35]&#8288;</code> </td><td style="text-align: left;"> <code>LIGHTPOWER</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Optical stimulation power</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,36]&#8288;</code> </td><td style="text-align: left;"> <code>LOW</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Low (temperature, time, wavelength)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,37]&#8288;</code> </td><td style="text-align: left;"> <code>HIGH</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> High (temperature, time, wavelength)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,38]&#8288;</code> </td><td style="text-align: left;"> <code>RATE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Rate (heating rate, scan rate)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,39]&#8288;</code> </td><td style="text-align: left;"> <code>TEMPERATURE</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Sample temperature</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,40]&#8288;</code> </td><td style="text-align: left;"> <code>MEASTEMP</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Measured temperature</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,41]&#8288;</code> </td><td style="text-align: left;"> <code>AN_TEMP</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Annealing temperature</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,42]&#8288;</code> </td><td style="text-align: left;"> <code>AN_TIME</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Annealing time</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,43]&#8288;</code> </td><td style="text-align: left;"> <code>TOLDELAY</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> TOL 'delay' channels</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,44]&#8288;</code> </td><td style="text-align: left;"> <code>TOLON</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> TOL 'on' channels</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,45]&#8288;</code> </td><td style="text-align: left;"> <code>TOLOFF</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> TOL 'off' channels</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,46]&#8288;</code> </td><td style="text-align: left;"> <code>IRR_TIME</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Irradiation time</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,47]&#8288;</code> </td><td style="text-align: left;"> <code>IRR_TYPE</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-08 </td><td style="text-align: left;"> Irradiation type (alpha, beta or gamma)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,48]&#8288;</code> </td><td style="text-align: left;"> <code>IRR_UNIT</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03-04 </td><td style="text-align: left;"> Irradiation unit (Gy, rad, s, min, h)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,49]&#8288;</code> </td><td style="text-align: left;"> <code>IRR_DOSERATE</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Irradiation dose rate (Gy/s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,50]&#8288;</code> </td><td style="text-align: left;"> <code>IRR_DOSERATEERR</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 06-08 </td><td style="text-align: left;"> Irradiation dose rate error (Gy/s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,51]&#8288;</code> </td><td style="text-align: left;"> <code>TIMESINCEIRR</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Time since irradiation (s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,52]&#8288;</code> </td><td style="text-align: left;"> <code>TIMETICK</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Time tick for pulsing (s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,53]&#8288;</code> </td><td style="text-align: left;"> <code>ONTIME</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> On-time for pulsing (in time ticks)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,54]&#8288;</code> </td><td style="text-align: left;"> <code>OFFTIME</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 03 </td><td style="text-align: left;"> Off-time for pulsed stimulation (in s) </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,55]&#8288;</code> </td><td style="text-align: left;"> <code>STIMPERIOD</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Stimulation period (on+off in time ticks)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,56]&#8288;</code> </td><td style="text-align: left;"> <code>GATE_ENABLED</code> </td><td style="text-align: left;"> <code>raw</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> PMT signal gating enabled</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,57]&#8288;</code> </td><td style="text-align: left;"> <code>ENABLE_FLAGS</code> </td><td style="text-align: left;"> <code>raw</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> PMT signal gating  enabled</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,58]&#8288;</code> </td><td style="text-align: left;"> <code>GATE_START</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Start gating (in time ticks)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,59]&#8288;</code> </td><td style="text-align: left;"> <code>GATE_STOP</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Stop gating (in time ticks), <code>'Gateend'</code> for version 04, here only GATE_STOP is used</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,60]&#8288;</code> </td><td style="text-align: left;"> <code>PTENABLED</code> </td><td style="text-align: left;"> <code>raw</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Photon time enabled</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,61]&#8288;</code> </td><td style="text-align: left;"> <code>DTENABLED</code> </td><td style="text-align: left;"> <code>raw</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> PMT dead time correction enabled</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,62]&#8288;</code> </td><td style="text-align: left;"> <code>DEADTIME</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> PMT dead time (s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,63]&#8288;</code> </td><td style="text-align: left;"> <code>MAXLPOWER</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> Stimulation power to 100 percent (mW/cm^2)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,64]&#8288;</code> </td><td style="text-align: left;"> <code>XRF_ACQTIME</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> XRF acquisition time (s)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,65]&#8288;</code> </td><td style="text-align: left;"> <code>XRF_HV</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> XRF X-ray high voltage (V)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,66]&#8288;</code> </td><td style="text-align: left;"> <code>XRF_CURR</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> XRF X-ray current (¬µA)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,67]&#8288;</code> </td><td style="text-align: left;"> <code>XRF_DEADTIMEF</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 05-08 </td><td style="text-align: left;"> XRF dead time fraction</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,68]&#8288;</code> </td><td style="text-align: left;"> <code>DETECTOR_ID</code> </td><td style="text-align: left;"> <code>raw</code> </td><td style="text-align: left;"> 07-08 </td><td style="text-align: left;"> Detector ID</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,69]&#8288;</code> </td><td style="text-align: left;"> <code>LOWERFILTER_ID</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 07-08 </td><td style="text-align: left;"> Lower filter ID in reader</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,70]&#8288;</code> </td><td style="text-align: left;"> <code>UPPERFILTER_ID</code> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> 07-08 </td><td style="text-align: left;"> Upper filter ID in reader</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,71]&#8288;</code> </td><td style="text-align: left;"> <code>ENOISEFACTOR</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 07-08 </td><td style="text-align: left;"> Excess noise filter, usage unknown </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,72]&#8288;</code> </td><td style="text-align: left;"> <code>MARKPOS_X1</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Coordinates marker position 1 </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,73]&#8288;</code> </td><td style="text-align: left;"> <code>MARKPOS_Y1</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Coordinates marker position 1 </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,74]&#8288;</code> </td><td style="text-align: left;"> <code>MARKPOS_X2</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Coordinates marker position 2 </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,75]&#8288;</code> </td><td style="text-align: left;"> <code>MARKPOS_Y2</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Coordinates marker position 2 </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,76]&#8288;</code> </td><td style="text-align: left;"> <code>MARKPOS_X3</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Coordinates marker position 3 </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,77]&#8288;</code> </td><td style="text-align: left;"> <code>MARKPOS_Y3</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> Coordinates marker position 3 </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,78]&#8288;</code> </td><td style="text-align: left;"> <code>EXTR_START</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> usage unknown </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,79]&#8288;</code> </td><td style="text-align: left;"> <code>EXTR_END</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> usage unknown</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,80]&#8288;</code> </td><td style="text-align: left;"> <code>SEQUENCE</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> 03-04 </td><td style="text-align: left;"> Sequence name
</td>
</tr>

</table>

<p>V = BIN-file version (RLum means that it does not depend on a specific BIN version)
</p>
<p>Note that the <code>Risoe.BINfileData</code> object combines all values from
different versions from the BIN-file, reserved bits are skipped, however,
the function <a href="#topic+write_R2BIN">write_R2BIN</a> reset arbitrary reserved bits. Invalid
values for a specific version are set to <code>NA</code>. Furthermore, the
internal R data types do not necessarily match the required data types for
the BIN-file data import! Data types are converted during data import.<br />
</p>
<p><strong>LTYPE</strong> values
</p>

<table>
<tr>
 <td style="text-align: right;">
VALUE </td><td style="text-align: left;"> TYPE </td><td style="text-align: left;"> DESCRIPTION </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[0]&#8288;</code> </td><td style="text-align: left;"> <code>TL</code> </td><td style="text-align: left;">: Thermoluminescence </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[1]&#8288;</code> </td><td style="text-align: left;"> <code>OSL</code> </td><td style="text-align: left;">: Optically stimulated luminescence </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[2]&#8288;</code> </td><td style="text-align: left;"> <code>IRSL</code> </td><td style="text-align: left;">: Infrared stimulated luminescence </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[3]&#8288;</code> </td><td style="text-align: left;"> <code>M-IR</code> </td><td style="text-align: left;">: Infrared monochromator scan</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[4]&#8288;</code> </td><td style="text-align: left;"> <code>M-VIS</code> </td><td style="text-align: left;">: Visible monochromator scan</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[5]&#8288;</code> </td><td style="text-align: left;"> <code>TOL</code> </td><td style="text-align: left;">: Thermo-optical luminescence </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[6]&#8288;</code> </td><td style="text-align: left;"> <code>TRPOSL</code> </td><td style="text-align: left;">: Time Resolved Pulsed OSL</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[7]&#8288;</code> </td><td style="text-align: left;"> <code>RIR</code> </td><td style="text-align: left;">: Ramped IRSL</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[8]&#8288;</code> </td><td style="text-align: left;"> <code>RBR</code> </td><td style="text-align: left;">: Ramped (Blue) LEDs</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[9]&#8288;</code> </td><td style="text-align: left;"> <code>USER</code> </td><td style="text-align: left;">: User defined</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[10]&#8288;</code> </td><td style="text-align: left;"> <code>POSL</code> </td><td style="text-align: left;">: Pulsed OSL </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[11]&#8288;</code> </td><td style="text-align: left;"> <code>SGOSL</code> </td><td style="text-align: left;">: Single Grain OSL</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[12]&#8288;</code> </td><td style="text-align: left;"> <code>RL</code> </td><td style="text-align: left;">: Radio Luminescence </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[13]&#8288;</code> </td><td style="text-align: left;"> <code>XRF</code> </td><td style="text-align: left;">: X-ray Fluorescence
</td>
</tr>

</table>

<p><strong>DTYPE</strong> values
</p>

<table>
<tr>
 <td style="text-align: right;">
VALUE </td><td style="text-align: left;"> DESCRIPTION </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[0]&#8288;</code> </td><td style="text-align: left;"> Natural </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[1]&#8288;</code> </td><td style="text-align: left;"> N+dose </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[2]&#8288;</code> </td><td style="text-align: left;"> Bleach </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[3]&#8288;</code> </td><td style="text-align: left;"> Bleach+dose </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[4]&#8288;</code> </td><td style="text-align: left;"> Natural (Bleach) </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[5]&#8288;</code> </td><td style="text-align: left;"> N+dose (Bleach) </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[6]&#8288;</code> </td><td style="text-align: left;"> Dose </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[7]&#8288;</code> </td><td style="text-align: left;"> Background
</td>
</tr>

</table>

<p><strong>LIGHTSOURCE</strong> values
</p>

<table>
<tr>
 <td style="text-align: right;">
VALUE </td><td style="text-align: left;"> DESCRIPTION </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[0]&#8288;</code> </td><td style="text-align: left;"> None </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[1]&#8288;</code> </td><td style="text-align: left;"> Lamp </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[2]&#8288;</code> </td><td style="text-align: left;"> IR diodes/IR Laser </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[3]&#8288;</code> </td><td style="text-align: left;"> Calibration LED </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[4]&#8288;</code> </td><td style="text-align: left;"> Blue Diodes </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[5]&#8288;</code> </td><td style="text-align: left;"> White light </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[6]&#8288;</code> </td><td style="text-align: left;"> Green laser (single grain) </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[7]&#8288;</code> </td><td style="text-align: left;"> IR laser (single grain) </td>
</tr>

</table>

<p><strong>Internal DATA - object structure</strong>
</p>
<p>With version 8 of the BIN/BINX file format, slot <code style="white-space: pre;">&#8288;@DATA&#8288;</code> (byte array <code>DPOINTS</code>) can
contain two different values:
</p>

<ol>
<li> <p><code>DPOINTS</code> (standard for <code>RECTYPE</code> := (0,1)): is a vector with the length defined
through <code>NPOINTS</code>. This is the standard for xy-curves since version 03.
</p>
</li>
<li> <p><code>DPOINTS</code> (<code>RECTYPE</code> := 128) is contains no count values but information about
the definition of the regions of interest (ROI). Each definition is 504 bytes long.
The number of definitions is defined by <code>NPOINTS</code> in <code style="white-space: pre;">&#8288;@METADATA&#8288;</code>. The record
describes basically the geometric features of the regions of interest.
The representation in R is a nested <a href="base.html#topic+list">list</a>.
</p>
</li></ol>


<table>
<tr>
 <td style="text-align: right;">
<strong>#</strong> </td><td style="text-align: left;"> <strong>Name</strong> </td><td style="text-align: left;"> <strong>Data Type</strong> </td><td style="text-align: left;"> <strong>V</strong> </td><td style="text-align: left;"> <strong>Description</strong> </td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,1]&#8288;</code> </td><td style="text-align: left;"> <code>NOFPOINTS</code>  </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> number of points in the definition (e.g., if the ROI is a rectangle: 4)</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,2]&#8288;</code> </td><td style="text-align: left;"> <code>USEDFOR</code>  </td><td style="text-align: left;"> <code>logical</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> samples for which the ROI is used for; a maximum of 48 samples are allowed.</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,3]&#8288;</code> </td><td style="text-align: left;"> <code>SHOWNFOR</code>  </td><td style="text-align: left;"> <code>logical</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> samples for which the ROI is shown for; a maximum of 48 samples are allowed.</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,4]&#8288;</code> </td><td style="text-align: left;"> <code>COLOR</code>  </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> The colour values of the ROI.</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,5]&#8288;</code> </td><td style="text-align: left;"> <code>X</code>  </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> The x coordinates used to draw the ROI geometry (up to 50 points are allowed).</td>
</tr>
<tr>
 <td style="text-align: right;">
<code style="white-space: pre;">&#8288;[,6]&#8288;</code> </td><td style="text-align: left;"> <code>Y</code>  </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> 08 </td><td style="text-align: left;"> The y coordinates used to draw the ROI geometry (up to 50 points are allowed).</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>(information on the BIN/BINX file format are kindly provided by Ris√∏, DTU Nutech)
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
based on information provided by Torben Lapp and Karsten Bracht Nielsen (Ris√∏ DTU, Denmark)
, RLum Developer Team</p>


<h3>References</h3>

<p>Ris√∏ DTU, 2013. The Sequence Editor User Manual - Feb 2013 and Ris√∏ DTU, 2016.
</p>
<p>The Sequence Editor User Manual - February 2016
</p>
<p><a href="https://www.fysik.dtu.dk">https://www.fysik.dtu.dk</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+plot_Risoe.BINfileData">plot_Risoe.BINfileData</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+write_R2BIN">write_R2BIN</a>,
<a href="#topic+merge_Risoe.BINfileData">merge_Risoe.BINfileData</a>, <a href="#topic+Risoe.BINfileData2RLum.Analysis">Risoe.BINfileData2RLum.Analysis</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("Risoe.BINfileData")

</code></pre>

<hr>
<h2 id='Risoe.BINfileData2RLum.Analysis'>Convert Risoe.BINfileData object to an RLum.Analysis object</h2><span id='topic+Risoe.BINfileData2RLum.Analysis'></span>

<h3>Description</h3>

<p>Converts values from one specific position of a Risoe.BINfileData S4-class
object to an RLum.Analysis object.
</p>
<p>The <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object requires a set of curves for
specific further protocol analyses. However, the <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
usually contains a set of curves for different aliquots and different
protocol types that may be mixed up. Therefore, a conversion is needed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Risoe.BINfileData2RLum.Analysis(
  object,
  pos = NULL,
  grain = NULL,
  run = NULL,
  set = NULL,
  ltype = NULL,
  dtype = NULL,
  protocol = "unknown",
  keep.empty = TRUE,
  txtProgressBar = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
<code>Risoe.BINfileData</code> object</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_pos">pos</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): position number of the <code>Risoe.BINfileData</code>
object for which the curves are stored in the <code>RLum.Analysis</code> object.
If <code>length(position)&gt;1</code> a list of <code>RLum.Analysis</code> objects is returned.
If nothing is provided every position will be converted.
If the position is not valid <code>NULL</code> is returned.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_grain">grain</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a>, <a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
grain number from the measurement to limit the converted data set
(e.g., <code>grain = c(1:48)</code>). Please be aware that this option may lead to
unwanted effects, as the output is strictly limited to the chosen grain
number for all position numbers</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_run">run</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a>, <a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
run number from the measurement to limit the converted data set
(e.g., <code>run = c(1:48)</code>).</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_set">set</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a>, <a href="base.html#topic+numeric">numeric</a> (<em>optional</em>):
set number from the measurement to limit the converted data set
(e.g., <code>set = c(1:48)</code>).</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_ltype">ltype</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a>, <a href="base.html#topic+character">character</a> (<em>optional</em>):
curve type to limit the converted data. Commonly allowed values are:
<code>IRSL</code>, <code>OSL</code>, <code>TL</code>, <code>RIR</code>, <code>RBR</code> and <code>USER</code>
(see also <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>)</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_dtype">dtype</code></td>
<td>
<p><a href="base.html#topic+vector">vector</a>, <a href="base.html#topic+character">character</a> (<em>optional</em>):
data type to limit the converted data. Commonly allowed values are
listed in <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a></p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_protocol">protocol</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
sets protocol type for analysis object. Value may be used by subsequent
analysis functions.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_keep.empty">keep.empty</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
If <code>TRUE</code> (default) an <code>RLum.Analysis</code> object is returned even if it does
not contain any records. Set to <code>FALSE</code> to discard all empty objects.</p>
</td></tr>
<tr><td><code id="Risoe.BINfileData2RLum.Analysis_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.
</p>


<h3>Function version</h3>

<p>0.4.3
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. Risoe.BINfileData2RLum.Analysis(): Convert Risoe.BINfileData object to an RLum.Analysis object. Function version 0.4.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The <code>protocol</code> argument of the <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
object is set to 'unknown' if not stated otherwise.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##load data
data(ExampleData.BINfileData, envir = environment())

##convert values for position 1
Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos = 1)

</code></pre>

<hr>
<h2 id='RLum-class'>Class <code>"RLum"</code></h2><span id='topic+RLum-class'></span><span id='topic+replicate_RLum+2CRLum-method'></span>

<h3>Description</h3>

<p>Abstract class for data in the package Luminescence
Subclasses are:
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'RLum'
replicate_RLum(object, times = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum-class_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
an object of class <a href="#topic+RLum-class">RLum</a></p>
</td></tr>
<tr><td><code id="RLum-class_+3A_times">times</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
number for times each element is repeated element</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>RLum-class</strong><br />
|<br />
|&mdash;-<a href="#topic+RLum.Data-class">RLum.Data</a><br />
|&mdash;-|&ndash; <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a><br />
|&mdash;-|&ndash; <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a><br />
|&mdash;-|&ndash; <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a><br />
|&mdash;-<a href="#topic+RLum.Analysis-class">RLum.Analysis</a><br />
|&mdash;-<a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>replicate_RLum(RLum)</code>: Replication method RLum-objects
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>originator</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> containing the name of the producing
function for the object. Set automatically by using the function <a href="#topic+set_RLum">set_RLum</a>.</p>
</dd>
<dt><code>info</code></dt><dd><p>Object of class <a href="base.html#topic+list">list</a> for additional information on the object itself</p>
</dd>
<dt><code>.uid</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> for a unique object identifier. This id is
usually calculated using the internal function <code>create_UID()</code> if the function <a href="#topic+set_RLum">set_RLum</a>
is called.</p>
</dd>
<dt><code>.pid</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> for a parent id. This allows nesting RLum-objects
at will. The parent id can be the uid of another object.</p>
</dd>
</dl>


<h3>Objects from the Class</h3>

<p>A virtual Class: No objects can be created from it.
</p>


<h3>Class version</h3>

<p>0.4.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum-class(): Class 'RLum'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><code>RLum</code> is a virtual class.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>, <a href="#topic+methods_RLum">methods_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum")

</code></pre>

<hr>
<h2 id='RLum.Analysis-class'>Class <code>"RLum.Analysis"</code></h2><span id='topic+RLum.Analysis-class'></span><span id='topic+show+2CRLum.Analysis-method'></span><span id='topic+set_RLum+2CRLum.Analysis-method'></span><span id='topic+get_RLum+2CRLum.Analysis-method'></span><span id='topic+structure_RLum+2CRLum.Analysis-method'></span><span id='topic+length_RLum+2CRLum.Analysis-method'></span><span id='topic+names_RLum+2CRLum.Analysis-method'></span><span id='topic+add_metadata+3C-+2CRLum.Analysis-method'></span><span id='topic+rename_metadata+3C-+2CRLum.Analysis-method'></span><span id='topic+replace_metadata+3C-+2CRLum.Analysis-method'></span><span id='topic+smooth_RLum+2CRLum.Analysis-method'></span><span id='topic+sort_RLum+2CRLum.Analysis-method'></span><span id='topic+melt_RLum+2CRLum.Analysis-method'></span><span id='topic+view+2CRLum.Analysis-method'></span>

<h3>Description</h3>

<p>Object class to represent analysis data for protocol analysis, i.e. all curves,
spectra etc. from one measurements. Objects from this class are produced,
by e.g. <a href="#topic+read_XSYG2R">read_XSYG2R</a>, <a href="#topic+read_Daybreak2R">read_Daybreak2R</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'RLum.Analysis'
show(object)

## S4 method for signature 'RLum.Analysis'
set_RLum(
  class,
  originator,
  .uid,
  .pid,
  protocol = NA_character_,
  records = list(),
  info = list()
)

## S4 method for signature 'RLum.Analysis'
get_RLum(
  object,
  record.id = NULL,
  recordType = NULL,
  curveType = NULL,
  RLum.type = NULL,
  protocol = "UNKNOWN",
  get.index = NULL,
  drop = TRUE,
  recursive = TRUE,
  info.object = NULL,
  subset = NULL,
  env = parent.frame(2)
)

## S4 method for signature 'RLum.Analysis'
structure_RLum(object, fullExtent = FALSE)

## S4 method for signature 'RLum.Analysis'
length_RLum(object)

## S4 method for signature 'RLum.Analysis'
names_RLum(object)

## S4 replacement method for signature 'RLum.Analysis'
add_metadata(object, info_element) &lt;- value

## S4 replacement method for signature 'RLum.Analysis'
rename_metadata(object, info_element) &lt;- value

## S4 replacement method for signature 'RLum.Analysis'
replace_metadata(object, info_element, subset = NULL) &lt;- value

## S4 method for signature 'RLum.Analysis'
smooth_RLum(object, ...)

## S4 method for signature 'RLum.Analysis'
sort_RLum(object, slot = NULL, info_element = NULL, decreasing = FALSE, ...)

## S4 method for signature 'RLum.Analysis'
melt_RLum(object)

## S4 method for signature 'RLum.Analysis'
view(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum.Analysis-class_+3A_object">object</code></td>
<td>
<p>(<strong>required</strong>):
an object of class <a href="#topic+RLum.Analysis-class">RLum.Analysis</a></p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_class">class</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the <code>RLum</code> class to be created</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_originator">originator</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+character">character</a> (<em>automatic</em>):
contains the name of the calling function (the function that produces this object);
can be set manually.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_.uid">.uid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+character">character</a> (<em>automatic</em>):
sets an unique ID for this object using the internal C++ function <code>create_UID</code>.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_.pid">.pid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide a parent id for nesting at will.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_protocol">protocol</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+character">character</a> (<em>optional</em>):
sets protocol type for analysis object. Value may be used by subsequent analysis functions.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_records">records</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+list">list</a> (<strong>required</strong>):
list of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_info">info</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+list">list</a> (<em>optional</em>):
a list containing additional info data for the object
</p>
<p><strong><code>set_RLum</code></strong>:
</p>
<p>Returns an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_record.id">record.id</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+numeric">numeric</a> or <a href="base.html#topic+logical">logical</a> (<em>optional</em>):
IDs of specific records. If of type <code>logical</code> the entire id range is assumed
and <code>TRUE</code> and <code>FALSE</code> indicates the selection.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_recordtype">recordType</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+character">character</a> (<em>optional</em>):
record type (e.g., &quot;OSL&quot;). Can be also a vector, for multiple matching,
e.g., <code>recordType = c("OSL", "IRSL")</code></p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_curvetype">curveType</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+character">character</a> (<em>optional</em>):
curve type (e.g. &quot;predefined&quot; or &quot;measured&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_rlum.type">RLum.type</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+character">character</a> (<em>optional</em>):
RLum object type. Defaults to &quot;RLum.Data.Curve&quot; and &quot;RLum.Data.Spectrum&quot;.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_get.index">get.index</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+logical">logical</a> (<em>optional</em>):
return a numeric vector with the index of each element in the RLum.Analysis object.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_drop">drop</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+logical">logical</a> (<em>with default</em>):
coerce to the next possible layer (which are <code>RLum.Data</code>-objects),
<code>drop = FALSE</code> keeps the original <code>RLum.Analysis</code></p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_recursive">recursive</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if <code>TRUE</code> (default) when the result of the <code>get_RLum()</code> request is a single
object, the object itself will be returned directly, rather than being
wrapped in a list. Mostly this makes things easier, but this might be
undesired if this method is used within a loop.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_info.object">info.object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+character">character</a> (<em>optional</em>):
name of the wanted info element</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_subset">subset</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+expression">expression</a> (<em>optional</em>):
logical expression indicating elements or rows to keep: missing values are
taken as false. This argument takes precedence over all other arguments,
meaning they are not considered when subsetting the object.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_env">env</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>: <a href="base.html#topic+environment">environment</a> (<em>with default</em>):
An environment passed to <a href="base.html#topic+eval">eval</a> as the enclosure. This argument is only
relevant when subsetting the object and should not be used manually.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_fullextent">fullExtent</code></td>
<td>
<p><a href="#topic+structure_RLum">structure_RLum</a>; <a href="base.html#topic+logical">logical</a> (<em>with default</em>):
extends the returned <code>data.frame</code> to its full extent, i.e. all info elements
are part of the return as well. The default value is <code>FALSE</code> as the data
frame might become rather big.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_info_element">info_element</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): name of the <code>info</code> field
to use in sorting.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_value">value</code></td>
<td>
<p>(<strong>required</strong>) The value assigned to the selected elements
of the metadata field. If <code>NULL</code> the elements named in <code>info_element</code>
will be removed.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_...">...</code></td>
<td>
<p>other arguments that might be passed</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_slot">slot</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): slot name to use in sorting.</p>
</td></tr>
<tr><td><code id="RLum.Analysis-class_+3A_decreasing">decreasing</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): whether the sort order should
be decreasing (<code>FALSE</code> by default).</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>get_RLum</code></strong>:
</p>
<p>Returns:
</p>

<ol>
<li> <p><a href="base.html#topic+list">list</a> of <a href="#topic+RLum.Data-class">RLum.Data</a> objects or
</p>
</li>
<li><p> Single <a href="#topic+RLum.Data-class">RLum.Data</a> object, if only one object is contained and <code>recursive = FALSE</code> or
</p>
</li>
<li> <p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects for <code>drop = FALSE</code>
</p>
</li></ol>

<p><strong><code>structure_RLum</code></strong>:
</p>
<p>Returns <a href="methods.html#topic+data.frame-class">data.frame</a> showing the structure.
</p>
<p><strong><code>length_RLum</code></strong>
</p>
<p>Returns the number records in this object.
</p>
<p><strong><code>names_RLum</code></strong>
</p>
<p>Returns the names of the record types (<code>recordType</code>) in this object.
</p>
<p><strong><code>smooth_RLum</code></strong>
</p>
<p>Same object as input, after smoothing
</p>
<p><strong><code>sort_RLum</code></strong>
</p>
<p>Same object as input, but sorted according to the specified parameters.
</p>
<p><strong><code>melt_RLum</code></strong>
</p>
<p>Flat <a href="base.html#topic+data.frame">data.frame</a> with <code>X</code>, <code>Y</code>, <code>TYPE</code>, <code>UID</code>
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show(RLum.Analysis)</code>: Show structure of <code>RLum.Analysis</code> object
</p>
</li>
<li> <p><code>set_RLum(RLum.Analysis)</code>: Construction method for <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects.
</p>
</li>
<li> <p><code>get_RLum(RLum.Analysis)</code>: Accessor method for RLum.Analysis object.
</p>
<p>The slots record.id, <code style="white-space: pre;">&#8288;@recordType&#8288;</code>, <code style="white-space: pre;">&#8288;@curveType&#8288;</code> and <code style="white-space: pre;">&#8288;@RLum.type&#8288;</code> are optional to allow for records
limited by their id (list index number), their record type (e.g. <code>recordType = "OSL"</code>)
or object type.
</p>
<p>Example: curve type (e.g. <code>curveType = "predefined"</code> or <code>curveType ="measured"</code>)
</p>
<p>The selection of a specific RLum.type object superimposes the default selection.
Currently supported objects are: RLum.Data.Curve and RLum.Data.Spectrum
</p>
</li>
<li> <p><code>structure_RLum(RLum.Analysis)</code>: Method to show the structure of an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.
</p>
</li>
<li> <p><code>length_RLum(RLum.Analysis)</code>: Returns the length of the object, i.e., number of stored records.
</p>
</li>
<li> <p><code>names_RLum(RLum.Analysis)</code>: Returns the names of the <a href="#topic+RLum.Data-class">RLum.Data</a> objects (same as shown with the
<code>show</code> method)
</p>
</li>
<li> <p><code>add_metadata(RLum.Analysis) &lt;- value</code>: Adds metadata to <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
</p>
</li>
<li> <p><code>rename_metadata(RLum.Analysis) &lt;- value</code>: Renames a metadata entry of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
</p>
</li>
<li> <p><code>replace_metadata(RLum.Analysis) &lt;- value</code>: Replaces or removes metadata of <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
</p>
</li>
<li> <p><code>smooth_RLum(RLum.Analysis)</code>: Smoothing of <code>RLum.Data</code> objects contained in this <code>RLum.Analysis</code> object
using the internal function <code>.smoothing</code>.
</p>
</li>
<li> <p><code>sort_RLum(RLum.Analysis)</code>: Sorting of <code>RLum.Data</code> objects contained in this <code>RLum.Analysis</code> object.
At least one of <code>slot</code> and <code>info_element</code> must be provided. If both are
given, ordering by <code>slot</code> always supersedes ordering by <code>info_element</code>.
</p>
</li>
<li> <p><code>melt_RLum(RLum.Analysis)</code>: Melts <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects into a flat data.frame to be used
in combination with other packages such as <code>ggplot2</code>.
</p>
</li>
<li> <p><code>view(RLum.Analysis)</code>: View method for <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>protocol</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> describing the applied measurement protocol</p>
</dd>
<dt><code>records</code></dt><dd><p>Object of class <a href="base.html#topic+list">list</a> containing objects of class <a href="#topic+RLum.Data-class">RLum.Data</a></p>
</dd>
</dl>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>set_RLum("RLum.Analysis", ...)</code>.
</p>


<h3>Class version</h3>

<p>0.4.18
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum.Analysis-class(): Class 'RLum.Analysis'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The method <a href="#topic+structure_RLum">structure_RLum</a> is currently just available for objects
containing <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData2RLum.Analysis">Risoe.BINfileData2RLum.Analysis</a>,
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+RLum-class">RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum.Analysis")

##set empty object
set_RLum(class = "RLum.Analysis")

###use example data
##load data
data(ExampleData.RLum.Analysis, envir = environment())

##show curves in object
get_RLum(IRSAR.RF.Data)

##show only the first object, but by keeping the object
get_RLum(IRSAR.RF.Data, record.id = 1, drop = FALSE)

</code></pre>

<hr>
<h2 id='RLum.Data-class'>Class <code>"RLum.Data"</code></h2><span id='topic+RLum.Data-class'></span><span id='topic+add_metadata+3C-+2CRLum.Data-method'></span><span id='topic+rename_metadata+3C-+2CRLum.Data-method'></span><span id='topic+replace_metadata+3C-+2CRLum.Data-method'></span><span id='topic+view+2CRLum.Data-method'></span>

<h3>Description</h3>

<p>Generalized virtual data class for luminescence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 replacement method for signature 'RLum.Data'
add_metadata(object, info_element) &lt;- value

## S4 replacement method for signature 'RLum.Data'
rename_metadata(object, info_element) &lt;- value

## S4 replacement method for signature 'RLum.Data'
replace_metadata(object, info_element, subset = NULL, verbose = TRUE) &lt;- value

## S4 method for signature 'RLum.Data'
view(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum.Data-class_+3A_object">object</code></td>
<td>
<p>(<strong>required</strong>) an object of class <a href="#topic+RLum.Data-class">RLum.Data</a></p>
</td></tr>
<tr><td><code id="RLum.Data-class_+3A_info_element">info_element</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>) name of the metadata field
to replace or remove</p>
</td></tr>
<tr><td><code id="RLum.Data-class_+3A_value">value</code></td>
<td>
<p>(<strong>required</strong>) The value assigned to the selected elements
of the metadata field. If <code>NULL</code> the elements named in <code>info_element</code>
will be removed.</p>
</td></tr>
<tr><td><code id="RLum.Data-class_+3A_subset">subset</code></td>
<td>
<p><a href="base.html#topic+expression">expression</a> (<em>optional</em>) logical expression to limit the
substitution only to the selected subset of elements</p>
</td></tr>
<tr><td><code id="RLum.Data-class_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>) enable/disable output to the
terminal.</p>
</td></tr>
<tr><td><code id="RLum.Data-class_+3A_...">...</code></td>
<td>
<p>other arguments that might be passed</p>
</td></tr>
</table>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>add_metadata(RLum.Data) &lt;- value</code>: Add metadata entries to <a href="#topic+RLum.Data-class">RLum.Data</a> objects
</p>
</li>
<li> <p><code>rename_metadata(RLum.Data) &lt;- value</code>: Rename a metadata entry of <a href="#topic+RLum.Data-class">RLum.Data</a> objects
</p>
</li>
<li> <p><code>replace_metadata(RLum.Data) &lt;- value</code>: Replaces or removes metadata of <a href="#topic+RLum.Data-class">RLum.Data</a> objects
</p>
</li>
<li> <p><code>view(RLum.Data)</code>: View method for <a href="#topic+RLum.Data-class">RLum.Data</a> objects
</p>
</li></ul>


<h3>Objects from the Class</h3>

<p>A virtual class: no objects can be created from it.
</p>


<h3>Class version</h3>

<p>0.2.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum.Data-class(): Class 'RLum.Data'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum-class">RLum</a>, <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum.Data")

</code></pre>

<hr>
<h2 id='RLum.Data.Curve-class'>Class <code>"RLum.Data.Curve"</code></h2><span id='topic+RLum.Data.Curve-class'></span><span id='topic+show+2CRLum.Data.Curve-method'></span><span id='topic+set_RLum+2CRLum.Data.Curve-method'></span><span id='topic+get_RLum+2CRLum.Data.Curve-method'></span><span id='topic+length_RLum+2CRLum.Data.Curve-method'></span><span id='topic+names_RLum+2CRLum.Data.Curve-method'></span><span id='topic+bin_RLum.Data+2CRLum.Data.Curve-method'></span><span id='topic+smooth_RLum+2CRLum.Data.Curve-method'></span><span id='topic+melt_RLum+2CRLum.Data.Curve-method'></span>

<h3>Description</h3>

<p>Class for representing luminescence curve data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'RLum.Data.Curve'
show(object)

## S4 method for signature 'RLum.Data.Curve'
set_RLum(
  class,
  originator,
  .uid,
  .pid,
  recordType = NA_character_,
  curveType = NA_character_,
  data = matrix(0, ncol = 2),
  info = list()
)

## S4 method for signature 'RLum.Data.Curve'
get_RLum(object, info.object = NULL)

## S4 method for signature 'RLum.Data.Curve'
length_RLum(object)

## S4 method for signature 'RLum.Data.Curve'
names_RLum(object)

## S4 method for signature 'RLum.Data.Curve'
bin_RLum.Data(object, bin_size = 2)

## S4 method for signature 'RLum.Data.Curve'
smooth_RLum(object, k = NULL, fill = NA, align = "right", method = "mean")

## S4 method for signature 'RLum.Data.Curve'
melt_RLum(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum.Data.Curve-class_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>, <code><a href="#topic+length_RLum">length_RLum</a></code>, <code><a href="#topic+names_RLum">names_RLum</a></code> (<strong>required</strong>):
an object of class <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a></p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_class">class</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the <code>RLum</code> class to create</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_originator">originator</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
contains the name of the calling function (the function that produces this object);
can be set manually.</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_.uid">.uid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
sets an unique ID for this object using the internal C++ function <code>create_UID</code>.</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_.pid">.pid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide a parent id for nesting at will.</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_recordtype">recordType</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>optional</em>):
record type (e.g., &quot;OSL&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_curvetype">curveType</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>optional</em>):
curve type (e.g., &quot;predefined&quot; or &quot;measured&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_data">data</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+matrix">matrix</a> (<strong>required</strong>):
raw curve data. If <code>data</code> itself is a <code>RLum.Data.Curve</code>-object this can be
used to re-construct the object (see details), i.e. modified parameters except
<code>.uid</code>, <code>.pid</code> and <code>originator</code>. The rest will be subject to copy and paste unless provided.</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_info">info</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+list">list</a> (<em>optional</em>):
info elements</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_info.object">info.object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code> <a href="base.html#topic+character">character</a> (<em>optional</em>):
name of the wanted info element</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_bin_size">bin_size</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
set number of channels used for each bin, e.g. <code>bin_size = 2</code> means that
two channels are binned.</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_k">k</code></td>
<td>
<p><code><a href="#topic+smooth_RLum">smooth_RLum</a></code>; <a href="base.html#topic+integer">integer</a> (<em>with default</em>):
window for the rolling mean; must be odd for <code>rollmedian</code>.
If nothing is set k is set automatically</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_fill">fill</code></td>
<td>
<p><code><a href="#topic+smooth_RLum">smooth_RLum</a></code>; <a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
a vector defining the left and the right hand data</p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_align">align</code></td>
<td>
<p><code><a href="#topic+smooth_RLum">smooth_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>with default</em>):
specifying whether the index of the result should be left- or right-aligned
or centred (default) compared to the rolling window of observations, allowed
<code>"right"</code>, <code>"center"</code> and <code>"left"</code></p>
</td></tr>
<tr><td><code id="RLum.Data.Curve-class_+3A_method">method</code></td>
<td>
<p><code><a href="#topic+smooth_RLum">smooth_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>with default</em>):
defines which method should be applied for the smoothing: <code>"mean"</code> or <code>"median"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>set_RLum</code></strong>
</p>
<p>Returns an <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> object.
</p>
<p><strong><code>get_RLum</code></strong>
</p>

<ol>
<li><p> A <a href="base.html#topic+matrix">matrix</a> with the curve values or
</p>
</li>
<li><p> only the info object if <code>info.object</code> was set.
</p>
</li></ol>

<p><strong><code>length_RLum</code></strong>
</p>
<p>Number of channels in the curve (row number of the matrix)
</p>
<p><strong><code>names_RLum</code></strong>
</p>
<p>Names of the info elements (slot <code>info</code>)
</p>
<p><strong><code>bin_RLum.Data</code></strong>
</p>
<p>Same object as input, after applying the binning.
</p>
<p><strong><code>smooth_RLum</code></strong>
</p>
<p>Same object as input, after smoothing
</p>
<p><strong><code>melt_RLum</code></strong>
</p>
<p>Flat <a href="base.html#topic+data.frame">data.frame</a> with <code>X</code>, <code>Y</code>, <code>TYPE</code>, <code>UID</code>
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show(RLum.Data.Curve)</code>: Show structure of <code>RLum.Data.Curve</code> object
</p>
</li>
<li> <p><code>set_RLum(RLum.Data.Curve)</code>: Construction method for RLum.Data.Curve object. The slot info is optional
and predefined as empty list by default.
</p>
</li>
<li> <p><code>get_RLum(RLum.Data.Curve)</code>: Accessor method for RLum.Data.Curve object. The argument info.object is
optional to directly access the info elements. If no info element name is
provided, the raw curve data (matrix) will be returned.
</p>
</li>
<li> <p><code>length_RLum(RLum.Data.Curve)</code>: Returns the length of the curve object, which is the maximum of the
value time/temperature of the curve (corresponding to the stimulation length)
</p>
</li>
<li> <p><code>names_RLum(RLum.Data.Curve)</code>: Returns the names info elements coming along with this curve object
</p>
</li>
<li> <p><code>bin_RLum.Data(RLum.Data.Curve)</code>: Allows binning of specific objects
</p>
</li>
<li> <p><code>smooth_RLum(RLum.Data.Curve)</code>: Smoothing of RLum.Data.Curve objects using a rolling mean or median.
In particular the internal function <code>.smoothing</code> is used.
</p>
</li>
<li> <p><code>melt_RLum(RLum.Data.Curve)</code>: Melts <a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a> objects into a flat data.frame to be used
in combination with other packages such as <code>ggplot2</code>.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>recordType</code></dt><dd><p>Object of class &quot;character&quot; containing the type of the curve (e.g. &quot;TL&quot; or &quot;OSL&quot;)</p>
</dd>
<dt><code>curveType</code></dt><dd><p>Object of class &quot;character&quot; containing curve type, allowed values are
&quot;measured&quot; or &quot;predefined&quot;</p>
</dd>
<dt><code>data</code></dt><dd><p>Object of class <a href="base.html#topic+matrix">matrix</a> containing curve x and y data.
'data' can also be of type <code>RLum.Data.Curve</code> to change object values without
de-constructing the object. For example:
</p>
<div class="sourceCode"><pre>set_RLum(class = 'RLum.Data.Curve',
         data = Your.RLum.Data.Curve,
         recordType = 'never seen before')
</pre></div>
<p>would just change the <code>recordType</code>. Missing arguments  the value is taken
from the input object in 'data' (which is already an <code>RLum.Data.Curve</code>
object in this example)</p>
</dd>
</dl>


<h3>Create objects from this Class</h3>

<p>Objects can be created by calls of the form
<code>set_RLum(class = "RLum.Data.Curve", ...)</code>.
</p>


<h3>Class version</h3>

<p>0.5.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum.Data.Curve-class(): Class 'RLum.Data.Curve'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The class should only contain data for a single curve. For additional
elements the slot <code>info</code> can be used (e.g. providing additional heating
ramp curve). Objects from the class <code>RLum.Data.Curve</code> are produced by other
functions (partly within <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects),
namely: <a href="#topic+Risoe.BINfileData2RLum.Analysis">Risoe.BINfileData2RLum.Analysis</a>, <a href="#topic+read_XSYG2R">read_XSYG2R</a>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum-class">RLum</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+plot_RLum">plot_RLum</a>, <a href="#topic+merge_RLum">merge_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum.Data.Curve")

##set empty curve object
set_RLum(class = "RLum.Data.Curve")

</code></pre>

<hr>
<h2 id='RLum.Data.Image-class'>Class <code>"RLum.Data.Image"</code></h2><span id='topic+RLum.Data.Image-class'></span><span id='topic+show+2CRLum.Data.Image-method'></span><span id='topic+set_RLum+2CRLum.Data.Image-method'></span><span id='topic+get_RLum+2CRLum.Data.Image-method'></span><span id='topic+names_RLum+2CRLum.Data.Image-method'></span>

<h3>Description</h3>

<p>Class for representing luminescence image data (TL/OSL/RF). Such data are for example produced
by the function <a href="#topic+read_SPE2R">read_SPE2R</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'RLum.Data.Image'
show(object)

## S4 method for signature 'RLum.Data.Image'
set_RLum(
  class,
  originator,
  .uid,
  .pid,
  recordType = "Image",
  curveType = NA_character_,
  data = array(),
  info = list()
)

## S4 method for signature 'RLum.Data.Image'
get_RLum(object, info.object)

## S4 method for signature 'RLum.Data.Image'
names_RLum(object)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum.Data.Image-class_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>, <code><a href="#topic+names_RLum">names_RLum</a></code> (<strong>required</strong>):
an object of class <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a></p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_class">class</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a>: name of the <code>RLum</code> class to create</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_originator">originator</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
contains the name of the calling function (the function that produces this object);
can be set manually.</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_.uid">.uid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
sets an unique ID for this object using the internal C++ function <code>create_UID</code>.</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_.pid">.pid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide a parent id for nesting at will.</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_recordtype">recordType</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a>:
record type (e.g. &quot;OSL&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_curvetype">curveType</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a>:
curve type (e.g. &quot;predefined&quot; or &quot;measured&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_data">data</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+matrix">matrix</a>:
raw curve data. If data is of type <code>RLum.Data.Image</code> this can be used to
re-construct the object, i.e. modified parameters except <code>.uid</code> and <code>.pid</code>. The rest
will be subject to copy and paste unless provided.</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_info">info</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+list">list</a>:
info elements</p>
</td></tr>
<tr><td><code id="RLum.Data.Image-class_+3A_info.object">info.object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>; <a href="base.html#topic+character">character</a>:
name of the info object to returned</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>set_RLum</code></strong>
</p>
<p>Returns an object from class <code>RLum.Data.Image</code>
</p>
<p><strong><code>get_RLum</code></strong>
</p>

<ol>
<li><p> Returns the data object (<a href="base.html#topic+array">array</a>)
</p>
</li>
<li><p> only the info object if <code>info.object</code> was set.
</p>
</li></ol>

<p><strong><code>names_RLum</code></strong>
</p>
<p>Returns the names of the info elements
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show(RLum.Data.Image)</code>: Show structure of <code>RLum.Data.Image</code> object
</p>
</li>
<li> <p><code>set_RLum(RLum.Data.Image)</code>: Construction method for RLum.Data.Image object. The slot info is optional
and predefined as empty list by default.
</p>
</li>
<li> <p><code>get_RLum(RLum.Data.Image)</code>: Accessor method for <code>RLum.Data.Image</code> object. The argument <code>info.object</code> is
optional to directly access the info elements. If no info element name is
provided, the raw image data (<code>array</code>) will be returned.
</p>
</li>
<li> <p><code>names_RLum(RLum.Data.Image)</code>: Returns the names info elements coming along with this curve object
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>recordType</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> containing the type of the curve (e.g. &quot;OSL image&quot;, &quot;TL image&quot;)</p>
</dd>
<dt><code>curveType</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> containing curve type, allowed values
are measured or predefined</p>
</dd>
<dt><code>data</code></dt><dd><p>Object of class <a href="base.html#topic+array">array</a> containing image data.</p>
</dd>
<dt><code>info</code></dt><dd><p>Object of class <a href="base.html#topic+list">list</a> containing further meta information objects</p>
</dd>
</dl>


<h3>Objects from the class</h3>

<p>Objects can be created by calls of the form <code>set_RLum("RLum.Data.Image", ...)</code>.
</p>


<h3>Class version</h3>

<p>0.5.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum.Data.Image-class(): Class 'RLum.Data.Image'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The class should only contain data for a set of images. For additional
elements the slot <code>info</code> can be used.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum-class">RLum</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+plot_RLum">plot_RLum</a>, <a href="#topic+read_SPE2R">read_SPE2R</a>, <a href="#topic+read_TIFF2R">read_TIFF2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum.Data.Image")

##create empty RLum.Data.Image object
set_RLum(class = "RLum.Data.Image")

</code></pre>

<hr>
<h2 id='RLum.Data.Spectrum-class'>Class <code>"RLum.Data.Spectrum"</code></h2><span id='topic+RLum.Data.Spectrum-class'></span><span id='topic+show+2CRLum.Data.Spectrum-method'></span><span id='topic+set_RLum+2CRLum.Data.Spectrum-method'></span><span id='topic+get_RLum+2CRLum.Data.Spectrum-method'></span><span id='topic+names_RLum+2CRLum.Data.Spectrum-method'></span><span id='topic+bin_RLum.Data+2CRLum.Data.Spectrum-method'></span>

<h3>Description</h3>

<p>Class for representing luminescence spectra data (TL/OSL/RF).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'RLum.Data.Spectrum'
show(object)

## S4 method for signature 'RLum.Data.Spectrum'
set_RLum(
  class,
  originator,
  .uid,
  .pid,
  recordType = "Spectrum",
  curveType = NA_character_,
  data = matrix(),
  info = list()
)

## S4 method for signature 'RLum.Data.Spectrum'
get_RLum(object, info.object)

## S4 method for signature 'RLum.Data.Spectrum'
names_RLum(object)

## S4 method for signature 'RLum.Data.Spectrum'
bin_RLum.Data(object, bin_size.col = 1, bin_size.row = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum.Data.Spectrum-class_+3A_object">object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>, <code><a href="#topic+names_RLum">names_RLum</a></code> (<strong>required</strong>):
an object of class <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a></p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_class">class</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
name of the <code>RLum</code> class to create.</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_originator">originator</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>automatic</em>):
contains the name of the calling function (the function that produces this object);
can be set manually.</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_.uid">.uid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
sets an unique ID for this object using the internal C++ function <code>create_UID</code>.</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_.pid">.pid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide a parent id for nesting at will.</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_recordtype">recordType</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a>:
record type (e.g. &quot;OSL&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_curvetype">curveType</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a>:
curve type (e.g. &quot;predefined&quot; or &quot;measured&quot;)</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_data">data</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+matrix">matrix</a>:
raw curve data. If data is of type <code>RLum.Data.Spectrum</code>, this can be used
to re-construct the object. If the object is reconstructed, <code>.uid</code>, <code>.pid</code>
and <code>originator</code> are always taken from the input object</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_info">info</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code> <a href="base.html#topic+list">list</a>:
info elements</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_info.object">info.object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>optional</em>):
the name of the info object to be called. If no info element name
is provided, the raw curve data (matrix) will be returned</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_bin_size.col">bin_size.col</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
set number of channels used for each bin, e.g. <code>bin_size.col = 2</code> means that
two channels are binned. Note: The function does not check the input, very large values
mean a full column binning (a single sum)</p>
</td></tr>
<tr><td><code id="RLum.Data.Spectrum-class_+3A_bin_size.row">bin_size.row</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
set number of channels used for each bin, e.g. <code>bin_size.row = 2</code> means that
two channels are binned. Note: The function does not check the input, very large values
mean a full row binning (a single sum)</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code style="white-space: pre;">&#8288;[set_RLum]&#8288;</code></strong>
</p>
<p>An object from the class <code>RLum.Data.Spectrum</code>
</p>
<p><strong><code style="white-space: pre;">&#8288;[get_RLum]&#8288;</code></strong>
</p>

<ol>
<li><p> A <a href="base.html#topic+matrix">matrix</a> with the spectrum values or
</p>
</li>
<li><p> only the info object if <code>info.object</code> was set.
</p>
</li></ol>

<p><strong><code style="white-space: pre;">&#8288;[names_RLum]&#8288;</code></strong>
</p>
<p>The names of the info objects
</p>
<p><strong><code style="white-space: pre;">&#8288;[bin_RLum.Data]&#8288;</code></strong>
</p>
<p>Same object as input, after applying the binning.
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show(RLum.Data.Spectrum)</code>: Show structure of <code>RLum.Data.Spectrum</code> object
</p>
</li>
<li> <p><code>set_RLum(RLum.Data.Spectrum)</code>: Construction method for RLum.Data.Spectrum object. The <code>info</code> slot is
optional and by default it is set to an empty list
</p>
</li>
<li> <p><code>get_RLum(RLum.Data.Spectrum)</code>: Accessor method for <code>RLum.Data.Spectrum</code> object.
</p>
</li>
<li> <p><code>names_RLum(RLum.Data.Spectrum)</code>: Returns the names info elements coming along with this curve object
</p>
</li>
<li> <p><code>bin_RLum.Data(RLum.Data.Spectrum)</code>: Allows binning of RLum.Data.Spectrum data. Count values and values on the x-axis are summed-up;
for wavelength/energy values the mean is calculated.
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>recordType</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> containing the type of the curve (e.g. &quot;TL&quot; or &quot;OSL&quot;)</p>
</dd>
<dt><code>curveType</code></dt><dd><p>Object of class <a href="base.html#topic+character">character</a> containing curve type, allowed values are
&quot;measured&quot; or &quot;predefined&quot;</p>
</dd>
<dt><code>data</code></dt><dd><p>Object of class <a href="base.html#topic+matrix">matrix</a> containing spectrum (count) values.
Row labels indicate wavelength/pixel values, column labels are temperature or time values.</p>
</dd>
<dt><code>info</code></dt><dd><p>Object of class <a href="base.html#topic+list">list</a> containing further meta information objects</p>
</dd>
</dl>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>set_RLum("RLum.Data.Spectrum", ...)</code>.
</p>


<h3>Class version</h3>

<p>0.5.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum.Data.Spectrum-class(): Class 'RLum.Data.Spectrum'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The class should only contain data for a single spectra data set. For
additional elements the slot <code>info</code> can be used. Objects from this class are automatically
created by, e.g., <a href="#topic+read_XSYG2R">read_XSYG2R</a>
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum-class">RLum</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+plot_RLum">plot_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum.Data.Spectrum")

##show example data
data(ExampleData.XSYG, envir = environment())
TL.Spectrum

##show data matrix
get_RLum(TL.Spectrum)

##plot spectrum
## Not run: 
plot_RLum(TL.Spectrum)

## End(Not run)

</code></pre>

<hr>
<h2 id='RLum.Results-class'>Class <code>"RLum.Results"</code></h2><span id='topic+RLum.Results-class'></span><span id='topic+show+2CRLum.Results-method'></span><span id='topic+set_RLum+2CRLum.Results-method'></span><span id='topic+get_RLum+2CRLum.Results-method'></span><span id='topic+length_RLum+2CRLum.Results-method'></span><span id='topic+names_RLum+2CRLum.Results-method'></span><span id='topic+view+2CRLum.Results-method'></span>

<h3>Description</h3>

<p>Object class contains results data from functions (e.g., <a href="#topic+analyse_SAR.CWOSL">analyse_SAR.CWOSL</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'RLum.Results'
show(object)

## S4 method for signature 'RLum.Results'
set_RLum(class, originator, .uid, .pid, data = list(), info = list())

## S4 method for signature 'RLum.Results'
get_RLum(object, data.object, info.object = NULL, drop = TRUE)

## S4 method for signature 'RLum.Results'
length_RLum(object)

## S4 method for signature 'RLum.Results'
names_RLum(object)

## S4 method for signature 'RLum.Results'
view(object, element = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RLum.Results-class_+3A_object">object</code></td>
<td>
<p>an object of class <a href="#topic+RLum.Results-class">RLum.Results</a></p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_class">class</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the <code>RLum</code> class to create</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_originator">originator</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
contains the name of the calling function (the function that produces this object);
can be set manually.</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_.uid">.uid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>automatic</em>):
sets an unique ID for this object using the internal C++ function <code>create_UID</code>.</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_.pid">.pid</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide a parent id for nesting at will.</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_data">data</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+list">list</a> (<em>optional</em>):
a list containing the data to
be stored in the object</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_info">info</code></td>
<td>
<p><code><a href="#topic+set_RLum">set_RLum</a></code>; <a href="base.html#topic+list">list</a> (<em>optional</em>):
a list containing additional info data for the object</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_data.object">data.object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>; <a href="base.html#topic+character">character</a> or <a href="base.html#topic+numeric">numeric</a>:
name or index of the data slot to be returned</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_info.object">info.object</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>; <a href="base.html#topic+character">character</a> (<em>optional</em>):
name of the wanted info element</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_drop">drop</code></td>
<td>
<p><code><a href="#topic+get_RLum">get_RLum</a></code>; <a href="base.html#topic+logical">logical</a> (<em>with default</em>):
coerce to the next possible layer (which are data objects, <code>drop = FALSE</code>
keeps the original <code>RLum.Results</code></p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_element">element</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>): index of the element to display</p>
</td></tr>
<tr><td><code id="RLum.Results-class_+3A_...">...</code></td>
<td>
<p>other arguments that might be passed</p>
</td></tr>
</table>


<h3>Value</h3>

<p><strong><code>set_RLum</code></strong>:
</p>
<p>Returns an object from the class <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>
<p><strong><code>get_RLum</code></strong>:
</p>
<p>Returns:
</p>

<ol>
<li><p> Data object from the specified slot
</p>
</li>
<li> <p><a href="base.html#topic+list">list</a> of data objects from the slots if 'data.object' is vector or
</p>
</li>
<li><p> an <a href="#topic+RLum.Results-class">RLum.Results</a> for <code>drop = FALSE</code>.
</p>
</li></ol>

<p><strong><code>length_RLum</code></strong>
</p>
<p>Returns the number of data elements in the <code>RLum.Results</code> object.
</p>
<p><strong><code>names_RLum</code></strong>
</p>
<p>Returns the names of the data elements in the object.
</p>


<h3>Methods (by generic)</h3>


<ul>
<li> <p><code>show(RLum.Results)</code>: Show structure of <code>RLum.Results</code> object
</p>
</li>
<li> <p><code>set_RLum(RLum.Results)</code>: Construction method for an RLum.Results object.
</p>
</li>
<li> <p><code>get_RLum(RLum.Results)</code>: Accessor method for RLum.Results object. The argument data.object allows
directly accessing objects delivered within the slot data. The default
return object depends on the object originator (e.g., <code>fit_LMCurve</code>).
If nothing is specified always the first <code>data.object</code> will be returned.
</p>
<p>Note: Detailed specification should be made in combination with the originator slot in the
receiving function if results are pipped.
</p>
</li>
<li> <p><code>length_RLum(RLum.Results)</code>: Returns the length of the object, i.e., number of stored data.objects
</p>
</li>
<li> <p><code>names_RLum(RLum.Results)</code>: Returns the names data.objects
</p>
</li>
<li> <p><code>view(RLum.Results)</code>: View method for <a href="#topic+RLum.Results-class">RLum.Results</a> objects
</p>
</li></ul>


<h3>Slots</h3>


<dl>
<dt><code>data</code></dt><dd><p>Object of class <a href="base.html#topic+list">list</a> containing output data</p>
</dd>
</dl>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("RLum.Results", ...)</code>.
</p>


<h3>Class version</h3>

<p>0.5.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. RLum.Results-class(): Class 'RLum.Results'. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The class is intended to store results from functions to be used by
other functions. The data in the object should always be accessed by the
method <code>get_RLum</code>.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum-class">RLum</a>, <a href="#topic+plot_RLum">plot_RLum</a>, <a href="#topic+merge_RLum">merge_RLum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
showClass("RLum.Results")

##create an empty object from this class
set_RLum(class = "RLum.Results")

##use another function to show how it works

##Basic calculation of the dose rate for a specific date
 dose.rate &lt;-  calc_SourceDoseRate(
   measurement.date = "2012-01-27",
   calib.date = "2014-12-19",
   calib.dose.rate = 0.0438,
   calib.error = 0.0019)

##show object
dose.rate

##get results
get_RLum(dose.rate)

##get parameters used for the calcualtion from the same object
get_RLum(dose.rate, data.object = "parameters")

##alternatively objects can be accessed using S3 generics, such as
dose.rate$parameters

</code></pre>

<hr>
<h2 id='scale_GammaDose'>Calculate the gamma dose deposited within a sample taking layer-to-layer
variations in radioactivity into account (according to Aitken, 1985)</h2><span id='topic+scale_GammaDose'></span>

<h3>Description</h3>

<p>This function calculates the gamma dose deposited in a luminescence sample
taking into account layer-to-layer variations in sediment radioactivity.
The function scales user inputs of uranium, thorium and potassium based on
input parameters for sediment density, water content and given layer
thicknesses and distances to the sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scale_GammaDose(
  data,
  conversion_factors = c("Cresswelletal2018", "Guerinetal2011", "AdamiecAitken1998",
    "Liritzisetal2013")[1],
  fractional_gamma_dose = c("Aitken1985")[1],
  verbose = TRUE,
  plot = TRUE,
  plot_singlePanels = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scale_GammaDose_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
A table containing all relevant information for each individual layer. The
table must have the following named columns:
</p>

<ul>
<li> <p><code>id</code> (<a href="base.html#topic+character">character</a>): an arbitrary id or name of each layer
</p>
</li>
<li> <p><code>thickness</code> (<a href="base.html#topic+numeric">numeric</a>): vertical extent of each layer in cm
</p>
</li>
<li> <p><code>sample_offset</code> (<a href="base.html#topic+logical">logical</a>): distance of the sample in cm,
<strong>measured from the BOTTOM OF THE TARGET LAYER</strong>. Except for the target layer
all values must be <code>NA</code>.
</p>
</li>
<li> <p><code>K</code> (<a href="base.html#topic+numeric">numeric</a>): K nuclide content in %
</p>
</li>
<li> <p><code>K_se</code> (<a href="base.html#topic+numeric">numeric</a>): error on the K content
</p>
</li>
<li> <p><code>Th</code> (<a href="base.html#topic+numeric">numeric</a>): Th nuclide content in ppm
</p>
</li>
<li> <p><code>Th_se</code> (<a href="base.html#topic+numeric">numeric</a>): error on the Th content
</p>
</li>
<li> <p><code>U</code> (<a href="base.html#topic+numeric">numeric</a>): U nuclide content in ppm
</p>
</li>
<li> <p><code>U_se</code> (<a href="base.html#topic+numeric">numeric</a>): error on the U content
</p>
</li>
<li> <p><code>water_content</code> (<a href="base.html#topic+numeric">numeric</a>): water content of each layer in %
</p>
</li>
<li> <p><code>water_content_se</code> (<a href="base.html#topic+numeric">numeric</a>):  error on the water content
</p>
</li>
<li> <p><code>density</code> (<a href="base.html#topic+numeric">numeric</a>): bulk density of each layer in g/cm^-3
</p>
</li></ul>
</td></tr>
<tr><td><code id="scale_GammaDose_+3A_conversion_factors">conversion_factors</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
The conversion factors used to calculate the dose rate from sediment
nuclide contents. Valid options are:
</p>

<ul>
<li> <p><code>"Cresswelletal2018"</code> (default)
</p>
</li>
<li> <p><code>"Liritzisetal2013"</code>
</p>
</li>
<li> <p><code>"Guerinetal2011"</code>
</p>
</li>
<li> <p><code>"AdamiecAitken1998"</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="scale_GammaDose_+3A_fractional_gamma_dose">fractional_gamma_dose</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
Factors to scale gamma dose rate values. Valid options are:
</p>

<ul>
<li> <p><code>"Aitken1985"</code> (default): Table H1 in the appendix
</p>
</li></ul>
</td></tr>
<tr><td><code id="scale_GammaDose_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="scale_GammaDose_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="scale_GammaDose_+3A_plot_singlepanels">plot_singlePanels</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable single plot mode, i.e. one plot window per plot.</p>
</td></tr>
<tr><td><code id="scale_GammaDose_+3A_...">...</code></td>
<td>
<p>Further parameters passed to <a href="graphics.html#topic+barplot">barplot</a>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>User Input</strong>
</p>
<p>To calculate the gamma dose which is deposited in a sample, the user needs
to provide information on those samples influencing the luminescence sample.
As a rule of thumb, all sediment layers within at least 30 cm radius from
the luminescence sample taken should be taken into account when calculating
the gamma dose rate. However, the actual range of gamma radiation might be
different, depending on the emitting radioelement, the water content and the
sediment density of each layer (Aitken, 1985). Therefore the user is
advised to provide as much detail as possible and physically sensible.
</p>
<p>The function requires a <a href="base.html#topic+data.frame">data.frame</a> that is to be structured
in columns and rows, with samples listed in rows. The first column contains
information on the layer/sample ID, the second on the thickness (in cm) of
each layer, whilst column 3 should contain <code>NA</code> for all layers that are not
sampled for OSL/TL. For the layer the OSL/TL sample was taken from a numerical
value must be provided, which is the distance (in cm) measured from <strong>bottom</strong>
of the layer of interest. If the whole layer was sampled insert <code>0</code>. If the
sample was taken from <em>within</em> the layer, insert a numerical value <code style="white-space: pre;">&#8288;&gt;0&#8288;</code>,
which describes the distance from the middle of the sample to the bottom of
the layer in cm. Columns 4 to 9 should contain radionuclide concentrations
and their standard errors for
potassium (in %), thorium (in ppm) and uranium (in ppm). Columns 10 and 11
give information on the water content and its uncertainty (standard error)
in %. The layer density (in g/cm3) should be given in column 12. No cell
should be left blank. Please ensure to keep the column titles as given in
the example dataset (<code>data('ExampleData.ScaleGammaDose')</code>, see examples).
</p>
<p>The user can decide which dose rate
conversion factors should be used to calculate the gamma dose rates.
The options are:
</p>

<ul>
<li> <p><code>"Cresswelletal2018"</code> (Cresswell et al., 2018)
</p>
</li>
<li> <p><code>"Liritzisetal2013"</code> (Liritzis et al., 2013)
</p>
</li>
<li> <p><code>"Guerinetal2011"</code> (Guerin et al., 2011)
</p>
</li>
<li> <p><code>"AdamiecAitken1998"</code> (Adamiec and Aitken, 1998)
</p>
</li></ul>

<p><strong>Water content</strong>
</p>
<p>The water content provided by the user should be calculated according to:
</p>
<p style="text-align: center;"><code class="reqn"> ( Wet weight [g] - Dry weight [g] ) / Dry weight [g] * 100 </code>
</p>

<p><strong>Calculations</strong>
</p>
<p>After converting the radionuclide concentrations into dose rates, the
function will scale the dose rates based on the thickness of the layers,
the distances to the sample, the water content and the density of the sediment.
The calculations are based on Aitken (1985, Appendix H). As an example
(equivalent to Aitken, 1985), assuming three layers of sediment, where <strong>L</strong> is
inert and positioned in between the infinite thick and equally active
layers <strong>A</strong> and <strong>B</strong>, the dose in <strong>L</strong> and <strong>B</strong> due to <strong>A</strong> is given by
</p>
<p style="text-align: center;"><code class="reqn"> {1-f(x)}D_A </code>
</p>

<p>Where <code>x</code> is the distance into the inert medium, so <code>f(x)</code> is the weighted
average fractional dose at <code>x</code> and <code>D_A</code> denotes that the dose is delivered by <strong>A</strong>.
<code>f(x)</code> is derived from table H1 (Aitken, 1985), when setting <code>z = x</code>.
Consequently, the dose in <strong>A</strong> and <strong>L</strong> due to <strong>B</strong> is given by
</p>
<p style="text-align: center;"><code class="reqn"> {1 - f(t-x)}D_B </code>
</p>

<p>Here <code>t</code> is the thickness of <strong>L</strong> and the other parameters are denoted as above,
just for the dose being delivered by B. <code>f(t-x)</code> is derived from table H1
(Aitken, 1985), when setting <code>z</code> equal to <code>t-x</code>. Following this, the dose in <strong>L</strong>
delivered by <strong>A</strong> and <strong>B</strong> is given by
</p>
<p style="text-align: center;"><code class="reqn"> {2 - f(x) - f(t-x)}D_{AB} </code>
</p>

<p>Since <strong>A</strong> and <strong>B</strong> are equally active <code style="white-space: pre;">&#8288;D_{AB} = D_A = D_B&#8288;</code>.
</p>
<p>The function uses the value of the fractional dose rate at the layer
boundary to start the calculation for the next layer. This way, the function
is able to scale the gamma dose rate accurately for distant layers when the
density and water content is not constant for the entire section.
</p>


<h3>Value</h3>

<p>After performing the calculations the user is provided with different outputs.
</p>

<ol>
<li><p> The total gamma dose rate received by the sample (+/- uncertainties) as a
print in the console.
</p>
</li>
<li><p> A plot showing the sediment sequence, the user input sample information
and the contribution to total gamma dose rate.
</p>
</li>
<li><p> RLum Results. If the user wishes to save these results, writing a script
to run the function and to save the results would look like this:
</p>
</li></ol>

<div class="sourceCode"><pre>mydata &lt;- read.table("c:/path/to/input/file.txt")
results &lt;- scale_GammaDose(mydata)
table &lt;- get_RLum(results)
write.csv(table, "c:/path/to/results.csv")
</pre></div>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$summary&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> summary of the model results </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$data&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> the original input data </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$dose_rates&#8288;</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> two <code>data.frames</code> for the scaled and infinite matrix dose rates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$tables&#8288;</code> </td><td style="text-align: left;"> <code>list</code> </td><td style="text-align: left;"> several <code>data.frames</code> containing intermediate results </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$args&#8288;</code> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> arguments of the call </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$call&#8288;</code> </td><td style="text-align: left;"> <code>call</code> </td><td style="text-align: left;"> the original function call </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>slot:</strong> <strong><code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>Currently unused.
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
<code style="white-space: pre;">&#8288;[ PLOT OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;<br />
</p>
<p>Three plots are produced:
</p>

<ul>
<li><p> A visualisation of the provided sediment layer structure to quickly
assess whether the data was provided and interpreted correctly.
</p>
</li>
<li><p> A scatter plot of the nuclide contents per layer (K, Th, U) as well as the
water content. This may help to correlate the dose rate contribution of
specific layers to the layer of interest.
</p>
</li>
<li><p> A barplot visualising the contribution of each layer to the total dose rate
received by the sample in the target layer.
</p>
</li></ul>



<h3>Function version</h3>

<p>0.1.3
</p>


<h3>Acknowledgements</h3>

<p>We thank Dr Ian Bailiff for the provision of an excel spreadsheet, which has
been very helpful when writing this function.
</p>


<h3>How to cite</h3>

<p>Riedesel, S., Autzen, M., Burow, C., 2025. scale_GammaDose(): Calculate the gamma dose deposited within a sample taking layer-to-layer variations in radioactivity into account (according to Aitken, 1985). Function version 0.1.3. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p><strong>This function has BETA status. If possible, results should be</strong>
<strong>cross-checked.</strong>
</p>


<h3>Author(s)</h3>

<p>Svenja Riedesel, Aberystwyth University (United Kingdom) <br />
Martin Autzen, DTU NUTECH Center for Nuclear Technologies (Denmark) <br />
Christoph Burow, University of Cologne (Germany) <br />
Based on an excel spreadsheet and accompanying macro written by Ian Bailiff.
, RLum Developer Team</p>


<h3>References</h3>

<p>Aitken, M.J., 1985. Thermoluminescence Dating. Academic Press, London.
</p>
<p>Adamiec, G., Aitken, M.J., 1998. Dose-rate conversion factors: update.
Ancient TL 16, 37-46.
</p>
<p>Cresswell., A.J., Carter, J., Sanderson, D.C.W., 2018.
Dose rate conversion parameters: Assessment of nuclear data.
Radiation Measurements 120, 195-201.
</p>
<p>Guerin, G., Mercier, N., Adamiec, G., 2011. Dose-rate conversion
factors: update. Ancient TL, 29, 5-8.
</p>
<p>Liritzis, I., Stamoulis, K., Papachristodoulou, C., Ioannides, K., 2013.
A re-evaluation of radiation dose-rate conversion factors. Mediterranean
Archaeology and Archaeometry 13, 1-15.
</p>


<h3>See Also</h3>

<p><a href="#topic+ExampleData.ScaleGammaDose">ExampleData.ScaleGammaDose</a>,
<a href="#topic+BaseDataSet.ConversionFactors">BaseDataSet.ConversionFactors</a>, <a href="stats.html#topic+approx">approx</a>, <a href="graphics.html#topic+barplot">barplot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load example data
data("ExampleData.ScaleGammaDose", envir = environment())
x &lt;- ExampleData.ScaleGammaDose

# Scale gamma dose rate
results &lt;- scale_GammaDose(data = x,
                           conversion_factors = "Cresswelletal2018",
                           fractional_gamma_dose = "Aitken1985",
                           verbose = TRUE,
                           plot = TRUE)

get_RLum(results)

</code></pre>

<hr>
<h2 id='Second2Gray'>Converting equivalent dose values from seconds (s) to Gray (Gy)</h2><span id='topic+Second2Gray'></span>

<h3>Description</h3>

<p>This function is defunct, use <a href="#topic+convert_Second2Gray">convert_Second2Gray</a> instead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Second2Gray(...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Second2Gray_+3A_...">...</code></td>
<td>
<p>Unused.</p>
</td></tr>
</table>

<hr>
<h2 id='set_Risoe.BINfileData'>General setter function for Risoe.BINfileData objects</h2><span id='topic+set_Risoe.BINfileData'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects. Depending on the input object, the
corresponding function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_Risoe.BINfileData(
  METADATA = data.frame(),
  DATA = list(),
  .RESERVED = list()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_Risoe.BINfileData_+3A_metadata">METADATA</code></td>
<td>
<p>x</p>
</td></tr>
<tr><td><code id="set_Risoe.BINfileData_+3A_data">DATA</code></td>
<td>
<p>x</p>
</td></tr>
<tr><td><code id="set_Risoe.BINfileData_+3A_.reserved">.RESERVED</code></td>
<td>
<p>x</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object.
</p>


<h3>Function version</h3>

<p>0.1
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. set_Risoe.BINfileData(): General setter function for Risoe.BINfileData objects. Function version 0.1. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>

<hr>
<h2 id='set_RLum'>General setter function for RLum S4 class objects</h2><span id='topic+set_RLum'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the given class, the corresponding
method to create an object from this class will be selected.
Allowed additional arguments can be found in the documentations of the
corresponding <a href="#topic+RLum-class">RLum</a> class:
</p>

<ul>
<li> <p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>,
</p>
</li>
<li> <p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
</p>
</li>
<li> <p><a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>,
</p>
</li>
<li> <p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>,
</p>
</li>
<li> <p><a href="#topic+RLum.Results-class">RLum.Results</a>
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>set_RLum(class, originator, .uid = create_UID(), .pid = NA_character_, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_RLum_+3A_class">class</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of the S4 class to create, must correspond to one of the <a href="#topic+RLum-class">RLum</a>
classes.</p>
</td></tr>
<tr><td><code id="set_RLum_+3A_originator">originator</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>automatic</em>):
contains the name of the calling function (the function that produces this object);
can be set manually.</p>
</td></tr>
<tr><td><code id="set_RLum_+3A_.uid">.uid</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>automatic</em>):
unique ID for this object, by default set using the internal C++ function
<code>create_UID</code>.</p>
</td></tr>
<tr><td><code id="set_RLum_+3A_.pid">.pid</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
option to provide a parent id for nesting at will.</p>
</td></tr>
<tr><td><code id="set_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the specified <a href="#topic+RLum-class">RLum</a> class.
</p>


<h3>Function version</h3>

<p>0.3.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. set_RLum(): General setter function for RLum S4 class objects. Function version 0.3.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## produce empty objects from each class
set_RLum(class = "RLum.Data.Curve")
set_RLum(class = "RLum.Data.Spectrum")
set_RLum(class = "RLum.Data.Spectrum")
set_RLum(class = "RLum.Analysis")
set_RLum(class = "RLum.Results")

## produce a curve object with arbitrary curve values
object &lt;- set_RLum(
class = "RLum.Data.Curve",
curveType = "arbitrary",
recordType = "OSL",
data = matrix(c(1:100,exp(-c(1:100))),ncol = 2))

## plot this curve object
plot_RLum(object)

</code></pre>

<hr>
<h2 id='smooth_RLum'>Smoothing of data for RLum S4-class objects</h2><span id='topic+smooth_RLum'></span><span id='topic+smooth_RLum+2Clist-method'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum-class">RLum</a> class. The smoothing is based on an internal function
called <code>.smoothing</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smooth_RLum(object, ...)

## S4 method for signature 'list'
smooth_RLum(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="smooth_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code></p>
</td></tr>
<tr><td><code id="smooth_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same type as the input object provided.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>smooth_RLum(list)</code>: Returns a list of <a href="#topic+RLum-class">RLum</a> objects that had been passed to <a href="#topic+smooth_RLum">smooth_RLum</a>
</p>
</li></ul>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. smooth_RLum(): Smoothing of data for RLum S4-class objects. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>Currently only <code>RLum</code> objects of class <code>RLum.Data.Curve</code> and <code>RLum.Analysis</code>
(with curve data) are supported.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.CW_OSL_Curve, envir = environment())

## create RLum.Data.Curve object from this example
curve &lt;-
  set_RLum(
      class = "RLum.Data.Curve",
      recordType = "OSL",
      data = as.matrix(ExampleData.CW_OSL_Curve)
  )

## plot data without and with smoothing
plot_RLum(curve)
plot_RLum(smooth_RLum(curve))

</code></pre>

<hr>
<h2 id='sort_RLum'>Sort data for RLum S4-class objects</h2><span id='topic+sort_RLum'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum-class">RLum</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_RLum(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sort_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> or <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
S4 object of class <code>RLum.Analysis</code> or <code>Risoe.BINfileData</code></p>
</td></tr>
<tr><td><code id="sort_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of the same type as the input object provided.
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Colombo, M., 2025. sort_RLum(): Sort data for RLum S4-class objects. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Marco Colombo, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.XSYG, envir = environment())
obj &lt;- OSL.SARMeasurement$Sequence.Object[1:9]

sort_RLum(obj, slot = "recordType")
sort_RLum(obj, info_element = "curveDescripter")

</code></pre>

<hr>
<h2 id='sTeve'>sTeve - sophisticated tool for efficient data validation and evaluation</h2><span id='topic+sTeve'></span>

<h3>Description</h3>

<p>This function provides a sophisticated routine for comprehensive
luminescence dating data analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sTeve(n_frames = 10, t_animation = 2, n.tree = 7, type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sTeve_+3A_n_frames">n_frames</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
n frames</p>
</td></tr>
<tr><td><code id="sTeve_+3A_t_animation">t_animation</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
t animation</p>
</td></tr>
<tr><td><code id="sTeve_+3A_n.tree">n.tree</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
how many trees do you want to cut?</p>
</td></tr>
<tr><td><code id="sTeve_+3A_type">type</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>optional</em>):
Make a decision: 1, 2 or 3</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This amazing sophisticated function validates your data seriously.
</p>


<h3>Value</h3>

<p>Validates your data.
</p>


<h3>How to cite</h3>

<p>NA, NA, , , 2025. sTeve(): sTeve - sophisticated tool for efficient data validation and evaluation. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function should not be taken too seriously.
</p>


<h3>Author(s)</h3>

<p>R Luminescence Team, 2012-2046
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+plot_KDE">plot_KDE</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##no example available

</code></pre>

<hr>
<h2 id='structure_RLum'>General structure function for RLum S4 class objects</h2><span id='topic+structure_RLum'></span><span id='topic+structure_RLum+2Clist-method'></span>

<h3>Description</h3>

<p>The function provides a generalised access point for specific
<a href="#topic+RLum-class">RLum</a> objects. Depending on the input object, the corresponding
function will be selected.
Allowed arguments can be found in the documentations of the corresponding
<a href="#topic+RLum-class">RLum</a> class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>structure_RLum(object, ...)

## S4 method for signature 'list'
structure_RLum(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="structure_RLum_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> (<strong>required</strong>):
S4 object of class <code>RLum</code></p>
</td></tr>
<tr><td><code id="structure_RLum_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <a href="base.html#topic+data.frame">data.frame</a> with structure of the object.
</p>


<h3>Functions</h3>


<ul>
<li> <p><code>structure_RLum(list)</code>: Returns a list of <a href="#topic+RLum-class">RLum</a> objects that were passed to <a href="#topic+structure_RLum">structure_RLum</a>
</p>
</li></ul>


<h3>Function version</h3>

<p>0.2.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. structure_RLum(): General structure function for RLum S4 class objects. Function version 0.2.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data.Curve-class">RLum.Data.Curve</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>,
<a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.XSYG, envir = environment())

## show structure
structure_RLum(OSL.SARMeasurement$Sequence.Object)

</code></pre>

<hr>
<h2 id='subset_SingleGrainData'>Simple Subsetting of Single Grain Data from Ris√∏ BIN/BINX files</h2><span id='topic+subset_SingleGrainData'></span>

<h3>Description</h3>

<p>Most measured single grains do not exhibit light and it makes
usually sense to subset single grain datasets using a table of
position and grain pairs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_SingleGrainData(object, selection)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="subset_SingleGrainData_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>): input object with the
data to subset</p>
</td></tr>
<tr><td><code id="subset_SingleGrainData_+3A_selection">selection</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>): selection table with two columns
for position (1st column) and grain (2nd column) (columns names do not matter)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A subset <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. subset_SingleGrainData(): Simple Subsetting of Single Grain Data from Ris√∏ BIN/BINX files. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+verify_SingleGrainData">verify_SingleGrainData</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data
data(ExampleData.BINfileData, envir = environment())

## set POSITION/GRAIN pair dataset
selection &lt;- data.frame(POSITION = c(1,5,7), GRAIN = c(0,0,0))

##subset
subset_SingleGrainData(object = CWOSL.SAR.Data, selection = selection)

</code></pre>

<hr>
<h2 id='template_DRAC'>Create a DRAC input data template (v1.2)</h2><span id='topic+template_DRAC'></span>

<h3>Description</h3>

<p>This function returns a DRAC input template (v1.2) to be used in conjunction
with the <a href="#topic+use_DRAC">use_DRAC</a> function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>template_DRAC(nrow = 1L, preset = NULL, file_input = NULL, notification = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="template_DRAC_+3A_nrow">nrow</code></td>
<td>
<p><a href="base.html#topic+integer">integer</a> (<em>with default</em>):
specifies the number of rows of the template (i.e., the number of data
sets you want to submit).</p>
</td></tr>
<tr><td><code id="template_DRAC_+3A_preset">preset</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
By default, all values of the template are set to <code>NA</code>, which means that
the user needs to fill in <strong>all</strong> data first before submitting to DRAC
using <code>use_DRAC()</code>. To reduce the number of values that need to be
provided, <code>preset</code> can be used to create a template with at least
a minimum of reasonable preset values.
</p>
<p><code>preset</code> can be one of the following:
</p>

<ul>
<li> <p><code>quartz_coarse</code>
</p>
</li>
<li> <p><code>quartz_fine</code>
</p>
</li>
<li> <p><code>feldspar_coarse</code>
</p>
</li>
<li> <p><code>polymineral_fine</code>
</p>
</li>
<li> <p><code>DRAC-example_quartz</code>
</p>
</li>
<li> <p><code>DRAC-example_feldspar</code>
</p>
</li>
<li> <p><code>DRAC-example_polymineral</code>
</p>
</li></ul>

<p>Note that the last three options can be used to produce a template
with values directly taken from the official DRAC input <code>.csv</code> file.</p>
</td></tr>
<tr><td><code id="template_DRAC_+3A_file_input">file_input</code></td>
<td>
<p><a href="base.html#topic+character">character</a> file connection to a DRAC <code>.csv</code> file, the file
will be imported and translated to the template that can be used by <a href="#topic+use_DRAC">use_DRAC</a>.
Please note that there is not check on validity of the <code>.csv</code> file.</p>
</td></tr>
<tr><td><code id="template_DRAC_+3A_notification">notification</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
show or hide the notification</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A  list of class <code>DRAC.list</code>.
</p>


<h3>How to cite</h3>

<p>Burow, C., Kreutzer, S., 2025. template_DRAC(): Create a DRAC input data template (v1.2). In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Christoph Burow, University of Cologne (Germany), Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Durcan, J.A., King, G.E., Duller, G.A.T., 2015. DRAC: Dose Rate and Age Calculator for trapped charge dating.
Quaternary Geochronology 28, 54-61. doi:10.1016/j.quageo.2015.03.012
</p>


<h3>See Also</h3>

<p><a href="base.html#topic+as.data.frame">as.data.frame</a>, <a href="base.html#topic+list">list</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# create a new DRAC input input
input &lt;- template_DRAC(preset = "DRAC-example_quartz")

# show content of the input
print(input)
print(input$`Project ID`)
print(input[[4]])


## Example: DRAC Quartz example
# note that you only have to assign new values where they
# are different to the default values
input$`Project ID` &lt;- "DRAC-Example"
input$`Sample ID` &lt;- "Quartz"
input$`Conversion factors` &lt;- "AdamiecAitken1998"
input$`External U (ppm)` &lt;- 3.4
input$`errExternal U (ppm)` &lt;- 0.51
input$`External Th (ppm)` &lt;- 14.47
input$`errExternal Th (ppm)` &lt;- 1.69
input$`External K (%)` &lt;- 1.2
input$`errExternal K (%)` &lt;- 0.14
input$`Calculate external Rb from K conc?` &lt;- "N"
input$`Calculate internal Rb from K conc?` &lt;- "N"
input$`Scale gammadoserate at shallow depths?` &lt;- "N"
input$`Grain size min (microns)` &lt;- 90
input$`Grain size max (microns)` &lt;- 125
input$`Water content ((wet weight - dry weight)/dry weight) %` &lt;- 5
input$`errWater content %` &lt;- 2
input$`Depth (m)` &lt;- 2.2
input$`errDepth (m)` &lt;- 0.22
input$`Overburden density (g cm-3)` &lt;- 1.8
input$`errOverburden density (g cm-3)` &lt;- 0.1
input$`Latitude (decimal degrees)` &lt;- 30.0000
input$`Longitude (decimal degrees)` &lt;- 70.0000
input$`Altitude (m)` &lt;- 150
input$`De (Gy)` &lt;- 20
input$`errDe (Gy)` &lt;- 0.2

# use DRAC
## Not run: 
output &lt;- use_DRAC(input)

## End(Not run)

</code></pre>

<hr>
<h2 id='trim_RLum.Data'>Trim Channels of RLum.Data-class Objects</h2><span id='topic+trim_RLum.Data'></span>

<h3>Description</h3>

<p>Trim off the number of channels of <a href="#topic+RLum.Data-class">RLum.Data</a> objects of similar record type
on the time domain. This function is useful in cases where objects have different lengths (short/longer
measurement time) but should be analysed jointly by other functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>trim_RLum.Data(object, recordType = NULL, trim_range = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="trim_RLum.Data_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data-class">RLum.Data</a> <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>): input object,
can be a <a href="base.html#topic+list">list</a> of objects. Please note that in the latter case the function works
only isolated on each element of the <a href="base.html#topic+list">list</a>.</p>
</td></tr>
<tr><td><code id="trim_RLum.Data_+3A_recordtype">recordType</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>): type of the record where the trim
should be applied. If not set, the types are determined automatically and applied
for each record type classes. Can be provided as <a href="base.html#topic+list">list</a>.</p>
</td></tr>
<tr><td><code id="trim_RLum.Data_+3A_trim_range">trim_range</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>optional</em>): sets the range of indices to
keep. If only one value is given, this is taken to be the minimum; if two
values are given, then the range is defined between the two values
(inclusive). Any value beyond the second is silently ignored. If nothing
is set (default), then all curves are trimmed to the same maximum length.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function has two modes of operation:
</p>

<ol>
<li><p> Single <a href="#topic+RLum.Data-class">RLum.Data</a> objects or a <a href="base.html#topic+list">list</a> of such objects:
the function is applied separately over each object.
</p>
</li>
<li><p> Multiple curves via <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> or a <a href="base.html#topic+list">list</a> of such objects:
in this mode, the function first determines the minimum number of channels
for each category of records and then jointly processes them. For instance,
if the object contains one TL curve with 100 channels and two OSL curves
with 100 and 99 channels, respectively, then the minimum would be set to
100 channels for the TL curve and to 99 for the OSL curves. If no further
parameters are applied, the function will shorten all OSL curves to 99
channels, but leave the TL curve untouched.
</p>
</li></ol>



<h3>Value</h3>

<p>A trimmed object or <a href="base.html#topic+list">list</a> of such objects similar to the input objects
</p>


<h3>Function version</h3>

<p>0.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. trim_RLum.Data(): Trim Channels of RLum.Data-class Objects. Function version 0.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## trim all TL curves in the object to channels 10 to 20
data(ExampleData.BINfileData, envir = environment())
temp &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data, pos = 1)

c &lt;- trim_RLum.Data(
object = temp,
recordType = "TL",
trim_range = c(10,20))

plot_RLum.Analysis(
object = c,
combine = TRUE,
subset = list(recordType = "TL"))

## simulate a situation where one OSL curve
## in the dataset has only 999 channels instead of 1000
## all curves should be limited to 999
temp@records[[2]]@data &lt;- temp@records[[2]]@data[-nrow(temp[[2]]@data),]

c &lt;- trim_RLum.Data(object = temp)
nrow(c@records[[4]]@data)


</code></pre>

<hr>
<h2 id='tune_Data'>Tune data for experimental purpose</h2><span id='topic+tune_Data'></span>

<h3>Description</h3>

<p>The error can be reduced and sample size increased for specific purpose.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tune_Data(data, decrease.error = 0, increase.data = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="tune_Data_+3A_data">data</code></td>
<td>
<p><a href="base.html#topic+data.frame">data.frame</a> (<strong>required</strong>):
input values, structure: data (<code>values[,1]</code>) and data error (<code>values [,2]</code>)
are required</p>
</td></tr>
<tr><td><code id="tune_Data_+3A_decrease.error">decrease.error</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
factor by which the error is decreased, ranges between 0 and 1.</p>
</td></tr>
<tr><td><code id="tune_Data_+3A_increase.data">increase.data</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a>:
factor by which the error is decreased, ranges between 0 and <code>Inf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a <a href="base.html#topic+data.frame">data.frame</a> with tuned values.
</p>


<h3>Function version</h3>

<p>0.5.0
</p>


<h3>How to cite</h3>

<p>Dietze, M., 2025. tune_Data(): Tune data for experimental purpose. Function version 0.5.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>You should not use this function to improve your poor data set!
</p>


<h3>Author(s)</h3>

<p>Michael Dietze, GFZ Potsdam (Germany)
, RLum Developer Team</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## load example data set
data(ExampleData.DeValues, envir = environment())
x &lt;- ExampleData.DeValues$CA1

## plot original data
plot_AbanicoPlot(data = x,
                 summary = c("n", "mean"))

## decrease error by 10 %
plot_AbanicoPlot(data = tune_Data(x, decrease.error = 0.1),
                 summary = c("n", "mean"))

## increase sample size by 200 %
#plot_AbanicoPlot(data = tune_Data(x, increase.data = 2) ,
#                summary = c("n", "mean"))

</code></pre>

<hr>
<h2 id='use_DRAC'>Use DRAC to calculate dose rate data</h2><span id='topic+use_DRAC'></span>

<h3>Description</h3>

<p>The function provides an interface from R to DRAC. An R-object or a
CSV file is passed to the DRAC website and results are re-imported into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>use_DRAC(file, name, print_references = TRUE, citation_style = "text", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="use_DRAC_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
name of a CSV file (formatted according to the DRAC v1.2 CSV template) to
be sent to the DRAC website for calculation. It can also be a DRAC template
object obtained from <code><a href="#topic+template_DRAC">template_DRAC()</a></code>.</p>
</td></tr>
<tr><td><code id="use_DRAC_+3A_name">name</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
Optional user name submitted to DRAC. If omitted, a random name will be generated</p>
</td></tr>
<tr><td><code id="use_DRAC_+3A_print_references">print_references</code></td>
<td>
<p>(<em>with default</em>):
Print all references used in the input data table to the console.</p>
</td></tr>
<tr><td><code id="use_DRAC_+3A_citation_style">citation_style</code></td>
<td>
<p>(<em>with default</em>):
If <code>print_references = TRUE</code> this argument determines the output style of the
used references. Valid options are <code>"Bibtex"</code>, <code>"citation"</code>, <code>"html"</code>, <code>"latex"</code>
or <code>"R"</code>. Default is <code>"text"</code>.</p>
</td></tr>
<tr><td><code id="use_DRAC_+3A_...">...</code></td>
<td>
<p>Further arguments.
</p>

<ul>
<li> <p><code>url</code> <a href="base.html#topic+character">character</a>: provide an alternative URL to DRAC
</p>
</li>
<li> <p><code>ignore_version</code> <a href="base.html#topic+logical">logical</a>: ignores the version check, this might come in handy
if the version has changed, but not the column order
</p>
</li>
<li> <p><code>user</code> <a href="base.html#topic+character">character</a>: option to provide username for secured site
</p>
</li>
<li> <p><code>password</code> <a href="base.html#topic+character">character</a>: password for secured site, only works jointly with <code>user</code>
</p>
</li>
<li> <p><code>verbose</code> <a href="base.html#topic+logical">logical</a>: show or hide console output
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns an <a href="#topic+RLum.Results-class">RLum.Results</a> object containing the following elements:
</p>
<table role = "presentation">
<tr><td><code>DRAC</code></td>
<td>
<p><a href="base.html#topic+list">list</a>: a named list containing the following elements in slot <code style="white-space: pre;">&#8288;@data&#8288;</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$highlights&#8288;</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> summary of 25 most important input/output fields </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$header&#8288;</code> </td><td style="text-align: left;"> <a href="base.html#topic+character">character</a> </td><td style="text-align: left;"> HTTP header from the DRAC server response </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$labels&#8288;</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> descriptive headers of all input/output fields </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$content&#8288;</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> complete DRAC input/output table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$input&#8288;</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> DRAC input table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$output&#8288;</code> </td><td style="text-align: left;"> <a href="base.html#topic+data.frame">data.frame</a> </td><td style="text-align: left;"> DRAC output table </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>references</code></td><td style="text-align: left;"> <a href="base.html#topic+list">list</a> </td><td style="text-align: left;"> A list of bib entries of used references </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

</td></tr>
<tr><td><code>data</code></td>
<td>
<p><a href="base.html#topic+character">character</a> or <a href="base.html#topic+list">list</a> path to the input spreadsheet or a DRAC template</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p><a href="base.html#topic+call">call</a> the function call</p>
</td></tr>
<tr><td><code>args</code></td>
<td>
<p><a href="base.html#topic+list">list</a> used arguments</p>
</td></tr>
</table>
<p>The output should be accessed using the function <a href="#topic+get_RLum">get_RLum</a>.
</p>


<h3>Function version</h3>

<p>0.15
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., Dietze, M., Burow, C., 2025. use_DRAC(): Use DRAC to calculate dose rate data. Function version 0.15. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)<br />
Michael Dietze, GFZ Potsdam (Germany)<br />
Christoph Burow, University of Cologne (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>Durcan, J.A., King, G.E., Duller, G.A.T., 2015. DRAC: Dose Rate and Age Calculator for trapped charge dating.
Quaternary Geochronology 28, 54-61. doi:10.1016/j.quageo.2015.03.012
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## (1) Method using the DRAC spreadsheet

file &lt;-  "/PATH/TO/DRAC_Input_Template.csv"

# send the actual IO template spreadsheet to DRAC
## Not run: 
use_DRAC(file = file)

## End(Not run)



## (2) Method using an R template object

# Create a template
input &lt;- template_DRAC(preset = "DRAC-example_quartz")

# Fill the template with values
input$`Project ID` &lt;- "DRAC-Example"
input$`Sample ID` &lt;- "Quartz"
input$`Conversion factors` &lt;- "AdamiecAitken1998"
input$`External U (ppm)` &lt;- 3.4
input$`errExternal U (ppm)` &lt;- 0.51
input$`External Th (ppm)` &lt;- 14.47
input$`errExternal Th (ppm)` &lt;- 1.69
input$`External K (%)` &lt;- 1.2
input$`errExternal K (%)` &lt;- 0.14
input$`Calculate external Rb from K conc?` &lt;- "N"
input$`Calculate internal Rb from K conc?` &lt;- "N"
input$`Scale gammadoserate at shallow depths?` &lt;- "N"
input$`Grain size min (microns)` &lt;- 90
input$`Grain size max (microns)` &lt;- 125
input$`Water content ((wet weight - dry weight)/dry weight) %` &lt;- 5
input$`errWater content %` &lt;- 2
input$`Depth (m)` &lt;- 2.2
input$`errDepth (m)` &lt;- 0.22
input$`Overburden density (g cm-3)` &lt;- 1.8
input$`errOverburden density (g cm-3)` &lt;- 0.1
input$`Latitude (decimal degrees)` &lt;- 30.0000
input$`Longitude (decimal degrees)` &lt;- 70.0000
input$`Altitude (m)` &lt;- 150
input$`De (Gy)` &lt;- 20
input$`errDe (Gy)` &lt;- 0.2

# use DRAC
## Not run: 
output &lt;- use_DRAC(input)

## End(Not run)

</code></pre>

<hr>
<h2 id='verify_SingleGrainData'>Verify single grain data sets and check for invalid grains, i.e.
zero-light level grains</h2><span id='topic+verify_SingleGrainData'></span>

<h3>Description</h3>

<p>This function tries to identify automatically zero-light level curves (grains)
from single grain data measurements.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verify_SingleGrainData(
  object,
  threshold = 10,
  use_fft = FALSE,
  cleanup = FALSE,
  cleanup_level = "aliquot",
  verbose = TRUE,
  plot = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="verify_SingleGrainData_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> or <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> (<strong>required</strong>):
input object. The function also accepts a list with objects of allowed type.</p>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_threshold">threshold</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>):
numeric threshold value for the allowed difference between the <code>mean</code> and
the <code>var</code> of the count values (see details)</p>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_use_fft">use_fft</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): applies an additional approach based on <a href="stats.html#topic+fft">stats::fft</a>.
The threshold is fixed and cannot be changed.</p>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_cleanup">cleanup</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
if set to <code>TRUE</code>, curves/aliquots identified as zero light level curves/aliquots are
automatically removed. Output is an object as same type as the input, i.e.
either <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> or <a href="#topic+RLum.Analysis-class">RLum.Analysis</a></p>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_cleanup_level">cleanup_level</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
selects the level for the clean-up of the input data sets.
Two options are allowed: <code>"curve"</code> or <code>"aliquot"</code>:
</p>

<ul>
<li><p> If  <code>"curve"</code> is selected, every single curve marked as <code>invalid</code> is removed.
</p>
</li>
<li><p> If <code>"aliquot"</code> is selected, curves of one aliquot (grain or disc) can be
marked as invalid, but will not be removed. An aliquot will be only removed
if all curves of this aliquot are marked as invalid.
</p>
</li></ul>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disables output to the terminal.</p>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_plot">plot</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the plot output.</p>
</td></tr>
<tr><td><code id="verify_SingleGrainData_+3A_...">...</code></td>
<td>
<p>further parameters to control the plot output; if selected.
Supported arguments <code>main</code>, <code>ylim</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p><strong>How does the method work?</strong>
</p>
<p>The function compares the expected values (<code class="reqn">E(X)</code>) and the variance
(<code class="reqn">Var(X)</code>) of the count values for each curve. Assuming that the
background roughly follows a Poisson distribution, the absolute difference
of both values should be zero or at least around zero as
</p>
<p style="text-align: center;"><code class="reqn">E(x) = Var(x) = \lambda</code>
</p>

<p>Thus the function checks for:
</p>
<p style="text-align: center;"><code class="reqn">abs(E(x) - Var(x)) &gt;= \Theta</code>
</p>

<p>With <code class="reqn">\Theta</code> an arbitrary, user defined, threshold. Values above the
threshold indicate curves comprising a signal.
</p>
<p>Note: the absolute difference of <code class="reqn">E(X)</code> and <code class="reqn">Var(x)</code> instead of the
ratio was chosen as both terms can become 0 which would result in 0 or <code>Inf</code>,
if the ratio is calculated.
</p>


<h3>Value</h3>

<p>The function returns
</p>
<p>&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
<code style="white-space: pre;">&#8288;[ NUMERICAL OUTPUT ]&#8288;</code><br />
&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&mdash;&ndash;<br />
</p>
<p><strong><code>RLum.Results</code></strong>-object
</p>
<p><strong>slot:****<code style="white-space: pre;">&#8288;@data&#8288;</code></strong>
</p>

<table>
<tr>
 <td style="text-align: left;">
<strong>Element</strong> </td><td style="text-align: left;"> <strong>Type</strong> </td><td style="text-align: left;"> <strong>Description</strong></td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$unique_pairs&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> the unique position and grain pairs </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$selection_id&#8288;</code> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> the selection as record ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<code style="white-space: pre;">&#8288;$selection_full&#8288;</code> </td><td style="text-align: left;"> <code>data.frame</code> </td><td style="text-align: left;"> implemented models used in the baSAR-model core </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><strong>slot:****<code style="white-space: pre;">&#8288;@info&#8288;</code></strong>
</p>
<p>The original function call
</p>
<p><strong>Output variation</strong>
</p>
<p>For <code>cleanup = TRUE</code> the same object as the input is returned, but cleaned up
(invalid curves were removed). This means: Either a <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>
or an <a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object is returned in such cases.
A <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> object can be exported to a BINX-file by
using the function <a href="#topic+write_R2BIN">write_R2BIN</a>.
</p>


<h3>Function version</h3>

<p>0.2.5
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. verify_SingleGrainData(): Verify single grain data sets and check for invalid grains, i.e. zero-light level grains. Function version 0.2.5. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>This function can work with <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects or
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> objects (or a list of it). However, the function is
highly optimised for <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> objects as it make sense to
remove identify invalid grains before the conversion to an
<a href="#topic+RLum.Analysis-class">RLum.Analysis</a> object.
</p>
<p>The function checking for invalid curves works rather robust and it is likely
that Reg0 curves within a SAR cycle are removed as well. Therefore it is
strongly recommended to use the argument <code>cleanup = TRUE</code> carefully if
the cleanup works only on curves.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+write_R2BIN">write_R2BIN</a>,
<a href="#topic+read_BIN2R">read_BIN2R</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##01 - basic example I
##just show how to apply the function
data(ExampleData.XSYG, envir = environment())

##verify and get data.frame out of it
verify_SingleGrainData(OSL.SARMeasurement$Sequence.Object)$selection_full

##02 - basic example II
data(ExampleData.BINfileData, envir = environment())
id &lt;- verify_SingleGrainData(object = CWOSL.SAR.Data,
cleanup_level = "aliquot")$selection_id

## Not run: 
##03 - advanced example I
##importing and exporting a BIN-file

##select and import file
file &lt;- file.choose()
object &lt;- read_BIN2R(file)

##remove invalid aliquots(!)
object &lt;- verify_SingleGrainData(object, cleanup = TRUE)

##export to new BIN-file
write_R2BIN(object, paste0(dirname(file),"/", basename(file), "_CLEANED.BIN"))

## End(Not run)

</code></pre>

<hr>
<h2 id='view'>Convenience data visualisation function</h2><span id='topic+view'></span>

<h3>Description</h3>

<p>Invokes the <a href="utils.html#topic+View">utils::View</a> function tailored to objects in the package.
If started from RStudio, it uses the RStudio viewer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>view(object, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="view_+3A_object">object</code></td>
<td>
<p>(<strong>required</strong>) object to view</p>
</td></tr>
<tr><td><code id="view_+3A_...">...</code></td>
<td>
<p>further arguments passed to the specific class method</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>NULL</code> and opens the data viewer.
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. view(): Convenience data visualisation function. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><code><a href="utils.html#topic+View">utils::View()</a></code>
</p>

<hr>
<h2 id='write_R2BIN'>Export Risoe.BINfileData into Ris√∏ BIN/BINX-file</h2><span id='topic+write_R2BIN'></span>

<h3>Description</h3>

<p>Exports a <code>Risoe.BINfileData</code> object in a <code style="white-space: pre;">&#8288;*.bin&#8288;</code> or <code style="white-space: pre;">&#8288;*.binx&#8288;</code> file that can be
opened by the Analyst software or other Ris√∏ software.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_R2BIN(
  object,
  file,
  version,
  compatibility.mode = FALSE,
  verbose = TRUE,
  txtProgressBar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_R2BIN_+3A_object">object</code></td>
<td>
<p><a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> (<strong>required</strong>):
input object to be stored in a bin file.</p>
</td></tr>
<tr><td><code id="write_R2BIN_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>):
file name and path of the output file
</p>

<ul>
<li> <p><code style="white-space: pre;">&#8288;[WIN]&#8288;</code>: <code>write_R2BIN(object, "C:/Desktop/test.bin")</code>
</p>
</li>
<li> <p><code style="white-space: pre;">&#8288;[MAC/LINUX]&#8288;</code>: <code>write_R2BIN("/User/test/Desktop/test.bin")</code>
</p>
</li></ul>
</td></tr>
<tr><td><code id="write_R2BIN_+3A_version">version</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
version number for the output file. If no value is provided, the highest
version number from the <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a> is taken automatically.
</p>
<p><strong>Note:</strong>
This argument can be used to convert BIN-file versions.</p>
</td></tr>
<tr><td><code id="write_R2BIN_+3A_compatibility.mode">compatibility.mode</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
this option recalculates the position values if necessary and set the max.
value to 48. The old position number is appended as comment (e.g., 'OP: 70).
This option accounts for potential compatibility problems with the Analyst software.
It further limits the maximum number of points per curve to 9,999. If a curve contains more
data the curve data get binned using the smallest possible bin width.</p>
</td></tr>
<tr><td><code id="write_R2BIN_+3A_verbose">verbose</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable output to the terminal.</p>
</td></tr>
<tr><td><code id="write_R2BIN_+3A_txtprogressbar">txtProgressBar</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the progress bar. Ignored if <code>verbose = FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The structure of the exported binary data follows the data structure
published in the Appendices of the <em>Analyst</em> manual p. 42.
</p>
<p>If
<code>LTYPE</code>, <code>DTYPE</code> and <code>LIGHTSOURCE</code> are not of type
<a href="base.html#topic+character">character</a>, no transformation into numeric values is done.
</p>


<h3>Value</h3>

<p>Write a binary file and returns the name and path of the file as <a href="base.html#topic+character">character</a>.
</p>


<h3>Function version</h3>

<p>0.5.4
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. write_R2BIN(): Export Risoe.BINfileData into Ris√∏ BIN/BINX-file. Function version 0.5.4. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Note</h3>

<p>The function just roughly checks the data structures. The validity of
the output data depends on the user.
</p>
<p>The validity of the file path is not further checked. BIN-file conversions
using the argument <code>version</code> may be a lossy conversion, depending on the
chosen input and output data (e.g., conversion from version 08 to 07 to 06 to 05 to 04 or 03).
</p>
<p><strong>Warning</strong>
</p>
<p>Although the coding was done carefully, it seems that the BIN/BINX-files
produced by Ris√∏ DA 15/20 TL/OSL readers slightly differ on the byte level.
No obvious differences are observed in the METADATA, however, the
BIN/BINX-file may not fully compatible, at least not similar to the ones
directly produced by the Ris√∏ readers!
</p>
<p>ROI definitions (introduced in BIN-file version 8) are not supported!
There are furthermore ignored by the function <a href="#topic+read_BIN2R">read_BIN2R</a>.
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>References</h3>

<p>DTU Nutech, 2016. The Sequence Editor, Users Manual, February, 2016.
<a href="https://www.fysik.dtu.dk">https://www.fysik.dtu.dk</a>
</p>


<h3>See Also</h3>

<p><a href="#topic+read_BIN2R">read_BIN2R</a>, <a href="#topic+Risoe.BINfileData-class">Risoe.BINfileData</a>, <a href="base.html#topic+writeBin">writeBin</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##load example dataset
file &lt;- system.file("extdata/BINfile_V8.binx", package = "Luminescence")
temp &lt;- read_BIN2R(file)

##create temporary file path
##(for usage replace by own path)
temp_file &lt;- tempfile(pattern = "output", fileext = ".binx")

##export to temporary file path
write_R2BIN(temp, file = temp_file)

</code></pre>

<hr>
<h2 id='write_R2TIFF'>Export RLum.Data.Image and RLum.Data.Spectrum objects to TIFF Images</h2><span id='topic+write_R2TIFF'></span>

<h3>Description</h3>

<p>Simple wrapper around <a href="tiff.html#topic+writeTIFF">tiff::writeTIFF</a> to export suitable
RLum-class objects to TIFF images. Per default 16-bit TIFF files are exported.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_R2TIFF(object, file = tempfile(), norm = 65535, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_R2TIFF_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a> or <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a> object (<strong>required</strong>):
input object, can be a <a href="base.html#topic+list">list</a> of such objects</p>
</td></tr>
<tr><td><code id="write_R2TIFF_+3A_file">file</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<strong>required</strong>): the file name and path</p>
</td></tr>
<tr><td><code id="write_R2TIFF_+3A_norm">norm</code></td>
<td>
<p><a href="base.html#topic+numeric">numeric</a> (<em>with default</em>): normalisation values. Values in TIFF files must range between 0-1, however, usually
in imaging applications the pixel values are real integer count values. The normalisation to the
to the highest 16-bit integer values -1 ensures that the numerical values are retained in the exported
image. If <code>1</code> nothing is normalised.</p>
</td></tr>
<tr><td><code id="write_R2TIFF_+3A_...">...</code></td>
<td>
<p>further arguments to be passed to <a href="tiff.html#topic+writeTIFF">tiff::writeTIFF</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A TIFF file
</p>


<h3>Function version</h3>

<p>0.1.0
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. write_R2TIFF(): Export RLum.Data.Image and RLum.Data.Spectrum objects to TIFF Images. Function version 0.1.0. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Institute of Geography, Heidelberg University (Germany)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="tiff.html#topic+writeTIFF">tiff::writeTIFF</a>, <a href="#topic+RLum.Data.Image-class">RLum.Data.Image</a>, <a href="#topic+RLum.Data.Spectrum-class">RLum.Data.Spectrum</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ExampleData.RLum.Data.Image, envir = environment())
write_R2TIFF(ExampleData.RLum.Data.Image, file = tempfile())

</code></pre>

<hr>
<h2 id='write_RLum2CSV'>Export RLum-objects to CSV</h2><span id='topic+write_RLum2CSV'></span>

<h3>Description</h3>

<p>This function exports <a href="#topic+RLum-class">RLum</a>-objects to CSV-files using the R function
<a href="utils.html#topic+write.table">utils::write.table</a>. All <a href="#topic+RLum-class">RLum</a>-objects are supported, but the
export is lossy, i.e. the pure numerical values are exported only. Information
that cannot be coerced to a <a href="base.html#topic+data.frame">data.frame</a> or a <a href="base.html#topic+matrix">matrix</a> are discarded as well as
metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_RLum2CSV(
  object,
  path = NULL,
  prefix = "",
  export = TRUE,
  compact = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="write_RLum2CSV_+3A_object">object</code></td>
<td>
<p><a href="#topic+RLum-class">RLum</a> or a <a href="base.html#topic+list">list</a> of <code>RLum</code> objects (<strong>required</strong>):
objects to be written. Can be a <a href="base.html#topic+data.frame">data.frame</a> if needed internally.</p>
</td></tr>
<tr><td><code id="write_RLum2CSV_+3A_path">path</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>optional</em>):
character string naming folder for the output to be written. If nothing
is provided <code>path</code> will be set to the working directory.
<strong>Note:</strong> this argument is ignored when <code>export = FALSE</code>.</p>
</td></tr>
<tr><td><code id="write_RLum2CSV_+3A_prefix">prefix</code></td>
<td>
<p><a href="base.html#topic+character">character</a> (<em>with default</em>):
optional prefix to name the files. This prefix is valid for all written files</p>
</td></tr>
<tr><td><code id="write_RLum2CSV_+3A_export">export</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>):
enable/disable the file export. If set to <code>FALSE</code> nothing is written to
the file connection, but a list comprising objects of type <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
is returned instead.</p>
</td></tr>
<tr><td><code id="write_RLum2CSV_+3A_compact">compact</code></td>
<td>
<p><a href="base.html#topic+logical">logical</a> (<em>with default</em>): if <code>TRUE</code> (default) the output
will be simpler but less comprehensive, that is not all elements in the
objects will be fully broken down. This is in particular useful for writing
<code>RLum.Results</code> objects to CSV files, as such objects can be rather complex
and not all information are needed in a CSV file or can be meaningfully
translated to CSV format.</p>
</td></tr>
<tr><td><code id="write_RLum2CSV_+3A_...">...</code></td>
<td>
<p>further arguments that will be passed to the function
<a href="utils.html#topic+write.table">utils::write.table</a>. All arguments except the argument <code>file</code> are supported</p>
</td></tr>
</table>


<h3>Details</h3>

<p>However, in combination with the implemented import functions, nearly every
supported import data format can be exported to CSV-files, this gives a great
deal of freedom in terms of compatibility with other tools.
</p>
<p><strong>Input is a list of objects</strong>
</p>
<p>If the input is a <a href="base.html#topic+list">list</a> of objects all explicit function arguments can be provided
as <a href="base.html#topic+list">list</a>.
</p>


<h3>Value</h3>

<p>The function returns either a CSV-file (or many of them) or for the
option <code>export == FALSE</code> a list comprising objects of type <a href="base.html#topic+data.frame">data.frame</a> and <a href="base.html#topic+matrix">matrix</a>
</p>


<h3>Function version</h3>

<p>0.2.2
</p>


<h3>How to cite</h3>

<p>Kreutzer, S., 2025. write_RLum2CSV(): Export RLum-objects to CSV. Function version 0.2.2. In: Kreutzer, S., Burow, C., Dietze, M., Fuchs, M.C., Schmidt, C., Fischer, M., Friedrich, J., Mercier, N., Philippe, A., Riedesel, S., Autzen, M., Mittelstrass, D., Gray, H.J., Galharret, J., Colombo, M., Steinbuch, L., Boer, A.d., 2025. Luminescence: Comprehensive Luminescence Dating Data Analysis. R package version 1.0.1. https://r-lum.github.io/Luminescence/
</p>


<h3>Author(s)</h3>

<p>Sebastian Kreutzer, Geography &amp; Earth Science, Aberystwyth University (United Kingdom)
, RLum Developer Team</p>


<h3>See Also</h3>

<p><a href="#topic+RLum.Analysis-class">RLum.Analysis</a>, <a href="#topic+RLum.Data-class">RLum.Data</a>, <a href="#topic+RLum.Results-class">RLum.Results</a>,
<a href="utils.html#topic+write.table">utils::write.table</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
##transform values to a list (and do not write)
data(ExampleData.BINfileData, envir = environment())
object &lt;- Risoe.BINfileData2RLum.Analysis(CWOSL.SAR.Data)[[1]]
write_RLum2CSV(object, export = FALSE)

## Not run: 

##create temporary filepath
##(for usage replace by own path)
temp_file &lt;- tempfile(pattern = "output", fileext = ".csv")

##write CSV-file to working directory
write_RLum2CSV(temp_file)


## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
