<!DOCTYPE html><html lang="en"><head><title>Help for package MAAPER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MAAPER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#maaper'><p>Model-based analysis of alternative polyadenylation using 3’ end-linked reads</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Analysis of Alternative Polyadenylation Using 3' End-Linked
Reads</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A computational method developed for model-based analysis of alternative polyadenylation (APA) using 3' end-linked reads. It accurately assigns 3' RNA-seq reads to polyA sites through statistical modeling, and generates multiple statistics for APA analysis. Please also see Li WV, Zheng D, Wang R, Tian B (2021) &lt;<a href="https://doi.org/10.1186%2Fs13059-021-02429-5">doi:10.1186/s13059-021-02429-5</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>parallel, GenomicRanges, GenomicAlignments, GenomicFeatures,
GenomeInfoDb, stats, utils, Rsamtools, IRanges, MASS</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Vivianstats/MAAPER">https://github.com/Vivianstats/MAAPER</a>,
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02429-5">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02429-5</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-14 00:15:35 UTC; wei</td>
</tr>
<tr>
<td>Author:</td>
<td>Wei Vivian Li <a href="https://orcid.org/0000-0002-2087-2709"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wei Vivian Li &lt;vivian.li@rutgers.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-14 14:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='maaper'>Model-based analysis of alternative polyadenylation using 3’ end-linked reads</h2><span id='topic+maaper'></span>

<h3>Description</h3>

<p>Model-based analysis of alternative polyadenylation using 3’ end-linked reads
</p>


<h3>Usage</h3>

<pre><code class='language-R'>maaper(
  gtf,
  pas_annotation,
  output_dir,
  bam_c1,
  bam_c2,
  read_len,
  ncores = 1,
  num_pas_thre = 25,
  frac_pas_thre = 0.05,
  dist_thre = 600,
  num_thre = 50,
  run = "all",
  subset = NULL,
  region = "all",
  gtf_rds = NULL,
  verbose = FALSE,
  paired = FALSE,
  bed = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="maaper_+3A_gtf">gtf</code></td>
<td>
<p>A character specifying the full path of the GTF file (reference genome);</p>
</td></tr>
<tr><td><code id="maaper_+3A_pas_annotation">pas_annotation</code></td>
<td>
<p>A list containing the pas annotation. MAAPER provides processed annotation
information from PolyA_DB v3 on its Github page.</p>
</td></tr>
<tr><td><code id="maaper_+3A_output_dir">output_dir</code></td>
<td>
<p>A character specifying the full path of the output directory,
which is used to store all intermdediate and final outputs.</p>
</td></tr>
<tr><td><code id="maaper_+3A_bam_c1">bam_c1</code></td>
<td>
<p>A character vector specifying the full paths to the bam files for
condition 1 (control). The length of the vector equals the number of samples.</p>
</td></tr>
<tr><td><code id="maaper_+3A_bam_c2">bam_c2</code></td>
<td>
<p>A character vector specifying the full paths to the bam files for
condition 2 (experiment). The length of the vector equals the number of samples.</p>
</td></tr>
<tr><td><code id="maaper_+3A_read_len">read_len</code></td>
<td>
<p>An integer specifying the read length.</p>
</td></tr>
<tr><td><code id="maaper_+3A_ncores">ncores</code></td>
<td>
<p>An integer specifying the number of cores used in parallel computation.</p>
</td></tr>
<tr><td><code id="maaper_+3A_num_pas_thre">num_pas_thre</code></td>
<td>
<p>An integer specifying the threhold on PAS's read number. Defaults to 25.</p>
</td></tr>
<tr><td><code id="maaper_+3A_frac_pas_thre">frac_pas_thre</code></td>
<td>
<p>A numeric specifying the threshold on PAS's fraction. Defaults to 0.05.</p>
</td></tr>
<tr><td><code id="maaper_+3A_dist_thre">dist_thre</code></td>
<td>
<p>An integer specifying the threshold on fragment length. Defaults to 600.</p>
</td></tr>
<tr><td><code id="maaper_+3A_num_thre">num_thre</code></td>
<td>
<p>An integer specifying the threhold on gene's read number. Defaults to 50.</p>
</td></tr>
<tr><td><code id="maaper_+3A_run">run</code></td>
<td>
<p>&quot;all&quot; (default) or &quot;skip-train&quot;. For test and debug only.</p>
</td></tr>
<tr><td><code id="maaper_+3A_subset">subset</code></td>
<td>
<p>A character vector specifying genes' Ensembl IDs if only a subset of genes need to be analyzed.
Check the <code>pas_annotation</code> files for ID formats.</p>
</td></tr>
<tr><td><code id="maaper_+3A_region">region</code></td>
<td>
<p>&quot;all&quot; (default). For test and debug only.</p>
</td></tr>
<tr><td><code id="maaper_+3A_gtf_rds">gtf_rds</code></td>
<td>
<p>NULL (default). For test and debug only.</p>
</td></tr>
<tr><td><code id="maaper_+3A_verbose">verbose</code></td>
<td>
<p>FALSE (default). For test and debug only.</p>
</td></tr>
<tr><td><code id="maaper_+3A_paired">paired</code></td>
<td>
<p>A boolean indicating whether to perform paired test instead of unpaired test (defaults to FALSE).</p>
</td></tr>
<tr><td><code id="maaper_+3A_bed">bed</code></td>
<td>
<p>Aboolean indicating whether bedGraph files should be output for visualization in genome browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>maaper</code> saves two text files, gene.txt and pas.txt, to <code>out_dir</code>.
pas.txt contains the gene names, predicted PASs, and their corresponding fractions in the two conditions.
gene.txt contains the genes' PAS number, p values, RED, RLDu, and RLDi scores.
</p>


<h3>Author(s)</h3>

<p>Wei Vivian Li, <a href="mailto:vivian.li@rutgers.edu">vivian.li@rutgers.edu</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
# data used in this example can be found on the package's Github page
pas_annotation = readRDS("./mouse.PAS.mm9.rds")
gtf = "./gencode.mm9.chr19.gtf"
bam_c1 = "./NT_chr19_example.bam"
bam_c2 = "./AS_4h_chr19_example.bam"
maaper(gtf, pas_annotation, output_dir = "./",
       bam_c1, bam_c2, read_len = 76, ncores = 1)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
