<!DOCTYPE html><html lang="en-GB"><head><title>Help for package MACER</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MACER}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#align_to_ref'><p>Align and Trim MSA Against a Reference</p></a></li>
<li><a href='#auto_seq_download'><p>Automatic Sequence Download</p></a></li>
<li><a href='#barcode_clean'><p>DNA Barcode Clean</p></a></li>
<li><a href='#create_fastas'><p>Table To FASTA</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Molecular Acquisition, Cleaning, and Evaluation in R 'MACER'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Robert G Young &lt;rgyoung6@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>To assist biological researchers in assembling taxonomically and marker focused molecular sequence data sets. 'MACER' accepts a list of genera as a user input and uses NCBI-GenBank and BOLD as resources to download and assemble molecular sequence datasets. These datasets are then assembled by marker, aligned, trimmed, and cleaned. The use of this package allows the publication of specific parameters to ensure reproducibility. The 'MACER' package has four core functions and an example run through using all of these functions can be found in the associated repository <a href="https://github.com/rgyoung6/MACER_example">https://github.com/rgyoung6/MACER_example</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td>&lt;<a href="https://github.com/rgyoung6/MACER&amp;gt;">https://github.com/rgyoung6/MACER&gt;</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>rentrez, ape, httr, stats, utils, ggplot2, parallel, pbapply,
grDevices, png</td>
</tr>
<tr>
<td>Language:</td>
<td>en-GB</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-12-02 23:20:20 UTC; robgy</td>
</tr>
<tr>
<td>Author:</td>
<td>Robert G Young <a href="https://orcid.org/0000-0002-6731-2506"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Rekkab Gill [aut],
  Daniel Gillis [aut],
  Robert H Hanner [aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-12-02 23:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='align_to_ref'>Align and Trim MSA Against a Reference</h2><span id='topic+align_to_ref'></span>

<h3>Description</h3>

<p>Takes a FASTA file with target sequences and aligns them against a reference sequence submitted to the program.
The output is an aligned fasta file that is trimmed to the length of the reference sequence. Sequences without full coverage (records having sequences with leading or trailing gaps) are removed.
Records with characters other than IUPAC are also removed. Finally, internal gaps are removed from the sequence based on the submitted multiple sequence alignment percent
coverage of the character position as provided in the pigl argument supplied by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>align_to_ref(
  data_folder = NULL,
  ref_seq_file = NULL,
  MAFFT_loc = NULL,
  output_file = NULL,
  pigl = 0.95,
  op = 1.53
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="align_to_ref_+3A_data_folder">data_folder</code></td>
<td>
<p>This variable can be used to provide a location for the file containing all of the fasta files wanting to be aligned. The default value is set to NULL where the program will prompt the user to select the folder through point-and-click.</p>
</td></tr>
<tr><td><code id="align_to_ref_+3A_ref_seq_file">ref_seq_file</code></td>
<td>
<p>This variable can be used to provide a location for the reference sequence file. The default value is set to NULL where the program will prompt the user to select the folder through point-and-click.</p>
</td></tr>
<tr><td><code id="align_to_ref_+3A_mafft_loc">MAFFT_loc</code></td>
<td>
<p>This variable can be used to provide a location for the MAFFT program. The default value is set to NULL where the program will prompt the user to select the folder through point-and-click.</p>
</td></tr>
<tr><td><code id="align_to_ref_+3A_output_file">output_file</code></td>
<td>
<p>This variable can be used to set the location of the output files from the program. The default value is set to NULL where the program will place the output files in the same location as the target files.</p>
</td></tr>
<tr><td><code id="align_to_ref_+3A_pigl">pigl</code></td>
<td>
<p>This is the percent internal gap loop argument. This provides a percent that will remove records causing internal gaps if more than the percent value assigned
to this argument is reached. If this value is set to 0 then internal gaps are not removed. The default for this value is 0.95.</p>
</td></tr>
<tr><td><code id="align_to_ref_+3A_op">op</code></td>
<td>
<p>This is the gap opening penalty for the use of MAFFT. The higher the value the larger penalty in the alignment. The default for this value is set to 1.53 which is the
default value in the MAFFT program. For alignment of highly conserved regions where no gaps are expected this should be set to a much higher number and 10 is recommended for coding regions like the COI-5P.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>User Input:
1. A file folder location with the fasta files that need to be aligned and trimmed using the supplied reference sequence. Please note that any and all fasta files (named *.fas) in this folder will be analyzed.
2. A reference sequence file with a sequence or MSA with all sequences having the same length.
3. The location of the MAFFT executable file &lt;https://mafft.cbrc.jp/alignment/software/&gt;
</p>


<h3>Value</h3>

<p>Output:
1. In the submitted file folder location there will be a log file titled MAFFT_log.
2. The sequence output files from this script are placed into two subfolders. These folders are in the submitted file location where the fasta files of interest are located.
The two folders created are MAFFT and MAFFT_trimmed. In the MAFFT folder there will be files with name of the files in the submitted file folder appended with &quot;_MAFFT&quot;.
The MAFFT_trimmed file will contain files with the same naming convention as the files in the submitted folder and appended with &quot;_MAFFT_trimmed&quot;.
</p>


<h3>Author(s)</h3>

<p>Robert G. Young
</p>


<h3>References</h3>

<p>&lt;https://github.com/rgyoung6/MACER&gt;
Young RG, Gill R, Gillis D, Hanner RH (2021) Molecular Acquisition, Cleaning and Evaluation in R (MACER) - A tool to assemble molecular marker datasets from BOLD and GenBank. Biodiversity Data Journal 9: e71378. &lt;https://doi.org/10.3897/BDJ.9.e71378&gt;
</p>


<h3>See Also</h3>

<p>auto_seq_download()
create_fastas()
barcode_clean()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
align_to_ref(pigl=0.75)
align_to_ref(pigl=0.95, op=10)
align_to_ref(pigl=0)

## End(Not run)

</code></pre>

<hr>
<h2 id='auto_seq_download'>Automatic Sequence Download</h2><span id='topic+auto_seq_download'></span>

<h3>Description</h3>

<p>Takes a list of genera, as supplied by the user, and searches and downloads molecular sequence data from BOLD and Genbank.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>auto_seq_download(
  BOLD_database = TRUE,
  NCBI_database = TRUE,
  search_str = NULL,
  input_file = NULL,
  output_file = NULL,
  seq_min = 100,
  seq_max = 2500
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="auto_seq_download_+3A_bold_database">BOLD_database</code></td>
<td>
<p>TRUE is to include, FALSE is to exclude; default TRUE</p>
</td></tr>
<tr><td><code id="auto_seq_download_+3A_ncbi_database">NCBI_database</code></td>
<td>
<p>TRUE is to include, FALSE is to exclude; default TRUE</p>
</td></tr>
<tr><td><code id="auto_seq_download_+3A_search_str">search_str</code></td>
<td>
<p>NULL uses the default string, anything other than NULL then that string will be used for the GenBank search; default NULL.
The Default String is: (genus[ORGN]) NOT (shotgun[ALL] OR genome[ALL] OR assembled[ALL] OR microsatellite[ALL])</p>
</td></tr>
<tr><td><code id="auto_seq_download_+3A_input_file">input_file</code></td>
<td>
<p>NULL prompts the user to indicate the location of the input file through point and click prompts, anything other than NULL then the string supplied will be used for the location; default NULL</p>
</td></tr>
<tr><td><code id="auto_seq_download_+3A_output_file">output_file</code></td>
<td>
<p>NULL prompts the user to indicate the location of the output file through point and click prompts, anything other than NULL then the string supplied will be used for the location; default NULL</p>
</td></tr>
<tr><td><code id="auto_seq_download_+3A_seq_min">seq_min</code></td>
<td>
<p>holds the minimum length value to not flag the sequence; default 100</p>
</td></tr>
<tr><td><code id="auto_seq_download_+3A_seq_max">seq_max</code></td>
<td>
<p>holds the maximum length value to not flag the sequence; default 2500</p>
</td></tr>
</table>


<h3>Details</h3>

<p>User Input: A list of genera in a text file in a single column with a new line at the end of the list.
</p>


<h3>Value</h3>

<p>Outputs: One main folder containing three other folders.
Main folder - Seq_auto_dl_TTTTTT_MMM_DD
Three subfolders:
1. BOLD - Contains a file for every genus downloaded with the raw data from the BOLD system.
2. NCBI - Contains a file for every genus downloaded with the raw data from GenBank.
3. Total_tables - Contains files for the running of the function which include...
A_Summary.txt - This file contains information about the downloads.
A_Total_Table.tsv - A file with a single table containing the accumulated data for all genera searched.
</p>


<h3>Note</h3>

<p>When using a custom search string for NCBI only a single genus at a time can be used.
</p>


<h3>Author(s)</h3>

<p>Robert G. Young
</p>


<h3>References</h3>

<p>&lt;https://github.com/rgyoung6/MACER&gt;
Young, R. G., Gill, R., Gillis, D., Hanner, R. H. (Submitted June 2021). Molecular Acquisition, Cleaning, and Evaluation in R (MACER) - A tool to assemble molecular marker datasets from BOLD and GenBank. Biodiversity Data Journal.
</p>


<h3>See Also</h3>

<p>create_fastas()
align_to_ref()
barcode_clean()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
auto_seq_download()
auto_seq_download(BOLD_database = TRUE, NCBI_database = FALSE)
auto_seq_download(BOLD_database = FALSE, NCBI_database = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='barcode_clean'>DNA Barcode Clean</h2><span id='topic+barcode_clean'></span>

<h3>Description</h3>

<p>Takes an input fasta file and identifies genus level outliers and species outliers based on the 1.5 x greater than the interquartile range.
It also, if selected, checks the sequence using amino acid translation and has the option to eliminate sequences that have non-IUPAC codes.
Finally, the program calculates the barcode gap for the species in the submitted dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>barcode_clean(
  AA_code = "invert",
  AGCT_only = TRUE,
  data_folder = NULL,
  outliers = TRUE,
  dist_model = "raw",
  replicates = 1000,
  replacement = TRUE,
  conf_level = 1,
  numCores = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="barcode_clean_+3A_aa_code">AA_code</code></td>
<td>
<p>This is the amino acid translation matrix (as implemented through ape) used to check the sequences for stop codons. The following codes are available std, vert, invert, F. The default is invert.</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_agct_only">AGCT_only</code></td>
<td>
<p>This indicates if records with characters other than AGCT are kept, the default is TRUE. TRUE removes records with non-AGCT FALSE is accepting all IUPAC characters</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_data_folder">data_folder</code></td>
<td>
<p>This variable can be used to provide a location for the MSA fasta files to be cleaned. The default value is set to NULL where the program will prompt the user to select the folder through point-and-click.</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_outliers">outliers</code></td>
<td>
<p>This is the variable to indicate if the use would like to remove suspected sequence record outliers using 1.5X the genetic distance. If set to TRUE genus and species level outliers will be removed. If FALSE this will not occur. Default TRUE.</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_dist_model">dist_model</code></td>
<td>
<p>This is the model of nucleotide evolution that the ape program will use (see ape documentation for options. Default is &quot;raw&quot;</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_replicates">replicates</code></td>
<td>
<p>This is the number of replicates that the bootstrapping will perform. Note: more replicates will take longer. Default is 1000</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_replacement">replacement</code></td>
<td>
<p>This indicates that the replacement of MSA nucleotide columns will be replaced in the random resampling. Default is set to TRUE</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_conf_level">conf_level</code></td>
<td>
<p>This is a percentage of the initial MSA nucleotide length. When set to 1 the bootstrapped resampling will have the same length as the initial MSA. Default is set to 1</p>
</td></tr>
<tr><td><code id="barcode_clean_+3A_numcores">numCores</code></td>
<td>
<p>This is the number of cores that the user would like to use where multithreading is available. Default is set to 1, indicating only a single thread will be used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input: A file folder with one or more fasta files of interest
</p>


<h3>Value</h3>

<p>Output:
A single log file for the running of the function with the name A_Clean_File_YYYY-DD-TTTTTTTT.
The function will also output three files for each fasta file submitted. The first is the distance matrix that was calculated and used to assess the DNA barcode gaps.
This file is named the same as the input file with dist_table.dat appended to the end of the name. The second file is the total data table file which provides a table
of all submitted records for each data set accompanied with the results from each section of the analysis. This file is named the same as the input fasta with data_table.dat appended
to the end, Finally, a fasta file with all outliers and flagged records removed is generated for each input fasta file. This output file is named the same as the input fasta with no_outlier.fas appended to the end.
</p>


<h3>Author(s)</h3>

<p>Robert G. Young
</p>


<h3>References</h3>

<p>&lt;https://github.com/rgyoung6/MACER&gt;
Young RG, Gill R, Gillis D, Hanner RH (2021) Molecular Acquisition, Cleaning and Evaluation in R (MACER) - A tool to assemble molecular marker datasets from BOLD and GenBank. Biodiversity Data Journal 9: e71378. &lt;https://doi.org/10.3897/BDJ.9.e71378&gt;
</p>


<h3>See Also</h3>

<p>auto_seq_download()
create_fastas()
align_to_ref()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
barcode_clean(),
barcode_clean(AA_code = "vert", AGCT_only = TRUE),
barcode_clean(AA_code = "vert")

## End(Not run)

</code></pre>

<hr>
<h2 id='create_fastas'>Table To FASTA</h2><span id='topic+create_fastas'></span>

<h3>Description</h3>

<p>Using the output table from the download script and the user built genus-marker name parameter file to take the downloaded data and place them into fasta files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_fastas(
  data_file = NULL,
  input_file = NULL,
  output_folder = NULL,
  no_marker = FALSE,
  no_taxa = FALSE,
  no_seq = FALSE,
  name_issue = FALSE,
  taxa_digits = FALSE,
  taxa_punct = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_fastas_+3A_data_file">data_file</code></td>
<td>
<p>NULL prompts the user to indicate the location of the data file in the format of the auto_seq_download output, anything other than NULL then the string supplied will be used for the location; default NULL</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_input_file">input_file</code></td>
<td>
<p>NULL prompts the user to indicate the location of the input file used to select through point and click prompts, anything other than NULL then the string supplied will be used for the location; default NULL</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_output_folder">output_folder</code></td>
<td>
<p>NULL prompts the user to indicate the location of the output file through point and click prompts, anything other than NULL then the string supplied will be used for the location; default NULL</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_no_marker">no_marker</code></td>
<td>
<p>If set to TRUE then will include records filtered out due to no marker data. Default is FALSE to not include records with no marker data.</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_no_taxa">no_taxa</code></td>
<td>
<p>If set to TRUE then will include records filtered out due to no taxa data. Default is FALSE to not include records with no taxa data.</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_no_seq">no_seq</code></td>
<td>
<p>If set to TRUE then will include records filtered out due to no sequence data. Default is FALSE to not include records with no sequence data.</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_name_issue">name_issue</code></td>
<td>
<p>If set to TRUE then will include records filtered out due to genus and species names with more than two terms. Default is FALSE to not include records with taxonomic naming issues.</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_taxa_digits">taxa_digits</code></td>
<td>
<p>If set to TRUE then will include records filtered out due to genus or species names containing digits. Default is FALSE to not include records with digits in the taxonomic naming.</p>
</td></tr>
<tr><td><code id="create_fastas_+3A_taxa_punct">taxa_punct</code></td>
<td>
<p>If set to TRUE then will include records filtered out due to the presence of punctuation in the genus or species names. Default is FALSE to not include records with punctuation in the taxonomic naming.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Input: File with list of genera with the molecular markers names below the taxa. The information to create this parameters file can be obtained from A_Summary.txt file from the download script results.
For further details please see the documentation.
</p>


<h3>Value</h3>

<p>This script outputs a fasta file of sequences for each column in the submitted parameters file. These files are named with the genera of interest and the first marker name in the column of the parameters file.
These files are located in the folder where the Total_tables.txt file is located.
</p>


<h3>Author(s)</h3>

<p>Rekkab Singh Gill
</p>


<h3>References</h3>

<p>&lt;https://github.com/rgyoung6/MACER&gt;
Young RG, Gill R, Gillis D, Hanner RH (2021) Molecular Acquisition, Cleaning and Evaluation in R (MACER) - A tool to assemble molecular marker datasets from BOLD and GenBank. Biodiversity Data Journal 9: e71378. &lt;https://doi.org/10.3897/BDJ.9.e71378&gt;
</p>


<h3>See Also</h3>

<p>create_fastas()
align_to_ref()
barcode_clean()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
create_fastas()
create_fastas(no_marker = TRUE, no_taxa = TRUE)
create_fastas(no_seq  = TRUE, name_issue = TRUE)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
