<!DOCTYPE html><html><head><title>Help for package MACP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MACP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calculate_PPIscore'><p>Calculate Pairwise Protein Profile Similarity using Different</p>
Metrics</a></li>
<li><a href='#Clust_Valid'><p>Cluster Evaluation by External Measures</p></a></li>
<li><a href='#cluster_tuning'><p>ClusterONE Hyperparameters Tuning</p></a></li>
<li><a href='#data_filtering'><p>Data Filtering</p></a></li>
<li><a href='#EliminateCpxRedundance'><p>Hierarchical Clustering of Modules</p></a></li>
<li><a href='#enrichmentCPX'><p>Functional Enrichment Analysis for Predicted Complexes</p></a></li>
<li><a href='#ensemble_model'><p>Predict Interactions via Ensemble Learning Method</p></a></li>
<li><a href='#exampleData'><p>Demo CF-MS data</p></a></li>
<li><a href='#generate_refInt'><p>Generate Class Labels for Data Input Based on Gold Reference Set</p></a></li>
<li><a href='#get_clusters'><p>Predict Complexes</p></a></li>
<li><a href='#get_DenoisedNet'><p>Denoising Predicted Protein-Protein Interactions</p></a></li>
<li><a href='#getCPX'><p>Fetch Complexes from the CORUM Database</p></a></li>
<li><a href='#impute_MissingData'><p>Impute missing Values in Elution Profile Matrix</p></a></li>
<li><a href='#keepMT'><p>Keep Mitochondrial (mt) Proteins</p></a></li>
<li><a href='#MCL_clustering'><p>MCL clustering</p></a></li>
<li><a href='#MCL_tuning'><p>MCL Hyperparameters Tuning</p></a></li>
<li><a href='#orthMappingCpx'><p>Protein Complex Ortholog Mapping</p></a></li>
<li><a href='#predPPI_MACP'><p>Predict Protein-Protein Interactions and Putative Complexes</p></a></li>
<li><a href='#refcpx'><p>CORUM reference complexes</p></a></li>
<li><a href='#scaling'><p>Column and Row-wise Normalization</p></a></li>
<li><a href='#subcellular.mtPPI'><p>Keep Mitochondrial (mt) Proteins</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Macromolecular Assemblies from Co-Elution Profile (MACP)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>The MACP employs machine learning algorithm for automated scoring 
   of co-fractionation mass spectrometry (CF-MS) and then systematically 
   map multi-protein complexes from these high-confidence protein-protein 
   interactions (PPIs) using unsupervised learning (i.e., clustering).</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, zoo, utils, dplyr, lsa, WGCNA, tidyr, tibble, Hmisc,
igraph, PRROC, pROC, ggplot2, grDevices, fmsb, stringr, caret</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Matineh Rahmatbakhsh &lt;matinerb.94@gmail.com&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/BSD-3-Clause">BSD_3_clause</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mrbakhsh/MACP">https://github.com/mrbakhsh/MACP</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mrbakhsh/MACP/issues">https://github.com/mrbakhsh/MACP/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, ptw, e1071, kernlab, ranger, proxy, infotheo,
gridExtra, philentropy, randomForest, gprofiler2, purrr, minet,
entropy, MCL, orthogene, protti, arules, rmarkdown, BiocStyle</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-27 17:42:22 UTC; Matine</td>
</tr>
<tr>
<td>Author:</td>
<td>Matineh Rahmatbakhsh [aut, trl, cre],
  Mohan Babu [led]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-28 17:32:30 UTC</td>
</tr>
</table>
<hr>
<h2 id='calculate_PPIscore'>Calculate Pairwise Protein Profile Similarity using Different
Metrics</h2><span id='topic+calculate_PPIscore'></span>

<h3>Description</h3>

<p>This function first removes proteins pairs for which two
proteins never occurred in the same fractions, then computes
pairwise protein similarity using up to 18 metrics (by default all the 18
measures are activated). This function also
provides users with an option to choose an appropriate co-fractionation
correlation score cut-off using the 'corr_cutoff' argument,
if argument 'corr_removal' is set to TRUE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calculate_PPIscore(
  x,
  pcc = TRUE,
  PCCN = TRUE,
  pcc_p = TRUE,
  spearman = TRUE,
  kendall = TRUE,
  bicor = TRUE,
  weighted_rank = TRUE,
  cosine = TRUE,
  jaccard = TRUE,
  dice = TRUE,
  apex = TRUE,
  minfo = TRUE,
  bayesian = TRUE,
  wcc = TRUE,
  euclidean = TRUE,
  manhattan = TRUE,
  canberra = TRUE,
  avg.distance = TRUE,
  rept = 10,
  corr_removal = FALSE,
  corr_cutoff = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calculate_PPIscore_+3A_x">x</code></td>
<td>
<p>A co-elution data matrix with proteins in rows and
fractions in columns.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_pcc">pcc</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using Pearson correlation metric.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_pccn">PCCN</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity using
Pearson correlation plus noise.This function is adapted from the
PCCN function in the SMED package.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_pcc_p">pcc_p</code></td>
<td>
<p>If TRUE, computes P-value of the Pearson correlation.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_spearman">spearman</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity using
spearman correlation.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_kendall">kendall</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity using
kendall correlation.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_bicor">bicor</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity using
biweight midcorrealtion (bicor) correlation.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_weighted_rank">weighted_rank</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using weighted rank measure.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_cosine">cosine</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using cosine metric.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_jaccard">jaccard</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using jaccard metric.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_dice">dice</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using dice measure.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_apex">apex</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using apex.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_minfo">minfo</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using mutual information.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_bayesian">bayesian</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity using
Bayes correlation based on zero-count distribution.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_wcc">wcc</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using weighted cross correlation.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_euclidean">euclidean</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using euclidean measure.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_manhattan">manhattan</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using manhattan measure.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_canberra">canberra</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using canberra measure.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_avg.distance">avg.distance</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using avg.distance measure.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_rept">rept</code></td>
<td>
<p>Poisson iterations, defaults to 10.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_corr_removal">corr_removal</code></td>
<td>
<p>If TRUE, removes protein pairs with
correlation scores &lt; the user defined threshold ; defaults to FALSE.</p>
</td></tr>
<tr><td><code id="calculate_PPIscore_+3A_corr_cutoff">corr_cutoff</code></td>
<td>
<p>user defined threshold for correlation similarity
scores. Defaults to 0.5.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>calculate_PPIscore
</p>


<h3>Value</h3>

<p>A data frame containing the calculated features for all possible
protein pairs.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>M1&lt;-matrix(rnorm(36),nrow=6)
M1 &lt;- abs(M1)
rownames(M1) &lt;- c("A","B","C","D","E","F")
scored_Data &lt;- calculate_PPIscore(M1)
</code></pre>

<hr>
<h2 id='Clust_Valid'>Cluster Evaluation by External Measures</h2><span id='topic+Clust_Valid'></span>

<h3>Description</h3>

<p>This function evaluate the quality of clusters by comparing
clustering-derived partitions to known labels (i.e., CORUM complexes) and
assess the similarity between them using quality measures including
overlap score (O), sensitivity (Sn), clustering-wise positive predictive
value (PPV), geometric accuracy (Acc), and maximum matching ratio (MMR).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Clust_Valid(predcpx, refcpx)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Clust_Valid_+3A_predcpx">predcpx</code></td>
<td>
<p>A list containing predicted complexes.</p>
</td></tr>
<tr><td><code id="Clust_Valid_+3A_refcpx">refcpx</code></td>
<td>
<p>A list containing reference complexes (i.e.,
CORUM complexes).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Clust_Valid
</p>


<h3>Value</h3>

<p>A list containing the numerical values for each
evaluation metrics.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load known reference complexes
data(refcpx)
# Select subset of complexes to be used as an instance sets for predicted
# complexes
predcpx &lt;- refcpx[5:15]
Eval_result &lt;- Clust_Valid(predcpx,refcpx)
</code></pre>

<hr>
<h2 id='cluster_tuning'>ClusterONE Hyperparameters Tuning</h2><span id='topic+cluster_tuning'></span>

<h3>Description</h3>

<p>This function optimizes the choice of ClusterONE algorithm
parameters such as density, node penalty, and overlap score by comparing
clustering-derived partitions for each combination of parameters to known
labels (i.e., CORUM complexes) and assess the similarity between them
using quality measures including overlap score, sensitivity (Sn),
clustering-wise positive predictive value (PPV), geometric accuracy (Acc),
and maximum matching raio (MMR).It is recommended to first reduce
redundancy in the known reference complexes via
<code><a href="#topic+EliminateCpxRedundance">EliminateCpxRedundance</a></code>,then performs parameter tuning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cluster_tuning(
  refcpx,
  csize = 2,
  d = c(0.3, 0.5),
  p = c(2),
  max_overlap = c(0.5, 0.6),
  tpath = file.path(system.file("extdata", package = "MACP"))
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cluster_tuning_+3A_refcpx">refcpx</code></td>
<td>
<p>A list containing reference complexes
(i.e., corum complexes).</p>
</td></tr>
<tr><td><code id="cluster_tuning_+3A_csize">csize</code></td>
<td>
<p>An integer, the minimum size of the predicted complexes.
Defualts to 2.</p>
</td></tr>
<tr><td><code id="cluster_tuning_+3A_d">d</code></td>
<td>
<p>A vector of number, density of predicted complexes.</p>
</td></tr>
<tr><td><code id="cluster_tuning_+3A_p">p</code></td>
<td>
<p>A vector of integer, penalty value for the inclusion of each node.</p>
</td></tr>
<tr><td><code id="cluster_tuning_+3A_max_overlap">max_overlap</code></td>
<td>
<p>A vector of number, specifies the maximum allowed
overlap between two clusters.</p>
</td></tr>
<tr><td><code id="cluster_tuning_+3A_tpath">tpath</code></td>
<td>
<p>A character string indicating the path to the project
directory that contains the interaction data. Interaction data must be
stored as .txt file and containing id1-id2-weight triplets. Defaults
to MACP/inst/extdata directory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>cluster_tuning
</p>


<h3>Value</h3>

<p>A data.frame containing clustering performance across different
combination of parameters.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>References</h3>

<p>Nepusz, T., Yu, H., and Paccanaro, A. (2012a).
Detecting overlapping protein complexes in protein-protein interaction
networks. Nat. Methods 9, 471.
</p>

<hr>
<h2 id='data_filtering'>Data Filtering</h2><span id='topic+data_filtering'></span>

<h3>Description</h3>

<p>This function removes proteins for which peptide only
detected in one fraction (i.e., &quot;one-hit-wonders&quot;) across the
co-elution table, common contaminants (e.g., keratins) only for mouse
and human organisms and frequent flyers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data_filtering(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="data_filtering_+3A_x">x</code></td>
<td>
<p>A data matrix  object with rows
including proteins and fractions along the columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>data_filtering
</p>


<h3>Value</h3>

<p>Filtered matrix.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the co-elution data
data("exampleData")
# Perform raw data pre-processing
datOut &lt;- data_filtering(exampleData)
</code></pre>

<hr>
<h2 id='EliminateCpxRedundance'>Hierarchical Clustering of Modules</h2><span id='topic+EliminateCpxRedundance'></span>

<h3>Description</h3>

<p>This function reduces redundancy in the reference complexes
by first computing the overlap of two complexes via Jaccard index,
followed by merging overlapping complexes with user-defined threshold
(here is 0.2).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EliminateCpxRedundance(
  rawCpx,
  custom_bg = NULL,
  sim_method = "euclidean",
  linkage = "average",
  h = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EliminateCpxRedundance_+3A_rawcpx">rawCpx</code></td>
<td>
<p>A list containing protein complexes</p>
</td></tr>
<tr><td><code id="EliminateCpxRedundance_+3A_custom_bg">custom_bg</code></td>
<td>
<p>Vector of proteins names to use as a background.
If given, refcpx will be first mapped to the background proteisn, followed
by removing redundancy in the refcpx.</p>
</td></tr>
<tr><td><code id="EliminateCpxRedundance_+3A_sim_method">sim_method</code></td>
<td>
<p>c(euclidean&quot;,
&quot;maximum&quot;, &quot;manhattan&quot;, &quot;canberra&quot;, &quot;binary&quot;, or &quot;minkowski&quot;); Default
is euclidean</p>
</td></tr>
<tr><td><code id="EliminateCpxRedundance_+3A_linkage">linkage</code></td>
<td>
<p>c(&quot;average&quot;, &quot;ward&quot;, &quot;single&quot;, &quot;complete&quot;, &quot;mcquitty&quot;,
&quot;median&quot;, &quot;centroid&quot;); Default is average.</p>
</td></tr>
<tr><td><code id="EliminateCpxRedundance_+3A_h">h</code></td>
<td>
<p>numeric scalar or vector with heights where the tree should be
cut; Defaults to 0.2</p>
</td></tr>
</table>


<h3>Details</h3>

<p>EliminateCpxRedundance
</p>


<h3>Value</h3>

<p>List of unique complexes.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh
</p>


<h3>Examples</h3>

<pre><code class='language-R'># predicted interactions
pred_ppi &lt;- read.table(
system.file("extdata/ppi_input_ClusterONE.txt", package = "MACP"),
header = FALSE)
# get all the proteins in the predicted network
custom_bg &lt;- union(pred_ppi$V1, pred_ppi$V2)
# reference complexes
data("refcpx")
# reduce redundancy in reference complexes
filt_cpx &lt;- EliminateCpxRedundance(refcpx,
custom_bg,
sim_method = "euclidean",
linkage="average",
h = 0.2)
</code></pre>

<hr>
<h2 id='enrichmentCPX'>Functional Enrichment Analysis for Predicted Complexes</h2><span id='topic+enrichmentCPX'></span>

<h3>Description</h3>

<p>This function uses <code><a href="gprofiler2.html#topic+gost">gost</a></code> function
in <code>gprofiler2</code> package to perform functional enrichment analysis
for predicted modules.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>enrichmentCPX(
  predcpx,
  threshold = 0.05,
  sources = c("GO", "KEGG", "CORUM", "REAC", "CORUM"),
  p.corrction.method = "bonferroni",
  custom_bg = NULL,
  org = "mmusculus"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="enrichmentCPX_+3A_predcpx">predcpx</code></td>
<td>
<p>A data.frame containing predicted complexes resulted from
<code><a href="#topic+get_clusters">get_clusters</a></code> or <code><a href="#topic+MCL_clustering">MCL_clustering</a></code>.</p>
</td></tr>
<tr><td><code id="enrichmentCPX_+3A_threshold">threshold</code></td>
<td>
<p>Custom p-value threshold for significance.</p>
</td></tr>
<tr><td><code id="enrichmentCPX_+3A_sources">sources</code></td>
<td>
<p>A vector of data sources to use.
See <code><a href="gprofiler2.html#topic+gost">gost</a></code> for more details.</p>
</td></tr>
<tr><td><code id="enrichmentCPX_+3A_p.corrction.method">p.corrction.method</code></td>
<td>
<p>The algorithm used for multiple testing
correction;defaults to 'bonferroni'.
See <code><a href="gprofiler2.html#topic+gost">gost</a></code> for more details.</p>
</td></tr>
<tr><td><code id="enrichmentCPX_+3A_custom_bg">custom_bg</code></td>
<td>
<p>vector of gene names to use as a statistical background.
Defaults to NULL.</p>
</td></tr>
<tr><td><code id="enrichmentCPX_+3A_org">org</code></td>
<td>
<p>An organism name;defaults to 'mmusculus'.
See <code><a href="gprofiler2.html#topic+gost">gost</a></code> for more details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>enrichmentCPX
</p>


<h3>Value</h3>

<p>A data.frame with the enrichment analysis results.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>

<hr>
<h2 id='ensemble_model'>Predict Interactions via Ensemble Learning Method</h2><span id='topic+ensemble_model'></span>

<h3>Description</h3>

<p>This function uses individual or an ensemble of classifiers to
predict interactions from CF-MS data. This ensemble
algorithm combines different results generated from individual
classifiers within the ensemble via average to enhance prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ensemble_model(
  features,
  gd,
  classifier = c("glm", "svmRadial", "ranger"),
  cv_fold = 2,
  verboseIter = TRUE,
  plots = FALSE,
  filename = file.path(tempdir(), "plots.pdf")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ensemble_model_+3A_features">features</code></td>
<td>
<p>A data frame with protein-protein associations in the
first column, and features to be passed to the classifier in the remaining
columns.</p>
</td></tr>
<tr><td><code id="ensemble_model_+3A_gd">gd</code></td>
<td>
<p>A gold reference set including true associations with class
labels indicating if such PPIs are positive or negative.</p>
</td></tr>
<tr><td><code id="ensemble_model_+3A_classifier">classifier</code></td>
<td>
<p>The type of classifier to use. See <code>caret</code>
for the available classifiers.</p>
</td></tr>
<tr><td><code id="ensemble_model_+3A_cv_fold">cv_fold</code></td>
<td>
<p>Number of partitions for cross-validation; defaults to 5.</p>
</td></tr>
<tr><td><code id="ensemble_model_+3A_verboseiter">verboseIter</code></td>
<td>
<p>Logical value, indicating whether to check the status
of training process;defaults to FALSE.</p>
</td></tr>
<tr><td><code id="ensemble_model_+3A_plots">plots</code></td>
<td>
<p>Logical value, indicating whether to plot the performance
of ensemble learning algorithm as compared to individual classifiers;
defaults to FALSE.If the argument set to TRUE, plots will be saved in
the current working directory.
These plots are :
</p>

<ul>
<li><p>pr_plot - Precision-recall plot of ensemble classifier vs
selected individual classifiers.
</p>
</li>
<li><p>roc_plot - ROC plot of ensemble classifier vs selected individual
classifiers.
</p>
</li>
<li><p>points_plot - Plot accuracy, F1-score ,positive predictive value
(PPV),sensitivity (SE), and Matthews correlation coefficient (MCC) of
ensemble classifier vs selected individual classifiers.
</p>
</li></ul>
<p>.</p>
</td></tr>
<tr><td><code id="ensemble_model_+3A_filename">filename</code></td>
<td>
<p>character string, indicating the location and output
pdf filename for for performance plots. Defaults to tempdir().</p>
</td></tr>
</table>


<h3>Details</h3>

<p>ensemble_model
</p>


<h3>Value</h3>

<p>Ensemble_training_output
</p>

<ul>
<li><p>prediction score - Prediction scores for whole dataset from each
individual classifier.
</p>
</li>
<li><p>Best - Selected hyper parameters.
</p>
</li>
<li><p>Parameter range - Tested hyper parameters.
</p>
</li>
<li><p>prediction_score_test - Scores probabilities for test data from
each individual classifier.
</p>
</li>
<li><p>class_label -  Class probabilities for test data from each
individual classifier.
</p>
</li></ul>

<p>classifier_performance
</p>

<ul>
<li><p>cm - A confusion matrix.
</p>
</li>
<li><p>ACC - Accuracy.
</p>
</li>
<li><p>SE - Sensitivity.
</p>
</li>
<li><p>SP - Specificity.
</p>
</li>
<li><p>PPV - Positive Predictive Value.
</p>
</li>
<li><p>F1 - F1-score.
</p>
</li>
<li><p>MCC - Matthews correlation coefficient.
</p>
</li>
<li><p>Roc_Object - A list of elements.
See <code><a href="pROC.html#topic+roc">roc</a></code> for more details.
</p>
</li>
<li><p>PR_Object - A list of elements.
See <code><a href="PRROC.html#topic+pr.curve">pr.curve</a></code> for more details.
</p>
</li></ul>

<p>predicted_interactions - The input data frame of pairwise interactions,
including classifier scores averaged across all models.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>

<hr>
<h2 id='exampleData'>Demo CF-MS data</h2><span id='topic+exampleData'></span>

<h3>Description</h3>

<p>Co-elution profiles derived from mitochondrial (mt) extracts of
mouse brain culture,fractionated by (size-exclusion chromatography, SEC)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(exampleData)
</code></pre>


<h3>Format</h3>

<p>A matrix with 284 rows and 83 columns,
with proteins in rows and biochemical fractions in columns
</p>

<hr>
<h2 id='generate_refInt'>Generate Class Labels for Data Input Based on Gold Reference Set</h2><span id='topic+generate_refInt'></span>

<h3>Description</h3>

<p>This function creates class labels for protein pairs in the
same order in the data input based on gold reference set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_refInt(x, refcpx)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_refInt_+3A_x">x</code></td>
<td>
<p>A data frame with interacting proteins in the first two
columns.</p>
</td></tr>
<tr><td><code id="generate_refInt_+3A_refcpx">refcpx</code></td>
<td>
<p>A list containing gold standard protein complexes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>generate_refInt
</p>


<h3>Value</h3>

<p>A Data frame containing class labels for protein pairs in the
data input. If protein pairs involve in same protein complexes are
assigned to Positive, otherwise Negative.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>

<hr>
<h2 id='get_clusters'>Predict Complexes</h2><span id='topic+get_clusters'></span>

<h3>Description</h3>

<p>This function partitions high-confidence network to
putative complexes via ClusterONE clustering algorithm to identify
protein complex membership.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_clusters(
  csize = 2,
  d = 0.3,
  p = 2,
  max_overlap = 0.8,
  tpath = file.path(system.file("extdata", package = "MACP"))
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_clusters_+3A_csize">csize</code></td>
<td>
<p>An integer, the minimum size of the predicted complexes.
Defaults to 2.</p>
</td></tr>
<tr><td><code id="get_clusters_+3A_d">d</code></td>
<td>
<p>A number, density of predicted complexes. Defaults to 0.3.</p>
</td></tr>
<tr><td><code id="get_clusters_+3A_p">p</code></td>
<td>
<p>An integer, penalty value for the inclusion of each node.
Defaults to 2.</p>
</td></tr>
<tr><td><code id="get_clusters_+3A_max_overlap">max_overlap</code></td>
<td>
<p>A number, specifies the maximum allowed
overlap between two clusters. Defaults to 0.8.</p>
</td></tr>
<tr><td><code id="get_clusters_+3A_tpath">tpath</code></td>
<td>
<p>A character string indicating the path to the project
directory that contains the interaction data. Interactions data must be
stored as .txt file and containing id1-id2-weight triplets.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>get_clusters
</p>


<h3>Value</h3>

<p>A data.frame containing predicted complexes
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>References</h3>

<p>Nepusz, T., Yu, H., and Paccanaro, A. (2012a).
Detecting overlapping protein complexes in protein-protein interaction
networks. Nat. Methods 9, 471.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>predcpx &lt;-
get_clusters(csize = 3, d = 0.3, p = 2,
max_overlap = 0.8,
tpath = file.path(system.file("extdata", package = "MACP")))
</code></pre>

<hr>
<h2 id='get_DenoisedNet'>Denoising Predicted Protein-Protein Interactions</h2><span id='topic+get_DenoisedNet'></span>

<h3>Description</h3>

<p>This function removes the noise in the form of false
positive edges in the predicted networks using network topology.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_DenoisedNet(ppi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_DenoisedNet_+3A_ppi">ppi</code></td>
<td>
<p>Interactions data containing id1-id2-weight triplets.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>get_DenoisedNet
</p>


<h3>Value</h3>

<p>A data.frame containing denoised network.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># high-confidence network as input
ppi &lt;-
read.table(system.file("extdata/ppi_input_ClusterONE.txt",
package="MACP"),
quote="\"", comment.char="")
# Perform network denoising
denoisetNet &lt;- get_DenoisedNet(ppi)
</code></pre>

<hr>
<h2 id='getCPX'>Fetch Complexes from the CORUM Database</h2><span id='topic+getCPX'></span>

<h3>Description</h3>

<p>This function retrieves protein complexes directly from the
CORUM database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getCPX(org = "Mouse", tpath = tempdir())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getCPX_+3A_org">org</code></td>
<td>
<p>Mammalian model organisms including: &quot;Human&quot;, &quot;Mouse&quot;,
&quot;Pig&quot;, &quot;Bovine&quot;, &quot;Rat&quot;, &quot;Mammalia&quot;, &quot;Rabbit&quot;, &quot;Dog&quot;, &quot;Hamster&quot;,
and &quot;MINK&quot;. Defaults to &quot;Mouse&quot;.</p>
</td></tr>
<tr><td><code id="getCPX_+3A_tpath">tpath</code></td>
<td>
<p>A character string indicating the path to the project
directory. If the directory is
missing, it will be stored in the temp directory.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>getCPX
</p>


<h3>Value</h3>

<p>A list containing protein complexes for mammalian organisms.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>

<hr>
<h2 id='impute_MissingData'>Impute missing Values in Elution Profile Matrix</h2><span id='topic+impute_MissingData'></span>

<h3>Description</h3>

<p>This function imputes missing values in protein elution
profile matrix via average of adjacent rows. This function is
not applicable for missing values present in the first or last column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute_MissingData(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="impute_MissingData_+3A_x">x</code></td>
<td>
<p>A data matrix with rows
including proteins and fractions along the columns, while some
fractions may contain missing values.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>impute_MissingData
</p>


<h3>Value</h3>

<p>Imputed matrix.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the co-elution data
data("exampleData")
# Replace the values with NAs in the 10th column
exampleData[, 10] &lt;- NA
# Impute missing value
datOut &lt;- impute_MissingData(exampleData)
</code></pre>

<hr>
<h2 id='keepMT'>Keep Mitochondrial (mt) Proteins</h2><span id='topic+keepMT'></span>

<h3>Description</h3>

<p>This function removes all the non-mitochondrial proteins by
mapping the co-eluted proteins from chromatography fractions to
MitoCarta database.
Note that this function is only applicable to mouse or human organisms.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>keepMT(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="keepMT_+3A_x">x</code></td>
<td>
<p>A data matrix  object with rows
including proteins and fractions along the columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>keepMT
</p>


<h3>Value</h3>

<p>Matrix containing mt proteins.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the co-elution data
data("exampleData")
# Removes non-mitochondtial proteins
datOut &lt;- keepMT(exampleData)
</code></pre>

<hr>
<h2 id='MCL_clustering'>MCL clustering</h2><span id='topic+MCL_clustering'></span>

<h3>Description</h3>

<p>This function applies MCL clustering to further refine
the predicted subnetworks produced by ClusterONE.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MCL_clustering(hc_ppi, predcpx, inflation = 9, csize = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MCL_clustering_+3A_hc_ppi">hc_ppi</code></td>
<td>
<p>High-confidence interactions data containing
id1-id2-weight triplets.</p>
</td></tr>
<tr><td><code id="MCL_clustering_+3A_predcpx">predcpx</code></td>
<td>
<p>A data.frame containing predicted complexes resulted from
<code><a href="#topic+get_clusters">get_clusters</a></code>.</p>
</td></tr>
<tr><td><code id="MCL_clustering_+3A_inflation">inflation</code></td>
<td>
<p>MCL inflation parameter. Defaults to 9.</p>
</td></tr>
<tr><td><code id="MCL_clustering_+3A_csize">csize</code></td>
<td>
<p>An integer, the minimum size of the predicted complexes.
Defaults to 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MCL_clustering
</p>


<h3>Value</h3>

<p>List of refined complexes.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh
</p>


<h3>Examples</h3>

<pre><code class='language-R'># open high-confidence network
hc_ppi &lt;-
read.delim(
system.file("extdata/ppi_input_ClusterONE.txt", package = "MACP"),
header = FALSE)
# predict complexes by ClusterONE
predcpx &lt;-
get_clusters(csize = 3, d = 0.3, p = 2,
max_overlap = 0.8,
tpath = file.path(system.file("extdata", package = "MACP")))
# Break down big complexes by MCL
MCL_clusters &lt;- MCL_clustering(hc_ppi, predcpx, inflation = 4, csize = 2)
</code></pre>

<hr>
<h2 id='MCL_tuning'>MCL Hyperparameters Tuning</h2><span id='topic+MCL_tuning'></span>

<h3>Description</h3>

<p>This function optimize the choice of MCL algorithm
parameter (inflation) by comparing clustering-derived partitions for each
paramter values to known labels (i.e., CORUM complexes) and
assess the similarity between them using quality measures including
overlap score, sensitivity (Sn),
clustering-wise positive predictive value (PPV), geometric accuracy (Acc),
and maximum matching raio (MMR). It is recommended to first reduce
redundancy in the known reference complexes
via <code><a href="#topic+EliminateCpxRedundance">EliminateCpxRedundance</a></code>, then performs parameter tuning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MCL_tuning(hc_ppi, predcpx, refcpx, inflation = c(6, 8, 9), csize = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MCL_tuning_+3A_hc_ppi">hc_ppi</code></td>
<td>
<p>Interactions data containing id1-id2-weight triplets.</p>
</td></tr>
<tr><td><code id="MCL_tuning_+3A_predcpx">predcpx</code></td>
<td>
<p>A data.frame containing predicted modules resulted from
<code><a href="#topic+get_clusters">get_clusters</a></code>.</p>
</td></tr>
<tr><td><code id="MCL_tuning_+3A_refcpx">refcpx</code></td>
<td>
<p>A list containing reference complexes
(i.e., corum complexes).</p>
</td></tr>
<tr><td><code id="MCL_tuning_+3A_inflation">inflation</code></td>
<td>
<p>A vector of integer, representing
MCL inflation parameter</p>
</td></tr>
<tr><td><code id="MCL_tuning_+3A_csize">csize</code></td>
<td>
<p>An integer, the minimum size of the predicted complexes.
Defaults to 2.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MCL_tuning
</p>


<h3>Value</h3>

<p>A data.frame containing clustering performance across different
inflation values.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>

<hr>
<h2 id='orthMappingCpx'>Protein Complex Ortholog Mapping</h2><span id='topic+orthMappingCpx'></span>

<h3>Description</h3>

<p>This function uses <code><a href="orthogene.html#topic+convert_orthologs">convert_orthologs</a></code>
function to support ortholog mapping of protein complexes between any pair
of 700+ species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>orthMappingCpx(
  datInput,
  input_species,
  output_species,
  input_taxid,
  output_taxid
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="orthMappingCpx_+3A_datinput">datInput</code></td>
<td>
<p>A list containing reference complexes (i.e.,
CORUM complexes). Note that the members of each complexes must be
represented by UniProt accession identifier.</p>
</td></tr>
<tr><td><code id="orthMappingCpx_+3A_input_species">input_species</code></td>
<td>
<p>Name of the input species (e.g., &quot;mouse&quot;,&quot;fly&quot;).
See <code><a href="orthogene.html#topic+map_species">map_species</a></code> to return a full list of
available species.</p>
</td></tr>
<tr><td><code id="orthMappingCpx_+3A_output_species">output_species</code></td>
<td>
<p>Name of the output species (e.g., &quot;human&quot;).
See <code><a href="orthogene.html#topic+map_species">map_species</a></code> to return a full list of
available species.</p>
</td></tr>
<tr><td><code id="orthMappingCpx_+3A_input_taxid">input_taxid</code></td>
<td>
<p>A numeric value that specifies the NCBI
taxonomy identifier (TaxId) for input organism (e.g., 10090).</p>
</td></tr>
<tr><td><code id="orthMappingCpx_+3A_output_taxid">output_taxid</code></td>
<td>
<p>A numeric value that specifies the NCBI
taxonomy identifier (TaxId) for output organism.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>orthMappingCpx
</p>


<h3>Value</h3>

<p>A list containing complexes, whose members
converted to output_species.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>

<hr>
<h2 id='predPPI_MACP'>Predict Protein-Protein Interactions and Putative Complexes</h2><span id='topic+predPPI_MACP'></span>

<h3>Description</h3>

<p>This function first begins by executing several
pre-processing steps to improve the quality of the raw data, followed
by computing similarity between protein pairs using their co-elution
profiles. Computed features and and class labels generated
from reference complexes are then fed into an individual or
ensemble of ML classifiers.These models then generate a weighted protein
interaction network in which edge weights between protein nodes represent
the ML model's probability estimate for interaction.
High-confidence PPIs resulted from ROC-curve cutoff analysis is then
denoised and finally are partitioned via two-stage clustering,
first by ClusterONE,then by MCL clustering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predPPI_MACP(
  data,
  refcpx,
  tpath = tempdir(),
  data_processing = TRUE,
  data_imputing = TRUE,
  scaling = TRUE,
  keepMT = FALSE,
  pcc = TRUE,
  PCCN = TRUE,
  pcc_p = TRUE,
  spearman = TRUE,
  kendall = TRUE,
  bicor = TRUE,
  weighted_rank = TRUE,
  cosine = TRUE,
  jaccard = TRUE,
  dice = TRUE,
  apex = TRUE,
  minfo = TRUE,
  bayesian = TRUE,
  wcc = TRUE,
  euclidean = TRUE,
  manhattan = TRUE,
  canberra = TRUE,
  avg.distance = TRUE,
  rept = 10,
  corr_removal = FALSE,
  corr_cutoff = 0.5,
  classifier = c("glm", "svmRadial", "ranger"),
  verboseIter = TRUE,
  cv_fold = 5,
  plots = FALSE,
  subcellular_mtPPI = FALSE,
  organism = "mouse",
  csize = 3,
  d = 0.3,
  p = 2,
  max_overlap = 0.8,
  inflation = 9
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="predPPI_MACP_+3A_data">data</code></td>
<td>
<p>A data matrix with rows including proteins and fractions
along the columns. see <code><a href="#topic+exampleData">exampleData</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_refcpx">refcpx</code></td>
<td>
<p>A list of known reference complexes.
see <code><a href="#topic+getCPX">getCPX</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_tpath">tpath</code></td>
<td>
<p>A character string indicating the path to the project
directory. If the directory is
missing, it will be stored in the Temp directory.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_data_processing">data_processing</code></td>
<td>
<p>If TRUE, removes proteins for which peptide only
detected in one fraction (i.e., &quot;one-hit-wonders&quot;) across the
co-elution table, common contaminants (e.g., keratins) only for mouse
and human organisms and frequent flyers. Defaults to TRUE.
See <code><a href="#topic+data_filtering">data_filtering</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_data_imputing">data_imputing</code></td>
<td>
<p>if TRUE, imputes missing values in protein elution
profile matrix via average of adjacent rows. This function is
not applicable for missing values present in the first or last column.
Defaults to TRUE. See <code><a href="#topic+impute_MissingData">impute_MissingData</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_scaling">scaling</code></td>
<td>
<p>If TRUE, performs column and row-wise normalization.
Defaults to TRUE. See <code><a href="#topic+scaling">scaling</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_keepmt">keepMT</code></td>
<td>
<p>if TRUE, removes all the non-mitochondrial proteins by
mapping the co-eluted proteins from chromatography fractions to MitoCarta
database. Note that this function is only applicable to
mouse or human organisms.Defaults to FALSE. See <code><a href="#topic+keepMT">keepMT</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_pcc">pcc</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using Pearson correlation metric. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_pccn">PCCN</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity using
Pearson correlation plus noise. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_pcc_p">pcc_p</code></td>
<td>
<p>If TRUE, computes P-value of the Pearson correlation.
Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_spearman">spearman</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity using
spearman correlation. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_kendall">kendall</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity using
kendall correlation. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_bicor">bicor</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity using
biweight midcorrealtion (bicor) correlation. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_weighted_rank">weighted_rank</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using weighted rank measure. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_cosine">cosine</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using cosine metric. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_jaccard">jaccard</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using jaccard metric. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_dice">dice</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using dice measure. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_apex">apex</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using apex. Defaults to TRUE. See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_minfo">minfo</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using mutual information. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_bayesian">bayesian</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity using
Bayes correlation based on zero-count distribution. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_wcc">wcc</code></td>
<td>
<p>If TRUE, computes pairwise protein profile similarity
using weighted cross correlation. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_euclidean">euclidean</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using euclidean measure. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_manhattan">manhattan</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using manhattan measure. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_canberra">canberra</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using canberra measure. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_avg.distance">avg.distance</code></td>
<td>
<p>if TRUE, computes pairwise protein profile similarity
using avg.distance measure. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_rept">rept</code></td>
<td>
<p>Poisson iterations, defaults to 10. Defaults to TRUE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_corr_removal">corr_removal</code></td>
<td>
<p>If TRUE, removes protein pairs with
correlation scores &lt; the user defined threshold; defaults to FALSE.
See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_corr_cutoff">corr_cutoff</code></td>
<td>
<p>user defined threshold for correlation similarity
scores. Defaults to 0.5.See <code><a href="#topic+calculate_PPIscore">calculate_PPIscore</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_classifier">classifier</code></td>
<td>
<p>The type of classifier to use for ensemble or
individual model. See <code>caret</code> for the available classifiers.
Defaults to c(&quot;glm&quot;, &quot;svmRadial&quot;, &quot;ranger&quot;).
See <code><a href="#topic+ensemble_model">ensemble_model</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_verboseiter">verboseIter</code></td>
<td>
<p>Logical value, indicating whether to check the status
of training process;defaults to FALSE. See <code><a href="#topic+ensemble_model">ensemble_model</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_cv_fold">cv_fold</code></td>
<td>
<p>Number of partitions for cross-validation; defaults to 5.
See <code><a href="#topic+ensemble_model">ensemble_model</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_plots">plots</code></td>
<td>
<p>Logical value, indicating whether to plot the performance
of the learning algorithm using k-fold cross-validation;
defaults to FALSE. These plots are :
</p>
 <ul>
<li><p>pr_plot - Precision-recall PLOT
</p>
</li>
<li><p>roc_plot - ROC plot
</p>
</li>
<li><p>point_plot - Point plot showing accuracy,
F1-score , positive predictive value (PPV), sensitivity (SE) and MCC.</p>
</li></ul>

<p>See <code><a href="#topic+ensemble_model">ensemble_model</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_subcellular_mtppi">subcellular_mtPPI</code></td>
<td>
<p>if TRUE, removes PPIs occurring between outer mt
membrane (OMM) and matrix, between intermembrane space (IMS) and matrix,
as well as between any subcellular mt compartment (except OMM) and
cytosolic proteins as they deemed to be erroneous. Defaults to FALSE.
See <code><a href="#topic+subcellular.mtPPI">subcellular.mtPPI</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_organism">organism</code></td>
<td>
<p>Organism under study (i.e., mouse or human).
Defaults to mouse. See <code><a href="#topic+subcellular.mtPPI">subcellular.mtPPI</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_csize">csize</code></td>
<td>
<p>An integer, the minimum size of the predicted complexes.
Defaults to 2. See <code><a href="#topic+get_clusters">get_clusters</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_d">d</code></td>
<td>
<p>A number, density of predicted complexes. Defaults to 0.3.
See <code><a href="#topic+get_clusters">get_clusters</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_p">p</code></td>
<td>
<p>An integer, penalty value for the inclusion of each node.
Defaults to 2.
See <code><a href="#topic+get_clusters">get_clusters</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_max_overlap">max_overlap</code></td>
<td>
<p>A number, specifies the maximum allowed
overlap between two clusters. Defaults to 0.8.
See <code><a href="#topic+get_clusters">get_clusters</a></code>.</p>
</td></tr>
<tr><td><code id="predPPI_MACP_+3A_inflation">inflation</code></td>
<td>
<p>MCL inflation parameter. Defaults to 9.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>predPPI_MACP
</p>


<h3>Value</h3>

<p>Return following data sets in the current directory including:
</p>

<ul>
<li><p>unfilteredPPIs - Unfiltered interactions
</p>
</li>
<li><p>filteredPPI - High-confidence interactions
defined by ROC threshold.
</p>
</li>
<li><p>High_confidence interactions_with_mt_sublocalization - if
subcellular.mtPPI is TRUE, it return high-confidene PPIs with mt
sublocalization status.
</p>
</li>
<li><p>predicted_cpx_clusterONE - Putative complexes generated
by clusterONE.
</p>
</li>
<li><p>predicted_cpx_clusterONE_MCL - Putative complexes generated
by clusterONE and MCL.
</p>
</li>
<li><p>Best_roc_curve_cutoff - Best cutoff generated from ROC curve.</p>
</li></ul>


<hr>
<h2 id='refcpx'>CORUM reference complexes</h2><span id='topic+refcpx'></span>

<h3>Description</h3>

<p>A list containing CORUM reference complexes for mouse
organism.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(refcpx)
</code></pre>


<h3>Source</h3>

<p><a href="https://mips.helmholtz-muenchen.de/corum/">https://mips.helmholtz-muenchen.de/corum/</a>
</p>

<hr>
<h2 id='scaling'>Column and Row-wise Normalization</h2><span id='topic+scaling'></span>

<h3>Description</h3>

<p>This function performs column and row-wise normalization.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scaling(data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="scaling_+3A_data">data</code></td>
<td>
<p>A data matrix with rows
including proteins and fractions along the columns.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>scaling
</p>


<h3>Value</h3>

<p>Scaled data matrix.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load the co-elution data
data("exampleData")
# Normalize the data
datOut &lt;- scaling(exampleData)
</code></pre>

<hr>
<h2 id='subcellular.mtPPI'>Keep Mitochondrial (mt) Proteins</h2><span id='topic+subcellular.mtPPI'></span>

<h3>Description</h3>

<p>This function removes PPIs occurring between outer mt
membrane (OMM) and matrix, between intermembrane space (IMS) and matrix,
as well as between any subcellular mt compartment (except OMM) and
cytosolic proteins as they deemed to be erroneous
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subcellular.mtPPI(ppi, organism = "mouse")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subcellular.mtPPI_+3A_ppi">ppi</code></td>
<td>
<p>Interactions data containing id1-id2-weight triplets.</p>
</td></tr>
<tr><td><code id="subcellular.mtPPI_+3A_organism">organism</code></td>
<td>
<p>Organism under study (i.e., mouse or human).
Defaults to mouse.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>subcellular.mtPPI
</p>


<h3>Value</h3>

<p>Filtered PPI netwrok.
</p>


<h3>Author(s)</h3>

<p>Matineh Rahmatbakhsh, <a href="mailto:matinerb.94@gmail.com">matinerb.94@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ppi &lt;-
read.table(system.file("extdata/ppi_input_ClusterONE.txt",
package="MACP"),
quote="\"", comment.char="")
filtered_mtEdges &lt;- subcellular.mtPPI(ppi)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
