<!DOCTYPE html><html><head><title>Help for package macrosyntR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {macrosyntR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#compute_linkage_groups'><p>Compute Linkage groups</p></a></li>
<li><a href='#compute_macrosynteny'><p>Compute significant macrosynteny blocks</p></a></li>
<li><a href='#get_syntenic_genes'><p>get the syntenic genes as a table</p></a></li>
<li><a href='#load_orthologs'><p>load orthologs with their genomic coordinates.</p></a></li>
<li><a href='#plot_chord_diagram'><p>plot the Macro-synteny as a chord diagram</p></a></li>
<li><a href='#plot_macrosynteny'><p>Plot Macro-synteny</p></a></li>
<li><a href='#plot_oxford_grid'><p>plot the Macro-synteny as an oxford grid.</p></a></li>
<li><a href='#reorder_macrosynteny'><p>Reorder the mbh_df before plotting</p></a></li>
<li><a href='#reorder_multiple_macrosyntenies'><p>Reorder the chromosomes of two or more species before plotting</p></a></li>
<li><a href='#reverse_species_order'><p>Reverse order of the species in an orthologs_df.</p></a></li>
<li><a href='#subset_linkage_orthologs'><p>Subset Orthologs contained in conserved linkage groups</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Draw Ordered Oxford Grids and Chord Diagrams</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sami El Hilali &lt;elhilali.sami@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, ggplot2, igraph, tidyr, reshape2, dplyr,
stringr, rlang</td>
</tr>
<tr>
<td>Description:</td>
<td>Use standard genomics file format (BED) and a table of orthologs to 
 illustrate synteny conservation at the genome-wide scale. 
 Significantly conserved linkage groups are identified as described in Simakov et al. (2020) &lt;<a href="https://doi.org/10.1038%2Fs41559-020-1156-z">doi:10.1038/s41559-020-1156-z</a>&gt; 
 and displayed on an Oxford Grid (Edwards (1991) &lt;<a href="https://doi.org/10.1111%2Fj.1469-1809.1991.tb00394.x">doi:10.1111/j.1469-1809.1991.tb00394.x</a>&gt;) or a chord diagram as in Simakov et al. (2022) &lt;<a href="https://doi.org/10.1126%2Fsciadv.abi5884">doi:10.1126/sciadv.abi5884</a>&gt;. 
 The package provides a function that uses a network-based greedy algorithm to find communities (Clauset et al. (2004) &lt;<a href="https://doi.org/10.1103%2FPhysRevE.70.066111">doi:10.1103/PhysRevE.70.066111</a>&gt;) 
 and so automatically order the chromosomes on the plot to improve interpretability.</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/SamiLhll/macrosyntR">https://github.com/SamiLhll/macrosyntR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/SamiLhll/macrosyntR/issues">https://github.com/SamiLhll/macrosyntR/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-14 15:05:34 UTC; sami</td>
</tr>
<tr>
<td>Author:</td>
<td>Sami El Hilali <a href="https://orcid.org/0000-0003-4417-8399"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Richard Copley <a href="https://orcid.org/0000-0001-7846-4954"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-14 16:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='compute_linkage_groups'>Compute Linkage groups</h2><span id='topic+compute_linkage_groups'></span>

<h3>Description</h3>

<p>This is a function to compute the conserved linkage groups shared between two or more species.
It computes the significant associations between chromosomes of all species versus all (pairwise)
using the fischer test implemented in compute_macrosynteny().
It outputs a dataframe shaped as following :
sp1.Chr,sp2.Chr,..., spN.chr,n,LGs
where n is the number of shared orthologs in the group and LGs are the IDs for the linkage groups
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_linkage_groups(orthologs_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_linkage_groups_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded with load_orthologs()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe object
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of compute_linkage_groups: 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_macrosynteny &lt;- compute_linkage_groups(my_orthologs)

</code></pre>

<hr>
<h2 id='compute_macrosynteny'>Compute significant macrosynteny blocks</h2><span id='topic+compute_macrosynteny'></span>

<h3>Description</h3>

<p>This is a function to generate the contingency table of an orthologs dataframe and
apply fischer test to calculate the significant associations. It outputs a dataframe shaped as following :
sp1.Chr,sp2.Chr,a,pval,significant,pval_adj
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compute_macrosynteny(orthologs_df, pvalue_threshold = 0.001)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compute_macrosynteny_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded with load_orthologs()</p>
</td></tr>
<tr><td><code id="compute_macrosynteny_+3A_pvalue_threshold">pvalue_threshold</code></td>
<td>
<p>numeric. threshold for significancy. (default equals 0.001)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe object
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of compute_macrosynteny : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_macrosynteny &lt;- compute_macrosynteny(my_orthologs)

</code></pre>

<hr>
<h2 id='get_syntenic_genes'>get the syntenic genes as a table</h2><span id='topic+get_syntenic_genes'></span>

<h3>Description</h3>

<p>This is a function to extract all the syntenic genes from an orthologs_df. It requires as input an orthologs_df loaded by load_orthologs().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_syntenic_genes(orthologs_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_syntenic_genes_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded by load_orthologs()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe composed of details for each detected syntenic block of genes. It contains the following columns : sp1.Chr, sp1.Start, sp1.End, sp2.Chr, sp2.Start, sp2.End, size, sp1.IDs, sp2.IDs
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of get_syntenic_genes :

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_syntenic_block_of_genes &lt;- get_syntenic_genes(my_orthologs)

</code></pre>

<hr>
<h2 id='load_orthologs'>load orthologs with their genomic coordinates.</h2><span id='topic+load_orthologs'></span>

<h3>Description</h3>

<p>Puts together the table of orthologous genes with their genomic coordinates
in the two or more species. It outputs a data.frame shaped as following :
sp1.ID,sp1.Chr,sp1.Start,sp1.End,sp1.Index,sp2.ID,sp2.Chr,sp2.Start,sp2.End,sp2.Index,...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_orthologs(
  orthologs_table,
  sp1_bed = NULL,
  sp2_bed = NULL,
  bedfiles = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_orthologs_+3A_orthologs_table">orthologs_table</code></td>
<td>
<p>character. Full path to the orthologs table (format : geneID_on_species1   geneID_on_species2   geneID_on_speciesN)</p>
</td></tr>
<tr><td><code id="load_orthologs_+3A_sp1_bed">sp1_bed</code></td>
<td>
<p>(deprecated) character. Full path to the genomic coordinates of the genes on species1</p>
</td></tr>
<tr><td><code id="load_orthologs_+3A_sp2_bed">sp2_bed</code></td>
<td>
<p>(deprecated) character. Full path to the genomic coordinates of the genes on species2</p>
</td></tr>
<tr><td><code id="load_orthologs_+3A_bedfiles">bedfiles</code></td>
<td>
<p>array. List of full paths to the genomic coordinates ordered as in the appearing order of the orthologs_table (BED format)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe composed of genomic coordinates and relative index of orthologs on both species
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of load_orthologs for two species :

orthologs_file &lt;- system.file("extdata","Bflo_vs_Pyes.tab",package="macrosyntR")
bedfile_sp1 &lt;- system.file("extdata","Bflo.bed",package="macrosyntR")
bedfile_sp2 &lt;- system.file("extdata","Pyes.bed",package="macrosyntR")


my_orthologs &lt;- load_orthologs(orthologs_table = orthologs_file,
                               bedfiles = c(bedfile_sp1,bedfile_sp2))
# example with 3 species :
orthologs_file &lt;- system.file("extdata","Single_copy_orthologs.tsv",package="macrosyntR")
bedfile_sp3 &lt;- system.file("extdata","Pech.bed",package="macrosyntR")

my_orthologs &lt;- load_orthologs(orthologs_table = orthologs_file,
                               bedfiles = c(bedfile_sp1,bedfile_sp2,bedfile_sp3))


</code></pre>

<hr>
<h2 id='plot_chord_diagram'>plot the Macro-synteny as a chord diagram</h2><span id='topic+plot_chord_diagram'></span>

<h3>Description</h3>

<p>This is a function to plot the chord diagrams to compare the macro synteny of two or more species.
It requires as input an orthologs_df loaded by load_orthologs()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_chord_diagram(
  orthologs_df,
  species_labels = NULL,
  species_labels_size = 5,
  color_by = "sp1.Chr",
  custom_color_palette = NULL,
  reorder_chromosomes = TRUE,
  remove_non_linkage_orthologs = TRUE,
  species_labels_hpos = -400,
  label_size = 2,
  ideogram_fill = "white",
  ideogram_color = "black",
  ideogram_height = 4,
  ribbons_curvature = 0.1,
  ribbons_alpha = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_chord_diagram_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded by the load_orthologs()</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_species_labels">species_labels</code></td>
<td>
<p>list of characters. names of the species to display on the plot</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_species_labels_size">species_labels_size</code></td>
<td>
<p>integer. size of the labels (default = 2)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_color_by">color_by</code></td>
<td>
<p>string. name of the column in the orthologs_df to color the links by (default = &quot;sp1.Chr&quot;)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_custom_color_palette">custom_color_palette</code></td>
<td>
<p>list of characters. palette to use for the coloring of the links following the argument color_by</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_reorder_chromosomes">reorder_chromosomes</code></td>
<td>
<p>logical. (default = TRUE) tells whether to reorder the chromosomes in clusters as implemented in reorder_macrosynteny()</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_remove_non_linkage_orthologs">remove_non_linkage_orthologs</code></td>
<td>
<p>logical. (default = TRUE) tells wether to remove the orthologs that are not within significant linkage groups as calculated by comput_linkage_groups().</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_species_labels_hpos">species_labels_hpos</code></td>
<td>
<p>(default =-400)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_label_size">label_size</code></td>
<td>
<p>integer. size of the labels to display on the ideograms (default = 2)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_ideogram_fill">ideogram_fill</code></td>
<td>
<p>character. name of the colors to fill the ideograms with (default = &quot;white&quot;)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_ideogram_color">ideogram_color</code></td>
<td>
<p>character. name of the colors to draw the borders of the ideograms with (default = &quot;black&quot;)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_ideogram_height">ideogram_height</code></td>
<td>
<p>integer. height of the ideograms (default = 4)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_ribbons_curvature">ribbons_curvature</code></td>
<td>
<p>float. curvature of the ribbons (default = 0.1)</p>
</td></tr>
<tr><td><code id="plot_chord_diagram_+3A_ribbons_alpha">ribbons_alpha</code></td>
<td>
<p>float. alpha of the ribbons (default = 0.5)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>
<p><code><a href="#topic+reorder_macrosynteny">reorder_macrosynteny()</a></code>
</p>
<p><code><a href="#topic+compute_linkage_groups">compute_linkage_groups()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of plot_oxford_grid : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

plot_chord_diagram(my_orthologs,species_labels = c("B. flo","P. ech"))

</code></pre>

<hr>
<h2 id='plot_macrosynteny'>Plot Macro-synteny</h2><span id='topic+plot_macrosynteny'></span>

<h3>Description</h3>

<p>This is a function to generate the contingency table of an MBH dataframe and apply fischer test to calculate the significant associations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_macrosynteny(macrosynt_df, sp1_label = "", sp2_label = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_macrosynteny_+3A_macrosynt_df">macrosynt_df</code></td>
<td>
<p>dataframe of contingency table with p-values calculated by the compute_macrosynteny() function</p>
</td></tr>
<tr><td><code id="plot_macrosynteny_+3A_sp1_label">sp1_label</code></td>
<td>
<p>character. The name of the species1 to display on the plot</p>
</td></tr>
<tr><td><code id="plot_macrosynteny_+3A_sp2_label">sp2_label</code></td>
<td>
<p>character. The name of the species2 to put on the plot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+compute_macrosynteny">compute_macrosynteny()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of plot_macrosynteny : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_macrosynteny &lt;- compute_macrosynteny(my_orthologs)

plot_macrosynteny(my_macrosynteny,
                  sp1_label = "B. floridae",
                  sp2_label = "P. yessoensis")

</code></pre>

<hr>
<h2 id='plot_oxford_grid'>plot the Macro-synteny as an oxford grid.</h2><span id='topic+plot_oxford_grid'></span>

<h3>Description</h3>

<p>This is a function to plot the oxford grided plot to compare the macro synteny of two species. It requires as input an orthologs_df loaded by load_orthologs()
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_oxford_grid(
  orthologs_df,
  sp1_label = "",
  sp2_label = "",
  dot_size = 0.5,
  dot_alpha = 0.4,
  reorder = FALSE,
  keep_only_significant = FALSE,
  color_by = NULL,
  pvalue_threshold = 0.001,
  color_palette = NULL,
  shade_non_significant = TRUE,
  reverse_species = FALSE,
  keep_sp1_raw_order = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_oxford_grid_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded by the load_orthologs()</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_sp1_label">sp1_label</code></td>
<td>
<p>character. name of 1st species to display on the plot</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_sp2_label">sp2_label</code></td>
<td>
<p>character. name of 2nd species to display on the plot</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_dot_size">dot_size</code></td>
<td>
<p>numeric. (default = 0.5)</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_dot_alpha">dot_alpha</code></td>
<td>
<p>numeric. (default = 0.4)</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_reorder">reorder</code></td>
<td>
<p>logical. (default = FALSE) tells whether to reorder the chromosomes in clusters as implemented in reorder_macrosynteny()</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_keep_only_significant">keep_only_significant</code></td>
<td>
<p>logical. (default = FALSE)</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_color_by">color_by</code></td>
<td>
<p>string/variable name. (default = NULL) column of the orthologs_df to use to color the dots.</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_pvalue_threshold">pvalue_threshold</code></td>
<td>
<p>numeric. (default = 0.001)</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_color_palette">color_palette</code></td>
<td>
<p>vector. (default = NULL) list of colors (as string under double quote) for the clusters. The amount of colors must match the amount of clusters.</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_shade_non_significant">shade_non_significant</code></td>
<td>
<p>logical. (default = TRUE) When TRUE the orthologs located on non-significant linkage groups are displayed in &quot;grey&quot;</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_reverse_species">reverse_species</code></td>
<td>
<p>logical. (default = FALSE) When TRUE the x and y axis of the plot are reversed. sp1 is displayed on the y axis and sp2 is displayed on the x axis.</p>
</td></tr>
<tr><td><code id="plot_oxford_grid_+3A_keep_sp1_raw_order">keep_sp1_raw_order</code></td>
<td>
<p>logical.(default equals FALSE) tells if the reordering should be constrained on the species1 and change just the order of the species2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>
<p><code><a href="#topic+reorder_macrosynteny">reorder_macrosynteny()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of plot_oxford_grid : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

plot_oxford_grid(my_orthologs,
                 sp1_label = "B. floridae",
                 sp2_label = "P. echinospica")

# plot a reordered Oxford Grid and color by cluster :

plot_oxford_grid(my_orthologs,
                 sp1_label = "B. floridae",
                 sp2_label = "P. echinospica",
                 reorder = TRUE,
                 color_by = "clust")
 
</code></pre>

<hr>
<h2 id='reorder_macrosynteny'>Reorder the mbh_df before plotting</h2><span id='topic+reorder_macrosynteny'></span>

<h3>Description</h3>

<p>This is a function to reorder an orthologs_df, that was generated with load_orthologs(). It retrieves communities using igraph::cluster_fast_greedy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reorder_macrosynteny(
  orthologs_df,
  pvalue_threshold = 0.001,
  keep_only_significant = FALSE,
  keep_sp1_raw_order = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reorder_macrosynteny_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. mutual best hits with genomic coordinates loaded with load_orthologs()</p>
</td></tr>
<tr><td><code id="reorder_macrosynteny_+3A_pvalue_threshold">pvalue_threshold</code></td>
<td>
<p>numeric. threshold for significancy. (default equals 0.001)</p>
</td></tr>
<tr><td><code id="reorder_macrosynteny_+3A_keep_only_significant">keep_only_significant</code></td>
<td>
<p>logical. (default equals FALSE) tells if the non significant linkage groups should be removed. It drastically speeds up the computation when using one highly fragmented genome.</p>
</td></tr>
<tr><td><code id="reorder_macrosynteny_+3A_keep_sp1_raw_order">keep_sp1_raw_order</code></td>
<td>
<p>logical. (default equals FALSE) tells if the reordering should be constrained on the species1 and change just the order of the species2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>
<p><code><a href="#topic+compute_macrosynteny">compute_macrosynteny()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of reorder_macrosynteny : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

my_orthologs_reordered &lt;- reorder_macrosynteny(my_orthologs)

</code></pre>

<hr>
<h2 id='reorder_multiple_macrosyntenies'>Reorder the chromosomes of two or more species before plotting</h2><span id='topic+reorder_multiple_macrosyntenies'></span>

<h3>Description</h3>

<p>This is a function to reorder an orthologs_df, same as reorder_macrosynteny,
but it handles tables with more than 2 species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reorder_multiple_macrosyntenies(orthologs_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reorder_multiple_macrosyntenies_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded with load_orthologs()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>
<p><code><a href="#topic+compute_macrosynteny">compute_macrosynteny()</a></code>
</p>
<p><code><a href="#topic+reorder_macrosynteny">reorder_macrosynteny()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of reorder_macrosynteny : 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

my_orthologs_reordered &lt;- reorder_multiple_macrosyntenies(my_orthologs)

</code></pre>

<hr>
<h2 id='reverse_species_order'>Reverse order of the species in an orthologs_df.</h2><span id='topic+reverse_species_order'></span>

<h3>Description</h3>

<p>Returns an orthologs_df (data.frame) with reversed species order compared to the inputted orthologs_df.
sp1 becomes sp2 and the otherway around. It intends at facilitating the integration of more than just two datasets.
It outputs a data.frame shaped as following :
sp1.ID,sp1.Chr,sp1.Start,sp1.End,sp1.Index,sp2.ID,sp2.Chr,sp2.Start,sp2.End,sp2.Index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reverse_species_order(orthologs_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reverse_species_order_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>orthologs_df dataframe. mutual best hits with genomic coordinates loaded with load_orthologs()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dataframe composed of genomic coordinates and relative index of orthologs on both species
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of reverse_species_order :

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)

my_orthologs_reversed &lt;- reverse_species_order(my_orthologs)

</code></pre>

<hr>
<h2 id='subset_linkage_orthologs'>Subset Orthologs contained in conserved linkage groups</h2><span id='topic+subset_linkage_orthologs'></span>

<h3>Description</h3>

<p>This is a function to subset an orthologs_df and keep only the orthologs that are within significant
linkage groups computed by the function compute_linkage_groups().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_linkage_orthologs(orthologs_df, linkages = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_linkage_orthologs_+3A_orthologs_df">orthologs_df</code></td>
<td>
<p>dataframe. orthologs with genomic coordinates loaded with load_orthologs()</p>
</td></tr>
<tr><td><code id="subset_linkage_orthologs_+3A_linkages">linkages</code></td>
<td>
<p>dataframe. table listing the linkage groups as returned by the function compute_linkage_groups()</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe object
</p>


<h3>See Also</h3>

<p><code><a href="#topic+load_orthologs">load_orthologs()</a></code>
</p>
<p><code><a href="#topic+compute_linkage_groups">compute_linkage_groups()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># basic usage of compute_linkage_groups: 

orthologs_table &lt;- system.file("extdata","my_orthologs.tab",package="macrosyntR")

my_orthologs &lt;- read.table(orthologs_table,header=TRUE)
                               
my_macrosynteny &lt;- compute_linkage_groups(my_orthologs)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
