<!DOCTYPE html><html lang="en"><head><title>Help for package MADPop</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MADPop}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MADPop-package'><p>(M)HC (A)llele-Based (D)ifferencing between (Pop)ulations</p></a></li>
<li><a href='#chi2.stat'><p>Chi-squared test statistic for contingency tables</p></a></li>
<li><a href='#colSort'><p>Matrix Sort</p></a></li>
<li><a href='#fish215'><p>Genotypes of lake trout from Ontario, Canada</p></a></li>
<li><a href='#geno.format'><p>Format genotype matrix</p></a></li>
<li><a href='#hUM.post'><p>Posterior sampling from a hierarchical Unconstrained-Multinomial model</p></a></li>
<li><a href='#LRT.stat'><p>Likelihood ratio test statistic for contingency tables</p></a></li>
<li><a href='#UM.eqtest'><p>Equality tests for two multinomial samples</p></a></li>
<li><a href='#UM.suff'><p>Sufficient statistics for the Unconstrained-Multinomial model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>MHC Allele-Based Differencing Between Populations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-26</td>
</tr>
<tr>
<td>Description:</td>
<td>Tools for the analysis of population differences
    using the Major Histocompatibility Complex (MHC) genotypes of samples
    having a variable number of alleles (1-4) recorded for each
    individual.  A hierarchical Dirichlet-Multinomial model on the
    genotype counts is used to pool small samples from multiple
    populations for pairwise tests of equality.  Bayesian inference is
    implemented via the 'rstan' package.  Bootstrapped and posterior
    p-values are provided for chi-squared and likelihood ratio tests of
    equal genotype probabilities.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mlysy/MADPop">https://github.com/mlysy/MADPop</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mlysy/MADPop/issues">https://github.com/mlysy/MADPop/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), rstan (&ge; 2.26.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>methods, Rcpp (&ge; 0.12.0), stats, rstantools</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>BH (&ge; 1.66.0), Rcpp (&ge; 0.12.0), RcppEigen (&ge; 0.3.3.3.0),
rstan (&ge; 2.26.0), StanHeaders (&ge; 2.26.0), RcppParallel (&ge;
5.0.1)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-26 19:36:43 UTC; mlysy</td>
</tr>
<tr>
<td>Author:</td>
<td>Martin Lysy [cre, aut],
  Peter W.J. Kim [aut],
  Terin Robinson [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Martin Lysy &lt;mlysy@uwaterloo.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-27 00:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='MADPop-package'>(M)HC (A)llele-Based (D)ifferencing between (Pop)ulations</h2><span id='topic+MADPop'></span><span id='topic+MADPop-package'></span>

<h3>Description</h3>

<p>Tools for the analysis of population differences using the Major Histocompatibility Complex (MHC) genotypes of samples having a variable number of alleles (1-4) recorded for each individual.
</p>


<h3>Details</h3>

<p>For a full tutorial see package vignette: <code>vignette("MADPop-quicktut")</code>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Martin Lysy <a href="mailto:mlysy@uwaterloo.ca">mlysy@uwaterloo.ca</a>
</p>
<p>Authors:
</p>

<ul>
<li><p> Peter W.J. Kim
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Terin Robinson [contributor]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/mlysy/MADPop">https://github.com/mlysy/MADPop</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/mlysy/MADPop/issues">https://github.com/mlysy/MADPop/issues</a>
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># typical dataset
head(fish215[sample(nrow(fish215)),])
table(fish215$Lake) # number of samples per lake

# contingency table on two lakes
iLakes &lt;- c("Michipicoten", "Simcoe")
Xsuff &lt;- UM.suff(X = fish215[fish215$Lake %in% iLakes,])
ctab &lt;- Xsuff$tab
ctab

# bootstrapped p-value calculation for chi^2 and LR tests
p.MLE &lt;- colSums(ctab)/sum(ctab)
N1 &lt;- sum(ctab[1,])
N2 &lt;- sum(ctab[2,])
# bootstrapped test statistics (chi^2 and LRT)
T.boot &lt;- UM.eqtest(N1 = N1, N2 = N2, p0 = p.MLE, nreps = 1e3)

# observed test statistics
T.obs &lt;- c(chi2 = chi2.stat(ctab), LRT = LRT.stat(ctab))
# p-values
rowMeans(t(T.boot) &gt; T.obs)

# posterior sampler for hierarchical model

# output posterior probability for each genotype in lake Simcoe
rhoId &lt;- "Simcoe"
nsamples &lt;- 500
hUM.fit &lt;- hUM.post(nsamples = nsamples, X = fish215,
                    rhoId = rhoId, chains = 1)

# first 20 genotype posterior probabilities in lake Simcoe
rho.post &lt;- hUM.fit$rho[,1,]
boxplot(rho.post[,1:20], las = 2,
        xlab = "Genotype", ylab = "Posterior Probability",
        pch = ".", col = "grey")
</code></pre>

<hr>
<h2 id='chi2.stat'>Chi-squared test statistic for contingency tables</h2><span id='topic+chi2.stat'></span>

<h3>Description</h3>

<p>Calculates the chi-squared test statistic for a two-way contingency table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chi2.stat(tab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chi2.stat_+3A_tab">tab</code></td>
<td>
<p>A <code>K x C</code> matrix (contingency table) of counts. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose that <code>tab</code> consists of counts from <code class="reqn">K</code> populations (rows) in <code class="reqn">C</code> categories.  The chi-squared test statistic is computed as
</p>
<p style="text-align: center;"><code class="reqn">
  \sum_{i=1}^K \sum_{j=1}^C (E_{ij} - O_{ij})^2/E_{ij},
</code>
</p>

<p>where <code class="reqn">O_{ij}</code> is the observed number of counts in the <code class="reqn">i</code>th row and <code class="reqn">j</code>th column of <code>tab</code>, and <code class="reqn">E_{ij}</code> is the expected number of counts under <code class="reqn">H_0</code> that the populations have indentical proportions in each category:
</p>
<p style="text-align: center;"><code class="reqn">
  E_{ij} = \frac 1 N \sum_{i=1}^K O_{ij} \times \sum_{j=1}^C O_{ij}.
</code>
</p>

<p>where <code class="reqn">N</code> is the total number of counts in <code>tab</code>.
</p>


<h3>Value</h3>

<p>The calculated value of the chi-squared statistic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simple contingency table
ctab &lt;- rbind(pop1 = c(5, 3, 0, 3),
                pop2 = c(4, 10, 2, 5))
colnames(ctab) &lt;- LETTERS[1:4]
ctab
chi2.stat(ctab) # chi^2 test statistic
</code></pre>

<hr>
<h2 id='colSort'>Matrix Sort</h2><span id='topic+colSort'></span>

<h3>Description</h3>

<p>Sorts a matrix by first column, breaking ties with second column, breaking those ties with 3rd, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>colSort(X)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="colSort_+3A_x">X</code></td>
<td>
<p>matrix to sort.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Same matrix with rows permuted according to sort order.
</p>

<hr>
<h2 id='fish215'>Genotypes of lake trout from Ontario, Canada</h2><span id='topic+fish215'></span>

<h3>Description</h3>

<p>Observable genotypes (up to possibly duplicated alleles) of 215 lake trout (<em>Salvelinus namaycush</em>) collected from 11 lakes in Ontario, Canada.
</p>


<h3>Format</h3>

<p>A <code>data.frame</code> with 215 rows and 5 columns.  The first column is an (optional) vector of population identifiers.  The next four columns contain the recorded genotype for each observation (row).  Each row contains up to four distinct allele identifiers in any order.  Missing alleles should be denoted by <code>NA</code>, or &quot;&quot;, but not both.
</p>


<h3>Details</h3>

<p>This <code>data.frame</code> is how a typical spreadsheet of genotype data gets imported into <strong>R</strong>.  Data must adhere to this format to be correctly processed by the functions in <span class="pkg">MADPop</span>.
</p>


<h3>Source</h3>

<p>Kuntz, S. (2014). <em>Population Differentiation of Ontario Lake trout (Salvelinus namaycush) using the Major Histocompatibility Complex class II <code class="reqn">\beta</code> gene</em> (<a href="https://uwspace.uwaterloo.ca/bitstream/handle/10012/8542/Kuntz_Steven.pdf?sequence=1">URL</a>).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>head(fish215)
</code></pre>

<hr>
<h2 id='geno.format'>Format genotype matrix</h2><span id='topic+geno.format'></span>

<h3>Description</h3>

<p>Convert a 4-column genotype matrix or data.frame into a simplified numerical format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>geno.format(X, Y.only = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="geno.format_+3A_x">X</code></td>
<td>
<p>Genotype matrix or data.frame.</p>
</td></tr>
<tr><td><code id="geno.format_+3A_y.only">Y.only</code></td>
<td>
<p>Logical.  Whether or not to output only <code>Y</code> (see return).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements
</p>

<p><code>Y</code>: A 4-column numeric matrix.  Each row is sorted with zeros indicating missing alleles.
<code>A</code>: Vector of allele names.
<code>G</code>: Unique allele combinations, i.e. <code>Y</code> without duplicated rows.

</p>

<hr>
<h2 id='hUM.post'>Posterior sampling from a hierarchical Unconstrained-Multinomial model</h2><span id='topic+hUM.post'></span>

<h3>Description</h3>

<p>MCMC sampling from a Dirichlet-Multinomial model using <code><a href="rstan.html#topic+stan">stan</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hUM.post(nsamples, X, popId, rhoId, full.stan.out = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="hUM.post_+3A_nsamples">nsamples</code></td>
<td>
<p>Number of posterior samples</p>
</td></tr>
<tr><td><code id="hUM.post_+3A_x">X</code></td>
<td>
<p>4-column or 5-column matrix of observations in the correct format.  See <code><a href="#topic+UM.suff">UM.suff</a></code>.</p>
</td></tr>
<tr><td><code id="hUM.post_+3A_popid">popId</code></td>
<td>
<p>Optional vector of population identifiers.  See <code><a href="#topic+UM.suff">UM.suff</a></code>.</p>
</td></tr>
<tr><td><code id="hUM.post_+3A_rhoid">rhoId</code></td>
<td>
<p>Populations for which posterior samples of the genotype probability vector <code>rho</code> are desired.  Defaults to all populations.  Set <code>rhoId = NULL</code> not to output these for any populations.</p>
</td></tr>
<tr><td><code id="hUM.post_+3A_full.stan.out">full.stan.out</code></td>
<td>
<p>Logical.  Whether or not to return the full <code>stan</code> output.  For monitoring convergence of the MCMC sampling.</p>
</td></tr>
<tr><td><code id="hUM.post_+3A_...">...</code></td>
<td>
<p>Further arguments to be passed to the <code><a href="rstan.html#topic+sampling">sampling</a></code> function in <span class="pkg">rstan</span>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The hierarchical Dirichlet-Multinomial model is given by
</p>
<p style="text-align: center;"><code class="reqn">
  Y_k \mid \rho_k \sim_{\textrm{ind}} \textrm{Multinomial}(\rho_k, N_k),
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  \rho_k \sim_{\textrm{iid}} \textrm{Dirichlet}(\alpha).
</code>
</p>

<p>where <code class="reqn">\alpha_0 = \sum_{i=1}^C \alpha_i</code> and <code class="reqn">\bar \alpha = \alpha/\alpha_0</code>.  MCMC sampling is achieved with the <span class="pkg">rstan</span> package, which is listed as a dependency for <span class="pkg">MADPop</span> so as to expose <span class="pkg">rstan</span>'s sophisticated tuning mechanism and convergence diagnostics.
</p>


<h3>Value</h3>

<p>A list with elements
</p>

<ul>
<li> <p><code>A</code>: The unique allele names.
</p>
</li>
<li> <p><code>G</code>: The 4-column matrix Package libcurl was not found in the pkg-config search path.of unique genotype combinations.
</p>
</li>
<li> <p><code>rho</code>: A matrix with <code>ncol(rho) == nrow(G)</code>, where each row is a draw from the posterior distribution of inheritance probabilities.
</p>
</li>
<li> <p><code>sfit</code>: If <code>full.stan.out = TRUE</code>, the fitted <code>stan</code> object.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># fit hierarchical model to fish215 data

# only output posterior samples for lake Simcoe
rhoId &lt;- "Simcoe"
nsamples &lt;- 500
hUM.fit &lt;- hUM.post(nsamples = nsamples, X = fish215,
                    rhoId = rhoId,
                    chains = 1) # number of MCMC chains

# plot first 20 posterior probabilities in lake Simcoe
rho.post &lt;- hUM.fit$rho[,1,]
boxplot(rho.post[,1:20], las = 2,
        xlab = "Genotype", ylab = "Posterior Probability",
        pch = ".", col = "grey")
</code></pre>

<hr>
<h2 id='LRT.stat'>Likelihood ratio test statistic for contingency tables</h2><span id='topic+LRT.stat'></span>

<h3>Description</h3>

<p>Calculate the likelihood ratio test statistic for general two-way contingency tables.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LRT.stat(tab)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="LRT.stat_+3A_tab">tab</code></td>
<td>
<p>A <code>K x C</code> matrix (contingency table) of counts. See details.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Suppose that <code>tab</code> consists of counts from <code class="reqn">K</code> populations (rows) in <code class="reqn">C</code> categories.  The likelihood ratio test statistic is computed as
</p>
<p style="text-align: center;"><code class="reqn">
  2 \sum_{i=1}^K \sum_{j=1}^N O_{ij} \log(p^A_{ij}/p^0_{j}),
</code>
</p>

<p>where <code class="reqn">O_{ij}</code> is the observed number of counts in the <code class="reqn">i</code>th row and <code class="reqn">j</code>th column of <code>tab</code>, <code class="reqn">p^A_{ij} = O_{ij}/\sum_{j=1}^C O_{ij}</code> is the unconstrained estimate of the proportion of category <code class="reqn">j</code> in population <code class="reqn">i</code>, and <code class="reqn">p^0_j = \sum_{i=1}^K O_{ij} / \sum_{i=1}^K\sum_{j=1}^C O_{ij}</code> is the estimate of this proportion under <code class="reqn">H_0</code> that the populations have indentical proportions in each category.  If any column has only zeros it is removed before calculating the LRT statistic.
</p>


<h3>Value</h3>

<p>The calculated value of the LRT statistic.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># simple contingency table
ctab &lt;- rbind(pop1 = c(5, 3, 0, 3),
                pop2 = c(4, 10, 2, 5))
colnames(ctab) &lt;- LETTERS[1:4]
ctab
LRT.stat(ctab) # likelihood ratio statistic
</code></pre>

<hr>
<h2 id='UM.eqtest'>Equality tests for two multinomial samples</h2><span id='topic+UM.eqtest'></span>

<h3>Description</h3>

<p>Generate multinomial samples from a common probability vector and calculate the Chi-square and Likelihood Ratio test statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UM.eqtest(N1, N2, p0, nreps, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UM.eqtest_+3A_n1">N1</code></td>
<td>
<p>Size of sample 1.</p>
</td></tr>
<tr><td><code id="UM.eqtest_+3A_n2">N2</code></td>
<td>
<p>Size of sample 2.</p>
</td></tr>
<tr><td><code id="UM.eqtest_+3A_p0">p0</code></td>
<td>
<p>Common probability vector from which to draw the multinomial samples.  Can also be a matrix, in which case each simulation randomly draws with replacement from the rows of p0.</p>
</td></tr>
<tr><td><code id="UM.eqtest_+3A_nreps">nreps</code></td>
<td>
<p>Number of replications of the simulation.</p>
</td></tr>
<tr><td><code id="UM.eqtest_+3A_verbose">verbose</code></td>
<td>
<p>Logical.  If <code>TRUE</code> prints message every <code>5000</code> replications.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The chi-squared and likelihood ratio test statistics are calculated from multinomial samples <code class="reqn">(Y_1^1, Y_2^1), \ldots, (Y_1^M, Y_2^M)</code>, where
</p>
<p style="text-align: center;"><code class="reqn">
  Y_k^m \stackrel{\textrm{ind}}{\sim} \textrm{Multinomial}(N_k, p_0^m),
</code>
</p>

<p>where <code class="reqn">p_0^m</code> is the <code class="reqn">m</code>th row of <code>p0</code>.
</p>


<h3>Value</h3>

<p>An <code>nreps x 2</code> matrix with the simulated chi-squared and LR values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># bootstrapped p-value calculation against equal genotype proportions
# in lakes Michipicoten and Simcoe

# contingency table
popId &lt;- c("Michipicoten", "Simcoe")
ctab &lt;- UM.suff(fish215[fish215$Lake %in% popId,])$tab
ctab

# MLE of probability vector
p.MLE &lt;- colSums(ctab)/sum(ctab)
# sample sizes
N1 &lt;- sum(ctab[1,]) # Michipicoten
N2 &lt;- sum(ctab[2,]) # Simcoe

# bootstrapped test statistics (chi^2 and LRT)
T.boot &lt;- UM.eqtest(N1 = N1, N2 = N2, p0 = p.MLE, nreps = 1e3)

# observed test statistics
T.obs &lt;- c(chi2 = chi2.stat(ctab), LRT = LRT.stat(ctab))
# p-values
rowMeans(t(T.boot) &gt; T.obs)
</code></pre>

<hr>
<h2 id='UM.suff'>Sufficient statistics for the Unconstrained-Multinomial model</h2><span id='topic+UM.suff'></span>

<h3>Description</h3>

<p>Converts a matrix or data.frame of genotype data into the sufficient statistics required to fit a Dirichlet-Multinomial hierarchical model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>UM.suff(X, popId)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="UM.suff_+3A_x">X</code></td>
<td>
<p>Genotype adata.  Either a <code>N x 4</code> matrix with <code>NA</code>'s indicating duplicates or a <code>N x 5</code> column data.frame with the first column being the <code>popId</code>.</p>
</td></tr>
<tr><td><code id="UM.suff_+3A_popid">popId</code></td>
<td>
<p>grouping variable for <code>X</code>. Must be supplied if <code>X</code> has 4 columns.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with elements:
</p>

<ul>
<li> <p><code>A</code>: Vector of unique alleles names.  The allele numbers in the following quantities correspond to the indices of <code>A</code>.
</p>
</li>
<li> <p><code>G</code>: 4-column matrix of unique genotype combinations.  The presence of 0's indicates that less than four alleles were amplified indicating that a given genotype either has less than 4 distinct alleles or that some alleles are duplicated.
</p>
</li>
<li> <p><code>tab</code>: Observed data in a simplified numerical format.  This is a contingency table with rows given by the unique elements of <code>popId</code> and columns given by each row of <code>G</code>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># sufficient statistics in 3 lakes

X &lt;- fish215[fish215$Lake %in% c("Hogan", "Manitou", "Simcoe"),]
suff &lt;- UM.suff(X)

suff$A # allele names
suff$G # unique genotypes
suff$tab # contingency table
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
