<!DOCTYPE html><html><head><title>Help for package MAGEE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MAGEE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MAGEE-package'>
<p>Mixed Model Association Test for GEne-Environment Interaction</p></a></li>
<li><a href='#example'><p>Example dataset</p></a></li>
<li><a href='#glmm.gei'>
<p>GLMM based single variant tests for gene-environment interactions</p></a></li>
<li><a href='#glmm.gei.meta'>
<p>GLMM based meta-analysis of single variant tests for gene-environment interactions</p></a></li>
<li><a href='#MAGEE'>
<p>Mixed model Association tests for GEne-Environment interactions</p></a></li>
<li><a href='#MAGEE.meta'>
<p>Mixed model Association tests for GEne-Environment interactions meta-analysis</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.3.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-11-17</td>
</tr>
<tr>
<td>Title:</td>
<td>Mixed Model Association Test for GEne-Environment Interaction</td>
</tr>
<tr>
<td>Description:</td>
<td>Use a 'glmmkin' class object (GMMAT package) from the null model to perform generalized linear mixed model-based single-variant and variant set main effect tests, gene-environment interaction tests, and joint tests for association, as proposed in Wang et al. (2020) &lt;<a href="https://doi.org/10.1002%2Fgepi.22351">doi:10.1002/gepi.22351</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Copyright:</td>
<td>See COPYRIGHTS for details.</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, Matrix, parallel, MASS, foreach, GMMAT, CompQuadForm,
data.table</td>
</tr>
<tr>
<td>Suggests:</td>
<td>doMC, SeqArray, SeqVarTools, testthat</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Xinyu Wang [aut],
  Han Chen [aut, cre],
  Duy Pham [aut],
  Kenneth Westerman [ctb],
  Cong Pan [aut],
  Eric Biggers [ctb, cph] (Author and copyright holder of included
    libdeflate library),
  Tino Reichardt [ctb, cph] (Author and copyright holder of threading
    code used in the included Zstandard (zstd) library),
  Meta Platforms, Inc. and affiliates [cph] (Copyright holder of included
    Zstandard (zstd) library)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Han Chen &lt;han.chen.2@uth.tmc.edu&gt;</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-17 08:35:35 UTC; hchen</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-17 09:14:00 UTC</td>
</tr>
</table>
<hr>
<h2 id='MAGEE-package'>
Mixed Model Association Test for GEne-Environment Interaction
</h2><span id='topic+MAGEE-package'></span>

<h3>Description</h3>

<p>An R package for performing generalized linear mixed model-based single-variant and variant set main effect tests, gene-environment interaction tests, and joint tests for association, as proposed in Wang et al. (2020) &lt;DOI:10.1002/gepi.22351&gt;.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> MAGEE</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.3.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2023-11-17</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 3)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Xinyu Wang, Han Chen, Duy T. Pham
</p>
<p>Maintainer: Han Chen &lt;Han.Chen.2@uth.tmc.edu&gt;
</p>


<h3>References</h3>

<p>Wang, X., Lim, E., Liu, C, Sung, Y.J., Rao, D.C., Morrison, A.C., Boerwinkle, E., Manning, A. K., and Chen, H. (2020) Efficient gene-environment interaction tests for large biobank-scale sequencing studies. Genetic Epidemiology, 44(8): 908-923.
</p>
<p>Chen, H., Huffman, J.E., Brody, J.A., Wang, C., Lee, S., Li, Z., Gogarten, S.M., Sofer, T., Bielak, L.F., Bis, J.C., et al. (2019) Efficient variant set mixed model association tests for continuous and binary traits in large-scale whole-genome sequencing studies. The American Journal of Human Genetics, 104 (2): 260-274.
</p>
<p>Wang, X., Pham, D.T., Westerman, K.E., Pan, C., Manning, A.K., Chen, H., (2022) Genomic summary statistics and meta-analysis for set-based gene-environment interaction tests in large-scale sequencing studies. medrxiv. 
</p>

<hr>
<h2 id='example'>Example dataset</h2><span id='topic+example'></span>

<h3>Description</h3>

<p>Example dataset for MAGEE.
</p>


<h3>Format</h3>

<p>Contains the following objects:
</p>

<dl>
<dt>pheno</dt><dd><p>a data frame of 400 observations from a cross-sectional study with 5 variables: id, disease, trait, age and sex.</p>
</dd>
<dt>pheno2</dt><dd><p>a data frame of 2,000 observations from a longitudinal study with 400 individuals and 5 variables: id, y.repeated, y.trend, time and sex.</p>
</dd>
<dt>GRM</dt><dd><p>a genetic relationship matrix for 400 observations.</p>
</dd>
</dl>


<hr>
<h2 id='glmm.gei'>
GLMM based single variant tests for gene-environment interactions
</h2><span id='topic+glmm.gei'></span>

<h3>Description</h3>

<p>Use a glmmkin class object from the null GLMM to perform single variant main effect score test, gene-environment interaction test, or joint test for association with genotypes in a GDS file .gds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm.gei(null.obj, interaction, geno.file, outfile, bgen.samplefile = NULL, 
  interaction.covariates = NULL,  meta.output = FALSE, center=TRUE, 
  MAF.range = c(1e-7, 0.5), miss.cutoff = 1, missing.method = "impute2mean", 
  nperbatch=100, ncores = 1, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmm.gei_+3A_null.obj">null.obj</code></td>
<td>

<p>a class glmmkin object, returned by fitting the null GLMM using <code>glmmkin( )</code>.
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_interaction">interaction</code></td>
<td>

<p>a numeric or a character vector indicating the environmental factors. If a numeric vector, it represents which indices in the order of covariates are the environmental factors; if a character vector, it represents the variable names of the environmental factors.
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_geno.file">geno.file</code></td>
<td>

<p>the full name of a GDS file (including the suffix .gds).
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_outfile">outfile</code></td>
<td>

<p>the output file name.
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_bgen.samplefile">bgen.samplefile</code></td>
<td>

<p>path to the BGEN .sample file. Required when the BGEN file does not contain sample identifiers.
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_interaction.covariates">interaction.covariates</code></td>
<td>

<p>a numeric or a character vector indicating the interaction covariates. If a numeric vector, it represents which indices in the order of covariates are the interaction covariates; if a character vector, it represents the variable names of the interaction covariates.
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_meta.output">meta.output</code></td>
<td>

<p>boolean value to modiy the output file.If TRUE, the GxE effect estimate and variance and covariance associated with the effect estimate are included in the output file. (default = FALSE)
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_center">center</code></td>
<td>

<p>a logical switch for centering genotypes before tests. If TRUE, genotypes will be centered to have mean 0 before tests, otherwise raw values will be directly used in tests (default = TRUE).
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_maf.range">MAF.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum minor allele frequencies of variants that should be included in the analysis (default = c(1e-7, 0.5)).
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_miss.cutoff">miss.cutoff</code></td>
<td>

<p>the maximum missing rate allowed for a variant to be included (default = 1, including all variants).
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_missing.method">missing.method</code></td>
<td>

<p>method of handling missing genotypes.Either &quot;impute2mean&quot; or &quot;omit&quot; (default = &quot;impute2mean&quot;).
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_nperbatch">nperbatch</code></td>
<td>

<p>an integer for how many SNPs should be tested in a batch (default = 100). The computational time can increase dramatically if this value is either small or large. The optimal value for best performance depends on the user's system.
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_ncores">ncores</code></td>
<td>

<p>a positive integer indicating the number of cores to be used in parallel computing (default = 1).
</p>
</td></tr>
<tr><td><code id="glmm.gei_+3A_verbose">verbose</code></td>
<td>

<p>a logical switch for whether failed matrix inversions should be written to <code>outfile</code>.err for debugging (default = FALSE).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL
</p>


<h3>Author(s)</h3>

<p>Xinyu Wang, Han Chen, Duy Pham
</p>


<h3>References</h3>

<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celedón, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control forpopulation structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  library(GMMAT)
  data(example)
  attach(example)

  model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM,
                    id = "id", family = binomial(link = "logit"))
                   
  if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
    quietly = TRUE)) {
    infile &lt;- system.file("extdata", "geno.gds", package = "MAGEE")
    gds_outfile &lt;- tempfile()
    glmm.gei(model0, interaction='sex', geno.file = infile, outfile = gds_outfile)
    unlink(gds_outfile)
  }
  infile &lt;- system.file("extdata", "geno.bgen", package = "MAGEE")
  samplefile &lt;- system.file("extdata", "geno.sample", package = "MAGEE")
  bgen_outfile &lt;- tempfile()
  glmm.gei(model0, interaction='sex', geno.file = infile, outfile = bgen_outfile,
           bgen.samplefile = samplefile)
  unlink(bgen_outfile)
</code></pre>

<hr>
<h2 id='glmm.gei.meta'>
GLMM based meta-analysis of single variant tests for gene-environment interactions
</h2><span id='topic+glmm.gei.meta'></span>

<h3>Description</h3>

<p>Use a glmmkin class object from the null GLMM to perform meta-analysis of single variant main effect score test, gene-environment interaction test, or joint test for association with genotypes in a GDS file .gds.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmm.gei.meta(files, outfile, interaction, SNPID = rep("SNPID", length(files)), 
              CHR = rep("CHR", length(files)), POS = rep("POS", length(files)), 
              Non_Effect_Allele = rep("Non_Effect_Allele", length(files)), 
              Effect_Allele = rep("Effect_Allele", length(files))) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmm.gei.meta_+3A_files">files</code></td>
<td>

<p>tab or space delimited plain text files (or compressed files that can be recognized by the R function read.table) with at least the following columns: SNPID, CHR, POS, Non_Effect_Allele, Effect_Allele, N_Samples, AF, Beta_Marginal, SE_Beta_Marginal, P_Value_Marginal, Beta_G, Beta_G_sex, SE_Beta_G, SE_Beta_G_sex, Cov_Beta_G_G.sex, P_Value_Interaction, P_Value_Joint. Generally, if each study performs score tests using genotypes in PLINK binary PED format or GDS format, the score test output from glmm.score can be directly used as input files. 
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_outfile">outfile</code></td>
<td>

<p>the output file name.
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_interaction">interaction</code></td>
<td>

<p>a numeric or a character vector indicating the environmental factors. If a numeric vector, it represents which indices in the order of covariates are the environmental factors; if a character vector, it represents the variable names of the environmental factors.
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_snpid">SNPID</code></td>
<td>

<p>a character vector of SNPID column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;SNPID&quot;, length(files))).
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_chr">CHR</code></td>
<td>

<p>a character vector of CHR column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;CHR&quot;, length(files))).
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_pos">POS</code></td>
<td>

<p>a character vector of POS column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;POS&quot;, length(files))).
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_non_effect_allele">Non_Effect_Allele</code></td>
<td>

<p>a character vector of Non_Effect_Allele column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;Non_Effect_Allele&quot;, length(files))).
</p>
</td></tr>
<tr><td><code id="glmm.gei.meta_+3A_effect_allele">Effect_Allele</code></td>
<td>

<p>a character vector of Effect_Allele column names in each input file. The length and order must match the length and order of <code>files</code> (default = rep(&quot;Effect_Allele&quot;, length(files))).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing the following:
</p>
<table>
<tr><td><code>SNPID</code></td>
<td>
<p>SNP name.</p>
</td></tr>
<tr><td><code>CHR</code></td>
<td>
<p>chromosome.</p>
</td></tr>
<tr><td><code>POS</code></td>
<td>
<p>physical position.</p>
</td></tr>
<tr><td><code>Non_Effect_Allele</code></td>
<td>
<p>non_effect allele frequency.</p>
</td></tr>
<tr><td><code>Effect_Allele</code></td>
<td>
<p>effect allele frequency.</p>
</td></tr>
<tr><td><code>N_Samples</code></td>
<td>
<p>number of samples.</p>
</td></tr>
<tr><td><code>AF</code></td>
<td>
<p>allele frequency.</p>
</td></tr>
<tr><td><code>Beta_Marginal</code></td>
<td>
<p>coefficient estimate for the marginal genetic effect.</p>
</td></tr>
<tr><td><code>SE_Beta_Marginal</code></td>
<td>
<p>standard error of the marginal genetic effect.</p>
</td></tr>
<tr><td><code>P_Value_Marginal</code></td>
<td>
<p>marginal effect score test p-value.</p>
</td></tr>
<tr><td><code>Beta_G</code></td>
<td>
<p>coefficient estimate for the genetic main effect.</p>
</td></tr>
<tr><td><code>Beta_G-*</code></td>
<td>
<p>coefficient estimate for the interaction terms.</p>
</td></tr>
<tr><td><code>SE_Beta_G</code></td>
<td>
<p>model-based standard error associated with the the genetic main effect.</p>
</td></tr>
<tr><td><code>SE_Beta_G-*</code></td>
<td>
<p>mdel-based standard error associated with any GxE or interaction covariate terms.</p>
</td></tr>
<tr><td><code>Cov_Beta_G_G-*</code></td>
<td>
<p>model-based covariance between the genetic main effect and any GxE or interaction covariate terms.</p>
</td></tr>
<tr><td><code>P_Value_Interaction</code></td>
<td>
<p>gene-environment interaction test p-value.</p>
</td></tr>
<tr><td><code>P_Value_Joint</code></td>
<td>
<p>joint test p-value.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xinyu Wang, Han Chen, Duy Pham
</p>


<h3>References</h3>

<p>Chen, H., Wang, C., Conomos, M.P., Stilp, A.M., Li, Z., Sofer, T., Szpiro, A.A., Chen, W., Brehm, J.M., Celedón, J.C., Redline, S., Papanicolaou, G.J., Thornton, T.A., Laurie, C.C., Rice, K. and Lin, X. (2016) Control forpopulation structure and relatedness for binary traits in genetic association studies via logistic mixed models. The American Journal of Human Genetics 98, 653-666.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
  infile1 &lt;- system.file("extdata", "meta1.txt", package = "MAGEE")
  infile2 &lt;- system.file("extdata", "meta2.txt", package = "MAGEE")
  infile3 &lt;- system.file("extdata", "meta3.txt", package = "MAGEE")
  infile4 &lt;- system.file("extdata", "meta4.txt", package = "MAGEE")
  infile5 &lt;- system.file("extdata", "meta5.txt", package = "MAGEE")
  outfile &lt;- tempfile()
  glmm.gei.meta(files = c(infile1, infile2, infile3, infile4, infile5),
		outfile = outfile, interaction="sex")
  unlink(outfile)

</code></pre>

<hr>
<h2 id='MAGEE'>
Mixed model Association tests for GEne-Environment interactions
</h2><span id='topic+MAGEE'></span><span id='topic+MAGEE.prep'></span><span id='topic+MAGEE.lowmem'></span>

<h3>Description</h3>

<p>Use a glmmkin class object from the null GLMM to perform variant set-based main effect tests, gene-environment interaction tests, and joint tests for association with genotypes in a GDS file (.gds). 7 user-defined tests are included: Main effect variance component test (MV), Main effect hybrid test of burden and variance component test using Fisher's method (MF), Interaction variance component test (IV), Interaction hybrid test of burden and variance component test using Fisher's method (IF), Joint variance component test (JV), Joint hybrid test of burden and variance component test using Fisher's method (JF), and Joint hybrid test of burden and variance component test using double Fisher's procedures (JD). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAGEE(null.obj, interaction, geno.file, group.file,  group.file.sep = "\t", 
      bgen.samplefile = NULL, interaction.covariates = NULL, meta.file.prefix = NULL,
      MAF.range = c(1e-7, 0.5), AF.strata.range = c(0, 1), MAF.weights.beta = c(1, 25), 
      miss.cutoff = 1, missing.method = "impute2mean", method = "davies", tests = "JF", 
      use.minor.allele = FALSE, auto.flip = FALSE, 
      Garbage.Collection = FALSE, is.dosage = FALSE, ncores = 1)

MAGEE.prep(null.obj, interaction, geno.file, group.file, interaction.covariates = NULL,
           group.file.sep = "\t", auto.flip = FALSE)
           
MAGEE.lowmem(MAGEE.prep.obj, geno.file = NULL, meta.file.prefix = NULL, 
             MAF.range = c(1e-7, 0.5), AF.strata.range = c(0, 1), 
             MAF.weights.beta = c(1, 25), miss.cutoff = 1, missing.method = "impute2mean",
             method = "davies", tests = "JF", use.minor.allele = FALSE, 
	     Garbage.Collection = FALSE, is.dosage = FALSE, ncores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAGEE_+3A_null.obj">null.obj</code></td>
<td>

<p>a class glmmkin object, returned by fitting the null GLMM using <code>glmmkin( )</code>.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_interaction">interaction</code></td>
<td>

<p>a numeric or a character vector indicating the environmental factors. If a numberic vector, it represents which indices in the order of covariates are the environmental factors; if a character vector, it represents the variable names of the environmental factors.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_geno.file">geno.file</code></td>
<td>

<p>a .gds file for the full genotypes. The <code>sample.id</code> in <code>geno.file</code> should overlap <code>id_include</code> in <code>null.obj</code>. It is recommended that <code>sample.id</code> in <code>geno.file</code> include the full samples (at least all samples as specified in <code>id_include</code> of <code>null.obj</code>). It is not necessary for the user to take a subset of <code>geno.file</code> before running the analysis.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_group.file">group.file</code></td>
<td>

<p>a plain text file with 6 columns defining the test units. There should be no headers in the file, and the columns are group name, chromosome, position, reference allele, alternative allele and weight, respectively.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_group.file.sep">group.file.sep</code></td>
<td>

<p>the delimiter in group.file (default = <code>"\t"</code>).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_bgen.samplefile">bgen.samplefile</code></td>
<td>

<p>path to the BGEN .sample file. Required when the BGEN file does not contain sample identifiers.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_interaction.covariates">interaction.covariates</code></td>
<td>

<p>a numeric or a character vector indicating the interaction covariates. If a numeric vector, it represents which indices in the order of covariates are the interaction covariates; if a character vector, it represents the variable names of the interaction covariates.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_meta.file.prefix">meta.file.prefix</code></td>
<td>

<p>the prefix for meta-analysis (default = <code>"NULL"</code>).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_maf.range">MAF.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum minor allele frequencies of variants that should be included in the analysis (default = c(1e-7, 0.5)).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_af.strata.range">AF.strata.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum coding allele frequencies of variants in each stratum that should be included in the analysis, if the environmental factor is categorical (default = c(0, 1)).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_maf.weights.beta">MAF.weights.beta</code></td>
<td>

<p>a numeric vector of length 2 defining the beta probability density function parameters on the minor allele frequencies. This internal minor allele frequency weight is multiplied by the external weight given by the group.file. To turn off internal minor allele frequency weight and only use the external weight given by the group.file, use c(1, 1) to assign flat weights (default = c(1, 25)).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_miss.cutoff">miss.cutoff</code></td>
<td>

<p>the maximum missing rate allowed for a variant to be included (default = 1, including all variants).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_missing.method">missing.method</code></td>
<td>

<p>method of handling missing genotypes. Either &quot;impute2mean&quot; or &quot;impute2zero&quot; (default = &quot;impute2mean&quot;).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_method">method</code></td>
<td>

<p>a method to compute p-values for the test statistics (default = &quot;davies&quot;). &quot;davies&quot; represents an exact method that computes a p-value by inverting the characteristic function of the mixture chisq distribution, with an accuracy of 1e-6. When &quot;davies&quot; p-value is less than 1e-5, it defaults to method &quot;kuonen&quot;. &quot;kuonen&quot; represents a saddlepoint approximation method that computes the tail probabilities of the mixture chisq distribution. When &quot;kuonen&quot; fails to compute a p-value, it defaults to method &quot;liu&quot;. &quot;liu&quot; is a moment-matching approximation method for the mixture chisq distribution.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_tests">tests</code></td>
<td>

<p>a character vector indicating which MAGEE tests should be performed (&quot;MV&quot; for the main effect variance component test, &quot;MF&quot; for the main effect combined test of the burden and variance component tests using Fisher's method, &quot;IV&quot; for the interaction variance component test, &quot;IF&quot; for the interaction combined test of the burden and variance component tests using Fisher's method, &quot;JV&quot; for the joint variance component test for main effect and interaction, &quot;JF&quot; for the joint combined test of the burden and variance component tests for main effect and interaction using Fisher's method, or &quot;JD&quot; for the joint combined test of the burden and variance component tests for main effect and interaction using double Fisher's method.). The &quot;MV&quot; and &quot;IV&quot; test are automatically included when performing &quot;JV&quot;, and the &quot;MF&quot; and &quot;IF&quot; test are automatically included when performing &quot;JF&quot; or &quot;JD&quot; (default = &quot;JF&quot;).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_use.minor.allele">use.minor.allele</code></td>
<td>

<p>a logical switch for whether to use the minor allele (instead of the alt allele) as the coding allele (default = FALSE). It does not change &quot;MV&quot;, &quot;IV&quot;, and &quot;JV&quot; results, but &quot;MF&quot;, &quot;IF&quot;, &quot;JF&quot;, and &quot;JD&quot; results will be affected.Along with the MAF filter, this option is useful for combining rare mutations, assuming rare allele effects are in the same direction.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_auto.flip">auto.flip</code></td>
<td>

<p>a logical switch for whether to enable automatic allele flipping if a variant with alleles ref/alt is not found at a position, but a variant at the same position with alleles alt/ref is found (default = FALSE). Use with caution for whole genome sequence data, as both ref/alt and alt/ref variants at the same position are not uncommon, and they are likely two different variants, rather than allele flipping.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_garbage.collection">Garbage.Collection</code></td>
<td>

<p>a logical switch for whether to enable garbage collection in each test (default = FALSE). Pay for memory efficiency with slower computation speed.
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_is.dosage">is.dosage</code></td>
<td>

<p>a logical switch (default = FALSE).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_ncores">ncores</code></td>
<td>

<p>a positive integer indicating the number of cores to be used in parallel computing (default = 1).
</p>
</td></tr>
<tr><td><code id="MAGEE_+3A_magee.prep.obj">MAGEE.prep.obj</code></td>
<td>

<p>a class MAGEE.prep object, returned by <code>MAGEE.prep</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame with the following components:
</p>
<table>
<tr><td><code>group</code></td>
<td>

<p>name of the test unit group.
</p>
</td></tr>
<tr><td><code>n.variants</code></td>
<td>

<p>number of variants in the test unit group that pass the missing rate and allele frequency filters.
</p>
</td></tr>
<tr><td><code>miss.min</code></td>
<td>

<p>minimum missing rate for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>miss.mean</code></td>
<td>

<p>mean missing rate for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>miss.max</code></td>
<td>

<p>maximum missing rate for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.min</code></td>
<td>

<p>minimum coding allele frequency for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.mean</code></td>
<td>

<p>mean coding allele frequency for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.max</code></td>
<td>

<p>maximum coding allele frequency for variants in the test unit group.
</p>
</td></tr>
<tr><td><code>freq.strata.min</code></td>
<td>

<p>minimum coding allele frequency of each stratum if the environmental factor is categorical.
</p>
</td></tr>
<tr><td><code>freq.strata.max</code></td>
<td>

<p>maximum coding allele frequency of each stratum if the environmental factor is categorical.
</p>
</td></tr>
<tr><td><code>MV.pval</code></td>
<td>

<p>MV test p-value.
</p>
</td></tr>
<tr><td><code>MF.pval</code></td>
<td>

<p>MF test p-value.
</p>
</td></tr>
<tr><td><code>IV.pval</code></td>
<td>

<p>IV test p-value.
</p>
</td></tr>
<tr><td><code>IF.pval</code></td>
<td>

<p>IF test p-value.
</p>
</td></tr>
<tr><td><code>JV.pval</code></td>
<td>

<p>JV test p-value.
</p>
</td></tr>
<tr><td><code>JF.pval</code></td>
<td>

<p>JF test p-value.
</p>
</td></tr>
<tr><td><code>JD.pval</code></td>
<td>

<p>JD test p-value.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xinyu Wang, Han Chen, Duy Pham
</p>


<h3>References</h3>

<p>Wang, X., Lim, E., Liu, C, Sung, Y.J., Rao, D.C., Morrison, A.C., Boerwinkle, E., Manning, A. K., and Chen, H. (2020) Efficient gene-environment interaction tests for large biobank-scale sequencing studies. Genetic Epidemiology, 44(8): 908-923.
Chen, H., Huffman, J.E., Brody, J.A., Wang, C., Lee, S., Li, Z., Gogarten, S.M., Sofer, T., Bielak, L.F., Bis, J.C., et al. (2019) Efficient variant set mixed model association tests for continuous and binary traits in large-scale whole-genome sequencing studies. The American Journal of Human Genetics, 104 (2): 260-274.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
  quietly = TRUE)) {
  library(GMMAT)
  data(example)
  attach(example)
  model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM,
    id = "id", family = binomial(link = "logit"))
  geno.file &lt;- system.file("extdata", "geno.gds", package = "MAGEE")
  group.file &lt;- system.file("extdata", "SetID.withweights.txt", package = "MAGEE")
  out &lt;- MAGEE(model0, interaction='sex', geno.file, group.file, group.file.sep = "\t",
    tests=c("JV", "JF", "JD"))
  print(out)
}

</code></pre>

<hr>
<h2 id='MAGEE.meta'>
Mixed model Association tests for GEne-Environment interactions meta-analysis
</h2><span id='topic+MAGEE.meta'></span>

<h3>Description</h3>

<p>Use a glmmkin class object from the null GLMM to perform meta-analysis of variant set-based main effect tests, gene-environment interaction tests, and joint tests for association with genotypes in a GDS file (.gds). 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAGEE.meta(meta.files.prefix, n.files = rep(1, length(meta.files.prefix)), 
            cohort.group.idx = NULL, group.file, group.file.sep = "\t", 
            E.match=NULL, MAF.range = c(1e-7, 0.5), 
            MAF.weights.beta = c(1, 25), miss.cutoff = 1, method = "davies", 
            tests = "JF", use.minor.allele = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MAGEE.meta_+3A_meta.files.prefix">meta.files.prefix</code></td>
<td>

<p>a character vector for prefix of intermediate files (.score.* and .var.*) required in a meta-analysis. Each element represents the prefix of .score.* and .var.* from one cohort. The length of vector should be equal to the number of cohorts.
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_n.files">n.files</code></td>
<td>

<p>an integer vector with the same length as meta.files.prefix, indicating how many sets of intermediate files (.score.* and .var.*) are expected from each cohort, usually as the result of multi-threading in creating the intermediate files (default = rep(1, length(meta.files.prefix))).
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_cohort.group.idx">cohort.group.idx</code></td>
<td>

<p>a vector with the same length as meta.files.prefix, indicating which cohorts should be grouped together in the meta-analysis assuming homogeneous genetic effects. For example, c(&quot;a&quot;,&quot;b&quot;,&quot;a&quot;,&quot;a&quot;,&quot;b&quot;) means cohorts 1, 3, 4 are assumed to have homogeneous genetic effects, and cohorts 2, 5 are in another group with homogeneous genetic effects (but possibly heterogeneous with group &quot;a&quot;). If NULL, all cohorts are in the same group (default = NULL).
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_group.file">group.file</code></td>
<td>

<p>a plain text file with 6 columns defining the test units. There should be no headers in the file, and the columns are group name, chromosome, position, reference allele, alternative allele and weight, respectively.
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_group.file.sep">group.file.sep</code></td>
<td>

<p>the delimiter in group.file (default = &quot;\t&quot;).
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_e.match">E.match</code></td>
<td>

<p>environmental factors that match the interactions (default = &quot;\t&quot;).
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_maf.range">MAF.range</code></td>
<td>

<p>a numeric vector of length 2 defining the minimum and maximum minor allele frequencies of variants that should be included in the analysis (default = c(1e-7, 0.5)). Filter applied to the combined samples.
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_maf.weights.beta">MAF.weights.beta</code></td>
<td>

<p>a numeric vector of length 2 defining the beta probability density function parameters on the minor allele frequencies. This internal minor allele frequency weight is multiplied by the external weight given by the group.file. To turn off internal minor allele frequency weight and only use the external weight given by the group.file, use c(1, 1) to assign flat weights (default = c(1, 25)). Applied to the combined samples.
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_miss.cutoff">miss.cutoff</code></td>
<td>

<p>the maximum missing rate allowed for a variant to be included (default = 1, including all variants). Filter applied to the combined samples.
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_method">method</code></td>
<td>

<p>a method to compute p-values for SKAT-type test statistics (default = &quot;davies&quot;). &quot;davies&quot; represents an exact method that computes a p-value by inverting the characteristic function of the mixture chisq distribution, with an accuracy of 1e-6. When &quot;davies&quot; p-value is less than 1e-5, it defaults to method &quot;kuonen&quot;. &quot;kuonen&quot; represents a saddlepoint approximation method that computes the tail probabilities of the mixture chisq distribution. When &quot;kuonen&quot; fails to compute a p-value, it defaults to method &quot;liu&quot;. &quot;liu&quot; is a moment-matching approximation method for the mixture chisq distribution.
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_tests">tests</code></td>
<td>

<p>a character vector indicating which MAGEE tests should be performed (&quot;MV&quot; for the main effect variance component test, &quot;MF&quot; for the main effect combined test of the burden and variance component tests using Fisher's method, &quot;IV&quot; for the interaction variance component test, &quot;IF&quot; for the interaction combined test of the burden and variance component tests using Fisher's method, &quot;JV&quot; for the joint variance component test for main effect and interaction, &quot;JF&quot; for the joint combined test of the burden and variance component tests for main effect and interaction using Fisher's method, or &quot;JD&quot; for the joint combined test of the burden and variance component tests for main effect and interaction using double Fisher's method.). The &quot;MV&quot; and &quot;IV&quot; test are automatically included when performing &quot;JV&quot;, and the &quot;MF&quot; and &quot;IF&quot; test are automatically included when performing &quot;JF&quot; or &quot;JD&quot; (default = &quot;JF&quot;).
</p>
</td></tr>
<tr><td><code id="MAGEE.meta_+3A_use.minor.allele">use.minor.allele</code></td>
<td>

<p>a logical switch for whether to use the minor allele (instead of the alt allele) as the coding allele (default = FALSE). It does not change &quot;MV&quot;, &quot;IV&quot;, and &quot;JV&quot; results, but &quot;MF&quot;, &quot;IF&quot;, &quot;JF&quot;, and &quot;JD&quot; results will be affected. Along with the MAF filter, this option is useful for combining rare mutations, assuming rare allele effects are in the same direction.
</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>group</code></td>
<td>

<p>name of the test unit group.
</p>
</td></tr>
<tr><td><code>n.variants</code></td>
<td>

<p>number of variants in the test unit group that pass the missing rate and allele frequency filters.
</p>
</td></tr>
<tr><td><code>MV.pval</code></td>
<td>

<p>MV test p-value.
</p>
</td></tr>
<tr><td><code>MF.pval</code></td>
<td>

<p>MF test p-value.
</p>
</td></tr>
<tr><td><code>IV.pval</code></td>
<td>

<p>IV test p-value.
</p>
</td></tr>
<tr><td><code>IF.pval</code></td>
<td>

<p>IF test p-value.
</p>
</td></tr>
<tr><td><code>JV.pval</code></td>
<td>

<p>JV test p-value.
</p>
</td></tr>
<tr><td><code>JF.pval</code></td>
<td>

<p>JF test p-value.
</p>
</td></tr>
<tr><td><code>JD.pval</code></td>
<td>

<p>JD test p-value.
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Xinyu Wang, Han Chen, Duy Pham
</p>


<h3>References</h3>

<p>Wang, X., Lim, E., Liu, C, Sung, Y.J., Rao, D.C., Morrison, A.C., Boerwinkle, E., Manning, A. K., and Chen, H. (2020) Efficient gene-environment interaction tests for large biobank-scale sequencing studies. Genetic Epidemiology, 44(8): 908-923.
Chen, H., Huffman, J.E., Brody, J.A., Wang, C., Lee, S., Li, Z., Gogarten, S.M., Sofer, T., Bielak, L.F., Bis, J.C., et al. (2019) Efficient variant set mixed model association tests for continuous and binary traits in large-scale whole-genome sequencing studies. The American Journal of Human Genetics, 104 (2): 260-274.
Wang, X., Pham, D.T., Westerman, K.E., Pan, C., Manning, A.K., Chen, H., (2022) Genomic summary statistics and meta-analysis for set-based gene-environment interaction tests in large-scale sequencing studies. medrxiv.  
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if(requireNamespace("SeqArray", quietly = TRUE) &amp;&amp; requireNamespace("SeqVarTools",
  quietly = TRUE)) {
  library(GMMAT)
  data(example)
  attach(example)
  model0 &lt;- glmmkin(disease ~ age + sex, data = pheno, kins = GRM,
  id = "id", family = binomial(link = "logit"))
  geno.file &lt;- system.file("extdata", "geno.gds", package = "MAGEE")
  group.file &lt;- system.file("extdata", "SetID.withweights.txt", package = "MAGEE")
  meta.file.prefix &lt;- tempfile()
  out &lt;- MAGEE(model0, interaction="sex",geno.file=geno.file,
	       group.file=group.file, meta.file.prefix = meta.file.prefix,
	       MAF.range = c(0, 0.5), miss.cutoff = 1,
	       method = "davies", tests = c("JV","JF","JD"))
  print(out)
  out1 &lt;- MAGEE.meta(meta.file.prefix, group.file = group.file, tests = c("JV","JF","JD"))
  print(out1)
  unlink(paste0(meta.file.prefix, c(".score", ".cov"), ".1"))
}

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
