<!DOCTYPE html><html><head><title>Help for package MAGNAMWAR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MAGNAMWAR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#after_ortho_format'><p>Formatted output of OrthoMCL.</p></a></li>
<li><a href='#after_ortho_format_grps'><p>Formatted output of OrthoMCL.</p></a></li>
<li><a href='#AnalyzeOrthoMCL'><p>Main OrthoMCL Analysis</p></a></li>
<li><a href='#CalculatePrincipalCoordinates'><p>Show Principal Components Breakdown</p></a></li>
<li><a href='#FormatAfterOrtho'><p>Format file from output of OrthoMCL algorithm before use in AnalyzeOrthoMCL</p></a></li>
<li><a href='#FormatMCLFastas'><p>Format all raw GenBank fastas to single OrthoMCL compatible fasta file</p></a></li>
<li><a href='#joined_mtrx'><p>Final output of join_repset.</p></a></li>
<li><a href='#joined_mtrx_grps'><p>Final output of join_repset.</p></a></li>
<li><a href='#JoinRepSeq'><p>Join Representative Sequences</p></a></li>
<li><a href='#ManhatGrp'><p>Manhattan Plot of All Taxa</p></a></li>
<li><a href='#mcl_mtrx'><p>Final output of AnalyzeOrthoMCL</p></a></li>
<li><a href='#mcl_mtrx_grps'><p>Final output of AnalyzeOrthoMCL</p></a></li>
<li><a href='#PDGPlot'><p>Plot of a PDG and Data with Standard Error Bars</p></a></li>
<li><a href='#PDGvOG'><p>Number of PDGs vs OGs/PDG</p></a></li>
<li><a href='#pheno_data'><p>Triglyceride (TAG) content of fruit flies dataset.</p></a></li>
<li><a href='#PhyDataError'><p>Phylogenetic Tree with Attached Bar Plot and Standard Error Bars</p></a></li>
<li><a href='#PrintOGSeqs'><p>Print OG Sequences</p></a></li>
<li><a href='#QQPlotter'><p>QQPlot</p></a></li>
<li><a href='#RASTtoGBK'><p>Write RAST files to Genbank formats OrthoMCL Analysis</p></a></li>
<li><a href='#starv_pheno_data'><p>Starvation rate of fruit flies dataset.</p></a></li>
<li><a href='#SurvAppendMatrix'><p>Append Survival Test Outputs</p></a></li>
<li><a href='#WriteMCL'><p>Print analyzed matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Pipeline for Meta-Genome Wide Association</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-06-15</td>
</tr>
<tr>
<td>Description:</td>
<td>Correlates variation within the meta-genome to target species
    phenotype variations in meta-genome with association studies. Follows
    the pipeline described in Chaston, J.M. et al. (2014) &lt;<a href="https://doi.org/10.1128%2FmBio.01631-14">doi:10.1128/mBio.01631-14</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, coxme, doParallel, dplyr, foreach, iterators, lme4,
multcomp, parallel, plyr, qqman, survival, seqinr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-16 03:47:53 UTC; coripenrod</td>
</tr>
<tr>
<td>Author:</td>
<td>Corinne Sexton [aut],
  John Chaston [aut, cre],
  Hayden Smith [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>John Chaston &lt;john_chaston@byu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-12 07:20:17 UTC</td>
</tr>
</table>
<hr>
<h2 id='after_ortho_format'>Formatted output of OrthoMCL.</h2><span id='topic+after_ortho_format'></span>

<h3>Description</h3>

<p>A list created by inputting the output of OrthoMCL clusters
into the FormatAfterOrtho function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>after_ortho_format
</code></pre>


<h3>Format</h3>

<p>List of 2: (1) presence absence matrix, (2) protein ids:
</p>

<dl>
<dt>pa_matrix</dt><dd><p>matrix showing taxa presence/absence in OG</p>
</dd>
<dt>proteins</dt><dd><p>matrix listing protein_id contained in each OG</p>
</dd>
</dl>

<hr>
<h2 id='after_ortho_format_grps'>Formatted output of OrthoMCL.</h2><span id='topic+after_ortho_format_grps'></span>

<h3>Description</h3>

<p>A list created by inputting the output of OrthoMCL clusters
into the FormatAfterOrtho function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>after_ortho_format_grps
</code></pre>


<h3>Format</h3>

<p>List of 2: (1) presence absence matrix, (2) protein ids:
</p>

<dl>
<dt>pa_matrix</dt><dd><p>matrix showing taxa presence/absence in OG</p>
</dd>
<dt>proteins</dt><dd><p>matrix listing protein_id contained in each OG</p>
</dd>
</dl>

<hr>
<h2 id='AnalyzeOrthoMCL'>Main OrthoMCL Analysis</h2><span id='topic+AnalyzeOrthoMCL'></span>

<h3>Description</h3>

<p>Main function for analyzing the statistical association of OG (orthologous group) presence with phenotype data
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnalyzeOrthoMCL(mcl_data, pheno_data, model, species_name, resp = NULL,
  fix2 = NULL, rndm1 = NULL, rndm2 = NULL, multi = 1, time = NULL,
  event = NULL, time2 = NULL, startnum = 1, stopnum = "end",
  output_dir = NULL, sig_digits = NULL, princ_coord = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AnalyzeOrthoMCL_+3A_mcl_data">mcl_data</code></td>
<td>
<p>output of FormatAfterOrtho; a list of matrices; (1) a presence/absence matrix of taxa per OG, (2) a list of the specific protein ids within each OG</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_pheno_data">pheno_data</code></td>
<td>
<p>a data frame of phenotypic data with specific column names used to specify response variable as well as other fixed and random effects</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_model">model</code></td>
<td>
<p>linear model with gene presence as fixed effect (lm), linear mixed mffect models with gene presence as fixed effect and additional variables specified as: one random effect (lmeR1); two independent random effects (lmeR2ind); two random effects with rndm2 nested in rndm1 (lmeR2nest); or two independent random effects with one additional fixed effect (lmeF2), Wilcox Test with gene presence as fixed effect (wx), Survival Tests with support for multi core design: with two random effects (survmulti), and with two times as well as an additional fixed variable (survmulticensor)</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_species_name">species_name</code></td>
<td>
<p>Column name in pheno_data containing 4-letter species designations</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_resp">resp</code></td>
<td>
<p>Column name in pheno_data containing response variable</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_fix2">fix2</code></td>
<td>
<p>Column name in pheno_data containing second fixed effect</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_rndm1">rndm1</code></td>
<td>
<p>Column name in pheno_data containing first random variable</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_rndm2">rndm2</code></td>
<td>
<p>Column name in pheno_data containing second random variable</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_multi">multi</code></td>
<td>
<p>(can only be used with survival tests) Number of cores</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_time">time</code></td>
<td>
<p>(can only be used with survival tests) Column name in pheno_data containing first time</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_event">event</code></td>
<td>
<p>(can only be used with survival tests) Column name in pheno_data containing event</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_time2">time2</code></td>
<td>
<p>(can only be used with survival tests) Column name in pheno_data containing second time</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_startnum">startnum</code></td>
<td>
<p>number of test to start on</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_stopnum">stopnum</code></td>
<td>
<p>number of test to stop on</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_output_dir">output_dir</code></td>
<td>
<p>(if using survival tests) directory where small output files will be placed before using SurvAppendMatrix. Must specify a directory if choosing to output small files, else only written as a matrix</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_sig_digits">sig_digits</code></td>
<td>
<p>amount of digits to display for p-values and means of data; default to NULL (no rounding)</p>
</td></tr>
<tr><td><code id="AnalyzeOrthoMCL_+3A_princ_coord">princ_coord</code></td>
<td>
<p>the number of principle coordinates to be included in model as fixed effects (1, 2, or 3), if a decimal is specified, as many principal coordinates as are needed to account for that percentage of the variance will be included in the analysis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with the following columns: OG, p-values, Bonferroni corrected p-values, mean phenotype of OG-containing taxa, mean pheotype of OG-lacking taxa, taxa included in OG, taxa not included in OG
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Linear Model

## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, pheno_data, 'lm',
 'Treatment', resp='RespVar')

## End(Not run)

# the rest of the examples are not run for time's sake
#Linear Mixed Effect with one random effect
## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, pheno_data, 'lmeR1',
'Treatment', resp='RespVar', rndm1='Experiment')

## End(Not run)


#Linear Mixed Effect with two independent random effects
## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, pheno_data, 'lmeR2ind',
 'Treatment', resp='RespVar', rndm1='Experiment', rndm2='Vial')

## End(Not run)


#Linear Mixed Effect with rndm2 nested in rndm1
## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, pheno_data, 'lmeR2nest',
 'Treatment',  resp='RespVar', rndm1='Experiment', rndm2='Vial')

## End(Not run) 


#Linear Mixed Effect with two independent random effects and one additional fixed effect
## Not run: 
mcl_mtrx3 &lt;- AnalyzeOrthoMCL(after_ortho_format, pheno_data, 'lmeF2',
 'Treatment', resp='RespVar', fix2='Treatment', rndm1='Experiment', rndm2='Vial', princ_coord = 4)

## End(Not run)


#Wilcoxon Test
## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, pheno_data, 'wx',
 'Treatment', resp='RespVar')

## End(Not run)

# ~ 5 minutes
#Survival with two independent random effects, run on multiple cores
## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, starv_pheno_data, 'TRT', model='survmulti',
 time='t2', event='event', rndm1='EXP', rndm2='VIAL', multi=1)

## End(Not run)

# ~ 5 minutes
#Survival with two independent random effects and one additional fixed effect,
#including drops on multi cores
## Not run: 
mcl_mtrx &lt;- AnalyzeOrthoMCL(after_ortho_format, starv_pheno_data, 'TRT', model='survmulticensor',
 time='t1', time2='t2', event='event', rndm1='EXP', rndm2='VIAL', fix2='BACLO', multi=1)
 
## End(Not run)
#to be appended with SurvAppendMatrix
</code></pre>

<hr>
<h2 id='CalculatePrincipalCoordinates'>Show Principal Components Breakdown</h2><span id='topic+CalculatePrincipalCoordinates'></span>

<h3>Description</h3>

<p>Function to show Principal Components statistics based on the OrthoMCL presence absence groupings.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CalculatePrincipalCoordinates(mcl_data)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CalculatePrincipalCoordinates_+3A_mcl_data">mcl_data</code></td>
<td>
<p>output of FormatAfterOrtho &ndash;list of 2 things&ndash; 1: binary matrix indicating the presence / absence of genes in each OG and 2: vector of names of OGs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a named list of principal components and accompanying proportion of variance for each
</p>


<h3>Examples</h3>

<pre><code class='language-R'>CalculatePrincipalCoordinates(after_ortho_format)
</code></pre>

<hr>
<h2 id='FormatAfterOrtho'>Format file from output of OrthoMCL algorithm before use in AnalyzeOrthoMCL</h2><span id='topic+FormatAfterOrtho'></span>

<h3>Description</h3>

<p>After running OrthoMCL and/or submitting to www.orthomcl.org, formats the output file to be used in AnalyzeOrthoMCL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FormatAfterOrtho(file, format = "ortho")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FormatAfterOrtho_+3A_file">file</code></td>
<td>
<p>Path to the OrthoMCL output file</p>
</td></tr>
<tr><td><code id="FormatAfterOrtho_+3A_format">format</code></td>
<td>
<p>Specification of the method by which file was obtained: defaults to 'ortho' for output from orthomcl.org. Other option is 'groups' for output from local run of OrthoMCL software.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of matrices; (1) a presence/absence matrix of taxa per OG, (2) a list of the specific protein ids within each OG
</p>


<h3>Examples</h3>

<pre><code class='language-R'>file &lt;- system.file('extdata', 'orthologGroups.txt', package='MAGNAMWAR')
after_ortho_format &lt;- FormatAfterOrtho(file)

file_grps &lt;- system.file('extdata', 'groups_example_r.txt', package='MAGNAMWAR')
after_ortho_format_grps &lt;- FormatAfterOrtho(file_grps, format = 'groups')

</code></pre>

<hr>
<h2 id='FormatMCLFastas'>Format all raw GenBank fastas to single OrthoMCL compatible fasta file</h2><span id='topic+FormatMCLFastas'></span>

<h3>Description</h3>

<p>Creates the composite fasta file for use in running OrthoMCL and/or submitting to www.orthomcl.org
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FormatMCLFastas(fa_dir, genbnk_id = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FormatMCLFastas_+3A_fa_dir">fa_dir</code></td>
<td>
<p>Path to the directory where all raw GenBank files are stored. Note, all file names must be changed to a 4-letter code representing each species and have '.fasta' file descriptor</p>
</td></tr>
<tr><td><code id="FormatMCLFastas_+3A_genbnk_id">genbnk_id</code></td>
<td>
<p>(Only necessary for the deprecated version of fasta headers) The index of the sequence ID in the GenBank pipe-separated annotation line (default: 4)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns nothing, but prints the path to the final OrthoMCL compatible fasta file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
dir &lt;- system.file('extdata', 'fasta_dir', package='MAGNAMWAR')
dir &lt;- paste(dir,'/',sep='')
formatted_file &lt;- FormatMCLFastas(dir)

## End(Not run)
</code></pre>

<hr>
<h2 id='joined_mtrx'>Final output of join_repset.</h2><span id='topic+joined_mtrx'></span>

<h3>Description</h3>

<p>A data frame containing the final results of statistical analysis with 
protein ids, annotations, and sequences added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>joined_mtrx
</code></pre>


<h3>Format</h3>

<p>A data frame with 17 rows and 11 variables:
</p>

<dl>
<dt>OG</dt><dd><p>taxa cluster id, as defined by OrthoMCL</p>
</dd>
<dt>pval1</dt><dd><p>p-value, based on presence absence</p>
</dd>
<dt>corrected_pval1</dt><dd><p>Bonferroni p-value, corrected by number of tests</p>
</dd>
<dt>mean_OGContain</dt><dd><p>mean of all taxa phenotypes in that OG</p>
</dd>
<dt>mean_OGLack</dt><dd><p>mean of all taxa phenotypes not in that OG</p>
</dd>
<dt>taxa_contain</dt><dd><p>taxa in that cluster</p>
</dd>
<dt>taxa_miss</dt><dd><p>taxa not in that cluster</p>
</dd>
<dt>rep_taxon</dt><dd><p>randomly selected representative taxa from the cluster</p>
</dd>
<dt>rep_id</dt><dd><p>protein id, from randomly selected representative taxa</p>
</dd>
<dt>rep_annot</dt><dd><p>fasta annotation, from randomly selected representative taxa</p>
</dd>
<dt>rep_seq</dt><dd><p>AA sequence, from randomly selected representative taxa</p>
</dd>
</dl>

<hr>
<h2 id='joined_mtrx_grps'>Final output of join_repset.</h2><span id='topic+joined_mtrx_grps'></span>

<h3>Description</h3>

<p>A data frame containing the final results of statistical analysis with 
protein ids, annotations, and sequences added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>joined_mtrx_grps
</code></pre>


<h3>Format</h3>

<p>A data frame with 10 rows and 11 variables:
</p>

<dl>
<dt>OG</dt><dd><p>taxa cluster id, as defined by OrthoMCL</p>
</dd>
<dt>pval1</dt><dd><p>p-value, based on presence absence</p>
</dd>
<dt>corrected_pval1</dt><dd><p>Bonferroni p-value, corrected by number of tests</p>
</dd>
<dt>mean_OGContain</dt><dd><p>mean of all taxa phenotypes in that OG</p>
</dd>
<dt>mean_OGLack</dt><dd><p>mean of all taxa phenotypes not in that OG</p>
</dd>
<dt>taxa_contain</dt><dd><p>taxa in that cluster</p>
</dd>
<dt>taxa_miss</dt><dd><p>taxa not in that cluster</p>
</dd>
<dt>rep_taxon</dt><dd><p>randomly selected representative taxa from the cluster</p>
</dd>
<dt>rep_id</dt><dd><p>protein id, from randomly selected representative taxa</p>
</dd>
<dt>rep_annot</dt><dd><p>fasta annotation, from randomly selected representative taxa</p>
</dd>
<dt>rep_seq</dt><dd><p>AA sequence, from randomly selected representative taxa</p>
</dd>
</dl>

<hr>
<h2 id='JoinRepSeq'>Join Representative Sequences</h2><span id='topic+JoinRepSeq'></span>

<h3>Description</h3>

<p>Joins the OrthoMCL output matrix to representative sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JoinRepSeq(mcl_data, fa_dir, mcl_mtrx, fastaformat = "new")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JoinRepSeq_+3A_mcl_data">mcl_data</code></td>
<td>
<p>output of FormatAfterOrtho; a list of matrices; (1) a presence/absence matrix of taxa per OG, (2) a list of the specific protein ids within each OG</p>
</td></tr>
<tr><td><code id="JoinRepSeq_+3A_fa_dir">fa_dir</code></td>
<td>
<p>Path to the directory where all raw GenBank files are stored. Note, all file names must be changed to a 4-letter code representing each species and have '.fasta' file descriptor</p>
</td></tr>
<tr><td><code id="JoinRepSeq_+3A_mcl_mtrx">mcl_mtrx</code></td>
<td>
<p>OrthoMCL output matrix from AnalyzeOrthoMCL()</p>
</td></tr>
<tr><td><code id="JoinRepSeq_+3A_fastaformat">fastaformat</code></td>
<td>
<p>options: new &amp; old; new = no GI numbers included; defaults to new</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns the original OrthoMCL output matrix with additional columns: representative sequence taxon, representative sequence id, representative sequence annotation, representative sequence
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
dir &lt;- system.file('extdata', 'fasta_dir', package='MAGNAMWAR')
dir &lt;- paste(dir,'/',sep='')
joined_mtrx_grps &lt;- JoinRepSeq(after_ortho_format_grps, dir, mcl_mtrx_grps, fastaformat = 'old')

## End(Not run)
</code></pre>

<hr>
<h2 id='ManhatGrp'>Manhattan Plot of All Taxa</h2><span id='topic+ManhatGrp'></span>

<h3>Description</h3>

<p>Manhattan plot that graphs all p-values for taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ManhatGrp(mcl_data, mcl_mtrx, tree = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ManhatGrp_+3A_mcl_data">mcl_data</code></td>
<td>
<p>FormatAfterOrtho output</p>
</td></tr>
<tr><td><code id="ManhatGrp_+3A_mcl_mtrx">mcl_mtrx</code></td>
<td>
<p>output of AnalyzeOrthoMCL()</p>
</td></tr>
<tr><td><code id="ManhatGrp_+3A_tree">tree</code></td>
<td>
<p>tree file optional, used for ordering taxa along x axis</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a manhattan plot
</p>


<h3>References</h3>

<p>Some sort of reference
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ManhatGrp(after_ortho_format, mcl_mtrx)

#@param equation of line of significance, defaults to -log10((.05)/dim(pdgs)[1])
</code></pre>

<hr>
<h2 id='mcl_mtrx'>Final output of AnalyzeOrthoMCL</h2><span id='topic+mcl_mtrx'></span>

<h3>Description</h3>

<p>A matrix containing the final results of statistical analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcl_mtrx
</code></pre>


<h3>Format</h3>

<p>A matrix with 17 rows and 7 variables:
</p>

<dl>
<dt>OG</dt><dd><p>taxa cluster id, as defined by OrthoMCL</p>
</dd>
<dt>pval1</dt><dd><p>p-value, based on presence absence</p>
</dd>
<dt>corrected_pval1</dt><dd><p>Bonferroni p-value, corrected by number of tests</p>
</dd>
<dt>mean_OGContain</dt><dd><p>mean of all taxa phenotypes in that OG</p>
</dd>
<dt>mean_OGLack</dt><dd><p>mean of all taxa phenotypes not in that OG</p>
</dd>
<dt>taxa_contain</dt><dd><p>taxa in that cluster</p>
</dd>
<dt>taxa_miss</dt><dd><p>taxa not in that cluster</p>
</dd>
</dl>

<hr>
<h2 id='mcl_mtrx_grps'>Final output of AnalyzeOrthoMCL</h2><span id='topic+mcl_mtrx_grps'></span>

<h3>Description</h3>

<p>A matrix containing the final results of statistical analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcl_mtrx_grps
</code></pre>


<h3>Format</h3>

<p>A matrix with 10 rows and 7 variables:
</p>

<dl>
<dt>OG</dt><dd><p>taxa cluster id, as defined by OrthoMCL</p>
</dd>
<dt>pval1</dt><dd><p>p-value, based on presence absence</p>
</dd>
<dt>corrected_pval1</dt><dd><p>Bonferroni p-value, corrected by number of tests</p>
</dd>
<dt>mean_OGContain</dt><dd><p>mean of all taxa phenotypes in that OG</p>
</dd>
<dt>mean_OGLack</dt><dd><p>mean of all taxa phenotypes not in that OG</p>
</dd>
<dt>taxa_contain</dt><dd><p>taxa in that cluster</p>
</dd>
<dt>taxa_miss</dt><dd><p>taxa not in that cluster</p>
</dd>
</dl>

<hr>
<h2 id='PDGPlot'>Plot of a PDG and Data with Standard Error Bars</h2><span id='topic+PDGPlot'></span>

<h3>Description</h3>

<p>Bar plot of PDG vs phenotype data with presence of taxa in PDG indicated by color
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PDGPlot(data, mcl_matrix, OG = "NONE", species_colname, data_colname,
  xlab = "Taxa", ylab = "Data", ylimit = NULL, tree = NULL,
  order = NULL, main_title = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PDGPlot_+3A_data">data</code></td>
<td>
<p>R object of phenotype data</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_mcl_matrix">mcl_matrix</code></td>
<td>
<p>AnalyzeOrthoMCL output</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_og">OG</code></td>
<td>
<p>optional parameter, a string with the name of chosen group (OG) to be colored</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_species_colname">species_colname</code></td>
<td>
<p>name of column in phenotypic data file with taxa designations</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_data_colname">data_colname</code></td>
<td>
<p>name of column in phenotypic data file with data observations</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_xlab">xlab</code></td>
<td>
<p>string to label barplot's x axis</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_ylab">ylab</code></td>
<td>
<p>string to label barplot's y axis</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_ylimit">ylimit</code></td>
<td>
<p>optional parameter to limit y axis</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_tree">tree</code></td>
<td>
<p>optional parameter (defaults to NULL) Path to tree file, orders the taxa by phylogenetic distribution, else it defaults to alphabetical</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_order">order</code></td>
<td>
<p>vector with order of taxa names for across the x axis (defaults to alpha ordering)</p>
</td></tr>
<tr><td><code id="PDGPlot_+3A_main_title">main_title</code></td>
<td>
<p>string for title of the plot (defaults to OG)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a barplot with taxa vs phenotypic data complete with standard error bars
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
PDGPlot(pheno_data, mcl_mtrx, 'OG5_126778', 'Treatment', 'RespVar', ylimit=12)

</code></pre>

<hr>
<h2 id='PDGvOG'>Number of PDGs vs OGs/PDG</h2><span id='topic+PDGvOG'></span>

<h3>Description</h3>

<p>Barplot that indicates the number of PDGs vs OGs(clustered orthologous groups) in a PDG
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PDGvOG(mcl_data, num = 40, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PDGvOG_+3A_mcl_data">mcl_data</code></td>
<td>
<p>FormatAfterOrtho output</p>
</td></tr>
<tr><td><code id="PDGvOG_+3A_num">num</code></td>
<td>
<p>an integer indicating where the x axis should end and be compiled</p>
</td></tr>
<tr><td><code id="PDGvOG_+3A_...">...</code></td>
<td>
<p>args to be passed to barplot</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a barplot with a height determined by the second column and the first column abbreviated to accomodate visual spacing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
PDGvOG(after_ortho_format_grps,2)

</code></pre>

<hr>
<h2 id='pheno_data'>Triglyceride (TAG) content of fruit flies dataset.</h2><span id='topic+pheno_data'></span>

<h3>Description</h3>

<p>A subset of the TAG content of fruit flies, collected in the Chaston Lab,
to be used as a brief example for tests in AnalyzeOrthoMCL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pheno_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 586 rows and 4 variables:
</p>

<dl>
<dt>Treatment</dt><dd><p>4-letter taxa designation of associated bacteria</p>
</dd>
<dt>RespVar</dt><dd><p>response variable, TAG content</p>
</dd>
<dt>Vial</dt><dd><p>random effect variable, vial number of flies</p>
</dd>
<dt>Experiment</dt><dd><p>random effect variable, experiment number of flies</p>
</dd>
</dl>

<hr>
<h2 id='PhyDataError'>Phylogenetic Tree with Attached Bar Plot and Standard Error Bars</h2><span id='topic+PhyDataError'></span>

<h3>Description</h3>

<p>Presents data for each taxa including standard error bars next to a phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PhyDataError(phy, data, mcl_matrix, species_colname, data_colname,
  color = NULL, OG = NULL, xlabel = "xlabel", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PhyDataError_+3A_phy">phy</code></td>
<td>
<p>Path to tree file</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_data">data</code></td>
<td>
<p>R object of phenotype data</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_mcl_matrix">mcl_matrix</code></td>
<td>
<p>AnalyzeOrthoMCL output</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_species_colname">species_colname</code></td>
<td>
<p>name of column in data file with taxa designations</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_data_colname">data_colname</code></td>
<td>
<p>name of column in data file with data observations</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_color">color</code></td>
<td>
<p>optional parameter, (defaults to NULL) assign colors to individual taxa by providing file (format: Taxa | Color)</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_og">OG</code></td>
<td>
<p>optional parameter, (defaults to NULL) a string with the names of chosen group to be colored</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_xlabel">xlabel</code></td>
<td>
<p>string to label barplot's x axis</p>
</td></tr>
<tr><td><code id="PhyDataError_+3A_...">...</code></td>
<td>
<p>argument to be passed from other methods such as parameters from barplot() function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A phylogenetic tree with a barplot of the data (with standard error bars) provided matched by taxa.
</p>


<h3>References</h3>

<p>Some sort of reference
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
file &lt;- system.file('extdata', 'muscle_tree2.dnd', package='MAGNAMWAR')
PhyDataError(file, pheno_data, mcl_mtrx, species_colname = 'Treatment', data_colname = 'RespVar',
 OG='OG5_126778', xlabel='TAG Content')

</code></pre>

<hr>
<h2 id='PrintOGSeqs'>Print OG Sequences</h2><span id='topic+PrintOGSeqs'></span>

<h3>Description</h3>

<p>Print all protein sequences and annotations in a given OG
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PrintOGSeqs(after_ortho, OG, fasta_dir, out_dir = NULL, outfile = "none")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PrintOGSeqs_+3A_after_ortho">after_ortho</code></td>
<td>
<p>output from FormatAfterOrtho</p>
</td></tr>
<tr><td><code id="PrintOGSeqs_+3A_og">OG</code></td>
<td>
<p>name of OG</p>
</td></tr>
<tr><td><code id="PrintOGSeqs_+3A_fasta_dir">fasta_dir</code></td>
<td>
<p>directory to fastas</p>
</td></tr>
<tr><td><code id="PrintOGSeqs_+3A_out_dir">out_dir</code></td>
<td>
<p>complete path to output directory</p>
</td></tr>
<tr><td><code id="PrintOGSeqs_+3A_outfile">outfile</code></td>
<td>
<p>name of file that will be written to</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A fasta file with all protein sequences and ids for a given OG
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
OG &lt;- 'OG5_126968'
dir &lt;- system.file('extdata', 'fasta_dir', package='MAGNAMWAR')
dir &lt;- paste(dir,'/',sep='')

PrintOGSeqs(after_ortho_format, OG, dir)

## End(Not run)

</code></pre>

<hr>
<h2 id='QQPlotter'>QQPlot</h2><span id='topic+QQPlotter'></span>

<h3>Description</h3>

<p>Makes a qqplot of the p-values obtained through AnalyzeOrthoMCL
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QQPlotter(mcl_mtrx)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="QQPlotter_+3A_mcl_mtrx">mcl_mtrx</code></td>
<td>
<p>matrix generated by AnalyzeOrthoMCL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a qqplot of the p-values obtained through AnalyzeOrthoMCL
</p>


<h3>References</h3>

<p>Some sore of reference
</p>


<h3>Examples</h3>

<pre><code class='language-R'>QQPlotter(mcl_mtrx)
</code></pre>

<hr>
<h2 id='RASTtoGBK'>Write RAST files to Genbank formats OrthoMCL Analysis</h2><span id='topic+RASTtoGBK'></span>

<h3>Description</h3>

<p>Useful for reformating RAST files to GBK format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RASTtoGBK(input_fasta, input_reference, out_name_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RASTtoGBK_+3A_input_fasta">input_fasta</code></td>
<td>
<p>path to input fasta file</p>
</td></tr>
<tr><td><code id="RASTtoGBK_+3A_input_reference">input_reference</code></td>
<td>
<p>path to a .csv file; it should be downloaded from RAST as excel format, saved as a .csv (saved as the tab-delimited version has compatibility problems)</p>
</td></tr>
<tr><td><code id="RASTtoGBK_+3A_out_name_path">out_name_path</code></td>
<td>
<p>name and path of the file to write to</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
lfrc_fasta &lt;- system.file('extdata', 'RASTtoGBK//lfrc.fasta', package='MAGNAMWAR')
lfrc_reference &lt;- system.file('extdata', 'RASTtoGBK//lfrc_lookup.csv', package='MAGNAMWAR')
lfrc_path &lt;- system.file('extdata', 'RASTtoGBK//lfrc_out.fasta', package='MAGNAMWAR')

RASTtoGBK(lfrc_fasta,lfrc_reference,lfrc_path)

## End(Not run)

</code></pre>

<hr>
<h2 id='starv_pheno_data'>Starvation rate of fruit flies dataset.</h2><span id='topic+starv_pheno_data'></span>

<h3>Description</h3>

<p>A subset of the Starvation rate of fruit flies, collected in the Chaston Lab,
to be used as a brief example for survival tests in AnalyzeOrthoMCL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>starv_pheno_data
</code></pre>


<h3>Format</h3>

<p>A matrix with 543 rows and 7 variables:
</p>

<dl>
<dt>EXP</dt><dd><p>random effect variable, experiment number of flies</p>
</dd>
<dt>VIAL</dt><dd><p>random effect variable, vial number of flies</p>
</dd>
<dt>BACLO</dt><dd><p>fixed effect variable, loss of bacteria in flies</p>
</dd>
<dt>TRT</dt><dd><p>4-letter taxa designation of associated bacteria</p>
</dd>
<dt>t1</dt><dd><p>time 1</p>
</dd>
<dt>t2</dt><dd><p>time 2</p>
</dd>
<dt>event</dt><dd><p>event</p>
</dd>
</dl>

<hr>
<h2 id='SurvAppendMatrix'>Append Survival Test Outputs</h2><span id='topic+SurvAppendMatrix'></span>

<h3>Description</h3>

<p>Function used to append all .csv files that are outputted from AnalyzeOrthoMCL into one matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SurvAppendMatrix(work_dir, out_name = "surv_matrix.csv", out_dir = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SurvAppendMatrix_+3A_work_dir">work_dir</code></td>
<td>
<p>the directory where the output files of AnalyzeOrthoMCL are located</p>
</td></tr>
<tr><td><code id="SurvAppendMatrix_+3A_out_name">out_name</code></td>
<td>
<p>file name of outputted matrix</p>
</td></tr>
<tr><td><code id="SurvAppendMatrix_+3A_out_dir">out_dir</code></td>
<td>
<p>the directory where the outputted matrix is placed</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A csv file containing a matrix with the following columns: OG, p-values, Bonferroni corrected p-values, mean phenotype of OG-containing taxa, mean pheotype of OG-lacking taxa, taxa included in OG, taxa not included in OG
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
file &lt;- system.file('extdata', 'outputs', package='MAGNAMWAR')
directory &lt;- paste(file, '/', sep = '')
SurvAppendMatrix(directory)

## End(Not run)

</code></pre>

<hr>
<h2 id='WriteMCL'>Print analyzed matrix</h2><span id='topic+WriteMCL'></span>

<h3>Description</h3>

<p>Writes a tab separated version of the analyzed OrthoMCL data with or without the joined representative sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>WriteMCL(mtrx, filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="WriteMCL_+3A_mtrx">mtrx</code></td>
<td>
<p>Matrix derived from AnalyzeOrthoMCL</p>
</td></tr>
<tr><td><code id="WriteMCL_+3A_filename">filename</code></td>
<td>
<p>File name to save final output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The path to the written file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
WriteMCL(mcl_mtrx, 'matrix.tsv')
#mcl_mtrx previously derived from AnalyzeOrthoMCL() or join_repset()

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
