<!DOCTYPE html><html lang="en"><head><title>Help for package malani</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {malani}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dat'><p>A matrix of expression values.</p></a></li>
<li><a href='#grp'><p>A vector of class labels for <code>dat</code>.</p></a></li>
<li><a href='#Gsvmod'><p>G SVM models.</p></a></li>
<li><a href='#intGenes'><p>Select initial gene list from original data matrix.</p></a></li>
<li><a href='#malanidata'><p>Dataset for malani package</p></a></li>
<li><a href='#pairmod'><p>Find best performing pairs</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Machine Learning Assisted Network Inference</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Mehrab Ghanat Bari</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mehrab Ghanat Bari &lt;m.ghanatbari@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Find dark genes. These genes are often disregarded due to no detected mutation or differential expression, but are important in coordinating the functionality in cancer networks.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Depends:</td>
<td>e1071, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-09-26 17:54:07 UTC; m157258</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-09-26 23:44:53</td>
</tr>
</table>
<hr>
<h2 id='dat'>A matrix of expression values.</h2><span id='topic+dat'></span>

<h3>Description</h3>

<p>A numeric matrix 100*20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat
</code></pre>


<h3>Format</h3>

<p>matrix.</p>

<hr>
<h2 id='grp'>A vector of class labels for <code><a href="#topic+dat">dat</a></code>.</h2><span id='topic+grp'></span>

<h3>Description</h3>

<p>Vector length of 20.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grp
</code></pre>


<h3>Format</h3>

<p>vector</p>

<hr>
<h2 id='Gsvmod'>G SVM models.</h2><span id='topic+Gsvmod'></span>

<h3>Description</h3>

<p>Returns accuracy performance of all genes. G support vector machine (SVM) classifiers trained using G different data matrixes, are used to predict labels in test data. Models are ranked based on prediction performances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gsvmod(dat.train, lab.train, dat.test, lab.test)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Gsvmod_+3A_dat.train">dat.train</code></td>
<td>
<p>Train data with G features and (k-1)*S/k samples. Parameter k comes from cross-validation scheme and is specified by user (default is 2).</p>
</td></tr>
<tr><td><code id="Gsvmod_+3A_lab.train">lab.train</code></td>
<td>
<p>Class labels for train data.</p>
</td></tr>
<tr><td><code id="Gsvmod_+3A_dat.test">dat.test</code></td>
<td>
<p>Test data with G features and S/k samples.</p>
</td></tr>
<tr><td><code id="Gsvmod_+3A_lab.test">lab.test</code></td>
<td>
<p>Class labels for test data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Accuracy scores for models. Each model represents one gene.
</p>

<hr>
<h2 id='intGenes'>Select initial gene list from original data matrix.</h2><span id='topic+intGenes'></span>

<h3>Description</h3>

<p>Train G-1 SVM models in <code>k-fold</code> cross validation scheme to select initial genes list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intGenes(dat, grp, nfolds.out = 2, top.per = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="intGenes_+3A_dat">dat</code></td>
<td>
<p>Original gene expression data matrix with G rows (number of genes) and S column (number of samples).</p>
</td></tr>
<tr><td><code id="intGenes_+3A_grp">grp</code></td>
<td>
<p>Class labels.</p>
</td></tr>
<tr><td><code id="intGenes_+3A_nfolds.out">nfolds.out</code></td>
<td>
<p>Outer cross validation number (default is 2).</p>
</td></tr>
<tr><td><code id="intGenes_+3A_top.per">top.per</code></td>
<td>
<p>All genes are ranked based on their models performance and <code> top.per%</code> of them are selected as initial genes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Selected initial genes.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(malanidata)
int &lt;- intGenes(dat,grp)
print(int$top.genes)
</code></pre>

<hr>
<h2 id='malanidata'>Dataset for malani package</h2><span id='topic+malanidata'></span>

<h3>Description</h3>

<p>A numeric matrix G*S contains gene expressions data.
G are the genes (rows) and S are the samples (columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>malanidata
</code></pre>


<h3>Format</h3>

<p>A matrix of numeric values, 100 genes, 20 samples and class labels.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(malanidata)
</code></pre>

<hr>
<h2 id='pairmod'>Find best performing pairs</h2><span id='topic+pairmod'></span>

<h3>Description</h3>

<p>Combine each gene in initial set with all genes in the original set. Top <code>npair</code>  pairs are selected to construct the <code>Q</code> matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairmod(X, LX, theta, npair = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairmod_+3A_x">X</code></td>
<td>
<p>Original gene expression data matrix. With G rows (number of genes) and S column (number of samples).</p>
</td></tr>
<tr><td><code id="pairmod_+3A_lx">LX</code></td>
<td>
<p>Class labels.</p>
</td></tr>
<tr><td><code id="pairmod_+3A_theta">theta</code></td>
<td>
<p>Initial gene set.</p>
</td></tr>
<tr><td><code id="pairmod_+3A_npair">npair</code></td>
<td>
<p>Given a gene in initial set, top <code>npair</code> best performing pairs correspond to that gene are selected (Default is 10).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Best (<code>npair*G/20</code>) performing pairs.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
