<!DOCTYPE html><html><head><title>Help for package malan</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {malan}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#[[.malan_pedigreelist'><p>Get pedigree from pedigree list</p></a></li>
<li><a href='#[[.malan_population'><p>Get individual from population by pid</p></a></li>
<li><a href='#analyse_mixture_result'><p>Analyse mixture results</p></a></li>
<li><a href='#analyse_mixture_results'><p>Analyse mixture results in a vectorised fashion</p></a></li>
<li><a href='#as_tbl_graph.malan_pedigreelist'><p>Get tidy graph object</p></a></li>
<li><a href='#brothers_matching'><p>Number of brothers with matching haplotype</p></a></li>
<li><a href='#build_haplotype_hashmap'><p>Build hashmap of haplotype to individuals</p></a></li>
<li><a href='#build_pedigrees'><p>Build pedigrees from (individuals in) a population.</p></a></li>
<li><a href='#calc_autosomal_genotype_conditional_cumdist'><p>Calculate conditional genotype cumulative probabilities with theta</p></a></li>
<li><a href='#calc_autosomal_genotype_probs'><p>Calculate genotype probabilities with theta</p></a></li>
<li><a href='#construct_M'><p>Construct M matrix</p></a></li>
<li><a href='#count_brothers'><p>Number of brothers</p></a></li>
<li><a href='#count_haplotype_near_matches_individuals'><p>Count near haplotype matches in list of individuals</p></a></li>
<li><a href='#count_haplotype_occurrences_individuals'><p>Count haplotypes occurrences in list of individuals</p></a></li>
<li><a href='#count_haplotype_occurrences_pedigree'><p>Count haplotypes occurrences in pedigree</p></a></li>
<li><a href='#count_uncles'><p>Number of uncles</p></a></li>
<li><a href='#delete_haplotypeids_hashmap'><p>Delete haplotype hashmap</p></a></li>
<li><a href='#estimate_autotheta_1subpop_genotypes'><p>Estimate autosomal theta from genotypes</p></a></li>
<li><a href='#estimate_autotheta_1subpop_individuals'><p>Estimate autosomal theta from individuals</p></a></li>
<li><a href='#estimate_autotheta_subpops_genotypes'><p>Estimate autosomal F, theta, and f from subpopulations of genotypes</p></a></li>
<li><a href='#estimate_autotheta_subpops_individuals'><p>Estimate autosomal F, theta, and f from subpopulations of individuals</p></a></li>
<li><a href='#estimate_autotheta_subpops_pids'><p>Estimate autosomal F, theta, and f from subpopulations of individual ids</p></a></li>
<li><a href='#estimate_autotheta_subpops_unweighted_genotypes'><p>Unweighted estimate of autosomal theta from subpopulations of genotypes</p></a></li>
<li><a href='#estimate_autotheta_subpops_unweighted_pids'><p>Unweighted estimate of autosomal theta from subpopulations of individual ids</p></a></li>
<li><a href='#father_matches'><p>Father matches</p></a></li>
<li><a href='#from_igraph'><p>Convert igraph to population</p></a></li>
<li><a href='#from_igraph_rcpp'><p>Generate paternal brothers population</p></a></li>
<li><a href='#generate_get_founder_haplotype_db'><p>Generate a function to simulate pedigree founder haplotype based on a haplotype databasep</p></a></li>
<li><a href='#generate_get_founder_haplotype_ladder'><p>Generate a function to simulate pedigree founder haplotype based on ladder information</p></a></li>
<li><a href='#get_allele_counts_genotypes'><p>Get autosomal allele counts from subpopulations of genotypes</p></a></li>
<li><a href='#get_allele_counts_pids'><p>Get autosomal allele counts from subpopulations given by pids</p></a></li>
<li><a href='#get_brothers'><p>Get brothers</p></a></li>
<li><a href='#get_children'><p>Get children</p></a></li>
<li><a href='#get_cousins'><p>Get cousins</p></a></li>
<li><a href='#get_family_info'><p>Get individual's family information</p></a></li>
<li><a href='#get_father'><p>Get father</p></a></li>
<li><a href='#get_generation'><p>Get individual's generation number</p></a></li>
<li><a href='#get_haplotype'><p>Get haplotype from an individual</p></a></li>
<li><a href='#get_haplotypes_in_pedigree'><p>Get haplotypes in pedigree</p></a></li>
<li><a href='#get_haplotypes_individuals'><p>Get haplotype matrix from list of individuals</p></a></li>
<li><a href='#get_haplotypes_pids'><p>Get haplotypes from a vector of pids.</p></a></li>
<li><a href='#get_individual'><p>Get individual by pid</p></a></li>
<li><a href='#get_individuals'><p>Get all individuals in population</p></a></li>
<li><a href='#get_matching_pids_from_hashmap'><p>Get individuals with a certain haplotype id by hashmap lookup</p></a></li>
<li><a href='#get_nodes_edges'><p>Get nodes and edges</p></a></li>
<li><a href='#get_pedigree_as_graph'><p>Get pedigree information as graph (mainly intended for plotting)</p></a></li>
<li><a href='#get_pedigree_from_individual'><p>Get pedigree from individual</p></a></li>
<li><a href='#get_pedigree_id'><p>Get pedigree id</p></a></li>
<li><a href='#get_pedigree_id_from_pid'><p>Get pedigree ids from pids</p></a></li>
<li><a href='#get_pedigrees_tidy'><p>Get pedigrees information in tidy format</p></a></li>
<li><a href='#get_pid'><p>Get pid from individual</p></a></li>
<li><a href='#get_pids_in_pedigree'><p>Get pids in pedigree</p></a></li>
<li><a href='#get_uncles'><p>Get uncles</p></a></li>
<li><a href='#get_zero_haplotype_generator'><p>Generate a function to generate the zero haplotype</p></a></li>
<li><a href='#grandfather_matches'><p>Grandfather matches</p></a></li>
<li><a href='#haplotype_matches_individuals'><p>Get individuals matching from list of individuals</p></a></li>
<li><a href='#haplotype_partially_matches_individuals'><p>Get individuals partially matching from list of individuals</p></a></li>
<li><a href='#haplotypes_to_hashes'><p>Convert haplotypes to hashes (integers)</p></a></li>
<li><a href='#infer_generation'><p>Infer individual's generation number</p></a></li>
<li><a href='#infer_generations'><p>Infer generation numbers from pedigrees</p></a></li>
<li><a href='#load_haplotypes'><p>Load haplotypes to individuals</p></a></li>
<li><a href='#load_individuals'><p>Construct a population from data</p></a></li>
<li><a href='#malan-package'><p>MAle Lineage ANalysis</p></a></li>
<li><a href='#meioses_generation_distribution'><p>Meiotic distribution</p></a></li>
<li><a href='#meiotic_dist'><p>Meiotic distance between two individuals</p></a></li>
<li><a href='#meiotic_dist_threshold'><p>Meiotic distance between two individuals (with threshold)</p></a></li>
<li><a href='#meiotic_radius'><p>Meiotic radius</p></a></li>
<li><a href='#mixture_info_by_individuals_2pers'><p>Mixture information about 2 persons' mixture of donor1 and donor2.</p></a></li>
<li><a href='#mixture_info_by_individuals_3pers'><p>Mixture information about 3 persons' mixture of donor1, donor2 and donor3.</p></a></li>
<li><a href='#mixture_info_by_individuals_4pers'><p>Mixture information about 4 persons' mixture of donor1, donor2, donor3 and donor4.</p></a></li>
<li><a href='#mixture_info_by_individuals_5pers'><p>Mixture information about 5 persons' mixture of donor1, donor2, donor3, donor4 and donor5.</p></a></li>
<li><a href='#pedigree_as_igraph'><p>Convert pedigree to igraph</p></a></li>
<li><a href='#pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists'><p>Information about matching individuals</p></a></li>
<li><a href='#pedigree_haplotype_near_matches_meiosis'><p>Information about almost matching individuals</p></a></li>
<li><a href='#pedigree_size'><p>Get pedigree size</p></a></li>
<li><a href='#pedigree_size_generation'><p>Size of pedigree</p></a></li>
<li><a href='#pedigrees_all_populate_autosomal'><p>Populate 1-locus autosomal DNA profile in pedigrees with single-step mutation model.</p></a></li>
<li><a href='#pedigrees_all_populate_haplotypes'><p>Populate haplotypes in pedigrees (0-founder/unbounded).</p></a></li>
<li><a href='#pedigrees_all_populate_haplotypes_custom_founders'><p>Populate haplotypes in pedigrees (custom founder/unbounded).</p></a></li>
<li><a href='#pedigrees_all_populate_haplotypes_ladder_bounded'><p>Populate haplotypes in pedigrees (custom founder/bounded).</p></a></li>
<li><a href='#pedigrees_count'><p>Get number of pedigrees</p></a></li>
<li><a href='#pedigrees_table'><p>Get distribution of pedigree sizes</p></a></li>
<li><a href='#plot.malan_pedigree'><p>Plot pedigree</p></a></li>
<li><a href='#plot.malan_pedigreelist'><p>Plot pedigree list</p></a></li>
<li><a href='#population_populate_autosomal_infinite_alleles'><p>Populate 1-locus autosomal DNA profile in pedigrees with infinite alleles mutation model.</p></a></li>
<li><a href='#population_size_generation'><p>Size of population</p></a></li>
<li><a href='#print_individual'><p>Print individual</p></a></li>
<li><a href='#print.malan_pedigree'><p>Print pedigree</p></a></li>
<li><a href='#print.malan_pedigreelist'><p>Print pedigree list</p></a></li>
<li><a href='#print.malan_population'><p>Print population</p></a></li>
<li><a href='#print.malan_population_abort'><p>Print <code>malan_population_abort</code></p></a></li>
<li><a href='#relationship_allele_diff_dist'><p>Calculate distribution of allele difference</p></a></li>
<li><a href='#relationship_allele_diff_dist_sym'><p>Calculate distribution of allele difference for symmetric mutation rates</p></a></li>
<li><a href='#sample_autosomal_genotype'><p>Sample genotype with theta</p></a></li>
<li><a href='#sample_geneology'><p>Simulate a geneology with constant population size.</p></a></li>
<li><a href='#sample_geneology_varying_size'><p>Simulate a geneology with varying population size.</p></a></li>
<li><a href='#set_generation'><p>Set individual's generation number</p></a></li>
<li><a href='#split_by_haplotypes'><p>Split pids by haplotype</p></a></li>
<li><a href='#test_create_population'><p>Generate test population</p></a></li>
<li><a href='#ystr_kits'><p>Kit information about Y-STR markers</p></a></li>
<li><a href='#ystr_markers'><p>Mutational information about Y-STR markers</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>MAle Lineage ANalysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Author:</td>
<td>Mikkel Meyer Andersen [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mikkel Meyer Andersen &lt;mikl@math.aau.dk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>MAle Lineage ANalysis by simulating 
    genealogies backwards and imposing short tandem repeats (STR) mutations forwards. 
    Intended for forensic Y chromosomal STR (Y-STR) haplotype analyses. 
    Numerous analyses are possible, e.g. number of matches and meiotic distance to matches. 
    Refer to papers mentioned in citation("malan") (DOI's: 
    &lt;<a href="https://doi.org/10.1371%2Fjournal.pgen.1007028">doi:10.1371/journal.pgen.1007028</a>&gt;, 
    &lt;<a href="https://doi.org/10.21105%2Fjoss.00684">doi:10.21105/joss.00684</a>&gt; and 
    &lt;<a href="https://doi.org/10.1016%2Fj.fsigen.2018.10.004">doi:10.1016/j.fsigen.2018.10.004</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | file LICENSE</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.7), RcppProgress (&ge; 0.2.1), RcppArmadillo (&ge;
0.9.880.1.0), igraph (&ge; 1.0.1), tibble (&ge; 1.1), magrittr (&ge;
1.5), methods</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6), dplyr (&ge; 0.7.3), tidygraph (&ge; 1.0.0.9999)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo, RcppProgress</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, ggraph, dirmult, Rmpfr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mikldk/malan/issues">https://github.com/mikldk/malan/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-11-28 17:28:26 UTC; mikl</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-11-28 17:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='+5B+5B.malan_pedigreelist'>Get pedigree from pedigree list</h2><span id='topic++5B+5B.malan_pedigreelist'></span>

<h3>Description</h3>

<p>Get pedigree from <code>malan_pedigreelist</code> generated by
<code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_pedigreelist'
x[[...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.malan_pedigreelist_+3A_x">x</code></td>
<td>
<p>Element id</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.malan_pedigreelist_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Pedigree
</p>

<hr>
<h2 id='+5B+5B.malan_population'>Get individual from population by pid</h2><span id='topic++5B+5B.malan_population'></span>

<h3>Description</h3>

<p>Get individual from population by pid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_population'
x[[...]]
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B5B+2B5B.malan_population_+3A_x">x</code></td>
<td>
<p>pid</p>
</td></tr>
<tr><td><code id="+2B5B+2B5B.malan_population_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Individual
</p>

<hr>
<h2 id='analyse_mixture_result'>Analyse mixture results</h2><span id='topic+analyse_mixture_result'></span>

<h3>Description</h3>

<p>Calculate LR-like quantities by haplotype counts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_mixture_result(
  mix_res,
  unique_haps_in_mixture,
  unique_haps_in_mixture_counts
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyse_mixture_result_+3A_mix_res">mix_res</code></td>
<td>
<p>Mixture result from <code><a href="#topic+mixture_info_by_individuals_2pers">mixture_info_by_individuals_2pers()</a></code>,
<code><a href="#topic+mixture_info_by_individuals_3pers">mixture_info_by_individuals_3pers()</a></code>, <code><a href="#topic+mixture_info_by_individuals_4pers">mixture_info_by_individuals_4pers()</a></code>,
<code><a href="#topic+mixture_info_by_individuals_5pers">mixture_info_by_individuals_5pers()</a></code></p>
</td></tr>
<tr><td><code id="analyse_mixture_result_+3A_unique_haps_in_mixture">unique_haps_in_mixture</code></td>
<td>
<p>Included unique haplotypes to use as elements in contributor sets.</p>
</td></tr>
<tr><td><code id="analyse_mixture_result_+3A_unique_haps_in_mixture_counts">unique_haps_in_mixture_counts</code></td>
<td>
<p>Population counts of the included haplotypes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>NOTE: Only takes up to 9 contributors!
</p>


<h3>Value</h3>

<p>A list with numeric quantities
</p>

<hr>
<h2 id='analyse_mixture_results'>Analyse mixture results in a vectorised fashion</h2><span id='topic+analyse_mixture_results'></span>

<h3>Description</h3>

<p>Refer to <code><a href="#topic+analyse_mixture_result">analyse_mixture_result()</a></code> for details.
Essentially, <code><a href="#topic+analyse_mixture_result">analyse_mixture_result()</a></code> is run on each element of <code>mixture_results</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>analyse_mixture_results(
  mixture_results,
  unique_haps_in_mixture_list,
  unique_haps_in_mixture_counts_list
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="analyse_mixture_results_+3A_mixture_results">mixture_results</code></td>
<td>
<p>List of <code>n</code> mixture results from <code><a href="#topic+mixture_info_by_individuals_2pers">mixture_info_by_individuals_2pers()</a></code>,
<code><a href="#topic+mixture_info_by_individuals_3pers">mixture_info_by_individuals_3pers()</a></code>, <code><a href="#topic+mixture_info_by_individuals_4pers">mixture_info_by_individuals_4pers()</a></code>,
<code><a href="#topic+mixture_info_by_individuals_5pers">mixture_info_by_individuals_5pers()</a></code></p>
</td></tr>
<tr><td><code id="analyse_mixture_results_+3A_unique_haps_in_mixture_list">unique_haps_in_mixture_list</code></td>
<td>
<p>List of <code>n</code> included unique haplotypes, one for each element in <code>mix_res</code></p>
</td></tr>
<tr><td><code id="analyse_mixture_results_+3A_unique_haps_in_mixture_counts_list">unique_haps_in_mixture_counts_list</code></td>
<td>
<p>List of <code>n</code> population counts of the included unique haplotypes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>NOTE: Only takes up to 9 contributors!
</p>


<h3>Value</h3>

<p>A list with lists of numeric quantities
</p>

<hr>
<h2 id='as_tbl_graph.malan_pedigreelist'>Get tidy graph object</h2><span id='topic+as_tbl_graph.malan_pedigreelist'></span>

<h3>Description</h3>

<p>Get tidy graph object <code><a href="tidygraph.html#topic+tbl_graph">tbl_graph()</a></code>, e.g.
to plot with <code>ggraph()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_pedigreelist'
as_tbl_graph(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_tbl_graph.malan_pedigreelist_+3A_x">x</code></td>
<td>
<p><code>malan_pedigreelist</code></p>
</td></tr>
<tr><td><code id="as_tbl_graph.malan_pedigreelist_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code><a href="tidygraph.html#topic+tbl_graph">tbl_graph()</a></code> object
</p>

<hr>
<h2 id='brothers_matching'>Number of brothers with matching haplotype</h2><span id='topic+brothers_matching'></span>

<h3>Description</h3>

<p>Get individual's number of brothers that matches <code>individual</code>'s haplotype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>brothers_matching(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="brothers_matching_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of brothers that matches <code>individual</code>'s haplotype
</p>

<hr>
<h2 id='build_haplotype_hashmap'>Build hashmap of haplotype to individuals</h2><span id='topic+build_haplotype_hashmap'></span>

<h3>Description</h3>

<p>Makes it possible to find all individuals' pid with a certain haplotype.
Must be used with e.g. <code><a href="#topic+get_matching_pids_from_hashmap">get_matching_pids_from_hashmap()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_haplotype_hashmap(individuals, progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_haplotype_hashmap_+3A_individuals">individuals</code></td>
<td>
<p>List of individuals to build hashmap of</p>
</td></tr>
<tr><td><code id="build_haplotype_hashmap_+3A_progress">progress</code></td>
<td>
<p>Show progress?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>External pointer to hashmap with haplotype as keys and vector of individuals' pid as value
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_matching_pids_from_hashmap">get_matching_pids_from_hashmap()</a></code>.
</p>

<hr>
<h2 id='build_pedigrees'>Build pedigrees from (individuals in) a population.</h2><span id='topic+build_pedigrees'></span>

<h3>Description</h3>

<p>In a newly simulated population, each individual only knows its father and children.
Using this information, this function builds pedigrees.
This makes it easier to e.g. population haplotypes, find path between two individuals
(if they are not in the same pedigree, they are not connected).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_pedigrees(population, progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_pedigrees_+3A_population">population</code></td>
<td>
<p>Population generated by <code><a href="#topic+sample_geneology">sample_geneology()</a></code> or <code><a href="#topic+sample_geneology_varying_size">sample_geneology_varying_size()</a></code>.</p>
</td></tr>
<tr><td><code id="build_pedigrees_+3A_progress">progress</code></td>
<td>
<p>Show progress.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object with class <code>malan_pedigreelist</code> (an internal list of external pointers to pedigrees).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sample_geneology">sample_geneology()</a></code> and <code><a href="#topic+sample_geneology_varying_size">sample_geneology_varying_size()</a></code> for simulating populations.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
str(sim, 1)
sim$population
peds &lt;- build_pedigrees(sim$population)
peds

</code></pre>

<hr>
<h2 id='calc_autosomal_genotype_conditional_cumdist'>Calculate conditional genotype cumulative probabilities with theta</h2><span id='topic+calc_autosomal_genotype_conditional_cumdist'></span>

<h3>Description</h3>

<p>Calculate conditional genotype cumulative probabilities with theta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_autosomal_genotype_conditional_cumdist(allele_dist, theta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_autosomal_genotype_conditional_cumdist_+3A_allele_dist">allele_dist</code></td>
<td>
<p>Allele distribution (probabilities) &ndash; gets normalised</p>
</td></tr>
<tr><td><code id="calc_autosomal_genotype_conditional_cumdist_+3A_theta">theta</code></td>
<td>
<p>Theta correction between 0 and 1 (both included)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix: row i: conditional cumulative distribution of alleles given allele i
</p>

<hr>
<h2 id='calc_autosomal_genotype_probs'>Calculate genotype probabilities with theta</h2><span id='topic+calc_autosomal_genotype_probs'></span>

<h3>Description</h3>

<p>Calculate genotype probabilities with theta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_autosomal_genotype_probs(allele_dist, theta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calc_autosomal_genotype_probs_+3A_allele_dist">allele_dist</code></td>
<td>
<p>Allele distribution (probabilities) &ndash; gets normalised</p>
</td></tr>
<tr><td><code id="calc_autosomal_genotype_probs_+3A_theta">theta</code></td>
<td>
<p>Theta correction between 0 and 1 (both included)</p>
</td></tr>
</table>

<hr>
<h2 id='construct_M'>Construct M matrix</h2><span id='topic+construct_M'></span>

<h3>Description</h3>

<p>Construct M matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>construct_M(meioses, mu_dw, mu_up)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="construct_M_+3A_meioses">meioses</code></td>
<td>
<p>number of meioses separating the two individuals</p>
</td></tr>
<tr><td><code id="construct_M_+3A_mu_dw">mu_dw</code></td>
<td>
<p>mutation rate for 1-step down-mutation</p>
</td></tr>
<tr><td><code id="construct_M_+3A_mu_up">mu_up</code></td>
<td>
<p>mutation rate for 1-step up-mutation</p>
</td></tr>
</table>

<hr>
<h2 id='count_brothers'>Number of brothers</h2><span id='topic+count_brothers'></span>

<h3>Description</h3>

<p>Get individual's number of brothers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_brothers(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_brothers_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of brothers
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_brothers">get_brothers()</a></code>
</p>

<hr>
<h2 id='count_haplotype_near_matches_individuals'>Count near haplotype matches in list of individuals</h2><span id='topic+count_haplotype_near_matches_individuals'></span>

<h3>Description</h3>

<p>Counts the number of types close to <code>haplotype</code> in <code>individuals</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_haplotype_near_matches_individuals(individuals, haplotype, max_dist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_haplotype_near_matches_individuals_+3A_individuals">individuals</code></td>
<td>
<p>List of individuals to count occurrences in.</p>
</td></tr>
<tr><td><code id="count_haplotype_near_matches_individuals_+3A_haplotype">haplotype</code></td>
<td>
<p>Haplotype to count near-matches occurrences of.</p>
</td></tr>
<tr><td><code id="count_haplotype_near_matches_individuals_+3A_max_dist">max_dist</code></td>
<td>
<p>Maximum distance (0 = match, 1 = 1 STR allele difference, ...)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of times that a haplotype within a radius of <code>max_dist</code> of
<code>haplotype</code> occurred amongst <code>individuals</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+count_haplotype_occurrences_individuals">count_haplotype_occurrences_individuals()</a></code>,
<code><a href="#topic+pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists">pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists()</a></code>.
</p>

<hr>
<h2 id='count_haplotype_occurrences_individuals'>Count haplotypes occurrences in list of individuals</h2><span id='topic+count_haplotype_occurrences_individuals'></span>

<h3>Description</h3>

<p>Counts the number of types <code>haplotype</code> appears in <code>individuals</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_haplotype_occurrences_individuals(individuals, haplotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_haplotype_occurrences_individuals_+3A_individuals">individuals</code></td>
<td>
<p>List of individuals to count occurrences in.</p>
</td></tr>
<tr><td><code id="count_haplotype_occurrences_individuals_+3A_haplotype">haplotype</code></td>
<td>
<p>Haplotype to count occurrences of.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of times that <code>haplotype</code> occurred amongst <code>individuals</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists">pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists()</a></code>,
<code><a href="#topic+count_haplotype_near_matches_individuals">count_haplotype_near_matches_individuals()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes(peds, 2, c(0, 0))
count_haplotype_occurrences_individuals(sim$end_generation_individuals, c(0, 0))

</code></pre>

<hr>
<h2 id='count_haplotype_occurrences_pedigree'>Count haplotypes occurrences in pedigree</h2><span id='topic+count_haplotype_occurrences_pedigree'></span>

<h3>Description</h3>

<p>Counts the number of types <code>haplotype</code> appears in <code>pedigree</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_haplotype_occurrences_pedigree(
  pedigree,
  haplotype,
  generation_upper_bound_in_result = -1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_haplotype_occurrences_pedigree_+3A_pedigree">pedigree</code></td>
<td>
<p>Pedigree to count occurrences in.</p>
</td></tr>
<tr><td><code id="count_haplotype_occurrences_pedigree_+3A_haplotype">haplotype</code></td>
<td>
<p>Haplotype to count occurrences of.</p>
</td></tr>
<tr><td><code id="count_haplotype_occurrences_pedigree_+3A_generation_upper_bound_in_result">generation_upper_bound_in_result</code></td>
<td>
<p>Only consider matches in
generation 0, 1, ... generation_upper_bound_in_result.
-1 means disabled, consider all generations.
End generation is generation 0.
Second last generation is 1.
And so on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of times that <code>haplotype</code> occurred in <code>pedigree</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists">pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists()</a></code>.
</p>

<hr>
<h2 id='count_uncles'>Number of uncles</h2><span id='topic+count_uncles'></span>

<h3>Description</h3>

<p>Get individual's number of uncles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_uncles(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_uncles_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of uncles
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_uncles">get_uncles()</a></code>
</p>

<hr>
<h2 id='delete_haplotypeids_hashmap'>Delete haplotype hashmap</h2><span id='topic+delete_haplotypeids_hashmap'></span>

<h3>Description</h3>

<p>Delete hashmap made by <code><a href="#topic+build_haplotype_hashmap">build_haplotype_hashmap()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delete_haplotypeids_hashmap(hashmap)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delete_haplotypeids_hashmap_+3A_hashmap">hashmap</code></td>
<td>
<p>Hashmap made by <code><a href="#topic+build_haplotype_hashmap">build_haplotype_hashmap()</a></code></p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+get_matching_pids_from_hashmap">get_matching_pids_from_hashmap()</a></code>
and <code><a href="#topic+build_haplotype_hashmap">build_haplotype_hashmap()</a></code>.
</p>

<hr>
<h2 id='estimate_autotheta_1subpop_genotypes'>Estimate autosomal theta from genotypes</h2><span id='topic+estimate_autotheta_1subpop_genotypes'></span>

<h3>Description</h3>

<p>Estimate autosomal theta for one subpopulation given a sample of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_1subpop_genotypes(genotypes, return_estimation_info = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_1subpop_genotypes_+3A_genotypes">genotypes</code></td>
<td>
<p>Matrix of genotypes: two columns (allele1 and allele2) and a row per individual</p>
</td></tr>
<tr><td><code id="estimate_autotheta_1subpop_genotypes_+3A_return_estimation_info">return_estimation_info</code></td>
<td>
<p>Whether to return the quantities used to estimate <code>theta</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>


<h3>Value</h3>

<p>List:
</p>

<ul>
<li> <p><code>theta</code>
</p>

<ul>
<li> <p><code>estimate</code>: Vector of length 1 containing estimate of theta or NA if it could not be estimated
</p>
</li>
<li> <p><code>error</code>: true if an error happened, false otherwise
</p>
</li>
<li> <p><code>details</code>: contains description if an error happened
</p>
</li>
<li> <p><code>estimation_info</code>: If <code>return_estimation_info = true</code>: a list with information used to estimate <code>theta</code>. Else <code>NULL</code>.
</p>
</li></ul>

</li></ul>


<hr>
<h2 id='estimate_autotheta_1subpop_individuals'>Estimate autosomal theta from individuals</h2><span id='topic+estimate_autotheta_1subpop_individuals'></span>

<h3>Description</h3>

<p>Estimate autosomal theta for one subpopulation given a list of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_1subpop_individuals(
  individuals,
  return_estimation_info = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_1subpop_individuals_+3A_individuals">individuals</code></td>
<td>
<p>Individuals to get haplotypes for.</p>
</td></tr>
<tr><td><code id="estimate_autotheta_1subpop_individuals_+3A_return_estimation_info">return_estimation_info</code></td>
<td>
<p>Whether to return the quantities used to estimate <code>theta</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>


<h3>Value</h3>

<p>List:
</p>

<ul>
<li> <p><code>theta</code>
</p>

<ul>
<li> <p><code>estimate</code>: Vector of length 1 containing estimate of theta or NA if it could not be estimated
</p>
</li>
<li> <p><code>error</code>: true if an error happened, false otherwise
</p>
</li>
<li> <p><code>details</code>: contains description if an error happened
</p>
</li>
<li> <p><code>estimation_info</code>: If <code>return_estimation_info = true</code>: a list with information used to estimate <code>theta</code>. Else <code>NULL</code>.
</p>
</li></ul>

</li></ul>


<hr>
<h2 id='estimate_autotheta_subpops_genotypes'>Estimate autosomal F, theta, and f from subpopulations of genotypes</h2><span id='topic+estimate_autotheta_subpops_genotypes'></span>

<h3>Description</h3>

<p>Estimates autosomal F, theta, and f for a number of subpopulations given a list of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_subpops_genotypes(subpops, subpops_sizes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_subpops_genotypes_+3A_subpops">subpops</code></td>
<td>
<p>List of subpopulations, each a list of individuals</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_genotypes_+3A_subpops_sizes">subpops_sizes</code></td>
<td>
<p>Size of each subpopulation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>
<p>Based on Bruce S Weir, Genetic Data Analysis 2, 1996. (GDA2).
</p>


<h3>Value</h3>

<p>Estimates of autosomal F, theta, and f as well as additional information
</p>

<hr>
<h2 id='estimate_autotheta_subpops_individuals'>Estimate autosomal F, theta, and f from subpopulations of individuals</h2><span id='topic+estimate_autotheta_subpops_individuals'></span>

<h3>Description</h3>

<p>Estimates autosomal F, theta, and f for a number of subpopulations given a list of individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_subpops_individuals(subpops, subpops_sizes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_subpops_individuals_+3A_subpops">subpops</code></td>
<td>
<p>List of subpopulations, each a list of individuals</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_individuals_+3A_subpops_sizes">subpops_sizes</code></td>
<td>
<p>Size of each subpopulation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>
<p>Based on Bruce S Weir, Genetic Data Analysis 2, 1996. (GDA2).
</p>


<h3>Value</h3>

<p>Estimates of autosomal F, theta, and f as well as additional information
</p>

<hr>
<h2 id='estimate_autotheta_subpops_pids'>Estimate autosomal F, theta, and f from subpopulations of individual ids</h2><span id='topic+estimate_autotheta_subpops_pids'></span>

<h3>Description</h3>

<p>Estimates autosomal F, theta, and f for a number of subpopulations given a list of pids (individual ids).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_subpops_pids(population, subpops, subpops_sizes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_subpops_pids_+3A_population">population</code></td>
<td>
<p>Population obtain from simulation</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_pids_+3A_subpops">subpops</code></td>
<td>
<p>List of individual pids</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_pids_+3A_subpops_sizes">subpops_sizes</code></td>
<td>
<p>Size of each subpopulation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>
<p>Based on Bruce S Weir, Genetic Data Analysis 2, 1996. (GDA2).
</p>


<h3>Value</h3>

<p>Estimates of autosomal F, theta, and f as well as additional information
</p>

<hr>
<h2 id='estimate_autotheta_subpops_unweighted_genotypes'>Unweighted estimate of autosomal theta from subpopulations of genotypes</h2><span id='topic+estimate_autotheta_subpops_unweighted_genotypes'></span>

<h3>Description</h3>

<p>Estimates unweighted autosomal theta for a number of subpopulations given a list of subpopulations of genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_subpops_unweighted_genotypes(subpops, assume_HWE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_subpops_unweighted_genotypes_+3A_subpops">subpops</code></td>
<td>
<p>List of individual genotypes</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_unweighted_genotypes_+3A_assume_hwe">assume_HWE</code></td>
<td>
<p>if the alleles themselves are used instead of genotypes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>
<p>Based on Weir and Goudet, Genetics 2017:
http://www.genetics.org/content/early/2017/05/26/genetics.116.198424
</p>


<h3>Value</h3>

<p>Estimate of autosomal theta
</p>

<hr>
<h2 id='estimate_autotheta_subpops_unweighted_pids'>Unweighted estimate of autosomal theta from subpopulations of individual ids</h2><span id='topic+estimate_autotheta_subpops_unweighted_pids'></span>

<h3>Description</h3>

<p>Estimates unweighted autosomal theta for a number of subpopulations given a list of pids (individual ids).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_autotheta_subpops_unweighted_pids(population, subpops, assume_HWE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_autotheta_subpops_unweighted_pids_+3A_population">population</code></td>
<td>
<p>Population obtain from simulation</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_unweighted_pids_+3A_subpops">subpops</code></td>
<td>
<p>List of individual pids</p>
</td></tr>
<tr><td><code id="estimate_autotheta_subpops_unweighted_pids_+3A_assume_hwe">assume_HWE</code></td>
<td>
<p>if the alleles themselves are used instead of genotypes</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>
<p>Based on Weir and Goudet, Genetics 2017:
http://www.genetics.org/content/early/2017/05/26/genetics.116.198424
</p>


<h3>Value</h3>

<p>Estimate of autosomal theta
</p>

<hr>
<h2 id='father_matches'>Father matches</h2><span id='topic+father_matches'></span>

<h3>Description</h3>

<p>Does the father have the same profile as <code>individual</code>?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>father_matches(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="father_matches_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Whether father has the same profile as <code>individual</code> or not
</p>

<hr>
<h2 id='from_igraph'>Convert igraph to population</h2><span id='topic+from_igraph'></span>

<h3>Description</h3>

<p>Convert igraph to population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>from_igraph(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="from_igraph_+3A_x">x</code></td>
<td>
<p>igraph, must be a forest of directed trees with unique positive integer names (as they will be pid's)</p>
</td></tr>
<tr><td><code id="from_igraph_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A population
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g &lt;- igraph::graph_from_literal( 2 +- 1 -+ 3, 4 -+ 5 )
plot(g)
pop &lt;- from_igraph(g)
peds &lt;- build_pedigrees(pop, progress = FALSE)
plot(peds)
infer_generations(peds)
get_generation(get_individual(pop, 1))
get_generation(get_individual(pop, 2))
get_generation(get_individual(pop, 3))
get_generation(get_individual(pop, 4))
get_generation(get_individual(pop, 5))

</code></pre>

<hr>
<h2 id='from_igraph_rcpp'>Generate paternal brothers population</h2><span id='topic+from_igraph_rcpp'></span>

<h3>Description</h3>

<p>Generate paternal brothers population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>from_igraph_rcpp(vertices, edges)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="from_igraph_rcpp_+3A_vertices">vertices</code></td>
<td>
<p>vector of vertices</p>
</td></tr>
<tr><td><code id="from_igraph_rcpp_+3A_edges">edges</code></td>
<td>
<p>matrix with edges</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An external pointer to the population.
</p>

<hr>
<h2 id='generate_get_founder_haplotype_db'>Generate a function to simulate pedigree founder haplotype based on a haplotype databasep</h2><span id='topic+generate_get_founder_haplotype_db'></span>

<h3>Description</h3>

<p>Generate a function to simulate pedigree founder haplotype based on a haplotype databasep
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_get_founder_haplotype_db(db)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_get_founder_haplotype_db_+3A_db">db</code></td>
<td>
<p>data frame or matrix with haplotypes from which the founder is randomly simulated</p>
</td></tr>
</table>

<hr>
<h2 id='generate_get_founder_haplotype_ladder'>Generate a function to simulate pedigree founder haplotype based on ladder information</h2><span id='topic+generate_get_founder_haplotype_ladder'></span>

<h3>Description</h3>

<p>Generate a function to simulate pedigree founder haplotype based on ladder information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate_get_founder_haplotype_ladder(ladder_min, ladder_max)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="generate_get_founder_haplotype_ladder_+3A_ladder_min">ladder_min</code></td>
<td>
<p>vector of minimum alleles; <code>ladder_min[i]</code> is the minimum allele at locus i</p>
</td></tr>
<tr><td><code id="generate_get_founder_haplotype_ladder_+3A_ladder_max">ladder_max</code></td>
<td>
<p>vector of minimum alleles; <code>ladder_max[i]</code> is the maximum allele at locus i</p>
</td></tr>
</table>

<hr>
<h2 id='get_allele_counts_genotypes'>Get autosomal allele counts from subpopulations of genotypes</h2><span id='topic+get_allele_counts_genotypes'></span>

<h3>Description</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_counts_genotypes(subpops)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_counts_genotypes_+3A_subpops">subpops</code></td>
<td>
<p>List of individual genotypes</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with allele counts
</p>

<hr>
<h2 id='get_allele_counts_pids'>Get autosomal allele counts from subpopulations given by pids</h2><span id='topic+get_allele_counts_pids'></span>

<h3>Description</h3>

<p>Assumes that <code><a href="#topic+pedigrees_all_populate_autosomal">pedigrees_all_populate_autosomal()</a></code> was used first to populate autosomal genotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_allele_counts_pids(population, subpops)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_allele_counts_pids_+3A_population">population</code></td>
<td>
<p>Population obtain from simulation</p>
</td></tr>
<tr><td><code id="get_allele_counts_pids_+3A_subpops">subpops</code></td>
<td>
<p>List of individual pids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with allele counts
</p>

<hr>
<h2 id='get_brothers'>Get brothers</h2><span id='topic+get_brothers'></span>

<h3>Description</h3>

<p>Get individual's brothers
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_brothers(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_brothers_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with brothers
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_father">get_father()</a></code>, <code><a href="#topic+get_uncles">get_uncles()</a></code>, <code><a href="#topic+get_children">get_children()</a></code>, <code><a href="#topic+get_cousins">get_cousins()</a></code>
</p>

<hr>
<h2 id='get_children'>Get children</h2><span id='topic+get_children'></span>

<h3>Description</h3>

<p>Get individual's children
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_children(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_children_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with children
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_father">get_father()</a></code>, <code><a href="#topic+get_brothers">get_brothers()</a></code>, <code><a href="#topic+get_uncles">get_uncles()</a></code>, <code><a href="#topic+get_cousins">get_cousins()</a></code>
</p>

<hr>
<h2 id='get_cousins'>Get cousins</h2><span id='topic+get_cousins'></span>

<h3>Description</h3>

<p>Get individual's cousins
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cousins(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cousins_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with cousins
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_brothers">get_brothers()</a></code>, <code><a href="#topic+get_uncles">get_uncles()</a></code>, <code><a href="#topic+get_children">get_children()</a></code>
</p>

<hr>
<h2 id='get_family_info'>Get individual's family information</h2><span id='topic+get_family_info'></span>

<h3>Description</h3>

<p>Get individual's family information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_family_info(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_family_info_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with family information
</p>

<hr>
<h2 id='get_father'>Get father</h2><span id='topic+get_father'></span>

<h3>Description</h3>

<p>Get individual's father
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_father(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_father_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Father
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_brothers">get_brothers()</a></code>, <code><a href="#topic+get_uncles">get_uncles()</a></code>, <code><a href="#topic+get_children">get_children()</a></code>, <code><a href="#topic+get_cousins">get_cousins()</a></code>
</p>

<hr>
<h2 id='get_generation'>Get individual's generation number</h2><span id='topic+get_generation'></span>

<h3>Description</h3>

<p>Note that generation 0 is final, end generation.
1 is second last generation etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_generation(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_generation_+3A_individual">individual</code></td>
<td>
<p>Individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>generation
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
indv &lt;- get_individual(sim$population, 1)
get_generation(indv)

</code></pre>

<hr>
<h2 id='get_haplotype'>Get haplotype from an individual</h2><span id='topic+get_haplotype'></span>

<h3>Description</h3>

<p>Requires that haplotypes are first populated, e.g.
with <code><a href="#topic+pedigrees_all_populate_haplotypes">pedigrees_all_populate_haplotypes()</a></code>,
<code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code>, or
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_haplotype(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_haplotype_+3A_individual">individual</code></td>
<td>
<p>Individual to get haplotypes for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Haplotype for <code>individual</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_haplotypes_individuals">get_haplotypes_individuals()</a></code> and <code><a href="#topic+get_haplotypes_pids">get_haplotypes_pids()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes(peds, 2, c(1, 1))
get_haplotype(sim$end_generation_individuals[[1]])

</code></pre>

<hr>
<h2 id='get_haplotypes_in_pedigree'>Get haplotypes in pedigree</h2><span id='topic+get_haplotypes_in_pedigree'></span>

<h3>Description</h3>

<p>Get haplotypes in pedigree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_haplotypes_in_pedigree(ped)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_haplotypes_in_pedigree_+3A_ped">ped</code></td>
<td>
<p>Pedigree</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with haplotypes
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes(peds, 2, c(1, 1))
get_haplotypes_in_pedigree(peds[[1]])

</code></pre>

<hr>
<h2 id='get_haplotypes_individuals'>Get haplotype matrix from list of individuals</h2><span id='topic+get_haplotypes_individuals'></span>

<h3>Description</h3>

<p>Requires that haplotypes are first populated, e.g.
with <code><a href="#topic+pedigrees_all_populate_haplotypes">pedigrees_all_populate_haplotypes()</a></code>,
<code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code>, or
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_haplotypes_individuals(individuals)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_haplotypes_individuals_+3A_individuals">individuals</code></td>
<td>
<p>Individuals to get haplotypes for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of haplotypes where row <code>i</code> is the haplotype of <code>individuals[[i]]</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_haplotypes_pids">get_haplotypes_pids()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes(peds, 2, c(1, 1))
get_haplotypes_individuals(sim$end_generation_individuals)

</code></pre>

<hr>
<h2 id='get_haplotypes_pids'>Get haplotypes from a vector of pids.</h2><span id='topic+get_haplotypes_pids'></span>

<h3>Description</h3>

<p>Requires that haplotypes are first populated, e.g.
with <code><a href="#topic+pedigrees_all_populate_haplotypes">pedigrees_all_populate_haplotypes()</a></code>,
<code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code>, or
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_haplotypes_pids(population, pids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_haplotypes_pids_+3A_population">population</code></td>
<td>
<p>Population</p>
</td></tr>
<tr><td><code id="get_haplotypes_pids_+3A_pids">pids</code></td>
<td>
<p>Vector of pids to get haplotypes for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix of haplotypes where row <code>i</code> is the haplotype of <code>individuals[[i]]</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_haplotypes_individuals">get_haplotypes_individuals()</a></code>.
</p>

<hr>
<h2 id='get_individual'>Get individual by pid</h2><span id='topic+get_individual'></span>

<h3>Description</h3>

<p>Get individual by pid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_individual(population, pid)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_individual_+3A_population">population</code></td>
<td>
<p>Population</p>
</td></tr>
<tr><td><code id="get_individual_+3A_pid">pid</code></td>
<td>
<p>pid</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Individual
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
indv &lt;- get_individual(sim$population, 1)
get_pid(indv)

</code></pre>

<hr>
<h2 id='get_individuals'>Get all individuals in population</h2><span id='topic+get_individuals'></span>

<h3>Description</h3>

<p>Get all individuals in population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_individuals(population)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_individuals_+3A_population">population</code></td>
<td>
<p>Population</p>
</td></tr>
</table>

<hr>
<h2 id='get_matching_pids_from_hashmap'>Get individuals with a certain haplotype id by hashmap lookup</h2><span id='topic+get_matching_pids_from_hashmap'></span>

<h3>Description</h3>

<p>By using hashmap made by <code><a href="#topic+build_haplotype_hashmap">build_haplotype_hashmap()</a></code>,
it is easy to get all individuals with a certain haplotype id.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_matching_pids_from_hashmap(hashmap, haplotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_matching_pids_from_hashmap_+3A_hashmap">hashmap</code></td>
<td>
<p>Hashmap to make lookup in, made by <code><a href="#topic+build_haplotype_hashmap">build_haplotype_hashmap()</a></code></p>
</td></tr>
<tr><td><code id="get_matching_pids_from_hashmap_+3A_haplotype">haplotype</code></td>
<td>
<p>to get individuals that has this haplotype id</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of individuals with a given haplotype id
</p>


<h3>See Also</h3>

<p><code><a href="#topic+build_haplotype_hashmap">build_haplotype_hashmap()</a></code>.
</p>

<hr>
<h2 id='get_nodes_edges'>Get nodes and edges</h2><span id='topic+get_nodes_edges'></span>

<h3>Description</h3>

<p>Get nodes and edges in <code>malan_pedigreelist</code>.
For example to plot via <code><a href="tidygraph.html#topic+as_tbl_graph">as_tbl_graph()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_nodes_edges(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_nodes_edges_+3A_x">x</code></td>
<td>
<p><code>malan_pedigreelist</code></p>
</td></tr>
<tr><td><code id="get_nodes_edges_+3A_...">...</code></td>
<td>
<p>Ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with entries <code>nodes</code> and <code>edges</code>
</p>

<hr>
<h2 id='get_pedigree_as_graph'>Get pedigree information as graph (mainly intended for plotting)</h2><span id='topic+get_pedigree_as_graph'></span>

<h3>Description</h3>

<p>Get pedigree information as graph (mainly intended for plotting)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigree_as_graph(ped)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pedigree_as_graph_+3A_ped">ped</code></td>
<td>
<p>Pedigree</p>
</td></tr>
</table>

<hr>
<h2 id='get_pedigree_from_individual'>Get pedigree from individual</h2><span id='topic+get_pedigree_from_individual'></span>

<h3>Description</h3>

<p>Get pedigree from individual
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigree_from_individual(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pedigree_from_individual_+3A_individual">individual</code></td>
<td>
<p>Individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pedigree
</p>

<hr>
<h2 id='get_pedigree_id'>Get pedigree id</h2><span id='topic+get_pedigree_id'></span>

<h3>Description</h3>

<p>Get pedigree id
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigree_id(ped)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pedigree_id_+3A_ped">ped</code></td>
<td>
<p>Pedigree</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
get_pedigree_id(peds[[1]])

</code></pre>

<hr>
<h2 id='get_pedigree_id_from_pid'>Get pedigree ids from pids</h2><span id='topic+get_pedigree_id_from_pid'></span>

<h3>Description</h3>

<p>Get pedigree ids from pids
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigree_id_from_pid(population, pids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pedigree_id_from_pid_+3A_population">population</code></td>
<td>
<p>Population</p>
</td></tr>
<tr><td><code id="get_pedigree_id_from_pid_+3A_pids">pids</code></td>
<td>
<p>Pids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector with pedigree ids
</p>

<hr>
<h2 id='get_pedigrees_tidy'>Get pedigrees information in tidy format</h2><span id='topic+get_pedigrees_tidy'></span>

<h3>Description</h3>

<p>Get pedigrees information in tidy format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pedigrees_tidy(pedigrees)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pedigrees_tidy_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigrees</p>
</td></tr>
</table>

<hr>
<h2 id='get_pid'>Get pid from individual</h2><span id='topic+get_pid'></span>

<h3>Description</h3>

<p>Get pid from individual
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pid(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pid_+3A_individual">individual</code></td>
<td>
<p>Individual to get pid of</p>
</td></tr>
</table>


<h3>Value</h3>

<p>pid
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
indv &lt;- get_individual(sim$population, 1)
get_pid(indv)

</code></pre>

<hr>
<h2 id='get_pids_in_pedigree'>Get pids in pedigree</h2><span id='topic+get_pids_in_pedigree'></span>

<h3>Description</h3>

<p>Get pids in pedigree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pids_in_pedigree(ped)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_pids_in_pedigree_+3A_ped">ped</code></td>
<td>
<p>Pedigree</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
get_pids_in_pedigree(peds[[1]])

</code></pre>

<hr>
<h2 id='get_uncles'>Get uncles</h2><span id='topic+get_uncles'></span>

<h3>Description</h3>

<p>Get individual's uncles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_uncles(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_uncles_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with uncles
</p>


<h3>See Also</h3>

<p><code><a href="#topic+get_brothers">get_brothers()</a></code>, <code><a href="#topic+get_children">get_children()</a></code>, <code><a href="#topic+get_cousins">get_cousins()</a></code>
</p>

<hr>
<h2 id='get_zero_haplotype_generator'>Generate a function to generate the zero haplotype</h2><span id='topic+get_zero_haplotype_generator'></span>

<h3>Description</h3>

<p>Generate a function to generate the zero haplotype
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_zero_haplotype_generator(loci)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_zero_haplotype_generator_+3A_loci">loci</code></td>
<td>
<p>Number of loci</p>
</td></tr>
</table>

<hr>
<h2 id='grandfather_matches'>Grandfather matches</h2><span id='topic+grandfather_matches'></span>

<h3>Description</h3>

<p>Does the frandfather have the same profile as <code>individual</code>?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>grandfather_matches(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="grandfather_matches_+3A_individual">individual</code></td>
<td>
<p>individual</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Whether grandfather has the same profile as <code>individual</code> or not
</p>

<hr>
<h2 id='haplotype_matches_individuals'>Get individuals matching from list of individuals</h2><span id='topic+haplotype_matches_individuals'></span>

<h3>Description</h3>

<p>Get the indvididuals that matches <code>haplotype</code> in <code>individuals</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotype_matches_individuals(individuals, haplotype)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplotype_matches_individuals_+3A_individuals">individuals</code></td>
<td>
<p>List of individuals to count occurrences in.</p>
</td></tr>
<tr><td><code id="haplotype_matches_individuals_+3A_haplotype">haplotype</code></td>
<td>
<p>Haplotype to count occurrences of.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of individuals that matches <code>haplotype</code> amongst <code>individuals</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists">pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists()</a></code>.
</p>

<hr>
<h2 id='haplotype_partially_matches_individuals'>Get individuals partially matching from list of individuals</h2><span id='topic+haplotype_partially_matches_individuals'></span>

<h3>Description</h3>

<p>Get the indvididuals that partially matches <code>haplotype</code> in <code>individuals</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotype_partially_matches_individuals(
  individuals,
  haplotype,
  ignore_loci = as.integer(c())
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplotype_partially_matches_individuals_+3A_individuals">individuals</code></td>
<td>
<p>List of individuals to count occurrences in.</p>
</td></tr>
<tr><td><code id="haplotype_partially_matches_individuals_+3A_haplotype">haplotype</code></td>
<td>
<p>Haplotype to count occurrences of.</p>
</td></tr>
<tr><td><code id="haplotype_partially_matches_individuals_+3A_ignore_loci">ignore_loci</code></td>
<td>
<p>Vector of loci to ignore (1 = ignore first locus etc.)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of individuals that partially matches <code>haplotype</code> amongst <code>individuals</code>.
</p>

<hr>
<h2 id='haplotypes_to_hashes'>Convert haplotypes to hashes (integers)</h2><span id='topic+haplotypes_to_hashes'></span>

<h3>Description</h3>

<p>Individuals with the same haplotype will have the same hash (integer)
and individuals with different haplotypes will have different hashes (integers).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>haplotypes_to_hashes(population, pids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="haplotypes_to_hashes_+3A_population">population</code></td>
<td>
<p>Population obtained from simulation</p>
</td></tr>
<tr><td><code id="haplotypes_to_hashes_+3A_pids">pids</code></td>
<td>
<p>Vector of individual pids</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can be useful if for example using haplotypes to define groups
and the haplotype itself is not of interest.
</p>


<h3>Value</h3>

<p>Integer vector with haplotype hashes
</p>

<hr>
<h2 id='infer_generation'>Infer individual's generation number</h2><span id='topic+infer_generation'></span>

<h3>Description</h3>

<p>Takes as input final generation, then moves up in pedigree and increments
generation number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_generation(final_generation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infer_generation_+3A_final_generation">final_generation</code></td>
<td>
<p>Individuals in final generation</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: Only works when all final generation individuals are provided.
</p>

<hr>
<h2 id='infer_generations'>Infer generation numbers from pedigrees</h2><span id='topic+infer_generations'></span>

<h3>Description</h3>

<p>Infer generation numbers from pedigrees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>infer_generations(peds)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="infer_generations_+3A_peds">peds</code></td>
<td>
<p>Pedigrees infered by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>

<hr>
<h2 id='load_haplotypes'>Load haplotypes to individuals</h2><span id='topic+load_haplotypes'></span>

<h3>Description</h3>

<p>Note that individuals loaded this way does not have information about generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_haplotypes(population, pid, haplotypes, progress = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_haplotypes_+3A_population">population</code></td>
<td>
<p>of individuals</p>
</td></tr>
<tr><td><code id="load_haplotypes_+3A_pid">pid</code></td>
<td>
<p>ID of male</p>
</td></tr>
<tr><td><code id="load_haplotypes_+3A_haplotypes">haplotypes</code></td>
<td>

<ul>
<li><p> row <code>i</code> has <code>pid[i]</code> ID
</p>
</li></ul>
</td></tr>
<tr><td><code id="load_haplotypes_+3A_progress">progress</code></td>
<td>
<p>Show progress.</p>
</td></tr>
</table>

<hr>
<h2 id='load_individuals'>Construct a population from data</h2><span id='topic+load_individuals'></span>

<h3>Description</h3>

<p>Note that individuals loaded this way does not have information about generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_individuals(pid, pid_dad, progress = TRUE, error_on_pid_not_found = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="load_individuals_+3A_pid">pid</code></td>
<td>
<p>ID of male</p>
</td></tr>
<tr><td><code id="load_individuals_+3A_pid_dad">pid_dad</code></td>
<td>
<p>ID of male's father, 0 if not known</p>
</td></tr>
<tr><td><code id="load_individuals_+3A_progress">progress</code></td>
<td>
<p>Show progress.</p>
</td></tr>
<tr><td><code id="load_individuals_+3A_error_on_pid_not_found">error_on_pid_not_found</code></td>
<td>
<p>Error if pid not found</p>
</td></tr>
</table>

<hr>
<h2 id='malan-package'>MAle Lineage ANalysis</h2><span id='topic+malan'></span><span id='topic+malan-package'></span>

<h3>Description</h3>

<p>Simulating genealogies backwards and imposing STR mutations forwards.
</p>


<h3>Details</h3>

<p>See vignettes and manual for documentation.
</p>
<p>Disclaimer:
THE SOFTWARE IS PROVIDED &quot;AS IS&quot;, WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT.
IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR ANYONE DISTRIBUTING
THE SOFTWARE BE LIABLE FOR ANY DAMAGES OR OTHER LIABILITY, WHETHER IN CONTRACT,
TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Mikkel Meyer Andersen <a href="mailto:mikl@math.aau.dk">mikl@math.aau.dk</a>
</p>


<h3>References</h3>

<p>Andersen MM, Balding DJ (2017) How convincing is a matching Y-chromosome profile?
PLoS Genet 13(11): e1007028. <a href="https://doi.org/10.1371/journal.pgen.1007028">doi:10.1371/journal.pgen.1007028</a>.
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li><p> Report bugs at <a href="https://github.com/mikldk/malan/issues">https://github.com/mikldk/malan/issues</a>
</p>
</li></ul>


<hr>
<h2 id='meioses_generation_distribution'>Meiotic distribution</h2><span id='topic+meioses_generation_distribution'></span>

<h3>Description</h3>

<p>Get the distribution of number of meioses from <code>individual</code>
to all individuals in <code>individual</code>'s pedigree.
Note the <code>generation_upper_bound_in_result</code> parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meioses_generation_distribution(
  individual,
  generation_upper_bound_in_result = -1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meioses_generation_distribution_+3A_individual">individual</code></td>
<td>
<p>Individual to calculate all meiotic distances from</p>
</td></tr>
<tr><td><code id="meioses_generation_distribution_+3A_generation_upper_bound_in_result">generation_upper_bound_in_result</code></td>
<td>
<p>Limit on distribution; -1 means no limit.
0 is the final generation. 1 second last generation etc.</p>
</td></tr>
</table>

<hr>
<h2 id='meiotic_dist'>Meiotic distance between two individuals</h2><span id='topic+meiotic_dist'></span>

<h3>Description</h3>

<p>Get the number of meioses between two individuals.
Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meiotic_dist(ind1, ind2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meiotic_dist_+3A_ind1">ind1</code></td>
<td>
<p>Individual 1</p>
</td></tr>
<tr><td><code id="meiotic_dist_+3A_ind2">ind2</code></td>
<td>
<p>Individual 2</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of meioses between <code>ind1</code> and <code>ind2</code> if they are in the same pedigree, else -1.
</p>

<hr>
<h2 id='meiotic_dist_threshold'>Meiotic distance between two individuals (with threshold)</h2><span id='topic+meiotic_dist_threshold'></span>

<h3>Description</h3>

<p>Get the number of meioses between two individuals.
Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meiotic_dist_threshold(ind1, ind2, threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meiotic_dist_threshold_+3A_ind1">ind1</code></td>
<td>
<p>Individual 1</p>
</td></tr>
<tr><td><code id="meiotic_dist_threshold_+3A_ind2">ind2</code></td>
<td>
<p>Individual 2</p>
</td></tr>
<tr><td><code id="meiotic_dist_threshold_+3A_threshold">threshold</code></td>
<td>
<p>Max search radius, if exceeding, return -1</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Number of meioses between <code>ind1</code> and <code>ind2</code> if they are in the same pedigree, else -1.
</p>

<hr>
<h2 id='meiotic_radius'>Meiotic radius</h2><span id='topic+meiotic_radius'></span>

<h3>Description</h3>

<p>Get all individual IDs within a meiotic radius
Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meiotic_radius(ind, radius)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meiotic_radius_+3A_ind">ind</code></td>
<td>
<p>Individual</p>
</td></tr>
<tr><td><code id="meiotic_radius_+3A_radius">radius</code></td>
<td>
<p>Max radius</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with ID and meiotic radius
</p>

<hr>
<h2 id='mixture_info_by_individuals_2pers'>Mixture information about 2 persons' mixture of donor1 and donor2.</h2><span id='topic+mixture_info_by_individuals_2pers'></span>

<h3>Description</h3>

<p>Mixture information about 2 persons' mixture of donor1 and donor2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixture_info_by_individuals_2pers(
  individuals,
  donor1,
  donor2,
  include_genealogy_info = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixture_info_by_individuals_2pers_+3A_individuals">individuals</code></td>
<td>
<p>Individuals to consider as possible contributors and thereby get information from.</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_2pers_+3A_donor1">donor1</code></td>
<td>
<p>Contributor1/donor 1</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_2pers_+3A_donor2">donor2</code></td>
<td>
<p>Contributor2/donor 2</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_2pers_+3A_include_genealogy_info">include_genealogy_info</code></td>
<td>
<p>Include information about meiotic distances and family info</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with mixture information about the mixture <code>donor1</code>+<code>donor2</code>+<code>donor3</code> from <code>individuals</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixture_info_by_individuals_3pers">mixture_info_by_individuals_3pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_4pers">mixture_info_by_individuals_4pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_5pers">mixture_info_by_individuals_5pers</a></code>
</p>

<hr>
<h2 id='mixture_info_by_individuals_3pers'>Mixture information about 3 persons' mixture of donor1, donor2 and donor3.</h2><span id='topic+mixture_info_by_individuals_3pers'></span>

<h3>Description</h3>

<p>Mixture information about 3 persons' mixture of donor1, donor2 and donor3.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixture_info_by_individuals_3pers(individuals, donor1, donor2, donor3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixture_info_by_individuals_3pers_+3A_individuals">individuals</code></td>
<td>
<p>Individuals to consider as possible contributors and thereby get information from.</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_3pers_+3A_donor1">donor1</code></td>
<td>
<p>Contributor1/donor 1</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_3pers_+3A_donor2">donor2</code></td>
<td>
<p>Contributor2/donor 2</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_3pers_+3A_donor3">donor3</code></td>
<td>
<p>Contributor3/donor 3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with mixture information about the mixture <code>donor1</code>+<code>donor2</code>+<code>donor3</code> from <code>individuals</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixture_info_by_individuals_2pers">mixture_info_by_individuals_2pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_4pers">mixture_info_by_individuals_4pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_5pers">mixture_info_by_individuals_5pers</a></code>
</p>

<hr>
<h2 id='mixture_info_by_individuals_4pers'>Mixture information about 4 persons' mixture of donor1, donor2, donor3 and donor4.</h2><span id='topic+mixture_info_by_individuals_4pers'></span>

<h3>Description</h3>

<p>Mixture information about 4 persons' mixture of donor1, donor2, donor3 and donor4.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixture_info_by_individuals_4pers(individuals, donor1, donor2, donor3, donor4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixture_info_by_individuals_4pers_+3A_individuals">individuals</code></td>
<td>
<p>Individuals to consider as possible contributors and thereby get information from.</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_4pers_+3A_donor1">donor1</code></td>
<td>
<p>Contributor1/donor 1</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_4pers_+3A_donor2">donor2</code></td>
<td>
<p>Contributor2/donor 2</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_4pers_+3A_donor3">donor3</code></td>
<td>
<p>Contributor3/donor 3</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_4pers_+3A_donor4">donor4</code></td>
<td>
<p>Contributor4/donor 4</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with mixture information about the mixture <code>donor1</code>+<code>donor2</code>+<code>donor3</code> from <code>individuals</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixture_info_by_individuals_2pers">mixture_info_by_individuals_2pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_3pers">mixture_info_by_individuals_3pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_5pers">mixture_info_by_individuals_5pers</a></code>
</p>

<hr>
<h2 id='mixture_info_by_individuals_5pers'>Mixture information about 5 persons' mixture of donor1, donor2, donor3, donor4 and donor5.</h2><span id='topic+mixture_info_by_individuals_5pers'></span>

<h3>Description</h3>

<p>Mixture information about 5 persons' mixture of donor1, donor2, donor3, donor4 and donor5.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixture_info_by_individuals_5pers(
  individuals,
  donor1,
  donor2,
  donor3,
  donor4,
  donor5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixture_info_by_individuals_5pers_+3A_individuals">individuals</code></td>
<td>
<p>Individuals to consider as possible contributors and thereby get information from.</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_5pers_+3A_donor1">donor1</code></td>
<td>
<p>Contributor1/donor 1</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_5pers_+3A_donor2">donor2</code></td>
<td>
<p>Contributor2/donor 2</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_5pers_+3A_donor3">donor3</code></td>
<td>
<p>Contributor3/donor 3</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_5pers_+3A_donor4">donor4</code></td>
<td>
<p>Contributor4/donor 4</p>
</td></tr>
<tr><td><code id="mixture_info_by_individuals_5pers_+3A_donor5">donor5</code></td>
<td>
<p>Contributor5/donor 5</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with mixture information about the mixture <code>donor1</code>+<code>donor2</code>+<code>donor3</code> from <code>individuals</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixture_info_by_individuals_2pers">mixture_info_by_individuals_2pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_3pers">mixture_info_by_individuals_3pers</a></code>,
<code><a href="#topic+mixture_info_by_individuals_4pers">mixture_info_by_individuals_4pers</a></code>
</p>

<hr>
<h2 id='pedigree_as_igraph'>Convert pedigree to igraph</h2><span id='topic+pedigree_as_igraph'></span>

<h3>Description</h3>

<p>Convert pedigree to igraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigree_as_igraph(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigree_as_igraph_+3A_x">x</code></td>
<td>
<p>Pedigree</p>
</td></tr>
<tr><td><code id="pedigree_as_igraph_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>igraph</code> object
</p>

<hr>
<h2 id='pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists'>Information about matching individuals</h2><span id='topic+pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists'></span>

<h3>Description</h3>

<p>Gives information about all individuals in pedigree that matches an individual.
Just as <code><a href="#topic+count_haplotype_occurrences_individuals">count_haplotype_occurrences_individuals()</a></code> counts the number of
occurrences amongst a list of individuals,
this gives detailed information about matching individuals in the pedigree,
e.g. meiotic distances and maximum L1 distance on the path as some of these
matches may have (back)mutations between in between them (but often this will be 0).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists(
  suspect,
  generation_upper_bound_in_result = -1L,
  error_on_no_haplotype = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists_+3A_suspect">suspect</code></td>
<td>
<p>Individual that others must match the profile of.</p>
</td></tr>
<tr><td><code id="pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists_+3A_generation_upper_bound_in_result">generation_upper_bound_in_result</code></td>
<td>
<p>Only consider matches in
generation 0, 1, ... generation_upper_bound_in_result.
-1 means disabled, consider all generations.
End generation is generation 0.
Second last generation is 1.
And so on.</p>
</td></tr>
<tr><td><code id="pedigree_haplotype_matches_in_pedigree_meiosis_L1_dists_+3A_error_on_no_haplotype">error_on_no_haplotype</code></td>
<td>
<p>raise error or silently ignore individuals
with no haplotype</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with information about matching individuals.
Columns in order: meioses (meiotic distance to <code>suspect</code>),
max_L1 (on the path between the matching individual and <code>suspect</code>,
what is the maximum L1 distance between the <code>suspect</code>'s profile and the
profiles of the individuals on the path),
pid (pid of matching individual)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+count_haplotype_occurrences_individuals">count_haplotype_occurrences_individuals()</a></code>.
</p>

<hr>
<h2 id='pedigree_haplotype_near_matches_meiosis'>Information about almost matching individuals</h2><span id='topic+pedigree_haplotype_near_matches_meiosis'></span>

<h3>Description</h3>

<p>Gives information about all individuals in pedigree that almost matches
an individual.
Just as <code><a href="#topic+count_haplotype_near_matches_individuals">count_haplotype_near_matches_individuals()</a></code> counts the number of
occurrences amongst a list of individuals,
this gives detailed information about almost matching individuals in
the pedigree: for now, the meiotic distances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigree_haplotype_near_matches_meiosis(
  suspect,
  max_dist,
  generation_upper_bound_in_result = -1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigree_haplotype_near_matches_meiosis_+3A_suspect">suspect</code></td>
<td>
<p>Individual that others must match the profile of.</p>
</td></tr>
<tr><td><code id="pedigree_haplotype_near_matches_meiosis_+3A_max_dist">max_dist</code></td>
<td>
<p>Maximum distance (0 = match, 1 = 1 STR allele difference, ...)</p>
</td></tr>
<tr><td><code id="pedigree_haplotype_near_matches_meiosis_+3A_generation_upper_bound_in_result">generation_upper_bound_in_result</code></td>
<td>
<p>Only consider matches in
generation 0, 1, ... generation_upper_bound_in_result.
-1 means disabled, consider all generations.
End generation is generation 0.
Second last generation is 1.
And so on.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with information about matching individuals.
Columns in order: 1) meioses (meiotic distance to <code>suspect</code>),
2) haplotype distance, 3) pid (pid of matching individual)
</p>


<h3>See Also</h3>

<p><code><a href="#topic+count_haplotype_near_matches_individuals">count_haplotype_near_matches_individuals()</a></code>.
</p>

<hr>
<h2 id='pedigree_size'>Get pedigree size</h2><span id='topic+pedigree_size'></span>

<h3>Description</h3>

<p>Get pedigree size
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigree_size(ped)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigree_size_+3A_ped">ped</code></td>
<td>
<p>Pedigree</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigree_size(peds[[1]])

</code></pre>

<hr>
<h2 id='pedigree_size_generation'>Size of pedigree</h2><span id='topic+pedigree_size_generation'></span>

<h3>Description</h3>

<p>Get the size of the pedigree.
Note the <code>generation_upper_bound_in_result</code> parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigree_size_generation(pedigree, generation_upper_bound_in_result = -1L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigree_size_generation_+3A_pedigree">pedigree</code></td>
<td>
<p>Pedigree to get size of</p>
</td></tr>
<tr><td><code id="pedigree_size_generation_+3A_generation_upper_bound_in_result">generation_upper_bound_in_result</code></td>
<td>
<p>Limit on generation to include in count; -1 means no limit.
0 only include the final generation. 1 only second last generation etc.</p>
</td></tr>
</table>

<hr>
<h2 id='pedigrees_all_populate_autosomal'>Populate 1-locus autosomal DNA profile in pedigrees with single-step mutation model.</h2><span id='topic+pedigrees_all_populate_autosomal'></span>

<h3>Description</h3>

<p>Populate 1-locus autosomal DNA profile from founder and down in all pedigrees.
Note, that only alleles from ladder is assigned and
that all founders draw type randomly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigrees_all_populate_autosomal(
  pedigrees,
  allele_dist,
  theta,
  mutation_rate,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigrees_all_populate_autosomal_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigree list in which to populate genotypes</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_autosomal_+3A_allele_dist">allele_dist</code></td>
<td>
<p>Allele distribution (probabilities) &ndash; gets normalised</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_autosomal_+3A_theta">theta</code></td>
<td>
<p>Theta correction between 0 and 1 (both included)</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_autosomal_+3A_mutation_rate">mutation_rate</code></td>
<td>
<p>Mutation rate between 0 and 1 (both included)</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_autosomal_+3A_progress">progress</code></td>
<td>
<p>Show progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code> and
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>

<hr>
<h2 id='pedigrees_all_populate_haplotypes'>Populate haplotypes in pedigrees (0-founder/unbounded).</h2><span id='topic+pedigrees_all_populate_haplotypes'></span>

<h3>Description</h3>

<p>Populate haplotypes from founder and down in all pedigrees.
Note, that haplotypes are unbounded and
that all founders get haplotype <code>rep(0L, loci)</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigrees_all_populate_haplotypes(
  pedigrees,
  loci,
  mutation_rates,
  prob_two_step = 0,
  prob_genealogical_error = 0,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigrees_all_populate_haplotypes_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigree list in which to populate haplotypes</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_+3A_loci">loci</code></td>
<td>
<p>Number of loci</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_+3A_mutation_rates">mutation_rates</code></td>
<td>
<p>Vector with mutation rates, length <code>loci</code></p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_+3A_prob_two_step">prob_two_step</code></td>
<td>
<p>Given a mutation happens, this is the probability that the mutation is a two-step mutation</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_+3A_prob_genealogical_error">prob_genealogical_error</code></td>
<td>
<p>Probability that a genealogical error happens: if so, give individual haplotype <code>rep(0L, loci)</code> instead of father's</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_+3A_progress">progress</code></td>
<td>
<p>Show progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code> and
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes(peds, 2, c(1, 1))
get_haplotype(sim$end_generation_individuals[[1]])

</code></pre>

<hr>
<h2 id='pedigrees_all_populate_haplotypes_custom_founders'>Populate haplotypes in pedigrees (custom founder/unbounded).</h2><span id='topic+pedigrees_all_populate_haplotypes_custom_founders'></span>

<h3>Description</h3>

<p>Populate haplotypes from founder and down in all pedigrees.
Note, that haplotypes are unbounded.
All founders get a haplotype from calling the user
provided function <code>get_founder_haplotype()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigrees_all_populate_haplotypes_custom_founders(
  pedigrees,
  mutation_rates,
  get_founder_haplotype = NULL,
  prob_two_step = 0,
  prob_genealogical_error = 0,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigrees_all_populate_haplotypes_custom_founders_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigree list in which to populate haplotypes</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_custom_founders_+3A_mutation_rates">mutation_rates</code></td>
<td>
<p>Vector with mutation rates</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_custom_founders_+3A_get_founder_haplotype">get_founder_haplotype</code></td>
<td>
<p>Function taking no arguments returning a haplotype of <code>length(mutation_rates)</code></p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_custom_founders_+3A_prob_two_step">prob_two_step</code></td>
<td>
<p>Given a mutation happens, this is the probability that the mutation is a two-step mutation</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_custom_founders_+3A_prob_genealogical_error">prob_genealogical_error</code></td>
<td>
<p>Probability that a genealogical error happens: if so, give individual haplotype <code>get_founder_haplotype()</code> instead of father's</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_custom_founders_+3A_progress">progress</code></td>
<td>
<p>Show progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigrees_all_populate_haplotypes">pedigrees_all_populate_haplotypes()</a></code> and
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes_custom_founders(
  peds, c(1, 1), function(x) c(10, 10))
get_haplotype(sim$end_generation_individuals[[1]])

</code></pre>

<hr>
<h2 id='pedigrees_all_populate_haplotypes_ladder_bounded'>Populate haplotypes in pedigrees (custom founder/bounded).</h2><span id='topic+pedigrees_all_populate_haplotypes_ladder_bounded'></span>

<h3>Description</h3>

<p>Populate haplotypes from founder and down in all pedigrees.
Note, that haplotypes are bounded by <code>ladder_min</code> and <code>ladder_max</code>.
All founders get a haplotype from calling the user
provided function <code>get_founder_haplotype()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigrees_all_populate_haplotypes_ladder_bounded(
  pedigrees,
  mutation_rates,
  ladder_min,
  ladder_max,
  get_founder_haplotype = NULL,
  prob_two_step = 0,
  prob_genealogical_error = 0,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigree list in which to populate haplotypes</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_mutation_rates">mutation_rates</code></td>
<td>
<p>Vector with mutation rates</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_ladder_min">ladder_min</code></td>
<td>
<p>Lower bounds for haplotypes, same length as <code>mutation_rates</code></p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_ladder_max">ladder_max</code></td>
<td>
<p>Upper bounds for haplotypes, same length as <code>mutation_rates</code>; all entries must be strictly greater than <code>ladder_min</code></p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_get_founder_haplotype">get_founder_haplotype</code></td>
<td>
<p>Function taking no arguments returning a haplotype of <code>length(mutation_rates)</code></p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_prob_two_step">prob_two_step</code></td>
<td>
<p>Given a mutation happens, this is the probability that the mutation is a two-step mutation; refer to details for information about behaviour around ladder boundaries</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_prob_genealogical_error">prob_genealogical_error</code></td>
<td>
<p>Probability that a genealogical error happens: if so, give individual haplotype <code>get_founder_haplotype()</code> instead of father's</p>
</td></tr>
<tr><td><code id="pedigrees_all_populate_haplotypes_ladder_bounded_+3A_progress">progress</code></td>
<td>
<p>Show progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given that a two step mutation should happen (probability specified by <code>prob_two_step</code>):
With distances &gt;= 2 to ladder bounds, mutations happen as usual.
At distance = 0 or 1 to a ladder bound, the mutation is forced to move away from the boundary.
</p>
<p>Note, that pedigrees must first have been inferred by <code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigrees_all_populate_haplotypes">pedigrees_all_populate_haplotypes()</a></code> and
<code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_all_populate_haplotypes_ladder_bounded(
  peds, c(1, 1), c(0L, 0L), c(10L, 10L), 
  function(x) c(10, 10))
get_haplotype(sim$end_generation_individuals[[1]])

</code></pre>

<hr>
<h2 id='pedigrees_count'>Get number of pedigrees</h2><span id='topic+pedigrees_count'></span>

<h3>Description</h3>

<p>Get number of pedigrees
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigrees_count(pedigrees)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigrees_count_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigrees</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_count(peds)

</code></pre>

<hr>
<h2 id='pedigrees_table'>Get distribution of pedigree sizes</h2><span id='topic+pedigrees_table'></span>

<h3>Description</h3>

<p>Get distribution of pedigree sizes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pedigrees_table(pedigrees)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pedigrees_table_+3A_pedigrees">pedigrees</code></td>
<td>
<p>Pedigrees</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
peds &lt;- build_pedigrees(sim$population)
pedigrees_table(peds)

</code></pre>

<hr>
<h2 id='plot.malan_pedigree'>Plot pedigree</h2><span id='topic+plot.malan_pedigree'></span>

<h3>Description</h3>

<p>Plot <code>malan_pedigree</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_pedigree'
plot(
  x,
  ids = TRUE,
  haplotypes = FALSE,
  locus_sep = " ",
  mark_pids = NULL,
  label_color = "black",
  node_color = "lightgray",
  mark_color = "orange",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.malan_pedigree_+3A_x">x</code></td>
<td>
<p>Pedigree</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_ids">ids</code></td>
<td>
<p>Show pids</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_haplotypes">haplotypes</code></td>
<td>
<p>Show haplotypes</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_locus_sep">locus_sep</code></td>
<td>
<p>Locus separator in haplotypes</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_mark_pids">mark_pids</code></td>
<td>
<p>Vector of pids to highlight</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_label_color">label_color</code></td>
<td>
<p>Label color</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_node_color">node_color</code></td>
<td>
<p>Node color</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_mark_color">mark_color</code></td>
<td>
<p>Highlight color</p>
</td></tr>
<tr><td><code id="plot.malan_pedigree_+3A_...">...</code></td>
<td>
<p>Passed to <code>igraph::plot.igraph</code></p>
</td></tr>
</table>

<hr>
<h2 id='plot.malan_pedigreelist'>Plot pedigree list</h2><span id='topic+plot.malan_pedigreelist'></span>

<h3>Description</h3>

<p>Plot <code>malan_pedigreelist</code> generated by
<code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_pedigreelist'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.malan_pedigreelist_+3A_x">x</code></td>
<td>
<p>Pedigree list</p>
</td></tr>
<tr><td><code id="plot.malan_pedigreelist_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='population_populate_autosomal_infinite_alleles'>Populate 1-locus autosomal DNA profile in pedigrees with infinite alleles mutation model.</h2><span id='topic+population_populate_autosomal_infinite_alleles'></span>

<h3>Description</h3>

<p>Populate 1-locus autosomal DNA profile from founder and down in all pedigrees.
Note, that all founders have type 0 to begin with.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>population_populate_autosomal_infinite_alleles(
  population,
  mutation_rate,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="population_populate_autosomal_infinite_alleles_+3A_population">population</code></td>
<td>
<p>Population in which to populate genotypes</p>
</td></tr>
<tr><td><code id="population_populate_autosomal_infinite_alleles_+3A_mutation_rate">mutation_rate</code></td>
<td>
<p>Mutation rate between 0 and 1 (both included)</p>
</td></tr>
<tr><td><code id="population_populate_autosomal_infinite_alleles_+3A_progress">progress</code></td>
<td>
<p>Show progress</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The maternal allele is taken by random from
the <code>2*N[g]</code> alleles in the previous generation consisting of <code>N[g]</code> males
with descendants in the live population.
</p>
<p>This is also why this is not using pedigrees but instead the population.
</p>
<p>Note, that pedigrees need not be inferred.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+pedigrees_all_populate_haplotypes_custom_founders">pedigrees_all_populate_haplotypes_custom_founders()</a></code> and
<code><a href="#topic+pedigrees_all_populate_haplotypes_ladder_bounded">pedigrees_all_populate_haplotypes_ladder_bounded()</a></code>.
</p>

<hr>
<h2 id='population_size_generation'>Size of population</h2><span id='topic+population_size_generation'></span>

<h3>Description</h3>

<p>Get the size of the population.
Note the <code>generation_upper_bound_in_result</code> parameter.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>population_size_generation(population, generation_upper_bound_in_result = -1L)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="population_size_generation_+3A_population">population</code></td>
<td>
<p>Population to get size of</p>
</td></tr>
<tr><td><code id="population_size_generation_+3A_generation_upper_bound_in_result">generation_upper_bound_in_result</code></td>
<td>
<p>Limit on generation to include in count; -1 means no limit.
0 only include the final generation. 1 only second last generation etc.</p>
</td></tr>
</table>

<hr>
<h2 id='print_individual'>Print individual</h2><span id='topic+print_individual'></span>

<h3>Description</h3>

<p>Print individual
</p>


<h3>Usage</h3>

<pre><code class='language-R'>print_individual(individual)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print_individual_+3A_individual">individual</code></td>
<td>
<p>Individual</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
indv &lt;- get_individual(sim$population, 1)
print_individual(indv)

</code></pre>

<hr>
<h2 id='print.malan_pedigree'>Print pedigree</h2><span id='topic+print.malan_pedigree'></span>

<h3>Description</h3>

<p>Print pedigree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_pedigree'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.malan_pedigree_+3A_x">x</code></td>
<td>
<p>Pedigree</p>
</td></tr>
<tr><td><code id="print.malan_pedigree_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='print.malan_pedigreelist'>Print pedigree list</h2><span id='topic+print.malan_pedigreelist'></span>

<h3>Description</h3>

<p>Print <code>malan_pedigreelist</code> generated by
<code><a href="#topic+build_pedigrees">build_pedigrees()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_pedigreelist'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.malan_pedigreelist_+3A_x">x</code></td>
<td>
<p>Pedigrees (<code>malan_pedigreelist</code>)</p>
</td></tr>
<tr><td><code id="print.malan_pedigreelist_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='print.malan_population'>Print population</h2><span id='topic+print.malan_population'></span>

<h3>Description</h3>

<p>Print <code>malan_population</code> generated by
<code><a href="#topic+sample_geneology">sample_geneology()</a></code> or <code><a href="#topic+sample_geneology_varying_size">sample_geneology_varying_size()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_population'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.malan_population_+3A_x">x</code></td>
<td>
<p>Population (<code>malan_population</code>)</p>
</td></tr>
<tr><td><code id="print.malan_population_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='print.malan_population_abort'>Print <code>malan_population_abort</code></h2><span id='topic+print.malan_population_abort'></span>

<h3>Description</h3>

<p>Print <code>malan_population_abort</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'malan_population_abort'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.malan_population_abort_+3A_x">x</code></td>
<td>
<p><code>malan_population_abort</code></p>
</td></tr>
<tr><td><code id="print.malan_population_abort_+3A_...">...</code></td>
<td>
<p>ignored</p>
</td></tr>
</table>

<hr>
<h2 id='relationship_allele_diff_dist'>Calculate distribution of allele difference</h2><span id='topic+relationship_allele_diff_dist'></span>

<h3>Description</h3>

<p>Calculate distribution of allele difference after <code>m</code> meioses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relationship_allele_diff_dist(meioses, mu_dw, mu_up, method = "explicit")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relationship_allele_diff_dist_+3A_meioses">meioses</code></td>
<td>
<p>number of meioses separating the two individuals</p>
</td></tr>
<tr><td><code id="relationship_allele_diff_dist_+3A_mu_dw">mu_dw</code></td>
<td>
<p>mutation rate for 1-step down-mutation</p>
</td></tr>
<tr><td><code id="relationship_allele_diff_dist_+3A_mu_up">mu_up</code></td>
<td>
<p>mutation rate for 1-step up-mutation</p>
</td></tr>
<tr><td><code id="relationship_allele_diff_dist_+3A_method">method</code></td>
<td>
<p><code>"explicit"</code> (default): use known formulas for eigenvalues and eigenvectors.
Can cause numerical problems.
<code>"matmult"</code>: do matrix multiplication instead of diagonalisation.
<code>"matmult_mpfr"</code>: as <code>"matmult"</code> but with the <code>Rmpfr</code> library (note that this returns <code>list</code> instead of <code>data.frame</code>).
<code>"r_eigen"</code>: use <code>R</code>'s <code><a href="base.html#topic+eigen">eigen()</a></code> function to find eigen values. Mostly for debugging.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> with columns <code>d</code> (allele difference) and <code>p</code> (prob)
</p>

<hr>
<h2 id='relationship_allele_diff_dist_sym'>Calculate distribution of allele difference for symmetric mutation rates</h2><span id='topic+relationship_allele_diff_dist_sym'></span>

<h3>Description</h3>

<p>Calculate distribution of allele difference after <code>m</code> meioses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>relationship_allele_diff_dist_sym(meioses, mu_updw, method = "explicit")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="relationship_allele_diff_dist_sym_+3A_meioses">meioses</code></td>
<td>
<p>number of meioses separating the two individuals</p>
</td></tr>
<tr><td><code id="relationship_allele_diff_dist_sym_+3A_mu_updw">mu_updw</code></td>
<td>
<p>mutation rate for 1-step down- and up-mutations, i.e. total mutation rate is <code>2*mu_updw</code></p>
</td></tr>
<tr><td><code id="relationship_allele_diff_dist_sym_+3A_method">method</code></td>
<td>
<p><code>"explicit"</code> (default): use known formulas for eigenvalues and eigenvectors.
Can cause numerical problems.
<code>"matmult"</code>: do matrix multiplication instead of diagonalisation.
<code>"matmult_mpfr"</code>: as <code>"matmult"</code> but with the <code>Rmpfr</code> library (note that this returns <code>list</code> instead of <code>data.frame</code>).
<code>"r_eigen"</code>: use <code>R</code>'s <code><a href="base.html#topic+eigen">eigen()</a></code> function to find eigen values. Mostly for debugging.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><code>data.frame</code> with columns <code>d</code> (allele difference) and <code>p</code> (prob)
</p>

<hr>
<h2 id='sample_autosomal_genotype'>Sample genotype with theta</h2><span id='topic+sample_autosomal_genotype'></span>

<h3>Description</h3>

<p>Sample genotype with theta
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_autosomal_genotype(allele_dist, theta)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_autosomal_genotype_+3A_allele_dist">allele_dist</code></td>
<td>
<p>Allele distribution (probabilities) &ndash; gets normalised</p>
</td></tr>
<tr><td><code id="sample_autosomal_genotype_+3A_theta">theta</code></td>
<td>
<p>Theta correction between 0 and 1 (both included)</p>
</td></tr>
</table>

<hr>
<h2 id='sample_geneology'>Simulate a geneology with constant population size.</h2><span id='topic+sample_geneology'></span>

<h3>Description</h3>

<p>This function simulates a geneology where the last generation has <code>population_size</code> individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_geneology(
  population_size,
  generations,
  generations_full = 1L,
  generations_return = 3L,
  enable_gamma_variance_extension = FALSE,
  gamma_parameter_shape = 5,
  gamma_parameter_scale = 1/5,
  progress = TRUE,
  verbose_result = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_geneology_+3A_population_size">population_size</code></td>
<td>
<p>The size of the population.</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_generations">generations</code></td>
<td>
<p>The number of generations to simulate:
</p>

<ul>
<li><p> -1 for simulate to 1 founder
</p>
</li>
<li><p> else simulate this number of generations.
</p>
</li></ul>
</td></tr>
<tr><td><code id="sample_geneology_+3A_generations_full">generations_full</code></td>
<td>
<p>Number of full generations to be simulated.</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_generations_return">generations_return</code></td>
<td>
<p>How many generations to return (pointers to) individuals for.</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_enable_gamma_variance_extension">enable_gamma_variance_extension</code></td>
<td>
<p>Enable symmetric Dirichlet (and disable standard Wright-Fisher).</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_gamma_parameter_shape">gamma_parameter_shape</code></td>
<td>
<p>Parameter related to symmetric Dirichlet distribution for each man's probability to be father. Refer to details.</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_gamma_parameter_scale">gamma_parameter_scale</code></td>
<td>
<p>Parameter realted to symmetric Dirichlet distribution for each man's probability to be father. Refer to details.</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_progress">progress</code></td>
<td>
<p>Show progress.</p>
</td></tr>
<tr><td><code id="sample_geneology_+3A_verbose_result">verbose_result</code></td>
<td>
<p>Verbose result.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By the backwards simulating process of the Wright-Fisher model,
individuals with no descendants in the end population are not simulated.
If for some reason additional full generations should be simulated,
the number can be specified via the <code>generations_full</code> parameter.
This can for example be useful if one wants to simulate the
final 3 generations although some of these may not get (male) children.
</p>
<p>Let <code class="reqn">\alpha</code> be the parameter of a symmetric Dirichlet distribution
specifying each man's probability to be the father of an arbitrary
male in the next generation. When <code class="reqn">\alpha = 5</code>, a man's relative probability
to be the father has 95\
constant 1 under the standard Wright-Fisher model and the standard deviation in
the number of male offspring per man is 1.10 (standard Wright-Fisher = 1).
</p>
<p>This symmetric Dirichlet distribution is implemented by drawing
father (unscaled) probabilities from a Gamma distribution with
parameters <code>gamma_parameter_shape</code> and <code>gamma_parameter_scale</code>
that are then normalised to sum to 1.
To obtain a symmetric Dirichlet distribution with parameter <code class="reqn">\alpha</code>,
the following must be used:
<code class="reqn">`gamma_parameter_shape` = \alpha</code>
and
<code class="reqn">`gamma_parameter_scale` = 1/\alpha</code>.
</p>


<h3>Value</h3>

<p>A malan_simulation / list with the following entries:
</p>

<ul>
<li> <p><code>population</code>. An external pointer to the population.
</p>
</li>
<li> <p><code>generations</code>. Generations actually simulated, mostly useful when parameter <code>generations = -1</code>.
</p>
</li>
<li> <p><code>founders</code>. Number of founders after the simulated <code>generations</code>.
</p>
</li>
<li> <p><code>growth_type</code>. Growth type model.
</p>
</li>
<li> <p><code>sdo_type</code>. Standard deviation in a man's number of male offspring. StandardWF or GammaVariation depending on <code>enable_gamma_variance_extension</code>.
</p>
</li>
<li> <p><code>end_generation_individuals</code>. Pointers to individuals in end generation.
</p>
</li>
<li> <p><code>individuals_generations</code>. Pointers to individuals in last <code>generations_return</code> generation (if <code>generations_return = 3</code>, then individuals in the last three generations are returned).
</p>
</li></ul>

<p>If <code>verbose_result</code> is true, then these additional components are also returned:
</p>

<ul>
<li> <p><code>individual_pids</code>. A matrix with pid (person id) for each individual.
</p>
</li>
<li> <p><code>father_pids</code>. A matrix with pid (person id) for each individual's father.
</p>
</li>
<li> <p><code>father_indices</code>. A matrix with indices for fathers.
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+sample_geneology_varying_size">sample_geneology_varying_size()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
str(sim, 1)
sim$population
peds &lt;- build_pedigrees(sim$population)
peds

</code></pre>

<hr>
<h2 id='sample_geneology_varying_size'>Simulate a geneology with varying population size.</h2><span id='topic+sample_geneology_varying_size'></span>

<h3>Description</h3>

<p>This function simulates a geneology with varying population size specified
by a vector of population sizes, one for each generation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_geneology_varying_size(
  population_sizes,
  generations_full = 1L,
  generations_return = 3L,
  enable_gamma_variance_extension = FALSE,
  gamma_parameter_shape = 5,
  gamma_parameter_scale = 1/5,
  progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_geneology_varying_size_+3A_population_sizes">population_sizes</code></td>
<td>
<p>The size of the population at each generation, <code>g</code>.
<code>population_sizes[g]</code> is the population size at generation <code>g</code>.
The length of population_sizes is the number of generations being simulated.</p>
</td></tr>
<tr><td><code id="sample_geneology_varying_size_+3A_generations_full">generations_full</code></td>
<td>
<p>Number of full generations to be simulated.</p>
</td></tr>
<tr><td><code id="sample_geneology_varying_size_+3A_generations_return">generations_return</code></td>
<td>
<p>How many generations to return (pointers to) individuals for.</p>
</td></tr>
<tr><td><code id="sample_geneology_varying_size_+3A_enable_gamma_variance_extension">enable_gamma_variance_extension</code></td>
<td>
<p>Enable symmetric Dirichlet (and disable standard Wright-Fisher).</p>
</td></tr>
<tr><td><code id="sample_geneology_varying_size_+3A_gamma_parameter_shape">gamma_parameter_shape</code></td>
<td>
<p>Parameter related to symmetric Dirichlet distribution for each man's probability to be father. Refer to details.</p>
</td></tr>
<tr><td><code id="sample_geneology_varying_size_+3A_gamma_parameter_scale">gamma_parameter_scale</code></td>
<td>
<p>Parameter realted to symmetric Dirichlet distribution for each man's probability to be father. Refer to details.</p>
</td></tr>
<tr><td><code id="sample_geneology_varying_size_+3A_progress">progress</code></td>
<td>
<p>Show progress.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>By the backwards simulating process of the Wright-Fisher model,
individuals with no descendants in the end population are not simulated
If for some reason additional full generations should be simulated,
the number can be specified via the <code>generations_full</code> parameter.
This can for example be useful if one wants to simulate the
final 3 generations although some of these may not get (male) children.
</p>
<p>Let <code class="reqn">\alpha</code> be the parameter of a symmetric Dirichlet distribution
specifying each man's probability to be the father of an arbitrary
male in the next generation. When <code class="reqn">\alpha = 5</code>, a man's relative probability
to be the father has 95\
constant 1 under the standard Wright-Fisher model and the standard deviation in
the number of male offspring per man is 1.10 (standard Wright-Fisher = 1).
</p>
<p>This symmetric Dirichlet distribution is implemented by drawing
father (unscaled) probabilities from a Gamma distribution with
parameters <code>gamma_parameter_shape</code> and <code>gamma_parameter_scale</code>
that are then normalised to sum to 1.
To obtain a symmetric Dirichlet distribution with parameter <code class="reqn">\alpha</code>,
the following must be used:
<code class="reqn">`gamma_parameter_shape` = \alpha</code>
and
<code class="reqn">`gamma_parameter_scale` = 1/\alpha</code>.
</p>


<h3>Value</h3>

<p>A malan_simulation / list with the following entries:
</p>

<ul>
<li> <p><code>population</code>. An external pointer to the population.
</p>
</li>
<li> <p><code>generations</code>. Generations actually simulated, mostly useful when parameter <code>generations = -1</code>.
</p>
</li>
<li> <p><code>founders</code>. Number of founders after the simulated <code>generations</code>.
</p>
</li>
<li> <p><code>growth_type</code>. Growth type model.
</p>
</li>
<li> <p><code>sdo_type</code>. Standard deviation in a man's number of male offspring. StandardWF or GammaVariation depending on <code>enable_gamma_variance_extension</code>.
</p>
</li>
<li> <p><code>end_generation_individuals</code>. Pointers to individuals in end generation.
</p>
</li>
<li> <p><code>individuals_generations</code>. Pointers to individuals in last <code>generations_return</code> generation (if <code>generations_return = 3</code>, then individuals in the last three generations are returned).
</p>
</li></ul>



<h3>See Also</h3>

<p><code><a href="#topic+sample_geneology">sample_geneology()</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology_varying_size(10*(1:10))
str(sim, 1)
sim$population
peds &lt;- build_pedigrees(sim$population)
peds

</code></pre>

<hr>
<h2 id='set_generation'>Set individual's generation number</h2><span id='topic+set_generation'></span>

<h3>Description</h3>

<p>Note that generation 0 is final, end generation.
1 is second last generation etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_generation(individual, generation)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_generation_+3A_individual">individual</code></td>
<td>
<p>Individual</p>
</td></tr>
<tr><td><code id="set_generation_+3A_generation">generation</code></td>
<td>
<p>Generation to assign</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>sim &lt;- sample_geneology(100, 10)
indv &lt;- get_individual(sim$population, 1)
get_generation(indv)
set_generation(indv, 100)
get_generation(indv)

</code></pre>

<hr>
<h2 id='split_by_haplotypes'>Split pids by haplotype</h2><span id='topic+split_by_haplotypes'></span>

<h3>Description</h3>

<p>Individuals with the same haplotype will be in the same group
and individuals with different haplotypes will be in different groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>split_by_haplotypes(population, pids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="split_by_haplotypes_+3A_population">population</code></td>
<td>
<p>Population obtained from simulation</p>
</td></tr>
<tr><td><code id="split_by_haplotypes_+3A_pids">pids</code></td>
<td>
<p>Vector of individual pids</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of integer vector, element i is an IntegerVector
with all pids from <code>pids</code> with the same haplotype
</p>

<hr>
<h2 id='test_create_population'>Generate test population</h2><span id='topic+test_create_population'></span>

<h3>Description</h3>

<p>Generate test population
</p>


<h3>Usage</h3>

<pre><code class='language-R'>test_create_population()
</code></pre>


<h3>Value</h3>

<p>An external pointer to the population.
</p>

<hr>
<h2 id='ystr_kits'>Kit information about Y-STR markers</h2><span id='topic+ystr_kits'></span>

<h3>Description</h3>

<p>A dataset containing information about the
Y chromosomal short tandem repeat (Y-STR) markers that are present in the kit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ystr_kits
</code></pre>


<h3>Format</h3>

<p>A data frame with 88 rows and 2 variables:
</p>

<dl>
<dt>Marker</dt><dd><p>name of Y-STR marker</p>
</dd>
<dt>Kit</dt><dd><p>name of Y-STR kit</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://www.yhrd.org">https://www.yhrd.org</a>
</p>

<hr>
<h2 id='ystr_markers'>Mutational information about Y-STR markers</h2><span id='topic+ystr_markers'></span>

<h3>Description</h3>

<p>A dataset from yhrd.org (and their sources) containing mutational information about
Y chromosomal short tandem repeat (Y-STR) markers used in forensic genetics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ystr_markers
</code></pre>


<h3>Format</h3>

<p>A data frame with 29 rows and 5 variables:
</p>

<dl>
<dt>Marker</dt><dd><p>name of Y-STR marker</p>
</dd>
<dt>Meioses</dt><dd><p>number of meioses observed</p>
</dd>
<dt>Mutations</dt><dd><p>number of mutations observed in the corresponding number of Meioses</p>
</dd>
<dt>MutProb</dt><dd><p>point estimate of mutation probability, MutProb = Mutations/Meioses</p>
</dd>
<dt>Alleles</dt><dd><p>observed alleles</p>
</dd>
</dl>



<h3>Details</h3>

<p>Note, that loci with duplications (DYS385a/b as well as
DYF387S1a/b have been split into two loci).
</p>


<h3>Source</h3>

<p><a href="https://www.yhrd.org">https://www.yhrd.org</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
