<!DOCTYPE html><html><head><title>Help for package MALDIrppa</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MALDIrppa}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MALDIrppa-package'>
<p>MALDI mass spectra robust pre-processing and analysis</p></a></li>
<li><a href='#addMetadata'>
<p>Add metadata to <code>AbstractMassObject</code> class objects</p></a></li>
<li><a href='#alignPeaks'>
<p>Compact peak alignment process for <code>MassPeaks</code> objects</p></a></li>
<li><a href='#countPeaks'>
<p>Count the number of peaks in <code>MassPeaks</code> objects</p></a></li>
<li><a href='#deletePeaks'>
<p>Delete peaks from a <code>MassPeaks</code> objects</p></a></li>
<li><a href='#detectOutliers'>
<p>Detection of outlying mass peak profiles</p></a></li>
<li><a href='#importSpectra'>
<p>Convert mass spectra from text files into <code>MassSpectrum</code> objects</p></a></li>
<li><a href='#peakPatterns'>
<p>Display peak presence/absence patterns</p></a></li>
<li><a href='#plot.scSpectra'>
<p>Graphical summary of <code>screenSpectra</code> objects</p></a></li>
<li><a href='#rawToPeaks'>
<p>Create list of <code>MassPeaks</code> objects</p></a></li>
<li><a href='#rawToSpectra'>
<p>Create list of <code>MassSpectrum</code> objects</p></a></li>
<li><a href='#redResolution'>
<p>Reduce resolution of <code>MassSpectrum</code> objects</p></a></li>
<li><a href='#screenSpectra'>
<p>Identification of potentially low-quality raw mass spectra</p></a></li>
<li><a href='#snrPeaks'>
<p>Extract signal-to-noise ratio thresholds from <code>MassPeaks</code> objects</p></a></li>
<li><a href='#spectra'>
<p>Example mass spectra profiles</p></a></li>
<li><a href='#summary.scSpectra'>
<p>Numerical summary of <code>screenSpectra</code> objects</p></a></li>
<li><a href='#summaryPeaks'>
<p>Summary of mass peak profiles</p></a></li>
<li><a href='#summarySpectra'>
<p>Summary of mass spectra</p></a></li>
<li><a href='#transfIntensity'>
<p>Transform intensity of <code>MassSpectrum</code> objects</p></a></li>
<li><a href='#type'>
<p>Example mass spectra metadata</p></a></li>
<li><a href='#wavSmoothing'>
<p>Discrete wavelet transformation for <code>MassSpectrum</code> objects</p></a></li>
<li><a href='#writeIntensity'>
<p>Write intensity matrix in different formats</p></a></li>
<li><a href='#writeMetadata'>
<p>Write metadata in different formats</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>MALDI Mass Spectrometry Data Robust Pre-Processing and Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.0-2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-23</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Javier Palarea-Albaladejo &lt;javier.palarea@udg.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.5), MALDIquant, signal, robustbase, lattice,
waveslim</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>yes</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides methods for quality control and robust pre-processing and analysis of MALDI mass spectrometry data (Palarea-Albaladejo et al. (2018) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtx628">doi:10.1093/bioinformatics/btx628</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Japal/MALDIrppa">https://github.com/Japal/MALDIrppa</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Japal/MALDIrppa/issues">https://github.com/Japal/MALDIrppa/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 16:12:30 UTC; Japal</td>
</tr>
<tr>
<td>Author:</td>
<td>Javier Palarea-Albaladejo
    <a href="https://orcid.org/0000-0003-0162-669X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [cre, aut],
  Paolo Ribeca <a href="https://orcid.org/0000-0001-5599-3933"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-25 17:30:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='MALDIrppa-package'>
MALDI mass spectra robust pre-processing and analysis
</h2><span id='topic+MALDIrppa-package'></span><span id='topic+MALDIrppa'></span>

<h3>Description</h3>

<p>This package provides procedures for quality control and robust pre-processing and analysis of MALDI mass spectrometry data based on objects and methods from the <code><a href="MALDIquant.html#topic+MALDIquant">MALDIquant</a></code> package. Moreover, it includes some additional functionalities and data summary and management tools (see vignette).
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> MALDIrppa</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1.0-2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-01-23</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Javier Palarea-Albaladejo
</p>
<p>Maintainer: Javier Palarea-Albaladejo &lt;javier.palarea@udg.edu&gt;
</p>

<hr>
<h2 id='addMetadata'>
Add metadata to <code>AbstractMassObject</code> class objects
</h2><span id='topic+addMetadata'></span>

<h3>Description</h3>

<p>This function adds metadata to the <code>metaData</code> slot of an <code><a href="MALDIquant.html#topic+AbstractMassObject-class">AbstractMassObject-class</a></code> class object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>addMetadata(x, metadata, pos)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="addMetadata_+3A_x">x</code></td>
<td>

<p>List of <code><a href="MALDIquant.html#topic+AbstractMassObject-class">AbstractMassObject-class</a></code> (<code>MassSpectra</code> or <code>MassPeaks</code>) class objects.
</p>
</td></tr>
<tr><td><code id="addMetadata_+3A_metadata">metadata</code></td>
<td>

<p>Vector containing the metadata to be included for each element of <code>x</code> (same length as <code>x</code>).
</p>
</td></tr>
<tr><td><code id="addMetadata_+3A_pos">pos</code></td>
<td>

<p>Position of the new metadata within the <code>metaData</code> slot list of each element of <code>x</code>.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List of <code><a href="MALDIquant.html#topic+AbstractMassObject-class">AbstractMassObject-class</a></code> class objects including the new metadata in their <code>metaData</code> slot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects
data(type)    # metadata

# Add metadata

info &lt;- paste("Spectrum No.",1:length(spectra)) # Artificial metadata vector

spectra2 &lt;- addMetadata(spectra,info,1)

# Check info in metaData slot

spectra2[[1]]@metaData

</code></pre>

<hr>
<h2 id='alignPeaks'>
Compact peak alignment process for <code>MassPeaks</code> objects
</h2><span id='topic+alignPeaks'></span>

<h3>Description</h3>

<p>This function provides a single command for selecting anchor peaks, peak alignment and binning of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> class objects (<code><a href="MALDIquant.html#topic+MALDIquant">MALDIquant</a></code> package). It also deals with alignment-related issues found in high-resolution mass spectrometry data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alignPeaks(x, minFreq = 0.9, tolerance = 0.003, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alignPeaks_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> class objects.
</p>
</td></tr>
<tr><td><code id="alignPeaks_+3A_minfreq">minFreq</code></td>
<td>

<p>Minimum relative frequency of a peak over <code>x</code> to be considered as anchor peak for alignment (see <code><a href="MALDIquant.html#topic+referencePeaks">referencePeaks</a></code>).
</p>
</td></tr>
<tr><td><code id="alignPeaks_+3A_tolerance">tolerance</code></td>
<td>

<p>Maximal deviation in peak masses to be considered as identical (see <code><a href="MALDIquant.html#topic+referencePeaks">referencePeaks</a></code>, <code><a href="MALDIquant.html#topic+binPeaks">binPeaks</a></code>).
</p>
</td></tr>
<tr><td><code id="alignPeaks_+3A_...">...</code></td>
<td>
<p>Other arguments from the original functions in <code><a href="MALDIquant.html#topic+MALDIquant">MALDIquant</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code><a href="MALDIquant.html#topic+warpMassPeaks">warpMassPeaks</a></code> and <code><a href="MALDIquant.html#topic+binPeaks">binPeaks</a></code> in the <code><a href="MALDIquant.html#topic+MALDIquant">MALDIquant</a></code> package for details about the alignment and binning algorithms. Note that <code>alignPeaks</code> applies an additional binning round which helps to correct for misalignment issues found after using the default <code>strict</code> or <code>relaxed</code> bin creation rules in high-resolution mass spectrometry data.
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> class objects with aligned peaks along a common m/z range. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Some pre-processing

spectra &lt;- screenSpectra(spectra)$fspectra
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)

# Peak alignment

peaks &lt;- alignPeaks(peaks, minFreq = 0.8)

</code></pre>

<hr>
<h2 id='countPeaks'>
Count the number of peaks in <code>MassPeaks</code> objects
</h2><span id='topic+countPeaks'></span>

<h3>Description</h3>

<p>This function provides the number of peaks of each element of a list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>countPeaks(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="countPeaks_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector consisting of the number of peaks for each peak profile in <code>x</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Some pre-processing

spectra &lt;- screenSpectra(spectra)$fspectra
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)

# Count peaks

npeaks &lt;- countPeaks(peaks)
</code></pre>

<hr>
<h2 id='deletePeaks'>
Delete peaks from a <code>MassPeaks</code> objects
</h2><span id='topic+deletePeaks'></span>

<h3>Description</h3>

<p>This function deletes peaks of height (intensity) below a given value in <code>MassPeaks</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>deletePeaks(x, min = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="deletePeaks_+3A_x">x</code></td>
<td>

<p>A list of <code>MassPeaks</code> objects.
</p>
</td></tr>
<tr><td><code id="deletePeaks_+3A_min">min</code></td>
<td>

<p>Lower threshold used to discard a peak.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions takes a list of <code>MassPeaks</code> objects and filters out peaks of height (intensity) falling below the given minimum value.
</p>


<h3>Value</h3>

<p>A filtered list of <code>MassPeaks</code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Some pre-processing

spectra &lt;- screenSpectra(spectra)$fspectra
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)

# Delete peaks of intensity &lt; 30

peaks &lt;- deletePeaks(peaks, min = 30)
</code></pre>

<hr>
<h2 id='detectOutliers'>
Detection of outlying mass peak profiles
</h2><span id='topic+detectOutliers'></span>

<h3>Description</h3>

<p>This function identifies outlying cases in a collection of processed mass peak profiles. It can be applied either on peak intensities or binary data (peak presence/absence patterns). It allows to specify a grouping factor in order to execute the procedure at the desired level of aggregation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detectOutliers(x, by = NULL, binary = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="detectOutliers_+3A_x">x</code></td>
<td>
<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects containing processed peaks.</p>
</td></tr>
<tr><td><code id="detectOutliers_+3A_by">by</code></td>
<td>
<p>If given, a grouping variable (<code><a href="base.html#topic+factor">factor</a></code> or <code><a href="base.html#topic+numeric">numeric</a></code>) subsetting the data.</p>
</td></tr>
<tr><td><code id="detectOutliers_+3A_binary">binary</code></td>
<td>
<p>Logical value. It indicates whether the procedure must be applied on either peak intensities (<code>FALSE</code>, default) or on binary peak presence/absence patterns (<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="detectOutliers_+3A_...">...</code></td>
<td>
<p>Optional arguments for the robust outlier detection method.</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>This function marks samples with mass peak profiles that largely deviates from other samples at the given aggregation level. It uses robust methods for the detection of multivariate outliers applied on metric multidimensional scaling (MDS) coordinates (Euclidean distance is used for peak intensities and binary distance for binary profiles; see <code><a href="stats.html#topic+dist">dist</a></code>). The number of MDS coordinates used is generally set to p = <code>floor</code>(n/2), where n is the number of samples in the target subset. This is an upper cap recommended for the computation of the robust MCD estimator by <code><a href="robustbase.html#topic+covMcd">covMcd</a></code>. However, that rule of thumb can still generate matrix singularity problems with <code><a href="robustbase.html#topic+covMcd">covMcd</a></code> in some cases. When this occurs <code>detectOutliers</code> further reduces p to use the maximum number of MDS coordinates giving rise to a non-singular covariance matrix (min(p) = 2 in any case). The adaptive multivariate outlier detection algorithm was adapted from the <code>mvoutlier</code> package.
</p>


<h3>Value</h3>

<p>If <code>by = NULL</code>, a logical vector of length equal to the number of elements of <code>x</code> indicating outlying samples by <code>TRUE</code>. Otherwise, a 2-column <code><a href="base.html#topic+data.frame">data.frame</a></code> is generated which includes such a logical vector along with the grouping variable given in <code>by</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects
data(type) # metadata

# Some pre-processing

sc.results &lt;- screenSpectra(spectra,meta=type)
spectra &lt;- sc.results$fspectra # filtered mass spectra
type &lt;- sc.results$fmeta    # filtered metadata  
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)
peaks &lt;- alignPeaks(peaks, minFreq = 0.8)

# Find outlying samples at isolate level

out &lt;- detectOutliers(peaks, by = type$isolate)

# From peak presence/absence patterns

out.binary &lt;- detectOutliers(peaks, by = type$isolate, binary = TRUE)

  </code></pre>

<hr>
<h2 id='importSpectra'>
Convert mass spectra from text files into <code>MassSpectrum</code> objects
</h2><span id='topic+importSpectra'></span>

<h3>Description</h3>

<p>This function allows to import collections of mass spectra stored in individual text files into a list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>importSpectra(where = getwd())
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="importSpectra_+3A_where">where</code></td>
<td>

<p>Path to the folder where the text files are stored (default: current working directory).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions works with dat, csv or txt file types containing two columns: the first one referring to common m/z values and the second one to intensities (using single-space separator between both and no column names). It reads all the .dat, .csv or .txt files in the given folder (so unrelated files should better not be there) and creates a list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects. For importing data from more specialised file formats we refer the reader to the package <code>MALDIquantForeign</code>.
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create fake mass spectrometry data

s1 &lt;- cbind(1:20, rlnorm(20))
s2 &lt;- cbind(1:20, rlnorm(20))
s3 &lt;- cbind(1:20, rlnorm(20))

# Save as csv files in temporary directory

path &lt;- tempdir()

write.table(s1, file = file.path(path, "s1.csv"),
            row.names = FALSE, col.names = FALSE, sep=" ")
write.table(s2, file = file.path(path, "s2.csv"),
            row.names = FALSE, col.names = FALSE, sep=" ")
write.table(s3, file = file.path(path, "s3.csv"),
            row.names = FALSE, col.names = FALSE, sep=" ")

# Import files and arrange into a list of MassSpectrum objects

spectra &lt;- importSpectra(where = path)

</code></pre>

<hr>
<h2 id='peakPatterns'>
Display peak presence/absence patterns
</h2><span id='topic+peakPatterns'></span>

<h3>Description</h3>

<p>This function displays the patterns of peak presence and absence in an intensity matrix as generated from <code><a href="MALDIquant.html#topic+intensityMatrix">intensityMatrix</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>peakPatterns(x, abs.lab = NA, barplot = TRUE,
             axis.lab = c("m/z", "Index"), bar.col = "red3",
             cell.col = c("white", "dodgerblue"), grid = FALSE,
             grid.col = "black", grid.lty = "dotted", cex.axis = 0.5,
             cex.lab = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="peakPatterns_+3A_x">x</code></td>
<td>

<p>A <code><a href="base.html#topic+matrix">matrix</a></code>, <code><a href="base.html#topic+data.frame">data.frame</a></code> or a list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_abs.lab">abs.lab</code></td>
<td>

<p>Unique label used to denote peak absence in <code>x</code> (<code>NA</code>, default).
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_barplot">barplot</code></td>
<td>

<p>Logical value indicating whether a barplot of relative peak frequency across samples is displayed (<code>TRUE</code>, default).
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_axis.lab">axis.lab</code></td>
<td>

<p>Vector of axis labels in the <code>c("x", "y")</code> format.
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_bar.col">bar.col</code></td>
<td>

<p>Colour of the bars in the barplot.
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_cell.col">cell.col</code></td>
<td>

<p>Vector of colours for the table cells (format <code>c("col.absence","col.presence")</code>).
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_grid">grid</code></td>
<td>

<p>Logical value indicating whether gridlines are added (<code>FALSE</code>, default).
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_grid.col">grid.col</code></td>
<td>

<p>Colour of the gridlines (<code>"black"</code>, default).
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_grid.lty">grid.lty</code></td>
<td>

<p>Style of the gridlines (<code>"dotted"</code>, default. See <code>lty</code> in <code><a href="graphics.html#topic+par">par</a></code>).
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_cex.axis">cex.axis</code></td>
<td>

<p>Axis tick labels scaling factor relative to default.
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_cex.lab">cex.lab</code></td>
<td>

<p>Axis labels scaling factor relative to default.
</p>
</td></tr>
<tr><td><code id="peakPatterns_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The peak presence/absence patterns are displayed by rows from the first (top) to the last (bottom) sample in the data set <code>x</code> over the range of common m/z points. Positive peaks are by default represented by coloured cells whereas zero or absent peaks are left blank. A barplot on the top margins shows the relative frequency of a peak at each m/z point across samples.
</p>


<h3>Value</h3>

<p>No return value, graphical output.</p>


<h3>See Also</h3>

<p>See <code><a href="MALDIquant.html#topic+intensityMatrix">intensityMatrix</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects
data(type) # metadata

# Some pre-processing

sc.results &lt;- screenSpectra(spectra,meta=type)
spectra &lt;- sc.results$fspectra # filtered mass spectra
type &lt;- sc.results$fmeta    # filtered metadata  
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)
peaks &lt;- alignPeaks(peaks, minFreq = 0.8)

# Display patterns across all data

peakPatterns(peaks)

# Check results within isolate 280

peakPatterns(peaks[type$Isolate=="280"])

</code></pre>

<hr>
<h2 id='plot.scSpectra'>
Graphical summary of <code>screenSpectra</code> objects
</h2><span id='topic+plot.scSpectra'></span>

<h3>Description</h3>

<p>This is a <code>plot</code> method for <code>scSpectra</code> objects that displays the results from applying <code>screenSpectra</code> to identify potential faulty, low-quality raw mass spectra.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'scSpectra'
plot(x, type = c("index", "hist", "casewise"), breaks = 30,
        labels = FALSE, col = "green3", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.scSpectra_+3A_x">x</code></td>
<td>

<p>A <code>scSpectra</code> object as generated by <code><a href="#topic+screenSpectra">screenSpectra</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.scSpectra_+3A_type">type</code></td>
<td>

<p>Type of graphical display.
</p>
</td></tr>
<tr><td><code id="plot.scSpectra_+3A_breaks">breaks</code></td>
<td>

<p>Number of break points for plotting a histogram when <code>type = "hist"</code> (default = 30).
</p>
</td></tr>
<tr><td><code id="plot.scSpectra_+3A_labels">labels</code></td>
<td>

<p>Vector of labels for the mass spectra (default = <code>FALSE</code>, no labels).
</p>
</td></tr>
<tr><td><code id="plot.scSpectra_+3A_col">col</code></td>
<td>

<p>Colour for the histogram bars when <code>type = "hist"</code>.
</p>
</td></tr>
<tr><td><code id="plot.scSpectra_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For <code>type = "index"</code> (default) the upper and lower fences used to declare a mass spectrum as potentially low-quality are shown along with their A scores (see <code><a href="#topic+screenSpectra">screenSpectra</a></code> for details). Point labels can be added using the <code>labels</code> argument (either a position index when <code>labels = TRUE</code> or a given vector of labels; see examples below). For <code>type = "hist"</code> a histogram of the distribution of the A scores is produced along with the tolerance fences. Finally, <code>type = "casewise"</code> displays interactively the flagged spectra for visual inspection.
</p>


<h3>Value</h3>

<p>No return value, graphical output.</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+screenSpectra">screenSpectra</a></code> and <code><a href="#topic+summary.scSpectra">summary.scSpectra</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra objects
data(type)    # metadata

sc.results &lt;- screenSpectra(spectra)
plot(sc.results)
plot(sc.results, labels = TRUE)
plot(sc.results, labels = type$SpectID)
plot(sc.results, type = "hist")
</code></pre>

<hr>
<h2 id='rawToPeaks'>
Create list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects
</h2><span id='topic+rawToPeaks'></span>

<h3>Description</h3>

<p>This is an auxiliary function to create a list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects from raw data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rawToPeaks(mz, I)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rawToPeaks_+3A_mz">mz</code></td>
<td>

<p>Vector of m/z values.
</p>
</td></tr>
<tr><td><code id="rawToPeaks_+3A_i">I</code></td>
<td>

<p>Matrix of peak intensity values.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions creates a list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects from a vector of common m/z values and a matrix of column vectors of peak intensities for a collections of mass peak profiles. The column names are used to label the elements of the list.
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>

<hr>
<h2 id='rawToSpectra'>
Create list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects
</h2><span id='topic+rawToSpectra'></span>

<h3>Description</h3>

<p>This is an auxiliary function to create a list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects from raw data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rawToSpectra(mz, I)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rawToSpectra_+3A_mz">mz</code></td>
<td>

<p>Vector of m/z values.
</p>
</td></tr>
<tr><td><code id="rawToSpectra_+3A_i">I</code></td>
<td>

<p>Matrix of intensity values.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions creates a list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects from a vector of common m/z values and a matrix of column vectors of intensities for a collections of mass spectra. The column names are used to label the elements of the list.
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+importSpectra">importSpectra</a></code>.
</p>

<hr>
<h2 id='redResolution'>
Reduce resolution of <code>MassSpectrum</code> objects
</h2><span id='topic+redResolution'></span>

<h3>Description</h3>

<p>This function allows to obtain a lighter version of a list of <code><a href="MALDIquant.html#topic+MassSpectrum-class">MassSpectrum</a></code> objects by decreasing their m/z resolution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>redResolution(x, by = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="redResolution_+3A_x">x</code></td>
<td>
<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum-class">MassSpectrum</a></code> objects.
</p>
</td></tr>
<tr><td><code id="redResolution_+3A_by">by</code></td>
<td>
<p>Number of times reduction (<code>by = 1</code>, default).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function reduces the resolution of mass spectra by eliminating a regular sequence of m/z sampling points in steps given by the argument <code>by</code>. For example, specifiying <code>by = 2</code> means to reduce the length and memory usage of the signal by a half approximately.
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum-class">MassSpectrum</a></code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Reduce resolution by a half

spectra.LowRes &lt;- redResolution(spectra, by = 2)

</code></pre>

<hr>
<h2 id='screenSpectra'>
Identification of potentially low-quality raw mass spectra
</h2><span id='topic+screenSpectra'></span>

<h3>Description</h3>

<p>This function implements a quality control check to help in the identification of possibly faulty, low-quality raw mass spectra. It computes an atypicality score and labels suspicious profiles for further inspection and filtering.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>screenSpectra(x, meta = NULL, threshold = 1.5, estimator = c("Q", "MAD"),
                 method = c("adj.boxplot", "boxplot", "ESD", "Hampel", "RC"),
                 nd = 1, lambda = 0.5, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="screenSpectra_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_meta">meta</code></td>
<td>
<p>(optional) Matrix or vector containing metadata associated to <code>x</code>. Typically a data matrix including spectrum ID, biotype, replicate number, etc. for each element of <code>x</code>.
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_threshold">threshold</code></td>
<td>

<p>Multiplicative factor used in computing the upper and lower fences to determine passes and failures. It is related to the actual method used to compute the fences (see <code>method</code>). Typically, <code>threshold = 1.5</code> (default value) for the boxplot rules, and <code>threshold = 3</code> for the others.
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_estimator">estimator</code></td>
<td>
<p>Robust scale estimator used:
</p>
<p><code>Q</code>: robust location-free scale estimate (default, see <code><a href="robustbase.html#topic+Qn">Qn</a></code> function in <code>robustbase</code> package). More efficient than MAD and adequate for non-symmetric distributions.
</p>
<p><code>MAD</code>: median absolute deviance scale estimate. Very robust and preferred for fairly symmetric distributions.
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_method">method</code></td>
<td>

<p>Method used to compute upper and lower fences for the identification of atypical mass spectra.
</p>
<p><code>boxplot</code>: standard boxplot rule based on the first and third quartiles and the interquartile range. 
</p>
<p><code>adj.boxplot</code>: extension of boxplot rule for strongly asymmetric data (default).
</p>
<p><code>ESD</code>: extreme studentized deviation method. Based on the mean and the standard deviation of the data. Typically used with <code>threshold = 3</code> (three-sigma rule).
</p>
<p><code>Hampel</code>: robust version of the ESD method based on the median and the median absolute deviance estimate (MAD).
</p>
<p><code>RC</code>: as Hampel's but replacing MAD by Rousseeuw &amp; Croux (1993)'s <code><a href="robustbase.html#topic+Qn">Qn</a></code> as scale estimate.
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_nd">nd</code></td>
<td>

<p>Order for the derivative function of the mass spectra (default = 1).
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_lambda">lambda</code></td>
<td>

<p>Weight given to each component of the atypicality score (values in [0, 1], default = 0.5, see details below).
</p>
</td></tr>
<tr><td><code id="screenSpectra_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The procedure computes an atypicality score (A score) based on a weighted function of two components: (1) a robust scale estimator (<code>Q</code> or <code>MAD</code>) of the n-order derivative (computed using Savitzky-Golay smoothing filter) of scaled mass spectra and (2) the median intensity of the signals. Given a <code>method</code> to determine tolerance fences, a mass spectrum is labelled as potentially faulty, low-quality according to the magnitude of its A score. The <code>adj.boxplot</code> method based on the <code>Q</code> scale estimator and equal weights to both components (<code>lambda</code> = 0.5) are the default options. The greater <code>lambda</code> the higher the weight given to the scale estimator in the A score. The function produces summaries and a list of mass spectra and (if given) associated metadata in which the identified cases were filtered out.
</p>


<h3>Value</h3>

<p>An object of class <code>scSpectra</code> with elements:
</p>
<table>
<tr><td><code>fspectra</code></td>
<td>
<p>List of mass spectra (<code>MassSpectrum</code> class) with potential low-quality cases filtered out.</p>
</td></tr>
<tr><td><code>fmeta</code></td>
<td>
<p>Associated filtered metadata (<code>data.frame</code> object).</p>
</td></tr>
<tr><td><code>est.table</code></td>
<td>
<p>Results table showing the mass spectra ID, A score and label (pass/failure).</p>
</td></tr>
<tr><td><code>...</code></td>
<td>
<p>Other details (see method <code><a href="#topic+summary.scSpectra">summary.scSpectra</a></code> for <code>scSpectra</code> objects).</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>See methods <code><a href="#topic+summary.scSpectra">summary.scSpectra</a></code> and <code><a href="#topic+plot.scSpectra">plot.scSpectra</a></code> for <code>scSpectra</code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load example data

data(spectra) # list of MassSpectra objects
data(type)    # metadata

# Results using different settings

sc.results &lt;- screenSpectra(spectra)
sc.results &lt;- screenSpectra(spectra, type)
sc.results &lt;- screenSpectra(spectra, type, method = "RC")
sc.results &lt;- screenSpectra(spectra, type, threshold = 3, estimator = "MAD", method = "Hampel")

# Numerical and graphical summary

summary(sc.results)
plot(sc.results)

# Save filtered data for further pre-processing

filtered.spectra &lt;- sc.results$fspectra
filtered.type &lt;- sc.results$fmeta
</code></pre>

<hr>
<h2 id='snrPeaks'>
Extract signal-to-noise ratio thresholds from <code>MassPeaks</code> objects
</h2><span id='topic+snrPeaks'></span>

<h3>Description</h3>

<p>This function extracts the thresholds used to determine peaks from mass spectra based on signal-to-noise ratio (SNR) (threshold equal to <code>SNR*noise</code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>snrPeaks(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="snrPeaks_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Given a collection of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects as obtained from <code><a href="MALDIquant.html#topic+detectPeaks">detectPeaks</a></code>, this function provides the thresholds used in each case to determine peaks from the original mass spectra. The thresholds are calculated as the product of a SNR value set by the user and the estimated noise of the signal (see <code><a href="MALDIquant.html#topic+detectPeaks">detectPeaks</a></code>).
</p>


<h3>Value</h3>

<p>A list of vectors of SNR-based thresholds, one for each sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Some pre-processing

spectra &lt;- screenSpectra(spectra)$fspectra
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)

# Extract thresholds for each mass peak profile

SNRs &lt;- snrPeaks(peaks)
</code></pre>

<hr>
<h2 id='spectra'>
Example mass spectra profiles
</h2><span id='topic+spectra'></span>

<h3>Description</h3>

<p>List of mass spectra (<code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> class).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(spectra)</code></pre>


<h3>Details</h3>

<p>Low-resolution version of a MALDI-TOF mass spectrometry data set in the range [2500, 13000] m/z provided for illustration purposes. It consists of 4 technical replicates of 5 biological replicates from 19 bacterial isolates (see <code><a href="#topic+type">type</a></code> for associated metadata).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(spectra)

str(spectra[[1]])

plot(spectra[[1]])

</code></pre>

<hr>
<h2 id='summary.scSpectra'>
Numerical summary of <code>screenSpectra</code> objects
</h2><span id='topic+summary.scSpectra'></span>

<h3>Description</h3>

<p>This is a <code>summary</code> method for <code>scSpectra</code> objects that generates a numerical summary of the settings and results from applying <code>screenSpectra</code> to identify potential faulty, low-quality raw mass spectra.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'scSpectra'
summary(object, ncases = 10, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.scSpectra_+3A_object">object</code></td>
<td>

<p>A <code>scSpectra</code> object as generated from <code><a href="#topic+screenSpectra">screenSpectra</a></code>.
</p>
</td></tr>
<tr><td><code id="summary.scSpectra_+3A_ncases">ncases</code></td>
<td>

<p>Number of cases shown in the results table.
</p>
</td></tr>
<tr><td><code id="summary.scSpectra_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A table is generated that includes details of the numerical estimations along with mass spectra ID, A score and the label for each mass spectra, either potentially low-quality (<code>failure</code>) or good-quality (<code>success</code>).
</p>


<h3>Value</h3>

<p>No return value, text printed on console.</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+screenSpectra">screenSpectra</a></code> and <code><a href="#topic+plot.scSpectra">plot.scSpectra</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra objects

sc.results &lt;- screenSpectra(spectra)
summary(sc.results)
</code></pre>

<hr>
<h2 id='summaryPeaks'>
Summary of mass peak profiles
</h2><span id='topic+summaryPeaks'></span>

<h3>Description</h3>

<p>This function generates a numerical summary of a collection of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryPeaks(x, digits = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summaryPeaks_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>
</td></tr>
<tr><td><code id="summaryPeaks_+3A_digits">digits</code></td>
<td>

<p>Integer indicating the number of decimal places to be used.
</p>
</td></tr></table>


<h3>Details</h3>

<p>For each <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> on the list this function provides summary statistics of m/z points, peak intensities and SNR thresholds (number, minimum, mean, standard deviation, median, mean absolute deviation, maximum).
</p>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> containing summary information of a collection of <code><a href="MALDIquant.html#topic+MassPeaks">MassPeaks</a></code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects
data(type)    # metadata

# Some pre-processing

sc.results &lt;- screenSpectra(spectra, meta = type)

spectra &lt;- sc.results$fspectra
type &lt;- sc.results$fmeta

spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)

names(peaks) &lt;- type$SpectID # spectra IDs are lost after removeBaseline()

# Summary of peak profile features (results for positions 10 to 20)

summaryPeaks(peaks[10:20])

</code></pre>

<hr>
<h2 id='summarySpectra'>
Summary of mass spectra
</h2><span id='topic+summarySpectra'></span>

<h3>Description</h3>

<p>This function generates a numerical summary of a collection of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarySpectra(x, digits = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summarySpectra_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>
</td></tr>
<tr><td><code id="summarySpectra_+3A_digits">digits</code></td>
<td>

<p>Integer indicating the number of decimal places to be used.
</p>
</td></tr></table>


<h3>Details</h3>

<p>For each <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> on the list this function provides summary statistics of m/z points and signal intensities (number, minimum, mean, standard deviation, median, mean absolute deviation, maximum).
</p>


<h3>Value</h3>

<p>A <code><a href="base.html#topic+data.frame">data.frame</a></code> containing summary information of a collection of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects
data(type)    # metadata

# Summary of spectra features (results for 20 first mass spectra)

summarySpectra(spectra[1:20])

# Some pre-processing

sc.results &lt;- screenSpectra(spectra, meta = type)

spectra &lt;- sc.results$fspectra
type &lt;- sc.results$fmeta

spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)

names(spectra) &lt;- type$SpectID # spectra IDs are lost with removeBaseline()

# Summary of spectra features (results for positions 10 to 20)

summarySpectra(spectra[10:20])
</code></pre>

<hr>
<h2 id='transfIntensity'>
Transform intensity of <code>MassSpectrum</code> objects
</h2><span id='topic+transfIntensity'></span>

<h3>Description</h3>

<p>This function applies user-defined transformations on the intensities of <code><a href="MALDIquant.html#topic+MassSpectrum-class">MassSpectrum</a></code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transfIntensity(x, fun = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transfIntensity_+3A_x">x</code></td>
<td>
<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum-class">MassSpectrum</a></code> objects.
</p>
</td></tr>
<tr><td><code id="transfIntensity_+3A_fun">fun</code></td>
<td>
<p>Name of an user-defined transformation function or any other pre-defined one in R.
</p>
</td></tr>
<tr><td><code id="transfIntensity_+3A_...">...</code></td>
<td>
<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function allows the user to define any sensible function to be applied on signal intensities. For logarithm and square root transformations it is equivalent to <code><a href="MALDIquant.html#topic+transformIntensity">transformIntensity</a></code> in the MALDIquant package. 
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum-class">MassSpectrum</a></code> objects with signal intensities transformed according to <code>fun</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Scale intensities into [0, 1] by dividing by their maximum value

scale.max &lt;- function(x){x/max(x)} # define scaling function

scaled.spectra &lt;- transfIntensity(spectra, fun = scale.max)
    
# Compute natural logarithm of intensity values (using the pre-defined sqrt R function)

log.spectra &lt;- transfIntensity(spectra, sqrt)


</code></pre>

<hr>
<h2 id='type'>
Example mass spectra metadata
</h2><span id='topic+type'></span>

<h3>Description</h3>

<p>Metadata associated to the <code><a href="#topic+spectra">spectra</a></code> data set containing information about isolate, biological and technical replicate numbers and mass spectra IDs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(type)</code></pre>


<h3>Format</h3>

<p>The format is:
</p>
<p>Isolate: Factor w/ 14 levels &quot;280&quot;,&quot;43&quot;,&quot;45&quot;,..: 2 2 2 2 2 2 2 2 2 2 ...
</p>
<p>BioRep : int  1 1 1 1 2 2 2 2 3 3 ...
</p>
<p>TechRep: int  1 2 3 4 1 2 3 4 1 2 ...
</p>
<p>SpectID: Factor w/ 315 levels &quot;160408C13&quot;,&quot;160408C14&quot;,..: 1 2 3 4 5 6 7 8 9 10 ...
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(type)
str(type)
</code></pre>

<hr>
<h2 id='wavSmoothing'>
Discrete wavelet transformation for <code>MassSpectrum</code> objects
</h2><span id='topic+wavSmoothing'></span>

<h3>Description</h3>

<p>This function performs undecimated wavelet transform (UDWT) on mass spectra in <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> format. Alternatively, smoothing methods included in the <code>MALDIquant</code> package can be called.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>wavSmoothing(x, method = c("Wavelet", "SavitzkyGolay", "MovingAverage"),
             n.levels = 4, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="wavSmoothing_+3A_x">x</code></td>
<td>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects.
</p>
</td></tr>
<tr><td><code id="wavSmoothing_+3A_method">method</code></td>
<td>

<p>Smoothing method used.
</p>
</td></tr>
<tr><td><code id="wavSmoothing_+3A_n.levels">n.levels</code></td>
<td>

<p>Depth of the decomposition for wavelet-based smoothing.
</p>
</td></tr>
<tr><td><code id="wavSmoothing_+3A_...">...</code></td>
<td>
<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that from version 1.1.0 of <code>MALDIrppa</code> wavelet smoothing is conducted by maximal overlap discrete wavelet transformation and universal thresholding of coefficients based on methods available on the <code>waveslim</code> package. The optimal level of smoothing is determined by model-driven estimates of the thresholds. The parameter <code>n.levels</code> (values &gt; 0 and &lt;= log(length(x),2)) can be used to tweak the levels to obtain a smoother or rougher result.
</p>
<p>Alternatively, smoothing methods <code>SavitzkyGolay</code> and <code>MovingAverage</code> from the <code>MALDIquant</code> package can be called directly from this function.
</p>
<p>If the previous implementation of the wavelet method is required please download and install manually source files of version 1.0.5-1 from the archive of old sources of the package (<a href="https://CRAN.R-project.org/package=MALDIrppa">https://CRAN.R-project.org/package=MALDIrppa</a>).
</p>


<h3>Value</h3>

<p>A list of <code><a href="MALDIquant.html#topic+MassSpectrum">MassSpectrum</a></code> objects with denoised signal intensities.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# sqrt transformation and signal smoothing using UDWT

spectra &lt;- transfIntensity(spectra, fun = "sqrt")
spectra &lt;- wavSmoothing(spectra)
</code></pre>

<hr>
<h2 id='writeIntensity'>
Write intensity matrix in different formats
</h2><span id='topic+writeIntensity'></span>

<h3>Description</h3>

<p>This function writes an intensity matrix as generated by <code><a href="MALDIquant.html#topic+intensityMatrix">intensityMatrix</a></code> into a file in the R, csv, NEXUS or FASTA formats. For NEXUS format it allows to specify weights for peaks.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeIntensity(x, filename = "intMatrix", format = c("R", "csv", "NEXUS", "FASTA"),
               binary = FALSE, labels = NULL, weights=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeIntensity_+3A_x">x</code></td>
<td>

<p>Intensity matrix as obtained from <code><a href="MALDIquant.html#topic+intensityMatrix">intensityMatrix</a></code>.
</p>
</td></tr>
<tr><td><code id="writeIntensity_+3A_filename">filename</code></td>
<td>

<p>A character string specifying a name for the destination file (filename extension not required).
</p>
</td></tr>
<tr><td><code id="writeIntensity_+3A_format">format</code></td>
<td>

<p>One of R (default <code>.RData</code> file), text (comma-separated <code>.csv</code> file), NEXUS (<code>.nex</code> file) or FASTA (<code>.fas</code> file).
</p>
</td></tr>
<tr><td><code id="writeIntensity_+3A_binary">binary</code></td>
<td>

<p>Logical value. If <code>TRUE</code>, a binary version (1: peak presence, 0: peak absence) of <code>x</code> is saved (default <code>FALSE</code>).
</p>
</td></tr>
<tr><td><code id="writeIntensity_+3A_labels">labels</code></td>
<td>

<p>Optional vector of ID labels for the samples.
</p>
</td></tr>
<tr><td><code id="writeIntensity_+3A_weights">weights</code></td>
<td>

<p>Optional numeric vector of peak weights (NEXUS format).
</p>
</td></tr>
<tr><td><code id="writeIntensity_+3A_...">...</code></td>
<td>

<p>Additional arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a wrapper function to simplify the writing of an intensity matrix in different formats while adding some extra features. It includes the common NEXUS and FASTA formats as an extension of functions in the <code>ape</code> package to handle peak intensity data. It also allows for taxa/sample pre-computed peak weights to be included in the NEXUS file. It checks whether the names meet NEXUS name conventions and gives them adequate format if not. A binary intensity matrix is always internally generated (<code>binary = TRUE</code>) when either the NEXUS or FASTA format is chosen. If any, <code>NA</code> values in <code>x</code> are assumed to denote zero intensity/peak absence and are then converted into zeros.
</p>


<h3>Value</h3>

<p>No return value, file in selected format created on destination folder.</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects

# Some pre-processing

spectra &lt;- screenSpectra(spectra)$fspectra
spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)
peaks &lt;- alignPeaks(peaks, minFreq = 0.8)

# Intensity matrix

int &lt;- intensityMatrix(peaks)

# Save as R file (saved to a temporary location as an example)

writeIntensity(int, file = file.path(tempdir(), "int"))

# Save as binary NEXUS file (saved to a temporary location as an example)

writeIntensity(int, file = file.path(tempdir(),"int.binary"),
               format = "NEXUS", interleaved = FALSE)

</code></pre>

<hr>
<h2 id='writeMetadata'>
Write metadata in different formats
</h2><span id='topic+writeMetadata'></span>

<h3>Description</h3>

<p>This function is simply a wrapper to write the metadata associated with a collection of mass spectra into a file in either the R or csv format.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>writeMetadata(x, filename = "Metadata", format = c("R", "csv"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="writeMetadata_+3A_x">x</code></td>
<td>

<p>Metadata in any sensible data format, preferably <code><a href="base.html#topic+matrix">matrix</a></code> or <code><a href="base.html#topic+data.frame">data.frame</a></code>.
</p>
</td></tr>
<tr><td><code id="writeMetadata_+3A_filename">filename</code></td>
<td>

<p>A character string specifying a name for the destination file (filename extension not required).
</p>
</td></tr>
<tr><td><code id="writeMetadata_+3A_format">format</code></td>
<td>

<p>One of R (default <code>.RData</code> file) or text (comma-separated <code>.csv</code> file).
</p>
</td></tr>
<tr><td><code id="writeMetadata_+3A_...">...</code></td>
<td>

<p>Other arguments.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>It uses either <code><a href="base.html#topic+save">save</a></code> or <code><a href="utils.html#topic+write.table">write.table</a></code> to store the metadata. Check these functions for adequate data formats.
</p>


<h3>Value</h3>

<p>No return value, file in selected format created on destination folder.</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load example data

data(spectra) # list of MassSpectra class objects
data(type)    # metadata

# Some pre-processing

sc.spectra &lt;- screenSpectra(spectra, meta = type)

spectra &lt;- sc.spectra$fspectra # filtered spectra
type &lt;- sc.spectra$fmeta # filtered metadata

spectra &lt;- transformIntensity(spectra, method = "sqrt")
spectra &lt;- wavSmoothing(spectra)
spectra &lt;- removeBaseline(spectra)
peaks &lt;- detectPeaks(spectra)
peaks &lt;- alignPeaks(peaks, minFreq = 0.8)

# Intensity matrix

int &lt;- intensityMatrix(peaks)

# Save resulting data in R format (to a temporary location as an example)

writeIntensity(int, filename = file.path(tempdir(),"MyintMatrix"))
writeMetadata(type, filename = file.path(tempdir(),"MyMetadata"))

# Save resulting data in csv format (to a temporary location as an example)

writeIntensity(int, filename = file.path(tempdir(),"MyintMatrix"),
               format = "csv")
writeMetadata(type, filename = file.path(tempdir(),"MyMetadata"),
               format = "csv")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
