<!DOCTYPE html><html><head><title>Help for package maldipickr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {maldipickr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#check_spectra'><p>Evaluate the spectra regularities</p></a></li>
<li><a href='#delineate_with_identification'><p>Delineate clusters from taxonomic identifications</p></a></li>
<li><a href='#delineate_with_similarity'><p>Delineate clusters from a similarity matrix</p></a></li>
<li><a href='#get_spectra_names'><p>Extract spectra names and check for uniqueness</p></a></li>
<li><a href='#import_biotyper_spectra'><p>Importing spectra from the Bruker MALDI Biotyper device</p></a></li>
<li><a href='#import_spede_clusters'><p>Import clusters results generated by SPeDE</p></a></li>
<li><a href='#is_well_on_edge'><p>Identify the wells on the plate's edge</p></a></li>
<li><a href='#maldipickr-package'><p>maldipickr: Dereplicate and Cherry-Pick Mass Spectrometry Spectra</p></a></li>
<li><a href='#merge_processed_spectra'><p>Merge multiple processed spectra and peaks</p></a></li>
<li><a href='#pick_spectra'><p>Cherry-pick Bruker MALDI Biotyper spectra</p></a></li>
<li><a href='#process_spectra'><p>Process Bruker MALDI Biotyper spectra <em>à la</em> Strejcek et al. (2018)</p></a></li>
<li><a href='#read_biotyper_report'><p>Importing Bruker MALDI Biotyper CSV report</p></a></li>
<li><a href='#read_many_biotyper_reports'><p>Importing a list of Bruker MALDI Biotyper CSV reports</p></a></li>
<li><a href='#remove_spectra'><p>Remove (raw or processed) spectra</p></a></li>
<li><a href='#set_reference_spectra'><p>Set a reference spectrum for each cluster</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Dereplicate and Cherry-Pick Mass Spectrometry Spectra</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Convenient wrapper functions for the analysis of
    matrix-assisted laser desorption/ionization-time-of-flight (MALDI-TOF)
    spectra data in order to select only representative spectra (also
    called cherry-pick). The package covers the preprocessing and
    dereplication steps (based on Strejcek, Smrhova, Junkova and Uhlik
    (2018) &lt;<a href="https://doi.org/10.3389%2Ffmicb.2018.01294">doi:10.3389/fmicb.2018.01294</a>&gt;) needed to cluster MALDI-TOF
    spectra before the final cherry-picking step. It enables the easy
    exclusion of spectra and/or clusters to accommodate complex
    cherry-picking strategies. Alternatively, cherry-picking using
    taxonomic identification MALDI-TOF data is made easy with functions to
    import inconsistently formatted reports.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ClavelLab/maldipickr">https://github.com/ClavelLab/maldipickr</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ClavelLab/maldipickr/issues">https://github.com/ClavelLab/maldipickr/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, lifecycle, magrittr, MALDIquant, readBrukerFlexData,
rlang, stats, tibble, tidyr, tidyselect, tools, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, spelling, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Config/fusen/version:</td>
<td>0.5.2</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-14 11:48:23 UTC; cpauvert</td>
</tr>
<tr>
<td>Author:</td>
<td>Charlie Pauvert <a href="https://orcid.org/0000-0001-9832-2507"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  David Wylensek <a href="https://orcid.org/0000-0002-8424-5712"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Selina Nüchtern [ctb],
  Thomas Clavel <a href="https://orcid.org/0000-0002-7229-5595"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb, fnd, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Charlie Pauvert &lt;cpauvert@ukaachen.de&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-14 12:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='check_spectra'>Evaluate the spectra regularities</h2><span id='topic+check_spectra'></span>

<h3>Description</h3>

<p>Assess whether all the spectra in the list are not empty, of the same length and correspond to profile data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_spectra(spectra_list, tolerance = sqrt(.Machine$double.eps))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_spectra_+3A_spectra_list">spectra_list</code></td>
<td>
<p>A list of <a href="MALDIquant.html#topic+MassSpectrum-class">MALDIquant::MassSpectrum</a> objects</p>
</td></tr>
<tr><td><code id="check_spectra_+3A_tolerance">tolerance</code></td>
<td>
<p>A numeric indicating the accepted tolerance to the spectra length.
The default value is the machine numerical precision and is close to 1.5e-8.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of logical vectors of length <code>spectra_list</code> indicating if the spectra are empty (<code>is_empty</code>), of an odd length (<code>is_outlier_length</code>) or not a profile spectra (<code>is_not_regular</code>).
</p>


<h3>See Also</h3>

<p><a href="#topic+process_spectra">process_spectra</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
directory_biotyper_spectra &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
)
# Import the six spectra
spectra_list &lt;- import_biotyper_spectra(directory_biotyper_spectra)
# Display the list of checks, with FALSE where no anomaly is detected
check_spectra(spectra_list)
# The overall sanity can be checked with Reduce
Reduce(any, check_spectra(spectra_list)) # Should be FALSE
</code></pre>

<hr>
<h2 id='delineate_with_identification'>Delineate clusters from taxonomic identifications</h2><span id='topic+delineate_with_identification'></span>

<h3>Description</h3>

<p>From the report of taxonomic identification produced by the Bruker MALDI Biotyper
spectra sharing the same identification are labeled in the same cluster.
Spectra with unknown identification (e.g., due to database completeness) are set in unique cluster.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delineate_with_identification(tibble_report)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delineate_with_identification_+3A_tibble_report">tibble_report</code></td>
<td>
<p>A tibble of <em>n</em> rows, with <em>n</em> the number of spectra,
produced by <code><a href="#topic+read_biotyper_report">read_biotyper_report()</a></code> or <code><a href="#topic+read_many_biotyper_reports">read_many_biotyper_reports()</a></code>. The long format and the best hits options are expected to be used in these functions to produce a compliant input tibble.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As all unknown identification are considered unique clusters <em>within one input tibble</em>, it is important to consider whether the taxonomic identifications come from a single report or multiple reports, depending on the research question. A message is displayed to confirm from which type of reports the delineation was done.
</p>


<h3>Value</h3>

<p>A tibble of <em>n</em> rows for each spectra and 3 columns:
</p>

<ul>
<li> <p><code>name</code>: the spectra names from the <code>name</code> column from the output of either <code><a href="#topic+read_biotyper_report">read_biotyper_report()</a></code> or <code><a href="#topic+read_many_biotyper_reports">read_many_biotyper_reports()</a></code>.
</p>
</li>
<li> <p><code>membership</code>: integers stating the cluster number to which the spectra belong to. It starts from 1 to <em>c</em>, the total number of clusters.
</p>
</li>
<li> <p><code>cluster_size</code>: integers indicating the total number of spectra in the corresponding cluster.
</p>
</li></ul>



<h3>See Also</h3>

<p><a href="#topic+delineate_with_similarity">delineate_with_similarity</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>report_unknown &lt;- read_biotyper_report(
  system.file("biotyper_unknown.csv", package = "maldipickr")
)
delineate_with_identification(report_unknown)
</code></pre>

<hr>
<h2 id='delineate_with_similarity'>Delineate clusters from a similarity matrix</h2><span id='topic+delineate_with_similarity'></span>

<h3>Description</h3>

<p>From a matrix of spectra similarity (e.g., with the cosine metric,
or Pearson product moment), infer the species clusters based on a
threshold <strong>above</strong> (or <strong>equal to</strong>) which spectra are considered alike.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>delineate_with_similarity(sim_matrix, threshold, method = "complete")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="delineate_with_similarity_+3A_sim_matrix">sim_matrix</code></td>
<td>
<p>A <code class="reqn">n \times n</code> similarity matrix, with <code class="reqn">n</code> the number of spectra. Columns should be named as the rows.</p>
</td></tr>
<tr><td><code id="delineate_with_similarity_+3A_threshold">threshold</code></td>
<td>
<p>A numeric value indicating the minimal similarity between two spectra. Adjust accordingly to the similarity metric used.</p>
</td></tr>
<tr><td><code id="delineate_with_similarity_+3A_method">method</code></td>
<td>
<p>The method of hierarchical clustering to use. The default and recommended method is &quot;complete&quot;, but any methods from <a href="stats.html#topic+hclust">stats::hclust</a> are valid.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The similarity matrix is converted to a distance matrix by subtracting the value one. This approach works for cosine similarity and positive correlations that have an upper bound of 1. Clusters are then delineated using hierarchical clustering. The default method of hierarchical clustering is the complete linkage (also known as farthest neighbor clustering) to ensure that the within-group minimum similarity of each cluster respects the threshold. See the Details section of <a href="stats.html#topic+hclust">stats::hclust</a> for others valid methods to use.
</p>


<h3>Value</h3>

<p>A tibble of <code class="reqn">n</code> rows for each spectra and 3 columns:
</p>

<ul>
<li> <p><code>name</code>: the rownames of the similarity matrix indicating the spectra names
</p>
</li>
<li> <p><code>membership</code>: integers stating the cluster number to which the spectra belong to. It starts from 1 to <code class="reqn">c</code>, the total number of clusters.
</p>
</li>
<li> <p><code>cluster_size</code>: integers indicating the total number of spectra in the corresponding cluster.
</p>
</li></ul>



<h3>See Also</h3>

<p>For similarity metrics: <a href="https://rdrr.io/cran/coop/man/cosine.html"><code>coop::tcosine</code></a>, <a href="https://rdrr.io/r/stats/cor.html"><code>stats::cor</code></a>, <a href="https://rdrr.io/cran/Hmisc/man/rcorr.html"><code>Hmisc::rcorr</code></a>. For using taxonomic identifications for clusters : <a href="#topic+delineate_with_identification">delineate_with_identification</a>. For further analyses: <a href="#topic+set_reference_spectra">set_reference_spectra</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Toy similarity matrix between the six example spectra of
#  three species. The cosine metric is used and a value of
#  zero indicates dissimilar spectra and a value of one
#  indicates identical spectra.
cosine_similarity &lt;- matrix(
  c(
    1, 0.79, 0.77, 0.99, 0.98, 0.98,
    0.79, 1, 0.98, 0.79, 0.8, 0.8,
    0.77, 0.98, 1, 0.77, 0.77, 0.77,
    0.99, 0.79, 0.77, 1, 1, 0.99,
    0.98, 0.8, 0.77, 1, 1, 1,
    0.98, 0.8, 0.77, 0.99, 1, 1
  ),
  nrow = 6,
  dimnames = list(
    c(
      "species1_G2", "species2_E11", "species2_E12",
      "species3_F7", "species3_F8", "species3_F9"
    ),
    c(
      "species1_G2", "species2_E11", "species2_E12",
      "species3_F7", "species3_F8", "species3_F9"
    )
  )
)
# Delineate clusters based on a 0.92 threshold applied
#  to the similarity matrix
delineate_with_similarity(cosine_similarity, threshold = 0.92)
</code></pre>

<hr>
<h2 id='get_spectra_names'>Extract spectra names and check for uniqueness</h2><span id='topic+get_spectra_names'></span>

<h3>Description</h3>

<p>Given the list of raw spectra, <code>get_spectra_names()</code> extracts the spectra names
using the file metadata, and warns if the associated sanitized names are not unique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_spectra_names(spectra_list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_spectra_names_+3A_spectra_list">spectra_list</code></td>
<td>
<p>A list of <a href="MALDIquant.html#topic+MassSpectrum-class">MALDIquant::MassSpectrum</a> objects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with four columns
</p>

<ul>
<li> <p><code>sanitized_name</code>: spectra names based on <code>fullName</code> where dots and dashes are converted to underscores
</p>
</li>
<li> <p><code>name</code>: spectra name using the <code>name</code> label in the spectra metadata
</p>
</li>
<li> <p><code>fullName</code>: spectra full name using the <code>fullName</code> label in the spectra metadata
</p>
</li>
<li> <p><code>file</code>: the path to the raw spectra data
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
directory_biotyper_spectra &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
)
# Import the six spectra
spectra_list &lt;- import_biotyper_spectra(directory_biotyper_spectra)
# Extract the names
get_spectra_names(spectra_list)

# Artificially create duplicated entries to show the warning
get_spectra_names(spectra_list[c(1,1)])
</code></pre>

<hr>
<h2 id='import_biotyper_spectra'>Importing spectra from the Bruker MALDI Biotyper device</h2><span id='topic+import_biotyper_spectra'></span>

<h3>Description</h3>

<p>This function is a wrapper around the <code><a href="readBrukerFlexData.html#topic+readBrukerFlexDir">readBrukerFlexData::readBrukerFlexDir()</a></code> to read both <code>acqus</code> and <code>acqu</code> MALDI files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_biotyper_spectra(
  biotyper_directory,
  remove_calibration = c("BTS", "Autocalibration")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_biotyper_spectra_+3A_biotyper_directory">biotyper_directory</code></td>
<td>
<p>A path to the folder tree with the spectra to be imported.</p>
</td></tr>
<tr><td><code id="import_biotyper_spectra_+3A_remove_calibration">remove_calibration</code></td>
<td>
<p>A vector of characters used as regex to indicate which (calibration) spectra are going to be removed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When using <code><a href="readBrukerFlexData.html#topic+readBrukerFlexDir">readBrukerFlexData::readBrukerFlexDir()</a></code> on <code>acqus</code> files (instead of the native <code>acqu</code> files), the function will fail with the following error message:
</p>
<div class="sourceCode"><pre>Error in .readAcquFile(fidFile = fidFile, verbose = verbose) :
File ‘/data/maldi_dir/targetA/0_D10/1/1SLin/acqu’ doesn't exists!
</pre></div>
<p>But it turns out that <code>acqu</code> and <code>acqus</code> files <a href="https://github.com/sgibb/readBrukerFlexData/wiki/acqu-file">are the same</a>, so the function here create <code>acqu</code> symbolic links that point to <code>acqus</code> files.
</p>


<h3>Value</h3>

<p>A list of <a href="MALDIquant.html#topic+MassSpectrum-class">MALDIquant::MassSpectrum</a> objects
</p>


<h3>See Also</h3>

<p><a href="#topic+check_spectra">check_spectra</a>, <a href="#topic+process_spectra">process_spectra</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
directory_biotyper_spectra &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
)
# Import the six spectra
spectra_list &lt;- import_biotyper_spectra(directory_biotyper_spectra)
# Display the list of spectra
spectra_list
</code></pre>

<hr>
<h2 id='import_spede_clusters'>Import clusters results generated by SPeDE</h2><span id='topic+import_spede_clusters'></span>

<h3>Description</h3>

<p>Reformat the table output from the analysis of raw Bruker MALDI Biotyper
spectra by the SPeDE tool from Dumolin et al. (2019) to be consistent with the Strejcek et al. (2018) procedure followed
in the <a href="#topic+maldipickr">maldipickr</a> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>import_spede_clusters(path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="import_spede_clusters_+3A_path">path</code></td>
<td>
<p>Path to the comma separated table generated by SPeDE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with the following columns:
</p>

<ul>
<li> <p><code>name</code>: a character denoting the spectra name (all spaces, dashes and dots are replaced by underscores &quot;_&quot; in SPeDE)
</p>
</li>
<li> <p><code>membership</code>: integers stating the cluster number to which the spectra belong to. It starts from 1 to <em>c</em>, the total number of clusters.
</p>
</li>
<li> <p><code>cluster_size</code>: integers indicating the total number of spectra in the corresponding cluster.
</p>
</li>
<li> <p><code>quality</code>: a character indicating the spectra quality category by SPeDE, out of GREEN, ORANGE and RED.
</p>
</li>
<li> <p><code>is_reference</code>: a logical indicating whether the corresponding spectra is a reference spectra of the cluster.
</p>
</li></ul>



<h3>References</h3>

<p>Dumolin C, Aerts M, Verheyde B, Schellaert S, Vandamme T, Van Der Jeugt F, De Canck E, Cnockaert M, Wieme AD, Cleenwerck I, Peiren J, Dawyndt P, Vandamme P, &amp; Carlier A. (2019). &quot;Introducing SPeDE: High-Throughput Dereplication and Accurate Determination of Microbial Diversity from Matrix-Assisted Laser Desorption–Ionization Time of Flight Mass Spectrometry Data&quot;. <em>MSystems</em> 4(5). <a href="doi:10.1128/msystems.00437-19">doi:10.1128/msystems.00437-19</a>.
</p>


<h3>See Also</h3>

<p><a href="https://github.com/LM-UGent/SPeDE">https://github.com/LM-UGent/SPeDE</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Reformat the output from SPeDE table
# https://github.com/LM-UGent/SPeDE
import_spede_clusters(
  system.file("spede.csv", package = "maldipickr")
)
</code></pre>

<hr>
<h2 id='is_well_on_edge'>Identify the wells on the plate's edge</h2><span id='topic+is_well_on_edge'></span>

<h3>Description</h3>

<p>Identify the wells on the plate's edge
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_well_on_edge(
  well_number,
  plate_layout = c(96, 384),
  edges = c("top", "bottom", "left", "right"),
  details = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_well_on_edge_+3A_well_number">well_number</code></td>
<td>
<p>A vector of positive numeric well identifier</p>
</td></tr>
<tr><td><code id="is_well_on_edge_+3A_plate_layout">plate_layout</code></td>
<td>
<p>An integer indicating the maximum number of well on the plate</p>
</td></tr>
<tr><td><code id="is_well_on_edge_+3A_edges">edges</code></td>
<td>
<p>A character vector pointing which plate edges should be considered</p>
</td></tr>
<tr><td><code id="is_well_on_edge_+3A_details">details</code></td>
<td>
<p>A logical controlling whether a <a href="base.html#topic+data.frame">data.frame</a> with more details should be returned</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Flag the wells located on the edges of a 96- or 384-well plate, based on the following well numbering:
</p>

<ul>
<li><p> Well numbers start at 1
</p>
</li>
<li><p> Well are numbered from left to right and then top to bottom of the plate.
</p>
</li></ul>



<h3>Value</h3>

<p>A logical vector, the same length as <code>well_number</code> indicating whether the well is on the edge. If <code>details = TRUE</code>, the function returns a <a href="base.html#topic+data.frame">data.frame</a> that complements the logical vector with the <code>well_number</code>, row and column positions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Logical vector indicating whether the wells are on the four edges
is_well_on_edge(1:96, plate_layout = 96)
# More details can be obtained to verify the results
well_df &lt;- is_well_on_edge(1:96, plate_layout = 96, details = TRUE)
# And the resulting prediction displayed
matrix(well_df$is_edge, ncol = max(well_df$col), byrow = TRUE)
</code></pre>

<hr>
<h2 id='maldipickr-package'>maldipickr: Dereplicate and Cherry-Pick Mass Spectrometry Spectra</h2><span id='topic+maldipickr'></span><span id='topic+maldipickr-package'></span>

<h3>Description</h3>

<p><img src="../help/figures/logo.png" style='float: right' alt='logo' width='120' />
</p>
<p>Convenient wrapper functions for the analysis of matrix-assisted laser desorption/ionization-time-of-flight (MALDI-TOF) spectra data in order to select only representative spectra (also called cherry-pick). The package covers the preprocessing and dereplication steps (based on Strejcek, Smrhova, Junkova and Uhlik (2018) <a href="https://doi.org/10.3389/fmicb.2018.01294">doi:10.3389/fmicb.2018.01294</a>) needed to cluster MALDI-TOF spectra before the final cherry-picking step. It enables the easy exclusion of spectra and/or clusters to accommodate complex cherry-picking strategies. Alternatively, cherry-picking using taxonomic identification MALDI-TOF data is made easy with functions to import inconsistently formatted reports.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Charlie Pauvert <a href="mailto:cpauvert@ukaachen.de">cpauvert@ukaachen.de</a> (<a href="https://orcid.org/0000-0001-9832-2507">ORCID</a>) [copyright holder]
</p>
<p>Other contributors:
</p>

<ul>
<li><p> David Wylensek (<a href="https://orcid.org/0000-0002-8424-5712">ORCID</a>) [contributor]
</p>
</li>
<li><p> Selina Nüchtern [contributor]
</p>
</li>
<li><p> Thomas Clavel (<a href="https://orcid.org/0000-0002-7229-5595">ORCID</a>) [contributor, funder, copyright holder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/ClavelLab/maldipickr">https://github.com/ClavelLab/maldipickr</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/ClavelLab/maldipickr/issues">https://github.com/ClavelLab/maldipickr/issues</a>
</p>
</li></ul>


<hr>
<h2 id='merge_processed_spectra'>Merge multiple processed spectra and peaks</h2><span id='topic+merge_processed_spectra'></span>

<h3>Description</h3>

<p>Aggregate multiple processed spectra, their associated peaks and metadata into a feature matrix and a concatenated metadata table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_processed_spectra(
  processed_spectra,
  remove_peakless_spectra = TRUE,
  interpolate_missing = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge_processed_spectra_+3A_processed_spectra">processed_spectra</code></td>
<td>
<p>A <a href="base.html#topic+list">list</a> of the processed spectra and associated peaks and metadata in two possible formats:
</p>

<ul>
<li><p> A list of <strong>in-memory objects</strong> (named <code>spectra</code>, <code>peaks</code>, <code>metadata</code>) produced by <a href="#topic+process_spectra">process_spectra</a>.
</p>
</li>
<li> <p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> A list of <strong>paths</strong> to RDS files produced by <a href="#topic+process_spectra">process_spectra</a> when using the <code>rds_prefix</code> option.
</p>
</li></ul>
</td></tr>
<tr><td><code id="merge_processed_spectra_+3A_remove_peakless_spectra">remove_peakless_spectra</code></td>
<td>
<p>A logical indicating whether to discard the spectra without detected peaks.</p>
</td></tr>
<tr><td><code id="merge_processed_spectra_+3A_interpolate_missing">interpolate_missing</code></td>
<td>
<p>A logical indicating if intensity values for missing peaks should be interpolated from the processed spectra signal or left NA which would then be converted to 0.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <em>n</em>×<em>p</em> matrix, with <em>n</em> spectra as rows and <em>p</em> features as columns that are the peaks found in all the processed spectra.
</p>


<h3>See Also</h3>

<p><a href="#topic+process_spectra">process_spectra</a>, the &quot;Value&quot; section in <a href="https://rdrr.io/cran/MALDIquant/man/intensityMatrix-functions.html"><code>MALDIquant::intensityMatrix</code></a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
directory_biotyper_spectra &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
)
# Import the six spectra
spectra_list &lt;- import_biotyper_spectra(directory_biotyper_spectra)
# Transform the spectra signals according to Strejcek et al. (2018)
processed &lt;- process_spectra(spectra_list)
# Merge the spectra to produce the feature matrix
fm &lt;- merge_processed_spectra(list(processed))
# The feature matrix has 6 spectra as rows and
#  35 peaks as columns
dim(fm)
# Notice the difference when the interpolation is turned off
fm_no_interpolation &lt;- merge_processed_spectra(
  list(processed),
  interpolate_missing = FALSE
)
sum(fm == 0) # 0
sum(fm_no_interpolation == 0) # 68

# Multiple runs can be aggregated using list()
# Merge the spectra to produce the feature matrix
fm_all &lt;- merge_processed_spectra(list(processed, processed, processed))
# The feature matrix has 3×6=18 spectra as rows and
#  35 peaks as columns
dim(fm_all)
</code></pre>

<hr>
<h2 id='pick_spectra'>Cherry-pick Bruker MALDI Biotyper spectra</h2><span id='topic+pick_spectra'></span>

<h3>Description</h3>

<p>Using the clusters information, and potential additional metadata as external
criteria, spectra are labeled as to be picked for each cluster. Note that some
spectra and therefore clusters can be explicitly removed (<em>masked</em>)
from the picking decision if they have been previously picked
or should be discarded, using logical columns in the metadata table.
If no metadata are provided, the reference spectra of each cluster will be picked.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pick_spectra(
  cluster_df,
  metadata_df = NULL,
  criteria_column = NULL,
  hard_mask_column = NULL,
  soft_mask_column = NULL,
  is_descending_order = TRUE,
  is_sorted = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pick_spectra_+3A_cluster_df">cluster_df</code></td>
<td>
<p>A tibble with clusters information
from the <a href="#topic+delineate_with_similarity">delineate_with_similarity</a> or the <a href="#topic+import_spede_clusters">import_spede_clusters</a> function.</p>
</td></tr>
<tr><td><code id="pick_spectra_+3A_metadata_df">metadata_df</code></td>
<td>
<p>Optional tibble with relevant metadata to guide the
picking process (e.g., OD600).</p>
</td></tr>
<tr><td><code id="pick_spectra_+3A_criteria_column">criteria_column</code></td>
<td>
<p>Optional character indicating the column in <code>metadata_df</code>
to be used as a criteria.</p>
</td></tr>
<tr><td><code id="pick_spectra_+3A_hard_mask_column">hard_mask_column</code></td>
<td>
<p>Column name in the <code>cluster_df</code> or <code>metadata_df</code> tibble indicating whether the spectra, <strong>and the clusters to which they belong</strong> should be discarded (<code>TRUE</code>) or not (<code>FALSE</code>) before the picking decision.</p>
</td></tr>
<tr><td><code id="pick_spectra_+3A_soft_mask_column">soft_mask_column</code></td>
<td>
<p>Column name in the <code>cluster_df</code> or <code>metadata_df</code> tibble indicating whether the spectra should be discarded (<code>TRUE</code>) or not (<code>FALSE</code>) before the picking decision.</p>
</td></tr>
<tr><td><code id="pick_spectra_+3A_is_descending_order">is_descending_order</code></td>
<td>
<p>Optional logical indicating whether to sort the <code>criteria_column</code> from the highest-to-lowest value (<code>TRUE</code>) or lowest-to-highest (<code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="pick_spectra_+3A_is_sorted">is_sorted</code></td>
<td>
<p>Optional logical to indicate that the <code>cluster_df</code> is
already sorted by cluster based on (usually multiple) internal criteria to
pick the first of each cluster. This flag is <strong>overridden</strong> if a <code>metadata_df</code>
is provided.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with as many rows as <code>cluster_df</code> with an additional logical
column named <code>to_pick</code> to indicate whether the colony associated to the spectra
should be picked. If <code>metadata_df</code> is provided, then additional columns from
this tibble are added to the returned tibble.
</p>


<h3>See Also</h3>

<p><a href="#topic+delineate_with_similarity">delineate_with_similarity</a>, <a href="#topic+set_reference_spectra">set_reference_spectra</a>. For a useful utility function to soft-mask specific spectra: <a href="#topic+is_well_on_edge">is_well_on_edge</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># 0. Load a toy example of a tibble of clusters created by
#   the `delineate_with_similarity` function.
clusters &lt;- readRDS(
  system.file("clusters_tibble.RDS",
    package = "maldipickr"
  )
)
# 1. By default and if no other metadata are provided,
#   the function picks reference spectra for each clusters.
#
# N.B: The spectra `name` and `to_pick` columns are moved to the left
# only for clarity using the `relocate()` function.
#
pick_spectra(clusters) %&gt;%
  dplyr::relocate(name, to_pick) # only for clarity

# 2.1 Simulate OD600 values with uniform distribution
#  for each of the colonies we measured with
#  the Bruker MALDI Biotyper
set.seed(104)
metadata &lt;- dplyr::transmute(
  clusters,
  name = name, OD600 = runif(n = nrow(clusters))
)
metadata

# 2.2 Pick the spectra based on the highest
#   OD600 value per cluster
pick_spectra(clusters, metadata, "OD600") %&gt;%
  dplyr::relocate(name, to_pick) # only for clarity

# 3.1 Say that the wells on the right side of the plate are
#   used for negative controls and should not be picked.
metadata &lt;- metadata %&gt;% dplyr::mutate(
  well = gsub(".*[A-Z]([0-9]{1,2}$)", "\\1", name) %&gt;%
    strtoi(),
  is_edge = is_well_on_edge(
    well_number = well, plate_layout = 96, edges = "right"
  )
)

# 3.2 Pick the spectra after discarding (or soft masking)
#   the spectra indicated by the `is_edge` column.
pick_spectra(clusters, metadata, "OD600",
  soft_mask_column = "is_edge"
) %&gt;%
  dplyr::relocate(name, to_pick) # only for clarity

# 4.1 Say that some spectra were picked before
#   (e.g., in the column F) in a previous experiment.
# We do not want to pick clusters with those spectra
#   included to limit redundancy.
metadata &lt;- metadata %&gt;% dplyr::mutate(
  picked_before = grepl("_F", name)
)
# 4.2 Pick the spectra from clusters without spectra
#   labeled as `picked_before` (hard masking).
pick_spectra(clusters, metadata, "OD600",
  hard_mask_column = "picked_before"
) %&gt;%
  dplyr::relocate(name, to_pick) # only for clarity
</code></pre>

<hr>
<h2 id='process_spectra'>Process Bruker MALDI Biotyper spectra <em>à la</em> Strejcek et al. (2018)</h2><span id='topic+process_spectra'></span>

<h3>Description</h3>

<p>Process Bruker MALDI Biotyper spectra <em>à la</em> Strejcek et al. (2018)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>process_spectra(
  spectra_list,
  spectra_names = get_spectra_names(spectra_list),
  rds_prefix = deprecated()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="process_spectra_+3A_spectra_list">spectra_list</code></td>
<td>
<p>A list of <a href="MALDIquant.html#topic+MassSpectrum-class">MALDIquant::MassSpectrum</a> objects.</p>
</td></tr>
<tr><td><code id="process_spectra_+3A_spectra_names">spectra_names</code></td>
<td>
<p>A <a href="tibble.html#topic+tibble">tibble::tibble</a> (or <a href="base.html#topic+data.frame">data.frame</a>) of sanitized spectra names by default from <a href="#topic+get_spectra_names">get_spectra_names</a>. If provided manually, the column <code>sanitized_name</code> will be used to name the spectra.</p>
</td></tr>
<tr><td><code id="process_spectra_+3A_rds_prefix">rds_prefix</code></td>
<td>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#deprecated"><img src="../help/figures/lifecycle-deprecated.svg" alt='[Deprecated]' /></a> Writing to disk as RDS is no longer supported. A character indicating the prefix for the <code>.RDS</code> output files to be written in the <code>processed</code> directory. By default, no prefix are given and thus no files are written.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Based on the original implementation, the function performs the following tasks:
</p>

<ol>
<li><p> Square-root transformation
</p>
</li>
<li><p> Mass range trimming to 4-10 kDa as they were deemed most determinant by Strejcek et al. (2018)
</p>
</li>
<li><p> Signal smoothing using the Savitzky-Golay method and a half window size of 20
</p>
</li>
<li><p> Baseline correction with the SNIP procedure
</p>
</li>
<li><p> Normalization by Total Ion Current
</p>
</li>
<li><p> Peak detection using the SuperSmoother procedure and with a signal-to-noise ratio above 3
</p>
</li>
<li><p> Peak filtering. This step has been added to discard peaks with a negative signal-to-noise ratio probably due to being on the edge of the mass range.
</p>
</li></ol>



<h3>Value</h3>

<p>A named list of three objects:
</p>

<ul>
<li> <p><code>spectra</code>: a list the length of the spectra list of <a href="MALDIquant.html#topic+MassSpectrum-class">MALDIquant::MassSpectrum</a> objects.
</p>
</li>
<li> <p><code>peaks</code>: a list the length of the spectra list of <a href="MALDIquant.html#topic+MassPeaks-class">MALDIquant::MassPeaks</a> objects.
</p>
</li>
<li> <p><code>metadata</code>: a tibble indicating the median signal-to-noise ratio (<code>SNR</code>) and peaks number for all spectra list (<code>peaks</code>), with spectra names in the <code>name</code> column.
</p>
</li></ul>



<h3>Note</h3>

<p>The original R code on which this function is based is accessible at: <a href="https://github.com/strejcem/MALDIvs16S">https://github.com/strejcem/MALDIvs16S</a>
</p>


<h3>References</h3>

<p>Strejcek M, Smrhova T, Junkova P &amp; Uhlik O (2018). “Whole-Cell MALDI-TOF MS versus 16S rRNA Gene Analysis for Identification and Dereplication of Recurrent Bacterial Isolates.” <em>Frontiers in Microbiology</em> 9 <a href="doi:10.3389/fmicb.2018.01294">doi:10.3389/fmicb.2018.01294</a>.
</p>


<h3>See Also</h3>

<p><a href="#topic+import_biotyper_spectra">import_biotyper_spectra</a> and <a href="#topic+check_spectra">check_spectra</a> for the inputs and <a href="#topic+merge_processed_spectra">merge_processed_spectra</a> for further analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
directory_biotyper_spectra &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
)
# Import the six spectra
spectra_list &lt;- import_biotyper_spectra(directory_biotyper_spectra)
# Transform the spectra signals according to Strejcek et al. (2018)
processed &lt;- process_spectra(spectra_list)
# Overview of the list architecture that is returned
#  with the list of processed spectra, peaks identified and the
#  metadata table
str(processed, max.level = 2)
# A detailed view of the metadata with the median signal-to-noise
#  ratio (SNR) and the number of peaks
processed$metadata
</code></pre>

<hr>
<h2 id='read_biotyper_report'>Importing Bruker MALDI Biotyper CSV report</h2><span id='topic+read_biotyper_report'></span>

<h3>Description</h3>

<p>The header-less table exported by the Compass software in the Bruker MALDI
Biotyper device is separated by semi-colons and has empty columns which prevent
an easy import in R. This function reads the report correctly as a tibble.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_biotyper_report(path, best_hits = TRUE, long_format = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_biotyper_report_+3A_path">path</code></td>
<td>
<p>Path to the semi-colon separated table</p>
</td></tr>
<tr><td><code id="read_biotyper_report_+3A_best_hits">best_hits</code></td>
<td>
<p>A logical indicating whether to return only the best hits for each target analyzed</p>
</td></tr>
<tr><td><code id="read_biotyper_report_+3A_long_format">long_format</code></td>
<td>
<p>A logical indicating whether the table is in the long format (many rows) or wide format (many columns) when showing all the hits. This option has no effect when <code>best_hits = TRUE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The header-less table contains identification information for each target processed by
the Biotyper device and once processed by the <code>read_biotyper_report</code>,
the following seven columns are available in the tibble, <em>when using the <code>best_hits = TRUE</code> option</em>:
</p>

<ul>
<li> <p><code>name</code>: a character indicating the name of the spot of the MALDI target (i.e., plate)
</p>
</li>
<li> <p><code>sample_name</code>: the character string provided during the preparation of the MALDI target (i.e., plate)
</p>
</li>
<li> <p><code>hit_rank</code>: an integer indicating the rank of the hit for the corresponding target and identification
</p>
</li>
<li> <p><code>bruker_quality</code>: a character encoding the quality of the identification with potentially multiple &quot;+&quot; symbol or only one &quot;-&quot;
</p>
</li>
<li> <p><code>bruker_species</code>: the species name associated with the MALDI spectrum analyzed.
</p>
</li>
<li> <p><code>bruker_taxid</code>: the NCBI Taxonomy Identifier of the species name in the column species
</p>
</li>
<li> <p><code>bruker_hash</code>: a hash from an undocumented checksum function probably to encode the database entry.
</p>
</li>
<li> <p><code>bruker_log</code>: the log-score of the identification.
</p>
</li></ul>

<p>When all hits are returned (with <code>best_hits = FALSE</code>), the default output format is the long format (<code>long_format = TRUE</code>), meaning that the previous columns remain
unchanged, but all hits are now returned, thus increasing the number of rows.
</p>
<p>When all hits are returned (with <code>best_hits = FALSE</code>) <em>using the wide format</em> (<code style="white-space: pre;">&#8288;long_format = FALSE), the two columns &#8288;</code>name<code>and</code>sample_name<code style="white-space: pre;">&#8288;remains unchanged, but the five columns prefixed by&#8288;</code>bruker_' contain the hit rank, <strong>creating a tibble of 52 columns</strong>:
</p>

<ul>
<li> <p><code>bruker_01_quality</code>
</p>
</li>
<li> <p><code>bruker_01_species</code>
</p>
</li>
<li> <p><code>bruker_01_taxid</code>
</p>
</li>
<li> <p><code>bruker_01_hash</code>
</p>
</li>
<li> <p><code>bruker_01_log</code>
</p>
</li>
<li> <p><code>bruker_02_quality</code>
</p>
</li>
<li><p> ...
</p>
</li>
<li> <p><code>bruker_10_species</code>
</p>
</li>
<li> <p><code>bruker_10_taxid</code>
</p>
</li>
<li> <p><code>bruker_10_hash</code>
</p>
</li>
<li> <p><code>bruker_10_log</code>
</p>
</li></ul>



<h3>Value</h3>

<p>A tibble of 7 columns (<code>best_hits = TRUE</code>) or 52 columns (<code>best_hits = FALSE</code>). See Details for the description of the columns.
</p>


<h3>Note</h3>

<p>A report that contains only spectra with no peaks found will return a tibble of 0 rows and a warning message.
</p>


<h3>See Also</h3>

<p><a href="#topic+read_many_biotyper_reports">read_many_biotyper_reports</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get a example Bruker report
biotyper &lt;- system.file("biotyper.csv", package = "maldipickr")
# Import the report as a tibble
report_tibble &lt;- read_biotyper_report(biotyper)
# Display the tibble
report_tibble
</code></pre>

<hr>
<h2 id='read_many_biotyper_reports'>Importing a list of Bruker MALDI Biotyper CSV reports</h2><span id='topic+read_many_biotyper_reports'></span>

<h3>Description</h3>

<p>Importing a list of Bruker MALDI Biotyper CSV reports
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_many_biotyper_reports(path_to_reports, report_ids, best_hits = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_many_biotyper_reports_+3A_path_to_reports">path_to_reports</code></td>
<td>
<p>A vector of paths to the csv files to be imported by <code><a href="#topic+read_biotyper_report">read_biotyper_report()</a></code>.</p>
</td></tr>
<tr><td><code id="read_many_biotyper_reports_+3A_report_ids">report_ids</code></td>
<td>
<p>A vector of character names for each of the reports.</p>
</td></tr>
<tr><td><code id="read_many_biotyper_reports_+3A_best_hits">best_hits</code></td>
<td>
<p>A logical indicating whether to return only the best hit in the <code><a href="#topic+read_biotyper_report">read_biotyper_report()</a></code> function.</p>
</td></tr>
<tr><td><code id="read_many_biotyper_reports_+3A_...">...</code></td>
<td>
<p>Name-value pairs to be passed on to <code><a href="dplyr.html#topic+mutate">dplyr::mutate()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble just like the one returned by the <code><a href="#topic+read_biotyper_report">read_biotyper_report()</a></code> function, except that the name of the spot of the MALDI target (i.e., plate) is registered to the <code>original_name</code> column (instead of the <code>name</code> column), and the column <code>name</code> consist in the provided <code>report_ids</code> used as a prefix of the <code>original_name</code> column.
</p>


<h3>Note</h3>

<p>The report identifiers are sanitized to convert all dashes (<code>-</code>) as underscores (<code style="white-space: pre;">&#8288;_&#8288;</code>).
</p>


<h3>See Also</h3>

<p><a href="#topic+read_biotyper_report">read_biotyper_report</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># List of Bruker MALDI Biotyper reports
reports_paths &lt;- system.file(
  c("biotyper.csv", "biotyper.csv", "biotyper.csv"),
  package = "maldipickr"
)
# Read the list of reports and combine them in a single tibble
read_many_biotyper_reports(
  reports_paths,
  report_ids = c("first", "second", "third"),
  # Additional metadata below are passed to dplyr::mutate
  growth_temperature = 37.0
)
</code></pre>

<hr>
<h2 id='remove_spectra'>Remove (raw or processed) spectra</h2><span id='topic+remove_spectra'></span>

<h3>Description</h3>

<p>The <code>remove_spectra()</code> function is used to discard specific spectra from (1) raw spectra list by removing them, or (2) processed spectra by removing them from the spectra, peaks and metadata objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_spectra(spectra_list, to_remove)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="remove_spectra_+3A_spectra_list">spectra_list</code></td>
<td>
<p>A list of <a href="MALDIquant.html#topic+MassSpectrum-class">MALDIquant::MassSpectrum</a> objects OR A list of processed spectra from <a href="#topic+process_spectra">process_spectra</a></p>
</td></tr>
<tr><td><code id="remove_spectra_+3A_to_remove">to_remove</code></td>
<td>
<p>The spectra to be removed. In the case of raw spectra: a logical vector same size of <code>spectra_list</code> or from <a href="#topic+check_spectra">check_spectra</a> function. In the case of processed spectra: names of the spectra as formatted in <a href="#topic+get_spectra_names">get_spectra_names</a> in the <code>sanitized_name</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The same object as <code>spectra_list</code> minus the spectra in <code>to_remove</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
directory_biotyper_spectra &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
)
# Import only the first two spectra
spectra_list &lt;- import_biotyper_spectra(directory_biotyper_spectra)[1:2]
# Introduce artificially an empty raw spectra
spectra_list &lt;- c(spectra_list, MALDIquant::createMassSpectrum(0, 0))
# Empty spectra are detected by `check_spectra()`
#   and can be removed by `remove_spectra()`
spectra_list %&gt;% 
  remove_spectra(to_remove = check_spectra(.))

# Get an example processed spectra
processed_path &lt;- system.file(
    "three_processed_spectra_with_one_peakless.RDS",
    package = "maldipickr")
processed &lt;- readRDS(processed_path) %&gt;% list()

# Remove a specific spectra
remove_spectra(processed, "empty_H12")
</code></pre>

<hr>
<h2 id='set_reference_spectra'>Set a reference spectrum for each cluster</h2><span id='topic+set_reference_spectra'></span>

<h3>Description</h3>

<p>Define a high-quality spectra as a representative
spectra of the cluster based on the highest median signal-to-noise ratio
and the number of detected peaks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_reference_spectra(cluster_df, metadata_df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="set_reference_spectra_+3A_cluster_df">cluster_df</code></td>
<td>
<p>A tibble of <em>n</em> rows for each spectra produced by <a href="#topic+delineate_with_similarity">delineate_with_similarity</a> function with at least the following columns:
</p>

<ul>
<li> <p><code>name</code>: the rownames of the similarity matrix indicating the spectra names
</p>
</li>
<li> <p><code>membership</code>: integers stating the cluster number to which the spectra belong to. It starts from 1 to <em>c</em>, the total number of clusters.
</p>
</li>
<li> <p><code>cluster_size</code>: integers indicating the total number of spectra in the corresponding cluster.
</p>
</li></ul>
</td></tr>
<tr><td><code id="set_reference_spectra_+3A_metadata_df">metadata_df</code></td>
<td>
<p>A tibble of <em>n</em> rows for each spectra produced by the <a href="#topic+process_spectra">process_spectra</a> function with median signal-to-noise ratio (<code>SNR</code>), peaks number (<code>peaks</code>), and spectra names in the <code>name</code> column.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A merged tibble in the same order as <code>cluster_df</code> with both the columns of <code>cluster_df</code> and <code>metadata_df</code>, as well as a logical column <code>is_reference</code> indicating if the spectrum is the reference spectra of the cluster.
</p>


<h3>See Also</h3>

<p><a href="#topic+delineate_with_similarity">delineate_with_similarity</a>, <a href="#topic+pick_spectra">pick_spectra</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get an example directory of six Bruker MALDI Biotyper spectra
# Import the six spectra and
# Transform the spectra signals according to Strejcek et al. (2018)
processed &lt;- system.file(
  "toy-species-spectra",
  package = "maldipickr"
) %&gt;%
  import_biotyper_spectra() %&gt;%
  process_spectra()

# Toy similarity matrix between the six example spectra of
#  three species. The cosine metric is used and a value of
#  zero indicates dissimilar spectra and a value of one
#  indicates identical spectra.
cosine_similarity &lt;- matrix(
  c(
    1, 0.79, 0.77, 0.99, 0.98, 0.98,
    0.79, 1, 0.98, 0.79, 0.8, 0.8,
    0.77, 0.98, 1, 0.77, 0.77, 0.77,
    0.99, 0.79, 0.77, 1, 1, 0.99,
    0.98, 0.8, 0.77, 1, 1, 1,
    0.98, 0.8, 0.77, 0.99, 1, 1
  ),
  nrow = 6,
  dimnames = list(
    c(
      "species1_G2", "species2_E11", "species2_E12",
      "species3_F7", "species3_F8", "species3_F9"
    ),
    c(
      "species1_G2", "species2_E11", "species2_E12",
      "species3_F7", "species3_F8", "species3_F9"
    )
  )
)
# Delineate clusters based on a 0.92 threshold applied
#  to the similarity matrix
clusters &lt;- delineate_with_similarity(
  cosine_similarity,
  threshold = 0.92
)

# Set reference spectra with the toy example
set_reference_spectra(clusters, processed$metadata)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
