<!DOCTYPE html><html lang="en"><head><title>Help for package MANCIE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MANCIE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mancie'><p> Matrix Analysis and Normalization by Concordant Information Enhancement</p></a></li>
<li><a href='#mancie_example'><p> The demo dataset for the <code>MANCIE</code> package</p></a></li>
<li><a href='#summarize_mat'><p> Summarize information in the supplementary matrix</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Matrix Analysis and Normalization by Concordant Information
Enhancement</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-03-01</td>
</tr>
<tr>
<td>Author:</td>
<td>Tao Wang, Chongzhi Zang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Tao Wang &lt;tao.wang@utsouthwestern.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.0)</td>
</tr>
<tr>
<td>Description:</td>
<td>High-dimensional data integration is a critical but difficult problem in genomics research because of potential biases from high-throughput experiments. We present MANCIE, a computational method for integrating two genomic data sets with homogenous dimensions from different sources based on a PCA procedure as an approximation to a Bayesian approach.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-01 18:28:38 UTC; twang6</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-02 01:14:19</td>
</tr>
</table>
<hr>
<h2 id='mancie'> Matrix Analysis and Normalization by Concordant Information Enhancement </h2><span id='topic+mancie'></span>

<h3>Description</h3>

<p>This function removes noise in the main matrix by utilizing information available from the supplementary matrix or summarized supplementary matrix. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mancie(mat_main,mat_supp,cutoff1=0.5,cutoff2=0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mancie_+3A_mat_main">mat_main</code></td>
<td>

<p>The main matrix or data frame. Rows are features (genes/peaks/etc) and cols are samples (conditions/replicates)
</p>
</td></tr>
<tr><td><code id="mancie_+3A_mat_supp">mat_supp</code></td>
<td>

<p>The supplementary matrix or data frame. <code>mat_supp</code> must have the same dimensions as <code>mat_main</code>
</p>
</td></tr>
<tr><td><code id="mancie_+3A_cutoff1">cutoff1</code></td>
<td>

<p>The higher cutoff. See below for explanation.
</p>
</td></tr>
<tr><td><code id="mancie_+3A_cutoff2">cutoff2</code></td>
<td>

<p>The lower cutoff. See below for explanation.
</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>If the supplementary dataset have the same genomic features on rows and samples on columns as <code>mat_main</code>, it can be directly fed to <code>mancie</code>. An example is RNA-Seq data of the same cell lines from two labs. If the supplementary dataset has different rows from <code>mat_main</code>. It need to be first summarized using <code>summarize_mat</code> to be compatible with <code>mat_main</code>. An example is RNA-Seq data and DNase-seq data of the same tissue types.
</p>
<p>The underlying rationale for using MANCIE is that the variation of genomic features in <code>mat_supp</code> are concordant with and can be used to remove noise in the variation of genomic features in <code>mat_main</code>.
</p>
<p>(a) If the correlation between row i of <code>mat_main</code> and row i of <code>mat_supp</code> is larger than <code>cutoff1</code>, the new row vector will be the first PC of the matrix formed by these two row vectors.
(b) If the correlation is between <code>cutoff1</code> and <code>cutoff2</code>, the new row vector will be the weighted average of these two rows. The weight for row i of <code>mat_main</code> is 1 and the weight for row i of <code>mat_supp</code> is the correlation between these two row vectors.
(c)If the correlation is smaller than <code>cutoff2</code>, the new row vector is the original row i of <code>mat_main</code>
</p>
<p>There should be a reasonable portion of rows that fall into the first and second category. If not, the user should check if the data they would like to try MANCIE on really fits the aforementioned rationale. The user may also vary the default values of <code>cutoff1</code> and <code>cutoff2</code> if they see fit. The <code>mancie</code> function will report percentage of rows falling into each category.
</p>


<h3>Value</h3>

<p>A modified matrix with the same dimensions as the main matrix
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summarize_mat">summarize_mat</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(mancie_example,package="MANCIE")
  sum_DNase=summarize_mat(exp,ann_exp,DNase,ann_DNase) 
  lev_exp=mancie(exp,sum_DNase) 
</code></pre>

<hr>
<h2 id='mancie_example'> The demo dataset for the <code>MANCIE</code> package
</h2><span id='topic+mancie_example'></span><span id='topic+exp'></span><span id='topic+ann_exp'></span><span id='topic+DNase'></span><span id='topic+ann_DNase'></span>

<h3>Description</h3>

<p> This demo dataset is a small portion of the Encode dataset used in our publication.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mancie_example)</code></pre>


<h3>Format</h3>

<p>  4 data frames
</p>

<hr>
<h2 id='summarize_mat'> Summarize information in the supplementary matrix </h2><span id='topic+summarize_mat'></span>

<h3>Description</h3>

<p>Summarize information in the supplementary matrix according to physical location into a new matrix with the same dimensions as the main matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarize_mat(mat_main,ann_main,mat_supp,ann_supp,n_limit=50,extend=100000,method="pca")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarize_mat_+3A_mat_main">mat_main</code></td>
<td>

<p>The main matrix or data frame. Rows are features (genes/peaks/etc) and cols are samples (conditions/replicates)
</p>
</td></tr>
<tr><td><code id="summarize_mat_+3A_ann_main">ann_main</code></td>
<td>

<p><code>ann_main</code> is a data frame that contains the genomic locations of features in <code>mat_main</code>. It must have the same number of rows as <code>mat_main</code>, and must have columns named as &quot;chr&quot;, &quot;start&quot; and &quot;end&quot;.
</p>
</td></tr>
<tr><td><code id="summarize_mat_+3A_mat_supp">mat_supp</code></td>
<td>

<p>The supplementary matrix or data frame. Rows are features (genes/peaks/etc) and cols are samples (conditions/replicates)
</p>
</td></tr>
<tr><td><code id="summarize_mat_+3A_ann_supp">ann_supp</code></td>
<td>

<p><code>ann_supp</code> os a data frame that contains the genomic locations of features in <code>mat_supp</code>. It must have the same number of rows as <code>mat_supp</code>, and must have columns named as &quot;chr&quot;, &quot;start&quot; and &quot;end&quot;.
</p>
</td></tr>
<tr><td><code id="summarize_mat_+3A_n_limit">n_limit</code></td>
<td>

<p>The most number of closet features in the supplemenatry matrix that can be used for summarization for each feature in the main matrix
</p>
</td></tr> 
<tr><td><code id="summarize_mat_+3A_extend">extend</code></td>
<td>

<p>The genomic features in the supplemenatry matrix that are no farther away than <code>extend</code> bp from the feature in question in the main matrix will be used for summarization
</p>
</td></tr>
<tr><td><code id="summarize_mat_+3A_method">method</code></td>
<td>

<p>Which method to summarize the information in the supplementary matrix when there are &gt;1 neighboring row vectors associated with the row vector in the main matrix. &quot;pca&quot; (default) or &quot;max&quot;. In the &quot;max&quot; method, the row vector of these neighboring vectors with the highest correlation with the row vector in the main matrix is used. In the &quot;pca&quot; method, PCA is caculated for these row vectors and the first principal component is used.
</p>
</td></tr> 
</table>


<h3>Details</h3>

<p>The main matrix and supplementary matrix must have the same columns corresponding to conditions or replicates. They have different features on rows that can be linked by physical location on genomes. The basic assumption is that one feature's variation in the main matrix is correlated with nearby feature(s)' principal variation in the supplementary matrix.
</p>


<h3>Value</h3>

<p>A modified matrix with the same dimensions as the main matrix
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mancie">mancie</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  data(mancie_example,package="MANCIE")
  sum_DNase=summarize_mat(exp,ann_exp,DNase,ann_DNase) 
  lev_exp=mancie(exp,sum_DNase) 
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
