<!DOCTYPE html><html><head><title>Help for package Mangrove</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Mangrove}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#getVarExp'>
<p>Calculating liability-scale variance explained by genetic risk variants</p></a></li>
<li><a href='#Mangrove-internal'><p>Internal Mangrove objects</p></a></li>
<li><a href='#Mangrove-package'>
<p>Mangrove: Risk prediction on trees</p></a></li>
<li><a href='#MangroveBetas'>
<p>The MangroveBetas class</p></a></li>
<li><a href='#MangroveContPreds'>
<p>The MangroveContPreds class</p></a></li>
<li><a href='#MangroveExamples'>
<p>Example data for the Mangrove vignette</p></a></li>
<li><a href='#MangroveORs'>
<p>The MangroveORs class</p></a></li>
<li><a href='#MangrovePed'>
<p>The MangrovePed class</p></a></li>
<li><a href='#MangroveRiskPreds'>
<p>The MangroveRiskPreds class</p></a></li>
<li><a href='#MangroveSample'>
<p>The MangroveSample class</p></a></li>
<li><a href='#MangroveTree'>
<p>The MangroveTree class</p></a></li>
<li><a href='#plotNaivePrev'>
<p>Plotting expected</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Risk Prediction on Trees</td>
</tr>
<tr>
<td>Version:</td>
<td>1.21</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-02-22</td>
</tr>
<tr>
<td>Author:</td>
<td>Luke Jostins</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Luke Jostins &lt;luke.jostins@kennedy.ox.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Methods for performing genetic risk
        prediction from genotype data.  You can use it to perform risk
        prediction for individuals, or for families with missing data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Suggests:</td>
<td>kinship2</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-02-22 13:14:51 UTC; lj4</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-02-22 17:15:37</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
</table>
<hr>
<h2 id='getVarExp'>
Calculating liability-scale variance explained by genetic risk variants
</h2><span id='topic+getVarExp'></span><span id='topic+getVarExpSim'></span>

<h3>Description</h3>

<p>These functions calculate the variance explained on the liability scale by the genetic risk variants described by a <code><a href="#topic+MangroveORs">MangroveORs</a></code> object. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getVarExp(ORs, K)
getVarExpSim(ORs,K,iter=1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getVarExp_+3A_ors">ORs</code></td>
<td>

<p>An object of class <code><a href="#topic+MangroveORs">MangroveORs</a></code>	
</p>
</td></tr>
<tr><td><code id="getVarExp_+3A_k">K</code></td>
<td>

<p>The prevalence of the disease.
</p>
</td></tr>
<tr><td><code id="getVarExp_+3A_iter">iter</code></td>
<td>

<p>The number of individuals sampled for variance estimation.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>getVarExp</code> calculates the variance explained analytically, by converting the odds ratios to liability-scale units, and combining them together additively. <code>getVarExpSim</code> simulates genotypes from the allele frequencies, performs risk prediction on them, converts the resulting posteriors to liabilities, and directly measures the variance of the sample.
</p>


<h3>Value</h3>

<p>The proportion of variance explained, as a single numeric value.</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleORs)
getVarExp(exampleORs,0.02)
getVarExpSim(exampleORs,0.02)
</code></pre>

<hr>
<h2 id='Mangrove-internal'>Internal Mangrove objects</h2><span id='topic+mendT'></span>

<h3>Description</h3>

<p>Internal Mangrove objects.</p>


<h3>Details</h3>

<p>These are not to be called by the user.</p>

<hr>
<h2 id='Mangrove-package'>
Mangrove: Risk prediction on trees
</h2><span id='topic+Mangrove-package'></span><span id='topic+Mangrove'></span>

<h3>Description</h3>

<p><code>Mangrove</code> is an R package for performing genetic risk
prediction from genotype data.  You can use it to perform risk prediction for
individuals, or for families with missing data.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> Mangrove</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.21</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2017-02-22</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>For a detailed example of how to use <code>Mangrove</code>, see <code>vignette("Mangrove")</code>.
</p>


<h3>Author(s)</h3>

<p>Luke Jostins &lt;luke.jostins@kennedy.ox.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleORs)
data(famped)
tree &lt;- initialiseTree()
tree$addPed(famped,exampleORs)
sam &lt;- tree$getPrevs(exampleORs,K=0.02)
plotNaivePrev(famped,0.02)
plot(sam)

</code></pre>

<hr>
<h2 id='MangroveBetas'>
The MangroveBetas class
</h2><span id='topic+MangroveBetas'></span><span id='topic+readBetas'></span><span id='topic+print.MangroveBetas'></span><span id='topic+summary.MangroveBetas'></span><span id='topic+plot.MangroveBetas'></span>

<h3>Description</h3>

<p>Functions to read in and use MangroveBetas objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readBetas(betafile,header=TRUE)

## S3 method for class 'MangroveBetas'
print(x, ...)
## S3 method for class 'MangroveBetas'
summary(object, ...)
## S3 method for class 'MangroveBetas'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangroveBetas_+3A_betafile">betafile</code></td>
<td>

<p>The location of a beta file to read in
</p>
</td></tr>
<tr><td><code id="MangroveBetas_+3A_header">header</code></td>
<td>

<p>Logical. Indicates whether the betafile above has a header.
</p>
</td></tr>
<tr><td><code id="MangroveBetas_+3A_x">x</code></td>
<td>

<p>A <code>MangroveBetas</code> object
</p>
</td></tr>
<tr><td><code id="MangroveBetas_+3A_object">object</code></td>
<td>

<p>A <code>MangroveBetas</code> object
</p>
</td></tr>
<tr><td><code id="MangroveBetas_+3A_...">...</code></td>
<td>

<p>Additional arguments to be passed to downstream methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A <code>MangroveBetas</code> object is used to hold the risk alleles, beta values and frequencies for a set of genetic variants used to predict a quantitative trait. You can read a beta file from disk using <code>readBetas</code>.
</p>
<p>The format of the beta file is a text file with 4 columns. The first should be the variant ID, the second the risk allele, the third the additive beta value and the fourth the frequency. 
</p>
<p>The <code>print</code> method shows the data that the object holds. The <code>summary</code> method gives some measures of how predictive the variants are, including the variance explained by the variants in the object. The <code>plot</code> method show the cumulative variance explained as the variants are added in one-at-a-time (in order of most-to-least variance explained). 
</p>
<p>See <code>vignette("Mangrove")</code> for an example of a beta file, and usage of a beta object.
</p>


<h3>Value</h3>

<p>An object of class <code>MangroveBetas</code> for <code>readBetas</code>. Nothing is returned for the methods.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveContPreds">MangroveContPreds</a></code> for performing quantitative trait prediction using this object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleBetas)
summary(exampleBetas)
plot(exampleBetas)
</code></pre>

<hr>
<h2 id='MangroveContPreds'>
The MangroveContPreds class
</h2><span id='topic+MangroveContPreds'></span><span id='topic+calcBetas'></span><span id='topic+applyBetas'></span><span id='topic+print.MangroveContPreds'></span><span id='topic+summary.MangroveContPreds'></span><span id='topic+plot.MangroveContPreds'></span>

<h3>Description</h3>

<p>Methods to generate, summarise and apply MangroveContPreds objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcBetas(ped,betas)
applyBetas(contpred,mu,sigma)

## S3 method for class 'MangroveContPreds'
print(x, ...)
## S3 method for class 'MangroveContPreds'
summary(object, ...)
## S3 method for class 'MangroveContPreds'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangroveContPreds_+3A_ped">ped</code></td>
<td>

<p>A <code><a href="#topic+MangrovePed">MangrovePed</a></code> object
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_betas">betas</code></td>
<td>

<p>A <code><a href="#topic+MangroveBetas">MangroveBetas</a></code> object
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_contpred">contpred</code></td>
<td>

<p>A <code>MangroveContPreds</code> object
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_object">object</code></td>
<td>

<p>A <code>MangroveContPreds</code> object
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_x">x</code></td>
<td>

<p>A <code>MangroveContPreds</code> object
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_mu">mu</code></td>
<td>

<p>The population mean of the quantitative trait being predicted
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_sigma">sigma</code></td>
<td>

<p>The population standard deviation of the quantitative trait being predicted
</p>
</td></tr>
<tr><td><code id="MangroveContPreds_+3A_...">...</code></td>
<td>

<p>Arguments to be passed on to downstream methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>MangroveContPreds</code> objects are used to hold quantitative (or continuous) trait predictions for a number of individuals. This object contains the beta value for the trait (i.e. the person's value of the trait on a scale with the population mean being zero and the population standard deviation being one). You can generate per-individual beta values from a pedigree object and some per-variant beta values using <code>calcBetas</code>, and you can transform these to quantitative trait predictions given a population mean and standard deviation using <code>applyBetas</code>. 
</p>
<p>The <code>print</code> method writes the beta value for each individual to screen. The <code>summary</code> method gives some quantile information, and the <code>plot</code> method plots a histogram of predicted values, which should be approximately normal.
</p>


<h3>Value</h3>

<p><code>calcBetas</code> returns a <code>MangroveContPreds</code> object, <code>applyBetas</code> returns a numeric vector containing posterior probabilities. The methods do not return anything.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveRiskPreds">MangroveRiskPreds</a></code> for doing risk prediction for binary traits.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(contped)
data(exampleBetas)
predbetas &lt;- calcBetas(contped,exampleBetas)

summary(predbetas)
plot(predbetas)

contpreds &lt;- applyBetas(predbetas,162,6.4)

</code></pre>

<hr>
<h2 id='MangroveExamples'>
Example data for the Mangrove vignette
</h2><span id='topic+MangroveExamples'></span><span id='topic+ccped'></span><span id='topic+contped'></span><span id='topic+famped'></span><span id='topic+exampleORs'></span><span id='topic+exampleBetas'></span>

<h3>Description</h3>

<p>Examples of the <code><a href="#topic+MangroveORs">MangroveORs</a></code>, <code><a href="#topic+MangroveBetas">MangroveBetas</a></code> and <code><a href="#topic+MangrovePed">MangrovePed</a></code> classes from the <code><a href="#topic+Mangrove">Mangrove</a></code> package. To be used in combination with the Mangrove vignette.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(ccped)
data(famped)
data(exampleORs)
data(contped)
data(exampleBetas)</code></pre>


<h3>Details</h3>

<p>See vignette(&quot;Mangrove&quot;) for details on these example objects.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ccped)
summary(ccped)

data(contped)
summary(contped)

data(famped)
summary(famped)

data(exampleORs)
summary(exampleORs)

data(exampleBetas)
summary(exampleBetas)
</code></pre>

<hr>
<h2 id='MangroveORs'>
The MangroveORs class
</h2><span id='topic+MangroveORs'></span><span id='topic+readORs'></span><span id='topic+print.MangroveORs'></span><span id='topic+summary.MangroveORs'></span><span id='topic+plot.MangroveORs'></span>

<h3>Description</h3>

<p>Functions to read in and use MangroveORs objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>readORs(ORfile,header=TRUE)

## S3 method for class 'MangroveORs'
print(x, ...)
## S3 method for class 'MangroveORs'
summary(object, K = NULL, ...)
## S3 method for class 'MangroveORs'
plot(x, K = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangroveORs_+3A_orfile">ORfile</code></td>
<td>

<p>The location of an odds ratio file to read in
</p>
</td></tr>
<tr><td><code id="MangroveORs_+3A_header">header</code></td>
<td>

<p>Logical. Indicates whether the ORfile above has a header.
</p>
</td></tr>
<tr><td><code id="MangroveORs_+3A_x">x</code></td>
<td>

<p>A <code>MangroveORs</code> object
</p>
</td></tr>
<tr><td><code id="MangroveORs_+3A_object">object</code></td>
<td>

<p>A <code>MangroveORs</code> object
</p>
</td></tr>
<tr><td><code id="MangroveORs_+3A_k">K</code></td>
<td>

<p>The prevalence of the disease that the odds ratios predict. If NULL, data is returned for a few example prevalences.
</p>
</td></tr>
<tr><td><code id="MangroveORs_+3A_...">...</code></td>
<td>

<p>Additional arguments to be passed to downstream methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A <code>MangroveORs</code> object is used to hold the risk alleles, odds ratios and frequencies for a set of genetic variants used to predict disease. You can read an odds ratio file from disk using <code>readORs</code>.
</p>
<p>The format of the odds ratio file is a text file with 4 or 5 columns. If the file has 4 columns, the first should be the variant ID, the second the risk allele, the third the additive odds ratio and the fourth the frequency. If 5 columns exist, column three is the heterozygous odds ratio, column four is the homozygous risk odds ratio, and five is the allele frequency.
</p>
<p>The <code>print</code> method shows the data that the object holds. The <code>summary</code> method gives some measures of how predictive the variants are, including the  variance explained by the variants in the object on the liability scale. The <code>plot</code> method show the cumulative variance explained on the liability scale as the variants are added in one-at-a-time (in order of most-to-least variance explained). 
</p>
<p>See <code>vignette("Mangrove")</code> for an example of an odds ratio file, and usage of an odds ratio object.
</p>


<h3>Value</h3>

<p>An object of class <code>MangroveORs</code> for <code>readORs</code>. Nothing is returned for the methods.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveRiskPreds">MangroveRiskPreds</a></code> for performing risk prediction using this object, and <code><a href="#topic+getVarExp">getVarExp</a></code> for more on assessing variance explained.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(exampleORs)
summary(exampleORs)
plot(exampleORs)
</code></pre>

<hr>
<h2 id='MangrovePed'>
The MangrovePed class
</h2><span id='topic+MangrovePed'></span><span id='topic+readPed'></span><span id='topic+summary.MangrovePed'></span><span id='topic+print.MangrovePed'></span><span id='topic+plot.MangrovePed'></span>

<h3>Description</h3>

<p>Functions to read in and summarise MangrovePed objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
readPed(prefix)

## S3 method for class 'MangrovePed'
summary(object, ...)
## S3 method for class 'MangrovePed'
print(x, ...)
## S3 method for class 'MangrovePed'
plot(x, ...)

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangrovePed_+3A_prefix">prefix</code></td>
<td>

<p>The prefix for the file locations of a pair pedigree and map files, such as produced by the program <code>Plink</code>. Requires prefix.ped and prefix.map to both exist.
</p>
</td></tr>
<tr><td><code id="MangrovePed_+3A_x">x</code></td>
<td>

<p>An object of class <code>MangrovePed</code>
</p>
</td></tr> 
<tr><td><code id="MangrovePed_+3A_object">object</code></td>
<td>

<p>An object of class <code>MangrovePed</code>
</p>
</td></tr> 
<tr><td><code id="MangrovePed_+3A_...">...</code></td>
<td>

<p>Arguments to be passed on to other methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A MangrovePed object holds genotypes and family relationships for a number of individuals. You can read in such data from a Plink pedigree file using <code>readPed</code>.
</p>
<p>You can show the raw pedigree data using the <code>print</code> method and get summary information (such as number of samples, cases and variants) using the <code>summary</code> method. The <code>plot</code> function produces an error.
</p>


<h3>Value</h3>

<p>For <code>readPed</code>, an object of class <code>MangrovePed</code>. For methods, nothing is returned.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveORs">MangroveORs</a></code>, <code><a href="#topic+MangroveTree">MangroveTree</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(famped)
summary(famped)
print(famped)
</code></pre>

<hr>
<h2 id='MangroveRiskPreds'>
The MangroveRiskPreds class
</h2><span id='topic+MangroveRiskPreds'></span><span id='topic+calcORs'></span><span id='topic+applyORs'></span><span id='topic+print.MangroveRiskPreds'></span><span id='topic+summary.MangroveRiskPreds'></span><span id='topic+plot.MangroveRiskPreds'></span>

<h3>Description</h3>

<p>Methods to generate, summarise and apply MangroveRiskPreds objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcORs(ped,ORs)
applyORs(riskpred,K)

## S3 method for class 'MangroveRiskPreds'
print(x, ...)
## S3 method for class 'MangroveRiskPreds'
summary(object, ...)
## S3 method for class 'MangroveRiskPreds'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangroveRiskPreds_+3A_ped">ped</code></td>
<td>

<p>A <code><a href="#topic+MangrovePed">MangrovePed</a></code> object
</p>
</td></tr>
<tr><td><code id="MangroveRiskPreds_+3A_ors">ORs</code></td>
<td>

<p>A <code><a href="#topic+MangroveORs">MangroveORs</a></code> object
</p>
</td></tr>
<tr><td><code id="MangroveRiskPreds_+3A_riskpred">riskpred</code></td>
<td>

<p>A <code>MangroveRiskPreds</code> object
</p>
</td></tr>
<tr><td><code id="MangroveRiskPreds_+3A_object">object</code></td>
<td>

<p>A <code>MangroveRiskPreds</code> object
</p>
</td></tr>
<tr><td><code id="MangroveRiskPreds_+3A_x">x</code></td>
<td>

<p>A <code>MangroveRiskPreds</code> object
</p>
</td></tr>
<tr><td><code id="MangroveRiskPreds_+3A_k">K</code></td>
<td>

<p>The prevalence of the disease being predicted.
</p>
</td></tr>
<tr><td><code id="MangroveRiskPreds_+3A_...">...</code></td>
<td>

<p>Arguments to be passed on to downstream methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>MangroveRiskPreds</code> objects are used to hold risk predictions for a number of individuals. This object contains the odds ratio for each individual of developing the disease, relative to the population average. You can generate risk predictions from a pedigree object and some per-variant odds ratios using <code>calcORs</code>, and you can use these to calculate posterior probabilities of developing the disease given a prevalence using <code>applyORs</code>.
</p>
<p>The <code>print</code> method writes the odds ratio for each individual to screen. The <code>summary</code> method gives some quantile information, and the <code>plot</code> method prints a histogram of log odds ratios, which should be approximately normal.
</p>


<h3>Value</h3>

<p><code>calcORs</code> returns a <code>MangroveRiskPreds</code> object, <code>applyORs</code> returns a numeric vector containing posterior probabilities. The methods do not return anything.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveTree">MangroveTree</a></code> for doing risk prediction for a whole family. <code><a href="#topic+MangroveContPreds">MangroveContPreds</a></code> for doing quantitative trait prediction.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(ccped)
data(exampleORs)
ccors &lt;- calcORs(ccped,exampleORs)

summary(ccors)
plot(ccors)

ccposts &lt;- applyORs(ccors,0.02)

</code></pre>

<hr>
<h2 id='MangroveSample'>
The MangroveSample class
</h2><span id='topic+MangroveSample'></span><span id='topic+print.MangroveSample'></span><span id='topic+summary.MangroveSample'></span><span id='topic+plot.MangroveSample'></span>

<h3>Description</h3>

<p>Functions for manipulating MangroveSample objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MangroveSample'
print(x,...)
## S3 method for class 'MangroveSample'
summary(object, ...)
## S3 method for class 'MangroveSample'
plot(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangroveSample_+3A_x">x</code></td>
<td>

<p>A <code>MangroveSample</code> object.
</p>
</td></tr>
<tr><td><code id="MangroveSample_+3A_object">object</code></td>
<td>

<p>A <code>MangroveSample</code> object.
</p>
</td></tr>
<tr><td><code id="MangroveSample_+3A_...">...</code></td>
<td>

<p>Further arguments to be passed on to downstream methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MangroveSample objects are produced from a <code><a href="#topic+MangroveTree">MangroveTree</a></code> object via a call to <code>tree$getPrevs</code>. They contain samples from the posterior distribution of number of affecteds in a family conditional on that family's genotypes.
</p>
<p>The <code>print</code> method writes all the samples to screen. You can plot the distribution of number of affecteds using the <code>plot</code> method (which calls <code><a href="#topic+plotSampledPrev">plotSampledPrev</a></code>), and perform a significance test for whether the observed number of affecteds is greater than would be expected using the <code>summary</code> method. 
</p>
<p>You can see a detailed example of how these methods are applied by calling <code>vignette("Mangrove")</code>
</p>


<h3>Value</h3>

<p>None of the methods return anything.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveTree">MangroveTree</a></code> to generate <code>MangroveSample</code> objects. <code><a href="#topic+plotSampledPrev">plotSampledPrev</a></code> for more on plotting expected distributions.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(famped)
data(exampleORs)
tree &lt;- initialiseTree()
tree$addPed(famped,exampleORs)
sam &lt;- tree$getPrevs(exampleORs,0.02)

summary(sam)
plot(sam)

</code></pre>

<hr>
<h2 id='MangroveTree'>
The MangroveTree class
</h2><span id='topic+MangroveTree'></span><span id='topic+initialiseTree'></span><span id='topic+getPrevs'></span><span id='topic+addPed'></span><span id='topic+summary.MangroveTree'></span><span id='topic+print.MangroveTree'></span><span id='topic+plot.MangroveTree'></span>

<h3>Description</h3>

<p>Functions to create, populate, summarise and utilise MangroveTree objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>initialiseTree()

# Member functions: 
# tree$addPed(ped,ORs)
# tree$getPrevs(ORs = NULL,K = NULL,overwrite=FALSE,iter=1000)

## S3 method for class 'MangroveTree'
print(x,...)
## S3 method for class 'MangroveTree'
summary(object,...)
## S3 method for class 'MangroveTree'
plot(x,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MangroveTree_+3A_tree">tree</code></td>
<td>

<p>An object of class <code><a href="#topic+MangroveTree">MangroveTree</a></code>
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_x">x</code></td>
<td>

<p>An object of class <code><a href="#topic+MangroveTree">MangroveTree</a></code>
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_object">object</code></td>
<td>

<p>An object of class <code><a href="#topic+MangroveTree">MangroveTree</a></code>
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_ped">ped</code></td>
<td>

<p>An object of class \code<a href="#topic+MangrovePed">MangrovePed</a>
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_ors">ORs</code></td>
<td>

<p>An object of class <code><a href="#topic+MangroveORs">MangroveORs</a></code>
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_k">K</code></td>
<td>

<p>The prevalence of the disease (between 0 and 1, or NULL if not known)
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_overwrite">overwrite</code></td>
<td>

<p>Logical. If set to FALSE, and sampling has already been performed, then existing samples will be used. If TRUE, sampling is performed again.
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_iter">iter</code></td>
<td>

<p>Number of samples to draw from the posterior distribution of number of affecteds.
</p>
</td></tr>
<tr><td><code id="MangroveTree_+3A_...">...</code></td>
<td>

<p>Arguments to be passed on to other methods.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The MangroveTree class is used for holding family trees, and for performing risk prediction on them. The <code>initialiseTree</code> function creates an empty tree, and the <code>tree$addPed</code> populates it from genetic data. You can then sample from the posterior distribution of number of affecteds using the <code>tree$getPrevs</code> function.
</p>
<p>The <code>print</code> method writes a text version of the tree to screen, and the <code>summary</code> method writes some basic information about what the tree contains, along with what calculations have and have not been performed on it. The <code>plot</code> method is not implemented, and throws an error message.
</p>
<p>To see an example of the use of this class, use <code>vignette("Mangrove")</code>.
</p>


<h3>Value</h3>

<p>For <code>initialiseTree</code>, an empty object of class <code>MangroveTree</code>. For the <code>tree$getPrevs</code> class function, an object of class <code><a href="#topic+MangroveSample">MangroveSample</a></code>. 
</p>
<p>For all other functions, nothing is returned.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangrovePed">MangrovePed</a></code>, <code><a href="#topic+MangroveORs">MangroveORs</a></code>, <code><a href="#topic+MangroveSample">MangroveSample</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(famped)
data(exampleORs)

tree &lt;- initialiseTree()
tree$addPed(famped,exampleORs)

print(tree)
summary(tree)

sam &lt;- tree$getPrevs(exampleORs,K=0.02)
summary(sam)
</code></pre>

<hr>
<h2 id='plotNaivePrev'>
Plotting expected
</h2><span id='topic+plotNaivePrev'></span><span id='topic+plotSampledPrev'></span>

<h3>Description</h3>

<p>Two functions for plotting the expected distribution of affecteds in a family, either assuming no genetic risk factors, or using sampled from a custom distribution of affecteds (such as produced by <code><a href="#topic+MangroveTree">MangroveTree</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotNaivePrev(ped,K,maxN = NULL,...)
plotSampledPrev(samples, obs_prev, exp_prev, maxN = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotNaivePrev_+3A_ped">ped</code></td>
<td>

<p>A <code><a href="#topic+MangrovePed">MangrovePed</a></code> object.
</p>
</td></tr>
<tr><td><code id="plotNaivePrev_+3A_k">K</code></td>
<td>

<p>The prevalence of the disease.  
</p>
</td></tr>
<tr><td><code id="plotNaivePrev_+3A_samples">samples</code></td>
<td>

<p>Samples from the distribution of number of affecteds.
</p>
</td></tr>
<tr><td><code id="plotNaivePrev_+3A_obs_prev">obs_prev</code></td>
<td>

<p>The number of affecteds actually observed.
</p>
</td></tr>
<tr><td><code id="plotNaivePrev_+3A_exp_prev">exp_prev</code></td>
<td>

<p>The expected number of affecteds under a naive model.
</p>
</td></tr>
<tr><td><code id="plotNaivePrev_+3A_maxn">maxN</code></td>
<td>

<p>The maximum number of affecteds to be shown on the graph. If <code>NULL</code>, an appropriate maximum is selected from the data.
</p>
</td></tr>
<tr><td><code id="plotNaivePrev_+3A_...">...</code></td>
<td>

<p>Additional arguments to <code>plot</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions are used to assess how &quot;unusual&quot; a family is in terms of the number of affected individuals it contains. <code>plotNaivePrev</code> plots the distribution of affected individuals in the family assuming no genetic risk factors (i.e. under a binomial model). <code>plotSampledPrev</code> is more general, and takes in a set of samples from the expected distribution. 
</p>
<p>Note that <code>plotSampledPrev</code> is called by <code><a href="#topic+plot.MangroveSample">plot.MangroveSample</a></code> to plot the results of a MangroveTree sampling. It will be easier to use the <code>print.MangroveSample</code> method rather than using <code>plotSampledPrev</code> under most circumstances.
</p>


<h3>Value</h3>

<p>Neither function returns anything.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MangroveSample">MangroveSample</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(famped)

plotNaivePrev(famped,0.02) # is this unexpected for a 2% disease?
plotNaivePrev(famped,0.04) # how about for a 4% disease?

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
