<!DOCTYPE html><html lang="en"><head><title>Help for package manhattanly</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {manhattanly}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#HapMap'><p>Subset of HapMap data with simulated GWAS results</p></a></li>
<li><a href='#manhattanly'><p>Creates a plotly manhattan plot</p></a></li>
<li><a href='#manhattanr'><p>Creates a manhattanr object</p></a></li>
<li><a href='#qqly'><p>Creates a plotly Q-Q plot</p></a></li>
<li><a href='#qqr'><p>Creates a qq object</p></a></li>
<li><a href='#significantSNP'><p>Character vector of SNPs to highlight</p></a></li>
<li><a href='#volcanoly'><p>Creates a plotly volcano plot</p></a></li>
<li><a href='#volcanor'><p>Creates a volcano object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Interactive Q-Q and Manhattan Plots Using 'plotly.js'</td>
</tr>
<tr>
<td>Version:</td>
<td>0.3.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-04-06</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Description:</td>
<td>Create interactive manhattan, Q-Q and volcano plots that are usable from the R console, 
    in 'Dash' apps, in the 'RStudio' viewer pane, in 'R Markdown' documents, and in 'Shiny' apps.
    Hover the mouse pointer over a point to show details or drag a rectangle to
    zoom. A manhattan plot is a popular graphical method for visualizing results
    from high-dimensional data analysis such as a (epi)genome wide association study
    (GWAS or EWAS), in which p-values, Z-scores, test statistics are plotted on a scatter
    plot against their genomic position. Manhattan plots are used for visualizing
    potential regions of interest in the genome that are associated with a phenotype.
    Interactive manhattan plots allow the inspection of specific value (e.g. rs number or
    gene name) by hovering the mouse over a cell, as well as zooming into a region of the
    genome (e.g. a chromosome) by dragging a rectangle around the relevant area.
    This work is based on the 'qqman' package and the 'plotly.js'
    engine. It produces similar manhattan and Q-Q plots as the 'manhattan' and 'qq'
    functions in the 'qqman' package, with the advantage of including extra annotation 
    information and interactive web-based visualizations directly from R. 
    Once uploaded to a 'plotly' account, 'plotly' graphs (and the data behind them) 
    can be viewed and modified in a web browser.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, magrittr, plotly, ggplot2</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), knitr, rmarkdown, covr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sahirbhatnagar/manhattanly/">https://github.com/sahirbhatnagar/manhattanly/</a>,
<a href="https://sahirbhatnagar.com/manhattanly/">https://sahirbhatnagar.com/manhattanly/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sahirbhatnagar/manhattanly/issues">https://github.com/sahirbhatnagar/manhattanly/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-04-26 01:48:15 UTC; sahir</td>
</tr>
<tr>
<td>Author:</td>
<td>Sahir Bhatnagar [aut, cre] (http://sahirbhatnagar.com/)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sahir Bhatnagar &lt;sahir.bhatnagar@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-04-26 06:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='HapMap'>Subset of HapMap data with simulated GWAS results</h2><span id='topic+HapMap'></span>

<h3>Description</h3>

<p>A dataset containing a subset of the draft release 2 for genome-wide SNP
genotyping in DNA samples from 11 human populations (sometimes referred to as
the &quot;HapMap 3&quot; samples). Only the PLINK .map file was used. Approximately
2.5% of the SNPs in each chromosome were retained. The p-values, zscores,
and effectsizes were simulated using random distributions in R. Annotation
information (nearest gene and distance to nearest gene) was obtained from the
UCSC genome annotation database for the Mar. 2006 GenBank freeze assembled by
NCBI (hg18, Build 36.1)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HapMap
</code></pre>


<h3>Format</h3>

<p>A data frame with 14412 rows and 8 variables: </p>

<dl>
<dt>CHR</dt><dd><p>chromosome number. Autosomes coded 1 through 22, and 23 is the X
chromosome (<code>integer</code>)</p>
</dd> <dt>BP</dt><dd><p>genomic base-pair position
(<code>integer</code>)</p>
</dd> <dt>P</dt><dd><p>p-value (<code>numeric</code>)</p>
</dd> <dt>SNP</dt><dd><p>rs# or snp
identifier (<code>character</code>)</p>
</dd> <dt>ZSCORE</dt><dd><p>z-score (<code>numeric</code>)</p>
</dd>
<dt>EFFECTSIZE</dt><dd><p>effect size (<code>numeric</code>)</p>
</dd> <dt>GENE</dt><dd><p>nearest gene to
the SNP (<code>character</code>)</p>
</dd> <dt>DISTANCE</dt><dd><p>distance between the SNP and
GENE. if <code>DISTANCE=0</code> then the SNP is located in the GENE
(<code>integer</code>)</p>
</dd> </dl>



<h3>Source</h3>

<p><a href="ftp://ftp.ncbi.nlm.nih.gov/hapmap/genotypes/2009-01_phaseIII/plink_format/">ftp://ftp.ncbi.nlm.nih.gov/hapmap/genotypes/2009-01_phaseIII/plink_format/</a>
</p>
<p><a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/">http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/</a>
</p>

<hr>
<h2 id='manhattanly'>Creates a plotly manhattan plot</h2><span id='topic+manhattanly'></span><span id='topic+manhattanly.default'></span><span id='topic+manhattanly.manhattanr'></span>

<h3>Description</h3>

<p>Creates an interactive manhattan plot with multiple annotation options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattanly(
  x,
  ...,
  col = c("#969696", "#252525"),
  point_size = 5,
  labelChr = NULL,
  suggestiveline = -log10(1e-05),
  suggestiveline_color = "blue",
  suggestiveline_width = 1,
  genomewideline = -log10(5e-08),
  genomewideline_color = "red",
  genomewideline_width = 1,
  highlight = NULL,
  highlight_color = "#00FF00",
  showlegend = FALSE,
  showgrid = FALSE,
  xlab = NULL,
  ylab = "-log10(p)",
  title = "Manhattan Plot"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="manhattanly_+3A_x">x</code></td>
<td>
<p>Can be an object of class <code>manhattanr</code> produced by the
<code><a href="#topic+manhattanr">manhattanr</a></code> function or a <code>data.frame</code> which must contain
at least the following three columns: </p>
 <ul>
<li><p>the chromosome
number </p>
</li>
<li><p>genomic base-pair position </p>
</li>
<li><p>a numeric quantity to plot
such as a p-value or zscore </p>
</li></ul>
</td></tr>
<tr><td><code id="manhattanly_+3A_...">...</code></td>
<td>
<p>other parameters passed to <code><a href="#topic+manhattanr">manhattanr</a></code></p>
</td></tr>
<tr><td><code id="manhattanly_+3A_col">col</code></td>
<td>
<p>A character vector indicating the colors of each chromosome. If
the number of colors specified is less than the number of unique
chromosomes, then the elements will be recycled. Can be
<a href="https://www.rapidtables.com/web/color/RGB_Color.html">Hex Codes</a> as
well.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_point_size">point_size</code></td>
<td>
<p>A <code>numeric</code> indicating the size of the points on the
plot. Default is 5</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_labelchr">labelChr</code></td>
<td>
<p>A character vector equal to the number of chromosomes
specifying the chromosome labels (e.g., <code>c(1:22, "X", "Y", "MT")</code>).
Default is <code>NULL</code>, meaning that the actual chromosome numbers will be
used.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_suggestiveline">suggestiveline</code></td>
<td>
<p>Where to draw a &quot;suggestive&quot; line. Default is
<code>-log10(1e-5)</code>. Set to <code>FALSE</code> to disable.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_suggestiveline_color">suggestiveline_color</code></td>
<td>
<p>color of &quot;suggestive&quot; line. Only used if
<code>suggestiveline</code> is not set to <code>FALSE</code>. Default is <code>"blue"</code>.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_suggestiveline_width">suggestiveline_width</code></td>
<td>
<p>Width of <code>suggestiveline</code>. Default is 1.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_genomewideline">genomewideline</code></td>
<td>
<p>Where to draw a &quot;genome-wide sigificant&quot; line. Default
<code>-log10(5e-8)</code>. Set to <code>FALSE</code> to disable.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_genomewideline_color">genomewideline_color</code></td>
<td>
<p>color of &quot;genome-wide sigificant&quot; line. Only used
if <code>genomewideline</code> is not set to <code>FALSE</code>. Default is
<code>"red"</code>.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_genomewideline_width">genomewideline_width</code></td>
<td>
<p>Width of <code>genomewideline</code>. Default is 1.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of SNPs in your dataset to highlight.
These SNPs should all be in your dataset. Default is <code>NULL</code> which
means that nothing is highlighted.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color used to highlight points. Only used if
<code>highlight</code> argument has been specified</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_showlegend">showlegend</code></td>
<td>
<p>Should a legend be shown. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_showgrid">showgrid</code></td>
<td>
<p>Should gridlines be shown. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_xlab">xlab</code></td>
<td>
<p>X-axis label. Default is <code>NULL</code> which means that the label
is automatically determined by the <code><a href="#topic+manhattanr">manhattanr</a></code> function.
Specify here to overwrite the default.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_ylab">ylab</code></td>
<td>
<p>Y-axis label. Default is <code>"-log10(p)"</code>.</p>
</td></tr>
<tr><td><code id="manhattanly_+3A_title">title</code></td>
<td>
<p>Title of the plot. Default is <code>"Manhattan Plot"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An interactive manhattan plot.
</p>


<h3>Note</h3>

<p>This package is inspired by the
<a href="https://github.com/stephenturner/qqman"><code>qqman</code></a> package. This
package provides additional annotation options and builds on the
<code>plotly</code> <code>d3.js</code> engine. These plots can be included in
Dash apps, Shiny apps, Rmarkdown documents or embedded in websites using
simple HTML code.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+manhattanr">manhattanr</a></code>, <code><a href="#topic+HapMap">HapMap</a></code>,
<code><a href="#topic+significantSNP">significantSNP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(manhattanly)
manhattanly(HapMap)

# highlight SNPs of interest
# 'signigicantSNP' is a character vector of SNPs included in this package
manhattanly(HapMap, snp = "SNP", highlight = significantSNP)

## End(Not run)
</code></pre>

<hr>
<h2 id='manhattanr'>Creates a manhattanr object</h2><span id='topic+manhattanr'></span>

<h3>Description</h3>

<p>An object of class manhattanr includes all the needed information for
producing a manhattan plot. The goal is to seperate the pre-processing of the
manhattan plot elements from the graphical rendaring of the object, which
could be done using any graphical device including
<code><a href="plotly.html#topic+plot_ly">plot_ly</a></code> and <code><a href="graphics.html#topic+plot">plot</a></code> in base
<code>R</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattanr(
  x,
  chr = "CHR",
  bp = "BP",
  p = "P",
  snp,
  gene,
  annotation1,
  annotation2,
  logp = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="manhattanr_+3A_x">x</code></td>
<td>
<p>A <code>data.frame</code> which must contain at least the following three
columns: </p>
 <ul>
<li><p>the chromosome number </p>
</li>
<li><p>genomic base-pair
position </p>
</li>
<li><p>a numeric quantity to plot such as a p-value or zscore </p>
</li></ul>
</td></tr>
<tr><td><code id="manhattanr_+3A_chr">chr</code></td>
<td>
<p>A string denoting the column name for the chromosome. Default is
<code>chr = "CHR"</code>. This column must be <code>numeric</code> or <code>integer</code>.
Minimum number of chromosomes required is 1. If you have X, Y, or MT
chromosomes, be sure to renumber these 23, 24, 25, etc.</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_bp">bp</code></td>
<td>
<p>A string denoting the column name for the chromosomal position.
Default is <code>bp = "BP"</code>. This column must be <code>numeric</code> or
<code>integer</code>.</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_p">p</code></td>
<td>
<p>A string denoting the column name for the numeric quantity to be
plotted on the y-axis. Default is <code>p = "P"</code>. This column must be
<code>numeric</code> or <code>integer</code>. This does not have to be a p-value. It
can be any numeric quantity such as peak heights, bayes factors, test
statistics. If it is not a p-value, make sure to set <code>logp = FALSE</code>.</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_snp">snp</code></td>
<td>
<p>A string denoting the column name for the SNP names (e.g. rs
number). More generally, this column could be anything that identifies each
point being plotted. For example, in an Epigenomewide association study
(EWAS) this could be the probe name or cg number. This column should be a
<code>character</code>. This argument is optional, however it is necessary to
specify if you want to highlight points on the plot using the
<code>highlight</code> argument in the <code><a href="#topic+manhattanly">manhattanly</a></code> function</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_gene">gene</code></td>
<td>
<p>A string denoting the column name for the GENE names. This column
could be a <code>character</code> or <code>numeric</code>. More generally this could be
any annotation information that you want to include in the plot. This
argument is optional.</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_annotation1">annotation1</code></td>
<td>
<p>A string denoting the column name for an annotation. This
column could be a <code>character</code> or <code>numeric</code>. This could be any
annotation information that you want to include in the plot (e.g. zscore,
effect size, minor allele frequency). This argument is optional.</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_annotation2">annotation2</code></td>
<td>
<p>A string denoting the column name for an annotation. This
column could be a <code>character</code> or <code>numeric</code>. This could be any
annotation information that you want to include in the plot (e.g. zscore,
effect size, minor allele frequency). This argument is optional.</p>
</td></tr>
<tr><td><code id="manhattanr_+3A_logp">logp</code></td>
<td>
<p>If TRUE, the -log10 of the p-value is plotted. It isn't very
useful to plot raw p-values, but plotting the raw value could be useful for
other genome-wide plots, for example, peak heights, bayes factors, test
statistics, other &quot;scores&quot; etc.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>list</code> object of class <code>manhattanr</code> with the following
elements </p>
 <dl>
<dt>data</dt><dd><p>processed data to be used for plotting</p>
</dd>
<dt>xlabel</dt><dd><p>The label of the x-axis which is determined by the number of
chromosomes present in the data</p>
</dd> <dt>ticks</dt><dd><p>the coordinates on the x-axis
of where the tick marks should be placed</p>
</dd> <dt>labs</dt><dd><p>the labels for each
tick. This defaults to the chromosome number but can be changed in the
<code><a href="#topic+manhattanly">manhattanly</a></code> function </p>
</dd> <dt>nchr</dt><dd><p>the number of unique
chromosomes present in the data</p>
</dd> <dt>pName, snpName, geneName,
annotation1Name, annotation2Name</dt><dd><p>The names of the columns corresponding to
the data provided. This information is used for annotating the plot in the
<code><a href="#topic+manhattanly">manhattanly</a></code> function </p>
</dd> </dl>



<h3>Source</h3>

<p>The pre-processing is mostly the same as the
manhattan function from the
<a href="https://github.com/stephenturner/qqman"><code>qqman</code></a> package
</p>


<h3>See Also</h3>

<p><code><a href="#topic+manhattanly">manhattanly</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># HapMap dataset included in this package already has columns named P, CHR and BP
library(manhattanly)
DT &lt;- manhattanr(HapMap)
class(DT)
head(DT[["data"]])

# include snp and gene information
DT2 &lt;- manhattanr(HapMap, snp = "SNP", gene = "GENE")
head(DT2[["data"]])
</code></pre>

<hr>
<h2 id='qqly'>Creates a plotly Q-Q plot</h2><span id='topic+qqly'></span><span id='topic+qqly.default'></span><span id='topic+qqly.qqr'></span>

<h3>Description</h3>

<p>Creates an interactive Q-Q plot with multiple annotation options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqly(
  x,
  ...,
  col = "#252525",
  size = 5,
  type = 20,
  abline_col = "red",
  abline_size = 1,
  abline_type = "solid",
  highlight = NULL,
  highlight_color = "#00FF00",
  xlab = "Expected -log10(p)",
  ylab = "Observed -log10(p)",
  title = "Q-Q Plot"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qqly_+3A_x">x</code></td>
<td>
<p>Can be an object of class <code>qqr</code> produced by the
<code><a href="#topic+qqr">qqr</a></code> function or a <code>data.frame</code> which must contain at
least the following column: </p>
 <ul>
<li><p>a p-value, must be numeric </p>
</li></ul>
</td></tr>
<tr><td><code id="qqly_+3A_...">...</code></td>
<td>
<p>other parameters passed to <code><a href="#topic+qqr">qqr</a></code></p>
</td></tr>
<tr><td><code id="qqly_+3A_col">col</code></td>
<td>
<p>A <code>character</code> indicating the color of the points. Can
be <a href="https://www.rapidtables.com/web/color/RGB_Color.html">Hex Codes</a> as
well.</p>
</td></tr>
<tr><td><code id="qqly_+3A_size">size</code></td>
<td>
<p>A <code>numeric</code> specifying the size of the points. Default
is 1</p>
</td></tr>
<tr><td><code id="qqly_+3A_type">type</code></td>
<td>
<p>An integer between 0 and 25 specifying the point shape.
Default is 20 (filled circle). Deprecated.</p>
</td></tr>
<tr><td><code id="qqly_+3A_abline_col">abline_col</code></td>
<td>
<p>A <code>character</code> indicating the color of the 45 degree
diagonal line. Can be
<a href="https://www.rapidtables.com/web/color/RGB_Color.html">Hex Codes</a> as
well. Default is <code>"red"</code>.</p>
</td></tr>
<tr><td><code id="qqly_+3A_abline_size">abline_size</code></td>
<td>
<p>A <code>numeric</code> indicating the size of the 45 degree
diagonal line. Default is 0.5.</p>
</td></tr>
<tr><td><code id="qqly_+3A_abline_type">abline_type</code></td>
<td>
<p>Sets the line type of the
45 degree line. Set to a dash type character among &quot;solid&quot;, &quot;dot&quot;,
&quot;dash&quot;, &quot;longdash&quot;, &quot;dashdot&quot;, or &quot;longdashdot&quot;, or a dash length list in
px (eg &quot;5px&quot;,&quot;10px&quot;,&quot;2px&quot;). Can also be a positive numeric value (e.g 5,
10, 2). Default is &quot;dash&quot;. See
<a href="https://plotly.com/r/reference/#layout-shapes">plotly help page on
layouts</a> for complete list and more details</p>
</td></tr>
<tr><td><code id="qqly_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of SNPs in your dataset to highlight.
These SNPs should all be in your dataset. Default is <code>NULL</code> which
means that nothing is highlighted.</p>
</td></tr>
<tr><td><code id="qqly_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color used to highlight points. Only used if
<code>highlight</code> argument has been specified</p>
</td></tr>
<tr><td><code id="qqly_+3A_xlab">xlab</code></td>
<td>
<p>X-axis label. Default is <code>"Expected -log10(p)"</code></p>
</td></tr>
<tr><td><code id="qqly_+3A_ylab">ylab</code></td>
<td>
<p>Y-axis label. Default is <code>"Observed -log10(p)"</code></p>
</td></tr>
<tr><td><code id="qqly_+3A_title">title</code></td>
<td>
<p>Title of the plot. Default is <code>"Q-Q Plot"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An interactive Q-Q plot.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+qqr">qqr</a></code>, <code><a href="#topic+HapMap">HapMap</a></code>,
<code><a href="#topic+significantSNP">significantSNP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
library(manhattanly)
qqly(HapMap)

# highlight SNPs of interest
# 'signigicantSNP' is a character vector of SNPs included in this package
qqly(HapMap, snp = "SNP", highlight = significantSNP)

## End(Not run)

</code></pre>

<hr>
<h2 id='qqr'>Creates a qq object</h2><span id='topic+qqr'></span>

<h3>Description</h3>

<p>An object of class qq includes all the needed information for producing a
quantile-quantile plot of p-values. The goal is to seperate the
pre-processing of the quantile-quantile plot elements from the graphical
rendaring of the object, which could be done using any graphical device
including <code><a href="plotly.html#topic+plot_ly">plot_ly</a></code> and <code><a href="graphics.html#topic+plot">plot</a></code> in
base <code>R</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqr(x, p = "P", snp, gene, annotation1, annotation2, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="qqr_+3A_x">x</code></td>
<td>
<p>A <code>data.frame</code> which must contain at least the following
column: </p>
 <ul>
<li><p>a p-value, must be numeric </p>
</li></ul>
</td></tr>
<tr><td><code id="qqr_+3A_p">p</code></td>
<td>
<p>A string denoting the column name for the p-values. Default is
<code>p = "P"</code>. This column must be <code>numeric</code> or <code>integer</code>.
Should not have missing, NA, NaN, or NULL values and should be between 0
and 1.</p>
</td></tr>
<tr><td><code id="qqr_+3A_snp">snp</code></td>
<td>
<p>A string denoting the column name for the SNP names (e.g. rs
number). More generally, this column could be anything that identifies each
point being plotted. For example, in an Epigenomewide association study
(EWAS) this could be the probe name or cg number. This column should be a
<code>character</code>. This argument is optional, however it is necessary to
specify if you want to highlight points on the plot using the
<code>highlight</code> argument in the <code><a href="#topic+qqly">qqly</a></code> function</p>
</td></tr>
<tr><td><code id="qqr_+3A_gene">gene</code></td>
<td>
<p>A string denoting the column name for the GENE names. This column
could be a <code>character</code> or <code>numeric</code>. More generally this could be
any annotation information that you want to include in the plot. This
argument is optional.</p>
</td></tr>
<tr><td><code id="qqr_+3A_annotation1">annotation1</code></td>
<td>
<p>A string denoting the column name for an annotation. This
column could be a <code>character</code> or <code>numeric</code>. This could be any
annotation information that you want to include in the plot (e.g. zscore,
effect size, minor allele frequency). This argument is optional.</p>
</td></tr>
<tr><td><code id="qqr_+3A_annotation2">annotation2</code></td>
<td>
<p>A string denoting the column name for an annotation. This
column could be a <code>character</code> or <code>numeric</code>. This could be any
annotation information that you want to include in the plot (e.g. zscore,
effect size, minor allele frequency). This argument is optional.</p>
</td></tr>
<tr><td><code id="qqr_+3A_...">...</code></td>
<td>
<p>currently ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An list object of class <code>qqr</code> with the following elements
</p>
  <dl>
<dt>data</dt><dd><p>processed data to be used for plotting the Q-Q plot
including the observed and expected p-values on the -log10 scale</p>
</dd>
<dt>pName, snpName, geneName, annotation1Name, annotation2Name</dt><dd><p>The names
of the columns corresponding to the data provided. This information is used
for annotating the plot in the <code><a href="#topic+qqly">qqly</a></code> function </p>
</dd> </dl>



<h3>Note</h3>

<p>This function will return an error if any of the p-values are NA, less
than 0 or greater than 1
</p>


<h3>Source</h3>

<p>The calculation of the expected p-value is taken from the
qq function from the
<a href="https://github.com/stephenturner/qqman"><code>qqman</code></a> package
</p>


<h3>See Also</h3>

<p><code><a href="#topic+qqly">qqly</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(manhattanly)
qqrObj &lt;- qqr(HapMap, snp = "SNP", highlight = significantSNP)
class(qqrObj)
head(qqrObj[["data"]])
</code></pre>

<hr>
<h2 id='significantSNP'>Character vector of SNPs to highlight</h2><span id='topic+significantSNP'></span>

<h3>Description</h3>

<p>SNP rs identifiers from <code><a href="#topic+HapMap">HapMap</a></code> dataset that are significant at <code>p-value
&lt; 1e-6</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>significantSNP
</code></pre>


<h3>Format</h3>

<p>A character vector with 20 elements
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HapMap">HapMap</a></code>
</p>

<hr>
<h2 id='volcanoly'>Creates a plotly volcano plot</h2><span id='topic+volcanoly'></span><span id='topic+volcanoly.default'></span><span id='topic+volcanoly.volcanor'></span>

<h3>Description</h3>

<p>Creates an interactive volcano plot with multiple annotation options
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcanoly(
  x,
  ...,
  col = c("#252525"),
  point_size = 5,
  effect_size_line = c(-1, 1),
  effect_size_line_color = "grey",
  effect_size_line_width = 0.5,
  effect_size_line_type = "dash",
  genomewideline = -log10(1e-05),
  genomewideline_color = "grey",
  genomewideline_width = 0.5,
  genomewideline_type = "dash",
  highlight = NULL,
  highlight_color = "red",
  xlab = NULL,
  ylab = "-log10(p)",
  title = "Volcano Plot"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="volcanoly_+3A_x">x</code></td>
<td>
<p>Can be an object of class <code>volcanor</code> produced by the
<code><a href="#topic+volcanor">volcanor</a></code> function or a <code>data.frame</code> which must contain
at least the following two columns: </p>
 <ul>
<li><p>a p-value, must be
numeric </p>
</li>
<li><p>a measure of the strength of association, typically an odds
ratio, regression coefficient or log fold change. Must be numeric </p>
</li></ul>
</td></tr>
<tr><td><code id="volcanoly_+3A_...">...</code></td>
<td>
<p>other parameters passed to <code><a href="#topic+volcanor">volcanor</a></code></p>
</td></tr>
<tr><td><code id="volcanoly_+3A_col">col</code></td>
<td>
<p>A character of length 1 indicating the color of the points. Only
the first argument will be used if more than one color is supplied. Can be
<a href="https://www.rapidtables.com/web/color/RGB_Color.html">Hex Codes</a> as
well.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_point_size">point_size</code></td>
<td>
<p>A <code>numeric</code> indicating the size of the points on the
plot. Default is 5</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_effect_size_line">effect_size_line</code></td>
<td>
<p>Where to draw a &quot;suggestive&quot; line on the x-axis.
Default is <code>-1 and +1</code>. Must be a vector of length 2. If a longer
vector is supplied, only the first two elements will be used. First element
must be smaller than second element. Set to <code>FALSE</code> to disable.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_effect_size_line_color">effect_size_line_color</code></td>
<td>
<p>color of &quot;suggestive&quot; line. Only used if
<code>effect_size_line</code> is not set to <code>FALSE</code>. Default is
<code>"blue"</code>.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_effect_size_line_width">effect_size_line_width</code></td>
<td>
<p>Width of <code>effect_size_line</code>. Default is 1.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_effect_size_line_type">effect_size_line_type</code></td>
<td>
<p>Sets the line type of the
<code>effect_size_line</code>. Set to a dash type character among &quot;solid&quot;, &quot;dot&quot;,
&quot;dash&quot;, &quot;longdash&quot;, &quot;dashdot&quot;, or &quot;longdashdot&quot;, or a dash length list in
px (eg &quot;5px&quot;,&quot;10px&quot;,&quot;2px&quot;). Can also be a positive numeric value (e.g 5,
10, 2). Default is &quot;dash&quot;. See
<a href="https://plotly.com/r/reference/#layout-shapes">plotly help page on
layouts</a> for complete list and more details</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_genomewideline">genomewideline</code></td>
<td>
<p>Where to draw a &quot;genome-wide sigificant&quot; line. Default
<code>-log10(1e-5)</code>. Set to <code>FALSE</code> to disable. If more than one
element is provided, only the first will be used</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_genomewideline_color">genomewideline_color</code></td>
<td>
<p>color of &quot;genome-wide sigificant&quot; line. Only used
if <code>genomewideline</code> is not set to <code>FALSE</code>. Default is
<code>"red"</code>.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_genomewideline_width">genomewideline_width</code></td>
<td>
<p>Width of <code>genomewideline</code>. Default is 1.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_genomewideline_type">genomewideline_type</code></td>
<td>
<p>Sets the line type of the <code>genomewideline</code>.
Set to a dash type character among &quot;solid&quot;, &quot;dot&quot;, &quot;dash&quot;, &quot;longdash&quot;,
&quot;dashdot&quot;, or &quot;longdashdot&quot;, or a dash length list in px (eg
&quot;5px&quot;,&quot;10px&quot;,&quot;2px&quot;). Can also be a positive numeric value (e.g 5, 10, 2).
Default is &quot;dash&quot;. See
<a href="https://plotly.com/r/reference/#layout-shapes">plotly help page on
layouts</a> for complete list and more details</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_highlight">highlight</code></td>
<td>
<p>A character vector of SNPs in your dataset to highlight.
These SNPs should all be in your dataset. Default is <code>NULL</code> which
means that all points that are both beyond <code>genomewideline</code> and
<code>effect_size_line</code> are highlighted. Set to <code>FALSE</code> if you don't
want any points highlighted.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_highlight_color">highlight_color</code></td>
<td>
<p>Color used to highlight points. Only used if
<code>highlight</code> argument has been specified</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_xlab">xlab</code></td>
<td>
<p>X-axis label. Default is <code>NULL</code> which means that the label
is automatically determined by the <code><a href="#topic+volcanor">volcanor</a></code> function. Specify
here to overwrite the default.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_ylab">ylab</code></td>
<td>
<p>Y-axis label. Default is <code>"-log10(p)"</code>.</p>
</td></tr>
<tr><td><code id="volcanoly_+3A_title">title</code></td>
<td>
<p>Title of the plot. Default is <code>"Volcano Plot"</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An interactive volcano plot.
</p>


<h3>Note</h3>

<p>This package provides additional annotation options and builds on the
<code>plotly</code> <code>d3.js</code> engine. These plots can be included in
Shiny apps, Dash apps, Rmarkdown documents or embeded in websites using
simple HTML code.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+volcanor">volcanor</a></code>, <code><a href="#topic+HapMap">HapMap</a></code>,
<code><a href="#topic+significantSNP">significantSNP</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>volcanorObj &lt;- volcanor(HapMap,
  p = "P",
  effect_size = "EFFECTSIZE",
  snp = "SNP",
  gene = "GENE"
)
class(volcanorObj)
head(volcanorObj$data)
</code></pre>

<hr>
<h2 id='volcanor'>Creates a volcano object</h2><span id='topic+volcanor'></span>

<h3>Description</h3>

<p>An object of class volcano includes all the needed information for producing
a volcano plot of p-values against effect sizes or fold-changes. The goal is
to seperate the pre-processing of the volcano plot elements from the
graphical rendaring of the object, which could be done using any graphical
device including <code><a href="plotly.html#topic+plot_ly">plot_ly</a></code> and
<code><a href="graphics.html#topic+plot">plot</a></code> in base <code>R</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>volcanor(
  x,
  p = "P",
  effect_size = "EFFECTSIZE",
  snp,
  gene,
  annotation1,
  annotation2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="volcanor_+3A_x">x</code></td>
<td>
<p>A <code>data.frame</code> which must contain at least the following
columns: </p>
 <ul>
<li><p>a p-value, must be numeric </p>
</li>
<li><p>a measure of the
strength of association, typically an odds ratio, regression coefficient or
log fold change. Must be numeric </p>
</li></ul>
</td></tr>
<tr><td><code id="volcanor_+3A_p">p</code></td>
<td>
<p>A chracter string denoting the column name for the p-values. Default
is <code>p = "P"</code>. This column must be <code>numeric</code> or <code>integer</code>.
Should not have missing, NA, NaN, or NULL values and should be between 0
and 1.</p>
</td></tr>
<tr><td><code id="volcanor_+3A_effect_size">effect_size</code></td>
<td>
<p>A string denoting the column name for the effect size.
Default is <code>effect_size = "EFFECTSIZE"</code>. This column must be <code>numeric</code> or
<code>integer</code>. Should not have missing, NA, NaN, or NULL values.</p>
</td></tr>
<tr><td><code id="volcanor_+3A_snp">snp</code></td>
<td>
<p>A string denoting the column name for the SNP names (e.g. rs
number). This argument is optional but required if you want to highlight
any points. More generally, this column could be anything that identifies
each point being plotted. For example, in an Epigenomewide association
study (EWAS) this could be the probe name or cg number. This column should
contain <code>characters</code>. This argument is necessary.
<code><a href="#topic+volcanoly">volcanoly</a></code> function</p>
</td></tr>
<tr><td><code id="volcanor_+3A_gene">gene</code></td>
<td>
<p>A string denoting the column name for the GENE names. This column
could be a <code>character</code> or <code>numeric</code>. More generally this could be
any annotation information that you want to include in the plot. This
argument is optional.</p>
</td></tr>
<tr><td><code id="volcanor_+3A_annotation1">annotation1</code></td>
<td>
<p>A string denoting the column name for an annotation. This
column could be a <code>character</code> or <code>numeric</code>. This could be any
annotation information that you want to include in the plot (e.g. zscore,
effect size, minor allele frequency). This argument is optional.</p>
</td></tr>
<tr><td><code id="volcanor_+3A_annotation2">annotation2</code></td>
<td>
<p>A string denoting the column name for an annotation. This
column could be a <code>character</code> or <code>numeric</code>. This could be any
annotation information that you want to include in the plot (e.g. zscore,
effect size, minor allele frequency). This argument is optional.</p>
</td></tr>
<tr><td><code id="volcanor_+3A_...">...</code></td>
<td>
<p>currently ignored</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An list object of class <code>volcanor</code> with the following elements
</p>
  <dl>
<dt>data</dt><dd><p>processed data to be used for plotting the volcano
plot including the observed and expected p-values on the -log10 scale</p>
</dd>
<dt>pName, snpName, geneName, annotation1Name, annotation2Name</dt><dd><p>The names
of the columns corresponding to the data provided. This information is used
for annotating the plot in the <code><a href="#topic+volcanoly">volcanoly</a></code> function </p>
</dd> </dl>



<h3>Note</h3>

<p>This function will return an error if any of the p-values are NA, less
than 0 or greater than 1
</p>


<h3>See Also</h3>

<p><code><a href="#topic+volcanoly">volcanoly</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(manhattanly)
volcanorObj &lt;- volcanor(HapMap)
class(volcanorObj)
head(volcanorObj)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
