<!DOCTYPE html><html><head><title>Help for package manhplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {manhplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#manhplot-package'>
<p>The Manhattan++ Plot</p></a></li>
<li><a href='#manhplusplot'><p>Generate the manhattan++ plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The Manhattan++ Plot</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0)</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-11-25</td>
</tr>
<tr>
<td>Author:</td>
<td>Chris Grace &lt;cgrace@well.ox.ac.uk&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chris Grace &lt;cgrace@well.ox.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>This plot integrates annotation into a manhattan plot. The plot is implemented as a heatmap, which is binned using -log10(p-value) and chromosome position. Annotation currently supported is minor allele frequency and gene function high impact variants.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2, ggplot2, ggrepel, gridExtra</td>
</tr>
<tr>
<td>Suggests:</td>
<td>R.utils, testthat</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/cgrace1978/manhplot/">https://github.com/cgrace1978/manhplot/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/cgrace1978/manhplot/issues">https://github.com/cgrace1978/manhplot/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-11-25 15:53:19 UTC; cgrace</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-11-25 16:40:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='manhplot-package'>
The Manhattan++ Plot
</h2><span id='topic+manhplot-package'></span><span id='topic+manhplot'></span>

<h3>Description</h3>

<p>This plot integrates annotation into a manhattan plot. The plot is implemented as a heatmap, which is binned using -log10(p-value) and chromosome position. Annotation currently supported is minor allele frequency and gene function high impact variants.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> manhplot</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Title: </td><td style="text-align: left;"> The Manhattan++ Plot</td>
</tr>
<tr>
 <td style="text-align: left;">
Depends: </td><td style="text-align: left;"> R (&gt;= 3.4.0)</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2019-11-25</td>
</tr>
<tr>
 <td style="text-align: left;">
Author: </td><td style="text-align: left;"> Chris Grace &lt;cgrace@well.ox.ac.uk&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Maintainer: </td><td style="text-align: left;"> Chris Grace &lt;cgrace@well.ox.ac.uk&gt;</td>
</tr>
<tr>
 <td style="text-align: left;">
Description: </td><td style="text-align: left;"> This plot integrates annotation into a manhattan plot. The plot is implemented as a heatmap, which is binned using -log10(p-value) and chromosome position. Annotation currently supported is minor allele frequency and gene function high impact variants.</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
RoxygenNote: </td><td style="text-align: left;"> 6.1.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Imports: </td><td style="text-align: left;"> reshape2, ggplot2, ggrepel, gridExtra</td>
</tr>
<tr>
 <td style="text-align: left;">
Suggests: </td><td style="text-align: left;"> R.utils, testthat</td>
</tr>
<tr>
 <td style="text-align: left;">
URL: </td><td style="text-align: left;"> https://github.com/cgrace1978/manhplot/</td>
</tr>
<tr>
 <td style="text-align: left;">
BugReports: </td><td style="text-align: left;"> https://github.com/cgrace1978/manhplot/issues</td>
</tr>
<tr>
 <td style="text-align: left;">
NeedsCompilation: </td><td style="text-align: left;"> no</td>
</tr>
<tr>
 <td style="text-align: left;">
Packaged: </td><td style="text-align: left;"> 2019-11-25 15:02:16 UTC; cgrace</td>
</tr>
<tr>
 <td style="text-align: left;">
Repository: </td><td style="text-align: left;"> CRAN</td>
</tr>
<tr>
 <td style="text-align: left;">
Date/Publication: </td><td style="text-align: left;"> 2019-06-04 11:40:03 UTC</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<p>Index of help topics:
</p>
<pre>
manhplot-package        The Manhattan++ Plot
manhplusplot            Generate the manhattan++ plot
</pre>


<h3>Author(s)</h3>

<p>Chris Grace &lt;cgrace@well.ox.ac.uk&gt;
</p>
<p>Maintainer: Chris Grace &lt;cgrace@well.ox.ac.uk&gt;
</p>

<hr>
<h2 id='manhplusplot'>Generate the manhattan++ plot</h2><span id='topic+manhplusplot'></span>

<h3>Description</h3>

<p>Generate the manhattan++ plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhplusplot(infile, outfile, configfile, snpfile, drawastiff = F,
  GWS = 5e-08, FDR = 0.001, MAF = 0.05, chrname = "chr",
  posname = "pos", pvalname = "pvalue", frqname = "maf",
  conseqname = "conseq", showgenes = F, showrsids = F,
  pos.split = 3e+06, pval.split = 0.125, max.pval = 20)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhplusplot_+3A_infile">infile</code></td>
<td>
<p>Input GWAS summary statistics</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_outfile">outfile</code></td>
<td>
<p>Output file prefix for the manhattan++ plot</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_configfile">configfile</code></td>
<td>
<p>Configuration file</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_snpfile">snpfile</code></td>
<td>
<p>Table of SNPs to visualize</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_drawastiff">drawastiff</code></td>
<td>
<p>If TRUE draw a Tiff file, if FALSE draw a PDF file</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_gws">GWS</code></td>
<td>
<p>Genome wise significance pvalue threshold (5E-8 by default)</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_fdr">FDR</code></td>
<td>
<p>False discovery Rate pvalue threshold (1E-3 by default)</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_maf">MAF</code></td>
<td>
<p>Minor Allele Frequency threshold</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_chrname">chrname</code></td>
<td>
<p>Column name for chromosome in GWAS infile</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_posname">posname</code></td>
<td>
<p>Column name for position in GWAS infile</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_pvalname">pvalname</code></td>
<td>
<p>Column name for pvalue in GWAS infile</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_frqname">frqname</code></td>
<td>
<p>column name for allele frequency in GWAS infile</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_conseqname">conseqname</code></td>
<td>
<p>column name for variant annotation consequence in GWAS infile</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_showgenes">showgenes</code></td>
<td>
<p>If T shows known genes as bubbles on main manhattan plot, if F show positions of interest as bubbles</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_showrsids">showrsids</code></td>
<td>
<p>If showgenes is T, then show the rsids, rather than genes</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_pos.split">pos.split</code></td>
<td>
<p>The bin lengths for positions</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_pval.split">pval.split</code></td>
<td>
<p>The bin lengths for pvalues</p>
</td></tr>
<tr><td><code id="manhplusplot_+3A_max.pval">max.pval</code></td>
<td>
<p>The maximum pvalue to display</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For file formats see github page <a href="https://github.com/cgrace1978/manhplot">https://github.com/cgrace1978/manhplot</a>
</p>


<h3>Author(s)</h3>

<p>Chris Grace
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

library(manhplot)
## Load R.utils for gzip functionality
library(R.utils)

## unzip the data included with this package
gunzip(system.file("extdata","cad.add.160614_manhformat.txt.gz",package = "manhplot"))

infile&lt;-system.file("extdata","cad.add.160614_manhformat.txt",package = "manhplot")
configfile&lt;-system.file("extdata","config.txt", package = "manhplot")
snpfile&lt;-system.file("extdata","56cad.add.160614.variants.txt", package = "manhplot")

manhplusplot(infile = infile,outfile = file.path(tempdir(), "default-plot"), 
               configfile = configfile, snpfile = snpfile)

## zip the data
gzip(system.file("extdata","cad.add.160614_manhformat.txt",package = "manhplot"))


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
