<!DOCTYPE html><html><head><title>Help for package Map2NCBI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {Map2NCBI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Example10MarkerFile'>
<p>Example Marker File with 10 Markers on BTA 1</p></a></li>
<li><a href='#GeneList_BTA1'>
<p>Output from GetGeneList function for BTA 1</p></a></li>
<li><a href='#GetGeneList'>
<p>A Function to Filter and Save Genomic Features from NCBI (all builds)</p></a></li>
<li><a href='#Map2NCBI-package'>
<p>Mapping Markers to the Nearest Genomic Feature</p></a></li>
<li><a href='#MapMarkers'>
<p>Mapping SNP Markers to Closest Genomic Feature</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Mapping Markers to the Nearest Genomic Feature</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-01-23</td>
</tr>
<tr>
<td>Author:</td>
<td>Lauren L. Hulsman Hanna and David G. Riley</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lauren Hanna &lt;Lauren.Hanna@ndsu.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows the user to generate a list of features (gene, pseudo, RNA,
    CDS, and/or UTR) directly from NCBI database for any species with a current 
    build available. Option to save downloaded and formatted files is available, 
    and the user can prioritize the feature list based on type and assembly builds 
    present in the current build used. The user can then use the list of features 
    generated or provide a list to map a set of markers (designed for SNP markers 
    with a single base pair position available) to the closest feature based on 
    the map build. This function does require map positions of the markers to be 
    provided and the positions should be based on the build being queried through 
    NCBI. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5), rentrez (&ge; 1.2)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-01-24 18:31:27 UTC; lauren.hanna</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-01-24 19:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='Example10MarkerFile'>
Example Marker File with 10 Markers on BTA 1
</h2><span id='topic+Example10MarkerFile'></span>

<h3>Description</h3>

<p>This file contains marker map placement based on UCD 1.2 build and is meant to be used as an example of the <code><a href="#topic+MapMarkers">MapMarkers</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Example10MarkerFile)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;Example10MarkerFile&quot;
</p>


<h3>Details</h3>

<p>Markers are SNP from BovSNP50 assay by Illumina, Inc.(San Diego, CA). These markers were used in a Nellore-Angus crossbred population described in the reference below. Map positions are based on the current Bos taurus assembly.
</p>


<h3>References</h3>

<p>Riley, D.G., Welsh Jr., T.H., Gill, C.A., Hulsman, L.L., Herring, A.D., Riggs, P.K., Sawyer, J.E., Sanders, J.O., 2013. Whole genome association of SNP with newborn calf cannon bone length. Liv. Sci. 155: 186-196. doi:<a href="https://doi.org/10.1016/j.livsci.2013.05.022">10.1016/j.livsci.2013.05.022</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Example10MarkerFile)
</code></pre>

<hr>
<h2 id='GeneList_BTA1'>
Output from GetGeneList function for BTA 1
</h2><span id='topic+GeneList_BTA1'></span>

<h3>Description</h3>

<p>This output is provided to run the <code><a href="#topic+MapMarkers">MapMarkers</a></code> function example. It was generated using the <code><a href="#topic+GetGeneList">GetGeneList</a></code> function example and truncated to only include BTA 1 data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(GeneList_BTA1)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;GeneList_BTA1&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(GeneList_BTA1)
</code></pre>

<hr>
<h2 id='GetGeneList'>
A Function to Filter and Save Genomic Features from NCBI (all builds)
</h2><span id='topic+GetGeneList'></span>

<h3>Description</h3>

<p><code>GetGeneList</code> allows the user to access the NCBI database for the species specified using the secure ftp site, download feature information as well as filter and save feature information for future use. This update now allows users to specify if the latest assembly build should be used or not using the <span class="pkg">rentrez</span> package. Once the <code>GetGeneList</code> function is complete, no other access to NCBI or the internet is required. This function requires user input to determine the feature and class types that will be retained during the filtering process. Note: The requirements for this function have changed slightly due to NCBI ftp site organization changes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetGeneList(Species,latest = TRUE, savefiles = TRUE, destfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetGeneList_+3A_species">Species</code></td>
<td>

<p>This term designates the species to be used in the function and is dependent on the scientific name.
Options: Must include in quotation marks, where the genus and species should be separated by a space (e.g., &quot;Bos taurus&quot;).
</p>
</td></tr>
<tr><td><code id="GetGeneList_+3A_latest">latest</code></td>
<td>

<p>Default is true. This term indicates if the most recent (latest) assembly build for that species should be used to get genomic features for. If set to false, the user will be prompted to idenify the assembly to use. In some species, the same assembly link may be listed more than once (e.g. GCF_000003055.6_Bos_taurus_UMD_3.1.1 vs. GCF_000003055.5_Bos_taurus_UMD_3.1.1). In any case, there is a number that designates one with a higher file number (e.g., &quot;3055.6&quot; vs. &quot;3055.5&quot; for Bos taurus 3.1). Always start with the higher file number for that build as it likely contains the feature table. If this fails, then try the other version. The assembly build should always match the marker map file build.
</p>
</td></tr>
<tr><td><code id="GetGeneList_+3A_savefiles">savefiles</code></td>
<td>

<p>Default is true. This term allows you to save the original feature list downloaded from the NCBI database as a text file as well as the filtered feature list produced from the function only if set to TRUE.
Options: Must be either TRUE or FALSE.
</p>
</td></tr>
<tr><td><code id="GetGeneList_+3A_destfile">destfile</code></td>
<td>

<p>This is the pathway to the computer location in which files will be saved and must be specified using quotation marks (e.g., <code>getwd()</code>).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In running this function, the user will be prompted to enter feedback after the file downloads. Items that will be requested, if multiples are present include 1) primary feature type and 2) primary class type to prioritize filtering the dataset on. In each case, the user can opt to keep all feature and class types. This will mean that duplicate information is available per gene ID. If filtered, all unique gene ID will be returned, where preference is given to the class feature and class types specified. Gene ID without the preferred feature and class types will be queried for their available information and added while still removing duplicates. The file returned contains 20 columns based on the current NCBI file structure. Those column headings and descriptions are provided below in the <code>Value</code> section.
</p>
<p>Note: While waiting for the function to run, if the user presses &quot;Enter&quot; prematurely, this will result in the function not running correctly and it will have to be started over. Please read instructions carefully.
</p>
<p>If <code>savefiles = TRUE</code>, then both the original file from NCBI and the filtered file the user specified will be saved in the <code>destfile</code> location. Once the function has run, the user can choose to either use the information at that time or call it later using the saved file. In either case, the output from the filtered file can be used with marker data to run the <code><a href="#topic+MapMarkers">MapMarkers</a></code> function that is also a part of this package.
</p>


<h3>Value</h3>

<p>Column headings and descriptions returned to the user from the <code>GetGeneList</code> function.
</p>
<table>
<tr><td><code>feature</code></td>
<td>
<p>The type of feature based on INSDC, which can include GENE, RNA (various types), and CDS.</p>
</td></tr>
<tr><td><code>class</code></td>
<td>
<p>Gene features are subdivided into classes according to the gene biotype. ncRNA features are subdivided according to the ncRNA_class. CDS features are subdivided into with_protein and without_protein, depending on whether the CDS feature has a protein accession assigned or not. CDS features marked as without_protein include CDS features for C regions and V/D/J segments of immunoglobulin and similar genes that undergo genomic rearrangement, and pseudogenes.</p>
</td></tr>
<tr><td><code>assembly</code></td>
<td>
<p>Accession.version of the assembly.</p>
</td></tr>
<tr><td><code>assembly_unit</code></td>
<td>
<p>The name of the assembly unit, such as &quot;Primary Assembly&quot;, &quot;ALT_REF_LOCI_1&quot;, or &quot;non-nuclear&quot;.</p>
</td></tr>
<tr><td><code>seq_type</code></td>
<td>
<p>The type of sequence the feature is from. Typically include chromosome, mitochondrion, plasmid, or unplaced scaffold.</p>
</td></tr>
<tr><td><code>chromosome</code></td>
<td>
<p>The chromosome the feature is located on, which can include mitochondrial DNA or unknown (blank) if applicable.</p>
</td></tr>
<tr><td><code>genomic_accession</code></td>
<td>
<p>The accession.version of that genome the feature is found on.</p>
</td></tr>
<tr><td><code>start</code></td>
<td>
<p>The start position of the feature on the chromosome.</p>
</td></tr>
<tr><td><code>end</code></td>
<td>
<p>The end position of the feature on the chromosome.</p>
</td></tr>
<tr><td><code>strand</code></td>
<td>
<p>The orientation of the feature on the chromosome (can be + or -).</p>
</td></tr>
<tr><td><code>product_accession</code></td>
<td>
<p>The accession.version of the product referenced by this feature, if it exists.</p>
</td></tr>
<tr><td><code>non-redundant_refseq</code></td>
<td>
<p>For bacteria and archaea assemblies, this column contains the non-redundant WP_ protein accession corresponding to the CDS feature. This may be the same as the previous column for RefSeq genomes annotated directly with WP_ RefSeq proteins, or may be different for genomes annotated with genome-specific protein accessions (e.g. NP_ or YP_ RefSeq proteins) that reference a WP_ RefSeq accession.</p>
</td></tr>
<tr><td><code>related_accession</code></td>
<td>
<p>For eukaryotic RefSeq annotations, this is the RefSeq protein accession corresponding to the transcript feature, or the RefSeq transcript accession corresponding to the protein feature.</p>
</td></tr>
<tr><td><code>name</code></td>
<td>
<p>For genes, this is the gene description or full name. For RNA, CDS, and some other features, this is the product name.</p>
</td></tr>
<tr><td><code>symbol</code></td>
<td>
<p>The gene symbol.</p>
</td></tr>
<tr><td><code>GeneID</code></td>
<td>
<p>The corresponding gene ID on the NCBI database the feature is located in.</p>
</td></tr>
<tr><td><code>locus_tag</code></td>
<td>
<p>No description available from NCBI. Typically a blank column.</p>
</td></tr>
<tr><td><code>feature_interval_length</code></td>
<td>
<p>This is the sum of the lengths of all intervals for the feature (i.e. the length without introns for a joined feature).</p>
</td></tr>
<tr><td><code>product_length</code></td>
<td>
<p>This is the length of the product corresponding to the accession.version in product_accession&quot; column. Protein product lengths are in amino acid units and do not include the stop codon which is included in &quot;feature_interval_length&quot; column. Additionally, product_length may differ from feature_interval_length if the product contains sequence differences vs. the genome, as found for some RefSeq transcript and protein products based on mRNA sequences and also for INSDC proteins that are submitted to correct genome discrepancies.</p>
</td></tr>
<tr><td><code>attributes</code></td>
<td>
<p>A semi-colon delimited list of a controlled set of qualifiers, if available. The list currently includes: partial, pseudo, pseudogene, ribosomal_slippage, trans_splicing, anticodon=NNN (for tRNAs), old_locus_tag=XXX.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For issues or problems with this function, please contact Lauren Hanna at <a href="mailto:Lauren.Hanna@ndsu.edu">Lauren.Hanna@ndsu.edu</a>.
</p>


<h3>Author(s)</h3>

<p>Lauren L. Hulsman Hanna and
David G. Riley
</p>


<h3>References</h3>

<p>Hulsman Hanna, L. L., and D. G. Riley. 2014. Mapping genomic markers to closest feature using the R package Map2NCBI. Livest. Sci. 162:59-65. doi:<a href="https://doi.org/10.1016/j.livsci.2014.01.019">10.1016/j.livsci.2014.01.019</a>
</p>
<p>National Center for Biotechnology Information. 2018. Latest assembly version 'README' file, last updated 26 February 2018. Available at: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/README.txt">https://ftp.ncbi.nlm.nih.gov/genomes/refseq/README.txt</a> (Accessed 23 Jan 2020).
</p>


<h3>See Also</h3>

<p>Function: <code><a href="#topic+MapMarkers">MapMarkers</a></code>,
Package: <span class="pkg">rentrez</span>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example 1: Run the following example and, when prompted,
#choose [n],[1],[n],[3] to filter the build and feature
#information. This example is interactive and requires
#user input. Please note that pressing "Enter" prematurely
#can cause the function to not run properly.
## Not run: 
GeneList = GetGeneList("Bos taurus",destfile=getwd())

## End(Not run)
</code></pre>

<hr>
<h2 id='Map2NCBI-package'>
Mapping Markers to the Nearest Genomic Feature
</h2><span id='topic+Map2NCBI-package'></span><span id='topic+Map2NCBI'></span>

<h3>Description</h3>

<p>Allows the user to generate a list of features (gene, pseudo, RNA, CDS, and/or UTR) directly from NCBI database for any species with a build available. Option to save downloaded and formatted files is available, and the user can prioritize the feature list based on type and assembly builds present in the build used. The GetGeneList function can now handle query of the NCBI for genome builds released prior to 2018 as well as the latest build for that species. The user can then use the list of features generated or provide a list to map a set of markers (designed for SNP markers with a single base pair position available) to the closest feature based on the map build. This function does require map positions of the markers to be provided and the positions should be based on the build being queried through NCBI.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> Map2NCBI</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.4</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-01-23</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>This package can be used as a two part process with the <code><a href="#topic+GetGeneList">GetGeneList</a></code> function followed by the <code><a href="#topic+MapMarkers">MapMarkers</a></code> function. See individual function documentation for more information.
</p>


<h3>Author(s)</h3>

<p>Lauren L. Hulsman Hanna and David G. Riley
</p>
<p>Maintainer: Lauren Hanna &lt;<a href="mailto:Lauren.Hanna@ndsu.edu">Lauren.Hanna@ndsu.edu</a>&gt;
</p>


<h3>References</h3>

<p>Hulsman Hanna, L. L., and D. G. Riley. 2014. Mapping genomic markers to closest feature using the R package Map2NCBI. Livest. Sci. 162:59-65. doi:<a href="https://doi.org/10.1016/j.livsci.2014.01.019">10.1016/j.livsci.2014.01.019</a>
</p>
<p>National Center for Biotechnology Information. 2018. Latest assembly version 'README' file, last updated 26 February 2018. Available at: <a href="https://ftp.ncbi.nlm.nih.gov/genomes/refseq/README.txt">https://ftp.ncbi.nlm.nih.gov/genomes/refseq/README.txt</a> (Accessed 23 Jan 2020).
</p>


<h3>See Also</h3>

<p>Functions: <code><a href="#topic+GetGeneList">GetGeneList</a></code> &amp; <code><a href="#topic+MapMarkers">MapMarkers</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#See individual function documentation for applied examples.
</code></pre>

<hr>
<h2 id='MapMarkers'>
Mapping SNP Markers to Closest Genomic Feature
</h2><span id='topic+MapMarkers'></span>

<h3>Description</h3>

<p><code>MapMarkers</code> allows the user to map the supplied DNA markers (primarily designed for SNP markers) to the genomic feature in closest proximity based on the feature list generated using the <code><a href="#topic+GetGeneList">GetGeneList</a></code> function or a properly formated feature list (see <code>Values</code> section).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MapMarkers(features, markers, nAut, other = c("X"), savefiles = TRUE, destfile)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MapMarkers_+3A_features">features</code></td>
<td>

<p>This is the table or matrix in the current R session that will be used to map the marker list to. If using the <code>GetGeneList</code> function, the name given to the output should be supplied here (e.g., &quot;GeneList&quot; from the example provided in the <code><a href="#topic+GetGeneList">GetGeneList</a></code> documentation file). Note: If using a feature list generated using version 1.1 of this package, please review column names below for changes in format to ensure the function runs properly.
</p>
</td></tr>
<tr><td><code id="MapMarkers_+3A_markers">markers</code></td>
<td>

<p>This is the table or matrix in the current R session that will provide marker map information to use for the function. See <code>Values</code> for format of the marker file needed.
</p>
</td></tr>
<tr><td><code id="MapMarkers_+3A_naut">nAut</code></td>
<td>

<p>The number of autosomes in the species. This should reflect the total number of autosomes in the species, not the number of autosomes in the marker file.
</p>
</td></tr>
<tr><td><code id="MapMarkers_+3A_other">other</code></td>
<td>

<p>The sex chromosomes or other genomic information available (e.g., for eukaryotes this could include mitochondrial DNA). These must be specified inside quotation marks. If sex chromosomes or other genomic information is not provided in the marker file, set other=FALSE.
</p>
</td></tr>
<tr><td><code id="MapMarkers_+3A_savefiles">savefiles</code></td>
<td>

<p>Default is TRUE. This term allows you to save the final marker file with genomic feature information in the destfile location as &quot;MappedMarkers.txt&quot; format. Any markers that cannot be mapped due to lack of feature information are saved as &quot;NotMapped.txt&quot;. Options: Must be either TRUE or FALSE.
</p>
</td></tr>
<tr><td><code id="MapMarkers_+3A_destfile">destfile</code></td>
<td>

<p>This is the pathway to the folder in which files will be saved and must be specified using quotation marks (e.g., <code>"C:/Temp/"</code> or <code>paste(getwd(),"/",sep="")</code>).
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The <code>MapMarkers</code> function processes each chromosome individually to search for features that fall closest to the markers provided based on the map information included. Map positions of the markers must match the assembly being used in the feature list. Once the closest feature has been found, the marker and feature information are saved together and take the format of binding the marker map file (which include at a minimum 3 columns) with the feature list columns provided (20 columns if using the <code>GetGeneList</code> function or a minimum of 4 columns if formatting yourself). The function also adds 2 additional columns described in <code>Value</code> section to identify the distance the marker is from the feature and a category to group the marker's proximity to the feature by.
</p>


<h3>Value</h3>

<p>1) Format for feature list if not generated using the <code>GetGeneList</code> function:
</p>
<table>
<tr><td><code>FeatureName</code></td>
<td>
<p>The name of the feature provided. Column heading name can be changed, but should be included to identify the feature once the <code>MapMarkers</code> function is completed.</p>
</td></tr>
<tr><td><code>chromosome</code></td>
<td>
<p>The chromosome in which the genomic feature is located on. The column heading name must be given this name. If including sex chromosomes or other genomic information, label based on letters or abbreviation (e.g., &quot;X&quot;).</p>
</td></tr>
<tr><td><code>start</code></td>
<td>
<p>The start position of the genomic feature based on the build used. The column heading name must be given this name. This used to be called &quot;chr_start&quot; in version 1.1 of this package.</p>
</td></tr>
<tr><td><code>end</code></td>
<td>
<p>The end or stop position of the genomic feature based on the build used. The column heading name must be given this name. This used to be called &quot;chr_stop&quot; in version 1.1 of this package.</p>
</td></tr>
</table>
<p>2) Format for the marker map file:
</p>
<table>
<tr><td><code>Marker</code></td>
<td>
<p>Name of the marker. Be aware of R language and its restrictions. The name of this column heading can be changed to something else.</p>
</td></tr>
<tr><td><code>chromosome</code></td>
<td>
<p>The chromosome in which the marker is mapped to. The name of this column is required and must be exact. This must be numeric. If including sex chromosomes or other genomic information, assign numbers to each. Number the sex chromosomes or other genomic information in the order that matches the order listed in the other=c() statement (e.g., X and Y chromosomes are labeled 30 and 31, respectively, so other=c(&quot;X&quot;,&quot;Y&quot;) to follow that order). The function will automatically align the letter with the correct number as long as they are included in the order specified.</p>
</td></tr>
<tr><td><code>position</code></td>
<td>
<p>The base pair position of the marker based on the map build used. This build must also match the build in which you generated genomic feature from using the <code>GetGeneList</code> function or other method. The name of this column is required and must be exact. This function was designed for SNP markers, but if using other types of markers, you should choose the base pair location that best represents the marker (e.g., center position) and include that in this column.</p>
</td></tr>
</table>
<p>NOTE: Order of the columns in both files are not necessarily important, but correct column heading names are essential. R programming is case sensitive, so make sure it matches exactly unless otherwise noted. Other columns may be included, but will not be used by the function. Any columns included in this file will be returned with the final marker file after the <code>MapMarkers</code> function is completed.
</p>
<p>3) Additional columns included in the output file of the <code>MapMarkers</code> function:
</p>
<table>
<tr><td><code>Distance</code></td>
<td>
<p>The base pair distance of the marker from the closest feature identified. If the marker is located inside the feature, the distance is set to zero.</p>
</td></tr>
<tr><td><code>Inside?</code></td>
<td>
<p>The category in which the marker and feature pair fall into. This is based on the distance between the Marker and the closest feature, which is broken into 11 categories described in the next section.</p>
</td></tr>
</table>
<p>4) Categories that are included in the &quot;Inside?&quot; column:
</p>
<table>
<tr><td><code>Yes</code>, <code>_Inside_Gene</code></td>
<td>
<p>Marker is located in the closest feature.</p>
</td></tr>
<tr><td><code>Marker_is_&lt;=_2500_bp_Before_Feature</code></td>
<td>
<p>The closest feature is located after the marker position and is within 2,500 base pairs (bp).</p>
</td></tr>
<tr><td><code>Marker_is_&lt;=_2500_bp_After_Feature</code></td>
<td>
<p>The closest feature is located before the marker position and is within 2,500 bp.</p>
</td></tr>
<tr><td><code>Marker_is_&gt;_2500_bp_&lt;=5000_bp_Before_Feature</code></td>
<td>
<p>The closest feature is located before the marker position and is between 2,500 bp and 5,000 bp from the marker.</p>
</td></tr>
<tr><td><code>Marker_is_&gt;_2500_bp_&lt;=5000_bp_After_Feature</code></td>
<td>
<p>The closest feature is located after the marker position and is between 2,500 bp and 5,000 bp from the marker.</p>
</td></tr>
<tr><td><code>Marker_is_&gt;_5000_bp_&lt;=25000_bp_Before_Feature</code></td>
<td>
<p>The closest feature is located before the marker position and is between 5,000 bp and 25,000 bp from the marker.</p>
</td></tr>
<tr><td><code>Marker_is_&gt;_5000_bp_&lt;=25000_bp_After_Feature</code></td>
<td>
<p>The closest feature is located after the marker position and is between 5,000 bp and 25,000 bp from the marker.</p>
</td></tr>
<tr><td><code>Nearest_feature_is_&gt;_25</code>, <code>000_bp_before_marker</code></td>
<td>
<p>The closest feature is located before the marker position and is more than 25,000 bp from the marker.</p>
</td></tr>
<tr><td><code>Nearest_feature_is_&gt;_25</code>, <code>000_bp_after_marker</code></td>
<td>
<p>The closest feature is located after the marker position and is more than 25,000 bp from the marker.</p>
</td></tr>
<tr><td><code>Nearest_feature_is_&gt;_1_Mb_before_marker</code></td>
<td>
<p>The closest feature is located before the marker position and is more than 1,000,000 bp (1 Mb) from the marker.</p>
</td></tr>
<tr><td><code>Nearest_feature_is_&gt;_1_Mp_after_marker</code></td>
<td>
<p>The closest feature is located after the marker position and is more than 1,000,000 bp (1 Mb) from the marker.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>For issues or problems with this function, please contact
Lauren Hanna at <a href="mailto:Lauren.Hanna@ndsu.edu">Lauren.Hanna@ndsu.edu</a>.
</p>


<h3>Author(s)</h3>

<p>Lauren L. Hulsman Hanna and David G. Riley
</p>


<h3>References</h3>

<p>Hulsman Hanna, L. L., and D. G. Riley. 2014. Mapping genomic markers to closest feature using the R package Map2NCBI. Livest. Sci. 162:59-65. doi:<a href="https://doi.org/10.1016/j.livsci.2014.01.019">10.1016/j.livsci.2014.01.019</a>
</p>


<h3>See Also</h3>

<p>Function: <code><a href="#topic+GetGeneList">GetGeneList</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Example 1: Step 1 includes running "GetGeneList" function.
#As this step is interactive, a dataset from Bos taurus has
#been generated and available to use in the \data folder as
#well as a subset of marker information from BTA 1. Use the
#following code to run this example:

data(GeneList_BTA1)
data(Example10MarkerFile)
Example1 = MapMarkers(GeneList_BTA1, Example10MarkerFile,
    nAut=29,other="X",savefiles = FALSE)

#Note, this example will not save the output to the working
#directory, but will return the information to "Example1"
#variable.
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
