<!DOCTYPE html><html lang="en"><head><title>Help for package MAPITR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MAPITR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MAPITR'><p>MArginal ePIstasis Test for Regions</p>
(<code>MAPITR</code>)</a></li>
<li><a href='#MAPITR_SimData_Genotypes'><p>MAPITR SimData Genotypes</p></a></li>
<li><a href='#MAPITR_SimData_Pathways'><p>MAPITR SimData Pathways</p></a></li>
<li><a href='#MAPITR_SimData_PCs'><p>MAPITR SimData PCs</p></a></li>
<li><a href='#MAPITR_SimData_Phenotype'><p>MAPITR SimData Phenotype</p></a></li>
<li><a href='#MAPITR_TestData_Genotypes'><p>MAPITR TestData Genotypes</p></a></li>
<li><a href='#MAPITR_TestData_Pathways'><p>MAPITR TestData Pathways</p></a></li>
<li><a href='#MAPITR_TestData_PCs'><p>MAPITR TestData PCs</p></a></li>
<li><a href='#MAPITR_TestData_Phenotype'><p>MAPITR TestData Phenotype</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>MArginal ePIstasis Test for Regions</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-09-24</td>
</tr>
<tr>
<td>Description:</td>
<td>A genetic analysis tool and variance component 
    model for identifying marginal epistasis between pathways 
    and the rest of the genome. 'MAPITR' uses as input a matrix 
    of genotypes, a vector of phenotypes, and a list of 
    pathways. 'MAPITR' then iteratively tests each pathway for 
    epistasis between any variants within the pathway versus 
    any variants remaining in the rest of the genome. 'MAPITR'
    returns results in the form of p-values for every pathway 
    indicating whether the null model of there being no 
    epistatic interactions between a pathway and the rest of 
    the genome can be rejected.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mturchin20/MAPITR">https://github.com/mturchin20/MAPITR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/mturchin20/MAPITR/issues">https://github.com/mturchin20/MAPITR/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, doParallel, Rcpp (&ge; 1.0.5), CompQuadForm</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-09-25 03:52:13 UTC; mturchin20</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Turchin <a href="https://orcid.org/0000-0003-3569-1529"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Gregory Darnell <a href="https://orcid.org/0000-0003-0425-940X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb],
  Lorin Crawford <a href="https://orcid.org/0000-0003-0178-8242"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ctb],
  Sohini Ramachandran
    <a href="https://orcid.org/0000-0002-9588-7964"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Peter Carbonetto <a href="https://orcid.org/0000-0003-1144-6780"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Turchin &lt;michael_turchin@brown.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-09-28 13:20:12 UTC</td>
</tr>
</table>
<hr>
<h2 id='MAPITR'>MArginal ePIstasis Test for Regions
(<code>MAPITR</code>)</h2><span id='topic+MAPITR'></span>

<h3>Description</h3>

<p>Run <code>MAPITR</code> for a group of pathways on a single
phenotype and a set of genome-wide SNPs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MAPITR(
  Genotypes,
  Phenotype,
  Pathways,
  Covariates = NULL,
  CenterStandardize = TRUE,
  OpenMP = FALSE,
  cores = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MAPITR_+3A_genotypes">Genotypes</code></td>
<td>
<p>A n x p matrix containing the genotypes (0/1/2) for
all p SNPs across all n individuals. No default value.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_phenotype">Phenotype</code></td>
<td>
<p>A vector containing phenotypic values for all
individuals being analyzed. No default value.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_pathways">Pathways</code></td>
<td>
<p>A r x 2 matrix containing the pathway names and then
a comma-separated list of the <code>Genotypes</code> column indices
representing each SNP in the associated pathway. Note, this second 
column of comma-separated indices are the numeric positions for each 
SNP in <code>Genotypes</code> and not the SNP IDs or column names. No
default value.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_covariates">Covariates</code></td>
<td>
<p>A n x q matrix containing any q additional covariates
that should be included in the M-projection matrix of the model. 
See Turchin et al. 2020 for details. Note that these are covariates
which are applied to both sides of the model, ie the phenotype as well
as the genotypes. A y-intercept term is automatically included and does
not need to be part of this n x q matrix. No default value.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_centerstandardize">CenterStandardize</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that
indicates whether the genotype matrix <code>Genotypes</code> should be
centered and standardized before analysis. This is a recommended step.
Indicate <code>FALSE</code> if this is a preprocessing step that has
already been done prior to running <code>MAPITR</code>. The default value 
is TRUE.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_openmp">OpenMP</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that indicates
whether OpenMP versions of <code>MAPITR</code> should be implemented.
OpenMP versions of the underlying Rcpp code will run more quickly, 
but requires the user to be operating an R version that has 
been installed with OpenMP access. The default value is FALSE.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_cores">cores</code></td>
<td>
<p>A numeric value providing the expected number of cores
if the OpenMP version of the code is being used. 
<code>parallel::detectCores()</code> is used by default to assign this 
variable when no value is given. A value generally should only be 
given when needing finer control of the code or for testing purposes. 
The default value is NULL.</p>
</td></tr>
<tr><td><code id="MAPITR_+3A_...">...</code></td>
<td>
<p>Additional optional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing two entries. First, a dataframe (<code>Results</code>)
containing in the first column the list of pathways that were analyzed, 
in the second column the associated <code>MAPITR</code> p-values for each 
pathway, in the third column the associated <code>MAPITR</code> variance 
component estimates for each pathway, and in the fourth column the 
associated <code>MAPITR</code> PVEs for each pathway. Second, a matrix
(<code>Eigenvalues</code>) containing the n associated <code>MAPITR</code> 
eigenvalues for each pathway per column.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MAPITR_TestData_Genotypes, MAPITR_TestData_Phenotype, 
MAPITR_TestData_Pathways)
MAPITROutput &lt;- MAPITR(MAPITR_TestData_Genotypes, MAPITR_TestData_Phenotype, 
MAPITR_TestData_Pathways, OpenMP=FALSE)
MAPITROutput$Results

</code></pre>

<hr>
<h2 id='MAPITR_SimData_Genotypes'>MAPITR SimData Genotypes</h2><span id='topic+MAPITR_SimData_Genotypes'></span>

<h3>Description</h3>

<p>A simulated dataset of genotypes for use in 
vignettes
</p>


<h3>Format</h3>

<p>A data frame with 500 rows (individuals) and 750 
variables (SNPs)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_SimData_Pathways'>MAPITR SimData Pathways</h2><span id='topic+MAPITR_SimData_Pathways'></span>

<h3>Description</h3>

<p>A simulated set of pathways for use in 
vignettes
</p>


<h3>Format</h3>

<p>A data frame with 5 rows (pathways) and 2 variables 
(pathway name, comma-separated list of pathway column indices)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_SimData_PCs'>MAPITR SimData PCs</h2><span id='topic+MAPITR_SimData_PCs'></span>

<h3>Description</h3>

<p>Top 10 principal components from the simulated
genotypes for for use in vignettes
</p>


<h3>Format</h3>

<p>A data frame with 500 rows (individuals) and 10 
variables (PCs)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_SimData_Phenotype'>MAPITR SimData Phenotype</h2><span id='topic+MAPITR_SimData_Phenotype'></span>

<h3>Description</h3>

<p>A simulated set of phenotypes for use in
vignettes
</p>


<h3>Format</h3>

<p>A data frame with 500 rows (individuals) and 1 variables 
(phenotype)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_TestData_Genotypes'>MAPITR TestData Genotypes</h2><span id='topic+MAPITR_TestData_Genotypes'></span>

<h3>Description</h3>

<p>A simulated dataset of genotypes for use in unit
tests
</p>


<h3>Format</h3>

<p>A data frame with 500 rows (individuals) and 750 
variables (SNPs)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_TestData_Pathways'>MAPITR TestData Pathways</h2><span id='topic+MAPITR_TestData_Pathways'></span>

<h3>Description</h3>

<p>A simulated set of pathways for use in unit
tests
</p>


<h3>Format</h3>

<p>A data frame with 5 rows (pathways) and 2 variables 
(pathway name, comma-separated list of pathway column indices)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_TestData_PCs'>MAPITR TestData PCs</h2><span id='topic+MAPITR_TestData_PCs'></span>

<h3>Description</h3>

<p>Top 10 principal components from the simulated
genotypes for for use in unit tests
</p>


<h3>Format</h3>

<p>A data frame with 500 rows (individuals) and 10 
variables (PCs)
</p>


<h3>Source</h3>

<p>Manually created
</p>

<hr>
<h2 id='MAPITR_TestData_Phenotype'>MAPITR TestData Phenotype</h2><span id='topic+MAPITR_TestData_Phenotype'></span>

<h3>Description</h3>

<p>A simulated set of phenotypes for use in unit
tests
</p>


<h3>Format</h3>

<p>A data frame with 500 rows (individuals) and 1 
variables (phenotype)
</p>


<h3>Source</h3>

<p>Manually created
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
