<!DOCTYPE html><html><head><title>Help for package marble</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {marble}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dat'><p>simulated data for demonstrating the features of marble.</p></a></li>
<li><a href='#GxESelection'><p>Variable selection for a marble object</p></a></li>
<li><a href='#marble'><p>fit a robust Bayesian variable selection model for G×E interactions.</p></a></li>
<li><a href='#marble-package'><p>Robust Marginal Bayesian Variable Selection for Gene-Environment Interactions</p></a></li>
<li><a href='#print.GxESelection'><p>print a GxESelection object</p></a></li>
<li><a href='#print.marble'><p>print a marble object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Robust Marginal Bayesian Variable Selection for Gene-Environment
Interactions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-05-09</td>
</tr>
<tr>
<td>Description:</td>
<td>Recently, multiple marginal variable selection methods have been developed and shown to be effective in Gene-Environment interactions studies. We propose a novel marginal Bayesian variable selection method for Gene-Environment interactions studies. In particular, our marginal Bayesian method is robust to data contamination and outliers in the outcome variables. With the incorporation of spike-and-slab priors, we have implemented the Gibbs sampler based on Markov Chain Monte Carlo. The core algorithms of the package have been developed in 'C++'.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/xilustat/marble">https://github.com/xilustat/marble</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp, stats</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-10 19:03:12 UTC; grimf</td>
</tr>
<tr>
<td>Author:</td>
<td>Xi Lu [aut, cre],
  Cen Wu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Xi Lu &lt;xilu@ksu.edu&gt;</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-10 19:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='dat'>simulated data for demonstrating the features of marble.</h2><span id='topic+dat'></span><span id='topic+X'></span><span id='topic+Y'></span><span id='topic+E'></span><span id='topic+clin'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of marble.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("dat")
</code></pre>


<h3>Format</h3>

<p>dat consists of four components: X, Y, E, clin.
</p>


<h3>Details</h3>

<p><strong>The data model for generating Y</strong>
</p>
<p>Use subscript <code class="reqn">i</code> to denote the <code class="reqn">i</code>th subject. Let <code class="reqn">(Y_{i}, X_{i}, E_{i}, clin_{i})</code> (<code class="reqn">i=1,\ldots,n</code>) be
independent and identically distributed random vectors. <code class="reqn">Y_{i}</code> is a continuous response variable representing the
phenotype. <code class="reqn">X_{i}</code> is the <code class="reqn">p</code>&ndash;dimensional vector of genetic factors. The environmental factors and clinical factors
are denoted as the <code class="reqn">q</code>-dimensional vector <code class="reqn">E_{i}</code> and the <code class="reqn">m</code>-dimensional vector <code class="reqn">clin_{i}</code>, respectively.
The <code class="reqn">\epsilon</code> follows some heavy-tailed distribution. For <code class="reqn">X_{ij}</code> (<code class="reqn">j = 1,\ldots,p</code>), the measurement of the <code class="reqn">j</code>th genetic factor on the <code class="reqn">j</code>th subject, 
considering the following model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{k=1}^{q}\alpha_{k}E_{ik}+\sum_{t=1}^{m}\gamma_{t}clin_{it}+\beta_{j}X_{ij}+\sum_{k=1}^{q}\eta_{jk}X_{ij}E_{ik}+\epsilon_{i},</code>
</p>

<p>where <code class="reqn">\alpha_{0}</code> is the intercept, <code class="reqn">\alpha_{k}</code>'s and <code class="reqn">\gamma_{t}</code>'s are the regression coefficients corresponding to effects of environmental and clinical factors, respectively.
The <code class="reqn">\beta_{j}</code>'s and <code class="reqn">\eta_{jk}</code>'s are the regression coefficients of the genetic variants and G<code class="reqn">\times</code>E interactions effects, correspondingly. 
The G<code class="reqn">\times</code>E interactions effects are defined with <code class="reqn">W_{j} = (X_{j}E_{1},\ldots,X_{j}E_{q}).</code> With a slight abuse of notation, denote <code class="reqn">\tilde{W} = W_{j}.</code>  
Denote <code class="reqn">\alpha=(\alpha_{1}, \ldots, \alpha_{q})^{T}</code>, <code class="reqn">\gamma=(\gamma_{1}, \ldots, \gamma_{m})^{T}</code>, <code class="reqn">\beta=(\beta_{1}, \ldots, \beta_{p})^{T}</code>, <code class="reqn">\eta=(\eta_{1}^{T}, \ldots, \eta_{p}^{T})^{T}</code>, <code class="reqn">\tilde{W} = (\tilde{W_{1}}, \dots, \tilde{W_{p}})</code>. 
Then model can be written as 
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = E_{i}\alpha + clin_{i}\gamma + X_{ij}\beta_{j} + \tilde{W}_{i}\eta_{j} + \epsilon_{i}.</code>
</p>



<h3>See Also</h3>

<p><code><a href="#topic+marble">marble</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
dim(X)
</code></pre>

<hr>
<h2 id='GxESelection'>Variable selection for a marble object</h2><span id='topic+GxESelection'></span>

<h3>Description</h3>

<p>Variable selection for a marble object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GxESelection(obj, sparse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GxESelection_+3A_obj">obj</code></td>
<td>
<p>marble object.</p>
</td></tr>
<tr><td><code id="GxESelection_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For class &lsquo;Sparse&rsquo;,  the inclusion probability is used to indicate the importance of predictors. 
Here we use a binary indicator <code class="reqn">\phi</code> to denote the membership of the non-spike distribution. 
Take the main effect of the <code class="reqn">j</code>th genetic factor, <code class="reqn">X_{j}</code>, as an example. 
Suppose we have collected H posterior samples from MCMC after burn-ins. The <code class="reqn">j</code>th G factor is included 
in the marginal G<code class="reqn">\times</code>E model at the <code class="reqn">j</code>th MCMC iteration if the corresponding indicator is 1, i.e., <code class="reqn">\phi_j^{(h)} = 1</code>. 
Subsequently, the posterior probability of retaining the <code class="reqn">j</code>th genetic main effect in the final marginal model is defined as the average of all the indicators for the <code class="reqn">j</code>th G factor among the H posterior samples. 
That is, <code class="reqn">p_j = \hat{\pi} (\phi_j = 1|y) = \frac{1}{H} \sum_{h=1}^{H} \phi_j^{(h)}, \;  j = 1, \dots,p.</code>
A larger posterior inclusion probability of <code class="reqn">j</code>th indicates a stronger empirical evidence that the <code class="reqn">j</code>th genetic main effect has a non-zero coefficient, i.e., a stronger association with the phenotypic trait. 
Here, we use 0.5 as a cutting-off point. If <code class="reqn">p_j &gt; 0.5</code>, then the <code class="reqn">j</code>th genetic main effect is included in the final model. Otherwise, the <code class="reqn">j</code>th genetic main effect is excluded in the final model.
For class &lsquo;NonSparse&rsquo;, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;GxESelection&rsquo; is returned, which is a list with components:
</p>
<table>
<tr><td><code>method</code></td>
<td>
<p>method used for identifying important effects.</p>
</td></tr>
<tr><td><code>effects</code></td>
<td>
<p>a list of indicators of selected effects.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lu, X., Fan, K., Ren, J., and Wu, C. (2021). Identifying Gene–Environment Interactions With Robust Marginal Bayesian Variable Selection.
<em>Frontiers in Genetics</em>, 12:667074 <a href="https://doi.org/10.3389/fgene.2021.667074">doi:10.3389/fgene.2021.667074</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+marble">marble</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)
max.steps=5000
## sparse
fit=marble(X, Y, E, clin, max.steps=max.steps)
selected=GxESelection(fit,sparse=TRUE)
selected

## non-sparse
fit=marble(X, Y, E, clin, max.steps=max.steps, sparse=FALSE)
selected=GxESelection(fit,sparse=FALSE)
selected



</code></pre>

<hr>
<h2 id='marble'>fit a robust Bayesian variable selection model for G×E interactions.</h2><span id='topic+marble'></span>

<h3>Description</h3>

<p>fit a robust Bayesian variable selection model for G×E interactions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marble(
  X,
  Y,
  E,
  clin,
  max.steps = 10000,
  robust = TRUE,
  sparse = TRUE,
  debugging = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="marble_+3A_x">X</code></td>
<td>
<p>the matrix of predictors (genetic factors). Each row should be an observation vector.</p>
</td></tr>
<tr><td><code id="marble_+3A_y">Y</code></td>
<td>
<p>the continuous response variable.</p>
</td></tr>
<tr><td><code id="marble_+3A_e">E</code></td>
<td>
<p>a matrix of environmental factors. E will be centered. The interaction terms between X (genetic factors) and E will be automatically created and included in the model.</p>
</td></tr>
<tr><td><code id="marble_+3A_clin">clin</code></td>
<td>
<p>a matrix of clinical variables. Clinical variables are not subject to penalize. Clinical variables will be centered and a column of 1 will be added to the Clinical matrix as the intercept.</p>
</td></tr>
<tr><td><code id="marble_+3A_max.steps">max.steps</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="marble_+3A_robust">robust</code></td>
<td>
<p>logical flag. If TRUE, robust methods will be used.</p>
</td></tr>
<tr><td><code id="marble_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
<tr><td><code id="marble_+3A_debugging">debugging</code></td>
<td>
<p>logical flag. If TRUE, progress will be output to the console and extra information will be returned.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Consider the data model described in &quot;<code><a href="#topic+dat">dat</a></code>&quot;:
</p>
<p style="text-align: center;"><code class="reqn">Y_{i} = \alpha_{0} + \sum_{k=1}^{q}\alpha_{k}E_{ik}+\sum_{t=1}^{m}\gamma_{t}clin_{it}+\beta_{j}X_{ij}+\sum_{k=1}^{q}\eta_{jk}X_{ij}E_{ik}+\epsilon_{i},</code>
</p>

<p>Where <code class="reqn">\alpha_{0}</code> is the intercept, <code class="reqn">\alpha_{k}</code>'s and <code class="reqn">\gamma_{t}</code>'s are the regression coefficients corresponding to effects of environmental and clinical factors.
And <code class="reqn">\beta_{j}</code>'s and <code class="reqn">\eta_{jk}</code>'s are the regression coefficients of the genetic variants and G<code class="reqn">\times</code>E interactions effects, correspondingly. 
</p>
<p>When sparse=TRUE (default), spike&ndash;and&ndash;slab priors are imposed to identify important main and interaction effects. If sparse=FALSE, Laplacian shrinkage will be used.
</p>
<p>When robust=TRUE (default), the distribution of <code class="reqn">\epsilon_{i}</code> is defined as a Laplace distribution with density
<code class="reqn">
f(\epsilon_{i}|\nu) = \frac{\nu}{2}\exp\left\{-\nu |\epsilon_{i}|\right\}
</code>, (<code class="reqn">i=1,\dots,n</code>), which leads to a Bayesian formulation of LAD regression. If robust=FALSE, <code class="reqn">\epsilon_{i}</code> follows a normal distribution.
</p>
<p>Here, a rank list of the main and interaction effects is provided. For method incorporating spike-and-slab priors,  
the inclusion probability is used to indicate the importance of predictors. 
We use a binary indicator <code class="reqn">\phi</code> to denote the membership of the non-spike distribution. 
Take the main effect of the <code class="reqn">j</code>th genetic factor, <code class="reqn">X_{j}</code>, as an example. 
Suppose we have collected H posterior samples from MCMC after burn-ins. The <code class="reqn">j</code>th G factor is included 
in the marginal G<code class="reqn">\times</code>E model at the <code class="reqn">j</code>th MCMC iteration if the corresponding indicator is 1, i.e., <code class="reqn">\phi_j^{(h)} = 1</code>. 
Subsequently, the posterior probability of retaining the <code class="reqn">j</code>th genetic main effect in the final marginal model is defined as the average of all the indicators for the <code class="reqn">j</code>th G factor among the H posterior samples. 
That is, <code class="reqn">p_j = \hat{\pi} (\phi_j = 1|y) = \frac{1}{H} \sum_{h=1}^{H} \phi_j^{(h)}, \;  j = 1, \dots,p.</code>
A larger posterior inclusion probability <code class="reqn">j</code>th indicates a stronger empirical evidence that the <code class="reqn">j</code>th genetic main effect has a non-zero coefficient, i.e., a stronger association with the phenotypic trait. 
For method without spike-and-slab priors, variable selection is based on different level of credible intervals.
</p>
<p>Both <code class="reqn">X</code>, <code class="reqn">clin</code> and <code class="reqn">E</code> will be standardized before the generation of interaction terms to avoid the multicollinearity between main effects and interaction terms.
</p>
<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;marble&rsquo; is returned, which is a list with component:
</p>
<table>
<tr><td><code>posterior</code></td>
<td>
<p>the posterior samples of coefficients from the MCMC.</p>
</td></tr>
<tr><td><code>coefficient</code></td>
<td>
<p>the estimated value of coefficients.</p>
</td></tr>
<tr><td><code>ranklist</code></td>
<td>
<p>the rank list of main and interaction effects.</p>
</td></tr>
<tr><td><code>burn.in</code></td>
<td>
<p>the total number of burn-ins.</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>the total number of iterations.</p>
</td></tr>
<tr><td><code>design</code></td>
<td>
<p>the design matrix of all effects.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lu, X., Fan, K., Ren, J., and Wu, C. (2021). Identifying Gene–Environment Interactions With Robust Marginal Bayesian Variable Selection.
<em>Frontiers in Genetics</em>, 12:667074 <a href="https://doi.org/10.3389/fgene.2021.667074">doi:10.3389/fgene.2021.667074</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GxESelection">GxESelection</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dat)

## default method
max.steps=5000
fit=marble(X, Y, E, clin, max.steps=max.steps)

## coefficients of parameters
fit$coefficient

## Estimated values of main G effects 
fit$coefficient$G

## Estimated values of interactions effects 
fit$coefficient$GE

## Rank list of main G effects and interactions 
fit$ranklist


## alternative: robust selection
fit=marble(X, Y, E, clin, max.steps=max.steps, robust=TRUE, sparse=FALSE)
fit$coefficient
fit$ranklist

## alternative: non-robust sparse selection
fit=marble(X, Y, E, clin, max.steps=max.steps, robust=FALSE, sparse=FALSE)
fit$coefficient
fit$ranklist


</code></pre>

<hr>
<h2 id='marble-package'>Robust Marginal Bayesian Variable Selection for Gene-Environment Interactions</h2><span id='topic+marble-package'></span>

<h3>Description</h3>

<p>In this package, we provide a set of robust marginal Bayesian variable selection methods for gene-environment interaction analysis. A Bayesian formulation of the quantile regression has been adopted to accommodate data contamination and heavy-tailed distributions in the response. The proposed method conducts a robust marginal variable selection by accounting for structural sparsity. In particular, the spike-and-slab priors are imposed to identify important main and interaction effects.
In addition to the default method, users can also choose different structures (robust or non-robust), methods without spike-and-slab priors.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface <strong>marble()</strong> allows users to flexibly choose the fitting methods they prefer. There are two arguments in marble() that control the fitting method:
robust: whether to use robust methods; sparse: whether to use the spike-and-slab priors to create sparsity.
The function marble() returns a marble object that contains the posterior estimates of each coefficients. Moreover, it also provides a rank list of the genetic factors and gene-environment interactions.
Functions GxESelection() and print.marble() are implemented for marble objects.
GxESelection() takes a marble object and returns the variable selection results.
</p>


<h3>References</h3>

<p>Lu, X., Fan, K., Ren, J., and Wu, C. (2021). Identifying Gene–Environment Interactions With Robust Marginal Bayesian Variable Selection.
<em>Frontiers in Genetics</em>, 12:667074 <a href="https://doi.org/10.3389/fgene.2021.667074">doi:10.3389/fgene.2021.667074</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2020). Robust Bayesian variable selection for gene-environment interactions.
<a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2020). Gene–Environment Interaction: a Variable Selection Perspective. Epistasis. Methods in Molecular Biology.
<em>Humana Press</em> (Accepted) <a href="https://arxiv.org/abs/2003.02930">https://arxiv.org/abs/2003.02930</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Shi, X., Liu, J., Huang, J., Zhou, Y., Xie, Y. and Ma, S. (2014). A penalized robust method for identifying gene–environment interactions. 
<em>Genetic epidemiology</em>, 38(3), 220-230 <a href="https://doi.org/10.1002/gepi.21795">doi:10.1002/gepi.21795</a>
</p>
<p>Chai, H., Zhang, Q., Jiang, Y., Wang, G., Zhang, S., Ahmed, S. E. and Ma, S. (2017). Identifying gene-environment interactions for prognosis using a robust approach. 
<em>Econometrics and statistics</em>, 4, 105-120 <a href="https://doi.org/10.1016/j.ecosta.2016.10.004">doi:10.1016/j.ecosta.2016.10.004</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+marble">marble</a></code>
</p>

<hr>
<h2 id='print.GxESelection'>print a GxESelection object</h2><span id='topic+print.GxESelection'></span>

<h3>Description</h3>

<p>Print a summary of a GxESelection object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'GxESelection'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.GxESelection_+3A_x">x</code></td>
<td>
<p>GxESelection object.</p>
</td></tr>
<tr><td><code id="print.GxESelection_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.GxESelection_+3A_...">...</code></td>
<td>
<p>other print arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GxESelection">GxESelection</a></code>
</p>

<hr>
<h2 id='print.marble'>print a marble object</h2><span id='topic+print.marble'></span>

<h3>Description</h3>

<p>Print a summary of a marble object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'marble'
print(x, digits = max(3, getOption("digits") - 3), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.marble_+3A_x">x</code></td>
<td>
<p>marble object.</p>
</td></tr>
<tr><td><code id="print.marble_+3A_digits">digits</code></td>
<td>
<p>significant digits in printout.</p>
</td></tr>
<tr><td><code id="print.marble_+3A_...">...</code></td>
<td>
<p>other print arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, called for side effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+marble">marble</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
