<!DOCTYPE html><html lang="en"><head><title>Help for package MareyMap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MareyMap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Arabidopsis_thaliana'><p>The marey map for the species Arabidopsis thaliana</p></a></li>
<li><a href='#argList-methods'><p>Methods for function argList in package 'MareyMap'</p></a></li>
<li><a href='#Caenorhabditis_elegans'><p>The marey map for the species Caenorhabditis_elegans</p></a></li>
<li><a href='#color-methods'><p>Accessors for slot color in package 'MareyMap'</p></a></li>
<li><a href='#createOrder-methods'><p>createOrder</p></a></li>
<li><a href='#Drosophila_melanogaster'><p>The marey map for the species Drosophila melanogaster</p></a></li>
<li><a href='#geneticDistances-methods'><p>Accessors for slot geneticDistances</p></a></li>
<li><a href='#getInterList-methods'><p>returns a list of interpolation methods.</p></a></li>
<li><a href='#Homo_sapiens'><p>The marey map for the species Homo sapiens (female, male and mean of the two)</p></a></li>
<li><a href='#interpolate-methods'><p>interpolate</p></a></li>
<li><a href='#Interpolation-class'><p>Class &quot;Interpolation&quot;</p></a></li>
<li><a href='#interpolation-methods'><p>set or returns an interpolation.</p></a></li>
<li><a href='#InterpolationParam-class'><p>Class &quot;InterpolationParam&quot;</p></a></li>
<li><a href='#InterpolationParam-methods'><p>Methods for Function InterpolationParam in Package 'MareyMap'</p></a></li>
<li><a href='#interpolations-methods'><p>accessors for slot interpolations</p></a></li>
<li><a href='#MapCollection-class'><p>Class &quot;MapCollection&quot;</p></a></li>
<li><a href='#MapCollection-methods'><p>Constructor for class MapCollection</p></a></li>
<li><a href='#mapName-methods'><p> accessors for the slot mapName</p></a></li>
<li><a href='#mapNames-methods'><p>mapNames</p></a></li>
<li><a href='#MapSet-class'><p>Class &quot;MapSet&quot;</p></a></li>
<li><a href='#MapSet-methods'><p> Constructor for the class MapSet.</p></a></li>
<li><a href='#MareyMap-class'><p>Class &quot;MareyMap&quot;</p></a></li>
<li><a href='#MareyMap-methods'><p>Constructor for object of the class MareyMap.</p></a></li>
<li><a href='#markerNames-methods'><p>Accessors for the class markerNames.</p></a></li>
<li><a href='#markerValidity-methods'><p>accessors for slot markerValidity</p></a></li>
<li><a href='#MMLoess-class'><p>Class &quot;MMLoess&quot;. Interpolation method plug-in for MareyMap</p></a></li>
<li><a href='#MMSlidingWindow-class'><p>Class &quot;MMSlidingWindow&quot; Interpolation method plug-in for MareyMap</p></a></li>
<li><a href='#MMSpline3-class'><p>Class &quot;MMSpline3&quot; Interpolation using cubic splines</p></a></li>
<li><a href='#name-methods'><p>Accessor for slot name.</p></a></li>
<li><a href='#paramDefault-methods'><p>Accessors for slot paramDefault</p></a></li>
<li><a href='#paramDesc-methods'><p>Accessors for the slot paramDesc</p></a></li>
<li><a href='#paramFun-methods'><p> accessors for the slot paramFun</p></a></li>
<li><a href='#paramMax-methods'><p>Accessors for the slot paramMax.</p></a></li>
<li><a href='#paramMin-methods'><p>Accessors for slot paramMin.</p></a></li>
<li><a href='#paramName-methods'><p>Accessors for the slot paramName.</p></a></li>
<li><a href='#paramType-methods'><p>Accessors for the slot paramType.</p></a></li>
<li><a href='#paramValues-methods'><p>Accessor for the slot paramValues.</p></a></li>
<li><a href='#persistent-methods'><p>Accessors for the slot persistent.</p></a></li>
<li><a href='#physicalPositions-methods'><p>Accessors for the slot physicalPositions</p></a></li>
<li><a href='#plotMarkers-methods'><p>plotMarkers</p></a></li>
<li><a href='#plotModel-methods'><p>plotModel</p></a></li>
<li><a href='#plotModels-methods'><p>plotModels</p></a></li>
<li><a href='#plotRate-methods'><p> plotRate</p></a></li>
<li><a href='#plotRates-methods'><p> plotRates</p></a></li>
<li><a href='#query-methods'><p>  Methods for Function query</p></a></li>
<li><a href='#rates-methods'><p>Accessors for the slot rates</p></a></li>
<li><a href='#registerInterpolationMethod-methods'><p> registerInterpolationMethods</p></a></li>
<li><a href='#removeMarker-methods'><p> removeMarker</p></a></li>
<li><a href='#setName-methods'><p>Accessors for the slot setName.</p></a></li>
<li><a href='#setNames-methods'><p>setNames</p></a></li>
<li><a href='#startMareyMapGUI-methods'><p>MareyMapGUI launcher</p></a></li>
<li><a href='#textFile-methods'><p>textFile</p></a></li>
<li><a href='#userParam-methods'><p> userParam</p></a></li>
<li><a href='#validPositions-methods'><p>validPositions</p></a></li>
<li><a href='#visible-methods'><p>Accessors for the slot visible</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Estimation of Meiotic Recombination Rates Using Marey Maps</td>
</tr>
<tr>
<td>Version:</td>
<td>1.3.7</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, grDevices, methods, stats, tcltk, tools, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>tkrplot</td>
</tr>
<tr>
<td>Author:</td>
<td>Aurélie Siberchicot, Clément Rezvoy, Delphine Charif, Laurent Gueguen and Gabriel Marais</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aurélie Siberchicot &lt;aurelie.siberchicot@univ-lyon1.fr&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Description:</td>
<td>Local recombination rates are graphically estimated across a genome using Marey maps.</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/lbbe-software/MareyMap/issues">https://github.com/lbbe-software/MareyMap/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://lbbe.univ-lyon1.fr/fr/mareymap">https://lbbe.univ-lyon1.fr/fr/mareymap</a> ;
<a href="https://lbbe-shiny.univ-lyon1.fr/MareyMapOnline/">https://lbbe-shiny.univ-lyon1.fr/MareyMapOnline/</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 20:17:11 UTC; siberchicot</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-23 20:42:46 UTC</td>
</tr>
</table>
<hr>
<h2 id='Arabidopsis_thaliana'>The marey map for the species Arabidopsis thaliana</h2><span id='topic+Arabidopsis_thaliana'></span>

<h3>Description</h3>

<p>A marey map is an interpolation between genetical and physical maps of a mapomosome. This data file contains marey map for the species Arabidopsis thaliana.
This map is held into an object of class MapSet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Arabidopsis_thaliana)</code></pre>


<h3>Format</h3>

<p>Arabidopsis_thaliana contains an object of the class <code><a href="#topic+MapSet-class">MapSet-class</a></code>. The maps inside this object (objects of the class <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>) can be accessed using the operator &quot;[[&quot; (see example)
This set contains 5 maps called &quot;Chromosome 1&quot;, ..., &quot;Chromosome 5&quot;.
</p>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>Source</h3>

<p>The physical positions were also recorded for those genetic markers that have both been mapped to the Recombinant Inbred (RI) recombination map 
(see <a href="https://arabidopsis.info/">https://arabidopsis.info/</a>), and have been precisely physically mapped on the basis of flanking sequence, 
using the marker position information from TAIR. (release 1). 
Wright SI, Agrawal N, Bureau TE. Effects of recombination rate and gene density on transposable element distributions in Arabidopsis thaliana.Genome Res. 2003 Aug;13(8):1897-903.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(Arabidopsis_thaliana)
#map &lt;- Arabidopsis_thaliana[["Chromosome 1"]]
#plot(map)
#names(Arabidopsis_thaliana@maps)
</code></pre>

<hr>
<h2 id='argList-methods'>Methods for function argList in package 'MareyMap'</h2><span id='topic+argList'></span><span id='topic+argList-methods'></span>

<h3>Description</h3>

<p>Methods for function <code>argList</code> in package <span class="pkg">MareyMap</span> returns the argument list for object
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>returns the basic interpolation parameters (for use in createOrder)</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='Caenorhabditis_elegans'>The marey map for the species Caenorhabditis_elegans</h2><span id='topic+Caenorhabditis_elegans'></span>

<h3>Description</h3>

<p>A marey map is an interpolation between genetical and physical maps of a mapomosome. This data file contains marey map for the species Caenorhabditis_elegans.
This map is held into an object of class MapSet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Caenorhabditis_elegans)</code></pre>


<h3>Format</h3>

<p>Caenorhabditis_elegans contains an object of the class <code><a href="#topic+MapSet-class">MapSet-class</a></code>. The maps inside this object (objects of the class <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>) can be accessed using the operator &quot;[[&quot; (see example)
This set contains 6 maps called &quot;I&quot;, &quot;II&quot;, &quot;III&quot;, &quot;IV&quot;, &quot;V&quot; and &quot;X&quot;.
</p>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>Source</h3>

<p>Wormbase Release WS160, <a href="https://wormbase.org//">https://wormbase.org//</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Caenorhabditis_elegans)
map &lt;- Caenorhabditis_elegans[["I"]]
plot(map)
names(Caenorhabditis_elegans@maps)
</code></pre>

<hr>
<h2 id='color-methods'>Accessors for slot color in package 'MareyMap'</h2><span id='topic+color'></span><span id='topic+color-methods'></span><span id='topic+color+3C-'></span><span id='topic+color+3C--methods'></span>

<h3>Description</h3>

<p>Methods for function <code>color</code> in package <span class="pkg">MareyMap</span>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>Returns or sets the value of the slot color in the <code><a href="#topic+interpolation">interpolation</a></code> object</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='createOrder-methods'>createOrder</h2><span id='topic+createOrder'></span><span id='topic+createOrder-methods'></span>

<h3>Description</h3>

<p>Returns a character containing the code to recreate the object as it is at the moment of the call (e.g. &quot;new(objCLass, arg1=...)&quot;)
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MMSlidingWindow&quot;</dt><dd><p>createOrder for method MMSlidingWindow</p>
</dd>
<dt>object = &quot;MMLoess&quot;</dt><dd><p>createOrder for method MMLoess</p>
</dd>
<dt>object = &quot;MMSpline3&quot;</dt><dd><p>createOrder for method MMSpline3</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='Drosophila_melanogaster'>The marey map for the species Drosophila melanogaster</h2><span id='topic+Drosophila_melanogaster'></span>

<h3>Description</h3>

<p>A marey map is an interpolation between genetical and physical maps of a mapomosome. This data file contains marey map for the species Drosophila melanogaster.
This map is held into an object of class MapSet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Drosophila_melanogaster)</code></pre>


<h3>Format</h3>

<p>Drosophila_melanogaster contains an object of the class <code><a href="#topic+MapSet-class">MapSet-class</a></code>. The maps inside this object (objects of the class <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>) can be accessed using the operator &quot;[[&quot; (see example)
This set contains 6 maps called &quot;2L&quot;, &quot;2R&quot;, &quot;3L&quot;, &quot;3R&quot;, &quot;4&quot;, &quot;X&quot;.
</p>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>Source</h3>

<p>2nd release of D. mel genome, v 01/1998 Flybase. Marais G, Piganeau G.Hill-Robertson interference is a minor determinant of variations in codon bias across Drosophila melanogaster and Caenorhabditis elegans genomes. Mol Biol Evol. 2002 Sep;19(9):1399-406.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Drosophila_melanogaster)
map &lt;- Drosophila_melanogaster[["2L"]]
plot(map)
names(Drosophila_melanogaster@maps)
</code></pre>

<hr>
<h2 id='geneticDistances-methods'>Accessors for slot geneticDistances</h2><span id='topic+geneticDistances'></span><span id='topic+geneticDistances-methods'></span><span id='topic+geneticDistances+3C-'></span><span id='topic+geneticDistances+3C--methods'></span>

<h3>Description</h3>

<p>Methods for function <code>geneticDistances</code> in package <span class="pkg">MareyMap</span>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>returns or set the value of the slot geneticDistances in MareyMap</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='getInterList-methods'>returns a list of interpolation methods.</h2><span id='topic+getInterList'></span><span id='topic+getInterList-methods'></span><span id='topic+getInterList+2Cmissing-method'></span>

<h3>Description</h3>

<p>returns the list of the interpolation methods present in the interpolation register.
</p>


<h3>Methods</h3>


<dl>
<dt>dummy = &quot;missing&quot;</dt><dd><p>default method.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>

<hr>
<h2 id='Homo_sapiens'>The marey map for the species Homo sapiens (female, male and mean of the two)</h2><span id='topic+Homo_sapiens_female'></span><span id='topic+Homo_sapiens_male'></span><span id='topic+Homo_sapiens_mean'></span>

<h3>Description</h3>

<p>A marey map is an interpolation between genetical and physical maps of a mapomosome. This data file contains marey map for the species Homo sapiens.
This map is held into an object of class MapSet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Homo_sapiens_mean)</code></pre>


<h3>Format</h3>

<p>Homo sapiens (female, male and mean) contains an object of the class <code><a href="#topic+MapSet-class">MapSet-class</a></code>. The maps inside this object (objects of the class <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>) can be accessed using the operator &quot;[[&quot; (see example)
Each of the 3 sets (female, male and mean) contains 22 maps called &quot;Chromosome 01&quot;, ... , &quot;Chromosome 22&quot;.
</p>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>Source</h3>

<p>Rutgers Combined Linkage-Physical Maps, version 2.0 (Build 35). Xiangyang Komg and Tara Matise 12/08/2004
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_female)
map &lt;- Homo_sapiens_female[["Chromosome 01"]]
plot(map)
names(Homo_sapiens_female@maps)
</code></pre>

<hr>
<h2 id='interpolate-methods'>interpolate</h2><span id='topic+interpolate'></span><span id='topic+interpolate-methods'></span>

<h3>Description</h3>

<p>performs an interpolation on <code>map</code> using the interpolation method provided by <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;, map = &quot;MareyMap&quot;</dt><dd><p>DO NOT USE in an 'real' programming language, this would be a virtual function</p>
</dd>
<dt>object = &quot;MMSlidingWindow&quot;, map = &quot;MareyMap&quot;</dt><dd><p>interpolate using sliding windows</p>
</dd>
<dt>object = &quot;MMLoess&quot;, map = &quot;MareyMap&quot;</dt><dd><p>interpolate using loess</p>
</dd>
<dt>object = &quot;MMSpline3&quot;, map = &quot;MareyMap&quot;</dt><dd><p>interpolate using cubic splines.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='Interpolation-class'>Class &quot;Interpolation&quot;</h2><span id='topic+Interpolation'></span><span id='topic+Interpolation-class'></span><span id='topic+interpolate+2CInterpolation+2CMareyMap-method'></span><span id='topic+userParam+2CInterpolation-method'></span><span id='topic+query+2CInterpolation+2Cinteger-method'></span><span id='topic+query+2CInterpolation+2Cnumeric-method'></span><span id='topic+argList+2CInterpolation-method'></span><span id='topic+plot+2CMareyMap+2CInterpolation-method'></span><span id='topic+plotModel+2CInterpolation-method'></span><span id='topic+plotRate+2CInterpolation-method'></span><span id='topic+name+3C-+2CInterpolation-method'></span><span id='topic+name+2CInterpolation-method'></span><span id='topic+rates+3C-+2CInterpolation-method'></span><span id='topic+rates+2CInterpolation-method'></span><span id='topic+visible+3C-+2CInterpolation-method'></span><span id='topic+visible+2CInterpolation-method'></span><span id='topic+color+3C-+2CInterpolation-method'></span><span id='topic+color+2CInterpolation-method'></span><span id='topic+persistent+3C-+2CInterpolation-method'></span><span id='topic+persistent+2CInterpolation-method'></span>

<h3>Description</h3>

<p>This class serves as a base for interpolation methods. 
Although it is not defined as virtual for technical reason, it should not be used as is and is only useful after having been extended.
</p>


<h3>Objects from the Class</h3>

<p>Objects from the class should not be created
</p>


<h3>Slots</h3>


<dl>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> holding the name of the interpolation </p>
</dd>
<dt><code>color</code>:</dt><dd><p>Object of class <code>"character"</code> holding the name or the code (#RRGGBB) of the color of the interpolation on the plot </p>
</dd>
<dt><code>physicalPositions</code>:</dt><dd><p>Object of class <code>"vector"</code> holding the valid physical position of the map on which the interpolation is calculated. This redundancy is necessary to cary out the plotting of the map. </p>
</dd>
<dt><code>rates</code>:</dt><dd><p>Object of class <code>"vector"</code> holding the local recombination rate estimations at the location of each markers. </p>
</dd>
<dt><code>visible</code>:</dt><dd><p>Object of class <code>"logical"</code>, whether or not the interpolation should be plotted </p>
</dd>
<dt><code>persistent</code>:</dt><dd><p>Object of class <code>"logical"</code>, whether or not the interpolation should be saved when the map is saved to text file. </p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>argList</dt><dd><p><code>signature(object = "Interpolation")</code>: This function is used in the function createList (c.f. code examples).</p>
</dd>
<dt>color&lt;-</dt><dd><p><code>signature(object = "Interpolation")</code>: changes the color of the interpolation.</p>
</dd>
<dt>color</dt><dd><p><code>signature(object = "Interpolation")</code>: returns the color of the interpolation.</p>
</dd>
<dt>interpolate</dt><dd><p><code>signature(object = "Interpolation", map = "MareyMap")</code>: recalculate the interpolation.</p>
</dd>
<dt>name&lt;-</dt><dd><p><code>signature(object = "Interpolation")</code>: changes the name of the interpolation.</p>
</dd>
<dt>name</dt><dd><p><code>signature(object = "Interpolation")</code>: returns the name of the interpolation.</p>
</dd>
<dt>persistent&lt;-</dt><dd><p><code>signature(object = "Interpolation")</code>: changes the persistence state of the interpolation.</p>
</dd>
<dt>persistent</dt><dd><p><code>signature(object = "Interpolation")</code>: returns the persistence state of the interpolation.</p>
</dd>
<dt>plotModel</dt><dd><p><code>signature(object = "Interpolation")</code>: this function plots only the model of the marey map calculated by the method.</p>
</dd>
<dt>plotRate</dt><dd><p><code>signature(object = "Interpolation")</code>: this function plots the recombination rates calculated by the method.</p>
</dd>
<dt>query</dt><dd><p><code>signature(object = "Interpolation", pos = "integer")</code>: returns the recombination rate calculated by this method on a given location on the sequence.</p>
</dd>
<dt>rates&lt;-</dt><dd><p><code>signature(object = "Interpolation")</code>: replaces the rates with a new vector.</p>
</dd>
<dt>rates</dt><dd><p><code>signature(object = "Interpolation")</code>: returns the rates of the interpolation.</p>
</dd>
<dt>userParam</dt><dd><p><code>signature(object = "Interpolation")</code>: function returning information about the parameters modifiable by the user (c.f. code example.)</p>
</dd>
<dt>visible&lt;-</dt><dd><p><code>signature(object = "Interpolation")</code>: changes the visibility of the interpolation.</p>
</dd>
<dt>visible</dt><dd><p><code>signature(object = "Interpolation")</code>: returns the visibility of the interpolation</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>
<code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='interpolation-methods'>set or returns an interpolation.</h2><span id='topic+interpolation'></span><span id='topic+interpolation-methods'></span><span id='topic+interpolation+3C-'></span><span id='topic+interpolation+3C--methods'></span>

<h3>Description</h3>

<p>Methods for function <code>interpolation</code> in package <span class="pkg">MareyMap</span>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;, inter_name = &quot;character&quot;</dt><dd><p>set or returns the interpolation <code>inter_name</code> in object <code><a href="#topic+MareyMap-class">MareyMap-class</a></code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='InterpolationParam-class'>Class &quot;InterpolationParam&quot;</h2><span id='topic+InterpolationParam'></span><span id='topic+InterpolationParam-class'></span><span id='topic+paramDefault+3C-+2CInterpolationParam-method'></span><span id='topic+paramDefault+2CInterpolationParam-method'></span><span id='topic+paramDesc+3C-+2CInterpolationParam+2Ccharacter-method'></span><span id='topic+paramDesc+2CInterpolationParam-method'></span><span id='topic+paramFun+3C-+2CInterpolationParam+2Ccharacter-method'></span><span id='topic+paramFun+2CInterpolationParam-method'></span><span id='topic+paramMax+3C-+2CInterpolationParam-method'></span><span id='topic+paramMax+2CInterpolationParam-method'></span><span id='topic+paramMin+3C-+2CInterpolationParam-method'></span><span id='topic+paramMin+2CInterpolationParam-method'></span><span id='topic+paramName+3C-+2CInterpolationParam+2Ccharacter-method'></span><span id='topic+paramName+2CInterpolationParam-method'></span><span id='topic+paramType+3C-+2CInterpolationParam+2Ccharacter-method'></span><span id='topic+paramType+2CInterpolationParam-method'></span><span id='topic+paramValues+3C-+2CInterpolationParam+2Cvector-method'></span><span id='topic+paramValues+2CInterpolationParam-method'></span>

<h3>Description</h3>

<p>Class holding informations about a parameter of an interpolation method. This class is used in the function <code><a href="#topic+userParam+2CInterpolation-method">userParam,Interpolation-method</a></code>. These informations are used by the GUI to dynamically create the dialog to modify this parameter.</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("InterpolationParam", ...)</code> or using the method<code>InterpolationParam()</code>. 
</p>


<h3>Slots</h3>


<dl>
<dt><code>paramName</code>:</dt><dd><p>Object of class <code>"character"</code> holding the name of the parameter.</p>
</dd>
<dt><code>paramType</code>:</dt><dd><p>Object of class <code>"character"</code> holding the type of the parameter, currently supported parameter types are: <code>logical</code>, <code>numeric</code>, <code>integer</code> and <code>color</code>.</p>
</dd>
<dt><code>paramDesc</code>:</dt><dd><p>Object of class <code>"character"</code> giving a small description of the parameter's signification.</p>
</dd>
<dt><code>paramDefault</code>:</dt><dd><p>Object of class <code>"ANY"</code>, default value for the parameter.</p>
</dd>
<dt><code>paramValues</code>:</dt><dd><p>Object of class <code>"vector"</code> containing the set of valid values for the parameter (can be NULL).</p>
</dd>
<dt><code>paramMin</code>:</dt><dd><p>Object of class <code>"ANY"</code>, a minimum value for the parameter, or minimal string length if the parameter is of type <code>character</code></p>
</dd>
<dt><code>paramMax</code>:</dt><dd><p>Object of class <code>"ANY"</code> , a minimum value for the parameter, or maximum string length if the parameter is of type <code>character</code></p>
</dd>
<dt><code>paramFun</code>:</dt><dd><p>Object of class <code>"character"</code>, the nae of function that is to be called to update the parameter's value. ex if paramFun is <code>foo</code> the parameter's value will be set using <code>foo(interpolation)&lt;-new_value</code></p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>paramDefault&lt;-</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramDefault</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramDesc&lt;-</dt><dd><p><code>signature(object = "InterpolationParam", value = "character")</code>: ...</p>
</dd>
<dt>paramDesc</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramFun&lt;-</dt><dd><p><code>signature(object = "InterpolationParam", value = "character")</code>: ...</p>
</dd>
<dt>paramFun</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramMax&lt;-</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramMax</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramMin&lt;-</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramMin</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramName&lt;-</dt><dd><p><code>signature(object = "InterpolationParam", value = "character")</code>: ...</p>
</dd>
<dt>paramName</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramType&lt;-</dt><dd><p><code>signature(object = "InterpolationParam", value = "character")</code>: ...</p>
</dd>
<dt>paramType</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
<dt>paramValues&lt;-</dt><dd><p><code>signature(object = "InterpolationParam", value = "vector")</code>: ...</p>
</dd>
<dt>paramValues</dt><dd><p><code>signature(object = "InterpolationParam")</code>: ...</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  nam &lt;- InterpolationParam()
	paramName(nam) &lt;- "name"
	paramDesc(nam) &lt;- "The name of the interpolation.\n"
	paramType(nam) &lt;- "character"
	paramDefault(nam) &lt;- "default name"
	paramMin(nam) &lt;- 1
	paramMax(nam) &lt;- NULL	
	paramFun(nam) &lt;- "name"

	vis &lt;- InterpolationParam()
	paramName(vis) &lt;- "visible"
	paramDesc(vis) &lt;- "Whether the line is visible on the plot or not"
	paramType(vis) &lt;- "logical"
	paramDefault(vis) &lt;- TRUE
	paramFun(vis) &lt;- "visible"
	
	sav &lt;- InterpolationParam()
	paramName(sav) &lt;- "persistent"
	paramDesc(sav) &lt;- "Indicate if the interpolation is to be be kept \n\
  when the map is saved to text file"
	paramType(sav) &lt;- "logical"
	paramDefault(sav) &lt;- TRUE
	paramFun(sav) &lt;- "persistent"
	
	col &lt;- InterpolationParam()
	paramName(col) &lt;- "line color"
	paramDesc(col) &lt;- "color of the line"
	paramType(col) &lt;- "color"
	paramDefault(col) &lt;- "#000000"
	paramFun(col) &lt;- "color"
</code></pre>

<hr>
<h2 id='InterpolationParam-methods'>Methods for Function InterpolationParam in Package 'MareyMap'</h2><span id='topic+InterpolationParam-methods'></span><span id='topic+InterpolationParam+2Cmissing-method'></span>

<h3>Description</h3>

<p>constructor for class <code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code> in package <span class="pkg">MareyMap</span>
</p>


<h3>Methods</h3>


<dl>
<dt>dummy = &quot;missing&quot;</dt><dd><p>default constructor.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code>link{InterpolationParam-class}</code>
</p>

<hr>
<h2 id='interpolations-methods'>accessors for slot interpolations</h2><span id='topic+interpolations'></span><span id='topic+interpolations-methods'></span><span id='topic+interpolations+3C-'></span><span id='topic+interpolations+3C--methods'></span>

<h3>Description</h3>

<p>set or get the value of the slot <code>interpolation</code> in <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p> set or get slot <code>interpolations</code> in <code><a href="#topic+MareyMap-class">MareyMap-class</a></code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap">MareyMap</a></code>
</p>

<hr>
<h2 id='MapCollection-class'>Class &quot;MapCollection&quot;</h2><span id='topic+MapCollection+2Cmissing-method'></span><span id='topic+MapCollection'></span><span id='topic+MapCollection-class'></span><span id='topic++5B+2CMapCollection-method'></span><span id='topic++5B+5B+3C-+2CMapCollection+2CANY+2CANY+2CMapSet-method'></span><span id='topic++5B+5B+3C-+2CMapCollection+2CANY+2CANY+2CMareyMap-method'></span><span id='topic++5B+5B+2CMapCollection-method'></span><span id='topic++24+2CMapCollection-method'></span><span id='topic++2B+2CMapCollection+2CMareyMap-method'></span><span id='topic++2B+2CMapCollection+2CMapSet-method'></span><span id='topic++2B+2CMapCollection+2CMapCollection-method'></span><span id='topic+-+2CMapCollection+2Ccharacter-method'></span><span id='topic+coerce+2CMapCollection+2Cdata.frame-method'></span><span id='topic+setNames+2CMapCollection-method'></span><span id='topic+length+2CMapCollection-method'></span><span id='topic+textFile+2CMapCollection+2Ccharacter-method'></span>

<h3>Description</h3>

<p>This class holds several <code><a href="#topic+MareyMap-class">MareyMap-class</a></code> objects organised in sets (using <code><a href="#topic+MapSet-class">MapSet-class</a></code> object).</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("MapCollection", ...)</code> or using the function MapCollection. You may as well use the default collection provided by the package (as <code>data()</code>). 
</p>


<h3>Slots</h3>


<dl>
<dt><code>sets</code>:</dt><dd><p>Object of class <code>"list"</code> containing a <code><a href="#topic+MapSet-class">MapSet-class</a></code> for each set$* of the collection.</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>[</dt><dd><p><code>signature(x = "MapCollection")</code>: ...</p>
</dd>
<dt>[[&lt;-</dt><dd><p><code>signature(x = "MapCollection", i = "ANY", j = "ANY", value = "MapSet")</code>: ...</p>
</dd>
<dt>[[&lt;-</dt><dd><p><code>signature(x = "MapCollection", i = "ANY", j = "ANY", value = "MareyMap")</code>: ...</p>
</dd>
<dt>[[</dt><dd><p><code>signature(x = "MapCollection")</code>: obtain the MapSet corresponding to a set name</p>
</dd>
<dt>$</dt><dd><p><code>signature(x = "MapCollection")</code>: subset by set name</p>
</dd>
<dt>+</dt><dd><p><code>signature(e1 = "MapCollection", e2 = "MareyMap")</code>: adds a map to the collection</p>
</dd>
<dt>+</dt><dd><p><code>signature(e1 = "MapCollection", e2 = "MapSet")</code>: adds a set t the collection</p>
</dd>
<dt>-</dt><dd><p><code>signature(e1 = "MapCollection", e2 = "character")</code>: removes the map named e2 from the object e1.</p>
</dd>
<dt>coerce</dt><dd><p><code>signature(from = "MapCollection", to = "data.frame")</code>: <code>--&gt; as(object,"data.frame")</code></p>
</dd>
<dt>setNames</dt><dd><p><code>signature(object = "MapCollection")</code>: returns the list of the names of all the sets present in the collection</p>
</dd>
<dt>length</dt><dd><p><code>signature(object = "MapCollection")</code>: returns the length of the collection</p>
</dd>
<dt>textFile</dt><dd><p><code>signature(object = "MapCollection", file = "character")</code>: Writes the content of the collection to text file <code>text</code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MapSet-class">MapSet-class</a></code>
<code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_mean)
map1 &lt;- Homo_sapiens_mean[["Chromosome 01"]]
# do some useful changes to the map...
mapName(map1)&lt;- "Chrom 1 (modified)"
</code></pre>

<hr>
<h2 id='MapCollection-methods'>Constructor for class MapCollection</h2><span id='topic+MapCollection-methods'></span>

<h3>Description</h3>

<p>Creates an object of class <code><a href="#topic+MapCollection">MapCollection</a></code>
</p>


<h3>Methods</h3>


<dl>
<dt>x = &quot;missing&quot;</dt><dd><p>Default constructor.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MapCollection-class">MapCollection-class</a></code>
</p>

<hr>
<h2 id='mapName-methods'> accessors for the slot mapName</h2><span id='topic+mapName'></span><span id='topic+mapName-methods'></span><span id='topic+mapName+3C-'></span><span id='topic+mapName+3C--methods'></span>

<h3>Description</h3>

<p>Set or get the value of the slot <code>mapName</code> in <code>object</code>. 
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>  Set or get the value of the slot <code>mapName</code> in <code>object</code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='mapNames-methods'>mapNames</h2><span id='topic+mapNames'></span><span id='topic+mapNames-methods'></span>

<h3>Description</h3>

<p>Returns the names of the maps contained in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MapSet&quot;</dt><dd><p>Returns the names of the maps contained in <code>MapSet</code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MapSet">MapSet</a></code>
</p>

<hr>
<h2 id='MapSet-class'>Class &quot;MapSet&quot;</h2><span id='topic+MapSet'></span><span id='topic+MapSet-class'></span><span id='topic+-+2CMapSet+2Ccharacter-method'></span><span id='topic++5B+2CMapSet-method'></span><span id='topic++5B+5B+3C-+2CMapSet+2CANY+2CANY+2CMareyMap-method'></span><span id='topic++5B+5B+2CMapSet-method'></span><span id='topic++24+2CMapSet-method'></span><span id='topic++2B+2CMapSet+2CMareyMap-method'></span><span id='topic+coerce+2CMapSet+2Cdata.frame-method'></span><span id='topic+mapNames+2CMapSet-method'></span><span id='topic+setName+3C-+2CMapSet+2Ccharacter-method'></span><span id='topic+setName+2CMapSet-method'></span><span id='topic+length+2CMapSet-method'></span><span id='topic+textFile+2CMapSet+2Ccharacter-method'></span>

<h3>Description</h3>

<p>This class is a collection of MareyMap for a given set. Not that useful on its own, it is used for internal organization inside a MapCollection</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("MapSet", ...)</code>. Or using the function <code>MapSet()</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>maps</code>:</dt><dd><p>Object of class <code>"list"</code> containing all the <code>MareyMap</code> of the set.</p>
</dd>
<dt><code>setName</code>:</dt><dd><p>Object of class <code>"character"</code> containing the name of the set</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>[</dt><dd><p><code>signature(x = "MapSet")</code>: returns a map from the list  </p>
</dd>
<dt>[[&lt;-</dt><dd><p><code>signature(x = "MapSet", i = "ANY", j = "ANY", value = "MareyMap")</code>: replaces a map in the list with a new one </p>
</dd>
<dt>[[</dt><dd><p><code>signature(x = "MapSet")</code>: returns a map from the list.  </p>
</dd>
<dt>$</dt><dd><p><code>signature(x = "MapSet")</code>: access a map by name in the list. </p>
</dd>
<dt>+</dt><dd><p><code>signature(e1 = "MapSet", e2 = "MareyMap")</code>: adds a map to the collection. </p>
</dd>
<dt>-</dt><dd><p><code>signature(e1 = "MapSet", e2 = "character")</code>: removes the map named e2 from the object e1.</p>
</dd>
<dt>coerce</dt><dd><p><code>signature(from = "MapSet", to = "data.frame")</code>: converts the specieMaps object into a data.frame. </p>
</dd>
<dt>mapNames</dt><dd><p><code>signature(object = "MapSet")</code>: returns a vector containing the names of all the maps. </p>
</dd>
<dt>setName&lt;-</dt><dd><p><code>signature(object = "MapSet", value = "character")</code>: changes the name of the set. Also changes the named of all MareyMap objects inside the list. </p>
</dd>
<dt>setName</dt><dd><p><code>signature(object = "MapSet")</code>: returns the name of the set. </p>
</dd>
<dt>length</dt><dd><p><code>signature(object = "MapCollection")</code>: returns the length of the set</p>
</dd>
<dt>textFile</dt><dd><p><code>signature(object = "MapSet", file = "character")</code>: Writes the content of the MapSet object to file. </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MapCollection-class">MapCollection-class</a></code>
<code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_mean)
set &lt;- Homo_sapiens_mean
setName(set) &lt;- "Human"
set[["Chromosome 01"]]
</code></pre>

<hr>
<h2 id='MapSet-methods'> Constructor for the class MapSet.</h2><span id='topic+MapSet-methods'></span><span id='topic+MapSet+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Creates an object of the class <code><a href="#topic+MapSet">MapSet</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt>set_name = &quot;character&quot;</dt><dd><p> creates a new object of the class <code><a href="#topic+MapSet">MapSet</a></code> with the name set_name. </p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>

<hr>
<h2 id='MareyMap-class'>Class &quot;MareyMap&quot; </h2><span id='topic+MareyMap'></span><span id='topic+MareyMap+2Cdata.frame+2CANY+2CANY+2CANY-method'></span><span id='topic+MareyMap-class'></span><span id='topic++5B+2CMareyMap-method'></span><span id='topic++5B+5B+3C-+2CMareyMap-method'></span><span id='topic++5B+5B+3C-+2CMareyMap+2CANY+2CANY+2CANY-method'></span><span id='topic++5B+5B+2CMareyMap-method'></span><span id='topic++2B+2CMareyMap+2CInterpolation-method'></span><span id='topic+coerce+2CMareyMap+2Cdata.frame-method'></span><span id='topic+coerce+2Cdata.frame+2CMareyMap-method'></span><span id='topic+coerce+2CMareyMap+2CNULL-method'></span><span id='topic+geneticDistances+3C-+2CMareyMap-method'></span><span id='topic+geneticDistances+2CMareyMap-method'></span><span id='topic+interpolation+3C-+2CMareyMap+2Ccharacter+2CInterpolation-method'></span><span id='topic+interpolation+2CMareyMap+2Ccharacter-method'></span><span id='topic+interpolations+3C-+2CMareyMap-method'></span><span id='topic+interpolations+2CMareyMap-method'></span><span id='topic+mapName+3C-+2CMareyMap-method'></span><span id='topic+mapName+2CMareyMap-method'></span><span id='topic+markerNames+3C-+2CMareyMap-method'></span><span id='topic+markerNames+2CMareyMap-method'></span><span id='topic+markerValidity+3C-+2CMareyMap-method'></span><span id='topic+markerValidity+2CMareyMap-method'></span><span id='topic+physicalPositions+3C-+2CMareyMap-method'></span><span id='topic+physicalPositions+2CMareyMap-method'></span><span id='topic+plot+2CMareyMap+2Cmissing-method'></span><span id='topic+plotMarkers+2CMareyMap-method'></span><span id='topic+plotModels+2CMareyMap-method'></span><span id='topic+plotRates+2CMareyMap-method'></span><span id='topic+query+2CMareyMap+2Cnumeric-method'></span><span id='topic+removeMarker+2CMareyMap+2Cinteger-method'></span><span id='topic+setName+3C-+2CMareyMap+2CANY-method'></span><span id='topic+setName+2CMareyMap-method'></span><span id='topic+textFile+2CMareyMap+2Ccharacter-method'></span><span id='topic+validPositions+2CMareyMap-method'></span>

<h3>Description</h3>

<p> The class holds the physical and genetic positions of a marey map</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("MareyMap", ...)</code> or using the function <code>MareyMap()</code>.
However, most of the time you can more conveniently use already constructed objects contained in (c.f. code examples)
</p>


<h3>Slots</h3>


<dl>
<dt><code>setName</code>:</dt><dd><p>Object of class <code>"character"</code>  </p>
</dd>
<dt><code>mapName</code>:</dt><dd><p>Object of class <code>"character"</code> containing the name of the map, usually the name of the mapomosome it is mapping. </p>
</dd>
<dt><code>markerNames</code>:</dt><dd><p>Object of class <code>"vector"</code> containing the names of the markers of the map. </p>
</dd>
<dt><code>physicalPositions</code>:</dt><dd><p>Object of class <code>"vector"</code> containing the physical positions of the markers. </p>
</dd>
<dt><code>geneticDistances</code>:</dt><dd><p>Object of class <code>"vector"</code> containing the positions of the markers on the genetic map. </p>
</dd>
<dt><code>markerValidity</code>:</dt><dd><p>Object of class <code>"vector"</code> indicating for each marker if it is to be taken into account for interpolations. </p>
</dd>
<dt><code>interpolations</code>:</dt><dd><p>Object of class <code>"list"</code> containing the <code>interpolation</code>s calculated on the map. </p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt>[</dt><dd><p><code>signature(x = "MareyMap")</code>: returns a vector combining marker name, physical position, genetic distance and marker validity for the i th marker </p>
</dd>
<dt>[[&lt;-</dt><dd><p><code>signature(x = "MareyMap")</code>: replaces marker name, physical position, genetic distance and marker validity for the i th marker. value must be a vector containing 4 column named &quot;name&quot;, &quot;phys&quot;, &quot;gen&quot; and &quot;valid&quot;. </p>
</dd>
<dt>[[&lt;-</dt><dd><p><code>signature(x = "MareyMap", i = "ANY", j = "ANY", value = "ANY")</code>: replaces marker name, physical position, genetic distance and marker validity for the i th marker. value must be a vector containing 4 column named &quot;name&quot;, &quot;phys&quot;, &quot;gen&quot; and &quot;valid&quot;. </p>
</dd>
<dt>[[</dt><dd><p><code>signature(x = "MareyMap")</code>: returns a vector combining marker name, physical position, genetic distance and marker validity for the i th marker </p>
</dd>
<dt>+</dt><dd><p><code>signature(e1 = "MareyMap", e2 = "Interpolation")</code>: adds an interpolation to the map </p>
</dd>
<dt>coerce</dt><dd><p><code>signature(from = "MareyMap", to = "data.frame")</code>: convert the Marey map into a data.frame. </p>
</dd>
<dt>coerce</dt><dd><p><code>signature(from = "MareyMap", to = "NULL")</code>: returns NULL </p>
</dd>
<dt>coerce</dt><dd><p><code>signature(from = "data.frame", to = "MareyMap")</code>: create a MareyMap from a data.frame. </p>
</dd>
<dt>geneticDistances&lt;-</dt><dd><p><code>signature(object = "MareyMap")</code>: replace the genetic distance with a new vector. </p>
</dd>
<dt>geneticDistances</dt><dd><p><code>signature(object = "MareyMap")</code>: returns the genetic distances of the markers </p>
</dd>
<dt>interpolation&lt;-</dt><dd><p><code>signature(object = "MareyMap", inter_name = "character", value = "Interpolation")</code>:  replaces the interpolation which name matches <code>inter_name</code> with the content of the parameter <code>value</code>.</p>
</dd>
<dt>interpolation</dt><dd><p><code>signature(object = "MareyMap", inter_name = "character")</code>: returns the interpolation which name matches <code>inter_name</code>. </p>
</dd>
<dt>interpolations&lt;-</dt><dd><p><code>signature(object = "MareyMap")</code>: replace the list of interpolation with a new list. </p>
</dd>
<dt>interpolations</dt><dd><p><code>signature(object = "MareyMap")</code>: returns the list of interpolations. </p>
</dd>
<dt>mapName&lt;-</dt><dd><p><code>signature(object = "MareyMap")</code>: replaces the name of the map </p>
</dd>
<dt>mapName</dt><dd><p><code>signature(object = "MareyMap")</code>: returns the name of the map </p>
</dd>
<dt>markerNames&lt;-</dt><dd><p><code>signature(object = "MareyMap")</code>: replaces the marker names with a new vector of  names</p>
</dd>
<dt>markerNames</dt><dd><p><code>signature(object = "MareyMap")</code>: returns a vector containing the names of the markers</p>
</dd>
<dt>markerValidity&lt;-</dt><dd><p><code>signature(object = "MareyMap")</code>: replaces the marker validity with a new vector</p>
</dd>
<dt>markerValidity</dt><dd><p><code>signature(object = "MareyMap")</code>: returns a vector containing information about the validity of the markers</p>
</dd>
<dt>physicalPositions&lt;-</dt><dd><p><code>signature(object = "MareyMap")</code>: replaces the vector of physical positions with a new vector</p>
</dd>
<dt>physicalPositions</dt><dd><p><code>signature(object = "MareyMap")</code>: returns a vector containing the physical positions of the markers</p>
</dd>
<dt>plot</dt><dd><p><code>signature(x = "MareyMap", y = "missing")</code>: plots the map as well as the interpolations (if any)</p>
</dd>
<dt>plotMarkers</dt><dd><p><code>signature(object = "MareyMap")</code>: plots only the markers of the map, without plotting the interpolations</p>
</dd>
<dt>plotModels</dt><dd><p><code>signature(object = "MareyMap")</code>: plot only the models of the interpolations</p>
</dd>
<dt>plotRates</dt><dd><p><code>signature(object = "MareyMap")</code>: plots only the recombination rates</p>
</dd>
<dt>query</dt><dd><p><code>signature(object = "MareyMap", pos = "numeric")</code>: returns the value of the local  recombination rate at physical position <code>pos</code> as estimated by the interpolations defined on the map. Returns a list of integers (or NAs). each individual result may be accessed via result[[interpolation_name]]</p>
</dd>
<dt>removeMarker</dt><dd><p><code>signature(object = "MareyMap", value = "integer")</code>: remove the marker number <code>value</code>.</p>
</dd>
<dt>setName&lt;-</dt><dd><p><code>signature(object = "MareyMap", value = "ANY")</code>: replaces the name of the set with a new <code>character</code>.</p>
</dd>
<dt>setName</dt><dd><p><code>signature(object = "MareyMap")</code>: returns the name of the set.</p>
</dd>
<dt>validPositions</dt><dd><p><code>signature(object = "MareyMap")</code>:</p>
</dd>
<dt>textFile</dt><dd><p><code>signature(object = "MareyMap", file = "character")</code>: Writes the map to text file.</p>
</dd>
</dl>



<h3>Warning</h3>

<p>vectors <code>markerNames</code>, <code>physicalPositions</code>, <code>geneticDistances</code> and <code>markerValidity</code> must always retain the same length.</p>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MapCollection-class">MapCollection-class</a></code>
<code><a href="#topic+MapSet-class">MapSet-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_male)
chr7 &lt;- Homo_sapiens_male[["Chromosome 07"]]
chr7 &lt;- chr7 + MMLoess()
par(mfrow = c(2, 1))
plotMarkers(chr7)
</code></pre>

<hr>
<h2 id='MareyMap-methods'>Constructor for object of the class MareyMap.</h2><span id='topic+MareyMap-methods'></span><span id='topic+MareyMap+2Cmissing+2Cmissing+2Cmissing+2Cmissing-method'></span>

<h3>Description</h3>

<p>Creates an object of the class <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>.
</p>


<h3>Methods</h3>


<dl>
<dt>data_table = &quot;missing&quot;, column_names = &quot;missing&quot;, set_name = &quot;missing&quot;, map_name = &quot;missing&quot;</dt><dd><p>Default constructor</p>
</dd>
<dt>data_table = &quot;data.frame&quot;</dt><dd><p>fill the class using the provided data.frame</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='markerNames-methods'>Accessors for the class markerNames.</h2><span id='topic+markerNames'></span><span id='topic+markerNames-methods'></span><span id='topic+markerNames+3C-'></span><span id='topic+markerNames+3C--methods'></span>

<h3>Description</h3>

<p>Get or Set slot <code>markerNames</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>Get or Set slot <code>markerNames</code> in <code><a href="#topic+MareyMap">MareyMap</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='markerValidity-methods'>accessors for slot markerValidity</h2><span id='topic+markerValidity'></span><span id='topic+markerValidity-methods'></span><span id='topic+markerValidity+3C-'></span><span id='topic+markerValidity+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>MarkerValidity</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>Get or set the value of the slot <code>MarkerValidity</code> in <code><a href="#topic+MareyMap">MareyMap</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='MMLoess-class'>Class &quot;MMLoess&quot;. Interpolation method plug-in for MareyMap </h2><span id='topic+MMLoess-methods'></span><span id='topic+MMLoess+2Cmissing-method'></span><span id='topic+MMLoess'></span><span id='topic+MMLoess-class'></span><span id='topic+createOrder+2CMMLoess-method'></span><span id='topic+degree+3C-'></span><span id='topic+degree+3C-+2CMMLoess-method'></span><span id='topic+degree'></span><span id='topic+degree+2CMMLoess-method'></span><span id='topic+interpolate+2CMMLoess+2CMareyMap-method'></span><span id='topic+plot+2CMareyMap+2CMMLoess-method'></span><span id='topic+plotModel+2CMMLoess-method'></span><span id='topic+plotRate+2CMMLoess-method'></span><span id='topic+query+2CMMLoess+2Cnumeric-method'></span><span id='topic+span+3C-'></span><span id='topic+span+3C-+2CMMLoess-method'></span><span id='topic+span'></span><span id='topic+span+2CMMLoess-method'></span><span id='topic+userParam+2CMMLoess-method'></span>

<h3>Description</h3>

<p> This packages provides an interpolation method that estimates the recombination rates using <code><a href="stats.html#topic+loess">loess</a></code>. 
For more details on how this method works, see <code><a href="stats.html#topic+loess">loess</a></code>.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("MMLoess", ...)</code> or simply by calling <code>MMLoess()</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>span</code>:</dt><dd><p>Object of class <code>"numeric"</code>, parameters that controls the degree of smoothing. </p>
</dd>
<dt><code>degree</code>:</dt><dd><p>Object of class <code>"integer"</code>, the degree of the polynomials to be used, up to 2. </p>
</dd>
<dt><code>model</code>:</dt><dd><p>Object of class <code>"ANY"</code>  </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code>, name of the interpolation. </p>
</dd>
<dt><code>color</code>:</dt><dd><p>Object of class <code>"character"</code>, color of the interpolation on the map. </p>
</dd>
<dt><code>physicalPositions</code>:</dt><dd><p>Object of class <code>"vector"</code>, physical positions of the markers of the map on which the interpolation is calculated. </p>
</dd>
<dt><code>rates</code>:</dt><dd><p>Object of class <code>"vector"</code>, the local estimation of the recombination rate at the position of each markers. </p>
</dd>
<dt><code>visible</code>:</dt><dd><p>Object of class <code>"logical"</code>, whether the interpolation should be plotted or not. </p>
</dd>
<dt><code>persistent</code>:</dt><dd><p>Object of class <code>"logical"</code>, whether the interpolation should be saved along with the map in the text file </p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"Interpolation"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>createOrder</dt><dd><p><code>signature(object = "MMLoess")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code>.  </p>
</dd>
<dt>degree&lt;-</dt><dd><p><code>signature(object = "MMLoess")</code>: changes the degree of the polynomials. </p>
</dd>
<dt>degree</dt><dd><p><code>signature(object = "MMLoess")</code>: returns the degree of the polynomials. </p>
</dd>
<dt>interpolate</dt><dd><p><code>signature(object = "MMLoess", map = "MareyMap")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code>. </p>
</dd>
<dt>plotModel</dt><dd><p><code>signature(object = "MMLoess")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code>. </p>
</dd>
<dt>plotRate</dt><dd><p><code>signature(object = "MMLoess")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code>. </p>
</dd>
<dt>query</dt><dd><p><code>signature(object = "MMLoess", pos = "numeric")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code>. </p>
</dd>
<dt>span&lt;-</dt><dd><p><code>signature(object = "MMLoess")</code>: changes the value of the span. </p>
</dd>
<dt>span</dt><dd><p><code>signature(object = "MMLoess")</code>: returns the value of the span. </p>
</dd>
<dt>userParam</dt><dd><p><code>signature(object = "MMLoess")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code>. </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+loess">loess</a></code>
<code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_male)
chr7 &lt;- Homo_sapiens_male[["Chromosome 07"]]
itr1 &lt;- MMLoess()
color(itr1) &lt;- "red"
span(itr1) &lt;- 0.05
chr7 &lt;- chr7 + itr1
</code></pre>

<hr>
<h2 id='MMSlidingWindow-class'>Class &quot;MMSlidingWindow&quot; Interpolation method plug-in for MareyMap </h2><span id='topic+MMSlidingWindow-methods'></span><span id='topic+MMSlidingWindow+2Cmissing-method'></span><span id='topic+MMSlidingWindow'></span><span id='topic+MMSlidingWindow-class'></span><span id='topic+createOrder+2CMMSlidingWindow-method'></span><span id='topic+interpolate+2CMMSlidingWindow+2CMareyMap-method'></span><span id='topic+plot+2CMareyMap+2CMMSlidingWindow-method'></span><span id='topic+plotModel+2CMMSlidingWindow-method'></span><span id='topic+plotRate+2CMMSlidingWindow-method'></span><span id='topic+query+2CMMSlidingWindow+2Cnumeric-method'></span><span id='topic+shift+3C-'></span><span id='topic+shift+3C-+2CMMSlidingWindow-method'></span><span id='topic+shift'></span><span id='topic+shift+2CMMSlidingWindow-method'></span><span id='topic+size+3C-'></span><span id='topic+size+3C-+2CMMSlidingWindow-method'></span><span id='topic+size'></span><span id='topic+size+2CMMSlidingWindow-method'></span><span id='topic+threshold+3C-'></span><span id='topic+threshold+3C-+2CMMSlidingWindow-method'></span><span id='topic+threshold'></span><span id='topic+threshold+2CMMSlidingWindow-method'></span><span id='topic+userParam+2CMMSlidingWindow-method'></span>

<h3>Description</h3>

<p> This package provides an interpolation method that estimates the recombination rates using a sliding window approach.  </p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("MMSlidingWindow", ...)</code> or using the method <code>SlidingWindow()</code>.
</p>


<h3>Slots</h3>


<dl>
<dt><code>size</code>:</dt><dd><p>Object of class <code>"integer"</code>, the size of the windows in base pairs </p>
</dd>
<dt><code>shift</code>:</dt><dd><p>Object of class <code>"integer"</code>, the distance in number of base pairs between the centers of two consecutive windows. </p>
</dd>
<dt><code>threshold</code>:</dt><dd><p>Object of class <code>"integer"</code> minimum number of markers in a window to perform the regression </p>
</dd>
<dt><code>results</code>:</dt><dd><p>Object of class <code>"vector"</code> position of the model curve estimated at the center f each window. </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code> see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt><code>color</code>:</dt><dd><p>Object of class <code>"character"</code> see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt><code>physicalPositions</code>:</dt><dd><p>Object of class <code>"vector"</code> see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt><code>rates</code>:</dt><dd><p>Object of class <code>"vector"</code> see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt><code>visible</code>:</dt><dd><p>Object of class <code>"logical"</code> see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt><code>persistent</code>:</dt><dd><p>Object of class <code>"logical"</code> see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt><code>geneticalDistances</code>:</dt><dd></dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"Interpolation"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>createOrder</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt>interpolate</dt><dd><p><code>signature(object = "MMSlidingWindow", map = "MareyMap")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt>plotModel</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt>plotRate</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt>query</dt><dd><p><code>signature(object = "MMSlidingWindow", pos = "numeric")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
<dt>shift&lt;-</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: updates the value of the parameter <code>shift</code> </p>
</dd>
<dt>shift</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: returns the value of the parameter <code>shift</code> </p>
</dd>
<dt>size&lt;-</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: updates the value of the parameter <code>size</code> </p>
</dd>
<dt>size</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: returns the value of the parameter <code>size</code> </p>
</dd>
<dt>threshold&lt;-</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: updates the value of the parameter <code>threshold</code> </p>
</dd>
<dt>threshold</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: returns the value of the parameter <code>threshold</code> </p>
</dd>
<dt>userParam</dt><dd><p><code>signature(object = "MMSlidingWindow")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code> </p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_mean)
human7 &lt;- Homo_sapiens_mean[["Chromosome 07"]]
itr1 &lt;- MMSlidingWindow()
color(itr1) &lt;- "red"
size(itr1) &lt;- 5000000
shift(itr1) &lt;- 1000000
human7 &lt;- human7 + itr1
plot(human7)
</code></pre>

<hr>
<h2 id='MMSpline3-class'>Class &quot;MMSpline3&quot; Interpolation using cubic splines </h2><span id='topic+MMSpline3-methods'></span><span id='topic+MMSpline3+2Cmissing-method'></span><span id='topic+MMSpline3'></span><span id='topic+MMSpline3-class'></span><span id='topic+df+3C-'></span><span id='topic+df+3C-+2CMMSpline3+2Cnumeric-method'></span><span id='topic+df'></span><span id='topic+df+2CMMSpline3-method'></span><span id='topic+gcv+3C-'></span><span id='topic+gcv+3C-+2CMMSpline3+2Clogical-method'></span><span id='topic+gcv'></span><span id='topic+gcv+2CMMSpline3-method'></span><span id='topic+createOrder+2CMMSpline3-method'></span><span id='topic+interpolate+2CMMSpline3+2CMareyMap-method'></span><span id='topic+plot+2CMareyMap+2CMMSpline3-method'></span><span id='topic+plotModel+2CMMSpline3-method'></span><span id='topic+plotRate+2CMMSpline3-method'></span><span id='topic+query+2CMMSpline3+2Cnumeric-method'></span><span id='topic+spar+3C-'></span><span id='topic+spar+3C-+2CMMSpline3+2Cnumeric-method'></span><span id='topic+spar'></span><span id='topic+spar+2CMMSpline3-method'></span><span id='topic+type+3C-'></span><span id='topic+type+3C-+2CMMSpline3+2Ccharacter-method'></span><span id='topic+type'></span><span id='topic+type+2CMMSpline3-method'></span><span id='topic+userParam+2CMMSpline3-method'></span>

<h3>Description</h3>

<p>MMSPline3 is a S4 class providing an interpolation method for MareyMap. Interpolation is done using cubic splines. The class uses
smooth.spline to carry out the interpolation.
</p>


<h3>Objects from the Class</h3>

<p>Objects can be created by calls of the form <code>new("MMSpline3", ...)</code>.
However you may instead use the function <code><a href="#topic+MMSpline3">MMSpline3</a></code>() and adjust
the parameters afterwards using the accessor functions.
</p>


<h3>Slots</h3>


<dl>
<dt><code>type</code>:</dt><dd><p>Object of class <code>"character"</code> indicate which type of smoothing is applied, applicable values are &quot;cross-validation&quot;, &quot;spar&quot; and	&quot;degree of freedom&quot;</p>
</dd>
<dt><code>gcv</code>:</dt><dd><p>Object of class <code>"logical"</code> indicating whether	generalized cross validation is to be used when smoothing via cross	validation. </p>
</dd>
<dt><code>df</code>:</dt><dd><p>Object of class <code>"numeric"</code> holding the degree of freedom to	use when the smoothing is done using degree of freedom</p>
</dd>
<dt><code>spar</code>:</dt><dd><p>Object of class <code>"numeric"</code> value of the &quot;spar&quot; to be taken when smoothing the &quot;spar&quot; method.</p>
</dd>
<dt><code>model</code>:</dt><dd><p>Object of class <code>"ANY"</code> slot used to keep the	smooth.spline object after the interpolation. </p>
</dd>
<dt><code>name</code>:</dt><dd><p>Object of class <code>"character"</code>  holding the name of	the interpolation.</p>
</dd>
<dt><code>color</code>:</dt><dd><p>Object of class <code>"character"</code> holding the color of the interpolation</p>
</dd>
<dt><code>physicalPositions</code>:</dt><dd><p>Object of class <code>"vector"</code> holding the valid physical positions of the map on which the interpolation is applied.</p>
</dd>
<dt><code>rates</code>:</dt><dd><p>Object of class <code>"vector"</code> holding the local recombination rates at the position of each marker.</p>
</dd>
<dt><code>visible</code>:</dt><dd><p>Object of class <code>"logical"</code> indicating whether the interpolation is to be drawn or not. </p>
</dd>
<dt><code>persistent</code>:</dt><dd><p>Object of class <code>"logical"</code> indicating whether or not the interpolation should be taken into account when saving to	text file.</p>
</dd>
</dl>



<h3>Extends</h3>

<p>Class <code>"Interpolation"</code>, directly.
</p>


<h3>Methods</h3>


<dl>
<dt>df&lt;-</dt><dd><p><code>signature(object = "MMSpline3", value = "numeric")</code>: Changes the value of the degree of freedom.</p>
</dd>
<dt>df</dt><dd><p><code>signature(object = "MMSpline3")</code>: returns the value of the degree of freedom.</p>
</dd>
<dt>gcv&lt;-</dt><dd><p><code>signature(object = "MMSpline3", value = "logical")</code>: Turns on or off the generalized cross validation.</p>
</dd>
<dt>gcv</dt><dd><p><code>signature(object = "MMSpline3")</code>: indicates whether the generalized cross validation is on or not.</p>
</dd>
<dt>createOrder</dt><dd><p><code>signature(object = "MMSpline3")</code>: see <code><a href="#topic+Interpolation-class">Interpolation-class</a></code></p>
</dd>
<dt>interpolate</dt><dd><p><code>signature(object = "MMSpline3", map = "MareyMap")</code>: carries out the interpolation on the map passed as parameter. Called automatically when an interpolation is added to a map </p>
</dd>
<dt>plotModel</dt><dd><p><code>signature(object = "MMSpline3")</code>: Plot the interpolation's model of the Marey curve.</p>
</dd>
<dt>plotRate</dt><dd><p><code>signature(object = "MMSpline3")</code>: Plots the variations of the recombination rates across a mapomosome.</p>
</dd>
<dt>query</dt><dd><p><code>signature(object = "MMSpline3", pos = "numeric")</code>: query	the estimation of the recombination rate in a specific position (or a	vector of specific positions)</p>
</dd>
<dt>spar&lt;-</dt><dd><p><code>signature(object = "MMSpline3", value = "numeric")</code>: Changes the value of the spar.</p>
</dd>
<dt>spar</dt><dd><p><code>signature(object = "MMSpline3")</code>: returns the value of	the spar.</p>
</dd>
<dt>type&lt;-</dt><dd><p><code>signature(object = "MMSpline3", value = "character")</code>: Changes the type of smoothing to be used.</p>
</dd>
<dt>type</dt><dd><p><code>signature(object = "MMSpline3")</code>: returns the smoothing	method used.</p>
</dd>
<dt>userParam</dt><dd><p><code>signature(object = "MMSpline3")</code>: returns informations about the user parameter, used by tcl/tk interface but not that	useful for a command line user.</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+smooth.spline">smooth.spline</a></code>
<code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
<code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Homo_sapiens_mean)
human7 &lt;- Homo_sapiens_mean[["Chromosome 07"]]
itr1 &lt;- MMSpline3()
color(itr1) &lt;- "red"
spar(itr1) &lt;- 0.05
human7 &lt;- human7 + itr1
plot(human7)
</code></pre>

<hr>
<h2 id='name-methods'>Accessor for slot name.</h2><span id='topic+name'></span><span id='topic+name-methods'></span><span id='topic+name+3C-'></span><span id='topic+name+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>name</code> in <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>Get or set the value of the slot <code>name</code> in <code>Interpolation</code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation">Interpolation</a></code>
</p>

<hr>
<h2 id='paramDefault-methods'>Accessors for slot paramDefault</h2><span id='topic+paramDefault'></span><span id='topic+paramDefault-methods'></span><span id='topic+paramDefault+3C-'></span><span id='topic+paramDefault+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>paramDefault</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p>Get or set the value of the slot <code>paramDefault</code> in <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam">InterpolationParam</a></code>
</p>

<hr>
<h2 id='paramDesc-methods'>Accessors for the slot paramDesc</h2><span id='topic+paramDesc'></span><span id='topic+paramDesc-methods'></span><span id='topic+paramDesc+3C-'></span><span id='topic+paramDesc+3C--methods'></span>

<h3>Description</h3>

<p>Accessors for the slot <code>paramDesc</code> in <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p>Accessors for the slot <code>paramDesc</code> in <code><a href="#topic+InterpolationParam">InterpolationParam</a></code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam">InterpolationParam</a></code>
</p>

<hr>
<h2 id='paramFun-methods'> accessors for the slot paramFun</h2><span id='topic+paramFun'></span><span id='topic+paramFun-methods'></span><span id='topic+paramFun+3C-'></span><span id='topic+paramFun+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>paramFun</code> in <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p>Get or set the value of the slot <code>paramFun</code> in <code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam">InterpolationParam</a></code>
</p>

<hr>
<h2 id='paramMax-methods'>Accessors for the slot paramMax.</h2><span id='topic+paramMax'></span><span id='topic+paramMax-methods'></span><span id='topic+paramMax+3C-'></span><span id='topic+paramMax+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>paramMax</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p>Get or set the value of the slot <code>paramMax</code> in <code><a href="#topic+InterpolationParam">InterpolationParam</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam">InterpolationParam</a></code>
</p>

<hr>
<h2 id='paramMin-methods'>Accessors for slot paramMin.</h2><span id='topic+paramMin'></span><span id='topic+paramMin-methods'></span><span id='topic+paramMin+3C-'></span><span id='topic+paramMin+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the parameter <code>paramMin</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p>Get or set the value of the parameter <code>paramMin</code> in <code>InterpolationParam</code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>
</p>

<hr>
<h2 id='paramName-methods'>Accessors for the slot paramName.</h2><span id='topic+paramName'></span><span id='topic+paramName-methods'></span><span id='topic+paramName+3C-'></span><span id='topic+paramName+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>paramName</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p> Get or set the value of the slot <code>paramName</code> in <code><a href="#topic+InterpolationParam">InterpolationParam</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>
</p>

<hr>
<h2 id='paramType-methods'>Accessors for the slot paramType.</h2><span id='topic+paramType'></span><span id='topic+paramType-methods'></span><span id='topic+paramType+3C-'></span><span id='topic+paramType+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>paramType</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p> Get or set the value of the slot <code>paramType</code> in <code><a href="#topic+InterpolationParam">InterpolationParam</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>
</p>

<hr>
<h2 id='paramValues-methods'>Accessor for the slot paramValues.</h2><span id='topic+paramValues'></span><span id='topic+paramValues-methods'></span><span id='topic+paramValues+3C-'></span><span id='topic+paramValues+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>paramValues</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;InterpolationParam&quot;</dt><dd><p> Get or set the value of the slot <code>paramValues</code> in <code><a href="#topic+InterpolationParam">InterpolationParam</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>
</p>

<hr>
<h2 id='persistent-methods'>Accessors for the slot persistent.</h2><span id='topic+persistent'></span><span id='topic+persistent-methods'></span><span id='topic+persistent+3C-'></span><span id='topic+persistent+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>persistent</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p> Get or set the value of the slot <code>persistent</code> in <code><a href="#topic+Interpolation">Interpolation</a></code>. </p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='physicalPositions-methods'>Accessors for the slot physicalPositions</h2><span id='topic+physicalPositions'></span><span id='topic+physicalPositions-methods'></span><span id='topic+physicalPositions+3C-'></span><span id='topic+physicalPositions+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>physicalPositions</code> <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>Get or set the value of the slot <code>physicalPositions</code> <code><a href="#topic+MareyMap-class">MareyMap-class</a></code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='plotMarkers-methods'>plotMarkers</h2><span id='topic+plotMarkers'></span><span id='topic+plotMarkers-methods'></span>

<h3>Description</h3>

<p>Plots the markers contained in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>Plots the markers of the map <code><a href="#topic+MareyMap">MareyMap</a></code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='plotModel-methods'>plotModel</h2><span id='topic+plotModel'></span><span id='topic+plotModel-methods'></span>

<h3>Description</h3>

<p>Plot the model of <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p> Default function, plots nothing </p>
</dd>
<dt>object = &quot;MMSlidingWindow&quot;</dt><dd><p> Plots the model of an interpolation of type MMSlidingWindow </p>
</dd>
<dt>object = &quot;MMLoess&quot;</dt><dd><p>  Plots the model of an interpolation of type MMLoess-class </p>
</dd>
<dt>object = &quot;MMSpline3&quot;</dt><dd><p> Plots the model of an interpolation of type MMSpline3-class </p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='plotModels-methods'>plotModels </h2><span id='topic+plotModels'></span><span id='topic+plotModels-methods'></span>

<h3>Description</h3>

<p>plot models from <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p> Calls plotModel on all the interpolation defined on the <code><a href="#topic+MareyMap">MareyMap</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='plotRate-methods'> plotRate </h2><span id='topic+plotRate'></span><span id='topic+plotRate-methods'></span>

<h3>Description</h3>

<p>Plots the variation of the recombination rate as calculated by <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>   Plots the variation of the recombination rate as calculated by the <code><a href="#topic+Interpolation">Interpolation</a></code>. </p>
</dd>
<dt>object = &quot;MMSlidingWindow&quot;</dt><dd><p>  Plots the variation of the recombination rate as calculated by MMSlidingWindow.</p>
</dd>
<dt>object = &quot;MMLoess&quot;</dt><dd><p> Plots the variation of the recombination rate as calculated by MMLoess. </p>
</dd>
<dt>object = &quot;MMSpline3&quot;</dt><dd><p> Plots the variation of the recombination rate as calculated by MMSpline3. </p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='plotRates-methods'> plotRates</h2><span id='topic+plotRates'></span><span id='topic+plotRates-methods'></span>

<h3>Description</h3>

<p>Plots the variations of the recombination rate as calculated by all the
interpolations defined on the object.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p> Plots the variations of the recombination rate as calculated by all the interpolations defined on the <code><a href="#topic+MareyMap">MareyMap</a></code>. </p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='query-methods'>  Methods for Function query </h2><span id='topic+query'></span><span id='topic+query-methods'></span>

<h3>Description</h3>

<p>query an <code>object</code> for a given (or several) <code>pos</code>ition(s).
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;, pos = &quot;integer&quot;</dt><dd><p>Default method.</p>
</dd> 
<dt>object = &quot;MareyMap&quot;, pos = &quot;numeric&quot;</dt><dd><p>Query the local recombination rate as calculated by all the Interpolation defined on the map.</p>
</dd>
<dt>object = &quot;MMSlidingWindow&quot;, pos = &quot;numeric&quot;</dt><dd><p>Query the local recombination rate as calculated by MMSlidingWindow.</p>
</dd>
<dt>object = &quot;MMLoess&quot;, pos = &quot;numeric&quot;</dt><dd><p>Query the local recombination rate as calculated by MMLoess.</p>
</dd>
<dt>object = &quot;MMSpline3&quot;, pos = &quot;numeric&quot;</dt><dd><p>Query the local recombination rate as calculated by MMSpline3.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='rates-methods'>Accessors for the slot rates</h2><span id='topic+rates'></span><span id='topic+rates-methods'></span><span id='topic+rates+3C-'></span><span id='topic+rates+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>rates</code> in <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>Get or set the value of the slot <code>rates</code> in <code><a href="#topic+Interpolation">Interpolation</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>

<hr>
<h2 id='registerInterpolationMethod-methods'> registerInterpolationMethods</h2><span id='topic+registerInterpolationMethod'></span><span id='topic+registerInterpolationMethod-methods'></span><span id='topic+registerInterpolationMethod+2Ccharacter+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Adds an <code><a href="#topic+Interpolation">Interpolation</a></code> method to the interpolation register.
</p>


<h3>Methods</h3>


<dl>
<dt>name = &quot;character&quot;, classname = &quot;character&quot;</dt><dd><p>Register an interpolation method with a given <code>name</code> and a contructor function <code>classname</code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
</p>

<hr>
<h2 id='removeMarker-methods'> removeMarker </h2><span id='topic+removeMarker'></span><span id='topic+removeMarker-methods'></span>

<h3>Description</h3>

<p>Removes a marker from an <code>object</code> 
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;, value = &quot;integer&quot;</dt><dd><p>Removes the marker number <code>value</code> from a <code><a href="#topic+MareyMap-class">MareyMap-class</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='setName-methods'>Accessors for the slot setName.</h2><span id='topic+setName'></span><span id='topic+setName-methods'></span><span id='topic+setName+3C-'></span><span id='topic+setName+3C--methods'></span>

<h3>Description</h3>

<p>Get or set the value of the slot <code>setName</code> in <code>object</code>.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>Get or set the value of the slot <code>setName</code> in <code><a href="#topic+MareyMap">MareyMap</a></code>.</p>
</dd>
<dt>object = &quot;MapSet&quot;</dt><dd><p>Get or set the value of the slot <code>setName</code> in <code><a href="#topic+MapSet">MapSet</a></code>.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap">MareyMap</a></code>
<code><a href="#topic+MapSet">MapSet</a></code>
</p>

<hr>
<h2 id='setNames-methods'>setNames</h2><span id='topic+setNames'></span><span id='topic+setNames-methods'></span>

<h3>Description</h3>

<p>returns the names of the set contained in object.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MapCollection&quot;</dt><dd><p> Returns the name of the set contained in <code><a href="#topic+MapCollection">MapCollection</a></code>. </p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MapCollection">MapCollection</a></code>.
</p>

<hr>
<h2 id='startMareyMapGUI-methods'>MareyMapGUI launcher</h2><span id='topic+startMareyMapGUI'></span><span id='topic+startMareyMapGUI-methods'></span><span id='topic+startMareyMapGUI+2Cmissing-method'></span>

<h3>Description</h3>

<p>launches the graphical user interface of MareyMapGUI
</p>


<h3>Methods</h3>


<dl>
<dt>dummy = &quot;missing&quot;</dt><dd><p>Default method.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>

<hr>
<h2 id='textFile-methods'>textFile</h2><span id='topic+textFile'></span><span id='topic+textFile-methods'></span>

<h3>Description</h3>

<p>Writes object to text file.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MapCollection&quot;, file = &quot;character&quot;</dt><dd><p>Writes a <code><a href="#topic+MapCollection">MapCollection</a></code> to text file.</p>
</dd>
<dt>object = &quot;MareyMap&quot;, file = &quot;character&quot;</dt><dd><p>Writes a <code><a href="#topic+MareyMap">MareyMap</a></code> to text file.</p>
</dd>
<dt>object = &quot;MapSet&quot;, file = &quot;character&quot;</dt><dd><p>Writes a <code><a href="#topic+MapSet">MapSet</a></code> to text file.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
<code><a href="#topic+MapCollection-class">MapCollection-class</a></code>
<code><a href="#topic+MapSet-class">MapSet-class</a></code>
</p>

<hr>
<h2 id='userParam-methods'> userParam</h2><span id='topic+userParam'></span><span id='topic+userParam-methods'></span>

<h3>Description</h3>

<p>Returns information about the parameters modifiable by the user in object
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>Returns information about the parameters modifiable by the user in <code><a href="#topic+Interpolation">Interpolation</a></code></p>
</dd>
<dt>object = &quot;MMSlidingWindow&quot;</dt><dd><p> Returns information about the parameters modifiable by the user in MMSlidingWindow</p>
</dd> 
<dt>object = &quot;MMLoess&quot;</dt><dd><p> Returns information about the parameters modifiable by the user in MMLoess</p>
</dd>
<dt>object = &quot;MMSpline3&quot;</dt><dd><p> Returns information about the parameters modifiable by the user in  MMSpline3</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+Interpolation-class">Interpolation-class</a></code>
<code><a href="#topic+InterpolationParam-class">InterpolationParam-class</a></code>
</p>

<hr>
<h2 id='validPositions-methods'>validPositions</h2><span id='topic+validPositions'></span><span id='topic+validPositions-methods'></span>

<h3>Description</h3>

<p>Returns the positions of object which are valid.
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;MareyMap&quot;</dt><dd><p>Returns the physical positions of MareyMap which are valid.</p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>


<h3>See Also</h3>

<p><code><a href="#topic+MareyMap-class">MareyMap-class</a></code>
</p>

<hr>
<h2 id='visible-methods'>Accessors for the slot visible</h2><span id='topic+visible'></span><span id='topic+visible-methods'></span><span id='topic+visible+3C-'></span><span id='topic+visible+3C--methods'></span>

<h3>Description</h3>

<p>Get of set the value of the slot <code>visible</code> in <code>object</code>
</p>


<h3>Methods</h3>


<dl>
<dt>object = &quot;Interpolation&quot;</dt><dd><p>Get of set the value of the slot <code>visible</code> in <code><a href="#topic+Interpolation-class">Interpolation-class</a></code></p>
</dd>
</dl>


<h3>Author(s)</h3>

<p>Aurélie Siberchicot <a href="mailto:aurelie.siberchicot@univ-lyon1.fr">aurelie.siberchicot@univ-lyon1.fr</a> and Clément Rezvoy</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
