<!DOCTYPE html><html><head><title>Help for package markerpen</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {markerpen}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gene_mapping'><p>Mapping gene names to Ensembl IDs</p></a></li>
<li><a href='#pca_pen'><p>Penalized Principal Component Analysis for Marker Gene Selection</p></a></li>
<li><a href='#refine_markers'><p>Marker Gene Selection via Penalized Principal Component Analysis</p></a></li>
<li><a href='#sort_markers'><p>Post-processing Selected Marker Genes</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Marker Gene Detection via Penalized Principal Component Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-03-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Yixuan Qiu, Jiebiao Wang, Jing Lei, and Kathryn Roeder</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yixuan Qiu &lt;yixuan.qiu@cos.name&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of the 'MarkerPen' algorithm, short for marker gene detection
    via penalized principal component analysis, described in the paper by Qiu, Wang, Lei,
    and Roeder (2020, &lt;<a href="https://doi.org/10.1101%2F2020.11.07.373043">doi:10.1101/2020.11.07.373043</a>&gt;). 'MarkerPen' is a semi-supervised
    algorithm for detecting marker genes by combining prior marker information with bulk
    transcriptome data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.1), RSpectra, stats</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen, RSpectra</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, prettydoc, scales</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>C++11</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-14 11:41:12 UTC; qyx</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-17 00:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='gene_mapping'>Mapping gene names to Ensembl IDs</h2><span id='topic+gene_mapping'></span>

<h3>Description</h3>

<p>A data set showing the mapping between gene names and Ensembl gene IDs,
derived from the <span class="pkg">EnsDb.Hsapiens.v79</span> Bioconductor package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gene_mapping
</code></pre>


<h3>Format</h3>

<p>A data frame with 59074 rows and 2 variables:
</p>

<dl>
<dt>ensembl</dt><dd><p>Ensembl gene IDs</p>
</dd>
<dt>name</dt><dd><p>corresponding gene names</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://bioconductor.org/packages/release/data/annotation/html/EnsDb.Hsapiens.v79.html">https://bioconductor.org/packages/release/data/annotation/html/EnsDb.Hsapiens.v79.html</a>
</p>

<hr>
<h2 id='pca_pen'>Penalized Principal Component Analysis for Marker Gene Selection</h2><span id='topic+pca_pen'></span>

<h3>Description</h3>

<p>This function solves the optimization problem
</p>
<p style="text-align: center;"><code class="reqn">\min\quad-\mathrm{tr}(SX) + \lambda p(X),</code>
</p>

<p style="text-align: center;"><code class="reqn">s.t.\quad O\preceq X \preceq I, \quad X \ge 0, \quad\mathrm{and}\quad \mathrm{tr}(X)=1,</code>
</p>

<p>where <code class="reqn">O\preceq X \preceq I</code> means all eigenvalues of <code class="reqn">X</code> are
between 0 and 1, <code class="reqn">X \ge 0</code> means all elements of <code class="reqn">X</code> are nonnegative,
and <code class="reqn">p(X)</code> is a penalty function defined in the article
(see the <strong>References</strong> section).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pca_pen(
  S,
  gr,
  lambda,
  w = 1.5,
  alpha = 0.01,
  maxit = 1000,
  eps = 1e-04,
  verbose = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pca_pen_+3A_s">S</code></td>
<td>
<p>The sample correlation matrix of gene expression.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_gr">gr</code></td>
<td>
<p>Indices of genes that are treated as markers in the prior information.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_lambda">lambda</code></td>
<td>
<p>Tuning parameter to control the sparsity of eigenvectors.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_w">w</code></td>
<td>
<p>Tuning parameter to control the weight on prior information.
Larger <code class="reqn">w</code> means genes not in the prior list are less likely
to be selected as markers.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_alpha">alpha</code></td>
<td>
<p>Step size of the optimization algorithm.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_eps">eps</code></td>
<td>
<p>Tolerance parameter for convergence.</p>
</td></tr>
<tr><td><code id="pca_pen_+3A_verbose">verbose</code></td>
<td>
<p>Level of verbosity.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>

<dl>
<dt>projection</dt><dd><p>The estimated projection matrix.</p>
</dd>
<dt>evecs</dt><dd><p>The estimated eigenvectors.</p>
</dd>
<dt>niter</dt><dd><p>Number of iterations used in the optimization process.</p>
</dd>
<dt>err_v</dt><dd><p>The optimization error in each iteration.</p>
</dd>
</dl>



<h3>References</h3>

<p>Qiu, Y., Wang, J., Lei, J., &amp; Roeder, K. (2020).
Identification of cell-type-specific marker genes from co-expression patterns in tissue samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
n = 200  # Sample size
p = 500  # Number of genes
s = 50   # Number of true signals

# The first s genes are true markers, and others are noise
Sigma = matrix(0, p, p)
Sigma[1:s, 1:s] = 0.9
diag(Sigma) = 1

# Simulate data from the covariance matrix
x = matrix(rnorm(n * p), n) %*% chol(Sigma)

# Sample correlation matrix
S = cor(x)

# Indices of prior marker genes
# Note that we have omitted 10 true markers, and included 10 false markers
gr = c(1:(s - 10), (s + 11):(s + 20))

# Run the algorithm
res = pca_pen(S, gr, lambda = 0.1, verbose = 1)

# See if we can recover the true correlation structure
image(res$projection, asp = 1)

</code></pre>

<hr>
<h2 id='refine_markers'>Marker Gene Selection via Penalized Principal Component Analysis</h2><span id='topic+refine_markers'></span>

<h3>Description</h3>

<p>This function refines a prior marker gene list by combining information from bulk
tissue data, based on the penalized principal component analysis. The current
implementation computes on one cell type at a time. To get marker genes for
multiple cell types, call this function iteratively.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>refine_markers(
  mat_exp,
  range,
  markers,
  lambda,
  w = 1.5,
  thresh = 0.001,
  alpha = 0.01,
  maxit = 1000,
  eps = 1e-04,
  verbose = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="refine_markers_+3A_mat_exp">mat_exp</code></td>
<td>
<p>The gene expression matrix in the original scale
(not logarithm-transformed), with rows standing for observations and
columns for genes. The matrix should include gene names as column names.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_range">range</code></td>
<td>
<p>A character vector of gene names, representing the range of genes in which
markers are sought.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_markers">markers</code></td>
<td>
<p>A character vector of gene names giving the prior marker gene list.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_lambda">lambda</code></td>
<td>
<p>A tuning parameter to control the number of selected marker genes. A larger
value typically means a smaller number of genes.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_w">w</code></td>
<td>
<p>Tuning parameter to control the weight on prior information.
Larger <code class="reqn">w</code> means genes not in the prior list are less likely
to be selected as markers.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_thresh">thresh</code></td>
<td>
<p>Below this threshold small factor loadings are treated as zeros.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_alpha">alpha</code></td>
<td>
<p>Step size of the optimization algorithm.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_eps">eps</code></td>
<td>
<p>Tolerance parameter for convergence.</p>
</td></tr>
<tr><td><code id="refine_markers_+3A_verbose">verbose</code></td>
<td>
<p>Level of verbosity.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the following components:
</p>

<dl>
<dt>spca</dt><dd><p>The sparse PCA result as in <code><a href="#topic+pca_pen">pca_pen</a>()</code>.</p>
</dd>
<dt>markers</dt><dd><p>A character vector of selected markers genes.</p>
</dd>
<dt>markers_coef</dt><dd><p>The estimated factor loadings for the associated genes.</p>
</dd>
</dl>



<h3>References</h3>

<p>Qiu, Y., Wang, J., Lei, J., &amp; Roeder, K. (2020).
Identification of cell-type-specific marker genes from co-expression patterns in tissue samples.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Data used in the vignette
load(system.file("examples", "gene_expr.RData", package = "markerpen"))
load(system.file("examples", "published_markers.RData", package = "markerpen"))
load(system.file("examples", "markers_range.RData", package = "markerpen"))

# Get expression matrix - rows are observations, columns are genes
ind = match(rownames(dat), markerpen::gene_mapping$name)
ind = na.omit(ind)
ensembl = markerpen::gene_mapping$ensembl[ind]
mat_exp = t(dat[markerpen::gene_mapping$name[ind], ])
colnames(mat_exp) = ensembl

# We compute the marker genes for two cell types with a reduced problem size
# See the vignette for the full example

# Markers for astrocytes
set.seed(123)
search_range = intersect(markers_range$astrocytes, ensembl)
search_range = sample(search_range, 300)
prior_markers = intersect(pub_markers$astrocytes, search_range)
ast_re = refine_markers(
    mat_exp, search_range, prior_markers,
    lambda = 0.35, w = 1.5, maxit = 500, eps = 1e-3, verbose = 0
)
# Remove selected markers from the expression matrix
mat_rest = mat_exp[, setdiff(colnames(mat_exp), ast_re$markers)]

# Markers for microglia
search_range = intersect(markers_range$microglia, ensembl)
search_range = sample(search_range, 300)
prior_markers = intersect(pub_markers$microglia, search_range)
mic_re = refine_markers(
    mat_exp, search_range, prior_markers,
    lambda = 0.35, w = 1.5, maxit = 500, eps = 1e-3, verbose = 0
)

# Refined markers
markers_re = list(astrocytes = ast_re$markers,
                  microglia  = mic_re$markers)
# Visualize the correlation matrix
cor_markers = cor(mat_exp[, unlist(markers_re)])
image(cor_markers, asp = 1)

# Post-process the selected markers
# Pick the first 20 ordered markers
markers_ord = sort_markers(cor_markers, markers_re)
markers_ord = lapply(markers_ord, head, n = 20)
# Visualize the correlation matrix
image(cor(mat_exp[, unlist(markers_ord)]), asp = 1)

</code></pre>

<hr>
<h2 id='sort_markers'>Post-processing Selected Marker Genes</h2><span id='topic+sort_markers'></span>

<h3>Description</h3>

<p>This function reorders the selected marker genes using information of the sample
correlation matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sort_markers(corr, markers)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sort_markers_+3A_corr">corr</code></td>
<td>
<p>The sample correlation matrix, whose row and column names are gene names.</p>
</td></tr>
<tr><td><code id="sort_markers_+3A_markers">markers</code></td>
<td>
<p>A list of marker genes. Each component of the list is a vector of marker
gene names corresponding to a cell type. All the gene names in this list
must appear in the row/column names of <code>corr</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list that has the same structure as the input <code>markers</code> argument, with
the elements in each component reordered. See the example in
<code><a href="#topic+refine_markers">refine_markers</a>()</code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
