<!DOCTYPE html><html><head><title>Help for package markophylo</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {markophylo}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#markophylo-package'>
<p>Markov chain models for phylogenetic trees.</p></a></li>
<li><a href='#estimaterates'>
<p>Estimate substitution rate matrix.</p></a></li>
<li><a href='#estimaterates_f'>
<p>Estimate substitution rate matrix.</p></a></li>
<li><a href='#patterns'>
<p>Unique phyletic patterns with counts.</p></a></li>
<li><a href='#plottree'>
<p>Plot the tree used with different hypothesized branch groupings (or clades) following unique rates coloured differently.</p></a></li>
<li><a href='#print.markophylo'>
<p>Print summary information for the model fit.</p></a></li>
<li><a href='#simdata1'>
<p>Simulated data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Markov Chain Models for Phylogenetic Trees</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-14</td>
</tr>
<tr>
<td>Author:</td>
<td>Utkarsh J. Dang and G. Brian Golding</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Utkarsh J. Dang &lt;utkarshdang@cunet.carleton.ca&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows for fitting of maximum likelihood models using Markov chains
    on phylogenetic trees for analysis of discrete character data. Examples of such
    discrete character data include restriction sites, gene family presence/absence,
    intron presence/absence, and gene family size data. Hypothesis-driven user-
    specified substitution rate matrices can be estimated. Allows for biologically
    realistic models combining constrained substitution rate matrices, site rate
    variation, site partitioning, branch-specific rates, allowing for non-stationary
    prior root probabilities, correcting for sampling bias, etc. See Dang and Golding 
    (2016) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtv541">doi:10.1093/bioinformatics/btv541</a>&gt; for more details.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.4), ape (&ge; 3.2), numDeriv (&ge; 2012.9.1),
phangorn (&ge; 1.99.13), geiger</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr (&ge; 1.10), testthat (&ge; 0.9.1), markdown (&gt; 1.0)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-19 11:28:03 UTC; udang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-19 12:20:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='markophylo-package'>
Markov chain models for phylogenetic trees.
</h2><span id='topic+markophylo-package'></span><span id='topic+markophylo'></span>

<h3>Description</h3>

<p>Allows for fitting of maximum likelihood models using Markov chains on phylogenetic trees for analysis of discrete character data. Examples of such discrete character data include restriction sites, gene family presence/absence, intron presence/absence, and gene family size data. Hypothesis-driven user-specified substitution rate matrices can be estimated. Allows for biologically realistic models combining constrained substitution rate matrices, site rate variation, site partitioning, branch-specific rates, allowing for non-stationary prior root probabilities, correcting for sampling bias, etc. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> markophylo</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.5</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-01-19</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;= 2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Function <code><a href="#topic+estimaterates">estimaterates</a></code> is the main workhorse of the package, allowing users to specify a substitution rate matrix and estimate it. Partition analysis, gamma rate variation, estimating root frequencies of the discrete characters, and specifying clades (or a set of branches) with their own unique rates are all possible.
</p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a>
</p>


<h3>References</h3>

<p>Dirk Eddelbuettel and Conrad Sanderson (2014). RcppArmadillo: Accelerating R with high-performance C++ linear algebra. Computational Statistics and Data Analysis, 71, 1054&ndash;1063.
</p>
<p>Eddelbuettel, Dirk and Romain Francois (2011). Rcpp: Seamless R and C++ Integration. Journal of Statistical Software, 40(8), 1&ndash;18.
</p>
<p>Gilbert, Paul and Ravi Varadhan (2012). numDeriv: Accurate Numerical Derivatives. R package version 2014.2-1.
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008. GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131.
</p>
<p>Paradis, E., Claude, J. &amp; Strimmer, K. 2004. APE: analyses of phylogenetics and evolution in R language. Bioinformatics 20: 289-290. R package version 3.2.
</p>
<p>Schliep, K.P. 2011. phangorn: phylogenetic analysis in R. Bioinformatics, 27(4) 592-593. R package version 1.99.13.
</p>
<p>Yang, Z. (1994). Maximum likelihood phylogenetic estimation from DNA sequences with variable rates over sites: approximate methods. Journal of Molecular evolution, 39(3), 306&ndash;314.
</p>

<hr>
<h2 id='estimaterates'>
Estimate substitution rate matrix.
</h2><span id='topic+estimaterates'></span>

<h3>Description</h3>

<p>Estimate a substitution rate matrix. Users can specify a substitution rate matrix and estimate it. Partition analysis, gamma rate variation, estimating root frequencies of the discrete characters, and specifying clades (or a set of branches) with their own unique rates are also possible. See arguments below. See vignette for detailed examples.</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimaterates(usertree = NULL, userphyl = NULL, matchtipstodata = FALSE,
              unobserved = NULL, alphabet = NULL, modelmat = NULL, 
              bgtype = "listofnodes", bg = NULL, partition = NULL, 
              ratevar = FALSE, nocat = 4, reversible = FALSE, numhessian = TRUE, 
              rootprob = NULL, rpvec = NULL, init = 0.9, lowli = 0.001, upli = 100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimaterates_+3A_usertree">usertree</code></td>
<td>

<p>Rooted binary tree of class &quot;phylo&quot;. Read in Newick tree using read.tree() from package ape before passing this argument. The branch lengths must be in expected substitutions per site (but see lowli and upli arguments). Trees estimated using MrBayes and BEAST, for instance, yield branch lengths using that scale. If an unrooted tree is available, look into the &ldquo;root&quot; and &ldquo;midpoint.root&quot; functions from the APE and phytools packages, respectively.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_userphyl">userphyl</code></td>
<td>

<p>A matrix or data frame of phyletic patterns. Rows represent the discrete character patterns and columns represent the taxa. These data can be numeric or character (but not both). 
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_matchtipstodata">matchtipstodata</code></td>
<td>

<p>The default is FALSE, which means that the user must ensure that the ordering of the taxa in the data matrix must match the internal ordering of tip labels of the tree. Set to TRUE, if the column names of the data matrix, i.e., the taxa names, are all present, with the same spelling and notation in the tip labels of the tree provided, in which case, the restriction on ordering is not necessary.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_unobserved">unobserved</code></td>
<td>

<p>A matrix of unobserved phyletic patterns, representing possible sampling or acquisition bias. Each row should be a unique phyletic pattern.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_alphabet">alphabet</code></td>
<td>

<p>The set of discrete characters used. May be integer only or character only.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_modelmat">modelmat</code></td>
<td>

<p>Can be one of two options: pre-built or user-specified. For the pre-built matrices, use:
</p>

<ul>
<li><p> &quot;ER&quot; for an equal rates matrix, e.g., matrix(c(NA, 1, 1, 1, 1, NA, 1, 1, 1, 1, NA, 1, 1, 1, 1, NA
), nrow = 4, ncol = 4)
</p>
</li>
<li><p> &quot;SYM&quot; for a symmetric matrix, e.g., matrix(c(NA, 1, 2, 3, 1, NA, 4, 5, 2, 4, NA, 6, 3, 5, 6, NA
), 4,4)
</p>
</li>
<li><p> &quot;ARD&quot; for an all rates different matrix, e.g., matrix(c(NA, 1, 2, 3, 4, NA, 5, 6, 7, 8, NA, 9, 10, 11, 12, 
NA), 4, 4)
</p>
</li>
<li><p> &quot;GTR&quot; for a general time reversible model. This sets reversible to be TRUE, rootprob to be &quot;maxlik&quot;, and the rate matrix supplied to the function is symmetric. The estimated rate matrix can be written as the product of a symmetric matrix multiplied by a diagonal matrix (consisting of the estimated root probabilities).
</p>
</li>
<li><p> &quot;BD&quot; for a standard birth death matrix, e.g., matrix(c(NA, 1, 0, 0, 2, NA, 1, 0, 0, 2, NA, 1, 0, 0, 2, NA
), 4, 4)
</p>
</li>
<li><p> &quot;BDER&quot; for a standard birth death matrix, e.g., matrix(c(NA, 1, 0, 0, 1, NA, 1, 0, 0, 1, NA, 1, 0, 0, 1, NA
), 4, 4)
</p>
</li>
<li><p> &quot;BDSYM&quot; for a standard birth death matrix, e.g., matrix(c(NA, 1, 0, 0, 1, NA, 2, 0, 0, 2, NA, 3, 0, 0, 3, NA
), 4, 4)
</p>
</li>
<li><p> &quot;BDARD&quot; for an all rates different birth death matrix, e.g., matrix(c(NA, 1, 0, 0, 4, NA, 2, 0, 0, 5, NA, 3, 0, 0, 6, NA
), 4, 4)
</p>
</li></ul>

<p>These pre-built matrices are inspired by the APE and DiscML packages.
</p>
<p>A square matrix can also be input that consists of integer values denoting the indices of the substitution rates to be estimated. The number of rows and columns corresponds to the total number of discete states possible in the data. For example, using matrix(c(NA, 1, 2, NA), 2, 2) means that two rates must be estimated corresponding to the entries 1 and 2, respectively. Using matrix(c(NA, 1, 0, NA), 2, 2) means that only one rate must be estimated and the entry 0 corresponds to a substitution that is not permitted (hence, does not need to be estimated). See examples and vignette for more examples.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_bgtype">bgtype</code></td>
<td>

<p>Use this to group branches hypothesized to follow the same rates (but differ from other branches). If clade-specific insertion and deletion rates are required to be estimated, use argument option &quot;ancestornodes&quot;. If, on the other hand, a group of branches (not in a clade) are hypothesized to follow the same rates, use argument option &quot;listofnodes&quot;.</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_bg">bg</code></td>
<td>

<p>A vector of nodes should be provided if the &quot;ancestornodes&quot; option was chosen for argument &quot;bgtype&quot;. If, on the other hand, &quot;listofnodes&quot; was chosen, a list should be provided with each element of the list being a vector of nodes that limit the branches that follow the same rates. See examples and vignette.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_partition">partition</code></td>
<td>

<p>A list of vectors (of sites) subject to different evolutionary constraints. For example, supplying list(c(1:2500), c(2501:5000)) means that sites 1 through 2500 follow their own substitution rates distinct from sites 2501 through 5000. These sites correspond to the rows of the data supplied in userphyl. Partition models can be fitted with a common (or unique) gamma distribution for rate variation over all partitions, and/or common root probabilities can be estimated over all partitions.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_ratevar">ratevar</code></td>
<td>

<p>Default option is FALSE. Specifying &quot;discgamma&quot; implements the discrete gamma approximation model of Yang, 1994. Even if a partition of sites is specified, this option uses the same <code class="reqn">\alpha</code> parameter for the gamma distribution over all partitions. Specifying &quot;partitionspecificgamma&quot; implements the discrete gamma approximation model in each partition separately.
The number of categories can be specified using the &quot;nocat&quot; argument. See examples and vignette.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_nocat">nocat</code></td>
<td>

<p>The number of categories for the discrete gamma approximation. 
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_reversible">reversible</code></td>
<td>

<p>This option forces a model to be reversible, i.e., the flow from state 'a' to 'b' is the same as the flow from state 'b' to 'a'. Only symmetric transition matrices can be specified in modelmat with this option. Inspired by the DiscML package.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_numhessian">numhessian</code></td>
<td>

<p>Set to FALSE if standard errors are not required. This speeds up the algorithm. Default is TRUE. Although the function being used to calculate these errors is reliable, it is rare but possible that the errors are not calculated (due to approximation-associated issues while calculating the Hessian). In this case, try bootstrapping with numhessian=FALSE.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_rootprob">rootprob</code></td>
<td>

<p>Four options are available: &quot;equal&quot;, &quot;stationary&quot;, &quot;maxlik&quot;, and &quot;user&quot;. 
</p>

<ul>
<li><p> Option &quot;equal&quot; means that all the discrete characters are given equal weight at the root. 
</p>
</li>
<li><p> Using &quot;stationary&quot; means that the discrete characters are weighted at the root by the stationary frequencies implied by the substitution rate matrix. Note that these can differ based on the partition. In the case that the &quot;bgtype&quot; argument is also provided, an average of the stationary frequencies of all branch groupings is used.
</p>
</li>
<li><p> If &quot;maxlik&quot; is supplied, then the root frequencies are also estimated. These do not differ based on the partition.
</p>
</li>
<li><p> If &quot;user&quot; is supplied here, a vector of root frequency parameters can be provided to argument &quot;rpvec&quot;.
</p>
</li></ul>

</td></tr>
<tr><td><code id="estimaterates_+3A_rpvec">rpvec</code></td>
<td>

<p>If option &quot;user&quot; is specified for argument &quot;rootprob&quot;, supply a vector of the same length as that provided to argument &quot;alphabet&quot;, representing the root frequency parameters.
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_init">init</code></td>
<td>
	
<p>Initial value for the rates. The default value is 0.9.</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_lowli">lowli</code></td>
<td>

<p>For finer control of the boundaries of the optimization problem. The default value is 0.001. This usually suffices if the branch lengths are in expected substitutions per site. However, if branch lengths are in different units, this should be changed accordingly. 
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_upli">upli</code></td>
<td>

<p>For finer control of the boundaries of the optimization problem. The default value is 100. This usually suffices if the branch lengths are in expected substitutions per site. However, if branch lengths are in different units, this should be changed accordingly. 
</p>
</td></tr>
<tr><td><code id="estimaterates_+3A_...">...</code></td>
<td>

<p>Passing other arguments to the optimization algorithm &quot;nlminb&quot;. For example, control = list(trace = 5) will print progress at every 5th iteration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See vignette for detailed examples.
</p>


<h3>Value</h3>

<p>All arguments used while calling the estimaterates function are attached in a list. The following components are also returned in the same list:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>Function call used.</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>A vector of convergence indicators for the model run. 0 denotes successful convergence.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Time taken in seconds.</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>The phylogenetic tree used.</p>
</td></tr>
<tr><td><code>bg</code></td>
<td>
<p>List of group of nodes that capture branches that follow unique substitution rates.</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>List of results including the parameter estimates for each unique entry of modelmat (excluding zeros), standard errors, number of parameters fit, and AIC and BIC values. Furthermore, details from the optimization routine applied are also available. Use ...$results$wop$parsep.</p>
</td></tr>
<tr><td><code>data_red</code></td>
<td>
<p>Unique phyletic patterns observed.</p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>List of number of times each gene phyletic pattern was observed.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>See the vignette for an in-depth look at some examples.
</p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a>
</p>


<h3>References</h3>

<p>Paradis, E., Claude, J. &amp; Strimmer, K. 2004. APE: analyses of phylogenetics and evolution in R language. Bioinformatics 20: 289-290. R package version 3.2.
</p>
<p>Tane Kim and Weilong Hao 2014. DiscML: An R package for estimating evolutionary rates of discrete characters using maximum likelihood. R package version 1.0.1. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(markophylo)
##############
data(simdata1) #example data for a 2-state continuous time markov chain model.
#Now, plot example tree.
ape::plot.phylo(simdata1$tree, edge.width = 2, show.tip.label = FALSE, no.margin = TRUE) 
ape::nodelabels(frame = "circle", cex = 0.7)
ape::tiplabels(frame = "circle", cex = 0.7)
print(simdata1$Q) #substitution matrix used to simulate example data
print(table(simdata1$data)) #states and frequencies
model1 &lt;- estimaterates(usertree = simdata1$tree, userphyl = simdata1$data, 
                        alphabet = c(1, 2), rootprob = "equal", 
                        modelmat = matrix(c(NA, 1, 2, NA), 2, 2))
print(model1)
####

#If the data is known to contain sampling bias such that certain phyletic
#patterns are not observed, then these unobserved data can be corrected for
#easily. First, let's create a filtered version of the data following which
#a correction will be applied within the function. Here, any patterns with all
#ones or twos are filtered out.
filterall1 &lt;- which(apply(simdata1$data, MARGIN = 1, FUN = 
                            function(x) isTRUE(all.equal(as.vector(x), c(1, 1, 1, 1)))))
filterall2 &lt;- which(apply(simdata1$data, MARGIN = 1, FUN = 
                            function(x) isTRUE(all.equal(as.vector(x), c(2, 2, 2, 2)))))
filteredsimdata1 &lt;- simdata1$data[-c(filterall1, filterall2), ]
model1_f_corrected &lt;- estimaterates(usertree = simdata1$tree, userphyl = filteredsimdata1, 
                          unobserved = matrix(c(1, 1, 1, 1, 2, 2, 2, 2), nrow = 2,  byrow = TRUE), 
                          alphabet = c(1, 2), rootprob = "equal", 
                          modelmat = matrix(c(NA, 1, 2, NA), 2, 2))
print(model1_f_corrected)
##############
data(simdata2)
print(simdata2$Q)
#While simulating the data found in simdata2, the clade with node 7 as its
#most recent common ancestor (MRCA) was constrained to have twice the 
#substitution rates as the rest of the branches in the tree.
print(table(simdata2$data))
model2 &lt;- estimaterates(usertree = simdata2$tree, userphyl = simdata2$data, 
                        alphabet = c(1, 2), bgtype = "ancestornodes", bg = c(7),
                        rootprob = "equal", modelmat = matrix(c(NA, 1, 2, NA), 2, 2))
print(model2)
plottree(model2, colors=c("blue", "darkgreen"), edge.width = 2, show.tip.label = FALSE, 
         no.margin = TRUE)
ape::nodelabels(frame = "circle", cex = 0.7)
ape::tiplabels(frame = "circle", cex = 0.7)
##############
#Nucleotide data was simulated such that the first half of sites followed
#substitution rates different from the other half of sites. Data was simulated
#in the two partitions with rates 0.33 and 0.99.
data(simdata3)
print(dim(simdata3$data))
print(table(simdata3$data))
model3 &lt;- estimaterates(usertree = simdata3$tree, userphyl = simdata3$data, 
                        alphabet = c("a", "c", "g", "t"), rootprob = "equal", 
                        partition = list(c(1:2500), c(2501:5000)), 
                        modelmat = matrix(c(NA, 1, 1, 1, 1, NA, 1, 1, 
                                            1, 1, NA, 1, 1, 1, 1, NA), 4, 4))
print(model3)

#More examples in the vignette.
</code></pre>

<hr>
<h2 id='estimaterates_f'>
Estimate substitution rate matrix.
</h2><span id='topic+estimaterates_f'></span>

<h3>Description</h3>

<p>Beta version of code; still in development. More or less identical to <code><a href="#topic+estimaterates">estimaterates</a></code>. Here, some entries of the substitution rate matrix can be specified by the user to be held fixed during the parameter estimation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimaterates_f(usertree = NULL, userphyl = NULL, matchtipstodata = FALSE,
              unobserved = NULL, alphabet = NULL, modelmat = NULL, 
              bgtype = "listofnodes", bg = NULL, partition = NULL, 
              ratevar = FALSE, nocat = 4, reversible = FALSE, numhessian = TRUE, 
              rootprob = NULL, rpvec = NULL, init = 0.9, lowli = 0.001, upli = 100, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimaterates_f_+3A_usertree">usertree</code></td>
<td>

<p>Rooted binary tree of class &quot;phylo&quot;. Read in Newick tree using read.tree() from package ape before passing this argument. The branch lengths must be in expected substitutions per site (but see lowli and upli arguments). Trees estimated using MrBayes and BEAST, for instance, yield branch lengths using that scale. If an unrooted tree is available, look into the &ldquo;root&quot; and &ldquo;midpoint.root&quot; functions from the APE and phytools packages, respectively.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_userphyl">userphyl</code></td>
<td>

<p>A matrix or data frame of phyletic patterns. Rows represent the discrete character patterns and columns represent the taxa. These data can be numeric or character (but not both). 
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_matchtipstodata">matchtipstodata</code></td>
<td>

<p>The default is FALSE, which means that the user must ensure that the ordering of the taxa in the data matrix must match the internal ordering of tip labels of the tree. Set to TRUE, if the column names of the data matrix, i.e., the taxa names, are all present, with the same spelling and notation in the tip labels of the tree provided, in which case, the restriction on ordering is not necessary.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_unobserved">unobserved</code></td>
<td>

<p>A matrix of unobserved phyletic patterns, representing possible sampling or acquisition bias. Each row should be a unique phyletic pattern.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_alphabet">alphabet</code></td>
<td>

<p>The set of discrete characters used. May be integer only or character only.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_modelmat">modelmat</code></td>
<td>

<p>Accepts the same arguments as <code><a href="#topic+estimaterates">estimaterates</a></code>.  Here, some entries of the substitution rate matrix can be specified by the user to be held fixed during the parameter estimation. A square matrix can be input that consists of integer values denoting the indices of the substitution rates to be estimated and entries with decimal values (that are not equivalent to an integer in R, i.e., something other than, for example, 1.0, 2.0, etc.) to be held fixed. See example.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_bgtype">bgtype</code></td>
<td>

<p>Use this to group branches hypothesized to follow the same rates (but differ from other branches). If clade-specific insertion and deletion rates are required to be estimated, use argument option &quot;ancestornodes&quot;. If, on the other hand, a group of branches (not in a clade) are hypothesized to follow the same rates, use argument option &quot;listofnodes&quot;.</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_bg">bg</code></td>
<td>

<p>A vector of nodes should be provided if the &quot;ancestornodes&quot; option was chosen for argument &quot;bgtype&quot;. If, on the other hand, &quot;listofnodes&quot; was chosen, a list should be provided with each element of the list being a vector of nodes that limit the branches that follow the same rates. See examples and vignette.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_partition">partition</code></td>
<td>

<p>A list of vectors (of sites) subject to different evolutionary constraints. For example, supplying list(c(1:2500), c(2501:5000)) means that sites 1 through 2500 follow their own substitution rates distinct from sites 2501 through 5000. These sites correspond to the rows of the data supplied in userphyl. Partition models can be fitted with a common (or unique) gamma distribution for rate variation over all partitions, and/or common root probabilities can be estimated over all partitions.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_ratevar">ratevar</code></td>
<td>

<p>Default option is FALSE. Specifying &quot;discgamma&quot; implements the discrete gamma approximation model of Yang, 1994. Even if a partition of sites is specified, this option uses the same <code class="reqn">\alpha</code> parameter for the gamma distribution over all partitions. Specifying &quot;partitionspecificgamma&quot; implements the discrete gamma approximation model in each partition separately.
The number of categories can be specified using the &quot;nocat&quot; argument. See examples and vignette.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_nocat">nocat</code></td>
<td>

<p>The number of categories for the discrete gamma approximation. 
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_reversible">reversible</code></td>
<td>

<p>This option forces a model to be reversible, i.e., the flow from state 'a' to 'b' is the same as the flow from state 'b' to 'a'. Only symmetric transition matrices can be specified in modelmat with this option. Inspired by the DiscML package.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_numhessian">numhessian</code></td>
<td>

<p>Set to FALSE if standard errors are not required. This speeds up the algorithm. Default is TRUE. Although the function being used to calculate these errors is reliable, it is rare but possible that the errors are not calculated (due to approximation-associated issues while calculating the Hessian). In this case, try bootstrapping with numhessian=FALSE.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_rootprob">rootprob</code></td>
<td>

<p>Four options are available: &quot;equal&quot;, &quot;stationary&quot;, &quot;maxlik&quot;, and &quot;user&quot;. 
</p>

<ul>
<li><p> Option &quot;equal&quot; means that all the discrete characters are given equal weight at the root. 
</p>
</li>
<li><p> Using &quot;stationary&quot; means that the discrete characters are weighted at the root by the stationary frequencies implied by the substitution rate matrix. Note that these can differ based on the partition. In the case that the &quot;bgtype&quot; argument is also provided, an average of the stationary frequencies of all branch groupings is used.
</p>
</li>
<li><p> If &quot;maxlik&quot; is supplied, then the root frequencies are also estimated. These do not differ based on the partition.
</p>
</li>
<li><p> If &quot;user&quot; is supplied here, a vector of root frequency parameters can be provided to argument &quot;rpvec&quot;.
</p>
</li></ul>

</td></tr>
<tr><td><code id="estimaterates_f_+3A_rpvec">rpvec</code></td>
<td>

<p>If option &quot;user&quot; is specified for argument &quot;rootprob&quot;, supply a vector of the same length as that provided to argument &quot;alphabet&quot;, representing the root frequency parameters.
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_init">init</code></td>
<td>
	
<p>Initial value for the rates. The default value is 0.9.</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_lowli">lowli</code></td>
<td>

<p>For finer control of the boundaries of the optimization problem. The default value is 0.001. This usually suffices if the branch lengths are in expected substitutions per site. However, if branch lengths are in different units, this should be changed accordingly. 
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_upli">upli</code></td>
<td>

<p>For finer control of the boundaries of the optimization problem. The default value is 100. This usually suffices if the branch lengths are in expected substitutions per site. However, if branch lengths are in different units, this should be changed accordingly. 
</p>
</td></tr>
<tr><td><code id="estimaterates_f_+3A_...">...</code></td>
<td>

<p>Passing other arguments to the optimization algorithm &quot;nlminb&quot;. For example, control = list(trace = 5) will print progress at every 5th iteration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See vignette for detailed examples.
</p>


<h3>Value</h3>

<p>All arguments used while calling the estimaterates function are attached in a list. The following components are also returned in the same list:
</p>
<table>
<tr><td><code>call</code></td>
<td>
<p>Function call used.</p>
</td></tr>
<tr><td><code>conv</code></td>
<td>
<p>A vector of convergence indicators for the model run. 0 denotes successful convergence.</p>
</td></tr>
<tr><td><code>time</code></td>
<td>
<p>Time taken in seconds.</p>
</td></tr>
<tr><td><code>tree</code></td>
<td>
<p>The phylogenetic tree used.</p>
</td></tr>
<tr><td><code>bg</code></td>
<td>
<p>List of group of nodes that capture branches that follow unique substitution rates.</p>
</td></tr>
<tr><td><code>results</code></td>
<td>
<p>List of results including the parameter estimates for each unique entry of modelmat (excluding zeros), standard errors, number of parameters fit, and AIC and BIC values. Furthermore, details from the optimization routine applied are also available. Use ...$results$wop$parsep.</p>
</td></tr>
<tr><td><code>data_red</code></td>
<td>
<p>Unique phyletic patterns observed.</p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>List of number of times each gene phyletic pattern was observed.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>Experimental function
</p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a>
</p>


<h3>References</h3>

<p>Paradis, E., Claude, J. &amp; Strimmer, K. 2004. APE: analyses of phylogenetics and evolution in R language. Bioinformatics 20: 289-290. R package version 3.2.
</p>
<p>Tane Kim and Weilong Hao 2014. DiscML: An R package for estimating evolutionary rates of discrete characters using maximum likelihood. R package version 1.0.1. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>##############

rm(list=ls())
set.seed(1) #set seed
library(markophylo)
tree1 &lt;- ape::rtree(4, br = runif(6, 0.01, 0.3)) #generate a random 4-taxa tree (6 branches) 
bf &lt;- c(0.33, 0.33, 0.34) #vector of equal root frequencies
gf &lt;- 5000 #number of character patterns to be simulated
rootseq &lt;- sample(1:3, gf, replace = TRUE, prob = bf) #simulate the MRCA sequence (1s and 2s).
par1 &lt;- 1.5 
par2 &lt;- 3   
par3 &lt;- 5
Q_sim &lt;- matrix(c(NA, 0.6, par2, par1, NA, 0.8, par2, par3, NA), 3, 3) 
#generated substitution rate matrix
diag(Q_sim) &lt;- -rowSums(Q_sim, na.rm = TRUE)
print(Q_sim)
#simulate data using the geiger package and Q_sim
data_sim &lt;- sapply(1:gf, FUN = function(X) geiger::sim.char(tree1, 
Q_sim, model = "discrete", root = rootseq[X])) 
simulateddat &lt;- list(data = t(data_sim), Q = Q_sim, tree = tree1)

model1 &lt;- markophylo::estimaterates_f(usertree = simulateddat$tree, 
userphyl = simulateddat$data, alphabet = c(1, 2, 3), rootprob = "equal", 
modelmat = matrix(c(NA, 0.6, 2, 1, NA, 0.8, 2, 3, NA), 3, 3))

</code></pre>

<hr>
<h2 id='patterns'>
Unique phyletic patterns with counts.
</h2><span id='topic+patterns'></span>

<h3>Description</h3>

<p>Generate a reduced dataset consisting of unique phyletic patterns
</p>


<h3>Usage</h3>

<pre><code class='language-R'>patterns(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="patterns_+3A_x">x</code></td>
<td>
<p>A matrix or data frame of phyletic patterns. Rows represent the discrete character patterns and columns represent the taxa. These data can be numeric or character. </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>w</code></td>
<td>
<p>Counts of unique phyletic patterns.</p>
</td></tr>
<tr><td><code>databp_red</code></td>
<td>
<p>Reduced data set corresponding to w.</p>
</td></tr>
<tr><td><code>names</code></td>
<td>
<p>Column names of x.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a>
</p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+estimaterates">estimaterates</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simdata1)
uniq_patt &lt;- patterns(simdata1$data)
print(uniq_patt)
</code></pre>

<hr>
<h2 id='plottree'>
Plot the tree used with different hypothesized branch groupings (or clades) following unique rates coloured differently.
</h2><span id='topic+plottree'></span>

<h3>Description</h3>

<p>Plotting command for use on an object of class &quot;markophylo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plottree(x, colors = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plottree_+3A_x">x</code></td>
<td>
<p>An object of class &quot;markophylo&quot;.</p>
</td></tr>
<tr><td><code id="plottree_+3A_colors">colors</code></td>
<td>
<p>Default works well. However, a custom vector of colours can be specified here&mdash;should be the same length as length(x$bg). Note that these colours are used to colour the different branch groupings following their own rates.</p>
</td></tr>
<tr><td><code id="plottree_+3A_...">...</code></td>
<td>
<p>Any further commands to ape::plot.phylo.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Plotting function. See examples</p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+estimaterates">estimaterates</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simdata2)
model2 &lt;- estimaterates(usertree = simdata2$tree, userphyl = simdata2$data, 
                        alphabet = c(1, 2), bgtype = "ancestornodes", bg = c(7),
                        rootprob = "equal", modelmat = matrix(c(NA, 1, 2, NA), 2, 2))
plottree(model2, colors=c("blue", "darkgreen"), edge.width = 2, show.tip.label = FALSE, 
         no.margin = TRUE)
ape::nodelabels(frame = "circle", cex = 0.7)
ape::tiplabels(frame = "circle", cex = 0.7)
</code></pre>

<hr>
<h2 id='print.markophylo'>
Print summary information for the model fit.
</h2><span id='topic+print.markophylo'></span>

<h3>Description</h3>

<p>Summary command for use on an object of class &quot;markophylo&quot;. The estimated substitution rates for each partition specified along with the estimated <code class="reqn">\alpha</code>(s) for the gamma distribution (if any) and prior probability of discrete characters at the root (if estimated) are printed. If branch groupings (or clades) were specified, then the rates (and corresponding standard errors) are displayed in a matrix with the columns representing the different branch groupings (ordered by the subsets of x$bg where x is an object of class &quot;idea&quot;). The rows represent the indices of the model matrix.</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'markophylo'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.markophylo_+3A_x">x</code></td>
<td>

<p>An object of class &quot;markophylo&quot;.</p>
</td></tr>
<tr><td><code id="print.markophylo_+3A_...">...</code></td>
<td>

<p>Ignore this
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Printing function</p>


<h3>Author(s)</h3>

<p>Utkarsh J. Dang and G. Brian Golding
</p>
<p><a href="mailto:utkarshdang@cunet.carleton.ca">utkarshdang@cunet.carleton.ca</a></p>


<h3>See Also</h3>

<p>See also <code><a href="#topic+estimaterates">estimaterates</a></code>.
</p>

<hr>
<h2 id='simdata1'>
Simulated data.
</h2><span id='topic+simdata1'></span><span id='topic+simdata2'></span><span id='topic+simdata3'></span><span id='topic+simdata4'></span><span id='topic+simdata5'></span>

<h3>Description</h3>

<p>Five simulated data sets are included. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("simdata1")
data("simdata2")
data("simdata3")
data("simdata4")
data("simdata5")
</code></pre>


<h3>Format</h3>

<p>Each data set contains a list that comprises a tree (called &quot;tree&quot;) and data (called &quot;data&quot;) as its components. The tree is in the ape package phylo format. The data component consists of a matrix of discrete character patterns with the different patterns as the rows and the taxa as the columns. 
</p>


<h3>Details</h3>

<p>These data were simulated with either the geiger or the phangorn packages.
</p>


<h3>Value</h3>

<p>Simulated datasets. See examples</p>


<h3>References</h3>

<p>Schliep K.P. 2011. phangorn: phylogenetic analysis in R. Bioinformatics, 27(4)  592&ndash;593. R package version 1.99-12.
</p>
<p>Harmon Luke J, Jason T Weir, Chad D Brock, Richard E Glor, and Wendell Challenger. 2008. GEIGER: investigating evolutionary radiations. Bioinformatics 24:129-131. R package version 2.0.3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(simdata1)
data(simdata2)
data(simdata3)
data(simdata4)
data(simdata5)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
