<!DOCTYPE html><html><head><title>Help for package MARVEL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MARVEL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#adhocGene.DE.Gene.10x'><p>Differential gene expression analysis of specified gene</p></a></li>
<li><a href='#adhocGene.DE.PSI.10x'><p>Differential splice junction analysis of specified gene</p></a></li>
<li><a href='#adhocGene.PlotDEValues.10x'><p>Plot differential splice junction analysis results for a specified gene</p></a></li>
<li><a href='#adhocGene.PlotSJPosition.10x'><p>Plots the locations of specified splice junction relative to isoforms</p></a></li>
<li><a href='#adhocGene.TabulateExpression.Gene.10x'><p>Dotplot of gene expression values for a specified gene</p></a></li>
<li><a href='#adhocGene.TabulateExpression.PSI.10x'><p>Dotplot of splice junction expression values for a specified gene</p></a></li>
<li><a href='#AnnotateGenes.10x'><p>Annotate splice junctions</p></a></li>
<li><a href='#AnnotateSJ.10x'><p>Annotate splice junctions</p></a></li>
<li><a href='#AnnoVolcanoPlot'><p>Annotate volcano plot with nonsense-mediated decay (NMD) genes</p></a></li>
<li><a href='#AssignModality'><p>Assign modalities</p></a></li>
<li><a href='#BioPathways'><p>Pathway enrichment analysis</p></a></li>
<li><a href='#BioPathways.10x'><p>Pathway enrichment analysis</p></a></li>
<li><a href='#BioPathways.Plot'><p>Plot pathway enrichment analysis results</p></a></li>
<li><a href='#BioPathways.Plot.10x'><p>Plot pathway enrichment analysis results</p></a></li>
<li><a href='#CheckAlignment'><p>Pre-flight check</p></a></li>
<li><a href='#CheckAlignment.10x'><p>Pre-flight check</p></a></li>
<li><a href='#CheckAlignment.Exp'><p>Check gene data</p></a></li>
<li><a href='#CheckAlignment.PSI'><p>Check splicing data</p></a></li>
<li><a href='#CheckAlignment.PSI.Exp'><p>Check splicing and gene data against each other</p></a></li>
<li><a href='#CheckAlignment.SJ'><p>Check splice junction data</p></a></li>
<li><a href='#CompareExpr'><p>Compares gene expression changes based on nonsense-mediated decay (NMD) status</p></a></li>
<li><a href='#CompareValues'><p>Differential splicing and gene expression analysis</p></a></li>
<li><a href='#CompareValues.Exp'><p>Differential gene expression analysis</p></a></li>
<li><a href='#CompareValues.Exp.Spliced'><p>Differential gene expression analysis for differentially spliced genes</p></a></li>
<li><a href='#CompareValues.Genes.10x'><p>Differential gene expression analysis</p></a></li>
<li><a href='#CompareValues.PSI'><p>Differential splicing analysis</p></a></li>
<li><a href='#CompareValues.SJ.10x'><p>Differential splice junction analysis</p></a></li>
<li><a href='#ComputePSI'><p>Compute percent spliced-in (PSI) values</p></a></li>
<li><a href='#ComputePSI.A3SS'><p>Compute Alternative 3' Splice Site (A3SS) Percent Spliced-in (PSI) Values</p></a></li>
<li><a href='#ComputePSI.A5SS'><p>Compute alternative 5' splice site (A5SS) percent spliced-in (PSI) values</p></a></li>
<li><a href='#ComputePSI.AFE'><p>Compute alternative first exon (AFE) percent spliced-in (PSI) values</p></a></li>
<li><a href='#ComputePSI.ALE'><p>Compute alternative last exon (ALE) percent spliced-in (PSI) values</p></a></li>
<li><a href='#ComputePSI.MXE'><p>Compute mutually exclusive exons (MXE) percent spliced-in (PSI) values</p></a></li>
<li><a href='#ComputePSI.RI'><p>Compute retained-intron (RI) percent spliced-in (PSI) values</p></a></li>
<li><a href='#ComputePSI.SE'><p>Compute skipped-exon (SE) percent spliced-in (PSI) values</p></a></li>
<li><a href='#CountEvents'><p>Tabulate the number of expressed splicing events</p></a></li>
<li><a href='#CreateMarvelObject'><p>Create Marvel object for plate-based RNA-sequencing data</p></a></li>
<li><a href='#CreateMarvelObject.10x'><p>Create Marvel object for droplet-based RNA-sequencing data</p></a></li>
<li><a href='#DetectEvents'><p>Detect Splicing Events</p></a></li>
<li><a href='#DetectEvents.AFE'><p>Detect alternative first exons</p></a></li>
<li><a href='#DetectEvents.AFE.NegStrand'><p>Detect alternative first exons on negative strand</p></a></li>
<li><a href='#DetectEvents.AFE.PosStrand'><p>Detect alternative first exons on positive strand</p></a></li>
<li><a href='#DetectEvents.ALE'><p>Detect alternative last exons</p></a></li>
<li><a href='#DetectEvents.ALE.NegStrand'><p>Detect alternative last exons on negative strand</p></a></li>
<li><a href='#DetectEvents.ALE.PosStrand'><p>Detect alternative last exons on positive strand</p></a></li>
<li><a href='#FilterGenes.10x'><p>Filter specific gene types</p></a></li>
<li><a href='#FindPTC'><p>Find premature terminal codons (PTCs)</p></a></li>
<li><a href='#FindPTC.A3SS.NegStrand'><p>Find premature terminal codon (PTC) for alternative 3' splice site (A3SS) located on the negative strand of the transcript</p></a></li>
<li><a href='#FindPTC.A3SS.PosStrand'><p>Find premature terminal codon (PTC) for alternative 3' splice site (A3SS) located on the positive strand of the transcript</p></a></li>
<li><a href='#FindPTC.A5SS.NegStrand'><p>Find premature terminal codon (PTC) for alternative 5' splice site (A5SS) located on the negative strand of the transcript</p></a></li>
<li><a href='#FindPTC.A5SS.PosStrand'><p>Find premature terminal codon (PTC) for alternative 5' splice site (A5SS) located on the positive strand of the transcript</p></a></li>
<li><a href='#FindPTC.RI.NegStrand'><p>Find premature terminal codon (PTC) for retained-intron (RI) located on the negative strand of the transcript</p></a></li>
<li><a href='#FindPTC.RI.PosStrand'><p>Find premature terminal codon (PTC) for retained-Intron (RI) located on the positive strand of the transcript</p></a></li>
<li><a href='#FindPTC.SE.NegStrand'><p>Find premature terminal codon (PTC) for skipped-exon (SE) located on the negative Strand of the yranscript</p></a></li>
<li><a href='#FindPTC.SE.PosStrand'><p>Find premature terminal codon (PTC) for skipped-exon (SE) located on the positive strand of the transcript</p></a></li>
<li><a href='#IsoSwitch'><p>Classify gene-splicing relationship</p></a></li>
<li><a href='#IsoSwitch.10x'><p>Classify gene-splicing relationship</p></a></li>
<li><a href='#IsoSwitch.PlotExpr'><p>Plot gene-splicing relative change</p></a></li>
<li><a href='#ModalityChange'><p>Classify modality changes</p></a></li>
<li><a href='#ParseGTF'><p>Parse gene transfer file (GTF)</p></a></li>
<li><a href='#PctASE'><p>Tabulate differentially spliced splicing event</p></a></li>
<li><a href='#PlotDEValues'><p>Plot differential splicing and gene expression analysis results</p></a></li>
<li><a href='#PlotDEValues.Exp.Global'><p>Plot global differential gene expression analysis results</p></a></li>
<li><a href='#PlotDEValues.Exp.Spliced'><p>Plot differential gene expression analysis of differentially spliced genes</p></a></li>
<li><a href='#PlotDEValues.Genes.10x'><p>Plot differential gene analysis results</p></a></li>
<li><a href='#PlotDEValues.PSI.Distance'><p>Plot differential splicing analysis results based on distance statistics.</p></a></li>
<li><a href='#PlotDEValues.PSI.Mean'><p>Plot differential splicing analysis results based on mean PSI difference</p></a></li>
<li><a href='#PlotDEValues.PSI.Mean.g2vsg1'><p>Plot differential splicing analysis results based on mean PSI difference</p></a></li>
<li><a href='#PlotDEValues.SJ.10x'><p>Plot differential splice junction analysis results</p></a></li>
<li><a href='#PlotPctExprCells.Genes.10x'><p>Plot gene expression distribution</p></a></li>
<li><a href='#PlotPctExprCells.SJ.10x'><p>Plot splice junction expression distribution</p></a></li>
<li><a href='#PlotValues'><p>Plot percent spliced-in (PSI) or gene expression values</p></a></li>
<li><a href='#PlotValues.Exp'><p>Plot gene expression values</p></a></li>
<li><a href='#PlotValues.PCA.CellGroup.10x'><p>Annotate reduced dimension space with cell feature</p></a></li>
<li><a href='#PlotValues.PCA.Gene.10x'><p>Annotate reduced dimension space with gene expression values</p></a></li>
<li><a href='#PlotValues.PCA.PSI.10x'><p>Annotate reduced dimension space with PSI values</p></a></li>
<li><a href='#PlotValues.PSI'><p>Plot percent spliced-in (PSI) values</p></a></li>
<li><a href='#PropModality'><p>Tabulate modality proportion</p></a></li>
<li><a href='#PropModality.Bar'><p>Modality proportion broken down by event type</p></a></li>
<li><a href='#PropModality.Doughnut'><p>Overall modality proportion</p></a></li>
<li><a href='#PropPTC'><p>Tabulate proportion of transcripts with PTC</p></a></li>
<li><a href='#RunPCA'><p>Principle component analysis</p></a></li>
<li><a href='#RunPCA.Exp'><p>Principle component analysis for gene Data</p></a></li>
<li><a href='#RunPCA.PSI'><p>Principle component analysis for splicing data</p></a></li>
<li><a href='#SubsetCrypticA3SS'><p>Differential gene expression analysis for differentially spliced genes</p></a></li>
<li><a href='#SubsetSamples'><p>Subset samples (cells)</p></a></li>
<li><a href='#TransformExpValues'><p>Transform gene expression Values</p></a></li>
<li><a href='#ValidateSJ.10x'><p>Validate splice junctions</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Revealing Splicing Dynamics at Single-Cell Resolution</td>
</tr>
<tr>
<td>Version:</td>
<td>1.4.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Alternative splicing represents an additional and underappreciated layer of complexity underlying gene expression profiles. Nevertheless, there remains hitherto a paucity of software to investigate splicing dynamics at single-cell resolution. 'MARVEL' enables splicing analysis of single-cell RNA-sequencing data generated from plate- and droplet-based library preparation methods.</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2 (&ge; 3.3.2), Matrix (&ge; 1.3-3), methods, plyr (&ge;
1.8.4), scales (&ge; 1.1.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>AnnotationDbi (&ge; 1.48.0), Biostrings (&ge; 2.56.0), BSgenome
(&ge; 1.56.0), BSgenome.Hsapiens.NCBI.GRCh38 (&ge; 1.3.1000),
clusterProfiler (&ge; 3.16.0), factoextra (&ge; 1.0.7), FactoMineR
(&ge; 2.3), fitdistrplus (&ge; 1.1-1), GenomicRanges (&ge; 1.42.0),
ggnewscale (&ge; 0.4.5), ggrepel (&ge; 0.9.1), gridExtra (&ge; 2.3),
gtools (&ge; 3.9.2), IRanges (&ge; 2.24.1), kableExtra (&ge; 1.3.1),
knitr (&ge; 1.29), kSamples (&ge; 1.2-9), markdown (&ge; 1.1), MAST
(&ge; 1.16.0), org.Hs.eg.db (&ge; 3.10.0), org.Mm.eg.db (&ge;
3.11.4), parallel, pheatmap (&ge; 1.0.12), reshape2 (&ge; 1.4.4),
rmarkdown (&ge; 2.3), S4Vectors (&ge; 0.26.1), stringr (&ge; 1.4.0),
textclean (&ge; 0.9.3), twosamples (&ge; 1.1.1), wiggleplotr (&ge;
1.12.1)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-30 21:43:47 UTC; seanwen</td>
</tr>
<tr>
<td>Author:</td>
<td>Sean Wen [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Sean Wen &lt;sean.wenwx@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-31 10:22:50 UTC</td>
</tr>
</table>
<hr>
<h2 id='adhocGene.DE.Gene.10x'>Differential gene expression analysis of specified gene</h2><span id='topic+adhocGene.DE.Gene.10x'></span>

<h3>Description</h3>

<p>Performs differential gene expression analysis specified gene across for all possible pairs of cell groups. The gene and cell groups were defined earlier in <code>adhocGene.TabulateExpression.Gene.10x</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adhocGene.DE.Gene.10x(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adhocGene.DE.Gene.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>adhocGene.TabulateExpression.Gene.10x</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$adhocGene$DE$Gene$Data</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- adhocGene.DE.Gene.10x(MarvelObject=marvel.demo.10x)

# Check output
marvel.demo.10x$adhocGene$DE$Gene$Data
</code></pre>

<hr>
<h2 id='adhocGene.DE.PSI.10x'>Differential splice junction analysis of specified gene</h2><span id='topic+adhocGene.DE.PSI.10x'></span>

<h3>Description</h3>

<p>Performs differential splice junction analysis specified gene across for all possible pairs of cell groups. The gene and cell groups were defined earlier in <code>adhocGene.TabulateExpression.Gene.10x</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adhocGene.DE.PSI.10x(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adhocGene.DE.PSI.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>adhocGene.TabulateExpression.PSI.10x</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$adhocGene$DE$PSI$Data</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- adhocGene.DE.PSI.10x(MarvelObject=marvel.demo.10x)

# Check output
marvel.demo.10x$adhocGene$DE$PSI$Data
</code></pre>

<hr>
<h2 id='adhocGene.PlotDEValues.10x'>Plot differential splice junction analysis results for a specified gene</h2><span id='topic+adhocGene.PlotDEValues.10x'></span>

<h3>Description</h3>

<p>Scatterplot of results from differential gene and splice junction analysis. x-axis represents the gene expression log2 fold change between the different pairs of cell groups. y-axis represents the PSI differences or log2 fold change between the different pairs of cell groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adhocGene.PlotDEValues.10x(
  MarvelObject,
  coord.intron,
  log2fc.gene = 0.5,
  delta.sj = 5,
  label.size = 2,
  point.size = 2,
  xmin = NULL,
  xmax = NULL,
  ymin = NULL,
  ymax = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>adhocGene.DE.Gene.10x</code> and <code>adhocGene.DE.PSI.10x</code> functions.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_coord.intron">coord.intron</code></td>
<td>
<p>Character string. Coordinates of splice junction whose differential splice junction results will be plotted.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_log2fc.gene">log2fc.gene</code></td>
<td>
<p>Numeric value. Absolute log2 fold change, above which, the gene is considered differentially expressed.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_delta.sj">delta.sj</code></td>
<td>
<p>Numeric value. Absolute differences in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. The font size of the group comparison labels on the plot will be adjusted to the size specified here. Default is <code>2</code>.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>2</code>.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_xmin">xmin</code></td>
<td>
<p>Numeric value. Minimum x-axis value.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_xmax">xmax</code></td>
<td>
<p>Numeric value. Maximum x-axis value.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_ymin">ymin</code></td>
<td>
<p>Numeric value. Minimum y-axis value.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotDEValues.10x_+3A_ymax">ymax</code></td>
<td>
<p>Numeric value. Maximum y-axis value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$adhocGene$DE$VolcanoPlot$Plot</code> and <code>MarvelObject$adhocGene$DE$VolcanoPlot$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define SJ to plot
coord.intron &lt;- marvel.demo.10x$adhocGene$DE$PSI$Data$coord.intron[1]

# Plot SJ vs gene
marvel.demo.10x &lt;- adhocGene.PlotDEValues.10x(
                        MarvelObject=marvel.demo.10x,
                        coord.intron=coord.intron,
                        log2fc.gene=0.5,
                        delta.sj=5,
                        label.size=2,
                        point.size=2,
                        xmin=-2.0,
                        xmax=2.0,
                        ymin=-25,
                        ymax=25
                        )

# Check output
marvel.demo.10x$adhocGene$DE$VolcanoPlot$Plot
</code></pre>

<hr>
<h2 id='adhocGene.PlotSJPosition.10x'>Plots the locations of specified splice junction relative to isoforms</h2><span id='topic+adhocGene.PlotSJPosition.10x'></span>

<h3>Description</h3>

<p>Plots the locations of specified splice junction relative to isoforms. List of isoforms are retrieved from GTF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adhocGene.PlotSJPosition.10x(
  MarvelObject,
  coord.intron,
  coord.intron.ext = 50,
  rescale_introns = FALSE,
  show.protein.coding.only = TRUE,
  anno.label.size = 3,
  anno.colors = c("black", "gray", "red")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_coord.intron">coord.intron</code></td>
<td>
<p>Character string. Coordinates of splice junction whose splice junction will be plotted.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_coord.intron.ext">coord.intron.ext</code></td>
<td>
<p>Numeric value. Number of bases to extend the splice junction start and end coordinates into the exons. Helpful to enhance splice junction locations on the plot. Default is <code>50</code>.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_rescale_introns">rescale_introns</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the intron length will be shorten. Helpful when introns are very long and focus visualisation of exons and splice junctions. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_show.protein.coding.only">show.protein.coding.only</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), only protein-coding isoforms will be displayed.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_anno.label.size">anno.label.size</code></td>
<td>
<p>Numeric value. Font size of isoform ID labels. Default is <code>3</code>.</p>
</td></tr>
<tr><td><code id="adhocGene.PlotSJPosition.10x_+3A_anno.colors">anno.colors</code></td>
<td>
<p>Vector of character strings. Colors for non-coding UTRs, coding exons, and splice junctions, respectively. Default is <code>c("black", "gray", "red")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$adhocGene$SJPosition$Plot</code>, <code>MarvelObject$adhocGene$SJPosition$metadata</code>, <code>MarvelObject$adhocGene$SJPosition$exonfile</code>, and <code>MarvelObject$adhocGene$SJPosition$cdsfile</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- adhocGene.PlotSJPosition.10x(
                        MarvelObject=marvel.demo.10x,
                        coord.intron="chr1:100:1001",
                        rescale_introns=FALSE,
                        show.protein.coding.only=TRUE,
                        anno.label.size=1.5
                        )
</code></pre>

<hr>
<h2 id='adhocGene.TabulateExpression.Gene.10x'>Dotplot of gene expression values for a specified gene</h2><span id='topic+adhocGene.TabulateExpression.Gene.10x'></span>

<h3>Description</h3>

<p>Creates a dotplot of average expression value of a specified gene across different cell groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adhocGene.TabulateExpression.Gene.10x(
  MarvelObject,
  cell.group.list,
  gene_short_name,
  log2.transform = TRUE,
  min.pct.cells = 10,
  downsample = FALSE,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_cell.group.list">cell.group.list</code></td>
<td>
<p>List of character strings. Each element of the list is a vector of cell IDs corresponding to a cell group.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_gene_short_name">gene_short_name</code></td>
<td>
<p>Character string. Gene names whose expression will be plotted.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_log2.transform">log2.transform</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), normalised gene expression values will be off-set by 1 and then log2-transformed prior to plotting.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_min.pct.cells">min.pct.cells</code></td>
<td>
<p>Numeric value. Percentage of cell expressing the gene in a cell group, below which, the value be re-coded as missing and appear will be omitted from the plot. A gene is considered to be expressed in a given cell if it has non-zero normalised count.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the number of cells in each cell group will be down-sampled so that all cell groups will have the same number of cells. The number of cells to down-sample will be based on the smallest cell group. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.Gene.10x_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Random number generator to be fixed for down-sampling.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$adhocGene$Expression$Gene$Table</code>, <code>MarvelObject$adhocGene$Expression$Gene$Plot</code>, <code>MarvelObject$adhocGene$cell.group.list</code>, and <code>MarvelObject$adhocGene$gene_short_name</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # iPSC
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Cardio day 10
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

    # Save into list
    cell.group.list &lt;- list("iPSC"=cell.ids.1,
                            "Cardio d10"=cell.ids.2
                            )

# Gene expression profiling
marvel.demo.10x &lt;- adhocGene.TabulateExpression.Gene.10x(
                        MarvelObject=marvel.demo.10x,
                        cell.group.list=cell.group.list,
                        gene_short_name="TPM2",
                        min.pct.cells=10,
                        downsample=TRUE
                        )

# Check output
marvel.demo.10x$adhocGene$Expression$Gene$Plot
marvel.demo.10x$adhocGene$Expression$Gene$Table
</code></pre>

<hr>
<h2 id='adhocGene.TabulateExpression.PSI.10x'>Dotplot of splice junction expression values for a specified gene</h2><span id='topic+adhocGene.TabulateExpression.PSI.10x'></span>

<h3>Description</h3>

<p>Creates a dotplot of splice junction expression value of a specified gene across different cell groups. The gene and cell groups were defined earlier in <code>adhocGene.TabulateExpression.Gene.10x</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adhocGene.TabulateExpression.PSI.10x(MarvelObject, min.pct.cells = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adhocGene.TabulateExpression.PSI.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>adhocGene.TabulateExpression.Gene.10x</code> function.</p>
</td></tr>
<tr><td><code id="adhocGene.TabulateExpression.PSI.10x_+3A_min.pct.cells">min.pct.cells</code></td>
<td>
<p>Numeric value. Percentage of cell expressing the splice junction in a cell group, below which, the value be re-coded as missing and appear will be omitted from the plot. A splice junction is considered to be expressed in a given cell if it has count &gt;=1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$adhocGene$Expression$PSI$Table</code> and  <code>MarvelObject$adhocGene$Expression$PSI$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# SJ usage profiling
marvel.demo.10x &lt;- adhocGene.TabulateExpression.PSI.10x(
                        MarvelObject=marvel.demo.10x,
                        min.pct.cells=10
                        )

# Check output
marvel.demo.10x$adhocGene$Expression$PSI$Plot
</code></pre>

<hr>
<h2 id='AnnotateGenes.10x'>Annotate splice junctions</h2><span id='topic+AnnotateGenes.10x'></span>

<h3>Description</h3>

<p>Annotates the each gene in the gene metadata with the gene type, e.g. protein-coding, antisense etc.. Annotations are retrieved from GTF. Only genes found in gene metadata and GTF will be retained.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnnotateGenes.10x(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AnnotateGenes.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject.10x</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing the updated slots <code>MarvelObject$gene.metadata</code> and <code>gene.norm.matrix</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load un-processed MARVEL object
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )

# Annotate gene metadata
marvel.demo.10x &lt;- AnnotateGenes.10x(MarvelObject=marvel.demo.10x.raw)
</code></pre>

<hr>
<h2 id='AnnotateSJ.10x'>Annotate splice junctions</h2><span id='topic+AnnotateSJ.10x'></span>

<h3>Description</h3>

<p>Annotates the splice junctions by assigning the gene name to the start and end of the splice junction. Annotations are retrieved from GTF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnnotateSJ.10x(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AnnotateSJ.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>AnnotateGenes.10x</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing the updated slot <code>MarvelObject$sj.metadata</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load un-processed MARVEL object
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )

# Annotate gene metadata
marvel.demo.10x &lt;- AnnotateGenes.10x(MarvelObject=marvel.demo.10x.raw)

# Annotate junction metadata
marvel.demo.10x &lt;- AnnotateSJ.10x(MarvelObject=marvel.demo.10x)
</code></pre>

<hr>
<h2 id='AnnoVolcanoPlot'>Annotate volcano plot with nonsense-mediated decay (NMD) genes</h2><span id='topic+AnnoVolcanoPlot'></span>

<h3>Description</h3>

<p>Annotate volcano plot generated from differential gene expression analysis with genes predicted to undergo splicing-induced NMD.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AnnoVolcanoPlot(
  MarvelObject,
  anno = FALSE,
  anno.gene_short_name = NULL,
  label.size = NULL,
  point.size = 1,
  xlabel.size = 8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AnnoVolcanoPlot_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareExpr</code> function.</p>
</td></tr>
<tr><td><code id="AnnoVolcanoPlot_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, selected gene names will be annotated on the plot as defined in <code>gene.label.x.below</code> and <code>gene.label.y.above</code>.</p>
</td></tr>
<tr><td><code id="AnnoVolcanoPlot_+3A_anno.gene_short_name">anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the gene names to annotate on the plot.</p>
</td></tr>
<tr><td><code id="AnnoVolcanoPlot_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. When <code>anno</code> set to <code>TRUE</code>, the size of gene labels.</p>
</td></tr>
<tr><td><code id="AnnoVolcanoPlot_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="AnnoVolcanoPlot_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$NMD$AnnoVolcanoPlot$Table</code> and <code>MarvelObject$NMD$AnnoVolcanoPlot$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- AnnoVolcanoPlot(MarvelObject=marvel.demo)

# Check outputs
head(marvel.demo$NMD$AnnoVolcanoPlot$Table)
marvel.demo$NMD$AnnoVolcanoPlot$Plot
</code></pre>

<hr>
<h2 id='AssignModality'>Assign modalities</h2><span id='topic+AssignModality'></span>

<h3>Description</h3>

<p>Assigns modalities to each splicing event for a specified group of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AssignModality(
  MarvelObject,
  sample.ids,
  min.cells = 25,
  sigma.sq = 0.001,
  bimodal.adjust = TRUE,
  bimodal.adjust.fc = 3,
  bimodal.adjust.diff = 50,
  seed = 1,
  tran_ids = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="AssignModality_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_sample.ids">sample.ids</code></td>
<td>
<p>Vector of character strings. Sample IDs that constitute the cell group.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the splicing event for the event to be included for modality assignment.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_sigma.sq">sigma.sq</code></td>
<td>
<p>Numeric value. The variance threshold below which the included/excluded modality will be defined as primary sub-modality, and above which it will be defined as dispersed sub-modality.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_bimodal.adjust">bimodal.adjust</code></td>
<td>
<p>Logical. When set to <code>TRUE</code>, MARVEL will identify false bimodal modalities and reassign them as included/excluded modality.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_bimodal.adjust.fc">bimodal.adjust.fc</code></td>
<td>
<p>Numeric value. The ratio between the proportion of cells with &gt;0.75 PSI vs &lt;0.25 PSI (and vice versa) below which the splicing event will be classified as bimodal. Only applicable when <code>bimodal.adjust</code> set to <code>TRUE</code>. To be used in conjunction with <code>bimodal.adjust.diff</code>.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_bimodal.adjust.diff">bimodal.adjust.diff</code></td>
<td>
<p>Numeric value. The difference between the percentage of cells with &gt;0.75 PSI vs &lt;0.25 PSI (and vice versa) below which the splicing event will be classified as bimodal. Only applicable when <code>bimodal.adjust</code> set to <code>TRUE</code>. To be used in conjunction with <code>bimodal.adjust.fc</code>.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Ensure the <code>fitdist</code> function returns the same values for alpha and beta paramters each time this function is executed using the same random number generator.</p>
</td></tr>
<tr><td><code id="AssignModality_+3A_tran_ids">tran_ids</code></td>
<td>
<p>Character strings. Specific vector of transcript IDs for modality assignment. This will be a subset of all transcripts expressed in sufficient number of cells as defined in <code>min.cells</code> option.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>MarvelObject$Modality$Results</code>.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

df.pheno &lt;- marvel.demo$SplicePheno
sample.ids &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]

# Assign modality
marvel.demo &lt;- AssignModality(MarvelObject=marvel.demo,
                              sample.ids=sample.ids,
                              min.cells=5
                              )

# Check output
head(marvel.demo$Modality$Results)
</code></pre>

<hr>
<h2 id='BioPathways'>Pathway enrichment analysis</h2><span id='topic+BioPathways'></span>

<h3>Description</h3>

<p>Performs pathway enrichment analysis on differentially spliced genes or user-specified custom set of genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioPathways(
  MarvelObject,
  method = NULL,
  pval = NULL,
  delta = 0,
  n.top = NULL,
  method.adjust = "fdr",
  custom.genes = NULL,
  species = "human"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioPathways_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Alternative to <code>n.top</code> and <code>custom.genes</code>, i.e. choose one of these three options. Adjusted p-value below which the splicing events are considered differentially spliced and their corresponding genes are included for gene ontology analysis. If this argument is specified, then <code>n.top</code> must not be specified.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_delta">delta</code></td>
<td>
<p>Numeric value. The absolute difference between the means PSI values of cell group 1 and 2, above which, the splicing event is considered differentially spliced and their corresponding genes are included for gene ontology analysis.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_n.top">n.top</code></td>
<td>
<p>Numeric value. Alternative to <code>pval</code> to <code>custom.genes</code>, i.e. choose one of these three options.. Indicate the top n splicing events with the smallest adjusted p-values are differentially spliced and their corresponding genes are included for gene ontology analysis. If this argument is specified, then <code>pval</code> must not be specified.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_method.adjust">method.adjust</code></td>
<td>
<p>Character string. Adjust p-values for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_custom.genes">custom.genes</code></td>
<td>
<p>Character strings. Alternative to <code>pval</code> and <code>n.top</code>, i.e. choose one of these three options.. Vector of gene names to be assessed for enrichment of biological pathways.</p>
</td></tr>
<tr><td><code id="BioPathways_+3A_species">species</code></td>
<td>
<p>Character strings. Takes the value <code>"human"</code> or <code>"mouse"</code>, which corresponds to human and mouse genes, respectively. Default value is <code>"human"</code>. This will enable <code>MARVEL</code> to retrieve the relevant database for GO analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$DE$BioPathways$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- BioPathways(MarvelObject=marvel.demo,
                           method="ad",
                           custom.genes=c("RPL26", "SNRPN")
                           )
</code></pre>

<hr>
<h2 id='BioPathways.10x'>Pathway enrichment analysis</h2><span id='topic+BioPathways.10x'></span>

<h3>Description</h3>

<p>Performs pathway enrichment analysis on differentially spliced genes or user-specified custom set of genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioPathways.10x(
  MarvelObject,
  pval = 0.05,
  log2fc = NULL,
  delta = 5,
  min.gene.norm = 0,
  method.adjust = "fdr",
  custom.genes = NULL,
  species = "human",
  remove.ribo = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioPathways.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_pval">pval</code></td>
<td>
<p>Numeric value. p-value, above which, the splice junction is considered differentially spliced. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_log2fc">log2fc</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential splicing analysis, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>delta</code> has been specified.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_delta">delta</code></td>
<td>
<p>Numeric value. Absolute difference in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>log2fc</code> has been specified.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_min.gene.norm">min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction is considered differentially spliced. Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_method.adjust">method.adjust</code></td>
<td>
<p>Character string. Adjust p-values for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_custom.genes">custom.genes</code></td>
<td>
<p>Character strings. Alternative to <code>pval</code> and <code>delta</code>. Vector of gene names to be assessed for enrichment of biological pathways.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_species">species</code></td>
<td>
<p>Character strings. Takes the value <code>"human"</code> or <code>"mouse"</code>, which corresponds to human and mouse genes, respectively. Default value is <code>"human"</code>.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_remove.ribo">remove.ribo</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, ribosomal genes will be removed prior to GO analysis. This may prevent high-expressing ribosomal genes from overshadowing more biological relevant genes for GO analysis. Default value is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="BioPathways.10x_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing new slot <code>MarvelObject$DE$BioPathways$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- BioPathways.10x(
                        MarvelObject=marvel.demo.10x,
                        custom.genes=c("TPM2", "GNAS"),
                        species="human"
                        )
</code></pre>

<hr>
<h2 id='BioPathways.Plot'>Plot pathway enrichment analysis results</h2><span id='topic+BioPathways.Plot'></span>

<h3>Description</h3>

<p>Plots user-specified enriched pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioPathways.Plot(
  MarvelObject,
  go.terms,
  y.label.size = 10,
  offset = 0.5,
  x.axis = "enrichment"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioPathways.Plot_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>BioPathways</code> function.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot_+3A_go.terms">go.terms</code></td>
<td>
<p>Vector of character strings. Names of pathways to plot. Should match pathway names in column <code>Description</code> of <code>MarvelObject$DE$BioPathways$Table</code>.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot_+3A_y.label.size">y.label.size</code></td>
<td>
<p>Numeric value. Size of y-axis tick labels, i.e. gene set names.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot_+3A_offset">offset</code></td>
<td>
<p>Numeric value. The -log10(p-value) on the x-axis to substract or add to increase the plot margins.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot_+3A_x.axis">x.axis</code></td>
<td>
<p>Character string. If set to <code>"enrichment"</code> (default) the pathway enrichment will be displayed on the x-axis while the color intensity of the data points will reflect the -log10(adjusted p-value). If set to <code>"pval"</code> the -log10(adjusted p-value) will be displayed on the x-axis while the color intensity of the data points will reflect the pathway enrichment.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function plots selected gene sets returned from gene ontology analysis performed previously using <code>BioPathways</code>
</p>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$DE$BioPathways$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define go terms to plot
df &lt;- marvel.demo$DE$BioPathways$Table
go.terms &lt;- df$Description[c(1:10)]

# Plot
marvel.demo &lt;- BioPathways.Plot(MarvelObject=marvel.demo,
                                go.terms=go.terms,
                                offset=10
                                )

# Check output
marvel.demo$DE$BioPathways$Plot
</code></pre>

<hr>
<h2 id='BioPathways.Plot.10x'>Plot pathway enrichment analysis results</h2><span id='topic+BioPathways.Plot.10x'></span>

<h3>Description</h3>

<p>Plots user-specified enriched pathways.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BioPathways.Plot.10x(MarvelObject, go.terms, y.label.size = 10, offset = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BioPathways.Plot.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>BioPathways.10x</code> function.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot.10x_+3A_go.terms">go.terms</code></td>
<td>
<p>Vector of character strings. Names of pathways to plot. Should match pathway names in column <code>Description</code> of <code>MarvelObject$DE$BioPathways$Table</code>.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot.10x_+3A_y.label.size">y.label.size</code></td>
<td>
<p>Numeric value. Size of y-axis tick labels, i.e. pathway names.</p>
</td></tr>
<tr><td><code id="BioPathways.Plot.10x_+3A_offset">offset</code></td>
<td>
<p>Numeric value. The value on the x-axis to substract or add to increase the plot margins.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>MarvelObject$DE$BioPathways$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define top pathways to plot
go.terms &lt;- marvel.demo.10x$DE$BioPathways$Table$Description
go.terms &lt;- go.terms[c(1:10)]

# Plot
marvel.demo.10x &lt;- BioPathways.Plot.10x(
                            MarvelObject=marvel.demo.10x,
                            go.terms=go.terms
                            )
# Check outpout
marvel.demo.10x$DE$BioPathways$Plot
</code></pre>

<hr>
<h2 id='CheckAlignment'>Pre-flight check</h2><span id='topic+CheckAlignment'></span>

<h3>Description</h3>

<p>Checks if the metadata aligns with the columns and rows of the matrix for splicing or gene data. This is a wrapper function for <code>CheckAlignment.PSI</code>, <code>CheckAlignment.Exp</code>, <code>CheckAlignment.PSI.Exp</code>, and <code>CheckAlignment.SJ</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlignment(MarvelObject, level)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckAlignment_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="CheckAlignment_+3A_level">level</code></td>
<td>
<p>Character string. Indicate <code>"SJ"</code>, <code>"splicing"</code> or <code>"gene"</code> for splice junction, splicing or gene data, respectively. <code>"SJ"</code> typically specified before computing PSI values. <code>"splicing"</code> or <code>"gene"</code> typically specified after computing PSI values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slots <code>MarvelObject$SpliceJunction</code>, <code>MarvelObject$IntronCoverage</code>, <code>MarvelObject$SplicePheno</code>, <code>MarvelObject$SpliceFeatureValidated</code>, and <code>MarvelObject$PSI</code> or <code>MarvelObject$GenePheno</code>, <code>MarvelObject$GeneFeature</code>, and <code>MarvelObject$Gene</code> are updated for splicing or gene data, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- CheckAlignment(MarvelObject=marvel.demo,
                              level="SJ"
                              )
</code></pre>

<hr>
<h2 id='CheckAlignment.10x'>Pre-flight check</h2><span id='topic+CheckAlignment.10x'></span>

<h3>Description</h3>

<p>Ensures only overlapping cells found in both gene and splice junction data are retained. Also ensures matrix columns matches cell IDs in sample metadata and matrix rows matches gene name or splice junction coordinates in feature metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlignment.10x(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckAlignment.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>FilterGenes.10x</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing updated slots <code>MarvelObject$gene.norm.matrix</code>, <code>MarvelObject$sample.metadata</code>, <code>MarvelObject$gene.metadata</code>, <code>MarvelObject$gene.count.matrix</code>, <code>MarvelObject$sj.count.matrix</code>, <code>MarvelObject$sj.metadata</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load un-processed MARVEL object
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )

# Annotate gene metadata
marvel.demo.10x &lt;- AnnotateGenes.10x(MarvelObject=marvel.demo.10x.raw)

# Annotate junction metadata
marvel.demo.10x &lt;- AnnotateSJ.10x(MarvelObject=marvel.demo.10x)

# Validate junctions
marvel.demo.10x &lt;- ValidateSJ.10x(MarvelObject=marvel.demo.10x)

# Subset CDS genes
marvel.demo.10x &lt;- FilterGenes.10x(MarvelObject=marvel.demo.10x,
                          gene.type="protein_coding"
                          )

# Pre-flight check
marvel.demo.10x &lt;- CheckAlignment.10x(MarvelObject=marvel.demo.10x)
</code></pre>

<hr>
<h2 id='CheckAlignment.Exp'>Check gene data</h2><span id='topic+CheckAlignment.Exp'></span>

<h3>Description</h3>

<p>Checks if the metadata aligns with the columns and rows of the matrix for gene data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlignment.Exp(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckAlignment.Exp_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slots <code>MarvelObject$SplicePheno</code>, <code>MarvelObject$SpliceFeature</code>, and <code>MarvelObject$PSI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- CheckAlignment.Exp(MarvelObject=marvel.demo)
</code></pre>

<hr>
<h2 id='CheckAlignment.PSI'>Check splicing data</h2><span id='topic+CheckAlignment.PSI'></span>

<h3>Description</h3>

<p>Checks if the metadata aligns with the columns and rows of the matrix for splicing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlignment.PSI(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckAlignment.PSI_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slots <code>MarvelObject$SplicePheno</code>, <code>MarvelObject$SpliceFeature</code>, and <code>MarvelObject$PSI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- CheckAlignment.PSI(MarvelObject=marvel.demo)
</code></pre>

<hr>
<h2 id='CheckAlignment.PSI.Exp'>Check splicing and gene data against each other</h2><span id='topic+CheckAlignment.PSI.Exp'></span>

<h3>Description</h3>

<p>Subsets overlapping samples between splicing and gene data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlignment.PSI.Exp(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckAlignment.PSI.Exp_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CheckAlignment.PSI</code> and <code>CheckAlignment.Exp</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slots <code>MarvelObject$SplicePheno</code>, <code>MarvelObject$PSI</code>, <code>MarvelObject$GenePheno</code>, and <code>MarvelObject$Exp</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- CheckAlignment.PSI.Exp(MarvelObject=marvel.demo)
</code></pre>

<hr>
<h2 id='CheckAlignment.SJ'>Check splice junction data</h2><span id='topic+CheckAlignment.SJ'></span>

<h3>Description</h3>

<p>Checks if the metadata aligns with the columns and rows of the matrix for splice junction data prior to PSI computation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckAlignment.SJ(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckAlignment.SJ_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slots <code>MarvelObject$SplicePheno</code>, <code>MarvelObject$PSI</code> and <code>MarvelObject$IntronCounts</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- CheckAlignment.SJ(MarvelObject=marvel.demo)
</code></pre>

<hr>
<h2 id='CompareExpr'>Compares gene expression changes based on nonsense-mediated decay (NMD) status</h2><span id='topic+CompareExpr'></span>

<h3>Description</h3>

<p>Compares gene expression changes based on NMD status for each splicing event type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareExpr(MarvelObject, xlabels.size = 8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareExpr_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>FindPTC</code> function.</p>
</td></tr>
<tr><td><code id="CompareExpr_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Size of the x-axis tick labels. Default is 8.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 new slots <code>MarvelObject$NMD$NMD.Expr$Table</code>, <code>MarvelObject$NMD$NMD.Expr$Plot</code>, and <code>MarvelObject$NMD$NMD.Expr$Plot.Stats</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- CompareExpr(MarvelObject=marvel.demo)

# Check outputs
head(marvel.demo$NMD$NMD.Expr$Table)
marvel.demo$NMD$NMD.Expr$Plot
marvel.demo$NMD$NMD.Expr$Plot.Stats
</code></pre>

<hr>
<h2 id='CompareValues'>Differential splicing and gene expression analysis</h2><span id='topic+CompareValues'></span>

<h3>Description</h3>

<p>Performs differential splicing and gene expression analysis between 2 groups of cells. This is a wrapper function for <code>CompareValues.PSI</code> and <code>CompareValues.Exp</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareValues(
  MarvelObject,
  cell.group.g1 = NULL,
  cell.group.g2 = NULL,
  downsample = FALSE,
  seed = 1,
  min.cells = 25,
  pct.cells = NULL,
  method = NULL,
  nboots = 1000,
  n.permutations = 1000,
  method.adjust = "fdr",
  level,
  event.type = NULL,
  show.progress = TRUE,
  annotate.outliers = TRUE,
  n.cells.outliers = 10,
  assign.modality = TRUE,
  custom.gene_ids = NULL,
  psi.method = NULL,
  psi.pval = NULL,
  psi.delta = NULL,
  method.de.gene = NULL,
  method.adjust.de.gene = NULL,
  mast.method = "bayesglm",
  mast.ebayes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareValues_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 1 (reference group).</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 2.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the number of cells in each cell group will be downsampled to the sample size of the smaller cell group so that both cell groups will have the sample size prior to differential expression analysis. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_seed">seed</code></td>
<td>
<p>Numeric value. The seed number for the random number generator to ensure reproducibility during during down-sampling of cells when <code>downsample</code> set to <code>TRUE</code>, during permutation testing when <code>method</code> set to <code>"permutation"</code>, and during modality assignment which will be performed automatically.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the splicing event or genes for the event or genes to be included for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_pct.cells">pct.cells</code></td>
<td>
<p>Numeric value. The minimum percentage of cells expressing the splicing event or genes for the event or genes to be included for differential splicing analysis. If <code>pct.cells</code> is specified, then <code>pct.cells</code> will be used as threshold instead of <code>min.cells</code>.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_method">method</code></td>
<td>
<p>Character string. Statistical test to compare the 2 groups of cells. <code>"ks"</code>, <code>"kuiper"</code>, <code>"ad"</code>, <code>"dts"</code>, <code>"wilcox"</code>, and <code>"t.test"</code> for Kolmogorov-Smirnov, Kuiper, Anderson-Darling, DTS, Wilcox, and t-test, respectively. Additional <code>"mast"</code> option is available for differential gene expression analysis. If <code>"mast"</code> is specified, the log2fc and p-values will be corrected using the gene detection rate as per the <code>MAST</code> package tutorial.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_nboots">nboots</code></td>
<td>
<p>Numeric value.Only applicable when <code>level</code> set to <code>"splicing"</code>.  When <code>method</code> set to <code>"dts"</code>, the number of bootstrap iterations for computing the p-value.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_n.permutations">n.permutations</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. When <code>method</code> set to <code>"permutation"</code>, this argument indicates the number of permutations to perform for generating the null distribution for subsequent p-value inference. Default is <code>1000</code> times.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_method.adjust">method.adjust</code></td>
<td>
<p>Character string. Adjust p-values for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_level">level</code></td>
<td>
<p>Character string. Indicate <code>"splicing"</code> or <code>"gene"</code> for differential splicing or gene expression analysis, respectively.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_event.type">event.type</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"splicing"</code>. Indicate which splicing event type to include for analysis. Can take value <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, or <code>"A3SS"</code> which represents skipped-exon (SE), mutually-exclusive exons (MXE), retained-intron (RI), alternative 5' splice site (A5SS), and alternative 3' splice site (A3SS), respectively.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_show.progress">show.progress</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, progress bar will be displayed so that users can estimate the time needed for differential analysis. Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_annotate.outliers">annotate.outliers</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. When set to <code>TRUE</code>, statistical difference in PSI values between the two cell groups that is driven by outlier cells will be annotated.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_n.cells.outliers">n.cells.outliers</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. When <code>method</code> set to <code>"dts"</code>, the minimum number of cells with non-1 or non-0 PSI values for included-to-included or excluded-to-excluded modality change, respectively. The p-values will be re-coded to 1 when both cell groups have less than this minimum number of cells. This is to avoid false positive results.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_assign.modality">assign.modality</code></td>
<td>
<p>Logical value. Only applicable when <code>level</code> set to <code>"splicing"</code>. If set to <code>TRUE</code> (default), modalities will be assigned to each cell group.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_custom.gene_ids">custom.gene_ids</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"gene"</code>. Instead of specified the genes to include for DE analysis with <code>min.cells</code>, users may input a custom vector of gene IDs to include for DE analysis.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_psi.method">psi.method</code></td>
<td>
<p>Vector of character string(s). Only applicable when <code>level</code> set to <code>"gene.spliced"</code> and when <code>CompareValues</code> function has been ran with <code>level</code> set to <code>"splicing"</code> earlier. To include significant events from these method(s) for differential gene expression analysis.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Vector of numeric value(s). Only applicable when <code>level</code> set to <code>"gene.spliced"</code> and when <code>CompareValues</code> function has been ran with <code>level</code> set to <code>"splicing"</code> earlier. The adjusted p-value, below which, the splicing event is considered differentially spliced, and the corresponding genes will be included for differential gene expression analysis.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_psi.delta">psi.delta</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code> and when <code>CompareValues</code> function has been ran with <code>level</code> set to <code>"splicing"</code> earlier. The absolute difference in mean PSI values between <code>cell.group.g1</code> and <code>cell.group.g1</code>, above which, the splicing event is considered differentially spliced, and the corresponding genes will be included for differential gene expression analysis.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_method.de.gene">method.de.gene</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"gene.spliced"</code> and when <code>CompareValues</code> function has been ran with <code>level</code> set to <code>"splicing"</code> earlier. Same as <code>method</code>.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_method.adjust.de.gene">method.adjust.de.gene</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"gene.spliced"</code> and when <code>CompareValues</code> function has been ran with <code>level</code> set to <code>"splicing"</code> earlier. Same as <code>method.adjust</code>.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_mast.method">mast.method</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"gene"</code> or <code>"gene.spliced"</code>. As per the <code>method</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>"bayesglm"</code>, other options are <code>"glm"</code> and <code>"glmer"</code>.</p>
</td></tr>
<tr><td><code id="CompareValues_+3A_mast.ebayes">mast.ebayes</code></td>
<td>
<p>Logical value. Only applicable when <code>level</code> set to <code>"gene"</code> or <code>"gene.spliced"</code>. As per the <code>ebayes</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>MarvelObject$DE$PSI$Table[["method"]]</code> or <code>MarvelObject$DE$Exp$Table</code> when <code>level</code> option specified as <code>"splicing"</code> or <code>"gene"</code>, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell groups for analysis
df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]

# DE
marvel.demo &lt;- CompareValues(MarvelObject=marvel.demo,
                             cell.group.g1=cell.group.g1,
                             cell.group.g2=cell.group.g2,
                             min.cells=5,
                             method="t.test",
                             method.adjust="fdr",
                             level="splicing",
                             event.type=c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
                             show.progress=FALSE
                             )

# Check output
head(marvel.demo$DE$PSI$Table[["ad"]])
</code></pre>

<hr>
<h2 id='CompareValues.Exp'>Differential gene expression analysis</h2><span id='topic+CompareValues.Exp'></span>

<h3>Description</h3>

<p>Performs differential gene expression analysis between 2 groups of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareValues.Exp(
  MarvelObject,
  cell.group.g1 = NULL,
  cell.group.g2 = NULL,
  downsample = FALSE,
  seed = 1,
  min.cells = 25,
  pct.cells = NULL,
  method,
  method.adjust,
  show.progress = TRUE,
  nboots = 1000,
  custom.gene_ids = NULL,
  mast.method = "bayesglm",
  mast.ebayes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareValues.Exp_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 1 (reference group).</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 2.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the number of cells in each cell group will be downsampled to the sample size of the smaller cell group so that both cell groups will have the sample size prior to differential expression analysis. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_seed">seed</code></td>
<td>
<p>Numeric value. The seed number for the random number generator to ensure reproducibility during during down-sampling of cells when <code>downsample</code> set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the gene for the gene to be included for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_pct.cells">pct.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the gene for the gene to be included for differential splicing analysis. If <code>pct.cells</code> is specified, then <code>pct.cells</code> will be used as threshold instead of <code>min.cells</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_method">method</code></td>
<td>
<p>Character string. Statistical test to compare the 2 groups of cells. <code>"ks"</code>, <code>"kuiper"</code>, <code>"ad"</code>, <code>"dts"</code>, <code>"wilcox"</code>, and <code>"t.test"</code> for Kolmogorov-Smirnov, Kuiper, Anderson-Darling, DTS, Wilcox, and t-test, respectively. Additional option is <code>"mast"</code>. If set to <code>"mast"</code> is specified, the log2fc and p-values will be corrected using the gene detection rate as per the <code>MAST</code> package tutorial.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_method.adjust">method.adjust</code></td>
<td>
<p>Character string. Adjust p-values for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_show.progress">show.progress</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, progress bar will be displayed so that users can estimate the time needed for differential analysis. Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_nboots">nboots</code></td>
<td>
<p>Numeric value. When <code>method</code> set to <code>"dts"</code>, the number of bootstrap iterations for computing the p-value.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_custom.gene_ids">custom.gene_ids</code></td>
<td>
<p>Character string. Instead of specified the genes to include for DE analysis with <code>min.cells</code>, users may input a custom vector of gene IDs to include for DE analysis.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_mast.method">mast.method</code></td>
<td>
<p>Character string. As per the <code>method</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>"bayesglm"</code>, other options are <code>"glm"</code> and <code>"glmer"</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp_+3A_mast.ebayes">mast.ebayes</code></td>
<td>
<p>Logical value. As per the <code>ebayes</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 new slot <code>MarvelObject$DE$Exp$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell groups for analysis
df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]

# DE
marvel.demo &lt;- CompareValues.Exp(MarvelObject=marvel.demo,
                                 cell.group.g1=cell.group.g1,
                                 cell.group.g2=cell.group.g2,
                                 min.cells=5,
                                 method="t.test",
                                 method.adjust="fdr",
                                 show.progress=FALSE
                                 )

# Check output
head(marvel.demo$DE$Exp$Table)
</code></pre>

<hr>
<h2 id='CompareValues.Exp.Spliced'>Differential gene expression analysis for differentially spliced genes</h2><span id='topic+CompareValues.Exp.Spliced'></span>

<h3>Description</h3>

<p>Performs differential gene expression analysis between 2 groups of cells only on differentially spliced genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareValues.Exp.Spliced(
  MarvelObject,
  cell.group.g1 = NULL,
  cell.group.g2 = NULL,
  psi.method,
  psi.pval,
  psi.delta,
  method.de.gene = "wilcox",
  method.adjust.de.gene = "fdr",
  downsample = FALSE,
  seed = 1,
  show.progress = TRUE,
  mast.method = "bayesglm",
  mast.ebayes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 1 (reference group).</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 2.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_psi.method">psi.method</code></td>
<td>
<p>Vector of character string(s). To include significant events from these method(s) for differential gene expression analysis.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Vector of numeric value(s). The adjusted p-value, below which, the splicing event is considered differentially spliced, and the corresponding genes will be included for differential gene expression analysis.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_psi.delta">psi.delta</code></td>
<td>
<p>Numeric value. The absolute difference in mean PSI values between <code>cell.group.g1</code> and <code>cell.group.g1</code>, above which, the splicing event is considered differentially spliced, and the corresponding genes will be included for differential gene expression analysis.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_method.de.gene">method.de.gene</code></td>
<td>
<p>Character string. Same as <code>method</code> in <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_method.adjust.de.gene">method.adjust.de.gene</code></td>
<td>
<p>Character string. Same as <code>method</code> in <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the number of cells in each cell group will be downsampled to the sample size of the smaller cell group so that both cell groups will have the sample size prior to differential expression analysis. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_seed">seed</code></td>
<td>
<p>Numeric value. The seed number for the random number generator to ensure reproducibility during during down-sampling of cells when <code>downsample</code> set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_show.progress">show.progress</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, progress bar will be displayed so that users can estimate the time needed for differential analysis. Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_mast.method">mast.method</code></td>
<td>
<p>Character string. As per the <code>method</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>"bayesglm"</code>, other options are <code>"glm"</code> and <code>"glmer"</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Exp.Spliced_+3A_mast.ebayes">mast.ebayes</code></td>
<td>
<p>Logical value. As per the <code>ebayes</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 new slot <code>MarvelObject$DE$Exp$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell groups for analysis
df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]

# DE
marvel.demo &lt;- CompareValues.Exp.Spliced(MarvelObject=marvel.demo,
                                         cell.group.g1=cell.group.g1,
                                         cell.group.g2=cell.group.g2,
                                         psi.method="ad",
                                         psi.pval=0.10,
                                         psi.delta=0,
                                         method.de.gene="t.test",
                                         method.adjust.de.gene="fdr",
                                         show.progress=FALSE
                                         )

# Check output
head(marvel.demo$DE$Exp.Spliced$Table)
</code></pre>

<hr>
<h2 id='CompareValues.Genes.10x'>Differential gene expression analysis</h2><span id='topic+CompareValues.Genes.10x'></span>

<h3>Description</h3>

<p>Performs differential gene expression analysis between two groups of cells. Only among cells and genes previously included for splice junction analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareValues.Genes.10x(
  MarvelObject,
  log2.transform = TRUE,
  show.progress = TRUE,
  method = "wilcox",
  mast.method = "bayesglm",
  mast.ebayes = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareValues.Genes.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.SJ.10x</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.Genes.10x_+3A_log2.transform">log2.transform</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), normalised gene expression values will be off-set by 1 and then log2-transformed prior to analysis. This option is automatically set to <code>TRUE</code> if <code>method</code> option is set to <code>"mast"</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Genes.10x_+3A_show.progress">show.progress</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), the progress bar will appear.</p>
</td></tr>
<tr><td><code id="CompareValues.Genes.10x_+3A_method">method</code></td>
<td>
<p>Character string. Statistical test to compare the 2 groups of cells. Default is <code>"wilcox"</code> as recommended by Seurat. Another option is <code>"mast"</code>. If <code>"mast"</code> is specified, the log2fc and p-values will be corrected using the gene detection rate as per the <code>MAST</code> package tutorial.</p>
</td></tr>
<tr><td><code id="CompareValues.Genes.10x_+3A_mast.method">mast.method</code></td>
<td>
<p>Character string. As per the <code>method</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>"bayesglm"</code>, other options are <code>"glm"</code> and <code>"glmer"</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.Genes.10x_+3A_mast.ebayes">mast.ebayes</code></td>
<td>
<p>Logical value. As per the <code>ebayes</code> option of the <code>zlm</code> function from the <code>MAST</code> package. Default is <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a updated slot <code>MarvelObject$DE$SJ$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- CompareValues.Genes.10x(
                        MarvelObject=marvel.demo.10x,
                        show.progress=FALSE
                        )

# Check output
head(marvel.demo.10x$DE$SJ$Table)
</code></pre>

<hr>
<h2 id='CompareValues.PSI'>Differential splicing analysis</h2><span id='topic+CompareValues.PSI'></span>

<h3>Description</h3>

<p>Performs differentially splicing analysis between 2 groups of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareValues.PSI(
  MarvelObject,
  cell.group.g1,
  cell.group.g2,
  downsample = FALSE,
  seed = 1,
  min.cells = 25,
  pct.cells = NULL,
  method,
  nboots = 1000,
  n.permutations = 1000,
  method.adjust = "fdr",
  event.type,
  show.progress = TRUE,
  annotate.outliers = TRUE,
  n.cells.outliers = 10,
  assign.modality = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareValues.PSI_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 1 (reference group).</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 2.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the number of cells in each cell group will be downsampled to the sample size of the smaller cell group so that both cell groups will have the sample size prior to differential expression analysis. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_seed">seed</code></td>
<td>
<p>Numeric value. The seed number for the random number generator to ensure reproducibility during during down-sampling of cells when <code>downsample</code> set to <code>TRUE</code>, during permutation testing when <code>method</code> set to <code>"permutation"</code>, and during modality assignment which will be performed automatically.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the splicing event for the event to be included for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_pct.cells">pct.cells</code></td>
<td>
<p>Numeric value. The minimum percentage of cells expressing the splicing event for the event to be included for differential splicing analysis. If <code>pct.cells</code> is specified, then <code>pct.cells</code> will be used as threshold instead of <code>min.cells</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_method">method</code></td>
<td>
<p>Character string. Statistical test to compare the 2 groups of cells. <code>"ks"</code>, <code>"kuiper"</code>, <code>"ad"</code>, <code>"dts"</code>, <code>"wilcox"</code>, <code>"t.test"</code>, and <code>"permutation"</code> for Kolmogorov-Smirnov, Kuiper, Anderson-Darling, DTS, Wilcox, t-test, and, permutation approach respectively.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_nboots">nboots</code></td>
<td>
<p>Numeric value. When <code>method</code> set to <code>"dts"</code>, the number of bootstrap iterations for computing the p-value.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_n.permutations">n.permutations</code></td>
<td>
<p>Numeric value. When <code>method</code> set to <code>"permutation"</code>, this argument indicates the number of permutations to perform for generating the null distribution for subsequent p-value inference. Default is <code>1000</code> times.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_method.adjust">method.adjust</code></td>
<td>
<p>Character string. Adjust p-values for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_event.type">event.type</code></td>
<td>
<p>Character string. Indicate which splicing event type to include for analysis. Can take value <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, or <code>"A3SS"</code> which represents skipped-exon (SE), mutually-exclusive exons (MXE), retained-intron (RI), alternative 5' splice site (A5SS), and alternative 3' splice site (A3SS), respectively.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_show.progress">show.progress</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, progress bar will be displayed so that users can estimate the time needed for differential analysis. Default value is <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_annotate.outliers">annotate.outliers</code></td>
<td>
<p>Numeric value. When set to <code>TRUE</code>, statistical difference in PSI values between the two cell groups that is driven by outlier cells will be annotated.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_n.cells.outliers">n.cells.outliers</code></td>
<td>
<p>Numeric value. When <code>annotate.outliers</code> set to <code>TRUE</code>, the minimum number of cells with non-1 or non-0 PSI values for included-to-included or excluded-to-excluded modality change, respectively. The p-values will be re-coded to 1 when both cell groups have less than this minimum number of cells. This is to avoid false positive results.</p>
</td></tr>
<tr><td><code id="CompareValues.PSI_+3A_assign.modality">assign.modality</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), modalities will be assigned to each cell group.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class data frame containing the output of the differential splicing analysis.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell groups for analysis
df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]

# DE
results &lt;- CompareValues.PSI(MarvelObject=marvel.demo,
                             cell.group.g1=cell.group.g1,
                             cell.group.g2=cell.group.g2,
                             min.cells=5,
                             method="t.test",
                             method.adjust="fdr",
                             event.type=c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
                             show.progress=FALSE
                             )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='CompareValues.SJ.10x'>Differential splice junction analysis</h2><span id='topic+CompareValues.SJ.10x'></span>

<h3>Description</h3>

<p>Performs differential splice junction analysis between two groups of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CompareValues.SJ.10x(
  MarvelObject,
  coord.introns = NULL,
  cell.group.g1,
  cell.group.g2,
  min.pct.cells.genes = 10,
  min.pct.cells.sj = 10,
  min.gene.norm = 1,
  seed = 1,
  n.iterations = 100,
  downsample = FALSE,
  show.progress = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CompareValues.SJ.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_coord.introns">coord.introns</code></td>
<td>
<p>Character strings. Specific splice junctions to be included for analysis. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of Character strings. Cell IDs corresponding to Group 1 (reference group).</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of Character strings. Cell IDs corresponding to Group 2.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_min.pct.cells.genes">min.pct.cells.genes</code></td>
<td>
<p>Numeric value. Minimum percentage of cells in which the gene is expressed for that gene to be included for splice junction expression distribution analysis. Expressed genes defined as genes with non-zero normalised UMI counts. This threshold may be determined from <code>PlotPctExprCells.SJ.10x</code> function. Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_min.pct.cells.sj">min.pct.cells.sj</code></td>
<td>
<p>Numeric value. Minimum percentage of cells in which the splice junction is expressed for that splice junction to be included for splice junction expression distribution analysis. Expressed splice junctions defined as splice junctions with raw UMI counts &gt;= 1. This threshold may be determined from <code>PlotPctExprCells.SJ.10x</code> function. Default is <code>10</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_min.gene.norm">min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction will be included for analysis. Default is <code>1.0</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Random number generator to be fixed for permutations test and down-sampling.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_n.iterations">n.iterations</code></td>
<td>
<p>Numeric value. Number of times to shuffle the cell group labels when building the null distribution. Default is <code>100</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, both cell groups will be down-sampled so that both cell groups will have the same number of cells. The number of cells to downsample will be based on the smallest cell group. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="CompareValues.SJ.10x_+3A_show.progress">show.progress</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), the progress bar will appear.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$DE$SJ$Table</code>, <code>MarvelObject$DE$SJ$cell.group.g1</code>, and <code>MarvelObject$DE$SJ$cell.group.g2</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # Group 1 (reference)
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Group 2
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

# DE
marvel.demo.10x &lt;- CompareValues.SJ.10x(
                        MarvelObject=marvel.demo.10x,
                        cell.group.g1=cell.ids.1,
                        cell.group.g2=cell.ids.2,
                        min.pct.cells.genes=10,
                        min.pct.cells.sj=10,
                        min.gene.norm=1.0,
                        seed=1,
                        n.iterations=100,
                        downsample=TRUE,
                        show.progress=FALSE
                        )

# Check output
head(marvel.demo.10x$DE$SJ$Table)
</code></pre>

<hr>
<h2 id='ComputePSI'>Compute percent spliced-in (PSI) values</h2><span id='topic+ComputePSI'></span>

<h3>Description</h3>

<p>Validate splicing events and subsequently computes percent spliced-in (PSI) values these high-quality splicing events. This is a wrapper function for <code>ComputePSI.SE</code>, <code>ComputePSI.MXE</code>, <code>ComputePSI.A5SS</code>, <code>ComputePSI.A3SS</code>, <code>ComputePSI.RI</code>, <code>ComputePSI.AFE</code>, and <code>ComputePSI.ALE</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI(
  MarvelObject,
  CoverageThreshold,
  EventType,
  thread = NULL,
  UnevenCoverageMultiplier = 10,
  read.length = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_eventtype">EventType</code></td>
<td>
<p>Character string. Indicate which splicing event type to calculate the PSI values for. Can take value <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, or <code>"A3SS"</code> which represents skipped-exon (SE), mutually-exclusive exons (MXE), retained-intron (RI), alternative 5' splice site (A5SS), and alternative 3' splice site (A3SS), respectively.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_thread">thread</code></td>
<td>
<p>Numeric value. Only applicable when <code>EventType</code> set to <code>"RI"</code> Set number of threads..</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_unevencoveragemultiplier">UnevenCoverageMultiplier</code></td>
<td>
<p>Numeric value. Maximum allowable fold difference between two included junction counts for SE or two included or two excluded junction counts for MXE. Only applicable when <code>EventType</code> set to <code>"SE"</code> or <code>"MXE"</code>, respectively.</p>
</td></tr>
<tr><td><code id="ComputePSI_+3A_read.length">read.length</code></td>
<td>
<p>Numeric value. The length of read.Only applicable when <code>EventType</code> set to <code>"RI"</code>. This number will be specific to the sequencing mode. E.g. read length should be set to <code>150</code> when samples were sequenced in 150bp paired-end or single-end. This option should only be specified when users used read-counting approach for computing intron counts. The option should be left with its default value <code>1</code> when users tabulated the per-base count and summed them up to arrive at the intron counts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>$SpliceFeatureValidated</code> and <code>$PSI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI(MarvelObject=marvel.demo,
                          CoverageThreshold=10,
                          EventType="SE",
                          UnevenCoverageMultiplier=10
                          )
</code></pre>

<hr>
<h2 id='ComputePSI.A3SS'>Compute Alternative 3' Splice Site (A3SS) Percent Spliced-in (PSI) Values</h2><span id='topic+ComputePSI.A3SS'></span>

<h3>Description</h3>

<p>Validate A3SS splicing events and subsequently computes percent spliced-in (PSI) values these high-quality splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.A3SS(MarvelObject, CoverageThreshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.A3SS_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.A3SS_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>$SpliceFeatureValidated$A3SS</code> and <code>$PSI$A3SS</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.A3SS(MarvelObject=marvel.demo,
                               CoverageThreshold=10
                               )
</code></pre>

<hr>
<h2 id='ComputePSI.A5SS'>Compute alternative 5' splice site (A5SS) percent spliced-in (PSI) values</h2><span id='topic+ComputePSI.A5SS'></span>

<h3>Description</h3>

<p>Validate A5SS splicing events and subsequently computes percent spliced-in (PSI) values these high-quality splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.A5SS(MarvelObject, CoverageThreshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.A5SS_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.A5SS_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>$SpliceFeatureValidated$A5SS</code> and <code>$PSI$A5SS</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.A5SS(MarvelObject=marvel.demo,
                               CoverageThreshold=10
                               )
</code></pre>

<hr>
<h2 id='ComputePSI.AFE'>Compute alternative first exon (AFE) percent spliced-in (PSI) values</h2><span id='topic+ComputePSI.AFE'></span>

<h3>Description</h3>

<p>Computes percent spliced-in (PSI) for alternative first exon (ALE) splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.AFE(MarvelObject, CoverageThreshold = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.AFE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>DetectEvents</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.AFE_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>$SpliceFeatureValidated$AFE</code> and <code>$PSI$AFE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.AFE(MarvelObject=marvel.demo,
                              CoverageThreshold=10
                              )
</code></pre>

<hr>
<h2 id='ComputePSI.ALE'>Compute alternative last exon (ALE) percent spliced-in (PSI) values</h2><span id='topic+ComputePSI.ALE'></span>

<h3>Description</h3>

<p>Computes percent spliced-in (PSI) for alternative last exon (ALE) splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.ALE(MarvelObject, CoverageThreshold = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.ALE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>DetectEvents</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.ALE_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>$SpliceFeatureValidated$ALE</code> and <code>$PSI$ALE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.ALE(MarvelObject=marvel.demo,
                              CoverageThreshold=10
                              )
</code></pre>

<hr>
<h2 id='ComputePSI.MXE'>Compute mutually exclusive exons (MXE) percent spliced-in (PSI) values</h2><span id='topic+ComputePSI.MXE'></span>

<h3>Description</h3>

<p>Validate MXE splicing events and subsequently computes percent spliced-in (PSI) values these high-quality splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.MXE(MarvelObject, CoverageThreshold, UnevenCoverageMultiplier = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.MXE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.MXE_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
<tr><td><code id="ComputePSI.MXE_+3A_unevencoveragemultiplier">UnevenCoverageMultiplier</code></td>
<td>
<p>Numeric value. Maximum allowable fold difference between two included or two excluded junction counts for MXE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the PSI for each MXE splicing event. Splicing events provided in <code>SpliceFeature</code> data frame will first be cross-checked against the splice junctions provided in <code>SpliceJunction</code> data frame. Only events whose junctions are found in <code>SpliceJunction</code> are retained. The formula for computing PSI is the number of junction reads supporting the included isoform divided by the total number of reads supporting both included and excluded isoforms.
</p>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>$SpliceFeatureValidated$MXE</code> and <code>$PSI$MXE</code>.
</p>


<h3>Author(s)</h3>

<p>Sean Wen &lt;sean.wenwx@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.MXE(MarvelObject=marvel.demo,
                              CoverageThreshold=10,
                              UnevenCoverageMultiplier=10
                              )
</code></pre>

<hr>
<h2 id='ComputePSI.RI'>Compute retained-intron (RI) percent spliced-in (PSI) values</h2><span id='topic+ComputePSI.RI'></span>

<h3>Description</h3>

<p>Validate RI splicing events and subsequently computes percent spliced-in (PSI) values these high-quality splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.RI(
  MarvelObject,
  CoverageThreshold,
  IntronCounts,
  thread,
  read.length = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.RI_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.RI_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
<tr><td><code id="ComputePSI.RI_+3A_introncounts">IntronCounts</code></td>
<td>
<p>Data frame. Columns indicate sample IDs, rows indicate intron coordinates, and values indicate total intron coverage. The first column needs to be named <code>coord.intron</code>. These values will be combined with splice junction counts in the MARVEL object to compute PSI values.</p>
</td></tr>
<tr><td><code id="ComputePSI.RI_+3A_thread">thread</code></td>
<td>
<p>Numeric value. Set number of threads.</p>
</td></tr>
<tr><td><code id="ComputePSI.RI_+3A_read.length">read.length</code></td>
<td>
<p>Numeric value. The length of read. This number will be specific to the sequencing mode. E.g. read length should be set to <code>150</code> when samples were sequenced in 150bp paired-end or single-end. This option should only be specified when users used read-counting approach for computing intron counts. The option should be left with its default value <code>1</code> when users tabulated the per-base count and summed them up to arrive at the intron counts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>$SpliceFeatureValidated$RI</code> and <code>$PSI$RI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.RI(MarvelObject=marvel.demo,
                             CoverageThreshold=10,
                             IntronCounts=marvel.demo$IntronCounts,
                             thread=1
                             )
</code></pre>

<hr>
<h2 id='ComputePSI.SE'>Compute skipped-exon (SE) percent spliced-in (PSI) values</h2><span id='topic+ComputePSI.SE'></span>

<h3>Description</h3>

<p>Validate SE splicing events and subsequently computes percent spliced-in (PSI) values these high-quality splicing events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComputePSI.SE(MarvelObject, CoverageThreshold, UnevenCoverageMultiplier = 10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ComputePSI.SE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="ComputePSI.SE_+3A_coveragethreshold">CoverageThreshold</code></td>
<td>
<p>Numeric value. Coverage threshold below which the PSI of the splicing event will be censored, i.e. annotated as missing (NA). Coverage defined as the total number of reads supporting both included and excluded isoforms.</p>
</td></tr>
<tr><td><code id="ComputePSI.SE_+3A_unevencoveragemultiplier">UnevenCoverageMultiplier</code></td>
<td>
<p>Numeric value. Maximum allowable fold difference between two included junction counts.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function computes the PSI for each SE splicing event. Splicing events provided in <code>SpliceFeature</code> data frame will first be cross-checked against the splice junctions provided in <code>SpliceJunction</code> data frame. Only events whose junctions are found in <code>SpliceJunction</code> are retained. The formula for computing PSI is the number of junction reads supporting the included isoform divided by the total number of reads supporting both included and excluded isoforms.
</p>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>$SpliceFeatureValidated$SE</code>  <code>$PSI$SE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ComputePSI.SE(MarvelObject=marvel.demo,
                             CoverageThreshold=10,
                             UnevenCoverageMultiplier=10
                             )
</code></pre>

<hr>
<h2 id='CountEvents'>Tabulate the number of expressed splicing events</h2><span id='topic+CountEvents'></span>

<h3>Description</h3>

<p>Tabulates and plots the number of expressed splicing events for each splicing event category for a specified cell group.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CountEvents(MarvelObject, sample.ids, min.cells, event.group.colors = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CountEvents_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="CountEvents_+3A_sample.ids">sample.ids</code></td>
<td>
<p>Vector of character strings. Sample IDs that constitute the cell group.</p>
</td></tr>
<tr><td><code id="CountEvents_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. Minimum number of cells expressing the splicing event for the event to be included for tabulation. A splicing event is defined as expressed when it has a non-missing PSI value.</p>
</td></tr>
<tr><td><code id="CountEvents_+3A_event.group.colors">event.group.colors</code></td>
<td>
<p>Vector of character strings. Colors for the event groups. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$N.Events$Table</code> and <code>MarvelObject$N.Events$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell group for analysis
df.pheno &lt;- marvel.demo$SplicePheno
sample.ids &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]

# Tabulate expressed events
marvel.demo &lt;- CountEvents(MarvelObject=marvel.demo,
                           sample.ids=sample.ids,
                           min.cells=5,
                           event.group.colors=NULL
                           )

# Check outputs
marvel.demo$N.Events$Table
marvel.demo$N.Events$Plot
</code></pre>

<hr>
<h2 id='CreateMarvelObject'>Create Marvel object for plate-based RNA-sequencing data</h2><span id='topic+CreateMarvelObject'></span>

<h3>Description</h3>

<p>Creates an S3 object named <code>Marvel</code> for downstream analysis, specifically for plate-based RNA-sequencing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateMarvelObject(
  SplicePheno = NULL,
  SpliceJunction = NULL,
  IntronCounts = NULL,
  SpliceFeature = NULL,
  SpliceFeatureValidated = NULL,
  PSI = NULL,
  GeneFeature = NULL,
  Exp = NULL,
  GTF = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateMarvelObject_+3A_splicepheno">SplicePheno</code></td>
<td>
<p>Data frame. Sample metadata.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_splicejunction">SpliceJunction</code></td>
<td>
<p>Data frame. Splice junction counts matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_introncounts">IntronCounts</code></td>
<td>
<p>Data frame. Intron coverage matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_splicefeature">SpliceFeature</code></td>
<td>
<p>List of data frames. Each data frame is the exon-level alternative splicing event metadata.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_splicefeaturevalidated">SpliceFeatureValidated</code></td>
<td>
<p>List of data frames. Each data frame is the validated (high-quality) exon-level alternative splicing event metadata.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_psi">PSI</code></td>
<td>
<p>Data frame. PSI matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_genefeature">GeneFeature</code></td>
<td>
<p>Data frame. Gene metadata.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_exp">Exp</code></td>
<td>
<p>Data frame. Normalised, non-log2-transformed gene expression matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject_+3A_gtf">GTF</code></td>
<td>
<p>Data frame. GTF used for generating the exon-level alternative splicing event metadata.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

SpliceJunction &lt;- marvel.demo$SpliceJunction
SpliceJunction[1:5,1:5]

SplicePheno &lt;- marvel.demo$SplicePheno
SplicePheno[1:5,]

SpliceFeature &lt;- marvel.demo$SpliceFeature
SpliceFeature[["SE"]][1:5, ]

IntronCounts &lt;- marvel.demo$IntronCounts
IntronCounts[1:5,1:5]

GeneFeature &lt;- marvel.demo$GeneFeature
GeneFeature[1:5, ]

Exp &lt;- marvel.demo$Exp
Exp[1:5,1:5]

marvel &lt;- CreateMarvelObject(SpliceJunction=SpliceJunction,
                             SplicePheno=SplicePheno,
                             SpliceFeature=SpliceFeature,
                             IntronCounts=IntronCounts,
                             GeneFeature=GeneFeature,
                             Exp=Exp
                             )
class(marvel)
</code></pre>

<hr>
<h2 id='CreateMarvelObject.10x'>Create Marvel object for droplet-based RNA-sequencing data</h2><span id='topic+CreateMarvelObject.10x'></span>

<h3>Description</h3>

<p>Creates an S3 object named <code>Marvel</code> for downstream analysis, specifically for droplet-based RNA-sequencing data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateMarvelObject.10x(
  gene.norm.matrix = NULL,
  gene.norm.pheno = NULL,
  gene.norm.feature = NULL,
  gene.count.matrix = NULL,
  gene.count.pheno = NULL,
  gene.count.feature = NULL,
  sj.count.matrix = NULL,
  sj.count.pheno = NULL,
  sj.count.feature = NULL,
  pca = NULL,
  gtf = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CreateMarvelObject.10x_+3A_gene.norm.matrix">gene.norm.matrix</code></td>
<td>
<p>Sparse matrix. UMI-collapsed, normalised, non-log2-transformed gene expression matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_gene.norm.pheno">gene.norm.pheno</code></td>
<td>
<p>Data frame. Sample metadata for annotating <code>gene.norm.matrix</code> columns with cell IDs.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_gene.norm.feature">gene.norm.feature</code></td>
<td>
<p>Data frame. Gene metadata for annotating <code>gene.norm.matrix</code> rows with gene names.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_gene.count.matrix">gene.count.matrix</code></td>
<td>
<p>Sparse matrix. UMI-collapsed, non-normalised (raw counts), non-log2-transformed gene expression matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_gene.count.pheno">gene.count.pheno</code></td>
<td>
<p>Data frame. Sample metadata for annotating <code>gene.count.matrix</code> columsn with cell IDs.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_gene.count.feature">gene.count.feature</code></td>
<td>
<p>Data frame. Gene metadata for annotating <code>gene.count.matrix</code> rows with gene names.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_sj.count.matrix">sj.count.matrix</code></td>
<td>
<p>Sparse matrix. UMI-collapsed, non-normalised (raw counts), non-log2-transformed splice junction expression matrix.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_sj.count.pheno">sj.count.pheno</code></td>
<td>
<p>Data frame. Sample metadata for annotating <code>sj.count.matrix</code> columsn with cell IDs.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_sj.count.feature">sj.count.feature</code></td>
<td>
<p>Data frame. Splice junction metadata for annotating <code>sj.count.matrix</code> rows with splice junction coordinates.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_pca">pca</code></td>
<td>
<p>Data frame. Coordinates of PCA/tSNE/UMAP.</p>
</td></tr>
<tr><td><code id="CreateMarvelObject.10x_+3A_gtf">gtf</code></td>
<td>
<p>Data frame. GTF used in cellranger. Will be used for annotating splice junctions downstream.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Retrieve, observe format of pre-saved input files
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )
# Gene expression (Normalised)
    # Matrix
    df.gene.norm &lt;- marvel.demo.10x.raw$gene.norm.matrix
    df.gene.norm[1:5, 1:5]

    # phenoData
    df.gene.norm.pheno &lt;- marvel.demo.10x.raw$sample.metadata
    head(df.gene.norm.pheno)

    # featureData
    df.gene.norm.feature &lt;- data.frame("gene_short_name"=rownames(df.gene.norm),
                                       stringsAsFactors=FALSE
                                       )
    head(df.gene.norm.feature)

# Gene expression (Counts)
    # Matrix
    df.gene.count &lt;- marvel.demo.10x.raw$gene.count.matrix
    df.gene.count[1:5, 1:5]

    # phenoData
    df.gene.count.pheno &lt;- data.frame("cell.id"=colnames(df.gene.count),
                                       stringsAsFactors=FALSE
                                       )
    head(df.gene.count.pheno)

    # featureData
    df.gene.count.feature &lt;- data.frame("gene_short_name"=rownames(df.gene.count),
                                       stringsAsFactors=FALSE
                                       )
    head(df.gene.count.feature)

# SJ (Counts)
    # Matrix
    df.sj.count &lt;- marvel.demo.10x.raw$sj.count.matrix
    df.sj.count[1:5, 1:5]

    # phenoData
    df.sj.count.pheno &lt;- data.frame("cell.id"=colnames(df.sj.count),
                                     stringsAsFactors=FALSE
                                     )
    head(df.sj.count.pheno)

    # featureData
    df.sj.count.feature &lt;- data.frame("coord.intron"=rownames(df.sj.count),
                                       stringsAsFactors=FALSE
                                       )
    head(df.sj.count.feature)

# tSNE coordinates
df.coord &lt;- marvel.demo.10x.raw$pca
head(df.coord)

# GTF
gtf &lt;- marvel.demo.10x.raw$gtf
head(gtf)

# Create MARVEL object
marvel.demo.10x &lt;- CreateMarvelObject.10x(gene.norm.matrix=df.gene.norm,
                     gene.norm.pheno=df.gene.norm.pheno,
                     gene.norm.feature=df.gene.norm.feature,
                     gene.count.matrix=df.gene.count,
                     gene.count.pheno=df.gene.count.pheno,
                     gene.count.feature=df.gene.count.feature,
                     sj.count.matrix=df.sj.count,
                     sj.count.pheno=df.sj.count.pheno,
                     sj.count.feature=df.sj.count.feature,
                     pca=df.coord,
                     gtf=gtf
                     )
</code></pre>

<hr>
<h2 id='DetectEvents'>Detect Splicing Events</h2><span id='topic+DetectEvents'></span>

<h3>Description</h3>

<p>Detects splicing events, specifically alternative first and last exons (AFE, ALE) from GTF. This is a wrapper function for <code>DetectEvents.ALE</code> and <code>DetectEvents.AFE</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents(
  MarvelObject,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE,
  EventType
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>. Only applicable when <code>EventType</code> set to <code>"ALE"</code> or <code>"AFE"</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents_+3A_eventtype">EventType</code></td>
<td>
<p>Character string. Indicate which splicing event type to calculate the PSI values for. Can take value <code>"ALE"</code> or <code>"AFE"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$ALE</code> or <code>MarvelObject$SpliceFeature$AFE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents(MarvelObject=marvel.demo,
                            min.cells=5,
                            min.expr=1,
                            track.progress=FALSE,
                            EventType="AFE"
                            )
</code></pre>

<hr>
<h2 id='DetectEvents.AFE'>Detect alternative first exons</h2><span id='topic+DetectEvents.AFE'></span>

<h3>Description</h3>

<p>Detects alternative first exons from GTF. This is a wrapper function for <code>DetectEvents.AFE.PosStrand</code> and <code>DetectEvents.AFE.NegStrand</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents.AFE(
  MarvelObject,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents.AFE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$AFE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents.AFE(MarvelObject=marvel.demo,
                                min.cells=5,
                                min.expr=1,
                                track.progress=FALSE
                                )
</code></pre>

<hr>
<h2 id='DetectEvents.AFE.NegStrand'>Detect alternative first exons on negative strand</h2><span id='topic+DetectEvents.AFE.NegStrand'></span>

<h3>Description</h3>

<p>Detects alternative first exons, specifically for genes transcribed on the negative strand of the DNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents.AFE.NegStrand(
  MarvelObject,
  parsed.gtf = NULL,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents.AFE.NegStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.NegStrand_+3A_parsed.gtf">parsed.gtf</code></td>
<td>
<p>Data frame. GTF file with the gene_id parsed. Generated from the <code>DetectEvents.AFE</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.NegStrand_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.NegStrand_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.NegStrand_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$AFE.NegStrand</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents.AFE.NegStrand(MarvelObject=marvel.demo,
                                          parsed.gtf=NULL,
                                          min.cells=5,
                                          min.expr=1,
                                          track.progress=FALSE
                                          )
</code></pre>

<hr>
<h2 id='DetectEvents.AFE.PosStrand'>Detect alternative first exons on positive strand</h2><span id='topic+DetectEvents.AFE.PosStrand'></span>

<h3>Description</h3>

<p>Detects alternative first exons, specifically for genes transcribed on the positive strand of the DNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents.AFE.PosStrand(
  MarvelObject,
  parsed.gtf = NULL,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents.AFE.PosStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.PosStrand_+3A_parsed.gtf">parsed.gtf</code></td>
<td>
<p>Data frame. GTF file with the gene_id parsed. Generated from the <code>DetectEvents.AFE</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.PosStrand_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.PosStrand_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.AFE.PosStrand_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$AFE.PosStrand</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents.AFE.PosStrand(MarvelObject=marvel.demo,
                                          parsed.gtf=NULL,
                                          min.cells=5,
                                          min.expr=1,
                                          track.progress=FALSE
                                          )
</code></pre>

<hr>
<h2 id='DetectEvents.ALE'>Detect alternative last exons</h2><span id='topic+DetectEvents.ALE'></span>

<h3>Description</h3>

<p>Detects alternative last exons from GTF. This is a wrapper function for <code>DetectEvents.ALE.PosStrand</code> and <code>DetectEvents.ALE.NegStrand</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents.ALE(
  MarvelObject,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents.ALE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$ALE</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents.ALE(MarvelObject=marvel.demo,
                                min.cells=5,
                                min.expr=1,
                                track.progress=FALSE
                                )
</code></pre>

<hr>
<h2 id='DetectEvents.ALE.NegStrand'>Detect alternative last exons on negative strand</h2><span id='topic+DetectEvents.ALE.NegStrand'></span>

<h3>Description</h3>

<p>Detects alternative last exons, specifically for genes transcribed on the negative strand of the DNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents.ALE.NegStrand(
  MarvelObject,
  parsed.gtf = NULL,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents.ALE.NegStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.NegStrand_+3A_parsed.gtf">parsed.gtf</code></td>
<td>
<p>Data frame. GTF file with the gene_id parsed. Generated from the <code>DetectEvents.ALE</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.NegStrand_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.NegStrand_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.NegStrand_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$ALE.NegStrand</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents.ALE.NegStrand(MarvelObject=marvel.demo,
                                          parsed.gtf=NULL,
                                          min.cells=5,
                                          min.expr=1,
                                          track.progress=FALSE
                                          )
</code></pre>

<hr>
<h2 id='DetectEvents.ALE.PosStrand'>Detect alternative last exons on positive strand</h2><span id='topic+DetectEvents.ALE.PosStrand'></span>

<h3>Description</h3>

<p>Detects alternative last exons, specifically for genes transcribed on the positive strand of the DNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DetectEvents.ALE.PosStrand(
  MarvelObject,
  parsed.gtf = NULL,
  min.cells = 50,
  min.expr = 1,
  track.progress = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="DetectEvents.ALE.PosStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.PosStrand_+3A_parsed.gtf">parsed.gtf</code></td>
<td>
<p>Data frame. GTF file with the gene_id parsed. Generated from the <code>DetectEvents.ALE</code> function.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.PosStrand_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum number of cells in which the gene is expressed for the gene to included for splicing event detected and quantification. To be used in conjunction with <code>min.expr</code> argument. Default value is <code>50</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.PosStrand_+3A_min.expr">min.expr</code></td>
<td>
<p>Numeric value. The minimum expression value for the gene to be considered to be expressed in a cell. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="DetectEvents.ALE.PosStrand_+3A_track.progress">track.progress</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, progress bar will appear to track the progress of the rate-limiting step of this function, which is the extraction of the final exon-exon junctions. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$SpliceFeature$ALE.PosStrand</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- DetectEvents.ALE.PosStrand(MarvelObject=marvel.demo,
                                          parsed.gtf=NULL,
                                          min.cells=5,
                                          min.expr=1,
                                          track.progress=FALSE
                                          )
</code></pre>

<hr>
<h2 id='FilterGenes.10x'>Filter specific gene types</h2><span id='topic+FilterGenes.10x'></span>

<h3>Description</h3>

<p>Retain genes of specific type, e.g., protein-coding genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FilterGenes.10x(MarvelObject, gene.type = "protein_coding")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FilterGenes.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>AnnotateGenes.10x</code> function.</p>
</td></tr>
<tr><td><code id="FilterGenes.10x_+3A_gene.type">gene.type</code></td>
<td>
<p>Character string. Gene type to keep. Specification should match that of GTF.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing the updated slots <code>MarvelObject$gene.metadata</code>, <code>MarvelObject$gene.norm.matrix</code>, <code>MarvelObject$sj.metadata</code>, and <code>MarvelObject$sj.count.matrix</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load un-processed MARVEL object
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )

# Annotate gene metadata
marvel.demo.10x &lt;- AnnotateGenes.10x(MarvelObject=marvel.demo.10x.raw)

# Annotate junction metadata
marvel.demo.10x &lt;- AnnotateSJ.10x(MarvelObject=marvel.demo.10x)

# Validate junctions
marvel.demo.10x &lt;- ValidateSJ.10x(MarvelObject=marvel.demo.10x)

# Subset CDS genes
marvel.demo.10x &lt;- FilterGenes.10x(MarvelObject=marvel.demo.10x,
                          gene.type="protein_coding"
                          )
</code></pre>

<hr>
<h2 id='FindPTC'>Find premature terminal codons (PTCs)</h2><span id='topic+FindPTC'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC(MarvelObject, method, pval, delta, custom.tran_ids = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> functions.</p>
</td></tr>
<tr><td><code id="FindPTC_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="FindPTC_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splicing event will be analysed for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC_+3A_delta">delta</code></td>
<td>
<p>Numeric value. Positive delta percent spliced-in (PSI) value above which the splicing event will be analysed for PTCs. &quot;Positive&quot; because only an increased in PSI value leads to increased alternative exon inclusion in the transcript.</p>
</td></tr>
<tr><td><code id="FindPTC_+3A_custom.tran_ids">custom.tran_ids</code></td>
<td>
<p>Vector of character strings. Subset of tran_ids to be brought forward for analysis after filtering based on <code>pval</code> and <code>delta</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$NMD$Prediction</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- FindPTC(MarvelObject=marvel.demo,
                       method="ad",
                       pval=0.1,
                       delta=90
                       )
</code></pre>

<hr>
<h2 id='FindPTC.A3SS.NegStrand'>Find premature terminal codon (PTC) for alternative 3' splice site (A3SS) located on the negative strand of the transcript</h2><span id='topic+FindPTC.A3SS.NegStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.A3SS.NegStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.A3SS.NegStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.A3SS.NegStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.A3SS.NegStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="A3SS")
index.2 &lt;- grep(":-@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[3]
gene_id &lt;- results$gene_id[3]

# Find PTC
results &lt;- FindPTC.A3SS.NegStrand(MarvelObject=marvel.demo,
                                  tran_id=NULL,
                                  gene_id=gene_id
                                  )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.A3SS.PosStrand'>Find premature terminal codon (PTC) for alternative 3' splice site (A3SS) located on the positive strand of the transcript</h2><span id='topic+FindPTC.A3SS.PosStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.A3SS.PosStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.A3SS.PosStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.A3SS.PosStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.A3SS.PosStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="A3SS")
index.2 &lt;- grep(":+@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.A3SS.PosStrand(MarvelObject=marvel.demo,
                                tran_id=tran_id,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.A5SS.NegStrand'>Find premature terminal codon (PTC) for alternative 5' splice site (A5SS) located on the negative strand of the transcript</h2><span id='topic+FindPTC.A5SS.NegStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.A5SS.NegStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.A5SS.NegStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.A5SS.NegStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.A5SS.NegStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="A5SS")
index.2 &lt;- grep(":-@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.A5SS.NegStrand(MarvelObject=marvel.demo,
                                tran_id=tran_id,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.A5SS.PosStrand'>Find premature terminal codon (PTC) for alternative 5' splice site (A5SS) located on the positive strand of the transcript</h2><span id='topic+FindPTC.A5SS.PosStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.A5SS.PosStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.A5SS.PosStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.A5SS.PosStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.A5SS.PosStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="A5SS")
index.2 &lt;- grep(":+@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.A5SS.PosStrand(MarvelObject=marvel.demo,
                                tran_id=tran_id,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.RI.NegStrand'>Find premature terminal codon (PTC) for retained-intron (RI) located on the negative strand of the transcript</h2><span id='topic+FindPTC.RI.NegStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.RI.NegStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.RI.NegStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.RI.NegStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.RI.NegStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="RI")
index.2 &lt;- grep(":-@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.RI.NegStrand(MarvelObject=marvel.demo,
                                tran_id=tran_id,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.RI.PosStrand'>Find premature terminal codon (PTC) for retained-Intron (RI) located on the positive strand of the transcript</h2><span id='topic+FindPTC.RI.PosStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.RI.PosStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.RI.PosStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.RI.PosStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.RI.PosStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="RI")
index.2 &lt;- grep(":+@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.RI.PosStrand(MarvelObject=marvel.demo,
                                tran_id=tran_id,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.SE.NegStrand'>Find premature terminal codon (PTC) for skipped-exon (SE) located on the negative Strand of the yranscript</h2><span id='topic+FindPTC.SE.NegStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.SE.NegStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.SE.NegStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.SE.NegStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.SE.NegStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.#'</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="SE")
index.2 &lt;- grep(":-@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.SE.NegStrand(MarvelObject=marvel.demo,
                                tran_id=NULL,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='FindPTC.SE.PosStrand'>Find premature terminal codon (PTC) for skipped-exon (SE) located on the positive strand of the transcript</h2><span id='topic+FindPTC.SE.PosStrand'></span>

<h3>Description</h3>

<p>Finds PTC(s) introduced by alternative exons into protein-coding transcripts.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FindPTC.SE.PosStrand(MarvelObject, tran_id, gene_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FindPTC.SE.PosStrand_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CompareValues.PSI</code> and <code>ParseGTF</code> function.</p>
</td></tr>
<tr><td><code id="FindPTC.SE.PosStrand_+3A_tran_id">tran_id</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> to look for PTCs.</p>
</td></tr>
<tr><td><code id="FindPTC.SE.PosStrand_+3A_gene_id">gene_id</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> corresponding to the <code>tran_id</code> argument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function finds PTC(s) introduced by alternative exons into protein-coding transcripts. It also records the distance between a PTCs and the final splice junction for a given protein-coding transcript. Non-protein-coding transcripts or transcripts in which splicing events are located outside of the transcripts' open-reading frame (ORF) are not analysed for PTCs but are noted.
</p>


<h3>Value</h3>

<p>A data frame of transcripts containing splicing events meeting the <code>psi.de.sig</code> and <code>psi.de.diff</code> criteria are categorised based on the presence or absence of PTCs.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define relevant event type
results &lt;- marvel.demo$DE$PSI$Table[["ad"]]
index.1 &lt;- which(results$event_type=="SE")
index.2 &lt;- grep(":+@", results$tran_id, fixed=TRUE)
index &lt;- intersect(index.1, index.2)
results &lt;- results[index, ]
tran_id &lt;- results$tran_id[1]
gene_id &lt;- results$gene_id[1]

# Find PTC
results &lt;- FindPTC.SE.PosStrand(MarvelObject=marvel.demo,
                                tran_id=NULL,
                                gene_id=gene_id
                                )

# Check output
head(results)
</code></pre>

<hr>
<h2 id='IsoSwitch'>Classify gene-splicing relationship</h2><span id='topic+IsoSwitch'></span>

<h3>Description</h3>

<p>Classify gene-splicing relative changes to each other from cell group 1 to group 2. Classifications are coordinated, opposing, isoform-switching, and complex. In coordinated relationship, both gene and splicing changes in the same direction from cell group 1 to group 2. In opposing relationship, gene changes in the opposite direction relative to splicing from cell group 1 to group 2. In isoform-switching, there is differential splice junction usage without differential expression of the corresponding gene between cell group 1 and group 2. Complex relationship involves genes with both coordinated and opposing relationships with splicing. Only differentially spliced junctions are included for analysis here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsoSwitch(
  MarvelObject,
  method,
  psi.pval = 0.1,
  psi.delta = 0,
  gene.pval = 0.1,
  gene.log2fc = 0.5,
  event.type = NULL,
  custom.tran_ids = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsoSwitch_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splicing event is considered differentially spliced and included for isoform switching analysis. To be used in conjunction with <code>psi.delta</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_psi.delta">psi.delta</code></td>
<td>
<p>Numeric value. The absolute mininum difference in PSI values between the two cell groups above which the splicing event is considered differentially spliced nd included for isoform switching analysis. To be used in conjunction with <code>psi.pval</code>. Specify <code>0</code> (default) to switch this threshold off.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_gene.pval">gene.pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the gene is considered differentially expressed. Default value is <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_gene.log2fc">gene.log2fc</code></td>
<td>
<p>Numeric value. The absolute log2 fold change in mean gene expression values between the two cell groups above which the gene is considered differentially expressed. To be used in conjunction with <code>gene.pval</code>. Specify <code>0</code> to switch this threshold off. Default value is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_event.type">event.type</code></td>
<td>
<p>Character string. Indicate which splicing event type to include for analysis. Can take any combination of values: <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, <code>"A3SS"</code>, <code>"AFE}, or \code{"ALE</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch_+3A_custom.tran_ids">custom.tran_ids</code></td>
<td>
<p>Vector of character strings. Subset of tran_ids to be brought forward for analysis after filtering based on <code>psi.pval</code> and <code>psi.delta</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>MarvelObject$DE$Cor$Table</code>, <code>MarvelObject$DE$Cor$Plot</code>, and <code>MarvelObject$DE$Cor$Plot.Stats</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- IsoSwitch(MarvelObject=marvel.demo,
                         method="ad",
                         psi.pval=0.1,
                         psi.delta=0,
                         gene.pval=0.1,
                         gene.log2fc=0.5
                         )

# Check outputs
head(marvel.demo$DE$Cor$Table_Raw)
head(marvel.demo$DE$Cor$Table)
marvel.demo$DE$Cor$Plot
marvel.demo$DE$Cor$Plot.Stats
</code></pre>

<hr>
<h2 id='IsoSwitch.10x'>Classify gene-splicing relationship</h2><span id='topic+IsoSwitch.10x'></span>

<h3>Description</h3>

<p>Classify gene-splicing relative changes to each other from cell group 1 to group 2. Classifications are coordinated, opposing, isoform-switching, and complex. In coordinated relationship, both gene and splicing changes in the same direction from cell group 1 to group 2. In opposing relationship, gene changes in the opposite direction relative to splicing from cell group 1 to group 2. In isoform-switching, there is differential splice junction usage without differential expression of the corresponding gene between cell group 1 and group 2. Complex relationship involves genes with both coordinated and opposing relationships with splicing. Only differentially spliced junctions are included for analysis here.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsoSwitch.10x(
  MarvelObject,
  pval.sj = 0.05,
  log2fc.sj = NULL,
  delta.sj = 5,
  min.gene.norm = 0,
  pval.adj.gene = 0.05,
  log2fc.gene = 0.5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsoSwitch.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td></tr>
<tr><td><code id="IsoSwitch.10x_+3A_pval.sj">pval.sj</code></td>
<td>
<p>Numeric value. p-value from differential splicing analysis, below which, the splice junction is considered differentially spliced. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch.10x_+3A_log2fc.sj">log2fc.sj</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential splicing analysis, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>delta.sj</code> has been specified.</p>
</td></tr>
<tr><td><code id="IsoSwitch.10x_+3A_delta.sj">delta.sj</code></td>
<td>
<p>Numeric value. Absolute difference in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>log2fc.sj</code> has been specified.</p>
</td></tr>
<tr><td><code id="IsoSwitch.10x_+3A_min.gene.norm">min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction is considered differentially spliced. Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch.10x_+3A_pval.adj.gene">pval.adj.gene</code></td>
<td>
<p>Numeric value. Adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="IsoSwitch.10x_+3A_log2fc.gene">log2fc.gene</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential gene expression analysis, above which, the gene is considered differentially expressed. This option should be <code>NULL</code> if <code>delta.sj</code> has been specified.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing new slots <code>MarvelObject$SJ.Gene.Cor$Data</code>, <code>MarvelObject$SJ.Gene.Cor$Proportion$Plot</code>, and <code>MarvelObject$SJ.Gene.Cor$Proportion$Table</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- IsoSwitch.10x(
                        MarvelObject=marvel.demo.10x,
                        pval.sj=0.05,
                        delta.sj=5,
                        min.gene.norm=1.0,
                        pval.adj.gene=0.05,
                        log2fc.gene=0.5
                        )

# Check outputs
marvel.demo.10x$SJ.Gene.Cor$Proportion$Plot
marvel.demo.10x$SJ.Gene.Cor$Proportion$Table
cols &lt;- c("coord.intron", "gene_short_name", "cor.complete")
head(marvel.demo.10x$SJ.Gene.Cor$Data[,cols])
</code></pre>

<hr>
<h2 id='IsoSwitch.PlotExpr'>Plot gene-splicing relative change</h2><span id='topic+IsoSwitch.PlotExpr'></span>

<h3>Description</h3>

<p>Plots delta PSI vs gene log2-fold change
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IsoSwitch.PlotExpr(MarvelObject, anno = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IsoSwitch.PlotExpr_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td></tr>
<tr><td><code id="IsoSwitch.PlotExpr_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, genes with coordinated, opposing or complex change relative to splicing change will be annotated on the plot. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>MarvelObject$DE$Cor$PSIvsExpr$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- IsoSwitch.PlotExpr(MarvelObject=marvel.demo, anno=TRUE)

# Check output
marvel.demo$DE$Cor$PSIvsExpr$Plot
</code></pre>

<hr>
<h2 id='ModalityChange'>Classify modality changes</h2><span id='topic+ModalityChange'></span>

<h3>Description</h3>

<p>Classifies the type of modality change for each splicing event that has taken place between 2 groups of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ModalityChange(MarvelObject, method, psi.pval, psi.delta = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ModalityChange_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="ModalityChange_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="ModalityChange_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splicing event is considered differentially spliced and included for modality analysis.</p>
</td></tr>
<tr><td><code id="ModalityChange_+3A_psi.delta">psi.delta</code></td>
<td>
<p>Numeric value. The absolute difference between the means PSI values of cell group 1 and 2, above which, the splicing event is considered differentially spliced and included for modality analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$DE$Modality$Table</code>, <code>MarvelObject$DE$Modality$Plot</code>, and <code>MarvelObject$DE$Modality$Plot.Stats</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ModalityChange(MarvelObject=marvel.demo,
                              method="ad",
                              psi.pval=0.1,
                              psi.delta=0
                              )

# Check outputs
head(marvel.demo$DE$Modality$Table)
marvel.demo$DE$Modality$Plot
marvel.demo$DE$Modality$Plot.Stats
</code></pre>

<hr>
<h2 id='ParseGTF'>Parse gene transfer file (GTF)</h2><span id='topic+ParseGTF'></span>

<h3>Description</h3>

<p>Parses the gene transfer file (GTF) for downstream nonsense-mediated decay (NMD) prediction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ParseGTF(MarvelObject)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ParseGTF_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.PSI</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function parses the GTF in order to generate new columns for gene IDs, transcript IDs, and transcript type. These information are extracted from the attribute (9th) column for a standard GTF. These information will be used for downstream NMD prediction.
</p>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$NMD$GTF</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- ParseGTF(MarvelObject=marvel.demo)
</code></pre>

<hr>
<h2 id='PctASE'>Tabulate differentially spliced splicing event</h2><span id='topic+PctASE'></span>

<h3>Description</h3>

<p>Tabulates the percentage or absoluate number of significant splicing events for each splicing type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PctASE(
  MarvelObject,
  method,
  psi.pval,
  psi.mean.diff,
  ylabels.size = 8,
  barlabels.size = 3,
  x.offset = 0,
  direction.color = NULL,
  mode = "percentage"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PctASE_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splicing event is considered differentially spliced and included for tabulation.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_psi.mean.diff">psi.mean.diff</code></td>
<td>
<p>Numeric value. The minimum absolute differences in PSI values between the two cell groups above which the splicing event is considered differentially spliced and included for tabulation.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_ylabels.size">ylabels.size</code></td>
<td>
<p>Numeric value. Size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_barlabels.size">barlabels.size</code></td>
<td>
<p>Numeric value. Size of the labels above each bar. Default is <code>3</code></p>
</td></tr>
<tr><td><code id="PctASE_+3A_x.offset">x.offset</code></td>
<td>
<p>Numeric value. The values on the x-axis to offset by. Useful when right margin overshadow the numbers above the bars. Default value is <code>0</code>.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_direction.color">direction.color</code></td>
<td>
<p>Character strings. Vector of length 2 to specify the colors for significanly down- and up-regulated splicing events. Default is <code>NULL</code>, which corresponds to default <code>ggplot2</code> color scheme.</p>
</td></tr>
<tr><td><code id="PctASE_+3A_mode">mode</code></td>
<td>
<p>Character strings. When set to <code>"percentage"</code> (default), percentage of significant splicing events over total splicing events detected will be tabulate. When set to <code>absolute</code>, the number of significant splicing events will be tabulated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$DE$AbsASE$Table</code> and <code>MarvelObject$DE$AbsASE$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PctASE(MarvelObject=marvel.demo,
                      method="ad",
                      psi.pval=0.1,
                      psi.mean.diff=0
                      )
</code></pre>

<hr>
<h2 id='PlotDEValues'>Plot differential splicing and gene expression analysis results</h2><span id='topic+PlotDEValues'></span>

<h3>Description</h3>

<p>Volcano plot of differential splicing and gene expression analysis results. This is a wrapper function for <code>PlotDEValues.PSI.Mean</code>, <code>PlotDEValues.Exp.Global</code>, and <code>PlotDEValues.Exp.Spliced</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues(
  MarvelObject,
  method = NULL,
  pval,
  level,
  delta = NULL,
  log2fc = NULL,
  psi.pval = NULL,
  psi.delta = NULL,
  gene.pval = NULL,
  gene.log2fc = NULL,
  point.size = 1,
  xlabel.size = 8,
  point.alpha = 1,
  anno = FALSE,
  anno.gene_short_name = NULL,
  anno.tran_id = NULL,
  label.size = 2.5,
  y.upper.offset = 5,
  event.types = c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
  event.types.colors = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing.mean"</code>, <code>"splicing.distance"</code>, and <code>"gene.global"</code>. Adjusted p-value below which the splcing events or genes are considered as statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_level">level</code></td>
<td>
<p>Character string. Indicate <code>"splicing.distance"</code> if the percent spliced-in (PSI) values' distribution was previously tested between 2 groups of cells using the <code>CompareValues</code> function. Statistical tests for distribution include Kolmogorov-Smirnov, Kuiper, and Anderson-Darling test. Indicate <code>"splicing.mean"</code> or <code>gene</code> if the PSI or gene expression values' mean was previously tested between 2 groups of cells using the <code>CompareValues</code> function. Statistical tests for comparing mean are t-test and Wilcoxon rank-sum test.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_delta">delta</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing.mean"</code>. The positive (and negative) value specified above (and below) which the splicing events are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_log2fc">log2fc</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.global"</code>. The positive (and negative) value specified above (and below) which the genes are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The adjusted p-value from differential splicing analysis, below which, the splicing event is considered differentially spliced. Default is <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_psi.delta">psi.delta</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The absolute differences in average PSI value between two cell groups from differential splicing analysis, above which, the splicing event is considered differentially spliced.  Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_gene.pval">gene.pval</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_gene.log2fc">gene.log2fc</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"gene.spliced"</code>. The absolute log2 fold change in gene expression betwene two cell groups from differential splicing analysis, above which, the gene is considered differentially expressed. Default is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing.mean.g2vsg1"</code>. Transpancy of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names or splicing events will be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_anno.gene_short_name">anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the gene names to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_anno.tran_id">anno.tran_id</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the coordinates of the splicing events to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_y.upper.offset">y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_event.types">event.types</code></td>
<td>
<p>Vector of character string(s). Only applicable when <code>level</code> set to <code>"splicing.mean.g2vsg1"</code>. The specific splicing event to plot. May take any one or more of the following values <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, <code>"A3SS"</code>, <code>"AFE"</code>, and  <code>"ALE"</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues_+3A_event.types.colors">event.types.colors</code></td>
<td>
<p>Vector of character string(s). Only applicable when <code>level</code> set to <code>"splicing.mean.g2vsg1"</code>. Customise colors as per splicing event type specified in <code>event.types</code> option. Should be of same length as <code>event.types</code> option.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$DE$PSI$Plot[["method"]]</code> when level set to <code>"splicing.mean"</code> or <code>"splicing.distance"</code> or <code>MarvelObject$DE$Exp.Global$Table</code> and <code>MarvelObject$DE$Exp.Global$Plot</code> when level set to <code>"gene.global"</code> or <code>MarvelObject$DE$Exp.Spliced$Table</code> and <code>MarvelObject$DE$Exp.Spliced$Plot</code> when level set to <code>"gene.spliced"</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues(MarvelObject=marvel.demo,
                            method="ad",
                            pval=0.10,
                            level="splicing.distance"
                            )

# Check output
marvel.demo$DE$PSI$Plot[["ad"]]
</code></pre>

<hr>
<h2 id='PlotDEValues.Exp.Global'>Plot global differential gene expression analysis results</h2><span id='topic+PlotDEValues.Exp.Global'></span>

<h3>Description</h3>

<p>Volcano plot of differential splicing analysis results based on all expressed genes between 2 groups of cells. x-axis represents the log2 fold change in gene expression. y-axis represents the adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.Exp.Global(
  MarvelObject,
  pval = 0.1,
  log2fc = 0.5,
  point.size = 1,
  anno = FALSE,
  anno.gene_short_name = NULL,
  label.size = 2.5,
  y.upper.offset = 5,
  xlabel.size = 8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the genes are considered as statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_log2fc">log2fc</code></td>
<td>
<p>Numeric value. The positive (and negative) value specified above (and below) which the genes are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. The point size for the data points. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names will be annotated on the plot as defined in <code>anno.gene_short_name</code> option.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_anno.gene_short_name">anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the gene names to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_y.upper.offset">y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Global_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$DE$Exp.Global$Table</code>, <code>MarvelObject$DE$Exp.Global$Summary</code>, and <code>MarvelObject$DE$Exp.Global$Plot</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues.Exp.Global(MarvelObject=marvel.demo,
                                       pval=0.10,
                                       log2fc=0.5
                                       )

# Check output
head(marvel.demo$DE$Exp.Global$Table)
marvel.demo$DE$Exp.Global$Plot
marvel.demo$DE$Exp.Global$Summary
</code></pre>

<hr>
<h2 id='PlotDEValues.Exp.Spliced'>Plot differential gene expression analysis of differentially spliced genes</h2><span id='topic+PlotDEValues.Exp.Spliced'></span>

<h3>Description</h3>

<p>Volcano plot of differential splicing analysis results based on differentially spliced genes between 2 groups of cells. x-axis represents the log2 fold change in gene expression. y-axis represents the adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.Exp.Spliced(
  MarvelObject,
  method,
  psi.pval = 0.1,
  psi.delta = 0,
  gene.pval = 0.1,
  gene.log2fc = 0.5,
  point.size = 1,
  anno = FALSE,
  anno.gene_short_name = NULL,
  label.size = 2.5,
  y.upper.offset = 5,
  xlabel.size = 8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_method">method</code></td>
<td>
<p>(Vector of) Character string(s). The <code>method</code> specified in <code>CompareValues</code> function when <code>level</code> option set to <code>"splicing"</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_psi.pval">psi.pval</code></td>
<td>
<p>Numeric value. The adjusted p-value from differential splicing analysis, below which, the splicing event is considered differentially spliced. Default is <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_psi.delta">psi.delta</code></td>
<td>
<p>Numeric value. The absolute differences in average PSI value between two cell groups from differential splicing analysis, above which, the splicing event is considered differentially spliced.  Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_gene.pval">gene.pval</code></td>
<td>
<p>Numeric value. The adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_gene.log2fc">gene.log2fc</code></td>
<td>
<p>Numeric value. The absolute log2 fold change in gene expression betwene two cell groups from differential splicing analysis, above which, the gene is considered differentially expressed. Default is <code>0.5</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names will be annotated on the plot as defined in <code>anno.gene_short_name</code> option.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_anno.gene_short_name">anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the gene names to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_y.upper.offset">y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Exp.Spliced_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$DE$Exp.Spliced$Table</code>, <code>MarvelObject$DE$Exp.Spliced$Summary</code>, and <code>MarvelObject$DE$Exp.Spliced$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues.Exp.Spliced(MarvelObject=marvel.demo,
                                        method="ad",
                                        psi.pval=0.1,
                                        psi.delta=0,
                                        gene.pval=0.1,
                                        gene.log2fc=0.5
                                        )
# Check output
marvel.demo$DE$Exp.Spliced$Summary
marvel.demo$DE$Exp.Spliced$Plot
</code></pre>

<hr>
<h2 id='PlotDEValues.Genes.10x'>Plot differential gene analysis results</h2><span id='topic+PlotDEValues.Genes.10x'></span>

<h3>Description</h3>

<p>Volcano plot of results from differential gene expression analysis. x-axis represents the log2 fold change between two cell groups. y-axis represents -log10(adjusted p-value). Only genes whose splice junctions were considered to be differentially spliced are included for plotting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.Genes.10x(
  MarvelObject,
  pval.sj = 0.05,
  log2fc.sj = NULL,
  delta.sj = 5,
  min.gene.norm = 0,
  pval.adj.gene = 0.05,
  log2fc.gene = 0.5,
  anno = FALSE,
  anno.gene_short_name = NULL,
  label.size = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_pval.sj">pval.sj</code></td>
<td>
<p>Numeric value. p-value from differential splicing analysis, below which, the splice junction is considered differentially spliced. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_log2fc.sj">log2fc.sj</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential splicing analysis, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>delta.sj</code> has been specified.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_delta.sj">delta.sj</code></td>
<td>
<p>Numeric value. Absolute difference in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>log2fc.sj</code> has been specified.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_min.gene.norm">min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction is considered differentially spliced. Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_pval.adj.gene">pval.adj.gene</code></td>
<td>
<p>Numeric value. Adjusted p-value from differential gene expression analysis, below which, the gene is considered differentially expressed. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_log2fc.gene">log2fc.gene</code></td>
<td>
<p>Numeric value. Absolute log2 fold change from differential gene expression analysis, above which, the gene is considered differentially expressed. This option should be <code>NULL</code> if <code>delta.sj</code> has been specified.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, user-specific genes in <code>anno.gene_short_name</code> will be annotated on the plot. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_anno.gene_short_name">anno.gene_short_name</code></td>
<td>
<p>Vector of character strings. If <code>anno</code> set to <code>TRUE</code>, genes specified here will be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.Genes.10x_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. If <code>anno</code> set to <code>TRUE</code>, the font size of the annotations on the plot will be adjusted to the size specified here. Default is <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$DE$SJ$VolcanoPlot$Gene$Plot</code> and <code>MarvelObject$DE$SJ$VolcanoPlot$Gene$Data</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- PlotDEValues.Genes.10x(
                        MarvelObject=marvel.demo.10x,
                        pval.sj=0.05,
                        delta.sj=5,
                        min.gene.norm=1.0,
                        pval.adj.gene=0.05,
                        log2fc.gene=0.5
                        )

# Check outputs
marvel.demo.10x$DE$SJ$VolcanoPlot$Gene$Plot
head(marvel.demo.10x$DE$SJ$VolcanoPlot$Gene$Data)
</code></pre>

<hr>
<h2 id='PlotDEValues.PSI.Distance'>Plot differential splicing analysis results based on distance statistics.</h2><span id='topic+PlotDEValues.PSI.Distance'></span>

<h3>Description</h3>

<p>Ranked plot for differential splicing analysis results based on distance statistics. Only statistical test that assess the overall PSI distribution between two cell groups will be eligible for plotting here, e.g., Anderson-Darling and DTS. x-axis represents the distance statistics. y-axis represents the adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.PSI.Distance(
  MarvelObject,
  method,
  pval,
  point.size = 1,
  xlabel.size = 8,
  anno = FALSE,
  anno.tran_id = NULL,
  label.size = 2.5,
  y.upper.offset = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splcing events are considered as statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. The point size for the data points. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names will be annotated on the plot. Speficified together with <code>anno.tran_id</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_anno.tran_id">anno.tran_id</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the coordinates of the splicing events to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Distance_+3A_y.upper.offset">y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>MarvelObject$DE$PSI$Plot[["method"]]</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues.PSI.Distance(MarvelObject=marvel.demo,
                                         method="ad",
                                         pval=0.10
                                         )

# Check output
marvel.demo$DE$PSI$Plot[["ad"]]
</code></pre>

<hr>
<h2 id='PlotDEValues.PSI.Mean'>Plot differential splicing analysis results based on mean PSI difference</h2><span id='topic+PlotDEValues.PSI.Mean'></span>

<h3>Description</h3>

<p>Volcano plot of differential splicing analysis results based on mean PSI difference between 2 groups of cells. x-axis represents the mean delta PSI. y-axis represents the adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.PSI.Mean(
  MarvelObject,
  method,
  pval = 0.1,
  delta = 5,
  point.size = 1,
  xlabel.size = 8,
  anno = FALSE,
  anno.tran_id = NULL,
  label.size = 2.5,
  y.upper.offset = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splcing event are considered as statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_delta">delta</code></td>
<td>
<p>Numeric value. The positive (and negative) value specified above (and below) which the splicing events are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. The point size for the data points. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names will be annotated on the plot. Speficified together with <code>anno.tran_id</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_anno.tran_id">anno.tran_id</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the coordinates of the splicing events to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean_+3A_y.upper.offset">y.upper.offset</code></td>
<td>
<p>Numeric value. The value in -log10(p-value) to increase the upper limit of the y-axis. To be used when <code>anno</code> set to TRUE so that gene labels will not be truncated at the upper limit of the y-axis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>MarvelObject$DE$PSI$Plot[["method"]]</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues.PSI.Mean(MarvelObject=marvel.demo,
                                     method="ad",
                                     pval=0.10,
                                     delta=5
                                     )

# Check output
marvel.demo$DE$PSI$Plot[["ad"]]
</code></pre>

<hr>
<h2 id='PlotDEValues.PSI.Mean.g2vsg1'>Plot differential splicing analysis results based on mean PSI difference</h2><span id='topic+PlotDEValues.PSI.Mean.g2vsg1'></span>

<h3>Description</h3>

<p>Scatterplot of differential splicing analysis results based on mean PSI difference between 2 groups of cells. x-axis represents the mean PSI values of cell group 1. y-axis represents the mean PSI values of cell group 2.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.PSI.Mean.g2vsg1(
  MarvelObject,
  method,
  pval,
  delta = 5,
  point.size = 1,
  xlabel.size = 8,
  anno = FALSE,
  anno.tran_id = NULL,
  label.size = 2.5,
  point.alpha = 1,
  event.types = c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
  event.types.colors = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_method">method</code></td>
<td>
<p>Character string. The statistical method used for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_pval">pval</code></td>
<td>
<p>Numeric value. Adjusted p-value below which the splcing event are considered as statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_delta">delta</code></td>
<td>
<p>Numeric value. The positive (and negative) value specified above (and below) which the splicing events are considered to be statistically significant and will consequently be color-annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. The point size for the data points. Default value is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_xlabel.size">xlabel.size</code></td>
<td>
<p>Numeric value. Font size of the xtick labels. Default is <code>8</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the specific gene names will be annotated on the plot. Speficified together with <code>anno.tran_id</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_anno.tran_id">anno.tran_id</code></td>
<td>
<p>Vector of character strings. When <code>anno</code> set to <code>TRUE</code>, the coordinates of the splicing events to be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. Only applicable if <code>anno</code> set to TRUE. Size of the gene name labels.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transpancy of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_event.types">event.types</code></td>
<td>
<p>Vector of character string(s). The specific splicing event to plot. May take any one or more of the following values <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code>, <code>"A3SS"</code>, <code>"AFE"</code>, and  <code>"ALE"</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.PSI.Mean.g2vsg1_+3A_event.types.colors">event.types.colors</code></td>
<td>
<p>Vector of character string(s). Customise colors as per splicing event type specified in <code>event.types</code> option. Should be of same length as <code>event.types</code> option.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>MarvelObject$DE$PSI$Plot[["method"]]</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PlotDEValues.PSI.Mean.g2vsg1(MarvelObject=marvel.demo,
                                     method="ad",
                                     pval=0.10,
                                     delta=5
                                     )

# Check output
marvel.demo$DE$PSI$Plot
marvel.demo$DE$PSI$Summary
</code></pre>

<hr>
<h2 id='PlotDEValues.SJ.10x'>Plot differential splice junction analysis results</h2><span id='topic+PlotDEValues.SJ.10x'></span>

<h3>Description</h3>

<p>Volcano plot of results from differential splice junction analysis. x-axis represents the average normalised gene expression across the two cell groups. y-axis represents the differences or log2 fold change between the two cell groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotDEValues.SJ.10x(
  MarvelObject,
  pval = 0.05,
  log2fc = NULL,
  delta = 5,
  min.gene.norm = 0,
  anno = FALSE,
  anno.coord.intron = NULL,
  label.size = 2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CompareValues.Genes.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_pval">pval</code></td>
<td>
<p>Numeric value. p-value, below which, the splice junction is considered differentially spliced. To be used in conjunction with <code>log2fc</code>, <code>delta</code>, and <code>min.gene.norm</code>. Default is <code>0.05</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_log2fc">log2fc</code></td>
<td>
<p>Numeric value. Absolute log2 fold change, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>delta</code> has been specified.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_delta">delta</code></td>
<td>
<p>Numeric value. Absolute differences in average PSI values between the two cell groups, above which, the splice junction is considered differentially spliced. This option should be <code>NULL</code> if <code>log2fc</code> has been specified.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_min.gene.norm">min.gene.norm</code></td>
<td>
<p>Numeric value. The average normalised gene expression across the two cell groups above which the splice junction is considered differentially spliced. Default is <code>0</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_anno">anno</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, user-specific spliced genes in <code>anno.coord.intron</code> will be annotated on the plot. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_anno.coord.intron">anno.coord.intron</code></td>
<td>
<p>Vector of character strings. If <code>anno</code> set to <code>TRUE</code>, splice junction coordinates specified here will be annotated on the plot.</p>
</td></tr>
<tr><td><code id="PlotDEValues.SJ.10x_+3A_label.size">label.size</code></td>
<td>
<p>Numeric value. If <code>anno</code> set to <code>TRUE</code>, the font size of the annotations on the plot will be adjusted to the size specified here. Default is <code>2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$DE$SJ$VolcanoPlot$SJ$Plot</code> and <code>MarvelObject$DE$SJ$VolcanoPlot$SJ$Data</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

marvel.demo.10x &lt;- PlotDEValues.SJ.10x(
                        MarvelObject=marvel.demo.10x,
                        pval=0.05,
                        delta=5,
                        min.gene.norm=1.0,
                        anno=FALSE
                        )

# Check outputs
marvel.demo.10x$DE$SJ$VolcanoPlot$SJ$Plot
head(marvel.demo.10x$DE$SJ$VolcanoPlot$SJ$Data)
</code></pre>

<hr>
<h2 id='PlotPctExprCells.Genes.10x'>Plot gene expression distribution</h2><span id='topic+PlotPctExprCells.Genes.10x'></span>

<h3>Description</h3>

<p>Generates a plot of gene expression distribution (percentage of cells expressing a particular gene) to determine normalised gene expression threshold for downstream differential splice junction analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPctExprCells.Genes.10x(
  MarvelObject,
  cell.group.g1,
  cell.group.g2,
  min.pct.cells = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPctExprCells.Genes.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.Genes.10x_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 1 (reference group) of downstream differential splice junction analysis.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.Genes.10x_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 2 of downstream differential splice junction analysis.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.Genes.10x_+3A_min.pct.cells">min.pct.cells</code></td>
<td>
<p>Numeric value. Minimum percentage of cells in which the gene is expressed for that gene to be included for gene expression distribution analysis. Expressed genes defined as genes with non-zero normalised UMI counts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$pct.cells.expr$Gene$Plot</code> and <code>MarvelObject$pct.cells.expr$Gene$Data</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # Group 1 (reference)
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Group 2
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

# Explore % of cells expressing genes
marvel.demo.10x &lt;- PlotPctExprCells.Genes.10x(
                        MarvelObject=marvel.demo.10x,
                        cell.group.g1=cell.ids.1,
                        cell.group.g2=cell.ids.2,
                        min.pct.cells=5
                        )

# Check output
marvel.demo.10x $pct.cells.expr$Gene$Plot
head(marvel.demo.10x $pct.cells.expr$Gene$Data)
</code></pre>

<hr>
<h2 id='PlotPctExprCells.SJ.10x'>Plot splice junction expression distribution</h2><span id='topic+PlotPctExprCells.SJ.10x'></span>

<h3>Description</h3>

<p>Generates a plot of splice junction expression distribution (percentage of cells expressing a particular splice junction) to determine splice junction expression threshold for downstream differential splice junction analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotPctExprCells.SJ.10x(
  MarvelObject,
  cell.group.g1,
  cell.group.g2,
  min.pct.cells.genes = 10,
  min.pct.cells.sj = 10,
  downsample = FALSE,
  downsample.pct.sj = 10,
  seed = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_cell.group.g1">cell.group.g1</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 1 (reference group) of downstream differential splice junction analysis.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_cell.group.g2">cell.group.g2</code></td>
<td>
<p>Vector of character strings. Cell IDs corresponding to Group 2 of downstream differential splice junction analysis.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_min.pct.cells.genes">min.pct.cells.genes</code></td>
<td>
<p>Numeric value. Minimum percentage of cells in which the gene is expressed for that gene to be included for splice junction expression distribution analysis. Expressed genes defined as genes with non-zero normalised UMI counts. This threshold may be determined from <code>PlotPctExprCells.SJ.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_min.pct.cells.sj">min.pct.cells.sj</code></td>
<td>
<p>Numeric value. Minimum percentage of cells in which the splice junction is expressed for that splice junction to be included for splice junction expression distribution analysis. Expressed splice junctions defined as splice junctions with raw UMI counts &gt;= 1.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_downsample">downsample</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, the splice junctions will be downsampled so that only a smaller number of splice junctions will be included for expression exploration analysis here. Default value is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_downsample.pct.sj">downsample.pct.sj</code></td>
<td>
<p>Numeric value. If <code>downsample</code> set to <code>TRUE</code>, the minimum percentage of splice junctions to include for expression exploration analysis here.</p>
</td></tr>
<tr><td><code id="PlotPctExprCells.SJ.10x_+3A_seed">seed</code></td>
<td>
<p>Numeric value. To ensure the splice junctions downsampled will always be reproducible.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with a new slots <code>MarvelObject$pct.cells.expr$SJ$Plot</code> and <code>MarvelObject$pct.cells.expr$SJ$Data</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # Group 1 (reference)
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Group 2
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

# Explore % of cells expressing SJ
marvel.demo.10x &lt;- PlotPctExprCells.SJ.10x(
                    MarvelObject=marvel.demo.10x,
                    cell.group.g1=cell.ids.1,
                    cell.group.g2=cell.ids.2,
                    min.pct.cells.genes=5,
                    min.pct.cells.sj=5,
                    downsample=TRUE,
                    downsample.pct.sj=100
                    )

marvel.demo.10x$pct.cells.expr$SJ$Plot
head(marvel.demo.10x$pct.cells.expr$SJ$Data)
</code></pre>

<hr>
<h2 id='PlotValues'>Plot percent spliced-in (PSI) or gene expression values</h2><span id='topic+PlotValues'></span>

<h3>Description</h3>

<p>Plots percent spliced-in (PSI) or gene expression values across different groups of cells. This is a wrapper function for <code>PlotValues.Exp</code> and <code>PlotValues.PSI</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotValues(
  MarvelObject,
  cell.group.list,
  feature,
  maintitle = "gene_short_name",
  xlabels.size = 8,
  level,
  min.cells = NULL,
  sigma.sq = 0.001,
  bimodal.adjust = NULL,
  seed = NULL,
  modality.column = "modality.bimodal.adj",
  scale.y.log = FALSE,
  max.cells.jitter = 10000,
  max.cells.jitter.seed = 1,
  cell.group.colors = NULL,
  point.alpha = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotValues_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_cell.group.list">cell.group.list</code></td>
<td>
<p>List of character strings. Each element of the list is a vector of cell IDs corresponding to a cell group. The name of the element will be the cell group label.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_feature">feature</code></td>
<td>
<p>Character string. <code>tran_id</code> or <code>gene_id</code> for plotting. Should match <code>tran_id</code> or <code>gene_id</code> column of <code>MarvelObject$ValidatedSpliceFeature</code> or <code>MarvelObject$GeneFeature</code> slot when <code>level</code> set to <code>"splicing"</code> or <code>"gene"</code>, respectively.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_maintitle">maintitle</code></td>
<td>
<p>Character string. Column to use as plot main title as per <code>MarvelObject$ValidatedSpliceFeature</code> or <code>MarvelObject$GeneFeature</code> when <code>level</code> set to <code>"splicing"</code> or <code>"gene"</code>, respectively. Default is <code>"gene_short_name"</code> column.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Size of x-axis labels as per <code>ggplot2</code> function. Default is 8.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_level">level</code></td>
<td>
<p>Character string. Indicate <code>"splicing"</code> or <code>"gene"</code> for PSI or gene expression value plotting, respectively.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. The minimum no. of cells expressing the splicing event to be included for analysis.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_sigma.sq">sigma.sq</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. The variance threshold below which the included/excluded modality will be defined as primary sub-modality, and above which it will be defined as dispersed sub-modality. Please refer to <code>AssignModality</code> function help page for more details. Default is 0.001.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_bimodal.adjust">bimodal.adjust</code></td>
<td>
<p>Logical. Only applicable when <code>level</code> set to <code>"splicing"</code>. When set to <code>TRUE</code>, MARVEL will identify false bimodal modalities and reassign them as included/excluded modality. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. Ensure the <code>fitdist</code> function returns the same values for alpha and beta paramters each time this function is executed using the same random number generator. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_modality.column">modality.column</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"splicing"</code>. Can take the value <code>"modality"</code>, <code>"modality.var"</code> or <code>"modality.bimodal.adj"</code>. Please refer to <code>AssignModality</code> function help page for more details. Default is <code>"modality.bimodal.adj"</code>.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_scale.y.log">scale.y.log</code></td>
<td>
<p>Logical value. Only applicable when <code>level</code> set to <code>"splicing"</code>. If set to <code>TRUE</code>, the y-axis of will log10-scaled. Useful when most PSI values are extremely small (&lt; 0.02) or big (&gt; 0.98). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_max.cells.jitter">max.cells.jitter</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. Maximum number of cells for jitter points. Cells are randomly downsampled to show on jitter plot. Useful when there are large number of cells so that individual jitter points do not overcrowd the violin plot. Specified together with <code>max.cells.jitter.seed</code>. To disable this option, specify a value large than the number of cells in each cell group.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_max.cells.jitter.seed">max.cells.jitter.seed</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>. Cells downsampled are reproducible. Specified together with <code>max.cells.jitter</code>.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_cell.group.colors">cell.group.colors</code></td>
<td>
<p>Character string. Vector of colors for the cell groups specified for PCA analysis using <code>cell.type.columns</code>, <code>cell.type.variable</code>, and <code>cell.type.labels</code>. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
<tr><td><code id="PlotValues_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points. Takes any values between 0-1. Default value is <code>0.2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>$adhocPlot$PSI</code> or <code>MarvelObject$adhocPlot$Exp</code> when <code>level</code> set to <code>"splicing"</code> or <code>"gene"</code>, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell groups to plot
df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]
cell.group.list &lt;- list(cell.group.g1, cell.group.g2)
names(cell.group.list) &lt;- c("iPSC", "Endoderm")

# Plot
marvel.demo &lt;- PlotValues(MarvelObject=marvel.demo,
                          cell.group.list=cell.group.list,
                          feature="chr17:8383254:8382781|8383157:-@chr17:8382143:8382315",
                          level="splicing",
                          min.cells=5,
                          xlabels.size=5
                          )

# Check output
marvel.demo$adhocPlot$PSI
</code></pre>

<hr>
<h2 id='PlotValues.Exp'>Plot gene expression values</h2><span id='topic+PlotValues.Exp'></span>

<h3>Description</h3>

<p>Boxplot of gene expression values across different groups of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotValues.Exp(
  MarvelObject,
  cell.group.list,
  feature,
  maintitle = "gene_short_name",
  xlabels.size = 8,
  cell.group.colors = NULL,
  point.alpha = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotValues.Exp_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotValues.Exp_+3A_cell.group.list">cell.group.list</code></td>
<td>
<p>List of character strings. Each element of the list is a vector of cell IDs corresponding to a cell group. The name of the element will be the cell group label.</p>
</td></tr>
<tr><td><code id="PlotValues.Exp_+3A_feature">feature</code></td>
<td>
<p>Character string. <code>gene_id</code> for plotting. Should match <code>gene_id</code> column of <code>MarvelObject$GeneFeature</code> slot.</p>
</td></tr>
<tr><td><code id="PlotValues.Exp_+3A_maintitle">maintitle</code></td>
<td>
<p>Character string. Column to use as plot main title as per <code>MarvelObject$GeneFeature</code>. Default is <code>"gene_short_name"</code> column.</p>
</td></tr>
<tr><td><code id="PlotValues.Exp_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Size of x-axis labels as per <code>ggplot2</code> function. Default is 8.</p>
</td></tr>
<tr><td><code id="PlotValues.Exp_+3A_cell.group.colors">cell.group.colors</code></td>
<td>
<p>Character string. Vector of colors for the cell groups specified for PCA analysis using <code>cell.type.columns</code>, <code>cell.type.variable</code>, and <code>cell.type.labels</code>. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
<tr><td><code id="PlotValues.Exp_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points. Takes any values between 0-1. Default value is <code>0.2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$adhocPlot$Exp</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]
cell.group.list &lt;- list(cell.group.g1, cell.group.g2)
names(cell.group.list) &lt;- c("iPSC", "Endoderm")

# Plot
marvel.demo &lt;- PlotValues.Exp(MarvelObject=marvel.demo,
                              cell.group.list=cell.group.list,
                              feature="ENSG00000161970.15",
                              xlabels.size=8
                              )

# Check output
marvel.demo$adhocPlot$Exp
</code></pre>

<hr>
<h2 id='PlotValues.PCA.CellGroup.10x'>Annotate reduced dimension space with cell feature</h2><span id='topic+PlotValues.PCA.CellGroup.10x'></span>

<h3>Description</h3>

<p>Annotates reduced dimension space, e.g., UMAP and tSNE, with cell features such as cell group, donor ID, sample ID, etc.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotValues.PCA.CellGroup.10x(
  MarvelObject,
  cell.group.list,
  legendtitle = "Cell group",
  alpha = 0.75,
  point.size = 1,
  point.stroke = 0.1,
  point.colors = NULL,
  point.size.legend = 2,
  type
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_cell.group.list">cell.group.list</code></td>
<td>
<p>List of character strings. Each element of the list is a vector of cell IDs corresponding to a feature, e.g. cell group. The names of the element will be the cell feature label.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_legendtitle">legendtitle</code></td>
<td>
<p>Character string. Legend title. Default is <code>"Cell group"</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_alpha">alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points. Takes any values between 0-1 whereby 0 is totally transparent and 1 is opaque. Default is <code>0.75</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_point.stroke">point.stroke</code></td>
<td>
<p>Numeric value. Outline thickness of data points.  Default is <code>0.1</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_point.colors">point.colors</code></td>
<td>
<p>Vector of character strings. Colors of cell groups and should be same length as <code>cell.group.list</code>. Default <code>ggplot2</code> colors are used.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_point.size.legend">point.size.legend</code></td>
<td>
<p>Numeric value. Size of legend keys. Default is <code>2</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.CellGroup.10x_+3A_type">type</code></td>
<td>
<p>Character string. Type of reduced dimension space. Options are <code>"umap"</code> and <code>"tsne"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$adhocPlot$PCA$CellGroup</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # iPSC
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Cardio day 10
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

    # Save into list
    cell.group.list &lt;- list("iPSC"=cell.ids.1,
                            "Cardio d10"=cell.ids.2
                            )

# Plot cell groups
marvel.demo.10x &lt;- PlotValues.PCA.CellGroup.10x(
                            MarvelObject=marvel.demo.10x,
                            cell.group.list=cell.group.list,
                            legendtitle="Cell group",
                            type="tsne"
                            )

# Check output
marvel.demo.10x$adhocPlot$PCA$CellGroup
</code></pre>

<hr>
<h2 id='PlotValues.PCA.Gene.10x'>Annotate reduced dimension space with gene expression values</h2><span id='topic+PlotValues.PCA.Gene.10x'></span>

<h3>Description</h3>

<p>Annotates reduced dimension space, e.g., UMAP and tSNE, with gene expression values. Values will be automatically be log2-transformed prior to plotting.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotValues.PCA.Gene.10x(
  MarvelObject,
  cell.ids = NULL,
  gene_short_name,
  log2.transform = TRUE,
  point.size = 0.1,
  color.gradient = c("grey90", "blue", "red"),
  type
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_cell.ids">cell.ids</code></td>
<td>
<p>Vector of character strings. Specify specific cells to plot.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_gene_short_name">gene_short_name</code></td>
<td>
<p>Character string. Gene name whose expression will be plotting.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_log2.transform">log2.transform</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> (default), normalised gene expression values will be off-set by 1 and then log2-transformed prior to analysis.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_color.gradient">color.gradient</code></td>
<td>
<p>Vector of character strings. Colors to indicate low, moderate, and high expression. Default is <code>c("grey90","blue","red")</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.Gene.10x_+3A_type">type</code></td>
<td>
<p>Character string. Type of reduced dimension space. Options are <code>"umap"</code> and <code>"tsne"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$adhocPlot$PCA$Gene</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # iPSC
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Cardio day 10
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

    # Save into list
    cell.group.list &lt;- list("iPSC"=cell.ids.1,
                            "Cardio d10"=cell.ids.2
                            )

# Plot expression
marvel.demo.10x &lt;- PlotValues.PCA.Gene.10x(
                      MarvelObject=marvel.demo.10x,
                      gene_short_name="TPM2",
                      color.gradient=c("grey","cyan","green","yellow","red"),
                      type="tsne"
                      )

# Check output
marvel.demo.10x$adhocPlot$PCA$Gene
</code></pre>

<hr>
<h2 id='PlotValues.PCA.PSI.10x'>Annotate reduced dimension space with PSI values</h2><span id='topic+PlotValues.PCA.PSI.10x'></span>

<h3>Description</h3>

<p>Annotates reduced dimension space, e.g., UMAP and tSNE, with PSI values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotValues.PCA.PSI.10x(
  MarvelObject,
  cell.ids = NULL,
  coord.intron,
  min.gene.count = 3,
  point.size = 0.1,
  log2.transform = FALSE,
  color.gradient = c("grey90", "blue", "red"),
  type
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment.10x</code> function.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_cell.ids">cell.ids</code></td>
<td>
<p>Vector of character strings. Specific set of cells to plot.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_coord.intron">coord.intron</code></td>
<td>
<p>Character string. Coordinates of splice junction whose expression will be plotted.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_min.gene.count">min.gene.count</code></td>
<td>
<p>Numeric value. Minimum raw gene count, above which, the PSI value will be calculate for the cell. Default is <code>3</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points. Default is <code>1</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_log2.transform">log2.transform</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, PSI values will be log2-transformed. Useful for highlighting small changes in PSI values between cell groups. Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_color.gradient">color.gradient</code></td>
<td>
<p>Vector of character strings. Colors to indicate low, moderate, and high expression. Default is <code>c("grey90","blue","red")</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PCA.PSI.10x_+3A_type">type</code></td>
<td>
<p>Character string. Type of reduced dimension space. Options are <code>"umap"</code> and <code>"tsne"</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$adhocPlot$PCA$PSI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
marvel.demo.10x &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.rds",
                               package="MARVEL")
                               )

# Define cell groups
    # Retrieve sample metadata
    sample.metadata &lt;- marvel.demo.10x$sample.metadata

    # iPSC
    index &lt;- which(sample.metadata$cell.type=="iPSC")
    cell.ids.1 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.1)

    # Cardio day 10
    index &lt;- which(sample.metadata$cell.type=="Cardio day 10")
    cell.ids.2 &lt;- sample.metadata[index, "cell.id"]
    length(cell.ids.2)

    # Save into list
    cell.group.list &lt;- list("iPSC"=cell.ids.1,
                            "Cardio d10"=cell.ids.2
                            )

# Plot expression
marvel.demo.10x &lt;- PlotValues.PCA.PSI.10x(
                        MarvelObject=marvel.demo.10x,
                        coord.intron="chr1:23693914:23694659",
                        min.gene.count=3,
                        log2.transform=FALSE,
                        color.gradient=c("grey","cyan","green","yellow","red"),
                        type="tsne"
                        )

# Check output
marvel.demo.10x$adhocPlot$PCA$PSI
</code></pre>

<hr>
<h2 id='PlotValues.PSI'>Plot percent spliced-in (PSI) values</h2><span id='topic+PlotValues.PSI'></span>

<h3>Description</h3>

<p>Violin plot of percent spliced-in (PSI) values across different groups of cells.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotValues.PSI(
  MarvelObject,
  cell.group.list,
  feature,
  maintitle = "gene_short_name",
  xlabels.size = 8,
  max.cells.jitter = 10000,
  max.cells.jitter.seed = 1,
  min.cells = 25,
  sigma.sq = 0.001,
  bimodal.adjust = TRUE,
  seed = 1,
  modality.column = "modality.bimodal.adj",
  scale.y.log = FALSE,
  cell.group.colors = NULL,
  point.alpha = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotValues.PSI_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_cell.group.list">cell.group.list</code></td>
<td>
<p>List of character strings. Each element of the list is a vector of cell IDs corresponding to a cell group. The name of the element will be the cell group label.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_feature">feature</code></td>
<td>
<p>Character string. Coordinates of splicing event to plot.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_maintitle">maintitle</code></td>
<td>
<p>Character string. Column to use as plot main title as per <code>MarvelObject$ValidatedSpliceFeature</code>. Default is <code>"gene_short_name"</code> column.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Size of x-axis labels as per <code>ggplot2</code> function. Default is 8.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_max.cells.jitter">max.cells.jitter</code></td>
<td>
<p>Numeric value. Maximum number of cells for jitter points. Cells are randomly downsampled to show on jitter plot. Useful when there are large number of cells so that individual jitter points do not overcrowd the violin plot.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_max.cells.jitter.seed">max.cells.jitter.seed</code></td>
<td>
<p>Numeric value. Cells downsampled are reproducible.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the splicing event to be included for analysis. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_sigma.sq">sigma.sq</code></td>
<td>
<p>Numeric value. The variance threshold below which the included/excluded modality will be defined as primary sub-modality, and above which it will be defined as dispersed sub-modality. Please refer to <code>AssignModality</code> function help page for more details. Default is 0.001.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_bimodal.adjust">bimodal.adjust</code></td>
<td>
<p>Logical. When set to <code>TRUE</code>, MARVEL will identify false bimodal modalities and reassign them as included/excluded modality. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Ensure the <code>fitdist</code> function returns the same values for alpha and beta paramters each time this function is executed using the same random number generator. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_modality.column">modality.column</code></td>
<td>
<p>Character string. Can take the value <code>"modality"</code>, <code>"modality.var"</code> or <code>"modality.bimodal.adj"</code>. Please refer to <code>AssignModality</code> function help page for more details. Default is <code>"modality.bimodal.adj"</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_scale.y.log">scale.y.log</code></td>
<td>
<p>Logical value. Only applicable when <code>level</code> set to <code>"splicing"</code>. If set to <code>TRUE</code>, the y-axis of will log10-scaled. Useful when most PSI values are extremely small (&lt; 0.02) or big (&gt; 0.98). Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_cell.group.colors">cell.group.colors</code></td>
<td>
<p>Character string. Vector of colors for the cell groups specified for PCA analysis using <code>cell.type.columns</code>, <code>cell.type.variable</code>, and <code>cell.type.labels</code>. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
<tr><td><code id="PlotValues.PSI_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points. Takes any values between 0-1. Default value is <code>0.2</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slot <code>MarvelObject$adhocPlot$PSI</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define cell groups to plot
df.pheno &lt;- marvel.demo$SplicePheno
cell.group.g1 &lt;- df.pheno[which(df.pheno$cell.type=="iPSC"), "sample.id"]
cell.group.g2 &lt;- df.pheno[which(df.pheno$cell.type=="Endoderm"), "sample.id"]
cell.group.list &lt;- list(cell.group.g1, cell.group.g2)
names(cell.group.list) &lt;- c("iPSC", "Endoderm")

# Plot
marvel.demo &lt;- PlotValues.PSI(MarvelObject=marvel.demo,
                              cell.group.list=cell.group.list,
                              feature="chr17:8383254:8382781|8383157:-@chr17:8382143:8382315",
                              min.cells=5,
                              xlabels.size=5
                              )

# Check output
marvel.demo$adhocPlot$PSI
</code></pre>

<hr>
<h2 id='PropModality'>Tabulate modality proportion</h2><span id='topic+PropModality'></span>

<h3>Description</h3>

<p>Tabulates and plots the proportion of each modality. This is a wrapper function for <code>PropModality.Doughnut</code> and <code>PropModality.Bar</code> functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PropModality(
  MarvelObject,
  modality.column,
  modality.type,
  event.type,
  across.event.type,
  prop.test = NULL,
  prop.adj = NULL,
  xlabels.size = 8,
  zoom = FALSE,
  yinterval = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PropModality_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>AssignModality</code> function.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_modality.column">modality.column</code></td>
<td>
<p>Character string. Can take the value <code>"modality"</code>, <code>"modality.var"</code> or <code>"modality.bimodal.adj"</code>. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_modality.type">modality.type</code></td>
<td>
<p>Character string. <code>basic</code> indicates that only the main modalities (included, excluded, bimodal, middle, multimodal) are analysed. Sub-modalities (primary and dispersed) will be merged. <code>complete</code> indicates that both main and sub-modalities are analysed. Sub-modalities will not be merged.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_event.type">event.type</code></td>
<td>
<p>Character string. To indicate which event type to analyse. Can take the value <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code> or <code>"A3SS"</code>. Specify <code>"all"</code> to include all event types.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_across.event.type">across.event.type</code></td>
<td>
<p>Logical. If set to <code>TRUE</code>, the proportion of modality will be compared across the specified event types</p>
</td></tr>
<tr><td><code id="PropModality_+3A_prop.test">prop.test</code></td>
<td>
<p>Character string. Only applicable when <code>across.event.type</code> set to <code>TRUE</code>. <code>chisq</code> Chi-squared test used to compare the proportion of modalities across the different event splicing type. <code>fisher</code> Fisher test used to compare the proportion of modalities across the different splicing event type.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_prop.adj">prop.adj</code></td>
<td>
<p>Character string. Only applicable when <code>across.event.type</code> set to <code>TRUE</code>. Adjust p-values generated from <code>prop.test</code> for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Only applicable when <code>across.event.type</code> set to <code>TRUE</code>. Size of x-axis labels as per <code>ggplot2</code> function. Default is 8.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_zoom">zoom</code></td>
<td>
<p>Logical value. Only applicable if <code>across.event.type</code> set to <code>TRUE</code>. If set to <code>TRUE</code>, users can specify the range of the y-axis using <code>yinterval</code> argument. Useful when scrutinasing low-frequency event types, e.g. middle and multimodal.</p>
</td></tr>
<tr><td><code id="PropModality_+3A_yinterval">yinterval</code></td>
<td>
<p>Logical value. Only applicable if <code>across.event.type</code> set to <code>TRUE</code> and <code>zoom</code> set to <code>TRUE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slot <code>$Modality$Prop$DoughnutChart</code> or <code>$Modality$Prop$BarChart</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PropModality(MarvelObject=marvel.demo,
                            modality.column="modality.bimodal.adj",
                            modality.type="extended",
                            event.type=c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
                            across.event.type=FALSE
                            )

# Check outputs
marvel.demo$Modality$Prop$DoughnutChart$Table
marvel.demo$Modality$Prop$DoughnutChart$Plot
</code></pre>

<hr>
<h2 id='PropModality.Bar'>Modality proportion broken down by event type</h2><span id='topic+PropModality.Bar'></span>

<h3>Description</h3>

<p>Tabulates and plots the proportion of each modality broken down by splicing event type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PropModality.Bar(
  MarvelObject,
  modality.column,
  modality.type,
  event.type,
  xlabels.size = 8,
  zoom = FALSE,
  yinterval = NULL,
  prop.test,
  prop.adj
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PropModality.Bar_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>AssignModality</code> function.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_modality.column">modality.column</code></td>
<td>
<p>Character string. Can take the value <code>"modality"</code>, <code>"modality.var"</code> or <code>"modality.bimodal.adj"</code>. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_modality.type">modality.type</code></td>
<td>
<p>Character string. <code>basic</code> indicates that only the main modalities (included, excluded, bimodal, middle, multimodal) are analysed. Sub-modalities (primary and dispersed) will be merged. <code>extended</code> indicates that both main and sub-modalities are analysed. Sub-modalities will not be merged.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_event.type">event.type</code></td>
<td>
<p>Character string. To indicate which event type to analyse. Can take the value <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code> or <code>"A3SS"</code>. Specify <code>"all"</code> to include all event types.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Size of x-axis labels as per <code>ggplot2</code> function. Default is 8.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_zoom">zoom</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, users can specify the range of the y-axis using <code>yinterval</code> argument. Useful when scrutinasing low-frequency event types, e.g. middle and multimodal.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_yinterval">yinterval</code></td>
<td>
<p>Logical value. Only applicable when <code>zoom</code> is set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_prop.test">prop.test</code></td>
<td>
<p>Character string. Only applicable when <code>across.event.type</code> set to <code>TRUE</code>. <code>chisq</code> Chi-squared test used to compare the proportion of modalities across the different event splicing type. <code>fisher</code> Fisher test used to compare the proportion of modalities across the different splicing event type.</p>
</td></tr>
<tr><td><code id="PropModality.Bar_+3A_prop.adj">prop.adj</code></td>
<td>
<p>Character string. Only applicable when <code>across.event.type</code> set to <code>TRUE</code>. Adjust p-values generated from <code>prop.test</code> for multiple testing. Options available as per <code>p.adjust</code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing new slots <code>MarvelObject$Modality$Prop$BarChart$Table</code> and <code>MarvelObject$Modality$Prop$BarChart$Stats</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PropModality.Bar(MarvelObject=marvel.demo,
                                modality.column="modality.bimodal.adj",
                                modality.type="extended",
                                event.type=c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE"),
                                prop.test="fisher",
                                prop.adj="fdr"
                                )

# Check outputs
head(marvel.demo$Modality$Prop$BarChart$Table)
marvel.demo$Modality$Prop$BarChart$Plot
marvel.demo$Modality$Prop$BarChart$Stats
</code></pre>

<hr>
<h2 id='PropModality.Doughnut'>Overall modality proportion</h2><span id='topic+PropModality.Doughnut'></span>

<h3>Description</h3>

<p>Tabulates and plots the proportion of each modality without breaking down by splicing event type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PropModality.Doughnut(MarvelObject, modality.column, modality.type, event.type)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PropModality.Doughnut_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>AssignModality</code> function.</p>
</td></tr>
<tr><td><code id="PropModality.Doughnut_+3A_modality.column">modality.column</code></td>
<td>
<p>Character string. Can take the value <code>"modality"</code>, <code>"modality.var"</code> or <code>"modality.bimodal.adj"</code>. Please refer to <code>AssignModality</code> function help page for more details.</p>
</td></tr>
<tr><td><code id="PropModality.Doughnut_+3A_modality.type">modality.type</code></td>
<td>
<p>Character string. <code>basic</code> indicates that only the main modalities (included, excluded, bimodal, middle, multimodal) are analysed. Sub-modalities (primary and dispersed) will be merged. <code>complete</code> indicates that both main and sub-modalities are analysed. Sub-modalities will not be merged.</p>
</td></tr>
<tr><td><code id="PropModality.Doughnut_+3A_event.type">event.type</code></td>
<td>
<p>Character string. To indicate which event type to analyse. Can take the value <code>"SE"</code>, <code>"MXE"</code>, <code>"RI"</code>, <code>"A5SS"</code> or <code>"A3SS"</code>. Specify <code>"all"</code> to include all event types.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$Modality$Prop$DoughnutChart$Table</code> and <code>MarvelObject$Modality$Prop$DoughnutChart$Plot</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PropModality.Doughnut(MarvelObject=marvel.demo,
                                     modality.column="modality.bimodal.adj",
                                     modality.type="extended",
                                     event.type=c("SE", "MXE", "RI", "A5SS", "A3SS", "AFE", "ALE")
                                     )

# Check outputs
marvel.demo$Modality$Prop$DoughnutChart$Table
marvel.demo$Modality$Prop$DoughnutChart$Plot
</code></pre>

<hr>
<h2 id='PropPTC'>Tabulate proportion of transcripts with PTC</h2><span id='topic+PropPTC'></span>

<h3>Description</h3>

<p>Tabulates and plots the proportion of transcripts with PTC for each splicing event type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PropPTC(MarvelObject, xlabels.size = 8, show.NovelSJ.NoCDS = TRUE, prop.test)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PropPTC_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>FindPTC</code> function.</p>
</td></tr>
<tr><td><code id="PropPTC_+3A_xlabels.size">xlabels.size</code></td>
<td>
<p>Numeric value. Size of the x-axis tick labels. Default is 8.</p>
</td></tr>
<tr><td><code id="PropPTC_+3A_show.novelsj.nocds">show.NovelSJ.NoCDS</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code> transcripts not analysed for premature terminal codon (PTC), e.g. non-protein-coding transcripts are tabulated and plotted.</p>
</td></tr>
<tr><td><code id="PropPTC_+3A_prop.test">prop.test</code></td>
<td>
<p>Character string. <code>chisq</code> Chi-squared test used to compare the proportion of transcripts with PTC across the different event splicing type. <code>fisher</code> Fisher test used to compare the proportion of transcripts with PTC across the different splicing event type.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$NMD$PTC.Prop$Table</code>, <code>MarvelObject$NMD$PTC.Prop$Plot</code>, and <code>MarvelObject$NMD$PTC.Prop$Plot.Stats</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- PropPTC(MarvelObject=marvel.demo,
                       xlabels.size=8,
                       show.NovelSJ.NoCDS=TRUE,
                       prop.test="fisher"
                       )

# Check outputs
head(marvel.demo$NMD$PTC.Prop$Table)
marvel.demo$NMD$PTC.Prop$Plot
marvel.demo$NMD$PTC.Prop$Plot.Stats
</code></pre>

<hr>
<h2 id='RunPCA'>Principle component analysis</h2><span id='topic+RunPCA'></span>

<h3>Description</h3>

<p>Performs principle component analysis on splicing or gene data. This is a wrapper function for <code>RunPCA.PSI</code> and <code>RunPCA.Exp</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunPCA(
  MarvelObject,
  cell.group.column,
  cell.group.order = NULL,
  cell.group.colors = NULL,
  sample.ids = NULL,
  min.cells = 25,
  features,
  point.size = 0.5,
  point.alpha = 0.75,
  point.stroke = 0.1,
  seed = 1,
  method.impute = "random",
  cell.group.column.impute = NULL,
  level
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunPCA_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_cell.group.column">cell.group.column</code></td>
<td>
<p>Character string. The name of the sample metadata column in which the variables will be used to label the cell groups on the PCA.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_cell.group.order">cell.group.order</code></td>
<td>
<p>Character string. The order of the variables under the sample metadata column specified in <code>cell.group.column</code> to appear in the PCA cell group legend.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_cell.group.colors">cell.group.colors</code></td>
<td>
<p>Character string. Vector of colors for the cell groups specified for PCA analysis using <code>cell.type.columns</code> and <code>cell.group.order</code>. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_sample.ids">sample.ids</code></td>
<td>
<p>Character strings. Specific cells to plot.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the splicing event or gene for the event or gene, respectively, to be included for analysis.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_features">features</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> or <code>gene_id</code> for analysis. Should match <code>tran_id</code> or <code>gene_id</code> column of <code>MarvelObject$ValidatedSpliceFeature</code> or <code>MarvelObject$GeneFeature</code> when <code>level</code> set to <code>"splicing"</code> or <code>"gene"</code>, respectively.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points on reduced dimension space.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points on reduced dimension space. Take any values between 0 to 1. The smaller the value, the more transparent the data points will be.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_point.stroke">point.stroke</code></td>
<td>
<p>Numeric value. The thickness of the outline of the data points. The larger the value, the thicker the outline of the data points.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Only applicable when <code>level</code> set to <code>"splicing"</code>.  Ensures imputed values for NA PSIs are reproducible.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_method.impute">method.impute</code></td>
<td>
<p>Character string. Only applicable when <code>level</code> set to <code>"splicing"</code>. Indicate the method for imputing missing PSI values (low coverage). <code>"random"</code> method randomly assigns any values between 0-1. <code>"population.mean"</code> method uses the mean PSI value for each cell population. Default option is <code>"population.mean"</code>.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_cell.group.column.impute">cell.group.column.impute</code></td>
<td>
<p>Character string. Only applicable when <code>method.impute</code> set to <code>"population.mean"</code>. The name of the sample metadata column in which the variables will be used to impute missing values.</p>
</td></tr>
<tr><td><code id="RunPCA_+3A_level">level</code></td>
<td>
<p>Character string. Indicate <code>"splicing"</code> or <code>"gene"</code> for splicing or gene expression analysis, respectively</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with new slots <code>MarvelObject$PCA$PSI$Results</code>, <code>MarvelObject$PCA$PSI$Plot</code>, and <code>MarvelObject$PCA$PSI$Plot.Elbow</code> or <code>MarvelObject$PCA$Exp$Results</code>, <code>MarvelObject$PCA$Exp$Plot</code>, and <code>MarvelObject$PCA$Exp$Plot.Elbow</code>, when <code>level</code> option specified as <code>"splicing"</code> or <code>"gene"</code>, respectively.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define splicing events for analysis
df &lt;- do.call(rbind.data.frame, marvel.demo$PSI)
tran_ids &lt;- df$tran_id

# PCA
marvel.demo &lt;- RunPCA(MarvelObject=marvel.demo,
                      sample.ids=marvel.demo$SplicePheno$sample.id,
                      cell.group.column="cell.type",
                      cell.group.order=c("iPSC", "Endoderm"),
                      cell.group.colors=NULL,
                      min.cells=5,
                      features=tran_ids,
                      level="splicing",
                      point.size=2
                      )

# Check outputs
head(marvel.demo$PCA$PSI$Results$ind$coord)
marvel.demo$PCA$PSI$Plot
</code></pre>

<hr>
<h2 id='RunPCA.Exp'>Principle component analysis for gene Data</h2><span id='topic+RunPCA.Exp'></span>

<h3>Description</h3>

<p>Performs principle component analysis using gene expression values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunPCA.Exp(
  MarvelObject,
  sample.ids = NULL,
  cell.group.column,
  cell.group.order = NULL,
  cell.group.colors = NULL,
  features,
  min.cells = 25,
  point.size = 0.5,
  point.alpha = 0.75,
  point.stroke = 0.1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunPCA.Exp_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_sample.ids">sample.ids</code></td>
<td>
<p>Character strings. Specific cells to plot.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_cell.group.column">cell.group.column</code></td>
<td>
<p>Character string. The name of the sample metadata column in which the variables will be used to label the cell groups on the PCA.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_cell.group.order">cell.group.order</code></td>
<td>
<p>Character string. The order of the variables under the sample metadata column specified in <code>cell.group.column</code> to appear in the PCA cell group legend.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_cell.group.colors">cell.group.colors</code></td>
<td>
<p>Character string. Vector of colors for the cell groups specified for PCA analysis using <code>cell.type.columns</code> and <code>cell.group.order</code>. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_features">features</code></td>
<td>
<p>Character string. Vector of <code>gene_id</code> for analysis. Should match <code>gene_id</code> column of <code>MarvelObject$GeneFeature</code>.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the gene to be included for analysis.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points on reduced dimension space.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points on reduced dimension space. Take any values between 0 to 1. The smaller the value, the more transparent the data points will be.</p>
</td></tr>
<tr><td><code id="RunPCA.Exp_+3A_point.stroke">point.stroke</code></td>
<td>
<p>Numeric value. The thickness of the outline of the data points. The larger the value, the thicker the outline of the data points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>MarvelObject$PCA$Exp$Results</code>, <code>MarvelObject$PCA$Exp$Plot</code>, and <code>MarvelObject$PCA$Exp$Plot.Elbow</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define genes for analysis
gene_ids &lt;- marvel.demo$Exp$gene_id

# PCA
marvel.demo &lt;- RunPCA.Exp(MarvelObject=marvel.demo,
                          sample.ids=marvel.demo$SplicePheno$sample.id,
                          cell.group.column="cell.type",
                          cell.group.order=c("iPSC", "Endoderm"),
                          min.cells=5,
                          features=gene_ids,
                          point.size=2
                          )

# Check outputs
head(marvel.demo$PCA$Exp$Results$ind$coord)
marvel.demo$PCA$Exp$Plot
</code></pre>

<hr>
<h2 id='RunPCA.PSI'>Principle component analysis for splicing data</h2><span id='topic+RunPCA.PSI'></span>

<h3>Description</h3>

<p>Performs principle component analysis using PSI values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RunPCA.PSI(
  MarvelObject,
  sample.ids = NULL,
  cell.group.column,
  cell.group.order,
  cell.group.colors = NULL,
  features,
  min.cells = 25,
  point.size = 0.5,
  point.alpha = 0.75,
  point.stroke = 0.1,
  seed = 1,
  method.impute = "random",
  cell.group.column.impute = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RunPCA.PSI_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_sample.ids">sample.ids</code></td>
<td>
<p>Character strings. Specific cells to plot.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_cell.group.column">cell.group.column</code></td>
<td>
<p>Character string. The name of the sample metadata column in which the variables will be used to label the cell groups on the PCA.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_cell.group.order">cell.group.order</code></td>
<td>
<p>Character string. The order of the variables under the sample metadata column specified in <code>cell.group.column</code> to appear in the PCA cell group legend.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_cell.group.colors">cell.group.colors</code></td>
<td>
<p>Character string. Vector of colors for the cell groups specified for PCA analysis using <code>cell.type.columns</code> and <code>cell.group.order</code>. If not specified, default <code>ggplot2</code> colors will be used.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_features">features</code></td>
<td>
<p>Character string. Vector of <code>tran_id</code> for analysis. Should match <code>tran_id</code> column of <code>MarvelObject$ValidatedSpliceFeature</code>.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_min.cells">min.cells</code></td>
<td>
<p>Numeric value. The minimum no. of cells expressing the splicing event to be included for analysis.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_point.size">point.size</code></td>
<td>
<p>Numeric value. Size of data points on reduced dimension space.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_point.alpha">point.alpha</code></td>
<td>
<p>Numeric value. Transparency of the data points on reduced dimension space. Take any values between 0 to 1. The smaller the value, the more transparent the data points will be.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_point.stroke">point.stroke</code></td>
<td>
<p>Numeric value. The thickness of the outline of the data points. The larger the value, the thicker the outline of the data points.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_seed">seed</code></td>
<td>
<p>Numeric value. Ensures imputed values for NA PSIs are reproducible.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_method.impute">method.impute</code></td>
<td>
<p>Character string. Indicate the method for imputing missing PSI values (low coverage). <code>"random"</code> method randomly assigns any values between 0-1. <code>"population.mean"</code> method uses the mean PSI value for each cell population. Default option is <code>"population.mean"</code>.</p>
</td></tr>
<tr><td><code id="RunPCA.PSI_+3A_cell.group.column.impute">cell.group.column.impute</code></td>
<td>
<p>Character string. Only applicable when <code>method.impute</code> set to <code>"population.mean"</code>. The name of the sample metadata column in which the variables will be used to impute missing values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing with new slots <code>MarvelObject$PCA$PSI$Results</code> and  <code>MarvelObject$PCA$PSI$Plot</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

# Define splicing events for analysis
df &lt;- do.call(rbind.data.frame, marvel.demo$PSI)
tran_ids &lt;- df$tran_id

# PCA
marvel.demo &lt;- RunPCA.PSI(MarvelObject=marvel.demo,
                          sample.ids=marvel.demo$SplicePheno$sample.id,
                          cell.group.column="cell.type",
                          cell.group.order=c("iPSC", "Endoderm"),
                          cell.group.colors=NULL,
                          min.cells=5,
                          features=tran_ids,
                          point.size=2
                          )

# Check outputs
head(marvel.demo$PCA$PSI$Results$ind$coord)
marvel.demo$PCA$PSI$Plot
</code></pre>

<hr>
<h2 id='SubsetCrypticA3SS'>Differential gene expression analysis for differentially spliced genes</h2><span id='topic+SubsetCrypticA3SS'></span>

<h3>Description</h3>

<p>Performs differential gene expression analysis between 2 groups of cells only on differentially spliced genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubsetCrypticA3SS(MarvelObject, method, distance.to.ss = c(1, 100))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SubsetCrypticA3SS_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>TransformExpValues</code> function.</p>
</td></tr>
<tr><td><code id="SubsetCrypticA3SS_+3A_method">method</code></td>
<td>
<p>Vector of character string(s). To include splicing events from these method(s) for differential splicing analysis.</p>
</td></tr>
<tr><td><code id="SubsetCrypticA3SS_+3A_distance.to.ss">distance.to.ss</code></td>
<td>
<p>Character string. Range of distances between A3SS and canonical splice site to consider A3SS to be cryptic. Default value <code>c(1, 100)</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 updated slot <code>MarvelObject$DE$PSI$Table</code> and new slot <code>MarvelObject$DE$PSI$A3SS.dist.to.ss</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- SubsetCrypticA3SS(MarvelObject=marvel.demo,
                                 method="ad",
                                 distance.to.ss=c(1,100)
                                 )

# Check output
head(marvel.demo$DE$PSI$Table[["ad"]])
</code></pre>

<hr>
<h2 id='SubsetSamples'>Subset samples (cells)</h2><span id='topic+SubsetSamples'></span>

<h3>Description</h3>

<p>Subsets specific samples (cells) from sample metadata.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SubsetSamples(MarvelObject, sample.ids)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SubsetSamples_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CreateMarvelObject</code> function.</p>
</td></tr>
<tr><td><code id="SubsetSamples_+3A_sample.ids">sample.ids</code></td>
<td>
<p>Vector of character strings. Sample IDs to subset.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slot <code>MarvelObject$SplicePheno</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

sample.ids &lt;- sample(marvel.demo$SplicePheno$sample.id, size=10)

marvel.demo &lt;- SubsetSamples(MarvelObject=marvel.demo,
                             sample.ids=sample.ids
                             )
</code></pre>

<hr>
<h2 id='TransformExpValues'>Transform gene expression Values</h2><span id='topic+TransformExpValues'></span>

<h3>Description</h3>

<p>Transforms gene expression values and censor lowly-expressing genes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TransformExpValues(
  MarvelObject,
  offset = 1,
  transformation = "log2",
  threshold.lower = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TransformExpValues_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>CheckAlignment</code> function.</p>
</td></tr>
<tr><td><code id="TransformExpValues_+3A_offset">offset</code></td>
<td>
<p>Numeric value. To indicate the value to add to the expression values before log transformation. The only option for this argument is 1.</p>
</td></tr>
<tr><td><code id="TransformExpValues_+3A_transformation">transformation</code></td>
<td>
<p>Character string. To indicate the type of transformation to use on the expression values after offsetting the values. The only option for this argument is <code>log2</code>.</p>
</td></tr>
<tr><td><code id="TransformExpValues_+3A_threshold.lower">threshold.lower</code></td>
<td>
<p>Numeric value. To indicate the value below which the expression values will be censored, i.e. re-coded as 0, after offsetting and transforming the values. The only option for this argument is <code>1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 with updated slot <code>MarvelObject$Exp</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marvel.demo &lt;- readRDS(system.file("extdata/data", "marvel.demo.rds", package="MARVEL"))

marvel.demo &lt;- TransformExpValues(MarvelObject=marvel.demo,
                                  offset=1,
                                  transformation="log2",
                                  threshold.lower=1
                                  )
</code></pre>

<hr>
<h2 id='ValidateSJ.10x'>Validate splice junctions</h2><span id='topic+ValidateSJ.10x'></span>

<h3>Description</h3>

<p>Retains splice junctions whose start and end belong to the same gene.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ValidateSJ.10x(MarvelObject, keep.novel.sj = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ValidateSJ.10x_+3A_marvelobject">MarvelObject</code></td>
<td>
<p>Marvel object. S3 object generated from <code>AnnotateSJ.10x</code> function.</p>
</td></tr>
<tr><td><code id="ValidateSJ.10x_+3A_keep.novel.sj">keep.novel.sj</code></td>
<td>
<p>Logical value. If set to <code>TRUE</code>, novel splice junctions will be retained for downstream analysis. Novel splice junctions are defined as splice junctions with one end reported in GTF while the other was not reported in GTF. Default value is <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class S3 containing the updated slots <code>MarvelObject$sj.metadata</code> and <code>MarvelObject$sj.count.matrix</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Load un-processed MARVEL object
marvel.demo.10x.raw &lt;- readRDS(system.file("extdata/data",
                               "marvel.demo.10x.raw.rds",
                               package="MARVEL")
                               )

# Annotate gene metadata
marvel.demo.10x &lt;- AnnotateGenes.10x(MarvelObject=marvel.demo.10x.raw)

# Annotate junction metadata
marvel.demo.10x &lt;- AnnotateSJ.10x(MarvelObject=marvel.demo.10x)

# Validate junctions
marvel.demo.10x &lt;- ValidateSJ.10x(MarvelObject=marvel.demo.10x)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
