<!DOCTYPE html><html><head><title>Help for package MatrixEQTL</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MatrixEQTL}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Covariates'>
<p>Artificial data for Matrix eQTL sample code: Covariates.</p></a></li>
<li><a href='#GE'>
<p>Artificial data for Matrix eQTL sample code: Gene expression.</p></a></li>
<li><a href='#geneloc'>
<p>Artificial data for Matrix eQTL sample code: Gene location file.</p></a></li>
<li><a href='#Matrix_eQTL_main'>
<p>Main function for fast eQTL analysis in MatrixEQTL package</p></a></li>
<li><a href='#MatrixEQTL_cis_code'><p>Sample code for cis/trans-eQTL analysis with Matrix eQTL</p></a></li>
<li><a href='#MatrixEQTL-package'>
<p>Matrix eQTL: Ultra Fast eQTL Analysis via Large Matrix Operations</p></a></li>
<li><a href='#modelANOVA'>
<p>Constant for <code>Matrix_eQTL_engine</code>.</p></a></li>
<li><a href='#modelLINEAR'>
<p>Constant for <code>Matrix_eQTL_engine</code>.</p></a></li>
<li><a href='#modelLINEAR_CROSS'>
<p>Constant for <code>Matrix_eQTL_engine</code>.</p></a></li>
<li><a href='#plot.MatrixEQTL'><p>Plot histogram or QQ-plot of all p-values</p></a></li>
<li><a href='#SlicedData-class'><p>Class <code>SlicedData</code> for storing large matrices</p></a></li>
<li><a href='#SNP'>
<p>Artificial data for Matrix eQTL sample code: Genotype.</p></a></li>
<li><a href='#snpsloc'>
<p>Artificial data for Matrix eQTL sample code: SNP location file.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Matrix eQTL: Ultra Fast eQTL Analysis via Large Matrix
Operations</td>
</tr>
<tr>
<td>Version:</td>
<td>2.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-12-04</td>
</tr>
<tr>
<td>Description:</td>
<td>Matrix eQTL is designed for fast eQTL analysis on large datasets.
        Matrix eQTL can test for association between genotype 
        and gene expression using linear regression 
        with either additive or ANOVA genotype effects.
        The models can include covariates to account for factors 
        as population stratification, gender, and clinical variables. 
        It also supports models with heteroscedastic and/or correlated errors,
        false discovery rate estimation and 
        separate treatment of local (cis) and distant (trans) eQTLs.
        For more details see Shabalin (2012) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbts163">doi:10.1093/bioinformatics/bts163</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/LGPL-3">LGPL-3</a></td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/andreyshabalin/MatrixEQTL/issues">https://github.com/andreyshabalin/MatrixEQTL/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.12.0), methods, utils, graphics, stats</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-12-05 00:23:41 UTC; Andrey</td>
</tr>
<tr>
<td>Author:</td>
<td>Andrey A Shabalin <a href="https://orcid.org/0000-0003-0309-6821"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Andrey A Shabalin &lt;andrey.shabalin@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-12-22 10:06:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='Covariates'>
Artificial data for Matrix eQTL sample code: Covariates.
</h2><span id='topic+Covariates'></span>

<h3>Description</h3>

<p>Artificial data set with 2 covariates across 15 samples. 
Columns of the file must match to those of the 
expression and genotype data sets.
</p>


<h3>Format</h3>


<table>
<tr>
 <td style="text-align: left;">
        id     </td><td style="text-align: left;"> Sam_01   </td><td style="text-align: left;"> Sam_02 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        gender </td><td style="text-align: left;"> 0        </td><td style="text-align: left;"> 0      </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        age    </td><td style="text-align: left;"> 40       </td><td style="text-align: left;"> 46     </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website: 
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>

<hr>
<h2 id='GE'>
Artificial data for Matrix eQTL sample code: Gene expression.
</h2><span id='topic+GE'></span>

<h3>Description</h3>

<p>Artificial data set with expression of 10 genes across 15 samples. 
Columns of the file must match to those of the 
genotype and covariates data sets.
</p>


<h3>Format</h3>


<table>
<tr>
 <td style="text-align: left;">
        geneid  </td><td style="text-align: left;"> Sam_01   </td><td style="text-align: left;"> Sam_02 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        Gene_01 </td><td style="text-align: left;"> 4.91     </td><td style="text-align: left;"> 4.63   </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        Gene_02 </td><td style="text-align: left;"> 13.78    </td><td style="text-align: left;"> 13.14  </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        ...  </td><td style="text-align: left;"> ...   </td><td style="text-align: left;"> ... </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website: 
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>

<hr>
<h2 id='geneloc'>
Artificial data for Matrix eQTL sample code: Gene location file.
</h2><span id='topic+geneloc'></span>

<h3>Description</h3>

<p>Artificial Gene location file for 10 genes. 
</p>

<table>
<tr>
 <td style="text-align: left;">
        geneid  </td><td style="text-align: left;"> chr      </td><td style="text-align: left;"> left   </td><td style="text-align: left;"> right  </td>
</tr>
<tr>
 <td style="text-align: left;">
        Gene_01 </td><td style="text-align: left;"> chr1     </td><td style="text-align: left;"> 721289 </td><td style="text-align: left;"> 731289 </td>
</tr>
<tr>
 <td style="text-align: left;">
        Gene_02 </td><td style="text-align: left;"> chr1     </td><td style="text-align: left;"> 752565 </td><td style="text-align: left;"> 762565 </td>
</tr>
<tr>
 <td style="text-align: left;">
        ...  </td><td style="text-align: left;"> ...   </td><td style="text-align: left;"> ... </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>



<h3>Format</h3>

<p>A data frame with 4 columns.
</p>

<dl>
<dt><code>geneid</code></dt><dd>
<p>A column with gene names. 
The order does not have to match the gene expression data set.
</p>
</dd>
<dt><code>chr</code></dt><dd>
<p>Chromosome number, i.e. chr1.
</p>
</dd>
<dt><code>left</code></dt><dd>
<p>Lower (smaller) coordinate of the gene begining/end.
</p>
</dd>
<dt><code>right</code></dt><dd>
<p>Upper (larger) coordinate of the gene begining/end.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>

<hr>
<h2 id='Matrix_eQTL_main'>
Main function for fast eQTL analysis in MatrixEQTL package
</h2><span id='topic+Matrix_eQTL_main'></span><span id='topic+Matrix_eQTL_engine'></span>

<h3>Description</h3>

<p><code>Matrix_eQTL_engine</code> function tests association of
every row of the <code>snps</code> dataset with every row of the
<code>gene</code> dataset using a linear regression model
defined by the <code>useModel</code> parameter (see below).
</p>
<p>The testing procedure accounts for extra 
covariates specified by the <code>cvrt</code> parameter.
</p>
<p>The <code>errorCovariance</code> parameter can be set to the 
error variance-covariance matrix to account 
for heteroskedastic and/or correlated errors.
</p>
<p>Associations significant at <code>pvOutputThreshold</code> 
(<code>pvOutputThreshold.cis</code>) levels are saved to 
<code>output_file_name</code> (<code>output_file_name.cis</code>), 
with corresponding estimates of effect size (slope coefficient),
test statistics, p-values, and q-values (false discovery rate).
</p>
<p>Matrix eQTL can perform separate analysis for
local (cis) and distant (trans) eQTLs.
For such analysis one has to set the cis-analysis specific
parameters <code>pvOutputThreshold.cis &gt; 0</code>,
<code>cisDist</code>, <code>snpspos</code> and 
genepos in the call of <code>Matrix_eQTL_main</code> function.
A gene-SNP pair is considered local if the 
distance between them is less or equal to <code>cisDist</code>.
The genomic location of genes and SNPs is defined by 
the data frames <code>snpspos</code> and genepos.
Depending on p-value thresholds <code>pvOutputThreshold</code> and
<code>pvOutputThreshold.cis</code> Matrix eQTL runs in 
one of three different modes:
</p>

<ul>
<li><p> Set <code>pvOutputThreshold &gt; 0</code> and 
<code>pvOutputThreshold.cis = 0</code> (or use <code>Matrix_eQTL_engine</code>)
to perform eQTL analysis without using gene/SNP locations.
Associations significant at the <code>pvOutputThreshold</code> level
are be recorded in <code>output_file_name</code>
and in the returned object.
</p>
</li>
<li><p> Set <code>pvOutputThreshold = 0</code> and
<code>pvOutputThreshold.cis &gt; 0</code> to perform eQTL analysis for
local gene-SNP pairs only. Local associations significant at
<code>pvOutputThreshold.cis</code> level will be recorded in
<code>output_file_name.cis</code> and in the returned object.
</p>
</li>
<li><p> Set <code>pvOutputThreshold &gt; 0</code> and
<code>pvOutputThreshold.cis &gt; 0</code> to perform eQTL analysis
with separate p-value thresholds for local and distant eQTLs.
Distant and local associations significant at corresponding
thresholds are recorded in <code>output_file_name</code> and
<code>output_file_name.cis</code> respectively and in the returned object.
In this case the false discovery rate is calculated
separately for these two sets of eQTLs.
</p>
</li></ul>

<p><code>Matrix_eQTL_engine</code> is a wrapper for <code>Matrix_eQTL_main</code>
for eQTL analysis without regard to gene/SNP location and provided
for compatibility with the previous versions of the package.
</p>
<p>The parameter <code>pvalue.hist</code> allows to record information sufficient
to create a histogram or QQ-plot of all the p-values
(see <code><a href="#topic+plot.MatrixEQTL">plot</a></code>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Matrix_eQTL_main(
        snps, 
        gene, 
        cvrt = SlicedData$new(), 
        output_file_name = "", 
        pvOutputThreshold = 1e-5,
        useModel = modelLINEAR, 
        errorCovariance = numeric(), 
        verbose = TRUE, 
        output_file_name.cis = "", 
        pvOutputThreshold.cis = 0,
        snpspos = NULL, 
        genepos = NULL,
        cisDist = 1e6,
        pvalue.hist = FALSE,
        min.pv.by.genesnp = FALSE,
        noFDRsaveMemory = FALSE)

Matrix_eQTL_engine(
        snps, 
        gene, 
        cvrt = SlicedData$new(), 
        output_file_name, 
        pvOutputThreshold = 1e-5, 
        useModel = modelLINEAR, 
        errorCovariance = numeric(), 
        verbose = TRUE,
        pvalue.hist = FALSE,
        min.pv.by.genesnp = FALSE,
        noFDRsaveMemory = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Matrix_eQTL_main_+3A_snps">snps</code></td>
<td>

<p><code><a href="#topic+SlicedData-class">SlicedData</a></code> object with genotype information. 
Can be real-valued for linear models and
must take at most 3 distinct values for ANOVA unless
the number of ANOVA categories is set to
a higher number (see <code>useModel</code> parameter).
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_gene">gene</code></td>
<td>

<p><code><a href="#topic+SlicedData-class">SlicedData</a></code> object with gene expression information.
Must have columns matching those of <code>snps</code>.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_cvrt">cvrt</code></td>
<td>

<p><code><a href="#topic+SlicedData-class">SlicedData</a></code> object with additional covariates. 
Can be an empty <code>SlicedData</code> object in case of no covariates.
The constant is always included in the model and
would cause an error if included in <code>cvrt</code>.
The order of columns must match those in <code>snps</code> and <code>gene</code>.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_output_file_name">output_file_name</code></td>
<td>

<p><code>character</code>, <code>connection</code>, or <code>NULL</code>.
If not <code>NULL</code>, significant associations are saved to
this file (all significant associations if <code>pvOutputThreshold=0</code>
or only distant if <code>pvOutputThreshold&gt;0</code>).
If the file with this name exists, it is overwritten.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_output_file_name.cis">output_file_name.cis</code></td>
<td>

<p><code>character</code>, <code>connection</code>, or <code>NULL</code>.
If not <code>NULL</code>, significant local associations
are saved to this file.
If the file with this name exists, it is overwritten.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_pvoutputthreshold">pvOutputThreshold</code></td>
<td>

<p><code>numeric</code>. Significance threshold for all/distant tests.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_pvoutputthreshold.cis">pvOutputThreshold.cis</code></td>
<td>
 
<p><code>numeric</code>. Same as <code>pvOutputThreshold</code>, but for local eQTLs.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_usemodel">useModel</code></td>
<td>

<p><code>integer</code>. 
Eigher <code>modelLINEAR</code>, 
<code>modelANOVA</code>, or
<code>modelLINEAR_CROSS</code>.
</p>

<ol>
<li><p> Set <code>useModel = <a href="#topic+modelLINEAR">modelLINEAR</a></code> to model
the effect of the genotype as additive linear and
test for its significance using t-statistic.
</p>
</li>
<li><p> Set <code>useModel = <a href="#topic+modelANOVA">modelANOVA</a></code> to treat genotype
as a categorical variables and use ANOVA model and
test for its significance using F-test.
The default number of ANOVA categories is 3.
Set otherwise like this:
<code>options(MatrixEQTL.ANOVA.categories=4)</code>.
</p>
</li>
<li><p> Set <code>useModel = <a href="#topic+modelLINEAR_CROSS">modelLINEAR_CROSS</a></code> to add
a new term to the model equal to the product of genotype and
the last covariate; the significance of this term is
then tested using t-statistic.
</p>
</li></ol>

</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_errorcovariance">errorCovariance</code></td>
<td>

<p><code>numeric</code>. The error covariance matrix.
Use <code>numeric()</code> for homoskedastic independent errors. 
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_verbose">verbose</code></td>
<td>

<p><code>logical</code>. Set to <code>TRUE</code> to display
more detailed report about the progress.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_snpspos">snpspos</code></td>
<td>

<p><code>data.frame</code> object with information about SNP locations,
must have 3 columns - SNP name, chromosome, and position, like this:
</p>

<table>
<tr>
 <td style="text-align: center;">
            snpid   </td><td style="text-align: center;"> chr    </td><td style="text-align: center;"> pos     </td>
</tr>
<tr>
 <td style="text-align: center;">
            Snp_01  </td><td style="text-align: center;"> 1      </td><td style="text-align: center;"> 721289  </td>
</tr>
<tr>
 <td style="text-align: center;">
            Snp_02  </td><td style="text-align: center;"> 1      </td><td style="text-align: center;"> 752565  </td>
</tr>
<tr>
 <td style="text-align: center;">
            ...  </td><td style="text-align: center;"> ... </td><td style="text-align: center;"> ...  </td>
</tr>
<tr>
 <td style="text-align: center;">
        </td>
</tr>

</table>

</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_genepos">genepos</code></td>
<td>

<p><code>data.frame</code> with information about transcript locations,
must have 4 columns - the name, chromosome, and 
positions of the left and right ends, like this:
</p>

<table>
<tr>
 <td style="text-align: center;">
            geneid  </td><td style="text-align: center;">  chr    </td><td style="text-align: center;"> left   </td><td style="text-align: center;"> right  </td>
</tr>
<tr>
 <td style="text-align: center;">
            Gene_01 </td><td style="text-align: center;">   1     </td><td style="text-align: center;"> 721289 </td><td style="text-align: center;"> 731289 </td>
</tr>
<tr>
 <td style="text-align: center;">
            Gene_02 </td><td style="text-align: center;">   1     </td><td style="text-align: center;"> 752565 </td><td style="text-align: center;"> 762565 </td>
</tr>
<tr>
 <td style="text-align: center;">
            ...  </td><td style="text-align: center;"> ...  </td><td style="text-align: center;"> ... </td><td style="text-align: center;"> ... </td>
</tr>
<tr>
 <td style="text-align: center;">
        </td>
</tr>

</table>

</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_cisdist">cisDist</code></td>
<td>

<p><code>numeric</code>. SNP-gene pairs within this distance are 
considered local. The distance is measured from the
nearest end of the gene. SNPs within a gene are always considered local.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_pvalue.hist">pvalue.hist</code></td>
<td>

<p><code>logical</code>, <code>numerical</code>, or <code>"qqplot"</code>.
Defines whether and how the distribution of p-values is recorded
in the returned object.
If <code>pvalue.hist = FALSE</code>, the information is not recorded
and the analysis is performed faster. 
Alternatively, set <code>pvalue.hist = "qqplot"</code> to record information
sufficient to create a QQ-plot of the p-values
(use <code><a href="#topic+plot.MatrixEQTL">plot</a></code> on the returned
object to create the plot).
To record information for a histogram set <code>pvalue.hist</code>
to the desired number of bins of equal size.
Finally, <code>pvalue.hist</code> can also be set to a
custom set of bin edges.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_min.pv.by.genesnp">min.pv.by.genesnp</code></td>
<td>

<p><code>logical</code>. Set <code>min.pv.by.genesnp = TRUE</code> to record
the minimum p-value for each SNP and each gene in the returned object.
The minimum p-values are recorded even if if they are above the
corresponding thresholds of <code>pvOutputThreshold</code> and
<code>pvOutputThreshold.cis</code>.
The analysis runs faster when the parameter is set to <code>FALSE</code>.
</p>
</td></tr>
<tr><td><code id="Matrix_eQTL_main_+3A_nofdrsavememory">noFDRsaveMemory</code></td>
<td>

<p><code>logical</code>. Set <code>noFDRsaveMemory = TRUE</code> to save
significant gene-SNP pairs directly to the output files,
reduce memory footprint and skip FDR calculation.
The eQTLs are not recorded in the returned object
if <code>noFDRsaveMemory = TRUE</code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that the columns of 
<code>gene</code>, <code>snps</code>, and <code>cvrt</code> must match.
If they do not match in the input files,
use <code>ColumnSubsample</code> method to subset and/or reorder them.
</p>


<h3>Value</h3>

<p>The detected eQTLs are saved in <code>output_file_name</code>
and/or <code>output_file_name.cis</code> if they are not <code>NULL</code>.
The method also returns a list with a summary of the performed analysis.
</p>
<table>
<tr><td><code>param</code></td>
<td>

<p>Keeps all input parameters and also records
the number of degrees of freedom for the full model.
</p>
</td></tr>
<tr><td><code>time.in.sec</code></td>
<td>

<p>Time difference between the start and 
the end of the analysis (in seconds).
</p>
</td></tr>
<tr><td><code>all</code></td>
<td>

<p>Information about all detected eQTLs.
</p>
</td></tr>
<tr><td><code>cis</code></td>
<td>

<p>Information about detected local eQTLs.
</p>
</td></tr>
<tr><td><code>trans</code></td>
<td>

<p>Information about detected distant eQTLs.
</p>
</td></tr>
</table>
<p>The elements <code>all</code>, <code>cis</code>, and <code>trans</code>
may contain the following components
</p>

<dl>
<dt><code>ntests</code></dt><dd>
<p>Total number of tests performed. This is used for FDR calculation.
</p>
</dd>
<dt><code>eqtls</code></dt><dd>
<p>Data frame with recorded significant associations.
Not available if <code>noFDRsaveMemory=FALSE</code>
</p>
</dd>
<dt><code>neqtls</code></dt><dd>
<p>Number of significant associations recorded.
</p>
</dd>
<dt><code>hist.bins</code></dt><dd>
<p>Histogram bins used for recording p-value distribution.
See <code>pvalue.hist</code> parameter.
</p>
</dd>
<dt><code>hist.counts</code></dt><dd>
<p>Number of p-value that fell in each histogram bin.
See <code>pvalue.hist</code> parameter.
</p>
</dd>
<dt><code>min.pv.snps</code></dt><dd>
<p>Vector with the best p-value for each SNP.
See <code>min.pv.by.genesnp</code> parameter.
</p>
</dd>
<dt><code>min.pv.gene</code></dt><dd>
<p>Vector with the best p-value for each gene.
See <code>min.pv.by.genesnp</code> parameter.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>The code below is the sample code for eQTL analysis NOT using gene/SNP locations.    
</p>
<p>See <code><a href="#topic+MatrixEQTL_cis_code">MatrixEQTL_cis_code</a></code> for sample code for
eQTL analysis that separates local and distant tests.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Matrix eQTL by Andrey A. Shabalin
# http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/
# 
# Be sure to use an up to date version of R and Matrix eQTL.

# source("Matrix_eQTL_R/Matrix_eQTL_engine.r");
library(MatrixEQTL)

## Location of the package with the data files.
base.dir = find.package('MatrixEQTL');

## Settings

# Linear model to use, modelANOVA, modelLINEAR, or modelLINEAR_CROSS
useModel = modelLINEAR; # modelANOVA, modelLINEAR, or modelLINEAR_CROSS

# Genotype file name
SNP_file_name = paste0(base.dir, "/data/SNP.txt");

# Gene expression file name
expression_file_name = paste0(base.dir, "/data/GE.txt");

# Covariates file name
# Set to character() for no covariates
covariates_file_name = paste0(base.dir, "/data/Covariates.txt");

# Output file name
output_file_name = tempfile();

# Only associations significant at this level will be saved
pvOutputThreshold = 1e-2;

# Error covariance matrix
# Set to numeric() for identity.
errorCovariance = numeric();
# errorCovariance = read.table("Sample_Data/errorCovariance.txt");


## Load genotype data

snps = SlicedData$new();
snps$fileDelimiter = "\t";      # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1;          # one row of column labels
snps$fileSkipColumns = 1;       # one column of row labels
snps$fileSliceSize = 2000;      # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);

## Load gene expression data

gene = SlicedData$new();
gene$fileDelimiter = "\t";      # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1;          # one row of column labels
gene$fileSkipColumns = 1;       # one column of row labels
gene$fileSliceSize = 2000;      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);

## Load covariates

cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t";      # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1;          # one row of column labels
cvrt$fileSkipColumns = 1;       # one column of row labels
if(length(covariates_file_name)&gt;0){
    cvrt$LoadFile(covariates_file_name);
}

## Run the analysis

me = Matrix_eQTL_engine(
    snps = snps,
    gene = gene,
    cvrt = cvrt,
    output_file_name = output_file_name,
    pvOutputThreshold = pvOutputThreshold,
    useModel = useModel, 
    errorCovariance = errorCovariance, 
    verbose = TRUE,
    pvalue.hist = TRUE,
    min.pv.by.genesnp = FALSE,
    noFDRsaveMemory = FALSE);
        
unlink(output_file_name);

## Results:

cat('Analysis done in: ', me$time.in.sec, ' seconds', '\n');
cat('Detected eQTLs:', '\n');
show(me$all$eqtls)

## Plot the histogram of all p-values

plot(me)
</code></pre>

<hr>
<h2 id='MatrixEQTL_cis_code'>Sample code for cis/trans-eQTL analysis with Matrix eQTL</h2><span id='topic+MatrixEQTL_cis_code'></span>

<h3>Description</h3>

<p>The following code is the best starting point for those 
who want to perform cis-/trans-eQTL analysis with Matrix eQTL.
</p>


<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and other sample code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Matrix eQTL by Andrey A. Shabalin
# http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/
# 
# Be sure to use an up to date version of R and Matrix eQTL.

# source("Matrix_eQTL_R/Matrix_eQTL_engine.r");
library(MatrixEQTL)

## Location of the package with the data files.
base.dir = find.package('MatrixEQTL');

## Settings

# Linear model to use, modelANOVA, modelLINEAR, or modelLINEAR_CROSS
useModel = modelLINEAR; # modelANOVA, modelLINEAR, or modelLINEAR_CROSS

# Genotype file name
SNP_file_name = paste0(base.dir, "/data/SNP.txt");
snps_location_file_name = paste0(base.dir, "/data/snpsloc.txt");

# Gene expression file name
expression_file_name = paste0(base.dir, "/data/GE.txt");
gene_location_file_name = paste0(base.dir, "/data/geneloc.txt");

# Covariates file name
# Set to character() for no covariates
covariates_file_name = paste0(base.dir, "/data/Covariates.txt");

# Output file name
output_file_name_cis = tempfile();
output_file_name_tra = tempfile();

# Only associations significant at this level will be saved
pvOutputThreshold_cis = 2e-2;
pvOutputThreshold_tra = 1e-2;

# Error covariance matrix
# Set to numeric() for identity.
errorCovariance = numeric();
# errorCovariance = read.table("Sample_Data/errorCovariance.txt");

# Distance for local gene-SNP pairs
cisDist = 1e6;

## Load genotype data

snps = SlicedData$new();
snps$fileDelimiter = "\t";      # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1;          # one row of column labels
snps$fileSkipColumns = 1;       # one column of row labels
snps$fileSliceSize = 2000;      # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);

## Load gene expression data

gene = SlicedData$new();
gene$fileDelimiter = "\t";      # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1;          # one row of column labels
gene$fileSkipColumns = 1;       # one column of row labels
gene$fileSliceSize = 2000;      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);

## Load covariates

cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t";      # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1;          # one row of column labels
cvrt$fileSkipColumns = 1;       # one column of row labels
if(length(covariates_file_name)&gt;0) {
  cvrt$LoadFile(covariates_file_name);
}

## Run the analysis
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE);
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE);

me = Matrix_eQTL_main(
    snps = snps, 
    gene = gene, 
    cvrt = cvrt,
    output_file_name      = output_file_name_tra,
    pvOutputThreshold     = pvOutputThreshold_tra,
    useModel = useModel, 
    errorCovariance = errorCovariance, 
    verbose = TRUE, 
    output_file_name.cis  = output_file_name_cis,
    pvOutputThreshold.cis = pvOutputThreshold_cis,
    snpspos = snpspos, 
    genepos = genepos,
    cisDist = cisDist,
    pvalue.hist = TRUE,
    min.pv.by.genesnp = FALSE,
    noFDRsaveMemory = FALSE);

unlink(output_file_name_tra);
unlink(output_file_name_cis);

## Results:

cat('Analysis done in: ', me$time.in.sec, ' seconds', '\n');
cat('Detected local eQTLs:', '\n');
show(me$cis$eqtls)
cat('Detected distant eQTLs:', '\n');
show(me$trans$eqtls)

## Make the histogram of local and distant p-values

plot(me)
</code></pre>

<hr>
<h2 id='MatrixEQTL-package'>
Matrix eQTL: Ultra Fast eQTL Analysis via Large Matrix Operations
</h2><span id='topic+MatrixEQTL-package'></span><span id='topic+MatrixEQTL'></span>

<h3>Description</h3>

<p>Matrix eQTL is designed for fast eQTL analysis of large datasets.
Matrix eQTL can test for association between genotype and
gene expression using linear regression 
with either additive or ANOVA (additive and dominant) genotype effects.
The models can include covariates to account for such factors as
population stratification, gender, clinical variables,
and surrogate variables.
Matrix eQTL also supports models with heteroscedastic
and/or correlated errors,
false discovery rate estimation and separate treatment of
local (cis) and distant (trans) eQTLs.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
        Package: </td><td style="text-align: left;"> MatrixEQTL</td>
</tr>
<tr>
 <td style="text-align: left;">
        Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
        License: </td><td style="text-align: left;"> LGPL-3 </td>
</tr>
<tr>
 <td style="text-align: left;">
        LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
        Depends: </td><td style="text-align: left;"> methods</td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>
<p>Maintainer: Andrey A Shabalin &lt;andrey.shabalin@gmail.com&gt;
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>

<hr>
<h2 id='modelANOVA'>
Constant for <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code>.
</h2><span id='topic+modelANOVA'></span>

<h3>Description</h3>

<p>Set parameter <code>useModel = modelANOVA</code> in the call of
<code><a href="#topic+Matrix_eQTL_main">Matrix_eQTL_main</a></code> to indicate that the genotype
should be treated as a categorical variable.
</p>


<h3>Note</h3>

<p>By default, the number of ANOVA categories is fixed to be 3.
</p>
<p>To set it to a different number (say, 4) use the following command:
</p>
<p><code>options(MatrixEQTL.ANOVA.categories=4)</code>
</p>
<p>To check the current settings run:
</p>
<p><code>getOption("MatrixEQTL.ANOVA.categories", 3);</code>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library("MatrixEQTL")    
            
# Number of columns (samples)
n = 100;

# Number of covariates
nc = 10;

# Generate the standard deviation of the noise
noise.std = 0.1 + rnorm(n)^2;

# Generate the covariates
cvrt.mat = 2 + matrix(rnorm(n*nc), ncol = nc);

# Generate the vectors with single genotype and expression variables
snps.mat = floor(runif(n, min = 0, max = 3));
gene.mat = 1 + (snps.mat==1) + cvrt.mat %*% rnorm(nc) + rnorm(n) * noise.std;

# Create 3 SlicedData objects for the analysis
snps1 = SlicedData$new( matrix( snps.mat, nrow = 1 ) );
gene1 = SlicedData$new( matrix( gene.mat, nrow = 1 ) );
cvrt1 = SlicedData$new( t(cvrt.mat) );

# name of temporary output file
filename = tempfile();

snps1
gene1

# Call the main analysis function
me = Matrix_eQTL_main(
    snps = snps1, 
    gene = gene1, 
    cvrt = cvrt1, 
    output_file_name = filename, 
    pvOutputThreshold = 1, 
    useModel = modelANOVA, 
    errorCovariance = diag(noise.std^2), 
    verbose = TRUE,
    pvalue.hist = FALSE );
# remove the output file
unlink( filename );

# Pull Matrix eQTL results - t-statistic and p-value

fstat = me$all$eqtls$statistic;
pvalue = me$all$eqtls$pvalue;
rez = c( Fstat = fstat, pvalue = pvalue)
# And compare to those from ANOVA in R
{
    cat("\n\n Matrix eQTL: \n"); 
    print(rez);
    cat("\n R anova(lm()) output: \n")
    lmodel = lm( gene.mat ~ cvrt.mat + factor(snps.mat), weights = 1/noise.std^2);
    lmout = anova(lmodel)[2, c("F value", "Pr(&gt;F)")];
    print( lmout )
}

# Results from Matrix eQTL and "lm" must agree
stopifnot(all.equal(lmout, rez, check.attributes = FALSE));
</code></pre>

<hr>
<h2 id='modelLINEAR'>
Constant for <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code>.
</h2><span id='topic+modelLINEAR'></span>

<h3>Description</h3>

<p>Set parameter <code>useModel = modelLINEAR</code> in the call of
<code><a href="#topic+Matrix_eQTL_main">Matrix_eQTL_main</a></code> to indicate that the effect of
genotype on expression should be assumed to be additive linear.
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('MatrixEQTL')    

# Number of columns (samples)
n = 100;

# Number of covariates
nc = 10;

# Generate the standard deviation of the noise
noise.std = 0.1 + rnorm(n)^2;

# Generate the covariates
cvrt.mat = 2 + matrix(rnorm(n*nc), ncol = nc);

# Generate the vectors with genotype and expression variables
snps.mat = cvrt.mat %*% rnorm(nc) + rnorm(n);
gene.mat = cvrt.mat %*% rnorm(nc) + rnorm(n) * noise.std + 0.5 * snps.mat + 1;

# Create 3 SlicedData objects for the analysis
snps1 = SlicedData$new( matrix( snps.mat, nrow = 1 ) );
gene1 = SlicedData$new( matrix( gene.mat, nrow = 1 ) );
cvrt1 = SlicedData$new( t(cvrt.mat) );

# name of temporary output file
filename = tempfile();

# Call the main analysis function
me = Matrix_eQTL_main(
    snps = snps1, 
    gene = gene1, 
    cvrt = cvrt1, 
    output_file_name = filename, 
    pvOutputThreshold = 1, 
    useModel = modelLINEAR, 
    errorCovariance = diag(noise.std^2), 
    verbose = TRUE,
    pvalue.hist = FALSE );
# remove the output file
unlink( filename );

# Pull Matrix eQTL results - t-statistic and p-value
beta = me$all$eqtls$beta;
tstat = me$all$eqtls$statistic;
pvalue = me$all$eqtls$pvalue;
rez = c(beta = beta, tstat = tstat, pvalue = pvalue)
# And compare to those from the linear regression in R
{
    cat('\n\n Matrix eQTL: \n'); 
    print(rez);
    cat('\n R summary(lm()) output: \n');
    lmodel = lm( gene.mat ~ snps.mat + cvrt.mat, weights = 1/noise.std^2 );
    lmout = summary(lmodel)$coefficients[2, c("Estimate", "t value", "Pr(&gt;|t|)")];
    print( lmout )
}

# Results from Matrix eQTL and 'lm' must agree
stopifnot(all.equal(lmout, rez, check.attributes = FALSE));
</code></pre>

<hr>
<h2 id='modelLINEAR_CROSS'>
Constant for <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code>.
</h2><span id='topic+modelLINEAR_CROSS'></span>

<h3>Description</h3>

<p>Set parameter <code>useModel = modelLINEAR_CROSS</code> in the call of
<code><a href="#topic+Matrix_eQTL_main">Matrix_eQTL_main</a></code> to indicate that Matrix eQTL should
include the interaction of SNP and the
last covariate in the model and test for its significance.
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library('MatrixEQTL')    

# Number of columns (samples)
n = 25;

# Number of covariates
nc = 10;

# Generate the standard deviation of the noise
noise.std = 0.1 + rnorm(n)^2;

# Generate the covariates
cvrt.mat = 2 + matrix(rnorm(n*nc), ncol = nc);

# Generate the vectors with single genotype and expression variables
snps.mat = cvrt.mat %*% rnorm(nc) + rnorm(n);
gene.mat = cvrt.mat %*% rnorm(nc) + rnorm(n) * noise.std + 
            1 + 0.5 * snps.mat + snps.mat * cvrt.mat[,nc];

# Create 3 SlicedData objects for the analysis
snps1 = SlicedData$new( matrix( snps.mat, nrow = 1 ) );
gene1 = SlicedData$new( matrix( gene.mat, nrow = 1 ) );
cvrt1 = SlicedData$new( t(cvrt.mat) );

# name of temporary output file
filename = tempfile();

# Call the main analysis function
me = Matrix_eQTL_main(
    snps = snps1, 
    gene = gene1, 
    cvrt = cvrt1, 
    output_file_name = filename, 
    pvOutputThreshold = 1, 
    useModel = modelLINEAR_CROSS, 
    errorCovariance = diag(noise.std^2), 
    verbose = TRUE,
    pvalue.hist = FALSE );
# remove the output file
unlink( filename );

# Pull Matrix eQTL results - t-statistic and p-value
beta = me$all$eqtls$beta;
tstat = me$all$eqtls$statistic;
pvalue = me$all$eqtls$pvalue;
rez = c(beta = beta, tstat = tstat, pvalue = pvalue)
# And compare to those from the linear regression in R
{
    cat('\n\n Matrix eQTL: \n'); 
    print(rez);
    cat('\n R summary(lm()) output: \n')
    lmodel = lm( gene.mat ~ snps.mat + cvrt.mat + snps.mat*cvrt.mat[,nc], 
                    weights = 1/noise.std^2);
    lmout = tail(summary(lmodel)$coefficients,1)[, c("Estimate", "t value", "Pr(&gt;|t|)")];
    print( tail(lmout) );
}

# Results from Matrix eQTL and 'lm' must agree
stopifnot(all.equal(lmout, rez, check.attributes = FALSE));
</code></pre>

<hr>
<h2 id='plot.MatrixEQTL'>Plot histogram or QQ-plot of all p-values
</h2><span id='topic+plot.MatrixEQTL'></span>

<h3>Description</h3>

<p>This method plots a histogram or QQ-plot of p-values
for all tests performed by <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'MatrixEQTL'
plot(
    x,
    cex = 0.5,
    pch = 19,
    xlim = NULL,
    ylim = NULL,
    main = NULL,
    ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.MatrixEQTL_+3A_x">x</code></td>
<td>

<p>An object returned by <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.MatrixEQTL_+3A_cex">cex</code></td>
<td>

<p>A numerical value giving the amount by which
plotting text and symbols should be magnified relative to the default.
</p>
</td></tr>
<tr><td><code id="plot.MatrixEQTL_+3A_pch">pch</code></td>
<td>

<p>Plotting &quot;character&quot;, i.e., symbol to use.
See <code><a href="graphics.html#topic+points">points</a></code>.
</p>
</td></tr>
<tr><td><code id="plot.MatrixEQTL_+3A_xlim">xlim</code></td>
<td>

<p>Set the range of the horisontal axis.
</p>
</td></tr>
<tr><td><code id="plot.MatrixEQTL_+3A_ylim">ylim</code></td>
<td>

<p>Set the range of the vertical axis.
</p>
</td></tr>
<tr><td><code id="plot.MatrixEQTL_+3A_main">main</code></td>
<td>

<p>Plot title.
</p>
</td></tr>
<tr><td><code id="plot.MatrixEQTL_+3A_...">...</code></td>
<td>

<p>Further graphical parameters passed to 
<code><a href="graphics.html#topic+lines">lines</a></code> and <code><a href="graphics.html#topic+points">points</a></code>.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The plot type (histogram vs. QQ-plot) is determined by the
<code>pvalue.hist</code> parameter in the call of
<code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> function.
</p>


<h3>Value</h3>

<p>The method does not return any value.
</p>


<h3>Note</h3>

<p>The sample code below produces figures like these:
</p>
<p>Histogram: <img src="../help/figures/histogram.png" alt="histogram.png" />
</p>
<p>QQ-plot: <img src="../help/figures/QQplot.png" alt="QQplot.png" />
</p>


<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(MatrixEQTL)

# Number of samples
n = 100 

# Number of variables
ngs = 2000

# Common signal in all variables
pop = 0.2*rnorm(n)

# data matrices
snps.mat = matrix(rnorm(n*ngs), ncol = ngs) + pop
gene.mat = matrix(rnorm(n*ngs), ncol = ngs) + pop + snps.mat*((1:ngs)/ngs)^9/2

# data objects for Matrix eQTL engine
snps1 = SlicedData$new( t( snps.mat ) )
gene1 = SlicedData$new( t( gene.mat ) )
cvrt1 = SlicedData$new( )
rm(snps.mat, gene.mat)

# Slice data in blocks of 500 variables
snps1$ResliceCombined(500)
gene1$ResliceCombined(500)

# Produce no output files
filename = NULL # tempfile()

# Perform analysis recording information for a histogram
meh = Matrix_eQTL_engine(
    snps = snps1, 
    gene = gene1, 
    cvrt = cvrt1, 
    output_file_name = filename, 
    pvOutputThreshold = 1e-100, 
    useModel = modelLINEAR, 
    errorCovariance = numeric(), 
    verbose = TRUE,
    pvalue.hist = 100)
    
plot(meh, col="grey")

# Perform analysis recording information for a QQ-plot
meq = Matrix_eQTL_engine(
    snps = snps1, 
    gene = gene1, 
    cvrt = cvrt1, 
    output_file_name = filename,
    pvOutputThreshold = 1e-6, 
    useModel = modelLINEAR, 
    errorCovariance = numeric(), 
    verbose = TRUE,
    pvalue.hist = "qqplot")
    
plot(meq)
</code></pre>

<hr>
<h2 id='SlicedData-class'>Class <code>SlicedData</code> for storing large matrices</h2><span id='topic+SlicedData'></span><span id='topic+SlicedData-class'></span><span id='topic++5B+5B+2CSlicedData-method'></span><span id='topic++5B+5B+3C-+2CSlicedData-method'></span><span id='topic+colnames+2CSlicedData-method'></span><span id='topic+colnames+3C-+2CSlicedData-method'></span><span id='topic+dim+2CSlicedData-method'></span><span id='topic+length+2CSlicedData-method'></span><span id='topic+ncol+2CSlicedData-method'></span><span id='topic+NCOL+2CSlicedData-method'></span><span id='topic+nrow+2CSlicedData-method'></span><span id='topic+NROW+2CSlicedData-method'></span><span id='topic+rownames+2CSlicedData-method'></span><span id='topic+rownames+3C-+2CSlicedData-method'></span><span id='topic+show+2CSlicedData-method'></span><span id='topic+as.matrix+2CSlicedData-method'></span><span id='topic+rowMeans+2CSlicedData-method'></span><span id='topic+rowSums+2CSlicedData-method'></span><span id='topic+colMeans+2CSlicedData-method'></span><span id='topic+colSums+2CSlicedData-method'></span><span id='topic+summary.SlicedData'></span>

<h3>Description</h3>

<p>This class is created for fast and memory efficient manipulations with
large datasets presented in matrix form.
It is used to load, store, and manipulate large datasets,
e.g. genotype and gene expression matrices.
When a dataset is loaded,
it is sliced in blocks of 1,000 rows (default size).
This allows imputing, standardizing, and performing other operations
with the data with minimal memory overhead.
</p>


<h3>Usage</h3>

<pre><code class='language-R'># x[[i]] indexing allows easy access to individual slices.
# It is equivalent to x$GetSlice(i) and x$SetSlice(i,value)
## S4 method for signature 'SlicedData'
x[[i]]
## S4 replacement method for signature 'SlicedData'
x[[i]] &lt;- value

# The following commands work as if x was a simple matrix object
## S4 method for signature 'SlicedData'
nrow(x)
## S4 method for signature 'SlicedData'
ncol(x)
## S4 method for signature 'SlicedData'
dim(x)
## S4 method for signature 'SlicedData'
rownames(x)
## S4 method for signature 'SlicedData'
colnames(x)
## S4 replacement method for signature 'SlicedData'
rownames(x) &lt;- value
## S4 replacement method for signature 'SlicedData'
colnames(x) &lt;- value

# SlicedData object can be easily transformed into a matrix
# preserving row and column names
## S4 method for signature 'SlicedData'
as.matrix(x)

# length(x) can be used in place of x$nSlices()
# to get the number of slices in the object
## S4 method for signature 'SlicedData'
length(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SlicedData-class_+3A_x">x</code></td>
<td>

<p><code><a href="#topic+SlicedData-class">SlicedData</a></code> object.
</p>
</td></tr>
<tr><td><code id="SlicedData-class_+3A_i">i</code></td>
<td>

<p>Number of a slice.
</p>
</td></tr>
<tr><td><code id="SlicedData-class_+3A_value">value</code></td>
<td>

<p>New content for the slice / new row or column names.
</p>
</td></tr>
</table>


<h3>Extends</h3>

<p><code>SlicedData</code> is a reference classes (<code><a href="methods.html#topic+envRefClass-class">envRefClass</a></code>).
Its methods can change the values of the fields of the class.
</p>


<h3>Fields</h3>


<dl>
<dt><code>dataEnv</code>:</dt><dd>
<p><code>environment</code>. Stores the slices of the data matrix.
The slices should be accessed via
<code>getSlice()</code> and <code>setSlice()</code> methods.
</p>
</dd>
<dt><code>nSlices1</code>:</dt><dd>
<p><code>numeric</code>. Number of slices. For internal use.
The value should be access via <code>nSlices()</code> method.
</p>
</dd>
<dt><code>rowNameSlices</code>:</dt><dd>
<p><code>list</code>. Slices of row names.
</p>
</dd>
<dt><code>columnNames</code>:</dt><dd>
<p><code>character</code>. Column names.
</p>
</dd>
<dt><code>fileDelimiter</code>:</dt><dd>
<p><code>character</code>. Delimiter separating values in the input file.
</p>
</dd>
<dt><code>fileSkipColumns</code>:</dt><dd>
<p><code>numeric</code>. Number of columns with row labels in the input file.
</p>
</dd>
<dt><code>fileSkipRows</code>:</dt><dd>
<p><code>numeric</code>. Number of rows with column labels in the input file.
</p>
</dd>
<dt><code>fileSliceSize</code>:</dt><dd>
<p><code>numeric</code>. Maximum number of rows in a slice.
</p>
</dd>
<dt><code>fileOmitCharacters</code>:</dt><dd>
<p><code>character</code>. 
Missing value (NaN) representation in the input file.
</p>
</dd>
</dl>



<h3>Methods</h3>


<dl>
<dt><code>initialize(mat)</code>:</dt><dd>
<p>Create the object from a matrix.
</p>
</dd>
<dt><code>nSlices()</code>:</dt><dd>
<p>Returns the number of slices.
</p>
</dd>
<dt><code>nCols()</code>:</dt><dd>
<p>Returns the number of columns in the matrix.
</p>
</dd>
<dt><code>nRows()</code>:</dt><dd>
<p>Returns the number of rows in the matrix.
</p>
</dd>
<dt><code>Clear()</code>:</dt><dd>
<p>Clears the object.
Removes the data slices and row and column names.
</p>
</dd>
<dt><code>Clone()</code>:</dt><dd>
<p>Makes a copy of the object.
Changes to the copy do not affect the source object.
</p>
</dd>
<dt><code>CreateFromMatrix(mat)</code>:</dt><dd>
<p>Creates <code>SlicedData</code> object from a <code><a href="base.html#topic+matrix">matrix</a></code>.
</p>
</dd>
<dt><code>LoadFile(filename, skipRows = NULL, skipColumns = NULL,</code> <br /> 
<code>sliceSize = NULL, omitCharacters = NULL,
            delimiter = NULL, rowNamesColumn = 1)</code>:</dt><dd>
<p>Loads data matrix from a file.
<code>filename</code> should be a character string.
The remaining parameters specify the file format and have the same
meaning as <code>file*</code> fields.
Additional <code>rowNamesColumn</code> parameter specifies which of the
columns of row labels to use as row names.
</p>
</dd>
<dt><code>SaveFile(filename)</code>:</dt><dd>
<p>Saves the data to a file.
<code>filename</code> should be a character string.
</p>
</dd>
<dt><code>getSlice(sl)</code>:</dt><dd>
<p>Retrieves <code>sl</code>-th slice of the matrix.
</p>
</dd>
<dt><code>setSlice(sl, value)</code>:</dt><dd>
<p>Set <code>sl</code>-th slice of the matrix.
</p>
</dd>
<dt><code>ColumnSubsample(subset)</code>:</dt><dd>
<p>Reorders/subsets the columns according to <code>subset</code>. <br />
Acts as <code>M = M[ ,subset]</code> for a matrix <code>M</code>.
</p>
</dd>
<dt><code>RowReorder(ordr)</code>:</dt><dd>
<p>Reorders rows according to <code>ordr</code>. <br />
Acts as <code>M = M[ordr, ]</code> for a matrix <code>M</code>.
</p>
</dd>
<dt><code>RowMatrixMultiply(multiplier)</code>:</dt><dd>
<p>Multiply each row by the <code>multiplier</code>. <br />
Acts as <code>M = M %*% multiplier</code> for a matrix <code>M</code>.
</p>
</dd>
<dt><code>CombineInOneSlice()</code>:</dt><dd>
<p>Combines all slices into one.
The whole matrix can then be obtained via <code>$getSlice(1)</code>.
</p>
</dd>
<dt><code>IsCombined()</code>:</dt><dd>
<p>Returns <code>TRUE</code> if the number of slices is 1 or 0.
</p>
</dd>
<dt><code>ResliceCombined(sliceSize = -1)</code>:</dt><dd>
<p>Cuts the data into slices of <code>sliceSize</code> rows.
If <code>sliceSize</code> is not defined,
the value of <code>fileSliceSize</code> field is used.
</p>
</dd>
<dt><code>GetAllRowNames()</code>:</dt><dd>
<p>Returns all row names in one vector.
</p>
</dd>    
<dt><code>RowStandardizeCentered()</code>:</dt><dd>
<p>Set the mean of each row to zero and the sum of squares to one.
</p>
</dd>
<dt><code>SetNanRowMean()</code>:</dt><dd>
<p>Impute rows with row mean.
Rows full of NaN values are imputed with zeros.
</p>
</dd>
<dt><code>RowRemoveZeroEps()</code>:</dt><dd>
<p>Removes rows of zeros and those that are nearly zero.
</p>
</dd>
<dt><code>FindRow(rowname)</code>:</dt><dd>
<p>Finds row by name.
Returns a pair of slice number an row number within the slice.
If no row is found, the function returns <code>NULL</code>.
</p>
</dd>
<dt><code>rowMeans(x, na.rm = FALSE, dims = 1L)</code>:</dt><dd>
<p>Returns a vector of row means.
Works as <a href="base.html#topic+colSums">rowMeans</a> but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
<dt><code>rowSums(x, na.rm = FALSE, dims = 1L)</code>:</dt><dd>
<p>Returns a vector of row sums.
Works as <a href="base.html#topic+colSums">rowSums</a> but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
<dt><code>colMeans(x, na.rm = FALSE, dims = 1L)</code>:</dt><dd>
<p>Returns a vector of column means.
Works as <a href="base.html#topic+colSums">colMeans</a> but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
<dt><code>colSums(x, na.rm = FALSE, dims = 1L)</code>:</dt><dd>
<p>Returns a vector of column sums.
Works as <a href="base.html#topic+colSums">colSums</a> but
requires <code>dims</code> to be equal to <code>1L</code>.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>This class is used to load data for eQTL analysis by
<code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create a SlicedData variable
sd = SlicedData$new()    

# Show the details of the empty object
show(sd)    

# Create a matrix of values and assign to sd
mat = matrix(1:12, 3, 4)    
rownames(mat) = c("row1", "row2", "row3")    
colnames(mat) = c("col1", "col2", "col3", "col4")    
sd$CreateFromMatrix( mat )    

# Show the detail of the object (one slice)
show(sd)    

# Slice it in pieces of 2 rows
sd$ResliceCombined(sliceSize = 2L)    

# Show the number of slices (equivalent function calls)
sd$nSlices()
length(sd)

# Is it all in one slice? (No)
sd$IsCombined()

# Show the column names (equivalent function calls)
sd$columnNames
colnames(sd)

# Show row name slices
sd$rowNameSlices

# Show all row names (equivalent function calls)
sd$GetAllRowNames()
rownames(sd)

# Print the second slice
print(sd[[2]])

# Reorder and subset columns
sd$ColumnSubsample( c(1,3,4) )    

# Reorder and subset rows
sd$RowReorder( c(3,1) )    

# Show the detail of the object (one slice again)
show(sd)    

# Is it all in one slice? (Yes)
sd$IsCombined()

# Find the row with name "row1" (it is second in the first slice)
sd$FindRow("row1")    

</code></pre>

<hr>
<h2 id='SNP'>
Artificial data for Matrix eQTL sample code: Genotype.
</h2><span id='topic+SNP'></span>

<h3>Description</h3>

<p>Artificial data set with genotype information for 
15 markers across 15 samples.
Columns of the file must match to those of the 
gene expression and covariates data sets.
</p>


<h3>Format</h3>


<table>
<tr>
 <td style="text-align: left;">
        snpnid  </td><td style="text-align: left;"> Sam_01   </td><td style="text-align: left;"> Sam_02 </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        Snp_01  </td><td style="text-align: left;"> 2        </td><td style="text-align: left;"> 0      </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        Snp_02  </td><td style="text-align: left;"> 0        </td><td style="text-align: left;"> 1      </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
        ...  </td><td style="text-align: left;"> ...   </td><td style="text-align: left;"> ... </td><td style="text-align: left;"> ... </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>

<hr>
<h2 id='snpsloc'>
Artificial data for Matrix eQTL sample code: SNP location file.
</h2><span id='topic+snpsloc'></span>

<h3>Description</h3>

<p>Artificial SNP location file for 15 markers. 
</p>

<table>
<tr>
 <td style="text-align: left;">
        snpid   </td><td style="text-align: left;"> chr      </td><td style="text-align: left;"> pos     </td>
</tr>
<tr>
 <td style="text-align: left;">
        Snp_01  </td><td style="text-align: left;"> chr1     </td><td style="text-align: left;"> 721289  </td>
</tr>
<tr>
 <td style="text-align: left;">
        Snp_02  </td><td style="text-align: left;"> chr1     </td><td style="text-align: left;"> 752565  </td>
</tr>
<tr>
 <td style="text-align: left;">
        ...  </td><td style="text-align: left;"> ...   </td><td style="text-align: left;"> ...  </td>
</tr>
<tr>
 <td style="text-align: left;">
    </td>
</tr>

</table>



<h3>Format</h3>

<p>A data frame with 3 columns.
</p>

<dl>
<dt><code>snpid</code></dt><dd>
<p>A column with SNP names. 
The order does not have to match the genotype data set.
</p>
</dd>
<dt><code>chr</code></dt><dd>
<p>Chromosome number, i.e. chr1.
</p>
</dd>
<dt><code>pos</code></dt><dd>
<p>Coordinate of the SNP.
</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Andrey A Shabalin <a href="mailto:andrey.shabalin@gmail.com">andrey.shabalin@gmail.com</a>
</p>


<h3>References</h3>

<p>The package website:
<a href="http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/">http://www.bios.unc.edu/research/genomic_software/Matrix_eQTL/</a>
</p>


<h3>See Also</h3>

<p>See <code><a href="#topic+Matrix_eQTL_engine">Matrix_eQTL_engine</a></code> for reference and sample code.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
