<!DOCTYPE html><html lang="en"><head><title>Help for package maxaltall</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {maxaltall}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#max_altall'>
<p>'FASTA' ML and ‘altall’ Sequences from IQ-TREE .state Files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>'FASTA' ML and ‘altall’ Sequences from IQ-TREE .state Files</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Philip D. Kiser &lt;pkiser@uci.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Takes a .state file generated by IQ-TREE as an input and, for each ancestral node present in the file, generates a FASTA-formatted maximum likelihood (ML) sequence as well as an ‘AltAll’ sequence in which uncertain sites, determined by the two parameters thres_1 and thres_2, have the maximum likelihood state swapped with the next most likely state as described in Geeta N. Eick, Jamie T. Bridgham, Douglas P. Anderson, Michael J. Harms, and Joseph W. Thornton (2017), "Robustness of Reconstructed Ancestral Protein Functions to Statistical Uncertainty" &lt;<a href="https://doi.org/10.1093%2Fmolbev%2Fmsw223">doi:10.1093/molbev/msw223</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3)</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, dplyr, magrittr, tidyr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-24 13:54:18 UTC; Philip Kiser</td>
</tr>
<tr>
<td>Author:</td>
<td>Alec S. Chu [aut],
  Philip D. Kiser [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-25 09:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='max_altall'>
'FASTA' ML and ‘altall’ Sequences from IQ-TREE .state Files
</h2><span id='topic+max_altall'></span>

<h3>Description</h3>

<p>Takes a .state file generated by IQ-TREE as an input and, for each ancestral node present in the file, generates a FASTA-formatted maximum likelihood (ML) sequence as well as an ‘AltAll’ sequence in which uncertain sites, determined by the two parameters thres_1 and thres_2, have the maximum likelihood state swapped with the next most likely state as described in Geeta N. Eick, Jamie T. Bridgham, Douglas P. Anderson, Michael J. Harms, and Joseph W. Thornton (2017), &quot;Robustness of Reconstructed Ancestral Protein Functions to Statistical Uncertainty&quot; &lt;doi:10.1093/molbev/msw223&gt;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>max_altall(file, type, thres_1, thres_2, export, export_dir)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="max_altall_+3A_file">file</code></td>
<td>

<p>This argument specifies the IQ-TREE .state file containing the data the function will use.
</p>
</td></tr>
<tr><td><code id="max_altall_+3A_type">type</code></td>
<td>

<p>This argument is either ‘aa’ for amino acid states or ‘nuc’ for nucleotide states.
</p>
</td></tr>
<tr><td><code id="max_altall_+3A_thres_1">thres_1</code></td>
<td>

<p>This argument specifies the probability threshold at which the most probable state will be considered ambiguous enough for possible substitution (as determined by thres_2) by the second most probable state. Permissible values obey the following inequalities:
</p>
<p>0.05 &lt;= thres_1 &lt;= 1 for amino acid states
0.25 &lt;= thres_1 &lt;= 1 for nucleotide states
</p>
<p>The default value is 0.8 (see Eick et al).
</p>
</td></tr>
<tr><td><code id="max_altall_+3A_thres_2">thres_2</code></td>
<td>

<p>This argument specifies the probability threshold at which the second most probable state will be substituted for the most probable state. Permissible values obey the following inequalities:
</p>
<p>thres_2 &lt;= thres_1, thres_2 &lt;= (1 - thres_1), thres_2 &gt;= (1 - thres_1 )/19) for amino acid states
thres_2 &lt;= thres_1, thres_2 &lt;= (1 - thres_1), thres_2 &gt;= (1 - thres_1 )/3) for nucleotide states
</p>
<p>The default value is 0.2 (see Eick et al).
</p>
</td></tr>
<tr><td><code id="max_altall_+3A_export">export</code></td>
<td>

<p>This argument is either &quot;TRUE&quot; or &quot;FALSE.&quot; A &quot;TRUE&quot; input will cause the sequences to be saved to a new text file called 'node_sequences_all.txt' in a directory specified by the export_dir argument described below. A &quot;FALSE&quot; input will cause the sequences to be printed to the R console. The default value is &quot;FALSE&quot;
</p>
</td></tr>
<tr><td><code id="max_altall_+3A_export_dir">export_dir</code></td>
<td>

<p>This argument specifies the directory where the new text file containing the deposited FASTA sequences will be located. The default directory is tempdir().
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An IQ-TREE .state file contains posterior probabilities for each possible character state at each position of an amino acid or nucleotide alignment and for each ancestral node of the phylogenic tree used for the ancestral state reconstruction calculation. The purpose of this R script is to extract the maximum likelihood sequence and a user-defined ‘AltAll’ sequence and output the sequences in FASTA format for facile downstream use. The ‘AltAll’ sequence concept is described by Eick et al.
</p>


<h3>Value</h3>

<p>Returns a new text file called node_sequences_all.txt with the ML and altall sequences for all nodes present in the input .state file.
</p>


<h3>Author(s)</h3>

<p>Alec Chu and Philip D. Kiser at the University of California, Irvine
</p>


<h3>References</h3>

<p>Paper describing IQ-TREE
1) B.Q. Minh, H.A. Schmidt, O. Chernomor, D. Schrempf, M.D. Woodhams, A. von Haeseler, R. Lanfear (2020) IQ-TREE 2: New models and efficient methods for phylogenetic inference in the genomic era. Mol. Biol. Evol., 37:1530-1534.
https://doi.org/10.1093/molbev/msaa015
</p>
<p>Online IQ-TREE command reference
2) http://www.iqtree.org/doc/Command-Reference
</p>
<p>Paper describing the concept of an AltAll sequence
3) Geeta N. Eick, Jamie T. Bridgham, Douglas P. Anderson, Michael J. Harms, Joseph W. Thornton, Robustness of Reconstructed Ancestral Protein Functions to Statistical Uncertainty, Molecular Biology and Evolution, Volume 34, Issue 2, February 2017, Pages 247–261, https://doi.org/10.1093/molbev/msw223
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

example_aa.state &lt;- system.file("extdata", "example_aa.state", package = "maxaltall")
max_altall(example_aa.state, "aa", 0.8, 0.2, "TRUE", tempdir())


example_nuc.state &lt;- system.file("extdata", "example_nuc.state", package = "maxaltall")
max_altall(example_nuc.state, "nuc", 0.8, 0.2, "TRUE", tempdir())
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
