<!DOCTYPE html><html><head><title>Help for package MBMethPred</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MBMethPred}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BoxPlot'><p>Box plot</p></a></li>
<li><a href='#ConfusionMatrix'><p>Confusion matrix</p></a></li>
<li><a href='#Data1'><p>Training data</p></a></li>
<li><a href='#Data2'><p>Data2</p></a></li>
<li><a href='#Data3'><p>Data3</p></a></li>
<li><a href='#KNearestNeighborModel'><p>K nearest neighbor model</p></a></li>
<li><a href='#LinearDiscriminantAnalysisModel'><p>Linear discriminant analysis model</p></a></li>
<li><a href='#ModelMetrics'><p>Model metrics</p></a></li>
<li><a href='#NaiveBayesModel'><p>Naive bayes model</p></a></li>
<li><a href='#NeuralNetworkModel'><p>Artificial neural network model</p></a></li>
<li><a href='#NewDataPredictionResult'><p>New data prediction result</p></a></li>
<li><a href='#RandomForestModel'><p>Random forest model</p></a></li>
<li><a href='#ReadMethylFile'><p>Input file for prediction</p></a></li>
<li><a href='#ReadSNFData'><p>Input file for similarity network fusion (SNF)</p></a></li>
<li><a href='#RLabels'><p>RLabels</p></a></li>
<li><a href='#SimilarityNetworkFusion'><p>Similarity network fusion (SNF)</p></a></li>
<li><a href='#SupportVectorMachineModel'><p>Support vector machine model</p></a></li>
<li><a href='#TSNEPlot'><p>t-SNE 3D plot</p></a></li>
<li><a href='#XGBoostModel'><p>XGBoost model</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Medulloblastoma Subgroups Prediction</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.4.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-09-08</td>
</tr>
<tr>
<td>Description:</td>
<td>Utilizing a combination of machine learning models (Random Forest, Naive Bayes, K-Nearest Neighbor, Support Vector Machines, Extreme Gradient Boosting, and Linear Discriminant Analysis) and a deep Artificial Neural Network model, 'MBMethPred' can predict medulloblastoma subgroups, including wingless (WNT), sonic hedgehog (SHH), Group 3, and Group 4 from DNA methylation beta values. See Sharif Rahmani E, Lawarde A, Lingasamy P, Moreno SV, Salumets A and Modhukur V (2023), MBMethPred: a computational framework for the accurate classification of childhood medulloblastoma subgroups using data integration and AI-based approaches. Front. Genet. 14:1233657. &lt;<a href="https://doi.org/10.3389%2Ffgene.2023.1233657">doi:10.3389/fgene.2023.1233657</a>&gt; for more details.</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/sharifrahmanie/MBMethPred">https://github.com/sharifrahmanie/MBMethPred</a></td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Edris Sharif Rahmani &lt;rahmani.biotech@gmail.com&gt;</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/sharifrahmanie/MBMethPred/issues">https://github.com/sharifrahmanie/MBMethPred/issues</a></td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>LazyDataCompression:</td>
<td>xz</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>stringr, ggplot2, parallel, caTools, caret, keras, MASS,
Rtsne, SNFtool, class, dplyr, e1071, pROC, randomForest, readr,
reshape2, reticulate, rgl, tensorflow, xgboost</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, utils, stats, scales</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-18 13:38:07 UTC; edi</td>
</tr>
<tr>
<td>Author:</td>
<td>Edris Sharif Rahmani
    <a href="https://orcid.org/0000-0002-7899-1663"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ctb,
    cre],
  Ankita Sunil Lawarde
    <a href="https://orcid.org/0000-0001-7572-4431"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ctb],
  Vijayachitra Modhukur
    <a href="https://orcid.org/0000-0002-7123-9903"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-18 14:10:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='BoxPlot'>Box plot</h2><span id='topic+BoxPlot'></span>

<h3>Description</h3>

<p>A function to draw a box plot for the DNA methylation dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BoxPlot(File, Projname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BoxPlot_+3A_file">File</code></td>
<td>
<p>The output of ReadMethylFile function.</p>
</td></tr>
<tr><td><code id="BoxPlot_+3A_projname">Projname</code></td>
<td>
<p>A name used to name the plot. The default is null.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data &lt;- Data2[1:10,]
data &lt;- cbind(rownames(data), data)
colnames(data)[1] &lt;- "ID"
BoxPlot(File = data)
</code></pre>

<hr>
<h2 id='ConfusionMatrix'>Confusion matrix</h2><span id='topic+ConfusionMatrix'></span>

<h3>Description</h3>

<p>A function to calculate the confusion matrix of the machine and deep learning models. It outputs Accuracy, Precision, Sensitivity, F1-Score, Specificity, and AUC_average.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ConfusionMatrix(y_true, y_pred)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ConfusionMatrix_+3A_y_true">y_true</code></td>
<td>
<p>True labels</p>
</td></tr>
<tr><td><code id="ConfusionMatrix_+3A_y_pred">y_pred</code></td>
<td>
<p>Predicted labels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(1234)
data &lt;- Data1[1:10,]
data$subgroup &lt;- factor(data$subgroup)
fac &lt;- ncol(data)
split &lt;- caTools::sample.split(data[, fac], SplitRatio = 0.8)
training_set &lt;- subset(data, split == TRUE)
test_set &lt;- subset(data, split == FALSE)
rf &lt;- randomForest::randomForest(x = training_set[-fac],
                                 y = training_set[, fac],
                                 ntree = 10)
y_pred &lt;- predict(rf, newdata = test_set[-fac])
ConfusionMatrix(y_true = test_set[, fac],
                y_pred = y_pred)
</code></pre>

<hr>
<h2 id='Data1'>Training data</h2><span id='topic+Data1'></span><span id='topic+dataset1'></span>

<h3>Description</h3>

<p>Data1 is a medulloblastoma DNA methylation beta values from a GEO series (GSE85212) and focuses on 399 as the most important probes. This dataset is used to train and test the machine and deep learning models.
</p>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Source</h3>

<p>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85212
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Data1)
</code></pre>

<hr>
<h2 id='Data2'>Data2</h2><span id='topic+Data2'></span><span id='topic+dataset2'></span>

<h3>Description</h3>

<p>Data2 is a medulloblastoma DNA methylation beta values (GSE85212, 50 samples) including 10000 most variable probes used for similarity network fusion.
</p>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Source</h3>

<p>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85212
</p>


<h3>References</h3>

<p>Cavalli FMG, Remke M, Rampasek L, Peacock J et al. Intertumoral Heterogeneity within Medulloblastoma Subgroups. Cancer Cell 2017 Jun 12;31(6):737-754.e6. PMID: 28609654
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Data2)
</code></pre>

<hr>
<h2 id='Data3'>Data3</h2><span id='topic+Data3'></span><span id='topic+dataset3'></span>

<h3>Description</h3>

<p>Data3 is an gene expression dataset from primary medulloblastoma samples (GSE85217, 50 samples) used for similarity network fusion.
</p>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Source</h3>

<p>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85217
</p>


<h3>References</h3>

<p>Cavalli FMG, Remke M, Rampasek L, Peacock J et al. Intertumoral Heterogeneity within Medulloblastoma Subgroups. Cancer Cell 2017 Jun 12;31(6):737-754.e6. PMID: 28609654
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Data3)
</code></pre>

<hr>
<h2 id='KNearestNeighborModel'>K nearest neighbor model</h2><span id='topic+KNearestNeighborModel'></span>

<h3>Description</h3>

<p>A function to train a K nearest neighbor model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="KNearestNeighborModel_+3A_splitratio">SplitRatio</code></td>
<td>
<p>Train and test split ratio. A value greater or equal to zero and less than one.</p>
</td></tr>
<tr><td><code id="KNearestNeighborModel_+3A_cv">CV</code></td>
<td>
<p>The number of folds for cross-validation. It should be greater than one.</p>
</td></tr>
<tr><td><code id="KNearestNeighborModel_+3A_k">K</code></td>
<td>
<p>The number of nearest neighbors.</p>
</td></tr>
<tr><td><code id="KNearestNeighborModel_+3A_ncores">NCores</code></td>
<td>
<p>The number of cores for parallel computing.</p>
</td></tr>
<tr><td><code id="KNearestNeighborModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(111)
knn &lt;- KNearestNeighborModel(SplitRatio = 0.8,
                             CV = 3,
                             K = 3,
                             NCores = 1,
                             NewData = NULL)
</code></pre>

<hr>
<h2 id='LinearDiscriminantAnalysisModel'>Linear discriminant analysis model</h2><span id='topic+LinearDiscriminantAnalysisModel'></span>

<h3>Description</h3>

<p>A function to train a linear discriminant analysis model to classify medulloblastoma subgroups using the DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="LinearDiscriminantAnalysisModel_+3A_splitratio">SplitRatio</code></td>
<td>
<p>Train and test split ratio. A value greater or equal than zero and less than one.</p>
</td></tr>
<tr><td><code id="LinearDiscriminantAnalysisModel_+3A_cv">CV</code></td>
<td>
<p>The number of folds for cross validation. It should be greater than one.</p>
</td></tr>
<tr><td><code id="LinearDiscriminantAnalysisModel_+3A_ncores">NCores</code></td>
<td>
<p>The number of cores for parallel computing.</p>
</td></tr>
<tr><td><code id="LinearDiscriminantAnalysisModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
lda &lt;- LinearDiscriminantAnalysisModel(SplitRatio = 0.8,
                                       CV = 2,
                                       NCores = 1,
                                       NewData = NULL)
</code></pre>

<hr>
<h2 id='ModelMetrics'>Model metrics</h2><span id='topic+ModelMetrics'></span>

<h3>Description</h3>

<p>A function to extract the confusion matrix information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ModelMetrics(Model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ModelMetrics_+3A_model">Model</code></td>
<td>
<p>A trained model.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>xgboost &lt;- XGBoostModel(SplitRatio = 0.2,
                        CV = 2,
                        NCores = 1,
                        NewData = NULL)
ModelMetrics(Model = xgboost)
</code></pre>

<hr>
<h2 id='NaiveBayesModel'>Naive bayes model</h2><span id='topic+NaiveBayesModel'></span>

<h3>Description</h3>

<p>A function to train a Naive Bayes model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="NaiveBayesModel_+3A_splitratio">SplitRatio</code></td>
<td>
<p>Train and test split ratio. A value greater or equal to zero and less than one.</p>
</td></tr>
<tr><td><code id="NaiveBayesModel_+3A_cv">CV</code></td>
<td>
<p>The number of folds for cross-validation. It should be greater than one.</p>
</td></tr>
<tr><td><code id="NaiveBayesModel_+3A_threshold">Threshold</code></td>
<td>
<p>The threshold for deciding class probability. A value greater or equal to zero and less than one.</p>
</td></tr>
<tr><td><code id="NaiveBayesModel_+3A_ncores">NCores</code></td>
<td>
<p>The number of cores for parallel computing.</p>
</td></tr>
<tr><td><code id="NaiveBayesModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
nb &lt;- NaiveBayesModel(SplitRatio = 0.8,
                      CV = 2,
                      Threshold = 0.8,
                      NCores = 1,
                      NewData = NULL)
</code></pre>

<hr>
<h2 id='NeuralNetworkModel'>Artificial neural network model</h2><span id='topic+NeuralNetworkModel'></span>

<h3>Description</h3>

<p>A function to train an artificial neural network model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="NeuralNetworkModel_+3A_epochs">Epochs</code></td>
<td>
<p>The number of epochs.</p>
</td></tr>
<tr><td><code id="NeuralNetworkModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
<tr><td><code id="NeuralNetworkModel_+3A_installtensorflow">InstallTensorFlow</code></td>
<td>
<p>Logical. Running this function for the first time, you need to install TensorFlow library (V 2.10-cpu). Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
set.seed(1234)
ann &lt;- NeuralNetworkModel(Epochs = 100,
                          NewData = NULL,
                          InstallTensorFlow = TRUE)

## End(Not run)
</code></pre>

<hr>
<h2 id='NewDataPredictionResult'>New data prediction result</h2><span id='topic+NewDataPredictionResult'></span>

<h3>Description</h3>

<p>A function to output the predicted medulloblastoma subgroups by trained models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NewDataPredictionResult(Model)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="NewDataPredictionResult_+3A_model">Model</code></td>
<td>
<p>A trained model</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(10)
fac &lt;- ncol(Data1)
NewData &lt;- sample(data.frame(t(Data1[,-fac])),10)
NewData &lt;- cbind(rownames(NewData), NewData)
colnames(NewData)[1] &lt;- "ID"
xgboost &lt;- XGBoostModel(SplitRatio = 0.2,
                        CV = 2,
                        NCores = 1,
                        NewData = NewData)
NewDataPredictionResult(Model = xgboost)
</code></pre>

<hr>
<h2 id='RandomForestModel'>Random forest model</h2><span id='topic+RandomForestModel'></span>

<h3>Description</h3>

<p>A function to train a random forest model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="RandomForestModel_+3A_splitratio">SplitRatio</code></td>
<td>
<p>Train and test split ratio. A value greater or equal to zero and less than one.</p>
</td></tr>
<tr><td><code id="RandomForestModel_+3A_cv">CV</code></td>
<td>
<p>The number of folds for cross-validation. It should be greater than one.</p>
</td></tr>
<tr><td><code id="RandomForestModel_+3A_ntree">NTree</code></td>
<td>
<p>The number of trees to be grown.</p>
</td></tr>
<tr><td><code id="RandomForestModel_+3A_ncores">NCores</code></td>
<td>
<p>The number of cores for parallel computing.</p>
</td></tr>
<tr><td><code id="RandomForestModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(21)
rf &lt;- RandomForestModel(SplitRatio = 0.8,
                        CV = 3,
                        NTree = 10,
                        NCores = 1,
                        NewData = NULL)
</code></pre>

<hr>
<h2 id='ReadMethylFile'>Input file for prediction</h2><span id='topic+ReadMethylFile'></span>

<h3>Description</h3>

<p>A function to read DNA methylation files. It can be used as the new data for prediction by every model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadMethylFile(File)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadMethylFile_+3A_file">File</code></td>
<td>
<p>A data frame with tsv or csv file extension. The first column of the data frame is the CpG methylation probe that starts with cg characters and is followed by a number (e.g., cg100091). Other columns are samples with methylation beta values. All columns in the data frame should have a name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
methyl &lt;- ReadMethylFile(File = "file.csv")

## End(Not run)
</code></pre>

<hr>
<h2 id='ReadSNFData'>Input file for similarity network fusion (SNF)</h2><span id='topic+ReadSNFData'></span>

<h3>Description</h3>

<p>A function to read user-provided file feeding into the SNF function (from the SNFtools package).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadSNFData(File)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadSNFData_+3A_file">File</code></td>
<td>
<p>A data frame with tsv or csv file extension. The first column of the data frame is the CpG methylation probe that starts with cg characters and is followed by a number (e.g., cg100091). Other columns are samples with methylation beta values. All columns in the data frame should have a name.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data &lt;- ReadSNFData(File = "file.csv")

## End(Not run)
</code></pre>

<hr>
<h2 id='RLabels'>RLabels</h2><span id='topic+RLabels'></span><span id='topic+Subgroups'></span>

<h3>Description</h3>

<p>The actual labels from the medulloblastoma DNA methylation beta values (GSE85212, 50 samples) that was used for similarity network fusion.
</p>


<h3>Value</h3>

<p>Factor
</p>


<h3>Source</h3>

<p>https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE85212
</p>


<h3>References</h3>

<p>Cavalli FMG, Remke M, Rampasek L, Peacock J et al. Intertumoral Heterogeneity within Medulloblastoma Subgroups. Cancer Cell 2017 Jun 12;31(6):737-754.e6. PMID: 28609654
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RLabels)
</code></pre>

<hr>
<h2 id='SimilarityNetworkFusion'>Similarity network fusion (SNF)</h2><span id='topic+SimilarityNetworkFusion'></span>

<h3>Description</h3>

<p>A function to perform SNF function (from SNFtool package) and output clusters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SimilarityNetworkFusion(
  Files = NULL,
  NNeighbors,
  Sigma,
  NClusters,
  CLabels = NULL,
  RLabels = NULL,
  Niterations
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SimilarityNetworkFusion_+3A_files">Files</code></td>
<td>
<p>A list of data frames created using the ReadSNFData function or matrices.</p>
</td></tr>
<tr><td><code id="SimilarityNetworkFusion_+3A_nneighbors">NNeighbors</code></td>
<td>
<p>The number of nearest neighbors.</p>
</td></tr>
<tr><td><code id="SimilarityNetworkFusion_+3A_sigma">Sigma</code></td>
<td>
<p>The variance for local model.</p>
</td></tr>
<tr><td><code id="SimilarityNetworkFusion_+3A_nclusters">NClusters</code></td>
<td>
<p>The number of clusters.</p>
</td></tr>
<tr><td><code id="SimilarityNetworkFusion_+3A_clabels">CLabels</code></td>
<td>
<p>A string vector to name the clusters. Optional.</p>
</td></tr>
<tr><td><code id="SimilarityNetworkFusion_+3A_rlabels">RLabels</code></td>
<td>
<p>The actual label of samples to calculate the Normalized Mutual Information (NMI) score. Optional.</p>
</td></tr>
<tr><td><code id="SimilarityNetworkFusion_+3A_niterations">Niterations</code></td>
<td>
<p>The number of iterations for the diffusion process.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Factor
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(RLabels) # Real labels
data(Data2) # Methylation
data(Data3) # Gene expression
snf &lt;- SimilarityNetworkFusion(Files = list(Data2, Data3),
                               NNeighbors  = 13,
                               Sigma = 0.75,
                               NClusters = 4,
                               CLabels = c("Group4", "SHH", "WNT", "Group3"),
                               RLabels = RLabels,
                               Niterations = 10)
snf
</code></pre>

<hr>
<h2 id='SupportVectorMachineModel'>Support vector machine model</h2><span id='topic+SupportVectorMachineModel'></span>

<h3>Description</h3>

<p>A function to train a support vector machine model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="SupportVectorMachineModel_+3A_splitratio">SplitRatio</code></td>
<td>
<p>Train and test split ratio. A value greater or equal to zero and less than one.</p>
</td></tr>
<tr><td><code id="SupportVectorMachineModel_+3A_cv">CV</code></td>
<td>
<p>The number of folds for cross-validation. It should be greater than one.</p>
</td></tr>
<tr><td><code id="SupportVectorMachineModel_+3A_ncores">NCores</code></td>
<td>
<p>The number of cores for parallel computing.</p>
</td></tr>
<tr><td><code id="SupportVectorMachineModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(56)
svm &lt;- SupportVectorMachineModel(SplitRatio = 0.8,
                                 CV = 3,
                                 NCores = 1,
                                 NewData = NULL)
</code></pre>

<hr>
<h2 id='TSNEPlot'>t-SNE 3D plot</h2><span id='topic+TSNEPlot'></span>

<h3>Description</h3>

<p>A function to draw a 3D t-SNE plot for DNA methylation beta values using the K-means clustering technique.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>TSNEPlot(File, NCluster = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="TSNEPlot_+3A_file">File</code></td>
<td>
<p>The output of ReadMethylFile function.</p>
</td></tr>
<tr><td><code id="TSNEPlot_+3A_ncluster">NCluster</code></td>
<td>
<p>The number of cluster.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Objects of rgl
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
set.seed(123)
data &lt;- Data2[1:100,]
data &lt;- data.frame(t(data))
data &lt;- cbind(rownames(data), data)
colnames(data)[1] &lt;- "ID"
TSNEPlot(File = data, NCluster = 4)

</code></pre>

<hr>
<h2 id='XGBoostModel'>XGBoost model</h2><span id='topic+XGBoostModel'></span>

<h3>Description</h3>

<p>A function to train an XGBoost model to classify medulloblastoma subgroups using DNA methylation beta values (Illumina Infinium HumanMethylation450). Prediction is followed by training if new data is provided.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="XGBoostModel_+3A_splitratio">SplitRatio</code></td>
<td>
<p>Train and test split ratio. A value greater or equal to zero and less than one.</p>
</td></tr>
<tr><td><code id="XGBoostModel_+3A_cv">CV</code></td>
<td>
<p>The number of folds for cross-validation. It should be greater than one.</p>
</td></tr>
<tr><td><code id="XGBoostModel_+3A_ncores">NCores</code></td>
<td>
<p>The number of cores for parallel computing.</p>
</td></tr>
<tr><td><code id="XGBoostModel_+3A_newdata">NewData</code></td>
<td>
<p>A methylation beta values input from the ReadMethylFile function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>


<h3>Examples</h3>

<pre><code class='language-R'>set.seed(123)
xgboost &lt;- XGBoostModel(SplitRatio = 0.2,
                        CV = 2,
                        NCores = 1,
                        NewData = NULL)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
