<!DOCTYPE html><html lang="en"><head><title>Help for package mbmixture</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mbmixture}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bootstrapNull'><p>Bootstrap to assess significance</p></a></li>
<li><a href='#bootstrapSE'><p>Bootstrap to get standard errors</p></a></li>
<li><a href='#mbmix_loglik'><p>log likelihood function for microbiome mixture</p></a></li>
<li><a href='#mbmixdata'><p>Example dataset for mbmixture package</p></a></li>
<li><a href='#mle_e'><p>MLE of e for fixed p</p></a></li>
<li><a href='#mle_p'><p>MLE of p for fixed e</p></a></li>
<li><a href='#mle_pe'><p>Find MLEs for microbiome mixture</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.6</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-11-27</td>
</tr>
<tr>
<td>Title:</td>
<td>Microbiome Mixture Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Evaluate whether a microbiome sample is a mixture of two
    samples, by fitting a model for the number of read counts as a
    function of single nucleotide polymorphism (SNP) allele and the
    genotypes of two potential source samples.
    Lobo et al. (2021) &lt;<a href="https://doi.org/10.1093%2Fg3journal%2Fjkab308">doi:10.1093/g3journal/jkab308</a>&gt;.</td>
</tr>
<tr>
<td>Author:</td>
<td>Karl W Broman <a href="https://orcid.org/0000-0002-4914-6671"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karl W Broman &lt;broman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, parallel, numDeriv</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, testthat, devtools, roxygen2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kbroman/mbmixture">https://github.com/kbroman/mbmixture</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/kbroman/mbmixture/issues">https://github.com/kbroman/mbmixture/issues</a></td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-11-27 22:51:22 UTC; kbroman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-11-27 23:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bootstrapNull'>Bootstrap to assess significance</h2><span id='topic+bootstrapNull'></span>

<h3>Description</h3>

<p>Perform a parametric bootstrap to assess whether there is significant evidence that a sample is a mixture.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrapNull(
  tab,
  n_rep = 1000,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = TRUE,
  cores = 1,
  return_raw = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bootstrapNull_+3A_tab">tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td></tr>
<tr><td><code id="bootstrapNull_+3A_n_rep">n_rep</code></td>
<td>
<p>Number of bootstrap replicates</p>
</td></tr>
<tr><td><code id="bootstrapNull_+3A_interval">interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td></tr>
<tr><td><code id="bootstrapNull_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td></tr>
<tr><td><code id="bootstrapNull_+3A_check_boundary">check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td></tr>
<tr><td><code id="bootstrapNull_+3A_cores">cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code><a href="parallel.html#topic+makeCluster">parallel::makeCluster()</a></code>.</p>
</td></tr>
<tr><td><code id="bootstrapNull_+3A_return_raw">return_raw</code></td>
<td>
<p>If TRUE, return the raw results. If FALSE, just return the p-value.
Unlink <code><a href="#topic+bootstrapSE">bootstrapSE()</a></code>, here the default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>return_raw=FALSE</code>, a single numeric value (the p-value).If
<code>return_raw=TRUE</code>, a vector of length <code>n_rep</code> with the LRT statistics from each
bootstrap replicate.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bootstrapSE">bootstrapSE()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
# just 100 bootstrap replicates, as an illustration
bootstrapNull(mbmixdata, n_rep=100)

</code></pre>

<hr>
<h2 id='bootstrapSE'>Bootstrap to get standard errors</h2><span id='topic+bootstrapSE'></span>

<h3>Description</h3>

<p>Perform a parametric bootstrap to get estimated standard errors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>bootstrapSE(
  tab,
  n_rep = 1000,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = FALSE,
  cores = 1,
  return_raw = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="bootstrapSE_+3A_tab">tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td></tr>
<tr><td><code id="bootstrapSE_+3A_n_rep">n_rep</code></td>
<td>
<p>Number of bootstrap replicates</p>
</td></tr>
<tr><td><code id="bootstrapSE_+3A_interval">interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td></tr>
<tr><td><code id="bootstrapSE_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td></tr>
<tr><td><code id="bootstrapSE_+3A_check_boundary">check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td></tr>
<tr><td><code id="bootstrapSE_+3A_cores">cores</code></td>
<td>
<p>Number of CPU cores to use, for parallel calculations.
(If <code>0</code>, use <code><a href="parallel.html#topic+detectCores">parallel::detectCores()</a></code>.)
Alternatively, this can be links to a set of cluster sockets, as
produced by <code><a href="parallel.html#topic+makeCluster">parallel::makeCluster()</a></code>.</p>
</td></tr>
<tr><td><code id="bootstrapSE_+3A_return_raw">return_raw</code></td>
<td>
<p>If TRUE, return the raw results. If FALSE, just return the estimated standard errors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If <code>return_raw=FALSE</code>, a vector of two standard errors. If
<code>return_raw=TRUE</code>, an matrix of size <code>n_rep</code> x 2 with the detailed
bootstrap results.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bootstrapNull">bootstrapNull()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
# just 100 bootstrap replicates, as an illustration
bootstrapSE(mbmixdata, n_rep=100)

</code></pre>

<hr>
<h2 id='mbmix_loglik'>log likelihood function for microbiome mixture</h2><span id='topic+mbmix_loglik'></span>

<h3>Description</h3>

<p>Calculate log likelihood function for microbiome sample mixture model at particular values of <code>p</code> and <code>e</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mbmix_loglik(tab, p, e = 0)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mbmix_loglik_+3A_tab">tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td></tr>
<tr><td><code id="mbmix_loglik_+3A_p">p</code></td>
<td>
<p>Contaminant probability (proportion of mixture coming from the second sample).</p>
</td></tr>
<tr><td><code id="mbmix_loglik_+3A_e">e</code></td>
<td>
<p>Sequencing error rate.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The log likelihood evaluated at <code>p</code> and <code>e</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
mbmix_loglik(mbmixdata, p=0.74, e=0.002)

</code></pre>

<hr>
<h2 id='mbmixdata'>Example dataset for mbmixture package</h2><span id='topic+mbmixdata'></span>

<h3>Description</h3>

<p>Example dataset for mbmixture package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mbmixdata)
</code></pre>


<h3>Format</h3>

<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
mle_pe(mbmixdata)
</code></pre>

<hr>
<h2 id='mle_e'>MLE of e for fixed p</h2><span id='topic+mle_e'></span>

<h3>Description</h3>

<p>Calculate the MLE of the sequencing error rate e for a fixed value of the contaminant probability p.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle_e(
  tab,
  p = 0.05,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_e_+3A_tab">tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td></tr>
<tr><td><code id="mle_e_+3A_p">p</code></td>
<td>
<p>Assumed value for the contaminant probability</p>
</td></tr>
<tr><td><code id="mle_e_+3A_interval">interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td></tr>
<tr><td><code id="mle_e_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td></tr>
<tr><td><code id="mle_e_+3A_check_boundary">check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single numeric value, the MLE of <code>e</code>, with the log likelihood as an attribute.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
mle_e(mbmixdata, p=0.74)

</code></pre>

<hr>
<h2 id='mle_p'>MLE of p for fixed e</h2><span id='topic+mle_p'></span>

<h3>Description</h3>

<p>Calculate the MLE of the contaminant probability p for a fixed value of the sequencing error rate e.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle_p(
  tab,
  e = 0.002,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_p_+3A_tab">tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td></tr>
<tr><td><code id="mle_p_+3A_e">e</code></td>
<td>
<p>Assumed value for the sequencing error rate</p>
</td></tr>
<tr><td><code id="mle_p_+3A_interval">interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td></tr>
<tr><td><code id="mle_p_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td></tr>
<tr><td><code id="mle_p_+3A_check_boundary">check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single numeric value, the MLE of <code>p</code>, with the log likelihood as an attribute.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
mle_p(mbmixdata, e=0.002)

</code></pre>

<hr>
<h2 id='mle_pe'>Find MLEs for microbiome mixture</h2><span id='topic+mle_pe'></span>

<h3>Description</h3>

<p>Find joint MLEs of p and e for microbiome mixture model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle_pe(
  tab,
  interval = c(0, 1),
  tol = 0.000001,
  check_boundary = FALSE,
  SE = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_pe_+3A_tab">tab</code></td>
<td>
<p>Dataset of read counts as 3d array of size 3x3x2, genotype in first sample x genotype in second sample x allele in read.</p>
</td></tr>
<tr><td><code id="mle_pe_+3A_interval">interval</code></td>
<td>
<p>Interval to which each parameter should be constrained</p>
</td></tr>
<tr><td><code id="mle_pe_+3A_tol">tol</code></td>
<td>
<p>Tolerance for convergence</p>
</td></tr>
<tr><td><code id="mle_pe_+3A_check_boundary">check_boundary</code></td>
<td>
<p>If TRUE, explicitly check the boundaries of <code>interval</code>.</p>
</td></tr>
<tr><td><code id="mle_pe_+3A_se">SE</code></td>
<td>
<p>If TRUE, get estimated standard errors.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the estimates of <code>p</code> and <code>e</code> along with the evaluated log likelihood and likelihood ratio test statistics for the hypotheses p=0 and p=1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mbmixdata)
mle_pe(mbmixdata)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
