<!DOCTYPE html><html lang="en-US"><head><title>Help for package mcbette</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mcbette}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mcbette-package'><p>mcbette: Model Comparison Using Babette</p></a></li>
<li><a href='#calc_weights'><p>Calculate the weights for each marginal likelihood</p></a></li>
<li><a href='#can_run_mcbette'><p>Can 'mcbette' run?</p></a></li>
<li><a href='#check_beast2_ns_pkg'><p>Checks if the BEAST2 'NS' package is installed.</p></a></li>
<li><a href='#check_marg_liks'><p>Check if the <code>marg_liks</code> are of the same type as returned</p>
by est_marg_liks.</a></li>
<li><a href='#check_mcbette_state'><p>Check if the <code>mcbette_state</code> is valid.</p></a></li>
<li><a href='#default_params_doc'><p>Documentation of general function arguments.</p>
This function does nothing.
It is intended to inherit function argument documentation.</a></li>
<li><a href='#est_marg_lik'><p>Estimate the marginal likelihood for an inference model.</p></a></li>
<li><a href='#est_marg_liks'><p>Estimate the marginal likelihoods for one or more inference models</p></a></li>
<li><a href='#get_mcbette_state'><p>Get the current state of mcbette</p></a></li>
<li><a href='#get_test_marg_liks'><p>Get testing <code>marg_liks</code></p></a></li>
<li><a href='#interpret_bayes_factor'><p>Interpret a Bayes factor</p></a></li>
<li><a href='#interpret_marg_lik_estimates'><p>Interpret the marginal likelihood estimates</p></a></li>
<li><a href='#is_marg_liks'><p>Determine if the <code>marg_liks</code> is valid</p></a></li>
<li><a href='#mcbette_report'><p>Create a mcbette report,</p>
to be used when reporting bugs</a></li>
<li><a href='#mcbette_self_test'><p>Performs a minimal mcbette run</p></a></li>
<li><a href='#plot_marg_liks'><p>Plot the <code>marg_liks</code></p></a></li>
<li><a href='#set_mcbette_state'><p>Set the mcbette state.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Model Comparison Using 'babette'</td>
</tr>
<tr>
<td>Version:</td>
<td>1.15.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richèl J.C. Bilderbeek &lt;rjcbilderbeek@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>'BEAST2' (<a href="https://www.beast2.org">https://www.beast2.org</a>) is a widely used
    Bayesian phylogenetic tool, that uses DNA/RNA/protein data
    and many model priors to create a posterior of jointly estimated 
    phylogenies and parameters.
    'mcbette' allows to do a Bayesian model comparison over some
    site and clock models, 
    using 'babette' (<a href="https://github.com/ropensci/babette/">https://github.com/ropensci/babette/</a>).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ropensci/mcbette/">https://github.com/ropensci/mcbette/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ropensci/mcbette/issues">https://github.com/ropensci/mcbette/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>babette (&ge; 2.3), beautier (&ge; 2.6.2), beastier (&ge; 2.4.6),
curl, devtools, mauricer (&ge; 2.5), Rmpfr, testit, txtplot</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ape, ggplot2, hunspell, knitr, lintr, markdown, nLTT,
phangorn, rappdirs, rmarkdown, spelling, stringr, testthat (&ge;
2.1.0), tracerer</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>BEAST2 (https://www.beast2.org/)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-31 14:31:04 UTC; richel</td>
</tr>
<tr>
<td>Author:</td>
<td>Richèl J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Joëlle Barido-Sottani [rev] (Joëlle reviewed the package for rOpenSci,
    see https://github.com/ropensci/software-review/issues/360),
  Vikram Baliga <a href="https://orcid.org/0000-0002-9367-8974"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [rev] (Vikram reviewed the package for rOpenSci, see
    https://github.com/ropensci/software-review/issues/360)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-08-19 08:40:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='mcbette-package'>mcbette: Model Comparison Using Babette</h2><span id='topic+mcbette'></span><span id='topic+mcbette-package'></span>

<h3>Description</h3>

<p>'mcbette' does a model comparing using <a href="babette.html#topic+babette">babette</a>,
where the models are Bayesian phylogenetic models,
as created by <a href="beautier.html#topic+create_inference_model">create_inference_model</a>.
</p>


<h3>Details</h3>

<p>The main function is <a href="#topic+est_marg_liks">est_marg_liks</a>,
which estimate the marginal likelihoods (aka evidence)
for one or more inference models, based on a single alignment.
Also, the marginal likelihoods are compared, resulting in a
relative weight for each model, where a relative weight of a model
close to <code>1.0</code> means that that model is way likelier than
the others.
</p>
<p>In the process, multiple (temporary) files are created (where
<code>[x]</code> denotes the index in a list)
</p>

<ul>
<li> <p><code>beast2_optionses[x]$input_filename</code>
path to the the BEAST2 XML input file
</p>
</li>
<li> <p><code>beast2_optionses[x]$output_state_filename</code>
path to the BEAST2 XML state file
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$tracelog$filename</code>
path to the BEAST2 trace file with parameter estimates
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$treelog$filename</code>
path to the BEAST2 <code>trees</code> file with the posterior trees
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$screenlog$filename</code>
path to the BEAST2 screen output file
</p>
</li></ul>

<p>These file can be deleted manually by <a href="babette.html#topic+bbt_delete_temp_files">bbt_delete_temp_files</a>,
else these will be deleted automatically by the operating system.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <a href="#topic+can_run_mcbette">can_run_mcbette</a> to see if 'mcbette' can run.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (can_run_mcbette()) {

  # An example FASTA file
  fasta_filename &lt;- system.file("extdata", "simple.fas", package = "mcbette")

  inference_model_1 &lt;- beautier::create_ns_inference_model(
    site_model = beautier::create_jc69_site_model()
  )
  inference_model_2 &lt;- beautier::create_ns_inference_model(
    site_model = beautier::create_gtr_site_model()
  )

  # Shorten the run, by doing a short (dirty, unreliable) MCMC
  inference_model_1$mcmc &lt;- beautier::create_test_ns_mcmc()
  inference_model_2$mcmc &lt;- beautier::create_test_ns_mcmc()

  inference_models &lt;- c(list(inference_model_1), list(inference_model_2))

  # Estimate the marginal log-likelihoods of the two models
  marg_liks &lt;- est_marg_liks(
    fasta_filename = fasta_filename,
    inference_models = inference_models
  )

  # Interpret the results
  interpret_marg_lik_estimates(marg_liks)

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='calc_weights'>Calculate the weights for each marginal likelihood</h2><span id='topic+calc_weights'></span>

<h3>Description</h3>

<p>Calculate the weights for each marginal likelihood
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc_weights(marg_liks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc_weights_+3A_marg_liks">marg_liks</code></td>
<td>
<p>(non-log) marginal likelihood estimates</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the weight of each marginal likelihood estimate,
which will sum up to 1.0
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Evidences (aka marginal likelihoods) can be very small
evidences &lt;- c(0.0001, 0.0002, 0.0003, 0.0004)

# Sum will be 1.0
calc_weights(evidences)

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='can_run_mcbette'>Can 'mcbette' run?</h2><span id='topic+can_run_mcbette'></span>

<h3>Description</h3>

<p>Can 'mcbette' run?
Will return <a href="base.html#topic+TRUE">TRUE</a> if:
</p>

<ul>
<li><p> (1) Running on Linux or MacOS
</p>
</li>
<li><p> (2) BEAST2 is installed
</p>
</li>
<li><p> (3) The BEAST2 NS package is installed
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>can_run_mcbette(beast2_folder = beastier::get_default_beast2_folder())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="can_run_mcbette_+3A_beast2_folder">beast2_folder</code></td>
<td>
<p>the folder where the BEAST2 is installed.
Note that this is not the folder where the BEAST2 executable is installed:
the BEAST2 executable is in a subfolder.
Use <a href="beastier.html#topic+get_default_beast2_folder">get_default_beast2_folder</a>
to get the default BEAST2 folder.
Use <a href="beastier.html#topic+get_default_beast2_bin_path">get_default_beast2_bin_path</a>
to get the full path to the default BEAST2 executable.
Use <a href="beastier.html#topic+get_default_beast2_jar_path">get_default_beast2_jar_path</a>
to get the full path to the default BEAST2 jar file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p><a href="base.html#topic+TRUE">TRUE</a> if 'mcbette' can run.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>can_run_mcbette()

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='check_beast2_ns_pkg'>Checks if the BEAST2 'NS' package is installed.</h2><span id='topic+check_beast2_ns_pkg'></span>

<h3>Description</h3>

<p>Checks if the BEAST2 'NS' package is installed.
Will <a href="base.html#topic+stop">stop</a> if not
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_beast2_ns_pkg(beast2_bin_path = beastier::get_default_beast2_bin_path())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_beast2_ns_pkg_+3A_beast2_bin_path">beast2_bin_path</code></td>
<td>
<p>path to the the BEAST2 binary file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. The function will <a href="base.html#topic+stop">stop</a> with an error message if
the BEAST2 'NS' package is not installed.
</p>

<hr>
<h2 id='check_marg_liks'>Check if the <code>marg_liks</code> are of the same type as returned
by <a href="#topic+est_marg_liks">est_marg_liks</a>.</h2><span id='topic+check_marg_liks'></span>

<h3>Description</h3>

<p><a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_marg_liks(marg_liks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_marg_liks_+3A_marg_liks">marg_liks</code></td>
<td>
<p>a table of (estimated) marginal likelihoods,
as, for example, created by <a href="#topic+est_marg_liks">est_marg_liks</a>.
This <a href="base.html#topic+data.frame">data.frame</a> has the following columns:
</p>

<ul>
<li> <p><code>site_model_name</code>: name of the site model,
must be an element of <a href="beautier.html#topic+get_site_model_names">get_site_model_names</a>
</p>
</li>
<li> <p><code>clock_model_name</code>: name of the clock model,
must be an element of <a href="beautier.html#topic+get_clock_model_names">get_clock_model_names</a>
</p>
</li>
<li> <p><code>tree_prior_name</code>: name of the tree prior,
must be an element of <a href="beautier.html#topic+get_tree_prior_names">get_tree_prior_names</a>
</p>
</li>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>weight</code>: relative model weight, a value from 1.0 (all
evidence is in favor of this model combination) to 0.0 (no
evidence in favor of this model combination)
</p>
</li>
<li> <p><code>ess</code>: effective sample size of the marginal likelihood
estimation
</p>
</li></ul>

<p>Use <a href="#topic+get_test_marg_liks">get_test_marg_liks</a> to get a test <code>marg_liks</code>.
Use <a href="#topic+is_marg_liks">is_marg_liks</a> to determine if a <code>marg_liks</code> is valid.
Use <a href="#topic+check_marg_liks">check_marg_liks</a> to check that a <code>marg_liks</code> is valid.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will <a href="base.html#topic+stop">stop</a> with an error message if there
is a problem with the input.
</p>

<hr>
<h2 id='check_mcbette_state'>Check if the <code>mcbette_state</code> is valid.</h2><span id='topic+check_mcbette_state'></span>

<h3>Description</h3>

<p>Check if the <code>mcbette_state</code> is valid.
Will <a href="base.html#topic+stop">stop</a> otherwise.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mcbette_state(mcbette_state)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_mcbette_state_+3A_mcbette_state">mcbette_state</code></td>
<td>
<p>the <a href="#topic+mcbette">mcbette</a> state,
which is a <a href="base.html#topic+list">list</a> with the following elements:
</p>

<ul>
<li>
<p>beast2_installed
<a href="base.html#topic+TRUE">TRUE</a> if BEAST2 is installed,
<a href="base.html#topic+FALSE">FALSE</a> otherwise

</p>
</li>
<li>
<p>ns_installed
<a href="base.html#topic+NA">NA</a> if BEAST2 is not installed.
<a href="base.html#topic+TRUE">TRUE</a> if the BEAST2 NS package is installed
<a href="base.html#topic+FALSE">FALSE</a> if the BEAST2 NS package is not installed

</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will <a href="base.html#topic+stop">stop</a> if the input is invalid.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='default_params_doc'>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.</h2><span id='topic+default_params_doc'></span>

<h3>Description</h3>

<p>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_params_doc(
  beast2_bin_path,
  beast2_folder,
  beast2_working_dir,
  beast2_options,
  beast2_optionses,
  clock_model,
  clock_models,
  epsilon,
  fasta_filename,
  inference_model,
  inference_models,
  marg_liks,
  mcbette_state,
  mcmc,
  os,
  rng_seed,
  site_model,
  site_models,
  tree_prior,
  tree_priors,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="default_params_doc_+3A_beast2_bin_path">beast2_bin_path</code></td>
<td>
<p>path to the the BEAST2 binary file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_folder">beast2_folder</code></td>
<td>
<p>the folder where the BEAST2 is installed.
Note that this is not the folder where the BEAST2 executable is installed:
the BEAST2 executable is in a subfolder.
Use <a href="beastier.html#topic+get_default_beast2_folder">get_default_beast2_folder</a>
to get the default BEAST2 folder.
Use <a href="beastier.html#topic+get_default_beast2_bin_path">get_default_beast2_bin_path</a>
to get the full path to the default BEAST2 executable.
Use <a href="beastier.html#topic+get_default_beast2_jar_path">get_default_beast2_jar_path</a>
to get the full path to the default BEAST2 jar file.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_working_dir">beast2_working_dir</code></td>
<td>
<p>folder in which BEAST2 will run and
produce intermediate files.
By default, this is a temporary folder</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_options">beast2_options</code></td>
<td>
<p>a <code>beast2_options</code> structure,
as can be created by <a href="beastier.html#topic+create_mcbette_beast2_options">create_mcbette_beast2_options</a>.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_beast2_optionses">beast2_optionses</code></td>
<td>
<p>list of one or more <code>beast2_options</code>
structures,
as can be created by <a href="beastier.html#topic+create_mcbette_beast2_options">create_mcbette_beast2_options</a>.
Use of reduplicated plural to achieve difference with
<code>beast2_options</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_model">clock_model</code></td>
<td>
<p>a clock model,
as can be created by <a href="beautier.html#topic+create_clock_model">create_clock_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_clock_models">clock_models</code></td>
<td>
<p>a list of one or more clock models,
as can be created by <a href="beautier.html#topic+create_clock_models">create_clock_models</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_epsilon">epsilon</code></td>
<td>
<p>measure of relative accuracy.
Smaller values result in longer, more precise estimations</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>name of the FASTA file</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_inference_model">inference_model</code></td>
<td>
<p>an inference model,
as can be created by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_inference_models">inference_models</code></td>
<td>
<p>a list of one or more inference models,
as can be created by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_marg_liks">marg_liks</code></td>
<td>
<p>a table of (estimated) marginal likelihoods,
as, for example, created by <a href="#topic+est_marg_liks">est_marg_liks</a>.
This <a href="base.html#topic+data.frame">data.frame</a> has the following columns:
</p>

<ul>
<li> <p><code>site_model_name</code>: name of the site model,
must be an element of <a href="beautier.html#topic+get_site_model_names">get_site_model_names</a>
</p>
</li>
<li> <p><code>clock_model_name</code>: name of the clock model,
must be an element of <a href="beautier.html#topic+get_clock_model_names">get_clock_model_names</a>
</p>
</li>
<li> <p><code>tree_prior_name</code>: name of the tree prior,
must be an element of <a href="beautier.html#topic+get_tree_prior_names">get_tree_prior_names</a>
</p>
</li>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>weight</code>: relative model weight, a value from 1.0 (all
evidence is in favor of this model combination) to 0.0 (no
evidence in favor of this model combination)
</p>
</li>
<li> <p><code>ess</code>: effective sample size of the marginal likelihood
estimation
</p>
</li></ul>

<p>Use <a href="#topic+get_test_marg_liks">get_test_marg_liks</a> to get a test <code>marg_liks</code>.
Use <a href="#topic+is_marg_liks">is_marg_liks</a> to determine if a <code>marg_liks</code> is valid.
Use <a href="#topic+check_marg_liks">check_marg_liks</a> to check that a <code>marg_liks</code> is valid.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mcbette_state">mcbette_state</code></td>
<td>
<p>the <a href="#topic+mcbette">mcbette</a> state,
which is a <a href="base.html#topic+list">list</a> with the following elements:
</p>

<ul>
<li>
<p>beast2_installed
<a href="base.html#topic+TRUE">TRUE</a> if BEAST2 is installed,
<a href="base.html#topic+FALSE">FALSE</a> otherwise

</p>
</li>
<li>
<p>ns_installed
<a href="base.html#topic+NA">NA</a> if BEAST2 is not installed.
<a href="base.html#topic+TRUE">TRUE</a> if the BEAST2 NS package is installed
<a href="base.html#topic+FALSE">FALSE</a> if the BEAST2 NS package is not installed

</p>
</li></ul>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mcmc">mcmc</code></td>
<td>
<p>an MCMC for the Nested Sampling run,
as can be created by <a href="beautier.html#topic+create_mcmc_nested_sampling">create_mcmc_nested_sampling</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_os">os</code></td>
<td>
<p>name of the operating system,
must be <code>unix</code> (Linux, Mac) or <code>win</code> (Windows)</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_rng_seed">rng_seed</code></td>
<td>
<p>a random number generator seed used for the BEAST2
inference</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_model">site_model</code></td>
<td>
<p>a site model,
as can be created by <a href="beautier.html#topic+create_site_model">create_site_model</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_site_models">site_models</code></td>
<td>
<p>a list of one or more site models,
as can be created by <a href="beautier.html#topic+create_site_models">create_site_models</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_prior">tree_prior</code></td>
<td>
<p>a tree prior,
as can be created by <a href="beautier.html#topic+create_tree_prior">create_tree_prior</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_tree_priors">tree_priors</code></td>
<td>
<p>a list of one or more tree priors,
as can be created by <a href="beautier.html#topic+create_tree_priors">create_tree_priors</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE show debug output</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='est_marg_lik'>Estimate the marginal likelihood for an inference model.</h2><span id='topic+est_marg_lik'></span>

<h3>Description</h3>

<p>Estimate the marginal likelihood for an inference model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_marg_lik(
  fasta_filename,
  inference_model = beautier::create_ns_inference_model(),
  beast2_options = beastier::create_mcbette_beast2_options(),
  os = rappdirs::app_dir()$os
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est_marg_lik_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>name of the FASTA file</p>
</td></tr>
<tr><td><code id="est_marg_lik_+3A_inference_model">inference_model</code></td>
<td>
<p>an inference model,
as can be created by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="est_marg_lik_+3A_beast2_options">beast2_options</code></td>
<td>
<p>a <code>beast2_options</code> structure,
as can be created by <a href="beastier.html#topic+create_mcbette_beast2_options">create_mcbette_beast2_options</a>.</p>
</td></tr>
<tr><td><code id="est_marg_lik_+3A_os">os</code></td>
<td>
<p>name of the operating system,
must be <code>unix</code> (Linux, Mac) or <code>win</code> (Windows)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="base.html#topic+list">list</a> showing the estimated marginal likelihoods
(and its estimated error), its items are::
</p>

<ul>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>esses</code> the Effective Sample Size
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="#topic+can_run_mcbette">can_run_mcbette</a>: see if 'mcbette' can run
</p>
</li>
<li> <p><a href="#topic+est_marg_liks">est_marg_liks</a>: estimate multiple marginal likelihoods
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (can_run_mcbette()) {

  # An example FASTA file
  fasta_filename &lt;- system.file("extdata", "simple.fas", package = "mcbette")

  # A testing inference model with inaccurate (thus fast) marginal
  # likelihood estimation
  inference_model &lt;- beautier::create_ns_inference_model()

  # Shorten the run, by doing a short (dirty, unreliable) MCMC
  inference_model$mcmc &lt;- beautier::create_test_ns_mcmc()

  # Setup the options for BEAST2 to be able to call BEAST2 packages
  beast2_options &lt;- beastier::create_mcbette_beast2_options()

  # Estimate the marginal likelihood
  est_marg_lik(
    fasta_filename = fasta_filename,
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  beastier::remove_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='est_marg_liks'>Estimate the marginal likelihoods for one or more inference models</h2><span id='topic+est_marg_liks'></span>

<h3>Description</h3>

<p>Estimate the marginal likelihoods (aka evidence)
for one or more inference models, based on a single alignment.
Also, the marginal likelihoods are compared, resulting in a
relative weight for each model, where a relative weight of a model
close to <code>1.0</code> means that that model is way likelier than
the others.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>est_marg_liks(
  fasta_filename,
  inference_models = list(beautier::create_inference_model(mcmc =
    beautier::create_ns_mcmc())),
  beast2_optionses = rep(list(beastier::create_mcbette_beast2_options()), times =
    length(inference_models)),
  verbose = FALSE,
  os = rappdirs::app_dir()$os
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="est_marg_liks_+3A_fasta_filename">fasta_filename</code></td>
<td>
<p>name of the FASTA file</p>
</td></tr>
<tr><td><code id="est_marg_liks_+3A_inference_models">inference_models</code></td>
<td>
<p>a list of one or more inference models,
as can be created by <a href="beautier.html#topic+create_inference_model">create_inference_model</a></p>
</td></tr>
<tr><td><code id="est_marg_liks_+3A_beast2_optionses">beast2_optionses</code></td>
<td>
<p>list of one or more <code>beast2_options</code>
structures,
as can be created by <a href="beastier.html#topic+create_mcbette_beast2_options">create_mcbette_beast2_options</a>.
Use of reduplicated plural to achieve difference with
<code>beast2_options</code></p>
</td></tr>
<tr><td><code id="est_marg_liks_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE show debug output</p>
</td></tr>
<tr><td><code id="est_marg_liks_+3A_os">os</code></td>
<td>
<p>name of the operating system,
must be <code>unix</code> (Linux, Mac) or <code>win</code> (Windows)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In the process, multiple (temporary) files are created (where
<code>[x]</code> denotes the index in a list)
</p>

<ul>
<li> <p><code>beast2_optionses[x]$input_filename</code>
path to the the BEAST2 XML input file
</p>
</li>
<li> <p><code>beast2_optionses[x]$output_state_filename</code>
path to the BEAST2 XML state file
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$tracelog$filename</code>
path to the BEAST2 trace file with parameter estimates
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$treelog$filename</code>
path to the BEAST2 <code>trees</code> file with the posterior trees
</p>
</li>
<li> <p><code>inference_models[x]$mcmc$screenlog$filename</code>
path to the BEAST2 screen output file
</p>
</li></ul>

<p>These file can be deleted manually by <a href="babette.html#topic+bbt_delete_temp_files">bbt_delete_temp_files</a>,
else these will be deleted automatically by the operating system.
</p>


<h3>Value</h3>

<p>a <a href="base.html#topic+data.frame">data.frame</a> showing the estimated marginal likelihoods
(and its estimated error) per combination of models. Columns are:
</p>

<ul>
<li> <p><code>site_model_name</code>: name of the site model
</p>
</li>
<li> <p><code>clock_model_name</code>: name of the clock model
</p>
</li>
<li> <p><code>tree_prior_name</code>: name of the tree prior
</p>
</li>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>weight</code>: relative model weight, a value from 1.0 (all
evidence is in favor of this model combination) to 0.0 (no
evidence in favor of this model combination)
</p>
</li>
<li> <p><code>ess</code>: effective sample size of the marginal likelihood
estimation
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>


<ul>
<li> <p><a href="#topic+can_run_mcbette">can_run_mcbette</a>: see if 'mcbette' can run
</p>
</li>
<li> <p><a href="#topic+est_marg_liks">est_marg_liks</a>: estimate multiple marginal likelihood of a
single inference mode
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>if (can_run_mcbette()) {

  # Use an example FASTA file
  fasta_filename &lt;- system.file("extdata", "simple.fas", package = "mcbette")

  # Create two inference models
  inference_model_1 &lt;- beautier::create_ns_inference_model(
    site_model = beautier::create_jc69_site_model()
  )
  inference_model_2 &lt;- beautier::create_ns_inference_model(
    site_model = beautier::create_hky_site_model()
  )

  # Shorten the run, by doing a short (dirty, unreliable) MCMC
  inference_model_1$mcmc &lt;- beautier::create_test_ns_mcmc()
  inference_model_2$mcmc &lt;- beautier::create_test_ns_mcmc()

  # Combine the inference models
  inference_models &lt;- list(inference_model_1, inference_model_2)

  # Create the BEAST2 options, that will write the output
  # to different (temporary) filanems
  beast2_options_1 &lt;- beastier::create_mcbette_beast2_options()
  beast2_options_2 &lt;- beastier::create_mcbette_beast2_options()

  # Combine the two BEAST2 options sets,
  # use reduplicated plural
  beast2_optionses &lt;- list(beast2_options_1, beast2_options_2)

  # Compare the models
  marg_liks &lt;- est_marg_liks(
    fasta_filename,
    inference_models = inference_models,
    beast2_optionses = beast2_optionses
  )

  # Interpret the results
  interpret_marg_lik_estimates(marg_liks)

  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>

<hr>
<h2 id='get_mcbette_state'>Get the current state of <a href="#topic+mcbette">mcbette</a></h2><span id='topic+get_mcbette_state'></span>

<h3>Description</h3>

<p>Get the current state of <a href="#topic+mcbette">mcbette</a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mcbette_state(beast2_folder = beastier::get_default_beast2_folder())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mcbette_state_+3A_beast2_folder">beast2_folder</code></td>
<td>
<p>the folder where the BEAST2 is installed.
Note that this is not the folder where the BEAST2 executable is installed:
the BEAST2 executable is in a subfolder.
Use <a href="beastier.html#topic+get_default_beast2_folder">get_default_beast2_folder</a>
to get the default BEAST2 folder.
Use <a href="beastier.html#topic+get_default_beast2_bin_path">get_default_beast2_bin_path</a>
to get the full path to the default BEAST2 executable.
Use <a href="beastier.html#topic+get_default_beast2_jar_path">get_default_beast2_jar_path</a>
to get the full path to the default BEAST2 jar file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="base.html#topic+list">list</a> with the following elements:
</p>

<ul>
<li>
<p>beast2_installed
<a href="base.html#topic+TRUE">TRUE</a> if BEAST2 is installed,
<a href="base.html#topic+FALSE">FALSE</a> otherwise

</p>
</li>
<li>
<p>ns_installed
<a href="base.html#topic+TRUE">TRUE</a> if the BEAST2 NS package is installed
<a href="base.html#topic+FALSE">FALSE</a> if the BEAST2 or the BEAST2 NS package is not installed

</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>get_mcbette_state()

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='get_test_marg_liks'>Get testing <code>marg_liks</code></h2><span id='topic+get_test_marg_liks'></span>

<h3>Description</h3>

<p>Get testing <code>marg_liks</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_test_marg_liks()
</code></pre>


<h3>Value</h3>

<p>A data frame with marginal likelihoods.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_test_marg_liks()

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='interpret_bayes_factor'>Interpret a Bayes factor</h2><span id='topic+interpret_bayes_factor'></span>

<h3>Description</h3>

<p>Interpret a Bayes factor, using the interpretation from [1].
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interpret_bayes_factor(bayes_factor)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="interpret_bayes_factor_+3A_bayes_factor">bayes_factor</code></td>
<td>
<p>Bayes factor to be interpreted</p>
</td></tr>
</table>


<h3>Details</h3>


<ul>
<li><p> [1] H. Jeffreys (1961). The Theory of Probability (3rd ed.).
Oxford. p. 432
</p>
</li></ul>



<h3>Value</h3>

<p>a string with the interpretation in English
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>interpret_bayes_factor(0.5)

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='interpret_marg_lik_estimates'>Interpret the marginal likelihood estimates</h2><span id='topic+interpret_marg_lik_estimates'></span>

<h3>Description</h3>

<p>Interpret the marginal likelihood estimates
as created by <a href="#topic+est_marg_liks">est_marg_liks</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>interpret_marg_lik_estimates(marg_liks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="interpret_marg_lik_estimates_+3A_marg_liks">marg_liks</code></td>
<td>
<p>a table of (estimated) marginal likelihoods,
as, for example, created by <a href="#topic+est_marg_liks">est_marg_liks</a>.
This <a href="base.html#topic+data.frame">data.frame</a> has the following columns:
</p>

<ul>
<li> <p><code>site_model_name</code>: name of the site model,
must be an element of <a href="beautier.html#topic+get_site_model_names">get_site_model_names</a>
</p>
</li>
<li> <p><code>clock_model_name</code>: name of the clock model,
must be an element of <a href="beautier.html#topic+get_clock_model_names">get_clock_model_names</a>
</p>
</li>
<li> <p><code>tree_prior_name</code>: name of the tree prior,
must be an element of <a href="beautier.html#topic+get_tree_prior_names">get_tree_prior_names</a>
</p>
</li>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>weight</code>: relative model weight, a value from 1.0 (all
evidence is in favor of this model combination) to 0.0 (no
evidence in favor of this model combination)
</p>
</li>
<li> <p><code>ess</code>: effective sample size of the marginal likelihood
estimation
</p>
</li></ul>

<p>Use <a href="#topic+get_test_marg_liks">get_test_marg_liks</a> to get a test <code>marg_liks</code>.
Use <a href="#topic+is_marg_liks">is_marg_liks</a> to determine if a <code>marg_liks</code> is valid.
Use <a href="#topic+check_marg_liks">check_marg_liks</a> to check that a <code>marg_liks</code> is valid.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. This function shows the interpretation
via <a href="base.html#topic+message">message</a>.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='is_marg_liks'>Determine if the <code>marg_liks</code> is valid</h2><span id='topic+is_marg_liks'></span>

<h3>Description</h3>

<p>Determine if the <code>marg_liks</code> is valid
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_marg_liks(marg_liks, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_marg_liks_+3A_marg_liks">marg_liks</code></td>
<td>
<p>a table of (estimated) marginal likelihoods,
as, for example, created by <a href="#topic+est_marg_liks">est_marg_liks</a>.
This <a href="base.html#topic+data.frame">data.frame</a> has the following columns:
</p>

<ul>
<li> <p><code>site_model_name</code>: name of the site model,
must be an element of <a href="beautier.html#topic+get_site_model_names">get_site_model_names</a>
</p>
</li>
<li> <p><code>clock_model_name</code>: name of the clock model,
must be an element of <a href="beautier.html#topic+get_clock_model_names">get_clock_model_names</a>
</p>
</li>
<li> <p><code>tree_prior_name</code>: name of the tree prior,
must be an element of <a href="beautier.html#topic+get_tree_prior_names">get_tree_prior_names</a>
</p>
</li>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>weight</code>: relative model weight, a value from 1.0 (all
evidence is in favor of this model combination) to 0.0 (no
evidence in favor of this model combination)
</p>
</li>
<li> <p><code>ess</code>: effective sample size of the marginal likelihood
estimation
</p>
</li></ul>

<p>Use <a href="#topic+get_test_marg_liks">get_test_marg_liks</a> to get a test <code>marg_liks</code>.
Use <a href="#topic+is_marg_liks">is_marg_liks</a> to determine if a <code>marg_liks</code> is valid.
Use <a href="#topic+check_marg_liks">check_marg_liks</a> to check that a <code>marg_liks</code> is valid.</p>
</td></tr>
<tr><td><code id="is_marg_liks_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE show debug output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the argument is a valid <code>marg_liks</code>,
FALSE otherwise
</p>

<hr>
<h2 id='mcbette_report'>Create a <a href="#topic+mcbette">mcbette</a> report,
to be used when reporting bugs</h2><span id='topic+mcbette_report'></span>

<h3>Description</h3>

<p>Create a <a href="#topic+mcbette">mcbette</a> report,
to be used when reporting bugs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcbette_report(beast2_folder = beastier::get_default_beast2_folder())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcbette_report_+3A_beast2_folder">beast2_folder</code></td>
<td>
<p>the folder where the BEAST2 is installed.
Note that this is not the folder where the BEAST2 executable is installed:
the BEAST2 executable is in a subfolder.
Use <a href="beastier.html#topic+get_default_beast2_folder">get_default_beast2_folder</a>
to get the default BEAST2 folder.
Use <a href="beastier.html#topic+get_default_beast2_bin_path">get_default_beast2_bin_path</a>
to get the full path to the default BEAST2 executable.
Use <a href="beastier.html#topic+get_default_beast2_jar_path">get_default_beast2_jar_path</a>
to get the full path to the default BEAST2 jar file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>nothing. It is intended that the output (not
the return value) is copy-pasted from screen.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(beautier::is_on_ci()) {
  mcbette_report()
}
</code></pre>

<hr>
<h2 id='mcbette_self_test'>Performs a minimal <a href="#topic+mcbette">mcbette</a> run</h2><span id='topic+mcbette_self_test'></span>

<h3>Description</h3>

<p>Performs a minimal <a href="#topic+mcbette">mcbette</a> run
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcbette_self_test(beast2_folder = beastier::get_default_beast2_folder())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcbette_self_test_+3A_beast2_folder">beast2_folder</code></td>
<td>
<p>the folder where the BEAST2 is installed.
Note that this is not the folder where the BEAST2 executable is installed:
the BEAST2 executable is in a subfolder.
Use <a href="beastier.html#topic+get_default_beast2_folder">get_default_beast2_folder</a>
to get the default BEAST2 folder.
Use <a href="beastier.html#topic+get_default_beast2_bin_path">get_default_beast2_bin_path</a>
to get the full path to the default BEAST2 executable.
Use <a href="beastier.html#topic+get_default_beast2_jar_path">get_default_beast2_jar_path</a>
to get the full path to the default BEAST2 jar file.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Will <a href="base.html#topic+stop">stop</a> is 'mcbette' cannot run on a minimal,
standard input.
</p>

<hr>
<h2 id='plot_marg_liks'>Plot the <code>marg_liks</code></h2><span id='topic+plot_marg_liks'></span>

<h3>Description</h3>

<p>Plot the <code>marg_liks</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_marg_liks(marg_liks)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_marg_liks_+3A_marg_liks">marg_liks</code></td>
<td>
<p>a table of (estimated) marginal likelihoods,
as, for example, created by <a href="#topic+est_marg_liks">est_marg_liks</a>.
This <a href="base.html#topic+data.frame">data.frame</a> has the following columns:
</p>

<ul>
<li> <p><code>site_model_name</code>: name of the site model,
must be an element of <a href="beautier.html#topic+get_site_model_names">get_site_model_names</a>
</p>
</li>
<li> <p><code>clock_model_name</code>: name of the clock model,
must be an element of <a href="beautier.html#topic+get_clock_model_names">get_clock_model_names</a>
</p>
</li>
<li> <p><code>tree_prior_name</code>: name of the tree prior,
must be an element of <a href="beautier.html#topic+get_tree_prior_names">get_tree_prior_names</a>
</p>
</li>
<li> <p><code>marg_log_lik</code>: estimated marginal (natural) log likelihood
</p>
</li>
<li> <p><code>marg_log_lik_sd</code>: estimated error of <code>marg_log_lik</code>
</p>
</li>
<li> <p><code>weight</code>: relative model weight, a value from 1.0 (all
evidence is in favor of this model combination) to 0.0 (no
evidence in favor of this model combination)
</p>
</li>
<li> <p><code>ess</code>: effective sample size of the marginal likelihood
estimation
</p>
</li></ul>

<p>Use <a href="#topic+get_test_marg_liks">get_test_marg_liks</a> to get a test <code>marg_liks</code>.
Use <a href="#topic+is_marg_liks">is_marg_liks</a> to determine if a <code>marg_liks</code> is valid.
Use <a href="#topic+check_marg_liks">check_marg_liks</a> to check that a <code>marg_liks</code> is valid.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <a href="ggplot2.html#topic+ggplot">ggplot</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>plot_marg_liks(get_test_marg_liks())

beastier::remove_beaustier_folders()
beastier::check_empty_beaustier_folders()
</code></pre>

<hr>
<h2 id='set_mcbette_state'>Set the <a href="#topic+mcbette">mcbette</a> state.</h2><span id='topic+set_mcbette_state'></span>

<h3>Description</h3>

<p>Set the <a href="#topic+mcbette">mcbette</a> state to having BEAST2 installed with
or without installing the BEAST2 NS package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_mcbette_state(
  mcbette_state,
  beast2_folder = beastier::get_default_beast2_folder(),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_mcbette_state_+3A_mcbette_state">mcbette_state</code></td>
<td>
<p>the <a href="#topic+mcbette">mcbette</a> state,
which is a <a href="base.html#topic+list">list</a> with the following elements:
</p>

<ul>
<li>
<p>beast2_installed
<a href="base.html#topic+TRUE">TRUE</a> if BEAST2 is installed,
<a href="base.html#topic+FALSE">FALSE</a> otherwise

</p>
</li>
<li>
<p>ns_installed
<a href="base.html#topic+NA">NA</a> if BEAST2 is not installed.
<a href="base.html#topic+TRUE">TRUE</a> if the BEAST2 NS package is installed
<a href="base.html#topic+FALSE">FALSE</a> if the BEAST2 NS package is not installed

</p>
</li></ul>
</td></tr>
<tr><td><code id="set_mcbette_state_+3A_beast2_folder">beast2_folder</code></td>
<td>
<p>the folder where the BEAST2 is installed.
Note that this is not the folder where the BEAST2 executable is installed:
the BEAST2 executable is in a subfolder.
Use <a href="beastier.html#topic+get_default_beast2_folder">get_default_beast2_folder</a>
to get the default BEAST2 folder.
Use <a href="beastier.html#topic+get_default_beast2_bin_path">get_default_beast2_bin_path</a>
to get the full path to the default BEAST2 executable.
Use <a href="beastier.html#topic+get_default_beast2_jar_path">get_default_beast2_jar_path</a>
to get the full path to the default BEAST2 jar file.</p>
</td></tr>
<tr><td><code id="set_mcbette_state_+3A_verbose">verbose</code></td>
<td>
<p>if TRUE show debug output</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Note</h3>

<p>In newer versions of BEAST2, BEAST2 comes pre-installed with the
BEAST2 NS package. For such a version, one cannot install BEAST2
without NS. A warning will be issues if one intends to only install
BEAST2 (i.e. without the BEAST2 NS package) and gets the BEAST2
NS package installed as a side effect as well.
</p>
<p>Also, installing or uninstalling a BEAST2 package from a BEAST2
installation will affect all installations.
</p>


<h3>See Also</h3>


<ul>
<li><p> Use <a href="#topic+get_mcbette_state">get_mcbette_state</a> to
get the current <a href="#topic+mcbette">mcbette</a> state
</p>
</li>
<li><p> Use <a href="#topic+check_mcbette_state">check_mcbette_state</a> to
check the current <a href="#topic+mcbette">mcbette</a> state
</p>
</li></ul>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
