<!DOCTYPE html><html><head><title>Help for package MCMC.OTU</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MCMC.OTU}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MCMC.OTU-package'>
<p>Bayesian analysis of multivariate counts data in DNA metabarcoding and ecology.</p></a></li>
<li><a href='#getNormalizedOTUdata'>
<p>Extracts mcmc.otu model predictions</p></a></li>
<li><a href='#green.data'>
<p>Symbiodinium sp. ITS2 OTUs from Orbicella franksi and O. faveolata</p></a></li>
<li><a href='#logLin'>
<p>prepares OTU counts data for PCA analysis using log-linear-hybrid transform</p></a></li>
<li><a href='#mcmc.otu'>
<p>Analyzes multivariate counts data using poisson-lognormal mixed model</p></a></li>
<li><a href='#mcmc.pval'>
<p>calculates p-value based on Bayesian z-score or MCMC sampling</p></a></li>
<li><a href='#otuByAutocorr'>
<p>Selects OTUs for which MCMC-based parameter estimates are reliable.</p></a></li>
<li><a href='#otuStack'>
<p>Prepares OTU counts data for MCMC model fitting using mcmc.otu().</p></a></li>
<li><a href='#OTUsummary'>
<p>Summarizes and plots results of mcmc.otu() function series.</p></a></li>
<li><a href='#padjustOTU'>
<p>Adjusts p-values in the OTU summary for multiple comparisons.</p></a></li>
<li><a href='#panel.cor'>
<p>accessory function for pairs() to display Pearson correlations</p></a></li>
<li><a href='#panel.cor.pval'>
<p>accessory function for pairs() to display pvalue of the Pearson correlation</p></a></li>
<li><a href='#purgeOutliers'>
<p>Removes outlier samples and OTUs.</p></a></li>
<li><a href='#signifOTU'>
<p>Finds differentially represented OTUs.</p></a></li>
<li><a href='#startedLog'>
<p>prepares OTU counts data for PCA analysis using started-log transform</p></a></li>
<li><a href='#summaryPlotOTU'>
<p>Wrapper function for ggplot2 to make bar and line graphs of</p>
mcmc.otu() results</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Analysis of Multivariate Counts Data in DNA
Metabarcoding and Ecology</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.10</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-02-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Mikhail V. Matz</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Mikhail V. Matz &lt;matz@utexas.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Poisson-lognormal generalized linear mixed model analysis of multivariate counts data using MCMC, aiming to infer the changes in relative proportions of individual variables. The package was originally designed for sequence-based analysis of microbial communities ("metabarcoding", variables = operational taxonomic units, OTUs), but can be used for other types of multivariate counts, such as in ecological applications (variables = species). The results are summarized and plotted using 'ggplot2' functions. Includes functions to remove sample and variable outliers and reformat counts into normalized log-transformed values for correlation and principal component/coordinate analysis. Walkthrough and examples: http://www.bio.utexas.edu/research/matz_lab/matzlab/Methods_files/walkthroughExample_mcmcOTU_R.txt. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>MCMCglmm,ggplot2,coda</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-02-11 19:23:29 UTC; c-monstr</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-02-12 00:53:04</td>
</tr>
</table>
<hr>
<h2 id='MCMC.OTU-package'>
Bayesian analysis of multivariate counts data in DNA metabarcoding and ecology.
</h2><span id='topic+MCMC.OTU-package'></span><span id='topic+MCMC.OTU'></span>

<h3>Description</h3>

<p>This package enables MCMC-based generalized linear mixed model analysis of multivariate counts data, such as common in DNA metabarcoding and community ecology. The results are summarized and plotted using ggplot2 functions.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> MCMC.OTU</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.10</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-02-10</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-3</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>At the moment, the package handles experimental design with a single multilevel
fixed factor or two fully crossed multilevel fixed factors. Any number of scalar (OTU-specific)
random factors (i.e. blocking factors) are allowed. By default, it is assumed that variation in total counts per sample is not biologically relevant (refects sequencing or survey effort). See help for the core function mcmc.otu() for more details. 
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz &lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Quantifying cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 2:e386 https://peerj.com/articles/386/
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Symbiodinium sp diversity in two coral species at two reefs (banks)
data(green.data)

# removing outliers
goods=purgeOutliers(
	data=green.data,
	count.columns=c(4:length(green.data[1,])),
	zero.cut=0.25 # remove this line for real analysis
)

# stacking the data table
gs=otuStack(
	data=goods,
	count.columns=c(4:length(goods[1,])),
	condition.columns=c(1:3)
	)

# fitting the model
mm=mcmc.otu(
	fixed="bank+species+bank:species",
	data=gs,
	nitt=3000,burnin=2000 # remove this line for real analysis!
	)

# selecting the OTUs that were modeled reliably
acpass=otuByAutocorr(mm,gs)

# calculating effect sizes and p-values:
ss=OTUsummary(mm,gs,summ.plot=FALSE)

# correcting for mutliple comparisons (FDR)
ss=padjustOTU(ss)

# getting significatly changing OTUs (FDR&lt;0.05)
sigs=signifOTU(ss)

# plotting them
ss2=OTUsummary(mm,gs,otus=sigs)

# bar-whiskers graph of relative changes:
# ssr=OTUsummary(mm,gs,otus=signifOTU(ss),relative=TRUE)

# displaying effect sizes and p-values for significant OTUs
ss$otuWise[sigs]

</code></pre>

<hr>
<h2 id='getNormalizedOTUdata'>
Extracts mcmc.otu model predictions
</h2><span id='topic+getNormalizedOTUdata'></span>

<h3>Description</h3>

<p>Generates a table of model-derived log2-transformed OTU abundances 
without sample effects (i.e., corresponding to predicted normalized data)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getNormalizedOTUdata(model, data, log.base=10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getNormalizedOTUdata_+3A_model">model</code></td>
<td>

<p>mcmc.otu model: the output of mcmc.otu function fitted with pr=TRUE option 
</p>
</td></tr>
<tr><td><code id="getNormalizedOTUdata_+3A_data">data</code></td>
<td>

<p>The dataset that was analysed to generate the model (output of otuStack function)
</p>
</td></tr>
<tr><td><code id="getNormalizedOTUdata_+3A_log.base">log.base</code></td>
<td>

<p>base of the logarithm to use
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The function returns a list of two data frames. The first one, normData, is the model-predicted log2-transformed OTU abundances table. It has one column per OTU and one row per sample. The second data frame, conditions, is a table of experimental conditions corresponding to the normData table.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='green.data'>
Symbiodinium sp. ITS2 OTUs from Orbicella franksi and O. faveolata 
</h2><span id='topic+green.data'></span>

<h3>Description</h3>

<p>Symbiodinium sp. ITS2 OTUs from Orbicella franksi and O. faveolata corals at the Flower Garden Banks. Two banks (east and west), two coral species, five OTUs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(green.data)</code></pre>


<h3>Format</h3>

<p>A data frame with 58 observations on the following 156 variables.
</p>

<dl>
<dt><code>species</code></dt><dd><p>a factor with levels <code>faveolata</code> <code>franksi</code></p>
</dd>
<dt><code>bank</code></dt><dd><p>a factor with levels <code>east</code> <code>west</code></p>
</dd>
<dt><code>sample</code></dt><dd><p>a factor with levels <code>EFAV117</code> <code>EFAV123</code> <code>EFAV128</code> <code>EFAV137</code> <code>EFAV140</code> <code>EFAV142</code> <code>EFAV143</code> <code>EFAV145</code> <code>EFAV150</code> <code>EFAV153</code> <code>EFAV154</code> <code>EFAV157</code> <code>EFAV161</code> <code>EFAV162</code> <code>EFRA101</code> <code>EFRA102</code> <code>EFRA105</code> <code>EFRA216</code> <code>EFRA218</code> <code>EFRA221</code> <code>EFRA222</code> <code>EFRA224</code> <code>EFRA231</code> <code>EFRA232</code> <code>EFRA233</code> <code>EFRA236</code> <code>EFRA82</code> <code>EFRA90</code> <code>EFRA91</code> <code>WFAV102</code> <code>WFAV103</code> <code>WFAV107</code> <code>WFAV111</code> <code>WFAV112</code> <code>WFAV188</code> <code>WFAV191</code> <code>WFAV89</code> <code>WFAV91</code> <code>WFAV94</code> <code>WFAV95</code> <code>WFAV97</code> <code>WFAV98</code> <code>WFAV99</code> <code>WFRA106</code> <code>WFRA107</code> <code>WFRA112</code> <code>WFRA114</code> <code>WFRA115</code> <code>WFRA116</code> <code>WFRA117</code> <code>WFRA198</code> <code>WFRA199</code> <code>WFRA200</code> <code>WFRA201</code> <code>WFRA202</code> <code>WFRA203</code> <code>WFRA205</code> <code>WFRA210</code></p>
</dd>
<dt><code>H35JRAZ01A0C9J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A0D1B</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A0RQO</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A1FKL</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A1GT3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A1NMH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A1SO3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A1VUD</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A22NW</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A2ZJK</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A2ZYY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A30IE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A32BS</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A3DXY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A3UJS</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A4BHG</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A4F3O</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A4G6O</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A4O8A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A4U5O</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A5LTB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A5O9U</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A6ITE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A771F</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A7EP4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A7SMY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A8A59</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A8X2Y</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A93HV</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01A9TZY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AMZP0</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AN7JF</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ANHED</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ANUPT</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AO7GS</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AON2S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01APP5A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01APPG5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01APWEH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01APYLM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AQIU4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AR1R8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AR8Q4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ARD6M</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ARKYW</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ARSTD</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ARZ7L</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AS8ZF</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ASV28</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ATSK2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01ATZCY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AU38T</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AUMOE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AUTH5</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AUX8X</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AVB0Q</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AVFAX</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AVX70</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AWGQC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AXA3Z</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AXGLD</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AXVIR</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AY2DE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AY6KB</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AZFWS</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01AZTST</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BB9FU</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BBITP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BBSV6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BD2HV</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BDBD6</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BDTVF</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BE95X</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BEHEK</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BEMYI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BEP0J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BFNRT</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BFVYJ</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BG87N</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BGG1A</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BGN2J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BGUP9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BH7VV</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ01BIHI3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C08W3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C0K1K</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C141R</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C16B4</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C1INY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C1WD2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C2344</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C2919</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C2CBI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C2DMJ</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C2MH3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C2PQY</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C32WG</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C3ANP</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C5910</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C5CKR</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C5G30</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C60I8</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C6AF1</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C6E0W</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C6LU3</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C6Q3V</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C75OM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C7GCE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C7P6J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C8JJT</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03C9MVI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03CX46J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03CX5BC</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03CXVYS</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DA80G</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DALPN</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DAW5J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DAXEI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DC5UZ</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DCCVA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DCGCE</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DCXOX</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DE7WW</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DED1S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DEJ2S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DF9NA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DFE9W</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DFZXI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DG0DK</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DGHF2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DHRCR</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DIOTH</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DJAMF</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DJLV9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DJU1V</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DKE4J</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DKRO9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DLA3U</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DLJW2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DLRGI</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DMDRM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DMGA2</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DMGAO</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DO292</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DOWHJ</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DP0DM</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DQ55S</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DQK7F</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DR9O9</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DRJ6E</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DSA3E</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DSJQA</code></dt><dd><p>a numeric vector</p>
</dd>
<dt><code>H35JRAZ03DSR5L</code></dt><dd><p>a numeric vector</p>
</dd>
</dl>



<h3>Source</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='logLin'>
prepares OTU counts data for PCA analysis using log-linear-hybrid transform
</h2><span id='topic+logLin'></span>

<h3>Description</h3>

<p>Normalizes counts based on total counts per sample, then uses log-linear-hybrid
transformation on the data. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>logLin(data,count.columns,k=10,zero.na=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="logLin_+3A_data">data</code></td>
<td>

<p>OTU counts table. Rows: samples, columns: OTUs.
</p>
</td></tr>
<tr><td><code id="logLin_+3A_count.columns">count.columns</code></td>
<td>

<p>numbers of colums containing OTU counts.
</p>
</td></tr>
<tr><td><code id="logLin_+3A_k">k</code></td>
<td>

<p>count threshold to transition from linear to log transform.
</p>
</td></tr>
<tr><td><code id="logLin_+3A_zero.na">zero.na</code></td>
<td>

<p>With 'zero.na=TRUE', the original zero count datapoints become 'NA' 
(useful for analysis of pairwise correlations between OTUs).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table of transformed normalized values.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Log-linear-hybrid transform: 
Rocke DM and Durbin B Bioinformatics 2003 19:966-972 (DOI: 10.1093/bioinformatics/btg107)
MCMC.OTU:
Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='mcmc.otu'>
Analyzes multivariate counts data using poisson-lognormal mixed model
</h2><span id='topic+mcmc.otu'></span>

<h3>Description</h3>

<p>Wrapper function for MCMCglmm by Jarrod Hadfield, designed for multivariate counts data
such as in sequence-based analysis of microbial communities (&quot;metabarcoding&quot;, variables = 
operational taxonomic units, OTUs), or in ecological applications (variables = species). 
The function aims to infer the changes in relative proportions of individual variables.
The maximum number of variables that can be processed on a laptop computer is about 200; 
more memory is required for larger numbers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmc.otu(fixed=NULL, random=NULL, data, y.scale="proportion", 
globalMainEffects="remove", vprior="uninf",...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcmc.otu_+3A_fixed">fixed</code></td>
<td>

<p>combination of fixed effects, as a text string. Do not use &quot;*&quot; symbol, list
it fully, such as: 'factor1+factor2+factor1:factor2'. 
</p>
</td></tr>
<tr><td><code id="mcmc.otu_+3A_random">random</code></td>
<td>

<p>A vector of names for variable-specific scalar random effects, such as 'c(&quot;effect1&quot;,&quot;effect2&quot;)'.
</p>
</td></tr>
<tr><td><code id="mcmc.otu_+3A_data">data</code></td>
<td>

<p>output of the otuStack() function
</p>
</td></tr>
<tr><td><code id="mcmc.otu_+3A_y.scale">y.scale</code></td>
<td>

<p>By default, the modeled abundances will be expressed relative to the total
counts in the sample, effectively corresponding to proportions of total. Specify 'y.scale=&quot;absolute&quot;' to express the results as absolute abundances. 
</p>
</td></tr>
<tr><td><code id="mcmc.otu_+3A_globalmaineffects">globalMainEffects</code></td>
<td>

<p>By default, the model will assume that the samples can vary systematically in abundance among factor combinations (i.e., there is an effect of a factor combination applicable to all variables) and remove these effects; this is analogous to normalizing the samples to total counts. Specify 'globalMainEffects=&quot;keep&quot;' to switch this off.  
</p>
</td></tr>
<tr><td><code id="mcmc.otu_+3A_vprior">vprior</code></td>
<td>

<p>Prior for variance of user-specified random effects. By default an inverse Wishart prior with assumed variance (V) at 1 and the degree of belief parameter (nu) at 0. With 'prior=&quot;iw&quot;' and 'prior=&quot;iw01&quot;' nu is the number of OTUs minus 0.998, resulting in a weakly informative prior that is commonly used in this type of inference. vprior=&quot;iw&quot; will assume large prior variance (1), vprior=&quot;iw01&quot; will assume small prior variance (0.1). If the model has trouble converging, specify vptior=&quot;iw&quot;.
</p>
</td></tr>
<tr><td><code id="mcmc.otu_+3A_...">...</code></td>
<td>

<p>other options for MCMCglmm function, such as nitt (number of iterations), thin (tinning interval), and burnin (number of initial iterations to disregard). For a more precise inference, specify 'nitt=50000, thin=25, burnin=5000'. See MCMCglmm documentation for more details.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function constructs priors and runs an MCMC chain to fit a Poisson-lognormal generalized linear mixed model to the multivariate  counts data.
</p>
<p>The fixed effects for the model by default include a variable-specific intercept, global 
(non-variable-specific) main effects of fixed factors, and variable-specific effect 
for each of the listed fixed factors. With globalMainEffects=&quot;keep&quot; the model
will not include the global main effects, resulting in them being absorbed into
the variable-specific effects.
</p>
<p>The user-specified random effects are all assumed to be variable-specific with no covariances. 
</p>
<p>The model includes one universal random factor: the scalar random effect of sample,
which accounts for the unequal counting effort among samples. 
</p>
<p>Residual variances are assumed to be variable-specific with no covariances, with weakly informative 
inverse Wishart prior with variance=1 and nu=(number of variables)-0.998. 
</p>
<p>The priors for fixed effects are diffuse gaussians with a mean at 0 and very large variances (1e+8),  
</p>


<h3>Value</h3>

<p>An MCMCglmm object. OTUsummary() function within this package summaizes these data, 
calculates all variable-wise credible intervals and p-values, and plots the results either 
as line-point-whiskers graph or a bar-whiskers graph using ggplot2 functions. 
</p>
<p>OTUsummary() only works for experiments with a single multilevel factor or two fully 
crossed multilevel factors. 
</p>
<p>For more useful operations on MCMCglmm objects, such as posterior.mode(), HPDinterval(), and plot(), see documentation for MCMCglmm package.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>


<h3>See Also</h3>

<p>OTUsummary(),MCMCglmm()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>	
# Symbiodinium sp diversity in two coral species at two reefs (banks)
data(green.data)

# removing outliers
goods=purgeOutliers(
	data=green.data,
	count.columns=c(4:length(green.data[1,])),
	zero.cut=0.25 # remove this line for real analysis
)

# stacking the data table
gs=otuStack(
	data=goods,
	count.columns=c(4:length(goods[1,])),
	condition.columns=c(1:3)
	)

# fitting the model
mm=mcmc.otu(
	fixed="bank+species+bank:species",
	data=gs,
	nitt=3000,burnin=2000 # remove this line for real analysis!
	)

# selecting the OTUs that were modeled reliably
acpass=otuByAutocorr(mm,gs)

# calculating effect sizes and p-values:
ss=OTUsummary(mm,gs,summ.plot=FALSE)

# correcting for mutliple comparisons (FDR)
ss=padjustOTU(ss)

# getting significatly changing OTUs (FDR&lt;0.05)
sigs=signifOTU(ss)

# plotting them
ss2=OTUsummary(mm,gs,otus=sigs)

# bar-whiskers graph of relative changes:
# ssr=OTUsummary(mm,gs,otus=signifOTU(ss),relative=TRUE)

# displaying effect sizes and p-values for significant OTUs
ss$otuWise[sigs]

</code></pre>

<hr>
<h2 id='mcmc.pval'>
calculates p-value based on Bayesian z-score or MCMC sampling
</h2><span id='topic+mcmc.pval'></span>

<h3>Description</h3>

<p>Accessory function to OTUsummary function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcmc.pval(dat, testlim = 0, sided = 2, ptype="z")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mcmc.pval_+3A_dat">dat</code></td>
<td>

<p>a table of MCMC samples
</p>
</td></tr>
<tr><td><code id="mcmc.pval_+3A_testlim">testlim</code></td>
<td>

<p>positive value to test whether the posterior crosses it (for variance components)
</p>
</td></tr>
<tr><td><code id="mcmc.pval_+3A_sided">sided</code></td>
<td>

<p>sidedness of the test
</p>
</td></tr>
<tr><td><code id="mcmc.pval_+3A_ptype">ptype</code></td>
<td>

<p>'z' for z-score based p-values, 'mcmc' for standard sampling based p-values
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of calculated Bayesian p-values. For standard 'mcmc' p=values, the minimal possible value is 1/(MCMC sample size).
z-score based p-values are useful to approximate very low p-values with limited MCMC sample sizes.
</p>


<h3>Author(s)</h3>

<p>Mikhail Matz, UT Austin
&lt;matz@utexas.edu&gt;
</p>

<hr>
<h2 id='otuByAutocorr'>
Selects OTUs for which MCMC-based parameter estimates are reliable.
</h2><span id='topic+otuByAutocorr'></span>

<h3>Description</h3>

<p>Looks at autocorrelation among sampled parameter values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otuByAutocorr(model, data, ac.cut=0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="otuByAutocorr_+3A_model">model</code></td>
<td>

<p>Model generated by mcmc.otu()
</p>
</td></tr>
<tr><td><code id="otuByAutocorr_+3A_data">data</code></td>
<td>

<p>Dataset used to fit the model
</p>
</td></tr>
<tr><td><code id="otuByAutocorr_+3A_ac.cut">ac.cut</code></td>
<td>

<p>Autocorrelation cutoff
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of OTU names for which autocorrelation is less than ac.cut argument.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='otuStack'>
Prepares OTU counts data for MCMC model fitting using mcmc.otu().
</h2><span id='topic+otuStack'></span>

<h3>Description</h3>

<p>Stacks all OTU columns into one (count), creates a column for OTU names and
columns for experimental conditions. Adds a fake OTU called &quot;summ&quot; - sum of all
counts per sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otuStack(data, count.columns, condition.columns)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="otuStack_+3A_data">data</code></td>
<td>

<p>The input data table: one column per OTU, plus columns containg experimental 
conditions and the required column 'sample' denoting biological replicates.
See green.data included with the package.
</p>
</td></tr>
<tr><td><code id="otuStack_+3A_count.columns">count.columns</code></td>
<td>

<p>vector of numbers identifyng OTU counts columns
</p>
</td></tr>
<tr><td><code id="otuStack_+3A_condition.columns">condition.columns</code></td>
<td>

<p>vector of numbers identifyng columns containing experimental conditions
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset that can be fed into mcmc.otu() function.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Symbiodinium sp diversity in two coral species at two reefs (banks)
data(green.data)
green.data

# stacking the data table
gs=otuStack(green.data,count.columns=c(4:8),condition.columns=c(1:3))
head(gs,30)

</code></pre>

<hr>
<h2 id='OTUsummary'>
Summarizes and plots results of mcmc.otu() function series.
</h2><span id='topic+OTUsummary'></span>

<h3>Description</h3>

<p>Calculates abundances of each OTU across factor combinations;
calculates pairwise differences between all factor combinations
and their significances for each OTU;
plots results as bar or line graphs with credible intervals (ggplot2)
NOTE: only works for experiments involving a single multi-level fixed factor
or two fully crossed multi-level fixed factors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OTUsummary(model, data, otus = NA, relative = FALSE, 
log.base = 10, summ.plot = TRUE, ptype = "z", xgroup=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OTUsummary_+3A_model">model</code></td>
<td>

<p>Model generated by mcmc.otu() or mcmc.otu.normalized() 
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_data">data</code></td>
<td>

<p>Dataset used to build the model (returned by otuStack() or otuStackNormalize())
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_otus">otus</code></td>
<td>

<p>A vector of OTU names to summarize and plot. If left unspecified, all 
OTUs will be summarized.
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_relative">relative</code></td>
<td>

<p>Whether to plot OTU abundances as log(proportion of total) (default) or fold-
changes relative to the sample that is considered to be &quot;global control&quot; 
(relative = TRUE). The &quot;global control&quot; is the combination of factors 
that served as a reference during model fitting, either because it is 
alphanumerically first (that happens by default) or because it has been 
explicitly designated as such using relevel() function. 
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_log.base">log.base</code></td>
<td>

<p>Base of the logarithm to use.
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_summ.plot">summ.plot</code></td>
<td>

<p>By default, the function generates a summary plot, which is a line-points-95% 
credible intervals plot of log(fraction of total) with 'relative=FALSE' 
and a bar graph of log(fold change relative to the control), 
again with 95%  credible intervals, with 'relative=TRUE'. Specify 
'summ.plot=FALSE' if you don't want the summary plot.
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_ptype">ptype</code></td>
<td>

<p>Which type of p-values to use. By default p-values based on the Bayesian 
z-score are used. Specify 'ptype=&quot;mcmc&quot;' to output more conventional p-values 
based on MCMC sampling (these will be limited on the lower end by the size of 
MCMC sample).
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_xgroup">xgroup</code></td>
<td>

<p>For two-factor designs: which of the factors to use to form the x-axis. The 
other one will be used to form facets.
</p>
</td></tr>
<tr><td><code id="OTUsummary_+3A_...">...</code></td>
<td>

<p>Additional options for summaryPlotOTU() function. Among those, 'x.order' can be a 
vector specifying the order of factor levels on the x-axis.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three items:
</p>
<table>
<tr><td><code>summary</code></td>
<td>
<p> Summary table containing calculated abundances, their SD 
and 95% credible limits</p>
</td></tr>
<tr><td><code>otuWise</code></td>
<td>
<p>A series of matrices listing pairwise differences between 
factor combinations (upper triangle) and corresponding p-values (lower triangle)</p>
</td></tr>
<tr><td><code>ggPlot</code></td>
<td>
<p>the ggplot2 object for plotting. See http://docs.ggplot2.org/0.9.2.1/theme.html for ways 
to modify it, such as add text, rotate labels, change fonts, etc. </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>


<h3>See Also</h3>

<p>mcmc.otu(),MCMCglmm()
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# see example in ?MCMC.OTU

</code></pre>

<hr>
<h2 id='padjustOTU'>
Adjusts p-values in the OTU summary for multiple comparisons.
</h2><span id='topic+padjustOTU'></span>

<h3>Description</h3>

<p>Collects all p-values from the summary, performs FDR correction (Benjamini-Hochberg),
returns the summary with adjusted p-values.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>padjustOTU(otu.summary, method="BH")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="padjustOTU_+3A_otu.summary">otu.summary</code></td>
<td>

<p>The output of OTUsummary function.
</p>
</td></tr>
<tr><td><code id="padjustOTU_+3A_method">method</code></td>
<td>

<p>Method of adjustment for multiple comparisons (see ?p.adjust)
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An OTU summary list (see ?OTUsummary) with adjusted p-values
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='panel.cor'>
accessory function for pairs() to display Pearson correlations
</h2><span id='topic+panel.cor'></span>

<h3>Description</h3>

<p>works as upper.panel or lower.panel argument of pairs() (package graphics).  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel.cor(x, y, digits=2, cex.cor) 
  </code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panel.cor_+3A_x">x</code></td>
<td>

<p>x element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor_+3A_y">y</code></td>
<td>

<p>y element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor_+3A_digits">digits</code></td>
<td>

<p>number of non-zero digits to leave at the end
</p>
</td></tr>
<tr><td><code id="panel.cor_+3A_cex.cor">cex.cor</code></td>
<td>

<p>scaling factor for displayed text
</p>
</td></tr>
</table>


<h3>References</h3>

<p>cannibalized from an example in ?pairs (package graphics)
</p>

<hr>
<h2 id='panel.cor.pval'>
accessory function for pairs() to display pvalue of the Pearson correlation
</h2><span id='topic+panel.cor.pval'></span>

<h3>Description</h3>

<p>works as upper.panel or lower.panel argument of pairs() (package graphics). Displays pvalues better than 0.1.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>panel.cor.pval(x, y, digits = 2, cex.cor, p.cut=0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="panel.cor.pval_+3A_x">x</code></td>
<td>

<p>x element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_y">y</code></td>
<td>

<p>y element of the pairs() matrix
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_digits">digits</code></td>
<td>

<p>number of non-zero digits to leave at the end
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_cex.cor">cex.cor</code></td>
<td>

<p>scaling factor for displayed text
</p>
</td></tr>
<tr><td><code id="panel.cor.pval_+3A_p.cut">p.cut</code></td>
<td>

<p>p-value cutoff
</p>
</td></tr>
</table>


<h3>References</h3>

<p>cannibalized from an example in ?pairs (package graphics)
</p>

<hr>
<h2 id='purgeOutliers'>
Removes outlier samples and OTUs.
</h2><span id='topic+purgeOutliers'></span>

<h3>Description</h3>

<p>Samples are removed if their total counts fall below the specified z-score cutoff (default -2.5); OTUs are removed if their sum of counts is less than specified fraction of total counts (default 0.001), or if they are observed in less than specified fraction of all samples (default 0.2). Samples are removed first, then OTUs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>purgeOutliers(data, count.columns, sampleZcut=(-2.5), otu.cut=0.001, zero.cut=0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="purgeOutliers_+3A_data">data</code></td>
<td>

<p>A data frame containing counts, with samples as rows and OTUs as columns
</p>
</td></tr>
<tr><td><code id="purgeOutliers_+3A_count.columns">count.columns</code></td>
<td>

<p>Vector of OTU column numbers.
</p>
</td></tr>
<tr><td><code id="purgeOutliers_+3A_samplezcut">sampleZcut</code></td>
<td>

<p>z-score cutoff for removing low-coverage outlier samples.
</p>
</td></tr>
<tr><td><code id="purgeOutliers_+3A_otu.cut">otu.cut</code></td>
<td>

<p>cutoff for a fraction of total counts the OTU has to represent. This would depend on the methodology and nature of OTUs; for coral symbionts (Symbiodinium sp) the reasonable cutoff is 0.001 (Quigley et al, PLoS ONE 2014)
</p>
</td></tr>
<tr><td><code id="purgeOutliers_+3A_zero.cut">zero.cut</code></td>
<td>

<p>fraction of samples in which the OTU is found.  
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The data table with outlier samples and OTUs removed.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='signifOTU'>
Finds differentially represented OTUs.
</h2><span id='topic+signifOTU'></span>

<h3>Description</h3>

<p>Trawls otuWise summary tables in the data returned by OTUsummary
and returns the names of OTUs that have p-values better than 
the specified cutoff.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>signifOTU(otu.summary, p.cutoff=0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="signifOTU_+3A_otu.summary">otu.summary</code></td>
<td>

<p>The output of OTUsummary function.
</p>
</td></tr>
<tr><td><code id="signifOTU_+3A_p.cutoff">p.cutoff</code></td>
<td>

<p>p-value cutoff to report OTUs.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of significant OTUs.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='startedLog'>
prepares OTU counts data for PCA analysis using started-log transform
</h2><span id='topic+startedLog'></span>

<h3>Description</h3>

<p>Normalizes counts based on total counts per sample, adds a constant number
(argument logstart) to get rid of zeroes, then log-transforms the data.  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>startedLog(data,count.columns,logstart=0.1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="startedLog_+3A_data">data</code></td>
<td>

<p>OTU counts table. Rows: samples, columns: OTUs.
</p>
</td></tr>
<tr><td><code id="startedLog_+3A_count.columns">count.columns</code></td>
<td>

<p>numbers of colums containing OTU counts.
</p>
</td></tr>
<tr><td><code id="startedLog_+3A_logstart">logstart</code></td>
<td>

<p>number to add to zero counts before log transform. With 'logstart=0', the zero count datapoints become 'NA' (useful for analysis of pairwise correlations between OTUs).
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A table of log-transformed normalized values.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Started Log transform: 
Rocke DM and Durbin B Bioinformatics 2003 19:966-972 (DOI: 10.1093/bioinformatics/btg107)
MCMC.OTU:
Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

<hr>
<h2 id='summaryPlotOTU'>
Wrapper function for ggplot2 to make bar and line graphs of 
mcmc.otu() results
</h2><span id='topic+summaryPlotOTU'></span>

<h3>Description</h3>

<p>This function is called automatically by OTUsummary() and also
can be used separately to plot the results produced by OTUsummary() 
with more plotting options 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summaryPlotOTU(data, xgroup, facet = NA, type = "bar", 
x.order = NA, whiskers = "ci", otus = NA,log.base=10)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summaryPlotOTU_+3A_data">data</code></td>
<td>

<p>A summary table generated by HPDplot(), it is the first element in the 
returned list.
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_xgroup">xgroup</code></td>
<td>

<p>Which factor will be used to form the x axis (for 2-way designs).
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_facet">facet</code></td>
<td>

<p>The factor by which the plot will be split into facets (for 2-way designs).
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_type">type</code></td>
<td>

<p>Two types are supported: &quot;bar&quot; and &quot;line&quot; (&quot;line&quot; also has points). 
&quot;bar&quot; is more useful to plot fold-changes returned when OTUsummary() 
is run with the option 'relative=TRUE'. &quot;line&quot; is better for plotting 
actual inferred OTU abundances across factor levels; it is particularly 
good for time courses and other cases when multiple factor levels must be 
compared to each other. &quot;bar&quot; is good to plot log(fold-changes) relative to 
global control.  
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_x.order">x.order</code></td>
<td>

<p>A vector giving the order of factor levels on the x-axis. If unspecified, 
alphanumeric order will be used.
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_whiskers">whiskers</code></td>
<td>

<p>The interval indicated by the whiskers. Default is &quot;ci&quot;, the 95% credible 
interval; another option is &quot;sd&quot; - standard deviation of the posterior.
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_otus">otus</code></td>
<td>

<p>Vector of OTU names to plot. By default, all OTUs in the summary will be plotted.
</p>
</td></tr>
<tr><td><code id="summaryPlotOTU_+3A_log.base">log.base</code></td>
<td>

<p>Base of the logarithm to indicate on the y-axis label.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function invokes ggplot() functon from the ggplot2 package to plot 
the results either as a single panel (one-way designs) or a multi-panel 
(2-way designs, one panel per level of the factor specified by 'facet' argument).
</p>


<h3>Value</h3>

<p>A ggplot object. See http://docs.ggplot2.org/0.9.2.1/theme.html for ways 
to modify it, such as add text, rotate labels, change fonts, etc.
</p>


<h3>Author(s)</h3>

<p>Mikhail V. Matz, University of Texas at Austin
&lt;matz@utexas.edu&gt;
</p>


<h3>References</h3>

<p>Elizabeth A. Green, Sarah W. Davies, Mikhail V. Matz, Monica Medina Next-generation sequencing reveals cryptic Symbiodinium diversity within Orbicella faveolata and Orbicella franksi at the Flower Garden Banks, Gulf of Mexico. PeerJ 2014 https://peerj.com/preprints/246/ 
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
