<!DOCTYPE html><html lang="en"><head><title>Help for package MDSMap</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MDSMap}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MDSMap-package'><p>High density Genetic Linkage Mapping using Multidimensional Scaling</p></a></li>
<li><a href='#calc.maps.pc'><p>Estimate marker positions using Principal Curves</p></a></li>
<li><a href='#calc.maps.sphere'><p>Estimate marker positions using spherically constrained weighted MDS</p></a></li>
<li><a href='#calc.nnfit'><p>Calculate the nearest neighbour fit.</p></a></li>
<li><a href='#calc.nnfit.from.file'><p>Nearest neighbour fit from estimated map and file of pairwise recombination fractions.</p></a></li>
<li><a href='#calc.nnfit.loci'><p>Calculates the nearest neighbour fit for an individual marker.</p></a></li>
<li><a href='#calc.nswaps'><p>Calculates the number of swaps required to move from one order to another.</p></a></li>
<li><a href='#calc.pair.rf.lod'><p>Create recombination matrix from pairwise data file.</p></a></li>
<li><a href='#convert.polar'><p>Convert Cartesian coordinates from wMDS coordinates to polar coordinates.</p></a></li>
<li><a href='#dmap'><p>Calculates pairwise map distances from the recombination fraction.</p></a></li>
<li><a href='#dmap.check'><p>Reorders a distance map by a new marker order.</p></a></li>
<li><a href='#estimate.map'><p>Load data, estimate a linkage map and plot diagnostics for the fit.</p></a></li>
<li><a href='#get.dist.loci'><p>Calculates the distance of a marker from some objective &quot;truth&quot;.</p></a></li>
<li><a href='#get.nearest.informative'><p>For a given marker finds the nearest neighbours with LOD scores &gt; 0.</p></a></li>
<li><a href='#invert.map'><p>Invert the order of locimap from an estimated map.</p></a></li>
<li><a href='#lgI.txt'><p>Dataset lgI.txt: pairwise recombination fractions for 143 markers.</p></a></li>
<li><a href='#lgV.txt'><p>Dataset lgV.txt: pairwise recombination fractions for 238 markers.</p></a></li>
<li><a href='#map.to.interval'><p>Map points from MDS final configuration to interval starting at 0.</p></a></li>
<li><a href='#meandist.from.truth'><p>Calculates mean square distance between marker positions in two different maps.</p></a></li>
<li><a href='#plot.pcmap'><p>Diagnostic plots for the map estimation using calc.maps.pc with 2 dimensions.</p></a></li>
<li><a href='#plot.pcmap3d'><p>Diagnostic plots for the map estimation using calc.maps.pc with 3 dimensions.</p></a></li>
<li><a href='#plot.spheremap'><p>Produces diagnostic plots for the estimated map using <code>calc.maps.sphere</code>.</p></a></li>
<li><a href='#recalc.nnfit.from.map'><p>Calculate a nearest neighbour fit from an estimated map object.</p></a></li>
<li><a href='#sim.bc.rflod.file'><p>Simulate a backcross population from homozygous parents.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>High Density Genetic Linkage Mapping using Multidimensional
Scaling</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-08-07</td>
</tr>
<tr>
<td>Author:</td>
<td>Katharine F. Preedy &lt;Katharine.preedy@bioss.ac.uk&gt;, Christine A. Hackett &lt;christine.hackett@bioss.ac.uk&gt;</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bram Boskamp &lt;bram.boskamp@bioss.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Estimate genetic linkage maps for markers on a single chromosome (or in a single linkage group) from pairwise recombination fractions or intermarker distances using weighted metric multidimensional scaling. The methods are suitable for autotetraploid as well as diploid populations. Options for assessing the fit to a known map are also provided. Methods are discussed in detail in Preedy and Hackett (2016) &lt;<a href="https://doi.org/10.1007%2Fs00122-016-2761-8">doi:10.1007/s00122-016-2761-8</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> | file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>smacof (&ge; 1.9.0), princurve (&ge; 2.1.2), rgl, reshape</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.0</td>
</tr>
<tr>
<td>Suggests:</td>
<td>qtl, R.rsp</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-08-07 14:12:12 UTC; Bram Boskamp</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-08-09 10:10:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='MDSMap-package'>High density Genetic Linkage Mapping using Multidimensional Scaling</h2><span id='topic+MDSMap-package'></span>

<h3>Description</h3>

<p>MDSmap provides functions for estimating genetic linkage maps for
markers from a single linkage group from pairwise intermarker map
distances using the Haldane or Kosambi map function; or recombination 
fractions. It either uses constrained weighted metric multidimensional 
scaling (cMDS) in 2 dimensions or unconstrained weighted metric 
multidimensional scaling (MDS) followed by fitting a principal curve 
(PC) in either 2 or 3 dimensions. Pairwise distances can be weighted 
either by the LOD score or LOD2. There are functions for diagnostic 
plots, estimating the difference between the observed and estimated 
difference between points and their nearest informative neighbour, 
which may be useful in deciding which weights to use and also for 
testing estimated maps against a map estimated externally.
</p>


<h3>Details</h3>

<p>The main top level functions to use: <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code> and 
<code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>, and use <code><a href="#topic+plot.pcmap">plot.pcmap</a></code>, 
<code><a href="#topic+plot.spheremap">plot.spheremap</a></code> or <code><a href="#topic+plot.pcmap3d">plot.pcmap3d</a></code> to visualize 
the result.
</p>


<h3>Author(s)</h3>

<p>Katharine F. Preedy &lt;Katharine.preedy@bioss.ac.uk&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map&lt;-calc.maps.pc(system.file("extdata", "lgV.txt", package="MDSMap"),
ndim=2,weightfn='lod2',mapfn='kosambi')
plot(map)


</code></pre>

<hr>
<h2 id='calc.maps.pc'>Estimate marker positions using Principal Curves</h2><span id='topic+calc.maps.pc'></span>

<h3>Description</h3>

<p>Reads a text file of pairwise recombination fractions and LOD scores, reduces 
to 2 or 3 dimensions using wMDS and projects onto a single dimension using 
principal curves to estimate marker positions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.maps.pc(fname, spar = NULL, n = NULL, ndim = 2,
  weightfn = "lod2", mapfn = "haldane")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.maps.pc_+3A_fname">fname</code></td>
<td>
<p>Character string the name of the file of recombination fractions 
and scores it should not contain any suffices (the file should be a .txt file 
as described below).</p>
</td></tr>
<tr><td><code id="calc.maps.pc_+3A_spar">spar</code></td>
<td>
<p>Integer - the smoothing parameter for the principal curve. If 
NULL this will be done using leave one out cross validation.</p>
</td></tr>
<tr><td><code id="calc.maps.pc_+3A_n">n</code></td>
<td>
<p>Vector of integers or character strings containing markers to be 
omitted from the analysis.</p>
</td></tr>
<tr><td><code id="calc.maps.pc_+3A_ndim">ndim</code></td>
<td>
<p>Number of dimensions in which to perform the wMDS and fit the 
curve - can be 2 or 3.</p>
</td></tr>
<tr><td><code id="calc.maps.pc_+3A_weightfn">weightfn</code></td>
<td>
<p>Character string specifying the values to use for the weight 
matrix in the MDS <code>'lod2'</code> or <code>'lod'</code>.</p>
</td></tr>
<tr><td><code id="calc.maps.pc_+3A_mapfn">mapfn</code></td>
<td>
<p>Character string specifying the map function to use on the 
recombination fractions <code>'haldane'</code> is default, <code>'kosambi'</code> or <code>'none'</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads a file of the form described below and casts the data into matrices of 
pairwise recombination fractions and weights determined by the <code>weightfn</code> 
parameter (<code>LOD</code> or <code>LOD^2^</code>) calculates a distance matrix from the map function. 
Haldane is the default map function, none just uses recombination fractions 
and the other alternative is Kosambi (see <code><a href="#topic+dmap">dmap</a></code> for details). 
</p>
<p>Performs both an weighted MDS on the distance matrix using <code><a href="smacof.html#topic+smacofSym">smacofSym</a></code> and 
<code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code> (<cite>de Leeuw &amp; Mair 2009</cite>) and fits a 
principal curve to map this to an interval (<code><a href="princurve.html#topic+principal_curve">principal_curve</a></code> for details).
</p>
<p>File names should be of the form <code>fname.txt</code> and it is assumed that they are in 
a tab or space separated file of the format displayed below. The first entry on 
the first row is the number of markers to be analysed. Underneath this is a 
table in which the first two columns contain marker names, the third column 
contains the pairwise recombination fractions between the markers and the 
fourth column the associated lod score.  Note that marker names in the first 
column vary more slowly than in the second column. Missing recombination pairs 
are acceptable. Recombination fractions greater than 0.499999 are set to that 
value.
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>A list (S3 class pcmap or pcmap3d depending on ndim) with the following elements:
</p>
<table role = "presentation">
<tr><td><code>smacofsym</code></td>
<td>
<p>The unconstrained wMDS results.</p>
</td></tr>
<tr><td><code>pc</code></td>
<td>
<p>The results from the principal curve fit.</p>
</td></tr>
<tr><td><code>distmap</code></td>
<td>
<p>A symmetric matrix of pairwise distances between markers where 
the columns are in the estimated order.</p>
</td></tr>
<tr><td><code>lodmap</code></td>
<td>
<p>A symmetric matrix of lod scores associated with the distances 
in distmap.</p>
</td></tr>
<tr><td><code>locimap</code></td>
<td>
<p>A data frame of the markers containing the name of each marker, 
the number in the configuration plot if that is being used, the position of each 
marker in order of increasing distance and the nearest neighbour fit of the marker.</p>
</td></tr>
<tr><td><code>length</code></td>
<td>
<p>Integer giving the total length of the segment.</p>
</td></tr>
<tr><td><code>removed</code></td>
<td>
<p>A vector of the names of markers removed from the analysis.</p>
</td></tr>
<tr><td><code>locikey</code></td>
<td>
<p>A data frame showing the number associated with each marker name 
for interpreting the wMDS configuration plots.</p>
</td></tr>
<tr><td><code>meannnfit</code></td>
<td>
<p>The mean across all markers of the nearest neighbour fits.</p>
</td></tr>
</table>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF in R. J Stat Softw 31: 1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>
<p><cite>Hastie T, Weingessel A, Bengtsson H, Cannoodt R (1999) princurve: Fits a Principal Curve in Arbitrary Dimension. ) R package version 2.1.2.</cite> <a href="https://CRAN.R-project.org/package=princurve">https://CRAN.R-project.org/package=princurve</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>, <code><a href="#topic+calc.pair.rf.lod">calc.pair.rf.lod</a></code>, <code><a href="smacof.html#topic+smacofSym">smacofSym</a></code>, <code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code>, <code><a href="#topic+map.to.interval">map.to.interval</a></code>, <code><a href="#topic+dmap">dmap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map&lt;-calc.maps.pc(system.file("extdata", "lgV.txt", package="MDSMap"),
ndim=2,weightfn='lod2',mapfn='kosambi')
plot(map)
</code></pre>

<hr>
<h2 id='calc.maps.sphere'>Estimate marker positions using spherically constrained weighted MDS</h2><span id='topic+calc.maps.sphere'></span>

<h3>Description</h3>

<p>Reads a text file of pairwise recombination fractions and LOD scores, 
estimates marker positions using spherically constrained weighted MDS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.maps.sphere(fname, p = 100, n = NULL, weightfn = "lod2",
  mapfn = "haldane")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.maps.sphere_+3A_fname">fname</code></td>
<td>
<p>Character string specifying the base name of the file 
fname.txt which contains the data to be analysed. The file should be 
white space or tab separated.</p>
</td></tr>
<tr><td><code id="calc.maps.sphere_+3A_p">p</code></td>
<td>
<p>Integer - the penalty for deviations from the sphere - higher p 
forces points more closely onto a sphere.</p>
</td></tr>
<tr><td><code id="calc.maps.sphere_+3A_n">n</code></td>
<td>
<p>Vector of integers or strings containing markers to be omitted from 
the analysis.</p>
</td></tr>
<tr><td><code id="calc.maps.sphere_+3A_weightfn">weightfn</code></td>
<td>
<p>Character string specifying the values to use for the weight 
matrix in the MDS 'lod2' or 'lod'.</p>
</td></tr>
<tr><td><code id="calc.maps.sphere_+3A_mapfn">mapfn</code></td>
<td>
<p>Character string specifying the map function to use on the 
recombination fractions 'haldane' is default, 'kosambi' or 'none'.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This can be very slow with large sets of markers, in which case it may be 
better to consider <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>.
</p>
<p>Reads a file of the form described below and casts the data into matrices of 
pairwise recombination fractions and weights determined by the <code>weightfn</code> 
parameter (<code>LOD</code> or <code>LOD^2^</code>) calculates a distance matrix from the map 
function. Haldane is the default map function, None just uses recombination 
fractions and the other alternative is Kosambi (see <code>link{dmap}</code> 
for details). 
</p>
<p>Performs both an unconstrained and dual spherically constrained weighted MDS 
on the distance matrix using <code><a href="smacof.html#topic+smacofSym">smacofSym</a></code> and 
<code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code> (<cite>de Leeuw &amp; Mair 2009</cite>) 
and maps this to an interval (see <code><a href="#topic+map.to.interval">map.to.interval</a></code> for details).
</p>
<p>Inevitably the constrained MDS has higher stress than the unconstrained MDS and 
a good rule of thumb is that this should not be more than about 10
</p>
<p>File names should be of the form <code>fname.txt</code> and it is assumed that they are in 
a tab or space separated file of the format displayed below. The first entry on 
the first row is the number of markers to be analysed. Underneath this is a 
table in which the first two columns contain marker names, the third column 
contains the pairwise recombination fractions between the markers and the 
fourth column the associated LOD score.  Note that marker names in the first 
column vary more slowly than in the second column. Missing recombination pairs 
are acceptable. Recombination fractions greater than 0.499999 are set to that 
value.
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>A list (S3 class 'spheremap') with the following elements:
</p>
<table role = "presentation">
<tr><td><code>smacofsym</code></td>
<td>
<p>The unconstrained wMDS results.</p>
</td></tr>
<tr><td><code>smacofsphere</code></td>
<td>
<p>The spherically constrained wMDS results.</p>
</td></tr>
<tr><td><code>mapsphere</code></td>
<td>
<p>Map of the markers onto an interval containing order-the 
rank of each marker.</p>
</td></tr>
<tr><td><code>distmap</code></td>
<td>
<p>A symmetric matrix of pairwise distances between markers where 
the columns are in the estimated order.</p>
</td></tr>
<tr><td><code>lodmap</code></td>
<td>
<p>A symmetric matrix of lod scores associated with the distances in distmap.</p>
</td></tr>
<tr><td><code>locimap</code></td>
<td>
<p>A data frame of the markers containing the name of each marker, 
the number in the configuration plot if that is being used, the position of 
each marker in order of increasing distance and the nearest neighbour fit of 
the marker.</p>
</td></tr>
<tr><td><code>length</code></td>
<td>
<p>Integer giving the total length of the segment.</p>
</td></tr>
<tr><td><code>removed</code></td>
<td>
<p>A vector of the names of markers removed from the analysis.</p>
</td></tr>
<tr><td><code>locikey</code></td>
<td>
<p>A data frame showing the number associated with each marker 
name for interpreting the wMDS configuration plots.</p>
</td></tr>
<tr><td><code>stressratio</code></td>
<td>
<p>The ratio of the constrained to unconstrained stress.</p>
</td></tr>
<tr><td><code>ssphere</code></td>
<td>
<p>The stress per point of the spherically constrained wMDS.</p>
</td></tr>
<tr><td><code>ssym</code></td>
<td>
<p>Stress per point of the unconstrained wMDS.</p>
</td></tr>
<tr><td><code>meannnfit</code></td>
<td>
<p>The mean across all markers of the nearest neighbour fits.</p>
</td></tr>
</table>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF 
in R. J Stat Softw 31: 1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>, <code><a href="#topic+calc.pair.rf.lod">calc.pair.rf.lod</a></code>, <code><a href="smacof.html#topic+smacofSym">smacofSym</a></code>, <code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code>, <code><a href="#topic+map.to.interval">map.to.interval</a></code>, <code><a href="#topic+dmap">dmap</a></code>, <code><a href="#topic+calc.nnfit">calc.nnfit</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>smap&lt;-calc.maps.sphere(system.file("extdata", "lgI.txt", package="MDSMap"),
weightfn='lod',mapfn='kosambi')
</code></pre>

<hr>
<h2 id='calc.nnfit'>Calculate the nearest neighbour fit.</h2><span id='topic+calc.nnfit'></span>

<h3>Description</h3>

<p>Calculates the total, mean and individual differences between the observed and 
estimated distances from all loci and their nearest neighbours with non-zero 
LOD scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.nnfit(distmap, lodmap, estmap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.nnfit_+3A_distmap">distmap</code></td>
<td>
<p>Symmetric matrix of pairwise inter-marker distances with 
columns and rows corresponding to the estimated map order.</p>
</td></tr>
<tr><td><code id="calc.nnfit_+3A_lodmap">lodmap</code></td>
<td>
<p>Symmetric matrix of pairwise lod scores with columns and
rows corresponding to the estimated map order.</p>
</td></tr>
<tr><td><code id="calc.nnfit_+3A_estmap">estmap</code></td>
<td>
<p>Vector of estimated marker positions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The nearest neighbour fit for a marker is the sum of the difference between
the observed and estimated distances between the marker and its nearest
informative neighbour. A neighbour is informative if the LOD score for the 
inter-marker distance is greater than zero. This function calculates the nearest 
neighbour fit for each marker and returns the fit for each point and the sum 
of all the fits.
</p>


<h3>Value</h3>

<p>A list with the elements:
</p>
<table role = "presentation">
<tr><td><code>fit</code></td>
<td>
<p>Sum of the nearest neighbour fits over all markers.</p>
</td></tr>
<tr><td><code>pointfits</code></td>
<td>
<p>Vector of nearest neighbour fits for each marker.</p>
</td></tr>
<tr><td><code>meanfit</code></td>
<td>
<p>Mean of the nearest neighbour fits over all markers.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+calc.nnfit.loci">calc.nnfit.loci</a></code>
</p>

<hr>
<h2 id='calc.nnfit.from.file'>Nearest neighbour fit from estimated map and file of pairwise recombination fractions.</h2><span id='topic+calc.nnfit.from.file'></span>

<h3>Description</h3>

<p>Calculates a nearest neighbour fit based from an estimated map and a file 
containing pairwise recombination fractions and LOD scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.nnfit.from.file(estmap, fname, mapfn = "haldane", n = NULL,
  header = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.nnfit.from.file_+3A_estmap">estmap</code></td>
<td>
<p>A character string indicating the name of a comma separated 
value file with  the first column containing marker names in the order 
of their estimated position.</p>
</td></tr>
<tr><td><code id="calc.nnfit.from.file_+3A_fname">fname</code></td>
<td>
<p>A character string specifying the base name of the file 
<code>fname.txt</code> which contains the data to be analysed the file should be 
white space or tab separated.</p>
</td></tr>
<tr><td><code id="calc.nnfit.from.file_+3A_mapfn">mapfn</code></td>
<td>
<p>Character string, <code>'haldane'</code>, <code>'kosambi'</code> or <code>'none'</code>
specifying the values to use to estimate the map distance from the 
recombination fractions. Default is <code>'haldane'</code>.</p>
</td></tr>
<tr><td><code id="calc.nnfit.from.file_+3A_n">n</code></td>
<td>
<p>Vector of character strings or numbers specifying the markers to be 
omitted from the analysis. Default is <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="calc.nnfit.from.file_+3A_header">header</code></td>
<td>
<p>Logical argument indicating whether the .csv file estmap contains headers - default is TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads in two files <code>fname.txt</code> and <code>estmap</code>. 
</p>
<p>The data is cast the data into symmetric matrices of pairwise recombination 
fractions and LOD scores with the order of columns and rows in the matrix 
determined by the order specified in <code>estmap</code>. A distance matrix is 
calculated according to the method specified by <code>mapfn</code>. Haldane is the 
default map function, None just uses recombination fractions and the other 
alternative is Kosambi (see <code><a href="#topic+dmap">dmap</a></code> for details). The nearest 
neighbour fit is then calculated (see <code><a href="#topic+calc.nnfit">calc.nnfit</a></code> for details)
</p>
<p><code>estmap</code> should contain marker names in the first column in the order 
of the estimated map.
</p>
<p><code>fname</code> should be of the form <code>fname.txt</code> and it is assumed that they are in 
a tab or space separated file of the format displayed below. The first entry on 
the first row is the number of markers to be analysed. Underneath this is a 
table in which the first two columns contain marker names, the third column 
contains the pairwise recombination fractions between the markers and the 
fourth column the associated LOD score.  Note that marker names in the first 
column vary more slowly than in the second column. Missing recombination pairs 
are acceptable. Recombination fractions greater than 0.499999 are set to that 
value.
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>A list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>fit</code></td>
<td>
<p>Sum over all markers of the nearest neighbour fits.</p>
</td></tr>
<tr><td><code>pointfits</code></td>
<td>
<p>The nearest neighbour fit for each marker.</p>
</td></tr>
<tr><td><code>meanfit</code></td>
<td>
<p>Mean of the nearest neighbour fits over all markers.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+dmap">dmap</a></code>, <code><a href="#topic+calc.nnfit">calc.nnfit</a></code>, <code><a href="#topic+calc.pair.rf.lod">calc.pair.rf.lod</a></code>
</p>

<hr>
<h2 id='calc.nnfit.loci'>Calculates the nearest neighbour fit for an individual marker.</h2><span id='topic+calc.nnfit.loci'></span>

<h3>Description</h3>

<p>Calculates the nearest neighbour fit for an individual marker.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.nnfit.loci(loci, distmap, lodmap, estmap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.nnfit.loci_+3A_loci">loci</code></td>
<td>
<p>Scalar indicating the estimated rank position of the marker.</p>
</td></tr>
<tr><td><code id="calc.nnfit.loci_+3A_distmap">distmap</code></td>
<td>
<p>Symmetric matrix of pairwise inter-marker distances with columns
and rows corresponding to the estimated map order.</p>
</td></tr>
<tr><td><code id="calc.nnfit.loci_+3A_lodmap">lodmap</code></td>
<td>
<p>Symmetric matrix of pairwise lod scores with columns and rows
corresponding to the estimated map order.</p>
</td></tr>
<tr><td><code id="calc.nnfit.loci_+3A_estmap">estmap</code></td>
<td>
<p>Vector of estimated marker positions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The nearest neighbour fit for a marker is the sum of the difference between 
the observed and estimated distances between the marker and its nearest 
informative neighbour. A neighbour is informative if the LOD score for the 
inter-marker distance is non zero. This function finds the nearest markers 
with a non-zero LOD score (this may be one or two markers). Calculates the 
estimated distances between these markers and the marker of interest and 
returns the sum of the absolute values of the difference between the observed 
and estimated distances.
</p>


<h3>Value</h3>

<p>Scalar corresponding to the difference between the observed and 
estimated intermarker differences.
</p>

<hr>
<h2 id='calc.nswaps'>Calculates the number of swaps required to move from one order to another.</h2><span id='topic+calc.nswaps'></span>

<h3>Description</h3>

<p>Calculates the number of swaps required to move from one order to another.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.nswaps(map1, map2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.nswaps_+3A_map1">map1</code></td>
<td>
<p>Vector of marker positions or ranks.</p>
</td></tr>
<tr><td><code id="calc.nswaps_+3A_map2">map2</code></td>
<td>
<p>Vector of marker positions or ranks.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is intended to be used when comparing an estimated marker ordering to 
some perceived &quot;truth&quot;. It is most likely to be useful when dealing with 
simulated data where the concept of truth makes most sense. It calculates 
the minimum number of single place swaps that would be needed to move from 
<code>map1</code> to <code>map2</code> and it does this by reverse engineering kendall's tau b 
correlation coefficient 
</p>
<p style="text-align: center;"><code class="reqn">\tau=\frac{2(C-D)}{N}</code>
</p>

<p>where <code class="reqn">N</code> is the total number of pairs of markers, <code class="reqn">C</code> the number of concordant pairs and <code class="reqn">D</code> the number 
of discordant pairs. If there are <code class="reqn">n</code> markers then the total number of pairs 
<code class="reqn">N={{n}\choose{2}}</code> and <code class="reqn">C=N-D</code> so <code class="reqn">D=0.5 {{n}\choose{2}}(1-\tau)</code> and the minimum 
number of swaps is the minimum of <code class="reqn">D</code> and <code class="reqn">N-D</code>
</p>


<h3>Value</h3>

<p>Scalar giving the number of swaps.
</p>

<hr>
<h2 id='calc.pair.rf.lod'>Create recombination matrix from pairwise data file.</h2><span id='topic+calc.pair.rf.lod'></span>

<h3>Description</h3>

<p>Reads a text file of pairwise recombination fractions and LOD scores and casts 
it into a matrix of recombination fractions and weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calc.pair.rf.lod(fname, weightfn = "lod", ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="calc.pair.rf.lod_+3A_fname">fname</code></td>
<td>
<p>Character string specifying the base name of the file <code>fname.txt</code> which 
contains the data to be analysed the file should be white space or tab separated.</p>
</td></tr>
<tr><td><code id="calc.pair.rf.lod_+3A_weightfn">weightfn</code></td>
<td>
<p>Character string specifiying the values to use for the weight 
matrix <code>'lod2'</code> or <code>'lod'</code>.</p>
</td></tr>
<tr><td><code id="calc.pair.rf.lod_+3A_...">...</code></td>
<td>
<p><code><a href="utils.html#topic+read.table">read.table</a></code> arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>File names should be of the form <code>fname.txt</code> and it is assumed that they are in 
a tab or space separated file of the format displayed below. The first entry on 
the first row is the number of markers to be analysed. Underneath this is a 
table in which the first two columns contain marker names, the third column 
contains the pairwise recombination fractions between the markers and the 
fourth column the associated LOD score.  Note that marker names in the first 
column vary more slowly than in the second column. Missing recombination pairs 
are acceptable. Recombination fractions greater than 0.499999 are set to that 
value
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>A list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>rf</code></td>
<td>
<p>A symmetric matrix of recombination fractions.</p>
</td></tr>
<tr><td><code>nloci</code></td>
<td>
<p>The number of markers in the analysis.</p>
</td></tr>
<tr><td><code>locinames</code></td>
<td>
<p>The names of the markers in the analysis.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>lodrf&lt;-calc.pair.rf.lod(system.file("extdata", "lgV.txt", package="MDSMap"), 
"lod2")
</code></pre>

<hr>
<h2 id='convert.polar'>Convert Cartesian coordinates from wMDS coordinates to polar coordinates.</h2><span id='topic+convert.polar'></span>

<h3>Description</h3>

<p>Converts the coordinates of points in the final configuration of a spherically 
constrained wMDS from Cartesian to polar coordinates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>convert.polar(mdsobject, nloci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="convert.polar_+3A_mdsobject">mdsobject</code></td>
<td>
<p>Output from <code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code> using the dual method in the <span class="pkg">smacof</span> package.</p>
</td></tr>
<tr><td><code id="convert.polar_+3A_nloci">nloci</code></td>
<td>
<p>The number of markers in the configuration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Centres the circle on zero if necessary, finds a the most natural break in 
the points to start as 0, then calculates the angle of each point relative 
to this. The radius is the median distance of points from the centre.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>theta</code></td>
<td>
<p>A vector of angles one for each point.</p>
</td></tr>
<tr><td><code>radius</code></td>
<td>
<p>A scalar the radius of sphere.</p>
</td></tr>
</table>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF in R. J Stat Softw 31:1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>


<h3>See Also</h3>

<p><code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#M and lod should be n x n symmetric matrices of the same dimensions where n 
#is the number markers to be analysed
## Not run: 
mds1&lt;-smacofSphere(M,ndim=2,algorithm="dual",weightmat=lod,penalty=100)
pol&lt;-convert.polar(mds1,n)

## End(Not run)
</code></pre>

<hr>
<h2 id='dmap'>Calculates pairwise map distances from the recombination fraction.</h2><span id='topic+dmap'></span>

<h3>Description</h3>

<p>Calculates pairwise map distances from the recombination fraction.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmap(rf, mapfn = "haldane")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dmap_+3A_rf">rf</code></td>
<td>
<p>A symmetric matrix of pairwise recombination fractions.</p>
</td></tr>
<tr><td><code id="dmap_+3A_mapfn">mapfn</code></td>
<td>
<p>A character string specifying the map function to be 
used in calculated the distance <code>'haldane'</code>, <code>'kosambi'</code>, <code>'none'</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The default is the <code>'haldane'</code> map function <code class="reqn">0.5\ln(1-2rf)</code>, <code>'kosambi'</code> returns 
<code class="reqn">0.25\ln((1+2rf)/(1-2rf))</code> and <code>'none'</code> returns <code class="reqn">rf</code>, the recombination fraction.
</p>


<h3>Value</h3>

<p>a symmetric matrix of pairwise map distances in the same format as the recombination matrix supplied.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>lodrf&lt;-calc.pair.rf.lod(system.file("extdata", "lgV.txt", package="MDSMap"))
mdist=dmap(lodrf$rf,mapfn="haldane")
</code></pre>

<hr>
<h2 id='dmap.check'>Reorders a distance map by a new marker order.</h2><span id='topic+dmap.check'></span>

<h3>Description</h3>

<p>Reorders a distance map by a new marker order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dmap.check(distmap, newrank)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dmap.check_+3A_distmap">distmap</code></td>
<td>
<p>A symmetric matrix of pairwise inter-marker distances.</p>
</td></tr>
<tr><td><code id="dmap.check_+3A_newrank">newrank</code></td>
<td>
<p>A vector of scalars giving the new rank of each marker, markers 
should appear in the same order as in the distmap.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The rows and columns in distmap are reordered such that if entry i in <code>newrank</code> 
has value j then row j and column j in the new matrix are row i and column i 
from distmap.
</p>


<h3>Value</h3>

<p>Matrix of pairwise inter-marker distances.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>s&lt;-matrix(1:25,nrow=5)
s&lt;-0.5*(s+t(s))
rank&lt;-c(1,3,4,2,5)
dmap.check(s,rank)
</code></pre>

<hr>
<h2 id='estimate.map'>Load data, estimate a linkage map and plot diagnostics for the fit.</h2><span id='topic+estimate.map'></span>

<h3>Description</h3>

<p>Load data, estimate a linkage map and plot diagnostics for the fit.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate.map(fname, p = NULL, n = NULL, ispc = TRUE, ndim = 2,
  weightfn = "lod2", mapfn = "haldane", D1lim = NULL, D2lim = NULL,
  D3lim = NULL, displaytext = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="estimate.map_+3A_fname">fname</code></td>
<td>
<p>Character string containing the base file from which the data 
should be read - should contain the complete file name excluding the suffix 
which should be <code>.txt</code></p>
</td></tr>
<tr><td><code id="estimate.map_+3A_p">p</code></td>
<td>
<p>Smoothing parameter.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_n">n</code></td>
<td>
<p>Vector of integers or character strings containing the name or 
position in the input list of loci to be excluded from the analysis.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_ispc">ispc</code></td>
<td>
<p>Logical determining the method to be used to estimate the map. By 
default this is <code>TRUE</code> and the method of principal curves will be used. If 
<code>FALSE</code> then the constrained MDS method will be used.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_ndim">ndim</code></td>
<td>
<p>Integer the number of dimensions to use if the Principal curves 
method is used. By default this is 2, but it can also be 3.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_weightfn">weightfn</code></td>
<td>
<p>Character string specifying the values to use for the weight 
matrix in the MDS <code>lod2</code> or <code>lod</code>.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_mapfn">mapfn</code></td>
<td>
<p>Character string specifying the map function to use on the 
recombination fractions <code>'haldane'</code> is default, <code>'kosambi'</code> or <code>'none'</code>.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_d1lim">D1lim</code></td>
<td>
<p>Numeric vector specifying the limits of the axis relating to 
dimension 1 of the wMDS used to estimate the map.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_d2lim">D2lim</code></td>
<td>
<p>Numeric vector specifying the limits of the axis relating to 
dimension 1 of the wMDS used to estimate the map.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_d3lim">D3lim</code></td>
<td>
<p>Numeric vector specifying the limits of the axis relating to 
dimension 1 of the wMDS used to estimate the map.</p>
</td></tr>
<tr><td><code id="estimate.map_+3A_displaytext">displaytext</code></td>
<td>
<p>Logical argument determining how markers should be labelled 
in the wMDS configuration plot. If <code>TRUE</code> then marker names are used. If 
<code>FALSE</code> then numbers are used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Data is read from a text file which should be of the form described below. 
By default, <code>ispc=TRUE</code>, in which case maps are estimated using unconstrained 
weighted MDS followed by fitting a principal curve. Details can be found in 
the description of the function <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>. If <code>ispc=FALSE</code> 
maps are estimated using spherically constrained weighted MDS. Details can be 
found in the description of the function <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>.
</p>
<p><code>ndim</code> is only relevant if <code>ispc=TRUE</code>, in which case it specifies the number of 
dimensions to be used, the default is 2 but it can also be 3 dimensions. 
</p>
<p>Diagnostic plots are then produced using <code><a href="#topic+plot.pcmap">plot.pcmap</a></code> for the method of 
principal curves in 2 dimensions, <code><a href="#topic+plot.pcmap3d">plot.pcmap3d</a></code> for the method of principal 
curves in 3 dimensions and <code><a href="#topic+plot.spheremap">plot.spheremap</a></code> for the method using spherically 
constrained MDS.
</p>
<p><code>n</code> specifies markers to be omitted from the analysis. It can be a vector of 
character strings specifying makers to be omitted, or a vector of integers 
specifying the markers to omit. The latter method is likely to be useful when 
removing outliers after inspection of the diagnostic plot, because the output 
contains a dataframe, locikey, which associates each marker with its 
identifying number. By default this is NULL and all markers in the file will 
be analysed.
</p>
<p><code>p</code> is a smoothing parameter which operates quite differently depending on 
whether map estimation is performed using Principal Curves or Constrained 
MDS. If the PC method is used, <code>p</code> determines the smoothing parameter spar in 
the function <code><a href="princurve.html#topic+principal_curve">principal_curve</a></code> from the package 
<span class="pkg">princurve</span>. If <code>NULL</code> then the most appropriate value will be determined 
using leave one out cross validation. 
If Constrained MDS is used then <code>p</code> must be set to a number which specifies the 
penalty for deviations from the sphere in the function <code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code> from the 
<span class="pkg">smacof</span> package. Something between 50 and 100 is generally appropriate and this 
penalty can be decreased if stress from the constrained analysis is more than 
about 10
for details)
</p>
<p>File names should be of the form <code>fname.txt</code> and it is assumed that they are in 
a tab or space separated file of the format displayed below. The first entry on 
the first row is the number of markers to be analysed. Underneath this is a 
table in which the first two columns contain marker names, the third column 
contains the pairwise recombination fractions between the markers and the 
fourth column the associated LOD score.  Note that marker names in the first 
column vary more slowly than in the second column. Missing recombination pairs 
are acceptable. Recombination fractions greater than 0.499999 are set to that 
value.
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>map (s3 class pcmap, pcmap3d or spheremap) from <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code> if <code>ispc=TRUE</code> or 
<code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code> if <code>ispc=FALSE</code>.
</p>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF in R. J Stat Softw 31: 1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>
<p><cite>Hastie T, Weingessel A, Bengtsson H, Cannoodt R (1999) princurve: Fits a Principal Curve in Arbitrary Dimension. ) R package version 2.1.2.</cite> <a href="https://CRAN.R-project.org/package=princurve">https://CRAN.R-project.org/package=princurve</a>
</p>


<h3>See Also</h3>

<p><code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code>, <code><a href="princurve.html#topic+principal_curve">principal_curve</a></code>, <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>, <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>, <code><a href="#topic+plot.pcmap">plot.pcmap</a></code>, <code><a href="#topic+plot.pcmap3d">plot.pcmap3d</a></code>, <code><a href="#topic+plot.spheremap">plot.spheremap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>estimate.map(system.file("extdata", "lgI.txt", package="MDSMap"),
ndim=3)
</code></pre>

<hr>
<h2 id='get.dist.loci'>Calculates the distance of a marker from some objective &quot;truth&quot;.</h2><span id='topic+get.dist.loci'></span>

<h3>Description</h3>

<p>Calculates the distance of a marker from some objective &quot;truth&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.dist.loci(loci, estmap, realmap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.dist.loci_+3A_loci">loci</code></td>
<td>
<p>Character string or number specifying the marker name.</p>
</td></tr>
<tr><td><code id="get.dist.loci_+3A_estmap">estmap</code></td>
<td>
<p>Data frame in which has a column called &quot;names&quot; containing marker 
names and a column called &quot;position&quot; containing marker positions.</p>
</td></tr>
<tr><td><code id="get.dist.loci_+3A_realmap">realmap</code></td>
<td>
<p>Data frame in which the first column contains marker names and 
the second column marker positions. Column names are not necessary.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Both the first column of <code>realmap</code> and <code>estmap$name</code> must contain 
<code>markername</code>, but aside from this they do not have to have identical entries.
</p>


<h3>Value</h3>

<p>position in estmap-position in realmap
</p>

<hr>
<h2 id='get.nearest.informative'>For a given marker finds the nearest neighbours with LOD scores &gt; 0.</h2><span id='topic+get.nearest.informative'></span>

<h3>Description</h3>

<p>Finds the nearest neighbours of a marker with LOD scores &gt; 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get.nearest.informative(loci, lodmap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get.nearest.informative_+3A_loci">loci</code></td>
<td>
<p>Scalar indicating a marker number</p>
</td></tr>
<tr><td><code id="get.nearest.informative_+3A_lodmap">lodmap</code></td>
<td>
<p>Symmetric matrix of pairwise LOD scores</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The columns and rows of the matrix should be in the order corresponding to 
the estimated map order. The function then returns the ranks of first markers 
to the left and right of the marker of interest with non-zero lod scores.
</p>


<h3>Value</h3>

<p>A vector of length 1 or 2 containing the rank of the nearest informative markers.
</p>

<hr>
<h2 id='invert.map'>Invert the order of locimap from an estimated map.</h2><span id='topic+invert.map'></span>

<h3>Description</h3>

<p>Takes the locimap from estimate.maps a dataframe containing names and positions 
and any other information in increasing order of distance and inverts the order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>invert.map(locimap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="invert.map_+3A_locimap">locimap</code></td>
<td>
<p>a data frame containing the markers names and positions</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The map should be a data frame with a column called 'position'. It should have a starting marker a position zero. 
The function then inverts the distances from so that the marker at maximum distance from the starting marker (the end marker) is at distance 0 
and the original starting marker is now at the maximum distance. It also inverts the order of the rows in the data frame. 
Thus if the markers were originally in order of increasing distance from the starting marker they will now be in order of increasing distance 
from the end marker.
</p>


<h3>Value</h3>

<p>The original data frame in inverted order with the distances inverted so that the end marker is now the starting marker.
</p>

<hr>
<h2 id='lgI.txt'>Dataset lgI.txt: pairwise recombination fractions for 143 markers.</h2><span id='topic+lgI.txt'></span>

<h3>Description</h3>

<p>A dataset containing the pairwise recombinations fractions for 143 SNP markers 
from linkage group I of potato. These are derived from the genotypes of 190 
offspring from a cross between potato cultivar Stirling and the breeding line 
12601ab1. Further details are available in Hackett et al. (2013).
</p>


<h3>Format</h3>

<p>An ascii text file in the format described in <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code> 
and <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>. The first line contains the number of markers
and the number of combinations. Then follow the space-separated combinations with
their recombination fractions and LOD scores:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<h3>lgI.txt</h3>

<p>NA
</p>


<h3>Source</h3>

<p><cite>Hackett, C.A., McLean, K. and Bryan, G.J. (2013). Linkage analysis and QTL mapping using SNP dosage data in a tetraploid potato mapping population. PLoS ONE 8, e63939</cite>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata", "lgI.txt", package="MDSMap")

</code></pre>

<hr>
<h2 id='lgV.txt'>Dataset lgV.txt: pairwise recombination fractions for 238 markers.</h2><span id='topic+lgV.txt'></span>

<h3>Description</h3>

<p>A dataset containing the pairwise recombinations fractions for 238 SNP markers 
from linkage group V of potato. These are derived from the genotypes of 190 
offspring from a cross between potato cultivar Stirling and the breeding line 
12601ab1. Further details are available in Hackett et al. (2013).
</p>


<h3>Format</h3>

<p>An ascii text file in the format described in <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code> 
and <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>. The first line contains the number of markers
and the number of combinations. Then follow the space-separated combinations with
their recombination fractions and LOD scores:
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>LOD</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>


<h3>lgV.txt</h3>

<p>NA
</p>


<h3>Source</h3>

<p><cite>Hackett, C.A., McLean, K. and Bryan, G.J. (2013). Linkage analysis and QTL mapping using SNP dosage data in a tetraploid potato mapping population. PLoS ONE 8, e63939</cite>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>system.file("extdata", "lgV.txt", package="MDSMap")

</code></pre>

<hr>
<h2 id='map.to.interval'>Map points from MDS final configuration to interval starting at 0.</h2><span id='topic+map.to.interval'></span>

<h3>Description</h3>

<p>Maps points from the final configuration of a 2-dimensional spherically 
constrained wMDS to an interval with a starting point at 0.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>map.to.interval(mdsobject, nloci)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="map.to.interval_+3A_mdsobject">mdsobject</code></td>
<td>
<p>The output from <code><a href="smacof.html#topic+smacofSphere">smacofSphere</a></code>.</p>
</td></tr>
<tr><td><code id="map.to.interval_+3A_nloci">nloci</code></td>
<td>
<p>The number of markers in the configuration.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Centres the configuration on zero and calculates the median distance of the points from the origin.
Finds the largest gap in the spherical configuration and assigns the marker on the right hand side of it angle 0. 
Converts Cartesian coordinates to polar coordinates and projects points onto 
the arc centred on 0 with radius the median distance from the origin.
</p>


<h3>Value</h3>

<p>A list with the elements:
</p>
<table role = "presentation">
<tr><td><code>chromlength</code></td>
<td>
<p>A named vector giving the position of each marker.</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>A named vector giving the rank order of the markers.</p>
</td></tr>
<tr><td><code>locilength</code></td>
<td>
<p>A named vector giving the position of each marker in order 
of increasing distance along the segment.</p>
</td></tr>
<tr><td><code>maporder</code></td>
<td>
<p>A named vector of the position in the input list of each 
marker in order of increasing distance along the segment.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+convert.polar">convert.polar</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># M and lod should be n x n symmetric matrices of the same dimensions where
# n is the number markers to be analysed
## Not run: 
mds1&lt;-smacofSphere(M,ndim=2,algorithm=dual,weightmat=lod,penalty=100)
pol&lt;-map.to.interval (m1,n)

## End(Not run)
</code></pre>

<hr>
<h2 id='meandist.from.truth'>Calculates mean square distance between marker positions in two different maps.</h2><span id='topic+meandist.from.truth'></span>

<h3>Description</h3>

<p>Calculates mean square distance of markers in the analysis from some objective 
&quot;truth&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meandist.from.truth(estmap, realmap)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="meandist.from.truth_+3A_estmap">estmap</code></td>
<td>
<p>Estimated map with 2 columns, <code>name</code> and <code>position</code> 
contain marker names and positions.</p>
</td></tr>
<tr><td><code id="meandist.from.truth_+3A_realmap">realmap</code></td>
<td>
<p>Map in which the first column contains marker names and the 
second contains marker positions. Column names are not necessary.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The first column of <code>realmap</code> must contain identical entries to 
<code>estmap$name</code>. However, the order of entries can be different.
</p>
<p>For every marker the difference between the position stated in <code>estmap</code> 
and in <code>realmap</code> is calculated (see <code><a href="#topic+get.dist.loci">get.dist.loci</a></code>).
</p>
<p>Every difference is squared and the mean of the square differences is returned.
</p>
<p>Note that where different weights are used in estimating maps, it is valid to 
compare the mean distance from the truth. However, if different map functions 
are used then the distances are not comparable. 
</p>
<p>Therefore, if there is some knowledge of markers on a chromosome and data is 
simulated so that there is some objective knowledge of the truth then this 
function could be used to decide whether to use <code>lod</code> or <code>lod2</code> 
weightings to estimate maps attempting to locate additional markers. However, 
it is not suitable for deciding on the map functions used to calculated the 
pairwise marker distances.
</p>


<h3>Value</h3>

<p>A list with the following elements:
</p>
<table role = "presentation">
<tr><td><code>pointdist</code></td>
<td>
<p>Data frame containing marker names and the distance between 
the estimated position and the &quot;real&quot; position.</p>
</td></tr>
<tr><td><code>meansquaredist</code></td>
<td>
<p>mean square distance between the estimated real position 
of markers.</p>
</td></tr>
</table>

<hr>
<h2 id='plot.pcmap'>Diagnostic plots for the map estimation using calc.maps.pc with 2 dimensions.</h2><span id='topic+plot.pcmap'></span>

<h3>Description</h3>

<p>Diagnostic plots for the map estimation using calc.maps.pc with 2 dimensions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pcmap'
plot(x, D1lim = NULL, D2lim = NULL,
  displaytext = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.pcmap_+3A_x">x</code></td>
<td>
<p>Map object from <code><a href="#topic+calc.maps.pc">calc.maps.pc()</a></code> with 2 dimensions.</p>
</td></tr>
<tr><td><code id="plot.pcmap_+3A_d1lim">D1lim</code></td>
<td>
<p>Numeric vector specifying the limits of the horizontal axis.</p>
</td></tr>
<tr><td><code id="plot.pcmap_+3A_d2lim">D2lim</code></td>
<td>
<p>Numeric vector specifying the limits of the vertical axis.</p>
</td></tr>
<tr><td><code id="plot.pcmap_+3A_displaytext">displaytext</code></td>
<td>
<p>Logical argument determining how markers should be labelled 
in the wMDS configuration plot. If TRUE then marker names are used. If FALSE 
then numbers are used.</p>
</td></tr>
<tr><td><code id="plot.pcmap_+3A_...">...</code></td>
<td>
<p>Further arguments are ignored. (accepted for compatibility with generic plot)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots 2 panels:
</p>
<p>Panel 1 the final MDS configuration and the fitted principal curve from the 
<code><a href="#topic+calc.maps.pc">calc.maps.pc()</a></code> in 2 dimensions. If D1lim or D2lim 
is not specified, then limits are defined by <code><a href="smacof.html#topic+plot.smacof">plot.smacof</a></code>. 
</p>
<p>Panel 2 the pointwise nearest neighbour fits in order of the position in the 
estimated map.
</p>
<p>Markers are assigned numbers according to the order in which they occur in 
the input file. The locikey output of the map object is a data frame 
associating marker names with their numbers. This can be accessed using 
<code>pcmap$locikey</code>. If <code>displaytext=FALSE</code> then markers will be labelled 
by these numbers. By default displaytext=TRUE and markers are labelled by 
marker name.
</p>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF in R. J Stat Softw 31: 1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.pcmap3d">plot.pcmap3d</a></code>, <code><a href="#topic+plot.spheremap">plot.spheremap</a></code>,<code><a href="smacof.html#topic+plot.smacof">plot.smacof</a></code>, <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map&lt;-calc.maps.pc(system.file("extdata", "lgV.txt", package="MDSMap"),
ndim=2,weightfn='lod2',mapfn='haldane')
plot(map)
</code></pre>

<hr>
<h2 id='plot.pcmap3d'>Diagnostic plots for the map estimation using calc.maps.pc with 3 dimensions.</h2><span id='topic+plot.pcmap3d'></span>

<h3>Description</h3>

<p>Diagnostic plots for the map estimation using calc.maps.pc with 3 dimensions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'pcmap3d'
plot(x, D1lim = NULL, D2lim = NULL, D3lim = NULL,
  displaytext = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.pcmap3d_+3A_x">x</code></td>
<td>
<p>Map object from calc.maps.pc() with 3 dimensions.</p>
</td></tr>
<tr><td><code id="plot.pcmap3d_+3A_d1lim">D1lim</code></td>
<td>
<p>Numeric vector specifying the limits of the axis relating to dimension 1 of the wMDS used to obtain pcmap3d.</p>
</td></tr>
<tr><td><code id="plot.pcmap3d_+3A_d2lim">D2lim</code></td>
<td>
<p>Numeric vector specifying the limits of the axis relating to dimension 1 of the wMDS used to obtain pcmap3d.</p>
</td></tr>
<tr><td><code id="plot.pcmap3d_+3A_d3lim">D3lim</code></td>
<td>
<p>Numeric vector specifying the limits of the axis relating to dimension 1 of the wMDS used to obtain pcmap3d.</p>
</td></tr>
<tr><td><code id="plot.pcmap3d_+3A_displaytext">displaytext</code></td>
<td>
<p>Logical argument determining how markers should be labelled 
in the wMDS configuration plot. If TRUE then marker names are used. If FALSE then numbers are used.</p>
</td></tr>
<tr><td><code id="plot.pcmap3d_+3A_...">...</code></td>
<td>
<p>Further arguments are ignored. (accepted for compatibility with generic plot)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots 4 panels
</p>
<p>Panels 1-3 show the final MDS configuration and the fitted principal curve from 
the <code><a href="#topic+calc.maps.pc">calc.maps.pc()</a></code> in 3 dimensions. plots <code>D1</code> vs 
<code>D2</code>, <code>D1</code> vs <code>D3</code> and <code>D2</code> vs <code>D3</code>.  If <code>D1lim</code>, 
<code>D2lim</code> or <code>D3lim</code> is not specified, then limits are defined by <code><a href="smacof.html#topic+plot.smacof">plot.smacof</a></code>.
</p>
<p>Panel 4 shows the pointwise nearest neighbour fits in order of the position 
in the estimated map.
</p>
<p>Also plots a 3 dimensional scatterplot of the final MDS configuration and the 
fitted principal curve in a new window using <code><a href="rgl.html#topic+plot3d">plot3d</a></code> from the 
<code><a href="rgl.html#topic+rgl">rgl package</a></code>. 
</p>
<p>Markers are assigned numbers according to the order in which they occur in the 
input file. The locikey output of the map object is a data frame associating 
marker names with their numbers. This can be accessed using <code>pcmap3d$locikey</code>. 
If <code>displaytext=FALSE</code> then markers will be labelled by these numbers. 
By default <code>displaytext=TRUE</code> and markers are labelled by marker name.
</p>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF in R. J Stat Softw 31: 1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.pcmap">plot.pcmap</a></code>, <code><a href="#topic+plot.spheremap">plot.spheremap</a></code>,<code><a href="smacof.html#topic+plot.smacof">plot.smacof</a></code>, <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>, <code><a href="rgl.html#topic+plot3d">plot3d</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map&lt;-calc.maps.pc(system.file("extdata", "lgV.txt", package="MDSMap"),
ndim=3,weightfn='lod2',mapfn='haldane')
plot(map)
</code></pre>

<hr>
<h2 id='plot.spheremap'>Produces diagnostic plots for the estimated map using <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>.</h2><span id='topic+plot.spheremap'></span>

<h3>Description</h3>

<p>Produces diagnostic plots for the estimated map using <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'spheremap'
plot(x, displaytext = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot.spheremap_+3A_x">x</code></td>
<td>
<p>Map object from <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code></p>
</td></tr>
<tr><td><code id="plot.spheremap_+3A_displaytext">displaytext</code></td>
<td>
<p>Logical argument determining how markers should be labelled in the MDS 
configuration plot. If TRUE then marker names are used. If FALSE then numbers are used.</p>
</td></tr>
<tr><td><code id="plot.spheremap_+3A_...">...</code></td>
<td>
<p>Further arguments are ignored. (accepted for compatibility with generic plot)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Produces a figure with 3 panels from a map object produced by <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>.
</p>
<p>Panel one shows the stress of the unconstrained MDS, the stress of the constrained 
MDS and the ratio of the two. A good rule of thumb is the stress from the constrained 
MDS should not be more than 10
</p>
<p>Panel 2 shows the final configuration of the unconstrained MDS which can be used to 
identify outliers.
</p>
<p>Panel 3 shows the final configuration of the constrained MDS in black and the 
unconstrained MDS in red. This can be used to check that the constrained fit 
is not distorting the data - large changes in the rank of a point in either 
dimension 1 or dimension 2 are indications of a problem with the fit.
</p>
<p>Panel 4 shows the pointwise nearest neighbour fits in order of the position 
in the estimated map.
</p>
<p>If <code>D1lim</code> or <code>D2lim</code> is not specified, then limits of panels 2 and 
3 are defined by <code><a href="smacof.html#topic+plot.smacof">plot.smacof</a></code>. 
</p>
<p>Markers are assigned numbers according to the order in which they occur in the 
input file. The locikey output of the map object is a data frame associating 
marker names with their numbers. This can be accessed using <code>pcmap3d$locikey</code>. 
If <code>displaytext=FALSE</code> then in panels 2 and 3 markers will be labelled by 
these numbers. By default <code>displaytext=TRUE</code> and markers are labelled by 
marker name.
</p>


<h3>References</h3>

<p><cite>de Leeuw J, Mair P (2009) Multidimensional scaling using majorization: SMACOF in R. J Stat Softw 31: 1-30</cite> <a href="http://www.jstatsoft.org/v31/i03/">http://www.jstatsoft.org/v31/i03/</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot.pcmap">plot.pcmap</a></code>, <code><a href="#topic+plot.pcmap3d">plot.pcmap3d</a></code>, <code><a href="smacof.html#topic+plot.smacof">plot.smacof</a></code>, <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>map&lt;-calc.maps.sphere(system.file("extdata", "lgI.txt", package="MDSMap"),
weightfn='lod', mapfn='kosambi')
plot(map)
</code></pre>

<hr>
<h2 id='recalc.nnfit.from.map'>Calculate a nearest neighbour fit from an estimated map object.</h2><span id='topic+recalc.nnfit.from.map'></span>

<h3>Description</h3>

<p>Calculates a new nearest neighbour fit based on a new order from a map object 
generated by <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>, <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code> or 
<code><a href="#topic+estimate.map">estimate.map</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>recalc.nnfit.from.map(estmap, mapobject, header = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="recalc.nnfit.from.map_+3A_estmap">estmap</code></td>
<td>
<p>A character string indicating the name of a comma separated value 
file with  the first column containing marker names in the order of their estimated position.</p>
</td></tr>
<tr><td><code id="recalc.nnfit.from.map_+3A_mapobject">mapobject</code></td>
<td>
<p>A map object generated by <code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>, 
<code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code> or <code><a href="#topic+estimate.map">estimate.map</a></code>.</p>
</td></tr>
<tr><td><code id="recalc.nnfit.from.map_+3A_header">header</code></td>
<td>
<p>Logical argument indicating whether the .csv file <code>estmap</code> 
contains headers - default is <code>TRUE</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads in a new estimated order, reorders the distance map and LOD scores by 
the new order and recalculates the nearest neighbour fit.
</p>


<h3>Value</h3>

<p>A list with the elements:
</p>
<table role = "presentation">
<tr><td><code>fit</code></td>
<td>
<p>Sum over all markers of the nearest neighbour fits.</p>
</td></tr>
<tr><td><code>pointfits</code></td>
<td>
<p>The nearest neighbour fit for each marker.</p>
</td></tr>
<tr><td><code>meanfit</code></td>
<td>
<p>Meanv of the nearest neighbour fits over all markers.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+calc.maps.pc">calc.maps.pc</a></code>, <code><a href="#topic+calc.maps.sphere">calc.maps.sphere</a></code>, <code><a href="#topic+estimate.map">estimate.map</a></code>, <code><a href="#topic+calc.nnfit">calc.nnfit</a></code>
</p>

<hr>
<h2 id='sim.bc.rflod.file'>Simulate a backcross population from homozygous parents.</h2><span id='topic+sim.bc.rflod.file'></span>

<h3>Description</h3>

<p>Simulates a backcross population from homozygous parents and writes a file 
containing the number of markers and observed pairwise distances, the pairwise 
recombination fractions and LOD scores in a text file suitable for analysis by 
other functions in the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.bc.rflod.file(fname)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim.bc.rflod.file_+3A_fname">fname</code></td>
<td>
<p>a character string specifying the base name of the file fname.txt to which the data should be written</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function simply generates data for use with the vignette. The R/qtl package is used to simulate a backcross #'population of 200 individuals from homozygous parents with 200 markers in a single linkage group of length #'100cM. The recombination fractions and LOD scores are calculated. The data is written to a text file in the #'format of output from JoinMap 4.  In particular, the data is cast into a data frame with marker names in the #'first two columns, pairwise recombination fractions in the third column and associated LOD scores in the fourth #'column. The data is written to a text file 'fname.txt' where the first row contains two entries - the number of #'markers and the number of pairwise observations. Below this the data frame containing the distance data is #'appended with no column headings. 
</p>

<table>
<tr>
 <td style="text-align: left;">
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>nmarkers</code> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>marker_1</code> </td><td style="text-align: left;"> <code>marker_2</code> </td><td style="text-align: left;"> <code>recombination fraction</code> </td><td style="text-align: left;"> <code>lod</code></td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>2</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>1</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>3</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>2</code> </td><td style="text-align: left;"> <code>4</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td><td style="text-align: left;"> <code>.</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Value</h3>

<p>No output - just the text file as above
</p>


<h3>References</h3>

<p><cite>Broman KW, Wu H, Sen S, Churchill GA (2003) R/qtl: QTL mapping in experimental crosses. Bioinformatics. 189: 889-890</cite>
<cite>Van Ooijen JW (2006) JoinMap 4; Software for the calculation of genetic linkage maps in experimental populations. Wageningen; Netherlands: Kyazma B.V</cite>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
