<!DOCTYPE html><html><head><title>Help for package medicaldata</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {medicaldata}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#blood_storage'><p>Retrospective Cohort Study of the Effects of Blood Storage on Prostate Cancer</p></a></li>
<li><a href='#covid_testing'><p>Deidentified Results of COVID-19 testing at the Children's Hospital of Pennsylvania (CHOP) in 2020</p></a></li>
<li><a href='#cytomegalovirus'><p>Retrospective Cohort Study of the Effects of Donor KIR genotype on the reactivation of cytomegalovirus (CMV) after myeloablative allogeneic hematopoietic stem cell transplant.</p>
<br /></a></li>
<li><a href='#esoph_ca'><p>esoph_ca: Esophageal Cancer dataset</p></a></li>
<li><a href='#indo_rct'><p>RCT of Indomethacin for Prevention of Post-ERCP Pancreatitis</p></a></li>
<li><a href='#indometh'><p>Cohort Study of the Pharmacokinetics of Intravenous Indomethacin</p></a></li>
<li><a href='#laryngoscope'><p>Randomized, Comparison Trial of Video vs. Standard Laryngoscope</p></a></li>
<li><a href='#licorice_gargle'><p>Randomized, Controlled Trial of Licorice Gargle before Intubation for Elective Thoracic Surgery</p></a></li>
<li><a href='#opt'><p>Obstetrics and Periodontal Therapy Dataset <br /></p></a></li>
<li><a href='#polyps'><p>RCT of Sulindac for Polyp Prevention in Familial Adenomatous Polyposis</p></a></li>
<li><a href='#scurvy'><p>Randomized Trial of Six Therapies for Scurvy</p></a></li>
<li><a href='#smartpill'><p>Prospective Cohort Study of Intestinal Transit using a SmartPill to Compare Trauma Patients to Healthy Volunteers</p></a></li>
<li><a href='#strep_tb'><p>RCT of Streptomycin Therapy for Tuberculosis</p></a></li>
<li><a href='#supraclavicular'><p>Study of Supraclavicular Anesthesia</p></a></li>
<li><a href='#theoph'><p>Cohort Study of the Pharmacokinetics of Oral Theophylline</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Data Package for Medical Datasets</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-08-08</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides access to well-documented medical datasets for teaching.
    Featuring several from the Teaching of Statistics in the Health Sciences 
    website <a href="https://www.causeweb.org/tshs/category/dataset/">https://www.causeweb.org/tshs/category/dataset/</a>, a few reconstructed datasets of historical significance in medical
    research, some reformatted and extended from existing R packages, 
    and some data donations. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://higgi13425.github.io/medicaldata/">https://higgi13425.github.io/medicaldata/</a>,
<a href="https://github.com/higgi13425/medicaldata/">https://github.com/higgi13425/medicaldata/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/higgi13425/medicaldata/issues">https://github.com/higgi13425/medicaldata/issues</a></td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, markdown, learnr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-08-12 12:06:51 UTC; peterhiggins</td>
</tr>
<tr>
<td>Author:</td>
<td>Peter Higgins <a href="https://orcid.org/0000-0002-7111-0077"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Peter Higgins &lt;higgi13425@yahoo.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-08-16 07:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='blood_storage'>Retrospective Cohort Study of the Effects of Blood Storage on Prostate Cancer</h2><span id='topic+blood_storage'></span>

<h3>Description</h3>

<p>This data set contains data on 316 men who had undergone radical prostatectomy and received transfusion during or within 30 days of the surgical procedure and had available prostate serum antigen (PSA) follow-up data. The main exposure of interest was RBC storage duration group. A number of demographic, baseline and prognostic factors were also collected. The outcome was time to biochemical (PSA) cancer recurrence. The dataset is cleaned and complete. There are no outliers or data problems (more details after variable information).
<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blood_storage
</code></pre>


<h3>Format</h3>

<p>A data frame with 316 observations and 20 variables
</p>

<dl>
<dt><code>RBC.Age.Group</code></dt><dd><p>NA, numeric, range: 1.00-  3</p>
</dd>
<dt><code>Median.RBC.Age</code></dt><dd><p>NA, numeric, range:10.00- 25</p>
</dd>
<dt><code>Age</code></dt><dd><p>NA, numeric, range:38.40- 79</p>
</dd>
<dt><code>AA</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>FamHx</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>PVol</code></dt><dd><p>NA, numeric, range:19.40-274</p>
</dd>
<dt><code>TVol</code></dt><dd><p>NA, numeric, range: 1.00-  3</p>
</dd>
<dt><code>T.Stage</code></dt><dd><p>NA, numeric, range: 1.00-  2</p>
</dd>
<dt><code>bGS</code></dt><dd><p>NA, numeric, range: 1.00-  3</p>
</dd>
<dt><code>BN+</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>OrganConfined</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>PreopPSA</code></dt><dd><p>NA, numeric, range: 1.30- 40</p>
</dd>
<dt><code>PreopTherapy</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>Units</code></dt><dd><p>NA, numeric, range: 1.00- 19</p>
</dd>
<dt><code>sGS</code></dt><dd><p>NA, numeric, range: 1.00-  4</p>
</dd>
<dt><code>AnyAdjTherapy</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>AdjRadTherapy</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>Recurrence</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>Censor</code></dt><dd><p>NA, numeric, range: 0.00-  1</p>
</dd>
<dt><code>TimeToRecurrence</code></dt><dd><p>NA, numeric, range: 0.27-104</p>
</dd></dl>



<h3>Details</h3>



<h4>Background</h4>

<p>Prostate cancer is the most common malignant neoplasm in men, and radical prostatectomy is among the primary therapies for localized prostate cancer. The biochemical recurrence-free survival rate 5 years after prostatectomy ranges from 70% to 90%. Improvements in the surgical technique have decreased the amount of intraoperative blood loss occurring during radical prostatectomy; however, substantial numbers of patients still require perioperative blood transfusions.<br />
Blood transfusions are associated with adverse reactions, including postoperative infections and transfusion-related immune perturbations. Allogeneic leukocytes present in the transfused blood are thought to suppress host cellular immune responses. Furthermore, the immunodepressant effect is secondary to an imbalance of accumulated cytokines and proinflammatory mediators in the transfused blood against decreased production of lymphocyte stimulating cell-mediated cytokines, such as interleukin 2 and increased release of immunosuppressive prostaglandins in the patient undergoing transfusion.<br />
In cancer patients, perioperative blood transfusion has long been suspected of reducing long-term survival, but available evidence is inconsistent. It is also unclear which components of transfused blood underlie the cancer-promoting effects reported by some studies. An important factor associated with the deleterious effects of blood transfusion is the storage age of the transfused blood units. It is suspected that cancer recurrence may be worsened after the transfusion of older blood.<br />
<br />
</p>



<h4>Study Objective</h4>

<p>This study evaluated the association between red blood cells (RBC) storage duration and biochemical prostate cancer recurrence after radical prostatectomy. Specifically, tested was the hypothesis that perioperative transfusion of allogeneic RBCs stored for a prolonged period is associated with earlier biochemical recurrence of prostate cancer after prostatectomy.<br />
<br />
</p>



<h4>Subjects and Variables</h4>

<p>Patients were assigned to 1 of 3 RBC age exposure groups on the basis of the terciles (ie, the 33rd and 66th percentiles) of the overall distribution of RBC storage duration if all their transfused units could be loosely characterized as of 'younger,' 'middle,' or 'older' age. Although this approach resulted in the removal of certain patients with wide RBC age distributions, it has the advantage of defining an essentially random and clearly separable exposure.<br />
Prostate-specific antigen (PSA) was used as a biochemical marker of prostate cancer recurrence after prostatectomy. A PSA value of at least 0.4 ng/mL (to convert to microg/L, multiply by 1.0) followed by another increase was considered biochemical cancer recurrence.<br />
<br />
The initial population consisted of 865 men who had undergone radical prostatectomy and received transfusion during or within 30 days of the surgical procedure at Cleveland Clinic and had available PSA follow-up data. Of these patients, 110 were excluded from the analysis because they received a combination of allogeneic and autologous blood products. Of the remaining 755 patients, 405 (54%) received solely allogeneic and 350 patients (46%) received solely autologous RBC units. Of the 405 patients who received allogeneic RBC transfusion, 89 were excluded because their transfused RBC age distribution included more than one of the terciles. Thus, this dataset consists of the 316 patients who received solely allogeneic blood products and could be classified into an RBC age exposure group.<br />
</p>



<h3>Source</h3>

<p>Cata et al. 'Blood Storage Duration and Biochemical Recurrence of Cancer after Radical Prostatectomy'. Mayo Clin Proc 2011; 86(2): 120-127.
</p>

<hr>
<h2 id='covid_testing'>Deidentified Results of COVID-19 testing at the Children's Hospital of Pennsylvania (CHOP) in 2020</h2><span id='topic+covid_testing'></span>

<h3>Description</h3>

<p>A dataset containing details of SARS-CoV-2 testing in 2020 at CHOP
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covid_testing
</code></pre>


<h3>Format</h3>

<p>A data frame with 15524 observations and 17 variables
</p>

<dl>
<dt>subject_id</dt><dd><p>id number for each subject; type: numeric</p>
</dd>
<dt>fake_first_name</dt><dd><p>an auto-generated fake first name; type: character</p>
</dd>
<dt>fake_last_name</dt><dd><p>an auto-generated fake last name; character</p>
</dd>
<dt>gender</dt><dd><p>anonymized Gender, levels: female, male; type: character</p>
</dd>
<dt>pan_day</dt><dd><p>day after start of pandemic; type: numeric</p>
</dd>
<dt>test_id</dt><dd><p>test that was performed, levels: covid, xcvd1; type: character</p>
</dd>
<dt>clinic_name</dt><dd><p>Clinic or ward where the specimen was collected, 88 levels; type: character</p>
</dd>
<dt>result</dt><dd><p>result of test, levels: positive, negative, invalid; type: character</p>
</dd>
<dt>demo_group</dt><dd><p>patient group, levels: patient, misc_adult, client, other adult, unidentified; type: character</p>
</dd>
<dt>age</dt><dd><p>Age of subject at time of specimen collection (Anonymized), units = years; type: numeric</p>
</dd>
<dt>drive_thru_ind</dt><dd><p>Whether the specimen was collected via a drive-thru site, levels: 1: Collected at drive-thru site; 0: Not collected at drive-thru site; type: numeric</p>
</dd>
<dt>ct_result</dt><dd><p>Cycle at which threshold reached during PCR, range: 14.05-45; type: numeric</p>
</dd>
<dt>orderset</dt><dd><p>Whether an order set was used for test order, levels: 1: Collected via orderset; 0: Not collected via orderset; numeric</p>
</dd>
<dt>payor_group</dt><dd><p>Payor associated with order, levels: commercial, government, unassigned, medical assistance, self pay, charity care, other; type: character</p>
</dd>
<dt>patient_class</dt><dd><p>Disposition of subject at time of collection, levels: inpatient, emergency, observation, recurring outpatient, outpatient, not applicable, day surgery, admit after surgery-obs, admit after surgery-ip; type: character</p>
</dd>
<dt>col_rec_tat</dt><dd><p>Time elapsed between collect time and receive time, range: 0 - 61370.2, units = hours; type: numeric</p>
</dd>
<dt>rec_ver_tat</dt><dd><p>Time elapsed between receive time and verification time, range: -18.6 - 218.2, units = hours; type: numeric</p>
</dd>
</dl>
<p>...

</p>


<h3>Details</h3>

<p>Data on testing for SARS-CoV2 from days 4-107 of the COVID pandemic in 2020. CHOP is a pediatric hospital in Philadelphia, Pennsylvania, USA. These data have been anonymized, time-shifted, and permuted.
</p>


<h3>Source</h3>

<p>This data set is from Amrom E. Obstfeld, who de-identified data on COVID-19 testing during 2020 at CHOP (Children's Hospital of Pennsylvania). This data set contains data concerning testing for SARS-CoV2 via PCR as well as associated metadata. These data have been anonymized, time-shifted, and permuted.
</p>

<hr>
<h2 id='cytomegalovirus'>Retrospective Cohort Study of the Effects of Donor KIR genotype on the reactivation of cytomegalovirus (CMV) after myeloablative allogeneic hematopoietic stem cell transplant.
<br /></h2><span id='topic+cytomegalovirus'></span>

<h3>Description</h3>

<p>This data set contains 64 consecutive patients who underwent T-cell replete, matched sibling donor reduced-intensity conditioning allogeneic hematopoietic stem cell transplant. The primary risk factor of interest was the number of activating killer immunoglobulin-like receptors (aKIRs: 1-4 vs. 5-6). (more details after variable information). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cytomegalovirus
</code></pre>


<h3>Format</h3>

<p>A data frame with 64 observations and 26 variables
</p>

<dl>
<dt><code>ID</code></dt><dd><p>Patient ID, numeric, range:  1-64</p>
</dd>
<dt><code>age</code></dt><dd><p>Recipient age at transplant, numeric, range: 29-67</p>
</dd>
<dt><code>sex</code></dt><dd><p>Recipient sex, numeric, range:  0 (male) - 1(female)</p>
</dd>
<dt><code>race</code></dt><dd><p>Recipient race, numeric, range:  0 (white) - 1 (african-american)</p>
</dd>
<dt><code>diagnosis</code></dt><dd><p>type: character, levels: 13</p>
</dd>
<dt><code>diagnosis.type</code></dt><dd><p>Category of cancer diagnosis, numeric, range:  0 (myeloid) - 1 (lymphoid)</p>
</dd>
<dt><code>time.to.transplant</code></dt><dd><p>Time from cancer diagnosis to transplant (months), numeric, range:  1.84-173.8</p>
</dd>
<dt><code>prior.radiation</code></dt><dd><p>Prior radiation therapy, numeric, range:  0 (no) - 1 (yes)</p>
</dd>
<dt><code>prior.chemo</code></dt><dd><p>Number of prior chemotherapy regimens, numeric, range:  0-8</p>
</dd>
<dt><code>prior.transplant</code></dt><dd><p>Prior stem cell transplant, numeric, range:  0 (no) - 1 (yes)</p>
</dd>
<dt><code>recipient.cmv</code></dt><dd><p>Recipient cytomegalovirus seropositive status, numeric, range:  0 (negative) - 1 (positive)</p>
</dd>
<dt><code>donor.cmv</code></dt><dd><p>Donor cytomegalovirus seropositive status, numeric, range:  0 (negative) - 1 (positive)</p>
</dd>
<dt><code>donor.sex</code></dt><dd><p>Donor sex, numeric, range:  0 (female) - 1 (male)</p>
</dd>
<dt><code>TNC.dose</code></dt><dd><p>Total nucleated cell dose (x 10^8/kg), numeric, range:  2.06- 21.0</p>
</dd>
<dt><code>CD34.dose</code></dt><dd><p>Total CD34+ (stem) cell dose (x 10^8/kg), numeric, range:  2.04- 12.5</p>
</dd>
<dt><code>CD3.dose</code></dt><dd><p>Total CD3+ (T) cell dose (x 10^8/kg), numeric, range:  1.08-  8.2</p>
</dd>
<dt><code>CD8.dose</code></dt><dd><p>Total CD8+ cell dose (x 10^8/kg), numeric, range:  0.16-  3.2</p>
</dd>
<dt><code>TBI.dose</code></dt><dd><p>Total body irradiation dosage (centiGrays), numeric, range:200.00-400.0</p>
</dd>
<dt><code>C1/C2</code></dt><dd><p>HLA-Cw group, numeric, range:  0 (heterozygous) - 1 (homozygous)</p>
</dd>
<dt><code>aKIRs</code></dt><dd><p>Number of donor activating killer immunoglobulin-like receptors (hypothesized Predictor), numeric, range:  1.00-  6.0</p>
</dd>
<dt><code>cmv</code></dt><dd><p>cytomegalovirus reactivation posttransplant (hypothesized Outcome), numeric, range:  0 (No) - 1 (Yes)</p>
</dd>
<dt><code>time.to.cmv</code></dt><dd><p>Time to cytomegalovirus reactivation (months), numeric, range:  0.43- 84.5</p>
</dd>
<dt><code>agvhd</code></dt><dd><p>Acute level 2-4 graft versus host disease, numeric, range:  0 (no) - 1 (yes)</p>
</dd>
<dt><code>time.to.agvhd</code></dt><dd><p>Time to acute level 2-4 graft versus host disease (months), numeric, range:  0.66- 85.2</p>
</dd>
<dt><code>cgvhd</code></dt><dd><p>Chronic graft versus host disease, numeric, range:  0 (no) - 1 (yes)</p>
</dd>
<dt><code>time.to.cgvhd</code></dt><dd><p>Time to chronic graft versus host disease (months), numeric, range:  0.82- 65.1</p>
</dd></dl>



<h3>Details</h3>

<p>A number of demographic, baseline and transplant characteristics were also collected. The primary outcome is presence of and time to cytomegalovirus reactivation. The dataset is cleaned and relatively complete. There are no outliers or data problems.
<br />
<br />
</p>


<h4>Background<br /></h4>

<p>Hematopoietic stem cell transplantation (HSCT) is the transplantation of multipotent hematopoietic stem cells, from bone marrow, peripheral blood, or umbilical cord blood. It is a medical procedure most often performed for patients with certain cancers of the blood or bone marrow, such as multiple myeloma or leukemia. Allogeneic HSCT involves two people: the (healthy) donor and the (patient) recipient. Allogeneic HSC donors must have a tissue (HLA) type that matches the recipient. <br />
In myeloablative allogeneic HSCT, chemotherapy or irradiation is given immediately prior to a transplant (the conditioning regimen) with the purpose of eradicating the patient's disease prior to the infusion of HSC and to suppress immune reactions. The bone marrow can be ablated (destroyed) with dose- levels that cause minimal injury to other tissues. For many patients who are at high risk for transplant-related mortality with myeloablative allogeneic HSCT, reduced- intensity conditioning allogeneic hematopoietic stem cell transplant has proven effective. Although the reduced-intensity conditioning allogeneic HSCT may avoid many of the organ toxicities associated with myeloablative conditioning, the risk for developing graft-versus-host disease and infection including cytomegalovirus remains significant.<br />
Cytomegalovirus (CMV) is a common virus that can infect almost anyone. Once infected, your body retains the virus for life. Most people don't know they have CMV because it rarely causes problems in healthy people. But if pregnant or having a weakened immune system, CMV is cause for concern. For people with compromised immunity, such as after allogeneic HSCT, CMV infection can be fatal. Natural killer (NK) and T cells provide protection against CMV reactivation. The reactivity of NK cells and some T-cell subsets are regulated by the interaction of killer immunoglobulin-like receptors (KIRs) with target cell HLA class 1 molecules. The donor activating KIR genotype has been implicated as a contributing factor for CMV reactivation after myeloablative allogeneic HSCT.<br />
<br />
</p>



<h4>Study Objective</h4>

<p>This study investigates whether donor KIR genotype influences reactivation of CMV after T-cell replete, matched sibling donor reduced-intensity conditioning allogeneic HSCT.<br />
<br />
</p>



<h4>Subjects and Variables</h4>

<p>The study included 64 consecutive patients who underwent T-cell replete, matched sibling donor reduced-intensity conditioning allogeneic hematopoietic stem cell transplant between January 16, 2000 and April 24, 2007 at the Cleveland Clinic.
Human leucocyte antigen (HLA) typing on donors and recipients was performed to allow assessment of killer immunoglobulin-like receptor ligands (KIRs). To allow for comparison with previous studies, donors were categorized as having 1-4 or 5-6 activating killer immunoglobulin-like receptor genes (aKIRs).<br />
CMV reactivation was defined as any detection of cytomegalovirus DNA in the blood; the lower detection limit for this assay was 600 copies/mL.<br />
<br />
The initial population consisted of 865 men who had undergone radical prostatectomy and received transfusion during or within 30 days of the surgical procedure at Cleveland Clinic and had available PSA follow-up data. Of these patients, 110 were excluded from the analysis because they received a combination of allogeneic and autologous blood products. Of the remaining 755 patients, 405 (54%) received solely allogeneic and 350 patients (46%) received solely autologous RBC units. Of the 405 patients who received allogeneic RBC transfusion, 89 were excluded because their transfused RBC age distribution included more than one of the terciles. Thus, this dataset consists of the 316 patients who received solely allogeneic blood products and could be classified into an RBC age exposure group.<br />
</p>



<h3>Source</h3>

<p>Sobecks et al. 'Cytomegalovirus Reactivation After Matched Sibling Donor Reduced-Intensity Conditioning Allogeneic Hematopoietic Stem Cell Transplant Correlates With Donor Killer Immunoglobulin-like Receptor Genotype'. Exp Clin Transplant 2011; 1: 7-13.
</p>

<hr>
<h2 id='esoph_ca'>esoph_ca: Esophageal Cancer dataset</h2><span id='topic+esoph_ca'></span>

<h3>Description</h3>

<p>Data from a case-control study of esophageal cancer in Ille-et-Vilaine, France, evaluating the effects of smoking and alcohol on the incidence of esophageal cancer. Smoking and alcohol are associated risk factors for squamous cell cancer of the esophagus, rather than adenocarcinoma of the esophagus, which is associated with obesity and esophageal reflux (more details available below the variable definitions).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>esoph_ca
</code></pre>


<h3>Format</h3>

<p>A data frame with 88 rows and 5 variables, with 200 cases and 975 controls.
</p>

<dl>
<dt>agegp</dt><dd><p>6 levels of age: &quot;25-34&quot;, &quot;35-44&quot;, &quot;45-54&quot;, &quot;55-64&quot;, &quot;65-74&quot;, &quot;75+&quot;; type: ordinal factor</p>
</dd>
<dt>alcgp</dt><dd><p>4 levels of alcohol consumption: &quot;0-39g/day&quot;, &quot;40-79&quot;, &quot;80-119&quot;, &quot;120+&quot;; type: ordinal factor</p>
</dd>
<dt>tobgp</dt><dd><p>4 levels of tobacco consumption: &quot;0-9g/day&quot;, &quot;10-19&quot;, '20-29&quot;, &quot;30+&quot;; type: ordinal factor</p>
</dd>
<dt>ncases</dt><dd><p>Number of cases; type: integer</p>
</dd>
<dt>ncontrols</dt><dd><p>Number of controls; type: integer</p>
</dd>
</dl>



<h3>Details</h3>

<p>An original base R dataset, though of somewhat unclear origin. The statistical textbook source is clear, though it is not clear which of the original epidemiological papers on esophageal cancer in Ille-et-Vilaine is referred to by this dataset. The original authors of the medical study were <strong>not</strong> credited in the base R dataset. There are several possible papers in PubMed, none of which quite match up with this dataset. This could be from Tuyns, AJ, et al., Bull Cancer, 1977;64(1):45-60, but this paper reports 778 controls, rather than the 975 found here. A 1975 paper from the same group reported 718 cases (Int J Epidemiol, 1975 Mar;4(1):55-9.  doi: 10.1093/ije/4.1.55.). There is also another possible source - a 1975 paper from the same group, <em>Usefulness of population controls in retrospective studies of alcohol consumption. Experience from a case&ndash;control study of esophageal cancer in Ille-et-Vilaine, France</em>, Journal of Studies on Alcohol, 39(1): 175-182 (1978), which is behind a publisher paywall.
</p>


<h3>Source</h3>

<p>Breslow, N. E. and Day, N. E. (1980) Statistical Methods in Cancer Research. Volume 1: The Analysis of Case-Control Studies. IARC Lyon / Oxford University Press.
Originally in base R datasets.
</p>

<hr>
<h2 id='indo_rct'>RCT of Indomethacin for Prevention of Post-ERCP Pancreatitis</h2><span id='topic+indo_rct'></span>

<h3>Description</h3>

<p>Results of a randomized, placebo-controlled, prospective 2-arm trial of rectal indomethacin (100 mg) vs. placebo prevent post-ERCP pancreatitis in 602 participants, as reported by Elmunzer, Higgins, et al. in 2012 in the New England Journal of Medicine (more details available below the variable definitions).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indo_rct
</code></pre>


<h3>Format</h3>

<p>A data frame with 602 observations and 33 variables
</p>

<dl>
<dt>id</dt><dd><p>subject id, first integer indicates center, integer, range:1001-4003</p>
</dd>
<dt>site</dt><dd><p>study site (center), factor, 1 = University of Michigan, 2= Indiana University, 3 = University of Kentucky, 4 = Case Western</p>
</dd>
<dt>age</dt><dd><p>age in years, numeric, range:  19-90</p>
</dd>
<dt>risk</dt><dd><p>risk score, numeric, range:   1-5.5</p>
</dd>
<dt>gender</dt><dd><p>male or female, factor, levels: 1_female, 2_male</p>
</dd>
<dt>sod</dt><dd><p>sphincter of oddi dysfunction was present, a risk factor favoring post-ERCP pancreatitis, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>pep</dt><dd><p>previous post-ERCP pancreatitis (PEP), a risk factor for future PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>recpanc</dt><dd><p>Recurrent Pancreatitis, a risk factor for future PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>psphinc</dt><dd><p>a Pancreatic Sphincterotomy was performed, a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>precut</dt><dd><p>a sphincter pre-cut was needed to enter the papilla, a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>difcan</dt><dd><p>Cannulation of the papilla was difficult, a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>pneudil</dt><dd><p>Pneumatic dilation of the papilla was performed, a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>amp</dt><dd><p>An Ampullectomy was performed for dysplasia or cancer, which could be a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>paninj</dt><dd><p>Contrast was injected into the pancreas during the procedure, a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>acinar</dt><dd><p>The pancreas appeared to have acinarization on imaging, which could be a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>brush</dt><dd><p>Brushings were taken from the pancreatic duct, a possible risk factor favoring post-ERCP pancreatitis. factor, levels: 0_no, 1_yes</p>
</dd>
<dt>asa81</dt><dd><p>Aspirin was used at a dose of 81 mg per day, which may increase the risk of bleeding. factor, levels: 0_no, 1_yes</p>
</dd>
<dt>asa325</dt><dd><p>Aspirin was used at a dose of 325 mg per day, which may increase the risk of bleeding. factor, levels: 0_no, 1_yes</p>
</dd>
<dt>asa</dt><dd><p>Aspirin was used (at a dose of 325 mg per day(at any dose), which may increase the risk of bleeding. factor, levels: 0_no, 1_yes</p>
</dd>
<dt>prophystent</dt><dd><p>A pancreatic duct stent was placed at the end of the procedure per the judgement of the endoscopist (more often in high-risk cases), a potential protective effect against PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>therastent</dt><dd><p>A pancreatic duct stent was placed in order to treat a clinically significant narrowing of the pancreatic duct, a potential protective effect against PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>pdstent</dt><dd><p>A pancreatic duct stent was placed at the end of the procedure for any reason, a potential protective effect against PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>sodsom</dt><dd><p>Sphincter of oddi manometry was performed during the procedure for SOD, a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>bsphinc</dt><dd><p>A biliary sphincterotomy was performed, which could be a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>bstent</dt><dd><p>A biliary stent was placed to relieve significant biliary obstruction, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>chole</dt><dd><p>Choledocholithiasis (gallstones blocking the biliary duct) was present, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>pbmal</dt><dd><p>Malignancy of the biliary duct or pancreas was found, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>train</dt><dd><p>A trainee participated in the ERCP, which could be a risk factor for PEP, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>outcome</dt><dd><p>outcome of post-ercp pancreatitis, factor, levels: 0_no, 1_yes</p>
</dd>
<dt>status</dt><dd><p>outpatient status, factor, levels: 0_inpatient, 1_outpatient</p>
</dd>
<dt>type</dt><dd><p>Sphincter of Oddi dysfunction type/level - higher numbers are more severe with greater association with PEP, factor, levels: 0_no SOC, 1_type 1, 2_type 2, 3_type 3</p>
</dd>
<dt>rx</dt><dd><p>treatment arm, factor, levels: 0_placebo, 1_indomethacin</p>
</dd>
<dt>bleed</dt><dd><p>A gastrointestinal bleed occurred (which could be a complication of indomethacin therapy), factor, levels: 1. no, 2. yes</p>
</dd>
</dl>



<h3>Details</h3>

<p>ERCP, or endoscopic retrograde cholangio-pancreatogram, is a procedure performed by threading an endoscope through the mouth to the opening in the duodenum where bile and pancreatic digestive juices are released into the intestine. ERCP is helpful for treating blockages of flow of bile (gallstones, cancer), or diagnosing cancers of the pancreas, but has a high rate of complications (15-25%). <br />
The occurrence of post-ERCP pancreatitis is a common and feared complication, as pancreatitis can result in multisystem organ failure and death, and can occur in ~ 16% of ERCP procedures. <br />
The inflammatory cytokine storm that can result from this procedural complication can be quite severe. Several small randomized trials suggested that anti-inflammatory NSAID therapies at the time of ERCP could reduce the rate of this complication, but all were rather small single-center studies, and were not sufficiently convincing to change practice. <br />
Elmunzer, Higgins, and colleagues performed a <a href="https://pubmed.ncbi.nlm.nih.gov/18375470/">meta-analysis</a> of these small trials, which suggested that this was a significant effect, and that indomethacin could result in a 64% reduction in post-ERCP pancreatitis. <br />
The investigators took this as a possible over-estimate of the effect (due to publication bias), and designed a multicenter RCT of a planned 948 patients to see a reduction of 50% from a placebo rate of 10% to an indomethacin rate of 5%. Two interim analyses were performed, after 400 and 600 patients were enrolled, using an alpha spending function. The Data and Safety Monitoring Board stopped the study after 602 participants were enrolled because of the significantly positive effect of indomethacin, which reduced post-ERCP pancreatitis from 16% in the placebo group to 9% in the indomethacin group. <br />
You can find the manuscript at <a href="https://www.nejm.org/doi/full/10.1056/NEJMoa1111103">Indomethacin to Prevent Post-ERCP Pancreatitis</a>. <br />
</p>


<h3>Source</h3>

<p>This data set is sourced from the authors of the 2012 manuscript in the New England Journal of Medicine, entitled, A Randomized Trial of Rectal Indomethacin to Prevent Post-ERCP Pancreatitis, pages 1414-1422 volume 366, in the April 12, 2012 edition, authored by the Elmunzer, BJ, Higgins PDR, et al. You can find the manuscript at <a href="https://www.nejm.org/doi/full/10.1056/NEJMoa1111103">Indomethacin to Prevent Post-ERCP Pancreatitis</a>.
</p>

<hr>
<h2 id='indometh'>Cohort Study of the Pharmacokinetics of Intravenous Indomethacin</h2><span id='topic+indometh'></span>

<h3>Description</h3>

<p>Results of a Cohort Study of the Pharmacokinetics of Intravenous Indomethacin, with plasma concentrations over time (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>indometh
</code></pre>


<h3>Format</h3>

<p>A data frame with 66 observations and 3 variables
</p>

<dl>
<dt>Subject</dt><dd><p>subject id number for each participant; type: character</p>
</dd>
<dt>time</dt><dd><p>Time from initial dose in hours; type: double</p>
</dd>
<dt>conc</dt><dd><p>Concentration of indomethacin in the plasma in micrograms per milliliter' type: double</p>
</dd>
</dl>



<h3>Details</h3>

<p>This data set contains data on 6 healthy volunteer subjects who participated in a pharmacokinetic study of intravenous indomethacin. Indomethacin is an anti-inflammatory and pain-relieving non-steroidal medication. It can be administered by the intravenous, oral, or rectal suppository routes. Some of the indomethacin is excreted in the bile and reabsorbed by the intestine. This phenomenon, called enterohepatic circulation, keeps the drug around longer than would be expected otherwise. <br />
Each subject in Study 1 (intravenous route) received a single 50 mg dose of radioactively labeled indomethacin (^14^-carbon-labeled, with each dose containing 25 microCuries of radioactivity). Subjects received a standard meal (one 8-oz can of Metrecal, 8 oz of whole milk, and one medium-size apple) 30 rain prior to medication and 8 oz of water every 2 hr throughout the waking hours to ensure adequate urine output.<br />
Blood samples were taken at frequent intervals over the first 8 hours after dosing, and the quantity of indomethacin in the plasma (as well as stool and urine) at each time point was measured in micrograms per milliliter. This data set only contains the plasma measurements from Table 1 on page 258 of the manuscript. While this paper was published in 1976 (post-Tuskegee reveal), there is no mention of ethics review, IRB review, or consent of the healthy volunteers. <br />
The abstract from the original manuscript: <br />
There are no discernible quantitative differences in the biotransformation and the excretion of indomethacin following oral, rectal, and intravenous administration of indomethacin-2-^14^C. Approximately 50% (range 24-115% for n = 6) of an intravenous dose undergoes enterohepatic circulation. Thus the bioavailability of indomethacin to the systemic circulation may exceed the administered dose. Relative to the intravenous dose, indomethacin is 80 and 100% bioavailable from suppositories and capsules, respectively. Absorption and/or reabsorption appears to be more rapid and uniform by the rectal route. Recognition of the attributes of biliary recycling also helps to explain the observed variability in apparent plasma half-life, while their neglect requires alternative explanations for anomalies between the disappearance rate from plasma and the corresponding appearance rate in urine.<br />
</p>


<h3>Source</h3>

<p>Kwan, Breault, Umbenhauer, McMahon and Duggan (1976) Kinetics of Indomethacin absorption, elimination, and enterohepatic circulation in man. Pharmacokinetics and Biopharmaceutics. 1976 Jun;4(3):255-80.  doi: 10.1007/BF01063617.
</p>

<hr>
<h2 id='laryngoscope'>Randomized, Comparison Trial of Video vs. Standard Laryngoscope</h2><span id='topic+laryngoscope'></span>

<h3>Description</h3>

<p>This data set contains 99 adult patients with a body mass index between 30 and 50 kg/m2 who required orotracheal intubation for elective surgery. Patient demographics, airway assessment data, intubation success rate, time to intubation, ease of intubation, and occurrence of complications were recorded. The dataset is cleaned and complete. There are no outliers or data problems (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>laryngoscope
</code></pre>


<h3>Format</h3>

<p>A data frame with 99 observations and 22 variables
</p>

<dl>
<dt><code>age</code></dt><dd><p>Age (years), numeric, range: 20-77</p>
</dd>
<dt><code>gender</code></dt><dd><p>Gender, numeric, 0 = female; 1 = male</p>
</dd>
<dt><code>asa</code></dt><dd><p>American Society of Anesthesiologists physical status(1-4), range:  2-4</p>
</dd>
<dt><code>BMI</code></dt><dd><p>Body Mass Index (kg/m^2), numeric, range: 31-61</p>
</dd>
<dt><code>Mallampati</code></dt><dd><p>Mallampati score predicting ease of intubation 1 = Full visibility of tonsils, uvula and soft palate (easy intubation); 2 = Visibility of hard and soft palate, upper portion of tonsils and uvula; 3 = Soft and hard palate and base of the uvula are visible; 4 = Only Hard Palate visible (difficult intubation), numeric, range:  1-4</p>
</dd>
<dt><code>Randomization</code></dt><dd><p>Laryngoscope randomized, numeric, range:  0 = Standard Macintosh #4, 1 = AWS Pentaz Video</p>
</dd>
<dt><code>attempt1_time</code></dt><dd><p>First intubation attempt time (seconds), numeric, range:  9-113</p>
</dd>
<dt><code>attempt1_S_F</code></dt><dd><p>Successful intubation first attempt, numeric, 0 = no, 1 = yes</p>
</dd>
<dt><code>attempt2_time</code></dt><dd><p>Second intubation attempt time (seconds), numeric, range: 11- 60</p>
</dd>
<dt><code>attempt2_assigned_method</code></dt><dd><p>Second intubation attempt made with assigned laryngoscope, numeric, 0 = no, 1 = yes</p>
</dd>
<dt><code>attempt2_S_F</code></dt><dd><p>Successful intubation second attempt, numeric, 0 = no, 1 = yes, numeric, range:  0 = no, 1 = yes</p>
</dd>
<dt><code>attempt3_time</code></dt><dd><p>Third intubation attempt time (seconds), numeric, range: 15- 30</p>
</dd>
<dt><code>attempt3_assigned_method</code></dt><dd><p>Third intubation attempt made with assigned laryngoscope, numeric, 0 = no, 1 = yes</p>
</dd>
<dt><code>attempt3_S_F</code></dt><dd><p>Successful intubation third attempt, numeric, 0 = no, 1 = yes, numeric, range:  1-1</p>
</dd>
<dt><code>attempts</code></dt><dd><p>Number of intubation attempts, numeric, range:  1-3</p>
</dd>
<dt><code>failures</code></dt><dd><p>Number of intubation failures, numeric, range:  0-2</p>
</dd>
<dt><code>total_intubation_time</code></dt><dd><p> Total Intubation time (second), numeric, range:  9-100</p>
</dd>
<dt><code>intubation_overall_S_F</code></dt><dd><p>Overall successful intubation, numeric, 0 = no, 1 = yes</p>
</dd>
<dt><code>bleeding</code></dt><dd><p>Bleeding (trace), numeric, 0 = no, 1 = yes</p>
</dd>
<dt><code>ease</code></dt><dd><p>Ease of tracheal intubation, 0 = extremely easy to 100 = extremely difficult, numeric, range:  0-100</p>
</dd>
<dt><code>sore_throat</code></dt><dd><p>Severity of postoperative sore throat, 0 = none; 1 = mild; 2 = moderate; 3 = severe, numeric, range:  0-  3</p>
</dd>
<dt><code>view</code></dt><dd><p>Cormack-Lehane grade of glottic view 0 = &quot;not good&quot; Cormack- Lehane grade 1 or 2; 1 = &quot;good&quot; Cormack-Lehane grade 3 or 4,  numeric, range:  0-  1</p>
</dd>
</dl>



<h3>Details</h3>

<p>The Laryngoscope dataset was contributed by Dr. Amy Nowacki, Associate Professor, Cleveland Clinic. Please refer to this resource as: Amy S. Nowacki, 'Laryngoscope Dataset', TSHS Resources Portal (2017). Available at https://www.causeweb.org/tshs/laryngoscope/. <br />
Difficult and failed tracheal intubations are among the principal causes of anesthetic-related mortality and morbidity. Because a good laryngeal view facilitates successful tracheal intubation, new technologies have been introduced to improve visualization. Video laryngoscopes, for example, often use miniature cameras to facilitate visualization of the laryngeal inlet with no need to align the oral, pharyngeal, and tracheal axes.<br />
The Pentax AWS is a novel video laryngoscope, available in Japan since 2006, which is designed to facilitate intubation by providing a video image of the glottis. It incorporates a miniature video camera and a battery-powered, built-in LCD monitor. A disposable blade is attached to the base system. Incorporation of an LCD display makes it possible to view the glottis simultaneously with insertion of the endotracheal tube (ETT). In this regard, it differs from some other video laryngoscope designs that use external monitors. The Pentax AWS also differs in having a side channel that positions and guides the ETT. Reports suggest that the Pentax AWS can help intubate, but randomized data remain sparse.cr
This study tested the hypothesis that intubation with the Pentax AWS would be easier and faster than with a standard Macintosh laryngoscope with a #4 blade.<br />
<br />
</p>


<h3>Source</h3>

<p>These are data from a study by Abdallah et al. A Randomized Comparison between the Pentax AWS Video Laryngoscope and the Macintosh Laryngoscope in Morbidly Obese Patients. Anesthesia Analgesia 2011; 113: 1082-7.
</p>

<hr>
<h2 id='licorice_gargle'>Randomized, Controlled Trial of Licorice Gargle before Intubation for Elective Thoracic Surgery</h2><span id='topic+licorice_gargle'></span>

<h3>Description</h3>

<p>This study enrolled 236 adult patients undergoing elective thoracic surgery requiring a double-lumen endotracheal tube. Gender, physical status, BMI, age, Mallampati score, smoking status, preoperative pain, surgery size, intervention and the outcomes (cough, sore throat and pain swallowing at various time points) are provided. The dataset is cleaned and complete (missing outcomes for 2 patients). There are no outliers or data problems (<strong>more details</strong> available below the variable definitions).<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>licorice_gargle
</code></pre>


<h3>Format</h3>

<p>A data frame with 235 observations and 19 variables
</p>

<dl>
<dt><code>preOp_gender</code></dt><dd><p>Gender, numeric, 0 = Male; 1 = Female</p>
</dd>
<dt><code>preOp_asa</code></dt><dd><p>American Society of Anesthesiologists physical status, numeric, 1 = a normal healthy patient; 2 = a patient with mild systemic disease; 3 = a patient with severe systemic disease</p>
</dd>
<dt><code>preOp_calcBMI</code></dt><dd><p>Body mass index (kg/m^2), numeric, range:16-36</p>
</dd>
<dt><code>preOp_age</code></dt><dd><p>Age (years), numeric, range:18-86</p>
</dd>
<dt><code>preOp_mallampati</code></dt><dd><p>Mallampati score, with 1 = easy to intubate, 4= difficult intubation, numeric, 1 = soft palate, fauces, uvula, pillars visible; 2 = soft palate, fauces, uvula visible; 3 = soft palate, base of uvula visible; 4 = soft palate not visible at all</p>
</dd>
<dt><code>preOp_smoking</code></dt><dd><p>Smoking status, numeric, 1 = Current; 2 = Past; 3 = Never</p>
</dd>
<dt><code>preOp_pain</code></dt><dd><p>Preoperative pain, numeric, 0 = No; 1 = Yes</p>
</dd>
<dt><code>treat</code></dt><dd><p>Intervention, 0 = Sugar 5g; 1 = Licorice 0.5g</p>
</dd>
<dt><code>intraOp_surgerySize</code></dt><dd><p>Surgery size, numeric, 1 = Small (thoracoscopy); 2 = Medium (thoracotomy &lt; 3 h); 3 = Large (thoracotomy &gt; 3 h or blood loss &gt; 1000 mL)</p>
</dd>
<dt><code>extubation_cough</code></dt><dd><p>Amount of coughing immediately after extubation, numeric, 0 = No cough; 1 = Mild; 2 = Moderate; 3 = Severe</p>
</dd>
<dt><code>pacu30min_cough</code></dt><dd><p>Amount of coughing at 30 minutes after arrival in PACU, numeric, 0 = No cough; 1 = Mild; 2 = Moderate; 3 = Severe</p>
</dd>
<dt><code>pacu30min_throatPain</code></dt><dd><p>Sore throat pain score at rest at 30 minutes after arrival in PACU (11 point Likert scale, 0=no pain, 10 = worst pain)</p>
</dd>
<dt><code>pacu30min_swallowPain</code></dt><dd><p>Sore throat pain score during swallowing at 30 minutes after arrival in PACU (11 point Likert scale, 0=no pain, 10 = worst pain), numeric, range: 0-10</p>
</dd>
<dt><code>pacu90min_cough</code></dt><dd><p>Amount of coughing at 90 minutes after arrival in PACU, numeric, 0 = No cough; 1 = Mild; 2 = Moderate; 3 = Severe</p>
</dd>
<dt><code>pacu90min_throatPain</code></dt><dd><p>Sore throat pain score at rest at 90 minutes after arrival in PACU (11 point Likert scale, 0=no pain, 10 = worst pain), numeric, range: 0-6)</p>
</dd>
<dt><code>postOp4hour_cough</code></dt><dd><p>Amount of coughing at 4 hours after surgery, numeric, 0 = No cough; 1 = Mild; 2 = Moderate; 3 = Severe, range: 0-2</p>
</dd>
<dt><code>postOp4hour_throatPain</code></dt><dd><p>Sore throat pain score at rest at 4 hours after surgery (11 point Likert scale, 0=no pain, 10 = worst pain), numeric, range: 0-6), numeric, range: 0- 7</p>
</dd>
<dt><code>pod1am_cough</code></dt><dd><p>Amount of coughing on the first postoperative morning, 0 = No cough; 1 = Mild; 2 = Moderate; 3 = Severe, numeric, range: 0- 3</p>
</dd>
<dt><code>pod1am_throatPain</code></dt><dd><p>Sore throat pain score at rest on the first postoperative morning (11 point Likert scale, 0=no pain, 10 = worst pain), numeric, range: 0-6), numeric, range: 0- 6</p>
</dd>
</dl>



<h3>Details</h3>

<p>The Licorice Gargle dataset was contributed by Dr. Amy Nowacki, Associate Professor, Cleveland Clinic. Please refer to this resource as: Amy S. Nowacki, 'Licorice Gargle Dataset', TSHS Resources Portal (2017). Available at https://www.causeweb.org/tshs/licorice-gargle/.<br />
Postoperative sore throat is a common and annoying complication of endotracheal intubation. Intubation with double-lumen tubes, which are much larger than conventional single-lumen tubes, are especially likely to provoke sore throats, with a reported incidence up to 90%. Presumably, postoperative sore throats are a consequence of local tissue trauma, due to laryngoscopy and/or endotracheal intubation, leading to inflammation of pharyngeal mucosa.<br />
Nonpharmacological methods for preventing an intubation-related sore throat include using smaller-sized endotracheal tubes, lubricating the endotracheal tube with water-soluble jelly, and careful airway instrumentation as examples. Pharmacological measures for attenuating postoperative sore throats include inhalation of beclomethasone or fluticasone propionate; gargling with azulene sulfonate, aspirin, or ketamine; and gargling or spraying benzydamine hydrochloride on the endotracheal cuff for example. Each of these approaches and others not listed, however, has limitations and variable success rates; thus none has become established or is in routine clinical use.<br />
Recently, a study reported that gargling with licorice halves the risk of sore throat after intubation with conventional endotracheal tubes, based on a study of just 40 patients. A number of active ingredients have been isolated from licorice, including glycyrrhizin, liquilitin, liquiritigenin, and glabridin. The glycyrrhizin component reportedly has anti-inflammatory and antiallergic properties. Liquilitin and liquiritigenin have peripheral and central antitussive properties. Glabridin has significant antioxidant and ulcer-healing properties, which might help heal pharyngeal and tracheal mucosa after minor injuries that often complicate laryngoscopy, intubation, and endotracheal tube cuff inflation.<br />
This study tested the hypothesis that gargling with licorice solution immediately before induction of anesthesia prevents sore throat and postextubation coughing in patients intubated with double-lumen tubes.<br />
</p>


<h3>Source</h3>

<p>These are data from a study by Ruetzler et al. 'A Randomized, Double-Blind Comparison of Licorice Versus Sugar-Water Gargle for Prevention of Postoperative Sore Throat and Postextubation Coughing'. Anesth Analg 2013; 117: 614 – 21.
</p>

<hr>
<h2 id='opt'>Obstetrics and Periodontal Therapy Dataset <br /></h2><span id='topic+opt'></span>

<h3>Description</h3>

<p>The objective of this randomized controlled trial was to determine whether treatment of maternal periodontal disease can reduce risk of preterm birth and low birth weight (<strong>more details</strong> available below the variable definitions).<br />
<br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>opt
</code></pre>


<h3>Format</h3>

<p>A data frame with 823 observations and 171 variables
</p>

<dl>
<dt><code>PID</code></dt><dd><p>Participant ID, First digit indicates enrollment center (1 = NY, 2 = MN, 3 = KY, 4 = MS); Next 4 digits are sequential; Sixth digit is a check digit; There are no missing data, numeric, range: 100034-402477</p>
</dd>
<dt><code>Clinic</code></dt><dd><p>Enrollment Center, factor, NY = Harlem Hospital, MN = Hennepin County Center; KY = University of Kentucky; MS = University of Mississippi Medical Center; There are no missing data</p>
</dd>
<dt><code>Group</code></dt><dd><p>Randomized treatment assignment, factor, T = Intervention; C = Control; There are no missing data</p>
</dd>
<dt><code>Age</code></dt><dd><p>Age of participant at baseline (years), numeric, range: 16-44</p>
</dd>
<dt><code>Black</code></dt><dd><p>Black participant (self-identified), factor; Yes, No</p>
</dd>
<dt><code>White</code></dt><dd><p>White participant (self-identified), factor; Yes, No</p>
</dd>
<dt><code>Nat.Am</code></dt><dd><p>Native American participant, incl. Latin Americans with aboriginal origin(self-identified), factor; Yes, No</p>
</dd>
<dt><code>Asian</code></dt><dd><p>Asian participant (self-identified), factor; Yes, No</p>
</dd>
<dt><code>Hisp</code></dt><dd><p>Hispanic participant (self-identified), factor; Yes, No</p>
</dd>
<dt><code>Education</code></dt><dd><p>Education level of participant, factor;  LT 8 yrs = Less than 8 years; 8-12 yrs = 8 to 12 years; MT 12 yrs = More than 12 yrs; blank = Missing</p>
</dd>
<dt><code>Public.Asstce</code></dt><dd><p>Public Assistance: Whether a government agency paid for the delivery, factor;  Yes, No;</p>
</dd>
<dt><code>Hypertension</code></dt><dd><p>Whether participant had chronic hypertension at baseline, factor; Yes, No</p>
</dd>
<dt><code>Diabetes</code></dt><dd><p>Whether participant had diabetes at baseline (self-reported), factor; Yes, No</p>
</dd>
<dt><code>BL.Diab.Type</code></dt><dd><p>Baseline Diabetes Type: Type of diabetes, for participants having diabetes at baseline (self-reported), factor; Type I; Type II; Blank = No diabetes at baseline (variable 13 = No)</p>
</dd>
<dt><code>BMI</code></dt><dd><p>NA, numeric, range: 15.000-68.0</p>
</dd>
<dt><code>Use.Tob</code></dt><dd><p>Self-reported participant history of tobacco use, factor; Yes, No; Blank = Missing</p>
</dd>
<dt><code>BL.Cig.Day</code></dt><dd><p>Self-reported number of cigarettes per day for those with tobacco use history, numeric, range: 1-30; Blank = Missing (variable 16= Yes or blank) or non-smoker (variable 16 = No)</p>
</dd>
<dt><code>Use.Alc</code></dt><dd><p>Self-reported participant history of alcohol use, factor; Yes, No; Blank = Missing</p>
</dd>
<dt><code>BL.Drks.Day</code></dt><dd><p>, Blank = Missing (variable 18 = Yes or blank) or non-drinker (variable 18 = No)</p>
</dd>
<dt><code>Drug.Add</code></dt><dd><p>Self-reported participant history of drug addiction, factor; Yes, No; Blank = Missing</p>
</dd>
<dt><code>Prev.preg</code></dt><dd><p>Any previous pregnancy, factor; Yes, No; No missing data</p>
</dd>
<dt><code>N.prev.preg</code></dt><dd><p>Number of previous pregnancies for those with any previous pregnancy, numeric, range:  1-11; Blank = Missing (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>Live.PTB</code></dt><dd><p>Previous live preterm birth for those with any previous pregnancy, factor; Yes; No = No previous live preterm birth (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>Any.stillbirth</code></dt><dd><p>Previous stillbirth, factor; Yes; No = No previous stillbirth (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>Spont.ab</code></dt><dd><p>Previous spontaneous abortion, factor; Yes; No; Blank = Missing (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>Induced.ab</code></dt><dd><p>Previous induced abortion, factor; Yes; No; Blank = Missing (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>Any.live.ptb.sb.sp.ab.in.ab</code></dt><dd><p>Any previous live pre-term birth, stillbirth, spontaneous abortion, or induced abortion, factor; Yes; No = No live pre-term birth/stillbirth/abortion (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>N.living.kids</code></dt><dd><p>Number of living children the subject had at baseline, numeric, range: 0-9; Blank = Missing (variable 21 = Yes) or no previous pregnancies (variable 21 = No)</p>
</dd>
<dt><code>Tx.comp.</code></dt><dd><p>Whether treatment plans were completed by participants in treatment group, factor, Yes = Completed; No = Not completed; Und = Some therapy (unknown whether completed); Blank = Withdrew from treatment (variable 3 = T) or no periodontal therapy (variable 3 = C)</p>
</dd>
<dt><code>Local.anes</code></dt><dd><p>Whether any local anesthetic used during periodontal therapy for participants in treatment group, factor, Yes; No = No local anesthetic used or withdrew from treatment (variable 3 = T); Blank = No periodontal therapy (variable 3 = C)</p>
</dd>
<dt><code>Topical.Anest</code></dt><dd><p>Whether any topical anesthetic used during periodontal therapy for participants in treatment group, factor, Yes; No = No topical anesthetic used or withdrew from treatment (variable 3 = T); Blank = No periodontal therapy (variable 3 = C)</p>
</dd>
<dt><code>Tx.time</code></dt><dd><p>Total treatment time for participants in treatment group (hours), numeric, range: 0.117-5.8; Blank = Withdrew from treatment (variable 3 = T and variable 29 = blank) or no periodontal therapy (variable 3 = C)</p>
</dd>
<dt><code>EDC.necessary.</code></dt><dd><p>Whether patient required1 essential dental care (EDC), factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Completed.EDC</code></dt><dd><p>Did patient complete EDC before 20 weeks gestational age?, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>N.extractions</code></dt><dd><p>Number of teeth extracted during EDC, numeric, range: 0-20; Blank = Missing</p>
</dd>
<dt><code>N.perm.restorations</code></dt><dd><p>Number of permanent restorations carried out as a part of EDC, numeric, range: 0-18; Blank = Missing</p>
</dd>
<dt><code>N.qualifying.teeth</code></dt><dd><p>Number of teeth meeting OPT (Obstetrics and Periodontal Therapy Study) criteria for having periodontal disease at baseline, numeric, range: 3.000-28.0</p>
</dd>
<dt><code>BL.GE</code></dt><dd><p>Whole-mouth average gingival index at baseline, numeric, range: 0.429-3.0, Silness-Lowe Gingival Index: Higher value indicates more severe inflammation; 0 = Normal gingiva; There are no missing data</p>
</dd>
<dt><code>BL..BOP</code></dt><dd><p>Percentage of sites bleeding on probing at baseline, numeric, range:33.951-100.0</p>
</dd>
<dt><code>BL.PD.avg</code></dt><dd><p>Whole-mouth average pocket depth at baseline (mm), numeric, range: 1.851-7.0</p>
</dd>
<dt><code>BL..PD.4</code></dt><dd><p>Percentage of sites with pocket depth greater than or equal to 4mm at baseline, numeric, range: 3.571-99.2</p>
</dd>
<dt><code>BL..PD.5</code></dt><dd><p>Percentage of sites with pocket depth greater than or equal to 5mm at baseline, numeric, range:  0-91.7</p>
</dd>
<dt><code>BL.CAL.avg</code></dt><dd><p>Whole-mouth average clinical attachment level at baseline (mm), numeric, range: 0.185-5.1</p>
</dd>
<dt><code>BL..CAL.2</code></dt><dd><p>Percentage of sites with clinical attachment level greater than or equal to 2 mm at baseline, numeric, range: 2.381-100.0</p>
</dd>
<dt><code>BL..CAL.3</code></dt><dd><p>Percentage of site with clinical attachment level greater than or equal to 3 mm at baseline, numeric, range: 0-94.9</p>
</dd>
<dt><code>BL.Calc.I</code></dt><dd><p>Whole-mouth average calculus index at baseline, Simplified Oral Hygiene Index (OHI-S): Higher value indicates more calculus; 0 = No calculus present; numeric, range: 0-3.0</p>
</dd>
<dt><code>BL.Pl.I</code></dt><dd><p>Whole-mouth average plaque index at baseline, Silness-Lowe Gingival Index:Higher value indicates more severe inflammation, 0= normal gingiva, numeric, range: 0.056-3.0</p>
</dd>
<dt><code>V3.GE</code></dt><dd><p>Whole-mouth average gingival index at Visit 3, numeric, range:      0.030-3.0</p>
</dd>
<dt><code>V3..BOP</code></dt><dd><p>Percentage of sites bleeding on probing at Visit 3, numeric, range: 0.725-100.0, Blank = Missing</p>
</dd>
<dt><code>V3.PD.avg</code></dt><dd><p>Whole-mouth average pocket depth at Visit 3 (mm), numeric, range: 1.601-5.5, Blank = Missing</p>
</dd>
<dt><code>V3..PD.4</code></dt><dd><p>Percentage of sites with pocket depth greater than or equal to 4mm at Visit 3, numeric, range: 0-83.9, Blank = Missing</p>
</dd>
<dt><code>V3..PD.5</code></dt><dd><p>Percentage of sites with pocket depth greater than or equal to 5mm at Visit 3, numeric, range: 0-77.4, Blank = Missing</p>
</dd>
<dt><code>V3.CAL.avg</code></dt><dd><p>Whole-mouth average clinical attachment level at Visit 3 (mm), numeric, range: 0.036-3.9, Blank = Missing</p>
</dd>
<dt><code>V3..CAL.2</code></dt><dd><p>Percentage of sites with clinical attachment level greater than or equal to 2 mm at visit 3, numeric, range: 0-97.8, Blank = Missing</p>
</dd>
<dt><code>V3..CAL.3</code></dt><dd><p>Percentage of sites with clinical attachment level greater than or equal to 3 mm at visit 3, numeric, range: 0-85.7, Blank = Missing</p>
</dd>
<dt><code>V3.Calc.I</code></dt><dd><p>Whole-mouth average calculus index at visit 3, numeric, range:  0-2.6, Simplified Oral Hygiene Index (OHI-S): Higher value indicates more calculus; 0 = No calculus present; Blank = Missing</p>
</dd>
<dt><code>V3.Pl.I</code></dt><dd><p>Whole-mouth average plaque index at visit 3, numeric, range: 0-2.6, Silness-Lowe Plaque Index: Higher value indicates more abundant plaque; 0 = No plaque in gingival area; Blank = Missing</p>
</dd>
<dt><code>V5.GE</code></dt><dd><p>Whole-mouth average gingival index at visit 5, numeric, range: 0.190-2.7, Silness-Lowe Gingival Index: Higher value indicates more severe inflammation; 0 = Normal gingiva; Blank = Missing</p>
</dd>
<dt><code>V5..BOP</code></dt><dd><p>Percentage of sites bleeding on probing at visit 5, numeric, range: 3.571-100.0, Blank = Missing</p>
</dd>
<dt><code>V5.PD.avg</code></dt><dd><p>Whole-mouth average pocket depth at visit 5, numeric, range: 1.536-5.4, Blank = Missing</p>
</dd>
<dt><code>V5..PD.4</code></dt><dd><p>Percentage of sites with pocket depth greater than or equal to 4mm at Visit 5, numeric, range:  0-83, Blank = Missing</p>
</dd>
<dt><code>V5..PD.5</code></dt><dd><p>Percentage of sites with pocket depth greater than or equal to 5mm at Visit 3, numeric, range: 0-75.6, Blank = Missing</p>
</dd>
<dt><code>V5.CAL.avg</code></dt><dd><p>Whole-mouth average clinical attachment level at visit 5 (mm), numeric, range: 0.018-4.3, Blank = Missing</p>
</dd>
<dt><code>V5..CAL.2</code></dt><dd><p>Percentage of sites with clinical attachment level greater than or equal to 2 mm at visit 5, numeric, range:  0.000-99.2, Blank = Missing</p>
</dd>
<dt><code>V5..CAL.3</code></dt><dd><p>Percentage of sites with clinical attachment level greater than or equal to 3 mm at visit 5, numeric, range: 0.000-85.0, Blank = Missing</p>
</dd>
<dt><code>V5.Calc.I</code></dt><dd><p>Whole-mouth average calculus index at visit 5, numeric, range: 0.0-2.6, Simplified Oral Hygiene Index (OHI-S): Higher value indicates more calculus; 0 = No calculus present; Blank = Missing</p>
</dd>
<dt><code>V5.Pl.I</code></dt><dd><p>Whole-mouth average plaque index at visit 5, numeric, range: 0.0-2.5, Silness-Lowe Plaque Index: Higher value indicates more abundant plaque; 0 = No plaque in gingival area; Blank = Missing</p>
</dd>
<dt><code>N.PAL.sites</code></dt><dd><p>Number of sites for which attachment loss increased from baseline by greater than or equal to 3 mm, numeric, range: 0-33, 0 = No sites; Blank = Missing</p>
</dd>
<dt><code>Birth.outcome</code></dt><dd><p>Birth outcome, factor, Elective abortion; Live birth; Lost to FU = Lost to Follow-Up; Non-live birth = Stillbirth or spontaneous abortion; There are no missing data</p>
</dd>
<dt><code>Preg.ended...37.wk</code></dt><dd><p>Whether the pregnancy ended before gestational age 37 weeks (259 days), factor, Yes; No; Blank = Lost to Follow-Up</p>
</dd>
<dt><code>GA.at.outcome</code></dt><dd><p>Gestational age at end of pregnancy, or at mother's last follow-up visit if lost to follow-up, numeric, range:  103-302</p>
</dd>
<dt><code>Birthweight</code></dt><dd><p>Infant birth weight at time of birth, abstracted from obstetrical records (grams), numeric, range:  101-5160, Blank = Missing</p>
</dd>
<dt><code>Fetal.congenital.anomaly</code></dt><dd><p>Fetal/congenital anomaly identified at birth or during pregnancy?, factor, Yes; No; There are no missing data</p>
</dd>
<dt><code>Apgar1</code></dt><dd><p>Apgar score, a summary of a newborn infant's 'Appearance, Pulse, Grimace, Activity, Respiration' at 1 minute Score interpretation: less than or equal to 3: Critically low 4-6: Fairly low greater than or equal to 7: Normal, numeric, range: 0-10, Blank = Missing</p>
</dd>
<dt><code>Apgar5</code></dt><dd><p>Apgar score at 5 minutes, numeric, range: 0-10, Blank = Missing</p>
</dd>
<dt><code>Any.SAE.</code></dt><dd><p>Whether participant experienced any serious adverse events (e.g. lost pregnancies) factor, Yes; No; There are no missing data</p>
</dd>
<dt><code>GA...1st.SAE</code></dt><dd><p>Gestational age of first SAE (serious adverse event), integer, range: 96-467, 259 = No SAE (variable 76 must = No); There are no missing data</p>
</dd>
<dt><code>Bact.vag</code></dt><dd><p>Whether mother had bacterial vaginosis during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Gest.diab</code></dt><dd><p>Whether mother had gestational diabetes during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Oligo</code></dt><dd><p>Whether mother had oligohydramnios during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Polyhyd</code></dt><dd><p>Whether mother had polyhydramnios during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Gonorrhea</code></dt><dd><p>Whether mother had gonorrhea during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Chlamydia</code></dt><dd><p>Whether mother had chlamydia during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Strep.B</code></dt><dd><p>Whether mother had strep B colonization during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Traumatic.Inj</code></dt><dd><p>Whether mother had a traumatic injury during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>UTI</code></dt><dd><p>Whether mother had a urinary tract infection during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Pre.eclamp</code></dt><dd><p>Whether mother had pre-eclampsia, a pregnancy condition characterized by high blood pressure and associated with fetal growth restriction during pregnancy, factor, Yes; No; Blank = Missing</p>
</dd>
<dt><code>Mom.HIV.status</code></dt><dd><p>HIV status of mother during pregnancy, factor, Yes = HIV-positive; No = HIV-negative or unknown (question answered but HIV status at delivery not recorded); Blank = Missing (question not answered)</p>
</dd>
<dt><code>BL.Anti.inf</code></dt><dd><p>Did participant report use of antiinflammatory medication at or less than 6 months before baseline?, integer,  0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>BL.Cortico</code></dt><dd><p>Did participant report use of corticosteroids at or less than 6 months before baseline?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>BL.Antibio</code></dt><dd><p>Did participant report use of antibiotics at or less than 6 months before baseline?, integer,  0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>BL.Bac.vag</code></dt><dd><p>Did participant report use of bacterial vaginitis treatments at or less than 6 months before baseline?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V3.Anti.inf</code></dt><dd><p>Did participant report use of antiinflammatory medication between baseline and visit 3?,integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V3.Cortico</code></dt><dd><p>Did participant report use of corticosteroids between baseline and visit 3?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V3.Antibio</code></dt><dd><p>Did participant report use of antibiotics between baseline and visit 3?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V3.Bac.vag</code></dt><dd><p>Did participant report use of bacterial vaginitis treatments between baseline and visit 3?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V5.Anti.inf</code></dt><dd><p>Did participant report use of antiinflammatory medication between visit 3 and visit 5?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V5.Cortico</code></dt><dd><p>Did participant report use of corticosteroids between visit 3 and visit 5?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V5.Antibio</code></dt><dd><p>Did participant report use of antibiotics between visit 3 and visit 5?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>V5.Bac.vag</code></dt><dd><p>Did participant report use of bacterial vaginitis treatments between visit 3 and visit 5?, integer, 0 = No; 1 = Yes; There are no missing data</p>
</dd>
<dt><code>X..Vis.Att</code></dt><dd><p>Visit attendance: Number of study visits attended AFTER baseline, integer, Range: 0-5</p>
</dd>
<dt><code>X..Vis.Elig</code></dt><dd><p>Number of visits for which participant was eligible (could become ineligible after miscarriage or early delivery), integer, Range: 0-5</p>
</dd>
<dt><code>X1st.Miss.Vis</code></dt><dd><p>First missed visit. No one missed the baseline visit, so this variable takes values 2, 3, 4, 5, 6, and 100 (no eligible visits missed), integer, Range: 2-6, 100</p>
</dd>
<dt><code>OAA1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to A. actinomycetemcomitans at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OCR1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to C. rectus at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OFN1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to F. nucleatum at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OPG1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to P. gingivalis at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OPI1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to P. intermedia at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OTD1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to T. denticola at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OTF1</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to T. forsythus at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OCRP1</code></dt><dd><p>Serum measure for C-reactive protein (CRP) at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>O1B1</code></dt><dd><p>Serum measure for Interleukin(IL)-1b at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>O61</code></dt><dd><p>Serum measure for Interleukin(IL)-6 at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>O81</code></dt><dd><p>Serum measure forInterleukin(IL)-8 at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OPGE21</code></dt><dd><p>Serum measure for Prostaglandin E2 at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OTNF1</code></dt><dd><p>Serum measure for tumor necrosis factor (TNF)-alpha at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OMMP91</code></dt><dd><p>Serum measure for gelatinase (MMP9) at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>ETXU_CAT1</code></dt><dd><p>Serum endotoxin level at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OFIBRIN1</code></dt><dd><p>Serum measure for fibrinogen at baseline, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OAA5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to A. actinomycetemcomitans at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OCR5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to C. rectus at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OFN5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to F. nucleatum at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OPG5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to P. gingivalis at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OPI5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to P. intermedia at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OTD5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to T. denticola at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OTF5</code></dt><dd><p>Serum IgG (immunoglobulin) antibodies to T. forsythus at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OCRP5</code></dt><dd><p>Serum measure for C-reactive protein (CRP) at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>O1B5</code></dt><dd><p>Serum measure for Interleukin(IL)-1b at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>O65</code></dt><dd><p>Serum measure forInterleukin(IL)-6 at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>O85</code></dt><dd><p>Serum measure forInterleukin(IL)-8 at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OPGE25</code></dt><dd><p>Serum measure for Prostaglandin E2 at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OTNF5</code></dt><dd><p>Serum measure for tumor necrosis factor (TNF)-alpha at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OMMP95</code></dt><dd><p>Serum measure for gelatinase (MMP9) at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>ETXU_CAT5</code></dt><dd><p>Serum endotoxin level at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>OFIBRIN5</code></dt><dd><p>Serum measure for fibrinogen at visit 5, factor (actually numeric or missing), dot(.) = Missing</p>
</dd>
<dt><code>BL.DNA</code></dt><dd><p>Total amount of bacterial DNA extracted from plaque as a measure of total bacterial concentration at baseline (ng/mL), numeric, range: 0-5750.0</p>
</dd>
<dt><code>BL.Univ</code></dt><dd><p>Count of all bacteria detected by universal primer at baseline, numeric, range: 1,890,000-1,070,000,000, Blank = Missing</p>
</dd>
<dt><code>BL.AA</code></dt><dd><p>Count of A. actinomycetemcomitans bacteria at baseline, numeric, range:  0-7,970,000, Blank = Missing</p>
</dd>
<dt><code>BL.PG</code></dt><dd><p>Count of P. gingivalis bacteria at baseline, numeric, range:  0-167,000,000, Blank = Missing</p>
</dd>
<dt><code>BL.TD</code></dt><dd><p>Count of T. denticola bacteria at baseline, numeric, range:  0-50,500,000, Blank = Missing</p>
</dd>
<dt><code>BL.TF</code></dt><dd><p>Count of T. forsythus bacteria at baseline, numeric, range:  0-40,200,000, Blank = Missing</p>
</dd>
<dt><code>BL.PI</code></dt><dd><p>Count of P. intermedia bacteria at baseline, numeric, range:  0-87,500,000, Blank = Missing</p>
</dd>
<dt><code>BL.CR</code></dt><dd><p>Count of C. rectus bacteria at baseline, numeric, range:  0-32,600,000, Blank = Missing</p>
</dd>
<dt><code>BL.FN</code></dt><dd><p>Count of F. nucleatum bacteria at baseline, numeric, range:  67,300- 152,000,000, Blank = Missing</p>
</dd>
<dt><code>BL.S7</code></dt><dd><p>Sum of the 7 species-specific bacterial counts (variables 138-144) at baseline, rounded to 3 significant figures, numeric, range:  87,000-391,000,000, Blank = Missing</p>
</dd>
<dt><code>V5.DNA</code></dt><dd><p>Total amount of bacterial DNA extracted from plaque as a measure of total bacterial concentration at visit 5 (ng/mL), numeric, range: 0-5750.0</p>
</dd>
<dt><code>V5.Univ</code></dt><dd><p>Count of all bacteria detected by universal primer at visit 5, numeric, range: 1,890,000-1,070,000,000, Blank = Missing</p>
</dd>
<dt><code>V5.AA</code></dt><dd><p>Count of A. actinomycetemcomitans bacteria at visit 5, numeric, range:  0-40,200,000, Blank = Missing</p>
</dd>
<dt><code>V5.PG</code></dt><dd><p>Count of P. gingivalis bacteria at visit 5, numeric, range:  0-40,200,000, Blank = Missing</p>
</dd>
<dt><code>V5.TD</code></dt><dd><p>Count of T. forsythus bacteria at visit 5, numeric, range:  0-40,200,000, Blank = Missing</p>
</dd>
<dt><code>V5.TF</code></dt><dd><p>Count of T. forsythus bacteria at visit 5, numeric, range:  0-40,200,000, Blank = Missing</p>
</dd>
<dt><code>V5.PI</code></dt><dd><p>Count of P. intermedia bacteria at visit 5, numeric, range:  0-87,500,000, Blank = Missing</p>
</dd>
<dt><code>V5.CR</code></dt><dd><p>Count of C. rectus bacteria at visit 5, numeric, range:  0-32,600,000, Blank = Missing</p>
</dd>
<dt><code>V5.FN</code></dt><dd><p>Count of F. nucleatum bacteria at visit 5, numeric, range:  67,300- 152,000,000, Blank = Missing</p>
</dd>
<dt><code>V5.S7</code></dt><dd><p>Sum of the 7 species-specific bacterial counts (variables 138-144) at visit 5, rounded to 3 significant figures, numeric, range:  87,000-391,000,000, Blank = Missing</p>
</dd>
<dt><code>BL..AA</code></dt><dd><p>Percent of A. actinomycetemcomitans out of total DNA (variable 146) at baseline, numeric, range:  0-8.9, Blank = Missing</p>
</dd>
<dt><code>BL..PG</code></dt><dd><p>Percent of P. gingivalis out of total DNA at baseline, numeric, range:  0-37.3, Blank = Missing</p>
</dd>
<dt><code>BL..TD</code></dt><dd><p>Percent of T. denticola out of total DNA at baseline, numeric, range:  0-13.2, Blank = Missing</p>
</dd>
<dt><code>BL..TF</code></dt><dd><p>Percent of T. forsythus out of total DNA at baseline, numeric, range:  0-17.7, Blank = Missing</p>
</dd>
<dt><code>BL..PI</code></dt><dd><p>Percent of P. intermedia out of total DNA at baseline, numeric, range:  0-46.3, Blank = Missing</p>
</dd>
<dt><code>BL..CR</code></dt><dd><p>Percent of C. rectus out of total DNA at baseline, numeric, range:  0-10.5, Blank = Missing</p>
</dd>
<dt><code>BL..FN</code></dt><dd><p>Percent of F. nucleatum out of total DNA at baseline, numeric, range: 0.330-63.2, Blank = Missing</p>
</dd>
<dt><code>BL..S7</code></dt><dd><p>Sum of the percents for the 7 species (AA, PG, TD, TF, PI, CR, and FN) at baseline, numeric, range: 0.420-86.3, Blank = Missing</p>
</dd>
<dt><code>V5..AA</code></dt><dd><p>Percent of A. actinomycetemcomitans out of total DNA at visit 5, numeric, range: 0-16.1, Blank = Missing</p>
</dd>
<dt><code>V5..PG</code></dt><dd><p>Percent of P. gingivalis out of total DNA at visit 5, numeric, range: 0-59.7, Blank = Missing</p>
</dd>
<dt><code>V5..TD</code></dt><dd><p>Percent of T. denticola out of total DNA at visit 5, numeric, range: 0-20.5, Blank = Missing</p>
</dd>
<dt><code>V5..TF</code></dt><dd><p>Percent of T. forsythus out of total DNA at visit 5, numeric, range: 0-19.3, Blank = Missing</p>
</dd>
<dt><code>V5..PI</code></dt><dd><p>Percent of P. intermedia out of total DNA at visit 5, numeric, range: 0-40.7, Blank = Missing</p>
</dd>
<dt><code>V5..CR</code></dt><dd><p>Percent of C. rectus out of total DNA at visit 5, numeric, range: 0-14.6, Blank = Missing</p>
</dd>
<dt><code>V5..FN</code></dt><dd><p>Percent of F. nucleatum out of total DNA at visit 5, numeric, range: 0-49.9, Blank = Missing</p>
</dd>
<dt><code>V5..S7</code></dt><dd><p>Sum of the percents for the 7 species (AA, PG, TD, TF, PI, CR, and FN) at visit 5, numeric, range: 2.560-80.8, Blank = Missing</p>
</dd></dl>



<h3>Details</h3>



<h4>Background:</h4>

<p>Randomized Clinical Trial on the Effect of Treatment of Maternal Periodontal Disease Can Reduce Preterm Birth Risk.<br />
Maternal periodontal disease has been linked in observational studies to preterm birth (&lt; 37 weeks) and low birth weight (&lt; 2500 g) outcomes. The Obstetrics and Periodontal Therapy study was a multi-center randomized trial evaluating the effect of nonsurgical periodontal treatment intervention on preterm birth, comparing outcomes of women treated before 21 weeks gestation (treatment) to those treated after delivery (control). <br />
Preterm birth, defined as delivery before 37 weeks of gestation, is a growing problem. In some cases, preterm birth can lead to infant death; in others, its consequences may include neurodevelopmental disabilities, cognitive impairment, and/or respiratory disorders in the child. Many risk factors for preterm birth have already been identified, including maternal age, drug use, and diabetes. However, such factors are exhibited in only about half of preterm birth mothers, highlighting a need to expand our understanding of what contributes to preterm birth risk.<br />
Several observational studies have suggested an association between maternal periodontal disease and preterm birth. Periodontal disease is an inflammatory condition characterized by the destruction of tissue and/or bone around the teeth. A major component of periodontal disease is oral colonization by gram-negative bacteria; systemic release of cytokines and/or lipopolysaccharides from these bacteria may impact fetal condition.<br />
Inoculation of the periodontal pathogen P. gingivalis into pregnant animals does have a dose-dependent effect on birth weight and preterm birth signaling, but no such causal link has been shown in humans, only some associations. Though not definitive, the possibility of a significant relationship raises the question of whether treatment of maternal periodontal disease can decrease preterm birth risk.<br />
</p>



<h4>Participants:</h4>

<p>823 participants enrolled at 4 centers underwent stratified randomization, resulting in 413 women assigned to the treatment group and 410 to control. All participants were 13-16 weeks pregnant at time of randomization (baseline/visit 1) and went on to attend monthly follow-up visits defined as visits 2, 3, 4, and 5 corresponding to gestational age ranges of 17-20, 21-24, 25-28, and 29-32 weeks.<br />
</p>



<h4>Treatment:</h4>

<p>The treatment group received periodontal treatment, oral hygiene instruction, and tooth polishing at their follow-ups, while those assigned to control underwent only brief oral exams. Data collection occurred at visits 1 (baseline), 3, and 5. The primary outcome of interest is gestational age at end of pregnancy. Additional outcomes include birthweight, clinical measures of periodontal disease, and various microbiological and immunological outcomes.<br />
</p>



<h4>Analysis:</h4>

<p>Statistical analyses were carried out on an intent-to-treat basis. Gestational age can be thought of as 'time until end of pregnancy,' for which certain survival analysis methods would be appropriate. The study used a log-rank test stratified by center to compare time until end of pregnancy for treatment and control groups.<br />
A semiparametric proportional hazards model was also used for this purpose and incorporated maternal risk factors as predictors. For the study's main analyses, gestational age was censored at 37 weeks (259 days) because the interest was in extending pregnancies that would otherwise end pre-term, not extending pregnancies generally.<br />
Though not used in the study itself, logistic regression is another method that could be applied: for example, to gestational age, dichotomized as 'preterm' or 'not preterm' according to a gestational age cutoff, or to birthweight dichotomized as 'low' or 'high' at the 2500 g or other cutoff (2500 g would be in keeping with the World Health Organization's definition for low birth weight). Changes in clinical measures of periodontal disease from baseline to visits 3 or 5 could be analyzed using mixed effects linear models. The dataset also features a number of baseline characteristics, which could be compared in treatment and control groups via Student t-tests, Wilcoxon rank sum tests, Fisher's exact tests or Pearson's chi-square tests, as appropriate.<br />
</p>



<h4>Publishing:</h4>

<p>The nonsurgical periodontal treatment involving scaling and root planing induced significant improvements in periodontal health. The study did not however find a significant relation between periodontal treatment and preterm birth risk. The results of this study were published in 2006 by Michalowicz et al., 'Treatment of periodontal disease and the risk of preterm birth', in The New England Journal of Medicine. The Obstetrics and Periodontal Therapy Dataset contains the data used in this study.<br />
The obstetrics and periodontal therapy dataset was contributed by Dr. Ann Brearley, Assistant Professor, Division of Biostatistics, School of Public Health, University of Minnesota and her colleagues. Please refer to this resource as: Meredith Hyun, James S. Hodges and Ann M. Brearley, 'Obstetrics and Periodontal Therapy Dataset', TSHS Resources Portal (2019). Available at https://www.causeweb.org/tshs/obstetrics-and-periodontal-therapy/.<br />
<br />
</p>



<h3>Source</h3>

<p>Michalowicz et al., 'Treatment of periodontal disease and the risk of preterm birth', N Engl J Med 2006; 355:1885-1894. DOI: 10.1056/NEJMoa062249
</p>

<hr>
<h2 id='polyps'>RCT of Sulindac for Polyp Prevention in Familial Adenomatous Polyposis</h2><span id='topic+polyps'></span>

<h3>Description</h3>

<p>Results of a randomized, placebo-controlled trial of sulindac in the reduction of colonic polyps in Familial Adenomatous Polyposis  (FAP) (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>polyps
</code></pre>


<h3>Format</h3>

<p>A data frame with 22 observations and 7 variables
</p>

<dl>
<dt>participant_id</dt><dd><p>id number for each participant; type: character</p>
</dd>
<dt>sex</dt><dd><p>participant sex, levels: female, male; type: factor</p>
</dd>
<dt>age</dt><dd><p>age in years; type: numeric</p>
</dd>
<dt>baseline</dt><dd><p>number of colonic polyps at baseline; type: numeric</p>
</dd>
<dt>treatment</dt><dd><p>treatment assignment, levels: sulindac, placebo; type: factor</p>
</dd>
<dt>number3m</dt><dd><p>number of colonic polyps at 3 months; type: numeric</p>
</dd>
<dt>number12m</dt><dd><p>number of colonic polyps at 12 months; type: numeric</p>
</dd>
</dl>



<h3>Details</h3>

<p>FAP is an inherited condition caused by mutations in the APC  (Adenomatous Polyposis Coli) gene that leads to early and frequent formation of precancerous polyps of the colon at a young age, and invariably leads to the development of colon cancer at a young age.  <br />
Early, frequent surveillance colonoscopy and polyp removal is helpful, but this study examined whether there is a beneficial effect of preventive medical therapy with the nonsteroidal pain reliever, sulindac, versus placebo in a RCT vs placebo in 22 participants, with polyp number measured (via colonoscopy) at baseline, 3 months, and 12 months after starting the study drug. Note that one subject did not return for the 12 month colonoscopy.
</p>


<h3>Source</h3>

<p>This data set is from a study published in 1993 in the New England Journal of Medicine,
</p>
<p>F. M. Giardiello, S. R. Hamilton, A. J. Krush, S. Piantadosi, L. M. Hylind, P. Celano, S. V. Booker, C. R. Robinson and G. J. A. Offerhaus (1993), Treatment of colonic and rectal adenomas with sulindac in familial adenomatous polyposis. New England Journal of Medicine, 328(18), 1313-1316.
</p>
<p>This dataset is derived from and improved upon from the HSAUR package.
</p>

<hr>
<h2 id='scurvy'>Randomized Trial of Six Therapies for Scurvy</h2><span id='topic+scurvy'></span>

<h3>Description</h3>

<p>Results of a randomized, 6-arm comparator-controlled trial of 6 interventions to treat scurvy in 12 disabled seamen, as reported by James Lind in 1757 (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scurvy
</code></pre>


<h3>Format</h3>

<p>A data frame with 12 observations and 8 variables
</p>

<dl>
<dt>study_id</dt><dd><p>invented id number for each participant; type: character</p>
</dd>
<dt>treatment</dt><dd><p>assigned treatment, levels: cider, dilute_sulfuric_acid, vinegar, sea_water, citrus, purgative_mixture; type: factor</p>
</dd>
<dt>dosing_regimen_for_scurvy</dt><dd><p>details on daily dosing and schedule; type: character</p>
</dd>
<dt>gum_rot_d6</dt><dd><p>rating of symptom of rotting of gums; type: factor, with levels: 0=none, 1=mild, 2=moderate, 3=severe</p>
</dd>
<dt>skin_sores_d6</dt><dd><p>rating of symptom of skin sores; type: factor, with levels: 0=none, 1=mild, 2=moderate, 3=severe</p>
</dd>
<dt>weakness_of_the_knees_d6</dt><dd><p>rating of symptom of weakness of the knees (ability to stand); type: factor, with levels: 0=none, 1=mild, 2=moderate, 3=severe</p>
</dd>
<dt>lassitude_d6</dt><dd><p>rating of symptom of lassitude (generalized weakness); type: factor, with levels: 0=none, 1=mild, 2=moderate, 3=severe</p>
</dd>
<dt>fit_for_duty_d6</dt><dd><p>dichotomous fitness for duty as a seaman; type: factor: 0_no, 1_yes</p>
</dd>
</dl>



<h3>Details</h3>

<p>Scurvy was a common affliction of seamen on long voyages, leading to mouth sores, skin lesions, weakness of the knees, and lassitude. Scurvy could be fatal on long voyages. James Lind reported the treatment of 12 seamen with scurvy in 1757, in <em>A Treatise on the Scurvy in Three Parts</em>. This 476 page bloviation can be found scanned to the Google Books website <a href="https://www.google.com/books/edition/A_Treatise_on_the_Scurvy/oP1UEXWU7fsC?hl=en&amp;gbpv=1&amp;printsec=frontcover">A Treatise on the Scurvy</a>.    Pages 149-153 are a rare gem among what can be generously described as 400+ pages of evidence-free blathering, and these 4 pages may represent the first report of a controlled clinical trial. <br />
Lind was the ship's surgeon on board the HMS Salisbury, and had a number of scurvy-affected seamen at his disposal. Many remedies had been described and advocated for, with no more than anecdotal evidence. On May 20, 1747, Lind decided to try the 6 available therapies at his disposal in a comparative study in 12 affected seamen. He selected 12 with roughly similar severity, with notable skin and mouth sores, weakness of the knees, and significant lassitude, making them unfit for duty. They each received the standard shipboard diet of gruel and mutton broth, supplemented with occasional biscuits and puddings. Each treatment was a dietary supplement (including citrus fruits) or a medicinal. <br />
This data frame was reconstructed from Lind's account as recorded on these 4 pages, with his estimates of severity translated to a 4 point Likert scale (0-3) for each of the symptoms he described at his chosen endpoint on day 6. A fanciful study_id variable was added, along with detailed descriptions of the dosing schedule of each treatment. <br />
Of note, there is some dispute about whether this was truly the first clinical trial, or whether it actually happened. See link about the <a href="https://www.bbvaopenmind.com/en/science/leading-figures/james-lind-and-scurvy-the-first-clinical-trial-in-history/">historical debate</a>.     Lind reported that the seamen treated with 2 lemons and an orange daily did best, followed by those treated with cider. Those treated with elixir of vitriol only had improvement in mouth sores. One imagines that acidic substances (like dilute sulfuric acid, vinegar, cider, and citrus fruits) might have been rather painful on these mouth sores. Unfortunately, the burial of 4 valuable pages of data in 476 pages of noise, a publication delay of 10 years, and Lind's half-hearted conclusions, meant that it took until 1795 before the British Navy mandated daily limes for seamen. <br />
</p>


<h3>Source</h3>

<p>This data set is faithfully reconstructed from a report published in 1757 as <em>A Treatise on the Scurvy in 3 Parts</em>, by James Lind, pp. 149-153, and you can find a scan of the source document that you can read yourself on Google Books <a href="https://www.google.com/books/edition/A_Treatise_on_the_Scurvy/oP1UEXWU7fsC?hl=en&amp;gbpv=1&amp;printsec=frontcover">here</a>.
</p>

<hr>
<h2 id='smartpill'>Prospective Cohort Study of Intestinal Transit using a SmartPill to Compare Trauma Patients to Healthy Volunteers</h2><span id='topic+smartpill'></span>

<h3>Description</h3>

<p>This study evaluated gastric emptying, small bowel transit time, and total intestinal transit time in 8 critically ill trauma patients. These data were compared with those obtained in 87 healthy volunteers from a separate trial. Data were obtained with a motility capsule that wirelessly transmitted pH, pressure, and temperature to a recorder attached to each subject's abdomen. Transit times were available for almost all patients, however, pH, pressure and temperature data is missing for all critically ill patients and sparsely missing for the healthy volunteers (<strong>more details</strong> available below the variable definitions) <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smartpill
</code></pre>


<h3>Format</h3>

<p>A data frame with 95 obsrvations and 22 variables
</p>

<dl>
<dt><code>Group</code></dt><dd><p>Study group, numeric, 0 = Critically Ill Trama Patient, 1 = Healthy Volunteer</p>
</dd>
<dt><code>Gender</code></dt><dd><p>Gender, numeric, range: 0 = Female, 1 = Male</p>
</dd>
<dt><code>Race</code></dt><dd><p>Race, numeric, 1 = White, 2 = Black, 3 = Asian/Pacific Islander, 4 = Hispanic, 5 = Other</p>
</dd>
<dt><code>Height</code></dt><dd><p>Height (centimeters), numeric, range: 132.1-193.0</p>
</dd>
<dt><code>Weight</code></dt><dd><p> Weight (kilograms), numeric, range:  44.9-127.0</p>
</dd>
<dt><code>Age</code></dt><dd><p>Age (years), numeric, range:  18.0-72.0</p>
</dd>
<dt><code>GE.Time</code></dt><dd><p>Gastric Emptying Time is time from ingestion to gastric emptying (hours), numeric, range:   1.7-74.3</p>
</dd>
<dt><code>SB.Time</code></dt><dd><p>Small Bowel Transit Time is time from gastric emptying to ileocecal junction (hours), numeric, range:   1.8-13.8</p>
</dd>
<dt><code>C.Time</code></dt><dd><p>Colonic Transit Time is time from ileocecal junction to body exit (hours), numeric, range:   0.7-118.9</p>
</dd>
<dt><code>WG.Time</code></dt><dd><p>Whole Gut Time is time from ingestion to body exit (hours), numeric, range:   6.0-816.0</p>
</dd>
<dt><code>S.Contractions</code></dt><dd><p>Stomach contractions are counted if the peak amplitude of the contraction is over 10 mmHg and under 300 mmHg, numeric, range:  47.0-1665.0</p>
</dd>
<dt><code>S.Sum.of.Amplitudes</code></dt><dd><p>Stomach sum of amplitudes (mm Hg), numeric, range: 655.6-33800.3</p>
</dd>
<dt><code>S.Mean.Peak.Amplitude</code></dt><dd><p>Stomach mean peak amplitude is the sum of amplitudes divided by number of contractions (mm Hg), numeric, range:   4.6-43.4</p>
</dd>
<dt><code>S.Mean.pH</code></dt><dd><p>Stomach mean pH is the average pH over the whole recording time in the stomach with normal ~ 1.5-3.5, numeric, range:   1.5-5.9</p>
</dd>
<dt><code>SB.Contractions</code></dt><dd><p>Small Bowel contractions are counted if the peak amplitude of the contraction is over 10 mmHg and under 300 mmHg, numeric, range: 223.0-2375.0</p>
</dd>
<dt><code>SB.Sum.of.Amplitudes</code></dt><dd><p>Small Bowel sum of amplitudes (mm Hg), numeric, range:3899.4-41122.5</p>
</dd>
<dt><code>SB.Mean.Peak.Amplitude</code></dt><dd><p>Small Bowell mean peak amplitude is the sum of amplitudes divided by number of contractions (mm Hg), numeric, range:  15.0-27.9</p>
</dd>
<dt><code>SB.Mean.pH</code></dt><dd><p>Small Bowel mean pH is the average pH over the whole recording time in the small bowel, normal ~ 6-7.4, numeric, range:   4.7-8.6</p>
</dd>
<dt><code>Colon.Contractions</code></dt><dd><p>Colon contractions are counted if the peak amplitude of the contraction is over 10 mmHg and under 300 mmHg, numeric, range:  41.0-2672.0</p>
</dd>
<dt><code>Colon.Sum.of.Amplitudes</code></dt><dd><p>Colon sum of amplitudes (mm Hg), numeric, range:1872.6-117707.5</p>
</dd>
<dt><code>C.Mean.Peak.Amplitude</code></dt><dd><p>Colon mean peak amplitude is the sum of amplitudes divided by number of contractions (mm Hg), numeric, range:  32.8-    64.2</p>
</dd>
<dt><code>C.Mean.pH</code></dt><dd><p>Colon mean pH is the average pH over the whole recording time in the colon, normal ~ 5-7-6.7, numeric, range:   3.9-8.1</p>
</dd>
</dl>



<h3>Details</h3>

<p>The Smart Pill dataset was contributed by Dr. Amy Nowacki, Associate Professor, Cleveland Clinic. Please refer to this resource as: Amy S. Nowacki, 'Smart Pill Dataset', TSHS Resources Portal (2017). Available at https://www.causeweb.org/tshs/smart-pill/. <br />
Delayed gastric emptying is a well-known problem in critically ill patients and is associated with feeding disturbances and inadequate nutrition. However, evaluating gastrointestinal function remains challenging in critically ill patients who are mechanically ventilated. Many tests that are practical and accurate under standardized, controlled conditions often fail in the critical care setting. For example, the consensus recommendations for gastric emptying scintigraphy are impractical in intubated patients because they recommend low-fat, egg white meal with imaging at 0, 1, 2, and 4 hours after meal ingestion. Another test, the lactulose hydrogen breath test, relies on prompt bacterial breakdown of lactulose in the colon; however, changes in bacterial flora - which are presumably common in critical care patients - can produce false transit times.<br />
The 13C-octanoic acid breath test was reported as successful when used bedside to measure gastric emptying. However, manometry only assesses the upper gastrointestinal function, mainly esophagus, stomach, and proximal small bowel. Finally, video capsule technology has been used to determine small bowel transit time and pathomorphology in critically ill patients, although inadequate battery lifespan of the capsule (approximately 8-10 hours) could prevent complete examination in some cases.<br />
An alternative technique, wireless capsule technology, may be useful for evaluating gastrointestinal motility in critical care patients. A newly developed motility capsule for assessing gastric emptying in patients with suspected gastroparesis has been available since 2006. It is a wireless capsule that transmits pH, pressure, and temperature.<br />
This study describes the first use of a novel motility capsule to compare gastric emptying and small bowel transit times in critically ill trauma patients with intracranial hemorrhage with times recorded previously in healthy volunteers. Secondly, this study compares critically ill patients and volunteers on whole-gut transit time. <br />
</p>


<h3>Source</h3>

<p>Rauch et al. 'Use of Wireless Utility Capsule to Determine Gastric Emptying and Small Intestinal Transit Times in Critically Ill Trauma Patients'. Journal of Critical Care 2012; 27(5): 534.e7-534.e12.
</p>

<hr>
<h2 id='strep_tb'>RCT of Streptomycin Therapy for Tuberculosis</h2><span id='topic+strep_tb'></span>

<h3>Description</h3>

<p>Results of a randomized, placebo-controlled, prospective 2-arm trial of streptomycin 2 grams daily (arm A2) vs. placebo (arm A1) to treat tuberculosis in 107 young patients, as reported by the Streptomycin in Tuberculosis Trials Committee in 1948 in the British Medical Journal (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>strep_tb
</code></pre>


<h3>Format</h3>

<p>A data frame with 107 observations and 13 variables
</p>

<dl>
<dt>patient_id</dt><dd><p>invented id number for each participant; type: character</p>
</dd>
<dt>arm</dt><dd><p>assigned treatment arm, Streptomycin or Control; type: factor</p>
</dd>
<dt>dose_strep_g</dt><dd><p>grams, dose of Streptomycin: numeric, 0, 1, or 2 grams</p>
</dd>
<dt>dose_PAS_g</dt><dd><p>grams, dose of PAS (Para-Amino-Salicylate): numeric, 5, 10, or 20 grams. Note that no one in this intial study (study A) received PAS. This was added for combination therapy in studies B and C, as reported in 1952.</p>
</dd>
<dt>gender</dt><dd><p>gender, dichotomous (this was in 1948); type: factor, with levels: M = Male, F= Female</p>
</dd>
<dt>baseline_condition</dt><dd><p>Condition of the Patient at Baseline, 3 levels, 1_Good, 2_Fair, 3_Poor; type: factor</p>
</dd>
<dt>baseline_temp</dt><dd><p>temperature at baseline in degrees fahrenheit or celsius, but categorized into 4 levels (afebrile level apparently were cases not measured with a thermometer): factor, with levels: 1_afebrile, 2_&lt;99F/&lt;37.2C, 3_99-99.9F/37.2-37.75C, 4_100F+/37.7C+</p>
</dd>
<dt>baseline_esr</dt><dd><p>Erythrocyte Sedimentation Rate in mm per hour, categorized into 4 levels, from 0-51+ mm per hour; type: factor, with levels: 1_0-10, 2_11-20, 3_21-50, 4_51+</p>
</dd>
<dt>baseline_cavitation</dt><dd><p>dichotomous presence of cavitation on the baseline chest x-ray; type: factor: 0_no, 1_yes</p>
</dd>
<dt>strep_resistance</dt><dd><p>streptomycin resistance after 6 months of therapy, measured on a 0-100+ scale, categorized into 3 levels - sensitive, moderate, and resistant; type: factor: 1_sens_0-8, 2_mod_8-99, 3_resist_100+</p>
</dd>
<dt>radiologic_6m</dt><dd><p>Likert score rating of radiologic response on chest x-ray at 6 months; type: factor: 1_Death, 2_Considerable_deterioration, 3_Moderate_deterioration, 4_No_change, 5_Moderate_improvement, 6_Considerable_improvement</p>
</dd>
<dt>rad_num</dt><dd><p>Likert score numeric rating of radiologic response on chest x-ray at 6 months; type: numeric: 1-6, from Death to Considerable Improvement</p>
</dd>
<dt>improved</dt><dd><p>Dichotomous outcome of improvement (equal to rad_num of 5-6); type: logical, TRUE or FALSE. 55 of the 107 participants were improved.</p>
</dd>
</dl>



<h3>Details</h3>

<p>The Streptomycin for Tuberculosis trial in 1948 was considered the first modern randomized, placebo-controlled clinical trial, which could be done in part because there were very limited supplies of streptomycin in the UK after World War II. <br />
This publication seems a bit primitive today, without standard features like a proper Table 1, and some creative use of graphs to display baseline characteristics of the study sample <br />
More strikingly, there is no ethics committee approval, or consent. <br />
You can read the pdf of the original journal article at <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2091872/pdf/brmedj03701-0007.pdf">Streptomycin in TB Study</a>. <br />
This was the first of a series of 3 trials, in which the initial effectiveness of Streptomycin was established, but rapid resistance developed, and significant side effects occurred at a dose of 2 grams of streptomycin. This type of resistance also occurred with another new anti-tubercular therapy at the time, PAS (Para-Amino-Salicylate). Subsequent trials B and C evaluated different doses and combinations of Streptomycin and PAS, and were published together in 1952 in the BMJ, with the pdf available here <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2023677/pdf/brmedj03496-0024.pdf">1952 Three Streptomycin in TB Studies Summarized</a>. <br />
Commentary on the conduct of these trials from one of the MD investigators can be found at <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1592068/">MD Clinical Trialist Commentary</a>. <br />
Commentary on the design and analysis of these trials from statistician A. Bradford Hill can be found at <a href="https://www.sciencedirect.com/science/article/abs/pii/019724569090001I">Statistican Commentary</a>.
</p>


<h3>Source</h3>

<p>This data set is reconstructed to the best of my ability from the paper in the British Medical Journal from 1948, entitled, Streptomycin Treatment of Pulmonary Tuberculosis, pages 769-782 in the October 30, 1948 edition, authored by the Streptomycin in Tuberculosis Trials Committee.
You can find the pdf at <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2091872/pdf/brmedj03701-0007.pdf">Streptomycin in TB</a>.
</p>

<hr>
<h2 id='supraclavicular'>Study of Supraclavicular Anesthesia</h2><span id='topic+supraclavicular'></span>

<h3>Description</h3>

<p>This data set contains 103 patients who were scheduled to undergo an upper extremity procedure suitable for supraclavicular anesthesia. Patients were randomly assigned to either (1) combined group-ropivacaine and mepivacaine mixture; or (2) sequential group-mepivacaine followed by ropivacaine. A number of demographic and post-op pain medication variables (fentanyl, alfentanil, midazolam) were collected. The primary outcome is time to 4-nerve sensory block onset. The dataset is cleaned and relatively complete. There are no outliers or data problems (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>supraclavicular
</code></pre>


<h3>Format</h3>

<p>A data frame with 103 observations and 17 variables
</p>

<dl>
<dt><code>subject</code></dt><dd><p>Subject ID, numeric, range: 1-103</p>
</dd>
<dt><code>group</code></dt><dd><p>Anesthetic group, numeric, 1 = Mixture; 2 = Sequential</p>
</dd>
<dt><code>gender</code></dt><dd><p>Gender, numeric, 1 = Male; 0= Female</p>
</dd>
<dt><code>bmi</code></dt><dd><p>Body mass index (kg/m^2), numeric, range:19-43.5</p>
</dd>
<dt><code>age</code></dt><dd><p>Age (years), numeric, range:18-74</p>
</dd>
<dt><code>fentanyl</code></dt><dd><p>Fentanyl pain medication (micrograms), numeric, range: 0-250.0</p>
</dd>
<dt><code>alfentanil</code></dt><dd><p>Alfentanil pain medication (milligrams), numeric, range: 0-4.3</p>
</dd>
<dt><code>midazolam</code></dt><dd><p>Midazolam hypnotic-sedative medication, numeric, range: 0-9.0</p>
</dd>
<dt><code>onset_sensory</code></dt><dd><p>Time to 4 nerve sensory block onset or, if onset_sensory block failed the observed worst outcome of minutes for any patient (50 minutes), numeric, range: 0-50.0</p>
</dd>
<dt><code>onset_first_sensory</code></dt><dd><p>Time to first sensory block in minutes, or if block failed, a value of 15 minutes, numeric, range: 6-15.0</p>
</dd>
<dt><code>onset_motor</code></dt><dd><p>Time to complete motor block or, if motor block failed, the observed worst outcome of minutes for any patient (50 minutes), numeric, range: 1-50.0</p>
</dd>
<dt><code>nerve_block_censor</code></dt><dd><p>block failed, numeric, 0 = nerve block succeeded, 1 = block failed (censored)</p>
</dd>
<dt><code>med_duration</code></dt><dd><p>Time from the onset of 4 nerve sensory block until the first request for an analgesic medication (hours), numeric, range: 0-48.0</p>
</dd>
<dt><code>med_censor</code></dt><dd><p>Patients who did not take an analgesic were censored at 48 hours, numeric, 0 = nerve  succeeded, 1 = block failed (censored)</p>
</dd>
<dt><code>vps_rest</code></dt><dd><p>Maximum postop verbal pain score (at rest), on  11 point Likert scale (0-10), numeric, range: 0-10</p>
</dd>
<dt><code>vps_movement</code></dt><dd><p>Maximum postop verbal pain score (with movement), on  11 point Likert scale (0-10), numeric, range: 0-10</p>
</dd>
<dt><code>opioid_total</code></dt><dd><p>Total opioid consumption in milligrams, numeric, range: 0-225.0</p>
</dd>
</dl>



<h3>Details</h3>

<p>The choice of anesthetic technique combined with a suitable plan for postoperative analgesia can facilitate early discharge, improve patient comfort, and increase overall satisfaction. Patients having painful procedures who undergo general anesthesia have a 2- to 5-fold greater risk of unplanned overnight admissions compared with those having regional anesthesia. Regional anesthetic techniques and peripheral nerve blocks are especially favored for surgeries on the extremities. Both rapid onset of the block and prolonged postoperative analgesia are desired characteristics of regional anesthesia.<br />
The choice of local anesthetics or combinations thereof can greatly influence the effectiveness of the block, onset time, duration of postoperative analgesia, need for opioid use, and patient satisfaction. Mepivacaine and ropivacaine are commonly used in peripheral nerve blocks, their drawbacks being a short duration with 1.5% mepivacaine and a delayed onset with 0.5% ropivacaine. An ideal local anesthetic with high potency, low toxicity, rapid onset, and prolonged duration does not exist yet. Investigators have therefore tried mixtures of local anesthetics in an attempt to combine their advantages with conflicting results. A potential problem is that mixing drugs dilutes the effects of each. Thus, a mixture of a rapid-onset drug such as mepivacaine with a long-acting one such as ropivacaine may well result in slower onset than mepivacaine alone and shorter duration of action than ropivacaine alone. In contrast, sequential administration of the same amounts of the same drugs may preserve the desirable features of each.<br />
Objective: This study investigates whether sequential supraclavicular injection of 1.5% mepivacaine followed 90 seconds later by 0.5% ropivacaine provides a quicker onset and a longer duration of analgesia than an equidose combination of the 2 local anesthetics.<br />
</p>


<h3>Source</h3>

<p>These are data from a study by Roberman et al. 'Combined Versus Sequential Injection of Mepivacaine and Ropivacaine for Supraclavicular Nerve Blocks'. Reg Anesth Pain Med 2011; 36:145-50.
</p>

<hr>
<h2 id='theoph'>Cohort Study of the Pharmacokinetics of Oral Theophylline</h2><span id='topic+theoph'></span>

<h3>Description</h3>

<p>Results of a Cohort Study of the Pharmacokinetics of Oral Theophylline, with plasma concentrations over time (<strong>more details</strong> available below the variable definitions). <br />
</p>


<h3>Usage</h3>

<pre><code class='language-R'>theoph
</code></pre>


<h3>Format</h3>

<p>A data frame with 132 observations and 5 variables
</p>

<dl>
<dt>Subject</dt><dd><p>subject id number for each participant; type: ordinal factor</p>
</dd>
<dt>Wt</dt><dd><p>Weight in kilograms; type: double</p>
</dd>
<dt>Dose</dt><dd><p>Dose in milligrams per kilogram; type: double</p>
</dd>
<dt>Time</dt><dd><p>Time from initial dose in hours; type: double</p>
</dd>
<dt>conc</dt><dd><p>Concentration of theophylline in the plasma in micrograms per milliliter' type: double</p>
</dd>
</dl>



<h3>Details</h3>

<p>This data set is from a pharmacokinetic study of oral dosing of the anti-asthma medication, theophylline, in 12 subjects over 25 hours, published By Dr. Robert A. Upton around 1980. The original publication, if any, is unclear and not cited. These data were used in a package named <code>nlme</code>, and reported in Boeckmann, A.J., et al.Dr. Upton did publish several papers on theophylline pharmacokinetics around 1980-1984, and these data could have been from one of these.<br />
Theophylline is an methylxanthine anti-asthma medication, which acts as a bronchodilator, with secondary effects to strengthen diaphragm contraction, reduce pulmonary artery pressures, and reduce mast cell release. It can be administered by the intravenous, oral, or rectal suppository routes.<br />
Each subject in this Study (oral route) received a single oral dose of theophylline.
Blood samples were taken at frequent intervals over the first 25 hours after dosing, and the quantity of theophylline in the plasma at each time point was measured in micrograms per milliliter. <br />
Unfortunately, the theophylline plasma level in blood varies considerably between patients, because of differences in drug clearance, which is affected by body mass, age, smoking, liver and heart function, and viral infections. To complicate this drug further, it has important interactions with a number of other common medicines which can increase or decrease the drug level. Each subject in this study received a single oral dose of 300 mg of theophylline, which has been converted to a milligrams per kilogram dose. Blood samples were taken at frequent intervals over the next 25 hours after dosing, and the quantity of theophylline in the plasma at each time point was measured in micrograms per milliliter of plasma. <br />
</p>


<h3>Source</h3>

<p>Boeckmann, A. J., Sheiner, L. B. and Beal, S. L. (1994), NONMEM Users Guide: Part V, NONMEM Project Group, University of California, San Francisco.
Note that the original data collector, Robert A. Upton, is not credited, nor is the original work cited.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
