<!DOCTYPE html><html><head><title>Help for package MendelianRandomization</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MendelianRandomization}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BF_dist'><p>Generate bootstrap samples for the bootstrapping Fieller's confidence interval of the penalized inverse-variance weighted (pIVW) method</p></a></li>
<li><a href='#calcium'><p>Data on effect of calcium on fasting glucose (correlated variants)</p></a></li>
<li><a href='#ci_normal'><p>Calculate confidence intervals using the normal distribution</p></a></li>
<li><a href='#ci_t'><p>Calculate confidence intervals using the t-distribution</p></a></li>
<li><a href='#CLR-class'><p>CLR Class</p></a></li>
<li><a href='#cML_estimate'><p>Estimate with Regular Likelihood</p></a></li>
<li><a href='#cML_estimate_random'><p>Estimate with Regular Likelihood Using Multiple Random Start Points</p></a></li>
<li><a href='#cML_SdTheta'><p>Standard Error of Estimated Theta</p></a></li>
<li><a href='#condFstat'><p>Calculates conditional F-statistic for each risk factor using summarized data</p></a></li>
<li><a href='#coursedata'><p>Course data</p></a></li>
<li><a href='#decimals'><p>Produce nicely rounded numbers</p></a></li>
<li><a href='#DIVW-class'><p>DIVW Class</p></a></li>
<li><a href='#Egger-class'><p>Egger Class</p></a></li>
<li><a href='#egger.bounds'><p>Calculates confidence intervals for the MR-Egger method</p></a></li>
<li><a href='#getter'><p>Applies method $ to different classes</p></a></li>
<li><a href='#het.weight'><p>Heterogeneity-penalized weight function</p></a></li>
<li><a href='#invcov_mvmr'><p>Generate the list of inverse of covariance matrices used in <code>MVMR-cML-DP</code></p></a></li>
<li><a href='#IVW-class'><p>IVW Class</p></a></li>
<li><a href='#ldlc'><p>Data on lipid effects on coronary artery disease (uncorrelated variants)</p></a></li>
<li><a href='#loglikelihood'><p>Calculates log-likelihood with uncorrelated variants in two-sample setting (no correlation from sample overlap)</p></a></li>
<li><a href='#loglikelihoodcorrel'><p>Calculates log-likelihood with correlated variants in two-sample setting (no correlation from sample overlap)</p></a></li>
<li><a href='#loglikelihoodrhocorrel'><p>Calculates log-likelihood with correlation from sample overlap</p></a></li>
<li><a href='#MaxLik-class'><p>MaxLik Class</p></a></li>
<li><a href='#mbe_boot'><p>Mode-based estimate (Hartwig) bootstrap function</p></a></li>
<li><a href='#mbe_est'><p>Mode-based estimate (Hartwig) estimation function</p></a></li>
<li><a href='#model.prior'><p>Prior weight function</p></a></li>
<li><a href='#mr_allmethods'><p>Mendelian randomization estimation using all methods</p></a></li>
<li><a href='#mr_clr'><p>Conditional likelihood ratio (CLR) method</p></a></li>
<li><a href='#mr_cML'><p>Constrained maximum likelihood (cML) method</p></a></li>
<li><a href='#mr_conmix'><p>Contamination mixture method</p></a></li>
<li><a href='#mr_divw'><p>Debiased inverse-variance weighted method</p></a></li>
<li><a href='#mr_egger'><p>MR-Egger method</p></a></li>
<li><a href='#mr_forest'><p>Draw a forest plot of causal estimates</p></a></li>
<li><a href='#mr_funnel'><p>Draw a funnel plot of variant-specific estimates</p></a></li>
<li><a href='#mr_hetpen'><p>Heterogeneity-penalized method</p></a></li>
<li><a href='#mr_input'><p>Inputting and formatting data for use in causal estimation</p></a></li>
<li><a href='#mr_ivw'><p>Inverse-variance weighted method</p></a></li>
<li><a href='#mr_lasso'><p>MR-Lasso method</p></a></li>
<li><a href='#mr_loo'><p>Leave-one-out estimates</p></a></li>
<li><a href='#mr_maxlik'><p>Maximum-likelihood method</p></a></li>
<li><a href='#mr_mbe'><p>Mode-based method of Hartwig</p></a></li>
<li><a href='#mr_median'><p>Median-based method</p></a></li>
<li><a href='#mr_mvcML'><p>Multivariable constrained maximum likelihood method</p></a></li>
<li><a href='#mr_mvegger'><p>Multivariable MR-Egger method</p></a></li>
<li><a href='#mr_mvgmm'><p>Multivariable generalized method of moments (GMM) method</p></a></li>
<li><a href='#mr_mvinput'><p>Inputting and formatting data for use in causal estimation</p></a></li>
<li><a href='#mr_mvivw'><p>Multivariable inverse-variance weighted method</p></a></li>
<li><a href='#mr_mvivwme'><p>Multivariable inverse-variance weighted method with measurement error</p></a></li>
<li><a href='#mr_mvlasso'><p>Multivariable MR-Lasso method</p></a></li>
<li><a href='#mr_mvmedian'><p>Multivariable median-based method</p></a></li>
<li><a href='#mr_mvpcgmm'><p>Multivariable principal components generalized method of moments (PC-GMM) method</p></a></li>
<li><a href='#mr_pcgmm'><p>Univariable principal components generalized method of moments (PC-GMM) method</p></a></li>
<li><a href='#mr_pivw'><p>Penalized inverse-variance weighted method</p></a></li>
<li><a href='#mr_plot'><p>Draw a scatter plot of the genetic associations and/or causal estimates</p></a></li>
<li><a href='#MRAll-class'><p>MRAll Class</p></a></li>
<li><a href='#MRcML-class'><p>MRcML Class</p></a></li>
<li><a href='#MRConMix-class'><p>MRConMix Class</p></a></li>
<li><a href='#MRHetPen-class'><p>MRHetPen Class</p></a></li>
<li><a href='#MRInput-class'><p>MRInput Class</p></a></li>
<li><a href='#MRLasso-class'><p>MRLasso class</p></a></li>
<li><a href='#MRMBE-class'><p>MRMBE Class</p></a></li>
<li><a href='#MRMVInput-class'><p>MRMVInput Class</p></a></li>
<li><a href='#mv_norm'><p>Sampling from multivariate normal distribution</p></a></li>
<li><a href='#MVcML_SdTheta'><p>Standard error estimate for MVMR-cML-BIC</p></a></li>
<li><a href='#MVEgger-class'><p>MVEgger Class</p></a></li>
<li><a href='#MVGMM-class'><p>MVGMM Class</p></a></li>
<li><a href='#MVIVW-class'><p>MVIVW Class</p></a></li>
<li><a href='#MVIVWME-class'><p>MVIVWME Class</p></a></li>
<li><a href='#MVLasso-class'><p>MRMVLasso class</p></a></li>
<li><a href='#MVMedian-class'><p>MRMVMedian class</p></a></li>
<li><a href='#MVmr_cML'><p>MVMRcML method with Data Perturbation</p></a></li>
<li><a href='#MVmr_cML_DP'><p>MVMRcML method with Data Perturbation</p></a></li>
<li><a href='#MVMRcML-class'><p>MVMRcML Class</p></a></li>
<li><a href='#MVPCGMM-class'><p>MVPCGMM Class</p></a></li>
<li><a href='#PCGMM-class'><p>PCGMM Class</p></a></li>
<li><a href='#penalised.weights'><p>Calculates p-values for penalization of weights</p></a></li>
<li><a href='#penalised.weights.delta'><p>Calculates p-values for penalization of weights with second-order weights</p></a></li>
<li><a href='#pheno_input'><p>Extract summarized data from PhenoScanner</p></a></li>
<li><a href='#phenoscanner'><p>PhenoScanner</p></a></li>
<li><a href='#PIVW-class'><p>PIVW Class</p></a></li>
<li><a href='#pl'><p>Profile likelihood of valid IVs</p></a></li>
<li><a href='#r.weights'><p>Calculates p-values for penalization of weights</p></a></li>
<li><a href='#r.weights.delta'><p>Calculates p-values for penalization of weights with second-order weights</p></a></li>
<li><a href='#simpleCap'><p>Capitalize a word</p></a></li>
<li><a href='#values'><p>Applies method values() to different output classes</p></a></li>
<li><a href='#weighted.median'><p>Weighted median function</p></a></li>
<li><a href='#weighted.median.boot.se'><p>Weighted median standard error function</p></a></li>
<li><a href='#WeightedMedian-class'><p>WeightedMedian Class</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Mendelian Randomization Package</td>
</tr>
<tr>
<td>Version:</td>
<td>0.10.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-04-12</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Stephen Burgess &lt;sb452@medschl.cam.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Encodes several methods for performing Mendelian randomization
    analyses with summarized data. Summarized data on genetic associations with the
    exposure and with the outcome can be obtained from large consortia. These data
    can be used for obtaining causal estimates using instrumental variable methods.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.1), methods</td>
</tr>
<tr>
<td>Imports:</td>
<td>knitr, rmarkdown, plotly (&ge; 3.6.0), ggplot2 (&ge; 1.0.1),
robustbase (&ge; 0.92-6), Matrix (&ge; 1.2), iterpc (&ge; 0.3),
quantreg (&ge; 5.01), rjson, glmnet, numDeriv, Rcpp</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-12 09:48:50 UTC; sb452</td>
</tr>
<tr>
<td>Author:</td>
<td>Stephen Burgess <a href="https://orcid.org/0000-0001-5365-8760"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Olena Yavorska [aut],
  James Staley <a href="https://orcid.org/0009-0001-9520-5011"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Fernando Hartwig <a href="https://orcid.org/0000-0003-3729-0710"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Jim Broadbent <a href="https://orcid.org/0000-0002-5685-6241"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Christopher Foley <a href="https://orcid.org/0000-0002-0970-2610"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Andrew Grant [ctb],
  Amy Mason <a href="https://orcid.org/0000-0002-8019-0777"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Ting Ye <a href="https://orcid.org/0000-0001-6009-641X"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Haoran Xue <a href="https://orcid.org/0000-0001-7923-6128"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [ctb],
  Zhaotong Lin <a href="https://orcid.org/0000-0001-8723-4392"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Siqi Xu [ctb],
  Ashish Patel <a href="https://orcid.org/0000-0002-0350-4365"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb],
  Hyunseung Kang [ctb],
  Sheng Wang [ctb],
  Ville Karhunen <a href="https://orcid.org/0000-0001-6064-1588"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-12 10:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='BF_dist'>Generate bootstrap samples for the bootstrapping Fieller's confidence interval of the penalized inverse-variance weighted (pIVW) method</h2><span id='topic+BF_dist'></span>

<h3>Description</h3>

<p>Internal function of the penalized inverse-variance weighted (pIVW) method, which generates bootstrap samples for the bootstrapping Fieller's confidence interval.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BF_dist(
  object,
  beta_hat = 0,
  tau2 = 0,
  lambda = 1,
  n_boot = 1000,
  seed_boot = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="BF_dist_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="BF_dist_+3A_beta_hat">beta_hat</code></td>
<td>
<p>The causal effect estimate.</p>
</td></tr>
<tr><td><code id="BF_dist_+3A_tau2">tau2</code></td>
<td>
<p>The estimated variance of the horizontal pleiotropy.</p>
</td></tr>
<tr><td><code id="BF_dist_+3A_lambda">lambda</code></td>
<td>
<p>The penalty parameter in the pIVW estimator. By default, <code>lambda=1</code>.</p>
</td></tr>
<tr><td><code id="BF_dist_+3A_n_boot">n_boot</code></td>
<td>
<p>The sample size of the bootstrap samples. By default, <code>n_boot=1000</code>.</p>
</td></tr>
<tr><td><code id="BF_dist_+3A_seed_boot">seed_boot</code></td>
<td>
<p>The seed for random sampling in the bootstrap method. By default, <code>seed_boot=1</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>z_b</code></td>
<td>
<p>A vector containing the bootstrap samples for the bootstrapping Fieller's confidence interval.</p>
</td></tr>
</table>

<hr>
<h2 id='calcium'>Data on effect of calcium on fasting glucose (correlated variants)</h2><span id='topic+calcium'></span><span id='topic+calciumse'></span><span id='topic+fastgluc'></span><span id='topic+fastglucse'></span><span id='topic+calc.rho'></span>

<h3>Description</h3>

<p>Two sets of example data are included in the package: one illustrating uncorrelated variants, and the other correlated variants. These are the data on correlated variants.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcium

calciumse

fastgluc

fastglucse

calc.rho
</code></pre>


<h3>Format</h3>

<p>An object of class <code>numeric</code> of length 6.
</p>
<p>An object of class <code>numeric</code> of length 6.
</p>
<p>An object of class <code>numeric</code> of length 6.
</p>
<p>An object of class <code>numeric</code> of length 6.
</p>
<p>An object of class <code>matrix</code> (inherits from <code>array</code>) with 6 rows and 6 columns.
</p>


<h3>Details</h3>

<p>The variables <code>calcium</code>, and <code>fastgluc</code> are the genetic associations with calcium and fasting glucose for 6 genetic variants reported by Burgess et al (2015). The respective standard errors of the associations are given as <code>calciumse</code> and <code>fastglucse</code>. The matrix of correlations between the genetic variants is given as <code>calc.rho</code>.
</p>
<p>These data can be used to test out the various functions in the package.
</p>


<h3>References</h3>

<p>Stephen Burgess, Robert A Scott, Nic J Timpson, George Davey Smith, Simon G Thompson. Using published data in Mendelian randomization: a blueprint for efficient identification of causal risk factors. Eur J Epidemiol 2015; 30(7):543-552. doi: 10.1007/s10654-015-0011-z.
</p>

<hr>
<h2 id='ci_normal'>Calculate confidence intervals using the normal distribution</h2><span id='topic+ci_normal'></span>

<h3>Description</h3>

<p>Internal function for calculating confidence intervals using the normal distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ci_normal(type, mean, se, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ci_normal_+3A_type">type</code></td>
<td>
<p>&quot;l&quot; for lower, &quot;u&quot; for upper.</p>
</td></tr>
<tr><td><code id="ci_normal_+3A_mean">mean</code></td>
<td>
<p>Causal estimate.</p>
</td></tr>
<tr><td><code id="ci_normal_+3A_se">se</code></td>
<td>
<p>Standard error of estimate.</p>
</td></tr>
<tr><td><code id="ci_normal_+3A_alpha">alpha</code></td>
<td>
<p>Significance level.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Numeric value of confidence interval limit.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ci_normal(type = "l", mean = 0, se = 1, alpha = 0.05)

</code></pre>

<hr>
<h2 id='ci_t'>Calculate confidence intervals using the t-distribution</h2><span id='topic+ci_t'></span>

<h3>Description</h3>

<p>Internal function for calculating confidence intervals using the t-distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ci_t(type, mean, se, df, alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ci_t_+3A_type">type</code></td>
<td>
<p>&quot;l&quot; for lower, &quot;u&quot; for upper.</p>
</td></tr>
<tr><td><code id="ci_t_+3A_mean">mean</code></td>
<td>
<p>Causal estimate.</p>
</td></tr>
<tr><td><code id="ci_t_+3A_se">se</code></td>
<td>
<p>Standard error of estimate.</p>
</td></tr>
<tr><td><code id="ci_t_+3A_df">df</code></td>
<td>
<p>Number of degrees of freedom.</p>
</td></tr>
<tr><td><code id="ci_t_+3A_alpha">alpha</code></td>
<td>
<p>Significance level.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Numeric value of confidence interval limit.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>ci_t(type = "l", mean = 0, se = 1, df = 8, alpha = 0.05)

</code></pre>

<hr>
<h2 id='CLR-class'>CLR Class</h2><span id='topic+CLR-class'></span>

<h3>Description</h3>

<p>An object containing the confidence intervals produced using the conditional likelihood ratio method (CLR) and other identification-robust methods.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>ARlower</code></dt><dd><p>The lower bounds of the causal estimate based on inverting Anderson and Rubin's test.</p>
</dd>
<dt><code>ARupper</code></dt><dd><p>The upper bounds of the causal estimate based on inverting Anderson and Rubin's test.</p>
</dd>
<dt><code>Klower</code></dt><dd><p>The lower bounds of the causal estimate based on inverting Kleibergen's test.</p>
</dd>
<dt><code>Kupper</code></dt><dd><p>The upper bounds of the causal estimate based on inverting Kleibergen's test.</p>
</dd>
<dt><code>CLRlower</code></dt><dd><p>The lower bounds of the causal estimate based on inverting Moreira's conditional likelihood ratio test.</p>
</dd>
<dt><code>CLRupper</code></dt><dd><p>The upper bounds of the causal estimate based on inverting Moreira's conditional likelihood ratio test.</p>
</dd>
<dt><code>CIMin</code></dt><dd><p>The smallest value used in the search to find the confidence interval.</p>
</dd>
<dt><code>CIMax</code></dt><dd><p>The largest value used in the search to find the confidence interval.</p>
</dd>
<dt><code>CIStep</code></dt><dd><p>The step size used in the search to find the confidence interval.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
</dl>

<hr>
<h2 id='cML_estimate'>Estimate with Regular Likelihood</h2><span id='topic+cML_estimate'></span>

<h3>Description</h3>

<p>Internal function of mr_cML.
Estimate theta, b vector, r vector with constrained maximum likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cML_estimate(
  b_exp,
  b_out,
  se_exp,
  se_out,
  K,
  initial_theta = 0,
  initial_mu = rep(0, length(b_exp)),
  maxit = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cML_estimate_+3A_b_exp">b_exp</code></td>
<td>
<p>Vector of estimated effects for exposure.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_b_out">b_out</code></td>
<td>
<p>Vector or estimated effects for outcome.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_se_exp">se_exp</code></td>
<td>
<p>Vector of standard errors for exposure.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_se_out">se_out</code></td>
<td>
<p>Vector of standard errors for outcome.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_k">K</code></td>
<td>
<p>Constraint parameter, number of invalid IVs.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_initial_theta">initial_theta</code></td>
<td>
<p>Starting point for theta.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_initial_mu">initial_mu</code></td>
<td>
<p>Starting point for mu.</p>
</td></tr>
<tr><td><code id="cML_estimate_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iteration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains: theta is the estimate causal effect,
b_vec is the estimated vector of b,
r_vec is the estimated vector of r.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cML_estimate(b_exp = ldlc,b_out = chdlodds,se_exp = ldlcse,
se_out = chdloddsse, K = 5)
</code></pre>

<hr>
<h2 id='cML_estimate_random'>Estimate with Regular Likelihood Using Multiple Random Start Points</h2><span id='topic+cML_estimate_random'></span>

<h3>Description</h3>

<p>Internal function of mr_cML. 
Get estimated theta, se of estimated theta
and negative log-likelihood,
using multiple random starting points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cML_estimate_random(
  b_exp,
  b_out,
  se_exp,
  se_out,
  K,
  random_start = 0,
  maxit = 100
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cML_estimate_random_+3A_b_exp">b_exp</code></td>
<td>
<p>Vector of estimated effects for exposure.</p>
</td></tr>
<tr><td><code id="cML_estimate_random_+3A_b_out">b_out</code></td>
<td>
<p>Vector or estimated effects for outcome.</p>
</td></tr>
<tr><td><code id="cML_estimate_random_+3A_se_exp">se_exp</code></td>
<td>
<p>Vector of standard errors for exposure.</p>
</td></tr>
<tr><td><code id="cML_estimate_random_+3A_se_out">se_out</code></td>
<td>
<p>Vector of standard errors for outcome.</p>
</td></tr>
<tr><td><code id="cML_estimate_random_+3A_k">K</code></td>
<td>
<p>Constraint parameter, number of invalid IVs.</p>
</td></tr>
<tr><td><code id="cML_estimate_random_+3A_random_start">random_start</code></td>
<td>
<p>Number of random starting points, default is 0.</p>
</td></tr>
<tr><td><code id="cML_estimate_random_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iteration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains: theta is the estimate causal effect,
se is standard error of estimated theta,
l is negative log-likelihood,
r_est is estimated r vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>cML_estimate_random(b_exp = ldlc,b_out = chdlodds,se_exp = ldlcse,
se_out = chdloddsse, K = 5)
</code></pre>

<hr>
<h2 id='cML_SdTheta'>Standard Error of Estimated Theta</h2><span id='topic+cML_SdTheta'></span>

<h3>Description</h3>

<p>Internal function of mr_cML.
Get the standard error of estimated theta from constrained maximum likelihood.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cML_SdTheta(b_exp, b_out, se_exp, se_out, theta, b_vec, r_vec)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cML_SdTheta_+3A_b_exp">b_exp</code></td>
<td>
<p>Vector of estimated effects for exposure.</p>
</td></tr>
<tr><td><code id="cML_SdTheta_+3A_b_out">b_out</code></td>
<td>
<p>Vector or estimated effects for outcome.</p>
</td></tr>
<tr><td><code id="cML_SdTheta_+3A_se_exp">se_exp</code></td>
<td>
<p>Vector of standard errors for exposure.</p>
</td></tr>
<tr><td><code id="cML_SdTheta_+3A_se_out">se_out</code></td>
<td>
<p>Vector of standard errors for outcome.</p>
</td></tr>
<tr><td><code id="cML_SdTheta_+3A_theta">theta</code></td>
<td>
<p>Estimated theta from cML.</p>
</td></tr>
<tr><td><code id="cML_SdTheta_+3A_b_vec">b_vec</code></td>
<td>
<p>Estimated vector of b from cML.</p>
</td></tr>
<tr><td><code id="cML_SdTheta_+3A_r_vec">r_vec</code></td>
<td>
<p>Estimated vector of r from cML.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Standard error of theta.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># First get estimates:
MLE_result = cML_estimate(b_exp = ldlc,b_out = chdlodds,se_exp = ldlcse,
se_out = chdloddsse, K = 5)

# Calculate standard error:
cML_SdTheta(b_exp = ldlc,b_out = chdlodds,se_exp = ldlcse,
se_out = chdloddsse, theta = MLE_result$theta, b_vec = MLE_result$b_vec, r_vec = MLE_result$r_vec)
</code></pre>

<hr>
<h2 id='condFstat'>Calculates conditional F-statistic for each risk factor using summarized data</h2><span id='topic+condFstat'></span>

<h3>Description</h3>

<p>Internal function for calculating conditional F-statistics for the output <code>condFstat</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>condFstat(.Bx, .Bxse, .nx, .ld, .cor.x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="condFstat_+3A_.bx">.Bx</code></td>
<td>
<p>Beta-coefficient for genetic associations with the risk factors.</p>
</td></tr>
<tr><td><code id="condFstat_+3A_.bxse">.Bxse</code></td>
<td>
<p>Standard error of genetic associations with the risk factors.</p>
</td></tr>
<tr><td><code id="condFstat_+3A_.nx">.nx</code></td>
<td>
<p>sample sizes used to compute genetic associations with the risk factors.</p>
</td></tr>
<tr><td><code id="condFstat_+3A_.ld">.ld</code></td>
<td>
<p>genetic variant correlation matrix</p>
</td></tr>
<tr><td><code id="condFstat_+3A_.cor.x">.cor.x</code></td>
<td>
<p>Risk factor correlation matrix</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Conditional F-statistic for each exposure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>condFstat(.Bx = cbind(ldlc, hdlc, trig), .Bxse = cbind(ldlcse, hdlcse, trigse),
 .nx = rep(17723,3), .ld=diag(length(ldlc)), .cor.x=diag(3))

</code></pre>

<hr>
<h2 id='coursedata'>Course data</h2><span id='topic+coursedata'></span>

<h3>Description</h3>

<p>Data required for Mendelian randomization course.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coursedata
</code></pre>


<h3>Format</h3>

<p>An object of class <code>list</code> of length 11.
</p>

<hr>
<h2 id='decimals'>Produce nicely rounded numbers</h2><span id='topic+decimals'></span>

<h3>Description</h3>

<p>Internal function for putting extra zeros on numbers (if needed).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decimals(number, places)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="decimals_+3A_number">number</code></td>
<td>
<p>A number to be rounded.</p>
</td></tr>
<tr><td><code id="decimals_+3A_places">places</code></td>
<td>
<p>Number of decimal places.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Returns a nicely formatted number. For example, 0.3 to three decimal places would be rendered as 0.300.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>decimals(number = 0.3, places = 3)

</code></pre>

<hr>
<h2 id='DIVW-class'>DIVW Class</h2><span id='topic+DIVW-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the debiased inverse-variance weighted (dIVW) method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Over.dispersion</code></dt><dd><p>Should the method consider overdispersion (balanced horizontal pleiotropy)? Default is TRUE.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal point estimate from the median-based method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code> (obtained from bootstrapping).</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate from the Wald method.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that used in the calculation.</p>
</dd>
<dt><code>Condition</code></dt><dd><p>A measure defined as (average F-statistic -1)*sqrt(# snps) that needs to be large for reliable asymptotic approximation based on the dIVW estimator. It is recommended to be greater than 20.</p>
</dd>
</dl>

<hr>
<h2 id='Egger-class'>Egger Class</h2><span id='topic+Egger-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the MR-Egger method as well as various statistics.
</p>
<p>The MR-Egger model uses a random-effects model; a fixed-effect model does not make sense as pleiotropy leads to heterogeneity between the causal estimates targeted by the genetic variants. The (multiplicative) random-effects model allows over-dispersion in the regression model. Under-dispersion is not permitted (in case of under-dispersion, the residual standard error is set to 1).
</p>


<h3>Slots</h3>


<dl>
<dt><code>Model</code></dt><dd><p>Model always takes the value <code>random</code>, as only random-effects analyses are permitted.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Robust</code></dt><dd><p>Whether robust regression was used in the regression model relating the genetic associations with the outcome and those with the exposure.</p>
</dd>
<dt><code>Penalized</code></dt><dd><p>Whether weights in the regression model were penalized for variants with heterogeneous causal estimates.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal point estimate from the MR-Egger method.</p>
</dd>
<dt><code>StdError.Est</code></dt><dd><p>The standard error associated with <code>Estimate</code>.</p>
</dd>
<dt><code>Pvalue.Est</code></dt><dd><p>P-value associated with the causal estimate from the Wald method.</p>
</dd>
<dt><code>CILower.Est</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code> based on <code>StdError.Est</code>.</p>
</dd>
<dt><code>CIUpper.Est</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError.Est</code>.</p>
</dd>
<dt><code>Intercept</code></dt><dd><p>The intercept estimate from the MR-Egger method. Under the InSIDE assumption, the intercept represents the average pleiotropic effect (average direct effect on the outcome) of a genetic variant. If the intercept differs from zero, this is evidence that the genetic variants are not all valid instruments; specifically, there is directional pleiotropy.</p>
</dd>
<dt><code>StdError.Int</code></dt><dd><p>The standard error associated with <code>Intercept</code>.</p>
</dd>
<dt><code>Pvalue.Int</code></dt><dd><p>P-value associated with the intercept from the Wald method.</p>
</dd>
<dt><code>CILower.Int</code></dt><dd><p>The lower bound of the confidence interval for <code>Intercept</code> based on <code>StdError.Int</code>.</p>
</dd>
<dt><code>CIUpper.Int</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError.Int</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>Causal.pval</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>Pleio.pval</code></dt><dd><p>P-value associated with the intercept (p-value for the MR-Egger intercept test of directional pleiotropy).</p>
</dd>
<dt><code>RSE</code></dt><dd><p>The estimated residual standard error from the regression model.</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that the MR-Egger regression model describes the associations with the outcome with no excess heterogeneity.</p>
</dd>
<dt><code>I.sq</code></dt><dd><p>A measure of heterogeneity between the genetic associations with the exposure (see Bowden IJE 2016: &quot;Assessing the suitability of summary data for Mendelian randomization analyses using MR-Egger regression: The role of the I2 statistic.&quot;). Low values of <code>I.sq</code> relate both to large differences in precision between MR-Egger and IVW estimates, and to more weak instrument bias (in a two-sample setting, this is attenuation of MR-Egger estimate towards the null).</p>
</dd>
</dl>

<hr>
<h2 id='egger.bounds'>Calculates confidence intervals for the MR-Egger method</h2><span id='topic+egger.bounds'></span>

<h3>Description</h3>

<p>Internal function for calculating confidence intervals for the MR-Egger method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>egger.bounds(type, dist, .theta, .thetase, df = 0, .rse, .alpha)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="egger.bounds_+3A_type">type</code></td>
<td>
<p>&quot;l&quot; for lower, &quot;u&quot; for upper.</p>
</td></tr>
<tr><td><code id="egger.bounds_+3A_dist">dist</code></td>
<td>
<p>&quot;normal&quot; for normal distribution, &quot;t-dist&quot; for t-distribution.</p>
</td></tr>
<tr><td><code id="egger.bounds_+3A_.theta">.theta</code></td>
<td>
<p>Causal estimate.</p>
</td></tr>
<tr><td><code id="egger.bounds_+3A_.thetase">.thetase</code></td>
<td>
<p>Standard error of estimate.</p>
</td></tr>
<tr><td><code id="egger.bounds_+3A_df">df</code></td>
<td>
<p>Degrees of freedom (for t-distribution).</p>
</td></tr>
<tr><td><code id="egger.bounds_+3A_.rse">.rse</code></td>
<td>
<p>Residual standard error.</p>
</td></tr>
<tr><td><code id="egger.bounds_+3A_.alpha">.alpha</code></td>
<td>
<p>Significance level.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The slight complication of this function is that when the estimate of the residual standard error (RSE) is less than one, the t-distribution confidence interval is calculated as either the confidence interval using a normal deviate and setting the RSE to 1, or a t-distribution deviate and using the estimated RSE. The wider of the two intervals is reported. This ensures that under-dispersion is not doubly penalized, while also making sure that the estimate is no more precise than that from a fixed-effect analysis.
</p>
<p>None.
</p>


<h3>Value</h3>

<p>Numeric value of confidence interval limit.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>egger.bounds(type = "l", dist = "normal", .theta = 0, .thetase = 1, df = 0, .rse = 1, .alpha = 0.05)

</code></pre>

<hr>
<h2 id='getter'>Applies method $ to different classes</h2><span id='topic+getter'></span><span id='topic++24+2CWeightedMedian-method'></span><span id='topic++24+2CIVW-method'></span><span id='topic++24+2CMRMBE-method'></span><span id='topic++24+2CEgger-method'></span><span id='topic++24+2CMRAll-method'></span><span id='topic++24+2CMRInput-method'></span><span id='topic++24+2CMaxLik-method'></span><span id='topic++24+2CMVIVW-method'></span><span id='topic++24+2CMVIVWME-method'></span><span id='topic++24+2CMVEgger-method'></span><span id='topic++24+2CMRMVInput-method'></span><span id='topic++24+2CMRHetPen-method'></span><span id='topic++24+2CMRConMix-method'></span><span id='topic++24+2CMVMedian-method'></span><span id='topic++24+2CMVLasso-method'></span><span id='topic++24+2CMRLasso-method'></span><span id='topic++24+2CDIVW-method'></span><span id='topic++24+2CMRcML-method'></span><span id='topic++24+2CPIVW-method'></span><span id='topic++24+2CMVMRcML-method'></span><span id='topic++24+2CPCGMM-method'></span><span id='topic++24+2CMVPCGMM-method'></span><span id='topic++24+2CMVGMM-method'></span><span id='topic++24+2CCLR-method'></span>

<h3>Description</h3>

<p>Enables slots of objects in this package to be referenced easily.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'WeightedMedian'
x$name

## S4 method for signature 'IVW'
x$name

## S4 method for signature 'MRMBE'
x$name

## S4 method for signature 'Egger'
x$name

## S4 method for signature 'MRAll'
x$name

## S4 method for signature 'MRInput'
x$name

## S4 method for signature 'MaxLik'
x$name

## S4 method for signature 'MVIVW'
x$name

## S4 method for signature 'MVIVWME'
x$name

## S4 method for signature 'MVEgger'
x$name

## S4 method for signature 'MRMVInput'
x$name

## S4 method for signature 'MRHetPen'
x$name

## S4 method for signature 'MRConMix'
x$name

## S4 method for signature 'MVMedian'
x$name

## S4 method for signature 'MVLasso'
x$name

## S4 method for signature 'MRLasso'
x$name

## S4 method for signature 'DIVW'
x$name

## S4 method for signature 'MRcML'
x$name

## S4 method for signature 'PIVW'
x$name

## S4 method for signature 'MVMRcML'
x$name

## S4 method for signature 'PCGMM'
x$name

## S4 method for signature 'MVPCGMM'
x$name

## S4 method for signature 'MVGMM'
x$name

## S4 method for signature 'CLR'
x$name
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getter_+3A_x">x</code></td>
<td>
<p>Object.</p>
</td></tr>
<tr><td><code id="getter_+3A_name">name</code></td>
<td>
<p>Name of slot.</p>
</td></tr>
</table>

<hr>
<h2 id='het.weight'>Heterogeneity-penalized weight function</h2><span id='topic+het.weight'></span>

<h3>Description</h3>

<p>Internal function for calculating heterogeneity-penalized weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>het.weight(prob.valid.inst, bx, by, byse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="het.weight_+3A_prob.valid.inst">prob.valid.inst</code></td>
<td>
<p>Probability (prior) of a genetic variant being a valid instrument.</p>
</td></tr>
<tr><td><code id="het.weight_+3A_bx">bx</code></td>
<td>
<p>Genetic associations with risk factor.</p>
</td></tr>
<tr><td><code id="het.weight_+3A_by">by</code></td>
<td>
<p>Genetic associations with outcome.</p>
</td></tr>
<tr><td><code id="het.weight_+3A_byse">byse</code></td>
<td>
<p>Standard errors of genetic associations with outcome.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Heterogeneity-penalized weight.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>het.weight(0.5, calcium, fastgluc, fastglucse)

</code></pre>

<hr>
<h2 id='invcov_mvmr'>Generate the list of inverse of covariance matrices used in <code>MVMR-cML-DP</code></h2><span id='topic+invcov_mvmr'></span>

<h3>Description</h3>

<p>Generate the list of inverse of covariance matrices used in <code>MVMR-cML-DP</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>invcov_mvmr(se_bx, se_by, rho_mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="invcov_mvmr_+3A_se_bx">se_bx</code></td>
<td>
<p>A m*L matrix of standard errors of SNP-exposure association</p>
</td></tr>
<tr><td><code id="invcov_mvmr_+3A_se_by">se_by</code></td>
<td>
<p>A vector of standard errors of SNP-outcome association</p>
</td></tr>
<tr><td><code id="invcov_mvmr_+3A_rho_mat">rho_mat</code></td>
<td>
<p>The correlation matrix among the L exposures and the outcome</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of inverse of covariance matrices with respect to each genetic variant, retaining the ordering in <code>se_bx</code>
</p>

<hr>
<h2 id='IVW-class'>IVW Class</h2><span id='topic+IVW-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the inverse-variance weighted (IVW) method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Model</code></dt><dd><p>The model used for estimation: random-effects (<code>"random"</code>) or fixed-effect (<code>"fixed"</code>). The default option (<code>"default"</code>) is to use a fixed-effect model when there are three or fewer genetic variants, and a random-effects model when there are four or more. The (multiplicative) random-effects model allows for heterogeneity between the causal estimates targeted by the genetic variants by allowing over-dispersion in the regression model. Under-dispersion is not permitted (in case of under-dispersion, the residual standard error is set to 1, as in a fixed-effect analysis).</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Robust</code></dt><dd><p>Whether robust regression was used in the regression model relating the genetic associations with the outcome and those with the exposure.</p>
</dd>
<dt><code>Penalized</code></dt><dd><p>Whether weights in the regression model were penalized for variants with heterogeneous causal estimates.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal point estimate from the inverse-variance weighted method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>RSE</code></dt><dd><p>The estimated residual standard error from the regression model.</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</dd>
<dt><code>Fstat</code></dt><dd><p>An approximation of the first-stage F statistic for all variants based on the summarized data.</p>
</dd>
</dl>

<hr>
<h2 id='ldlc'>Data on lipid effects on coronary artery disease (uncorrelated variants)</h2><span id='topic+ldlc'></span><span id='topic+hdlc'></span><span id='topic+hdlcse'></span><span id='topic+ldlcse'></span><span id='topic+trig'></span><span id='topic+trigse'></span><span id='topic+chdlodds'></span><span id='topic+chdloddsse'></span><span id='topic+lipid_effect'></span><span id='topic+lipid_other'></span><span id='topic+lipid_eaf'></span>

<h3>Description</h3>

<p>Two sets of example data are included in the package: one illustrating uncorrelated variants, and the other correlated variants. These are the data on uncorrelated variants.
</p>
<p>The variables <code>ldlc</code>, <code>hdlc</code>, <code>trig</code>, and <code>chdlodds</code> are the genetic associations with (respectively) LDL-cholesterol, HDL-cholesterol, triglycerides, and coronary heart disease (CHD) risk for 28 genetic variants reported by Waterworth et al (2010). The respective standard errors of the associations are given as <code>ldlcse</code>, <code>hdlcse</code>, <code>trigse</code>, and <code>chdloddsse</code>.
</p>
<p>These data can be used to test out the various functions in the package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldlc

hdlc

hdlcse

ldlcse

trig

trigse

chdlodds

chdloddsse

lipid_effect

lipid_other

lipid_eaf
</code></pre>


<h3>Format</h3>

<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>
<p>An object of class <code>character</code> of length 28.
</p>
<p>An object of class <code>character</code> of length 28.
</p>
<p>An object of class <code>numeric</code> of length 28.
</p>


<h3>References</h3>

<p>Dawn Waterworth, Sally Ricketts, ..., Manj Sandhu: Genetic variants influencing circulating lipid levels and risk of coronary artery disease. Arterioscler Thromb Vasc Biol 2010; 30:2264-227. doi: 10.1161/atvbaha.109.201020.
</p>

<hr>
<h2 id='loglikelihood'>Calculates log-likelihood with uncorrelated variants in two-sample setting (no correlation from sample overlap)</h2><span id='topic+loglikelihood'></span>

<h3>Description</h3>

<p>Internal function for calculating log-likelihood for for maximum-likelihood method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loglikelihood(param, x, sigmax, y, sigmay)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loglikelihood_+3A_param">param</code></td>
<td>
<p>Parameters in the model.</p>
</td></tr>
<tr><td><code id="loglikelihood_+3A_x">x</code></td>
<td>
<p>Genetic associations with the exposure.</p>
</td></tr>
<tr><td><code id="loglikelihood_+3A_sigmax">sigmax</code></td>
<td>
<p>Standard errors of associations with the exposure.</p>
</td></tr>
<tr><td><code id="loglikelihood_+3A_y">y</code></td>
<td>
<p>Genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="loglikelihood_+3A_sigmay">sigmay</code></td>
<td>
<p>Standard errors of genetic associations with the outcome.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Log-likelihood.
</p>

<hr>
<h2 id='loglikelihoodcorrel'>Calculates log-likelihood with correlated variants in two-sample setting (no correlation from sample overlap)</h2><span id='topic+loglikelihoodcorrel'></span>

<h3>Description</h3>

<p>Internal function for calculating log-likelihood for for maximum-likelihood method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loglikelihoodcorrel(param, x, Taux, y, Tauy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loglikelihoodcorrel_+3A_param">param</code></td>
<td>
<p>Parameters in the model.</p>
</td></tr>
<tr><td><code id="loglikelihoodcorrel_+3A_x">x</code></td>
<td>
<p>Genetic associations with the exposure.</p>
</td></tr>
<tr><td><code id="loglikelihoodcorrel_+3A_taux">Taux</code></td>
<td>
<p>Precision matrix for associations with the exposure.</p>
</td></tr>
<tr><td><code id="loglikelihoodcorrel_+3A_y">y</code></td>
<td>
<p>Genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="loglikelihoodcorrel_+3A_tauy">Tauy</code></td>
<td>
<p>Precision matrix for associations with the outcome.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Log-likelihood.
</p>

<hr>
<h2 id='loglikelihoodrhocorrel'>Calculates log-likelihood with correlation from sample overlap</h2><span id='topic+loglikelihoodrhocorrel'></span>

<h3>Description</h3>

<p>Internal function for calculating log-likelihood for for maximum-likelihood method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>loglikelihoodrhocorrel(param, x, y, Tauxy)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="loglikelihoodrhocorrel_+3A_param">param</code></td>
<td>
<p>Parameters in the model.</p>
</td></tr>
<tr><td><code id="loglikelihoodrhocorrel_+3A_x">x</code></td>
<td>
<p>Genetic associations with the exposure.</p>
</td></tr>
<tr><td><code id="loglikelihoodrhocorrel_+3A_y">y</code></td>
<td>
<p>Genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="loglikelihoodrhocorrel_+3A_tauxy">Tauxy</code></td>
<td>
<p>Precision matrix for associations with the exposure and outcome.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Log-likelihood.
</p>

<hr>
<h2 id='MaxLik-class'>MaxLik Class</h2><span id='topic+MaxLik-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the maximum-likelihood method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Model</code></dt><dd><p>The model used for estimation: fixed-effect (<code>"fixed"</code>) or random-effects (<code>"random"</code>).</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Psi</code></dt><dd><p>The correlations between genetic associations with the exposure and with the outcome.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal point estimate from the inverse-variance weighted method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>RSE</code></dt><dd><p>The estimated residual standard error from the regression model.</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (likelihood ratio statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</dd>
</dl>

<hr>
<h2 id='mbe_boot'>Mode-based estimate (Hartwig) bootstrap function</h2><span id='topic+mbe_boot'></span>

<h3>Description</h3>

<p>Internal function for calculating standard error of mode-based estimate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mbe_boot(BetaIV.in, seBetaIV.in, weighting.in, iterations.in, phi.in)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mbe_boot_+3A_betaiv.in">BetaIV.in</code></td>
<td>
<p>Ratio causal estimates for each genetic variant considered individually.</p>
</td></tr>
<tr><td><code id="mbe_boot_+3A_sebetaiv.in">seBetaIV.in</code></td>
<td>
<p>Standard errors of ratio causal estimates.</p>
</td></tr>
<tr><td><code id="mbe_boot_+3A_iterations.in">iterations.in</code></td>
<td>
<p>Number of bootstrap iterations.</p>
</td></tr>
<tr><td><code id="mbe_boot_+3A_phi.in">phi.in</code></td>
<td>
<p>Bandwidth multiplication factor.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Bootstrapped mode-based estimates.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mbe_boot(BetaIV.in = chdlodds/ldlc, seBetaIV.in = abs(chdloddsse/ldlc),
   weighting.in = "simple", iterations.in = 100)

</code></pre>

<hr>
<h2 id='mbe_est'>Mode-based estimate (Hartwig) estimation function</h2><span id='topic+mbe_est'></span>

<h3>Description</h3>

<p>Internal function for calculating mode-based estimate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mbe_est(BetaIV.in, seBetaIV.in, phi.in)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mbe_est_+3A_betaiv.in">BetaIV.in</code></td>
<td>
<p>Ratio causal estimates for each genetic variant considered individually.</p>
</td></tr>
<tr><td><code id="mbe_est_+3A_sebetaiv.in">seBetaIV.in</code></td>
<td>
<p>Standard errors of ratio causal estimates.</p>
</td></tr>
<tr><td><code id="mbe_est_+3A_phi.in">phi.in</code></td>
<td>
<p>Bandwidth multiplication factor.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Mode-based estimate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mbe_est(BetaIV.in = chdlodds/ldlc, seBetaIV.in = abs(chdloddsse/ldlc), phi.in = 1)

</code></pre>

<hr>
<h2 id='model.prior'>Prior weight function</h2><span id='topic+model.prior'></span>

<h3>Description</h3>

<p>Internal function for prior weights.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>model.prior(model.size, N.obs, prob.valid.inst)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="model.prior_+3A_model.size">model.size</code></td>
<td>
<p>Size of subset.</p>
</td></tr>
<tr><td><code id="model.prior_+3A_n.obs">N.obs</code></td>
<td>
<p>Number of genetic variants.</p>
</td></tr>
<tr><td><code id="model.prior_+3A_prob.valid.inst">prob.valid.inst</code></td>
<td>
<p>Probability (prior) of a genetic variant being a valid instrument.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Prior weight.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>model.prior(8, 15, 0.5)

</code></pre>

<hr>
<h2 id='mr_allmethods'>Mendelian randomization estimation using all methods</h2><span id='topic+mr_allmethods'></span><span id='topic+mr_allmethods+2CMRInput-method'></span>

<h3>Description</h3>

<p>The function <code>mr_allmethods</code> implements Mendelian randomization analyses using summarized data to calculate estimates (as well as standard
errors and confidence interval limits) for all the methods included in the package (or alternatively for the group of methods chosen).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_allmethods(object, method = "all", ...)

## S4 method for signature 'MRInput'
mr_allmethods(object, method = "all", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_allmethods_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_allmethods_+3A_method">method</code></td>
<td>
<p>Which estimation method should be included in the calculation. By default, all estimates are computed (<code>"all"</code>), but one can choose to show only the results of median-based, inverse-variance weighted, or MR-Egger methods separately through specifying <code>"median"</code>, <code>"ivw"</code>, <code>"egger"</code>, or <code>"main"</code> (gives main results only, that is simple and weighted median, IVW, and MR-Egger).</p>
</td></tr>
<tr><td><code id="mr_allmethods_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See <code>mr_median</code>, <code>mr_egger</code>, and <code>mr_ivw</code> for details of how each of the methods is implemented.
</p>


<h3>Value</h3>

<p>An object of type <code>MRAll</code> with the following slots :
</p>
<table>
<tr><td><code>Data</code></td>
<td>
<p>The MRInput object used to calculate the various values.</p>
</td></tr>
<tr><td><code>Values</code></td>
<td>
<p>A data.frame containing the various estimates.</p>
</td></tr>
<tr><td><code>Method</code></td>
<td>
<p>The choice of methods estimated (default is <code>"all"</code>).</p>
</td></tr>
</table>


<h3>References</h3>

<p>See <code>mr_median</code>, <code>mr_egger</code>, and <code>mr_ivw</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_allmethods(mr_input(bx = ldlc, bxse = ldlcse,
  by = chdlodds, byse = chdloddsse), method="main", iterations = 100)
  # iterations is set to 100 to reduce runtime for the mr_median method,
  # at least 10000 iterations are recommended in practice

</code></pre>

<hr>
<h2 id='mr_clr'>Conditional likelihood ratio (CLR) method</h2><span id='topic+mr_clr'></span><span id='topic+mr_clr+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_clr</code> function calculates confidence intervals based on inverting the conditional likelihood ratio and other identification-robust tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_clr(object, nx, ny, alpha = 0.05, CIMin = -10, CIMax = 10, CIStep = 0.01)

## S4 method for signature 'MRInput'
mr_clr(object, nx, ny, alpha = 0.05, CIMin = -10, CIMax = 10, CIStep = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_clr_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_clr_+3A_nx">nx</code></td>
<td>
<p>The sample size used to compute genetic associations with the exposure.</p>
</td></tr>
<tr><td><code id="mr_clr_+3A_ny">ny</code></td>
<td>
<p>The sample size used to compute genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="mr_clr_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_clr_+3A_cimin">CIMin</code></td>
<td>
<p>The smallest value to use in the search to find the confidence interval (default is -10).</p>
</td></tr>
<tr><td><code id="mr_clr_+3A_cimax">CIMax</code></td>
<td>
<p>The largest value to use in the search to find the confidence interval (default is +10).</p>
</td></tr>
<tr><td><code id="mr_clr_+3A_cistep">CIStep</code></td>
<td>
<p>The step size to use in the search to find the confidence interval (default is 0.01). Using a lower value (such as 0.001) will give more precise confidence intervals, but increase run time.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In weak instrument settings, usual inference based on point estimates and standard errors may not be accurate. This method calculates confidence intervals based on inverting identification-robust tests proposed in Wang and Kang (2021, Biometrics) that provide valid inferences regardless of instrument strength.
</p>
<p>This includes conditional likelihood ratio (CLR), Kleibergen (K), and Anderson and Rubin (AR) tests.
</p>
<p>Evidence from the econometrics literature suggests that CLR inference is the best option in terms of power under a wide range of settings.
</p>
<p>Please note that these methods do not provide point estimates, only confidence intervals. While most examples provide a confidence interval that is a single range of values, in some cases the confidence interval may comprise multiple ranges of values. In other cases, a valid confidence interval may not exist.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>CLR</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string with the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>ARlower</code></td>
<td>
<p>The lower bounds of the causal estimate based on inverting Anderson and Rubin's test.</p>
</td></tr>
<tr><td><code>ARupper</code></td>
<td>
<p>The upper bounds of the causal estimate based on inverting Anderson and Rubin's test.</p>
</td></tr>
<tr><td><code>Klower</code></td>
<td>
<p>The lower bounds of the causal estimate based on inverting Kleibergen's test.</p>
</td></tr>
<tr><td><code>Kupper</code></td>
<td>
<p>The upper bounds of the causal estimate based on inverting Kleibergen's test.</p>
</td></tr>
<tr><td><code>CLRlower</code></td>
<td>
<p>The lower bounds of the causal estimate based on inverting Moreira's conditional likelihood ratio test.</p>
</td></tr>
<tr><td><code>CLRupper</code></td>
<td>
<p>The upper bounds of the causal estimate based on inverting Moreira's conditional likelihood ratio test.</p>
</td></tr>
<tr><td><code>CIMin</code></td>
<td>
<p>The smallest value used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>CIMax</code></td>
<td>
<p>The largest value used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>CIStep</code></td>
<td>
<p>The step size used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Description of the CLR method: &quot;Weak-instrument robust tests in two-sample summary-data Mendelian randomization&quot;, S. Wang and H. Kang, Biometrics, 2021.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_clr(mr_input(bx = calcium, bxse = calciumse,
   by = fastgluc, byse = fastglucse, correl = calc.rho), nx=6351, ny=133010)

</code></pre>

<hr>
<h2 id='mr_cML'>Constrained maximum likelihood (cML) method</h2><span id='topic+mr_cML'></span><span id='topic+mr_cML+2CMRInput-method'></span>

<h3>Description</h3>

<p>Constrained maximum likelihood (cML) based Mendelian
Randomization method robust to both 
correlated and uncorrelated pleiotropy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_cML(
  object,
  MA = TRUE,
  DP = TRUE,
  K_vec = 0:(length(object@betaX) - 2),
  random_start = 0,
  num_pert = 200,
  random_start_pert = 0,
  maxit = 100,
  random_seed = 314,
  n,
  Alpha = 0.05
)

## S4 method for signature 'MRInput'
mr_cML(
  object,
  MA = TRUE,
  DP = TRUE,
  K_vec = 0:(length(object@betaX) - 2),
  random_start = 0,
  num_pert = 200,
  random_start_pert = 0,
  maxit = 100,
  random_seed = 314,
  n,
  Alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_cML_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_ma">MA</code></td>
<td>
<p>Whether model average is applied or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_dp">DP</code></td>
<td>
<p>Whether data perturbation is applied or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_k_vec">K_vec</code></td>
<td>
<p>Set of candidate K's, the constraint parameter 
representing number of invalid IVs. Default is from 0 to (#IV - 2).</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_random_start">random_start</code></td>
<td>
<p>Number of random starting points for cML, default is 0.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_num_pert">num_pert</code></td>
<td>
<p>Number of perturbation when DP is TRUE, default is 200.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_random_start_pert">random_start_pert</code></td>
<td>
<p>Number of random start points for 
cML with data perturbation, default is 0.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations for each optimization. 
Default is 100.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_random_seed">random_seed</code></td>
<td>
<p>Random seed, default is 314. When <code>random_seed=NULL</code>, no random seed 
will be used and the results may not be reproducible.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_n">n</code></td>
<td>
<p>Sample size. When sample sizes of GWAS for exposure and outcome are different,
and/or when sample sizes of different SNPs are different, 
the smallest sample size is recommended to get conservative result and avoid type-I error.
See reference for more discussions.</p>
</td></tr>
<tr><td><code id="mr_cML_+3A_alpha">Alpha</code></td>
<td>
<p>Significance level for the confidence interval for estimate, default is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The MRcML method selects invalid IVs with correlated
and/or uncorrelated peliotropic effects using constrained maximum
likelihood. <code>cML-BIC</code> gives results of the selected model with 
original data, while <code>cML-MA-BIC</code> averages over all candidate models.
<code>cML-BIC-DP</code> and <code>cML-MA-BIC-DP</code> are the versions with 
data-perturbation to account for selection uncertainty when 
many invalid IVs have weak pleiotropic effects. 
</p>
<p>When DP is performed,
two goodness-of-fit (GOF) tests are developed to check whether the 
model-based and DP- based variance estimates converge to the same estimate.
Small p-values of GOF tests indicate selection uncertainty is not ignorable, and 
results from DP is more reliable. See reference for more details.
</p>
<p>As the constrained maximum likelihood function is non-convex, multiple
random starting points could be used to find a global minimum. For some 
starting points the algorithm may not converge and a warning message will 
be prompted, typically this will not affect the results.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MRcML</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>Estimate of theta.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>Standard error of estimate.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>p-value of estimate.</p>
</td></tr>
<tr><td><code>BIC_invalid</code></td>
<td>
<p>Set of selected invalid IVs if cML-BIC is performed, i.e. without MA or DP.</p>
</td></tr>
<tr><td><code>GOF1_p</code></td>
<td>
<p>p-value of the first goodness-of-fit test.</p>
</td></tr>
<tr><td><code>GOF2_p</code></td>
<td>
<p>p-value of the second goodness-of-fit test.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of SNPs that were used in the calculation.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>Significance level for the confidence interval for estimate, default is 0.05.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>Lower bound of the confidence interval for estimate.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>Upper bound of the confidence interval for estimate.</p>
</td></tr>
<tr><td><code>MA</code></td>
<td>
<p>Indicator of whether model average is applied.</p>
</td></tr>
<tr><td><code>DP</code></td>
<td>
<p>Indicator of whether data perturbation is applied.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xue, H., Shen, X., &amp; Pan, W. (2021). 
Constrained maximum likelihood-based Mendelian randomization 
robust to both correlated and uncorrelated pleiotropic effects. 
The American Journal of Human Genetics, 108(7), 1251-1269.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perform cML-MA-BIC-DP:
mr_cML(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds,
byse = chdloddsse), num_pert=5, MA = TRUE, DP = TRUE, n = 17723)
# num_pert is set to 5 to reduce computational time
# the default value of 200 is recommended in practice

# Perform cML-BIC-DP:
mr_cML(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds,
byse = chdloddsse), MA = TRUE, DP = FALSE,, n = 17723)
   

</code></pre>

<hr>
<h2 id='mr_conmix'>Contamination mixture method</h2><span id='topic+mr_conmix'></span><span id='topic+mr_conmix+2CMRInput-method'></span>

<h3>Description</h3>

<p>Contamination mixture method for robust and efficient estimation under the 'plurality valid' assumption.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_conmix(object, psi = 0, CIMin = NA, CIMax = NA, CIStep = 0.01, alpha = 0.05)

## S4 method for signature 'MRInput'
mr_conmix(object, psi = 0, CIMin = NA, CIMax = NA, CIStep = 0.01, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_conmix_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_conmix_+3A_psi">psi</code></td>
<td>
<p>The value of the standard deviation of the distribution of invalid estimands (default value is 0, corresponding to 1.5 times the standard deviation of the ratio estimates).</p>
</td></tr>
<tr><td><code id="mr_conmix_+3A_cimin">CIMin</code></td>
<td>
<p>The smallest value to use in the search to find the confidence interval. The default value is NA, which means that the method uses the smallest value of the lower bound of the 95% confidence interval for the variant-specific ratio estimates as the smallest value.</p>
</td></tr>
<tr><td><code id="mr_conmix_+3A_cimax">CIMax</code></td>
<td>
<p>The largest value to use in the search to find the confidence interval. The default value is NA, which means that the method uses the greatest value of the upper bound of the 95% confidence interval for the variant-specific ratio estimates as the largest value.</p>
</td></tr>
<tr><td><code id="mr_conmix_+3A_cistep">CIStep</code></td>
<td>
<p>The step size to use in the search to find the confidence interval (default is 0.01). The confidence interval is determined by a grid search algorithm. Using the default settings, we calculate the likelihood at all values from -1 to +1 increasing in units of 0.01. If this range is too large or the step size is too small, then the method will take a long time to run.</p>
</td></tr>
<tr><td><code id="mr_conmix_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The contamination mixture method is implemented by constructing a likelihood function based on the variant-specific causal estimates. If a genetic variant is a valid instrument, then its causal estimate will be normally distributed about the true value of the causal effect. If a genetic variant is not a valid instrument, then its causal estimate will be normally distributed about some other value. We assume that the values estimated by invalid instruments are normally distributed about zero with a large standard deviation. This enables a likelihood function to be specified that is a product of two-component mixture distributions, with one mixture distribution for each variant. The computational time for maximizing this likelihood directly is exponential in the number of genetic variants. We use a profile likelihood approach to reduce the computational complexity to be linear in the number of variants.
</p>
<p>We consider different values of the causal effect in turn. For each value, we calculate the contribution to the likelihood for each genetic variant as a valid instrument and as an invalid instrument. If the contribution to the likelihood as a valid instrument is greater, then we take the variant's contribution as a valid instrument; if less, then its contribution is taken as an invalid instrument. This gives us the configuration of valid and invalid instruments that maximizes the likelihood for the given value of the causal effect. This is a profile likelihood, a one-dimensional function of the causal effect. The point estimate is then taken as the value of the causal effect that maximizes the profile likelihood.
</p>
<p>Confidence intervals are evaluated by calculating the log-likelihood function, and finding all points within a given vertical distance of the maximum of the log-likelihood function (which is the causal estimate). As such, if the log-likelihood function is multimodal, then the confidence interval may include multiple disjoint ranges. This may indicate the presence of multiple causal mechanisms by which the exposure may influence the outcome with different magnitudes of causal effect. As the confidence interval is determined by a grid search, care must be taken when chosing the minimum (<code>CIMin</code>) and maximum (<code>CIMax</code>) values in the search, as well as the step size (<code>CIStep</code>). The default values will not be suitable for all applications.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MRConMix</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Psi</code></td>
<td>
<p>The value of the standard deviation parameter.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>CIRange</code></td>
<td>
<p>The range of values in the confidence interval based on a grid search between the minimum and maximum values for the causal effect provided.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower limit of the confidence interval. If the confidence interval contains multiple ranges, then lower limits of all ranges will be reported.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper limit of the confidence interval. If the confidence interval contains multiple ranges, then upper limits of all ranges will be reported.</p>
</td></tr>
<tr><td><code>CIMin</code></td>
<td>
<p>The smallest value used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>CIMax</code></td>
<td>
<p>The largest value used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>CIStep</code></td>
<td>
<p>The step size used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the estimate calculated using the likelihood function and a chi-squared distribution.</p>
</td></tr>
<tr><td><code>Valid</code></td>
<td>
<p>The numbers of genetic variants that were considered valid instruments at the causal estimate.</p>
</td></tr>
<tr><td><code>ValidSNPs</code></td>
<td>
<p>The names of genetic variants that were considered valid instruments at the causal estimate.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Stephen Burgess, Christopher N Foley, Elias Allara, Joanna Howson. A robust and efficient method for Mendelian randomization with hundreds of genetic variants: unravelling mechanisms linking HDL-cholesterol and coronary heart disease. Nat Comms 2020. doi: 10.1038/s41467-019-14156-4.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_conmix(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds,
   byse = chdloddsse), psi = 3, CIMin = -1, CIMax = 5, CIStep = 0.01)

</code></pre>

<hr>
<h2 id='mr_divw'>Debiased inverse-variance weighted method</h2><span id='topic+mr_divw'></span><span id='topic+mr_divw+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_divw</code> function implements the debiased inverse-variance weighted method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_divw(object, over.dispersion = TRUE, alpha = 0.05, diagnostics = FALSE)

## S4 method for signature 'MRInput'
mr_divw(object, over.dispersion = TRUE, alpha = 0.05, diagnostics = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_divw_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_divw_+3A_over.dispersion">over.dispersion</code></td>
<td>
<p>Should the method consider overdispersion (balanced horizontal pleiotropy)? Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_divw_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_divw_+3A_diagnostics">diagnostics</code></td>
<td>
<p>Should the function returns the q-q plot for assumption diagnosis. Default is FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The debiased inverse-variance weighted method (dIVW) removes the weak instrument bias of the IVW method and is more robust under many weak instruments.
</p>


<h3>Value</h3>

<p>The output from the function is a <code>DIVW</code> object containing:
</p>
<table>
<tr><td><code>Over.dispersion</code></td>
<td>
<p><code>TRUE</code> if the method has considered balanced horizontal pleiotropy, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>Standard error of the causal estimate calculated using bootstrapping.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound for the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound for the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the estimate (calculated using <code>Estimate/StdError</code> as per a Wald test) using a normal distribution.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>Condition</code></td>
<td>
<p>A measure (average F-statistic -1)*sqrt(# snps) that needs to be large for reliable asymptotic approximation based on the dIVW estimator. It is recommended to be greater than 20.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ting Ye, Jun Shao, Hyunseung Kang (2021). Debiased Inverse-Variance Weighted Estimator in
Two-Sample Summary-Data Mendelian Randomization. The Annals of Statistics, 49(4), 2079-2100. Also available at <a href="https://arxiv.org/abs/1911.09802">https://arxiv.org/abs/1911.09802</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_divw(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_egger'>MR-Egger method</h2><span id='topic+mr_egger'></span><span id='topic+mr_egger+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_egger</code> function implements the MR-Egger method introduced by Bowden et al (2015).
</p>
<p>This method provides: 1) a test of the for directional pleiotropy (the MR-Egger intercept test), 2) a test for a
causal effect, and 3) an estimate of the causal effect.
If the intercept term differs from zero, then the genetic variants are not all valid instrumental variables and
the standard (inverse-variance weighted) estimate is biased. If the InSIDE (Instrument Strength Independent of Direct Effect) assumption holds, then the MR-Egger slope parameter provides a test for a causal effect, and a consistent estimate of the causal effect even if the intercept differs from zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_egger(
  object,
  robust = FALSE,
  penalized = FALSE,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05,
  ...
)

## S4 method for signature 'MRInput'
mr_egger(
  object,
  robust = FALSE,
  penalized = FALSE,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_egger_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_egger_+3A_robust">robust</code></td>
<td>
<p>Indicates whether robust regression using the <code>lmrob()</code> function from the package <code>robustbase</code> should be used in the method.</p>
</td></tr>
<tr><td><code id="mr_egger_+3A_penalized">penalized</code></td>
<td>
<p>Indicates whether a penalty should be applied to the weights to downweight the contribution of genetic variants with outlying ratio estimates to the analysis.</p>
</td></tr>
<tr><td><code id="mr_egger_+3A_correl">correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for. The matrix of correlations between must be provided: the elements of this matrix are the correlations between the individual variants (diagonal elements are 1). If a correlation is specified, then the values of <code>"robust"</code> and <code>"penalized"</code> are taken as <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mr_egger_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals, can be <code>"normal"</code> (the default option) or <code>"t-dist"</code>. If the distribution is <code>"t-dist"</code>, then a t-distribution is used in case of over-dispersion. In case of under-dispersion, the confidence interval is the wider of that using the estimated residual standard error and a t-distribution, or that using a residual standard error of 1 and a normal distribution. This ensures that under-dispersion is not &quot;doubly penalized&quot; by setting the residual standard error to 1 and using a t-distribution, and also that the random-effects analysis is no more precise than a fixed-effect analysis would be.</p>
</td></tr>
<tr><td><code id="mr_egger_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_egger_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the regression method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The causal estimate is obtained by regression of the associations with the outcome on the associations with the risk factor, with weights being the inverse-variances of the associations with the outcome. The intercept is estimated (in contrast with the inverse-variance weighted method, where the intercept is set to zero).
</p>
<p>As part of the analysis, the genetic variants are orientated so that all of the associations with the risk factor are positive (and signs of associations with the outcome are changed to keep the orientation consistent if required). Re-orientation of the genetic variants is performed automatically as part of the function.
</p>
<p>The MR-Egger model uses a random-effects model (<code>"random"</code>); a fixed-effect model does not make sense as pleiotropy leads to heterogeneity between the causal estimates targeted by the genetic variants. The (multiplicative) random-effects model allows over-dispersion in the regression model. Under-dispersion is not permitted (in case of under-dispersion, the residual standard error is set to 1).
</p>


<h3>Value</h3>

<p>The output of the function is an <code>Egger</code> object containing:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"random"</code>).</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>Robust</code></td>
<td>
<p><code>TRUE</code> if robust estimate has been calculated, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Penalized</code></td>
<td>
<p><code>TRUE</code> if weights have been penalized, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate (slope coefficient).</p>
</td></tr>
<tr><td><code>StdError.Est</code></td>
<td>
<p>Standard error of the causal estimate.</p>
</td></tr>
<tr><td><code>Pvalue.Est</code></td>
<td>
<p>The p-value associated with the estimate (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>CILower.Est</code></td>
<td>
<p>The lower bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper.Est</code></td>
<td>
<p>The upper bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>The value of the intercept estimate.</p>
</td></tr>
<tr><td><code>StdError.Int</code></td>
<td>
<p>Standard error of the intercept estimate.</p>
</td></tr>
<tr><td><code>Pvalue.Int</code></td>
<td>
<p>The p-value associated with the intercept.</p>
</td></tr>
<tr><td><code>CILower.Int</code></td>
<td>
<p>The lower bound of the intercept based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper.Int</code></td>
<td>
<p>The upper bound of the intercept based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals (same as <code>alpha</code> above).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>Causal.pval</code></td>
<td>
<p>The p-value for the MR-Egger causal estimate.</p>
</td></tr>
<tr><td><code>Pleio.pval</code></td>
<td>
<p>The p-value for the MR-Egger intercept test (a low p-value suggests either directional pleiotropy or failure of the InSIDE assumption, and indicates that the IVW estimate is biased).</p>
</td></tr>
<tr><td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that the regression model (including an intercept) fits the regression model with no additional variability. Rejection of the null hypothesis is expected if genetic variants are pleiotropic, and doesn't mean that the MR-Egger analysis or the InSIDE assumption is invalid.</p>
</td></tr>
<tr><td><code>I.sq</code></td>
<td>
<p>A measure of heterogeneity between the genetic associations with the exposure (see Bowden IJE 2016). Low values of <code>I.sq</code> relate both to large differences in precision between MR-Egger and IVW estimates, and to more weak instrument bias (in a two-sample setting, this is attenuation of MR-Egger estimate towards the null).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Jack Bowden, George Davey Smith, Stephen Burgess. Mendelian randomization with invalid instruments: effect estimation and bias detection through Egger regression. International Journal of Epidemiology 2015; 44:512&ndash;525. doi: 10.1093/ije/dyv080.
</p>
<p>Confidence intervals, and robust and penalized weights: Stephen Burgess, Jack Bowden, Frank Dudbridge, Simon G Thompson. Robust instrumental variable methods using multiple candidate instruments with application to Mendelian randomization. arXiv 2016; 1606.03729.
</p>
<p>I-squared statistic: Jack Bowden and others. Assessing the suitability of summary data for Mendelian randomization analyses using MR-Egger regression: The role of the I2 statistic. Int J Epidemiol 2016 (to appear).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_egger(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))
mr_egger(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  robust = TRUE)
mr_egger(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  penalized = TRUE)
mr_egger(mr_input(calcium, calciumse, fastgluc, fastglucse, corr=calc.rho))
  ## correlated variants

</code></pre>

<hr>
<h2 id='mr_forest'>Draw a forest plot of causal estimates</h2><span id='topic+mr_forest'></span><span id='topic+mr_forest+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_forest</code> function draws a forest plot of causal estimates. The default option plots the variant-specific causal estimates (<code>by/bx</code>) and the estimate from the <code>mr_ivw</code> function using default settings (assuming variants are uncorrelated, random-effects for 4+ variants). Options allow users to plot estimates from a variety of different methods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_forest(
  object,
  alpha = 0.05,
  snp_estimates = TRUE,
  methods = "ivw",
  ordered = FALSE
)

## S4 method for signature 'MRInput'
mr_forest(
  object,
  alpha = 0.05,
  snp_estimates = TRUE,
  methods = "ivw",
  ordered = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_forest_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_forest_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05, corresponding to 95% confidence intervals.</p>
</td></tr>
<tr><td><code id="mr_forest_+3A_snp_estimates">snp_estimates</code></td>
<td>
<p>Whether to plot the variant-specific estimates. Defaults to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="mr_forest_+3A_methods">methods</code></td>
<td>
<p>Takes a string of computation methods used to calculate estimates. Defaults to <code>"ivw"</code>. Options are: <code>"median"</code> (simple median estimate), <code>"wmedian"</code> (weighted median estimate), <code>"egger"</code> (MR-Egger estimate), <code>"mbe"</code> (mode-based estimate), <code>"conmix"</code> (contamination mixture estimate), and <code>"maxlik"</code> (maximum likelihood estimate).</p>
</td></tr>
<tr><td><code id="mr_forest_+3A_ordered">ordered</code></td>
<td>
<p>Determines by whether to arrange the variant-specific estimates in ascending order. Defaults to <code>FALSE</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As the function produces a <code>ggplot</code> object, graphical parameters can be changed by adding commands from the <code>ggplot2</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_forest(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  alpha = 0.01, ordered = TRUE)
mr_forest(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  methods = c("ivw", "wmedian", "egger"), snp_estimates = FALSE)
forest = mr_forest(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))
# how to change x-axis limits
# library(ggplot2)
# forest2 = forest + coord_cartesian(xlim=c(-5,5))
# forest2

</code></pre>

<hr>
<h2 id='mr_funnel'>Draw a funnel plot of variant-specific estimates</h2><span id='topic+mr_funnel'></span><span id='topic+mr_funnel+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_funnel</code> function draws a funnel plot of variant-specific causal estimates. Estimates (<code>by/bx</code>) are plotted against the precision of the estimates (<code>abs(bx)/byse</code>). Precision is the reciprocal of standard error. A vertical dashed line is plotted at the estimate from the <code>mr_ivw</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_funnel(object, CI = TRUE)

## S4 method for signature 'MRInput'
mr_funnel(object, CI = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_funnel_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_funnel_+3A_ci">CI</code></td>
<td>
<p>A <code>logical</code> variable dicating as to whether to plot the confidence interval associated with each point. Default value is TRUE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As the function produces a <code>ggplot</code> object, graphical parameters can be changed by adding commands from the <code>ggplot2</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_funnel(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_hetpen'>Heterogeneity-penalized method</h2><span id='topic+mr_hetpen'></span><span id='topic+mr_hetpen+2CMRInput-method'></span>

<h3>Description</h3>

<p>Heterogeneity-penalized model-averaging method for efficient modal-based estimation.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_hetpen(
  object,
  prior = 0.5,
  CIMin = -1,
  CIMax = 1,
  CIStep = 0.001,
  alpha = 0.05
)

## S4 method for signature 'MRInput'
mr_hetpen(
  object,
  prior = 0.5,
  CIMin = -1,
  CIMax = 1,
  CIStep = 0.001,
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_hetpen_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_hetpen_+3A_prior">prior</code></td>
<td>
<p>The prior probability of a genetic variant being a valid instrument (default is 0.5).</p>
</td></tr>
<tr><td><code id="mr_hetpen_+3A_cimin">CIMin</code></td>
<td>
<p>The smallest value to use in the search to find the confidence interval (default is -1).</p>
</td></tr>
<tr><td><code id="mr_hetpen_+3A_cimax">CIMax</code></td>
<td>
<p>The largest value to use in the search to find the confidence interval (default is +1).</p>
</td></tr>
<tr><td><code id="mr_hetpen_+3A_cistep">CIStep</code></td>
<td>
<p>The step size to use in the search to find the confidence interval (default is 0.001). The confidence interval is determined by a grid search algorithm. Using the default settings, we calculate the likelihood at all values from -1 to +1 increasing in units of 0.001. If this range is too large or the step size is too small, then the grid search algorithm will take a long time to converge.</p>
</td></tr>
<tr><td><code id="mr_hetpen_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This method was developed as a more efficient version of the mode-based estimation method of Hartwig et al. It proceeds by evaluating weights for all subsets of genetic variants (excluding the null set and singletons). Subsets receive greater weight if they include more variants, but are severely downweighted if the variants in the subset have heterogeneous causal estimates. As such, the method will identify the subset with the largest number (by weight) of variants having similar causal estimates.
</p>
<p>Confidence intervals are evaluated by calculating a log-likelihood function, and finding all points within a given vertical distance of the maximum of the log-likelihood function (which is the causal estimate). As such, if the log-likelihood function is multimodal, then the confidence interval may include multiple disjoint ranges. This may indicate the presence of multiple causal mechanisms by which the exposure may influence the outcome with different magnitudes of causal effect. As the confidence interval is determined by a grid search, care must be taken when chosing the minimum (<code>CIMin</code>) and maximum (<code>CIMax</code>) values in the search, as well as the step size (<code>CIStep</code>). The default values will not be suitable for all applications.
</p>
<p>The method should give consistent estimates as the sample size increases if a weighted plurality of the genetic variants are valid instruments. This means that the largest group of variants with the same causal estimate in the asymptotic limit are the valid instruments.
</p>
<p>The current implementation of the method evaluates a weight and an estimate for each of the subsets of genetic variants. This means that the method complexity doubles for each additional genetic variant included in the analysis. Currently, the method provides a warning message when used with 25+ variants, and fails to run with 30+.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MRHetPen</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Prior</code></td>
<td>
<p>The value of the bandwidth factor.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>CIRange</code></td>
<td>
<p>The range of values in the confidence interval based on a grid search between the minimum and maximum values for the causal effect provided.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower limit of the confidence interval. If the confidence interval contains multiple ranges, then lower limits of all ranges will be reported.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper limit of the confidence interval. If the confidence interval contains multiple ranges, then upper limits of all ranges will be reported.</p>
</td></tr>
<tr><td><code>CIMin</code></td>
<td>
<p>The smallest value used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>CIMax</code></td>
<td>
<p>The largest value used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>CIStep</code></td>
<td>
<p>The step size used in the search to find the confidence interval.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Stephen Burgess, Verena Zuber, Apostolos Gkatzionis, Christopher N Foley. Improving on a modal-based estimation method: model averaging for consistent and efficient estimation in Mendelian randomization when a plurality of candidate instruments are valid. bioRxiv 2017. doi: 10.1101/175372.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_hetpen(mr_input(bx = ldlc[1:10], bxse = ldlcse[1:10], by = chdlodds[1:10],
   byse = chdloddsse[1:10]), CIMin = -1, CIMax = 5, CIStep = 0.01)

</code></pre>

<hr>
<h2 id='mr_input'>Inputting and formatting data for use in causal estimation</h2><span id='topic+mr_input'></span>

<h3>Description</h3>

<p>The <code>mr_input</code> function is required for inputting and formatting data for use in any of the estimation functions provided in this package. The <code>MRInput</code> class outputted by the function can also be viewed graphically using the <code>mr_plot</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_input(
  bx = 0,
  bxse = 0,
  by = 0,
  byse = 0,
  correlation = matrix(),
  exposure = "exposure",
  outcome = "outcome",
  snps = "snp",
  effect_allele = NA,
  other_allele = NA,
  eaf = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_input_+3A_bx">bx</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the first variable (often referred to as the exposure, risk factor, or modifiable phenotype).</p>
</td></tr>
<tr><td><code id="mr_input_+3A_bxse">bxse</code></td>
<td>
<p>The standard errors associated with the beta-coefficients <code>bx</code>.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_by">by</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the second variable (often referred to as the outcome). For a disease outcome, the beta coefficients are log odds estimates from logistic regression analyses.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_byse">byse</code></td>
<td>
<p>The standard errors associated with the beta-coefficients in <code>by</code>.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_correlation">correlation</code></td>
<td>
<p>The matrix of correlations between genetic variants. If this variable is not provided, then we assume that genetic variants are uncorrelated.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_exposure">exposure</code></td>
<td>
<p>The name of the exposure variable.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_outcome">outcome</code></td>
<td>
<p>The name of the outcome variable.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_snps">snps</code></td>
<td>
<p>The names of the genetic variants (SNPs) included in the analysis. The inputs <code>exposure</code>, <code>outcome</code>, and <code>snps</code> are not required, but may be useful for keeping track of various <code>MRInput</code> objects. They are also used by the <code>mr_plot</code> function.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_effect_allele">effect_allele</code></td>
<td>
<p>The name of the effect allele for each SNP. The beta-coefficients are the associations with the exposure and outcome per additional copy of the effect allele.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_other_allele">other_allele</code></td>
<td>
<p>The name of the non-effect allele.</p>
</td></tr>
<tr><td><code id="mr_input_+3A_eaf">eaf</code></td>
<td>
<p>The expected allele frequencies (numeric). The slots <code>effect_allele</code>, <code>other_allele</code>, and <code>eaf</code> are neither required, nor currently used in the MendelianRandomization package. They are included for future compatibility with the MR-Base suite of functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The beta-coefficients are assumed to be estimated for uncorrelated (independent) genetic variants, although a correlation matrix can be specified if the variants are correlated in their distributions. We also assume that the beta-coefficients for associations with the exposure and with the outcome are uncorrelated (corresponding to a two-sample Mendelian randomization analysis), although correlation between associations with the exposure and with the outcome generally have little impact on causal estimates or standard errors.
</p>
<p>If the four variables are not all the same length, then an error message will be reported. The analyses will still try to run, but the output may be misleading. However, in some analyses (for example, the standard IVW and MR-Egger methods), the values of <code>bxse</code> are not used in the analysis, and can therefore safely be omitted (provided that the other variables are correctly labelled).
</p>


<h3>Value</h3>

<p>An MRInput object containing:
</p>
<table>
<tr><td><code>betaX</code></td>
<td>
<p>The genetic associations with the exposure.</p>
</td></tr>
<tr><td><code>betaXse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td></tr>
<tr><td><code>betaY</code></td>
<td>
<p>The genetic associations with the outcome.</p>
</td></tr>
<tr><td><code>betaYse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td></tr>
<tr><td><code>correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>snps</code></td>
<td>
<p>A vector of character strings with the names of the genetic variants.</p>
</td></tr>
<tr><td><code>effect_allele</code></td>
<td>
<p>A vector of character strings with the names of the effect alleles.</p>
</td></tr>
<tr><td><code>other_allele</code></td>
<td>
<p>A vector of character strings with the names of the non-effect alleles.</p>
</td></tr>
<tr><td><code>eaf</code></td>
<td>
<p>A numeric vector with the effect allele frequencies.</p>
</td></tr>
</table>

<hr>
<h2 id='mr_ivw'>Inverse-variance weighted method</h2><span id='topic+mr_ivw'></span><span id='topic+mr_ivw+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_ivw</code> function implements the inverse-variance method, informally known as the &quot;Toby Johnson&quot; method. With a single
genetic variant, this is simply the ratio method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_ivw(
  object,
  model = "default",
  robust = FALSE,
  penalized = FALSE,
  weights = "simple",
  psi = 0,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05,
  ...
)

## S4 method for signature 'MRInput'
mr_ivw(
  object,
  model = "default",
  robust = FALSE,
  penalized = FALSE,
  weights = "simple",
  psi = 0,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_ivw_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_model">model</code></td>
<td>
<p>What type of model should be used: <code>"default"</code>, <code>"random"</code> or <code>"fixed"</code>. The random-effects model (<code>"random"</code>) is a multiplicative random-effects model, allowing overdispersion in the weighted linear regression (the residual standard error is not fixed to be 1, but is not allowed to take values below 1). The fixed-effect model (<code>"fixed"</code>) sets the residual standard error to be 1. The <code>"default"</code> setting is to use a fixed-effect model with 3 genetic variants or fewer, and otherwise to use a random-effects model.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_robust">robust</code></td>
<td>
<p>Indicates whether robust regression using the <code>lmrob()</code> function from the package <code>robustbase</code> should be used in the method rather than standard linear regression (<code>lm</code>).</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_penalized">penalized</code></td>
<td>
<p>Indicates whether a penalty should be applied to the weights to downweight the contribution of genetic variants with outlying ratio estimates to the analysis.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_weights">weights</code></td>
<td>
<p>Which weights to use in the weighted regression. If <code>"simple"</code> (the default option), then the IVW estimate is equivalent to meta-analysing the ratio estimates from each variant using inverse-variance weights based on the simplest expression of the variance for the ratio estimate (first-order term from the delta expansion - standard error of the association with the outcome divided by the association with the exposure). If <code>"delta"</code>, then the variance expression is the second-order term from the delta expansion. The second-order term incorporates uncertainty in the genetic association with the exposure &ndash; this uncertainty is ignored using the simple weighting.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_psi">psi</code></td>
<td>
<p>The correlation between the genetic associations with the exposure and the association with the outcome for each variant resulting from sample overlap. The default value is <code>0</code>, corresponding to a strict two-sample Mendelian randomization analysis (no overlap). If there is complete overlap between the samples, then the correlation should be set to the observational correlation between the exposure and the outcome. This correlation is only used in the calculation of standard errors if the option <code>weights</code> is set to <code>"delta"</code>.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_correl">correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for. The matrix of correlations between must be provided in the <code>MRInput</code> object: the elements of this matrix are the correlations between the individual variants (diagonal elements are 1). If a correlation matrix is specified in the <code>MRInput</code> object, then <code>correl</code> is set to <code>TRUE</code>. If <code>correl</code> is set to <code>TRUE</code>, then the values of <code>robust</code> and <code>penalized</code> are taken as <code>FALSE</code>, and <code>weights</code> is set to <code>"simple"</code>.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_ivw_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the regression method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>With multiple uncorrelated genetic variants, this estimate can be thought of as: 1) the inverse-variance
weighted combination of the ratio estimates from a meta-analysis; 2) the ratio estimate from combining the
genetic variants into a weighted score and then using this score as an instrumental variable (the same estimate
is obtained from the two-stage least squares method using individual-level data); 3) the coefficient from
weighted linear regression of the associations with the outcome on the associations with the risk factor fixing
the intercept to zero and using the inverse-variance weights.
</p>
<p>Here, we implement the method using weighted linear regression. If the variants are correlated, the method is implemented using generalized weighted linear regression; this is hard coded using matrix algebra.
</p>
<p>The causal estimate is obtained by regression of the associations with the outcome on the associations with the risk factor, with the intercept set to zero and weights being the inverse-variances of the associations with the outcome.
</p>
<p>With a single genetic variant, the estimate is the ratio of coefficients betaY/betaX and the standard error is the first term of the delta method approximation betaYse/betaX.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>IVW</code> object containing:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"fixed"</code>, <code>"random"</code>, or <code>"default"</code>).</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>Robust</code></td>
<td>
<p><code>TRUE</code> if robust regression has been used to calculate the estimate, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Penalized</code></td>
<td>
<p><code>TRUE</code> if weights have been penalized, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>Standard error of the causal estimate.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the estimate (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</td></tr>
<tr><td><code>Fstat</code></td>
<td>
<p>An approximation of the first-stage F statistic for all variants based on the summarized data.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Original implementation: The International Consortium for Blood Pressure Genome-Wide Association Studies. Genetic variants in novel pathways influence blood pressure and cardiovascular disease risk. Nature 2011; 478:103-109. doi: 10.1038/nature10405.
</p>
<p>Detailed description of method: Stephen Burgess, Adam S Butterworth, Simon G Thompson. Mendelian randomization analysis with multiple genetic variants using summarized data. Genetic Epidemiology 2013; 37:658-665. doi: 10.1002/gepi.21758.
</p>
<p>Robust and penalized weights: Stephen Burgess, Jack Bowden, Frank Dudbridge, Simon G Thompson. Robust instrumental variable methods using multiple candidate instruments with application to Mendelian randomization. arXiv 2016; 1606.03729.
</p>
<p>Heterogeneity test: Fabiola del Greco, Cosetta Minelli, Nuala A Sheehan, John R Thompson. Detecting pleiotropy in Mendelian randomisation studies with summary data and a continuous outcome. Stat Med 2015; 34(21):2926-2940. doi: 10.1002/sim.6522.
</p>
<p>Simple versus delta weights (first-order versus second-order): Stephen Burgess, Jack Bowden. Integrating summarized data from multiple genetic variants in Mendelian randomization: bias and coverage properties of inverse-variance weighted methods. arXiv:1512.04486.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_ivw(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))
mr_ivw(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  robust = TRUE)
mr_ivw(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  penalized = TRUE)
mr_ivw(mr_input(calcium, calciumse, fastgluc, fastglucse, corr=calc.rho))
  ## correlated variants

</code></pre>

<hr>
<h2 id='mr_lasso'>MR-Lasso method</h2><span id='topic+mr_lasso'></span><span id='topic+mr_lasso+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_lasso</code> function performs the MR-Lasso method, which applies lasso-type penalization to the direct effects of genetic variants on the outcome.
The causal estimate is described as a post-lasso estimate, and is obtained by performing the IVW method using only those genetic variants that are identified as valid by the lasso procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_lasso(object, distribution = "normal", alpha = 0.05, lambda = numeric(0))

## S4 method for signature 'MRInput'
mr_lasso(object, distribution = "normal", alpha = 0.05, lambda = numeric(0))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_lasso_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_lasso_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_lasso_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_lasso_+3A_lambda">lambda</code></td>
<td>
<p>The value of the tuning parameter used by the lasso procedure which controls the level of sparsity. If not specified, the tuning parameter will be calculated by the heterogeneity stopping rule.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MR-Lasso extends the IVW model to include an intercept term for each genetic variant. These intercept terms represent associations between the
genetic variants and the outcome which bypass the risk factor. The causal effect estimates are estimated by weighted linear regression where the intercept terms are subject
to lasso-type penalization. The lasso penalization will tend to shrink the intercept terms corresponding to the valid instruments to zero.
</p>
<p>The lasso penalty relies on a tuning parameter which controls the level of sparsity. The default is to use a heterogeneity stopping rule, but a fixed value may be specified.
</p>
<p>As part of the analysis, the genetic variants are orientated so that all of the associations with the risk factor are positive (and signs of associations with the outcome are
changed to keep the orientation consistent if required). Re-orientation of the genetic variants is performed automatically as part of the function.
</p>
<p>The MR-Lasso method is performed in two steps. First, a regularized regression model is fitted, and some genetic variants are identified as valid instruments. Second, the causal effect is estimated using standard IVW with only the valid genetic variants.
The post-lasso method will be performed as long as at least two genetic variants are identified as valid instruments. The default heterogeneity stopping rule will always return at least two
genetic variants as valid instruments.
The main estimate given by the method is the post-lasso estimate. However, parameter estimates from the regularized regression model used to identify invalid variants are also provided for completeness.
</p>
<p>If a substantial proportion of genetic variants are removed from the analysis, the MR-Lasso method may give a false impression of confidence in the causal estimate due to homogeneity of the variant-specific causal estimates amongst the remaining variants. However, it is not reasonable to claim that there is strong evidence for a causal effect after a large number of variants with heterogeneous estimates have been removed from the analysis.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MRLasso</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The causal estimate from the MR-Lasso method. This is the post-lasso estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>The standard error of the causal estimate from the MR-Lasso method.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound of the confidence interval for the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound of the confidence interval for the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the causal estimate using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RegEstimate</code></td>
<td>
<p>The estimate from the regularized regression model used in the MR-Lasso method.</p>
</td></tr>
<tr><td><code>RegIntercept</code></td>
<td>
<p>The intercept estimates from the regularized regression model used in the MR-Lasso method.</p>
</td></tr>
<tr><td><code>Valid</code></td>
<td>
<p>The number of genetic variants that have been identified as valid instruments.</p>
</td></tr>
<tr><td><code>ValidSNPs</code></td>
<td>
<p>The names of genetic variants that have been identified as valid instruments.</p>
</td></tr>
<tr><td><code>Lambda</code></td>
<td>
<p>The value of the tuning parameter used to compute <code>RegEstimate</code></p>
</td></tr>
</table>


<h3>References</h3>

<p>Jessica MB Rees, Angela M Wood, Frank Dudbridge, Stephen Burgess. Robust methods in Mendelian randomization via penalization of heterogeneous causal estimates. PLoS ONE 2019; 14(9):e0222362
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_lasso(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_loo'>Leave-one-out estimates</h2><span id='topic+mr_loo'></span><span id='topic+mr_loo+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_loo</code> function draws a forest plot of causal estimates from the <code>mr_ivw</code> function using default settings (assuming variants are uncorrelated, random-effects for 4+ variants) omitting each variant in turn. So the estimate labelled <code>snp_1</code> includes all variants except the labelled variant, and so on. The <code>mr_ivw</code> estimate including all variants (&quot;IVW estimate&quot;) is also provided for reference.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_loo(object, alpha = 0.05)

## S4 method for signature 'MRInput'
mr_loo(object, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_loo_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_loo_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05, corresponding to 95% confidence intervals.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>As the function produces a <code>ggplot</code> object, graphical parameters can be changed by adding commands from the <code>ggplot2</code> package.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_loo(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  alpha = 0.01)

</code></pre>

<hr>
<h2 id='mr_maxlik'>Maximum-likelihood method</h2><span id='topic+mr_maxlik'></span><span id='topic+mr_maxlik+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_maxlik</code> function implements the maximum-likelihood method introduced by Burgess et al (2013).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_maxlik(
  object,
  model = "default",
  correl = FALSE,
  psi = 0,
  distribution = "normal",
  alpha = 0.05,
  ...
)

## S4 method for signature 'MRInput'
mr_maxlik(
  object,
  model = "default",
  correl = FALSE,
  psi = 0,
  distribution = "normal",
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_maxlik_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_maxlik_+3A_model">model</code></td>
<td>
<p>What type of model should be used: <code>"default"</code>, <code>"random"</code> or <code>"fixed"</code>. The method naturally estimates a fixed-effect model, assuming that the same causal effect is estimated by each of the genetic variants. However, if there is heterogeneity in the causal estimates of the different variants, then confidence intervals under a fixed-effect model will be overly narrow. The random-effects model adds additional uncertainty by multiplying the standard error by the square-root of the likelihood ratio heterogeneity statistic divided by the number of genetic variants less one (unless this quantity is less than 1, in which case no modification to the standard error is made). This parallels the residual standard error in a regression model (the Cochran Q heterogeneity test statistic is equal to the square of the RSE multiplied by the number of genetic variants less one). The default setting (<code>"default"</code>) is to use a fixed-effect model with 3 genetic variants or fewer, and otherwise to use a random-effects model.</p>
</td></tr>
<tr><td><code id="mr_maxlik_+3A_correl">correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for. The matrix of correlations between must be provided in the <code>MRInput</code> object: the elements of this matrix are the correlations between the individual variants (diagonal elements are 1).</p>
</td></tr>
<tr><td><code id="mr_maxlik_+3A_psi">psi</code></td>
<td>
<p>The correlation between the association with the exposure and the association with the outcome for each variant resulting from sample overlap.</p>
</td></tr>
<tr><td><code id="mr_maxlik_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals, can be <code>"normal"</code> (the default option) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_maxlik_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_maxlik_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the optimization method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A likelihood function is defined by assuming that the summarized data for each genetic variant are normally distributed. A bivariate normal distribution is assumed for the associations of each genetic variant with the exposure and with the outcome. The mean of the association with the outcome is taken as the mean association with the exposure multiplied by the causal effect parameter.
</p>
<p>Thus, if there are <code>K</code> genetic variants, then <code>K+1</code> parameters are estimated by the method: one for each gene&ndash;exposure association, plus the causal parameter. If the number of genetic variants is large, then maximization of this function may be an issue. If the maximum likelihood estimate substantially differs from the inverse-variance weighted estimate, this may indicate that convergence has not occurred in the optimization algorithm.
</p>
<p>The variance-covariance matrices for the bivariate normal distributions are obtained from the standard error estimates provided. The correlation <code>psi</code> between genetic associations with the exposure and with the outcome due to sample overlap can be specified; its default value is zero.
</p>
<p>Two features why this method may be preferred over the inverse-variance weighted method are the incorporation in the model of uncertainty in the genetic associations with the exposure, and of correlation between the genetic association estimates with exposure and outcome for each variant. The method is implemented both for uncorrelated and correlated genetic variants. It can also be used for a single genetic variant.
</p>
<p>The original version of the maximum-likelihood method assumed that all genetic variants identify the same causal estimate; a fixed-effect model. The causal estimate may be overly precise if the fixed-effect model is incorrect and there is substantial heterogeneity in the causal estimates from the different variants. The random-effects analysis implemented here is an ad hoc solution to the problem of heterogeneity, but one that should result in reasonable confidence intervals that incorporate this heterogeneity.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MaxLik</code> object containing:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"fixed"</code>, <code>"random"</code>, or <code>"default"</code>).</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>Psi</code></td>
<td>
<p>The correlation between genetic associations with the exposure and with the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>Standard error of the causal estimate.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the estimate (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model (always equal to 1, as a fixed-effect model is required.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (likelihood ratio statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Stephen Burgess, Adam S Butterworth, Simon G Thompson. Mendelian randomization analysis with multiple genetic variants using summarized data. Genetic Epidemiology 2013; 37:658-665. doi: 10.1002/gepi.21758.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_maxlik(mr_input(bx = ldlc[1:10], bxse = ldlcse[1:10],
  by = chdlodds[1:10], byse = chdloddsse[1:10]))
mr_maxlik(mr_input(bx = ldlc[1:10], bxse = ldlcse[1:10],
  by = chdlodds[1:10], byse = chdloddsse[1:10]), psi=0.2)
mr_maxlik(mr_input(calcium, calciumse, fastgluc, fastglucse, corr=calc.rho))
  ## correlated variants

</code></pre>

<hr>
<h2 id='mr_mbe'>Mode-based method of Hartwig</h2><span id='topic+mr_mbe'></span><span id='topic+mr_mbe+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mbe</code> function implements the mode-based method introduced by Hartwig, Bowden and Davey Smith (2017).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mbe(
  object,
  weighting = "weighted",
  stderror = "simple",
  phi = 1,
  seed = 314159265,
  iterations = 10000,
  distribution = "normal",
  alpha = 0.05
)

## S4 method for signature 'MRInput'
mr_mbe(
  object,
  weighting = "weighted",
  stderror = "delta",
  phi = 1,
  seed = 314159265,
  iterations = 10000,
  distribution = "normal",
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mbe_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_weighting">weighting</code></td>
<td>
<p>Whether the analysis should be <code>"weighted"</code> (the default option) or <code>"unweighted"</code>.</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_stderror">stderror</code></td>
<td>
<p>Whether standard error estimates should be i) <code>"simple"</code> - calculated as the first-order term from the delta expansion - standard error of the association with the outcome divided by the association with the exposure), or ii) <code>"delta"</code> - calculated as the second-order term from the delta expansion (the default option). The second-order term incorporates uncertainty in the genetic association with the exposure &ndash; this uncertainty is ignored using the simple weighting. The <code>"simple"</code> option is referred to by Hartwig et al as &quot;assuming NOME&quot;, and the <code>"delta"</code> option as &quot;not assuming NOME&quot;.</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_phi">phi</code></td>
<td>
<p>The choice of bandwidth in the kernel-smoothly density method. A value of 1 (the default value) represents the bandwidth value selected by the modified Silverman's bandwidth rule, as recommended by Hartwig et al. A value of 0.5 represents half that value, and so on.</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_seed">seed</code></td>
<td>
<p>The random seed to use when generating the bootstrap samples used to calculate the confidence intervals (for reproducibility). The default value is 314159265. If set to <code>NA</code>, the random seed will not be set (for example, if the function is used as part of a larger simulation).</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_iterations">iterations</code></td>
<td>
<p>Number of iterations to use in the bootstrap procedure.</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals, can be <code>"normal"</code> (the default option) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_mbe_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The mode-based estimation (MBE) method takes the variant-specific ratio estimates from each genetic variant in turn, and calculates the modal estimate. This is implemented by constructing a kernel-smoothed density out of the ratio estimates, and taking the maximum value as the modal estimate. The standard error is calculated by a bootstrap procedure, and confidence intervals based on the estimate having a normal distribution.
</p>
<p>The method should give consistent estimates as the sample size increases if a plurality (or weighted plurality) of the genetic variants are valid instruments. This means that the largest group of variants with the same causal estimate in the asymptotic limit are the valid instruments.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MRMBE</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Weighting</code></td>
<td>
<p>A character string <code>"weighted"</code> or <code>"unweighted"</code>.</p>
</td></tr>
<tr><td><code>StdErr</code></td>
<td>
<p>A character string <code>"simple"</code> or <code>"delta"</code>.</p>
</td></tr>
<tr><td><code>Phi</code></td>
<td>
<p>The value of the bandwidth factor.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>Standard error of the causal estimate.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the estimate (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Fernando Pires Hartwig, George Davey Smith, Jack Bowden. Robust inference in summary data Mendelian randomization via the zero modal pleiotropy assumption. International Journal of Epidemiology 2017; 46(6): 1985-1998. doi: 10.1093/ije/dyx102.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mbe(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse), iterations=100)
mr_mbe(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
   phi=0.5, iterations=100)
mr_mbe(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
   weighting="weighted", stderror="delta", iterations=100)
# iterations set to 100 to reduce computational time,
#  more iterations are recommended in practice

</code></pre>

<hr>
<h2 id='mr_median'>Median-based method</h2><span id='topic+mr_median'></span><span id='topic+mr_median+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_median</code> function implements the weighted median (default) or simple median method introduced by Bowden et al (2016) to calculate
the median of the ratio instrumental variable estimates evaluated using each genetic variant individually.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_median(
  object,
  weighting = "weighted",
  distribution = "normal",
  alpha = 0.05,
  iterations = 10000,
  seed = 314159265
)

## S4 method for signature 'MRInput'
mr_median(
  object,
  weighting = "weighted",
  distribution = "normal",
  alpha = 0.05,
  iterations = 10000,
  seed = 314159265
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_median_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_median_+3A_weighting">weighting</code></td>
<td>
<p>The type of weighting applied. The default option is to calculate the weighted median (<code>"weighted"</code>); other options are <code>"simple"</code> and <code>"penalized"</code>.</p>
</td></tr>
<tr><td><code id="mr_median_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution to use to calculate the 95% confidence intervals, can be <code>"normal"</code> or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_median_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_median_+3A_iterations">iterations</code></td>
<td>
<p>The number of bootstrap samples to generate when calculating the estimated standard error. The default value is 10000.</p>
</td></tr>
<tr><td><code id="mr_median_+3A_seed">seed</code></td>
<td>
<p>The random seed to use when generating the bootstrap samples (for reproducibility). The default value is 314159265. If set to <code>NA</code>, the random seed will not be set (for example, if the function is used as part of a larger simulation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The median-based methods have greater robustness to individual genetic variants with strongly outlying causal estimates
compared with the inverse-variance weighted and MR-Egger methods. Formally, the simple median method gives a
consistent estimate of the causal effect when at least 50% of the genetic variants are valid instrumental variables
(for the weighted median method, when 50% of the weight comes from valid instrumental variables).
</p>
<p>When the weighting is <code>"simple"</code>, the estimate is obtained by calculating the ratio causal estimates
from each genetic variants theta = betaY/betaX, and finding the median estimate.
</p>
<p>When the weighting is <code>"weighted"</code>, the estimate is obtained by:
</p>

<ol>
<li><p> Calculating the ratio causal estimates and ordering the genetic variants according to the magnitude of their estimates, i.e. </p>
<p style="text-align: center;"><code class="reqn">\theta_1 &lt; \theta_2 &lt; ... &lt; \theta_J</code>
</p>

</li>
<li><p> Calculate normalized inverse-variance weights for each genetic variant <code class="reqn">w_1, w_2, ..., w_J</code>, as:
</p>
<p style="text-align: center;"><code class="reqn">w_j = \frac{\beta_{Xj}^2}{se(\beta_{Yj})^2} / \sum_{i=1}^{J} \frac{\beta_{Xi}^2}{se(\beta_{Yi})^2}</code>
</p>

</li>
<li><p> Find k such that
</p>
<p style="text-align: center;"><code class="reqn">s_k = \sum_{i = 1}^k w_i &lt; 0.5</code>
</p>

<p>and
</p>
<p style="text-align: center;"><code class="reqn">s_{k+1}  = \sum_{i = 1}^{k+1} w_i &gt; 0.5</code>
</p>

</li>
<li><p> Calculate the weighted median estimate by extrapolation as:
</p>
<p style="text-align: center;"><code class="reqn">\theta_{WM} = \theta_k + (\theta_{k+1} - \theta_k) \times \frac{0.5 - s_k}{s_{k+1} - s_k}</code>
</p>
 </li></ol>

<p>The simple median estimate is the same as the weighted median estimate when all the weights are equal. Standard errors
for both the simple and weighted median methods are calculated through bootstrapping.
</p>
<p>When the weighting is <code>"penalized"</code>, the weighted method is used, but the contribution of genetic variants with outlying (heterogeneous) ratio estimates to the analysis is downweighted.
</p>


<h3>Value</h3>

<p>The output from the function is a <code>WeightedMedian</code> object containing:
</p>
<table>
<tr><td><code>Type</code></td>
<td>
<p>The type of weights used: <code>"weighted"</code>, <code>"simple"</code>, or <code>"penalized"</code>.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The value of the causal estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>Standard error of the causal estimate calculated using bootstrapping.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound for the causal estimate based on the estimated bootstrapped standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound for the causal estimate based on the estimated bootstrapped standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-value associated with the estimate (calculated using <code>Estimate/StdError</code> as per a Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Jack Bowden, George Davey Smith, Philip C Haycock, Stephen Burgess. Consistent estimation in Mendelian randomization with
some invalid instruments using a weighted median estimator. Genetic Epidemiology 2016; 40(4):304-314. doi: 10.1002/gepi.21965.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_median(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  weighting = "weighted", iterations = 100)
  # iterations is set to 100 to reduce runtime for the mr_median method,
  # 10000 iterations are recommended in practice
mr_median(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  weighting = "simple", iterations = 100)
mr_median(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  weighting = "penalized", iterations = 100)

</code></pre>

<hr>
<h2 id='mr_mvcML'>Multivariable constrained maximum likelihood method</h2><span id='topic+mr_mvcML'></span><span id='topic+mr_mvcML+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvcML</code> function performs multivariable Mendelian randomization via the constrained maximum likelihood method, which is robust to both correlated and uncorrelated pleiotropy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvcML(
  object,
  n,
  DP = TRUE,
  rho_mat = diag(ncol(object@betaX) + 1),
  K_vec = 0:(ceiling(nrow(object@betaX)/2)),
  random_start = 0,
  num_pert = 100,
  min_theta_range = -0.5,
  max_theta_range = 0.5,
  maxit = 100,
  alpha = 0.05,
  seed = 314159265
)

## S4 method for signature 'MRMVInput'
mr_mvcML(
  object,
  n,
  DP = TRUE,
  rho_mat = diag(ncol(object@betaX) + 1),
  K_vec = 0:(ceiling(nrow(object@betaX)/2)),
  random_start = 0,
  num_pert = 200,
  min_theta_range = -0.5,
  max_theta_range = 0.5,
  maxit = 100,
  alpha = 0.05,
  seed = 314159265
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvcML_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_n">n</code></td>
<td>
<p>Sample size. The smallest sample size among all (both exposures and outcome) GWAS used in the analysis is recommended.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_dp">DP</code></td>
<td>
<p>Whether data perturbation is applied or not. Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_rho_mat">rho_mat</code></td>
<td>
<p>The correlation matrix among the exposures and outcome GWAS estimates, which can be estimated by the intercept term from bivariate LDSC. See reference for more discussions. Default is the identify matrix, for example, in the absence of overlapping samples among GWAS datasets.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_k_vec">K_vec</code></td>
<td>
<p>Set of candidate K's, the constraint parameter 
representing number of invalid IVs. It can range from 0 up to #IV - (#exposure + 1). Default is from 0 to (#IV/2).</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_random_start">random_start</code></td>
<td>
<p>Number of random starting points for MVMRcML, default is 0.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_num_pert">num_pert</code></td>
<td>
<p>Number of perturbation when DP is TRUE, default is 200.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_min_theta_range">min_theta_range</code></td>
<td>
<p>The lower bound of the uniform distribution for each initial value for theta generated from, default is -0.5.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_max_theta_range">max_theta_range</code></td>
<td>
<p>The uppder bound of the uniform distribution for each initial value for theta generated from, default is 0.5.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations for each optimization. Default is 100.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_alpha">alpha</code></td>
<td>
<p>Significance level for the confidence interval for estimate, default is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvcML_+3A_seed">seed</code></td>
<td>
<p>The random seed to use when generating the perturbed samples (for reproducibility). The default value is 314159265. If set to <code>NA</code>, the random seed will not be set (for example, if the function is used as part of a larger simulation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multivariable MRcML (MVMRcML) is an extension of MRcML to deal with multiple exposures of interest. It is robust to both correlated and uncorrelated pleiotropy as its univariable version.
</p>
<p>In practice, the data perturbation (DP) version is preferred in practice for a more robust inference as it can account for the uncertainty in model selection.
However, it may take a longer time especially when the number of IVs is large (so the range of <code>K_vec</code> can be large too).
One strategy is to try a small range of K (the number of invalid IVs) first (with a small <code>num_pert</code>), 
then adjust it if the number of selected invalid IVs fall close to the boundary.
You can also use other methods, e.g. <code>mr_mvlasso</code>, to get a rough sense of the number of invalid IVs.
</p>
<p>Similar to <code>mr_cML</code>, multiple random starting points could be used to find a global minimum.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVMRcML</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal distribution.</p>
</td></tr>
<tr><td><code>BIC_invalid</code></td>
<td>
<p>Set of selected invalid IVs by MVMRcML-BIC.</p>
</td></tr>
<tr><td><code>K_hat</code></td>
<td>
<p>The number of selected invalid IVs by MVMRcML-BIC, or a vector for each data perturbation in MVMRcML-DP.</p>
</td></tr>
<tr><td><code>eff_DP_B</code></td>
<td>
<p>The number of data perturbations with successful convergence in MVMRcML-DP.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lin, Z., Xue, H., &amp; Pan, W. (2023). Robust multivariable Mendelian randomization based on constrained maximum likelihood. The American Journal of Human Genetics, 110(4), 592-605.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Perform MVMRcML-DP:
mr_mvcML(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), n = 17723, num_pert = 5, random_start = 5)
  # num_pert is set to 5 to reduce runtime for the mr_mvcML method,
  # At least 100 perturbations should be used and more is preferred for a stable result.

rho_mat = matrix(c(1,-0.1,0.2,0,-0.1,1,-0.3,0,
 				  0.2,-0.3,1,0,0,0,0,1),ncol=4) ## Toy example of rho_mat
mr_mvcML(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), n = 17723, num_pert = 5, rho_mat = rho_mat)

# Perform MVMRcML-BIC:
mr_mvcML(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), n = 17723, DP = FALSE)

</code></pre>

<hr>
<h2 id='mr_mvegger'>Multivariable MR-Egger method</h2><span id='topic+mr_mvegger'></span><span id='topic+mr_mvegger+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvegger</code> function performs multivariable Mendelian randomization via the MR-Egger method. This is implemented by multivariable weighted linear regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvegger(
  object,
  orientate = 1,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05
)

## S4 method for signature 'MRMVInput'
mr_mvegger(
  object,
  orientate = 1,
  correl = FALSE,
  distribution = "normal",
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvegger_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvegger_+3A_orientate">orientate</code></td>
<td>
<p>The risk factor that genetic associations are orientated to. The univariable and multivariable versions of MR-Egger are both sensitive to the choice of parameterization of the genetic associations - which allele the associations are orientated with respect to (in other words, which allele is the effect allele). For univariable MR-Egger, this is resolved by setting the genetic associations with the exposure all to be positive. In multivariable MR-Egger, we have to choose which of the exposures to orientate the genetic associations to. The default option is <code>1</code>, meaning that genetic associations with the first exposure are set to be positive.</p>
</td></tr>
<tr><td><code id="mr_mvegger_+3A_correl">correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for. The matrix of correlations between must be provided in the <code>MRInput</code> object: the elements of this matrix are the correlations between the individual variants (diagonal elements are 1). If a correlation matrix is specified in the <code>MRInput</code> object, then <code>correl</code> is set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="mr_mvegger_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_mvegger_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multivariable MR-Egger is an extension of the MR-Egger method to deal with genetic variants that are associated with multiple risk factors.
</p>
<p>We implement the method using multivariable weighted linear regression. If the variants are correlated, the method is implemented using generalized weighted linear regression; this is hard coded using matrix algebra.
</p>
<p>The causal estimate is obtained by regression of the associations with the outcome on the associations with the risk factors, with the intercept estimated and weights being the inverse-variances of the associations with the outcome.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVEgger</code> object containing:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"random"</code>).</p>
</td></tr>
<tr><td><code>Orientate</code></td>
<td>
<p>The number corresponding to the risk factor that the genetic associations are orientated to.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of the causal estimates (slope coefficient).</p>
</td></tr>
<tr><td><code>StdError.Est</code></td>
<td>
<p>Standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>Pvalue.Est</code></td>
<td>
<p>The p-values associated with the estimates using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>CILower.Est</code></td>
<td>
<p>The lower bound of the causal estimates based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper.Est</code></td>
<td>
<p>The upper bound of the causal estimates based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Intercept</code></td>
<td>
<p>The value of the intercept estimate.</p>
</td></tr>
<tr><td><code>StdError.Int</code></td>
<td>
<p>Standard error of the intercept estimate.</p>
</td></tr>
<tr><td><code>Pvalue.Int</code></td>
<td>
<p>The p-value associated with the intercept.</p>
</td></tr>
<tr><td><code>CILower.Int</code></td>
<td>
<p>The lower bound of the intercept based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper.Int</code></td>
<td>
<p>The upper bound of the intercept based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Jessica Rees, Angela Wood, Stephen Burgess. Extending the MR-Egger method for multivariable Mendelian randomization to correct for both measured and unmeasured pleiotropy. Statistics in Medicine 2017; 36(29): 4705-4718. doi: 10.1002/sim.7492.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvegger(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), orientate = 1)

</code></pre>

<hr>
<h2 id='mr_mvgmm'>Multivariable generalized method of moments (GMM) method</h2><span id='topic+mr_mvgmm'></span><span id='topic+mr_mvgmm+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvgmm</code> function performs multivariable Mendelian randomization via the generalized method of moments method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvgmm(object, nx, ny, cor.x = NULL, robust = TRUE, alpha = 0.05, ...)

## S4 method for signature 'MRMVInput'
mr_mvgmm(object, nx, ny, cor.x = NULL, robust = TRUE, alpha = 0.05, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvgmm_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvgmm_+3A_nx">nx</code></td>
<td>
<p>Vector of sample sizes used to compute genetic associations with the exposure (one for each exposure).</p>
</td></tr>
<tr><td><code id="mr_mvgmm_+3A_ny">ny</code></td>
<td>
<p>The sample size used to compute genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="mr_mvgmm_+3A_cor.x">cor.x</code></td>
<td>
<p>Correlation matrix for exposures. Default is to assume the exposures are uncorrelated.</p>
</td></tr>
<tr><td><code id="mr_mvgmm_+3A_robust">robust</code></td>
<td>
<p>Indicates whether overdispersion heterogeneity is accounted for in the model. Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_mvgmm_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvgmm_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the optimization routines to calculate GMM estimates and overdispersion parameter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Robust inference in two-sample multivariable Mendelian randomization using the generalized method of moments. The method accounts for overdispersion heterogeneity in genetic variant-outcome associations.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVGMM</code> object containing:
</p>
<table>
<tr><td><code>Robust</code></td>
<td>
<p><code>TRUE</code> if overdispersion heterogeneity was included in the model, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations if supplied. If not supplied, then an identity matrix will be returned.</p>
</td></tr>
<tr><td><code>ExpCorrelation</code></td>
<td>
<p><code>TRUE</code> if an exposure correlation matrix was specified, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>CondFstat</code></td>
<td>
<p>A vector of conditional F-statistics (one for each exposure).</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Overdispersion</code></td>
<td>
<p>The estimate of the overdispersion parameter. If this is negative, then a value of zero is used in the method.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal distribution.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value (for non-robust model only): the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more genetic variants may be pleiotropic. If the robust option is set to <code>TRUE</code>, then this represents excess heterogeneity beyond the overdispersion model.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Description of the generalized method of moments: Hansen, L. P. (1982). Large sample properties of generalized method of moments estimators. Econometrica, pp.1029-1054.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvgmm(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), nx=rep(17723,3), ny=17723)

</code></pre>

<hr>
<h2 id='mr_mvinput'>Inputting and formatting data for use in causal estimation</h2><span id='topic+mr_mvinput'></span>

<h3>Description</h3>

<p>The <code>mr_mvinput</code> function is required for inputting and formatting data for use in the multivariable Mendelian randomization functions provided in this package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvinput(
  bx = matrix(),
  bxse = matrix(),
  by = 0,
  byse = 0,
  correlation = matrix(),
  exposure = "exposure",
  outcome = "outcome",
  snps = "snp",
  effect_allele = NA,
  other_allele = NA,
  eaf = NA
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvinput_+3A_bx">bx</code></td>
<td>
<p>A matrix of beta-coefficient values for genetic associations with the risk factor variables. These should be arranged so that column 1 are the beta-coefficients for risk factor 1, and row 1 are the beta-coefficients for genetic variant 1.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_bxse">bxse</code></td>
<td>
<p>The matrix of standard errors associated with the beta-coefficients <code>bx</code>.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_by">by</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the second variable (often referred to as the outcome). For a disease outcome, the beta coefficients are log odds estimates from logistic regression analyses.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_byse">byse</code></td>
<td>
<p>The vector standard errors associated with the beta-coefficients in <code>by</code>.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_correlation">correlation</code></td>
<td>
<p>The matrix of correlations between genetic variants. If this variable is not provided, then we assume that genetic variants are uncorrelated.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_exposure">exposure</code></td>
<td>
<p>The names of the exposure variables.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_outcome">outcome</code></td>
<td>
<p>The name of the outcome variable.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_snps">snps</code></td>
<td>
<p>The names of the genetic variants (SNPs) included in the analysis. The inputs <code>exposure</code>, <code>outcome</code>, and <code>snps</code> are not required, but may be useful for keeping track of various <code>MRInput</code> objects. They are also used by the <code>mr_plot</code> function.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_effect_allele">effect_allele</code></td>
<td>
<p>The name of the effect allele for each SNP. The beta-coefficients are the associations with the exposure and outcome per additional copy of the effect allele.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_other_allele">other_allele</code></td>
<td>
<p>The name of the non-effect allele.</p>
</td></tr>
<tr><td><code id="mr_mvinput_+3A_eaf">eaf</code></td>
<td>
<p>The expected allele frequencies (numeric). The slots <code>effect_allele</code>, <code>other_allele</code>, and <code>eaf</code> are neither required, nor currently used in the MendelianRandomization package. They are included for future compatibility with the MR-Base suite of functions.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The beta-coefficients are assumed to be estimated for uncorrelated (independent) genetic variants, although a correlation matrix can be specified if the variants are correlated in their distributions. We also assume that the beta-coefficients for associations with the exposure and with the outcome are uncorrelated (corresponding to a two-sample Mendelian randomization analysis), although correlation between associations with the exposure and with the outcome generally have little impact on causal estimates or standard errors.
</p>
<p>If the variables are not all the same length, then an error message will be reported. The analyses will still try to run, but the output may be misleading. However, in some analyses (for example, the standard IVW and MR-Egger methods), the values of <code>bxse</code> are not used in the analysis, and can therefore safely be omitted (provided that the other variables are correctly labelled).
</p>


<h3>Value</h3>

<p>An MRMVInput object containing:
</p>
<table>
<tr><td><code>betaX</code></td>
<td>
<p>The genetic associations with the exposures.</p>
</td></tr>
<tr><td><code>betaXse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td></tr>
<tr><td><code>betaY</code></td>
<td>
<p>The genetic associations with the outcome.</p>
</td></tr>
<tr><td><code>betaYse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td></tr>
<tr><td><code>correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>exposure</code></td>
<td>
<p>Character strings with the names given to the exposures.</p>
</td></tr>
<tr><td><code>outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>snps</code></td>
<td>
<p>A vector of character strings with the names of the genetic variants.</p>
</td></tr>
<tr><td><code>effect_allele</code></td>
<td>
<p>A vector of character strings with the names of the effect alleles.</p>
</td></tr>
<tr><td><code>other_allele</code></td>
<td>
<p>A vector of character strings with the names of the non-effect alleles.</p>
</td></tr>
<tr><td><code>eaf</code></td>
<td>
<p>A numeric vector with the effect allele frequencies.</p>
</td></tr>
</table>

<hr>
<h2 id='mr_mvivw'>Multivariable inverse-variance weighted method</h2><span id='topic+mr_mvivw'></span><span id='topic+mr_mvivw+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvivw</code> function performs multivariable Mendelian randomization via the inverse-variance method. This is implemented by multivariable weighted linear regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvivw(
  object,
  model = "default",
  robust = FALSE,
  correl = FALSE,
  correl.x = NULL,
  nx = NA,
  distribution = "normal",
  alpha = 0.05,
  ...
)

## S4 method for signature 'MRMVInput'
mr_mvivw(
  object,
  model = "default",
  robust = FALSE,
  correl = FALSE,
  correl.x = NULL,
  nx = NA,
  distribution = "normal",
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvivw_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_model">model</code></td>
<td>
<p>What type of model should be used: <code>"default"</code>, <code>"random"</code> or <code>"fixed"</code>. The random-effects model (<code>"random"</code>) is a multiplicative random-effects model, allowing overdispersion in the weighted linear regression (the residual standard error is not fixed to be 1, but is not allowed to take values below 1). The fixed-effect model (<code>"fixed"</code>) sets the residual standard error to be 1. The <code>"default"</code> setting is to use a fixed-effect model with 3 genetic variants or fewer, and otherwise to use a random-effects model.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_robust">robust</code></td>
<td>
<p>Indicates whether robust regression using the <code>lmrob()</code> function from the package <code>robustbase</code> should be used in the method rather than standard linear regression (<code>lm</code>).</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_correl">correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for. The matrix of correlations between must be provided in the <code>MRMVInput</code> object: the elements of this matrix are the correlations between the individual variants (diagonal elements are 1). If a correlation matrix is specified in the <code>MRMVInput</code> object, then <code>correl</code> is set to <code>TRUE</code>.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_correl.x">correl.x</code></td>
<td>
<p>Correlation matrix for exposures (Optional). Default is to assume the exposures are uncorrelated. This is only used in the computation of conditional F-statistics.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_nx">nx</code></td>
<td>
<p>Either a single value, or a vector of sample sizes for the genetic associations with the exposures (one for each exposure, and assumed equal for all variants). If a single value is provided, it is assumed this is the sample size for all exposures. This is optional, and it is only used in the calculation of conditional F statistics. If not supplied, then conditional F statistics are not reported.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvivw_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the regression method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multivariable Mendelian randomization is an extension of Mendelian randomization to deal with genetic variants that are associated with multiple risk factors. Two scenarios are envisioned for its use: 1) risk factors that are biologically related, such as lipid fractions; and 2) risk factors where there is potentially a network of causal effects (mediation) from one risk factor to another. In both cases, under the extended assumptions of multivariable Mendelian randomization, coefficients represent the direct causal effects of each risk factor in turn with the other risk factors being fixed.
</p>
<p>We implement the method using multivariable weighted linear regression. If the variants are correlated, the method is implemented using generalized weighted linear regression; this is hard coded using matrix algebra.
</p>
<p>The causal estimate is obtained by regression of the associations with the outcome on the associations with the risk factors, with the intercept set to zero and weights being the inverse-variances of the associations with the outcome.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVIVW</code> object containing:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"fixed"</code>, <code>"random"</code>, or <code>"default"</code>).</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Robust</code></td>
<td>
<p><code>TRUE</code> if robust regression has been used to calculate the estimate, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</td></tr>
<tr><td><code>CondFstat</code></td>
<td>
<p>Conditional F statistics: An approximation of the first-stage conditional F statistics for all variants based on the summarized data. This represents the instrument strength for each exposure conditional on other exposures in the model. This is only reported if the sample sizes for the genetic associations with the exposures are provided.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Description of approach: Stephen Burgess, Simon G Thompson. Multivariable Mendelian Randomization: the use of pleiotropic genetic variants to estimate causal effects. American Journal of Epidemiology 2015; 181(4):251-260. doi: 10.1093/aje/kwu283.
</p>
<p>Description of inverse-variance weighted method: Stephen Burgess, Frank Dudbridge, Simon G Thompson. Re: &quot;Multivariable Mendelian randomization: the use of pleiotropic genetic variants to estimate causal effects.&quot; American Journal of Epidemiology 2015; 181(4):290-291. doi: 10.1093/aje/kwv017.
</p>
<p>Use for mediation analysis: Stephen Burgess, Deborah J Thompson, Jessica MB Rees, Felix R Day, John R Perry, Ken K Ong. Dissecting causal pathways using Mendelian randomization with summarized genetic data: Application to age at menarche and risk of breast cancer. Genetics 2017; 207(2):481-487. doi: 10.1534/genetics.117.300191.
</p>
<p>Calculation of conditional F statistics: Ashish Patel, Dipender Gill, Dmitry Shungin, Christos Mantzoros, Lotte Bjerre Knudsen, Jack Bowden, Stephen Burgess. Robust use of phenotypic heterogeneity at drug target genes for mechanistic insights: application of multivariable Mendelian randomization to <em>GLP1R</em> gene region. Pre-print.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvivw(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_mvivwme'>Multivariable inverse-variance weighted method with measurement error</h2><span id='topic+mr_mvivwme'></span><span id='topic+mr_mvivwme+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvivwme</code> function performs multivariable Mendelian randomization via the inverse-variance method with measurement error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvivwme(
  object,
  model = "default",
  correl = FALSE,
  correl.x = NULL,
  distribution = "normal",
  alpha = 0.05,
  max_iter = 100,
  no_ini = 1,
  seed = 20201201,
  ...
)

## S4 method for signature 'MRMVInput'
mr_mvivwme(
  object,
  model = "default",
  correl = FALSE,
  correl.x = NULL,
  distribution = "normal",
  alpha = 0.05,
  max_iter = 100,
  no_ini = 1,
  seed = 20201201,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvivwme_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_model">model</code></td>
<td>
<p>What type of model should be used: <code>"default"</code>, <code>"random"</code> or <code>"fixed"</code>. The random-effects model (<code>"random"</code>) is a multiplicative random-effects model, allowing overdispersion in the weighted linear regression (the residual standard error is not fixed to be 1, but is not allowed to take values below 1). The fixed-effect model (<code>"fixed"</code>) sets the residual standard error to be 1. The <code>"default"</code> setting is to use a fixed-effect model with 3 genetic variants or fewer, and otherwise to use a random-effects model.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_correl">correl</code></td>
<td>
<p>If the genetic variants are correlated, then this correlation can be accounted for.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_correl.x">correl.x</code></td>
<td>
<p>Correlation matrix for exposures. Default is to assume the exposures are uncorrelated.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_max_iter">max_iter</code></td>
<td>
<p>The maximum number of iterations in the optimisation procedure.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_no_ini">no_ini</code></td>
<td>
<p>The number of initial values for the optimisation procedure.</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_seed">seed</code></td>
<td>
<p>The random seed to use for the optimisation procedure. The default value is 20201201. If set to <code>NA</code>, the random seed will not be set (for example, if the function is used as part of a larger simulation).</p>
</td></tr>
<tr><td><code id="mr_mvivwme_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the regression method.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The extension of multivariable Mendelian randomization to account for measurement error in the genetic associations with the exposure traits.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVIVWME</code> object containing:
</p>
<table>
<tr><td><code>Model</code></td>
<td>
<p>A character string giving the type of model used (<code>"fixed"</code>, <code>"random"</code>, or <code>"default"</code>).</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RSE</code></td>
<td>
<p>The estimated residual standard error from the regression model.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Zhu, Jiazheng, Stephen Burgess, and Andrew J. Grant. Bias in Multivariable Mendelian Randomization Studies Due to Measurement Error on Exposures, 2022. https://doi.org/10.48550/arXiv.2203.08668.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvivwme(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_mvlasso'>Multivariable MR-Lasso method</h2><span id='topic+mr_mvlasso'></span><span id='topic+mr_mvlasso+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvlasso</code> function performs the multivariable MR-Lasso method, which applies lasso-type penalization to the direct effects of genetic variants on the outcome.
The causal estimates are described as post-lasso estimates, and are obtained by performing the multivariable IVW method using only those genetic variants that are identified as valid by the lasso procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvlasso(
  object,
  orientate = 1,
  distribution = "normal",
  alpha = 0.05,
  lambda = numeric(0)
)

## S4 method for signature 'MRMVInput'
mr_mvlasso(
  object,
  orientate = 1,
  distribution = "normal",
  alpha = 0.05,
  lambda = numeric(0)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvlasso_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvlasso_+3A_orientate">orientate</code></td>
<td>
<p>The risk factor that genetic associations are orientated to. The default option is <code>1</code>, meaning that genetic associations with the first risk factor are set to be positive.</p>
</td></tr>
<tr><td><code id="mr_mvlasso_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_mvlasso_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvlasso_+3A_lambda">lambda</code></td>
<td>
<p>The value of the tuning parameter used by the lasso procedure which controls the level of sparsity. If not specified, the tuning parameter will be calculated by the heterogeneity stopping rule.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Multivariable MR-Lasso extends the multivariable IVW model to include an intercept term for each genetic variant. These intercept terms represent associations between the
genetic variants and the outcome which bypass the risk factors. The regularized regression model is estimated by multivariable weighted linear regression where the intercept terms are subject
to lasso-type penalization. The lasso penalization will tend to shrink the intercept terms corresponding to the valid instruments to zero.
</p>
<p>The lasso penalty relies on a tuning parameter which controls the level of sparsity. The default is to use a heterogeneity stopping rule, but a fixed value may be specified.
</p>
<p>As part of the analysis, the genetic variants are orientated so that all of the associations with one of the risk factors are positive (the first risk factor is used by default). Re-orientation
of the genetic variants is performed automatically as part of the function.
</p>
<p>The MR-Lasso method is performed in two steps. First, a regularized regression model is fitted, and some genetic variants are identified as valid instruments. Second, causal effects are estimated using standard multivariable IVW with only the valid genetic variants.
The post-lasso method will be performed as long as the number of genetic variants identified as valid instruments is greater than the number of risk factors.
The default heterogeneity stopping rule will always return more genetic variants as valid instruments than risk factors for identification.
The main estimates given by the method are the post-lasso estimates. However, parameter estimates from the regularized regression model used to identify invalid variants are also provided for completeness.
</p>
<p>If a substantial proportion of genetic variants are removed from the analysis, the multivariable MR-Lasso method may give a false impression of confidence in the causal estimate due to homogeneity of the variant-specific causal estimates amongst the remaining variants. However, it is not reasonable to claim that there is strong evidence for a causal effect after a large number of variants with heterogeneous estimates have been removed from the analysis.
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVLasso</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates from the multivariable MR-Lasso method. These are the post-lasso estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates from the multivariable MR-Lasso method.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the confidence intervals for the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the confidence intervals for the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the (post-lasso) causal estimates using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
<tr><td><code>RegEstimate</code></td>
<td>
<p>The estimates from the regularized regression model used in the multivariable MR-Lasso method.</p>
</td></tr>
<tr><td><code>RegIntercept</code></td>
<td>
<p>The intercept estimates estimates from the regularized regression model used in the multivariable MR-Lasso method.</p>
</td></tr>
<tr><td><code>Valid</code></td>
<td>
<p>The number of genetic variants that have been identified as valid instruments.</p>
</td></tr>
<tr><td><code>ValidSNPs</code></td>
<td>
<p>The names of genetic variants that have been identified as valid instruments.</p>
</td></tr>
<tr><td><code>Lambda</code></td>
<td>
<p>The value of the tuning parameter used to compute <code>RegEstimate</code>.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Andrew J Grant, Stephen Burgess. Pleiotropy robust methods for multivariable Mendelian randomization. arXiv 2020; 2008.11997
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvlasso(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_mvmedian'>Multivariable median-based method</h2><span id='topic+mr_mvmedian'></span><span id='topic+mr_mvmedian+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvmedian</code> function performs multivariable Mendelian randomization via the median method. This is implemented by multivariable weighted quantile regression, with the quantile set to 0.5 (the median).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvmedian(
  object,
  distribution = "normal",
  alpha = 0.05,
  iterations = 10000,
  seed = 314159265
)

## S4 method for signature 'MRMVInput'
mr_mvmedian(
  object,
  distribution = "normal",
  alpha = 0.05,
  iterations = 10000,
  seed = 314159265
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvmedian_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvmedian_+3A_distribution">distribution</code></td>
<td>
<p>The type of distribution used to calculate the confidence intervals. Options are <code>"normal"</code> (default) or <code>"t-dist"</code>.</p>
</td></tr>
<tr><td><code id="mr_mvmedian_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvmedian_+3A_iterations">iterations</code></td>
<td>
<p>The number of bootstrap samples to generate when calculating the estimated standard error. The default value is 10000.</p>
</td></tr>
<tr><td><code id="mr_mvmedian_+3A_seed">seed</code></td>
<td>
<p>The random seed to use when generating the bootstrap samples (for reproducibility). The default value is 314159265. If set to <code>NA</code>, the random seed will not be set (for example, if the function is used as part of a larger simulation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The multivariable median method is similar to the univariable weighted median method, except that it is implemented using quantile regression. The regression model is multivariable and weighted by the inverse of the variances of the variant-specific estimates. Confidence intervals are calculated by parametric bootstrap to estimate the standard error of the estimates, and then using quantiles of a normal or t-distribution (depending on the value of <code>distribution</code>).
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVMedian</code> object containing:
</p>
<table>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal or t-distribution (as specified in <code>distribution</code>).</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of genetic variants (SNPs) included in the analysis.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvmedian(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse), iterations = 100)
  # iterations is set to 100 to reduce runtime for the mr_mvmedian method,
  # 10000 iterations are recommended in practice

</code></pre>

<hr>
<h2 id='mr_mvpcgmm'>Multivariable principal components generalized method of moments (PC-GMM) method</h2><span id='topic+mr_mvpcgmm'></span><span id='topic+mr_mvpcgmm+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_mvpcgmm</code> function performs multivariable Mendelian randomization via the principal components generalized method of moments method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_mvpcgmm(
  object,
  nx,
  ny,
  cor.x = NULL,
  r = NULL,
  thres = 0.99,
  robust = TRUE,
  alpha = 0.05,
  ...
)

## S4 method for signature 'MRMVInput'
mr_mvpcgmm(
  object,
  nx,
  ny,
  cor.x = NULL,
  r = NULL,
  thres = 0.99,
  robust = TRUE,
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_mvpcgmm_+3A_object">object</code></td>
<td>
<p>An <code>MRMVInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_nx">nx</code></td>
<td>
<p>Vector of sample sizes used to compute genetic associations with the exposure (one for each exposure).</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_ny">ny</code></td>
<td>
<p>The sample size used to compute genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_cor.x">cor.x</code></td>
<td>
<p>Correlation matrix for exposures. Default is to assume the exposures are uncorrelated.</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_r">r</code></td>
<td>
<p>The number of genetic principal components to be used to instrument the exposures. Default chooses <code>r</code> to explain 99.9% of variation in a sample weighted genetic correlation matrix (this can be varied by setting the <code>thres</code> parameter).</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_thres">thres</code></td>
<td>
<p>The threshold value of variation in the sample weighted genetic correlation matrix explained by the genetic principal components. The default value is 0.99, indicating that 99% of variation is explained by the principal components. Note that if <code>r</code> and <code>thres</code> are both specified, then <code>r</code> will take precedence and <code>thres</code> will be ignored.</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_robust">robust</code></td>
<td>
<p>Indicates whether overdispersion heterogeneity is accounted for in the model. Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_mvpcgmm_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the optimization routines to calculate GMM estimates and overdispersion parameter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When a Mendelian randomization analysis is performed using correlated genetic variants from a single gene region, there is a tradeoff between using too few variants (and compromising on power) and using too many variants (in which case, estimates can be highly sensitive to small variation in the correlation matrix). This method performs dimension reduction on a weighted version of the genetic correlation matrix to form principal components based on the genetic variants, which are then used as instruments. It is recommended not to include very highly correlated variants in this method (say, <code>r^2 &gt; 0.95</code>), but the method should cope well with variants correlated below this level.
</p>
<p>This function runs a multivariable version of the PC-GMM method, which can be used when there are distinct exposures associated with variants at a single gene region. Phenotypic heterogeneity (that is, the genetic associations with the exposures are not collinear) at genomic loci encoding drug targets can be exploited by multivariable Mendelian randomization to provide insight on the pathways by which pharmacological interventions may affect disease risk.
</p>
<p>This method provides two-sample multivariable Mendelian randomization estimates and associated confidence intervals that account for overdispersion heterogeneity in dimension-reduced genetic associations (when <code>robust = TRUE</code>).
</p>


<h3>Value</h3>

<p>The output from the function is an <code>MVPCGMM</code> object containing:
</p>
<table>
<tr><td><code>Robust</code></td>
<td>
<p><code>TRUE</code> if overdispersion heterogeneity was included in the model, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character vector with the names given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>ExpCorrelation</code></td>
<td>
<p><code>TRUE</code> if an exposure correlation matrix was specified, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>CondFstat</code></td>
<td>
<p>A vector of conditional F-statistics (one for each exposure).</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>A vector of causal estimates.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>A vector of standard errors of the causal estimates.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bounds of the causal estimates based on the estimated standard errors and the significance level provided.</p>
</td></tr>
<tr><td><code>Overdispersion</code></td>
<td>
<p>The estimate of the overdispersion parameter.</p>
</td></tr>
<tr><td><code>PCs</code></td>
<td>
<p>The number of genetic principal components used to instrument the exposures.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal distribution.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value (for non-robust model only): the null hypothesis is that all genetic principal components estimate the same causal parameter; rejection of the null is an indication that one or more principal components may be pleiotropic.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Description of multivariable Mendelian randomization: Stephen Burgess, Simon G Thompson. Multivariable Mendelian Randomization: the use of pleiotropic genetic variants to estimate causal effects. American Journal of Epidemiology 2015; 181(4):251-260. doi: 10.1093/aje/kwu283.
</p>
<p>Description of the PC-GMM method: &quot;Robust use of phenotypic heterogeneity at drug target genes for mechanistic insights: application of cis-multivariable Mendelian randomization to GLP1R gene region&quot; (Preprint).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_mvpcgmm(mr_mvinput(bx = cbind(ldlc, hdlc, trig), bxse = cbind(ldlcse, hdlcse, trigse),
   by = chdlodds, byse = chdloddsse, correlation = diag(length(ldlc))), nx=rep(17723,3), ny=17723)
# Note this example does not use variants from a single gene region, and is provided
#  to demonstrate that the code works, rather than to illustrate a recommended use case.

</code></pre>

<hr>
<h2 id='mr_pcgmm'>Univariable principal components generalized method of moments (PC-GMM) method</h2><span id='topic+mr_pcgmm'></span><span id='topic+mr_pcgmm+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_pcgmm</code> function performs multivariable Mendelian randomization via the principal components generalized method of moments method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_pcgmm(
  object,
  nx,
  ny,
  r = NULL,
  thres = 0.99,
  robust = TRUE,
  alpha = 0.05,
  ...
)

## S4 method for signature 'MRInput'
mr_pcgmm(
  object,
  nx,
  ny,
  r = NULL,
  thres = 0.99,
  robust = TRUE,
  alpha = 0.05,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_pcgmm_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_nx">nx</code></td>
<td>
<p>The sample size used to compute genetic associations with the exposure.</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_ny">ny</code></td>
<td>
<p>The sample size used to compute genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_r">r</code></td>
<td>
<p>The number of genetic principal components to be used to instrument the exposure. Default chooses <code>r</code> to explain 99.9% of variation in a sample weighted genetic correlation matrix (this can be varied by setting the <code>thres</code> parameter).</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_thres">thres</code></td>
<td>
<p>The threshold value of variation in the sample weighted genetic correlation matrix explained by the genetic principal components. The default value is 0.99, indicating that 99% of variation is explained by the principal components. Note that if <code>r</code> and <code>thres</code> are both specified, then <code>r</code> will take precedence and <code>thres</code> will be ignored.</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_robust">robust</code></td>
<td>
<p>Indicates whether overdispersion heterogeneity is accounted for in the model. Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence interval. The default value is 0.05.</p>
</td></tr>
<tr><td><code id="mr_pcgmm_+3A_...">...</code></td>
<td>
<p>Additional arguments to be passed to the optimization routines to calculate the GMM estimate and overdispersion parameter.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When a Mendelian randomization analysis is performed using correlated genetic variants from a single gene region, there is a tradeoff between using too few variants (and compromising on power) and using too many variants (in which case, estimates can be highly sensitive to small variation in the correlation matrix). This method performs dimension reduction on a weighted version of the genetic correlation matrix to form principal components based on the genetic variants, which are then used as instruments. It is recommended not to include very highly correlated variants in this method (say, <code>r^2 &gt; 0.95</code>), but the method should cope well with variants correlated below this level.
</p>
<p>This function runs a univariable version of the PC-GMM method, which can be used when there are a single exposure associated with variants at a given gene region.
</p>
<p>This method provides two-sample univariable Mendelian randomization estimates and associated confidence intervals that account for overdispersion heterogeneity in dimension-reduced genetic associations (when <code>robust = TRUE</code>).
</p>


<h3>Value</h3>

<p>The output from the function is an <code>PCGMM</code> object containing:
</p>
<table>
<tr><td><code>Robust</code></td>
<td>
<p><code>TRUE</code> if overdispersion heterogeneity was included in the model, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string with the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string with the names given to the outcome.</p>
</td></tr>
<tr><td><code>Correlation</code></td>
<td>
<p>The matrix of genetic correlations.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The causal estimate.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>The standard error of the causal estimate.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound of the causal estimate based on the estimated standard error and the significance level provided.</p>
</td></tr>
<tr><td><code>Fstat</code></td>
<td>
<p>The first-stage F statistic for all genetic principal components used as instruments.</p>
</td></tr>
<tr><td><code>Overdispersion</code></td>
<td>
<p>The estimate of the overdispersion parameter.</p>
</td></tr>
<tr><td><code>PCs</code></td>
<td>
<p>The number of genetic principal components used to instrument the exposure.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>The p-values associated with the estimates (calculated as Estimate/StdError as per Wald test) using a normal distribution.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used when calculating the confidence intervals.</p>
</td></tr>
<tr><td><code>Heter.Stat</code></td>
<td>
<p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value (for non-robust model only): the null hypothesis is that all genetic principal components estimate the same causal parameter; rejection of the null is an indication that one or more principal components may be pleiotropic.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Description of the PC-GMM method: &quot;Robust use of phenotypic heterogeneity at drug target genes for mechanistic insights: application of cis-multivariable Mendelian randomization to GLP1R gene region&quot; (Preprint).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_pcgmm(mr_input(bx = calcium, bxse = calciumse,
   by = fastgluc, byse = fastglucse, correlation = calc.rho), nx=6351, ny=133010)

</code></pre>

<hr>
<h2 id='mr_pivw'>Penalized inverse-variance weighted method</h2><span id='topic+mr_pivw'></span><span id='topic+mr_pivw+2CMRInput-method'></span>

<h3>Description</h3>

<p>The <code>mr_pivw</code> function implements the penalized inverse-variance weighted (pIVW) method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_pivw(
  object,
  lambda = 1,
  over.dispersion = TRUE,
  delta = 0,
  sel.pval = NULL,
  Boot.Fieller = TRUE,
  alpha = 0.05
)

## S4 method for signature 'MRInput'
mr_pivw(
  object,
  lambda = 1,
  over.dispersion = TRUE,
  delta = 0,
  sel.pval = NULL,
  Boot.Fieller = TRUE,
  alpha = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_pivw_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object.</p>
</td></tr>
<tr><td><code id="mr_pivw_+3A_lambda">lambda</code></td>
<td>
<p>The penalty parameter in the pIVW estimator. It plays a role in the bias-variance trade-off of the estimator. It is recommended to choose <code>lambda=1</code> to achieve the smallest bias and valid inference. By default, <code>lambda=1</code>.</p>
</td></tr>
<tr><td><code id="mr_pivw_+3A_over.dispersion">over.dispersion</code></td>
<td>
<p>Should the method consider overdispersion (balanced horizontal pleiotropy)? Default is TRUE.</p>
</td></tr>
<tr><td><code id="mr_pivw_+3A_delta">delta</code></td>
<td>
<p>The z-score threshold for IV selection. <code>delta</code> should be greater than or equal to zero. By default, <code>delta=0</code> (i.e., no IV selection will be conducted).  See 'Details'.</p>
</td></tr>
<tr><td><code id="mr_pivw_+3A_sel.pval">sel.pval</code></td>
<td>
<p>A numeric vector containing the P-values of the SNP effects on the exposure, which will be used for the IV selection. <code>sel.pval</code> should be provided when <code>delta</code> is not zero. See 'Details'.</p>
</td></tr>
<tr><td><code id="mr_pivw_+3A_boot.fieller">Boot.Fieller</code></td>
<td>
<p>If <code>Boot.Fieller=TRUE</code>, then the P-value and the confidence interval of the causal effect will be calculated based on the bootstrapping Fieller method. Otherwise, the P-value and the confidence interval of the causal effect will be calculated from the normal distribution. It is recommended to use the bootstrapping Fieller method when <code>Condition</code> is smaller than 10 (see 'Details'). By default, <code>Boot.Fieller=TRUE</code>.</p>
</td></tr>
<tr><td><code id="mr_pivw_+3A_alpha">alpha</code></td>
<td>
<p>The significance level used to calculate the confidence intervals. The default value is 0.05.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The penalized inverse-variance weighted (pIVW) estimator accounts for weak instruments and balanced horizontal pleiotropy simultaneously
in two-sample MR with summary statistics, i.e., an exposure sample (with IV-exposure effect <code>Bx</code> and standard error <code>Bxse</code>) and
an outcome sample (with IV-outcome effect <code>By</code> and standard error <code>Byse</code>).
</p>
<p>The pIVW estimator also allows for IV selection in three-sample MR, where weak IVs are screened out using
an extra sample (with IV-exposure effect <code>Bx*</code> and standard error <code>Bxse*</code>) independent of the exposure sample and outcome sample.
Generally, the P-value for <code>Bx*</code> can be computed by <code>sel.pval=2*pnorm(abs(Bx*/Bxse*), lower.tail = FALSE)</code>,
Given <code>sel.pval</code> and a z-score threshold <code>delta</code>, the variants kept in the analysis will be those
with <code>sel.pval&lt;2*pnorm(delta,lower.tail = FALSE)</code>.
</p>
<p>The <code>mr_pivw</code> function outputs a measure <code>Condition</code> that needs to be large for reliable asymptotic properties of the pIVW estimator.
We also refer to <code>Condition</code> as effective sample size, which is a function of a measure of IV strength and the number of IVs.
When <code>delta</code> is zero (i.e., no IV selection), <code>Condition = (average F-statistic -1)*sqrt(# snps)</code>. When <code>delta</code> is not zero
(i.e., IV selection is conducted), <code>Condition = [(average F-statistic -1)*sqrt(# snps)]/c</code>,
where the numerator is computed using the selected variants, and the denominator <code>c</code> involves the selection probabilities
of all variants (see more details in the paper). We suggest that <code>Condition</code> should be greater than 5 for the pIVW estimator to achieve reliable asymptotic properties.
</p>


<h3>Value</h3>

<p>The output from the function is a <code>PIVW</code> object containing:
</p>
<table>
<tr><td><code>Over.dispersion</code></td>
<td>
<p><code>TRUE</code> if the method has considered balanced horizontal pleiotropy, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Boot.Fieller</code></td>
<td>
<p><code>TRUE</code> if the bootstrapping Fieller method is used to calculate the P-value and the confidence interval of the causal effect, <code>FALSE</code> otherwise.</p>
</td></tr>
<tr><td><code>Lambda</code></td>
<td>
<p>The penalty parameter in the pIVW estimator.</p>
</td></tr>
<tr><td><code>Delta</code></td>
<td>
<p>The z-score threshold for IV selection.</p>
</td></tr>
<tr><td><code>Exposure</code></td>
<td>
<p>A character string giving the name given to the exposure.</p>
</td></tr>
<tr><td><code>Outcome</code></td>
<td>
<p>A character string giving the name given to the outcome.</p>
</td></tr>
<tr><td><code>Estimate</code></td>
<td>
<p>The causal point estimate from the pIVW estimator.</p>
</td></tr>
<tr><td><code>StdError</code></td>
<td>
<p>The standard error associated with <code>Estimate</code>.</p>
</td></tr>
<tr><td><code>CILower</code></td>
<td>
<p>The lower bound of the confidence interval for <code>Estimate</code>, which is derived from the bootstrapping Fieller method or normal distribution. For the bootstrapping Fieller's interval, if it contains multiple ranges, then lower limits of all ranges will be reported.</p>
</td></tr>
<tr><td><code>CIUpper</code></td>
<td>
<p>The upper bound of the confidence interval for <code>Estimate</code>, which is derived from the bootstrapping Fieller method or normal distribution. For the bootstrapping Fieller's interval, if it contains multiple ranges, then upper limits of all ranges will be reported.</p>
</td></tr>
<tr><td><code>Alpha</code></td>
<td>
<p>The significance level used in constructing the confidence interval.</p>
</td></tr>
<tr><td><code>Pvalue</code></td>
<td>
<p>P-value associated with the causal estimate from the pIVW estimator, which is derived from the bootstrapping Fieller method or normal distribution.</p>
</td></tr>
<tr><td><code>Tau2</code></td>
<td>
<p>The variance of the balanced horizontal pleiotropy. <code>Tau2</code> is calculated by using all IVs in the data before conducting the IV selection.</p>
</td></tr>
<tr><td><code>SNPs</code></td>
<td>
<p>The number of SNPs after IV selection.</p>
</td></tr>
<tr><td><code>Condition</code></td>
<td>
<p>The estimated effective sample size. It is recommended to be greater than 5 for the pIVW estimator to achieve reliable asymptotic properties. See 'Details'.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Xu S., Wang P., Fung W.K. and Liu Z. (2022). A Novel Penalized Inverse-Variance Weighted Estimator for Mendelian Randomization with Applications to COVID-19 Outcomes. Biometrics. doi: 10.1111/biom.13732.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_pivw(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse))

</code></pre>

<hr>
<h2 id='mr_plot'>Draw a scatter plot of the genetic associations and/or causal estimates</h2><span id='topic+mr_plot'></span><span id='topic+mr_plot+2CMRInput-method'></span><span id='topic+mr_plot+2CMRAll-method'></span><span id='topic+mr_plot+2CMRMVInput-method'></span>

<h3>Description</h3>

<p>The function <code>mr_plot</code> has three functionalities. It can generate a visual representation of <code>MRInput</code>, <code>MRMVInput</code>  and <code>MRAll</code> objects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mr_plot(
  object,
  error = TRUE,
  line = "ivw",
  orientate = FALSE,
  interactive = TRUE,
  labels = FALSE
)

## S4 method for signature 'MRInput'
mr_plot(
  object,
  error = TRUE,
  line = "ivw",
  orientate = FALSE,
  interactive = TRUE,
  labels = FALSE
)

## S4 method for signature 'MRAll'
mr_plot(object)

## S4 method for signature 'MRMVInput'
mr_plot(
  object,
  error = TRUE,
  line = TRUE,
  orientate = FALSE,
  interactive = TRUE,
  labels = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mr_plot_+3A_object">object</code></td>
<td>
<p>An <code>MRInput</code> object or an <code>MRMVInput</code> object or an <code>MRAll</code> object.</p>
</td></tr>
<tr><td><code id="mr_plot_+3A_error">error</code></td>
<td>
<p>When viewing an <code>MRInput</code> or <code>MRMVInput</code> object, one can choose whether to include error bars (default is to include). For an <code>MRMVInput</code> object, the horizontal error bars only take into account uncertainty in the causal estimates.</p>
</td></tr>
<tr><td><code id="mr_plot_+3A_line">line</code></td>
<td>
<p>When viewing an <code>MRInput</code> object, one can choose whether to include the IVW estimate (<code>line = "ivw"</code>) or the MR-Egger estimate (<code>line = "egger"</code>). When viewing an <code>MRMVInput</code>, one can choose whether to include a line through the origin with gradient 1 (<code>line = TRUE</code>) or not.</p>
</td></tr>
<tr><td><code id="mr_plot_+3A_orientate">orientate</code></td>
<td>
<p>When viewing an <code>MRInput</code> or <code>MRMVInput</code> object, one can choose whether to orientate all genetic variants so that the associations with the risk factor are all positive. This is recommended particularly when plotting the MR-Egger estimate, although the default setting is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="mr_plot_+3A_interactive">interactive</code></td>
<td>
<p>When viewing an <code>MRInput</code> or <code>MRMVInput</code> object, one can choose whether to produce an interactive graph using the <code>plotly</code> package, or a static graph using the regular <code>plot</code> command.</p>
</td></tr>
<tr><td><code id="mr_plot_+3A_labels">labels</code></td>
<td>
<p>When viewing an <code>MRInput</code> or <code>MRMVInput</code> object with <code>interactive</code> set to <code>FALSE</code>, setting <code>labels</code> to <code>TRUE</code> means that the name of each genetic variants appears above the corresponding datapoint.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The result is dependent on the type of object passed to <code>mr_plot</code>.
When the object is an <code>MRInput</code> object, the function uses either the <code>plot</code> command (if <code>interactive</code> is set to <code>FALSE</code>) or <code>plotly</code> syntax (if <code>interactive</code> is set to <code>TRUE</code>) to plot the association estimates against each other.
When the object is an <code>MRMVInput</code> object, functionality is similar except that we plot the estimated associations with the outcome on the y-axis, and fitted values of the associations with the outcome from the inverse-variance weighted method on the x-axis.
If <code>interactive</code> is set to <code>FALSE</code>, then a static graph is produced. By setting <code>labels</code> to <code>TRUE</code>, the names of the genetic variants appear above the points. This produces a less visually appealing graph, but one where it is easier to identify the individual genetic variants.
If <code>interactive</code> is set to <code>TRUE</code>, then the plot is interactive and the user can hover over the various points to see the name of the associated genetic variant and its association estimates.
When the object is an <code>MRAll</code> object, the function generates a <code>ggplot</code> to compare the causal estimates proposed by different methods.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mr_plot(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  line="egger", orientate = TRUE)
mr_plot(mr_input(bx = ldlc, bxse = ldlcse, by = chdlodds, byse = chdloddsse),
  line="ivw", interactive=FALSE) # produces a static graph
mr_plot(mr_allmethods(mr_input(bx = ldlc, bxse = ldlcse,
   by = chdlodds, byse = chdloddsse), method="all", iterations = 50))
  # iterations is set to 50 to reduce runtime for the mr_median method,
  # 10000 iterations are recommended in practice

</code></pre>

<hr>
<h2 id='MRAll-class'>MRAll Class</h2><span id='topic+MRAll-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the <code>mr_allmethods</code> function.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Data</code></dt><dd><p>The <code>mr_input</code> object that was used as an input to the <code>mr_allmethods</code> function. This includes the original data, so that a call to <code>mr_plot</code> can plot the original data and the various causal estimates.</p>
</dd>
<dt><code>Values</code></dt><dd><p>A data.frame object comprising estimates from the various methods called by the <code>mr_allmethods</code> function. The first column gives the names of the methods, then the causal estimates, standard errors, 95% confidence intervals, and p-values.</p>
</dd>
<dt><code>Method</code></dt><dd><p>A string indicating whether all methods are implemented (<code>"all"</code>, the default option), or just main methods (<code>"main"</code>), or only a subset of methods (<code>"ivw"</code>, <code>"egger"</code>, or <code>"median"</code>).</p>
</dd>
</dl>

<hr>
<h2 id='MRcML-class'>MRcML Class</h2><span id='topic+MRcML-class'></span>

<h3>Description</h3>

<p>An object containing the results of MRcML.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>Estimate of theta.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>Standard error of estimate.</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>p-value of estimate.</p>
</dd>
<dt><code>BIC_invalid</code></dt><dd><p>Set of selected invalid IVs if cML-BIC is performed, i.e. without MA or DP.</p>
</dd>
<dt><code>GOF1_p</code></dt><dd><p>p-value of the first goodness-of-fit test.</p>
</dd>
<dt><code>GOF2_p</code></dt><dd><p>p-value of the second goodness-of-fit test.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>Significance level for the confidence interval for estimate, default is 0.05.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>Lower bound of the confidence interval for estimate.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>Upper bound of the confidence interval for estimate.</p>
</dd>
<dt><code>MA</code></dt><dd><p>Indicator of whether model average is applied.</p>
</dd>
<dt><code>DP</code></dt><dd><p>Indicator of whether data perturbation is applied.</p>
</dd>
</dl>

<hr>
<h2 id='MRConMix-class'>MRConMix Class</h2><span id='topic+MRConMix-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the contamination mixture method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Psi</code></dt><dd><p>The value of the standard deviation of the distribution of invalid estimands (default is 1.5 times the standard deviation of the ratio estimates).</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimate from the contamination mixture method.</p>
</dd>
<dt><code>CIRange</code></dt><dd><p>The confidence interval for <code>Estimate</code> based on a grid search.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower limit of the confidence interval. If the confidence interval contains multiple ranges, then lower limits of all ranges will be reported.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper limit of the confidence interval. If the confidence interval contains multiple ranges, then upper limits of all ranges will be reported.</p>
</dd>
<dt><code>CIMin</code></dt><dd><p>The smallest value used in the search to find the confidence interval.</p>
</dd>
<dt><code>CIMax</code></dt><dd><p>The largest value used in the search to find the confidence interval.</p>
</dd>
<dt><code>CIStep</code></dt><dd><p>The step size used in the search to find the confidence interval.</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>The p-value associated with the estimate calculated using the likelihood function and a chi-squared distribution.</p>
</dd>
<dt><code>Valid</code></dt><dd><p>The numbers of genetic variants that were considered valid instruments at the causal estimate.</p>
</dd>
<dt><code>ValidSNPs</code></dt><dd><p>The names of genetic variants that were considered valid instruments at the causal estimate.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
</dl>

<hr>
<h2 id='MRHetPen-class'>MRHetPen Class</h2><span id='topic+MRHetPen-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the heterogeneity-penalized model-averaging mode-based estimation method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Prior</code></dt><dd><p>The value of the prior probability of a genetic variant being a valid instrument (default is 0.5).</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimate from the heterogeneity-penalized method.</p>
</dd>
<dt><code>CIRange</code></dt><dd><p>The confidence interval for <code>Estimate</code> based on a grid search.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower limit of the confidence interval. If the confidence interval contains multiple ranges, then lower limits of all ranges will be reported.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper limit of the confidence interval. If the confidence interval contains multiple ranges, then upper limits of all ranges will be reported.</p>
</dd>
<dt><code>CIMin</code></dt><dd><p>The smallest value used in the search to find the confidence interval.</p>
</dd>
<dt><code>CIMax</code></dt><dd><p>The largest value used in the search to find the confidence interval.</p>
</dd>
<dt><code>CIStep</code></dt><dd><p>The step size used in the search to find the confidence interval.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
</dl>

<hr>
<h2 id='MRInput-class'>MRInput Class</h2><span id='topic+MRInput-class'></span>

<h3>Description</h3>

<p>An object containing the four vectors of summary statistics required to calculate Mendelian randomization estimates.
</p>


<h3>Details</h3>

<p>The beta-coefficients are assumed to be estimated for uncorrelated (independent) genetic variants, although a correlation matrix can be specified if the variants are correlated in their distributions. We also assume that the beta-coefficients for associations with the exposure and with the outcome are uncorrelated (corresponding to a two-sample Mendelian randomization analysis), although correlation between associations with the exposure and with the outcome generally have little impact on causal estimates or standard errors.
Estimates can either be specified by the user, or extracted from the PhenoScanner tool.
</p>


<h3>Slots</h3>


<dl>
<dt><code>betaX</code></dt><dd><p>A numeric vector of beta-coefficient values for genetic associations with the first variable (often referred to as the exposure, risk factor, or modifiable phenotype).</p>
</dd>
<dt><code>betaY</code></dt><dd><p>A numeric vector of beta-coefficient values for genetic associations with the second variable (often referred to as the outcome). For a disease outcome, the beta coefficients are log odds estimates from logistic regression analyses.</p>
</dd>
<dt><code>betaXse</code></dt><dd><p>The standard errors associated with the beta-coefficients in <code>betaX</code>.</p>
</dd>
<dt><code>betaYse</code></dt><dd><p>The standard errors associated with the beta-coefficients in <code>betaY</code>.</p>
</dd>
<dt><code>correlation</code></dt><dd><p>The matrix of correlations between genetic variants. If this variable is not provided, then we assume that genetic variants are uncorrelated.</p>
</dd>
<dt><code>exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>snps</code></dt><dd><p>The names of the genetic variants (SNPs) included in the analysis. The slots <code>exposure</code>, <code>outcome</code>, and <code>snps</code> are not required, but may be useful for keeping track of various <code>MRInput</code> objects. They are also used by the <code>mr_plot</code> function.</p>
</dd>
<dt><code>effect_allele</code></dt><dd><p>The name of the effect allele for each SNP. The beta-coefficients are the associations with the exposure and outcome per additional copy of the effect allele.</p>
</dd>
<dt><code>other_allele</code></dt><dd><p>The name of the non-effect allele.</p>
</dd>
<dt><code>eaf</code></dt><dd><p>The expected allele frequencies (numeric). The slots <code>effect_allele</code>, <code>other_allele</code>, and <code>eaf</code> are neither required, nor currently used in the MendelianRandomization package. They are included for future compatibility with the MR-Base suite of functions.</p>
</dd>
</dl>

<hr>
<h2 id='MRLasso-class'>MRLasso class</h2><span id='topic+MRLasso-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the MR-Lasso method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimate from the MR-Lasso method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate from the MR-Lasso method.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs used in the calculation.</p>
</dd>
<dt><code>RegEstimate</code></dt><dd><p>The estimate from the regularized regression model used in the MR-Lasso method.</p>
</dd>
<dt><code>RegIntercept</code></dt><dd><p>The intercept estimates from the regularized regression model used in the MR-Lasso method. An intercept estimate of zero identifies the corresponding genetic variant as a valid instrument. Genetic variants with non-zero intercept estimates will be excluded from the post-lasso estimator.</p>
</dd>
<dt><code>Valid</code></dt><dd><p>The number of genetic variants that have been identified as valid instruments.</p>
</dd>
<dt><code>ValidSNPs</code></dt><dd><p>The names of genetic variants that have been identified as valid instruments.</p>
</dd>
<dt><code>Lambda</code></dt><dd><p>The value of the tuning parameter used to compute <code>RegEstimate</code> (default is to calulate <code>Lambda</code> using the heterogeneity stopping rule).</p>
</dd>
</dl>

<hr>
<h2 id='MRMBE-class'>MRMBE Class</h2><span id='topic+MRMBE-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the mode-based estimation method of Hartwig et al as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Weighting</code></dt><dd><p>Whether the analysis was <code>weighted</code> or <code>unweighted</code>.</p>
</dd>
<dt><code>StdErr</code></dt><dd><p>Whether the <code>simple</code> or <code>delta</code> version of the standard errors were used.</p>
</dd>
<dt><code>Phi</code></dt><dd><p>The value of the bandwidth factor.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimate from the mode-based estimation method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
</dl>

<hr>
<h2 id='MRMVInput-class'>MRMVInput Class</h2><span id='topic+MRMVInput-class'></span>

<h3>Description</h3>

<p>An object containing the summary statistics required to calculate multivariable Mendelian randomization estimates.
</p>


<h3>Details</h3>

<p>The beta-coefficients are assumed to be estimated for uncorrelated (independent) genetic variants, although a correlation matrix can be specified if the variants are correlated in their distributions. We also assume that the beta-coefficients for associations with the exposure and with the outcome are uncorrelated (corresponding to a two-sample Mendelian randomization analysis), although correlation between associations with the exposure and with the outcome generally have little impact on causal estimates or standard errors.
</p>


<h3>Slots</h3>


<dl>
<dt><code>betaX</code></dt><dd><p>A matrix of beta-coefficient values for genetic associations with the risk factor variables. These should be arranged so that column 1 are the beta-coefficients for risk factor 1, and row 1 are the beta-coefficients for genetic variant 1.</p>
</dd>
<dt><code>betaY</code></dt><dd><p>A numeric vector of beta-coefficient values for genetic associations with the second variable (often referred to as the outcome). For a disease outcome, the beta coefficients are log odds estimates from logistic regression analyses.</p>
</dd>
<dt><code>betaXse</code></dt><dd><p>The matrix of standard errors associated with the beta-coefficients in <code>betaX</code>.</p>
</dd>
<dt><code>betaYse</code></dt><dd><p>The vector of standard errors associated with the beta-coefficients in <code>betaY</code>.</p>
</dd>
<dt><code>correlation</code></dt><dd><p>The matrix of correlations between genetic variants. If this variable is not provided, then we assume that genetic variants are uncorrelated.</p>
</dd>
<dt><code>exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>snps</code></dt><dd><p>The names of the genetic variants (SNPs) included in the analysis. The slots <code>exposure</code>, <code>outcome</code>, and <code>snps</code> are not required, but may be useful for keeping track of various <code>MRInput</code> objects. They are also used by the <code>mr_plot</code> function.</p>
</dd>
<dt><code>effect_allele</code></dt><dd><p>The name of the effect allele for each SNP. The beta-coefficients are the associations with the exposure and outcome per additional copy of the effect allele.</p>
</dd>
<dt><code>other_allele</code></dt><dd><p>The name of the non-effect allele.</p>
</dd>
<dt><code>eaf</code></dt><dd><p>The expected allele frequencies (numeric). The slots <code>effect_allele</code>, <code>other_allele</code>, and <code>eaf</code> are neither required, nor currently used in the MendelianRandomization package. They are included for future compatibility with the MR-Base suite of functions.</p>
</dd>
</dl>

<hr>
<h2 id='mv_norm'>Sampling from multivariate normal distribution</h2><span id='topic+mv_norm'></span>

<h3>Description</h3>

<p>Sampling from multivariate normal distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mv_norm(n, mu, Sigma)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mv_norm_+3A_n">n</code></td>
<td>
<p>Number of draws.</p>
</td></tr>
<tr><td><code id="mv_norm_+3A_mu">mu</code></td>
<td>
<p>Mean vector.</p>
</td></tr>
<tr><td><code id="mv_norm_+3A_sigma">Sigma</code></td>
<td>
<p>Covariance matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Random vector of length n.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>mv_norm(1, 0.2, matrix(0.1))

</code></pre>

<hr>
<h2 id='MVcML_SdTheta'>Standard error estimate for MVMR-cML-BIC</h2><span id='topic+MVcML_SdTheta'></span>

<h3>Description</h3>

<p>This is based on the profile likelihood of the set of valid IVs, which is not robust to uncertainty in model selection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVcML_SdTheta(b_exp, b_out, Sig_inv_l, theta, zero_ind, r_vec = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVcML_SdTheta_+3A_b_exp">b_exp</code></td>
<td>
<p>A matrix of SNP effects on the exposure variable.</p>
</td></tr>
<tr><td><code id="MVcML_SdTheta_+3A_b_out">b_out</code></td>
<td>
<p>A vector of SNP effects on the outcome variable.</p>
</td></tr>
<tr><td><code id="MVcML_SdTheta_+3A_sig_inv_l">Sig_inv_l</code></td>
<td>
<p>A list of inverse of covariance matrix.</p>
</td></tr>
<tr><td><code id="MVcML_SdTheta_+3A_theta">theta</code></td>
<td>
<p>A vector of final estimates of causal effect of each exposure by MVMR-cML-BIC obtained from <code>MVmr_cML_DP</code>.</p>
</td></tr>
<tr><td><code id="MVcML_SdTheta_+3A_zero_ind">zero_ind</code></td>
<td>
<p>A vector of the index of valid IVs.</p>
</td></tr>
<tr><td><code id="MVcML_SdTheta_+3A_r_vec">r_vec</code></td>
<td>
<p>A vector of estimated horizontal pleiotropic effects.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector
</p>

<hr>
<h2 id='MVEgger-class'>MVEgger Class</h2><span id='topic+MVEgger-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable MR-Egger method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Model</code></dt><dd><p>Model always takes the value <code>random</code>, as only random-effects analyses are permitted.</p>
</dd>
<dt><code>Orientate</code></dt><dd><p>The number of the risk factor that genetic associations are orientated to. The default value is <code>1</code>, meaning that genetic associations with the first risk factor are set to be positive.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the inverse-variance weighted method.</p>
</dd>
<dt><code>StdError.Est</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower.Est</code></dt><dd><p>The lower bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper.Est</code></dt><dd><p>The upper bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Pvalue.Est</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>Intercept</code></dt><dd><p>The intercept estimate from the MR-Egger method. Under the InSIDE assumption, the intercept represents the average pleiotropic effect (average direct effect on the outcome) of a genetic variant. If the intercept differs from zero, this is evidence that the genetic variants are not all valid instruments; specifically, there is directional pleiotropy.</p>
</dd>
<dt><code>StdError.Int</code></dt><dd><p>The standard error associated with <code>Intercept</code>.</p>
</dd>
<dt><code>CILower.Int</code></dt><dd><p>The lower bound of the confidence interval for <code>Intercept</code> based on <code>StdError.Int</code>.</p>
</dd>
<dt><code>CIUpper.Int</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError.Int</code>.</p>
</dd>
<dt><code>Pvalue.Int</code></dt><dd><p>P-value associated with the intercept.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>RSE</code></dt><dd><p>The estimated residual standard error from the regression model.</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</dd>
</dl>

<hr>
<h2 id='MVGMM-class'>MVGMM Class</h2><span id='topic+MVGMM-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable generalized method of methods (GMM) method.
</p>


<h3>Slots</h3>


<dl>
<dt><code>robust</code></dt><dd><p>Whether the robust model with overdispersion heterogeneity is estimated.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants if specified. If not specified, an identity matrix will be returned.</p>
</dd>
<dt><code>ExpCorrelation</code></dt><dd><p>Whether an exposure correlation matrix was specified.</p>
</dd>
<dt><code>CondFstat</code></dt><dd><p>The conditional F-statistic for each exposure.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the GMM method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Overdispersion</code></dt><dd><p>The estimate of the overdispersion parameter for the robust model. If this is negative, then a value of zero is used in the method.</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimates.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing confidence intervals (default is 0.05).</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value (for non-robust model): the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more genetic variants may be pleiotropic.</p>
</dd>
</dl>

<hr>
<h2 id='MVIVW-class'>MVIVW Class</h2><span id='topic+MVIVW-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable inverse-variance weighted (IVW) method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Model</code></dt><dd><p>The model used for estimation: random-effects (<code>"random"</code>) or fixed-effect (<code>"fixed"</code>). The default option (<code>"default"</code>) is to use a fixed-effect model when there are three or fewer genetic variants, and a random-effects model when there are four or more. The (multiplicative) random-effects model allows for heterogeneity between the causal estimates targeted by the genetic variants by allowing over-dispersion in the regression model. Under-dispersion is not permitted (in case of under-dispersion, the residual standard error is set to 1, as in a fixed-effect analysis).</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Robust</code></dt><dd><p>Whether robust regression was used in the regression model relating the genetic associations with the outcome and those with the exposure.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the inverse-variance weighted method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>RSE</code></dt><dd><p>The estimated residual standard error from the regression model.</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</dd>
<dt><code>CondFstat</code></dt><dd><p>Conditional F statistics: An approximation of the first-stage conditional F statistics for all variants based on the summarized data. This represents the instrument strength for each exposure conditional on other exposures in the model. This is only reported if the sample sizes for the genetic associations with the exposures are provided.</p>
</dd>
</dl>

<hr>
<h2 id='MVIVWME-class'>MVIVWME Class</h2><span id='topic+MVIVWME-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable inverse-variance weighted (IVW) method with measurement error, as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Model</code></dt><dd><p>The model used for estimation: random-effects (<code>"random"</code>) or fixed-effect (<code>"fixed"</code>). The default option (<code>"default"</code>) is to use a fixed-effect model when there are three or fewer genetic variants, and a random-effects model when there are four or more. The (multiplicative) random-effects model allows for heterogeneity between the causal estimates targeted by the genetic variants by allowing over-dispersion in the regression model. Under-dispersion is not permitted (in case of under-dispersion, the residual standard error is set to 1, as in a fixed-effect analysis).</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the inverse-variance weighted method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>RSE</code></dt><dd><p>The estimated residual standard error from the regression model.</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value: the null hypothesis is that all genetic variants estimate the same causal parameter; rejection of the null is an indication that one or more variants may be pleiotropic.</p>
</dd>
</dl>

<hr>
<h2 id='MVLasso-class'>MRMVLasso class</h2><span id='topic+MVLasso-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable MR-Lasso method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Orientate</code></dt><dd><p>The number of the risk factor that genetic associations are orientated to. The default value is <code>1</code>, meaning that genetic associations with the first risk factor are set to be positive.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the multivariable MR-Lasso method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-values associated with the causal estimates from the multivariable MR-Lasso method.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs used in the calculation.</p>
</dd>
<dt><code>RegEstimate</code></dt><dd><p>The estimates from the regularized regression model used in the multivariable MR-Lasso method.</p>
</dd>
<dt><code>RegIntercept</code></dt><dd><p>The intercept estimates from the regularized regression model used in the multivariable MR-Lasso method. An intercept estimate of zero identifies the corresponding genetic variant as a valid instrument. Genetic variants with non-zero intercept estimates will be excluded from the MR-Lasso method, which is obtained as a post-lasso estimator.</p>
</dd>
<dt><code>Valid</code></dt><dd><p>The number of genetic variants that have been identified as valid instruments.</p>
</dd>
<dt><code>ValidSNPs</code></dt><dd><p>The names of genetic variants that have been identified as valid instruments.</p>
</dd>
<dt><code>Lambda</code></dt><dd><p>The value of the tuning parameter used to compute <code>RegEstimate</code> (default is to calulate <code>Lambda</code> using the heterogeneity stopping rule).</p>
</dd>
</dl>

<hr>
<h2 id='MVMedian-class'>MRMVMedian class</h2><span id='topic+MVMedian-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable median method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the multivariable median method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code> (obtained from bootstrapping).</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-values associated with the causal estimates from the Wald method.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that used in the calculation.</p>
</dd>
</dl>

<hr>
<h2 id='MVmr_cML'>MVMRcML method with Data Perturbation</h2><span id='topic+MVmr_cML'></span>

<h3>Description</h3>

<p>This is the internal MVMRcML-BIC function of mr_mvcML.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVmr_cML(
  b_exp,
  b_out,
  se_bx,
  Sig_inv_l,
  n,
  K_vec = as.numeric(c()),
  random_start = 1L,
  min_theta_range = -0.5,
  max_theta_range = 0.5,
  maxit = 100L,
  thres = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVmr_cML_+3A_b_exp">b_exp</code></td>
<td>
<p>A m*L matrix of SNP effects on the exposure variable.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_b_out">b_out</code></td>
<td>
<p>A m*1 matrix of SNP effects on the outcome variable.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_se_bx">se_bx</code></td>
<td>
<p>A m*L matrix of standard errors of <code>b_exp</code>.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_sig_inv_l">Sig_inv_l</code></td>
<td>
<p>A list of the inverse of m covariance matrices.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_n">n</code></td>
<td>
<p>The smallest sample size of the L+1 GWAS dataset.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_k_vec">K_vec</code></td>
<td>
<p>Sets of candidate K's, the constraint parameter representing number of invalid IVs.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_random_start">random_start</code></td>
<td>
<p>Number of random start points, default is 1.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_min_theta_range">min_theta_range</code></td>
<td>
<p>The lower bound of the uniform distribution for each initial value for theta generated from.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_max_theta_range">max_theta_range</code></td>
<td>
<p>The upper bound of the uniform distribution for each initial value for theta generated from.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations for each optimization, default is 100.</p>
</td></tr>
<tr><td><code id="MVmr_cML_+3A_thres">thres</code></td>
<td>
<p>Threshold for convergence criterion.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>

<dl>
<dt>BIC_theta</dt><dd><p>Estimated causal effect from MVMR-cML-BIC</p>
</dd>
<dt>BIC_invalid</dt><dd><p>Invalid IVs selected by MVMR-cML-BIC</p>
</dd>
<dt>l_vec</dt><dd><p>A vector of negative log-likelihood corresponding to each <code>K</code>.</p>
</dd>
<dt>K_vec</dt><dd><p>A vector of candidate K's</p>
</dd>
<dt>theta_vec</dt><dd><p>A matrix of causal parameter estimates, each column corresponds to a candidate <code>K</code>.</p>
</dd>
<dt>Conv_vec</dt><dd><p>A vector of successful convergence indicators corresponding to each <code>K</code>.</p>
</dd>
<dt>Converge</dt><dd><p>Indicator of successful convergence, 0 means success, 1 means failure.</p>
</dd>
<dt>BIC_vec</dt><dd><p>Data perturbation with successful convergence</p>
</dd>
<dt>Khat</dt><dd><p>The length of <code>BIC_invalid</code>.</p>
</dd>
</dl>


<hr>
<h2 id='MVmr_cML_DP'>MVMRcML method with Data Perturbation</h2><span id='topic+MVmr_cML_DP'></span>

<h3>Description</h3>

<p>This is the internal MVMRcML-DP function of mr_mvcML.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MVmr_cML_DP(
  b_exp,
  b_out,
  se_bx,
  Sig_inv_l,
  n,
  K_vec = as.numeric(c()),
  random_start = 1L,
  num_pert = 100L,
  min_theta_range = -0.5,
  max_theta_range = 0.5,
  maxit = 100L,
  thres = 1e-04
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MVmr_cML_DP_+3A_b_exp">b_exp</code></td>
<td>
<p>A m*L matrix of SNP effects on the exposure variable.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_b_out">b_out</code></td>
<td>
<p>A m*1 matrix of SNP effects on the outcome variable.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_se_bx">se_bx</code></td>
<td>
<p>A m*L matrix of standard errors of <code>b_exp</code>.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_sig_inv_l">Sig_inv_l</code></td>
<td>
<p>A list of the inverse of m covariance matrices.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_n">n</code></td>
<td>
<p>The smallest sample size of the L+1 GWAS dataset.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_k_vec">K_vec</code></td>
<td>
<p>Sets of candidate K's, the constraint parameter representing number of invalid IVs.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_random_start">random_start</code></td>
<td>
<p>Number of random start points, default is 1.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_num_pert">num_pert</code></td>
<td>
<p>Number of perturbation, default is 100.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_min_theta_range">min_theta_range</code></td>
<td>
<p>The lower bound of the uniform distribution for each initial value for theta generated from.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_max_theta_range">max_theta_range</code></td>
<td>
<p>The upper bound of the uniform distribution for each initial value for theta generated from.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_maxit">maxit</code></td>
<td>
<p>Maximum number of iterations for each optimization, default is 100.</p>
</td></tr>
<tr><td><code id="MVmr_cML_DP_+3A_thres">thres</code></td>
<td>
<p>Threshold for convergence criterion.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list
</p>

<dl>
<dt>BIC_theta</dt><dd><p>Estimated causal effect from MVMR-cML-BIC</p>
</dd>
<dt>BIC_invalid</dt><dd><p>Invalid IVs selected by MVMR-cML-BIC</p>
</dd>
<dt>BIC_DP_theta</dt><dd><p>Estimated causal effect from MVMR-cML-DP </p>
</dd>
<dt>BIC_DP_se</dt><dd><p>Estimate standard error for <code>BIC_DP_theta</code></p>
</dd>
<dt>eff_DP_B</dt><dd><p>Data perturbation with successful convergence</p>
</dd>
<dt>DP_ninvalid</dt><dd><p>A vector of the number of selected invalid IVs by MVMRcML-BIC in each data perturbation.</p>
</dd>
</dl>


<hr>
<h2 id='MVMRcML-class'>MVMRcML Class</h2><span id='topic+MVMRcML-class'></span>

<h3>Description</h3>

<p>An object containing the results of MVMRcML.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the multivariable MRcML method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence intervals for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-values associated with the causal estimates from the multivariable MRcML method.</p>
</dd>
<dt><code>BIC_invalid</code></dt><dd><p>The index set of selected invalid IVs by MVMRcML-BIC.</p>
</dd>
<dt><code>K_hat</code></dt><dd><p>The number of selected invalid IVs by MVMRcML-BIC, or a vector for each data perturbation in MVMRcML-DP.</p>
</dd>
<dt><code>eff_DP_B</code></dt><dd><p>The number of data perturbations with successful convergence in MVMRcML-DP.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that were used in the calculation.</p>
</dd>
<dt><code>DP</code></dt><dd><p>Indicator of whether data perturbation is applied.</p>
</dd>
</dl>

<hr>
<h2 id='MVPCGMM-class'>MVPCGMM Class</h2><span id='topic+MVPCGMM-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the multivariable principal components generalized method of methods (PC-GMM) method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>robust</code></dt><dd><p>Whether the robust model with overdispersion heterogeneity is estimated.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The names of the exposure variables.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>ExpCorrelation</code></dt><dd><p>Whether an exposure correlation matrix was specified.</p>
</dd>
<dt><code>CondFstat</code></dt><dd><p>The conditional F-statistic for each exposure.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimates from the PC-GMM method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard errors associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bounds of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Overdispersion</code></dt><dd><p>The estimate of the overdispersion parameter for the robust model.</p>
</dd>
<dt><code>PCs</code></dt><dd><p>The number of genetic principal components used to instrument the exposures.</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimates.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing confidence intervals (default is 0.05).</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value (for non-robust model): the null hypothesis is that all principal components estimate the same causal parameter; rejection of the null is an indication that one or more principal components may be pleiotropic.</p>
</dd>
</dl>

<hr>
<h2 id='PCGMM-class'>PCGMM Class</h2><span id='topic+PCGMM-class'></span>

<h3>Description</h3>

<p>An object containing the estimates produced using the univariable principal components generalized method of methods (PC-GMM) method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>robust</code></dt><dd><p>Whether the robust model with overdispersion heterogeneity is estimated.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Correlation</code></dt><dd><p>The matrix of correlations between genetic variants.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal estimate from the PC-GMM method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Fstat</code></dt><dd><p>The first-stage F statistic for all genetic principal components used as instruments.</p>
</dd>
<dt><code>Overdispersion</code></dt><dd><p>The estimate of the overdispersion parameter for the robust model.</p>
</dd>
<dt><code>PCs</code></dt><dd><p>The number of genetic principal components used to instrument the exposure.</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Heter.Stat</code></dt><dd><p>Heterogeneity statistic (Cochran's Q statistic) and associated p-value (for non-robust model): the null hypothesis is that all principal components estimate the same causal parameter; rejection of the null is an indication that one or more principal components may be pleiotropic.</p>
</dd>
</dl>

<hr>
<h2 id='penalised.weights'>Calculates p-values for penalization of weights</h2><span id='topic+penalised.weights'></span>

<h3>Description</h3>

<p>Internal function for calculating penalized weights in conjunction with <code>r.weights</code>.
</p>
<p>These weights are used in either the <code>mr_ivw</code> or <code>mr_egger</code> functions when <code>penalized = TRUE</code>, or in the <code>mr_median</code> function when <code>method = "penalized"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penalised.weights(.Bx, .Bxse, .By, .Byse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penalised.weights_+3A_.bx">.Bx</code></td>
<td>
<p>Beta-coefficient for genetic association with the risk factor.</p>
</td></tr>
<tr><td><code id="penalised.weights_+3A_.bxse">.Bxse</code></td>
<td>
<p>Standard error of genetic association with the risk factor.</p>
</td></tr>
<tr><td><code id="penalised.weights_+3A_.by">.By</code></td>
<td>
<p>Beta-coefficient for genetic association with the outcome.</p>
</td></tr>
<tr><td><code id="penalised.weights_+3A_.byse">.Byse</code></td>
<td>
<p>Standard error of genetic association with the outcome.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>P-values corresponding to heterogeneity test for each genetic variant in turn (based on a chi-squared-1 distribution).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>penalised.weights(ldlc, ldlcse, chdlodds, chdloddsse)

</code></pre>

<hr>
<h2 id='penalised.weights.delta'>Calculates p-values for penalization of weights with second-order weights</h2><span id='topic+penalised.weights.delta'></span>

<h3>Description</h3>

<p>Internal function for calculating penalized weights in conjunction with <code>r.weights</code> with <code>weights</code> equal to <code>"delta"</code>.
</p>
<p>These weights are used in either the <code>mr_ivw</code> or <code>mr_egger</code> functions when <code>penalized = TRUE</code>, or in the <code>mr_median</code> function when <code>method = "penalized"</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>penalised.weights.delta(.Bx, .Bxse, .By, .Byse, .psi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="penalised.weights.delta_+3A_.bx">.Bx</code></td>
<td>
<p>Beta-coefficient for genetic association with the risk factor.</p>
</td></tr>
<tr><td><code id="penalised.weights.delta_+3A_.bxse">.Bxse</code></td>
<td>
<p>Standard error of genetic association with the risk factor.</p>
</td></tr>
<tr><td><code id="penalised.weights.delta_+3A_.by">.By</code></td>
<td>
<p>Beta-coefficient for genetic association with the outcome.</p>
</td></tr>
<tr><td><code id="penalised.weights.delta_+3A_.byse">.Byse</code></td>
<td>
<p>Standard error of genetic association with the outcome.</p>
</td></tr>
<tr><td><code id="penalised.weights.delta_+3A_.psi">.psi</code></td>
<td>
<p>The correlation between the genetic associations with the exposure and the association with the outcome for each variant resulting from sample overlap.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>P-values corresponding to heterogeneity test for each genetic variant in turn (based on a chi-squared-1 distribution).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>penalised.weights.delta(ldlc, ldlcse, chdlodds, chdloddsse, 0)

</code></pre>

<hr>
<h2 id='pheno_input'>Extract summarized data from PhenoScanner</h2><span id='topic+pheno_input'></span>

<h3>Description</h3>

<p>The function <code>pheno_input</code> extracts summarized data on associations with named exposure and outcome variables from PhenoScanner.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pheno_input(
  snps,
  exposure,
  pmidE,
  ancestryE,
  outcome,
  pmidO,
  ancestryO,
  correl = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pheno_input_+3A_snps">snps</code></td>
<td>
<p>The names (rsid) of the genetic variants to be included in the analysis.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_exposure">exposure</code></td>
<td>
<p>The name of the exposure variable.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_pmide">pmidE</code></td>
<td>
<p>The PubMed ID (PMID) of the publication in which the genetic association estimates with the exposure were originally reported. Some variables are reported in multiple consortia (for example, associations with coronary artery disease by CARDIoGRAM in 2011 [PMID:21378990], by CARDIoGRAMplusC4D in 2013, and again by CARDIoGRAMplusC4D in 2015 [PMID:26343387]). Equally, some publications reported associations on multiple variables (for example, CARDIoGRAMplusC4D in 2015 [PMID:26343387] reported associations with coronary artery disease and with myocardial infarction). By providing the variable name and the PubMed ID, the set of associations is (almost) uniquely identified.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_ancestrye">ancestryE</code></td>
<td>
<p>The ancestry of individuals in which estimates were obtained. A small number of studies reported genetic association estimates for a single variable in a single publication for multiple ethnicities (for example, associations with log(eGFR creatinine) from CKD-Gen in 2016 [PMID:26831199] were reported for both Europeans and Africans). The combination of exposure name, PubMed ID, and ancestry uniquely defines the set of associations. Providing the ancestry also reminds analysts of the additional complication of conducting Mendelian randomization when associations with the exposure and with the outcome are in individuals of different ancestry. Most association estimates are obtained in <code>"European"</code> or <code>"Mixed"</code> populations, although some are obtained in <code>"African"</code>, <code>"Asian"</code>, or <code>"Hispanic"</code> populations.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_outcome">outcome</code></td>
<td>
<p>The name of the outcome variable.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_pmido">pmidO</code></td>
<td>
<p>The PubMed ID of the publication in which the genetic association estimates with the outcome were originally reported.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_ancestryo">ancestryO</code></td>
<td>
<p>The ancestry of individuals in which genetic association estimates with the outcome were obtained.</p>
</td></tr>
<tr><td><code id="pheno_input_+3A_correl">correl</code></td>
<td>
<p>The correlations between the genetic variants. If this is not specified, then the genetic variants are assumed to be uncorrelated. Note that for the correlations to reference the correct variants, the list of genetic variants needs to be in alphabetical order.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PhenoScanner bioinformatic tool is a curated database of publicly available results from large-scale genetic association studies. Queries can be made for individual genetic variants (SNPs and small indels), or for multiple variants in a single batch query. These association estimates and their standard errors can be used in Mendelian randomization analyses.
</p>
<p>The <code>phenoscanner</code> command is included in the <code>MendelianRandomization</code> package with permission of James Staley. The function is also available in a standalone package from github: <a href="https://github.com/phenoscanner/phenoscanner">https://github.com/phenoscanner/phenoscanner</a>.
</p>


<h3>Value</h3>

<p>The output of the <code>pheno_input</code> function is an <code>MRInput</code> object that can be used directly in any of the estimation functions (such as <code>mr_ivw</code>) or in the plotting function <code>mr_plot</code>. The output contains:
</p>
<table>
<tr><td><code>bx</code></td>
<td>
<p>The genetic associations with the exposure.</p>
</td></tr>
<tr><td><code>bxse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td></tr>
<tr><td><code>by</code></td>
<td>
<p>The genetic associations with the outcome.</p>
</td></tr>
<tr><td><code>byse</code></td>
<td>
<p>The corresponding standard errors.</p>
</td></tr>
<tr><td><code>correlation</code></td>
<td>
<p>The matrix of genetic correlations as specified by the user.</p>
</td></tr>
<tr><td><code>exposure</code></td>
<td>
<p>A character string giving the name of the exposure as provided in the PhenoScanner database.</p>
</td></tr>
<tr><td><code>outcome</code></td>
<td>
<p>A character string giving the name of the outcome as provided in the PhenoScanner database.</p>
</td></tr>
<tr><td><code>snps</code></td>
<td>
<p>A vector of character strings with the names of the genetic variants.</p>
</td></tr>
</table>


<h3>References</h3>

<p>James R Staley, James Blackshow, Mihir A Kamat, Steve Ellis, Prvaeen Surendran, Benjamin B Sun, Dirk S Paul, Daniel Freitag, Stephen Burgess, John Danesh, Robin Young, and Adam S Butterworth. PhenoScanner: a database of human genotype&ndash;phenotype associations. Bioinformatics 2016. doi: 10.1093/bioinformatics/btw373.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># pheno_input(snps=c("rs12916", "rs2479409", "rs217434", "rs1367117"),
# exposure = "Low density lipoprotein", pmidE = "24097068", ancestryE = "European",
# outcome = "Coronary artery disease", pmidO = "26343387", ancestryO = "Mixed")

</code></pre>

<hr>
<h2 id='phenoscanner'>PhenoScanner</h2><span id='topic+phenoscanner'></span>

<h3>Description</h3>

<p>The <code>phenoscanner</code> function queries the PhenoScanner database of genotype-phenotype associations from inside R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phenoscanner(
  snpquery = NULL,
  genequery = NULL,
  regionquery = NULL,
  catalogue = "GWAS",
  pvalue = 1e-05,
  proxies = "None",
  r2 = 0.8,
  build = 37
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phenoscanner_+3A_snpquery">snpquery</code></td>
<td>
<p>a vector of SNPs.</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_genequery">genequery</code></td>
<td>
<p>a vector of gene names.</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_regionquery">regionquery</code></td>
<td>
<p>a vector of genomic regions.</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_catalogue">catalogue</code></td>
<td>
<p>the catalogue to be searched (options: None, GWAS, eQTL, pQTl, mQTL, methQTL).</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_pvalue">pvalue</code></td>
<td>
<p>the p-value threshold.</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_proxies">proxies</code></td>
<td>
<p>the proxies database to be searched (options: None, AFR, AMR, EAS, EUR, SAS).</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_r2">r2</code></td>
<td>
<p>the r2 threshold.</p>
</td></tr>
<tr><td><code id="phenoscanner_+3A_build">build</code></td>
<td>
<p>the genome build (options: 37, 38).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list containing a data.frame of association results and a data.frame of SNP/Region/Gene information from PhenoScanner.
</p>


<h3>Author(s)</h3>

<p>PhenoScanner &lt;phenoscanner@gmail.com&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'># SNP
# res &lt;- phenoscanner(snpquery="rs10840293")
# head(res$results)
# res$snps

# Gene
# res &lt;- phenoscanner(genequery="SWAP70")
# head(res$results)
# res$snps

# Region
# res &lt;- phenoscanner(regionquery="chr11:9685624-9774538")
# head(res$results)
# res$regions
</code></pre>

<hr>
<h2 id='PIVW-class'>PIVW Class</h2><span id='topic+PIVW-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the penalized inverse-variance weighted (pIVW) method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Over.dispersion</code></dt><dd><p>Should the method consider overdispersion (balanced horizontal pleiotropy)? Default is TRUE.</p>
</dd>
<dt><code>Boot.Fieller</code></dt><dd><p>If <code>Boot.Fieller=TRUE</code>, then the P-value and the confidence interval of the causal effect will be calculated based on the bootstrapping Fieller method. Otherwise, the P-value and the confidence interval of the causal effect will be calculated from the normal distribution. It is recommended to use the bootstrapping Fieller method when <code>Condition</code> (the estimated effective sample size) is smaller than 10. By default, <code>Boot.Fieller=TRUE</code>.</p>
</dd>
<dt><code>Lambda</code></dt><dd><p>The penalty parameter in the pIVW estimator. The penalty parameter plays a role in the bias-variance trade-off of the estimator. It is recommended to choose <code>lambda=1</code> to achieve the smallest bias and valid inference. By default, <code>lambda=1</code>.</p>
</dd>
<dt><code>Delta</code></dt><dd><p>The z-score threshold for IV selection. By default, <code>delta=0</code> (i.e., no IV selection will be conducted).</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal point estimate from the pIVW estimator.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code>.</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code>, which is derived from the bootstrapping Fieller method or normal distribution. For the bootstrapping Fieller's interval, if it contains multiple ranges, then lower limits of all ranges will be reported.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code>, which is derived from the bootstrapping Fieller method or normal distribution. For the bootstrapping Fieller's interval, if it contains multiple ranges, then upper limits of all ranges will be reported.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate from the pIVW estimator.</p>
</dd>
<dt><code>Tau2</code></dt><dd><p>The variance of the balanced horizontal pleiotropy. <code>Tau2</code> is calculated by using all IVs in the data before conducting the IV selection.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs after IV selection.</p>
</dd>
<dt><code>Condition</code></dt><dd><p>The estimated effective sample size. It is recommended to be greater than 5 for the pIVW estimator to achieve reliable asymptotic properties.</p>
</dd>
</dl>

<hr>
<h2 id='pl'>Profile likelihood of valid IVs</h2><span id='topic+pl'></span>

<h3>Description</h3>

<p>calculate the profile likelihood of valid IVs up to a constant
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pl(x, b_exp_v, b_out_v, Sig_inv_v)
</code></pre>

<hr>
<h2 id='r.weights'>Calculates p-values for penalization of weights</h2><span id='topic+r.weights'></span>

<h3>Description</h3>

<p>Internal function for calculating penalized weights in conjunction with <code>penalised.weights</code>.
</p>
<p>These weights are used in either the <code>mr_ivw</code> or <code>mr_egger</code> functions when <code>penalized = TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r.weights(.Byse, .pen.weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="r.weights_+3A_.byse">.Byse</code></td>
<td>
<p>Standard error of genetic association with the outcome.</p>
</td></tr>
<tr><td><code id="r.weights_+3A_.pen.weights">.pen.weights</code></td>
<td>
<p>Factors for penalizing weights.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Penalized weights.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>r.weights(chdloddsse, penalised.weights(ldlc, ldlcse, chdlodds, chdloddsse))

</code></pre>

<hr>
<h2 id='r.weights.delta'>Calculates p-values for penalization of weights with second-order weights</h2><span id='topic+r.weights.delta'></span>

<h3>Description</h3>

<p>Internal function for calculating penalized weights in conjunction with <code>penalised.weights</code> with <code>weights</code> equal to <code>"delta"</code>.
</p>
<p>These weights are used in either the <code>mr_ivw</code> or <code>mr_egger</code> functions when <code>penalized = TRUE</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>r.weights.delta(.Bx, .Bxse, .By, .Byse, .psi, .pen.weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="r.weights.delta_+3A_.bx">.Bx</code></td>
<td>
<p>Beta-coefficient for genetic association with the risk factor.</p>
</td></tr>
<tr><td><code id="r.weights.delta_+3A_.bxse">.Bxse</code></td>
<td>
<p>Standard error of genetic association with the risk factor.</p>
</td></tr>
<tr><td><code id="r.weights.delta_+3A_.by">.By</code></td>
<td>
<p>Beta-coefficient for genetic association with the outcome.</p>
</td></tr>
<tr><td><code id="r.weights.delta_+3A_.byse">.Byse</code></td>
<td>
<p>Standard error of genetic association with the outcome.</p>
</td></tr>
<tr><td><code id="r.weights.delta_+3A_.psi">.psi</code></td>
<td>
<p>The correlation between the genetic associations with the exposure and the association with the outcome for each variant resulting from sample overlap.</p>
</td></tr>
<tr><td><code id="r.weights.delta_+3A_.pen.weights">.pen.weights</code></td>
<td>
<p>Factors for penalizing weights.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Penalized weights.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>r.weights.delta(ldlc, ldlcse, chdlodds, chdloddsse, 0,
  penalised.weights.delta(ldlc, ldlcse, chdlodds, chdloddsse, 0))

</code></pre>

<hr>
<h2 id='simpleCap'>Capitalize a word</h2><span id='topic+simpleCap'></span>

<h3>Description</h3>

<p>Internal function for capitalizing a word.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simpleCap(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simpleCap_+3A_x">x</code></td>
<td>
<p>A word.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Returns a capitalized word.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>simpleCap(x = "weighted")

</code></pre>

<hr>
<h2 id='values'>Applies method values() to different output classes</h2><span id='topic+values'></span><span id='topic+values+2CWeightedMedian-method'></span><span id='topic+values+2CIVW-method'></span><span id='topic+values+2CEgger-method'></span>

<h3>Description</h3>

<p>Enables the internal function <code>values</code>, used in the <code>mr_allmethods</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S4 method for signature 'WeightedMedian'
values(object)

## S4 method for signature 'IVW'
values(object)

## S4 method for signature 'Egger'
values(object)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="values_+3A_object">object</code></td>
<td>
<p>Object (could be an object of class &quot;WeightedMedian&quot;, &quot;Egger&quot;, or &quot;IVW&quot;).</p>
</td></tr>
</table>

<hr>
<h2 id='weighted.median'>Weighted median function</h2><span id='topic+weighted.median'></span>

<h3>Description</h3>

<p>Internal function for calculating weighted median estimate (or simple median estimate if weights are all equal).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weighted.median(theta, weights)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weighted.median_+3A_theta">theta</code></td>
<td>
<p>Ratio causal estimates for each genetic variant considered individually.</p>
</td></tr>
<tr><td><code id="weighted.median_+3A_weights">weights</code></td>
<td>
<p>Weights.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Causal estimate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>weighted.median(theta = chdlodds/ldlc, weights = abs(chdloddsse/ldlc))

</code></pre>

<hr>
<h2 id='weighted.median.boot.se'>Weighted median standard error function</h2><span id='topic+weighted.median.boot.se'></span>

<h3>Description</h3>

<p>Internal function for calculating standard error of weighted median estimate (or simple median estimator if weights are all equal) using bootstrapping. The number of iterations and initial value of the random seed can also be set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>weighted.median.boot.se(Bx, By, Bxse, Byse, weights, iter, seed)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weighted.median.boot.se_+3A_bx">Bx</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the exposure.</p>
</td></tr>
<tr><td><code id="weighted.median.boot.se_+3A_by">By</code></td>
<td>
<p>A numeric vector of beta-coefficient values for genetic associations with the outcome.</p>
</td></tr>
<tr><td><code id="weighted.median.boot.se_+3A_bxse">Bxse</code></td>
<td>
<p>The standard errors associated with the beta-coefficients in <code>Bx</code>.</p>
</td></tr>
<tr><td><code id="weighted.median.boot.se_+3A_byse">Byse</code></td>
<td>
<p>The standard errors associated with the beta-coefficients in <code>By</code>.</p>
</td></tr>
<tr><td><code id="weighted.median.boot.se_+3A_weights">weights</code></td>
<td>
<p>Weights.</p>
</td></tr>
<tr><td><code id="weighted.median.boot.se_+3A_iter">iter</code></td>
<td>
<p>The number of bootstrap samples to generate when calculating the standard error.</p>
</td></tr>
<tr><td><code id="weighted.median.boot.se_+3A_seed">seed</code></td>
<td>
<p>The random seed to use when generating the bootstrap samples (for reproducibility). If set to <code>NA</code>, the random seed will not be set (for example, if the function is used as part of a larger simulation).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>None.
</p>


<h3>Value</h3>

<p>Causal estimate.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>weighted.median.boot.se(Bx = ldlc, By = chdlodds, Bxse = ldlcse, Byse = chdloddsse,
weights = chdloddsse, iter = 100, seed = 314)

</code></pre>

<hr>
<h2 id='WeightedMedian-class'>WeightedMedian Class</h2><span id='topic+WeightedMedian-class'></span>

<h3>Description</h3>

<p>An object containing the estimate produced using the median-based method as well as various statistics.
</p>


<h3>Slots</h3>


<dl>
<dt><code>Type</code></dt><dd><p>The type of median that has been calculated, <code>"simple"</code>, <code>"weighted"</code>, or <code>"penalized"</code>.</p>
</dd>
<dt><code>Exposure</code></dt><dd><p>The name of the exposure variable.</p>
</dd>
<dt><code>Outcome</code></dt><dd><p>The name of the outcome variable.</p>
</dd>
<dt><code>Estimate</code></dt><dd><p>The causal point estimate from the median-based method.</p>
</dd>
<dt><code>StdError</code></dt><dd><p>The standard error associated with <code>Estimate</code> (obtained from bootstrapping).</p>
</dd>
<dt><code>CILower</code></dt><dd><p>The lower bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>CIUpper</code></dt><dd><p>The upper bound of the confidence interval for <code>Estimate</code> based on <code>StdError</code>.</p>
</dd>
<dt><code>Alpha</code></dt><dd><p>The significance level used in constructing the confidence interval (default is 0.05).</p>
</dd>
<dt><code>Pvalue</code></dt><dd><p>P-value associated with the causal estimate from the Wald method.</p>
</dd>
<dt><code>SNPs</code></dt><dd><p>The number of SNPs that used in the calculation.</p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
