<!DOCTYPE html><html lang="en"><head><title>Help for package MetaComp</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MetaComp}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#load_edge_assignment'><p>Efficiently loads an EDGE-produced taxonomic assignment from a file.</p>
An assumption has been made &ndash; since EDGE tables are generated in an automated fashion,
they should be properly formatted &ndash; thus the code doesn't check for any inconsistencies except
for the very file existence. Note however, the unassigned to taxa entries are removed.
This implementation fully relies on the fread function from data.table package
gaining performance over traditional R techniques.</a></li>
<li><a href='#load_edge_assignments'><p>Efficiently loads a BWA (or other EDGE-like taxonomic assignment) tables from a list</p>
of files. Outputs a named list of assignments.</a></li>
<li><a href='#merge_edge_assignments'><p>Merges two or more EDGE-like taxonomical assignments. The input data frames are assumed to</p>
have the following columns: LEVEL, TAXA, and ABUNDANCE &ndash; these will be used in the merge
procedure, all other columns will be ignored.</a></li>
<li><a href='#merge_edge_counts'><p>Merges two or more EDGE-like taxonomical assignments. The input data frames are assumed to</p>
have the following columns: LEVEL, TAXA, and COUNT &ndash; these will be used in the merge
procedure, all other columns will be ignored.</a></li>
<li><a href='#plot_edge_assignment'><p>Generates a single column ggplot for a taxonomic assignment table and also outputs a PDF.</p></a></li>
<li><a href='#plot_merged_assignment'><p>Generates a single column ggplot for a taxonomic assignment table.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.1.2</td>
</tr>
<tr>
<td>Title:</td>
<td>EDGE Taxonomy Assignments Visualization</td>
</tr>
<tr>
<td>Description:</td>
<td>Implements routines for metagenome sample taxonomy assignments collection, 
    aggregation, and visualization. Accepts the EDGE-formatted output from GOTTCHA/GOTTCHA2, 
    BWA, Kraken, MetaPhlAn, DIAMOND, and Pangia. Produces SVG and PDF heatmap-like plots 
    comparing taxa abundances across projects. </td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/seninp-bioinfo/MetaComp">https://github.com/seninp-bioinfo/MetaComp</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/seninp-bioinfo/MetaComp/issues">https://github.com/seninp-bioinfo/MetaComp/issues</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>reshape2, plyr, dplyr, data.table, ggplot2, Cairo</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-06-18 19:33:19 UTC; psenin</td>
</tr>
<tr>
<td>Author:</td>
<td>Pavel Senin [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Pavel Senin &lt;senin@hawaii.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-06-18 20:06:45 UTC</td>
</tr>
</table>
<hr>
<h2 id='load_edge_assignment'>Efficiently loads an EDGE-produced taxonomic assignment from a file.
An assumption has been made &ndash; since EDGE tables are generated in an automated fashion,
they should be properly formatted &ndash; thus the code doesn't check for any inconsistencies except
for the very file existence. Note however, the unassigned to taxa entries are removed.
This implementation fully relies on the fread function from data.table package
gaining performance over traditional R techniques.</h2><span id='topic+load_edge_assignment'></span>

<h3>Description</h3>

<p>Efficiently loads an EDGE-produced taxonomic assignment from a file.
An assumption has been made &ndash; since EDGE tables are generated in an automated fashion,
they should be properly formatted &ndash; thus the code doesn't check for any inconsistencies except
for the very file existence. Note however, the unassigned to taxa entries are removed.
This implementation fully relies on the fread function from data.table package
gaining performance over traditional R techniques.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_edge_assignment(filepath, type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_edge_assignment_+3A_filepath">filepath</code></td>
<td>
<p>the path to EDGE-generated tab-delimited taxonomy assignment file.</p>
</td></tr>
<tr><td><code id="load_edge_assignment_+3A_type">type</code></td>
<td>
<p>the assignment type. Following types are recognized: 'bwa', 'diamond',
'gottcha', 'gottcha2', 'kraken', 'metaphlan', and 'pangia'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame containing four columns: TAXA, LEVEL, COUNT, and ABUNDANCE, representing
taxonomically anchored sequences from the sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pa_fpath &lt;- system.file("extdata", "HMP_even//allReads-pangia.list.txt", package="MetaComp")
pangia_assignment = load_edge_assignment(pa_fpath, type = "pangia")

table(pangia_assignment$LEVEL)

pangia_assignment[pangia_assignment$LEVEL == "phylum",]

</code></pre>

<hr>
<h2 id='load_edge_assignments'>Efficiently loads a BWA (or other EDGE-like taxonomic assignment) tables from a list
of files. Outputs a named list of assignments.</h2><span id='topic+load_edge_assignments'></span>

<h3>Description</h3>

<p>Efficiently loads a BWA (or other EDGE-like taxonomic assignment) tables from a list
of files. Outputs a named list of assignments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_edge_assignments(filepath, type)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_edge_assignments_+3A_filepath">filepath</code></td>
<td>
<p>the path to tab delimited, two-column file whose first column is a project_id
(which will be used to name this assignment) and the second column is the assignment filename.</p>
</td></tr>
<tr><td><code id="load_edge_assignments_+3A_type">type</code></td>
<td>
<p>the type of assignments to be loaded. Following types are recognized: 'bwa',
'diamond', 'gottcha', 'gottcha2', 'kraken', 'metaphlan', and 'pangia'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of all read assignments.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hmp_even_fp &lt;- system.file("extdata", "HMP_even", package="MetaComp")
hmp_stagger_fp &lt;- system.file("extdata", "HMP_stagger", package="MetaComp")
data_files &lt;- data.frame(V1 = c("HMP_even", "HMP_stagger"),
                         V2 = c(file.path(hmp_even_fp, "allReads-gottcha2-speDB-b.list.txt"),
                                file.path(hmp_stagger_fp, "allReads-gottcha2-speDB-b.list.txt")))
write.table(data_files, file.path(tempdir(), "assignments.txt"),
                                                 row.names = FALSE, col.names = FALSE)
gottcha2_assignments = load_edge_assignments(file.path(tempdir(), "assignments.txt"),
                                                                            type = "gottcha2")

names(gottcha2_assignments)
table(gottcha2_assignments[[1]]$LEVEL)

</code></pre>

<hr>
<h2 id='merge_edge_assignments'>Merges two or more EDGE-like taxonomical assignments. The input data frames are assumed to
have the following columns: LEVEL, TAXA, and ABUNDANCE &ndash; these will be used in the merge
procedure, all other columns will be ignored.</h2><span id='topic+merge_edge_assignments'></span>

<h3>Description</h3>

<p>Merges two or more EDGE-like taxonomical assignments. The input data frames are assumed to
have the following columns: LEVEL, TAXA, and ABUNDANCE &ndash; these will be used in the merge
procedure, all other columns will be ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_edge_assignments(assignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_edge_assignments_+3A_assignments">assignments</code></td>
<td>
<p>A named list of assignments (the list element's name will be used as a
resulting data frame column name).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A merged table, which is a data frame whose rows are taxonomical ids and columns are
the input assignments ids.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
hmp_even_fp &lt;- system.file("extdata", "HMP_even", package="MetaComp")
hmp_stagger_fp &lt;- system.file("extdata", "HMP_stagger", package="MetaComp")
data_files &lt;- data.frame(V1 = c("HMP_even", "HMP_stagger"),
                         V2 = c(file.path(hmp_even_fp, "allReads-gottcha2-speDB-b.list.txt"),
                                file.path(hmp_stagger_fp, "allReads-gottcha2-speDB-b.list.txt")))
write.table(data_files, file.path(tempdir(), "assignments.txt"),
                                                 row.names = FALSE, col.names = FALSE)
gottcha2_assignments = merge_edge_assignments(
                         load_edge_assignments(
                           file.path(tempdir(), "assignments.txt"), type = "gottcha2"))

## End(Not run)

</code></pre>

<hr>
<h2 id='merge_edge_counts'>Merges two or more EDGE-like taxonomical assignments. The input data frames are assumed to
have the following columns: LEVEL, TAXA, and COUNT &ndash; these will be used in the merge
procedure, all other columns will be ignored.</h2><span id='topic+merge_edge_counts'></span>

<h3>Description</h3>

<p>Merges two or more EDGE-like taxonomical assignments. The input data frames are assumed to
have the following columns: LEVEL, TAXA, and COUNT &ndash; these will be used in the merge
procedure, all other columns will be ignored.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_edge_counts(assignments)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="merge_edge_counts_+3A_assignments">assignments</code></td>
<td>
<p>A named list of assignments (the list element's name will be used as a
resulting data frame column name).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A merged table, which is a data frame whose rows are taxonomical ids and columns are
the input assignments ids.
</p>

<hr>
<h2 id='plot_edge_assignment'>Generates a single column ggplot for a taxonomic assignment table and also outputs a PDF.</h2><span id='topic+plot_edge_assignment'></span>

<h3>Description</h3>

<p>This implementation is built upon ggplot geom_tile.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_edge_assignment(assignment, level, plot_title, column_title, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_edge_assignment_+3A_assignment">assignment</code></td>
<td>
<p>The EDGE-like assignment table.</p>
</td></tr>
<tr><td><code id="plot_edge_assignment_+3A_level">level</code></td>
<td>
<p>The taxonomic level to plot (i.e., family, strain, etc...).</p>
</td></tr>
<tr><td><code id="plot_edge_assignment_+3A_plot_title">plot_title</code></td>
<td>
<p>The plot title, e.g., &quot;Project XX, Run YY&quot;.</p>
</td></tr>
<tr><td><code id="plot_edge_assignment_+3A_column_title">column_title</code></td>
<td>
<p>The column title.</p>
</td></tr>
<tr><td><code id="plot_edge_assignment_+3A_filename">filename</code></td>
<td>
<p>The PDF file name mask.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the ggplot2 plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>pa_fpath &lt;- system.file("extdata", "HMP_even//allReads-pangia.list.txt", package="MetaComp")
pangia_assignment = load_edge_assignment(pa_fpath, type = "pangia")

plot_edge_assignment(pangia_assignment, "phylum", "Pangia", "HMP Even",
                                                     file.path(tempdir(), "assignment.pdf"))

</code></pre>

<hr>
<h2 id='plot_merged_assignment'>Generates a single column ggplot for a taxonomic assignment table.</h2><span id='topic+plot_merged_assignment'></span>

<h3>Description</h3>

<p>This implementation...
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_merged_assignment(assignment, taxonomy_level,
  sorting_order = "abundance", row_limit = 60, min_row_abundance = 0,
  plot_title, filename)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_merged_assignment_+3A_assignment">assignment</code></td>
<td>
<p>The gottcha-like merged assignment table.</p>
</td></tr>
<tr><td><code id="plot_merged_assignment_+3A_taxonomy_level">taxonomy_level</code></td>
<td>
<p>The level which need to be plotted.</p>
</td></tr>
<tr><td><code id="plot_merged_assignment_+3A_sorting_order">sorting_order</code></td>
<td>
<p>the order in which rows shall be sorted, &quot;abundance&quot; is defult,
&quot;alphabetical&quot; is an alternative.</p>
</td></tr>
<tr><td><code id="plot_merged_assignment_+3A_row_limit">row_limit</code></td>
<td>
<p>the max amount of rows to plot (default is 60).</p>
</td></tr>
<tr><td><code id="plot_merged_assignment_+3A_min_row_abundance">min_row_abundance</code></td>
<td>
<p>the minimal sum of abundances in a row required to plot.
Rows whose sum is less than this value are dropped even if row_limit is specified.
Ignored for &quot;alphabetical&quot; order. (default 0.0).</p>
</td></tr>
<tr><td><code id="plot_merged_assignment_+3A_plot_title">plot_title</code></td>
<td>
<p>The plot title.</p>
</td></tr>
<tr><td><code id="plot_merged_assignment_+3A_filename">filename</code></td>
<td>
<p>The output file mask, PDF and SVG files will be produced with Cairo device.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
hmp_even_fp &lt;- system.file("extdata", "HMP_even", package="MetaComp")
hmp_stagger_fp &lt;- system.file("extdata", "HMP_stagger", package="MetaComp")
data_files &lt;- data.frame(V1 = c("HMP_even", "HMP_stagger"),
                         V2 = c(file.path(hmp_even_fp, "allReads-gottcha2-speDB-b.list.txt"),
                                file.path(hmp_stagger_fp, "allReads-gottcha2-speDB-b.list.txt")))
write.table(data_files, file.path(tempdir(), "assignments.txt"),
                                                 row.names = FALSE, col.names = FALSE)
gottcha2_assignments = merge_edge_assignments(
                         load_edge_assignments(
                           file.path(tempdir(), "assignments.txt"), type = "gottcha2"))
plot_merged_assignment(gottcha2_assignments, "family", 'alphabetical', 100, 0,
                                       "HMP side-to-side", file.path(tempdir(), "assignment.pdf"))

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
