<!DOCTYPE html><html lang="en"><head><title>Help for package metaRNASeq</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metaRNASeq}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#metaRNASeq-package'>
<p>Meta-analysis for RNA-seq data.</p></a></li>
<li><a href='#adjpval'>
<p>Simulated adjusted p-values</p></a></li>
<li><a href='#dispFuncs'>
<p>Gamma regression parameters describing the mean-dispersion relationship for two real datasets.</p></a></li>
<li><a href='#FC'>
<p>Simulated fold changes (FC)</p></a></li>
<li><a href='#fishercomb'>
<p>P-value combination using Fisher's method</p></a></li>
<li><a href='#IDD.IRR'><p> Integration-driven Discovery and Integration-driven Revision Rates</p></a></li>
<li><a href='#invnorm'>
<p>P-value combination using the inverse normal method</p></a></li>
<li><a href='#param'>
<p>Mean simulation parameters</p></a></li>
<li><a href='#rawpval'>
<p>Simulated p-values</p></a></li>
<li><a href='#sim.function'>
<p>Simulation of multiple RNA-seq experiments</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Meta-Analysis of RNA-Seq Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-09-30</td>
</tr>
<tr>
<td>Author:</td>
<td>Guillemette Marot  [aut, cre], Andrea Rau  [aut, cre], Florence Jaffrezic [aut], Samuel Blanck [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>samuel Blanck &lt;samuel.blanck@univ-lille.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DESeq2 (&ge; 1.0.17), VennDiagram (&ge; 1.6.20)</td>
</tr>
<tr>
<td>Description:</td>
<td>Implementation of two p-value combination techniques (inverse normal and Fisher methods). A vignette is provided to explain how to perform a meta-analysis from two independent RNA-seq experiments.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>biocViews:</td>
<td>HighThroughputSequencing, RNAseq, DifferentialExpression</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-09-30 14:13:40 UTC; sblanck</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-10-01 15:00:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='metaRNASeq-package'>
Meta-analysis for RNA-seq data.
</h2><span id='topic+metaRNASeq-package'></span><span id='topic+metaRNASeq'></span>

<h3>Description</h3>

<p>Implementation of two p-value combination techniques (inverse normal and Fisher methods). A vignette is provided to explain how to perform a meta-analysis from two independent RNA-seq experiments.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> metaRNASeq</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-01-26</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Andrea Rau, Guillemette Marot, Florence Jaffrezic 
</p>
<p>Maintainer: Guillemette Marot &lt;guillemette.marot@inria.fr&gt;
</p>


<h3>References</h3>

<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>See Also</h3>

<p><code><a href="#topic+invnorm">invnorm</a></code>
<code><a href="#topic+fishercomb">fishercomb</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#An User's guide with detailed examples can be downloaded in interactive R sessions
if(interactive()){
	vignette("metaRNASeq")
	}
</code></pre>

<hr>
<h2 id='adjpval'>
Simulated adjusted p-values
</h2><span id='topic+adjpval'></span>

<h3>Description</h3>

<p>The adjusted p-values provided here result from the following procedure:
1) simulation of two RNA-seq experiments with four replicates in each condition via the <code><a href="#topic+sim.function">sim.function</a></code>, 2) analysis of differentially expressed tags using the DESeq package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(adjpval)</code></pre>


<h3>Format</h3>

<p>List of length 2, where each list is a vector containing the adjusted p-values for 14,456 genes from
individual differential analyses (obtained using DESeq v1.8.3) of each of the simulated RNA-seq datasets. 
</p>


<h3>Details</h3>

<p>It is possible to reproduce these adjusted p-values using the procedure described in the package vignette.
</p>


<h3>References</h3>

<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(adjpval)
## Maybe str(adjpval)
</code></pre>

<hr>
<h2 id='dispFuncs'>
Gamma regression parameters describing the mean-dispersion relationship for two real datasets.
</h2><span id='topic+dispFuncs'></span>

<h3>Description</h3>

<p>Gamma regression parameters describing the mean-dispersion relationship for each of the 
two real datasets considered in the associated paper, as estimated using the DESeq
package version 1.8.3 (Anders and Huber, 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dispFuncs)</code></pre>


<h3>Format</h3>

<p>List of length 2, where each list is a vector containing the two estimated coefficients (<code class="reqn">\alpha_0</code> and 
<code class="reqn">\alpha_1</code>) for the gamma regression in each study (see details below).
</p>


<h3>Details</h3>

<p>The <code>dispFuncs</code> object contains the estimated coefficients from the parametric gamma regressions
describing the mean-dispersion relationship for the two real datasets considered in the associated paper.
The gamma regressions were estimated using the DESeq package version 1.8.3 (Anders and Huber, 2010).
</p>
<p>Briefly, after estimating a per-gene mean expression and dispersion values, the DESeq package fits a curve 
through these estimates. These fitted values correspond to an estimation of the typical relationship between 
mean expression values <code class="reqn">\mu</code> and dispersions <code class="reqn">\alpha</code> within a given dataset. By default, this relationship 
is estimated using a gamma-family generalized linear model (GLM), where two coefficients <code class="reqn">\alpha_0</code> and <code class="reqn">\alpha_1</code> 
are found to parameterize the fit as <code class="reqn">\alpha = \alpha_0 + \alpha_1 / \mu</code>.
</p>
<p>For the first dataset (F078), the estimated mean-dispersion relationship is described using the following
gamma-family GLM:
</p>
<p style="text-align: center;"><code class="reqn">\alpha = 0.024 + 14.896 / \mu.</code>
</p>

<p>For the second dataset (F088), the estimated mean-dispersion relationship is described using the following
gamma-family GLM:
</p>
<p style="text-align: center;"><code class="reqn">\alpha = 0.00557 + 1.54247 / \mu.</code>
</p>

<p>These gamma-family GLMs describing the mean-dispersions relationship in each of the two datasets are used in this
package to simulate data using dispersion parameters that are as realistic as possible.
</p>


<h3>References</h3>

<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>
<p>S. Anders and W. Huber (2010). Differential expression analysis for sequence count data. 
<em>Genome Biology</em>, 11:R106.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sim.function">sim.function</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dispFuncs)
</code></pre>

<hr>
<h2 id='FC'>
Simulated fold changes (FC)
</h2><span id='topic+FC'></span>

<h3>Description</h3>

<p>The FC provided here result from the following procedure:
1) simulation of two RNA-seq experiments with four replicates in each condition via the <code><a href="#topic+sim.function">sim.function</a></code>, 2) analysis of differentially expressed tags using the DESeq package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(FC)</code></pre>


<h3>Format</h3>

<p>List of length 2, where each list is a vector containing the FC for 14,456 genes from
individual differential analyses (obtained using DESeq v1.8.3) of each of the simulated RNA-seq datasets. 
</p>


<h3>Details</h3>

<p>It is possible to reproduce these FC using the procedure described in the package vignette.
</p>


<h3>References</h3>

<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(FC)
## Maybe str(FC)
</code></pre>

<hr>
<h2 id='fishercomb'>
P-value combination using Fisher's method
</h2><span id='topic+fishercomb'></span>

<h3>Description</h3>

<p>Combines one sided p-values using Fisher's method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fishercomb(indpval, BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fishercomb_+3A_indpval">indpval</code></td>
<td>

<p>List of vectors of one sided p-values to be combined.
</p>
</td></tr>
<tr><td><code id="fishercomb_+3A_bhth">BHth</code></td>
<td>

<p>Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The test statistic for each gene <em>g</em> is defined as
</p>
<p style="text-align: center;"><code class="reqn">F_g = -2 \sum_{s=1}^S ln(p_{gs})</code>
</p>

<p>where <code class="reqn">p_{gs}</code> corresponds to the raw <em>p</em>-value obtained for gene <em>g</em> in a differential
analysis for study <em>s</em> (assumed to be uniformly distributed under the null hypothesis). Under the
null hypothesis, the test statistic <code class="reqn">F_g</code> follows a <code class="reqn">\chi^2</code> distribution with <em>2S</em>
degrees of freedom. Classical procedures for the correction of multiple testing, such as that of Benjamini
and Hochberg (1995) may subsequently be applied to the obtained <em>p</em>-values to control the false
discovery rate at a desired rate <code class="reqn">\alpha</code>. 
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>DEindices</code></td>
<td>
<p>Indices of differentially expressed genes at the chosen Benjamini Hochberg threshold.</p>
</td></tr>
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td></tr>
<tr><td><code>rawpval</code></td>
<td>
<p>Vector with raw p-values for differential expression in the meta-analysis.</p>
</td></tr>
<tr><td><code>adjpval</code></td>
<td>
<p>Vector with adjusted p-values for differential expression in the meta-analysis.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Y. Benjamini and Y. Hochberg (1995). Controlling the false discovery rate: a pratical and powerful approach
to multiple testing. <em>JRSS B</em> (57): 289-300.
</p>
<p>M. Brown (1975). A method for combining non-independent, one-sided tests of significance. <em>Biometrics</em> <b>31</b>(4): 987-992.
</p>
<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metaRNASeq">metaRNASeq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rawpval)
fishcomb &lt;- fishercomb(rawpval, BHth = 0.05)
DE &lt;- ifelse(fishcomb$adjpval&lt;=0.05,1,0)
hist(fishcomb$rawpval,nclass=100)

## A more detailed example is given in the vignette of the package:
## vignette("metaRNASeq")
</code></pre>

<hr>
<h2 id='IDD.IRR'> Integration-driven Discovery and Integration-driven Revision Rates </h2><span id='topic+IDD.IRR'></span>

<h3>Description</h3>

<p>Calculates the gain or the loss of differentially expressed genes due to meta-analysis
compared to individual studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IDD.IRR(meta_de, ind_de)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="IDD.IRR_+3A_meta_de">meta_de</code></td>
<td>
<p> Vector of differentially expressed tags (or indices of these tags) with the meta-analysis</p>
</td></tr>
<tr><td><code id="IDD.IRR_+3A_ind_de">ind_de</code></td>
<td>
<p> List of vectors storing differentially expressed tags (or indices of these tags) in each individual study </p>
</td></tr>
</table>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>DE</code></td>
<td>
<p>Number of Differentially Expressed (DE) genes </p>
</td></tr>
<tr><td><code>IDD</code></td>
<td>
<p>Integration Driven Discoveries: number of genes that are declared DE in
the meta-analysis that were not identified in any of
the individual studies alone.</p>
</td></tr>
<tr><td><code>Loss</code></td>
<td>
<p>Number of genes that are declared DE in individual studies but not in meta-analysis.</p>
</td></tr>
<tr><td><code>IDR</code></td>
<td>
<p>Integration-driven Discovery Rate: proportion of genes that are identified as DE in the meta-analysis that were not
identified in any of the individual studies alone.</p>
</td></tr>
<tr><td><code>IRR</code></td>
<td>
<p>Integration-driven Revision Rate: percentage of genes
that are declared DE in individual studies but not in meta-analysis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rawpval)
adjpval&lt;-lapply(rawpval, FUN=function(x) p.adjust(x, method="BH"))
ind_smalladjp&lt;-lapply(adjpval, FUN=function(x) which(x &lt;= 0.05))
#indicators corresponding to the inverse normal p-value combination
invnormcomb &lt;- invnorm(rawpval,nrep=c(8,8), BHth = 0.05) 
IDD.IRR(invnormcomb$DEindices,ind_smalladjp)
#indicators corresponding to the p-value combination with Fisher's method
fishcomb &lt;- fishercomb(rawpval, BHth = 0.05)
IDD.IRR(fishcomb$DEindices,ind_smalladjp)
</code></pre>

<hr>
<h2 id='invnorm'>
P-value combination using the inverse normal method
</h2><span id='topic+invnorm'></span>

<h3>Description</h3>

<p>Combines one sided p-values using the inverse normal method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>invnorm(indpval, nrep, BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="invnorm_+3A_indpval">indpval</code></td>
<td>

<p>List of vectors of one sided p-values to be combined.
</p>
</td></tr>
<tr><td><code id="invnorm_+3A_nrep">nrep</code></td>
<td>

<p>Vector of numbers of replicates used in each study to calculate the previous one-sided p-values. 
</p>
</td></tr>
<tr><td><code id="invnorm_+3A_bhth">BHth</code></td>
<td>

<p>Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each gene <em>g</em>, let
</p>
<p style="text-align: center;"><code class="reqn">N_g = \sum_{s=1}^S \omega_s \Phi^{-1}(1-p_{gs}),</code>
</p>

<p>where <code class="reqn">p_{gs}</code> corresponds to the raw <em>p</em>-value obtained for gene <em>g</em> in a differential
analysis for study <em>s</em> (assumed to be uniformly distributed under the null hypothesis), <code class="reqn">\Phi</code> the
cumulative distribution function of the standard normal distribution, and <code class="reqn">\omega_s</code> a set of weights.
We define the weights <code class="reqn">\omega_s</code> as in Marot and Mayer (2009):
</p>
<p style="text-align: center;"><code class="reqn">\omega_s = \sqrt{\frac{\sum_c R_{cs}}{\sum_\ell \sum_c R_{c\ell}}},</code>
</p>

<p>where <code class="reqn">\sum_c R_{cs}</code> is the total number of biological replicates in study <em>s</em>. This allows
studies with large numbers of biological replicates to be attributed a larger weight than smaller studies.
</p>
<p>Under the null hypothesis, the test statistic <code class="reqn">N_g</code> follows a N(0,1) distribution. A unilateral
test on the righthand tail of the distribution may then be performed, and classical procedures for the
correction of multiple testing, such as that of Benjamini and Hochberg (1995), may subsequently be applied to
the obtained <em>p</em>-values to control the false discovery rate at a desired level <code class="reqn">\alpha</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>DEindices</code></td>
<td>
<p>Indices of differentially expressed genes at the chosen Benjamini Hochberg threshold.</p>
</td></tr>
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td></tr>
<tr><td><code>rawpval</code></td>
<td>
<p>Vector with raw p-values for differential expression in the meta-analysis.</p>
</td></tr>
<tr><td><code>adjpval</code></td>
<td>
<p>Vector with adjusted p-values for differential expression in the meta-analysis.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function resembles the function <code>directpvalcombi</code> in the <em>metaMA</em> R package; there is, however, one 
important difference in the calculation of <em>p</em>-values. In particular, for microarray data, it is typically 
advised to separate under- and over-expressed genes prior to the meta-analysis. In the case of RNA-seq data, 
differential analyses from individual studies typically make use of negative binomial models and exact tests, 
which lead to one-sided, rather than two-sided, p-values. As such, we suggest performing a meta-analysis over 
the full set of genes, followed by an a posteriori check, and if necessary filter, of genes with conflicting 
results (over vs. under expression) among studies.
</p>


<h3>References</h3>

 
<p>Y. Benjamini and Y. Hochberg (1995). Controlling the false discovery rate: a pratical and powerful approach
to multiple testing. <em>JRSS B</em> (57): 289-300.
</p>
<p>Hedges, L. and Olkin, I. (1985). Statistical Methods for Meta-Analysis. Academic Press. 
</p>
<p>Marot, G. and Mayer, C.-D. (2009). Sequential analysis for microarray data based on sensitivity and meta-analysis.
<em>SAGMB</em> 8(1): 1-33.
</p>
<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91</p>


<h3>See Also</h3>

<p><code><a href="#topic+metaRNASeq">metaRNASeq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rawpval)
## 8 replicates simulated in each study
invnormcomb &lt;- invnorm(rawpval,nrep=c(8,8), BHth = 0.05)       
DE &lt;- ifelse(invnormcomb$adjpval&lt;=0.05,1,0)
hist(invnormcomb$rawpval,nclass=100)

## A more detailed example is given in the vignette of the package:
## vignette("metaRNASeq")
</code></pre>

<hr>
<h2 id='param'>
Mean simulation parameters
</h2><span id='topic+param'></span>

<h3>Description</h3>

<p>Mean simulation parameters obtained from the analysis of a real dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(param)</code></pre>


<h3>Format</h3>

<p>A data frame with 26408 observations on the following 3 variables.
</p>

<dl>
<dt><code>mucond1</code></dt><dd><p>a numeric vector with mean parameters for condition 1</p>
</dd>
<dt><code>mucond2</code></dt><dd><p>a numeric vector with mean parameters for condition 2</p>
</dd>
<dt><code>DE</code></dt><dd><p>a logical vector indicating which tags are differentially expressed (value 1)</p>
</dd>
</dl>



<h3>Details</h3>

<p>Mean parameters provided in this package are empirical means (obtained after normalization for library size differences) of real data described in the following references.
</p>


<h3>Source</h3>

<p>Supplementary material of (Dillies et al., 2013) paper.
</p>


<h3>References</h3>

<p>M.A. Dillies, A. Rau, J. Aubert, C. Hennequet-Antier, M. Jeanmougin, N. Servant, C. Keime, G. Marot, D. Castel, J. Estelle, G. Guernec, B. Jagla, L. Jouneau, D. Laloe, C. Le Gall, B. Schaeffer, S. Le Crom, M. Guedj and F. Jaffrezic, on behalf of the French StatOmique Consortium (2013) 
A comprehensive evaluation of normalization methods for Illumina high-throughput RNA sequencing data analysis. <em>Briefings in Bioinformatics</em> <b>14</b>(6):671-83 .
</p>
<p>T. Strub, S. Giuliano, T. Ye, et al. (2011) Essential role of microphthalmia transcription factor for DNA replication, mitosis and genomic stability in melanoma. emphOncogene <b>30</b>:2319-32.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(param)
</code></pre>

<hr>
<h2 id='rawpval'>
Simulated p-values
</h2><span id='topic+rawpval'></span>

<h3>Description</h3>

<p>The p-values provided here result from the following procedure:
1) simulation of two RNA-seq experiments with four replicates in each condition via the <code><a href="#topic+sim.function">sim.function</a></code>, 2) analysis of differentially expressed tags using the DESeq package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(rawpval)</code></pre>


<h3>Format</h3>

<p>List of length 2, where each list is a vector containing the raw p-values for 14,456 genes from
individual differential analyses (obtained using DESeq v1.8.3) of each of the simulated RNA-seq datasets. 
</p>


<h3>Details</h3>

<p>It is possible to reproduce these p-values using the procedure described in the package vignette.
</p>


<h3>References</h3>

<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(rawpval)
## Maybe str(rawpval)
</code></pre>

<hr>
<h2 id='sim.function'>
Simulation of multiple RNA-seq experiments
</h2><span id='topic+sim.function'></span><span id='topic+extractfromsim'></span>

<h3>Description</h3>

<p>Simulate data arising from multiple independent RNA-seq experiments 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sim.function(param, dispFuncs, nrep = 4, classes = NULL, inter.sd = 0.3)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="sim.function_+3A_param">param</code></td>
<td>
<p> Mean expression levels: <code>param</code> must be a data frame containing at least two columns named 
&quot;mucond1&quot; and &quot;mucond2&quot; and one row per gene.
</p>
</td></tr>
<tr><td><code id="sim.function_+3A_dispfuncs">dispFuncs</code></td>
<td>

<p>List of length equal to the number of studies to be simulated, containing the gamma regression parameters 
describing the mean-dispersion relationship for each one; these are the mean-dispersion functions linking mean 
and intra-study variability for each independent experiment. 
</p>
</td></tr>
<tr><td><code id="sim.function_+3A_nrep">nrep</code></td>
<td>

<p>Number of replicates to be simulated in each condition in each study. Ignored if <code>classes</code> is filled.
</p>
</td></tr>
<tr><td><code id="sim.function_+3A_classes">classes</code></td>
<td>

<p>List of class memberships, one per study (maximum 20 studies). Each vector or factor of the list can only 
contain two levels which correspond to the two conditions studied. If NULL, <code>classes</code> is built as a 
list of two vectors with <code>nrep</code> labels 1 (for condition 1) and <code>nrep</code> labels 2 (for condition 2).
</p>
</td></tr>
<tr><td><code id="sim.function_+3A_inter.sd">inter.sd</code></td>
<td>

<p>Inter-study variability. By default, <code>inter.sd</code> is set to 0.3, which corresponds to a moderate inter-study 
variability in the case where <code><a href="#topic+param">param</a></code> and <code><a href="#topic+dispFuncs">dispFuncs</a></code> parameters are used to simulate data.
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Details about the simulation procedure are given in the following paper: 
</p>


<h3>Value</h3>

<p>A matrix with simulated expression levels, one row per gene and one column per replicate. Names of studies are given 
in the column names of the matrix. 
</p>


<h3>Note</h3>

<p>If the <code><a href="#topic+param">param</a></code> data provided in this package are not used to simulate data, one should check that the 
per-condition means in <code>param</code> are reasonable. Note also that for genes to be simulated as non-differentially
expressed, the values of &quot;mucond1&quot; and &quot;mucond2&quot; in <code>param</code> should be equal.
</p>


<h3>References</h3>

<p>A. Rau, G. Marot and F. Jaffrezic (2014). Differential meta-analysis of RNA-seq data. <em>BMC Bioinformatics</em> <b>15</b>:91
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metaRNASeq">metaRNASeq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Load simulation parameters
data(param)
data(dispFuncs)

## Simulate data
matsim &lt;- sim.function(param = param, dispFuncs = dispFuncs)
sim.conds &lt;- colnames(matsim)
rownames(matsim) &lt;-paste("tag", 1:dim(matsim)[1],sep="")

# extract simulated data from one study
simstudy1 &lt;- extractfromsim(matsim,"study1")
head(simstudy1$study)
simstudy1$pheno
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
