<!DOCTYPE html><html><head><title>Help for package MetaCycle</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MetaCycle}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#cycHumanBloodData'><p>cycHumanBloodData</p></a></li>
<li><a href='#cycHumanBloodDesign'><p>cycHumanBloodDesign</p></a></li>
<li><a href='#cycMouseLiverProtein'><p>cycMouseLiverProtein</p></a></li>
<li><a href='#cycMouseLiverRNA'><p>cycMouseLiverRNA</p></a></li>
<li><a href='#cycSimu4h2d'><p>cycSimu4h2d</p></a></li>
<li><a href='#cycVignettesAMP'><p>cycVignettesAMP</p></a></li>
<li><a href='#cycYeastCycle'><p>cycYeastCycle</p></a></li>
<li><a href='#meta2d'><p>Detect rhythmic signals from time-series datasets with multiple methods</p></a></li>
<li><a href='#meta3d'><p>Detect rhythmic signals from time-series datasets with individual</p>
information</a></li>
<li><a href='#MetaCycle'><p>MetaCycle</p>
Evaluate Periodicity in Large Scale Data</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Evaluate Periodicity in Large Scale Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.0</td>
</tr>
<tr>
<td>Description:</td>
<td>There are two functions-meta2d and meta3d for
    detecting rhythmic signals from time-series datasets. For analyzing
    time-series datasets without individual information, 'meta2d' is 
    suggested, which could incorporates multiple methods from ARSER, 
    JTK_CYCLE and Lomb-Scargle in the detection of interested rhythms. For 
    analyzing time-series datasets with individual information, 'meta3d' is 
    suggested, which takes use of any one of these three methods to analyze 
	time-series data individual by individual and gives out integrated values 
    based on analysis result of each individual.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gnm</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, parallel</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Author:</td>
<td>Gang Wu [aut, cre],
  Ron Anafi [aut, ctb],
  John Hogenesch [aut, ctb],
  Michael Hughes [aut, ctb],
  Karl Kornacker [aut, ctb],
  Xavier Li [aut, ctb],
  Matthew Carlucci [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gang Wu &lt;wggucas@gmail.com&gt;</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-04-18 21:27:44 UTC; wug8hn</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-04-18 22:00:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='cycHumanBloodData'>cycHumanBloodData</h2><span id='topic+cycHumanBloodData'></span>

<h3>Description</h3>

<p>This data set lists time-series profiles of 10 transcripts
sampled from  multiple individuals under different sleep conditions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycHumanBloodData
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 439 columns (column 1 = transcript name,
column 2 to 439 = samples from individuals at different time points and
sleep conditions).</p>


<h3>Source</h3>

<p>Moller-Levet C. S., et al. (2013). Effects of insufficient sleep on
circadian rhythmicity and expression amplitude of the human blood
transcriptome. <em>Proc Natl Acad Sci U S A</em>, <b>110(12)</b>,
E1132&ndash;1141.
</p>

<hr>
<h2 id='cycHumanBloodDesign'>cycHumanBloodDesign</h2><span id='topic+cycHumanBloodDesign'></span>

<h3>Description</h3>

<p>This data set describes individual information, sleep
condition and sampling time corresponding to each sample in
'cycHumanBloodData'.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycHumanBloodDesign
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 4 columns described as below:
</p>

<table>
<tr>
 <td style="text-align: right;">
    [,1] </td><td style="text-align: left;"> sample_library  </td><td style="text-align: left;"> character  </td><td style="text-align: left;"> sample ID</td>
</tr>
<tr>
 <td style="text-align: right;">
    [,2] </td><td style="text-align: left;"> subject         </td><td style="text-align: left;"> character  </td><td style="text-align: left;"> individual ID</td>
</tr>
<tr>
 <td style="text-align: right;">
    [,3] </td><td style="text-align: left;"> group           </td><td style="text-align: left;"> character  </td><td style="text-align: left;"> sleep condition</td>
</tr>
<tr>
 <td style="text-align: right;">
    [,4] </td><td style="text-align: left;"> time_hoursawake </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> hours after awake
  </td>
</tr>

</table>


<h3>Source</h3>

<p>Moller-Levet C. S., et al. (2013). Effects of insufficient sleep on
circadian rhythmicity and expression amplitude of the human blood
transcriptome. <em>Proc Natl Acad Sci U S A</em>, <b>110(12)</b>,
E1132&ndash;1141.
</p>

<hr>
<h2 id='cycMouseLiverProtein'>cycMouseLiverProtein</h2><span id='topic+cycMouseLiverProtein'></span>

<h3>Description</h3>

<p>This data set lists expression profiles of 5 circadian
proteins with 3h-resolution covering two days.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycMouseLiverProtein
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 49 columns(column 1 = protein name,
column 2 to 49 = time points from CT0 to CT45 with three replicates
at each time point).</p>


<h3>Source</h3>

<p>Robles M. S., Cox J., Mann M. (2014). In-vivo quantitative
proteomics reveals a key contribution of post-transcriptional
mechanisms to the circadian regulation of liver metabolism.
<em>PLoS Genet</em>, <b>10(1)</b>, e1004047.
</p>

<hr>
<h2 id='cycMouseLiverRNA'>cycMouseLiverRNA</h2><span id='topic+cycMouseLiverRNA'></span>

<h3>Description</h3>

<p>This data set lists expression profiles of 10 circadian
transcripts with 1h-resolution covering two days.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycMouseLiverRNA
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 49 columns(column 1 = transcript name,
column 2 to 49 = time points from CT18 to CT65).</p>


<h3>Source</h3>

<p>Hughes M. E., et al. (2009). Harmonics of circadian gene
transcription in mammals. <em>PLoS Genet</em>, <b>5(4)</b>, e1000442.
</p>

<hr>
<h2 id='cycSimu4h2d'>cycSimu4h2d</h2><span id='topic+cycSimu4h2d'></span>

<h3>Description</h3>

<p>This data set lists 20 simulated profiles(periodic and
non-periodic) with 4h-resolution covering two periods.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycSimu4h2d
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 13 columns(column 1 = curve ID, column 2 to
13 = time points from 0 to 44).</p>


<h3>Source</h3>

<p>Wu G., Zhu J., Yu J., Zhou L., Huang J. Z. and  Zhang Z. (2014).
Evaluation of five methods for genome-wide circadian gene identification.
<em>Journal of Biological Rhythms</em>, <b>29(4)</b>, 231&ndash;242.
</p>

<hr>
<h2 id='cycVignettesAMP'>cycVignettesAMP</h2><span id='topic+cycVignettesAMP'></span>

<h3>Description</h3>

<p>This data set lists meta2d's analysis results of three
circadian transcripts selected from the same source dataset used by
cycMouseLiverRNA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycVignettesAMP
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 71 columns described as below:
</p>

<table>
<tr>
 <td style="text-align: right;">
        [,1] </td><td style="text-align: left;"> CycID                 </td><td style="text-align: left;"> character  </td><td style="text-align: left;"> transcript name</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,2] </td><td style="text-align: left;"> ARS_pvalue            </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> pvalue from ARS</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,3] </td><td style="text-align: left;"> ARS_BH.Q              </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> FDR from ARS</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,4] </td><td style="text-align: left;"> ARS_period            </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> period from ARS</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,5] </td><td style="text-align: left;"> ARS_adjphase          </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> adjusted phase from ARS</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,6] </td><td style="text-align: left;"> ARS_amplitude         </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> amplitude from ARS</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,7] </td><td style="text-align: left;"> JTK_pvalue            </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> pvalue from JTK</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,8] </td><td style="text-align: left;"> JTK_BH.Q              </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> FDR from JTK</td>
</tr>
<tr>
 <td style="text-align: right;">
        [,9] </td><td style="text-align: left;"> JTK_period            </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> period from JTK</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,10] </td><td style="text-align: left;"> JTK_adjphase          </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> adjusted phase from JTK</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,11] </td><td style="text-align: left;"> JTK_amplitude         </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> amplitude from JTK</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,12] </td><td style="text-align: left;"> LS_pvalue             </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> pvalue from LS</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,13] </td><td style="text-align: left;"> LS_BH.Q               </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> FDR from JTK</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,14] </td><td style="text-align: left;"> LS_period             </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> period from LS</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,15] </td><td style="text-align: left;"> LS_adjphase           </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> adjusted phase from LS</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,16] </td><td style="text-align: left;"> LS_amplitude          </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> amplitude from LS</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,17] </td><td style="text-align: left;"> meta2d_pvalue         </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> integrated pvalue</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,18] </td><td style="text-align: left;"> meta2d_BH.Q           </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> FDR based on integrated pvalue</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,19] </td><td style="text-align: left;"> meta2d_period         </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> averaged period of three methods</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,20] </td><td style="text-align: left;"> meta2d_phase          </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> integrated phase</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,21] </td><td style="text-align: left;"> meta2d_Base           </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> baseline value given by meta2d</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,22] </td><td style="text-align: left;"> meta2d_AMP            </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> amplitude given by meta2d</td>
</tr>
<tr>
 <td style="text-align: right;">
       [,23] </td><td style="text-align: left;"> meta2d_rAMP           </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> relative amplitude</td>
</tr>
<tr>
 <td style="text-align: right;">
    [,24:71] </td><td style="text-align: left;"> CT18 to CT65          </td><td style="text-align: left;"> numeric    </td><td style="text-align: left;"> sampling time point
  </td>
</tr>

</table>


<h3>Source</h3>

<p>Hughes M. E., et al. (2009). Harmonics of circadian gene
transcription in mammals. <em>PLoS Genet</em>, <b>5(4)</b>, e1000442.
</p>

<hr>
<h2 id='cycYeastCycle'>cycYeastCycle</h2><span id='topic+cycYeastCycle'></span>

<h3>Description</h3>

<p>This data set lists expression profiles of 10 cycling
transcripts with 16-minutes resolution covering about two yeast cell
cycles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cycYeastCycle
</code></pre>


<h3>Format</h3>

<p>A dataframe containing 12 columns(column 1 = transcript name,
column 2 to 12 = time points from 2 minutes to 162 minutes after
recovery phase).</p>


<h3>Source</h3>

<p>Orlando D. A., et al. (2008). Global control of cell-cycle
transcription by coupled CDK and network oscillators. <em>Nature</em>,
<b>453(7197)</b>, 944&ndash;947.
</p>

<hr>
<h2 id='meta2d'>Detect rhythmic signals from time-series datasets with multiple methods</h2><span id='topic+meta2d'></span>

<h3>Description</h3>

<p>This is a function that incorporates ARSER, JTK_CYCLE and Lomb-Scargle to
detect rhythmic signals from time-series datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta2d(infile, outdir = "metaout", filestyle, timepoints, minper = 20,
  maxper = 28, cycMethod = c("ARS", "JTK", "LS"),
  analysisStrategy = "auto", outputFile = TRUE,
  outIntegration = "both", adjustPhase = "predictedPer",
  combinePvalue = "fisher", weightedPerPha = FALSE, ARSmle = "auto",
  ARSdefaultPer = 24, outRawData = FALSE, releaseNote = TRUE,
  outSymbol = "", parallelize = FALSE, nCores = 1, inDF = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta2d_+3A_infile">infile</code></td>
<td>
<p>a character string. The name of input file containing
time-series data.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_outdir">outdir</code></td>
<td>
<p>a character string. The name of directory used to store
output files.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_filestyle">filestyle</code></td>
<td>
<p>a character vector(length 1 or 3). The data format of
input file, must be <code>"txt"</code>, or <code>"csv"</code>, or a character vector
containing field separator character(<code>sep</code>), quoting character
(<code>quote</code>), and the character used for decimal points(<code>dec</code>,
for details see <code><a href="utils.html#topic+read.table">read.table</a></code>).</p>
</td></tr>
<tr><td><code id="meta2d_+3A_timepoints">timepoints</code></td>
<td>
<p>a numeric vector corresponding to sampling time points
of input time-series data; if sampling time points are in the first line
of input file, it could be set as a character sting-&quot;Line1&quot; or &quot;line1&quot;.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_minper">minper</code></td>
<td>
<p>a numeric value. The minimum period length of interested
rhythms. The default is <code>20</code> for circadian rhythms.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_maxper">maxper</code></td>
<td>
<p>a numeric value. The maximum period length of interested
rhythms. The default is <code>28</code> for circadian rhythms.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_cycmethod">cycMethod</code></td>
<td>
<p>a character vector(length 1 or 2 or 3). User-defined
methods for detecting rhythmic signals, must be selected as any one, any
two or all three methods(default) from <code>"ARS"</code>(ARSER),
<code>"JTK"</code>(JTK_CYCLE) and <code>"LS"</code>(Lomb-Scargle).</p>
</td></tr>
<tr><td><code id="meta2d_+3A_analysisstrategy">analysisStrategy</code></td>
<td>
<p>a character string. The strategy used to select
proper methods from <code>cycMethod</code> for analyzing input time-series
data, must be <code>"auto"</code>(default), or <code>"selfUSE"</code>. See
<code>Details</code> part for more information.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_outputfile">outputFile</code></td>
<td>
<p>logical. If <code>TRUE</code>, analysis results will be wrote
in the output files. If <code>FALSE</code>, analysis results will be returned
as an R list.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_outintegration">outIntegration</code></td>
<td>
<p>a character string. This parameter controls what
kinds of analysis results will be outputted, must be one of <code>"both"</code>
(default), <code>"onlyIntegration"</code>(only output integration file), or
<code>"noIntegration"</code>(not output integration file).</p>
</td></tr>
<tr><td><code id="meta2d_+3A_adjustphase">adjustPhase</code></td>
<td>
<p>a character string. The method used to adjust original
phase calculated by each method in integration file, must be one of
<code>"predictedPer"</code>(adjust phase with predicted period length) or
<code>"notAdjusted"</code>(not adjust phase).</p>
</td></tr>
<tr><td><code id="meta2d_+3A_combinepvalue">combinePvalue</code></td>
<td>
<p>a character string. The method used to integrate
multiple p-values, must be one of <code>"bonferroni"</code>(Bonferroni
correction), or <code>"fisher"</code>(Fisher's method).</p>
</td></tr>
<tr><td><code id="meta2d_+3A_weightedperpha">weightedPerPha</code></td>
<td>
<p>logical. If <code>TRUE</code>, weighted scores based on
p-value given by each method will be used to calculate the integrated
period length and phase.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_arsmle">ARSmle</code></td>
<td>
<p>a character string. The strategy of using MLE method in
<code><a href="stats.html#topic+ar">ar</a></code> fit of <code>"ARS"</code>, must be one of
<code>"auto"</code>(use MLE depending the number of time points), <code>"mle"</code>
(always use MLE), or <code>"nomle"</code>(never use MLE).</p>
</td></tr>
<tr><td><code id="meta2d_+3A_arsdefaultper">ARSdefaultPer</code></td>
<td>
<p>a numeric value. The expected period length of
interested rhythm, which is a necessary parameter for <code>ARS</code>. The
default is <code>24</code>(for circadian rhythms). Set it to another proper
numeric value for other rhythms.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_outrawdata">outRawData</code></td>
<td>
<p>logical. If <code>TRUE</code>, raw time-series data will be
added in the output files.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_releasenote">releaseNote</code></td>
<td>
<p>logical. If <code>TRUE</code>, reminding or warning notes
during the analysis will be released on the screen.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_outsymbol">outSymbol</code></td>
<td>
<p>a character string. A common prefix exists in the names of
output files.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_parallelize">parallelize</code></td>
<td>
<p>logical. If <code>TRUE</code>, computation will be done in paralleL
Doesn't work in windows machine.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_ncores">nCores</code></td>
<td>
<p>a integer. Bigger or equal to one, number of cores to use.</p>
</td></tr>
<tr><td><code id="meta2d_+3A_indf">inDF</code></td>
<td>
<p>data.frame. If <code>!is.null(inDF)</code> and timepoints is a numeric meta2d will use this data.frame instead of loading from <code>infile</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><a href="https://github.com/cauyrd/ARSER">ARSER</a>(Yang, 2010),
<a href="http://openwetware.org/wiki/HughesLab:JTK_Cycle">JTK_CYCLE</a>(
Hughes, 2010), and
<a href="https://academic.oup.com/bioinformatics/article/22/3/310/220284">
Lomb-Scargle</a>(Glynn, 2006) are three popular methods of detecting
rhythmic signals. <code>ARS</code> can not analyze unevenly sampled datasets,
or evenly sampled datasets but with missing values, or with replicate
samples, or with non-integer sampling interval. <code>JTK</code> is not
suitable to analyze unevenly sampled datasets or evenly sampled datasets
but with non-integer sampling interval. If set <code>analysisStrategy</code>
as <code>"auto"</code>(default), <code>meta2d</code> will automatically select
proper method from <code>cycMethod</code> for each input dataset. If the user
clearly know that the dataset could be analyzed by each method defined
by <code>cycMethod</code> and do not hope to output integrated values,
<code>analysisStrategy</code> can be set as <code>"selfUSE"</code>.
</p>
<p><code>ARS</code> used here is translated from its python version which always
uses <code>"yule-walker"</code>, <code>"burg"</code>, and <code>"mle"</code> methods(see
<code><a href="stats.html#topic+ar">ar</a></code>) to fit autoregressive models to time-series
data. Fitting by <code>"mle"</code> will be very slow for datasets
with many time points. If <code>ARSmle = "auto"</code> is used,
<code>meta2d</code> will only include <code>"mle"</code> when number of time points
is smaller than 24. In addition, one evaluation work(Wu, 2014) indicates
that <code>ARS</code> shows relative high false positive rate in analyzing
high-resolution datasets (1h/2days and 2h/2days). <code>JTK</code>(version 3)
used here is the latest version, which improves its p-value calculation
in analyzing datasets with missing values.
</p>
<p>The power of detecting rhythmic signals for an algorithm is associated
with the nature of data and interested periodic pattern(Deckard, 2013),
which indicates that integrating analysis results from multiple methods
may be helpful to rhythmic detection. For integrating p-values,
Bonferroni correction(<code>"bonferroni"</code>) and Fisher's method(
<code>"fisher"</code>) (Fisher, 1925; implementation code from <span class="pkg">MADAM</span>)
could be selected, and <code>"bonferroni"</code> is usually more conservative
than <code>"fisher"</code>. The integrated period is arithmetic mean of
multiple periods. For integrating phase, <code>meta2d</code> takes use of
<a href="https://en.wikipedia.org/wiki/Mean_of_circular_quantities">
mean of circular quantities</a>. Integrated period and phase is further
used to calculate the baseline value and amplitude through fitting a
constructed periodic model.
</p>
<p>Phases given by <code>JTK</code> and <code>LS</code> need to be adjusted with their
predicted period (<code>adjustedPhase = "predictedPer"</code>) before
integration. If <code>adjustedPhas = "notAdjusted"</code> is selected, no
integrated phase will be calculated. If set <code>weightedPerPha</code> as
<code>TRUE</code>, weighted scores will be used in averaging periods and
phases. Weighted scores for one method are based on all its reported
p-values, which means a weighted score assigned to any one profile will
be affected by all other profiles. It is always a problem of averaging
phases with quite different period lengths(eg. averaging two phases
with 16-hours' and 30-hours' period length). Currently, setting
<code>minper</code>, <code>maxper</code> and <code>ARSdefaultPer</code> to a same value
may be the only way of completely eliminating such problem.
</p>
<p>This function is originally aimed to analyze large scale periodic data(
eg. circadian transcriptome data) without individual information.
Please pay attention to data format of input file(see <code>Examples</code>
part). Except the first column and first row, others are time-series
experimental values(setting missing values as <code>NA</code>).
</p>


<h3>Value</h3>

<p><code>meta2d</code> will write analysis results in different files under
<code>outdir</code> if set <code>outputFile = TRUE</code>. Files named with
&quot;ARSresult&quot;, &quot;JTKresult&quot; and &quot;LSreult&quot; store analysis results from
<code>ARS</code>, <code>JTK</code> and <code>LS</code> respectively. The file named with
&quot;meta2d&quot; is the integration file, and it stores integrated values in
columns with a common name tag-&quot;meta2d&quot;. The integration file also
contains p-value, FDR value, period, phase(adjusted phase if
<code>adjustedPhase = "predictedPer"</code>) and amplitude values calculated
by each method.
If <code>outputFile = FALSE</code> is selected, <code>meta2d</code> will return a
list containing the following components:
</p>

<table>
<tr>
 <td style="text-align: right;">
       ARS   </td><td style="text-align: left;">  analysis results from <code>ARS</code> method</td>
</tr>
<tr>
 <td style="text-align: right;">
       JTK   </td><td style="text-align: left;">  analysis results from <code>JTK</code> method</td>
</tr>
<tr>
 <td style="text-align: right;">
       LS    </td><td style="text-align: left;">  analysis results from <code>LS</code> method</td>
</tr>
<tr>
 <td style="text-align: right;">
       meta  </td><td style="text-align: left;">  the integrated analysis results as mentioned above
  </td>
</tr>

</table>



<h3>References</h3>

<p>Yang R. and  Su Z. (2010). Analyzing circadian expression data by
harmonic regression based on autoregressive spectral estimation.
<em>Bioinformatics</em>, <b>26(12)</b>, i168&ndash;i174.
</p>
<p>Hughes M. E., Hogenesch J. B. and Kornacker K. (2010). JTK_CYCLE: an
efficient nonparametric algorithm for detecting rhythmic components in
genome-scale data sets. <em>Journal of Biological Rhythms</em>,
<b>25(5)</b>, 372&ndash;380.
</p>
<p>Glynn E. F., Chen J. and Mushegian A. R. (2006). Detecting periodic
patterns in unevenly spaced gene expression time series using
Lomb-Scargle periodograms. <em>Bioinformatics</em>, <b>22(3)</b>,
310&ndash;316.
</p>
<p>Wu G., Zhu J., Yu J., Zhou L., Huang J. Z. and  Zhang Z. (2014). Evaluation
of five methods for genome-wide circadian gene identification.
<em>Journal of Biological Rhythms</em>, <b>29(4)</b>, 231&ndash;242.
</p>
<p>Deckard A., Anafi R. C., Hogenesch J. B., Haase S.B. and Harer J. (2013).
Design and analysis of large-scale biological rhythm studies:
a comparison of algorithms for detecting periodic signals in biological
data. <em>Bioinformatics</em>, <b>29(24)</b>, 3174&ndash;3180.
</p>
<p>Fisher, R.A. (1925). <em>Statistical methods for research workers</em>.
Oliver and Boyd (Edinburgh).
</p>
<p>Kugler K. G., Mueller L.A. and Graber A. (2010). MADAM - an open source
toolbox for meta-analysis. <em>Source Code for Biology and Medicine</em>,
<b>5</b>, 3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># write 'cycSimu4h2d', 'cycMouseLiverRNA' and 'cycYeastCycle' into three
# 'csv' files
write.csv(cycSimu4h2d, file="cycSimu4h2d.csv", row.names=FALSE)
write.csv(cycMouseLiverRNA, file="cycMouseLiverRNA.csv", row.names=FALSE)
write.csv(cycYeastCycle, file="cycYeastCycle.csv", row.names=FALSE)

# write 'cycMouseLiverProtein' into a 'txt' file
write.table(cycMouseLiverProtein, file="cycMouseLiverProtein.txt",
  sep="\t", quote=FALSE, row.names=FALSE)

# analyze 'cycMouseLiverRNA.csv' with JTK_CYCLE
# this is masked for keeping the total running time within 10s required by CRAN check
# meta2d(infile="cycMouseLiverRNA.csv", filestyle="csv", outdir="example",
#  timepoints=18:65, cycMethod="JTK", outIntegration="noIntegration")

# analyze 'cycMouseLiverProtein.txt' with JTK_CYCLE and Lomb-Scargle
meta2d(infile="cycMouseLiverProtein.txt", filestyle="txt",
  outdir="example", timepoints=rep(seq(0, 45, by=3), each=3),
  cycMethod=c("JTK","LS"), outIntegration="noIntegration")

# analyze 'cycSimu4h2d.csv' with ARSER, JTK_CYCLE and Lomb-Scargle and
# output integration file with analysis results from each method
meta2d(infile="cycSimu4h2d.csv", filestyle="csv", outdir="example",
  timepoints="Line1")

# analyze 'cycYeastCycle.csv' with ARSER, JTK_CYCLE and Lomb-Scargle to
# detect transcripts associated with cell cycle, and only output
# integration file
meta2d(infile="cycYeastCycle.csv",filestyle="csv", outdir="example",
  minper=80, maxper=96, timepoints=seq(2, 162, by=16),
  outIntegration="onlyIntegration", ARSdefaultPer=85,
  outRawData=TRUE)
# return analysis results instead of output them into files
cyc &lt;- meta2d(infile="cycYeastCycle.csv",filestyle="csv",
  minper=80, maxper=96, timepoints=seq(2, 162, by=16),
  outputFile=FALSE, ARSdefaultPer=85, outRawData=TRUE)
head(cyc$ARS)
head(cyc$JTK)
head(cyc$LS)
head(cyc$meta)
</code></pre>

<hr>
<h2 id='meta3d'>Detect rhythmic signals from time-series datasets with individual
information</h2><span id='topic+meta3d'></span>

<h3>Description</h3>

<p>This is a function that takes use of any one method from ARSER, JTK_CYCLE
and Lomb-Scargle to detect rhythmic signals from time-series datasets
containing individual information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta3d(datafile, designfile, outdir = "metaout", filestyle,
  design_libColm, design_subjectColm, minper = 20, maxper = 28,
  cycMethodOne = "JTK", timeUnit = "hour", design_hrColm,
  design_dayColm = NULL, design_minColm = NULL,
  design_secColm = NULL, design_groupColm = NULL,
  design_libIDrename = NULL, adjustPhase = "predictedPer",
  combinePvalue = "fisher", weightedMethod = TRUE,
  outIntegration = "both", ARSmle = "auto", ARSdefaultPer = 24,
  dayZeroBased = FALSE, outSymbol = "", parallelize = FALSE,
  nCores = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta3d_+3A_datafile">datafile</code></td>
<td>
<p>a character string. The name of data file containing
time-series experimental values of all individuals.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_designfile">designfile</code></td>
<td>
<p>a character string. The name of experimental design file,
at least containing the library ID(column names of <code>datafile</code>),
subject ID(the individual corresponding to each library ID), and
sampling time information of each library ID.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_outdir">outdir</code></td>
<td>
<p>a character string. The name of directory used to store
output files.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_filestyle">filestyle</code></td>
<td>
<p>a character vector(length 1 or 3). The data format of
input files, must be <code>"txt"</code>, or <code>"csv"</code>, or a character
vector containing field separator character(<code>sep</code>), quoting
character(<code>quote</code>), and the character used for decimal
points(<code>dec</code>, for details see <code><a href="utils.html#topic+read.table">read.table</a></code>).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_libcolm">design_libColm</code></td>
<td>
<p>a numeric value. The order index(from left to right)
of the column storing library ID in <code>designfile</code>.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_subjectcolm">design_subjectColm</code></td>
<td>
<p>a numeric value. The order index(from left to
right) of the column storing subject ID in <code>designfile</code>.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_minper">minper</code></td>
<td>
<p>a numeric value. The minimum period length of interested
rhythms. The default is <code>20</code> for circadian rhythms.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_maxper">maxper</code></td>
<td>
<p>a numeric value. The maximum period length of interested
rhythms. The default is <code>28</code> for circadian rhythms.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_cycmethodone">cycMethodOne</code></td>
<td>
<p>a character string. The selected method for analyzing
time-series data of each individual, must be one of <code>"ARS"</code>(ARSER),
<code>"JTK"</code>(JTK_CYCLE), or <code>"LS"</code>(Lomb-Scargle).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_timeunit">timeUnit</code></td>
<td>
<p>a character string. The basic time-unit, must be one of
<code>"day"</code>, <code>"hour"</code>(default for circadian study),
<code>"minute"</code>, or <code>"second"</code> depending on specific experimental
design.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_hrcolm">design_hrColm</code></td>
<td>
<p>a numeric value. The order index(from left to right)
of the column storing time point value-sampling hour information in
<code>designfile</code>. If there is no such column in <code>designfile</code>,
set it as <code>NULL</code>.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_daycolm">design_dayColm</code></td>
<td>
<p>a numeric value. The order index(from left to right)
of the column storing time point value-sampling day information in
<code>designfile</code>. If there is no such column in <code>designfile</code>,
set it as <code>NULL</code>(default).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_mincolm">design_minColm</code></td>
<td>
<p>a numeric value. The order index(from left to right)
of the column storing time point value-sampling minute information in
<code>designfile</code>. If there is no such column in <code>designfile</code>,
set it as <code>NULL</code>(default).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_seccolm">design_secColm</code></td>
<td>
<p>a numeric value. The order index(from left to right)
of the column storing time point value-sampling second information in
<code>designfile</code>. If there is no such column in <code>designfile</code>,
set it as <code>NULL</code>(default).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_groupcolm">design_groupColm</code></td>
<td>
<p>a numeric value. The order index(from left to
right) of the column storing experimental group information of each
individual in <code>designfile</code>. If there is no such column in
<code>designfile</code>, set it as <code>NULL</code>(default) and take all
individuals as one group.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_design_libidrename">design_libIDrename</code></td>
<td>
<p>a character vector(length 2) containing  a
matchable character string in each library ID of <code>designfile</code>, and
a replacement character string. If it is not necessary to replace
characters in library ID of <code>designfile</code>, set it as <code>NULL</code>(
default).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_adjustphase">adjustPhase</code></td>
<td>
<p>a character string. The method used to adjust each
calculated phase before getting integrated phase, must be one of
<code>"predictedPer"</code>(adjust phase with predicted period length)
or <code>"notAdjusted"</code>(not adjust phase).</p>
</td></tr>
<tr><td><code id="meta3d_+3A_combinepvalue">combinePvalue</code></td>
<td>
<p>a character string. The method used to integrate
p-values of multiple individuals, currently only <code>"fisher"</code>(
Fisher's method) could be selected.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_weightedmethod">weightedMethod</code></td>
<td>
<p>logical. If <code>TRUE</code>(default), weighted score
based on p-value of each individual will be used to integrate period,
phase and amplitude values of multiple individuals.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_outintegration">outIntegration</code></td>
<td>
<p>a character string. This parameter controls what
kinds of analysis results will be outputted, must be one of
<code>"both"</code>, <code>"onlyIntegration"</code>, or <code>"noIntegration"</code>.
See <code><a href="#topic+meta2d">meta2d</a></code> for more information.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_arsmle">ARSmle</code></td>
<td>
<p>a character string. The strategy of using MLE method in
<code>"ARS"</code>, must be one of <code>"auto"</code>, <code>"mle"</code>, or
<code>"nomle"</code>. See <code><a href="#topic+meta2d">meta2d</a></code> for more information.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_arsdefaultper">ARSdefaultPer</code></td>
<td>
<p>a numeric value. The expected period length of
interested rhythm, which is a necessary parameter for <code>ARS</code>. See
<code><a href="#topic+meta2d">meta2d</a></code> for more information.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_dayzerobased">dayZeroBased</code></td>
<td>
<p>logical. If <code>TRUE</code>, the first sampling day is
recorded as day zero in the <code>designfile</code>.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_outsymbol">outSymbol</code></td>
<td>
<p>a character string. A common prefix exists in the names of
output files.</p>
</td></tr>
<tr><td><code id="meta3d_+3A_parallelize">parallelize</code></td>
<td>
<p>logical. If <code>TRUE</code>, computation will be done in paralleL
Doesn't work in windows machine</p>
</td></tr>
<tr><td><code id="meta3d_+3A_ncores">nCores</code></td>
<td>
<p>a integer. Bigger or equal to one, number of cores to use</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is originally aimed to analyze large scale periodic data with
individual information. Please pay attention to the data format of
<code>datafile</code> and <code>designfile</code>(see <code>Examples</code> part).
Time-series experimental values(missing values as <code>NA</code>) from
all individuals should be stored in <code>datafile</code>, with the first row
containing all library ID(unique identification number for each sample)
and the first column containing all detected molecular names(eg.
transcript or gene name). The <code>designfile</code> should at least have
three columns-library ID, subject ID and sampling time column.
Experimental group information of each subject ID may be in another
column. In addition, sampling time information may be stored in multiple
columns instead of one column. For example, sampling time-&quot;36 hours&quot; may
be recorded as &quot;day 2&quot;(sampling day column, <code>design_dayColm</code>) plus
&quot;12 hours&quot;(sampling hour column, <code>design_hrColm</code>). The library ID
in <code>datafile</code> and <code>designfile</code> should be same. If there are
different characters between library ID in these two files, try
<code>design_libIDrename</code> to keep them same.
</p>
<p><code>ARS</code>, <code>JTK</code> or <code>LS</code> could be used to analyze time-series
profiles individual by individual. <code>meta3d</code> requires that all
individuals should be analyzed by the same method before integrating
calculated p-value, period, phase, baseline value, amplitude and relative
amplitude values group by group. However, the sampling pattern among
individuals may be different and the requirement of sampling pattern for
each method is not same(see more information about these methods and their
limitations in <code><a href="#topic+meta2d">meta2d</a></code>). Please carefully select a proper
method for the specific dataset. <code>meta3d</code> also help users select
the suitable method through warning notes.
</p>
<p>P-values from different individuals are integrated with Fisher's method
(<code>"fisher"</code>)(Fisher,1925; implementation code from <span class="pkg">MADAM</span>).For
short time-series profiles(eg. 10 time points or less), p-values given by
Lomb-Scargle may be over conservative, which will also lead to
conservative integrated p-values. The integrated period, baseline,
amplitude and relative amplitude values are arithmetic mean of multiple
individuals, respectively. The phase is
<a href="https://en.wikipedia.org/wiki/Mean_of_circular_quantities">
mean of circular quantities</a>(<code>adjustPhase = "predictedPer"</code>)
or a arithmetic mean (<code>adjustPhase = "notAdjusted"</code>) of multiple
individual phases. For completely removing the potential problem of
averaging phases with quite different period length(also mentioned
in <code><a href="#topic+meta2d">meta2d</a></code>), setting <code>minper</code>, <code>maxper</code> and
<code>ARSdefaultPer</code> to a same value may be the only known way. If
<code>weightedMethod = TRUE</code> is selected, weighted scores(
<code>-log10(p-values)</code>) will be taken into account in integrating
period, phase, baseline, amplitude and relative amplitude.
</p>


<h3>Value</h3>

<p><code>meta3d</code> will write analysis results to <code>outdir</code> instead of
returning them as objects. Output files with &quot;meta3dSubjectID&quot; in
the file name are analysis results for each individual. Files named with
&quot;meta3dGroupID&quot; store integrated p-values, period, phase, baseline,
amplitude and relative amplitude values from multiple individuals of
each group and calculated FDR values based on integrated p-values.
</p>


<h3>References</h3>

<p>Glynn E. F., Chen J., and Mushegian  A. R. (2006). Detecting periodic
patterns in unevenly spaced gene expression time series using
Lomb-Scargle periodograms. <em>Bioinformatics</em>, <b>22(3)</b>,
310&ndash;316
</p>
<p>Fisher, R.A. (1925). <em>Statistical methods for research workers</em>.
Oliver and Boyd (Edinburgh).
</p>
<p>Kugler K. G., Mueller L.A., and Graber A. (2010). MADAM - an open source
toolbox for meta-analysis. <em>Source Code for Biology and Medicine</em>,
<b>5</b>, 3.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># write 'cycHumanBloodData' and 'cycHumanBloodDesign' into two 'csv' files
write.csv(cycHumanBloodData, file="cycHumanBloodData.csv",
  row.names=FALSE)
write.csv(cycHumanBloodDesign, file="cycHumanBloodDesign.csv",
  row.names=FALSE)

# detect circadian transcripts with JTK in studied individuals
meta3d(datafile="cycHumanBloodData.csv", cycMethodOne="JTK",
  designfile="cycHumanBloodDesign.csv", outdir="example",
  filestyle="csv", design_libColm=1, design_subjectColm=2,
  design_hrColm=4, design_groupColm=3)
</code></pre>

<hr>
<h2 id='MetaCycle'>MetaCycle
Evaluate Periodicity in Large Scale Data</h2><span id='topic+MetaCycle'></span><span id='topic+MetaCycle-package'></span>

<h3>Description</h3>

<p>MetaCycle
Evaluate Periodicity in Large Scale Data
</p>


<h3>Author(s)</h3>

<p>Gang Wu <a href="mailto:wggucas@gmail.com">wggucas@gmail.com</a>,
Ron Anafi <a href="mailto:ranafi@gmail.com">ranafi@gmail.com</a>,
John Hogenesch <a href="mailto:hogenesch@gmail.com">hogenesch@gmail.com</a>,
Michael Hughes <a href="mailto:michael.evan.hughes@gmail.com">michael.evan.hughes@gmail.com</a>,
Karl Kornacker <a href="mailto:kornacker@midohio.twcbc.com">kornacker@midohio.twcbc.com</a>,
Xavier Li <a href="mailto:haohan.li.97@gmail.com">haohan.li.97@gmail.com</a>,
Matthew Carlucci <a href="mailto:matt.carlucci.96@gmail.com">matt.carlucci.96@gmail.com</a>
Maintainer: Gang Wu <a href="mailto:wggucas@gmail.com">wggucas@gmail.com</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
