<!DOCTYPE html><html><head><title>Help for package metaMA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metaMA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#calcfit2Diffrep'><p> Empirical Bayes statistics from limma analysis with unpaired data</p></a></li>
<li><a href='#directEScombi'><p> Direct effect size combination</p></a></li>
<li><a href='#directpvalcombi'><p> Direct p-value combination</p></a></li>
<li><a href='#effectsize'><p> Calculates effect sizes from given t or moderated t statistics</p></a></li>
<li><a href='#EScombination'><p> Effect size combination for unpaired data</p></a></li>
<li><a href='#EScombination.paired'><p> Effect size combination for paired data</p></a></li>
<li><a href='#IDDIRR'><p> Integration-driven Discovery and Integration-driven Revision Rates</p></a></li>
<li><a href='#metaMA-package'><p> Meta-analysis for MicroArrays</p></a></li>
<li><a href='#pvalcombination'><p> P-value combination for unpaired data</p></a></li>
<li><a href='#pvalcombination.paired'><p> P-value combination for paired data</p></a></li>
<li><a href='#row.ttest.stat'><p> Row t-tests</p></a></li>
<li><a href='#row.ttest.statp'><p> Row paired t-tests</p></a></li>
<li><a href='#rowVars'><p> Row variance of an array</p></a></li>
<li><a href='#Singhdata'><p> Singh dataset</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Meta-Analysis for MicroArrays</td>
</tr>
<tr>
<td>Version:</td>
<td>3.1.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-12</td>
</tr>
<tr>
<td>Author:</td>
<td>Guillemette Marot [aut,cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Samuel Blanck &lt;samuel.blanck@univ-lille.fr&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>limma, SMVar</td>
</tr>
<tr>
<td>Suggests:</td>
<td>GEOquery, org.Hs.eg.db, VennDiagram, annaffy, hgu133plus2.db,
hgu133a.db, hgu95av2.db</td>
</tr>
<tr>
<td>Description:</td>
<td>Combination of either p-values or modified effect sizes from different
    studies to find differentially expressed genes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-12 13:29:12 UTC; sblanck</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-12 16:02:36 UTC</td>
</tr>
</table>
<hr>
<h2 id='calcfit2Diffrep'> Empirical Bayes statistics from limma analysis with unpaired data </h2><span id='topic+calcfit2Diffrep'></span>

<h3>Description</h3>

<p>Computes empirical Bayes statistics from limma analysis with only one group effect. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>calcfit2Diffrep(C1, C2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="calcfit2Diffrep_+3A_c1">C1</code></td>
<td>
<p> Gene expression data of the arrays in the first condition. Each row of <code>C1</code> corresponds to one spot, each column to one replicate. </p>
</td></tr>
<tr><td><code id="calcfit2Diffrep_+3A_c2">C2</code></td>
<td>
<p> Gene expression data of the arrays in the second condition. Each row of <code>C2</code> corresponds to one spot, each column to one replicate. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Returns fit2 described in limma vignette. To be used with unpaired data.
</p>


<h3>Value</h3>

<p>fit2
</p>


<h3>Note</h3>

 
<p>see Bioconductor limma vignette  
</p>

<hr>
<h2 id='directEScombi'> Direct effect size combination </h2><span id='topic+directEScombi'></span>

<h3>Description</h3>

<p>Combines effect sizes already calculated.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>directEScombi(ES, varES, BHth = 0.05, useREM = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="directEScombi_+3A_es">ES</code></td>
<td>
<p>Matrix of effect sizes. Each column of <code>ES</code> corresponds to one study and each row to one gene. </p>
</td></tr>
<tr><td><code id="directEScombi_+3A_vares">varES</code></td>
<td>
<p> Matrix of effect size variances. Each column of <code>varES</code> corresponds to one study and each row to one gene. </p>
</td></tr>
<tr><td><code id="directEScombi_+3A_bhth">BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td></tr>
<tr><td><code id="directEScombi_+3A_userem">useREM</code></td>
<td>
<p> A logical value indicating whether or not to include the between-study variance into the model. </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Combines effect sizes with the method presented in (Choi et al., 2003).
</p>


<h3>Value</h3>

<p>List
</p>
<table>
<tr><td><code>DEindices</code></td>
<td>
<p> Indices of differentially expressed genes at the chosen Benjamini Hochberg threshold.</p>
</td></tr> 
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td></tr> 
</table>


<h3>References</h3>

<p> Choi, J. K., Yu, U., Kim, S., and Yoo, O. J. (2003). Combining multiple microarray
studies and modeling interstudy variation. Bioinformatics, 19 Suppl 1. </p>

<hr>
<h2 id='directpvalcombi'> Direct p-value combination </h2><span id='topic+directpvalcombi'></span>

<h3>Description</h3>

<p>Combines one sided p-values with the inverse normal method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>directpvalcombi(pvalonesided, nrep, BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="directpvalcombi_+3A_pvalonesided">pvalonesided</code></td>
<td>
<p> List of vectors of one sided p-values to be combined. </p>
</td></tr>
<tr><td><code id="directpvalcombi_+3A_nrep">nrep</code></td>
<td>
<p> Vector of numbers of replicates used in each study to calculate the previous one-sided p-values. </p>
</td></tr>
<tr><td><code id="directpvalcombi_+3A_bhth">BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>List
</p>
<table>
<tr><td><code>DEindices</code></td>
<td>
<p> Indices of differentially expressed genes at the chosen Benjamini Hochberg threshold.</p>
</td></tr> 
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td></tr> 
</table>


<h3>Note</h3>

<p> One-sided p-values are required to avoid directional conflicts. Then a two-sided test is performed
to find differentially expressed genes. 
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p> Hedges, L. and Olkin, I. (1985). Statistical Methods for Meta-Analysis. Academic
Press. </p>

<hr>
<h2 id='effectsize'> Calculates effect sizes from given t or moderated t statistics </h2><span id='topic+effectsize'></span>

<h3>Description</h3>

<p>Function not to be used separately.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effectsize(tstat, ntilde, m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="effectsize_+3A_tstat">tstat</code></td>
<td>
<p> Vector of test statistics and effect sizes. </p>
</td></tr>
<tr><td><code id="effectsize_+3A_ntilde">ntilde</code></td>
<td>
<p> Proportion factor between a test statistic and its corresponding effect size. </p>
</td></tr>
<tr><td><code id="effectsize_+3A_m">m</code></td>
<td>
<p> Number of degrees of freedom. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>Matrix with one row per gene, and in column:
</p>
<table>
<tr><td><code>d</code></td>
<td>
<p>Commonly used effect size (which is biased)</p>
</td></tr>
<tr><td><code>vard</code></td>
<td>
<p> Variance of the commonly used effect size</p>
</td></tr>
<tr><td><code>dprime</code></td>
<td>
<p> Unbiased effect size </p>
</td></tr>
<tr><td><code>vardprime</code></td>
<td>
<p> Variance of the unbiased effect size</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Guillemette Marot with contribution from Ankur Ravinarayana Chakravarthy</p>


<h3>References</h3>

<p> Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. Moderated effect size combination for microarray meta-analyses and comparison study. Submitted. </p>


<h3>Examples</h3>

<pre><code class='language-R'>#for SMVar: 
#stati$TestStat[order(stati$GeneId)],length(classes[[i]]),stati$DegOfFreedom[order(stati$GeneId)])
#for Limma
#effectsize(fit2i$t,length(classes[[i]]),(fit2i$df.prior+fit2i$df.residual))
</code></pre>

<hr>
<h2 id='EScombination'> Effect size combination for unpaired data  </h2><span id='topic+EScombination'></span>

<h3>Description</h3>

<p>Calculates effect sizes from unpaired data either from classical or moderated
t-tests (Limma, SMVar) for each study and combines these effect sizes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EScombination(esets, classes, moderated = c("limma", "SMVar", "t")[1], BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EScombination_+3A_esets">esets</code></td>
<td>
<p> List of matrices (or data frames), one matrix per study. Each matrix has one row per gene and one column per replicate and gives the expression data for both conditions with the order specified in the <code>classes</code> argument. All studies must have the same genes.
If the data are already stored as ExpressionSets objects (cf. Bioconductor project), then <code>exprs(yourdata)</code> will give an appropriate element of the list <code>esets</code> used for this function. </p>
</td></tr>
<tr><td><code id="EScombination_+3A_classes">classes</code></td>
<td>
<p> List of class memberships, one per study. Each vector or factor of the list can only contain two levels which correspond to the two conditions studied. </p>
</td></tr>
<tr><td><code id="EScombination_+3A_moderated">moderated</code></td>
<td>
<p> Method to calculate the test statistic inside each study from which the effect size is computed. <code>moderated</code> has to be chosen between &quot;limma&quot;, &quot;SMVar&quot; and &quot;t&quot;. </p>
</td></tr>
<tr><td><code id="EScombination_+3A_bhth">BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>List
</p>
<table>
<tr><td><code>Study1</code></td>
<td>
<p>Vector of indices of differentially expressed genes in study 1. Similar names are given for the other individual studies.</p>
</td></tr>
<tr><td><code>AllIndStudies</code></td>
<td>
<p>Vector of indices of differentially expressed genes found by at least one of the individual studies.</p>
</td></tr>
<tr><td><code>Meta</code></td>
<td>
<p>Vector of indices of differentially expressed genes in the meta-analysis.</p>
</td></tr>
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis. </p>
</td></tr>
</table>


<h3>Note</h3>

<p>While the invisible object resulting from this function contains 
the values described previously, other quantities of interest are printed: 
DE,IDD,Loss,IDR,IRR.
All these quantities are defined in function <code>IDDIDR</code> and in (Marot et al., 2009) 
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p> Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Singhdata)
#Meta-analysis
res=EScombination(esets=Singhdata$esets,classes=Singhdata$classes)
#Number of differentially expressed genes in the meta-analysis
length(res$Meta)
#To plot an histogram of raw p-values
rawpval=2*(1-pnorm(abs(res$TestStatistic)))
hist(rawpval,nclass=100)
</code></pre>

<hr>
<h2 id='EScombination.paired'> Effect size combination for paired data </h2><span id='topic+EScombination.paired'></span>

<h3>Description</h3>

<p>Calculates effect sizes from paired data either from classical or moderated
t-tests (Limma, SMVar) for each study and combines these effect sizes. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>EScombination.paired(logratios, moderated = c("limma", "SMVar", "t")[1], BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EScombination.paired_+3A_logratios">logratios</code></td>
<td>
<p> List of matrices (or data frames). Each matrix has one row per gene and one column per replicate and gives the logratios of one study. All studies must have the same genes.</p>
</td></tr>
<tr><td><code id="EScombination.paired_+3A_moderated">moderated</code></td>
<td>
<p> Method to calculate the test statistic inside each study from which the effect size is computed. <code>moderated</code> has to be chosen between &quot;limma&quot;, &quot;SMVar&quot; and &quot;t&quot;. </p>
</td></tr>
<tr><td><code id="EScombination.paired_+3A_bhth">BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>List
</p>
<table>
<tr><td><code>Study1</code></td>
<td>
<p>Vector of indices of differentially expressed genes in study 1. Similar names are given for the other individual studies.</p>
</td></tr>
<tr><td><code>AllIndStudies</code></td>
<td>
<p>Vector of indices of differentially expressed genes found by at least one of the individual studies.</p>
</td></tr>
<tr><td><code>Meta</code></td>
<td>
<p>Vector of indices of differentially expressed genes in the meta-analysis.</p>
</td></tr>
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis. </p>
</td></tr>
</table>


<h3>Note</h3>

<p>While the invisible object resulting from this function contains 
the values described previously, other quantities of interest are printed: 
DE,IDD,Loss,IDR,IRR.
All these quantities are defined in function <code>IDDIDR</code> and in (Marot et al., 2009) 
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Singhdata)
#create artificially paired data:
artificialdata=lapply(Singhdata$esets,FUN=function(x) (x[,1:10]-x[,11:20]))
#Meta-analysis
res=EScombination.paired(artificialdata)
#Number of differentially expressed genes in the meta-analysis
length(res$Meta)
#To plot an histogram of raw p-values
rawpval=2*(1-pnorm(abs(res$TestStatistic)))
hist(rawpval,nclass=100)
</code></pre>

<hr>
<h2 id='IDDIRR'> Integration-driven Discovery and Integration-driven Revision Rates </h2><span id='topic+IDDIRR'></span>

<h3>Description</h3>

<p>Calculates the gain or the loss of differentially expressed genes due to meta-analysis
compared to individual studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>IDDIRR(finalde, deindst)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="IDDIRR_+3A_finalde">finalde</code></td>
<td>
<p> Vector of indices of differentially expressed genes after meta-analysis </p>
</td></tr>
<tr><td><code id="IDDIRR_+3A_deindst">deindst</code></td>
<td>
<p> Vector of indices of differentially expressed genes found at least in one study </p>
</td></tr>
</table>


<h3>Value</h3>

<table>
<tr><td><code>DE</code></td>
<td>
<p>Number of Differentially Expressed (DE) genes </p>
</td></tr>
<tr><td><code>IDD</code></td>
<td>
<p>Integration Driven Discoveries: number of genes that are declared DE in
the meta-analysis that were not identified in any of
the individual studies alone.</p>
</td></tr>
<tr><td><code>Loss</code></td>
<td>
<p>Number of genes that are declared DE in individual studies but not in meta-analysis.</p>
</td></tr>
<tr><td><code>IDR</code></td>
<td>
<p>Integration-driven Discovery Rate: proportion of genes that are identified as DE in the meta-analysis that were not
identified in any of the individual studies alone.</p>
</td></tr>
<tr><td><code>IRR</code></td>
<td>
<p>Integration-driven Revision Rate: percentage of genes
that are declared DE in individual studies but not in meta-analysis.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Singhdata)
out=EScombination(esets=Singhdata$esets,classes=Singhdata$classes)
IDDIRR(out$Meta,out$AllIndStudies)

## The function is currently defined as
#function(finalde,deindst)
#{
#DE=length(finalde)
#gains=finalde[which(!(finalde %in% deindst))]
#IDD=length(gains)
#IDR=IDD/DE*100
#perte=which(!(deindst %in% finalde))
#Loss=length(perte)
#IRR=Loss/length(deindst)*100
#res=c(DE,IDD,Loss,IDR,IRR)
#names(res)=c("DE","IDD","Loss","IDR","IRR")
#res
#}
</code></pre>

<hr>
<h2 id='metaMA-package'> Meta-analysis for MicroArrays </h2><span id='topic+metaMA-package'></span><span id='topic+metaMA'></span>

<h3>Description</h3>

<p>Combines either p-values or moderated effect sizes from different studies to find differentially expressed genes.</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> metaMA</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 3.1.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2015-01-28</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p><code>pvalcombination</code> and <code>EScombination</code> are the most important functions to combine unpaired data.
</p>
<p><code>pvalcombination</code> combines p-values from individual studies. 
</p>
<p><code>EScombination</code> combines effect sizes from individual studies.
</p>
<p><code>pvalcombination.paired</code> and <code>EScombination.paired</code> are to be used for paired data.
</p>
<p><code>IDDIDR</code> can help in the interpretation of gain and loss of information due to meta-analysis.
</p>


<h3>Author(s)</h3>

<p>Guillemette Marot &lt;guillemette.marot@inria.fr&gt;
</p>


<h3>References</h3>

<p>Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>library(metaMA)
data(Singhdata)
EScombination(esets=Singhdata$esets,classes=Singhdata$classes)
pvalcombination(esets=Singhdata$esets,classes=Singhdata$classes)
#more details are provided in the vignette; only open it in interactive R sessions
if(interactive()){
  vignette("metaMA")
  }
</code></pre>

<hr>
<h2 id='pvalcombination'> P-value combination for unpaired data  </h2><span id='topic+pvalcombination'></span>

<h3>Description</h3>

<p>Calculates differential expression p-values from unpaired data either from classical or moderated
t-tests (Limma, SMVar) for each study and combines these p-values by the inverse normal method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalcombination(esets, classes, moderated = c("limma", "SMVar", "t")[1], BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalcombination_+3A_esets">esets</code></td>
<td>
<p> List of matrices (or data frames), one matrix per study. Each matrix has one row per gene and one column per replicate and gives the expression data for both conditions with the order specified in the <code>classes</code> argument. All studies must have the same genes.
If the data are already stored as ExpressionSets objects (cf. Bioconductor project), then <code>exprs(yourdata)</code> will give an appropriate element of the list <code>esets</code> used for this function. </p>
</td></tr>
<tr><td><code id="pvalcombination_+3A_classes">classes</code></td>
<td>
<p> List of class memberships, one per study. Each vector or factor of the list can only contain two levels which correspond to the two conditions studied. </p>
</td></tr>
<tr><td><code id="pvalcombination_+3A_moderated">moderated</code></td>
<td>
<p> Method to calculate the test statistic inside each study from which the p-value is computed. <code>moderated</code> has to be chosen between &quot;limma&quot;, &quot;SMVar&quot; and &quot;t&quot;. </p>
</td></tr>
<tr><td><code id="pvalcombination_+3A_bhth">BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>List
</p>
<table>
<tr><td><code>Study1</code></td>
<td>
<p>Vector of indices of differentially expressed genes in study 1. Similar names are given for the other individual studies.</p>
</td></tr>
<tr><td><code>AllIndStudies</code></td>
<td>
<p>Vector of indices of differentially expressed genes found by at least one of the individual studies.</p>
</td></tr>
<tr><td><code>Meta</code></td>
<td>
<p>Vector of indices of differentially expressed genes in the meta-analysis.</p>
</td></tr>
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td></tr> 
</table>


<h3>Note</h3>

<p>While the invisible object resulting from this function contains 
the values described previously, other quantities of interest are printed: 
DE,IDD,Loss,IDR,IRR.
All these quantities are defined in function <code>IDDIDR</code> and in (Marot et al., 2009) 
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Singhdata)
#Meta-analysis
res=pvalcombination(esets=Singhdata$esets,classes=Singhdata$classes)
#Number of differentially expressed genes in the meta-analysis
length(res$Meta)
#To plot an histogram of raw p-values
rawpval=2*(1-pnorm(abs(res$TestStatistic)))
hist(rawpval,nclass=100)
</code></pre>

<hr>
<h2 id='pvalcombination.paired'> P-value combination for paired data </h2><span id='topic+pvalcombination.paired'></span>

<h3>Description</h3>

<p>Calculates differential expression p-values from paired data either from classical or moderated
t-tests (Limma, SMVar) for each study and combines these p-values by the inverse normal method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pvalcombination.paired(logratios, moderated = c("limma", "SMVar", "t")[1], BHth = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pvalcombination.paired_+3A_logratios">logratios</code></td>
<td>
<p> List of matrices. Each matrix has one row per gene and one column per replicate and gives the logratios of one study. All studies must have the same genes.</p>
</td></tr>
<tr><td><code id="pvalcombination.paired_+3A_moderated">moderated</code></td>
<td>
<p> Method to calculate the test statistic inside each study from which the effect size is computed. <code>moderated</code> has to be chosen between &quot;limma&quot;, &quot;SMVar&quot; and &quot;t&quot;. </p>
</td></tr>
<tr><td><code id="pvalcombination.paired_+3A_bhth">BHth</code></td>
<td>
<p> Benjamini Hochberg threshold. By default, the False Discovery Rate is controlled at 5%. </p>
</td></tr>
</table>


<h3>Value</h3>

<p>List
</p>
<table>
<tr><td><code>Study1</code></td>
<td>
<p>Vector of indices of differentially expressed genes in study 1. Similar names are given for the other individual studies.</p>
</td></tr>
<tr><td><code>AllIndStudies</code></td>
<td>
<p>Vector of indices of differentially expressed genes found by at least one of the individual studies.</p>
</td></tr>
<tr><td><code>Meta</code></td>
<td>
<p>Vector of indices of differentially expressed genes in the meta-analysis.</p>
</td></tr>
<tr><td><code>TestStatistic</code></td>
<td>
<p>Vector with test statistics for differential expression in the meta-analysis.</p>
</td></tr> 
</table>


<h3>Note</h3>

<p>While the invisible object resulting from this function contains 
the values described previously, other quantities of interest are printed: 
DE,IDD,Loss,IDR,IRR.
All these quantities are defined in function <code>IDDIDR</code> and in (Marot et al., 2009) 
</p>


<h3>Author(s)</h3>

<p> Guillemette Marot </p>


<h3>References</h3>

<p>  Marot, G., Foulley, J.-L., Mayer, C.-D., Jaffrezic, F. (2009) Moderated effect size and p-value combinations for microarray meta-analyses. Bioinformatics. 25 (20): 2692-2699. </p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Singhdata)
#create artificially paired data:
artificialdata=lapply(Singhdata$esets,FUN=function(x) (x[,1:10]-x[,11:20]))
#Meta-analysis
res=pvalcombination.paired(artificialdata)
#Number of differentially expressed genes in the meta-analysis
length(res$Meta)
#To plot an histogram of raw p-values
rawpval=2*(1-pnorm(abs(res$TestStatistic)))
hist(rawpval,nclass=100)
</code></pre>

<hr>
<h2 id='row.ttest.stat'> Row t-tests </h2><span id='topic+row.ttest.stat'></span>

<h3>Description</h3>

<p>Performs t-tests for unpaired data row by row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>row.ttest.stat(mat1, mat2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="row.ttest.stat_+3A_mat1">mat1</code></td>
<td>
<p> Matrix with data for the first condition </p>
</td></tr>
<tr><td><code id="row.ttest.stat_+3A_mat2">mat2</code></td>
<td>
<p> Matrix with data for the second condition </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is much faster than employing apply with FUN=t.test
</p>


<h3>Value</h3>

<p>Vector with t-test statistics
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The function is currently defined as
function(mat1,mat2){ 
n1&lt;-dim(mat1)[2]
n2&lt;-dim(mat2)[2] 
n&lt;-n1+n2 
m1&lt;-rowMeans(mat1,na.rm=TRUE) 
m2&lt;-rowMeans(mat2,na.rm=TRUE) 
v1&lt;-rowVars(mat1,na.rm=TRUE) 
v2&lt;-rowVars(mat2,na.rm=TRUE) 
vpool&lt;-(n1-1)/(n-2)*v1 + (n2-1)/(n-2)*v2 
tstat&lt;-sqrt(n1*n2/n)*(m2-m1)/sqrt(vpool) 
return(tstat)}
</code></pre>

<hr>
<h2 id='row.ttest.statp'> Row paired t-tests </h2><span id='topic+row.ttest.statp'></span>

<h3>Description</h3>

<p>Performs t-tests for paired data row by row.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>row.ttest.statp(mat)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="row.ttest.statp_+3A_mat">mat</code></td>
<td>
<p> Matrix with data to be tested (for example, log-ratios in microarray experiments). </p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is much faster than employing apply with FUN=t.test.
</p>


<h3>Value</h3>

<p>Vector with t-test statistics.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The function is currently defined as
function(mat){ 
m&lt;-rowMeans(mat,na.rm=TRUE) 
sd&lt;-rowSds(mat,na.rm=TRUE)  
tstat&lt;-m/(sd*sqrt(1/dim(mat)[2])) 
return(tstat)}
</code></pre>

<hr>
<h2 id='rowVars'> Row variance of an array </h2><span id='topic+rowVars'></span>

<h3>Description</h3>

<p>Calculates variances of each row of an array
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rowVars(x, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rowVars_+3A_x">x</code></td>
<td>
<p> Array of two or more dimensions, containing numeric, complex, integer or logical values, or a numeric data frame. </p>
</td></tr>
<tr><td><code id="rowVars_+3A_na.rm">na.rm</code></td>
<td>
<p> Logical. Should missing values (including NaN) be omitted from the calculations? </p>
</td></tr>
</table>


<h3>Details</h3>

<p> This function is the same as applying apply with FUN=var but is a lot faster.</p>


<h3>Value</h3>

<p>A numeric or complex array of suitable size, or a vector if the result is one-dimensional. The dimnames (or names for a vector result) are taken from the original array.</p>


<h3>Examples</h3>

<pre><code class='language-R'>## The function is currently defined as
function (x,na.rm = TRUE) 
{
    sqr = function(x) x * x
    n = rowSums(!is.na(x))
    n[n &lt;= 1] = NA
    return(rowSums(sqr(x - rowMeans(x,na.rm = na.rm)), na.rm = na.rm)/(n - 1))
  }
</code></pre>

<hr>
<h2 id='Singhdata'> Singh dataset</h2><span id='topic+Singhdata'></span>

<h3>Description</h3>

<p>Publicly available microarray dataset artificially split in 5 studies </p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Singhdata)</code></pre>


<h3>Format</h3>

<p>List of 3 elements:
</p>

<dl>
<dt> esets </dt><dd><p>List of 5 data frames corresponding to 5 artificial studies, each with 12625 genes and 20 replicates (10 normal samples and 10 tumoral samples)</p>
</dd>
<dt> classes </dt><dd><p>List of 5 numeric vectors with class memberships, one per study</p>
</dd> 
<dt> geneNames </dt><dd><p>Factor with 12625 levels corresponding to gene names</p>
</dd>
</dl>



<h3>Source</h3>

<p> These data are available on the website <a href="http://www.bioinf.ucd.ie/people/ian/">http://www.bioinf.ucd.ie/people/ian/</a>. We considered 50 normal samples and 50 tumoral samples, leaving out the 2 last tumoral samples.
Data are already normalized. </p>


<h3>References</h3>

<p>Singh, D., Febbo, P. G., Ross, K., Jackson, D. G., Manola, J., Ladd, C., Tamayo, P.,
Renshaw, A. A., D'Amico, A. V., Richie, J. P., Lander, E. S., Loda, M., Kantoff,
P.W., Golub, T. R., and Sellers,W. R. (2002). Gene expression correlates of clinical prostate cancer behavior. Cancer Cell, 1(2). 203:209.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Singhdata)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
