<!DOCTYPE html><html><head><title>Help for package metaGE</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metaGE}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#metaGE-package'><p>metaGE: Meta-Analysis for Detecting Genotype x Environment Associations</p></a></li>
<li><a href='#autocor'><p>Computation of the autocorrelation</p></a></li>
<li><a href='#CheckContrast'><p>Check and reformat the matrix of contrast</p></a></li>
<li><a href='#CheckIncidence'><p>Check and reformat the matrix of incidence</p></a></li>
<li><a href='#ContrastStatTest'><p>Compute the statistic of the contrast test.</p></a></li>
<li><a href='#ContrastStatTest.NA'><p>Compute the statistic of the contrast test in presence of missing values</p></a></li>
<li><a href='#envDesc'><p>Description of the environments.</p></a></li>
<li><a href='#FastKerFdr'><p>FastKerFdr</p></a></li>
<li><a href='#GetH0Items'><p>GetH0Items</p></a></li>
<li><a href='#lindley'><p>Computation of the lindley process from scores.</p></a></li>
<li><a href='#LLikelihoodT_vect'><p>LLikelihoodT_vect</p></a></li>
<li><a href='#MakeQQplot'><p>Drawing a QQplot</p></a></li>
<li><a href='#metaData'><p>Results of different GWAS.</p></a></li>
<li><a href='#metaGE.collect'><p>Collect the results of GWAS data from different files</p></a></li>
<li><a href='#metaGE.cor'><p>Infer inter-environment correlation matrix</p></a></li>
<li><a href='#metaGE.fit'><p>Meta-analysis procedure: Fixed or Random effect.</p></a></li>
<li><a href='#metaGE.heatmap'><p>Draw the heatmap to see markers effects across environments.</p></a></li>
<li><a href='#metaGE.incidence'><p>Create the matrix of incidence.</p></a></li>
<li><a href='#metaGE.lscore'><p>Compute the local score from a set of pvalues.</p></a></li>
<li><a href='#metaGE.manhattan'><p>Draw the Manhattan plot.</p></a></li>
<li><a href='#metaGE.pvalplot'><p>Display visual checks of pvalues.</p></a></li>
<li><a href='#metaGE.regplot'><p>Plot the z-score of a marker according to a covariate.</p></a></li>
<li><a href='#metaGE.test'><p>Meta-analysis test for Genotype x Environment interactions: Contrast or Regression.</p></a></li>
<li><a href='#ReadData'><p>ReadData</p></a></li>
<li><a href='#RegressionStatTest'><p>Compute the pvalue of the meta-regression test.</p></a></li>
<li><a href='#RegressionStatTestNA'><p>Compute the pvalue of the regression test in presence of missing values.</p></a></li>
<li><a href='#sig_sl'><p>Computation of the significative regions</p></a></li>
<li><a href='#thresUnif'><p>Computation of the significance threshold</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Meta-Analysis for Detecting Genotype x Environment Associations</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Annaïg De Walsche &lt;annaig.de-walsche@inrae.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Meta-analysis of genome-wide association studies for studying
    Genotype x Environment interactions.  The 4 main functions of the
    package metaGE.collect(), metaGE.cor(), metaGE.fit() and metaGE.test()
    correspond to 4 steps to perform the meta-analysis: Collecting the
    results of genome-wide association studies data from different files;
    Inferring the inter-environment correlation matrix; Performing global
    test procedure for quantitative trait loci detection (using a Fixed or
    Random effect model); Performing tests of contrast or meta-regression
    using an environmental co-factor. (De Walsche, A., et al. (2023)
    &lt;<a href="https://doi.org/10.1101%2F2023.03.01.530237">doi:10.1101/2023.03.01.530237</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>corrplot, data.table, dplyr, emdbook, future, ggplot2,
ggrepel, gplots, graphics, grDevices, ks, purrr, qqman, Rfast,
stats, stringr, tibble, tidyr, utils, viridis</td>
</tr>
<tr>
<td>Suggests:</td>
<td>DT, knitr, rmarkdown, tidyverse</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-09-28 08:31:12 UTC; Annaig</td>
</tr>
<tr>
<td>Author:</td>
<td>Annaïg De Walsche <a href="https://orcid.org/0000-0003-0603-1716"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Tristan Mary-Huard
    <a href="https://orcid.org/0000-0002-3839-9067"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-09-28 08:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='metaGE-package'>metaGE: Meta-Analysis for Detecting Genotype x Environment Associations</h2><span id='topic+metaGE'></span><span id='topic+metaGE-package'></span>

<h3>Description</h3>

<p>Meta-analysis of genome-wide association studies for studying Genotype x Environment interactions. The 4 main functions of the package metaGE.collect(), metaGE.cor(), metaGE.fit() and metaGE.test() correspond to 4 steps to perform the meta-analysis: Collecting the results of genome-wide association studies data from different files; Inferring the inter-environment correlation matrix; Performing global test procedure for quantitative trait loci detection (using a Fixed or Random effect model); Performing tests of contrast or meta-regression using an environmental co-factor. (De Walsche, A., et al. (2023) <a href="https://doi.org/10.1101/2023.03.01.530237">doi:10.1101/2023.03.01.530237</a>).
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Annaïg De Walsche <a href="mailto:annaig.de-walsche@inrae.fr">annaig.de-walsche@inrae.fr</a> (<a href="https://orcid.org/0000-0003-0603-1716">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> Tristan Mary-Huard <a href="mailto:tristan.mary-huard@agroparistech.fr">tristan.mary-huard@agroparistech.fr</a> (<a href="https://orcid.org/0000-0002-3839-9067">ORCID</a>)
</p>
</li></ul>


<hr>
<h2 id='autocor'>Computation of the autocorrelation</h2><span id='topic+autocor'></span>

<h3>Description</h3>

<p>The function autocor computes the autocorrelation.(function from localscore)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>autocor(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="autocor_+3A_x">x</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the autocorrelation.
</p>

<hr>
<h2 id='CheckContrast'>Check and reformat the matrix of contrast</h2><span id='topic+CheckContrast'></span>

<h3>Description</h3>

<p>The function CheckContrast check and reformat the matrix of  contrast.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckContrast(Contrast, ContrastName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckContrast_+3A_contrast">Contrast</code></td>
<td>
<p>A matrix of contrast.</p>
</td></tr>
<tr><td><code id="CheckContrast_+3A_contrastname">ContrastName</code></td>
<td>
<p>The name of the contrast.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The matrix of contrast in the right format.
</p>

<hr>
<h2 id='CheckIncidence'>Check and reformat the matrix of incidence</h2><span id='topic+CheckIncidence'></span>

<h3>Description</h3>

<p>The function CheckIncidence check and reformat the matrix of incidence.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CheckIncidence(Incidence, IncidenceName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CheckIncidence_+3A_incidence">Incidence</code></td>
<td>
<p>A matrix of incidence, as obtained from metaGE.incidence.</p>
</td></tr>
<tr><td><code id="CheckIncidence_+3A_incidencename">IncidenceName</code></td>
<td>
<p>The name of the incidence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The matrix of incidence in the right format.
</p>

<hr>
<h2 id='ContrastStatTest'>Compute the statistic of the contrast test.</h2><span id='topic+ContrastStatTest'></span>

<h3>Description</h3>

<p>The function ContrastStatTest compute the statistic of the contrast test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContrastStatTest(Incidence, Contrast = NULL, Zmat, MatCorr, IncidenceName)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContrastStatTest_+3A_incidence">Incidence</code></td>
<td>
<p>A matrix of incidence, as obtained from metaGE.incidence.</p>
</td></tr>
<tr><td><code id="ContrastStatTest_+3A_contrast">Contrast</code></td>
<td>
<p>A matrix of contrast, if NULL the identity matrix is used. (NULL by default)</p>
</td></tr>
<tr><td><code id="ContrastStatTest_+3A_zmat">Zmat</code></td>
<td>
<p>A matrix containing the Zscores of all markers (in rows) in each environment (in columns).</p>
</td></tr>
<tr><td><code id="ContrastStatTest_+3A_matcorr">MatCorr</code></td>
<td>
<p>The inter-environments correlation matrix. Can be computed using MetaGE.cor.</p>
</td></tr>
<tr><td><code id="ContrastStatTest_+3A_incidencename">IncidenceName</code></td>
<td>
<p>The name of the incidence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset of two columns containing the pvalue of the test of contrast and the minimum number of environment per group of all markers.
</p>

<hr>
<h2 id='ContrastStatTest.NA'>Compute the statistic of the contrast test in presence of missing values</h2><span id='topic+ContrastStatTest.NA'></span>

<h3>Description</h3>

<p>The function ContrastStatTest compute the statistic of the contrast test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ContrastStatTest.NA(
  Incidence,
  Contrast = NULL,
  Zmat,
  MatCorr,
  Data,
  Configs.list,
  IncidenceName
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ContrastStatTest.NA_+3A_incidence">Incidence</code></td>
<td>
<p>A matrix of incidence, as obtained from metaGE.incidence.</p>
</td></tr>
<tr><td><code id="ContrastStatTest.NA_+3A_contrast">Contrast</code></td>
<td>
<p>A matrix of contrast, if NULL the identity matrix is used. (NULL by default)</p>
</td></tr>
<tr><td><code id="ContrastStatTest.NA_+3A_zmat">Zmat</code></td>
<td>
<p>A matrix containing the Zscores of all markers (in rows) in each environment (in columns).</p>
</td></tr>
<tr><td><code id="ContrastStatTest.NA_+3A_matcorr">MatCorr</code></td>
<td>
<p>The inter-environments correlation matrix. Can be computed using MetaGE.cor.</p>
</td></tr>
<tr><td><code id="ContrastStatTest.NA_+3A_data">Data</code></td>
<td>
<p>A dataset containing the effect, the pvalues and the na configuration for all marker</p>
</td></tr>
<tr><td><code id="ContrastStatTest.NA_+3A_configs.list">Configs.list</code></td>
<td>
<p>A vector containing the NA configurations present in the dataset</p>
</td></tr>
<tr><td><code id="ContrastStatTest.NA_+3A_incidencename">IncidenceName</code></td>
<td>
<p>The name of the incidence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset of two columns containing the pvalue of the test of contrast and the minimum number of environment per group of all markers.
</p>

<hr>
<h2 id='envDesc'>Description of the environments.</h2><span id='topic+envDesc'></span>

<h3>Description</h3>

<p>A dataset containing variables describing the 22 environments.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>envDesc
</code></pre>


<h3>Format</h3>

<p>A data frame with 22 rows and 3 variables:
</p>

<ul>
<li><p> FileName: environment name
</p>
</li>
<li><p> Temp: temperature
</p>
</li>
<li><p> Water: water condition
</p>
</li></ul>


<hr>
<h2 id='FastKerFdr'>FastKerFdr</h2><span id='topic+FastKerFdr'></span>

<h3>Description</h3>

<p>Computes H1 posteriors of the Z-scores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FastKerFdr(Z, p0, plotting = FALSE, NbKnot = 1e+05, tol = 1e-05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="FastKerFdr_+3A_z">Z</code></td>
<td>
<p>A vector containing Zscores</p>
</td></tr>
<tr><td><code id="FastKerFdr_+3A_p0">p0</code></td>
<td>
<p>A double between 0 and 1. A priori proportion of H0 hypotheses</p>
</td></tr>
<tr><td><code id="FastKerFdr_+3A_plotting">plotting</code></td>
<td>
<p>A boolean saying to plot or not (FALSE by default)</p>
</td></tr>
<tr><td><code id="FastKerFdr_+3A_nbknot">NbKnot</code></td>
<td>
<p>The (maximum) number of knot for the kde procedure.(1e5 by default)</p>
</td></tr>
<tr><td><code id="FastKerFdr_+3A_tol">tol</code></td>
<td>
<p>a tolerance value for convergence (1e-5 by default)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tau is the vector of H1 posteriors
</p>

<hr>
<h2 id='GetH0Items'>GetH0Items</h2><span id='topic+GetH0Items'></span>

<h3>Description</h3>

<p>This function give the index of the markers which seems not significant (under H0)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetH0Items(Zmat, Threshold = 0.8, plotting = FALSE, Cores = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetH0Items_+3A_zmat">Zmat</code></td>
<td>
<p>A matrix containing the Zscore (in rows) for each environment (in columns)</p>
</td></tr>
<tr><td><code id="GetH0Items_+3A_threshold">Threshold</code></td>
<td>
<p>Threshold on posteriors (to be H1) to filter markers for correlation computation (0.6 by default)</p>
</td></tr>
<tr><td><code id="GetH0Items_+3A_plotting">plotting</code></td>
<td>
<p>A boolean saying to plot or not (FALSE by default)</p>
</td></tr>
<tr><td><code id="GetH0Items_+3A_cores">Cores</code></td>
<td>
<p>The number of cores to used, optional. By default, availableCores()-1 cores is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of index of markers which seems not significant (under H0)
</p>

<hr>
<h2 id='lindley'>Computation of the lindley process from scores.</h2><span id='topic+lindley'></span>

<h3>Description</h3>

<p>The function lindley computes the lindley process from scores.(function from localscore)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lindley(scores)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lindley_+3A_scores">scores</code></td>
<td>
<p>A numeric vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the lindley.
</p>

<hr>
<h2 id='LLikelihoodT_vect'>LLikelihoodT_vect</h2><span id='topic+LLikelihoodT_vect'></span>

<h3>Description</h3>

<p>This function compute the values of loglikelihood for all markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LLikelihoodT_vect(Zmat, Delta, P, Mu, Tau)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LLikelihoodT_vect_+3A_zmat">Zmat</code></td>
<td>
<p>A matrix containing the Zscores of all markers (in rows) in each environment (in columns)</p>
</td></tr>
<tr><td><code id="LLikelihoodT_vect_+3A_delta">Delta</code></td>
<td>
<p>A vector containing the diagonal coefficients of the diagonal matrix obtained by the diagonalization of the correlation matrix</p>
</td></tr>
<tr><td><code id="LLikelihoodT_vect_+3A_p">P</code></td>
<td>
<p>Matrix such that MatCorr = P Delta t(P), with Delta diagonal</p>
</td></tr>
<tr><td><code id="LLikelihoodT_vect_+3A_mu">Mu</code></td>
<td>
<p>A vector containing the average effect of the markers</p>
</td></tr>
<tr><td><code id="LLikelihoodT_vect_+3A_tau">Tau</code></td>
<td>
<p>A vector containing the heterogeneity between environments of the markers</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector containing the value of the Log-Likelihood of all markers
</p>

<hr>
<h2 id='MakeQQplot'>Drawing a QQplot</h2><span id='topic+MakeQQplot'></span>

<h3>Description</h3>

<p>The function MakeQQplot displays the QQplot of the -log10(pvalues).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MakeQQplot(Pvalues, Name = NULL, Xrange = NULL, Yrange = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MakeQQplot_+3A_pvalues">Pvalues</code></td>
<td>
<p>A vector containing pvalues.</p>
</td></tr>
<tr><td><code id="MakeQQplot_+3A_name">Name</code></td>
<td>
<p>A name of the corresponding test. (optional)</p>
</td></tr>
<tr><td><code id="MakeQQplot_+3A_xrange">Xrange</code></td>
<td>
<p>A range for the x axis. (optional)</p>
</td></tr>
<tr><td><code id="MakeQQplot_+3A_yrange">Yrange</code></td>
<td>
<p>A range for the y axis. (optional)</p>
</td></tr>
</table>

<hr>
<h2 id='metaData'>Results of different GWAS.</h2><span id='topic+metaData'></span>

<h3>Description</h3>

<p>A dataset containing the results of 10 different genetic association studies testing the association between a set of 25,436 markers and the grain yield.
The data are extrated from: Drops Amaizing available on the https://doi.org/10.15454/6TL2N4 website.
This dataset were obtained thanks to the <em>metaGE.collect</em> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaData
</code></pre>


<h3>Format</h3>

<p>A data frame with 25,436 rows and 35 variables:
</p>

<ul>
<li><p> CHR: chromosome of the marker
</p>
</li>
<li><p> POS: position of the marker
</p>
</li>
<li><p> MARKER: name of the marker
</p>
</li>
<li><p> FREQ.env: maf of the marker in the environment env
</p>
</li>
<li><p> EFFECT.env: regression coefficient of the marker in the environment env
</p>
</li>
<li><p> EFFECT_SE.env: standard error of the regression coefficient of the marker in the environment env
</p>
</li>
<li><p> PVAL.env: pvalue of the marker in the environment env
</p>
</li>
<li><p> WEIGHT.env: weight of the marker in the environment env
</p>
</li>
<li><p> ALLELE0: allele0
</p>
</li>
<li><p> ALLELE1: allele1
</p>
</li></ul>


<hr>
<h2 id='metaGE.collect'>Collect the results of GWAS data from different files</h2><span id='topic+metaGE.collect'></span>

<h3>Description</h3>

<p>This function merges files containing the summary statistics of GWAS in different environments (one file per environment).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.collect(
  FileNames,
  VariableNames,
  MinFreq = 0,
  DropDuplicates = TRUE,
  Verbose = FALSE,
  NA.rmv = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.collect_+3A_filenames">FileNames</code></td>
<td>
<p>A list containing the file paths to merge (one trait only) or a list of such lists</p>
</td></tr>
<tr><td><code id="metaGE.collect_+3A_variablenames">VariableNames</code></td>
<td>
<p>A named list containing the column names in the original files corresponding to the variables below:
</p>

<ul>
<li><p> MARKER, CHR, POS, EFFECT, EFFECT_SE, PVAL,
</p>
</li>
<li><p> (optional: FREQ, ALLELE0, ALLELE1, WEIGHT)
or a list of such lists.
</p>
</li></ul>
</td></tr>
<tr><td><code id="metaGE.collect_+3A_minfreq">MinFreq</code></td>
<td>
<p>A numeric value allowing to filter markers based on the maf. (optional)</p>
</td></tr>
<tr><td><code id="metaGE.collect_+3A_dropduplicates">DropDuplicates</code></td>
<td>
<p>A boolean indicating whether duplicate markers should be removed or not. (TRUE by default)</p>
</td></tr>
<tr><td><code id="metaGE.collect_+3A_verbose">Verbose</code></td>
<td>
<p>A boolean indicating whether progression messages should be printed or not. (FALSE by default)</p>
</td></tr>
<tr><td><code id="metaGE.collect_+3A_na.rmv">NA.rmv</code></td>
<td>
<p>A boolean should the NA be removed or not (TRUE by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each file MUST contain the variables below:
</p>

<ul>
<li><p> MARKER =  marker,
</p>
</li>
<li><p> CHR = the chromosome ,
</p>
</li>
<li><p> POS = the position of the marker on the chromosome,
</p>
</li>
<li><p> EFFECT = the mean effect of the marker,
</p>
</li>
<li><p> EFFECT_SE = the standard error of the mean effect,
</p>
</li>
<li><p> PVAL = the pvalue of the mean effect.
Each file might contain the variables:
</p>
</li>
<li><p> FREQ = MAF
</p>
</li>
<li><p> ALLELE0
</p>
</li>
<li><p> ALLELE1
</p>
</li>
<li><p> WEIGHT
</p>
</li></ul>



<h3>Value</h3>

<p>A list of:
</p>

<ul>
<li><p> Data -&gt;  a tibble containing all the columns of interest of all the files from FileNames,
</p>
</li>
<li><p> RemovedMarkers -&gt; same kind of tibble, but containing the markers that have been removed due to unclear allele coding.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>require(dplyr)
require(tibble)
require(stringr)
RepData &lt;- system.file("extdata", package = "metaGE")
# Get the complete list of association files
File.list &lt;- list.files(RepData ,full.names = TRUE) %&gt;%
            tibble(Names = .) %&gt;%
            mutate(ShortNames = Names %&gt;%
                  str_remove(pattern = paste0(RepData,"/")) %&gt;%
                  str_remove(pattern = "_DF.txt"))  %&gt;%
            select(ShortNames,Names) %&gt;%
            deframe
###Build the dataset
## First provide the list of variable names
Names.list &lt;- list(MARKER="Marker_Name",
                  CHR="Chromosome",
                  POS="Marker_Position",
                  FREQ="Maf",
                  EFFECT="SNP_Weight",
                  PVAL="Pvalue",
                  ALLELE0="Allele1",
                  ALLELE1="Allele2")

MinFreq &lt;- 0.07

## Now collect
metaData &lt;- metaGE.collect(File.list, Names.list,MinFreq = MinFreq)
</code></pre>

<hr>
<h2 id='metaGE.cor'>Infer inter-environment correlation matrix</h2><span id='topic+metaGE.cor'></span>

<h3>Description</h3>

<p>This function infer the inter-environment correlation matrix from the z-scores after filtering markers with high probability of being under H1..
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.cor(Data, Threshold = 0.6, NA.omit = TRUE, Cores = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.cor_+3A_data">Data</code></td>
<td>
<p>A dataset containing the effects and pvalues of each marker (in rows) in each environment (in columns) as obtained by metaGE.collect</p>
</td></tr>
<tr><td><code id="metaGE.cor_+3A_threshold">Threshold</code></td>
<td>
<p>Threshold on posteriors (to be H1) to filter markers before computing correlation (0.6 by default)</p>
</td></tr>
<tr><td><code id="metaGE.cor_+3A_na.omit">NA.omit</code></td>
<td>
<p>A boolean: should the NA be removed for the inter-environment correlation matrix computation (default=TRUE)</p>
</td></tr>
<tr><td><code id="metaGE.cor_+3A_cores">Cores</code></td>
<td>
<p>The number of cores to used, optional. By default, availableCores()-1 cores is used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The inter-environment correlation matrix
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(corrplot)
data("metaData")
Threshold &lt;- 0.8
matCorr &lt;- metaGE.cor(metaData, Threshold = Threshold)
#corrplot(matCorr,order = "hclust")
</code></pre>

<hr>
<h2 id='metaGE.fit'>Meta-analysis procedure: Fixed or Random effect.</h2><span id='topic+metaGE.fit'></span>

<h3>Description</h3>

<p>Quantitative trait loci detection via Fixed or Random effect meta-analysis GWAS procedure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.fit(Data, MatCorr, Method, NA.omit = TRUE, DropZScores = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.fit_+3A_data">Data</code></td>
<td>
<p>A dataset containing the estimated marker effect and its associated pvalue of each marker (in rows) in each environment (in columns), as obtained from metaGE.collect.</p>
</td></tr>
<tr><td><code id="metaGE.fit_+3A_matcorr">MatCorr</code></td>
<td>
<p>The inter-environments correlation matrix. Can be computed using metaGE.cor.</p>
</td></tr>
<tr><td><code id="metaGE.fit_+3A_method">Method</code></td>
<td>
<p>A string specifying the method to be performed: either 'Fe' or 'Re'.</p>
</td></tr>
<tr><td><code id="metaGE.fit_+3A_na.omit">NA.omit</code></td>
<td>
<p>A boolean specifying whether the markers with some NA values should be removed. (TRUE by default)</p>
</td></tr>
<tr><td><code id="metaGE.fit_+3A_dropzscores">DropZScores</code></td>
<td>
<p>A boolean specifying whether the Zscores should be dropped from the dataset or not.(FALSE by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Different tests may be performed:
</p>

<ul>
<li><p> Fixed Effect (Fe), to identify markers with a stable effect across environments.
</p>
</li>
<li><p> Random Effect (Re), to identify markers whose effects may be unstable across environments.
</p>
</li></ul>



<h3>Value</h3>

<p>The dataset Data with supplementary  columns:
</p>

<ul>
<li><p> Mu: Estimation of Mu,
</p>
</li>
<li><p> Tau: Estimation of Tau, the heterogeneity,
</p>
</li>
<li><p> Pvalue: The Pvalue of the test,
</p>
</li>
<li><p> the Zscores for each environment if DropLocalScores = FALSE.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>require(dplyr)
# Import the data
data("metaData")

# Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

# Fixed Effect
FeDF &lt;- metaGE.fit(metaData, matCorr, Method = "Fe")
head(FeDF %&gt;% select(CHR, POS, MARKER, Mu, Tau, PVALUE))

# Random Effect
ReDF &lt;- metaGE.fit(metaData, matCorr, Method = "Re")
head(ReDF %&gt;% select(CHR, POS, MARKER, Mu, Tau, PVALUE))

</code></pre>

<hr>
<h2 id='metaGE.heatmap'>Draw the heatmap to see markers effects across environments.</h2><span id='topic+metaGE.heatmap'></span>

<h3>Description</h3>

<p>The function metaGE.heatmap displays the heatmap of the zscores, the estimated marker effects or the pvalues of each markers (in rows) in each environments (in columns).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.heatmap(
  Data,
  Prefix = "Z.",
  EnvGroups = NULL,
  QTLsVarName = NULL,
  RowOrder = TRUE,
  ColOrder = TRUE,
  ShowDendrogram = FALSE,
  Colors = c("red", "black", "green"),
  Main = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.heatmap_+3A_data">Data</code></td>
<td>
<p>A dataset containing the zscores, the effects or the pvalues of each marker (in rows) in each environment (in columns), as obtained from metaGE.fit.</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_prefix">Prefix</code></td>
<td>
<p>The prefix of the score to display in the heatmap: 'Z.' for the zscores, 'EFFECT.' for the effects and 'PVAL.' for the pvalues.('Z.' by default)</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_envgroups">EnvGroups</code></td>
<td>
<p>A dataset containing the names of the environments (in the first column) and the groups to which the environments belong (in the second column). (optional)</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_qtlsvarname">QTLsVarName</code></td>
<td>
<p>The name of the column indicating to which QTL the marker belongs.  (optional)</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_roworder">RowOrder</code></td>
<td>
<p>A boolean specifying whether to reorder the markers or not. (TRUE by default)</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_colorder">ColOrder</code></td>
<td>
<p>A boolean specifying whether to reorder the environments or not. (TRUE by default)</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_showdendrogram">ShowDendrogram</code></td>
<td>
<p>A boolean specifying wether to show the clustering of the rows and/or the columns. (FALSE by default)</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_colors">Colors</code></td>
<td>
<p>A vector of three colors corresponding to the color scale of the Heatmap.</p>
</td></tr>
<tr><td><code id="metaGE.heatmap_+3A_main">Main</code></td>
<td>
<p>The main to display.(optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The heatmap
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(dplyr)
# Import the data
data("metaData")

# Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

# Fit the Fixed Effect model
FeDF &lt;- metaGE.fit(metaData, matCorr, Method = "Fe")

# Control the FDR (here Benjamini-Hochberg)
Alpha &lt;- 0.05
Signif &lt;-  FeDF$PVALUE %&gt;% p.adjust(method = "BH") %&gt;% `&lt;`(Alpha) %&gt;% which

# Draw the z-scores heatmap of the significant markers
heatmap &lt;- metaGE.heatmap(Data = FeDF[Signif,],
                         Prefix = "Z.")
</code></pre>

<hr>
<h2 id='metaGE.incidence'>Create the matrix of incidence.</h2><span id='topic+metaGE.incidence'></span>

<h3>Description</h3>

<p>The function metaGE.incidence convert categorical variable describing the environments into a matrix of dummy variables with in rows the levels of the variables and in columns the environment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.incidence(VarName, Covariate, EnvName, Data, AtLeast = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.incidence_+3A_varname">VarName</code></td>
<td>
<p>The name of the column containing the categorical variable in the Covariate dataset.</p>
</td></tr>
<tr><td><code id="metaGE.incidence_+3A_covariate">Covariate</code></td>
<td>
<p>A dataset containing categorical variables (in columns) describing the environments (in rows).</p>
</td></tr>
<tr><td><code id="metaGE.incidence_+3A_envname">EnvName</code></td>
<td>
<p>The name of the column containing the names of the environment in the Covariate dataset.</p>
</td></tr>
<tr><td><code id="metaGE.incidence_+3A_data">Data</code></td>
<td>
<p>A dataset containing the effects and pvalues of each marker (in rows) in each environment (in columns), as obtained from metaGE.collect.</p>
</td></tr>
<tr><td><code id="metaGE.incidence_+3A_atleast">AtLeast</code></td>
<td>
<p>A numeric value indicating the minimum number of environments must belong to each level (equals 1 by default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The names of the environment must be the same as used in the Data dataset.
</p>


<h3>Value</h3>

<p>A  binary matrix containing indicator variables with in rows the levels of the variables and in columns the environment.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import the data
data("metaData")
data("envDesc")

# Build the matrix of incidence
(Incidence.Temp &lt;- metaGE.incidence(VarName = "Temp",Covariate = envDesc,
                                    EnvName = "ShortName", Data = metaData))
</code></pre>

<hr>
<h2 id='metaGE.lscore'>Compute the local score from a set of pvalues.</h2><span id='topic+metaGE.lscore'></span>

<h3>Description</h3>

<p>The function metaGE.lscore computes the local score and the significant regions from
a set of pvalues.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.lscore(Data, PvalName, xi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.lscore_+3A_data">Data</code></td>
<td>
<p>A dataset containing the following columns: CHR, POS, MARKER and PvalName.</p>
</td></tr>
<tr><td><code id="metaGE.lscore_+3A_pvalname">PvalName</code></td>
<td>
<p>The name of the column containing the p-value.</p>
</td></tr>
<tr><td><code id="metaGE.lscore_+3A_xi">xi</code></td>
<td>
<p>The threshold of the score, xi = 1,2,3 or 4.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is directly inherited from the scorelocalfunctions.R R code file of Fariello MI, Boitard S, Mercier S, et al.,
as available on the https://forge-dga.jouy.inra.fr/projects/local-score website.
The technical details of the computation can be found in Fariello MI, Boitard S, Mercier S, et al.
Accounting for linkage disequilibrium in genome scans for selection without individual genotypes: The local score approach.
https://doi.org/10.1111/mec.14141.
The function computes a local score for the detection of significant regions based on the hypothesis that the H0 distribution of the pvalues is
uniform. Under this hypothesis the local score follows a Gumbel distribution (under H0) whose parameters depend
on the threshold <code>xi</code> and on the autocorrelation between pvalues within each chromosome. The threshold has
to be selected in 1,2,3,4 and the autocorrelation is computed internally.
</p>


<h3>Value</h3>

<p>A list of:
</p>

<ul>
<li><p> Data the dataset Data with the local score as supplementary column.
</p>
</li>
<li><p> SigZones a dataset containing informations about the significative regions.
</p>
</li>
<li><p> SigMarker a dataset containing the signicatives markers.
</p>
</li>
<li><p> ChrThreshold a dataset containing the chromosome-wide significance thresholds.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>require(dplyr)
## Not run: 
# Import the data
data("metaData")

# Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

# Fit the Fixed Effect model
FeDF &lt;- metaGE.fit(metaData, matCorr, Method = "Fe")

# Compute the score local
xi &lt;- 2
FeScore &lt;- metaGE.lscore(FeDF,"PVALUE", xi)
#FeScore$SigZones
## End(Not run)
</code></pre>

<hr>
<h2 id='metaGE.manhattan'>Draw the Manhattan plot.</h2><span id='topic+metaGE.manhattan'></span>

<h3>Description</h3>

<p>The function metaGE.manhattan displays the Manhattan plot of the -log10(p-value) or the local score of each marker along the genome.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.manhattan(
  Data,
  VarName,
  Threshold = NULL,
  SigZones = NULL,
  Score = FALSE,
  AnnotateMarker = NULL,
  Main = "",
  col = c("grey", "black"),
  colSigZones = "blue",
  Ylim = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.manhattan_+3A_data">Data</code></td>
<td>
<p>A dataset containing the columns: CHR, POS, MARKER and the variable to plot for each marker, as obtained from metaGE.fit.</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_varname">VarName</code></td>
<td>
<p>The name of the column containing the variable to plot, generally the p-value or a score.</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_threshold">Threshold</code></td>
<td>
<p>A threshold in order to draw a &quot;genome-wide significant&quot; line.  (optional)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_sigzones">SigZones</code></td>
<td>
<p>A dataset containing the significant zones to plot, as obtained from metaGE.lscore. Must have columns: CHR, POS, START, END. (optional)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_score">Score</code></td>
<td>
<p>A boolean. If FALSE, the -log10 of the variable is plotted, useful for plotting p-values. If TRUE, the raw values of the variable is plotted, useful for plotting scores. (FALSE by default)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_annotatemarker">AnnotateMarker</code></td>
<td>
<p>A list of markers name to annotate in the plot. (optional)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_main">Main</code></td>
<td>
<p>The main to display. (optional)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_col">col</code></td>
<td>
<p>A character vector indicating which colors to alternate for different chromosomes.(c('grey', 'black') by default)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_colsigzones">colSigZones</code></td>
<td>
<p>A character indicating which color to plot the significant zones.('blue' by default)</p>
</td></tr>
<tr><td><code id="metaGE.manhattan_+3A_ylim">Ylim</code></td>
<td>
<p>Two numeric values, specifying the lower limit and the upper limit of the  y-axe scale. (optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The Manhattan plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(dplyr)
# Import the data
data("metaData")

# Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

# Fit the Fixed Effect model
FeDF &lt;- metaGE.fit(metaData, matCorr, Method = "Fe")

# Control the FDR (here Benjamini-Hochberg)
Alpha &lt;- 0.05
Signif &lt;-  FeDF$PVALUE %&gt;% p.adjust(method = "BH") %&gt;% `&lt;`(Alpha) %&gt;% which

# Draw the corresponding manhattan plot
#PvalThresholdFe &lt;- FeDF[Signif,]$PVALUE%&gt;% max %&gt;% max(.,0)
#manhattan_pval &lt;- metaGE.manhattan(Data = FeDF,VarName = 'PVALUE',
#                             Threshold = PvalThresholdFe,
#                              Main = '-log10(Pval) alongside the chromosome Fe method')


# Compute the score local
## Not run: xi &lt;- 2
FeScore &lt;- metaGE.lscore(FeDF,"PVALUE", xi)

# Draw the corresponding manhattan plot
manhattan_lscore &lt;- metaGE.manhattan(Data = FeScore$Data,VarName = 'SCORE',
                                     SigZones = FeScore$SigZones, Score = TRUE,
                                     Main = 'Local score alongside the chromosome Fe method')

## End(Not run)



</code></pre>

<hr>
<h2 id='metaGE.pvalplot'>Display visual checks of pvalues.</h2><span id='topic+metaGE.pvalplot'></span>

<h3>Description</h3>

<p>The function metaGE.pvalplot displays the pvalue distribution and the QQplot of the -log10(pvalues).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.pvalplot(Pvalues, Main = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.pvalplot_+3A_pvalues">Pvalues</code></td>
<td>
<p>A vector containing pvalues.</p>
</td></tr>
<tr><td><code id="metaGE.pvalplot_+3A_main">Main</code></td>
<td>
<p>The main to display.(optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value, the plot is displayed in the active graphics window.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Import the data
data("metaData")

# Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

# Fit the Fixed Effect model
FeDF &lt;- metaGE.fit(metaData, matCorr, Method = "Fe")

# Check the pvalues
metaGE.pvalplot(Pvalues = FeDF$PVALUE, Main= "Pvalue Fe")
</code></pre>

<hr>
<h2 id='metaGE.regplot'>Plot the z-score of a marker according to a covariate.</h2><span id='topic+metaGE.regplot'></span>

<h3>Description</h3>

<p>The function metaGE.regplot displays the graph of the z-scores of a marker according to a covariate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.regplot(
  Data,
  Covariate,
  EnvName,
  MarkerName,
  VarName,
  Zscore = FALSE,
  aesCol = NULL,
  Main = ""
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.regplot_+3A_data">Data</code></td>
<td>
<p>A dataset containing the columns: MARKER and the z-scores of each marker (in rows) in each environment (in columns), as obtained from metaGE.collect.</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_covariate">Covariate</code></td>
<td>
<p>A dataset containing the values of one or more covariates (in columns) in each environment (in rows).</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_envname">EnvName</code></td>
<td>
<p>The name of the column containing the names of the environment in the Covariate dataset.</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_markername">MarkerName</code></td>
<td>
<p>The name of the marker.</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_varname">VarName</code></td>
<td>
<p>The name of the column containing the covariable to plot.</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_zscore">Zscore</code></td>
<td>
<p>A boolean. If FALSE, the estimated marker effects is plotted. If TRUE, the z-scores of the marker is plotted. (FALSE by default)</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_aescol">aesCol</code></td>
<td>
<p>The name of the column containing a qualitative covariable to specify the color of the points. (optional)</p>
</td></tr>
<tr><td><code id="metaGE.regplot_+3A_main">Main</code></td>
<td>
<p>The main to display.(optional)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("metaData")
data("envDesc")
metaGE.regplot(Data = metaData, Covariate = envDesc, EnvName = "ShortName",
               MarkerName = "AX-91369217", VarName = "Tnight.mean", aesCol = "Classification")

</code></pre>

<hr>
<h2 id='metaGE.test'>Meta-analysis test for Genotype x Environment interactions: Contrast or Regression.</h2><span id='topic+metaGE.test'></span>

<h3>Description</h3>

<p>The function metaGE.test compute meta-analysis contrast or regression test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaGE.test(
  Data,
  MatCorr,
  Incidence = NULL,
  Contrast = NULL,
  Covariate = NULL,
  EnvName = NULL,
  NA.omit = TRUE,
  DropZScores = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaGE.test_+3A_data">Data</code></td>
<td>
<p>A dataset containing the estimated marker effect and its associated pvalue of each marker (in rows) in each environment (in columns), as obtained from metaGE.collect.</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_matcorr">MatCorr</code></td>
<td>
<p>The inter-environment correlation matrix. It can be compute by the metaGE.cor function.</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_incidence">Incidence</code></td>
<td>
<p>A matrix of incidence, as obtained from metaGE.incidence or a list of such matrix.</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_contrast">Contrast</code></td>
<td>
<p>A matrix of contrast, or a list of such matrix.</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_covariate">Covariate</code></td>
<td>
<p>A dataset containing the values of one or more covariates (in columns) in each environment (in rows).</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_envname">EnvName</code></td>
<td>
<p>The name of the column containing the names of the environment in the covariate dataset.</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_na.omit">NA.omit</code></td>
<td>
<p>A boolean specifying whether the markers with some NA values should be removed from the test procedure. (TRUE by default)</p>
</td></tr>
<tr><td><code id="metaGE.test_+3A_dropzscores">DropZScores</code></td>
<td>
<p>A boolean specifying whether the Zscores should be dropped from the dataset or not. (FALSE by default)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If Incidence is provided, the function will perform all the corresponding tests of contrast. If Covariate is provided, the function will perform all the corresponding meta-regression tests.
The Contrast can be NULL, in this case the identity matrix is used.
</p>


<h3>Value</h3>

<p>The dataset Data with supplementary columns containing the Pvalue of each test performed.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(dplyr)

 # Import the data
data("metaData")
data("envDesc")

#' # Compute the inter-environment correlation matrix
matCorr &lt;- metaGE.cor(metaData, Threshold = 0.8)

#### Contrast test
# Build the matrix of incidence
Incidence.Water &lt;- metaGE.incidence(VarName = "Water",Covariate = envDesc,
                                    EnvName = "ShortName", Data = metaData)

# Perform the contrast test
ContrastDF &lt;- metaGE.test(metaData, matCorr,Incidence = Incidence.Water, Contrast = NULL)
head(ContrastDF %&gt;% select(CHR, POS, MARKER, PVALUE.Contrast1))

#### Regression test
RegressionDF &lt;- metaGE.test(metaData,matCorr, Covariate = envDesc[,c(1,5)],EnvName = "ShortName" )
head(RegressionDF %&gt;% select(CHR, POS, MARKER, PVALUE.Tnight.mean))
</code></pre>

<hr>
<h2 id='ReadData'>ReadData</h2><span id='topic+ReadData'></span>

<h3>Description</h3>

<p>This function read the one file, select interesting columns and rename them.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReadData(ListN, FileN, VarN, MinFreq = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ReadData_+3A_listn">ListN</code></td>
<td>
<p>The name of the list of files where the file to read belongs or NULL if there is only one list of files</p>
</td></tr>
<tr><td><code id="ReadData_+3A_filen">FileN</code></td>
<td>
<p>The name of the file to read</p>
</td></tr>
<tr><td><code id="ReadData_+3A_varn">VarN</code></td>
<td>
<p>A named list containing the column names in the file corresponding to the variables below: MARKER, CHR, POS, EFFECT, EFFECT_SE, PVAL,(optional: FREQ, ALLELE0, ALLELE1, wEIGHT)</p>
</td></tr>
<tr><td><code id="ReadData_+3A_minfreq">MinFreq</code></td>
<td>
<p>A numeric value allowing to filter to keep markers with MAF &gt; MinFreq</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with the interesting columns selected and renamed.
</p>

<hr>
<h2 id='RegressionStatTest'>Compute the pvalue of the meta-regression test.</h2><span id='topic+RegressionStatTest'></span>

<h3>Description</h3>

<p>The function RegressionStatTest compute the statistic and the pvalue of the regression test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RegressionStatTest(Covariate, CovName, Zmat, MatCorr)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RegressionStatTest_+3A_covariate">Covariate</code></td>
<td>
<p>A dataset containing the values of one Covariate (in columns) in each environment (in rows).</p>
</td></tr>
<tr><td><code id="RegressionStatTest_+3A_covname">CovName</code></td>
<td>
<p>The name the Covariate.</p>
</td></tr>
<tr><td><code id="RegressionStatTest_+3A_zmat">Zmat</code></td>
<td>
<p>A matrix containing the Zscores of all markers (in rows) in each environment (in columns).</p>
</td></tr>
<tr><td><code id="RegressionStatTest_+3A_matcorr">MatCorr</code></td>
<td>
<p>The inter-environments correlation matrix. Can be computed using MetaGE.cor.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset of two columns containing the pvalue of the meta-regression test and the number of environment used to perform the test of all markers.
</p>

<hr>
<h2 id='RegressionStatTestNA'>Compute the pvalue of the regression test in presence of missing values.</h2><span id='topic+RegressionStatTestNA'></span>

<h3>Description</h3>

<p>The function RegressionStatTest compute the statistic and the pvalue of the regression test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RegressionStatTestNA(Covariate, CovName, Zmat, MatCorr, Data, Configs.list)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RegressionStatTestNA_+3A_covariate">Covariate</code></td>
<td>
<p>A dataset containing the values of one covariate (in columns) in each environment (in rows).</p>
</td></tr>
<tr><td><code id="RegressionStatTestNA_+3A_covname">CovName</code></td>
<td>
<p>The name the covariate.</p>
</td></tr>
<tr><td><code id="RegressionStatTestNA_+3A_zmat">Zmat</code></td>
<td>
<p>A matrix containing the Zscores of all markers (in rows) in each environment (in columns).</p>
</td></tr>
<tr><td><code id="RegressionStatTestNA_+3A_matcorr">MatCorr</code></td>
<td>
<p>The inter-environments correlation matrix. Can be computed using MetaGE.cor.</p>
</td></tr>
<tr><td><code id="RegressionStatTestNA_+3A_data">Data</code></td>
<td>
<p>A dataset containing the effect, the pvalues and the na configuration for all marker</p>
</td></tr>
<tr><td><code id="RegressionStatTestNA_+3A_configs.list">Configs.list</code></td>
<td>
<p>A vector containing the NA configurations present in the dataset</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataset of two columns containing the pvalue of the meta-regression test and the number of environment used to perform the test of all markers.
</p>

<hr>
<h2 id='sig_sl'>Computation of the significative regions</h2><span id='topic+sig_sl'></span>

<h3>Description</h3>

<p>The function sig_sl computes the significative regions from a lindley process given a significance threshold.(function from localscore)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sig_sl(lind, pos, th)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sig_sl_+3A_lind">lind</code></td>
<td>
<p>The lindley</p>
</td></tr>
<tr><td><code id="sig_sl_+3A_pos">pos</code></td>
<td>
<p>The position</p>
</td></tr>
<tr><td><code id="sig_sl_+3A_th">th</code></td>
<td>
<p>The threshold</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the significance threshold.
</p>

<hr>
<h2 id='thresUnif'>Computation of the significance threshold</h2><span id='topic+thresUnif'></span>

<h3>Description</h3>

<p>The function thresUnif computes the significance threshold.(function from localscore)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>thresUnif(L, cor, xi, alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="thresUnif_+3A_l">L</code></td>
<td>
<p>The length of the chromosome</p>
</td></tr>
<tr><td><code id="thresUnif_+3A_cor">cor</code></td>
<td>
<p>The autocorrelation of the chromosome</p>
</td></tr>
<tr><td><code id="thresUnif_+3A_xi">xi</code></td>
<td>
<p>The threshold of the score, xi = 1,2,3 or 4.</p>
</td></tr>
<tr><td><code id="thresUnif_+3A_alpha">alpha</code></td>
<td>
<p>The nominal threshold.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The distribution of the p-values is uniform, the local score follows a Gumbel distribution under the null.
</p>


<h3>Value</h3>

<p>the significance threshold.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
