<!DOCTYPE html><html><head><title>Help for package metamicrobiomeR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metamicrobiomeR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#alpha.compare'><p>Compare multiple alpha diversity indexes between groups</p></a></li>
<li><a href='#alphadat'><p>Alpha diversity data.</p></a></li>
<li><a href='#asum4'><p>Combined alpha diversity data for meta-analysis.</p></a></li>
<li><a href='#covar.rm'><p>Covariate data.</p></a></li>
<li><a href='#gtab.3stud'><p>Test datasets for microbiome age prediction.</p></a></li>
<li><a href='#kegg.12'><p>Pathway abundance data.</p></a></li>
<li><a href='#meta.niceplot'><p>Nice meta-analysis plots.</p></a></li>
<li><a href='#meta.taxa'><p>Meta-analysis of taxa/pathway abundance comparison.</p></a></li>
<li><a href='#metatab.show'><p>Display meta-analysis results.</p></a></li>
<li><a href='#microbiomeage'><p>Predict microbiome age.</p></a></li>
<li><a href='#pathway.compare'><p>Compare (kegg) pathway abundance</p></a></li>
<li><a href='#read.multi'><p>Read multiple files</p></a></li>
<li><a href='#tabsex4'><p>Combined data for meta-analysis.</p></a></li>
<li><a href='#taxa.compare'><p>Compare taxa relative abundance</p></a></li>
<li><a href='#taxa.filter'><p>Filter relative abundance data</p></a></li>
<li><a href='#taxa.mean.plot'><p>Plot mean taxa abundance</p></a></li>
<li><a href='#taxa.meansdn'><p>Summarize abundance by group</p></a></li>
<li><a href='#taxcomtab.show'><p>Display abundance comparison results.</p></a></li>
<li><a href='#taxtab6'><p>Taxonomic relative abundance data.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Microbiome Data Analysis &amp; Meta-Analysis with GAMLSS-BEZI &amp;
Random Effects</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Generalized Additive Model for Location, Scale and Shape (GAMLSS) 
    with zero inflated beta (BEZI) family for analysis of microbiome relative abundance data 
    (with various options for data transformation/normalization to address compositional effects) and 
    random effects meta-analysis models for meta-analysis pooling estimates across microbiome studies 
    are implemented. 
    Random Forest model to predict microbiome age based on relative abundances of  
    shared bacterial genera with the Bangladesh data (Subramanian et al 2014), 
    comparison of multiple diversity indexes using linear/linear mixed effect models 
    and some data display/visualization are also implemented.
    The reference paper is published by 
    Ho NT, Li F, Wang S, Kuhn L (2019) &lt;<a href="https://doi.org/10.1186%2Fs12859-019-2744-2">doi:10.1186/s12859-019-2744-2</a>&gt; . </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), gamlss</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>meta, lme4, gdata, plyr, dplyr, tidyr, ggplot2, gridExtra,
lmerTest, matrixStats, zCompositions, compositions</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, RCurl, httr, repmis, jsonlite, knitr, rmarkdown,
grDevices, gplots, magrittr, tools, foreign, mgcv, reshape2,
caret, randomForest, tsibble, RColorBrewer</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/nhanhocu/metamicrobiomeR">https://github.com/nhanhocu/metamicrobiomeR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/nhanhocu/metamicrobiomeR/issues">https://github.com/nhanhocu/metamicrobiomeR/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-31 01:41:11 UTC; nhanht6</td>
</tr>
<tr>
<td>Author:</td>
<td>Nhan Ho [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Nhan Ho &lt;nhanhocumc@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-09 11:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='alpha.compare'>Compare multiple alpha diversity indexes between groups</h2><span id='topic+alpha.compare'></span>

<h3>Description</h3>

<p>This function calculates average of alpha diversity indexes for a specific rarefaction depth, standardize diversity indexes and compare between groups
using linear/linear mixed effect model and adjust for covariates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>alpha.compare(
  datlist,
  depth,
  mapfile,
  mapsampleid,
  comvar,
  adjustvar,
  personid = "personid",
  longitudinal = "yes",
  age.limit = 1e+06,
  standardize = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="alpha.compare_+3A_datlist">datlist</code></td>
<td>
<p>the list of your dataframe.</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_depth">depth</code></td>
<td>
<p>the rarefaction depth of choice. Depth can be &quot;max&quot; (highest depth) or an order (number) of the depth in the list generated by alpha_rarefaction.py</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_mapfile">mapfile</code></td>
<td>
<p>mapping file</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_mapsampleid">mapsampleid</code></td>
<td>
<p>sample id in the mapping file</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_comvar">comvar</code></td>
<td>
<p>variable for comparison</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_adjustvar">adjustvar</code></td>
<td>
<p>variables that need to be adjusted in the model</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_personid">personid</code></td>
<td>
<p>name of variable for person id. Default is &quot;personid&quot; (applicable when longitudinal=&quot;yes&quot;).</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_longitudinal">longitudinal</code></td>
<td>
<p>longitudinal data or one time data. Options are c(&quot;yes&quot;,&quot;no&quot;). Default is &quot;yes&quot;.</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_age.limit">age.limit</code></td>
<td>
<p>age upper limit for included samples. Default is 1000000 months (~no upper limit).</p>
</td></tr>
<tr><td><code id="alpha.compare_+3A_standardize">standardize</code></td>
<td>
<p>standardization of diversity indexes before comparison or not. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of comparison result matrices and mean diversity of all indexes for each samples (with or without standardization)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alphadat)
data(covar.rm)
covar.rm$sampleid&lt;-tolower(covar.rm$sampleid)
#comparison of standardized alpha diversity indexes between genders adjusting for
#breastfeeding and infant age at sample collection in infants &lt;=6 months of age
alphacom&lt;-alpha.compare(datlist=alphadat,depth=3,mapfile=covar.rm,
mapsampleid="sampleid", comvar="gender",adjustvar=c("age.sample","bf"),
longitudinal="yes", age.limit=6,standardize=TRUE)
alphacom$alphasum[,1:5]
</code></pre>

<hr>
<h2 id='alphadat'>Alpha diversity data.</h2><span id='topic+alphadat'></span>

<h3>Description</h3>

<p>Alpha diversity data from alpha_rarefaction output from QIIME.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(alphadat)
</code></pre>


<h3>Format</h3>

<p>A list of 4 dataframes for four indexes: Chao1, Observed_species, PD_whole_tree, Shannon.
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(alphadat)
# Load covariate data
data(covar.rm)
covar.rm$sampleid&lt;-tolower(covar.rm$sampleid)
#comparison of standardized alpha diversity indexes between genders adjusting for
#breastfeeding and infant age at sample collection in infants &lt;=6 months of age
alphacom&lt;-alpha.compare(datlist=alphadat,depth=3,mapfile=covar.rm,
mapsampleid="sampleid", comvar="gender",adjustvar=c("age.sample","bf"),
longitudinal="yes", age.limit=6,standardize=TRUE)
alphacom$alphasum[,1:5]
</code></pre>

<hr>
<h2 id='asum4'>Combined alpha diversity data for meta-analysis.</h2><span id='topic+asum4'></span>

<h3>Description</h3>

<p>Result outputs of differential analysis of alpha diversity
using linear or linear mixed effects model
from &quot;alpha.compare&quot; function combined from
4 studies for meta-analysis. The comparison was between gender
adjusted for age of infants at sample collection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(asum4)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 16 rows and 18 variables.
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>
<p>Bender et al. Sci Transl Med. 2016 Jul 27; 8(349): 349ra100.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5301310/">PubMed</a>)
</p>
<p>Pannaraj et al. JAMA Pediatr. 2017;90095(7):647–54.
(<a href="https://pubmed.ncbi.nlm.nih.gov/28492938/">PubMed</a>)
</p>
<p>Thompson et al. Front Cell Infect Microbiol. 2015;5:3.
(<a href="https://pubmed.ncbi.nlm.nih.gov/25705611/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(asum4)
</code></pre>

<hr>
<h2 id='covar.rm'>Covariate data.</h2><span id='topic+covar.rm'></span>

<h3>Description</h3>

<p>Monthly longitudinal clinical data of 50 infants from
birth to 2 years of life.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(covar.rm)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 996 rows and 32 variables.
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(covar.rm)
# Load KEGG pathway data
data(kegg.12)
# Comparison of pathway relative abundances for some first pathways of level 1 only
# and assuming crosssectional data (to save running time)
path1&lt;-pathway.compare(pathtab=list(kegg.12[[1]][, 1:2]),
mapfile=covar.rm,sampleid="sampleid",pathsum="rel", stat.med="gamlss",
comvar="gender",adjustvar=c("age.sample","bf"), longitudinal="no",
p.adjust.method="fdr", percent.filter=0.05,relabund.filter=0.00005)
taxcomtab.show(taxcomtab=path1$l1, sumvar="path",tax.lev="l2",
tax.select="none", showvar="genderMale", p.adjust.method="fdr",p.cutoff=1)
</code></pre>

<hr>
<h2 id='gtab.3stud'>Test datasets for microbiome age prediction.</h2><span id='topic+gtab.3stud'></span>

<h3>Description</h3>

<p>A list of three datasets to be used as test datasets
for microbiome age prediction using Random Forest model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(gtab.3stud)
</code></pre>


<h3>Format</h3>

<p>A list of three dataframes.
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>
<p>Bender et al. Sci Transl Med. 2016 Jul 27; 8(349): 349ra100.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5301310/">PubMed</a>)
</p>
<p>Pannaraj et al. JAMA Pediatr. 2017;90095(7):647–54.
(<a href="https://pubmed.ncbi.nlm.nih.gov/28492938/">PubMed</a>)
</p>
<p>Thompson et al. Front Cell Infect Microbiol. 2015;5:3.
(<a href="https://pubmed.ncbi.nlm.nih.gov/25705611/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(gtab.3stud)
</code></pre>

<hr>
<h2 id='kegg.12'>Pathway abundance data.</h2><span id='topic+kegg.12'></span>

<h3>Description</h3>

<p>KEGG pathway abundance data from PICRUSt analysis.
This is monthly longitudinal data of 50 infants from birth to
2 years of life.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(kegg.12)
</code></pre>


<h3>Format</h3>

<p>A list of 2 dataframes for level 1 and level 2 of KEGG pathways.
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(kegg.12)
# Load covariate data
data(covar.rm)
# Comparison of pathway relative abundances for some first pathways of level 1 only
# and assuming crosssectional data (to save running time)
path1&lt;-pathway.compare(pathtab=list(kegg.12[[1]][, 1:2]),
mapfile=covar.rm,sampleid="sampleid",pathsum="rel", stat.med="gamlss",
comvar="gender",adjustvar=c("age.sample","bf"), longitudinal="no",
p.adjust.method="fdr", percent.filter=0.05,relabund.filter=0.00005)
taxcomtab.show(taxcomtab=path1$l1, sumvar="path",tax.lev="l2",
tax.select="none", showvar="genderMale", p.adjust.method="fdr",p.cutoff=1)
</code></pre>

<hr>
<h2 id='meta.niceplot'>Nice meta-analysis plots.</h2><span id='topic+meta.niceplot'></span>

<h3>Description</h3>

<p>This function displays meta-analysis results of relative abundance as a nice combined heatmap and forest plot. More flexibility/options for plot will be added.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta.niceplot(
  metadat,
  sumtype = "taxa",
  level = "main",
  p,
  p.adjust,
  phyla.col = "rainbow",
  phyla.select = c("actinobacteria", "bacteroidetes", "cyanobacteria", "firmicutes",
    "fusobacteria", "proteobacteria", "verrucomicrobia", ".thermi."),
  col.select = c("#dd1c77", "#31a354", "#91003f", "#d95f0e", "#636363", "#2ef0e7",
    "#862ef0", "#000"),
  est.break = c(-Inf, -1, -0.5, -0.1, 0, 0.1, 0.5, 1, Inf),
  est.break.label = c("&lt;-1", "[-1,-0.5)", "[-0.5,-0.1)", "[-0.1,0)", "[0,0.1)",
    "[0.1,0.5)", "[0.5,1)", "&gt;=1"),
  neg.palette = "PuBu",
  pos.palette = "YlOrRd",
  p.sig.heat = "no",
  p.break = c(0, 1e-04, 0.05, 1),
  p.break.label = c("**", "*", ""),
  p.pool.break = c(0, 0.05, 1),
  p.pool.break.label = c("[0,0.05)", "[0.05,1]"),
  padjust.pool.break = c(0, 0.1, 1),
  padjust.pool.break.label = c("[0,0.1)", "[0.1,1]"),
  forest.est.shape = c("17", "16"),
  forest.est.col = c("red", "black"),
  forest.col = "by.pvalue",
  leg.key.size = 1,
  leg.text.size = 8,
  heat.text.x.size = 8,
  heat.text.x.angle = 0,
  forest.axis.text.y = 8,
  forest.axis.text.x = 8,
  heat.forest.width.ratio = c(1, 1),
  point.ratio = c(3, 1),
  line.ratio = c(2, 1)
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta.niceplot_+3A_metadat">metadat</code></td>
<td>
<p>output data from metatab.show.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_sumtype">sumtype</code></td>
<td>
<p>Either &quot;taxa&quot; for taxa and &quot;path&quot; for pathway.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_level">level</code></td>
<td>
<p>&quot;main&quot; for main level such as phylum or &quot;sub&quot; for higher level such as species. Default is &quot;main&quot;.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p">p</code></td>
<td>
<p>name of variable for p-values</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p.adjust">p.adjust</code></td>
<td>
<p>name of variable for multiple testing adjusted p-values</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_phyla.col">phyla.col</code></td>
<td>
<p>type of color for main level (phylum). Options are &quot;rainbow&quot; (default) or &quot;select&quot;.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_phyla.select">phyla.select</code></td>
<td>
<p>selected phyla for selected colors (only when phyla.col=&quot;select&quot;). Default are c(&quot;actinobacteria&quot;,&quot;bacteroidetes&quot;,&quot;cyanobacteria&quot;,&quot;firmicutes&quot;,&quot;fusobacteria&quot;,&quot;proteobacteria&quot;,&quot;verrucomicrobia&quot;,&quot;.thermi.&quot;).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_col.select">col.select</code></td>
<td>
<p>selected colors for selected phyla (only when phyla.col=&quot;select&quot;). Corresponding default are c(&quot;#dd1c77&quot;,&quot;#31a354&quot;,&quot;#91003f&quot;,&quot;#d95f0e&quot;,&quot;#636363&quot;,&quot;#2ef0e7&quot;,&quot;#862ef0&quot;,&quot;#000&quot;).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_est.break">est.break</code></td>
<td>
<p>breaks for estimates to generate color categories on heatmap. Default are c(-Inf, -1,-0.5,-0.1,0,0.1,0.5,1, Inf). For pathway, recommended breaks are c(-Inf, -0.5,-0.1,-0.05,0,0.05,0.1,0.5, Inf).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_est.break.label">est.break.label</code></td>
<td>
<p>labels for corresponding color categories on heatmap generated by est.break. Default corresponding to default est.break are c(&quot;&lt;-1&quot;,&quot;[-1,-0.5)&quot;,&quot;[-0.5,-0.1)&quot;,&quot;[-0.1,0)&quot;, &quot;[0,0.1)&quot;, &quot;[01,0.5)&quot;, &quot;[0.5,1)&quot;, &quot;&gt;=1&quot;). For pathway, corresponding recommended labels are c(&quot;&lt;-0.5)&quot;, &quot;[-0.5,-0.1)&quot;,&quot;[-0.1,-0.05)&quot;,&quot;[-0.05,0)&quot;,&quot;[0,0.05)&quot;,&quot;[0.05,0.1)&quot;, &quot;[0.1,0.5)&quot;, &quot;&gt;=0.5&quot;).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_neg.palette">neg.palette</code></td>
<td>
<p>color palette for negative estimate values. Default is &quot;PuBu&quot;. Use display.brewer.all() of RColorBrewer package for other options.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_pos.palette">pos.palette</code></td>
<td>
<p>color palette for positive estimate values. Default is &quot;YlOrRd&quot;. Use display.brewer.all() of RColorBrewer package for other options.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p.sig.heat">p.sig.heat</code></td>
<td>
<p>whether or not show significant p values on heatmap. Default is &quot;yes&quot;.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p.break">p.break</code></td>
<td>
<p>breaks for significant levels of p values. Default is c(0, 0.0001,0.05, 1).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p.break.label">p.break.label</code></td>
<td>
<p>labels to be showed on heatmap for different levels of p-values from p.break. Default is c(&quot;**&quot;, &quot;*&quot;,&quot;&quot;) for p breaks at c(0, 0.0001,0.05, 1).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p.pool.break">p.pool.break</code></td>
<td>
<p>breaks for pooled p-values to be distinguished in forest plot. Default are c(0,0.05,1).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_p.pool.break.label">p.pool.break.label</code></td>
<td>
<p>labels for pooled p-value breaks. Corresponding default are c(&quot;[0,0.05)&quot;,&quot;[0.05,1]&quot;).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_padjust.pool.break">padjust.pool.break</code></td>
<td>
<p>breaks for multiple testing adjusted p-values to be distinguished in forest plot. Default are c(0,0.1,1).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_padjust.pool.break.label">padjust.pool.break.label</code></td>
<td>
<p>labels for  multiple testing adjusted p-value breaks. Corresponding default are c(&quot;[0,0.1)&quot;,&quot;[0.1,1]&quot;).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_forest.est.shape">forest.est.shape</code></td>
<td>
<p>point shape of pooled estimates in forest plot. Default are c(&quot;17&quot;,&quot;16&quot;) for corresponding significant and non-significant pooled estimates.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_forest.est.col">forest.est.col</code></td>
<td>
<p>colors of point (pooled estimates) and 95 CI bars in forest plot. Default are c(&quot;red&quot;, &quot;black&quot;) for significant and non-significant estimates.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_forest.col">forest.col</code></td>
<td>
<p>Color of forest plot (point estimates and 95 CI). Options are &quot;by.pvalue&quot; (distinguished by signficant vs. non-significant p-value) or &quot;by.estimate&quot; (color scaled similarly to heatmap color).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_leg.key.size">leg.key.size</code></td>
<td>
<p>legdend key size for heatmap.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_leg.text.size">leg.text.size</code></td>
<td>
<p>legend text size for heatmap.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_heat.text.x.size">heat.text.x.size</code></td>
<td>
<p>heatmap x label text size.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_heat.text.x.angle">heat.text.x.angle</code></td>
<td>
<p>heatmap x label text angle.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_forest.axis.text.y">forest.axis.text.y</code></td>
<td>
<p>forest plot y label text size.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_forest.axis.text.x">forest.axis.text.x</code></td>
<td>
<p>forest plot x label text size.</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_heat.forest.width.ratio">heat.forest.width.ratio</code></td>
<td>
<p>ratio of width between heatmap and forest plot to be used in grid.arrange. Dedault is c(1,1).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_point.ratio">point.ratio</code></td>
<td>
<p>ratio of point size between significant pooled estimate and non-significant pooled estimate. Default is c(3,1).</p>
</td></tr>
<tr><td><code id="meta.niceplot_+3A_line.ratio">line.ratio</code></td>
<td>
<p>ratio of error bar line size between significant pooled estimate and non-significant pooled estimate. Default is=c(2,1).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>combined heatmap forest plot.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load saved GAMLSS-BEZI results of four studies
# for the comparison of bacterial taxa relative abundance
# between genders adjusted for breastfeeding and
# infant age at sample collection
data(tabsex4)
#select only taxonomies of a small phylum for meta-analysis example
# (to save running time)
tlm&lt;-tabsex4$id[grep("k__bacteria.p__fusobacteria",tabsex4$id)]
# meta-analysis
metab.sex&lt;-meta.taxa(taxcomdat=tabsex4[tabsex4$id %in% tlm,],
summary.measure="RR", pool.var="id", studylab="study", backtransform=FALSE,
percent.meta=0.5, p.adjust.method="fdr")
#show results by table and plot
#phylum
#table
metatab.show(metatab=metab.sex$random,com.pooled.tab=tabsex4[tabsex4$id %in% tlm,],
tax.lev="l2",showvar="genderMale",p.cutoff.type="p", p.cutoff=1,display="table")
#plot
metadat&lt;-metatab.show(metatab=metab.sex$random, com.pooled.tab=tabsex4[tabsex4$id %in% tlm,],
tax.lev="l2", showvar="genderMale",p.cutoff.type="p", p.cutoff=1,display="data")
meta.niceplot(metadat=metadat,sumtype="taxa",level="main",
p="p",p.adjust="p.adjust", phyla.col="rainbow", p.sig.heat="yes",
heat.forest.width.ratio =c(1.5,1), leg.key.size=0.8, leg.text.size=10,
heat.text.x.size=10, heat.text.x.angle=0, forest.axis.text.y=8,
forest.axis.text.x=10, point.ratio = c(4,2),line.ratio = c(2,1))
</code></pre>

<hr>
<h2 id='meta.taxa'>Meta-analysis of taxa/pathway abundance comparison.</h2><span id='topic+meta.taxa'></span>

<h3>Description</h3>

<p>This function does meta-analysis based on estimates and standard errors from taxa/pathway abundance comparison using random effect and fixed effect meta-analysis models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meta.taxa(
  taxcomdat,
  estimate.pattern = "Estimate.",
  se.pattern = "Std. Error.",
  summary.measure = "RR",
  pool.var = "id",
  studylab = "study",
  backtransform = FALSE,
  percent.meta = 0.5,
  p.adjust.method = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta.taxa_+3A_taxcomdat">taxcomdat</code></td>
<td>
<p>matrice of estimates and SE of all taxa/pathways combined from all included studies.</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_estimate.pattern">estimate.pattern</code></td>
<td>
<p>string pattern for estimates. Default is &quot;Estimate.&quot;.</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_se.pattern">se.pattern</code></td>
<td>
<p>string pattern for standard error. Default is &quot;Std. Error.&quot;.</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_summary.measure">summary.measure</code></td>
<td>
<p>&quot;RR&quot; for estimates from GAMLSS with BEZI family and &quot;RD&quot; for estimates from Linear/linear mixed effect model. Default is &quot;RR&quot;</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_pool.var">pool.var</code></td>
<td>
<p>name of id variable for meta-analysis. Default is &quot;id&quot;.</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_studylab">studylab</code></td>
<td>
<p>name of variable characterizing included studies. Default is &quot;study&quot;.</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_backtransform">backtransform</code></td>
<td>
<p>whether or not to perform backtransformation of the estimates. Default is FALSE.</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_percent.meta">percent.meta</code></td>
<td>
<p>the threshold percentage of number of studies that a taxa is available to do meta-analysis. Default is 0.5</p>
</td></tr>
<tr><td><code id="meta.taxa_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for multiple testing adjustment (available methods of the function p.adjust). Default is &quot;fdr&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of matrices of results for all variables in the comparison models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load saved GAMLSS-BEZI results of four studies
# for the comparison of bacterial taxa relative abundance between
# genders adjusted for breastfeeding and infant age at sample collection
data(tabsex4)
#select only taxonomies of a small phylum for meta-analysis example
# (to save running time)
tlm&lt;-tabsex4$id[grep("k__bacteria.p__fusobacteria",tabsex4$id)]
# meta-analysis
metab.sex&lt;-meta.taxa(taxcomdat=tabsex4[tabsex4$id %in% tlm,],
summary.measure="RR", pool.var="id", studylab="study",
backtransform=FALSE, percent.meta=0.5, p.adjust.method="fdr")
#show results by table and plot
#phylum
#table
metatab.show(metatab=metab.sex$random,com.pooled.tab=tabsex4[tabsex4$id %in% tlm,],
tax.lev="l2",showvar="genderMale",p.cutoff.type="p", p.cutoff=1,display="table")
</code></pre>

<hr>
<h2 id='metatab.show'>Display meta-analysis results.</h2><span id='topic+metatab.show'></span>

<h3>Description</h3>

<p>This function displays meta-analysis results of relative abundance as heatmap, forest plot, table or data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metatab.show(
  metatab,
  com.pooled.tab,
  sumvar = "taxa",
  highest.lev = "g",
  tax.lev = "l2",
  showvar,
  estimate.pattern = "Estimate.",
  se.pattern = "Std. Error.",
  p.pattern = "Pr(&gt;|t|)",
  readjust.p = FALSE,
  p.cutoff.type = "p",
  p.cutoff = 0.05,
  display = "plot",
  plot = "heatmap",
  fill.value = "log(OR)",
  grid = FALSE,
  digit = 2,
  p.digit = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metatab.show_+3A_metatab">metatab</code></td>
<td>
<p>matrice of taxa/pathway abundance comparison meta-analysis results generated from meta.taxa.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_com.pooled.tab">com.pooled.tab</code></td>
<td>
<p>matrice of taxa/pathway abundance comparison generated from taxa.compare or pathway.compare combined from all included studies.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_sumvar">sumvar</code></td>
<td>
<p>Either &quot;taxa&quot; for taxa and &quot;path&quot; for pathway.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_highest.lev">highest.lev</code></td>
<td>
<p>Highest level of bacterial taxonomies available for analysis. Options are &quot;g&quot; for genus (usually for 16S data) and &quot;s&quot; for species (usually for shortgun data).</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_tax.lev">tax.lev</code></td>
<td>
<p>taxa level to be displayed. Options are from &quot;l2&quot; (phylum) to &quot;l7&quot; (species). Default is &quot;l2&quot;.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_showvar">showvar</code></td>
<td>
<p>variable (string pattern) in the model to be displayed.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_estimate.pattern">estimate.pattern</code></td>
<td>
<p>string pattern for estimates. Default is &quot;Estimate.&quot;.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_se.pattern">se.pattern</code></td>
<td>
<p>string pattern for standard error variable. Default is &quot;Std. Error.&quot;.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_p.pattern">p.pattern</code></td>
<td>
<p>string pattern for p-value variable. Default is &quot;Pr(&gt;|t|)&quot;.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_readjust.p">readjust.p</code></td>
<td>
<p>multiple testing re-adjustment for only the level to be displayed (TRUE) or keep original multiple testing adjustment for all taxa of all levels (FALSE).Default is FALSE.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_p.cutoff.type">p.cutoff.type</code></td>
<td>
<p>type of p-value for cutoff. Options are &quot;p&quot; for p-value or &quot;p.adjust&quot; for multiple testing adjusted p-value. Default is &quot;p&quot;.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_p.cutoff">p.cutoff</code></td>
<td>
<p>cutoff p-value to be displayed. Default is 0.05.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_display">display</code></td>
<td>
<p>type of display. Options are display=c(&quot;plot&quot;,&quot;table&quot;,&quot;data&quot;)</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_plot">plot</code></td>
<td>
<p>type of plot. Options are plot=c(&quot;heatmap&quot;,&quot;forest&quot;).</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_fill.value">fill.value</code></td>
<td>
<p>name of legend.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_grid">grid</code></td>
<td>
<p>whether multiple plots will be displayed alongside. Default is FALSE.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_digit">digit</code></td>
<td>
<p>digit for estimates and 95 CI. Default is 2.</p>
</td></tr>
<tr><td><code id="metatab.show_+3A_p.digit">p.digit</code></td>
<td>
<p>digit for p-values. Default is 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot table or data.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Load saved GAMLSS-BEZI results of four studies for the comparison of
# bacterial taxa relative abundance between genders adjusted for
# breastfeeding and infant age at sample collection
data(tabsex4)
#select only taxonomies of a small phylum for meta-analysis example
# (to save running time)
tlm&lt;-tabsex4$id[grep("k__bacteria.p__fusobacteria",tabsex4$id)]
# meta-analysis
metab.sex&lt;-meta.taxa(taxcomdat=tabsex4[tabsex4$id %in% tlm,],
summary.measure="RR", pool.var="id", studylab="study",
backtransform=FALSE, percent.meta=0.5, p.adjust.method="fdr")
#show results by table and plot
#phylum
#table
metatab.show(metatab=metab.sex$random,com.pooled.tab=tabsex4[tabsex4$id %in% tlm,],
tax.lev="l2",showvar="genderMale",p.cutoff.type="p", p.cutoff=1,display="table")
#plot
metadat&lt;-metatab.show(metatab=metab.sex$random,com.pooled.tab=tabsex4[tabsex4$id %in% tlm,],
tax.lev="l2",showvar="genderMale",p.cutoff.type="p", p.cutoff=1,display="data")
meta.niceplot(metadat=metadat,sumtype="taxa",level="main",p="p",
p.adjust="p.adjust",phyla.col="rainbow",p.sig.heat="yes",
heat.forest.width.ratio =c(1.5,1), leg.key.size=0.8,
leg.text.size=10, heat.text.x.size=10, heat.text.x.angle=0,
forest.axis.text.y=8,forest.axis.text.x=10,
point.ratio = c(4,2),line.ratio = c(2,1))
</code></pre>

<hr>
<h2 id='microbiomeage'>Predict microbiome age.</h2><span id='topic+microbiomeage'></span>

<h3>Description</h3>

<p>This function predicts microbiome age using Random Forest model based on relative abundances of bacterial genera shared with the Bangladesh study (Subramanian et al 2014).
This function gets the shared genera list between the Bangladesh study and all other included studies,
get the training and test sets from Bangladesh data based on the shared genera list,
fit the train Random Forest model and predict microbiome age in the test set of Bangladesh data and data from all included studies,
check for performance of the model based on the shared genera list on Bangladesh healthy cohort data, reproduce the findings of the Bangladesh malnutrition study.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microbiomeage(l6.relabundtab, bal6)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="microbiomeage_+3A_l6.relabundtab">l6.relabundtab</code></td>
<td>
<p>list of taxa summary table from phylum up to genus level merged to mapping file outputed from QIIME of all included studies.</p>
</td></tr>
<tr><td><code id="microbiomeage_+3A_bal6">bal6</code></td>
<td>
<p>reference data for model training (taxa summary table from phylum up to genus level merged to mapping file outputed from QIIME of the Bangladesh study).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of training and test sets of Bangladesh data, shared genera list, relative abundance data of shared genera, randomforest fit, RF model performance plot,predicted microbiome age of Bangladesh data and data of other included studies.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data used for this example are available
# in the "metamicrobiomeR" package version in Github.
# Download example data from the package github repo
#setwd("your directory") #put your working directory inside the quotation marks
download.file(url = "https://github.com/nhanhocu/metamicrobiomeR/archive/master.zip",
destfile = "metamicrobiomeR-master.zip")
# unzip the .zip file
unzip(zipfile = "metamicrobiomeR-master.zip")
#Load data from each study and put in a list
#Load Bangladesh train data
patht&lt;-paste(getwd(),
"metamicrobiomeR-master/inst/extdata/QIIME_outputs/Bangladesh/tax_mapping7", sep="/")
bal6 &lt;- utils::read.delim(paste(patht, "Subramanian_et_al_mapping_file_L6.txt", sep="/"))
colnames(bal6)&lt;-tolower(colnames(bal6))
# Load data of 3 other studies
#format for data of other studies should be similar to Bangladesh data,
# must have 'age.sample' variable as age of infant at stool sample collection
data(gtab.3stud)
names(gtab.3stud)
#predict microbiome age on Bangladesh data and
# data of other three studies based on shared genera across 4 studies
#Predict microbiome age on train and test data (take time to run)
miage&lt;-microbiomeage(l6.relabundtab=gtab.3stud, bal6=bal6)
#list of shared genera that are available in the Bangladesh study
# and other included studies
miage$sharedgenera.importance
#check performance
gridExtra::grid.arrange(miage$performanceplot$ptrain, miage$performanceplot$ptest,nrow=1)
#replicate the findings of Subramanian et al paper
ggplot2::ggplot() +geom_point(data=miage$microbiomeage.bangladesh$all,
aes(x=age.sample, y=age.predicted, colour=health_analysis_groups))

</code></pre>

<hr>
<h2 id='pathway.compare'>Compare (kegg) pathway abundance</h2><span id='topic+pathway.compare'></span>

<h3>Description</h3>

<p>This is a slightly modified version of the taxa.compare function.
It compares pathway abundance generated from PICRUSt analysis between groups using different methods (apply to pathway summary tables already merged to mapping file and put in a list (e.g.level 1, 2 and 3)).
Specifically, it compares relative abundances of bacterial functional pathways at all levels using GAMLSS or LM/LMEM and compares of log(absolute abundances) of bacterial functional pathways at all levels using LM/LMEM.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pathway.compare(
  pathtab,
  mapfile,
  sampleid = "sampleid",
  pathsum = "rel",
  stat.med = "gamlss",
  transform = "none",
  comvar,
  adjustvar,
  personid = "personid",
  longitudinal = "yes",
  p.adjust.method = "fdr",
  percent.filter = 0.05,
  relabund.filter = 5e-05,
  pooldata = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="pathway.compare_+3A_pathtab">pathtab</code></td>
<td>
<p>list of pathway abundance table of all levels.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_mapfile">mapfile</code></td>
<td>
<p>mapping file or file containing covariates.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_sampleid">sampleid</code></td>
<td>
<p>variable containing sample id to be matched between pathway abundance table and mapping file.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_pathsum">pathsum</code></td>
<td>
<p>type of abundance to be compared. Options are &quot;rel&quot; for relative abundance or &quot;log&quot; for log of absolute abundance. Default is &quot;rel&quot;.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_stat.med">stat.med</code></td>
<td>
<p>statistical method for comparison. stat.med can be &quot;lm&quot; for LM/LMEM (usable for both pathsum=&quot;rel&quot; or &quot;log&quot;) or &quot;gamlss&quot; for GAMLSS with BEZI family (gamlss only make sense if pathsum=&quot;rel&quot; ).</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_transform">transform</code></td>
<td>
<p>transformation of relative abundance data. Options are &quot;none&quot; for no transformation, &quot;asin.sqrt&quot; for arcsine transformation, &quot;logit&quot; for logit transformation. Default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_comvar">comvar</code></td>
<td>
<p>main variable for comparison.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_adjustvar">adjustvar</code></td>
<td>
<p>variables to be adjusted.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_personid">personid</code></td>
<td>
<p>name of variable for person id in mapping file (applicable for longitudinal data)</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_longitudinal">longitudinal</code></td>
<td>
<p>whether the data is longitudinal. Default is &quot;yes&quot;.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for multiple testing adjustment. Available options are those of the p.adjust function. Default is &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_percent.filter">percent.filter</code></td>
<td>
<p>prevalence threshold (the percentage of number of samples the taxa/pathway available). Default is 0.05.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_relabund.filter">relabund.filter</code></td>
<td>
<p>relative abundance threshold (the minimum of the average relative abundance for a taxa/pathway to be retained). Default is 0.00005.</p>
</td></tr>
<tr><td><code id="pathway.compare_+3A_pooldata">pooldata</code></td>
<td>
<p>whether the data is pooled from multiple studies. Default is FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrice of coefficients, standard errors, p-values and multiple testing adjusted p-values of all variables in the models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load Bangladesh pathway and metadata
data(kegg.12)
data(covar.rm)
# Comparison of pathway relative abundances for some first pathways of level 1 only
# and assuming crosssectional data (to save running time)
path1&lt;-pathway.compare(pathtab=list(kegg.12[[1]][, 1:2]),
mapfile=covar.rm,sampleid="sampleid",pathsum="rel", stat.med="gamlss",
comvar="gender",adjustvar=c("age.sample","bf"), longitudinal="no",
p.adjust.method="fdr", percent.filter=0.05,relabund.filter=0.00005)
taxcomtab.show(taxcomtab=path1$l1, sumvar="path",tax.lev="l2",
tax.select="none", showvar="genderMale", p.adjust.method="fdr",p.cutoff=1)
</code></pre>

<hr>
<h2 id='read.multi'>Read multiple files</h2><span id='topic+read.multi'></span>

<h3>Description</h3>

<p>This function reads multiple files of the same pattern in a directory into R.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.multi(
  patht,
  patternt = ".txt",
  assignt = "no",
  study = NULL,
  tolower.colnames = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.multi_+3A_patht">patht</code></td>
<td>
<p>path to files.</p>
</td></tr>
<tr><td><code id="read.multi_+3A_patternt">patternt</code></td>
<td>
<p>pattern of files. Options are &quot;.txt&quot; or &quot;.csv&quot;.</p>
</td></tr>
<tr><td><code id="read.multi_+3A_assignt">assignt</code></td>
<td>
<p>assign files. Default is &quot;no&quot;.</p>
</td></tr>
<tr><td><code id="read.multi_+3A_study">study</code></td>
<td>
<p>name of the study. Default is NULL.</p>
</td></tr>
<tr><td><code id="read.multi_+3A_tolower.colnames">tolower.colnames</code></td>
<td>
<p>turn all column names to lower case. Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of all data files in the path
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# The data used for this example are available
# in the "metamicrobiomeR" package version in Github.
# Download example data from the package github repo
#setwd("your directory") #put your working directory inside the quotation marks
download.file(url = "https://github.com/nhanhocu/metamicrobiomeR/archive/master.zip",
destfile = "metamicrobiomeR-master.zip")
# unzip the .zip file
unzip(zipfile = "metamicrobiomeR-master.zip")
patht&lt;-paste(getwd(),
"metamicrobiomeR-master/inst/extdata/QIIME_outputs/Bangladesh/alpha_div_collated", sep="/")
alpha.ba&lt;-read.multi(patht=patht,patternt=".txt", assignt="no",study="Bangladesh")

</code></pre>

<hr>
<h2 id='tabsex4'>Combined data for meta-analysis.</h2><span id='topic+tabsex4'></span>

<h3>Description</h3>

<p>Result outputs of differential abundance analysis using GAMLSS_BEZI
from &quot;taxa.compare&quot; function combined from
4 studies for meta-analysis. The comparison was between gender
adjusted for age of infants at sample collection.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(tabsex4)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 701 rows and 23 variables.
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>
<p>Bender et al. Sci Transl Med. 2016 Jul 27; 8(349): 349ra100.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5301310/">PubMed</a>)
</p>
<p>Pannaraj et al. JAMA Pediatr. 2017;90095(7):647–54.
(<a href="https://pubmed.ncbi.nlm.nih.gov/28492938/">PubMed</a>)
</p>
<p>Thompson et al. Front Cell Infect Microbiol. 2015;5:3.
(<a href="https://pubmed.ncbi.nlm.nih.gov/25705611/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># load saved GAMLSS-BEZI results of four studies
# for the comparison of bacterial taxa relative abundance between
# genders adjusted for breastfeeding and infant age at sample collection
data(tabsex4)
#select only taxonomies of a small phylum for meta-analysis example
# (to save running time)
tlm&lt;-tabsex4$id[grep("k__bacteria.p__fusobacteria",tabsex4$id)]
# meta-analysis
metab.sex&lt;-meta.taxa(taxcomdat=tabsex4[tabsex4$id %in% tlm,],
summary.measure="RR", pool.var="id", studylab="study",
backtransform=FALSE, percent.meta=0.5, p.adjust.method="fdr")
#show results by table and plot
#phylum table
metatab.show(metatab=metab.sex$random,com.pooled.tab=tabsex4[tabsex4$id %in% tlm,],
tax.lev="l2",showvar="genderMale",p.cutoff.type="p", p.cutoff=1,display="table")
</code></pre>

<hr>
<h2 id='taxa.compare'>Compare taxa relative abundance</h2><span id='topic+taxa.compare'></span>

<h3>Description</h3>

<p>This function compares taxa relative abundance summary tables at all levels between groups using GAMLSS with BEZI or Linear/Linear Mixed Effect models (LM/LMEM) after filtering (using prevalence and relative abundance thresholds).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.compare(
  taxtab,
  propmed.rel = "gamlss",
  transform = "none",
  zeroreplace.method = "none",
  comvar,
  adjustvar,
  personid = "personid",
  longitudinal = "yes",
  percent.filter = 0.05,
  relabund.filter = 5e-05,
  p.adjust.method = "fdr"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.compare_+3A_taxtab">taxtab</code></td>
<td>
<p>taxa relative abundance table (already merged to mapping file) from phylum to species or any preferred highest taxa level.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_propmed.rel">propmed.rel</code></td>
<td>
<p>statistical method for comparing relative abundance. Options are &quot;lm&quot; for LM/LMEM or &quot;gamlss&quot; for GAMLSS with BEZI family.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_transform">transform</code></td>
<td>
<p>transformation of relative abundance data. Options are &quot;none&quot; for no transformation, &quot;gmpr&quot; for Geometric Mean of Pairwise Ratios (GMPR) normalization, &quot;asin.sqrt&quot; for arcsine transformation, &quot;logit&quot; for logit transformation, &quot;clr&quot; for centered log ratio transformation. Default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_zeroreplace.method">zeroreplace.method</code></td>
<td>
<p>Method for zero replacement implemented in R package *zCompositions*. Options are &quot;none&quot; for no replacement, &quot;multKM&quot; for Multiplicative Kaplan-Meier smoothing spline (KMSS) replacement, &quot;multLN&quot; for Multiplicative lognormal replacement, &quot;multRepl&quot; for Multiplicative simple replacement, &quot;lrEM&quot; for Log-ratio EM algorithm, &quot;lrDA&quot; for Log-ratio DA algorithm. Default is &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_comvar">comvar</code></td>
<td>
<p>main variable for comparison</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_adjustvar">adjustvar</code></td>
<td>
<p>variables to be adjusted.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_personid">personid</code></td>
<td>
<p>name of variable for person id (applicable for longitudinal data)</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_longitudinal">longitudinal</code></td>
<td>
<p>whether data is longitudinal? Options are &quot;yes&quot; or &quot;no&quot;. Default is &quot;yes&quot;.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_percent.filter">percent.filter</code></td>
<td>
<p>prevalence threshold (the percentage of number of samples the taxa/pathway available). Default is 0.05.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_relabund.filter">relabund.filter</code></td>
<td>
<p>relative abundance threshold (the minimum of the average relative abundance for a taxa/pathway to be retained). Default is 0.00005.</p>
</td></tr>
<tr><td><code id="taxa.compare_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for multiple testing adjustment. Options are those of the p.adjust.methods of stats:: p.adjust function. Default for this function is &quot;fdr&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>matrice of coefficients, standard errors, p-values and multiple testing adjusted p-values of all variables in the models.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load summary tables of bacterial taxa relative abundance from Bangladesh data
data(taxtab6)
tab6&lt;-as.data.frame(taxtab6)
tl&lt;-colnames(taxtab6)[grep("k__bacteria.p__fusobacteria",colnames(taxtab6))]
taxacom.ex&lt;-taxa.compare(taxtab=tab6[,c("personid","x.sampleid","bf","age.sample",tl)],
propmed.rel="gamlss",comvar="bf",adjustvar="age.sample",
longitudinal="yes",p.adjust.method="fdr")
</code></pre>

<hr>
<h2 id='taxa.filter'>Filter relative abundance data</h2><span id='topic+taxa.filter'></span>

<h3>Description</h3>

<p>This function filters bacterial taxa or pathway relative abundance tables based on the percentage of samples with their availability (prevalence) and relative abundance thresholds.
It will remove taxa/pathway with relative abundance &lt;relabund.filter and available in &lt;percent.filter of number of samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.filter(taxtab, percent.filter = 0.05, relabund.filter = 5e-05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.filter_+3A_taxtab">taxtab</code></td>
<td>
<p>taxa/pathway relative abundance table.</p>
</td></tr>
<tr><td><code id="taxa.filter_+3A_percent.filter">percent.filter</code></td>
<td>
<p>prevalence threshold (the percentage of number of samples the taxa/pathway available). Default is 0.05.</p>
</td></tr>
<tr><td><code id="taxa.filter_+3A_relabund.filter">relabund.filter</code></td>
<td>
<p>relative abundance threshold (the minimum of the average relative abundance for a taxa/pathway to be retained). Default is 0.00005.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of all taxa/pathways retained after filtering.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load summary tables of bacterial taxa relative abundance from Bangladesh data
data(taxtab6)
taxlist.rm&lt;-taxa.filter(taxtab=taxtab6,percent.filter = 0.05,
relabund.filter = 0.00005)
</code></pre>

<hr>
<h2 id='taxa.mean.plot'>Plot mean taxa abundance</h2><span id='topic+taxa.mean.plot'></span>

<h3>Description</h3>

<p>This function visualize mean relative abundance by group as stacked plots.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.mean.plot(
  tabmean,
  sumvar = "taxa",
  tax.select = "none",
  tax.lev = "l2",
  comvar,
  groupvar,
  mean.filter = 0.005,
  pallete.by.phylum = FALSE,
  show.taxname = "full",
  legend.position = "right",
  xlab = "",
  ylab = "Relative abundance"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.mean.plot_+3A_tabmean">tabmean</code></td>
<td>
<p>table of mean abundance generated from taxa.meansdn.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_sumvar">sumvar</code></td>
<td>
<p>variable to be plotted. Options are c(&quot;taxa&quot;,&quot;path&quot;). Default is &quot;taxa&quot;</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_tax.select">tax.select</code></td>
<td>
<p>list of selected taxa/pathways to be plotted. Default is &quot;none&quot; or plot all taxa/pathways.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_tax.lev">tax.lev</code></td>
<td>
<p>taxa level to be visualized. Options are from &quot;l2&quot; (phylum) to &quot;l7&quot; (species). Default is &quot;l2&quot;. If sumvar=&quot;path&quot;, all pathways will be visualized.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_comvar">comvar</code></td>
<td>
<p>main variable for comparison.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_groupvar">groupvar</code></td>
<td>
<p>variable for stratifying.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_mean.filter">mean.filter</code></td>
<td>
<p>mean abundance filtering threshold (only plot those with mean abundance&gt;threshold and plot all those with mean abundance &lt;threshold as &quot;other&quot;).</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_pallete.by.phylum">pallete.by.phylum</code></td>
<td>
<p>whether each pallete of color for each phylum. Default is FALSE (plot distinc colors).</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_show.taxname">show.taxname</code></td>
<td>
<p>whether show &quot;full&quot; taxa name or &quot;short&quot; name. Default is &quot;full&quot;.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_legend.position">legend.position</code></td>
<td>
<p>position of legend. Options are c(&quot;right&quot;, &quot;left&quot;,&quot;bottom&quot;,&quot;top&quot;,&quot;none&quot;) as in ggplot2. Default is &quot;right&quot;.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_xlab">xlab</code></td>
<td>
<p>label for x-axis. Default is &quot;Chronological age (month)&quot;.</p>
</td></tr>
<tr><td><code id="taxa.mean.plot_+3A_ylab">ylab</code></td>
<td>
<p>label for y-axis. Default is &quot;Relative abundance&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of ggplot2 object and list of taxa/pathways plotted (those with mean abundance &gt;mean.filter).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load summary tables of bacterial taxa relative abundance from Bangladesh data
data(taxtab6)
taxlist.rm&lt;-taxa.filter(taxtab=taxtab6, percent.filter = 0.05, relabund.filter = 0.00005)
taxa.meansdn.rm&lt;-taxa.meansdn(taxtab=taxtab6,sumvar="bf",groupvar="age.sample")
taxa.meansdn.rm&lt;-taxa.meansdn.rm[taxa.meansdn.rm$bf!="No_BF",]
taxa.meansdn.rm$bf&lt;-gdata::drop.levels(taxa.meansdn.rm$bf,reorder=FALSE)
#phylum
p.bf.l2&lt;-taxa.mean.plot(tabmean=taxa.meansdn.rm, tax.lev="l2",
comvar="bf", groupvar="age.sample",mean.filter=0.005, show.taxname="short")
p.bf.l2$p
</code></pre>

<hr>
<h2 id='taxa.meansdn'>Summarize abundance by group</h2><span id='topic+taxa.meansdn'></span>

<h3>Description</h3>

<p>This function summarizes taxa/pathway abundance tables to provide mean, sd, count by groups.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa.meansdn(
  taxtab,
  sumvar,
  groupvar,
  percent.filter = 0.05,
  relabund.filter = 5e-05,
  othervar = "none"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa.meansdn_+3A_taxtab">taxtab</code></td>
<td>
<p>taxa/pathway  abundance table from phylum to species or any preferred highest taxa level.</p>
</td></tr>
<tr><td><code id="taxa.meansdn_+3A_sumvar">sumvar</code></td>
<td>
<p>main variable for summary</p>
</td></tr>
<tr><td><code id="taxa.meansdn_+3A_groupvar">groupvar</code></td>
<td>
<p>variable to be stratified.</p>
</td></tr>
<tr><td><code id="taxa.meansdn_+3A_percent.filter">percent.filter</code></td>
<td>
<p>prevalence threshold (the percentage of number of samples the taxa/pathway available). Default is 0.05.</p>
</td></tr>
<tr><td><code id="taxa.meansdn_+3A_relabund.filter">relabund.filter</code></td>
<td>
<p>relative abundance threshold (the minimum of the average relative abundance for a taxa/pathway to be retained). Default is 0.00005.</p>
</td></tr>
<tr><td><code id="taxa.meansdn_+3A_othervar">othervar</code></td>
<td>
<p>vector of variables that are not abundance variables to be summarized. Default is &quot;none&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>table of mean, sd, count by group.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load summary tables of bacterial taxa relative abundance from Bangladesh data
data(taxtab6)
taxa.meansdn.rm&lt;-taxa.meansdn(taxtab=taxtab6,sumvar="bf",groupvar="age.sample")
</code></pre>

<hr>
<h2 id='taxcomtab.show'>Display abundance comparison results.</h2><span id='topic+taxcomtab.show'></span>

<h3>Description</h3>

<p>This function displays taxa/pathway abundance comparison results as table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxcomtab.show(
  taxcomtab,
  sumvar = "taxa",
  tax.lev = "l2",
  tax.select = "none",
  showvar,
  readjust.p = FALSE,
  p.adjust.method = "fdr",
  p.cutoff = 0.05,
  digit = 2,
  p.digit = 4
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxcomtab.show_+3A_taxcomtab">taxcomtab</code></td>
<td>
<p>table of taxa abundance comparison generated from taxa.compare.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_sumvar">sumvar</code></td>
<td>
<p>Options are &quot;taxa&quot; for bacterial taxa and &quot;path&quot; for pathway. Default is &quot;taxa&quot;</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_tax.lev">tax.lev</code></td>
<td>
<p>taxa level to be displayed. Options are from &quot;l2&quot; (phylum) to &quot;l7&quot; (species). Default is &quot;l2&quot;.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_tax.select">tax.select</code></td>
<td>
<p>selected list of taxa to be displayed. Default is &quot;none&quot; or display all available taxa.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_showvar">showvar</code></td>
<td>
<p>variable (pattern) in the model to be displayed.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_readjust.p">readjust.p</code></td>
<td>
<p>multiple testing re-adjustment for only the level to be displayed (TRUE) or keep original multiple testing adjustment for all taxa of all levels (FALSE).</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>method for multiple testing adjustment. Available options are those of the p.adjust function. Default is &quot;fdr&quot;.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_p.cutoff">p.cutoff</code></td>
<td>
<p>cutoff p-value to be displayed. Default is 0.05.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_digit">digit</code></td>
<td>
<p>digit for estimates and 95 CI. Default is 2.</p>
</td></tr>
<tr><td><code id="taxcomtab.show_+3A_p.digit">p.digit</code></td>
<td>
<p>digit for p-values. Default is 4.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table of results.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load summary tables of bacterial taxa relative abundance from Bangladesh data
data(taxtab6)
tab6&lt;-as.data.frame(taxtab6)
#Comparison of bacterial taxa relative abundance using GAMLSS
# Only run on a few taxa of a phylum to save running time
tl&lt;-colnames(taxtab6)[grep("k__bacteria.p__fusobacteria",colnames(taxtab6))]
taxacom.ex&lt;-taxa.compare(taxtab=tab6[,c("personid","x.sampleid","bf","age.sample",tl)],
propmed.rel="gamlss",comvar="bf",adjustvar="age.sample",
longitudinal="yes",p.adjust.method="fdr")
# show phylum results
taxcomtab.show(taxcomtab=taxacom.ex,tax.select="none",
showvar="bfNon_exclusiveBF", tax.lev="l2",
readjust.p=TRUE,p.adjust.method="fdr",p.cutoff = 1)
</code></pre>

<hr>
<h2 id='taxtab6'>Taxonomic relative abundance data.</h2><span id='topic+taxtab6'></span>

<h3>Description</h3>

<p>Monthly longitudinal relative abundance data from phylum to genus level of
50 infants from birth to 2 year of life.
Mapping file is merged to the data for ready use.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(taxtab6)
</code></pre>


<h3>Format</h3>

<p>A data frame with 322 row (samples) and 803 variables (including mapping varilable and bacterial taxonomies from phylum to genus level).
</p>


<h3>Source</h3>

<p><a href="https://gordonlab.wustl.edu/supplemental-data/supplemental-data-portal/subramanian-et-al-2014/">Gordon Lab</a>
</p>


<h3>References</h3>

<p>Subramanian et al. Nature. 2014 Jun 19; 510(7505): 417–421.
(<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4189846/">PubMed</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#Load summary tables of bacterial taxa relative abundance from Bangladesh data
data(taxtab6)
tab6&lt;-as.data.frame(taxtab6)
tl&lt;-colnames(taxtab6)[grep("k__bacteria.p__fusobacteria",colnames(taxtab6))]
taxacom.ex&lt;-taxa.compare(taxtab=tab6[,c("personid","x.sampleid","bf","age.sample",tl)],
propmed.rel="gamlss",comvar="bf",adjustvar="age.sample",
longitudinal="yes",p.adjust.method="fdr")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
