<!DOCTYPE html><html><head><title>Help for package MetaNet</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MetaNet}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#%&lt;&gt;%'><p>Assignment pipe</p></a></li>
<li><a href='#anno_edge'><p>Use dataframe to annotate edges of an igraph</p></a></li>
<li><a href='#anno_vertex'><p>Use data.frame to annotate vertexes of metanet</p></a></li>
<li><a href='#arc_count'><p>Edgelist</p></a></li>
<li><a href='#arc_taxonomy'><p>Edgelist</p></a></li>
<li><a href='#as_arc'><p>Layout as a arc</p></a></li>
<li><a href='#as_circle_tree'><p>Layout as a circle_tree</p></a></li>
<li><a href='#as_line'><p>Layout as a line</p></a></li>
<li><a href='#as_polyarc'><p>Layout as a polyarc</p></a></li>
<li><a href='#as_polycircle'><p>Layout as a polycircle</p></a></li>
<li><a href='#as_polygon'><p>Layout as a polygon</p></a></li>
<li><a href='#as.ggig'><p>Transfer an igraph object to a ggig</p></a></li>
<li><a href='#c_net_annotate'><p>Annotate a metanet</p></a></li>
<li><a href='#c_net_build'><p>Construct a metanet from a corr object</p></a></li>
<li><a href='#c_net_calculate'><p>Calculate correlation for one or two t(otutab), or distance for one t(otutab).</p></a></li>
<li><a href='#c_net_filter'><p>Filter a network according to some attributes</p></a></li>
<li><a href='#c_net_from_edgelist'><p>Construct a network from edge_list dataframe</p></a></li>
<li><a href='#c_net_layout'><p>Layout coordinates</p></a></li>
<li><a href='#c_net_plot'><p>Plot a metanet</p></a></li>
<li><a href='#c_net_save'><p>Save network file</p></a></li>
<li><a href='#c_net_set'><p>Set basic attributes from totu table</p></a></li>
<li><a href='#c_net_stability'><p>Evaluate the stability of a network</p></a></li>
<li><a href='#c_net_union'><p>Union two networks</p></a></li>
<li><a href='#c_net_update'><p>Update a metanet object or transform igraph object to metanet object</p></a></li>
<li><a href='#cal_sim'><p>Calculate similarity for one t(otutab)</p></a></li>
<li><a href='#check_tabs'><p>Check tables and extract common samples</p></a></li>
<li><a href='#clean_igraph'><p>Clean a igraph object</p></a></li>
<li><a href='#co_net'><p>MetaNet networks</p></a></li>
<li><a href='#co_net_rmt'><p>MetaNet networks</p></a></li>
<li><a href='#co_net2'><p>MetaNet networks</p></a></li>
<li><a href='#Cohesion'><p>Cohesion calculation</p></a></li>
<li><a href='#df2net_tree'><p>Transform a dataframe to a network edgelist.</p></a></li>
<li><a href='#extract_sub_net'><p>Extract sub-network from the whole network</p></a></li>
<li><a href='#fast_cor'><p>Fast correlation calculation</p></a></li>
<li><a href='#filter_n_module'><p>Filter some modules as others</p></a></li>
<li><a href='#fit_power'><p>Fit power-law distribution for an igraph</p></a></li>
<li><a href='#g_layout'><p>Layout with group</p></a></li>
<li><a href='#g_layout_polygon'><p>Layout with group as a polygon</p></a></li>
<li><a href='#get_community'><p>Get community</p></a></li>
<li><a href='#get_e'><p>Get edge information</p></a></li>
<li><a href='#get_group_skeleton'><p>Get skeleton network according to a group</p></a></li>
<li><a href='#get_module'><p>Get module</p></a></li>
<li><a href='#get_module_eigen'><p>Get module_eigen</p></a></li>
<li><a href='#get_n'><p>Get network information</p></a></li>
<li><a href='#get_v'><p>Get vertex information</p></a></li>
<li><a href='#input_corr'><p>Import corr from .csv file</p></a></li>
<li><a href='#input_gephi'><p>Input a graphml file exported by Gephi</p></a></li>
<li><a href='#is_metanet'><p>Is this object a metanet object?</p></a></li>
<li><a href='#links_stat'><p>Link summary of the network</p></a></li>
<li><a href='#metab'><p>MetaNet networks abundance</p></a></li>
<li><a href='#metab_g'><p>MetaNet networks metadata</p></a></li>
<li><a href='#MetaNet-package'><p>MetaNet: Network Analysis for Omics Data</p></a></li>
<li><a href='#micro'><p>MetaNet networks abundance</p></a></li>
<li><a href='#micro_g'><p>MetaNet networks metadata</p></a></li>
<li><a href='#module_detect'><p>Detect the modules</p></a></li>
<li><a href='#module_eigen'><p>Calculate the eigenvalue of each module and correlation of nodes and eigenvalue (node_eigen_cor).</p></a></li>
<li><a href='#module_net'><p>Generate a n-modules network</p></a></li>
<li><a href='#multi_net_build'><p>Multi-omics network build</p></a></li>
<li><a href='#multi1'><p>MetaNet networks</p></a></li>
<li><a href='#nc'><p>Calculate natural_connectivity</p></a></li>
<li><a href='#net_par'><p>Calculate all topological indexes of a network</p></a></li>
<li><a href='#netD3plot'><p>plot use networkD3</p></a></li>
<li><a href='#olympic_rings_net'><p>Plot olympic rings using network</p></a></li>
<li><a href='#p.adjust.table'><p>p.adjust apply on a correlation table (matrix or data.frame)</p></a></li>
<li><a href='#plot.ggig'><p>Plot a ggig</p></a></li>
<li><a href='#plot.metanet'><p>Plot a metanet</p></a></li>
<li><a href='#plot.rmt_res'><p>Plot a rmt_res</p></a></li>
<li><a href='#plot.robust'><p>Plot robust</p></a></li>
<li><a href='#plot.robustness'><p>Plot robustness</p></a></li>
<li><a href='#plot.vulnerability'><p>Plot vulnerability</p></a></li>
<li><a href='#print.cohesion'><p>Print method for 'cohesion' objects</p></a></li>
<li><a href='#print.coors'><p>Print method for 'coors' objects</p></a></li>
<li><a href='#print.corr'><p>Print method for 'corr' objects</p></a></li>
<li><a href='#print.ggig'><p>Print method for 'ggig' objects</p></a></li>
<li><a href='#print.metanet'><p>Print method for 'metanet' objects</p></a></li>
<li><a href='#print.robust'><p>Print method for 'robust' objects</p></a></li>
<li><a href='#print.robustness'><p>Print method for 'robustness' objects</p></a></li>
<li><a href='#print.vulnerability'><p>Print method for 'vulnerability' objects</p></a></li>
<li><a href='#rand_net'><p>Degree distribution comparison with random network</p></a></li>
<li><a href='#rand_net_par'><p>Net_pars of many random network</p></a></li>
<li><a href='#RMT_threshold'><p>Get RMT threshold for a correlation matrix</p></a></li>
<li><a href='#show_MetaNet_logo'><p>Show MetaNet logo</p></a></li>
<li><a href='#smallworldness'><p>Calculate small-world coefficient</p></a></li>
<li><a href='#summ_2col'><p>Summaries two columns information</p></a></li>
<li><a href='#summary_module'><p>Summary module index</p></a></li>
<li><a href='#to_module_net'><p>Transformation a network to a module network</p></a></li>
<li><a href='#transc'><p>MetaNet networks abundance</p></a></li>
<li><a href='#transc_g'><p>MetaNet networks metadata</p></a></li>
<li><a href='#twocol_edgelist'><p>Quick build a metanet from two columns table</p></a></li>
<li><a href='#venn_net'><p>Venn network</p></a></li>
<li><a href='#zp_analyse'><p>Zi-Pi calculate</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Network Analysis for Omics Data</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Description:</td>
<td>Comprehensive network analysis package.
    Calculate correlation network fastly, accelerate lots of analysis by parallel computing.
    Support for multi-omics data, search sub-nets fluently.
    Handle bigger data, more than 10,000 nodes in each omics.
    Offer various layout method for multi-omics network and some interfaces to other software ('Gephi', 'Cytoscape', 'ggplot2'), easy to visualize.
    Provide comprehensive topology indexes calculation, including ecological network stability.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0), igraph (&ge; 1.3.5)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>graphics, dplyr, ggplot2 (&ge; 3.2.0), ggnewscale, ggrepel,
RColorBrewer, grDevices, magrittr, reshape2, stats, tibble,
utils, pcutils (&ge; 0.2.5), rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>pheatmap, vegan, stringr, foreach, doSNOW, snow, knitr,
rmarkdown, prettydoc, Hmisc, gifski, ggraph, networkD3,
ggpmisc, ggtree, treeio, circlize, ggpubr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/Asa12138/MetaNet/issues">https://github.com/Asa12138/MetaNet/issues</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/Asa12138/MetaNet">https://github.com/Asa12138/MetaNet</a></td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>biocViews:</td>
<td>DataImport, Network analysis, Omics, Software, Visualization</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-03-21 16:16:27 UTC; asa</td>
</tr>
<tr>
<td>Author:</td>
<td>Chen Peng <a href="https://orcid.org/0000-0002-9449-7606"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chen Peng &lt;pengchen2001@zju.edu.cn&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-03-25 20:40:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='+25+26lt+3B+26gt+3B+25'>Assignment pipe</h2><span id='topic++25+3C+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&lt;&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&lt;&gt;% rhs
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_lhs">lhs</code></td>
<td>
<p>A value or the magrittr placeholder.</p>
</td></tr>
<tr><td><code id="+2B25+2B26lt+2B3B+2B26gt+2B3B+2B25_+3A_rhs">rhs</code></td>
<td>
<p>A function call using the magrittr semantics.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of calling <code>rhs(lhs)</code>.
</p>

<hr>
<h2 id='anno_edge'>Use dataframe to annotate edges of an igraph</h2><span id='topic+anno_edge'></span>

<h3>Description</h3>

<p>Use dataframe to annotate edges of an igraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anno_edge(go, anno_tab, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anno_edge_+3A_go">go</code></td>
<td>
<p>metanet an igraph object</p>
</td></tr>
<tr><td><code id="anno_edge_+3A_anno_tab">anno_tab</code></td>
<td>
<p>a dataframe using to annotate (with rowname or a name column)</p>
</td></tr>
<tr><td><code id="anno_edge_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a annotated igraph object
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
anno &lt;- data.frame("from" = "s__Pelomonas_puraquae", "to" = "s__un_g__Rhizobium", new_atr = "new")
anno_edge(co_net, anno) -&gt; anno_net
</code></pre>

<hr>
<h2 id='anno_vertex'>Use data.frame to annotate vertexes of metanet</h2><span id='topic+anno_vertex'></span><span id='topic+anno_node'></span>

<h3>Description</h3>

<p>Use data.frame to annotate vertexes of metanet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>anno_vertex(go, anno_tab, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="anno_vertex_+3A_go">go</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="anno_vertex_+3A_anno_tab">anno_tab</code></td>
<td>
<p>a dataframe using to annotate (with rowname or a &quot;name&quot; column)</p>
</td></tr>
<tr><td><code id="anno_vertex_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a annotated metanet object
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
data("otutab", package = "pcutils")
anno_vertex(co_net, taxonomy)
</code></pre>

<hr>
<h2 id='arc_count'>Edgelist</h2><span id='topic+arc_count'></span>

<h3>Description</h3>

<p>Edgelist for <code>c_net_from_edgelist()</code>
</p>

<hr>
<h2 id='arc_taxonomy'>Edgelist</h2><span id='topic+arc_taxonomy'></span>

<h3>Description</h3>

<p>Edgelist for <code>c_net_from_edgelist()</code>
</p>

<hr>
<h2 id='as_arc'>Layout as a arc</h2><span id='topic+as_arc'></span>

<h3>Description</h3>

<p>Layout as a arc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_arc(angle = 0, arc = pi)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_arc_+3A_angle">angle</code></td>
<td>
<p>anticlockwise rotation angle</p>
</td></tr>
<tr><td><code id="as_arc_+3A_arc">arc</code></td>
<td>
<p>the radian of arc</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-column matrix, each row giving the coordinates of a vertex, according to the ids of the vertex ids.
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_circle_tree">as_circle_tree</a>()</code>,
<code><a href="#topic+as_line">as_line</a>()</code>,
<code><a href="#topic+as_polyarc">as_polyarc</a>()</code>,
<code><a href="#topic+as_polycircle">as_polycircle</a>()</code>,
<code><a href="#topic+as_polygon">as_polygon</a>()</code>,
<code><a href="#topic+c_net_layout">c_net_layout</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as_arc()(co_net)
c_net_plot(co_net, coors = as_arc(pi / 2), rescale = FALSE)
</code></pre>

<hr>
<h2 id='as_circle_tree'>Layout as a circle_tree</h2><span id='topic+as_circle_tree'></span>

<h3>Description</h3>

<p>Layout as a circle_tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_circle_tree()
</code></pre>


<h3>Value</h3>

<p>A two-column matrix, each row giving the coordinates of a vertex, according to the ids of the vertex ids.
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_arc">as_arc</a>()</code>,
<code><a href="#topic+as_line">as_line</a>()</code>,
<code><a href="#topic+as_polyarc">as_polyarc</a>()</code>,
<code><a href="#topic+as_polycircle">as_polycircle</a>()</code>,
<code><a href="#topic+as_polygon">as_polygon</a>()</code>,
<code><a href="#topic+c_net_layout">c_net_layout</a>()</code>
</p>

<hr>
<h2 id='as_line'>Layout as a line</h2><span id='topic+as_line'></span>

<h3>Description</h3>

<p>Layout as a line
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_line(angle = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_line_+3A_angle">angle</code></td>
<td>
<p>anticlockwise rotation angle</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-column matrix, each row giving the coordinates of a vertex, according to the ids of the vertex ids.
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_arc">as_arc</a>()</code>,
<code><a href="#topic+as_circle_tree">as_circle_tree</a>()</code>,
<code><a href="#topic+as_polyarc">as_polyarc</a>()</code>,
<code><a href="#topic+as_polycircle">as_polycircle</a>()</code>,
<code><a href="#topic+as_polygon">as_polygon</a>()</code>,
<code><a href="#topic+c_net_layout">c_net_layout</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as_line()(co_net)
c_net_plot(co_net, coors = as_line(pi / 2))
</code></pre>

<hr>
<h2 id='as_polyarc'>Layout as a polyarc</h2><span id='topic+as_polyarc'></span>

<h3>Description</h3>

<p>Layout as a polyarc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_polyarc(n = 3, space = pi/3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_polyarc_+3A_n">n</code></td>
<td>
<p>how many arcs of this poly_arc</p>
</td></tr>
<tr><td><code id="as_polyarc_+3A_space">space</code></td>
<td>
<p>the space between each arc, default: pi/3</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-column matrix, each row giving the coordinates of a vertex, according to the ids of the vertex ids.
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_arc">as_arc</a>()</code>,
<code><a href="#topic+as_circle_tree">as_circle_tree</a>()</code>,
<code><a href="#topic+as_line">as_line</a>()</code>,
<code><a href="#topic+as_polycircle">as_polycircle</a>()</code>,
<code><a href="#topic+as_polygon">as_polygon</a>()</code>,
<code><a href="#topic+c_net_layout">c_net_layout</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as_polyarc()(co_net)
</code></pre>

<hr>
<h2 id='as_polycircle'>Layout as a polycircle</h2><span id='topic+as_polycircle'></span>

<h3>Description</h3>

<p>Layout as a polycircle
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_polycircle(n = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_polycircle_+3A_n">n</code></td>
<td>
<p>how many circles of this polycircle</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-column matrix, each row giving the coordinates of a vertex, according to the ids of the vertex ids.
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_arc">as_arc</a>()</code>,
<code><a href="#topic+as_circle_tree">as_circle_tree</a>()</code>,
<code><a href="#topic+as_line">as_line</a>()</code>,
<code><a href="#topic+as_polyarc">as_polyarc</a>()</code>,
<code><a href="#topic+as_polygon">as_polygon</a>()</code>,
<code><a href="#topic+c_net_layout">c_net_layout</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as_polycircle()(co_net)
</code></pre>

<hr>
<h2 id='as_polygon'>Layout as a polygon</h2><span id='topic+as_polygon'></span>

<h3>Description</h3>

<p>Layout as a polygon
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_polygon(n = 3, line_curved = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_polygon_+3A_n">n</code></td>
<td>
<p>how many edges of this polygon</p>
</td></tr>
<tr><td><code id="as_polygon_+3A_line_curved">line_curved</code></td>
<td>
<p>line_curved 0~0.5</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A two-column matrix, each row giving the coordinates of a vertex, according to the ids of the vertex ids.
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_arc">as_arc</a>()</code>,
<code><a href="#topic+as_circle_tree">as_circle_tree</a>()</code>,
<code><a href="#topic+as_line">as_line</a>()</code>,
<code><a href="#topic+as_polyarc">as_polyarc</a>()</code>,
<code><a href="#topic+as_polycircle">as_polycircle</a>()</code>,
<code><a href="#topic+c_net_layout">c_net_layout</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as_polygon()(co_net)
</code></pre>

<hr>
<h2 id='as.ggig'>Transfer an igraph object to a ggig</h2><span id='topic+as.ggig'></span>

<h3>Description</h3>

<p>Transfer an igraph object to a ggig
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.ggig(go, coors = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.ggig_+3A_go">go</code></td>
<td>
<p>igraph or meatnet</p>
</td></tr>
<tr><td><code id="as.ggig_+3A_coors">coors</code></td>
<td>
<p>coordinates for nodes,columns: name, X, Y</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggig object
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>as.ggig(co_net, coors = c_net_layout(co_net)) -&gt; ggig
plot(ggig)
as.ggig(multi1, coors = c_net_layout(multi1)) -&gt; ggig
plot(ggig, labels_num = 0.3)
</code></pre>

<hr>
<h2 id='c_net_annotate'>Annotate a metanet</h2><span id='topic+c_net_annotate'></span>

<h3>Description</h3>

<p>Annotate a metanet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_annotate(go, anno_tab, mode = "v", verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_annotate_+3A_go">go</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="c_net_annotate_+3A_anno_tab">anno_tab</code></td>
<td>
<p>a dataframe using to annotate (mode v, e), or a list (mode n)</p>
</td></tr>
<tr><td><code id="c_net_annotate_+3A_mode">mode</code></td>
<td>
<p>&quot;v&quot; for vertex, &quot;e&quot; for edge, &quot;n&quot; for network</p>
</td></tr>
<tr><td><code id="c_net_annotate_+3A_verbose">verbose</code></td>
<td>
<p>logical</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a annotated metanet object
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
anno &lt;- data.frame("name" = "s__Pelomonas_puraquae", new_atr = "new")
co_net_new &lt;- c_net_annotate(co_net, anno, mode = "v")
get_v(co_net_new, c("name", "new_atr"))

anno &lt;- data.frame("from" = "s__Pelomonas_puraquae", "to" = "s__un_g__Rhizobium", new_atr = "new")
co_net_new &lt;- c_net_annotate(co_net, anno, mode = "e")
get_e(co_net_new, c("from", "to", "new_atr"))

co_net_new &lt;- c_net_annotate(co_net, list(new_atr = "new"), mode = "n")
get_n(co_net_new)
</code></pre>

<hr>
<h2 id='c_net_build'>Construct a metanet from a corr object</h2><span id='topic+c_net_build'></span>

<h3>Description</h3>

<p>Construct a metanet from a corr object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_build(
  corr,
  r_threshold = 0.6,
  p_threshold = 0.05,
  use_p_adj = TRUE,
  delete_single = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_build_+3A_corr">corr</code></td>
<td>
<p>corr object from <code>c_net_calculate()</code> or <code>input_corr()</code>.</p>
</td></tr>
<tr><td><code id="c_net_build_+3A_r_threshold">r_threshold</code></td>
<td>
<p>r_threshold (default: &gt;0.6).</p>
</td></tr>
<tr><td><code id="c_net_build_+3A_p_threshold">p_threshold</code></td>
<td>
<p>p_threshold (default: &lt;0.05).</p>
</td></tr>
<tr><td><code id="c_net_build_+3A_use_p_adj">use_p_adj</code></td>
<td>
<p>use the p.adjust instead of p.value (default: TRUE), if p.adjust not in the corr object, use p.value.</p>
</td></tr>
<tr><td><code id="c_net_build_+3A_delete_single">delete_single</code></td>
<td>
<p>should delete single vertexes?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an metanet object
</p>


<h3>See Also</h3>

<p>Other build: 
<code><a href="#topic+c_net_from_edgelist">c_net_from_edgelist</a>()</code>,
<code><a href="#topic+c_net_set">c_net_set</a>()</code>,
<code><a href="#topic+c_net_update">c_net_update</a>()</code>,
<code><a href="#topic+multi_net_build">multi_net_build</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
t(otutab) -&gt; totu
metadata[, 3:10] -&gt; env
c_net_calculate(totu) -&gt; corr
c_net_build(corr, r_threshold = 0.65) -&gt; co_net

c_net_calculate(totu, env) -&gt; corr2
c_net_build(corr2) -&gt; co_net2
</code></pre>

<hr>
<h2 id='c_net_calculate'>Calculate correlation for one or two t(otutab), or distance for one t(otutab).</h2><span id='topic+c_net_calculate'></span><span id='topic+c_net_cal'></span>

<h3>Description</h3>

<p>Calculate correlation for one or two t(otutab), or distance for one t(otutab).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_calculate(
  totu,
  totu2 = NULL,
  method = "spearman",
  filename = FALSE,
  p.adjust.method = NULL,
  p.adjust.mode = "all",
  threads = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_calculate_+3A_totu">totu</code></td>
<td>
<p>t(otutab), row are samples, column are features.</p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_totu2">totu2</code></td>
<td>
<p>t(otutab2) or NULL, row are samples, column are features.</p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_method">method</code></td>
<td>
<p>&quot;spearman&quot; (default), &quot;pearson&quot;, &quot;sparcc&quot;, or distance index from <code><a href="vegan.html#topic+vegdist">vegdist</a></code>.</p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_filename">filename</code></td>
<td>
<p>the prefix of saved .corr file or FALSE.</p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_p.adjust.mode">p.adjust.mode</code></td>
<td>
<p>see <code><a href="#topic+p.adjust.table">p.adjust.table</a></code></p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_threads">threads</code></td>
<td>
<p>threads, default: 1.</p>
</td></tr>
<tr><td><code id="c_net_calculate_+3A_verbose">verbose</code></td>
<td>
<p>verbose, default: TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a corr object with 3 elements:
</p>
<table>
<tr><td><code>r</code></td>
<td>
<p>default: spearman correlation</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>default: p-value of spearman correlation</p>
</td></tr>
<tr><td><code>p.adjust</code></td>
<td>
<p>default p.adjust.method = NULL</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other calculate: 
<code><a href="#topic+cal_sim">cal_sim</a>()</code>,
<code><a href="#topic+fast_cor">fast_cor</a>()</code>,
<code><a href="#topic+input_corr">input_corr</a>()</code>,
<code><a href="#topic+p.adjust.table">p.adjust.table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
t(otutab) -&gt; totu
c_net_calculate(totu) -&gt; corr
metadata[, 3:10] -&gt; env
c_net_calculate(totu, env) -&gt; corr2
</code></pre>

<hr>
<h2 id='c_net_filter'>Filter a network according to some attributes</h2><span id='topic+c_net_filter'></span>

<h3>Description</h3>

<p>Filter a network according to some attributes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_filter(go, ..., mode = "v")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_filter_+3A_go">go</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="c_net_filter_+3A_...">...</code></td>
<td>
<p>some attributes of vertex and edge</p>
</td></tr>
<tr><td><code id="c_net_filter_+3A_mode">mode</code></td>
<td>
<p>&quot;v&quot; or &quot;e&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("multi_net")
c_net_filter(multi1, v_group %in% c("omic1", "omic2"))
</code></pre>

<hr>
<h2 id='c_net_from_edgelist'>Construct a network from edge_list dataframe</h2><span id='topic+c_net_from_edgelist'></span>

<h3>Description</h3>

<p>Construct a network from edge_list dataframe
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_from_edgelist(
  edgelist,
  vertex = NULL,
  direct = FALSE,
  e_type = NULL,
  e_class = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_from_edgelist_+3A_edgelist">edgelist</code></td>
<td>
<p>first is source, second is target, others are annotation</p>
</td></tr>
<tr><td><code id="c_net_from_edgelist_+3A_vertex">vertex</code></td>
<td>
<p>vertex metadata</p>
</td></tr>
<tr><td><code id="c_net_from_edgelist_+3A_direct">direct</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="c_net_from_edgelist_+3A_e_type">e_type</code></td>
<td>
<p>set e_type</p>
</td></tr>
<tr><td><code id="c_net_from_edgelist_+3A_e_class">e_class</code></td>
<td>
<p>set e_class</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>See Also</h3>

<p>Other build: 
<code><a href="#topic+c_net_build">c_net_build</a>()</code>,
<code><a href="#topic+c_net_set">c_net_set</a>()</code>,
<code><a href="#topic+c_net_update">c_net_update</a>()</code>,
<code><a href="#topic+multi_net_build">multi_net_build</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(edgelist)
edge_net &lt;- c_net_from_edgelist(arc_count, vertex = arc_taxonomy)
edge_net &lt;- c_net_set(edge_net, vertex_class = "Phylum", edge_width = "n")
c_net_plot(edge_net)
</code></pre>

<hr>
<h2 id='c_net_layout'>Layout coordinates</h2><span id='topic+c_net_layout'></span><span id='topic+c_net_lay'></span>

<h3>Description</h3>

<p>Layout coordinates
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_layout(
  go,
  method = igraph::nicely(),
  order_by = NULL,
  order_ls = NULL,
  seed = 1234,
  line_curved = 0.5,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_layout_+3A_go">go</code></td>
<td>
<p>igraph or metanet</p>
</td></tr>
<tr><td><code id="c_net_layout_+3A_method">method</code></td>
<td>
<p>(1) as_line(), as_arc(), as_polygon(), as_polyarc(), as_polycircle(), as_circle_tree();
(2) as_star(), as_tree(), in_circle(), nicely(), on_grid(), on_sphere(),randomly(), with_dh(), with_fr(), with_gem(), with_graphopt(), with_kk(),with_lgl(), with_mds(),. see <code><a href="igraph.html#topic+layout_">layout_</a></code>;
(3) a character, &quot;auto&quot;,&quot;backbone&quot;,&quot;centrality&quot;,&quot;circlepack&quot;,&quot;dendrogram&quot;,
&quot;eigen&quot;,&quot;focus&quot;,&quot;hive&quot;,&quot;igraph&quot;,&quot;linear&quot;,&quot;manual&quot;,&quot;matrix&quot;,
&quot;partition&quot;,&quot;pmds&quot;,&quot;stress&quot;,&quot;treemap&quot;,&quot;unrooted&quot;. see <code><a href="ggraph.html#topic+create_layout">create_layout</a></code></p>
</td></tr>
<tr><td><code id="c_net_layout_+3A_order_by">order_by</code></td>
<td>
<p>order nodes according to a node attribute</p>
</td></tr>
<tr><td><code id="c_net_layout_+3A_order_ls">order_ls</code></td>
<td>
<p>manual the discrete variable with a vector, or continuous variable with &quot;desc&quot; to decreasing</p>
</td></tr>
<tr><td><code id="c_net_layout_+3A_seed">seed</code></td>
<td>
<p>random seed</p>
</td></tr>
<tr><td><code id="c_net_layout_+3A_line_curved">line_curved</code></td>
<td>
<p>consider line curved, only for some layout methods like as_line(), as_polygon().default:0</p>
</td></tr>
<tr><td><code id="c_net_layout_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>coors object: coordinates for nodes, columns: name, X, Y; curved for edges, columns: from, to, curved;
</p>


<h3>See Also</h3>

<p>Other layout: 
<code><a href="#topic+as_arc">as_arc</a>()</code>,
<code><a href="#topic+as_circle_tree">as_circle_tree</a>()</code>,
<code><a href="#topic+as_line">as_line</a>()</code>,
<code><a href="#topic+as_polyarc">as_polyarc</a>()</code>,
<code><a href="#topic+as_polycircle">as_polycircle</a>()</code>,
<code><a href="#topic+as_polygon">as_polygon</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(igraph)
c_net_layout(co_net) -&gt; coors
c_net_plot(co_net, coors)
c_net_plot(co_net, c_net_layout(co_net, in_circle()), vertex.size = 2)
c_net_plot(co_net, c_net_layout(co_net, in_circle(), order_by = "v_class"), vertex.size = 2)
c_net_plot(co_net, c_net_layout(co_net, in_circle(), order_by = "size", order_ls = "desc"))
c_net_plot(co_net, c_net_layout(co_net, as_polygon(3)))
</code></pre>

<hr>
<h2 id='c_net_plot'>Plot a metanet</h2><span id='topic+c_net_plot'></span>

<h3>Description</h3>

<p>Plot a metanet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_plot(
  go,
  coors = NULL,
  ...,
  labels_num = 5,
  vertex_size_range = NULL,
  edge_width_range = NULL,
  plot_module = FALSE,
  mark_module = FALSE,
  mark_color = NULL,
  mark_alpha = 0.3,
  module_label = FALSE,
  module_label_cex = 2,
  module_label_color = "black",
  module_label_just = c(0.5, 0.5),
  legend = TRUE,
  legend_number = FALSE,
  legend_cex = 1,
  legend_position = c(left_leg_x = -2, left_leg_y = 1, right_leg_x = 1.2, right_leg_y =
    1),
  group_legend_title = NULL,
  group_legend_order = NULL,
  color_legend = TRUE,
  color_legend_order = NULL,
  size_legend = FALSE,
  size_legend_title = "Node Size",
  edge_legend = TRUE,
  edge_legend_title = "Edge type",
  edge_legend_order = NULL,
  width_legend = FALSE,
  width_legend_title = "Edge width",
  lty_legend = FALSE,
  lty_legend_title = "Edge class",
  lty_legend_order = NULL,
  seed = 1234
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_plot_+3A_go">go</code></td>
<td>
<p>an igraph or metanet object</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_coors">coors</code></td>
<td>
<p>the coordinates you saved</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_...">...</code></td>
<td>
<p>additional parameters for <code><a href="igraph.html#topic+igraph.plotting">igraph.plotting</a></code></p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_labels_num">labels_num</code></td>
<td>
<p>show how many labels,&gt;1 indicates number, &lt;1 indicates fraction, &quot;all&quot; indicates all, default:5</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_vertex_size_range">vertex_size_range</code></td>
<td>
<p>the vertex size range, e.g. c(1,10)</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_edge_width_range">edge_width_range</code></td>
<td>
<p>the edge width range, e.g. c(1,10)</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_plot_module">plot_module</code></td>
<td>
<p>logical, plot module?</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_mark_module">mark_module</code></td>
<td>
<p>logical, mark the modules?</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_mark_color">mark_color</code></td>
<td>
<p>mark colors</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_mark_alpha">mark_alpha</code></td>
<td>
<p>mark fill alpha, default 0.3</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_module_label">module_label</code></td>
<td>
<p>module_label</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_module_label_cex">module_label_cex</code></td>
<td>
<p>module_label_cex</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_module_label_color">module_label_color</code></td>
<td>
<p>module_label_color</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_module_label_just">module_label_just</code></td>
<td>
<p>module_label_just</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_legend">legend</code></td>
<td>
<p>all legends</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_legend_number">legend_number</code></td>
<td>
<p>legend with numbers</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_legend_cex">legend_cex</code></td>
<td>
<p>character expansion factor relative to current par(&quot;cex&quot;), default: 1</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_legend_position">legend_position</code></td>
<td>
<p>legend_position, default: c(left_leg_x=-1.9,left_leg_y=1,right_leg_x=1.2,right_leg_y=1)</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_group_legend_title">group_legend_title</code></td>
<td>
<p>group_legend_title, length must same to the numbers of v_group</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_group_legend_order">group_legend_order</code></td>
<td>
<p>group_legend_order vector</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_color_legend">color_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_color_legend_order">color_legend_order</code></td>
<td>
<p>color_legend_order vector</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_size_legend">size_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_size_legend_title">size_legend_title</code></td>
<td>
<p>size_legend_title</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_edge_legend">edge_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_edge_legend_title">edge_legend_title</code></td>
<td>
<p>edge_legend_title</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_edge_legend_order">edge_legend_order</code></td>
<td>
<p>edge_legend_order vector, e.g. c(&quot;positive&quot;,&quot;negative&quot;)</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_width_legend">width_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_width_legend_title">width_legend_title</code></td>
<td>
<p>width_legend_title</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_lty_legend">lty_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_lty_legend_title">lty_legend_title</code></td>
<td>
<p>lty_legend_title</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_lty_legend_order">lty_legend_order</code></td>
<td>
<p>lty_legend_order</p>
</td></tr>
<tr><td><code id="c_net_plot_+3A_seed">seed</code></td>
<td>
<p>random seed, default:1234, make sure each plot is the same.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a network plot
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
c_net_plot(co_net)
c_net_plot(co_net2)
c_net_plot(multi1)
</code></pre>

<hr>
<h2 id='c_net_save'>Save network file</h2><span id='topic+c_net_save'></span>

<h3>Description</h3>

<p>Save network file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_save(go, filename = "net", format = "data.frame")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_save_+3A_go">go</code></td>
<td>
<p>metanet network</p>
</td></tr>
<tr><td><code id="c_net_save_+3A_filename">filename</code></td>
<td>
<p>filename</p>
</td></tr>
<tr><td><code id="c_net_save_+3A_format">format</code></td>
<td>
<p>&quot;data.frame&quot;,&quot;graphml&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>

<hr>
<h2 id='c_net_set'>Set basic attributes from totu table</h2><span id='topic+c_net_set'></span>

<h3>Description</h3>

<p>Set basic attributes from totu table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_set(
  go,
  ...,
  vertex_group = "v_group",
  vertex_class = "v_class",
  vertex_size = "size",
  edge_type = "e_type",
  edge_class = "e_class",
  edge_width = "width",
  node_break = 5,
  edge_break = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_set_+3A_go">go</code></td>
<td>
<p>metanet an igraph object</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_...">...</code></td>
<td>
<p>some data.frames to annotate go</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_vertex_group">vertex_group</code></td>
<td>
<p>choose which column to be vertex_group (map to vertex_shape)</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_vertex_class">vertex_class</code></td>
<td>
<p>choose which column to be vertex_class (map to vertex_color)</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_vertex_size">vertex_size</code></td>
<td>
<p>choose which column to be vertex_size (map to vertex_size)</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_edge_type">edge_type</code></td>
<td>
<p>choose which column to be edge_type (map to edge_color)</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_edge_class">edge_class</code></td>
<td>
<p>choose which column to be edge_class (map to edge_linetype)</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_edge_width">edge_width</code></td>
<td>
<p>choose which column to be edge_width (map to edge_width)</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_node_break">node_break</code></td>
<td>
<p>node_break if v_class is numeric, default: 5</p>
</td></tr>
<tr><td><code id="c_net_set_+3A_edge_break">edge_break</code></td>
<td>
<p>edge_break if e_type is numeric, default: 5</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a metanet object
</p>


<h3>See Also</h3>

<p>Other build: 
<code><a href="#topic+c_net_build">c_net_build</a>()</code>,
<code><a href="#topic+c_net_from_edgelist">c_net_from_edgelist</a>()</code>,
<code><a href="#topic+c_net_update">c_net_update</a>()</code>,
<code><a href="#topic+multi_net_build">multi_net_build</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
t(otutab) -&gt; totu
metadata[, 3:10] -&gt; env

data("c_net")
co_net &lt;- c_net_set(co_net, taxonomy, data.frame("Abundance" = colSums(totu)),
  vertex_class = "Phylum", vertex_size = "Abundance"
)
co_net2 &lt;- c_net_set(co_net2, taxonomy, data.frame(name = colnames(env), env = colnames(env)),
  vertex_class = c("Phylum", "env")
)
co_net2 &lt;- c_net_set(co_net2, data.frame("Abundance" = colSums(totu)), vertex_size = "Abundance")
</code></pre>

<hr>
<h2 id='c_net_stability'>Evaluate the stability of a network</h2><span id='topic+c_net_stability'></span><span id='topic+robust_test'></span><span id='topic+vulnerability'></span><span id='topic+robustness'></span>

<h3>Description</h3>

<p style="text-align: center;"><code class="reqn">Vi=\frac{E-Ei}{E}</code>
</p>

<p>E is the global efficiency and Ei is the global efficiency after the removal of the node i and its entire links.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_stability(
  go_ls,
  mode = "robust_test",
  partial = 0.5,
  step = 10,
  reps = 9,
  threads = 1,
  verbose = TRUE,
  keystone = FALSE
)

robust_test(
  go_ls,
  partial = 0.5,
  step = 10,
  reps = 9,
  threads = 1,
  verbose = TRUE
)

vulnerability(go_ls, threads = 1, verbose = TRUE)

robustness(go_ls, keystone = FALSE, reps = 9, threads = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_stability_+3A_go_ls">go_ls</code></td>
<td>
<p>an igraph object or igraph list.</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_mode">mode</code></td>
<td>
<p>&quot;robust_test&quot;, &quot;vulnerability&quot;, &quot;robustness&quot;</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_partial">partial</code></td>
<td>
<p>how much percent vertexes be removed in total (default: 0.5, only for robust_test)</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_step">step</code></td>
<td>
<p>how many nodes be removed each time? (default: 10, only for robust_test)</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_reps">reps</code></td>
<td>
<p>simulation number (default: 9)</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_threads">threads</code></td>
<td>
<p>threads</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="c_net_stability_+3A_keystone">keystone</code></td>
<td>
<p>remove 70%% keystones instead of remove 50%% nodes (default: False, only for robustness)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data.frame
</p>
<p>data.frame (robustness class)
</p>
<p>a vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("c_net")
if (requireNamespace("ggpmisc")) {
  c_net_stability(co_net, mode = "robust_test", step = 20, reps = 9) -&gt; robust_res
  plot(robust_res, index = "Average_degree", mode = 2)
}

c_net_stability(co_net, mode = "vulnerability") -&gt; vulnerability_res
plot(vulnerability_res)

robustness(co_net) -&gt; robustness_res
plot(robustness_res)

module_detect(co_net) -&gt; co_net_modu
zp_analyse(co_net_modu, mode = 2) -&gt; co_net_modu

c_net_stability(co_net_modu, mode = "robustness", keystone = TRUE) -&gt; robustness_res
plot(robustness_res)

</code></pre>

<hr>
<h2 id='c_net_union'>Union two networks</h2><span id='topic+c_net_union'></span>

<h3>Description</h3>

<p>Union two networks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_union(go1, go2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_union_+3A_go1">go1</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="c_net_union_+3A_go2">go2</code></td>
<td>
<p>metanet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
co_net_union &lt;- c_net_union(co_net, co_net2)
c_net_plot(co_net_union)
</code></pre>

<hr>
<h2 id='c_net_update'>Update a metanet object or transform igraph object to metanet object</h2><span id='topic+c_net_update'></span><span id='topic+as.metanet'></span>

<h3>Description</h3>

<p>Update a metanet object or transform igraph object to metanet object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>c_net_update(go, node_break = 5, edge_break = 5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="c_net_update_+3A_go">go</code></td>
<td>
<p>a metanet object or igraph object</p>
</td></tr>
<tr><td><code id="c_net_update_+3A_node_break">node_break</code></td>
<td>
<p>node_break if v_class is numeric, default: 5</p>
</td></tr>
<tr><td><code id="c_net_update_+3A_edge_break">edge_break</code></td>
<td>
<p>edge_break if e_type is numeric, default: 5</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>See Also</h3>

<p>Other build: 
<code><a href="#topic+c_net_build">c_net_build</a>()</code>,
<code><a href="#topic+c_net_from_edgelist">c_net_from_edgelist</a>()</code>,
<code><a href="#topic+c_net_set">c_net_set</a>()</code>,
<code><a href="#topic+multi_net_build">multi_net_build</a>()</code>
</p>

<hr>
<h2 id='cal_sim'>Calculate similarity for one t(otutab)</h2><span id='topic+cal_sim'></span>

<h3>Description</h3>

<p>Calculate similarity for one t(otutab)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cal_sim(totu, method = "bray", norm = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cal_sim_+3A_totu">totu</code></td>
<td>
<p>t(otutab), row are samples, column are features.</p>
</td></tr>
<tr><td><code id="cal_sim_+3A_method">method</code></td>
<td>
<p>Dissimilarity index, see <code><a href="vegan.html#topic+vegdist">vegdist</a></code>.</p>
</td></tr>
<tr><td><code id="cal_sim_+3A_norm">norm</code></td>
<td>
<p>hellinger normalization in features (default: FALSE).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>similarity = 1-distance
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+vegdist">vegdist</a></code>
</p>
<p>Other calculate: 
<code><a href="#topic+c_net_calculate">c_net_calculate</a>()</code>,
<code><a href="#topic+fast_cor">fast_cor</a>()</code>,
<code><a href="#topic+input_corr">input_corr</a>()</code>,
<code><a href="#topic+p.adjust.table">p.adjust.table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("vegan")) {
  data("otutab", package = "pcutils")
  t(otutab) -&gt; totu
  cal_sim(totu) -&gt; sim_corr
}
</code></pre>

<hr>
<h2 id='check_tabs'>Check tables and extract common samples</h2><span id='topic+check_tabs'></span>

<h3>Description</h3>

<p>Check tables and extract common samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_tabs(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="check_tabs_+3A_...">...</code></td>
<td>
<p>tables</p>
</td></tr>
</table>


<h3>Value</h3>

<p>formatted tables
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
check_tabs(otutab)
</code></pre>

<hr>
<h2 id='clean_igraph'>Clean a igraph object</h2><span id='topic+clean_igraph'></span>

<h3>Description</h3>

<p>Clean a igraph object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_igraph(go, direct = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_igraph_+3A_go">go</code></td>
<td>
<p>igraph, metanet objects</p>
</td></tr>
<tr><td><code id="clean_igraph_+3A_direct">direct</code></td>
<td>
<p>direct?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a igraph object
</p>

<hr>
<h2 id='co_net'>MetaNet networks</h2><span id='topic+co_net'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='co_net_rmt'>MetaNet networks</h2><span id='topic+co_net_rmt'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='co_net2'>MetaNet networks</h2><span id='topic+co_net2'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='Cohesion'>Cohesion calculation</h2><span id='topic+Cohesion'></span><span id='topic+plot.cohesion'></span>

<h3>Description</h3>

<p>Cohesion calculation
</p>
<p>Plot cohesion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Cohesion(otutab, reps = 200, threads = 1, mycor = NULL, verbose = TRUE)

## S3 method for class 'cohesion'
plot(x, group, metadata, mode = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Cohesion_+3A_otutab">otutab</code></td>
<td>
<p>otutab</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_reps">reps</code></td>
<td>
<p>iteration time</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_threads">threads</code></td>
<td>
<p>threads</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_mycor">mycor</code></td>
<td>
<p>a correlation matrix you want to use, skip the null model build when mycor is not NULL, default: NULL</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_x">x</code></td>
<td>
<p><code>Cohesion()</code> result (cohesion object)</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_group">group</code></td>
<td>
<p>group name in colnames(metadata)</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_metadata">metadata</code></td>
<td>
<p>metadata</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_mode">mode</code></td>
<td>
<p>plot mode, 1~2</p>
</td></tr>
<tr><td><code id="Cohesion_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="pcutils.html#topic+group_box">group_box</a></code> (mode=1) or <code><a href="pcutils.html#topic+group_box">group_box</a></code> (mode=2)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Cohesion object: a list with two dataframe
</p>
<p>a ggplot
</p>


<h3>References</h3>

<p>Herren, C. M. &amp; McMahon, K. (2017) Cohesion: a method for quantifying the connectivity of microbial communities. doi:10.1038/ismej.2017.91.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("otutab", package = "pcutils")
# set reps at least 99 when you run.
Cohesion(otutab[1:50, ], reps = 19) -&gt; cohesion_res
if (requireNamespace("ggpubr")) {
  plot(cohesion_res, group = "Group", metadata = metadata, mode = 1)
  plot(cohesion_res, group = "Group", metadata = metadata, mode = 2)
}

</code></pre>

<hr>
<h2 id='df2net_tree'>Transform a dataframe to a network edgelist.</h2><span id='topic+df2net_tree'></span>

<h3>Description</h3>

<p>Transform a dataframe to a network edgelist.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>df2net_tree(test, fun = sum)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="df2net_tree_+3A_test">test</code></td>
<td>
<p>df</p>
</td></tr>
<tr><td><code id="df2net_tree_+3A_fun">fun</code></td>
<td>
<p>default: sum</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
cbind(taxonomy, num = rowSums(otutab))[1:20, ] -&gt; test
df2net_tree(test) -&gt; ttt
plot(ttt)
if (requireNamespace("ggraph")) plot(ttt, coors = as_circle_tree())
</code></pre>

<hr>
<h2 id='extract_sub_net'>Extract sub-network from the whole network</h2><span id='topic+extract_sub_net'></span>

<h3>Description</h3>

<p>Extract sub-network from the whole network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>extract_sub_net(
  whole_net,
  otutab,
  threads = 1,
  save_net = FALSE,
  fast = TRUE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="extract_sub_net_+3A_whole_net">whole_net</code></td>
<td>
<p>the whole network</p>
</td></tr>
<tr><td><code id="extract_sub_net_+3A_otutab">otutab</code></td>
<td>
<p>otutab, these columns will be extract</p>
</td></tr>
<tr><td><code id="extract_sub_net_+3A_threads">threads</code></td>
<td>
<p>threads, default: 1</p>
</td></tr>
<tr><td><code id="extract_sub_net_+3A_save_net">save_net</code></td>
<td>
<p>should save these sub_nets? FALSE or a filename</p>
</td></tr>
<tr><td><code id="extract_sub_net_+3A_fast">fast</code></td>
<td>
<p>less indexes for faster calculate ?</p>
</td></tr>
<tr><td><code id="extract_sub_net_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe contains all sub_net parameters
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
extract_sub_net(co_net, otutab) -&gt; sub_net_pars
</code></pre>

<hr>
<h2 id='fast_cor'>Fast correlation calculation</h2><span id='topic+fast_cor'></span>

<h3>Description</h3>

<p>Fast correlation calculation
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fast_cor(totu, totu2 = NULL, method = c("pearson", "spearman"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fast_cor_+3A_totu">totu</code></td>
<td>
<p>t(otutab), row are samples, column are features.</p>
</td></tr>
<tr><td><code id="fast_cor_+3A_totu2">totu2</code></td>
<td>
<p>t(otutab) or NULL, row are samples, column are features.</p>
</td></tr>
<tr><td><code id="fast_cor_+3A_method">method</code></td>
<td>
<p>&quot;spearman&quot; or &quot;pearson&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with 2 elements:
</p>
<table>
<tr><td><code>r</code></td>
<td>
<p>default: spearman correlation</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>default: p-value of spearman correlation</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other calculate: 
<code><a href="#topic+c_net_calculate">c_net_calculate</a>()</code>,
<code><a href="#topic+cal_sim">cal_sim</a>()</code>,
<code><a href="#topic+input_corr">input_corr</a>()</code>,
<code><a href="#topic+p.adjust.table">p.adjust.table</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
t(otutab[1:100, ]) -&gt; totu
fast_cor(totu, method = "spearman") -&gt; corr
</code></pre>

<hr>
<h2 id='filter_n_module'>Filter some modules as others</h2><span id='topic+filter_n_module'></span><span id='topic+combine_n_module'></span><span id='topic+plot_module_tree'></span>

<h3>Description</h3>

<p>Filter some modules as others
</p>
<p>Combine or cut modules to module_number
</p>
<p>Plot module tree
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_n_module(go_m, n_node_in_module = 0, keep_id = NULL, delete = FALSE)

combine_n_module(go_m, module_number = 5)

plot_module_tree(go_m, module = "module", community = NULL, label.size = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_n_module_+3A_go_m">go_m</code></td>
<td>
<p>module metanet</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_n_node_in_module">n_node_in_module</code></td>
<td>
<p>transfer the modules less than n_node_in_module to &quot;others&quot;</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_keep_id">keep_id</code></td>
<td>
<p>keep modules ids, will not be &quot;others&quot;</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_delete">delete</code></td>
<td>
<p>logical, delete others modules? default:FALSE, the others module will be &quot;others&quot;.</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_module_number">module_number</code></td>
<td>
<p>number of modules</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_module">module</code></td>
<td>
<p>which column name is module. default: &quot;module&quot;</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_community">community</code></td>
<td>
<p>community object, default: NULL, use the community of go_m</p>
</td></tr>
<tr><td><code id="filter_n_module_+3A_label.size">label.size</code></td>
<td>
<p>label.size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet with modules
</p>
<p>ggplot
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
module_detect(co_net) -&gt; co_net_modu
filter_n_module(co_net_modu, n_node_in_module = 30) -&gt; co_net_modu
if (requireNamespace("ggtree") &amp;&amp; requireNamespace("treeio")) plot_module_tree(co_net_modu)
combine_n_module(co_net_modu, 20) -&gt; co_net_modu1
if (requireNamespace("ggtree") &amp;&amp; requireNamespace("treeio")) plot_module_tree(co_net_modu1)
</code></pre>

<hr>
<h2 id='fit_power'>Fit power-law distribution for an igraph</h2><span id='topic+fit_power'></span>

<h3>Description</h3>

<p>Fit power-law distribution for an igraph
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fit_power(go, p.value = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fit_power_+3A_go">go</code></td>
<td>
<p>igraph</p>
</td></tr>
<tr><td><code id="fit_power_+3A_p.value">p.value</code></td>
<td>
<p>calculate p.value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>fit_power(co_net)
</code></pre>

<hr>
<h2 id='g_layout'>Layout with group</h2><span id='topic+g_layout'></span><span id='topic+g_layout_nice'></span>

<h3>Description</h3>

<p>Layout with group
</p>
<p>Layout with group nicely
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g_layout(
  go,
  group = "module",
  group_order = NULL,
  layout1 = in_circle(),
  zoom1 = 20,
  layout2 = in_circle(),
  zoom2 = 3,
  show_big_layout = FALSE,
  ...
)

g_layout_nice(go, group = "module", mode = "circlepack", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="g_layout_+3A_go">go</code></td>
<td>
<p>igraph or metanet</p>
</td></tr>
<tr><td><code id="g_layout_+3A_group">group</code></td>
<td>
<p>group name (default: module)</p>
</td></tr>
<tr><td><code id="g_layout_+3A_group_order">group_order</code></td>
<td>
<p>group_order</p>
</td></tr>
<tr><td><code id="g_layout_+3A_layout1">layout1</code></td>
<td>
<p>layout1 method, one of
(1) a dataframe or matrix: rowname is group, two columns are X and Y
(2) function: layout method for <code><a href="#topic+c_net_layout">c_net_layout</a></code> default: in_circle()</p>
</td></tr>
<tr><td><code id="g_layout_+3A_zoom1">zoom1</code></td>
<td>
<p>big network layout size</p>
</td></tr>
<tr><td><code id="g_layout_+3A_layout2">layout2</code></td>
<td>
<p>one of functions: layout method for <code><a href="#topic+c_net_layout">c_net_layout</a></code>, or a list of functions.</p>
</td></tr>
<tr><td><code id="g_layout_+3A_zoom2">zoom2</code></td>
<td>
<p>average sub_network layout size, or numeric vector, or &quot;auto&quot;</p>
</td></tr>
<tr><td><code id="g_layout_+3A_show_big_layout">show_big_layout</code></td>
<td>
<p>show the big layout to help you adjust.</p>
</td></tr>
<tr><td><code id="g_layout_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
<tr><td><code id="g_layout_+3A_mode">mode</code></td>
<td>
<p>circlepack, treemap, backbone, stress</p>
</td></tr>
</table>


<h3>Value</h3>

<p>coors
</p>


<h3>See Also</h3>

<p>Other g_layout: 
<code><a href="#topic+g_layout_polygon">g_layout_polygon</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("c_net")
module_detect(co_net, method = "cluster_fast_greedy") -&gt; co_net_modu
g_layout(co_net_modu, group = "module", zoom1 = 30, zoom2 = "auto", layout2 = as_line()) -&gt; oridata
plot(co_net_modu, coors = oridata)


data("c_net")
module_detect(co_net, method = "cluster_fast_greedy") -&gt; co_net_modu
if (requireNamespace("ggraph")) {
  plot(co_net_modu, coors = g_layout_nice(co_net_modu, group = "module"))
  plot(co_net_modu, coors = g_layout_nice(co_net_modu, group = "module", mode = "treemap"))
}

</code></pre>

<hr>
<h2 id='g_layout_polygon'>Layout with group as a polygon</h2><span id='topic+g_layout_polygon'></span><span id='topic+g_layout_polyarc'></span><span id='topic+g_layout_polycircle'></span>

<h3>Description</h3>

<p>Layout with group as a polygon
</p>
<p>Layout with group as a polyarc
</p>
<p>Layout with group as a polyarc
</p>


<h3>Usage</h3>

<pre><code class='language-R'>g_layout_polygon(
  go,
  group = "v_group",
  group_order = NULL,
  group2 = NULL,
  group2_order = NULL,
  line_curved = 0.5
)

g_layout_polyarc(
  go,
  group = "v_group",
  group_order = NULL,
  group2 = NULL,
  group2_order = NULL,
  space = pi/4,
  scale_node_num = TRUE
)

g_layout_polycircle(
  go,
  group = "v_group",
  group_order = NULL,
  group2 = NULL,
  group2_order = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="g_layout_polygon_+3A_go">go</code></td>
<td>
<p>igraph</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_group">group</code></td>
<td>
<p>group name (default:v_group)</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_group_order">group_order</code></td>
<td>
<p>group_order</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_group2">group2</code></td>
<td>
<p>group2 name, will order nodes in each group according to group2_order</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_group2_order">group2_order</code></td>
<td>
<p>group2_order</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_line_curved">line_curved</code></td>
<td>
<p>line_curved 0~1</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_space">space</code></td>
<td>
<p>the space between each arc, default: pi/4</p>
</td></tr>
<tr><td><code id="g_layout_polygon_+3A_scale_node_num">scale_node_num</code></td>
<td>
<p>scale with the node number in each group</p>
</td></tr>
</table>


<h3>Value</h3>

<p>coors
</p>


<h3>See Also</h3>

<p>Other g_layout: 
<code><a href="#topic+g_layout">g_layout</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g_layout_polygon(multi1) -&gt; oridata
c_net_plot(multi1, oridata)
g_layout_polyarc(multi1, group2 = "v_class", group2_order = c(LETTERS[4:1])) -&gt; oridata
c_net_plot(multi1, oridata)
g_layout_polycircle(co_net2, group2 = "v_class") -&gt; oridata
c_net_plot(co_net2, oridata)
</code></pre>

<hr>
<h2 id='get_community'>Get community</h2><span id='topic+get_community'></span>

<h3>Description</h3>

<p>Get community
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_community(go_m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_community_+3A_go_m">go_m</code></td>
<td>
<p>module metanet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>community
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>

<hr>
<h2 id='get_e'>Get edge information</h2><span id='topic+get_e'></span>

<h3>Description</h3>

<p>Get edge information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_e(go, name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_e_+3A_go">go</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="get_e_+3A_name">name</code></td>
<td>
<p>attribute name, default: NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>

<hr>
<h2 id='get_group_skeleton'>Get skeleton network according to a group</h2><span id='topic+get_group_skeleton'></span><span id='topic+skeleton_plot'></span>

<h3>Description</h3>

<p>Get skeleton network according to a group
</p>
<p>Skeleton plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_group_skeleton(go, Group = "v_class", count = NULL, top_N = 8)

skeleton_plot(ske_net, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_group_skeleton_+3A_go">go</code></td>
<td>
<p>network</p>
</td></tr>
<tr><td><code id="get_group_skeleton_+3A_group">Group</code></td>
<td>
<p>vertex column name</p>
</td></tr>
<tr><td><code id="get_group_skeleton_+3A_count">count</code></td>
<td>
<p>take which column count, default: NULL</p>
</td></tr>
<tr><td><code id="get_group_skeleton_+3A_top_n">top_N</code></td>
<td>
<p>top_N</p>
</td></tr>
<tr><td><code id="get_group_skeleton_+3A_ske_net">ske_net</code></td>
<td>
<p>skeleton</p>
</td></tr>
<tr><td><code id="get_group_skeleton_+3A_...">...</code></td>
<td>
<p>additional parameters for <code><a href="igraph.html#topic+igraph.plotting">igraph.plotting</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>skeleton network
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_group_skeleton(co_net) -&gt; ske_net
skeleton_plot(ske_net)
</code></pre>

<hr>
<h2 id='get_module'>Get module</h2><span id='topic+get_module'></span>

<h3>Description</h3>

<p>Get module
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_module(go_m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_module_+3A_go_m">go_m</code></td>
<td>
<p>module metanet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>module
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>

<hr>
<h2 id='get_module_eigen'>Get module_eigen</h2><span id='topic+get_module_eigen'></span>

<h3>Description</h3>

<p>Get module_eigen
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_module_eigen(go_m)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_module_eigen_+3A_go_m">go_m</code></td>
<td>
<p>module metanet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>module_eigen
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>

<hr>
<h2 id='get_n'>Get network information</h2><span id='topic+get_n'></span>

<h3>Description</h3>

<p>Get network information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_n(go, name = NULL, simple = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_n_+3A_go">go</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="get_n_+3A_name">name</code></td>
<td>
<p>attribute name, default: NULL</p>
</td></tr>
<tr><td><code id="get_n_+3A_simple">simple</code></td>
<td>
<p>logical, get simple index</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>

<hr>
<h2 id='get_v'>Get vertex information</h2><span id='topic+get_v'></span>

<h3>Description</h3>

<p>Get vertex information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_v(go, name = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_v_+3A_go">go</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="get_v_+3A_name">name</code></td>
<td>
<p>attribute name, default: NULL</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+is_metanet">is_metanet</a>()</code>
</p>

<hr>
<h2 id='input_corr'>Import corr from .csv file</h2><span id='topic+input_corr'></span>

<h3>Description</h3>

<p>Import corr from .csv file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_corr(filename)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="input_corr_+3A_filename">filename</code></td>
<td>
<p>filename of .corr</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a corr object
</p>


<h3>See Also</h3>

<p>Other calculate: 
<code><a href="#topic+c_net_calculate">c_net_calculate</a>()</code>,
<code><a href="#topic+cal_sim">cal_sim</a>()</code>,
<code><a href="#topic+fast_cor">fast_cor</a>()</code>,
<code><a href="#topic+p.adjust.table">p.adjust.table</a>()</code>
</p>

<hr>
<h2 id='input_gephi'>Input a graphml file exported by Gephi</h2><span id='topic+input_gephi'></span>

<h3>Description</h3>

<p>Input a graphml file exported by Gephi
</p>


<h3>Usage</h3>

<pre><code class='language-R'>input_gephi(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="input_gephi_+3A_file">file</code></td>
<td>
<p>graphml file exported by Gephi</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list contains the igraph object and coordinates
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>

<hr>
<h2 id='is_metanet'>Is this object a metanet object?</h2><span id='topic+is_metanet'></span><span id='topic+is.metanet'></span>

<h3>Description</h3>

<p>Is this object a metanet object?
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_metanet(go)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_metanet_+3A_go">go</code></td>
<td>
<p>a test object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>logical
</p>


<h3>See Also</h3>

<p>Other manipulate: 
<code><a href="#topic+anno_edge">anno_edge</a>()</code>,
<code><a href="#topic+anno_vertex">anno_vertex</a>()</code>,
<code><a href="#topic+c_net_annotate">c_net_annotate</a>()</code>,
<code><a href="#topic+c_net_filter">c_net_filter</a>()</code>,
<code><a href="#topic+c_net_save">c_net_save</a>()</code>,
<code><a href="#topic+c_net_union">c_net_union</a>()</code>,
<code><a href="#topic+get_e">get_e</a>()</code>,
<code><a href="#topic+get_n">get_n</a>()</code>,
<code><a href="#topic+get_v">get_v</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(c_net)
is_metanet(co_net)
</code></pre>

<hr>
<h2 id='links_stat'>Link summary of the network</h2><span id='topic+links_stat'></span>

<h3>Description</h3>

<p>Link summary of the network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>links_stat(
  go,
  group = "v_class",
  e_type = "all",
  topN = 6,
  colors = NULL,
  legend_number = FALSE,
  legend = TRUE,
  legend_cex = 1,
  legend_position = c(left_leg_x = -1.6, left_leg_y = 1, right_leg_x = 1.2, right_leg_y =
    1),
  col_legend_order = NULL,
  group_legend_title = NULL,
  group_legend_order = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="links_stat_+3A_go">go</code></td>
<td>
<p>igraph or metanet</p>
</td></tr>
<tr><td><code id="links_stat_+3A_group">group</code></td>
<td>
<p>summary which group of vertex attribution in names(vertex_attr(go))</p>
</td></tr>
<tr><td><code id="links_stat_+3A_e_type">e_type</code></td>
<td>
<p>&quot;positive&quot;, &quot;negative&quot;, &quot;all&quot;</p>
</td></tr>
<tr><td><code id="links_stat_+3A_topn">topN</code></td>
<td>
<p>topN of group, default:5</p>
</td></tr>
<tr><td><code id="links_stat_+3A_colors">colors</code></td>
<td>
<p>colors</p>
</td></tr>
<tr><td><code id="links_stat_+3A_legend_number">legend_number</code></td>
<td>
<p>legend with numbers</p>
</td></tr>
<tr><td><code id="links_stat_+3A_legend">legend</code></td>
<td>
<p>all legends</p>
</td></tr>
<tr><td><code id="links_stat_+3A_legend_cex">legend_cex</code></td>
<td>
<p>character expansion factor relative to current par(&quot;cex&quot;), default: 1</p>
</td></tr>
<tr><td><code id="links_stat_+3A_legend_position">legend_position</code></td>
<td>
<p>legend_position, default: c(left_leg_x=-1.9,left_leg_y=1,right_leg_x=1.2,right_leg_y=1)</p>
</td></tr>
<tr><td><code id="links_stat_+3A_col_legend_order">col_legend_order</code></td>
<td>
<p>col_legend_order vector,</p>
</td></tr>
<tr><td><code id="links_stat_+3A_group_legend_title">group_legend_title</code></td>
<td>
<p>group_legend_title, length must same to the numbers of v_group</p>
</td></tr>
<tr><td><code id="links_stat_+3A_group_legend_order">group_legend_order</code></td>
<td>
<p>group_legend_order vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("circlize")) {
  links_stat(co_net, topN = 10)
  module_detect(co_net) -&gt; co_net_modu
  links_stat(co_net_modu, group = "module")
}
</code></pre>

<hr>
<h2 id='metab'>MetaNet networks abundance</h2><span id='topic+metab'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='metab_g'>MetaNet networks metadata</h2><span id='topic+metab_g'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='MetaNet-package'>MetaNet: Network Analysis for Omics Data</h2><span id='topic+MetaNet'></span><span id='topic+MetaNet-package'></span>

<h3>Description</h3>

<p>Comprehensive network analysis package. Calculate correlation network fastly, accelerate lots of analysis by parallel computing. Support for multi-omics data, search sub-nets fluently. Handle bigger data, more than 10,000 nodes in each omics. Offer various layout method for multi-omics network and some interfaces to other software ('Gephi', 'Cytoscape', 'ggplot2'), easy to visualize. Provide comprehensive topology indexes calculation, including ecological network stability.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Chen Peng <a href="mailto:pengchen2001@zju.edu.cn">pengchen2001@zju.edu.cn</a> (<a href="https://orcid.org/0000-0002-9449-7606">ORCID</a>)
</p>


<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://github.com/Asa12138/MetaNet">https://github.com/Asa12138/MetaNet</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/Asa12138/MetaNet/issues">https://github.com/Asa12138/MetaNet/issues</a>
</p>
</li></ul>


<hr>
<h2 id='micro'>MetaNet networks abundance</h2><span id='topic+micro'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='micro_g'>MetaNet networks metadata</h2><span id='topic+micro_g'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='module_detect'>Detect the modules</h2><span id='topic+module_detect'></span><span id='topic+c_net_module'></span>

<h3>Description</h3>

<p>Detect the modules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>module_detect(
  go,
  method = "cluster_fast_greedy",
  n_node_in_module = 0,
  delete = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="module_detect_+3A_go">go</code></td>
<td>
<p>an igraph object</p>
</td></tr>
<tr><td><code id="module_detect_+3A_method">method</code></td>
<td>
<p>cluster_method: &quot;cluster_walktrap&quot;, &quot;cluster_edge_betweenness&quot;, &quot;cluster_fast_greedy&quot;, &quot;cluster_spinglass&quot;</p>
</td></tr>
<tr><td><code id="module_detect_+3A_n_node_in_module">n_node_in_module</code></td>
<td>
<p>transfer the modules less than n_node_in_module to &quot;others&quot;</p>
</td></tr>
<tr><td><code id="module_detect_+3A_delete">delete</code></td>
<td>
<p>logical, delete others modules? default:FALSE, the others module will be &quot;others&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>an igraph object
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
module_detect(co_net) -&gt; co_net_modu
</code></pre>

<hr>
<h2 id='module_eigen'>Calculate the eigenvalue of each module and correlation of nodes and eigenvalue (node_eigen_cor).</h2><span id='topic+module_eigen'></span><span id='topic+module_expression'></span>

<h3>Description</h3>

<p>Calculate the eigenvalue of each module and correlation of nodes and eigenvalue (node_eigen_cor).
</p>
<p>Plot the expression of each modules
</p>


<h3>Usage</h3>

<pre><code class='language-R'>module_eigen(go_m, totu, cor_method = "spearman")

module_expression(
  go_m,
  totu,
  group = NULL,
  r_threshold = 0.6,
  x_order = NULL,
  facet_param = NULL,
  plot_eigen = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="module_eigen_+3A_go_m">go_m</code></td>
<td>
<p>module metanet</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_totu">totu</code></td>
<td>
<p>original abundance table used for module_eigen().</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_cor_method">cor_method</code></td>
<td>
<p>&quot;pearson&quot;, &quot;kendall&quot;, &quot;spearman&quot;</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_group">group</code></td>
<td>
<p>group variable for totu</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_r_threshold">r_threshold</code></td>
<td>
<p>the threshold for node_eigen_cor, default: 0.6.</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_x_order">x_order</code></td>
<td>
<p>order the x axis.</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_facet_param">facet_param</code></td>
<td>
<p>parameters parse to <code><a href="ggplot2.html#topic+facet_wrap">facet_wrap</a></code>, e.g. nrow=2.</p>
</td></tr>
<tr><td><code id="module_eigen_+3A_plot_eigen">plot_eigen</code></td>
<td>
<p>plot the eigen value line.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>module metanet with module_eigen
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("otutab", package = "pcutils")
t(otutab) -&gt; totu
data("c_net")
module_detect(co_net, n_node_in_module = 30) -&gt; co_net_modu
module_eigen(co_net_modu, totu) -&gt; co_net_modu
module_expression(co_net_modu, totu)
</code></pre>

<hr>
<h2 id='module_net'>Generate a n-modules network</h2><span id='topic+module_net'></span>

<h3>Description</h3>

<p>this is just a random generation method, the module number of result is not exactly the module_number, you can change the inter_module_density and intra_module_density to get the proper result.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>module_net(
  module_number = 3,
  n_node_in_module = 30,
  intra_module_density = 0.3,
  inter_module_density = 0.01
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="module_net_+3A_module_number">module_number</code></td>
<td>
<p>number of modules</p>
</td></tr>
<tr><td><code id="module_net_+3A_n_node_in_module">n_node_in_module</code></td>
<td>
<p>number of nodes in each modules</p>
</td></tr>
<tr><td><code id="module_net_+3A_intra_module_density">intra_module_density</code></td>
<td>
<p>intra_module_density, recommend bigger than 20*inter_module_density, default:0.3</p>
</td></tr>
<tr><td><code id="module_net_+3A_inter_module_density">inter_module_density</code></td>
<td>
<p>inter_module_density, default:0.01</p>
</td></tr>
</table>


<h3>Value</h3>

<p>n-modules metanet
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>g1 &lt;- module_net()
get_n(g1)
plot(g1, mark_module = TRUE)
plot(g1, coors = g_layout(g1, zoom2 = 20))
plot(g1, coors = g_layout_polyarc(g1, group = "module"))
plot(g1, coors = g_layout_polygon(g1, group = "module"))
</code></pre>

<hr>
<h2 id='multi_net_build'>Multi-omics network build</h2><span id='topic+multi_net_build'></span>

<h3>Description</h3>

<p>Multi-omics network build
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_net_build(
  ...,
  mode = "full",
  method = "spearman",
  filename = FALSE,
  p.adjust.method = NULL,
  r_threshold = 0.6,
  p_threshold = 0.05,
  use_p_adj = TRUE,
  delete_single = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_net_build_+3A_...">...</code></td>
<td>
<p>some omics abundance tables</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_mode">mode</code></td>
<td>
<p>&quot;full&quot;</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_method">method</code></td>
<td>
<p>&quot;spearman&quot; or &quot;pearson&quot;</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_filename">filename</code></td>
<td>
<p>the prefix of saved .corr file or FALSE</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_p.adjust.method">p.adjust.method</code></td>
<td>
<p>see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code></p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_r_threshold">r_threshold</code></td>
<td>
<p>r_threshold (default: &gt;0.6)</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_p_threshold">p_threshold</code></td>
<td>
<p>p_threshold (default: &lt;0.05)</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_use_p_adj">use_p_adj</code></td>
<td>
<p>use the p.adjust instead of p-value (default: TRUE)</p>
</td></tr>
<tr><td><code id="multi_net_build_+3A_delete_single">delete_single</code></td>
<td>
<p>should delete single vertexes?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>See Also</h3>

<p>Other build: 
<code><a href="#topic+c_net_build">c_net_build</a>()</code>,
<code><a href="#topic+c_net_from_edgelist">c_net_from_edgelist</a>()</code>,
<code><a href="#topic+c_net_set">c_net_set</a>()</code>,
<code><a href="#topic+c_net_update">c_net_update</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("multi_test")
multi1 &lt;- multi_net_build(list(Microbiome = micro, Metabolome = metab, Transcriptome = transc))
multi1 &lt;- c_net_set(multi1, micro_g, metab_g, transc_g,
  vertex_class = c("Phylum", "kingdom", "type")
)
multi1 &lt;- c_net_set(multi1, data.frame("Abundance1" = colSums(micro)),
  data.frame("Abundance2" = colSums(metab)), data.frame("Abundance3" = colSums(transc)),
  vertex_size = paste0("Abundance", 1:3)
)
c_net_plot(multi1)
</code></pre>

<hr>
<h2 id='multi1'>MetaNet networks</h2><span id='topic+multi1'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='nc'>Calculate natural_connectivity</h2><span id='topic+nc'></span>

<h3>Description</h3>

<p>Calculate natural_connectivity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nc(p)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nc_+3A_p">p</code></td>
<td>
<p>an igraph or metanet object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>natural_connectivity (numeric)
</p>


<h3>References</h3>

<p><code>`nc` in `ggClusterNet`</code>
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>igraph::make_ring(10) %&gt;% nc()
</code></pre>

<hr>
<h2 id='net_par'>Calculate all topological indexes of a network</h2><span id='topic+net_par'></span><span id='topic+c_net_index'></span>

<h3>Description</h3>

<p>Calculate all topological indexes of a network
</p>
<p>Add topological indexes for a network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>net_par(go, mode = c("v", "e", "n", "all"), fast = TRUE)

c_net_index(go, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="net_par_+3A_go">go</code></td>
<td>
<p>igraph or metanet</p>
</td></tr>
<tr><td><code id="net_par_+3A_mode">mode</code></td>
<td>
<p>calculate what? c(&quot;v&quot;, &quot;e&quot;, &quot;n&quot;, &quot;all&quot;)</p>
</td></tr>
<tr><td><code id="net_par_+3A_fast">fast</code></td>
<td>
<p>less indexes for faster calculate ?</p>
</td></tr>
<tr><td><code id="net_par_+3A_force">force</code></td>
<td>
<p>replace existed net_par</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a 3-elements list
</p>
<table>
<tr><td><code>n_index</code></td>
<td>
<p>indexs of the whole network</p>
</td></tr>
<tr><td><code>v_index</code></td>
<td>
<p>indexs of each vertex</p>
</td></tr>
<tr><td><code>e_index</code></td>
<td>
<p>indexs of each edge</p>
</td></tr>
</table>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>igraph::make_graph("Walther") %&gt;% net_par()
c_net_index(co_net) -&gt; co_net_with_par
</code></pre>

<hr>
<h2 id='netD3plot'>plot use networkD3</h2><span id='topic+netD3plot'></span>

<h3>Description</h3>

<p>plot use networkD3
</p>


<h3>Usage</h3>

<pre><code class='language-R'>netD3plot(go, v_class = "v_class", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="netD3plot_+3A_go">go</code></td>
<td>
<p>metanet</p>
</td></tr>
<tr><td><code id="netD3plot_+3A_v_class">v_class</code></td>
<td>
<p>which attributes use to be v_class</p>
</td></tr>
<tr><td><code id="netD3plot_+3A_...">...</code></td>
<td>
<p>see <code><a href="networkD3.html#topic+forceNetwork">forceNetwork</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>D3 plot
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
plot(co_net2)
if (requireNamespace("networkD3")) {
  netD3plot(co_net2)
}
</code></pre>

<hr>
<h2 id='olympic_rings_net'>Plot olympic rings using network</h2><span id='topic+olympic_rings_net'></span>

<h3>Description</h3>

<p>Plot olympic rings using network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>olympic_rings_net()
</code></pre>


<h3>Value</h3>

<p>network plot
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>olympic_rings_net()
</code></pre>

<hr>
<h2 id='p.adjust.table'>p.adjust apply on a correlation table (matrix or data.frame)</h2><span id='topic+p.adjust.table'></span>

<h3>Description</h3>

<p>p.adjust apply on a correlation table (matrix or data.frame)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>p.adjust.table(pp, method = "BH", mode = "all")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="p.adjust.table_+3A_pp">pp</code></td>
<td>
<p>table of p-values</p>
</td></tr>
<tr><td><code id="p.adjust.table_+3A_method">method</code></td>
<td>
<p>see <code><a href="stats.html#topic+p.adjust">p.adjust</a></code>, default: &quot;BH&quot;.</p>
</td></tr>
<tr><td><code id="p.adjust.table_+3A_mode">mode</code></td>
<td>
<p>&quot;all&quot; for all values; &quot;rows&quot; adjust each row one by one; &quot;columns&quot; adjust each column one by one. Default: &quot;all&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a table of adjusted p-values
</p>


<h3>See Also</h3>

<p>Other calculate: 
<code><a href="#topic+c_net_calculate">c_net_calculate</a>()</code>,
<code><a href="#topic+cal_sim">cal_sim</a>()</code>,
<code><a href="#topic+fast_cor">fast_cor</a>()</code>,
<code><a href="#topic+input_corr">input_corr</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>matrix(abs(rnorm(100, 0.01, 0.1)), 10, 10) -&gt; pp
p.adjust.table(pp, method = "BH", mode = "all") -&gt; pp_adj
</code></pre>

<hr>
<h2 id='plot.ggig'>Plot a ggig</h2><span id='topic+plot.ggig'></span>

<h3>Description</h3>

<p>Plot a ggig
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggig'
plot(
  x,
  coors = NULL,
  ...,
  labels_num = 5,
  vertex_size_range = NULL,
  edge_width_range = NULL,
  plot_module = FALSE,
  mark_module = FALSE,
  mark_color = NULL,
  mark_alpha = 0.3,
  module_label = FALSE,
  module_label_cex = 2,
  module_label_color = "black",
  module_label_just = c(0.5, 0.5),
  legend_number = FALSE,
  legend = TRUE,
  legend_cex = 1,
  legend_position = c(left_leg_x = -2, left_leg_y = 1, right_leg_x = 1.2, right_leg_y =
    1),
  group_legend_title = NULL,
  group_legend_order = NULL,
  color_legend = TRUE,
  color_legend_order = NULL,
  size_legend = FALSE,
  size_legend_title = "Node Size",
  edge_legend = TRUE,
  edge_legend_title = "Edge type",
  edge_legend_order = NULL,
  width_legend = FALSE,
  width_legend_title = "Edge width",
  lty_legend = FALSE,
  lty_legend_title = "Edge class",
  lty_legend_order = NULL,
  seed = 1234
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.ggig_+3A_x">x</code></td>
<td>
<p>ggig object</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_coors">coors</code></td>
<td>
<p>the coordinates you saved</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_...">...</code></td>
<td>
<p>additional parameters for <code><a href="igraph.html#topic+igraph.plotting">igraph.plotting</a></code></p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_labels_num">labels_num</code></td>
<td>
<p>show how many labels,&gt;1 indicates number, &lt;1 indicates fraction, &quot;all&quot; indicates all, default:5</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_vertex_size_range">vertex_size_range</code></td>
<td>
<p>the vertex size range, e.g. c(1,10)</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_edge_width_range">edge_width_range</code></td>
<td>
<p>the edge width range, e.g. c(1,10)</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_plot_module">plot_module</code></td>
<td>
<p>logical, plot module?</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_mark_module">mark_module</code></td>
<td>
<p>logical, mark the modules?</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_mark_color">mark_color</code></td>
<td>
<p>mark colors</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_mark_alpha">mark_alpha</code></td>
<td>
<p>mark fill alpha, default 0.3</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_module_label">module_label</code></td>
<td>
<p>module_label</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_module_label_cex">module_label_cex</code></td>
<td>
<p>module_label_cex</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_module_label_color">module_label_color</code></td>
<td>
<p>module_label_color</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_module_label_just">module_label_just</code></td>
<td>
<p>module_label_just</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_legend_number">legend_number</code></td>
<td>
<p>legend with numbers</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_legend">legend</code></td>
<td>
<p>all legends</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_legend_cex">legend_cex</code></td>
<td>
<p>character expansion factor relative to current par(&quot;cex&quot;), default: 1</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_legend_position">legend_position</code></td>
<td>
<p>legend_position, default: c(left_leg_x=-1.9,left_leg_y=1,right_leg_x=1.2,right_leg_y=1)</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_group_legend_title">group_legend_title</code></td>
<td>
<p>group_legend_title, length must same to the numbers of v_group</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_group_legend_order">group_legend_order</code></td>
<td>
<p>group_legend_order vector</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_color_legend">color_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_color_legend_order">color_legend_order</code></td>
<td>
<p>color_legend_order vector</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_size_legend">size_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_size_legend_title">size_legend_title</code></td>
<td>
<p>size_legend_title</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_edge_legend">edge_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_edge_legend_title">edge_legend_title</code></td>
<td>
<p>edge_legend_title</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_edge_legend_order">edge_legend_order</code></td>
<td>
<p>edge_legend_order vector, e.g. c(&quot;positive&quot;,&quot;negative&quot;)</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_width_legend">width_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_width_legend_title">width_legend_title</code></td>
<td>
<p>width_legend_title</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_lty_legend">lty_legend</code></td>
<td>
<p>logical</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_lty_legend_title">lty_legend_title</code></td>
<td>
<p>lty_legend_title</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_lty_legend_order">lty_legend_order</code></td>
<td>
<p>lty_legend_order</p>
</td></tr>
<tr><td><code id="plot.ggig_+3A_seed">seed</code></td>
<td>
<p>random seed, default:1234, make sure each plot is the same.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>

<hr>
<h2 id='plot.metanet'>Plot a metanet</h2><span id='topic+plot.metanet'></span>

<h3>Description</h3>

<p>Plot a metanet
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metanet'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.metanet_+3A_x">x</code></td>
<td>
<p>metanet object</p>
</td></tr>
<tr><td><code id="plot.metanet_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>

<hr>
<h2 id='plot.rmt_res'>Plot a rmt_res</h2><span id='topic+plot.rmt_res'></span>

<h3>Description</h3>

<p>Plot a rmt_res
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rmt_res'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.rmt_res_+3A_x">x</code></td>
<td>
<p>rmt_res</p>
</td></tr>
<tr><td><code id="plot.rmt_res_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>

<hr>
<h2 id='plot.robust'>Plot robust</h2><span id='topic+plot.robust'></span>

<h3>Description</h3>

<p>Plot robust
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'robust'
plot(
  x,
  indexes = c("Natural_connectivity", "Average_path_length", "Average_degree"),
  use_ratio = FALSE,
  mode = 1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.robust_+3A_x">x</code></td>
<td>
<p><code>robust_test()</code> result (robust object)</p>
</td></tr>
<tr><td><code id="plot.robust_+3A_indexes">indexes</code></td>
<td>
<p>indexes selected to show</p>
</td></tr>
<tr><td><code id="plot.robust_+3A_use_ratio">use_ratio</code></td>
<td>
<p>use the delete nodes ratio rather than nodes number</p>
</td></tr>
<tr><td><code id="plot.robust_+3A_mode">mode</code></td>
<td>
<p>plot mode, 1~3</p>
</td></tr>
<tr><td><code id="plot.robust_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="pcutils.html#topic+group_box">group_box</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot
</p>

<hr>
<h2 id='plot.robustness'>Plot robustness</h2><span id='topic+plot.robustness'></span>

<h3>Description</h3>

<p>Plot robustness
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'robustness'
plot(x, indexes = "Node_number", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.robustness_+3A_x">x</code></td>
<td>
<p><code>robustness()</code> result (robustness object)</p>
</td></tr>
<tr><td><code id="plot.robustness_+3A_indexes">indexes</code></td>
<td>
<p>indexes selected to show</p>
</td></tr>
<tr><td><code id="plot.robustness_+3A_...">...</code></td>
<td>
<p>additional arguments for <code><a href="pcutils.html#topic+group_box">group_box</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot
</p>

<hr>
<h2 id='plot.vulnerability'>Plot vulnerability</h2><span id='topic+plot.vulnerability'></span>

<h3>Description</h3>

<p>Plot vulnerability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vulnerability'
plot(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot.vulnerability_+3A_x">x</code></td>
<td>
<p><code>vulnerability()</code> result (vulnerability object)</p>
</td></tr>
<tr><td><code id="plot.vulnerability_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a ggplot
</p>

<hr>
<h2 id='print.cohesion'>Print method for 'cohesion' objects</h2><span id='topic+print.cohesion'></span>

<h3>Description</h3>

<p>Print method for 'cohesion' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'cohesion'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.cohesion_+3A_x">x</code></td>
<td>
<p>'cohesion' object</p>
</td></tr>
<tr><td><code id="print.cohesion_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.coors'>Print method for 'coors' objects</h2><span id='topic+print.coors'></span>

<h3>Description</h3>

<p>Print method for 'coors' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'coors'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.coors_+3A_x">x</code></td>
<td>
<p>'coors' object</p>
</td></tr>
<tr><td><code id="print.coors_+3A_...">...</code></td>
<td>
<p>additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.corr'>Print method for 'corr' objects</h2><span id='topic+print.corr'></span>

<h3>Description</h3>

<p>Print method for 'corr' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'corr'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.corr_+3A_x">x</code></td>
<td>
<p>'corr' object</p>
</td></tr>
<tr><td><code id="print.corr_+3A_...">...</code></td>
<td>
<p>additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.ggig'>Print method for 'ggig' objects</h2><span id='topic+print.ggig'></span>

<h3>Description</h3>

<p>Print method for 'ggig' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'ggig'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.ggig_+3A_x">x</code></td>
<td>
<p>'ggig' object</p>
</td></tr>
<tr><td><code id="print.ggig_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.metanet'>Print method for 'metanet' objects</h2><span id='topic+print.metanet'></span>

<h3>Description</h3>

<p>Print method for 'metanet' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metanet'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.metanet_+3A_x">x</code></td>
<td>
<p>'metanet' object</p>
</td></tr>
<tr><td><code id="print.metanet_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.robust'>Print method for 'robust' objects</h2><span id='topic+print.robust'></span>

<h3>Description</h3>

<p>Print method for 'robust' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'robust'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.robust_+3A_x">x</code></td>
<td>
<p>'robust' object</p>
</td></tr>
<tr><td><code id="print.robust_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.robustness'>Print method for 'robustness' objects</h2><span id='topic+print.robustness'></span>

<h3>Description</h3>

<p>Print method for 'robustness' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'robustness'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.robustness_+3A_x">x</code></td>
<td>
<p>'robustness' object</p>
</td></tr>
<tr><td><code id="print.robustness_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='print.vulnerability'>Print method for 'vulnerability' objects</h2><span id='topic+print.vulnerability'></span>

<h3>Description</h3>

<p>Print method for 'vulnerability' objects
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'vulnerability'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.vulnerability_+3A_x">x</code></td>
<td>
<p>'vulnerability' object</p>
</td></tr>
<tr><td><code id="print.vulnerability_+3A_...">...</code></td>
<td>
<p>Additional arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No value
</p>

<hr>
<h2 id='rand_net'>Degree distribution comparison with random network</h2><span id='topic+rand_net'></span>

<h3>Description</h3>

<p>Degree distribution comparison with random network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rand_net(go = go)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rand_net_+3A_go">go</code></td>
<td>
<p>igraph object</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rand_net(co_net)
</code></pre>

<hr>
<h2 id='rand_net_par'>Net_pars of many random network</h2><span id='topic+rand_net_par'></span><span id='topic+compare_rand'></span>

<h3>Description</h3>

<p>Net_pars of many random network
</p>
<p>Compare some indexes between your net with random networks
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rand_net_par(go, reps = 99, threads = 1, verbose = TRUE)

compare_rand(
  pars,
  randp,
  index = c("Average_path_length", "Clustering_coefficent")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rand_net_par_+3A_go">go</code></td>
<td>
<p>igraph</p>
</td></tr>
<tr><td><code id="rand_net_par_+3A_reps">reps</code></td>
<td>
<p>simulation time</p>
</td></tr>
<tr><td><code id="rand_net_par_+3A_threads">threads</code></td>
<td>
<p>threads</p>
</td></tr>
<tr><td><code id="rand_net_par_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
<tr><td><code id="rand_net_par_+3A_pars">pars</code></td>
<td>
<p>your net pars resulted by net_pars()</p>
</td></tr>
<tr><td><code id="rand_net_par_+3A_randp">randp</code></td>
<td>
<p>random networks pars resulted by rand_net_par()</p>
</td></tr>
<tr><td><code id="rand_net_par_+3A_index">index</code></td>
<td>
<p>compared indexes: &quot;Average_path_length&quot;,&quot;Clustering_coefficent&quot; or else</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>,
<code><a href="#topic+smallworldness">smallworldness</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
rand_net_par(co_net_rmt, reps = 30) -&gt; randp
net_par(co_net_rmt, fast = FALSE) -&gt; pars
compare_rand(pars, randp)
</code></pre>

<hr>
<h2 id='RMT_threshold'>Get RMT threshold for a correlation matrix</h2><span id='topic+RMT_threshold'></span><span id='topic+rmt'></span>

<h3>Description</h3>

<p>Get RMT threshold for a correlation matrix
</p>
<p>Get RMT threshold for a correlation matrix roughly
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RMT_threshold(
  occor.r,
  out_dir,
  min_threshold = 0.5,
  max_threshold = 0.8,
  step = 0.02,
  gif = FALSE,
  verbose = FALSE
)

rmt(occor.r, min_threshold = 0.5, max_threshold = 0.85, step = 0.01)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="RMT_threshold_+3A_occor.r">occor.r</code></td>
<td>
<p>a corr object or a correlation matrix</p>
</td></tr>
<tr><td><code id="RMT_threshold_+3A_out_dir">out_dir</code></td>
<td>
<p>output dir</p>
</td></tr>
<tr><td><code id="RMT_threshold_+3A_min_threshold">min_threshold</code></td>
<td>
<p>min_threshold</p>
</td></tr>
<tr><td><code id="RMT_threshold_+3A_max_threshold">max_threshold</code></td>
<td>
<p>max_threshold</p>
</td></tr>
<tr><td><code id="RMT_threshold_+3A_step">step</code></td>
<td>
<p>step</p>
</td></tr>
<tr><td><code id="RMT_threshold_+3A_gif">gif</code></td>
<td>
<p>render a .gif file?</p>
</td></tr>
<tr><td><code id="RMT_threshold_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a r-threshold
</p>
<p>recommend threshold
</p>


<h3>References</h3>

<p>J. Zhou, Y. Deng, FALSE. Luo, Z. He, Q. Tu, X. Zhi, (2010) Functional Molecular Ecological Networks, doi:10.1128/mBio.00169-10.
<a href="https://matstat.org/content_en/RMT/RMThreshold_Intro.pdf">https://matstat.org/content_en/RMT/RMThreshold_Intro.pdf</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(otutab, package = "pcutils")
t(otutab) -&gt; totu
c_net_calculate(totu) -&gt; corr
rmt(corr)
# recommend: 0.69
c_net_build(corr, r_threshold = 0.69) -&gt; co_net_rmt

</code></pre>

<hr>
<h2 id='show_MetaNet_logo'>Show MetaNet logo</h2><span id='topic+show_MetaNet_logo'></span>

<h3>Description</h3>

<p>Show MetaNet logo
</p>


<h3>Usage</h3>

<pre><code class='language-R'>show_MetaNet_logo()
</code></pre>


<h3>Value</h3>

<p>picture
</p>

<hr>
<h2 id='smallworldness'>Calculate small-world coefficient</h2><span id='topic+smallworldness'></span>

<h3>Description</h3>

<p>Calculate small-world coefficient
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smallworldness(go, reps = 99, threads = 1, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smallworldness_+3A_go">go</code></td>
<td>
<p>igraph or metanet</p>
</td></tr>
<tr><td><code id="smallworldness_+3A_reps">reps</code></td>
<td>
<p>simulation time</p>
</td></tr>
<tr><td><code id="smallworldness_+3A_threads">threads</code></td>
<td>
<p>threads</p>
</td></tr>
<tr><td><code id="smallworldness_+3A_verbose">verbose</code></td>
<td>
<p>verbose</p>
</td></tr>
</table>


<h3>Value</h3>

<p>number
</p>


<h3>See Also</h3>

<p>Other topological: 
<code><a href="#topic+extract_sub_net">extract_sub_net</a>()</code>,
<code><a href="#topic+fit_power">fit_power</a>()</code>,
<code><a href="#topic+get_group_skeleton">get_group_skeleton</a>()</code>,
<code><a href="#topic+links_stat">links_stat</a>()</code>,
<code><a href="#topic+nc">nc</a>()</code>,
<code><a href="#topic+net_par">net_par</a>()</code>,
<code><a href="#topic+rand_net_par">rand_net_par</a>()</code>,
<code><a href="#topic+rand_net">rand_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# set reps at least 99 when you run.
smallworldness(co_net, reps = 9)

</code></pre>

<hr>
<h2 id='summ_2col'>Summaries two columns information</h2><span id='topic+summ_2col'></span>

<h3>Description</h3>

<p>Summaries two columns information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summ_2col(df, from = 1, to = 2, count = 3, direct = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summ_2col_+3A_df">df</code></td>
<td>
<p>data.frame</p>
</td></tr>
<tr><td><code id="summ_2col_+3A_from">from</code></td>
<td>
<p>first column name or index</p>
</td></tr>
<tr><td><code id="summ_2col_+3A_to">to</code></td>
<td>
<p>second column name or index</p>
</td></tr>
<tr><td><code id="summ_2col_+3A_count">count</code></td>
<td>
<p>(optional) weight column, if no, each equal to 1</p>
</td></tr>
<tr><td><code id="summ_2col_+3A_direct">direct</code></td>
<td>
<p>consider direct? default: FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame
</p>


<h3>Examples</h3>

<pre><code class='language-R'>test &lt;- data.frame(
  a = sample(letters[1:4], 10, replace = TRUE),
  b = sample(letters[1:4], 10, replace = TRUE)
)
summ_2col(test, direct = TRUE)
summ_2col(test, direct = FALSE)
if (requireNamespace("circlize")) {
  summ_2col(test, direct = TRUE) %&gt;% pcutils::my_circo()
}
</code></pre>

<hr>
<h2 id='summary_module'>Summary module index</h2><span id='topic+summary_module'></span>

<h3>Description</h3>

<p>Summary module index
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_module(go_m, var = "v_class", module = "module", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_module_+3A_go_m">go_m</code></td>
<td>
<p>module metanet</p>
</td></tr>
<tr><td><code id="summary_module_+3A_var">var</code></td>
<td>
<p>variable name</p>
</td></tr>
<tr><td><code id="summary_module_+3A_module">module</code></td>
<td>
<p>which column name is module. default: &quot;module&quot;</p>
</td></tr>
<tr><td><code id="summary_module_+3A_...">...</code></td>
<td>
<p>add</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
module_detect(co_net, n_node_in_module = 30) -&gt; co_net_modu
summary_module(co_net_modu, var = "v_class", module = "module")
summary_module(co_net_modu, var = "Abundance", module = "module")
</code></pre>

<hr>
<h2 id='to_module_net'>Transformation a network to a module network</h2><span id='topic+to_module_net'></span>

<h3>Description</h3>

<p>Transformation a network to a module network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_module_net(go)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="to_module_net_+3A_go">go</code></td>
<td>
<p>metanet</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet with modules
</p>


<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+zp_analyse">zp_analyse</a>()</code>
</p>

<hr>
<h2 id='transc'>MetaNet networks abundance</h2><span id='topic+transc'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='transc_g'>MetaNet networks metadata</h2><span id='topic+transc_g'></span>

<h3>Description</h3>

<p>MetaNet co_nets
</p>

<hr>
<h2 id='twocol_edgelist'>Quick build a metanet from two columns table</h2><span id='topic+twocol_edgelist'></span>

<h3>Description</h3>

<p>Quick build a metanet from two columns table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>twocol_edgelist(edgelist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="twocol_edgelist_+3A_edgelist">edgelist</code></td>
<td>
<p>two columns table (no elements exist in two columns at same time)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>metanet
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+venn_net">venn_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>twocol &lt;- data.frame(
  "col1" = sample(letters, 30, replace = TRUE),
  "col2" = sample(c("A", "B"), 30, replace = TRUE)
)
twocol_net &lt;- twocol_edgelist(twocol)
plot(twocol_net)
c_net_plot(twocol_net, g_layout_polygon(twocol_net))
</code></pre>

<hr>
<h2 id='venn_net'>Venn network</h2><span id='topic+venn_net'></span>

<h3>Description</h3>

<p>Venn network
</p>


<h3>Usage</h3>

<pre><code class='language-R'>venn_net(tab)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="venn_net_+3A_tab">tab</code></td>
<td>
<p>data.frame (row is elements, column is group), or a list (names is group, value is elements)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>plot
</p>


<h3>See Also</h3>

<p>Other plot: 
<code><a href="#topic+as.ggig">as.ggig</a>()</code>,
<code><a href="#topic+c_net_plot">c_net_plot</a>()</code>,
<code><a href="#topic+input_gephi">input_gephi</a>()</code>,
<code><a href="#topic+netD3plot">netD3plot</a>()</code>,
<code><a href="#topic+olympic_rings_net">olympic_rings_net</a>()</code>,
<code><a href="#topic+plot.ggig">plot.ggig</a>()</code>,
<code><a href="#topic+twocol_edgelist">twocol_edgelist</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(otutab, package = "pcutils")
tab &lt;- otutab[400:485, 1:3]
venn_net(tab) -&gt; v_net
plot(v_net)
</code></pre>

<hr>
<h2 id='zp_analyse'>Zi-Pi calculate</h2><span id='topic+zp_analyse'></span><span id='topic+zp_plot'></span>

<h3>Description</h3>

<p>Zi-Pi calculate
</p>
<p>Zi-Pi plot of vertexes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>zp_analyse(go_m, mode = 2, use_origin = TRUE)

zp_plot(go, label = TRUE, mode = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="zp_analyse_+3A_go_m">go_m</code></td>
<td>
<p>igraph object after <code>module_detect()</code></p>
</td></tr>
<tr><td><code id="zp_analyse_+3A_mode">mode</code></td>
<td>
<p>plot style, 1~3</p>
</td></tr>
<tr><td><code id="zp_analyse_+3A_use_origin">use_origin</code></td>
<td>
<p>use original_module, default:TRUE, if FALSE, use module</p>
</td></tr>
<tr><td><code id="zp_analyse_+3A_go">go</code></td>
<td>
<p>igraph object after zp_analyse()</p>
</td></tr>
<tr><td><code id="zp_analyse_+3A_label">label</code></td>
<td>
<p>show label or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>igraph
</p>
<p>a ggplot object
</p>


<h3>References</h3>


<ol>
<li><p> Guimer, R. &amp; Amaral, L. Functional cartography of complex metabolic networks. (2005) doi:10.1038/nature03288.
</p>
</li></ol>



<h3>See Also</h3>

<p>Other module: 
<code><a href="#topic+filter_n_module">filter_n_module</a>()</code>,
<code><a href="#topic+get_community">get_community</a>()</code>,
<code><a href="#topic+get_module_eigen">get_module_eigen</a>()</code>,
<code><a href="#topic+get_module">get_module</a>()</code>,
<code><a href="#topic+module_detect">module_detect</a>()</code>,
<code><a href="#topic+module_eigen">module_eigen</a>()</code>,
<code><a href="#topic+module_net">module_net</a>()</code>,
<code><a href="#topic+summary_module">summary_module</a>()</code>,
<code><a href="#topic+to_module_net">to_module_net</a>()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("c_net")
module_detect(co_net) -&gt; co_net_modu
zp_analyse(co_net_modu) -&gt; co_net_modu
if (requireNamespace("ggrepel")) {
  zp_plot(co_net_modu)
  zp_plot(co_net_modu, mode = 3)
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
