<!DOCTYPE html><html><head><title>Help for package MetaLonDA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MetaLonDA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#areaPermutation'><p>Calculate Area Ratio (AR) of each feature's time interval for all permutations</p></a></li>
<li><a href='#curveFitting'><p>Fit longitudinal data</p></a></li>
<li><a href='#findSigInterval'><p>Find significant time intervals</p></a></li>
<li><a href='#intervalArea'><p>Calculate Area Ratio (AR) of each feature's time interval</p></a></li>
<li><a href='#metalonda'><p>Metagenomic Longitudinal Differential Abundance Analysis for one feature</p></a></li>
<li><a href='#metalonda_test_data'><p>Simulated data of OTU abundance for 2 phenotypes each has 5 subjects at 10 time-points</p></a></li>
<li><a href='#metalondaAll'><p>Metagenomic Longitudinal Differential Abundance Analysis for all Features</p></a></li>
<li><a href='#normalize'><p>Normalize count matrix</p></a></li>
<li><a href='#permutation'><p>Permute group labels</p></a></li>
<li><a href='#visualizeArea'><p>Visualize significant time interval</p></a></li>
<li><a href='#visualizeARHistogram'><p>Visualize Area Ratio (AR) empirical distribution</p></a></li>
<li><a href='#visualizeFeature'><p>Visualize Longitudinal Feature</p></a></li>
<li><a href='#visualizeFeatureSpline'><p>Visualize the feature trajectory with the fitted Splines</p></a></li>
<li><a href='#visualizeTimeIntervals'><p>Visualize all significant time intervals for all tested features</p></a></li>
<li><a href='#visualizeVolcanoPlot'><p>Visualize log2 fold-change and significance of each interval as volcano plot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Metagenomics Longitudinal Differential Abundance Method</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.8</td>
</tr>
<tr>
<td>Date:</td>
<td>2019-12-17</td>
</tr>
<tr>
<td>Author:</td>
<td>Ahmed A. Metwally, Yang Dai, Patricia W. Finn, David L. Perkins</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ahmed A. Metwally &lt;ametwall@stanford.edu&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/aametwally/MetaLonDA">https://github.com/aametwally/MetaLonDA</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/aametwally/MetaLonDA/issues">https://github.com/aametwally/MetaLonDA/issues</a></td>
</tr>
<tr>
<td>Description:</td>
<td>Identify time intervals of differentially abundant metagenomic features in longitudinal studies (Metwally AA, et al., Microbiome, 2018 &lt;<a href="https://doi.org/10.1186%2Fs40168-018-0402-y">doi:10.1186/s40168-018-0402-y</a>&gt;).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Depends:</td>
<td>R(&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>gss, plyr, zoo, pracma, ggplot2, parallel, doParallel,
metagenomeSeq, DESeq2, edgeR</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-12-17 20:11:48 UTC; ahmedmetwally</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-12-18 02:10:26 UTC</td>
</tr>
</table>
<hr>
<h2 id='areaPermutation'>Calculate Area Ratio (AR) of each feature's time interval for all permutations</h2><span id='topic+areaPermutation'></span>

<h3>Description</h3>

<p>Fits longitudinal samples from the same group using negative binomial or LOWESS for all permutations
</p>


<h3>Usage</h3>

<pre><code class='language-R'>areaPermutation(perm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="areaPermutation_+3A_perm">perm</code></td>
<td>
<p>list has all the permutated models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of all permutation area ratio
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metalonda_test_data)
n.sample = 5 # sample size;
n.timepoints = 10 # time point;
n.perm = 3
n.group= 2 # number of group;
Group = factor(c(rep(0,n.sample*n.timepoints), rep(1,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
aggretage.df = data.frame(Count = metalonda_test_data[1,], Time = Time, Group = Group, ID = ID)
perm = permutation(aggretage.df, n.perm = 3, method = "nbinomial", points)
areaPermutation(perm)
</code></pre>

<hr>
<h2 id='curveFitting'>Fit longitudinal data</h2><span id='topic+curveFitting'></span>

<h3>Description</h3>

<p>Fits longitudinal samples from the same group using negative binomial smoothing splines or LOWESS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>curveFitting(df, method = "nbinomial", points)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="curveFitting_+3A_df">df</code></td>
<td>
<p>dataframe has the Count, Group, ID, Time</p>
</td></tr>
<tr><td><code id="curveFitting_+3A_method">method</code></td>
<td>
<p>fitting method (nbinomial, lowess)</p>
</td></tr>
<tr><td><code id="curveFitting_+3A_points">points</code></td>
<td>
<p>points at which the prediction should happen</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the fitted model
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metalonda_test_data)
n.sample = 5 
n.timepoints = 10 
n.group = 2 
Group = factor(c(rep(0, n.sample*n.timepoints), rep(1, n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
aggretage.df = data.frame(Count = metalonda_test_data[1,], Time = Time, Group = Group, ID = ID)
cf = curveFitting(df = aggretage.df, method= "nbinomial", points)
</code></pre>

<hr>
<h2 id='findSigInterval'>Find significant time intervals</h2><span id='topic+findSigInterval'></span>

<h3>Description</h3>

<p>Identify significant time intervals
</p>


<h3>Usage</h3>

<pre><code class='language-R'>findSigInterval(adjusted.pvalue, threshold = 0.05, sign)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="findSigInterval_+3A_adjusted.pvalue">adjusted.pvalue</code></td>
<td>
<p>vector of the adjusted p-value</p>
</td></tr>
<tr><td><code id="findSigInterval_+3A_threshold">threshold</code></td>
<td>
<p>p-value cut off</p>
</td></tr>
<tr><td><code id="findSigInterval_+3A_sign">sign</code></td>
<td>
<p>vector hold area sign of each time interval</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of the start and end points of all significant time intervals
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>p = c(0.04, 0.01, 0.02, 0.04, 0.06, 0.2, 0.06, 0.04)
sign = c(1, 1, 1, 1, -1, -1, 1, 1)
findSigInterval(p, threshold = 0.05, sign)
</code></pre>

<hr>
<h2 id='intervalArea'>Calculate Area Ratio (AR) of each feature's time interval</h2><span id='topic+intervalArea'></span>

<h3>Description</h3>

<p>Calculate Area Ratio (AR) of each feature's time interval
</p>


<h3>Usage</h3>

<pre><code class='language-R'>intervalArea(curve.fit.df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="intervalArea_+3A_curve.fit.df">curve.fit.df</code></td>
<td>
<p>gss data object of the fitted spline</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the area ratio for all time intervals
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metalonda_test_data)
n.sample = 5
n.timepoints = 10
n.group= 2
Group = factor(c(rep(0,n.sample*n.timepoints), rep(1,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
aggregate.df = data.frame(Count = metalonda_test_data[1,], Time = Time, Group = Group, ID = ID)
model = curveFitting(df = aggregate.df, method= "nbinomial", points)
intervalArea(model)
</code></pre>

<hr>
<h2 id='metalonda'>Metagenomic Longitudinal Differential Abundance Analysis for one feature</h2><span id='topic+metalonda'></span>

<h3>Description</h3>

<p>Find significant time intervals of the one feature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metalonda(
  Count,
  Time,
  Group,
  ID,
  n.perm = 500,
  fit.method = "nbinomial",
  points,
  text = 0,
  parall = FALSE,
  pvalue.threshold = 0.05,
  adjust.method = "BH",
  time.unit = "days",
  ylabel = "Normalized Count",
  col = c("blue", "firebrick"),
  prefix = "Test"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metalonda_+3A_count">Count</code></td>
<td>
<p>matrix has the number of reads that mapped to each feature in each sample.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_time">Time</code></td>
<td>
<p>vector of the time label of each sample.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_group">Group</code></td>
<td>
<p>vector of the group label of each sample.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_id">ID</code></td>
<td>
<p>vector of the subject ID label of each sample.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_fit.method">fit.method</code></td>
<td>
<p>fitting method (nbinomial, lowess).</p>
</td></tr>
<tr><td><code id="metalonda_+3A_points">points</code></td>
<td>
<p>points at which the prediction should happen.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_text">text</code></td>
<td>
<p>Feature's name.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_parall">parall</code></td>
<td>
<p>boolean to indicate whether to use multicore.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_pvalue.threshold">pvalue.threshold</code></td>
<td>
<p>p-value threshold cutoff for identifing significant time intervals.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_adjust.method">adjust.method</code></td>
<td>
<p>multiple testing correction method.</p>
</td></tr>
<tr><td><code id="metalonda_+3A_time.unit">time.unit</code></td>
<td>
<p>time unit used in the Time vector (hours, days, weeks, months, etc.)</p>
</td></tr>
<tr><td><code id="metalonda_+3A_ylabel">ylabel</code></td>
<td>
<p>text to be shown on the y-axis of all generated figures (default: &quot;Normalized Count&quot;)</p>
</td></tr>
<tr><td><code id="metalonda_+3A_col">col</code></td>
<td>
<p>two color to be used for the two groups (eg., c(&quot;red&quot;, &quot;blue&quot;)).</p>
</td></tr>
<tr><td><code id="metalonda_+3A_prefix">prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns a list of the significant time intervals for the tested feature.
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metalonda_test_data)
n.sample = 5
n.timepoints = 10
n.group = 2
Group = factor(c(rep(0, n.sample*n.timepoints), rep(1,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
## Not run: 
output.nbinomial = metalonda(Count = metalonda_test_data[1,], Time = Time, Group = Group,
ID = ID, fit.method =  "nbinomial", n.perm = 10, points = points,
text = rownames(metalonda_test_data)[1], parall = FALSE, pvalue.threshold = 0.05, 
adjust.method = "BH", time.unit = "hours", ylabel = "Normalized Count", col = c("black", "green"))

## End(Not run)
</code></pre>

<hr>
<h2 id='metalonda_test_data'>Simulated data of OTU abundance for 2 phenotypes each has 5 subjects at 10 time-points</h2><span id='topic+metalonda_test_data'></span>

<h3>Description</h3>

<p>The dataset is used for testing the MetaLonDA
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metalonda_test_data
</code></pre>


<h3>Format</h3>

<p>A data frame with 2 OTUs patterns</p>

<hr>
<h2 id='metalondaAll'>Metagenomic Longitudinal Differential Abundance Analysis for all Features</h2><span id='topic+metalondaAll'></span>

<h3>Description</h3>

<p>Identify significant features and their significant time interval
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metalondaAll(
  Count,
  Time,
  Group,
  ID,
  n.perm = 500,
  fit.method = "nbinomial",
  num.intervals = 100,
  parall = FALSE,
  pvalue.threshold = 0.05,
  adjust.method = "BH",
  time.unit = "days",
  norm.method = "none",
  prefix = "Output",
  ylabel = "Normalized Count",
  col = c("blue", "firebrick")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metalondaAll_+3A_count">Count</code></td>
<td>
<p>Count matrix of all features</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_time">Time</code></td>
<td>
<p>Time label of all samples</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_group">Group</code></td>
<td>
<p>Group label of all samples</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_id">ID</code></td>
<td>
<p>individual ID label for samples</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_fit.method">fit.method</code></td>
<td>
<p>The fitting method (nbinomial, lowess)</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_num.intervals">num.intervals</code></td>
<td>
<p>The number of time intervals at which metalonda test differential abundance</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_parall">parall</code></td>
<td>
<p>logic to indicate whether to use multicore</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_pvalue.threshold">pvalue.threshold</code></td>
<td>
<p>p-value threshold cutoff</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_adjust.method">adjust.method</code></td>
<td>
<p>Multiple testing correction methods</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_time.unit">time.unit</code></td>
<td>
<p>time unit used in the Time vector (hours, days, weeks, months, etc.)</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_norm.method">norm.method</code></td>
<td>
<p>normalization method to be used to normalize count matrix (css, tmm, ra, log10, median_ratio)</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_prefix">prefix</code></td>
<td>
<p>prefix for the output figure</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_ylabel">ylabel</code></td>
<td>
<p>text to be shown on the y-axis of all generated figures (default: &quot;Normalized Count&quot;)</p>
</td></tr>
<tr><td><code id="metalondaAll_+3A_col">col</code></td>
<td>
<p>two color to be used for the two groups (eg., c(&quot;red&quot;, &quot;blue&quot;)).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list of the significant features a long with their significant time intervals
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(metalonda_test_data)
n.sample = 5
n.timepoints = 10
n.group = 2
Group = factor(c(rep(0, n.sample*n.timepoints), rep(1,n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
output.nbinomial = metalondaAll(Count = metalonda_test_data, Time = Time, Group = Group,
ID = ID, n.perm = 10, fit.method =  "nbinomial", num.intervals = 100, 
parall = FALSE, pvalue.threshold = 0.05, adjust.method = "BH", 
time.unit = "hours", norm.method = "none", prefix = "Test",  time.unit = "hours", 
ylabel = "Normalized Count", col = c("black", "green"))

## End(Not run)
</code></pre>

<hr>
<h2 id='normalize'>Normalize count matrix</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>Normalize count matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(count, method = "css")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_+3A_count">count</code></td>
<td>
<p>count matrix</p>
</td></tr>
<tr><td><code id="normalize_+3A_method">method</code></td>
<td>
<p>normalization method</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>

<hr>
<h2 id='permutation'>Permute group labels</h2><span id='topic+permutation'></span>

<h3>Description</h3>

<p>Permutes the group label of the samples in order to construct the AR empirical distibution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>permutation(
  perm.dat,
  n.perm = 500,
  method = "nbinomial",
  points,
  lev,
  parall = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="permutation_+3A_perm.dat">perm.dat</code></td>
<td>
<p>dataframe has the Count, Group, ID, Time</p>
</td></tr>
<tr><td><code id="permutation_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations</p>
</td></tr>
<tr><td><code id="permutation_+3A_method">method</code></td>
<td>
<p>The fitting method (negative binomial, LOWESS)</p>
</td></tr>
<tr><td><code id="permutation_+3A_points">points</code></td>
<td>
<p>The points at which the prediction should happen</p>
</td></tr>
<tr><td><code id="permutation_+3A_lev">lev</code></td>
<td>
<p>the two level's name</p>
</td></tr>
<tr><td><code id="permutation_+3A_parall">parall</code></td>
<td>
<p>boolean to indicate whether to use multicore.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>returns the fitted model for all the permutations
</p>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metalonda_test_data)
n.sample = 5
n.timepoints = 10
n.perm = 3
n.group = 2
Group = factor(c(rep(0, n.sample*n.timepoints), rep(1, n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
aggregate.df = data.frame(Count = metalonda_test_data[1,], Time = Time, Group = Group, ID = ID)
prm = permutation(aggregate.df, n.perm = 3, method = "nbinomial", points)
</code></pre>

<hr>
<h2 id='visualizeArea'>Visualize significant time interval</h2><span id='topic+visualizeArea'></span>

<h3>Description</h3>

<p>Visualize significant time interval
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeArea(
  aggregate.df,
  model.ss,
  method,
  start,
  end,
  text,
  group.levels,
  unit = "days",
  ylabel = "Normalized Count",
  col = c("blue", "firebrick"),
  prefix = "Test"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeArea_+3A_aggregate.df">aggregate.df</code></td>
<td>
<p>Dataframe has the Count, Group, ID, Time</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_model.ss">model.ss</code></td>
<td>
<p>The fitted model</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_method">method</code></td>
<td>
<p>Fitting method (negative binomial or LOWESS)</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_start">start</code></td>
<td>
<p>Vector of the start points of the time intervals</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_end">end</code></td>
<td>
<p>Vector of the end points of the time intervals</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_text">text</code></td>
<td>
<p>Feature name</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_group.levels">group.levels</code></td>
<td>
<p>Level's name</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_unit">unit</code></td>
<td>
<p>time unit used in the Time vector (hours, days, weeks, months, etc.)</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_ylabel">ylabel</code></td>
<td>
<p>text to be shown on the y-axis of all generated figures (default: &quot;Normalized Count&quot;)</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_col">col</code></td>
<td>
<p>two color to be used for the two groups (eg., c(&quot;red&quot;, &quot;blue&quot;)).</p>
</td></tr>
<tr><td><code id="visualizeArea_+3A_prefix">prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>

<hr>
<h2 id='visualizeARHistogram'>Visualize Area Ratio (AR) empirical distribution</h2><span id='topic+visualizeARHistogram'></span>

<h3>Description</h3>

<p>Visualize Area Ratio (AR) empirical distribution for each time interval
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeARHistogram(permuted, text, method, prefix = "Test")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeARHistogram_+3A_permuted">permuted</code></td>
<td>
<p>Permutation of the permuted data</p>
</td></tr>
<tr><td><code id="visualizeARHistogram_+3A_text">text</code></td>
<td>
<p>Feature name</p>
</td></tr>
<tr><td><code id="visualizeARHistogram_+3A_method">method</code></td>
<td>
<p>fitting method</p>
</td></tr>
<tr><td><code id="visualizeARHistogram_+3A_prefix">prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>

<hr>
<h2 id='visualizeFeature'>Visualize Longitudinal Feature</h2><span id='topic+visualizeFeature'></span>

<h3>Description</h3>

<p>Visualize Longitudinal Feature
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeFeature(
  df,
  text,
  group.levels,
  unit = "days",
  ylabel = "Normalized Count",
  col = c("blue", "firebrick"),
  prefix = "Test"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeFeature_+3A_df">df</code></td>
<td>
<p>dataframe has the Count, Group, ID, Time</p>
</td></tr>
<tr><td><code id="visualizeFeature_+3A_text">text</code></td>
<td>
<p>feature name</p>
</td></tr>
<tr><td><code id="visualizeFeature_+3A_group.levels">group.levels</code></td>
<td>
<p>The two level's name</p>
</td></tr>
<tr><td><code id="visualizeFeature_+3A_unit">unit</code></td>
<td>
<p>time interval unit</p>
</td></tr>
<tr><td><code id="visualizeFeature_+3A_ylabel">ylabel</code></td>
<td>
<p>text to be shown on the y-axis of all generated figures (default: &quot;Normalized Count&quot;)</p>
</td></tr>
<tr><td><code id="visualizeFeature_+3A_col">col</code></td>
<td>
<p>two color to be used for the two groups (eg., c(&quot;red&quot;, &quot;blue&quot;)).</p>
</td></tr>
<tr><td><code id="visualizeFeature_+3A_prefix">prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(metalonda_test_data)
pfx = tempfile()
dir.create(file.path(pfx))
n.sample = 5
n.timepoints = 10
n.group = 2
Group = factor(c(rep(0, n.sample*n.timepoints), rep(1, n.sample*n.timepoints)))
Time = rep(rep(1:n.timepoints, times = n.sample), 2)
ID = factor(rep(1:(2*n.sample), each = n.timepoints))
points = seq(1, 10, length.out = 10)
aggregate.df = data.frame(Count = metalonda_test_data[1,], Time = Time, Group = Group, ID = ID)
visualizeFeature(df = aggregate.df, text = rownames(metalonda_test_data)[1], 
group.levels = Group, prefix = pfx)
</code></pre>

<hr>
<h2 id='visualizeFeatureSpline'>Visualize the feature trajectory with the fitted Splines</h2><span id='topic+visualizeFeatureSpline'></span>

<h3>Description</h3>

<p>Plot the longitudinal features along with the fitted splines
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeFeatureSpline(
  df,
  model,
  method,
  text,
  group.levels,
  unit = "days",
  ylabel = "Normalized Count",
  col = c("blue", "firebrick"),
  prefix = "Test"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeFeatureSpline_+3A_df">df</code></td>
<td>
<p>dataframe has the Count , Group, ID, Time</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_model">model</code></td>
<td>
<p>the fitted model</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_method">method</code></td>
<td>
<p>The fitting method (negative binomial, LOWESS)</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_text">text</code></td>
<td>
<p>feature name</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_group.levels">group.levels</code></td>
<td>
<p>The two level's name</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_unit">unit</code></td>
<td>
<p>time unit used in the Time vector (hours, days, weeks, months, etc.)</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_ylabel">ylabel</code></td>
<td>
<p>text to be shown on the y-axis of all generated figures (default: &quot;Normalized Count&quot;)</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_col">col</code></td>
<td>
<p>two color to be used for the two groups (eg., c(&quot;red&quot;, &quot;blue&quot;)).</p>
</td></tr>
<tr><td><code id="visualizeFeatureSpline_+3A_prefix">prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>

<hr>
<h2 id='visualizeTimeIntervals'>Visualize all significant time intervals for all tested features</h2><span id='topic+visualizeTimeIntervals'></span>

<h3>Description</h3>

<p>Visualize all significant time intervals for all tested features
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeTimeIntervals(
  interval.details,
  prefix = "Test",
  unit = "days",
  col = c("blue", "firebrick")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeTimeIntervals_+3A_interval.details">interval.details</code></td>
<td>
<p>Dataframe has infomation about significant interval (feature name, start, end, dominant, p-value)</p>
</td></tr>
<tr><td><code id="visualizeTimeIntervals_+3A_prefix">prefix</code></td>
<td>
<p>prefix for the output figure</p>
</td></tr>
<tr><td><code id="visualizeTimeIntervals_+3A_unit">unit</code></td>
<td>
<p>time unit used in the Time vector (hours, days, weeks, months, etc.)</p>
</td></tr>
<tr><td><code id="visualizeTimeIntervals_+3A_col">col</code></td>
<td>
<p>two color to be used for the two groups (eg., c(&quot;red&quot;, &quot;blue&quot;)).</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>

<hr>
<h2 id='visualizeVolcanoPlot'>Visualize log2 fold-change and significance of each interval as volcano plot</h2><span id='topic+visualizeVolcanoPlot'></span>

<h3>Description</h3>

<p>Visualize log2 fold-change and significance of each interval as volcano plot
</p>


<h3>Usage</h3>

<pre><code class='language-R'>visualizeVolcanoPlot(df, text, prefix = "Test")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="visualizeVolcanoPlot_+3A_df">df</code></td>
<td>
<p>Dataframe has a detailed summary about feature's significant intervals</p>
</td></tr>
<tr><td><code id="visualizeVolcanoPlot_+3A_text">text</code></td>
<td>
<p>Feature name</p>
</td></tr>
<tr><td><code id="visualizeVolcanoPlot_+3A_prefix">prefix</code></td>
<td>
<p>prefix to be used to create directory for the analysis results</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ahmed Metwally (ametwall@stanford.edu)
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
