<!DOCTYPE html><html><head><title>Help for package meta</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {meta}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#amlodipine'><p>Amlodipine for Work Capacity</p></a></li>
<li><a href='#as.data.frame.meta'><p>Additional functions for objects of class meta</p></a></li>
<li><a href='#barplot.rob'><p>Produce weighted bar plot of risk of bias assessment</p></a></li>
<li><a href='#baujat.meta'><p>Baujat plot to explore heterogeneity in meta-analysis</p></a></li>
<li><a href='#blup.meta'><p>Calculate best linear unbiased predictor for <code>meta</code> object</p></a></li>
<li><a href='#bubble.metareg'><p>Bubble plot to display the result of a meta-regression</p></a></li>
<li><a href='#caffeine'><p>Caffeine for daytime drowsiness</p></a></li>
<li><a href='#ci'><p>Calculation of confidence intervals (based on normal approximation</p>
or t-distribution)</a></li>
<li><a href='#cisapride'><p>Cisapride in Non-Ulcer Dispepsia</p></a></li>
<li><a href='#drapery'><p>Drapery plot</p></a></li>
<li><a href='#estimates.meta'><p>Extract results from meta-analysis object</p></a></li>
<li><a href='#Fleiss1993bin'><p>Aspirin after Myocardial Infarction</p></a></li>
<li><a href='#Fleiss1993cont'><p>Mental Health Treatment</p></a></li>
<li><a href='#forest.meta'><p>Forest plot to display the result of a meta-analysis</p></a></li>
<li><a href='#forest.metabind'><p>Forest plot to display the result of a meta-analysis</p></a></li>
<li><a href='#funnel.meta'><p>Funnel plot</p></a></li>
<li><a href='#gs'><p>Get default for a meta-analysis setting.</p></a></li>
<li><a href='#JAMAlabels'><p>Create study labels in JAMA layout (deprecated function)</p></a></li>
<li><a href='#labbe.metabin'><p>L'Abbé plot for meta-analysis with binary outcomes</p></a></li>
<li><a href='#labels.meta'><p>Create study labels for forest plot</p></a></li>
<li><a href='#longarm'><p>Transform data from pairwise comparisons to long arm-based format</p></a></li>
<li><a href='#meta-object'><p>Description of R object of class &quot;meta&quot;</p></a></li>
<li><a href='#meta-package'><p>meta: Brief overview of methods and general hints</p></a></li>
<li><a href='#meta-sm'><p>Description of summary measures available in R package <b>meta</b></p></a></li>
<li><a href='#meta-transf'><p>Auxiliary functions for (back) transformations</p></a></li>
<li><a href='#metaadd'><p>Add pooled results from external analysis to meta-analysis</p></a></li>
<li><a href='#metabias.meta'><p>Test for funnel plot asymmetry</p></a></li>
<li><a href='#metabias.rm5'><p>Cochrane review: Test for funnel plot asymmetry</p></a></li>
<li><a href='#metabin'><p>Meta-analysis of binary outcome data</p></a></li>
<li><a href='#metabind'><p>Combine and summarize meta-analysis objects</p></a></li>
<li><a href='#metacont'><p>Meta-analysis of continuous outcome data</p></a></li>
<li><a href='#metacor'><p>Meta-analysis of correlations</p></a></li>
<li><a href='#metacr'><p>Meta-analysis of outcome data from Cochrane review</p></a></li>
<li><a href='#metacum.meta'><p>Cumulative meta-analysis</p></a></li>
<li><a href='#metagen'><p>Generic inverse variance meta-analysis</p></a></li>
<li><a href='#metainc'><p>Meta-analysis of incidence rates</p></a></li>
<li><a href='#metainf.meta'><p>Influence analysis in meta-analysis using leave-one-out method</p></a></li>
<li><a href='#metamean'><p>Meta-analysis of single means</p></a></li>
<li><a href='#metamerge'><p>Merge results of two meta-analyses on the same data set</p></a></li>
<li><a href='#metaprop'><p>Meta-analysis of single proportions</p></a></li>
<li><a href='#metarate'><p>Meta-analysis of single incidence rates</p></a></li>
<li><a href='#metareg.meta'><p>Meta-regression</p></a></li>
<li><a href='#nnt'><p>Calculate the number needed to treat (NNT)</p></a></li>
<li><a href='#Olkin1995'><p>Thrombolytic Therapy after Acute Myocardial Infarction</p></a></li>
<li><a href='#or2smd'><p>Conversion from log odds ratio to standardised mean difference</p></a></li>
<li><a href='#Pagliaro1992'><p>Meta-analysis on Prevention of First Bleeding in Cirrhosis</p></a></li>
<li><a href='#print.meta'><p>Print meta-analysis results</p></a></li>
<li><a href='#print.rm5'><p>Cochrane review: summary of meta-analyses</p></a></li>
<li><a href='#print.summary.meta'><p>Print detailed meta-analysis results</p></a></li>
<li><a href='#radial.meta'><p>Radial plot</p></a></li>
<li><a href='#read.cdir'><p>Import data of Cochrane intervention review</p></a></li>
<li><a href='#read.mtv'><p>Import RevMan 4 data files (.mtv)</p></a></li>
<li><a href='#read.rm5'><p>Import RevMan 5 analysis data</p></a></li>
<li><a href='#rob'><p>Risk of bias assessment</p></a></li>
<li><a href='#settings.meta'><p>Print and change default settings to conduct and print or plot</p>
meta-analyses in R package <b>meta</b>.</a></li>
<li><a href='#smd2or'><p>Conversion from standardised mean difference to log odds ratio</p></a></li>
<li><a href='#smoking'><p>Smoking example</p></a></li>
<li><a href='#subset.longarm'><p>Return subset of longarm object</p></a></li>
<li><a href='#summary.meta'><p>Summary of meta-analysis results</p></a></li>
<li><a href='#summary.rm5'><p>Cochrane review: detailed summary of meta-analyses</p></a></li>
<li><a href='#traffic_light'><p>Produce traffic light plot of risk of bias assessment</p></a></li>
<li><a href='#trimfill.meta'><p>Trim-and-fill method to adjust for bias in meta-analysis</p></a></li>
<li><a href='#trimfill.rm5'><p>Cochrane review: trim-and-fill method</p></a></li>
<li><a href='#update.meta'><p>Update a meta-analysis object</p></a></li>
<li><a href='#weights.meta'><p>Calculate absolute and percentage weights for meta-analysis</p></a></li>
<li><a href='#woodyplants'><p>Elevated CO_2 and total biomass of woody plants</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>General Package for Meta-Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>7.0-0</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-01-11</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0), metadat</td>
</tr>
<tr>
<td>Imports:</td>
<td>metafor (&ge; 3.0-0), grid, lme4, CompQuadForm, xml2, dplyr,
methods, readr, magrittr, purrr, stringr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>BiasedUrn, pimeta, estmeansd, writexl, rmarkdown, knitr,
robvis</td>
</tr>
<tr>
<td>Author:</td>
<td>Guido Schwarzer <a href="https://orcid.org/0000-0001-6214-9087"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [cre, aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Guido Schwarzer &lt;guido.schwarzer@uniklinik-freiburg.de&gt;</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/guido-s/meta/">https://github.com/guido-s/meta/</a>
<a href="https://link.springer.com/book/10.1007/978-3-319-21416-0">https://link.springer.com/book/10.1007/978-3-319-21416-0</a></td>
</tr>
<tr>
<td>Description:</td>
<td>User-friendly general package providing standard methods for meta-analysis and supporting Schwarzer, Carpenter, and Rücker &lt;<a href="https://doi.org/10.1007%2F978-3-319-21416-0">doi:10.1007/978-3-319-21416-0</a>&gt;, "Meta-Analysis with R" (2015):
 - common effect and random effects meta-analysis;
 - several plots (forest, funnel, Galbraith / radial, L'Abbe, Baujat, bubble);
 - three-level meta-analysis model;
 - generalised linear mixed model;
 - Hartung-Knapp method for random effects model;
 - Kenward-Roger method for random effects model;
 - prediction interval;
 - statistical tests  for funnel plot asymmetry;
 - trim-and-fill method to evaluate bias in meta-analysis;
 - meta-regression;
 - cumulative meta-analysis and leave-one-out meta-analysis;
 - import data from 'RevMan 5';
 - produce forest plot summarising several (subgroup) meta-analyses.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-11 20:20:43 UTC; sc</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-12 13:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='amlodipine'>Amlodipine for Work Capacity</h2><span id='topic+amlodipine'></span>

<h3>Description</h3>

<p>Meta-analysis on the effect of amlodipine on work capacity.
</p>
<p>This meta-analysis is used as a data example in Hartung and Knapp (2001).
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>study</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.amlo</em></b></td><td style="text-align: left;"> number of observations in amlodipine group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean.amlo</em></b></td><td style="text-align: left;"> estimated mean in amlodipine group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>var.amlo</em></b></td><td style="text-align: left;"> variance in amlodipine group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.plac</em></b></td><td style="text-align: left;"> number of observations in placebo group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean.plac</em></b></td><td style="text-align: left;"> estimated mean in placebo group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>var.plac</em></b></td><td style="text-align: left;"> variance in placebo group
</td>
</tr>

</table>



<h3>Source</h3>

<p>Hartung J &amp; Knapp G (2001):
On tests of the overall treatment effect in meta-analysis with
normally distributed responses.
<em>Statistics in Medicine</em>,
<b>20</b>, 1771&ndash;82
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metacont">metacont</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(amlodipine)

m &lt;- metacont(n.amlo, mean.amlo, sqrt(var.amlo),
  n.plac, mean.plac, sqrt(var.plac),
  data = amlodipine, studlab = study,
  method.tau = "DL")
m.hk &lt;- update(m, method.random.ci = "HK")

# Same results for mean difference as in Table III in Hartung and
# Knapp (2001)
#
vars.common &lt;- c("TE.common", "lower.common", "upper.common")
vars.random &lt;- c("TE.random", "lower.random", "upper.random")
#
res.common &lt;- as.data.frame(m[vars.common])
names(res.common) &lt;- vars.random
#
res.md &lt;- rbind(res.common,
  as.data.frame(m[vars.random]),
  as.data.frame(m.hk[vars.random]))
#
res.md &lt;- round(res.md, 5)
#
row.names(res.md) &lt;- c("CE", "RE", "RE (HaKn)")
names(res.md) &lt;- c("Absolute difference", "CI lower", "CI upper")
#
res.md
</code></pre>

<hr>
<h2 id='as.data.frame.meta'>Additional functions for objects of class meta</h2><span id='topic+as.data.frame.meta'></span>

<h3>Description</h3>

<p>The <code>as.data.frame</code> method returns a data frame containing
information on individual studies, e.g., estimated treatment effect
and its standard error.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
as.data.frame(x, row.names = NULL, optional = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as.data.frame.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="as.data.frame.meta_+3A_row.names">row.names</code></td>
<td>
<p><code>NULL</code> or a character vector giving the row
names for the data frame.</p>
</td></tr>
<tr><td><code id="as.data.frame.meta_+3A_optional">optional</code></td>
<td>
<p>logical. If <code>TRUE</code>, setting row names and
converting column names (to syntactic names) is optional.</p>
</td></tr>
<tr><td><code id="as.data.frame.meta_+3A_...">...</code></td>
<td>
<p>other arguments</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame is returned by the function <code>as.data.frame</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+forest.meta">forest.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
#
# Generate additional variable with grouping information
#
Fleiss1993cont$group &lt;- c(1, 2, 1, 1, 2)
#
# Do meta-analysis without grouping information
#
m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD", studlab = paste(study, year))
#
# Update meta-analysis object and do subgroup analyses
#
update(m1, subgroup = group)

# Same result using metacont function directly
#
m2 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD", studlab = paste(study, year),
  subgroup = group)
m2

# Compare printout of the following two commands
#
as.data.frame(m1)
m1$data

</code></pre>

<hr>
<h2 id='barplot.rob'>Produce weighted bar plot of risk of bias assessment</h2><span id='topic+barplot.rob'></span>

<h3>Description</h3>

<p>Produce weighted bar plot of risk of bias assessment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rob'
barplot(
  height,
  overall = FALSE,
  weighted = TRUE,
  colour = "cochrane",
  quiet = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="barplot.rob_+3A_height">height</code></td>
<td>
<p>An object of class <code>rob</code>.</p>
</td></tr>
<tr><td><code id="barplot.rob_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether to include a bar for
overall risk of bias in the figure.</p>
</td></tr>
<tr><td><code id="barplot.rob_+3A_weighted">weighted</code></td>
<td>
<p>A logical indicating whether weights should be used
in the bar plot.</p>
</td></tr>
<tr><td><code id="barplot.rob_+3A_colour">colour</code></td>
<td>
<p>Specify colour scheme for the bar plot; see
<code><a href="robvis.html#topic+rob_summary">rob_summary</a></code>.</p>
</td></tr>
<tr><td><code id="barplot.rob_+3A_quiet">quiet</code></td>
<td>
<p>A logical to suppress the display of the bar plot.</p>
</td></tr>
<tr><td><code id="barplot.rob_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a wrapper function for <code><a href="robvis.html#topic+rob_summary">rob_summary</a></code> of
R package <b>robvis</b> to produce a weighted bar plot of risk of
bias assessment.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rob">rob</a></code>, <code><a href="#topic+traffic_light">traffic_light</a></code>,
<code><a href="robvis.html#topic+rob_summary">rob_summary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Use RevMan 5 settings
oldset &lt;- settings.meta("RevMan5")

data(caffeine)

m1 &lt;- metabin(h.caf, n.caf, h.decaf, n.decaf, sm = "OR",
  data = caffeine, studlab = paste(study, year))

# Add risk of bias assessment to meta-analysis
m2 &lt;- rob(D1, D2, D3, D4, D5, overall = rob, data = m1, tool = "rob2")

# Print risk of bias assessment
rob(m2)

## Not run: 
# Weighted bar plot (if R package robvis has been installed)
barplot(rob(m2))

## End(Not run)

# Use previous settings
settings.meta(oldset)

</code></pre>

<hr>
<h2 id='baujat.meta'>Baujat plot to explore heterogeneity in meta-analysis</h2><span id='topic+baujat.meta'></span><span id='topic+baujat'></span>

<h3>Description</h3>

<p>Draw a Baujat plot to explore heterogeneity in meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
baujat(
  x,
  yscale = 1,
  xlim,
  ylim,
  xlab = "Contribution to overall heterogeneity",
  ylab = "Influence on overall result",
  pch = 21,
  cex = 1,
  col = "black",
  bg = "darkgray",
  studlab = TRUE,
  cex.studlab = 0.8,
  pos.studlab,
  offset = 0.5,
  xmin = 0,
  ymin = 0,
  grid = TRUE,
  col.grid = "lightgray",
  lty.grid = "dotted",
  lwd.grid = par("lwd"),
  pty = "s",
  pooled,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="baujat.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_yscale">yscale</code></td>
<td>
<p>Scaling factor for values on y-axis.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min,max) of the plot.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_ylim">ylim</code></td>
<td>
<p>The y limits (min,max) of the plot.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y-axis.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_pch">pch</code></td>
<td>
<p>The plotting symbol used for individual studies.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_cex">cex</code></td>
<td>
<p>The magnification to be used for plotting symbol.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_col">col</code></td>
<td>
<p>A vector with colour of plotting symbols.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_bg">bg</code></td>
<td>
<p>A vector with background colour of plotting symbols (only
used if <code>pch</code> in <code>21:25</code>).</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_studlab">studlab</code></td>
<td>
<p>A logical indicating whether study labels should be
printed in the graph. A vector with study labels can also be
provided (must be of same length as <code>x$TE</code> then).</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_cex.studlab">cex.studlab</code></td>
<td>
<p>The magnification for study labels.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_pos.studlab">pos.studlab</code></td>
<td>
<p>Position of study labels, see argument
<code>pos</code> in <code><a href="graphics.html#topic+text">text</a></code>.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_offset">offset</code></td>
<td>
<p>Offset for study labels (see <code><a href="graphics.html#topic+text">text</a></code>).</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_xmin">xmin</code></td>
<td>
<p>A numeric specifying minimal value to print study
labels (on x-axis).</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_ymin">ymin</code></td>
<td>
<p>A numeric specifying minimal value to print study
labels (on y-axis).</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_grid">grid</code></td>
<td>
<p>A logical indicating whether a grid is printed in the
plot.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_col.grid">col.grid</code></td>
<td>
<p>Colour for grid lines.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_lty.grid">lty.grid</code></td>
<td>
<p>The line type for grid lines.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_lwd.grid">lwd.grid</code></td>
<td>
<p>The line width for grid lines.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_pty">pty</code></td>
<td>
<p>A character specifying type of plot region (see
<code><a href="graphics.html#topic+par">par</a></code>).</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_pooled">pooled</code></td>
<td>
<p>A character string indicating whether a common effect
or random effects model is used for pooling. Either missing (see
Details), <code>"common"</code> or <code>"random"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="baujat.meta_+3A_...">...</code></td>
<td>
<p>Graphical arguments as in <code>par</code> may also be
passed as arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Baujat et al. (2002) introduced a scatter plot to explore
heterogeneity in meta-analysis. On the x-axis the contribution of
each study to the overall heterogeneity statistic (see list object
<code>Q</code> of the meta-analysis object <code>x</code>) is plotted. On the
y-axis the standardised difference of the overall treatment effect
with and without each study is plotted; this quantity describes the
influence of each study on the overal treatment effect.
</p>
<p>Information from object <code>x</code> is utilised if argument
<code>pooled</code> is missing. A common effect model is assumed
(<code>pooled="common"</code>) if argument <code>x$common</code> is
<code>TRUE</code>; a random effects model is assumed
(<code>pooled="random"</code>) if argument <code>x$random</code> is
<code>TRUE</code> and <code>x$common</code> is <code>FALSE</code>.
</p>
<p>Internally, the <code><a href="#topic+metainf">metainf</a></code> function is used to calculate
the values on the y-axis.
</p>


<h3>Value</h3>

<p>A data.frame with the following variables:
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>Coordinate on x-axis (contribution to heterogeneity
statistic)</p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>Coordinate on y-axis (influence on overall treatment
effect)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Baujat B, Mahé C, Pignon JP, Hill C (2002):
A graphical method for exploring heterogeneity in meta-analyses:
Application to a meta-analysis of 65 trials.
<em>Statistics in Medicine</em>,
<b>30</b>, 2641&ndash;52
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metainf">metainf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)

# Only consider first ten studies
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, sm = "OR", method = "I", studlab = paste(author, year),
  subset = 1:10)

# Generate Baujat plot
baujat(m1)

## Not run: 
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, sm = "OR", method = "I", studlab = paste(author, year))

# Do not print study labels if the x-value is smaller than 4 and
# the y-value is smaller than 1
baujat(m1, yscale = 10, xmin = 4, ymin = 1)

# Change position of study labels
baujat(m1, yscale = 10, xmin = 4, ymin = 1,
       pos = 1, xlim = c(0, 6.5))

# Generate Baujat plot and assign x- and y- coordinates to R object
# b1
b1 &lt;- baujat(m1)

# Calculate overall heterogeneity statistic
sum(b1$x)
m1$Q

## End(Not run)

</code></pre>

<hr>
<h2 id='blup.meta'>Calculate best linear unbiased predictor for <code>meta</code> object</h2><span id='topic+blup.meta'></span><span id='topic+blup'></span><span id='topic+print.blup.meta'></span><span id='topic+estimates.blup.meta'></span><span id='topic+print.estimates.blup.meta'></span>

<h3>Description</h3>

<p>Calculate best linear unbiased predictors (BLUPs) for meta-analysis object
created with R package <b>meta</b>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
blup(x, level = x$level, backtransf = x$backtransf, ...)

## S3 method for class 'blup.meta'
print(
  x,
  backtransf = attr(x, "x")$backtransf,
  digits = gs("digits"),
  digits.se = gs("digits.se"),
  digits.tau2 = gs("digits.tau2"),
  digits.tau = gs("digits.tau"),
  big.mark = gs("big.mark"),
  se = FALSE,
  print.tau2 = gs("print.tau2"),
  print.tau = gs("print.tau"),
  details = gs("details"),
  ...
)

## S3 method for class 'blup.meta'
estimates(
  x,
  se = FALSE,
  backtransf = attr(x, "x")$backtransf,
  digits = gs("digits"),
  digits.se = gs("digits.se"),
  digits.tau2 = gs("digits.tau2"),
  digits.tau = gs("digits.tau"),
  writexl = !missing(path),
  path = "estimates_blup.xlsx",
  overwrite = FALSE,
  ...
)

## S3 method for class 'estimates.blup.meta'
print(x, big.mark = gs("big.mark"), details = gs("details"), ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blup.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>, <code>blup.meta</code>, or
<code>estimates.blup.meta</code>.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_level">level</code></td>
<td>
<p>The level used to calculate prediction intervals for BLUPs.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether BLUPs should be
back transformed. If <code>backtransf = TRUE</code>, results for
<code>sm = "OR"</code> will be odds ratios rather than log odds ratios,
for example.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to <code><a href="base.html#topic+prmatrix">prmatrix</a></code>).</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_digits.se">digits.se</code></td>
<td>
<p>Minimal number of significant digits for standard errors.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_digits.tau2">digits.tau2</code></td>
<td>
<p>Minimal number of significant digits for
between-study variance <code class="reqn">\tau^2</code>, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_digits.tau">digits.tau</code></td>
<td>
<p>Minimal number of significant digits for
<code class="reqn">\tau</code>, the square root of the between-study variance
<code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_se">se</code></td>
<td>
<p>A logical indicating whether standard errors should be
printed / extracted.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_print.tau2">print.tau2</code></td>
<td>
<p>A logical specifying whether between-study
variance <code class="reqn">\tau^2</code> should be printed.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_print.tau">print.tau</code></td>
<td>
<p>A logical specifying whether <code class="reqn">\tau</code>, the
square root of the between-study variance <code class="reqn">\tau^2</code>, should be
printed.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_details">details</code></td>
<td>
<p>A logical specifying whether details on
statistical methods should be printed.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_writexl">writexl</code></td>
<td>
<p>A logical indicating whether an Excel file should be
created (R package <b>writexl</b> must be available).</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_path">path</code></td>
<td>
<p>A character string specifying the filename of the Excel
file.</p>
</td></tr>
<tr><td><code id="blup.meta_+3A_overwrite">overwrite</code></td>
<td>
<p>A logical indicating whether an existing Excel
file should be overwritten.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with variables
</p>
<table>
<tr><td><code>studlab</code></td>
<td>
<p>Study label</p>
</td></tr>
<tr><td><code>blup</code></td>
<td>
<p>estimated best linear unbiased predictor</p>
</td></tr>
<tr><td><code>se.blup</code></td>
<td>
<p>standard error (only if argument <code>backtransf = FALSE</code>)</p>
</td></tr>
<tr><td><code>lower</code></td>
<td>
<p>lower prediction limits</p>
</td></tr>
<tr><td><code>upper</code></td>
<td>
<p>upper prediction limits</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("dat.bcg", package = "metadat")
m1 &lt;- metabin(tpos, tpos + tneg, cpos, cpos + cneg,
  data = dat.bcg, studlab = paste(author, year), method = "Inverse")
summary(m1)
blup(m1)

## Not run: 
estimates(blup(m1), path = "blup_m1.xlsx")

## End(Not run)

</code></pre>

<hr>
<h2 id='bubble.metareg'>Bubble plot to display the result of a meta-regression</h2><span id='topic+bubble.metareg'></span><span id='topic+bubble'></span>

<h3>Description</h3>

<p>Draw a bubble plot to display the result of a meta-regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metareg'
bubble(
  x,
  xlim,
  ylim,
  xlab,
  ylab,
  cex,
  min.cex = 0.5,
  max.cex = 5,
  pch = 21,
  col = "black",
  bg = "darkgray",
  lty = 1,
  lwd = 1,
  col.line = "black",
  studlab = FALSE,
  cex.studlab = 0.8,
  pos.studlab = 2,
  offset = 0.5,
  regline = TRUE,
  backtransf = x$.meta$x$backtransf,
  ref,
  col.ref = "lightgray",
  lty.ref = 1,
  lwd.ref = 1,
  pscale = x$.meta$x$pscale,
  irscale = x$.meta$x$irscale,
  axes = TRUE,
  box = TRUE,
  ...
)

bubble(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="bubble.metareg_+3A_x">x</code></td>
<td>
<p>An object of class <code>metareg</code>.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min,max) of the plot.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_ylim">ylim</code></td>
<td>
<p>The y limits (min,max) of the plot.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y-axis.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_cex">cex</code></td>
<td>
<p>The magnification to be used for plotting symbols.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_min.cex">min.cex</code></td>
<td>
<p>Minimal magnification for plotting symbols.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_max.cex">max.cex</code></td>
<td>
<p>Maximal magnification for plotting symbols.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_pch">pch</code></td>
<td>
<p>The plotting symbol used for individual studies.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_col">col</code></td>
<td>
<p>A vector with colour of plotting symbols.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_bg">bg</code></td>
<td>
<p>A vector with background colour of plotting symbols (only
used if <code>pch</code> in <code>21:25</code>).</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_lty">lty</code></td>
<td>
<p>The line type for the meta-regression line.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_lwd">lwd</code></td>
<td>
<p>The line width for the meta-regression line.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_col.line">col.line</code></td>
<td>
<p>Colour for the meta-regression line.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_studlab">studlab</code></td>
<td>
<p>A logical indicating whether study labels should be
printed in the graph. A vector with study labels can also be
provided (must be of same length as the numer of studies in the
meta-analysis then).</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_cex.studlab">cex.studlab</code></td>
<td>
<p>The magnification for study labels.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_pos.studlab">pos.studlab</code></td>
<td>
<p>Position of study labels, see argument
<code>pos</code> in <code><a href="graphics.html#topic+text">text</a></code>.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_offset">offset</code></td>
<td>
<p>Offset for study labels (see <code><a href="graphics.html#topic+text">text</a></code>).</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_regline">regline</code></td>
<td>
<p>A logical indicating whether a regression line
should be added to the bubble plot.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for relative
summary measures (argument <code>sm</code> equal to <code>"OR"</code>,
<code>"RR"</code>, <code>"HR"</code>, or <code>"IRR"</code>) should be back
transformed. If <code>backtransf=TRUE</code>, results for
<code>sm="OR"</code> are printed as odds ratios rather than log odds
ratios, for example.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_ref">ref</code></td>
<td>
<p>A numerical giving the reference value to be plotted as
a line in the bubble plot. No reference line is plotted if
argument <code>ref</code> is equal to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_col.ref">col.ref</code></td>
<td>
<p>Colour of the reference line.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_lty.ref">lty.ref</code></td>
<td>
<p>The line type for the reference line.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_lwd.ref">lwd.ref</code></td>
<td>
<p>The line width for the reference line.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
probabilities.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
incidence rates.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_axes">axes</code></td>
<td>
<p>A logical indicating whether axes should be printed.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_box">box</code></td>
<td>
<p>A logical indicating whether a box should be printed.</p>
</td></tr>
<tr><td><code id="bubble.metareg_+3A_...">...</code></td>
<td>
<p>Graphical arguments as in <code>par</code> may also be
passed as arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A bubble plot can be used to display the result of a
meta-regression. It is a scatter plot with the treatment effect for
each study on the y-axis and the covariate used in the
meta-regression on the x-axis. Typically, the size of the plotting
symbol is inversely proportional to the variance of the estimated
treatment effect (Thompson &amp; Higgins, 2002).
</p>
<p>Argument <code>cex</code> specifies the plotting size for each individual
study. If this argument is missing the weights from the
meta-regression model will be used (which typically is a random
effects model). Use <code>cex="common"</code> in order to utilise weights
from a common effect model to define the size of the plotted
symbols (even for a random effects meta-regression). If a vector
with individual study weights is provided, the length of this
vector must be of the same length as the number of studies.
</p>
<p>Arguments <code>min.cex</code> and <code>max.cex</code> can be used to define
the size of the smallest and largest plotting symbol. The plotting
size of the most precise study is set to <code>max.cex</code> whereas the
plotting size of all studies with a plotting size smaller than
<code>min.cex</code> will be set to <code>min.cex</code>.
</p>
<p>For a meta-regression with more than one covariate. Only a scatter plot of
the first covariate in the regression model is shown. In this case the
effect of the first covariate adjusted for other covariates in the
meta-regression model is shown.
</p>
<p>For a factor or categorial covariate separate bubble plots for each
group compared to the baseline group are plotted.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Thompson SG, Higgins JP (2002):
How should meta-regression analyses be undertaken and interpreted?
<em>Statistics in Medicine</em>,
<b>21</b>, 1559&ndash;73
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metainf">metainf</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)

# Add some (fictitious) grouping variables:
Fleiss1993cont$age &lt;- c(55, 65, 52, 65, 58)
Fleiss1993cont$region &lt;- c("Europe", "Europe", "Asia", "Asia", "Europe")

m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")

mr1 &lt;- metareg(m1, region)
mr1

bubble(mr1)
bubble(mr1, lwd = 2, col.line = "blue")

mr2 &lt;- metareg(m1, age)
mr2

bubble(mr2, lwd = 2, col.line = "blue", xlim = c(50, 70))
bubble(mr2, lwd = 2, col.line = "blue", xlim = c(50, 70), cex = "common")

# Do not print regression line
#
bubble(mr2, lwd = 2, col.line = "blue", xlim = c(50, 70), regline = FALSE)

</code></pre>

<hr>
<h2 id='caffeine'>Caffeine for daytime drowsiness</h2><span id='topic+caffeine'></span>

<h3>Description</h3>

<p>Caffeine for daytime drowsiness (Cochrane Practice review)
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>study</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>year</em></b></td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>h.caf</em></b></td><td style="text-align: left;"> Number of participants with headaches (caffeine group) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.caf</em></b></td><td style="text-align: left;"> Number of participants (caffeine group) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>h.decaf</em></b></td><td style="text-align: left;"> Number of participants with headaches (decaf group) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.decaf</em></b></td><td style="text-align: left;"> Number of participants (decaf group) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>D1</em></b></td><td style="text-align: left;"> Domain 1 of risk of bias 2 tool (RoB 2) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>D2</em></b></td><td style="text-align: left;"> Domain 2 (RoB 2) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>D3</em></b></td><td style="text-align: left;"> Domain 3 (RoB 2) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>D4</em></b></td><td style="text-align: left;"> Domain 4 (RoB 2) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>D5</em></b></td><td style="text-align: left;"> Domain 5 (RoB 2) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>rob</em></b></td><td style="text-align: left;"> Overall RoB 2 assessment
</td>
</tr>

</table>



<h3>Details</h3>

<p>Data come from the Cochrane Practice review on caffeine for daytime
drowsiness. Eight fictitous studies evaluate the risk of headaches
after drinking either caffeinated or decaffeinated coffee.
</p>


<h3>References</h3>

<p>Higgins JPT, Savović J, Page MJ, Sterne JA on behalf of the RoB2
Development Group (2019):
Revised Cochrane risk-of-bias tool for randomized trials.
<a href="https://www.riskofbias.info/welcome/rob-2-0-tool">https://www.riskofbias.info/welcome/rob-2-0-tool</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+rob">rob</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>oldset &lt;- settings.meta("RevMan5")

data(caffeine)
head(caffeine)

m1 &lt;- metabin(h.caf, n.caf, h.decaf, n.decaf, sm = "OR",
  data = caffeine, studlab = paste(study, year))

# Add risk of bias assessment to meta-analysis
m1 &lt;- rob(D1, D2, D3, D4, D5, overall = rob, data = m1, tool = "rob2")

# Print risk of bias assessment
rob(m1)

# Forest plot with risk of bias assessment
forest(m1)

settings.meta(oldset)
</code></pre>

<hr>
<h2 id='ci'>Calculation of confidence intervals (based on normal approximation
or t-distribution)</h2><span id='topic+ci'></span>

<h3>Description</h3>

<p>Calculation of confidence intervals; based on normal approximation
or t-distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ci(TE, seTE, level = 0.95, df = NULL, null.effect = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ci_+3A_te">TE</code></td>
<td>
<p>Estimated treatment effect.</p>
</td></tr>
<tr><td><code id="ci_+3A_sete">seTE</code></td>
<td>
<p>Standard error of treatment estimate.</p>
</td></tr>
<tr><td><code id="ci_+3A_level">level</code></td>
<td>
<p>The confidence level required.</p>
</td></tr>
<tr><td><code id="ci_+3A_df">df</code></td>
<td>
<p>Degrees of freedom (for confidence intervals based on
t-distribution).</p>
</td></tr>
<tr><td><code id="ci_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>List with components
</p>
<table>
<tr><td><code>TE</code></td>
<td>
<p>Estimated treatment effect</p>
</td></tr>
<tr><td><code>seTE</code></td>
<td>
<p>Standard error of treatment estimate</p>
</td></tr>
<tr><td><code>lower</code></td>
<td>
<p>Lower confidence limits</p>
</td></tr>
<tr><td><code>upper</code></td>
<td>
<p>Upper confidence limits</p>
</td></tr>
<tr><td><code>statistic</code></td>
<td>
<p>Test statistic (either z-score or t-score)</p>
</td></tr>
<tr><td><code>p</code></td>
<td>
<p>P-value of text.with null hypothesis <code>TE=0</code></p>
</td></tr>
<tr><td><code>level</code></td>
<td>
<p>The confidence level required</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>Degrees of freedom (t-distribution)</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This function is primarily called from other functions of the
library <code>meta</code>, e.g. <code>forest.meta</code>, <code>summary.meta</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data.frame(ci(170, 10))
data.frame(ci(170, 10, 0.99))
data.frame(ci(1.959964, 1))
data.frame(ci(2.2621571628, 1, df = 9))

</code></pre>

<hr>
<h2 id='cisapride'>Cisapride in Non-Ulcer Dispepsia</h2><span id='topic+cisapride'></span>

<h3>Description</h3>

<p>Meta-analysis on cisapride in non-ulcer dispepsia.
</p>
<p>This meta-analysis is used as a data example in Hartung and Knapp
(2001).
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>study</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>event.cisa</em></b></td><td style="text-align: left;"> number of events in cisapride group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.cisa</em></b></td><td style="text-align: left;"> number of observations in cisapride group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>event.plac</em></b></td><td style="text-align: left;"> number of events in placebo group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.plac</em></b></td><td style="text-align: left;"> number of observations in placebo group
</td>
</tr>

</table>



<h3>Source</h3>

<p>Hartung J &amp; Knapp G (2001):
A refined method for the meta-analysis of controlled clinical
trials with binary outcome.
<em>Statistics in Medicine</em>,
<b>20</b>, 3875&ndash;89
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(cisapride)

m.or &lt;- metabin(event.cisa, n.cisa, event.plac, n.plac,
  data = cisapride, sm = "OR",
  method = "Inverse", method.tau = "DL",
  studlab = study, method.incr = "all")

m.or.hk &lt;- update(m.or, method.random.ci = "HK")
m.rr &lt;- update(m.or, sm = "RR")
m.rr.hk &lt;- update(m.or, sm = "RR", method.random.ci = "HK")

vars.common &lt;- c("TE.common", "lower.common", "upper.common")
vars.random &lt;- c("TE.random", "lower.random", "upper.random")
#
res.common.or &lt;- as.data.frame(m.or[vars.common])
names(res.common.or) &lt;- vars.random
#
res.common.rr &lt;- as.data.frame(m.rr[vars.common])
names(res.common.rr) &lt;- vars.random

# Results for log risk ratio - see Table VII in Hartung and Knapp (2001) 
#
res.rr &lt;- rbind(res.common.rr,
  as.data.frame(m.rr[vars.random]),
  as.data.frame(m.rr.hk[vars.random]))
#
row.names(res.rr) &lt;- c("CE", "RE", "RE (HaKn)")
names(res.rr) &lt;- c("Log risk ratio", "CI lower", "CI upper")
#
res.rr


# Results for log odds ratio (Table VII in Hartung and Knapp 2001) 
#
res.or &lt;- rbind(res.common.or,
  as.data.frame(m.or[vars.random]),
  as.data.frame(m.or.hk[vars.random]))
#
row.names(res.or) &lt;- c("CE", "RE", "RE (HaKn)")
names(res.or) &lt;- c("Log odds ratio", "CI lower", "CI upper")
#
res.or
</code></pre>

<hr>
<h2 id='drapery'>Drapery plot</h2><span id='topic+drapery'></span>

<h3>Description</h3>

<p>Draw a drapery plot with (scaled) p-value curves for individual
studies and meta-analysis estimates.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>drapery(
  x,
  type = "zvalue",
  layout = "grayscale",
  study.results = TRUE,
  lty.study = 1,
  lwd.study = 1,
  col.study = "darkgray",
  labels,
  col.labels = "black",
  cex.labels = 0.7,
  subset.labels,
  srt.labels,
  common = x$common,
  random = x$random,
  lty.common = 1,
  lwd.common = max(3, lwd.study),
  col.common = "blue",
  lty.random = 1,
  lwd.random = lwd.common,
  col.random = "red",
  sign = NULL,
  lty.sign = 1,
  lwd.sign = 1,
  col.sign = "black",
  prediction = random,
  col.predict = "lightblue",
  alpha = if (type == "zvalue") c(0.001, 0.01, 0.05, 0.1) else c(0.01, 0.05, 0.1),
  lty.alpha = 2,
  lwd.alpha = 1,
  col.alpha = "black",
  cex.alpha = 0.7,
  col.null.effect = "black",
  legend = TRUE,
  pos.legend = "topleft",
  bg = "white",
  bty = "o",
  backtransf = x$backtransf,
  xlab,
  ylab,
  xlim,
  ylim,
  lwd.max = 2.5,
  lwd.study.weight = if (random) "random" else "common",
  at = NULL,
  n.grid = if (type == "zvalue") 10000 else 1000,
  mar = c(5.1, 4.1, 4.1, 4.1),
  plot = TRUE,
  warn.deprecated = gs("warn.deprecated"),
  fixed,
  lwd.fixed,
  lty.fixed,
  col.fixed,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="drapery_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="drapery_+3A_type">type</code></td>
<td>
<p>A character string indicating whether to plot test
statistics (<code>"zvalue"</code>) or p-values (<code>"pvalue"</code>), can
be abbreviated.</p>
</td></tr>
<tr><td><code id="drapery_+3A_layout">layout</code></td>
<td>
<p>A character string for the line layout of individual
studies: <code>"grayscale"</code>, <code>"equal"</code>, or
<code>"linewidth"</code> (see Details), can be abbreviated.</p>
</td></tr>
<tr><td><code id="drapery_+3A_study.results">study.results</code></td>
<td>
<p>A logical indicating whether results for
individual studies should be shown in the figure.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lty.study">lty.study</code></td>
<td>
<p>Line type for individual studies.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.study">lwd.study</code></td>
<td>
<p>Line width for individual studies.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.study">col.study</code></td>
<td>
<p>Colour of lines for individual studies.</p>
</td></tr>
<tr><td><code id="drapery_+3A_labels">labels</code></td>
<td>
<p>A logical or character string indicating whether
study labels should be shown at the top of the drapery plot;
either <code>FALSE</code>, <code>"id"</code>, or <code>"studlab"</code>; see
Details.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.labels">col.labels</code></td>
<td>
<p>Colour of study labels.</p>
</td></tr>
<tr><td><code id="drapery_+3A_cex.labels">cex.labels</code></td>
<td>
<p>The magnification for study labels.</p>
</td></tr>
<tr><td><code id="drapery_+3A_subset.labels">subset.labels</code></td>
<td>
<p>A vector specifying which study labels should
be shown in the drapery plot.</p>
</td></tr>
<tr><td><code id="drapery_+3A_srt.labels">srt.labels</code></td>
<td>
<p>A numerical vector or single numeric (between 0
and 90) specifying the angle to rotate study labels; see Details.</p>
</td></tr>
<tr><td><code id="drapery_+3A_common">common</code></td>
<td>
<p>A logical indicating whether to show result for the
common effect model.</p>
</td></tr>
<tr><td><code id="drapery_+3A_random">random</code></td>
<td>
<p>A logical indicating whether to show result for the
random effects model.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lty.common">lty.common</code></td>
<td>
<p>Line type for common effect meta-analysis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.common">lwd.common</code></td>
<td>
<p>Line width for common effect meta-analysis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.common">col.common</code></td>
<td>
<p>Colour of lines for common effect meta-analysis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lty.random">lty.random</code></td>
<td>
<p>Line type for random effects meta-analysis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.random">lwd.random</code></td>
<td>
<p>Line width for random effects meta-analysis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.random">col.random</code></td>
<td>
<p>Colour of lines for random effects meta-analysis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_sign">sign</code></td>
<td>
<p>Significance level used to highlight significant values
in curves.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lty.sign">lty.sign</code></td>
<td>
<p>Line type for significant values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.sign">lwd.sign</code></td>
<td>
<p>Line width for significant values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.sign">col.sign</code></td>
<td>
<p>Line colour for significant values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether to show prediction
region.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.predict">col.predict</code></td>
<td>
<p>Colour of prediction region</p>
</td></tr>
<tr><td><code id="drapery_+3A_alpha">alpha</code></td>
<td>
<p>Horizonal lines are printed for the specified alpha
values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lty.alpha">lty.alpha</code></td>
<td>
<p>Line type of horizonal lines for alpha values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.alpha">lwd.alpha</code></td>
<td>
<p>Line width of horizonal lines for alpha values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.alpha">col.alpha</code></td>
<td>
<p>Colour of horizonal lines for alpha values.</p>
</td></tr>
<tr><td><code id="drapery_+3A_cex.alpha">cex.alpha</code></td>
<td>
<p>The magnification for the text of the alpha</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.null.effect">col.null.effect</code></td>
<td>
<p>Colour of vertical line indicating null
effect.</p>
</td></tr>
<tr><td><code id="drapery_+3A_legend">legend</code></td>
<td>
<p>A logical indicating whether a legend should be
printed.</p>
</td></tr>
<tr><td><code id="drapery_+3A_pos.legend">pos.legend</code></td>
<td>
<p>A character string with position of legend (see
<code><a href="graphics.html#topic+legend">legend</a></code>).</p>
</td></tr>
<tr><td><code id="drapery_+3A_bg">bg</code></td>
<td>
<p>Background colour of legend (see <code><a href="graphics.html#topic+legend">legend</a></code>).</p>
</td></tr>
<tr><td><code id="drapery_+3A_bty">bty</code></td>
<td>
<p>Type of the box around the legend; either <code>"o"</code> or
<code>"n"</code> (see <code><a href="graphics.html#topic+legend">legend</a></code>).</p>
</td></tr>
<tr><td><code id="drapery_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed on the x-axis. For example, if <code>backtransf
= FALSE</code>, log odds ratios instead of odds ratios are shown on the
x-axis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y-axis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min, max) of the plot.</p>
</td></tr>
<tr><td><code id="drapery_+3A_ylim">ylim</code></td>
<td>
<p>The y limits (min, max) of the plot (ignored if
<code>type = "pvalue"</code>).</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.max">lwd.max</code></td>
<td>
<p>The maximum line width (only considered if argument
<code>layout</code> is equal to <code>"linewidth"</code>).</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.study.weight">lwd.study.weight</code></td>
<td>
<p>A character string indicating whether to
determine line width for individual studies using weights from
common effect (<code>"common"</code>) or random effects model
(<code>"random"</code>), can be abbreviated (only considered if
argument <code>layout</code> is equal to <code>"linewidth"</code>).</p>
</td></tr>
<tr><td><code id="drapery_+3A_at">at</code></td>
<td>
<p>Points at which tick-marks are to be drawn on the x-axis.</p>
</td></tr>
<tr><td><code id="drapery_+3A_n.grid">n.grid</code></td>
<td>
<p>The number of grid points to calculate the p-value or
test statistic functions.</p>
</td></tr>
<tr><td><code id="drapery_+3A_mar">mar</code></td>
<td>
<p>Physical plot margin, see <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="drapery_+3A_plot">plot</code></td>
<td>
<p>A logical indicating whether to generate a figure.</p>
</td></tr>
<tr><td><code id="drapery_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="drapery_+3A_fixed">fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'common').</p>
</td></tr>
<tr><td><code id="drapery_+3A_lwd.fixed">lwd.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'lwd.common').</p>
</td></tr>
<tr><td><code id="drapery_+3A_lty.fixed">lty.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'lty.common').</p>
</td></tr>
<tr><td><code id="drapery_+3A_col.fixed">col.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'col.common').</p>
</td></tr>
<tr><td><code id="drapery_+3A_...">...</code></td>
<td>
<p>Graphical arguments as in <code>par</code> may also be
passed as arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The concept of a p-value function, also called confidence curve,
goes back to Birnbaum (1961). A drapery plot, showing p-value
functions (or a scaled version based on the corresponding test
statistics) for individual studies as well as meta-analysis
estimates, is drawn in the active graphics window. Furthermore, a
prediction region for a single future study is shown as a shaded
area. In contrast to a forest plot, a drapery plot does not provide
information for a single confidence level however for any
confidence level.
</p>
<p>Argument <code>type</code> can be used to either show p-value functions
(Birnbaum, 1961) or a scaled version (Infanger, 2019) with test
statistics (default).
</p>
<p>Argument <code>layout</code> determines how curves for individual studies
are presented:
</p>

<ul>
<li><p> darker gray tones with increasing precision (<code>layout =
  "grayscale"</code>)
</p>
</li>
<li><p> thicker lines with increasing precision (<code>layout =
  "linewidth"</code>)
</p>
</li>
<li><p> equal lines (<code>layout = "equal"</code>)
</p>
</li></ul>

<p>Argument <code>labels</code> determines how curves of individual studies
are labelled:
</p>

<ul>
<li><p> number of the study in the (unsorted) forest plot / printout
of a meta-analysis (<code>labels = "id"</code>)
</p>
</li>
<li><p> study labels provided by argument <code>studlab</code> in
meta-analysis functions (<code>labels = "studlab"</code>)
</p>
</li>
<li><p> no study labels (<code>labels = FALSE</code>)
</p>
</li></ul>

<p>By default, study labels are used (<code>labels = "studlab"</code>) if no
label has more than three characters; otherwise IDs are used
(<code>labels = "id"</code>). The connection between IDs and study labels
(among other information) is part of a data frame which is
invisibly returned (if argument <code> study.results = TRUE</code>).
</p>
<p>Argument <code>srt.labels</code> can be used to change the rotation of
IDs or study labels. By default, study labels are rotated by +/- 45
degrees if at least one study label has more than three characters;
otherwise labels are not rotated.
</p>
<p>If <code>labels = "studlab"</code>, labels are rotated by -45 degrees for
studies with a treatment estimate below the common effect estimate
and otherwise by 45 degrees.
</p>


<h3>Author(s)</h3>

<p>Gerta Rücker <a href="mailto:gerta.ruecker@uniklinik-freiburg.de">gerta.ruecker@uniklinik-freiburg.de</a>, Guido
Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Birnbaum A (1961):
Confidence Curves: An Omnibus Technique for Estimation and Testing
Statistical Hypotheses.
<em>Journal of the American Statistical Association</em>,
<b>56</b>, 246&ndash;9
</p>
<p>Infanger D and Schmidt-Trucksäss A (2019):
P value functions: An underused method to present research results
and to promote quantitative reasoning
<em>Statistics in Medicine</em>,
<b>38</b>, 4189&ndash;97
</p>


<h3>See Also</h3>

<p><code><a href="#topic+forest">forest</a></code>, <code><a href="#topic+radial">radial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("lungcancer")
m1 &lt;- metainc(d.smokers, py.smokers, d.nonsmokers, py.nonsmokers,
  data = lungcancer, studlab = study)

# Drapery plot
#
drapery(m1, xlim = c(0.5, 50))

## Not run: 
data(Fleiss1993bin)
m2 &lt;- metabin(d.asp, n.asp, d.plac, n.plac,
  data = Fleiss1993bin, studlab = paste(study, year),
  sm = "OR", random = FALSE)

# Produce drapery plot and print data frame with connection between
# IDs and study labels
#
(drapery(m2))

# For studies with a significant effect (p &lt; 0.05), show
# study labels and print labels and lines in red
#
drapery(m2,
  labels = "studlab", subset.labels = pval &lt; 0.05,
  srt.labels = 0, col.labels = "red",
  col.study = ifelse(pval &lt; 0.05, "red", "darkgray"))

## End(Not run)

</code></pre>

<hr>
<h2 id='estimates.meta'>Extract results from meta-analysis object</h2><span id='topic+estimates.meta'></span><span id='topic+estimates'></span><span id='topic+print.estimates.meta'></span>

<h3>Description</h3>

<p>Extract study and meta-analysis results from meta-analysis object
which can be stored in an Excel file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
estimates(
  x,
  sortvar,
  study.results = TRUE,
  common = x$common,
  random = x$random,
  prediction = x$prediction,
  overall = x$overall,
  subgroup,
  prediction.subgroup = x$prediction.subgroup,
  se = FALSE,
  ci = TRUE,
  statistic = FALSE,
  pval = FALSE,
  n = TRUE,
  backtransf = x$backtransf,
  digits = gs("digits"),
  digits.se = gs("digits.se"),
  digits.stat = gs("digits.stat"),
  digits.pval = gs("digits.pval"),
  writexl = !missing(path),
  path = "estimates.xlsx",
  overwrite = FALSE,
  ...
)

estimates(x, ...)

## S3 method for class 'estimates.meta'
print(
  x,
  digits.tau = gs("digits.tau"),
  text.tau2 = gs("text.tau2"),
  text.tau = gs("text.tau"),
  big.mark = gs("big.mark"),
  details = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimates.meta_+3A_x">x</code></td>
<td>
<p>A meta-analysis object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_sortvar">sortvar</code></td>
<td>
<p>An optional vector used to sort the individual
studies (must be of same length as <code>x$TE</code>).</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_study.results">study.results</code></td>
<td>
<p>A logical indicating whether study results
should be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether results of common effect
meta-analysis should be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether results of random
effects meta-analysis should be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether prediction interval
should be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be extracted. This argument is useful in a meta-analysis
with subgroups if overall results should not be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_subgroup">subgroup</code></td>
<td>
<p>A logical indicating whether subgroup results
should be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which prediction intervals should be
extracted for subgroups.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_se">se</code></td>
<td>
<p>A logical indicating whether standard errors should be
extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_ci">ci</code></td>
<td>
<p>A logical indicating whether confidence / prediction
interval should be extracted.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_statistic">statistic</code></td>
<td>
<p>A logical indicating whether to extract statistic
of test for overall effect.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_pval">pval</code></td>
<td>
<p>A logical indicating whether to extract p-value of test
for overall effect.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_n">n</code></td>
<td>
<p>A logical indicating whether sample sizes should be
extracted (if available).</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether extracted results
should be back transformed.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_digits.se">digits.se</code></td>
<td>
<p>Minimal number of significant digits for standard
errors, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_digits.stat">digits.stat</code></td>
<td>
<p>Minimal number of significant digits for z- or
t-statistic for test of overall effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_digits.pval">digits.pval</code></td>
<td>
<p>Minimal number of significant digits for p-value
of overall treatment effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_writexl">writexl</code></td>
<td>
<p>A logical indicating whether an Excel file should be
created (R package <b>writexl</b> must be available).</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_path">path</code></td>
<td>
<p>A character string specifying the filename of the Excel
file.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_overwrite">overwrite</code></td>
<td>
<p>A logical indicating whether an existing Excel
file should be overwritten.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code>prmatrix</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_digits.tau">digits.tau</code></td>
<td>
<p>Minimal number of significant digits for square
root of between-study variance, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_text.tau2">text.tau2</code></td>
<td>
<p>Text printed to identify between-study variance
<code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_text.tau">text.tau</code></td>
<td>
<p>Text printed to identify <code class="reqn">\tau</code>, the square
root of the between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="estimates.meta_+3A_details">details</code></td>
<td>
<p>A logical specifying whether details on statistical
methods should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Extract study and meta-analysis results from meta-analysis object.
By default, a data frame with the results is
generated. Alternatively, an Excel file is created if argument
<code>writexl = TRUE</code>.
</p>
<p>The following information is extracted.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Variable</b> </td><td style="text-align: left;"> <b>Content</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
studlab </td><td style="text-align: left;"> Study label / descriptor of meta-analysis result </td>
</tr>
<tr>
 <td style="text-align: left;">
estimate </td><td style="text-align: left;"> (Back transformed) estimate for individual studies /
  meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
se </td><td style="text-align: left;"> Standard error </td>
</tr>
<tr>
 <td style="text-align: left;">
lower </td><td style="text-align: left;"> Lower (back transformed) confidence / prediction
  interval limit </td>
</tr>
<tr>
 <td style="text-align: left;">
upper </td><td style="text-align: left;"> Upper (back transformed) confidence / prediction
  interval limit </td>
</tr>
<tr>
 <td style="text-align: left;">
k </td><td style="text-align: left;"> Number of studies </td>
</tr>
<tr>
 <td style="text-align: left;">
df </td><td style="text-align: left;"> Degrees of freedom for confidence / prediction intervals </td>
</tr>
<tr>
 <td style="text-align: left;">
statistic </td><td style="text-align: left;"> Statistic for test of effect </td>
</tr>
<tr>
 <td style="text-align: left;">
pval </td><td style="text-align: left;"> P-value for test of effect </td>
</tr>
<tr>
 <td style="text-align: left;">
n </td><td style="text-align: left;"> Total sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
n.e </td><td style="text-align: left;"> Sample size in first (experimental) group </td>
</tr>
<tr>
 <td style="text-align: left;">
n.c </td><td style="text-align: left;"> Sample size in second (control) group
</td>
</tr>

</table>

<p>Some variables are only extracted if the corresponding logical
argument <code>se</code>, <code>ci</code>, <code>statistic</code> or <code>n</code> is
<code>TRUE</code>. Furthermore, (group) sample sizes are only extracted
if available in the meta-analysis object.
</p>
<p>The variables <code>estimate</code>, <code>lower</code> and <code>upper</code>
contain the back transformed effects and confidence interval
limits, e.g., odds ratio (argument <code>sm = "OR"</code> in
<code><a href="#topic+metabin">metabin</a></code> or <code><a href="#topic+metagen">metagen</a></code>) or correlation
(<code>sm = "ZCOR"</code> in <code><a href="#topic+metacor">metacor</a></code> or
<code><a href="#topic+metagen">metagen</a></code>), if argument <code>backtransf</code> is
<code>TRUE</code>. Otherwise, these variables contain the transformed
values, e.g., log odds ratio (<code>sm = "OR"</code>) or Fisher's Z
transformed correlations (<code>sm = "ZCOR"</code>). See
<code><a href="#topic+meta-sm">meta-sm</a></code> for available summary measures and
<code><a href="#topic+meta-transf">meta-transf</a></code> for the corresponding transformations and
back transformations.
</p>


<h3>Value</h3>

<p>A data frame with additional class 'extract.meta' or an Excel file.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-transf">meta-transf</a></code>, <code><a href="#topic+meta-sm">meta-sm</a></code>,
<code><a href="#topic+as.data.frame.meta">as.data.frame.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
m1 &lt;- metacor(c(0.85, 0.7, 0.95), c(20, 40, 10))
summary(m1)
estimates(m1)
estimates(m1, backtransf = FALSE)
estimates(update(m1, common = FALSE, random = FALSE))
estimates(update(m1, prediction = TRUE))
estimates(update(m1, prediction = TRUE,
  level.ma = 0.99, level.predict = 0.9))

## Not run: 
# Create Excel file with extracted results
# (if R package writexl is available)
#
fname1 &lt;- tempfile(fileext = ".xlsx")
estimates(m1, path = fname1)
# An existing Excel file is not overwritten but a warning is printed
estimates(m1, path = fname1)
# Overwrite an existing Excel file
estimates(m1, path = fname1, overwrite = TRUE)
# Suppress message on file creation and overwrite existing file
suppressMessages(estimates(m1, path = fname1, overwrite = TRUE))

# Save the extracted results in a text file
fname2 &lt;- tempfile(fileext = ".csv")
fname2
write.csv(estimates(m1), file = fname2, row.names = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='Fleiss1993bin'>Aspirin after Myocardial Infarction</h2><span id='topic+Fleiss1993bin'></span><span id='topic+Fleiss93'></span>

<h3>Description</h3>

<p>Meta-analysis on aspirin in preventing death after myocardial
infarction.
</p>
<p>Data example in Fleiss (1993) for meta-analysis with binary
outcomes.
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>study</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>year</em></b></td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>d.asp</em></b></td><td style="text-align: left;"> number of deaths in aspirin group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.asp</em></b></td><td style="text-align: left;"> number of observations in aspirin group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>d.plac</em></b></td><td style="text-align: left;"> number of deaths in placebo group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.plac</em></b></td><td style="text-align: left;"> number of observations in placebo group
</td>
</tr>

</table>



<h3>Source</h3>

<p>Fleiss JL (1993):
The statistical basis of meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>2</b>, 121&ndash;45
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)
metabin(d.asp, n.asp, d.plac, n.plac, data = Fleiss1993bin,
  studlab = paste(study, year), sm = "OR", random = FALSE)
</code></pre>

<hr>
<h2 id='Fleiss1993cont'>Mental Health Treatment</h2><span id='topic+Fleiss1993cont'></span><span id='topic+Fleiss93cont'></span>

<h3>Description</h3>

<p>Meta-analysis on the Effect of Mental Health Treatment on Medical
Utilisation.
</p>
<p>Data example in Fleiss (1993) for meta-analysis with continuous outcomes.
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>study</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>year</em></b></td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.psyc</em></b></td><td style="text-align: left;"> number of observations in psychotherapy
  group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean.psyc</em></b></td><td style="text-align: left;"> estimated mean in psychotherapy group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>sd.psyc</em></b></td><td style="text-align: left;"> standard deviation in psychotherapy group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.cont</em></b></td><td style="text-align: left;"> number of observations in control group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean.cont</em></b></td><td style="text-align: left;"> estimated mean in control group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>sd.cont</em></b></td><td style="text-align: left;"> standard deviation in control group
</td>
</tr>

</table>



<h3>Source</h3>

<p>Fleiss JL (1993):
The statistical basis of meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>2</b>, 121&ndash;45
</p>


<h3>See Also</h3>

<p><code><a href="#topic+Fleiss1993bin">Fleiss1993bin</a></code>
</p>
<p><code><a href="#topic+metacont">metacont</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
# Note, the following command uses the bias-corrected version of
# Hedges' g. Accordingly, results differ from Fleiss (1993), section 3,
# using the uncorrected version of Hedges' g.
metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, studlab = paste(study, year),
  random = FALSE, sm = "SMD")
</code></pre>

<hr>
<h2 id='forest.meta'>Forest plot to display the result of a meta-analysis</h2><span id='topic+forest.meta'></span><span id='topic+forest'></span><span id='topic+plot.meta'></span><span id='topic+.forestArgs'></span>

<h3>Description</h3>

<p>Draw a forest plot (using grid graphics system) in the active
graphics window or store the forest plot in a file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
forest(
  x,
  sortvar,
  studlab = TRUE,
  layout = gs("layout"),
  common = x$common,
  random = x$random,
  overall = x$overall,
  text.common = x$text.common,
  text.random = x$text.random,
  lty.common = gs("lty.common"),
  lty.random = gs("lty.random"),
  col.common = gs("col.common"),
  col.random = gs("col.random"),
  text.w.common = x$text.w.common,
  text.w.random = x$text.w.random,
  prediction = x$prediction,
  text.predict = x$text.predict,
  subgroup = TRUE,
  subgroup.hetstat = subgroup &amp; (is.character(hetstat) || hetstat),
  print.subgroup.labels = TRUE,
  subgroup.name = x$subgroup.name,
  print.subgroup.name = x$print.subgroup.name,
  sep.subgroup = x$sep.subgroup,
  text.common.w = text.common,
  text.random.w = text.random,
  text.predict.w = text.predict,
  sort.subgroup = gs("sort.subgroup"),
  pooled.totals = common | random,
  pooled.events = gs("pooled.events"),
  pooled.times = gs("pooled.times"),
  study.results = gs("study.results"),
  rob = x$rob,
  rob.text = "Risk of Bias",
  rob.xpos = 0,
  rob.legend = TRUE,
  rob.only = FALSE,
  xlab = "",
  xlab.pos,
  smlab = NULL,
  smlab.pos,
  xlim,
  allstudies = TRUE,
  weight.study = NULL,
  pscale = x$pscale,
  irscale = x$irscale,
  irunit = x$irunit,
  file = NULL,
  width = gs("width"),
  rows.gr = NULL,
  func.gr = NULL,
  args.gr = NULL,
  dev.off = NULL,
  ref,
  lower.equi = gs("lower.equi"),
  upper.equi = gs("upper.equi"),
  lty.equi = gs("lty.equi"),
  col.equi = gs("col.equi"),
  fill.equi = gs("fill.equi"),
  fill.lower.equi = fill.equi,
  fill.upper.equi = rev(fill.equi),
  fill = gs("fill"),
  leftcols = gs("leftcols"),
  rightcols = gs("rightcols"),
  leftlabs = gs("leftlabs"),
  rightlabs = gs("rightlabs"),
  label.e = x$label.e,
  label.c = x$label.c,
  label.e.attach = gs("label.e.attach"),
  label.c.attach = gs("label.c.attach"),
  label.right = x$label.right,
  label.left = x$label.left,
  bottom.lr = gs("bottom.lr"),
  lab.NA = gs("lab.NA"),
  lab.NA.effect = gs("lab.NA.effect"),
  lab.NA.weight = gs("lab.NA.weight"),
  lwd = gs("lwd"),
  at = NULL,
  label = TRUE,
  col.label = gs("col.label"),
  type.study = gs("type.study"),
  type.common = gs("type.common"),
  type.random = type.common,
  type.subgroup = ifelse(study.results, "diamond", "square"),
  type.subgroup.common = type.subgroup,
  type.subgroup.random = type.subgroup,
  col.study = gs("col.study"),
  col.square = gs("col.square"),
  col.square.lines = gs("col.square.lines"),
  col.circle = gs("col.circle"),
  col.circle.lines = col.circle,
  col.inside = gs("col.inside"),
  col.inside.common = col.inside,
  col.inside.random = col.inside,
  col.diamond = gs("col.diamond"),
  col.diamond.common = col.diamond,
  col.diamond.random = col.diamond,
  col.diamond.lines = gs("col.diamond.lines"),
  col.diamond.lines.common = col.diamond.lines,
  col.diamond.lines.random = col.diamond.lines,
  col.predict = gs("col.predict"),
  col.predict.lines = gs("col.predict.lines"),
  col.subgroup = gs("col.subgroup"),
  col.label.right = gs("col.label.right"),
  col.label.left = gs("col.label.left"),
  hetstat = common | random | overall.hetstat,
  overall.hetstat = x$overall.hetstat &amp; !inherits(x, "metamerge"),
  hetlab = gs("hetlab"),
  resid.hetstat = gs("resid.hetstat"),
  resid.hetlab = gs("resid.hetlab"),
  print.I2 = gs("forest.I2"),
  print.I2.ci = gs("forest.I2.ci"),
  print.tau2 = gs("forest.tau2"),
  print.tau2.ci = gs("forest.tau2.ci"),
  print.tau = gs("forest.tau"),
  print.tau.ci = gs("forest.tau.ci"),
  print.Q = gs("forest.Q"),
  print.pval.Q = gs("forest.pval.Q"),
  print.Rb = gs("forest.Rb"),
  print.Rb.ci = gs("forest.Rb.ci"),
  text.subgroup.nohet = gs("text.subgroup.nohet"),
  LRT = gs("LRT"),
  test.overall = gs("test.overall"),
  test.overall.common = common &amp; overall &amp; test.overall,
  test.overall.random = random &amp; overall &amp; test.overall,
  label.test.overall.common,
  label.test.overall.random,
  print.stat = gs("forest.stat"),
  test.subgroup = x$test.subgroup,
  test.subgroup.common = test.subgroup &amp; common,
  test.subgroup.random = test.subgroup &amp; random,
  common.subgroup = common,
  random.subgroup = random,
  prediction.subgroup = x$prediction.subgroup,
  print.Q.subgroup = gs("forest.Q.subgroup"),
  label.test.subgroup.common,
  label.test.subgroup.random,
  test.effect.subgroup = gs("test.effect.subgroup"),
  test.effect.subgroup.common,
  test.effect.subgroup.random,
  label.test.effect.subgroup.common,
  label.test.effect.subgroup.random,
  text.addline1,
  text.addline2,
  details = gs("forest.details"),
  col.lines = gs("col.lines"),
  header.line,
  col.header.line = col.lines,
  col.jama.line = col.subgroup,
  fontsize = gs("fontsize"),
  fontfamily = gs("fontfamily"),
  fs.heading = fontsize,
  fs.common = gs("fs.common"),
  fs.random = gs("fs.random"),
  fs.predict = gs("fs.predict"),
  fs.common.labels = gs("fs.common.labels"),
  fs.random.labels = gs("fs.random.labels"),
  fs.predict.labels = gs("fs.predict.labels"),
  fs.study = fontsize,
  fs.study.labels = fs.study,
  fs.hetstat = gs("fs.hetstat"),
  fs.test.overall = gs("fs.test.overall"),
  fs.test.subgroup = gs("fs.test.subgroup"),
  fs.test.effect.subgroup = gs("fs.test.effect.subgroup"),
  fs.addline = gs("fs.addline"),
  fs.axis = fontsize,
  fs.smlab = fontsize,
  fs.xlab = fontsize,
  fs.lr = fontsize,
  fs.rob = fontsize,
  fs.rob.symbols = fontsize,
  fs.details = fontsize,
  ff.heading = "bold",
  ff.common = gs("ff.common"),
  ff.random = gs("ff.random"),
  ff.predict = gs("ff.predict"),
  ff.common.labels = gs("ff.common.labels"),
  ff.random.labels = gs("ff.random.labels"),
  ff.predict.labels = gs("ff.predict.labels"),
  ff.study = "plain",
  ff.study.labels = ff.study,
  ff.hetstat = gs("ff.hetstat"),
  ff.test.overall = gs("ff.test.overall"),
  ff.test.subgroup = gs("ff.test.subgroup"),
  ff.test.effect.subgroup = gs("ff.test.effect.subgroup"),
  ff.addline = gs("ff.addline"),
  ff.axis = gs("ff.axis"),
  ff.smlab = gs("ff.smlab"),
  ff.xlab = gs("ff.xlab"),
  ff.lr = gs("ff.lr"),
  ff.rob = "plain",
  ff.rob.symbols = "bold",
  ff.details = "plain",
  squaresize = if (layout == "BMJ") 0.9/spacing else 0.8/spacing,
  lwd.square = gs("lwd.square"),
  lwd.diamond = gs("lwd.diamond"),
  arrow.type = gs("arrow.type"),
  arrow.length = gs("arrow.length"),
  plotwidth = if (layout %in% c("BMJ", "JAMA")) "8cm" else "6cm",
  colgap = gs("colgap"),
  colgap.left = colgap,
  colgap.right = colgap,
  colgap.studlab = colgap.left,
  colgap.forest = gs("colgap.forest"),
  colgap.forest.left = colgap.forest,
  colgap.forest.right = colgap.forest,
  colgap.rob = "1mm",
  colgap.rob.overall = "2mm",
  calcwidth.pooled = (common | random) &amp; (overall | !is.null(x$subgroup)),
  calcwidth.common = calcwidth.pooled,
  calcwidth.random = calcwidth.pooled,
  calcwidth.predict = gs("calcwidth.predict"),
  calcwidth.hetstat = gs("calcwidth.hetstat"),
  calcwidth.tests = gs("calcwidth.tests"),
  calcwidth.subgroup = gs("calcwidth.subgroup"),
  calcwidth.addline = gs("calcwidth.addline"),
  just = if (layout == "JAMA") "left" else "right",
  just.studlab = gs("just.studlab"),
  just.addcols = gs("just.addcols"),
  just.addcols.left = just.addcols,
  just.addcols.right = just.addcols,
  bmj.text = NULL,
  bmj.xpos = 0,
  bmj.sep = " / ",
  spacing = gs("spacing"),
  addrow = gs("addrow"),
  addrow.overall = gs("addrow.overall"),
  addrow.subgroups = gs("addrow.subgroups"),
  addrows.below.overall = gs("addrows.below.overall"),
  new = TRUE,
  backtransf = x$backtransf,
  digits = gs("digits.forest"),
  digits.se = gs("digits.se"),
  digits.stat = gs("digits.stat"),
  digits.pval = max(gs("digits.pval") - 2, 2),
  digits.pval.Q = max(gs("digits.pval.Q") - 2, 2),
  digits.Q = gs("digits.Q"),
  digits.tau2 = gs("digits.tau2"),
  digits.tau = gs("digits.tau"),
  digits.I2 = max(gs("digits.I2") - 1, 0),
  digits.weight = gs("digits.weight"),
  digits.mean = gs("digits.mean"),
  digits.sd = gs("digits.sd"),
  digits.cor = digits,
  digits.time = digits,
  digits.n = 0,
  digits.event = 0,
  digits.TE = gs("digits.TE.forest"),
  digits.addcols = digits,
  digits.addcols.right = digits.addcols,
  digits.addcols.left = digits.addcols,
  scientific.pval = gs("scientific.pval"),
  big.mark = gs("big.mark"),
  zero.pval = if (layout == "JAMA") FALSE else gs("zero.pval"),
  JAMA.pval = if (layout == "JAMA") TRUE else gs("JAMA.pval"),
  warn.deprecated = gs("warn.deprecated"),
  ...
)

## S3 method for class 'meta'
plot(x, ...)

.forestArgs()
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forest.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_sortvar">sortvar</code></td>
<td>
<p>An optional vector used to sort the individual
studies (must be of same length as <code>x$TE</code>).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_studlab">studlab</code></td>
<td>
<p>A logical indicating whether study labels should be
printed in the graph. A vector with study labels can also be
provided (must be of same length as <code>x$TE</code> then).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_layout">layout</code></td>
<td>
<p>A character string specifying the layout of the
forest plot (see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether common effect estimate
should be plotted.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether random effects estimate
should be plotted.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be plotted. This argument is useful in a meta-analysis
with subgroups if summaries should only be plotted on group
level.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in the plot to label the
pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in the plot to label the
pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lty.common">lty.common</code></td>
<td>
<p>Line type of pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lty.random">lty.random</code></td>
<td>
<p>Line type of pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.common">col.common</code></td>
<td>
<p>Line colour of pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.random">col.random</code></td>
<td>
<p>Line colour of pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in the plot to label
the prediction interval.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_subgroup">subgroup</code></td>
<td>
<p>A single logical or logical vector indicating
whether / which subgroup results should be shown in forest
plot. This argument is useful in a meta-analysis with subgroups
if summaries should not be plotted for (some) subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_subgroup.hetstat">subgroup.hetstat</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which information on heterogeneity in
subgroups should be shown in forest plot. This argument is useful
in a meta-analysis with subgroups if heterogeneity statistics
should not be printed for (some) subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.subgroup.labels">print.subgroup.labels</code></td>
<td>
<p>A logical indicating whether subgroup
label should be printed.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a label for the
grouping variable.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the grouping variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between label and levels of grouping variable.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.common.w">text.common.w</code></td>
<td>
<p>A character string to label the pooled common
effect estimate within subgroups, or a character vector of same
length as number of subgroups with corresponging labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.random.w">text.random.w</code></td>
<td>
<p>A character string to label the pooled random
effect estimate within subgroups, or a character vector of same
length as number of subgroups with corresponging labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.predict.w">text.predict.w</code></td>
<td>
<p>A character string to label the prediction
interval within subgroups, or a character vector of same length
as number of subgroups with corresponging labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_sort.subgroup">sort.subgroup</code></td>
<td>
<p>A logical indicating whether groups should be
ordered alphabetically.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_pooled.totals">pooled.totals</code></td>
<td>
<p>A logical indicating whether total number of
observations should be given in the figure.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_pooled.events">pooled.events</code></td>
<td>
<p>A logical indicating whether total number of
events should be given in the figure.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_pooled.times">pooled.times</code></td>
<td>
<p>A logical indicating whether total person time
at risk should be given in the figure.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_study.results">study.results</code></td>
<td>
<p>A logical indicating whether results for
individual studies should be shown in the figure (useful to only
plot subgroup results).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rob">rob</code></td>
<td>
<p>Risk of bias (RoB) assessment.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rob.text">rob.text</code></td>
<td>
<p>Column heading for RoB table.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rob.xpos">rob.xpos</code></td>
<td>
<p>A numeric specifying the horizontal position of the
risk of bias label in RoB table heading. The value is a so called
normalised parent coordinate in the horizontal direction (see
<code><a href="grid.html#topic+unit">unit</a></code>).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rob.legend">rob.legend</code></td>
<td>
<p>A logical specifying whether a legend with RoB
domains should be printed.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rob.only">rob.only</code></td>
<td>
<p>A logical indicating whether the risk of bias
assessment is the only information printed on the right side of
the forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_xlab.pos">xlab.pos</code></td>
<td>
<p>A numeric specifying the center of the label on the
x-axis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_smlab">smlab</code></td>
<td>
<p>A label for the summary measure (printed at top of
figure).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_smlab.pos">smlab.pos</code></td>
<td>
<p>A numeric specifying the center of the label for
the summary measure.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min,max) of the plot, or the character
string &quot;symmetric&quot; to produce symmetric forest plots.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_allstudies">allstudies</code></td>
<td>
<p>A logical indicating whether studies with
inestimable treatment effects should be included in the forest
plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_weight.study">weight.study</code></td>
<td>
<p>A character string indicating weighting used to
determine size of squares or diamonds (argument
<code>type.study</code>) to plot individual study results. One of
missing, <code>"same"</code>, <code>"common"</code>, or <code>"random"</code>, can
be abbreviated. Plot symbols have the same size for all studies
or represent study weights from common effect or random effects
model.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
single event probabilities or risk differences, i.e. if argument
<code>sm</code> is equal to <code>"PLOGIT"</code>, <code>"PLN"</code>,
<code>"PRAW"</code>, <code>"PAS"</code>, <code>"PFT"</code>, or <code>"RD"</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single incidence rates or incidence rate differences, i.e. if
argument <code>sm</code> is equal to <code>"IR"</code>, <code>"IRLN"</code>,
<code>"IRS"</code>, <code>"IRFT"</code>, or <code>"IRD"</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_irunit">irunit</code></td>
<td>
<p>A character specifying the time unit used to
calculate rates, e.g., person-years.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_file">file</code></td>
<td>
<p>File name.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_width">width</code></td>
<td>
<p>Width of graphics file.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rows.gr">rows.gr</code></td>
<td>
<p>Additional rows in forest plot to change height of
graphics file (e.g., in order to add a title at the top of the
forest plot).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_func.gr">func.gr</code></td>
<td>
<p>Name of graphics function, e.g., <code><a href="grDevices.html#topic+pdf">pdf</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_args.gr">args.gr</code></td>
<td>
<p>List with additional graphical parameters passed on
to graphics function (argument 'height' cannot be provided as the
height is calculated internally; use instead argument 'rows.gr').</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_dev.off">dev.off</code></td>
<td>
<p>A logical to specify whether current graphics device
should be shut down, i.e., whether file should be stored.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ref">ref</code></td>
<td>
<p>A numerical giving the reference value to be plotted as
a line in the forest plot. No reference line is plotted if
argument <code>ref</code> is equal to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lower.equi">lower.equi</code></td>
<td>
<p>A numerical giving the lower limit of equivalence
to be plotted as a line in the forest plot. Or a vector to
provide several limits, e.g., for large, moderate and small
effects. No line is plotted if argument <code>lower.equi</code> is
equal to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_upper.equi">upper.equi</code></td>
<td>
<p>A numerical giving the upper limit of equivalence
to be plotted as a line in the forest plot. Or a vector to
provide several limits, e.g., for small, moderate and large
effects. No line is plotted if argument <code>upper.equi</code> is
equal to <code>NA</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lty.equi">lty.equi</code></td>
<td>
<p>Line type (limits of equivalence).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.equi">col.equi</code></td>
<td>
<p>Line colour (limits of equivalence).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fill.equi">fill.equi</code></td>
<td>
<p>Colour(s) for area between limits of equivalence
or more general limits.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fill.lower.equi">fill.lower.equi</code></td>
<td>
<p>Colour of area between lower limit(s) and
reference value. Can be equal to the number of lower limits or
the number of limits plus 1 (in this case the the region between
minimum and smallest limit is also filled).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fill.upper.equi">fill.upper.equi</code></td>
<td>
<p>Colour of area between reference value and
upper limit(s). Can be equal to the number of upper limits or the
number of limits plus 1 (in this case the region between largest
limit and maximum is also filled).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fill">fill</code></td>
<td>
<p>Colour for background of confidence interval plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_leftcols">leftcols</code></td>
<td>
<p>A character vector specifying (additional) columns
to be printed on the left side of the forest plot or a logical
value (see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rightcols">rightcols</code></td>
<td>
<p>A character vector specifying (additional) columns
to be printed on the right side of the forest plot or a logical
value (see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_leftlabs">leftlabs</code></td>
<td>
<p>A character vector specifying labels for
(additional) columns on left side of the forest plot (see
Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_rightlabs">rightlabs</code></td>
<td>
<p>A character vector specifying labels for
(additional) columns on right side of the forest plot (see
Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.e">label.e</code></td>
<td>
<p>Label to be used for experimental group in table
heading.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.c">label.c</code></td>
<td>
<p>Label to be used for control group in table heading.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.e.attach">label.e.attach</code></td>
<td>
<p>A character specifying the column name where
label <code>label.e</code> should be attached to in table heading.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.c.attach">label.c.attach</code></td>
<td>
<p>A character specifying the column name where
label <code>label.c</code> should be attached to in table heading.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.right">label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.left">label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_bottom.lr">bottom.lr</code></td>
<td>
<p>A logical indicating whether labels on right and
left side should be printed at bottom or top of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lab.na">lab.NA</code></td>
<td>
<p>A character string to label missing values.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lab.na.effect">lab.NA.effect</code></td>
<td>
<p>A character string to label missing values in
individual treatment estimates and confidence intervals.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lab.na.weight">lab.NA.weight</code></td>
<td>
<p>A character string to label missing weights.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lwd">lwd</code></td>
<td>
<p>The line width, see <code><a href="graphics.html#topic+par">par</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_at">at</code></td>
<td>
<p>The points at which tick-marks are to be drawn, see
<code>grid.xaxis</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label">label</code></td>
<td>
<p>A logical value indicating whether to draw the labels
on the tick marks, or an expression or character vector which
specify the labels to use. See <code><a href="grid.html#topic+grid.xaxis">grid.xaxis</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.label">col.label</code></td>
<td>
<p>The colour of labels on the x-axis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_type.study">type.study</code></td>
<td>
<p>A character string or vector specifying how to
plot treatment effects and confidence intervals for individual
studies (see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_type.common">type.common</code></td>
<td>
<p>A character string specifying how to plot
treatment effect and confidence interval for common effect
meta-analysis (see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_type.random">type.random</code></td>
<td>
<p>A character string specifying how to plot
treatment effect and confidence interval for random effects
meta-analysis (see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_type.subgroup">type.subgroup</code></td>
<td>
<p>A character string specifying how to plot
treatment effect and confidence interval for subgroup results
(see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_type.subgroup.common">type.subgroup.common</code></td>
<td>
<p>A character string specifying how to
plot treatment effect and confidence interval for subgroup
results (common effect model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_type.subgroup.random">type.subgroup.random</code></td>
<td>
<p>A character string specifying how to
plot treatment effect and confidence interval for subgroup
results (random effects model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.study">col.study</code></td>
<td>
<p>The colour for individual study results and
confidence limits.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.square">col.square</code></td>
<td>
<p>The colour for squares reflecting study's weight
in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.square.lines">col.square.lines</code></td>
<td>
<p>The colour for the outer lines of squares
reflecting study's weight in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.circle">col.circle</code></td>
<td>
<p>The colour for circles reflecting study weights
in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.circle.lines">col.circle.lines</code></td>
<td>
<p>The colour for the outer lines of circles
reflecting study's weight in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.inside">col.inside</code></td>
<td>
<p>The colour for individual study results and
confidence limits if confidence limits are completely within
squares.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.inside.common">col.inside.common</code></td>
<td>
<p>The colour for result of common effect
meta-analysis if confidence limit lies completely within square.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.inside.random">col.inside.random</code></td>
<td>
<p>The colour for result of random effects
meta-analysis if confidence limit lies completely within square.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.diamond">col.diamond</code></td>
<td>
<p>The colour of diamonds representing the results
for common effect and random effects models.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.diamond.common">col.diamond.common</code></td>
<td>
<p>The colour of diamonds for common effect
estimates.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.diamond.random">col.diamond.random</code></td>
<td>
<p>The colour of diamonds for random effects
estimates.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.diamond.lines">col.diamond.lines</code></td>
<td>
<p>The colour of the outer lines of diamonds
representing the results for common effect and random effects
models.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.diamond.lines.common">col.diamond.lines.common</code></td>
<td>
<p>The colour of the outer lines of
diamond for common effect estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.diamond.lines.random">col.diamond.lines.random</code></td>
<td>
<p>The colour of the outer lines of
diamond for random effects estimate.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.predict">col.predict</code></td>
<td>
<p>Background colour of prediction interval.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.predict.lines">col.predict.lines</code></td>
<td>
<p>Colour of outer lines of prediction
interval.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.subgroup">col.subgroup</code></td>
<td>
<p>The colour to print information on subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.label.right">col.label.right</code></td>
<td>
<p>The colour for label on right side of null
effect.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.label.left">col.label.left</code></td>
<td>
<p>The colour for label on left side of null
effect.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_hetstat">hetstat</code></td>
<td>
<p>Either a logical value indicating whether to print
results for heterogeneity measures at all or a character string
(see Details).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_hetlab">hetlab</code></td>
<td>
<p>Label printed in front of results for heterogeneity
measures.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_resid.hetstat">resid.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
measures of residual heterogeneity in a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_resid.hetlab">resid.hetlab</code></td>
<td>
<p>Label printed in front of results for residual
heterogeneity measures.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.i2">print.I2</code></td>
<td>
<p>A logical value indicating whether to print the
value of the I-squared statistic.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.i2.ci">print.I2.ci</code></td>
<td>
<p>A logical value indicating whether to print the
confidence interval of the I-squared statistic.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.tau2">print.tau2</code></td>
<td>
<p>A logical value indicating whether to print the
value of the between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.tau2.ci">print.tau2.ci</code></td>
<td>
<p>A logical value indicating whether to print
the confidence interval of <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.tau">print.tau</code></td>
<td>
<p>A logical value indicating whether to print
<code class="reqn">\tau</code>, the square root of the between-study variance
<code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.tau.ci">print.tau.ci</code></td>
<td>
<p>A logical value indicating whether to print the
confidence interval of <code class="reqn">\tau</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.q">print.Q</code></td>
<td>
<p>A logical value indicating whether to print the
value of the heterogeneity statistic Q.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.pval.q">print.pval.Q</code></td>
<td>
<p>A logical value indicating whether to print the
p-value of the heterogeneity statistic Q.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.rb">print.Rb</code></td>
<td>
<p>A logical value indicating whether to print the
value of the I-squared statistic.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.rb.ci">print.Rb.ci</code></td>
<td>
<p>A logical value indicating whether to print the
confidence interval of the I-squared statistic.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.subgroup.nohet">text.subgroup.nohet</code></td>
<td>
<p>A logical value or character string
which is printed to indicate subgroups with less than two studies
contributing to meta-analysis (and thus without
heterogeneity). If FALSE, heterogeneity statistics are printed
(with NAs).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lrt">LRT</code></td>
<td>
<p>A logical value indicating whether to report
Likelihood-Ratio or Wald-type test of heterogeneity for
generalized linear mixed models.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.overall">test.overall</code></td>
<td>
<p>A logical value indicating whether to print
results of test for overall effect.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.overall.common">test.overall.common</code></td>
<td>
<p>A logical value indicating whether to
print results of test for overall effect (common effect model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.overall.random">test.overall.random</code></td>
<td>
<p>A logical value indicating whether to
print results of test for overall effect (random effects model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.test.overall.common">label.test.overall.common</code></td>
<td>
<p>Label printed in front of results
of test for overall effect (common effect model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.test.overall.random">label.test.overall.random</code></td>
<td>
<p>Label printed in front of results
of test for overall effect (random effects model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.stat">print.stat</code></td>
<td>
<p>A logical value indicating whether z- or t-value
for test of treatment effect should be printed.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.subgroup.common">test.subgroup.common</code></td>
<td>
<p>A logical value indicating whether to
print results of test for subgroup differences (common effect
model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.subgroup.random">test.subgroup.random</code></td>
<td>
<p>A logical value indicating whether to
print results of test for subgroup differences (random effects
model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_common.subgroup">common.subgroup</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which common effect estimates should be
printed for subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_random.subgroup">random.subgroup</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which random effects estimates should be
printed for subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which prediction intervals should be printed
for subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_print.q.subgroup">print.Q.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
the value of the heterogeneity statistic Q (test for subgroup
differences).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.test.subgroup.common">label.test.subgroup.common</code></td>
<td>
<p>Label printed in front of results
of test for subgroup differences (common effect model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.test.subgroup.random">label.test.subgroup.random</code></td>
<td>
<p>Label printed in front of results
of test for subgroup differences (random effects model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.effect.subgroup">test.effect.subgroup</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which tests for effect in subgroups should
be printed.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.effect.subgroup.common">test.effect.subgroup.common</code></td>
<td>
<p>A single logical or logical
vector indicating whether / which tests for effect in subgroups
should be printed (common effect model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_test.effect.subgroup.random">test.effect.subgroup.random</code></td>
<td>
<p>A single logical or logical
vector indicating whether / which tests for effect in subgroups
should be printed (random effects model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.test.effect.subgroup.common">label.test.effect.subgroup.common</code></td>
<td>
<p>Label printed in front of
results of test for effect in subgroups (common effect model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_label.test.effect.subgroup.random">label.test.effect.subgroup.random</code></td>
<td>
<p>Label printed in front of
results of test for effect in subgroups (random effects model).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.addline1">text.addline1</code></td>
<td>
<p>Text for first additional line (below
meta-analysis results).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_text.addline2">text.addline2</code></td>
<td>
<p>Text for second additional line (below
meta-analysis results).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_details">details</code></td>
<td>
<p>A logical specifying whether details on statistical
methods should be printed.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.lines">col.lines</code></td>
<td>
<p>The colour of lines.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_header.line">header.line</code></td>
<td>
<p>A logical value indicating whether to print a
header line or a character string (&quot;both&quot;, &quot;below&quot;, &quot;&quot;).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.header.line">col.header.line</code></td>
<td>
<p>Colour of the header line(s).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_col.jama.line">col.jama.line</code></td>
<td>
<p>Colour of the additional JAMA lines.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fontsize">fontsize</code></td>
<td>
<p>The size of text (in points), see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fontfamily">fontfamily</code></td>
<td>
<p>The font family, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.heading">fs.heading</code></td>
<td>
<p>The size of text for column headings, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.common">fs.common</code></td>
<td>
<p>The size of text for results of common effect
model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.random">fs.random</code></td>
<td>
<p>The size of text for results of random effects
model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.predict">fs.predict</code></td>
<td>
<p>The size of text for results of prediction
interval, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.common.labels">fs.common.labels</code></td>
<td>
<p>The size of text for label of common effect
model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.random.labels">fs.random.labels</code></td>
<td>
<p>The size of text for label of random
effects model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.predict.labels">fs.predict.labels</code></td>
<td>
<p>The size of text for label of prediction
interval, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.study">fs.study</code></td>
<td>
<p>The size of text for results of individual studies,
see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.study.labels">fs.study.labels</code></td>
<td>
<p>The size of text for labels of individual
studies, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.hetstat">fs.hetstat</code></td>
<td>
<p>The size of text for heterogeneity measures, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.test.overall">fs.test.overall</code></td>
<td>
<p>The size of text of test for overall effect,
see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.test.subgroup">fs.test.subgroup</code></td>
<td>
<p>The size of text of test of subgroup
differences, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.test.effect.subgroup">fs.test.effect.subgroup</code></td>
<td>
<p>The size of text of test of effect
in subgroups, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.addline">fs.addline</code></td>
<td>
<p>The size of text for additional lines, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.axis">fs.axis</code></td>
<td>
<p>The size of text on x-axis, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.smlab">fs.smlab</code></td>
<td>
<p>The size of text of label for summary measure, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.xlab">fs.xlab</code></td>
<td>
<p>The size of text of label on x-axis, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.lr">fs.lr</code></td>
<td>
<p>The size of text of label on left and right side of
forest plot, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.rob">fs.rob</code></td>
<td>
<p>The size of text of risk of bias items in the legend,
see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.rob.symbols">fs.rob.symbols</code></td>
<td>
<p>The size of risk of bias symbols, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_fs.details">fs.details</code></td>
<td>
<p>The size of text for details on (meta-analysis)
methods, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.heading">ff.heading</code></td>
<td>
<p>The fontface for column headings, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.common">ff.common</code></td>
<td>
<p>The fontface of text for results of common effect
model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.random">ff.random</code></td>
<td>
<p>The fontface of text for results of random effects
model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.predict">ff.predict</code></td>
<td>
<p>The fontface of text for results of prediction
interval, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.common.labels">ff.common.labels</code></td>
<td>
<p>The fontface of text for label of common
effect model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.random.labels">ff.random.labels</code></td>
<td>
<p>The fontface of text for label of random
effects model, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.predict.labels">ff.predict.labels</code></td>
<td>
<p>The fontface of text for label of
prediction interval, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.study">ff.study</code></td>
<td>
<p>The fontface of text for results of individual
studies, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.study.labels">ff.study.labels</code></td>
<td>
<p>The fontface of text for labels of
individual studies, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.hetstat">ff.hetstat</code></td>
<td>
<p>The fontface of text for heterogeneity measures,
see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.test.overall">ff.test.overall</code></td>
<td>
<p>The fontface of text of test for overall
effect, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.test.subgroup">ff.test.subgroup</code></td>
<td>
<p>The fontface of text for test of subgroup
differences, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.test.effect.subgroup">ff.test.effect.subgroup</code></td>
<td>
<p>The fontface of text for test of
effect in subgroups, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.addline">ff.addline</code></td>
<td>
<p>The fontface of text for additional lines, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.axis">ff.axis</code></td>
<td>
<p>The fontface of text on x-axis, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.smlab">ff.smlab</code></td>
<td>
<p>The fontface of text of label for summary measure,
see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.xlab">ff.xlab</code></td>
<td>
<p>The fontface of text of label on x-axis, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.lr">ff.lr</code></td>
<td>
<p>The fontface of text of label on left and right side
of forest plot, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.rob">ff.rob</code></td>
<td>
<p>The fontface of text of risk of bias items, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.rob.symbols">ff.rob.symbols</code></td>
<td>
<p>The fontface of risk of bias symbols, see
<code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_ff.details">ff.details</code></td>
<td>
<p>The fontface for details on (meta-analysis)
methods, see <code><a href="grid.html#topic+gpar">gpar</a></code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_squaresize">squaresize</code></td>
<td>
<p>A numeric used to increase or decrease the size
of squares in the forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lwd.square">lwd.square</code></td>
<td>
<p>The line width of the border around squares.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_lwd.diamond">lwd.diamond</code></td>
<td>
<p>The line width of the border around diamonds.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_arrow.type">arrow.type</code></td>
<td>
<p>A character string indicating whether arrows
printed for results outside the forest plot should be
<code>"open"</code>, or <code>"closed"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_arrow.length">arrow.length</code></td>
<td>
<p>The length of arrows in inches.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_plotwidth">plotwidth</code></td>
<td>
<p>Either a character string, e.g., &quot;8cm&quot;, &quot;60mm&quot;, or
&quot;3inch&quot;, or a <code><a href="grid.html#topic+unit">unit</a></code> object specifying width of
the forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap">colgap</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between columns
printed on left and right side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.left">colgap.left</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between columns
printed on left side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.right">colgap.right</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between columns
printed on right side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.studlab">colgap.studlab</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between column
with study labels and subsequent column.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.forest">colgap.forest</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between column
adjacent to forest plot and the forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.forest.left">colgap.forest.left</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between column on
the left side of forest plot and the forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.forest.right">colgap.forest.right</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between column on
the right side of forest plot and the forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.rob">colgap.rob</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap between risk of
bias columns.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_colgap.rob.overall">colgap.rob.overall</code></td>
<td>
<p>Either a character string or a
<code><a href="grid.html#topic+unit">unit</a></code> object specifying gap before column with
overall risk of bias assessment.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.pooled">calcwidth.pooled</code></td>
<td>
<p>A logical indicating whether text for
common effect and random effects model should be considered to
calculate width of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.common">calcwidth.common</code></td>
<td>
<p>A logical indicating whether text given in
arguments <code>text.common</code> and <code>text.common.w</code> should be
considered to calculate width of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.random">calcwidth.random</code></td>
<td>
<p>A logical indicating whether text given in
arguments <code>text.random</code> and <code>text.random.w</code> should be
considered to calculate width of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.predict">calcwidth.predict</code></td>
<td>
<p>A logical indicating whether text given in
argument <code>text.predict</code> and <code>text.predict.w</code> should be
considered to calculate width of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.hetstat">calcwidth.hetstat</code></td>
<td>
<p>A logical indicating whether text for
heterogeneity statistics should be considered to calculate width
of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.tests">calcwidth.tests</code></td>
<td>
<p>A logical indicating whether text for tests
of overall effect or subgroup differences should be considered to
calculate width of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.subgroup">calcwidth.subgroup</code></td>
<td>
<p>A logical indicating whether text with
subgroup labels should be considered to calculate width of the
column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_calcwidth.addline">calcwidth.addline</code></td>
<td>
<p>A logical indicating whether text for
additional lines should be considered to calculate width of the
column with study labels.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_just">just</code></td>
<td>
<p>Justification of text in all columns but columns with
study labels and additional variables (possible values: &quot;left&quot;,
&quot;right&quot;, &quot;center&quot;).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_just.studlab">just.studlab</code></td>
<td>
<p>Justification of text for study labels
(possible values: &quot;left&quot;, &quot;right&quot;, &quot;center&quot;).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_just.addcols">just.addcols</code></td>
<td>
<p>Justification of text for additional columns
(possible values: &quot;left&quot;, &quot;right&quot;, &quot;center&quot;).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_just.addcols.left">just.addcols.left</code></td>
<td>
<p>Justification of text for additional
columns on left side of forest plot (possible values: &quot;left&quot;,
&quot;right&quot;, &quot;center&quot;). Can be of same length as number of additional
columns on left side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_just.addcols.right">just.addcols.right</code></td>
<td>
<p>Justification of text for additional
columns on right side of forest plot (possible values: &quot;left&quot;,
&quot;right&quot;, &quot;center&quot;). Can be of same length as number of additional
columns on right side of forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_bmj.text">bmj.text</code></td>
<td>
<p>A character string used in the plot with BMJ layout
to label the group specific information.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_bmj.xpos">bmj.xpos</code></td>
<td>
<p>A numeric specifying the horizontal position of the
BMJ label. The value is a so called normalised parent coordinate
in the horizontal direction (see <code><a href="grid.html#topic+unit">unit</a></code>).</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_bmj.sep">bmj.sep</code></td>
<td>
<p>A character string used to separate sample sizes
from number of events or means / standard deviations.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_spacing">spacing</code></td>
<td>
<p>A numeric determining line spacing in a forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_addrow">addrow</code></td>
<td>
<p>A logical value indicating whether an empty row is
printed above study results.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_addrow.overall">addrow.overall</code></td>
<td>
<p>A logical value indicating whether an empty
row is printed above overall meta-analysis results.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_addrow.subgroups">addrow.subgroups</code></td>
<td>
<p>A logical value indicating whether an empty
row is printed between results for subgroups.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_addrows.below.overall">addrows.below.overall</code></td>
<td>
<p>A numeric value indicating how many
empty rows are printed between meta-analysis results and
heterogeneity statistics and test results.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_new">new</code></td>
<td>
<p>A logical value indicating whether a new figure should
be printed in an existing graphics window.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in forest plots. If <code>backtransf = TRUE</code>,
results for <code>sm = "OR"</code> are presented as odds ratios rather
than log odds ratios and results for <code>sm = "ZCOR"</code> are
presented as correlations rather than Fisher's z transformed
correlations, for example.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits for treatment
effects, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.se">digits.se</code></td>
<td>
<p>Minimal number of significant digits for standard
errors.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.stat">digits.stat</code></td>
<td>
<p>Minimal number of significant digits for z- or
t-statistic for test of overall effect.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.pval">digits.pval</code></td>
<td>
<p>Minimal number of significant digits for p-value
of overall treatment effect.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.pval.q">digits.pval.Q</code></td>
<td>
<p>Minimal number of significant digits for
p-value of heterogeneity test.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.q">digits.Q</code></td>
<td>
<p>Minimal number of significant digits for
heterogeneity statistic Q.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.tau2">digits.tau2</code></td>
<td>
<p>Minimal number of significant digits for
between-study variance.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.tau">digits.tau</code></td>
<td>
<p>Minimal number of significant digits for square
root of between-study variance.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.i2">digits.I2</code></td>
<td>
<p>Minimal number of significant digits for I-squared
statistic.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.weight">digits.weight</code></td>
<td>
<p>Minimal number of significant digits for
weights.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.mean">digits.mean</code></td>
<td>
<p>Minimal number of significant digits for means;
only applies to <code><a href="#topic+metacont">metacont</a></code> objects.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.sd">digits.sd</code></td>
<td>
<p>Minimal number of significant digits for standard
deviations; only applies to <code><a href="#topic+metacont">metacont</a></code> objects.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.cor">digits.cor</code></td>
<td>
<p>Minimal number of significant digits for
correlations; only applies to <code><a href="#topic+metacor">metacor</a></code> objects.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.time">digits.time</code></td>
<td>
<p>Minimal number of significant digits for times;
only applies to <code><a href="#topic+metainc">metainc</a></code> and <code><a href="#topic+metarate">metarate</a></code>
objects.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.n">digits.n</code></td>
<td>
<p>Minimal number of significant digits for sample
sizes.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.event">digits.event</code></td>
<td>
<p>Minimal number of significant digits for event
numbers.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.te">digits.TE</code></td>
<td>
<p>Minimal number of significant digits for list
element 'TE'.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.addcols">digits.addcols</code></td>
<td>
<p>A vector or scalar with minimal number of
significant digits for additional columns.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.addcols.right">digits.addcols.right</code></td>
<td>
<p>A vector or scalar with minimal number
of significant digits for additional columns on right side of
forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_digits.addcols.left">digits.addcols.left</code></td>
<td>
<p>A vector or scalar with minimal number
of significant digits for additional columns on left side of
forest plot.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_scientific.pval">scientific.pval</code></td>
<td>
<p>A logical specifying whether p-values should
be printed in scientific notation, e.g., 1.2345e-01 instead of
0.12345.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_zero.pval">zero.pval</code></td>
<td>
<p>A logical specifying whether p-values should be
printed with a leading zero.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_jama.pval">JAMA.pval</code></td>
<td>
<p>A logical specifying whether p-values for test of
overall effect should be printed according to JAMA reporting
standards.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="forest.meta_+3A_...">...</code></td>
<td>
<p>Additional graphical arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A forest plot, also called confidence interval plot, is drawn in
the active graphics window. The forest functions in R package
<b>meta</b> are based on the grid graphics system. Resize the
graphics windows if the forest plot is too large or too small for
the graphics window. Alternatively, save the forest plot in a file.
</p>


<h4>Saving forest plots</h4>

<p>A forest plot can be directly stored in a file using argument
<code>file</code> or specifying the R function for the graphics device
driver using argument <code>func.gr</code>, e.g., <code><a href="grDevices.html#topic+pdf">pdf</a></code>. If
only the filename is provided, the extension is checked and matched
against the most common graphics device drivers.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Extension</b> </td><td style="text-align: left;"> <b>Graphics device</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.pdf</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+pdf">pdf</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.ps</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+postscript">postscript</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.svg</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+svg">svg</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.bmp</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+bmp">bmp</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.jpg</code> / <code>.jpeg</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+jpeg">jpeg</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.png</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+png">png</a></code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>.tif</code> / <code>.tiff</code> </td><td style="text-align: left;"> <code><a href="grDevices.html#topic+tiff">tiff</a></code>
</td>
</tr>

</table>

<p>The height of the graphics device is automatically determined if
the forest plot is saved to a file. Argument <code>rows.gr</code> can be
used to increase or decrease the number of rows shown in the forest
plot (either to show missing information or to remove
whitespace). The width of the graphics device can be specified with
argument <code>width</code>, see, for example, <code><a href="grDevices.html#topic+pdf">pdf</a></code> or
<code><a href="grDevices.html#topic+jpeg">jpeg</a></code>. Other arguments of graphics device functions
can be provided as a list in argument <code>args.gr</code>.
</p>
<p>Alternatively, the (resized) graphics window can be stored to a
file using either <code><a href="grDevices.html#topic+dev.copy2eps">dev.copy2eps</a></code> or
<code><a href="grDevices.html#topic+dev.copy2pdf">dev.copy2pdf</a></code>. It is also possible to manually create
a file using, for example, <code><a href="grDevices.html#topic+pdf">pdf</a></code>, <code><a href="grDevices.html#topic+png">png</a></code>, or
<code><a href="grDevices.html#topic+svg">svg</a></code> and to specify the width and height of the
graphic (see Examples).
</p>



<h4>Default layout for studies and pooled effects</h4>

<p>By default, treatment estimates and confidence intervals are
plotted in the following way:
</p>

<ul>
<li><p> For an individual study, a square with treatment estimate in
the center and confidence interval as line extending either side
of the square (<code>type.study = "square"</code>)
</p>
</li>
<li><p> For meta-analysis results, a diamond with treatment estimate
in the center and right and left side corresponding to lower and
upper confidence limits (<code>type.common = "diamond"</code>,
<code>type.random = "diamond"</code>, and <code>type.subgroup = "diamond"</code>)
</p>
</li></ul>

<p>In a forest plot, size of the squares typically reflects the precision of
individual treatment estimates based either on the common effect
(<code>weight.study = "common"</code>) or random effects meta-analysis
(<code>weight.study = "random"</code>). Information from meta-analysis object
<code>x</code> is utilised if argument <code>weight.study</code> is missing. Weights
from the common effect model are used if argument <code>x$common</code> is
<code>TRUE</code>; weights from the random effects model are used if argument
<code>x$random</code> is <code>TRUE</code> and <code>x$common</code> is <code>FALSE</code>.
The same square sizes are used if <code>weight.study = "same"</code>.
</p>
<p>A prediction interval for treatment effect of a new study (Higgins
et al., 2009) is given in the forest plot if arguments
<code>prediction</code> and <code>random</code> are <code>TRUE</code>. For
graphical presentation of prediction intervals the approach by
Guddat et al. (2012) is used.
</p>



<h4>Columns printed on left side of forest plot</h4>

<p>Argument <code>leftcols</code> can be used to specify columns which are
printed on the left side of the forest plot. By default, i.e. if
argument <code>leftcols</code> is <code>NULL</code> and <code>layout = "meta"</code>,
and depending on the class of the meta-analysis object (which is
defined by the R function used to generate the object) a different
set of <em><b>columns</b></em> is printed <em><b>on the left
side of the forest plot</b></em>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Function</b> </td><td style="text-align: left;"> <b>Value of argument leftcols</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metabin">metabin</a></code> </td><td style="text-align: left;"> <code>c("studlab", "event.e", "n.e",
  "event.c", "n.c")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metacont">metacont</a></code> </td><td style="text-align: left;"> <code>c("studlab", "n.e", "mean.e",
  "sd.e", "n.c", "mean.c", "sd.c")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metacor">metacor</a></code> </td><td style="text-align: left;"> <code>c("studlab", "n")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metagen">metagen</a></code> </td><td style="text-align: left;"> <code>c("studlab", "TE", "seTE")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metainc">metainc</a></code> </td><td style="text-align: left;"> <code>c("studlab", "event.e", "time.e",
  "event.c", "time.c")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metamean">metamean</a></code> </td><td style="text-align: left;"> <code>c("studlab", "n", "mean", "sd")</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metaprop">metaprop</a></code> </td><td style="text-align: left;"> <code>c("studlab", "event", "n")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metarate">metarate</a></code> </td><td style="text-align: left;"> <code>c("studlab", "event", "time", "n")</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metacum">metacum</a></code> </td><td style="text-align: left;"> <code>"studlab"</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+metainf">metainf</a></code> </td><td style="text-align: left;"> <code>"studlab"</code>
</td>
</tr>

</table>

<p>For three-level models, the cluster variable is printed next to the
study labels (value <code>"cluster"</code> in argument <code>leftcols</code>).
</p>
<p>By default, study labels and labels for pooled estimates and
heterogeneity statistics will be printed in the first column on the
left side of the forest plot. The character string <code>"studlab"</code>
is used to identify study labels as this is the name of the list
element of a meta-analysis object.
</p>
<p>If the character string <code>"studlab"</code> is not provided in
<code>leftcols</code> and <code>rightcols</code>, the first <em>additional</em>
variable specified by the user is used as study labels (and labels
for pooled estimates are printed in this column). Additional
variables are any variables not mentioned in the section on
predefined column names below. For example, <code>leftcols =
"studlab"</code> and <code>leftcols = "study"</code> would result in the same
forest plot if the variable <code>"study"</code> was used in the command
to conduct the meta-analysis. If no additional variable is provided
by the user, no study labels will be printed.
</p>



<h4>Overlapping information on left side of forest plot</h4>

<p>Depending on the number of columns printed on the left side of the
forest plot, information on heterogeneity measures or statistical
tests (see below) can be overlapping with the x-axis. Argument
<code>addrows.below.overall</code> can be used to specify the number of
empty rows that are printed between meta-analysis results and
information on heterogeneity measures and statistical tests. By
default, no additional rows are added to the forest plot. If
<code>addrows.below.overall = NULL</code>, the function tries to add a
sufficient number of empty rows to prevent overlapping
text. Another possibility is to manually increase the space between
the columns on the left side (argument <code>colgap.left</code>) or
between the columns on the left side and the forest plot (argument
<code>colgap.forest.left</code>).
</p>



<h4>Columns printed on right side of forest plot</h4>

<p>Argument <code>rightcols</code> can be used to
specify columns which are printed on the right side of the
forest plot. If argument <code>rightcols</code> is
<code>FALSE</code>, no columns will be printed on the right side. By
default, i.e. if argument <code>rightcols</code> is
<code>NULL</code> and <code>layout = "meta"</code>, the following
<em><b>columns</b></em> will be printed <em><b>on the right side
of the forest plot</b></em>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Meta-analysis results</b> </td><td style="text-align: left;"> <b>Value of argument
rightcols</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
No summary </td><td style="text-align: left;"> <code>c("effect", "ci")</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
Only common effect model </td><td style="text-align: left;"> <code>c("effect", "ci", "w.common")</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
Only random effects model </td><td style="text-align: left;"> <code>c("effect", "ci", "w.random")</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
Both models </td><td style="text-align: left;"> <code>c("effect", "ci", "w.common", "w.random")</code>
</td>
</tr>

</table>

<p>By default, estimated treatment effect and corresponding confidence
interval will be printed. Depending on arguments <code>common</code> and
<code>random</code>, weights of the common effect and/or random effects
model will be given too.
</p>
<p>For an object of class <code><a href="#topic+metacum">metacum</a></code> or
<code><a href="#topic+metainf">metainf</a></code> the following columns will be printed:
<code>c("effect", "ci", "pval", "tau2", "tau", "I2")</code>. This
information corresponds to the printout with
<code><a href="#topic+print.meta">print.meta</a></code>.
</p>



<h4>Predefined column names</h4>

<p>The arguments <code>leftlabs</code> and <code>rightlabs</code> can be used to
specify column headings which are printed on left or right side of
the forest plot. For certain columns predefined labels exist which
are used by default, i.e., if arguments <code>leftlabs</code> and
<code>rightlabs</code> are <code>NULL</code>:
</p>

<table>
<tr>
 <td style="text-align: right;">
Column: </td><td style="text-align: center;"> <code>studlab</code> </td><td style="text-align: center;"> <code>TE</code> </td><td style="text-align: center;"> <code>seTE</code> </td><td style="text-align: center;">
  <code>cluster</code> </td><td style="text-align: center;"> <code>n.e</code> </td><td style="text-align: center;"> <code>n.c</code> </td>
</tr>
<tr>
 <td style="text-align: right;"> 
Label: </td><td style="text-align: center;"> "Study" </td><td style="text-align: center;"> "TE" </td><td style="text-align: center;"> "seTE" </td><td style="text-align: center;"> "Cluster" </td><td style="text-align: center;">
  "Total" </td><td style="text-align: center;"> "Total" </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>
<tr>
 <td style="text-align: right;">
Column: </td><td style="text-align: center;"> <code>n</code> </td><td style="text-align: center;"> <code>event.e</code> </td><td style="text-align: center;"> <code>event.c</code> </td><td style="text-align: center;">
  <code>event</code> </td><td style="text-align: center;"> <code>mean.e</code> </td><td style="text-align: center;"> <code>mean.c</code> </td>
</tr>
<tr>
 <td style="text-align: right;">
Label: </td><td style="text-align: center;"> "Total" </td><td style="text-align: center;"> "Events" </td><td style="text-align: center;"> "Events" </td><td style="text-align: center;"> "Events" </td><td style="text-align: center;">
  "Mean" </td><td style="text-align: center;"> "Mean" </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>
<tr>
 <td style="text-align: right;">
Column: </td><td style="text-align: center;"> <code>sd.e</code> </td><td style="text-align: center;"> <code>sd.c</code> </td><td style="text-align: center;"> <code>time.e</code>
  </td><td style="text-align: center;"> <code>time.c</code> </td><td style="text-align: center;"> <code>effect</code> </td><td style="text-align: center;"> </td>
</tr>
<tr>
 <td style="text-align: right;">
Label: </td><td style="text-align: center;"> "SD" </td><td style="text-align: center;"> "SD" </td><td style="text-align: center;"> "Time" </td><td style="text-align: center;"> "Time" </td><td style="text-align: center;">
  <code>x$sm</code> </td><td style="text-align: center;"> </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>
<tr>
 <td style="text-align: right;">
Column: </td><td style="text-align: center;"> <code>ci</code> </td><td style="text-align: center;"> <code>effect.ci</code> </td><td style="text-align: center;">
  <code>w.common</code> </td><td style="text-align: center;"> <code>w.random</code> </td><td style="text-align: center;"> </td><td style="text-align: center;"> </td>
</tr>
<tr>
 <td style="text-align: right;">
Label: </td><td style="text-align: center;"> <code>x$level</code>"%-CI" </td><td style="text-align: center;"> <em>effect+ci</em> </td><td style="text-align: center;">
  "W(common)" </td><td style="text-align: center;"> "W(random)" </td><td style="text-align: center;"> </td><td style="text-align: center;">
</td>
</tr>

</table>

<p>For other columns, the column name will be used as a label if no
column label is defined. It is possible to only provide labels for
new columns (see Examples). Otherwise the length of <code>leftlabs</code>
and <code>rightlabs</code> must be the same as the number of printed
columns. The value <code>NA</code> can be used to specify columns which
should use default labels (see Examples).
</p>
<p>In pairwise meta-analysis comparing two groups (i.e.,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, and <code><a href="#topic+metagen">metagen</a></code> depending on the
outcome), arguments <code>label.e</code> and <code>label.c</code> are used to
label columns belonging to the two treatment groups. By default,
labels defined in the meta-analysis object are used. The columns
where treatment labels are attached can be changed using arguments
<code>label.e.attach</code> and <code>label.c.attach</code>.
</p>



<h4>Risk of bias assessment</h4>

<p>A risk of bias (RoB) assessment can be shown in the forest plot by
either using a meta-analysis object with an RoB assessment as main
input or providing a suitable object created with
<code><a href="#topic+rob">rob</a></code>. Argument <code>rob = FALSE</code> can be used to
suppress the print of the risk of bias information.
</p>
<p>RoB assessments are shown as the only information on the right side
of the forest plot. Thus, arguments <code>rightcols</code> and
<code>rightlabs</code> should not be used. Predefined columns shown by
default on the right side of a forest plot will be moved to the
left side.
</p>



<h4>Information on heterogeneity and statistical tests</h4>

<p>Argument <code>hetstat</code> can be a character string to specify where
to print heterogeneity information:
</p>

<ul>
<li><p> row with results for common effect model (<code>hetstat =
"common"</code>),
</p>
</li>
<li><p> row with results for random effects model (<code>hetstat =
"random"</code>).
</p>
</li></ul>

<p>Otherwise, information on heterogeneity measures is printed below
the meta-analysis results if argument <code>overall.hetstat = TRUE</code>
(default). The heterogeneity measures to print can be specified
(see list of arguments following <code>overall.hetstat</code>).
</p>
<p>In addition, the following arguments can be used to print results
for various statistical tests:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Statistical test</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.overall.common</code> </td><td style="text-align: left;"> Test for overall effect (common
  effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.overall.random</code> </td><td style="text-align: left;"> Test for overall effect (random
  effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.effect.subgroup.common</code> </td><td style="text-align: left;"> Test for effect in subgroup
  (CE model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.effect.subgroup.random</code> </td><td style="text-align: left;"> Test for effect in subgroup
  (RE model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.subgroup.common</code> </td><td style="text-align: left;"> Test for subgroup differences (CE
  model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.subgroup.random</code> </td><td style="text-align: left;"> Test for subgroup differences (RE
  model)
</td>
</tr>

</table>

<p>By default, these arguments are <code>FALSE</code> with exception of
tests for subgroup differences which are <code>TRUE</code>. R function
<code><a href="#topic+settings.meta">settings.meta</a></code> can be used to change this default for
the entire R session. For example, use the following command to
always print results of tests for an overall effect:
<code>settings.meta(test.overall = TRUE)</code>.
</p>



<h4>Flexible printing of subgroup results</h4>

<p>Argument <code>subgroup</code> determines whether summary results are
printed for subgroups. A logical vector of length equal to the
number of subgroups can be provided to determine which subgroup
summaries are printed. By default, only subgroup results based on
at least two studies are printed which is identical to use argument
<code>subgroup = k.w &gt; 1</code>. The order of the logical vector
corresponds to the order of subgroups in list element 'subgroup.levels' of a
meta-analysis object. Argument <code>subgroup = k.w &gt;= 1</code> can be
used to show results for all subgroups (including those with a
single study).
</p>
<p>The following arguments can be used in a similar way:
</p>

<ul>
<li> <p><code>subgroup.hetstat</code> (heterogeneity statistic in
subgroups),
</p>
</li>
<li> <p><code>common.subgroup</code> (common effect estimates in
subgroups),
</p>
</li>
<li> <p><code>random.subgroup</code> (random effects estimates in
subgroups),
</p>
</li>
<li> <p><code>prediction.subgroup</code> (prediction interval in
subgroups),
</p>
</li>
<li> <p><code>test.effect.subgroup</code> (test for effect in subgroups),
</p>
</li>
<li> <p><code>test.effect.subgroup.common</code> (test for effect in
subgroups, common effect model),
</p>
</li>
<li> <p><code>test.effect.subgroup.random</code> (test for effect in
subgroups, random effects model).
</p>
</li></ul>




<h4>Additional general settings</h4>

<p>Arguments <code>text.common</code>, <code>text.random</code>, and
<code>text.predict</code> can be used to change the label to identify
overall results (common effect and random effects model as well as
prediction interval). By default the following text is printed:
</p>

<ul>
<li><p> &quot;Common effect model&quot; (argument <code>text.common</code>)
</p>
</li>
<li><p> &quot;Random effects model&quot; (<code>text.random</code>)
</p>
</li>
<li><p> &quot;Prediction interval&quot; (<code>text.predict</code>)
</p>
</li></ul>

<p>If confidence interval levels are different for individual studies,
meta-analysis, and prediction interval (arguments <code>level</code>,
<code>level.ma</code>, <code>level.predict</code> in meta-analysis functions,
e.g., <code><a href="#topic+metabin">metabin</a></code>), additional information is printed,
e.g., &quot; (99%-CI)&quot; for a 99% confidence interval in the
meta-analysis.
</p>
<p>Argument <code>pscale</code> can be used to rescale single proportions or
risk differences, e.g., <code>pscale = 1000</code> means that proportions
are expressed as events per 1000 observations. This is useful in
situations with (very) low event probabilities.
</p>
<p>Argument <code>irscale</code> can be used to rescale single rates or rate
differences, e.g., <code>irscale = 1000</code> means that rates are
expressed as events per 1000 time units, e.g., person-years. This is
useful in situations with (very) low rates. Argument <code>irunit</code>
can be used to specify the time unit used in individual studies
(default: &quot;person-years&quot;). This information is printed in summaries
and forest plots if argument <code>irscale</code> is not equal to 1.
</p>



<h4>Forest plots in RevMan5 layout</h4>

<p>If argument <code>layout = "RevMan5"</code> (and arguments <code>leftcols</code> and
<code>rightcols</code> are <code>NULL</code>), the layout for forest plots used for
Cochrane reviews (which are generated with Review Manager 5,
<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>)
is reproduced:
</p>

<ol>
<li><p> All columns are printed on the left side of the forest plot
(see arguments <code>leftcols</code> and <code>rightcols</code>)
</p>
</li>
<li><p> Tests for overall effect and subgroup differences are printed
(<code>test.overall</code>, <code>test.effect.subgroup</code>,
<code>test.subgroup</code>)
</p>
</li>
<li><p> Diamonds representing meta-analysis results are printed in
black (<code>diamond.common</code>, <code>diamond.random</code>)
</p>
</li>
<li><p> Colour of squares depends on the meta-analysis object
(<code>col.square</code>, <code>col.square.lines</code>)
</p>
</li>
<li><p> Information on effect measure and meta-analysis method is
printed above the forest plot (<code>smlab</code>)
</p>
</li>
<li><p> Label &quot;Study or Subgroup&quot; is printed for meta-analysis with
subgroups (<code>leftlabs</code>)
</p>
</li></ol>




<h4>Forest plots in JAMA layout</h4>

<p>If argument <code>layout = "JAMA"</code> (and arguments <code>leftcols</code> and
<code>rightcols</code> are <code>NULL</code>), instructions for authors of the
<em>Journal of the American Medical Association</em>, see
<a href="https://jamanetwork.com/journals/jama/pages/instructions-for-authors/">https://jamanetwork.com/journals/jama/pages/instructions-for-authors/</a>,
are taken into account:
</p>

<ol>
<li><p> Graph labels on right and left side are printed in bold font
at top of forest plot (see arguments <code>bottom.lr</code> and
<code>ff.lr</code>)
</p>
</li>
<li><p> Information on effect measure and level of confidence
interval is printed at bottom of forest plot (<code>xlab</code>)
</p>
</li>
<li><p> Tests for overall effect are printed (<code>test.overall</code>)
</p>
</li>
<li><p> Diamonds representing meta-analysis results are printed in
lightblue (<code>diamond.common</code>, <code>diamond.random</code>)
</p>
</li>
<li><p> Squares representing individual study results are printed in
darkblue (<code>col.square</code>, <code>col.square.lines</code>)
</p>
</li>
<li><p> Between-study variance <code class="reqn">\tau^2</code> is not printed
</p>
</li>
<li><p> Empty rows are omitted (<code>addrow</code>, <code>addrow.overall</code>,
<code>addrow.subgroups</code>)
</p>
</li>
<li><p> Label &quot;Source&quot; is printed instead of &quot;Study&quot; (<code>leftlabs</code>)
</p>
</li>
<li><p> P-values are printed without leading zeros (<code>zero.pval</code>)
</p>
</li>
<li><p> P-values are rounded to three digits (for 0.001 &lt; p <code class="reqn">\le</code>
0.01) or two digits (p &gt; 0.01) (<code>JAMA.pval</code>)
</p>
</li></ol>

<p>Study labels according to JAMA guidelines can be generated using
<code><a href="#topic+labels.meta">labels.meta</a></code>.
</p>



<h4>Forest plots showing results of subgroups</h4>

<p>The following changes are conducted if argument
<code>layout = "subgroup"</code> (and arguments <code>leftcols</code> and
<code>rightcols</code> are <code>NULL</code>) and a subgroup analysis was
conducted:
</p>

<ol>
<li><p> Individual study results are omitted (see argument
<code>study.results</code>)
</p>
</li>
<li><p> Total number of observations is not printed
(<code>pooled.totals</code>)
</p>
</li>
<li><p> Label &quot;Subgroup&quot; is printed instead of &quot;Study&quot;
(<code>leftlabs</code>)
</p>
</li></ol>




<h3>Note</h3>

<p>R function <code>.forestArgs</code> generates a character vector with all
arguments of <code>forest.meta</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Guddat C, Grouven U, Bender R, Skipka G (2012):
A note on the graphical presentation of prediction intervals in
random-effects meta-analyses.
<em>Systematic Reviews</em>,
<b>1</b>, 34
</p>
<p>Higgins JPT, Thompson SG, Spiegelhalter DJ (2009): 
A re-evaluation of random-effects meta-analysis.
<em>Journal of the Royal Statistical Society: Series A</em>,
<b>172</b>, 137-59
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+forest.metabind">forest.metabind</a></code>,
<code><a href="#topic+settings.meta">settings.meta</a></code>, <code><a href="#topic+labels.meta">labels.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = c(41, 47, 51, 59),
  sm = "RR", method = "I",
  studlab = paste(author, year))


## Not run: 
# Do standard (symmetric) forest plot
#
forest(m1)

## End(Not run)

# Layout of forest plot similar to Review Manager 5
#
# Furthermore, add labels on both sides of forest plot and
# prediction interval
#
forest(m1, layout = "RevMan5", common = FALSE,
  label.right = "Favours control", col.label.right = "red",
  label.left = "Favours experimental", col.label.left = "green",
  prediction = TRUE)


## Not run: 
# Create PDF files with the forest plot
#
# - specify filename (R function pdf() is used due to extension .pdf)
# - height of the figure is automatically determined
# - width is set to 10 inches
forest(m1, file = "forest-m1-1.pdf", width = 10)
#
# - specify graphics device function
#   (filename "Rplots.pdf" used, see help page of R function pdf())
# - height of the figure is automatically determined
# - width is set to 10 inches
# - set title for PDF file
# - set background of forest plot
forest(m1, func.gr = pdf, width = 10,
  args.gr = list(title = "My Forest Plot", bg = "green"))
#
# - manually specify the height of the figure
pdf("forest-m1-2.pdf", width = 10, height = 3)
forest(m1)
dev.off()

# Define equivalence limits: 0.75 and 1 / 0.75
#
forest(m1, layout = "RevMan5", common = FALSE,
  lower.equi = 0.75, upper.equi = 1 / 0.75, fill.equi = "lightgray")

# Fill areas with beneficial and detrimental effects
#
forest(m1, layout = "RevMan5", common = FALSE,
  lower.equi = 0.75, upper.equi = 1 / 0.75,
  fill.lower.equi = c("green", "lightgray"),
  fill.upper.equi = c("lightgray", "red"))

# Define thresholds for small, moderate and large effects
# and use hcl.colors() to define colours to fill areas
#
thresholds &lt;- c(0.25, 0.5, 0.75)
n.cols &lt;- length(thresholds) + 1
forest(m1, layout = "RevMan5", common = FALSE,
  label.right = "Undesirable effect", 
  label.left = "Desirable effect", 
  lty.equi = 3, col.equi = "darkgray",
  lower.equi = thresholds, upper.equi = 1 / rev(thresholds),
  fill.lower.equi =
    hcl.colors(n.cols, palette = "Blues 2", alpha = 0.6),
  fill.upper.equi =
    hcl.colors(n.cols, palette = "Oranges", alpha = 0.6, rev = TRUE))

# Conduct subgroup meta-analysis
#
m2 &lt;- update(m1,
  subgroup = ifelse(year &lt; 1987, "Before 1987", "1987 and later"),
  print.subgroup.name = FALSE)

# Show summary results for subgroups with at least two studies
#
forest(m2, sortvar = -TE, random = FALSE)

# Show results for all subgroups
#
forest(m2, sortvar = -TE, random = FALSE, subgroup = k.w &gt;= 1)

# Forest plot specifying argument xlim
#
forest(m1, xlim = c(0.01, 10))

# Print results of test for overall effect
#
forest(m1, test.overall.common = TRUE, test.overall.random = TRUE)

# Forest plot with 'classic' layout used in R package meta,
# version &lt; 1.6-0
#
forest(m1, col.square = "black", hetstat = FALSE)

# Change set of columns printed on left side of forest plot
# (resulting in overlapping text)
#
forest(m1, random = FALSE, leftcols = "studlab")

# Use argument 'calcwidth.hetstat' to consider text for heterogeneity
# measures in width of column with study labels
#
forest(m1, random = FALSE, leftcols = "studlab",
  calcwidth.hetstat = TRUE)

# Use argument 'addrows.below.overall' to manually add two empty
# rows
#
forest(m1, random = FALSE, leftcols = "studlab", addrows = 2)

# Do not print columns on right side of forest plot
#
forest(m1, rightcols = FALSE)

# Change study label to "Author"
#
forest(m1, random = FALSE, leftlabs = c("Author", NA, NA, NA, NA))

# Just give effect estimate and 95% confidence interval on right
# side of forest plot (in one column)
#
forest(m1, rightcols = "effect.ci")

# Just give effect estimate and 95% confidence interval on right
# side of forest plot
#
forest(m1, rightcols = c("effect", "ci"))

# 1. Change order of columns on left side
# 2. Attach labels to columns 'event.e' and 'event.c' instead of
#    columns 'n.e' and 'n.c'
#
forest(m1,
  leftcols = c("studlab", "n.e", "event.e", "n.c", "event.c"),
  label.e.attach = "event.e", label.c.attach = "event.c")

# Specify column labels only for variables 'year' and 'author'
# (and define digits for additional variables)
#
forest(m1,
  leftcols = c("studlab", "event.e", "n.e", "event.c", "n.c", "author", "year"),
  leftlabs = c("Author", "Year of Publ"))

# Center text in all columns
#
forest(m1,
  leftcols = c("studlab", "event.e", "n.e", "event.c", "n.c",
               "author", "year"),
  leftlabs = c("Author", "Year of Publ"), hetstat = FALSE,
  just = "center", just.addcols = "center", just.studlab = "center")

# Same result
#
forest(m1,
  leftcols = c("studlab", "event.e", "n.e", "event.c", "n.c",
             "author", "year"),
  leftlabs = c("Author", "Year of Publ"), hetstat = FALSE,
  just = "c", just.addcols = "c", just.studlab = "c")

# Change some fontsizes and fontfaces
#
forest(m1,
  fs.study = 10, ff.study = "italic",
  fs.study.label = 11, ff.study.label = "bold",
  fs.axis = 5, ff.axis = "italic",
  ff.smlab = "bold.italic",
  ff.common = "plain", ff.hetstat = "plain")

# Change some colours
#
forest(m1,
  col.diamond = "green", col.diamond.lines = "red",
  col.study = c("green", "blue", "red", "orange"),
  col.square = "pink", col.square.lines = "black")

# Sort by weight in common effect model
#
forest(m1, sortvar = w.common, random = FALSE)

# Sort by decreasing weight in common effect model
#
forest(m1, sortvar = -w.common, random = FALSE)

# Sort by size of treatment effect
#
forest(m1, sortvar = TE, random = FALSE)

# Sort by size of treatment effect
#
forest(m1, sortvar = -TE, random = FALSE)

# Sort by decreasing year of publication
#
forest(m1, sortvar = -year, random = FALSE)

# Print results of test for subgroup differences (random effects
# model)
#
forest(m2, sortvar = -TE, common = FALSE)

# Print only subgroup results
#
forest(m2, layout = "subgroup")

# Print only subgroup results (and consider text for tests of
# subgroup differences in width of subgroup column)
#
forest(m2, layout = "subgroup", calcwidth.tests = TRUE)

# Print only subgroup results (and consider text for heterogeneity
# in width of subgroup column)
#
forest(m2, layout = "subgroup", calcwidth.hetstat = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='forest.metabind'>Forest plot to display the result of a meta-analysis</h2><span id='topic+forest.metabind'></span>

<h3>Description</h3>

<p>Draws a forest plot in the active graphics window (using grid
graphics system).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metabind'
forest(
  x,
  leftcols,
  leftlabs,
  rightcols = c("effect", "ci"),
  rightlabs,
  common = x$common,
  random = x$random,
  overall = x$overall,
  subgroup = FALSE,
  hetstat = FALSE,
  overall.hetstat = x$overall.hetstat,
  prediction = x$prediction,
  lab.NA = "",
  col.square = gs("col.square"),
  col.square.lines = col.square,
  col.circle = gs("col.circle"),
  col.circle.lines = col.circle,
  col.diamond = gs("col.diamond"),
  col.diamond.common = col.diamond,
  col.diamond.random = col.diamond,
  col.diamond.lines = gs("col.diamond.lines"),
  col.diamond.lines.common = col.diamond.lines,
  col.diamond.lines.random = col.diamond.lines,
  col.predict = gs("col.predict"),
  col.predict.lines = gs("col.predict.lines"),
  type.common = NULL,
  type.random = NULL,
  type.predict = NULL,
  digits = gs("digits.forest"),
  digits.se = gs("digits.se"),
  digits.stat = gs("digits.stat"),
  digits.pval = max(gs("digits.pval") - 2, 2),
  digits.pval.Q = max(gs("digits.pval.Q") - 2, 2),
  digits.Q = gs("digits.Q"),
  digits.tau2 = gs("digits.tau2"),
  digits.tau = gs("digits.tau"),
  digits.I2 = max(gs("digits.I2") - 1, 0),
  scientific.pval = gs("scientific.pval"),
  big.mark = gs("big.mark"),
  print.subgroup.labels = x$with.subgroups,
  addrow.subgroups = print.subgroup.labels,
  smlab,
  calcwidth.pooled = overall,
  warn.deprecated = gs("warn.deprecated"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="forest.metabind_+3A_x">x</code></td>
<td>
<p>An object of class <code><a href="#topic+metabind">metabind</a></code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_leftcols">leftcols</code></td>
<td>
<p>A character vector specifying (additional) columns
to be plotted on the left side of the forest plot or a logical
value (see Details).</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_leftlabs">leftlabs</code></td>
<td>
<p>A character vector specifying labels for
(additional) columns on left side of the forest plot (see
Details).</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_rightcols">rightcols</code></td>
<td>
<p>A character vector specifying (additional) columns
to be plotted on the right side of the forest plot or a logical
value (see Details).</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_rightlabs">rightlabs</code></td>
<td>
<p>A character vector specifying labels for
(additional) columns on right side of the forest plot (see
Details).</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_common">common</code></td>
<td>
<p>A logical indicating whether common effect estimates
should be plotted.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_random">random</code></td>
<td>
<p>A logical indicating whether random effects estimates
should be plotted.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be plotted. This argument is useful in a meta-analysis
with subgroups if summaries should only be plotted on group
level.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_subgroup">subgroup</code></td>
<td>
<p>A logical indicating whether subgroup results
should be shown in forest plot. This argument is useful in a
meta-analysis with subgroups if summaries should not be plotted
on group level.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_hetstat">hetstat</code></td>
<td>
<p>Either a logical value indicating whether to print
results for heterogeneity measures at all or a character string
(see Details).</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether prediction
interval(s) should be printed.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_lab.na">lab.NA</code></td>
<td>
<p>A character string to label missing values.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.square">col.square</code></td>
<td>
<p>The colour for squares reflecting study's weight
in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.square.lines">col.square.lines</code></td>
<td>
<p>The colour for the outer lines of squares
reflecting study's weight in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.circle">col.circle</code></td>
<td>
<p>The colour for circles reflecting study weights
in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.circle.lines">col.circle.lines</code></td>
<td>
<p>The colour for the outer lines of circles
reflecting study's weight in the meta-analysis.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.diamond">col.diamond</code></td>
<td>
<p>The colour of diamonds representing the results
for common effect and random effects models.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.diamond.common">col.diamond.common</code></td>
<td>
<p>The colour of diamonds for common effect
estimates.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.diamond.random">col.diamond.random</code></td>
<td>
<p>The colour of diamonds for random effects
estimates.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.diamond.lines">col.diamond.lines</code></td>
<td>
<p>The colour of the outer lines of diamonds
representing the results for common effect and random effects
models.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.diamond.lines.common">col.diamond.lines.common</code></td>
<td>
<p>The colour of the outer lines of
diamond for common effect estimates.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.diamond.lines.random">col.diamond.lines.random</code></td>
<td>
<p>The colour of the outer lines of
diamond for random effects estimates.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.predict">col.predict</code></td>
<td>
<p>Background colour of prediction intervals.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_col.predict.lines">col.predict.lines</code></td>
<td>
<p>Colour of outer lines of prediction
intervals.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_type.common">type.common</code></td>
<td>
<p>A character string or vector specifying how to
plot common effect estimates.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_type.random">type.random</code></td>
<td>
<p>A character string or vector specifying how to
plot random effects estimates.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_type.predict">type.predict</code></td>
<td>
<p>A character string or vector specifying how to
plot prediction intervals.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits for treatment
effects, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.se">digits.se</code></td>
<td>
<p>Minimal number of significant digits for standard
errors, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.stat">digits.stat</code></td>
<td>
<p>Minimal number of significant digits for z- or
t-statistic for test of overall effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.pval">digits.pval</code></td>
<td>
<p>Minimal number of significant digits for p-value
of overall treatment effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.pval.q">digits.pval.Q</code></td>
<td>
<p>Minimal number of significant digits for
p-value of heterogeneity test, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.q">digits.Q</code></td>
<td>
<p>Minimal number of significant digits for
heterogeneity statistic Q, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.tau2">digits.tau2</code></td>
<td>
<p>Minimal number of significant digits for
between-study variance, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.tau">digits.tau</code></td>
<td>
<p>Minimal number of significant digits for square
root of between-study variance, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_digits.i2">digits.I2</code></td>
<td>
<p>Minimal number of significant digits for I-squared
statistic, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_scientific.pval">scientific.pval</code></td>
<td>
<p>A logical specifying whether p-values should
be printed in scientific notation, e.g., 1.2345e-01 instead of
0.12345.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_print.subgroup.labels">print.subgroup.labels</code></td>
<td>
<p>A logical indicating whether subgroup
label should be printed.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_addrow.subgroups">addrow.subgroups</code></td>
<td>
<p>A logical value indicating whether an empty
row is printed between results for subgroups.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_smlab">smlab</code></td>
<td>
<p>A label for the summary measurex (printed at top of
figure).</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_calcwidth.pooled">calcwidth.pooled</code></td>
<td>
<p>A logical indicating whether text for
common effect and random effects model should be considered to
calculate width of the column with study labels.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="forest.metabind_+3A_...">...</code></td>
<td>
<p>Additional graphical arguments (passed on to
<code><a href="#topic+forest.meta">forest.meta</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A forest plot, also called confidence interval plot, is drawn in
the active graphics window. The forest functions in R package
<b>meta</b> are based on the grid graphics system. In order to
print the forest plot, resize the graphics window and either use
<code><a href="grDevices.html#topic+dev.copy2eps">dev.copy2eps</a></code> or <code><a href="grDevices.html#topic+dev.copy2pdf">dev.copy2pdf</a></code>. Another
possibility is to create a file using <code><a href="grDevices.html#topic+pdf">pdf</a></code>,
<code><a href="grDevices.html#topic+png">png</a></code>, or <code><a href="grDevices.html#topic+svg">svg</a></code> and to specify the width
and height of the graphic (see <code><a href="#topic+forest.meta">forest.meta</a></code> examples).
</p>
<p>The arguments <code>leftcols</code> and <code>rightcols</code> can be used to
specify columns which are plotted on the left and right side of the
forest plot, respectively.
</p>
<p>The arguments <code>leftlabs</code> and <code>rightlabs</code> can be used to
specify column headings which are plotted on left and right side of
the forest plot, respectively. For certain columns predefined
labels exist. For other columns, the column name will be used as a
label. It is possible to only provide labels for new columns (see
<code><a href="#topic+forest.meta">forest.meta</a></code> examples). Otherwise the length of
<code>leftlabs</code> and <code>rightlabs</code> must be the same as the number
of printed columns, respectively. The value <code>NA</code> can be used
to specify columns which should use default labels.
</p>
<p>Argument <code>hetstat</code> can be a character string to specify where
to print heterogeneity information:
</p>

<ul>
<li><p> row with results for common effect model (<code>hetstat =
"common"</code>),
</p>
</li>
<li><p> row with results for random effects model (<code>hetstat =
"random"</code>),
</p>
</li>
<li><p> rows with 'study' information (<code>hetstat = "study"</code>).
</p>
</li></ul>

<p>Otherwise, information on heterogeneity is printed in dedicated rows.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+forest.meta">forest.meta</a></code>, <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metacont">metacont</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+metabind">metabind</a></code>, <code><a href="#topic+settings.meta">settings.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)

# Add some (fictitious) grouping variables:
#
Fleiss1993cont$age &lt;- c(55, 65, 55, 65, 55)
Fleiss1993cont$region &lt;- c("Europe", "Europe", "Asia", "Asia", "Europe")

m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")

# Conduct two subgroup analyses
#
mu1 &lt;- update(m1, subgroup = age, bylab = "Age group")
mu2 &lt;- update(m1, subgroup = region, bylab = "Region")

# Combine subgroup meta-analyses and show forest plot with subgroup
# results
#
mb1 &lt;- metabind(mu1, mu2)
mb1
forest(mb1)

</code></pre>

<hr>
<h2 id='funnel.meta'>Funnel plot</h2><span id='topic+funnel.meta'></span><span id='topic+funnel'></span>

<h3>Description</h3>

<p>Draw a funnel plot which can be used to assess small study effects
in meta-analysis. A contour-enhanced funnel plot can also be
produced to assess causes of funnel plot asymmetry.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
funnel(
  x,
  xlim = NULL,
  ylim = NULL,
  xlab = NULL,
  ylab = NULL,
  common = x$common,
  random = x$random,
  axes = TRUE,
  pch = if (!inherits(x, "trimfill")) 21 else ifelse(x$trimfill, 1, 21),
  text = NULL,
  cex = 1,
  lty.common = 2,
  lty.random = 9,
  lwd = 1,
  lwd.common = lwd,
  lwd.random = lwd,
  col = "black",
  bg = "darkgray",
  col.common = "black",
  col.random = "black",
  log,
  yaxis,
  contour.levels = NULL,
  col.contour,
  ref = ifelse(is_relative_effect(x$sm), 1, 0),
  level = if (common | random) x$level else NULL,
  studlab = FALSE,
  cex.studlab = 0.8,
  pos.studlab = 2,
  ref.triangle = FALSE,
  lty.ref = 1,
  lwd.ref = lwd,
  col.ref = "black",
  lty.ref.triangle = 5,
  backtransf = x$backtransf,
  warn.deprecated = gs("warn.deprecated"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="funnel.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min,max) of the plot.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_ylim">ylim</code></td>
<td>
<p>The y limits (min,max) of the plot.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y-axis.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether the common effect
estimate should be plotted.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether the random effects
estimate should be plotted.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_axes">axes</code></td>
<td>
<p>A logical indicating whether axes should be drawn on
the plot.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_pch">pch</code></td>
<td>
<p>The plotting symbol used for individual studies.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_text">text</code></td>
<td>
<p>A character vector specifying the text to be used
instead of plotting symbol.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_cex">cex</code></td>
<td>
<p>The magnification to be used for plotting symbol.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lty.common">lty.common</code></td>
<td>
<p>Line type (common effect estimate).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lty.random">lty.random</code></td>
<td>
<p>Line type (random effects estimate).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lwd">lwd</code></td>
<td>
<p>The line width for confidence intervals (if <code>level</code>
is not <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lwd.common">lwd.common</code></td>
<td>
<p>The line width for common effect estimate (if
<code>common</code> is not <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lwd.random">lwd.random</code></td>
<td>
<p>The line width for random effects estimate (if
<code>random</code> is not <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_col">col</code></td>
<td>
<p>A vector with colour of plotting symbols.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_bg">bg</code></td>
<td>
<p>A vector with background colour of plotting symbols (only
used if <code>pch</code> in <code>21:25</code>).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_col.common">col.common</code></td>
<td>
<p>Colour of line representing common effect estimate.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_col.random">col.random</code></td>
<td>
<p>Colour of line representing random effects
estimate.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_log">log</code></td>
<td>
<p>A character string which contains <code>"x"</code> if the
x-axis is to be logarithmic, <code>"y"</code> if the y-axis is to be
logarithmic and <code>"xy"</code> or <code>"yx"</code> if both axes are to be
logarithmic.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_yaxis">yaxis</code></td>
<td>
<p>A character string indicating which type of weights
are to be used. Either <code>"se"</code>, <code>"invvar"</code>,
<code>"invse"</code>, <code>"size"</code>, <code>"invsqrtsize"</code>, or
<code>"ess"</code>.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_contour.levels">contour.levels</code></td>
<td>
<p>A numeric vector specifying contour levels to
produce contour-enhanced funnel plot.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_col.contour">col.contour</code></td>
<td>
<p>Colour of contours.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_ref">ref</code></td>
<td>
<p>Reference value (null effect) used to produce
contour-enhanced funnel plot.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_level">level</code></td>
<td>
<p>The confidence level utilised in the plot. For the
funnel plot, confidence limits are not drawn if
<code>yaxis="size"</code> or <code>yaxis="invsqrtsize"</code>.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_studlab">studlab</code></td>
<td>
<p>A logical indicating whether study labels should be
printed in the graph. A vector with study labels can also be
provided (must be of same length as <code>x$TE</code> then).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_cex.studlab">cex.studlab</code></td>
<td>
<p>Size of study labels, see argument <code>cex</code> in
<code><a href="graphics.html#topic+text">text</a></code>.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_pos.studlab">pos.studlab</code></td>
<td>
<p>Position of study labels, see argument
<code>pos</code> in <code><a href="graphics.html#topic+text">text</a></code>.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_ref.triangle">ref.triangle</code></td>
<td>
<p>A logical indicating whether approximate
confidence limits should be printed around reference value (null
effect).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lty.ref">lty.ref</code></td>
<td>
<p>Line type (reference value).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lwd.ref">lwd.ref</code></td>
<td>
<p>The line width for the reference value and
corresponding confidence intervals (if <code>ref.triangle</code> is
TRUE and <code>level</code> is not <code>NULL</code>).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_col.ref">col.ref</code></td>
<td>
<p>Colour of line representing reference value.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_lty.ref.triangle">lty.ref.triangle</code></td>
<td>
<p>Line type (confidence intervals of
reference value).</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for relative
summary measures (argument <code>sm</code> equal to <code>"OR"</code>,
<code>"RR"</code>, <code>"HR"</code>, or <code>"IRR"</code>) should be back
transformed in funnel plots. If <code>backtransf=TRUE</code>, results
for <code>sm="OR"</code> are printed as odds ratios rather than log
odds ratios, for example.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="funnel.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).
moment).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A funnel plot (Light &amp; Pillemer, 1984) is drawn in the active
graphics window. If <code>common</code> is TRUE, the estimate of the
common effect model is plotted as a vertical line. Similarly, if
<code>random</code> is TRUE, the estimate of the random effects model is
plotted. If <code>level</code> is not NULL, the corresponding approximate
confidence limits are drawn around the common effect estimate (if
<code>common</code> is TRUE) or the random effects estimate (if
<code>random</code> is TRUE and <code>common</code> is FALSE).
</p>
<p>In the funnel plot, the standard error of the treatment estimates
is plotted on the y-axis by default (<code>yaxis = "se"</code>) which is
likely to be the best choice (Sterne &amp; Egger, 2001). Only exception
is meta-analysis of diagnostic test accuracy studies (Deeks et al.,
2005) where the inverse of the square root of the <em>effective
study size</em> is used (<code>yaxis = "ess"</code>). Other possible choices
for <code>yaxis</code> are <code>"invvar"</code> (inverse of the variance),
<code>"invse"</code> (inverse of the standard error), <code>"size"</code>
(study size), and <code>"invsqrtsize"</code> (1 / sqrt(study size)).
</p>
<p>If argument <code>yaxis</code> is not equal to <code>"size"</code>,
<code>"invsqrtsize"</code> or <code>"ess"</code>, contour-enhanced funnel plots
can be produced (Peters et al., 2008) by specifying the contour
levels (argument <code>contour.levels</code>). By default (argument
<code>col.contour</code> missing), suitable gray levels will be used to
distinguish the contours. Different colours can be chosen by
argument <code>col.contour</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>, Petra
Graham <a href="mailto:pgraham@efs.mq.edu.au">pgraham@efs.mq.edu.au</a>
</p>


<h3>References</h3>

<p>Deeks JJ, Macaskill P, Irwig L (2005):
The performance of tests of publication bias and other sample size
effects in systematic reviews of diagnostic test accuracy was
assessed.
<em>Journal of Clinical Epidemiology</em>,
<b>58</b>:882&ndash;93
</p>
<p>Light RJ &amp; Pillemer DB (1984):
<em>Summing Up. The Science of Reviewing Research</em>.
Cambridge: Harvard University Press
</p>
<p>Peters JL, Sutton AJ, Jones DR, Abrams KR, Rushton L (2008):
Contour-enhanced meta-analysis funnel plots help distinguish
publication bias from other causes of asymmetry.
<em>Journal of Clinical Epidemiology</em>,
<b>61</b>, 991&ndash;6
</p>
<p>Sterne JAC &amp; Egger M (2001):
Funnel plots for detecting bias in meta-analysis: Guidelines on
choice of axis.
<em>Journal of Clinical Epidemiology</em>,
<b>54</b>, 1046&ndash;55
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabias">metabias</a></code>, <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+radial">radial</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = c(41, 47, 51, 59),
  studlab = paste(author, year),
  sm = "RR", method = "I")

oldpar &lt;- par(mfrow = c(2, 2))

# Standard funnel plot
#
funnel(m1)

# Funnel plot with confidence intervals, common effect estimate and
# contours
#
cc &lt;- funnel(m1, common = TRUE,
             level = 0.95, contour = c(0.9, 0.95, 0.99))$col.contour
legend(0.05, 0.05,
  c("0.1 &gt; p &gt; 0.05", "0.05 &gt; p &gt; 0.01", "&lt; 0.01"), fill = cc)

# Contour-enhanced funnel plot with user-chosen colours
#
funnel(m1, common = TRUE,
  level = 0.95, contour = c(0.9, 0.95, 0.99),
  col.contour = c("darkgreen", "green", "lightgreen"),
  lwd = 2, cex = 2, pch = 16, studlab = TRUE, cex.studlab = 1.25)
legend(0.05, 0.05,
  c("0.1 &gt; p &gt; 0.05", "0.05 &gt; p &gt; 0.01", "&lt; 0.01"),
  fill = c("darkgreen", "green", "lightgreen"))

par(oldpar)

</code></pre>

<hr>
<h2 id='gs'>Get default for a meta-analysis setting.</h2><span id='topic+gs'></span>

<h3>Description</h3>

<p>Get default for a meta-analysis setting in R package <b>meta</b>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gs(x = NULL, unname = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gs_+3A_x">x</code></td>
<td>
<p>A character string or vector with setting name(s).</p>
</td></tr>
<tr><td><code id="gs_+3A_unname">unname</code></td>
<td>
<p>A logical whether to remove names from attributes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to get the default for a meta-analysis
setting defined using <code><a href="#topic+settings.meta">settings.meta</a></code>.
</p>
<p>This function is primarily used to define default settings in
meta-analysis functions, e.g., <code><a href="#topic+metabin">metabin</a></code> or
<code><a href="#topic+metacont">metacont</a></code>. A list of all arguments with current
settings is printed using the command
<code>settings.meta("print")</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+settings.meta">settings.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get default setting for confidence interval of random effects
# model
#
gs("method.random.ci")

# Get default setting for summary measure in metabin()
#
gs("smbin")

</code></pre>

<hr>
<h2 id='JAMAlabels'>Create study labels in JAMA layout (deprecated function)</h2><span id='topic+JAMAlabels'></span>

<h3>Description</h3>

<p>Deprecated function to create study labels in JAMA layout (for
forest plot). Replaced by <code><a href="#topic+labels.meta">labels.meta</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>JAMAlabels(author, year, citation, data = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="JAMAlabels_+3A_author">author</code></td>
<td>
<p>A vector providing study authors.</p>
</td></tr>
<tr><td><code id="JAMAlabels_+3A_year">year</code></td>
<td>
<p>A vector providing year of publication.</p>
</td></tr>
<tr><td><code id="JAMAlabels_+3A_citation">citation</code></td>
<td>
<p>A vector providing citation numbers.</p>
</td></tr>
<tr><td><code id="JAMAlabels_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This auxiliary function can be used to create study labels in JAMA
layout which can be added to a forest plot using argument
'studlab'.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+labels.meta">labels.meta</a></code>, <code><a href="#topic+forest.meta">forest.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)

refs &lt;- 20 + 1:7

Fleiss1993bin$mylabs &lt;-
  JAMAlabels(study, year, refs, data = Fleiss1993bin)

m &lt;- metabin(d.asp, n.asp, d.plac, n.plac, data = Fleiss1993bin,
  studlab = paste(study, year),
  sm = "OR", random = FALSE)

forest(m, studlab = mylabs, layout = "JAMA",
  fontfamily = "Times", fontsize = 10)

</code></pre>

<hr>
<h2 id='labbe.metabin'>L'Abbé plot for meta-analysis with binary outcomes</h2><span id='topic+labbe.metabin'></span><span id='topic+labbe'></span><span id='topic+labbe.default'></span>

<h3>Description</h3>

<p>Draw a L'Abbé plot for meta-analysis with binary outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'metabin'
labbe(
  x,
  xlim,
  ylim,
  xlab = NULL,
  ylab = NULL,
  TE.common = x$TE.common,
  TE.random = x$TE.random,
  common = x$common,
  random = x$random,
  backtransf = x$backtransf,
  axes = TRUE,
  pch = 21,
  text = NULL,
  cex = 1,
  col = "black",
  bg = "lightgray",
  lwd = 1,
  lwd.common = lwd,
  lwd.random = lwd,
  lty.common = 2,
  lty.random = 9,
  col.common = col,
  col.random = col,
  nulleffect = TRUE,
  lwd.nulleffect = lwd,
  col.nulleffect = "lightgray",
  sm = x$sm,
  weight,
  studlab = FALSE,
  cex.studlab = 0.8,
  pos.studlab = 2,
  label.e = x$label.e,
  label.c = x$label.c,
  warn.deprecated = gs("warn.deprecated"),
  TE.fixed,
  fixed,
  lwd.fixed,
  lty.fixed,
  col.fixed,
  ...
)

## Default S3 method:
labbe(
  x,
  y,
  xlim,
  ylim,
  xlab = NULL,
  ylab = NULL,
  TE.common = NULL,
  TE.random = NULL,
  common = !is.null(TE.common),
  random = !is.null(TE.random),
  backtransf = TRUE,
  axes = TRUE,
  pch = 21,
  text = NULL,
  cex = 1,
  col = "black",
  bg = "lightgray",
  lwd = 1,
  lwd.common = lwd,
  lwd.random = lwd,
  lty.common = 2,
  lty.random = 9,
  col.common = col,
  col.random = col,
  nulleffect = TRUE,
  lwd.nulleffect = lwd,
  col.nulleffect = "lightgray",
  sm = "",
  weight,
  studlab = FALSE,
  cex.studlab = 0.8,
  pos.studlab = 2,
  label.e = NULL,
  label.c = NULL,
  warn.deprecated = gs("warn.deprecated"),
  TE.fixed,
  fixed,
  lwd.fixed,
  lty.fixed,
  col.fixed,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labbe.metabin_+3A_x">x</code></td>
<td>
<p>An object of class <code>metabin</code>. Alternatively, the x
coordinates of points of the L'Abbé plot.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min, max) of the plot.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_ylim">ylim</code></td>
<td>
<p>The y limits (min, max) of the plot.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y-axis.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_te.common">TE.common</code></td>
<td>
<p>A numeric or vector specifying combined common
effect estimate(s).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_te.random">TE.random</code></td>
<td>
<p>A numeric or vector specifying combined random
effects estimate(s).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_common">common</code></td>
<td>
<p>A logical indicating whether the common effect
estimate should be plotted.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_random">random</code></td>
<td>
<p>A logical indicating whether the random effects
estimate should be plotted.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating which values should be
printed on x- and y-axis (see Details).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_axes">axes</code></td>
<td>
<p>A logical indicating whether axes should be drawn on
the plot.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_pch">pch</code></td>
<td>
<p>The plotting symbol used for individual studies.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_text">text</code></td>
<td>
<p>A character vector specifying the text to be used
instead of plotting symbol.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_cex">cex</code></td>
<td>
<p>The magnification to be used for plotting symbol.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_col">col</code></td>
<td>
<p>A vector with colour of plotting symbols.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_bg">bg</code></td>
<td>
<p>A vector with background colour of plotting symbols (only
used if <code>pch</code> in <code>21:25</code>).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lwd">lwd</code></td>
<td>
<p>The line width.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lwd.common">lwd.common</code></td>
<td>
<p>The line width(s) for common effect estimate(s)
(if <code>common</code> is not <code>NULL</code> or <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lwd.random">lwd.random</code></td>
<td>
<p>The line width(s) for random effects estimate(s)
(if <code>random</code> is not <code>NULL</code> or <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lty.common">lty.common</code></td>
<td>
<p>Line type(s) for common effect estimate(s).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lty.random">lty.random</code></td>
<td>
<p>Line type(s) for random effects estimate(s).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_col.common">col.common</code></td>
<td>
<p>Colour of line(s) for common effect estimate(s).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_col.random">col.random</code></td>
<td>
<p>Colour of line(s) for random effects estimate(s).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_nulleffect">nulleffect</code></td>
<td>
<p>A logical indicating whether line for null effect
should be added to the plot..</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lwd.nulleffect">lwd.nulleffect</code></td>
<td>
<p>Width of line for null effect.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_col.nulleffect">col.nulleffect</code></td>
<td>
<p>Colour of line for null effect.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_sm">sm</code></td>
<td>
<p>A character string indicating underlying summary measure,
i.e., <code>"RD"</code>, <code>"RR"</code>, <code>"OR"</code>, or <code>"ASD"</code>.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_weight">weight</code></td>
<td>
<p>Either a numeric vector specifying relative sizes of
plotting symbols or a character string indicating which type of
plotting symbols is to be used for individual treatment
estimates. One of missing (see Details), <code>"same"</code>,
<code>"common"</code>, or <code>"random"</code>, can be abbreviated. Plot
symbols have the same size for all studies or represent study
weights from common effect or random effects model.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_studlab">studlab</code></td>
<td>
<p>A logical indicating whether study labels should be
printed in the graph. A vector with study labels can also be
provided (must be of same length as <code>x$event.e</code> then).</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_cex.studlab">cex.studlab</code></td>
<td>
<p>Size of study labels.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_pos.studlab">pos.studlab</code></td>
<td>
<p>Position of study labels, see argument
<code>pos</code> in <code><a href="graphics.html#topic+text">text</a></code>.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_label.e">label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_label.c">label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_te.fixed">TE.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'TE.common').</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_fixed">fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'common').</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lwd.fixed">lwd.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'lwd.common').</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_lty.fixed">lty.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'lty.common').</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_col.fixed">col.fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'col.common').</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_...">...</code></td>
<td>
<p>Graphical arguments as in <code>par</code> may also be
passed as arguments.</p>
</td></tr>
<tr><td><code id="labbe.metabin_+3A_y">y</code></td>
<td>
<p>The y coordinates of the L'Abbé plot, if argument <code>x</code>
is not an object of class <code>metabin</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A L'Abbé plot is a scatter plot with the risk in the control group
on the x-axis and the risk in the experimental group on the y-axis
(L'Abbé et al., 1987). It can be used to evaluate heterogeneity in
meta-analysis.  Furthermore, this plot can aid to choose a summary
measure (odds ratio, risk ratio, risk difference) that will result
in more consistent results (Jiménez et al., 1997; Deeks, 2002).
</p>
<p>If argument <code>backtransf</code> is TRUE (default), event
probabilities will be printed on x- and y-axis. Otherwise,
transformed event probabilities will be printed as defined by the
summary measure, i.e., log odds of probabilities for odds ratio as
summary measure (<code>sm = "OR"</code>), log probabilities for <code>sm
= "RR"</code>, and arcsine-transformed probabilities for <code>sm =
"ASD"</code>.
</p>
<p>If <code>common</code> is TRUE, the estimate of the common effct model is
plotted as a line. If <code>random</code> is TRUE, the estimate of the
random effects model is plotted as a line.
</p>
<p>Information from object <code>x</code> is utilised if argument
<code>weight</code> is missing. Weights from the common effect model are
used (<code>weight = "common"</code>) if argument <code>x$common</code> is
<code>TRUE</code>; weights from the random effects model are used
(<code>weight = "random"</code>) if argument <code>x$random</code> is
<code>TRUE</code> and <code>x$common</code> is <code>FALSE</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Deeks JJ (2002):
Issues in the selection of a summary statistic for meta-analysis of
clinical trials with binary outcomes.
<em>Statistics in Medicine</em>,
<b>21</b>, 1575&ndash;600
</p>
<p>Jiménez FJ, Guallar E, Martín-Moreno JM (1997):
A graphical display useful for meta-analysis.
<em>European Journal of Public Health</em>,
<b>1</b>, 101&ndash;5
</p>
<p>L'Abbé KA, Detsky AS, O'Rourke K (1987):
Meta-analysis in clinical research.
<em>Annals of Internal Medicine</em>,
<b>107</b>, 224&ndash;33
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, studlab = paste(author, year),
  sm = "RR", method = "I")

# L'Abbe plot for risk ratio
#
labbe(m1)

# L'Abbe plot for odds ratio
#
labbe(m1, sm = "OR")
# same plot
labbe(update(m1, sm = "OR"))

# L'Abbe plot for risk difference
#
labbe(m1, sm = "RD")

# L'Abbe plot on log odds scale
#
labbe(m1, sm = "OR", backtransf = FALSE)

# L'Abbe plot for odds ratio with coloured lines for various
# treatment effects (defined as log odds ratios)
#
mycols &lt;- c("blue", "yellow", "green", "red", "green", "yellow", "blue")
labbe(m1, sm = "OR", random = FALSE,
  TE.common = log(c(1 / 10, 1 / 5, 1 / 2, 1, 2, 5, 10)),
  col.common = mycols, lwd.common = 2)

# L'Abbe plot on log odds scale with coloured lines for various
# treatment effects (defined as log odds ratios)
#
labbe(m1, sm = "OR", random = FALSE, backtransf = FALSE,
  TE.common = log(c(1 / 10, 1 / 5, 1 / 2, 1, 2, 5, 10)),
  col.common = mycols, lwd.common = 2)

</code></pre>

<hr>
<h2 id='labels.meta'>Create study labels for forest plot</h2><span id='topic+labels.meta'></span>

<h3>Description</h3>

<p>Create study labels for forest plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
labels(
  object,
  author = object$studlab,
  year = "",
  citation = NULL,
  layout = "JAMA",
  data = object$data,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="labels.meta_+3A_object">object</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="labels.meta_+3A_author">author</code></td>
<td>
<p>An optional vector providing study authors.</p>
</td></tr>
<tr><td><code id="labels.meta_+3A_year">year</code></td>
<td>
<p>An optional vector providing year of publication.</p>
</td></tr>
<tr><td><code id="labels.meta_+3A_citation">citation</code></td>
<td>
<p>An optional vector providing citation numbers.</p>
</td></tr>
<tr><td><code id="labels.meta_+3A_layout">layout</code></td>
<td>
<p>A character string specifying layout. Either &quot;JAMA&quot;
or &quot;Lancet&quot;.</p>
</td></tr>
<tr><td><code id="labels.meta_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information.</p>
</td></tr>
<tr><td><code id="labels.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored at the moment).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This auxiliary function can be used to create study labels in JAMA
or Lancet layout which can be added to a forest plot using argument
'studlab'.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+forest.meta">forest.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)

refs &lt;- 20 + 1:7

m &lt;- metabin(d.asp, n.asp, d.plac, n.plac, data = Fleiss1993bin,
 studlab = study, sm = "OR", random = FALSE)

forest(m,
  studlab = labels(m, year = year, citation = refs, layout = "JAMA"),
  layout = "JAMA", fontfamily = "Times", fontsize = 10)

forest(m,
  studlab = labels(m, year = year, citation = refs, layout = "Lancet"))

</code></pre>

<hr>
<h2 id='longarm'>Transform data from pairwise comparisons to long arm-based format</h2><span id='topic+longarm'></span>

<h3>Description</h3>

<p>This function transforms data from pairwise comparisons to a long
arm-based format, i.e., two rows for a pairwise comparison.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>longarm(
  treat1,
  treat2,
  event1,
  n1,
  event2,
  n2,
  mean1,
  sd1,
  mean2,
  sd2,
  time1,
  time2,
  data = NULL,
  studlab,
  id1 = NULL,
  id2 = NULL,
  append = TRUE,
  keep.duplicated = FALSE,
  keep.internal = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="longarm_+3A_treat1">treat1</code></td>
<td>
<p>Either label for first treatment or a meta-analysis
or pairwise object (see Details).</p>
</td></tr>
<tr><td><code id="longarm_+3A_treat2">treat2</code></td>
<td>
<p>Label for second treatment.</p>
</td></tr>
<tr><td><code id="longarm_+3A_event1">event1</code></td>
<td>
<p>Number of events (first treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_n1">n1</code></td>
<td>
<p>Number of observations (first treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_event2">event2</code></td>
<td>
<p>Number of events (second treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_n2">n2</code></td>
<td>
<p>Number of observations (second treatment)</p>
</td></tr>
<tr><td><code id="longarm_+3A_mean1">mean1</code></td>
<td>
<p>Estimated mean (first treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_sd1">sd1</code></td>
<td>
<p>Standard deviation (first treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_mean2">mean2</code></td>
<td>
<p>Estimated mean (second treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_sd2">sd2</code></td>
<td>
<p>Standard deviation (second treatment).</p>
</td></tr>
<tr><td><code id="longarm_+3A_time1">time1</code></td>
<td>
<p>Person time at risk (first treatment)</p>
</td></tr>
<tr><td><code id="longarm_+3A_time2">time2</code></td>
<td>
<p>Person time at risk (second treatment)</p>
</td></tr>
<tr><td><code id="longarm_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information.</p>
</td></tr>
<tr><td><code id="longarm_+3A_studlab">studlab</code></td>
<td>
<p>A vector with study labels (optional).</p>
</td></tr>
<tr><td><code id="longarm_+3A_id1">id1</code></td>
<td>
<p>Last character(s) of variable names for additional
variables with group specific information for first treatment.</p>
</td></tr>
<tr><td><code id="longarm_+3A_id2">id2</code></td>
<td>
<p>Last character(s) of variable names for additional
variables with group specific information for second treatment.</p>
</td></tr>
<tr><td><code id="longarm_+3A_append">append</code></td>
<td>
<p>A logical indicating if data frame provided in
argument 'data' should be returned.</p>
</td></tr>
<tr><td><code id="longarm_+3A_keep.duplicated">keep.duplicated</code></td>
<td>
<p>A logical indicating if duplicated rows
should be returned (see Details).</p>
</td></tr>
<tr><td><code id="longarm_+3A_keep.internal">keep.internal</code></td>
<td>
<p>A logical indicating if variables generated
internally should be returned (typically only relevant for data
checking).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function transforms data given as one pairwise comparison per
row to a long arm-based format with one row per treatment arm. The
long arm-based format is, for example, the required input format
for WinBUGS.
</p>
<p>The function can be used to transform data with a binary,
continuous or count outcome. The corresponding meta-analysis
functions are <code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code> and
<code><a href="#topic+metainc">metainc</a></code>. Accordingly, a meta-analysis object created
with one of these functions can be provided as argument
<code>treat1</code>. It is also possible to use the longarm function with
an R objected created with <code><a href="netmeta.html#topic+pairwise">pairwise</a></code> from R
package <b>netmeta</b>.
</p>
<p>Otherwise, arguments <code>treat1</code> and <code>treat2</code> are mandatory
to identify the individual treatments and, depending on the
outcome, the following additional arguments are mandatory:
</p>

<ul>
<li><p> event1, n1, event2, n2 (binary outcome);
</p>
</li>
<li><p> n1, mean1, sd1, n2, mean2, sd2 (continuous outcome);
</p>
</li>
<li><p> time1, n1, time2, n2 (count outcome).
</p>
</li></ul>

<p>Argument <code>studlab</code> must be provided if several pairwise
comparisons come from a single study with more than two treatments.
</p>
<p>The following variables will be returned:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>studlab</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>treat</em></b></td><td style="text-align: left;"> treatment label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n</em></b></td><td style="text-align: left;"> group sample size (count outcome only if provided) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>events</em></b></td><td style="text-align: left;"> number of events (binary or count outcome) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>nonevents</em></b></td><td style="text-align: left;"> number of non-events (binary outcome) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean</em></b></td><td style="text-align: left;"> estimated mean (continuous outcome) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>sd</em></b></td><td style="text-align: left;"> standard deviation (continuous outcome) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>time</em></b></td><td style="text-align: left;"> person time at risk (count outcome) </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>In addition, the data set provided in argument <code>data</code> will be
returned if argument <code>append = TRUE</code> (default).
</p>
<p>Argument <code>keep.duplicated</code> can be used to keep duplicated rows
from the data set. Duplicated rows can occur, for example, in a
three-arm study comparing treatments A and B with placebo. In this
situation, the placebo arm will be returned twice in the data set
in long arm-based format if <code>keep.duplicated = TRUE</code>. By
default, duplicated rows with not be kept in the data set.
</p>


<h3>Value</h3>

<p>A data frame in long arm-based format.
</p>


<h3>Note</h3>

<p>R function <code><a href="metafor.html#topic+to.long">to.long</a></code> from R package
<b>metafor</b> is called internally.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="netmeta.html#topic+pairwise">pairwise</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Artificial example with three studies
m &lt;- metabin(1:3, 100:102, 4:6, 200:202, studlab = LETTERS[1:3])
# Transform data to long arm-based format
longarm(m)
# Keep internal variables
longarm(m, keep.internal = TRUE)

</code></pre>

<hr>
<h2 id='meta-object'>Description of R object of class &quot;meta&quot;</h2><span id='topic+meta-object'></span>

<h3>Description</h3>

<p>Detailed description of R objects of class &quot;meta&quot;.
</p>


<h3>Details</h3>

<p>The following R functions create an object of class <code>"meta"</code>:
</p>

<ul>
<li> <p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metacor">metacor</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metamean">metamean</a></code>,
<code><a href="#topic+metaprop">metaprop</a></code>, <code><a href="#topic+metarate">metarate</a></code>,
<code><a href="#topic+metacr">metacr</a></code>, <code><a href="#topic+metamerge">metamerge</a></code>,
<code><a href="#topic+trimfill">trimfill</a></code>
</p>
</li></ul>

<p>The following generic functions are available for an object of
class <code>"meta"</code>:
</p>

<ul>
<li> <p><code><a href="#topic+as.data.frame.meta">as.data.frame.meta</a></code>, <code><a href="#topic+labels.meta">labels.meta</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>, <code><a href="#topic+print.summary.meta">print.summary.meta</a></code>,
<code><a href="#topic+summary.meta">summary.meta</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+weights.meta">weights.meta</a></code>
</p>
</li></ul>

<p>An object of class <code>"meta"</code> is a list containing the following
components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>studlab</code> </td><td style="text-align: left;"> Study labels </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm</code> </td><td style="text-align: left;"> Effect measure </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>null.effect</code> </td><td style="text-align: left;"> Effect under the null hypothesis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE</code> </td><td style="text-align: left;"> Effect estimates (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE</code> </td><td style="text-align: left;"> Standard error of effect estimates (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic</code> </td><td style="text-align: left;"> Statistics for test of effect (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval</code> </td><td style="text-align: left;"> P-values for test of effect (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df</code> </td><td style="text-align: left;"> Degrees of freedom (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>level</code> </td><td style="text-align: left;"> Level of confidence intervals for individual studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower</code> </td><td style="text-align: left;"> Lower confidence limits (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper</code> </td><td style="text-align: left;"> Upper confidence limits (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>three.level</code> </td><td style="text-align: left;"> Indicator variable for three-level
  meta-analysis model </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>cluster</code> </td><td style="text-align: left;"> Cluster variable (three-level meta-analysis
  model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>rho</code> </td><td style="text-align: left;"> Within-cluster correlation (three-level
  meta-analysis model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k</code> </td><td style="text-align: left;"> Number of estimates combined in meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.study</code> </td><td style="text-align: left;"> Number of studies combined in meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.all</code> </td><td style="text-align: left;"> Number of all studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.TE</code> </td><td style="text-align: left;"> Number of studies with estimable effects </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>overall</code> </td><td style="text-align: left;"> Print meta-analysis results </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>overall.hetstat</code> </td><td style="text-align: left;"> Print overall heterogeneity statistics </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>common</code> </td><td style="text-align: left;"> Print results for common effect meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>random</code> </td><td style="text-align: left;"> Print results for random effects meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prediction</code> </td><td style="text-align: left;"> Print prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>backtransf</code> </td><td style="text-align: left;"> Back transform results in printouts and plots </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method</code> </td><td style="text-align: left;"> Meta-analysis method (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random</code> </td><td style="text-align: left;"> Meta-analysis method (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>w.common</code> </td><td style="text-align: left;"> Weights for common effect model (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.common</code> </td><td style="text-align: left;"> Estimated overall effect (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.common</code> </td><td style="text-align: left;"> Standard error of overall effect (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic.common</code> </td><td style="text-align: left;"> Statistic for test of overall effect (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.common</code> </td><td style="text-align: left;"> P-value for test of overall effect (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>level.ma</code> </td><td style="text-align: left;"> Level of confidence interval for meta-analysis
  estimates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.common</code> </td><td style="text-align: left;"> Lower confidence limit (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.common</code> </td><td style="text-align: left;"> Upper confidence limit (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>w.random</code> </td><td style="text-align: left;"> Weight for random effects model (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.random</code> </td><td style="text-align: left;"> Estimated overall effect (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.random</code> </td><td style="text-align: left;"> Standard error of overall effect (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic.random</code> </td><td style="text-align: left;"> Statistic for test of overall effect (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.random</code> </td><td style="text-align: left;"> P-value for test of overall effect (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci</code> </td><td style="text-align: left;"> Confidence interval method (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.random</code> </td><td style="text-align: left;"> Degrees of freedom (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.random</code> </td><td style="text-align: left;"> Lower confidence limit (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.random</code> </td><td style="text-align: left;"> Upper confidence limit (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.classic</code> </td><td style="text-align: left;"> Standard error (classic random effects method) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.ci</code> </td><td style="text-align: left;"> <em>Ad hoc</em> correction for
  Hartung-Knapp method (confidence interval) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.hakn.ci</code> </td><td style="text-align: left;"> Degrees of freedom for Hartung-Knapp method
  </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> (if used in meta-analysis) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.ci</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp method
  </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (not taking <em>ad hoc</em> variance correction into account)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.adhoc.ci</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp
  method </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (taking <em>ad hoc</em> variance correction into account) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.kero</code> </td><td style="text-align: left;"> Degrees of freedom for Kenward-Roger method </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;">  (if used in meta-analysis) </td>
</tr>
<tr>
 <td style="text-align: left;"> 
<code>seTE.kero</code> </td><td style="text-align: left;"> Standard error for Kenward-Roger
  method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.predict</code> </td><td style="text-align: left;"> Method to calculate prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.pi</code> </td><td style="text-align: left;"> <em>Ad hoc</em> correction for
  Hartung-Knapp method (prediction interval) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.hakn.ci</code> </td><td style="text-align: left;"> Degrees of freedom for Hartung-Knapp method
  </td>
</tr>
<tr>
 <td style="text-align: left;"> </td><td style="text-align: left;"> (prediction interval) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.predict</code> </td><td style="text-align: left;"> Standard error used to calculate
  prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.predict</code> </td><td style="text-align: left;"> Degrees of freedom for prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>level.predict</code> </td><td style="text-align: left;"> Level of prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.predict</code> </td><td style="text-align: left;"> Lower limit of prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.predict</code> </td><td style="text-align: left;"> Upper limit of prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.pi</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp method
  </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (not taking <em>ad hoc</em> variance correction into account)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.adhoc.pi</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp
  method </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (taking <em>ad hoc</em> variance correction into account) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q</code> </td><td style="text-align: left;"> Heterogeneity statistic </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.Q</code> </td><td style="text-align: left;"> Degrees of freedom for heterogeneity statistic
  <code>Q</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q</code> </td><td style="text-align: left;"> P-value of heterogeneity test </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau</code> </td><td style="text-align: left;"> Method to estimate between-study variance <code class="reqn">\tau^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>control</code> </td><td style="text-align: left;"> Additional arguments for iterative estimation
  of <code class="reqn">\tau^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau.ci</code> </td><td style="text-align: left;"> Method for confidence interval of <code class="reqn">\tau^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau2</code> </td><td style="text-align: left;"> Between-study variance <code class="reqn">\tau^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>se.tau2</code> </td><td style="text-align: left;"> Standard error of <code class="reqn">\tau^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.tau2</code> </td><td style="text-align: left;"> Lower confidence limit (<code class="reqn">\tau^2</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.tau2</code> </td><td style="text-align: left;"> Upper confidence limit (<code class="reqn">\tau^2</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau</code> </td><td style="text-align: left;"> Square-root of between-study variance <code class="reqn">\tau</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.tau</code> </td><td style="text-align: left;"> Lower confidence limit (<code class="reqn">\tau</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.tau</code> </td><td style="text-align: left;"> Upper confidence limit (<code class="reqn">\tau</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau.preset</code> </td><td style="text-align: left;"> Prespecified value for <code class="reqn">\tau</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.tau</code> </td><td style="text-align: left;"> Effect estimate used to estimate <code class="reqn">\tau^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>detail.tau</code> </td><td style="text-align: left;"> Detail on between-study variance estimate </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>H</code> </td><td style="text-align: left;"> Heterogeneity statistic H </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.H</code> </td><td style="text-align: left;"> Lower confidence limit (heterogeneity statistic H) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.H</code> </td><td style="text-align: left;"> Upper confidence limit (heterogeneity statistic H) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>I2</code> </td><td style="text-align: left;"> Heterogeneity statistic I<code class="reqn">^2</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.I2</code> </td><td style="text-align: left;"> Lower confidence limit (heterogeneity statistic I<code class="reqn">^2</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.I2</code> </td><td style="text-align: left;"> Upper confidence limit (heterogeneity statistic I<code class="reqn">^2</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Rb</code> </td><td style="text-align: left;"> Heterogeneity statistic R<code class="reqn">_b</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.Rb</code> </td><td style="text-align: left;"> Lower confidence limit (heterogeneity statistic R<code class="reqn">_b</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.Rb</code> </td><td style="text-align: left;"> Upper confidence limit (heterogeneity statistic R<code class="reqn">_b</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.bias</code> </td><td style="text-align: left;"> Method to test for funnel plot asymmetry </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>text.common</code> </td><td style="text-align: left;"> Label for common effect model </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>text.random</code> </td><td style="text-align: left;"> Label for random effects model </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>text.predict</code> </td><td style="text-align: left;"> Label for prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>text.w.common</code> </td><td style="text-align: left;"> Label for weights (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>text.w.random</code> </td><td style="text-align: left;"> Label for weights (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>title</code> </td><td style="text-align: left;"> Title of meta-analysis / systematic review </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>complab</code> </td><td style="text-align: left;"> Comparison label </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>outclab</code> </td><td style="text-align: left;"> Outcome label </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>label.e</code> </td><td style="text-align: left;"> Label for experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>label.c</code> </td><td style="text-align: left;"> Label for control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>label.left</code> </td><td style="text-align: left;"> Graph label on left side of forest plot </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>label.right</code> </td><td style="text-align: left;"> Graph label on right side of forest plot </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>keepdata</code> </td><td style="text-align: left;"> Keep original data </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>data</code> </td><td style="text-align: left;"> Original data (set) used in function call (if
  <code>keepdata = TRUE</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>subset</code> </td><td style="text-align: left;"> Information on subset of original data used in
  meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (if <code>keepdata = TRUE</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>exclude</code> </td><td style="text-align: left;"> Studies excluded from meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>warn</code> </td><td style="text-align: left;"> Print warnings </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>call</code> </td><td style="text-align: left;"> Function call </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>version</code> </td><td style="text-align: left;"> Version of R package <b>meta</b> used to create
  object
</td>
</tr>

</table>

<p>For subgroup analysis (argument <code>subgroup</code>), the following
additional components are added to the list.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>subgroup</code> </td><td style="text-align: left;"> Subgroup information (for individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>subgroup.name</code> </td><td style="text-align: left;"> Name of subgroup variable </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>print.subgroup.name</code> </td><td style="text-align: left;"> Print name of subgroup variable </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sep.subgroup</code> </td><td style="text-align: left;"> Separator between name of subgroup variable and value </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.subgroup</code> </td><td style="text-align: left;"> Print test for subgroup differences </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prediction.subgroup</code> </td><td style="text-align: left;"> Print prediction interval for subgroup(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau.common</code> </td><td style="text-align: left;"> Assumption of common between-study variance in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>subgroup.levels</code> </td><td style="text-align: left;"> Levels of grouping variable </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.w</code> </td><td style="text-align: left;"> Number of estimates combined in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.study.w</code> </td><td style="text-align: left;"> Number of studies combined in subgroups
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.all.w</code> </td><td style="text-align: left;"> Number of studies in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.TE.w</code> </td><td style="text-align: left;"> Number of studies with estimable effects
  in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.common.w</code> </td><td style="text-align: left;"> Estimated effect in subgroups (common
  effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.common.w</code> </td><td style="text-align: left;"> Standard error in subgroups (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic.common.w</code> </td><td style="text-align: left;"> Statistic for test of effect in subgroups (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.common.w</code> </td><td style="text-align: left;"> P-value for test of effect in subgroups (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.common.w</code> </td><td style="text-align: left;"> Lower confidence limit in subgroups (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.common.w</code> </td><td style="text-align: left;"> Upper confidence limit in subgroups (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>w.common.w</code> </td><td style="text-align: left;"> Total weight in subgroups (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.random.w</code> </td><td style="text-align: left;"> Estimated effect in subgroups (random
  effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.random.w</code> </td><td style="text-align: left;"> Standard error in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic.random.w</code> </td><td style="text-align: left;"> Statistic for test of effect in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.random.w</code> </td><td style="text-align: left;"> P-value for test of effect in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.random.w</code> </td><td style="text-align: left;"> Degrees of freedom in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.random.w</code> </td><td style="text-align: left;"> Lower confidence limit in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.random.w</code> </td><td style="text-align: left;"> Upper confidence limit in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>w.random.w</code> </td><td style="text-align: left;"> Total weight in subgroups (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.classic.w</code> </td><td style="text-align: left;"> Standard error (classic random effects method) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.hakn.ci.w</code> </td><td style="text-align: left;"> Degrees of freedom for Hartung-Knapp
  method in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.ci.w</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp method in subgroups
  </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (not taking <em>ad hoc</em> variance correction into account)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.adhoc.ci.w</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp
  method in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.kero.w</code> </td><td style="text-align: left;"> Degrees of freedom for Kenward-Roger method in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;"> 
<code>seTE.kero.w</code> </td><td style="text-align: left;"> Standard error for Kenward-Roger
  method in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.predict.w</code> </td><td style="text-align: left;"> Standard error for prediction interval in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.predict.w</code> </td><td style="text-align: left;"> Degrees of freedom for prediction interval in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.predict.w</code> </td><td style="text-align: left;"> Lower limit of prediction interval in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.predict.w</code> </td><td style="text-align: left;"> Upper limit of prediction interval in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.pi.w</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp method in subgroups (prediction intervals)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (not taking <em>ad hoc</em> variance correction into account)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.hakn.adhoc.pi.w</code> </td><td style="text-align: left;"> Standard error for Hartung-Knapp
  method in subgroups (prediction intervals) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.w</code> </td><td style="text-align: left;"> Heterogeneity statistic Q in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q.w</code> </td><td style="text-align: left;"> P-value for test of heterogeneity in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau2.w</code> </td><td style="text-align: left;"> Between-study variance <code class="reqn">\tau^2</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau.w</code> </td><td style="text-align: left;"> Square-root of between-study variance <code class="reqn">\tau</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>H.w</code> </td><td style="text-align: left;"> Heterogeneity statistic H in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.H.w</code> </td><td style="text-align: left;"> Lower confidence limit for H in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.H.w</code> </td><td style="text-align: left;"> Upper confidence limit for H in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>I2.w</code> </td><td style="text-align: left;"> Heterogeneity statistic I<code class="reqn">^2</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.I2.w</code> </td><td style="text-align: left;"> Lower confidence limit for I<code class="reqn">^2</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.I2.w</code> </td><td style="text-align: left;"> Upper confidence limit for I<code class="reqn">^2</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Rb.w</code> </td><td style="text-align: left;"> Heterogeneity statistic R<code class="reqn">_b</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.Rb.w</code> </td><td style="text-align: left;"> Lower confidence limit for R<code class="reqn">_b</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.Rb.w</code> </td><td style="text-align: left;"> Upper confidence limit for R<code class="reqn">_b</code> in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.w.common</code> </td><td style="text-align: left;"> Within-group heterogeneity statistic Q
  (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.w.random</code> </td><td style="text-align: left;"> Within-group heterogeneity statistic
  Q (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (only calculated if argument <code>tau.common = TRUE</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.Q.w</code> </td><td style="text-align: left;"> Degrees of freedom for <code>Q.w.common</code> and <code>Q.w.random</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q.w.common</code> </td><td style="text-align: left;"> P-value of test for residual
  heterogeneity (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q.w.random</code> </td><td style="text-align: left;"> P-value of test for residual
  heterogeneity (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.b.common</code> </td><td style="text-align: left;"> Between-groups heterogeneity statistic
  Q (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.Q.b.common</code> </td><td style="text-align: left;"> Degrees of freedom for <code>Q.b.common</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q.b.common</code> </td><td style="text-align: left;"> P-value of test for subgroup
  differences (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.b.random</code> </td><td style="text-align: left;"> Between-groups heterogeneity statistic
  Q (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.Q.b.random</code> </td><td style="text-align: left;"> Degrees of freedom for <code>Q.b.random</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q.b.random</code> </td><td style="text-align: left;"> P-value of test for subgroup
  differences (random effects model)
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metabin">metabin</a></code> has the additional
class <code>"metabin"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>event.e</code> </td><td style="text-align: left;"> Events in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.e</code> </td><td style="text-align: left;"> Sample size in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>event.e</code> </td><td style="text-align: left;"> Events in control group (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.e</code> </td><td style="text-align: left;"> Sample size in control group (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr</code> </td><td style="text-align: left;"> Increment added to zero cells </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.incr</code> </td><td style="text-align: left;"> Continuity correction method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sparse</code> </td><td style="text-align: left;"> Continuity correction applied </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>allstudies</code> </td><td style="text-align: left;"> Include studies with double zeros </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>doublezeros</code> </td><td style="text-align: left;"> Indicator for studies with double zeros </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MH.exact</code> </td><td style="text-align: left;"> Exact Mantel-Haenszel method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RR.Cochrane</code> </td><td style="text-align: left;"> Cochrane method to calculate risk ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.Cochrane</code> </td><td style="text-align: left;"> Cochrane method to calculate <code class="reqn">\tau^2</code>
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.CMH</code> </td><td style="text-align: left;"> Cochran-Mantel-Haenszel statistic </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>df.Q.CMH</code> </td><td style="text-align: left;"> Degrees of freedom for <code>Q.CMH</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.Q.CMH</code> </td><td style="text-align: left;"> P-value of Cochran-Mantel-Haenszel test </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>print.CMH</code> </td><td style="text-align: left;"> Print results for Cochran-Mantel-Haenszel
  statistic </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr.e</code> </td><td style="text-align: left;"> Continuity correction in experimental group
  (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr.c</code> </td><td style="text-align: left;"> Continuity correction in control group
  (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.MH</code> </td><td style="text-align: left;"> Number of studies (Mantel-Haenszel method)
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metacont">metacont</a></code> has the additional
class <code>"metacont"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>n.e</code> </td><td style="text-align: left;"> Sample size in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mean.e</code> </td><td style="text-align: left;"> Estimated mean in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sd.e</code> </td><td style="text-align: left;"> Standard deviation in experimental group
  (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.c</code> </td><td style="text-align: left;"> Sample size in control group (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mean.c</code> </td><td style="text-align: left;"> Estimated mean in control group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sd.c</code> </td><td style="text-align: left;"> Standard deviation in control group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pooledvar</code> </td><td style="text-align: left;"> Use pooled variance for mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.smd</code> </td><td style="text-align: left;"> Method for standardised mean difference
  (SMD) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sd.glass</code> </td><td style="text-align: left;"> Denominator in Glass' method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>exact.smd</code> </td><td style="text-align: left;"> Use exact formulae for SMD </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.ci</code> </td><td style="text-align: left;"> Method to calculate confidence limits </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.mean</code> </td><td style="text-align: left;"> Method to approximate mean </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.sd</code> </td><td style="text-align: left;"> Method to approximate standard deviation
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metacor">metacor</a></code> has the additional
class <code>"metacor"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>cor</code> </td><td style="text-align: left;"> Correlation (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n</code> </td><td style="text-align: left;"> Sample size (individual studies)
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metainc">metainc</a></code> has the additional
class <code>"metainc"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>event.e</code> </td><td style="text-align: left;"> Events in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>time.e</code> </td><td style="text-align: left;"> Person time in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.e</code> </td><td style="text-align: left;"> Sample size in experimental group (individual
  studies) </td>
</tr>
<tr>
 <td style="text-align: left;"> 
<code>event.c</code> </td><td style="text-align: left;"> Events in control group (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>time.c</code> </td><td style="text-align: left;"> Person time in control group (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.c</code> </td><td style="text-align: left;"> Sample size in control group (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr</code> </td><td style="text-align: left;"> Increment added to zero cells </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.incr</code> </td><td style="text-align: left;"> Continuity correction method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sparse</code> </td><td style="text-align: left;"> Continuity correction applied </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr.event</code> </td><td style="text-align: left;"> Continuity correction (individual studies)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.MH</code> </td><td style="text-align: left;"> Number of studies (Mantel-Haenszel method)
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metamean">metamean</a></code> has the additional
class <code>"metamean"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>n</code> </td><td style="text-align: left;"> Sample size (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>mean</code> </td><td style="text-align: left;"> Estimated mean (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sd</code> </td><td style="text-align: left;"> Standard deviation (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.ci</code> </td><td style="text-align: left;"> Method to calculate confidence limits </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.mean</code> </td><td style="text-align: left;"> Method to approximate mean </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.sd</code> </td><td style="text-align: left;"> Method to approximate standard deviation </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metaprop">metaprop</a></code> has the additional
class <code>"metaprop"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>event</code> </td><td style="text-align: left;"> Events (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n</code> </td><td style="text-align: left;"> Sample size (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr</code> </td><td style="text-align: left;"> Increment added to zero cells </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.incr</code> </td><td style="text-align: left;"> Continuity correction method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sparse</code> </td><td style="text-align: left;"> Continuity correction applied </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.ci</code> </td><td style="text-align: left;"> Method to calculate confidence limits </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr.event</code> </td><td style="text-align: left;"> Continuity correction (individual studies)
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metarate">metarate</a></code> has the additional
class <code>"metarate"</code> and the following components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>event</code> </td><td style="text-align: left;"> Events (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>time</code> </td><td style="text-align: left;"> Person time (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n</code> </td><td style="text-align: left;"> Sample size (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr</code> </td><td style="text-align: left;"> Increment added to zero cells </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.incr</code> </td><td style="text-align: left;"> Continuity correction method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sparse</code> </td><td style="text-align: left;"> Continuity correction applied </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.ci</code> </td><td style="text-align: left;"> Method to calculate confidence limits </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>incr.event</code> </td><td style="text-align: left;"> Continuity correction (individual studies)
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+trimfill">trimfill</a></code> has the additional
classes <code>"trimfill"</code> and <code>"metagen"</code> and the following
components.
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>k0</code> </td><td style="text-align: left;"> Number of added studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>left</code> </td><td style="text-align: left;"> Studies missing on left side </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>ma.common</code> </td><td style="text-align: left;"> Use common effect or random effects model to
  estimate </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> number of missing studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>type</code> </td><td style="text-align: left;"> Method to estimate missing studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.iter.max</code> </td><td style="text-align: left;"> Maximum number of iterations </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>n.iter</code> </td><td style="text-align: left;"> Number of iterations </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>trimfill</code> </td><td style="text-align: left;"> Filled studies (individual studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>class.x</code> </td><td style="text-align: left;"> Primary class of meta-analysis object
</td>
</tr>

</table>

<p>An object created with <code><a href="#topic+metamerge">metamerge</a></code> has the additional
class <code>"metamerge"</code>. Furthermore, the following components
have a different meaning:
</p>

<table>
<tr>
 <td style="text-align: left;">
<code>k</code> </td><td style="text-align: left;"> Vector with number of estimates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.study</code> </td><td style="text-align: left;"> Vector with number of studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.all</code> </td><td style="text-align: left;"> Vector with total number of studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.TE</code> </td><td style="text-align: left;"> Vector with number of studies with estimable
  effects </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>k.MH</code> </td><td style="text-align: left;"> Vector with number of studies combined with
  Mantel-Haenszel method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.common</code> </td><td style="text-align: left;"> Vector with common effect estimates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.common</code> </td><td style="text-align: left;"> Vector with standard errors of common effect
  estimates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.common</code> </td><td style="text-align: left;"> Vector with lower confidence limits (common
  effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.common</code> </td><td style="text-align: left;"> Vector with upper confidence limits (common
  effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic.common</code> </td><td style="text-align: left;"> Vector with test statistics for test of
  overall effect (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.common</code> </td><td style="text-align: left;"> Vector with p-value of test for overall effect
  (common effect model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>TE.random</code> </td><td style="text-align: left;"> Vector with random effects estimates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>seTE.random</code> </td><td style="text-align: left;"> Vector with standard errors of random effects
  estimates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>lower.random</code> </td><td style="text-align: left;"> Vector with lower confidence limits (random
  effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>upper.random</code> </td><td style="text-align: left;"> Vector with upper confidence limits (random
  effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>statistic.random</code> </td><td style="text-align: left;"> Vector with test statistics for test of
  overall effect (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>pval.random</code> </td><td style="text-align: left;"> Vector with p-value of test for overall effect
  (random effects model) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>w.common</code> </td><td style="text-align: left;"> Vector or matrix with common effect weights </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>w.random</code> </td><td style="text-align: left;"> Vector or matrix with random effects weights
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+meta-sm">meta-sm</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>, <code><a href="#topic+summary.meta">summary.meta</a></code>,
<code><a href="#topic+forest.meta">forest.meta</a></code>
</p>

<hr>
<h2 id='meta-package'>meta: Brief overview of methods and general hints</h2><span id='topic+meta-package'></span><span id='topic+meta'></span>

<h3>Description</h3>

<p>R package <b>meta</b> is a user-friendly general package providing
standard methods for meta-analysis and supporting Schwarzer et
al. (2015),
<a href="https://link.springer.com/book/10.1007/978-3-319-21416-0">https://link.springer.com/book/10.1007/978-3-319-21416-0</a>.
</p>


<h3>Details</h3>

<p>R package <b>meta</b> (Schwarzer, 2007; Balduzzi et al., 2019)
provides the following statistical methods for meta-analysis.
</p>

<ol>
<li><p> Common effect (also called fixed effect) and random effects model:
</p>

<ul>
<li><p> Meta-analysis of continuous outcome data (<code><a href="#topic+metacont">metacont</a></code>)
</p>
</li>
<li><p> Meta-analysis of binary outcome data (<code><a href="#topic+metabin">metabin</a></code>)
</p>
</li>
<li><p> Meta-analysis of incidence rates (<code><a href="#topic+metainc">metainc</a></code>)
</p>
</li>
<li><p> Generic inverse variance meta-analysis (<code><a href="#topic+metagen">metagen</a></code>)
</p>
</li>
<li><p> Meta-analysis of single correlations (<code><a href="#topic+metacor">metacor</a></code>)
</p>
</li>
<li><p> Meta-analysis of single means (<code><a href="#topic+metamean">metamean</a></code>)
</p>
</li>
<li><p> Meta-analysis of single proportions (<code><a href="#topic+metaprop">metaprop</a></code>)
</p>
</li>
<li><p> Meta-analysis of single incidence rates (<code><a href="#topic+metarate">metarate</a></code>)
</p>
</li></ul>

</li>
<li><p> Several plots for meta-analysis:
</p>

<ul>
<li><p> Forest plot (<code><a href="#topic+forest.meta">forest.meta</a></code>, <code><a href="#topic+forest.metabind">forest.metabind</a></code>)
</p>
</li>
<li><p> Funnel plot (<code><a href="#topic+funnel.meta">funnel.meta</a></code>)
</p>
</li>
<li><p> Galbraith plot / radial plot (<code><a href="#topic+radial.meta">radial.meta</a></code>)
</p>
</li>
<li><p> L'Abbe plot for meta-analysis with binary outcome data
(<code><a href="#topic+labbe.metabin">labbe.metabin</a></code>, <code><a href="#topic+labbe.default">labbe.default</a></code>)
</p>
</li>
<li><p> Baujat plot to explore heterogeneity in meta-analysis
(<code><a href="#topic+baujat.meta">baujat.meta</a></code>)
</p>
</li>
<li><p> Bubble plot to display the result of a meta-regression
(<code><a href="#topic+bubble.metareg">bubble.metareg</a></code>)
</p>
</li></ul>

</li>
<li><p> Three-level meta-analysis model (Van den Noortgate et al.,
2013)
</p>
</li>
<li><p> Generalised linear mixed models (GLMMs) for binary and count
data (Stijnen et al., 2010) (<code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metaprop">metaprop</a></code>, and
<code><a href="#topic+metarate">metarate</a></code>)
</p>
</li>
<li><p> Various estimators for the between-study variance
<code class="reqn">\tau^2</code> in a random effects model (Veroniki et al., 2016);
see description of argument <code>method.tau</code> below
</p>
</li>
<li><p> Hartung-Knapp method for random effects meta-analysis
(Hartung &amp; Knapp, 2001a,b), see description of arguments
<code>method.random.ci</code> and <code>adhoc.hakn.ci</code> below
</p>
</li>
<li><p> Kenward-Roger method for random effects meta-analysis
(Partlett and Riley, 2017), see description of arguments
<code>method.random.ci</code> and <code>method.predict</code> below
</p>
</li>
<li><p> Prediction interval for the treatment effect of a new study
(Higgins et al., 2009; Partlett and Riley, 2017; Nagashima et al.,
2019), see description of argument <code>method.predict</code> below
</p>
</li>
<li><p> Statistical tests for funnel plot asymmetry
(<code><a href="#topic+metabias.meta">metabias.meta</a></code>, <code><a href="#topic+metabias.rm5">metabias.rm5</a></code>) and
trim-and-fill method (<code><a href="#topic+trimfill.meta">trimfill.meta</a></code>,
<code><a href="#topic+trimfill.default">trimfill.default</a></code>) to evaluate bias in meta-analysis
</p>
</li>
<li><p> Meta-regression (<code><a href="#topic+metareg">metareg</a></code>)
</p>
</li>
<li><p> Cumulative meta-analysis (<code><a href="#topic+metacum">metacum</a></code>) and
leave-one-out meta-analysis (<code><a href="#topic+metainf">metainf</a></code>)
</p>
</li>
<li><p> Import data from RevMan Web (<code><a href="#topic+read.cdir">read.cdir</a></code>), RevMan
5 (<code><a href="#topic+read.rm5">read.rm5</a></code>), see also <code><a href="#topic+metacr">metacr</a></code> to
conduct meta-analysis for a single comparison and outcome from a
Cochrane review
</p>
</li></ol>

<p>R package <b>meta</b> provides two vignettes:
</p>

<ul>
<li> <p><code>vignette("meta-workflow")</code> with an overview of main
functions,
</p>
</li>
<li> <p><code>vignette("meta-tutorial")</code> with up-to-date commands for
Balduzzi et al. (2019).
</p>
</li></ul>

<p>Additional statistical meta-analysis methods are provided by add-on
R packages:
</p>

<ul>
<li><p> Frequentist methods for network meta-analysis (R package
<b>netmeta</b>)
</p>
</li>
<li><p> Statistical methods for sensitivity analysis in meta-analysis
(R package <b>metasens</b>)
</p>
</li>
<li><p> Statistical methods for meta-analysis of diagnostic accuracy
studies with several cutpoints (R package <b>diagmeta</b>)
</p>
</li></ul>

<p>In the following, more details on available and default statistical
meta-analysis methods are provided and R function
<code><a href="#topic+settings.meta">settings.meta</a></code> is briefly described which can be used
to change the default settings. Additional information on
meta-analysis objects and available summary measures can be found
on the help pages <code><a href="#topic+meta-object">meta-object</a></code> and
<code><a href="#topic+meta-sm">meta-sm</a></code>.
</p>


<h4>Estimation of between-study variance</h4>

<p>The following methods are available in all meta-analysis functions
to estimate the between-study variance <code class="reqn">\tau^2</code>.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b></td><td style="text-align: left;"> <b>Method</b> </td>
</tr>
<tr>
 <td style="text-align: left;"> 
<code>method.tau = "REML"</code>
 </td><td style="text-align: left;"> Restricted maximum-likelihood estimator (Viechtbauer, 2005) </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (default) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "PM"</code>
 </td><td style="text-align: left;"> Paule-Mandel estimator (Paule and Mandel, 1982) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "DL"</code>
 </td><td style="text-align: left;"> DerSimonian-Laird estimator (DerSimonian and Laird, 1986) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "ML"</code>
 </td><td style="text-align: left;"> Maximum-likelihood estimator (Viechtbauer, 2005) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "HS"</code>
 </td><td style="text-align: left;"> Hunter-Schmidt estimator (Hunter and Schmidt, 2015) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "SJ"</code>
 </td><td style="text-align: left;"> Sidik-Jonkman estimator (Sidik and Jonkman, 2005) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "HE"</code>
 </td><td style="text-align: left;"> Hedges estimator (Hedges and Olkin, 1985) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau = "EB"</code>
 </td><td style="text-align: left;"> Empirical Bayes estimator (Morris, 1983)
</td>
</tr>

</table>

<p>For GLMMs, only the maximum-likelihood method is available.
</p>
<p>Historically, the DerSimonian-Laird method was the de facto
standard to estimate the between-study variance <code class="reqn">\tau^2</code> and is
the default in some software packages including Review Manager 5
(RevMan 5) and R package <b>meta</b>, version 4 and below. However,
its role has been challenged and especially the REML and
Paule-Mandel estimators have been recommended (Veroniki et al.,
2016; Langan et al., 2019). Accordingly, the currenct default in R
package <b>meta</b> is the REML estimator.
</p>
<p>The following R command could be used to employ the Paule-Mandel
instead of the REML estimator in all meta-analyses of the current R
session:
</p>

<ul>
<li> <p><code>settings.meta(method.tau = "PM")</code>
</p>
</li></ul>

<p>Other estimators for <code class="reqn">\tau^2</code> could be selected in a similar
way.
</p>
<p>Note, for binary outcomes, two variants of the DerSimonian-Laird
estimator are available if the Mantel-Haenszel method is used for
pooling. If argument <code>Q.Cochrane = TRUE</code> (default), the
heterogeneity statistic Q is based on the Mantel-Haenszel instead
of the inverse variance estimator under the common effect
model. This is the estimator for <code class="reqn">\tau^2</code> implemented in RevMan
5.
</p>



<h4>Confidence interval for random effects estimate</h4>

<p>The following methods are available in all meta-analysis functions
to calculate a confidence interval for the random effects estimate.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b></td><td style="text-align: left;"> <b>Method</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci = "classic"</code></td><td style="text-align: left;"> Based on standard normal
  quantile </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (DerSimonian and Laird, 1986) (default) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci = "HK"</code></td><td style="text-align: left;"> Method by Hartung and Knapp
  (2001a/b) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci = "KR"</code></td><td style="text-align: left;"> Kenward-Roger method (Partlett and
  Riley, 2017)
</td>
</tr>

</table>

<p>DerSimonian and Laird (1986) introduced the classic random effects
model using a quantile of the standard normal distribution to
calculate a confidence interval for the random effects
estimate. This method implicitly assumes that the weights in the
random effects meta-analysis are not estimated but
given. Particularly, the uncertainty in the estimation of the
between-study variance <code class="reqn">\tau^2</code> is ignored.
</p>
<p>Hartung and Knapp (2001a,b) proposed an alternative method for
random effects meta-analysis based on a refined variance estimator
for the treatment estimate and a quantile of a
<em>t</em>-distribution with <em>k-1</em> degrees of freedom where
<em>k</em> corresponds to the number of studies in the
meta-analysis.
</p>
<p>The Kenward-Roger method is only available for the REML estimator
(<code>method.tau = "REML"</code>) of the between-study variance
<code class="reqn">\tau^2</code> (Partlett and Riley, 2017). This method is based on an
adjusted variance estimate for the random effects
estimate. Furthermore, a quantile of a <em>t</em>-distribution with
adequately modified degrees of freedom is used to calculate the
confidence interval.
</p>
<p>For GLMMs and three-level models, the Kenward-Roger method is not
available, but a method similar to Knapp and Hartung (2003) is used
if <code>method.random.ci = "HK"</code>. For this method, the variance
estimator is not modified, however, a quantile of a
<em>t</em>-distribution with <em>k-1</em> degrees of freedom is used;
see description of argument <code>test</code> in
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> and <code><a href="metafor.html#topic+rma.mv">rma.mv</a></code>.
</p>
<p>Simulation studies (Hartung and Knapp, 2001a,b; IntHout et al.,
2014; Langan et al., 2019) show improved coverage probabilities of
the Hartung-Knapp method compared to the classic random effects
method. However, in rare settings with very homogeneous treatment
estimates, the Hartung-Knapp variance estimate can be arbitrarily
small resulting in a very narrow confidence interval (Knapp and
Hartung, 2003; Wiksten et al., 2016). In such cases, an <em>ad
hoc</em> variance correction has been proposed by utilising the
variance estimate from the classic random effects model with the
Hartung-Knapp method (Knapp and Hartung, 2003; IQWiQ, 2022). An
alternative <em>ad hoc</em> approach is to use the confidence
interval of the classic common or random effects meta-analysis if
it is wider than the interval from the Hartung-Knapp method
(Wiksten et al., 2016; Jackson et al., 2017).
</p>
<p>Argument <code>adhoc.hakn.ci</code> can be used to choose the <em>ad
hoc</em> correction for the Hartung-Knapp (HK) method:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b></td><td style="text-align: left;"> <b><em>Ad hoc</em> method</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.ci = ""</code></td><td style="text-align: left;"> no <em>ad hoc</em> correction (default)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.ci = "se"</code></td><td style="text-align: left;"> use variance correction if HK standard
 error is smaller </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> than standard error from classic random effects
 </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> meta-analysis (Knapp and Hartung, 2003) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.ci = "IQWiG6"</code></td><td style="text-align: left;"> use variance correction if HK
 confidence interval </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> is narrower than CI from classic random effects model </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> with DerSimonian-Laird estimator (IQWiG, 2022) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.ci = "ci"</code></td><td style="text-align: left;"> use wider confidence interval of
 classic random effects </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> and HK meta-analysis </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> (Hybrid method 2 in Jackson et al., 2017)
</td>
</tr>

</table>

<p>For GLMMs and three-level models, the <em>ad hoc</em> variance
corrections are not available.
</p>



<h4>Prediction interval</h4>

<p>The following methods are available in all meta-analysis functions
to calculate a prediction interval for the treatment effect in a
single new study.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b></td><td style="text-align: left;"> <b>Method</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.predict = "HTS"</code></td><td style="text-align: left;"> Based on <em>t</em>-distribution
  (Higgins et al., 2009) (default) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.predict = "HK"</code></td><td style="text-align: left;"> Hartung-Knapp method (Partlett and
  Riley, 2017) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.predict = "KR"</code></td><td style="text-align: left;"> Kenward-Roger method (Partlett and
  Riley, 2017) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.predict = "NNF"</code></td><td style="text-align: left;"> Bootstrap approach (Nagashima et
  al., 2019) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.predict = "S"</code></td><td style="text-align: left;"> Based on standard normal quantile
  (Skipka, 2006)
</td>
</tr>

</table>

<p>By default (<code>method.predict = "HTS"</code>), the prediction interval
is based on a <em>t</em>-distribution with <em>k-2</em> degrees of
freedom where <em>k</em> corresponds to the number of studies in the
meta-analysis, see equation (12) in Higgins et al. (2009).
</p>
<p>The Kenward-Roger method is only available for the REML estimator
(<code>method.tau = "REML"</code>) of the between-study variance
<code class="reqn">\tau^2</code> (Partlett and Riley, 2017). This method is based on an
adjusted variance estimate for the random effects
estimate. Furthermore, a quantile of a <em>t</em>-distribution with
adequately modified degrees of freedom minus 1 is used to calculate
the prediction interval.
</p>
<p>The bootstrap approach is only available if R package <b>pimeta</b>
is installed (Nagashima et al., 2019). Internally, the
<code><a href="pimeta.html#topic+pima">pima</a></code> function is called with argument
<code>method = "boot"</code>. Argument <code>seed.predict</code> can be used to
get a reproducible bootstrap prediction interval and argument
<code>seed.predict.subgroup</code> for reproducible bootstrap prediction
intervals in subgroups.
</p>
<p>The method of Skipka (2006) ignores the uncertainty in the
estimation of the between-study variance <code class="reqn">\tau^2</code> and thus has
too narrow limits for meta-analyses with a small number of studies.
</p>
<p>For GLMMs and three-level models, the Kenward-Roger method and the
bootstrap approach are not available, but a method similar to Knapp
and Hartung (2003) is used if <code>method.random.ci = "HK"</code>. For
this method, the variance estimator is not modified, however, a
quantile of a <em>t</em>-distribution with <em>k-2</em> degrees of
freedom is used; see description of argument <code>test</code> in
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> and <code><a href="metafor.html#topic+rma.mv">rma.mv</a></code>.
</p>
<p>Argument <code>adhoc.hakn.pi</code> can be used to choose the <em>ad
hoc</em> correction for the Hartung-Knapp method:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b></td><td style="text-align: left;"> <b><em>Ad hoc</em> method</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.pi = ""</code></td><td style="text-align: left;"> no <em>ad hoc</em> correction (default)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.pi = "se"</code></td><td style="text-align: left;"> use variance correction if HK
 standard error is smaller
</td>
</tr>

</table>

<p>For GLMMs and three-level models, the <em>ad hoc</em> variance
corrections are not available.
</p>



<h4>Confidence interval for the between-study variance</h4>

<p>The following methods are available in all meta-analysis functions
to calculate a confidence interval for <code class="reqn">\tau^2</code> and <code class="reqn">\tau</code>.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b></td><td style="text-align: left;"> <b>Method</b> </td>
</tr>
<tr>
 <td style="text-align: left;"> 
<code>method.tau.ci = "J"</code></td><td style="text-align: left;"> Method by Jackson (2013) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau.ci = "BJ"</code></td><td style="text-align: left;"> Method by Biggerstaff and Jackson (2008) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau.ci = "QP"</code></td><td style="text-align: left;"> Q-Profile method (Viechtbauer, 2007) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau.ci = "PL"</code></td><td style="text-align: left;"> Profile-Likelihood method for three-level </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> meta-analysis model (Van den Noortgate et al., 2013) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau.ci = ""</code></td><td style="text-align: left;"> No confidence interval
</td>
</tr>

</table>

<p>The first three methods have been recommended by Veroniki et
al. (2016). By default, the Jackson method is used for the
DerSimonian-Laird estimator of <code class="reqn">\tau^2</code> and the Q-profile
method for all other estimators of <code class="reqn">\tau^2</code>.
</p>
<p>The Profile-Likelihood method is the only method available for the
three-level meta-analysis model.
</p>
<p>For GLMMs, no confidence intervals for <code class="reqn">\tau^2</code> and <code class="reqn">\tau</code>
are calculated.
</p>



<h4>Change default settings for R session</h4>

<p>R function <code><a href="#topic+settings.meta">settings.meta</a></code> can be used to change the
previously described and several other default settings for the
current R session.
</p>
<p>Some pre-defined general settings are available:
</p>

<ul>
<li> <p><code>settings.meta("RevMan5")</code>
</p>
</li>
<li> <p><code>settings.meta("JAMA")</code>
</p>
</li>
<li> <p><code>settings.meta("BMJ")</code>
</p>
</li>
<li> <p><code>settings.meta("IQWiG5")</code>
</p>
</li>
<li> <p><code>settings.meta("IQWiG6")</code>
</p>
</li>
<li> <p><code>settings.meta("geneexpr")</code>
</p>
</li></ul>

<p>The first command can be used to reproduce meta-analyses from
Cochrane reviews conducted with <em>Review Manager 5</em> (RevMan 5,
<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>)
and specifies to use a RevMan 5 layout in forest plots.
</p>
<p>The second command can be used to generate forest plots following
instructions for authors of the <em>Journal of the American
Medical Association</em>
(<a href="https://jamanetwork.com/journals/jama/pages/instructions-for-authors/">https://jamanetwork.com/journals/jama/pages/instructions-for-authors/</a>). Study
labels according to JAMA guidelines can be generated using
<code><a href="#topic+labels.meta">labels.meta</a></code>.
</p>
<p>The third command can be used to generate forest plots in the current layout
of the <em>British Medical Journal</em>.
</p>
<p>The next two commands implement the recommendations of the
Institute for Quality and Efficiency in Health Care (IQWiG),
Germany accordinging to General Methods 5 and 6, respectively
(<a href="https://www.iqwig.de/en/about-us/methods/methods-paper/">https://www.iqwig.de/en/about-us/methods/methods-paper/</a>).
</p>
<p>The last setting can be used to print p-values in scientific
notation and to suppress the calculation of confidence intervals
for the between-study variance.
</p>
<p>See <code><a href="#topic+settings.meta">settings.meta</a></code> for more details on these
pre-defined general settings.
</p>
<p>In addition, <code><a href="#topic+settings.meta">settings.meta</a></code> can be used to define
individual settings for the current R session. For example, the
following R command specifies the use of Hartung-Knapp and
Paule-Mandel method, and the printing of prediction intervals for
any meta-analysis generated after execution of this command:
</p>

<ul>
<li> <p><code>settings.meta(method.random.ci = "HK", method.tau =
  "PM", prediction = TRUE)</code>
</p>
</li></ul>




<h4>Data sets</h4>

<p>The following data sets are available in R package <b>meta</b>.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Data set</b></td><td style="text-align: left;"> <b>Description</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+Fleiss1993bin">Fleiss1993bin</a></code></td><td style="text-align: left;"> Aspirin after myocardial infarction </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+Fleiss1993cont">Fleiss1993cont</a></code></td><td style="text-align: left;"> Mental health treatment on medical utilisation</td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+Olkin1995">Olkin1995</a></code></td><td style="text-align: left;"> Thrombolytic therapy after acute myocardial infarction </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+Pagliaro1992">Pagliaro1992</a></code></td><td style="text-align: left;"> Prevention of first bleeding in cirrhosis </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+amlodipine">amlodipine</a></code></td><td style="text-align: left;"> Amlodipine for work capacity </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+caffeine">caffeine</a></code></td><td style="text-align: left;"> Caffeine for daytime drowsiness (Cochrane Practice review) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+cisapride">cisapride</a></code></td><td style="text-align: left;"> Cisapride in non-ulcer dispepsia </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+lungcancer">lungcancer</a></code></td><td style="text-align: left;"> Smoking example </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+smoking">smoking</a></code></td><td style="text-align: left;"> Smoking example </td>
</tr>
<tr>
 <td style="text-align: left;">
<code><a href="#topic+woodyplants">woodyplants</a></code></td><td style="text-align: left;"> Elevated CO$_2$ and total biomass of woody plants
</td>
</tr>

</table>

<p><b>R</b> package <b>metadat</b> has a large collection of meta-analysis data
sets.
</p>



<h3>Note</h3>

<p>Balduzzi et al. (2019) is the preferred citation in publications
for <b>meta</b>. Type <code>citation("meta")</code> for a BibTeX entry of
this publication.
</p>
<p>Type <code>help(package = "meta")</code> for a listing of all R functions
and datasets available in <b>meta</b>. For example, results of
several meta-analyses can be combined with <code><a href="#topic+metabind">metabind</a></code>
which is useful to generate a forest plot with results of several
subgroup analyses.
</p>
<p>R package <b>meta</b> imports R functions from <b>metafor</b>
(Viechtbauer, 2010) to
</p>

<ul>
<li><p> estimate the between-study variance <code class="reqn">\tau^2</code>,
</p>
</li>
<li><p> conduct meta-regression,
</p>
</li>
<li><p> estimate three-level models,
</p>
</li>
<li><p> estimate generalised linear mixed models.
</p>
</li></ul>

<p>To report problems and bugs
</p>

<ul>
<li><p> type <code>bug.report(package = "meta")</code> if you do not use
RStudio,
</p>
</li>
<li><p> send an email to Guido Schwarzer
<a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a> if you use RStudio.
</p>
</li></ul>

<p>The development version of <b>meta</b> is available on GitHub
<a href="https://github.com/guido-s/meta/">https://github.com/guido-s/meta/</a>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Balduzzi S, Rücker G, Schwarzer G (2019):
How to perform a meta-analysis with R: a practical tutorial.
<em>Evidence-Based Mental Health</em>,
<b>22</b>, 153&ndash;160
</p>
<p>Biggerstaff BJ, Jackson D (2008):
The exact distribution of Cochran’s heterogeneity statistic in
one-way random effects meta-analysis.
<em>Statistics in Medicine</em>,
<b>27</b>, 6093&ndash;110
</p>
<p>DerSimonian R &amp; Laird N (1986):
Meta-analysis in clinical trials.
<em>Controlled Clinical Trials</em>,
<b>7</b>, 177&ndash;88
</p>
<p>Hartung J, Knapp G (2001a):
On tests of the overall treatment effect in meta-analysis with
normally distributed responses.
<em>Statistics in Medicine</em>,
<b>20</b>, 1771&ndash;82
</p>
<p>Hartung J, Knapp G (2001b):
A refined method for the meta-analysis of controlled clinical
trials with binary outcome.
<em>Statistics in Medicine</em>,
<b>20</b>, 3875&ndash;89
</p>
<p>Hedges LV &amp; Olkin I (1985):
<em>Statistical methods for meta-analysis</em>.
San Diego, CA: Academic Press
</p>
<p>Higgins JPT, Thompson SG, Spiegelhalter DJ (2009):
A re-evaluation of random-effects meta-analysis.
<em>Journal of the Royal Statistical Society: Series A</em>,
<b>172</b>, 137&ndash;59
</p>
<p>Hunter JE &amp; Schmidt FL (2015):
<em>Methods of Meta-Analysis: Correcting Error and Bias in
Research Findings</em> (Third edition).
Thousand Oaks, CA: Sage
</p>
<p>IntHout J, Ioannidis JPA, Borm GF (2014):
The Hartung-Knapp-Sidik-Jonkman method for random effects
meta-analysis is straightforward and considerably outperforms the
standard DerSimonian-Laird method.
<em>BMC Medical Research Methodology</em>,
<b>14</b>, 25
</p>
<p>IQWiG (2022):
General Methods: Version 6.1.
<a href="https://www.iqwig.de/en/about-us/methods/methods-paper/">https://www.iqwig.de/en/about-us/methods/methods-paper/</a>
</p>
<p>Jackson D (2013):
Confidence intervals for the between-study variance in random
effects meta-analysis using generalised Cochran heterogeneity
statistics.
<em>Research Synthesis Methods</em>,
<b>4</b>, 220&ndash;229
</p>
<p>Jackson D, Law M, Rücker G, Schwarzer G (2017): 
The Hartung-Knapp modification for random-effects meta-analysis: A
useful refinement but are there any residual concerns?
<em>Statistics in Medicine</em>,
<b>36</b>, 3923&ndash;34
</p>
<p>Knapp G &amp; Hartung J (2003):
Improved tests for a random effects meta-regression with a single
covariate.
<em>Statistics in Medicine</em>,
<b>22</b>, 2693&ndash;710
</p>
<p>Langan D, Higgins JPT, Jackson D, Bowden J, Veroniki AA,
Kontopantelis E, et al. (2019):
A comparison of heterogeneity variance estimators in simulated
random-effects meta-analyses.
<em>Research Synthesis Methods</em>,
<b>10</b>, 83&ndash;98
</p>
<p>Schwarzer G (2007):
meta: An R package for meta-analysis.
<em>R News</em>,
<b>7</b>, 40&ndash;5
</p>
<p>Schwarzer G, Carpenter JR and Rücker G (2015):
<em>Meta-Analysis with R (Use-R!)</em>.
Springer International Publishing, Switzerland
</p>
<p>Skipka G (2006):
The inclusion of the estimated inter-study variation into forest
plots for random effects meta-analysis - a suggestion for a
graphical representation [abstract].
<em>XIV Cochrane Colloquium, Dublin</em>, 23-26.
</p>
<p>Stijnen T, Hamza TH, Ozdemir P (2010):
Random effects meta-analysis of event outcome in the framework of
the generalized linear mixed model with applications in sparse
data.
<em>Statistics in Medicine</em>,
<b>29</b>, 3046&ndash;67
</p>
<p>Veroniki AA, Jackson D, Viechtbauer W, Bender R, Bowden J, Knapp G,
et al. (2016):
Methods to estimate the between-study variance and its uncertainty
in meta-analysis.
<em>Research Synthesis Methods</em>,
<b>7</b>, 55&ndash;79 
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Viechtbauer W (2005):
Bias and efficiency of meta-analytic variance estimators in the
random-effects model.
<em>Journal of Educational and Behavioral Statistics</em>,
<b>30</b>, 261&ndash;93
</p>
<p>Viechtbauer W (2007):
Confidence intervals for the amount of heterogeneity in
meta-analysis.
<em>Statistics in Medicine</em>,
<b>26</b>, 37&ndash;52
</p>
<p>Viechtbauer W (2010):
Conducting Meta-Analyses in R with the metafor Package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>
<p>Wiksten A, Rücker G, Schwarzer G (2016):
Hartung-Knapp method is not always conservative compared with
fixed-effect meta-analysis.
<em>Statistics in Medicine</em>,
<b>35</b>, 2503&ndash;15
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-object">meta-object</a></code>, <code><a href="#topic+meta-sm">meta-sm</a></code>
</p>

<hr>
<h2 id='meta-sm'>Description of summary measures available in R package <b>meta</b></h2><span id='topic+meta-sm'></span>

<h3>Description</h3>

<p>Description of summary measures available in R package <b>meta</b>
</p>


<h3>Details</h3>

<p>The following summary measures (argument <code>sm</code>) are recognized
in R package <b>meta</b>.
</p>


<h4>Meta-analysis of binary outcome data (<code><a href="#topic+metabin">metabin</a>)</code></h4>


<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "OR"</code> </td><td style="text-align: left;"> Odds ratio (Fleiss, 1993)</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "RR"</code> </td><td style="text-align: left;"> Risk ratio (Fleiss, 1993) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "RD"</code> </td><td style="text-align: left;"> Risk difference (Fleiss, 1993) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "ASD"</code> </td><td style="text-align: left;"> Arcsine difference (Rücker et al., 2009) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "DOR"</code> </td><td style="text-align: left;"> Diagnostic odds ratio (Moses et al., 1993)
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "VE"</code> </td><td style="text-align: left;"> Vaccine efficacy or vaccine effectiveness
</td>
</tr>

</table>

<p>Note, mathematically, odds ratios and diagnostic odds ratios are
identical, however, the labels in printouts and figures
differ. Furthermore, log risk ratio (logRR) and log vaccine ratio
(logVR) are mathematical identical, however, back-transformed
results differ as vaccine efficacy or effectiveness is defined as
<code>VE = 100 * (1 - RR)</code>.
</p>
<p>A continuity correction is used for some summary measures in the
case of a zero cell count (see <code><a href="#topic+metabin">metabin</a></code>). 
</p>
<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain transformed values, e.g., log odds ratios, log risk ratios
or log vaccine ratios.  In printouts and plots transformed values
are back transformed if argument <code>backtransf = TRUE</code>
(default), with exception of the arcsine difference where no
back-transformation exists. Auxiliary function
<code><a href="#topic+logVR2VE">logVR2VE</a></code> is used to back-transform log vaccine ratios
to vaccine efficacy or effectiveness while <code><a href="base.html#topic+exp">exp</a></code>
is used to back-transform log odds or risk ratios.
</p>



<h4>Meta-analysis of continuous outcome data (<code><a href="#topic+metacont">metacont</a>)</code></h4>


<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "MD"</code> </td><td style="text-align: left;"> Mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "SMD"</code> </td><td style="text-align: left;"> Standardised mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "ROM"</code> </td><td style="text-align: left;"> Ratio of means
</td>
</tr>

</table>

<p>Three variants to calculate the standardised mean difference are
available (see <code><a href="#topic+metacont">metacont</a></code>).
</p>
<p>For the ratio of means, list elements <code>TE</code>, <code>TE.common</code>,
<code>TE.random</code>, etc., contain the log transformed ratio of
means. In printouts and plots these values are back transformed
using <code><a href="base.html#topic+exp">exp</a></code> if argument <code>backtransf = TRUE</code>
(default).
</p>



<h4>Meta-analysis of correlations (<code><a href="#topic+metacor">metacor</a>)</code></h4>


<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "ZCOR"</code> </td><td style="text-align: left;"> Fisher's z transformed correlation </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "COR"</code> </td><td style="text-align: left;"> Untransformed correlations
</td>
</tr>

</table>

<p>For Fisher's z transformed correlations, list elements <code>TE</code>,
<code>TE.common</code>, <code>TE.random</code>, etc., contain the transformed
correlations. In printouts and plots these values are back
transformed using auxiliary function <code><a href="#topic+z2cor">z2cor</a></code> if
argument <code>backtransf = TRUE</code> (default).
</p>



<h4>Meta-analysis of incidence rates (<code><a href="#topic+metainc">metainc</a>)</code></h4>


<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IRR"</code> </td><td style="text-align: left;"> Incidence rate ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IRD"</code> </td><td style="text-align: left;"> Incidence rate difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IRSD"</code> </td><td style="text-align: left;"> Square root transformed incidence rate
  difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "VE"</code> </td><td style="text-align: left;"> Vaccine efficacy or vaccine effectiveness
</td>
</tr>

</table>

<p>Note, log incidence rate ratio (logIRR) and log vaccine ratio
(logVR) are mathematical identical, however, back-transformed
results differ as vaccine efficacy or effectiveness is defined as
<code>VE = 100 * (1 - IRR)</code>.
</p>
<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain the transformed incidence rates. In printouts and plots
these values are back transformed if argument <code>backtransf =
TRUE</code> (default). For back-transformation, <code><a href="base.html#topic+exp">exp</a></code>
is used for the incidence rate ratio, power of 2 is used for square
root transformed rates and <code><a href="#topic+logVR2VE">logVR2VE</a></code> is used for
vaccine efficacy / effectiveness.
</p>



<h4>Meta-analysis of single means (<code><a href="#topic+metamean">metamean</a>)</code></h4>


<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "MRAW"</code> </td><td style="text-align: left;"> Raw, i.e. untransformed, means </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "MLN"</code> </td><td style="text-align: left;"> Log transformed means
</td>
</tr>

</table>

<p>Calculations are conducted on the log scale if <code>sm =
"MLN"</code>. Accordingly, list elements <code>TE</code>, <code>TE.common</code>, and
<code>TE.random</code> contain the logarithm of means. In printouts and
plots these values are back transformed using
<code><a href="base.html#topic+exp">exp</a></code> if argument <code>backtransf = TRUE</code>.
</p>



<h4>Meta-analysis of single proportions (<code><a href="#topic+metaprop">metaprop</a>)</code></h4>

<p>The following transformations of proportions are
implemented to calculate an overall proportion:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "PLOGIT"</code> </td><td style="text-align: left;"> Logit transformation </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "PAS"</code> </td><td style="text-align: left;"> Arcsine transformation </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "PFT"</code> </td><td style="text-align: left;"> Freeman-Tukey Double arcsine transformation
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "PLN"</code> </td><td style="text-align: left;"> Log transformation </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "PRAW"</code> </td><td style="text-align: left;"> No transformation
</td>
</tr>

</table>

<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain the transformed proportions. In printouts and plots these
values are back transformed if argument <code>backtransf = TRUE</code>
(default). For back-transformation, <code><a href="#topic+logit2p">logit2p</a></code> is used
for logit transformed proportions, <code><a href="#topic+asin2p">asin2p</a></code> is used for
(Freeman-Tukey) arcsine transformed proportions and
<code><a href="base.html#topic+exp">exp</a></code> is used for log transformed proportions.
</p>



<h4>Meta-analysis of single rates (<code><a href="#topic+metarate">metarate</a>)</code></h4>

<p>The following transformations of incidence rates are implemented to
calculate an overall rate:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IRLN"</code> </td><td style="text-align: left;"> Log transformation </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IRS"</code> </td><td style="text-align: left;"> Square root transformation </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IRFT"</code> </td><td style="text-align: left;"> Freeman-Tukey Double arcsine transformation
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "IR"</code> </td><td style="text-align: left;"> No transformation
</td>
</tr>

</table>

<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain the transformed incidence rates. In printouts and plots
these values are back transformed if argument <code>backtransf =
TRUE</code> (default). For back-transformation, <code><a href="base.html#topic+exp">exp</a></code>
is used for log transformed rates, power of 2 is used for square
root transformed rates and <code><a href="#topic+asin2ir">asin2ir</a></code> is used for
Freeman-Tukey arcsine transformed rates.
</p>



<h4>Generic inverse variance method (<code><a href="#topic+metagen">metagen</a>)</code></h4>

<p>The following summary measures are recognised in addition to the
above mentioned summary measures:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Summary measure</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "HR"</code> </td><td style="text-align: left;"> Hazard ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>sm = "VE"</code> </td><td style="text-align: left;"> Vaccine efficacy or vaccine effectiveness
</td>
</tr>

</table>

<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain transformed values, i.e., log hazard ratios and log vaccine
ratios. In printouts and plots these values are back transformed if
argument <code>backtransf = TRUE</code> (default); see also
<a href="#topic+meta-transf">meta-transf</a>.
</p>



<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Borenstein M, Hedges LV, Higgins JP, Rothstein HR (2010):
A basic introduction to fixed-effect and random-effects models for
meta-analysis.
<em>Research Synthesis Methods</em>,
<b>1</b>, 97&ndash;111
</p>
<p>Fleiss JL (1993):
The statistical basis of meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>2</b>, 121&ndash;45
</p>
<p>Moses LE, Shapiro D, Littenberg B (1993):
Combining Independent Studies of a Diagnostic Test into a Summary
Roc Curve: Data-Analytic Approaches and Some Additional
Considerations.
<em>Statistics in Medicine</em>,
<b>12</b>, 1293&ndash;1316
</p>
<p>Rücker G, Schwarzer G, Carpenter J, Olkin I (2009):
Why add anything to nothing? The arcsine difference as a measure of
treatment effect in meta-analysis with zero cells.
<em>Statistics in Medicine</em>,
<b>28</b>, 721&ndash;38
</p>
<p>Stijnen T, Hamza TH, Ozdemir P (2010):
Random effects meta-analysis of event outcome in the framework of
the generalized linear mixed model with applications in sparse
data.
<em>Statistics in Medicine</em>,
<b>29</b>, 3046&ndash;67
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+meta-transf">meta-transf</a></code>,
<code><a href="#topic+meta-object">meta-object</a></code>, <code><a href="#topic+print.meta">print.meta</a></code>,
<code><a href="#topic+summary.meta">summary.meta</a></code>, <code><a href="#topic+forest.meta">forest.meta</a></code>
</p>

<hr>
<h2 id='meta-transf'>Auxiliary functions for (back) transformations</h2><span id='topic+meta-transf'></span><span id='topic+transf'></span><span id='topic+cor2z'></span><span id='topic+p2asin'></span><span id='topic+p2logit'></span><span id='topic+VE2logVR'></span><span id='topic+backtransf'></span><span id='topic+asin2ir'></span><span id='topic+asin2p'></span><span id='topic+logit2p'></span><span id='topic+logVR2VE'></span><span id='topic+z2cor'></span>

<h3>Description</h3>

<p>Auxiliary functions to (back) transform effect estimates or
confidence / prediction interval limit(s).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transf(x, sm, func = NULL, args = NULL)

cor2z(x)

p2asin(x)

p2logit(x)

VE2logVR(x)

backtransf(x, sm, n, time, func = NULL, args = NULL)

asin2ir(x, time = NULL)

asin2p(x, n = NULL)

logit2p(x)

logVR2VE(x)

z2cor(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meta-transf_+3A_x">x</code></td>
<td>
<p>Numerical vector with effect estimates, lower or upper
confidence / prediction interval limit(s).</p>
</td></tr>
<tr><td><code id="meta-transf_+3A_sm">sm</code></td>
<td>
<p>Summary measure.</p>
</td></tr>
<tr><td><code id="meta-transf_+3A_func">func</code></td>
<td>
<p>User-specified function for (back) transformation.</p>
</td></tr>
<tr><td><code id="meta-transf_+3A_args">args</code></td>
<td>
<p>Function arguments for user-specified function.</p>
</td></tr>
<tr><td><code id="meta-transf_+3A_n">n</code></td>
<td>
<p>Sample size(s) to back transform Freeman-Tukey transformed
proportions.</p>
</td></tr>
<tr><td><code id="meta-transf_+3A_time">time</code></td>
<td>
<p>Time(s) to back transform Freeman-Tukey transformed
incidence rates.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Often in a meta-analysis, effect estimates are transformed before
calculating a weighted average. For example, the log odds ratio and
its standard error is used instead of the odds ratio in R function
<code><a href="#topic+metagen">metagen</a></code>. To report the results of a meta-analysis,
effect estimates are typically back transformed to the original
scale. R package <b>meta</b> provides some auxiliary functions for
(back) transformations.
</p>


<h4>Transformations</h4>

<p>The following auxiliary functions are provided by R package
<b>meta</b> to transform effect estimates or confidence /
prediction interval limits.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Function</b> </td><td style="text-align: left;"> <b>Transformation</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>cor2z</code> </td><td style="text-align: left;"> Correlations to Fisher's Z transformed correlations </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p2logit</code> </td><td style="text-align: left;"> Proportions to logit transformed proportions </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>p2asin</code> </td><td style="text-align: left;"> Proportions to arcsine transformed proportions </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>VE2logVR</code> </td><td style="text-align: left;"> Vaccine efficacy / effectiveness to log vaccine ratio
</td>
</tr>

</table>

<p>Note, no function for the Freeman-Tukey arcsine transformation is
provided as this transformation is based on the number of events
and sample sizes instead of the effect estimates.
</p>
<p>R function <code>transf</code> is a wrapper function for the above and
additional transformations, e.g., the log transformation using
<code><a href="base.html#topic+log">log</a></code> for odds or risk ratios. Argument <code>sm</code>
is mandatory to specify the requested transformation. It is also
possible to specify a different function with arguments <code>func</code>
and <code>args</code>.
</p>



<h4>Back transformations</h4>

<p>The following auxiliary functions are available to back transform
effect estimates or confidence / prediction interval limits.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Function</b> </td><td style="text-align: left;"> <b>Transformation</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>asin2ir</code> </td><td style="text-align: left;"> Freeman-Tukey arcsine transformed rates to rates </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>asin2p</code> </td><td style="text-align: left;"> (Freeman-Tukey) arcsine transformed proportions to proportions </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>logit2p</code> </td><td style="text-align: left;"> Logit transformed proportions to proportions </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>logVR2VE</code> </td><td style="text-align: left;"> Log vaccine ratio to vaccine efficacy / effectiveness </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>z2cor</code> </td><td style="text-align: left;"> Fisher's Z transformed correlations  to correlations
</td>
</tr>

</table>

<p>Argument <code>time</code> is mandatory in R function <code>asin2ir</code>.
</p>
<p>If argument <code>n</code> is provided in R function <code>asin2p</code>,
Freeman-Tukey arcsine transformed proportions are
back transformed. Otherwise, arcsine transformed proportions are
back transformed.
</p>
<p>R function <code>backtransf</code> is a wrapper function for the above
and additional transformations, e.g., the exponential
transformation using <code><a href="base.html#topic+exp">exp</a></code> for log odds or log
risk ratios. Argument <code>sm</code> is mandatory to specify the
requested transformation. For the Freeman-Tukey transformations,
argument <code>n</code> or <code>time</code> is mandatory.
</p>
<p>It is also possible to specify a different function with arguments
<code>func</code> and <code>args</code>.
</p>



<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-sm">meta-sm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>logit2p(p2logit(0.5))

</code></pre>

<hr>
<h2 id='metaadd'>Add pooled results from external analysis to meta-analysis</h2><span id='topic+metaadd'></span>

<h3>Description</h3>

<p>Add pooled results from external analysis to an existing
meta-analysis object. This is useful, for example, to add results
from a Bayesian meta-analysis which is not implemented in R package
<b>meta</b>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaadd(
  x,
  type,
  TE,
  lower,
  upper,
  statistic = NA,
  pval = NA,
  text,
  data = NULL,
  method.common = "",
  method.random = "",
  method.tau = "",
  method.random.ci = "",
  method.predict = "",
  transf = gs("transf")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaadd_+3A_x">x</code></td>
<td>
<p>Meta-analysis object.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_type">type</code></td>
<td>
<p>A character string or vector indicating whether added
results are from common effect, random effects model or
prediction interval. Either <code>"common"</code>, <code>"random"</code> or
<code>"prediction"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_te">TE</code></td>
<td>
<p>Pooled estimate(s).</p>
</td></tr>
<tr><td><code id="metaadd_+3A_lower">lower</code></td>
<td>
<p>Lower limit(s) of confidence or prediction interval.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_upper">upper</code></td>
<td>
<p>Upper limit(s) of confidence or prediction interval.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_statistic">statistic</code></td>
<td>
<p>Test statistic(s).</p>
</td></tr>
<tr><td><code id="metaadd_+3A_pval">pval</code></td>
<td>
<p>P-value(s).</p>
</td></tr>
<tr><td><code id="metaadd_+3A_text">text</code></td>
<td>
<p>A character string or vector used in printouts and
forest plot to label the added results.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the new results or an
object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_method.common">method.common</code></td>
<td>
<p>A character string or vector to describe the
common effect method(s).</p>
</td></tr>
<tr><td><code id="metaadd_+3A_method.random">method.random</code></td>
<td>
<p>A character string or vector to describe the
random effects method(s).</p>
</td></tr>
<tr><td><code id="metaadd_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string or vector to describe the
estimator(s) of the between-study variance.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string or vector to describe
the method(s) to calculate confidence intervals under the random
effects model.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string or vector to describe the
method(s) used for prediction intervals.</p>
</td></tr>
<tr><td><code id="metaadd_+3A_transf">transf</code></td>
<td>
<p>A logical indicating whether inputs for arguments
<code>TE</code>, <code>lower</code> and <code>upper</code> are already
appropriately transformed to conduct the meta-analysis or on the
original scale. If <code>transf = TRUE</code> (default), inputs are
expected to be log odds ratios instead of odds ratios for
<code>sm = "OR"</code> and Fisher's z transformed correlations instead
of correlations for <code>sm = "ZCOR"</code>, for example.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In R package <b>meta</b>, objects of class <code>"meta"</code> contain
results of both common effect and random effects
meta-analyses. This function enables the user to add the pooled
results of an additional analysis to an existing meta-analysis
object. This is useful, for example, to add the result of a
Bayesian meta-analysis.
</p>
<p>If argument <code>data</code> is a meta-analysis object created with R
package <b>meta</b>, arguments <code>TE</code>, <code>lower</code>,
<code>upper</code>, <code>statistic</code> and <code>pval</code> are ignored as this
information is extracted from the meta-analysis.
</p>
<p>Otherwise, arguments <code>TE</code>, <code>lower</code> and <code>upper</code> have
to be provided if <code>type = "common"</code> or <code>type =
"random"</code>. For <code>type = "prediction"</code>, only arguments
<code>lower</code> and <code>upper</code> are mandatory.
</p>
<p>Note, R function <code><a href="#topic+metamerge">metamerge</a></code> can be used to add
meta-analysis results of another meta-analysis object (see
<code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Value</h3>

<p>An object of class <code>"meta"</code> with corresponding generic
functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metamerge">metamerge</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)

# Common effect and random effects meta-analysis
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac, data = Fleiss1993bin,
  studlab = paste(study, year), sm = "OR")

# Naive pooling
m2 &lt;- metabin(sum(d.asp), sum(n.asp), sum(d.plac), sum(n.plac),
  data = Fleiss1993bin, sm = "OR", text.common = "Naive pooling")

# Add results of second meta-analysis from common effect model
m12 &lt;- metaadd(m1, data = m2, method.common = "Naive pooling")
m12

forest(m12)

</code></pre>

<hr>
<h2 id='metabias.meta'>Test for funnel plot asymmetry</h2><span id='topic+metabias.meta'></span><span id='topic+metabias'></span><span id='topic+metabias.default'></span><span id='topic+print.metabias'></span>

<h3>Description</h3>

<p>Test for funnel plot asymmetry, based on rank correlation or linear
regression method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
metabias(
  x,
  method.bias = x$method.bias,
  plotit = FALSE,
  correct = FALSE,
  k.min = 10,
  ...
)

## S3 method for class 'metabias'
print(
  x,
  digits = gs("digits"),
  digits.stat = gs("digits.stat"),
  digits.pval = max(gs("digits.pval"), 2),
  digits.se = gs("digits.se"),
  digits.tau2 = gs("digits.tau2"),
  scientific.pval = gs("scientific.pval"),
  big.mark = gs("big.mark"),
  zero.pval = gs("zero.pval"),
  JAMA.pval = gs("JAMA.pval"),
  text.tau2 = gs("text.tau2"),
  ...
)

metabias(x, ...)

## Default S3 method:
metabias(
  x,
  seTE,
  method.bias = "Egger",
  plotit = FALSE,
  correct = FALSE,
  k.min = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metabias.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code> or estimated treatment
effect in individual studies.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used (see Details), can be abbreviated.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_plotit">plotit</code></td>
<td>
<p>A logical indicating whether a plot should be
produced (see Details).</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_correct">correct</code></td>
<td>
<p>A logical indicating whether a continuity corrected
statistic is used for rank correlation tests.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_k.min">k.min</code></td>
<td>
<p>Minimum number of studies to perform test for funnel
plot asymmetry.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits for estimates,
see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_digits.stat">digits.stat</code></td>
<td>
<p>Minimal number of significant digits for z- or
t-value of test for test of funnel plot asymmetry, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_digits.pval">digits.pval</code></td>
<td>
<p>Minimal number of significant digits for p-value
of test for test of funnel plot asymmetry, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_digits.se">digits.se</code></td>
<td>
<p>Minimal number of significant digits for standard
errors, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_digits.tau2">digits.tau2</code></td>
<td>
<p>Minimal number of significant digits for
residual heterogeneity variance, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_scientific.pval">scientific.pval</code></td>
<td>
<p>A logical specifying whether p-values should
be printed in scientific notation, e.g., 1.2345e-01 instead of
0.12345.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_zero.pval">zero.pval</code></td>
<td>
<p>A logical specifying whether p-values should be
printed with a leading zero.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_jama.pval">JAMA.pval</code></td>
<td>
<p>A logical specifying whether p-values for test of
overall effect should be printed according to JAMA reporting
standards.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_text.tau2">text.tau2</code></td>
<td>
<p>Text printed to identify residual heterogeneity
variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metabias.meta_+3A_sete">seTE</code></td>
<td>
<p>Standard error of estimated treatment effect (mandatory
if <code>x</code> not of class <code>meta</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Functions to conduct rank correlation or linear regression tests
for funnel plot asymmetry.
</p>


<h4>Classic generic tests</h4>

<p>The following tests are generic tests for funnel plot asymmetry
which only require estimates of the treatment effect and
corresponding standard errors. Accordingly, these are the only
tests provided by R function <code>metabias.default</code>.
</p>
<p>If argument <code>method.bias</code> is <code>"Begg"</code>, the test statistic
is based on the rank correlation between standardised treatment
estimates and variance estimates of estimated treatment effects;
Kendall's tau is used as correlation measure (Begg &amp; Mazumdar,
1994). The test statistic follows a standard normal
distribution. By default (if <code>correct</code> is FALSE), no
continuity correction is utilised (Kendall &amp; Gibbons, 1990).
</p>
<p>If argument <code>method.bias</code> is <code>"Egger"</code>, the test
statistic is based on a weighted linear regression of the treatment
effect on its standard error (Egger et al., 1997). The test
statistic follows a t distribution with <code>number of studies -
2</code> degrees of freedom.
</p>
<p>If argument <code>method.bias</code> is <code>"Thompson"</code>, the test
statistic is based on a weighted linear regression of the treatment
effect on its standard error using an additive between-study
variance component denoted as methods (3a) - (3d) in Thompson &amp;
Sharp (1999). The test statistic follows a t distribution with
<code>number of studies - 2</code> degrees of freedom.  </p>



<h4>Tests for meta-analysis with binary outcomes</h4>

<p>The following tests for funnel plot asymmetry are only available
for meta-analyses comparing two binary outcomes, i.e. meta-analyses
generated with the <code>metabin</code> function. The only exception is
the test by Peters et al. (2006) which can also be used in a
meta-analysis of single proportions generated with <code>metaprop</code>.
</p>
<p>If argument <code>method.bias</code> is <code>"Harbord"</code>, the test
statistic is based on a weighted linear regression utilising
efficient score and score variance (Harbord et al., 2006,
2009). The test statistic follows a t distribution with
<code>number of studies - 2</code> degrees of freedom.
</p>
<p>In order to calculate an arcsine test for funnel plot asymmetry
(Rücker et al., 2008), one has to use the <code>metabin</code> function
with argument <code>sm = "ASD"</code> as input to the <code>metabias</code>
command. The three arcsine tests described in Rücker et al. (2008)
can be calculated by setting <code>method.bias</code> to <code>"Begg"</code>,
<code>"Egger"</code> and <code>"Thompson"</code>, respectively.
</p>
<p>If argument <code>method.bias</code> is <code>"Macaskill"</code>, the test
statistic is based on a weighted linear regression of the treatment
effect on the total sample size with weights reciprocal to the
variance of the average event probability (Macaskill et al., 2001,
<em>method FPV</em>). The test statistic follows a t distribution
with <code>number of studies - 2</code> degrees of freedom.
</p>
<p>If argument <code>method.bias</code> is <code>"Peters"</code>, the test
statistic is based on a weighted linear regression of the treatment
effect on the inverse of the total sample size with weights
reciprocal to the variance of the average event probability (Peters
et al., 2006). The test statistic follows a t distribution with
<code>number of studies - 2</code> degrees of freedom. Note, this test is
a variant of Macaskill et al. (2001), <em>method FPV</em>, using the
inverse sample size as covariate.
</p>
<p>If argument <code>method.bias</code> is <code>"Schwarzer"</code>, the test
statistic is based on the rank correlation between a standardised
cell frequency and the inverse of the variance of the cell
frequency; Kendall's tau is used as correlation measure (Schwarzer
et al., 2007). The test statistic follows a standard normal
distribution. By default (if <code>correct</code> is FALSE), no
continuity correction is utilised (Kendall &amp; Gibbons, 1990).
</p>
<p>Finally, for meta-analysis of diagnostic test accuracy studies, if
argument <code>method.bias</code> is <code>"Deeks"</code>, the test statistic
is based on a weighted linear regression of the log diagnostic odds
ratio on the inverse of the squared effective sample size using the
effective sample size as weights (Deeks et al., 2005). The test
statistic follows a t distribution with <code>number of studies -
2</code> degrees of freedom.
</p>



<h4>Test for the standardised mean difference</h4>

<p>If argument <code>method.bias</code> is <code>"Pustejovsky"</code>, the test
statistic is based on a weighted linear regression of the treatment
effect on the square root of the sum of the inverse group sample
sizes using the treatment effect variance as weights (Pustejovsky &amp;
Rodgers, 2019). The test statistic follows a t distribution with
<code>number of studies - 2</code> degrees of freedom.
</p>



<h4>Recommendations and default settings</h4>

<p>Following recommendations by Sterne et al. (2011), by default, a
test for funnel plot asymmetry is only conducted if the number of
studies is ten or larger (argument <code>k.min = 10</code>). This
behaviour can be changed by setting a smaller value for argument
<code>k.min</code>. Note, the minimum number of studies is three.
</p>
<p>If argument <code>method.bias</code> is missing, the Harbord test
(<code>method.bias = "Harbord"</code>) is used in meta-analyses with a binary
outcome for the odds ratio and Deeks' test (<code>method.bias = "Deeks"</code>)
for the diagnostic odds ratios. In all other settings, the Egger test
(<code>method.bias = "Egger"</code>) is used (Sterne et al., 2011).
</p>
<p>No test for funnel plot asymmetry is conducted in meta-analyses
with subgroups.
</p>

<p>If argument <code>plotit = TRUE</code>, a scatter plot is shown if
argument <code>method.bias</code> is equal to <code>"Begg"</code>,
<code>"Egger"</code>, <code>"Thompson"</code>, <code>"Harbord"</code>, or
<code>"Deeks"</code>.
</p>


<h3>Value</h3>

<p>A list with class <code>metabias</code> containing the following
components if a test for funnel plot asymmetry is conducted:
</p>
<table>
<tr><td><code>statistic</code></td>
<td>
<p>Test statistic.</p>
</td></tr>
<tr><td><code>df</code></td>
<td>
<p>The degrees of freedom of the test statistic in
the case that it follows a t distribution.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>The p-value for the test.</p>
</td></tr>
<tr><td><code>estimate</code></td>
<td>
<p>Estimates used to calculate test statisic.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>A character string indicating what type of test was
used.</p>
</td></tr>
<tr><td><code>title</code></td>
<td>
<p>Title of Cochrane review.</p>
</td></tr>
<tr><td><code>complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code>outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code>var.model</code></td>
<td>
<p>A character string indicating whether none,
multiplicative, or additive residual heterogeneity variance was
assumed.</p>
</td></tr>
<tr><td><code>method.bias</code></td>
<td>
<p>As defined above.</p>
</td></tr>
<tr><td><code>x</code></td>
<td>
<p>Meta-analysis object.</p>
</td></tr>
<tr><td><code>version</code></td>
<td>
<p>Version of R package <b>meta</b> used to create
object.</p>
</td></tr>
</table>
<p>Or a list with the following elements if test is not conducted due
to the number of studies:
</p>
<table>
<tr><td><code>k</code></td>
<td>
<p>Number of studies in meta-analysis.</p>
</td></tr>
<tr><td><code>k.min</code></td>
<td>
<p>Minimum number of studies to perform test for funnel
plot asymmetry.</p>
</td></tr>
<tr><td><code>version</code></td>
<td>
<p>Version of R package <b>meta</b> used to create
object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Begg CB &amp; Mazumdar M (1994):
Operating characteristics of a rank correlation test for
publication bias.
<em>Biometrics</em>,
<b>50</b>, 1088&ndash;101
</p>
<p>Deeks JJ, Macaskill P, Irwig L (2005):
The performance of tests of publication bias and other sample size
effects in systematic reviews of diagnostic test accuracy was
assessed.
<em>Journal of Clinical Epidemiology</em>,
<b>58</b>:882&ndash;93
</p>
<p>Egger M, Smith GD, Schneider M &amp; Minder C (1997):
Bias in meta-analysis detected by a simple, graphical test.
<em>British Medical Journal</em>,
<b>315</b>, 629&ndash;34
</p>
<p>Harbord RM, Egger M &amp; Sterne J (2006):
A modified test for small-study effects in meta-analyses of
controlled trials with binary endpoints.
<em>Statistics in Medicine</em>,
<b>25</b>, 3443&ndash;57
</p>
<p>Harbord RM, Harris RJ, Sterne JAC (2009):
Updated tests for small-study effects in meta–analyses.
<em>The Stata Journal</em>,
<b>9</b>, 197&ndash;210
</p>
<p>Kendall M &amp; Gibbons JD (1990):
<em>Rank Correlation Methods</em>.
London: Edward Arnold
</p>
<p>Macaskill P, Walter SD, Irwig L (2001):
A comparison of methods to detect publication bias in
meta-analysis.
<em>Statistics in Medicine</em>,
<b>20</b>, 641&ndash;54
</p>
<p>Peters JL, Sutton AJ, Jones DR, Abrams KR &amp; Rushton L (2006):
Comparison of two methods to detect publication bias in
meta-analysis.
<em>Journal of the American Medical Association</em>,
<b>295</b>, 676&ndash;80
</p>
<p>Pustejovsky JE, Rodgers MA (2019):
Testing for funnel plot asymmetry of standardized mean differences.
<em>Research Synthesis Methods</em>,
<b>10</b>, 57&ndash;71
</p>
<p>Rücker G, Schwarzer G, Carpenter JR (2008):
Arcsine test for publication bias in meta-analyses with binary
outcomes.
<em>Statistics in Medicine</em>,
<b>27</b>, 746&ndash;63
</p>
<p>Schwarzer G, Antes G &amp; Schumacher M (2007):
A test for publication bias in meta-analysis with sparse binary
data.
<em>Statistics in Medicine</em>,
<b>26</b>, 721&ndash;33
</p>
<p>Sterne, JAC et al. (2011):
Recommendations for examining and interpreting funnel plot
asymmetry in meta-analyses of randomised controlled trials.
<em>BMJ (Clinical research ed.)</em>,
<b>343</b>, 1
</p>
<p>Thompson SG &amp; Sharp, SJ (1999):
Explaining heterogeneity in meta-analysis: a comparison of methods,
<em>Statistics in Medicine</em>,
<b>18</b>, 2693&ndash;708
</p>


<h3>See Also</h3>

<p><code><a href="#topic+funnel">funnel</a></code>, <code><a href="#topic+funnel.meta">funnel.meta</a></code>,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = 1:10, sm = "RR", method = "I")

metabias(m1)
metabias(m1, plotit = TRUE)

metabias(m1, method.bias = "Begg")
metabias(m1, method.bias = "Begg", correct = TRUE)

metabias(m1, method.bias = "Schwarzer")
metabias(m1, method.bias = "Egger")$pval

# Arcsine test (based on linear regression)
#
m1.as &lt;- update(m1, sm = "ASD")
metabias(m1.as)
# Same result (using function metabias.default)
metabias(m1.as$TE, m1.as$seTE)

# No test for funnel plot asymmetry calculated
#
m2 &lt;- update(m1, subset = 1:5)
metabias(m2)

m3 &lt;- update(m1, subset = 1:2)
metabias(m3)

# Test for funnel plot asymmetry calculated (use of argument k.min)
#
metabias(m2, k.min = 5)

</code></pre>

<hr>
<h2 id='metabias.rm5'>Cochrane review: Test for funnel plot asymmetry</h2><span id='topic+metabias.rm5'></span><span id='topic+metabias.cdir'></span>

<h3>Description</h3>

<p>Conduct a test for funnel plot asymmetry for all outcomes in a
Cochrane review of intervention studies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rm5'
metabias(
  x,
  comp.no,
  outcome.no,
  method.bias = "linreg",
  method.bias.binary = method.bias,
  method.bias.or = "score",
  k.min = 10,
  ...
)

## S3 method for class 'cdir'
metabias(
  x,
  comp.no,
  outcome.no,
  method.bias = "linreg",
  method.bias.binary = method.bias,
  method.bias.or = "score",
  k.min = 10,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metabias.rm5_+3A_x">x</code></td>
<td>
<p>An object of class <code>rm5</code> or <code>cdir</code>.</p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_comp.no">comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_outcome.no">outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test for
small-study effects is to be used for all outcomes. Either
<code>"rank"</code>, <code>"linreg"</code>, or <code>"mm"</code>, can be
abbreviated. See function <code><a href="#topic+metabias">metabias</a></code></p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_method.bias.binary">method.bias.binary</code></td>
<td>
<p>A character string indicating which test
is to be used for binary outcomes. Either <code>"rank"</code>,
<code>"linreg"</code>, <code>"mm"</code>, <code>"count"</code>, <code>"score"</code>, or
<code>"peters"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code></p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_method.bias.or">method.bias.or</code></td>
<td>
<p>A character string indicating which test is
to be used for binary outcomes with odds ratio as summary
measure. Either <code>"rank"</code>, <code>"linreg"</code>, <code>"mm"</code>,
<code>"count"</code>, <code>"score"</code>, or <code>"peters"</code>, can be
abbreviated. See function <code><a href="#topic+metabias">metabias</a></code></p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_k.min">k.min</code></td>
<td>
<p>Minimum number of studies to perform test for
small-study effects.</p>
</td></tr>
<tr><td><code id="metabias.rm5_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored at the moment)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to conduct a test for funnel plot
asymmetry for all or selected meta-analyses in a Cochrane review of
intervention studies (Higgins et al, 2023).
</p>
<p>The R function <code><a href="#topic+metacr">metacr</a></code> is called internally.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Higgins JPT, Thomas J, Chandler J, Cumpston M, Li T, Page MJ, Welch
VA (editors) (2023):
<em>Cochrane Handbook for Systematic Reviews of Interventions
Version 6.4</em>.
Available from <a href="https://training.cochrane.org/handbook">https://training.cochrane.org/handbook</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabias">metabias</a></code>, <code><a href="#topic+metacr">metacr</a></code>,
<code><a href="#topic+read.rm5">read.rm5</a></code>, <code><a href="#topic+read.cdir">read.cdir</a></code>,
<code><a href="#topic+summary.rm5">summary.rm5</a></code>, <code><a href="#topic+summary.cdir">summary.cdir</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate export data file "Fleiss1993_CR.csv" in sub-directory of
# package "meta"
#
filename &lt;- system.file("extdata", "Fleiss1993_CR.csv", package = "meta")
Fleiss1993_CR &lt;- read.rm5(filename)

# Print results for all tests of small-study effects
#
metabias(Fleiss1993_CR, k.min = 5)

# Print result of test of small-study effects for second outcome in
# first comparison
#
metabias(Fleiss1993_CR, comp.no = 1, outcome.no = 2, k.min = 5)

</code></pre>

<hr>
<h2 id='metabin'>Meta-analysis of binary outcome data</h2><span id='topic+metabin'></span>

<h3>Description</h3>

<p>Calculation of common effect and random effects estimates (risk
ratio, odds ratio, risk difference, arcsine difference, or
diagnostic odds ratio) for meta-analyses with binary outcome
data. Mantel-Haenszel, inverse variance, Peto method, generalised
linear mixed model (GLMM), and sample size method are available for
pooling. For GLMMs, the <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function
from R package <b>metafor</b> (Viechtbauer, 2010) is called
internally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metabin(
  event.e,
  n.e,
  event.c,
  n.c,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  method = ifelse(tau.common, "Inverse", gs("method")),
  sm = ifelse(!is.na(charmatch(tolower(method), c("peto", "glmm", "ssw"), nomatch = NA)),
    "OR", gs("smbin")),
  incr = gs("incr"),
  method.incr = gs("method.incr"),
  allstudies = gs("allstudies"),
  level = gs("level"),
  MH.exact = gs("MH.exact"),
  RR.Cochrane = gs("RR.Cochrane"),
  Q.Cochrane = gs("Q.Cochrane") &amp; method == "MH" &amp; method.tau == "DL",
  model.glmm = gs("model.glmm"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = ifelse(!is.na(charmatch(tolower(method), "glmm", nomatch = NA)), "ML",
    gs("method.tau")),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  method.bias = ifelse(sm == "OR", "Harbord", ifelse(sm == "DOR", "Deeks",
    gs("method.bias"))),
  backtransf = gs("backtransf"),
  pscale = 1,
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  label.e = gs("label.e"),
  label.c = gs("label.c"),
  label.left = gs("label.left"),
  label.right = gs("label.right"),
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  hakn,
  adhoc.hakn,
  print.CMH = gs("print.CMH"),
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metabin_+3A_event.e">event.e</code></td>
<td>
<p>Number of events in experimental group or true
positives in diagnostic study.</p>
</td></tr>
<tr><td><code id="metabin_+3A_n.e">n.e</code></td>
<td>
<p>Number of observations in experimental group or number
of ill participants in diagnostic study.</p>
</td></tr>
<tr><td><code id="metabin_+3A_event.c">event.c</code></td>
<td>
<p>Number of events in control group or false positives
in diagnostic study.</p>
</td></tr>
<tr><td><code id="metabin_+3A_n.c">n.c</code></td>
<td>
<p>Number of observations in control group or number of
healthy participants in diagnostic study.</p>
</td></tr>
<tr><td><code id="metabin_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metabin_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information, i.e., event.e, n.e, event.c, and n.c.</p>
</td></tr>
<tr><td><code id="metabin_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metabin_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metabin_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metabin_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metabin_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is to be
used for pooling of studies. One of <code>"Inverse"</code>,
<code>"MH"</code>, <code>"Peto"</code>, <code>"GLMM"</code>, or <code>"SSW"</code>, can
be abbreviated.</p>
</td></tr>
<tr><td><code id="metabin_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"RR"</code>, <code>"OR"</code>, <code>"RD"</code>, <code>"ASD"</code>,
<code>"DOR"</code>, or <code>"VE"</code>) is to be used for pooling of
studies, see Details.</p>
</td></tr>
<tr><td><code id="metabin_+3A_incr">incr</code></td>
<td>
<p>Could be either a numerical value which is added to
cell frequencies for studies with a zero cell count or the
character string <code>"TACC"</code> which stands for treatment arm
continuity correction, see Details.</p>
</td></tr>
<tr><td><code id="metabin_+3A_method.incr">method.incr</code></td>
<td>
<p>A character string indicating which continuity
correction method should be used (<code>"only0"</code>,
<code>"if0all"</code>, or <code>"all"</code>), see Details.</p>
</td></tr>
<tr><td><code id="metabin_+3A_allstudies">allstudies</code></td>
<td>
<p>A logical indicating if studies with zero or all
events in both groups are to be included in the meta-analysis
(applies only if <code>sm</code> is equal to <code>"RR"</code>, <code>"OR"</code>,
or <code>"DOR"</code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metabin_+3A_mh.exact">MH.exact</code></td>
<td>
<p>A logical indicating if <code>incr</code> is not to be
added to cell frequencies for studies with a zero cell count to
calculate the pooled estimate based on the Mantel-Haenszel
method.</p>
</td></tr>
<tr><td><code id="metabin_+3A_rr.cochrane">RR.Cochrane</code></td>
<td>
<p>A logical indicating if 2*<code>incr</code> instead of
1*<code>incr</code> is to be added to <code>n.e</code> and <code>n.c</code> in the
calculation of the risk ratio (i.e., <code>sm="RR"</code>) for studies
with a zero cell. This is used in RevMan 5, the program for
preparing and maintaining Cochrane reviews.</p>
</td></tr>
<tr><td><code id="metabin_+3A_q.cochrane">Q.Cochrane</code></td>
<td>
<p>A logical indicating if the Mantel-Haenszel
estimate is used in the calculation of the heterogeneity
statistic Q which is implemented in RevMan 5, the program for
preparing and maintaining Cochrane reviews.</p>
</td></tr>
<tr><td><code id="metabin_+3A_model.glmm">model.glmm</code></td>
<td>
<p>A character string indicating which GLMM should
be used.  One of <code>"UM.FS"</code>, <code>"UM.RS"</code>, <code>"CM.EL"</code>,
and <code>"CM.AL"</code>, see Details.</p>
</td></tr>
<tr><td><code id="metabin_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metabin_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metabin_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metabin_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metabin_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metabin_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metabin_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metabin_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metabin_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metabin_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metabin_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metabin_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test for
funnel plot asymmetry is to be used. Either <code>"Begg"</code>,
<code>"Egger"</code>, <code>"Thompson"</code>, <code>"Schwarzer"</code>,
<code>"Harbord"</code>, <code>"Peters"</code>, or <code>"Deeks"</code>, can be
abbreviated. See function <code><a href="#topic+metabias">metabias</a>.</code></p>
</td></tr>
<tr><td><code id="metabin_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for odds
ratio (<code>sm="OR"</code>), risk ratio (<code>sm="RR"</code>), or
diagnostic odds ratio (<code>sm="DOR"</code>) should be back
transformed in printouts and plots. If TRUE (default), results
will be presented as odds ratios and risk ratios; otherwise log
odds ratios and log risk ratios will be shown.</p>
</td></tr>
<tr><td><code id="metabin_+3A_pscale">pscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
risk differences.</p>
</td></tr>
<tr><td><code id="metabin_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metabin_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metabin_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metabin_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metabin_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metabin_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metabin_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metabin_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metabin_+3A_label.e">label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code id="metabin_+3A_label.c">label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code id="metabin_+3A_label.left">label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code id="metabin_+3A_label.right">label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code id="metabin_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metabin_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metabin_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metabin_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metabin_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metabin_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metabin_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metabin_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metabin_+3A_hakn">hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'method.random.ci').</p>
</td></tr>
<tr><td><code id="metabin_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by
'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metabin_+3A_print.cmh">print.CMH</code></td>
<td>
<p>A logical indicating whether result of the
Cochran-Mantel-Haenszel test for overall effect should be
printed.</p>
</td></tr>
<tr><td><code id="metabin_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metabin_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if <code>incr</code> is added to studies with zero cell
frequencies).</p>
</td></tr>
<tr><td><code id="metabin_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metabin_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> or
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="metabin_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function and to catch deprecated
arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculation of common and random effects estimates for meta-analyses
with binary outcome data.
</p>
<p>The following measures of treatment effect are available (Rücker et
al., 2009):
</p>

<ul>
<li><p> Risk ratio (<code>sm = "RR"</code>)
</p>
</li>
<li><p> Odds ratio (<code>sm = "OR"</code>)
</p>
</li>
<li><p> Risk difference (<code>sm = "RD"</code>)
</p>
</li>
<li><p> Arcsine difference (<code>sm = "ASD"</code>)
</p>
</li>
<li><p> Diagnostic Odds ratio (<code>sm = "DOR"</code>)
</p>
</li>
<li><p> Vaccine efficacy or vaccine effectiveness (<code>sm = "VE"</code>)
</p>
</li></ul>

<p>Note, mathematically, odds ratios and diagnostic odds ratios are
identical, however, the labels in printouts and figures
differ. Furthermore, log risk ratio (logRR) and log vaccine ratio
(logVR) are mathematical identical, however, back-transformed
results differ as vaccine efficacy or effectiveness is defined as
<code>VE = 100 * (1 - RR)</code>.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>


<h4>Meta-analysis method</h4>

<p>By default, both common effect (also called common effect) and
random effects models are considered (see arguments <code>common</code>
and <code>random</code>). If <code>method</code> is <code>"MH"</code> (default), the
Mantel-Haenszel method (Greenland &amp; Robins, 1985; Robins et al.,
1986) is used to calculate the common effect estimate; if
<code>method</code> is <code>"Inverse"</code>, inverse variance weighting is
used for pooling (Fleiss, 1993); if <code>method</code> is <code>"Peto"</code>,
the Peto method is used for pooling (Yusuf et al., 1985); if
<code>method</code> is <code>"SSW"</code>, the sample size method is used for
pooling (Bakbergenuly et al., 2020).
</p>
<p>While the Mantel-Haenszel and Peto method are defined under the
common effect model, random effects variants based on these methods
are also implemented in <code>metabin</code>. Following RevMan 5, the
Mantel-Haenszel estimator is used in the calculation of the
between-study heterogeneity statistic Q which is used in the
DerSimonian-Laird estimator (DerSimonian and Laird,
1986). Accordingly, the results for the random effects
meta-analysis using the Mantel-Haenszel or inverse variance method
are typically very similar. For the Peto method, Peto's log odds
ratio, i.e. <code>(O-E) / V</code> and its standard error <code>sqrt(1 /
V)</code> with <code>O-E</code> and <code>V</code> denoting &quot;Observed minus Expected&quot;
and its variance, are utilised in the random effects
model. Accordingly, results of a random effects model using
<code>sm = "Peto"</code> can be different to results from a random
effects model using <code>sm = "MH"</code> or <code>sm =
"Inverse"</code>. Note, the random effects estimate is based on the
inverse variance method for all methods discussed so far.
</p>
<p>A distinctive and frequently overlooked advantage of binary
endpoints is that individual patient data (IPD) can be extracted
from a two-by-two table.  Accordingly, statistical methods for IPD,
i.e., logistic regression and generalised linear mixed models, can
be utilised in a meta-analysis of binary outcomes (Stijnen et al.,
2010; Simmonds et al., 2016). These methods are available (argument
<code>method = "GLMM"</code>) for the odds ratio as summary measure for
the common effect and random effects model by calling the
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function from R package
<b>metafor</b> internally.
</p>
<p>Four different GLMMs are available for
meta-analysis with binary outcomes using argument <code>model.glmm</code>
(which corresponds to argument <code>model</code> in the
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function):
</p>

<table>
<tr>
 <td style="text-align: center;">
1. </td><td style="text-align: left;"> Logistic regression model with common study effects
 (default) </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "UM.FS"</code>, i.e., <b>U</b>nconditional
 <b>M</b>odel - <b>F</b>ixed <b>S</b>tudy effects) </td>
</tr>
<tr>
 <td style="text-align: center;">
2. </td><td style="text-align: left;"> Mixed-effects logistic regression model with random study
 effects </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "UM.RS"</code>, i.e., <b>U</b>nconditional
 <b>M</b>odel - <b>R</b>andom <b>S</b>tudy effects) </td>
</tr>
<tr>
 <td style="text-align: center;">
3. </td><td style="text-align: left;"> Generalised linear mixed model (conditional
 Hypergeometric-Normal) </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "CM.EL"</code>, i.e., <b>C</b>onditional
 <b>M</b>odel - <b>E</b>xact <b>L</b>ikelihood) </td>
</tr>
<tr>
 <td style="text-align: center;">
4. </td><td style="text-align: left;"> Generalised linear mixed model (conditional
  Binomial-Normal) </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "CM.AL"</code>, i.e., <b>C</b>onditional
  <b>M</b>odel - <b>A</b>pproximate <b>L</b>ikelihood)
</td>
</tr>

</table>

<p>Details on these four GLMMs as well as additional arguments which
can be provided using argument '<code>...</code>' in <code>metabin</code> are
described in <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> where you can also
find information on the iterative algorithms used for estimation.
Note, regardless of which value is used for argument
<code>model.glmm</code>, results for two different GLMMs are calculated:
common effect model (with fixed treatment effect) and random
effects model (with random treatment effects).
</p>



<h4>Continuity correction</h4>

<p>Three approaches are available to apply a continuity correction:
</p>

<ul>
<li><p> Only studies with a zero cell count (<code>method.incr =
  "only0"</code>)
</p>
</li>
<li><p> All studies if at least one study has a zero cell count
(<code>method.incr = "if0all"</code>)
</p>
</li>
<li><p> All studies irrespective of zero cell counts
(<code>method.incr = "all"</code>)
</p>
</li></ul>

<p>By default, a continuity correction is only applied to studies with
a zero cell count (<code>method.incr = "only0"</code>). This method
showed the best performance for the odds ratio in a simulation
study under the random effects model (Weber et al., 2020).
</p>
<p>The continuity correction method is used both to calculate
individual study results with confidence limits and to conduct
meta-analysis based on the inverse variance method. For the risk
difference, the method is only considered to calculate standard
errors and confidence limits.  For Peto method and GLMMs no
continuity correction is used in the meta-analysis. Furthermore,
the continuity correction is ignored for individual studies for the
Peto method.
</p>
<p>For studies with a zero cell count, by default, 0.5 (argument
<code>incr</code>) is added to all cell frequencies for the odds ratio or
only the number of events for the risk ratio (argument
<code>RR.Cochrane = FALSE</code>, default). The increment is added to all
cell frequencies for the risk ratio if argument <code>RR.Cochrane =
TRUE</code>. For the risk difference, <code>incr</code> is only added to all
cell frequencies to calculate the standard error. Finally, a
treatment arm continuity correction is used if <code>incr = "TACC"</code>
(Sweeting et al., 2004; Diamond et al., 2007).
</p>
<p>For odds ratio and risk ratio, treatment estimates and standard
errors are only calculated for studies with zero or all events in
both groups if <code>allstudies = TRUE</code>.
</p>
<p>For the Mantel-Haenszel method, by default (if <code>MH.exact</code> is
FALSE), <code>incr</code> is added to cell frequencies of a study with a
zero cell count in the calculation of the pooled risk ratio or odds
ratio as well as the estimation of the variance of the pooled risk
difference, risk ratio or odds ratio. This approach is also used in
other software, e.g. RevMan 5 and the Stata procedure
metan. According to Fleiss (in Cooper &amp; Hedges, 1994), there is no
need to add 0.5 to a cell frequency of zero to calculate the
Mantel-Haenszel estimate and he advocates the exact method
(<code>MH.exact</code> = TRUE). Note, estimates based on exact
Mantel-Haenszel method or GLMM are not defined if the number of
events is zero in all studies either in the experimental or control
group.
</p>



<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments
<code>common</code> and <code>random</code>. Accordingly, the estimate
for the random effects model can be extracted from component
<code>TE.random</code> of an object of class <code>"meta"</code> even if
argument <code>random = FALSE</code>. However, all functions in R
package <b>meta</b> will adequately consider the values for
<code>common</code> and <code>random</code>. E.g. function
<code><a href="#topic+print.meta">print.meta</a></code> will not print results for the random
effects model if <code>random = FALSE</code>.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metabin", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Bakbergenuly I, Hoaglin DC, Kulinskaya E (2020):
Methods for estimating between-study variance and overall
effect in meta-analysis of odds-ratios.
<em>Research Synthesis Methods</em>,
<b>11</b>, 426&ndash;42
</p>
<p>Cooper H &amp; Hedges LV (1994):
<em>The Handbook of Research Synthesis</em>.
Newbury Park, CA: Russell Sage Foundation
</p>
<p>Diamond GA, Bax L, Kaul S (2007):
Uncertain Effects of Rosiglitazone on the Risk for Myocardial
Infarction and Cardiovascular Death.
<em>Annals of Internal Medicine</em>,
<b>147</b>, 578&ndash;81
</p>
<p>DerSimonian R &amp; Laird N (1986):
Meta-analysis in clinical trials.
<em>Controlled Clinical Trials</em>,
<b>7</b>, 177&ndash;88
</p>
<p>Fleiss JL (1993):
The statistical basis of meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>2</b>, 121&ndash;45
</p>
<p>Greenland S &amp; Robins JM (1985):
Estimation of a common effect parameter from sparse follow-up data.
<em>Biometrics</em>,
<b>41</b>, 55&ndash;68
</p>
<p><em>Review Manager (RevMan)</em> [Computer program]. Version 5.4.
The Cochrane Collaboration, 2020
</p>
<p>Robins J, Breslow N, Greenland S (1986):
Estimators of the Mantel-Haenszel Variance Consistent in Both
Sparse Data and Large-Strata Limiting Models.
<em>Biometrics</em>,
<b>42</b>, 311&ndash;23
</p>
<p>Rücker G, Schwarzer G, Carpenter J, Olkin I (2009):
Why add anything to nothing? The arcsine difference as a measure of
treatment effect in meta-analysis with zero cells.
<em>Statistics in Medicine</em>,
<b>28</b>, 721&ndash;38
</p>
<p>Simmonds MC, Higgins JP (2016):
A general framework for the use of logistic regression models in
meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>25</b>, 2858&ndash;77
</p>
<p>StataCorp. 2011.
<em>Stata Statistical Software: Release 12</em>.
College Station, TX: StataCorp LP.
</p>
<p>Stijnen T, Hamza TH, Ozdemir P (2010):
Random effects meta-analysis of event outcome in the framework of
the generalized linear mixed model with applications in sparse
data.
<em>Statistics in Medicine</em>,
<b>29</b>, 3046&ndash;67
</p>
<p>Sweeting MJ, Sutton AJ, Lambert PC (2004):
What to add to nothing? Use and avoidance of continuity corrections
in meta-analysis of sparse data.
<em>Statistics in Medicine</em>,
<b>23</b>, 1351&ndash;75
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Viechtbauer W (2010):
Conducting meta-analyses in R with the metafor package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>
<p>Weber F, Knapp G, Ickstadt K, Kundt G, Glass Ä (2020):
Zero-cell corrections in random-effects meta-analyses.
<em>Research Synthesis Methods</em>,
<b>11</b>, 913&ndash;9
</p>
<p>Yusuf S, Peto R, Lewis J, Collins R, Sleight P (1985):
Beta blockade during and after myocardial infarction: An overview
of the randomized trials.
<em>Progress in Cardiovascular Diseases</em>,
<b>27</b>, 335&ndash;71
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+forest">forest</a></code>, <code><a href="#topic+funnel">funnel</a></code>,
<code><a href="#topic+metabias">metabias</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metareg">metareg</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Calculate odds ratio and confidence interval for a single study
#
metabin(10, 20, 15, 20, sm = "OR")

# Different results (due to handling of studies with double zeros)
#
metabin(0, 10, 0, 10, sm = "OR")
metabin(0, 10, 0, 10, sm = "OR", allstudies = TRUE)

# Use subset of Olkin (1995) to conduct meta-analysis based on
# inverse variance method (with risk ratio as summary measure)
#
data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = c(41, 47, 51, 59),
  studlab = paste(author, year),
  method = "Inverse")
m1
# Show results for individual studies
summary(m1)

# Use different subset of Olkin (1995)
#
m2 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = year &lt; 1970,
  studlab = paste(author, year),
  method = "Inverse")
m2
forest(m2)

# Meta-analysis with odds ratio as summary measure
#
m3 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = year &lt; 1970,
  studlab = paste(author, year),
  sm = "OR", method = "Inverse")
# Same meta-analysis result using 'update.meta' function
m3 &lt;- update(m2, sm = "OR")
m3

# Meta-analysis based on Mantel-Haenszel method (with odds ratio as
# summary measure)
#
m4 &lt;- update(m3, method = "MH")
m4

# Meta-analysis based on Peto method (only available for odds ratio
# as summary measure)
#
m5 &lt;- update(m3, method = "Peto")
m5

## Not run: 
# Meta-analysis using generalised linear mixed models
# (only if R package 'lme4' is available)
#

# Logistic regression model with (k = 4) fixed study effects
# (default: model.glmm = "UM.FS")
#
m6 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  studlab = paste(author, year),
  data = Olkin1995, subset = year &lt; 1970, method = "GLMM")
# Same results:
m6 &lt;- update(m2, method = "GLMM")
m6

# Mixed-effects logistic regression model with random study effects
# (warning message printed due to argument 'nAGQ')
#
m7 &lt;- update(m6, model.glmm = "UM.RS")
#
# Use additional argument 'nAGQ' for internal call of 'rma.glmm'
# function
#
m7 &lt;- update(m6, model.glmm = "UM.RS", nAGQ = 1)
m7

# Generalised linear mixed model (conditional Hypergeometric-Normal)
# (R package 'BiasedUrn' must be available)
#
m8 &lt;- update(m6, model.glmm = "CM.EL")
m8

# Generalised linear mixed model (conditional Binomial-Normal)
#
m9 &lt;- update(m6, model.glmm = "CM.AL")
m9

# Logistic regression model with (k = 70) fixed study effects
# (about 18 seconds with Intel Core i7-3667U, 2.0GHz)
#
m10 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
   studlab = paste(author, year),
   data = Olkin1995, method = "GLMM")
m10

# Mixed-effects logistic regression model with random study effects
# - about 50 seconds with Intel Core i7-3667U, 2.0GHz
# - several warning messages, e.g. "failure to converge, ..."
#
update(m10, model.glmm = "UM.RS")

# Conditional Hypergeometric-Normal GLMM
# - long computation time (about 12 minutes with Intel Core
#   i7-3667U, 2.0GHz)
# - estimation problems for this very large dataset:
#   * warning that Choleski factorization of Hessian failed
#   * confidence interval for treatment effect smaller in random
#     effects model compared to common effect model
#
system.time(m11 &lt;- update(m10, model.glmm = "CM.EL"))
m11

# Generalised linear mixed model (conditional Binomial-Normal)
# (less than 1 second with Intel Core i7-3667U, 2.0GHz)
#
update(m10, model.glmm = "CM.AL")

## End(Not run)

</code></pre>

<hr>
<h2 id='metabind'>Combine and summarize meta-analysis objects</h2><span id='topic+metabind'></span>

<h3>Description</h3>

<p>This function can be used to combine meta-analysis objects and is,
for example, useful to summarize results of various meta-analysis
methods or to generate a forest plot with results of several
subgroup analyses.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metabind(
  ...,
  name = NULL,
  common = NULL,
  random = NULL,
  prediction = NULL,
  backtransf = NULL,
  outclab = NULL,
  pooled = NULL,
  warn.deprecated = gs("warn.deprecated")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metabind_+3A_...">...</code></td>
<td>
<p>Any number of meta-analysis objects or a single list
with meta-analyses.</p>
</td></tr>
<tr><td><code id="metabind_+3A_name">name</code></td>
<td>
<p>An optional character vector providing descriptive
names for the meta-analysis objects.</p>
</td></tr>
<tr><td><code id="metabind_+3A_common">common</code></td>
<td>
<p>A logical vector indicating whether results of common
effect model should be considered.</p>
</td></tr>
<tr><td><code id="metabind_+3A_random">random</code></td>
<td>
<p>A logical vector indicating whether results of random
effects model should be considered.</p>
</td></tr>
<tr><td><code id="metabind_+3A_prediction">prediction</code></td>
<td>
<p>A logical vector indicating whether results of
prediction intervals should be considered.</p>
</td></tr>
<tr><td><code id="metabind_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots. If
<code>backtransf=TRUE</code> (default), results for <code>sm="OR"</code> are
printed as odds ratios rather than log odds ratios, for example.</p>
</td></tr>
<tr><td><code id="metabind_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label for all meta-analyis objects.</p>
</td></tr>
<tr><td><code id="metabind_+3A_pooled">pooled</code></td>
<td>
<p>Deprecated argument (replaced by <code>common</code> and
<code>random</code>.</p>
</td></tr>
<tr><td><code id="metabind_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to combine any number of meta-analysis
objects which is useful, for example, to summarize results of
various meta-analysis methods or to generate a forest plot with
results of several subgroup analyses (see Examples).
</p>
<p>Individual study results are not retained with <code>metabind</code> as
the function allows to combine meta-analyses from different data
sets (e.g., with randomized or observational studies). Individual
study results are retained with R function <code><a href="#topic+metamerge">metamerge</a></code>
which can be used to combine results of meta-analyses of the
same dataset.
</p>


<h3>Value</h3>

<p>An object of class <code>c("metabind", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+forest.metabind">forest.metabind</a></code>,
<code><a href="#topic+metamerge">metamerge</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)

# Add some (fictitious) grouping variables:
#
Fleiss1993cont$age &lt;- c(55, 65, 55, 65, 55)
Fleiss1993cont$region &lt;- c("Europe", "Europe", "Asia", "Asia", "Europe")

m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")

# Conduct two subgroup analyses
#
mu1 &lt;- update(m1, subgroup = age, subgroup.name = "Age group")
mu2 &lt;- update(m1, subgroup = region, subgroup.name = "Region")

# Combine random effects subgroup meta-analyses and show forest
# plot with subgroup results
#
mb1 &lt;- metabind(mu1, mu2, common = FALSE)
mb1
forest(mb1)

# Use various estimation methods for between-study heterogeneity
# variance
#
m1.pm &lt;- update(m1, method.tau = "PM")
m1.dl &lt;- update(m1, method.tau = "DL")
m1.ml &lt;- update(m1, method.tau = "ML")
m1.hs &lt;- update(m1, method.tau = "HS")
m1.sj &lt;- update(m1, method.tau = "SJ")
m1.he &lt;- update(m1, method.tau = "HE")
m1.eb &lt;- update(m1, method.tau = "EB")

# Combine meta-analyses and show results
#
taus &lt;- c("Restricted maximum-likelihood estimator",
  "Paule-Mandel estimator",
  "DerSimonian-Laird estimator",
  "Maximum-likelihood estimator",
  "Hunter-Schmidt estimator",
  "Sidik-Jonkman estimator",
  "Hedges estimator",
  "Empirical Bayes estimator")
#
m1.taus &lt;- metabind(m1, m1.pm, m1.dl, m1.ml, m1.hs, m1.sj, m1.he, m1.eb,
  name = taus, common = FALSE)
m1.taus
forest(m1.taus)

</code></pre>

<hr>
<h2 id='metacont'>Meta-analysis of continuous outcome data</h2><span id='topic+metacont'></span>

<h3>Description</h3>

<p>Calculation of common and random effects estimates for meta-analyses
with continuous outcome data; inverse variance weighting is used
for pooling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metacont(
  n.e,
  mean.e,
  sd.e,
  n.c,
  mean.c,
  sd.c,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  median.e,
  q1.e,
  q3.e,
  min.e,
  max.e,
  median.c,
  q1.c,
  q3.c,
  min.c,
  max.c,
  method.mean = "Luo",
  method.sd = "Shi",
  approx.mean.e,
  approx.mean.c = approx.mean.e,
  approx.sd.e,
  approx.sd.c = approx.sd.e,
  sm = gs("smcont"),
  method.ci = gs("method.ci.cont"),
  level = gs("level"),
  pooledvar = gs("pooledvar"),
  method.smd = gs("method.smd"),
  sd.glass = gs("sd.glass"),
  exact.smd = gs("exact.smd"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = gs("method.tau"),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  method.bias = gs("method.bias"),
  backtransf = gs("backtransf"),
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  label.e = gs("label.e"),
  label.c = gs("label.c"),
  label.left = gs("label.left"),
  label.right = gs("label.right"),
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  id,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metacont_+3A_n.e">n.e</code></td>
<td>
<p>Number of observations in experimental group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_mean.e">mean.e</code></td>
<td>
<p>Estimated mean in experimental group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_sd.e">sd.e</code></td>
<td>
<p>Standard deviation in experimental group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_n.c">n.c</code></td>
<td>
<p>Number of observations in control group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_mean.c">mean.c</code></td>
<td>
<p>Estimated mean in control group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_sd.c">sd.c</code></td>
<td>
<p>Standard deviation in control group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metacont_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information.</p>
</td></tr>
<tr><td><code id="metacont_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metacont_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metacont_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metacont_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metacont_+3A_median.e">median.e</code></td>
<td>
<p>Median in experimental group (used to estimate the
mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_q1.e">q1.e</code></td>
<td>
<p>First quartile in experimental group (used to estimate
the mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_q3.e">q3.e</code></td>
<td>
<p>Third quartile in experimental group (used to estimate
the mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_min.e">min.e</code></td>
<td>
<p>Minimum in experimental group (used to estimate the
mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_max.e">max.e</code></td>
<td>
<p>Maximum in experimental group (used to estimate the
mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_median.c">median.c</code></td>
<td>
<p>Median in control group (used to estimate the mean
and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_q1.c">q1.c</code></td>
<td>
<p>First quartile in control group (used to estimate the
mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_q3.c">q3.c</code></td>
<td>
<p>Third quartile in control group (used to estimate the
mean and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_min.c">min.c</code></td>
<td>
<p>Minimum in control group (used to estimate the mean
and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_max.c">max.c</code></td>
<td>
<p>Maximum in control group (used to estimate the mean
and standard deviation).</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.mean">method.mean</code></td>
<td>
<p>A character string indicating which method to
use to approximate the mean from the median and other statistics
(see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.sd">method.sd</code></td>
<td>
<p>A character string indicating which method to use
to approximate the standard deviation from sample size, median,
interquartile range and range (see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_approx.mean.e">approx.mean.e</code></td>
<td>
<p>Approximation method to estimate means in
experimental group (see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_approx.mean.c">approx.mean.c</code></td>
<td>
<p>Approximation method to estimate means in
control group (see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_approx.sd.e">approx.sd.e</code></td>
<td>
<p>Approximation method to estimate standard
deviations in experimental group (see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_approx.sd.c">approx.sd.c</code></td>
<td>
<p>Approximation method to estimate standard
deviations in control group (see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code>) is to be used for
pooling of studies.</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.ci">method.ci</code></td>
<td>
<p>A character string indicating which method is used
to calculate confidence intervals for individual studies (see
Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metacont_+3A_pooledvar">pooledvar</code></td>
<td>
<p>A logical indicating if a pooled variance should
be used for the mean difference (<code>sm="MD"</code>) or ratio of
means (<code>sm="ROM"</code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.smd">method.smd</code></td>
<td>
<p>A character string indicating which method is
used to estimate the standardised mean difference
(<code>sm="SMD"</code>). Either <code>"Hedges"</code> for Hedges' g
(default), <code>"Cohen"</code> for Cohen's d, or <code>"Glass"</code> for
Glass' delta, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metacont_+3A_sd.glass">sd.glass</code></td>
<td>
<p>A character string indicating which standard
deviation is used in the denominator for Glass' method to
estimate the standardised mean difference. Either
<code>"control"</code> using the standard deviation in the control
group (<code>sd.c</code>) or <code>"experimental"</code> using the standard
deviation in the experimental group (<code>sd.e</code>), can be
abbreviated.</p>
</td></tr>
<tr><td><code id="metacont_+3A_exact.smd">exact.smd</code></td>
<td>
<p>A logical indicating whether exact formulae should
be used in estimation of the standardised mean difference and its
standard error (see Details).</p>
</td></tr>
<tr><td><code id="metacont_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metacont_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metacont_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metacont_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metacont_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metacont_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metacont_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metacont_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacont_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, <code>"Thompson"</code>,
or <code>"Pustejovsky"</code> (see <code><a href="#topic+metabias">metabias</a></code>), can be
abbreviated.</p>
</td></tr>
<tr><td><code id="metacont_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for ratio of
means (<code>sm="ROM"</code>) should be back transformed in printouts
and plots. If TRUE (default), results will be presented as ratio
of means; otherwise log ratio of means will be shown.</p>
</td></tr>
<tr><td><code id="metacont_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metacont_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metacont_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metacont_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metacont_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metacont_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metacont_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metacont_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metacont_+3A_label.e">label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_label.c">label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code id="metacont_+3A_label.left">label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code id="metacont_+3A_label.right">label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code id="metacont_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metacont_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metacont_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metacont_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metacont_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metacont_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metacont_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metacont_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metacont_+3A_id">id</code></td>
<td>
<p>Deprecated argument (replaced by 'cluster').</p>
</td></tr>
<tr><td><code id="metacont_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by
'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metacont_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metacont_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if studies are excluded from meta-analysis due to zero
standard deviations).</p>
</td></tr>
<tr><td><code id="metacont_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metacont_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>.</p>
</td></tr>
<tr><td><code id="metacont_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculation of common and random effects estimates for meta-analyses
with continuous outcome data; inverse variance weighting is used
for pooling.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>
<p>Three different types of summary measures are available for continuous
outcomes:
</p>

<ul>
<li><p> mean difference (argument <code>sm = "MD"</code>)
</p>
</li>
<li><p> standardised mean difference (<code>sm = "SMD"</code>)
</p>
</li>
<li><p> ratio of means (<code>sm = "ROM"</code>)
</p>
</li></ul>



<h4>Standardised mean difference</h4>

<p>For the standardised mean difference three methods are implemented:
</p>

<ul>
<li><p> Hedges' g (default, <code>method.smd = "Hedges"</code>) - see
Hedges (1981)
</p>
</li>
<li><p> Cohen's d (<code>method.smd = "Cohen"</code>) - see Cohen (1988)
</p>
</li>
<li><p> Glass' delta (<code>method.smd = "Glass"</code>) - see Glass (1976)
</p>
</li></ul>

<p>Hedges (1981) calculated the exact bias in Cohen's d which is a
ratio of gamma distributions with the degrees of freedom,
i.e. total sample size minus two, as argument. By default (argument
<code>exact.smd = FALSE</code>), an accurate approximation of this bias
provided in Hedges (1981) is utilised for Hedges' g as well as its
standard error; these approximations are also used in RevMan
5. Following Borenstein et al. (2009) these approximations are not
used in the estimation of Cohen's d. White and Thomas (2005) argued
that approximations are unnecessary with modern software and
accordingly promote to use the exact formulae; this is possible
using argument <code>exact.smd = TRUE</code>. For Hedges' g the exact
formulae are used to calculate the standardised mean difference as
well as the standard error; for Cohen's d the exact formula is only
used to calculate the standard error. In typical applications (with
sample sizes above 10), the differences between using the exact
formulae and the approximation will be minimal.
</p>
<p>For Glass' delta, by default (argument <code>sd.glass =
"control"</code>), the standard deviation in the control group
(<code>sd.c</code>) is used in the denominator of the standard mean
difference. The standard deviation in the experimental group
(<code>sd.e</code>) can be used by specifying <code>sd.glass =
"experimental"</code>.
</p>



<h4>Ratio of means</h4>

<p>Meta-analysis of ratio of means &ndash; also called response ratios &ndash;
is described in Hedges et al. (1999) and Friedrich et al. (2008).
Calculations are conducted on the log scale and list elements
<code>TE</code>, <code>TE.common</code>, and <code>TE.random</code> contain the
logarithm of the ratio of means. In printouts and plots these
values are back transformed if argument <code>backtransf = TRUE</code>.
</p>



<h4>Approximate means from sample sizes, medians and other statistics</h4>

<p>Missing means in the experimental group (analogously for the
control group) can be derived from
</p>

<ol>
<li><p> sample size, median, interquartile range and range (arguments
<code>n.e</code>, <code>median.e</code>, <code>q1.e</code>, <code>q3.e</code>,
<code>min.e</code>, and <code>max.e</code>),
</p>
</li>
<li><p> sample size, median and interquartile range (arguments
<code>n.e</code>, <code>median.e</code>, <code>q1.e</code>, and <code>q3.e</code>), or
</p>
</li>
<li><p> sample size, median and range (arguments <code>n.e</code>,
<code>median.e</code>, <code>min.e</code>, and <code>max.e</code>).
</p>
</li></ol>

<p>By default, methods described in Luo et al. (2018) are utilized
(argument <code>method.mean = "Luo"</code>):
</p>

<ul>
<li><p> equation (15) if sample size, median, interquartile range and 
range are available,
</p>
</li>
<li><p> equation (11) if sample size, median and interquartile range
are available,
</p>
</li>
<li><p> equation (7) if sample size, median and range are available.
</p>
</li></ul>

<p>Instead the methods described in Wan et al. (2014) are used if
argument <code>method.mean = "Wan"</code>:
</p>

<ul>
<li><p> equation (10) if sample size, median, interquartile range and 
range are available,
</p>
</li>
<li><p> equation (14) if sample size, median and interquartile range
are available,
</p>
</li>
<li><p> equation (2) if sample size, median and range are available.
</p>
</li></ul>

<p>The following methods are also available to estimate means from
quantiles or ranges if R package <b>estmeansd</b> is installed:
</p>

<ul>
<li><p> Method for Unknown Non-Normal Distributions (MLN) approach
(Cai et al. (2021), argument <code>method.mean = "Cai"</code>),
</p>
</li>
<li><p> Quantile Estimation (QE) method (McGrath et al. (2020),
argument <code>method.mean = "QE-McGrath"</code>)),
</p>
</li>
<li><p> Box-Cox (BC) method (McGrath et al. (2020),
argument <code>method.mean = "BC-McGrath"</code>)).
</p>
</li></ul>

<p>By default, missing means are replaced successively using
interquartile ranges and ranges (if available), interquartile
ranges (if available) and finally ranges. Arguments
<code>approx.mean.e</code> and <code>approx.mean.c</code> can be used to
overwrite this behaviour for each individual study and treatment
arm:
</p>

<ul>
<li><p> use means directly (entry <code>""</code> in argument
<code>approx.mean.e</code> or <code>approx.mean.c</code>);
</p>
</li>
<li><p> median, interquartile range and range (<code>"iqr.range"</code>);
</p>
</li>
<li><p> median and interquartile range (<code>"iqr"</code>);
</p>
</li>
<li><p> median and range (<code>"range"</code>).
</p>
</li></ul>




<h4>Approximate standard deviations from sample sizes,
medians and other statistics</h4>

<p>Missing standard deviations in the experimental group (analogously
for the control group) can be derived from
</p>

<ol>
<li><p> sample size, median, interquartile range and range (arguments
<code>n.e</code>, <code>median.e</code>, <code>q1.e</code>, <code>q3.e</code>,
<code>min.e</code>, and <code>max.e</code>),
</p>
</li>
<li><p> sample size, median and interquartile range (arguments
<code>n.e</code>, <code>median.e</code>, <code>q1.e</code> and <code>q3.e</code>), or
</p>
</li>
<li><p> sample size, median and range (arguments <code>n.e</code>,
<code>median.e</code>, <code>min.e</code> and <code>max.e</code>).
</p>
</li></ol>

<p>Wan et al. (2014) describe methods to estimate the standard
deviation from the sample size, median and additional
statistics. Shi et al. (2020) provide an improved estimate of the
standard deviation if the interquartile range and range are
available in addition to the sample size and median. Accordingly,
equation (11) in Shi et al. (2020) is the default (argument
<code>method.sd = "Shi"</code>), if the median, interquartile range and
range are provided. The method by Wan et al. (2014) is used if
argument <code>method.sd = "Wan"</code> and, depending on the sample
size, either equation (12) or (13) is used. If only the
interquartile range or range is available, equations (15) / (16)
and (7) / (9) in Wan et al. (2014) are used, respectively.
</p>
<p>The following methods are also available to estimate standard
deviations from quantiles or ranges if R package <b>estmeansd</b>
is installed:
</p>

<ul>
<li><p> Method for Unknown Non-Normal Distributions (MLN) approach
(Cai et al. (2021), argument <code>method.mean = "Cai"</code>),
</p>
</li>
<li><p> Quantile Estimation (QE) method (McGrath et al. (2020),
argument <code>method.mean = "QE-McGrath"</code>)),
</p>
</li>
<li><p> Box-Cox (BC) method (McGrath et al. (2020),
argument <code>method.mean = "BC-McGrath"</code>)).
</p>
</li></ul>

<p>By default, missing standard deviations are replaced successively
using these method, i.e., interquartile ranges and ranges are used
before interquartile ranges before ranges. Arguments
<code>approx.sd.e</code> and <code>approx.sd.c</code> can be used to overwrite
this default for each individual study and treatment arms:
</p>

<ul>
<li><p> sample size, median, interquartile range and range
(<code>"iqr.range"</code>);
</p>
</li>
<li><p> sample size, median and interquartile range (<code>"iqr"</code>);
</p>
</li>
<li><p> sample size, median and range (<code>"range"</code>).
</p>
</li></ul>




<h4>Confidence intervals for individual studies</h4>

<p>For the mean difference (argument <code>sm = "MD"</code>), the confidence
interval for individual studies can be based on the
</p>

<ul>
<li><p> standard normal distribution (<code>method.ci = "z"</code>, default), or
</p>
</li>
<li><p> t-distribution (<code>method.ci = "t"</code>).
</p>
</li></ul>

<p>Note, this choice does not affect the results of the common effect
and random effects meta-analysis.
</p>



<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments <code>common</code>
and <code>random</code>. Accordingly, the estimate for the random effects
model can be extracted from component <code>TE.random</code> of an object
of class <code>"meta"</code> even if argument <code>random =
FALSE</code>. However, all functions in R package <b>meta</b> will
adequately consider the values for <code>common</code> and
<code>random</code>. E.g. function <code><a href="#topic+print.meta">print.meta</a></code> will not
print results for the random effects model if <code>random =
FALSE</code>.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metacont", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Note</h3>

<p>The function <code><a href="#topic+metagen">metagen</a></code> is called internally to
calculate individual and overall treatment estimates and standard
errors.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Borenstein M, Hedges LV, Higgins JPT, Rothstein HR (2009):
<em>Introduction to Meta-Analysis</em>.
Chichester: Wiley
</p>
<p>Cai S, Zhou J, Pan J (2021):
Estimating the sample mean and standard deviation from order
statistics and sample size in meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>30</b>, 2701&ndash;2719
</p>
<p>Cohen J (1988):
<em>Statistical Power Analysis for the Behavioral Sciences
(second ed.)</em>.
Lawrence Erlbaum Associates
</p>
<p>Friedrich JO, Adhikari NK, Beyene J (2008):
The ratio of means method as an alternative to mean differences for
analyzing continuous outcome variables in meta-analysis: A
simulation study.
<em>BMC Medical Research Methodology</em>,
<b>8</b>, 32
</p>
<p>Glass G (1976):
Primary, secondary, and meta-analysis of research.
<em>Educational Researcher</em>,
<b>5</b>, 3&ndash;8
</p>
<p>Hedges LV (1981):
Distribution theory for Glass's estimator of effect size and
related estimators.
<em>Journal of Educational and Behavioral Statistics</em>,
<b>6</b>, 107&ndash;28
</p>
<p>Hedges LV, Gurevitch J, Curtis PS (1999):
The meta-analysis of response ratios in experimental ecology.
<em>Ecology</em>,
<b>80</b>, 1150&ndash;6
</p>
<p>Luo D, Wan X, Liu J, Tong T (2018):
Optimally estimating the sample mean from the sample size, median,
mid-range, and/or mid-quartile range.
<em>Statistical Methods in Medical Research</em>,
<b>27</b>, 1785&ndash;805
</p>
<p>McGrath S, Zhao X, Steele R, et al. and the DEPRESsion Screening
Data (DEPRESSD) Collaboration (2020):
Estimating the sample mean and standard deviation from commonly
reported quantiles in meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>29</b>, 2520&ndash;2537
</p>
<p><em>Review Manager (RevMan)</em> [Computer program]. Version 5.4.
The Cochrane Collaboration, 2020
</p>
<p>Shi J, Luo D, Weng H, Zeng XT, Lin L, Chu H, Tong T (2020):
Optimally estimating the sample standard deviation from the
five-number summary.
<em>Research Synthesis Methods</em>,
<b>11</b>, 641&ndash;54
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Wan X, Wang W, Liu J, Tong T (2014):
Estimating the sample mean and standard deviation from the sample
size, median, range and/or interquartile range.
<em>BMC Medical Research Methodology</em>,
<b>14</b>, 135
</p>
<p>White IR, Thomas J (2005):
Standardized mean differences in individually-randomized and
cluster-randomized trials, with applications to meta-analysis.
<em>Clinical Trials</em>,
<b>2</b>, 141&ndash;51
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)

# Meta-analysis with Hedges' g as effect measure
#
m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")
m1
forest(m1)

# Use Cohen's d instead of Hedges' g as effect measure
#
update(m1, method.smd = "Cohen")

# Use Glass' delta instead of Hedges' g as effect measure
#
update(m1, method.smd = "Glass")

# Use Glass' delta based on the standard deviation in the experimental group
#
update(m1, method.smd = "Glass", sd.glass = "experimental")

# Calculate Hedges' g based on exact formulae
#
update(m1, exact.smd = TRUE)

data(amlodipine)
m2 &lt;- metacont(n.amlo, mean.amlo, sqrt(var.amlo),
  n.plac, mean.plac, sqrt(var.plac),
  data = amlodipine, studlab = study)
m2

# Use pooled variance
#
update(m2, pooledvar = TRUE)

# Meta-analysis of response ratios (Hedges et al., 1999)
#
data(woodyplants)
m3 &lt;- metacont(n.elev, mean.elev, sd.elev, n.amb, mean.amb, sd.amb,
  data = woodyplants, sm = "ROM")
m3
print(m3, backtransf = FALSE)

</code></pre>

<hr>
<h2 id='metacor'>Meta-analysis of correlations</h2><span id='topic+metacor'></span>

<h3>Description</h3>

<p>Calculation of common effect and random effects estimates for
meta-analyses with correlations; inverse variance weighting is used
for pooling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metacor(
  cor,
  n,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  sm = gs("smcor"),
  level = gs("level"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = gs("method.tau"),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  null.effect = 0,
  method.bias = gs("method.bias"),
  backtransf = gs("backtransf"),
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metacor_+3A_cor">cor</code></td>
<td>
<p>Correlation.</p>
</td></tr>
<tr><td><code id="metacor_+3A_n">n</code></td>
<td>
<p>Number of observations.</p>
</td></tr>
<tr><td><code id="metacor_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metacor_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information, i.e., cor and n.</p>
</td></tr>
<tr><td><code id="metacor_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metacor_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metacor_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metacor_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metacor_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"ZCOR"</code> or <code>"COR"</code>) is to be used for pooling of
studies.</p>
</td></tr>
<tr><td><code id="metacor_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metacor_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metacor_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metacor_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metacor_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metacor_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metacor_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metacor_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metacor_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metacor_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metacor_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metacor_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacor_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
<tr><td><code id="metacor_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, or
<code>"Thompson"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code>.</p>
</td></tr>
<tr><td><code id="metacor_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for Fisher's
z transformed correlations (<code>sm = "ZCOR"</code>) should be back
transformed in printouts and plots. If TRUE (default), results
will be presented as correlations; otherwise Fisher's z
transformed correlations will be shown.</p>
</td></tr>
<tr><td><code id="metacor_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metacor_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metacor_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metacor_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metacor_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metacor_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metacor_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metacor_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metacor_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metacor_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metacor_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metacor_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metacor_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metacor_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metacor_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metacor_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metacor_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metacor_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metacor_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metacor_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>.</p>
</td></tr>
<tr><td><code id="metacor_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Common effect and random effects meta-analysis of correlations
based either on Fisher's z transformation of correlations (<code>sm
= "ZCOR"</code>) or direct combination of (untransformed) correlations
(<code>sm = "COR"</code>) (see Cooper et al., 2009, p264-5 and
p273-4). Only few statisticians would advocate the use of
untransformed correlations unless sample sizes are very large (see
Cooper et al., 2009, p265). The artificial example given below
shows that the smallest study gets the largest weight if
correlations are combined directly because the correlation is
closest to 1.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>


<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments
<code>common</code> and <code>random</code>. Accordingly, the estimate
for the random effects model can be extracted from component
<code>TE.random</code> of an object of class <code>"meta"</code> even if
argument <code>random = FALSE</code>. However, all functions in R
package <b>meta</b> will adequately consider the values for
<code>common</code> and <code>random</code>. E.g. functions
<code><a href="#topic+print.meta">print.meta</a></code> and <code><a href="#topic+forest.meta">forest.meta</a></code> will not
print results for the random effects model if <code>random =
FALSE</code>.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metacor", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Note</h3>

<p>The function <code><a href="#topic+metagen">metagen</a></code> is called internally to
calculate individual and overall treatment estimates and standard
errors.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Cooper H, Hedges LV, Valentine JC (2009):
<em>The Handbook of Research Synthesis and Meta-Analysis</em>,
2nd Edition.
New York: Russell Sage Foundation
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metacont">metacont</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m1 &lt;- metacor(c(0.85, 0.7, 0.95), c(20, 40, 10))

# Print correlations (back transformed from Fisher's z
# transformation)
#
m1

# Print Fisher's z transformed correlations 
#
print(m1, backtransf = FALSE)

# Forest plot with back transformed correlations
#
forest(m1)

# Forest plot with Fisher's z transformed correlations
#
forest(m1, backtransf = FALSE)

m2 &lt;- update(m1, sm = "cor")
m2

## Not run: 
# Identical forest plots (as back transformation is the identity
# transformation)
forest(m2)
forest(m2, backtransf = FALSE)

## End(Not run)

</code></pre>

<hr>
<h2 id='metacr'>Meta-analysis of outcome data from Cochrane review</h2><span id='topic+metacr'></span>

<h3>Description</h3>

<p>Wrapper function to perform meta-analysis for a single outcome of a
Cochrane Intervention review.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metacr(
  x,
  comp.no = 1,
  outcome.no = 1,
  method,
  sm,
  level = gs("level"),
  common,
  random,
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = "DL",
  method.tau.ci = gs("method.tau.ci"),
  tau.common = FALSE,
  level.ma = gs("level.ma"),
  method.random.ci = "classic",
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  Q.Cochrane,
  swap.events,
  logscale,
  backtransf = gs("backtransf"),
  test.subgroup,
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  rob = NULL,
  tool = NULL,
  categories = NULL,
  col = NULL,
  symbols = NULL,
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title,
  complab,
  outclab,
  keepdata = gs("keepdata"),
  warn = FALSE,
  warn.deprecated = gs("warn.deprecated"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metacr_+3A_x">x</code></td>
<td>
<p>An object of class <code>rm5</code> or <code>cdir</code> created by R
function <code>read.rm5</code> or <code>read.cdir</code> .</p>
</td></tr>
<tr><td><code id="metacr_+3A_comp.no">comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code id="metacr_+3A_outcome.no">outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code id="metacr_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is to be
used for pooling of studies. One of <code>"Inverse"</code>,
<code>"MH"</code>, or <code>"Peto"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metacr_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"RR"</code>, <code>"OR"</code>, <code>"RD"</code>, <code>"ASD"</code>,
<code>"HR"</code>, <code>"MD"</code>, or <code>"SMD"</code>, or <code>"ROM"</code>) is to
be used for pooling of studies.</p>
</td></tr>
<tr><td><code id="metacr_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metacr_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metacr_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metacr_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metacr_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metacr_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metacr_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metacr_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metacr_+3A_q.cochrane">Q.Cochrane</code></td>
<td>
<p>A logical indicating if the Mantel-Haenszel
estimate is used in the calculation of the heterogeneity
statistic Q which is implemented in RevMan 5.</p>
</td></tr>
<tr><td><code id="metacr_+3A_swap.events">swap.events</code></td>
<td>
<p>A logical indicating whether events and
non-events should be interchanged.</p>
</td></tr>
<tr><td><code id="metacr_+3A_logscale">logscale</code></td>
<td>
<p>A logical indicating whether effect estimates are
entered on log-scale (ignored for <code>cdir</code> objects).</p>
</td></tr>
<tr><td><code id="metacr_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots. If
<code>backtransf=TRUE</code> (default), results for <code>sm="OR"</code> are
printed as odds ratios rather than log odds ratios and results
for <code>sm="ZCOR"</code> are printed as correlations rather than
Fisher's z transformed correlations, for example.</p>
</td></tr>
<tr><td><code id="metacr_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metacr_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metacr_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metacr_+3A_rob">rob</code></td>
<td>
<p>A logical indicating whether risk of bias (RoB)
assessment should be considered in meta-analysis (only for
<code>read.cdir</code> objects).</p>
</td></tr>
<tr><td><code id="metacr_+3A_tool">tool</code></td>
<td>
<p>Risk of bias (RoB) tool (only for <code>read.cdir</code>
objects).</p>
</td></tr>
<tr><td><code id="metacr_+3A_categories">categories</code></td>
<td>
<p>Possible RoB categories (only for
<code>read.cdir</code> objects).</p>
</td></tr>
<tr><td><code id="metacr_+3A_col">col</code></td>
<td>
<p>Colours for RoB categories (only for <code>read.cdir</code>
objects).</p>
</td></tr>
<tr><td><code id="metacr_+3A_symbols">symbols</code></td>
<td>
<p>Corresponding symbols for RoB categories (only for
<code>read.cdir</code> objects).</p>
</td></tr>
<tr><td><code id="metacr_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metacr_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metacr_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metacr_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metacr_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metacr_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metacr_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metacr_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metacr_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metacr_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if <code>incr</code> is added to studies with zero cell
frequencies).</p>
</td></tr>
<tr><td><code id="metacr_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metacr_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Cochrane intervention reviews are based on the comparison of two
interventions. Each Cochrane intervention review can have a
variable number of comparisons. For each comparison, a variable
number of outcomes can be define. For each outcome, a seperate
meta-analysis is conducted. Review Manager 5 (RevMan 5) was the
software used for preparing and maintaining Cochrane Reviews
(<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>).
</p>
<p>This wrapper function can be used to perform meta-analysis for a
single outcome of a Cochrane intervention review. Internally, R
functions <code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>, and
<code><a href="#topic+metagen">metagen</a></code> are called - depending on the definition of
the outcome in RevMan 5.
</p>
<p>Information on the risk of bias RoB) assessment can be provided
with arguments <code>tool</code>, <code>categories</code>, <code>col</code> and
<code>symbols</code>. This is not useful if an overall RoB assessment has
been done. In this case use <code><a href="#topic+rob">rob</a></code> to add the full
flexible RoB information to a <code><a href="#topic+metacr">metacr</a></code> object.
</p>
<p>Note, it is recommended to choose the RevMan 5 settings before
executing <code>metacr</code>, i.e., <code>settings.meta("revman5")</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"meta"</code> and - depending on outcome type
utilised in Cochrane intervention review for selected outcome -
<code>"metabin"</code>, <code>"metacont"</code>, or <code>"metagen"</code> with
corresponding generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p><em>Review Manager (RevMan)</em> [Computer program]. Version 5.4.
The Cochrane Collaboration, 2020
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+rob">rob</a></code>,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+read.cdir">read.cdir</a></code>,
<code><a href="#topic+read.rm5">read.rm5</a></code>, <code><a href="#topic+settings.meta">settings.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate export data file "Fleiss1993_CR.csv"
# in sub-directory of package "meta"
#
filename &lt;- system.file("extdata", "Fleiss1993_CR.csv", package = "meta")
#
Fleiss1993_CR &lt;- read.rm5(filename)

# Choose RevMan 5 settings and store old settings
#
oldset &lt;- settings.meta("revman5", quietly = FALSE)

# Same result as R command example(Fleiss1993bin)
#
metacr(Fleiss1993_CR)

# Same result as R command example(Fleiss1993cont)
#
metacr(Fleiss1993_CR, 1, 2)
forest(metacr(Fleiss1993_CR, 1, 2))

# Change summary measure to RR
#
m1 &lt;- metacr(Fleiss1993_CR)
update(m1, sm="RR")

# Use old settings
#
settings.meta(oldset)

</code></pre>

<hr>
<h2 id='metacum.meta'>Cumulative meta-analysis</h2><span id='topic+metacum.meta'></span><span id='topic+metacum'></span><span id='topic+metacum.default'></span>

<h3>Description</h3>

<p>Performs a cumulative meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
metacum(x, pooled, sortvar, no = 1, ...)

metacum(x, ...)

## Default S3 method:
metacum(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metacum.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="metacum.meta_+3A_pooled">pooled</code></td>
<td>
<p>A character string indicating whether a common effect
or random effects model is used for pooling. Either missing (see
Details), <code>"common"</code>, or <code>"random"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metacum.meta_+3A_sortvar">sortvar</code></td>
<td>
<p>An optional vector used to sort the individual
studies (must be of same length as <code>x$TE</code>).</p>
</td></tr>
<tr><td><code id="metacum.meta_+3A_no">no</code></td>
<td>
<p>A numeric specifying which meta-analysis results to
consider.</p>
</td></tr>
<tr><td><code id="metacum.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A cumulative meta-analysis is performed. Studies are included
sequentially as defined by <code>sortvar</code>.
</p>
<p>Information from object <code>x</code> is utilised if argument
<code>pooled</code> is missing. A common effect model is assumed
(<code>pooled = "common"</code>) if argument <code>x$common</code> is
<code>TRUE</code>; a random effects model is assumed (<code>pooled =
"random"</code>) if argument <code>x$random</code> is <code>TRUE</code> and
<code>x$common</code> is <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"meta"</code> and <code>"metacum"</code> with
corresponding generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>
<p>The following list elements have a different meaning:
</p>
<table>
<tr><td><code>TE</code>, <code>seTE</code></td>
<td>
<p>Estimated treatment effect and standard error of
pooled estimate in cumulative meta-analyses.</p>
</td></tr>
<tr><td><code>lower</code>, <code>upper</code></td>
<td>
<p>Lower and upper confidence interval limits.</p>
</td></tr>
<tr><td><code>statistic</code></td>
<td>
<p>Statistic for test of overall effect.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>P-value for test of overall effect.</p>
</td></tr>
<tr><td><code>studlab</code></td>
<td>
<p>Study label describing addition of studies.</p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>Sum of weights from common effect or random effects model.</p>
</td></tr>
<tr><td><code>TE.common</code>, <code>seTE.common</code></td>
<td>
<p>Value is <code>NA</code>.</p>
</td></tr>
<tr><td><code>TE.random</code>, <code>seTE.random</code></td>
<td>
<p>Value is <code>NA</code>.</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>Value is <code>NA</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Cooper H &amp; Hedges LV (1994):
<em>The Handbook of Research Synthesis</em>.
Newbury Park, CA: Russell Sage Foundation
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac,
  data = Fleiss1993bin, studlab = study, sm = "RR", method = "I")
m1
metacum(m1)
metacum(m1, pooled = "random")

forest(metacum(m1))
forest(metacum(m1, pooled = "random"))

metacum(m1, sortvar = study)
metacum(m1, sortvar = 7:1)

m2 &lt;- update(m1, title = "Fleiss1993bin meta-analysis", backtransf = FALSE)
metacum(m2)

data(Fleiss1993cont)
m3 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")
metacum(m3)

</code></pre>

<hr>
<h2 id='metagen'>Generic inverse variance meta-analysis</h2><span id='topic+metagen'></span>

<h3>Description</h3>

<p>Common effect and random effects meta-analysis based on estimates
(e.g. log hazard ratios) and their standard errors. The inverse
variance method is used for pooling.
</p>
<p>Three-level random effects meta-analysis (Van den Noortgate et al.,
2013) is available by internally calling
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> function from R package
<b>metafor</b> (Viechtbauer, 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metagen(
  TE,
  seTE,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  sm = "",
  method.ci = if (missing(df)) "z" else "t",
  level = gs("level"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = gs("method.tau"),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  detail.tau = "",
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  null.effect = 0,
  method.bias = gs("method.bias"),
  n.e = NULL,
  n.c = NULL,
  pval,
  df,
  lower,
  upper,
  level.ci = 0.95,
  median,
  q1,
  q3,
  min,
  max,
  method.mean = "Luo",
  method.sd = "Shi",
  approx.TE,
  approx.seTE,
  transf = gs("transf") &amp; missing(func.transf),
  backtransf = gs("backtransf") | !missing(func.backtransf),
  func.transf,
  func.backtransf,
  args.transf,
  args.backtransf,
  pscale = 1,
  irscale = 1,
  irunit = "person-years",
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  label.e = gs("label.e"),
  label.c = gs("label.c"),
  label.left = gs("label.left"),
  label.right = gs("label.right"),
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  id,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metagen_+3A_te">TE</code></td>
<td>
<p>Estimate of treatment effect, e.g., log hazard ratio or
risk difference.</p>
</td></tr>
<tr><td><code id="metagen_+3A_sete">seTE</code></td>
<td>
<p>Standard error of treatment estimate.</p>
</td></tr>
<tr><td><code id="metagen_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metagen_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information.</p>
</td></tr>
<tr><td><code id="metagen_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used (see Details).</p>
</td></tr>
<tr><td><code id="metagen_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots (see Details).</p>
</td></tr>
<tr><td><code id="metagen_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metagen_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metagen_+3A_sm">sm</code></td>
<td>
<p>A character string indicating underlying summary measure,
e.g., <code>"RD"</code>, <code>"RR"</code>, <code>"OR"</code>, <code>"ASD"</code>,
<code>"HR"</code>, <code>"MD"</code>, <code>"SMD"</code>, or <code>"ROM"</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.ci">method.ci</code></td>
<td>
<p>A character string indicating which method is used
to calculate confidence intervals for individual studies, see
Details.</p>
</td></tr>
<tr><td><code id="metagen_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metagen_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metagen_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metagen_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metagen_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metagen_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metagen_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metagen_+3A_detail.tau">detail.tau</code></td>
<td>
<p>Detail on between-study variance estimate.</p>
</td></tr>
<tr><td><code id="metagen_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metagen_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, or
<code>"Thompson"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_n.e">n.e</code></td>
<td>
<p>Number of observations in experimental group (or total
sample size in study).</p>
</td></tr>
<tr><td><code id="metagen_+3A_n.c">n.c</code></td>
<td>
<p>Number of observations in control group.</p>
</td></tr>
<tr><td><code id="metagen_+3A_pval">pval</code></td>
<td>
<p>P-value (used to estimate the standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_df">df</code></td>
<td>
<p>Degrees of freedom (used in test or to construct
confidence interval).</p>
</td></tr>
<tr><td><code id="metagen_+3A_lower">lower</code></td>
<td>
<p>Lower limit of confidence interval (used to estimate
the standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_upper">upper</code></td>
<td>
<p>Upper limit of confidence interval (used to estimate
the standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_level.ci">level.ci</code></td>
<td>
<p>Level of confidence interval.</p>
</td></tr>
<tr><td><code id="metagen_+3A_median">median</code></td>
<td>
<p>Median (used to estimate the treatment effect and
standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_q1">q1</code></td>
<td>
<p>First quartile (used to estimate the treatment effect and
standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_q3">q3</code></td>
<td>
<p>Third quartile (used to estimate the treatment effect and
standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_min">min</code></td>
<td>
<p>Minimum (used to estimate the treatment effect and
standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_max">max</code></td>
<td>
<p>Maximum (used to estimate the treatment effect and
standard error).</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.mean">method.mean</code></td>
<td>
<p>A character string indicating which method to
use to approximate the mean from the median and other statistics
(see Details).</p>
</td></tr>
<tr><td><code id="metagen_+3A_method.sd">method.sd</code></td>
<td>
<p>A character string indicating which method to use
to approximate the standard deviation from sample size, median,
interquartile range and range (see Details).</p>
</td></tr>
<tr><td><code id="metagen_+3A_approx.te">approx.TE</code></td>
<td>
<p>Approximation method to estimate treatment
estimate (see Details).</p>
</td></tr>
<tr><td><code id="metagen_+3A_approx.sete">approx.seTE</code></td>
<td>
<p>Approximation method to estimate standard error
(see Details).</p>
</td></tr>
<tr><td><code id="metagen_+3A_transf">transf</code></td>
<td>
<p>A logical indicating whether inputs for arguments
<code>TE</code>, <code>lower</code> and <code>upper</code> are already
appropriately transformed to conduct the meta-analysis or on the
original scale. If <code>transf = TRUE</code> (default), inputs are
expected to be log odds ratios instead of odds ratios for
<code>sm = "OR"</code> and Fisher's z transformed correlations instead
of correlations for <code>sm = "ZCOR"</code>, for example.</p>
</td></tr>
<tr><td><code id="metagen_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots. If <code>backtransf =
TRUE</code> (default), results for <code>sm = "OR"</code> are printed as odds
ratios rather than log odds ratios and results for <code>sm =
"ZCOR"</code> are printed as correlations rather than Fisher's z
transformed correlations, for example.</p>
</td></tr>
<tr><td><code id="metagen_+3A_func.transf">func.transf</code></td>
<td>
<p>A function used to transform inputs for
arguments <code>TE</code>, <code>lower</code> and <code>upper</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_func.backtransf">func.backtransf</code></td>
<td>
<p>A function used to back-transform results.</p>
</td></tr>
<tr><td><code id="metagen_+3A_args.transf">args.transf</code></td>
<td>
<p>An optional list to provide additional arguments
to <code>func.transf</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_args.backtransf">args.backtransf</code></td>
<td>
<p>An optional list to provide additional
arguments to <code>func.backtransf</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
single event probabilities or risk differences, i.e. if argument
<code>sm</code> is equal to <code>"PLOGIT"</code>, <code>"PLN"</code>,
<code>"PRAW"</code>, <code>"PAS"</code>, <code>"PFT"</code>, or <code>"RD"</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single incidence rates or incidence rate differences, i.e. if
argument <code>sm</code> is equal to <code>"IR"</code>, <code>"IRLN"</code>,
<code>"IRS"</code>, <code>"IRFT"</code>, or <code>"IRD"</code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_irunit">irunit</code></td>
<td>
<p>A character specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="metagen_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metagen_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metagen_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metagen_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metagen_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metagen_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metagen_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metagen_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metagen_+3A_label.e">label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code id="metagen_+3A_label.c">label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code id="metagen_+3A_label.left">label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code id="metagen_+3A_label.right">label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code id="metagen_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metagen_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metagen_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metagen_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metagen_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metagen_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metagen_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metagen_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metagen_+3A_id">id</code></td>
<td>
<p>Deprecated argument (replaced by 'cluster').</p>
</td></tr>
<tr><td><code id="metagen_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by
'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metagen_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metagen_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if studies are excluded from meta-analysis due to zero
standard errors).</p>
</td></tr>
<tr><td><code id="metagen_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metagen_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> or
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code>.</p>
</td></tr>
<tr><td><code id="metagen_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides the <em>generic inverse variance method</em>
for meta-analysis which requires treatment estimates and their
standard errors (Borenstein et al., 2010). The method is useful,
e.g., for pooling of survival data (using log hazard ratio and
standard errors as input). Arguments <code>TE</code> and <code>seTE</code> can
be used to provide treatment estimates and standard errors
directly. However, it is possible to derive these quantities from
other information.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>


<h4>Approximate treatment estimates</h4>

<p>Missing treatment estimates can be derived from
</p>

<ol>
<li><p> confidence limits provided by arguments <code>lower</code> and
<code>upper</code>;
</p>
</li>
<li><p> median, interquartile range and range (arguments
<code>median</code>, <code>q1</code>, <code>q3</code>, <code>min</code>, and <code>max</code>);
</p>
</li>
<li><p> median and interquartile range (arguments <code>median</code>,
<code>q1</code> and <code>q3</code>);
</p>
</li>
<li><p> median and range (arguments <code>median</code>, <code>min</code> and
<code>max</code>).
</p>
</li></ol>

<p>For confidence limits, the treatment estimate is defined as the
center of the confidence interval (on the log scale for relative
effect measures like the odds ratio or hazard ratio).
</p>
<p>If the treatment effect is a mean it can be approximated from
sample size, median, interquartile range and range.
</p>
<p>By default, methods described in Luo et al. (2018) are utilized
(argument <code>method.mean = "Luo"</code>):
</p>

<ul>
<li><p> equation (7) if sample size, median and range are available,
</p>
</li>
<li><p> equation (11) if sample size, median and interquartile range
are available,
</p>
</li>
<li><p> equation (15) if sample size, median, range and interquartile
range are available.
</p>
</li></ul>

<p>Instead the methods described in Wan et al. (2014) are used if
argument <code>method.mean = "Wan"</code>:
</p>

<ul>
<li><p> equation (2) if sample size, median and range are available,
</p>
</li>
<li><p> equation (14) if sample size, median and interquartile range
are available,
</p>
</li>
<li><p> equation (10) if sample size, median, range and interquartile
range are available.
</p>
</li></ul>

<p>The following methods are also available to estimate means from
quantiles or ranges if R package <b>estmeansd</b> is installed:
</p>

<ul>
<li><p> Method for Unknown Non-Normal Distributions (MLN) approach
(Cai et al. (2021), argument <code>method.mean = "Cai"</code>),
</p>
</li>
<li><p> Quantile Estimation (QE) method (McGrath et al. (2020),
argument <code>method.mean = "QE-McGrath"</code>)),
</p>
</li>
<li><p> Box-Cox (BC) method (McGrath et al. (2020),
argument <code>method.mean = "BC-McGrath"</code>)).
</p>
</li></ul>

<p>By default, missing treatment estimates are replaced successively
using these method, i.e., confidence limits are utilised before
interquartile ranges. Argument <code>approx.TE</code> can be used to
overwrite this default for each individual study:
</p>

<ul>
<li><p> Use treatment estimate directly (entry <code>""</code> in argument
<code>approx.TE</code>);
</p>
</li>
<li><p> confidence limits (<code>"ci"</code> in argument <code>approx.TE</code>);
</p>
</li>
<li><p> median, interquartile range and range (<code>"iqr.range"</code>);
</p>
</li>
<li><p> median and interquartile range (<code>"iqr"</code>);
</p>
</li>
<li><p> median and range (<code>"range"</code>).
</p>
</li></ul>




<h4>Approximate standard errors</h4>

<p>Missing standard errors can be derived from
</p>

<ol>
<li><p> p-value provided by arguments <code>pval</code> and (optional)
<code>df</code>;
</p>
</li>
<li><p> confidence limits (arguments <code>lower</code>, <code>upper</code>, and
(optional) <code>df</code>);
</p>
</li>
<li><p> sample size, median, interquartile range and range (arguments
<code>n.e</code> and / or <code>n.c</code>, <code>median</code>, <code>q1</code>,
<code>q3</code>, <code>min</code>, and <code>max</code>);
</p>
</li>
<li><p> sample size, median and interquartile range (arguments
<code>n.e</code> and / or <code>n.c</code>, <code>median</code>, <code>q1</code> and
<code>q3</code>);
</p>
</li>
<li><p> sample size, median and range (arguments <code>n.e</code> and / or
<code>n.c</code>, <code>median</code>, <code>min</code> and <code>max</code>).
</p>
</li></ol>

<p>For p-values and confidence limits, calculations are either based
on the standard normal or <em>t</em>-distribution if argument
<code>df</code> is provided. Furthermore, argument <code>level.ci</code> can be
used to provide the level of the confidence interval.
</p>
<p>Wan et al. (2014) describe methods to estimate the standard
deviation (and thus the standard error by deviding the standard
deviation with the square root of the sample size) from the sample
size, median and additional statistics. Shi et al. (2020) provide
an improved estimate of the standard deviation if the interquartile
range and range are available in addition to the sample size and
median. Accordingly, equation (11) in Shi et al. (2020) is the
default (argument <code>method.sd = "Shi"</code>), if the median,
interquartile range and range are provided (arguments
<code>median</code>, <code>q1</code>, <code>q3</code>, <code>min</code> and
<code>max</code>). The method by Wan et al. (2014) is used if argument
<code>method.sd = "Wan"</code> and, depending on the sample size, either
equation (12) or (13) is used. If only the interquartile range or
range is available, equations (15) / (16) and (7) / (9) in Wan et
al. (2014) are used, respectively. The sample size of individual
studies must be provided with arguments <code>n.e</code> and / or
<code>n.c</code>. The total sample size is calculated as <code>n.e</code> +
<code>n.c</code> if both arguments are provided.
</p>
<p>The following methods are also available to estimate standard
deviations from quantiles or ranges if R package <b>estmeansd</b>
is installed:
</p>

<ul>
<li><p> Method for Unknown Non-Normal Distributions (MLN) approach
(Cai et al. (2021), argument <code>method.mean = "Cai"</code>),
</p>
</li>
<li><p> Quantile Estimation (QE) method (McGrath et al. (2020),
argument <code>method.mean = "QE-McGrath"</code>)),
</p>
</li>
<li><p> Box-Cox (BC) method (McGrath et al. (2020),
argument <code>method.mean = "BC-McGrath"</code>)).
</p>
</li></ul>

<p>By default, missing standard errors are replaced successively using
these method, e.g., p-value before confidence limits before
interquartile range and range. Argument <code>approx.seTE</code> can be
used to overwrite this default for each individual study:
</p>

<ul>
<li><p> Use standard error directly (entry <code>""</code> in argument
<code>approx.seTE</code>);
</p>
</li>
<li><p> p-value (<code>"pval"</code> in argument <code>approx.seTE</code>);
</p>
</li>
<li><p> confidence limits (<code>"ci"</code>);
</p>
</li>
<li><p> median, interquartile range and range (<code>"iqr.range"</code>);
</p>
</li>
<li><p> median and interquartile range (<code>"iqr"</code>);
</p>
</li>
<li><p> median and range (<code>"range"</code>).
</p>
</li></ul>




<h4>Confidence intervals for individual studies</h4>

<p>For the mean difference (argument <code>sm = "MD"</code>), the confidence
interval for individual studies can be based on the
</p>

<ul>
<li><p> standard normal distribution (<code>method.ci = "z"</code>), or
</p>
</li>
<li> <p><em>t</em>-distribution (<code>method.ci = "t"</code>).
</p>
</li></ul>

<p>By default, the first method is used if argument <code>df</code> is
missing and the second method otherwise.
</p>
<p>Note, this choice does not affect the results of the common effect
and random effects meta-analysis.
</p>



<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Specify the null hypothesis of test for an overall effect</h4>

<p>Argument <code>null.effect</code> can be used to specify the (treatment)
effect under the null hypothesis in a test for an overall
effect.
</p>
<p>By default (<code>null.effect = 0</code>), the null hypothesis
corresponds to &quot;no difference&quot; (which is obvious for absolute
effect measures like the mean difference (<code>sm = "MD"</code>) or
standardised mean difference (<code>sm = "SMD"</code>)). For relative
effect measures, e.g., risk ratio (<code>sm = "RR"</code>) or odds ratio
(<code>sm = "OR"</code>), the null effect is defined on the log scale,
i.e., <em>log</em>(RR) = 0 or <em>log</em>(OR) = 0 which is equivalent
to testing RR = 1 or OR = 1.
</p>
<p>Use of argument <code>null.effect</code> is especially useful for summary
measures without a &quot;natural&quot; null effect, i.e., in situations
without a second (treatment) group. For example, an overall
proportion of 50% could be tested in the meta-analysis of single
proportions with argument <code>null.effect = 0.5</code>.
</p>
<p>Note, all tests for an overall effect are two-sided with the
alternative hypothesis that the effect is unequal to
<code>null.effect</code>.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments <code>common</code>
and <code>random</code>. Accordingly, the estimate for the random effects
model can be extracted from component <code>TE.random</code> of an object
of class <code>"meta"</code> even if argument <code>random =
FALSE</code>. However, all functions in R package <b>meta</b> will
adequately consider the values for <code>common</code> and
<code>random</code>. For example, functions <code><a href="#topic+print.meta">print.meta</a></code> and
<code><a href="#topic+forest.meta">forest.meta</a></code> will not show results for the random
effects model if <code>random = FALSE</code>.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>
<p>Argument <code>pscale</code> can be used to rescale single proportions or
risk differences, e.g. <code>pscale = 1000</code> means that proportions
are expressed as events per 1000 observations. This is useful in
situations with (very) low event probabilities.
</p>
<p>Argument <code>irscale</code> can be used to rescale single rates or rate
differences, e.g. <code>irscale = 1000</code> means that rates are
expressed as events per 1000 time units, e.g. person-years. This is
useful in situations with (very) low rates. Argument <code>irunit</code>
can be used to specify the time unit used in individual studies
(default: &quot;person-years&quot;). This information is printed in summaries
and forest plots if argument <code>irscale</code> is not equal to 1.
</p>
<p>Default settings for <code>common</code>, <code>random</code>,
<code>pscale</code>, <code>irscale</code>, <code>irunit</code> and several other
arguments can be set for the whole R session using
<code><a href="#topic+settings.meta">settings.meta</a></code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metagen", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Note</h3>

<p>R function <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> from R package
<span class="pkg">metafor</span> (Viechtbauer 2010) is called internally to estimate
the between-study variance <code class="reqn">\tau^2</code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Borenstein M, Hedges LV, Higgins JP, Rothstein HR (2010):
A basic introduction to fixed-effect and random-effects models for
meta-analysis.
<em>Research Synthesis Methods</em>,
<b>1</b>, 97&ndash;111
</p>
<p>Cai S, Zhou J, Pan J (2021):
Estimating the sample mean and standard deviation from order
statistics and sample size in meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>30</b>, 2701&ndash;2719
</p>
<p>Luo D, Wan X, Liu J, Tong T (2018):
Optimally estimating the sample mean from the sample size, median,
mid-range, and/or mid-quartile range.
<em>Statistical Methods in Medical Research</em>,
<b>27</b>, 1785&ndash;805
</p>
<p>McGrath S, Zhao X, Steele R, et al. and the DEPRESsion Screening
Data (DEPRESSD) Collaboration (2020):
Estimating the sample mean and standard deviation from commonly
reported quantiles in meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>29</b>, 2520&ndash;2537
</p>
<p>Shi J, Luo D, Weng H, Zeng X-T, Lin L, Chu H, et al. (2020):
Optimally estimating the sample standard deviation from the
five-number summary.
<em>Research Synthesis Methods</em>.
</p>
<p>Viechtbauer W (2010):
Conducting Meta-Analyses in R with the metafor Package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Wan X, Wang W, Liu J, Tong T (2014):
Estimating the sample mean and standard deviation from the sample
size, median, range and/or interquartile range.
<em>BMC Medical Research Methodology</em>,
<b>14</b>, 135
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>, <code><a href="#topic+settings.meta">settings.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac, study,
  data = Fleiss1993bin, sm = "RR", method = "I")
m1

# Identical results using the generic inverse variance method with
# log risk ratio and its standard error:
# Note, argument 'n.e' in metagen() is used to provide the total
# sample size which is calculated from the group sample sizes n.e
# and n.c in meta-analysis m1.
m1.gen &lt;- metagen(TE, seTE, studlab, n.e = n.e + n.c, data = m1, sm = "RR")
m1.gen
forest(m1.gen, leftcols = c("studlab", "n.e", "TE", "seTE"))

# Meta-analysis with prespecified between-study variance
#
metagen(m1$TE, m1$seTE, sm = "RR", tau.preset = sqrt(0.1))

# Meta-analysis of survival data:
#
logHR &lt;- log(c(0.95, 1.5))
selogHR &lt;- c(0.25, 0.35)
metagen(logHR, selogHR, sm = "HR")

# Paule-Mandel method to estimate between-study variance for data
# from Paule &amp; Mandel (1982)
#
average &lt;- c(27.044, 26.022, 26.340, 26.787, 26.796)
variance &lt;- c(0.003, 0.076, 0.464, 0.003, 0.014)
#
metagen(average, sqrt(variance), sm = "MD", method.tau = "PM")

# Conduct meta-analysis using hazard ratios and 95% confidence intervals
#
# Data from Steurer et al. (2006), Analysis 1.1 Overall survival
# https://www.cochranelibrary.com/cdsr/doi/10.1002/14651858.CD004270.pub2/abstract
#
study &lt;- c("FCG on CLL 1996", "Leporrier 2001", "Rai 2000", "Robak 2000")
HR &lt;- c(0.55, 0.92, 0.79, 1.18)
lower.HR &lt;- c(0.28, 0.79, 0.59, 0.64)
upper.HR &lt;- c(1.09, 1.08, 1.05, 2.17)
#
# Hazard ratios and confidence intervals as input
#
summary(metagen(HR, lower = lower.HR, upper = upper.HR,
  studlab = study, sm = "HR", transf = FALSE))
#
# Same result with log hazard ratios as input
#
summary(metagen(log(HR), lower = log(lower.HR), upper = log(upper.HR),
  studlab = study, sm = "HR"))
#
# Again, same result using an unknown summary measure and
# arguments 'func.transf' and 'func.backtransf'
#
summary(metagen(HR, lower = lower.HR, upper = upper.HR,
  studlab = study, sm = "Hazard ratio",
  func.transf = log, func.backtransf = exp))
#
# Finally, same result only providing argument 'func.transf' as the
# back-transformation for the logarithm is known
#
summary(metagen(HR, lower = lower.HR, upper = upper.HR,
  studlab = study, sm = "Hazard ratio",
  func.transf = log))

# Exclude MRC-1 and MRC-2 studies from meta-analysis, however,
# show them in printouts and forest plots
#
metabin(d.asp, n.asp, d.plac, n.plac, study,
  data = Fleiss1993bin, sm = "RR", method = "I",
  exclude = study %in% c("MRC-1", "MRC-2"))
#
# Exclude MRC-1 and MRC-2 studies completely from meta-analysis
#
metabin(d.asp, n.asp, d.plac, n.plac, study,
  data = Fleiss1993bin, sm = "RR", method = "I",
  subset = !(study %in% c("MRC-1", "MRC-2")))

# Exclude studies with total sample size above 1500
#
metabin(d.asp, n.asp, d.plac, n.plac, study,
  data = Fleiss1993bin, sm = "RR", method = "I",
  exclude = (n.asp + n.plac) &gt; 1500)

# Exclude studies containing "MRC" in study name
#
metabin(d.asp, n.asp, d.plac, n.plac, study,
  data = Fleiss1993bin, sm = "RR", method = "I",
  exclude = grep("MRC", study))

# Use both arguments 'subset' and 'exclude'
#
metabin(d.asp, n.asp, d.plac, n.plac, study,
  data = Fleiss1993bin, sm = "RR", method = "I",
  subset = (n.asp + n.plac) &gt; 1500,
  exclude = grep("MRC", study))

## Not run: 
# Three-level model: effects of modified school calendars on
# student achievement
data(dat.konstantopoulos2011, package = "metadat")
metagen(yi, sqrt(vi), studlab = study, data = dat.konstantopoulos2011,
  sm = "SMD",
  cluster = district, detail.tau = c("district", "district/school"))

## End(Not run)

</code></pre>

<hr>
<h2 id='metainc'>Meta-analysis of incidence rates</h2><span id='topic+metainc'></span>

<h3>Description</h3>

<p>Calculation of common effect and random effects estimates (incidence
rate ratio or incidence rate difference) for meta-analyses with
event counts.  Mantel-Haenszel, Cochran, inverse variance method,
and generalised linear mixed model (GLMM) are available for
pooling. For GLMMs, the <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function
from R package <b>metafor</b> (Viechtbauer 2010) is called
internally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metainc(
  event.e,
  time.e,
  event.c,
  time.c,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  method = if (sm == "IRSD") "Inverse" else "MH",
  sm = gs("sminc"),
  incr = gs("incr"),
  method.incr = gs("method.incr"),
  model.glmm = "UM.FS",
  level = gs("level"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = ifelse(!is.na(charmatch(tolower(method), "glmm", nomatch = NA)), "ML",
    gs("method.tau")),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  method.bias = gs("method.bias"),
  n.e = NULL,
  n.c = NULL,
  backtransf = if (sm == "IRSD") FALSE else gs("backtransf"),
  irscale = 1,
  irunit = "person-years",
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  label.e = gs("label.e"),
  label.c = gs("label.c"),
  label.left = gs("label.left"),
  label.right = gs("label.right"),
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  hakn,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metainc_+3A_event.e">event.e</code></td>
<td>
<p>Number of events in experimental group.</p>
</td></tr>
<tr><td><code id="metainc_+3A_time.e">time.e</code></td>
<td>
<p>Person time at risk in experimental group.</p>
</td></tr>
<tr><td><code id="metainc_+3A_event.c">event.c</code></td>
<td>
<p>Number of events in control group.</p>
</td></tr>
<tr><td><code id="metainc_+3A_time.c">time.c</code></td>
<td>
<p>Person time at risk in control group.</p>
</td></tr>
<tr><td><code id="metainc_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metainc_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information, i.e., event.e, time.e, event.c, and time.c.</p>
</td></tr>
<tr><td><code id="metainc_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metainc_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metainc_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metainc_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metainc_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is to be
used for pooling of studies. One of <code>"MH"</code>,
<code>"Inverse"</code>, <code>"Cochran"</code>, or <code>"GLMM"</code> can be
abbreviated.</p>
</td></tr>
<tr><td><code id="metainc_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"IRR"</code>, <code>"IRD"</code>, <code>"IRSD"</code>, or <code>"VE"</code>) is to
be used for pooling of studies, see Details.</p>
</td></tr>
<tr><td><code id="metainc_+3A_incr">incr</code></td>
<td>
<p>A numerical value which is added to cell frequencies
for studies with a zero cell count, see Details.</p>
</td></tr>
<tr><td><code id="metainc_+3A_method.incr">method.incr</code></td>
<td>
<p>A character string indicating which continuity
correction method should be used (<code>"only0"</code>,
<code>"if0all"</code>, or <code>"all"</code>), see Details.</p>
</td></tr>
<tr><td><code id="metainc_+3A_model.glmm">model.glmm</code></td>
<td>
<p>A character string indicating which GLMM should
be used. One of <code>"UM.FS"</code>, <code>"UM.RS"</code>, and
<code>"CM.EL"</code>, see Details.</p>
</td></tr>
<tr><td><code id="metainc_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metainc_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metainc_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metainc_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metainc_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metainc_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metainc_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metainc_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metainc_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metainc_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metainc_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metainc_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metainc_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, or
<code>"Thompson"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code>.</p>
</td></tr>
<tr><td><code id="metainc_+3A_n.e">n.e</code></td>
<td>
<p>Number of observations in experimental group (optional).</p>
</td></tr>
<tr><td><code id="metainc_+3A_n.c">n.c</code></td>
<td>
<p>Number of observations in control group (optional).</p>
</td></tr>
<tr><td><code id="metainc_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for
incidence rate ratio (<code>sm = "IRR"</code>) and vaccine efficacy or
vaccine effectiveness (<code>sm = "VE"</code>) should be back
transformed in printouts and plots. If TRUE (default), results
will be presented as incidence rate ratios or vaccine efficacy /
effectiveness; otherwise log incidence rate ratios or log vaccine
rate ratios will be shown.</p>
</td></tr>
<tr><td><code id="metainc_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
incidence rate differences.</p>
</td></tr>
<tr><td><code id="metainc_+3A_irunit">irunit</code></td>
<td>
<p>A character string specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="metainc_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metainc_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metainc_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metainc_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metainc_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metainc_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metainc_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metainc_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metainc_+3A_label.e">label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code id="metainc_+3A_label.c">label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code id="metainc_+3A_label.left">label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code id="metainc_+3A_label.right">label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code id="metainc_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metainc_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metainc_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metainc_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metainc_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metainc_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metainc_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metainc_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metainc_+3A_hakn">hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'method.random.ci').</p>
</td></tr>
<tr><td><code id="metainc_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by
'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metainc_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metainc_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if <code>incr</code> is added to studies with zero cell
frequencies).</p>
</td></tr>
<tr><td><code id="metainc_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metainc_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> or
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="metainc_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function and to catch deprecated
arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Calculation of common and random effects estimates for meta-analyses
comparing two incidence rates.
</p>
<p>The following measures of treatment effect are available:
</p>

<ul>
<li><p> Incidence Rate Ratio (<code>sm = "IRR"</code>)
</p>
</li>
<li><p> Incidence Rate Difference (<code>sm = "IRD"</code>)
</p>
</li>
<li><p> Square root transformed Incidence Rate Difference (<code>sm =
  "IRSD"</code>)
</p>
</li>
<li><p> Vaccine efficacy or vaccine effectiveness (<code>sm = "VE"</code>)
</p>
</li></ul>

<p>Note, log incidence rate ratio (logIRR) and log vaccine ratio
(logVR) are mathematical identical, however, back-transformed
results differ as vaccine efficacy or effectiveness is defined as
<code>VE = 100 * (1 - IRR)</code>.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>


<h4>Meta-analysis method</h4>

<p>By default, both common effect and random effects models are
considered (see arguments <code>common</code> and <code>random</code>). If
<code>method</code> is <code>"MH"</code> (default), the Mantel-Haenszel method
is used to calculate the common effect estimate (Greenland &amp;
Robbins, 1985); if <code>method</code> is <code>"Inverse"</code>, inverse
variance weighting is used for pooling; if <code>method</code> is
<code>"Cochran"</code>, the Cochran method is used for pooling
(Bayne-Jones, 1964, Chapter 8). For these three methods, the random
effects estimate is always based on the inverse variance method.
</p>
<p>A distinctive and frequently overlooked advantage of incidence
rates is that individual patient data (IPD) can be extracted from
count data. Accordingly, statistical methods for IPD, i.e.,
generalised linear mixed models, can be utilised in a meta-analysis
of incidence rate ratios (Stijnen et al., 2010). These methods are
available (argument <code>method = "GLMM"</code>) for the common effect
and random effects model by calling the
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function from R package
<b>metafor</b> internally.
</p>
<p>Three different GLMMs are available for meta-analysis of incidence
rate ratios using argument <code>model.glmm</code> (which corresponds to
argument <code>model</code> in the <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>
function):
</p>

<table>
<tr>
 <td style="text-align: center;">
1. </td><td style="text-align: left;"> Poisson regression model with fixed study effects (default)
 </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "UM.FS"</code>, i.e., <b>U</b>nconditional
 <b>M</b>odel - <b>F</b>ixed <b>S</b>tudy effects) </td>
</tr>
<tr>
 <td style="text-align: center;">
2. </td><td style="text-align: left;"> Mixed-effects Poisson regression model with random study
 effects </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "UM.RS"</code>, i.e., <b>U</b>nconditional
 <b>M</b>odel - <b>R</b>andom <b>S</b>tudy effects) </td>
</tr>
<tr>
 <td style="text-align: center;">
3. </td><td style="text-align: left;"> Generalised linear mixed model (conditional Poisson-Normal)
 </td>
</tr>
<tr>
 <td style="text-align: center;">
 </td><td style="text-align: left;"> (<code>model.glmm = "CM.EL"</code>, i.e., <b>C</b>onditional
  <b>M</b>odel - <b>E</b>xact <b>L</b>ikelihood)
</td>
</tr>

</table>

<p>Details on these three GLMMs as well as additional arguments which
can be provided using argument '<code>...</code>' in <code>metainc</code>
are described in <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> where you can also
find information on the iterative algorithms used for estimation.
Note, regardless of which value is used for argument
<code>model.glmm</code>, results for two different GLMMs are calculated:
common effect model (with fixed treatment effect) and random effects
model (with random treatment effects).
</p>



<h4>Continuity correction</h4>

<p>Three approaches are available to apply a continuity correction:
</p>

<ul>
<li><p> Only studies with a zero cell count (<code>method.incr =
  "only0", default</code>)
</p>
</li>
<li><p> All studies if at least one study has a zero cell count
(<code>method.incr = "if0all"</code>)
</p>
</li>
<li><p> All studies irrespective of zero cell counts
(<code>method.incr = "all"</code>)
</p>
</li></ul>

<p>For studies with a zero cell count, by default, 0.5 is added to all
cell frequencies of these studies (argument <code>incr</code>). This
continuity correction is used both to calculate individual study
results with confidence limits and to conduct meta-analysis based
on the inverse variance method. For Mantel-Haenszel method, Cochran
method, and GLMMs, nothing is added to zero cell counts.
Accordingly, estimates for these methods are not defined if the
number of events is zero in all studies either in the experimental
or control group.
</p>



<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments
<code>common</code> and <code>random</code>. Accordingly, the estimate
for the random effects model can be extracted from component
<code>TE.random</code> of an object of class <code>"meta"</code> even if
argument <code>random = FALSE</code>. However, all functions in R
package <b>meta</b> will adequately consider the values for
<code>common</code> and <code>random</code>. E.g. function
<code><a href="#topic+print.meta">print.meta</a></code> will not print results for the random
effects model if <code>random = FALSE</code>.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metainc", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Bayne-Jones S et al. (1964):
Smoking and Health: Report of the Advisory Committee to the Surgeon
General of the United States.
U-23 Department of Health, Education, and Welfare.
Public Health Service Publication No. 1103.
</p>
<p>Greenland S &amp; Robins JM (1985):
Estimation of a common effect parameter from sparse follow-up data.
<em>Biometrics</em>,
<b>41</b>, 55&ndash;68
</p>
<p>Stijnen T, Hamza TH, Ozdemir P (2010):
Random effects meta-analysis of event outcome in the framework of
the generalized linear mixed model with applications in sparse
data.
<em>Statistics in Medicine</em>,
<b>29</b>, 3046&ndash;67
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Viechtbauer W (2010):
Conducting Meta-Analyses in R with the Metafor Package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+update.meta">update.meta</a></code>, <code><a href="#topic+print.meta">print.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(smoking)
m1 &lt;- metainc(d.smokers, py.smokers, d.nonsmokers, py.nonsmokers,
  data = smoking, studlab = study)
print(m1, digits = 2)

m2 &lt;- update(m1, method = "Cochran")
print(m2, digits = 2)

data(lungcancer)
m3 &lt;- metainc(d.smokers, py.smokers, d.nonsmokers, py.nonsmokers,
  data = lungcancer, studlab = study)
print(m3, digits = 2)

# Redo Cochran meta-analysis with inflated standard errors
#
# All cause mortality
#
TEa &lt;- log((smoking$d.smokers/smoking$py.smokers) /
  (smoking$d.nonsmokers/smoking$py.nonsmokers))
seTEa &lt;- sqrt(1 / smoking$d.smokers + 1 / smoking$d.nonsmokers +
  2.5 / smoking$d.nonsmokers)
metagen(TEa, seTEa, sm = "IRR", studlab = smoking$study)

# Lung cancer mortality
#
TEl &lt;- log((lungcancer$d.smokers/lungcancer$py.smokers) /
  (lungcancer$d.nonsmokers/lungcancer$py.nonsmokers))
seTEl &lt;- sqrt(1 / lungcancer$d.smokers + 1 / lungcancer$d.nonsmokers +
  2.25 / lungcancer$d.nonsmokers)
metagen(TEl, seTEl, sm = "IRR", studlab = lungcancer$study)

## Not run: 
# Meta-analysis using generalised linear mixed models
# (only if R packages 'metafor' and 'lme4' are available)

# Poisson regression model (fixed study effects)
#
m4 &lt;- metainc(d.smokers, py.smokers, d.nonsmokers, py.nonsmokers,
  data = smoking, studlab = study, method = "GLMM")
m4

# Mixed-effects Poisson regression model (random study effects)
#
update(m4, model.glmm = "UM.RS", nAGQ = 1)
#
# Generalised linear mixed model (conditional Poisson-Normal)
#
update(m4, model.glmm = "CM.EL")

## End(Not run)

</code></pre>

<hr>
<h2 id='metainf.meta'>Influence analysis in meta-analysis using leave-one-out method</h2><span id='topic+metainf.meta'></span><span id='topic+metainf'></span><span id='topic+metainf.default'></span>

<h3>Description</h3>

<p>Performs an influence analysis. Pooled estimates are calculated
omitting one study at a time.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
metainf(x, pooled, sortvar, no = 1, ...)

metainf(x, ...)

## Default S3 method:
metainf(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metainf.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="metainf.meta_+3A_pooled">pooled</code></td>
<td>
<p>A character string indicating whether a common effect
or random effects model is used for pooling. Either missing (see
Details), <code>"common"</code> or <code>"random"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metainf.meta_+3A_sortvar">sortvar</code></td>
<td>
<p>An optional vector used to sort the individual
studies (must be of same length as <code>x$TE</code>).</p>
</td></tr>
<tr><td><code id="metainf.meta_+3A_no">no</code></td>
<td>
<p>A numeric specifying which meta-analysis results to
consider.</p>
</td></tr>
<tr><td><code id="metainf.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Performs a influence analysis; pooled estimates are calculated
omitting one study at a time. Studies are sorted according to
<code>sortvar</code>.
</p>
<p>Information from object <code>x</code> is utilised if argument
<code>pooled</code> is missing. A common effect model is assumed
(<code>pooled="common"</code>) if argument <code>x$common</code> is
<code>TRUE</code>; a random effects model is assumed
(<code>pooled="random"</code>) if argument <code>x$random</code> is
<code>TRUE</code> and <code>x$common</code> is <code>FALSE</code>.
</p>


<h3>Value</h3>

<p>An object of class <code>"meta"</code> and <code>"metainf"</code> with
corresponding generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>
<p>The following list elements have a different meaning:
</p>
<table>
<tr><td><code>TE</code>, <code>seTE</code></td>
<td>
<p>Estimated treatment effect and standard error of
pooled estimate in influence analysis.</p>
</td></tr>
<tr><td><code>lower</code>, <code>upper</code></td>
<td>
<p>Lower and upper confidence interval limits.</p>
</td></tr>
<tr><td><code>statistic</code></td>
<td>
<p>Statistic for test of overall effect.</p>
</td></tr>
<tr><td><code>pval</code></td>
<td>
<p>P-value for test of overall effect.</p>
</td></tr>
<tr><td><code>studlab</code></td>
<td>
<p>Study label describing omission of studies.</p>
</td></tr>
<tr><td><code>w</code></td>
<td>
<p>Sum of weights from common effect or random effects model.</p>
</td></tr>
<tr><td><code>TE.common</code>, <code>seTE.common</code></td>
<td>
<p>Value is <code>NA</code>.</p>
</td></tr>
<tr><td><code>TE.random</code>, <code>seTE.random</code></td>
<td>
<p>Value is <code>NA</code>.</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>Value is <code>NA</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Cooper H &amp; Hedges LV (1994):
<em>The Handbook of Research Synthesis</em>.
Newbury Park, CA: Russell Sage Foundation
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac,
  data = Fleiss1993bin, studlab = study, sm = "RR", method = "I")
m1
metainf(m1)
metainf(m1, pooled = "random")

forest(metainf(m1))
forest(metainf(m1), layout = "revman5")
forest(metainf(m1, pooled = "random"))

metainf(m1, sortvar = study)
metainf(m1, sortvar = 7:1)

m2 &lt;- update(m1, title = "Fleiss1993bin meta-analysis", backtransf = FALSE)
metainf(m2)

data(Fleiss1993cont)
m3 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")
metainf(m3)

</code></pre>

<hr>
<h2 id='metamean'>Meta-analysis of single means</h2><span id='topic+metamean'></span>

<h3>Description</h3>

<p>Calculation of an overall mean from studies reporting a single mean
using the inverse variance method for pooling; inverse variance
weighting is used for pooling.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metamean(
  n,
  mean,
  sd,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  median,
  q1,
  q3,
  min,
  max,
  method.mean = "Luo",
  method.sd = "Shi",
  approx.mean,
  approx.sd,
  sm = gs("smmean"),
  method.ci = gs("method.ci.cont"),
  level = gs("level"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = gs("method.tau"),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  null.effect = NA,
  method.bias = gs("method.bias"),
  backtransf = gs("backtransf"),
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metamean_+3A_n">n</code></td>
<td>
<p>Number of observations.</p>
</td></tr>
<tr><td><code id="metamean_+3A_mean">mean</code></td>
<td>
<p>Estimated mean.</p>
</td></tr>
<tr><td><code id="metamean_+3A_sd">sd</code></td>
<td>
<p>Standard deviation.</p>
</td></tr>
<tr><td><code id="metamean_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metamean_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information.</p>
</td></tr>
<tr><td><code id="metamean_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metamean_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metamean_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metamean_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metamean_+3A_median">median</code></td>
<td>
<p>Median (used to estimate the mean and standard
deviation).</p>
</td></tr>
<tr><td><code id="metamean_+3A_q1">q1</code></td>
<td>
<p>First quartile (used to estimate the mean and standard
deviation).</p>
</td></tr>
<tr><td><code id="metamean_+3A_q3">q3</code></td>
<td>
<p>Third quartile (used to estimate the mean and standard
deviation).</p>
</td></tr>
<tr><td><code id="metamean_+3A_min">min</code></td>
<td>
<p>Minimum (used to estimate the mean and standard
deviation).</p>
</td></tr>
<tr><td><code id="metamean_+3A_max">max</code></td>
<td>
<p>Maximum (used to estimate the mean and standard
deviation).</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.mean">method.mean</code></td>
<td>
<p>A character string indicating which method to
use to approximate the mean from the median and other statistics
(see Details).</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.sd">method.sd</code></td>
<td>
<p>A character string indicating which method to use
to approximate the standard deviation from sample size, median,
interquartile range and range (see Details).</p>
</td></tr>
<tr><td><code id="metamean_+3A_approx.mean">approx.mean</code></td>
<td>
<p>Approximation method to estimate means (see
Details).</p>
</td></tr>
<tr><td><code id="metamean_+3A_approx.sd">approx.sd</code></td>
<td>
<p>Approximation method to estimate standard
deviations (see Details).</p>
</td></tr>
<tr><td><code id="metamean_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"MRAW"</code> or <code>"MLN"</code>) is to be used for pooling of
studies.</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.ci">method.ci</code></td>
<td>
<p>A character string indicating which method is used
to calculate confidence intervals for individual studies, see
Details.</p>
</td></tr>
<tr><td><code id="metamean_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metamean_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metamean_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metamean_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metamean_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metamean_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metamean_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metamean_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metamean_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metamean_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
<tr><td><code id="metamean_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, or
<code>"Thompson"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code>.</p>
</td></tr>
<tr><td><code id="metamean_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots for <code>sm = "MLN"</code>. If
TRUE (default), results will be presented as means; otherwise
logarithm of means will be shown.</p>
</td></tr>
<tr><td><code id="metamean_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metamean_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metamean_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metamean_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metamean_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metamean_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metamean_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metamean_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metamean_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metamean_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metamean_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metamean_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metamean_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metamean_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metamean_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metamean_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metamean_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metamean_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metamean_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if studies are excluded from meta-analysis due to zero
standard deviations).</p>
</td></tr>
<tr><td><code id="metamean_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metamean_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>.</p>
</td></tr>
<tr><td><code id="metamean_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Common effect and random effects meta-analysis of single means to
calculate an overall mean; inverse variance weighting is used for
pooling. Note, you should use R function <code><a href="#topic+metacont">metacont</a></code> to
compare means of pairwise comparisons instead of using
<code>metamean</code> for each treatment arm separately which will break
randomisation in randomised controlled trials.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>
<p>The following transformations of means are implemented to
calculate an overall mean:
</p>

<ul>
<li><p> Raw, i.e. untransformed, means (<code>sm = "MRAW"</code>, default)
</p>
</li>
<li><p> Log transformed means (<code>sm = "MLN"</code>)
</p>
</li></ul>

<p>Calculations are conducted on the log scale if <code>sm =
"ROM"</code>. Accordingly, list elements <code>TE</code>, <code>TE.common</code>, and
<code>TE.random</code> contain the logarithm of means. In printouts and
plots these values are back transformed if argument
<code>backtransf = TRUE</code> (default).
</p>


<h4>Approximate means from sample sizes, medians and other statistics</h4>

<p>Missing means can be derived from
</p>

<ol>
<li><p> sample size, median, interquartile range and range (arguments
<code>n</code>, <code>median</code>, <code>q1</code>, <code>q3</code>, <code>min</code>, and
<code>max</code>),
</p>
</li>
<li><p> sample size, median and interquartile range (arguments
<code>n</code>, <code>median</code>, <code>q1</code>, and <code>q3</code>), or
</p>
</li>
<li><p> sample size, median and range (arguments <code>n</code>,
<code>median</code>, <code>min</code>, and <code>max</code>).
</p>
</li></ol>

<p>By default, methods described in Luo et al. (2018) are utilized
(argument <code>method.mean = "Luo"</code>):
</p>

<ul>
<li><p> equation (15) if sample size, median, interquartile range and 
range are available,
</p>
</li>
<li><p> equation (11) if sample size, median and interquartile range
are available,
</p>
</li>
<li><p> equation (7) if sample size, median and range are available.
</p>
</li></ul>

<p>Instead the methods described in Wan et al. (2014) are used if
argument <code>method.mean = "Wan"</code>:
</p>

<ul>
<li><p> equation (10) if sample size, median, interquartile range and 
range are available,
</p>
</li>
<li><p> equation (14) if sample size, median and interquartile range
are available,
</p>
</li>
<li><p> equation (2) if sample size, median and range are available.
</p>
</li></ul>

<p>The following methods are also available to estimate means from
quantiles or ranges if R package <b>estmeansd</b> is installed:
</p>

<ul>
<li><p> Method for Unknown Non-Normal Distributions (MLN) approach
(Cai et al. (2021), argument <code>method.mean = "Cai"</code>),
</p>
</li>
<li><p> Quantile Estimation (QE) method (McGrath et al. (2020),
argument <code>method.mean = "QE-McGrath"</code>)),
</p>
</li>
<li><p> Box-Cox (BC) method (McGrath et al. (2020),
argument <code>method.mean = "BC-McGrath"</code>)).
</p>
</li></ul>

<p>By default, missing means are replaced successively using
interquartile ranges and ranges (if available), interquartile
ranges (if available) and finally ranges. Argument
<code>approx.mean</code> can be used to overwrite this behaviour for each
individual study and treatment arm:
</p>

<ul>
<li><p> use means directly (entry <code>""</code> in argument
<code>approx.mean</code>);
</p>
</li>
<li><p> median, interquartile range and range (<code>"iqr.range"</code>);
</p>
</li>
<li><p> median and interquartile range (<code>"iqr"</code>);
</p>
</li>
<li><p> median and range (<code>"range"</code>).
</p>
</li></ul>




<h4>Approximate standard deviations from sample sizes, medians and other statistics</h4>

<p>Missing standard deviations can be derived from
</p>

<ol>
<li><p> sample size, median, interquartile range and range (arguments
<code>n</code>, <code>median</code>, <code>q1</code>, <code>q3</code>, <code>min</code>, and
<code>max</code>),
</p>
</li>
<li><p> sample size, median and interquartile range (arguments
<code>n</code>, <code>median</code>, <code>q1</code> and <code>q3</code>), or
</p>
</li>
<li><p> sample size, median and range (arguments <code>n</code>,
<code>median</code>, <code>min</code> and <code>max</code>).
</p>
</li></ol>

<p>Wan et al. (2014) describe methods to estimate the standard
deviation from the sample size, median and additional
statistics. Shi et al. (2020) provide an improved estimate of the
standard deviation if the interquartile range and range are
available in addition to the sample size and median. Accordingly,
equation (11) in Shi et al. (2020) is the default (argument
<code>method.sd = "Shi"</code>), if the median, interquartile range and
range are provided. The method by Wan et al. (2014) is used if
argument <code>method.sd = "Wan"</code> and, depending on the sample
size, either equation (12) or (13) is used. If only the
interquartile range or range is available, equations (15) / (16)
and (7) / (9) in Wan et al. (2014) are used, respectively.
</p>
<p>The following methods are also available to estimate standard
deviations from quantiles or ranges if R package <b>estmeansd</b>
is installed:
</p>

<ul>
<li><p> Method for Unknown Non-Normal Distributions (MLN) approach
(Cai et al. (2021), argument <code>method.mean = "Cai"</code>),
</p>
</li>
<li><p> Quantile Estimation (QE) method (McGrath et al. (2020),
argument <code>method.mean = "QE-McGrath"</code>)),
</p>
</li>
<li><p> Box-Cox (BC) method (McGrath et al. (2020),
argument <code>method.mean = "BC-McGrath"</code>)).
</p>
</li></ul>

<p>By default, missing standard deviations are replaced successively
using these method, i.e., interquartile ranges and ranges are used
before interquartile ranges before ranges. Argument
<code>approx.sd</code> can be used to overwrite this default for each
individual study and treatment arms:
</p>

<ul>
<li><p> sample size, median, interquartile range and range
(<code>"iqr.range"</code>);
</p>
</li>
<li><p> sample size, median and interquartile range (<code>"iqr"</code>);
</p>
</li>
<li><p> sample size, median and range (<code>"range"</code>).
</p>
</li></ul>




<h4>Confidence intervals for individual studies</h4>

<p>For untransformed means (argument <code>sm = "MRAW"</code>), the
confidence interval for individual studies can be based on the
</p>

<ul>
<li><p> standard normal distribution (<code>method.ci = "z"</code>, default), or
</p>
</li>
<li><p> t-distribution (<code>method.ci = "t"</code>).
</p>
</li></ul>




<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments
<code>common</code> and <code>random</code>. Accordingly, the estimate
for the random effects model can be extracted from component
<code>TE.random</code> of an object of class <code>"meta"</code> even if
argument <code>random = FALSE</code>. However, all functions in R
package <b>meta</b> will adequately consider the values for
<code>common</code> and <code>random</code>. E.g. functions
<code><a href="#topic+print.meta">print.meta</a></code> and <code><a href="#topic+forest.meta">forest.meta</a></code> will not
print results for the random effects model if <code>random =
FALSE</code>.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metamean", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Note</h3>

<p>The function <code><a href="#topic+metagen">metagen</a></code> is called internally to
calculate individual and overall treatment estimates and standard
errors.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Cai S, Zhou J, Pan J (2021):
Estimating the sample mean and standard deviation from order
statistics and sample size in meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>30</b>, 2701&ndash;2719
</p>
<p>Luo D, Wan X, Liu J, Tong T (2018):
Optimally estimating the sample mean from the sample size, median,
mid-range, and/or mid-quartile range.
<em>Statistical Methods in Medical Research</em>,
<b>27</b>, 1785&ndash;805
</p>
<p>McGrath S, Zhao X, Steele R, et al. and the DEPRESsion Screening
Data (DEPRESSD) Collaboration (2020):
Estimating the sample mean and standard deviation from commonly
reported quantiles in meta-analysis.
<em>Statistical Methods in Medical Research</em>,
<b>29</b>, 2520&ndash;2537
</p>
<p>Shi J, Luo D, Weng H, Zeng X-T, Lin L, Chu H, et al. (2020):
Optimally estimating the sample standard deviation from the
five-number summary.
<em>Research Synthesis Methods</em>.
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca
J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Wan X, Wang W, Liu J, Tong T (2014):
Estimating the sample mean and standard deviation from the sample
size, median, range and/or interquartile range.
<em>BMC Medical Research Methodology</em>,
<b>14</b>, 135
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metamean">metamean</a></code>, <code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>m1 &lt;- metamean(rep(100, 3), 1:3, rep(1, 3))
m1

m2 &lt;- update(m1, sm = "MLN")
m2

# With test for overall mean equal to 2
#
update(m1, null.effect = 2)
update(m2, null.effect = 2)

# Print results without back-transformation
#
update(m1, backtransf = FALSE)
update(m2, backtransf = FALSE)
update(m1, null.effect = 2, backtransf = FALSE)
update(m2, null.effect = 2, backtransf = FALSE)

</code></pre>

<hr>
<h2 id='metamerge'>Merge results of two meta-analyses on the same data set</h2><span id='topic+metamerge'></span>

<h3>Description</h3>

<p>This function can be used to merge results of two meta-analyses
into a single meta-analysis object if they are based on the same
data set. This is, for example, useful to produce a forest plot of
a random-effects meta-analysis with different estimates of the
between-study variance <code class="reqn">\tau^2</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metamerge(
  meta1,
  meta2,
  common1 = meta1$common,
  random1 = meta1$random,
  prediction1 = meta1$prediction,
  common2 = meta2$common,
  random2 = meta2$random,
  prediction2 = meta2$prediction,
  label1 = NULL,
  label2 = NULL,
  label1.common = label1,
  label2.common = label2,
  label1.random = label1,
  label2.random = label2,
  label1.predict = label1,
  label2.predict = label2,
  label1.subgroup = label1,
  label2.subgroup = label2,
  hetlabel1 = label1.random,
  hetlabel2 = label2.random,
  taulabel1 = label1.random,
  taulabel2 = label2.random,
  text.pooled1 = NULL,
  text.pooled2 = NULL,
  text.w.pooled1 = NULL,
  text.w.pooled2 = NULL,
  text.common1 = text.pooled1,
  text.common2 = text.pooled2,
  text.random1 = text.pooled1,
  text.random2 = text.pooled2,
  text.predict1 = text.pooled1,
  text.predict2 = text.pooled2,
  text.w.common1 = text.w.pooled1,
  text.w.common2 = text.w.pooled2,
  text.w.random1 = text.w.pooled1,
  text.w.random2 = text.w.pooled2,
  keep = FALSE,
  keep.Q = keep,
  keep.I2 = keep.Q,
  keep.w = keep,
  common = common1 | common2,
  random = random1 | random2,
  overall = common | random,
  overall.hetstat = common | random,
  prediction = prediction1 | prediction2,
  backtransf,
  warn.deprecated = gs("warn.deprecated"),
  pooled1,
  pooled2
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metamerge_+3A_meta1">meta1</code></td>
<td>
<p>First meta-analysis object (see Details).</p>
</td></tr>
<tr><td><code id="metamerge_+3A_meta2">meta2</code></td>
<td>
<p>Second meta-analysis object (see Details).</p>
</td></tr>
<tr><td><code id="metamerge_+3A_common1">common1</code></td>
<td>
<p>A logical indicating whether results of common
effect model should be considered for first meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_random1">random1</code></td>
<td>
<p>A logical indicating whether results of random
effects model should be considered for first meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_prediction1">prediction1</code></td>
<td>
<p>A logical indicating whether prediction interval
should be considered for first meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_common2">common2</code></td>
<td>
<p>A logical indicating whether results of common
effect model should be considered for second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_random2">random2</code></td>
<td>
<p>A logical indicating whether results of random
effects model should be considered for second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_prediction2">prediction2</code></td>
<td>
<p>A logical indicating whether prediction interval
should be considered for second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label1">label1</code></td>
<td>
<p>Default setting for arguments 'label1.common',
'label1.random', 'label1.predict' and 'label1.subgroup'.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label2">label2</code></td>
<td>
<p>Default setting for arguments 'label2.common',
'label2.random', 'label2.predict' and 'label2.subgroup'.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label1.common">label1.common</code></td>
<td>
<p>A character string ...</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label2.common">label2.common</code></td>
<td>
<p>A character string ...</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label1.random">label1.random</code></td>
<td>
<p>A character string ... (default label for
arguments 'hetlabel1' and 'taulabel1').</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label2.random">label2.random</code></td>
<td>
<p>A character string ... (default label for
arguments 'hetlabel2' and 'taulabel2').</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label1.predict">label1.predict</code></td>
<td>
<p>A character string ...</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label2.predict">label2.predict</code></td>
<td>
<p>A character string ...</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label1.subgroup">label1.subgroup</code></td>
<td>
<p>A character string ...</p>
</td></tr>
<tr><td><code id="metamerge_+3A_label2.subgroup">label2.subgroup</code></td>
<td>
<p>A character string ...</p>
</td></tr>
<tr><td><code id="metamerge_+3A_hetlabel1">hetlabel1</code></td>
<td>
<p>A character string used to label heterogeneity
statistics of the first meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_hetlabel2">hetlabel2</code></td>
<td>
<p>A character string used to label heterogeneity
statistics of the second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_taulabel1">taulabel1</code></td>
<td>
<p>A character string used to label estimate of
between-study variance of the first meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_taulabel2">taulabel2</code></td>
<td>
<p>A character string used to label estimate of
between-study variance of the second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.pooled1">text.pooled1</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the results from the first meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.pooled2">text.pooled2</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the results from the second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.w.pooled1">text.w.pooled1</code></td>
<td>
<p>A character string used to label weights of
the first meta-analysis; can be of same length as the number of
pooled estimates requested in argument <code>pooled1</code>.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.w.pooled2">text.w.pooled2</code></td>
<td>
<p>A character string used to label weights of
the second meta-analysis; can be of same length as the number of
pooled estimates requested in argument <code>pooled1</code>.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.common1">text.common1</code></td>
<td>
<p>A character string used in printouts and forest
plot to label results for common effect models from the first
meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.common2">text.common2</code></td>
<td>
<p>A character string used in printouts and forest
plot to label results for common effect models from the second
meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.random1">text.random1</code></td>
<td>
<p>A character string used in printouts and forest
plot to label results for random effects models from the first
meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.random2">text.random2</code></td>
<td>
<p>A character string used in printouts and forest
plot to label results for random effects models from the second
meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.predict1">text.predict1</code></td>
<td>
<p>A character string used in printouts and
forest plot to label prediction interval from the first
meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.predict2">text.predict2</code></td>
<td>
<p>A character string used in printouts and
forest plot to label prediction interval from the second
meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.w.common1">text.w.common1</code></td>
<td>
<p>A character string used to label common
effect weights of the first meta-analysis; can be of same length
as the number of common effect estimates.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.w.common2">text.w.common2</code></td>
<td>
<p>A character string used to label common
effect weights of the second meta-analysis; can be of same length
as the number of common effect estimates.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.w.random1">text.w.random1</code></td>
<td>
<p>A character string used to label random
effects weights of the first meta-analysis; can be of same length
as the number of random effects estimates.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_text.w.random2">text.w.random2</code></td>
<td>
<p>A character string used to label random
effects weights of the second meta-analysis; can be of same
length as the number of random effects estimates.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_keep">keep</code></td>
<td>
<p>A logical indicating whether to keep additional
information from second meta-analysis.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_keep.q">keep.Q</code></td>
<td>
<p>A logical indicating whether heterogeneity statistic
Q of second meta-analysis should be kept or ignored.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_keep.i2">keep.I2</code></td>
<td>
<p>A logical indicating whether heterogeneity statistic
I2 of second meta-analysis should be kept or ignored.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_keep.w">keep.w</code></td>
<td>
<p>A logical indicating whether weights of the second
meta-analysis should be kept or ignored.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_common">common</code></td>
<td>
<p>A logical indicating whether results of common effect
meta-analyses should be reported.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_random">random</code></td>
<td>
<p>A logical indicating whether results of random
effects meta-analyses should be reported.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether prediction intervals
should be reported.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots. If
<code>backtransf=TRUE</code> (default), results for <code>sm="OR"</code> are
printed as odds ratios rather than log odds ratios, for example.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_pooled1">pooled1</code></td>
<td>
<p>Deprecated argument (replaced by 'common1',
'random1', 'prediction1'). A character string indicating whether
results of common effect or random effects model should be
considered for first meta-analysis. Either <code>"both"</code>,
<code>"common"</code> or <code>"random"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metamerge_+3A_pooled2">pooled2</code></td>
<td>
<p>Deprecated argument (replaced by 'common2',
'random2', 'prediction2'). A character string indicating whether
results of common effect or random effects model should be
considered for second meta-analysis. Either <code>"both"</code>,
<code>"common"</code> or <code>"random"</code>, can be abbreviated.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>In R package <b>meta</b>, objects of class <code>"meta"</code> contain
results of both common effect and random effects
meta-analyses. This function enables the user to merge the results
of two meta-analysis object if they are based on the same data set.
</p>
<p>Applications of this function include printing and plotting results
of the common effect or random effects meta-analysis and the
</p>

<ul>
<li><p> trim-and-fill method (<code><a href="#topic+trimfill">trimfill</a></code>),
</p>
</li>
<li><p> limit meta-analyis (<code><a href="metasens.html#topic+limitmeta">limitmeta</a></code> from R
package <b>metasens</b>),
</p>
</li>
<li><p> Copas selection model (<code><a href="metasens.html#topic+copas">copas</a></code> from R
package <b>metasens</b>),
</p>
</li>
<li><p> robust variance meta-analysis model
(<code><a href="robumeta.html#topic+robu">robu</a></code> from R package <b>robumeta</b>).
</p>
</li></ul>

<p>The first argument (<code>meta1</code>) must be an object created by a
meta-analysis function (see <code><a href="#topic+meta-object">meta-object</a></code>). If an
object created with <code><a href="metasens.html#topic+limitmeta">limitmeta</a></code> or
<code><a href="metasens.html#topic+copas">copas</a></code> is provided as the first argument,
this object will be returned, i.e., argument <code>meta2</code> will be
ignored.
</p>
<p>The second meta-analysis could be an object created by a
meta-analysis function or with <code><a href="#topic+trimfill">trimfill</a></code>,
<code><a href="metasens.html#topic+limitmeta">limitmeta</a></code>, <code><a href="metasens.html#topic+copas">copas</a></code>,
or <code><a href="robumeta.html#topic+robu">robu</a></code>.
</p>
<p>The created meta-analysis object only contains the study results,
i.e., estimated effects and confidence intervals, from the first
meta-analysis which are shown in printouts and forest plots. This
only makes a difference for meta-analysis methods where individual
study results differ, e.g., Mantel-Haenszel and Peto method for
binary outcomes (see <code><a href="#topic+metabin">metabin</a></code>).
</p>
<p>R function <code><a href="#topic+metaadd">metaadd</a></code> can be used to add pooled results
from any (external) meta-analysis.
</p>
<p>R function <code><a href="#topic+metabind">metabind</a></code> can be used to print and plot the
results of several meta-analyses without the restriction that the
same data set has to be used. Accordingly, individual study results
are ignored.
</p>


<h3>Value</h3>

<p>An object of class <code>"meta"</code> and <code>"metamerge"</code> with
corresponding generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>
<p>The following list elements have a different meaning:
</p>
<table>
<tr><td><code>TE</code>, <code>seTE</code>, <code>studlab</code></td>
<td>
<p>Treatment estimate, standard error, and
study labels (first meta-analyis).</p>
</td></tr>
<tr><td><code>lower</code>, <code>upper</code></td>
<td>
<p>Lower and upper confidence interval limits for
individual studies (first meta-analysis).</p>
</td></tr>
<tr><td><code>statistic</code>, <code>pval</code></td>
<td>
<p>Statistic and p-value for test of treatment
effect for individual studies (first meta-analysis.</p>
</td></tr>
<tr><td><code>w.common</code></td>
<td>
<p>Vector or matrix with common effect weights.</p>
</td></tr>
<tr><td><code>w.random</code></td>
<td>
<p>Vector or matrix with random effects weights.</p>
</td></tr>
<tr><td><code>k</code></td>
<td>
<p>Vector with number of estimates (same length as number of
common effect and random effects estimates).</p>
</td></tr>
<tr><td><code>k.study</code></td>
<td>
<p>Vector with number of studies (same length as
number of common effect and random effects estimates).</p>
</td></tr>
<tr><td><code>k.all</code></td>
<td>
<p>Vector with total number of studies (same length as
number of common effect and random effects estimates).</p>
</td></tr>
<tr><td><code>k.TE</code></td>
<td>
<p>Vector with number of studies with estimable effects
(same length as number of common effect and random effects
estimates).</p>
</td></tr>
<tr><td><code>k.MH</code></td>
<td>
<p>Vector with number of studies combined with
Mantel-Haenszel method (same length as number of common effect
and random effects estimates).</p>
</td></tr>
<tr><td><code>TE.common</code></td>
<td>
<p>Vector with common effect estimates.</p>
</td></tr>
<tr><td><code>seTE.common</code></td>
<td>
<p>Vector with standard errors of common effect
estimates.</p>
</td></tr>
<tr><td><code>lower.common</code></td>
<td>
<p>Vector with lower confidence limits (common
effect model).</p>
</td></tr>
<tr><td><code>upper.common</code></td>
<td>
<p>Vector with upper confidence limits (common
effect model).</p>
</td></tr>
<tr><td><code>statistic.common</code></td>
<td>
<p>Vector with test statistics for test of
overall effect (common effect model).</p>
</td></tr>
<tr><td><code>pval.common</code></td>
<td>
<p>Vector with p-value of test for overall effect
(common effect model).</p>
</td></tr>
<tr><td><code>TE.random</code></td>
<td>
<p>Vector with random effects estimates.</p>
</td></tr>
<tr><td><code>seTE.random</code></td>
<td>
<p>Vector with standard errors of random effects
estimates.</p>
</td></tr>
<tr><td><code>lower.random</code></td>
<td>
<p>Vector with lower confidence limits (random
effects model).</p>
</td></tr>
<tr><td><code>upper.random</code></td>
<td>
<p>Vector with upper confidence limits (random
effects model).</p>
</td></tr>
<tr><td><code>statistic.random</code></td>
<td>
<p>Vector with test statistics for test of
overall effect (random effects model).</p>
</td></tr>
<tr><td><code>pval.random</code></td>
<td>
<p>Vector with p-value of test for overall effect
(random effects model).</p>
</td></tr>
</table>
<p>Furthermore, meta-analysis results of common effect or random
effects model are taken from first meta-analysis if only random
effects or common effects models are selected from both
meta-analyses (arguments <code>pooled1</code> and <code>pooled2</code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metabind">metabind</a></code>,
<code><a href="#topic+metaadd">metaadd</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Print results with more significant digits and do not show confidence
# intervals for tau^2 and tau
oldset &lt;- settings.meta(digits = 6, digits.stat = 4, digits.pval = 6,
  digits.Q = 6, digits.I2 = 4, digits.H = 4,
  print.tau2.ci = FALSE, print.tau.ci = FALSE)
oldopts &lt;- options(width = 120)

data(Fleiss1993bin)

# Mantel-Haenszel method
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac, data = Fleiss1993bin,
  studlab = paste(study, year), sm = "OR")
# Peto method
m2 &lt;- update(m1, method = "Peto")
# Inverse variance method (only common effect model)
m3 &lt;- update(m2, method = "Inverse", random = FALSE)

# Merge results from MH and Peto method
# - show individual results for MH method
#   (as this is the first meta-analysis)
# - keep all additional information from Peto meta-analysis (i.e.,
#   weights, Q statistic and I2 statistic)
m12 &lt;- metamerge(m1, m2,
  label1 = "REML", label2 = "REML-Peto",
  label1.common = "MH", label2.common = "Peto", 
  text.common1 = "Mantel-Haenszel method",
  text.common2 = "Peto method",
  text.w.random1 = "REML", text.w.random2 = "REML-Peto",
  hetlabel1 = "MH/IV", hetlabel2 = "Peto",
  keep = TRUE)

# Add common effect results from inverse variance method
# - keep weights from IV meta-analysis
# - Q and I2 statistic are identical for sm = "MH" and sm = "Inverse"
#   as inverse variance method is used for sm = "MH" under random
#   effects model
m123 &lt;- metamerge(m12, m3,
  label2 = "IV",
  text.common2 = "Inverse variance method",
  keep.w = TRUE)
summary(m123)
## Not run: 
forest(m123, digits = 6)

# Merge results (show individual results for Peto method)
m21 &lt;- metamerge(m2, m1,
  label1 = "REML-Peto", label2 = "REML",
  label1.common = "Peto", label2.common = "MH", 
  hetlabel1 = "Peto", hetlabel2 = "MH/IV",
  text.common1 = "Peto method",
  text.common2 = "Mantel-Haenszel method",
  keep = TRUE)

# Add results from inverse variance method
# - keep weights from IV meta-analysis
# - Q and I2 statistic are identical for sm = "MH" and sm = "Inverse"
#   as inverse variance method is used for sm = "MH" under random
#   effects model
m213 &lt;- metamerge(m21, m3,
  label2 = "IV",
  text.common2 = "Inverse variance method",
  keep.w = TRUE)
summary(m213)

# Random effects method using ML estimator for between-study variance tau2
m4 &lt;- update(m1, common = FALSE, method.tau = "ML")

# Use DerSimonian-Laird estimator for tau2
m5 &lt;- update(m4, method.tau = "DL")

# Use Paule-Mandel estimator for tau2
m6 &lt;- update(m4, method.tau = "PM")

# Merge random effects results for ML and DL estimators
# - keep weights for DL estimator (which are different from ML)
m45 &lt;- metamerge(m4, m5, label1 = "ML", label2 = "DL",
  text.w.random1 = "RE-ML", text.w.random2 = "RE-DL", keep.w = TRUE)
summary(m45)

# Add results for PM estimator
# - keep weights
m456 &lt;- metamerge(m45, m6, label2 = "PM",
  text.w.random2 = "RE-PM", keep.w = TRUE)
summary(m456)

m123456 &lt;- metamerge(m123, m456)
m123456

# Use Hartung-Knapp confidence intervals
# - do not keep information on Q, I2 and weights
m7 &lt;- update(m4, method.random.ci = "HK",
  text.random = "Hartung-Knapp method")
m8 &lt;- update(m5, method.random.ci = "HK",
  text.random = "Hartung-Knapp method")
m9 &lt;- update(m6, method.random.ci = "HK",
  text.random = "Hartung-Knapp method")

# Merge results for Hartung-Knapp method (with REML and DL estimator)
# - RE weights for REML estimator are shown
m78 &lt;- metamerge(m7, m8, label1 = "ML", label2 = "DL")
summary(m78)

m789 &lt;- metamerge(m78, m9, label2 = "PM")
summary(m789)

# Merge everything
m1to9 &lt;- metamerge(metamerge(m123, m456, keep.w = TRUE), m789)
summary(m1to9)

m10 &lt;- update(m1, method = "GLMM")

m.all &lt;- metamerge(m1to9, m10, keep.Q = TRUE,
  label2 = "GLMM", taulabel2 = "ML-GLMM")
summary(m.all)

forest(m.all, layout = "JAMA")
forest(m.all, details = TRUE)

## End(Not run)

settings.meta(oldset)
options(oldopts)

</code></pre>

<hr>
<h2 id='metaprop'>Meta-analysis of single proportions</h2><span id='topic+metaprop'></span>

<h3>Description</h3>

<p>Calculation of an overall proportion from studies reporting a
single proportion. Inverse variance method and generalised linear
mixed model (GLMM) are available for pooling. For GLMMs, the
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function from R package
<b>metafor</b> (Viechtbauer 2010) is called internally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metaprop(
  event,
  n,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  method,
  sm = gs("smprop"),
  incr = gs("incr"),
  method.incr = gs("method.incr"),
  method.ci = gs("method.ci.prop"),
  level = gs("level"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau = ifelse(!is.na(charmatch(tolower(method), "glmm", nomatch = NA)), "ML",
    gs("method.tau")),
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  null.effect = NA,
  method.bias = gs("method.bias"),
  backtransf = gs("backtransf"),
  pscale = 1,
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  hakn,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metaprop_+3A_event">event</code></td>
<td>
<p>Number of events.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_n">n</code></td>
<td>
<p>Number of observations.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information, i.e., event and n.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is to be
used for pooling of studies. One of <code>"Inverse"</code> and
<code>"GLMM"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"PLOGIT"</code>, <code>"PAS"</code>, <code>"PFT"</code>, <code>"PLN"</code>, or
<code>"PRAW"</code>) is to be used for pooling of studies, see Details.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_incr">incr</code></td>
<td>
<p>A numeric which is added to event number and sample
size of studies with zero or all events, i.e., studies with an
event probability of either 0 or 1.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.incr">method.incr</code></td>
<td>
<p>A character string indicating which continuity
correction method should be used (<code>"only0"</code>,
<code>"if0all"</code>, or <code>"all"</code>), see Details.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.ci">method.ci</code></td>
<td>
<p>A character string indicating which method is used
to calculate confidence intervals for individual studies, see
Details.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metaprop_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, or
<code>"Thompson"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code>.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for
transformed proportions (argument <code>sm != "PRAW"</code>) should be
back transformed in printouts and plots. If TRUE (default),
results will be presented as proportions; otherwise transformed
proportions will be shown. See Details for presentation of
confidence intervals.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_pscale">pscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single event probabilities.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metaprop_+3A_hakn">hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'method.random.ci').</p>
</td></tr>
<tr><td><code id="metaprop_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metaprop_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether the addition of
<code>incr</code> to studies with zero or all events should result in a
warning.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> or
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="metaprop_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function and to catch deprecated
arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides methods for common effect and random effects
meta-analysis of single proportions to calculate an overall
proportion. Note, you should use R function <code><a href="#topic+metabin">metabin</a></code>
to compare proportions of pairwise comparisons instead of using
<code>metaprop</code> for each treatment arm separately which will break
randomisation in randomised controlled trials.
</p>
<p>The following transformations of proportions are
implemented to calculate an overall proportion:
</p>

<ul>
<li><p> Logit transformation (<code>sm = "PLOGIT"</code>, default)
</p>
</li>
<li><p> Arcsine transformation (<code>sm = "PAS"</code>)
</p>
</li>
<li><p> Freeman-Tukey Double arcsine transformation (<code>sm = "PFT"</code>)
</p>
</li>
<li><p> Log transformation (<code>sm = "PLN"</code>)
</p>
</li>
<li><p> No transformation (<code>sm = "PRAW"</code>)
</p>
</li></ul>

<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain the transformed proportions. In printouts and plots these
values are back transformed if argument <code>backtransf = TRUE</code>
(default).
</p>
<p>A generalised linear mixed model (GLMM) - more specific, a random
intercept logistic regression model - can be utilised for the
meta-analysis of proportions (Stijnen et al., 2010). This is the
default method for the logit transformation (argument <code>sm =
"PLOGIT"</code>). Internally, the <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>
function from R package <b>metafor</b> is called to fit a GLMM.
</p>
<p>Classic meta-analysis (Borenstein et al., 2010) utilising the
(un)transformed proportions and corresponding standard errors in
the inverse variance method is conducted by calling the
<code><a href="#topic+metagen">metagen</a></code> function internally. This is the only
available method for all transformations but the logit
transformation. The classic meta-analysis model with logit
transformed proportions is used by setting argument <code>method =
"Inverse"</code>.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>


<h4>Choice of transformation / meta-analysis method</h4>

<p>Contradictory recommendations on the use of transformations of
proportions have been published in the literature. For example,
Barendregt et al. (2013) recommend the use of the Freeman-Tukey
double arcsine transformation instead of the logit transformation
whereas Warton &amp; Hui (2011) strongly advise to use generalised
linear mixed models with the logit transformation instead of the
arcsine transformation.
</p>
<p>Schwarzer et al. (2019) describe seriously misleading results in a
meta-analysis with very different sample sizes due to problems with
the back-transformation of the Freeman-Tukey transformation which
requires a single sample size (Miller, 1978). Accordingly,
Schwarzer et al. (2019) also recommend to use GLMMs for the
meta-analysis of single proportions, however, admit that individual
study weights are not available with this method. Meta-analysts
which require individual study weights should consider the inverse
variance method with the arcsine or logit transformation.
</p>
<p>In order to prevent misleading conclusions for the Freeman-Tukey
double arcsine transformation, sensitivity analyses using other
transformations or using a range of sample sizes should be
conducted (Schwarzer et al., 2019).
</p>



<h4>Continuity correction</h4>

<p>Three approaches are available to apply a continuity correction:
</p>

<ul>
<li><p> Only studies with a zero cell count (<code>method.incr =
  "only0"</code>)
</p>
</li>
<li><p> All studies if at least one study has a zero cell count
(<code>method.incr = "if0all"</code>)
</p>
</li>
<li><p> All studies irrespective of zero cell counts
(<code>method.incr = "all"</code>)
</p>
</li></ul>

<p>If the summary measure is equal to &quot;PLOGIT&quot;, &quot;PLN&quot;, or &quot;PRAW&quot;, the
continuity correction is applied if a study has either zero or all
events, i.e., an event probability of either 0 or 1.
</p>
<p>By default, 0.5 is used as continuity correction (argument
<code>incr</code>). This continuity correction is used both to calculate
individual study results with confidence limits and to conduct
meta-analysis based on the inverse variance method. For GLMMs no
continuity correction is used.
</p>



<h4>Confidence intervals for individual studies</h4>

<p>Various methods are available to calculate confidence intervals for
individual study results (see Agresti &amp; Coull 1998 and Newcombe
1988):
</p>

<ul>
<li><p> Clopper-Pearson interval also called 'exact' binomial
interval (<code>method.ci = "CP"</code>, default)
</p>
</li>
<li><p> Wilson Score interval (<code>method.ci = "WS"</code>)
</p>
</li>
<li><p> Wilson Score interval with continuity correction
(<code>method.ci = "WSCC"</code>)
</p>
</li>
<li><p> Agresti-Coull interval (<code>method.ci = "AC"</code>)
</p>
</li>
<li><p> Simple approximation interval (<code>method.ci = "SA"</code>)
</p>
</li>
<li><p> Simple approximation interval with continuity correction
(<code>method.ci = "SACC"</code>)
</p>
</li>
<li><p> Normal approximation interval based on summary measure,
i.e. defined by argument <code>sm</code> (<code>method.ci = "NAsm"</code>)
</p>
</li></ul>

<p>Note, with exception of the normal approximation based on the
summary measure, i.e. <code>method.ci = "NAsm"</code>, the same
confidence interval is calculated for individual studies for any
summary measure (argument <code>sm</code>) as only number of events and
observations are used in the calculation disregarding the chosen
transformation.
</p>
<p>Results will be presented for transformed proportions if argument
<code>backtransf = FALSE</code>. In this case, argument <code>method.ci =
"NAsm"</code> is used, i.e. confidence intervals based on the normal
approximation based on the summary measure.
</p>



<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Specify the null hypothesis of test for an overall proportion</h4>

<p>Argument <code>null.effect</code> can be used to specify the proportion
used under the null hypothesis in a test for an overall effect.
</p>
<p>By default (<code>null.effect = NA</code>), no hypothesis test is
conducted as it is unclear which value is a sensible choice for the
data at hand.  An overall proportion of 50%, for example, could be
tested by setting argument <code>null.effect = 0.5</code>.
</p>
<p>Note, all tests for an overall effect are two-sided with the
alternative hypothesis that the effect is unequal to
<code>null.effect</code>.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments
<code>common</code> and <code>random</code>. Accordingly, the estimate
for the random effects model can be extracted from component
<code>TE.random</code> of an object of class <code>"meta"</code> even if
argument <code>random = FALSE</code>. However, all functions in R
package <b>meta</b> will adequately consider the values for
<code>common</code> and <code>random</code>. E.g. function
<code><a href="#topic+print.meta">print.meta</a></code> will not print results for the random
effects model if <code>random = FALSE</code>.
</p>
<p>Argument <code>pscale</code> can be used to rescale proportions, e.g.
<code>pscale = 1000</code> means that proportions are expressed as events
per 1000 observations. This is useful in situations with (very) low
event probabilities.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metaprop", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Agresti A &amp; Coull BA (1998):
Approximate is better than &quot;exact&quot; for interval estimation of
binomial proportions.
<em>The American Statistician</em>,
<b>52</b>, 119&ndash;26
</p>
<p>Barendregt JJ, Doi SA, Lee YY, Norman RE, Vos T (2013):
Meta-analysis of prevalence.
<em>Journal of Epidemiology and Community Health</em>,
<b>67</b>, 974&ndash;8
</p>
<p>Borenstein M, Hedges LV, Higgins JP, Rothstein HR (2010):
A basic introduction to fixed-effect and random-effects models for
meta-analysis.
<em>Research Synthesis Methods</em>,
<b>1</b>, 97&ndash;111
</p>
<p>Freeman MF &amp; Tukey JW (1950):
Transformations related to the angular and the square root.
<em>Annals of Mathematical Statistics</em>,
<b>21</b>, 607&ndash;11
</p>
<p>Miller JJ (1978):
The inverse of the Freeman-Tukey double arcsine transformation.
<em>The American Statistician</em>,
<b>32</b>, 138
</p>
<p>Newcombe RG (1998):
Two-sided confidence intervals for the single proportion:
comparison of seven methods.
<em>Statistics in Medicine</em>,
<b>17</b>, 857&ndash;72
</p>
<p>Pettigrew HM, Gart JJ, Thomas DG (1986):
The bias and higher cumulants of the logarithm of a binomial
variate.
<em>Biometrika</em>,
<b>73</b>, 425&ndash;35
</p>
<p>Schwarzer G, Chemaitelly H, Abu-Raddad LJ, Rücker G (2019):
Seriously misleading results using inverse of Freeman-Tukey double
arcsine transformation in meta-analysis of single proportions.
<em>Research Synthesis Methods</em>,
<b>10</b>, 476&ndash;83
</p>
<p>Stijnen T, Hamza TH, Ozdemir P (2010):
Random effects meta-analysis of event outcome in the framework of
the generalized linear mixed model with applications in sparse
data.
<em>Statistics in Medicine</em>,
<b>29</b>, 3046&ndash;67
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Viechtbauer W (2010):
Conducting meta-analyses in R with the metafor package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>
<p>Warton DI, Hui FKC (2011):
The arcsine is asinine: the analysis of proportions in ecology.
<em>Ecology</em>,
<b>92</b>, 3&ndash;10
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metacont">metacont</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>, <code><a href="#topic+forest.meta">forest.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Meta-analysis using generalised linear mixed model
#
metaprop(4:1, 10 * 1:4)

# Apply various classic meta-analysis methods to estimate
# proportions
#
m1 &lt;- metaprop(4:1, 10 * 1:4, method = "Inverse")
m2 &lt;- update(m1, sm = "PAS")
m3 &lt;- update(m1, sm = "PRAW")
m4 &lt;- update(m1, sm = "PLN")
m5 &lt;- update(m1, sm = "PFT")
#
m1
m2
m3
m4
m5
#
forest(m1)
## Not run: 
forest(m2)
forest(m3)
forest(m3, pscale = 100)
forest(m4)
forest(m5)

## End(Not run)

# Do not back transform results, e.g. print logit transformed
# proportions if sm = "PLOGIT" and store old settings
#
oldset &lt;- settings.meta(backtransf = FALSE)
#
m6  &lt;- metaprop(4:1, c(10, 20, 30, 40), method = "Inverse")
m7  &lt;- update(m6, sm = "PAS")
m8  &lt;- update(m6, sm = "PRAW")
m9  &lt;- update(m6, sm = "PLN")
m10 &lt;- update(m6, sm = "PFT")
#
forest(m6)
## Not run: 
forest(m7)
forest(m8)
forest(m8, pscale = 100)
forest(m9)
forest(m10)

## End(Not run)

# Use old settings
#
settings.meta(oldset)

# Examples with zero events
#
m1 &lt;- metaprop(c(0, 0, 10, 10), rep(100, 4), method = "Inverse")
m2 &lt;- metaprop(c(0, 0, 10, 10), rep(100, 4), incr = 0.1, method = "Inverse")
#
m1
m2
#
## Not run: 
forest(m1)
forest(m2)

## End(Not run)

# Example from Miller (1978):
#
death &lt;- c(3, 6, 10, 1)
animals &lt;- c(11, 17, 21, 6)
#
m3 &lt;- metaprop(death, animals, sm = "PFT")
forest(m3)

# Data examples from Newcombe (1998)
# - apply various methods to estimate confidence intervals for
#   individual studies
#
event &lt;- c(81, 15, 0, 1)
n &lt;- c(263, 148, 20, 29)
#
m1 &lt;- metaprop(event, n, method.ci = "SA", method = "Inverse")
m2 &lt;- update(m1, method.ci = "SACC")
m3 &lt;- update(m1, method.ci = "WS")
m4 &lt;- update(m1, method.ci = "WSCC")
m5 &lt;- update(m1, method.ci = "CP")
#
lower &lt;- round(logit2p(rbind(NA, m1$lower, m2$lower, NA, m3$lower,
  m4$lower, NA, m5$lower)), 4)
upper &lt;- round(logit2p(rbind(NA, m1$upper, m2$upper, NA, m3$upper,
  m4$upper, NA, m5$upper)), 4)
#
tab1 &lt;- data.frame(
  scen1 = meta:::formatCI(lower[, 1], upper[, 1]),
  scen2 = meta:::formatCI(lower[, 2], upper[, 2]),
  scen3 = meta:::formatCI(lower[, 3], upper[, 3]),
  scen4 = meta:::formatCI(lower[, 4], upper[, 4])
  )
names(tab1) &lt;- c("r=81, n=263", "r=15, n=148",
  "r=0, n=20", "r=1, n=29")
row.names(tab1) &lt;- c("Simple", "- SA", "- SACC",
  "Score", "- WS", "- WSCC", "Binomial", "- CP")
tab1[is.na(tab1)] &lt;- ""
# Newcombe (1998), Table I, methods 1-5:
tab1

# Same confidence interval, i.e. unaffected by choice of summary
# measure
#
print(metaprop(event, n, method.ci = "WS", method = "Inverse"), ma = FALSE)
print(metaprop(event, n, sm = "PLN", method.ci = "WS"), ma = FALSE)
print(metaprop(event, n, sm = "PFT", method.ci = "WS"), ma = FALSE)
print(metaprop(event, n, sm = "PAS", method.ci = "WS"), ma = FALSE)
print(metaprop(event, n, sm = "PRAW", method.ci = "WS"), ma = FALSE)

# Different confidence intervals as argument sm = "NAsm"
#
print(metaprop(event, n, method.ci = "NAsm", method = "Inverse"), ma = FALSE)
print(metaprop(event, n, sm = "PLN", method.ci = "NAsm"), ma = FALSE)
print(metaprop(event, n, sm = "PFT", method.ci = "NAsm"), ma = FALSE)
print(metaprop(event, n, sm = "PAS", method.ci = "NAsm"), ma = FALSE)
print(metaprop(event, n, sm = "PRAW", method.ci = "NAsm"), ma = FALSE)

# Different confidence intervals as argument backtransf = FALSE.
# Accordingly, method.ci = "NAsm" used internally.
#
print(metaprop(event, n, method.ci = "WS", method = "Inverse"),
  ma = FALSE, backtransf = FALSE)
print(metaprop(event, n, sm = "PLN", method.ci = "WS"),
  ma = FALSE, backtransf = FALSE)
print(metaprop(event, n, sm = "PFT", method.ci = "WS"),
  ma = FALSE, backtransf = FALSE)
print(metaprop(event, n, sm = "PAS", method.ci = "WS"),
  ma = FALSE, backtransf = FALSE)
print(metaprop(event, n, sm = "PRAW", method.ci = "WS"),
  ma = FALSE, backtransf = FALSE)

# Same results (printed on original and log scale, respectively)
#
print(metaprop(event, n, sm = "PLN", method.ci = "NAsm"), ma = FALSE)
print(metaprop(event, n, sm = "PLN"), ma = FALSE, backtransf = FALSE)
# Results for first study (on log scale)
round(log(c(0.3079848, 0.2569522, 0.3691529)), 4)

# Print results as events per 1000 observations
#
print(metaprop(6:8, c(100, 1200, 1000), method = "Inverse"),
  pscale = 1000, digits = 1)

</code></pre>

<hr>
<h2 id='metarate'>Meta-analysis of single incidence rates</h2><span id='topic+metarate'></span>

<h3>Description</h3>

<p>Calculation of an overall incidence rate from studies reporting a
single incidence rate. Inverse variance method and generalised
linear mixed model (GLMM) are available for pooling. For GLMMs, the
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function from R package
<b>metafor</b> (Viechtbauer 2010) is called internally.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metarate(
  event,
  time,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  cluster = NULL,
  rho = 0,
  n = NULL,
  method = "Inverse",
  sm = gs("smrate"),
  incr = gs("incr"),
  method.incr = gs("method.incr"),
  method.ci = gs("method.ci.rate"),
  level = gs("level"),
  common = gs("common"),
  random = gs("random") | !is.null(tau.preset),
  overall = common | random,
  overall.hetstat = if (is.null(gs("overall.hetstat"))) common | random else
    gs("overall.hetstat"),
  prediction = gs("prediction") | !missing(method.predict),
  method.tau,
  method.tau.ci = gs("method.tau.ci"),
  tau.preset = NULL,
  TE.tau = NULL,
  tau.common = gs("tau.common"),
  level.ma = gs("level.ma"),
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  level.predict = gs("level.predict"),
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  null.effect = NA,
  method.bias = gs("method.bias"),
  backtransf = gs("backtransf"),
  irscale = 1,
  irunit = "person-years",
  text.common = gs("text.common"),
  text.random = gs("text.random"),
  text.predict = gs("text.predict"),
  text.w.common = gs("text.w.common"),
  text.w.random = gs("text.w.random"),
  title = gs("title"),
  complab = gs("complab"),
  outclab = "",
  subgroup,
  subgroup.name = NULL,
  print.subgroup.name = gs("print.subgroup.name"),
  sep.subgroup = gs("sep.subgroup"),
  test.subgroup = gs("test.subgroup"),
  prediction.subgroup = gs("prediction.subgroup"),
  seed.predict.subgroup = NULL,
  byvar,
  hakn,
  adhoc.hakn,
  keepdata = gs("keepdata"),
  warn = gs("warn"),
  warn.deprecated = gs("warn.deprecated"),
  control = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metarate_+3A_event">event</code></td>
<td>
<p>Number of events.</p>
</td></tr>
<tr><td><code id="metarate_+3A_time">time</code></td>
<td>
<p>Person time at risk.</p>
</td></tr>
<tr><td><code id="metarate_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels.</p>
</td></tr>
<tr><td><code id="metarate_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study
information, i.e., event and time.</p>
</td></tr>
<tr><td><code id="metarate_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used.</p>
</td></tr>
<tr><td><code id="metarate_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="metarate_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="metarate_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="metarate_+3A_n">n</code></td>
<td>
<p>Number of observations.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is to be
used for pooling of studies. One of <code>"Inverse"</code> and
<code>"GLMM"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="metarate_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure
(<code>"IR"</code>, <code>"IRLN"</code>, <code>"IRS"</code>, or <code>"IRFT"</code>) is
to be used for pooling of studies, see Details.</p>
</td></tr>
<tr><td><code id="metarate_+3A_incr">incr</code></td>
<td>
<p>A numeric which is added to the event number of studies
with zero events, i.e., studies with an incidence rate of 0.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.incr">method.incr</code></td>
<td>
<p>A character string indicating which continuity
correction method should be used (<code>"only0"</code>,
<code>"if0all"</code>, or <code>"all"</code>), see Details.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.ci">method.ci</code></td>
<td>
<p>A character string indicating whether to use
approximate normal (&quot;NAsm&quot;) or exact Poisson (&quot;Poisson&quot;)
confidence limits.</p>
</td></tr>
<tr><td><code id="metarate_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="metarate_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metarate_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="metarate_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="metarate_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="metarate_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="metarate_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="metarate_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="metarate_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="metarate_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
<tr><td><code id="metarate_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test is to
be used. Either <code>"Begg"</code>, <code>"Egger"</code>, or
<code>"Thompson"</code>, can be abbreviated. See function
<code><a href="#topic+metabias">metabias</a></code>.</p>
</td></tr>
<tr><td><code id="metarate_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results for
transformed rates (argument <code>sm != "IR"</code>) should be back
transformed in printouts and plots. If TRUE (default), results
will be presented as incidence rates; otherwise transformed rates
will be shown.</p>
</td></tr>
<tr><td><code id="metarate_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
rates.</p>
</td></tr>
<tr><td><code id="metarate_+3A_irunit">irunit</code></td>
<td>
<p>A character string specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="metarate_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="metarate_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="metarate_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="metarate_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="metarate_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="metarate_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="metarate_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="metarate_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="metarate_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="metarate_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="metarate_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="metarate_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="metarate_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="metarate_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="metarate_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="metarate_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="metarate_+3A_hakn">hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'method.random.ci').</p>
</td></tr>
<tr><td><code id="metarate_+3A_adhoc.hakn">adhoc.hakn</code></td>
<td>
<p>Deprecated argument (replaced by 'adhoc.hakn.ci').</p>
</td></tr>
<tr><td><code id="metarate_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="metarate_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether the addition of
<code>incr</code> to studies with zero events should result in a
warning.</p>
</td></tr>
<tr><td><code id="metarate_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="metarate_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> or
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="metarate_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function and to catch deprecated
arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides methods for common effect and random effects
meta-analysis of single incidence rates to calculate an overall
rate. Note, you should use R function <code><a href="#topic+metainc">metainc</a></code> to
compare incidence rates of pairwise comparisons instead of using
<code>metarate</code> for each treatment arm separately which will break
randomisation in randomised controlled trials.
</p>
<p>The following transformations of incidence rates are implemented to
calculate an overall rate:
</p>

<ul>
<li><p> Log transformation (<code>sm = "IRLN"</code>, default)
</p>
</li>
<li><p> Square root transformation (<code>sm = "IRS"</code>)
</p>
</li>
<li><p> Freeman-Tukey Double arcsine transformation (<code>sm =
  "IRFT"</code>)
</p>
</li>
<li><p> No transformation (<code>sm = "IR"</code>)
</p>
</li></ul>

<p>List elements <code>TE</code>, <code>TE.common</code>, <code>TE.random</code>, etc.,
contain the transformed incidence rates. In printouts and plots
these values are back transformed if argument <code>backtransf =
TRUE</code> (default).
</p>
<p>By default (argument <code>method = "Inverse"</code>), the inverse
variance method (Borenstein et al., 2010) is used for pooling by
calling <code><a href="#topic+metagen">metagen</a></code> internally. A random intercept
Poisson regression model (Stijnen et al., 2010) can be utilised
instead with argument <code>method = "GLMM"</code> which calls the
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function from R package
<b>metafor</b>.
</p>
<p>A three-level random effects meta-analysis model (Van den Noortgate
et al., 2013) is utilized if argument <code>cluster</code> is used and at
least one cluster provides more than one estimate. Internally,
<code><a href="metafor.html#topic+rma.mv">rma.mv</a></code> is called to conduct the analysis and
<code><a href="metafor.html#topic+weights.rma.mv">weights.rma.mv</a></code> with argument <code>type =
"rowsum"</code> is used to calculate random effects weights.
</p>
<p>Default settings are utilised for several arguments (assignments
using <code><a href="#topic+gs">gs</a></code> function). These defaults can be changed for
the current R session using the <code><a href="#topic+settings.meta">settings.meta</a></code>
function.
</p>
<p>Furthermore, R function <code><a href="#topic+update.meta">update.meta</a></code> can be used to
rerun a meta-analysis with different settings.
</p>


<h4>Continuity correction</h4>

<p>Three approaches are available to apply a continuity correction:
</p>

<ul>
<li><p> Only studies with a zero cell count (<code>method.incr =
  "only0"</code>)
</p>
</li>
<li><p> All studies if at least one study has a zero cell count
(<code>method.incr = "if0all"</code>)
</p>
</li>
<li><p> All studies irrespective of zero cell counts
(<code>method.incr = "all"</code>)
</p>
</li></ul>

<p>If the summary measure (argument <code>sm</code>) is equal to &quot;IR&quot; or
&quot;IRLN&quot;, the continuity correction is applied if a study has zero
events, i.e., an incidence rate of 0.
</p>
<p>By default, 0.5 is used as continuity correction (argument
<code>incr</code>). This continuity correction is used both to calculate
individual study results with confidence limits and to conduct
meta-analysis based on the inverse variance method.
</p>
<p>For the Freeman-Tukey (Freeman &amp; Tukey, 1950) and square root
transformation as well as GLMMs no continuity correction is used.
</p>



<h4>Subgroup analysis</h4>

<p>Argument <code>subgroup</code> can be used to conduct subgroup analysis for
a categorical covariate. The <code><a href="#topic+metareg">metareg</a></code> function can be
used instead for more than one categorical covariate or continuous
covariates.
</p>



<h4>Specify the null hypothesis of test for an overall effect</h4>

<p>Argument <code>null.effect</code> can be used to specify the rate used
under the null hypothesis in a test for an overall effect.
</p>
<p>By default (<code>null.effect = NA</code>), no hypothesis test is
conducted as it is unclear which value is a sensible choice for the
data at hand.  An overall rate of 2, for example, could be tested
by setting argument <code>null.effect = 2</code>.
</p>
<p>Note, all tests for an overall effect are two-sided with the
alternative hypothesis that the effect is unequal to
<code>null.effect</code>.
</p>



<h4>Exclusion of studies from meta-analysis</h4>

<p>Arguments <code>subset</code> and <code>exclude</code> can be used to exclude
studies from the meta-analysis. Studies are removed completely from
the meta-analysis using argument <code>subset</code>, while excluded
studies are shown in printouts and forest plots using argument
<code>exclude</code> (see Examples in <code><a href="#topic+metagen">metagen</a></code>).
Meta-analysis results are the same for both arguments.
</p>



<h4>Presentation of meta-analysis results</h4>

<p>Internally, both common effect and random effects models are
calculated regardless of values choosen for arguments
<code>common</code> and <code>random</code>. Accordingly, the estimate
for the random effects model can be extracted from component
<code>TE.random</code> of an object of class <code>"meta"</code> even if
argument <code>random = FALSE</code>. However, all functions in R
package <b>meta</b> will adequately consider the values for
<code>common</code> and <code>random</code>. E.g. function
<code><a href="#topic+print.meta">print.meta</a></code> will not print results for the random
effects model if <code>random = FALSE</code>.
</p>
<p>Argument <code>irscale</code> can be used to rescale rates, e.g.
<code>irscale = 1000</code> means that rates are expressed as events per
1000 time units, e.g. person-years. This is useful in situations
with (very) low rates. Argument <code>irunit</code> can be used to
specify the time unit used in individual studies (default:
&quot;person-years&quot;). This information is printed in summaries and
forest plots if argument <code>irscale</code> is not equal to 1.
</p>
<p>A prediction interval will only be shown if <code>prediction =
TRUE</code>.
</p>



<h3>Value</h3>

<p>An object of class <code>c("metarate", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Borenstein M, Hedges LV, Higgins JP, Rothstein HR (2010):
A basic introduction to fixed-effect and random-effects models for
meta-analysis.
<em>Research Synthesis Methods</em>,
<b>1</b>, 97&ndash;111
</p>
<p>Freeman MF &amp; Tukey JW (1950):
Transformations related to the angular and the square root.
<em>Annals of Mathematical Statistics</em>,
<b>21</b>, 607&ndash;11
</p>
<p>Stijnen T, Hamza TH, Ozdemir P (2010):
Random effects meta-analysis of event outcome in the framework of
the generalized linear mixed model with applications in sparse
data.
<em>Statistics in Medicine</em>,
<b>29</b>, 3046&ndash;67
</p>
<p>Van den Noortgate W, López-López JA, Marín-Martínez F, Sánchez-Meca J (2013):
Three-level meta-analysis of dependent effect sizes.
<em>Behavior Research Methods</em>,
<b>45</b>, 576&ndash;94
</p>
<p>Viechtbauer W (2010):
Conducting Meta-Analyses in R with the Metafor Package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>


<h3>See Also</h3>

<p><code><a href="#topic+meta-package">meta-package</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metacont">metacont</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Apply various meta-analysis methods to estimate incidence rates
#
m1 &lt;- metarate(4:1, c(10, 20, 30, 40))
m2 &lt;- update(m1, sm = "IR")
m3 &lt;- update(m1, sm = "IRS")
m4 &lt;- update(m1, sm = "IRFT")
#
m1
m2
m3
m4
#
forest(m1)
forest(m1, irscale = 100)
forest(m1, irscale = 100, irunit = "person-days")
forest(m1, backtransf = FALSE)
## Not run: 
forest(m2)
forest(m3)
forest(m4)

## End(Not run)

m5 &lt;- metarate(40:37, c(100, 200, 300, 400), sm = "IRFT")
m5

</code></pre>

<hr>
<h2 id='metareg.meta'>Meta-regression</h2><span id='topic+metareg.meta'></span><span id='topic+metareg'></span><span id='topic+metareg.default'></span>

<h3>Description</h3>

<p>Meta-regression for objects of class <code>meta</code>. This is a wrapper
function for the R function <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> in the R
package <b>metafor</b> (Viechtbauer 2010).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
metareg(
  x,
  formula,
  method.tau = x$method.tau,
  hakn = x$method.random.ci == "HK",
  level.ma = x$level.ma,
  intercept = TRUE,
  ...
)

metareg(x, ...)

## Default S3 method:
metareg(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="metareg.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="metareg.meta_+3A_formula">formula</code></td>
<td>
<p>Either a character string or a formula object.</p>
</td></tr>
<tr><td><code id="metareg.meta_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance tau-squared. Either
<code>"FE"</code>, <code>"DL"</code>, <code>"REML"</code>, <code>"ML"</code>,
<code>"HS"</code>, <code>"SJ"</code>, <code>"HE"</code>, or <code>"EB"</code>, can be
abbreviated.</p>
</td></tr>
<tr><td><code id="metareg.meta_+3A_hakn">hakn</code></td>
<td>
<p>A logical indicating whether the method by Hartung and
Knapp should be used to adjust test statistics and confidence
intervals.</p>
</td></tr>
<tr><td><code id="metareg.meta_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for parameter estimates in the meta-regression model.</p>
</td></tr>
<tr><td><code id="metareg.meta_+3A_intercept">intercept</code></td>
<td>
<p>A logical indicating whether an intercept should
be included in the meta-regression model.</p>
</td></tr>
<tr><td><code id="metareg.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments passed to R function
<code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This R function is a wrapper function for R function
<code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> in the R package <b>metafor</b>
(Viechtbauer 2010).
</p>
<p>Note, results are not back-transformed in printouts of
meta-analyses using summary measures with transformations, e.g.,
log risk ratios are printed instead of the risk ratio if argument
<code>sm = "RR"</code> and logit transformed proportions are printed if
argument <code>sm = "PLOGIT"</code>.
</p>
<p>Argument '...' can be used to pass additional arguments to R
function <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>. For example, argument
<code>control</code> to provide a list of control values for the
iterative estimation algorithm. See help page of R function
<code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> for more details.
</p>


<h3>Value</h3>

<p>An object of class <code>c("metareg", "rma.uni", "rma")</code>. Please
look at the help page of R function <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code>
for more details on the output from this function.
</p>
<p>In addition, a list <code>.meta</code> is added to the output containing
the following components:
</p>
<table>
<tr><td><code>x</code>, <code>formula</code>, <code>method.tau</code>, <code>hakn</code>, <code>level.ma</code>, <code>intercept</code></td>
<td>
<p>As
defined above.</p>
</td></tr>
<tr><td><code>dots</code></td>
<td>
<p>Information provided in argument '...'.</p>
</td></tr>
<tr><td><code>call</code></td>
<td>
<p>Function call.</p>
</td></tr>
<tr><td><code>version</code></td>
<td>
<p>Version of R package <b>meta</b> used to create
object.</p>
</td></tr>
<tr><td><code>version.metafor</code></td>
<td>
<p>Version of R package <b>metafor</b> used to
create object.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Viechtbauer W (2010):
Conducting Meta-Analyses in R with the Metafor Package.
<em>Journal of Statistical Software</em>,
<b>36</b>, 1&ndash;48
</p>


<h3>See Also</h3>

<p><code><a href="#topic+bubble">bubble</a></code>, <code><a href="#topic+summary.meta">summary.meta</a></code>,
<code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
# Add some (fictitious) grouping variables:
Fleiss1993cont$age &lt;- c(55, 65, 55, 65, 55)
Fleiss1993cont$region &lt;- c("Europe", "Europe", "Asia", "Asia", "Europe")

m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD")
## Not run: 
# Error due to wrong ordering of arguments (order has changed in
# R package meta, version 3.0-0)
#
try(metareg(~ region, m1))
try(metareg(~ region, data = m1))

# Warning as no information on covariate is available
#
metareg(m1)

## End(Not run)

# Do meta-regression for covariate region
#
mu2 &lt;- update(m1, subgroup = region, tau.common = TRUE, common = FALSE)
metareg(mu2)

# Same result for
# - tau-squared
# - test of heterogeneity
# - test for subgroup differences
# (as argument 'tau.common' was used to create mu2)
#
mu2
metareg(mu2, intercept = FALSE)
metareg(m1, region)

# Different result for
# - tau-squared
# - test of heterogeneity
# - test for subgroup differences
# (as argument 'tau.common' is - by default - FALSE)
#
mu1 &lt;- update(m1, subgroup = region)
mu1

# Generate bubble plot
#
bubble(metareg(mu2))

# Do meta-regression with two covariates
#
metareg(mu1, region + age)

# Do same meta-regressions using formula notation
#
metareg(m1, ~ region)
metareg(mu1, ~ region + age)

# Do meta-regression using REML method and print intermediate
# results for iterative estimation algorithm; furthermore print
# results with three digits.
#
metareg(mu1, region, method.tau = "REML",
  control = list(verbose = TRUE), digits = 3)

# Use Hartung-Knapp method
#
mu3 &lt;- update(mu2, method.random.ci = "HK")
mu3
metareg(mu3, intercept = FALSE)

</code></pre>

<hr>
<h2 id='nnt'>Calculate the number needed to treat (NNT)</h2><span id='topic+nnt'></span><span id='topic+nnt.default'></span><span id='topic+nnt.meta'></span><span id='topic+print.nnt.meta'></span>

<h3>Description</h3>

<p>Calculate the number needed to treat (NNT) from estimated risk
difference, risk ratio, or odds ratio, and a baseline risk.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nnt(x, ...)

## S3 method for class 'meta'
nnt(x, p.c, common = x$common, random = x$random, small.values = "good", ...)

## Default S3 method:
nnt(x, p.c, sm, lower, upper, small.values = "good", ...)

## S3 method for class 'nnt.meta'
print(
  x,
  common = x$common,
  random = x$random,
  digits = gs("digits"),
  digits.prop = gs("digits.prop"),
  big.mark = gs("big.mark"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="nnt_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>, or estimated treatment
effect, i.e., risk difference(s), risk ratio(s), or odds
ratio(s).</p>
</td></tr>
<tr><td><code id="nnt_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
<tr><td><code id="nnt_+3A_p.c">p.c</code></td>
<td>
<p>Baseline risk (control group event probability).</p>
</td></tr>
<tr><td><code id="nnt_+3A_common">common</code></td>
<td>
<p>A logical indicating whether NNTs should be
calculated based on common effect estimate.</p>
</td></tr>
<tr><td><code id="nnt_+3A_random">random</code></td>
<td>
<p>A logical indicating whether NNTs should be
calculated based on random effects estimate.</p>
</td></tr>
<tr><td><code id="nnt_+3A_small.values">small.values</code></td>
<td>
<p>A character string specifying whether small
treatment effects indicate a beneficial (<code>"good"</code>) or
harmful (<code>"bad"</code>) effect, can be abbreviated.</p>
</td></tr>
<tr><td><code id="nnt_+3A_sm">sm</code></td>
<td>
<p>Summary measure.</p>
</td></tr>
<tr><td><code id="nnt_+3A_lower">lower</code></td>
<td>
<p>Lower confidence interval limit.</p>
</td></tr>
<tr><td><code id="nnt_+3A_upper">upper</code></td>
<td>
<p>Upper confidence interval limit.</p>
</td></tr>
<tr><td><code id="nnt_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="nnt_+3A_digits.prop">digits.prop</code></td>
<td>
<p>Minimal number of significant digits for
proportions, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="nnt_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The number needed to treat (NNT) is the estimated number of
patients who need to be treated with a new treatment instead of a
standard for one additional patient to benefit (Laupacis et al.,
1988; Cook &amp; Sackett, 1995). This definition of the NNT implies
that the new treatment is more beneficial than the standard. If the
new treatment is indeed less beneficial than the standard, the NNT
gives the number of patients treated with the new treatment to
observe an additional harmful event. Accordingly, the abbreviations
NNTB and NNTH can be used to distinguish between beneficial and
harmful NNTs (Altman, 1998).
</p>
<p>NNTs can be easily computed from an estimated risk difference (RD),
risk ratio (RR), or odds ratio (OR) and a given baseline risk
(Higgins et al., 2022, section 15.4.4). It is also possible to
calculate NNTs from hazard ratios (HR) (Altman &amp; Andersen,
1999). Accordlingly, NNTs can be calculated for meta-analyses
generated with <code><a href="#topic+metabin">metabin</a></code> or <code><a href="#topic+metagen">metagen</a></code> if
argument <code>sm</code> was equal to <code>"RD"</code>, <code>"RR"</code>,
<code>"OR"</code>, or <code>"HR"</code>. It is also possible to provide only
estimated treatment effects and baseline risks (see Examples).
</p>
<p>The baseline risk can be specified using argument <code>p.c</code>. If
this argument is missing, the minimum, mean, and maximum of the
control event probabilities in the meta-analysis are used for
<code><a href="#topic+metabin">metabin</a></code> and control event probabilities of 0.1, 0.2,
..., 0.9 are used for <code><a href="#topic+metagen">metagen</a></code>.
</p>
<p>Argument <code>small.values</code> can be used to specify whether small
treatment effects indicate a beneficial (<code>"good"</code>) or harmful
(<code>"bad"</code>) effect. For <code>small.values = "small"</code>, odds and
risk ratios below 1 and risk differences below 0 indicate that the
new treatment is beneficial. For <code>small.values = "bad"</code>, odds
and risk ratios above 1 and risk differences above 0 indicate that
the new treatment is beneficial.
</p>


<h4>Interpretation of (positive and negative) NNTs</h4>

<p>A positive value for the estimated NNT indicates that the new
treatment is beneficial, i.e., the NNT is actually an NNTB. On the
other hand, a negative value for the estimated NNT indicates that
the new treatment is harmful, i.e., the NNT is actually an NNTH.
</p>
<p>The minimal value for the NNTB is 1. In this extreme case the new
treatment is 100% effective and the standard treatment is 0%
effective, i.e., only one patient has to be treated with the new
treatment for one additional patient to benefit. The NNTB increases
with decreasing difference between the two risks. If both risks are
equal, the NNTB is infinite.
</p>
<p>The other extreme is an NNT of -1 if the new treatment is 0%
effective and the standard is 100% effective. Here, one additional
harmful event is observed for each patient treated with the new
treatment. The NNT approaches minus infinity if the difference
between the two risks decreases to 0. Finally, an NNT of -1
translates to an NNTH of 1 with possible values from 1 to infinity.
</p>



<h4>Confidence interval for the NNT</h4>

<p>Confidence limits for an NNT are derived from the lower and upper
confidence limits of the summary measure using the same formulae as
for the NNT (Higgins et al., 2022, section 15.4.4).
</p>
<p>A peculiar problem arises if the confidence interval for the
summary measure includes the null effect (i.e., RR = 1, OR = 1, HR
= 1, or RD = 0). In this case the confidence interval for the NNT
contains both NNTB and NNTH values and it seemingly does not
include the estimated NNT.
</p>
<p>As described above, a positive NNT value corresponds to an NNTB and
the absolute value of a negative NNT is equal to an
NNTH. Accordingly, a confidence interval for the NNT from 20 to -5
translates to NNTB values between 20 and infinity and NNTH values
between 5 and infinity (Altman, 1998).
</p>



<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Altman DG (1998):
Confidence intervals for the number needed to treat.
<em>British Medical Journal</em>,
<b>317</b>, 1309&ndash;12
</p>
<p>Altman DG, Andersen PK (1999):
Calculating the Number Needed to Treat for Trials Where the Outcome
Is Time to an Event
<em>British Medical Journal</em>,
<b>319</b>, 1492&ndash;95
</p>
<p>Cook RJ, Sackett DL (1995):
The Number Needed to Treat: A Clinically Useful Measure of
Treatment Effect.
<em>British Medical Journal</em>,
<b>310</b>, 452&ndash;54
</p>
<p>Higgins JPT, Thomas J, Chandler J, Cumpston M, Li T, Page MJ, Welch
VA (editors) (2022):
<em>Cochrane Handbook for Systematic Reviews of Interventions
Version 6.3 (updated February 2022)</em>.
Available from www.training.cochrane.org/handbook
</p>
<p>Laupacis A, Sackett DL, Roberts RS (1988):
An Assessment of Clinically Useful Measures of the Consequences of
Treatment.
<em>New England Journal of Medicine</em>,
<b>318</b>, 1728&ndash;33
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Calculate NNT for RD = -0.12 and -0.22
# (Cochrane Handbook, version 6.3, subsection 15.4.4.1)
nnt(c(-0.12, -0.22), sm = "RD")

# Calculate NNT for RR = 0.92 and baseline risk p.c = 0.3
# (Cochrane Handbook, version 6.3, subsection 15.4.4.2)
nnt(0.92, p.c = 0.3, sm = "RR")

# Calculate NNT for OR = 0.73 and baseline risk p.c = 0.3
# (Cochrane Handbook, version 6.3, subsection 15.4.4.3)
nnt(0.73, p.c = 0.3, sm = "OR")

# Use Mantel-Haenszel odds ratio to calculate NNTs
data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont, data = Olkin1995,
              random = FALSE)
nnt(m1)

</code></pre>

<hr>
<h2 id='Olkin1995'>Thrombolytic Therapy after Acute Myocardial Infarction</h2><span id='topic+Olkin1995'></span><span id='topic+Olkin95'></span>

<h3>Description</h3>

<p>Meta-analysis on Thrombolytic Therapy after Acute Myocardial Infarction
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>author</em></b></td><td style="text-align: left;"> first author </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>year</em></b></td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>ev.exp</em></b></td><td style="text-align: left;"> number of events in experimental group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.exp</em></b></td><td style="text-align: left;"> number of observations in experimental group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>ev.cont</em></b></td><td style="text-align: left;"> number of events in control group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.cont</em></b></td><td style="text-align: left;"> number of observations in control group
</td>
</tr>

</table>



<h3>Source</h3>

<p>Olkin I (1995):
Statistical and theoretical considerations in meta-analysis.
<em>Journal of Clinical Epidemiology</em>,
<b>48</b>, 133&ndash;46
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)
metabin(ev.exp, n.exp, ev.cont, n.cont, data = Olkin1995)
</code></pre>

<hr>
<h2 id='or2smd'>Conversion from log odds ratio to standardised mean difference</h2><span id='topic+or2smd'></span>

<h3>Description</h3>

<p>Conversion from log odds ratio to standardised mean difference
using method by Hasselblad &amp; Hedges (1995) or Cox (1970).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>or2smd(
  lnOR,
  selnOR,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  method = "HH",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="or2smd_+3A_lnor">lnOR</code></td>
<td>
<p>Log odds ratio(s) or meta-analysis object.</p>
</td></tr>
<tr><td><code id="or2smd_+3A_selnor">selnOR</code></td>
<td>
<p>Standard error(s) of log odds ratio(s) (ignored if
argument <code>lnOR</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="or2smd_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels (ignored if
argument <code>lnOR</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="or2smd_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study information
(ignored if argument <code>lnOR</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="or2smd_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used (ignored if argument <code>lnOR</code> is a meta-analysis
object).</p>
</td></tr>
<tr><td><code id="or2smd_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots (ignored if argument <code>lnOR</code> is a meta-analysis
object).</p>
</td></tr>
<tr><td><code id="or2smd_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is used to
convert log odds ratios to standardised mean differences. Either
<code>"HH"</code> or <code>"CS"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="or2smd_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="#topic+metagen">metagen</a></code> (ignored if argument <code>lnOR</code> is a
meta-analysis object).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the following methods for the conversion
from log odds ratios to standardised mean difference:
</p>

<ul>
<li><p> Hasselblad &amp; Hedges (1995) assuming logistic distributions
(<code>method == "HH"</code>)
</p>
</li>
<li><p> Cox (1970) and Cox &amp; Snell (1989) assuming normal
distributions (<code>method == "CS"</code>)
</p>
</li></ul>

<p>Internally, <code><a href="#topic+metagen">metagen</a></code> is used to conduct a
meta-analysis with the standardised mean difference as summary
measure.
</p>
<p>Argument <code>lnOR</code> can be either a vector of log odds ratios or a
meta-analysis object created with <code><a href="#topic+metabin">metabin</a></code> or
<code><a href="#topic+metagen">metagen</a></code> and the odds ratio as summary measure.
</p>
<p>Argument <code>selnOR</code> is mandatory if argument <code>lnOR</code> is a
vector and ignored otherwise. Additional arguments in <code>...</code>
are only passed on to <code><a href="#topic+metagen">metagen</a></code> if argument <code>lnOR</code>
is a vector.
</p>


<h3>Value</h3>

<p>An object of class <code>c("metagen", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Borenstein M, Hedges LV, Higgins JPT, Rothstein HR (2009):
<em>Introduction to Meta-Analysis</em>.
Chichester: Wiley
</p>
<p>Cox DR (1970):
<em>Analysis of Binary Data</em>.
London: Chapman and Hall / CRC
</p>
<p>Cox DR, Snell EJ (1989):
<em>Analysis of Binary Data</em> (2nd edition).
London: Chapman and Hall / CRC
</p>
<p>Hasselblad V, Hedges LV (1995):
Meta-analysis of screening and diagnostic tests.
<em>Psychological Bulletin</em>,
<b>117</b>, 167&ndash;78
</p>


<h3>See Also</h3>

<p><code><a href="#topic+smd2or">smd2or</a></code>, <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metacont">metacont</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example from Borenstein et al. (2009), Chapter 7
#
mb &lt;- or2smd(0.9069, sqrt(0.0676))
# TE = standardised mean difference (SMD); seTE = standard error of SMD
data.frame(SMD = round(mb$TE, 4), varSMD = round(mb$seTE^2, 4))

# Use dataset from Fleiss (1993)
#
data(Fleiss1993bin)
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac,
  data = Fleiss1993bin, studlab = paste(study, year),
  sm = "OR", random = FALSE)
or2smd(m1)

</code></pre>

<hr>
<h2 id='Pagliaro1992'>Meta-analysis on Prevention of First Bleeding in Cirrhosis</h2><span id='topic+Pagliaro1992'></span>

<h3>Description</h3>

<p>Meta-analysis on Prevention of First Bleeding in Cirrhosis
comparing beta-blocker or sclerotherapy with placebo.
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>id</em></b></td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>treat.exp</em></b></td><td style="text-align: left;"> treatment in experimental group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>logOR</em></b></td><td style="text-align: left;"> log odds ratio
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>selogOR</em></b></td><td style="text-align: left;"> standard error of log odds ratio
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>bleed.exp</em></b></td><td style="text-align: left;"> number of bleedings in experimental group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.cont</em></b></td><td style="text-align: left;"> number of observations in experimental group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>bleed.plac</em></b></td><td style="text-align: left;"> number of bleedings in placebo group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.plac</em></b></td><td style="text-align: left;"> number of observations in placebo group
</td>
</tr>

</table>



<h3>Source</h3>

<p>Pagliaro L, D’Amico G et al. (1992):
Prevention of first bleeding in cirrhosis.
<em>Annals in Internal Medicine</em>,
<b>117</b>, 59&ndash;70
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Pagliaro1992)
sclero &lt;- subset(Pagliaro1992, treat.exp == "Sclerotherapy")

m &lt;- metagen(logOR, selogOR, data = sclero, sm = "OR")
m

# Thompson &amp; Sharp (1999), Table IV, method (2)
metabias(m, method = "Egger")

# Thompson &amp; Sharp (1999), Table IV, method (3a)
metabias(m, method = "Thompson")

# Thompson &amp; Sharp (1999), Table IV, method (3b)
update(m, method.tau = "ML")
metabias(update(m, method.tau = "ML"), method = "Thompson")

</code></pre>

<hr>
<h2 id='print.meta'>Print meta-analysis results</h2><span id='topic+print.meta'></span><span id='topic+cilayout'></span>

<h3>Description</h3>

<p>Print method for objects of class <code>meta</code>.
</p>
<p>R function cilayout can be utilised to change the layout to print
confidence intervals (both in printout from print.meta and
print.summary.meta function as well as in forest plots). The
default layout is &quot;[lower; upper]&quot;. Another popular layout is
&quot;(lower - upper)&quot; which is used throughout an R session by using R
command <code>cilayout("(", " - ")</code>.
</p>
<p>Argument <code>pscale</code> can be used to rescale single proportions or
risk differences, e.g. <code>pscale = 1000</code> means that proportions
are expressed as events per 1000 observations. This is useful in
situations with (very) low event probabilities.
</p>
<p>Argument <code>irscale</code> can be used to rescale single rates or rate
differences, e.g. <code>irscale = 1000</code> means that rates are
expressed as events per 1000 time units, e.g. person-years. This is
useful in situations with (very) low rates. Argument <code>irunit</code>
can be used to specify the time unit used in individual studies
(default: &quot;person-years&quot;). This information is printed in summaries
and forest plots if argument <code>irscale</code> is not equal to 1.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
print(
  x,
  common = x$common,
  random = x$random,
  prediction = x$prediction,
  overall = x$overall,
  overall.hetstat = x$overall.hetstat,
  test.subgroup = x$test.subgroup,
  test.subgroup.common = test.subgroup &amp; common,
  test.subgroup.random = test.subgroup &amp; random,
  prediction.subgroup = x$prediction.subgroup,
  backtransf = x$backtransf,
  pscale = x$pscale,
  irscale = x$irscale,
  irunit = x$irunit,
  subgroup.name = x$subgroup.name,
  print.subgroup.name = x$print.subgroup.name,
  sep.subgroup = x$sep.subgroup,
  nchar.subgroup = 35,
  sort.overall = NULL,
  sort.tau = NULL,
  sort.het = NULL,
  sort.Q = NULL,
  header = TRUE,
  print.CMH = x$print.CMH,
  digits = gs("digits"),
  digits.stat = gs("digits.stat"),
  digits.pval = max(gs("digits.pval"), 2),
  digits.pval.Q = max(gs("digits.pval.Q"), 2),
  digits.Q = gs("digits.Q"),
  digits.tau2 = gs("digits.tau2"),
  digits.tau = gs("digits.tau"),
  digits.H = gs("digits.H"),
  digits.I2 = gs("digits.I2"),
  scientific.pval = gs("scientific.pval"),
  big.mark = gs("big.mark"),
  zero.pval = gs("zero.pval"),
  JAMA.pval = gs("JAMA.pval"),
  digits.df = gs("digits.df"),
  print.tau2 = gs("print.tau2"),
  print.tau2.ci = gs("print.tau2.ci"),
  print.tau = gs("print.tau"),
  print.tau.ci = gs("print.tau.ci"),
  print.I2 = gs("print.I2"),
  print.H = gs("print.H"),
  print.Rb = gs("print.Rb"),
  text.tau2 = gs("text.tau2"),
  text.tau = gs("text.tau"),
  text.I2 = gs("text.I2"),
  text.Rb = gs("text.Rb"),
  print.Q = gs("print.Q"),
  details.methods = gs("details"),
  warn.backtransf = FALSE,
  func.backtransf = x$func.backtransf,
  warn.deprecated = gs("warn.deprecated"),
  ...
)

cilayout(
  bracket = gs("CIbracket"),
  separator = gs("CIseparator"),
  lower.blank = gs("CIlower.blank"),
  upper.blank = gs("CIupper.blank")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether results for common
effect meta-analysis should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether results for random
effects meta-analysis should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_test.subgroup.common">test.subgroup.common</code></td>
<td>
<p>A logical value indicating whether to
print results of test for subgroup differences (based on common
effect model).</p>
</td></tr>
<tr><td><code id="print.meta_+3A_test.subgroup.random">test.subgroup.random</code></td>
<td>
<p>A logical value indicating whether to
print results of test for subgroup differences (based on random
effects model).</p>
</td></tr>
<tr><td><code id="print.meta_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A single logical or logical vector
indicating whether / which prediction intervals should be printed
for subgroups.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether printed results
should be back transformed. If <code>backtransf=TRUE</code>, results
for <code>sm="OR"</code> are printed as odds ratios rather than log
odds ratios and results for <code>sm="ZCOR"</code> are printed as
correlations rather than Fisher's z transformed correlations, for
example.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
single event probabilities or risk differences, i.e. if argument
<code>sm</code> is equal to <code>"PLOGIT"</code>, <code>"PLN"</code>,
<code>"PRAW"</code>, <code>"PAS"</code>, <code>"PFT"</code>, or <code>"RD"</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single incidence rates or incidence rate differences, i.e. if
argument <code>sm</code> is equal to <code>"IR"</code>, <code>"IRLN"</code>,
<code>"IRS"</code>, <code>"IRFT"</code>, or <code>"IRD"</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_irunit">irunit</code></td>
<td>
<p>A character specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
grouping variable.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the grouping variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between label and levels of grouping variable.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_nchar.subgroup">nchar.subgroup</code></td>
<td>
<p>A numeric specifying the number of characters
to print from subgroup labels.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_sort.overall">sort.overall</code></td>
<td>
<p>An optional vector used to sort meta-analysis
results.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_sort.tau">sort.tau</code></td>
<td>
<p>An optional vector used to sort estimators of the
between-study heterogeneity variance.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_sort.het">sort.het</code></td>
<td>
<p>An optional vector used to sort heterogeneity
statistics.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_sort.q">sort.Q</code></td>
<td>
<p>An optional vector used to sort test of
heterogeneity.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_header">header</code></td>
<td>
<p>A logical indicating whether information on title of
meta-analysis, comparison and outcome should be printed at the
beginning of the printout.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.cmh">print.CMH</code></td>
<td>
<p>A logical indicating whether result of the
Cochran-Mantel-Haenszel test for overall effect should be
printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.stat">digits.stat</code></td>
<td>
<p>Minimal number of significant digits for z- or
t-value of test for overall effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.pval">digits.pval</code></td>
<td>
<p>Minimal number of significant digits for p-value
of overall treatment effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.pval.q">digits.pval.Q</code></td>
<td>
<p>Minimal number of significant digits for
p-value of heterogeneity test, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.q">digits.Q</code></td>
<td>
<p>Minimal number of significant digits for
heterogeneity statistic Q, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.tau2">digits.tau2</code></td>
<td>
<p>Minimal number of significant digits for
between-study variance <code class="reqn">\tau^2</code>, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.tau">digits.tau</code></td>
<td>
<p>Minimal number of significant digits for
<code class="reqn">\tau</code>, the square root of the between-study variance
<code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.h">digits.H</code></td>
<td>
<p>Minimal number of significant digits for H
statistic, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.i2">digits.I2</code></td>
<td>
<p>Minimal number of significant digits for I-squared
and Rb statistic, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_scientific.pval">scientific.pval</code></td>
<td>
<p>A logical specifying whether p-values should
be printed in scientific notation, e.g., 1.2345e-01 instead of
0.12345.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_zero.pval">zero.pval</code></td>
<td>
<p>A logical specifying whether p-values should be
printed with a leading zero.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_jama.pval">JAMA.pval</code></td>
<td>
<p>A logical specifying whether p-values for test of
overall effect should be printed according to JAMA reporting
standards.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_digits.df">digits.df</code></td>
<td>
<p>Minimal number of significant digits for degrees
of freedom.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.tau2">print.tau2</code></td>
<td>
<p>A logical specifying whether between-study
variance <code class="reqn">\tau^2</code> should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.tau2.ci">print.tau2.ci</code></td>
<td>
<p>A logical value indicating whether to print
the confidence interval of <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.tau">print.tau</code></td>
<td>
<p>A logical specifying whether <code class="reqn">\tau</code>, the
square root of the between-study variance <code class="reqn">\tau^2</code>, should be
printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.tau.ci">print.tau.ci</code></td>
<td>
<p>A logical value indicating whether to print the
confidence interval of <code class="reqn">\tau</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.i2">print.I2</code></td>
<td>
<p>A logical specifying whether heterogeneity
statistic I<code class="reqn">^2</code> should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.h">print.H</code></td>
<td>
<p>A logical specifying whether heterogeneity statistic
H should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.rb">print.Rb</code></td>
<td>
<p>A logical specifying whether heterogeneity
statistic R<code class="reqn">_b</code> should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_text.tau2">text.tau2</code></td>
<td>
<p>Text printed to identify between-study variance
<code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_text.tau">text.tau</code></td>
<td>
<p>Text printed to identify <code class="reqn">\tau</code>, the square
root of the between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_text.i2">text.I2</code></td>
<td>
<p>Text printed to identify heterogeneity statistic
I<code class="reqn">^2</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_text.rb">text.Rb</code></td>
<td>
<p>Text printed to identify heterogeneity statistic
R<code class="reqn">_b</code>.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_print.q">print.Q</code></td>
<td>
<p>A logical value indicating whether to print the
results of the test of heterogeneity.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_details.methods">details.methods</code></td>
<td>
<p>A logical specifying whether details on
statistical methods should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_warn.backtransf">warn.backtransf</code></td>
<td>
<p>Deprecated argument (ignored).</p>
</td></tr>
<tr><td><code id="print.meta_+3A_func.backtransf">func.backtransf</code></td>
<td>
<p>A function used to back-transform results.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to <code>prmatrix</code>).</p>
</td></tr>
<tr><td><code id="print.meta_+3A_bracket">bracket</code></td>
<td>
<p>A character with bracket symbol to print lower
confidence interval: &quot;[&quot;, &quot;(&quot;, &quot;{&quot;, &quot;&quot;.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_separator">separator</code></td>
<td>
<p>A character string with information on separator
between lower and upper confidence interval.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_lower.blank">lower.blank</code></td>
<td>
<p>A logical indicating whether blanks between left
bracket and lower confidence limit should be printed.</p>
</td></tr>
<tr><td><code id="print.meta_+3A_upper.blank">upper.blank</code></td>
<td>
<p>A logical indicating whether blanks between
separator and upper confidence limit should be printed.</p>
</td></tr>
</table>

<hr>
<h2 id='print.rm5'>Cochrane review: summary of meta-analyses</h2><span id='topic+print.rm5'></span>

<h3>Description</h3>

<p>Calculate and print a summary of all meta-analyses in a
Cochrane review.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rm5'
print(x, comp.no, outcome.no, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.rm5_+3A_x">x</code></td>
<td>
<p>An object of class <code>rm5</code>.</p>
</td></tr>
<tr><td><code id="print.rm5_+3A_comp.no">comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code id="print.rm5_+3A_outcome.no">outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code id="print.rm5_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to <code>metacr</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to redo all or selected meta-analyses of
a Cochrane Review.
</p>
<p>Review Manager 5 (RevMan 5) was the software used for preparing and
maintaining Cochrane Reviews
(<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>).
In RevMan 5, subgroup analyses can be defined and data from a
Cochrane review can be imported to R using the function
<code>read.rm5</code>.
</p>
<p>The R function <code><a href="#topic+metacr">metacr</a></code> is called internally.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Higgins, J.P.T and S. Green (2011):
<em>Cochrane Handbook for Systematic Reviews of Interventions
Version 5.1.0 [Updated March 2011]</em>.
The Cochrane Library: http://www.cochrane-handbook.org
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.meta">summary.meta</a></code>, <code><a href="#topic+metacr">metacr</a></code>,
<code><a href="#topic+read.rm5">read.rm5</a></code>, <code><a href="#topic+metabias.rm5">metabias.rm5</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate export data file "Fleiss1993_CR.csv"
# in sub-directory of package "meta"
#
filename &lt;- system.file("extdata", "Fleiss1993_CR.csv", package = "meta")
Fleiss1993_CR &lt;- read.rm5(filename)

# Print results for all meta-analysis
#
Fleiss1993_CR

# Print results only for second outcome of first comparison
#
print(Fleiss1993_CR, comp.no = 1, outcome.no = 2)

</code></pre>

<hr>
<h2 id='print.summary.meta'>Print detailed meta-analysis results</h2><span id='topic+print.summary.meta'></span>

<h3>Description</h3>

<p>Print method for objects of class <code>summary.meta</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'summary.meta'
print(
  x,
  sortvar,
  common = x$x$common,
  random = x$x$random,
  details = FALSE,
  ma = TRUE,
  overall = x$overall &amp; ma,
  backtransf = x$backtransf,
  pscale = x$pscale,
  irscale = x$irscale,
  irunit = x$irunit,
  digits = gs("digits"),
  digits.se = gs("digits.se"),
  digits.stat = gs("digits.stat"),
  digits.pval = max(gs("digits.pval"), 2),
  digits.tau2 = gs("digits.tau2"),
  digits.tau = gs("digits.tau"),
  digits.I2 = gs("digits.I2"),
  digits.prop = gs("digits.prop"),
  digits.weight = gs("digits.weight"),
  scientific.pval = gs("scientific.pval"),
  zero.pval = gs("zero.pval"),
  JAMA.pval = gs("JAMA.pval"),
  big.mark = gs("big.mark"),
  print.tau2 = gs("print.tau2"),
  print.tau = gs("print.tau"),
  print.I2 = gs("print.I2"),
  text.tau2 = gs("text.tau2"),
  text.tau = gs("text.tau"),
  text.I2 = gs("text.I2"),
  truncate,
  text.truncate = "*** Output truncated ***",
  details.methods = TRUE,
  warn.backtransf = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.summary.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>summary.meta</code></p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_sortvar">sortvar</code></td>
<td>
<p>An optional vector used to sort the individual
studies (must be of same length as <code>x$TE</code>).</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether results of common effect
meta-analysis should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether results of random
effects meta-analysis should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_details">details</code></td>
<td>
<p>A logical indicating whether further details of
individual studies should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_ma">ma</code></td>
<td>
<p>A logical indicating whether the summary results of the
meta-analysis should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether printed results
should be back transformed. If <code>backtransf = TRUE</code>, results
for <code>sm = "OR"</code> are printed as odds ratios rather than log
odds ratios and results for <code>sm = "ZCOR"</code> are printed as
correlations rather than Fisher's z transformed correlations, for
example.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
single event probabilities or risk differences, i.e. if argument
<code>sm</code> is equal to <code>"PLOGIT"</code>, <code>"PLN"</code>,
<code>"PRAW"</code>, <code>"PAS"</code>, <code>"PFT"</code>, or <code>"RD"</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single incidence rates or incidence rate differences, i.e. if
argument <code>sm</code> is equal to <code>"IR"</code>, <code>"IRLN"</code>,
<code>"IRS"</code>, <code>"IRFT"</code>, or <code>"IRD"</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_irunit">irunit</code></td>
<td>
<p>A character specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits">digits</code></td>
<td>
<p>Minimal number of significant digits, see
<code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.se">digits.se</code></td>
<td>
<p>Minimal number of significant digits for standard
deviations and standard errors, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.stat">digits.stat</code></td>
<td>
<p>Minimal number of significant digits for z- or
t-value of test for effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.pval">digits.pval</code></td>
<td>
<p>Minimal number of significant digits for p-value
of test of treatment effect, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.tau2">digits.tau2</code></td>
<td>
<p>Minimal number of significant digits for
between-study variance, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.tau">digits.tau</code></td>
<td>
<p>Minimal number of significant digits for square
root of between-study variance, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.i2">digits.I2</code></td>
<td>
<p>Minimal number of significant digits for I-squared
and Rb statistic, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.prop">digits.prop</code></td>
<td>
<p>Minimal number of significant digits for
proportions, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_digits.weight">digits.weight</code></td>
<td>
<p>Minimal number of significant digits for
weights, see <code>print.default</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_scientific.pval">scientific.pval</code></td>
<td>
<p>A logical specifying whether p-values should
be printed in scientific notation, e.g., 1.2345e-01 instead of
0.12345.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_zero.pval">zero.pval</code></td>
<td>
<p>A logical specifying whether p-values should be
printed with a leading zero.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_jama.pval">JAMA.pval</code></td>
<td>
<p>A logical specifying whether p-values for test of
overall effect should be printed according to JAMA reporting
standards.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_big.mark">big.mark</code></td>
<td>
<p>A character used as thousands separator.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_print.tau2">print.tau2</code></td>
<td>
<p>A logical specifying whether between-study
variance <code class="reqn">\tau^2</code> should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_print.tau">print.tau</code></td>
<td>
<p>A logical specifying whether <code class="reqn">\tau</code>, the
square root of the between-study variance <code class="reqn">\tau^2</code>, should be
printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_print.i2">print.I2</code></td>
<td>
<p>A logical specifying whether heterogeneity
statistic I<code class="reqn">^2</code> should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_text.tau2">text.tau2</code></td>
<td>
<p>Text printed to identify between-study variance
<code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_text.tau">text.tau</code></td>
<td>
<p>Text printed to identify <code class="reqn">\tau</code>, the square
root of the between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_text.i2">text.I2</code></td>
<td>
<p>Text printed to identify heterogeneity statistic
I<code class="reqn">^2</code>.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_truncate">truncate</code></td>
<td>
<p>An optional vector used to truncate the printout of
results for individual studies (must be a logical vector of same
length as <code>x$TE</code> or contain numerical values).</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_text.truncate">text.truncate</code></td>
<td>
<p>A character string printed if study results
were truncated from the printout.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_details.methods">details.methods</code></td>
<td>
<p>A logical specifying whether details on
statistical methods should be printed.</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_warn.backtransf">warn.backtransf</code></td>
<td>
<p>Deprecated argument (ignored).</p>
</td></tr>
<tr><td><code id="print.summary.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to
<code><a href="#topic+print.meta">print.meta</a></code> called internally).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Print method for objects of class <code>summary.meta</code> giving
detailed information on the meta-analysis.
</p>
<p>Argument <code>pscale</code> can be used to rescale single proportions or
risk differences, e.g. <code>pscale = 1000</code> means that proportions
are expressed as events per 1000 observations. This is useful in
situations with (very) low event probabilities.
</p>
<p>Argument <code>irscale</code> can be used to rescale single rates or rate
differences, e.g. <code>irscale = 1000</code> means that rates are
expressed as events per 1000 time units, e.g. person-years. This is
useful in situations with (very) low rates. Argument <code>irunit</code>
can be used to specify the time unit used in individual studies
(default: &quot;person-years&quot;). This information is printed in summaries
and forest plots if argument <code>irscale</code> is not equal to 1.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Cooper H &amp; Hedges LV (1994), <em>The Handbook of
Research Synthesis</em>.  Newbury Park, CA: Russell Sage Foundation.
</p>
<p>Crippa A, Khudyakov P, Wang M, Orsini N, Spiegelman D (2016), A new measure
of between-studies heterogeneity in meta-analysis.  <em>Statistics in
Medicine</em>, <b>35</b>, 3661&ndash;75.
</p>
<p>Higgins JPT &amp; Thompson SG (2002), Quantifying heterogeneity in a
meta-analysis.  <em>Statistics in Medicine</em>, <b>21</b>, 1539&ndash;58.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.meta">summary.meta</a></code>, <code><a href="#topic+update.meta">update.meta</a></code>,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD", studlab = paste(study, year))
sm1 &lt;- summary(m1)
sm1

print(sm1, digits = 2)

## Not run: 
# Use unicode characters to print tau^2, tau, and I^2 
print(sm1,
  text.tau2 = "\u03c4\u00b2",
  text.tau = "\u03c4", text.I2 = "I\u00b2")

## End(Not run)

</code></pre>

<hr>
<h2 id='radial.meta'>Radial plot</h2><span id='topic+radial.meta'></span><span id='topic+radial'></span><span id='topic+radial.default'></span>

<h3>Description</h3>

<p>Draw a radial plot (also called Galbraith plot) which can be used
to assess bias in meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
radial(
  x,
  xlim = NULL,
  ylim = NULL,
  xlab = "Inverse of standard error",
  ylab = "Standardised treatment effect (z-score)",
  common = TRUE,
  axes = TRUE,
  pch = 1,
  text = NULL,
  cex = 1,
  col = NULL,
  level = NULL,
  warn.deprecated = gs("warn.deprecated"),
  fixed,
  ...
)

## Default S3 method:
radial(
  x,
  y,
  xlim = NULL,
  ylim = NULL,
  xlab = "Inverse of standard error",
  ylab = "Standardised treatment effect (z-score)",
  common = TRUE,
  axes = TRUE,
  pch = 1,
  text = NULL,
  cex = 1,
  col = NULL,
  level = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="radial.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>, or estimated treatment
effect in individual studies.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_xlim">xlim</code></td>
<td>
<p>The x limits (min, max) of the plot.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_ylim">ylim</code></td>
<td>
<p>The y limits (min, max) of the plot.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_xlab">xlab</code></td>
<td>
<p>A label for the x-axis.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_ylab">ylab</code></td>
<td>
<p>A label for the y-axis.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether the pooled common effect
estimate should be plotted.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_axes">axes</code></td>
<td>
<p>A logical indicating whether axes should be drawn on
the plot.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_pch">pch</code></td>
<td>
<p>The plotting symbol used for individual studies.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_text">text</code></td>
<td>
<p>A character vector specifying the text to be used
instead of plotting symbol.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_cex">cex</code></td>
<td>
<p>The magnification to be used for plotting symbol.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_col">col</code></td>
<td>
<p>A vector with colour of plotting symbols.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_level">level</code></td>
<td>
<p>The confidence level utilised in the plot.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_fixed">fixed</code></td>
<td>
<p>Deprecated argument (replaced by 'common').</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_...">...</code></td>
<td>
<p>Graphical arguments as in <code>par</code> may also be
passed as arguments.</p>
</td></tr>
<tr><td><code id="radial.meta_+3A_y">y</code></td>
<td>
<p>Standard error of estimated treatment effect.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A radial plot (Galbraith 1988a,b), also called Galbraith plot, is
drawn in the active graphics window. If <code>common</code> is TRUE, the
pooled estimate of the common effect model is plotted. If
<code>level</code> is not NULL, the corresponding confidence limits are
drawn.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Galbraith RF (1988a):
Graphical display of estimates having differing standard errors.
<em>Technometrics</em>,
<b>30</b>, 271&ndash;81
</p>
<p>Galbraith RF (1988b):
A note on graphical presentation of estimated odds ratios from
several clinical trials.
<em>Statistics in Medicine</em>,
<b>7</b>, 889&ndash;94
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabias">metabias</a></code>, <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+funnel">funnel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Olkin1995)
m1 &lt;- metabin(ev.exp, n.exp, ev.cont, n.cont,
  data = Olkin1995, subset = c(41, 47, 51, 59),
  studlab = paste(author, year), sm = "RR", method = "I")

# Radial plot
#
radial(m1, level = 0.95)

</code></pre>

<hr>
<h2 id='read.cdir'>Import data of Cochrane intervention review</h2><span id='topic+read.cdir'></span><span id='topic+print.cdir'></span>

<h3>Description</h3>

<p>Reads Cochrane data package (version 1) of a Cochrane intervention
review and creates a data frame from it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.cdir(
  file,
  title = "Cochrane Review of Interventions",
  exdir = tempdir(),
  numbers.in.labels = TRUE,
  rob = !missing(tool) | !missing(categories) | !missing(col) | !missing(symbols),
  tool = NULL,
  categories = NULL,
  col = NULL,
  symbols = NULL,
  keep.orig = FALSE,
  ...
)

## S3 method for class 'cdir'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.cdir_+3A_file">file</code></td>
<td>
<p>The name of a file to read data values from.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_title">title</code></td>
<td>
<p>Title of Cochrane review.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_exdir">exdir</code></td>
<td>
<p>The directory to extract files to (the equivalent of
‘unzip -d’). It will be created if necessary.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_numbers.in.labels">numbers.in.labels</code></td>
<td>
<p>A logical indicating whether comparison
number and outcome number should be printed at the beginning of
the comparison (argument <code>complab</code>) and outcome label
(argument <code>outclab</code>); this is the default in RevMan Web.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_rob">rob</code></td>
<td>
<p>A logical indicating whether risk of bias (RoB)
assessment should be considered in meta-analyses.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_tool">tool</code></td>
<td>
<p>Risk of bias (RoB) tool.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_categories">categories</code></td>
<td>
<p>Possible RoB categories.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_col">col</code></td>
<td>
<p>Colours for RoB categories.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_symbols">symbols</code></td>
<td>
<p>Corresponding symbols for RoB categories.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_keep.orig">keep.orig</code></td>
<td>
<p>A logical indicating whether to return the
original data files.</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to
<code><a href="utils.html#topic+unzip">unzip</a></code>)</p>
</td></tr>
<tr><td><code id="read.cdir_+3A_x">x</code></td>
<td>
<p>An object of class &quot;cdir&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>RevMan Web is the current software used for preparing and
maintaining Cochrane reviews
(<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>).
RevMan Web includes the ability to write systematic reviews of
interventions or diagnostic test accuracy reviews.
</p>
<p>This function provides the ability to read the Cochrane data
package from a Cochrane intervention review created with RevMan
Web. The ZIP-file is extracted with <code><a href="utils.html#topic+unzip">unzip</a></code>.
</p>
<p>Argument <code>title</code> can be used to overwrite the title of the
Cochrane review.
</p>
<p>Information on the risk of bias (RoB) assessment can be provided
with arguments <code>tool</code>, <code>categories</code>, <code>col</code> and
<code>symbols</code>. This is only useful if (i) all outcomes are based
on the same RoB categories and (ii) an overall RoB assessment has
not been done. If no overall RoB assessment was conducted, R
function <code><a href="#topic+metacr">metacr</a></code> can be used to provide the RoB
information for a single outcome. R function <code><a href="#topic+rob">rob</a></code> is
the most flexible way to add RoB information to a meta-analysis
object.
</p>


<h4>Creation of Cochrane data package</h4>

<p>Two possible ways exist to create the ZIP-file.
</p>
<p>In RevMan Web (<a href="https://revman.cochrane.org/">https://revman.cochrane.org/</a>), press the
&quot;Export&quot; button at the bottom of the <em>Default view</em>
website. After a couple of seconds, the data package will be shown
at the bottom of the <em>Default view</em> website under &quot;Downloads&quot;.
</p>
<p>In the Cochrane Library (<a href="https://www.cochranelibrary.com/">https://www.cochranelibrary.com/</a>),
press on &quot;Download statistical data&quot; in the Contents menu to
download an rm5-file. This file can be converted to a data package
in RevMan Web using <em>Help</em> - <em>Convert a RevMan 5 file</em>.
</p>



<h3>Value</h3>

<p>A list consisting of a data frame 'data' with the study data and
(if available) a data frame 'rob' with information on the risk of
bias assessment. If <code>keep.orig = TRUE</code>, an additional list
'orig' is returned containing elements 'settings', 'datarows',
'subgroup' and 'rob' (if available).
</p>
<p>The data frame 'data' contains the following variables:
</p>
<table>
<tr><td><code>comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code>outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code>group.no</code></td>
<td>
<p>Group number.</p>
</td></tr>
<tr><td><code>studlab</code></td>
<td>
<p>Study label.</p>
</td></tr>
<tr><td><code>year</code></td>
<td>
<p>Year of publication.</p>
</td></tr>
<tr><td><code>event.e</code></td>
<td>
<p>Number of events in experimental group.</p>
</td></tr>
<tr><td><code>n.e</code></td>
<td>
<p>Number of observations in experimental group.</p>
</td></tr>
<tr><td><code>event.c</code></td>
<td>
<p>Number of events in control group.</p>
</td></tr>
<tr><td><code>n.c</code></td>
<td>
<p>Number of observations in control group.</p>
</td></tr>
<tr><td><code>mean.e</code></td>
<td>
<p>Estimated mean in experimental group.</p>
</td></tr>
<tr><td><code>sd.e</code></td>
<td>
<p>Standard deviation in experimental group.</p>
</td></tr>
<tr><td><code>mean.c</code></td>
<td>
<p>Estimated mean in control group.</p>
</td></tr>
<tr><td><code>sd.c</code></td>
<td>
<p>Standard deviation in control group.</p>
</td></tr>
<tr><td><code>O.E</code></td>
<td>
<p>Observed minus expected (IPD analysis).</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>Variance of <code>O.E</code> (IPD analysis).</p>
</td></tr>
<tr><td><code>TE</code>, <code>seTE</code></td>
<td>
<p>Estimated treatment effect and standard error of
individual studies.</p>
</td></tr>
<tr><td><code>lower</code>, <code>upper</code></td>
<td>
<p>Lower and upper limit of 95% confidence
interval for treatment effect in individual studies.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>Weight of individual studies (according to
meta-analytical method used in respective meta-analysis - see
details).</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>Ordering of studies.</p>
</td></tr>
<tr><td><code>grplab</code></td>
<td>
<p>Group label.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>Type of outcome. D = dichotomous, C = continuous, P =
IPD.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>A character string indicating which method has been
used for pooling of studies. One of <code>"Inverse"</code>,
<code>"MH"</code>, or <code>"Peto"</code>.</p>
</td></tr>
<tr><td><code>sm</code></td>
<td>
<p>A character string indicating which summary measure has
been used for pooling of studies.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>A character string indicating which meta-analytical
model has been used (either <code>"Fixed"</code> or <code>"Random"</code>).</p>
</td></tr>
<tr><td><code>common</code></td>
<td>
<p>A logical indicating whether common effect
meta-analysis has been used in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>random</code></td>
<td>
<p>A logical indicating whether random effects
meta-analysis has been used in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>title</code></td>
<td>
<p>Title of Cochrane review.</p>
</td></tr>
<tr><td><code>complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code>outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code>label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code>label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code>label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code>label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
</table>
<p>The data frame 'rob' contains the following variables:
</p>
<table>
<tr><td><code>studlab</code></td>
<td>
<p>Study label.</p>
</td></tr>
<tr><td><code>D1</code>, <code>D2</code>, <code>...</code></td>
<td>
<p>Risk of bias domain 1, 2, ...</p>
</td></tr>
<tr><td><code>D1.details</code>, <code>D2.details</code>, <code>...</code></td>
<td>
<p>Details on risk of bias domain
1, 2, ...</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p><a href="https://documentation.cochrane.org/revman-kb/data-package-user-guide-243761660.html">https://documentation.cochrane.org/revman-kb/data-package-user-guide-243761660.html</a>
</p>
<p><a href="https://documentation.cochrane.org/revman-kb/data-package-specification-249561249.html">https://documentation.cochrane.org/revman-kb/data-package-specification-249561249.html</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metacr">metacr</a></code>, <code><a href="#topic+rob">rob</a></code>,
<code><a href="#topic+read.rm5">read.rm5</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate file "Fleiss1993.zip" with Cochrane data package in
# sub-directory of R package meta
#
filename &lt;- system.file("extdata/Fleiss1993.zip", package = "meta")
Fleiss1993_CR &lt;- read.cdir(filename)
Fleiss1993_CR

# Same result as R Command example(Fleiss1993bin):
#
metacr(Fleiss1993_CR)

# Same result as R Command example(Fleiss1993cont):
#
metacr(Fleiss1993_CR, 1, 2)

</code></pre>

<hr>
<h2 id='read.mtv'>Import RevMan 4 data files (.mtv)</h2><span id='topic+read.mtv'></span>

<h3>Description</h3>

<p>Reads a file created with RevMan 4 and creates a data frame from
it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.mtv(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.mtv_+3A_file">file</code></td>
<td>
<p>The name of a file to read data values from.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Reads a file created with RevMan 4 (Menu: &quot;File&quot; - &quot;Export&quot; -
&quot;Analysis data file...&quot;) and creates a data frame from it.
</p>


<h3>Value</h3>

<p>A data frame containing the following components:
</p>
<table>
<tr><td><code>comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code>outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code>group.no</code></td>
<td>
<p>Group number.</p>
</td></tr>
<tr><td><code>studlab</code></td>
<td>
<p>Study label.</p>
</td></tr>
<tr><td><code>year</code></td>
<td>
<p>Year of publication.</p>
</td></tr>
<tr><td><code>event.e</code></td>
<td>
<p>Number of events in experimental group.</p>
</td></tr>
<tr><td><code>n.e</code></td>
<td>
<p>Number of observations in experimental group.</p>
</td></tr>
<tr><td><code>event.c</code></td>
<td>
<p>Number of events in control group.</p>
</td></tr>
<tr><td><code>n.c</code></td>
<td>
<p>Number of observations in control group.</p>
</td></tr>
<tr><td><code>mean.e</code></td>
<td>
<p>Estimated mean in experimental group.</p>
</td></tr>
<tr><td><code>sd.e</code></td>
<td>
<p>Standard deviation in experimental group.</p>
</td></tr>
<tr><td><code>mean.c</code></td>
<td>
<p>Estimated mean in control group.</p>
</td></tr>
<tr><td><code>sd.c</code></td>
<td>
<p>Standard deviation in control group.</p>
</td></tr>
<tr><td><code>O.E</code></td>
<td>
<p>Observed minus expected (IPD analysis).</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>Variance of <code>O.E</code> (IPD analysis).</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>Ordering of studies.</p>
</td></tr>
<tr><td><code>conceal</code></td>
<td>
<p>Concealment of treatment allocation.</p>
</td></tr>
<tr><td><code>grplab</code></td>
<td>
<p>Group label.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>Type of outcome. D = dichotomous, C = continuous, P =
IPD.</p>
</td></tr>
<tr><td><code>outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code>graph.exp</code></td>
<td>
<p>Graph label for experimental group.</p>
</td></tr>
<tr><td><code>graph.cont</code></td>
<td>
<p>Graph label for control group.</p>
</td></tr>
<tr><td><code>label.exp</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code>label.cont</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code>complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p><em>Review Manager (RevMan)</em> [Computer program]. Version 4.2.
Copenhagen: The Nordic Cochrane Centre, The Cochrane Collaboration, 2003
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate MTV-data file "FLEISS1993.MTV" in sub-directory of R package
# meta
#
filename &lt;- system.file("extdata/FLEISS1993.MTV", package = "meta")
fleiss1933.cc &lt;- read.mtv(filename)

# Same result as R Command example(Fleiss1993bin):
#
metabin(event.e, n.e, event.c, n.c,
  data = fleiss1933.cc, subset = type == "D",
  studlab = paste(studlab, year))

# Same result: example(Fleiss1993cont)
#
metacont(n.e, mean.e, sd.e, n.c, mean.c, sd.c,
  data = fleiss1933.cc, subset = type == "C",
  studlab = paste(studlab, year))

</code></pre>

<hr>
<h2 id='read.rm5'>Import RevMan 5 analysis data</h2><span id='topic+read.rm5'></span><span id='topic+Fleiss1993_CR'></span>

<h3>Description</h3>

<p>Reads analysis data from Cochrane intervention review created with
RevMan 5 and creates a data frame from it.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read.rm5(
  file,
  sep = ",",
  quote = "\"",
  title,
  numbers.in.labels = TRUE,
  debug = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read.rm5_+3A_file">file</code></td>
<td>
<p>The name of a file to read data values from.</p>
</td></tr>
<tr><td><code id="read.rm5_+3A_sep">sep</code></td>
<td>
<p>The field separator character (only considered for
CSV-files). Values on each line of the file are separated by this
character. The comma is the default field separator character in
RevMan 5.</p>
</td></tr>
<tr><td><code id="read.rm5_+3A_quote">quote</code></td>
<td>
<p>The set of quoting characters (only considered for
CSV-files). In RevMan 5 a &quot;\&quot;&quot; is the default quoting character.</p>
</td></tr>
<tr><td><code id="read.rm5_+3A_title">title</code></td>
<td>
<p>Title of Cochrane review.</p>
</td></tr>
<tr><td><code id="read.rm5_+3A_numbers.in.labels">numbers.in.labels</code></td>
<td>
<p>A logical indicating whether comparison
number and outcome number should be printed at the beginning of
the comparison (argument <code>complab</code>) and outcome label
(argument <code>outclab</code>); this is the default in RevMan 5.</p>
</td></tr>
<tr><td><code id="read.rm5_+3A_debug">debug</code></td>
<td>
<p>An integer between 0 and 3 indicating whether to print
debug messages (only considered for RM5-files).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Review Manager 5 (RevMan 5) was the software used for preparing and
maintaining Cochrane reviews
(<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>).
RevMan 5 includes the ability to write systematic reviews of
interventions, diagnostic test accuracy reviews, methodology
reviews and overviews of reviews.
</p>
<p>This function provides the ability to read the analysis data from a
Cochrane intervention review created with RevMan 5; a data frame is
created from it. Cochrane intervention reviews are based on
comparisons of two interventions.
</p>
<p>By default in RevMan 5, the name of the exported CSV data file is
the title of the Cochrane review. Furthermore, the title is part of
the RM5-file. Argument <code>title</code> can be used to overwrite the
title of the Cochrane review.
</p>


<h4>Import RM5-file</h4>

<p>A RM5-file (which is in a specific XML format) can be used directly
to import the analysis dataset.
</p>
<p>If the import fails, use argument <code>debug = 3</code> for more details.
</p>



<h4>Import CSV-file</h4>

<p>In the past, the following (rather complicated) procedure based on
a CSV-file generated within RevMan 5 was necessary - which is only
described here for backward compatibility.
</p>
<p>In order to generate a data analysis file in RevMan 5 use the
following Menu points: <code>"File"</code> - <code>"Export"</code> -
<code>"Data and analyses"</code>. It is mandatory to include the
following fields in the exported data file by selecting them with
the mouse cursor in the Export Analysis Data Wizard: (i) Comparison
Number, (ii) Outcome Number, (iii) Subgroup Number. When these
fields are not selected a corresponding error message will be
printed in R.  It is recommended to include all fields in the
exported data file except for the last field
&quot;Risk of bias tables&quot;. For example, in order to redo the
meta-analysis in R for the RevMan 5 data type
<code>"O-E and Variance"</code> the fields <code>"O-E"</code> and
<code>"Variance"</code> have to be selected in the Export Analysis Data
Wizard. If the last field &quot;Risk of bias tables&quot; is selected the
import in R fails with an error message
&quot;line X did not have Y elements&quot;.
</p>



<h3>Value</h3>

<p>A data frame containing the following components:
</p>
<table>
<tr><td><code>comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code>outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code>group.no</code></td>
<td>
<p>Group number.</p>
</td></tr>
<tr><td><code>studlab</code></td>
<td>
<p>Study label.</p>
</td></tr>
<tr><td><code>year</code></td>
<td>
<p>Year of publication.</p>
</td></tr>
<tr><td><code>event.e</code></td>
<td>
<p>Number of events in experimental group.</p>
</td></tr>
<tr><td><code>n.e</code></td>
<td>
<p>Number of observations in experimental group.</p>
</td></tr>
<tr><td><code>event.c</code></td>
<td>
<p>Number of events in control group.</p>
</td></tr>
<tr><td><code>n.c</code></td>
<td>
<p>Number of observations in control group.</p>
</td></tr>
<tr><td><code>mean.e</code></td>
<td>
<p>Estimated mean in experimental group.</p>
</td></tr>
<tr><td><code>sd.e</code></td>
<td>
<p>Standard deviation in experimental group.</p>
</td></tr>
<tr><td><code>mean.c</code></td>
<td>
<p>Estimated mean in control group.</p>
</td></tr>
<tr><td><code>sd.c</code></td>
<td>
<p>Standard deviation in control group.</p>
</td></tr>
<tr><td><code>O.E</code></td>
<td>
<p>Observed minus expected (IPD analysis).</p>
</td></tr>
<tr><td><code>V</code></td>
<td>
<p>Variance of <code>O.E</code> (IPD analysis).</p>
</td></tr>
<tr><td><code>TE</code>, <code>seTE</code></td>
<td>
<p>Estimated treatment effect and standard error of
individual studies.</p>
</td></tr>
<tr><td><code>lower</code>, <code>upper</code></td>
<td>
<p>Lower and upper limit of 95% confidence
interval for treatment effect in individual studies.</p>
</td></tr>
<tr><td><code>weight</code></td>
<td>
<p>Weight of individual studies (according to
meta-analytical method used in respective meta-analysis - see
details).</p>
</td></tr>
<tr><td><code>order</code></td>
<td>
<p>Ordering of studies.</p>
</td></tr>
<tr><td><code>grplab</code></td>
<td>
<p>Group label.</p>
</td></tr>
<tr><td><code>type</code></td>
<td>
<p>Type of outcome. D = dichotomous, C = continuous, P =
IPD.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>A character string indicating which method has been
used for pooling of studies. One of <code>"Inverse"</code>,
<code>"MH"</code>, or <code>"Peto"</code>.</p>
</td></tr>
<tr><td><code>sm</code></td>
<td>
<p>A character string indicating which summary measure has
been used for pooling of studies.</p>
</td></tr>
<tr><td><code>model</code></td>
<td>
<p>A character string indicating which meta-analytical
model has been used (either <code>"Fixed"</code> or <code>"Random"</code>).</p>
</td></tr>
<tr><td><code>common</code></td>
<td>
<p>A logical indicating whether common effect
meta-analysis has been used in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>random</code></td>
<td>
<p>A logical indicating whether random effects
meta-analysis has been used in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code>k</code></td>
<td>
<p>Total number of studies combined in respective
meta-analysis).</p>
</td></tr>
<tr><td><code>event.e.pooled</code></td>
<td>
<p>Number of events in experimental group in
respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>n.e.pooled</code></td>
<td>
<p>Number of observations in experimental group in
respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>event.c.pooled</code></td>
<td>
<p>Number of events in control group in
respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>n.c.pooled</code></td>
<td>
<p>Number of observations in control group in
respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>TE.pooled</code></td>
<td>
<p>Estimated treatment effect in respective
meta-analysis (see details).</p>
</td></tr>
<tr><td><code>lower</code>, <code>upper</code></td>
<td>
<p>Lower and upper limit of 95% confidence
interval for treatment effect in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>weight.pooled</code></td>
<td>
<p>Total weight in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>Z.pooled</code></td>
<td>
<p>Z-score for test of overall treatment effect in
respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>pval.pooled</code></td>
<td>
<p>P-value for test of overall treatment effect in
respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>Q</code></td>
<td>
<p>Heterogeneity statistic Q in respective meta-analysis (see
details).</p>
</td></tr>
<tr><td><code>pval.Q</code></td>
<td>
<p>P-value of heterogeneity statistic Q in respective
meta-analysis (see details).</p>
</td></tr>
<tr><td><code>I2</code></td>
<td>
<p>Heterogeneity statistic I<code class="reqn">^2</code> in respective meta-analysis
(see details).</p>
</td></tr>
<tr><td><code>tau2</code></td>
<td>
<p>Between-study variance (moment estimator of
DerSimonian-Laird) in respective meta-analysis.</p>
</td></tr>
<tr><td><code>Q.w</code></td>
<td>
<p>Heterogeneity statistic Q within groups in respective
meta-analysis (see details).</p>
</td></tr>
<tr><td><code>pval.Q.w</code></td>
<td>
<p>P-value of heterogeneity statistic Q within groups
in respective meta-analysis (see details).</p>
</td></tr>
<tr><td><code>I2.w</code></td>
<td>
<p>Heterogeneity statistic I<code class="reqn">^2</code> within groups in respective
meta-analysis (see details).</p>
</td></tr>
<tr><td><code>label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code>label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code>label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code>label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code>complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p><em>Review Manager (RevMan)</em> [Computer program]. Version 5.4.
The Cochrane Collaboration, 2020
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.rm5">summary.rm5</a></code>, <code><a href="#topic+metabias.rm5">metabias.rm5</a></code>,
<code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metacr">metacr</a></code>,
<code><a href="#topic+print.rm5">print.rm5</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate export data file "Fleiss1993_CR.csv"
# in sub-directory of package "meta"
#
filename &lt;- system.file("extdata", "Fleiss1993_CR.csv", package = "meta")
Fleiss1993_CR &lt;- read.rm5(filename)

# Same result as R command example(Fleiss1993bin):
#
metacr(Fleiss1993_CR)

# Same result as R command example(Fleiss1993cont):
#
metacr(Fleiss1993_CR, 1, 2)

## Not run: 
# Locate file "Fleiss1993.rm5" in sub-directory of R package meta
#
filename &lt;- system.file("extdata/Fleiss1993.rm5", package = "meta")
Fleiss1993_CR &lt;- read.cdir(filename)
Fleiss1993_CR

# Same result as R Command example(Fleiss1993bin):
#
metacr(Fleiss1993_CR)

## End(Not run)

</code></pre>

<hr>
<h2 id='rob'>Risk of bias assessment</h2><span id='topic+rob'></span><span id='topic+print.rob'></span>

<h3>Description</h3>

<p>Create table with risk of bias assessment or add table to existing
meta-analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rob(
  item1,
  item2 = NULL,
  item3 = NULL,
  item4 = NULL,
  item5 = NULL,
  item6 = NULL,
  item7 = NULL,
  item8 = NULL,
  item9 = NULL,
  item10 = NULL,
  studlab = NULL,
  overall = NULL,
  weight = NULL,
  data = NULL,
  tool = gs("tool.rob"),
  domains = NULL,
  categories = NULL,
  cat1 = categories,
  cat2 = categories,
  cat3 = categories,
  cat4 = categories,
  cat5 = categories,
  cat6 = categories,
  cat7 = categories,
  cat8 = categories,
  cat9 = categories,
  cat10 = categories,
  cat.overall = categories,
  col = NULL,
  col1 = col,
  col2 = col,
  col3 = col,
  col4 = col,
  col5 = col,
  col6 = col,
  col7 = col,
  col8 = col,
  col9 = col,
  col10 = col,
  col.overall = col,
  symbols = NULL,
  symb1 = symbols,
  symb2 = symbols,
  symb3 = symbols,
  symb4 = symbols,
  symb5 = symbols,
  symb6 = symbols,
  symb7 = symbols,
  symb8 = symbols,
  symb9 = symbols,
  symb10 = symbols,
  symb.overall = symbols,
  legend = TRUE,
  overwrite = FALSE,
  warn = TRUE
)

## S3 method for class 'rob'
print(x, legend = attr(x, "legend"), details = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rob_+3A_item1">item1</code></td>
<td>
<p>Risk of bias item 1 or a meta-analysis object of class
<code>meta</code> with information on risk of bias assessment.</p>
</td></tr>
<tr><td><code id="rob_+3A_item2">item2</code></td>
<td>
<p>Risk of bias item 2.</p>
</td></tr>
<tr><td><code id="rob_+3A_item3">item3</code></td>
<td>
<p>Risk of bias item 3.</p>
</td></tr>
<tr><td><code id="rob_+3A_item4">item4</code></td>
<td>
<p>Risk of bias item 4.</p>
</td></tr>
<tr><td><code id="rob_+3A_item5">item5</code></td>
<td>
<p>Risk of bias item 5.</p>
</td></tr>
<tr><td><code id="rob_+3A_item6">item6</code></td>
<td>
<p>Risk of bias item 6.</p>
</td></tr>
<tr><td><code id="rob_+3A_item7">item7</code></td>
<td>
<p>Risk of bias item 7.</p>
</td></tr>
<tr><td><code id="rob_+3A_item8">item8</code></td>
<td>
<p>Risk of bias item 8.</p>
</td></tr>
<tr><td><code id="rob_+3A_item9">item9</code></td>
<td>
<p>Risk of bias item 9.</p>
</td></tr>
<tr><td><code id="rob_+3A_item10">item10</code></td>
<td>
<p>Risk of bias item 10.</p>
</td></tr>
<tr><td><code id="rob_+3A_studlab">studlab</code></td>
<td>
<p>Study labels.</p>
</td></tr>
<tr><td><code id="rob_+3A_overall">overall</code></td>
<td>
<p>Overall risk of bias assess.</p>
</td></tr>
<tr><td><code id="rob_+3A_weight">weight</code></td>
<td>
<p>Weight for each study.</p>
</td></tr>
<tr><td><code id="rob_+3A_data">data</code></td>
<td>
<p>A data frame or a meta-analysis object of class
<code>meta</code>.</p>
</td></tr>
<tr><td><code id="rob_+3A_tool">tool</code></td>
<td>
<p>Risk of bias (RoB) tool.</p>
</td></tr>
<tr><td><code id="rob_+3A_domains">domains</code></td>
<td>
<p>A character vector with names of RoB domains.</p>
</td></tr>
<tr><td><code id="rob_+3A_categories">categories</code></td>
<td>
<p>Possible RoB categories.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat1">cat1</code></td>
<td>
<p>Possible categories for RoB item 1.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat2">cat2</code></td>
<td>
<p>Possible categories for RoB item 2.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat3">cat3</code></td>
<td>
<p>Possible categories for RoB item 3.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat4">cat4</code></td>
<td>
<p>Possible categories for RoB item 4.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat5">cat5</code></td>
<td>
<p>Possible categories for RoB item 5.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat6">cat6</code></td>
<td>
<p>Possible categories for RoB item 6.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat7">cat7</code></td>
<td>
<p>Possible categories for RoB item 7.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat8">cat8</code></td>
<td>
<p>Possible categories for RoB item 8.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat9">cat9</code></td>
<td>
<p>Possible categories for RoB item 9.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat10">cat10</code></td>
<td>
<p>Possible categories for RoB item 10.</p>
</td></tr>
<tr><td><code id="rob_+3A_cat.overall">cat.overall</code></td>
<td>
<p>Possible categories for overall RoB.</p>
</td></tr>
<tr><td><code id="rob_+3A_col">col</code></td>
<td>
<p>Colours for RoB categories.</p>
</td></tr>
<tr><td><code id="rob_+3A_col1">col1</code></td>
<td>
<p>Colours for categories for RoB item 1.</p>
</td></tr>
<tr><td><code id="rob_+3A_col2">col2</code></td>
<td>
<p>Colours for categories for RoB item 2.</p>
</td></tr>
<tr><td><code id="rob_+3A_col3">col3</code></td>
<td>
<p>Colours for categories for RoB item 3.</p>
</td></tr>
<tr><td><code id="rob_+3A_col4">col4</code></td>
<td>
<p>Colours for categories for RoB item 4.</p>
</td></tr>
<tr><td><code id="rob_+3A_col5">col5</code></td>
<td>
<p>Colours for categories for RoB item 5.</p>
</td></tr>
<tr><td><code id="rob_+3A_col6">col6</code></td>
<td>
<p>Colours for categories for RoB item 6.</p>
</td></tr>
<tr><td><code id="rob_+3A_col7">col7</code></td>
<td>
<p>Colours for categories for RoB item 7.</p>
</td></tr>
<tr><td><code id="rob_+3A_col8">col8</code></td>
<td>
<p>Colours for categories for RoB item 8.</p>
</td></tr>
<tr><td><code id="rob_+3A_col9">col9</code></td>
<td>
<p>Colours for categories for RoB item 9.</p>
</td></tr>
<tr><td><code id="rob_+3A_col10">col10</code></td>
<td>
<p>Colours for categories for RoB item 10.</p>
</td></tr>
<tr><td><code id="rob_+3A_col.overall">col.overall</code></td>
<td>
<p>Colours for categories for overall RoB.</p>
</td></tr>
<tr><td><code id="rob_+3A_symbols">symbols</code></td>
<td>
<p>Corresponding symbols for RoB categories.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb1">symb1</code></td>
<td>
<p>Corresponding symbols for RoB item 1.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb2">symb2</code></td>
<td>
<p>Corresponding symbols for RoB item 2.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb3">symb3</code></td>
<td>
<p>Corresponding symbols for RoB item 3.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb4">symb4</code></td>
<td>
<p>Corresponding symbols for RoB item 4.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb5">symb5</code></td>
<td>
<p>Corresponding symbols for RoB item 5.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb6">symb6</code></td>
<td>
<p>Corresponding symbols for RoB item 6.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb7">symb7</code></td>
<td>
<p>Corresponding symbols for RoB item 7.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb8">symb8</code></td>
<td>
<p>Corresponding symbols for RoB item 8.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb9">symb9</code></td>
<td>
<p>Corresponding symbols for RoB item 9.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb10">symb10</code></td>
<td>
<p>Corresponding symbols for RoB item 10.</p>
</td></tr>
<tr><td><code id="rob_+3A_symb.overall">symb.overall</code></td>
<td>
<p>Corresponding symbols for overall RoB.</p>
</td></tr>
<tr><td><code id="rob_+3A_legend">legend</code></td>
<td>
<p>A logical specifying whether legend with RoB domains
should be printed.</p>
</td></tr>
<tr><td><code id="rob_+3A_overwrite">overwrite</code></td>
<td>
<p>A logical indicating whether an existing risk of
bias table in a meta-analysis object should be overwritten.</p>
</td></tr>
<tr><td><code id="rob_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be
printed.</p>
</td></tr>
<tr><td><code id="rob_+3A_x">x</code></td>
<td>
<p>An object of class <code>rob</code>.</p>
</td></tr>
<tr><td><code id="rob_+3A_details">details</code></td>
<td>
<p>A logical indicating whether to print details on
categories and colours.</p>
</td></tr>
<tr><td><code id="rob_+3A_...">...</code></td>
<td>
<p>Additional printing arguments.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to define a risk of bias (RoB) assessment
for a meta-analysis which can be shown in a forest plot
(<code><a href="#topic+forest.meta">forest.meta</a></code>), summary weighted barplot
(<code><a href="#topic+barplot.rob">barplot.rob</a></code>) or traffic light plot
(<code><a href="#topic+traffic_light">traffic_light</a></code>). It is also possible to extract the
risk of bias assessment from a meta-analysis with RoB information.
</p>
<p>The risk of bias table contains
</p>

<ul>
<li><p> study labels;
</p>
</li>
<li><p> variables for individual RoB domains (with variable names A,
B, ...);
</p>
</li>
<li><p> an overall RoB assessment if argument <code>overall</code> is
provided;
</p>
</li>
<li><p> weights for individual studies used in summary weighted
barplots.
</p>
</li></ul>

<p>Note, an overall RoB assessment is mandatory to create a summary
weighted barplot or a traffic light plot.
</p>
<p>The RoB table is directly returned if argument <code>data</code> is a
data frame or argument <code>item1</code> is a meta-analysis with risk of
bias assessment. The RoB table is added as a new list element 'rob'
to a meta-analysis object if argument <code>data</code> is a
meta-analysis.
</p>
<p>The user must either specify the categories and (optionally)
domains of the RoB tool (using the eponymous arguments) or one of
the following RoB tools.
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Risk of bias tool</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tool = "RoB1"</code> </td><td style="text-align: left;"> RoB 1 tool for randomized studies
  (Higgins et al., 2011) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tool = "RoB2"</code> </td><td style="text-align: left;"> RoB 2 tool for randomized studies
  (Sterne et al., 2019) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tool = "RoB2-cluster"</code> </td><td style="text-align: left;"> RoB 2 tool for
  cluster-randomized trials </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tool = "RoB2-crossover"</code> </td><td style="text-align: left;"> RoB 2 tool for crossover
  trials </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tool = "ROBINS-I"</code> </td><td style="text-align: left;"> Risk Of Bias In Non-randomized
  Studies - of Interventions </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (Sterne et al., 2016) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tool = "ROBINS-E"</code> </td><td style="text-align: left;"> Risk Of Bias In Non-randomized
  Studies - of Exposures </td>
</tr>
<tr>
 <td style="text-align: left;">
</td><td style="text-align: left;"> (ROBINS-E Development Group, 2023)
</td>
</tr>

</table>

<p>These RoB tools are described on the website
<a href="https://www.riskofbias.info/">https://www.riskofbias.info/</a>.
</p>


<h4>Risk of bias domains</h4>

<p>By default, i.e., if argument <code>domains</code> is not provided by the
user, the following names are used for RoB domains.
</p>

<ul>
<li><p> RoB 1 tool for randomized studies (RoB1):
</p>

<ol>
<li><p> Random sequence generation (selection bias)
</p>
</li>
<li><p> Allocation concealment (selection bias)
</p>
</li>
<li><p> Blinding of participants and personnel (performance bias)
</p>
</li>
<li><p> Blinding of outcome assessment (detection bias)
</p>
</li>
<li><p> Incomplete outcome data (attrition bias)
</p>
</li>
<li><p> Selective reporting (reporting bias)
</p>
</li>
<li><p> Other bias
</p>
</li></ol>

</li>
<li><p> RoB 2 tool for randomized studies (RoB2):
</p>

<ol>
<li><p> Bias arising from the randomization process
</p>
</li>
<li><p> Bias due to deviations from intended intervention&quot;
</p>
</li>
<li><p> Bias due to missing outcome data
</p>
</li>
<li><p> Bias in measurement of the outcome
</p>
</li>
<li><p> Bias in selection of the reported result
</p>
</li></ol>

</li>
<li><p> RoB 2 tool for cluster-randomized trials (RoB2-cluster):
</p>

<ol>
<li><p> Bias arising from the randomization process
</p>
</li>
<li><p> Bias arising from the identification or recruitment of
participants into clusters
</p>
</li>
<li><p> Bias due to deviations from intended intervention
</p>
</li>
<li><p> Bias due to missing outcome data
</p>
</li>
<li><p> Bias in measurement of the outcome
</p>
</li>
<li><p> Bias in selection of the reported result
</p>
</li></ol>

</li>
<li><p> RoB 2 tool for crossover trials (RoB2-crossover)
</p>

<ol>
<li><p> Bias arising from the randomization process
</p>
</li>
<li><p> Bias arising from period and carryover effects
</p>
</li>
<li><p> Bias due to deviations from intended intervention
</p>
</li>
<li><p> Bias due to missing outcome data
</p>
</li>
<li><p> Bias in measurement of the outcome
</p>
</li>
<li><p> Bias in selection of the reported result
</p>
</li></ol>

</li>
<li><p> Risk Of Bias In Non-randomized Studies - of Intervention (ROBINS-I):
</p>

<ol>
<li><p> Risk of bias due to confounding
</p>
</li>
<li><p> Risk of bias in selection of participants into the study
</p>
</li>
<li><p> Risk of bias in classification of interventions
</p>
</li>
<li><p> Risk of bias due to deviations from intented interventions
</p>
</li>
<li><p> Risk of bias due to missing outcome data
</p>
</li>
<li><p> Risk of bias in measurement of the outcome
</p>
</li>
<li><p> Risk of bias in the selection of the reported results
</p>
</li></ol>

</li>
<li><p> Risk Of Bias In Non-randomized Studies - of Exposures (ROBINS-E):
</p>

<ol>
<li><p> Risk of bias due to confounding
</p>
</li>
<li><p> Risk of bias arising from measurement of the exposure into
the study (or into the analysis)
</p>
</li>
<li><p> Risk of bias due to post-exposure interventions
</p>
</li>
<li><p> Risk of bias due to deviations from intented interventions
</p>
</li>
<li><p> Risk of bias due to missing outcome data
</p>
</li>
<li><p> Risk of bias in measurement of the outcome
</p>
</li>
<li><p> Risk of bias in the selection of the reported results
</p>
</li></ol>

</li>
<li><p> User-defined RoB assessment:
</p>

<ol>
<li><p> First item
</p>
</li>
<li><p> Second item
</p>
</li>
<li> <p>...
</p>
</li></ol>

</li></ul>

<p>It is possible to define additional bias domains for the available
RoB tools. In this case, only the names for new RoB domains have to
be provided in argument <code>domains</code>. If argument <code>domains</code>
is not used to specify new domains, the names &quot;Additional item 1&quot;
etc. will be used. It is also possible to modify the pre-defined
domain names using argument <code>domains</code>.
</p>
<p>The maximum number of bias domains / items is ten (see arguments
<code>item1</code>, ..., <code>item10</code>).
</p>



<h4>Risk of bias categories, colours and symbols</h4>

<p>By default, the following settings are used.
</p>
<p>RoB 1 tool:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Values</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>categories</code> </td><td style="text-align: left;"> "Low risk of bias", "Unclear risk of bias",
  "High risk of bias" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>col</code> </td><td style="text-align: left;"> "green", "yellow", "red" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>symbols</code> </td><td style="text-align: left;"> "+", "?", "-"
</td>
</tr>

</table>

<p>RoB 2 tools:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Values</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>categories</code> </td><td style="text-align: left;"> "Low risk of bias", "Some concerns",
  "High risk of bias" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>col</code> </td><td style="text-align: left;"> "green", "yellow", "red" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>symbols</code> </td><td style="text-align: left;"> "+", "?", "-"
</td>
</tr>

</table>

<p>ROBINS tools:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Values</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>categories</code> </td><td style="text-align: left;"> "Low risk", "Some concerns", "High risk",
  "Very high risk", "NI" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>col</code> </td><td style="text-align: left;"> "green", "yellow", "red", "darkred", "darkgrey" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>symbols</code> </td><td style="text-align: left;"> none
</td>
</tr>

</table>

<p>User-defined RoB tools:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Values</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>categories</code> </td><td style="text-align: left;"> Must be specified by the user </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>col</code> </td><td style="text-align: left;"> 1, 2, ... </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>symbols</code> </td><td style="text-align: left;"> none
</td>
</tr>

</table>

<p>If colours (<code>col</code>) and symbols (<code>symbols</code>) are provided,
they must be of the same length as the number of categories. 
</p>



<h3>Value</h3>

<p>A data frame with study labels and risk of bias items and
additional class &quot;rob&quot;.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Higgins JPT, Altman DG, Gøtzsche PC, Jüni P, Moher D, Oxman AD et
al. (2011):
The Cochrane Collaboration's tool for assessing risk of bias in
randomised trials.
<em>British Medical Journal</em>, <b>343</b>: d5928
</p>
<p>ROBINS-E Development Group (Higgins J, Morgan R, Rooney A et al.) (2023):
Risk Of Bias In Non-randomized Studies - of Exposure (ROBINS-E)
Available from: <a href="https://www.riskofbias.info/welcome/robins-e-tool">https://www.riskofbias.info/welcome/robins-e-tool</a>.
</p>
<p>Sterne JA, Hernán MA, Reeves BC, Savović J, Berkman ND, Viswanathan
M, et al. (2016):
ROBINS-I: a tool for assessing risk of bias in non-randomised
studies of interventions.
<em>British Medical Journal</em>, <b>355</b>: i4919
</p>
<p>Sterne JAC, Savović J, Page MJ, Elbers RG, Blencowe NS, Boutron I,
et al. (2019):
RoB 2: a revised tool for assessing risk of bias in randomised trials.
<em>British Medical Journal</em>, <b>366</b>: l4898.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+forest.meta">forest.meta</a></code>, <code><a href="#topic+barplot.rob">barplot.rob</a></code>,
<code><a href="#topic+traffic_light">traffic_light</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Use RevMan 5 settings
oldset &lt;- settings.meta("RevMan5", quietly = FALSE)

data(caffeine)

m1 &lt;- metabin(h.caf, n.caf, h.decaf, n.decaf, sm = "OR",
  data = caffeine, studlab = paste(study, year))

# Add risk of bias assessment to meta-analysis
m2 &lt;- rob(D1, D2, D3, D4, D5, overall = rob, data = m1, tool = "rob2")

# Print risk of bias assessment
rob(m2)

# Forest plot with risk of bias assessment
forest(m2)

# Use previous settings
settings.meta(oldset)

</code></pre>

<hr>
<h2 id='settings.meta'>Print and change default settings to conduct and print or plot
meta-analyses in R package <b>meta</b>.</h2><span id='topic+settings.meta'></span>

<h3>Description</h3>

<p>Print and change default settings to conduct and print or plot
meta-analyses in R package <b>meta</b>. The following general
settings are available: <em>Review Manager 5</em>, <em>Journal of
the American Medical Association</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>settings.meta(..., quietly = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="settings.meta_+3A_...">...</code></td>
<td>
<p>Arguments to change default settings.</p>
</td></tr>
<tr><td><code id="settings.meta_+3A_quietly">quietly</code></td>
<td>
<p>A logical indicating whether information on settings
should be printed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to define defaults for several arguments
(i.e., assignments using <code><a href="#topic+gs">gs</a></code>) of the following R
functions: <code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metacor">metacor</a></code>, <code><a href="#topic+metacr">metacr</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metaprop">metaprop</a></code>,
<code><a href="#topic+metarate">metarate</a></code>
</p>
<p>Furthermore, some of these settings are considered to print
meta-analysis results and to produce forest plots.
</p>
<p>The function can be used to either change individual settings (see
Examples) or use one of the following general settings:
</p>

<ul>
<li> <p><code>settings.meta("RevMan5")</code>
</p>
</li>
<li> <p><code>settings.meta("BMJ")</code>
</p>
</li>
<li> <p><code>settings.meta("JAMA")</code>
</p>
</li>
<li> <p><code>settings.meta("IQWiG5")</code>
</p>
</li>
<li> <p><code>settings.meta("IQWiG6")</code>
</p>
</li>
<li> <p><code>settings.meta("geneexpr")</code>
</p>
</li>
<li> <p><code>settings.meta("meta4")</code>
</p>
</li></ul>

<p>The first command can be used to reproduce meta-analyses from
Cochrane reviews conducted with <em>Review Manager 5</em> (RevMan 5,
<a href="https://training.cochrane.org/online-learning/core-software/revman">https://training.cochrane.org/online-learning/core-software/revman</a>)
and specifies to use a RevMan 5 layout in forest plots.
</p>
<p>The second command can be used to generate forest plots in BMJ layout.
</p>
<p>The third command can be used to generate forest plots following
instructions for authors of the <em>Journal of the American
Medical Association</em>
(<a href="https://jamanetwork.com/journals/jama/pages/instructions-for-authors/">https://jamanetwork.com/journals/jama/pages/instructions-for-authors/</a>).Study
labels according to JAMA guidelines can be generated using
<code><a href="#topic+labels.meta">labels.meta</a></code>.
</p>
<p>The next commands implement the recommendations of the Institute
for Quality and Efficiency in Health Care, Germany (IQWiG)
accordinging to General Methods 5 and 6, respectively
(<a href="https://www.iqwig.de/en/about-us/methods/methods-paper/">https://www.iqwig.de/en/about-us/methods/methods-paper/</a>).
</p>
<p>The setting <code>"geneexpr"</code> can be used to print p-values in
scientific notation and to suppress the calculation of confidence
intervals for the between-study variance.
</p>
<p>The last setting uses the default settings of R package
<b>meta</b>, version 4 or below.
</p>
<p>RevMan 5 settings, in detail:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci</code> </td><td style="text-align: left;"> "classic" </td><td style="text-align: left;"> only available method
  in RevMan 5 </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau</code> </td><td style="text-align: left;"> "DL" </td><td style="text-align: left;"> only available method in RevMan 5
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>tau.common</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> common between-study variance in
  subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>MH.exact</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> exact Mantel-Haenszel method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>RR.Cochrane</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> calculation of risk ratios </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>Q.Cochrane</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> calculation of heterogeneity
  statistic </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>exact.smd</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> exact formulae for Hedges' g and
  Cohen's d </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>layout</code> </td><td style="text-align: left;"> "RevMan5" </td><td style="text-align: left;"> layout for forest plots </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prediction</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> no prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.overall</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print information on test of
  overall effect </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.subgroup</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print information on test for
  subgroup differences </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.effect.subgroup</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print information on
  test for effect in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.I2</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show heterogeneity statistic I2 in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.tau2</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show between-study heterogeneity variance
  in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.tau</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show between-study heterogeneity
  standard deviation in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.Q</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show heterogeneity statistic Q in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.pval.Q</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show p-value of test for heterogeneity
  in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.Rb</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show heterogeneity statistic Rb in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.tau2</code> </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> number of digits for tau-squared </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.tau</code> </td><td style="text-align: left;"> 4 </td><td style="text-align: left;"> number of digits for square root of
  tau-squared </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.I2</code> </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> number of digits for I-squared measure
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CIbracket</code>, </td><td style="text-align: left;"> "[" </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CIseparator</code> </td><td style="text-align: left;"> ", " </td><td style="text-align: left;"> print confidence intervals as
  "<code>[., .]</code>" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>header.line</code>, </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print header line
</td>
</tr>

</table>

<p>BMJ settings:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>layout</code> </td><td style="text-align: left;"> "BMJ" </td><td style="text-align: left;"> layout for forest plots </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.overall</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print information on test of
  overall effect </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.subgroup</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> print information on test for
  subgroup differences </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.effect.subgroup</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> print information on
  test for effect in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.I2</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show heterogeneity statistic I2 in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.tau2</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show between-study heterogeneity variance
  in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.tau</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show between-study heterogeneity
  standard deviation in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.Q</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show heterogeneity statistic Q in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.pval.Q</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show p-value of test for heterogeneity
  in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.Rb</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show heterogeneity statistic Rb in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.I2</code> </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> number of digits for I-squared measure
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.pval</code> </td><td style="text-align: left;"> 2 </td><td style="text-align: left;"> number of digits for p-values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CIbracket</code>, </td><td style="text-align: left;"> "(" </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CIseparator</code> </td><td style="text-align: left;"> " to " </td><td style="text-align: left;"> print confidence intervals as
  "<code>(. to .)</code>" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>hetlab</code>, </td><td style="text-align: left;"> </td><td style="text-align: left;"> "Test for heterogeneity: " </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>header.line</code>, </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print header line
</td>
</tr>

</table>

<p>JAMA settings:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>layout</code> </td><td style="text-align: left;"> "JAMA" </td><td style="text-align: left;"> layout for forest plots </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.overall</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print information on test of
  overall effect </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.subgroup</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> print information on test for
  subgroup differences </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>test.effect.subgroup</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> print information on
  test for effect in subgroups </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.I2</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show heterogeneity statistic I2 in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.tau2</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show between-study heterogeneity
  variance in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.tau</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show between-study heterogeneity
  standard deviation in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.Q</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show heterogeneity statistic Q in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.pval.Q</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> show p-value of test for heterogeneity
  in forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>forest.Rb</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> do not show heterogeneity statistic Rb in
  forest plots
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.I2</code> </td><td style="text-align: left;"> 0 </td><td style="text-align: left;"> number of digits for I-squared measure
  </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>digits.pval</code> </td><td style="text-align: left;"> 3 </td><td style="text-align: left;"> number of digits for p-values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CIbracket</code>, </td><td style="text-align: left;"> "(" </td><td style="text-align: left;"> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>CIseparator</code> </td><td style="text-align: left;"> "-" </td><td style="text-align: left;"> print confidence intervals as
  "<code>(.-.)</code>" </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>zero.pval</code>, </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> print p-values with leading zero
</td>
</tr>
<tr>
 <td style="text-align: left;">
<code>JAMA.pval</code>, </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> round p-values to three digits
  (for 0.001 &lt; p <code class="reqn">\le</code> 0.01) or two digits (p &gt; 0.01) </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>header.line</code>, </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> print header line
</td>
</tr>

</table>

<p>IQWiG, General Methods 5 settings:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci</code> </td><td style="text-align: left;"> "HK" </td><td style="text-align: left;"> Hartung-Knapp method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prediction</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> Prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>IQWiG, General Methods 6 settings:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.random.ci</code> </td><td style="text-align: left;"> "HK" </td><td style="text-align: left;"> Hartung-Knapp method </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>adhoc.hakn.ci</code> </td><td style="text-align: left;"> "IQWiG6" </td><td style="text-align: left;"> <em>ad hoc</em> variance correction </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau</code> </td><td style="text-align: left;"> "PM" </td><td style="text-align: left;"> Paule-Mandel estimator for
  between-study variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>prediction</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> Prediction interval </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Settings for gene expression data:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>scientific.pval</code> </td><td style="text-align: left;"> TRUE </td><td style="text-align: left;"> Scientific notation for p-values </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau.ci</code> </td><td style="text-align: left;"> FALSE </td><td style="text-align: left;"> no confidence interval for </td>
</tr>
<tr>
 <td style="text-align: left;">
 </td><td style="text-align: left;"> between-study heterogeneity variance </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>Settings for <b>meta</b>, version 4 or below:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Argument</b> </td><td style="text-align: left;"> <b>Value</b> </td><td style="text-align: left;"> <b>Comment</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
<code>method.tau</code> </td><td style="text-align: left;"> "DL" </td><td style="text-align: left;"> DerSimonian-Laird estimator </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>A list of all arguments with current settings is printed using the
command <code>settings.meta("print")</code>.
</p>
<p>In order to reset all settings of R package <b>meta</b> the command
<code>settings.meta("reset")</code> or <code>settings.meta(reset = TRUE)</code>
can be used.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+gs">gs</a></code>, <code><a href="#topic+forest.meta">forest.meta</a></code>,
<code><a href="#topic+print.meta">print.meta</a></code>, <code><a href="#topic+labels.meta">labels.meta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Get listing of current settings
#
settings.meta()

# Meta-analyses using default settings
#
metabin(10, 20, 15, 20)
metaprop(4, 20)
metabin(10, 20, 15, 20, sm = "RD")
metaprop(4, 20, sm = "PLN")

# Change summary measure for R functions metabin and metaprop
# and store old settings
#
oldset &lt;- settings.meta(smbin = "RD", smprop = "PLN")
#
metabin(10, 20, 15, 20)
metaprop(4, 20)

# Use old settings
#
settings.meta(oldset)

# Change level used to calculate confidence intervals
# (99%-CI for studies, 99.9%-CI for pooled effects)
#
metagen(1:3, 2:4 / 10, sm = "MD")
settings.meta(level = 0.99, level.ma = 0.999)
metagen(1:3, 2:4 / 10, sm = "MD")

# Always print a prediction interval
#
settings.meta(prediction = TRUE)
metagen(1:3, 2:4 / 10, sm = "MD")
metagen(4:6, 4:2 / 10, sm = "MD")

# Try to set unknown argument results in a warning
#
try(settings.meta(unknownarg = TRUE))

# Reset to default settings of R package meta
#
settings.meta("reset")
metabin(10, 20, 15, 20)
metaprop(4, 20)
metagen(1:3, 2:4 / 10, sm = "MD")

# Do not back transform results (e.g. print log odds ratios instead
# of odds ratios, print transformed correlations / proportions
# instead of correlations / proportions)
#
settings.meta(backtransf = FALSE)
metabin(10, 20, 15, 20)
metaprop(4, 20)
metacor(c(0.85, 0.7, 0.95), c(20, 40, 10))

# Forest plot using RevMan 5 style
#
settings.meta("RevMan5")
forest(metagen(1:3, 2:4 / 10, sm = "MD", common = FALSE),
  label.left = "Favours A", label.right = "Favours B",
  colgap.studlab = "2cm", colgap.forest.left = "0.2cm")

# Forest plot using JAMA style
#
settings.meta("JAMA")
forest(metagen(1:3, 2:4 / 10, sm = "MD", common = FALSE),
  label.left = "Favours A", label.right = "Favours B",
  colgap.studlab = "2cm", colgap.forest.left = "0.2cm")

# Use slightly different layout for confidence intervals
# (especially useful if upper confidence limit can be negative)
#
settings.meta(CIseparator = " - ")
forest(metagen(-(1:3), 2:4 / 10, sm = "MD", common = FALSE),
  label.left = "Favours A", label.right = "Favours B",
  colgap.studlab = "2cm", colgap.forest.left = "0.2cm")

# Use old settings
#
settings.meta(oldset)

</code></pre>

<hr>
<h2 id='smd2or'>Conversion from standardised mean difference to log odds ratio</h2><span id='topic+smd2or'></span>

<h3>Description</h3>

<p>Conversion from standardised mean difference to log odds ratio
using method by Hasselblad &amp; Hedges (1995) or Cox (1970).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>smd2or(
  smd,
  se.smd,
  studlab,
  data = NULL,
  subset = NULL,
  exclude = NULL,
  method = "HH",
  backtransf = gs("backtransf"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="smd2or_+3A_smd">smd</code></td>
<td>
<p>Standardised mean difference(s) (SMD) or meta-analysis
object.</p>
</td></tr>
<tr><td><code id="smd2or_+3A_se.smd">se.smd</code></td>
<td>
<p>Standard error(s) of SMD (ignored if argument
<code>smd</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="smd2or_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels (ignored if
argument <code>smd</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="smd2or_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the study information
(ignored if argument <code>smd</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="smd2or_+3A_subset">subset</code></td>
<td>
<p>An optional vector specifying a subset of studies to
be used (ignored if argument <code>smd</code> is a meta-analysis
object).</p>
</td></tr>
<tr><td><code id="smd2or_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots (ignored if argument <code>smd</code> is a meta-analysis object).</p>
</td></tr>
<tr><td><code id="smd2or_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is used to
convert SMDs to log odds ratios. Either <code>"HH"</code> or
<code>"CS"</code>, can be abbreviated.</p>
</td></tr>
<tr><td><code id="smd2or_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether odds ratios (if
TRUE) or log odds ratios (if FALSE) should be shown in printouts
and plots.</p>
</td></tr>
<tr><td><code id="smd2or_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="#topic+metagen">metagen</a></code> (ignored if argument <code>smd</code> is a
meta-analysis object).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function implements the following methods for the conversion
from standardised mean difference to log odds ratio:
</p>

<ul>
<li><p> Hasselblad &amp; Hedges (1995) assuming logistic distributions
(<code>method == "HH"</code>)
</p>
</li>
<li><p> Cox (1970) and Cox &amp; Snell (1989) assuming normal
distributions (<code>method == "CS"</code>)
</p>
</li></ul>

<p>Internally, <code><a href="#topic+metagen">metagen</a></code> is used to conduct a
meta-analysis with the odds ratio as summary measure.
</p>
<p>Argument <code>smd</code> can be either a vector of standardised mean
differences or a meta-analysis object created with
<code><a href="#topic+metacont">metacont</a></code> or <code><a href="#topic+metagen">metagen</a></code> and the
standardised mean difference as summary measure.
</p>
<p>Argument <code>se.smd</code> is mandatory if argument <code>smd</code> is a
vector and ignored otherwise. Additional arguments in <code>...</code>
are only passed on to <code><a href="#topic+metagen">metagen</a></code> if argument <code>smd</code>
is a vector.
</p>


<h3>Value</h3>

<p>An object of class <code>c("metagen", "meta")</code> with corresponding
generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Borenstein M, Hedges LV, Higgins JPT, Rothstein HR (2009):
<em>Introduction to Meta-Analysis</em>.
Chichester: Wiley
</p>
<p>Cox DR (1970):
<em>Analysis of Binary Data</em>.
London: Chapman and Hall / CRC
</p>
<p>Cox DR, Snell EJ (1989):
<em>Analysis of Binary Data</em> (2nd edition).
London: Chapman and Hall / CRC
</p>
<p>Hasselblad V, Hedges LV (1995):
Meta-analysis of screening and diagnostic tests.
<em>Psychological Bulletin</em>,
<b>117</b>, 167&ndash;78
</p>


<h3>See Also</h3>

<p><code><a href="#topic+or2smd">or2smd</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metabin">metabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Example from Borenstein et al. (2009), Chapter 7
#
mb &lt;- smd2or(0.5, sqrt(0.0205), backtransf = FALSE)
# TE = log odds ratio; seTE = standard error of log odds ratio
data.frame(lnOR = round(mb$TE, 4), varlnOR = round(mb$seTE^2, 4))

# Use dataset from Fleiss (1993)
#
data(Fleiss1993cont)
m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, sm = "SMD",
               studlab = paste(study, year))
smd2or(m1)

</code></pre>

<hr>
<h2 id='smoking'>Smoking example</h2><span id='topic+smoking'></span><span id='topic+lungcancer'></span>

<h3>Description</h3>

<p>Meta-analyses on the effect of smoking on mortality risk.
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>study</em></b></td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>participants</em></b></td><td style="text-align: left;"> total number of participants </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>d.smokers</em></b></td><td style="text-align: left;"> number of deaths in smokers' group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>py.smokers</em></b></td><td style="text-align: left;"> person years at risk in smokers' group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>d.nonsmokers</em></b></td><td style="text-align: left;"> number of deaths in non-smokers'
  group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>py.nonsmokers</em></b></td><td style="text-align: left;"> person years at risk in
  non-smokers' group
</td>
</tr>

</table>



<h3>Details</h3>

<p>Data have been reconstructed based on the famous Smoking and Health
Report to the Surgeon General (Bayne-Jones S et al., 1964). Data
sets can be used to evaluate the risk of smoking on overall
mortality (dataset <code>smoking</code>) and lung-cancer deaths (dataset
<code>lungcancer</code>), respectively.
</p>
<p>The person time is attributed such that the rate ratios are equal
to the reported mortality ratios implicitly assuming that the data
have arisen from a homogeneous age group; more detailed information
by age is not available from the report. Note, the group of
&quot;non-smokers&quot; actually consists of all participants except those
who are smokers of cigarettes only. Information on real non-smokers
is not available from the published Smoking and Health Report.
</p>


<h3>Source</h3>

<p>Bayne-Jones S et al. (1964):
Smoking and Health: Report of the Advisory Committee to the Surgeon
General of the United States.
U-23 Department of Health, Education, and Welfare.
Public Health Service Publication No. 1103.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metainc">metainc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(smoking)

m1 &lt;- metainc(d.smokers, py.smokers, d.nonsmokers, py.nonsmokers,
  data = smoking, studlab = study)
print(m1, digits = 2)

data(lungcancer)

m2 &lt;- metainc(d.smokers, py.smokers, d.nonsmokers, py.nonsmokers,
  data = lungcancer, studlab = study)
print(m2, digits = 2)
</code></pre>

<hr>
<h2 id='subset.longarm'>Return subset of longarm object</h2><span id='topic+subset.longarm'></span>

<h3>Description</h3>

<p>The <code>subset</code> method returns a subset of a longarm object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'longarm'
subset(x, subset, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset.longarm_+3A_x">x</code></td>
<td>
<p>An object of class <code>longarm</code>.</p>
</td></tr>
<tr><td><code id="subset.longarm_+3A_subset">subset</code></td>
<td>
<p>A logical expression indicating elements or rows to
keep: missing values are taken as false.</p>
</td></tr>
<tr><td><code id="subset.longarm_+3A_...">...</code></td>
<td>
<p>Additional arguments.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="#topic+longarm">longarm</a></code> object is returned.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+longarm">longarm</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Artificial example with three studies
m &lt;- metabin(1:3, 100:102, 4:6, 200:202, studlab = LETTERS[1:3])
# Transform data to long arm-based format
l1 &lt;- longarm(m)
l1

# Subset without Study B
subset(l1, studlab != "B")

</code></pre>

<hr>
<h2 id='summary.meta'>Summary of meta-analysis results</h2><span id='topic+summary.meta'></span>

<h3>Description</h3>

<p>Summary method for objects of class <code>meta</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
summary(object, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.meta_+3A_object">object</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="summary.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Summary method for objects of class <code>meta</code>.
</p>


<h3>Value</h3>

<p>An object of classes <code>summary.meta</code> and <code>meta</code> (see
<code><a href="#topic+meta-object">meta-object</a></code>.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Cooper H &amp; Hedges LV (1994):
<em>The Handbook of Research Synthesis</em>.
Newbury Park, CA: Russell Sage Foundation
</p>
<p>Crippa A, Khudyakov P, Wang M, Orsini N, Spiegelman D (2016):
A new measure of between-studies heterogeneity in meta-analysis.
<em>Statistics in Medicine</em>,
<b>35</b>, 3661&ndash;75
</p>
<p>Higgins JPT &amp; Thompson SG (2002):
Quantifying heterogeneity in a meta-analysis.
<em>Statistics in Medicine</em>,
<b>21</b>, 1539&ndash;58
</p>


<h3>See Also</h3>

<p><code><a href="#topic+print.summary.meta">print.summary.meta</a></code>, <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metacont">metacont</a></code>, <code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, studlab = paste(study, year), sm = "SMD")
summary(m1)

summary(update(m1, subgroup = c(1, 2, 1, 1, 2), subgroup.name = "group"))
forest(update(m1, subgroup = c(1, 2, 1, 1, 2), subgroup.name = "group"))

## Not run: 
# Use unicode characters to print tau^2, tau, and I^2
print(summary(m1),
  text.tau2 = "\u03c4\u00b2", text.tau = "\u03c4", text.I2 = "I\u00b2")

## End(Not run)

</code></pre>

<hr>
<h2 id='summary.rm5'>Cochrane review: detailed summary of meta-analyses</h2><span id='topic+summary.rm5'></span><span id='topic+summary.cdir'></span><span id='topic+print.summary.rm5'></span><span id='topic+print.summary.cdir'></span>

<h3>Description</h3>

<p>Calculate and print a detailed summary of all meta-analyses in a
Cochrane review.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rm5'
summary(object, comp.no, outcome.no, ...)

## S3 method for class 'cdir'
summary(object, comp.no, outcome.no, ...)

## S3 method for class 'summary.rm5'
print(x, ...)

## S3 method for class 'summary.cdir'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary.rm5_+3A_object">object</code></td>
<td>
<p>An object of class <code>rm5</code> or <code>cdir</code>.</p>
</td></tr>
<tr><td><code id="summary.rm5_+3A_comp.no">comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code id="summary.rm5_+3A_outcome.no">outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code id="summary.rm5_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to <code>metacr</code>).</p>
</td></tr>
<tr><td><code id="summary.rm5_+3A_x">x</code></td>
<td>
<p>An object of class <code>summary.rm5</code> or
<code>summary.cdir</code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to rerun all or selected meta-analyses of
a Cochrane Review of interventions (Higgins et al, 2023).
</p>
<p>The R function <code><a href="#topic+metacr">metacr</a></code> is called internally.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Higgins JPT, Thomas J, Chandler J, Cumpston M, Li T, Page MJ, Welch
VA (editors) (2023):
<em>Cochrane Handbook for Systematic Reviews of Interventions
Version 6.4</em>.
Available from <a href="https://training.cochrane.org/handbook">https://training.cochrane.org/handbook</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.meta">summary.meta</a></code>, <code><a href="#topic+metacr">metacr</a></code>,
<code><a href="#topic+read.rm5">read.rm5</a></code>, <code><a href="#topic+read.cdir">read.cdir</a></code>,
<code><a href="#topic+metabias.rm5">metabias.rm5</a></code>, <code><a href="#topic+metabias.cdir">metabias.cdir</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate export data file "Fleiss1993_CR.csv"
# in sub-directory of package "meta"
#
filename &lt;- system.file("extdata", "Fleiss1993_CR.csv", package = "meta")
Fleiss1993_CR &lt;- read.rm5(filename)

# Print summary results for all meta-analysis
#
summary(Fleiss1993_CR)

# Print summary results only for second outcome of first comparison
#
summary(Fleiss1993_CR, comp.no = 1, outcome.no = 2)

</code></pre>

<hr>
<h2 id='traffic_light'>Produce traffic light plot of risk of bias assessment</h2><span id='topic+traffic_light'></span>

<h3>Description</h3>

<p>Produce traffic light plot of risk of bias assessment
</p>


<h3>Usage</h3>

<pre><code class='language-R'>traffic_light(object, colour = "cochrane", psize = 15, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="traffic_light_+3A_object">object</code></td>
<td>
<p>An object of class <code>rob</code>.</p>
</td></tr>
<tr><td><code id="traffic_light_+3A_colour">colour</code></td>
<td>
<p>Specify colour scheme for the traffic light plot; see
<code><a href="robvis.html#topic+rob_summary">rob_summary</a></code>.</p>
</td></tr>
<tr><td><code id="traffic_light_+3A_psize">psize</code></td>
<td>
<p>Size of the traffic lights.</p>
</td></tr>
<tr><td><code id="traffic_light_+3A_quiet">quiet</code></td>
<td>
<p>A logical to suppress the display of the traffic light
plot.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a wrapper function for
<code><a href="robvis.html#topic+rob_traffic_light">rob_traffic_light</a></code> of R package <b>robvis</b>
to produce a traffic light plot of risk of bias assessment.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rob">rob</a></code>, <code><a href="#topic+barplot.rob">barplot.rob</a></code>,
<code><a href="robvis.html#topic+rob_traffic_light">rob_traffic_light</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Use RevMan 5 settings
oldset &lt;- settings.meta("RevMan5")

data(caffeine)

m1 &lt;- metabin(h.caf, n.caf, h.decaf, n.decaf, sm = "OR",
  data = caffeine, studlab = paste(study, year))

# Add risk of bias assessment to meta-analysis
m2 &lt;- rob(D1, D2, D3, D4, D5, overall = rob, data = m1, tool = "rob2")

# Print risk of bias assessment
rob(m2)

## Not run: 
# Traffic light plot (if R package robvis has been installed)
traffic_light(rob(m2))

## End(Not run)

# Use previous settings
settings.meta(oldset)

</code></pre>

<hr>
<h2 id='trimfill.meta'>Trim-and-fill method to adjust for bias in meta-analysis</h2><span id='topic+trimfill.meta'></span><span id='topic+trimfill'></span><span id='topic+trimfill.default'></span>

<h3>Description</h3>

<p>Trim-and-fill method for estimating and adjusting for the number
and outcomes of missing studies in a meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
trimfill(
  x,
  left = NULL,
  ma.common = TRUE,
  type = "L",
  n.iter.max = 50,
  common = FALSE,
  random = TRUE,
  prediction = x$prediction,
  backtransf = x$backtransf,
  pscale = x$pscale,
  irscale = x$irscale,
  irunit = x$irunit,
  silent = TRUE,
  warn.deprecated = gs("warn.deprecated"),
  ...
)

## Default S3 method:
trimfill(
  x,
  seTE,
  left = NULL,
  ma.common = TRUE,
  type = "L",
  n.iter.max = 50,
  sm = "",
  studlab = NULL,
  level = 0.95,
  level.ma = level,
  common = FALSE,
  random = TRUE,
  method.random.ci = gs("method.random.ci"),
  adhoc.hakn.ci = gs("adhoc.hakn.ci"),
  method.tau = gs("method.tau"),
  method.tau.ci = if (method.tau == "DL") "J" else "QP",
  prediction = FALSE,
  level.predict = level,
  method.predict = gs("method.predict"),
  adhoc.hakn.pi = gs("adhoc.hakn.pi"),
  seed.predict = NULL,
  backtransf = TRUE,
  pscale = 1,
  irscale = 1,
  irunit = "person-years",
  silent = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trimfill.meta_+3A_x">x</code></td>
<td>
<p>An object of class <code>meta</code>, or estimated treatment
effect in individual studies.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_left">left</code></td>
<td>
<p>A logical indicating whether studies are supposed to be
missing on the left or right side of the funnel plot. If NULL,
the linear regression test for funnel plot symmetry (i.e.,
function <code>metabias(..., method="Egger")</code>) is used to
determine whether studies are missing on the left or right side.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_ma.common">ma.common</code></td>
<td>
<p>A logical indicating whether a common effect or
random effects model is used to estimate the number of missing
studies.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_type">type</code></td>
<td>
<p>A character indicating which method is used to estimate
the number of missing studies. Either <code>"L"</code> or <code>"R"</code>.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_n.iter.max">n.iter.max</code></td>
<td>
<p>Maximum number of iterations to estimate number
of missing studies.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots. If
<code>backtransf=TRUE</code>, results for <code>sm="OR"</code> are printed as
odds ratios rather than log odds ratios and results for
<code>sm="ZCOR"</code> are printed as correlations rather than Fisher's
z transformed correlations, for example.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
single event probabilities or risk differences, i.e. if argument
<code>sm</code> is equal to <code>"PLOGIT"</code>, <code>"PLN"</code>,
<code>"PRAW"</code>, <code>"PAS"</code>, <code>"PFT"</code>, or <code>"RD"</code>.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single incidence rates or incidence rate differences, i.e. if
argument <code>sm</code> is equal to <code>"IR"</code>, <code>"IRLN"</code>,
<code>"IRS"</code>, <code>"IRFT"</code>, or <code>"IRD"</code>.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_irunit">irunit</code></td>
<td>
<p>A character specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_silent">silent</code></td>
<td>
<p>A logical indicating whether basic information on
iterations shown.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_sete">seTE</code></td>
<td>
<p>Standard error of estimated treatment effect.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_sm">sm</code></td>
<td>
<p>An optional character string indicating underlying
summary measure, e.g., <code>"RD"</code>, <code>"RR"</code>, <code>"OR"</code>,
<code>"ASD"</code>, <code>"HR"</code>, <code>"MD"</code>, <code>"SMD"</code>, or
<code>"ROM"</code>; ignored if <code>x</code> is of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_studlab">studlab</code></td>
<td>
<p>An optional vector with study labels; ignored if
<code>x</code> is of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies. If existing, <code>x$level</code> is used as value
for <code>level</code>; otherwise 0.95 is used.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence interval for
the pooled estimate. If existing, <code>x$level.ma</code> is used as
value for <code>level.ma</code>; otherwise 0.95 is used.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for the
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="trimfill.meta_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The trim-and-fill method (Duval, Tweedie 2000a, 2000b) can be used
for estimating and adjusting for the number and outcomes of missing
studies in a meta-analysis. The method relies on scrutiny of one
side of a funnel plot for asymmetry assumed due to publication
bias.
</p>
<p>Three different methods have been proposed originally to estimate
the number of missing studies. Two of these methods (L- and
R-estimator) have been shown to perform better in simulations, and
are available in this R function (argument <code>type</code>).
</p>
<p>A common effect or random effects model can be used to estimate the
number of missing studies (argument <code>ma.common</code>). Furthermore,
a common effect and/or random effects model can be used to
summaries study results (arguments <code>common</code> and
<code>random</code>). Simulation results (Peters et al. 2007) indicate
that the common-random model, i.e. using a common effect model to
estimate the number of missing studies and a random effects model
to summaries results, (i) performs better than the common-common
model, and (ii) performs no worse than and marginally better in
certain situations than the random-random model. Accordingly, the
common-random model is the default.
</p>
<p>An empirical comparison of the trim-and-fill method and the Copas
selection model (Schwarzer et al. 2010) indicates that the
trim-and-fill method leads to excessively conservative inference in
practice. The Copas selection model is available in R package
<b>metasens</b>.
</p>
<p>The function <code><a href="#topic+metagen">metagen</a></code> is called internally.
</p>


<h3>Value</h3>

<p>An object of class <code>c("trimfill", "metagen", "meta")</code> with
corresponding generic functions (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Duval S &amp; Tweedie R (2000a):
A nonparametric &quot;Trim and Fill&quot; method of accounting for
publication bias in meta-analysis.
<em>Journal of the American Statistical Association</em>,
<b>95</b>, 89&ndash;98
</p>
<p>Duval S &amp; Tweedie R (2000b):
Trim and Fill: A simple funnel-plot-based method of testing and
adjusting for publication bias in meta-analysis.
<em>Biometrics</em>,
<b>56</b>, 455&ndash;63
</p>
<p>Peters JL, Sutton AJ, Jones DR, Abrams KR, Rushton L (2007):
Performance of the trim and fill method in the presence of
publication bias and between-study heterogeneity.
<em>Statisics in Medicine</em>,
<b>10</b>, 4544&ndash;62
</p>
<p>Schwarzer G, Carpenter J, Rücker G (2010):
Empirical evaluation suggests Copas selection model preferable to
trim-and-fill method for selection bias in meta-analysis
<em>Journal of Clinical Epidemiology</em>,
<b>63</b>, 282&ndash;8
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metabias">metabias</a></code>,
<code><a href="#topic+funnel">funnel</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993bin)
m1 &lt;- metabin(d.asp, n.asp, d.plac, n.plac, data = Fleiss1993bin, sm = "OR")
tf1 &lt;- trimfill(m1)
tf1
funnel(tf1)
funnel(tf1, pch = ifelse(tf1$trimfill, 1, 16), level = 0.9, random = FALSE)
#
# Use log odds ratios on x-axis
#
funnel(tf1, backtransf = FALSE)
funnel(tf1, pch = ifelse(tf1$trimfill, 1, 16), level = 0.9, random = FALSE,
  backtransf = FALSE)

trimfill(m1$TE, m1$seTE, sm = m1$sm)

</code></pre>

<hr>
<h2 id='trimfill.rm5'>Cochrane review: trim-and-fill method</h2><span id='topic+trimfill.rm5'></span><span id='topic+trimfill.cdir'></span><span id='topic+print.trimfill.rm5'></span><span id='topic+print.trimfill.cdir'></span>

<h3>Description</h3>

<p>Conduct trim-and-fill analysis for all meta-analyses in a Cochrane
review.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'rm5'
trimfill(x, comp.no, outcome.no, ...)

## S3 method for class 'cdir'
trimfill(x, comp.no, outcome.no, ...)

## S3 method for class 'trimfill.rm5'
print(x, ...)

## S3 method for class 'trimfill.cdir'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="trimfill.rm5_+3A_x">x</code></td>
<td>
<p>An object of class <code>rm5</code>, <code>trimfill.rm5</code>,
<code>cdir</code> or <code>trimfill.cdir</code>.</p>
</td></tr>
<tr><td><code id="trimfill.rm5_+3A_comp.no">comp.no</code></td>
<td>
<p>Comparison number.</p>
</td></tr>
<tr><td><code id="trimfill.rm5_+3A_outcome.no">outcome.no</code></td>
<td>
<p>Outcome number.</p>
</td></tr>
<tr><td><code id="trimfill.rm5_+3A_...">...</code></td>
<td>
<p>Additional arguments (passed on to <code>metacr</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function can be used to conduct a trim-and-fill analysis for
all or selected meta-analyses in a Cochrane review of intervention
studies (Higgins et al, 2023).
</p>
<p>The R function <code><a href="#topic+metacr">metacr</a></code> is called internally.
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>References</h3>

<p>Higgins JPT, Thomas J, Chandler J, Cumpston M, Li T, Page MJ, Welch
VA (editors) (2023):
<em>Cochrane Handbook for Systematic Reviews of Interventions
Version 6.4</em>.
Available from <a href="https://training.cochrane.org/handbook">https://training.cochrane.org/handbook</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+summary.meta">summary.meta</a></code>, <code><a href="#topic+metacr">metacr</a></code>,
<code><a href="#topic+read.rm5">read.rm5</a></code>, <code><a href="#topic+read.cdir">read.cdir</a></code>,
<code><a href="#topic+metabias.rm5">metabias.rm5</a></code>, <code><a href="#topic+metabias.cdir">metabias.cdir</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Locate export data file "Fleiss1993_CR.csv"
# in sub-directory of package "meta"
#
filename &lt;- system.file("extdata", "Fleiss1993_CR.csv", package = "meta")
Fleiss1993_CR &lt;- read.rm5(filename)

# Conduct trim-and-fill analysis
#
trimfill(Fleiss1993_CR)

# Conduct trim-and-fill analysis only for second outcome of first
# comparison
#
trimfill(Fleiss1993_CR, comp.no = 1, outcome.no = 2)

</code></pre>

<hr>
<h2 id='update.meta'>Update a meta-analysis object</h2><span id='topic+update.meta'></span>

<h3>Description</h3>

<p>Update an existing meta-analysis object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
update(
  object,
  data = object$data,
  subset,
  studlab,
  exclude,
  cluster,
  rho = object$rho,
  method,
  sm = object$sm,
  incr,
  method.incr = object$method.incr,
  allstudies = object$allstudies,
  MH.exact = object$MH.exact,
  RR.Cochrane = object$RR.Cochrane,
  Q.Cochrane = object$Q.Cochrane,
  model.glmm = object$model.glmm,
  level = object$level,
  level.ma = object$level.ma,
  common = object$common,
  random = object$random,
  overall = object$overall,
  overall.hetstat = object$overall.hetstat,
  method.random.ci = object$method.random.ci,
  adhoc.hakn.ci = object$adhoc.hakn.ci,
  method.predict = object$method.predict,
  adhoc.hakn.pi = object$adhoc.hakn.pi,
  seed.predict = object$seed.predict,
  method.tau = object$method.tau,
  method.tau.ci = object$method.tau.ci,
  tau.preset = object$tau.preset,
  TE.tau = object$TE.tau,
  tau.common = object$tau.common,
  prediction = object$prediction,
  level.predict = object$level.predict,
  null.effect = object$null.effect,
  method.bias = object$method.bias,
  backtransf = object$backtransf,
  pscale = object$pscale,
  irscale = object$irscale,
  irunit = object$irunit,
  text.common = object$text.common,
  text.random = object$text.random,
  text.predict = object$text.predict,
  text.w.common = object$text.w.common,
  text.w.random = object$text.w.random,
  title = object$title,
  complab = object$complab,
  outclab = object$outclab,
  label.e = object$label.e,
  label.c = object$label.c,
  label.left = object$label.left,
  label.right = object$label.right,
  n.e = object$n.e,
  n.c = object$n.c,
  method.mean = object$method.mean,
  method.sd = object$method.sd,
  approx.mean.e = object$approx.mean.e,
  approx.mean.c = object$approx.mean.c,
  approx.sd.e = object$approx.sd.e,
  approx.sd.c = object$approx.sd.c,
  approx.mean = object$approx.mean,
  approx.sd = object$approx.sd,
  approx.TE = object$approx.TE,
  approx.seTE = object$approx.seTE,
  pooledvar = object$pooledvar,
  method.smd = object$method.smd,
  sd.glass = object$sd.glass,
  exact.smd = object$exact.smd,
  method.ci = object$method.ci,
  subgroup,
  subgroup.name = object$subgroup.name,
  print.subgroup.name = object$print.subgroup.name,
  sep.subgroup = object$sep.subgroup,
  test.subgroup = object$test.subgroup,
  prediction.subgroup = object$prediction.subgroup,
  seed.predict.subgroup = object$seed.predict.subgroup,
  byvar,
  id,
  print.CMH = object$print.CMH,
  keepdata = TRUE,
  left = object$left,
  ma.common = object$ma.common,
  type = object$type,
  n.iter.max = object$n.iter.max,
  warn = FALSE,
  warn.deprecated = gs("warn.deprecated"),
  verbose = FALSE,
  control = object$control,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="update.meta_+3A_object">object</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_data">data</code></td>
<td>
<p>Dataset.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_subset">subset</code></td>
<td>
<p>Subset.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_studlab">studlab</code></td>
<td>
<p>Study label.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_exclude">exclude</code></td>
<td>
<p>An optional vector specifying studies to exclude
from meta-analysis, however, to include in printouts and forest
plots.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_cluster">cluster</code></td>
<td>
<p>An optional vector specifying which estimates come
from the same cluster resulting in the use of a three-level
meta-analysis model.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_rho">rho</code></td>
<td>
<p>Assumed correlation of estimates within a cluster.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method">method</code></td>
<td>
<p>A character string indicating which method is to be
used for pooling of studies (see <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metaprop">metaprop</a></code> and
<code><a href="#topic+metarate">metarate</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_sm">sm</code></td>
<td>
<p>A character string indicating which summary measure is
used for pooling.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_incr">incr</code></td>
<td>
<p>Information on increment added to cell frequencies of
studies with zero cell counts (see <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metaprop">metaprop</a></code> and
<code><a href="#topic+metarate">metarate</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.incr">method.incr</code></td>
<td>
<p>A character string indicating which continuity
correction method should be used (see <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metaprop">metaprop</a></code> and
<code><a href="#topic+metarate">metarate</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_allstudies">allstudies</code></td>
<td>
<p>A logical indicating if studies with zero or all
events in both groups are to be included in the meta-analysis
(applies only to <code><a href="#topic+metabin">metabin</a></code> object with <code>sm</code>
equal to <code>"RR"</code> or <code>"OR"</code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_mh.exact">MH.exact</code></td>
<td>
<p>A logical indicating if <code>incr</code> is not to be
added to all cell frequencies for studies with a zero cell count
to calculate the pooled estimate based on the Mantel-Haenszel
method (applies only to <code><a href="#topic+metabin">metabin</a></code> object).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_rr.cochrane">RR.Cochrane</code></td>
<td>
<p>A logical indicating which method to use as
continuity correction for the risk ratio (see
<code><a href="#topic+metabin">metabin</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_q.cochrane">Q.Cochrane</code></td>
<td>
<p>A logical indicating which method to use to
calculate the heterogeneity statistic Q (see
<code><a href="#topic+metabin">metabin</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_model.glmm">model.glmm</code></td>
<td>
<p>A character string indicating which GLMM model
should be used (see <code><a href="#topic+metabin">metabin</a></code> and
<code><a href="#topic+metainc">metainc</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_level">level</code></td>
<td>
<p>The level used to calculate confidence intervals for
individual studies.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_level.ma">level.ma</code></td>
<td>
<p>The level used to calculate confidence intervals
for meta-analysis estimates.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether a common effect
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether a random effects
meta-analysis should be conducted.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_overall">overall</code></td>
<td>
<p>A logical indicating whether overall summaries
should be reported. This argument is useful in a meta-analysis
with subgroups if overall results should not be reported.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_overall.hetstat">overall.hetstat</code></td>
<td>
<p>A logical value indicating whether to print
heterogeneity measures for overall treatment comparisons. This
argument is useful in a meta-analysis with subgroups if
heterogeneity statistics should only be printed on subgroup
level.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.random.ci">method.random.ci</code></td>
<td>
<p>A character string indicating which method
is used to calculate confidence interval and test statistic for
random effects estimate (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_adhoc.hakn.ci">adhoc.hakn.ci</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied in the case
of an arbitrarily small Hartung-Knapp variance estimate (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.predict">method.predict</code></td>
<td>
<p>A character string indicating which method is
used to calculate a prediction interval (see
<code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_adhoc.hakn.pi">adhoc.hakn.pi</code></td>
<td>
<p>A character string indicating whether an
<em>ad hoc</em> variance correction should be applied for
prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_seed.predict">seed.predict</code></td>
<td>
<p>A numeric value used as seed to calculate
bootstrap prediction interval (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.tau">method.tau</code></td>
<td>
<p>A character string indicating which method is
used to estimate the between-study variance <code class="reqn">\tau^2</code> and its
square root <code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.tau.ci">method.tau.ci</code></td>
<td>
<p>A character string indicating which method is
used to estimate the confidence interval of <code class="reqn">\tau^2</code> and
<code class="reqn">\tau</code> (see <code><a href="#topic+meta-package">meta-package</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_tau.preset">tau.preset</code></td>
<td>
<p>Prespecified value for the square root of the
between-study variance <code class="reqn">\tau^2</code>.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_te.tau">TE.tau</code></td>
<td>
<p>Overall treatment effect used to estimate the
between-study variance tau-squared.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_tau.common">tau.common</code></td>
<td>
<p>A logical indicating whether tau-squared should
be the same across subgroups.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_prediction">prediction</code></td>
<td>
<p>A logical indicating whether a prediction
interval should be printed.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_level.predict">level.predict</code></td>
<td>
<p>The level used to calculate prediction
interval for a new study.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_null.effect">null.effect</code></td>
<td>
<p>A numeric value specifying the effect under the
null hypothesis.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.bias">method.bias</code></td>
<td>
<p>A character string indicating which test for
funnel plot asymmetry is to be used, can be abbreviated. See
function <code><a href="#topic+metabias">metabias</a>.</code></p>
</td></tr>
<tr><td><code id="update.meta_+3A_backtransf">backtransf</code></td>
<td>
<p>A logical indicating whether results should be
back transformed in printouts and plots. If <code>backtransf =
TRUE</code>, results for <code>sm = "OR"</code> are printed as odds ratios
rather than log odds ratios and results for <code>sm = "ZCOR"</code>
are printed as correlations rather than Fisher's z transformed
correlations, for example.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_pscale">pscale</code></td>
<td>
<p>A numeric giving scaling factor for printing of
single event probabilities or risk differences, i.e. if argument
<code>sm</code> is equal to <code>"PLOGIT"</code>, <code>"PLN"</code>,
<code>"PRAW"</code>, <code>"PAS"</code>, <code>"PFT"</code>, or <code>"RD"</code>.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_irscale">irscale</code></td>
<td>
<p>A numeric defining a scaling factor for printing of
single incidence rates or incidence rate differences, i.e. if
argument <code>sm</code> is equal to <code>"IR"</code>, <code>"IRLN"</code>,
<code>"IRS"</code>, <code>"IRFT"</code>, or <code>"IRD"</code>.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_irunit">irunit</code></td>
<td>
<p>A character specifying the time unit used to
calculate rates, e.g. person-years.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_text.common">text.common</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled common effect estimate.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_text.random">text.random</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the pooled random effects estimate.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_text.predict">text.predict</code></td>
<td>
<p>A character string used in printouts and forest
plot to label the prediction interval.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_text.w.common">text.w.common</code></td>
<td>
<p>A character string used to label weights of
common effect model.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_text.w.random">text.w.random</code></td>
<td>
<p>A character string used to label weights of
random effects model.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_title">title</code></td>
<td>
<p>Title of meta-analysis / systematic review.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_complab">complab</code></td>
<td>
<p>Comparison label.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_outclab">outclab</code></td>
<td>
<p>Outcome label.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_label.e">label.e</code></td>
<td>
<p>Label for experimental group.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_label.c">label.c</code></td>
<td>
<p>Label for control group.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_label.left">label.left</code></td>
<td>
<p>Graph label on left side of forest plot.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_label.right">label.right</code></td>
<td>
<p>Graph label on right side of forest plot.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_n.e">n.e</code></td>
<td>
<p>Number of observations in experimental group (only for
<code><a href="#topic+metagen">metagen</a></code> object).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_n.c">n.c</code></td>
<td>
<p>Number of observations in control group (only for
metagen object).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.mean">method.mean</code></td>
<td>
<p>A character string indicating which method to
use to approximate the mean from the median and other statistics
(see <code><a href="#topic+metacont">metacont</a></code> and <code><a href="#topic+metamean">metamean</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.sd">method.sd</code></td>
<td>
<p>A character string indicating which method to use
to approximate the standard deviation from sample size, median,
interquartile range and range (see <code><a href="#topic+metacont">metacont</a></code> and
<code><a href="#topic+metamean">metamean</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.mean.e">approx.mean.e</code></td>
<td>
<p>Approximation method to estimate means in
experimental group (see <code><a href="#topic+metacont">metacont</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.mean.c">approx.mean.c</code></td>
<td>
<p>Approximation method to estimate means in
control group (see <code><a href="#topic+metacont">metacont</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.sd.e">approx.sd.e</code></td>
<td>
<p>Approximation method to estimate standard
deviations in experimental group (see <code><a href="#topic+metacont">metacont</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.sd.c">approx.sd.c</code></td>
<td>
<p>Approximation method to estimate standard
deviations in control group (see <code><a href="#topic+metacont">metacont</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.mean">approx.mean</code></td>
<td>
<p>Approximation method to estimate means (see
<code><a href="#topic+metamean">metamean</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.sd">approx.sd</code></td>
<td>
<p>Approximation method to estimate standard
deviations (see <code><a href="#topic+metamean">metamean</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.te">approx.TE</code></td>
<td>
<p>Approximation method to estimate treatment
estimate (see <code><a href="#topic+metagen">metagen</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_approx.sete">approx.seTE</code></td>
<td>
<p>Approximation method to estimate standard error
(see <code><a href="#topic+metagen">metagen</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_pooledvar">pooledvar</code></td>
<td>
<p>A logical indicating if a pooled variance should
be used for the mean difference or ratio of means (see
<code><a href="#topic+metacont">metacont</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.smd">method.smd</code></td>
<td>
<p>A character string indicating which method is
used to estimate the standardised mean difference (see
<code><a href="#topic+metacont">metacont</a></code>).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_sd.glass">sd.glass</code></td>
<td>
<p>A character string indicating which standard
deviation is used in the denominator for Glass' method to
estimate the standardised mean difference (only for metacont
object with <code>sm = "SMD"</code>). Either <code>"control"</code> using the
standard deviation in the control group (<code>sd.c</code>) or
<code>"experimental"</code> using the standard deviation in the
experimental group (<code>sd.e</code>), can be abbreviated.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_exact.smd">exact.smd</code></td>
<td>
<p>A logical indicating whether exact formulae should
be used in estimation of the standardised mean difference and its
standard error.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_method.ci">method.ci</code></td>
<td>
<p>A character string indicating which method is used
to calculate confidence intervals for individual studies. Either
<code>"z"</code>, <code>"t"</code>, <code>"WS"</code>, <code>"WSCC"</code>, <code>"AC"</code>,
<code>"SA"</code>, <code>"SACC"</code>, <code>"NAsm"</code>, or <code>"Poisson"</code>,
can be abbreviated. See functions <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metaprop">metaprop</a></code> and <code><a href="#topic+metarate">metarate</a></code>.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_subgroup">subgroup</code></td>
<td>
<p>An optional vector to conduct a meta-analysis with
subgroups.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_subgroup.name">subgroup.name</code></td>
<td>
<p>A character string with a name for the
subgroup variable.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_print.subgroup.name">print.subgroup.name</code></td>
<td>
<p>A logical indicating whether the name of
the subgroup variable should be printed in front of the group
labels.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_sep.subgroup">sep.subgroup</code></td>
<td>
<p>A character string defining the separator
between name of subgroup variable and subgroup label.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_test.subgroup">test.subgroup</code></td>
<td>
<p>A logical value indicating whether to print
results of test for subgroup differences.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_prediction.subgroup">prediction.subgroup</code></td>
<td>
<p>A logical indicating whether prediction
intervals should be printed for subgroups.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_seed.predict.subgroup">seed.predict.subgroup</code></td>
<td>
<p>A numeric vector providing seeds to
calculate bootstrap prediction intervals within subgroups. Must
be of same length as the number of subgroups.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_byvar">byvar</code></td>
<td>
<p>Deprecated argument (replaced by 'subgroup').</p>
</td></tr>
<tr><td><code id="update.meta_+3A_id">id</code></td>
<td>
<p>Deprecated argument (replaced by 'cluster').</p>
</td></tr>
<tr><td><code id="update.meta_+3A_print.cmh">print.CMH</code></td>
<td>
<p>A logical indicating whether result of the
Cochran-Mantel-Haenszel test for overall effect should be
printed.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_keepdata">keepdata</code></td>
<td>
<p>A logical indicating whether original data (set)
should be kept in meta object.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_left">left</code></td>
<td>
<p>A logical indicating whether studies are supposed to be
missing on the left or right side of the funnel plot. If NULL,
the linear regression test for funnel plot symmetry (i.e.,
function <code>metabias(..., method = "linreg")</code>) is used to
determine whether studies are missing on the left or right side.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_ma.common">ma.common</code></td>
<td>
<p>A logical indicating whether a common effect or
random effects model is used to estimate the number of missing
studies.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_type">type</code></td>
<td>
<p>A character indicating which method is used to estimate
the number of missing studies. Either <code>"L"</code> or <code>"R"</code>.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_n.iter.max">n.iter.max</code></td>
<td>
<p>Maximum number of iterations to estimate number
of missing studies.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_warn">warn</code></td>
<td>
<p>A logical indicating whether warnings should be printed
(e.g., if <code>incr</code> is added to studies with zero cell
frequencies).</p>
</td></tr>
<tr><td><code id="update.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_verbose">verbose</code></td>
<td>
<p>A logical indicating whether to print information on
updates of older meta versions.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_control">control</code></td>
<td>
<p>An optional list to control the iterative process to
estimate the between-study variance <code class="reqn">\tau^2</code>. This argument
is passed on to <code><a href="metafor.html#topic+rma.uni">rma.uni</a></code> or
<code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code>, respectively.</p>
</td></tr>
<tr><td><code id="update.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (ignored at the moment).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Wrapper function to update an existing meta-analysis object which
was created with R function <code><a href="#topic+metabin">metabin</a></code>,
<code><a href="#topic+metacont">metacont</a></code>, <code><a href="#topic+metacor">metacor</a></code>,
<code><a href="#topic+metagen">metagen</a></code>, <code><a href="#topic+metainc">metainc</a></code>,
<code><a href="#topic+metamean">metamean</a></code>, <code><a href="#topic+metaprop">metaprop</a></code>, or
<code><a href="#topic+metarate">metarate</a></code>. More details on function arguments are
available in help files of respective R functions.
</p>
<p>This function can also be used for objects of class 'trimfill',
'metacum', and 'metainf'.
</p>


<h3>Value</h3>

<p>An object of class <code>"meta"</code> and <code>"metabin"</code>,
<code>"metacont"</code>, <code>"metacor"</code>, <code>"metainc"</code>,
<code>"metagen"</code>, <code>"metamean"</code>, <code>"metaprop"</code>, or
<code>"metarate"</code> (see <code><a href="#topic+meta-object">meta-object</a></code>).
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metacor">metacor</a></code>, <code><a href="#topic+metagen">metagen</a></code>,
<code><a href="#topic+metainc">metainc</a></code>, <code><a href="#topic+metamean">metamean</a></code>,
<code><a href="#topic+metaprop">metaprop</a></code>, <code><a href="#topic+metarate">metarate</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
m1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, studlab = paste(study, year), sm = "SMD")
m1

# Change summary measure (from 'SMD' to 'MD')
#
update(m1, sm = "MD")

# Restrict analysis to subset of studies
#
update(m1, subset = 1:2)

# Use different levels for confidence intervals
#
m2 &lt;- update(m1, level = 0.66, level.ma = 0.99)
print(m2, digits = 2)
forest(m2)

</code></pre>

<hr>
<h2 id='weights.meta'>Calculate absolute and percentage weights for meta-analysis</h2><span id='topic+weights.meta'></span>

<h3>Description</h3>

<p>This function returns a data frame containing information on
absolute and percentage weights of individual studies contributing
to common effect and random effects meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'meta'
weights(
  object,
  common = object$common,
  random = object$random,
  warn.deprecated = gs("warn.deprecated"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="weights.meta_+3A_object">object</code></td>
<td>
<p>An object of class <code>meta</code>.</p>
</td></tr>
<tr><td><code id="weights.meta_+3A_common">common</code></td>
<td>
<p>A logical indicating whether absolute and
percentage weights from the common effect model should be
calculated.</p>
</td></tr>
<tr><td><code id="weights.meta_+3A_random">random</code></td>
<td>
<p>A logical indicating whether absolute and
percentage weights from the random effects model should be
calculated.</p>
</td></tr>
<tr><td><code id="weights.meta_+3A_warn.deprecated">warn.deprecated</code></td>
<td>
<p>A logical indicating whether warnings should
be printed if deprecated arguments are used.</p>
</td></tr>
<tr><td><code id="weights.meta_+3A_...">...</code></td>
<td>
<p>Additional arguments (to catch deprecated arguments).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the following variables is returned:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Variable</b> </td><td style="text-align: left;"> <b>Definition</b> </td><td style="text-align: left;"> <b>Condition</b> </td>
</tr>
<tr>
 <td style="text-align: left;">
w.common </td><td style="text-align: left;"> absolute weights in common effect model </td><td style="text-align: left;"> (if
  <code>common = TRUE</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
p.common </td><td style="text-align: left;"> percentage weights in common effect model </td><td style="text-align: left;"> (if
  <code>common = TRUE</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
w.random </td><td style="text-align: left;"> absolute weights in random effects model </td><td style="text-align: left;"> (if
  <code>random = TRUE</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
p.random </td><td style="text-align: left;"> percentage weights in random effects model </td><td style="text-align: left;"> (if
  <code>random = TRUE</code>)
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Guido Schwarzer <a href="mailto:guido.schwarzer@uniklinik-freiburg.de">guido.schwarzer@uniklinik-freiburg.de</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+metabin">metabin</a></code>, <code><a href="#topic+metacont">metacont</a></code>,
<code><a href="#topic+metagen">metagen</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Fleiss1993cont)
# Do meta-analysis (common effect and random effects model)
#
meta1 &lt;- metacont(n.psyc, mean.psyc, sd.psyc, n.cont, mean.cont, sd.cont,
  data = Fleiss1993cont, studlab = paste(study, year), sm = "SMD")

# Print weights for common effect and random effects meta-analysis
#
weights(meta1)

# Do meta-analysis (only random effects model)
#
meta2 &lt;- update(meta1, common = FALSE)

# Print weights for random effects meta-analysis
#
weights(meta2)

# Print weights for common effect and random effects meta-analysis
#
weights(meta2, common = TRUE)

</code></pre>

<hr>
<h2 id='woodyplants'>Elevated CO_2 and total biomass of woody plants</h2><span id='topic+woodyplants'></span>

<h3>Description</h3>

<p>Meta-analysis on effects of elevated CO_2 on total biomass of woody
plants
</p>
<p>This dataset has been used as an example in Hedges et al. (1999) to
describe methods for the meta-analysis of response ratios. The
complete dataset with 102 observations and 26 variables is
available online as a supplement. Here only a subset of 10
variables is provided and used in the examples.
</p>


<h3>Format</h3>

<p>A data frame with the following columns:
</p>

<table>
<tr>
 <td style="text-align: right;">
<b><em>obsno</em></b></td><td style="text-align: left;"> observation number </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>papno</em></b></td><td style="text-align: left;"> database paper number </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>treat</em></b></td><td style="text-align: left;"> treatment code </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>level</em></b></td><td style="text-align: left;"> treatment level </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.elev</em></b></td><td style="text-align: left;"> number of observations in experimental
  group (elevated CO_2-level) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean.elev</em></b></td><td style="text-align: left;"> estimated mean in experimental group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>sd.elev</em></b></td><td style="text-align: left;"> standard deviation in experimental group
  </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>n.amb</em></b></td><td style="text-align: left;"> number of observations in control group
  (ambient CO_2-level) </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>mean.amb</em></b></td><td style="text-align: left;"> estimated mean in control group </td>
</tr>
<tr>
 <td style="text-align: right;">
<b><em>sd.amb</em></b></td><td style="text-align: left;"> standard deviation in control group
</td>
</tr>

</table>



<h3>References</h3>

<p>Hedges LV, Gurevitch J, Curtis PS (1999):
The meta-analysis of response ratios in experimental ecology.
<em>Ecology</em>,
<b>80</b>, 1150&ndash;6
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(woodyplants)

# Meta-analysis of response ratios (Hedges et al., 1999)
#
m1 &lt;- metacont(n.elev, mean.elev, sd.elev, n.amb, mean.amb, sd.amb,
  data = woodyplants, sm = "ROM", studlab = paste(obsno, papno, sep = " / "))
print(m1, prediction = TRUE)

# Meta-analysis for plants grown with low soil fertility treatment
#
m2 &lt;- update(m1, subset = (treat == "fert" &amp; level == "low"))
print(m2, prediction = TRUE)

# Meta-analysis for plants grown under low light conditions
#
m3 &lt;- update(m1, subset = (treat == "light" &amp; level == "low"))
print(m3, prediction = TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
