<!DOCTYPE html><html lang="en"><head><title>Help for package metagear</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metagear}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#metagear-package'><p>Research synthesis tools to facilitate systematic reviews, data extraction,</p>
and meta-analysis.</a></li>
<li><a href='#abstract_screener'><p>A GUI screener to quickly code candidate studies for inclusion/exclusion into</p>
a systematic review or meta-analysis.</a></li>
<li><a href='#browse_DOI'><p>Opens a web page associated with a DOI (digital object identifier).</p></a></li>
<li><a href='#chachi'><p>A small tribute to Chachi.</p></a></li>
<li><a href='#covariance_commonControl'><p>Generates a sampling variance-covariance matrix for modeling dependencies</p>
among effect sizes due to sharing a common control.</a></li>
<li><a href='#effort_distribute'><p>Assigns title/abstract screening efforts to a team.</p></a></li>
<li><a href='#effort_initialize'><p>Formats a reference dataset for title/abstract screening efforts.</p></a></li>
<li><a href='#effort_merge'><p>Merges multiple files that had title/abstract screening efforts</p>
distributed across a team.</a></li>
<li><a href='#effort_redistribute'><p>Redistributes title/abstract screening efforts among a review team.</p></a></li>
<li><a href='#effort_summary'><p>Provides a text summary of screening efforts among the reviewing team.</p></a></li>
<li><a href='#example_references_metagear'><p>A collection of bibliographic references</p></a></li>
<li><a href='#figure_add'><p>Manually add/detect points to a scatter plot figure.</p></a></li>
<li><a href='#figure_barPlot'><p>Detect and display all bar plot objects.</p></a></li>
<li><a href='#figure_detectAllPoints'><p>Automated detection of plotted points from a scatter-plot figure image.</p></a></li>
<li><a href='#figure_detectAxis'><p>Detect an axis from a figure image.</p></a></li>
<li><a href='#figure_display'><p>Displays an image plot.</p></a></li>
<li><a href='#figure_displayDetectedPoints'><p>Displays detected points on figure.</p></a></li>
<li><a href='#figure_displayDetections'><p>Displays the detected figure objects.</p></a></li>
<li><a href='#figure_extractDetectedPoints'><p>Extracts data points from a detected image.</p></a></li>
<li><a href='#figure_read'><p>Reads/loads a figure image from file.</p></a></li>
<li><a href='#figure_removeOutlyingPoints'><p>Remove outlier points from a figure.</p></a></li>
<li><a href='#figure_scatterPlot'><p>Detect and display all scatter plot objects.</p></a></li>
<li><a href='#figure_splitPlot'><p>Splits a composite figure that contains multiple plots.</p></a></li>
<li><a href='#figure_transformByColors'><p>Transforms RBG figure into list of binary images.</p></a></li>
<li><a href='#figure_transformToBinary'><p>Transforms figure to binary image.</p></a></li>
<li><a href='#figure_write'><p>Saves/writes a figure image.</p></a></li>
<li><a href='#impute_missingness'><p>Provides a summary of missingness in a dataset.</p></a></li>
<li><a href='#impute_SD'><p>Imputes missing standard deviations in a dataset.</p></a></li>
<li><a href='#isPDF'><p>Evaluates whether a file is a PDF document.</p></a></li>
<li><a href='#Kam_et_al_2003_Fig2.jpg'><p>An example image of a scatterplot figure</p></a></li>
<li><a href='#Kortum_and_Acymyan_2013_Fig4.jpg'><p>An example image of a bar plot figure</p></a></li>
<li><a href='#MA_effectsTable'><p>Generate an ANOVA-like effects table for a meta-analysis.</p></a></li>
<li><a href='#PDF_download'><p>Attempts to download a PDF using a DOI link.</p></a></li>
<li><a href='#PDF_extractImages'><p>Attempts to extract all images from a PDF</p></a></li>
<li><a href='#PDFs_collect'><p>Attempts to download PDFs from multiple DOI links.</p></a></li>
<li><a href='#plot_PRISMA'><p>Plots and creates a PRISMA flow diagram.</p></a></li>
<li><a href='#random_d'><p>Random generation of Hedges' d effect sizes.</p></a></li>
<li><a href='#random_missingness'><p>Random generation of missingness in a data frame.</p></a></li>
<li><a href='#random_N'><p>Random generation of sample sizes (N) for study outcomes.</p></a></li>
<li><a href='#random_OR'><p>Random generation of odds ratio (OR) effect sizes.</p></a></li>
<li><a href='#random_pairedN'><p>Random generation of paired sample sizes (N) for study outcomes.</p></a></li>
<li><a href='#random_r'><p>Random generation of correlation coefficients.</p></a></li>
<li><a href='#random_RR'><p>Random generation of log response ratio (RR) effect sizes.</p></a></li>
<li><a href='#replicate_MetaWin2.0'><p>Replicate meta-analysis results and summaries from MetaWin 2.0.</p></a></li>
<li><a href='#replicate_phyloMeta1.3'><p>Replicate phylogeneic meta-analysis results and summaries from phyloMeta 1.3.</p></a></li>
<li><a href='#scrape_bibliography'><p>Attempts to scrape/extract bibliographic data from Web of Science.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-02-15</td>
</tr>
<tr>
<td>Title:</td>
<td>Comprehensive Research Synthesis Tools for Systematic Reviews
and Meta-Analysis</td>
</tr>
<tr>
<td>Description:</td>
<td>Functionalities for facilitating systematic reviews, data
    extractions, and meta-analyses. It includes a GUI (graphical user interface)
    to help screen the abstracts and titles of bibliographic data; tools to assign
    screening effort across multiple collaborators/reviewers and to assess inter-
    reviewer reliability; tools to help automate the download and retrieval of
    journal PDF articles from online databases; figure and image extractions 
    from PDFs; web scraping of citations; automated and manual data extraction 
    from scatter-plot and bar-plot images; PRISMA (Preferred Reporting Items for
    Systematic Reviews and Meta-Analyses) flow diagrams; simple imputation tools
    to fill gaps in incomplete or missing study parameters; generation of random
    effects sizes for Hedges' d, log response ratio, odds ratio, and correlation
    coefficients for Monte Carlo experiments; covariance equations for modelling
    dependencies among multiple effect sizes (e.g., effect sizes with a common
    control); and finally summaries that replicate analyses and outputs from 
    widely used but no longer updated meta-analysis software (i.e., metawin).
	Funding for this package was supported by National Science Foundation (NSF) 
	grants DBI-1262545 and DEB-1451031. CITE: Lajeunesse, M.J. (2016) 
	Facilitating systematic reviews, data extraction and meta-analysis with the 
	metagear package for R. Methods in Ecology and Evolution 7, 323-330 
	&lt;<a href="https://doi.org/10.1111%2F2041-210X.12472">doi:10.1111/2041-210X.12472</a>&gt;.</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.2)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Matrix, MASS, metafor (&ge; 1.9-9), stringr</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Tcl/Tk toolkit (X11 Quarts for Mac)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>R.rsp</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="http://lajeunesse.myweb.usf.edu/">http://lajeunesse.myweb.usf.edu/</a> <a href="https://github.com/mjlajeunesse/">https://github.com/mjlajeunesse/</a>
<a href="https://www.youtube.com/c/LajeunesseLab/">https://www.youtube.com/c/LajeunesseLab/</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>EBImage, ape, hexView, RCurl, testthat, R.rsp</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-02-15 14:55:42 UTC; lajeunesse</td>
</tr>
<tr>
<td>Author:</td>
<td>Marc J. Lajeunesse
    <a href="https://orcid.org/0000-0002-9678-2080"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Marc J. Lajeunesse &lt;lajeunesse@usf.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-15 15:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='metagear-package'>Research synthesis tools to facilitate systematic reviews, data extraction, 
and meta-analysis.</h2><span id='topic+metagear-package'></span>

<h3>Description</h3>

<p><span class="pkg">metagear</span> is a comprehensive, multifunctional toolbox with capabilities 
aimed to cover much of the research synthesis taxonomy: from applying a 
systematic review approach to objectively assemble and screen the literature, 
to extracting data from studies, and to finally summarize and analyze these 
data with the statistics of meta-analysis.  More information about 
<span class="pkg">metagear</span> can be found at <a href="http://lajeunesse.myweb.usf.edu/">http://lajeunesse.myweb.usf.edu/</a>.
</p>


<h3>Details</h3>

<p><strong>What to cite?</strong><br /><br /> Lajeunesse, M.J. (2016) Facilitating 
systematic reviews, data extraction and meta-analysis with the metagear 
package for R. <em>Methods in Ecology and Evolution</em> 7: 323-330. [ download 
<a href="http://lajeunesse.myweb.usf.edu/papers/Lajeunesse_2016_Methods_in_Ecology_and_Evolution.pdf">here</a> ]
<br /><br /> <strong>Installation and Dependencies.</strong><br /><br /> <span class="pkg">metagear</span> has one 
external dependency that need to be installed and loaded prior to use in R.
This is the EBImage R package (Pau et al. 2010) available only from the 
Bioconductor repository: <a href="https://www.bioconductor.org/">https://www.bioconductor.org/</a>. 
<br /><br /> To properly install <span class="pkg">metagear</span>, start with the following
R script that loads the Bioconductor resources needed to install the EBImage 
(also accept all of its dependencies): <br /><br />
<code>install.packages("BiocManager");</code> <br /> <code>BiocManager::install("EBImage"))</code> 
<br /> <code>library(metagear)</code> <br /><br /> Finally for Mac OS users, installation 
is sometimes not straighforward as the abstract_screener() requires the
Tcl/Tk GUI toolkit to be installed. You can get this toolkit by making sure
the latest X11 application (xQuartz) is installed from here: 
<a href="https://www.xquartz.org/">https://www.xquartz.org/</a>.
</p>


<h3>Author(s)</h3>

<p>Marc J. Lajeunesse (University of South Florida, Tampa USA)
</p>


<h3>References</h3>

<p>Pau, G., Fuchs, F., Sklyar, O., Boutros, M. and Huber, W. (2010)
EBImage: an R package for image processing with applications to cellular
phenotypes. Bioinformatics 26: 979-981.
</p>

<hr>
<h2 id='abstract_screener'>A GUI screener to quickly code candidate studies for inclusion/exclusion into
a systematic review or meta-analysis.</h2><span id='topic+abstract_screener'></span>

<h3>Description</h3>

<p>A GUI screener to help scan and evaluate the title and abstract of studies to
be included in a systematic review or meta-analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>abstract_screener(
  file = file.choose(),
  aReviewer = NULL,
  reviewerColumnName = "REVIEWERS",
  unscreenedColumnName = "INCLUDE",
  unscreenedValue = "not vetted",
  abstractColumnName = "ABSTRACT",
  titleColumnName = "TITLE",
  browserSearch = "https://www.google.com/search?q=",
  fontSize = 13,
  windowWidth = 70,
  windowHeight = 16,
  theButtons = c("YES", "maybe", "NO"),
  keyBindingToButtons = c("y", "m", "n"),
  buttonSize = 10,
  highlightColor = "powderblue",
  highlightKeywords = NA
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="abstract_screener_+3A_file">file</code></td>
<td>
<p>The file name and location of a .csv file containing the
abstracts and titles.  The .csv file should have been initialized with
<code>effort_initialize</code> and populated with screeners (reviewers) using
<code>effort_distribute</code>.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_areviewer">aReviewer</code></td>
<td>
<p>The name (a string) of the reviewer to screen abstracts.
It is used when there are multiple reviewers assigned to screen abstracts.
The default column label is &quot;REVIEWERS&quot; as initialized with
<code>effort_distribute</code>.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_reviewercolumnname">reviewerColumnName</code></td>
<td>
<p>The name of the column heading in the .csv file
that contains the reviewer names that will screen abstracts.  The default
column label is &quot;REVIEWERS&quot;.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_unscreenedcolumnname">unscreenedColumnName</code></td>
<td>
<p>The name of the column heading in the .csv file
that contains the screening outcomes (i.e. vetting outcomes by a reviewer).
Unscreened references are by default labeled as &quot;not vetted&quot;.  The
reviewer then can code to &quot;YES&quot; (is a relevant study), &quot;NO&quot; is not relevant
and should be excluded, or &quot;MAYBE&quot; if the title/abstract is missing or
does not contains enough information to fully assess inclusivity.
The default label of this column is &quot;INCLUDE&quot;.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_unscreenedvalue">unscreenedValue</code></td>
<td>
<p>Changes the default coding (a string) of &quot;not vetted&quot;
that designates whether an abstract remains to be screened or vetted.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_abstractcolumnname">abstractColumnName</code></td>
<td>
<p>The name of the column heading in the .csv file
that contains the abstracts. The default label of this column is
&quot;ABSTRACT&quot;.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_titlecolumnname">titleColumnName</code></td>
<td>
<p>The name of the column heading in the .csv file
that contains the titles. The default label of this column is &quot;TITLE&quot;.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_browsersearch">browserSearch</code></td>
<td>
<p>Change the url for the browser title search; the
default is Google.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_fontsize">fontSize</code></td>
<td>
<p>Change the font gWidgets::size of the title and abstract text.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_windowwidth">windowWidth</code></td>
<td>
<p>Change the default width of the GUI window.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_windowheight">windowHeight</code></td>
<td>
<p>Change the default height of the GUI window.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_thebuttons">theButtons</code></td>
<td>
<p>A vector of coding buttons included on the screener. The
default is YES, maybe, and NO. Buttons can be removed as added by changing
this vector. For example, theButtons = c(&quot;YES&quot;, &quot;NO&quot;) to remove the
maybe-button, or theButtons = c(&quot;YES&quot;, &quot;maybe&quot;, NO&quot;, &quot;model&quot;) to add a
&quot;model&quot; button that tags studies specifically as &quot;model&quot;.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_keybindingtobuttons">keyBindingToButtons</code></td>
<td>
<p>A vector of specific keyboard bindings to buttons.
They are keyboard shortcuts to buttons and the default binding is y for
YES-button, m for maybe-button, and n for NO-button. If theButtons parameter
is modified then these keybindings should also be modified.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_buttonsize">buttonSize</code></td>
<td>
<p>Change the default gWidgets::size of buttons.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_highlightcolor">highlightColor</code></td>
<td>
<p>The color of keywords highlighted in title and abstract.
The default is blue, but for classic yellow use &quot;palegoldenrod&quot;.</p>
</td></tr>
<tr><td><code id="abstract_screener_+3A_highlightkeywords">highlightKeywords</code></td>
<td>
<p>A string or list of keywords that will be highlighted
in title and abstract.</p>
</td></tr>
</table>


<h3>Note</h3>

<p><strong>Installation and troubleshooting</strong><br /><br /> For Mac OS users,
installation is sometimes not straighforward as this screener requires the
Tcl/Tk GUI toolkit to be installed. You can get this toolkit by making sure
the latest X11 application (xQuartz) is installed, see here: 
<a href="https://www.xquartz.org/">https://www.xquartz.org/</a>. More information on
installation is found in <code>metagear</code>'s vignette.
<br /><br /> <strong>How to use the screener</strong> <br /><br /> The GUI itself will appear
as a single window with the first title/abstract listed in the .csv file.
If abstracts have already been screened/coded, it will begin at the
nearest reference labeled as &quot;not vetted&quot;. The SEARCH WEB button opens the
default browser and searches Google with the title of the reference. The
YES, MAYBE, NO buttons, which also have keyboard shortcuts y and n, are
used to code the inclusion/exclusion of the reference. Once clicked/coded
the next reference is loaded. The SAVE button is used to save the coding
progress of screening tasks. It will save coding progress directly to the
loaded .csv file. <strong>Closing the GUI, and not saving, will result in
the loss of screening efforts relative to last save.</strong>  <br /><br /> There is
also an ISSUE FIXES menu bar with quick corrections to screening errors.
These include ISSUE FIXES: REFRESH TITLE AND ABSTRACT TEXT which reloads
the text of the current abstract in case portions were deleted when
copying and pasting sections, ISSUE FIXES: STATUS OF CURRENT ABSTRACT
which provides information on whether or not the abstract was previously
screened, and ISSUE FIXES: RETURN TO PREVIOUS ABSTRACT that
backtracks to the previous abstract if a selection error occurred (note a
warning will appear of there is a change to its inclusion/exclusion
coding).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(example_references_metagear)
effort_distribute(example_references_metagear,
                  initialize = TRUE,
                  reviewers = "marc",
                  save_split = TRUE)
abstract_screener("effort_marc.csv",
                  aReviewer = "marc",
                  highlightKeywords = "and")

## End(Not run)

</code></pre>

<hr>
<h2 id='browse_DOI'>Opens a web page associated with a DOI (digital object identifier).</h2><span id='topic+browse_DOI'></span>

<h3>Description</h3>

<p>Uses the DOI name of a study reference to locate the e-journal website,
or reference/citation website in Web of Science, Google Scholar, or 
CrossRef.  Opens in default web-browser.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>browse_DOI(theDOI, host = "DOI")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="browse_DOI_+3A_thedoi">theDOI</code></td>
<td>
<p>A string that identifies an electronic document on the web.</p>
</td></tr>
<tr><td><code id="browse_DOI_+3A_host">host</code></td>
<td>
<p>A string that defines the domain link used to open the DOI.  The 
default, <code>"DOI"</code>, will open to the web page associated with 
the DOI (e.g., publisher website).  Other options include <code>"WOS"</code> that 
will open the DOI in Web of Science, <code>"GS"</code> in Google Scholar, and 
<code>"CRF"</code> in Crossref.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

browse_DOI("10.1086/603628")        

## End(Not run)

</code></pre>

<hr>
<h2 id='chachi'>A small tribute to Chachi.</h2><span id='topic+chachi'></span>

<h3>Description</h3>

<p>Rest easy little bud, 200?-2016.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chachi()
</code></pre>

<hr>
<h2 id='covariance_commonControl'>Generates a sampling variance-covariance matrix for modeling dependencies
among effect sizes due to sharing a common control.</h2><span id='topic+covariance_commonControl'></span>

<h3>Description</h3>

<p>Generates K by K sampling variance-covariance (VCV) matrix that models the
dependencies that arise due to using the same control group study parameters 
when estimating multiple effect sizes.  This VCV matrix can then be used in
meta-analysis.  Currently only supports VCV calculation for log response
ratios (see Lajeunesse 2011).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>covariance_commonControl(
  aDataFrame,
  control_ID,
  X_t,
  SD_t,
  N_t,
  X_c,
  SD_c,
  N_c,
  metric = "RR"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="covariance_commonControl_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data frame containing columns with all study parameters
used to estimate effect sizes (e.g., means, SD, N's for treatment and 
control groups).  Must also contain a column that codes which effect sizes
share a common control.  See example below.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_control_id">control_ID</code></td>
<td>
<p>Label of the column that codes groups of effect sizes that 
share the mean, SD, and N of a control group.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_x_t">X_t</code></td>
<td>
<p>Column label for the means of (t)reatment group used to estimate the
effect size.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_sd_t">SD_t</code></td>
<td>
<p>Column label for the standard deviations (SD) of the treatment
group used to estimate the effect size.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_n_t">N_t</code></td>
<td>
<p>Column label for the sample size (N) of the treatment group
used to estimate the effect size.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_x_c">X_c</code></td>
<td>
<p>Column label for the means of (c)ontrol group used to estimate the
effect size.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_sd_c">SD_c</code></td>
<td>
<p>Column label for the standard deviations (SD) of the control
group used to estimate the effect size.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_n_c">N_c</code></td>
<td>
<p>Column label for the sample size (N) of the control group
used to estimate the effect size.</p>
</td></tr>
<tr><td><code id="covariance_commonControl_+3A_metric">metric</code></td>
<td>
<p>Option to designate which effect size metric for which the 
common control VCV matrix is to be estimated.  Default is &quot;RR&quot; for log 
response ratio.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A K by K sampling variance-covariance matrix and a data frame aligned
with the block diagonal design of the sampling matrix.
</p>


<h3>Note</h3>

<p><strong>Response Ratio's (<code class="reqn">\mathit{RR}</code>) with a comon control group</strong><br /><br /> 
Following Lajeunesse (2011), when two (or more) reponse ratio 
(<code class="reqn">\mathit{RR}</code>) effect sizes share a common control mean 
(<code class="reqn">\bar{X}_C</code>), such as
<code class="reqn">\mathit{RR}_{A,C}=ln(\bar{X}_A/\bar{X}_C)</code> and 
<code class="reqn">\mathit{RR}_{B,C}=ln(\bar{X}_B/\bar{X}_C)</code>, then they share a 
sampling covariance of:
</p>
<p style="text-align: center;"><code class="reqn">cov(\mathit{\mathit{RR}}_{A,C},~\mathit{RR}_{B,C})=\frac{(\mathit{SD}_C)^2}{N_C\bar{X}_C^2},</code>
</p>

<p>where the <code class="reqn">\mathit{SD}</code> and <code class="reqn">N</code> are the standard deviation 
and sample size of <code class="reqn">\bar{X}_C</code>, respectively.
</p>


<h3>References</h3>

<p>Lajeunesse, M.J. 2011. On the meta-analysis of response ratios 
for studies with correlated and multi-group designs. Ecology 92: 2049-2055.
</p>

<hr>
<h2 id='effort_distribute'>Assigns title/abstract screening efforts to a team.</h2><span id='topic+effort_distribute'></span>

<h3>Description</h3>

<p>Randomly distributes screening tasks evenly or unevenly across multiple team
members.  It populates this effort in a data frame column that includes this
screening work (e.g., ABSTRACTS and TITLES).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effort_distribute(
  aDataFrame = NULL,
  dual = FALSE,
  reviewers = NULL,
  column_name = "REVIEWERS",
  effort = NULL,
  initialize = FALSE,
  save_split = FALSE,
  directory = getwd()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effort_distribute_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data.frame containing the titles and abstracts to be 
screened by a team.  The default assumes that the data.frame has already
been formatted using <code>effort_initialize</code>.  This data.frame will be
populated with screening efforts.  See example: 
<code><a href="#topic+example_references_metagear">example_references_metagear</a></code></p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_dual">dual</code></td>
<td>
<p>When <code>TRUE</code>, distributes effort using a dual screening 
design where two members will screen the same random collection of 
titles/abstracts.  Requires the team to have an even number of members.</p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_reviewers">reviewers</code></td>
<td>
<p>A vector with the names of each team member.</p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_column_name">column_name</code></td>
<td>
<p>Changes the default label of the &quot;REVIEWERS&quot; column 
that contains the screening efforts of each team member.</p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_effort">effort</code></td>
<td>
<p>A vector of percentages used to allocate screening
tasks among each team member.  When not called explicitly, assumes effort 
to be distributed evenly among all members.  Must be the same length as 
the number of team members, and also sum to 100.</p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_initialize">initialize</code></td>
<td>
<p>When <code>TRUE</code>, initializes the data.frame so that 
efforts could be distributed, calls: <code><a href="#topic+effort_initialize">effort_initialize</a></code>.  
Default is <code>FALSE</code>.</p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_save_split">save_split</code></td>
<td>
<p>Saves the allocated team effort into separate effort_*.csv
files for individual screening tasks.  These files can be given to each
member to screen their random title/abstract subset.  All files can be 
merged once all screening tasks have been completed using 
<code><a href="#topic+effort_merge">effort_merge</a></code>.</p>
</td></tr>
<tr><td><code id="effort_distribute_+3A_directory">directory</code></td>
<td>
<p>Changes the default location/directory for where the
effort_*.csv will be saved.  If not explicitly called, it will deposit 
files in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame with title/abstract screening efforts randomly 
distributed across a team.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+effort_initialize">effort_initialize</a></code>, <code><a href="#topic+effort_merge">effort_merge</a></code>, 
<code><a href="#topic+effort_summary">effort_summary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(example_references_metagear)
theTeam &lt;- c("Christina", "Luc")
effort_distribute(example_references_metagear, initialize = TRUE, reviewers = theTeam)

## End(Not run)

</code></pre>

<hr>
<h2 id='effort_initialize'>Formats a reference dataset for title/abstract screening efforts.</h2><span id='topic+effort_initialize'></span>

<h3>Description</h3>

<p>Adds columns with standardized labels to a data framw with bibliographic data
on journal articles.  These columns will be used to assign reviewers, 
implementation of dual screening design, and the coding of
inclusion/exclusions screening decisions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effort_initialize(
  aDataFrame,
  study_ID = TRUE,
  unscreenedValue = "not vetted",
  dual = FALSE,
  front = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effort_initialize_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data.frame object that includes the titles and 
abstracts to be screened.  It will be formatted for screening efforts.  
See example: <code><a href="#topic+example_references_metagear">example_references_metagear</a></code></p>
</td></tr>
<tr><td><code id="effort_initialize_+3A_study_id">study_ID</code></td>
<td>
<p>When <code>FALSE</code>, does not add a column &quot;STUDY_ID&quot; that
includes a unique identification number for each reference (row) in the
dataFrame.</p>
</td></tr>
<tr><td><code id="effort_initialize_+3A_unscreenedvalue">unscreenedValue</code></td>
<td>
<p>Changes the default coding (a string) of &quot;not vetted&quot;
that designates whether an abstract remains to be screened or vetted as
part of the &quot;INCLUDE&quot; column.</p>
</td></tr>
<tr><td><code id="effort_initialize_+3A_dual">dual</code></td>
<td>
<p>When <code>TRUE</code>, formats dataFrame for a dual screening (paired)
design.  Creates two reviewer teams: REVIEWERS_A and REVIEWERS_B.</p>
</td></tr>
<tr><td><code id="effort_initialize_+3A_front">front</code></td>
<td>
<p>When <code>FALSE</code>, adds new columns to the back end of the
dataframe.  When <code>TRUE</code>, adds columns to the front.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame formatted for title/abstract screening efforts.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+effort_distribute">effort_distribute</a></code>, <code><a href="#topic+effort_merge">effort_merge</a></code>, 
<code><a href="#topic+effort_summary">effort_summary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_references_metagear)
effort_initialize(example_references_metagear)        

</code></pre>

<hr>
<h2 id='effort_merge'>Merges multiple files that had title/abstract screening efforts 
distributed across a team.</h2><span id='topic+effort_merge'></span>

<h3>Description</h3>

<p>Combines (merges) multiple effort_*.csv files within the same directory that 
represent the completed screening efforts of multiple team members.  These
files were originally generated with <code><a href="#topic+effort_distribute">effort_distribute</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effort_merge(directory = getwd(), reviewers = NULL, dual = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effort_merge_+3A_directory">directory</code></td>
<td>
<p>The directory name for the location of multiple .csv files. 
Assumes the current working directory if none is explicitly called.
File names must include the &quot;effort_&quot; string as originally generated by
<code><a href="#topic+effort_distribute">effort_distribute</a></code>.</p>
</td></tr>
<tr><td><code id="effort_merge_+3A_reviewers">reviewers</code></td>
<td>
<p>A vector of reviewer names (strings) used to merge effort 
from a select group of team members.  Must be an even collection (e.g., 
pairs of reviewers) when a dual design was implemented.</p>
</td></tr>
<tr><td><code id="effort_merge_+3A_dual">dual</code></td>
<td>
<p>When <code>TRUE</code>, merges files implementing a dual screening
design.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single data.frame merged from multiple files.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+effort_initialize">effort_initialize</a></code>, <code><a href="#topic+effort_distribute">effort_distribute</a></code>, 
<code><a href="#topic+effort_summary">effort_summary</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(example_references_metagear)
theTeam &lt;- c("Christina", "Luc")
# warning effort_distribute below, will save two files to working 
# directory: effort_Christina.csv and effort_Luc.csv
effort_distribute(example_references_metagear, initialize = TRUE, 
                  reviewers = theTeam, save_split = TRUE)
effort_merge()

## End(Not run)

</code></pre>

<hr>
<h2 id='effort_redistribute'>Redistributes title/abstract screening efforts among a review team.</h2><span id='topic+effort_redistribute'></span>

<h3>Description</h3>

<p>Randomly re-distributes screening tasks from one reviewers to the rest of the 
reviewing team.  Used when screening effort needs to be re-allocated among
reviewing team members.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effort_redistribute(
  aDataFrame,
  column_name = "REVIEWERS",
  reviewer = NULL,
  remove_effort = 100,
  reviewers = NULL,
  effort = NULL,
  save_split = FALSE,
  directory = getwd()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effort_redistribute_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data.frame containing the titles and abstracts to be 
screened by a team.  The default assumes that the data.frame has already 
been formatted using <code>effort_initialize</code> and populated with 
<code>effort_distribute</code>.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_column_name">column_name</code></td>
<td>
<p>Changes the default label of the &quot;REVIEWERS&quot; column 
that contains the screening efforts of each team member.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_reviewer">reviewer</code></td>
<td>
<p>The name of the reviewer whose effort is to be redistributed.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_remove_effort">remove_effort</code></td>
<td>
<p>The percentage of effort to be redistributed among the team.
The default is that 100% of the effort will be re-distributed.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_reviewers">reviewers</code></td>
<td>
<p>A vector of the names of each team member that will take on
additional work.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_effort">effort</code></td>
<td>
<p>A vector of percentages used to allocate screening
tasks among each team member.  When not called explicitly, assumes effort 
to be distributed evenly among all members.  Must be the same length as 
the number of team members, and also sum to 100.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_save_split">save_split</code></td>
<td>
<p>Saves the allocated team effort into separate &quot;effort_*.csv&quot;
files for individual screening tasks.  These files can be given to each
member to screen their random title/abstract subset.  All files can be 
merged once all screening tasks have been completed using 
<code>effort_merge</code>.</p>
</td></tr>
<tr><td><code id="effort_redistribute_+3A_directory">directory</code></td>
<td>
<p>Changes the default location/directory for where the
&quot;effort_*.csv&quot; will be saved.  If not explicitly called, it will deposit 
files in the current working directory.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single data.frame with effort re-allocated among team members.
</p>

<hr>
<h2 id='effort_summary'>Provides a text summary of screening efforts among the reviewing team.</h2><span id='topic+effort_summary'></span>

<h3>Description</h3>

<p>Summarizes the number of studies screened, which were identified to be
included/excluded from the project, as well as those with conflicting
agreement on their inclusion/exclusion.  If a dual (paired) design was
implemented to screen references, then it also provides inter-reviewer
agreement estimate following Cohen (1960) that describes the agreement (or
repeatability) of screening/coding decisions. The magnitudes of
inter-reviewer agreement estimates are then interpreted following
Landis &amp; Koch (1977).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>effort_summary(
  aDataFrame,
  column_reviewers = "REVIEWERS",
  column_effort = "INCLUDE",
  dual = FALSE,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="effort_summary_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data.frame containing the titles and abstracts that were
screened by a team.  The default assumes that the data.frame is the
merged effort across the team using <code>effort_merge</code>.</p>
</td></tr>
<tr><td><code id="effort_summary_+3A_column_reviewers">column_reviewers</code></td>
<td>
<p>Changes the default label of the &quot;REVIEWERS&quot; column
that contains the screening efforts of each team member.</p>
</td></tr>
<tr><td><code id="effort_summary_+3A_column_effort">column_effort</code></td>
<td>
<p>Changes the default label of the &quot;INCLUDE&quot; column
that contains the screening decisions (coded references) of each team
member.</p>
</td></tr>
<tr><td><code id="effort_summary_+3A_dual">dual</code></td>
<td>
<p>When <code>TRUE</code>, provides a summary of the dual screening
effort as well as estimation of inter-reviewer agreements following
Cohen's (1960) kappa (K) and Landis and Koch's (1977) interpretation
benchmarks.</p>
</td></tr>
<tr><td><code id="effort_summary_+3A_quiet">quiet</code></td>
<td>
<p>When <code>TRUE</code>, does not print to console the summary table.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with summary information on the screening tasks of a
reviewing team.
</p>


<h3>References</h3>

<p>Cohen, J. 1960. A coefficient of agreement for nominal scales.
Educational and Psychological Measurement 20: 37-46.
</p>
<p>Landis, J.R., and Koch, G.G. 1977. The measurement of observer
agreement for categorical data. Biometrics 33: 159-174.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+effort_initialize">effort_initialize</a></code>, <code><a href="#topic+effort_distribute">effort_distribute</a></code>,
<code><a href="#topic+effort_merge">effort_merge</a></code>
</p>

<hr>
<h2 id='example_references_metagear'>A collection of bibliographic references</h2><span id='topic+example_references_metagear'></span>

<h3>Description</h3>

<p>An example dataset containing 11 journal references.  The variables are 
described below.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(example_references_metagear)
</code></pre>


<h3>Format</h3>

<p>A data frame with 12 rows and 9 variables.
</p>


<h3>Details</h3>


<ul>
<li><p> AUTHORS. Authors of the journal article
</p>
</li>
<li><p> YEAR. Publication year
</p>
</li>
<li><p> TITLE. Article title
</p>
</li>
<li><p> JOURNAL. Journal name
</p>
</li>
<li><p> VOLUME. Journal volume number
</p>
</li>
<li><p> LPAGES. Lower page number
</p>
</li>
<li><p> UPAGES. Upper page number
</p>
</li>
<li><p> DOI. Digital object identifier (DOI) of journal article
</p>
</li>
<li><p> ABSTRACT. Full text of the journal article abstract
</p>
</li></ul>


<hr>
<h2 id='figure_add'>Manually add/detect points to a scatter plot figure.</h2><span id='topic+figure_add'></span>

<h3>Description</h3>

<p>Allows for the user to manually add an unlimited number of points to a 
figure image, by left-clicking over a figure's point.  Click on the red  
upper-right box called &quot;EXIT&quot; to end recording the position of manually 
detected points.
</p>
<p>Allows for the user to manually add an unlimited number of points to a
figure image, by left-clicking over a figure/image point.  Click on the red
upper-right box called &quot;EXIT&quot; to end recording the position of manually
detected points.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_add(file = file.choose(), color = "#009900", size = 0.03)

figure_add(file = file.choose(), color = "#009900", size = 0.03)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_add_+3A_file">file</code></td>
<td>
<p>The file name and location of a figure.  Prompts for file name
if none is explicitly called.  Can also be a binary figure image with
detected points (an <code>EBImage</code> object). See: <code><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a></code></p>
</td></tr>
<tr><td><code id="figure_add_+3A_color">color</code></td>
<td>
<p>The color to paint the manually detected points; default is
green.</p>
</td></tr>
<tr><td><code id="figure_add_+3A_size">size</code></td>
<td>
<p>The radius of the painted points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with detected points.
</p>
<p>A data frame with detected points.
</p>


<h3>See Also</h3>

<p><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a>
</p>
<p><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a>
</p>

<hr>
<h2 id='figure_barPlot'>Detect and display all bar plot objects.</h2><span id='topic+figure_barPlot'></span>

<h3>Description</h3>

<p>Automated detection of grouped data displayed in a bar-plot/chart figure image.
The default returns these detected objects as an <code>EBImage</code> raster image, 
and as a vector of all the estimated lengths that are proportional to the values
presented on each bar (and their error bars, if they are present).  Note that
the extracted points will be sorted by their positioning on the X-axis (or 
Y if the plot is a horizontal bar plot).  For example, if there were error
bars in the figure these will be grouped with the detected bar column.  However,
within these X-axis positioning they will not be sorted.  See vignette for
worked several illustrations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_barPlot(
  file = file.choose(),
  horizontal = FALSE,
  binary_threshold = 0.6,
  axis_thickness = 3,
  axis_sensitivity = 0.2,
  axis_length = 0.75,
  axis_X_color = "#00ABAB",
  axis_Y_color = "#B0D36A",
  Y_min = 0,
  Y_max = 100,
  bar_width = 9,
  bar_sensitivity = 0.1,
  point_color = "#0098B2",
  point_size = 9,
  ignore = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_barPlot_+3A_file">file</code></td>
<td>
<p>The file name and location of a bar-plot figure.  Prompts
for file name if none is explicitly called.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_horizontal">horizontal</code></td>
<td>
<p>If <code>TRUE</code> then aims to detect objects from a bar-plot 
that depicts data horizontally (rather than vertically).</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_binary_threshold">binary_threshold</code></td>
<td>
<p>A proportion from zero to one designating the 
gray-scale threshold to convert pixels into black or white.  Pixel
intensities below the proportion will be converted to black, and those 
above white.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_axis_thickness">axis_thickness</code></td>
<td>
<p>An integer used to designate the thickness of the 
axis lines on a figure.  Close alignment to the thickness of the axis
on a figure will improve axis detection.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_axis_sensitivity">axis_sensitivity</code></td>
<td>
<p>A value designating the sensitivity of identifying 
straight lines on figure.  A smaller number results in a higher 
sensitivity to identify axes.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_axis_length">axis_length</code></td>
<td>
<p>The relative size of the axis to the figure.  The default
is that axis lengths are 0.75 (75 percent) the size of the figure.  This 
option is necessary since bar lengths may be similar to the axis length.  
Values should range between zero and one.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_axis_x_color">axis_X_color</code></td>
<td>
<p>The color to paint the detected X-axis.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_axis_y_color">axis_Y_color</code></td>
<td>
<p>The color to paint the detected Y-axis.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_y_min">Y_min</code></td>
<td>
<p>The minimum Y value displayed on the Y-axis (used to scale
detected data points).</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_y_max">Y_max</code></td>
<td>
<p>The maximum Y value displayed on the Y-axis (used to scale
detected data points).</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_bar_width">bar_width</code></td>
<td>
<p>An integer value designating the width of vertical lines on
bars.  A smaller number should be used when the width of bars are 
small (as well as the width of error bars).</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_bar_sensitivity">bar_sensitivity</code></td>
<td>
<p>A value designating the sensitivity of identifying 
the vertical lines on bars.  A smaller number should be used when the
thickness of bars are small (as well as the width of error bars).</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_point_color">point_color</code></td>
<td>
<p>The color to paint the circles identifying the detected 
levels on bar columns and error bars.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_point_size">point_size</code></td>
<td>
<p>An integer used to designate the size of the points painting
the detected bars on a figure.</p>
</td></tr>
<tr><td><code id="figure_barPlot_+3A_ignore">ignore</code></td>
<td>
<p>When <code>TRUE</code> does not display painted image with detections,
only returns the data frame with detected points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of scaled lengths for detected column and error bars.
</p>

<hr>
<h2 id='figure_detectAllPoints'>Automated detection of plotted points from a scatter-plot figure image.</h2><span id='topic+figure_detectAllPoints'></span>

<h3>Description</h3>

<p>Attempts to detect all points of a certain shape and size from a scatter-plot 
figure image (even those lying outside of the axis range).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_detectAllPoints(
  aBinaryPlot,
  sensitivity = 0.2,
  point_shape = "circle",
  point_size = 5
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_detectAllPoints_+3A_abinaryplot">aBinaryPlot</code></td>
<td>
<p>A binary figure image (an EBImage object).
See: <code>figure_transformToBinary</code></p>
</td></tr>
<tr><td><code id="figure_detectAllPoints_+3A_sensitivity">sensitivity</code></td>
<td>
<p>A value designating the sensitivity of identifying unique
points that overlap.  A smaller number results in a higher sensitivity to 
split overlapping points; a larger number will extract only a single point
from a cluster of overlapping points.</p>
</td></tr>
<tr><td><code id="figure_detectAllPoints_+3A_point_shape">point_shape</code></td>
<td>
<p>The shape of points on figure: can be <code>"circle"</code>,
<code>"square"</code>, or <code>"diamond"</code>.  If these options do not fit the
shape found in a figure, use the option that best approximates that
shape.</p>
</td></tr>
<tr><td><code id="figure_detectAllPoints_+3A_point_size">point_size</code></td>
<td>
<p>An integer used to designate the size of the points on
the figure.  Close alignment to the size of the points on a figure will
improve point detection.  See <code>EBImage</code> to help determine which
size to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>EBImage</code> object with detected scatter-plot points.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+figure_detectAxis">figure_detectAxis</a></code>
</p>

<hr>
<h2 id='figure_detectAxis'>Detect an axis from a figure image.</h2><span id='topic+figure_detectAxis'></span>

<h3>Description</h3>

<p>Attempts to detect either the X (horizontal) or Y (vertical) axis from 
a plotted figure.
</p>
<p>Attempts to detect either the X (horizontal) or Y (vertical) axis from 
a plotted figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_detectAxis(
  aBinaryPlot,
  axis_type = "X",
  axis_thickness = 5,
  sensitivity = 0.2
)

figure_detectAxis(
  aBinaryPlot,
  axis_type = "X",
  axis_thickness = 5,
  sensitivity = 0.2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_detectAxis_+3A_abinaryplot">aBinaryPlot</code></td>
<td>
<p>A binary figure image (an EBImage object).
See: <code><a href="#topic+figure_transformToBinary">figure_transformToBinary</a></code></p>
</td></tr>
<tr><td><code id="figure_detectAxis_+3A_axis_type">axis_type</code></td>
<td>
<p>The axis to be detected from a figure: can 
be <code>X</code> or <code>Y</code>.</p>
</td></tr>
<tr><td><code id="figure_detectAxis_+3A_axis_thickness">axis_thickness</code></td>
<td>
<p>An integer used to designate the thickness of the 
axis lines on a figure.  Close alignment to the thickness of the axis
on a figure will improve axis detection.</p>
</td></tr>
<tr><td><code id="figure_detectAxis_+3A_sensitivity">sensitivity</code></td>
<td>
<p>A value designating the sensitivity of identifying 
straight lines on a figure.  A smaller number results in a higher 
sensitivity to identify axes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>EBImage</code> object with detected points.
</p>
<p>An <code>EBImage</code> object with detected points.
</p>


<h3>See Also</h3>

<p><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a>
</p>
<p><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a>
</p>

<hr>
<h2 id='figure_display'>Displays an image plot.</h2><span id='topic+figure_display'></span>

<h3>Description</h3>

<p>Displays a .jpg, .jpeg, .png, or .tiff image file containing a plotted 
figure, or plots an EBImage object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_display(file = file.choose(), browser = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_display_+3A_file">file</code></td>
<td>
<p>The file name and location of a plot figure or EBImage object.
Prompts for file name if nothing is explicitly called.  Preferably in .jpg 
format.</p>
</td></tr>
<tr><td><code id="figure_display_+3A_browser">browser</code></td>
<td>
<p>When <code>"TRUE"</code>, displays the figure image in the default 
web browser.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>EBImage</code> object figure.
</p>


<h3>See Also</h3>

<p><a href="#topic+figure_read">figure_read</a>
</p>

<hr>
<h2 id='figure_displayDetectedPoints'>Displays detected points on figure.</h2><span id='topic+figure_displayDetectedPoints'></span>

<h3>Description</h3>

<p>Generates a raster image of a figure with the detected points painted on a 
background/reference figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_displayDetectedPoints(
  aDetectedPlot,
  background = NULL,
  color = "red",
  size = 2,
  ignore = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_displayDetectedPoints_+3A_adetectedplot">aDetectedPlot</code></td>
<td>
<p>A binary figure image with detected points 
(an <code>EBImage</code> object).  See: <code><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a></code></p>
</td></tr>
<tr><td><code id="figure_displayDetectedPoints_+3A_background">background</code></td>
<td>
<p>An EBImage figure of same size to be used as
background (e.g., the original (RGB/color) figure image).</p>
</td></tr>
<tr><td><code id="figure_displayDetectedPoints_+3A_color">color</code></td>
<td>
<p>The color to paint the detected points.</p>
</td></tr>
<tr><td><code id="figure_displayDetectedPoints_+3A_size">size</code></td>
<td>
<p>The radius of the painted points.</p>
</td></tr>
<tr><td><code id="figure_displayDetectedPoints_+3A_ignore">ignore</code></td>
<td>
<p>When <code>TRUE</code> does not display painted image, only 
returns painted image EBImage object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A RGB <code>EBImage</code> painted with detected points.
</p>


<h3>See Also</h3>

<p><a href="#topic+figure_displayDetections">figure_displayDetections</a>
</p>

<hr>
<h2 id='figure_displayDetections'>Displays the detected figure objects.</h2><span id='topic+figure_displayDetections'></span>

<h3>Description</h3>

<p>Generates a raster image of a figure with the detected objects painted on a 
background/reference figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_displayDetections(
  aDetectedPlot,
  background = NULL,
  color = "red",
  ignore = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_displayDetections_+3A_adetectedplot">aDetectedPlot</code></td>
<td>
<p>A binary figure image with detected objects
(an <code>EBImage</code> object).</p>
</td></tr>
<tr><td><code id="figure_displayDetections_+3A_background">background</code></td>
<td>
<p>An <code>EBImage</code> figure of same size to be used as
background (e.g., the original [RGB/color] figure image).</p>
</td></tr>
<tr><td><code id="figure_displayDetections_+3A_color">color</code></td>
<td>
<p>The color to paint the detected objects.</p>
</td></tr>
<tr><td><code id="figure_displayDetections_+3A_ignore">ignore</code></td>
<td>
<p>When <code>TRUE</code> does not display painted image, only 
returns painted image EBImage object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A RGB <code>EBImage</code> painted with detected figure objects.
</p>

<hr>
<h2 id='figure_extractDetectedPoints'>Extracts data points from a detected image.</h2><span id='topic+figure_extractDetectedPoints'></span>

<h3>Description</h3>

<p>Extracts raw X and Y data from the points detected in a scatter-plot figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_extractDetectedPoints(
  aDetectedPlot,
  xAxis = NULL,
  yAxis = NULL,
  X_min = NULL,
  X_max = NULL,
  Y_min = NULL,
  Y_max = NULL,
  summarize = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_extractDetectedPoints_+3A_adetectedplot">aDetectedPlot</code></td>
<td>
<p>A binary figure image with detected points (an 
<code>EBImage</code> object).  See: <code><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a></code></p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_xaxis">xAxis</code></td>
<td>
<p>A binary figure image with detected X-axis (an <code>EBImage</code>
object).  See: <code><a href="#topic+figure_detectAxis">figure_detectAxis</a></code>.</p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_yaxis">yAxis</code></td>
<td>
<p>A binary figure image with detected Y-axis (an <code>EBImage</code> 
object).  See: <code><a href="#topic+figure_detectAxis">figure_detectAxis</a></code>.</p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_x_min">X_min</code></td>
<td>
<p>The minimum value of X reported on the figure X-axis.</p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_x_max">X_max</code></td>
<td>
<p>The maximum value of X reported on the figure X-axis.</p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_y_min">Y_min</code></td>
<td>
<p>The minimum value of Y reported on the figure Y-axis.</p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_y_max">Y_max</code></td>
<td>
<p>The maximum value of Y reported on the figure Y-axis.</p>
</td></tr>
<tr><td><code id="figure_extractDetectedPoints_+3A_summarize">summarize</code></td>
<td>
<p>When <code>TRUE</code> returns a summary of the regression 
parameters (intercept + slope * X), R-squared, Pearson's product moment
correlation coefficient (r), and its variance (var_r) and sample size (N).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with the extracted X and Y values.
</p>

<hr>
<h2 id='figure_read'>Reads/loads a figure image from file.</h2><span id='topic+figure_read'></span>

<h3>Description</h3>

<p>Reads a .jpg, .jpeg, .png, or .tiff image file containing a plotted figure.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_read(file = file.choose(), display = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_read_+3A_file">file</code></td>
<td>
<p>The file name and location of a plot figure.  Prompts
for file name if none is explicitly called.  Preferably in .jpg format.</p>
</td></tr>
<tr><td><code id="figure_read_+3A_display">display</code></td>
<td>
<p>When <code>"TRUE"</code>, displays the figure as a raster image.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>EBImage</code> object figure.
</p>


<h3>See Also</h3>

<p><a href="#topic+figure_write">figure_write</a>
</p>

<hr>
<h2 id='figure_removeOutlyingPoints'>Remove outlier points from a figure.</h2><span id='topic+figure_removeOutlyingPoints'></span>

<h3>Description</h3>

<p>Removes all detected points outside of axis range.  Requires three detected
images: one based on <code><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a></code>, and two others based on 
detected X- and Y-axes (i.e. <code><a href="#topic+figure_detectAxis">figure_detectAxis</a></code>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_removeOutlyingPoints(aDetectedPlot, xAxis = NULL, yAxis = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_removeOutlyingPoints_+3A_adetectedplot">aDetectedPlot</code></td>
<td>
<p>A binary figure image with detected points (an 
<code>EBImage</code> object).
See: <code><a href="#topic+figure_detectAllPoints">figure_detectAllPoints</a></code></p>
</td></tr>
<tr><td><code id="figure_removeOutlyingPoints_+3A_xaxis">xAxis</code></td>
<td>
<p>A binary figure image with detected X-axis (an <code>EBImage</code> object).
See: <code><a href="#topic+figure_detectAxis">figure_detectAxis</a></code></p>
</td></tr>
<tr><td><code id="figure_removeOutlyingPoints_+3A_yaxis">yAxis</code></td>
<td>
<p>A binary figure image with detected Y-axis (an <code>EBImage</code> object).
See: <code><a href="#topic+figure_detectAxis">figure_detectAxis</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>EBImage</code> object with detected points within the specified X-
and Y-axis ranges.
</p>

<hr>
<h2 id='figure_scatterPlot'>Detect and display all scatter plot objects.</h2><span id='topic+figure_scatterPlot'></span>

<h3>Description</h3>

<p>Automated detection of the X-axis, Y-axis, and points on a scatter-plot
figure image.  The default returns these detected objects as an 
<code>EBImage</code> raster image, as well as the estimated effect size (correlation
coefficient or r) of the data within the scatter-plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_scatterPlot(
  file = file.choose(),
  binary_threshold = 0.6,
  binary_point_fill = FALSE,
  binary_point_tolerance = 2,
  axis_thickness = 5,
  axis_sensitivity = 0.2,
  axis_X_color = "#00ABAB",
  X_min = 40,
  X_max = 140,
  axis_Y_color = "#B0D36A",
  Y_min = 40,
  Y_max = 140,
  point_sensitivity = 0.2,
  point_shape = "circle",
  point_size = 3,
  point_color = "#0098B2",
  ignore = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_scatterPlot_+3A_file">file</code></td>
<td>
<p>The file name and location of a scatter plot figure.  Prompts
for file name if none is explicitly called.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_binary_threshold">binary_threshold</code></td>
<td>
<p>A proportion from zero to one designating the 
gray-scale threshold to convert pixels into black or white.  Pixel
intensities below the proportion will be converted to black, and those 
above white.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_binary_point_fill">binary_point_fill</code></td>
<td>
<p>If <code>TRUE</code> then fills empty points/symbols
in figure.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_binary_point_tolerance">binary_point_tolerance</code></td>
<td>
<p>An integer used to designate the size of the 
points to the fill.  Increase value to better fill empty points.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_axis_thickness">axis_thickness</code></td>
<td>
<p>An integer used to designate the thickness of the 
axis lines on a figure.  Close alignment to the thickness of the axis
on a figure will improve axis detection.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_axis_sensitivity">axis_sensitivity</code></td>
<td>
<p>A value designating the sensitivity of identifying 
straight lines on figure.  A smaller number results in a higher 
sensitivity to identify axes.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_axis_x_color">axis_X_color</code></td>
<td>
<p>The color to paint the detected X-axis.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_x_min">X_min</code></td>
<td>
<p>The minimum X value displayed on the X-axis (used to scale
detected data points).</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_x_max">X_max</code></td>
<td>
<p>The maximum X value displayed on the X-axis (used to scale
detected data points).</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_axis_y_color">axis_Y_color</code></td>
<td>
<p>The color to paint the detected Y-axis.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_y_min">Y_min</code></td>
<td>
<p>The minimum Y value displayed on the Y-axis (used to scale
detected data points).</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_y_max">Y_max</code></td>
<td>
<p>The maximum Y value displayed on the Y-axis (used to scale
detected data points).</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_point_sensitivity">point_sensitivity</code></td>
<td>
<p>A value designating the sensitivity of identifying 
unique points that overlap.  A smaller number results in a higher 
sensitivity to split overlapping points; a larger number will extract only
a single point from a cluster of overlapping points.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_point_shape">point_shape</code></td>
<td>
<p>The shape of points on figure: can be <code>"circle"</code>,
<code>"square"</code>, or <code>"diamond"</code>.  If these options do not fit the
shape found in a figure, use the option that best approximates that
shape.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_point_size">point_size</code></td>
<td>
<p>An integer used to designate the size of the points on
the figure.  Close alignment to the size of the points on a figure will
improve point detection.  See <code>EBImage</code> package to help determine which
size to use.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_point_color">point_color</code></td>
<td>
<p>The color to paint the detected scatter plot points.</p>
</td></tr>
<tr><td><code id="figure_scatterPlot_+3A_ignore">ignore</code></td>
<td>
<p>When <code>TRUE</code> does not display painted image, only 
returns painted image EBImage object.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with detected points.
</p>

<hr>
<h2 id='figure_splitPlot'>Splits a composite figure that contains multiple plots.</h2><span id='topic+figure_splitPlot'></span>

<h3>Description</h3>

<p>Automatically detects divisions among multiple plots found within a single
figure image file.  It then uses these divisions to split the image into 
multiple image files; each containing only a single X-Y plot.  Currently only
works on composite figures that have a matrix-style presentation where each 
sub-plot has the same size.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_splitPlot(
  file = file.choose(),
  binary_threshold = 0.6,
  space_sensitivity_X = 0.4,
  space_sensitivity_Y = 0.6,
  border_buffer = 5,
  guess_limit = 10,
  ignoreX = FALSE,
  ignoreY = FALSE,
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_splitPlot_+3A_file">file</code></td>
<td>
<p>The file name and location of a composite figure.  Prompts
for file name if none is explicitly called.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_binary_threshold">binary_threshold</code></td>
<td>
<p>A proportion from zero to one designating the 
gray-scale threshold to convert pixels into black or white.  Pixel
intensities below the proportion will be converted to black, and those 
above white.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_space_sensitivity_x">space_sensitivity_X</code></td>
<td>
<p>A proportion ranging from zero to one that 
designates the size of the separation among sub-plots along the X-axis 
relative to the largest empty space detected in the figure image.  As
space_sensitivity_X approaches 1, finer empty spaces (e.g., empty spaces
found in between plot captions and the axis line) will be treated as plot
divisions.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_space_sensitivity_y">space_sensitivity_Y</code></td>
<td>
<p>A proportion ranging from zero to one that 
designates the size of the seperation among sub-plots along the Y-axis 
relative to the largest empty space detected in the figure image.  As
space_sensitivity_Y approaches 1, finer empty spaces (e.g., empty spaces
found in between plot captions and the axis line) will be treated as plot
divisions.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_border_buffer">border_buffer</code></td>
<td>
<p>An integer value designating the amount of empty space
around the figure image that should be ignored.  As the number increases,
more blank space near the image's edge will be ignored.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_guess_limit">guess_limit</code></td>
<td>
<p>An integer value designating the number of guesses for 
within a figure image.  The default value designates the top 10 guesses of 
divisions.  Increase this number if there are more than 6 subplots per axis.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_ignorex">ignoreX</code></td>
<td>
<p>When <code>TRUE</code>, ignores detection of sub-plots along the 
X-axis.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_ignorey">ignoreY</code></td>
<td>
<p>When <code>TRUE</code>, ignores detection of sub-plots along the 
Y-axis.</p>
</td></tr>
<tr><td><code id="figure_splitPlot_+3A_quiet">quiet</code></td>
<td>
<p>When <code>TRUE</code>, does not print to console the saved file names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of sub-plots saved to file.
</p>

<hr>
<h2 id='figure_transformByColors'>Transforms RBG figure into list of binary images.</h2><span id='topic+figure_transformByColors'></span>

<h3>Description</h3>

<p>Generates a list of binary images relative to the number of colors in an 
RBG figure.  Useful to do when there are multiple objects in a figure 
presented with different colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_transformByColors(aFigure, colorsToSplit = 2)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_transformByColors_+3A_afigure">aFigure</code></td>
<td>
<p>The original (RBG/color) figure image (an <code>EBImage</code> object).</p>
</td></tr>
<tr><td><code id="figure_transformByColors_+3A_colorstosplit">colorsToSplit</code></td>
<td>
<p>An integer designating the number of colors in the
figure.  The number indicates the number of color intensities to 
divide into separate binary figures.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A colorsToSplit + 1 list of <code>EBImage</code> black and white objects.
The final item in this list will be an inverse binary of the original
figure.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+figure_transformToBinary">figure_transformToBinary</a></code>
</p>

<hr>
<h2 id='figure_transformToBinary'>Transforms figure to binary image.</h2><span id='topic+figure_transformToBinary'></span>

<h3>Description</h3>

<p>Transforms a figure into a black and white image.  This pre-processing of the
image is necessary to help identify objects within the figure (e.g., axes, 
plotted points).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_transformToBinary(
  aFigure,
  threshold = 0.6,
  point_fill = FALSE,
  point_tolerance = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_transformToBinary_+3A_afigure">aFigure</code></td>
<td>
<p>The original figure image (an <code>EBImage</code> object).</p>
</td></tr>
<tr><td><code id="figure_transformToBinary_+3A_threshold">threshold</code></td>
<td>
<p>A proportion from zero to one designating the gray-scale
threshold to convert pixels into black or white.  Pixel intensities below
the proportion will be converted to black, and those above white.  Helps
remove noise and increase contrast among candidate objects to detect.</p>
</td></tr>
<tr><td><code id="figure_transformToBinary_+3A_point_fill">point_fill</code></td>
<td>
<p>If <code>TRUE</code> then fills empty points/symbols in figure.</p>
</td></tr>
<tr><td><code id="figure_transformToBinary_+3A_point_tolerance">point_tolerance</code></td>
<td>
<p>An integer used to designate the size of the points 
to fill.  Increase value to better fill empty points.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An <code>EBImage</code> black and white object ready for object detection.
</p>

<hr>
<h2 id='figure_write'>Saves/writes a figure image.</h2><span id='topic+figure_write'></span>

<h3>Description</h3>

<p>Writes a figure image to file and returns the file name.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>figure_write(aFigure, file = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="figure_write_+3A_afigure">aFigure</code></td>
<td>
<p>The <code>EBImage</code> figure.</p>
</td></tr>
<tr><td><code id="figure_write_+3A_file">file</code></td>
<td>
<p>Name and location of file to save.  Supports .jpg, .png, and
.tiff image formats.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of file names.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+figure_read">figure_read</a></code>
</p>

<hr>
<h2 id='impute_missingness'>Provides a summary of missingness in a dataset.</h2><span id='topic+impute_missingness'></span>

<h3>Description</h3>

<p>Generates a summary of the percentage of missing data in a dataset.  Provides
insight on the appropriateness of imputation methods.  For example, if 30%
of data is missing, then perhaps this is too much to impute.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute_missingness(aDataFrame)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="impute_missingness_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data.frame containing columns that will be assessed for
missingness.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame that summarizes percent missingness for each column of 
a dataset.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(example_references_metagear)
impute_missingness(example_references_metagear)

</code></pre>

<hr>
<h2 id='impute_SD'>Imputes missing standard deviations in a dataset.</h2><span id='topic+impute_SD'></span>

<h3>Description</h3>

<p>Imputes (fills gaps) of missing standard deviations (SD) using simple imputation
methods following Bracken (1992) and Rubin and Schenker's (1991) &quot;hot deck&quot; 
approach.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>impute_SD(
  aDataFrame,
  columnSDnames,
  columnXnames,
  method = "Bracken1992",
  range = 3,
  M = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="impute_SD_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data frame containing columns with missing SD's (coded as 
<code>NA</code>) and their complete means (used only for nearest-neighbor method).</p>
</td></tr>
<tr><td><code id="impute_SD_+3A_columnsdnames">columnSDnames</code></td>
<td>
<p>Label of the column(s) with missing SD.  Can be a string 
or list of strings.</p>
</td></tr>
<tr><td><code id="impute_SD_+3A_columnxnames">columnXnames</code></td>
<td>
<p>Label of the column(s) with means (X) for each SD.  Can be 
a string or list of strings.  Must be complete with no missing data.</p>
</td></tr>
<tr><td><code id="impute_SD_+3A_method">method</code></td>
<td>
<p>The method used to impute the missing SD's.  The default is 
<code>"Bracken1992"</code> which applies Bracken's (1992) approach to impute SD using
the coefficient of variation from all complete cases.  Other options include:
<code>"HotDeck"</code> which applies Rubin and Schenker's (1991) resampling approach to
fill gaps of missing SD from the SD's with complete information, and 
<code>"HotDeck_NN"</code> which resamples from complete cases with means that are similar
to missing SD's.</p>
</td></tr>
<tr><td><code id="impute_SD_+3A_range">range</code></td>
<td>
<p>A positive number on the range of neighbours to sample from for
imputing SD's.  Used in combination with <code>"HotDeck_NN"</code>. The default 
is 3; which indicates that the 3 means that are most similar in rank order
to the mean with the missing SD will be resampled.</p>
</td></tr>
<tr><td><code id="impute_SD_+3A_m">M</code></td>
<td>
<p>The number of imputed datasets to return.  Currently only works
for <code>"HotDeck"</code> method.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An imputed (complete) dataset.
</p>


<h3>References</h3>

<p>Bracken, M.B. 1992. Statistical methods for analysis of effects
of treatment in overviews of randomized trials. Effective care of the
newborn infant (eds J.C. Sinclair and M.B. Bracken), pp. 
13-20. Oxford University Press, Oxford.
</p>
<p>Rubin, D.B. and Schenker, N. 1991. Multiple imputation in 
health-care databases: an overview and some applications.  Statistics 
in Medicine 10: 585-598.
</p>

<hr>
<h2 id='isPDF'>Evaluates whether a file is a PDF document.</h2><span id='topic+isPDF'></span>

<h3>Description</h3>

<p>Checks if provided file is in Portable Document Format (PDF).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>isPDF(aFileName, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="isPDF_+3A_afilename">aFileName</code></td>
<td>
<p>A string that identifies a file name (and directory path) of
the PDF candidate.</p>
</td></tr>
<tr><td><code id="isPDF_+3A_verbose">verbose</code></td>
<td>
<p>Provides more elaborate description of why the file could not 
be evaluated as a PDF (e.g., when validating a PDF online). When 
<code>"quiet"</code>, an error message is not generated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical value indicating whether the file is a PDF document.
</p>

<hr>
<h2 id='Kam_et_al_2003_Fig2.jpg'>An example image of a scatterplot figure</h2><span id='topic+Kam_et_al_2003_Fig2.jpg'></span>

<h3>Description</h3>

<p>A jpg image of a scatterplot from Figure 2 of Kam, M., Cohen-Gross, S., 
Khokhlova, I.S., Degen, A.A. and Geffen, E. 2003. Average daily metabolic 
rate, reproduction and energy allocation during lactation in the Sundevall
Jird Meriones crassus. Functional Ecology 17:496-503.
</p>


<h3>Format</h3>

<p>A raw jpg-formated image
</p>


<h3>Note</h3>

<p><strong>How to use</strong><br /><br /> 
<code>readImage(system.file("images", "Kam_et_al_2003_Fig2.jpg", package = "metagear"))</code>
</p>

<hr>
<h2 id='Kortum_and_Acymyan_2013_Fig4.jpg'>An example image of a bar plot figure</h2><span id='topic+Kortum_and_Acymyan_2013_Fig4.jpg'></span>

<h3>Description</h3>

<p>A jpg image of a bar plot from Figure 4 of Kortum, P., and Acymyan, C.Z. 
2013. How low can you go? Is the System Usability Scale range restricted? 
Journal of Usability Studies 9:14-24.
</p>


<h3>Format</h3>

<p>A raw jpg-formated image
</p>


<h3>Note</h3>

<p><strong>How to use</strong><br /><br /> 
<code>readImage(system.file("images", "Kortum_and_Acymyan_2013_Fig4.jpg", package = "metagear"))</code>
</p>

<hr>
<h2 id='MA_effectsTable'>Generate an ANOVA-like effects table for a meta-analysis.</h2><span id='topic+MA_effectsTable'></span>

<h3>Description</h3>

<p>Generates an ANOVA-like effects table that summarizes the within and 
between-study homogeneity tests (Q-tests), as well as moderator level 
Q-tests as originally described by Hedges and Olkin (1985; p. 156).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MA_effectsTable(model, weights, data, effects_model = "random")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MA_effectsTable_+3A_model">model</code></td>
<td>
<p>A two-sided linear formula object describing the model, with the
response (effect sizes) on the left of a ~ operator and the moderator 
variables, separated by +, :, * operators, on the right.</p>
</td></tr>
<tr><td><code id="MA_effectsTable_+3A_weights">weights</code></td>
<td>
<p>A column label from data.frame of variances to be used as weights.</p>
</td></tr>
<tr><td><code id="MA_effectsTable_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the variables named in the
model.</p>
</td></tr>
<tr><td><code id="MA_effectsTable_+3A_effects_model">effects_model</code></td>
<td>
<p>The default is <code>"random"</code>, which specifies a 
random-effects meta-analysis (DerSimonian and Laird method).  Other 
options include <code>"fixed"</code> which presents fixed-effect analyses.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An lm object of main effects.
</p>


<h3>References</h3>

<p>DerSimonian, R., and N. Laird. 1986. Meta-analysis in clinical 
trials. Controlled Clinical Trials, 7, 177-188.
</p>
<p>Hedges, L.V., and I. Olkin. 1985. Statistical methods for 
meta-analysis. Academic Press, New York, USA.
</p>

<hr>
<h2 id='PDF_download'>Attempts to download a PDF using a DOI link.</h2><span id='topic+PDF_download'></span>

<h3>Description</h3>

<p>Tries to download a PDF file using the digital objected identifier (DOI) link.
Uses ad hoc searches of journal HTML pages to detect candidate PDFs for
download.  Downloads all candidate pdfs. <strong>If running downloader
in Windows, having <code>"WindowsProxy = TRUE"</code> will significantly improve
download success.</strong>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PDF_download(
  DOI,
  directory = getwd(),
  theFileName = "temp",
  validatePDF = TRUE,
  quiet = FALSE,
  WindowsProxy = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PDF_download_+3A_doi">DOI</code></td>
<td>
<p>A string of the DOI (digital object identifier) used to identify
the source of a journal article PDF file(s).</p>
</td></tr>
<tr><td><code id="PDF_download_+3A_directory">directory</code></td>
<td>
<p>A string of the location (directory) were downloaded PDF
files are to be saved.  Directory name must end with &quot;\\&quot;.</p>
</td></tr>
<tr><td><code id="PDF_download_+3A_thefilename">theFileName</code></td>
<td>
<p>Used to rename the downloaded file.  No need to include
extension &quot;.pdf&quot;.</p>
</td></tr>
<tr><td><code id="PDF_download_+3A_validatepdf">validatePDF</code></td>
<td>
<p>When <code>"TRUE"</code> will only save to files that are valid
PDF documents.  When <code>"FALSE"</code> will save all candidate files, even if
they are not valid PDF formats.</p>
</td></tr>
<tr><td><code id="PDF_download_+3A_quiet">quiet</code></td>
<td>
<p>When <code>"FALSE"</code> does not print to console download progress
and summary.</p>
</td></tr>
<tr><td><code id="PDF_download_+3A_windowsproxy">WindowsProxy</code></td>
<td>
<p>When <code>TRUE</code> significantly improves download success
for computers running Windows; when <code>FALSE</code> on a Windows based
computer, you may only be able to download 30 to 50 PDFs at a time before
a connection error occurs and halts all downloads (e.g.,
<code>InternetOpenUrl failed</code> error).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A string describing the download success.  If unsuccessful,
returns the type of error during the download attempt.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PDFs_collect">PDFs_collect</a></code>
</p>

<hr>
<h2 id='PDF_extractImages'>Attempts to extract all images from a PDF</h2><span id='topic+PDF_extractImages'></span>

<h3>Description</h3>

<p>Tries to extract images within a PDF file.  Currently does not support 
decoding of images in CCITT compression formats. However, will still save 
these images to file; as a record of the number of images detected in the PDF.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PDF_extractImages(file = file.choose())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PDF_extractImages_+3A_file">file</code></td>
<td>
<p>The file name and location of a PDF file.  Prompts
for file name if none is explicitly called.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of file names saved as images.
</p>

<hr>
<h2 id='PDFs_collect'>Attempts to download PDFs from multiple DOI links.</h2><span id='topic+PDFs_collect'></span>

<h3>Description</h3>

<p>Tries to download a collection of PDF files using multiple digital object 
identifier (DOI) links.  Updates a data frame with the success of these 
downloads.  The function is a wrapper for <code><a href="#topic+PDF_download">PDF_download</a></code>.  NOTE: 
A single DOI may generate multiple PDF files.  <strong>If running downloader 
in Windows, having <code>"WindowsProxy = TRUE"</code> will significantly improve 
download success.</strong>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PDFs_collect(
  aDataFrame,
  DOIcolumn,
  FileNamecolumn,
  directory = getwd(),
  randomize = FALSE,
  seed = NULL,
  buffer = FALSE,
  validatePDF = TRUE,
  quiet = FALSE,
  showSummary = TRUE,
  WindowsProxy = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PDFs_collect_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data frame containing a column of DOIs and a column of
individual file names for each downloaded PDF.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_doicolumn">DOIcolumn</code></td>
<td>
<p>The label of the column containing all the DOI links.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_filenamecolumn">FileNamecolumn</code></td>
<td>
<p>The label of the column containing all the strings 
that will be used to rename the downloaded files.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_directory">directory</code></td>
<td>
<p>A string of the location (directory) were downloaded PDF
files are to be saved.  NOTE: helps to have this directory created before
initializing the <code>PDFs_collect</code> function.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_randomize">randomize</code></td>
<td>
<p>When <code>TRUE</code> will attempt to download PDFs in a random
order.  This may be necessary to ensure that host websites do not have
their HTML and files repeatedly accessed.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_seed">seed</code></td>
<td>
<p>An integer used to enforce repeatability when randomly 
downloading PDFs.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_buffer">buffer</code></td>
<td>
<p>When <code>TRUE</code> will randomly delay the downloads by a few
seconds (with a mean 4 seconds and a range of 1 to 20 seconds). Another
strategy to avoid quickly and repeatedly accessing host websites.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_validatepdf">validatePDF</code></td>
<td>
<p>When <code>TRUE</code> will only save to files that are valid
PDF documents.  When <code>FALSE</code> will save all candidate files, even if
they are not valid PDF formats.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_quiet">quiet</code></td>
<td>
<p>When <code>FALSE</code> does not print to console individual 
download progress and summary.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_showsummary">showSummary</code></td>
<td>
<p>When <code>FALSE</code> does not print overall summary of download
successes and failures.</p>
</td></tr>
<tr><td><code id="PDFs_collect_+3A_windowsproxy">WindowsProxy</code></td>
<td>
<p>When <code>TRUE</code> significantly improves download success 
for computers running Windows; when <code>FALSE</code> on a Windows based 
computer, you may only be able to download 30 to 50 PDFs at a time before 
a connection error occurs and halts all downloads (e.g., 
<code>InternetOpenUrl failed</code> error).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The data frame with new column containing download-outcome successes.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PDF_download">PDF_download</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

data(example_references_metagear)
someRefs &lt;- effort_initialize(example_references_metagear)  
dir.create("metagear_downloads")      
PDFs_collect(aDataFrame = someRefs, DOIcolumn = "DOI", 
             FileNamecolumn = "STUDY_ID", directory = "metagear_downloads",
			WindowsProxy = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='plot_PRISMA'>Plots and creates a PRISMA flow diagram.</h2><span id='topic+plot_PRISMA'></span>

<h3>Description</h3>

<p>Creates a PRISMA (Preferred Reporting Items for Systematic Reviews and 
Meta-Analyses) flow diagram depicting the 'flow' of study inclusions and
exclusions during various systematic review phases.  It is meant to 
describe the number of studies identified, included, excluded, reasons
for inclusion/exclusions, and final number of studies used in the
meta-analysis.  NOTE: currently only supports two start phases, and final
phase must not have an exclude branch.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_PRISMA(
  aPhaseVector,
  colWidth = 30,
  excludeDistance = 0.8,
  design = "classic",
  hide = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_PRISMA_+3A_aphasevector">aPhaseVector</code></td>
<td>
<p>A vector of ordered labels (strings) for each phase of
the PRISMA diagram.  Labels designating the beginning of the diagram
are commented with &quot;START_PHASE: &quot; and those designating exclusion phases
&quot;EXCLUDE_PHASE: &quot;.  These comments will be removed from the diagram.</p>
</td></tr>
<tr><td><code id="plot_PRISMA_+3A_colwidth">colWidth</code></td>
<td>
<p>An optional value (integer) designating the width of the text 
box of each phase.</p>
</td></tr>
<tr><td><code id="plot_PRISMA_+3A_excludedistance">excludeDistance</code></td>
<td>
<p>An optional value designating the the distance of 
exclude phase box from the main flow diagram.  Larger values (&gt; 0.8) 
increase this distance.</p>
</td></tr>
<tr><td><code id="plot_PRISMA_+3A_design">design</code></td>
<td>
<p>Designates the colorscheme and design of the the flow diagram.
The default is <code>classic</code> (as in versions of metagear prior to v. 0.4).
Others schemes are also available with color and more flat designs, and these
can be further customized; see NOTE below for these details.</p>
</td></tr>
<tr><td><code id="plot_PRISMA_+3A_hide">hide</code></td>
<td>
<p>When FALSE, the PRISMA flow diagram is not plotted.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a grid object (grob) list
</p>


<h3>Note</h3>

<p><strong>Using canned or custom PRISMA design layouts</strong><br /><br /> There are 
several color schemes and design layouts (e.g. curved or flat) available. 
These designs include: <code>cinnamonMint</code>, <code>sunSplash</code>, <code>pomegranate</code>,
<code>vintage</code>, <code>grey</code>, and <code>greyMono</code>. Custom schemes can also 
be developed by modifying each aspect of the design.  These are:
</p>

<dl>
<dt>S</dt><dd><p>color of start phases (default: white)</p>
</dd>
<dt>P</dt><dd><p>color of the main phases (default: white)</p>
</dd>
<dt>E</dt><dd><p>color of the exclusion phases (default: white)</p>
</dd>
<dt>F</dt><dd><p>color of the final phase (default: white)</p>
</dd>
<dt>fontSize</dt><dd><p>the size of the font (default: 12) </p>
</dd>
<dt>fontColor</dt><dd><p>the font color (default: black)</p>
</dd>
<dt>fontFace</dt><dd><p>either plain, bold, italic, or bold.italic (default: plain)</p>
</dd>
<dt>flatArrow</dt><dd><p>arrows curved when FALSE (default); arrows square when TRUE</p>
</dd>
<dt>flatBox</dt><dd><p>boxes curved when FALSE (default); Boxes square when TRUE</p>
</dd>
</dl>
 
<p>For example, changing the defaults to have red rather than white exclusion 
phases, and square boxes, would be: <code>design = c(E = "red", flatBox = TRUE)</code>.
</p>


<h3>References</h3>

<p>Moher, D., Liberati, A., Tetzlaff, J. and Altman, D.G.,
PRISMA Group. (2009) Preferred reporting items for systematic reviews and
meta-analyses: the PRISMA statement. BMJ 339, b2535.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>phases &lt;- c("START_PHASE: # of studies identified through database searching",
            "START_PHASE: # of additional studies identified through other sources",
            "# of studies after duplicates removed",
            "# of studies with title and abstract screened",
            "EXCLUDE_PHASE: # of studies excluded",
            "# of full-text articles assessed for eligibility",
            "EXCLUDE_PHASE: # of full-text excluded, not fitting eligibility criteria",
            "# of studies included in qualitative synthesis",
            "EXCLUDE_PHASE: # studies excluded, incomplete data reported",
            "final # of studies included in quantitative synthesis (meta-analysis)")
plot_PRISMA(phases, design = "cinnamonMint")

</code></pre>

<hr>
<h2 id='random_d'>Random generation of Hedges' d effect sizes.</h2><span id='topic+random_d'></span>

<h3>Description</h3>

<p>Generates random Hedges' d (1981, 1982) effect sizes and their variances.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_d(K, X_t, var_t, N_t, X_c, var_c, N_c, bias_correction = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_d_+3A_k">K</code></td>
<td>
<p>Number of effect sizes to generate.</p>
</td></tr>
<tr><td><code id="random_d_+3A_x_t">X_t</code></td>
<td>
<p>The population mean (mu) of the (t)reatment group.</p>
</td></tr>
<tr><td><code id="random_d_+3A_var_t">var_t</code></td>
<td>
<p>The population variance of the treatment group mean.</p>
</td></tr>
<tr><td><code id="random_d_+3A_n_t">N_t</code></td>
<td>
<p>The number of samples of the treatment mean.  When a non-negative
integer, all treatment means will be estimated using the same N.  A
vector of unequal N's can also be taken; if so, K will be ignored and the 
number of randomly generated means will equal the length of that vector,
and each mean will be based on each N within the vector.</p>
</td></tr>
<tr><td><code id="random_d_+3A_x_c">X_c</code></td>
<td>
<p>The population mean (mu) of the (c)ontrol group.</p>
</td></tr>
<tr><td><code id="random_d_+3A_var_c">var_c</code></td>
<td>
<p>The population variance of the control group mean.</p>
</td></tr>
<tr><td><code id="random_d_+3A_n_c">N_c</code></td>
<td>
<p>The number of samples of the control mean.  When a non-negative
integer, all control means will be estimated using the same N.  A
vector of unequal N's can also be taken; if so, K will be ignored and the 
number of randomly generated means will equal the length of that vector,
and each mean will be based on each N within the vector.</p>
</td></tr>
<tr><td><code id="random_d_+3A_bias_correction">bias_correction</code></td>
<td>
<p>When <code>"FALSE"</code>, returns Cohen's g effect sizes 
that are not adjusted using a small-sample correction (J).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table with columns of random effect sizes (d) and their 
variances (var_d).
</p>


<h3>References</h3>

<p>Hedges, L.V. 1981. Distribution theory for Glass's estimator of 
effect size and related estimators. Journal of Educational Statistics 
6: 107-128.
</p>
<p>Hedges, L.V. 1982. Estimation of effect size from a series of 
independent experiments. Psychological Bulletin 92: 490-499.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   random_d(K = 5, X_t = 25, var_t = 1, N_t = 15, X_c = 10, var_c = 1, N_c = 15)

</code></pre>

<hr>
<h2 id='random_missingness'>Random generation of missingness in a data frame.</h2><span id='topic+random_missingness'></span>

<h3>Description</h3>

<p>Generates random <code>NA</code>'s in in a column or groups of columns of a data frame.
Used in imputation simulations based on complete datasets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_missingness(aDataFrame, columnNames, percentMissing = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_missingness_+3A_adataframe">aDataFrame</code></td>
<td>
<p>A data.frame where missingness will be simulated.</p>
</td></tr>
<tr><td><code id="random_missingness_+3A_columnnames">columnNames</code></td>
<td>
<p>A string or a vector of strings that describe the column 
names (labels) where missingness will be simulated.</p>
</td></tr>
<tr><td><code id="random_missingness_+3A_percentmissing">percentMissing</code></td>
<td>
<p>The percentage of missingness within specified columns.
&quot;Percent missing&quot; uses a binomial distribution to simulate missing data.
Default is 10 (i.e. 10% missing).  Use <code><a href="#topic+impute_missingness">impute_missingness</a></code> for
a summary of these randomly generated missing data.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table with columns of missing data (specified as <code>NA</code>'s).
</p>

<hr>
<h2 id='random_N'>Random generation of sample sizes (N) for study outcomes.</h2><span id='topic+random_N'></span>

<h3>Description</h3>

<p>Generates random sample sizes (N) by either sampling from a Negative Binomial 
or Poisson distribution.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_N(K, method = "NegativeBinomial", mean = 15, min = 3, NB_size = 15)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_N_+3A_k">K</code></td>
<td>
<p>Number of sample sizes to generate.</p>
</td></tr>
<tr><td><code id="random_N_+3A_method">method</code></td>
<td>
<p>A string that defines what sampling distribution to generate
random N.  The default is <code>"NegativeBinomial"</code> but a <code>"Poisson"</code> 
distribution can also be used.</p>
</td></tr>
<tr><td><code id="random_N_+3A_mean">mean</code></td>
<td>
<p>The population mean (mu) if <code>"NegativeBinomial"</code>, or the lambda 
(dispersion parameter) if <code>"Poisson"</code>.  The default is 15, which will 
generate sample sizes that on average will center around N = 15.</p>
</td></tr>
<tr><td><code id="random_N_+3A_min">min</code></td>
<td>
<p>A non-negative integer that specifies the minimum sample size that
can be generated.  Default is N = 3.</p>
</td></tr>
<tr><td><code id="random_N_+3A_nb_size">NB_size</code></td>
<td>
<p>Dispersion parameter for the <code>"Negative Binomial"</code> distribution
that must be strictly positive, but need not be integer.  Default is 15, 
which creates a long tail for random N's ranging to about N = 60.  Increase
value to create a longer tail of random sample sizes.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of random sample sizes (N).
</p>

<hr>
<h2 id='random_OR'>Random generation of odds ratio (OR) effect sizes.</h2><span id='topic+random_OR'></span>

<h3>Description</h3>

<p>Generates random odds ratios, logged odds ratios, and their 
variances (Cornfield 1951).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_OR(K, p_A, N_A, p_B, N_B, continuity = 0.5, logged = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_OR_+3A_k">K</code></td>
<td>
<p>Number of effect sizes to generate.</p>
</td></tr>
<tr><td><code id="random_OR_+3A_p_a">p_A</code></td>
<td>
<p>The odds of the event of interest for Group A.  A probability 
ranging from zero to one.</p>
</td></tr>
<tr><td><code id="random_OR_+3A_n_a">N_A</code></td>
<td>
<p>The total number of samples of Group A.</p>
</td></tr>
<tr><td><code id="random_OR_+3A_p_b">p_B</code></td>
<td>
<p>The odds of the event of interest for Group B.  A probability 
ranging from zero to one.</p>
</td></tr>
<tr><td><code id="random_OR_+3A_n_b">N_B</code></td>
<td>
<p>The total number of samples of Group B.</p>
</td></tr>
<tr><td><code id="random_OR_+3A_continuity">continuity</code></td>
<td>
<p>Odds ratios with zero events cannot be computed.  Following, 
Cox (1970), a continuity correction can be added to each cell of the 2 by 2
table to help improve this problem of zero events within the table.  The
default value added is 0.5.</p>
</td></tr>
<tr><td><code id="random_OR_+3A_logged">logged</code></td>
<td>
<p>When <code>"FALSE"</code>, returns non-logged transformed 
odds ratios and appropriate variances.  Default is TRUE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table with columns of random effect sizes (OR) and their 
variances.
</p>


<h3>References</h3>

<p>Cornfield, J. 1951. A method for estimating comparative rates
from Clinical Data. Applications to cancer of the lung, breast, and cervix.
Journal of the National Cancer Institute 11: 1269-1275.
</p>
<p>Cox, D.R. 1970. The continuity correction. Biometrika 57: 217-219.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   random_OR(K = 5, p_A = 0.3, N_A = 100, p_B = 0.1, N_B = 60)

</code></pre>

<hr>
<h2 id='random_pairedN'>Random generation of paired sample sizes (N) for study outcomes.</h2><span id='topic+random_pairedN'></span>

<h3>Description</h3>

<p>Generates random paired sample sizes (N).  For example, sample sizes for a
treatment group and samples sizes for a control group.  These paired N 
are often correlated within studies.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_pairedN(K, mean = 15, min = 3, correlation = 0.95)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_pairedN_+3A_k">K</code></td>
<td>
<p>Number of paired sample sizes to generate.</p>
</td></tr>
<tr><td><code id="random_pairedN_+3A_mean">mean</code></td>
<td>
<p>The lambda (dispersion parameter) of a Poisson distribution.  The
default is 15, which will generate sample sizes that on average will center
around N = 15.</p>
</td></tr>
<tr><td><code id="random_pairedN_+3A_min">min</code></td>
<td>
<p>A non-negative integer that specifies the minimum sample size that
can be generated.  Default is N = 3.</p>
</td></tr>
<tr><td><code id="random_pairedN_+3A_correlation">correlation</code></td>
<td>
<p>A correlation ranging from zero to one that specifies how
'similar' the paired sample sizes will be to one another.  Default is 0.95 
(i.e. the paired sample sizes will be highly correlated).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table of paired random sample sizes (N).
</p>

<hr>
<h2 id='random_r'>Random generation of correlation coefficients.</h2><span id='topic+random_r'></span>

<h3>Description</h3>

<p>Generates random correlation coefficients (r or Pearson product-moment 
correlation coefficients) and their variances (Pearson 1895).  Also provides 
Fisher z-transformed correlation coefficients (Fisher 1915).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_r(K = 100, correlation = 0.5, N = 10, Fisher_Z = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_r_+3A_k">K</code></td>
<td>
<p>Number of effect sizes to generate.</p>
</td></tr>
<tr><td><code id="random_r_+3A_correlation">correlation</code></td>
<td>
<p>The mean population correlation coefficient (rho) to
simulate.  Must range between -1 to 1.</p>
</td></tr>
<tr><td><code id="random_r_+3A_n">N</code></td>
<td>
<p>The number of samples used to estimate each correlation coefficient.
When a non-negative integer, all r will be estimated using the same N.  A
vector of unequal N's can also be taken; if so, K will be ignored and the 
number of randomly generated r will equal the length of that vector.</p>
</td></tr>
<tr><td><code id="random_r_+3A_fisher_z">Fisher_Z</code></td>
<td>
<p>When <code>TRUE</code>, also returns the Fisher z-transformed 
correlation coefficients and their variances (Fisher 1915).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table with columns of random effect sizes (r), their variances
and sample sizes.
</p>


<h3>References</h3>

<p>Pearson, K. 1895. Notes on regression and inheritance in the 
case of two parents. Proceedings of the Royal Society of 
London 58: 240-242.
</p>
<p>Fisher, R.A. 1915. Frequency distribution of the values of the
correlation coefficient in samples of an indefinitely large 
population. Biometrika 10: 507-521.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   random_r(K = 5, correlation = 0.5, N = 50)

</code></pre>

<hr>
<h2 id='random_RR'>Random generation of log response ratio (RR) effect sizes.</h2><span id='topic+random_RR'></span>

<h3>Description</h3>

<p>Generates random log response ratios and their variances (Hedges et al. 1999).
NOTE: samples from a log-normal distribution to generate non-negative control
and treatment means (following Lajeunesse 2015).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>random_RR(K, X_t, var_t, N_t, X_c, var_c, N_c)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="random_RR_+3A_k">K</code></td>
<td>
<p>Number of effect sizes to generate.</p>
</td></tr>
<tr><td><code id="random_RR_+3A_x_t">X_t</code></td>
<td>
<p>The population mean (mu) of the (t)reatment group (numerator of 
ratio).</p>
</td></tr>
<tr><td><code id="random_RR_+3A_var_t">var_t</code></td>
<td>
<p>The population variance of the treatment group mean.</p>
</td></tr>
<tr><td><code id="random_RR_+3A_n_t">N_t</code></td>
<td>
<p>The number of samples of the treatment mean.  When a non-negative
integer, all treatment means will be estimated using the same N.  A
vector of unequal N's can also be taken; if so, K will be ignored and the 
number of randomly generated means will equal the length of that vector,
and each mean will be based on each N within the vector.</p>
</td></tr>
<tr><td><code id="random_RR_+3A_x_c">X_c</code></td>
<td>
<p>The population mean (mu) of the (c)ontrol group (denominator of 
ratio).</p>
</td></tr>
<tr><td><code id="random_RR_+3A_var_c">var_c</code></td>
<td>
<p>The population variance of the control group mean.</p>
</td></tr>
<tr><td><code id="random_RR_+3A_n_c">N_c</code></td>
<td>
<p>The number of samples of the control mean.  When a non-negative
integer, all control means will be estimated using the same N.  A
vector of unequal N's can also be taken; if so, K will be ignored and the 
number of randomly generated means will equal the length of that vector,
and each mean will be based on each N within the vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data table with columns of random effect sizes (RR) and their 
variances.
</p>


<h3>References</h3>

<p>Hedges, L.V., J. Gurevitch, and P.S. Curtis. 1999. The 
meta-analysis of response ratios in experimental ecology. Ecology 80: 
1150-1156.
</p>
<p>Lajeunesse, M.J. 2015. Bias and correction for the log response
ratio used in ecological meta-analysis. Ecology.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>   random_RR(K = 5, X_t = 25, var_t = 1, N_t = 15, X_c = 10, var_c = 1, N_c = 15)

</code></pre>

<hr>
<h2 id='replicate_MetaWin2.0'>Replicate meta-analysis results and summaries from MetaWin 2.0.</h2><span id='topic+replicate_MetaWin2.0'></span>

<h3>Description</h3>

<p>Replicate meta-analysis results and summaries from Rosenberg's et al. (2000)
software 'MetaWin' 2.0.  Currently only replicates moderator analyses and not
meta-regressions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replicate_MetaWin2.0(
  model,
  weights,
  effects_model = "random",
  data,
  bootstraps = 999
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replicate_MetaWin2.0_+3A_model">model</code></td>
<td>
<p>A two-sided linear formula object describing the model, with the
response (effect sizes) on the left of a ~ operator and the moderator 
variables, separated by +, :, * operators, on the right.  NOTE:  
MetaWin was limited to analyses with a single moderator variable.  
This function currently supports only categorical moderators.</p>
</td></tr>
<tr><td><code id="replicate_MetaWin2.0_+3A_weights">weights</code></td>
<td>
<p>A vector of effect size variances that will be used as 
weights for the meta-analysis.</p>
</td></tr>
<tr><td><code id="replicate_MetaWin2.0_+3A_effects_model">effects_model</code></td>
<td>
<p>The default is <code>"random"</code>, which specifies a 
random-effects meta-analysis.  Other options include <code>"fixed"</code> 
which presents fixed-effect analyses.</p>
</td></tr>
<tr><td><code id="replicate_MetaWin2.0_+3A_data">data</code></td>
<td>
<p>An optional data frame containing the variables named in model
and weights.</p>
</td></tr>
<tr><td><code id="replicate_MetaWin2.0_+3A_bootstraps">bootstraps</code></td>
<td>
<p>The number of bootstraps used to estimate confidence 
intervals.  As with 'MetaWin' 2.0, the default is 999.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Rosenberg, M.S., Adams, D.C., and Gurevitch, J. 2000. 
MetaWin: Statistical Software for Meta-Analysis. Sinauer Associates 
Sunderland, Massachusetts.
</p>

<hr>
<h2 id='replicate_phyloMeta1.3'>Replicate phylogeneic meta-analysis results and summaries from phyloMeta 1.3.</h2><span id='topic+replicate_phyloMeta1.3'></span>

<h3>Description</h3>

<p>Replicate phylogenetic meta-analysis results and summaries from Lajeunesse
(2011) software 'phyloMeta' 1.3.  Currently does not fully replicate all
functionality.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replicate_phyloMeta1.3(model, weights, data, phylogenyFile)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="replicate_phyloMeta1.3_+3A_model">model</code></td>
<td>
<p>A two-sided linear formula object describing the model, with the
response (effect sizes) on the left of a ~ operator and the moderator
variables, separated by +, :, * operators, on the right.  NOTE:
phyloMeta was limited to analyses with a single moderator variable.
This function currently supports only numerical categorical moderators.</p>
</td></tr>
<tr><td><code id="replicate_phyloMeta1.3_+3A_weights">weights</code></td>
<td>
<p>A vector of effect size variances that will be used as
weights for the meta-analysis.</p>
</td></tr>
<tr><td><code id="replicate_phyloMeta1.3_+3A_data">data</code></td>
<td>
<p>A data frame containing the variables named in model
and weights and species names (names must be exact as specified in
phylogeny).</p>
</td></tr>
<tr><td><code id="replicate_phyloMeta1.3_+3A_phylogenyfile">phylogenyFile</code></td>
<td>
<p>A text file containing a NEWICK phylogeny. The number of
species must be same as (k) number of effect sizes in data.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Lajeunesse, M.J. (2011) phyloMeta: a program for phylogenetic
comparative analyses with meta-analysis. Bioinformatics 27, 2603-2604.
</p>

<hr>
<h2 id='scrape_bibliography'>Attempts to scrape/extract bibliographic data from Web of Science.</h2><span id='topic+scrape_bibliography'></span>

<h3>Description</h3>

<p>A not so elegant way to extract bibliographic data of a research article by
scraping the contents of Web of Science (WOS).  Requires the DOI (digital
object identifier) of an article, as well as web access with an institutional
subscription to WOS.  Note: This function is not suited to extract data
for book chapters available on WOS.  Current extractions include: a vector
of authors (author), publication year (year), article title (title), journal
title (journal), journal volume (volume), page numbers (pages), abstract
(abstract), number of references (N_references), number of citations
(N_citations), journal impact factor (journal_IF), and the year the journal
impact factor was released (journal_IF_year).  Finally the date of the scrape
is also provided (date_scraped). Bulleted abstracts or those with subheadings
or subparagraphs will not be extracted properly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>scrape_bibliography(DOI, quiet = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="scrape_bibliography_+3A_doi">DOI</code></td>
<td>
<p>A string as the DOI (digital object identifier) of a research article.</p>
</td></tr>
<tr><td><code id="scrape_bibliography_+3A_quiet">quiet</code></td>
<td>
<p>When <code>TRUE</code>, does not print an MLA-style reference of the
extracted article.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of bibliographic extractions and a timestamp of the scrape.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# use DOI to scrape number of WOS citations of a research article
data(example_references_metagear)
someRefs &lt;- effort_initialize(example_references_metagear)
theWOSRef &lt;- scrape_bibliography(someRefs$DOI[1])
print(paste("citations = ", theWOSRef$N_citations))


## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
