<!DOCTYPE html><html><head><title>Help for package metadat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metadat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#dat.aloe2013'><p>Studies on the Association Between Supervision Quality and Various Outcomes in Social, Mental Health, and Child Welfare Workers</p></a></li>
<li><a href='#dat.anand1999'><p>Studies on the Effectiveness of Oral Anticoagulants in Patients with Coronary Artery Disease</p></a></li>
<li><a href='#dat.assink2016'><p>Studies on the Association between Recidivism and Mental Health</p></a></li>
<li><a href='#dat.axfors2021'><p>Mortality Outcomes with Hydroxychloroquine and Chloroquine in COVID-19 from an International Collaborative Meta-Analysis of Randomized Trials</p></a></li>
<li><a href='#dat.bakdash2021'><p>Dataset on Situation Awareness and Task Performance Associations</p></a></li>
<li><a href='#dat.baker2009'><p>Studies on Pharmacologic Treatments for Chronic Obstructive Pulmonary Disease</p></a></li>
<li><a href='#dat.bangertdrowns2004'><p>Studies on the Effectiveness of Writing-to-Learn Interventions</p></a></li>
<li><a href='#dat.baskerville2012'><p>Studies on the Effectiveness of Practice Facilitation Interventions</p></a></li>
<li><a href='#dat.bcg'><p>Studies on the Effectiveness of the BCG Vaccine Against Tuberculosis</p></a></li>
<li><a href='#dat.begg1989'><p>Studies on Bone-Marrow Transplantation versus Chemotherapy for the Treatment of Leukemia</p></a></li>
<li><a href='#dat.berkey1998'><p>Studies on Treatments for Periodontal Disease</p></a></li>
<li><a href='#dat.besson2016'><p>Dataset on How Maternal Diet Impacts Copying Styles in Rodents</p></a></li>
<li><a href='#dat.bonett2010'><p>Studies on the Reliability of the CES-D Scale</p></a></li>
<li><a href='#dat.bornmann2007'><p>Studies on Gender Differences in Grant and Fellowship Awards</p></a></li>
<li><a href='#dat.bourassa1996'><p>Studies on the Association between Handedness and Eye-Dominance</p></a></li>
<li><a href='#dat.cannon2006'><p>Studies on the Effectiveness of Intensive Versus Moderate Statin Therapy for Preventing Coronary Death or Myocardial Infarction</p></a></li>
<li><a href='#dat.cohen1981'><p>Studies on the Relationship between Course Instructor Ratings and Student Achievement</p></a></li>
<li><a href='#dat.colditz1994'><p>Studies on the Effectiveness of the BCG Vaccine Against Tuberculosis</p></a></li>
<li><a href='#dat.collins1985a'><p>Studies on the Treatment of Upper Gastrointestinal Bleeding by a Histamine H2 Antagonist</p></a></li>
<li><a href='#dat.collins1985b'><p>Studies on the Effects of Diuretics in Pregnancy</p></a></li>
<li><a href='#dat.craft2003'><p>Studies on the Relationship between the Competitive State Anxiety Inventory-2 and Sport Performance</p></a></li>
<li><a href='#dat.crede2010'><p>Studies on the Relationship between Class Attendance and Grades in College Students</p></a></li>
<li><a href='#dat.curtis1998'><p>Studies on the Effects of Elevated CO2 Levels on Woody Plant Mass</p></a></li>
<li><a href='#dat.dagostino1998'><p>Studies on the Effectiveness of Antihistamines in Reducing Symptoms of the Common Cold</p></a></li>
<li><a href='#dat.damico2009'><p>Studies on Topical plus Systemic Antibiotics to Prevent Respiratory Tract Infections</p></a></li>
<li><a href='#dat.debruin2009'><p>Studies on Standard Care Quality and HAART-Adherence</p></a></li>
<li><a href='#dat.dogliotti2014'><p>Studies on Antithrombotic Treatments to Prevent Strokes</p></a></li>
<li><a href='#dat.dong2013'><p>Studies on Safety of Inhaled Medications for Chronic Obstructive Pulmonary Disease</p></a></li>
<li><a href='#dat.dorn2007'><p>Studies on Complementary and Alternative Medicine for Irritable Bowel Syndrome</p></a></li>
<li><a href='#dat.egger2001'><p>Studies on the Effectiveness of Intravenous Magnesium in Acute Myocardial Infarction</p></a></li>
<li><a href='#dat.fine1993'><p>Studies on Radiation Therapy with or without Adjuvant Chemotherapy in Patients with Malignant Gliomas</p></a></li>
<li><a href='#dat.franchini2012'><p>Studies on Dopamine Agonists to Reduce &ldquo;Off-Time&rdquo; in Patients with Advanced Parkinson Disease</p></a></li>
<li><a href='#dat.frank2008'><p>Studies on the Association Between the CASP8 -652 6N del Promoter Polymorphism and Breast Cancer Risk</p></a></li>
<li><a href='#dat.gibson2002'><p>Studies on the Effectiveness of Self-Management Education and Regular Medical Review for Adults with Asthma</p></a></li>
<li><a href='#dat.graves2010'><p>Studies on the Effectiveness of Injected Cholera Vaccines</p></a></li>
<li><a href='#dat.gurusamy2011'><p>Studies on Interventions to Reduce Mortality after Liver Transplantation</p></a></li>
<li><a href='#dat.hackshaw1998'><p>Studies on the Risk of Lung Cancer in Women Exposed to Environmental Tobacco Smoke</p></a></li>
<li><a href='#dat.hahn2001'><p>Studies on the Effectiveness of Different Rehydration Solutions for the Prevention of Unscheduled Intravenous Infusion in Children with Diarrhoea</p></a></li>
<li><a href='#dat.hannum2020'><p>Studies Comparing Objective and Subjective Olfactory Loss in COVID-19 Patients</p></a></li>
<li><a href='#dat.hart1999'><p>Studies on the Effectiveness of Warfarin for Preventing Strokes</p></a></li>
<li><a href='#dat.hartmannboyce2018'><p>Studies on the Effectiveness of Nicotine Replacement Therapy for Smoking Cessation</p></a></li>
<li><a href='#dat.hasselblad1998'><p>Studies on the Effectiveness of Counseling for Smoking Cessation</p></a></li>
<li><a href='#dat.hine1989'><p>Studies on Prophylactic Use of Lidocaine After a Heart Attack</p></a></li>
<li><a href='#dat.ishak2007'><p>Studies on Deep-Brain Stimulation in Patients with Parkinson's disease</p></a></li>
<li><a href='#dat.kalaian1996'><p>Studies on the Effectiveness of Coaching for the SAT</p></a></li>
<li><a href='#dat.kearon1998'><p>Studies on the Accuracy of Venous Ultrasonography for the Diagnosis of Deep Venous Thrombosis</p></a></li>
<li><a href='#dat.knapp2017'><p>Studies on Differences in Planning Performance in Schizophrenia Patients versus Healthy Controls</p></a></li>
<li><a href='#dat.konstantopoulos2011'><p>Studies on the Effects of Modified School Calendars on Student Achievement</p></a></li>
<li><a href='#dat.landenberger2005'><p>Studies on the Effectiveness of CBT for Reducing Recidivism</p></a></li>
<li><a href='#dat.laopaiboon2015'><p>Studies on the Effectiveness of Azithromycin for Treating Lower Respiratory Tract Infections</p></a></li>
<li><a href='#dat.lau1992'><p>Studies on Intravenous Streptokinase for Acute Myocardial Infarction</p></a></li>
<li><a href='#dat.lee2004'><p>Studies on Acupoint P6 Stimulation for Preventing Nausea</p></a></li>
<li><a href='#dat.lehmann2018'><p>The Effect of Red on Perceived Attractiveness</p></a></li>
<li><a href='#dat.li2007'><p>Studies on the Effectiveness of Intravenous Magnesium in Acute Myocardial Infarction</p></a></li>
<li><a href='#dat.lim2014'><p>Studies on the Association Between Maternal Size, Offspring Size, and Number of Offsprings</p></a></li>
<li><a href='#dat.linde2005'><p>Studies on the Effectiveness of St. John's Wort for Treating Depression</p></a></li>
<li><a href='#dat.linde2015'><p>Studies on Classes of Antidepressants for the Primary Care Setting</p></a></li>
<li><a href='#dat.linde2016'><p>Studies on Antidepressants for the Primary Care Setting</p></a></li>
<li><a href='#dat.lopez2019'><p>Studies on the Effectiveness of CBT for Depression</p></a></li>
<li><a href='#dat.maire2019'><p>Studies on Temporal Trends in Fish Community Structures in French Rivers</p></a></li>
<li><a href='#dat.mccurdy2020'><p>Studies on the Generation Effect</p></a></li>
<li><a href='#dat.mcdaniel1994'><p>Studies on the Validity of Employment Interviews</p></a></li>
<li><a href='#dat.michael2013'><p>The Non-Persuasive Power of a Brain Image</p></a></li>
<li><a href='#dat.molloy2014'><p>Studies on the Relationship between Conscientiousness and Medication Adherence</p></a></li>
<li><a href='#dat.moura2021'><p>Studies on Assortative Mating</p></a></li>
<li><a href='#dat.nakagawa2007'><p>Assessing the Function of House Sparrows' Bib Size Using a Flexible Meta-Analysis Method</p></a></li>
<li><a href='#dat.nielweise2007'><p>Studies on Anti-Infective-Treated Central Venous Catheters for Prevention of Catheter-Related Bloodstream Infections</p></a></li>
<li><a href='#dat.nielweise2008'><p>Studies on Anti-Infective-Treated Central Venous Catheters for Prevention of Catheter-Related Bloodstream Infections</p></a></li>
<li><a href='#dat.normand1999'><p>Studies on the Length of Hospital Stay of Stroke Patients</p></a></li>
<li><a href='#dat.obrien2003'><p>Studies on the Relationship Between BMI and Risk of Preeclampsia</p></a></li>
<li><a href='#dat.pagliaro1992'><p>Studies on the Effectiveness of Nonsurgical Treatments in Cirrhosis</p></a></li>
<li><a href='#dat.pignon2000'><p>Studies on the Effectiveness of Locoregional Treatment plus Chemotherapy for Head and Neck Squamous-Cell Carcinoma</p></a></li>
<li><a href='#dat.pritz1997'><p>Studies on the Effectiveness of Hyperdynamic Therapy for Treating Cerebral Vasospasm</p></a></li>
<li><a href='#dat.raudenbush1985'><p>Studies on Assessing the Effects of Teacher Expectations on Pupil IQ</p></a></li>
<li><a href='#dat.riley2003'><p>Studies on MYC-N as a Prognostic Marker for Neuroblastoma</p></a></li>
<li><a href='#dat.senn2013'><p>Studies on the Effectiveness of Glucose-Lowering Agents</p></a></li>
<li><a href='#dat.stowe2010'><p>Studies on Adjuvant Treatments to Levodopa Therapy for Parkinson disease</p></a></li>
<li><a href='#dat.tannersmith2016'><p>Studies on the Relationship between School Motivation and Criminal Behavior</p></a></li>
<li><a href='#dat.vanhowe1999'><p>Studies on the Association between Circumcision and HIV Infection</p></a></li>
<li><a href='#dat.viechtbauer2021'><p>Studies to Illustrate Model Checking Methods</p></a></li>
<li><a href='#dat.white2020'><p>Studies on the Relationship between Sexual Signal Expression and Individual Quality</p></a></li>
<li><a href='#dat.woods2010'><p>Studies on Treatments for Chronic Obstructive Pulmonary Disease</p></a></li>
<li><a href='#dat.yusuf1985'><p>Studies of Beta Blockers During and After Myocardial Infarction</p></a></li>
<li><a href='#datsearch'><p>Search Function for the Datasets</p></a></li>
<li><a href='#metadat-package'><p>Meta-Analysis Datasets for R</p></a></li>
<li><a href='#prep_dat'><p>Data preparation function</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>1.2-0</td>
</tr>
<tr>
<td>Date:</td>
<td>2022-04-05</td>
</tr>
<tr>
<td>Title:</td>
<td>Meta-Analysis Datasets</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>utils, tools, mathjaxr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>metafor, numDeriv, BiasedUrn, dfoptim, igraph, ape, testthat,
digest, lme4, clubSandwich, meta, netmeta, mvtnorm, gridExtra,
rms</td>
</tr>
<tr>
<td>Description:</td>
<td>A collection of meta-analysis datasets for teaching purposes, illustrating/testing meta-analytic methods, and validating published analyses.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>TRUE</td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>mathjaxr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/wviechtb/metadat">https://github.com/wviechtb/metadat</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/wviechtb/metadat/issues">https://github.com/wviechtb/metadat/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-04-05 18:20:44 UTC; wviechtb</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas White <a href="https://orcid.org/0000-0002-3976-1734"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Daniel Noble <a href="https://orcid.org/0000-0001-9460-8743"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Alistair Senior <a href="https://orcid.org/0000-0001-9805-7280"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  W. Kyle Hamilton <a href="https://orcid.org/0000-0002-8642-7990"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Wolfgang Viechtbauer
    <a href="https://orcid.org/0000-0003-3463-4063"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre],
  Guido Schwarzer <a href="https://orcid.org/0000-0001-6214-9087"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [dtc]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wolfgang Viechtbauer &lt;wvb@metafor-project.org&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-04-06 11:52:37 UTC</td>
</tr>
</table>
<hr>
<h2 id='dat.aloe2013'>Studies on the Association Between Supervision Quality and Various Outcomes in Social, Mental Health, and Child Welfare Workers</h2><span id='topic+dat.aloe2013'></span>

<h3>Description</h3>

<p>Results from 5 studies examining the association between various measures of supervision quality and various work-related outcomes in social, mental health, and child welfare workers.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.aloe2013
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) and year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n</b>     </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tval</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> t-statistic for the test of the association/predictor </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>preds</b> </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of predictors included in the regression model </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>R2</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> the coefficient of determination (i.e., R-squared value) of the regression model </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>
  


<h3>Details</h3>

<p>The dataset is based on studies that used regression models to examine the association between some measure of perceived supervision quality (e.g., the quality of the relationship with one's supervisor) and some work-related outcome (e.g., job satisfaction) in social, mental health, and child welfare workers. The dataset was extracted from Aloe and Thompson (2013), which in turn is a subset of the studies included in the meta-analysis by Mor Barak et al. (2009).
</p>
<p>The dataset can be used to illustrate the meta-analysis of regression models, using measures such as the (semi-)partial correlation coefficient. For this, the t-statistic from the regression model for the association (i.e., predictor) of interest was extracted from each regression model (<code>tval</code>), as well as the sample size (<code>n</code>), the number of predictors included in the regression model (<code>preds</code>), and the coefficient of determination (i.e., R-squared value) of the regression model (<code>R2</code>). Based on this information, the (semi-)partial correlation coefficient can be computed for each study, as well as its corresponding sampling variance. These values can then be meta-analyzed using standard methods.
</p>


<h3>Concepts</h3>

<p>social work, (semi-)partial correlations, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Aloe, A. M., &amp; Thompson, C. G. (2013). The synthesis of partial effect sizes. <em>Journal of the Society for Social Work and Research</em>, <b>4</b>(4), 390&ndash;405. <code style="white-space: pre;">&#8288;https://doi.org/10.5243/jsswr.2013.24&#8288;</code>
</p>


<h3>References</h3>

<p>Mor Barak, M. E., Travis, D. J., Pyun, H., &amp; Xie, B. (2009). The impact of supervision on worker outcomes: A meta-analysis. <em>Social Service Review</em>, <b>83</b>(1), 3&ndash;32. <code style="white-space: pre;">&#8288;https://doi.org/10.1086/599028&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.aloe2013
dat

## Not run: 

### load metafor package
suppressPackageStartupMessages(library(metafor))

### compute the partial correlation coefficients and corresponding sampling variances
dat &lt;- escalc(measure="PCOR", ti=tval, ni=n, mi=preds, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### mixed-effects meta-regression model examining the relationship between the partial
### correlation coefficients and the number of predictors included in the models
res &lt;- rma(yi, vi, mods = ~ preds, data=dat)
res

### compute the r-to-z transformed partial correlation coefficients and their variances
dat &lt;- escalc(measure="ZPCOR", ti=tval, ni=n, mi=preds, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### back-transformation to the partial correlation scale
predict(res, transf=transf.ztor)

### compute the semi-partial correlation coefficients and their variances
dat &lt;- escalc(measure="SPCOR", ti=tval, ni=n, mi=preds, r2i=R2, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.anand1999'>Studies on the Effectiveness of Oral Anticoagulants in Patients with Coronary Artery Disease</h2><span id='topic+dat.anand1999'></span>

<h3>Description</h3>

<p>Results from 34 trials examining the effectiveness of oral anticoagulants in patients with coronary artery disease.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.anand1999
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author(s) or trial name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>intensity</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> intensity of anticoagulation (low, medium, or high) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>asp.t</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> concomitant use of aspirin in the treatment group (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>asp.c</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> concomitant use of aspirin in the control group (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 34 randomized clinical trials that examined the effectiveness of oral anticoagulants in patients with coronary artery disease. The results given here are focused on the total mortality in the treatment versus control groups.
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, odds ratios, Mantel-Haenszel method
</p>


<h3>Note</h3>

<p>Strictly speaking, there are only 31 trials, since Breddin et al. (1980) and ATACS (1990) are multiarm trials.
</p>
<p>According to a correction, <code>dat.anand1999$ci[29]</code> should be 1. But then <code>dat.anand1999$ci[21]</code> would also have to be 1 (if these data indeed refer to the same control group). This appears contradictory, so this correction was not made.
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Anand, S. S., &amp; Yusuf, S. (1999). Oral anticoagulant therapy in patients with coronary artery disease: A meta-analysis. <em>Journal of the American Medical Association</em>, <b>282</b>(21), 2058&ndash;2067. <code style="white-space: pre;">&#8288;https://doi.org/10.1001/jama.282.21.2058&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.anand1999
dat

## Not run: 

### load metafor package
library(metafor)

### High-Intensity OA vs Control
rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat,
       subset=(intensity=="high" &amp; asp.t==0 &amp; asp.c==0), digits=2)

### High- or Moderate-Intensity OA vs Aspirin
rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat,
       subset=(intensity %in% c("high","moderate") &amp; asp.t==0 &amp; asp.c==1), digits=2)

### Moderate-Intensity OA vs Control
rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat,
       subset=(intensity=="moderate" &amp; asp.t==0 &amp; asp.c==0), digits=2)

### High- or Moderate-Intensity OA and Aspirin vs Aspirin
rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat,
       subset=(intensity %in% c("high","moderate") &amp; asp.t==1 &amp; asp.c==1), digits=2)

### Low-Intensity OA and Aspirin vs Aspirin
rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat,
       subset=(intensity=="low" &amp; asp.t==1 &amp; asp.c==1), digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.assink2016'>Studies on the Association between Recidivism and Mental Health</h2><span id='topic+dat.assink2016'></span>

<h3>Description</h3>

<p>Results from 17 studies on the association between recidivism and mental health in delinquent juveniles.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.assink2016
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>esid</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> effect size within study id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>id</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> row id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pubstatus</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> published study (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year of the study (approximately mean centered) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>deltype</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of delinquent behavior in which juveniles could have recidivated (either general, overt, or covert)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies included in this dataset (which is a subset of the data used in Assink et al., 2015) compared the difference in recidivism between delinquent juveniles with a mental health disorder and a comparison group of juveniles without a mental health disorder. Since studies differed in the way recidivism was defined and assessed, results are given in terms of standardized mean differences, with positive values indicating a higher prevalence of recidivism in the group of juveniles with a mental health disorder.
</p>
<p>Multiple effect size estimates could be extracted from most studies (e.g., for different delinquent behaviors in which juveniles could have recidivated), necessitating the use of appropriate models/methods for the analysis. Assink and Wibbelink (2016) illustrate the use of multilevel meta-analysis models for this purpose.
</p>


<h3>Concepts</h3>

<p>psychology, criminology, standardized mean differences, multilevel models, cluster-robust inference
</p>


<h3>Note</h3>

<p>The <code>year</code> variable is not constant within study 3, as this study refers to two different publications using the same data.
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Assink, M., &amp; Wibbelink, C. J. M. (2016). Fitting three-level meta-analytic models in R: A step-by-step tutorial. <em>The Quantitative Methods for Psychology</em>, <b>12</b>(3), 154&ndash;174. <code style="white-space: pre;">&#8288;https://doi.org/10.20982/tqmp.12.3.p154&#8288;</code>
</p>


<h3>References</h3>

<p>Assink, M., van der Put, C. E., Hoeve, M., de Vries, S. L. A., Stams, G. J. J. M., &amp; Oort, F. J. (2015). Risk factors for persistent delinquent behavior among juveniles: A meta-analytic review. <em>Clinical Psychology Review</em>, <b>42</b>, 47&ndash;61. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.cpr.2015.08.002&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.assink2016
head(dat, 9)

## Not run: 

### load metafor package
library(metafor)

### fit multilevel model
res &lt;- rma.mv(yi, vi, random = ~ 1 | study/esid, data=dat)
res

### use cluster-robust inference methods
robust(res, cluster=study)

### LRTs for the variance components
res0 &lt;- rma.mv(yi, vi, random = ~ 1 | study/esid, data=dat, sigma2=c(0,NA))
anova(res0, res)
res0 &lt;- rma.mv(yi, vi, random = ~ 1 | study/esid, data=dat, sigma2=c(NA,0))
anova(res0, res)

### examine some potential moderators via meta-regression
rma.mv(yi, vi, mods = ~ pubstatus, random = ~ 1 | study/esid, data=dat)
rma.mv(yi, vi, mods = ~ year, random = ~ 1 | study/esid, data=dat)
dat$deltype &lt;- relevel(factor(dat$deltype), ref="general")
rma.mv(yi, vi, mods = ~ deltype, random = ~ 1 | study/esid, data=dat)
rma.mv(yi, vi, mods = ~ year + deltype, random = ~ 1 | study/esid, data=dat)

### assume that the effect sizes within studies are correlated with rho=0.6
V &lt;- vcalc(vi, cluster=study, obs=esid, data=dat, rho=0.6)
round(V[dat$study %in% c(1,2), dat$study %in% c(1,2)], 4)

### fit multilevel model using this approximate V matrix
res &lt;- rma.mv(yi, V, random = ~ 1 | study/esid, data=dat)
res

### use cluster-robust inference methods
robust(res, cluster=study)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.axfors2021'>Mortality Outcomes with Hydroxychloroquine and Chloroquine in COVID-19 from an International Collaborative Meta-Analysis of Randomized Trials</h2><span id='topic+dat.axfors2021'></span>

<h3>Description</h3>

<p>Results from 33 trials examining the effectiveness of hydroxychloroquine or chloroquine in patients with COVID-19.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.axfors2021
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>                </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> registry number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>acronym</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> shortened registry number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>patient_setting</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> patient setting </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>blinding_exact</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study blinding </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>high_dose</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> high or low dose of medication </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Published</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> publication status </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hcq_cq</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> medication type (hcq = hydroxychloroquine or cq = chloroquine) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hcq_arm_event</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hcq_arm_total</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>control_arm_event</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>control_arm_total</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Control</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> control group type (Standard of Care or Placebo)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 33 published and unpublished randomized clinical trials that examined the effectiveness of hydroxychloroquine or chloroquine in patients with COVID-19. The results given here are focused on the total mortality in the treatment versus control groups.
</p>


<h3>Concepts</h3>

<p>medicine, covid-19, odds ratios
</p>


<h3>Author(s)</h3>

<p>W. Kyle Hamilton <a href="mailto:whamilton@ucmerced.edu">whamilton@ucmerced.edu</a> <a href="https://kylehamilton.com">https://kylehamilton.com</a>
</p>


<h3>Source</h3>

<p>Axfors, C., Schmitt, A., Janiaud, P., van ’t Hooft, J., Moher, D., Goodman, S., … Hemkens, L. G. (2021, March 9). Hydroxychloroquine and chloroquine for survival in COVID-19: An international collaborative meta-analysis of randomized trials. <code style="white-space: pre;">&#8288;https://doi.org/10.17605/OSF.IO/QESV4&#8288;</code>
</p>


<h3>References</h3>

<p>Axfors, C., Schmitt, A. M., Janiaud, P., van’t Hooft, J., Abd-Elsalam, S., Abdo, E. F., Abella, B. S., Akram, J., Amaravadi, R. K., Angus, D. C., Arabi, Y. M., Azhar, S., Baden, L. R., Baker, A. W., Belkhir, L., Benfield, T., Berrevoets, M. A. H., Chen, C.-P., Chen, T.-C., … Hemkens, L. G. (2021). Mortality outcomes with hydroxychloroquine and chloroquine in COVID-19 from an international collaborative meta-analysis of randomized trials. Nature Communications, 12(1), 2349. <code style="white-space: pre;">&#8288;https://doi.org/10.1038/s41467-021-22446-z&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># copy data into 'dat' and examine data
dat &lt;- dat.axfors2021
dat

## Not run: 

# load metafor package
library(metafor)

# calculate log odds ratios and corresponding sampling variances
dat &lt;- escalc(measure="OR", ai=hcq_arm_event, n1i=hcq_arm_total,
                            ci=control_arm_event, n2i=control_arm_total, data=dat)

# meta-analysis Hydroxychloroquine
res_hcq &lt;- rma(yi, vi, subset=(hcq_cq=="hcq"), slab = id, data=dat)
print(res_hcq, digits=2)

# meta-analysis Chloroquine
res_cq &lt;- rma(yi, vi, subset=(hcq_cq=="cq"), slab = id, data=dat)
print(res_cq, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.bakdash2021'>Dataset on Situation Awareness and Task Performance Associations</h2><span id='topic+dat.bakdash2021'></span>

<h3>Description</h3>

<p>Results from 77 papers with 678 effects evaluating associations among measures of situation awareness and task performance.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.bakdash2021
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Author</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> paper author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Year</b>              </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> year of paper publication </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Title</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> title of paper </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DOI</b>               </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> digital object identifier (DOI) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DTIC.link</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> permanent link for Defense Technical Information Collection (DITC) reports; see: <code style="white-space: pre;">&#8288;https://www.dtic.mil&#8288;</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>SA.measure.type</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of SA measure </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Sample.size</b>       </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> reported sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Sample.size.stats</b> </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> reported sample size based on reported statistics (this reflects excluded participants) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>es.z</b>              </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> z-transformed correlation coefficient; includes ghost results (disclosed and undisclosed non-significant effects not reported in detail) imputed using the draw method described in Bakdash et al. (2021a) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi.z</b>              </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> variance for z-transformed correlation (calculated using <code>Sample.size.stats</code>, <em>not</em> <code>Sample.size</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>SampleID</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> unique identifier for each experiment/study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Outcome</b>           </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> unique value for each effect size
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset contains behavioral experiments from 77 papers/79 studies with a total of 678 effects, evaluating associations among measures of situation awareness (&ldquo;knowing what is going on&rdquo;) and task performance. Examples of situation awareness include knowledge of current vehicle speed in a simulated driving task and location and heading of aircraft in a simulated air traffic control task. Corresponding examples of task performance include &ldquo;the number of collisions in a simulated driving task&rdquo; and &ldquo;subject matter expert rating of conflict management in a simulated air control task&rdquo; (Bakdash et al. 2021a, p. 2). This dataset and the &lsquo;Examples&rsquo; are a highly simplified version of the data and code in Bakdash et al. (2021b; 2021c). The journal article by Bakdash et al. (2021a) describes the systematic review and meta-analysis in detail.
</p>
<p>This dataset is used to illustrate multilevel multivariate meta-analytic models for the overall pooled effect and pooled effects by situation awareness measure. We also adjust meta-analytic models using cluster-robust variance estimation / cluster-robust inference with the <code><a href="metafor.html#topic+robust">robust</a></code> function in <em>metafor</em>. Results are shown graphically in a customized forest plot with a prediction interval (estimated plausible range of individual effects). Last, we create a table summarizing the estimated meta-analytic heterogeneity parameters.
</p>
<p>The meta-analytic results show most pooled effect sizes in the positive medium range or less. There was also substantial meta-analytic heterogeneity (estimated systematic variance in true effects), nearing the magnitude of the overall pooled effect. We interpret the meta-analytic results as situation awareness typically having limited validity for task performance (i.e., good situation awareness does not tend to have strong probabilistic links with good performance and vice-versa). More formally, measures of situation awareness do not generally and meaningfully capture cognitive processes and other relevant factors underlying task performance.
</p>


<h4>Run-Time</h4>

<p>The code run-time can be greatly sped-up using a linear algebra library with <em>R</em> that makes use of multiple CPU cores. See: <a href="https://www.metafor-project.org/doku.php/tips:speeding_up_model_fitting">https://www.metafor-project.org/doku.php/tips:speeding_up_model_fitting</a>. To measure the run-time, uncomment these three lines: <code>start.time &lt;- Sys.time()</code>, <code>end.time &lt;- Sys.time()</code>, and <code>end.time - start.time</code>. Run-times on Windows 10 x64 with the Intel Math Kernel Library are:
</p>

<table>
<tr>
 <td style="text-align: right;">
   </td><td style="text-align: left;"> <em>CPU</em> </td><td style="text-align: left;"> <em>Run-Time (Minutes)</em> </td>
</tr>
<tr>
 <td style="text-align: right;">
   </td><td style="text-align: left;"> i7-11850H  </td><td style="text-align: left;"> 2.49 </td>
</tr>
<tr>
 <td style="text-align: right;">
   </td><td style="text-align: left;"> i7-4770    </td><td style="text-align: left;"> 5.38 </td>
</tr>
<tr>
 <td style="text-align: right;">
   </td>
</tr>

</table>




<h3>Concepts</h3>

<p>psychology, human factors, engineering, correlation coefficients, multilevel models, multivariate models, cluster-robust inference
</p>


<h3>Author(s)</h3>

<p>Jonathan Bakdash, <a href="mailto:jonathan.z.bakdash.civ@army.mil">jonathan.z.bakdash.civ@army.mil</a>, <a href="mailto:jbakdash@gmail.com">jbakdash@gmail.com</a> <br />
Laura Marusich, <a href="mailto:laura.m.cooper20.civ@army.mil">laura.m.cooper20.civ@army.mil</a>, <a href="mailto:lmarusich@gmail.com">lmarusich@gmail.com</a>
</p>


<h3>Source</h3>

<p>Bakdash, J. Z., Marusich, L. R., Cox, K. R., Geuss, M. N., Zaroukian, E. G., &amp; Morris, K. M. (2021b). The validity of situation awareness for performance: A meta-analysis (Code Ocean Capsule). <code style="white-space: pre;">&#8288;https://doi.org/10.24433/CO.1682542.v4&#8288;</code>
</p>
<p>Bakdash, J. Z., Marusich, L. R., Cox, K. R., Geuss, M. N., Zaroukian, E. G., &amp; Morris, K. M. (2021c). The validity of situation awareness for performance: A meta-analysis (Systematic Review, Data, and Code). <code style="white-space: pre;">&#8288;https://doi.org/10.17605/OSF.IO/4K7ZV&#8288;</code>
</p>


<h3>References</h3>

<p>Bakdash, J. Z., Marusich, L. R., Cox, K. R., Geuss, M. N., Zaroukian, E. G., &amp; Morris, K. M. (2021a). The validity of situation awareness for performance: A meta-analysis. <em>Theoretical Issues in Ergonomics Science</em>, 1&ndash;24. <code style="white-space: pre;">&#8288;https://doi.org/10.1080/1463922X.2021.1921310&#8288;</code>
</p>
<p>Supplemental materials: <code style="white-space: pre;">&#8288;https://www.tandfonline.com/doi/suppl/10.1080/1463922X.2021.1921310/suppl_file/ttie_a_1921310_sm5524.docx&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.bakdash2021
head(dat[c(1,2,6,8:12)])

## Not run: 
#start.time &lt;- Sys.time()

### load metafor
library(metafor)

### multilevel meta-analytic model to get the overall pooled effect
res.overall &lt;- rma.mv(es.z, vi.z, mods = ~ 1,
                      random = ~ 1 | SampleID / Outcome,
                      data = dat,
                      test = "t")
res.overall

### get prediction interval
predict(res.overall)

### cluster-robust variance estimation (CRVE) / cluster-robust inference
res.overall.crve &lt;- robust(res.overall, cluster = SampleID)
res.overall.crve

### get prediction interval
res.overall.crve.pred &lt;- predict(res.overall.crve)
res.overall.crve.pred

### multilevel meta-analytic model for SA measures
res.sa &lt;-  rma.mv(es.z, vi.z, mods = ~ SA.measure.type - 1,
                  random = ~ 1 | SampleID / Outcome,
                  data = dat,
                  test = "t")
res.sa

### cluster-robust variance estimation (CRVE) / cluster-robust inference
res.sa.crve &lt;- robust(res.sa, cluster = SampleID)
res.sa.crve

### profile likelihood plots
par(mfrow=c(2,1))
profile(res.sa.crve, progbar = FALSE)

### format and combine output of meta-analytic models for the forest plot
all.z        &lt;- c(res.sa.crve$beta,            # SA measures
                  res.overall.crve$beta,       # pooled effect for confidence interval (CI)
                  res.overall.crve$beta)       # pooled effect for prediction interval (PI)

all.ci.lower &lt;- c(res.sa.crve$ci.lb,           # SA measures
                  res.overall.crve.pred$ci.lb, # pooled effect, lower CI
                  res.overall.crve.pred$pi.lb) # pooled effect, lower PI

all.ci.upper &lt;- c(res.sa.crve$ci.ub,           # SA measures
                  res.overall.crve.pred$ci.ub, # pooled effect, upper CI
                  res.overall.crve.pred$pi.ub) # pooled effect, upper PI

### note: there is no p-value for the PI
all.pvals  &lt;- c(res.sa.crve$pval, res.overall.crve$pval)
all.labels &lt;- c(sort(unique(dat$SA.measure.type)), "Overall", "95% Prediction Interval")

### function to round p-values for the forest plot
pvals.round &lt;- function(input) {
  input &lt;- ifelse(input &lt; 0.001, "&lt; 0.001",
           ifelse(input &lt; 0.01, "&lt; 0.01",
           ifelse(input &lt; 0.05 &amp; input &gt;= 0.045, "&lt; 0.05",
           ifelse(round(input, 2) == 1.00, "0.99",
           sprintf("%.2f", round(input, 2))))))}

all.pvals.rounded &lt;- pvals.round(all.pvals)

### forest plot
plot.vals &lt;- data.frame(all.labels, all.z, all.ci.lower, all.ci.upper)

par(mfrow=c(1,1), cex = 1.05)
forest(plot.vals$all.z,
       ci.lb = plot.vals$all.ci.lower,
       ci.ub = plot.vals$all.ci.upper,
       slab  = plot.vals$all.labels,
       psize = 1,
       efac = 0, xlim = c(-1.8, 2.5), clim = c(-1, 1),
       transf = transf.ztor, # transform z to r
       at = seq(-0.5, 1, by = 0.25),
       xlab = expression("Correlation Coefficient"~"("*italic('r')*")"),
       main = "\n\n\nSA Measures",
       ilab = c(all.pvals.rounded, ""), ilab.xpos = 2.45, ilab.pos = 2.5,
       digits = 2, refline = 0, annotate = FALSE)

### keep trailing zero using sprintf
output &lt;- cbind(sprintf("%.2f", round(transf.ztor(plot.vals$all.z), 2)),
                sprintf("%.2f", round(transf.ztor(plot.vals$all.ci.lower), 2)),
                sprintf("%.2f", round(transf.ztor(plot.vals$all.ci.upper), 2)))

### alignment kludge
annotext &lt;- apply(output, 1, function(x) {paste0("  ", x[1], " [", x[2],", ", x[3], "]")})
text( 1.05, 12:1, annotext, pos = 4, cex = 1.05)
text(-1.475, 14.00, "SA Measure", cex = 1.05)
text( 2.30,  14.00, substitute(paste(italic('p-value'))), cex = 1.05)
text( 1.55,  14.00, "Correlation [95% CI]", cex = 1.05)
abline(h = 1.5)

### black polygon for overall mean CIs
addpoly(all.z[11], ci.lb = all.ci.lower[11], ci.ub = all.ci.upper[11],
        rows = 2, annotate = FALSE, efac = 1.5, transf = transf.ztor)

### white polygon for PI
addpoly(all.z[12], ci.lb = all.ci.lower[12], ci.ub = all.ci.upper[12],
        rows = 1, col = "white", border = "black",
        annotate = FALSE, efac = 1.5, transf = transf.ztor)

par(mfrow=c(1,1), cex = 1) # reset graph parameters to default

### confidence intervals for the variance components
re.CI.variances &lt;- confint(res.overall)
re.CI.variances

sigma1.z &lt;- data.frame(re.CI.variances[[1]]["random"])
sigma2.z &lt;- data.frame(re.CI.variances[[2]]["random"])

### fit model using alternative multivariate parameterization
res.overall.alt &lt;- rma.mv(es.z, vi.z, mods = ~ 1,
                          random = ~ factor(Outcome) | factor(SampleID),
                          data = dat,
                          test = "t")

### confidence intervals for the total amount of heterogeneity variance component
res.overall.alt.tau &lt;- confint(res.overall.alt, tau2=1)$random

### I^2: http://www.metafor-project.org/doku.php/tips:i2_multilevel_multivariate
W &lt;- diag(1/dat$vi.z)
X &lt;- model.matrix(res.overall)
P &lt;- W - W %*% X %*% solve(t(X) %*% W %*% X) %*% t(X) %*% W

### I^2 (variance due to heterogeneity): 61%
I2 &lt;- 100 * res.overall.alt$tau2 /
      (res.overall.alt$tau2 + (res.overall$k-res.overall$p)/sum(diag(P)))
I2

### 95% CI for I^2 using uncertainty around tau^2
I2.CI.lb &lt;- 100 * res.overall.alt.tau[1,2] /
            (res.overall.alt.tau[1,2] + (res.overall$k-res.overall$p)/sum(diag(P)))
I2.CI.lb

I2.CI.ub &lt;- 100 * res.overall.alt.tau[1,3] /
            (res.overall.alt.tau[1,3] + (res.overall$k-res.overall$p)/sum(diag(P)))
I2.CI.ub

### total amount of heterogeneity (tau)
sqrt(res.overall.alt$tau2)

### heterogeneity table
table.heterogeneity &lt;- data.frame(matrix(ncol = 3, nrow = 4))
colnames(table.heterogeneity) &lt;- c("Parameter Value",
                                   "Lower 95% CI",
                                   "Upper 95% CI")
rownames(table.heterogeneity) &lt;- c("Tau (Total)",
                                   "Tau1 (Between paper)",
                                   "Tau2 (Within paper)",
                                   "I2 (%)")

table.heterogeneity[1,] &lt;- res.overall.alt.tau[2,]
table.heterogeneity[2,] &lt;- sigma1.z[2,]
table.heterogeneity[3,] &lt;- sigma2.z[2,]
table.heterogeneity[4,] &lt;- c(I2, I2.CI.lb, I2.CI.ub)

round(table.heterogeneity, 2)

#end.time &lt;- Sys.time()
#end.time - start.time


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.baker2009'>Studies on Pharmacologic Treatments for Chronic Obstructive Pulmonary Disease</h2><span id='topic+dat.baker2009'></span>

<h3>Description</h3>

<p>Results from 39 trials examining pharmacologic treatments for chronic obstructive pulmonary disease (COPD).</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.baker2009
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>id</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>exac</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of individuals with one or more COPD exacerbations </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>total</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of individuals
</td>
</tr>

</table>



<h3>Details</h3>

<p>This data set comes from a systematic review of randomized controlled trials on pharmacologic treatments for chronic obstructive pulmonary disease (COPD) (Baker et al., 2009).
</p>
<p>The primary outcome, occurrence of one or more episodes of COPD exacerbation, is binary (yes / no). For this outcome, five drug treatments (fluticasone, budesonide, salmeterol, formoterol, tiotropium) and two combinations (fluticasone + salmeterol, budesonide + formoterol) were compared to placebo. The authors considered the two combinations as separate treatments instead of evaluating the individual components.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, network meta-analysis, component network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Baker, W. L., Baker, E. L., &amp; Coleman, C. I. (2009). Pharmacologic treatments for chronic obstructive pulmonary disease: A mixed-treatment comparison meta-analysis. <em>Pharmacotherapy: The Journal of Human Pharmacology and Drug Therapy</em>, <b>29</b>(8), 891&ndash;905. <code style="white-space: pre;">&#8288;https://doi.org/10.1592/phco.29.8.891&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="netmeta.html#topic+pairwise">pairwise</a></code>, <code><a href="meta.html#topic+metabin">metabin</a></code>,
<code><a href="netmeta.html#topic+netmeta">netmeta</a></code>, <code><a href="netmeta.html#topic+netcomb">netcomb</a></code>,
<code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show first 6 rows of the dataset
head(dat.baker2009)

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Transform data from long arm-based format to contrast-based
### format. Argument 'sm' has to be used for odds ratio as summary
### measure; by default the risk ratio is used in the metabin function
### called internally.
pw &lt;- pairwise(treatment, exac, total, studlab = paste(study, year),
  data = dat.baker2009, sm = "OR")

### Conduct random effects network meta-analysis (NMA)
### with placebo as reference
net &lt;- netmeta(pw, fixed = FALSE, ref = "plac")

### Show network graph
netgraph(net, seq = "optimal", start = "prcomp",
  labels = gsub("+", " +\n", trts, fixed = TRUE),
  plastic = TRUE, thickness = "se.fixed", number = TRUE,
  points = TRUE, cex.points = 5, col.points = "red",
  offset = 0.025)

### Print and plot results for network meta-analysis
net
forest(net)

### Conduct component network meta-analysis (CNMA)
cnet &lt;- netcomb(net)
cnet

### Compare results of NMA and additive CNMA
nb &lt;- netbind(net, cnet, name = c("Standard NMA", "Additive CNMA"))
forest(nb)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.bangertdrowns2004'>Studies on the Effectiveness of Writing-to-Learn Interventions</h2><span id='topic+dat.bangertdrowns2004'></span>

<h3>Description</h3>

<p>Results from 48 studies on the effectiveness of school-based writing-to-learn interventions on academic achievement. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.bangertdrowns2004
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>grade</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> grade level (1 = elementary; 2 = middle; 3 = high-school; 4 = college) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>length</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment length (in weeks) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>minutes</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> minutes per assignment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>wic</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> writing tasks were completed in class (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>feedback</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> feedback on writing was provided (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>info</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> writing contained informational components (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pers</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> writing contained personal components (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>imag</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> writing contained imaginative components (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>meta</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> prompts for metacognitive reflection (0 = no; 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>subject</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> subject matter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total sample size of the study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance
</td>
</tr>

</table>



<h3>Details</h3>

<p>In each of the studies included in this meta-analysis, an experimental group (i.e., a group of students that received instruction with increased emphasis on writing tasks) was compared against a control group (i.e., a group of students that received conventional instruction) with respect to some content-related measure of academic achievement (e.g., final grade, an exam/quiz/test score). The outcome measure for this meta-analysis was the standardized mean difference (with positive values indicating a higher mean level of academic achievement in the intervention group).
</p>
<p>The standardized mean differences given here are bias-corrected and therefore differ slightly from the values reported in the article. Also, since only the total sample size is given in the article, the sampling variances were computed under the assumption that \(n_{i1} = n_{i2} = n_i / 2\).
</p>


<h3>Concepts</h3>

<p>education, standardized mean differences, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Bangert-Drowns, R. L., Hurley, M. M., &amp; Wilkinson, B. (2004). The effects of school-based writing-to-learn interventions on academic achievement: A meta-analysis. <em>Review of Educational Research</em>, <b>74</b>(1), 29&ndash;58. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/00346543074001029&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.bangertdrowns2004
dat[1:10,-13]

## Not run: 

### load metafor package
library(metafor)

### fit random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### some examples of mixed-effects meta-regression models
res &lt;- rma(yi, vi, mods = ~ factor(grade), data=dat)
res
res &lt;- rma(yi, vi, mods = ~ length, data=dat)
res
res &lt;- rma(yi, vi, mods = ~ info + pers + imag + meta, data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.baskerville2012'>Studies on the Effectiveness of Practice Facilitation Interventions</h2><span id='topic+dat.baskerville2012'></span>

<h3>Description</h3>

<p>Results from 23 studies on the effectiveness of practice facilitation interventions within the primary care practice setting.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.baskerville2012
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>author</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>score</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> quality score (0 to 12 scale) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>design</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study design (cct = controlled clinical trial, rct = randomized clinical trial, crct = cluster randomized clinical trial) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>alloconc</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> allocation concealed (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>blind</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> single- or double-blind study (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>itt</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intention to treat analysis (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>fumonths</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> follow-up months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>retention</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> retention (in percent) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> country where study was conducted </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>outcomes</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of outcomes assessed </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>duration</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> duration of intervention </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pperf</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> practices per facilitator </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>meetings</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> (average) number of meetings </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hours</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> (average) hours per meeting </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tailor</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention tailored to the context and needs of the practice (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>smd</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>se</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding standard error
</td>
</tr>

</table>



<h3>Details</h3>

<p>Baskerville et al. (2012) describe outreach or practice facilitation as a &quot;multifaceted approach that involves skilled individuals who enable others, through a range of intervention components and approaches, to address the challenges in implementing evidence-based care guidelines within the primary care setting&quot;. The studies included in this dataset examined the effectiveness of practice facilitation interventions for improving some relevant evidence-based practice behavior. The effect was quantified in terms of a standardized mean difference, comparing the change (from pre- to post-intervention) in the intervention versus the comparison group (or the difference from baseline in prospective cohort studies).
</p>


<h3>Concepts</h3>

<p>medicine, primary care, standardized mean differences, publication bias, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Baskerville, N. B., Liddy, C., &amp; Hogg, W. (2012). Systematic review and meta-analysis of practice facilitation within primary care settings. <em>Annals of Family Medicine</em>, <b>10</b>(1), 63&ndash;74. <code style="white-space: pre;">&#8288;https://doi.org/10.1370/afm.1312&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.baskerville2012
dat

## Not run: 

### load metafor package
library(metafor)

### random-effects model
res &lt;- rma(smd, sei=se, data=dat, method="DL")
print(res, digits=2)

### funnel plot
funnel(res, xlab="Standardized Mean Difference", ylim=c(0,0.6))

### rank and regression tests for funnel plot asymmetry
ranktest(res)
regtest(res)

### meta-regression analyses examining various potential moderators
rma(smd, sei=se, mods = ~ score, data=dat, method="DL")
rma(smd, sei=se, mods = ~ alloconc, data=dat, method="DL")
rma(smd, sei=se, mods = ~ blind,    data=dat, method="DL")
rma(smd, sei=se, mods = ~ itt,      data=dat, method="DL")
rma(smd, sei=se, mods = ~ duration, data=dat, method="DL")
rma(smd, sei=se, mods = ~ tailor,   data=dat, method="DL")
rma(smd, sei=se, mods = ~ pperf,    data=dat, method="DL")
rma(smd, sei=se, mods = ~ I(meetings * hours), data=dat, method="DL")


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.bcg'>Studies on the Effectiveness of the BCG Vaccine Against Tuberculosis</h2><span id='topic+dat.bcg'></span>

<h3>Description</h3>

<p>Results from 13 studies examining the effectiveness of the Bacillus Calmette-Guerin (BCG) vaccine against tuberculosis. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.bcg
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tpos</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB positive cases in the treated (vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tneg</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB negative cases in the treated (vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cpos</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB positive cases in the control (non-vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cneg</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB negative cases in the control (non-vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ablat</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> absolute latitude of the study location (in degrees) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>alloc</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> method of treatment allocation (random, alternate, or systematic assignment)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 13 studies provide data in terms of \(2 \times 2\) tables in the form:
</p>

<table>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: center;"> TB positive </td><td style="text-align: center;"> TB negative </td>
</tr>
<tr>
 <td style="text-align: left;">
   vaccinated group </td><td style="text-align: center;"> <code>tpos</code> </td><td style="text-align: center;"> <code>tneg</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   control group    </td><td style="text-align: center;"> <code>cpos</code> </td><td style="text-align: center;"> <code>cneg</code>
   </td>
</tr>

</table>
<p> The goal of the meta-analysis was to examine the overall effectiveness of the BCG vaccine for preventing tuberculosis and to examine moderators that may potentially influence the size of the effect.
</p>
<p>The dataset has been used in several publications to illustrate meta-analytic methods (see &lsquo;References&rsquo;).
</p>


<h3>Concepts</h3>

<p>medicine, risk ratios, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Colditz, G. A., Brewer, T. F., Berkey, C. S., Wilson, M. E., Burdick, E., Fineberg, H. V., &amp; Mosteller, F. (1994). Efficacy of BCG vaccine in the prevention of tuberculosis: Meta-analysis of the published literature. <em>Journal of the American Medical Association</em>, <b>271</b>(9), 698&ndash;702. <code style="white-space: pre;">&#8288;https://doi.org/10.1001/jama.1994.03510330076038&#8288;</code>
</p>


<h3>References</h3>

<p>Berkey, C. S., Hoaglin, D. C., Mosteller, F., &amp; Colditz, G. A. (1995). A random-effects regression model for meta-analysis. <em>Statistics in Medicine</em>, <b>14</b>(4), 395&ndash;411. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.4780140406&#8288;</code>
</p>
<p>van Houwelingen, H. C., Arends, L. R., &amp; Stijnen, T. (2002). Advanced methods in meta-analysis: Multivariate approach and meta-regression. <em>Statistics in Medicine</em>, <b>21</b>(4), 589&ndash;624. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.1040&#8288;</code>
</p>
<p>Viechtbauer, W. (2010). Conducting meta-analyses in R with the metafor package. <em>Journal of Statistical Software</em>, <b>36</b>(3), 1&ndash;48. <code style="white-space: pre;">&#8288;https://doi.org/10.18637/jss.v036.i03&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.bcg
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log risk ratios and corresponding sampling variances
dat &lt;- escalc(measure="RR", ai=tpos, bi=tneg, ci=cpos, di=cneg, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### average risk ratio with 95% CI
predict(res, transf=exp)

### mixed-effects model with absolute latitude and publication year as moderators
res &lt;- rma(yi, vi, mods = ~ ablat + year, data=dat)
res

### predicted average risk ratios for 10-60 degrees absolute latitude
### holding the publication year constant at 1970
predict(res, newmods=cbind(seq(from=10, to=60, by=10), 1970), transf=exp)

### note: the interpretation of the results is difficult because absolute
### latitude and publication year are strongly correlated (the more recent
### studies were conducted closer to the equator)
plot(ablat ~ year, data=dat, pch=19, xlab="Publication Year", ylab="Absolute Lattitude")
cor(dat$ablat, dat$year)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.begg1989'>Studies on Bone-Marrow Transplantation versus Chemotherapy for the Treatment of Leukemia</h2><span id='topic+dat.begg1989'></span>

<h3>Description</h3>

<p>Results from controlled and uncontrolled studies on the effectiveness of allogeneic bone-marrow transplantation (BMT) and conventional chemotherapy (CMO) in the treatment of acute nonlymphocytic leukemia.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.begg1989
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trt</b>        </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> treatment (BMT or CMO) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>arms</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of arms in the study (1 = uncontrolled studies; 2 = controlled studies) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> 2-year disease-free survival rates </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sei</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> corresponding standard errors </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> corresponding sampling variances
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from controlled and uncontrolled studies on the 2-year disease-free survival rate in patients with acute nonlymphocytic leukemia receiving either allogeneic bone-marrow transplantation (BMT) or conventional chemotherapy (CMO). In the controlled (two-arm) studies (studies 1-4), a cohort of patients in complete remission and potentially eligible for BMT was assembled, and those who consented and for whom a donor could be found received BMT, with the remaining patients used as controls (receiving CMO). In the uncontrolled (one-arm) studies (studies 5-16), only a single group was studied, receiving either BMT or CMO.
</p>
<p>The data in this dataset were obtained from Table 1 in Begg and Pilote (1991, p. 902).
</p>


<h3>Concepts</h3>

<p>medicine, oncology, single-arm studies, multilevel models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Begg, C. B., &amp; Pilote, L. (1991). A model for incorporating historical controls into a meta-analysis. <em>Biometrics</em>, <b>47</b>(3), 899&ndash;906. <code style="white-space: pre;">&#8288;https://doi.org/10.2307/2532647&#8288;</code>
</p>


<h3>References</h3>

<p>Begg, C. B., Pilote, L., &amp; McGlave, P. B. (1989). Bone marrow transplantation versus chemotherapy in acute non-lymphocytic leukemia: A meta-analytic review. <em>European Journal of Cancer and Clinical Oncology</em>, <b>25</b>(11), 1519&ndash;1523. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/0277-5379(89)90291-5&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.begg1989
dat

## Not run: 

### load metafor package
library(metafor)

### turn trt and arms into factors and set reference levels
dat$trt  &lt;- relevel(factor(dat$trt), ref="CMO")
dat$arms &lt;- relevel(factor(dat$arms), ref="2")

### create data frame with the treatment differences for the controlled studies
dat2 &lt;- data.frame(yi = dat$yi[c(1,3,5,7)] - dat$yi[c(2,4,6,8)],
                   vi = dat$vi[c(1,3,5,7)] + dat$vi[c(2,4,6,8)])
dat2

### DerSimonian and Laird method using the treatment differences
res &lt;- rma(yi, vi, data=dat2, method="DL", digits=2)
res

### Begg &amp; Pilote (1991) model incorporating the uncontrolled studies
res &lt;- rma.mv(yi, vi, mods = ~ trt, random = ~ 1 | study,
              data=dat, method="ML", digits=2)
res

### model involving bias terms for the uncontrolled studies
res &lt;- rma.mv(yi, vi, mods = ~ trt + trt:arms, random = ~ 1 | study,
              data=dat, method="ML", digits=2)
res

### model with a random treatment effect
res &lt;- rma.mv(yi, vi, mods = ~ trt, random = list(~ 1 | study, ~ trt | study),
              struct="UN", tau2=c(0,NA), rho=0, data=dat, method="ML", digits=2)
res

### model with a random treatment effect, but with equal variances in both arms
res &lt;- rma.mv(yi, vi, mods = ~ trt, random = list(~ 1 | study, ~ trt | study),
              struct="CS", rho=0, data=dat, method="ML", digits=2)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.berkey1998'>Studies on Treatments for Periodontal Disease</h2><span id='topic+dat.berkey1998'></span>

<h3>Description</h3>

<p>Results from 5 trials comparing surgical and non-surgical treatments for medium-severity periodontal disease one year after treatment. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.berkey1998
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> trial number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of patients </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>outcome</b>    </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> outcome (PD = probing depth; AL = attachment level) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed mean difference in outcome (surgical versus non-surgical) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>v1i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> variances and covariances of the observed effects </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>v2i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> variances and covariances of the observed effects
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 5 trials that compared surgical and non-surgical methods for the treatment of medium-severity periodontal disease. Reported outcomes include the change in probing depth (PD) and attachment level (AL) one year after the treatment. The outcome measure used for this meta-analysis was the (raw) mean difference, calculated in such a way that positive values indicate that surgery was more effective than non-surgical treatment in decreasing the probing depth and increasing the attachment level (so, the results from the various trials indicate that surgery is preferable for reducing the probing depth, while non-surgical treatment is preferable for increasing the attachment level). Since each trial provides effect size estimates for both outcomes, the estimates are correlated. A multivariate model can be used to meta-analyze the two outcomes simultaneously.
</p>
<p>The <code>v1i</code> and <code>v2i</code> values are the variances and covariances of the observed effects. In particular, for each study, variables <code>v1i</code> and <code>v2i</code> form a \(2 \times 2\) variance-covariance matrix of the observed effects, with the diagonal elements corresponding to the sampling variances of the mean differences (the first for probing depth, the second for attachment level) and the off-diagonal value corresponding to the covariance of the two mean differences. Below, the full (block diagonal) variance-covariance for all studies is constructed from these two variables.
</p>


<h3>Concepts</h3>

<p>medicine, dentistry, raw mean differences, multivariate models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Berkey, C. S., Antczak-Bouckoms, A., Hoaglin, D. C., Mosteller, F., &amp; Pihlstrom, B. L. (1995). Multiple-outcomes meta-analysis of treatments for periodontal disease. <em>Journal of Dental Research</em>, <b>74</b>(4), 1030&ndash;1039. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/00220345950740040201&#8288;</code>
</p>
<p>Berkey, C. S., Hoaglin, D. C., Antczak-Bouckoms, A., Mosteller, F., &amp; Colditz, G. A. (1998). Meta-analysis of multiple outcomes by regression with random effects. <em>Statistics in Medicine</em>, <b>17</b>(22), 2537&ndash;2550. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/(sici)1097-0258(19981130)17:22&lt;2537::aid-sim953&gt;3.0.co;2-c&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.berkey1998
dat

## Not run: 

### load metafor package
library(metafor)

### construct block diagonal var-cov matrix of the observed outcomes based on variables v1i and v2i
V &lt;- vcalc(vi=1, cluster=author, rvars=c(v1i, v2i), data=dat)

### fit multiple outcomes (meta-regression) model (with REML estimation)
res &lt;- rma.mv(yi, V, mods = ~ outcome - 1, random = ~ outcome | trial, struct="UN", data=dat)
print(res, digits=3)

### test/estimate difference between the two outcomes
anova(res, X=c(1,-1))

### fit model including publication year as moderator for both outcomes (with ML estimation)
res &lt;- rma.mv(yi, V, mods = ~ outcome + outcome:I(year - 1983) - 1,
              random = ~ outcome | trial, struct="UN", data=dat, method="ML")
print(res, digits=3)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.besson2016'>Dataset on How Maternal Diet Impacts Copying Styles in Rodents</h2><span id='topic+dat.besson2016'></span>

<h3>Description</h3>

<p>Results from 46 studies synthesising maternal nutritional effects on coping styles in rodents.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.besson2016
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>comp_ID</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> effect-size unique identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study_ID</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study unique identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dam_ID</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> dam unique identifier (group of dams subjected to the same treatment) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>animal_ID</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> offspring unique identifier (group of offspring from the same dam group subjected to the same treatment) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Reference</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author’s names and date </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>species</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> species [rats or mice] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>strain</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> strain  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>manip_type</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> maternal nutritional manipulation type [protein or calorie] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>manip_direction</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> direction of maternal nutritional manipulation [- = restriction, + = overfeeding]  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nom_manip_val</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> degree of maternal nutritional manipulation as described in the original publications [% = percentage of caloric or protein restriction, # = increase in caloric intake]   </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>exp</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> percentage of caloric or protein maternal restriction or increase in caloric intake of the experimental group  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>control</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> percentage of caloric or protein maternal restriction or increase in caloric intake for the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>manip_parameter</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> protein content, percentage fat or intake  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vitmin_eql</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> were vitamins equalized across maternal diets? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>adlib_con</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> were maternal control groups fed ad libitum? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>adlib_exp</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> were maternal experimental groups fed ad libitum? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diet_con</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> name of maternal control diet?  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diet_exp</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> name of maternal experimental diet?  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dam_diet_start_dPC</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> start of the dam diet [in days post-conception] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dam_diet_end_dPC</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> end of the dam diet [in days post-conception] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diet_label</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> period of maternal diet manipulation [pregestation = pre-gestation, pre = pregnancy, lact = lactation, or pre+lact = pregnancy and lactation] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>age_mating</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> dam age at mating if known </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n_con_dam</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> sample size of the control dam groups </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n_exp_dam</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> sample size of the experimental dam groups </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>multi_use_con</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> were control groups used multiple time? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dam_housing</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> how were dams housed? [pair, group, or single] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>temperature</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> temperature during the experiment [°C] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>photoperiod</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> photoperiod during the experiment [number of hours of light] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>litter_size</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> size of the litter [number of pups per dam] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>litter_size_equalized</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> has litter size been equalized? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>crossfostered</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> have pups been cross-fostered? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sex</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> sex of the offspring that were tested [m = male, f = female, both = mixed sex] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>housing</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> offspring housing during the test period [dam, pair, single, or group] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bodymass_mean_contr</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> mean body mass of control offspring close to or during the testing period [g] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bodymass_SE_contr</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> S.E. for body mass of control offspring close to or during the testing period </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bodymass_mean_exp</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> mean body mass of experimental offspring close to or during the testing period [g] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bodymass_SE_exp</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> S.E. for body mass of experimental offspring close to or during the testing period </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bm_N_contr</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> sample size for body mass of control offspring close to or during the testing period </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bm_N_exp</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> sample size for body mass of experimental offspring close to or during the testing period </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bm_dPP</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> age of offspring when body mass was measured [in days post-parturition] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>offspring_diet</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> offspring diet after weaning [type of control diet] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>offspring_con_adlib</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> were control offspring fed ad libitum after weaning? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>offspring_diet_level</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> name of offspring diet after weaning  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>offspring_diet_end_dPP</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> end of the offspring diet [in days post-parturition]  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>post_diet_adlib</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> were experimental offspring fed ad libitum after weaning? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>response_age_dPP</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> offspring age when behavioural testing started [in days post-parturition] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>authors_behaviour_classification</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author's classification of offspring behaviour [anxiety, exploration, or activity] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>our_behaviour_classification</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> our classification of offspring behaviour [anxiety, exploration, or activity] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>response_test</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of test used [elevated T-maze (ETM), open field, etc.] to measure offspring behaviour </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>time_trial</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> duration of the testing [min] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>measure</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> measures taken during testing [total distance moved, time spent in open arm, etc.] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>unit</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> unit of the behavioural measure taken [min, s, m, number (#), etc.] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>high_better</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> for activity and exploration, a higher number is assumed to be better (i.e., animals were more active), but the opposite was assumed for anxiety (i.e., they were more anxious) [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>night.day</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> time of day when behaviours were measured [night or day]  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>comparison</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> for a given control-treatment group comparison, animal group codes as used in the original article [e.g., LP, HP]. This field allows identification of exactly which data (i.e., comparison of which pairs of groups) were extracted from the original paper, and is not used in our analyses. For our analyses the groups were re-coded as control/experimental. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>exp_mean</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> mean of the offspring behaviour measured for the experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>exp_se</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> S.E. of the offspring behaviour measured for the experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>exp_n</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> sample size for the offspring experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>con_mean</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> mean of offspring behaviour measured for the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>con_se</b> </td><td style="text-align: left;"> <code>numeric</code> </td><td style="text-align: left;"> S.E. of the offspring behaviour measured for the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>con_n</b> </td><td style="text-align: left;"> <code>integer</code> </td><td style="text-align: left;"> sample size for the offspring control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>con_ID</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> identifier for shared control groups within experiment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>percentage</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> is the offspring behaviour measure a percentage? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Data_source</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> figure or table number in the original paper from which the data were extracted </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>measure_comments</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> any comments on the offspring behaviour measures </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>SE_imputed</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> was S.E. imputed for the offspring behaviour measure? [yes or no] </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Comments</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> any comments on the data </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>
  


<h3>Details</h3>

<p>Data from experiments where dams were subject to caloric or protein restriction or were overfed around gestation were included. Offspring activity, exploration, or anxiety were measured outcomes variables from maternal experimental treatments. Multilevel meta-analysis and meta-regression models were used to analyze the meta-analytic data.
</p>


<h3>Concepts</h3>

<p>ecology, evolution, standardized mean differences
</p>


<h3>Author(s)</h3>

<p>Daniel Noble, <a href="mailto:daniel.noble@anu.edu.au">daniel.noble@anu.edu.au</a>
</p>


<h3>Source</h3>

<p>Besson, A. A., Lagisz, M., Senior, A. M., Hector, K. L., &amp; Nakagawa, S. (2016). Effect of maternal diet on offspring coping styles in rodents: A systematic review and meta-analysis. <em>Biological Reviews</em>, <b>91</b>(4), 1065&ndash;1080. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/brv.12210&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.besson2016
head(dat)

## Not run: 

### load metafor
library(metafor)

### compute SD from SE
dat$sd_c &lt;- with(dat, con_se * sqrt(con_n))
dat$sd_e &lt;- with(dat, exp_se * sqrt(exp_n))

### compute standardized mean differences and corresponding sampling variances
dat &lt;- escalc(measure="SMD", m1i=exp_mean, m2i=con_mean, sd1i=sd_e, sd2i=sd_c,
              n1i=exp_n, n2i=con_n, data=dat, add.measure=TRUE)

### fit model
mod1 &lt;- rma.mv(yi ~ 1, V = vi, random = list(~ 1 | study_ID, ~ 1 | comp_ID), data = dat)
mod1


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.bonett2010'>Studies on the Reliability of the CES-D Scale</h2><span id='topic+dat.bonett2010'></span>

<h3>Description</h3>

<p>Results from 9 studies on the reliability of the Center for Epidemiologic Studies Depression (CES-D) Scale administered to children providing care to an elderly parent.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.bonett2010
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>source</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> source of data </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of items in the scale </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed value of Cronbach's alpha </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>caregivers</b> </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> gender of the children in the sample
</td>
</tr>

</table>



<h3>Details</h3>

<p>The Center for Epidemiologic Studies Depression (CES-D) Scale is a 20-item questionnaire assessing various symptoms of depression, with each item scored on a 4-point scale. The scale has been used in several studies to examine depressive symptoms in children providing care to an elderly parent. The dataset includes information on the reliability of the scale as measured with Cronbach's alpha in 9 such studies. Also, the gender composition of the children in each sample is indicated.
</p>


<h3>Concepts</h3>

<p>psychology, Cronbach's alpha, reliability generalization, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Bonett, D. G. (2010). Varying coefficient meta-analytic methods for alpha reliability. <em>Psychological Methods</em>, <b>15</b>(4), 368&ndash;385. <code style="white-space: pre;">&#8288;https://doi.org/10.1037/a0020142&#8288;</code>
</p>


<h3>References</h3>

<p>Bonett, D. G. (2002). Sample size requirements for testing and estimating coefficient alpha. <em>Journal of Educational and Behavioral Statistics</em>, <b>27</b>(4), 335&ndash;340. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/10769986027004335&#8288;</code>
</p>
<p>Hakstian, A. R., &amp; Whalen, T. E. (1976). A k-sample significance test for independent alpha coefficients. <em>Psychometrika</em>, <b>41</b>(2), 219&ndash;231. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/BF02291840&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.bonett2010
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis using the raw alpha values
res &lt;- rma(measure="ARAW", ai=ai, mi=mi, ni=ni, data=dat)
res

### meta-analysis using transformed alpha values (using the
### transformation suggested by Hakstian &amp; Whalen, 1976)
res &lt;- rma(measure="AHW", ai=ai, mi=mi, ni=ni, data=dat)
res
predict(res, transf=transf.iahw)

### meta-analysis using transformed alpha values (using the
### transformation suggested by Bonett, 2002)
res &lt;- rma(measure="ABT", ai=ai, mi=mi, ni=ni, data=dat)
res
predict(res, transf=transf.iabt)

### forest plot
forest(res, slab=source, header=TRUE, atransf=transf.iabt, refline=coef(res))

### examine whether female/mixed samples yield different alphas (with raw alphas)
res &lt;- rma(measure="ARAW", ai=ai, mi=mi, ni=ni, mods = ~ caregivers, data=dat)
res
predict(res, newmods=c(0,1), digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.bornmann2007'>Studies on Gender Differences in Grant and Fellowship Awards</h2><span id='topic+dat.bornmann2007'></span>

<h3>Description</h3>

<p>Results from 21 studies on gender differences in grant and fellowship awards.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.bornmann2007
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study reference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>obs</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> observation within study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>doctype</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> document type </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>gender</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> gender of the study authors </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> (average) cohort year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>org</b>        </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> funding organization / program </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> country of the funding organization / program </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>type</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> fellowship or grant application </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>discipline</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> discipline / field </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>waward</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women who received a grant/fellowship award </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>wtotal</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women who applied for an award </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>maward</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of men who received a grant/fellowship award </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mtotal</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of men who applied for an award
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies in this dataset examine whether the chances of receiving a grant or fellowship award differs for men and women. Note that many studies provide multiple comparisons (e.g., for different years / cohorts / disciplines). A multilevel meta-analysis model can be used to account for the multilevel structure in these data.
</p>


<h3>Concepts</h3>

<p>sociology, odds ratios, multilevel models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Bornmann, L., Mutz, R., &amp; Daniel, H. (2007). Gender differences in grant peer review: A meta-analysis. <em>Journal of Informetrics</em>, <b>1</b>(3), 226&ndash;238. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.joi.2007.03.001&#8288;</code>
</p>


<h3>References</h3>

<p>Marsh, H. W., Bornmann, L., Mutz, R., Daniel, H.-D., &amp; O'Mara, A. (2009). Gender effects in the peer reviews of grant proposals: A comprehensive meta-analysis comparing traditional and multilevel approaches. <em>Review of Educational Research</em>, <b>79</b>(3), 1290&ndash;1326. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/0034654309334143&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.bornmann2007
head(dat, 16)

## Not run: 

### load metafor package
library(metafor)

### calculate log odds ratios and corresponding sampling variances
dat &lt;- escalc(measure="OR", ai=waward, n1i=wtotal, ci=maward, n2i=mtotal, data=dat)

### fit multilevel meta-analysis model
res &lt;- rma.mv(yi, vi, random = ~ 1 | study/obs, data=dat)
res

### estimated average odds ratio (with 95% CI/PI)
predict(res, transf=exp, digits=2)

### test for a difference between fellowship and grant applications
res &lt;- rma.mv(yi, vi, mods = ~ type, random = ~ 1 | study/obs, data=dat)
res
predict(res, newmods=0:1, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.bourassa1996'>Studies on the Association between Handedness and Eye-Dominance</h2><span id='topic+dat.bourassa1996'></span>

<h3>Description</h3>

<p>Results from 47 studies on the association between handedness and eye-dominance. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.bourassa1996
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sample</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>selection</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> selection of subjects on the basis of eyedness or handedness </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>investigator</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> investigator (psychologist, educationalist, or other) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hand_assess</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> method to assess handedness (questionnaire or performance based) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>eye_assess</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> method to assess eyedness (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mage</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean age of sample </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>lh.le</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of left-handed left-eyed individuals </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>lh.re</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of left-handed right-eyed individuals </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>rh.le</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of right-handed left-eyed individuals </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>rh.re</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of right-handed right-eyed individuals </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sex</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> sex of the sample (combined, male, or female)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 47 studies included in this meta-analysis examined the association between handedness and eye-dominance (ocular dominance or eyedness). Results are given in terms of \(2 \times 2\) tables, indicating the number of left-handed left-eyed, left-handed right-eyed, right-handed left-eyed, and right-handed right-eyed individuals. Note that some studies included multiple (independent) samples, so that the meta-analysis included 54 samples in total. Also, for some studies, the combined data of the males and females are further broken down into the two subgroups.
</p>
<p>In some studies, there was indication that the selection of subjects was not random with respect to handedness and/or eyedness. While this should not influence the size of the association as measured with the odds ratio, this invalidates those studies for assessing the overall percentage of left-eyed and left-handed individuals.
</p>
<p>Handedness was assessed in the individual studies either based on a questionnaire or inventory or based on task performance. Eyedness was assessed based on various methods: <code>E.1</code> methods are based on task performance, while <code>E.2.a</code> denotes assessment based on a questionnaire. The performance based methods could be further broken down into: <code>E.1.a.i</code> (monocular procedure with object/instrument held in one hand), <code>E.1.a.ii</code> (monocular procedure with object/instrument held in both hands), <code>E.1.b</code> (binocular procedure), <code>E.1.c</code> (a combination of the previous methods), and <code>E.1.d</code> (some other method).
</p>


<h3>Concepts</h3>

<p>psychology, odds ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Bourassa, D. C., McManus, I. C., &amp; Bryden, M. P. (1996). Handedness and eye-dominance: A meta-analysis of their relationship. <em>Laterality</em>, <b>1</b>(1), 5&ndash;34. <code style="white-space: pre;">&#8288;https://doi.org/10.1080/713754206&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.bourassa1996
head(dat, 10)

## Not run: 

### load metafor package
library(metafor)

### calculate log(OR) and corresponding sampling variance with 1/2 correction
dat &lt;- escalc(measure="OR", ai=lh.le, bi=lh.re, ci=rh.le, di=rh.re, data=dat, add=1/2, to="all")
head(dat, 10)

### overall association between handedness and eyedness
res &lt;- rma(yi, vi, data=dat, subset=sex=="combined")
res
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.cannon2006'>Studies on the Effectiveness of Intensive Versus Moderate Statin Therapy for Preventing Coronary Death or Myocardial Infarction</h2><span id='topic+dat.cannon2006'></span>

<h3>Description</h3>

<p>Results from 4 trials examining the effectiveness of intensive (high dose) versus moderate (standard dose) statin therapy for preventing coronary death or myocardial infarction.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.cannon2006
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> trial name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pop</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study population (post-ACS: post acute coronary syndrome; stable CAD: stable coronary artery disease) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nt</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the high dose group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nc</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the standard dose group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep1t</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the high dose group for end point 1: coronary death or non-fatal myocardial infarction </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep1c</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the standard dose group for end point 1: coronary death or non-fatal myocardial infarction </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep2t</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the high dose group for end point 2: coronary death or any cardiovascular event (MI, stroke, hospitalization for unstable angina, or revascularization) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep2c</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the standard dose group for end point 2: coronary death or any cardiovascular event (MI, stroke, hospitalization for unstable angina, or revascularization) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep3t</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the high dose group for end point 3: cardiovascular death </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep3c</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the standard dose group for end point 3: cardiovascular death </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep4t</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the high dose group for end point 4: non-cardiovascular death </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep4c</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the standard dose group for end point 4: non-cardiovascular death </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep5t</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the high dose group for end point 5: deaths (all-cause mortality) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep5c</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the standard dose group for end point 5: deaths (all-cause mortality) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep6t</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the high dose group for end point 6: stroke </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ep6c</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of events in the standard dose group for end point 6: stroke
</td>
</tr>

</table>



<h3>Details</h3>

<p>The data were obtained from Figures 2, 3, 4, and 5 in Cannon et al. (2006). The authors used the Mantel-Haenszel method for combining the results from the 4 trials. This approach is implemented in the <code><a href="metafor.html#topic+rma.mh">rma.mh</a></code> function.
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, odds ratios, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Cannon, C. P., Steinberg, B. A., Murphy, S. A., Mega, J. L., &amp; Braunwald, E. (2006). Meta-analysis of cardiovascular outcomes trials comparing intensive versus moderate statin therapy. <em>Journal of the American College of Cardiology</em>, <b>48</b>(3), 438&ndash;445. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.jacc.2006.04.070&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.cannon2006
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of log odds ratios using the MH method for endpoint 1
res &lt;- rma.mh(measure="OR", ai=ep1t, n1i=nt, ci=ep1c, n2i=nc, data=dat, slab=trial)
print(res, digits=2)

### forest plot
forest(res, xlim=c(-.8,.8), atransf=exp, at=log(c(2/3, 1, 3/2)),
       header=TRUE, top=2, cex=1.2, xlab="Odds Ratio")
mtext("(high dose better)", side=1, line=par("mgp")[1]-0.5, at=log(2/3), cex=1.2, font=3)
mtext("(standard dose better)", side=1, line=par("mgp")[1]-0.5, at=log(3/2), cex=1.2, font=3)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.cohen1981'>Studies on the Relationship between Course Instructor Ratings and Student Achievement</h2><span id='topic+dat.cohen1981'></span>

<h3>Description</h3>

<p>Results from 20 studies on the correlation between course instructor ratings and student achievement.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.cohen1981
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) and year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sample</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> course type </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>control</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> ability control </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size of the study (number of sections) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> observed correlation
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies included in this dataset examined to what extent students' ratings of a course instructor correlated with their achievement in the course. Instead of correlating individual ratings and achievement scores, the studies were carried out in multisection courses, in which the sections had different instructors but all sections used a common achievement measure (e.g., a final exam). The correlation coefficients reflect the correlation between the mean instructor rating and the mean achievement score of each section. Hence, the unit of analysis are the sections, not the individuals.
</p>
<p>Note that this dataset (extracted from Table A.3 in Cooper &amp; Hedges, 1994) only contains studies with at least 10 sections.
</p>


<h3>Concepts</h3>

<p>education, correlation coefficients
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Cooper, H., &amp; Hedges, L. V. (1994). Appendix A: Data Sets. In H. Cooper &amp; L. V. Hedges (Eds.), <em>The handbook of research synthesis</em> (pp. 543-547). New York: Russell Sage Foundation.
</p>


<h3>References</h3>

<p>Cohen, P. A. (1981). Student ratings of instruction and student achievement: A meta-analysis of multisection validity studies. <em>Review of Educational Research</em>, <b>51</b>(3), 281&ndash;309. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/00346543051003281&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.cohen1981
dat[c(1,4,5)]

## Not run: 

### load metafor package
library(metafor)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=ri, ni=ni, data=dat[c(1,4,5)])
dat

### meta-analysis of the transformed correlations using a random-effects model
res &lt;- rma(yi, vi, data=dat, digits=2)
res

### predicted average correlation with 95% CI
predict(res, transf=transf.ztor)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.colditz1994'>Studies on the Effectiveness of the BCG Vaccine Against Tuberculosis</h2><span id='topic+dat.colditz1994'></span>

<h3>Description</h3>

<p>Results from 13 studies examining the effectiveness of the Bacillus Calmette-Guerin (BCG) vaccine against tuberculosis. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.colditz1994
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tpos</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB positive cases in the treated (vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tneg</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB negative cases in the treated (vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cpos</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB positive cases in the control (non-vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cneg</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of TB negative cases in the control (non-vaccinated) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ablat</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> absolute latitude of the study location (in degrees) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>alloc</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> method of treatment allocation (random, alternate, or systematic assignment)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 13 studies provide data in terms of \(2 \times 2\) tables in the form:
</p>

<table>
<tr>
 <td style="text-align: left;">
                    </td><td style="text-align: center;"> TB positive </td><td style="text-align: center;"> TB negative </td>
</tr>
<tr>
 <td style="text-align: left;">
   vaccinated group </td><td style="text-align: center;"> <code>tpos</code> </td><td style="text-align: center;"> <code>tneg</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   control group    </td><td style="text-align: center;"> <code>cpos</code> </td><td style="text-align: center;"> <code>cneg</code>
   </td>
</tr>

</table>
<p> The goal of the meta-analysis was to examine the overall effectiveness of the BCG vaccine for preventing tuberculosis and to examine moderators that may potentially influence the size of the effect.
</p>
<p>The dataset has been used in several publications to illustrate meta-analytic methods (see &lsquo;References&rsquo;).
</p>


<h3>Concepts</h3>

<p>medicine, risk ratios, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Colditz, G. A., Brewer, T. F., Berkey, C. S., Wilson, M. E., Burdick, E., Fineberg, H. V., &amp; Mosteller, F. (1994). Efficacy of BCG vaccine in the prevention of tuberculosis: Meta-analysis of the published literature. <em>Journal of the American Medical Association</em>, <b>271</b>(9), 698&ndash;702. <code style="white-space: pre;">&#8288;https://doi.org/10.1001/jama.1994.03510330076038&#8288;</code>
</p>


<h3>References</h3>

<p>Berkey, C. S., Hoaglin, D. C., Mosteller, F., &amp; Colditz, G. A. (1995). A random-effects regression model for meta-analysis. <em>Statistics in Medicine</em>, <b>14</b>(4), 395&ndash;411. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.4780140406&#8288;</code>
</p>
<p>van Houwelingen, H. C., Arends, L. R., &amp; Stijnen, T. (2002). Advanced methods in meta-analysis: Multivariate approach and meta-regression. <em>Statistics in Medicine</em>, <b>21</b>(4), 589&ndash;624. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.1040&#8288;</code>
</p>
<p>Viechtbauer, W. (2010). Conducting meta-analyses in R with the metafor package. <em>Journal of Statistical Software</em>, <b>36</b>(3), 1&ndash;48. <code style="white-space: pre;">&#8288;https://doi.org/10.18637/jss.v036.i03&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.colditz1994
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log risk ratios and corresponding sampling variances
dat &lt;- escalc(measure="RR", ai=tpos, bi=tneg, ci=cpos, di=cneg, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### average risk ratio with 95% CI
predict(res, transf=exp)

### mixed-effects model with absolute latitude and publication year as moderators
res &lt;- rma(yi, vi, mods = ~ ablat + year, data=dat)
res

### predicted average risk ratios for 10-60 degrees absolute latitude
### holding the publication year constant at 1970
predict(res, newmods=cbind(seq(from=10, to=60, by=10), 1970), transf=exp)

### note: the interpretation of the results is difficult because absolute
### latitude and publication year are strongly correlated (the more recent
### studies were conducted closer to the equator)
plot(ablat ~ year, data=dat, pch=19, xlab="Publication Year", ylab="Absolute Lattitude")
cor(dat$ablat, dat$year)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.collins1985a'>Studies on the Treatment of Upper Gastrointestinal Bleeding by a Histamine H2 Antagonist</h2><span id='topic+dat.collins1985a'></span>

<h3>Description</h3>

<p>Results from studies examining the effectiveness of histamine H2 antagonists (cimetidine or ranitidine) in treating patients with acute upper gastrointestinal hemorrhage.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.collins1985a
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trial</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author of trial </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ref</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> reference number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trt</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> C = cimetidine, R = ranitidine </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ctrl</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> P = placebo, AA = antacids, UT = usual treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nti</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>b.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in treatment group with persistent or recurrent bleedings </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>o.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in treatment group in need of operation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>d.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in treatment group that died </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nci</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>b.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in control group with persistent or recurrent bleedings </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>o.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in control group in need of operation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>d.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in control group that died
</td>
</tr>

</table>



<h3>Details</h3>

<p>The data were obtained from Tables 1 and 2 in Collins and Langman (1985). The authors used Peto's (one-step) method for meta-analyzing the 27 trials. This approach is implemented in the <code><a href="metafor.html#topic+rma.peto">rma.peto</a></code> function. Using the same dataset, van Houwelingen, Zwinderman, and Stijnen (1993) describe some alternative approaches for analyzing these data, including fixed- and random-effects conditional logistic models. Those are implemented in the <code><a href="metafor.html#topic+rma.glmm">rma.glmm</a></code> function.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, Peto's method, generalized linear models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Collins, R., &amp; Langman, M. (1985). Treatment with histamine H2 antagonists in acute upper gastrointestinal hemorrhage. <em>New England Journal of Medicine</em>, <b>313</b>(11), 660&ndash;666. <code style="white-space: pre;">&#8288;https://doi.org/10.1056/NEJM198509123131104&#8288;</code>
</p>


<h3>References</h3>

<p>van Houwelingen, H. C., Zwinderman, K. H., &amp; Stijnen, T. (1993). A bivariate approach to meta-analysis. <em>Statistics in Medicine</em>, <b>12</b>(24), 2273&ndash;2284. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.4780122405&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.collins1985a
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of log ORs using Peto's method (outcome: persistent or recurrent bleedings)
res &lt;- rma.peto(ai=b.xti, n1i=nti, ci=b.xci, n2i=nci, data=dat)
print(res, digits=2)

### meta-analysis of log ORs using a conditional logistic regression model (FE model)
res &lt;- rma.glmm(measure="OR", ai=b.xti, n1i=nti, ci=b.xci, n2i=nci, data=dat,
                model="CM.EL", method="FE")
summary(res)
predict(res, transf=exp, digits=2)

### plot the likelihoods of the odds ratios
llplot(measure="OR", ai=b.xti, n1i=nti, ci=b.xci, n2i=nci, data=dat,
       lwd=1, refline=NA, xlim=c(-4,4), drop00=FALSE)

### meta-analysis of log odds ratios using a conditional logistic regression model (RE model)
res &lt;- rma.glmm(measure="OR", ai=b.xti, n1i=nti, ci=b.xci, n2i=nci, data=dat,
                model="CM.EL", method="ML")
summary(res)
predict(res, transf=exp, digits=2)

### meta-analysis of log ORs using Peto's method (outcome: need for surgery)
res &lt;- rma.peto(ai=o.xti, n1i=nti, ci=o.xci, n2i=nci, data=dat)
print(res, digits=2)

### meta-analysis of log ORs using Peto's method (outcome: death)
res &lt;- rma.peto(ai=d.xti, n1i=nti, ci=d.xci, n2i=nci, data=dat)
print(res, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.collins1985b'>Studies on the Effects of Diuretics in Pregnancy</h2><span id='topic+dat.collins1985b'></span>

<h3>Description</h3>

<p>Results from 9 studies examining the effects of diuretics in pregnancy on various outcomes.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.collins1985b
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pre.nti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women in treatment group followed up for pre-eclampsia outcome </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pre.nci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women in control/placebo group followed up for pre-eclampsia outcome </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pre.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women in treatment group with any form of pre-eclampsia </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pre.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women in control/placebo group with any form of pre-eclampsia </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>oedema</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> dummy variable indicating whether oedema was a diagnostic criterion </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>fup.nti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women in treatment group followed up for mortality outcomes </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>fup.nci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of women in control/placebo group followed up for mortality outcomes </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ped.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of perinatal deaths in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ped.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of perinatal deaths in control/placebo group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>stb.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of stillbirths in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>stb.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of stillbirths in control/placebo group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ned.xti</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of neonatal deaths in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ned.xci</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of neonatal deaths in control/placebo group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 9 studies in this dataset examined the effects of diuretics in pregnancy on various outcomes, including the presence of any form of pre-eclampsia, perinatal death, stillbirth, and neonatal death.
</p>


<h3>Concepts</h3>

<p>medicine, obstetrics, odds ratios, Peto's method
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Collins, R., Yusuf, S., &amp; Peto, R. (1985). Overview of randomised trials of diuretics in pregnancy. <em>British Medical Journal</em>, <b>290</b>(6461), 17&ndash;23. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/bmj.290.6461.17&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.collins1985b
dat

## Not run: 

### load metafor package
library(metafor)

### calculate (log) odds ratio and sampling variance
dat &lt;- escalc(measure="OR", n1i=pre.nti, n2i=pre.nci, ai=pre.xti, ci=pre.xci, data=dat)
summary(dat, digits=2, transf=exp)

### meta-analysis using Peto's method for any form of pre-eclampsia
rma.peto(n1i=pre.nti, n2i=pre.nci, ai=pre.xti, ci=pre.xci, data=dat, digits=2)

### meta-analysis including only studies where oedema was not a diagnostic criterion
rma.peto(n1i=pre.nti, n2i=pre.nci, ai=pre.xti, ci=pre.xci, data=dat, digits=2, subset=(oedema==0))

### meta-analyses of mortality outcomes (perinatal deaths, stillbirths, and neonatal deaths)
rma.peto(n1i=fup.nti, n2i=fup.nci, ai=ped.xti, ci=ped.xci, data=dat, digits=2)
rma.peto(n1i=fup.nti, n2i=fup.nci, ai=stb.xti, ci=stb.xci, data=dat, digits=2)
rma.peto(n1i=fup.nti, n2i=fup.nci, ai=ned.xti, ci=ned.xci, data=dat, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.craft2003'>Studies on the Relationship between the Competitive State Anxiety Inventory-2 and Sport Performance</h2><span id='topic+dat.craft2003'></span>

<h3>Description</h3>

<p>Results from 10 studies on the relationship between the Competitive State Anxiety Inventory-2 (CSAI-2) and sport performance.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.craft2003
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sport</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of sport (T = team sport, I = individual sport) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> correlation coefficient </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>var1</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> variable 1 of the correlation coefficient (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>var2</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> variable 2 of the correlation coefficient (see &lsquo;Details&rsquo;)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 10 studies included in this dataset are a subset of the studies included in the meta-analysis by Craft et al. (2003) on the relationship between the Competitive State Anxiety Inventory-2 (CSAI-2) and sport performance.
</p>
<p>The CSAI-2 has three subscales: cognitive anxiety (<code>acog</code>), somatic anxiety (<code>asom</code>), and self-confidence (<code>conf</code>). The studies included in this dataset administered the CSAI-2 prior to some sport competition and then measured sport performance based on the competition. Most studies provided all 6 correlations (3 for the correlations among the 3 subscales and 3 for the correlations between the subscales and sport performance), but 2 studies (with study numbers 6 and 17) only provided a subset.
</p>


<h3>Concepts</h3>

<p>psychology, correlation coefficients, multivariate models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Becker, B. J., &amp; Aloe, A. M. (2019). Model-based meta-analysis and related approaches. In H. Cooper, L. V. Hedges, &amp; J. C. Valentine (Eds.), <em>The handbook of research synthesis and meta-analysis</em> (3nd ed., pp. 339&ndash;363). New York: Russell Sage Foundation.
</p>


<h3>References</h3>

<p>Craft, L. L., Magyar, T. M., Becker, B. J., &amp; Feltz, D. L. (2003). The relationship between the Competitive State Anxiety Inventory-2 and sport performance: A meta-analysis. <em>Journal of Sport and Exercise Psychology</em>, <b>25</b>(1), 44&ndash;65. <code style="white-space: pre;">&#8288;https://doi.org/10.1123/jsep.25.1.44&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.craft2003
head(dat, 18)

## Not run: 

### load metafor package
library(metafor)

### construct dataset and var-cov matrix of the correlations
tmp &lt;- rcalc(ri ~ var1 + var2 | study, ni=ni, data=dat)
V &lt;- tmp$V
dat &lt;- tmp$dat

### examine data for study 1
dat[dat$study == 1,]
V[dat$study == 1, dat$study == 1]

### examine data for study 6
dat[dat$study == 6,]
V[dat$study == 6, dat$study == 6]

### examine data for study 17
dat[dat$study == 17,]
V[dat$study == 17, dat$study == 17]

### multivariate random-effects model
res &lt;- rma.mv(yi, V, mods = ~ var1.var2 - 1, random = ~ var1.var2 | study, struct="UN", data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.crede2010'>Studies on the Relationship between Class Attendance and Grades in College Students</h2><span id='topic+dat.crede2010'></span>

<h3>Description</h3>

<p>Results from 68 studies on the relationship between class attendence and class performance and/or grade point average in college students.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.crede2010
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>studyid</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>source</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study source (journal, dissertation, other) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sampleid</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample within study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>criterion</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> criterion variable (grade, gpa) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>class</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> class type (science, nonscience) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> observed correlation
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 68 studies included in this dataset provide information about the relationship between class attendance of college students and their performance (i.e., grade) in the class and/or their overall grade point average. Some studies included multiple samples and hence the dataset actually contains 97 correlation coefficients.
</p>
<p>The dataset was obtained via personal communication. Note that this dataset differs just slightly from the one used by Credé et al. (2010).
</p>


<h3>Concepts</h3>

<p>education, correlation coefficients, multilevel models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Personal communication.
</p>


<h3>References</h3>

<p>Credé, M., Roch, S. G., &amp; Kieszczynka, U. M. (2010). Class attendance in college: A meta-analytic review of the relationship of class attendance with grades and student characteristics. <em>Review of Educational Research</em>, <b>80</b>(2), 272&ndash;295. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/0034654310362998&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.crede2010
head(dat, 18)

## Not run: 

### load metafor package
library(metafor)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=ri, ni=ni, data=dat)

############################################################################

### meta-analysis for the relationship between attendance and grades
res &lt;- rma(yi, vi, data=dat, subset=criterion=="grade")
res

### estimated average correlation with 95% CI/PI
predict(res, transf=transf.ztor, digits=2)

### examine if relationship between attendance and grades differs for nonscience/science classes
res &lt;- rma(yi, vi, mods = ~ class, data=dat, subset=criterion=="grade")
res

### estimated average correlations for nonscience and science classes
predict(res, newmods=c(0,1), transf=transf.ztor, digits=2)

### examine if relationship between attendance and grades has changed over time
res &lt;- rma(yi, vi, mods = ~ year, data=dat, subset=criterion=="grade")
res

############################################################################

### meta-analysis for the relationship between attendance and GPA
res &lt;- rma(yi, vi, data=dat, subset=criterion=="gpa")
res

### estimated average correlation with 95% CI/PI
predict(res, transf=transf.ztor, digits=2)

### examine if relationship between attendance and GPA has changed over time
res &lt;- rma(yi, vi, mods = ~ year, data=dat, subset=criterion=="gpa")
res

############################################################################

### use a multilevel model to examine the relationship between attendance and grades
res &lt;- rma.mv(yi, vi, random = ~ 1 | studyid/sampleid, data=dat, subset=criterion=="grade")
res
predict(res, transf=transf.ztor, digits=2)

### use a multilevel model to examine the relationship between attendance and gpa
res &lt;- rma.mv(yi, vi, random = ~ 1 | studyid/sampleid, data=dat, subset=criterion=="gpa")
res
predict(res, transf=transf.ztor, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.curtis1998'>Studies on the Effects of Elevated CO2 Levels on Woody Plant Mass</h2><span id='topic+dat.curtis1998'></span>

<h3>Description</h3>

<p>Results from studies examining the effects of elevated CO2 levels on woody plant mass.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.curtis1998
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> observation number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>paper</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> paper number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>genus</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> genus name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>species</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> species name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>fungrp</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> plant functional group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>co2.ambi</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> ambient CO2 level (control group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>co2.elev</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> elevated CO2 level (treatment group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>units</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> units for CO2 exposure levels </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>time</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> maximum length of time (days) of CO2 exposure </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pot</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> growing method (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>method</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CO2 exposure facility (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>stock</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> planting stock code </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xtrt</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> interacting treatment code (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>level</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> interacting treatment level codes (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>m1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean plant mass under elevated CO2 level (treatment group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation of plant mass underelevated CO2 level (treatment group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of observations under elevated CO2 level (treatment group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>m2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean plant mass under ambient CO2 level (control group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation of plant mass under ambient CO2 level (control group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of observations under ambient CO2 level (control group)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies included in this dataset compared the total above- plus below-ground biomass (in grams) for plants that were either exposed to ambient (around 35 Pa) and elevated CO2 levels (around twice the ambient level). The <code>co2.ambi</code> and <code>co2.elev</code> variables indicate the CO2 levels in the control and treatment groups, respectively (with the <code>units</code> variable specifying the units for the CO2 exposure levels). Many of the studies also varied one or more additional environmental variables (defined by the <code>xtrt</code> and <code>level</code> variables):
</p>

<ul>
<li><p> NONE = no additional treatment factor
</p>
</li>
<li><p> FERT = soil fertility (either a <code>CONTROL</code>, <code>HIGH</code>, or <code>LOW</code> level)
</p>
</li>
<li><p> LIGHT = light treatment (always a <code>LOW</code> light level)
</p>
</li>
<li><p> FERT+L = soil fertility and light (a <code>LOW</code> light and soil fertility level)
</p>
</li>
<li><p> H2O = well watered vs drought (either a <code>WW</code> or <code>DRT</code> level)
</p>
</li>
<li><p> TEMP = temperature treatment (either a <code>HIGH</code> or <code>LOW</code> level)
</p>
</li>
<li><p> OZONE = ozone exposure (either a <code>HIGH</code> or <code>LOW</code> level)
</p>
</li>
<li><p> UVB = ultraviolet-B radiation exposure (either a <code>HIGH</code> or <code>LOW</code> level)
</p>
</li></ul>
<p> In addition, the studies differed with respect to various design variables, including CO2 exposure duration (<code>time</code>), growing method (<code>pot</code>: number = pot size in liters; <code>GRND</code> = plants rooted in ground; <code>HYDRO</code> = solution or aeroponic culture), CO2 exposure facility (<code>method</code>: <code>GC</code> = growth chamber; <code>GH</code> = greenhouse; <code>OTC</code> = field-based open-top chamber), and planting stock (<code>stock</code>: <code>SEED</code> = plants started from seeds; <code>SAP</code> = plants started from cuttings). The goal of the meta-analysis was to examine the effects of elevated CO2 levels on plant physiology and growth and the interacting effects of the environmental (and design) variables.
</p>


<h3>Concepts</h3>

<p>ecology, ratios of means
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Hedges, L. V., Gurevitch, J., &amp; Curtis, P. S. (1999). The meta-analysis of response ratios in experimental ecology. <em>Ecology</em>, <b>80</b>(4), 1150&ndash;1156. <code style="white-space: pre;">&#8288;https://doi.org/10.1890/0012-9658(1999)080[1150:TMAORR]2.0.CO;2&#8288;</code> (data obtained from <em>Ecological Archives</em>, E080-008-S1, at: <a href="https://esapubs.org/archive/ecol/E080/008/">https://esapubs.org/archive/ecol/E080/008/</a>)
</p>


<h3>References</h3>

<p>Curtis, P. S., &amp; Wang, X. (1998). A meta-analysis of elevated CO2 effects on woody plant mass, form, and physiology. <em>Oecologia</em>, <b>113</b>(3), 299&ndash;313. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s004420050381&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.curtis1998
head(dat)

## Not run: 

### load metafor package
library(metafor)

### calculate (log transformed) ratios of means and corresponding sampling variances
dat &lt;- escalc(measure="ROM", m1i=m1i, sd1i=sd1i, n1i=n1i,
                             m2i=m2i, sd2i=sd2i, n2i=n2i, data=dat)
head(dat)

### meta-analysis using a random-effects model
res &lt;- rma(yi, vi, method="DL", data=dat)
res

### average ratio of means with 95% CI
predict(res, transf=exp, digits=2)

### meta-analysis for plants grown under nutrient stress
res &lt;- rma(yi, vi, method="DL", data=dat, subset=(xtrt=="FERT" &amp; level=="LOW"))
predict(res, transf=exp, digits=2)

### meta-analysis for plants grown under low light conditions
res &lt;- rma(yi, vi, method="DL", data=dat, subset=(xtrt=="LIGHT" &amp; level=="LOW"))
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.dagostino1998'>Studies on the Effectiveness of Antihistamines in Reducing Symptoms of the Common Cold</h2><span id='topic+dat.dagostino1998'></span>

<h3>Description</h3>

<p>Results from 9 studies on the effectiveness of antihistamines in reducing the severity of runny nose and sneezing in the common cold.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.dagostino1998
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cold</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> natural or induced cold study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>scale.rn</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> scale for measuring runny nose severity </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>scale.sn</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> scale for measuring sneezing severity </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>drug</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of antihistamine studied </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tnt</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total sample size of the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tnc</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total sample size of the control (placebo) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>outcome</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> outcome variable (see &lsquo;Details&rsquo;)</td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mt</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sdt</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> SD in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mc</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sdc</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> SD in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xt</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients reaching the therapy goal in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xc</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients reaching the therapy goal in the control (placebo) group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nt</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size of the treatment group for measuring the outcome </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nc</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size of the control group for measuring the outcome
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies for this meta-analysis were assembled to examine the effectiveness of antihistamines in reducing the severity of runny nose and sneezing in the common cold. Effectiveness was measured after one and two days of treatment in terms of 4 different outcome variables:
</p>

<ol>
<li> <p><code>rnic1</code> and <code>rnic2</code> (continuous): incremental change (improvement) in runny nose severity at day 1 and day 2,
</p>
</li>
<li> <p><code>rngoal1</code> and <code>rngoal2</code> (dichotomous): reaching the goal of therapy (of at least a 50% reduction in runny nose severity) at day 1 and day 2,
</p>
</li>
<li> <p><code>snic1</code> and <code>snic2</code> (continuous): incremental change (improvement) in sneezing severity at day 1 and day 2, and
</p>
</li>
<li> <p><code>rngoal1</code> and <code>rngoal2</code> (dichotomous): reaching the goal of therapy (of at least a 50% reduction in sneezing severity) at day 1 and day 2.
</p>
</li></ol>

<p>For the continuous outcomes, standardized mean differences can be computed to quantify the difference between the treatment and control groups. For the dichotomous outcomes, one can compute (log) odds ratios to quantify the difference between the treatment and control groups.
</p>


<h3>Concepts</h3>

<p>medicine, standardized mean differences, odds ratios, multivariate models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>D'Agostino, R. B., Sr., Weintraub, M., Russell, H. K., Stepanians, M., D'Agostino, R. B., Jr., Cantilena, L. R., Jr., Graumlich, J. F., Maldonado, S., Honig, P., &amp; Anello, C. (1998). The effectiveness of antihistamines in reducing the severity of runny nose and sneezing: A meta-analysis. <em>Clinical Pharmacology &amp; Therapeutics</em>, <b>64</b>(6), 579&ndash;596. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/S0009-9236(98)90049-2&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.dagostino1998
head(dat, 16)

## Not run: 

### load metafor package
library(metafor)

### compute standardized mean differences and corresponding sampling variances
dat &lt;- escalc(measure="SMD", m1i=mt, m2i=mc, sd1i=sdt, sd2i=sdc, n1i=nt, n2i=nc, data=dat,
              add.measure=TRUE)

### compute log odds ratios and corresponding sampling variances
dat &lt;- escalc(measure="OR",  ai=xt, ci=xc, n1i=nt, n2i=nc, data=dat,
              replace=FALSE, add.measure=TRUE, add=1/2, to="all")

### inspect data for the first study
head(dat, 8)

### fit a random-effects model for incremental change in runny nose severity at day 1
res &lt;- rma(yi, vi, data=dat, subset=outcome=="rnic1")
res

### fit a random-effects model for reaching the goal of therapy for runny nose severity at day 1
res &lt;- rma(yi, vi, data=dat, subset=outcome=="rngoal1")
res
predict(res, transf=exp)

### construct approximate V matrix assuming a correlation of 0.7 for sampling errors within studies
dat$esid &lt;- ave(dat$study, dat$study, FUN=seq)
V &lt;- vcalc(vi, cluster=study, obs=esid, rho=0.7, data=dat)

### fit a model for incremental change in runny nose severity at day 1 and at day 2, allowing for
### correlated sampling errors (no random effects added, since there does not appear to be any
### noteworthy heterogeneity in these data)
res &lt;- rma.mv(yi, V, mods = ~ outcome - 1, data=dat, subset=outcome %in% c("rnic1","rnic2"))
res

### test if there is a difference in effects at day 1 and day 2
anova(res, X=c(1,-1))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.damico2009'>Studies on Topical plus Systemic Antibiotics to Prevent Respiratory Tract Infections</h2><span id='topic+dat.damico2009'></span>

<h3>Description</h3>

<p>Results from 16 studies examining the effectiveness of topical plus systemic antibiotics to prevent respiratory tract infections (RTIs).</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.damico2009
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xt</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of RTIs in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nt</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xc</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of RTIs in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nc</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>conceal</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> allocation concealment (0 = not adequate, 1 = adequate) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>blind</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> blinding (0 = open, 1 = double-blind)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 16 studies that examined the effectiveness of topical plus systemic antibiotics versus no prophylaxis to prevent respiratory tract infections (RTIs).
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>D'Amico, R., Pifferi, S., Torri, V., Brazzi, L., Parmelli, E., &amp; Liberati, A. (2009). Antibiotic prophylaxis to reduce respiratory tract infections and mortality in adults receiving intensive care. <em>Cochrane Database of Systematic Reviews</em>, <b>4</b>, CD000022. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD000022.pub3&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.damico2009
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of the (log) odds ratios using the Mantel-Haenszel method
rma.mh(measure="OR", ai=xt, n1i=nt, ci=xc, n2i=nc, data=dat, digits=2)

### calculate log odds ratios and corresponding sampling variances
dat &lt;- escalc(measure="OR", ai=xt, n1i=nt, ci=xc, n2i=nc, data=dat)

### meta-analysis using a random-effects model
res &lt;- rma(yi, vi, data=dat, method="DL")
res
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.debruin2009'>Studies on Standard Care Quality and HAART-Adherence</h2><span id='topic+dat.debruin2009'></span>

<h3>Description</h3>

<p>Results from 13 trials providing information about standard care quality and HAART-adherence in control groups.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.debruin2009
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>author</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author of study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>scq</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard care quality </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the standard care group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xi</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients with an undetectable viral load in standard care group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mi</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients with a detectable viral load in standard care group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ethnicity</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> dominant ethnicity of the patients in the standard care group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>patients</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> inclusion of patients continuing or starting (a new) treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>select</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> baseline selection of patients with adherence problems or no selection </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sens</b>        </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> sensitivity of viral load assessments (&lt;400 vs. &gt;=400 copies/ml)
</td>
</tr>

</table>



<h3>Details</h3>

<p>Highly active antiretroviral therapy (HAART) refers to a combination of multiple antiretroviral drugs that can effectively suppress the HIV virus. However, achieving viral suppression (to the point that the virus becomes essentially undetectable in a blood sample) requires high levels of adherence to an often complicated medication regimen. A number of trials have examined various interventions that aim to increase adherence levels. In each trial, patients receiving the intervention are compared to patients in a control group receiving standard care (often referred to as &lsquo;care as usual&rsquo;). However, the quality of standard care can vary substantially between these studies. de Bruin et al. (2009) assessed the quality of standard care provided (based on a quantification of the number of behavior change techniques applied) and examined to what extent the quality of standard care was related to the proportion of patients achieving effective viral suppression in the control groups.
</p>


<h3>Concepts</h3>

<p>psychology, medicine, proportions, single-arm studies, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>de Bruin, M., Viechtbauer, W., Hospers, H. J., Schaalma, H. P., &amp; Kok, G. (2009). Standard care quality determines treatment outcomes in control groups of HAART-adherence intervention studies: Implications for the interpretation and comparison of intervention effects. <em>Health Psychology</em>, <b>28</b>(6), 668&ndash;674. <code style="white-space: pre;">&#8288;https://doi.org/10.1037/a0015989&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.debruin2009
dat

## Not run: 

### load metafor package
library(metafor)

### calculate proportions and corresponding sampling variances
dat &lt;- escalc(measure="PR", xi=xi, ni=ni, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat)
print(res, digits=2)

### mixed-effects meta-regression model with all predictors/covariates
res &lt;- rma(yi, vi, mods = ~ scq + ethnicity + patients + select + sens, data=dat)
print(res, digits=3)

### mixed-effects meta-regression model with scq and ethnicity as predictors/covariates
res &lt;- rma(yi, vi, mods = ~ scq + ethnicity, data=dat)
print(res, digits=3)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.dogliotti2014'>Studies on Antithrombotic Treatments to Prevent Strokes</h2><span id='topic+dat.dogliotti2014'></span>

<h3>Description</h3>

<p>Results from 20 trials examining the effectiveness of antithrombotic treatments to prevent strokes in patients with non-valvular atrial fibrillation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.dogliotti2014
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>id</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>stroke</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of strokes </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>total</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of individuals
</td>
</tr>

</table>



<h3>Details</h3>

<p>This data set comes from a systematic review aiming to estimate the effects of eight antithrombotic treatments including placebo in reducing the incidence of major thrombotic events in patients with non-valvular atrial fibrillation (Dogliotti et al., 2014).
</p>
<p>The review included 20 studies with 79,808 participants, four studies are three-arm studies. The primary outcome is stroke reduction (yes / no).
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, network meta-analysis, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Dogliotti, A., Paolasso, E., &amp; Giugliano, R. P. (2014). Current and new oral antithrombotics in non-valvular atrial fibrillation: A network meta-analysis of 79808 patients. <em>Heart</em>, <b>100</b>(5), 396&ndash;405. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/heartjnl-2013-304347&#8288;</code>
</p>


<h3>See Also</h3>

  
<p><code><a href="netmeta.html#topic+pairwise">pairwise</a></code>, <code><a href="meta.html#topic+metabin">metabin</a></code>, <code><a href="netmeta.html#topic+netmeta">netmeta</a></code>, <code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show first 7 rows / 3 studies of the dataset
head(dat.dogliotti2014, 7)

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Change appearance of confidence intervals
cilayout("(", "-")

### Transform data from long arm-based format to contrast-based
### format. Argument 'sm' has to be used for odds ratio as summary
### measure; by default the risk ratio is used in the metabin function
### called internally.
pw &lt;- pairwise(treat = treatment, n = total, event = stroke,
  studlab = study, data = dat.dogliotti2014, sm = "OR")

### Print log odds ratios (TE) and standard errors (seTE)
head(pw, 5)[, 1:5]

### Conduct network meta-analysis (NMA) with placebo as reference
net &lt;- netmeta(pw, ref = "plac")

### Details on excluded study
selvars &lt;- c("studlab", "event1", "n1", "event2", "n2")
subset(pw, studlab == "WASPO, 2007")[, selvars]

### Show network graph
netgraph(net, seq = "optimal", number = TRUE)

### Conduct Mantel-Haenszel NMA
net.mh &lt;- netmetabin(pw, ref = "plac")

### Compare results of inverse variance and Mantel-Haenszel NMA
nb &lt;- netbind(net, net.mh, random = FALSE,
  name = c("Inverse variance", "Mantel-Haenszel"))
forest(nb, xlim = c(0.15, 2), at = c(0.2, 0.5, 1, 2))

### Print and plot results for inverse variance NMA
net
forest(net)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.dong2013'>Studies on Safety of Inhaled Medications for Chronic Obstructive Pulmonary Disease</h2><span id='topic+dat.dong2013'></span>

<h3>Description</h3>

<p>Results from 41 trials examining the safety of inhaled medications in patients with chronic obstructive pulmonary disease.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.dong2013
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> study ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>death</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> mortality </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>randomized</b> </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of individuals
</td>
</tr>

</table>



<h3>Details</h3>

<p>This network meta-analysis compared the safety of inhaled medications in patients with chronic obstructive pulmonary disease (Dong et al., 2013).
</p>
<p>Mortality was reported in 41 randomized trials, with a total of 52 462 patients. Mortality was low, with 2 408 deaths (4.6%) reported across all studies. There were nine studies that reported zero events in at least one of the treatment arms and three additional studies had zero events in all treatment arms.
</p>
<p>This data set was used in Efthimiou et al. (2019) to illustrate the Mantel-Haenszel method for network meta-analysis.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, network meta-analysis, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Dong, Y.-H., Lin, H.-H., Shau, W.-Y., Wu, Y.-C., Chang, C.-H., &amp; Lai, M.-S. (2013). Comparative safety of inhaled medications in patients with chronic obstructive pulmonary disease: Systematic review and mixed treatment comparison meta-analysis of randomised controlled trials. <em>Thorax</em>, <b>68</b>(1), 48&ndash;56. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/thoraxjnl-2012-201926&#8288;</code>
</p>


<h3>References</h3>

<p>Efthimiou, O., Rücker, G., Schwarzer, G., Higgins, J., Egger, M., &amp; Salanti, G. (2019). A Mantel-Haenszel model for network meta-analysis of rare events. <em>Statistics in Medicine</em>, <b>38</b>(16), 2992&ndash;3012. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.8158&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="netmeta.html#topic+pairwise">pairwise</a></code>, <code><a href="meta.html#topic+metabin">metabin</a></code>, <code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show first 6 rows / 3 studies of the dataset
head(dat.dong2013)

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Change appearance of confidence intervals
cilayout("(", "-")

### Transform data from long arm-based format to contrast-based
### format. Argument 'sm' has to be used for odds ratio as summary
### measure; by default the risk ratio is used in the metabin function
### called internally.
pw &lt;- pairwise(treatment, death, randomized, studlab = id,
  data = dat.dong2013, sm = "OR")

### Calculated log odds ratios (TE) and standard errors (seTE)
pw[1:3, 1:9]

### Conduct Mantel-Haenszel network meta-analysis (NMA)
net &lt;- netmetabin(pw, ref = "plac")

### Network graph
netgraph(net, seq = "optimal", col = "black", plastic = FALSE,
  points = TRUE, pch = 21, cex.points = 3, col.points = "black",
  bg.points = "gray", thickness = "se.fixed",
  number.of.studies = TRUE)

### Show results for Mantel-Haenszel NMA
net
forest(net)

### League table with network estimates in lower triangle and direct
### estimates in upper triangle
netleague(net)

### Assess inconsistency
print(netsplit(net), show = "both", ci = TRUE, overall = FALSE,
  nchar.trts = 6)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.dorn2007'>Studies on Complementary and Alternative Medicine for Irritable Bowel Syndrome</h2><span id='topic+dat.dorn2007'></span>

<h3>Description</h3>

<p>Results from 19 trials examining complementary and alternative medicine (CAM) for irritable bowel syndrome (IBS).</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.dorn2007
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> country where trial was conducted </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ibs.crit</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> IBS diagnostic criteria (Manning, Rome I, Rome II, or Other) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>days</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of treatment days </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>visits</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of practitioner visits </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>jada</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Jadad score </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x.a</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of responders in the active treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.a</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the active treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x.p</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of responders in the placebo group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.p</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the placebo group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 19 randomized clinical trials that examined the effectiveness of complementary and alternative medicine (CAM) for irritable bowel syndrome (IBS).
</p>


<h3>Concepts</h3>

<p>medicine, alternative medicine, risk ratios
</p>


<h3>Note</h3>

<p>The data were extracted from Table I in Dorn et al. (2009). Comparing the funnel plot in Figure 1 with the one obtained below indicates that the data for study 5 (Davis et al., 2006) in the table were not the ones that were used in the actual analyses.
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Dorn, S. D., Kaptchuk, T. J., Park, J. B., Nguyen, L. T., Canenguez, K., Nam, B. H., Woods, K. B., Conboy, L. A., Stason, W. B., &amp; Lembo, A. J. (2007). A meta-analysis of the placebo response in complementary and alternative medicine trials of irritable bowel syndrome. <em>Neurogastroenterology &amp; Motility</em>, <b>19</b>(8), 630&ndash;637. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/j.1365-2982.2007.00937.x&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.dorn2007
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log risk ratios and corresponding sampling variances
dat &lt;- escalc(measure="RR", ai=x.a, n1i=n.a, ci=x.p, n2i=n.p, data=dat)

### random-effects model
res &lt;- rma(yi, vi, data=dat, digits=2, method="DL")
res

### estimated average risk ratio
predict(res, transf=exp)

### funnel plot with study 5 highlighted in red
funnel(res, atransf=exp, at=log(c(.1, .2, .5, 1, 2, 5, 10)),
       ylim=c(0,1), steps=6, las=1, col=ifelse(id == 5, "red", "black"))

### change log risk ratio for study 5
dat$yi[5] &lt;- -0.44

### results are now more in line with what is reported in the paper
### (although the CI in the paper is not wide enough)
res &lt;- rma(yi, vi, data=dat, digits=2, method="DL")
predict(res, transf=exp)

### funnel plot with study 5 highlighted in red
funnel(res, atransf=exp, at=log(c(.1, .2, .5, 1, 2, 5, 10)),
       ylim=c(0,1), steps=6, las=1, col=ifelse(id == 5, "red", "black"))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.egger2001'>Studies on the Effectiveness of Intravenous Magnesium in Acute Myocardial Infarction</h2><span id='topic+dat.egger2001'></span>

<h3>Description</h3>

<p>Results from 16 trials examining the effectiveness of intravenous magnesium in the prevention of death following acute myocardial infarction.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.egger2001
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author or trial name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the magnesium group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the magnesium group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 16 randomized clinical trials that examined the effectiveness of intravenous magnesium in the prevention of death following acute myocardial infarction. Studies 1-7 were included in the meta-analyses by Teo et al. (1991) and Horner (1992) and were combined with the results from the LIMIT-2 trial (Woods et al., 1992) in Yusuf et al. (1993), suggesting that magnesium is an effective treatment for reducing mortality. However, the results from the ISIS-4 mega trial (ISIS-4 Collaborative Group, 1995) indicated no reduction in mortality with magnesium treatment. Publication bias has been suggested as one possible explanation for the conflicting findings (Egger &amp; Davey Smith, 1995).
</p>
<p>The present dataset includes some additional trials and are based on Table 18.2 from Egger, Davey Smith, and Altman (2001).
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, Peto's method, publication bias
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Egger, M., Davey Smith, G., &amp; Altman, D. G. (Eds.) (2001). <em>Systematic reviews in health care: Meta-analysis in context</em> (2nd ed.). London: BMJ Books.
</p>


<h3>References</h3>

<p>Egger, M., &amp; Davey Smith, G. (1995). Misleading meta-analysis: Lessons from &ldquo;an effective, safe, simple&rdquo; intervention that wasn't. <em>British Medical Journal</em>, <b>310</b>(6982), 752&ndash;754. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/bmj.310.6982.752&#8288;</code>
</p>
<p>Horner, S. M. (1992). Efficacy of intravenous magnesium in acute myocardial infarction in reducing arrhythmias and mortality: Meta-analysis of magnesium in acute myocardial infarction. <em>Circulation</em>, <b>86</b>(3), 774&ndash;779. <code style="white-space: pre;">&#8288;https://doi.org/10.1161/01.cir.86.3.774&#8288;</code>
</p>
<p>ISIS-4 Collaborative Group (1995). ISIS-4: A randomised factorial trial assessing early oral captopril, oral mononitrate, and intravenous magnesium sulphate in 58,050 patients with suspected acute myocardial infarction. <em>Lancet</em>, <b>345</b>(8951), 669&ndash;685. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/S0140-6736(95)90865-X&#8288;</code>
</p>
<p>Teo, K. K., Yusuf, S., Collins, R., Held, P. H., &amp; Peto, R. (1991). Effects of intravenous magnesium in suspected acute myocardial infarction: Overview of randomised trials. <em>British Medical Journal</em>, <b>303</b>(6816), 1499&ndash;1503. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/bmj.303.6816.1499&#8288;</code>
</p>
<p>Woods, K. L., Fletcher, S., Roffe, C., &amp; Haider, Y. (1992). Intravenous magnesium sulphate in suspected acute myocardial infarction: Results of the second Leicester Intravenous Magnesium Intervention Trial (LIMIT-2). <em>Lancet</em>, <b>339</b>(8809), 1553&ndash;1558. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/0140-6736(92)91828-v&#8288;</code>
</p>
<p>Yusuf, S., Teo, K., &amp; Woods, K. (1993). Intravenous magnesium in acute myocardial infarction: An effective, safe, simple, and inexpensive treatment. <em>Circulation</em>, <b>87</b>(6), 2043&ndash;2046. <code style="white-space: pre;">&#8288;https://doi.org/10.1161/01.cir.87.6.2043&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dat.li2007">dat.li2007</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.egger2001
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of trials 1-7 using Peto's method (as in Teo et al., 1991)
res &lt;- rma.peto(ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, subset=1:7)
print(res, digits=2)

### meta-analysis of trials 1-7 and LIMIT-2 (as in Yusuf et al., 1993)
res &lt;- rma.peto(ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, subset=c(1:7,14))
print(res, digits=2)

### meta-analysis of all trials except ISIS-4
res &lt;- rma.peto(ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, subset=-16)
print(res, digits=2)
predict(res, transf=exp, digits=2)

### meta-analysis of all trials including ISIS-4
res &lt;- rma.peto(ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat)
print(res, digits=2)
predict(res, transf=exp, digits=2)

### contour-enhanced funnel plot centered at 0
funnel(res, refline=0, level=c(90, 95, 99), shade=c("white", "gray", "darkgray"))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.fine1993'>Studies on Radiation Therapy with or without Adjuvant Chemotherapy in Patients with Malignant Gliomas</h2><span id='topic+dat.fine1993'></span>

<h3>Description</h3>

<p>Results from 17 trials comparing post-operative radiation therapy with and without adjuvant chemotherapy in patients with malignant gliomas.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.fine1993
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nei</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sample size in the experimental group receiving radiotherapy plus adjuvant chemotherapy </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nci</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sample size in the control group receiving radiotherapy alone </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>e1i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 6 months in the experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>c1i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 6 months in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>e2i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 12 months in the experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>c2i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 12 months in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>e3i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 18 months in the experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>c3i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 18 months in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>e4i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 24 months in the experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>c4i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of survivors at 24 months in the control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 17 trials report the post-operative survival of patients with malignant gliomas receiving either radiation therapy with adjuvant chemotherapy or radiation therapy alone. Survival was assessed at 6, 12, 18, and 24 months in all but one study (which assessed survival only at 12 and at 24 months).
</p>
<p>The data were reconstructed by Trikalinos and Olkin (2012) based on Table 2 in Fine et al. (1993) and Table 3 in Dear (1994). The data can be used to illustrate how a meta-analysis can be conducted of effect sizes reported at multiple time points using a multivariate model.
</p>


<h3>Concepts</h3>

<p>medicine, oncology, odds ratios, longitudinal models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Dear, K. B. G. (1994). Iterative generalized least squares for meta-analysis of survival data at multiple times. <em>Biometrics</em>, <b>50</b>(4), 989&ndash;1002. <code style="white-space: pre;">&#8288;https://doi.org/10.2307/2533438&#8288;</code>
</p>
<p>Trikalinos, T. A., &amp; Olkin, I. (2012). Meta-analysis of effect sizes reported at multiple time points: A multivariate approach. <em>Clinical Trials</em>, <b>9</b>(5), 610&ndash;620. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/1740774512453218&#8288;</code>
</p>


<h3>References</h3>

<p>Fine, H. A., Dear, K. B., Loeffler, J. S., Black, P. M., &amp; Canellos, G. P. (1993). Meta-analysis of radiation therapy with and without adjuvant chemotherapy for malignant gliomas in adults. <em>Cancer</em>, <b>71</b>(8), 2585&ndash;2597. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/1097-0142(19930415)71:8&lt;2585::aid-cncr2820710825&gt;3.0.co;2-s&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.fine1993
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log(ORs) and sampling variances for each time point
dat &lt;- escalc(measure="OR", ai=e1i, n1i=nei, ci=c1i, n2i=nci, data=dat, var.names=c("y1i","v1i"))
dat &lt;- escalc(measure="OR", ai=e2i, n1i=nei, ci=c2i, n2i=nci, data=dat, var.names=c("y2i","v2i"))
dat &lt;- escalc(measure="OR", ai=e3i, n1i=nei, ci=c3i, n2i=nci, data=dat, var.names=c("y3i","v3i"))
dat &lt;- escalc(measure="OR", ai=e4i, n1i=nei, ci=c4i, n2i=nci, data=dat, var.names=c("y4i","v4i"))

### calculate the covariances (equations in Appendix of Trikalinos &amp; Olkin, 2012)
dat$v12i &lt;- with(dat, nei / (e1i * (nei - e2i)) + nci / (c1i * (nci - c2i)))
dat$v13i &lt;- with(dat, nei / (e1i * (nei - e3i)) + nci / (c1i * (nci - c3i)))
dat$v14i &lt;- with(dat, nei / (e1i * (nei - e4i)) + nci / (c1i * (nci - c4i)))
dat$v23i &lt;- with(dat, nei / (e2i * (nei - e3i)) + nci / (c2i * (nci - c3i)))
dat$v24i &lt;- with(dat, nei / (e2i * (nei - e4i)) + nci / (c2i * (nci - c4i)))
dat$v34i &lt;- with(dat, nei / (e3i * (nei - e4i)) + nci / (c3i * (nci - c4i)))

### create dataset in long format
dat.long &lt;- data.frame(study=rep(1:nrow(dat), each=4), time=1:4,
                       yi=c(t(dat[c("y1i","y2i","y3i","y4i")])),
                       vi=c(t(dat[c("v1i","v2i","v3i","v4i")])))

### var-cov matrices of the sudies
V &lt;- lapply(split(dat, dat$study),
            function(x) matrix(c( x$v1i, x$v12i, x$v13i, x$v14i,
                                 x$v12i,  x$v2i, x$v23i, x$v24i,
                                 x$v13i, x$v23i,  x$v3i, x$v34i,
                                 x$v14i, x$v24i, x$v34i,  x$v4i), nrow=4, ncol=4, byrow=TRUE))

### remove rows for the missing time points in study 17
dat.long &lt;- na.omit(dat.long)

### remove corresponding rows/columns from var-cov matrix
V[[17]] &lt;- V[[17]][c(2,4),c(2,4)]

### make a copy of V
Vc &lt;- V

### replace any (near) singular var-cov matrices with ridge corrected versions
repl.Vi &lt;- function(Vi) {
   res &lt;- eigen(Vi)
   if (any(res$values &lt;= .08)) {
      round(res$vectors %*% diag(res$values + .08) %*% t(res$vectors), 12)
   } else {
      Vi
   }
}
Vc &lt;- lapply(Vc, repl.Vi)

### do not correct var-cov matrix of study 17
Vc[[17]] &lt;- V[[17]]

### construct block diagonal matrix
Vc &lt;- bldiag(Vc)

### multivariate fixed-effects model
res &lt;- rma.mv(yi, Vc, mods = ~ factor(time) - 1, method="FE", data=dat.long)
print(res, digits=3)

### multivariate random-effects model with heteroscedastic AR(1) structure for the true effects
res &lt;- rma.mv(yi, Vc, mods = ~ factor(time) - 1, random = ~ time | study,
              struct="HAR", data=dat.long, control=list(optimizer="hjk"))
print(res, digits=3)

### profile the variance components
par(mfrow=c(2,2))
profile(res, tau2=1, xlim=c( 0,.2))
profile(res, tau2=2, xlim=c( 0,.2))
profile(res, tau2=3, xlim=c( 0,.2))
profile(res, tau2=4, xlim=c(.1,.3))

### profile the autocorrelation coefficient
par(mfrow=c(1,1))
profile(res, rho=1)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.franchini2012'>Studies on Dopamine Agonists to Reduce &ldquo;Off-Time&rdquo; in Patients with Advanced Parkinson Disease</h2><span id='topic+dat.franchini2012'></span>

<h3>Description</h3>

<p>Results from 7 trials examining the effectiveness of four dopamine agonists and placebo to reduce &ldquo;off-time&rdquo; in patients with advanced Parkinson disease.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.franchini2012
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Study</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Treatment1</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y1</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment effect arm 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd1</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size arm 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Treatment2</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y2</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment effect arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd2</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size arm 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Treatment3</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y3</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment effect arm 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd3</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n3</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size arm 1
</td>
</tr>

</table>



<h3>Details</h3>

<p>This network meta-analysis compared the effectiveness of four active treatments and placebo in patients with advanced Parkinson disease (Franchini et al., 2012). The outcome is mean lost work-time reduction in patients given dopamine agonists as adjunct therapy. The data are given as sample size, mean, and standard deviation in each trial arm.
</p>
<p>This data set was used as an example in the supplemental material of Dias et al. (2013) where placebo is coded as 1 and the four active drugs as 2 to 5.
</p>


<h3>Concepts</h3>

<p>medicine, raw mean differences, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Dias, S., Sutton, A. J., Ades, A. E., &amp; Welton, N. J. (2013). Evidence synthesis for decision making 2: A generalized linear modeling framework for pairwise and network meta-analysis of randomized controlled trials. <em>Medical Decision Making</em>, <b>33</b>(5), 607&ndash;617. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/0272989X12458724&#8288;</code>
</p>
<p>Franchini, A. J., Dias, S., Ades, A. E., Jansen, J. P., &amp; Welton, N. J. (2012). Accounting for correlation in network meta-analysis with multi-arm trials. <em>Research Synthesis Methods</em>, <b>3</b>(2), 142&ndash;160. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/jrsm.1049&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show results from first three studies; third study is a three-arm
### study
head(dat.franchini2012, 3)

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print mean differences with two digits
settings.meta(digits = 2)

### Transform data from wide arm-based format to contrast-based
### format. Argument 'sm' must not be provided as the mean difference
### is the default in R function metacont() called internally.
pw &lt;- pairwise(list(Treatment1, Treatment2, Treatment3),
  n = list(n1, n2, n3),
  mean = list(y1, y2, y3),
  sd = list(sd1, sd2, sd3),
  data = dat.franchini2012, studlab = Study, sm = "MD")

### Show calculated mean differences (TE) for first three studies
pw[1:5, c(3:7, 10, 1)]

### Conduct network meta-analysis
net &lt;- netmeta(pw)
net

### Draw network graph
netgraph(net, points = TRUE, cex.points = 3, cex = 1.5,
  plastic = TRUE, thickness = "se.fixed",
  iterate = TRUE, start = "eigen")


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.frank2008'>Studies on the Association Between the CASP8 -652 6N del Promoter Polymorphism and Breast Cancer Risk</h2><span id='topic+dat.frank2008'></span>

<h3>Description</h3>

<p>Results from 4 case-control studies examining the association between the CASP8 -652 6N del promoter polymorphism and breast cancer risk.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.frank2008
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bc.ins.ins</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of cases who have a homozygous insertion polymorphism </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bc.ins.del</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of cases who have a heterozygous insertion/deletion polymorphism </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bc.del.del</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of cases who have a homozygous deletion polymorphism </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ct.ins.ins</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of controls who have a homozygous insertion polymorphism </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ct.ins.del</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of controls who are heterozygous insertion/deletion polymorphism </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ct.del.del</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of controls who have a homozygous deletion polymorphism
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 4 studies included in this dataset are case-control studies that have examined the association between the CASP8 -652 6N del promoter polymorphism and breast cancer risk. Breast cancer cases and controls were genotyped and either had a homozygous insertion, a heterozygous insertion/deletion, or a homozygous deletion polymorphism.
</p>
<p>Ziegler et al. (2011) used the same dataset to illustrate the use of meta-analytic methods to examine deviations from Hardy-Weinberg equilibrium across multiple studies. The relative excess heterozygosity (REH) is the proposed measure for such a meta-analysis, which can be computed by setting <code>measure="REH"</code>.
</p>


<h3>Concepts</h3>

<p>medicine, oncology, genetics, odds ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Frank, B., Rigas, S. H., Bermejo, J. L., Wiestler, M., Wagner, K., Hemminki, K., Reed, M. W., Sutter, C., Wappenschmidt, B., Balasubramanian, S. P., Meindl, A., Kiechle, M., Bugert, P., Schmutzler, R. K., Bartram, C. R., Justenhoven, C., Ko, Y.-D., Brüning, T., Brauch, H., Hamann, U., Pharoah, P. P. D., Dunning, A. M., Pooley, K. A., Easton, D. F., Cox, A. &amp; Burwinkel, B. (2008). The CASP8 -652 6N del promoter polymorphism and breast cancer risk: A multicenter study. <em>Breast Cancer Research and Treatment</em>, <b>111</b>(1), 139-144. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s10549-007-9752-z&#8288;</code>
</p>


<h3>References</h3>

<p>Ziegler, A., Steen, K. V. &amp; Wellek, S. (2011). Investigating Hardy-Weinberg equilibrium in case-control or cohort studies or meta-analysis. <em>Breast Cancer Research and Treatment</em>, <b>128</b>(1), 197&ndash;201. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s10549-010-1295-z&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.frank2008
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log odds ratios comparing ins/del versus ins/ins
dat &lt;- escalc(measure="OR", ai=bc.ins.del, bi=bc.ins.ins,
                            ci=ct.ins.del, di=ct.ins.ins, data=dat)

### fit random-effects model and get the pooled odds ratio (with 95% CI)
res &lt;- rma(yi, vi, data=dat)
res
predict(res, transf=exp, digits=2)

### calculate log odds ratios comparing del/del versus ins/ins
dat &lt;- escalc(measure="OR", ai=bc.del.del, bi=bc.ins.ins,
                            ci=ct.del.del, di=ct.ins.ins, data=dat)

### fit random-effects model and get the pooled odds ratio (with 95% CI)
res &lt;- rma(yi, vi, data=dat)
res
predict(res, transf=exp, digits=2)

### calculate log odds ratios comparing ins/del+del/del versus ins/ins
dat &lt;- escalc(measure="OR", ai=bc.ins.del+bc.del.del, bi=bc.ins.ins,
                            ci=ct.ins.del+ct.del.del, di=ct.ins.ins, data=dat)

### fit random-effects model and get the pooled odds ratio (with 95% CI)
res &lt;- rma(yi, vi, data=dat)
res
predict(res, transf=exp, digits=2)

############################################################################

### compute the relative excess heterozygosity in the controls
dat &lt;- escalc(measure="REH", ai=ct.ins.ins, bi=ct.ins.del, ci=ct.del.del,
              slab=study, data=dat)

### fit random-effects model and get the pooled REH value (with 90% CI)
res &lt;- rma(yi, vi, data=dat, level=90)
res
predict(res, transf=exp, digits=2)

### draw forest plot
forest(res, atransf=exp, header=TRUE, xlim=c(-1.5,1.5), at=log(c(0.5,5/7,1,7/5,2)))
segments(log(5/7), -2, log(5/7), res$k+1, lty="dotted")
segments(log(7/5), -2, log(7/5), res$k+1, lty="dotted")


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.gibson2002'>Studies on the Effectiveness of Self-Management Education and Regular Medical Review for Adults with Asthma</h2><span id='topic+dat.gibson2002'></span>

<h3>Description</h3>

<p>Results from 15 trials examining the effectiveness of self-management education and regular medical review for adults with asthma.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.gibson2002
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>author</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author of study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the intervention group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>m1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean number of days off work/school in the intervention group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation of the number of days off work/school in the intervention group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the control/comparison group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>m2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean number of days off work/school in the control/comparison group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation of the number of days off work/school in the control/comparison group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants who had one or more days off work/school in the intervention group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants who no days off work/school in the intervention group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants who had one or more days off work/school in the control/comparison group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>di</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants who no days off work/school in the control/comparison group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>type</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> numeric code for the intervention type (see &lsquo;Details&rsquo;)
</td>
</tr>

</table>



<h3>Details</h3>

<p>Asthma management guidelines typically recommend for patients to receive education and regular medical review. While self-management programs have been shown to increase patient knowledge, it is less clear to what extent they actually impact health outcomes. The systematic review by Gibson et al. (2002) examined the effectiveness of self-management education and regular medical review for adults with asthma. In each study, participants receiving a certain management intervention were compared against those in a control/comparison group with respect to a variety of health outcomes. One of the outcomes examined in a number of studies was the number of days off work/school.
</p>
<p>The majority of studies reporting this outcome provided means and standard deviations allowing a meta-analysis of standardized mean differences. Seven studies also reported the number of participants who had one or more days off work/school in each group. These studies could be meta-analyzed using, for example, (log) risk ratios. Finally, one could also consider a combined analysis based on standardized mean differences computed from the means and standard deviations where available and using probit transformed risk differences (which also provide estimates of the standardized mean difference) for the remaining studies.
</p>
<p>Some degree of patient education was provided in all studies. In addition, the <code>type</code> variable indicates what additional intervention components were included in each study:
</p>

<ol>
<li><p> optimal self-management (writing action plan, self-monitoring, regular medical review),
</p>
</li>
<li><p> self-monitoring and regular medical review,
</p>
</li>
<li><p> self-monitoring only,
</p>
</li>
<li><p> regular medical review only,
</p>
</li>
<li><p> written action plan only.
</p>
</li></ol>



<h3>Concepts</h3>

<p>medicine, primary care, risk ratios, standardized mean differences
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Gibson, P. G., Powell, H., Wilson, A., Abramson, M. J., Haywood, P., Bauman, A., Hensley, M. J., Walters, E. H., &amp; Roberts, J. J. L. (2002). Self-management education and regular practitioner review for adults with asthma. <em>Cochrane Database of Systematic Reviews</em>, <b>3</b>, CD001117. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD001117&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.gibson2002
dat

## Not run: 

### load metafor package
library(metafor)

### compute standardized mean differences and corresponding sampling variances
dat &lt;- escalc(measure="SMD", m1i=m1i, sd1i=sd1i, n1i=n1i, m2i=m2i, sd2i=sd2i, n2i=n2i, data=dat)
dat

### fit an equal-effects model to the standardized mean differences (as in Gibson et al., 2002)
res &lt;- rma(yi, vi, data=dat, method="EE")
print(res, digits=2)

### compute log risk ratios and corresponding sampling variances
dat &lt;- escalc(measure="RR", ai=ai, bi=bi, ci=ci, di=di, data=dat)
dat

### fit an equal-effects model to the log risk ratios
res &lt;- rma(yi, vi, data=dat, method="EE")
print(res, digits=2)
predict(res, transf=exp, digits=2)

### note: Gibson et al. (2002) used the Mantel-Haenszel method for their analysis
rma.mh(measure="RR", ai=ai, bi=bi, ci=ci, di=di, data=dat, digits=2)

### compute standardized mean differences where possible and otherwise probit transformed
### risk differences (which also provide estimates of the standardized mean differences)
dat &lt;- escalc(measure="SMD", m1i=m1i, sd1i=sd1i, n1i=n1i,
                             m2i=m2i, sd2i=sd2i, n2i=n2i, data=dat, add.measure=TRUE)
dat &lt;- escalc(measure="PBIT", ai=ai, bi=bi, ci=ci, di=di, data=dat, replace=FALSE, add.measure=TRUE)
dat

### fit a random-effects model to these estimates
res &lt;- rma(yi, vi, data=dat)
print(res, digits=2)

### meta-regression model examining if there are systematic differences based on the
### type of measure used (there are only 2 studies where measure="PBIT", so this isn't
### very conclusive here, but shown for illustration purposes)
res &lt;- rma(yi, vi, mods = ~ measure, data=dat)
print(res, digits=2)
predict(res, newmods=1, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.graves2010'>Studies on the Effectiveness of Injected Cholera Vaccines</h2><span id='topic+dat.graves2010'></span>

<h3>Description</h3>

<p>Results from 17 studies on the effectiveness of injected vaccines against cholera.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.graves2010
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author/study name and publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of cholera cases in the vaccinated group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of individuals in the vaccinated group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of cholera cases in the placebo group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of individuals in the placebo group
</td>
</tr>

</table>



<h3>Details</h3>

<p>Cholera is an infection caused by certain strains of the bacterium <em>Vibrio cholerae</em>. When untreated, mortality rates can be as high as 50-60%. Proper sanitation practices are usually effective in preventing outbreaks, but a number of oral and injectable vaccines have also been developed. The Cochrane review by Graves et al. (2010) examined the effectiveness of injectable vaccines for preventing cholera cases and death. The present dataset includes results from 17 studies that reported the number of cholera cases in vaccinated and placebo/comparison groups up to 7 months after the treatment.
</p>


<h3>Concepts</h3>

<p>medicine, risk ratios, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Graves, P. M., Deeks, J. J., Demicheli, V., &amp; Jefferson, T. (2010). Vaccines for preventing cholera: Killed whole cell or other subunit vaccines (injected). <em>Cochrane Database of Systematic Reviews</em>, <b>8</b>, CD000974. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD000974.pub2&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.graves2010
dat

## Not run: 

### load metafor package
library(metafor)

### analysis using the Mantel-Haenszel method
rma.mh(measure="RR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.gurusamy2011'>Studies on Interventions to Reduce Mortality after Liver Transplantation</h2><span id='topic+dat.gurusamy2011'></span>

<h3>Description</h3>

<p>Results from 14 trials examining the mortality risk of interventions for decreasing blood loss and blood transfusion requirements during liver transplantation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.gurusamy2011
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study information </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>death</b>     </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> mortality at 60 days post-transplantation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of individuals
</td>
</tr>

</table>



<h3>Details</h3>

<p>This network meta-analysis compared the effectiveness of seven interventions for decreasing blood loss and blood transfusion requirements during liver transplantation (Gurusamy et al., 2011).
</p>
<p>Fourteen studies reported mortality at 60 days, in 1,002 patients. Forty-five deaths were reported across all studies (4.5%). Six studies observed deaths in all treatment arms while three studies did not observe any deaths.
</p>
<p>This data set was used in Efthimiou et al. (2019) to introduce the Mantel-Haenszel method for network meta-analysis.
</p>
<p>One of the treatments (solvent detergent plasma) was only included in one study with zero events in both treatment arms; this study was excluded from all network meta-analyses. In addition, no death was observed in the antithrombin III arm of the only study evaluating this treatment which was excluded from the Mantel-Haenszel network meta-analysis.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, network meta-analysis, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Gurusamy, K. S., Pissanou, T., Pikhart, H., Vaughan, J., Burroughs, A. K., &amp; Davidson, B. R. (2011). Methods to decrease blood loss and transfusion requirements for liver transplantation. <em>Cochrane Database of Systematic Reviews</em>, <b>12</b>, CD009052. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD009052.pub2&#8288;</code>
</p>


<h3>References</h3>

<p>Efthimiou, O., Rücker, G., Schwarzer, G., Higgins, J., Egger, M., &amp; Salanti, G. (2019). A Mantel-Haenszel model for network meta-analysis of rare events. <em>Statistics in Medicine</em>, <b>38</b>(16), 2992&ndash;3012. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.8158&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="netmeta.html#topic+pairwise">pairwise</a></code>, <code><a href="meta.html#topic+metabin">metabin</a></code>, <code><a href="netmeta.html#topic+netmetabin">netmetabin</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show first 6 rows of the dataset
head(dat.gurusamy2011)

### Only study evaluating solvent detergent plasma
subset(dat.gurusamy2011, study == "Williamson 1999")

### Only study evaluating antithrombin III
subset(dat.gurusamy2011, study == "Baudo 1992")

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Change appearance of confidence intervals
cilayout("(", "-")

### Transform data from long arm-based format to contrast-based
### format. Argument 'sm' has to be used for odds ratio as summary
### measure; by default the risk ratio is used in the metabin function
### called internally.
pw &lt;- pairwise(treatment, death, n, studlab = study,
  data = dat.gurusamy2011, sm = "OR")

### Conduct Mantel-Haenszel network meta-analysis (NMA)
net.MH &lt;- netmetabin(pw, ref = "cont")

### Conduct inverse variance (IV) network meta-analysis
net.IV &lt;- netmeta(pw, ref = "cont")

### Network graph (Mantel-Haenszel NMA)
netgraph(net.MH, seq = "optimal", col = "black", plastic = FALSE,
  points = TRUE, pch = 21, cex.points = 3, col.points = "black",
  bg.points = "gray", thickness = "se.fixed",
  number.of.studies = TRUE)

### Full network graph (based on inverse variance method, including
### study comparing Antithrombin III with Control/Placebo)
netgraph(net.IV,
  seq = "optimal", col = "black", plastic = FALSE,
  points = TRUE, pch = 21, cex.points = 3, col.points = "black",
  bg.points = "gray", thickness = "se.fixed",
  number.of.studies = TRUE)

### Compare results for Mantel-Haenszel and IV NMA
forest(netbind(net.MH, net.IV,
  random = FALSE, name = c("MH NMA", "IV NMA")))

### Show results for Mantel-Haenszel NMA
net.MH
forest(net.MH)

### League table with network estimates in lower triangle and direct
### estimates in upper triangle
netleague(net.MH)

### Assess inconsistency
print(netsplit(net.MH), show = "both", ci = TRUE, overall = FALSE,
  nchar.trts = 6)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hackshaw1998'>Studies on the Risk of Lung Cancer in Women Exposed to Environmental Tobacco Smoke</h2><span id='topic+dat.hackshaw1998'></span>

<h3>Description</h3>

<p>Results from 37 studies on the risk of lung cancer in women exposed to environmental tobacco smoke (ETS) from their smoking spouse.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hackshaw1998
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> first author of study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>    </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> country where study was conducted </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>design</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> study design (either cohort or case-control) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cases</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of lung cancer cases </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>or</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> odds ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>or.lb</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> lower bound of 95% CI for the odds ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>or.ub</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> upper bound of 95% CI for the odds ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> log odds ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> corresponding sampling variance
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 37 studies (4 cohort, 33 case-control) examining if women (who are lifelong nonsmokers) have an elevated risk for lung cancer due to exposure to environmental tobacco smoke (ETS) from their smoking spouse. Values of the log odds ratio greater than 0 indicate an increased risk of cancer in exposed women compared to women not exposed to ETS from their spouse.
</p>
<p>Note that the log odds ratios and corresponding sampling variances were back-calculated from the reported odds ratios and confidence interval (CI) bounds (see &lsquo;Examples&rsquo;). Since the reported values were rounded to some extent, this introduces some minor inaccuracies into the back-calculations. The overall estimate reported in Hackshaw et al. (1997) and Hackshaw (1998) can be fully reproduced though.
</p>


<h3>Concepts</h3>

<p>medicine, oncology, epidemiology, smoking, odds ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Hackshaw, A. K., Law, M. R., &amp; Wald, N. J. (1997). The accumulated evidence on lung cancer and environmental tobacco smoke. <em>British Medical Journal</em>, <b>315</b>(7114), 980&ndash;988. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/bmj.315.7114.980&#8288;</code>
</p>
<p>Hackshaw, A. K. (1998). Lung cancer and passive smoking. <em>Statistical Methods in Medical Research</em>, <b>7</b>(2), 119&ndash;136. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/096228029800700203&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.hackshaw1998
head(dat, 10)

## Not run: 

### load metafor package
library(metafor)

### random-effects model using the log odds ratios
res &lt;- rma(yi, vi, data=dat, method="DL")
res

### estimated average odds ratio with CI (and prediction interval)
predict(res, transf=exp, digits=2)

### illustrate how the log odds ratios and corresponding sampling variances
### were back-calculated based on the reported odds ratios and CI bounds
dat$yi &lt;- log(dat$or)
dat$vi &lt;- ((log(dat$or.ub) - log(dat$or.lb)) / (2*qnorm(.975)))^2


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hahn2001'>Studies on the Effectiveness of Different Rehydration Solutions for the Prevention of Unscheduled Intravenous Infusion in Children with Diarrhoea</h2><span id='topic+dat.hahn2001'></span>

<h3>Description</h3>

<p>Results from 12 trials examining the effectiveness of a reduced versus standard rehydration solution for the prevention of unscheduled intravenous infusion in children with diarrhoea.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hahn2001
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> trial name and year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of children requiring unscheduled intravenous infusion in the reduced rehydration solution group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of children in the reduced rehydration solution group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of children requiring unscheduled intravenous infusion in the standard rehydration solution group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of children in the standard rehydration solution group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 12 randomized clinical trials that examined the effectiveness of a reduced osmolarity oral rehydration solution (total osmolarity &lt;250 mmol/l with reduced sodium) with a standard WHO oral rehydration solution (sodium 90 mmol/l, glucose 111mmol/l, total osmolarity 311 mmol/l) for the prevention of unscheduled intravenous infusion in children with diarrhoea.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Hahn, S., Kim, Y., &amp; Garner, P. (2001). Reduced osmolarity oral rehydration solution for treating dehydration due to diarrhoea in children: Systematic review. <em>British Medical Journal</em>, <b>323</b>(7304), 81&ndash;85. <code style="white-space: pre;">&#8288;https://doi.org/10.1136/bmj.323.7304.81&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.hahn2001
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of (log) odds rations using the Mantel-Haenszel method
res &lt;- rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, digits=2, slab=study)
res

### forest plot (also show studies that were excluded from the analysis)
options(na.action="na.pass")
forest(res, atransf=exp, at=log(c(.01, .1, 1, 10, 100)), header=TRUE)
options(na.action="na.omit")


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hannum2020'>Studies Comparing Objective and Subjective Olfactory Loss in COVID-19 Patients</h2><span id='topic+dat.hannum2020'></span>

<h3>Description</h3>

<p>Results from 35 studies measuring olfactory loss in COVID-19 patients using either objective or subjective measures.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hannum2020
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>authorName</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author of study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DOI</b>                  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> article DOI number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>                   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of Covid-19 positive patients in the study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xi</b>                   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of Covid-19 positive patients in the study with olfactory loss </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>percentOlfactoryLoss</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> percent of the sample with olfactory loss </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>objectivity</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> objective or subjective measure used </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>measured</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> outcome measure </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>testType</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of test used </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>              </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> country where patients were treated </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>patientType</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of patient information and location where being treated
</td>
</tr>

</table>



<h3>Details</h3>

<p>One of the symptoms of COVID-19 infection is olfactory loss (loss of smell) either recently acquired anosmia (complete loss of smell) or hyposmia (partial loss of smell). One challenge to reaching this symptom is the wide range of reported prevalence for this symptom ranging from 5 percent to 98 percent. In this dataset studies were grouped into one of two groups based on the type of method used to measure smell loss (either subjective measures, such as self-reported smell loss, or objective measures using rated stimuli).
</p>


<h3>Concepts</h3>

<p>medicine, covid-19, proportions
</p>


<h3>Author(s)</h3>

<p>W. Kyle Hamilton <a href="mailto:whamilton@ucmerced.edu">whamilton@ucmerced.edu</a> <a href="https://kylehamilton.com">https://kylehamilton.com</a>
</p>


<h3>Source</h3>

<p>Ramirez VA , Hannum ME, Lipson SJ, Herriman RD, Toskala AK, Lin C, Joseph PV, Reed DR. 2020. COVID-19 Smell Loss Prevalence Tracker. Available from: <code style="white-space: pre;">&#8288;https://vicente-ramirez.shinyapps.io/COVID19_Olfactory_Dashboard/&#8288;</code> and <code style="white-space: pre;">&#8288;https://github.com/vramirez4/OlfactoryLoss&#8288;</code> (accessed August 11, 2021)
</p>


<h3>References</h3>

<p>Hannum, M. E., Ramirez, V. A., Lipson, S. J., Herriman, R. D., Toskala, A. K., Lin, C., Joseph, P. V., &amp; Reed, D. R. (2020). Objective sensory testing methods reveal a higher prevalence of olfactory loss in COVID-19 positive patients compared to subjective methods: A systematic review and meta-analysis. <em>Chemical Senses</em>, <b>45</b>(9), 865&ndash;874. <code style="white-space: pre;">&#8288;https://doi.org/10.1093/chemse/bjaa064&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># copy data into 'dat' and examine data
dat &lt;- dat.hannum2020
dat

## Not run: 
   
# load metafor package
library(metafor)

# compute effect size
dat &lt;- escalc(measure="PR", xi=xi, ni=ni, data=dat)

# split data into objective and subjective datasets
dat_split &lt;- split(dat, dat$objectivity)
dat_objective &lt;- dat_split[["Objective"]]
dat_subjective &lt;- dat_split[["Subjective"]]

# random-effects model all studies
res_all &lt;- rma(yi, vi, data=dat)
print(res_all, digits=2)

# random-effects model objective
res_objective &lt;- rma(yi, vi, data=dat_objective)
print(res_objective, digits=2)

# random-effects model subjective
res_subjective &lt;- rma(yi, vi, data=dat_subjective)
print(res_subjective, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hart1999'>Studies on the Effectiveness of Warfarin for Preventing Strokes</h2><span id='topic+dat.hart1999'></span>

<h3>Description</h3>

<p>Results from 6 clinical trials examining the effectiveness of adjusted-dose warfarin for preventing strokes in patients with atrial fibrillation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hart1999
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study name (abbreviated) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of strokes in the warfarin group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the warfarin group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total person-time (in years) in the warfarin group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of strokes in the placebo/control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the placebo/control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total person-time (in years) in the placebo/control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>compgrp</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of comparison group (placebo or control) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>prevtype</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of prevention (primary or secondary) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trinr</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> target range for the international normalized ratio (INR)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 6 studies provide data with respect to the number of strokes in the warfarin and the comparison (placebo or control) group. In addition, the number of patients and the total person-time (in years) is provided for the two groups. The goal of the meta-analysis was to examine the effectiveness of adjusted-dose warfarin for preventing strokes in patients with atrial fibrillation.
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, incidence rates
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Hart, R. G., Benavente, O., McBride, R., &amp; Pearce, L. A. (1999). Antithrombotic therapy to prevent stroke in patients with atrial fibrillation: A meta-analysis. <em>Annals of Internal Medicine</em>, <b>131</b>(7), 492&ndash;501. <code style="white-space: pre;">&#8288;https://doi.org/10.7326/0003-4819-131-7-199910050-00003&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.hart1999
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log incidence rate ratios and corresponding sampling variances
dat &lt;- escalc(measure="IRR", x1i=x1i, x2i=x2i, t1i=t1i, t2i=t2i, data=dat)
dat

### meta-analysis of log incidence rate ratios using a random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### average incidence rate ratio with 95% CI
predict(res, transf=exp)

### forest plot with extra annotations
par(mar=c(5,4,1,2))
forest(res, xlim=c(-11, 5), at=log(c(.05, .25, 1, 4)), atransf=exp,
       slab=paste0(study, " (", year, ")"),
       ilab=cbind(paste(x1i, "/", t1i, sep=" "),
                  paste(x2i, "/", t2i, sep=" ")),
       ilab.xpos=c(-6.5,-4), cex=.85, header="Study (Year)")
op &lt;- par(cex=.85, font=2)
text(c(-6.5,-4), 8.5, c("Warfarin", "Control"))
text(c(-6.5,-4), 7.5, c("Strokes / PT", "Strokes / PT"))
segments(x0=-8, y0=8, x1=-2.75, y1=8)
par(op)

### meta-analysis of incidence rate differences using a random-effects model
res &lt;- rma(measure="IRD", x1i=x1i, x2i=x2i, t1i=t1i, t2i=t2i, data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hartmannboyce2018'>Studies on the Effectiveness of Nicotine Replacement Therapy for Smoking Cessation</h2><span id='topic+dat.hartmannboyce2018'></span>

<h3>Description</h3>

<p>Results from 133 studies examining the effectiveness of nicotine replacement therapy (NRT) for smoking cessation at 6+ months of follow-up.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hartmannboyce2018
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x.nrt</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the NRT group who were abstinent at the follow-up </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.nrt</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the NRT group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x.ctrl</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the control group who were abstinent at the follow-up </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.ctrl</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of NRT provided in the treatment group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 133 studies examining the effectiveness of nicotine replacement therapy (NRT) for smoking cessation. The results given in this dataset pertain to abstinence at 6+ months of follow-up. NRT was provided to participants in the treatment groups in various forms as indicated by the <code>treatment</code> variable (e.g., gum, patch, inhalator). Note that the dataset includes 136 rows, since a few studies included multiple treatments.
</p>


<h3>Concepts</h3>

<p>medicine, smoking, risk ratios, Mantel-Haenszel method
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Hartmann‐Boyce, J., Chepkin, S. C., Ye, W., Bullen, C. &amp; Lancaster, T. (2018). Nicotine replacement therapy versus control for smoking cessation. <em>Cochrane Database of Systematic Reviews</em>, <b>5</b>, CD000146. <code style="white-space: pre;">&#8288;https://doi.org//10.1002/14651858.CD000146.pub5&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.hartmannboyce2018
head(dat, 10)

## Not run: 

### load metafor package
library(metafor)

### turn treatment into a factor with the desired ordering
dat$treatment &lt;- factor(dat$treatment, levels=unique(dat$treatment))

### meta-analysis per treatment using the M-H method
lapply(split(dat, dat$treatment), function(x)
       rma.mh(measure="RR", ai=x.nrt,  n1i=n.nrt,
                            ci=x.ctrl, n2i=n.ctrl, data=x, digits=2))

### all combined
rma.mh(measure="RR", ai=x.nrt,  n1i=n.nrt,
                     ci=x.ctrl, n2i=n.ctrl, data=dat, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hasselblad1998'>Studies on the Effectiveness of Counseling for Smoking Cessation</h2><span id='topic+dat.hasselblad1998'></span>

<h3>Description</h3>

<p>Results from 24 studies on the effectiveness of various counseling types for smoking cessation.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hasselblad1998
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> id number for each treatment arm </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>authors</b>    </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trt</b>        </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> intervention group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of individuals abstinent </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of individuals in group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 24 studies on the effectiveness of various counseling types for smoking cessation (i.e., self-help, individual counseling, group counseling, and no contact). The dataset indicates the total number of individuals within each study arm and the number that were abstinent from 6 to 12 months. The majority of the studies compared two interventions types against each other, while 2 studies compared three types against each other simultaneously.
</p>
<p>The data can be used for a &lsquo;network meta-analysis&rsquo; (also called a &lsquo;mixed treatment comparison&rsquo;). The code below shows how such an analysis can be conducted using an arm-based and a contrast-based model (see Salanti et al., 2008, for more details).
</p>


<h3>Concepts</h3>

<p>medicine, psychology, smoking, odds ratios, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Hasselblad, V. (1998). Meta-analysis of multitreatment studies. <em>Medical Decision Making</em>, <b>18</b>(1), 37&ndash;43. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/0272989X9801800110&#8288;</code>
</p>


<h3>References</h3>

<p>Gleser, L. J., &amp; Olkin, I. (2009). Stochastically dependent effect sizes. In H. Cooper, L. V. Hedges, &amp; J. C. Valentine (Eds.), <em>The handbook of research synthesis and meta-analysis</em> (2nd ed., pp. 357&ndash;376). New York: Russell Sage Foundation.
</p>
<p>Law, M., Jackson, D., Turner, R., Rhodes, K., &amp; Viechtbauer, W. (2016). Two new methods to fit models for network meta-analysis with random inconsistency effects. <em>BMC Medical Research Methodology</em>, <b>16</b>, 87. <code style="white-space: pre;">&#8288;https://doi.org/10.1186/s12874-016-0184-5&#8288;</code>
</p>
<p>Salanti, G., Higgins, J. P. T., Ades, A. E., &amp; Ioannidis, J. P. A. (2008). Evaluation of networks of randomized trials. <em>Statistical Methods in Medical Research</em>, <b>17</b>(3), 279&ndash;301. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/0962280207080643&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.hasselblad1998
dat

## Not run: 

### load metafor package
library(metafor)

### create network graph ('igraph' package must be installed)
library(igraph, warn.conflicts=FALSE)
pairs &lt;- data.frame(do.call(rbind,
   sapply(split(dat$trt, dat$study), function(x) t(combn(x,2)))), stringsAsFactors=FALSE)
lvls &lt;- c("no_contact", "self_help", "ind_counseling", "grp_counseling")
pairs$X1 &lt;- factor(pairs$X1, levels=lvls)
pairs$X2 &lt;- factor(pairs$X2, levels=lvls)
tab &lt;- table(pairs[,1], pairs[,2])
tab # adjacency matrix
g &lt;- graph_from_adjacency_matrix(tab, mode = "plus", weighted=TRUE, diag=FALSE)
vertex_attr(g, "name") &lt;- c("No Contact", "Self-Help",
                            "Individual\nCounseling", "Group\nCounseling")
plot(g, edge.curved=FALSE, edge.width=E(g)$weight, layout=layout_on_grid,
     vertex.size=45, vertex.color="lightgray", vertex.label.color="black", vertex.label.font=2)

### calculate log odds for each study arm
dat &lt;- escalc(measure="PLO", xi=xi, ni=ni, add=1/2, to="all", data=dat)
dat

### convert trt variable to factor with desired ordering of levels
dat$trt &lt;- factor(dat$trt, levels=c("no_contact", "self_help", "ind_counseling", "grp_counseling"))

### add a space before each level (this makes the output a bit more legible)
levels(dat$trt) &lt;- paste0(" ", levels(dat$trt))

### network meta-analysis using an arm-based model with fixed study effects
### by setting rho=1/2, tau^2 reflects the amount of heterogeneity for all treatment comparisons
res &lt;- rma.mv(yi, vi, mods = ~ factor(study) + trt - 1,
              random = ~ trt | study, rho=1/2, data=dat, btt="trt")
res

### all pairwise odds ratios of interventions versus no contact
predict(res, newmods=cbind(matrix(0, nrow=3, ncol=24), diag(3)),
        intercept=FALSE, transf=exp, digits=2)

### all pairwise odds ratios comparing interventions (ic vs sh, gc vs sh, and gc vs ic)
predict(res, newmods=cbind(matrix(0, nrow=3, ncol=24), rbind(c(-1,1,0), c(-1,0,1), c(0,-1,1))),
        intercept=FALSE, transf=exp, digits=2)

### forest plot of ORs of interventions versus no contact
dev.new(width=7, height=4)
par(mar=c(5,4,1,2))
forest(c(0,res$beta[25:27]), sei=c(0,res$se[25:27]), psize=1, xlim=c(-3,4), digits=c(2,1), efac=2,
       slab=c("No Contact", "Self-Help", "Individual Counseling", "Group Counseling"),
       atransf=exp, at=log(c(.5, 1, 2, 4, 8)), xlab="Odds Ratio for Intervention vs. No Contact",
       header=c("Intervention", "Odds Ratio [95% CI]"))

############################################################################

### restructure dataset to a contrast-based format
dat &lt;- to.wide(dat.hasselblad1998, study="study", grp="trt", ref="no_contact", grpvars=6:7)

### calculate log odds ratios for each treatment comparison
dat &lt;- escalc(measure="OR", ai=xi.1, n1i=ni.1,
                            ci=xi.2, n2i=ni.2, add=1/2, to="all", data=dat)
dat

### calculate the variance-covariance matrix of the log odds ratios for multitreatment studies
### see Gleser &amp; Olkin (2009), equation (19.11), for the covariance equation
calc.v &lt;- function(x) {
   v &lt;- matrix(1/(x$xi.2[1] + 1/2) + 1/(x$ni.2[1] - x$xi.2[1] + 1/2), nrow=nrow(x), ncol=nrow(x))
   diag(v) &lt;- x$vi
   v
}
V &lt;- bldiag(lapply(split(dat, dat$study), calc.v))

### add contrast matrix to dataset
dat &lt;- contrmat(dat, grp1="trt.1", grp2="trt.2")
dat

### network meta-analysis using a contrast-based random-effects model
### by setting rho=1/2, tau^2 reflects the amount of heterogeneity for all treatment comparisons
res &lt;- rma.mv(yi, V, mods = ~ self_help + ind_counseling + grp_counseling - 1,
              random = ~ comp | study, rho=1/2, data=dat)
res

### predicted odds ratios of interventions versus no contact
predict(res, newmods=diag(3), transf=exp, digits=2)

### fit random inconsistency effects model (see Law et al., 2016)
res &lt;- rma.mv(yi, V, mods = ~ self_help + ind_counseling + grp_counseling - 1,
              random = list(~ comp | study, ~ comp | design), rho=1/2, phi=1/2, data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.hine1989'>Studies on Prophylactic Use of Lidocaine After a Heart Attack</h2><span id='topic+dat.hine1989'></span>

<h3>Description</h3>

<p>Results from 6 studies evaluating mortality from prophylactic use of lidocaine in acute myocardial infarction.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.hine1989
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>source</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> source of data </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in lidocaine group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in lidocaine group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>Hine et al. (1989) conducted a meta-analysis of death rates in randomized controlled trials in which prophylactic lidocaine was administered to patients with confirmed or suspected acute myocardial infarction. The dataset describes the mortality at the end of the assigned treatment period for control and intravenous lidocaine treatment groups for six studies. The question of interest is whether there is a detrimental effect of lidocaine. Because the studies were conducted to compare rates of arrhythmias following a heart attack, the studies, taken individually, are too small to detect important differences in mortality rates.
</p>
<p>The data in this dataset were obtained from Table I in Normand (1999, p. 322).
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, risk differences
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Normand, S. T. (1999). Meta-analysis: Formulating, evaluating, combining, and reporting. <em>Statistics in Medicine</em>, <b>18</b>(3), 321&ndash;359. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/(sici)1097-0258(19990215)18:3&lt;321::aid-sim28&gt;3.0.co;2-p&#8288;</code>
</p>


<h3>References</h3>

<p>Hine, L. K., Laird, N., Hewitt, P., &amp; Chalmers, T. C. (1989). Meta-analytic evidence against prophylactic use of lidocaine in acute myocardial infarction. <em>Archives of Internal Medicine</em>, <b>149</b>(12), 2694&ndash;2698. <code style="white-space: pre;">&#8288;https://doi.org/10.1001/archinte.1989.00390120056011&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.hine1989
dat

## Not run: 

### load metafor package
library(metafor)

### calculate risk differences and corresponding sampling variances
dat &lt;- escalc(measure="RD", n1i=n1i, n2i=n2i, ai=ai, ci=ci, data=dat)
dat

### meta-analysis of risk differences using a random-effects model
res &lt;- rma(yi, vi, data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.ishak2007'>Studies on Deep-Brain Stimulation in Patients with Parkinson's disease</h2><span id='topic+dat.ishak2007'></span>

<h3>Description</h3>

<p>Results from 46 studies examining the effects of deep-brain stimulation on motor skills of patients with Parkinson's disease.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.ishak2007
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> (first) author and year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y1i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed mean difference at 3 months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>v1i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sampling variance of the mean difference at 3 months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y2i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed mean difference at 6 months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>v2i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sampling variance of the mean difference at 6 months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y3i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed mean difference at 12 months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>v3i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sampling variance of the mean difference at 12 months </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y4i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed mean difference at the long-term follow-up </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>v4i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sampling variance of the mean difference at the long-term follow-up </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mdur</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> mean disease duration (in years) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mbase</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> mean baseline UPDRS score
</td>
</tr>

</table>



<h3>Details</h3>

<p>Deep-brain stimulation (DBS), which is delivered through thin surgically implanted wires in specific areas of the brain and controlled by the patient, is meant to provide relief of the debilitating symptoms of Parkinson's disease. The dataset includes the results from 46 studies examining the effects of DBS of the subthalamic nucleus on motor functioning, measured with the Unified Parkinson's Disease Rating Scale (UPDRS). The effect size measure for this meta-analysis was the mean difference of the scores while the stimulator is active and the baseline scores (before implantation of the stimulator). Since lower scores on the UPDRS indicate better functioning, negative numbers indicate improvements in motor skills. Effects were generally measured at 3, 6, and 12 months after implantation of the stimulator, with some studies also including a further long-term follow-up. However, the number of measurements differed between studies - hence the missing data on some of the measurement occasions.
</p>
<p>Since the same patients were followed over time within a study, effect size estimates from multiple measurement occasions are likely to be correlated. A multivariate model accounting for the correlation in the effects can be used to meta-analyze these data. A difficulty with this approach is the lack of information about the correlation of the measurements over time in the individual studies. The approach taken by Ishak et al. (2007) was to assume an autoregressive (AR1) structure for the estimates within the individual studies. In addition, the correlation in the true effects was modeled, again using an autoregressive structure.
</p>


<h3>Concepts</h3>

<p>medicine, raw mean differences, longitudinal models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Ishak, K. J., Platt, R. W., Joseph, L., Hanley, J. A., &amp; Caro, J. J. (2007). Meta-analysis of longitudinal studies. <em>Clinical Trials</em>, <b>4</b>(5), 525&ndash;539. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/1740774507083567&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.ishak2007
head(dat, 5)

## Not run: 

### load metafor package
library(metafor)

### create long format dataset
dat &lt;- reshape(dat, direction="long", idvar="study", v.names=c("yi","vi"),
               varying=list(c(2,4,6,8), c(3,5,7,9)))
dat &lt;- dat[order(study, time),]

### remove missing measurement occasions from dat.long
dat &lt;- dat[!is.na(yi),]
rownames(dat) &lt;- NULL
head(dat, 8)

### construct the full (block diagonal) V matrix with an AR(1) structure
### assuming an autocorrelation of 0.97 as estimated by Ishak et al. (2007)
V &lt;- vcalc(vi, cluster=study, time1=time, phi=0.97, data=dat)

### plot data
with(dat, interaction.plot(time, study, yi, type="b", pch=19, lty="solid", xaxt="n",
          legend=FALSE, xlab="Time Point", ylab="Mean Difference", bty="l"))
axis(side=1, at=1:4, lab=c("1 (3 months)", "2 (6 months)", "3 (12 months)", "4 (12+ months)"))

### multivariate model with heteroscedastic AR(1) structure for the true effects
res &lt;- rma.mv(yi, V, mods = ~ factor(time) - 1, random = ~ time | study,
              struct = "HAR", data = dat)
print(res, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.kalaian1996'>Studies on the Effectiveness of Coaching for the SAT</h2><span id='topic+dat.kalaian1996'></span>

<h3>Description</h3>

<p>Results from studies examining the effectiveness of coaching on the performance on the Scholastic Aptitude Test (SAT).</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.kalaian1996
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> row (effect) id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the coached group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants in the uncoached group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>outcome</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> subtest (verbal or math) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hrs</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> hours of coaching </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ets</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study conducted by the Educational Testing Service (ETS) (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>homework</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> assignment of homework outside of the coaching course (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>type</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study type (1 = randomized study, 2 = matched study, 3 = nonequivalent comparison study)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The effectiveness of coaching for the Scholastic Aptitude Test (SAT) has been examined in numerous studies. This dataset contains standardized mean differences comparing the performance of a coached versus uncoached group on the verbal and/or math subtest of the SAT. Studies may report a standardized mean difference for the verbal subtest, the math subtest, or both. In the latter case, the two standardized mean differences are not independent (since they were measured in the same group of subjects). The number of hours of coaching (variable <code>hrs</code>), whether the study was conducted by the Educational Testing Service (variable <code>ets</code>), whether homework was assigned outside of the coaching course (variable <code>homework</code>), and the study type (variable <code>type</code>) may be potential moderators of the treatment effect.
</p>


<h3>Concepts</h3>

<p>education, standardized mean differences, multivariate models, meta-regression
</p>


<h3>Note</h3>

<p>The dataset was obtained from Table 1 in Kalaian and Raudenbush (1996). However, there appear to be some inconsistencies between the data in the table and those that were actually used for the analyses (see &lsquo;Examples&rsquo;).
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Kalaian, H. A., &amp; Raudenbush, S. W. (1996). A multivariate mixed linear model for meta-analysis. <em>Psychological Methods</em>, <b>1</b>(3), 227&ndash;235. <code style="white-space: pre;">&#8288;https://doi.org/10.1037/1082-989X.1.3.227&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.kalaian1996
head(dat, 12)

## Not run: 

### load metafor package
library(metafor)

### check ranges
range(dat$yi[dat$outcome == "verbal"]) # -0.35 to 0.74 according to page 230
range(dat$yi[dat$outcome == "math"])   # -0.53 to 0.60 according to page 231

### comparing this with Figure 1 in the paper reveals some discrepancies
par(mfrow=c(1,2), mar=c(5,4,1,1))
plot(log(dat$hrs[dat$outcome == "verbal"]), dat$yi[dat$outcome == "verbal"],
     pch=19, xlab="Log(Coaching Hours)", ylab="Effect Size (verbal)",
     xlim=c(1,6), ylim=c(-0.5,1), xaxs="i", yaxs="i")
abline(h=c(-0.5,0,0.5), lty="dotted")
abline(v=log(c(5,18)),  lty="dotted")
plot(log(dat$hrs[dat$outcome == "math"]), dat$yi[dat$outcome == "math"],
     pch=19, xlab="Log(Coaching Hours)", ylab="Effect Size (math)",
     xlim=c(1,6), ylim=c(-1.0,1), xaxs="i", yaxs="i")
abline(h=c(-0.5,0,0.5), lty="dotted")
abline(v=log(c(5,18)),  lty="dotted")

### construct variance-covariance matrix assuming rho = 0.66 for effect sizes
### corresponding to the 'verbal' and 'math' outcome types
V &lt;- vcalc(vi, cluster=study, type=outcome, data=dat, rho=0.66)

### fit multivariate random-effects model
res &lt;- rma.mv(yi, V, mods = ~ outcome - 1,
              random = ~ outcome | study, struct="UN",
              data=dat, digits=3)
res

### test whether the effect differs for the math and verbal subtest
anova(res, X=c(1,-1))

### log-transform and mean center the hours of coaching variable
dat$loghrs &lt;- log(dat$hrs) - mean(log(dat$hrs), na.rm=TRUE)

### fit multivariate model with log(hrs) as moderator
res &lt;- rma.mv(yi, V, mods = ~ outcome + outcome:loghrs - 1,
              random = ~ outcome | study, struct="UN",
              data=dat, digits=3)
res

### fit model with tau2 = 0 for outcome verbal (which also constrains rho = 0)
res &lt;- rma.mv(yi, V, mods = ~ outcome + outcome:loghrs - 1,
              random = ~ outcome | study, struct="UN", tau2=c(NA,0),
              data=dat, digits=3)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.kearon1998'>Studies on the Accuracy of Venous Ultrasonography for the Diagnosis of Deep Venous Thrombosis</h2><span id='topic+dat.kearon1998'></span>

<h3>Description</h3>

<p>Results from diagnostic accuracy studies examining the accuracy of venous ultrasonography for the diagnosis of deep venous thrombosis.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.kearon1998
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>patients</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> patient group (either symptomatic or asymptomatic patients) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tp</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of true positives </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>np</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of positive patients (cases) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tn</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of true negatives </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nn</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of negative patients (non-cases)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies included in the dataset examined the accuracy of venous ultrasonography for the diagnossis of a first deep venous thrombosis in symptomatic and asymptomatic patients. Cases and non-cases were determined based on contrast venography. Venous ultrasonography was then used to make a diagnosis, leading to a given number of true positives and negatives.
</p>
<p>A subset of this dataset (using only the studies with asymptomatic patients) was used by Deeks et al. (2005) to illustrate methods for detecting publication bias (or small-study effects) in meta-analyses of diagnostic accuracy studies.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, diagnostic accuracy studies, multivariate models, publication bias
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Kearon, C., Julian, J. A., Math, M., Newman, T. E., &amp; Ginsberg, J. S. (1998). Noninvasive diagnosis of deep venous thrombosis. <em>Annals of Internal Medicine</em>, <b>128</b>(8), 663&ndash;677. <code style="white-space: pre;">&#8288;https://doi.org/10.7326/0003-4819-128-8-199804150-00011&#8288;</code>
</p>


<h3>References</h3>

<p>Deeks, J. J., Macaskill, P., &amp; Irwig, L. (2005). The performance of tests of publication bias and other sample size effects in systematic reviews of diagnostic test accuracy was assessed. <em>Journal of Clinical Epidemiology</em>, <b>58</b>(9), 882&ndash;893. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.jclinepi.2005.01.016&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.kearon1998
head(dat)

## Not run: 

### load metafor package
library(metafor)

### calculate diagnostic log odds ratios and corresponding sampling variances
dat &lt;- escalc(measure="OR", ai=tp, n1i=np, ci=nn-tn, n2i=nn, data=dat, add=1/2, to="all")
head(dat)

### fit random-effects model for the symptomatic patients
res &lt;- rma(yi, vi, data=dat, subset=patients=="symptomatic")
res

### fit random-effects model for the asymptomatic patients
res &lt;- rma(yi, vi, data=dat, subset=patients=="asymptomatic")
res

### estimated average diagnostic odds ratio (with 95% CI)
predict(res, transf=exp, digits=2)

### regression test for funnel plot asymmetry using SE as predictor
reg &lt;- regtest(res, model="lm")
reg

### corresponding funnel plot
funnel(res, atransf=exp, xlim=c(0,7), at=log(c(1,10,100,1000)), ylim=c(0,1.5), steps=4)
ys &lt;- seq(0, 2, length=100)
lines(coef(reg$fit)[1] + coef(reg$fit)[2]*ys, ys, lwd=2, lty=3)

### regression test for funnel plot asymmetry using total sample size as predictor
reg &lt;- regtest(res, model="lm", predictor="ni")
reg

### corresponding funnel plot
funnel(res, yaxis="ni", atransf=exp, xlim=c(0,7), at=log(c(1,10,100,1000)), ylim=c(0,300), steps=4)
ys &lt;- seq(0, 300, length=100)
lines(coef(reg$fit)[1] + coef(reg$fit)[2]*ys, ys, lwd=2, lty=3)

### regression test for funnel plot asymmetry using 1/sqrt(ESS) as predictor (Deeks et al., 2005)
dat$invessi &lt;- 1/(4*dat$np) + 1/(4*dat$nn)
tmp &lt;- rma(yi, invessi, data=dat, subset=patients=="asymptomatic")
reg &lt;- regtest(tmp, model="lm")
reg

### corresponding funnel plot
funnel(tmp, atransf=exp, xlim=c(0,7), at=log(c(1,10,100,1000)), ylim=c(0,.15), steps=4,
       refline=coef(res), level=0, ylab="1/root(ess)")
ys &lt;- seq(0, .20, length=100)
lines(coef(reg$fit)[1] + coef(reg$fit)[2]*ys, ys, lwd=2, lty=3)

### convert data to long format
dat &lt;- to.long(measure="OR", ai=tp, n1i=np, ci=tn, n2i=nn,
               data=dat.kearon1998, subset=patients=="asymptomatic")
dat &lt;- dat[9:12]
levels(dat$group) &lt;- c("sensitivity", "specificity")
dat

### calculate logit-transformed sensitivities
dat &lt;- escalc(measure="PLO", xi=out1, mi=out2, data=dat, add=1/2, to="all",
              include=group=="sensitivity")
dat

### calculate logit-transformed specificities
dat &lt;- escalc(measure="PLO", xi=out1, mi=out2, data=dat, add=1/2, to="all",
              include=group=="specificity")
dat

### bivariate random-effects model for logit sensitivity and specificity
res &lt;- rma.mv(yi, vi, mods = ~ group - 1, random = ~ group | study, struct="UN", data=dat)
res

### estimated average sensitivity and specificity based on the model
predict(res, newmods = rbind(c(1,0),c(0,1)), transf=transf.ilogit, tau2.levels=c(1,2), digits=2)

### estimated average diagnostic odds ratio based on the model
predict(res, newmods = c(1,1), transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.knapp2017'>Studies on Differences in Planning Performance in Schizophrenia Patients versus Healthy Controls</h2><span id='topic+dat.knapp2017'></span>

<h3>Description</h3>

<p>Results from 31 studies examining differences in planning performance in schizophrenia patients versus healthy controls.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.knapp2017
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>author</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>task</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of task </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>difficulty</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> task difficulty </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>group1</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> identifier for patient group within studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>group2</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> identifier for control group within studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>comp</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> identifier for comparisons within studies </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference for planning performance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n_sz</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of schizophrenic patients </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n_hc</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of healthy controls </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference for IQ </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance
</td>
</tr>

</table>



<h3>Details</h3>

<p>The studies included in this dataset examined differences between schizophrenia patients and healthy controls with respect to their performance on the tower of London test (<a href="https://en.wikipedia.org/wiki/Tower_of_London_test">https://en.wikipedia.org/wiki/Tower_of_London_test</a>) or a similar cognitive tasks measuring planning ability. The outcome measure for this meta-analysis was the standardized mean difference (with positive values indicating better performance in the healthy controls compared to the schizophrenia patients).
</p>
<p>The dataset has a more complex structure for several reasons:
</p>

<ol>
<li><p> Studies 2, 3, 9, and 20 included more than one schizophrenia patient group and the standardized mean differences were computed by comparing these groups against a single healthy control group.
</p>
</li>
<li><p> Studies 6, 12, 14, 15, 18, 19, 22, and 26 had the patients and controls complete different tasks of varying complexity (essentially the average number of moves required to complete a task). Study 6 also included two different task types.
</p>
</li>
<li><p> Study 24 provides two standardized mean differences, one for men and the other for women.
</p>
</li>
<li><p> Study 29 provides three standardized mean differences, corresponding to the three different COMT Val158Met genotypes (val/val, val/met, and met/met).
</p>
</li></ol>

<p>All 4 issues described above lead to a multilevel structure in the dataset, with multiple standardized mean differences nested within some of the studies. Issues 1. and 2. also lead to correlated sampling errors.
</p>


<h3>Concepts</h3>

<p>psychology, standardized mean differences, multilevel models, multivariate models, cluster-robust inference, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Knapp, F., Viechtbauer, W., Leonhart, R., Nitschke, K., &amp; Kaller, C. P. (2017). Planning performance in schizophrenia patients: A meta-analysis of the influence of task difficulty and clinical and sociodemographic variables. <em>Psychological Medicine</em>, <b>47</b>(11), 2002&ndash;2016. <code style="white-space: pre;">&#8288;https://doi.org/10.1017/S0033291717000459&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.knapp2017
dat[-c(1:2)]

## Not run: 

### load metafor package
library(metafor)

### fit a standard random-effects model ignoring the issues described above
res &lt;- rma(yi, vi, data=dat)
res

### fit a multilevel model with random effects for studies and comparisons within studies
### (but this ignored the correlation in the sampling errors)
res &lt;- rma.mv(yi, vi, random = ~ 1 | study/comp, data=dat)
res

### create variable that indicates the task and difficulty combination as increasing integers
dat$task.diff &lt;- unlist(lapply(split(dat, dat$study), function(x) {
   task.int &lt;- as.integer(factor(x$task))
   diff.int &lt;- as.integer(factor(x$difficulty))
   diff.int[is.na(diff.int)] &lt;- 1
   paste0(task.int, ".", diff.int)}))

### construct correlation matrix for two tasks with four different difficulties where the
### correlation is 0.4 for different difficulties of the same task, 0.7 for the same
### difficulty of different tasks, and 0.28 for different difficulties of different tasks
R &lt;- matrix(0.4, nrow=8, ncol=8)
R[5:8,1:4] &lt;- R[1:4,5:8] &lt;- 0.28
diag(R[1:4,5:8]) &lt;- 0.7
diag(R[5:8,1:4]) &lt;- 0.7
diag(R) &lt;- 1
rownames(R) &lt;- colnames(R) &lt;- paste0(rep(1:2, each=4), ".", 1:4)
R

### construct an approximate V matrix accounting for the use of shared groups and
### for correlations among tasks/difficulties as specified in the R matrix above
V &lt;- vcalc(vi, cluster=study, grp1=group1, grp2=group2, w1=n_sz, w2=n_hc,
           obs=task.diff, rho=R, data=dat)

### correlation matrix for study 3 with four patient groups and a single control group
round(cov2cor(V[dat$study == 3, dat$study == 3]), 2)

### correlation matrix for study 6 with two tasks with four difficulties
cov2cor(V[dat$study == 6, dat$study == 6])

### correlation matrix for study 24 with two independent groups
cov2cor(V[dat$study == 24, dat$study == 24])

### correlation matrix for study 29 with three independent groups
cov2cor(V[dat$study == 29, dat$study == 29])

### fit multilevel model as above, but now use this V matrix in the model
res &lt;- rma.mv(yi, V, random = ~ 1 | study/comp, data=dat)
res
predict(res, digits=2)

### use cluster-robust inference methods based on this model
robust(res, cluster=study)

### use methods from clubSandwich package
robust(res, cluster=study, clubSandwich=TRUE)

### examine if task difficulty is a potential moderator of the effect
res &lt;- rma.mv(yi, V, mods = ~ difficulty, random = ~ 1 | study/comp, data=dat)
res
sav &lt;- robust(res, cluster=study)
sav
sav &lt;- robust(res, cluster=study, clubSandwich=TRUE)
sav

### draw bubble plot
regplot(sav, xlab="Task Difficulty", ylab="Standardized Mean Difference", las=1, digits=1, bty="l")


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.konstantopoulos2011'>Studies on the Effects of Modified School Calendars on Student Achievement</h2><span id='topic+dat.konstantopoulos2011'></span>

<h3>Description</h3>

<p>Results from 56 studies on the effects of modified school calendars on student achievement.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.konstantopoulos2011
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>district</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> district id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>school</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> school id number (within district) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> year of the study
</td>
</tr>

</table>



<h3>Details</h3>

<p>Instead of following the more traditional school calendar with a long summer break (in addition to a short winter and spring break), some schools have switched to a modified school calendar comprising more frequent but shorter intermittent breaks (e.g., 9 weeks of school followed by 3 weeks off), while keeping the total number of days at school approximately the same. The effects of using such a modified calendar on student achievement have been examined in a number of studies and were meta-analyzed by Cooper et al. (2003).
</p>
<p>The dataset (taken from Konstantopoulos, 2011) contains the results from 56 studies, each comparing the level of academic achievement in a group of students following a modified school calendar with that of a group of students following a more traditional school calendar. The difference between the two groups was quantified in terms of a standardized mean difference (with positive values indicating a higher mean level of achievement in the group following the modified school calendar).
</p>
<p>The studies were conducted at various schools that were clustered within districts. The data therefore have a multilevel structure, with schools nested within districts. A multilevel meta-analysis of these data can be used to estimate and account for the amount of heterogeneity between districts and between schools within districts.
</p>


<h3>Concepts</h3>

<p>education, standardized mean differences, multilevel models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Konstantopoulos, S. (2011). Fixed effects and variance components estimation in three-level meta-analysis. <em>Research Synthesis Methods</em>, <b>2</b>(1), 61&ndash;76. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/jrsm.35&#8288;</code>
</p>


<h3>References</h3>

<p>Cooper, H., Valentine, J. C., Charlton, K., &amp; Melson, A. (2003). The effects of modified school calendars on student achievement and on school and community attitudes. <em>Review of Educational Research</em>, <b>73</b>(1), 1&ndash;52. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/00346543073001001&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.konstantopoulos2011
dat

## Not run: 

### load metafor package
library(metafor)

### regular random-effects model
res &lt;- rma(yi, vi, data=dat)
print(res, digits=3)

### regular random-effects model using rma.mv()
res &lt;- rma.mv(yi, vi, random = ~ 1 | study, data=dat)
print(res, digits=3)

### multilevel random-effects model
res.ml &lt;- rma.mv(yi, vi, random = ~ 1 | district/school, data=dat)
print(res.ml, digits=3)

### profile variance components
profile(res.ml, progbar=FALSE)

### multivariate parameterization of the model
res.mv &lt;- rma.mv(yi, vi, random = ~ school | district, data=dat)
print(res.mv, digits=3)

### tau^2 from multivariate model = sum of the two variance components from the multilevel model
round(sum(res.ml$sigma2), 3)

### rho from multivariate model = intraclass correlation coefficient based on the multilevel model
round(res.ml$sigma2[1] / sum(res.ml$sigma2), 3)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.landenberger2005'>Studies on the Effectiveness of CBT for Reducing Recidivism</h2><span id='topic+dat.landenberger2005'></span>

<h3>Description</h3>

<p>Results from 58 studies on the effectiveness of cognitive-behavioral therapy (CBT) for reducing recidivism in juvenile and adult offenders.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.landenberger2005
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author and year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pubtype</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> publication type (book chapter, journal article, report, or thesis) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> country where study was carried out (Canada, New Zealand, UK, or USA) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>design</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study design (matched groups, nonequivalent groups, or randomized trial) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>program</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> purpose of setting up the CBT program (for demonstration, practice, or research purposes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>setting</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment setting (community or prison) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>designprob</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> indication of study design problems (no, favors the control group, or favors the treatment group) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.ctrl.rec</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of recidivists in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.ctrl.non</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of non-recidivists in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.cbt.rec</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of recidivists in the CBT group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n.cbt.non</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of non-recidivists in the CBT group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>interval</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> recidivism interval (in months) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>group</b>            </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study group (adults or juveniles) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>age</b>              </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean age of the study group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>male</b>             </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> percentage of males in the study group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>minority</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> percentage of minorities in the study group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>length</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment length (in weeks) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sessions</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of CBT sessions per week </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hrs_week</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment hours per week </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>hrs_total</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total hours of treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.cogskills</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: cognitive skills (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.cogrestruct</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: cognitive restructuring (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.intpprbsolv</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: interpersonal problem solving (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.socskills</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: social skills (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.angerctrl</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: anger control (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.victimimpact</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: victim impact (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.subabuse</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: substance abuse (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.behavmod</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: behavior modification (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.relapseprev</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: relapse prevention (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.moralrsng</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: moral reasoning (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.roletaking</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: role taking (yes, no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cbt.other</b>        </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> CBT component: other (yes, no)
</td>
</tr>

</table>



<h3>Details</h3>

<p>Landenberger and Lipsey (2005) conducted a meta-analysis of 58 experimental and quasi-experimental studies of the effects of cognitive-behavioral therapy (CBT) on the recidivism rates of adult and juvenile offenders (see also Lipsey et al., 2007). The present dataset includes the results of these studies and a range of potential moderator variables to identify factors associated with variation in treatment effects.
</p>


<h3>Concepts</h3>

<p>psychology, criminology, odds ratios, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Personal communication.
</p>


<h3>References</h3>

<p>Landenberger, N. A., &amp; Lipsey, M. W. (2005). The positive effects of cognitive-behavioral programs for offenders: A meta-analysis of factors associated with effective treatment. <em>Journal of Experimental Criminology</em>, <b>1</b>, 451&ndash;476. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s11292-005-3541-7&#8288;</code>
</p>
<p>Lipsey, M. W., Landenberger, N. A., &amp; Wilson, S. J. (2007). Effects of cognitive-behavioral programs for criminal offenders. <em>Campbell Systematic Reviews</em>, <b>3</b>(1), 1&ndash;27. <code style="white-space: pre;">&#8288;https://doi.org/10.4073/csr.2007.6&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.landenberger2005
head(dat)

## Not run: 

### load metafor package
library(metafor)

### calculate log odds ratios (for non-recidivism in CBT vs. control groups) and sampling variances
dat &lt;- escalc(measure="OR", ai=n.cbt.non, bi=n.cbt.rec, ci=n.ctrl.non, di=n.ctrl.rec, data=dat)

### fit random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### estimated average OR and corresponding 95% CI/PI
predict(res, transf=exp, digits=2)

### examine if number of treatment sessions per week is a potential moderator
res &lt;- rma(yi, vi, mods = ~ sessions, data=dat)
res

### predicted ORs for 1, 2, 5, or 10 sessions per week
predict(res, newmods=c(1,2,5,10), transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.laopaiboon2015'>Studies on the Effectiveness of Azithromycin for Treating Lower Respiratory Tract Infections</h2><span id='topic+dat.laopaiboon2015'></span>

<h3>Description</h3>

<p>Results from 15 studies on the effectiveness of azithromycin versus amoxycillin or amoxycillin/clavulanic acid (amoxyclav) in the treatment of acute lower respiratory tract infections.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.laopaiboon2015
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>author</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of clinical failures in the group treated with azithromycin </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the group treated with azithromycin </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of clinical failures in the group treated with amoxycillin or amoxyclav </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the group treated with amoxycillin or amoxyclav </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>age</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the trial included adults or children </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diag.ab</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial included patients with a diagnosis of acute bacterial bronchitis </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diag.cb</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial included patients with a diagnosis of chronic bronchitis with acute exacerbation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diag.pn</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial included patients with a diagnosis of pneumonia </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ctrl</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> antibiotic in control group (amoxycillin or amoxyclav)
</td>
</tr>

</table>



<h3>Details</h3>

<p>Azithromycin is an antibiotic useful for the treatment of a number of bacterial infections. Laopaiboon et al. (2015) conducted a meta-analysis of trials comparing the effectiveness of azithromycin versus amoxycillin or amoxycillin/clavulanic acid (amoxyclav) in the treatment of acute lower respiratory tract infections, including acute bacterial bronchitis, acute exacerbations of chronic bronchitis, and pneumonia. The results from 15 trials are included in this dataset.
</p>


<h3>Concepts</h3>

<p>medicine, risk ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Laopaiboon, M., Panpanich, R., &amp; Swa Mya, K. (2015). Azithromycin for acute lower respiratory tract infections. <em>Cochrane Database of Systematic Reviews</em>, <b>3</b>, CD001954. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD001954.pub4&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.laopaiboon2015
dat

## Not run: 

### load metafor package
library(metafor)

### analysis using the Mantel-Haenszel method
rma.mh(measure="RR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, digits=3)

### calculate log risk ratios and corresponding sampling variances
dat &lt;- escalc(measure="RR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat)

### random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### average risk ratio with 95% CI
predict(res, transf=exp)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.lau1992'>Studies on Intravenous Streptokinase for Acute Myocardial Infarction</h2><span id='topic+dat.lau1992'></span>

<h3>Description</h3>

<p>Results from 33 trials comparing intravenous streptokinase versus placebo or no therapy in patients who had been hospitalized for acute myocardial infarction.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.lau1992
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> trial name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the streptokinase group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the streptokinase group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>In the paper by Lau et al. (1992), the data are used to illustrate the idea of a cumulative meta-analysis, where the results are updated as each trial is added to the dataset. See &lsquo;Examples&rsquo; for code that replicates the results and shows corresponding forest plots.
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, odds ratios, cumulative meta-analysis
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Lau, J., Antman, E. M., Jimenez-Silva, J., Kupelnick, B., Mosteller, F., &amp; Chalmers, T. C. (1992). Cumulative meta-analysis of therapeutic trials for myocardial infarction. <em>New England Journal of Medicine</em>, <b>327</b>(4), 248&ndash;254. <code style="white-space: pre;">&#8288;https://doi.org/10.1056/NEJM199207233270406&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.lau1992
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of log odds ratios using the MH method
res &lt;- rma.mh(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, slab=trial)
print(res, digits=2)

### forest plot
forest(res, xlim=c(-10,9), atransf=exp, at=log(c(.01, 0.1, 1, 10, 100)),
       header=TRUE, top=2, ilab=dat$year, ilab.xpos=-6)
text(-6, 35, "Year", font=2)

### cumulative meta-analysis
sav &lt;- cumul(res)

### forest plot of the cumulative results
forest(sav, xlim=c(-5,4), atransf=exp, at=log(c(0.1, 0.5, 1, 2, 10)),
       header=TRUE, top=2, ilab=dat$year, ilab.xpos=-3)
text(-3, 35, "Year", font=2)
id &lt;- c(4, 8, 15, 33) # rows for which the z/p-values should be shown (as in Lau et al., 1992)
text(1.1, (res$k:1)[id], paste0("z = ", formatC(sav$zval[id], format="f", digits=2),
                                ", p = ", formatC(sav$pval[id], format="f", digits=4)))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.lee2004'>Studies on Acupoint P6 Stimulation for Preventing Nausea</h2><span id='topic+dat.lee2004'></span>

<h3>Description</h3>

<p>Results from studies examining the effectiveness of wrist acupuncture point P6 stimulation for preventing postoperative nausea.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.lee2004
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients experiencing nausea in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total number of patients in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients experiencing nausea in the sham group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total number of patients in the sham group
</td>
</tr>

</table>



<h3>Details</h3>

<p>Postoperative nausea and vomiting are common complications following surgery and anaesthesia. As an alternative to drug therapy, acupuncture has been studied as a potential treatment in several trials. The dataset contains the results from 16 clinical trials examining the effectiveness of wrist acupuncture point P6 stimulation for preventing postoperative nausea.
</p>


<h3>Concepts</h3>

<p>medicine, alternative medicine, risk ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Lee, A., &amp; Done, M. L. (2004). Stimulation of the wrist acupuncture point P6 for preventing postoperative nausea and vomiting. <em>Cochrane Database of Systematic Reviews</em>, <b>3</b>, CD003281. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD003281.pub2&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.lee2004
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis based on log risk ratios
res &lt;- rma(measure="RR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat)
res
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.lehmann2018'>The Effect of Red on Perceived Attractiveness</h2><span id='topic+dat.lehmann2018'></span>

<h3>Description</h3>

<p>Results from studies in which participants rated the attractiveness of photos that featured red or a control color. See OSF project at <code style="white-space: pre;">&#8288;https://osf.io/xy47p/&#8288;</code>.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.lehmann2018
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Short_Title</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Shortened citation formatted Author name(s), year of publication - Experiment number. All cells in the column are unique for use as labels in the meta-analysis. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Full_Citation</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Full citation in APA format. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Short_Citation</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Shortened citation of different format, exactly as it would appear in an in-text citation. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Year</b>                   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Year study published (whether in journal or published online). </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Study</b>                  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Experiment number. If only one experiment presented in a paper, then &lsquo;Exp 1&rsquo;, otherwise numbered according to numbering within paper. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Peer_Reviewed</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether the experiment was published in a peer-reviewed journal or not. &lsquo;Yes&rsquo; = peer-reviewed journal, &lsquo;No&rsquo; can mean in press, online publication, or other. Column for moderator analysis. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Source_Type</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Location where experiment is available, including journal articles, conference proceedings, online-only, and other options. More specific than whether peer-reviewed or not. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Preregistered</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether experiment was pre-registered or not.  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Moderator_Group</b>        </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> In some studies, a moderator was intentionally investigated that was meant to reduce the red-romance effect. Data for studies where the red-romance effect is expected to be moderated are marked &lsquo;Yes&rsquo; in this column. All others are blank. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Gender</b>                 </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Gender of rater (male or female). In all cases, gender of stimuli will be opposite. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Color_Contrast</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> The color used as the contrast against red. In some cases, not every contrast color was listed. We chose to examine only contrasts that were present in the original studies, when possible. This column contains only the contrasts we examined in this meta-analysis.  </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Color_Form</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Location of color in photo. Background = background or border color manipulated; Face = facial redness manipulated; Shirt, Dress, Item = color of specified object manipulated; Dot = a dot of color on shirt manipulated. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Photo_Type</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Amount of body visible in photo. Head Shot = head only; Bust = head, shoulders, sometimes torso; Full Body = entire body visible. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DV_Type</b>                </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Scale used for DV. &lsquo;Perceived attractiveness&rsquo; = the perceived attractiveness scale used in the original studies; alternate scales are differentiated. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DV_Items</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Number of items in DV scale. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DV_Scale</b>               </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Full length of DV scale, if clear. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DV_ScaleBottom</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Lower anchor of DV scale. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>DV_ScaleTop</b>            </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Upper anchor of DV scale. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Location</b>               </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Country where study took place, if clear. &lsquo;Worldwide&rsquo; in some cases of online participation without IP filtering of participants. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Continent</b>              </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Continent where study took place, for the sake of creating larger categories for analysis. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Participants</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Basic notes about participants. Students = high school, undergraduate, or graduate students; online = participants were gathered online; adult = no other common identifying factor given. Put into fewer categories for ease of analysis. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Participant_Notes</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> A finer grained description of participant characteristics. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Design</b>                 </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether study was a between- or within-subjects design. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Eth_Majority</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Basic notes about participant ethnicity for ease of analysis. This represents the ethnic majority within the sample. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Eth_Majority_Detail</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> A finer grained description of participant characteristics, including in some cases participant counts when the ethnic majority was close to another category. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Eth_Stim</b>               </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Ethnicity of the people pictured in the stimulus materials. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Eth_Match</b>              </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether the ethnic majority of the participant pool matched the ethnicity of stimulus photos. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Red_Age</b>                </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mean age of participants in red group. If not given for specific group, then mean age overall. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Control_Age</b>            </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mean age of participants in control group. If not given for specific group, then mean age overall. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Color_Red</b>              </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Specific values of red color, if given. &lsquo;No data&rsquo; if not given or unclear. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Color_Control</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Specific values of control color, if given. &lsquo;No data&rsquo; if not given or unclear. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Red_Original</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether the red color used in the study is within 5 units of the LCh values for red used in the original study. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Color_Match</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether the control color used in the study is within 5 units of the red color on the L and C parameters. In cases where the control color used was white, it was not possible for the L and C parameters to match. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Presentation_Control</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Whether the color of the stimulus viewed by each participant was consistent, as in participants viewing everything on paper or the same computer, versus uncontrolled presentation of the stimulus, as in viewing stimulus on different computers. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Stimuli_Presentation</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Method for presenting stimuli. &lsquo;Paper&rsquo; = stimuli printed on paper, shown in-person; &lsquo;Screen&rsquo; = stimuli shown on-screen, not carefully controlled; &lsquo;Screen Control&rsquo; = stimuli shown on-screen, but screen carefully color-matched. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Red_N</b>                  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Number of participants in red group. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Red_M</b>                  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mean rating of DV in red group. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Red_SD</b>                 </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Standard deviation of DV in red group. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Control_N</b>              </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Number of participants in control group. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Control_M</b>              </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mean rating of DV in control group. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Control_SD</b>             </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Standard deviation of DV in control group. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>SD_diff</b>                </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Calculated for within-subjects studies, standard deviation of difference scores. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>RM_r</b>                   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Calculated for within-subjects studies, correlation between participant ratings of red and control attractiveness. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Control_Attractiveness</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Attractiveness of stimuli in control condition, calculated as <code>(Control_M - DV_ScaleBottom) / DV_ScaleTop</code>, in order to compare attractiveness ratings across different scales. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Notes</b>                  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Any additional notes on the study. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Total.SampleSize</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Total unique participants in the study. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pooled</b>                 </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Pooled standard deviation for within-subjects studies. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>                     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Standardized mean difference. </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>                     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Corresponding sampling variance. </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p>This is data from a meta-analysis of studies that test the red-romance hypothesis, which is that the color red enhances heterosexual attraction in romantic contexts. Analyzing male participants only, the meta-analysis should show a small, statistically significant effect (d = 0.26 [0.12, 0.40], p = .0004, N = 2,961).  Analyzing female participants only should show a very small effect (d = 0.13 [0.01, 0.25], p = .03, N = 2,739).  The analyses in the published meta-analysis found clear evidence of upward bias in the estimate for female participants and equivocal evidence for male participants.  Moderator analyses suggest effect sizes may have declined over time (both genders), may be largest when an original shade of red is used (men only), and may be smaller in pre-registered studies (women only).
</p>


<h3>Concepts</h3>

<p>psychology, attraction, standardized mean differences
</p>


<h3>Author(s)</h3>

<p>Robert Calin-Jageman, <a href="mailto:rcalinjageman@dom.edu">rcalinjageman@dom.edu</a>, <a href="https://calin-jageman.net">https://calin-jageman.net</a>
</p>


<h3>Source</h3>

<p>Lehmann, G. K., Elliot, A. J., &amp; Calin-Jageman, R. J. (2018). Meta-analysis of the effect of red on perceived attractiveness. <em>Evolutionary Psychology</em>, <b>16</b>(4). <code style="white-space: pre;">&#8288;https://doi.org/10.1177/1474704918802412&#8288;</code> <code style="white-space: pre;">&#8288;https://osf.io/xy47p/&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.lehmann2018
head(dat)

## Not run: 

### load metafor package
library(metafor)

### meta-analyses for male and female participants
red_romance_malep   &lt;- dat[dat$Gender == "Males", ]
red_romance_femalep &lt;- dat[dat$Gender == "Females", ]

res_malep &lt;- rma(yi, vi, data=red_romance_malep, test="knha")
res_malep
res_femalep &lt;- rma(yi, vi, data=red_romance_femalep, test="knha")
res_femalep


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.li2007'>Studies on the Effectiveness of Intravenous Magnesium in Acute Myocardial Infarction</h2><span id='topic+dat.li2007'></span>

<h3>Description</h3>

<p>Results from 22 trials examining the effectiveness of intravenous magnesium in the prevention of death following acute myocardial infarction.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.li2007
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author or trial name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the magnesium group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the magnesium group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 22 randomized clinical trials that examined the effectiveness of intravenous magnesium in the prevention of death following acute myocardial infarction. It is similar to the dataset <code><a href="#topic+dat.egger2001">dat.egger2001</a></code>, with some slight differences in the included trials and data used.
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, odds ratios, publication bias
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Li, J., Zhang, Q., Zhang, M., &amp; Egger, M. (2007). Intravenous magnesium for acute myocardial infarction. <em>Cochrane Database of Systematic Reviews</em>, <b>2</b>, CD002755. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD002755.pub2&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+dat.egger2001">dat.egger2001</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.li2007
dat

## Not run: 

### load metafor package
library(metafor)

### meta-analysis of all trials except ISIS-4
res &lt;- rma(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, method="EE", subset=-14)
print(res, digits=2)
predict(res, transf=exp, digits=2)

### meta-analysis of all trials including ISIS-4
res &lt;- rma(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, method="EE")
print(res, digits=2)
predict(res, transf=exp, digits=2)

### contour-enhanced funnel plot centered at 0
funnel(res, refline=0, level=c(90, 95, 99), shade=c("white", "gray", "darkgray"))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.lim2014'>Studies on the Association Between Maternal Size, Offspring Size, and Number of Offsprings</h2><span id='topic+dat.lim2014'></span>

<h3>Description</h3>

<p>Results from studies examining the association between maternal size, offspring size, and number of offsprings.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.lim2014
</code></pre>


<h3>Format</h3>

<p>The object is a list containing data frames <code>m_o_size</code>, <code>m_o_fecundity</code>, <code>o_o_unadj</code>, and <code>o_o_adj</code> that contain the following columns and the corresponding phylogenetic trees called <code>m_o_size_tree</code>, <code>m_o_fecundity_tree</code>, <code>o_o_unadj_tree</code>, and <code>o_o_adj_tree</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>article</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> article id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>species</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>amniotes</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the species was amniotic </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>environment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the species were wild or captive </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>reprounit</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the data were based on lifetime reproductive output or a single reproductive event (only in <code>m_o_size</code> and <code>m_o_fecundity</code>) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> correlation coefficient </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size
</td>
</tr>

</table>



<h3>Details</h3>

<p>The object <code>dat.lim2014</code> includes 4 datasets:
</p>

<table>
<tr>
 <td style="text-align: left;">
   <code>m_o_size</code>      </td><td style="text-align: left;"> on the correlation between maternal size and offspring size </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code>m_o_fecundity</code> </td><td style="text-align: left;"> on the correlation between maternal size and number of offsprings </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code>o_o_unadj</code>     </td><td style="text-align: left;"> on the correlation between offspring size and number of offsprings </td>
</tr>
<tr>
 <td style="text-align: left;">
   <code>o_o_adj</code>       </td><td style="text-align: left;"> on the correlation between offspring size and number of offsprings adjusted for maternal size
   </td>
</tr>

</table>

<p>Objects <code>m_o_size_tree</code>, <code>m_o_fecundity_tree</code>, <code>o_o_unadj_tree</code>, and <code>o_o_adj_tree</code> are the corresponding phylogenetic trees for the species included in each of these datasets.
</p>


<h3>Concepts</h3>

<p>ecology, evolution, correlation coefficients, multilevel models, phylogeny
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Lim, J. N., Senior, A. M., &amp; Nakagawa, S. (2014). Heterogeneity in individual quality and reproductive trade-offs within species. <em>Evolution</em>, <b>68</b>(8), 2306&ndash;2318. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/evo.12446&#8288;</code>
</p>


<h3>References</h3>

<p>Cinar, O., Nakagawa, S., &amp; Viechtbauer, W. (in press). Phylogenetic multilevel meta-analysis: A simulation study on the importance of modelling the phylogeny. <em>Methods in Ecology and Evolution</em>. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/2041-210X.13760&#8288;</code>
</p>
<p>Hadfield, J. D., &amp; Nakagawa, S. (2010). General quantitative genetic methods for comparative biology: Phylogenies, taxonomies and multi-trait models for continuous and categorical characters. <em>Journal of Evolutionary Biology</em>, <b>23</b>(3), 494&ndash;508. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/j.1420-9101.2009.01915.x&#8288;</code>
</p>
<p>Nakagawa, S., &amp; Santos, E. S. A. (2012). Methodological issues and advances in biological meta-analysis. <em>Evolutionary Ecology</em>, <b>26</b>(5), 1253&ndash;1274. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s10682-012-9555-5&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.lim2014$o_o_unadj
dat[1:14, -c(2:3)]

## Not run: 

### load metafor package
library(metafor)

### load ape package
library(ape, warn.conflicts=FALSE)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=ri, ni=ni, data=dat)

### copy tree to 'tree'
tree &lt;- dat.lim2014$o_o_unadj_tree

### compute branch lengths
tree &lt;- compute.brlen(tree)

### compute phylogenetic correlation matrix
A &lt;- vcv(tree, corr=TRUE)

### make copy of the species variable
dat$species.phy &lt;- dat$species

### create effect size id variable
dat$esid &lt;- 1:nrow(dat)

### fit multilevel phylogenetic meta-analytic model
res &lt;- rma.mv(yi, vi,
   random = list(~ 1 | article, ~ 1 | esid, ~ 1 | species, ~ 1 | species.phy),
   R=list(species.phy=A), data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.linde2005'>Studies on the Effectiveness of St. John's Wort for Treating Depression</h2><span id='topic+dat.linde2005'></span>

<h3>Description</h3>

<p>Results from 26 studies on the effectiveness of Hypericum perforatum extracts (St. John's wort) for treating depression.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.linde2005
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>country</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study location </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>major</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample restricted to patients who met criteria for major depression </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>baseline</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> HRSD baseline score </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>version</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> HRSD version (17 or 21 items) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>duration</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study duration (in weeks) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>prep</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> Hypericum extract preparation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dosage</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> dosage (in mg) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>response</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> definition of response (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of responses in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of responses in placebo group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in placebo group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>group</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> stratification variable used by the authors (see &lsquo;Details&rsquo;)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 26 double-blind placebo-controlled trials on the effectiveness of Hypericum perforatum extracts (St. John's wort) for treating depression (note that 2 studies did not provide sufficient response information).
</p>
<p>Data were extracted from Table 1 and Figure 3 from Linde et al. (2005). For study duration, the assessment week (instead of the total study duration) was coded for Philipp et al. (1999) and Montgomery et al. (2000). For dosage, the midpoint was coded when a range of values was given.
</p>
<p>The definition of what constitutes a <code>response</code> differed across studies and is coded as follows:
</p>

<ol>
<li><p> HRSD score reduction of at least 50% or HRSD score after therapy &lt;10,
</p>
</li>
<li><p> HRSD reduction of at least 50%,
</p>
</li>
<li><p> based on HRSD scale but exact definition not reported,
</p>
</li>
<li><p> global patient assessment of efficacy,
</p>
</li>
<li><p> at least &lsquo;much improved&rsquo; on the Clinical Global Impression sub-scale for global improvement.
</p>
</li></ol>

<p>The <code>group</code> variable corresponds to the variable used by Linde et al. (2005) to stratify their analyses and is coded as follows:
</p>

<ol>
<li><p> smaller trials restricted to major depression,
</p>
</li>
<li><p> larger trials restricted to major depression,
</p>
</li>
<li><p> smaller trials not restricted to major depression,
</p>
</li>
<li><p> larger trials not restricted to major depression.
</p>
</li></ol>



<h3>Concepts</h3>

<p>medicine, psychiatry, risk ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Linde, K., Berner, M., Egger, M., &amp; Mulrow, C. (2005). St John's wort for depression: Meta-analysis of randomised controlled trials. <em>British Journal of Psychiatry</em>, <b>186</b>(2), 99&ndash;107. <code style="white-space: pre;">&#8288;https://doi.org/10.1192/bjp.186.2.99&#8288;</code>
</p>


<h3>References</h3>

<p>Viechtbauer, W. (2007). Accounting for heterogeneity via random-effects models and moderator analyses in meta-analysis. <em>Zeitschrift für Psychologie / Journal of Psychology</em>, <b>215</b>(2), 104&ndash;121. <code style="white-space: pre;">&#8288;https://doi.org/10.1027/0044-3409.215.2.104&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.linde2005
head(dat)

## Not run: 

### load metafor package
library(metafor)

### remove studies with no response information and study with no responses in either group
dat &lt;- dat[-c(5,6,26),]

### calculate log risk ratios and corresponding sampling variances
dat &lt;- escalc(measure="RR", ai=ai, ci=ci, n1i=n1i, n2i=n2i, data=dat)
head(dat)

### meta-analysis of the log risk ratios using a random-effects model
res &lt;- rma(yi, vi, data=dat, method="DL")
res

### mixed-effects meta-regression model with stratification variable
res &lt;- rma(yi, vi, mods = ~ factor(group) - 1, data=dat, method="DL")
res

### predicted average risk ratio for each level of the stratification variable
predict(res, newmods=diag(4), transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.linde2015'>Studies on Classes of Antidepressants for the Primary Care Setting</h2><span id='topic+dat.linde2015'></span>

<h3>Description</h3>

<p>Results from 66 trials examining eight classes of antidepressants and placebo for the primary care setting.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.linde2015
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> study ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> year of publication </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment1</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment2</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment3</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients (arm 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>resp1</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of early responder (arm 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>remi1</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of early remissions (arm 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>loss1</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients loss to follow-up (arm 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>loss.ae1</b>   </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients loss to follow-up due to adverse events (arm 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ae1</b>        </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients with adverse events (arm 1) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients (arm 2) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>resp2</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of early responder (arm 2) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>remi2</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of early remissions (arm 2) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>loss2</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients loss to follow-up (arm 2) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>loss.ae2</b>   </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients loss to follow-up due to adverse events (arm 2) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ae2</b>        </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients with adverse events (arm 2) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n3</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients (arm 3) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>resp3</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of early responder (arm 3) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>remi3</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of early remissions (arm 3) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>loss3</b>      </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients loss to follow-up (arm 3) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>loss.ae3</b>   </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients loss to follow-up due to adverse events (arm 3) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ae3</b>        </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients with adverse events (arm 3)
</td>
</tr>

</table>



<h3>Details</h3>

<p>This data set comes from a systematic review of 8 pharmacological treatments of depression and placebo in primary care with 66 studies (8 of which were 3-arm studies) including 14,785 patients.
</p>
<p>The primary outcome is early response, defined as at least a 50% score reduction on a depression scale after completion of treatment. Secondary outcomes (also measured as dichotomous) were early remission (defined as having a symptom score below a fixed threshold after completion of treatment), lost to follow-up, lost to follow-up due to adverse events, and any adverse event. The odds ratio was used as effect measure.
</p>
<p>This data set was used as an example in Rücker and Schwarzer (2017) who introduced methods to resolve conflicting rankings of outcomes in network meta-analysis.
</p>


<h3>Concepts</h3>

<p>medicine, psychiatry, odds ratios, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Linde, K., Kriston, L., Rücker, G., Jamil, S., Schumann, I., Meissner, K., Sigterman, K., &amp; Schneider, A. (2015). Efficacy and acceptability of pharmacological treatments for depressive disorders in primary care: Systematic review and network meta-analysis. <em>Annals of Family Medicine</em>, <b>13</b>(1), 69&ndash;79. <code style="white-space: pre;">&#8288;https://doi.org/10.1370/afm.1687&#8288;</code>
</p>


<h3>References</h3>

<p>Rücker, G., &amp; Schwarzer, G. (2017). Resolve conflicting rankings of outcomes in network meta-analysis: Partial ordering of treatments. <em>Research Synthesis Methods</em>, <b>8</b>(4), 526&ndash;536. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/jrsm.1270&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="netmeta.html#topic+pairwise">pairwise</a></code>, <code><a href="meta.html#topic+metabin">metabin</a></code>, <code><a href="netmeta.html#topic+netmeta">netmeta</a></code>, <code><a href="netmeta.html#topic+netrank">netrank</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show results from first three studies (including three-arm study
### Lecrubier 1997)
head(dat.linde2015, 3)

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Change appearance of confidence intervals
cilayout("(", "-")

### Define order of treatments in printouts
trts &lt;- c("TCA", "SSRI", "SNRI", "NRI", "Low-dose SARI",
 "NaSSa", "rMAO-A", "Hypericum", "Placebo")

### Transform data from wide arm-based format to contrast-based format
### (outcome: early response). Argument 'sm' has to be used for odds
### ratio as summary measure; by default the risk ratio is used in the
### metabin function called internally.
pw1 &lt;- pairwise(list(treatment1, treatment2, treatment3),
  event = list(resp1, resp2, resp3),
  n = list(n1, n2, n3),
  studlab = id, data = dat.linde2015, sm = "OR")

### Conduct random effects network meta-analysis for primary outcome
### (early response); small number of early responses is bad (argument
### small.values)
net1 &lt;- netmeta(pw1, fixed = FALSE, reference = "Placebo", seq = trts,
  small.values = "bad")
net1

### Random effects NMA for early remission
pw2 &lt;- pairwise(treat = list(treatment1, treatment2, treatment3),
  event = list(remi1, remi2, remi3),
  n = list(n1, n2, n3),
  studlab = id, data = dat.linde2015, sm = "OR")
net2 &lt;- netmeta(pw2, fixed = FALSE,
   seq = trts, ref = "Placebo", small.values = "bad")
net2

### Ranking of treatments
nr1 &lt;- netrank(net1)
nr2 &lt;- netrank(net2)
nr1
nr2

### Partial order of treatment rankings (two outcomes)
outcomes &lt;- c("Early response", "Early remission")
po12 &lt;- netposet(nr1, nr2, outcomes = outcomes)
plot(po12)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.linde2016'>Studies on Antidepressants for the Primary Care Setting</h2><span id='topic+dat.linde2016'></span>

<h3>Description</h3>

<p>Results from 93 trials examining 22 interventions (including placebo and usual care) for the primary care of depression.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.linde2016
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>     </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> study ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>lnOR</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> response after treatment (log odds ratio) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>selnOR</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard error of log odds ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treat1</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treat2</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> second treatment
</td>
</tr>

</table>



<h3>Details</h3>

<p>This data set comes from a network meta-analysis of 22 treatments of depression in primary care (Linde et al., 2016), based on 93 trials (79 two-arm trials, 13 three-arm trials, and one four-arm trial). The primary outcome was response after treatment (yes/no), defined as a reduction from baseline by at least 50% on a depression scale. The data set contains log odds ratios with standard errors for all pairwise comparisons.
</p>
<p>The interventions comprised both medical and psychological treatments, also in combination, including placebo and usual care (UC) (Linde et al., 2016). Pharmacological interventions were tricyclic antidepressants (TCA), selective serotonin reuptake inhibitors (SSRI), serotonin-noradrenaline reuptake inhibitors (SNRI), noradrenaline reuptake inhibitors (NRI), low- dose serotonin (5-HT2) antagonists and reuptake inhibitors (low-dose SARI), noradrenergic and specific serotonergic agents (NaSSa), reversible inhibitors of monoaminoxidase A (rMAO-A), hypericum extracts, and an individualized drug. Psychological interventions were cognitive behavioral therapy (CBT; four forms: face-to-face CBT, remote therapist-led CBT, guided self-help CBT, and no or minimal contact CBT), face-to-face problem-solving therapy (PST), face-to-face interpersonal psychotherapy, face-to-face psychodynamic therapy, and &ldquo;other face-to-face therapy&rdquo;. Combination therapies were face-to-face CBT + SSRI, face-to-face PST + SSRI, and face-to-face interpersonal psychotherapy + SSRI.
</p>
<p>The data set was used as an example in Rücker et al. (2020) to illustrate component network meta-analysis using frequentist methods.
</p>


<h3>Concepts</h3>

<p>medicine, psychiatry, odds ratios, network meta-analysis, component network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Linde, K., Rücker, G., Schneider, A., &amp; Kriston, L. (2016). Questionable assumptions hampered interpretation of a network meta-analysis of primary care depression treatments. <em>Journal of Clinical Epidemiology</em>, <b>71</b>, 86&ndash;96. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.jclinepi.2015.10.010&#8288;</code>
</p>


<h3>References</h3>

<p>Rücker, G., Petropoulou, M., &amp; Schwarzer, G. (2020). Network meta-analysis of multicomponent interventions. <em>Biometrical Journal</em>, <b>62</b>(3), 808&ndash;821. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/bimj.201800167&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="netmeta.html#topic+netmeta">netmeta</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show results of first three studies (first study has three treatment
### arms)
head(dat.linde2016, 5)

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Define order of treatments in printouts and forest plots
trts &lt;- c("SSRI",
  "Face-to-face CBT", "Face-to-face interpsy", "Face-to-face PST",
  "Face-to-face CBT + SSRI", "Face-to-face interpsy + SSRI",
  "Face-to-face PST + SSRI",
  "Face-to-face psychodyn", "Other face-to-face",
  "TCA", "SNRI", "NRI", "Low-dose SARI", "NaSSa", "rMAO-A", "Ind drug",
  "Hypericum",
  "Remote CBT", "Self-help CBT", "No contact CBT",
  "UC", "Placebo")

### Conduct random effects network meta-analysis
net &lt;- netmeta(lnOR, selnOR, treat1, treat2, id,
  data = dat.linde2016, reference.group = "placebo",
  seq = trts, sm = "OR", fixed = FALSE)

### Network graph
netgraph(net, seq = "o", number = TRUE)

### Show results
net
forest(net, xlim = c(0.2, 50))

### Additive component network meta-analysis with placebo as inactive
### treatment
nc &lt;- netcomb(net, inactive = "placebo")
nc
forest(nc, xlim = c(0.2, 50))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.lopez2019'>Studies on the Effectiveness of CBT for Depression</h2><span id='topic+dat.lopez2019'></span>

<h3>Description</h3>

<p>Results from 76 studies examining the effectiveness of cognitive behavioral therapy (CBT) for depression in adults.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.lopez2019
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author and year of study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment provided (see &lsquo;Details&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>scale</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> scale used to measure depression symptoms </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> group size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>diff</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standardized mean change </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>se</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding standard error </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>group</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> type of therapy (0 = individual, 1 = group therapy) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tailored</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> whether the intervention was tailored to each patient (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sessions</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of sessions </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>length</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> average session length (in minutes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>intensity</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> product of sessions and length </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>multi</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included multimedia elements (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cog</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included cognitive techniques (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ba</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included behavioral activation (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>psed</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included psychoeducation (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>home</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included homework (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>prob</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included problem solving (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>soc</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included social skills training (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>relax</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included relaxation (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>goal</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included goal setting (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>final</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included a final session (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mind</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included mindfulness (0 = no, 1 = yes) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>act</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> intervention included acceptance and commitment therapy (0 = no, 1 = yes)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 76 studies examining the effectiveness of cognitive behavioral therapy (CBT) for treating depression in adults. Studies included two or more of the following treatments/conditions:
</p>

<ol>
<li><p> treatment as usual (TAU),
</p>
</li>
<li><p> no treatment,
</p>
</li>
<li><p> wait list,
</p>
</li>
<li><p> psychological or attention placebo,
</p>
</li>
<li><p> face-to-face CBT,
</p>
</li>
<li><p> multimedia CBT,
</p>
</li>
<li><p> hybrid CBT (i.e., multimedia CBT with one or more face-to-face sessions).
</p>
</li></ol>

<p>Multimedia CBT was defined as CBT delivered via self-help books, audio/video recordings, telephone, computer programs, apps, e-mail, or text messages.
</p>
<p>Variable <code>diff</code> is the standardized mean change within each group, with negative values indicating a decrease in depression symptoms.
</p>


<h3>Concepts</h3>

<p>psychiatry, standardized mean changes, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Personal communication.
</p>


<h3>References</h3>

<p>López-López, J. A., Davies, S. R., Caldwell, D. M., Churchill, R., Peters, T. J., Tallon, D., Dawson, S., Wu, Q., Li, J., Taylor, A., Lewis, G., Kessler, D. S., Wiles, N., &amp; Welton, N. J. (2019). The process and delivery of CBT for depression in adults: A systematic review and network meta-analysis. <em>Psychological Medicine</em>, <b>49</b>(12), 1937&ndash;1947. <code style="white-space: pre;">&#8288;https://doi.org/10.1017/S003329171900120X&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.lopez2019
dat[1:10,1:6]

## Not run: 

### load metafor package
library(metafor)

### create network graph ('igraph' package must be installed)
library(igraph, warn.conflicts=FALSE)
pairs &lt;- data.frame(do.call(rbind,
   sapply(split(dat$treatment, dat$study), function(x) t(combn(x,2)))), stringsAsFactors=FALSE)
pairs$X1 &lt;- factor(pairs$X1, levels=sort(unique(dat$treatment)))
pairs$X2 &lt;- factor(pairs$X2, levels=sort(unique(dat$treatment)))
tab &lt;- table(pairs[,1], pairs[,2])
tab # adjacency matrix
g &lt;- graph_from_adjacency_matrix(tab, mode = "plus", weighted=TRUE, diag=FALSE)
plot(g, edge.curved=FALSE, edge.width=E(g)$weight/2,
     layout=layout_in_circle(g, order=c("Wait list", "No treatment", "TAU", "Multimedia CBT",
                                        "Hybrid CBT", "F2F CBT", "Placebo")),
     vertex.size=45, vertex.color="lightgray", vertex.label.color="black", vertex.label.font=2)

### restructure data into wide format
dat &lt;- to.wide(dat, study="study", grp="treatment", ref="TAU",
               grpvars=c("diff","se","n"), postfix=c("1","2"))

### compute contrasts between treatment pairs and corresponding sampling variances
dat$yi &lt;- with(dat, diff1 - diff2)
dat$vi &lt;- with(dat, se1^2 + se2^2)

### calculate the variance-covariance matrix for multitreatment studies
calc.v &lt;- function(x) {
   v &lt;- matrix(x$se2[1]^2, nrow=nrow(x), ncol=nrow(x))
   diag(v) &lt;- x$vi
   v
}
V &lt;- bldiag(lapply(split(dat, dat$study), calc.v))

### add contrast matrix to the dataset
dat &lt;- contrmat(dat, grp1="treatment1", grp2="treatment2")

### network meta-analysis using a contrast-based random-effects model
### by setting rho=1/2, tau^2 reflects the amount of heterogeneity for all treatment comparisons
### the treatment left out (TAU) becomes the reference level for the treatment comparisons
res &lt;- rma.mv(yi, V, data=dat,
         mods = ~ No.treatment + Wait.list + Placebo + F2F.CBT + Hybrid.CBT + Multimedia.CBT - 1,
         random = ~ comp | study, rho=1/2)
res

### forest plot of the contrast estimates (treatments versus TAU)
forest(coef(res), diag(vcov(res)), slab=sub(".", " ", names(coef(res)), fixed=TRUE),
       xlim=c(-5,5), alim=c(-3,3), psize=1, header="Treatment",
       xlab="Difference in Standardized Mean Change (compared to TAU)")

### fit random inconsistency effects model
res &lt;- rma.mv(yi, V, data=dat,
         mods = ~ No.treatment + Wait.list + Placebo + F2F.CBT + Hybrid.CBT + Multimedia.CBT - 1,
         random = list(~ comp | study, ~ comp | design), rho=1/2, phi=1/2)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.maire2019'>Studies on Temporal Trends in Fish Community Structures in French Rivers</h2><span id='topic+dat.maire2019'></span>

<h3>Description</h3>

<p>Results from studies examining changes in the abundance of fish species in French rivers.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.maire2019
</code></pre>


<h3>Format</h3>

<p>The object is a list containing a data frame called <code>dat</code> that contains the following columns and distance matrix called <code>dmat</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>site</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study site </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>station</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> sampling station at site </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>site_station</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> site and station combined </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>s1</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mann-Kendal trend statistic for relative abundance of non-local species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vars1</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance (corrected for temporal autocorrelation) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>s2</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mann-Kendal trend statistic for relative abundance of northern species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vars2</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance (corrected for temporal autocorrelation) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>s3</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Mann-Kendal trend statistic for relative abundance of non-native species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vars3</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance (corrected for temporal autocorrelation) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>const</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> constant value of 1
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 35 sampling stations (at 11 sites along various French rivers) examining the abundance of various fish species over time (i.e., over 19-37 years, all until 2015). The temporal trend in these abundance data was quantified in terms of Mann-Kendal trend statistics, with positive values indicating monotonically increasing trends. The corresponding sampling variances were corrected for the temporal autocorrelation in the data (Hamed &amp; Rao, 1998).
</p>
<p>The distance matrix <code>dmat</code> indicates the distance of the sampling stations (1-423 river-km). For stations not connected through the river network, a high distance value of 10,000 river-km was set (effectively forcing the spatial correlation to be 0 for such stations).
</p>
<p>The dataset can be used to illustrate a meta-analysis allowing for spatial correlation in the outcomes.
</p>


<h3>Concepts</h3>

<p>ecology, climate change, spatial correlation
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Maire, A., Thierry, E., Viechtbauer, W., &amp; Daufresne, M. (2019). Poleward shift in large-river fish communities detected with a novel meta-analysis framework. <em>Freshwater Biology</em>, <b>64</b>(6), 1143&ndash;1156. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/fwb.13291&#8288;</code>
</p>


<h3>References</h3>

<p>Hamed, K. H., &amp; Rao, A. R. (1998). A modified Mann-Kendall trend test for autocorrelated data. <em>Journal of Hydrology</em>, <b>204</b>(1-4), 182&ndash;196. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/S0022-1694(97)00125-X&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.maire2019$dat
dat[-10]

### copy distance matrix into 'dmat' and examine first 5 rows/columns
dmat &lt;- dat.maire2019$dmat
dmat[1:5,1:5]

## Not run: 

### load metafor package
library(metafor)

### fit a standard random-effects model ignoring spatial correlation
res1 &lt;- rma.mv(s1, vars1, random = ~ 1 | site_station, data=dat)
res1

### fit model allowing for spatial correlation
res2 &lt;- rma.mv(s1, vars1, random = ~ site_station | const, struct="SPGAU",
               data=dat, dist=list(dmat), control=list(rho.init=10))
res2

### add random effects for sites and stations within sites
res3 &lt;- rma.mv(s1, vars1, random = list(~ 1 | site/station, ~ site_station | const), struct="SPGAU",
               data=dat, dist=list(dmat), control=list(rho.init=10))
res3

### likelihood ratio tests comparing the models
anova(res1, res2)
anova(res2, res3)

### profile likelihood plots for model res2
profile(res2, cline=TRUE)

### effective range (river-km for which the spatial correlation is &gt;= .05)
sqrt(3) * res2$rho

### note: it was necessary to adjust the starting value for rho in models
### res2 and res3 so that the optimizer does not get stuck in a local maximum
profile(res2, rho=1, xlim=c(0,200), steps=100)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.mccurdy2020'>Studies on the Generation Effect</h2><span id='topic+dat.mccurdy2020'></span>

<h3>Description</h3>

<p>Results from 126 articles that examined the so-called &lsquo;generation effect&rsquo;.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.mccurdy2020
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>article</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> article identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>experiment</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> experiment (within article) identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sample</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample (within experiment) identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>id</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> row identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pairing</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> identifier to indicate paired conditions within experiments </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean recall rate for the condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of participants for the condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>stimuli</b>          </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of stimuli for the condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>condition</b>        </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> condition (&lsquo;read&rsquo; or &lsquo;generate&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>gen_difficulty</b>   </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> generation difficulty (&lsquo;low&rsquo; or &lsquo;high&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>manip_type</b>       </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> manipulation type of the generate versus read condition (using a &lsquo;within&rsquo; or &lsquo;between&rsquo; subjects design) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>present_style</b>    </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> presentation style (&lsquo;mixed&rsquo; or &lsquo;pure&rsquo; list presentation) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>word_status</b>      </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> word status (&lsquo;words&rsquo;, &lsquo;non-words&rsquo;, or &lsquo;numbers&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>memory_test</b>      </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> memory test (&lsquo;recognition&rsquo;, &lsquo;cued recall&rsquo;, or &lsquo;free recall&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>memory_type</b>      </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> memory type (&lsquo;item&rsquo;, &lsquo;source&rsquo;, &lsquo;font color&rsquo;, &lsquo;font type&rsquo;, &lsquo;order&rsquo;, &lsquo;cue word&rsquo;, &lsquo;background color&rsquo;, or &lsquo;location&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>gen_constraint</b>   </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> generation constraint (&lsquo;low&rsquo;, &lsquo;medium&rsquo;, or &lsquo;high&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>learning_type</b>    </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> learning type (&lsquo;incidental&rsquo; or &lsquo;intentional&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>stimuli_relation</b> </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> stimuli relation (&lsquo;semantic&rsquo;, &lsquo;category&rsquo;, &lsquo;antonym&rsquo;, &lsquo;synonym&rsquo;, &lsquo;rhyme&rsquo;, &lsquo;compound words&rsquo;, &lsquo;definitions&rsquo;, or &lsquo;unrelated&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>gen_mode</b>         </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> generation mode (&lsquo;verbal/speaking&rsquo;, &lsquo;covert/thinking&rsquo;, or &lsquo;writing/typing&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>gen_task</b>         </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> generation task (&lsquo;anagram&rsquo;, &lsquo;letter transposition&rsquo;, &lsquo;word fragment&rsquo;, &lsquo;sentence completion&rsquo;, &lsquo;word stem&rsquo;, &lsquo;calculation&rsquo;, or &lsquo;cue only&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>attention</b>        </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> attention (&lsquo;divided&rsquo; or &lsquo;full&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pacing</b>           </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> pacing (&lsquo;self-paced&rsquo; or &lsquo;timed&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>filler_task</b>      </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> filler task (&lsquo;yes&rsquo; or &lsquo;no&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>age_grp</b>          </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> age group (&lsquo;younger&rsquo; or &lsquo;older&rsquo; adults) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>retention_delay</b>  </td><td style="text-align: left;"> <code>factor</code>    </td><td style="text-align: left;"> retention delay (&lsquo;immediate&rsquo;, &lsquo;short&rsquo;, or &lsquo;long&rsquo;) </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p>The generation effect is the memory benefit for self-generated compared with read or experimenter-provided information (Jacoby, 1978; Slamecka &amp; Graf, 1978). In a typical study, participants are presented with a list of stimuli (usually words or word pairs). For half of the stimuli, participants self-generate a target word (e.g., open–cl____), while for the other half, participants simply read an intact target word (e.g., above–below). On a later memory test for the target words, the common finding is that self-generated words are better remembered than read words (i.e., the generation effect).
</p>
<p>Although several theories have been proposed to explain the generation effect, there is still some debate on the underlying memory mechanism(s) contributing to this phenomenon. The meta-analysis by McCurdy et al. (2020) translated various theories on the generation effect into hypotheses that could then be tested in moderator analyses based on a dataset containing 126 articles, 310 experiments, and 1653 mean recall estimates collected under various conditions.
</p>
<p>Detailed explanations of the various variables coded (and how these can be used to test various hypotheses regarding the generation effect) can be found in the article. The most important variable is <code>condition</code>, which denotes whether a particular row of the dataset corresponds to the results of a &lsquo;read&rsquo; or a &lsquo;generate&rsquo; condition.
</p>
<p>The data structure is quite complex. Articles may have reported the findings from multiple experiments involving one or multiple samples that were examined under various conditions. The <code>pairing</code> variable indicates which rows of the dataset represent a pairing of a read condition with one or multiple corresponding generate conditions within an experiment. A pairing may involve the same sample of subjects (when using a within-subjects design for comparing the conditions) or different samples (when using a between-subjects design).
</p>


<h3>Concepts</h3>

<p>psychology, memory, proportions, raw means, multilevel models, cluster-robust inference
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>McCurdy, M. P., Viechtbauer, W., Sklenar, A. M., Frankenstein, A. N., &amp; Leshikar, E. D. (2020). Theories of the generation effect and the impact of generation constraint: A meta-analytic review. <em>Psychonomic Bulletin &amp; Review</em>, <b>27</b>(6), 1139&ndash;1165. <code style="white-space: pre;">&#8288;https://doi.org/10.3758/s13423-020-01762-3&#8288;</code>
</p>


<h3>References</h3>

<p>Slamecka, N. J., &amp; Graf, P. (1978). The generation effect: Delineation of a phenomenon. <em>Journal of Experimental Psychology: Human Learning and Memory</em>, <b>4</b>(6), 592&ndash;604. <code style="white-space: pre;">&#8288;https://doi.org/10.1037/0278-7393.4.6.592&#8288;</code>
</p>
<p>Jacoby, L. L. (1978). On interpreting the effects of repetition: Solving a problem versus remembering a solution. <em>Journal of Verbal Learning and Verbal Behavior</em>, <b>17</b>(6), 649&ndash;668. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/S0022-5371(78)90393-6&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.mccurdy2020
head(dat)

## Not run: 

### load metafor package
library(metafor)

### fit multilevel mixed-effects meta-regression model
res &lt;- rma.mv(yi, vi, mods = ~ condition,
              random = list(~ 1 | article/experiment/sample/id, ~ 1 | pairing),
              data=dat, sparse=TRUE, digits=3)
res

### proportion of total amount of heterogeneity due to each component
data.frame(source=res$s.names, sigma2=round(res$sigma2, 3),
   prop=round(res$sigma2 / sum(res$sigma2), 2))

### apply cluster-robust inference
sav &lt;- robust(res, cluster=article)
sav

### estimated average recall rate in read and generate conditions
predict(sav, newmods = c(0,1), digits=3)

### use methods from clubSandwich package
sav &lt;- robust(res, cluster=article, clubSandwich=TRUE)
sav


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.mcdaniel1994'>Studies on the Validity of Employment Interviews</h2><span id='topic+dat.mcdaniel1994'></span>

<h3>Description</h3>

<p>Results from 160 studies on the correlation between employment interview assessments and job performance.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.mcdaniel1994
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size of the study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> observed correlation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>type</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> interview type (j = job-related, s = situational, p = psychological) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>struct</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> interview structure (u = unstructured, s = structured)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 160 studies provide data in terms of the correlation between employment interview performance and actual job performance. In addition, the interview type and the interview structure are indicated.
</p>
<p>McDaniel et al. (1994) describe the interview type and structure variables as follows. &quot;Questions in situational interviews [...] focus on the individual's ability to project what his or her behavior would be in a given situation. [...] Job-related interviews are those in which the interviewer is a personnel officer or hiring authority and the questions attempt to assess past behaviors and job-related information, but most questions are not considered situational. Psychological interviews are conducted by a psychologist, and the questions are intended to assess personal traits, such as dependability.&quot; In structured interviews, &quot;the questions and acceptable responses were specified in advance and the responses were rated for appropriateness of content. [...] Unstructured interviews gather applicant information in a less systematic manner than do structured interviews. Although the questions may be specified in advance, they usually are not, and there is seldom a formalized scoring guide. Also, all persons being interviewed are not typically asked the same questions.&quot;
</p>
<p>The goal of the meta-analysis was to examine the overall criterion-related validity of employment interviews and to examine whether the validity depends on the type and structure of the interview.
</p>
<p>The data in this dataset were obtained from Table A.2 in Rothstein, Sutton, and Borenstein (2005, p. 325-329). Note that the <code>type</code> and <code>struct</code> variables contain some <code>NA</code>s.
</p>


<h3>Concepts</h3>

<p>psychology, correlation coefficients, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Rothstein, H. R., Sutton, A. J., &amp; Borenstein, M. (Eds.). (2005). <em>Publication bias in meta-analysis: Prevention, assessment, and adjustments</em>. Chichester, England: Wiley.
</p>


<h3>References</h3>

<p>McDaniel, M. A., Whetzel, D. L., Schmidt, F. L., &amp; Maurer, S. D. (1994). The validity of employment interviews: A comprehensive review and meta-analysis. <em>Journal of Applied Psychology</em>, <b>79</b>(4), 599&ndash;616. <code style="white-space: pre;">&#8288;https://doi.org/10.1037/0021-9010.79.4.599&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.mcdaniel1994
head(dat)

## Not run: 

### load metafor package
library(metafor)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=ri, ni=ni, data=dat)
head(dat)

### meta-analysis of the transformed correlations using a random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### average correlation with 95% CI
predict(res, transf=transf.ztor)

### mixed-effects model with interview type as factor
### note: job-related interviews is the reference level
rma(yi, vi, mods = ~ factor(type), data=dat)

### mixed-effects model with interview structure as factor
### note: structured interviews is the reference level
rma(yi, vi, mods = ~ factor(struct), data=dat)

### note: the interpretation of the results is difficult since all
### situational interviews were structured, almost all psychological
### interviews were unstructured, and actually for the majority of
### the psychological interviews it was unknown whether the interview
### was structured or unstructured
table(dat$type, dat$struct, useNA="always")

### meta-analysis of raw correlations using a random-effects model
res &lt;- rma(measure="COR", ri=ri, ni=ni, data=dat.mcdaniel1994)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.michael2013'>The Non-Persuasive Power of a Brain Image</h2><span id='topic+dat.michael2013'></span>

<h3>Description</h3>

<p>Results from studies exploring how a superfluous fMRI brain image influences the persuasiveness of a scientific claim.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.michael2013
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>Study</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> name of the study: Citation - Experiment - Subgroup </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>No_brain_n</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size for no-brain-image condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>No_brain_m</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean agreement rating for no-brain-image condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>No_brain_s</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation for no-brain-image condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Brain_n</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size for brain-image condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Brain_m</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean agreement rating for brain-image condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Brain_s</b>           </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation for brain-image condition </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Included_Critique</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> &lsquo;Critique&rsquo; if article included critical commentary on conclusions, otherwise &lsquo;No_critique&rsquo; </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Medium</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> &lsquo;Paper&rsquo; if conducted in person; &lsquo;Online&rsquo; if conducted online </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Compensation</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> notes on compensation provided to participants </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Participant_Pool</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> notes on where participants were recruited </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>                </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> raw mean difference, calculated as <code>Brain_m - No_brain_m</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>                </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset contains the data from the meta-analysis by Michael et al. (2013) of experiments on the persuasive power of a brain image. The meta-analysis analyzed an original study by McCabe and Castel (2008) as well as 10 replication attempts conducted by the authors of the meta-analysis.
</p>
<p>In each study, participants read an article about using brain imaging as a lie detector. The article either included a superfluous fMRI image of a brain (brain) or not (no_brain). After reading the article, all participants responded to the statement &ldquo;Do you agree or disagree with the conclusion that brain imaging can be used as a lie detector?&rdquo; on a scale from 1 (strongly disagree) to 4 (strongly agree).
</p>
<p>The original study by McCabe and Castel (2008) reported a relatively large increase in agreement due to the presence of brain images. Meta-analysis of the original study with the 10 replications suggests, however, a small, possibly null effect: an estimated average raw mean difference of 0.07 points, 95% CI [-0.00, 0.14], under a random-effects model.
</p>
<p>In some studies, the article included a passage critiquing the primary claims made in the article; this is coded in the <code>Included_Critique</code> column for analysis as a possible moderator. Note that Experiment 3 by McCabe and Castel (2008) was a 2x2 between subjects design: brain image presence was manipulated as well as the inclusion of a critique. The two different critique conditions are recorded as separate rows in this dataset. Analysis of this dataset with metafor yields the same results (given rounding) reported in the manuscript.
</p>


<h3>Concepts</h3>

<p>psychology, persuasion, raw mean differences
</p>


<h3>Author(s)</h3>

<p>Robert Calin-Jageman, <a href="mailto:rcalinjageman@dom.edu">rcalinjageman@dom.edu</a>, <a href="https://calin-jageman.net">https://calin-jageman.net</a>
</p>


<h3>Source</h3>

<p>Michael, R. B., Newman, E. J., Vuorre, M., Cumming, G., &amp; Garry, M. (2013). On the (non)persuasive power of a brain image. <em>Psychonomic Bulletin &amp; Review</em>, <b>20</b>(4), 720–-725. <code style="white-space: pre;">&#8288;https://doi.org/10.3758/s13423-013-0391-6&#8288;</code>
</p>


<h3>References</h3>

<p>McCabe, D. P., &amp; Castel, A. D. (2008). Seeing is believing: The effect of brain images on judgments of scientific reasoning. <em>Cognition</em>, <b>107</b>(1), 343&ndash;352. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.cognition.2007.07.017&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.michael2013
dat

## Not run: 

### load metafor package
library(metafor)

### Data prep
# yi and vi are already provided, but here's how you would use escalc() to obtain
# a raw-mean difference and its variance.
# Note the measure parameter is "MD" for 'raw mean difference'
dat &lt;- metafor::escalc(
  measure = "MD",
  m1i = Brain_m,
  m2i = No_brain_m,
  sd1i = Brain_s,
  sd2i = No_brain_s,
  n1i = Brain_n,
  n2i = No_brain_n,
  data = dat
)

### meta-analysis using a random-effects model of the raw mean differences
res &lt;- rma(yi, vi, data=dat)
print(res, digits=2)

### examine if Included_Critique is a potential moderator
res &lt;- rma(yi, vi, mods = ~ Included_Critique, data=dat)
print(res, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.molloy2014'>Studies on the Relationship between Conscientiousness and Medication Adherence</h2><span id='topic+dat.molloy2014'></span>

<h3>Description</h3>

<p>Results from 16 studies on the correlation between conscientiousness and medication adherence.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.molloy2014
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>authors</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study authors </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size of the study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> observed correlation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>controls</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> number of variables controlled for </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>design</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether a cross-sectional or prospective design was used </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>a_measure</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of adherence measure (self-report or other) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>c_measure</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> type of conscientiousness measure (NEO or other) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>meanage</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean age of the sample </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>quality</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> methodological quality
</td>
</tr>

</table>



<h3>Details</h3>

<p>Conscientiousness, one of the big-5 personality traits, can be defined as &ldquo;socially prescribed impulse control that facilitates task- and goal-directed behaviour, such as thinking before acting, delaying gratification, following norms and rules and planning, organising and prioritising tasks&rdquo; (John &amp; Srivastava, 1999). Conscientiousness has been shown to be related to a number of health-related behaviors (e.g., tobacco/alcohol/drug use, diet and activity patterns, risky behaviors). A recent meta-analysis by Molloy et al. (2014) examined to what extent conscientiousness is related to medication adherence, that is, the extent to which (typically chronically ill) patients follow a prescribed medication regimen (e.g., taking a daily dose of a cholesterol lowering drug in patients with high LDL serum cholesterol levels). The results from the 16 studies included in this meta-analysis are provided in this dataset.
</p>
<p>Variable <code>a_measure</code> indicates whether adherence was measured based on self-reports or a more &lsquo;objective&rsquo; measure (e.g., electronic monitoring of pill bottle openings, pill counts). Variable <code>c_measure</code> indicates whether conscientiousness was measured with some version of the NEO personality inventory or some other scale. Methodological quality was scored by the authors on a 1 to 4 scale with higher scores indicating higher quality (see article for details on how this score was derived).
</p>


<h3>Concepts</h3>

<p>psychology, medicine, correlation coefficients
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Molloy, G. J., O'Carroll, R. E., &amp; Ferguson, E. (2014). Conscientiousness and medication adherence: A meta-analysis. <em>Annals of Behavioral Medicine</em>, <b>47</b>(1), 92&ndash;101. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s12160-013-9524-4&#8288;</code>
</p>


<h3>References</h3>

<p>John, O. P., &amp; Srivastava, S. (1999). The Big Five Trait taxonomy: History, measurement, and theoretical perspectives. In L. A. Pervin &amp; O. P. John (Eds.), <em>Handbook of personality: Theory and research</em> (2nd ed., pp. 102-138). New York: Guilford Press.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.molloy2014
dat[-c(5:6)]

## Not run: 

### load metafor package
library(metafor)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=ri, ni=ni, data=dat, slab=paste(authors, year, sep=", "))
dat[-c(5:6)]

### meta-analysis of the transformed correlations using a random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### average correlation with 95% CI
predict(res, digits=3, transf=transf.ztor)

### forest plot
forest(res, addpred=TRUE, xlim=c(-1.6,1.6), atransf=transf.ztor,
       at=transf.rtoz(c(-.4,-.2,0,.2,.4,.6)), digits=c(2,1), cex=.8,
       header="Author(s), Year")

### funnel plot
funnel(res)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.moura2021'>Studies on Assortative Mating</h2><span id='topic+dat.moura2021'></span>

<h3>Description</h3>

<p>Results from 457 studies on assortative mating in various species.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.moura2021
</code></pre>


<h3>Format</h3>

<p>The object is a list containing a data frame called <code>dat</code> that contains the following columns and a phylogenetic tree called <code>tree</code>:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study.id</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>effect.size.id</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> effect size id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>species</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>species.id</b>        </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> species id (as in the Open Tree of Life reference taxonomy) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>subphylum</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> the subphyla of the species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>phylum</b>            </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> the phyla of the species </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>assortment.trait</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> the measure of body size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trait.dimensions</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> dimensionality of the measure </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>field.collection</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether data were collected in the field </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>publication.year</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year of the study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>pooled.data</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether data were pooled either spatially and/or temporally </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>spatially.pooled</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether data were pooled spatially </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>temporally.pooled</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether data were pooled temporally </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ri</b>                </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> correlation coefficient </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>                </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 457 studies included in this dataset provide 1828 correlation coefficients describing the similarity in some measure of body size in mating couples in 341 different species.
</p>


<h3>Concepts</h3>

<p>ecology, evolution, correlation coefficients, multivariate models, phylogeny, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Rios Moura, R., Oliveira Gonzaga, M., Silva Pinto, N., Vasconcellos-Neto, J., &amp; Requena, G. S. (2021). Assortative mating in space and time: Patterns and biases. <em>Ecology Letters</em>, <b>24</b>(5), 1089&ndash;1102. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/ele.13690&#8288;</code>
</p>


<h3>References</h3>

<p>Cinar, O., Nakagawa, S., &amp; Viechtbauer, W. (in press). Phylogenetic multilevel meta-analysis: A simulation study on the importance of modelling the phylogeny. <em>Methods in Ecology and Evolution</em>. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/2041-210X.13760&#8288;</code>
</p>
<p>Hadfield, J. D., &amp; Nakagawa, S. (2010). General quantitative genetic methods for comparative biology: Phylogenies, taxonomies and multi-trait models for continuous and categorical characters. <em>Journal of Evolutionary Biology</em>, <b>23</b>(3), 494&ndash;508. <code style="white-space: pre;">&#8288;https://doi.org/10.1111/j.1420-9101.2009.01915.x&#8288;</code>
</p>
<p>Nakagawa, S., &amp; Santos, E. S. A. (2012). Methodological issues and advances in biological meta-analysis. <em>Evolutionary Ecology</em>, <b>26</b>(5), 1253&ndash;1274. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s10682-012-9555-5&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.moura2021$dat
head(dat)

## Not run: 

### load metafor package
library(metafor)

### load ape package
library(ape, warn.conflicts=FALSE)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=ri, ni=ni, data=dat)

### copy tree to 'tree'
tree &lt;- dat.moura2021$tree

### turn tree into an ultrametric one
tree &lt;- compute.brlen(tree)

### compute phylogenetic correlation matrix
A &lt;- vcv(tree, corr=TRUE)

### make copy of the species.id variable
dat$species.id.phy &lt;- dat$species.id

### fit multilevel phylogenetic meta-analytic model
res &lt;- rma.mv(yi, vi,
   random = list(~ 1 | study.id, ~ 1 | effect.size.id, ~ 1 | species.id, ~ 1 | species.id.phy),
   R=list(species.id.phy=A), data=dat)
res

### examine if spatial and/or temporal pooling of data tends to yield larger correlations
res &lt;- rma.mv(yi, vi,
   mods = ~ spatially.pooled * temporally.pooled,
   random = list(~ 1 | study.id, ~ 1 | effect.size.id, ~ 1 | species.id, ~ 1 | species.id.phy),
   R=list(species.id.phy=A), data=dat)
res

### estimated average correlation without pooling, when pooling spatially,
### when pooling temporally, and when pooling spatially and temporally
predict(res, newmods = rbind(c(0,0,0),c(1,0,0),c(0,1,0),c(1,1,1)), transf=transf.ztor, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.nakagawa2007'>Assessing the Function of House Sparrows' Bib Size Using a Flexible Meta-Analysis Method</h2><span id='topic+dat.nakagawa2007'></span>

<h3>Description</h3>

<p>A meta-analysis on the association between the size of a male's bib and their social status in house sparrows (<em>Passer domesticus</em>).</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.nakagawa2007
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>StudyID</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> identity of primary study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Place</b>       </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> location of study population </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>Correlation</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> correlation coefficient </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>SampleSize</b>  </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size of population </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p>Each study measures the association between a sparrows bib size and its social status. Effects are quantified as correlation coefficients.
</p>


<h3>Concepts</h3>

<p>ecology, correlation coefficients
</p>


<h3>Author(s)</h3>

<p>Daniel Noble, <a href="mailto:daniel.noble@anu.edu.au">daniel.noble@anu.edu.au</a>
</p>


<h3>Source</h3>

<p>Nakagawa, S., Ockendon, N., Gillespie, D. O. S, Hatchwell, B. J., &amp; Burke, T. (2007). Assessing the function of house sparrows' bib size using a flexible meta-analysis method. <em>Behavioral Ecology</em>, <b>18</b>(5), 831&ndash;840. <code style="white-space: pre;">&#8288;https://doi.org/10.1093/beheco/arm050&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.nakagawa2007
dat

## Not run: 

### load metafor package
library(metafor)

### calculate Zr
dat &lt;- escalc(measure="ZCOR", ri=Correlation, ni=SampleSize, data=dat)

### fit meta-analytic model
res &lt;- rma.mv(yi, vi, random = list(~ 1 | StudyID), data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.nielweise2007'>Studies on Anti-Infective-Treated Central Venous Catheters for Prevention of Catheter-Related Bloodstream Infections</h2><span id='topic+dat.nielweise2007'></span>

<h3>Description</h3>

<p>Results from 18 studies comparing the risk of catheter-related bloodstream infection when using anti-infective-treated versus standard catheters in the acute care setting.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.nielweise2007
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of CRBSIs in patients receiving an anti-infective catheter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients receiving an anti-infective catheter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of CRBSIs in patients receiving a standard catheter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients receiving a standard catheter
</td>
</tr>

</table>



<h3>Details</h3>

<p>The use of a central venous catheter may lead to a catheter-related bloodstream infection (CRBSI), which in turn increases the risk of morbidity and mortality. Anti-infective-treated catheters have been developed that are meant to reduce the risk of CRBSIs. Niel-Weise et al. (2007) conducted a meta-analysis of studies comparing infection risk when using anti-infective-treated versus standard catheters in the acute care setting. The results from 18 such studies are included in this dataset.
</p>
<p>The dataset was used in the article by Stijnen et al. (2010) to illustrate various generalized linear mixed-effects models for the meta-analysis of proportions and odds ratios (see &lsquo;References&rsquo;).
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, generalized linear models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Niel-Weise, B. S., Stijnen, T., &amp; van den Broek, P. J. (2007). Anti-infective-treated central venous catheters: A systematic review of randomized controlled trials. <em>Intensive Care Medicine</em>, <b>33</b>(12), 2058&ndash;2068. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s00134-007-0897-3&#8288;</code>
</p>


<h3>References</h3>

<p>Stijnen, T., Hamza, T. H., &amp; Ozdemir, P. (2010). Random effects meta-analysis of event outcome in the framework of the generalized linear mixed model with applications in sparse data. <em>Statistics in Medicine</em>, <b>29</b>(29), 3046&ndash;3067. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.4040&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.nielweise2007
dat

## Not run: 

### load metafor package
library(metafor)

### standard (inverse-variance) random-effects model
res &lt;- rma(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, drop00=TRUE)
print(res, digits=3)
predict(res, transf=exp, digits=2)

### random-effects conditional logistic model
res &lt;- rma.glmm(measure="OR", ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, model="CM.EL")
print(res, digits=3)
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.nielweise2008'>Studies on Anti-Infective-Treated Central Venous Catheters for Prevention of Catheter-Related Bloodstream Infections</h2><span id='topic+dat.nielweise2008'></span>

<h3>Description</h3>

<p>Results from 18 studies comparing the risk of catheter-related bloodstream infection when using anti-infective-treated versus standard catheters for total parenteral nutrition or chemotherapy.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.nielweise2008
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>authors</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study authors </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of CRBSIs in patients receiving an anti-infective catheter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total number of catheter days for patients receiving an anti-infective catheter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>x2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of CRBSIs in patients receiving a standard catheter </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total number of catheter days for patients receiving a standard catheter
</td>
</tr>

</table>



<h3>Details</h3>

<p>The use of a central venous catheter may lead to a catheter-related bloodstream infection (CRBSI), which in turn increases the risk of morbidity and mortality. Anti-infective-treated catheters have been developed that are meant to reduce the risk of CRBSIs. Niel-Weise et al. (2008) conducted a meta-analysis of studies comparing infection risk when using anti-infective-treated versus standard catheters for total parenteral nutrition or chemotherapy. The results from 9 such studies are included in this dataset.
</p>
<p>The dataset was used in the article by Stijnen et al. (2010) to illustrate various generalized linear mixed-effects models for the meta-analysis of incidence rates and incidence rate ratios (see &lsquo;References&rsquo;).
</p>


<h3>Concepts</h3>

<p>medicine, incidence rates, generalized linear models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Niel-Weise, B. S., Stijnen, T., &amp; van den Broek, P. J. (2008). Anti-infective-treated central venous catheters for total parenteral nutrition or chemotherapy: A systematic review. <em>Journal of Hospital Infection</em>, <b>69</b>(2), 114&ndash;123. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.jhin.2008.02.020&#8288;</code>
</p>


<h3>References</h3>

<p>Stijnen, T., Hamza, T. H., &amp; Ozdemir, P. (2010). Random effects meta-analysis of event outcome in the framework of the generalized linear mixed model with applications in sparse data. <em>Statistics in Medicine</em>, <b>29</b>(29), 3046&ndash;3067. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.4040&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.nielweise2008
dat

## Not run: 

### load metafor package
library(metafor)

### standard (inverse-variance) random-effects model
res &lt;- rma(measure="IRR", x1i=x1i, t1i=t1i, x2i=x2i, t2i=t2i, data=dat)
print(res, digits=3)
predict(res, transf=exp, digits=2)

### random-effects conditional Poisson model
res &lt;- rma.glmm(measure="IRR", x1i=x1i, t1i=t1i, x2i=x2i, t2i=t2i, data=dat, model="CM.EL")
print(res, digits=3)
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.normand1999'>Studies on the Length of Hospital Stay of Stroke Patients</h2><span id='topic+dat.normand1999'></span>

<h3>Description</h3>

<p>Results from 9 studies on the length of the hospital stay of stroke patients under specialized care and under conventional/routine (non-specialist) care.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.normand1999
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>source</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> source of data </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients under specialized care </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>m1i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean length of stay (in days) under specialized care </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd1i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation of the length of stay under specialized care </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients under routine care </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>m2i</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> mean length of stay (in days) under routine care </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd2i</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation of the length of stay under routine care
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 9 studies provide data in terms of the mean length of the hospital stay (in days) of stroke patients under specialized care and under conventional/routine (non-specialist) care. The goal of the meta-analysis was to examine the hypothesis whether specialist stroke unit care will result in a shorter length of hospitalization compared to routine management.
</p>


<h3>Concepts</h3>

<p>medicine, raw mean differences, standardized mean differences
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Normand, S. T. (1999). Meta-analysis: Formulating, evaluating, combining, and reporting. <em>Statistics in Medicine</em>, <b>18</b>(3), 321&ndash;359. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/(sici)1097-0258(19990215)18:3&lt;321::aid-sim28&gt;3.0.co;2-p&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.normand1999
dat

## Not run: 

### load metafor package
library(metafor)

### calculate mean differences and corresponding sampling variances
dat &lt;- escalc(measure="MD", m1i=m1i, sd1i=sd1i, n1i=n1i, m2i=m2i, sd2i=sd2i, n2i=n2i, data=dat)
dat

### meta-analysis of mean differences using a random-effects model
res &lt;- rma(yi, vi, data=dat)
res

### meta-analysis of standardized mean differences using a random-effects model
res &lt;- rma(measure="SMD", m1i=m1i, sd1i=sd1i, n1i=n1i, m2i=m2i, sd2i=sd2i, n2i=n2i,
           data=dat, slab=source)
res

### draw forest plot
forest(res, xlim=c(-7,5), alim=c(-3,1), header="Study/Source")

### calculate (log transformed) ratios of means and corresponding sampling variances
dat &lt;- escalc(measure="ROM", m1i=m1i, sd1i=sd1i, n1i=n1i, m2i=m2i, sd2i=sd2i, n2i=n2i, data=dat)
dat

### meta-analysis of the (log transformed) ratios of means using a random-effects model
res &lt;- rma(yi, vi, data=dat)
res
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.obrien2003'>Studies on the Relationship Between BMI and Risk of Preeclampsia</h2><span id='topic+dat.obrien2003'></span>

<h3>Description</h3>

<p>Results from 13 studies on the relationship between maternal body mass index (BMI) and the risk of preeclampsia.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.obrien2003
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> (first) author of the study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ref</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> reference number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ch</b>         </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> exclusion due to chronic hypertension (yes/no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dm</b>         </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> exclusion due to diabetes mellitus (yes/no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mg</b>         </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> exclusion due to multiple gestation (yes/no) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bmi.lb</b>     </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> lower bound of the BMI interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bmi.ub</b>     </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> upper bound of the BMI interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>bmi</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> midpoint of the BMI interval </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cases</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of preeclampsia cases in the BMI group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>total</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> number of individuals in the BMI group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 13 studies examining the relationship between maternal body mass index (BMI) and the risk of preeclampsia. For each study, results are given in terms of the number of preeclampsia cases within two or more groups defined by the lower and upper BMI bounds as shown in the dataset (<code>NA</code> means that the interval is either open to the left or right). The <code>bmi</code> variable is the interval midpoint as defined by O'Brien et al. (2003).
</p>


<h3>Concepts</h3>

<p>medicine, obstetrics, risk ratios, proportions, multilevel models, dose-response models
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>O'Brien, T. E., Ray, J. G., &amp; Chan, W.-S. (2003). Maternal body mass index and the risk of preeclampsia: A systematic overview. <em>Epidemiology</em>, <b>14</b>(3), 368&ndash;374. <code style="white-space: pre;">&#8288;https://doi.org/10.1097/00001648-200305000-00020&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.obrien2003
dat

## Not run: 

### load metafor package
library(metafor)

### restructure the data into a wide format
dat2 &lt;- to.wide(dat, study="study", grp="grp", ref=1, grpvars=c("bmi","cases","total"),
                addid=FALSE, adddesign=FALSE, postfix=c(1,2))
dat2[1:10, -c(2:3)]

### calculate log risk ratios and corresponding sampling variances
dat2 &lt;- escalc(measure="RR", ai=cases1, n1i=total1, ci=cases2, n2i=total2, data=dat2)
dat2[1:10, -c(2:7)]

### forest plot of the risk ratios
dd &lt;- c(0,diff(dat2$study))
dd[dd &gt; 0] &lt;- 1
rows &lt;- (1:nrow(dat2)) + cumsum(dd)
rows &lt;- 1 + max(rows) - rows
slabs &lt;- mapply(function(x,y,z) as.expression(bquote(.(x)^.(y)~.(z))),
                dat2$author, dat2$ref, dat2$year)
with(dat2, forest(yi, vi, header=TRUE, slab=slabs, xlim=c(-7,5.5), fonts="mono", cex=0.8,
   psize=1, pch=19, efac=0, rows=rows, ylim=c(0,max(rows)+3), yaxs="i",
   atransf=exp, at=log(c(.05,0.1,0.2,0.5,1,2,5,10,20)), ilab=comp, ilab.xpos=-4, ilab.pos=4))
text(-4.4, max(rows)+2, "Comparison", font=2, cex=0.8, pos=4)

### within-study mean center the BMI variable
dat$bmicent &lt;- with(dat, bmi - ave(bmi, study))

### compute the proportion of preeclampsia cases and corresponding sampling variances
dat &lt;- escalc(measure="PR", xi=cases, ni=total, data=dat)

### convert the proportions to percentages (and convert the variances accordingly)
dat$yi &lt;- dat$yi*100
dat$vi &lt;- dat$vi*100^2
dat[1:10, -c(2:3)]

### fit multilevel meta-regression model to examine the relationship between the
### (centered) BMI variable and the risk of preeclampsia
res &lt;- rma.mv(yi, vi, mods = ~ bmicent, random = ~ 1 | study/grp, data=dat)
res

### draw scatterplot with regression line
res$slab &lt;- dat$ref
regplot(res, xlab=expression("Within-Study Mean Centered BMI"~(kg/m^2)),
        ylab="Preeclampsia Prevalence (%)", las=1, bty="l",
        at=seq(0,18,by=2), olim=c(0,100), psize=2, bg="gray90",
        label=TRUE, offset=0, labsize=0.6)

### fit model using a random slope for bmicent
res &lt;- rma.mv(yi, vi, mods = ~ bmicent, random = ~ bmicent | study, struct="GEN", data=dat)
res

### load rms package
library(rms)

### fit restricted cubic spline model
res &lt;- rma.mv(yi, vi, mods = ~ rcs(bmicent, 4), random = ~ 1 | study/grp, data=dat)
res

### get knot positions
knots &lt;- attr(rcs(model.matrix(res)[,2], 4), "parms")

### computed predicted values based on the model
xs &lt;- seq(-10, 10, length=1000)
sav &lt;- predict(res, newmods=rcspline.eval(xs, knots, inclx=TRUE))

### draw scatterplot with regression line based on the model
tmp &lt;- regplot(res, mod=2, pred=sav,
               xvals=xs, xlab=expression("Within-Study Mean Centered BMI"~(kg/m^2)),
               ylab="Preeclampsia Prevalence (%)", las=1, bty="l",
               at=seq(0,18,by=2), olim=c(0,100), psize=2, bg="gray90",
               label=TRUE, offset=0, labsize=0.6)
abline(v=knots, lty="dotted")
points(tmp)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.pagliaro1992'>Studies on the Effectiveness of Nonsurgical Treatments in Cirrhosis</h2><span id='topic+dat.pagliaro1992'></span>

<h3>Description</h3>

<p>Results from 26 trials examining the effectiveness of beta-blockers and sclerotherapy for the prevention of first bleeding in patients with cirrhosis</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.pagliaro1992
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trt</b>      </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> either beta-blockers, sclerotherapy, or control </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients with first bleeding </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients treated
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 26 randomized controlled trials examining the effectiveness of nonsurgical treatments for the prevention of first bleeding in patients with cirrhosis. Patients were either treated with beta-blockers, endoscopic sclerotherapy, or with a nonactive treatment (control). Two trials included all three treatment conditions, 7 trials compared beta-blockers against control, and 17 trials compared sclerotherapy against control. The dataset has been used in various papers to illustrate methods for conducting a network meta-analysis / mixed treatment comparison.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, Mantel-Haenszel method, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Pagliaro, L., D'Amico, G., Sörensen, T. I. A., Lebrec, D., Burroughs, A. K., Morabito, A., Tiné, F., Politi, F., &amp; Traina, M. (1992). Prevention of first bleeding in cirrhosis: A meta-analysis of randomized trials of nonsurgical treatment. <em>Annals of Internal Medicine</em>, <b>117</b>(1), 59&ndash;70. <code style="white-space: pre;">&#8288;https://doi.org/10.7326/0003-4819-117-1-59&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.pagliaro1992
dat

## Not run: 

### load metafor package
library(metafor)

### restructure dataset to a contrast-based format
dat.c &lt;- to.wide(dat, study="study", grp="trt", grpvars=3:4)
dat.c

### Mantel-Haenszel results for beta-blockers and sclerotherapy versus control, respectively
rma.mh(measure="OR", ai=xi.1, n1i=ni.1, ci=xi.2, n2i=ni.2,
       data=dat.c, subset=(trt.1=="beta-blockers"), digits=2)
rma.mh(measure="OR", ai=xi.1, n1i=ni.1, ci=xi.2, n2i=ni.2,
       data=dat.c, subset=(trt.1=="sclerotherapy"), digits=2)

### calculate log odds for each study arm
dat &lt;- escalc(measure="PLO", xi=xi, ni=ni, data=dat)
dat

### turn treatment variable into factor and set reference level
dat$trt &lt;- relevel(factor(dat$trt), ref="control")

### add a space before each level (this makes the output a bit more legible)
levels(dat$trt) &lt;- paste0(" ", levels(dat$trt))

### network meta-analysis using an arm-based random-effects model with fixed study effects
### (by setting rho=1/2, tau^2 reflects the amount of heterogeneity for all treatment comparisons)
res &lt;- rma.mv(yi, vi, mods = ~ factor(study) + trt - 1, random = ~ trt | study, rho=1/2, data=dat)
res

### average odds ratio comparing beta-blockers and sclerotherapy versus control, respectively
predict(res, newmods=c(rep(0,26), 1, 0), transf=exp, digits=2)
predict(res, newmods=c(rep(0,26), 0, 1), transf=exp, digits=2)

### average odds ratio comparing beta-blockers versus sclerotherapy
predict(res, newmods=c(rep(0,26), 1, -1), transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.pignon2000'>Studies on the Effectiveness of Locoregional Treatment plus Chemotherapy for Head and Neck Squamous-Cell Carcinoma</h2><span id='topic+dat.pignon2000'></span>

<h3>Description</h3>

<p>Results from studies examining mortality risk in patients with nonmetastatic head and neck squamous-cell carcinoma receiving either locoregional treatment plus chemotherapy versus locoregional treatment alone.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.pignon2000
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>id</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trial</b>      </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> trial abbreviation </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>OmE</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> observed minus expected number of deaths in the locoregional treatment plus chemotherapy group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>V</b>          </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> corresponding variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>grp</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> timing of chemotherapy: 1 = adjuvant, 2 = neoadjuvant, 3 = concomitant
</td>
</tr>

</table>



<h3>Details</h3>

<p>The purpose of this meta-analysis was to examine the mortality risk in patients with nonmetastatic head and neck squamous-cell carcinoma receiving either locoregional treatment plus chemotherapy versus locoregional treatment alone. For 65 trials, the dataset provides the observed minus expected number of deaths and corresponding variances in the locoregional treatment plus chemotherapy group. Based on these values, we can estimate the log hazard ratios with <code>OmE/V</code> and the corresponding sampling variance with <code>1/V</code>.
</p>
<p>The trials were also divided according to the timing of the chomotherapy: (1) adjuvant, after the locoregional treatment, (2) neoadjuvant, before the locoregional treatment, and (3) concomitant, chemotherapy given concomitantly or alternating with radiotherapy.
</p>


<h3>Concepts</h3>

<p>medicine, oncology, hazard ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Pignon, J. P., Bourhis, J., Domenge, C., &amp; Designe, L. (2000). Chemotherapy added to locoregional treatment for head and neck squamous-cell carcinoma: Three meta-analyses of updated individual data. <em>Lancet</em>, <b>355</b>(9208), 949&ndash;955. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/S0140-6736(00)90011-4&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.pignon2000
head(dat)

## Not run: 

### load metafor package
library(metafor)

### calculate log hazard ratios and sampling variances
dat$yi &lt;- with(dat, OmE/V)
dat$vi &lt;- with(dat, 1/V)
head(dat)

### meta-analysis based on all 65 trials
res &lt;- rma(yi, vi, data=dat, method="EE", digits=2)
res
predict(res, transf=exp)

### only adjuvant trials
res &lt;- rma(yi, vi, data=dat, method="EE", subset=grp==1, digits=2)
res
predict(res, transf=exp)

### only neoadjuvant trials
res &lt;- rma(yi, vi, data=dat, method="EE", subset=grp==2, digits=2)
res
predict(res, transf=exp)

### only concomitant trials
res &lt;- rma(yi, vi, data=dat, method="EE", subset=grp==3, digits=2)
res
predict(res, transf=exp)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.pritz1997'>Studies on the Effectiveness of Hyperdynamic Therapy for Treating Cerebral Vasospasm</h2><span id='topic+dat.pritz1997'></span>

<h3>Description</h3>

<p>Results from 14 studies on the effectiveness of hyperdynamic therapy for treating cerebral vasospasm.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.pritz1997
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>authors</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study authors </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients that improved with hyperdynamic therapy </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> total number of patients treated
</td>
</tr>

</table>



<h3>Details</h3>

<p>As described in Zhou et al. (1999), &quot;hyperdynamic therapy refers to induced hypertension and hypervolaemia (volume expansion) to treat ischaemic symptoms due to vasospasm, and the success of this therapy is defined as clinical improvement in terms of neurologic deficits.&quot; For each study that was included in the meta-analysis, the dataset includes information on the number of patients that improved under this form of therapy and the total number of patients that were treated. The goal of the meta-analysis is to estimate the true (average) success rate of hyperdynamic therapy.
</p>


<h3>Concepts</h3>

<p>medicine, single-arm studies, proportions
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Zhou, X.-H., Brizendine, E. J., &amp; Pritz, M. B. (1999). Methods for combining rates from several studies. <em>Statistics in Medicine</em>, <b>18</b>(5), 557&ndash;566. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/(SICI)1097-0258(19990315)18:5&lt;557::AID-SIM53&gt;3.0.CO;2-F&#8288;</code>
</p>


<h3>References</h3>

<p>Pritz M. B., Zhou, X.-H., &amp; Brizendine, E. J. (1996). Hyperdynamic therapy for cerebral vasospasm: A meta-analysis of 14 studies. <em>Journal of Neurovascular Disease</em>, <b>1</b>, 6&ndash;8.
</p>
<p>Pritz, M. B. (1997). Treatment of cerebral vasospasm due to aneurysmal subarachnoid hemorrhage: Past, present, and future of hyperdynamic therapy. <em>Neurosurgery Quarterly</em>, <b>7</b>(4), 273&ndash;285.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.pritz1997
dat

## Not run: 

### load metafor package
library(metafor)

### computation of "weighted average" in Zhou et al. (1999), Table IV
dat &lt;- escalc(measure="PR", xi=xi, ni=ni, data=dat, add=0)
theta.hat    &lt;- sum(dat$ni * dat$yi) / sum(dat$ni)
se.theta.hat &lt;- sqrt(sum(dat$ni^2 * dat$vi) / sum(dat$ni)^2)
ci.lb        &lt;- theta.hat - 1.96 * se.theta.hat
ci.ub        &lt;- theta.hat + 1.96 * se.theta.hat
round(c(estimate = theta.hat, se = se.theta.hat, ci.lb = ci.lb, ci.ub = ci.ub), 4)

### this is identical to an equal-effects model with sample size weights
rma(yi, vi, weights=ni, method="EE", data=dat)

### random-effects model with raw proportions
dat &lt;- escalc(measure="PR", xi=xi, ni=ni, data=dat)
res &lt;- rma(yi, vi, data=dat)
predict(res)

### random-effects model with logit transformed proportions
dat &lt;- escalc(measure="PLO", xi=xi, ni=ni, data=dat)
res &lt;- rma(yi, vi, data=dat)
predict(res, transf=transf.ilogit)

### mixed-effects logistic regression model
res &lt;- rma.glmm(measure="PLO", xi=xi, ni=ni, data=dat)
predict(res, transf=transf.ilogit)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.raudenbush1985'>Studies on Assessing the Effects of Teacher Expectations on Pupil IQ</h2><span id='topic+dat.raudenbush1985'></span>

<h3>Description</h3>

<p>Results from 19 studies examining how teachers' expectations about their pupils can influence actual IQ levels.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.raudenbush1985
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>author</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> study author(s) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>year</b>       </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> publication year </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>weeks</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> weeks of contact prior to expectancy induction </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>setting</b>    </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> whether tests were group or individually administered </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>tester</b>     </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> whether test administrator was aware or blind </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sample size of experimental group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sample size of control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> standardized mean difference </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> corresponding sampling variance
</td>
</tr>

</table>



<h3>Details</h3>

<p>In the so-called &lsquo;Pygmalion study&rsquo; (Rosenthal &amp; Jacobson, 1968), &ldquo;all of the predominantly poor children in the so-called Oak elementary school were administered a test pretentiously labeled the &lsquo;Harvard Test of Inflected Acquisition.&rsquo; After explaining that this newly designed instrument had identified those children most likely to show dramatic intellectual growth during the coming year, the experimenters gave the names of these &lsquo;bloomers&rsquo; to the teachers. In truth, the test was a traditional IQ test and the &lsquo;bloomers&rsquo; were a randomly selected 20% of the student population. After retesting the children 8 months later, the experimenters reported that those predicted to bloom had in fact gained significantly more in total IQ (nearly 4 points) and reasoning IQ (7 points) than the control group children. Further, at the end of the study, the teachers rated the experimental children as intellectually more curious, happier, better adjusted, and less in need of approval than their control group peers&rdquo; (Raudenbush, 1984).
</p>
<p>In the following years, a series of studies were conducted attempting to replicate this rather controversial finding. However, the great majority of those studies were unable to demonstrate a statistically significant difference between the two experimental groups in terms of IQ scores. Raudenbush (1984) conducted a meta-analysis based on 19 such studies to further examine the evidence for the existence of the &lsquo;Pygmalion effect&rsquo;. The dataset includes the results from these studies.
</p>
<p>The outcome measure used for the meta-analysis was the standardized mean difference (<code>yi</code>), with positive values indicating that the supposed &lsquo;bloomers&rsquo; had, on average, higher IQ scores than those in the control group. The <code>weeks</code> variable indicates the number of weeks of prior contact between teachers and students before the expectancy induction. Testing was done either in a group setting or individually, which is indicated by the <code>setting</code> variable. Finally, the <code>tester</code> variable indicates whether the test administrators were either aware or blind to the researcher-provided designations of the children's intellectual potential.
</p>
<p>The data in this dataset were obtained from Raudenbush and Bryk (1985) with information on the <code>setting</code> and <code>tester</code> variables extracted from Raudenbush (1984).
</p>


<h3>Concepts</h3>

<p>education, standardized mean differences, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Raudenbush, S. W. (1984). Magnitude of teacher expectancy effects on pupil IQ as a function of the credibility of expectancy induction: A synthesis of findings from 18 experiments. <em>Journal of Educational Psychology</em>, <b>76</b>(1), 85&ndash;97. <code style="white-space: pre;">&#8288;https://doi.org/10.1037/0022-0663.76.1.85&#8288;</code>
</p>
<p>Raudenbush, S. W., &amp; Bryk, A. S. (1985). Empirical Bayes meta-analysis. <em>Journal of Educational Statistics</em>, <b>10</b>(2), 75&ndash;98. <code style="white-space: pre;">&#8288;https://doi.org/10.3102/10769986010002075&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.raudenbush1985
dat

## Not run: 

### load metafor package
library(metafor)

### random-effects model
res &lt;- rma(yi, vi, data = dat)
res

### create weeks variable where values larger than 3 are set to 3
dat$weeks.c &lt;- ifelse(dat$weeks &gt; 3, 3, dat$weeks)

### mixed-effects model with weeks.c variable as moderator
res &lt;- rma(yi, vi, mods = ~ weeks.c, data = dat, digits = 3)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.riley2003'>Studies on MYC-N as a Prognostic Marker for Neuroblastoma</h2><span id='topic+dat.riley2003'></span>

<h3>Description</h3>

<p>Results from 81 studies examining overall and disease-free survival in neuroblastoma patients with amplified versus normal MYC-N protein levels.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.riley2003
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>      </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> study number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> log hazard ratio of the outcome in those with amplified versus normal MYC-N protein levels </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>         </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> sampling variance of the log hazard ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sei</b>        </td><td style="text-align: left;"> <code>numeric</code>    </td><td style="text-align: left;"> standard error of the log hazard ratio </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>outcome</b>    </td><td style="text-align: left;"> <code>character</code>  </td><td style="text-align: left;"> outcome (OS = overall survival; DFS = disease-free survival)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The meta-analysis by Riley et al. (2003) examined a variety of prognostic markers for overall and disease-free survival in patients with neuroblastoma. One of the markers examined was amplified levels of the MYC-N protein, with is associated with poorer outcomes.
</p>
<p>The dataset given here was extracted from Riley (2011) and has been used in several other publications (e.g., Riley et al., 2004, 2007). The dataset provides the (log) hazard ratios (and corresponding standard errors) with respect to these two outcomes in 81 studies, with positive values indicating a greater risk of death (for OS) or disease recurrence/death (for DFS) for patients with high MYC-N levels compared to those with normal/low levels. Note that information on both outcomes could only be extracted from 17 studies (39 studies only provided sufficient information to extract the OS estimate, while 25 studies only allowed for extraction of the DFS estimate).
</p>


<h3>Concepts</h3>

<p>medicine, oncology, hazard ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Riley, R. D., Sutton, A. J., Abrams, K. R., &amp; Lambert, P. C. (2004). Sensitivity analyses allowed more appropriate and reliable meta-analysis conclusions for multiple outcomes when missing data was present. <em>Journal of Clinical Epidemiology</em>, <b>57</b>(9), 911&ndash;924. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/j.jclinepi.2004.01.018&#8288;</code>
</p>
<p>Riley, R. D., Abrams, K. R., Lambert, P. C., Sutton, A. J., &amp; Thompson, J. R. (2007). An evaluation of bivariate random-effects meta-analysis for the joint synthesis of two correlated outcomes. <em>Statistics in Medicine</em>, <b>26</b>(1), 78&ndash;97. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.2524&#8288;</code>
</p>
<p>Riley, R. D. (2011). Erratum: An evaluation of bivariate random-effects meta-analysis for the joint synthesis of two correlated outcomes. <em>Statistics in Medicine</em>, <b>30</b>(4), 400. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/sim.4100&#8288;</code>
</p>


<h3>References</h3>

<p>Riley, R. D., Burchill, S. A., Abrams, K. R., Heney, D., Lambert, P. C., Jones, D. R., Sutton, A. J., Young, B., Wailoo, A. J., &amp; Lewis, I. J. (2003). A systematic review and evaluation of the use of tumour markers in paediatric oncology: Ewing's sarcoma and neuroblastoma. <em>Health Technology Assessment</em>, <b>7</b>(5), 1&ndash;162. <code style="white-space: pre;">&#8288;https://doi.org/10.3310/hta7050&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.riley2003
dat

## Not run: 

### load metafor package
library(metafor)

### random-effects model analysis for outcome DFS
res &lt;- rma(yi, sei=sei, data=dat, subset=(outcome == "DFS"), method="DL")
res
predict(res, transf=exp, digits=2)

### random-effects model analysis for outcome OS
res &lt;- rma(yi, sei=sei, data=dat, subset=(outcome == "OS"), method="DL")
res
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.senn2013'>Studies on the Effectiveness of Glucose-Lowering Agents</h2><span id='topic+dat.senn2013'></span>

<h3>Description</h3>

<p>Results from 26 trials examining the effectiveness of glucose-lowering agents in patients with type 2 diabetes</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.senn2013
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> (first) author and year of study </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ni</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> sample size of the study arm </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment given </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>comment</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether figures given are based on raw values at outcome or on change from baseline </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>mi</b>        </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> raw mean or mean change </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sdi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes the results from 26 randomized controlled trials examining the effectiveness of adding various oral glucose-lowering agents to a baseline sulfonylurea therapy in patients with type 2 diabetes. The outcome measured in the studies was either the mean HbA1c level at follow-up or the mean change in HbA1c level from baseline to follow-up. A total of 10 different treatment types were examined in these studies: acarbose, benfluorex, metformin, miglitol, pioglitazone, placebo, rosiglitazone, sitagliptin, sulfonylurea alone, and vildagliptin. One study included three treatment arms (Willms, 1999), while the rest of the studies included two treatment arms (hence, the dataset includes the results from 53 treatment arms).
</p>
<p>The data can be used for a network meta-analysis, either using an arm-based or a contrast-based model. See &lsquo;Examples&rsquo; below.
</p>


<h3>Concepts</h3>

<p>medicine, raw mean differences, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Senn, S., Gavini, F., Magrez, D., &amp; Scheen, A. (2013). Issues in performing a network meta-analysis. <em>Statistical Methods in Medical Research</em>, <b>22</b>(2), 169&ndash;189. <code style="white-space: pre;">&#8288;https://doi.org/10.1177/0962280211432220&#8288;</code>
</p>


<h3>References</h3>

<p>Law, M., Jackson, D., Turner, R., Rhodes, K., &amp; Viechtbauer, W. (2016). Two new methods to fit models for network meta-analysis with random inconsistency effects. <em>BMC Medical Research Methodology</em>, <b>16</b>, 87. <code style="white-space: pre;">&#8288;https://doi.org/10.1186/s12874-016-0184-5&#8288;</code>
</p>
<p>Rücker, G., &amp; Schwarzer, G. (2015). Ranking treatments in frequentist network meta-analysis works without resampling methods. <em>BMC Medical Research Methodology</em>, <b>15</b>, 58. <code style="white-space: pre;">&#8288;https://doi.org/10.1186/s12874-015-0060-8&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.senn2013
dat

## Not run: 

### load metafor package
library(metafor)

### create network graph ('igraph' package must be installed)
library(igraph, warn.conflicts=FALSE)
pairs &lt;- data.frame(do.call(rbind,
   sapply(split(dat$treatment, dat$study), function(x) t(combn(x,2)))), stringsAsFactors=FALSE)
pairs$X1 &lt;- factor(pairs$X1, levels=sort(unique(dat$treatment)))
pairs$X2 &lt;- factor(pairs$X2, levels=sort(unique(dat$treatment)))
tab &lt;- table(pairs[,1], pairs[,2])
tab # adjacency matrix
g &lt;- graph_from_adjacency_matrix(tab, mode = "plus", weighted=TRUE, diag=FALSE)
plot(g, edge.curved=FALSE, edge.width=E(g)$weight, layout=layout_as_star(g, center="placebo"),
     vertex.size=45, vertex.color="lightgray", vertex.label.color="black", vertex.label.font=2)

### table of studies versus treatments examined
print(addmargins(table(dat$study, dat$treatment)), zero.print="")

### table of frequencies with which treatment pairs were studied
print(as.table(crossprod(table(dat$study, dat$treatment))), zero.print="")

### add means and sampling variances of the means to the dataset
dat &lt;- escalc(measure="MN", mi=mi, sdi=sdi, ni=ni, data=dat)

### turn treatment variable into factor and set reference level
dat$treatment &lt;- relevel(factor(dat$treatment), ref="placebo")

### add a space before each level (this makes the output a bit more legible)
levels(dat$treatment) &lt;- paste0(" ", levels(dat$treatment))

### network meta-analysis using an arm-based fixed-effects model with fixed study effects
res.fe &lt;- rma.mv(yi, vi, mods = ~ study + treatment - 1, data=dat, slab=paste0(study, treatment))
res.fe

### test if treatment factor as a whole is significant
anova(res.fe, btt="treatment")

### forest plot of the contrast estimates (treatments versus placebos)
forest(tail(coef(res.fe), 9), tail(diag(vcov(res.fe)), 9), slab=levels(dat$treatment)[-1],
       xlim=c(-2.5, 2.0), alim=c(-1.5, 0.5), psize=1, xlab="Estimate", header="Treatment")

### weight matrix for the estimation of the fixed effects (leaving out the study effects)
w &lt;- t(tail(vcov(res.fe) %*% t(model.matrix(res.fe)) %*% weights(res.fe, type="matrix"), 9))
rownames(w) &lt;- res.fe$slab

### create shade plot for the diabetes network with placebo as the reference treatment
### negative values in blue shades, positive values in red shades
cols &lt;- colorRampPalette(c("blue", "gray95", "red"))(9)
heatmap(w, Rowv=NA, Colv=NA, scale="none", margins=c(6,11), col=cols,
        cexRow=.7, cexCol=1, labCol=levels(dat$treatment)[-1])

### network meta-analysis using an arm-based random-effects model with fixed study effects
### by setting rho=1/2, tau^2 reflects the amount of heterogeneity for all treatment comparisons
res.re &lt;- rma.mv(yi, vi, mods = ~ study + treatment - 1, random = ~ treatment | study, rho=1/2,
                 data=dat, slab=paste0(study, treatment))
res.re

### test if treatment factor as a whole is significant
anova(res.re, btt="treatment")

### forest plot of the contrast estimates (treatments versus placebos)
forest(tail(coef(res.re), 9), tail(diag(vcov(res.re)), 9), slab=levels(dat$treatment)[-1],
       xlim=c(-3.0, 2.5), alim=c(-1.5, 0.5), psize=1, xlab="Estimate", header="Treatment")

### compute the contribution of each study to the overall Q-test value
qi &lt;- sort(by((resid(res.fe) / sqrt(dat$vi))^2, dat$study, sum))

### check that the values add up
sum(qi)
res.fe$QE

### plot the values
s &lt;- length(qi)
par(mar=c(5,10,2,1))
plot(qi, 1:s, pch=19, xaxt="n", yaxt="n", xlim=c(0,40), xlab="Chi-Square Contribution", ylab="")
axis(side=1)
axis(side=2, at=1:s, labels=names(qi), las=1, tcl=0)
segments(rep(0,s), 1:s, qi, 1:s)

############################################################################

### restructure dataset to a contrast-based format
dat &lt;- dat.senn2013[c(1,4:2,5:6)] # reorder variables first
dat &lt;- to.wide(dat, study="study", grp="treatment", ref="placebo", grpvars=4:6)
dat

### calculate mean difference and corresponding sampling variance for each treatment comparison
dat &lt;- escalc(measure="MD", m1i=mi.1, sd1i=sdi.1, n1i=ni.1,
                            m2i=mi.2, sd2i=sdi.2, n2i=ni.2, data=dat)
dat

### calculate the variance-covariance matrix of the mean differences for the multitreatment studies
calc.v &lt;- function(x) {
   v &lt;- matrix(x$sdi.2[1]^2 / x$ni.2[1], nrow=nrow(x), ncol=nrow(x))
   diag(v) &lt;- x$vi
   v
}
V &lt;- bldiag(lapply(split(dat, dat$study), calc.v))

### add contrast matrix to dataset
dat &lt;- contrmat(dat, grp1="treatment.1", grp2="treatment.2")
dat

### network meta-analysis using a contrast-based random-effects model
### by setting rho=1/2, tau^2 reflects the amount of heterogeneity for all treatment comparisons
### the treatment left out (placebo) becomes the reference level for the treatment comparisons
res &lt;- rma.mv(yi, V, mods = ~ acarbose + benfluorex + metformin + miglitol + pioglitazone +
                              rosiglitazone + sitagliptin + sulfonylurea + vildagliptin - 1,
              random = ~ comp | study, rho=1/2, data=dat)
res

### forest plot of the contrast estimates (treatments versus placebos)
forest(coef(res), diag(vcov(res)), slab=names(coef(res)), order="obs",
       xlim=c(-3.0, 2.5), alim=c(-1.5, 0.5), psize=1, xlab="Estimate", header="Treatment")

### estimate all pairwise differences between treatments
contr &lt;- data.frame(t(combn(names(coef(res)), 2)))
contr &lt;- contrmat(contr, "X1", "X2", last="vildagliptin")
rownames(contr) &lt;- paste(contr$X1, "-", contr$X2)
contr &lt;- as.matrix(contr[-c(1:2)])
sav &lt;- predict(res, newmods=contr)
sav[["slab"]] &lt;- rownames(contr)
sav

### fit random inconsistency effects model (see Law et al., 2016)
inc &lt;- rma.mv(yi, V, mods = ~ acarbose + benfluorex + metformin + miglitol + pioglitazone +
                              rosiglitazone + sitagliptin + sulfonylurea + vildagliptin - 1,
              random = list(~ comp | study, ~ comp | design), rho=1/2, phi=1/2, data=dat)
inc

############################################################################

### compute P-scores (see Rücker &amp; Schwarzer, 2015)
contr &lt;- data.frame(t(combn(c(names(coef(res)),"placebo"), 2))) # add 'placebo' to contrast matrix
contr &lt;- contrmat(contr, "X1", "X2", last="placebo", append=FALSE)
b &lt;- c(coef(res),0) # add 0 for 'placebo' (the reference treatment)
vb &lt;- bldiag(vcov(res),0) # add 0 row/column for 'placebo' (the reference treatment)
pvals &lt;- apply(contr, 1, function(x) pnorm((x%*%b) / sqrt(t(x)%*%vb%*%x)))
tab &lt;- vec2mat(pvals, corr=FALSE)
tab[upper.tri(tab)] &lt;- t((1 - tab)[upper.tri(tab)])
rownames(tab) &lt;- colnames(tab) &lt;- colnames(contr)
round(tab, 2) # like Table 2 in the article
cbind(pscore=round(sort(apply(tab, 1, mean, na.rm=TRUE), decreasing=TRUE), 3))

# note: the values are slightly different from the ones given in Table 3 of Rücker and
# Schwarzer (2015) since model 'res' above is fitted using REML estimation while the
# results shown in the article are based on the 'netmeta' package, which uses a DL-type
# estimator for the amount of heterogeneity by default

############################################################################


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.stowe2010'>Studies on Adjuvant Treatments to Levodopa Therapy for Parkinson disease</h2><span id='topic+dat.stowe2010'></span>

<h3>Description</h3>

<p>Results from 29 trials assessing efficacy of three drug classes as adjuvant treatment to levodopa therapy in patients with Parkinson disease and motor complications.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.stowe2010
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study label </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>id</b>    </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> study id </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t1</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y1</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment effect arm 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd1</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation arm 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1</b>    </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size arm 1 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t2</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y2</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment effect arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd2</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2</b>    </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size arm 2 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>t3</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>y3</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> treatment effect arm 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sd3</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> standard deviation arm 3 </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n3</b>    </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> sample size arm 3
</td>
</tr>

</table>



<h3>Details</h3>

<p>This data set contains data from a Cochrane review assessing efficacy and safety of three drug classes as adjuvant treatment to levodopa therapy in patients with Parkinson disease and motor complications (Stowe et al., 2010).
</p>
<p>The authors conducted three pairwise meta-analyses comparing dopamine agonists, catechol-O-methyl transferase inhibitors (COMTI), and monoamine oxidase type B inhibitors (MAOBI) with placebo. The primary outcome was the mean reduction of the time spent in a relatively immobile &lsquo;off&rsquo; phase (mean off-time), calculated in hours per day. Relative treatment effects were expressed as mean difference. Data on this outcome were available for 5,331 patients from 28 studies comparing an active treatment with placebo and one three-arm study comparing two active treatments with placebo.
</p>


<h3>Concepts</h3>

<p>medicine, raw mean differences, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Stowe, R., Ives, N., Clarke, C. E., Deane, K., Hilten, V., Wheatley, K., Gray, R., Handley, K., &amp; Furmston, A. (2010). Evaluation of the efficacy and safety of adjuvant treatment to levodopa therapy in Parkinson's disease patients with motor complications. <em>Cochrane Database of Systematic Reviews</em>, <b>7</b>, CD007166. <code style="white-space: pre;">&#8288;https://doi.org/10.1002/14651858.CD007166.pub2&#8288;</code>
</p>


<h3>See Also</h3>

<p><code><a href="netmeta.html#topic+pairwise">pairwise</a></code>, <code><a href="meta.html#topic+metacont">metacont</a></code>, <code><a href="netmeta.html#topic+netmeta">netmeta</a></code>, <code><a href="netmeta.html#topic+netrank">netrank</a></code>, <code><a href="netmeta.html#topic+rankogram">rankogram</a></code>, <code><a href="netmeta.html#topic+netleague">netleague</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show results from three studies (including three-arm study LARGO)
dat.stowe2010[18:20, ]

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print mean differences with two digits and standard errors with 3
### digits
settings.meta(digits = 2, digits.se = 3)

### Transform data from wide arm-based format to contrast-based
### format. Argument 'sm' must not be provided as the mean difference
### is the default in R function metacont() called internally.
pw &lt;- pairwise(treat = list(t1, t2, t3), n = list(n1, n2, n3),
  mean = list(y1, y2, y3), sd = list(sd1, sd2, sd3),
  studlab = study, data = dat.stowe2010, sm = "MD")

### Show calculated mean differences (TE) for three studies
selstudy &lt;- c("COMTI(E) INT-OZ", "LARGO", "COMTI(E) Nomecomt")
subset(pw, studlab %in% selstudy)[, c(3:7, 10, 1)]

### Conduct random effects network meta-analysis (NMA)
### with placebo as reference
net &lt;- netmeta(pw, fixed = FALSE, ref = "plac")

### Show network graph
netgraph(net, number = TRUE, multiarm = TRUE,
  cex = 1.25, offset = 0.025,
  cex.number = 1, pos.number.of.studies = 0.3)

### Print NMA results
net

### Forest plot with NMA results
forest(net)

### Forest plot showing all network estimates of active treatments
### compared with other treatments
forest(net, ref = c("C", "D", "M"), baseline = FALSE, drop = TRUE)

### Treatment ranking using P-scores
netrank(net)

### Rankogram with all ranking probabilities
set.seed(1909)
ran &lt;- rankogram(net)
ran
plot(ran)

### Treatment ranking using SUCRAs
netrank(ran)

### League table showing network and direct estimates
netleague(net, seq = netrank(net), ci = FALSE)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.tannersmith2016'>Studies on the Relationship between School Motivation and Criminal Behavior</h2><span id='topic+dat.tannersmith2016'></span>

<h3>Description</h3>

<p>Results from 17 studies on the correlation between school motivation/attitudes and subsequent delinquent/criminal behavior.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.tannersmith2016
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>studyid</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study identifier </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>yi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> r-to-z transformed correlation coefficient </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>vi</b>       </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding sampling variance </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sei</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> corresponding standard error </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>aget1</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> age at which the school motivation/attitudes were assessed </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>aget2</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> age at which the delinquent/criminal behavior was assessed </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>propmale</b> </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> proportion of male participants in the sample </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sexmix</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the sample consisted only of males, only of females, or a mix
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset includes 113 r-to-z transformed correlation coefficients from 17 prospective longitudinal studies that examined the relationship between school motivation/attitudes and subsequent delinquent/criminal behavior.
</p>
<p>Multiple coefficients could be extracted from the studies &ldquo;given the numerous ways in which school motivation/attitudes variables could be operationalized (e.g., academic aspirations, academic self-efficacy) as well as the numerous ways in which crime/delinquency could be operationalized (e.g., property crime, violent crime)&rdquo; (Tanner-Smith et al., 2016).
</p>
<p>Since information to compute the covariance between multiple coefficients within studies is not available, Tanner-Smith et al. (2016) illustrate the use of cluster-robust inference methods for the analysis of this dataset.
</p>
<p>Note that this dataset is only meant to be used for pedagogical and demonstration purposes and does not constitute a proper review or synthesis of the complete and current research evidence on the given topic.
</p>


<h3>Concepts</h3>

<p>psychology, criminology, correlation coefficients, multilevel models, cluster-robust inference, meta-regression
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Tanner-Smith, E. E., Tipton, E. &amp; Polanin, J. R. (2016). Handling complex meta-analytic data structures using robust variance estimates: A tutorial in R. <em>Journal of Developmental and Life-Course Criminology</em>, <b>2</b>(1), 85&ndash;112. <code style="white-space: pre;">&#8288;https://doi.org/10.1007/s40865-016-0026-5&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.tannersmith2016
head(dat)

## Not run: 

### load metafor package
library(metafor)

### compute mean age variables within studies
dat$aget1 &lt;- ave(dat$aget1, dat$studyid)
dat$aget2 &lt;- ave(dat$aget2, dat$studyid)

### construct an effect size identifier variable
dat$esid &lt;- 1:nrow(dat)

### construct an approximate var-cov matrix assuming a correlation of 0.8
### for multiple coefficients arising from the same study
V &lt;- vcalc(vi, cluster=studyid, obs=esid, rho=0.8, data=dat)

### fit a multivariate random-effects model using the approximate var-cov matrix V
res &lt;- rma.mv(yi, V, random = ~ esid | studyid, data=dat)
res

### use cluster-robust inference methods
robust(res, cluster=studyid, clubSandwich=TRUE)

### note: the results obtained above and below are slightly different compared
### to those given by Tanner-Smith et al. (2016) since the approach illustrated
### here makes use a multivariate random-effects model for the 'working model'
### before applying the cluster-robust inference methods, while the results given
### in the paper are based on a somewhat simpler working model

### examine the main effects of the age variables
res &lt;- rma.mv(yi, V, mods = ~ aget1 + aget2,
              random = ~ 1 | studyid/esid, data=dat)
robust(res, cluster=studyid, clubSandwich=TRUE)

### also examine their interaction
res &lt;- rma.mv(yi, V, mods = ~ aget1 * aget2,
              random = ~ 1 | studyid/esid, data=dat)
robust(res, cluster=studyid, clubSandwich=TRUE)

### add the sexmix factor to the model
res &lt;- rma.mv(yi, V, mods = ~ aget1 * aget2 + sexmix,
              random = ~ 1 | studyid/esid, data=dat)
robust(res, cluster=studyid, clubSandwich=TRUE)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.vanhowe1999'>Studies on the Association between Circumcision and HIV Infection</h2><span id='topic+dat.vanhowe1999'></span>

<h3>Description</h3>

<p>Results from 33 studies examining the association between male circumcision and HIV infection. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.vanhowe1999
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>category</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study type (high-risk group, partner study, or population survey) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>non.pos</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of non-circumcised HIV positive cases </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>non.neg</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of non-circumcised HIV negative cases </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cir.pos</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of circumcised HIV positive cases </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>cir.neg</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of circumcised HIV negative cases
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 33 studies provide data in terms of \(2 \times 2\) tables in the form:
</p>

<table>
<tr>
 <td style="text-align: left;">
                   </td><td style="text-align: center;"> HIV positive   </td><td style="text-align: center;"> HIV negative </td>
</tr>
<tr>
 <td style="text-align: left;">
   non-circumcised </td><td style="text-align: center;"> <code>non.pos</code> </td><td style="text-align: center;"> <code>non.neg</code> </td>
</tr>
<tr>
 <td style="text-align: left;">
   circumcised     </td><td style="text-align: center;"> <code>cir.pos</code> </td><td style="text-align: center;"> <code>cir.neg</code>
   </td>
</tr>

</table>
<p> The goal of the meta-analysis was to examine if the risk of an HIV infection differs between non-circumcised versus circumcised men.
</p>
<p>The dataset is interesting because it can be used to illustrate the difference between naively pooling results by summing up the counts across studies and then computing the odds ratio based on the aggregated table (as was done by Van Howe, 1999) and conducting a proper meta-analysis (as illustrated by O'Farrell &amp; Egger, 2000). In fact, a proper meta-analysis shows that the HIV infection risk is on average higher in non-circumcised men, which is the opposite of what the naive pooling approach yields (which makes this an illustration of Simpson's paradox).
</p>


<h3>Concepts</h3>

<p>medicine, epidemiology, odds ratios
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Van Howe, R. S. (1999). Circumcision and HIV infection: Review of the literature and meta-analysis. <em>International Journal of STD &amp; AIDS</em>, <b>10</b>(1), 8&ndash;16. <code style="white-space: pre;">&#8288;https://doi.org/10.1258/0956462991913015&#8288;</code>
</p>


<h3>References</h3>

<p>O'Farrell, N., &amp; Egger, M. (2000). Circumcision in men and the prevention of HIV infection: A 'meta-analysis' revisited. <em>International Journal of STD &amp; AIDS</em>, <b>11</b>(3), 137&ndash;142. <code style="white-space: pre;">&#8288;https://doi.org/10.1258/0956462001915480&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.vanhowe1999
dat

## Not run: 

### load metafor package
library(metafor)

### naive pooling by summing up the counts within categories and then
### computing the odds ratios and corresponding confidence intervals
cat1 &lt;- with(dat[dat$category=="high-risk group",],
   escalc(measure="OR", ai=sum(non.pos), bi=sum(non.neg), ci=sum(cir.pos), di=sum(cir.neg)))
cat2 &lt;- with(dat[dat$category=="partner study",],
   escalc(measure="OR", ai=sum(non.pos), bi=sum(non.neg), ci=sum(cir.pos), di=sum(cir.neg)))
cat3 &lt;- with(dat[dat$category=="population survey",],
   escalc(measure="OR", ai=sum(non.pos), bi=sum(non.neg), ci=sum(cir.pos), di=sum(cir.neg)))
summary(cat1, transf=exp, digits=2)
summary(cat2, transf=exp, digits=2)
summary(cat3, transf=exp, digits=2)

### naive pooling across all studies
all &lt;- escalc(measure="OR", ai=sum(dat$non.pos), bi=sum(dat$non.neg),
                            ci=sum(dat$cir.pos), di=sum(dat$cir.neg))
summary(all, transf=exp, digits=2)

### calculate log odds ratios and corresponding sampling variances
dat &lt;- escalc(measure="OR", ai=non.pos, bi=non.neg, ci=cir.pos, di=cir.neg, data=dat)
dat

### random-effects model
res &lt;- rma(yi, vi, data=dat, method="DL")
res
predict(res, transf=exp, digits=2)

### random-effects model within subgroups
res &lt;- rma(yi, vi, data=dat, method="DL", subset=category=="high-risk group")
predict(res, transf=exp, digits=2)
res &lt;- rma(yi, vi, data=dat, method="DL", subset=category=="partner study")
predict(res, transf=exp, digits=2)
res &lt;- rma(yi, vi, data=dat, method="DL", subset=category=="population survey")
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.viechtbauer2021'>Studies to Illustrate Model Checking Methods</h2><span id='topic+dat.viechtbauer2021'></span>

<h3>Description</h3>

<p>Results from 20 hypothetical randomized clinical trials examining the effectiveness of a medication for treating some disease.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.viechtbauer2021
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>trial</b>  </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> trial number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nTi</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>nCi</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xTi</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the treatment group with remission </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>xCi</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in the control group with remission </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>dose</b>   </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> dosage of the medication provided to patients in the treatment group (in milligrams per day)
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset was constructed for the purposes of illustrating the model checking and diagnostic methods described in Viechtbauer (2021). The code below provides the results for many of the analyses and plots discussed in the book chapter.
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, outliers, model checks
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Viechtbauer, W. (2021). Model checking in meta-analysis. In C. H. Schmid, T. Stijnen, &amp; I. R. White (Eds.), <em>Handbook of meta-analysis</em> (pp. 219-254). Boca Raton, FL: CRC Press. <code style="white-space: pre;">&#8288;https://doi.org/10.1201/9781315119403&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.viechtbauer2021
dat

## Not run: 

### load metafor package
library(metafor)

### calculate log odds ratios and corresponding sampling variances

dat &lt;- escalc(measure="OR", ai=xTi, n1i=nTi, ci=xCi, n2i=nCi, add=1/2, to="all", data=dat)
dat

### number of studies

k &lt;- nrow(dat)

### fit models

res.CE &lt;- rma(yi, vi, data=dat, method="CE") # same as method="EE"
res.CE

res.RE &lt;- rma(yi, vi, data=dat, method="DL")
res.RE

res.MR &lt;- rma(yi, vi, mods = ~ dose, data=dat, method="FE")
res.MR

res.ME &lt;- rma(yi, vi, mods = ~ dose, data=dat, method="DL")
res.ME

### forest and bubble plot

par(mar=c(5,4,1,2))

forest(dat$yi, dat$vi, psize=0.8, efac=0, xlim=c(-4,6), ylim=c(-3,23),
       cex=1, width=c(5,5,5), xlab="Log Odds Ratio (LnOR)")
addpoly(res.CE, row=-1.5, mlab="CE Model")
addpoly(res.RE, row=-2.5, mlab="RE Model")
text(-4, 22, "Trial",         pos=4, font=2)
text( 6, 22, "LnOR [95% CI]", pos=2, font=2)
abline(h=0)

tmp &lt;- regplot(res.ME, xlim=c(0,250), ylim=c(-1,1.5), predlim=c(0,250), shade=FALSE, digits=1,
               xlab="Dosage (mg per day)", psize="seinv", plim=c(NA,5), bty="l", las=1,
               lty=c("solid", "dashed"), label=TRUE, labsize=0.8, offset=c(1,0.7))
res.sub &lt;- rma(yi, vi, mods = ~ dose, data=dat, method="DL", subset=-6)
abline(res.sub, lty="dotted")
points(tmp$xi, tmp$yi, pch=21, cex=tmp$psize, col="black", bg="darkgray")

par(mar=c(5,4,4,2))

### number of standardized deleted residuals larger than +-1.96 in each model

sum(abs(rstudent(res.CE)$z) &gt;= qnorm(.975))
sum(abs(rstudent(res.MR)$z) &gt;= qnorm(.975))
sum(abs(rstudent(res.RE)$z) &gt;= qnorm(.975))
sum(abs(rstudent(res.ME)$z) &gt;= qnorm(.975))

### plot of the standardized deleted residuals for the RE and ME models

plot(NA, NA, xlim=c(1,20), ylim=c(-4,4), xlab="Study", ylab="Standardized (Deleted) Residual",
     xaxt="n", main="Random-Effects Model", las=1)
axis(side=1, at=1:20)
abline(h=c(-1.96,1.96), lty="dotted")
abline(h=0)
points(1:20, rstandard(res.RE)$z, type="o", pch=19, col="gray70")
points(1:20, rstudent(res.RE)$z,  type="o", pch=19)
legend("top", pch=19, col=c("gray70","black"), lty="solid",
       legend=c("Standardized Residuals","Standardized Deleted Residuals"), bty="n")

plot(NA, NA, xlim=c(1,20), ylim=c(-4,4), xlab="Study", ylab="Standardized (Deleted) Residual",
     xaxt="n", main="Mixed-Effects Model", las=1)
axis(side=1, at=1:20)
abline(h=c(-1.96,1.96), lty="dotted")
abline(h=0)
points(1:20, rstandard(res.ME)$z, type="o", pch=19, col="gray70")
points(1:20, rstudent(res.ME)$z,  type="o", pch=19)
legend("top", pch=19, col=c("gray70","black"), lty="solid",
       legend=c("Standardized Residuals","Standardized Deleted Residuals"), bty="n")

### Baujat plots

baujat(res.CE, main="Common-Effects Model", xlab="Squared Pearson Residual", ylim=c(0,5), las=1)
baujat(res.ME, main="Mixed-Effects Model", ylim=c(0,2), las=1)

### GOSH plots (skipped because this takes quite some time to run)

if (FALSE) {

res.GOSH.CE &lt;- gosh(res.CE, subsets=10^7)
plot(res.GOSH.CE, cex=0.2, out=6, xlim=c(-0.25,1.25), breaks=c(200,100))

res.GOSH.ME &lt;- gosh(res.ME, subsets=10^7)
plot(res.GOSH.ME, het="tau2", out=6, breaks=50, adjust=0.6, las=1)

}

### plot of treatment dosage against the standardized residuals

plot(dat$dose, rstandard(res.ME)$z, pch=19, xlab="Dosage (mg per day)",
     ylab="Standardized Residual", xlim=c(0,250), ylim=c(-2.5,2.5), las=1)
abline(h=c(-1.96,1.96), lty="dotted", lwd=2)
abline(h=0)
title("Standardized Residual Plot")
text(dat$dose[6], rstandard(res.ME)$z[6], "6", pos=4, offset=0.4)

### quadratic polynomial model

rma(yi, vi, mods = ~ dose + I(dose^2), data=dat, method="DL")

### lack-of-fit model

resLOF &lt;- rma(yi, vi, mods = ~ dose + factor(dose), data=dat, method="DL", btt=3:9)
resLOF

### scatter plot to illustrate the lack-of-fit model

regplot(res.ME, xlim=c(0,250), ylim=c(-1.0,1.5), xlab="Dosage (mg per day)", ci=FALSE,
        predlim=c(0,250), psize=1, pch=19, col="gray60", digits=1, lwd=1, bty="l", las=1)
dosages &lt;- sort(unique(dat$dose))
lines(dosages, fitted(resLOF)[match(dosages, dat$dose)], type="o", pch=19, cex=2, lwd=2)
points(dat$dose, dat$yi, pch=19, col="gray60")
legend("bottomright", legend=c("Linear Model", "Lack-of-Fit Model"), pch=c(NA,19), col="black",
       lty="solid", lwd=c(1,2), pt.cex=c(1,2), seg.len=4, bty="n")

### checking normality of the standardized deleted residuals

qqnorm(res.ME, type="rstudent", main="Standardized Deleted Residuals", pch=19, label="out",
       lwd=2, pos=24, ylim=c(-4,3), lty=c("solid", "dotted"), las=1)

### checking normality of the random effects

sav &lt;- qqnorm(ranef(res.ME)$pred, main="BLUPs of the Random Effects", cex=1, pch=19,
              xlim=c(-2.2,2.2), ylim=c(-0.6,0.6), las=1)
abline(a=0, b=sd(ranef(res.ME)$pred), lwd=2)
text(sav$x[6], sav$y[6], "6", pos=4, offset=0.4)

### hat values for the CE and RE models

plot(NA, NA, xlim=c(1,20), ylim=c(0,0.21), xaxt="n", las=1, xlab="Study", ylab="Hat Value")
axis(1, 1:20, cex.axis=1)
points(hatvalues(res.CE), type="o", pch=19, col="gray70")
points(hatvalues(res.RE), type="o", pch=19)
abline(h=1/20, lty="dotted", lwd=2)
title("Hat Values for the CE/RE Models")
legend("topright", pch=19, col=c("gray70","black"), lty="solid",
       legend=c("Common-Effects Model", "Random-Effects Model"), bty="n")

### heatmap of the hat matrix for the ME model

cols &lt;- colorRampPalette(c("blue", "white", "red"))(101)
h &lt;- hatvalues(res.ME, type="matrix")
image(1:nrow(h), 1:ncol(h), t(h[nrow(h):1,]), axes=FALSE,
      xlab="Influence of the Observed Effect of Study ...", ylab="On the Fitted Value of Study ...",
      col=cols, zlim=c(-max(abs(h)),max(abs(h))))
axis(1, 1:20, tick=FALSE)
axis(2, 1:20, labels=20:1, las=1, tick=FALSE)
abline(h=seq(0.5,20.5,by=1), col="white")
abline(v=seq(0.5,20.5,by=1), col="white")
points(1:20, 20:1, pch=19, cex=0.4)
title("Heatmap for the Mixed-Effects Model")

### plot of leverages versus standardized residuals for the ME model

plot(hatvalues(res.ME), rstudent(res.ME)$z, pch=19, cex=0.2+3*sqrt(cooks.distance(res.ME)),
     las=1, xlab="Leverage (Hat Value)", ylab="Standardized Deleted Residual",
     xlim=c(0,0.35), ylim=c(-3.5,2.5))
abline(h=c(-1.96,1.96), lty="dotted", lwd=2)
abline(h=0, lwd=2)
ids &lt;- c(3,6,9)
text(hatvalues(res.ME)[ids] + c(0,0.013,0.010), rstudent(res.ME)$z[ids] - c(0.18,0,0), ids)
title("Leverage vs. Standardized Deleted Residuals")

### plot of the Cook's distances for the ME model

plot(1:20, cooks.distance(res.ME), ylim=c(0,1.6), type="o", pch=19, las=1, xaxt="n", yaxt="n",
     xlab="Study", ylab="Cook's Distance")
axis(1, 1:20, cex.axis=1)
axis(2, seq(0,1.6,by=0.4), las=1)
title("Cook's Distances")

### plot of the leave-one-out estimates of tau^2 for the ME model

x &lt;- influence(res.ME)

plot(1:20, x$inf$tau2.del,  ylim=c(0,0.15), type="o", pch=19, las=1, xaxt="n", xlab="Study",
     ylab=expression(paste("Estimate of ", tau^2, " without the ", italic(i), "th study")))
abline(h=res.ME$tau2, lty="dashed")
axis(1, 1:20)
title("Residual Heterogeneity Estimates")

### plot of the covariance ratios for the ME model

plot(1:20, x$inf$cov.r,  ylim=c(0,2.0), type="o", pch=19, las=1, xaxt="n",
     xlab="Study", ylab="Covariance Ratio")
abline(h=1, lty="dashed")
axis(1, 1:20)
title("Covariance Ratios")

### fit mixed-effects model without studies 3 and/or 6

rma(yi, vi, mods = ~ dose, data=dat, method="DL", subset=-3)
rma(yi, vi, mods = ~ dose, data=dat, method="DL", subset=-6)
rma(yi, vi, mods = ~ dose, data=dat, method="DL", subset=-c(3,6))


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.white2020'>Studies on the Relationship between Sexual Signal Expression and Individual Quality</h2><span id='topic+dat.white2020'></span>

<h3>Description</h3>

<p>Results from 41 studies examining the relationship between measures of individual quality and the expression of structurally coloured sexual signals.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.white2020
</code></pre>


<h3>Format</h3>

<p>The object is a data frame which contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>study_id</b>        </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> study-level ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>obs</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> observation-level ID </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>exp_obs</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the study is observational or experimental </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>control</b>         </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> whether the study did (1) or did not (0) include a non-sexual control trait </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>class</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> class of the study organisms </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>genus</b>           </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> class of the study organisms </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>species</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> species of the study organisms </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>sex</b>             </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> sex of the study organisms </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>iridescent</b>      </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> whether the colour signals were iridescent (1) or not (0) </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>col_var</b>         </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> the colour variable quantified </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>col_component</b>   </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> whether the colour variable is chromatic or achromatic </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>quality_measure</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> the measure of individual quality used </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>region</b>          </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> the body region from which colour was sampled </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> study sample size </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>r</b>               </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> Pearson's correlation coefficient </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Details</h3>

<p>The 186 rows in this dataset come from 41 experimental and observational studies reporting on the correlation between measures of individual quality (age, body condition, immune function, parasite resistance) and the expression of structurally coloured sexual signals across 28 species. The purpose of this meta-analysis was to test whether structural colour signals show heightened condition-dependent expression, as predicted by evolutionary models of 'honest' signalling.
</p>


<h3>Concepts</h3>

<p>ecology, evolution, correlation coefficients
</p>


<h3>Author(s)</h3>

<p>Thomas E. White, <a href="mailto:thomas.white@sydney.edu.au">thomas.white@sydney.edu.au</a>
</p>


<h3>Source</h3>

<p>White, T. E. (2020). Structural colours reflect individual quality: A meta-analysis. <em>Biology Letters</em>, <b>16</b>(4), 20200001. <code style="white-space: pre;">&#8288;https://doi.org/10.1098/rsbl.2020.0001&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat' and examine data
dat &lt;- dat.white2020
head(dat, 10)

## Not run: 

### load metafor package
library(metafor)

### calculate r-to-z transformed correlations and corresponding sampling variances
dat &lt;- escalc(measure="ZCOR", ri=r, ni=n, data=dat)

### fit multilevel meta-analytic model
res &lt;- rma.mv(yi, vi, random = list(~ 1 | study_id, ~ 1 | obs), data=dat)
res


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.woods2010'>Studies on Treatments for Chronic Obstructive Pulmonary Disease</h2><span id='topic+dat.woods2010'></span>

<h3>Description</h3>

<p>Results from 3 trials examining the mortality risk of three treatments and placebo in patients with chronic obstructive pulmonary disease.</p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.woods2010
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>author</b>    </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> first author / study name </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>treatment</b> </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> treatment </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>r</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of deaths </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>N</b>         </td><td style="text-align: left;"> <code>integer</code>   </td><td style="text-align: left;"> number of patients
</td>
</tr>

</table>



<h3>Details</h3>

<p>Count mortality statistics in randomised controlled trials of treatments for chronic obstructive pulmonary disease (Woods et al., 2010, Table 1).
</p>


<h3>Concepts</h3>

<p>medicine, odds ratios, network meta-analysis
</p>


<h3>Author(s)</h3>

<p>Guido Schwarzer, <a href="mailto:sc@imbi.uni-freiburg.de">sc@imbi.uni-freiburg.de</a>, <a href="https://github.com/guido-s/">https://github.com/guido-s/</a>
</p>


<h3>Source</h3>

<p>Woods, B. S., Hawkins, N., &amp; Scott, D. A. (2010). Network meta-analysis on the log-hazard scale, combining count and hazard ratio statistics accounting for multi-arm trials: A tutorial. <em>BMC Medical Research Methodology</em>, <b>10</b>, 54. <code style="white-space: pre;">&#8288;https://doi.org/10.1186/1471-2288-10-54&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Show full data set
dat.woods2010

## Not run: 

### Load netmeta package
suppressPackageStartupMessages(library(netmeta))

### Print odds ratios and confidence limits with two digits
settings.meta(digits = 2)

### Change appearance of confidence intervals
cilayout("(", "-")

### Transform data from long arm-based format to contrast-based
### format. Argument 'sm' has to be used for odds ratio as summary
### measure; by default the risk ratio is used in the metabin function
### called internally.
pw &lt;- pairwise(treatment, event = r, n = N,
  studlab = author, data = dat.woods2010, sm = "OR")
pw

### Conduct network meta-analysis
net &lt;- netmeta(pw)
net

### Show forest plot
forest(net, ref = "Placebo", drop = TRUE,
  leftlabs = "Contrast to Placebo")


## End(Not run)
</code></pre>

<hr>
<h2 id='dat.yusuf1985'>Studies of Beta Blockers During and After Myocardial Infarction</h2><span id='topic+dat.yusuf1985'></span>

<h3>Description</h3>

<p>Results from studies examining the effectiveness of beta blockers for reducing mortality and reinfarction. <script id="MathJax-script" async src="../../mathjaxr/doc/mathjax/es5/tex-chtml-full.js"></script></p>


<h3>Usage</h3>

<pre><code class='language-R'>dat.yusuf1985
</code></pre>


<h3>Format</h3>

<p>The data frame contains the following columns:
</p>

<table>
<tr>
 <td style="text-align: left;">
<b>table</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> table number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>id</b>     </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> trial id number </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>trial</b>  </td><td style="text-align: left;"> <code>character</code> </td><td style="text-align: left;"> trial name or first author </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ai</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths/reinfarctions in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n1i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in treatment group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>ci</b>     </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of deaths/reinfarctions in control group </td>
</tr>
<tr>
 <td style="text-align: left;">
<b>n2i</b>    </td><td style="text-align: left;"> <code>numeric</code>   </td><td style="text-align: left;"> number of patients in control group
</td>
</tr>

</table>



<h3>Details</h3>

<p>The dataset contains table 6 (total mortality from short-term trials of oral beta blockers), 9 (total mortality at one week from trials with an initial IV dose of a beta blocker), 10 (total mortality from long-term trials with treatment starting late and mortality from day 8 onwards in long-term trials that began early and continued after discharge), 11 (nonfatal reinfarction from long-term trials of beta blockers), 12a (sudden death in long-term beta blocker trials), and 12b (nonsudden death in long-term beta blocker trials) from the meta-analysis by Yusuf et al. (1985) on the effectiveness of of beta blockers for reducing mortality and reinfarction.
</p>
<p>The article also describes what is sometimes called Peto's one-step method for meta-analyzing \(2 \times 2\) table data. This method is implemented in the <code><a href="metafor.html#topic+rma.peto">rma.peto</a></code> function.
</p>


<h3>Concepts</h3>

<p>medicine, cardiology, odds ratios, Peto's method
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>, <a href="https://www.metafor-project.org">https://www.metafor-project.org</a>
</p>


<h3>Source</h3>

<p>Yusuf, S., Peto, R., Lewis, J., Collins, R., &amp; Sleight, P. (1985). Beta blockade during and after myocardial infarction: An overview of the randomized trials. <em>Progress in Cardiovascular Disease</em>, <b>27</b>(5), 335&ndash;371. <code style="white-space: pre;">&#8288;https://doi.org/10.1016/s0033-0620(85)80003-7&#8288;</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### copy data into 'dat'
dat &lt;- dat.yusuf1985
dat[dat$table == 6,]

## Not run: 

### load metafor package
library(metafor)

### to select a table for the analysis
tab &lt;- "6" # either: 6, 9, 10, 11, 12a, 12b

### to double-check total counts as reported in article
apply(dat[dat$table==tab,4:7], 2, sum, na.rm=TRUE)

### meta-analysis using Peto's one-step method
res &lt;- rma.peto(ai=ai, n1i=n1i, ci=ci, n2i=n2i, data=dat, subset=(table==tab))
res
predict(res, transf=exp, digits=2)


## End(Not run)
</code></pre>

<hr>
<h2 id='datsearch'>Search Function for the Datasets</h2><span id='topic+datsearch'></span>

<h3>Description</h3>

<p>Function to search among the existing datasets.</p>


<h3>Usage</h3>

<pre><code class='language-R'>datsearch(pattern, concept=TRUE, matchall=TRUE, fixed=TRUE, pkgdown=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="datsearch_+3A_pattern">pattern</code></td>
<td>
<p>character string or vector of strings specifying the terms to search for within the datasets. Can also be left unspecified to start the function in an interactive mode.</p>
</td></tr>
<tr><td><code id="datsearch_+3A_concept">concept</code></td>
<td>
<p>logical indicating whether the search should be confined to the concept terms (<code>TRUE</code> by default) or whether a full-text search should be conducted.</p>
</td></tr>
<tr><td><code id="datsearch_+3A_matchall">matchall</code></td>
<td>
<p>logical indicating whether only the datasets matching all terms (if multiple are specified) are returned (<code>TRUE</code> by default) or whether datasets matching any one of the terms are returned.</p>
</td></tr>
<tr><td><code id="datsearch_+3A_fixed">fixed</code></td>
<td>
<p>logical indicating whether a term is a string to be matched as is (<code>TRUE</code> by default). If <code>FALSE</code>, a search term is a regular expression that <code><a href="base.html#topic+grep">grep</a></code> will search for. Only relevant when <code>concept=FALSE</code> (i.e., when doing a full-text search).</p>
</td></tr>
<tr><td><code id="datsearch_+3A_pkgdown">pkgdown</code></td>
<td>
<p>logical indicating whether the standard help file or the pkgdown docs (at <a href="https://wviechtb.github.io/metadat/">https://wviechtb.github.io/metadat/</a>) should be shown for a chosen dataset (<code>FALSE</code> by default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can be used to search all existing datasets in the <span class="pkg">metadat</span> package based on their concept terms (see below) or based on a full-text search of their corresponding help files.
</p>
<p>When running <code>datsearch()</code> without the <code>pattern</code> argument specified, the function starts in an interactive mode and prompts for one or multiple search terms.
</p>
<p>Alternatively, one can specify a single search term via the <code>pattern</code> argument or multiple search terms by using a string vector as the <code>pattern</code> or by separating multiple search terms in a single string with &lsquo;,&rsquo;, &lsquo;;&rsquo;, or &lsquo;and&rsquo;.
</p>
<p>If <code>matchall=TRUE</code> (the default), only datasets matching all search terms (if multiple are specified) are returned. If <code>matchall=FALSE</code>, datasets matching any one of the search terms are returned.
</p>
<p>If a single match is found, the corresponding help file is directly shown. If multiple matches are found, the user is prompted to choose one of the matching datasets of interest.
</p>
<p><b>Concept Terms</b>
</p>
<p>Each dataset is tagged with one or multiple concept terms that refer to various aspects of a dataset, such as the field/topic of research, the outcome measure used for the analysis, the model(s) used for analyzing the data, and the methods/concepts that can be illustrated with the dataset.
</p>

<ul>
<li><p> In terms of &lsquo;fields/topics&rsquo;, the following terms have been used at least once: alternative medicine, attraction, cardiology, climate change, covid-19, criminology, dentistry, ecology, education, engineering, epidemiology, evolution, genetics, human factors, medicine, memory, obstetrics, oncology, persuasion, primary care, psychiatry, psychology, smoking, social work, sociology.
</p>
</li>
<li><p> In terms of &lsquo;outcome measures&rsquo;, the following terms have been used at least once: correlation coefficients, Cronbach's alpha, hazard ratios, incidence rates, raw mean differences, odds ratios, proportions, ratios of means, raw means, risk differences, risk ratios, (semi-)partial correlations, standardized mean changes, standardized mean differences.
</p>
</li>
<li><p> In terms of &lsquo;models/methods/concepts&rsquo;, the following terms have been used at least once: cluster-robust inference, component network meta-analysis, cumulative meta-analysis, diagnostic accuracy studies, dose response models, generalized linear models, longitudinal models, Mantel-Haenszel method, meta-regression, model checks, multilevel models, multivariate models, network meta-analysis, outliers, Peto's method, phylogeny, publication bias, reliability generalization, single-arm studies, spatial correlation.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Daniel Noble, <a href="mailto:daniel.noble@anu.edu.au">daniel.noble@anu.edu.au</a> <br />
Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># note: the examples below are not run since they require interactivity

if (FALSE) {

   # start the function in the interactive mode
   datsearch()

   # find all datasets tagged with the concept term 'standardized mean differences'
   datsearch("standardized mean differences")

   # find all datasets tagged with the concept terms 'odds ratio' and 'multilevel'
   datsearch("odds ratio, multilevel")

   # do a full-text search for the term 'infarct'
   datsearch("infarct", concept=FALSE)

   # do a full-text search for 'rma.mv(' (essentially finds all datasets where
   # the rma.mv() function was used in the examples section of a help file)
   datsearch("rma.mv(", concept=FALSE)

}
</code></pre>

<hr>
<h2 id='metadat-package'>Meta-Analysis Datasets for R</h2><span id='topic+metadat-package'></span><span id='topic+metadat'></span>

<h3>Description</h3>

<p>The <span class="pkg">metadat</span> package contains a large collection of meta-analysis datasets. These datasets are useful for teaching purposes, illustrating/testing meta-analytic methods, and validating published analyses.
</p>


<h3>Browsing and Searching for Datasets</h3>

<p>A listing of all datasets in the package can be obtained with <code>help(package=metadat)</code>. Each datasets is also tagged with one or multiple concept terms. These concept terms refer to various aspects of a dataset, such as the field/topic of research, the outcome measure used for the analysis, the model(s) used for analyzing the data, and the methods/concepts that can be illustrated with the dataset. The <code><a href="#topic+datsearch">datsearch</a></code> function can be used to search among the existing datasets in the package based on their concept terms or based on a full-text search of their corresponding help files.
</p>
<p>You can also read the documentation online at <a href="https://wviechtb.github.io/metadat/">https://wviechtb.github.io/metadat/</a> (where the output from the example analyses corresponding to each dataset is provided).
</p>


<h3>Contributing New Datasets</h3>

<p>We welcome contributions of new datasets to the package. For each dataset, there must be a citable reference, ideally in a peer-reviewed journal or publication. The general workflow for contributing a new dataset is as follows:
</p>

<ul>
<li><p> Install the <code>metadat</code> package in R in the usual manner (i.e., <code>install.packages("metadat")</code>).
</p>
</li>
<li><p> If you are familiar with Git/GitHub and making pull requests, fork the <a href="https://github.com/wviechtb/metadat">package repository</a>. Otherwise, <a href="https://github.com/wviechtb/metadat/archive/master.zip">download</a> the source version of the package from GitHub and unzip the file to some directory on your computer.
</p>
</li>
<li><p> Place the raw data (in a non-binary format) in the <code>data-raw</code> directory. The file should be named <code>dat.&lt;author&gt;&lt;year&gt;.&lt;ext&gt;</code>, where <code>&lt;author&gt;</code> is the last name of the first author of the publication from which the data come, <code>&lt;year&gt;</code> is the publication year, and <code>&lt;ext&gt;</code> is the file extension (e.g., <code>.txt</code>, <code>.csv</code>).
</p>
</li>
<li><p> Place a corresponding R script in the <code>data-raw</code> directory named <code>dat.&lt;author&gt;&lt;year&gt;.r</code> that reads in the data, possibly does some data cleaning/processing, and then saves the dataset to the <code>data</code> directory (using <code><a href="base.html#topic+save">save</a></code>), with name <code>dat.&lt;author&gt;&lt;year&gt;.rda</code>.
</p>
</li>
<li><p> Start R, load the <code>metadat</code> package (i.e., <code>library(metadat)</code>), and then run the <code><a href="#topic+prep_dat">prep_dat</a></code> function (either set the working directory to the location of the source package beforehand or use the <code>pkgdir</code> argument of the <code><a href="#topic+prep_dat">prep_dat</a></code> function to specify the source package location).
</p>
</li>
<li><p> For a new dataset, this should create a boilerplate template for a corresponding help file in the <code>man</code> directory, named <code>dat.&lt;author&gt;&lt;year&gt;.Rd</code>. Edit the help file, adding the title and a short description of the dataset in general, a description of each variable in the dataset, further details on the dataset (e.g., the field of research, how the data was collected, the purpose of the dataset / what it was used for, the effect size or outcome measure used in the analysis, the types of analyses/models that can be illustrated with the dataset), a reference for the source of the dataset, one or multiple concept terms, the name and email address of the contributor of the dataset, and (optionally) example code to illustrate the analysis of the dataset.
</p>
</li>
<li><p> Either make a pull request (if you are familiar with this workflow) or zip up the <code>dat.&lt;author&gt;&lt;year&gt;.&lt;ext&gt;</code>, <code>dat.&lt;author&gt;&lt;year&gt;.r</code>, <code>dat.&lt;author&gt;&lt;year&gt;.rda</code>, and <code>dat.&lt;author&gt;&lt;year&gt;.Rd</code> files and open up a new <a href="https://github.com/wviechtb/metadat/issues">issue at GitHub</a>, attaching the zip file.
</p>
</li>
<li><p> If the above makes no sense to you, you can also email one of the package authors with a cleaned, raw data file in <code>.txt</code> or <code>.csv</code> format, along with a meta-data file (format doesn't matter) that includes the information described above.
</p>
</li></ul>



<h3>Citing the Package</h3>

<p>If you use these data, please cite both the <span class="pkg">metadat</span> package (see <code>citation("metadat")</code> for the reference) and the original source of the data as given under the help file of a dataset.
</p>


<h3>Bug/Error Reports</h3>

<p>If you think you have found an error in an existing dataset or a bug in the package in general, please go to <a href="https://github.com/wviechtb/metadat/issues">https://github.com/wviechtb/metadat/issues</a> and open up a new issue.
</p>


<h3>Author(s)</h3>

<p>Thomas White, <a href="mailto:thomas.white@sydney.edu.au">thomas.white@sydney.edu.au</a> <br />
Daniel Noble, <a href="mailto:daniel.noble@anu.edu.au">daniel.noble@anu.edu.au</a> <br />
Alistair Senior, <a href="mailto:alistair.senior@sydney.edu.au">alistair.senior@sydney.edu.au</a> <br />
W. Kyle Hamilton, <a href="mailto:whamilton@ucmerced.edu">whamilton@ucmerced.edu</a> <br />
Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>
</p>

<hr>
<h2 id='prep_dat'>Data preparation function</h2><span id='topic+prep_dat'></span>

<h3>Description</h3>

<p>Function to run data processing scripts.</p>


<h3>Usage</h3>

<pre><code class='language-R'>prep_dat(rebuild=FALSE, overwrite, pkgdir)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prep_dat_+3A_rebuild">rebuild</code></td>
<td>
<p>logical indicating whether the entire database should be rebuild (default is <code>FALSE</code>).</p>
</td></tr>
<tr><td><code id="prep_dat_+3A_overwrite">overwrite</code></td>
<td>
<p>character vector with one or more <code>.Rd</code> filenames to overwrite (if they already exist). The default is to never overwrite any existing <code>.Rd</code> files.</p>
</td></tr>
<tr><td><code id="prep_dat_+3A_pkgdir">pkgdir</code></td>
<td>
<p>character string specifying the root directory of the source package (if unspecified, the current working directory is assumed to be the package root directory).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function is only for used for processing new datasets for inclusion in the package. It should be used as described on the <span class="pkg"><a href="#topic+metadat-package">metadat-package</a></span> help page.
</p>


<h3>Author(s)</h3>

<p>Wolfgang Viechtbauer, <a href="mailto:wvb@metafor-project.org">wvb@metafor-project.org</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
