<!DOCTYPE html><html lang="en"><head><title>Help for package metaprotr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metaprotr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_kegg'><p>add_kegg</p></a></li>
<li><a href='#add_taxonomy'><p>add_taxonomy</p></a></li>
<li><a href='#crumble_taxonomy'><p>crumble_taxonomy</p></a></li>
<li><a href='#export_ipath3'><p>export_ipath3</p></a></li>
<li><a href='#export_robject'><p>export_robject</p></a></li>
<li><a href='#export_vennlists'><p>export_vennlists</p></a></li>
<li><a href='#fecal_waters'><p>fecal waters</p></a></li>
<li><a href='#filter_shared'><p>filter_shared</p></a></li>
<li><a href='#filter_text'><p>filter_text</p></a></li>
<li><a href='#filter_unshared'><p>filter_unshared</p></a></li>
<li><a href='#getsc_specific'><p>getsc_specific</p></a></li>
<li><a href='#identify_differences'><p>identify_differences</p></a></li>
<li><a href='#inspect_sample_elements'><p>inspect_sample_elements</p></a></li>
<li><a href='#load_protspeps'><p>load_protspeps</p></a></li>
<li><a href='#plot_dendocluster'><p>plot_dendocluster</p></a></li>
<li><a href='#plot_fulltaxo'><p>plot_fulltaxo</p></a></li>
<li><a href='#plot_intensities'><p>plot_intensities</p></a></li>
<li><a href='#plot_intensities_ratio'><p>plot_intensities_ratio</p></a></li>
<li><a href='#plot_pca'><p>plot_pca</p></a></li>
<li><a href='#plot_pietaxo'><p>plot_pietaxo</p></a></li>
<li><a href='#plot_stackedtaxo'><p>plot_stackedtaxo</p></a></li>
<li><a href='#plot_venn'><p>plot_venn</p></a></li>
<li><a href='#remove_element'><p>remove_element</p></a></li>
<li><a href='#select_element'><p>select_element</p></a></li>
<li><a href='#species_annot_fw'><p>species_annot_fw</p></a></li>
<li><a href='#species_fw'><p>species_fw</p></a></li>
<li><a href='#venn_methods'><p>venn methods</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Metaproteomics Post-Processing Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2021-01-28</td>
</tr>
<tr>
<td>Author:</td>
<td>Aaron Millan-Oropeza [aut, cre],
  Catherine Juste [aut, ctb],
  Ariane Bassignani [aut, ctb],
  Céline Henry [aut, ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Aaron Millan-Oropeza &lt;aaron.ibt@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
    Set of tools for descriptive analysis of metaproteomics data generated from high-throughput 
    mass spectrometry instruments. These tools allow to cluster peptides and proteins abundance, 
    expressed as spectral counts, and to manipulate them in groups of metaproteins. This information 
    can be represented using multiple visualization functions to portray the global metaproteome 
    landscape and to differentiate samples or conditions, in terms of abundance of metaproteins, 
    taxonomic levels and/or functional annotation. The provided tools allow to implement flexible 
    analytical pipelines that can be easily applied to studies interested in metaproteomics analysis.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://forgemia.inra.fr/pappso/metaprotr">https://forgemia.inra.fr/pappso/metaprotr</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ade4, dendextend, dplyr, ggforce, ggrepel, reshape2, stringr,
tidyverse</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-28 20:20:48 UTC; amillan</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-02-05 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_kegg'>add_kegg</h2><span id='topic+add_kegg'></span>

<h3>Description</h3>

<p>Integrates a database containing the functional annotation of the identified 
metaproteins into a list defined as &quot;spectral_count_object&quot;. The proteins 
from the “spectral_count_object” must contain taxonomic information. 
The functional annotation was obtained from the Kyoto Encyclopedia of 
Genes and Genomes (KEGG) Orthology database. This database contains 
the molecular functions represented in terms of functional orthologs (KO terms).
Check <a href="https://www.genome.jp/kegg/">KEGG</a> for more details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_kegg(
  spectral_count_object,
  annotation_db,
  taxonomic_db,
  metaproteome_origin,
  protein_file,
  peptide_file,
  text_to_filter = "HUMAN",
  taxonomic_levels_allowed = 1
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_kegg_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing the abundance of the elements (groups, subgroups 
or peptides) expressed as spectral counts and organized by 
taxonomic levels. The format of this object is similar to 
that generated from the function &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_annotation_db">annotation_db</code></td>
<td>
<p>Dataframe containing the functional annotation 
of the proteins. This dataframe must contain two variables: 
i) &quot;gene_name&quot;: indicating the same protein names to those present in 
the variable  &quot;Accession&quot; from the &quot;peptides_proteins&quot;, third dataframe in the 
list defined as &quot;spectral_count_object&quot;; and, ii) &quot;ko&quot;: indicating the 
KEGG Orthology code assigned to a given protein. An example can be found in
this <a href="https://zenodo.org/record/3997093">repository</a>.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_taxonomic_db">taxonomic_db</code></td>
<td>
<p>Dataframe containing the taxonomic information 
for each protein. The first column must contain the same identifiers 
of those present in the column &quot;Accession&quot; from the dataframe 
&quot;peptides_proteins&quot; of the &quot;metaproteome_object&quot;. Two additional columns 
have to be present: i) one named &quot;organism&quot; containing the name of the strain
assigned to a given protein; and ii) the other named 
&quot;species.genus.family.order.class.phylum.superkingdom&quot;. 
The taxonomic classification can be obtained from a tool of 
sequences aligment and must be ordered as follows: species, genus, 
family, order, class, phylum and superkingdom. The characters inside must 
be concatenated by a comma without spaces 
(ex. &quot;Streptococcus anginosus,Streptococcus,Streptococcaceae,Lactobacillales,Bacilli,Firmicutes,Bacteria&quot;).
An example can be found in this <a href="https://zenodo.org/record/3997093">repository</a>.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_metaproteome_origin">metaproteome_origin</code></td>
<td>
<p>List defined as &quot;metaproteome_object&quot; generated
from the function 'load_protspeps'.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_protein_file">protein_file</code></td>
<td>
<p>Character indicating the location of a txt 
file containing the list of proteins generated in 
<a href="http://pappso.inrae.fr/bioinfo/xtandempipeline/">X!TandemPipeline</a> 
using an adapted iterative approach described by 
<a href="https://www.theses.fr/2019SORUS043">Bassignani, 2019</a>. 
Separation between columns should be indicated by tabulation. 
For more details regarding data input check 
<a href="https://forgemia.inra.fr/pappso/metaprotr#data-inputs">format examples</a>.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_peptide_file">peptide_file</code></td>
<td>
<p>Character indicating the location of a txt 
file containing peptides abundances expressed as spectral counts.
This file is generated from 
<a href="http://pappso.inrae.fr/bioinfo/xtandempipeline/">X!TandemPipeline</a> 
using an adapted iterative approach described by 
<a href="https://www.theses.fr/2019SORUS043">Bassignani, 2019</a>. 
Separation between columns should be indicated by tabulation. 
For more details regarding data input check 
<a href="https://forgemia.inra.fr/pappso/metaprotr#data-inputs">format examples</a>.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_text_to_filter">text_to_filter</code></td>
<td>
<p>Character containig a part of text to be searched in 
the &quot;Description&quot; of the protein file. All the elements containing this 
character will be removed. The default value was set to &quot;HUMAN&quot;.</p>
</td></tr>
<tr><td><code id="add_kegg_+3A_taxonomic_levels_allowed">taxonomic_levels_allowed</code></td>
<td>
<p>Numeric value indicating the maximal
number of taxonomic levels allowed per spectral group or subgroup (in function
of the type of spectral data). The default value is set to 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list defined as &quot;spectral_count_object&quot; with the functional 
annotation added to the identified proteins. A new column is added to 
the dataframe &quot;peptides_proteins&quot;. Two quality control plot are also generated,
one with the number of taxonomic entities per spectral level and another with
the number of KO terms per spectral level.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Download functional and taxonmical annotation db: https://zenodo.org/record/3997093#.X0UYI6Zb_mE 
meta99_full_taxo &lt;- read.csv2("full_taxonomy_MetaHIT99.tsv", header= TRUE, sep="\t")
kegg_db &lt;- read.csv2("hs_9_9_igc_vs_kegg89.table", header = TRUE, sep = "\t")

# Files with spectral abundance and proteins list from X!Tandempipeline
protein_file &lt;- "your/specific/location/protein_list.txt"
peptide_file &lt;- "your/specific/location/peptide_counting.txt"
metadata_file &lt;- "your/location/metadata.csv"

metaproteome_origin &lt;- load_protspeps(protein_file, peptide_file, metadata_file)

SCsgp_species &lt;- crumble_taxonomy(SC_subgroups, "species")

SCsgp_species_annot &lt;- add_kegg(
  SCsgp_species, 
  kegg_db, 
  meta99_full_taxo, 
  metaproteome_origin, 
  protein_file, 
  peptide_file, 
  text_to_filter = "HUMAN"
)


## End(Not run)
</code></pre>

<hr>
<h2 id='add_taxonomy'>add_taxonomy</h2><span id='topic+add_taxonomy'></span>

<h3>Description</h3>

<p>Integrates the database containing the taxonomic classification 
of the identified proteins in a &quot;metaproteome_object&quot;. 
The taxonomic classification is previously obtained by aligment 
algorithms and must include seven taxonomic levels assigned 
to a given protein: species, genus, family, order, class, phylum 
and superkingdom
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_taxonomy(metaproteome_object, taxonomic_database)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="add_taxonomy_+3A_metaproteome_object">metaproteome_object</code></td>
<td>
<p>List defined as &quot;metaproteome_object&quot; 
containing proteins and peptides abundances. The format of this 
object is similar to that generated from the function &quot;load_protspeps&quot;.</p>
</td></tr>
<tr><td><code id="add_taxonomy_+3A_taxonomic_database">taxonomic_database</code></td>
<td>
<p>Dataframe containing the taxonomic information 
for each protein. The first column must contain the same identifiers 
of those present in the column &quot;Accession&quot; from the dataframe 
&quot;peptides_proteins&quot; of the &quot;metaproteome_object&quot;. Two additional columns 
have to be present: i) one named &quot;organism&quot; containing the name of the strain
assigned to a given protein; and ii) the other named 
&quot;species.genus.family.order.class.phylum.superkingdom&quot;. 
The taxonomic classification can be obtained from a tool of 
sequences aligment and must be ordered as follows: species, genus, 
family, order, class, phylum and superkingdom. The characters inside must 
be concatenated by a comma 
(ex.&quot;Streptococcus anginosus,Streptococcus,Streptococcaceae,Lactobacillales,Bacilli,Firmicutes,Bacteria&quot;).
An example can be found in this <a href="https://zenodo.org/record/3997093">repository</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A &quot;metaproteome_object&quot;, which is a list of six elements with format
similar to that generated from the function &quot;load_protspeps&quot;. An additional 
column containing the taxonomic annotation is added to the dataframe 
named &quot;peptides_proteins&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# Download taxonmical annotation db: https://zenodo.org/record/3997093#.X0UYI6Zb_mE 
meta99_full_taxo &lt;- read.csv2("MetaHIT99_best_hit_taxo_complete.tsv", header = TRUE, sep="\t") 

# Files with spectral abundance and proteins list from X!Tandempipeline
protein_file &lt;- "your/specific/location/protein_list.txt"
peptide_file &lt;- "your/specific/location/peptide_counting.txt"
metadata_file &lt;- "your/location/metadata.csv"

metaproteome &lt;- load_protspeps(proteins_file, peptides_file, metadata_file)

metaproteome_taxo &lt;- add_taxonomy(metaproteome, meta99_full_taxo) 


## End(Not run)
</code></pre>

<hr>
<h2 id='crumble_taxonomy'>crumble_taxonomy</h2><span id='topic+crumble_taxonomy'></span>

<h3>Description</h3>

<p>Generates a list of four elements defined as &quot;spectral_count_object&quot;
containing taxonomic classification. The first element is a dataset 
that contains the spectral counts abundance organized by a provided 
taxonomic level. The possible taxonomic levels are: species, genus, 
family, order, class, phylum or superkingdom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>crumble_taxonomy(spectral_count_object, taxonomic_level, filter_rate = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="crumble_taxonomy_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot;
containing dataframes with abundance expressed as spectral counts
and organized by peptides, subgroups or groups. The format of 
this object is similar to that generated with the function 
&quot;getsc_specific&quot;. Taxonomy must be added previously 
with &quot;add_taxonomy&quot; function.</p>
</td></tr>
<tr><td><code id="crumble_taxonomy_+3A_taxonomic_level">taxonomic_level</code></td>
<td>
<p>Character indicating the taxonomic level to which
the spectral abundance will be arranged in the samples of the 
&quot;spectral_count_object&quot;. The possible options are: &quot;species&quot;, &quot;genus&quot;, 
&quot;family&quot;, &quot;order&quot;, &quot;class&quot;, &quot;phylum&quot; or &quot;superkingdom&quot;.</p>
</td></tr>
<tr><td><code id="crumble_taxonomy_+3A_filter_rate">filter_rate</code></td>
<td>
<p>Numeric value between 0 and 1 that indicates the 
minimal rate of consensual annotation desired by the user within each 
level of the spectral category (subgroup or group). This rate is defined 
as the ratio between the number of the most frequent annotation entity 
(&quot;species&quot;, &quot;genus&quot;, &quot;family&quot;, &quot;order&quot;, &quot;class&quot;, &quot;phylum&quot; or &quot;superkingdom&quot;) 
divided by the total number of entities within each level of the spectral 
category under study (subgroup or group). The default value is set to 1, 
if 100 % of consensus is desired.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of four elements defined as &quot;spectral_count_object&quot;, the first 
element is a dataframe with abundance expressed as spectral counts of entities 
(peptides, subgroups or groups) organized by the provided taxonomic level. 
The second element is a dataframe that contains the experiment information. 
The third element is a dataframe containing the information of peptides with 
their associated proteins. And the fourth element is a character indicating 
the type of object generated.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)

superkingdom_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "superkingdom")

phylum_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "phylum")

class_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "class")

order_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "order")

family_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "family")

genus_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "genus")

species_fecalwaters &lt;- crumble_taxonomy(fecal_waters, "species")


</code></pre>

<hr>
<h2 id='export_ipath3'>export_ipath3</h2><span id='topic+export_ipath3'></span>

<h3>Description</h3>

<p>Exports the KEGG Orthology (KO) terms in the adapted format to be used in 
the tool <a href="https://pathways.embl.de/">iPATH3</a>. The exported data is 
obtained from a &quot;spectral_count_object&quot; containing the functional 
annotation of the identified proteins from one condition or sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_ipath3(
  spectral_count_object,
  type_export,
  target_variable,
  sample_condition,
  hexadecimal_color,
  taxonomic_levels = NULL,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_ipath3_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing protein abundance expressed as spectral counts by a 
taxonomic level. The functional annotation must be added to this
object. The format of this object is similar to that generated from 
the function &quot;add_kegg&quot;.</p>
</td></tr>
<tr><td><code id="export_ipath3_+3A_type_export">type_export</code></td>
<td>
<p>Character indicating the type of export to be used. The
possible options are: i) &quot;all&quot; that selects all the KO terms from a given 
sample or a given condition; and, ii) &quot;selection&quot; that extracts the 
KO terms present in selected taxonomic entities (one or more).</p>
</td></tr>
<tr><td><code id="export_ipath3_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the column name from metadata 
containing the condition or sample to be analyzed.</p>
</td></tr>
<tr><td><code id="export_ipath3_+3A_sample_condition">sample_condition</code></td>
<td>
<p>Atomic vector indicating the sample from which 
the functional information will be extracted.</p>
</td></tr>
<tr><td><code id="export_ipath3_+3A_hexadecimal_color">hexadecimal_color</code></td>
<td>
<p>Character indicating the color to be used in 
iPATH3, this value must be indicated in hexadecimal format (eg. #ff0000).</p>
</td></tr>
<tr><td><code id="export_ipath3_+3A_taxonomic_levels">taxonomic_levels</code></td>
<td>
<p>Optional vector indicating the taxonomic levels
from which the KO terms will be extracted. This option is needed only
if the type of export is &quot;selection&quot;.</p>
</td></tr>
<tr><td><code id="export_ipath3_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create a file in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A csv file containing the KO terms present in a given sample or 
condition. The content of this file can be inserted directly in the tool  
<a href="https://pathways.embl.de/">iPATH3</a>. The width of the lines in 
iPATH3 will be displayed by the  percentage of spectra in the selected 
sample or condition. In this way, KO terms belonging to a given taxonomic 
level are represented in three intervals based on their abundace: i) 
below 2 percent, i) between 2 to 10 percent, or i) above 10 percent.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(species_annot_fw)

export_ipath3(
   species_annot_fw, 
   "all", 
   "SampleID", 
   "Q1_prot", 
   "#840AA3"
)

taxonomic_entities &lt;- c("Bacteroides caccae", "Coprococcus catus", "Merdimonas faecis")
export_ipath3(
   species_annot_fw, 
   "selection", 
   "SC_name", 
   "FW2", 
   "#28c1df", 
   taxonomic_entities
)


</code></pre>

<hr>
<h2 id='export_robject'>export_robject</h2><span id='topic+export_robject'></span>

<h3>Description</h3>

<p>Exports one of the dataframes present in a &quot;metaproteome_object&quot;
or in &quot;spectral_count_object&quot;. The export extensions can be RDATA or RDS.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_robject(entry_object, data_exported, format_data, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_robject_+3A_entry_object">entry_object</code></td>
<td>
<p>A &quot;metaproteome_object&quot; or a &quot;spectral_count_object&quot; with
similar format to that generated with the functions &quot;load_protspeps&quot; or 
&quot;getsc_specific&quot;, respectively.</p>
</td></tr>
<tr><td><code id="export_robject_+3A_data_exported">data_exported</code></td>
<td>
<p>Character indicating the type of data to be exported from 
a &quot;metaproteome_object&quot; or a &quot;spectral_count_object&quot;. The possible options 
are: i) &quot;proteins&quot;, ii) &quot;peptides&quot;, iii) &quot;pepProts&quot;, iv) &quot;spectral&quot;, 
v) &quot;spectral_percent&quot;, vi) &quot;metadata&quot;.</p>
</td></tr>
<tr><td><code id="export_robject_+3A_format_data">format_data</code></td>
<td>
<p>Character indicating the file extension, this can be 
either &quot;RDATA&quot; or &quot;RDS&quot;.</p>
</td></tr>
<tr><td><code id="export_robject_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask permission 
to create object in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A file with the extension &quot;RDATA&quot; or &quot;RDS&quot; containing the information 
from the selected dataframe from a &quot;metaproteome_object&quot; or a 
&quot;spectral_count_object&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
export_robject(fecal_waters, "pepProts", "rdata")

data(species_fw)
export_robject(species_fw, "spectral", "rds")


</code></pre>

<hr>
<h2 id='export_vennlists'>export_vennlists</h2><span id='topic+export_vennlists'></span>

<h3>Description</h3>

<p>Exports as csv files the elements (groups, subgroups, peptides or 
taxonomic levels) generated from the function &quot;plot_venn&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>export_vennlists(venn_lists_object, output_repo = NULL, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="export_vennlists_+3A_venn_lists_object">venn_lists_object</code></td>
<td>
<p>List defined as &quot;venn_lists_object&quot; containing the
elements (peptides, subgroups, groups or taxonomic elements) 
generated with the function &quot;plot_venn&quot;.</p>
</td></tr>
<tr><td><code id="export_vennlists_+3A_output_repo">output_repo</code></td>
<td>
<p>Character indicating the path of a previously 
created directory where the lists will be exported. 
This parameter is optional.</p>
</td></tr>
<tr><td><code id="export_vennlists_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order 
to ask permission to create csv files in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>csv files containing the elements present on each logic section 
(specific and intersections) from the list defined as &quot;venn_lists_object&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(venn_methods)

export_vennlists(venn_methods)


</code></pre>

<hr>
<h2 id='fecal_waters'>fecal waters</h2><span id='topic+fecal_waters'></span>

<h3>Description</h3>

<p>Data containing the abundance of 474 metaproteins 
expressed in spectral counts. Data generated from an Orbitrap 
Fusion Lumos  Tribrid Mass Spectrometer. The dataset contains 
the metaproteomes from three extraction methods: i) &quot;Q&quot; for Qiagen, 
ii) &quot;FW&quot; for fecal waters, and iii) &quot;Q_FW&quot; for the mixture 
of Qiagen and fecal waters. Data generated in the context of 
the project Microbiome Rapid Access (Université Paris-Saclay).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(fecal_waters)
</code></pre>


<h3>Format</h3>

<p>A list of four elements defined as &quot;spectral_count_object&quot;, 
generated with the function &quot;getsc_specific&quot;
</p>

<dl>
<dt>SC_subgroups</dt><dd><p>dataframe with 9 samples containing 474 
subgroups or metaproteins with abundance expressed as 
spectral counts</p>
</dd>
<dt>metadata</dt><dd><p>information related to the 9 samples from the 
experiment</p>
</dd>
<dt>peptides_proteins</dt><dd><p>information related to each of the 
1557 identified peptides</p>
</dd>
<dt>type_object</dt><dd><p>character indicating the type of object</p>
</dd>
</dl>


<hr>
<h2 id='filter_shared'>filter_shared</h2><span id='topic+filter_shared'></span>

<h3>Description</h3>

<p>Keeps the elements from a &quot;spectral_count_object&quot; that are common to all 
levels of a provided explanatory variable. This variable MUST correspond to 
the name of ONE column of the dataframe &quot;metadata&quot; (eg. conditions or samples). 
This function allows to identify the elements that are common to several 
conditions or samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_shared(spectral_count_object, metadata_feature)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_shared_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List containing dataframes with proteomics
elements whose abundance is expressed as spectral counts and are
organized by peptides, subgroups, groups or taxonomic levels. 
The format of this object is similar to that generated from 
the functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="filter_shared_+3A_metadata_feature">metadata_feature</code></td>
<td>
<p>Character indicating the name of one explanatory 
variable (ONE column name) of the dataframe &quot;metadata&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list defined as &quot;spectral_count_object&quot; in which 
the elements (groups, subgroups, peptides or taxonomic levels) 
of a given variable (condition or samples) from metadata 
will have at least ONE spectra per variable.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
data(species_fw)

common_elements_per_sample &lt;- filter_shared(fecal_waters, "SampleID")

common_elements_per_condition &lt;- filter_shared(species_fw, "Condition")


</code></pre>

<hr>
<h2 id='filter_text'>filter_text</h2><span id='topic+filter_text'></span>

<h3>Description</h3>

<p>Matches the entities containing a given chain of characters inside an explanatory 
variable (column name) of the dataframe &quot;peptides_proteins&quot; from a 
&quot;spectral_count_object&quot;. Based on the user's decision, the peptides, subgroups, 
groups or taxonomic levels containig the provided chain of characters will 
be kept or discarted in a newly generated object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_text(spectral_count_object, pepsprots_feature, text_to_filter, decision)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_text_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List containing dataframes with proteomics
elements whose abundance is expressed as spectral counts and are
organized by peptides, subgroups, groups or taxonomic levels. 
The format of this object is similar to that generated from 
the functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="filter_text_+3A_pepsprots_feature">pepsprots_feature</code></td>
<td>
<p>Character indicating the name of one explanatory 
variable (ONE column name) of the dataframe &quot;peptides_proteins&quot;.</p>
</td></tr>
<tr><td><code id="filter_text_+3A_text_to_filter">text_to_filter</code></td>
<td>
<p>Character containig the text to be searched in 
the &quot;pepsprots_feature&quot; content.</p>
</td></tr>
<tr><td><code id="filter_text_+3A_decision">decision</code></td>
<td>
<p>Character indicating wether the elements containing 
the matched text will be kept or dirscarted. The two allowed option 
are: &quot;keep&quot; or &quot;discard&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list defined as &quot;spectral_count_object&quot; with or without the 
elements (peptides, subgroups, groups, taxonomic items) that 
matched the provided text in a given variable of the 
&quot;peptides_proteins&quot; dataframe.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
data(species_fw)

cysteine_alkylations &lt;- filter_text(fecal_waters, "Modifs", "57.02146", "keep")

exclude_medimonas &lt;- filter_text(species_fw, "organism", "Merdimonas faecis BR31", "discard")


</code></pre>

<hr>
<h2 id='filter_unshared'>filter_unshared</h2><span id='topic+filter_unshared'></span>

<h3>Description</h3>

<p>Keeps the elements from a &quot;spectral_count_object&quot; that are specific to one
level of a provided explanatory variable. This variable MUST correspond to
the name of ONE column of the dataframe &quot;metadata&quot; (eg. conditions or samples).
This function allows to identify the elements that are specific to one 
condition or one sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_unshared(spectral_count_object, metadata_feature)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_unshared_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List containing dataframes with proteomics
elements whose abundance is expressed as spectral counts and are
organized by peptides, subgroups, groups or taxonomic levels. 
The format of this object is similar to that generated from 
the functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="filter_unshared_+3A_metadata_feature">metadata_feature</code></td>
<td>
<p>Character indicating the name of one explanatory 
variable (ONE column name) of the dataframe &quot;metadata&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list defined as &quot;spectral_count_object&quot; with the specific 
elements per sample or condition, having at least one spectra.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
data(species_fw)

specific_elements_per_sample &lt;- filter_unshared(fecal_waters, "SampleID")

specific_elements_per_condition &lt;- filter_unshared(species_fw, "Condition")


</code></pre>

<hr>
<h2 id='getsc_specific'>getsc_specific</h2><span id='topic+getsc_specific'></span>

<h3>Description</h3>

<p>Returns the abundances, expressed as spectral counts (SC), of the different 
peptides, subgroups (also referred as metaprotein) or groups within the 
samples of the experiment. The abundance corresponds to the sum of SC of the 
specific peptides present in a given subgroup or group. See 
<a href="http://pappso.inrae.fr/bioinfo/xtandempipeline/">X!TandemPipeline</a> 
for more details concerning the grouping algorithm.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getsc_specific(metaproteome_object, type_SCspecific)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="getsc_specific_+3A_metaproteome_object">metaproteome_object</code></td>
<td>
<p>List defined as &quot;metaproteome_object&quot; with 
dataframes having a similar format to that generated from 
&quot;load_protspeps&quot; function. It contains metaproteomics 
data such as peptide and protein abundances and sample 
information.</p>
</td></tr>
<tr><td><code id="getsc_specific_+3A_type_scspecific">type_SCspecific</code></td>
<td>
<p>Character indicating the type of data 
to be returned. The possible options are: i) &quot;sc_specific_peptides&quot; 
for peptides, or ii) &quot;sc_groups&quot; for groups, or iii) &quot;sc_subgroups&quot; 
for subgroups. For more details of the identification algorithm 
(peptides, subgroups, groups) check 
<a href="http://pappso.inrae.fr/bioinfo/xtandempipeline/">X!TandemPipeline</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of four elements defined as &quot;spectral_count_object&quot;. The first 
element is a dataframe organized in function of peptides OR subgroups 
(also referred as metaproteins) OR groups. The entities of this dataframe 
have their abundance expressed as SC from specific peptides. The second 
element is a dataframe of metadata containing the experiment information. 
The third element is a dataframe containing the information of peptides 
with their associated proteins. The fourth element is a character 
indicating the type of object generated.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

# From a given "metaproteome_object" add the taxonomic classification

metaproteome &lt;- load_protspeps(proteins_file, peptides_file, metadata_file)
metaproteome_taxo &lt;- add_taxonomy(metaproteome, meta99_full_taxo) 

# Organize proteomics data by peptides OR subgroups OR groups

SC_specific_peptides &lt;- getsc_specific(metaproteome_taxo, 'sc_specific_peptides') 
SC_specific_groups &lt;- getsc_specific(metaproteome_taxo, 'sc_groups') 
SC_specific_subgroups &lt;- getsc_specific(metaproteome_taxo, 'sc_subgroups') 


## End(Not run)
</code></pre>

<hr>
<h2 id='identify_differences'>identify_differences</h2><span id='topic+identify_differences'></span>

<h3>Description</h3>

<p>Shows the most differential taxonomic elements between two 
conditions or samples from a list defined as &quot;spectral_count_object&quot; 
with taxonomic classification. These elements are those with an 
absolute log2 (condition + 1 / reference + 1) &gt; 3. If a given condition has
several replicates the mean value is taken into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>identify_differences(
  spectral_count_object,
  target_variable,
  list_conditions,
  filter_ratio = 3,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="identify_differences_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts
and organized by taxonomic levels. The format of this object is 
similar to that generated from the function &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="identify_differences_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the variable name 
containing the conditions or samples to be compared. This value 
corresponds to the name of one column from metadata.</p>
</td></tr>
<tr><td><code id="identify_differences_+3A_list_conditions">list_conditions</code></td>
<td>
<p>Atomic vector indicating two conditions to 
be compared. The first element will be considered as the reference.</p>
</td></tr>
<tr><td><code id="identify_differences_+3A_filter_ratio">filter_ratio</code></td>
<td>
<p>Numeric value indicating the fold change filter 
to be considered for the pairwise comparison. The minimal value can be 
a fold change of 1.25. The default value is set at 3.</p>
</td></tr>
<tr><td><code id="identify_differences_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create an object in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Barplots (pdf) and a csv file with the defferential taxonomic 
elements between TWO conditions or sample. These elements are those that 
fulfill the ratio log2 (condition + 1 / reference + 1) &gt; filter_ratio.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(species_fw)
identify_differences(species_fw, "Methods", c("S", "S_EF"))

identify_differences(species_fw, "Methods", c("EF", "S_EF"), filter_ratio = 1.3)


</code></pre>

<hr>
<h2 id='inspect_sample_elements'>inspect_sample_elements</h2><span id='topic+inspect_sample_elements'></span>

<h3>Description</h3>

<p>Displays a graph that indicates the number of common elements from a 
&quot;spectral_count_object&quot; (peptides, subgroups, groups or taxonomic entities) 
per sample. This function is useful to distinguish heterogeneity between 
samples in an experimental design.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inspect_sample_elements(spectral_count_object, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inspect_sample_elements_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts. 
The spectral data can be organized by peptides, subgroups, 
groups or taxonomic levels.</p>
</td></tr>
<tr><td><code id="inspect_sample_elements_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create a pdf file in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Barplots (pdf) ilustrating the common spectral elements 
(peptides, subgroups, groups, taxonomic elements) per sample
in a &quot;spectral_count_object&quot;.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
inspect_sample_elements(fecal_waters)


</code></pre>

<hr>
<h2 id='load_protspeps'>load_protspeps</h2><span id='topic+load_protspeps'></span>

<h3>Description</h3>

<p>Loads three files: i) peptides abundances expressed as 
spectral counts, ii) proteins information, and iii) metadata of the
mass spectrometry samples. Combines the three files into a 
&quot;metaproteome_object&quot;, a list containing these dataframes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>load_protspeps(protein_file, peptide_file, metadata_file)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="load_protspeps_+3A_protein_file">protein_file</code></td>
<td>
<p>Character indicating the location of a txt 
file containing the list of proteins generated in 
<a href="http://pappso.inrae.fr/bioinfo/xtandempipeline/">X!TandemPipeline</a> 
using an adapted iterative approach described by 
<a href="https://www.theses.fr/2019SORUS043">Bassignani, 2019</a>. 
Separation between columns should be indicated by tabulation. 
For more details regarding data input check 
<a href="https://forgemia.inra.fr/pappso/metaprotr#data-inputs">format examples</a>.</p>
</td></tr>
<tr><td><code id="load_protspeps_+3A_peptide_file">peptide_file</code></td>
<td>
<p>Character indicating the location of a txt 
file containing peptides abundances expressed as spectral counts.
This file is generated from 
<a href="http://pappso.inrae.fr/bioinfo/xtandempipeline/">X!TandemPipeline</a> 
using an adapted iterative approach described by 
<a href="https://www.theses.fr/2019SORUS043">Bassignani, 2019</a>. 
Separation between columns should be indicated by tabulation. 
For more details regarding data input check 
<a href="https://forgemia.inra.fr/pappso/metaprotr#data-inputs">format examples</a>.</p>
</td></tr>
<tr><td><code id="load_protspeps_+3A_metadata_file">metadata_file</code></td>
<td>
<p>Character indicating the location of a csv file 
containing the samples information. The following columns names 
MUST be present: &quot;SC_name&quot; (sample ids assigned by the user), &quot;msrunfile&quot; 
(name of samples as indicated in mass spectrometry files and in the columns
of peptide_file) and &quot;SampleID&quot; (codes indicating the experimental group).
Additional columns containing complementary information can be added by 
the user (ex. replicates, order of injection, etc.). 
Separation between columns should be indicated by tabulation. 
For more details regarding data input check 
<a href="https://forgemia.inra.fr/pappso/metaprotr#data-inputs">format examples</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A &quot;metaproteome_object&quot;, which is a list of six elements containing: 
1) dataframe of the protein identifiers, 
2) dataframe of the peptide identifiers, 
3) dataframe containing the information of peptides with their associated proteins, 
4) dataframe of metadata containing the experiment information, 
5) dataframe of spectral counts per peptide on each sample, 
6) character indicating the type of object generated.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

protein_file &lt;- "location/peptides_abundances.csv"
peptide_file &lt;- "location/proteins_list.csv"
metadata &lt;- "location/metadata.csv"
metaproteome &lt;- load_protspeps(protein_file, peptide_file, metadata_file)


## End(Not run)
</code></pre>

<hr>
<h2 id='plot_dendocluster'>plot_dendocluster</h2><span id='topic+plot_dendocluster'></span>

<h3>Description</h3>

<p>Draws a dendogram where samples are clustered based on the number of elements 
present on each sample from a &quot;spectral_count_object&quot;. This graph is 
constructed based on Spearman correlations transformed into distances 
and plotted with the logic of the package 
<a href="https://CRAN.R-project.org/package=dendextend">dendextend</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_dendocluster(
  spectral_count_object,
  target_variable,
  file_title,
  hclust_method = "ward.D",
  correlation_method = "spearman",
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_dendocluster_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts 
from peptides, subgroups, groups or taxonomic levels. The format 
of this object is similar to that generated from the functions
&quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_dendocluster_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the name of one column 
from metadata. The different levels in this column will be represented 
as different colors in the final dendogram.</p>
</td></tr>
<tr><td><code id="plot_dendocluster_+3A_file_title">file_title</code></td>
<td>
<p>Character indicating the name of the generated file.</p>
</td></tr>
<tr><td><code id="plot_dendocluster_+3A_hclust_method">hclust_method</code></td>
<td>
<p>Character indicating the agglomeration method to be 
used for the hierarchical clustering. The possible methods are discribed 
on <a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/hclust">hclust</a>.
The default method is &quot;ward.D&quot;.</p>
</td></tr>
<tr><td><code id="plot_dendocluster_+3A_correlation_method">correlation_method</code></td>
<td>
<p>Character indicating the correlation coeficient to 
be computed. The possible options are discribed in the function 
<a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/cor">cor</a>. 
The default value is &quot;spearman&quot;.</p>
</td></tr>
<tr><td><code id="plot_dendocluster_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create a pdf file in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dendogram plot (pdf) indicating the number of elements per
sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
str(fecal_waters$metadata)

plot_dendocluster(fecal_waters, "Condition", "title_dendogram")

plot_dendocluster(fecal_waters, "Condition", "title_dendogram", 
   hclust_method = "mcquitty")

plot_dendocluster(fecal_waters, "Condition", "title_dendogram_groups", 
   correlation_method = "pearson")


</code></pre>

<hr>
<h2 id='plot_fulltaxo'>plot_fulltaxo</h2><span id='topic+plot_fulltaxo'></span>

<h3>Description</h3>

<p>Provides the number of taxonomic entities per sample in the different 
taxonomic levels. The taxonomic levels are: species, genus, family, order, 
class, phylum and superkingdom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_fulltaxo(spectral_count_object, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_fulltaxo_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with format similar to that generated with 
the function &quot;getsc_specific&quot;.  This object contains abundances 
expressed as spectral counts from peptides, subgroups (metaproteins) 
or groups. Taxonomy must have previously been added with the function 
&quot;add_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_fulltaxo_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create a pdf and a csv file in the workstation of 
the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Bar plots (pdf) and csv file with the number of taxonomic species, 
genus, family, order, class, phylum and superkingdom per sample. 
An additional csv file is generated providing the rate of assignment. 
The rate of assigment corresponds to the ratio between the number of 
the most frequent annotation (&quot;species&quot;, &quot;genus&quot;, &quot;family&quot;, &quot;order&quot;, 
&quot;class&quot;, &quot;phylum&quot; or &quot;superkingdom&quot;) and the total number of elements within 
each level of the spectral category under study (subgroup or group).
The csv file is generated only when the &quot;spectral_count_object&quot; is 
organized by subgroup or by group.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
plot_fulltaxo(fecal_waters)


</code></pre>

<hr>
<h2 id='plot_intensities'>plot_intensities</h2><span id='topic+plot_intensities'></span>

<h3>Description</h3>

<p>Draws violin plots containing the abundance intensities expressed as 
spectral counts per level (peptides, subgroups, groups or taxonomic entities) 
in provided samples or conditions from a &quot;spectral_count_object&quot;. If the 
provided conditions have several replicates the mean value is taken into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_intensities(
  spectral_count_object,
  target_variable,
  image_title = NULL,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_intensities_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts 
from peptides, subgroups, groups or taxonomic levels. The format of 
this object is similar to that generated from the functions 
&quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_intensities_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the name of one column from 
metadata, the column must contain the conditions to be displayed.</p>
</td></tr>
<tr><td><code id="plot_intensities_+3A_image_title">image_title</code></td>
<td>
<p>Character indicating the title to be displayed 
in the generated image.</p>
</td></tr>
<tr><td><code id="plot_intensities_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask
for permission to create a pdf file in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Violin plots (pdf) indicating the spectral counts of the different 
levels (peptides, subgroups, groups or taxonomic entities) per sample 
or condition.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
plot_intensities(fecal_waters, "SC_name", "Title to display inside the plot")

data(species_fw)
plot_intensities(species_fw, "Condition", "Abundance per condition")


</code></pre>

<hr>
<h2 id='plot_intensities_ratio'>plot_intensities_ratio</h2><span id='topic+plot_intensities_ratio'></span>

<h3>Description</h3>

<p>Generates a scatter plot of the log2 (ratio + 1) between two conditions considering
the spectral counts of each entity (peptides, subgroups, groups or taxonomic 
levels) from a &quot;spectral_count_object&quot;. If a given condition has several 
replicates the mean value is taken into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_intensities_ratio(
  spectral_count_object,
  target_variable,
  list_conditions,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_intensities_ratio_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts
organized by peptides, subgroups, groups or taxonomic levels. 
The format of this object is similar to that generated from 
the functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_intensities_ratio_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the variable name 
containing the conditions to be compared. This value corresponds 
to the name of one column from metadata.</p>
</td></tr>
<tr><td><code id="plot_intensities_ratio_+3A_list_conditions">list_conditions</code></td>
<td>
<p>Atomic vector indicating two conditions to A
be compared. The first element is considered as the reference (denominator) 
for ratio calculations.</p>
</td></tr>
<tr><td><code id="plot_intensities_ratio_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create object in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A scatter plot (pdf) indicating the log2 (ratio + 1) of the entities
(peptides, soubgroups, groups, taxonomic) between the two conditions provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)

plot_intensities_ratio(fecal_waters, "Methods", c("EF", "S"))

plot_intensities_ratio(fecal_waters, "SC_name", c("Q1", "Q2"))


</code></pre>

<hr>
<h2 id='plot_pca'>plot_pca</h2><span id='topic+plot_pca'></span>

<h3>Description</h3>

<p>Performs a Principal Components Analysis (PCA) from the spectral counts of the entities 
(peptides, subgroups, groups or taxonomic elements) in a 
&quot;spectral_count_object&quot; with or without taxonomy. PCA decomposition of
high dimensional data allows to observe global effects in two dimensions. For more 
details of the used function check dudi.pca from <a href="https://CRAN.R-project.org/package=ade4">ade4</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_pca(spectral_count_object, colors_var, pc_components, force = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_pca_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List described as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts from 
peptides, subgroups, groups or taxonomic levels. The format of this 
object is similar to that generated from the functions &quot;getsc_specific&quot; and 
&quot;crumble_taxonomy&quot;. The PCA projections will be applied to these observations.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_colors_var">colors_var</code></td>
<td>
<p>Character indicating the name of one column 
from metadata. The samples will be represented in different colors
in function of the levels of this variable (ex. conditions).</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_pc_components">pc_components</code></td>
<td>
<p>Two numeric values indicating two principal 
components to be analyzed.</p>
</td></tr>
<tr><td><code id="plot_pca_+3A_force">force</code></td>
<td>
<p>Logic value set as FALSE by default in order to ask 
permission to create a file in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pdf file containing the results of PCA applied to the two provided 
principal components. Including a bar plot indicating the percentage of variance
per principal component.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
plot_pca(fecal_waters, "Methods", c(1, 2))

data(species_fw)
plot_pca(species_fw, "Methods", c(1, 3))

data(species_annot_fw)
plot_pca(species_annot_fw, "Condition", c(1, 2))


</code></pre>

<hr>
<h2 id='plot_pietaxo'>plot_pietaxo</h2><span id='topic+plot_pietaxo'></span>

<h3>Description</h3>

<p>Generates a pie chart with taxonomic distribution of one selected
sample or condition. If the provided condition has several replicates 
the mean value is taken into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_pietaxo(
  spectral_count_object,
  target_variable,
  sampling,
  filter_percent = 1,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_pietaxo_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>list defined as &quot;spectral_count_object&quot; 
containing dataframes with spectral counts abundance of the 
samples organized by taxonomy (species, genus, family, 
order, class, phylum or superkingdom). This object is generated 
with the function &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_pietaxo_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the name of one column 
from metadata. This column must contain the identifiers of the sample 
or condition to be plotted.</p>
</td></tr>
<tr><td><code id="plot_pietaxo_+3A_sampling">sampling</code></td>
<td>
<p>Character indicating the name of sample or condition
to be plotted. This character must be present in the &quot;target_variable&quot;.</p>
</td></tr>
<tr><td><code id="plot_pietaxo_+3A_filter_percent">filter_percent</code></td>
<td>
<p>Optional numeric value between 0 and 99 that 
sets the minimal percentage of spectral counts at which the taxonomic 
elements will be displayed. The elements whose values are lower than 
this number will be gathered and displayed as &quot;others&quot;. 
The default value is set at 1.</p>
</td></tr>
<tr><td><code id="plot_pietaxo_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create files in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A pie chart (pdf) and a csv file with the taxonomic 
distribution of one sample or one condition. In the csv file, 
all the elements have their: a) spectral counts, b) percentage of 
spectral count in the sample, c) the taxonomic name and d)
the taxonomic elements that were assiged as 'others' in function 
of the filter provided.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(species_fw)

plot_pietaxo(species_fw, "Methods", "S")

plot_pietaxo(species_fw, "SC_name", "Q1")


</code></pre>

<hr>
<h2 id='plot_stackedtaxo'>plot_stackedtaxo</h2><span id='topic+plot_stackedtaxo'></span>

<h3>Description</h3>

<p>Generates stacked barplots of the spectral counts distributions among the different
taxonomic entities (&quot;species&quot;, &quot;genus&quot;, &quot;family&quot;, &quot;order&quot;, &quot;class&quot;, &quot;phylum&quot; 
or &quot;superkingdom&quot;) within the samples or conditions of a &quot;spectral_count_object&quot; 
with taxonomy. If the provided conditions have several replicates 
the mean value is taken into account.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_stackedtaxo(
  spectral_count_object,
  target_variable,
  bars_data,
  filter_percent = 1,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_stackedtaxo_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with spectral counts abundance organized by taxonomy 
(species, genus, family, order, class, phylum or superkingdom). This 
object is generated with the function &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_stackedtaxo_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the name of one column from 
metadata. The stacked barplots will be ordered by the levels of this variable.</p>
</td></tr>
<tr><td><code id="plot_stackedtaxo_+3A_bars_data">bars_data</code></td>
<td>
<p>Character indicating the type of labels to be displayed in 
the stacked bars. The possible options are &quot;percent&quot; or &quot;numbers&quot;.</p>
</td></tr>
<tr><td><code id="plot_stackedtaxo_+3A_filter_percent">filter_percent</code></td>
<td>
<p>Optional numeric value between 0 and 99 that 
sets the minimal percentage of spectral counts at which the taxonomic 
elements will be displayed. The elements whose values are lower than 
this number will be gathered and displayed as &quot;others&quot;. 
The default value is set at 1.</p>
</td></tr>
<tr><td><code id="plot_stackedtaxo_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create a pdf in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Barplots (pdf) of the taxonomic distribution of the samples present 
in a &quot;spectral_count_object&quot; with taxonomic levels.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(species_fw)

plot_stackedtaxo(species_fw, 'SampleID', 'percent', 2)

plot_stackedtaxo(species_fw, 'SC_name', 'numbers')


</code></pre>

<hr>
<h2 id='plot_venn'>plot_venn</h2><span id='topic+plot_venn'></span>

<h3>Description</h3>

<p>Generates a Venn diagram comparing up to 3 conditions. The lists of 
elements for each condition are also returned as a &quot;venn_lists_object&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_venn(
  spectral_count_object,
  target_variable,
  list_conditions,
  force = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_venn_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts 
from peptides, subgroups, groups or taxonomic levels. 
The format of this object is similar to that generated from the 
functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy&quot;.</p>
</td></tr>
<tr><td><code id="plot_venn_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the name of the explanatory 
variable that contains the conditions to be compared. This value corresponds 
to the name of one column from the metadata dataframe.</p>
</td></tr>
<tr><td><code id="plot_venn_+3A_list_conditions">list_conditions</code></td>
<td>
<p>Atomic vector indicating the conditions 
to be compared. The provided elements (2 or 3) must be present in the 
variable indicated as &quot;target_variable&quot;.</p>
</td></tr>
<tr><td><code id="plot_venn_+3A_force">force</code></td>
<td>
<p>Logic value set at FALSE by default in order to ask 
permission to create a pdf file in the workstation of the user.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A Venn diagram (pdf) and a list defined as &quot;venn_list_object&quot;
containing the elements (peptides, soubgroups, groups or taxonomic levels) 
for each logical section of the Venn diagram (specific and intersections).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(fecal_waters)
venn_QFW1_Q1 &lt;- plot_venn(fecal_waters, "SC_name", c("Q1_FW1", "Q1"))

data(species_fw)
venn_all &lt;- plot_venn(species_fw, "Methods", c("S_EF", "S", "EF"))

</code></pre>

<hr>
<h2 id='remove_element'>remove_element</h2><span id='topic+remove_element'></span>

<h3>Description</h3>

<p>Removes elements from a &quot;spectral_count_object&quot;. These elements can 
be: i) samples, ii) peptides, iii) proteins, iv) soubgroups, v) groups, 
vi) sequences, vii) species, viii) genus, ix) family, x) order, 
xi) class, xii) phylum or xiii) superkingdom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>remove_element(spectral_count_object, target_variable, list_elements)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="remove_element_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts 
from peptides, subgroups, groups or taxonomic levels. The format 
of this object is similar to that generated from the 
functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy.&quot;</p>
</td></tr>
<tr><td><code id="remove_element_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the variable that contains 
the elements to be removed. The options are : i) &quot;peptides&quot;, ii) &quot;proteins&quot;, 
iii) &quot;soubgroups&quot;, iv) &quot;groups&quot;, v) &quot;sequences&quot;, vi) &quot;species&quot;, vii) &quot;genus&quot;,
viii) &quot;family&quot;, ix) &quot;order&quot;, x) &quot;class&quot;, xi) &quot;phylum&quot; or xii) &quot;superkingdom&quot;. 
To select xiii) &quot;samples&quot;, it should be indicated the name of ONE column from 
metadata.</p>
</td></tr>
<tr><td><code id="remove_element_+3A_list_elements">list_elements</code></td>
<td>
<p>Atomic vector indicating the elements to be removed. 
For &quot;samples&quot;, indicate the element(s) present in the provided variable 
from metadata. For &quot;peptides&quot;, &quot;proteins&quot;, &quot;subgroups&quot; and &quot;groups&quot; 
provide the X!Tandem nomenclature. For &quot;sequences&quot;, provide the 
peptide sequences expressed as aminoacids. For any taxonomic level, provide
the taxonomic entities.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list defined as &quot;spectral_count_object&quot; without the elements 
provided in the second argument of the function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
data(species_fw)

data_selected_samples &lt;- remove_element(fecal_waters, "Methods", c("S_EF", "EF"))

data_selected_peptides &lt;- remove_element(fecal_waters, "peptides", c("pepa3c417", "pepd4664a1"))

data_selected_proteins &lt;- remove_element(species_fw, "proteins", c("a3.a9.a1", "a5.b81.a1"))

data_selected_subgroups &lt;- remove_element(species_fw, "subgroups", c("a3.a9", "b73.a5"))

data_selected_groups &lt;- remove_element(species_fw, "groups", c("a3", "b34", "c231"))

data_selected_sequences &lt;- remove_element(species_fw, "sequences", c("AQLNFGGTIENVVIRDEFPLEK"))


</code></pre>

<hr>
<h2 id='select_element'>select_element</h2><span id='topic+select_element'></span>

<h3>Description</h3>

<p>Keeps specific elements from a &quot;spectral_count_object&quot;. These elements can 
be: i) samples, ii) peptides, iii) proteins, iv) soubgroups, v) groups, 
vi) sequences, vii) species, viii) genus, ix) family, x) order, 
xi) class, xii) phylum or xiii) superkingdom.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_element(spectral_count_object, target_variable, list_elements)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="select_element_+3A_spectral_count_object">spectral_count_object</code></td>
<td>
<p>List defined as &quot;spectral_count_object&quot; 
containing dataframes with abundance expressed as spectral counts 
from peptides, subgroups, groups or taxonomic levels. The format 
of this object is similar to that generated from the 
functions &quot;getsc_specific&quot; and &quot;crumble_taxonomy.&quot;</p>
</td></tr>
<tr><td><code id="select_element_+3A_target_variable">target_variable</code></td>
<td>
<p>Character indicating the variable that contains 
the elements to be kept. The options are : i) &quot;peptides&quot;, ii) &quot;proteins&quot;, 
iii) &quot;soubgroups&quot;, iv) &quot;groups&quot;, v) &quot;sequences&quot;, vi) &quot;species&quot;, vii) &quot;genus&quot;,
viii) &quot;family&quot;, ix) &quot;order&quot;, x) &quot;class&quot;, xi) &quot;phylum&quot; or xii) &quot;superkingdom&quot;. 
To select xiii) &quot;samples&quot;, it should be indicated the name of ONE column from 
metadata.</p>
</td></tr>
<tr><td><code id="select_element_+3A_list_elements">list_elements</code></td>
<td>
<p>Atomic vector indicating the elements to be kept 
For &quot;samples&quot;, indicate the element(s) present in the provided variable 
from metadata. For &quot;peptides&quot;, &quot;proteins&quot;, &quot;subgroups&quot; and &quot;groups&quot; 
provide the X!Tandem nomenclature. For &quot;sequences&quot;, provide the 
peptide sequences expressed as aminoacids. For any taxonomic level, provide
the taxonomic entities.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list defined as &quot;spectral_count_object&quot; with the elements 
provided in the second argument of the function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(fecal_waters)
data(species_fw)

data_selected_samples &lt;- select_element(fecal_waters, "Methods", c("S_EF", "EF"))

data_selected_peptides &lt;- select_element(fecal_waters, "peptides", c("pepa3c417", "pepd4664a1"))

data_selected_proteins &lt;- select_element(species_fw, "proteins", c("a3.a9.a1", "a5.b81.a1"))

data_selected_subgroups &lt;- select_element(species_fw, "subgroups", c("a3.a9", "b73.a5"))

data_selected_groups &lt;- select_element(species_fw, "groups", c("a3", "b34", "c231"))

data_selected_sequences &lt;- select_element(species_fw, "sequences", c("AQLNFGGTIENVVIRDEFPLEK"))


</code></pre>

<hr>
<h2 id='species_annot_fw'>species_annot_fw</h2><span id='topic+species_annot_fw'></span>

<h3>Description</h3>

<p>Data containing the abundance of 15 species 
expressed in spectral counts and with functional annotation. 
Data generated from an Orbitrap Fusion Lumos Tribrid 
Mass Spectrometer. The dataset contains the metaproteomes 
from three extraction methods: i) &quot;Q&quot; for Qiagen, 
ii) &quot;FW&quot; for fecal waters, and iii) &quot;Q_FW&quot; for the mixture 
of Qiagen and fecal waters. Data generated in the context of 
the project Microbiome Rapid Access (Université Paris-Saclay).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(species_annot_fw)
</code></pre>


<h3>Format</h3>

<p>A list of four elements defined as &quot;spectral_count_object&quot; with 
functional annotation, generated with the function &quot;add_kegg&quot;
</p>

<dl>
<dt>SC_subgroups</dt><dd><p>dataframe with 9 samples containing 15 
especies with abundance expressed as spectral counts</p>
</dd>
<dt>metadata</dt><dd><p>information related to the 9 samples from the 
experiment</p>
</dd>
<dt>peptides_proteins</dt><dd><p>information related to each of the 
1315 identified peptides</p>
</dd>
<dt>type_object</dt><dd><p>character indicating the type of object</p>
</dd>
</dl>


<hr>
<h2 id='species_fw'>species_fw</h2><span id='topic+species_fw'></span>

<h3>Description</h3>

<p>Data containing the abundance of 17 species  
expressed in spectral counts. Data generated from an Orbitrap 
Fusion Lumos  Tribrid Mass Spectrometer. The dataset contains 
the metaproteomes from three extraction methods: i) &quot;Q&quot; for Qiagen, 
ii) &quot;FW&quot; for fecal waters, and iii) &quot;Q_FW&quot; for the mixture 
of Qiagen and fecal waters. Data generated in the context of 
the project Microbiome Rapid Access (Université Paris-Saclay).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(species_fw)
</code></pre>


<h3>Format</h3>

<p>A list of four elements defined as &quot;spectral_count_object&quot; with 
taxonomic annotation generated with the function &quot;crumble_taxonomy&quot;
</p>

<dl>
<dt>SC_subgroups</dt><dd><p>dataframe with 9 samples containing 17 
especies with abundance expressed as spectral counts</p>
</dd>
<dt>metadata</dt><dd><p>information related to the 9 samples from the 
experiment</p>
</dd>
<dt>peptides_proteins</dt><dd><p>information related to each of the 
1557 identified peptides</p>
</dd>
<dt>type_object</dt><dd><p>character indicating the type of object</p>
</dd>
</dl>


<hr>
<h2 id='venn_methods'>venn methods</h2><span id='topic+venn_methods'></span>

<h3>Description</h3>

<p>Data containing the subgroups for each logical section of 
the Venn diagram (specific and intersections) from two methods of extraction. 
The extraction methods are: i) &quot;S&quot; for Qiagen, and ii) &quot;S_EF&quot; for
the mixture of Qiagen and fecal waters.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(venn_methods)
</code></pre>


<h3>Format</h3>

<p>A list of six elements defined as &quot;venn_lists_object&quot;, 
generated with the function &quot;plot_venn&quot;
</p>

<dl>
<dt>S_EF</dt><dd><p> 385 subgroups found in the method 'S_EF'</p>
</dd>
<dt>S</dt><dd><p> 242 subgroups found the method 'S'</p>
</dd>
<dt>intersection</dt><dd><p> 201 subgroups in common between the methods 'S' and 'S_EF'</p>
</dd>
<dt>Specific in S_EF</dt><dd><p> 184 specific subgroups found in the method 'S_EF'</p>
</dd>
<dt>S</dt><dd><p> 41 specific subgroups found in the method 'S'</p>
</dd>
<dt>type_object</dt><dd><p>character indicating the type of object</p>
</dd>
</dl>


</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
