<!DOCTYPE html><html lang="en"><head><title>Help for package MetaSKAT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MetaSKAT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Example'><p>Example dataset</p></a></li>
<li><a href='#Generate_Meta_Files'><p>Generate summary statistics files</p></a></li>
<li><a href='#Meta_Null_Model'><p>Get parameters and residuals from H0</p></a></li>
<li><a href='#MetaSKAT_MSSD_OneSet'><p>Meta analysis SKAT with summary data from each study cohort.</p></a></li>
<li><a href='#MetaSKAT_wZ'><p>Meta analysis SKAT with individual level genotype data</p></a></li>
<li><a href='#Open_MSSD_File_2Read'><p>Read Meta SSD and Info files</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Meta Analysis for SNP-Set (Sequence) Kernel Association Test</td>
</tr>
<tr>
<td>Version:</td>
<td>0.90</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-10-08</td>
</tr>
<tr>
<td>Author:</td>
<td>Seunggeun (Shawn) Lee [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Seunggeun (Shawn) Lee &lt;lee7801@snu.ac.kr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for Meta-analysis Burden Test, Sequence Kernel Association Test (SKAT) and Optimal SKAT (SKAT-O) by  Lee et al. (2013) &lt;<a href="https://doi.org/10.1016%2Fj.ajhg.2013.05.010">doi:10.1016/j.ajhg.2013.05.010</a>&gt;. These methods use summary-level score statistics to carry out gene-based meta-analysis for rare variants.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>Little Endian</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.13.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>SKAT (&ge; 2.0.1)</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-10-08 13:08:42 UTC; lee7801</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-10-08 13:30:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='Example'>Example dataset</h2><span id='topic+Example'></span>

<h3>Description</h3>

<p>Example dataset
</p>


<h3>Format</h3>

<p>This example dataset has the following objects:
</p>

<dl>
<dt>y.list</dt><dd><p>a list object of binary phenotypes. 
It has 3 elements for 3 study cohorts. Each element is a vector of binary phenotypes.</p>
</dd>
<dt>x.list</dt><dd><p>a list object of covariates. 
It has 3 elements for 3 study cohorts. Each element is a matrix of covariates. 
The first and last elements have two covariates (two columns), and the second element has one covariate (one column).</p>
</dd>
<dt>n.g</dt><dd><p>a numeric value of the number of cohorts (3).</p>
</dd>
<dt>Z.list</dt><dd><p>a list object of genotypes of all samples. It has 10 elements for 10 genes. 
Each element is an nxp matrix with n being the total sample size (3000) and p being the number of SNPs.</p>
</dd>
</dl>


<hr>
<h2 id='Generate_Meta_Files'>Generate summary statistics files</h2><span id='topic+Generate_Meta_Files'></span><span id='topic+Generate_Meta_Files_FromDosage'></span>

<h3>Description</h3>

<p>Generate Meta SSD (MSSD) and Meta Info (MInfo) files. 
Both files are needed to run MetaSKAT with summary statistics.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
	Generate_Meta_Files(obj, File.Bed, File.Bim
	, File.SetID, File.MSSD, File.MInfo, N.Sample
	, File.Permu = NULL, data=NULL, impute.method="fixed")
	
	Generate_Meta_Files_FromDosage(obj, File.Dosage
	, File.SetID, File.MSSD, File.MInfo, N.Sample
	, File.Permu=NULL, data=NULL, impute.method="fixed")


 </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Generate_Meta_Files_+3A_obj">obj</code></td>
<td>
<p>returned object from SKAT_Null_Model. </p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.bed">File.Bed</code></td>
<td>
<p>name of the binary ped file (BED).</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.bim">File.Bim</code></td>
<td>
<p>name of the binary bim file (BIM).</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.setid">File.SetID</code></td>
<td>
<p>name of the SNP set ID file. The first column must be Set ID, and the second column must be SNP ID. There should be no header!!</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.mssd">File.MSSD</code></td>
<td>
<p>name of the MSSD file that will be generated.</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.minfo">File.MInfo</code></td>
<td>
<p>name of the MInfo file that will be generated.</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_n.sample">N.Sample</code></td>
<td>
<p>number of samples.</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.permu">File.Permu</code></td>
<td>
<p>name of a file that will have score statistics from permuted phenotypes (currently internal use only).</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_data">data</code></td>
<td>
<p>an optional data frame containing the variables in the model (default=NULL). If it is NULL, the variables are taken from environment(formula).</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_impute.method">impute.method</code></td>
<td>
<p>a method to impute missing genotypes (default= &quot;fixed&quot;). &quot;bestguess&quot; imputes missing genotypes as the most likely values(0,1,2), 
&quot;random&quot; imputes missing genotypes by generating binomial(2,p) random variables (p = MAF), and &quot;fixed&quot; imputes missing genotypes by assigning the mean genotype values (2p).</p>
</td></tr>
<tr><td><code id="Generate_Meta_Files_+3A_file.dosage">File.Dosage</code></td>
<td>
<p>name of the dosage file. The dosage file must not have a header.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>These functions generate summary statistic files (MSSD and MInfo files) from plink binary files. 
To run meta analysis, each study should provide both MSSD and MInfo files. 
The MSSD is a binary file with between-SNP information matrices, 
and MInfo is a text file with information on study cohorts and SNPsets. 
</p>
<p>If users want to use dosages instead of hard call genotypes, 
Generate_Meta_Files_FromDosage should be used instead of Generate_Meta_Files. 
The dosage file should follow the plink dosage file format with a single dosage value per each 
SNP (Format=1 in plink). The first three columns should be SNP ID, allele type1 (a1) and allele type2 (a2). 
After the first three columns, there should be N.Sample columns of dosage data. 
Each column represents each sample, and the order of samples should be matched with the order in
phenotypes and covariates used in SKAT_Null_Model.  
</p>
<p>ex) 
</p>
<p>rs0001	A	T 	0.1	0.2 
</p>
<p>rs0002	C	G	1.2	0 	
</p>
<p>Dosage value is the expected number of a2 copies, and 0 .. 2 scale. So the value 
0.1 indicates that the expected number of copy of a2 is 0.1. 
</p>


<h3>Author(s)</h3>

<p>Seunggeun Lee</p>

<hr>
<h2 id='Meta_Null_Model'>Get parameters and residuals from H0</h2><span id='topic+Meta_Null_Model'></span><span id='topic+Meta_Null_Model_EmmaX'></span>

<h3>Description</h3>

<p>Compute model parameters and residuals under the null model (H0) of no associations. 
It can be used only when individual level data are available.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
Meta_Null_Model(y.list, x.list, n.cohort, out_type="C",  n.Resampling=0)
Meta_Null_Model_EmmaX(y.list, x.list, n.cohort, K.list=NULL, Kin.File.list=NULL)

 </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Meta_Null_Model_+3A_y.list">y.list</code></td>
<td>
<p>a list object for phenotypes. Each element should be a vector of phenotypes. If you have 3 cohorts, it should have 3 elements.</p>
</td></tr>
<tr><td><code id="Meta_Null_Model_+3A_x.list">x.list</code></td>
<td>
<p>a list object for covariates. Each element should be a vector or a  matrix of covariates. If there are 3 cohorts, it should have 3 elements. 
If there are no covariates to adjust for, the element should be &quot;intercept&quot;. See the examples.</p>
</td></tr>
<tr><td><code id="Meta_Null_Model_+3A_n.cohort">n.cohort</code></td>
<td>
<p>a numeric value of the number of cohort.</p>
</td></tr>
<tr><td><code id="Meta_Null_Model_+3A_out_type">out_type</code></td>
<td>
<p>an indicator for the outcome type. &quot;C&quot; for continuous outcomes and &quot;D&quot; for dichotomous outcomes.</p>
</td></tr> 
<tr><td><code id="Meta_Null_Model_+3A_n.resampling">n.Resampling</code></td>
<td>
<p>internal use only.</p>
</td></tr>    
<tr><td><code id="Meta_Null_Model_+3A_k.list">K.list</code></td>
<td>
<p>a list object of the kinship matrices. If K.list=NULL, the function reads files in Kin.File.list to obtain kinship matrices.</p>
</td></tr>
<tr><td><code id="Meta_Null_Model_+3A_kin.file.list">Kin.File.list</code></td>
<td>
<p>a list object of emmax-kin output file names. </p>
</td></tr> 
</table>


<h3>Value</h3>

<p>It returns an object that has model parameters and residuals. The returned object will be used to run MetaSKAT_wZ. 
</p>


<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(Example)
attach(Example)

#############################################################
#	Compute a p-value of the first gene

obj&lt;-Meta_Null_Model(y.list, x.list, n.cohort=3, out_type="D")
MetaSKAT_wZ(Z.list[[1]], obj)$p.value

#############################################################
#	If you want to use the intercept-only model for the 2nd cohort

x.list[[2]]&lt;-"intercept"
obj&lt;-Meta_Null_Model(y.list, x.list, n.cohort=3, out_type="D")
MetaSKAT_wZ(Z.list[[1]], obj)$p.value


</code></pre>

<hr>
<h2 id='MetaSKAT_MSSD_OneSet'>Meta analysis SKAT with summary data from each study cohort.</h2><span id='topic+MetaSKAT_MSSD_OneSet'></span><span id='topic+MetaSKAT_MSSD_ALL'></span>

<h3>Description</h3>

<p>Meta analysis SKAT with Meta SSD (MSSD) and Info (MInfo) files. MetaSKAT_MSSD_OneSet computes a p-value for a given set,
and MetaSKAT_MSSD_ALL computes p-values for all sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
MetaSKAT_MSSD_OneSet(Cohort.Info, SetID, combined.weight=TRUE, weights.beta=c(1,25),
method="davies", r.corr=0, is.separate = FALSE, Group_Idx=NULL, MAF.cutoff=1, 
missing_cutoff=0.15)

MetaSKAT_MSSD_ALL(Cohort.Info, ...)

 </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_cohort.info">Cohort.Info</code></td>
<td>
<p>output object from Open_MSSD_File_2Read. </p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_setid">SetID</code></td>
<td>
<p>a character value of set id to test.</p>
</td></tr> 
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_combined.weight">combined.weight</code></td>
<td>
<p>a logical value (default=TRUE) for a type of weighting. See MetaSKAT_wZ page for details.</p>
</td></tr> 
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_weights.beta">weights.beta</code></td>
<td>
<p>a numeric vector of parameters for the beta weights (default=c(1,25))</p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_method">method</code></td>
<td>
<p>a method to compute a p-value (default= &quot;davies&quot;). See MetaSKAT_wZ page for details.</p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_r.corr">r.corr</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for the compound symmetric correlation structure kernels (default= 0). See MetaSKAT_wZ page for details.</p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_is.separate">is.separate</code></td>
<td>
<p>a logical value (default=FALSE) for homogeneous(=FALSE) or heterogeneous(=TRUE) genetic effects
of a SNP set across studies. See MetaSKAT_wZ page for details.</p>
</td></tr>  
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_group_idx">Group_Idx</code></td>
<td>
<p>a vector of group indicator (default=NULL). See MetaSKAT_wZ page for details.</p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_maf.cutoff">MAF.cutoff</code></td>
<td>
<p>a cutoff of the MAFs of SNPs (default=1). Any SNPs with MAFs &gt; MAF.cutoff will be excluded from the analysis.</p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_missing_cutoff">missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15). See MetaSKAT_wZ page for details.</p>
</td></tr>
<tr><td><code id="MetaSKAT_MSSD_OneSet_+3A_...">...</code></td>
<td>
<p>the same parameters of MetaSKAT_MSSD_OneSet after SetID.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please see MetaSKAT_wZ for details.
</p>


<h3>Value</h3>

<p>MetaSKAT_MSSD_OneSet and MetaSKAT_wZ return the same object. See MetaSKAT_wZ for details.
MetaSKAT_MSSD_ALL returns a dataframe with SetIDs (first column) and p-values (second column). 
</p>


<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>References</h3>

<p>Lee, S., Teslovich, T., Boehnke, M., Lin, X. (2013)
General framework for meta-analysis of rare variants in sequencing association studies.  <em>American Journal of Human Genetics</em>, 93, 42-53. 
</p>

<hr>
<h2 id='MetaSKAT_wZ'>Meta analysis SKAT with individual level genotype data</h2><span id='topic+MetaSKAT_wZ'></span>

<h3>Description</h3>

<p>Meta analysis SKAT with individual level genotype data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
MetaSKAT_wZ(Z, obj, combined.weight=TRUE, weights.beta=c(1,25), 
method="davies", r.corr=0, is.separate = FALSE, Group_Idx=NULL, 
impute.method="fixed",impute.estimate.maf=1, missing_cutoff=0.15)


 </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MetaSKAT_wZ_+3A_z">Z</code></td>
<td>
<p>a numeric genotype matrix with each row as a different individual and each column as a separate snp. 
Each genotype should be coded as 0, 1, 2, and 9 (or NA) for AA, Aa, aa, and missing, 
where A is a major allele and a is a minor allele. Missing genotypes will be imputed using observed MAFs. </p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_obj">obj</code></td>
<td>
<p>an output object from the Meta_Null_Model function. </p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_combined.weight">combined.weight</code></td>
<td>
<p>a logical value (default=TRUE) for the type of weighting. 
If it is TRUE, a weight for each SNP is computed using MAFs that are common across studies. 
If it is FALSE, group specific weights will be used based on group specific MAFs.</p>
</td></tr> 
<tr><td><code id="MetaSKAT_wZ_+3A_weights.beta">weights.beta</code></td>
<td>
<p>a numeric vector of parameters of beta weights (default=c(1,25))</p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_method">method</code></td>
<td>
<p>a method to compute a p-value (default= &quot;davies&quot;). 
&quot;davies&quot; represents an exact method that computes a p-value by inverting the characteristic function of the mixture chisq dist.,
and &quot;optimal&quot; represents the optimal test (SKAT-O) that is based on an optimal linear combination  of burden and SKAT statistics.
&quot;optimal&quot; is equivalent to &quot;optimal.adj&quot; in the SKAT function.</p>
</td></tr>      
<tr><td><code id="MetaSKAT_wZ_+3A_r.corr">r.corr</code></td>
<td>
<p>the <code class="reqn">\rho</code> parameter for the compound symmetric correlation structure kernel (default= 0). If r.corr=0, it does the SKAT test.  If r.corr=1, it does the burden test. 
If r.corr=(a vector of grid values between 0 and 1), it does SKAT-O. r.corr will be ignored if method=&quot;optimal&quot;. See the manual of SKAT.</p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_is.separate">is.separate</code></td>
<td>
<p>a logical value (default=FALSE) for homogeneous(=FALSE) or heterogeneous(=TRUE) genetic effects
of a SNP set across studies.
When FALSE, it is assumed that all studies share the same causal variants with the same effect size. 
When TRUE, it is assumed that studies/groups may have different causal variants. </p>
</td></tr>  
<tr><td><code id="MetaSKAT_wZ_+3A_group_idx">Group_Idx</code></td>
<td>
<p>a vector of group indicator (default=NULL). 
If a vector of integers are specified, it assumes causal variants are the same for studies with the same group index, and different for studies  with different group indexes.
When NULL, studies are assumed to be in different groups with different group indexes.
When is.separate=FALSE, it will be ignored.</p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_impute.method">impute.method</code></td>
<td>
<p>a method to impute missing genotypes (default= &quot;fixed&quot;). &quot;bestguess&quot; imputes missing genotypes as the most likely values(0,1,2), 
&quot;random&quot; imputes missing genotypes by generating binomial(2,p) random variables (p = MAF), and &quot;fixed&quot; imputes missing genotypes by assigning the mean genotype value (2p).</p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_impute.estimate.maf">impute.estimate.maf</code></td>
<td>
<p> a numeric value indicating how to estimate MAFs for the imputation. If
impute.estimate.maf=1 (default), MetaSKAT uses study-specific MAFs, in which each study MAFs will be used for the imputation. 
If impute.estimate.maf=2, all samples in the Z matrix will be used to calculate MAFs for
the imputation. If impute.estimate.maf=3, MetaSKAT uses group-specific MAFs.
Previous versions (&lt; ver 0.6) used impute.estimate.maf=2 as a default.</p>
</td></tr>
<tr><td><code id="MetaSKAT_wZ_+3A_missing_cutoff">missing_cutoff</code></td>
<td>
<p>a cutoff of the missing rates of SNPs (default=0.15). If the first study has SNPs with missing rates higher than the cutoff, 
these SNPs in the study will be excluded from the analysis. However, the same SNPs in other studies will not be excluded, if their missing rates are lower than the cutoff. The missing
rates are calculated study by study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The rows of Z should be matched with phenotypes and covariates. 
If there are 3 studies, and study 1,2, and 3 have n1, n2, and n3 samples, 
the first n1, n2, and n3 rows of Z should be genotypes of the first, second, and third studies, respectively.
</p>
<p>Group_Idx is a vector of group index. Suppose the first two studies are European-based and the last study is African American-based.
If you want to run MetaSKAT with assuming ancestry group specific heterogeneity, you can set Group_Idx=c(1,1,2), which indicates
the first two studies belong to the same group. 
</p>
<p>The four methods in the MetaSKAT paper can be run with the following parameters: 
</p>

<ol>
<li><p> Hom-Meta-SKAT:  combined.weight=TRUE, is.separate=FALSE 
</p>
</li>
<li><p> Hom-Meta-SKAT-O: combined.weight=TRUE, is.separate=FALSE, method=&quot;optimal&quot; 
</p>
</li>
<li><p> Het-Meta-SKAT:  combined.weight=FALSE, is.separate=TRUE 
</p>
</li>
<li><p> Het-Meta-SKAT-O: combined.weight=FALSE, is.separate=TRUE, method=&quot;optimal&quot; 
</p>
</li></ol>



<h3>Value</h3>

<table role = "presentation">
<tr><td><code>p.value</code></td>
<td>
<p>p-value. </p>
</td></tr>
<tr><td><code>param</code></td>
<td>
<p>estimated parameters of each method.</p>
</td></tr>   
<tr><td><code>param$Is_Converged</code></td>
<td>
<p> (only with method=&quot;davies&quot;) an indicator for the convergence. 1 indicates convergence and 0 otherwise. 
When 0 (not converged), &quot;liu&quot; method will used to compute a p-value. </p>
</td></tr>  
</table>


<h3>Author(s)</h3>

<p>Seunggeun Lee</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(Example)
attach(Example)

#############################################################
#	Compute a p-value of the first gene

obj&lt;-Meta_Null_Model(y.list, x.list, n.cohort=3, out_type="D")

# rho=0
MetaSKAT_wZ(Z.list[[1]], obj)$p.value

# rho=1 (burden test)
MetaSKAT_wZ(Z.list[[1]], obj, r.corr=1)$p.value


# optimal test
MetaSKAT_wZ(Z.list[[1]], obj, method="optimal")$p.value

# cohort specific weights
MetaSKAT_wZ(Z.list[[1]], obj, combined.weight=FALSE)$p.value


# Seperate = TRUE
# Assume heterogeneous genetic effect 
MetaSKAT_wZ(Z.list[[1]], obj, combined.weight=FALSE, is.separate = TRUE)$p.value


# Group

# the first two cohorts are in the same group.
Group_Idx=c(1,1,2)
MetaSKAT_wZ(Z.list[[1]], obj, combined.weight=FALSE, is.separate = TRUE,Group_Idx=Group_Idx)$p.value

# all three cohorts are in different group.
Group_Idx=c(1,2,3)
MetaSKAT_wZ(Z.list[[1]], obj, combined.weight=FALSE, is.separate = TRUE,Group_Idx=Group_Idx)$p.value



</code></pre>

<hr>
<h2 id='Open_MSSD_File_2Read'>Read Meta SSD and Info files</h2><span id='topic+Open_MSSD_File_2Read'></span>

<h3>Description</h3>

<p>Read Meta SSD (MSSD) and Meta Info (MInfo) files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
	Open_MSSD_File_2Read(File.MSSD.vec, File.MInfo.vec)

 </code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Open_MSSD_File_2Read_+3A_file.mssd.vec">File.MSSD.vec</code></td>
<td>
<p>a vector of MSSD files. Each element represents a MSSD file of each study.</p>
</td></tr>
<tr><td><code id="Open_MSSD_File_2Read_+3A_file.minfo.vec">File.MInfo.vec</code></td>
<td>
<p>a vector of MInfo files. Each element represents a MInfo file of each study.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Users should open MSSD and MInfo files to run MetaSKAT_MSSD_OneSet or MetaSKAT_MSSD_ALL.
If all individual level data are available, use MetaSKAT_wZ instead. 
</p>


<h3>Value</h3>

<p>This function returns an data object. The returned object will be used to run MetaSKAT_MSSD_OneSet or MetaSKAT_MSSD_ALL.
</p>


<h3>Author(s)</h3>

<p>Seunggeun Lee</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
