<!DOCTYPE html><html lang="en"><head><title>Help for package metevalue</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {metevalue}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#demo_biseq_DMR'><p>BiSeq Output Demo Dataset</p></a></li>
<li><a href='#demo_biseq_methyrate'><p>BiSeq Methyrate Demo Dataset</p></a></li>
<li><a href='#demo_desq_out'><p>DESeq Output Dataset</p></a></li>
<li><a href='#demo_DMRfinder_DMRs'><p>DMRfinder Output Demo Dataset</p></a></li>
<li><a href='#demo_DMRfinder_rate_combine'><p>DMRfinder Methyrate Demo Dataset</p></a></li>
<li><a href='#demo_methylkit_met_all'><p>Methyrate output dataset from methylKit</p></a></li>
<li><a href='#demo_methylkit_methyrate'><p>Methyrate Dataset</p></a></li>
<li><a href='#demo_metilene_input'><p>Metilene Methyrate Demo Dataset</p></a></li>
<li><a href='#demo_metilene_out'><p>Metilene Demo Output Dataset</p></a></li>
<li><a href='#evalue_buildin_sql'><p>Build-in data process function</p></a></li>
<li><a href='#evalue_buildin_var_fmt_nm'><p>Build-in check file format function</p>
Perform the format check and data clean for the  &quot;metilene&quot; or &quot;biseq&quot;, &quot;DMRfinder&quot; or &quot;methylKit&quot; method correspondingly.</a></li>
<li><a href='#metevalue.biseq'><p>Calculate E-value of the BiSeq data format</p></a></li>
<li><a href='#metevalue.biseq.chk'><p>Check the BiSeq data format</p></a></li>
<li><a href='#metevalue.DMRfinder'><p>Calculate E-value of the DMRfinder data format</p></a></li>
<li><a href='#metevalue.DMRfinder.chk'><p>Check the DMRfinder data format</p></a></li>
<li><a href='#metevalue.methylKit'><p>Calculate E-value of the methylKit data format</p></a></li>
<li><a href='#metevalue.methylKit.chk'><p>Check the methylKit data format</p></a></li>
<li><a href='#metevalue.metilene'><p>Calculate E-value of the Metilene data format</p></a></li>
<li><a href='#metevalue.metilene.chk'><p>Check the Metilene data format</p></a></li>
<li><a href='#metevalue.RNA_general'><p>A general method to calculate the e-value for RNA-seq data.</p></a></li>
<li><a href='#varevalue.metilene'><p>Calculate E-value of the Metilene data</p></a></li>
<li><a href='#varevalue.single_general'><p>A general method to calculate the e-value for other DNA methylation tools not described above. The input data is the DNA methylation rates using the similar format with Metilene.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>E-Value in the Omics Data Association Studies</td>
</tr>
<tr>
<td>Version:</td>
<td>0.2.4</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-05-26</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Yifan Yang &lt;yfyang.86@hotmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>In the omics data association studies, it is common to conduct the p-value corrections to control the false significance. Beyond the P-value corrections, E-value is recently studied to facilitate multiple testing correction based on V. Vovk and R. Wang (2021) &lt;<a href="https://doi.org/10.1214%2F20-AOS2020">doi:10.1214/20-AOS2020</a>&gt;. This package provides E-value calculation for DNA methylation data and RNA-seq data. Currently, five data formats are supported: DNA methylation levels using DMR detection tools (BiSeq, DMRfinder, MethylKit, Metilene and other DNA methylation tools) and RNA-seq data. The relevant references are listed below: Katja Hebestreit and Hans-Ulrich Klein (2022) &lt;<a href="https://doi.org/10.18129%2FB9.bioc.BiSeq">doi:10.18129/B9.bioc.BiSeq</a>&gt;; Altuna Akalin et.al (2012) &lt;<a href="https://doi.org/10.18129%2FB9.bioc.methylKit">doi:10.18129/B9.bioc.methylKit</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.apache.org/licenses/LICENSE-2.0">Apache License (&ge; 2)</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>sqldf, psych, dplyr, R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, prettydoc, knitr, ggplot2, tidyr, testthat (&ge;
3.0.0)</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Yifan Yang [aut, cre, cph],
  Xiaoqing Pan [aut],
  Haoyuan Liu [aut]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-05-26 13:20:20 UTC; yifanyang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-05-26 15:20:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='demo_biseq_DMR'>BiSeq Output Demo Dataset</h2><span id='topic+demo_biseq_DMR'></span>

<h3>Description</h3>

<p>The dummy output for BiSeq illustrating purpose. It is dummy.
</p>


<h3>Details</h3>

<p>- seqnames: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- width
</p>
<p>- strand: Strand
</p>
<p>- median.p
</p>
<p>- median.meth.group1
</p>
<p>- median.meth.group2
</p>
<p>- median.meth.diff
</p>
<p>Notice that there are &quot;NaN&quot; within the feature columns.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>

<hr>
<h2 id='demo_biseq_methyrate'>BiSeq Methyrate Demo Dataset</h2><span id='topic+demo_biseq_methyrate'></span>

<h3>Description</h3>

<p>The methyrate for BiSeq illustrating purpose. It is dummy.
</p>


<h3>Details</h3>

<p>The data includes 12 columns.
</p>
<p>- chr: string Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups, repeat 5 times.
Notice that there are &quot;NaN&quot; within the feature columns.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>

<hr>
<h2 id='demo_desq_out'>DESeq Output Dataset</h2><span id='topic+demo_desq_out'></span>

<h3>Description</h3>

<p>The output dummy data for &quot;RNA&quot; meythod illustrating purpose.
</p>


<h3>Details</h3>

<p>The data includes 10 columns.
</p>
<p>- treated1fb:
</p>
<p>- treated2fb:
</p>
<p>- treated3fb:
</p>
<p>- untreated1fb:
</p>
<p>- untreated2fb:
</p>
<p>- untreated3fb:
</p>
<p>- untreated4fb:
</p>
<p>This data contains 8166 rows and 7 columns.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># library("pasilla")
# pasCts &lt;- system.file("extdata",
#                       "pasilla_gene_counts.tsv",
#                       package="pasilla", mustWork=TRUE)
# pasAnno &lt;- system.file("extdata",
#                        "pasilla_sample_annotation.csv",
#                        package="pasilla", mustWork=TRUE)
# cts &lt;- as.matrix(read.csv(pasCts,sep="\t",row.names="gene_id"))
# coldata &lt;- read.csv(pasAnno, row.names=1)
# coldata &lt;- coldata[,c("condition","type")]
# coldata$condition &lt;- factor(coldata$condition)
# coldata$type &lt;- factor(coldata$type)
# 
# library("DESeq2")
# colnames(cts)=paste0(colnames(cts),'fb')
# cts = cts[,rownames(coldata)]
# dds &lt;- DESeqDataSetFromMatrix(countData = cts,
#                               colData = coldata,
#                               design = ~ condition)
# dds &lt;- DESeq(dds)
# 
# 
# dat &lt;- t(t(cts)/(dds$sizeFactor)) 
# dat.out &lt;- dat[rowSums(dat &gt;5)&gt;=0.8*ncol(dat),]
# 
# demo_desq_out &lt;- log(dat.out)
</code></pre>

<hr>
<h2 id='demo_DMRfinder_DMRs'>DMRfinder Output Demo Dataset</h2><span id='topic+demo_DMRfinder_DMRs'></span>

<h3>Description</h3>

<p>The output dummy dataset for DMRfinder illustrating purpose.
</p>


<h3>Details</h3>

<p>The data includes 6 columns.
</p>
<p>- chr: string Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups, repeat 2 times.
Notice that there are &quot;NaN&quot; within the feature columns.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>

<hr>
<h2 id='demo_DMRfinder_rate_combine'>DMRfinder Methyrate Demo Dataset</h2><span id='topic+demo_DMRfinder_rate_combine'></span>

<h3>Description</h3>

<p>The methyrate for BiSeq illustrating purpose. It is dummy.
</p>


<h3>Details</h3>

<p>The data includes 6 columns.
</p>
<p>- chr: string Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups, repeat 2 times.
Notice that there are &quot;NaN&quot; within the feature columns.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>

<hr>
<h2 id='demo_methylkit_met_all'>Methyrate output dataset from methylKit</h2><span id='topic+demo_methylkit_met_all'></span>

<h3>Description</h3>

<p>The methyrate dataset samples &quot;myCpG&quot; data from the methylKit (a bioconductor package) for illustrating purpose.
</p>


<h3>Details</h3>

<p>The data includes 7 columns:
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- strand: Strand
</p>
<p>- pvalue: The adjusted p-value based on BH method in MWU-test
</p>
<p>- qvalue: cutoff for qvalue of differential methylation statistic
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>


<h3>References</h3>

<p>Akalin, Altuna, et al. &quot;methylKit: a comprehensive R package for the analysis of genome-wide DNA methylation profiles.&quot; Genome biology 13.10 (2012): 1-9. doi: <a href="https://doi.org/10.1186/gb-2012-13-10-r87">10.1186/gb-2012-13-10-r87</a>
</p>

<hr>
<h2 id='demo_methylkit_methyrate'>Methyrate Dataset</h2><span id='topic+demo_methylkit_methyrate'></span>

<h3>Description</h3>

<p>The methyrate dataset samples &quot;myCpG&quot; data from the methylKit (a bioconductor package) for illustrating purpose.
</p>


<h3>Details</h3>

<p>The data includes 6 columns.
</p>
<p>- chr: string Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups (4 columns)
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>


<h3>References</h3>

<p>Akalin, Altuna, et al. &quot;methylKit: a comprehensive R package for the analysis of genome-wide DNA methylation profiles.&quot; Genome biology 13.10 (2012): 1-9. doi: <a href="https://doi.org/10.1186/gb-2012-13-10-r87">10.1186/gb-2012-13-10-r87</a>
</p>

<hr>
<h2 id='demo_metilene_input'>Metilene Methyrate Demo Dataset</h2><span id='topic+demo_metilene_input'></span>

<h3>Description</h3>

<p>The methyrate for metilene illustrating purpose. It is dummy.
</p>


<h3>Details</h3>

<p>The data includes 18 columns.
</p>
<p>- chr: string Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups.
</p>
<p>Notice that there are &quot;NaN&quot; within the feature columns.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>

<hr>
<h2 id='demo_metilene_out'>Metilene Demo Output Dataset</h2><span id='topic+demo_metilene_out'></span>

<h3>Description</h3>

<p>The output dummy data for &quot;metilene&quot; meythod illustrating purpose.
</p>


<h3>Details</h3>

<p>The data includes 10 columns.
</p>
<p>- V1: string Chromosome
</p>
<p>- V2: The positions of the start sites of the corresponding region
</p>
<p>- V3: The positions of the end sites of the corresponding region
</p>
<p>- V4- V10: data value.
</p>
<p>Please check the vignette &quot;metevalue&quot; for details.
</p>

<hr>
<h2 id='evalue_buildin_sql'>Build-in data process function</h2><span id='topic+evalue_buildin_sql'></span>

<h3>Description</h3>

<p>Build-in data process function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evalue_buildin_sql(a, b, method = "metilene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evalue_buildin_sql_+3A_a">a</code></td>
<td>
<p>data frame of the methylation rate</p>
</td></tr>
<tr><td><code id="evalue_buildin_sql_+3A_b">b</code></td>
<td>
<p>data frame of output data corresponding to the &quot;method&quot; option</p>
</td></tr>
<tr><td><code id="evalue_buildin_sql_+3A_method">method</code></td>
<td>
<p>&quot;metilene&quot; or &quot;biseq&quot;, &quot;DMRfinder&quot; or &quot;methylKit&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a data frame combines data frame a and b corresponding to the &quot;method&quot; option
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("demo_metilene_out")
data("demo_metilene_input")
result = evalue_buildin_var_fmt_nm(demo_metilene_input,
                                   demo_metilene_out, method="metilene")
result_sql = evalue_buildin_sql(result$a, result$b, method="metilene")

</code></pre>

<hr>
<h2 id='evalue_buildin_var_fmt_nm'>Build-in check file format function
Perform the format check and data clean for the  &quot;metilene&quot; or &quot;biseq&quot;, &quot;DMRfinder&quot; or &quot;methylKit&quot; method correspondingly.</h2><span id='topic+evalue_buildin_var_fmt_nm'></span>

<h3>Description</h3>

<p>Build-in check file format function
Perform the format check and data clean for the  &quot;metilene&quot; or &quot;biseq&quot;, &quot;DMRfinder&quot; or &quot;methylKit&quot; method correspondingly.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>evalue_buildin_var_fmt_nm(a, b, method = "metilene")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="evalue_buildin_var_fmt_nm_+3A_a">a</code></td>
<td>
<p>data frame of the methylation rate</p>
</td></tr>
<tr><td><code id="evalue_buildin_var_fmt_nm_+3A_b">b</code></td>
<td>
<p>data frame of output data corresponding to the &quot;method&quot; option</p>
</td></tr>
<tr><td><code id="evalue_buildin_var_fmt_nm_+3A_method">method</code></td>
<td>
<p>&quot;metilene&quot; or &quot;biseq&quot;, &quot;DMRfinder&quot; or &quot;methylKit&quot;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(a, b) which contains the cleaned data correspondingly
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("demo_metilene_out")
data("demo_metilene_input")
evalue_buildin_var_fmt_nm(demo_metilene_input,
                          demo_metilene_out, method="metilene")

</code></pre>

<hr>
<h2 id='metevalue.biseq'>Calculate E-value of the BiSeq data format</h2><span id='topic+metevalue.biseq'></span>

<h3>Description</h3>

<p>Please check vignette &quot;metevalue&quot; for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.biseq(
  methyrate,
  BiSeq.output,
  adjust.methods = "BH",
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.biseq_+3A_methyrate">methyrate</code></td>
<td>
<p>is the methyrate file.  For example: 
</p>

<table>
<tr>
 <td style="text-align: right;">
chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>
<p>The columns are (in order):
</p>
<p>- chr: Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups</p>
</td></tr>
<tr><td><code id="metevalue.biseq_+3A_biseq.output">BiSeq.output</code></td>
<td>
<p>is the output file of BiSeq. The columns are (in order):
</p>
<p>- seqnames: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- width: The number of CpG sites within the corresponding region
</p>
<p>- strand: Strand
</p>
<p>- median.p: The median p-value among CpG sites within the corresponding region
</p>
<p>- median.meth.group1: The median methylation rate in the first group among CpG sites within the corresponding region
</p>
<p>- median.meth.group2: The median methylation rate in the second group among CpG sites within the corresponding region
</p>
<p>- median.meth.diff: The median methylation difference between groups among CpG sites within the corresponding region</p>
</td></tr>
<tr><td><code id="metevalue.biseq_+3A_adjust.methods">adjust.methods</code></td>
<td>
<p>is the adjust methods of e-value. It can be 'bonferroni', 'hochberg', 'holm', 'hommel', 'BH', 'BY'</p>
</td></tr>
<tr><td><code id="metevalue.biseq_+3A_sep">sep</code></td>
<td>
<p>seperator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.biseq_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the BiSeq.output file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe, the columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2
</p>
<p>- e_value: The e-value of the corresponding region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#\donttest{
#data("demo_biseq_methyrate")
#data("demo_biseq_DMR")
#example_tempfiles = tempfile(c("demo_biseq_methyrate", "demo_biseq_DMR"))
#tempdir()
#### write to temp file ####
#write.table(demo_biseq_methyrate, file=example_tempfiles[1],row.names=FALSE,
#            col.names=TRUE, quote=FALSE, sep='\t')
#write.table(demo_biseq_DMR, file=example_tempfiles[2],
#             sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#### compute e-value and its adjustment ####
#result = metevalue.biseq(example_tempfiles[1],
#                         example_tempfiles[2], bheader = TRUE)
#}
</code></pre>

<hr>
<h2 id='metevalue.biseq.chk'>Check the BiSeq data format</h2><span id='topic+metevalue.biseq.chk'></span>

<h3>Description</h3>

<p>Check the BiSeq data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.biseq.chk(
  input_filename_a,
  input_filename_b,
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.biseq.chk_+3A_input_filename_a">input_filename_a</code></td>
<td>
<p>metilene input file path. This file is a sep (e.g. TAB) separated file with two key columns and several value columns:
For exampe:
</p>

<table>
<tr>
 <td style="text-align: right;">
chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>- chr and pos are keys;
</p>
<p>- g1~g2: methylation rate data in groups.</p>
</td></tr>
<tr><td><code id="metevalue.biseq.chk_+3A_input_filename_b">input_filename_b</code></td>
<td>
<p>metilene input file path. This file should stored as a sep(e.g. TAB) separated file with two key columns and several value columns:
The columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The position of the start site of the corresponding region
</p>
<p>- end: The position of the end site of the corresponding region
</p>
<p>- range: The range of the corresponding region
</p>
<p>- strand: Strand
</p>
<p>- median.p:  The median of p-values in the corresponding region
</p>
<p>- median.meth.group1 : The median of methylation level for the corresponding segment of group 1
</p>
<p>- median.meth.group2 : The median of methylation level for the corresponding segment of group 2
</p>
<p>- median.meth.diff:  The median of the difference between the methylation level</p>
</td></tr>
<tr><td><code id="metevalue.biseq.chk_+3A_sep">sep</code></td>
<td>
<p>separator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.biseq.chk_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the input_filename_b file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(file_a, file_b, file_a_b) returns a list with three pr-handled data.frames corresponding to the input_filename_a, input_filename_b file and a A JOIN B file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data("demo_biseq_methyrate")
#data("demo_biseq_DMR")
#example_tempfiles = tempfile(c("demo_biseq_methyrate", "demo_biseq_DMR"))
#tempdir()
#write.table(demo_biseq_methyrate, file=example_tempfiles[1],row.names=FALSE,
#            col.names=TRUE, quote=FALSE, sep='\t')
#write.table(demo_biseq_DMR, file=example_tempfiles[2],
#             sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#### compute e-value and its adjustment ####
#result = metevalue.biseq.chk(example_tempfiles[1],
#                         example_tempfiles[2], bheader = TRUE)
</code></pre>

<hr>
<h2 id='metevalue.DMRfinder'>Calculate E-value of the DMRfinder data format</h2><span id='topic+metevalue.DMRfinder'></span>

<h3>Description</h3>

<p>Calculate E-value of the DMRfinder data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.DMRfinder(
  methyrate,
  DMRfinder.output,
  adjust.methods = "BH",
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.DMRfinder_+3A_methyrate">methyrate</code></td>
<td>
<p>is the methyrate file. For example:
</p>

<table>
<tr>
 <td style="text-align: right;">
chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>The columns are (in order):
</p>
<p>- chr: Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder_+3A_dmrfinder.output">DMRfinder.output</code></td>
<td>
<p>is the output file of DMRfinder.
</p>
<p>- chr: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- CpG: The number of CpG sites within the corresponding region
</p>
<p>- Control.mu: The average methylation rate in control group
</p>
<p>- Expt1.mu: The average methylation rate in experiment group
</p>
<p>- Control.Expt1.diff: The methylation difference between control and experiment groups
</p>
<p>- Control.Expt1.pval: P-value based on Wald-test.</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder_+3A_adjust.methods">adjust.methods</code></td>
<td>
<p>is the adjust methods of e-value. It can be 'bonferroni', 'hochberg', 'holm', 'hommel', 'BH', 'BY'</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder_+3A_sep">sep</code></td>
<td>
<p>seperator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the DMRfinder.output file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe, the columns are (in order):
</p>
<p>- chr: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2
</p>
<p>- e_value: The e-value of the corresponding region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(demo_DMRfinder_rate_combine)
data(demo_DMRfinder_DMRs)
#example_tempfiles = tempfile(c("rate_combine", "DMRfinder_out"))
#tempdir()
#write.table(demo_DMRfinder_rate_combine, file=example_tempfiles[1],
#      row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
#write.table(demo_DMRfinder_DMRs, file=example_tempfiles[2],
#      sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#result = metevalue._DMRfinder(example_tempfiles[1], example_tempfiles[2],
#      bheader = TRUE)
#head(result)
</code></pre>

<hr>
<h2 id='metevalue.DMRfinder.chk'>Check the DMRfinder data format</h2><span id='topic+metevalue.DMRfinder.chk'></span>

<h3>Description</h3>

<p>Check the DMRfinder data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.DMRfinder.chk(
  input_filename_a,
  input_filename_b,
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.DMRfinder.chk_+3A_input_filename_a">input_filename_a</code></td>
<td>
<p>the combined data of methylation rate file. This file is a sep (e.g. TAB) separated file with two key columns and several value columns.
For exampe:
</p>

<table>
<tr>
 <td style="text-align: right;">
chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>The columns are (in order):
</p>
<p>- chr and pos are keys;
</p>
<p>- g1~g2: methylation rate data in groups.</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder.chk_+3A_input_filename_b">input_filename_b</code></td>
<td>
<p>the output file of DMRfinder.
The columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The position of the start sites of the corresponding region
</p>
<p>- end: The position of the end sites of the corresponding region
</p>
<p>- CpG: The number of CpG sites within the corresponding region
</p>
<p>- 'Control:mu': The absolute mean methylation level for the corresponding segment of the control group
</p>
<p>- 'Exptl:mu': The absolute mean methylation level for the corresponding segment of the experimental group
</p>
<p>- 'Control-&gt;Exptl:diff': The difference between the group means of methylation level
</p>
<p>- p: p-value</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder.chk_+3A_sep">sep</code></td>
<td>
<p>separator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.DMRfinder.chk_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the input_filename_b file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(file_a, file_b, file_a_b) returns a list with three pr-handled data.frames corresponding to the input_filename_a, input_filename_b file and a A JOIN B file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(demo_DMRfinder_rate_combine)
data(demo_DMRfinder_DMRs)
#example_tempfiles = tempfile(c("rate_combine", "DMRfinder_out"))
#tempdir()
#write.table(demo_DMRfinder_rate_combine, file=example_tempfiles[1],
#      row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
#write.table(demo_DMRfinder_DMRs, file=example_tempfiles[2],
#      sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#result = metevalue.DMRfinder.chk(example_tempfiles[1], example_tempfiles[2],
#      bheader = TRUE)
</code></pre>

<hr>
<h2 id='metevalue.methylKit'>Calculate E-value of the methylKit data format</h2><span id='topic+metevalue.methylKit'></span>

<h3>Description</h3>

<p>Calculate E-value of the methylKit data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.methylKit(
  methyrate,
  methylKit.output,
  adjust.methods = "BH",
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.methylKit_+3A_methyrate">methyrate</code></td>
<td>
<p>is the data of methylation rates of each sites and group. For example:
</p>

<table>
<tr>
 <td style="text-align: right;">
chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>The columns are (in order):
</p>
<p>- chr: Chromosome
</p>
<p>- pos: int Position
</p>
<p>- g1~g2: methylation rate data in groups</p>
</td></tr>
<tr><td><code id="metevalue.methylKit_+3A_methylkit.output">methylKit.output</code></td>
<td>
<p>is the output data with e-value of each region
</p>
<p>- chr: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- strand: Strand
</p>
<p>- pvalue: The adjusted p-value based on BH method in MWU-test
</p>
<p>- qvalue: cutoff for qvalue of differential methylation statistic
</p>
<p>- methyl.diff: The difference between the group means of methylation level</p>
</td></tr>
<tr><td><code id="metevalue.methylKit_+3A_adjust.methods">adjust.methods</code></td>
<td>
<p>is the adjust methods of e-value. It can be 'bonferroni', 'hochberg', 'holm', 'hommel', 'BH', 'BY'</p>
</td></tr>
<tr><td><code id="metevalue.methylKit_+3A_sep">sep</code></td>
<td>
<p>seperator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.methylKit_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the input_filename_b file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe, the columns are (in order):
</p>
<p>- chr: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2
</p>
<p>- e_value: The e-value of the corresponding region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(demo_methylkit_methyrate)
data(demo_methylkit_met_all)
## example_tempfiles = tempfile(c("rate_combine", "methylKit_DMR_raw"))
## tempdir()
## write.table(demo_methylkit_methyrate, file=example_tempfiles[1],
##       row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
## write.table(demo_methylkit_met_all, file=example_tempfiles[2],
##       sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
## result = metevalue.methylKit(example_tempfiles[1], example_tempfiles[2],
##       bheader = TRUE)
## str(result)
</code></pre>

<hr>
<h2 id='metevalue.methylKit.chk'>Check the methylKit data format</h2><span id='topic+metevalue.methylKit.chk'></span>

<h3>Description</h3>

<p>Check the methylKit data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.methylKit.chk(
  input_filename_a,
  input_filename_b,
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.methylKit.chk_+3A_input_filename_a">input_filename_a</code></td>
<td>
<p>the combined data of methylation rate file. This file is a sep (e.g. TAB) separated file with two key columns and several value columns:
For exampe:
</p>

<table>
<tr>
 <td style="text-align: right;">
chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>- chr and pos are keys;
</p>
<p>- g1~g2: methylation rate data in groups.</p>
</td></tr>
<tr><td><code id="metevalue.methylKit.chk_+3A_input_filename_b">input_filename_b</code></td>
<td>
<p>the output file of methylKit. a methylDiff or methylDiffDB object containing the differential methylated locations satisfying the criteria.
The columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The position of the start sites of the corresponding region
</p>
<p>- end: The position of the end sites of the corresponding region
</p>
<p>- strand: Strand
</p>
<p>- p: p-value
</p>
<p>- qvalue:  The adjusted p-value based on BH method
</p>
<p>- meth.diff : The difference between the group means of methylation level</p>
</td></tr>
<tr><td><code id="metevalue.methylKit.chk_+3A_sep">sep</code></td>
<td>
<p>separator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.methylKit.chk_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the input_filename_b file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(file_a, file_b, file_a_b) returns a list with three pr-handled data.frames corresponding to the input_filename_a, input_filename_b file and a A JOIN B file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(demo_methylkit_methyrate)
data(demo_methylkit_met_all)
## example_tempfiles = tempfile(c("rate_combine", "methylKit_DMR_raw"))
## tempdir()
## write.table(demo_methylkit_methyrate, file=example_tempfiles[1],
##       row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
## write.table(demo_methylkit_met_all, file=example_tempfiles[2],
##       sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
## result = metevalue.methylKit.chk(example_tempfiles[1], example_tempfiles[2],
##       bheader = TRUE)
</code></pre>

<hr>
<h2 id='metevalue.metilene'>Calculate E-value of the Metilene data format</h2><span id='topic+metevalue.metilene'></span>

<h3>Description</h3>

<p>Calculate E-value of the Metilene data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.metilene(
  methyrate,
  metilene.output,
  adjust.methods = "BH",
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.metilene_+3A_methyrate">methyrate</code></td>
<td>
<p>metilene input file path. This file is a sep (e.g. TAB) separated file with two key columns and several value columns.
For exampe:
</p>

<table>
<tr>
 <td style="text-align: right;">
 chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>The columns are (in order):
</p>
<p>- chr and pos are keys;
</p>
<p>- g1~g2: methylation rate data in groups.</p>
</td></tr>
<tr><td><code id="metevalue.metilene_+3A_metilene.output">metilene.output</code></td>
<td>
<p>metilene input file path. This file should stored as a sep(e.g. TAB) separated file with two key columns and several value columns:
The columns are (in order):
</p>
<p>- chr: Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2</p>
</td></tr>
<tr><td><code id="metevalue.metilene_+3A_adjust.methods">adjust.methods</code></td>
<td>
<p>is the adjust methods of e-value. It can be 'bonferroni', 'hochberg', 'holm', 'hommel', 'BH', 'BY'</p>
</td></tr>
<tr><td><code id="metevalue.metilene_+3A_sep">sep</code></td>
<td>
<p>seperator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.metilene_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the metilene.output file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe, the columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2
</p>
<p>- e_value: The e-value of the corresponding region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#### metilene example ####'
data(demo_metilene_input)
data(demo_metilene_out)
#example_tempfiles = tempfile(c("metilene_input", "metilene_out"))
#tempdir()
#write.table(demo_metilene_input, file=example_tempfiles[1],
#      row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
#write.table(demo_metilene_out, file=example_tempfiles[2],
#      sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#result = metevalue.metilene(example_tempfiles[1], example_tempfiles[2],
#      bheader = TRUE)
#head(result)
</code></pre>

<hr>
<h2 id='metevalue.metilene.chk'>Check the Metilene data format</h2><span id='topic+metevalue.metilene.chk'></span>

<h3>Description</h3>

<p>Check the Metilene data format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.metilene.chk(
  input_filename_a,
  input_filename_b,
  sep = "\t",
  bheader = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.metilene.chk_+3A_input_filename_a">input_filename_a</code></td>
<td>
<p>metilene input file path. This file is a sep (e.g. TAB) separated file with two key columns and several value columns.
For exampe:
</p>

<table>
<tr>
 <td style="text-align: right;">
 chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>The columns are (in order):
</p>
<p>- chr and pos are keys;
</p>
<p>- g1~g2: methylation rate data in groups.</p>
</td></tr>
<tr><td><code id="metevalue.metilene.chk_+3A_input_filename_b">input_filename_b</code></td>
<td>
<p>metilene input file path. This file should stored as a sep(e.g. TAB) separated file with two key columns and several value columns:
The columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The position of the start sites of the corresponding region
</p>
<p>- end: The position of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2</p>
</td></tr>
<tr><td><code id="metevalue.metilene.chk_+3A_sep">sep</code></td>
<td>
<p>separator, default is the TAB key.</p>
</td></tr>
<tr><td><code id="metevalue.metilene.chk_+3A_bheader">bheader</code></td>
<td>
<p>a logical value indicating whether the input_filename_b file contains the names of the variables as its first line. By default, bheader = FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list(file_a, file_b, file_a_b) returns a list with three pr-handled data.frames corresponding to the input_filename_a, input_filename_b file and a A JOIN B file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(demo_metilene_input)
#data(demo_metilene_out)
#example_tempfiles = tempfile(c("metilene_input", "metilene_out"))
#tempdir()
#write.table(demo_metilene_input, file=example_tempfiles[1],
#      row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
#write.table(demo_metilene_out, file=example_tempfiles[2],
#      sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#result = metevalue.metilene.chk(example_tempfiles[1], example_tempfiles[2],
#      bheader = TRUE)
</code></pre>

<hr>
<h2 id='metevalue.RNA_general'>A general method to calculate the e-value for RNA-seq data.</h2><span id='topic+metevalue.RNA_general'></span>

<h3>Description</h3>

<p>A general method to calculate the e-value for RNA-seq data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>metevalue.RNA_general(rna, group1_name, group2_name)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="metevalue.RNA_general_+3A_rna">rna</code></td>
<td>
<p>data.frame: A data.frame object of RNAseq data. For example:
</p>

<table>
<tr>
 <td style="text-align: right;">
TAG   </td><td style="text-align: right;">  treated1fb </td><td style="text-align: right;"> treated2fb  </td><td style="text-align: right;"> untreated1fb </td><td style="text-align: right;"> untreated2fb </td>
</tr>
<tr>
 <td style="text-align: right;">
TAG1  </td><td style="text-align: right;"> 4.449648    </td><td style="text-align: right;"> 4.750104    </td><td style="text-align: right;">   4.392285   </td><td style="text-align: right;">   4.497514 </td>
</tr>
<tr>
 <td style="text-align: right;">
TAG2  </td><td style="text-align: right;"> 8.241116    </td><td style="text-align: right;"> 8.302852    </td><td style="text-align: right;">  8.318125    </td><td style="text-align: right;">  8.488796 </td>
</tr>
<tr>
 <td style="text-align: right;">
...   </td><td style="text-align: right;"> ...         </td><td style="text-align: right;"> ...         </td><td style="text-align: right;">  ...         </td><td style="text-align: right;">  ...      </td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>Row names (TAG1 and TAG2 in the above example) is also suggested.</p>
</td></tr>
<tr><td><code id="metevalue.RNA_general_+3A_group1_name">group1_name</code></td>
<td>
<p>charactor: The name (pattern) of the first group. For example, &quot;treated&quot; in the above example.
For example 'treated_abc' and 'treated' will be considered as the same group if 'group1_name = &quot;treated&quot;'.  Use this with care in practice.</p>
</td></tr>
<tr><td><code id="metevalue.RNA_general_+3A_group2_name">group2_name</code></td>
<td>
<p>charactor: The name (pattern) of the second group. For example, &quot;untreated&quot; in the above example.
For example 'untreated_abc' and 'untreated' will be considered as the same group if 'group2_name = &quot;untreated&quot;'.  Use this with care in practice.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>evalue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("demo_desq_out")

evalue = metevalue.RNA_general(demo_desq_out, 'treated','untreated')

</code></pre>

<hr>
<h2 id='varevalue.metilene'>Calculate E-value of the Metilene data</h2><span id='topic+varevalue.metilene'></span>

<h3>Description</h3>

<p>The data file could be pre-handled by the evalue.metilene.chk function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varevalue.metilene(
  a,
  b,
  a_b,
  group1_name = "g1",
  group2_name = "g2",
  adjust.methods = "BH"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="varevalue.metilene_+3A_a">a</code></td>
<td>
<p>A data.frame object:
</p>

<table>
<tr>
 <td style="text-align: right;">
 chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>

<p>i.e two key columns (chrom, pos) with several value columns in groups.</p>
</td></tr>
<tr><td><code id="varevalue.metilene_+3A_b">b</code></td>
<td>
<p>A data.frame object stores the data, the columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2</p>
</td></tr>
<tr><td><code id="varevalue.metilene_+3A_a_b">a_b</code></td>
<td>
<p>A data.frame object of a join b with particular data clean processes. Check the function [evalue.methylKit.chk()] for more details.</p>
</td></tr>
<tr><td><code id="varevalue.metilene_+3A_group1_name">group1_name</code></td>
<td>
<p>charactor: The name of the first group. For example, &quot;g1&quot; in the above example.</p>
</td></tr>
<tr><td><code id="varevalue.metilene_+3A_group2_name">group2_name</code></td>
<td>
<p>charactor: The name of the second group. For example, &quot;g2&quot; in the above example.</p>
</td></tr>
<tr><td><code id="varevalue.metilene_+3A_adjust.methods">adjust.methods</code></td>
<td>
<p>is the adjust methods of e-value. It can be 'bonferroni', 'hochberg', 'holm', 'hommel', 'BH', 'BY'. The default value is 'BH'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe, the columns are (in order):
</p>
<p>- chr:   Chromosome
</p>
<p>- start: The positions of the start sites of the corresponding region
</p>
<p>- end: The positions of the end sites of the corresponding region
</p>
<p>- q-value: The adjusted p-value based on BH method in MWU-test
</p>
<p>- methyl.diff: The difference between the group means of methylation level
</p>
<p>- CpGs:  The number of CpG sites within the corresponding region
</p>
<p>- p : p-value based on MWU-test
</p>
<p>- p2: p-value based on 2D KS-test
</p>
<p>- m1:  The absolute mean methylation level for the corresponding segment of group 1
</p>
<p>- m2:  The absolute mean methylation level for the corresponding segment of group 2
</p>
<p>- e_value: The e-value of the corresponding region
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data(demo_metilene_input)
#data(demo_metilene_out)
#result = evalue_buildin_var_fmt_nm(demo_metilene_input, demo_metilene_out, method="metilene")
#result = list(a = result$a, 
#              b = result$b, 
#              a_b = evalue_buildin_sql(result$a, result$b, method="metilene"))
#result = varevalue.metilene(result$a, result$b, result$a_b)
</code></pre>

<hr>
<h2 id='varevalue.single_general'>A general method to calculate the e-value for other DNA methylation tools not described above. The input data is the DNA methylation rates using the similar format with Metilene.</h2><span id='topic+varevalue.single_general'></span>

<h3>Description</h3>

<p>The input data file is just the DNA methylation rates using the similar format above, with no need for another data file output by different tools.
The Chromosome name, start and end sites shoule be specified in the function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>varevalue.single_general(
  methyrate,
  group1_name = "g1",
  group2_name = "g2",
  chr,
  start,
  end
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="varevalue.single_general_+3A_methyrate">methyrate</code></td>
<td>
<p>data.frame: A data.frame object of methylation rates, the columns should be (name of groups can be self-defined)
</p>

<table>
<tr>
 <td style="text-align: right;">
 chr    </td><td style="text-align: right;">  pos     </td><td style="text-align: right;">   g1    </td><td style="text-align: right;"> ...  </td><td style="text-align: right;">  g1 </td><td style="text-align: right;">  g2 </td><td style="text-align: right;"> ... </td><td style="text-align: right;"> g2 </td>
</tr>
<tr>
 <td style="text-align: right;">
chr1 </td><td style="text-align: right;">  1    </td><td style="text-align: right;">  0.1 </td><td style="text-align: right;">  ... </td><td style="text-align: right;">   0.1</td><td style="text-align: right;">  0.2</td><td style="text-align: right;"> ... </td><td style="text-align: right;"> 0.2</td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>
</td></tr>
<tr><td><code id="varevalue.single_general_+3A_group1_name">group1_name</code></td>
<td>
<p>charactor: The name (pattern) of the first group. For example, &quot;g1&quot; in the above example.
For example 'g1_abc' and 'g1' will be considered as the same group if 'group1_name = &quot;g1&quot;'.  Use this with care in practice.</p>
</td></tr>
<tr><td><code id="varevalue.single_general_+3A_group2_name">group2_name</code></td>
<td>
<p>charactor: The name (pattern) of the second group. For example, &quot;g2&quot; in the above example.
For example 'g2_abc' and 'g2' will be considered as the same group if 'group2_name = &quot;g2&quot;'.  Use this with care in practice.</p>
</td></tr>
<tr><td><code id="varevalue.single_general_+3A_chr">chr</code></td>
<td>
<p>charactor: The Chromosome name. Typically, it is a string like &quot;chr21&quot; and so on.</p>
</td></tr>
<tr><td><code id="varevalue.single_general_+3A_start">start</code></td>
<td>
<p>integer:  The position of the start site of the corresponding region</p>
</td></tr>
<tr><td><code id="varevalue.single_general_+3A_end">end</code></td>
<td>
<p>integer: The position of the end site of the corresponding region</p>
</td></tr>
</table>


<h3>Value</h3>

<p>evalue
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#data("demo_metilene_input")
#varevalue.single_general(demo_metilene_input, chr = "chr21", start = 9437432, end = 9437540)
# [1] 2.626126e+43

#### Compare to `metevalue.metilene`  ####
data(demo_metilene_out)
#example_tempfiles = tempfile(c("metilene_input", "metilene_out"))
#tempdir()
#write.table(demo_metilene_input, file=example_tempfiles[1],
#      row.names=FALSE, col.names=TRUE, quote=FALSE, sep='\t')
#write.table (demo_metilene_out, file=example_tempfiles[2],
#      sep ="\t", row.names =FALSE, col.names =TRUE, quote =FALSE)
#result = metevalue.metilene(example_tempfiles[1], example_tempfiles[2],
#      bheader = TRUE)
# result[with(result, chr == 'chr21' &amp; start == '9437432' &amp; end == '9437540'), ncol(result)]
# [1] 2.626126e+43
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
