<!DOCTYPE html><html><head><title>Help for package methcon5</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {methcon5}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#%&gt;%'><p>Pipe operator</p></a></li>
<li><a href='#fake_methylation'><p>Simple simulated methylation dataset</p></a></li>
<li><a href='#meth_aggregate'><p>Calculate region wise summary statistics</p></a></li>
<li><a href='#meth_bootstrap'><p>Bootstrapped randomly samples values</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Identify and Rank CpG DNA Methylation Conservation Along the
Human Genome</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Identify and rank CpG DNA methylation conservation along the human 
    genome. Specifically it includes bootstrapping methods to provide ranking 
    which should adjust for the differences in length as without it short 
    regions tend to get higher conservation scores.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr, dplyr, purrr, rlang</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ggplot2, testthat (&ge; 2.1.0), covr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/EmilHvitfeldt/methcon5">https://github.com/EmilHvitfeldt/methcon5</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/EmilHvitfeldt/methcon5/issues">https://github.com/EmilHvitfeldt/methcon5/issues</a></td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-12-17 02:47:31 UTC; emilhvitfeldthansen</td>
</tr>
<tr>
<td>Author:</td>
<td>Emil Hvitfeldt <a href="https://orcid.org/0000-0002-0679-1945"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Emil Hvitfeldt &lt;emilhhvitfeldt@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-12-20 13:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='+25+26gt+3B+25'>Pipe operator</h2><span id='topic++25+3E+25'></span>

<h3>Description</h3>

<p>See <code>magrittr::<a href="magrittr.html#topic+pipe">%&gt;%</a></code> for details.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lhs %&gt;% rhs
</code></pre>

<hr>
<h2 id='fake_methylation'>Simple simulated methylation dataset</h2><span id='topic+fake_methylation'></span>

<h3>Description</h3>

<p>Simple simulated methylation dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fake_methylation
</code></pre>


<h3>Format</h3>

<p>A data frame with 2771 rows and 3 variables: <code>gene</code>,
<code>cons_level</code> and <code>meth</code>.</p>


<h3>Details</h3>

<p>This dataset is for example use only. It contains 500 genes
identified by <code>gene</code> each with one of 3 types of conservation levels
&quot;low&quot;, &quot;medium&quot; and &quot;high&quot;. The methylation values are independent randomly
distributed within each gene. Thus no spacial correlation is assumed.
</p>

<hr>
<h2 id='meth_aggregate'>Calculate region wise summary statistics</h2><span id='topic+meth_aggregate'></span>

<h3>Description</h3>

<p>Will take a data.frame and apply a function ('fun') to 'value' within the
groups defined by the 'id' column.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meth_aggregate(data, id, value, fun = mean, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meth_aggregate_+3A_data">data</code></td>
<td>
<p>a data.frame.</p>
</td></tr>
<tr><td><code id="meth_aggregate_+3A_id">id</code></td>
<td>
<p>variable name, to be aggregated around.</p>
</td></tr>
<tr><td><code id="meth_aggregate_+3A_value">value</code></td>
<td>
<p>variable name, contains the value to take mean over. Must be
a single column.</p>
</td></tr>
<tr><td><code id="meth_aggregate_+3A_fun">fun</code></td>
<td>
<p>function, summary statistic function to be calculated. Defaults
to 'mean'.</p>
</td></tr>
<tr><td><code id="meth_aggregate_+3A_...">...</code></td>
<td>
<p>Additional arguments for the function given to the argument fun.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please note the ordering of the data will matter depending on the choice of
aggregation function.
</p>


<h3>Value</h3>

<p>A methcon object. Contains the aggregated data along with original
data.frame and variable selections.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>meth_aggregate(fake_methylation, id = gene, value = meth, fun = mean)

meth_aggregate(fake_methylation, id = gene, value = meth, fun = var)

# custom functions can be used as well
mean_diff &lt;- function(x) {
  mean(diff(x))
}

meth_aggregate(fake_methylation, id = gene, value = meth, fun = mean_diff)
</code></pre>

<hr>
<h2 id='meth_bootstrap'>Bootstrapped randomly samples values</h2><span id='topic+meth_bootstrap'></span>

<h3>Description</h3>

<p>&quot;perm_v1&quot; (the default method) will sample the variables the rows
independently. &quot;perm_v2&quot; will sample regions of same size while allowing
overlap between different regions. &quot;perm_v3&quot; will sample regions under the
constraint that all sampled regions are contained in the region they are
sampled in.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>meth_bootstrap(data, reps, method = c("perm_v1", "perm_v2", "perm_v3"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="meth_bootstrap_+3A_data">data</code></td>
<td>
<p>a methcon data.frame output from 'meth_bootstrap'.</p>
</td></tr>
<tr><td><code id="meth_bootstrap_+3A_reps">reps</code></td>
<td>
<p>Number of reps, defaults to 1000.</p>
</td></tr>
<tr><td><code id="meth_bootstrap_+3A_method">method</code></td>
<td>
<p>Character, determining which method to use. See details for
information about methods. Defaults to &quot;perm_v1&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note that you can apply 'meth_bootstrap' multiple times to get values for
different methods.
</p>


<h3>Value</h3>

<p>A methcon object. Contains the aggregated data along with original
data.frame and variable selections and bootstrapped values.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Note that you likely want to do more than 10 repitions.
# rep = 10 was chosen to have the examples run fast.

fake_methylation %&gt;%
  meth_aggregate(id = gene, value = meth, fun = mean) %&gt;%
  meth_bootstrap(10)

fake_methylation %&gt;%
  meth_aggregate(id = gene, value = meth, fun = mean) %&gt;%
  meth_bootstrap(10, method = "perm_v2")

# Get multiple bootstraps
fake_methylation %&gt;%
  meth_aggregate(id = gene, value = meth, fun = mean) %&gt;%
  meth_bootstrap(10, method = "perm_v1") %&gt;%
  meth_bootstrap(10, method = "perm_v2") %&gt;%
  meth_bootstrap(10, method = "perm_v3")
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
