<!DOCTYPE html><html lang="en"><head><title>Help for package mExplorer</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mExplorer}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mExplorer'><p>Selection of process-specific regulators from high-throughput data using multinomial regression models.</p></a></li>
<li><a href='#prepare_gmt_input'><p>Creation of m:Explorer input data frame from GMT files</p></a></li>
<li><a href='#small_test_dframe'><p>Small sample of predictor data for testing m:Explorer</p></a></li>
<li><a href='#small_test_response_vec'><p>Small vector of yeast transcription factors for testing m:Explorer</p></a></li>
<li><a href='#yeastCCgenes'><p>Example vector of yeast transcription factors for m:Explorer</p></a></li>
<li><a href='#yeastTFdata'><p>Example predictor data for m:Explorer</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Description:</td>
<td>The method 'm:Explorer' associates a given list of target genes (e.g. those involved in a biological process) to gene regulators such as transcription factors. Transcription factors that bind DNA near significantly many target genes or correlate with target genes in transcriptional (microarray or RNAseq data) are selected. Selection of candidate master regulators is carried out using multinomial regression models, likelihood ratio tests and multiple testing correction. Reference: m:Explorer: multinomial regression models reveal positive and negative regulators of longevity in yeast quiescence. Juri Reimand, Anu Aun, Jaak Vilo, Juan M Vaquerizas, Juhan Sedman and Nicholas M Luscombe. Genome Biology (2012) 13:R55 &lt;<a href="https://doi.org/10.1186%2Fgb-2012-13-6-r55">doi:10.1186/gb-2012-13-6-r55</a>&gt;.</td>
</tr>
<tr>
<td>Title:</td>
<td>Identifying Master Gene Regulators from Gene Expression and
DNA-Binding Data</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, nnet, parallel, qusage</td>
</tr>
<tr>
<td>Collate:</td>
<td>'mExplorer.R'</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1.9000</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-08-24 13:27:51 UTC; dthompson</td>
</tr>
<tr>
<td>Author:</td>
<td>Juri Reimand [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Juri Reimand &lt;juri.reimand@utoronto.ca&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-08-24 14:51:44 UTC</td>
</tr>
</table>
<hr>
<h2 id='mExplorer'>Selection of process-specific regulators from high-throughput data using multinomial regression models.</h2><span id='topic+mExplorer'></span>

<h3>Description</h3>

<p>Selection of process-specific regulators from high-throughput data using multinomial regression models.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mExplorer(dframe, response, interactions = F, significance = 0.05,
  n_cores = 1, multitest = "BY")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mExplorer_+3A_dframe">dframe</code></td>
<td>
<p>Data frame of predictors. Row and column names are required for identifying samples (genes) and predictors (gene regulators), respectively.</p>
</td></tr>
<tr><td><code id="mExplorer_+3A_response">response</code></td>
<td>
<p>Vector of factors. Names of vector need to correspond to rownames in dframe.</p>
</td></tr>
<tr><td><code id="mExplorer_+3A_interactions">interactions</code></td>
<td>
<p>If enabled, pairs of predictors as interactions will be evaluated (much slower).</p>
</td></tr>
<tr><td><code id="mExplorer_+3A_significance">significance</code></td>
<td>
<p>Significance cutoff for p-values from log likelihood ratio tests.</p>
</td></tr>
<tr><td><code id="mExplorer_+3A_n_cores">n_cores</code></td>
<td>
<p>Number of processor cores to engage in computation. Use all available cores by default (n_cores=0).</p>
</td></tr>
<tr><td><code id="mExplorer_+3A_multitest">multitest</code></td>
<td>
<p>Method to perform multiple testing correction for p-values from predictor evaluation. See p.adjust() for details.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Vector of scores, with names corresponding to predictors.
</p>


<h3>Author(s)</h3>

<p>Juri Reimand &lt;juri.reimand@utoronto.ca&gt;
</p>


<h3>References</h3>

<p>m:Explorer - multinomial regression models reveal positive and negative regulators of longevity in yeast quiescence (2012, Genome Biology) by Juri Reimand, Anu Aun, Jaak Vilo, Juan M. Vaquerizas, Juhan Sedman, and Nicholas M. Luscombe
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(yeastCCgenes)
data(yeastTFdata)
mExplorer(yeastTFdata, yeastCCgenes)

data(mExplorer_small_test_data)
small_test_results = mExplorer(small_test_dframe, small_test_response_vec)
</code></pre>

<hr>
<h2 id='prepare_gmt_input'>Creation of m:Explorer input data frame from GMT files</h2><span id='topic+prepare_gmt_input'></span>

<h3>Description</h3>

<p>Creation of m:Explorer input data frame from GMT files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepare_gmt_input(gmt_filename, min_genes = NA, max_genes = NA)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepare_gmt_input_+3A_gmt_filename">gmt_filename</code></td>
<td>
<p>Path to GMT file to convert.</p>
</td></tr>
<tr><td><code id="prepare_gmt_input_+3A_min_genes">min_genes</code></td>
<td>
<p>Numeric indicating to discard pathways with less than min_genes genes. If NA, there is no lower bound on the number of genes. Default is NA.</p>
</td></tr>
<tr><td><code id="prepare_gmt_input_+3A_max_genes">max_genes</code></td>
<td>
<p>Numeric indicating to discard pathways with more than max_genes genes. If NA, there is no upper bound on the number of genes. Default is NA.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with pathways as columns, genes as rows. Gene/pathway combinations are marked with &quot;pw&quot; if that gene is in the pathway, or &quot;.&quot; if not.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Create m:Explorer input data frame from GMT "small_gmt.gmt," discarding
# pathways with less than 5 genes and more than 1000 genes
gmt_file = system.file("extdata", "small_gmt.gmt", package = "mExplorer")
gmt = prepare_gmt_input(gmt_file, 5, 1000)
</code></pre>

<hr>
<h2 id='small_test_dframe'>Small sample of predictor data for testing m:Explorer</h2><span id='topic+small_test_dframe'></span>

<h3>Description</h3>

<p>Small sample of predictor data for testing m:Explorer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mExplorer_small_test_data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 10 observations of 18 variables</p>

<hr>
<h2 id='small_test_response_vec'>Small vector of yeast transcription factors for testing m:Explorer</h2><span id='topic+small_test_response_vec'></span>

<h3>Description</h3>

<p>Small vector of yeast transcription factors for testing m:Explorer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mExplorer_small_test_data)
</code></pre>


<h3>Format</h3>

<p>A named character vector with 4 elements</p>

<hr>
<h2 id='yeastCCgenes'>Example vector of yeast transcription factors for m:Explorer</h2><span id='topic+yeastCCgenes'></span>

<h3>Description</h3>

<p>Example vector of yeast transcription factors for m:Explorer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yeastCCgenes)
</code></pre>


<h3>Format</h3>

<p>A named character vector with 186 elements</p>

<hr>
<h2 id='yeastTFdata'>Example predictor data for m:Explorer</h2><span id='topic+yeastTFdata'></span>

<h3>Description</h3>

<p>Example predictor data for m:Explorer
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(yeastTFdata)
</code></pre>


<h3>Format</h3>

<p>A data frame with 6253 observations of 18 variables</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
