<!DOCTYPE html><html><head><title>Help for package mgi.report.reader</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mgi.report.reader}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mgi.report.reader-package'><p>mgi.report.reader: Read Mouse Genome Informatics Reports</p></a></li>
<li><a href='#chromosomes'><p>Mouse chromosomes</p></a></li>
<li><a href='#common-args'><p>Common arguments</p></a></li>
<li><a href='#feature_type_definitions'><p>Genome Feature Type Definitions</p></a></li>
<li><a href='#feature_types'><p>Genome Feature types</p></a></li>
<li><a href='#is_mgi_identifier'><p>MGI accession identifier checking</p></a></li>
<li><a href='#marker_type_definitions'><p>Genetic Marker Type Definitions</p></a></li>
<li><a href='#marker_types'><p>Genetic marker types</p></a></li>
<li><a href='#mgi_base_url'><p>MGI base URL</p></a></li>
<li><a href='#mgi_reports_base_url'><p>MGI reports base URL</p></a></li>
<li><a href='#mgi_reports_index_url'><p>MGI reports index URL</p></a></li>
<li><a href='#open_marker_id_in_mgi'><p>Browse MGI markers identifiers online</p></a></li>
<li><a href='#open_marker_symbol_in_mgi'><p>Browse MGI markers symbols online</p></a></li>
<li><a href='#read_report'><p>Read an MGI report</p></a></li>
<li><a href='#read_symbol_map'><p>Read marker symbol mappings</p></a></li>
<li><a href='#read_symbol_to_id_map'><p>Read marker symbol to identifier mappings</p></a></li>
<li><a href='#report_example'><p>Report example</p></a></li>
<li><a href='#report_last_modified'><p>Report last modification date</p></a></li>
<li><a href='#report_source'><p>Report source</p></a></li>
<li><a href='#report-attributes'><p>Get MGI report specs by report key</p></a></li>
<li><a href='#reports'><p>Supported MGI reports</p></a></li>
<li><a href='#sure'><p>Are you sure?</p></a></li>
<li><a href='#symbol_to_identifier'><p>Convert marker symbols to updated marker identifiers</p></a></li>
<li><a href='#symbol_to_symbol'><p>Update marker symbols</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Read Mouse Genome Informatics Reports</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.3</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides readers for easy and consistent importing of
    Mouse Genome Informatics (MGI) report files:
    <a href="https://www.informatics.jax.org/downloads/reports/index.html">https://www.informatics.jax.org/downloads/reports/index.html</a>. These data
    are provided by Baldarelli RM, Smith CL, Ringwald M, Richardson JE, Bult CJ,
    Mouse Genome Informatics Group (2024) &lt;<a href="https://doi.org/10.1093%2Fgenetics%2Fiyae031">doi:10.1093/genetics/iyae031</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.10)</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>data.table, dplyr, httr2, memoise, rlang, stringr, tibble,
vroom</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://www.pattern.institute/mgi.report.reader/">https://www.pattern.institute/mgi.report.reader/</a>,
<a href="https://github.com/patterninstitute/mgi.report.reader/">https://github.com/patterninstitute/mgi.report.reader/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/patterninstitute/mgi.report.reader/issues">https://github.com/patterninstitute/mgi.report.reader/issues</a></td>
</tr>
<tr>
<td>Config/Needs/website:</td>
<td>rmarkdown, patterninstitute/chic</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat (&ge; 3.0.0), tidyr</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-07-07 01:17:19 UTC; rmagno</td>
</tr>
<tr>
<td>Author:</td>
<td>Ramiro Magno <a href="https://orcid.org/0000-0001-5226-3441"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  David Shaw <a href="https://orcid.org/0000-0002-9922-9934"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Isabel Duarte <a href="https://orcid.org/0000-0003-0060-2936"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ismail Gbadamosi <a href="https://orcid.org/0000-0002-1932-7519"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut],
  Ali Jawaid <a href="https://orcid.org/0000-0002-5126-6744"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut],
  Nencki Institute of Experimental Biology [fnd],
  University of Algarve [fnd],
  The Jackson Laboratory [fnd],
  Pattern Institute [cph, fnd]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Ramiro Magno &lt;rmagno@pattern.institute&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-07-07 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='mgi.report.reader-package'>mgi.report.reader: Read Mouse Genome Informatics Reports</h2><span id='topic+mgi.report.reader'></span><span id='topic+mgi.report.reader-package'></span>

<h3>Description</h3>

<p>Provides readers for easy and consistent importing of Mouse Genome Informatics (MGI) report files: <a href="https://www.informatics.jax.org/downloads/reports/index.html">https://www.informatics.jax.org/downloads/reports/index.html</a>. These data are provided by Baldarelli RM, Smith CL, Ringwald M, Richardson JE, Bult CJ, Mouse Genome Informatics Group (2024) <a href="https://doi.org/10.1093/genetics/iyae031">doi:10.1093/genetics/iyae031</a>.
</p>


<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Ramiro Magno <a href="mailto:rmagno@pattern.institute">rmagno@pattern.institute</a> (<a href="https://orcid.org/0000-0001-5226-3441">ORCID</a>)
</p>
<p>Authors:
</p>

<ul>
<li><p> David Shaw <a href="mailto:David.Shaw@jax.org">David.Shaw@jax.org</a> (<a href="https://orcid.org/0000-0002-9922-9934">ORCID</a>)
</p>
</li>
<li><p> Isabel Duarte <a href="mailto:iduarte@pattern.institute">iduarte@pattern.institute</a> (<a href="https://orcid.org/0000-0003-0060-2936">ORCID</a>)
</p>
</li>
<li><p> Ismail Gbadamosi <a href="mailto:i.gbadamosi@nencki.edu.pl">i.gbadamosi@nencki.edu.pl</a> (<a href="https://orcid.org/0000-0002-1932-7519">ORCID</a>)
</p>
</li>
<li><p> Ali Jawaid <a href="mailto:a.jawaid@nencki.edu.pl">a.jawaid@nencki.edu.pl</a> (<a href="https://orcid.org/0000-0002-5126-6744">ORCID</a>)
</p>
</li></ul>

<p>Other contributors:
</p>

<ul>
<li><p> Nencki Institute of Experimental Biology [funder]
</p>
</li>
<li><p> University of Algarve [funder]
</p>
</li>
<li><p> The Jackson Laboratory [funder]
</p>
</li>
<li><p> Pattern Institute [copyright holder, funder]
</p>
</li></ul>



<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://www.pattern.institute/mgi.report.reader/">https://www.pattern.institute/mgi.report.reader/</a>
</p>
</li>
<li> <p><a href="https://github.com/patterninstitute/mgi.report.reader/">https://github.com/patterninstitute/mgi.report.reader/</a>
</p>
</li>
<li><p> Report bugs at <a href="https://github.com/patterninstitute/mgi.report.reader/issues">https://github.com/patterninstitute/mgi.report.reader/issues</a>
</p>
</li></ul>


<hr>
<h2 id='chromosomes'>Mouse chromosomes</h2><span id='topic+chromosomes'></span>

<h3>Description</h3>

<p><code><a href="#topic+chromosomes">chromosomes()</a></code> returns mouse chromosome names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chromosomes(autosomal = TRUE, sexual = TRUE, mitochondrial = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chromosomes_+3A_autosomal">autosomal</code></td>
<td>
<p>Whether to include the autosomal chromosomes (1 thru 19).</p>
</td></tr>
<tr><td><code id="chromosomes_+3A_sexual">sexual</code></td>
<td>
<p>Whether to include the sexual chromosomes (X and Y).</p>
</td></tr>
<tr><td><code id="chromosomes_+3A_mitochondrial">mitochondrial</code></td>
<td>
<p>Whether to include the mitochondrial chromosome (MT).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of mouse chromosome names, or a subset thereof,
or an empty character vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># All chromosomes.
chromosomes()

# Autosomal chromosomes.
chromosomes(autosomal = TRUE, sexual = FALSE, mitochondrial = FALSE)

</code></pre>

<hr>
<h2 id='common-args'>Common arguments</h2><span id='topic+common-args'></span>

<h3>Description</h3>

<p>A set of arguments that are commonly reused across functions of
<code>{mgi.report.reader}</code>.
</p>


<h3>Arguments</h3>

<table>
<tr><td><code id="common-args_+3A_marker_id">marker_id</code></td>
<td>
<p>A character vector. MGI accession identifiers.</p>
</td></tr>
<tr><td><code id="common-args_+3A_marker_symbol">marker_symbol</code></td>
<td>
<p>A character vector. MGI marker symbols.</p>
</td></tr>
<tr><td><code id="common-args_+3A_report_key">report_key</code></td>
<td>
<p>A character vector. A key used to uniquely refer to an MGI
report.</p>
</td></tr>
<tr><td><code id="common-args_+3A_report_file">report_file</code></td>
<td>
<p>A character vector. The file path or URL to an MGI report
file.</p>
</td></tr>
<tr><td><code id="common-args_+3A_report_type">report_type</code></td>
<td>
<p>A character vector. The type of an MGI report.</p>
</td></tr>
</table>

<hr>
<h2 id='feature_type_definitions'>Genome Feature Type Definitions</h2><span id='topic+feature_type_definitions'></span>

<h3>Description</h3>

<p>A dataset containing different types of gene and genome features along with
their Sequence Ontology (SO) identifiers and definitions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature_type_definitions
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> with 71 rows
and 3 variables:
</p>

<dl>
<dt>feature_type</dt><dd><p>Character. The type of gene or genome feature.</p>
</dd>
<dt>so_id</dt><dd><p>Character. The Sequence Ontology identifier associated with the feature type.</p>
</dd>
<dt>definition</dt><dd><p>Character. The definition of the feature type.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The table in <a href="https://www.informatics.jax.org/userhelp/GENE_feature_types_help.shtml">https://www.informatics.jax.org/userhelp/GENE_feature_types_help.shtml</a>
and a few other terms found in MGI reports.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(feature_type_definitions, n = Inf)

</code></pre>

<hr>
<h2 id='feature_types'>Genome Feature types</h2><span id='topic+feature_types'></span>

<h3>Description</h3>

<p><code><a href="#topic+feature_types">feature_types()</a></code> returns different types of gene and genome features. For
feature type definitions, see <code>?feature_type_definitions</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>feature_types()
</code></pre>


<h3>Value</h3>

<p>A character vector of feature types' names.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>feature_types()

</code></pre>

<hr>
<h2 id='is_mgi_identifier'>MGI accession identifier checking</h2><span id='topic+is_mgi_identifier'></span>

<h3>Description</h3>

<p><code><a href="#topic+is_mgi_identifier">is_mgi_identifier()</a></code> checks whether the input is of the format
<code>"MGI:nnnnnn"</code>, where <code>"n"</code> is a digit and the number of digits can vary.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mgi_identifier(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mgi_identifier_+3A_x">x</code></td>
<td>
<p>A character vector.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical vector.
</p>

<hr>
<h2 id='marker_type_definitions'>Genetic Marker Type Definitions</h2><span id='topic+marker_type_definitions'></span>

<h3>Description</h3>

<p>A dataset of marker types definitions.
</p>
<p>Use instead <code><a href="#topic+marker_types">marker_types()</a></code> for the marker type names as a single character
vector.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marker_type_definitions
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> with
10 rows and 2
variables:
</p>

<dl>
<dt><code>marker_type</code></dt><dd><p>Character. The type of genetic marker.</p>
</dd>
<dt><code>definition</code></dt><dd><p>Character. The definition of the marker type.</p>
</dd>
</dl>



<h3>Source</h3>

<p>The cross-references in the entry definition for marker at MGI glossary:
<a href="https://www.informatics.jax.org/glossary/marker/">https://www.informatics.jax.org/glossary/marker/</a>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>print(marker_type_definitions, n = Inf)

</code></pre>

<hr>
<h2 id='marker_types'>Genetic marker types</h2><span id='topic+marker_types'></span>

<h3>Description</h3>

<p><code><a href="#topic+marker_types">marker_types()</a></code> returns MGI marker types. See <a href="#topic+marker_type_definitions">marker_type_definitions</a> for
the meaning of each type.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>marker_types()
</code></pre>


<h3>Value</h3>

<p>A character vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>marker_types()

</code></pre>

<hr>
<h2 id='mgi_base_url'>MGI base URL</h2><span id='topic+mgi_base_url'></span>

<h3>Description</h3>

<p><code><a href="#topic+mgi_base_url">mgi_base_url()</a></code> returns MGI base URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mgi_base_url()
</code></pre>


<h3>Value</h3>

<p>A scalar character vector with the URL.
</p>

<hr>
<h2 id='mgi_reports_base_url'>MGI reports base URL</h2><span id='topic+mgi_reports_base_url'></span>

<h3>Description</h3>

<p><code><a href="#topic+mgi_reports_base_url">mgi_reports_base_url()</a></code> returns MGI reports base URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mgi_reports_base_url()
</code></pre>


<h3>Value</h3>

<p>A scalar character vector with the URL.
</p>

<hr>
<h2 id='mgi_reports_index_url'>MGI reports index URL</h2><span id='topic+mgi_reports_index_url'></span>

<h3>Description</h3>

<p><code><a href="#topic+mgi_reports_index_url">mgi_reports_index_url()</a></code> returns MGI reports index URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mgi_reports_index_url()
</code></pre>


<h3>Value</h3>

<p>A scalar character vector with the URL.
</p>

<hr>
<h2 id='open_marker_id_in_mgi'>Browse MGI markers identifiers online</h2><span id='topic+open_marker_id_in_mgi'></span>

<h3>Description</h3>

<p><code><a href="#topic+open_marker_id_in_mgi">open_marker_id_in_mgi()</a></code> launches the web browser and opens a tab for each MGI
accession identifier on the Mouse Genome Informatics web interface:
<a href="https://www.informatics.jax.org">https://www.informatics.jax.org</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_marker_id_in_mgi(marker_id)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_marker_id_in_mgi_+3A_marker_id">marker_id</code></td>
<td>
<p>A character vector. MGI accession identifiers.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful, or <code>FALSE</code> otherwise. But note that
this function is run for its side effect of launching the browser.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read about Acta1 (actin alpha 1, skeletal muscle) online.
open_marker_id_in_mgi("MGI:87902")

# `open_marker_id_in_mgi()` is vectorized, so you can open multiple pages.
# NB: think twice if you really need to open many tabs at once.
open_marker_id_in_mgi(c("MGI:87902", "MGI:87909"))

</code></pre>

<hr>
<h2 id='open_marker_symbol_in_mgi'>Browse MGI markers symbols online</h2><span id='topic+open_marker_symbol_in_mgi'></span>

<h3>Description</h3>

<p><code><a href="#topic+open_marker_symbol_in_mgi">open_marker_symbol_in_mgi()</a></code> launches the web browser and opens a tab for each MGI
symbol on the Mouse Genome Informatics web interface:
<a href="https://www.informatics.jax.org">https://www.informatics.jax.org</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>open_marker_symbol_in_mgi(marker_symbol)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="open_marker_symbol_in_mgi_+3A_marker_symbol">marker_symbol</code></td>
<td>
<p>A character vector. MGI marker symbols.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns <code>TRUE</code> if successful, or <code>FALSE</code> otherwise. But note that
this function is run for its side effect of launching the browser.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Read about Acta1 (actin alpha 1, skeletal muscle) online.
open_marker_symbol_in_mgi("Acta1")

# `open_marker_symbol_in_mgi()` is vectorized, so you can open multiple pages.
# NB: think twice if you really need to open many tabs at once.
open_marker_symbol_in_mgi(c("Acta1", "Hes1"))

</code></pre>

<hr>
<h2 id='read_report'>Read an MGI report</h2><span id='topic+read_report'></span>

<h3>Description</h3>

<p><code><a href="#topic+read_report">read_report()</a></code> imports data from an MGI report into R as a tidy data set.
</p>
<p>You may call this function in two alternative ways:
</p>

<ul>
<li><p> Using <code>report_key</code>: this is the easiest approach. A report key maps to a
report currently hosted at MGI, e.g. <code>read_report("marker_list2")</code> reads
<code>MRK_List2.rpt</code> directly from MGI server into R. See Supported Reports below
for options.
</p>
</li>
<li><p> Using <code>report_file</code> and <code>report_type</code>: this approach is more flexible as
you can read directly from a file or URL.
</p>
</li></ul>



<h4>Supported Reports</h4>

<p>The set of currently supported reports:
</p>
<div class="sourceCode r"><pre>reports
#&gt; # A tibble: 13 x 4
#&gt;    report_key                 report_file              report_type   report_name
#&gt;    &lt;chr&gt;                      &lt;chr&gt;                    &lt;chr&gt;         &lt;chr&gt;      
#&gt;  1 marker_list1               MRK_List1.rpt            MRK_List1     Mouse Gene~
#&gt;  2 marker_list2               MRK_List2.rpt            MRK_List2     Mouse Gene~
#&gt;  3 marker_coordinates         MGI_MRK_Coord.rpt        MGI_MRK_Coord MGI Marker~
#&gt;  4 gene_model_coordinates     MGI_Gene_Model_Coord.rpt MGI_Gene_Mod~ MGI Gene M~
#&gt;  5 sequence_coordinates       MGI_GTGUP.gff            MGI_GTGUP     MGI Sequen~
#&gt;  6 genbank_refseq_ensembl_ids MRK_Sequence.rpt         MRK_Sequence  MGI Marker~
#&gt;  7 swiss_trembl_ids           MRK_SwissProt_TrEMBL.rpt MRK_SwissPro~ MGI Marker~
#&gt;  8 swiss_prot_ids             MRK_SwissProt.rpt        MRK_SwissProt MGI Marker~
#&gt;  9 gene_trap_ids              MRK_GeneTrap.rpt         MRK_GeneTrap  MGI Marker~
#&gt; 10 ensembl_ids                MRK_ENSEMBL.rpt          MRK_ENSEMBL   MGI Marker~
#&gt; 11 biotype_conflicts          MGI_BioTypeConflict.rpt  MGI_BioTypeC~ MGI Marker~
#&gt; 12 primers                    PRB_PrimerSeq.rpt        PRB_PrimerSeq MGI Marker~
#&gt; 13 interpro_domains           MGI_InterProDomains.rpt  MGI_InterPro~ InterPro d~
</pre></div>



<h3>Usage</h3>

<pre><code class='language-R'>read_report(
  report_key = NULL,
  report_file = NULL,
  report_type = NULL,
  n_max = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_report_+3A_report_key">report_key</code></td>
<td>
<p>A character vector. A key used to uniquely refer to an MGI
report.</p>
</td></tr>
<tr><td><code id="read_report_+3A_report_file">report_file</code></td>
<td>
<p>A character vector. The file path or URL to an MGI report
file.</p>
</td></tr>
<tr><td><code id="read_report_+3A_report_type">report_type</code></td>
<td>
<p>A character vector. The type of an MGI report.</p>
</td></tr>
<tr><td><code id="read_report_+3A_n_max">n_max</code></td>
<td>
<p>Maximum number of lines to read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> with report data in tidy
format. The set of variables is dependent on the specific report requested:
</p>

<ul>
<li><p> For <code>"marker_list1"</code>, see <code>vignette("marker_list1")</code>.
</p>
</li>
<li><p> For <code>"marker_list2"</code>, see <code>vignette("marker_list2")</code>.
</p>
</li>
<li><p> For <code>"marker_coordinates"</code>, see <code>vignette("marker_coordinates")</code>.
</p>
</li>
<li><p> For <code>"gene_model_coordinates"</code>, see <code>vignette("gene_model_coordinates")</code>.
</p>
</li>
<li><p> For <code>"sequence_coordinates"</code>, see <code>vignette("sequence_coordinates")</code>.
</p>
</li>
<li><p> For <code>"genbank_refseq_ensembl_ids"</code>, see <code>vignette("genbank_refseq_ensembl_ids")</code>.
</p>
</li>
<li><p> For <code>"swiss_trembl_ids"</code>, see <code>vignette("swiss_trembl_ids")</code>.
</p>
</li>
<li><p> For <code>"swiss_prot_ids"</code>, see <code>vignette("swiss_prot_ids")</code>.
</p>
</li>
<li><p> For <code>"gene_trap_ids"</code>, see <code>vignette("gene_trap_ids")</code>.
</p>
</li>
<li><p> For <code>"ensembl_ids"</code>, see <code>vignette("ensembl_ids")</code>.
</p>
</li>
<li><p> For <code>"biotype_conflicts"</code>, see <code>vignette("biotype_conflicts")</code>.
</p>
</li>
<li><p> For <code>"primers"</code>, see <code>vignette("primers")</code>.
</p>
</li>
<li><p> For <code>"interpro_domains"</code>, see <code>vignette("interpro_domains")</code>.
</p>
</li></ul>


<hr>
<h2 id='read_symbol_map'>Read marker symbol mappings</h2><span id='topic+read_symbol_map'></span>

<h3>Description</h3>

<p><code><a href="#topic+read_symbol_map">read_symbol_map()</a></code> reads MRK_List1.rpt data and returns an efficient
data table to be used for marker symbol remapping.
</p>
<p>This function is memoised.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_symbol_map(report_file = NULL, n_max = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_symbol_map_+3A_report_file">report_file</code></td>
<td>
<p>The path to a MRK_List1.rpt file. Leave this as <code>NULL</code>
and the function will automatically download the report from
<a href="https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt">https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt</a>.</p>
</td></tr>
<tr><td><code id="read_symbol_map_+3A_n_max">n_max</code></td>
<td>
<p>Maximum number of lines to read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="data.table.html#topic+data.table">data.table</a> where each row is a mapping,
from <code>marker_symbol</code> to <code>marker_symbol_now</code>:
</p>

<ul>
<li> <p><code>marker_symbol</code> (set as key): marker symbol is a unique abbreviation of the marker name.
</p>
</li>
<li> <p><code>marker_symbol_now</code>: genetic marker symbol replacement. If the record pertains a  <code>marker_symbol</code> that was withdrawn, then <code>marker_symbol_now</code> indicates the most recent in-use marker symbol that replaced it.
</p>
</li></ul>


<hr>
<h2 id='read_symbol_to_id_map'>Read marker symbol to identifier mappings</h2><span id='topic+read_symbol_to_id_map'></span>

<h3>Description</h3>

<p><code><a href="#topic+read_symbol_to_id_map">read_symbol_to_id_map()</a></code> reads MRK_List1.rpt data and returns an efficient
data table to be used for marker symbol remapping to marker identifiers. Note
that old symbols will be remapped to new symbols' ids, if applicable.
</p>
<p>This function is memoised.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_symbol_to_id_map(report_file = NULL, n_max = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_symbol_to_id_map_+3A_report_file">report_file</code></td>
<td>
<p>The path to a MRK_List1.rpt file. Leave this as <code>NULL</code>
and the function will automatically download the report from
<a href="https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt">https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt</a>.</p>
</td></tr>
<tr><td><code id="read_symbol_to_id_map_+3A_n_max">n_max</code></td>
<td>
<p>Maximum number of lines to read.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <a href="data.table.html#topic+data.table">data.table</a> where each row is a mapping,
from <code>marker_symbol</code> to <code>marker_id_now</code>:
</p>

<ul>
<li> <p><code>marker_symbol</code> (set as key): marker symbol is a unique abbreviation of the marker name.
</p>
</li>
<li> <p><code>marker_id_now</code>: genetic marker identifier replacement. If the record pertains a  <code>marker_symbol</code> that was withdrawn, then <code>marker_id_now</code> indicates the most recent in-use marker identifier that replaced it.
</p>
</li></ul>


<hr>
<h2 id='report_example'>Report example</h2><span id='topic+report_example'></span>

<h3>Description</h3>

<p><code><a href="#topic+report_example">report_example()</a></code> returns the local path of an example report file. These
files are typically very small and are useful for demonstrations. These
are mostly used in the Examples section of functions and in unit tests.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_example(report_file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report_example_+3A_report_file">report_file</code></td>
<td>
<p>File basename.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>report_example("MRK_List1-EX01.rpt")

report_example("MRK_List1-EX02.rpt")

report_example("MRK_List1-EX03.rpt")

</code></pre>

<hr>
<h2 id='report_last_modified'>Report last modification date</h2><span id='topic+report_last_modified'></span>

<h3>Description</h3>

<p><code><a href="#topic+report_last_modified">report_last_modified()</a></code> returns the last modified date and time of the
report source: local file or remote file. If a local file, the modification
date will be that indicated by the file system; if a remote file, the date
of last update is that provided by HTTP header <code>"last-modified"</code>.
</p>
<p>MGI updates its reports weekly, every Thursday. However, not all reports are
updated each week. The return value of this function is the closest you will
get to a versioning of MGI report files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_last_modified(tbl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report_last_modified_+3A_tbl">tbl</code></td>
<td>
<p>Report data as a <a href="tibble.html#topic+tibble-package">tibble</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A last modified date-time as a <a href="base.html#topic+DateTimeClasses">POSIXct</a>
object.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (FALSE) {
  markers &lt;- read_report("marker_list1", n_max = 10L)

  # When was the report file last updated?
  report_last_modified(markers)
}

</code></pre>

<hr>
<h2 id='report_source'>Report source</h2><span id='topic+report_source'></span>

<h3>Description</h3>

<p><code><a href="#topic+report_source">report_source()</a></code> returns the source used to obtain the report data:
a file path or an URL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_source(tbl)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report_source_+3A_tbl">tbl</code></td>
<td>
<p>Report data as a <a href="tibble.html#topic+tibble-package">tibble</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A single string with an absolute path to a file on disk or an URL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (FALSE) {
  markers &lt;- read_report("marker_list1", n_max = 10L)

  # Where did the data come from?
  report_source(markers)
}
</code></pre>

<hr>
<h2 id='report-attributes'>Get MGI report specs by report key</h2><span id='topic+report-attributes'></span><span id='topic+report_file'></span><span id='topic+report_name'></span><span id='topic+report_type'></span><span id='topic+report_url'></span>

<h3>Description</h3>

<p>Set of functions to retrieve metadata details of a MGI report.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>report_file(report_key)

report_name(report_key)

report_type(report_key)

report_url(report_key)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="report-attributes_+3A_report_key">report_key</code></td>
<td>
<p>A character vector. A key used to uniquely refer to an MGI
report.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector:
</p>

<ul>
<li> <p><code>report_file()</code>: report file name as hosted in <a href="https://www.informatics.jax.org/downloads/reports/">https://www.informatics.jax.org/downloads/reports/</a>.
</p>
</li>
<li> <p><code>report_name()</code>: report title.
</p>
</li>
<li> <p><code>report_type()</code>: report type.
</p>
</li>
<li> <p><code>report_url()</code>: report remote location.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>report_file("marker_list1")

report_name("marker_list1")

report_type("marker_list1")

report_url("marker_list1")

</code></pre>

<hr>
<h2 id='reports'>Supported MGI reports</h2><span id='topic+reports'></span>

<h3>Description</h3>

<p><code>reports</code> is a data set of supported MGI reports, meaning reports that
<code>{mgi.report.reader}</code> can currently read into R.
To browse all reports made available by MGI visit
<a href="https://www.informatics.jax.org/downloads/reports/">https://www.informatics.jax.org/downloads/reports/</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reports
</code></pre>


<h3>Format</h3>

<p>A <a href="tibble.html#topic+tibble-package">tibble</a> of 4 variables:
</p>

<dl>
<dt><code>report_key</code></dt><dd><p>A string key used to uniquely refer to an MGI report, which is only meaningful within the context of the <code>{mgi.report.reader}</code>.</p>
</dd>
<dt><code>report_file</code></dt><dd><p>MGI report file name as hosted at <a href="https://www.informatics.jax.org/downloads/reports/">https://www.informatics.jax.org/downloads/reports/</a>.</p>
</dd>
<dt><code>report_type</code></dt><dd><p>MGI report type. The type is used internally to find the appropriate reader for parsing, and is only meaningful within the context of <code>{mgi.report.reader}</code>.</p>
</dd>
<dt><code>report_name</code></dt><dd><p>MGI report name. Report names are taken from <a href="https://www.informatics.jax.org/downloads/reports/index.html">https://www.informatics.jax.org/downloads/reports/index.html</a>.</p>
</dd>
</dl>



<h3>Examples</h3>

<pre><code class='language-R'>reports

</code></pre>

<hr>
<h2 id='sure'>Are you sure?</h2><span id='topic+sure'></span>

<h3>Description</h3>

<p>This function asks you interactively for permission to continue or not. You
can specify a custom message before the question and also different messages
for a both a positive and negative answer.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sure(
  before_question = NULL,
  after_saying_no = NULL,
  after_saying_yes = NULL,
  default_answer = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sure_+3A_before_question">before_question</code></td>
<td>
<p>String with message to be printed before question.</p>
</td></tr>
<tr><td><code id="sure_+3A_after_saying_no">after_saying_no</code></td>
<td>
<p>String with message to be printed after answering
<code>'no'</code>.</p>
</td></tr>
<tr><td><code id="sure_+3A_after_saying_yes">after_saying_yes</code></td>
<td>
<p>String with message to be printed after answering
<code>'yes'</code>.</p>
</td></tr>
<tr><td><code id="sure_+3A_default_answer">default_answer</code></td>
<td>
<p>String with answer to question, if run in
non-interactive mode.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you run this function in non-interactive mode, you should pass an
automatic answer to <code>default_answer</code>: <code>'yes'</code> or <code>'no'</code>.
</p>


<h3>Value</h3>

<p>A logical indicating if answer was <code>'yes'</code>/<code>'y'</code>
(<code>TRUE</code>) or otherwise (<code>FALSE</code>).
</p>

<hr>
<h2 id='symbol_to_identifier'>Convert marker symbols to updated marker identifiers</h2><span id='topic+symbol_to_identifier'></span>

<h3>Description</h3>

<p><code><a href="#topic+symbol_to_identifier">symbol_to_identifier()</a></code> remaps old marker symbols to, in-use, most up
to date marker identifiers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symbol_to_identifier(x, report_file = NULL, n_max = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="symbol_to_identifier_+3A_x">x</code></td>
<td>
<p>A character vector of marker symbols to be remapped.</p>
</td></tr>
<tr><td><code id="symbol_to_identifier_+3A_report_file">report_file</code></td>
<td>
<p>The path to a MRK_List1.rpt file. Leave this as <code>NULL</code>
and the function will automatically download the report from
<a href="https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt">https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt</a>.</p>
</td></tr>
<tr><td><code id="symbol_to_identifier_+3A_n_max">n_max</code></td>
<td>
<p>Maximum number of lines to read from the <code>report_file</code>.</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>rpt_ex01 &lt;- report_example("MRK_List1-EX01.rpt")
read_report(report_file = rpt_ex01, report_type = "MRK_List1") |&gt;
  dplyr::select("marker_status", "marker_symbol", "marker_id_now")

# NB:
#   - "1700024N20Rik" has two conflicting mappings, so maps to `NA`.
#   - "Hes1" is not present in MRK_List1-EX01.rpt, so maps to `NA`.
#   - "Plpbp" (official) and "Prosc" (withdrawn) both map to "MGI:1891207"

marker_symbols &lt;- c("2200002F22Rik", "Plpbp", "Prosc", "1700024N20Rik", "Hes1")
symbol_to_identifier(x = marker_symbols, report_file = rpt_ex01)

</code></pre>

<hr>
<h2 id='symbol_to_symbol'>Update marker symbols</h2><span id='topic+symbol_to_symbol'></span>

<h3>Description</h3>

<p><code><a href="#topic+symbol_to_symbol">symbol_to_symbol()</a></code> remaps old marker symbols to, in-use, most up to
date symbols.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>symbol_to_symbol(x, report_file = NULL, n_max = Inf)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="symbol_to_symbol_+3A_x">x</code></td>
<td>
<p>A character vector of marker symbols to be remapped.</p>
</td></tr>
<tr><td><code id="symbol_to_symbol_+3A_report_file">report_file</code></td>
<td>
<p>The path to a MRK_List1.rpt file. Leave this as <code>NULL</code>
and the function will automatically download the report from
<a href="https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt">https://www.informatics.jax.org/downloads/reports/MRK_List1.rpt</a>.</p>
</td></tr>
<tr><td><code id="symbol_to_symbol_+3A_n_max">n_max</code></td>
<td>
<p>Maximum number of lines to read from the <code>report_file</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector of most up to date symbols.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>rpt_ex01 &lt;- report_example("MRK_List1-EX01.rpt")
read_report(report_file = rpt_ex01, report_type = "MRK_List1") |&gt;
  dplyr::select("marker_status", "marker_symbol", "marker_symbol_now")

# NB:
#   - "1700024N20Rik" has two conflicting mappings, so maps to `NA`.
#   - "Hes1" is not present in MRK_List1-EX01.rpt, so maps to `NA`.
#   - "Plpbp" (official) and "Prosc" (withdrawn) both map to "Plpbp"

marker_symbols &lt;- c("2200002F22Rik", "Plpbp", "Prosc", "1700024N20Rik", "Hes1")
symbol_to_symbol(x = marker_symbols, report_file = rpt_ex01)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
