<!DOCTYPE html><html lang="en-US"><head><title>Help for package mhcnuggetsr</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mhcnuggetsr}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#are_mhcnuggets_names'><p>Are these MHCnuggets names?</p></a></li>
<li><a href='#check_mhc_class'><p>Check the MHC class.</p></a></li>
<li><a href='#check_mhcnuggets_installation'><p>Check if MHCnuggets is installed.</p></a></li>
<li><a href='#check_mhcnuggets_options'><p>Check the MHCnuggets options.</p></a></li>
<li><a href='#check_mhcnuggets_options_names'><p>Check the names of the elements in an <code>mhcnuggets_options</code> list.</p></a></li>
<li><a href='#create_mhcnuggets_options'><p>Create a set of MHCnuggets options.</p></a></li>
<li><a href='#create_temp_peptides_path'><p>Create a path to a non-existing temporary file</p></a></li>
<li><a href='#create_test_mhcnuggets_options'><p>Create testing options for MHCnuggets</p></a></li>
<li><a href='#default_params_doc'><p>This function does nothing. It is intended to inherit the documentation</p>
of the parameters from.</a></li>
<li><a href='#downgrade_pip'><p>Downgrade pip.</p></a></li>
<li><a href='#get_default_mhcnuggets_folder'><p>Get the path to the folder where this package installs</p>
MHCnuggets by default</a></li>
<li><a href='#get_example_filename'><p>Get the full path to an MHCnuggets example file</p></a></li>
<li><a href='#get_example_filenames'><p>Get the full path to all MHCnuggets example files</p></a></li>
<li><a href='#get_mhc_1_haplotypes'><p>Get all the MHC-I haplotypes</p></a></li>
<li><a href='#get_mhc_2_haplotypes'><p>Get all the MHC-II haplotypes</p></a></li>
<li><a href='#get_mhcnuggets_url'><p>Get the URL of the MHCnuggets source code</p></a></li>
<li><a href='#get_mhcnuggets_version'><p>Get the MHCnuggets version</p></a></li>
<li><a href='#get_pip_version'><p>Get the version of pip</p></a></li>
<li><a href='#get_python_package_versions'><p>Get the version of all Python packages</p></a></li>
<li><a href='#get_trained_mhc_1_haplotypes'><p>Get all the MHC-I haplotypes that have been trained on a model</p></a></li>
<li><a href='#get_trained_mhc_2_haplotypes'><p>Get all the MHC-II haplotypes that have been trained on a model</p></a></li>
<li><a href='#install_mhcnuggets'><p>Install the <code>MHCnuggets</code> <code>Python</code> package.</p></a></li>
<li><a href='#install_pip'><p>Install pip.</p></a></li>
<li><a href='#is_mhcnuggets_installed'><p>Check if MHCnuggets is installed</p></a></li>
<li><a href='#is_mhcnuggets_name'><p>Is this an MHCnuggets name?</p></a></li>
<li><a href='#is_mhcnuggets_options'><p>Is this a <code>mhcnuggets_options</code>?</p></a></li>
<li><a href='#is_on_appveyor'><p>Determines if the environment is AppVeyor</p></a></li>
<li><a href='#is_on_ci'><p>Determines if the environment is a continuous integration service</p></a></li>
<li><a href='#is_on_travis'><p>Determines if the environment is Travis CI</p></a></li>
<li><a href='#is_pip_installed'><p>Determine if pip is installed</p></a></li>
<li><a href='#mhcnuggetsr'><p>mhcnuggetsr: estimate the topoplogy of membrane proteins</p></a></li>
<li><a href='#mhcnuggetsr_report'><p>Create a mhcnuggetsr report, to be used when reporting bugs</p></a></li>
<li><a href='#mhcnuggetsr_self_test'><p>Self-test the package</p></a></li>
<li><a href='#predict_ic50'><p>Predict the IC50 for peptides.</p></a></li>
<li><a href='#predict_ic50_from_file'><p>Predict the half maximal inhibitory concentration (aka IC50) (in nM)</p>
for one or more peptides as saved in a file.
Each peptide must be 15 amino acids at
most (use predict_ic50s to predict the IC50s for longer peptides)</a></li>
<li><a href='#predict_ic50s'><p>Predict the half maximal inhibitory concentrations (aka IC50s) (in nM)</p>
of all possible n-mers within a peptide</a></li>
<li><a href='#set_is_mhcnuggets_installed'><p>Set the MHCnuggets installation state to the desired one</p></a></li>
<li><a href='#set_pip_version'><p>Set the version of pip.</p></a></li>
<li><a href='#to_mhcnuggets_name'><p>Convert a standard haplotype name to the MHCnuggets name</p></a></li>
<li><a href='#to_mhcnuggets_names'><p>Convert one or more standard haplotype name to the MHCnuggets names</p></a></li>
<li><a href='#uninstall_mhcnuggets'><p>Uninstall the MHCnuggets Python package.</p></a></li>
<li><a href='#uninstall_pip'><p>Install pip.</p></a></li>
<li><a href='#upgrade_pip'><p>Uograde pip.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Title:</td>
<td>Call MHCnuggets</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Richèl J.C. Bilderbeek &lt;richel@richelbilderbeek.nl&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>
    MHCnuggets (<a href="https://github.com/KarchinLab/mhcnuggets">https://github.com/KarchinLab/mhcnuggets</a>) is a Python
    tool to predict MHC class I and MHC class II epitopes.
    This package allows one to call MHCnuggets from R.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>rappdirs, reticulate, stringr, tibble</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, testthat, rmarkdown</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/richelbilderbeek/mhcnuggetsr/">https://github.com/richelbilderbeek/mhcnuggetsr/</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/richelbilderbeek/mhcnuggetsr">https://github.com/richelbilderbeek/mhcnuggetsr</a></td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-10-27 09:13:47 UTC; richel</td>
</tr>
<tr>
<td>Author:</td>
<td>Richèl J.C. Bilderbeek
    <a href="https://orcid.org/0000-0003-1107-7049"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-11-04 11:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='are_mhcnuggets_names'>Are these MHCnuggets names?</h2><span id='topic+are_mhcnuggets_names'></span>

<h3>Description</h3>

<p>Determine if an HLA haplotype name follow the name format
that MHCnuggets uses internally
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_mhcnuggets_names(mhcs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="are_mhcnuggets_names_+3A_mhcs">mhcs</code></td>
<td>
<p>the MHC haplotype names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a vector of booleans, TRUE for HLA haplotypes that follow
the MHCnuggets naming convention
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  are_mhcnuggets_names(get_mhc_1_haplotypes())
  are_mhcnuggets_names(get_mhc_2_haplotypes())
}
</code></pre>

<hr>
<h2 id='check_mhc_class'>Check the MHC class.</h2><span id='topic+check_mhc_class'></span>

<h3>Description</h3>

<p>Check the MHC class. Will <a href="base.html#topic+stop">stop</a> if it is invalid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mhc_class(mhc_class)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_mhc_class_+3A_mhc_class">mhc_class</code></td>
<td>
<p>MHC class. Must be <code>I</code>, <code>II</code>
or <a href="base.html#topic+NA">NA</a>. Use <a href="base.html#topic+NA">NA</a> to let the class be deduced automatically</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>check_mhc_class("I")
check_mhc_class("II")
check_mhc_class(NA)
</code></pre>

<hr>
<h2 id='check_mhcnuggets_installation'>Check if MHCnuggets is installed.</h2><span id='topic+check_mhcnuggets_installation'></span>

<h3>Description</h3>

<p>Check if MHCnuggets is installed. Will <a href="base.html#topic+stop">stop</a> if not.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mhcnuggets_installation(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_mhcnuggets_installation_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="check_mhcnuggets_installation_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Details</h3>

<p>An MHCnuggets installation has two parts:
</p>

<ol>
<li><p> The installation of the Python package,
for running the code
</p>
</li>
<li><p> The download of the Python source code,
which allows the use of example files
</p>
</li></ol>



<h3>Value</h3>

<p>Nothing.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  check_mhcnuggets_installation()
}
</code></pre>

<hr>
<h2 id='check_mhcnuggets_options'>Check the MHCnuggets options.</h2><span id='topic+check_mhcnuggets_options'></span>

<h3>Description</h3>

<p>Check the MHCnuggets options.
Will <a href="base.html#topic+stop">stop</a> if the options are invalid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mhcnuggets_options(mhcnuggets_options)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_mhcnuggets_options_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>an <code>mhcnuggets_options</code> created
by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a> is always
checked by <a href="#topic+check_mhcnuggets_options">check_mhcnuggets_options</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  check_mhcnuggets_options(create_test_mhcnuggets_options())
}
</code></pre>

<hr>
<h2 id='check_mhcnuggets_options_names'>Check the names of the elements in an <code>mhcnuggets_options</code> list.</h2><span id='topic+check_mhcnuggets_options_names'></span>

<h3>Description</h3>

<p>Check the names of the elements in an <code>mhcnuggets_options</code> list.
Will <a href="base.html#topic+stop">stop</a> if an element is missing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>check_mhcnuggets_options_names(mhcnuggets_options)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="check_mhcnuggets_options_names_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  check_mhcnuggets_options_names(
    create_test_mhcnuggets_options()
  )
}
</code></pre>

<hr>
<h2 id='create_mhcnuggets_options'>Create a set of MHCnuggets options.</h2><span id='topic+create_mhcnuggets_options'></span>

<h3>Description</h3>

<p>Create a set of options to run MHCnuggets with.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_mhcnuggets_options(
  mhc_class = NA,
  mhc,
  ba_models = FALSE,
  verbose = FALSE,
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_mhcnuggets_options_+3A_mhc_class">mhc_class</code></td>
<td>
<p>MHC class. Must be <code>I</code>, <code>II</code>
or <a href="base.html#topic+NA">NA</a>. Use <a href="base.html#topic+NA">NA</a> to let the class be deduced automatically</p>
</td></tr>
<tr><td><code id="create_mhcnuggets_options_+3A_mhc">mhc</code></td>
<td>
<p>the MHC haplotype name</p>
</td></tr>
<tr><td><code id="create_mhcnuggets_options_+3A_ba_models">ba_models</code></td>
<td>
<p>Set to TRUE to use a pure BA model</p>
</td></tr>
<tr><td><code id="create_mhcnuggets_options_+3A_verbose">verbose</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> for more debug information</p>
</td></tr>
<tr><td><code id="create_mhcnuggets_options_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="create_mhcnuggets_options_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function will give an error message if the arguments are invalid.
</p>


<h3>Value</h3>

<p>an <code>mhcnuggets_options</code>
</p>


<h3>Note</h3>

<p>an <code>mhcnuggets_options</code> created
by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a> is always
checked by <a href="#topic+check_mhcnuggets_options">check_mhcnuggets_options</a>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+create_test_mhcnuggets_options">create_test_mhcnuggets_options</a> to create an
MHCnuggets object for testing
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  create_mhcnuggets_options(
    mhc = "HLA-A02:01"
  )
}
</code></pre>

<hr>
<h2 id='create_temp_peptides_path'>Create a path to a non-existing temporary file</h2><span id='topic+create_temp_peptides_path'></span>

<h3>Description</h3>

<p>Create a path to a non-existing temporary file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_temp_peptides_path(fileext = ".fasta")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_temp_peptides_path_+3A_fileext">fileext</code></td>
<td>
<p>file extension</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>create_temp_peptides_path()
</code></pre>

<hr>
<h2 id='create_test_mhcnuggets_options'>Create testing options for MHCnuggets</h2><span id='topic+create_test_mhcnuggets_options'></span>

<h3>Description</h3>

<p>Create a set of testing options to run MHCnuggets with.
The most important setting is the use of a specific haplotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_test_mhcnuggets_options(
  mhc_class = NA,
  mhc = "HLA-A02:01",
  ba_models = FALSE,
  verbose = FALSE,
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="create_test_mhcnuggets_options_+3A_mhc_class">mhc_class</code></td>
<td>
<p>MHC class. Must be <code>I</code>, <code>II</code>
or <a href="base.html#topic+NA">NA</a>. Use <a href="base.html#topic+NA">NA</a> to let the class be deduced automatically</p>
</td></tr>
<tr><td><code id="create_test_mhcnuggets_options_+3A_mhc">mhc</code></td>
<td>
<p>the MHC haplotype name</p>
</td></tr>
<tr><td><code id="create_test_mhcnuggets_options_+3A_ba_models">ba_models</code></td>
<td>
<p>Set to TRUE to use a pure BA model</p>
</td></tr>
<tr><td><code id="create_test_mhcnuggets_options_+3A_verbose">verbose</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> for more debug information</p>
</td></tr>
<tr><td><code id="create_test_mhcnuggets_options_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="create_test_mhcnuggets_options_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  create_test_mhcnuggets_options()
}
</code></pre>

<hr>
<h2 id='default_params_doc'>This function does nothing. It is intended to inherit the documentation
of the parameters from.</h2><span id='topic+default_params_doc'></span>

<h3>Description</h3>

<p>This function does nothing. It is intended to inherit the documentation
of the parameters from.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>default_params_doc(
  ba_models,
  folder_name,
  mhc,
  mhcs,
  mhc_class,
  mhcnuggets_options,
  mhcnuggets_url,
  n_aas,
  peptide,
  peptides,
  peptide_length,
  peptides_path,
  protein_sequence,
  verbose
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="default_params_doc_+3A_ba_models">ba_models</code></td>
<td>
<p>Set to TRUE to use a pure BA model</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mhc">mhc</code></td>
<td>
<p>the MHC haplotype name</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mhcs">mhcs</code></td>
<td>
<p>the MHC haplotype names</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mhc_class">mhc_class</code></td>
<td>
<p>MHC class. Must be <code>I</code>, <code>II</code>
or <a href="base.html#topic+NA">NA</a>. Use <a href="base.html#topic+NA">NA</a> to let the class be deduced automatically</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_n_aas">n_aas</code></td>
<td>
<p>number of amino acids</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_peptide">peptide</code></td>
<td>
<p>one peptide sequence</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_peptides">peptides</code></td>
<td>
<p>one of more peptide sequences</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_peptide_length">peptide_length</code></td>
<td>
<p>length of a peptide, in number of amino acids</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_peptides_path">peptides_path</code></td>
<td>
<p>the path to the peptides</p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>protein sequence, in uppercase,
for example <code>FAMILYVW</code></p>
</td></tr>
<tr><td><code id="default_params_doc_+3A_verbose">verbose</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> for more debug information</p>
</td></tr>
</table>


<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@noRd</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='downgrade_pip'>Downgrade pip.</h2><span id='topic+downgrade_pip'></span>

<h3>Description</h3>

<p>Set the version of pip to a specific earlier version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>downgrade_pip(version = "9.0.0")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="downgrade_pip_+3A_version">version</code></td>
<td>
<p>pip version</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+upgrade_pip">upgrade_pip</a> to set pip to the latest version.
Use <a href="#topic+set_pip_version">set_pip_version</a> to install a specific version of pip
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  if (is_pip_installed()) {
    downgrade_pip()
  }

## End(Not run)
</code></pre>

<hr>
<h2 id='get_default_mhcnuggets_folder'>Get the path to the folder where this package installs
MHCnuggets by default</h2><span id='topic+get_default_mhcnuggets_folder'></span>

<h3>Description</h3>

<p>Get the path to the folder where this package installs
MHCnuggets by default
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_default_mhcnuggets_folder()
</code></pre>


<h3>Value</h3>

<p>the path to the folder where this package installs
MHCnuggets by default
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_default_mhcnuggets_folder()
</code></pre>

<hr>
<h2 id='get_example_filename'>Get the full path to an MHCnuggets example file</h2><span id='topic+get_example_filename'></span>

<h3>Description</h3>

<p>Get the full path to an MHCnuggets example file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_filename(
  filename = "test_peptides.peps",
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_example_filename_+3A_filename">filename</code></td>
<td>
<p>name of the example file, without the path</p>
</td></tr>
<tr><td><code id="get_example_filename_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_example_filename_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the full path to an MHCnuggets example file
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+get_example_filenames">get_example_filenames</a> to get all MHCnuggets
example filenames
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_example_filename("test_peptides.peps")
}
</code></pre>

<hr>
<h2 id='get_example_filenames'>Get the full path to all MHCnuggets example files</h2><span id='topic+get_example_filenames'></span>

<h3>Description</h3>

<p>Get the full path to all MHCnuggets example files
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_example_filenames(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_example_filenames_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_example_filenames_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with all MHCnuggets example files
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+get_example_filename">get_example_filename</a> to get the full
path to a MHCnuggets example file
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_example_filenames()
}
</code></pre>

<hr>
<h2 id='get_mhc_1_haplotypes'>Get all the MHC-I haplotypes</h2><span id='topic+get_mhc_1_haplotypes'></span>

<h3>Description</h3>

<p>Get all the MHC-I haplotypes that MHCnuggets has been trained upon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mhc_1_haplotypes(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mhc_1_haplotypes_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_mhc_1_haplotypes_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with haplotype names in MHCnuggets format
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_mhc_1_haplotypes()
}
</code></pre>

<hr>
<h2 id='get_mhc_2_haplotypes'>Get all the MHC-II haplotypes</h2><span id='topic+get_mhc_2_haplotypes'></span>

<h3>Description</h3>

<p>Get all the MHC-II haplotypes that MHCnuggets has been trained upon.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mhc_2_haplotypes(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mhc_2_haplotypes_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_mhc_2_haplotypes_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with haplotype names in MHCnuggets format
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_mhc_2_haplotypes()
}
</code></pre>

<hr>
<h2 id='get_mhcnuggets_url'>Get the URL of the MHCnuggets source code</h2><span id='topic+get_mhcnuggets_url'></span>

<h3>Description</h3>

<p>Get the URL of the MHCnuggets source code
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mhcnuggets_url()
</code></pre>


<h3>Value</h3>

<p>a string that is a URL
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_mhcnuggets_url()
</code></pre>

<hr>
<h2 id='get_mhcnuggets_version'>Get the MHCnuggets version</h2><span id='topic+get_mhcnuggets_version'></span>

<h3>Description</h3>

<p>Get the MHCnuggets version
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_mhcnuggets_version(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_mhcnuggets_version_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_mhcnuggets_version_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a string that is a version, for example <code>2.3.2</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_mhcnuggets_version()
}
</code></pre>

<hr>
<h2 id='get_pip_version'>Get the version of pip</h2><span id='topic+get_pip_version'></span>

<h3>Description</h3>

<p>Get the version of pip
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_pip_version()
</code></pre>


<h3>Value</h3>

<p>a string that is a version, for example <code>20.2</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_pip_installed()) {
  get_pip_version()
}
</code></pre>

<hr>
<h2 id='get_python_package_versions'>Get the version of all Python packages</h2><span id='topic+get_python_package_versions'></span>

<h3>Description</h3>

<p>Get the version of all Python packages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_python_package_versions()
</code></pre>


<h3>Value</h3>

<p>a tibble with two columns: (1) <code>package</code>, the name of the
package, for example <code>absl-py</code>, (2) <code>version</code>, the version
of that package, for example <code>0.9.0</code>
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (rappdirs::app_dir()$os != "win" &amp;&amp; is_pip_installed()) {
  get_python_package_versions()
}
</code></pre>

<hr>
<h2 id='get_trained_mhc_1_haplotypes'>Get all the MHC-I haplotypes that have been trained on a model</h2><span id='topic+get_trained_mhc_1_haplotypes'></span>

<h3>Description</h3>

<p>Get all the MHC-I haplotypes that have been trained on a model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trained_mhc_1_haplotypes(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_trained_mhc_1_haplotypes_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_trained_mhc_1_haplotypes_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with haplotype names in MHCnuggets format
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_trained_mhc_1_haplotypes()
}
</code></pre>

<hr>
<h2 id='get_trained_mhc_2_haplotypes'>Get all the MHC-II haplotypes that have been trained on a model</h2><span id='topic+get_trained_mhc_2_haplotypes'></span>

<h3>Description</h3>

<p>Get all the MHC-II haplotypes that have been trained on a model
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_trained_mhc_2_haplotypes(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_trained_mhc_2_haplotypes_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="get_trained_mhc_2_haplotypes_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a character vector with haplotype names in MHCnuggets format
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  get_trained_mhc_2_haplotypes()
}
</code></pre>

<hr>
<h2 id='install_mhcnuggets'>Install the <code>MHCnuggets</code> <code>Python</code> package.</h2><span id='topic+install_mhcnuggets'></span>

<h3>Description</h3>

<p>Install the <code>MHCnuggets</code> <code>Python</code> package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_mhcnuggets(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="install_mhcnuggets_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="install_mhcnuggets_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  install_mhcnuggets()

## End(Not run)
</code></pre>

<hr>
<h2 id='install_pip'>Install pip.</h2><span id='topic+install_pip'></span>

<h3>Description</h3>

<p>Install pip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>install_pip()
</code></pre>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  install_pip()

## End(Not run)
</code></pre>

<hr>
<h2 id='is_mhcnuggets_installed'>Check if MHCnuggets is installed</h2><span id='topic+is_mhcnuggets_installed'></span>

<h3>Description</h3>

<p>Check if MHCnuggets is installed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mhcnuggets_installed(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_mhcnuggets_installed_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="is_mhcnuggets_installed_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if MHCnuggets is installed
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_mhcnuggets_installed()
</code></pre>

<hr>
<h2 id='is_mhcnuggets_name'>Is this an MHCnuggets name?</h2><span id='topic+is_mhcnuggets_name'></span>

<h3>Description</h3>

<p>Determine if an HLA haplotype name follow the name format
that MHCnuggets uses internally
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mhcnuggets_name(mhc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_mhcnuggets_name_+3A_mhc">mhc</code></td>
<td>
<p>the MHC haplotype name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if the name follows the MHCnuggets naming convention
</p>


<h3>Examples</h3>

<pre><code class='language-R'># The official name is not the name format used by MHCnuggets
is_mhcnuggets_name("HLA-A*01:01")

# MHCnuggets uses names without the asterisk
is_mhcnuggets_name("HLA-A01:01")
</code></pre>

<hr>
<h2 id='is_mhcnuggets_options'>Is this a <code>mhcnuggets_options</code>?</h2><span id='topic+is_mhcnuggets_options'></span>

<h3>Description</h3>

<p>Determine if the MHCnuggets options is valid.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mhcnuggets_options(mhcnuggets_options)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="is_mhcnuggets_options_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>TRUE if this a valid set of MHCnuggets options
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  is_mhcnuggets_options(create_test_mhcnuggets_options())
}
</code></pre>

<hr>
<h2 id='is_on_appveyor'>Determines if the environment is AppVeyor</h2><span id='topic+is_on_appveyor'></span>

<h3>Description</h3>

<p>Determines if the environment is AppVeyor
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_appveyor()
</code></pre>


<h3>Value</h3>

<p>TRUE if run on AppVeyor, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_on_appveyor()
</code></pre>

<hr>
<h2 id='is_on_ci'>Determines if the environment is a continuous integration service</h2><span id='topic+is_on_ci'></span>

<h3>Description</h3>

<p>Determines if the environment is a continuous integration service
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>


<h3>Value</h3>

<p>TRUE if run on AppVeyor or Travis CI, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_on_ci()
</code></pre>

<hr>
<h2 id='is_on_travis'>Determines if the environment is Travis CI</h2><span id='topic+is_on_travis'></span>

<h3>Description</h3>

<p>Determines if the environment is Travis CI
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_on_travis()
</code></pre>


<h3>Value</h3>

<p>TRUE if run on Travis CI, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_on_travis()
</code></pre>

<hr>
<h2 id='is_pip_installed'>Determine if pip is installed</h2><span id='topic+is_pip_installed'></span>

<h3>Description</h3>

<p>Determine if pip is installed
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_pip_installed()
</code></pre>


<h3>Value</h3>

<p>TRUE if pip is installed, FALSE otherwise
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>is_pip_installed()
</code></pre>

<hr>
<h2 id='mhcnuggetsr'>mhcnuggetsr: estimate the topoplogy of membrane proteins</h2><span id='topic+mhcnuggetsr'></span>

<h3>Description</h3>

<p>Proteins reside in either the cell plasma of in the
cell membrane. A membrane protein goes through the
membrane at least once. There are multiple ways to span this
hydrophobic layer. One common structure is the transmembrane (alpha)
helix (TMH). Given the amino acid sequence of a membrane protein, this
package predicts which parts of the protein are TMHs
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {

  peptides_path &lt;- get_example_filename("test_peptides.peps")
  mhc_1_haplotype &lt;- "HLA-A02:01"

  mhcnuggets_options &lt;- create_mhcnuggets_options(
    mhc = mhc_1_haplotype
  )

  predict_ic50_from_file(
    peptides_path = peptides_path,
    mhcnuggets_options = mhcnuggets_options
  )
}
</code></pre>

<hr>
<h2 id='mhcnuggetsr_report'>Create a <a href="#topic+mhcnuggetsr">mhcnuggetsr</a> report, to be used when reporting bugs</h2><span id='topic+mhcnuggetsr_report'></span>

<h3>Description</h3>

<p>Create a <a href="#topic+mhcnuggetsr">mhcnuggetsr</a> report, to be used when reporting bugs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mhcnuggetsr_report(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mhcnuggetsr_report_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="mhcnuggetsr_report_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  mhcnuggetsr_report()

## End(Not run)
</code></pre>

<hr>
<h2 id='mhcnuggetsr_self_test'>Self-test the package</h2><span id='topic+mhcnuggetsr_self_test'></span>

<h3>Description</h3>

<p>Self-test the package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mhcnuggetsr_self_test(mhcnuggets_options = create_test_mhcnuggets_options())
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mhcnuggetsr_self_test_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {
  mhcnuggetsr_self_test()
}
</code></pre>

<hr>
<h2 id='predict_ic50'>Predict the IC50 for peptides.</h2><span id='topic+predict_ic50'></span>

<h3>Description</h3>

<p>Predict the half maximal inhibitory concentration (aka IC50) (in nM)
for one or more peptides.
Each peptide must be 15 amino acids at
most (use <a href="#topic+predict_ic50s">predict_ic50s</a> to predict the IC50s for longer peptides)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_ic50(
  mhcnuggets_options,
  peptides,
  peptides_path = create_temp_peptides_path()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict_ic50_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
<tr><td><code id="predict_ic50_+3A_peptides">peptides</code></td>
<td>
<p>one of more peptide sequences</p>
</td></tr>
<tr><td><code id="predict_ic50_+3A_peptides_path">peptides_path</code></td>
<td>
<p>the path to the peptides</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble with two columns:
(1) <code>peptide</code>, which holds the peptide sequence, and
(2) <code>ic50</code>, which holds the predicted IC50
</p>


<h3>Note</h3>

<p>this function uses a temporary file, because MHCnuggets reads its
input from file. This temporary file is deleted after this function
passed successfully.
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {

  peptides &lt;- c("AIAACAMLLV", "ALVCYIVMPV", "ALEPRKEIDV")
  mhc_1_haplotype &lt;- "HLA-A02:01"

  mhcnuggets_options &lt;- create_mhcnuggets_options(
    mhc = mhc_1_haplotype
  )

  predict_ic50(
    peptides = peptides,
    mhcnuggets_options = mhcnuggets_options
  )
}
</code></pre>

<hr>
<h2 id='predict_ic50_from_file'>Predict the half maximal inhibitory concentration (aka IC50) (in nM)
for one or more peptides as saved in a file.
Each peptide must be 15 amino acids at
most (use <a href="#topic+predict_ic50s">predict_ic50s</a> to predict the IC50s for longer peptides)</h2><span id='topic+predict_ic50_from_file'></span>

<h3>Description</h3>

<p>Predict the half maximal inhibitory concentration (aka IC50) (in nM)
for one or more peptides as saved in a file.
Each peptide must be 15 amino acids at
most (use <a href="#topic+predict_ic50s">predict_ic50s</a> to predict the IC50s for longer peptides)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_ic50_from_file(
  mhcnuggets_options,
  peptides_path,
  mhcnuggets_output_filename = mhcnuggetsr::create_temp_peptides_path(fileext = ".csv")
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict_ic50_from_file_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
<tr><td><code id="predict_ic50_from_file_+3A_peptides_path">peptides_path</code></td>
<td>
<p>the path to the peptides</p>
</td></tr>
<tr><td><code id="predict_ic50_from_file_+3A_mhcnuggets_output_filename">mhcnuggets_output_filename</code></td>
<td>
<p>path to a temporary file to write
the MHCnuggets results to. This file will be deleted at the end of
the function if it passes successfully.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {

  peptides_path &lt;- get_example_filename("test_peptides.peps")
  mhc_1_haplotype &lt;- "HLA-A02:01"
  mhcnuggets_options &lt;- create_mhcnuggets_options(
    mhc = mhc_1_haplotype
  )

  predict_ic50_from_file(
    peptides_path = peptides_path,
    mhcnuggets_options = mhcnuggets_options
  )
}
</code></pre>

<hr>
<h2 id='predict_ic50s'>Predict the half maximal inhibitory concentrations (aka IC50s) (in nM)
of all possible n-mers within a peptide</h2><span id='topic+predict_ic50s'></span>

<h3>Description</h3>

<p>Predict the half maximal inhibitory concentrations (aka IC50s) (in nM)
of all possible n-mers within a peptide
</p>


<h3>Usage</h3>

<pre><code class='language-R'>predict_ic50s(
  protein_sequence,
  peptide_length,
  mhcnuggets_options,
  peptides_path = create_temp_peptides_path()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="predict_ic50s_+3A_protein_sequence">protein_sequence</code></td>
<td>
<p>protein sequence, in uppercase,
for example <code>FAMILYVW</code></p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_peptide_length">peptide_length</code></td>
<td>
<p>length of a peptide, in number of amino acids</p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_mhcnuggets_options">mhcnuggets_options</code></td>
<td>
<p>options to run MHCnuggets with,
as can be created by <a href="#topic+create_mhcnuggets_options">create_mhcnuggets_options</a>.</p>
</td></tr>
<tr><td><code id="predict_ic50s_+3A_peptides_path">peptides_path</code></td>
<td>
<p>the path to the peptides</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a tibble with columns:<br />
</p>

<ul>
<li><p> peptide the peptide fragment, each of length <code>peptide_length</code>
</p>
</li>
<li><p> ic50 the predicted IC50 (in nM)
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (is_mhcnuggets_installed()) {

  mhcnuggets_options &lt;- create_mhcnuggets_options(
    mhc = "HLA-A02:01"
  )

  predict_ic50s(
    protein_sequence = "AIAACAMLLVCCCCCC",
    peptide_length = 13,
    mhcnuggets_options = mhcnuggets_options
  )
}
</code></pre>

<hr>
<h2 id='set_is_mhcnuggets_installed'>Set the MHCnuggets installation state to the desired one</h2><span id='topic+set_is_mhcnuggets_installed'></span>

<h3>Description</h3>

<p>Set the MHCnuggets installation state to the desired one
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_is_mhcnuggets_installed(
  is_installed,
  verbose = FALSE,
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_is_mhcnuggets_installed_+3A_is_installed">is_installed</code></td>
<td>
<p>the desired installation state. Must
be <a href="base.html#topic+TRUE">TRUE</a> or <a href="base.html#topic+FALSE">FALSE</a></p>
</td></tr>
<tr><td><code id="set_is_mhcnuggets_installed_+3A_verbose">verbose</code></td>
<td>
<p>set to <a href="base.html#topic+TRUE">TRUE</a> for more debug information</p>
</td></tr>
<tr><td><code id="set_is_mhcnuggets_installed_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="set_is_mhcnuggets_installed_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='set_pip_version'>Set the version of pip.</h2><span id='topic+set_pip_version'></span>

<h3>Description</h3>

<p>Set the version of pip to a specific version,
by installing that version.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>set_pip_version(version)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="set_pip_version_+3A_version">version</code></td>
<td>
<p>pip version</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+upgrade_pip">upgrade_pip</a> to set pip to the latest version.
Use <a href="#topic+downgrade_pip">downgrade_pip</a> to set pip to a specific earlier version.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  if (is_pip_installed()) {
    set_pip_version("19.0")
  }

## End(Not run)
</code></pre>

<hr>
<h2 id='to_mhcnuggets_name'>Convert a standard haplotype name to the MHCnuggets name</h2><span id='topic+to_mhcnuggets_name'></span>

<h3>Description</h3>

<p>Convert a standard haplotype name to the MHCnuggets name.
Will <a href="base.html#topic+stop">stop</a> if this conversion fails.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_mhcnuggets_name(mhc)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="to_mhcnuggets_name_+3A_mhc">mhc</code></td>
<td>
<p>the MHC haplotype name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the MHCnuggets name for the haplotype
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>to_mhcnuggets_name("HLA-A*01:01")
</code></pre>

<hr>
<h2 id='to_mhcnuggets_names'>Convert one or more standard haplotype name to the MHCnuggets names</h2><span id='topic+to_mhcnuggets_names'></span>

<h3>Description</h3>

<p>Convert one or more standard haplotype names to the MHCnuggets names.
Will <a href="base.html#topic+stop">stop</a> if this conversion fails.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>to_mhcnuggets_names(mhcs)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="to_mhcnuggets_names_+3A_mhcs">mhcs</code></td>
<td>
<p>the MHC haplotype names</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the MHCnuggets names for the haplotypes
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>to_mhcnuggets_names("HLA-A*01:01")
</code></pre>

<hr>
<h2 id='uninstall_mhcnuggets'>Uninstall the MHCnuggets Python package.</h2><span id='topic+uninstall_mhcnuggets'></span>

<h3>Description</h3>

<p>Uninstall the MHCnuggets Python package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uninstall_mhcnuggets(
  folder_name = get_default_mhcnuggets_folder(),
  mhcnuggets_url = get_mhcnuggets_url()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="uninstall_mhcnuggets_+3A_folder_name">folder_name</code></td>
<td>
<p>superfolder of MHCnuggets.
The name of the superfolder is <code>/home/[user_name]/.local/share</code>
by default, as can be obtained by<a href="#topic+get_default_mhcnuggets_folder">get_default_mhcnuggets_folder</a></p>
</td></tr>
<tr><td><code id="uninstall_mhcnuggets_+3A_mhcnuggets_url">mhcnuggets_url</code></td>
<td>
<p>URL to the MHCnuggets GitHub repository</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>

<hr>
<h2 id='uninstall_pip'>Install pip.</h2><span id='topic+uninstall_pip'></span>

<h3>Description</h3>

<p>Install pip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>uninstall_pip()
</code></pre>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  uninstall_pip()

## End(Not run)
</code></pre>

<hr>
<h2 id='upgrade_pip'>Uograde pip.</h2><span id='topic+upgrade_pip'></span>

<h3>Description</h3>

<p>Uograde pip.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upgrade_pip()
</code></pre>


<h3>Value</h3>

<p>Nothing
</p>


<h3>Author(s)</h3>

<p>Richèl J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>use <a href="#topic+downgrade_pip">downgrade_pip</a> to set pip to an earlier version.
Use <a href="#topic+set_pip_version">set_pip_version</a> to install a specific version of pip
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
  install_pip()

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
