<!DOCTYPE html><html lang="en"><head><title>Help for package MHCtools</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MHCtools}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#BootKmeans'><p>BootKmeans() function</p></a></li>
<li><a href='#ClusterMatch'><p>ClusterMatch() function</p></a></li>
<li><a href='#CreateFas'><p>CreateFas() function</p></a></li>
<li><a href='#CreateHpltOccTable'><p>CreateHpltOccTable() function</p></a></li>
<li><a href='#CreateSamplesFas'><p>CreateSamplesFas() function</p></a></li>
<li><a href='#DistCalc'><p>DistCalc() function</p></a></li>
<li><a href='#GetHpltStats'><p>GetHpltStats() function</p></a></li>
<li><a href='#GetHpltTable'><p>GetHpltTable() function</p></a></li>
<li><a href='#GetReplStats'><p>GetReplStats function</p></a></li>
<li><a href='#GetReplTable'><p>GetReplTable function</p></a></li>
<li><a href='#hplt_occurrence_matrix'><p>Data hplt_occurrence_matrix</p></a></li>
<li><a href='#HpltFind'><p>HpltFind() function</p></a></li>
<li><a href='#HpltMatch'><p>HpltMatch() function</p></a></li>
<li><a href='#k_summary_table'><p>k_summary_table.rda</p></a></li>
<li><a href='#nest_table'><p>Data nest_table</p></a></li>
<li><a href='#NestTablesXL'><p>NestTablesXL() function</p></a></li>
<li><a href='#PapaDiv'><p>PapaDiv() function</p></a></li>
<li><a href='#parents_table'><p>Data parents_table</p></a></li>
<li><a href='#replicates_table'><p>Data replicates_table</p></a></li>
<li><a href='#ReplMatch'><p>ReplMatch() function</p></a></li>
<li><a href='#sequence_table'><p>Data sequence_table</p></a></li>
<li><a href='#sequence_table_fas'><p>Data sequence_table_fas</p></a></li>
<li><a href='#sequence_table_hplt'><p>Data sequence_table_hplt</p></a></li>
<li><a href='#sequence_table_repl'><p>Data sequence_table_repl</p></a></li>
<li><a href='#z1_matrix'><p>z1_matrix.rda</p></a></li>
<li><a href='#z2_matrix'><p>z2_matrix.rda</p></a></li>
<li><a href='#z3_matrix'><p>z3_matrix.rda</p></a></li>
<li><a href='#z4_matrix'><p>z4_matrix.rda</p></a></li>
<li><a href='#z5_matrix'><p>z5_matrix.rda</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Analysis of MHC Data in Non-Model Species</td>
</tr>
<tr>
<td>Version:</td>
<td>1.5.5</td>
</tr>
<tr>
<td>Description:</td>
<td>Fifteen tools for bioinformatics processing and analysis of major 
    histocompatibility complex (MHC) data. The functions are tailored for amplicon data 
    sets that have been filtered using the dada2 method (for more information on 
    dada2, visit <a href="https://benjjneb.github.io/dada2/">https://benjjneb.github.io/dada2/</a> ), but even other types of data 
    sets can be analyzed.
    The ReplMatch() function matches replicates in data sets in order to evaluate 
    genotyping success.
    The GetReplTable() and GetReplStats() functions perform such an evaluation.
    The CreateFas() function creates a fasta file with all the sequences in the data 
    set.
    The CreateSamplesFas() function creates individual fasta files for each sample in 
    the data set.
    The DistCalc() function calculates Grantham, Sandberg, or p-distances from pairwise 
    comparisons of all sequences in a data set, and mean distances of all pairwise 
    comparisons within each sample in a data set. The function additionally outputs five 
    tables with physico-chemical z-descriptor values (based on Sandberg et al. 1998) for 
    each amino acid position in all sequences in the data set. These tables may be useful 
    for further downstream analyses, such as estimation of MHC supertypes.
    The BootKmeans() function is a wrapper for the kmeans() function of the 'stats'
    package, which allows for bootstrapping. Bootstrapping k-estimates may be
    desirable in data sets, where e.g. BIC- vs. k-values do not produce clear
    inflection points ("elbows"). BootKmeans() performs multiple runs of kmeans() and 
    estimates optimal k-values based on a user-defined threshold of BIC reduction. The 
    method is an automated and bootstrapped version of visually inspecting elbow plots 
    of BIC- vs. k-values.
    The ClusterMatch() function is a tool for evaluating whether different k-means() 
    clustering models identify similar clusters, and summarize bootstrap model stats as 
    means for different estimated values of k. It is designed to take files produced by 
    the BootKmeans() function as input, but other data can be analyzed if the 
    descriptions of the required data formats are observed carefully.
    The PapaDiv() function compares parent pairs in the data set and calculate their 
    joint MHC diversity, taking into account sequence variants that occur in both 
    parents.
    The HpltFind() function infers putative haplotypes from families in the data 
    set. 
    The GetHpltTable() and GetHpltStats() functions evaluate the accuracy of 
    the haplotype inference.
    The CreateHpltOccTable() function creates a binary (logical) haplotype-sequence 
    occurrence matrix from the output of HpltFind(), for easy overview of which 
    sequences are present in which haplotypes. 
    The HpltMatch() function compares haplotypes to help identify overlapping and 
    potentially identical types.
    The NestTablesXL() function translates the output from HpltFind() to an Excel 
    workbook, that provides a convenient overview for evaluation and curating of the 
    inferred putative haplotypes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, utils, mgcv, grDevices, graphics, openxlsx</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2025-01-17 15:50:21 UTC; jacobroved</td>
</tr>
<tr>
<td>Author:</td>
<td>Jacob Roved [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jacob Roved &lt;jacob.roved@biol.lu.se&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2025-01-18 11:00:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='BootKmeans'>BootKmeans() function</h2><span id='topic+BootKmeans'></span>

<h3>Description</h3>

<p><code><a href="#topic+BootKmeans">BootKmeans</a></code> is a wrapper for the kmeans() function of the 'stats'
package, which allows for bootstrapping. Bootstrapping k-estimates may be
desirable in data sets, where the BIC- vs. k-values do not produce clear
inflection points (&quot;elbows&quot;).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>BootKmeans(
  z1_matrix,
  z2_matrix,
  z3_matrix,
  z4_matrix,
  z5_matrix,
  threshold = 0.01,
  no_scans = 1000,
  max_k = 40,
  iter.max = 1e+06,
  nstart = 200,
  algorithm = "Hartigan-Wong",
  path_out = path_out
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="BootKmeans_+3A_z1_matrix">z1_matrix</code></td>
<td>
<p>a matrix with numerical values of the first z-descriptor for
each amino acid position in all sequences in the data set.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_z2_matrix">z2_matrix</code></td>
<td>
<p>a matrix with numerical values of the second z-descriptor
for each amino acid position in all sequences in the data set.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_z3_matrix">z3_matrix</code></td>
<td>
<p>a matrix with numerical values of the third z-descriptor for
each amino acid position in all sequences in the data set.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_z4_matrix">z4_matrix</code></td>
<td>
<p>a matrix with numerical values of the fourth z-descriptor
for each amino acid position in all sequences in the data set.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_z5_matrix">z5_matrix</code></td>
<td>
<p>a matrix with numerical values of the fifth z-descriptor for
each amino acid position in all sequences in the data set.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_threshold">threshold</code></td>
<td>
<p>a numerical value between 0 and 1 specifying the threshold
of reduction in BIC for selecting a k estimate for each kmeans clustering
model. The value specifies a proportion of the max observed reduction in BIC
when increasing k by 1 (default 0.01).</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_no_scans">no_scans</code></td>
<td>
<p>an integer specifying the number of k estimation scans
to run (default 1,000).</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_max_k">max_k</code></td>
<td>
<p>an integer specifying the hypothetical maximum number of clusters
to detect (default 40). In each k estimation scan, the algorithm runs a
kmeans() clustering model for each value of k between 1 and max_k.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_iter.max">iter.max</code></td>
<td>
<p>an integer specifying the maximum number of iterations allowed
in each kmeans() clustering model (default 1,000,000).</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_nstart">nstart</code></td>
<td>
<p>an integer specifying the number of rows in the set of input
matrices that will be chosen as initial centers in the kmeans() clustering
models (default 200).</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_algorithm">algorithm</code></td>
<td>
<p>character vector, specifying the method for the kmeans()
clustering function, one of c(&quot;Hartigan-Wong&quot;, &quot;Lloyd&quot;, &quot;Forgy&quot;, &quot;MacQueen&quot;),
default is &quot;Hartigan-Wong&quot;.</p>
</td></tr>
<tr><td><code id="BootKmeans_+3A_path_out">path_out</code></td>
<td>
<p>a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>BootKmeans() performs multiple runs of kmeans() scanning k-values from 1 to a
maximum value defined by the user. In each scan, an optimal k-value is
estimated using a user-defined threshold of BIC reduction. The method is an
automated version of visually inspecting elbow plots of BIC- vs. k-values.
The number of scans to be performed is defined by the user.
</p>
<p>For each k-estimate scan, the algorithm produces a summary of the stats incl.
total within SS, AIC, and BIC, an elbow plot (BIC vs. k), and a set of cluster
files corresponding to the estimated optimal k-value. It also produces a table
summarizing the stats of the final selected kmeans() models corresponding to
the estimated optimal k-values of each scan.
</p>
<p>After running BootKmeans() on a data set, it is recommended to subsequently
evaluate the repeatability of the bootstrapped k-estimation scans with the
ClusterMatch() function also included in MHCtools.
</p>
<p>Input data format:
A set of five z-matrices containing numerical values of the z-descriptors
(z1-z5) for each amino acid position in a sequence alignment. Each column
should represent an amino acid position and each row one sequence in the
alignment.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>The function produces three folders in path_out, which contain for
each scan the estimated k-clusters saved as .RData files, an elbow plot saved
as .pdf, and a stats summary table saved as a .csv file. In path_out a summary
of all scans performed in the bootstrap run is also saved as .csv. This table
is also shown in the console.
Should alternative elbow plots be desired, they may be produced manually with
the stats presented in the summary tables for each scan.
</p>


<h3>Note</h3>

<p>If z-matrices were generated with the DistCalc() function, please make sure
to load the z-matrices from the .csv files exported by DistCalc(). Calling e.g.
'z1_matrix' without loading the exported tables will engage the default test
data set in MHCtools.
</p>
<p>Setting max_k too high can cause kmeans to fail with the error &quot;more cluster
centers than distinct data points&quot; - this problem can be solved by reducing
max_k.
</p>
<p>AIC and BIC are calculated from the kmeans model objects by the following
formulae:
- AIC = D + 2*m*k
- BIC = D + log(n)*m*k
in which:
- m = ncol(fit$centers)
- n = length(fit$cluster)
- k = nrow(fit$centers)
- D = fit$tot.withinss
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ClusterMatch">ClusterMatch</a></code>; <code><a href="#topic+DistCalc">DistCalc</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>z1_matrix &lt;- z1_matrix
z2_matrix &lt;- z2_matrix
z3_matrix &lt;- z3_matrix
z4_matrix &lt;- z4_matrix
z5_matrix &lt;- z5_matrix
path_out &lt;- tempdir()
BootKmeans(z1_matrix, z2_matrix, z3_matrix, z4_matrix, z5_matrix, threshold=0.01,
no_scans=10, max_k=20, iter.max=10, nstart=10, algorithm="Hartigan-Wong",
path_out=path_out)
</code></pre>

<hr>
<h2 id='ClusterMatch'>ClusterMatch() function</h2><span id='topic+ClusterMatch'></span>

<h3>Description</h3>

<p><code><a href="#topic+ClusterMatch">ClusterMatch</a></code> is a tool for evaluating whether k-means()
clustering models with similar estimated values of k identify similar
clusters. ClusterMatch() also summarizes model stats as means for
different estimated values of k. It is designed to take files produced
by the BootKmeans() function as input, but other data can be analyzed
if the descriptions of the data formats given below are observed
carefully.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ClusterMatch(filepath, path_out, k_summary_table)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ClusterMatch_+3A_filepath">filepath</code></td>
<td>
<p>a user defined path to a folder that contains the set of
K-cluster files to be matched against each other. The algorithm will attempt
to load all files in the folder, so it should contain only the relevant
K-cluster files. If the clusters were generated using the BootKmeans()
function, such a folder (named Clusters) was created by the algorithm in the
output path given by the user.
Each K-cluster file should correspond to the model$cluster object in kmeans()
saved as a .RData file. Such files are generated as part of the output from
BootKmeans(). ClusterMatch() assumes that the file names contain the string
&quot;model_&quot; followed by a model number, which must match the corresponding row
numbers in k_summary_table. If the data used was generated with the
BootKmeans() function, the formats and numbers will match by default.</p>
</td></tr>
<tr><td><code id="ClusterMatch_+3A_path_out">path_out</code></td>
<td>
<p>a user defined path to the folder where the output files will
be saved.</p>
</td></tr>
<tr><td><code id="ClusterMatch_+3A_k_summary_table">k_summary_table</code></td>
<td>
<p>a data frame summarizing the stats of the kmeans()
models that produced the clusters in the K-cluster files. If the data used
was generated with the BootKmeans() function, a compatible
k_summary_table was produced in the output path with the file name
&quot;k_means_bootstrap_summary_stats_&lt;date&gt;.csv&quot;.
If other data is analyzed, please observe these formatting requirements:
The k_summary_table must contain the data for each kmeans() model in rows
and as minimum the following columns:
- k-value (colname: k.est)
- residual total within sums-of-squares (colname: Tot.withinss.resid)
- residual AIC (colname: AIC.resid)
- residual BIC (colname: BIC.resid)
- delta BIC/max BIC (colname: prop.delta.BIC)
- delta BIC/k.est (colname: delta.BIC.over.k)
It is crucial that the models have the same numbers in the K-cluster file
names and in the k_summary_table, and that the rows of the table are ordered
by the model number.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>The function returns a summary table, which for each estimated number
of clusters (i.e. the k-values of the models) lists:
- number of models that found i clusters
- mean residual total within sums-of-squares
- mean residual AIC
- mean residual BIC
- mean delta BIC/max BIC
- mean delta BIC/k
- mean number of allele assignments that fall outside of the i most abundant
clusters across all pairwise comparisons between the models that found i
clusters
- mean proportion of allele assignments that fall outside of the i most
abundant clusters across all pairwise comparisons between the models that
found i clusters
The summary table is also saved as a .csv file in the output path.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+BootKmeans">BootKmeans</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filepath &lt;- system.file("extdata/ClusterMatch", package="MHCtools")
path_out &lt;- tempdir()
k_summary_table &lt;- k_summary_table
ClusterMatch(filepath, path_out, k_summary_table)
</code></pre>

<hr>
<h2 id='CreateFas'>CreateFas() function</h2><span id='topic+CreateFas'></span>

<h3>Description</h3>

<p><code><a href="#topic+CreateFas">CreateFas</a></code> creates a FASTA file with all the sequences in a
'dada2' sequence table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateFas(seq_table, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateFas_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td></tr>
<tr><td><code id="CreateFas_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A FASTA file with all the sequences in a 'dada2' sequence table. The
sequences are named in the FASTA file by an index number corresponding to
their column number in the sequence table.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreateSamplesFas">CreateSamplesFas</a></code>; for more information about 'dada2'
visit &lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq_table &lt;- sequence_table_fas
path_out &lt;- tempdir()
CreateFas(seq_table, path_out)
</code></pre>

<hr>
<h2 id='CreateHpltOccTable'>CreateHpltOccTable() function</h2><span id='topic+CreateHpltOccTable'></span>

<h3>Description</h3>

<p><code><a href="#topic+CreateHpltOccTable">CreateHpltOccTable</a></code> is designed to create a haplotype-sequence
occurrence matrix from the set of R lists with putative haplotypes output
by the HpltFind() function. CreateHpltOccTable() assumes that data originated
from a diploid species.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateHpltOccTable(seq_table, filepath, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateHpltOccTable_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td></tr>
<tr><td><code id="CreateHpltOccTable_+3A_filepath">filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the HpltFind() function have been saved.</p>
</td></tr>
<tr><td><code id="CreateHpltOccTable_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A binary (logical) occurrence matrix with the data set sequences
(inherited from seq_table) in columns and the putative haplotypes inferred
by the HpltFind() function in rows.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HpltFind">HpltFind</a></code>; for more information about 'dada2' visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq_table &lt;- sequence_table
filepath &lt;- system.file("extdata/HpltFindOut/", package="MHCtools")
path_out &lt;- tempdir()
CreateHpltOccTable(seq_table, filepath, path_out)
</code></pre>

<hr>
<h2 id='CreateSamplesFas'>CreateSamplesFas() function</h2><span id='topic+CreateSamplesFas'></span>

<h3>Description</h3>

<p><code><a href="#topic+CreateSamplesFas">CreateSamplesFas</a></code> creates a set of FASTA files with the
sequences present in each sample in a 'dada2' sequence table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CreateSamplesFas(seq_table, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="CreateSamplesFas_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td></tr>
<tr><td><code id="CreateSamplesFas_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A set of FASTA files with the sequences present in each sample in
the sequence table.The sequences are named in the FASTA files by an index
number corresponding to their column number in the sequence table, thus
identical sequences will have identical sample names in all the FASTA
files.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+CreateFas">CreateFas</a></code>; for more information about 'dada2' visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq_table &lt;- sequence_table_fas
path_out &lt;- tempdir()
CreateSamplesFas(seq_table, path_out)
</code></pre>

<hr>
<h2 id='DistCalc'>DistCalc() function</h2><span id='topic+DistCalc'></span>

<h3>Description</h3>

<p><code><a href="#topic+DistCalc">DistCalc</a></code> calculates Grantham distances, Sandberg distances,
or p-distances from pairwise comparisons of aligned sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistCalc(
  seq_file,
  path_out,
  input_fasta = NULL,
  input_seq = "aa",
  aa_dist = NULL,
  codon_pos = NULL,
  dist_type = "G"
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DistCalc_+3A_seq_file">seq_file</code></td>
<td>
<p>is a sequence occurrence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns. Optionally, a fasta file can be supplied as input in the format
rendered by read.fasta() from the package 'seqinr'.</p>
</td></tr>
<tr><td><code id="DistCalc_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
<tr><td><code id="DistCalc_+3A_input_fasta">input_fasta</code></td>
<td>
<p>optional, a logical (TRUE/FALSE) that indicates whether
the input file is a fasta file (TRUE) or a 'dada2'-style sequence table
(NULL/FALSE). The default is NULL/FALSE.</p>
</td></tr>
<tr><td><code id="DistCalc_+3A_input_seq">input_seq</code></td>
<td>
<p>defines the type of sequences in seq_file. It may take the
values 'nucl' or 'aa'.</p>
</td></tr>
<tr><td><code id="DistCalc_+3A_aa_dist">aa_dist</code></td>
<td>
<p>is optional, a logical (TRUE/FALSE) that determines whether
nucleotide sequences should be translated to amino acid sequences before
distance calculation, default is NULL/FALSE. Note that aa_dist must be set
to TRUE, if Grantham or Sandberg distances are calculated from an alignment
of nucleotide sequences.</p>
</td></tr>
<tr><td><code id="DistCalc_+3A_codon_pos">codon_pos</code></td>
<td>
<p>is optional, a vector of comma separated integers specifying
which codons to include in distance calculations. If omitted, distance
calculations are made using all codons. Note: When calculating nucleotide
P-distances, codon_pos should be specified as a vector of nucleotide
positions.</p>
</td></tr>
<tr><td><code id="DistCalc_+3A_dist_type">dist_type</code></td>
<td>
<p>is used to specify which kind of distances that are
calculated. It takes the values 'G' for Grantham distances, 'S'  for
Sandberg distances, or 'P' for p-distances. The argument is optional with
'G' as default setting.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The DistCalc() function takes a fasta file or a 'dada2'-style sequence
occurrence table (with aligned sequences as column names and samples in
rows) as input and produces a matrix with pairwise distances for all
sequences in the data set. If calculation of Sandberg distances is specified,
the function additionally outputs five tables with physico-chemical
z-descriptor values (based on Sandberg et al. 1998) for each amino acid
position in all sequences in the data set. These tables may be useful for
further downstream analyses, such as estimation of MHC supertypes. If a
sequence occurrence table is provided as input, the DistCalc() function
furthermore produces a table with the mean distances from all pairwise
comparisons of the sequences in each sample in the data set. (Note:
The mean distance will be NA for samples that have 0 or 1 sequence(s).)
</p>
<p>Grantham distances and Sandberg distances are calculated as described in
Pierini &amp; Lenz 2018. The Grantham distances produced by DistCalc() are
simply the mean Grantham distances (Grantham 1974) between all amino acid
codons in sequence pairs. When calculating Sandberg distances, DistCalc()
first computes Euclidian distances between all amino acid pairs based on
the five physico-chemical z-descriptors defined in Sandberg et al. 1998.
Sandberg distances are then calculated as the mean Euclidian distances
between all amino acid codons in sequence pairs. P-distances calculated
by DistCalc() are simply the proportion of varying codons between pairs
of sequences.
</p>
<p>The DistCalc() function includes an option for the user to specify which
codons to compare, which is useful e.g. if conducting the analysis only
on codons involved in specific functions, such as peptide binding of an MHC
molecule. Note: When calculating nucleotide P-distances, codon_pos is applied
directly on the nucleotide sequences. This allows the user to calculate
divergence in e.g. first, second, or third codon positions. Hence, codon_pos
should be specified as a vector of nucleotide positions when calculating
nucleotide P-distances.
</p>
<p>DistCalc() also accepts calculating amino acid distances directly from
protein-coding DNA sequences using the standard genetic code.
</p>
<p>The DistCalc() function accepts the following characters in the sequences:
Nucleotide sequences: A,T,G,C
Amino acid sequences: A,R,N,D,C,Q,E,G,H,I,L,K,M,F,P,S,T,W,Y,V
</p>
<p>It accepts gaps defined by '-'. Nucleotide triplets containing gaps are
translated to 'X', if amino acid distances are calculated directly from DNA
nucleotide sequences. Please note that '-' or 'X' are treated as unique
characters in p-distance calculations. The function will not accept 'X' or
gaps in Grantham or Sandberg distance calculations. If you wish to exclude
codons with 'X' or gaps from distance calculations, please use the
codon_pos option to specify which codons to compare.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>
<p>If you calculated Grantham or Sandberg distances, please additionally cite
either of the following references:
Grantham R. (1974). Amino acid difference formula to help explain protein
evolution. Science 185:862-864.
Sandberg M, Eriksson L, Jonsson J, Sjostrom M, Wold S. (1998). New chemical
descriptors relevant for the design of biologically active peptides. A
multivariate characterization of 87 amino acids. JMed Chem. 41(14):2481-2491.
</p>


<h3>Value</h3>

<p>The function returns a matrix with distances from all pairwise sequence
comparisons, where n is the number of sequences. If a sequence occurrence
table is given as input file, the function additionally returns a table with
the mean distance for each sample in the data set. If a sequence occurrence
table is given as input file, the sequences are named in the output matrix by
an index number that corresponds to their column number in the input file. If
calculation of Sandberg distances is specified, the function additionally
outputs five tables with physico-chemical z-descriptor values for each amino
acid position in all sequences in the data set. All output tables are saved
as .csv files in the output path.
</p>


<h3>See Also</h3>

<p>For more information about 'dada2', visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>seq_file &lt;- sequence_table_fas
path_out &lt;- tempdir()
DistCalc(seq_file, path_out, input_fasta=NULL, input_seq="nucl", aa_dist=NULL,
codon_pos=c(1,2,3,4,5,6,7,8), dist_type="P")
</code></pre>

<hr>
<h2 id='GetHpltStats'>GetHpltStats() function</h2><span id='topic+GetHpltStats'></span>

<h3>Description</h3>

<p><code><a href="#topic+GetHpltStats">GetHpltStats</a></code> uses the output files produced by the HpltFind()
function to calculate the mean of the mean proportion of incongruent
sequences across all nests in the data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetHpltStats(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetHpltStats_+3A_filepath">filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the HpltFind() function have been saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A mean of the mean proportion of incongruent sequences for each
nest.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HpltFind">HpltFind</a></code>; <code><a href="#topic+GetHpltTable">GetHpltTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filepath &lt;- system.file("extdata/HpltFindOut/", package="MHCtools")
GetHpltStats(filepath)
</code></pre>

<hr>
<h2 id='GetHpltTable'>GetHpltTable() function</h2><span id='topic+GetHpltTable'></span>

<h3>Description</h3>

<p><code><a href="#topic+GetHpltTable">GetHpltTable</a></code> uses the output files produced by the HpltFind()
function to produce a table with the mean proportion of incongruent sequences
for each nest. If the mean proportion of incongruent sequences is generally
low, but certain nests have many incongruent sequences, biological reasons
may be causing the mismatches, e.g. extra-pair fertilizations or
recombination events.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetHpltTable(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetHpltTable_+3A_filepath">filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the HpltFind() function have been saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A table with the mean proportion of incongruent sequences for each
nest.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HpltFind">HpltFind</a></code>; <code><a href="#topic+GetHpltStats">GetHpltStats</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filepath &lt;- system.file("extdata/HpltFindOut/", package="MHCtools")
GetHpltTable(filepath)
</code></pre>

<hr>
<h2 id='GetReplStats'>GetReplStats function</h2><span id='topic+GetReplStats'></span>

<h3>Description</h3>

<p><code><a href="#topic+GetReplStats">GetReplStats</a></code> uses the output files produced by the ReplMatch()
function to calculate statistics on the agreement between replicated samples
in the sequencing experiment.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetReplStats(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetReplStats_+3A_filepath">filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the ReplMatch() function have been saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A list containing the number of replicate sets with zero incongruent
sequences, the proportion of replicate sets with zero incongruent
sequences, the mean of the mean proportion of incongruent sequences across
all replicate sets, and the repeatability of the sequencing experiment.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReplMatch">ReplMatch</a></code>; <code><a href="#topic+GetReplTable">GetReplTable</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filepath &lt;- system.file("extdata/ReplMatchOut/", package="MHCtools")
GetReplStats(filepath)
</code></pre>

<hr>
<h2 id='GetReplTable'>GetReplTable function</h2><span id='topic+GetReplTable'></span>

<h3>Description</h3>

<p><code><a href="#topic+GetReplTable">GetReplTable</a></code> uses the output files produced by the ReplMatch()
function to produce a table with the replicate sets and their respective mean
proportion of incongruent sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GetReplTable(filepath)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GetReplTable_+3A_filepath">filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the ReplMatch() function have been saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A table with the mean proportion of incongruent sequences for each
replicate set.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ReplMatch">ReplMatch</a></code>; <code><a href="#topic+GetReplStats">GetReplStats</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>filepath &lt;- system.file("extdata/ReplMatchOut/", package="MHCtools")
GetReplTable(filepath)
</code></pre>

<hr>
<h2 id='hplt_occurrence_matrix'>Data hplt_occurrence_matrix</h2><span id='topic+hplt_occurrence_matrix'></span>

<h3>Description</h3>

<p>hplt_occurrence_matrix is an example of a binary occurrence matrix derived
from a randomized real major histocompatibility complex (MHC) data set. The
matrix was generated using the CreateHpltOccTable() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hplt_occurrence_matrix
</code></pre>


<h3>Format</h3>

<p>hplt_occurrence_matrix is a data frame with 136 putative haplotypes
in rows and 329 MHC sequence variants in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='HpltFind'>HpltFind() function</h2><span id='topic+HpltFind'></span>

<h3>Description</h3>

<p><code><a href="#topic+HpltFind">HpltFind</a></code> is designed to automatically infer major
histocompatibility complex (MHC) haplotypes from the genotypes of parents and
offspring in families (defined as nests) in non-model species, where MHC
sequence variants cannot be identified as belonging to individual loci.
HpltFind() assumes that data originated from a diploid species. The functions
GetHpltTable(), GetHpltStats(), and NestTablesXL() are designed to
evaluate the output files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HpltFind(nest_table, seq_table, alpha = 0.8, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HpltFind_+3A_nest_table">nest_table</code></td>
<td>
<p>is a table containing the sample names of parents and
offspring in each nest. This table should be organized so that the
individual names are in the first column (Sample_ID), and the nest number
is in the second column (Nest). For each nest, the first two rows should be
the parents, followed immediately by the offspring in the subsequent rows,
and then followed by the next nest, and so on. It is assumed that nests are
numbered consecutively beginning at 1.</p>
</td></tr>
<tr><td><code id="HpltFind_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td></tr>
<tr><td><code id="HpltFind_+3A_alpha">alpha</code></td>
<td>
<p>a numerical value between 0 and 1 (default 0.8) specifying a
threshold by which a set of sequences overlapping between a chick and a
parent will be assigned to the putative parental A haplotype or passed to
the B haplotype. Typical values are in the range 0.6-0.9. In data sets with
many different MHC alleles per individual (i.e. many MHC gene copies), alpha
may be set high. In data sets with fewer MHC alleles per individual, it
should be set lower. A range of alpha values may be tested to find the
optimal setting for a given data set, e.g. by evaluating the mean proportion
of incongruent sequences across the data set using GetHpltStats().</p>
</td></tr>
<tr><td><code id="HpltFind_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A set of R lists containing for each nest the putative haplotypes,
the names of sequences that could not be resolved with certainty in each
parent, the names of the sequences that were incongruent in the genotypes
of the nest, and the mean proportion of incongruent sequences (which is a
measure of the haplotype inference success and largely influenced by the
exactness of the genotyping experiment). The sequences are named in the
output by an index number corresponding to their column number in the
sequence table, thus identical sequences will have identical sample names
in all the output files. These files can be reopened in R e.g. using the
readRDS() function in the base package.
Note: HpltFind() will overwrite any existing files with the same output
file names in path_out.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GetHpltTable">GetHpltTable</a></code>; <code><a href="#topic+GetHpltStats">GetHpltStats</a></code>;
<code><a href="#topic+NestTablesXL">NestTablesXL</a></code>; <code><a href="#topic+CreateHpltOccTable">CreateHpltOccTable</a></code>; for more
information about 'dada2' visit &lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nest_table &lt;- nest_table
seq_table &lt;- sequence_table
path_out &lt;- tempdir()
HpltFind(nest_table, seq_table, alpha=0.8, path_out)
</code></pre>

<hr>
<h2 id='HpltMatch'>HpltMatch() function</h2><span id='topic+HpltMatch'></span>

<h3>Description</h3>

<p>Putative haplotypes may be identical to each other, or they may differ only by
incongruent or unresolved sequences. It is therefore useful to curate putative
haplotypes by comparing them to identify potentially overlapping types as
candidates for further investigation. <code><a href="#topic+HpltMatch">HpltMatch</a></code> calculates the
proportion of matching sequences between pairs of haplotypes and produces a
.csv table with values in a lower left matrix. If a threshold value is
specified, a list of haplotype matches where the proportion of matching
sequences exceeds the threshold will be produced.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HpltMatch(hplt_occ_matrix, path_out, threshold = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HpltMatch_+3A_hplt_occ_matrix">hplt_occ_matrix</code></td>
<td>
<p>A binary (logical) occurrence matrix with the data set
sequences in columns and the putative haplotypes in rows, as produced by the
CreateHpltOccTable() function.</p>
</td></tr>
<tr><td><code id="HpltMatch_+3A_path_out">path_out</code></td>
<td>
<p>a user defined path to the folder where the output file(s) will
be saved.</p>
</td></tr>
<tr><td><code id="HpltMatch_+3A_threshold">threshold</code></td>
<td>
<p>a numerical value between 0 and 1 (default NULL) specifying a
threshold for the proportion of matching sequences between haplotypes.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: The NestTablesXL() function provides a useful format for further
investigation of potentially overlapping haplotypes.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A table specifying the proportions of matching sequences between
pairs of haplotypes (in a lower left matrix). If a threshold value is
specified, a list of haplotype matches where the proportion of matching
sequences exceeds the threshold will be printed to the console. The list will
also be saved in the output path, and can be reopened in R e.g. using the
readRDS() function in the base package.
Note: HpltMatch() will overwrite any existing files with the same output
file names in path_out.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HpltFind">HpltFind</a></code>; <code><a href="#topic+CreateHpltOccTable">CreateHpltOccTable</a></code>;
<code><a href="#topic+NestTablesXL">NestTablesXL</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>hplt_occ_matrix &lt;- hplt_occurrence_matrix
path_out &lt;- tempdir()
HpltMatch(hplt_occ_matrix, path_out, threshold=NULL)
</code></pre>

<hr>
<h2 id='k_summary_table'>k_summary_table.rda</h2><span id='topic+k_summary_table'></span>

<h3>Description</h3>

<p>k_summary_table contains the results from a bootstrapped kmeans
clustering analysis performed on the test data in the tables
z1_matrix_test_data, z2_matrix_test_data, z3_matrix_test_data,
z4_matrix_test_data, and z5_matrix_test_data using BootKmeans().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>k_summary_table
</code></pre>


<h3>Format</h3>

<p>k_summary_table is a data frame with observations from
10 k-estimation scans in rows and their respective stats in 11 columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='nest_table'>Data nest_table</h2><span id='topic+nest_table'></span>

<h3>Description</h3>

<p>nest_table, parents_table, and sequence_table comprise a randomized dataset
derived from a real major histocompatibility complex (MHC) genotyping
experiment with data from parents and offspring. The nucleotide sequences
have been replaced with randomly generated sequences, and sample names have
been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>nest_table
</code></pre>


<h3>Format</h3>

<p>nest_table is a data frame with 213 samples in rows and 2 columns:
</p>
 <dl>
<dt>Sample_ID</dt><dd><p>Sample ID</p>
</dd> <dt>Nest</dt><dd><p>Nest index number</p>
</dd> </dl>



<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='NestTablesXL'>NestTablesXL() function</h2><span id='topic+NestTablesXL'></span>

<h3>Description</h3>

<p><code><a href="#topic+NestTablesXL">NestTablesXL</a></code> reads the R lists output by the HpltFind() function
and translates them to an Excel workbook for more convenient evaluation of the
inferred haplotypes and curation of unresolved and incongruent sequences. The
workbook contains separate tabs for each nest in the data set and provides an
overview of the genotypes of the samples in each nest and the inferred
haplotypes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>NestTablesXL(nest_table, seq_table, filepath, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="NestTablesXL_+3A_nest_table">nest_table</code></td>
<td>
<p>is a table containing the sample names of parents and
offspring in each nest. This table should be organized so that the
individual names are in the first column (Sample_ID), and the nest number
is in the second column (Nest). For each nest, the first two rows should be
the parents, followed immediately by the offspring in the subsequent rows,
and then followed by the next nest, and so on. It is assumed that nests are
numbered consecutively beginning at 1. Please use the same table as was used
to generate the haplotypes using HpltFind().</p>
</td></tr>
<tr><td><code id="NestTablesXL_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns. Please use the same table as was used to generate the haplotypes
using HpltFind().</p>
</td></tr>
<tr><td><code id="NestTablesXL_+3A_filepath">filepath</code></td>
<td>
<p>is a user defined path to the folder where the output files
from the HpltFind() function have been saved.</p>
</td></tr>
<tr><td><code id="NestTablesXL_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output file
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>An Excel workbook with individual tabs for each nest in nest_table.
Each tab contains a binary (logical) occurrence matrix with the samples from
each nest in columns and sequences (inherited from seq_table) in rows. The
order of the samples is derived from nest_table, with parents in the two
leftmost columns. Each tab also lists the putative haplotypes inferred by the
HpltFind() function and provides lists of unresolved sequences in haplotypes,
sequences with unidentified decent (i.e., present in parents but not in
offspring), sequences not assigned to haplotypes, and sequences with
unidentified origin (i.e., present in offspring but not in parents).
Note: NestTablesXL() will overwrite any existing file with the output file
name in path_out.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+HpltFind">HpltFind</a></code>; for more information about 'dada2' visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>nest_table &lt;- nest_table
seq_table &lt;- sequence_table_hplt
filepath &lt;- system.file("extdata/HpltFindOut/", package="MHCtools")
path_out &lt;- tempdir()
NestTablesXL(nest_table, seq_table, filepath, path_out)
</code></pre>

<hr>
<h2 id='PapaDiv'>PapaDiv() function</h2><span id='topic+PapaDiv'></span>

<h3>Description</h3>

<p><code><a href="#topic+PapaDiv">PapaDiv</a></code> calculates the joint major histocompatibility complex
(MHC) diversity in parent pairs, taking into account alleles that are shared
between the parents. The joint diversity in parent pairs is often of interest
in studies of mate choice, fitness, and heritability.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PapaDiv(parents_table, seq_table, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="PapaDiv_+3A_parents_table">parents_table</code></td>
<td>
<p>is a table containing the sample names of the parents in
each nest. This table should be organized so that each row represents one
nest, with the individual names of the mothers in the first column
(Mother), and the individual names of the fathers in the second column
(Father).</p>
</td></tr>
<tr><td><code id="PapaDiv_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td></tr>
<tr><td><code id="PapaDiv_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The PapaDiv() function outputs a set of R lists containing for the joint
diversity of each parent pair, the proportion of sequences that are shared
between the parents, the diversity of each of the parents, the observed sequence
variants in each parent, the matched sequence variants, and the incongruent
sequence variants in each parent.
</p>
<p>In addition, PapaDiv() produces a summary table with the names of the parents in
a pair, their respective MHC diversities, and the joint parent pair diversity.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>a set of R lists containing for the joint diversity of each parent
pair, the proportion of sequences that are shared between the parents, the
diversity of each of the parents, the observed sequence variants in each
parent, the matched sequence variants, and the incongruent sequence
variants in each parent. The sequences are named in the output by an index
number corresponding to their column number in the sequence table, thus
identical sequences will have identical sample names in all the output
files. These files are saved in a sub folder in the output path called
Parent_pairs (created by PapaDiv()) and can be reopened in R e.g. using
the readRDS() function in the base package. For downstream data analysis,
the PapaDiv() function also produces a summary table with the names of the
parents in a pair, their respective MHC diversities, and the joint parent
pair diversity. This table is saved as a .csv file in the output path.
</p>


<h3>See Also</h3>

<p>For more information about 'dada2' visit
&lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>parents_table &lt;- parents_table
seq_table &lt;- sequence_table
path_out &lt;- tempdir()
PapaDiv(parents_table, seq_table, path_out)
</code></pre>

<hr>
<h2 id='parents_table'>Data parents_table</h2><span id='topic+parents_table'></span>

<h3>Description</h3>

<p>nest_table, parents_table, and sequence_table comprise a randomized dataset
derived from a real major histocompatibility complex (MHC) genotyping
experiment with data from parents and offspring. The nucleotide sequences
have been replaced with randomly generated sequences, and sample names have
been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>parents_table
</code></pre>


<h3>Format</h3>

<p>parents_table is a data frame with 57 parent pairs in rows and 2
columns: </p>
 <dl>
<dt>Mother</dt><dd><p>Mother ID</p>
</dd> <dt>Father</dt><dd><p>Father ID</p>
</dd> </dl>



<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='replicates_table'>Data replicates_table</h2><span id='topic+replicates_table'></span>

<h3>Description</h3>

<p>replicates_table and sequence_table_repl comprise a randomized dataset
derived from a real major histocompatibility complex (MHC) genotyping
experiment with technical replicates. The nucleotide sequences have been
replaced with randomly generated sequences, and sample names have been
anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>replicates_table
</code></pre>


<h3>Format</h3>

<p>replicates_table is a data frame with 111 technical replicate samples
in rows and 2 columns: </p>
 <dl>
<dt>Sample_ID</dt><dd><p>Technical replicate
sample ID</p>
</dd> <dt>Replic_set</dt><dd><p>Index number of replicate set</p>
</dd> </dl>



<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='ReplMatch'>ReplMatch() function</h2><span id='topic+ReplMatch'></span>

<h3>Description</h3>

<p>In amplicon filtering it is sometimes valuable to compare technical
replicates in order to estimate the accuracy of a genotyping experiment. This
may be done both to optimize filtering settings and to estimate repeatability
to report in a publication. <code><a href="#topic+ReplMatch">ReplMatch</a></code> is designed to
automatically compare technical replicates in an amplicon filtering data set
and report the proportion of mismatches. The functions GetReplTable() and
GetReplStats() are designed to evaluate the output files.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ReplMatch(repl_table, seq_table, path_out)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ReplMatch_+3A_repl_table">repl_table</code></td>
<td>
<p>is a table containing the sample names of technical
replicates in the data set. This table should be organized so that the
individual names are in the first column (Sample_ID), and the index number
of the replicate set is in the second column (Replic_set). Replicate sets
may contain more than two replicates, but sets must be numbered consecutively
beginning at 1.</p>
</td></tr>
<tr><td><code id="ReplMatch_+3A_seq_table">seq_table</code></td>
<td>
<p>seq_table is a sequence table as output by the 'dada2'
pipeline, which has samples in rows and nucleotide sequence variants in
columns.</p>
</td></tr>
<tr><td><code id="ReplMatch_+3A_path_out">path_out</code></td>
<td>
<p>is a user defined path to the folder where the output files
will be saved.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note: ReplMatch() will throw a warning if all samples in a replicate set have
0 sequences. In that case, the mean_props for that replicate set and the
repeatability for the data set will be NaN, and ReplMatch() will report which
replicate set is problematic and suggest to remove it from the repl_table. If
removing replicate sets, beware that the replicate sets in repl_table must be
numbered consecutively beginning at 1.
</p>
<p>If you publish data or results produced with MHCtools, please cite both of
the following references:
Roved, J. (2022). MHCtools: Analysis of MHC data in non-model species. Cran.
Roved, J. (2024). MHCtools 1.5: Analysis of MHC sequencing data in R. In S.
Boegel (Ed.), HLA Typing: Methods and Protocols (2nd ed., pp. 275–295).
Humana Press. https://doi.org/10.1007/978-1-0716-3874-3_18
</p>


<h3>Value</h3>

<p>A set of R lists containing for each replicate set the observed
sequence variants, the names of the sequences that were incongruent in the
replicates, and the mean proportion of incongruent sequences (if 100
matches are expected between the replicates, this is equivalent of an error
rate in the sequencing process). The sequences are named in the output by
an index number corresponding to their column number in the sequence table,
thus identical sequences will have identical sample names in all the output
files. These files can be reopened in R e.g. using the readRDS() function
in the base package.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+GetReplTable">GetReplTable</a></code>; <code><a href="#topic+GetReplStats">GetReplStats</a></code>; for more
information about 'dada2' visit &lt;https://benjjneb.github.io/dada2/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>repl_table &lt;- replicates_table
seq_table &lt;- sequence_table_repl
path_out &lt;- tempdir()
ReplMatch(repl_table, seq_table, path_out)
</code></pre>

<hr>
<h2 id='sequence_table'>Data sequence_table</h2><span id='topic+sequence_table'></span>

<h3>Description</h3>

<p>nest_table, parents_table, and sequence_table comprise a randomized dataset
derived from a real major histocompatibility complex (MHC) genotyping
experiment with data from parents and offspring. The nucleotide sequences
have been replaced with randomly generated sequences, and sample names have
been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sequence_table
</code></pre>


<h3>Format</h3>

<p>sequence_table is a data frame with 334 samples in rows and 329 DNA
sequence variants in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='sequence_table_fas'>Data sequence_table_fas</h2><span id='topic+sequence_table_fas'></span>

<h3>Description</h3>

<p>sequence_table_fas is a randomized dataset derived from a real major
histocompatibility complex (MHC) genotyping experiment. The nucleotide
sequences have been replaced with randomly generated sequences, and sample
names have been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sequence_table_fas
</code></pre>


<h3>Format</h3>

<p>sequence_table_fas is a data frame with 100 samples in rows and 166
DNA sequence variants in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='sequence_table_hplt'>Data sequence_table_hplt</h2><span id='topic+sequence_table_hplt'></span>

<h3>Description</h3>

<p>sequence_table_hplt is a randomized test data set derived from a real major
histocompatibility complex (MHC) genotyping experiment. This table differs
from the sequence_table by having the nucleotide sequences replaced with
sequence names. Sample names have been anonymized from the real data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sequence_table_hplt
</code></pre>


<h3>Format</h3>

<p>sequence_table_hplt is a data frame with 334 samples in rows and 329
sequence variants in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='sequence_table_repl'>Data sequence_table_repl</h2><span id='topic+sequence_table_repl'></span>

<h3>Description</h3>

<p>replicates_table and sequence_table_repl comprise a randomized dataset
derived from a real major histocompatibility complex (MHC) genotyping
experiment with technical replicates. The nucleotide sequences have been
replaced with randomly generated sequences, and sample names have been
anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sequence_table_repl
</code></pre>


<h3>Format</h3>

<p>sequence_table_repl is a data frame with 412 samples in rows and 511
DNA sequence variants in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='z1_matrix'>z1_matrix.rda</h2><span id='topic+z1_matrix'></span>

<h3>Description</h3>

<p>z1_matrix comprise a randomized dataset derived from 70
nucleotide sequences of a real major histocompatibility complex (MHC)
genotyping experiment. z1-descriptor values have been extracted from a
subset of 8 amino acid codons and sequence names have been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z1_matrix
</code></pre>


<h3>Format</h3>

<p>z1_matrix is a data frame with 70 sequences in rows and
z1-descriptor variables for 8 sequence codons in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='z2_matrix'>z2_matrix.rda</h2><span id='topic+z2_matrix'></span>

<h3>Description</h3>

<p>z2_matrix comprise a randomized dataset derived from 70
nucleotide sequences of a real major histocompatibility complex (MHC)
genotyping experiment. z2-descriptor values have been extracted from a
subset of 8 amino acid codons and sequence names have been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z2_matrix
</code></pre>


<h3>Format</h3>

<p>z2_matrix is a data frame with 70 sequences in rows and
z2-descriptor variables for 8 sequence codons in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='z3_matrix'>z3_matrix.rda</h2><span id='topic+z3_matrix'></span>

<h3>Description</h3>

<p>z3_matrix comprise a randomized dataset derived from 70
nucleotide sequences of a real major histocompatibility complex (MHC)
genotyping experiment. z3-descriptor values have been extracted from a
subset of 8 amino acid codons and sequence names have been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z3_matrix
</code></pre>


<h3>Format</h3>

<p>z3_matrix is a data frame with 70 sequences in rows and
z3-descriptor variables for 8 sequence codons in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='z4_matrix'>z4_matrix.rda</h2><span id='topic+z4_matrix'></span>

<h3>Description</h3>

<p>z4_matrix comprise a randomized dataset derived from 70
nucleotide sequences of a real major histocompatibility complex (MHC)
genotyping experiment. z4-descriptor values have been extracted from a
subset of 8 amino acid codons and sequence names have been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z4_matrix
</code></pre>


<h3>Format</h3>

<p>z4_matrix is a data frame with 70 sequences in rows and
z4-descriptor variables for 8 sequence codons in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

<hr>
<h2 id='z5_matrix'>z5_matrix.rda</h2><span id='topic+z5_matrix'></span>

<h3>Description</h3>

<p>z5_matrix comprise a randomized dataset derived from 70
nucleotide sequences of a real major histocompatibility complex (MHC)
genotyping experiment. z5-descriptor values have been extracted from a
subset of 8 amino acid codons and sequence names have been anonymized.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>z5_matrix
</code></pre>


<h3>Format</h3>

<p>z5_matrix is a data frame with 70 sequences in rows and
z5-descriptor variables for 8 sequence codons in columns.
</p>


<h3>Source</h3>

<p>original data.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
