<!DOCTYPE html><html lang="en"><head><title>Help for package microbiomeMQC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {microbiomeMQC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#microbiomeMQC'><p>Calculate 4 key reporting measures</p></a></li>
<li><a href='#MQC_testdata'><p>MQC_testdata</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Calculate 4 Key Reporting Measures</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jacob Dehinsilu &lt;jacobdehinsilu@outlook.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Perform calculations for the WHO International Reference Reagents for the microbiome. Using strain, species or genera abundance tables generated through analysis of 16S ribosomal RNA sequencing or shotgun sequencing which included a reference reagent. This package will calculate measures of sensitivity, False positive relative abundance, diversity, and similarity based on mean average abundances with respect to the reference reagent.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Imports:</td>
<td>readxl, vegan</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-09-13 00:02:10 UTC; Jacob</td>
</tr>
<tr>
<td>Author:</td>
<td>Jacob Dehinsilu <a href="https://orcid.org/0000-0001-9041-970X"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Ravneet Bhuller [aut]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-09-16 07:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='microbiomeMQC'>Calculate 4 key reporting measures</h2><span id='topic+microbiomeMQC'></span>

<h3>Description</h3>

<p>Calculate the 4 key reporting measures for the WHO International Reference Reagents for the microbiome
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microbiomeMQC(input, taxonomic_level, output)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="microbiomeMQC_+3A_input">input</code></td>
<td>
<p>Path to the .xlsx or .csv file.e.g. &quot;C:\Users\joeblogs\Desktop\data.xlsx&quot;</p>
</td></tr>
<tr><td><code id="microbiomeMQC_+3A_taxonomic_level">taxonomic_level</code></td>
<td>
<p>The taxonomic level (&quot;strain&quot;, &quot;species&quot;, or &quot;genus&quot;) that you are using.</p>
</td></tr>
<tr><td><code id="microbiomeMQC_+3A_output">output</code></td>
<td>
<p>Path to save the output CSV file (file path must end with the file name e.g. &quot;C:\Users\joeblogs\Desktop\MQC.csv&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The 4 key reporting measures (using the example of species) are:
</p>
<p>Sensitivity: How many species from the reagent are correctly identified.
</p>
<p>Diversity: The total number of species detected.
</p>
<p>FPRA: The relative abundance of false positives.
</p>
<p>Similarity: The Bray-Curtis dissimilarity in composition between the species profile reported and the 'ground truth' profile of the reagent.
</p>
<p>To see how the data should be formatted do: data(MQC_testdata)
FP = false positive i.e. those bacteria that do not appear in the reagent
</p>


<h3>Value</h3>

<p>No return value. The function writes the results to the specified CSV file.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>input &lt;- system.file("extdata", "MQC_testdata.csv", package = "microbiomeMQC")
output &lt;- tempfile(fileext = ".csv")
microbiomeMQC(input, "species", output)
</code></pre>

<hr>
<h2 id='MQC_testdata'>MQC_testdata</h2><span id='topic+MQC_testdata'></span>

<h3>Description</h3>

<p>Data used for demonstrating the functionality of the microbiomeMQC package.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(MQC_testdata)
</code></pre>


<h3>Format</h3>

<p>An object of class <code>data.frame</code> with 26 rows and 6 columns:
</p>

<dl>
<dt>Column1</dt><dd><p>species</p>
</dd>
<dt>Column2</dt><dd><p>DNA_hilo_RR</p>
</dd>
<dt>Column3</dt><dd><p>perfect result</p>
</dd>
<dt>Column4</dt><dd><p>high FPRA example</p>
</dd>
<dt>Column5</dt><dd><p>high diversity example</p>
</dd>
<dt>Column6</dt><dd><p>low sensitivity example</p>
</dd>
</dl>



<h3>Source</h3>

<p>Generated by the package authors for demonstration purposes.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
