<!DOCTYPE html><html><head><title>Help for package microcontax</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {microcontax}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#contax.trim'><p>The ConTax data set</p></a></li>
<li><a href='#fullTaxonomy'><p>The full taxonomy of a genus</p></a></li>
<li><a href='#genusLookup'><p>Taxonomy lookup</p></a></li>
<li><a href='#getDomain'><p>Extractor functions for ConTax data</p></a></li>
<li><a href='#medoids'><p>The ConTax medoids</p></a></li>
<li><a href='#microcontax-package'><p>The ConTax data package</p></a></li>
<li><a href='#taxonomy.table'><p>Taxonomy look-up table</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>The ConTax Data Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-08-10</td>
</tr>
<tr>
<td>Author:</td>
<td>Hilde Vinje, Kristian Hovde Liland, Lars Snipen.</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lars Snipen &lt;lars.snipen@nmbu.no&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The consensus taxonomy for prokaryotes is a set of data-sets for
    best possible taxonomic classification based on 16S rRNA sequence data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>microseq</td>
</tr>
<tr>
<td>Suggests:</td>
<td>microcontax.data</td>
</tr>
<tr>
<td>Additional_repositories:</td>
<td><a href="https://khliland.github.io/drat/">https://khliland.github.io/drat/</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-11 13:19:00 UTC; larssn</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-11 14:50:06 UTC</td>
</tr>
</table>
<hr>
<h2 id='contax.trim'>The ConTax data set</h2><span id='topic+contax.trim'></span>

<h3>Description</h3>

<p>The trimmed version of the ConTax data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(contax.trim)
</code></pre>


<h3>Details</h3>

<p><code>contax.trim</code> is a <code>data.frame</code> object containing 38 781 full-length 16S rRNA 
sequences. It is the trimmed version of the full data set (see below). Large taxa (many sequences) have
been trimmed as described in Vinje et al. (2016) to obtain a data set with a more even representation of
the prokaryotic taxonomy.
</p>
<p>The <code>contax.full</code> is the full consensus taxonomy data set as described in Vinje et al. (2016). The data
set is too large for CRAN and thus available as a separate package <code>microcontax.data</code>. See example
below for how to obtain <code>contax.full</code>.
</p>
<p>The Header of every sequence starts with a unique tag, in this case the text &quot;ConTax&quot; and some integer.
This is followed by a token describing the origin of the sequence. It is typically
</p>
<p>&quot;Intersection=SRG&quot;
</p>
<p>meaning it is found in both the Silva, RDP and Greengenes data repository. Intersections can also be
SR, SG and RG if the sequence was found in two repositories only. The taxonomy information for each
sequence is found in the third token. It follows a commonly used format:
</p>
<p>&quot;k__&lt;...&gt;;p__&lt;...&gt;;c__&lt;...&gt;;o__&lt;...&gt;;f__&lt;...&gt;;g__&lt;...&gt;;&quot;
</p>
<p>where &lt;...&gt; is some proper text. The letters, followed by a double underscore, refer to the taxonomic levels
Domain (Kingdom), Phylum, Class, Order, Family and Genus.
Here is an example of a proper string:
</p>
<p>&quot;k__Bacteria;p__Firmicutes;c__Bacilli;o__Bacillales;f__Staphylococcaceae;g__Staphylococcus;&quot;
</p>
<p>As long as this format is used the taxonomy information can be extracted by the supplied 
extractor-functions <code><a href="#topic+getDomain">getDomain</a></code>, <code><a href="#topic+getPhylum">getPhylum</a></code>,...,<code><a href="#topic+getGenus">getGenus</a></code>.
</p>


<h3>Author(s)</h3>

<p>Hilde Vinje, Kristian Hovde Liland, Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+medoids">medoids</a></code>, <code><a href="#topic+getDomain">getDomain</a></code>, <code><a href="microcontax.data.html#topic+contax.full">contax.full</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(contax.trim)
dim(contax.trim)

# Write to FASTA-file
## Not run: 
writeFasta(contax.trim,out.file="ConTax_trim.fasta")

# Install microcontax.data with the BIG contax.full data set
if (!requireNamespace("microcontax.data", quietly = TRUE)) {
 install.packages("microcontax.data")
}
# Load data
data("contax.full", package = "microcontax.data")

## End(Not run)

</code></pre>

<hr>
<h2 id='fullTaxonomy'>The full taxonomy of a genus</h2><span id='topic+fullTaxonomy'></span>

<h3>Description</h3>

<p>Converts a genus to a string containing the full taxonomy.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fullTaxonomy(genera)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fullTaxonomy_+3A_genera">genera</code></td>
<td>
<p>A vector of texts, the genera names to look up.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The argument <code>genera</code> must consist of names in the <code>Genus</code> column of the data
set <code><a href="#topic+taxonomy.table">taxonomy.table</a></code>.
</p>
<p>&quot;k__&lt;...&gt;;p__&lt;...&gt;;c__&lt;...&gt;;o__&lt;...&gt;;f__&lt;...&gt;;g__&lt;...&gt;;&quot;
</p>
<p>where &lt;...&gt; is some proper text.
</p>


<h3>Value</h3>

<p>A character vector containing the taxonomy information.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+taxonomy.table">taxonomy.table</a></code>, <code><a href="#topic+genusLookup">genusLookup</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genera &lt;-  c("Bacillus","Clostridium","Hyphomonas") 
fullTaxonomy(genera)


</code></pre>

<hr>
<h2 id='genusLookup'>Taxonomy lookup</h2><span id='topic+genusLookup'></span>

<h3>Description</h3>

<p>Extracting taxonomic information from the <code><a href="#topic+taxonomy.table">taxonomy.table</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genusLookup(genera, rank = "Phylum")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genusLookup_+3A_genera">genera</code></td>
<td>
<p>A vector of texts, the genera names to look up.</p>
</td></tr>
<tr><td><code id="genusLookup_+3A_rank">rank</code></td>
<td>
<p>A single text, the level of the taxonomy to look up.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Function for looking up higher-level taxonomy of specified genera.
</p>
<p>The argument <code>genera</code> must consist of names in the <code>Genus</code> column of the data
set <code><a href="#topic+taxonomy.table">taxonomy.table</a></code>.
</p>


<h3>Value</h3>

<p>A character vector containing the taxonomy information. Names in <code>genera</code> not recognized will 
return <code>NA</code>. Please note that there are some cases of un-assigned taxonomy at some ranks 
(Class, Order or Family), this is returned as &quot;unknown&quot;.
</p>


<h3>Author(s)</h3>

<p>Hilde Vinje, Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+taxonomy.table">taxonomy.table</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>genus &lt;-  c("Acidilobus","Nitrosopumilus","Hyphomonas") 
genusLookup(genus, rank = "Phylum")
genusLookup(genus, rank = "Class")


</code></pre>

<hr>
<h2 id='getDomain'>Extractor functions for ConTax data</h2><span id='topic+getDomain'></span><span id='topic+getPhylum'></span><span id='topic+getClass'></span><span id='topic+getOrder'></span><span id='topic+getFamily'></span><span id='topic+getGenus'></span><span id='topic+getTag'></span><span id='topic+getTaxonomy'></span>

<h3>Description</h3>

<p>Extracting taxonomic information from ConTax data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>getDomain(header)
getPhylum(header)
getClass(header)
getOrder(header)
getFamily(header)
getGenus(header)
getTag(header)
getTaxonomy(header)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="getDomain_+3A_header">header</code></td>
<td>
<p>A vector of texts, typically the <code>Header</code> from a table,
containing taxonomy information in the proper format.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The ConTax data sets are tables in the FASTA format (see <code>readFasta</code>), 
where the <code>Header</code> column contains texts according to a strict format.
</p>
<p>The <code>header</code> always starts with a short text, a Tag, which is a unique identifier for every sequence.
The function <code>getTag</code> will extract this from the <code>header</code>.
</p>
<p>After the Tag follows one or more tokens. One of these tokens must be a string with the
following format:
</p>
<p>&quot;k__&lt;...&gt;;p__&lt;...&gt;;c__&lt;...&gt;;o__&lt;...&gt;;f__&lt;...&gt;;g__&lt;...&gt;;&quot;
</p>
<p>where &lt;...&gt; is some proper text. Here is an example of a proper string:
</p>
<p>&quot;k__Bacteria;p__Firmicutes;c__Bacilli;o__Bacillales;f__Staphylococcaceae;g__Staphylococcus;&quot;
</p>
<p>The functions <code>getDomain</code>, ..., <code>getGenus</code> extracts the
corresponding information from the <code>header</code>. <code>getTaxonomy</code>
combines all taxonomy extractors, combines these in a table
and imputes missing taxa with parent taxa.
</p>


<h3>Value</h3>

<p>A vector containing the sub-texts extracted from each <code>header</code> text, but 
<code>getTaxonomy</code> returns a table with the full taxonomy, one row for each input <code>header</code>
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+contax.trim">contax.trim</a></code>, <code><a href="#topic+medoids">medoids</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(contax.trim)
getTag(contax.trim$Header)
getGenus(contax.trim$Header)
getPhylum(contax.trim$Header)

</code></pre>

<hr>
<h2 id='medoids'>The ConTax medoids</h2><span id='topic+medoids'></span>

<h3>Description</h3>

<p>The genus medoids from the ConTax data set.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(medoids)
</code></pre>


<h3>Details</h3>

<p><code>medoids</code> is a <code>data.frame</code> object containing the medoide sequences for each genus in 
the ConTax data sets (both <code>contax.trim</code> and <code>contax.full</code>).
</p>
<p>The medoide sequence in a genus is the sequence having the smallest sum of distance to all other members
of the same genus. Thus, it is the sequence closest to the centre of the genus. The medoids can be used as
the representative of each genus, e.g. for building trees for the entire taxonomy.
</p>
<p>The taxonomy information for each sequence can be extracted from the <code>Header</code> column by the supplied
extractor-functions <code><a href="#topic+getDomain">getDomain</a></code>, <code><a href="#topic+getPhylum">getPhylum</a></code>,...,<code><a href="#topic+getGenus">getGenus</a></code>.
</p>


<h3>Author(s)</h3>

<p>Hilde Vinje, Kristian Hovde Liland, Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="microcontax.data.html#topic+contax.full">contax.full</a></code>, <code><a href="#topic+getDomain">getDomain</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(medoids)
summary(medoids)

</code></pre>

<hr>
<h2 id='microcontax-package'>The ConTax data package</h2><span id='topic+microcontax-package'></span><span id='topic+microcontax'></span>

<h3>Description</h3>

<p>The consensus taxonomy for prokaryotes is a package of data sets designed to be the best
possible for training taxonomic classifiers based on 16S rRNA sequence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microcontax()
</code></pre>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> microcontax</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.2</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2020-06-06</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Hilde Vinje, Kristian Liland, Lars Snipen.<br />
Maintainer: Lars Snipen &lt;lars.snipen@nmbu.no&gt;
</p>


<h3>See Also</h3>

<p><code><a href="microseq.html#topic+microseq-package">microseq</a></code>
</p>

<hr>
<h2 id='taxonomy.table'>Taxonomy look-up table</h2><span id='topic+taxonomy.table'></span>

<h3>Description</h3>

<p>A data frame consisting of the taxonomy information used in the ConTax data sets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(taxonomy.table)
</code></pre>


<h3>Details</h3>

<p><code>taxonomy.table</code> is a <code>data.frame</code> consisting of the seven columns Domain,
Phylum, Class, Order, Family, Genus and LPSN. The first six are taxonomy informations, the last
is &quot;Yes&quot; or &quot;No&quot; indiocating if the Genus listed is also found in the List of prokaryotic names
with standing in nomenclature (LPSN) database, see http://www.bacterio.net/.
</p>
<p>Each row contains the taxonomy information for a genus, hence the number of rows equals the number
of unique genera.
</p>
<p>To quickly look-up the higher rank taxonomy for a given genus, see the function <code><a href="#topic+genusLookup">genusLookup</a></code>.
</p>


<h3>Author(s)</h3>

<p>Hilde Vinje, Kristian Hovde Liland, Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+genusLookup">genusLookup</a></code>, <code><a href="microcontax.data.html#topic+contax.full">contax.full</a></code>,
<code><a href="#topic+contax.trim">contax.trim</a></code>, <code><a href="#topic+getDomain">getDomain</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(taxonomy.table)
dim(taxonomy.table)
taxonomy.table[1:10,]
genusLookup(taxonomy.table$Genus[1:10], rank = "Family")

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
