<!DOCTYPE html><html lang="en"><head><title>Help for package microhaplot</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {microhaplot}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mvShinyHaplot'><p>Transfer a copy of microhaplot app.</p></a></li>
<li><a href='#prepHaplotFiles'><p>Extracts haplotype from alignment reads.</p></a></li>
<li><a href='#runShinyHaplot'><p>Run shiny microhaplot</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Microhaplotype Constructor and Visualizer</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.1</td>
</tr>
<tr>
<td>Description:</td>
<td>A downstream bioinformatics tool to construct and assist 
    curation of microhaplotypes from short read sequences.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.2)</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>LazyData:</td>
<td>TRUE</td>
</tr>
<tr>
<td>Imports:</td>
<td>DT (&ge; 0.1), dplyr (&ge; 0.4.3), ggplot2 (&ge; 2.1.0), grid (&ge;
3.1.2), gtools (&ge; 3.5.0), magrittr (&ge; 1.5), scales (&ge;
0.4.0), shiny (&ge; 0.13.2), shinyBS (&ge; 0.61), tidyr (&ge; 0.4.1),
shinyWidgets (&ge; 0.4.3), ggiraph (&ge; 0.6.0)</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ngthomas/microhaplot">https://github.com/ngthomas/microhaplot</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/ngthomas/microhaplot/issues">https://github.com/ngthomas/microhaplot/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-09-28 00:53:53 UTC; thomasn</td>
</tr>
<tr>
<td>Author:</td>
<td>Thomas Ng [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Thomas Ng &lt;tngthomasng@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-10-03 07:40:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='mvShinyHaplot'>Transfer a copy of microhaplot app.</h2><span id='topic+mvShinyHaplot'></span>

<h3>Description</h3>

<p>Moves shiny microhaplot app to a different directory
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mvShinyHaplot(path)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mvShinyHaplot_+3A_path">path</code></td>
<td>
<p>string. directory path. Required</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a logical value of whether <code>file.copy</code> is successfully transferred Shiny app to its new directory
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
mvShinyHaplot(tempdir())

</code></pre>

<hr>
<h2 id='prepHaplotFiles'>Extracts haplotype from alignment reads.</h2><span id='topic+prepHaplotFiles'></span>

<h3>Description</h3>

<p>The function <code>microhaplot</code> extracts haplotype from sequence alignment files through perl script <code>hapture</code> and returns a summary table of the read depth and read quality associate with haplotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prepHaplotFiles(run.label, sam.path, label.path, vcf.path,
  out.path = tempdir(), add.filter = FALSE, app.path = tempdir(),
  n.jobs = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="prepHaplotFiles_+3A_run.label">run.label</code></td>
<td>
<p>character vector. Run label to be used to display in haPLOType. Required</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_sam.path">sam.path</code></td>
<td>
<p>string. Directory path folder containing all sequence alignment files (SAM). Required</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_label.path">label.path</code></td>
<td>
<p>string. Label file path. This customized label file is a tab-separate file that contains entries of SAM file name, individual ID, and group label. Required</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_vcf.path">vcf.path</code></td>
<td>
<p>string. VCF file path. Required</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_out.path">out.path</code></td>
<td>
<p>string. Optional. If not specified, the intermediate files are created under <code>TEMPDIR</code>, with the assumption that directory is granted for written permission.</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_add.filter">add.filter</code></td>
<td>
<p>boolean. Optional. If true, this removes any haplotype with unknown and deletion alignment characters i.e. &quot;*&quot; and &quot;_&quot;, removes any locus with large number of haplotypes ( # &gt; 40) , and remove any locus with fewer than half of the total individuals.</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_app.path">app.path</code></td>
<td>
<p>string. Path to shiny haPLOType app. Optional. If not specified, the path is default to <code>TEMPDIR</code>.</p>
</td></tr>
<tr><td><code id="prepHaplotFiles_+3A_n.jobs">n.jobs</code></td>
<td>
<p>positive integer. Number of SAM files to be parallel processed. Optional. This multithread is only available for non Window OS. Recommend two times the number of processors/core.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>This function returns a dataframe of 9 columns i.e group, id, locus, haplotype, depth, sum of Phred score, max of Phred score, allele balance and haplotype rank from highest to lowest read depth. This dataframe will also be saved in <code>out.path</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
run.label &lt;- "sebastes"

sam.path &lt;- tempdir()
untar(system.file("extdata",
                  "sebastes_sam.tar.gz",
                  package="microhaplot"),
      exdir = sam.path)


label.path &lt;- file.path(sam.path, "label.txt")
vcf.path &lt;- file.path(sam.path, "sebastes.vcf")

mvShinyHaplot(tempdir())
app.path &lt;- file.path(tempdir(), "microhaplot")

# retrieve system Perl version number
perl.version &lt;- as.numeric(system('perl -e "print $];"', intern=TRUE))

if (perl.version &gt;= 5.014) {
haplo.read.tbl &lt;- prepHaplotFiles(run.label = run.label,
                            sam.path = sam.path,
                            out.path = tempdir(),
                            label.path = label.path,
                            vcf.path = vcf.path,
                            app.path = app.path)
}else {
message("Perl version is outdated. Must &gt;= 5.014.")}

</code></pre>

<hr>
<h2 id='runShinyHaplot'>Run shiny microhaplot</h2><span id='topic+runShinyHaplot'></span>

<h3>Description</h3>

<p>Run shiny microhaplot app
</p>


<h3>Usage</h3>

<pre><code class='language-R'>runShinyHaplot(path = system.file("shiny", "microhaplot", package =
  "microhaplot"))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="runShinyHaplot_+3A_path">path</code></td>
<td>
<p>Path to shiny microhaplot app. Optional. If not specified, the path is default to local app path.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Runs shiny microhaplot application via <code>shiny::runApp</code> which typically doesn't return; interrupt R to stop the application (usually by pressing Ctrl+C or Esc).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if(interactive()){
runShinyHaplot()
}
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
