<!DOCTYPE html><html><head><title>Help for package microbial</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {microbial}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#.checkfile'><p>check file format</p></a></li>
<li><a href='#.getstar'><p>replace p value with star</p></a></li>
<li><a href='#.lda.fun'><p>LEfse function</p></a></li>
<li><a href='#betadiv'><p>calcaute beta diversity</p></a></li>
<li><a href='#betatest'><p>PERMANOVA test for phyloseq</p></a></li>
<li><a href='#biomarker'><p>Identify biomarker by using randomForest method</p></a></li>
<li><a href='#buildTree'><p>contruction of plylogenetic tree (extreme slow)</p></a></li>
<li><a href='#data-physeq'><p>The physeq data was modified from the (Data) Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample (2011)</p></a></li>
<li><a href='#difftest'><p>Calculate differential bacteria with DESeq2</p></a></li>
<li><a href='#distcolor'><p>distinguish colors for making figures</p></a></li>
<li><a href='#do_aov'><p>do anova test and return results as data.frame</p></a></li>
<li><a href='#do_ttest'><p>do t.test</p></a></li>
<li><a href='#do_wilcox'><p>do wilcox test</p></a></li>
<li><a href='#glmr'><p>Do the generalized linear model regression</p></a></li>
<li><a href='#ldamarker'><p>Identify biomarker by using LEfSe method</p></a></li>
<li><a href='#lightcolor'><p>light colors for making figures</p></a></li>
<li><a href='#normalize'><p>Normalize the phyloseq object with different methods</p></a></li>
<li><a href='#otu_table'><p>extract otu table</p></a></li>
<li><a href='#phy_tree'><p>Retrieve phylogenetic tree (phylo-class) from object.</p></a></li>
<li><a href='#plotalpha'><p>plot alpha diversity</p></a></li>
<li><a href='#plotbar'><p>plot bar for relative abundance for bacteria</p></a></li>
<li><a href='#plotbeta'><p>plot beta diversity</p></a></li>
<li><a href='#plotdiff'><p>plot differential results</p></a></li>
<li><a href='#plotLDA'><p>plot LEfSe results from ldamarker function</p></a></li>
<li><a href='#plotmarker'><p>plot the biomarker from the biomarker function with randomForest</p></a></li>
<li><a href='#plotquality'><p>plot the quality for the fastq file</p></a></li>
<li><a href='#prefilter'><p>filter the phyloseq</p></a></li>
<li><a href='#preRef'><p>Download the reference database</p></a></li>
<li><a href='#processSeq'><p>Perform dada2 analysis</p></a></li>
<li><a href='#psmelt'><p>Melt phyloseq data object into large data.frame</p></a></li>
<li><a href='#richness'><p>calculat the richness for the phyloseq object</p></a></li>
<li><a href='#sample_data'><p>extract sample information</p></a></li>
<li><a href='#subset_samples'><p>Subset the phyloseq based on sample</p></a></li>
<li><a href='#subset_taxa'><p>Subset species by taxonomic expression</p></a></li>
<li><a href='#tax_table'><p>extract taxonomy table</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Do 16s Data Analysis and Generate Figures</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.21</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions to enhance the available 
            statistical analysis procedures in R by providing simple functions to 
            analysis and visualize the 16S rRNA data.Here we present a tutorial 
            with minimum working examples to demonstrate usage and dependencies. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>dplyr, plyr, magrittr, broom, phyloseq, vegan, rlang, ggplot2,
ggpubr, DESeq2, SummarizedExperiment, S4Vectors, rstatix,
tidyr, phangorn, randomForest, edgeR</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>markdown,dada2,rmarkdown,knitr,tools,Biostrings, DECIPHER,
MASS,testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Software,GraphAndNetwork</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-05-15 17:44:20 UTC; bioguo</td>
</tr>
<tr>
<td>Author:</td>
<td>Kai Guo [aut, cre],
  Pan Gao [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kai Guo &lt;guokai8@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-05-15 18:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='.checkfile'>check file format</h2><span id='topic+.checkfile'></span>

<h3>Description</h3>

<p>check file format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.checkfile(file)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".checkfile_+3A_file">file</code></td>
<td>
<p>filename</p>
</td></tr>
</table>

<hr>
<h2 id='.getstar'>replace p value with star</h2><span id='topic+.getstar'></span>

<h3>Description</h3>

<p>replace p value with star
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.getstar(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".getstar_+3A_x">x</code></td>
<td>
<p>a (non-empty) numeric data values</p>
</td></tr>
</table>

<hr>
<h2 id='.lda.fun'>LEfse function</h2><span id='topic+.lda.fun'></span>

<h3>Description</h3>

<p>LEfse function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>.lda.fun(df)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id=".lda.fun_+3A_df">df</code></td>
<td>
<p>a dataframe with groups and bacteria abundance</p>
</td></tr>
</table>

<hr>
<h2 id='betadiv'>calcaute beta diversity</h2><span id='topic+betadiv'></span>

<h3>Description</h3>

<p>calcaute beta diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betadiv(physeq, distance = "bray", method = "PCoA")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betadiv_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="betadiv_+3A_distance">distance</code></td>
<td>
<p>A string character specifying dissimilarity index to be used in calculating pairwise distances (Default index is &quot;bray&quot;.).
&quot;unifrac&quot;,&quot;wunifrac&quot;,&quot;manhattan&quot;, &quot;euclidean&quot;, &quot;canberra&quot;, &quot;bray&quot;, &quot;kulczynski&quot;, &quot;jaccard&quot;, &quot;gower&quot;, &quot;altGower&quot;,
&quot;morisita&quot;, &quot;horn&quot;, &quot;mountford&quot;, &quot;raup&quot; , &quot;binomial&quot;, &quot;chao&quot;, &quot;cao&quot; or &quot;mahalanobis&quot;.</p>
</td></tr>
<tr><td><code id="betadiv_+3A_method">method</code></td>
<td>
<p>A character string specifying ordination method. All methods available to the <code>ordinate</code> function
of <code>phyloseq</code> are acceptable here as well.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with beta diversity data.frame and PCs
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("Physeq")
phy&lt;-normalize(physeq)
res &lt;- betadiv(phy)
}
</code></pre>

<hr>
<h2 id='betatest'>PERMANOVA test for phyloseq</h2><span id='topic+betatest'></span>

<h3>Description</h3>

<p>PERMANOVA test for phyloseq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>betatest(physeq, group, distance = "bray")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="betatest_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="betatest_+3A_group">group</code></td>
<td>
<p>(Required). Character string specifying name of a categorical variable that is preferred for grouping the information.
information.</p>
</td></tr>
<tr><td><code id="betatest_+3A_distance">distance</code></td>
<td>
<p>A string character specifying dissimilarity index to be used in calculating pairwise distances (Default index is &quot;bray&quot;.).
&quot;unifrac&quot;,&quot;wunifrac&quot;,&quot;manhattan&quot;, &quot;euclidean&quot;, &quot;canberra&quot;, &quot;bray&quot;, &quot;kulczynski&quot;, &quot;jaccard&quot;, &quot;gower&quot;, &quot;altGower&quot;,
&quot;morisita&quot;, &quot;horn&quot;, &quot;mountford&quot;, &quot;raup&quot; , &quot;binomial&quot;, &quot;chao&quot;, &quot;cao&quot; or &quot;mahalanobis&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>PERMANOVA test result
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("Physeq")
phy&lt;-normalize(physeq)
beta &lt;-betatest(phy,group="SampleType")
}
</code></pre>

<hr>
<h2 id='biomarker'>Identify biomarker by using randomForest method</h2><span id='topic+biomarker'></span>

<h3>Description</h3>

<p>Identify biomarker by using randomForest method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biomarker(
  physeq,
  group,
  ntree = 500,
  pvalue = 0.05,
  normalize = TRUE,
  method = "relative"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="biomarker_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="biomarker_+3A_group">group</code></td>
<td>
<p>group. A character string specifying the name of a categorical variable containing  grouping information.</p>
</td></tr>
<tr><td><code id="biomarker_+3A_ntree">ntree</code></td>
<td>
<p>Number of trees to grow. This should not be set to too small a number,
to ensure that every input row gets predicted at least a few times.</p>
</td></tr>
<tr><td><code id="biomarker_+3A_pvalue">pvalue</code></td>
<td>
<p>pvalue threshold for significant results from kruskal.test</p>
</td></tr>
<tr><td><code id="biomarker_+3A_normalize">normalize</code></td>
<td>
<p>to normalize the data before analysis(TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="biomarker_+3A_method">method</code></td>
<td>
<p>A list of character strings specifying <code>method</code> to be used to normalize the phyloseq object
Available methods are: &quot;relative&quot;,&quot;TMM&quot;,&quot;vst&quot;,&quot;log2&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data frame with significant biomarker
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
res &lt;- biomarker(physeq,group="group")

</code></pre>

<hr>
<h2 id='buildTree'>contruction of plylogenetic tree (extreme slow)</h2><span id='topic+buildTree'></span>

<h3>Description</h3>

<p>contruction of plylogenetic tree (extreme slow)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>buildTree(seqs)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="buildTree_+3A_seqs">seqs</code></td>
<td>
<p>DNA sequences</p>
</td></tr>
</table>


<h3>Value</h3>

<p>tree object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>

<hr>
<h2 id='data-physeq'>The physeq data was modified from the (Data) Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample (2011)</h2><span id='topic+data-physeq'></span><span id='topic+physeq'></span>

<h3>Description</h3>

<p>Published in PNAS in early 2011. This work compared the microbial
communities from 25 environmental samples and three known &ldquo;mock communities&rdquo;
&ndash; a total of 9 sample types &ndash; at a depth averaging 3.1 million reads per sample.
Authors were able to reproduce diversity patterns seen in many other
published studies, while also invesitigating technical issues/bias by
applying the same techniques to simulated microbial communities of known
</p>


<h3>References</h3>

<p>Caporaso, J. G., et al. (2011).
Global patterns of 16S rRNA diversity at a depth of millions of sequences per sample.
PNAS, 108, 4516-4522.
PMCID: PMC3063599
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(Physeq)
</code></pre>

<hr>
<h2 id='difftest'>Calculate differential bacteria with DESeq2</h2><span id='topic+difftest'></span>

<h3>Description</h3>

<p>Calculate differential bacteria with DESeq2
</p>


<h3>Usage</h3>

<pre><code class='language-R'>difftest(
  physeq,
  group,
  ref = NULL,
  pvalue = 0.05,
  padj = NULL,
  log2FC = 0,
  gm_mean = TRUE,
  fitType = "local",
  quiet = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="difftest_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="difftest_+3A_group">group</code></td>
<td>
<p>group (DESeq2). A character string specifying the name of a categorical variable containing  grouping information.</p>
</td></tr>
<tr><td><code id="difftest_+3A_ref">ref</code></td>
<td>
<p>reference group</p>
</td></tr>
<tr><td><code id="difftest_+3A_pvalue">pvalue</code></td>
<td>
<p>pvalue threshold for significant results</p>
</td></tr>
<tr><td><code id="difftest_+3A_padj">padj</code></td>
<td>
<p>adjust p value threshold for significant results</p>
</td></tr>
<tr><td><code id="difftest_+3A_log2fc">log2FC</code></td>
<td>
<p>log2 Fold Change threshold</p>
</td></tr>
<tr><td><code id="difftest_+3A_gm_mean">gm_mean</code></td>
<td>
<p>TRUE/FALSE calculate geometric means prior to estimate size factors</p>
</td></tr>
<tr><td><code id="difftest_+3A_fittype">fitType</code></td>
<td>
<p>either &quot;parametric&quot;, &quot;local&quot;, or &quot;mean&quot; for the type of fitting of dispersions to the mean intensity.</p>
</td></tr>
<tr><td><code id="difftest_+3A_quiet">quiet</code></td>
<td>
<p>whether to print messages at each step</p>
</td></tr>
</table>


<h3>Value</h3>

<p>datafame with differential test with DESeq2
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data("Physeq")
res &lt;- difftest(physeq,group="group")

</code></pre>

<hr>
<h2 id='distcolor'>distinguish colors for making figures</h2><span id='topic+distcolor'></span>

<h3>Description</h3>

<p>distinguish colors for making figures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distcolor
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 41.
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>

<hr>
<h2 id='do_aov'>do anova test and return results as data.frame</h2><span id='topic+do_aov'></span>

<h3>Description</h3>

<p>do anova test and return results as data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_aov(x, group, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="do_aov_+3A_x">x</code></td>
<td>
<p>data.frame with sample id as the column name, genes or otu as rownames</p>
</td></tr>
<tr><td><code id="do_aov_+3A_group">group</code></td>
<td>
<p>group factor used for comparison</p>
</td></tr>
<tr><td><code id="do_aov_+3A_...">...</code></td>
<td>
<p>parameters to anova_test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("ToothGrowth")
do_aov(ToothGrowth,group="supp")
}
</code></pre>

<hr>
<h2 id='do_ttest'>do t.test</h2><span id='topic+do_ttest'></span>

<h3>Description</h3>

<p>do t.test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_ttest(x, group, ref = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="do_ttest_+3A_x">x</code></td>
<td>
<p>data.frame with sample id as the column name, genes or otu as rownames</p>
</td></tr>
<tr><td><code id="do_ttest_+3A_group">group</code></td>
<td>
<p>group factor used for comparison</p>
</td></tr>
<tr><td><code id="do_ttest_+3A_ref">ref</code></td>
<td>
<p>reference group</p>
</td></tr>
<tr><td><code id="do_ttest_+3A_...">...</code></td>
<td>
<p>parameters to t_test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("mtcars")
do_ttest(mtcars,group="vs")
do_ttest(mtcars,group="cyl",ref="4")
}
</code></pre>

<hr>
<h2 id='do_wilcox'>do wilcox test</h2><span id='topic+do_wilcox'></span>

<h3>Description</h3>

<p>do wilcox test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>do_wilcox(x, group, ref = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="do_wilcox_+3A_x">x</code></td>
<td>
<p>data.frame with sample id as the column name, genes or otu as rownames</p>
</td></tr>
<tr><td><code id="do_wilcox_+3A_group">group</code></td>
<td>
<p>group factor used for comparison</p>
</td></tr>
<tr><td><code id="do_wilcox_+3A_ref">ref</code></td>
<td>
<p>reference group</p>
</td></tr>
<tr><td><code id="do_wilcox_+3A_...">...</code></td>
<td>
<p>parameters to wilcox_test</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("mtcars")
do_wilcox(mtcars,group="vs")
do_wilcox(mtcars,group="cyl",ref="4")
}
</code></pre>

<hr>
<h2 id='glmr'>Do the generalized linear model regression</h2><span id='topic+glmr'></span>

<h3>Description</h3>

<p>Do the generalized linear model regression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmr(
  physeq,
  group,
  factors = NULL,
  ref = NULL,
  family = binomial(link = "logit")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmr_+3A_physeq">physeq</code></td>
<td>
<p>phyloseq object</p>
</td></tr>
<tr><td><code id="glmr_+3A_group">group</code></td>
<td>
<p>the group factor to regression</p>
</td></tr>
<tr><td><code id="glmr_+3A_factors">factors</code></td>
<td>
<p>a vector to indicate adjuested factors</p>
</td></tr>
<tr><td><code id="glmr_+3A_ref">ref</code></td>
<td>
<p>the reference group</p>
</td></tr>
<tr><td><code id="glmr_+3A_family">family</code></td>
<td>
<p>binomial() or gaussian()</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
phy&lt;-normalize(physeq)
fit &lt;-glmr(phy,group="SampleType")

</code></pre>

<hr>
<h2 id='ldamarker'>Identify biomarker by using LEfSe method</h2><span id='topic+ldamarker'></span>

<h3>Description</h3>

<p>Identify biomarker by using LEfSe method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ldamarker(physeq, group, pvalue = 0.05, normalize = TRUE, method = "relative")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ldamarker_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="ldamarker_+3A_group">group</code></td>
<td>
<p>group. A character string specifying the name of a categorical variable containing  grouping information.</p>
</td></tr>
<tr><td><code id="ldamarker_+3A_pvalue">pvalue</code></td>
<td>
<p>pvalue threshold for significant results from kruskal.test</p>
</td></tr>
<tr><td><code id="ldamarker_+3A_normalize">normalize</code></td>
<td>
<p>to normalize the data before analysis(TRUE/FALSE)</p>
</td></tr>
<tr><td><code id="ldamarker_+3A_method">method</code></td>
<td>
<p>A list of character strings specifying <code>method</code> to be used to normalize the phyloseq object
Available methods are: &quot;relative&quot;,&quot;TMM&quot;,&quot;vst&quot;,&quot;log2&quot;.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
res &lt;- ldamarker(physeq,group="group")


</code></pre>

<hr>
<h2 id='lightcolor'>light colors for making figures</h2><span id='topic+lightcolor'></span>

<h3>Description</h3>

<p>light colors for making figures
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lightcolor
</code></pre>


<h3>Format</h3>

<p>An object of class <code>character</code> of length 56.
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>

<hr>
<h2 id='normalize'>Normalize the phyloseq object with different methods</h2><span id='topic+normalize'></span>

<h3>Description</h3>

<p>Normalize the phyloseq object with different methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize(physeq, group, method = "relative", table = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="normalize_+3A_group">group</code></td>
<td>
<p>group (DESeq2). A character string specifying the name of a categorical variable containing  grouping information.</p>
</td></tr>
<tr><td><code id="normalize_+3A_method">method</code></td>
<td>
<p>A list of character strings specifying <code>method</code> to be used to normalize the phyloseq object
Available methods are: &quot;relative&quot;,&quot;TMM&quot;,&quot;vst&quot;,&quot;log2&quot;.</p>
</td></tr>
<tr><td><code id="normalize_+3A_table">table</code></td>
<td>
<p>return a data.frame or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>phyloseq object with normalized data
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("Physeq")
phy&lt;-normalize(physeq)
}
</code></pre>

<hr>
<h2 id='otu_table'>extract otu table</h2><span id='topic+otu_table'></span>

<h3>Description</h3>

<p>extract otu table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>otu_table(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="otu_table_+3A_physeq">physeq</code></td>
<td>
<p>(Required). An integer matrix, otu_table-class, or phyloseq-class.</p>
</td></tr>
<tr><td><code id="otu_table_+3A_...">...</code></td>
<td>
<p>parameters for the otu_table function in phyloseq package</p>
</td></tr>
</table>

<hr>
<h2 id='phy_tree'>Retrieve phylogenetic tree (phylo-class) from object.</h2><span id='topic+phy_tree'></span>

<h3>Description</h3>

<p>Retrieve phylogenetic tree (phylo-class) from object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phy_tree(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phy_tree_+3A_physeq">physeq</code></td>
<td>
<p>(Required). An instance of phyloseq-class that contains a phylogenetic tree.
If physeq is a phylogenetic tree (a component data class), then it is returned as-is.</p>
</td></tr>
<tr><td><code id="phy_tree_+3A_...">...</code></td>
<td>
<p>parameters for the phy_tree function in phyloseq package</p>
</td></tr>
</table>

<hr>
<h2 id='plotalpha'>plot alpha diversity</h2><span id='topic+plotalpha'></span>

<h3>Description</h3>

<p>plot alpha diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotalpha(
  physeq,
  group,
  method = c("Observed", "Simpson", "Shannon"),
  color = NULL,
  geom = "boxplot",
  pvalue = 0.05,
  padj = NULL,
  sig.only = TRUE,
  wilcox = FALSE,
  show.number = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotalpha_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_group">group</code></td>
<td>
<p>group (Required). A character string specifying the name of a categorical variable containing  grouping information.</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_method">method</code></td>
<td>
<p>A list of character strings specifying <code>method</code> to be used to calculate for alpha diversity
in the data. Available methods are: &quot;Observed&quot;,&quot;Chao1&quot;,&quot;ACE&quot;,&quot;Richness&quot;, &quot;Fisher&quot;, &quot;Simpson&quot;, &quot;Shannon&quot;, &quot;Evenness&quot;,&quot;InvSimpson&quot;.</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_color">color</code></td>
<td>
<p>A vector of character use specifying the color</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_geom">geom</code></td>
<td>
<p>different geom to display(&quot;boxplot&quot;,&quot;violin&quot;,&quot;dotplot&quot;)</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_pvalue">pvalue</code></td>
<td>
<p>pvalue threshold for significant dispersion results</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_padj">padj</code></td>
<td>
<p>adjust p value threshold for significant dispersion results</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_sig.only">sig.only</code></td>
<td>
<p>display the significant comparsion only(TRUE/ FALSE)</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_wilcox">wilcox</code></td>
<td>
<p>use wilcoxon test or not</p>
</td></tr>
<tr><td><code id="plotalpha_+3A_show.number">show.number</code></td>
<td>
<p>to show the pvalue instead of significant symbol(TRUE/FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a ggplot object. This can further be manipulated as preferred by user.
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("Physeq")
plotalpha(physeq,group="SampleType")
}
</code></pre>

<hr>
<h2 id='plotbar'>plot bar for relative abundance for bacteria</h2><span id='topic+plotbar'></span>

<h3>Description</h3>

<p>plot bar for relative abundance for bacteria
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotbar(
  physeq,
  level = "Phylum",
  color = NULL,
  group = NULL,
  top = 5,
  return = FALSE,
  fontsize.x = 5,
  fontsize.y = 12
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotbar_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="plotbar_+3A_level">level</code></td>
<td>
<p>the level to plot</p>
</td></tr>
<tr><td><code id="plotbar_+3A_color">color</code></td>
<td>
<p>A vector of character use specifying the color</p>
</td></tr>
<tr><td><code id="plotbar_+3A_group">group</code></td>
<td>
<p>group (Optional). A character string specifying the name of a categorical variable containing  grouping information.</p>
</td></tr>
<tr><td><code id="plotbar_+3A_top">top</code></td>
<td>
<p>the number of most abundance bacteria to display</p>
</td></tr>
<tr><td><code id="plotbar_+3A_return">return</code></td>
<td>
<p>return the data with the relative abundance</p>
</td></tr>
<tr><td><code id="plotbar_+3A_fontsize.x">fontsize.x</code></td>
<td>
<p>the size of x axis label</p>
</td></tr>
<tr><td><code id="plotbar_+3A_fontsize.y">fontsize.y</code></td>
<td>
<p>the size of y axis label</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a ggplot object. This can further be manipulated as preferred by user.
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
phy&lt;-normalize(physeq)
plotbar(phy,level="Phylum")

</code></pre>

<hr>
<h2 id='plotbeta'>plot beta diversity</h2><span id='topic+plotbeta'></span>

<h3>Description</h3>

<p>plot beta diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotbeta(
  physeq,
  group,
  shape = NULL,
  distance = "bray",
  method = "PCoA",
  color = NULL,
  size = 3,
  ellipse = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotbeta_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_group">group</code></td>
<td>
<p>(Required). Character string specifying name of a categorical variable that is preferred for grouping the information.
information.</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_shape">shape</code></td>
<td>
<p>shape(Optional) Character string specifying shape of a categorical variable</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_distance">distance</code></td>
<td>
<p>A string character specifying dissimilarity index to be used in calculating pairwise distances (Default index is &quot;bray&quot;.).
&quot;unifrac&quot;,&quot;wunifrac&quot;,&quot;manhattan&quot;, &quot;euclidean&quot;, &quot;canberra&quot;, &quot;bray&quot;, &quot;kulczynski&quot;, &quot;jaccard&quot;, &quot;gower&quot;, &quot;altGower&quot;,
&quot;morisita&quot;, &quot;horn&quot;, &quot;mountford&quot;, &quot;raup&quot; , &quot;binomial&quot;, &quot;chao&quot;, &quot;cao&quot; or &quot;mahalanobis&quot;.</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_method">method</code></td>
<td>
<p>A character string specifying ordination method. All methods available to the <code>ordinate</code> function
of <code>phyloseq</code> are acceptable here as well.</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_color">color</code></td>
<td>
<p>user defined color for group</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_size">size</code></td>
<td>
<p>the point size</p>
</td></tr>
<tr><td><code id="plotbeta_+3A_ellipse">ellipse</code></td>
<td>
<p>draw ellipse or not</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("Physeq")
phy&lt;-normalize(physeq)
plotbeta(phy,group="SampleType")
}
</code></pre>

<hr>
<h2 id='plotdiff'>plot differential results</h2><span id='topic+plotdiff'></span>

<h3>Description</h3>

<p>plot differential results
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotdiff(
  res,
  level = "Genus",
  color = NULL,
  pvalue = 0.05,
  padj = NULL,
  log2FC = 0,
  size = 3,
  fontsize.x = 5,
  fontsize.y = 10,
  horiz = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotdiff_+3A_res">res</code></td>
<td>
<p>differential test results from diff_test</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_level">level</code></td>
<td>
<p>the level to plot</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_color">color</code></td>
<td>
<p>A vector of character use specifying the color</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_pvalue">pvalue</code></td>
<td>
<p>pvalue threshold for significant  results</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_padj">padj</code></td>
<td>
<p>adjust p value threshold for significant  results</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_log2fc">log2FC</code></td>
<td>
<p>log2 Fold Change threshold</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_size">size</code></td>
<td>
<p>size for the point</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_fontsize.x">fontsize.x</code></td>
<td>
<p>the size of x axis label</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_fontsize.y">fontsize.y</code></td>
<td>
<p>the size of y axis label</p>
</td></tr>
<tr><td><code id="plotdiff_+3A_horiz">horiz</code></td>
<td>
<p>horizontal or not (TRUE/FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
data("Physeq")
res &lt;- difftest(physeq,group="group")
plotdiff(res,level="Genus",padj=0.001)

</code></pre>

<hr>
<h2 id='plotLDA'>plot LEfSe results from ldamarker function</h2><span id='topic+plotLDA'></span>

<h3>Description</h3>

<p>plot LEfSe results from ldamarker function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotLDA(
  x,
  group,
  lda = 2,
  pvalue = 0.05,
  padj = NULL,
  color = NULL,
  fontsize.x = 4,
  fontsize.y = 5
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotLDA_+3A_x">x</code></td>
<td>
<p>LEfse results from ldamarker</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_group">group</code></td>
<td>
<p>a vector include two character to show the group comparsion</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_lda">lda</code></td>
<td>
<p>LDA threshold for significant biomarker</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_pvalue">pvalue</code></td>
<td>
<p>pvalue threshold for significant results</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_padj">padj</code></td>
<td>
<p>adjust p value threshold for significant results</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_color">color</code></td>
<td>
<p>A vector of character use specifying the color</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_fontsize.x">fontsize.x</code></td>
<td>
<p>the size of x axis label</p>
</td></tr>
<tr><td><code id="plotLDA_+3A_fontsize.y">fontsize.y</code></td>
<td>
<p>the size of y axis label</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
res &lt;- ldamarker(physeq,group="group")
plotLDA(res,group=c("A","B"),lda=5,pvalue=0.05)

</code></pre>

<hr>
<h2 id='plotmarker'>plot the biomarker from the biomarker function with randomForest</h2><span id='topic+plotmarker'></span>

<h3>Description</h3>

<p>plot the biomarker from the biomarker function with randomForest
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotmarker(
  x,
  level = "Genus",
  top = 30,
  rotate = FALSE,
  dot.size = 8,
  label.color = "black",
  label.size = 6
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotmarker_+3A_x">x</code></td>
<td>
<p>biomarker results from randomForest</p>
</td></tr>
<tr><td><code id="plotmarker_+3A_level">level</code></td>
<td>
<p>the bacteria level to display</p>
</td></tr>
<tr><td><code id="plotmarker_+3A_top">top</code></td>
<td>
<p>the number of important biomarker to draw</p>
</td></tr>
<tr><td><code id="plotmarker_+3A_rotate">rotate</code></td>
<td>
<p>TRUE/FALSE</p>
</td></tr>
<tr><td><code id="plotmarker_+3A_dot.size">dot.size</code></td>
<td>
<p>size for the dot</p>
</td></tr>
<tr><td><code id="plotmarker_+3A_label.color">label.color</code></td>
<td>
<p>label color</p>
</td></tr>
<tr><td><code id="plotmarker_+3A_label.size">label.size</code></td>
<td>
<p>label size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
res &lt;- biomarker(physeq,group="group")
plotmarker(res,level="Genus")

</code></pre>

<hr>
<h2 id='plotquality'>plot the quality for the fastq file</h2><span id='topic+plotquality'></span>

<h3>Description</h3>

<p>plot the quality for the fastq file
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotquality(file, n = 5e+05, aggregate = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotquality_+3A_file">file</code></td>
<td>
<p>(Required). character. File path(s) to fastq or fastq.gz file(s).</p>
</td></tr>
<tr><td><code id="plotquality_+3A_n">n</code></td>
<td>
<p>(Optional). Default 500,000. The number of records to sample from the fastq file.</p>
</td></tr>
<tr><td><code id="plotquality_+3A_aggregate">aggregate</code></td>
<td>
<p>(Optional). Default FALSE. If TRUE, compute an aggregate quality profile for all fastq files provided.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>figure
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
plotquality(system.file("extdata", "sam1F.fastq.gz", package="dada2"))

</code></pre>

<hr>
<h2 id='prefilter'>filter the phyloseq</h2><span id='topic+prefilter'></span>

<h3>Description</h3>

<p>filter the phyloseq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>prefilter(physeq, min = 10, perc = 0.05)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="prefilter_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="prefilter_+3A_min">min</code></td>
<td>
<p>Numeric, the threshold for mininal Phylum shown in samples</p>
</td></tr>
<tr><td><code id="prefilter_+3A_perc">perc</code></td>
<td>
<p>Numeric, input the percentage of samples for which to filter low counts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>filter phyloseq object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("Physeq")
physeqs&lt;-prefilter(physeq)

</code></pre>

<hr>
<h2 id='preRef'>Download the reference database</h2><span id='topic+preRef'></span>

<h3>Description</h3>

<p>Download the reference database
</p>


<h3>Usage</h3>

<pre><code class='language-R'>preRef(ref_db, path = ".")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="preRef_+3A_ref_db">ref_db</code></td>
<td>
<p>the reference database</p>
</td></tr>
<tr><td><code id="preRef_+3A_path">path</code></td>
<td>
<p>path for the  database</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the path of the database
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
preRef(ref_db="silva",path=tempdir())

</code></pre>

<hr>
<h2 id='processSeq'>Perform dada2 analysis</h2><span id='topic+processSeq'></span>

<h3>Description</h3>

<p>Perform dada2 analysis
</p>


<h3>Usage</h3>

<pre><code class='language-R'>processSeq(
  path = ".",
  truncLen = c(0, 0),
  trimLeft = 0,
  trimRight = 0,
  minLen = 20,
  maxLen = Inf,
  sample_info = NULL,
  train_data = "silva_nr99_v138_train_set.fa.gz",
  train_species = "silva_species_assignment_v138.fa.gz",
  outpath = NULL,
  saveobj = FALSE,
  buildtree = FALSE,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="processSeq_+3A_path">path</code></td>
<td>
<p>working dir for the input reads</p>
</td></tr>
<tr><td><code id="processSeq_+3A_trunclen">truncLen</code></td>
<td>
<p>(Optional). Default 0 (no truncation). Truncate reads after truncLen bases. Reads shorter than this are discarded.</p>
</td></tr>
<tr><td><code id="processSeq_+3A_trimleft">trimLeft</code></td>
<td>
<p>(Optional). The number of nucleotides to remove from the start of each read.</p>
</td></tr>
<tr><td><code id="processSeq_+3A_trimright">trimRight</code></td>
<td>
<p>(Optional). Default 0. The number of nucleotides to remove from the end of each read.
If both truncLen and trimRight are provided, truncation will be performed after trimRight is enforced.</p>
</td></tr>
<tr><td><code id="processSeq_+3A_minlen">minLen</code></td>
<td>
<p>(Optional). Default 20. Remove reads with length less than minLen. minLen is enforced after trimming and truncation.</p>
</td></tr>
<tr><td><code id="processSeq_+3A_maxlen">maxLen</code></td>
<td>
<p>Optional). Default Inf (no maximum). Remove reads with length greater than maxLen. maxLen is enforced before trimming and truncation.</p>
</td></tr>
<tr><td><code id="processSeq_+3A_sample_info">sample_info</code></td>
<td>
<p>(Optional).sample information for the sequence</p>
</td></tr>
<tr><td><code id="processSeq_+3A_train_data">train_data</code></td>
<td>
<p>(Required).training database</p>
</td></tr>
<tr><td><code id="processSeq_+3A_train_species">train_species</code></td>
<td>
<p>(Required). species database</p>
</td></tr>
<tr><td><code id="processSeq_+3A_outpath">outpath</code></td>
<td>
<p>(Optional).the path for the filtered reads and th out table</p>
</td></tr>
<tr><td><code id="processSeq_+3A_saveobj">saveobj</code></td>
<td>
<p>(Optional).Default FALSE. save the phyloseq object output.</p>
</td></tr>
<tr><td><code id="processSeq_+3A_buildtree">buildtree</code></td>
<td>
<p>build phylogenetic tree or not(default: FALSE)</p>
</td></tr>
<tr><td><code id="processSeq_+3A_verbose">verbose</code></td>
<td>
<p>(Optional). Default TRUE. Print verbose text output.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list include count table, summary table, taxonomy information and phyloseq object
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>

<hr>
<h2 id='psmelt'>Melt phyloseq data object into large data.frame</h2><span id='topic+psmelt'></span>

<h3>Description</h3>

<p>Melt phyloseq data object into large data.frame
</p>


<h3>Usage</h3>

<pre><code class='language-R'>psmelt(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="psmelt_+3A_physeq">physeq</code></td>
<td>
<p>A sample_data-class, or a phyloseq-class object with a sample_data.
If the sample_data slot is missing in physeq, then physeq will be returned as-is,
and a warning will be printed to screen.</p>
</td></tr>
<tr><td><code id="psmelt_+3A_...">...</code></td>
<td>
<p>parameters for the subset_samples function in phyloseq package</p>
</td></tr>
</table>

<hr>
<h2 id='richness'>calculat the richness for the phyloseq object</h2><span id='topic+richness'></span>

<h3>Description</h3>

<p>calculat the richness for the phyloseq object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>richness(physeq, method = c("Observed", "Simpson", "Shannon"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="richness_+3A_physeq">physeq</code></td>
<td>
<p>A <code>phyloseq</code> object containing merged information of abundance,
taxonomic assignment, sample data including the measured variables and categorical information
of the samples, and / or phylogenetic tree if available.</p>
</td></tr>
<tr><td><code id="richness_+3A_method">method</code></td>
<td>
<p>A list of character strings specifying <code>method</code> to be used to calculate for alpha diversity
in the data. Available methods are: &quot;Observed&quot;,&quot;Chao1&quot;,&quot;ACE&quot;,&quot;Richness&quot;, &quot;Fisher&quot;, &quot;Simpson&quot;, &quot;Shannon&quot;, &quot;Evenness&quot;,&quot;InvSimpson&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>data.frame of alpha diversity
</p>


<h3>Author(s)</h3>

<p>Kai Guo
</p>


<h3>Examples</h3>

<pre><code class='language-R'>{
data("Physeq")
rich &lt;-richness(physeq,method=c("Simpson", "Shannon"))
}
</code></pre>

<hr>
<h2 id='sample_data'>extract sample information</h2><span id='topic+sample_data'></span>

<h3>Description</h3>

<p>extract sample information
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_data(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_data_+3A_physeq">physeq</code></td>
<td>
<p>(Required). A data.frame-class, or a phyloseq-class object.</p>
</td></tr>
<tr><td><code id="sample_data_+3A_...">...</code></td>
<td>
<p>parameters for the sample_data function in phyloseq package</p>
</td></tr>
</table>

<hr>
<h2 id='subset_samples'>Subset the phyloseq based on sample</h2><span id='topic+subset_samples'></span>

<h3>Description</h3>

<p>Subset the phyloseq based on sample
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_samples(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_samples_+3A_physeq">physeq</code></td>
<td>
<p>A sample_data-class, or a phyloseq-class object with a sample_data.
If the sample_data slot is missing in physeq, then physeq will be returned as-is,
and a warning will be printed to screen.</p>
</td></tr>
<tr><td><code id="subset_samples_+3A_...">...</code></td>
<td>
<p>parameters for the subset_samples function in phyloseq package</p>
</td></tr>
</table>

<hr>
<h2 id='subset_taxa'>Subset species by taxonomic expression</h2><span id='topic+subset_taxa'></span>

<h3>Description</h3>

<p>Subset species by taxonomic expression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_taxa(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_taxa_+3A_physeq">physeq</code></td>
<td>
<p>A sample_data-class, or a phyloseq-class object with a sample_data.
If the sample_data slot is missing in physeq, then physeq will be returned as-is,
and a warning will be printed to screen.</p>
</td></tr>
<tr><td><code id="subset_taxa_+3A_...">...</code></td>
<td>
<p>parameters for the subset_taxa function in phyloseq package</p>
</td></tr>
</table>

<hr>
<h2 id='tax_table'>extract taxonomy table</h2><span id='topic+tax_table'></span>

<h3>Description</h3>

<p>extract taxonomy table
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_table(physeq, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tax_table_+3A_physeq">physeq</code></td>
<td>
<p>An object among the set of classes defined by the phyloseq package that contain taxonomyTable.</p>
</td></tr>
<tr><td><code id="tax_table_+3A_...">...</code></td>
<td>
<p>parameters for the tax_table function in phyloseq package</p>
</td></tr>
</table>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
