<!DOCTYPE html><html><head><title>Help for package MicrobiomeStat</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MicrobiomeStat}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#linda'><p>Linear (Lin) Model for Differential Abundance (DA) Analysis of High-dimensional Compositional Data</p></a></li>
<li><a href='#linda.plot'><p>Plot LinDA Results</p></a></li>
<li><a href='#linda.wald.test'><p>Wald test for bias-corrected regression coefficients</p></a></li>
<li><a href='#smokers'><p>Microbiome data from the human upper respiratory tract (left and right throat)</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Statistical Methods for Microbiome Compositional Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-03-13</td>
</tr>
<tr>
<td>Author:</td>
<td>Xianyang Zhang [aut],
  Jun Chen [aut, cre],
  Huijuan Zhou [ctb]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jun Chen &lt;chen.jun2@mayo.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A suite of methods for powerful and robust microbiome data analysis addressing zero-inflation, phylogenetic structure and compositional effects (Zhou et al. (2022)&lt;<a href="https://doi.org/10.1186%2Fs13059-022-02655-5">doi:10.1186/s13059-022-02655-5</a>&gt;).  The methods can be applied to the analysis of other (high-dimensional) compositional data arising from sequencing experiments.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>ggplot2, matrixStats, parallel, stats, utils, Matrix, statmod,
MASS, ggrepel, lmerTest, foreach, modeest</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-01 22:00:40 UTC; m123485</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-01 22:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='linda'>Linear (Lin) Model for Differential Abundance (DA) Analysis of High-dimensional Compositional Data</h2><span id='topic+linda'></span>

<h3>Description</h3>

<p>The function implements a simple, robust and highly scalable approach to tackle
the compositional effects in differential abundance analysis of high-dimensional compositional data. 
It fits linear regression models on the centered log2-ratio transformed data, identifies a bias term due to the transformation
and compositional effect, and corrects the bias using the mode of the regression coefficients.
It could fit mixed-effect models for analysis of correlated data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linda(
  feature.dat,
  meta.dat,
  formula,
  feature.dat.type = c('count', 'proportion'),
  prev.filter = 0,
  mean.abund.filter = 0, 
  max.abund.filter = 0,
  is.winsor = TRUE,
  outlier.pct = 0.03,
  adaptive = TRUE,
  zero.handling = c('pseudo-count', 'imputation'),
  pseudo.cnt = 0.5,
  corr.cut = 0.1,
  p.adj.method = "BH",
  alpha = 0.05,
  n.cores = 1, 
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linda_+3A_feature.dat">feature.dat</code></td>
<td>
<p>a matrix of counts/proportions, row - features (OTUs, genes, etc) , column - samples.</p>
</td></tr>
<tr><td><code id="linda_+3A_meta.dat">meta.dat</code></td>
<td>
<p>a data frame containing the sample meta data. If there are NAs, the corresponding samples
will be removed in the analysis.</p>
</td></tr>
<tr><td><code id="linda_+3A_formula">formula</code></td>
<td>
<p>a character string for the formula. The formula should conform to that used by <code>lm</code> (independent 
data) or <code>lmer</code> (correlated data).
For example: <code>formula = '~x1*x2+x3+(1|id)'</code>. At least one fixed effect is required.</p>
</td></tr>
<tr><td><code id="linda_+3A_feature.dat.type">feature.dat.type</code></td>
<td>
<p>the type of the feature data. It could be &quot;count&quot; or &quot;proportion&quot;.</p>
</td></tr>
<tr><td><code id="linda_+3A_prev.filter">prev.filter</code></td>
<td>
<p>the prevalence (percentage of non-zeros) cutoff, under which the features will  be filtered. The default is 0. </p>
</td></tr>
<tr><td><code id="linda_+3A_mean.abund.filter">mean.abund.filter</code></td>
<td>
<p>the mean relative abundance cutoff, under which the features will  be filtered. The default is 0.</p>
</td></tr>
<tr><td><code id="linda_+3A_max.abund.filter">max.abund.filter</code></td>
<td>
<p>the max relative abundance cutoff, under which the features will  be filtered. The default is 0.</p>
</td></tr>
<tr><td><code id="linda_+3A_is.winsor">is.winsor</code></td>
<td>
<p>a logical value indicating whether winsorization should be performed to replace outliers (high values).
The default is TRUE.</p>
</td></tr>
<tr><td><code id="linda_+3A_outlier.pct">outlier.pct</code></td>
<td>
<p>the expected percentage of outliers. These outliers will be winsorized. The default is 0.03.</p>
</td></tr>
<tr><td><code id="linda_+3A_adaptive">adaptive</code></td>
<td>
<p>a logical value indicating whether the approach to handle zeros (pseudo-count or imputation)
will be determined based on the correlations between the log(sequencing depth) and the explanatory variables
in <code>formula</code> when <code>feature.dat</code> is 'count'. If TRUE and the correlation p-value for any explanatory variable 
is smaller than or equal to <code>corr.cut</code>, the imputation approach will be used; otherwise, the pseudo-count approach will be used. </p>
</td></tr>
<tr><td><code id="linda_+3A_zero.handling">zero.handling</code></td>
<td>
<p>a character string of 'pseudo-count' or 'imputation' indicating the zero handling method
used when <code>feature.dat</code> is 'count'.  If 'pseudo-count', a<code>pseudo.cnt</code> will be added to each value in <code>feature.dat</code>. 
If 'imputation', then we use the imputation approach using the formula in the referenced paper. Basically,
zeros are imputed with values proportional to the sequencing depth. When <code>feature.dat</code> is 'proportion',
this parameter will be ignored and zeros will be imputed by half of the minimum for each feature.</p>
</td></tr>
<tr><td><code id="linda_+3A_pseudo.cnt">pseudo.cnt</code></td>
<td>
<p>a positive numeric value for the pseudo-count to be added if <code>zero.handling</code>
is 'pseudo-count'. Default is 0.5. </p>
</td></tr>
<tr><td><code id="linda_+3A_corr.cut">corr.cut</code></td>
<td>
<p>a numerical value between 0 and 1, indicating the significance level used for determining
the zero-handling approach when <code>adaptive</code> is TRUE. Default is 0.1.</p>
</td></tr>
<tr><td><code id="linda_+3A_p.adj.method">p.adj.method</code></td>
<td>
<p>a character string indicating the p-value adjustment approach for 
addressing multiple testing. See R function <code>p.adjust</code>. Default is 'BH'.</p>
</td></tr>
<tr><td><code id="linda_+3A_alpha">alpha</code></td>
<td>
<p>a numerical value between 0 and 1 indicating the significance level 
for declaring differential features. Default is 0.05.</p>
</td></tr>
<tr><td><code id="linda_+3A_n.cores">n.cores</code></td>
<td>
<p>a positive integer. If <code>n.cores &gt; 1</code> and formula is in a form of mixed-effect model,
<code>n.cores</code> parallels will be conducted. Default is 1.</p>
</td></tr>
<tr><td><code id="linda_+3A_verbose">verbose</code></td>
<td>
<p>a logical value indicating whether the trace information should be printed out.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with the elements
</p>
<table>
<tr><td><code>variables</code></td>
<td>
<p>A vector of variable names of all fixed effects in <code>formula</code>. For example: <code>formula = '~x1*x2+x3+(1|id)'</code>.
Suppose <code>x1</code> and <code>x2</code> are numerical, and <code>x3</code> is a categorical variable of three levels: a, b and c.
Then the elements of <code>variables</code> would be <code>('x1', 'x2', 'x3b', 'x3c', 'x1:x2')</code>.</p>
</td></tr>
<tr><td><code>bias</code></td>
<td>
<p>numeric vector; each element corresponds to one variable in <code>variables</code>;
the estimated bias of the regression coefficients due to the compositional effect.</p>
</td></tr>
<tr><td><code>output</code></td>
<td>
<p>a list of data frames with columns 'baseMean', 'log2FoldChange', 'lfcSE', 'stat', 'pvalue', 'padj', 'reject',
'df'; <code>names(output)</code> is equal to <code>variables</code>; the rows of the data frame corresponds to taxa.
Note: if there are taxa being excluded due to <code>prev.cut</code>, the number of the rows of the output data frame
will be not equal to the number of the rows of <code>otu.tab</code>. Taxa are identified by the rownames.
If the rownames of <code>otu.tab</code> are NULL, then <code>1 : nrow(otu.tab)</code> is set as the rownames of <code>otu.tab</code>.
</p>

<dl>
<dt>baseMean:</dt><dd><p> 2 to the power of the intercept coefficients (normalized by one million)</p>
</dd>
<dt>log2FoldChange:</dt><dd><p> bias-corrected coefficients</p>
</dd>
<dt>lfcSE:</dt><dd><p> standard errors of the coefficients</p>
</dd>
<dt>stat:</dt><dd> <p><code>log2FoldChange / lfcSE</code></p>
</dd>
<dt>pvalue:</dt><dd> <p><code>2 * pt(-abs(stat), df)</code></p>
</dd>
<dt>padj:</dt><dd> <p><code>p.adjust(pvalue, method = p.adj.method)</code></p>
</dd>
<dt>reject:</dt><dd> <p><code>padj &lt;= alpha</code></p>
</dd>
<dt>df:</dt><dd><p> degrees of freedom. The number of samples minus the number of explanatory variables (intercept included) for
fixed-effect models; estimates from R package <code>lmerTest</code> with Satterthwaite method of approximation for mixed-effect models.</p>
</dd>
</dl>

</td></tr>
<tr><td><code>covariance</code></td>
<td>
<p>a list of data frames; the data frame records the covariances between a regression coefficient with other coefficients;
<code>names(covariance)</code> is equal to <code>variables</code>; the rows of the data frame corresponds to taxa. If the length of <code>variables</code>
is equal to 1, then the <code>covariance</code> is NULL.</p>
</td></tr>
<tr><td><code>otu.tab.use</code></td>
<td>
<p>the OTU table used in the abundance analysis (the <code>otu.tab</code> after the preprocessing:
samples that have NAs in the variables in <code>formula</code> or have less than <code>lib.cut</code> read counts are removed;
taxa with prevalence less than <code>prev.cut</code> are removed and data is winsorized if <code>!is.null(winsor.quan)</code>;
and zeros are treated, i.e., imputed or pseudo-count added).</p>
</td></tr>
<tr><td><code>meta.use</code></td>
<td>
<p>the meta data used in the abundance analysis (only variables in <code>formula</code> are stored; samples that have NAs
or have less than <code>lib.cut</code> read counts are removed; numerical variables are scaled).</p>
</td></tr>
<tr><td><code>wald</code></td>
<td>
<p>a list for use in Wald test. If the fitting model is a linear model, then it includes
</p>

<dl>
<dt>beta:</dt><dd><p> a matrix of the biased regression coefficients including intercept and all fixed effects; the culumns correspond to taxa</p>
</dd>
<dt>sig:</dt><dd><p> the standard errors; the elements corresponding to taxa</p>
</dd>
<dt>X:</dt><dd><p> the design matrix of the fitting model</p>
</dd>
<dt>bias:</dt><dd><p> the estimated biases of the regression coefficients including intercept and all fixed effects</p>
</dd>
</dl>

<p>If the fitting model is a linear mixed-effect model, then it includes
</p>

<dl>
<dt>beta:</dt><dd><p> a matrix of the biased regression coefficients including intercept and all fixed effects; the culumns correspond to taxa</p>
</dd>
<dt>beta.cov:</dt><dd><p> a list of covariance matrices of <code>beta</code>; the elements corresponding to taxa</p>
</dd>
<dt>rand.cov:</dt><dd><p> a list with covariance matrices of variance parameters of random effects; the elements corresponding to taxa; see more details in the paper of 'lmerTest'</p>
</dd>
<dt>Joc.beta.cov.rand:</dt><dd><p> a list of a list of Jacobian matrices of <code>beta.cov</code> with respect to the variance parameters; the elements corresponding to taxa</p>
</dd>
<dt>bias:</dt><dd><p> the estimated biases of the regression coefficients including intercept and all fixed effects</p>
</dd>
</dl>

</td></tr>
</table>


<h3>Author(s)</h3>

<p>Huijuan Zhou,
Jun Chen,
Xianyang Zhang
</p>


<h3>References</h3>

<p>Zhou, H., He, K., Chen, J., Zhang, X. (2022). LinDA: linear models for differential abundance analysis of microbiome compositional data. Genome biology, 23(1), 95.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(smokers)

ind &lt;- smokers$meta$AIRWAYSITE == 'Throat'
otu.tab &lt;- as.data.frame(smokers$otu[, ind])
depth &lt;- colSums(otu.tab)
meta &lt;- cbind.data.frame(Smoke = factor(smokers$meta$SMOKER[ind]),
                         Sex = factor(smokers$meta$SEX[ind]),
                         Site = factor(smokers$meta$SIDEOFBODY[ind]),
                         SubjectID = factor(smokers$meta$HOST_SUBJECT_ID[ind]))

# Differential abundance analysis using the left throat data                       
ind1 &lt;- meta$Site == 'Left' &amp; depth &gt;= 1000
linda.obj  &lt;- linda(otu.tab[, ind1], meta[ind1, ], formula = '~Smoke+Sex', 
           feature.dat.type = 'count', 
           prev.filter = 0.1, is.winsor = TRUE, outlier.pct = 0.03,
           p.adj.method = "BH", alpha = 0.1)
           
           


linda.plot(linda.obj, c('Smokey', 'Sexmale'),
           titles = c('Smoke: n v.s. y', 'Sex: female v.s. male'), 
           alpha = 0.1, lfc.cut = 1, legend = TRUE, directory = NULL,
            width = 11, height = 8)
            
rownames(linda.obj $output[[1]])[which(linda.obj $output[[1]]$reject)]


# Differential abundance analysis pooling both the left and right throat data 
# Mixed effects model is used 

ind  &lt;- depth &gt;= 1000
linda.obj &lt;- linda(otu.tab[, ind], meta[ind, ], formula = '~Smoke+Sex+(1|SubjectID)',
           feature.dat.type = 'count', 
           prev.filter = 0.1, is.winsor = TRUE, outlier.pct = 0.03,
           p.adj.method = "BH", alpha = 0.1)

       
    
# For proportion data   
otu.tab.p &lt;- t(t(otu.tab) / colSums(otu.tab))
ind1 &lt;- meta$Site == 'Left' &amp; depth &gt;= 1000
lind.obj  &lt;- linda(otu.tab[, ind1], meta[ind1, ], formula = '~Smoke+Sex', 
           feature.dat.type = 'proportion', 
           prev.filter = 0.1, is.winsor = TRUE, outlier.pct = 0.03,
           p.adj.method = "BH", alpha = 0.1)

</code></pre>

<hr>
<h2 id='linda.plot'>Plot LinDA Results</h2><span id='topic+linda.plot'></span>

<h3>Description</h3>

<p>The function produces the effect size plot of the differential features and volcano plot based on the output from <code>linda</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linda.plot(
  linda.obj,
  variables.plot,
  titles = NULL,
  alpha = 0.05,
  lfc.cut = 1,
  legend = FALSE,
  directory = NULL,
  width = 11,
  height = 8
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linda.plot_+3A_linda.obj">linda.obj</code></td>
<td>
<p>return from function <code>linda</code>.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_variables.plot">variables.plot</code></td>
<td>
<p>vector; variables whose results are to be plotted. For example, suppose the return
value <code>variables</code> is equal to <code>('x1', 'x2', 'x3b', 'x3c', 'x1:x2')</code>, then one could set <code>variables.plot = c('x3b', 'x1:x2')</code>.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_titles">titles</code></td>
<td>
<p>vector; titles of the effect size plot and volcano plot for each variable in <code>variables.plot</code>.
Default is NULL. If NULL, the titles will be set as <code>variables.plot</code>.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_alpha">alpha</code></td>
<td>
<p>a numerical value between 0 and 1; cutoff for <code>padj</code>.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_lfc.cut">lfc.cut</code></td>
<td>
<p>a positive numerical value; cutoff for <code>log2FoldChange</code>.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_legend">legend</code></td>
<td>
<p>TRUE or FALSE; whether to show the legends of the effect size plot and volcano plot.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_directory">directory</code></td>
<td>
<p>character; the directory to save the figures, e.g., <code>getwd()</code>. Default is NULL. If NULL, figures will not be saved.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_width">width</code></td>
<td>
<p>the width of the graphics region in inches. See R function <code>pdf</code>.</p>
</td></tr>
<tr><td><code id="linda.plot_+3A_height">height</code></td>
<td>
<p>the height of the graphics region in inches. See R function <code>pdf</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of <code>ggplot2</code> objects.
</p>
<table>
<tr><td><code>plot.lfc</code></td>
<td>
<p>a list of effect size plots. Each plot corresponds to one variable in <code>variables.plot</code>.</p>
</td></tr>
<tr><td><code>plot.volcano</code></td>
<td>
<p>a list of volcano plots. Each plot corresponds to one variable in <code>variables.plot</code>.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Huijuan Zhou,
Jun Chen,
Xianyang Zhang
</p>


<h3>References</h3>

<p>Zhou, H., He, K., Chen, J., Zhang, X. (2022). LinDA: linear models for differential abundance analysis of microbiome compositional data. Genome biology, 23(1), 95.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(smokers)
ind &lt;- smokers$meta$AIRWAYSITE == 'Throat' &amp; smokers$meta$SIDEOFBODY == 'Left'
otu.tab &lt;- as.data.frame(smokers$otu[, ind])
depth &lt;- colSums(otu.tab)
meta &lt;- cbind.data.frame(Smoke = factor(smokers$meta$SMOKER[ind]),
                         Sex = factor(smokers$meta$SEX[ind]))
                         
ind  &lt;- depth &gt;= 1000
linda.obj &lt;- linda(otu.tab[, ind], meta[ind, ], formula = '~Smoke+Sex',
           feature.dat.type = 'count', 
           prev.filter = 0.1, is.winsor = TRUE, outlier.pct = 0.03,
           p.adj.method = "BH", alpha = 0.1)
           
linda.plot(linda.obj, c('Smokey', 'Sexmale'),
           titles = c('Smoke: n v.s. y', 'Sex: female v.s. male'), 
           alpha = 0.1, lfc.cut = 1, legend = TRUE, directory = NULL,
            width = 11, height = 8)
</code></pre>

<hr>
<h2 id='linda.wald.test'>Wald test for bias-corrected regression coefficients</h2><span id='topic+linda.wald.test'></span>

<h3>Description</h3>

<p>The function implements Wald test for bias-corrected regression coefficients generated from the <code>linda</code> function.
One can utilize the function to perform ANOVA-type analyses. For example, if you have a cateogrical variable with three levels, you can test whether all levels have the same effect.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>linda.wald.test(
  linda.obj,
  L,
  model = c("LM", "LMM"),
  alpha = 0.05,
  p.adj.method = "BH"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="linda.wald.test_+3A_linda.obj">linda.obj</code></td>
<td>
<p>return from the <code>linda</code> function.</p>
</td></tr>
<tr><td><code id="linda.wald.test_+3A_l">L</code></td>
<td>
<p>A matrix for testing <code>Lb = 0</code>, where <code>b</code> includes the intercept and all fixed effects from runing <code>linda</code>. Thus the number of columns of L must be equal to <code>length(variables)+1</code>, where <code>variables</code> is from <code>linda.obj</code>, which does not include the intercept.</p>
</td></tr>
<tr><td><code id="linda.wald.test_+3A_model">model</code></td>
<td>
<p><code>'LM'</code> or <code>'LMM'</code> indicating the model fitted in {linda} is linear model or linear mixed-effect model.</p>
</td></tr>
<tr><td><code id="linda.wald.test_+3A_alpha">alpha</code></td>
<td>
<p>significance level for testing <code>Lb = 0</code>.</p>
</td></tr>
<tr><td><code id="linda.wald.test_+3A_p.adj.method">p.adj.method</code></td>
<td>
<p>P-value adjustment approach. See R function <code>p.adjust</code>. The default is 'BH'.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns
</p>
<table>
<tr><td><code>Fstat</code></td>
<td>
<p>Wald statistics for each taxon</p>
</td></tr>
<tr><td><code>df1</code></td>
<td>
<p>The numerator degrees of freedom</p>
</td></tr>
<tr><td><code>df2</code></td>
<td>
<p>The denominator degrees of freedom</p>
</td></tr>
<tr><td><code>pvalue</code></td>
<td>
 <p><code>1 - pf(Fstat, df1, df2)</code></p>
</td></tr>
<tr><td><code>padj</code></td>
<td>
 <p><code>p.adjust(pvalue, method = p.adj.method)</code></p>
</td></tr>
<tr><td><code>reject</code></td>
<td>
 <p><code>padj &lt;= alpha</code></p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Huijuan Zhou <a href="mailto:huijuanzhou2019@gmail.com">huijuanzhou2019@gmail.com</a>
Jun Chen <a href="mailto:Chen.Jun2@mayo.edu">Chen.Jun2@mayo.edu</a>
Xianyang Zhang <a href="mailto:zhangxiany@stat.tamu.edu">zhangxiany@stat.tamu.edu</a>
</p>


<h3>References</h3>

<p>Zhou, H., He, K., Chen, J., Zhang, X. (2022). LinDA: linear models for differential abundance analysis of microbiome compositional data. Genome biology, 23(1), 95.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data(smokers)

ind &lt;- smokers$meta$AIRWAYSITE == 'Throat'
otu.tab &lt;- as.data.frame(smokers$otu[, ind])
depth &lt;- colSums(otu.tab)
meta &lt;- cbind.data.frame(Smoke = factor(smokers$meta$SMOKER[ind]),
                         Sex = factor(smokers$meta$SEX[ind]),
                         Site = factor(smokers$meta$SIDEOFBODY[ind]),
                         SubjectID = factor(smokers$meta$HOST_SUBJECT_ID[ind]))
ind  &lt;- depth &gt;= 1000
linda.obj &lt;- linda(otu.tab[, ind], meta[ind, ], formula = '~Smoke+Sex+(1|SubjectID)',
           feature.dat.type = 'count', 
           prev.filter = 0.1, is.winsor = TRUE, outlier.pct = 0.03,
           p.adj.method = "BH", alpha = 0.1)
#  L matrix (2x3) is designed to test the second (Smoke) and the third (Sex) coefficient to be 0.
# For a categorical variable &gt; two levels, similar L can be designed to do ANOVA-type test. 
L &lt;- matrix(c(0, 1, 0, 0, 0, 1), nrow = 2, byrow = TRUE)
result &lt;- linda.wald.test(linda.obj, L, 'LMM', alpha = 0.1)


</code></pre>

<hr>
<h2 id='smokers'>Microbiome data from the human upper respiratory tract (left and right throat)</h2><span id='topic+smokers'></span>

<h3>Description</h3>

<p>A dataset containing &quot;otu&quot;, &quot;tax&quot;, meta&quot;, &quot;genus&quot;, family&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(smokers)
</code></pre>


<h3>Format</h3>

<p>A  list with elements
</p>

<dl>
<dt>otu</dt><dd><p>otu table, 2156 taxa by 290 samples</p>
</dd>
<dt>tax</dt><dd><p>taxonomy table, 2156 taxa by 7 taxonomic ranks</p>
</dd>
<dt>meta</dt><dd><p>meta table, 290 samples by 53 sample variables</p>
</dd>
<dt>genus</dt><dd><p>304 by 290</p>
</dd>
<dt>family</dt><dd><p>113 by 290</p>
</dd>
</dl>



<h3>Source</h3>

<p><a href="https://qiita.ucsd.edu/">https://qiita.ucsd.edu/</a> study ID:524
Reference: Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
