<!DOCTYPE html><html lang="en"><head><title>Help for package microclass</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {microclass}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#microclass-package'><p>Methods for 16S based taxonomic classification of prokaryotes</p></a></li>
<li><a href='#blastClassify16S'><p>Classifying using BLAST</p></a></li>
<li><a href='#blastDbase16S'><p>Building a BLAST database</p></a></li>
<li><a href='#KmerCount'><p>K-mer counting</p></a></li>
<li><a href='#multinomClassify'><p>Classifying with a Multinomial model</p></a></li>
<li><a href='#multinomTrain'><p>Training multinomial model</p></a></li>
<li><a href='#rdpClassify'><p>Classifying with the RDP classifier</p></a></li>
<li><a href='#rdpTrain'><p>Training the RDP classifier</p></a></li>
<li><a href='#setParallel'><p>Set number of parallel threads</p></a></li>
<li><a href='#small.16S'><p>A small example data set</p></a></li>
<li><a href='#taxMachine'><p>Classifying 16S sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Methods for Taxonomic Classification of Prokaryotes</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2020-08-11</td>
</tr>
<tr>
<td>Author:</td>
<td>Kristian Hovde Liland, Hilde Vinje, Lars Snipen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Lars Snipen &lt;lars.snipen@nmbu.no&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Functions for assigning 16S sequence data to a
    taxonomic level in the tree-of-life for prokaryotes.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), microseq, microcontax, dplyr, stringr, rlang</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.11.1), RcppParallel, tibble</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp (&ge; 0.11.1), RcppEigen, RcppParallel</td>
</tr>
<tr>
<td>SystemRequirements:</td>
<td>GNU make</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-08-11 14:29:47 UTC; larssn</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-08-28 12:20:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='microclass-package'>Methods for 16S based taxonomic classification of prokaryotes</h2><span id='topic+microclass-package'></span><span id='topic+microclass'></span>

<h3>Description</h3>

<p>The package provides functions for assigning 16S sequence data to a
taxonomic level in the tree-of-life for prokaryotes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>microclass()
</code></pre>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> microclass</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-08-31</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Hilde Vinje, Kristian Hovde Liland, Lars Snipen.<br />
Maintainer: Lars Snipen &lt;lars.snipen@nmbu.no&gt;
</p>

<hr>
<h2 id='blastClassify16S'>Classifying using BLAST</h2><span id='topic+blastClassify16S'></span>

<h3>Description</h3>

<p>A 16S based classification based on BLAST.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blastClassify16S(sequence, bdb)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="blastClassify16S_+3A_sequence">sequence</code></td>
<td>
<p>Character vector of 16S sequences to classify.</p>
</td></tr>
<tr><td><code id="blastClassify16S_+3A_bdb">bdb</code></td>
<td>
<p>Name of BLAST data base, see <code><a href="#topic+blastDbase16S">blastDbase16S</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A vector of 16S sequences (DNA) are classified by first using BLAST <code>blastn</code> against
a database of 16S DNA sequences, and then classify according to the nearest-neighbour principle.
The nearest neighbour of a query sequence is the hit with the largest bitscore. The blast+
software  <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download</a>
must be installed on the system. Type <code>system("blastn -help")</code> in the Console window,
and a sensible Help-text should appear.
</p>
<p>The database must contain 16S sequences where the Header starts with a token specifying the taxon. 
More specifically, the tokens must look like:
</p>
<p>&lt;taxon&gt;_1
</p>
<p>&lt;taxon&gt;_2
</p>
<p>...etc
</p>
<p>where &lt;taxon&gt; is some proper taxon name. Use <code><a href="#topic+blastDbase16S">blastDbase16S</a></code> to make such databases.
</p>
<p>The identity of each alignment is also computed. This should be close to 1.0 for a classification
to be trusted. Identity values below 0.95 could indicate uncertain classifications, but this will
vary between taxa.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with two columns: Taxon is the predicted taxon for each <code>sequence</code>
and Identity is the corresponding identity-value. If no BLAST hit is seen, the sequence is 
<code>"unclassified"</code>.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+blastDbase16S">blastDbase16S</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("small.16S")
## Not run: 
dbase &lt;- blastDbase16S("test", small.16S$Sequence, word(small.16S$Header, 2, 2))
reads &lt;- str_sub(small.16S$Sequence, 100, 550)
blastClassify16S(reads, dbase) %&gt;% 
  bind_cols(small.16S) -&gt; tbl

## End(Not run)

</code></pre>

<hr>
<h2 id='blastDbase16S'>Building a BLAST database</h2><span id='topic+blastDbase16S'></span>

<h3>Description</h3>

<p>Building a BLAST database for 16S based classification.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blastDbase16S(name, sequence, taxon)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="blastDbase16S_+3A_name">name</code></td>
<td>
<p>The name of the database (text).</p>
</td></tr>
<tr><td><code id="blastDbase16S_+3A_sequence">sequence</code></td>
<td>
<p>A character vector with 16S sequence data.</p>
</td></tr>
<tr><td><code id="blastDbase16S_+3A_taxon">taxon</code></td>
<td>
<p>A character vector with taxon information.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This functions builds a database using the <code>makeblastdb</code> program of the
BLAST+ software <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download">https://blast.ncbi.nlm.nih.gov/Blast.cgi?PAGE_TYPE=BlastDocs&amp;DOC_TYPE=Download</a>.
Thus, this software must be available on the system when using this
function. If you type <code>system("makeblastdb -help")</code> in the Console window some meaningful
Help-text should be displayed.
</p>
<p>This function is most typically used prior to <code><a href="#topic+blastClassify16S">blastClassify16S</a></code> to set up the database
before searching and classifying. It can be seen as the 'training step' of a BLAST-based
classification procedure.
</p>
<p>The <code>sequence</code> must be a vector of DNA-sequences (16S sequences). The <code>taxon</code> is a vector of the same 
length as <code>sequence</code>, containing the correpsonding taxon information.
</p>


<h3>Value</h3>

<p>The database files are created, and the name of the database (<code>name</code>) is returned.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+blastClassify16S">blastClassify16S</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples for blastClassify16S.

</code></pre>

<hr>
<h2 id='KmerCount'>K-mer counting</h2><span id='topic+KmerCount'></span>

<h3>Description</h3>

<p>Counting overlapping words of length K in DNA/RNA sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KmerCount(sequences, K = 1, col.names = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="KmerCount_+3A_sequences">sequences</code></td>
<td>
<p>Vector of sequences (text).</p>
</td></tr>
<tr><td><code id="KmerCount_+3A_k">K</code></td>
<td>
<p>Word length (integer).</p>
</td></tr>
<tr><td><code id="KmerCount_+3A_col.names">col.names</code></td>
<td>
<p>Logical indicating if the words should be added as columns names.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For each input sequence, the frequency of every word of length <code>K</code> is counted. 
Counting is done with overlap. The counting itself is done by a C++ function.
</p>
<p>With <code>col.names=TRUE</code> the K-mers are added as column names, but this makes the
computations slower.
</p>


<h3>Value</h3>

<p>A matrix with one row for each sequence in <code>sequences</code> and one column for 
each possible word of length<code>K</code>.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+multinomTrain">multinomTrain</a></code>, <code><a href="#topic+multinomClassify">multinomClassify</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>KmerCount("ATGCCTGAACTGACCTGC",K=2)

</code></pre>

<hr>
<h2 id='multinomClassify'>Classifying with a Multinomial model</h2><span id='topic+multinomClassify'></span>

<h3>Description</h3>

<p>Classifying sequences by a trained Multinomial model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multinomClassify(sequence, trained.model, post.prob = FALSE, prior = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multinomClassify_+3A_sequence">sequence</code></td>
<td>
<p>Character vector of 16S sequences to classify.</p>
</td></tr>
<tr><td><code id="multinomClassify_+3A_trained.model">trained.model</code></td>
<td>
<p>A list with a trained model, see <code><a href="#topic+multinomTrain">multinomTrain</a></code>.</p>
</td></tr>
<tr><td><code id="multinomClassify_+3A_post.prob">post.prob</code></td>
<td>
<p>Logical indicating if posterior log-probabilities should be returned.</p>
</td></tr>
<tr><td><code id="multinomClassify_+3A_prior">prior</code></td>
<td>
<p>Logical indicating if classification should be done by flat priors (default)
or with empirical priors (prior=TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The classification step of the Multinomial method (Vinje et al, 2015) means counting 
K-mers on all sequences, and computing the posterior probabilities for each taxon in the trained model.
The predicted taxon for each input sequence is the one with the maximum posterior probability for
that sequence.
</p>
<p>By setting <code>post.prob=TRUE</code> you will get the log-probability of the best and second best taxon
for each sequence. This can be used for evaluating the certainty in the classifications,
see <code><a href="#topic+taxMachine">taxMachine</a></code>.
</p>
<p>The classification is parallelized through RcppParallel
employing Intel TBB and TinyThread. By default all available
processing cores are used. This can be changed using the
function <code><a href="#topic+setParallel">setParallel</a></code>.
</p>


<h3>Value</h3>

<p>If <code>post.prob=FALSE</code> a character vector of predicted taxa is returned.
</p>
<p>If <code>post.prob=TRUE</code> a <code>data.frame</code> with three columns is returned. Taxon
is the vector of predicted taxa, one for each sequence in <code>sequence</code>. The
Post.prob.1 and Post.prob.2 are vectors with the maximum and second largest posterior
log-probabilities for each sequence.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Vinje, H, Liland, KH, Almøy, T, Snipen, L. (2015). Comparing K-mer based methods for
improved classification of 16S sequences. BMC Bioinformatics, 16:205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+KmerCount">KmerCount</a></code>, <code><a href="#topic+multinomTrain">multinomTrain</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("small.16S")
seq &lt;- small.16S$Sequence
tax &lt;- sapply(strsplit(small.16S$Header,split=" "),function(x){x[2]})
## Not run: 
trn &lt;- multinomTrain(seq,tax)
primer.515f &lt;- "GTGYCAGCMGCCGCGGTAA"
primer.806rB &lt;- "GGACTACNVGGGTWTCTAAT"
reads &lt;- amplicon(seq, primer.515f, primer.806rB)
predicted &lt;- multinomClassify(unlist(reads[nchar(reads)&gt;0]),trn)
print(predicted)

## End(Not run)

</code></pre>

<hr>
<h2 id='multinomTrain'>Training multinomial model</h2><span id='topic+multinomTrain'></span>

<h3>Description</h3>

<p>Training the multinomial K-mer method on sequence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multinomTrain(sequence, taxon, K = 8, col.names = FALSE, n.pseudo = 100)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="multinomTrain_+3A_sequence">sequence</code></td>
<td>
<p>Character vector of 16S sequences.</p>
</td></tr>
<tr><td><code id="multinomTrain_+3A_taxon">taxon</code></td>
<td>
<p>Character vector of taxon labels for each sequence.</p>
</td></tr>
<tr><td><code id="multinomTrain_+3A_k">K</code></td>
<td>
<p>Word length (integer).</p>
</td></tr>
<tr><td><code id="multinomTrain_+3A_col.names">col.names</code></td>
<td>
<p>Logical indicating if column names should be added to the trained model matrix.</p>
</td></tr>
<tr><td><code id="multinomTrain_+3A_n.pseudo">n.pseudo</code></td>
<td>
<p>Number of pseudo-counts to use (positive numerics, need not be integer). Special case -1
will only return word counts, not log-probabilities.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The training step of the multinomial method (Vinje et al, 2015) means counting K-mers
on all sequences and compute the multinomial probabilities for each K-mer for each unique taxon. 
<code>n.pseudo</code> pseudo-counts are added, divided equally over all K-mers, before probabilities
are estimated. The optimal choice of <code>n.pseudo</code> will depend on <code>K</code> and the 
training data set. The default value <code>n.pseudo=100</code> has proven good for <code>K=8</code> and the
<code><a href="microcontax.html#topic+contax.trim">contax.trim</a></code> data set (see the <code>microcontax</code> R-package).
</p>
<p>Adding the actual K-mers as column names (<code>col.names=TRUE</code>) will slow down the
computations.
</p>
<p>The relative taxon sizes are also computed, and may be used as an empirical prior in the
classification step (see &quot;prior&quot; below).
</p>


<h3>Value</h3>

<p>A list with two elements. The first element is <code>Method</code>, which is the text 
<code>"multinom"</code> in this case. The second element is <code>Fitted</code>, which is a matrix
of probabilities with one row for each unique <code>taxon</code> and one column for each possible word of
length<code>K</code>. The sum of each row is 1.0. No probabilities are 0 if <code>n.pseudo</code>&gt;0.0.
</p>
<p>The matrix <code>Fitted</code> has an attribute <code>attr("prior",)</code>, that contains the relative
taxon sizes.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Vinje, H, Liland, KH, Almøy, T, Snipen, L. (2015). Comparing K-mer based methods for
improved classification of 16S sequences. BMC Bioinformatics, 16:205.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+KmerCount">KmerCount</a></code>, <code><a href="#topic+multinomClassify">multinomClassify</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples for multinomClassify

</code></pre>

<hr>
<h2 id='rdpClassify'>Classifying with the RDP classifier</h2><span id='topic+rdpClassify'></span>

<h3>Description</h3>

<p>Classifying sequences by a trained presence/absence K-mer model.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdpClassify(sequence, trained.model, post.prob = FALSE, prior = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rdpClassify_+3A_sequence">sequence</code></td>
<td>
<p>Character vector of sequences to classify.</p>
</td></tr>
<tr><td><code id="rdpClassify_+3A_trained.model">trained.model</code></td>
<td>
<p>A list with a trained model, see <code><a href="#topic+rdpTrain">rdpTrain</a></code>.</p>
</td></tr>
<tr><td><code id="rdpClassify_+3A_post.prob">post.prob</code></td>
<td>
<p>Logical indicating if posterior log-probabilities should be returned.</p>
</td></tr>
<tr><td><code id="rdpClassify_+3A_prior">prior</code></td>
<td>
<p>Logical indicating if classification should be done by flat priors (default)
or with empirical priors (prior=TRUE).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The classification step of the presence/absence method known as the RDP classifier
(Wang et al 2007) means looking for K-mers on all sequences, and computing the posterior
probabilities for each taxon using a trained model and a naive Bayes assumption. The predicted
taxon is the one producing the maximum posterior probability, for each <code>sequence</code>.
</p>
<p>The classification is parallelized through RcppParallel
employing Intel TBB and TinyThread. By default all available
processing cores are used. This can be changed using the
function <code><a href="#topic+setParallel">setParallel</a></code>.
</p>


<h3>Value</h3>

<p>A character vector with the predicted taxa, one for each <code>sequence</code>.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Wang, Q, Garrity, GM, Tiedje, JM, Cole, JR (2007). Naive Bayesian Classifier for 
Rapid Assignment of rRNA Sequences into the New Bacterial Taxonomy. Applied and Enviromental 
Microbiology, 73: 5261-5267.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rdpTrain">rdpTrain</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("small.16S")
seq &lt;- small.16S$Sequence
tax &lt;- sapply(strsplit(small.16S$Header,split=" "),function(x){x[2]})
## Not run: 
trn &lt;- rdpTrain(seq,tax)
primer.515f &lt;- "GTGYCAGCMGCCGCGGTAA"
primer.806rB &lt;- "GGACTACNVGGGTWTCTAAT"
reads &lt;- amplicon(seq, primer.515f, primer.806rB)
predicted &lt;- rdpClassify(unlist(reads[nchar(reads)&gt;0]),trn)
print(predicted)

## End(Not run)

</code></pre>

<hr>
<h2 id='rdpTrain'>Training the RDP classifier</h2><span id='topic+rdpTrain'></span>

<h3>Description</h3>

<p>Training the RDP presence/absence K-mer method on sequence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdpTrain(sequence, taxon, K = 8, cnames = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rdpTrain_+3A_sequence">sequence</code></td>
<td>
<p>Character vector of 16S sequences.</p>
</td></tr>
<tr><td><code id="rdpTrain_+3A_taxon">taxon</code></td>
<td>
<p>Character vector of taxon labels for each sequence.</p>
</td></tr>
<tr><td><code id="rdpTrain_+3A_k">K</code></td>
<td>
<p>Word length (integer).</p>
</td></tr>
<tr><td><code id="rdpTrain_+3A_cnames">cnames</code></td>
<td>
<p>Logical indicating if column names should be added to the trained model matrix.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The training step of the RDP method means looking for K-mers on all sequences,
and computing the probability of each K-mer being present for each unique taxon. This is an
attempt to re-implement the method described by Wang et tal (2007), but without the bootstrapping. 
See that publications for all details.
</p>
<p>The word-length <code>K</code> is by default 8, since this is the value used by Wang et al. Larger values
may lead to memory-problems since the trained model is a matrix with 4^K columns. Adding the K-mers 
as column names will slow down all computations.
</p>
<p>The relative taxon sizes are also computed, and returned as an attribute to the model matrix. They may 
be used as empirical priors in the classification step.
</p>


<h3>Value</h3>

<p>A list with two elements. The first element is <code>Method</code>, which is the text 
<code>"RDPclassifier"</code> in this case. The second element is <code>Fitted</code>, which is a 
matrix with one row for each unique <code>taxon</code> and one column for 
each possible word of length <code>K</code>. The value in row i and column j is the probability that
word j is present in taxon i.
</p>


<h3>Author(s)</h3>

<p>Kristian Hovde Liland and Lars Snipen.
</p>


<h3>References</h3>

<p>Wang, Q, Garrity, GM, Tiedje, JM, Cole, JR (2007). Naive Bayesian Classifier for 
Rapid Assignment of rRNA Sequences into the New Bacterial Taxonomy. Applied and Enviromental 
Microbiology, 73: 5261-5267.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rdpClassify">rdpClassify</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'># See examples for rdpClassify.

</code></pre>

<hr>
<h2 id='setParallel'>Set number of parallel threads</h2><span id='topic+setParallel'></span>

<h3>Description</h3>

<p>Simple function to set the number of threads to use in parallel
computations. The default equals all available logical cores. An integer is
interpreted as the number of threads. A numeric &lt; 1 is interpreted as a proportion
of the avialable logical cores.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>setParallel(C = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="setParallel_+3A_c">C</code></td>
<td>
<p>a scalar indicating the number of threads, default = NULL (#available logical cores)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>NULL, returned silently.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
setParallel() # Use all available logical cores.

## End(Not run)

</code></pre>

<hr>
<h2 id='small.16S'>A small example data set</h2><span id='topic+small.16S'></span>

<h3>Description</h3>

<p>A <code>tibble</code> object (<code>data.frame</code>) with some 16S sequences with taxon information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(small.16S)
</code></pre>


<h3>Details</h3>

<p>This is a <code>tibble</code> object (<code>data.frame</code>) with 71 sequences used in some examples.
The taxonomy information for each sequence follows the ConTax format, see the 
<code>microcontax</code> package for more details.
</p>


<h3>Author(s)</h3>

<p>Hilde Vinje, Kristian Hovde Liland, Lars Snipen.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(small.16S)
str(small.16S)

</code></pre>

<hr>
<h2 id='taxMachine'>Classifying 16S sequences</h2><span id='topic+taxMachine'></span>

<h3>Description</h3>

<p>Optimized classification of 16S sequence data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxMachine(
  sequence,
  model.in.memory = TRUE,
  model.on.disk = FALSE,
  verbose = TRUE,
  chunk.size = 10000
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="taxMachine_+3A_sequence">sequence</code></td>
<td>
<p>Character vector with DNA sequences.</p>
</td></tr>
<tr><td><code id="taxMachine_+3A_model.in.memory">model.in.memory</code></td>
<td>
<p>Logical indicating if model should be cached in memory (default=TRUE).</p>
</td></tr>
<tr><td><code id="taxMachine_+3A_model.on.disk">model.on.disk</code></td>
<td>
<p>Logical or text, for reading/saving models, see Deatils below (default=FALSE).</p>
</td></tr>
<tr><td><code id="taxMachine_+3A_verbose">verbose</code></td>
<td>
<p>Logical, if <code>TRUE</code> progress is reported during computations (default=TRUE).</p>
</td></tr>
<tr><td><code id="taxMachine_+3A_chunk.size">chunk.size</code></td>
<td>
<p>The number of sequence to classify in each iteration of the loop (default=10000).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function provides optimized taxonomy classifications from 16S sequence data.
</p>
<p>All sequences are classified to the genus level based on a Multinomial model (see <code><a href="#topic+multinomTrain">multinomTrain</a></code>)
trained on the designed consensus taxonomy data set <code><a href="microcontax.html#topic+contax.trim">contax.trim</a></code> found in
the R-package <code>microcontax</code>. The word length K=8 has been used in the model.
</p>
<p>To avoid saving fitted models in the package, a model is trained the first time you run <code>taxMachine</code> in an R session.
This takes only a few seconds, and the result is cached for latter use if <code>model.in.memory</code> is <code>TRUE</code>.
</p>
<p>If a path to an existing file with a trained model is supplied in <code>model.on.disk</code>, this Multinomial model is read
from the file and used. If a path to a new file is supplied, the trained Multinomial model will be saved to that file.
The default (<code>model.on.disk=FALSE</code>), means no files are read/saved, while <code>model.on.disk=TRUE</code> will attempt to load/save models from the
<code>microclass/extdata</code> directory.
</p>
<p>Both <code>verbose</code> and <code>chunk.size</code> are used to monitor the progress, which is nice when classifying huge data sets,
since this will take some time.
</p>


<h3>Value</h3>

<p>A <code>data.frame</code> with one row for each sequence. The columns are Genus, D.score, R.score and P.recognize.
</p>
<p>Genus is the predicted genus for each sequence. Note that all sequences get a prediction, but may still be
more or less reliable.
</p>
<p>The D.score is a measure of how the predicted genus wins over all other genera in the race for being the chosen one.
A large D.score means the winner stands out clearly, and we can be confident it is the correct genus. A D.score close to 0
means we have an uncertain classification. Only D.scores below 1.0, should be of any concern, see Liland et al
(2016) for details.
</p>
<p>The R.score is a measure of the models ability to recognize the sequence. The more negative the R.score gets, the more
unusual the sequence is compared to the training set (the <code><a href="microcontax.html#topic+contax.trim">contax.trim</a></code> data set). The P.recognize
is a rough probability of seing an R.score this small, or smaller, given the training data. Thus, a very small P.recognize means
the sequence is not really recognized, and the classification is worthless. A very negative R.score indicates either not 16S at all,
many sequencing errors that has destroyed the read, or a completely new taxon never seen before. See Liland et al (2016) for
details.
</p>


<h3>Author(s)</h3>

<p>Lars Snipen and Kristian Hovde Liland
</p>


<h3>References</h3>

<p>Liland, KH, Vinje, H, Snipen, L (2016). <code>microclass</code> - An 
R-package for 16S taxonomy classification. BMC Bioinformatics, xx:yy.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+KmerCount">KmerCount</a></code>, <code><a href="#topic+multinomClassify">multinomClassify</a></code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(small.16S)
tax.tab &lt;- taxMachine(small.16S$Sequence)

## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
