<!DOCTYPE html><html lang="en"><head><title>Help for package miLineage</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {miLineage}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#miLineage-package'>
<p>Association Tests for Microbial Lineages on a Taxonomic Tree</p></a></li>
<li><a href='#data.real'>
<p>Real data that include an OTU table, a taxonomy table, and a covariate table</p></a></li>
<li><a href='#data.toy'>
<p>Simulated data that include an OTU table, a taxonomy table, and a covariate table</p></a></li>
<li><a href='#QCAT'>
<p>Quasi-Conditional Association Test (QCAT)</p></a></li>
<li><a href='#QCAT_GEE'>
<p>QCAT+GEE Two-Part Test</p></a></li>
<li><a href='#ZIGDM'>
<p>Zero-Inflated Generalized Dirichlet Multinomial (ZIGDM) Tests</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Association Tests for Microbial Lineages on a Taxonomic Tree</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-03-09</td>
</tr>
<tr>
<td>Author:</td>
<td>Zheng-Zheng Tang</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Zheng-Zheng Tang &lt;tang@biostat.wisc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A variety of association tests for microbiome data analysis including Quasi-Conditional Association Tests (QCAT) described in Tang Z.-Z. et al.(2017) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtw804">doi:10.1093/bioinformatics/btw804</a>&gt; and Zero-Inflated Generalized Dirichlet Multinomial (ZIGDM) tests described in Tang Z.-Z. &amp; Chen G. (2017, submitted).</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.1), MASS, data.table, geepack</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-03-10 17:31:42 UTC; wzhou87</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-03-23 00:01:09 UTC</td>
</tr>
</table>
<hr>
<h2 id='miLineage-package'>
Association Tests for Microbial Lineages on a Taxonomic Tree
</h2><span id='topic+miLineage-package'></span><span id='topic+miLineage'></span>

<h3>Description</h3>

<p>A variety of association tests for microbiome data analysis described in Tang Z.-Z. et al.(2017) &lt;doi: 10.1093/bioinformatics/btw804&gt; and Tang Z.-Z. &amp; Chen G. (2017) &lt;Submitted&gt;.
miLineage allows users to (a) perform tests on multivariate taxon counts; (b) localize the covariate-associated lineages on the taxonomic tree; and (c) assess the overall association of the microbial community with the covariate of interest. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> miLineage</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2018-03-09</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL (&gt;=2)</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>QCAT, QCAT_GEE, ZIGDM
</p>


<h3>Author(s)</h3>

<p>Zheng-Zheng Tang
</p>
<p>Maintainer: Zheng-Zheng Tang &lt;tang@biostat.wisc.edu&gt;
</p>


<h3>References</h3>

<p>1. Tang, Z.-Z., Chen, G., Alekseyenko, A. V., and Li, H. (2017). A General Framework for Association Analysis of Microbial Communities on a Taxonomic Tree. Bioinformatics, 33(9), 1278-1285.<br />
2. Tang, Z.-Z. and Chen, G. (2017). Zero-Inflated Generalized Dirichlet Multinomial Regression Model for Microbiome Compositional Data Analysis. Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.toy)
OTU.toy = data.toy$OTU.toy
Tax.toy = data.toy$Tax.toy
case = data.toy$covariate.toy
# the OTUs should be consistent between the OTU table and the taxonomy table
OTU.toy.reorder = OTU.toy[,match(rownames(Tax.toy), colnames(OTU.toy))]
# perform QCAT test for detecting differential mean
QCAT(OTU.toy.reorder, case, 1, Tax.toy, fdr.alpha=0.05)
# perform ZIGDM test for detecting differential dispersion
ZIGDM(OTU.toy.reorder, NULL, NULL, case, test.type = "Disp", 1, 
  ZI.LB = 10, Tax.toy, fdr.alpha = 0.05)

</code></pre>

<hr>
<h2 id='data.real'>
Real data that include an OTU table, a taxonomy table, and a covariate table
</h2><span id='topic+data.real'></span>

<h3>Description</h3>

<p>The data are derived from a real gut microbiome study. The data contain an OTU table, a taxonomy table, and a covariate table. The data include 96 samples and 80 OTUs.   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("data.real")</code></pre>


<h3>Format</h3>

<p>data.real contains the following objects:
</p>

<ul>
<li><p> OTU.real: a matrix of OTU counts for 96 samples and 80 OTUs
</p>
</li>
<li><p> Tax.real: a matrix of taxonomy table from Rank1 (kingdom level) to Rank6 (genus level)  
</p>
</li>
<li><p> covariate.real: a matrix of three variables
</p>
</li></ul>



<h3>Source</h3>

<p>Wu, Gary D., et al. &quot;Linking long-term dietary patterns with gut microbial enterotypes.&quot; Science 334.6052 (2011): 105-108.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.real)
</code></pre>

<hr>
<h2 id='data.toy'>
Simulated data that include an OTU table, a taxonomy table, and a covariate table
</h2><span id='topic+data.toy'></span>

<h3>Description</h3>

<p>This data is simulated based on a mock taxonomic tree. The data contain an OTU table, a taxonomy table, and a covariate. The data include 50 samples (25 case-control pairs) and 7 OTUs.   
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data("data.toy")</code></pre>


<h3>Format</h3>

<p>data.toy contains the following objects:
</p>

<ul>
<li><p> OTU.toy: a matrix of OTU counts for 50 samples and 7 OTUs
</p>
</li>
<li><p> Tax.toy: a matrix of taxonomy table with 3 ranks  
</p>
</li>
<li><p> covariate.toy: a matrix of one variable
</p>
</li></ul>



<h3>Source</h3>

<p>This is a simulated data based on a mock taxonomic tree. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.toy)
</code></pre>

<hr>
<h2 id='QCAT'>
Quasi-Conditional Association Test (QCAT)
</h2><span id='topic+QCAT'></span>

<h3>Description</h3>

<p>This function allows users to (a) perform QCAT on multivariate taxon counts; (b) perform QCAT on the taxonomic tree to localize the covariate-associated lineages; and (c) assess the overall association of the microbial community with the covariate of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QCAT(OTU, X, X.index, Tax = NULL, 
min.depth = 0, n.resample = NULL, fdr.alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QCAT_+3A_otu">OTU</code></td>
<td>

<p>a numeric matrix contains counts with each row corresponds to a sample and each column corresponds to an OTU or a taxon. Column name is mandatory. No missing values are allowed.
</p>
</td></tr>
<tr><td><code id="QCAT_+3A_x">X</code></td>
<td>

<p>a numeric matrix contains covariates with each column pertains to one variable. Samples in the OTU and X matrices should be identical and in the same order. No missing values are allowed. 
</p>
</td></tr>
<tr><td><code id="QCAT_+3A_x.index">X.index</code></td>
<td>

<p>a vector indicates the columns in X for the covariate(s) of interest. The remaining columns in X will be treated as confounders.
</p>
</td></tr>
<tr><td><code id="QCAT_+3A_tax">Tax</code></td>
<td>

<p>a matrix define the taxonomic ranks with each row corresponds to an OTU or a taxon and each column corresponds to a rank (start from the higher taxonomic level, e.g., from kingdom to genus). Row name is mandatory and should be consistent with the column name of the OTU table,  Column name should be formatted as &quot;Rank1&quot;, &quot;Rank2&quot;, etc, representing the taxonomic levels from highest to lowest. 
</p>
</td></tr>
<tr><td><code id="QCAT_+3A_min.depth">min.depth</code></td>
<td>

<p>lower bound for sample read depth. Samples with read depth less than min.depth will be removed before the analysis.
</p>
</td></tr>
<tr><td><code id="QCAT_+3A_n.resample">n.resample</code></td>
<td>

<p>number of reamplings/permutations
</p>
</td></tr>
<tr><td><code id="QCAT_+3A_fdr.alpha">fdr.alpha</code></td>
<td>

<p>desired false discovery rate for multiple tests on the lineages.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If Tax=NULL (Default), a test is performed using all the OTUs/taxa.
</p>
<p>If Tax is provided, tests are performed for lineages derived from the taxonomic hierarchy. The output is a list that contains 3 components
</p>
<table role = "presentation">
<tr><td><code>lineage.pval</code></td>
<td>
<p>p-values for all lineages. By default (n.resample=NULL), only the asymptotic test will be performed.</p>
</td></tr>
<tr><td><code>sig.lineage</code></td>
<td>
<p>a vector of significant lineages</p>
</td></tr>
<tr><td><code>global.pval</code></td>
<td>
<p>p-values of the global tests</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zheng-Zheng Tang
</p>


<h3>References</h3>

<p>Tang, Z.-Z., Chen, G., Alekseyenko, A. V., and Li, H. (2017). A General Framework for Association Analysis of Microbial Communities on a Taxonomic Tree. Bioinformatics, 33(9), 1278-1285.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.toy)
OTU.toy = data.toy$OTU.toy
Tax.toy = data.toy$Tax.toy
case = data.toy$covariate.toy
# the OTUs should be consistent between the OTU table and the taxonomy table
OTU.toy.reorder = OTU.toy[,match(rownames(Tax.toy), colnames(OTU.toy))]
# perform QCAT
QCAT(OTU.toy.reorder, case, 1, Tax.toy, fdr.alpha=0.05)
</code></pre>

<hr>
<h2 id='QCAT_GEE'>
QCAT+GEE Two-Part Test
</h2><span id='topic+QCAT_GEE'></span>

<h3>Description</h3>

<p>This function performs Quasi-Conditional Association Test (QCAT) for the positive taxon counts and (Generalized Estimating Equation) GEE-based association test for the zero counts. Then, the two-part test is from the combination of those two tests.  
This function allows users to (a) perform QCAT+GEE on multivariate taxon counts; (b) perform QCAT+GEE on the taxonomic tree to localize the covariate-associated lineages; and (c) assess the overall association of the microbial community with the covariate of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>QCAT_GEE(OTU, X, X.index, Z, Z.index, Tax = NULL, 
min.depth = 0, n.resample = NULL, fdr.alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="QCAT_GEE_+3A_otu">OTU</code></td>
<td>

<p>a numeric matrix contains counts with each row corresponds to a sample and each column corresponds to an OTU or a taxon. Column name is mandatory. No missing values are allowed. 
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_x">X</code></td>
<td>

<p>a numeric matrix contains covariates for the positive-part test with each column pertains to one variable. Samples in the OTU and X matrices should be identical and in the same order. No missing values are allowed. 
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_x.index">X.index</code></td>
<td>

<p>a vector indicates the columns in X for the covariate(s) of interest in the positive-part test. The remaining columns in X will be treated as confounders in modeling the positive abundance.
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_z">Z</code></td>
<td>

<p>a numeric matrix contains covariates for the zero-part test with each column pertains to one variable. Samples in the OTU and Z matrices should be identical and in the same order. No missing values are allowed.   
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_z.index">Z.index</code></td>
<td>

<p>a vector indicates the columns in Z for the covariate(s) of interest in the zero-part test. The remaining columns in Z will be treated as confounders in modeling the proportion of zero.
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_tax">Tax</code></td>
<td>

<p>a matrix define the taxonomic ranks with each row corresponds to an OTU or a taxon and each column corresponds to a rank (start from the higher taxonomic level, e.g., from kingdom to genus). Row name is mandatory and should be consistent with the column name of the OTU table,  Column name should be formatted as &quot;Rank1&quot;, &quot;Rank2&quot;, etc, representing the taxonomic levels from highest to lowest. 
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_min.depth">min.depth</code></td>
<td>

<p>lower bound for sample read depth. Samples with read depth less than min.depth will be removed before the analysis.
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_n.resample">n.resample</code></td>
<td>

<p>number of reamplings/permutations
</p>
</td></tr>
<tr><td><code id="QCAT_GEE_+3A_fdr.alpha">fdr.alpha</code></td>
<td>

<p>desired false discovery rate for multiple tests on the lineages.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If Tax=NULL (Default), tests are performed using all the OTUs/taxa.
</p>
<p>If Tax is provided, tests are performed for lineages derived from the taxonomic hierarchy. The output is a list that contains 3 components
</p>
<table role = "presentation">
<tr><td><code>lineage.pval</code></td>
<td>
<p>a list of zero-part, positive-part and two-part (combined) p-values for all lineages. By default (n.resample=NULL), only the asymptotic test will be performed. </p>
</td></tr>
<tr><td><code>sig.lineage</code></td>
<td>
<p>a list of significant lineages based on the zero-part, positive-part and two-part p-values</p>
</td></tr>
<tr><td><code>global.pval</code></td>
<td>
<p>p-values of the global tests</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zheng-Zheng Tang
</p>


<h3>References</h3>

<p>Tang, Z.-Z., Chen, G., Alekseyenko, A. V., and Li, H. (2017). A General Framework for Association Analysis of Microbial Communities on a Taxonomic Tree. Bioinformatics, 33(9), 1278-1285.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.toy)
OTU.toy = data.toy$OTU.toy
Tax.toy = data.toy$Tax.toy
case = data.toy$covariate.toy
# the OTUs should be consistent between the OTU table and the taxonomy table
OTU.toy.reorder = OTU.toy[,match(rownames(Tax.toy), colnames(OTU.toy))]
# perform QCAT+GEE test
QCAT_GEE(OTU.toy.reorder, case, 1, case, 1, Tax.toy, fdr.alpha=0.05)
</code></pre>

<hr>
<h2 id='ZIGDM'>
Zero-Inflated Generalized Dirichlet Multinomial (ZIGDM) Tests
</h2><span id='topic+ZIGDM'></span>

<h3>Description</h3>

<p>Different from the distribution-free QCAT and QCAT+GEE, the ZIGDM tests are based on a parametric model (ZIGDM) for multivariate taxon counts. 
The ZIGDM tests can not only detect differential mean but also differential dispersion level or presence-absence frequency in microbial compositions.
This function allows users to (a) perform ZIGDM tests on multivariate taxon counts; (b) perform ZIGDM tests on the taxonomic tree to localize the covariate-associated lineages; and (c) assess the overall association of the microbial community with the covariate of interest.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ZIGDM(OTU, X4freq, X4mean, X4disp, test.type = "Mean", X.index, ZI.LB = 10, Tax = NULL, 
min.depth = 0, n.resample = NULL, fdr.alpha = 0.05)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ZIGDM_+3A_otu">OTU</code></td>
<td>

<p>a numeric matrix contains counts with each row corresponds to a sample and each column corresponds to an OTU or a taxon. Column name is mandatory. No missing values are allowed. 
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_x4freq">X4freq</code></td>
<td>

<p>a numeric matrix contains covariates that link to presence-absence frequency in microbial compositions. Each column pertains to one variable. Set X4freq=NULL if only intercept term is needed. Samples in the OTU and X4freq matrices should be identical and in the same order. No missing values are allowed. 
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_x4mean">X4mean</code></td>
<td>

<p>a numeric matrix contains covariates that link to mean abundance in microbial compositions. Each column pertains to one variable. Set X4mean=NULL if only intercept term is needed. Samples in the OTU and X4mean matrices should be identical and in the same order. No missing values are allowed.  
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_x4disp">X4disp</code></td>
<td>

<p>a numeric matrix contains covariates that link to dispersion level in microbial compositions. Each column pertains to one variable. Set X4disp=NULL if only intercept term is needed. Samples in the OTU and X4disp matrices should be identical and in the same order. No missing values are allowed.   
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_test.type">test.type</code></td>
<td>

<p>If test.type = &quot;Mean&quot;, the function will test for differential mean (Default). If test.type = &quot;Disp&quot;, the function will test for differential dispersion. If test.type = &quot;Freq&quot;, the function will test for differential presence-absence frequency. 
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_x.index">X.index</code></td>
<td>

<p>If test.type = &quot;Mean&quot;, X.index is a vector indicates the columns in X4mean for the covariate(s) of interest. The remaining columns in X4mean will be treated as confounders in modeling the abundance of the present taxa.
If test.type = &quot;Disp&quot;, X.index is a vector indicates the columns in X4disp for the covariate(s) of interest. The remaining columns in X4disp will be treated as confounders in modeling the dispersion level of the present taxa.
If test.type = &quot;Freq&quot;, X.index is a vector indicates the columns in X4freq for the covariate(s) of interest. The remaining columns in X4freq will be treated as confounders in modeling the presence-absence frequency.
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_zi.lb">ZI.LB</code></td>
<td>

<p>lower bound of zero counts for the taxon that needs zero-inflated model. 
The counts for a taxon is assumed to be zero-inflated if the number of zero observations for the taxon is greater than ZI.LB (Default is 10). If ZI.LB=NULL, GDM model (i.e. non-zero-inflated version of ZIGDM) will be applied.
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_tax">Tax</code></td>
<td>

<p>a matrix define the taxonomic ranks with each row corresponds to an OTU or a taxon and each column corresponds to a rank (start from the higher taxonomic level, e.g., from kingdom to genus). Row name is mandatory and should be consistent with the column name of the OTU table,  Column name should be formatted as &quot;Rank1&quot;, &quot;Rank2&quot;, etc, representing the taxonomic levels from highest to lowest. 
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_min.depth">min.depth</code></td>
<td>

<p>lower bound of sample read depth. Samples with read depth less than min.depth will be removed before the analysis.
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_n.resample">n.resample</code></td>
<td>

<p>number of reamplings/permutations
</p>
</td></tr>
<tr><td><code id="ZIGDM_+3A_fdr.alpha">fdr.alpha</code></td>
<td>

<p>desired false discovery rate for multiple tests on the lineages.
</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If Tax=NULL (Default), a test is performed using all the OTUs/taxa.
</p>
<p>If Tax is provided, tests are performed for lineages derived from the taxonomic hierarchy. The output is a list that contains 3 components
</p>
<table role = "presentation">
<tr><td><code>lineage.pval</code></td>
<td>
<p>p-values for all lineages. By default (n.resample=NULL), only the asymptotic test will be performed. </p>
</td></tr>
<tr><td><code>sig.lineage</code></td>
<td>
<p>a vector of significant lineages</p>
</td></tr>
<tr><td><code>global.pval</code></td>
<td>
<p>p-values of the global tests</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Zheng-Zheng Tang
</p>


<h3>References</h3>

<p>Tang, Z.-Z. and Chen, G. (2017). Zero-Inflated Generalized Dirichlet Multinomial Regression Model for Microbiome Compositional Data Analysis. Submitted.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data.toy)
OTU.toy = data.toy$OTU.toy
Tax.toy = data.toy$Tax.toy
case = data.toy$covariate.toy
# the OTUs should be consistent between the OTU table and the taxonomy table
OTU.toy.reorder = OTU.toy[,match(rownames(Tax.toy), colnames(OTU.toy))]
# perform ZIGDM test for detecting differential dispersion
ZIGDM(OTU.toy.reorder, NULL, NULL, case, test.type = "Disp", 1, 
  ZI.LB = 10, Tax.toy, fdr.alpha = 0.05)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
