<!DOCTYPE html><html><head><title>Help for package milorGWAS</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {milorGWAS}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#association.test.logistic'><p>Mixed logistic regression for GWAS</p></a></li>
<li><a href='#association.test.logistic.dosage'><p>Mixed logistic regression for GWAS, using dosages</p></a></li>
<li><a href='#qqplot.pvalues'><p>Stratified QQ-plot of p-values</p></a></li>
<li><a href='#SNP.category'><p>SNP.category</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Mixed Logistic Regression for Genome-Wide Analysis Studies
(GWAS)</td>
</tr>
<tr>
<td>Version:</td>
<td>0.7</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-06-12</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Hervé Perdry &lt;herve.perdry@universite-paris-saclay.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Fast approximate methods for mixed logistic regression in genome-wide analysis studies (GWAS).
  Two computationnally efficient methods are proposed for obtaining effect size estimates (beta) in 
  Mixed Logistic Regression in GWAS: the Approximate Maximum Likelihood Estimate (AMLE), and the Offset
  method. The wald test obtained with AMLE is identical to the score test. Data can be genotype matrices
  in plink format, or dosage (VCF files). The methods are described in details in 
  Milet et al (2020) &lt;<a href="https://doi.org/10.1101%2F2020.01.17.910109">doi:10.1101/2020.01.17.910109</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 1.0.2)</td>
</tr>
<tr>
<td>Depends:</td>
<td>gaston (&ge; 1.6)</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppEigen, gaston</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown, png</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-06-19 11:05:51 UTC; rv</td>
</tr>
<tr>
<td>Author:</td>
<td>Hervé Perdry [aut, cre],
  Jacqueline Milet [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-06-21 08:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='association.test.logistic'>Mixed logistic regression for GWAS</h2><span id='topic+association.test.logistic'></span>

<h3>Description</h3>

<p>Mixed logistic regression for GWAS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association.test.logistic(
  x,
  Y = x@ped$pheno,
  X = matrix(1, nrow(x)),
  K,
  beg = 1,
  end = ncol(x),
  algorithm = c("amle", "offset"),
  eigenK,
  p = 0,
  model = c("additive", "dominant", "recessive"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association.test.logistic_+3A_x">x</code></td>
<td>
<p>a bedmatrix</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_y">Y</code></td>
<td>
<p>phenotype vector. Default is column <code>pheno</code> of <code>x@ped</code></p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_x">X</code></td>
<td>
<p>A matrix of covariates (defaults to a column of ones for the intercept)</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_k">K</code></td>
<td>
<p>A genetic relationship matrix (or a list of such matrices)</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_beg">beg</code></td>
<td>
<p>Index of the first SNP tested for association</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_end">end</code></td>
<td>
<p>Index of the last SNP tested for association</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_algorithm">algorithm</code></td>
<td>
<p>Algorithm to use</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_eigenk">eigenK</code></td>
<td>
<p>eigen decomposition of K (only if <code>p</code> &gt; 0)</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_p">p</code></td>
<td>
<p>Number of principal components to include in the model</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_model">model</code></td>
<td>
<p>Model for the effect allele (allele A2)</p>
</td></tr>
<tr><td><code id="association.test.logistic_+3A_...">...</code></td>
<td>
<p>Additional parameter for <code>gaston::logistic.mm.aireml</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Tests the association between the phenotype and requested SNPs in <code>x</code>.
The phenotype <code>Y</code> is a binary trait. A Wald test is performed using an approximate
method defined by the parameter <code>algorithm</code>. 
</p>
<p>Parameter <code>model</code> allows to specify an additive model (genotypes A1 A1, A1 A2, and A2 A2
are recoded for analysis as 0, 1 and 2 respectively), a dominant model (genotypes recoded as 0, 1, and 1) or a recessive
model (recoded as 0, 0 and 1).
</p>
<p>All other arguments are as in <code>gaston::association.test</code>.
</p>


<h3>Value</h3>

<p>A data frame giving for each SNP the association statistics.
</p>


<h3>See Also</h3>

<p><code><a href="gaston.html#topic+association.test">association.test</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(TTN)
x &lt;- as.bed.matrix(TTN.gen, TTN.fam, TTN.bim)
## Simulation data ##
set.seed(1)
# some covariables
X &lt;- cbind(1, runif(nrow(x)))
# A random GRM
ran &lt;- random.pm( nrow(x))
# random effects (tau = 1)
omega &lt;- lmm.simu(1, 0, eigenK=ran$eigen)$omega
# linear term of the model
lin &lt;- X %*% c(0.1,-0.2) + omega
# vector of probabilitues
pi &lt;- 1/(1+exp( -lin ))
# vector of binary phenotypes
y &lt;- rbinom(nrow(x), 1, pi)
# testing association with 1) the score test, 2) the offset algorithm, 3) the 'amle' algorithm
a1 &lt;- association.test(x, y, X, K = ran$K, method = "lmm", response = "bin")
a2 &lt;- association.test.logistic(x, y, X, K = ran$K, algorithm = "offset")
a3 &lt;- association.test.logistic(x, y, X, K = ran$K, algorithm = "amle")

</code></pre>

<hr>
<h2 id='association.test.logistic.dosage'>Mixed logistic regression for GWAS, using dosages</h2><span id='topic+association.test.logistic.dosage'></span>

<h3>Description</h3>

<p>Mixed logistic regression for GWAS, using dosages
</p>


<h3>Usage</h3>

<pre><code class='language-R'>association.test.logistic.dosage(
  filename,
  Y,
  X,
  K,
  beg,
  end,
  algorithm = c("amle", "offset"),
  eigenK,
  p = 0,
  n.cores = 1L,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="association.test.logistic.dosage_+3A_filename">filename</code></td>
<td>
<p>Name of a dosage file</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_y">Y</code></td>
<td>
<p>phenotype vector. Default is column <code>pheno</code> of <code>x@ped</code></p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_x">X</code></td>
<td>
<p>A matrix of covariates (defaults to a column of ones for the intercept)</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_k">K</code></td>
<td>
<p>A genetic relationship matrix (or a list of such matrices)</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_beg">beg</code></td>
<td>
<p>Index of the first SNP tested for association</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_end">end</code></td>
<td>
<p>Index of the last SNP tested for association</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_algorithm">algorithm</code></td>
<td>
<p>Algorithm to use</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_eigenk">eigenK</code></td>
<td>
<p>eigen decomposition of K (only if <code>p</code> &gt; 0)</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_p">p</code></td>
<td>
<p>Number of principal components to include in the model</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_n.cores">n.cores</code></td>
<td>
<p>number of cores to use</p>
</td></tr>
<tr><td><code id="association.test.logistic.dosage_+3A_...">...</code></td>
<td>
<p>Additional parameter for <code>gaston::logistic.mm.aireml</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Dosage files can be VCF files with 'DS' or 'GP' fields. It is also possible to use a file
with columns 'id&quot;', 'chr', 'pos', 'A1', 'A2', 'sample1', 'sample2', etc. These files should have a 
header with column names.
</p>
<p>For more details refer to <code><a href="#topic+association.test.logistic">association.test.logistic</a></code> and <code><a href="gaston.html#topic+association.test">association.test</a></code>.
</p>


<h3>Value</h3>

<p>A data frame giving for each SNP the association statistics.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+association.test.logistic">association.test.logistic</a></code>, <code><a href="gaston.html#topic+association.test">association.test</a></code>
</p>

<hr>
<h2 id='qqplot.pvalues'>Stratified QQ-plot of p-values</h2><span id='topic+qqplot.pvalues'></span>

<h3>Description</h3>

<p>Draws a QQ plot of p-values
</p>


<h3>Usage</h3>

<pre><code class='language-R'>qqplot.pvalues(
  p,
  snp.cat,
  col.cat,
  col.abline = "red",
  CB = TRUE,
  col.CB = "gray80",
  CB.level = 0.95,
  thinning = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="qqplot.pvalues_+3A_p">p</code></td>
<td>
<p>vector of p-values, or a data.frame with a column named <code>p</code></p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_snp.cat">snp.cat</code></td>
<td>
<p>(optional) A factor giving the SNP categories.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_col.cat">col.cat</code></td>
<td>
<p>(optional) A vector of colors used to plot the SNP categories.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_col.abline">col.abline</code></td>
<td>
<p>Color of the line of slope 1. Set to <code>NA</code> to suppress.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_cb">CB</code></td>
<td>
<p><code>Logical</code>. If <code>TRUE</code>, a confidence band is included in the plot.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_col.cb">col.CB</code></td>
<td>
<p>The color of the confidence band.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_cb.level">CB.level</code></td>
<td>
<p>The level of the confidence band.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_thinning">thinning</code></td>
<td>
<p><code>Logical</code>. If <code>TRUE</code>, not all points are displayed.</p>
</td></tr>
<tr><td><code id="qqplot.pvalues_+3A_...">...</code></td>
<td>
<p>Graphical parameters to be passed to <code>plot</code> and <code>points</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function draws a QQ plot of <code class="reqn">p</code>-values, stratified by categories.
If the parameter <code>snp.cat</code> is missing, the function falls back on <code>gaston::qqplot.pvalues</code>.
</p>


<h3>Value</h3>

<p>Returns a 'NULL'
</p>


<h3>See Also</h3>

<p><code><a href="#topic+SNP.category">SNP.category</a></code>, <code><a href="gaston.html#topic+qqplot.pvalues">qqplot.pvalues</a></code> (in gaston)
</p>


<h3>Examples</h3>

<pre><code class='language-R'># a random vector of categories
ca &lt;- sample(c("A","B","C"), 1e6, TRUE, c(0.05, 0.9, 0.05))
# a vector of p-values, with different distribution depending on the strata
p &lt;- runif(1e6)**ifelse(ca == "A", .8, ifelse(ca == "B", 1, 1.2))
qqplot.pvalues(p, ca)

</code></pre>

<hr>
<h2 id='SNP.category'>SNP.category</h2><span id='topic+SNP.category'></span>

<h3>Description</h3>

<p>SNP.category
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SNP.category(bed, Z, threshold = 0.8)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SNP.category_+3A_bed">bed</code></td>
<td>
<p>A bed matrix</p>
</td></tr>
<tr><td><code id="SNP.category_+3A_z">Z</code></td>
<td>
<p>A vector of length <code>nrow(bed)</code></p>
</td></tr>
<tr><td><code id="SNP.category_+3A_threshold">threshold</code></td>
<td>
<p>Variance thresholds</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function determines a SNP Category from a covariable <code>Z</code>,
which can be for example an indicator variable for a population strata, 
or the first genomic principal component.
</p>


<h3>Value</h3>

<p>A factor giving the category of each SNP
</p>


<h3>See Also</h3>

<p><code><a href="#topic+qqplot.pvalues">qqplot.pvalues</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># a random vector of categories
ca &lt;- sample(c("A","B","C"), 1e6, TRUE, c(0.05, 0.9, 0.05))
# a vector of p-values, with different distribution depending on the strata
p &lt;- runif(1e6)**ifelse(ca == "A", .8, ifelse(ca == "B", 1, 1.2))
qqplot.pvalues(p, ca)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
