<!DOCTYPE html><html><head><title>Help for package MiRKAT</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MiRKAT}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#bindata'><p>Simulated DEPENDENT data with BINOMIAL traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)</p></a></li>
<li><a href='#CSKAT'><p>Small-sample SKAT for correlated (continuous) data ('c' stands for 'correlated'). Called within GLMM-MiRKAT.</p></a></li>
<li><a href='#D2K'><p>D2K</p></a></li>
<li><a href='#GLMMMiRKAT'><p>The Microbiome Regression-based Kernel Association Test Based on the Generalized Linear Mixed Model</p></a></li>
<li><a href='#inner.CSKAT'><p>Inner Function for CSKAT, Correlated Sequence Kernel Association Test</p></a></li>
<li><a href='#inner.KRV'><p>Kernel RV Coefficient Test; Inner Function</p></a></li>
<li><a href='#KRV'><p>Kernel RV Coefficient Test (KRV)</p></a></li>
<li><a href='#MiRKAT'><p>Microbiome Regression-based Kernel Association Test</p></a></li>
<li><a href='#MiRKAT_binary'><p>Microbiome Regression-Based Kernel Association Test for binary outcomes</p></a></li>
<li><a href='#MiRKAT_continuous'><p>Microbiome Regression-based Analysis Test for a continuous outcome variable</p></a></li>
<li><a href='#MiRKAT.iQ'><p>MiRKAT-iQ</p></a></li>
<li><a href='#MiRKAT.Q'><p>Robust MiRKAT (quantile regression)</p></a></li>
<li><a href='#MiRKAT.R'><p>Robust MiRKAT (robust regression)</p></a></li>
<li><a href='#MiRKATS'><p>Microiome Regression-based Kernel Association Test for Survival</p></a></li>
<li><a href='#MMiRKAT'><p>Multivariate Microbiome Regression-based Kernel Association Test</p></a></li>
<li><a href='#nordata'><p>Simulated DEPENDENT data with GAUSSIAN traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)</p></a></li>
<li><a href='#poisdata'><p>Simulated DEPENDENT data with POISSON (count) traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)</p></a></li>
<li><a href='#throat.meta'><p>Simulated metadata for microbiome regression-based analysis</p></a></li>
<li><a href='#throat.otu.tab'><p>Simulated OTU data for microbiome regression analysis</p></a></li>
<li><a href='#throat.tree'><p>Simulated rooted phylogenetic tree</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Microbiome Regression-Based Kernel Association Tests</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2.3</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Anna Plantinga &lt;amp9@williams.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Test for overall association between microbiome composition data 
  and phenotypes via phylogenetic kernels. The phenotype can be univariate 
  continuous or binary (Zhao et al. (2015) &lt;<a href="https://doi.org/10.1016%2Fj.ajhg.2015.04.003">doi:10.1016/j.ajhg.2015.04.003</a>&gt;), 
  survival outcomes (Plantinga et al. (2017) &lt;<a href="https://doi.org/10.1186%2Fs40168-017-0239-9">doi:10.1186/s40168-017-0239-9</a>&gt;), 
  multivariate (Zhan et al. (2017) &lt;<a href="https://doi.org/10.1002%2Fgepi.22030">doi:10.1002/gepi.22030</a>&gt;) and 
  structured phenotypes (Zhan et al. (2017) &lt;<a href="https://doi.org/10.1111%2Fbiom.12684">doi:10.1111/biom.12684</a>&gt;). 
  The package can also use robust regression (unpublished work) and 
  integrated quantile regression (Wang et al. (2021) &lt;<a href="https://doi.org/10.1093%2Fbioinformatics%2Fbtab668">doi:10.1093/bioinformatics/btab668</a>&gt;). 
  In each case, the microbiome community effect is modeled nonparametrically 
  through a kernel function, which can incorporate phylogenetic tree information. </td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.1.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, CompQuadForm, quantreg, GUniFrac, PearsonDS, lme4,
Matrix, permute, mixtools, survival, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, vegan, rmarkdown, magrittr, kableExtra</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-02-15 23:09:07 UTC; anna</td>
</tr>
<tr>
<td>Author:</td>
<td>Anna Plantinga [aut, cre],
  Nehemiah Wilson [aut, ctb],
  Haotian Zheng [aut, ctb],
  Tianying Wang [aut, ctb],
  Xiang Zhan [aut, ctb],
  Michael Wu [aut],
  Ni Zhao [aut, ctb],
  Jun Chen [aut]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-02-17 14:50:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='bindata'>Simulated DEPENDENT data with BINOMIAL traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)</h2><span id='topic+bindata'></span>

<h3>Description</h3>

<p>Simulated DEPENDENT data with BINOMIAL traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(bindata)
</code></pre>


<h3>Format</h3>

<p>A list containing three data objects for correlated microbiome data with binary response variable (described below). 
</p>

<dl>
<dt>bin.otu.tab</dt><dd><p>Simulated OTU data for correlated regression-based analysis; 59 rows and 730 columns, rows being patients and 
columns being OTUs</p>
</dd>
<dt>bin.meta</dt><dd><p>Simulated metadata for correlated regression-based analysis; 59 rows and 4 columns, rows being patients and 
columns being the outcome variable, subject identifier, and covariates to possibly account for in any regression modeling</p>
</dd>
<dt>bin.tree</dt><dd><p>Simulated rooted phylogenetic tree with 730 tips and 729 nodes</p>
</dd>
</dl>


<hr>
<h2 id='CSKAT'>Small-sample SKAT for correlated (continuous) data ('c' stands for 'correlated'). Called within GLMM-MiRKAT.</h2><span id='topic+CSKAT'></span>

<h3>Description</h3>

<p>Compute the adjusted score statistic and p-value
</p>


<h3>Usage</h3>

<pre><code class='language-R'>CSKAT(lmer.obj, Ks)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="CSKAT_+3A_lmer.obj">lmer.obj</code></td>
<td>
<p>A fitted lme4 object (model under H0)</p>
</td></tr>
<tr><td><code id="CSKAT_+3A_ks">Ks</code></td>
<td>
<p>A kernel matrix or list of kernels, quantifying the similarities between samples.</p>
</td></tr>
</table>


<h3>Value</h3>


<dl>
<dt>p.value</dt><dd><p>Association p-values</p>
</dd>
<dt>Q.adj</dt><dd><p>Adjusted score statistics</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Nehemiah Wilson, Anna Plantinga, Xiang Zhan, Jun Chen.
</p>


<h3>References</h3>

<p>Zhan X, et al. (2018) A small-sample kernel association test for correlated data with application to microbiome association studies.  
Genet Epidemiol.
</p>

<hr>
<h2 id='D2K'>D2K</h2><span id='topic+D2K'></span>

<h3>Description</h3>

<p>Construct kernel matrix from distance matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>D2K(D)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="D2K_+3A_d">D</code></td>
<td>
<p>An n by n matrix giving pairwise distances or dissimilarites, where n is sample size.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Converts a distance matrix (matrix of pairwise distances) into a kernel matrix for microbiome data. The kernel matrix is constructed as <code class="reqn">K = -(I-11'/n)D^2(I-11'/n)/2</code>, where D is the pairwise distance matrix, I is the identity
matrix, and 1 is a vector of ones.
</p>
<p><code class="reqn">D^2</code> represents element-wise square.
</p>
<p>To ensure that <code class="reqn">K</code> is positive semi-definite, a positive semi-definiteness correction is conducted
</p>


<h3>Value</h3>

<p>An n by n kernel or similarity matrix corresponding to the distance matrix given.
</p>


<h3>Author(s)</h3>

<p>Ni Zhao
</p>


<h3>References</h3>

<p>Zhao, Ni, et al. &quot;Testing in microbiome-profiling studies with MiRKAT, the microbiome regression-based kernel association test
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GUniFrac)

#Load in data and create a distance matrix
data(throat.tree)
data(throat.otu.tab)
unifracs &lt;- GUniFrac(throat.otu.tab, throat.tree, alpha=c(1))$unifracs
D1 &lt;- unifracs[,,"d_1"]

#Function call
K &lt;- D2K(D1)

</code></pre>

<hr>
<h2 id='GLMMMiRKAT'>The Microbiome Regression-based Kernel Association Test Based on the Generalized Linear Mixed Model</h2><span id='topic+GLMMMiRKAT'></span>

<h3>Description</h3>

<p>GLMMMiRKAT utilizes a generalized linear mixed model to allow dependence among samples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GLMMMiRKAT(
  y,
  X = NULL,
  Ks,
  id = NULL,
  time.pt = NULL,
  model,
  method = "perm",
  slope = FALSE,
  omnibus = "perm",
  nperm = 5000
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GLMMMiRKAT_+3A_y">y</code></td>
<td>
<p>A numeric vector of Gaussian (e.g., body mass index), Binomial (e.g., disease status, treatment/placebo) or 
Poisson (e.g., number of tumors/treatments) traits.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_x">X</code></td>
<td>
<p>A vector or matrix of numeric covariates, if applicable (default = NULL).</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_ks">Ks</code></td>
<td>
<p>A list of n-by-n OTU kernel matrices or one singular n-by-n OTU kernel matrix, where n is sample size.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_id">id</code></td>
<td>
<p>A vector of cluster (e.g., family or subject including repeated measurements) IDs. Defaults to NULL since it is 
unnecessary for the CSKAT call.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_time.pt">time.pt</code></td>
<td>
<p>A vector of time points for the longitudinal studies. 'time.pt' is not required (i.e., 'time.pt = NULL') 
for the random intercept model. Default is time.pt = NULL.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_model">model</code></td>
<td>
<p>A string declaring which model (&quot;gaussian&quot;, &quot;binomial&quot; or &quot;poisson&quot;) is to be used; should align with whether a 
Gaussian, Binomial, or Poisson trait is being inputted for the y argument.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_method">method</code></td>
<td>
<p>A string declaring which method (&quot;perm&quot; or &quot;davies) will be used to calculate the p-value. Davies is only 
available for Gaussian traits. Defaults to &quot;perm&quot;.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_slope">slope</code></td>
<td>
<p>An indicator to include random slopes in the model (slope = TRUE) or not (slope = FALSE). 'slope = FALSE' is for 
the random intercept model. 'slope = TRUE' is for the random slope model. For the random slope model (slope = TRUE), 'time.pt' 
is required.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;permutation&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or residual permutation to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="GLMMMiRKAT_+3A_nperm">nperm</code></td>
<td>
<p>The number of permutations used to calculate the p-values and omnibus p-value. Defaults to 5000.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Missing data is not permitted. Please remove all individuals with missing y, X, and Ks prior to input for analysis.
</p>
<p>y and X (if not NULL) should be numerical matrices or vectors with the same number of rows. 
</p>
<p>Ks should either be a list of n by n kernel matrices (where n is sample size) or a single kernel matrix. If you have distance
matrices from metagenomic data, each kernel can be constructed through function D2K. Each kernel can also be constructed 
through other mathematical approaches.
</p>
<p>If model=&quot;gaussian&quot; and method=&quot;davies&quot;, CSKAT is called. CSKAT utilizes the same omnibus test as GLMMMiRKAT. See ?CSKAT for more details.
</p>
<p>The &quot;method&quot; argument only determines kernel-specific p-values are generated. When Ks is a list of multiple kernels,
an omnibus p-value is computed via permutation.
</p>


<h3>Value</h3>

<p>Returns a p-value for each inputted kernel matrix, as well as an overall omnibus p-value if more than one kernel matrix
is inputted
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>p-value for each individual kernel matrix</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>overall omnibus p-value calculated by permutation for the adaptive GLMMMiRKAT analysis</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Hyunwook Koh
</p>


<h3>References</h3>

<p>Koh H, Li Y, Zhan X, Chen J, Zhao N. (2019) A distance-based kernel association test based on the generalized linear mixed 
model for correlated microbiome studies. Front. Genet. 458(10), 1-14.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

## Example with Gaussian (e.g., body mass index) traits
## For non-Gaussian traits, see vignette. 

# Import example microbiome data with Gaussian traits
data(nordata)
otu.tab &lt;- nordata$nor.otu.tab
meta &lt;- nordata$nor.meta

# Create kernel matrices and run analysis 
if (requireNamespace("vegan")) {
 library(vegan)
 D_BC = as.matrix(vegdist(otu.tab, 'bray'))
 K_BC = D2K(D_BC)
 GLMMMiRKAT(y = meta$y, X = cbind(meta$x1, meta$x2), id = meta$id, 
          Ks = K_BC, model = "gaussian", nperm = 500)
} else {
# Computation time is longer for phylogenetic kernels 
 tree &lt;- nordata$nor.tree
 unifracs &lt;- GUniFrac::GUniFrac(otu.tab, tree, alpha=c(1))$unifracs
 D_W &lt;- unifracs[,,"d_1"] 
 K_W = D2K(D_W)
 GLMMMiRKAT(y = meta$y, X = cbind(meta$x1, meta$x2), id = meta$id, 
          Ks = K_W, model = "gaussian", nperm = 500)
}

</code></pre>

<hr>
<h2 id='inner.CSKAT'>Inner Function for CSKAT, Correlated Sequence Kernel Association Test</h2><span id='topic+inner.CSKAT'></span>

<h3>Description</h3>

<p>Small-sample SKAT for correlated (continuous) data ('c' stands for 'correlated'). Computes the adjusted score statistic and p-value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inner.CSKAT(lmer.obj, K)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inner.CSKAT_+3A_lmer.obj">lmer.obj</code></td>
<td>
<p>A fitted lme4 object (model under H0)</p>
</td></tr>
<tr><td><code id="inner.CSKAT_+3A_k">K</code></td>
<td>
<p>the kernel matrix, which quantifies the similarities between samples</p>
</td></tr>
</table>


<h3>Value</h3>


<dl>
<dt>p.value</dt><dd><p>association p-value</p>
</dd>
<dt>Q.adj</dt><dd><p>adjusted score statistic</p>
</dd>
</dl>



<h3>References</h3>

<p>Zhan X, et al. (2018) A small-sample kernel association test for correlated data with application to microbiome association studies.  
Genet Epidemiol., submitted.
</p>

<hr>
<h2 id='inner.KRV'>Kernel RV Coefficient Test; Inner Function</h2><span id='topic+inner.KRV'></span>

<h3>Description</h3>

<p>Function called when user calls function KRV. For each kernel matrix inputted into KRV, KRV runs inner.KRV on that 
kernel with the inputted kernel.y outcome matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>inner.KRV(
  y = NULL,
  X = NULL,
  adjust.type,
  kernel.otu,
  kernel.y,
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="inner.KRV_+3A_y">y</code></td>
<td>
<p>A numeric n by p matrix of p continuous phenotype variables and 
sample size n (default = NULL). If it is NULL, a 
phenotype kernel matrix must be entered for &quot;kernel.y&quot;. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="inner.KRV_+3A_x">X</code></td>
<td>
<p>A numeric n by q matrix, containing q additional covariates 
(default = NULL). If NULL, an intercept only model is used. If the first
column of X is not uniformly 1, then an intercept column will be added.</p>
</td></tr>
<tr><td><code id="inner.KRV_+3A_adjust.type">adjust.type</code></td>
<td>
<p>Possible values are &quot;none&quot; (default if X is null), 
&quot;phenotype&quot; to adjust only the y variable (only possible if y is a numeric 
phenotype matrix rather than a pre-computed kernel), or &quot;both&quot; to adjust 
both the X and Y kernels.</p>
</td></tr>
<tr><td><code id="inner.KRV_+3A_kernel.otu">kernel.otu</code></td>
<td>
<p>A numeric OTU n by n kernel matrix or a list of matrices, 
where n is the sample size. It can be constructed from microbiome data, such 
as by transforming from a distance metric.</p>
</td></tr>
<tr><td><code id="inner.KRV_+3A_kernel.y">kernel.y</code></td>
<td>
<p>Either a numerical n by n kernel matrix for phenotypes or a 
method to compute the kernel of phenotype. Methods are &quot;Gaussian&quot; or &quot;linear&quot;. 
A Gaussian kernel (kernel.y=&quot;Gaussian&quot;) can capture the general relationship 
between microbiome and phenotypes; a linear kernel (kernel.y=&quot;linear&quot;) 
may be preferred if the underlying relationship is close to linear.</p>
</td></tr>
<tr><td><code id="inner.KRV_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="inner.KRV_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>y and X (if not NULL) should all be numerical matrices or vectors with the same number of rows.
</p>
<p>Ks should be a list of n by n matrices or a single matrix. If you have distance metric from metagenomic data, each kernel can be
constructed through function D2K. Each kernel can also be constructed through other mathematical approaches.
</p>
<p>Missing data is not permitted. Please remove all individuals with missing y, X, Ks prior to analysis
</p>
<p>Parameter &quot;method&quot; only concerns how kernel specific p-values are generated. When Ks is a list of multiple kernels, omnibus
p-value is computed through permutation from each individual p-value, which are calculated through method of choice.
</p>


<h3>Value</h3>

<p>Returns a p-value for the candidate kernel matrix
</p>
<table>
<tr><td><code>pv</code></td>
<td>
<p>p-value for the candidate kernel matrix</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>KRV statistic for the candidate kernel matrix. Only returned if returnKRV = TRUE.</p>
</td></tr>  
<tr><td><code>R2</code></td>
<td>
<p>R-squared for the candidate kernel matrix. Only returned if returnR2 = TRUE.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Haotian Zheng, Xiang Zhan, Ni Zhao
</p>


<h3>References</h3>

<p>Zhan, X., Plantinga, A., Zhao, N., and Wu, M.C. A Fast Small-Sample Kernel Independence Test for Microbiome Community-Level 
Association Analysis. Biometrics. 2017 Mar 10. doi: 10.1111/biom.12684.
</p>

<hr>
<h2 id='KRV'>Kernel RV Coefficient Test (KRV)</h2><span id='topic+KRV'></span>

<h3>Description</h3>

<p>Kernel RV coefficient test to evaluate the overall association between 
microbiome composition and high-dimensional or structured phenotype or genotype.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>KRV(
  y = NULL,
  X = NULL,
  adjust.type = NULL,
  kernels.otu,
  kernel.y,
  omnibus = "kernel_om",
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="KRV_+3A_y">y</code></td>
<td>
<p>A numeric n by p matrix of p continuous phenotype variables and 
sample size n (default = NULL). If it is NULL, a 
phenotype kernel matrix must be entered for &quot;kernel.y&quot;. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="KRV_+3A_x">X</code></td>
<td>
<p>A numeric n by q matrix, containing q additional covariates 
(default = NULL). If NULL, an intercept only model is used. If the first
column of X is not uniformly 1, then an intercept column will be added.</p>
</td></tr>
<tr><td><code id="KRV_+3A_adjust.type">adjust.type</code></td>
<td>
<p>Possible values are &quot;none&quot; (default if X is null), 
&quot;phenotype&quot; to adjust only the y variable (only possible if y is a numeric 
phenotype matrix rather than a pre-computed kernel), or &quot;both&quot; to adjust 
both the X and Y kernels.</p>
</td></tr>
<tr><td><code id="KRV_+3A_kernels.otu">kernels.otu</code></td>
<td>
<p>A numeric OTU n by n kernel matrix or a list of matrices, 
where n is the sample size. It can be constructed from microbiome data, such 
as by transforming from a distance metric.</p>
</td></tr>
<tr><td><code id="KRV_+3A_kernel.y">kernel.y</code></td>
<td>
<p>Either a numerical n by n kernel matrix for phenotypes or a 
method to compute the kernel of phenotype. Methods are &quot;Gaussian&quot; or &quot;linear&quot;. 
A Gaussian kernel (kernel.y=&quot;Gaussian&quot;) can capture the general relationship 
between microbiome and phenotypes; a linear kernel (kernel.y=&quot;linear&quot;) 
may be preferred if the underlying relationship is close to linear.</p>
</td></tr>
<tr><td><code id="KRV_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;kernel_om&quot; (or unambiguous 
abbreviations thereof), specifying whether to use the Cauchy combination test 
or an omnibus kernel to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="KRV_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="KRV_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>kernels.otu should be a list of numerical n by n kernel matrices, or a single 
n by n kernel matrix, where n is sample size.
</p>
<p>When kernel.y is a method (&quot;Gaussian&quot; or &quot;linear&quot;) to compute the kernel of 
phenotype, y should be a numerical phenotype matrix, and X (if not NULL) 
should be a numeric matrix of covariates. Both y and X should have n rows.
</p>
<p>When kernel.y is a kernel matrix for the phenotype, there is no need to provide 
X and y, and they will be ignored if provided. In this case, kernel.y and 
kernel.otu should both be numeric matrices with the same number of rows and columns.
</p>
<p>Missing data is not permitted. Please remove all individuals with missing 
kernel.otu, y (if not NULL), X (if not NULL), and kernel.y (if a matrix is 
entered) prior to analysis.
</p>


<h3>Value</h3>

<p>If only one candidate kernel matrix is considered, returns a list containing the p-value for the candidate kernel matrix. 
If more than one candidate kernel matrix is considered, returns a list of two elements: 
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>P-value for each candidate kernel matrix</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>Omnibus p-value</p>
</td></tr> 
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nehemiah Wilson, Haotian Zheng, Xiang Zhan, Ni Zhao
</p>


<h3>References</h3>

<p>Zheng, Haotian, Zhan, X., Plantinga, A., Zhao, N., and Wu, M.C. A Fast Small-Sample Kernel Independence Test for Microbiome 
Community-Level Association Analysis. Biometrics. 2017 Mar 10. doi: 10.1111/biom.12684.
</p>
<p>Liu, Hongjiao, Ling, W., Hua, X., Moon, J.Y., Williams-Nguyen, J., Zhan, X., Plantinga, A.M., Zhao, N., 
Zhang, A., Durazo-Arzivu, R.A., Knight, R., Qi, Q., Burk, R.D., Kaplan, R.C., and Wu, M.C. 
Kernel-based genetic association analysis for microbiome phenotypes identifies host genetic 
drivers of beta-diversity. 2021+
</p>


<h3>Examples</h3>

<pre><code class='language-R'>        
library(GUniFrac)
library(MASS)

data(throat.tree)
data(throat.otu.tab)
data(throat.meta)

## Simulate covariate data 
set.seed(123)
n = nrow(throat.otu.tab)
Sex &lt;- throat.meta$Sex
Smoker &lt;- throat.meta$SmokingStatus
anti &lt;- throat.meta$AntibioticUsePast3Months_TimeFromAntibioticUsage
Male = (Sex == "Male")**2
Smoker = (Smoker == "Smoker") **2
Anti =  (anti != "None")^2
cova = cbind(1, Male, Smoker, Anti)

## Simulate microbiome data
otu.tab.rff &lt;- Rarefy(throat.otu.tab)$otu.tab.rff
unifracs &lt;- GUniFrac(otu.tab.rff, throat.tree, alpha=c(0, 0.5, 1))$unifracs
# Distance matrices
D.weighted = unifracs[,,"d_1"]
D.unweighted = unifracs[,,"d_UW"]
# Kernel matrices 
K.weighted = D2K(D.weighted)
K.unweighted = D2K(D.unweighted)

if (requireNamespace("vegan")) {
 library(vegan)
 D.BC = as.matrix(vegdist(otu.tab.rff, method="bray"))
 K.BC = D2K(D.BC)
} 

# Simulate phenotype data 
rho = 0.2
Va = matrix(rep(rho, (2*n)^2), 2*n, 2*n)+diag(1-rho, 2*n)
Phe = mvrnorm(n, rep(0, 2*n), Va)  
K.y = Phe %*% t(Phe)  # phenotype kernel 

# Simulate genotype data
G = matrix(rbinom(n*10, 2, 0.1), n, 10) 
K.g = G %*% t(G)  # genotype kernel


## Unadjusted analysis (microbiome and phenotype)
KRV(y = Phe, kernels.otu = K.weighted, kernel.y = "Gaussian")  # numeric y 
KRV(kernels.otu = K.weighted, kernel.y = K.y)  # kernel y 

## Adjusted analysis (phenotype only)
KRV(kernels.otu = K.weighted, y = Phe, kernel.y = "linear", X = cova, adjust.type = "phenotype")

if (requireNamespace("vegan")) {
  ## Adjusted analysis (adjust both kernels; microbiome and phenotype) 
  KRV(kernels.otu = K.BC, kernel.y = K.y, X = cova, adjust.type='both')

  ## Adjusted analysis (adjust both kernels; microbiome and genotype)
  KRV(kernels.otu = K.BC, kernel.y = K.g, X = cova, adjust.type='both')
}


</code></pre>

<hr>
<h2 id='MiRKAT'>Microbiome Regression-based Kernel Association Test</h2><span id='topic+MiRKAT'></span>

<h3>Description</h3>

<p>Test for association between microbiome composition and a continuous or dichotomous outcome by incorporating 
phylogenetic or nonphylogenetic distance between different microbiomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKAT(
  y,
  X = NULL,
  Ks,
  out_type = "C",
  method = "davies",
  omnibus = "permutation",
  nperm = 999,
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKAT_+3A_y">y</code></td>
<td>
<p>A numeric vector of the a continuous or dichotomous outcome variable.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_x">X</code></td>
<td>
<p>A numeric matrix or data frame, containing additional covariates that you want to adjust for. If NULL, a intercept
only model is used. Defaults to NULL.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_ks">Ks</code></td>
<td>
<p>A list of n by n kernel matrices or a single n by n kernel matrix, where n is the sample size. It can be constructed 
from microbiome data through distance metric or other approaches, such as linear kernels or Gaussian kernels.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_out_type">out_type</code></td>
<td>
<p>An indicator of the outcome type (&quot;C&quot; for continuous, &quot;D&quot; for dichotomous).</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_method">method</code></td>
<td>
<p>Method used to compute the kernel specific p-value. &quot;davies&quot; represents an exact method that computes the p-value by
inverting the characteristic function of the mixture chisq. We adopt an exact variance component tests because most of the studies
concerning microbiome compositions have modest sample size. &quot;moment&quot; represents an approximation method that matches the first
two moments. &quot;permutation&quot; represents a permutation approach for p-value calculation. Defaults to &quot;davies&quot;.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;cauchy&quot; or &quot;permutation&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or residual permutation to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_nperm">nperm</code></td>
<td>
<p>The number of permutations if method = &quot;permutation&quot; or when multiple kernels are considered. If method = &quot;davies&quot; or 
&quot;moment&quot;, nperm is ignored. Defaults to 999.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic (a measure of effect size). Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MiRKAT_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return R-squared. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>y and X (if not NULL) should all be numeric matrices or vectors with the same number of rows.
</p>
<p>Ks should be a list of n by n matrices or a single matrix. If you have distance metric(s) from metagenomic data, each kernel can be
constructed through function D2K. Each kernel can also be constructed through other mathematical approaches.
</p>
<p>Missing data is not permitted. Please remove all individuals with missing y, X, Ks prior to analysis
</p>
<p>Parameter &quot;method&quot; only concerns with how kernel specific p-values are generated. When Ks is a list of multiple kernels, omnibus 
p-value is computed through permutation from each individual p-values, which are calculated through method of choice.
</p>


<h3>Value</h3>

<p>Returns a list containing the following elements: 
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>P-value for each candidate kernel matrix</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>Omnibus p-value considering multiple candidate kernel matrices</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>Kernel RV statistic (a measure of effect size). Only returned if returnKRV = TRUE.</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>R-squared. Only returned if returnR2 = TRUE.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ni Zhao
</p>


<h3>References</h3>

<p>Zhao, N., Chen, J.,Carroll, I. M., Ringel-Kulka, T., Epstein, M.P., Zhou, H., Zhou, J. J., Ringel, Y., Li, H. and Wu, M.C. (2015)).
Microbiome Regression-based Kernel Association Test (MiRKAT). American Journal of Human Genetics, 96(5):797-807
</p>
<p>Chen, J., Chen, W., Zhao, N., Wu, M~C.and Schaid, D~J. (2016) Small Sample Kernel Association Tests for Human Genetic and Microbiome
Association Studies. 40: 5-19. doi: 10.1002/gepi.21934
</p>
<p>Davies R.B. (1980) Algorithm AS 155: The Distribution of a Linear Combination of chi-2 Random Variables, Journal of the Royal 
Statistical Society. Series C , 29, 323-333.
</p>
<p>Satterthwaite, F. (1946). An approximate distribution of estimates of variance components. Biom. Bull. 2, 110-114.
</p>
<p>Lee S, Emond MJ, Bamshad MJ, Barnes KC, Rieder MJ, Nickerson DA; NHLBI GO Exome Sequencing Project-ESP Lung Project Team, 
Christiani DC, Wurfel MM, Lin X. (2012) Optimal unified approach for rare variant association testing with application to small 
sample case-control whole-exome sequencing studies. American Journal of Human Genetics, 91, 224-237.
</p>
<p>Zhou, J. J. and Zhou, H.(2015) Powerful Exact Variance Component Tests for the Small Sample Next Generation Sequencing Studies 
(eVCTest), in submission.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(GUniFrac)

data(throat.tree)
data(throat.otu.tab)
data(throat.meta)

unifracs = GUniFrac(throat.otu.tab, throat.tree, alpha = c(1))$unifracs
if (requireNamespace("vegan")) {
 library(vegan)
 BC= as.matrix(vegdist(throat.otu.tab, method="bray"))
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"], BC = BC)
} else {
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"])
}
Ks = lapply(Ds, FUN = function(d) D2K(d))

covar = cbind(throat.meta$Age, as.numeric(throat.meta$Sex == "Male"))

# Continuous phenotype
n = nrow(throat.meta)
y = rnorm(n)
MiRKAT(y, X = covar, Ks = Ks, out_type="C", method = "davies")

# Binary phenotype 
y = as.numeric(runif(n) &lt; 0.5)
MiRKAT(y, X = covar, Ks = Ks, out_type="D")


</code></pre>

<hr>
<h2 id='MiRKAT_binary'>Microbiome Regression-Based Kernel Association Test for binary outcomes</h2><span id='topic+MiRKAT_binary'></span>

<h3>Description</h3>

<p>Called by MiRKAT if the outcome variable is dichotomous (out_type=&quot;D&quot;)
</p>
<p>This function is called by the exported function MiRKAT if the argument &quot;out_type&quot; of MiRKAT is equal to &quot;D&quot; (for dichotomous).
</p>
<p>Each argument of MiRKAT_continuous is given the value of the corresponding argument given by the user to MiRKAT.
</p>
<p>Function not exported
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKAT_binary(
  y,
  X = NULL,
  Ks,
  method = "davies",
  family = "binomial",
  omnibus = "permutation",
  nperm = 999,
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKAT_binary_+3A_y">y</code></td>
<td>
<p>A numeric vector of the dichotomous outcome variable</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_x">X</code></td>
<td>
<p>A numerical matrix or data frame, containing additional covariates that you want to adjust for (Default = NULL). If it is 
NULL, a intercept only model was fit.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_ks">Ks</code></td>
<td>
<p>A list of n by n kernel matrices (or a single n by n kernel matrix), where n is the sample size. It can be constructed 
from microbiome data through distance metric or other approaches, such as linear kernels or Gaussian kernels.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_method">method</code></td>
<td>
<p>A string telling R which method to use to compute the kernel specific p-value (default = &quot;davies&quot;). &quot;davies&quot; 
represents an exact method that computes the p-value by inverting the characteristic function of the mixture chisq. We adopt an 
exact variance component tests because most of the studies concerning microbiome compositions have modest sample size. &quot;moment&quot; 
represents an approximation method that matches the first two moments. &quot;permutation&quot; represents a permutation approach for 
p-value calculation.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_family">family</code></td>
<td>
<p>A string describing the error distribution and link function to be used in the linear model.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;permutation&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or residual permutation to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_nperm">nperm</code></td>
<td>
<p>the number of permutations if method = &quot;permutation&quot; or when multiple kernels are considered. if method = &quot;davies&quot; 
or &quot;moment&quot;, nperm is ignored.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MiRKAT_binary_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If only one candidate kernel matrix is considered, returns a list containing the p-value for the candidate kernel matrix. 
If more than one candidate kernel matrix is considered, returns a list with two elements: the individual p-values for
each candidate kernel matrix, and the omnibus p-value. 
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>p-value for each candidate kernel matrix</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>omnibus p-value if multiple kernel matrices are considered</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ni Zhao
</p>


<h3>References</h3>

<p>Zhao, N., Chen, J.,Carroll, I. M., Ringel-Kulka, T., Epstein, M.P., Zhou, H., Zhou, J. J., Ringel, Y., Li, H. and Wu, M.C. (2015)).
Microbiome Regression-based Kernel Association Test (MiRKAT). American Journal of Human Genetics, 96(5):797-807
</p>
<p>Chen, J., Chen, W., Zhao, N., Wu, M~C.and Schaid, D~J. (2016) Small Sample Kernel Association Tests for Human Genetic and 
Microbiome Association Studies. 40: 5-19. doi: 10.1002/gepi.21934
</p>
<p>Davies R.B. (1980) Algorithm AS 155: The Distribution of a Linear Combination of chi-2 Random Variables, Journal of the Royal
Statistical Society. Series C , 29, 323-333.
</p>
<p>Satterthwaite, F. (1946). An approximate distribution of estimates of variance components. Biom. Bull. 2, 110-114.
</p>
<p>Lee S, Emond MJ, Bamshad MJ, Barnes KC, Rieder MJ, Nickerson DA; NHLBI GO Exome Sequencing Project-ESP Lung Project Team, 
Christiani DC, Wurfel MM, Lin X. (2012) Optimal unified approach for rare variant association testing with application to 
small sample case-control whole-exome sequencing studies. American Journal of Human Genetics, 91, 224-237.
</p>
<p>Zhou, J. J. and Zhou, H.(2015) Powerful Exact Variance Component Tests for the Small Sample Next Generation Sequencing Studies 
(eVCTest), in submission.
</p>

<hr>
<h2 id='MiRKAT_continuous'>Microbiome Regression-based Analysis Test for a continuous outcome variable</h2><span id='topic+MiRKAT_continuous'></span>

<h3>Description</h3>

<p>Inner function for MiRKAT; computes MiRKAT for continuous outcomes. Called by MiRKAT if out_type=&quot;C&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKAT_continuous(
  y,
  X = NULL,
  Ks,
  method,
  omnibus,
  nperm = 999,
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKAT_continuous_+3A_y">y</code></td>
<td>
<p>A numeric vector of the continuous outcome variable</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_x">X</code></td>
<td>
<p>A numeric matrix or data frame containing additional covariates (default = NULL). If NULL, an intercept only model is used.</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_ks">Ks</code></td>
<td>
<p>A list of n by n kernel matrices (or a single n by n kernel matrix), where n is the sample size. It can be constructed 
from microbiome data through distance metric or other approaches, such as linear kernels or Gaussian kernels.</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_method">method</code></td>
<td>
<p>A method to compute the kernel specific p-value (Default= &quot;davies&quot;). &quot;davies&quot; represents an exact method that computes
the p-value by inverting the characteristic function of the mixture chisq. We adopt an exact variance component tests because most 
of the studies concerning microbiome compositions have modest sample size. &quot;moment&quot; represents an approximation method that matches
the first two moments. &quot;permutation&quot; represents a permutation approach for p-value calculation.</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;permutation&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or residual permutation to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_nperm">nperm</code></td>
<td>
<p>the number of permutations if method = &quot;permutation&quot; or when multiple kernels are considered. If method = &quot;davies&quot; or 
&quot;moment&quot;, nperm is ignored. Defaults to 999.</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MiRKAT_continuous_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is called by the exported function &quot;MiRKAT&quot; when the argument of MiRKAT, out_type, is set equal to &quot;C&quot;. 
</p>
<p>Each argument of MiRKAT_continuous is given the value of the corresponding argument given by the user to MiRKAT.
</p>
<p>Function not exported
</p>


<h3>Value</h3>

<p>If only one candidate kernel matrix is considered, returns a list containing the p-value for the candidate kernel matrix. 
If more than one candidate kernel matrix is considered, returns a list of two elements: the individual p-values for
each candidate kernel matrix, and the omnibus p-value
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>p-value for each candidate kernel matrix</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>omnibus p-value considering multiple candidate kernel matrices</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Ni Zhao
</p>


<h3>References</h3>

<p>Zhao, N., Chen, J.,Carroll, I. M., Ringel-Kulka, T., Epstein, M.P., Zhou, H., Zhou, J. J., Ringel, Y., Li, H. and Wu, M.C. (2015)).
Microbiome Regression-based Kernel Association Test (MiRKAT). American Journal of Human Genetics, 96(5):797-807
</p>
<p>Chen, J., Chen, W., Zhao, N., Wu, M~C.and Schaid, D~J. (2016) Small Sample Kernel Association Tests for Human Genetic and 
Microbiome Association Studies. 40: 5-19. doi: 10.1002/gepi.21934
</p>
<p>Davies R.B. (1980) Algorithm AS 155: The Distribution of a Linear Combination of chi-2 Random Variables, Journal of the Royal
Statistical Society. Series C , 29, 323-333.
</p>
<p>Satterthwaite, F. (1946). An approximate distribution of estimates of variance components. Biom. Bull. 2, 110-114.
</p>
<p>Lee S, Emond MJ, Bamshad MJ, Barnes KC, Rieder MJ, Nickerson DA; NHLBI GO Exome Sequencing Project-ESP Lung Project Team, 
Christiani DC, Wurfel MM, Lin X. (2012) Optimal unified approach for rare variant association testing with application to small
sample case-control whole-exome sequencing studies. American Journal of Human Genetics, 91, 224-237.
</p>
<p>Zhou, J. J. and Zhou, H.(2015) Powerful Exact Variance Component Tests for the Small Sample Next Generation Sequencing Studies 
(eVCTest), in submission.
</p>

<hr>
<h2 id='MiRKAT.iQ'>MiRKAT-iQ</h2><span id='topic+MiRKAT.iQ'></span>

<h3>Description</h3>

<p>Integrated quantile regression-based kernel association test.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKAT.iQ(Y, X, K, weight = c(0.25, 0.25, 0.25, 0.25))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKAT.iQ_+3A_y">Y</code></td>
<td>
<p>A numeric vector of the continuous outcome variable.</p>
</td></tr>
<tr><td><code id="MiRKAT.iQ_+3A_x">X</code></td>
<td>
<p>A numeric matrix for additional covariates that you want to adjust for.</p>
</td></tr>
<tr><td><code id="MiRKAT.iQ_+3A_k">K</code></td>
<td>
<p>A list of n by n kernel matrices at a single n by n kernel matrix, where n is the sample size.</p>
</td></tr>
<tr><td><code id="MiRKAT.iQ_+3A_weight">weight</code></td>
<td>
<p>A length 4 vector specifying the weight for Cauchy combination, corresponding to wilcoxon/normal/inverselehmann/lehmann functions. The sum of the weight should be 1.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Returns a list containing the p values for single kernels, or the omnibus p-value if multiple candidate kernel matrices are provided.
</p>


<h3>Author(s)</h3>

<p>Tianying Wang, Xiang Zhan.
</p>


<h3>References</h3>

<p>Wang T, et al. (2021) Testing microbiome association using integrated quantile regression models. Bioinformatics (to appear).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(GUniFrac)
library(quantreg)
library(PearsonDS)
library(MiRKAT)

data(throat.tree)
data(throat.otu.tab)

## Create UniFrac and Bray-Curtis distance matrices 
unifracs = GUniFrac(throat.otu.tab, throat.tree, alpha = c(1))$unifracs
if (requireNamespace("vegan")) {
 library(vegan)
 BC= as.matrix(vegdist(throat.otu.tab, method="bray"))
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"], BC = BC)
} else {
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"])
}

## Convert to kernels           
Ks = lapply(Ds, FUN = function(d) D2K(d))
covar = cbind(throat.meta$Age, as.numeric(throat.meta$Sex == "Male"))
n = nrow(throat.meta)
y = rnorm(n)
result = MiRKAT.iQ(y, X = covar, K = Ks)

</code></pre>

<hr>
<h2 id='MiRKAT.Q'>Robust MiRKAT (quantile regression)</h2><span id='topic+MiRKAT.Q'></span>

<h3>Description</h3>

<p>A more robust version of MiRKAT utlizing a linear model that uses quantile regression.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKAT.Q(y, X, Ks, omnibus = "kernel_om", returnKRV = FALSE, returnR2 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKAT.Q_+3A_y">y</code></td>
<td>
<p>A numeric vector of the a continuous or dichotomous outcome variable.</p>
</td></tr>
<tr><td><code id="MiRKAT.Q_+3A_x">X</code></td>
<td>
<p>A numerical matrix or data frame, containing additional covariates that you want to adjust for. Mustn't be NULL.</p>
</td></tr>
<tr><td><code id="MiRKAT.Q_+3A_ks">Ks</code></td>
<td>
<p>A list of n by n kernel matrices (or a single n by n kernel matrix), where n is the sample size. If you have distance metric from metagenomic data, each kernel can 
be constructed through function D2K. Each kernel can also be constructed through other mathematical approaches, such as linear or Gaussian kernels.</p>
</td></tr>
<tr><td><code id="MiRKAT.Q_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;kernel_om&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or an omnibus kernel to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="MiRKAT.Q_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MiRKAT.Q_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MiRKAT.Q creates a kernel matrix using the linear model created with the function rq, a quantile regression function, then does 
the KRV analysis on Ks and the newly formed kernel matrix representing the outome traits. 
</p>
<p>Missing data is not permitted. Please remove all individuals with missing y, X, Ks prior to analysis
</p>


<h3>Value</h3>

<p>Returns p-values for each individual kernel matrix, an omnibus p-value if multiple kernels were provided, and measures of effect size KRV and R2. 
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>labeled individual p-values for each kernel</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>omnibus p_value, calculated as for the KRV test</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Weija Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    
library(GUniFrac)
library(quantreg)

# Generate data
data(throat.tree)
data(throat.otu.tab)
data(throat.meta)

unifracs = GUniFrac(throat.otu.tab, throat.tree, alpha = c(1))$unifracs
if (requireNamespace("vegan")) {
 library(vegan)
 BC= as.matrix(vegdist(throat.otu.tab, method="bray"))
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"], BC = BC)
} else {
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"])
}
Ks = lapply(Ds, FUN = function(d) D2K(d))

covar = scale(cbind(throat.meta$Age, as.numeric(throat.meta$Sex == "Male")))

# Continuous phenotype
n = nrow(throat.meta)
y = rchisq(n, 2) + apply(covar, 1, sum) 
MiRKAT.Q(y, X = covar, Ks = Ks)

</code></pre>

<hr>
<h2 id='MiRKAT.R'>Robust MiRKAT (robust regression)</h2><span id='topic+MiRKAT.R'></span>

<h3>Description</h3>

<p>A more robust version of MiRKAT utilizing a linear model by robust regression using an M estimator.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKAT.R(y, X, Ks, omnibus = "kernel_om", returnKRV = FALSE, returnR2 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKAT.R_+3A_y">y</code></td>
<td>
<p>A numeric vector of the a continuous or dichotomous outcome variable.</p>
</td></tr>
<tr><td><code id="MiRKAT.R_+3A_x">X</code></td>
<td>
<p>A numerical matrix or data frame, containing additional covariates that you want to adjust for Mustn't be NULL</p>
</td></tr>
<tr><td><code id="MiRKAT.R_+3A_ks">Ks</code></td>
<td>
<p>list of n by n kernel matrices (or a single n by n kernel matrix), where n is the sample size. It can be constructed from 
microbiome data through distance metric or other approaches, such as linear kernels or Gaussian kernels.</p>
</td></tr>
<tr><td><code id="MiRKAT.R_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;kernel_om&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or an omnibus kernel to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="MiRKAT.R_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MiRKAT.R_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MiRKAT.R creates a kernel matrix using the linear model created with the function rlm, a robust regression function, then does 
the KRV analysis on Ks and the newly formed kernel matrix representing the outome traits. 
</p>
<p>y and X should all be numerical matrices or vectors with the same number of rows, and mustn't be NULL.
</p>
<p>Ks should be a list of n by n matrices or a single matrix. If you have distance metric from metagenomic data, each kernel can be 
constructed through function D2K. Each kernel may also be constructed through other mathematical approaches.
</p>
<p>Missing data is not permitted. Please remove all individuals with missing y, X, Ks prior to analysis
</p>


<h3>Value</h3>

<p>Returns p-values for each individual kernel matrix, an omnibus p-value if multiple kernels were provided, and measures of effect size KRV and R2. 
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>labeled individual p-values for each kernel</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>omnibus p_value, calculated as for the KRV test</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Weijia Fu
</p>


<h3>Examples</h3>

<pre><code class='language-R'>    


# Generate data
library(GUniFrac)

data(throat.tree)
data(throat.otu.tab)
data(throat.meta)

unifracs = GUniFrac(throat.otu.tab, throat.tree, alpha = c(1))$unifracs
if (requireNamespace("vegan")) {
 library(vegan)
 BC= as.matrix(vegdist(throat.otu.tab, method="bray"))
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"], BC = BC)
} else {
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"])
}
Ks = lapply(Ds, FUN = function(d) D2K(d))

covar = cbind(throat.meta$Age, as.numeric(throat.meta$Sex == "Male"))

# Continuous phenotype
n = nrow(throat.meta)
y = rchisq(n, 2)
MiRKAT.R(y, X = covar, Ks = Ks)

</code></pre>

<hr>
<h2 id='MiRKATS'>Microiome Regression-based Kernel Association Test for Survival</h2><span id='topic+MiRKATS'></span>

<h3>Description</h3>

<p>Community level test for association between microbiome composition and survival outcomes (right-censored time-to-event data) 
using kernel matrices to compare similarity between microbiome profiles with similarity in survival times.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiRKATS(
  obstime,
  delta,
  X = NULL,
  Ks,
  beta = NULL,
  perm = FALSE,
  omnibus = "permutation",
  nperm = 999,
  returnKRV = FALSE,
  returnR2 = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiRKATS_+3A_obstime">obstime</code></td>
<td>
<p>A numeric vector of follow-up (survival/censoring) times.</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_delta">delta</code></td>
<td>
<p>Event indicator: a vector of 0/1, where 1 indicates that the event was observed for a subject (so &quot;obstime&quot; is 
survival time), and 0 indicates that the subject was censored.</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_x">X</code></td>
<td>
<p>A vector or matrix of numeric covariates, if applicable (default = NULL).</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_ks">Ks</code></td>
<td>
<p>A list of or a single numeric n by n kernel matrices or matrix (where n is the sample size).</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_beta">beta</code></td>
<td>
<p>A vector of coefficients associated with covariates. If beta is NULL and covariates are present, coxph is used to 
calculate coefficients (default = NULL).</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_perm">perm</code></td>
<td>
<p>Logical, indicating whether permutation should be used instead of analytic p-value calculation (default=FALSE). 
Not recommended for sample sizes of 100 or more.</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_omnibus">omnibus</code></td>
<td>
<p>A string equal to either &quot;Cauchy&quot; or &quot;permutation&quot; (or nonambiguous abbreviations thereof), specifying whether 
to use the Cauchy combination test or residual permutation to generate the omnibus p-value.</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_nperm">nperm</code></td>
<td>
<p>Integer, number of permutations used to calculate p-value if perm==TRUE (default=1000) and to calculate omnibus p-value if omnibus==&quot;permutation&quot;.</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MiRKATS_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>obstime, delta, and X should all have n rows, and the kernel or distance matrix should be a single n by n matrix. 
If a distance matrix is entered (distance=TRUE), a kernel matrix will be constructed from the distance matrix.
</p>
<p>Update in v1.1.0: MiRKATS also utilizes the OMiRKATS omnibus test if more than one kernel matrix is provided by the user. 
The OMiRKATS omnibus test calculates an overall p-value for the test via permutation. 
</p>
<p>Missing data is not permitted. Please remove individuals with missing data on y, X or in the kernel or distance matrix prior 
to using the function.
</p>
<p>The Efron approximation is used for tied survival times.
</p>


<h3>Value</h3>

<p>Return value depends on the number of kernel matrices inputted. If more than one kernel matrix is given, MiRKATS returns two
items; a vector of the labeled individual p-values for each kernel matrix, as well as an omnibus p-value from the Optimal-MiRKATS
omnibus test. If only one kernel matrix is given, then only its p-value will be given, as no omnibus test will be needed.
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>individual p-values for each inputted kernel matrix</p>
</td></tr>
<tr><td><code>omnibus_p</code></td>
<td>
<p>overall omnibus p-value</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nehemiah Wilson, Anna Plantinga
</p>


<h3>References</h3>

<p>Plantinga, A., Zhan, X., Zhao, N., Chen, J., Jenq, R., and Wu, M.C. MiRKAT-S: a distance-based test of association between
microbiome composition and survival times. Microbiome, 2017:5-17. doi: 10.1186/s40168-017-0239-9
</p>
<p>Zhao, N., Chen, J.,Carroll, I. M., Ringel-Kulka, T., Epstein, M.P., Zhou, H., Zhou, J. J., Ringel, Y., Li, H. and Wu, M.C. (2015)).
Microbiome Regression-based Kernel Association Test (MiRKAT). American Journal of Human Genetics, 96(5):797-807
</p>
<p>Chen, J., Chen, W., Zhao, N., Wu, M~C.and Schaid, D~J. (2016) Small Sample Kernel Association Tests for Human Genetic and
Microbiome Association Studies. 40:5-19. doi: 10.1002/gepi.21934
</p>
<p>Efron, B. (1977) &quot;The efficiency of Cox's likelihood function for censored data.&quot; Journal of the American statistical 
Association 72(359):557-565.
</p>
<p>Davies R.B. (1980) Algorithm AS 155: The Distribution of a Linear Combination of chi-2 Random Variables, Journal of the Royal 
Statistical Society Series C, 29:323-333
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
###################################
# Generate data
library(GUniFrac)


# Throat microbiome data
data(throat.tree)
data(throat.otu.tab)

unifracs = GUniFrac(throat.otu.tab, throat.tree, alpha = c(1))$unifracs
if (requireNamespace("vegan")) {
 library(vegan)
 BC= as.matrix(vegdist(throat.otu.tab, method="bray"))
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"], BC = BC)
} else {
 Ds = list(w = unifracs[,,"d_1"], uw = unifracs[,,"d_UW"])
}

Ks = lapply(Ds, FUN = function(d) D2K(d))

# Covariates and outcomes
covar &lt;- matrix(rnorm(120), nrow=60)
S &lt;- rexp(60, 3)   # survival time 
C &lt;- rexp(60, 1)   # censoring time 
D &lt;- (S&lt;=C)        # event indicator
U &lt;- pmin(S, C)    # observed follow-up time

MiRKATS(obstime = U, delta = D, X = covar, Ks = Ks, beta = NULL)


</code></pre>

<hr>
<h2 id='MMiRKAT'>Multivariate Microbiome Regression-based Kernel Association Test</h2><span id='topic+MMiRKAT'></span>

<h3>Description</h3>

<p>Test for association between overall microbiome composition and multiple continuous outcomes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMiRKAT(Y, X = NULL, Ks, returnKRV = FALSE, returnR2 = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMiRKAT_+3A_y">Y</code></td>
<td>
<p>A numerical n by p matrix of p continuous outcome variables, n being sample size.</p>
</td></tr>
<tr><td><code id="MMiRKAT_+3A_x">X</code></td>
<td>
<p>A numerical n by q matrix or data frame, containing q additional covariates that you want to adjust for (Default = NULL).
If it is NULL, an intercept only model is fit.</p>
</td></tr>
<tr><td><code id="MMiRKAT_+3A_ks">Ks</code></td>
<td>
<p>A list of numerical n by n kernel matrices, or a single n by n kernel matrix, where n is the sample size. Kernels can be 
constructed from distance matrices (such as Bray-Curtis or UniFrac distances) using the function D2K, or through other mathematical approaches.</p>
</td></tr>
<tr><td><code id="MMiRKAT_+3A_returnkrv">returnKRV</code></td>
<td>
<p>A logical indicating whether to return the KRV statistic. Defaults to FALSE.</p>
</td></tr>
<tr><td><code id="MMiRKAT_+3A_returnr2">returnR2</code></td>
<td>
<p>A logical indicating whether to return the R-squared coefficient. Defaults to FALSE.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Missing data is not permitted. Please remove all individuals with missing Y, X, K prior to analysis
</p>
<p>The method of generating kernel specific p-values is &quot;davies&quot;, which represents an exact method that computes the p-value by
inverting the characteristic function of the mixture chisq.
</p>


<h3>Value</h3>

<p>Returns a list of the MMiRKAT p-values for each inputted kernel matrix, labeled with the names of the kernels, if given.
</p>
<table>
<tr><td><code>p_values</code></td>
<td>
<p>list of the p-values for each individual kernel matrix inputted</p>
</td></tr>
<tr><td><code>KRV</code></td>
<td>
<p>A vector of kernel RV statistics (a measure of effect size), one for each candidate kernel matrix. Only returned if returnKRV = TRUE</p>
</td></tr>
<tr><td><code>R2</code></td>
<td>
<p>A vector of R-squared statistics, one for each candidate kernel matrix. Only returned if returnR2 = TRUE</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Nehemiah Wilson, Haotian Zheng, Xiang Zhan, Ni Zhao
</p>


<h3>References</h3>

<p>Zheng, H., Zhan, X., Tong, X., Zhao, N., Maity,A., Wu, M.C., and Chen,J. A small-sample multivariate kernel machine test for 
microbiome association studies. Genetic Epidemiology, 41(3), 210-220. DOI: 10.1002/gepi.22030
</p>


<h3>Examples</h3>

<pre><code class='language-R'> 
library(GUniFrac)
if(requireNamespace("vegan")) { library(vegan) }

data(throat.tree)
data(throat.otu.tab)
data(throat.meta)

unifracs &lt;- GUniFrac(throat.otu.tab, throat.tree, alpha=c(0, 0.5, 1))$unifracs

if(requireNamespace("vegan")) {
 BC= as.matrix(vegdist(throat.otu.tab , method="bray"))
 Ds = list(w = unifracs[,,"d_1"], u = unifracs[,,"d_UW"], BC = BC) 
} else {
 Ds = list(w = unifracs[,,"d_1"], u = unifracs[,,"d_UW"]) 
}
Ks = lapply(Ds, FUN = function(d) D2K(d)) 

n = nrow(throat.otu.tab)
Y = matrix(rnorm(n*3, 0, 1), n, 3)

covar = cbind(as.numeric(throat.meta$Sex == "Male"), as.numeric(throat.meta$PackYears))
MMiRKAT(Y = Y, X = covar, Ks = Ks) 

</code></pre>

<hr>
<h2 id='nordata'>Simulated DEPENDENT data with GAUSSIAN traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)</h2><span id='topic+nordata'></span>

<h3>Description</h3>

<p>Simulated DEPENDENT data with GAUSSIAN traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(nordata)
</code></pre>


<h3>Format</h3>

<p>A list containing three data objects for correlated microbiome data with continuous response variable (described below). 
</p>

<dl>
<dt>nor.otu.tab</dt><dd><p>Simulated OTU data for correlated regression-based analysis; 59 rows and 730 columns, rows being patients and 
columns being OTUs</p>
</dd>
<dt>nor.meta</dt><dd><p>Simulated metadata for correlated regression-based analysis; 59 rows and 4 columns, rows being patients and 
columns being the outcome variable, subject identifier, and covariates to possibly account for in any regression modeling</p>
</dd>
<dt>nor.tree</dt><dd><p>Simulated rooted phylogenetic tree with 730 tips and 729 nodes</p>
</dd>
</dl>


<hr>
<h2 id='poisdata'>Simulated DEPENDENT data with POISSON (count) traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)</h2><span id='topic+poisdata'></span>

<h3>Description</h3>

<p>Simulated DEPENDENT data with POISSON (count) traits for correlated regression-based analysis (i.e. CSKAT, GLMMMiRKAT)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(poisdata)
</code></pre>


<h3>Format</h3>

<p>A list containing three data objects for correlated microbiome data with binary response variable (described below). 
</p>

<dl>
<dt>pois.otu.tab</dt><dd><p>Simulated OTU data for correlated regression-based analysis; 59 rows and 730 columns, rows being patients and 
columns being OTUs</p>
</dd>
<dt>pois.meta</dt><dd><p>Simulated metadata for correlated regression-based analysis; 59 rows and 4 columns, rows being patients and 
columns being the outcome variable, subject identifier, and covariates to possibly account for in any regression modeling</p>
</dd>
<dt>pois.tree</dt><dd><p>Simulated rooted phylogenetic tree with 730 tips and 729 nodes</p>
</dd>
</dl>


<hr>
<h2 id='throat.meta'>Simulated metadata for microbiome regression-based analysis</h2><span id='topic+throat.meta'></span>

<h3>Description</h3>

<p>Simulation code can be seen in ?KRV
Corresponding OTU matrix is stored in &quot;throat.otu.tab&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.meta)
</code></pre>


<h3>Format</h3>

<p>A data frame with 59 rows and 16 columns, rows being participants and columns being different covariates to possibly be
accounted for in any utilized linear models.
</p>
<p>throat.meta is part of a microbiome data set for studying the effect of smoking on the upper respiratory tract microbiome. This data set 
comes from the throat microbiome of left body side. It contains 60 subjects consisting of 32 nonsmokers and 28 smokers.
</p>

<dl>
<dt>BarcodeSequence</dt><dd><p>Sequence of DNA that allows for the identification of the specific species of bacteria. See GUniFrac for 
more details</p>
</dd>
<dt>LinkerPrimerSequence</dt><dd><p>Sequence of DNA that aids in locating the Barcode Sequence. See GUniFrac for more details</p>
</dd>
<dt>SmokingStatus</dt><dd><p>whether or not each patient is a &quot;Smoker&quot; or a &quot;NonSmoker&quot;</p>
</dd>
<dt>PatientID</dt><dd><p>Identifying integer label given to each patient</p>
</dd>
<dt>SampleIndex</dt><dd><p>Labels each patient as being from this particular sample, so as possibly be able to use multiple samples
at once</p>
</dd>
<dt>AirwaySite</dt><dd><p>Part of body where our samples were taken from from in each participant</p>
</dd>
<dt>SideOfBody</dt><dd><p>Which side of the body the samples were taken from</p>
</dd>
<dt>SampleType</dt><dd><p>What kind of sample each one is; should all be patientsamples</p>
</dd>
<dt>RespiratoryDiseaseStatus_severity_timeframe</dt><dd><p>Whether or not the patient has had a respiratory disease, and if so which 
one_severity of said disease_whether or not that disease is still active. If there has been no such disease in the patient's
medical history, the patient's value is &quot;healthy&quot; in this column</p>
</dd>
<dt>AntibioticUsePast3Months_TimeFromAntibioticUsage</dt><dd><p>Whether or not the patient has used antibiotics in the past
month_if so, how long ago it was. If not antibiotics have been used in the past month, the patient's value is &quot;None&quot;
in this column</p>
</dd>
<dt>Age</dt><dd><p>Age of the patient</p>
</dd>
<dt>Sex</dt><dd><p>The sex of the patient</p>
</dd>
<dt>PackYears</dt><dd><p>Unit of measurement measuring the intensity of smoking; average number of packs per day times the number of 
years the patient has been smoking. If patient has never smoked, their value is 0 for this column</p>
</dd>
<dt>TimeFromLastCig</dt><dd><p>Minutes since the patient's last cigarette</p>
</dd>
<dt>TimeFromLastMeal</dt><dd><p>Minutes since the patient's last meal</p>
</dd>
<dt>Description</dt><dd><p>See Charleston paper and other sources</p>
</dd>
</dl>


<hr>
<h2 id='throat.otu.tab'>Simulated OTU data for microbiome regression analysis</h2><span id='topic+throat.otu.tab'></span>

<h3>Description</h3>

<p>Simulated code can be seen in ?KRV
Corresponding metadata is stored in &quot;throat.meta&quot;
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.otu.tab)
</code></pre>


<h3>Format</h3>

<p>60 rows and 856 columns, where rows are patients and columns are OTUs
</p>

<hr>
<h2 id='throat.tree'>Simulated rooted phylogenetic tree</h2><span id='topic+throat.tree'></span>

<h3>Description</h3>

<p>Simulation code can be seen in ?KRV
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.tree)
</code></pre>


<h3>Format</h3>

<p>Phylogenetic tree with 856 tips and 855 internal nodes
</p>


<h3>Details</h3>

<p>Corresponding OTU matrix stored in &quot;throat.otu.tab&quot;
See the GUniFrac package for more details
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
