<!DOCTYPE html><html><head><title>Help for package MiRSEA</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MiRSEA}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#Corrp2miRfile'><p>create a pathway-microRNA(miRNA) profile and a p value matrix</p></a></li>
<li><a href='#EnrichmentScore'><p>Computes the  enrichment score</p></a></li>
<li><a href='#EnrichmentScore2'><p>Computes the enrichment score faster</p></a></li>
<li><a href='#envData'><p>The variables in the environment include predefine pathway, target information of miRNAs,an expression profile and a example result of miRNA list</p></a></li>
<li><a href='#GetExampleData'><p>Get the example data</p></a></li>
<li><a href='#GetMiRTargetData'><p>Get the data of microRNA(miRNA) and target genes</p></a></li>
<li><a href='#GetPathwayData'><p>Get the gene sets of pathways</p></a></li>
<li><a href='#HeatMapPlot'><p>Plot a heatmap</p></a></li>
<li><a href='#MirSEA'><p>Identify dysregulated pathways based on microRNA (miRNA) set enrichment analysis</p></a></li>
<li><a href='#MiRSEA-internal'><p>MiRSEA internal functions</p></a></li>
<li><a href='#MiRSEA-package'><p>MicroRNA Set Enrichment Analysis</p></a></li>
<li><a href='#MsReport'><p>Produce report for a microRNA (abbreviated miRNA) set</p></a></li>
<li><a href='#PlotCorrelation'><p>Plot global microRNA(miRNA) correlation profile</p></a></li>
<li><a href='#PlotHeatMap'><p>Plot a heat map</p></a></li>
<li><a href='#PlotRunEnrichment'><p>Plot running microRNAs(miRNAs) enrichment score</p></a></li>
<li><a href='#S2N'>
<p>calculate signal to noise ratio for microRNAs(miRNAs)</p>
</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Version:</td>
<td>1.1.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2015-06-30</td>
</tr>
<tr>
<td>Title:</td>
<td>'MicroRNA' Set Enrichment Analysis</td>
</tr>
<tr>
<td>Author:</td>
<td>Junwei Han, Siyao Liu </td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Junwei Han &lt;hanjunwei1981@163.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The tools for 'MicroRNA Set Enrichment Analysis' can identify risk pathways(or prior gene sets) regulated by microRNA set in the context of microRNA expression data. (1) This package constructs a correlation profile of microRNA and pathways by the hypergeometric statistic test. The gene sets of pathways derived from the three public databases (Kyoto Encyclopedia of Genes and Genomes ('KEGG'); 'Reactome'; 'Biocarta') and the target gene sets of microRNA are provided by four databases('TarBaseV6.0'; 'mir2Disease'; 'miRecords'; 'miRTarBase';). (2) This package can quantify the change of correlation between microRNA for each pathway(or prior gene set) based on a microRNA expression data with cases and controls. (3) This package uses the weighted Kolmogorov-Smirnov statistic to calculate an enrichment score (ES) of a microRNA set that co-regulate to a pathway , which reflects the degree to which a given pathway is associated with the specific phenotype. (4) This package can provide the visualization of the results.</td>
</tr>
<tr>
<td>Collate:</td>
<td>Corrp2miRfile.R EnrichmentScore.R S2N.R MirSEA.R MsReport.R
GetExampleData.R getEnvironmentData.R GetMiRTargetData.R
GetPathwayData.R PlotCorrelation.R PlotHeatMap.R
PlotRunEnrichment.R</td>
</tr>
<tr>
<td>LazyData:</td>
<td>Yes</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.15.1)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>stats,graphics,grDevices</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Statistics, Pathways, microRNA, enrichment analysis</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-23 14:38:29 UTC; hornik</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-24 09:49:16 UTC</td>
</tr>
</table>
<hr>
<h2 id='Corrp2miRfile'>create a pathway-microRNA(miRNA) profile and a p value matrix </h2><span id='topic+Corrp2miRfile'></span>

<h3>Description</h3>

<p>The function  <code>Corrp2miRfile</code> create a p value matrix and a pathway-miRNA correlation profile  
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Corrp2miRfile(pathway="kegg", species = "hsa")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Corrp2miRfile_+3A_pathway">pathway</code></td>
<td>
<p>choose database of pathway,&quot;kegg&quot;,&quot;biocarta&quot; or&quot;reactome&quot;</p>
</td></tr>
<tr><td><code id="Corrp2miRfile_+3A_species">species</code></td>
<td>
<p>Species of miRNAs(default: hsa)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When users input interesting species and pathway, the function can calculate the p value between pathway and miRNA using hypergeometric.The p value can quantify the strength of the pathway regulated by each miRNA.The smaller p value is represent the bigger  strength of regulate.Then p2m can get miRNA set(pmSET)for each pathway,which is a co-regulated miRNA set of this pathway(w&gt;0).
</p>


<h3>Value</h3>

<table>
<tr><td><code>p</code></td>
<td>
<p>A p value weighted matrix (rows are pathway ,cols are miRNAs)</p>
</td></tr>
<tr><td><code>p2miR</code></td>
<td>
<p>pathway-miRNA correlation(pmSET) profile</p>
</td></tr> 
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>References</h3>

<p>Rivals I, Personnaz L, Taing L, &amp; Potier MC (2007) Enrichment or depletion of a GO category within a class of genes: which test? (Translated from eng) Bioinformatics 23(4):401-407 (in eng).
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MirSEA">MirSEA</a></code>,<code><a href="#topic+MsReport">MsReport</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
p2m&lt;-Corrp2miRfile(pathway="kegg", species = "example")

p2m$p[1,1:10]
p2m$p2miR[1,1:5]

## End(Not run)
</code></pre>

<hr>
<h2 id='EnrichmentScore'>Computes the  enrichment score </h2><span id='topic+EnrichmentScore'></span>

<h3>Description</h3>

<p>Computes the enrichment score of a microRNA(miRNA) set in a ordered miRNA list.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>     EnrichmentScore(miR.list, miR.set, weighted.score.type = 1, correl.vector = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EnrichmentScore_+3A_mir.list">miR.list</code></td>
<td>

<p>The ordered miRNA list ,integers indicating the original position in the input dataset.
</p>
</td></tr>
<tr><td><code id="EnrichmentScore_+3A_mir.set">miR.set</code></td>
<td>

<p>A miRNA set ,integers indicating the location of those miRNAs in the input dataset.
</p>
</td></tr>
<tr><td><code id="EnrichmentScore_+3A_weighted.score.type">weighted.score.type</code></td>
<td>

<p>Type of score,weight=0,ES reduces to the standard Kolmogorov-Smirnov statistic,when weight=1, we are weighting the miRNAs by their tw-score normalized by the sum of the tw-scores over all of the miRNAs in the miRNA set.
</p>
</td></tr>
<tr><td><code id="EnrichmentScore_+3A_correl.vector">correl.vector</code></td>
<td>

<p>A vector with the correlations(tw-scores) corresponding to the miRNAs in the miRNA list 
</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function can computes the enrichment score of a miRNA set in a miRNA list.  The weighted score type is the exponent of the correlation(e.g.tw-score)
(1) Rank order the miRNAs in a miRNA set to form a list according to the correlation(e.g.tw-score) of their expression profiles and regulated pathway
(2) Evaluate the fraction of miRNAs in the miRNA set(hits) weighted by their correlation and the fraction of miRNAs not in the miRNA set(misses)present up to a given position i in the miRNA list.The ES is the maximum deviation from zero of 'P(hit)-P(miss)'.
For a randomly distributed miRNA set,The enrichment score will be relatively small, but if it is concentrated at the top or bottom of the list,or otherwise nonrandomly distributed, then the Enrichment score will be correspondingly high.
</p>


<h3>Value</h3>

<table>
<tr><td><code>ES</code></td>
<td>
<p>Enrichment score.</p>
</td></tr>
<tr><td><code>arg.ES</code></td>
<td>
<p>Location in the miRNA list where the peak running enrichment occurs.</p>
</td></tr>
<tr><td><code>RES</code></td>
<td>
<p>Numerical vector containing the running enrichment score for all locations in the miRNA list.</p>
</td></tr>
<tr><td><code>tag.indicator</code></td>
<td>
<p>Binary vector indicating the location of the miRNA sets in the miRNA list.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+EnrichmentScore2">EnrichmentScore2</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Computes the enrichment score of a miRNA set in a ordered miRNA list.
E1&lt;-EnrichmentScore(miR.list=sample(1:1000),miR.set=c(39,281,37,381,39,11,3,34),
correl.vector=rep(0.3,1000))
#show results
#EnrichmentScore of this set
E1$ES
#peak running enrichment
E1$arg.ES
#running enrichment score of top ten miRNAs
E1$RES[1:10]
#Binary vector indicating the location of top ten miRNA in the miRNA list
E1$tag.indicator[1:10]

</code></pre>

<hr>
<h2 id='EnrichmentScore2'>Computes the enrichment score faster</h2><span id='topic+EnrichmentScore2'></span>

<h3>Description</h3>

<p>Computes the enrichment score of a microRNA(miRNA) set in miRNA list.</p>


<h3>Usage</h3>

<pre><code class='language-R'>EnrichmentScore2(miR.list, miR.set, weighted.score.type = 1, correl.vector = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="EnrichmentScore2_+3A_mir.list">miR.list</code></td>
<td>
<p>The ordered miRNA list,integers indicating the original position in the input dataset </p>
</td></tr>
<tr><td><code id="EnrichmentScore2_+3A_mir.set">miR.set</code></td>
<td>
<p>A miRNA set,integers indicating the location of those miRNAs in the input dataset</p>
</td></tr>
<tr><td><code id="EnrichmentScore2_+3A_weighted.score.type">weighted.score.type</code></td>
<td>
<p>Type of score, weight=0 (unweighted = Kolmogorov-Smirnov), 1 (weighted), and 2 (over-weighted)</p>
</td></tr>
<tr><td><code id="EnrichmentScore2_+3A_correl.vector">correl.vector</code></td>
<td>
<p>A vector with the correlations (e.g.tw-scores) corresponding to the miRNAs in the miRNA list </p>
</td></tr>
</table>


<h3>Details</h3>

<p>Computes the weighted enrichment score of a miRNA set in miRNA list. It is the same calculation as in 
EnrichmentScore but faster without producing the RES, arg.RES and tag.indicator outputs.
This call is intended to be used to asses the enrichment of random permutations rather than the 
observed one.The weighted score type is the exponent of the correlation. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>ES</code></td>
<td>
<p>Enrichment score (real number between -1 and +1)</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+EnrichmentScore">EnrichmentScore</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Computes the enrichment score of a miRNA set in miRNA list
R2&lt;-EnrichmentScore2(miR.list=sample(1:100),miR.set=c(39,28,37,38,11,3,34),
correl.vector=rep(0.04,100))
#show the result
R2$ES
</code></pre>

<hr>
<h2 id='envData'>The variables in the environment include predefine pathway, target information of miRNAs,an expression profile and a example result of miRNA list</h2><span id='topic+envData'></span><span id='topic+pathway'></span><span id='topic+mfile'></span><span id='topic+example.cls'></span><span id='topic+example.gct'></span><span id='topic+p'></span><span id='topic+p2miR'></span><span id='topic+miRList'></span>

<h3>Description</h3>

<p>The pathway information is download on the GSEA website,concluding three pathway database
(KEGG,Biocarta,Reactome).We arranged the data for miRNAs and their target genes,which is according to four database including miRTarBase,TarBaseV6.0,miRecords and mir2Disease.
example.GCT is an interesting miRNA expression data and example.cls is the vector of binary labels(class.labels)
p is a p value weighted matrix (rows are pathway ,cols are miRNAs).p2miR is a correlation profile between kegg pathways and each human miRNA.
miRList is a list of drawing parameters of KEGG ERBB signaling Pathway.
</p>


<h3>Format</h3>

<p>An environment variable</p>


<h3>Details</h3>

<p>The environment variable includes the variable <code>pathway</code>, <code>mfile</code>,<code>example.cls</code>,<code>example.gct</code>,<code>p</code>,<code>p2miR</code>,<code>miRList</code>
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>

<hr>
<h2 id='GetExampleData'>Get the example data</h2><span id='topic+GetExampleData'></span>

<h3>Description</h3>

<p>Get the example data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>  GetExampleData(exampleData)
  
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetExampleData_+3A_exampledata">exampleData</code></td>
<td>
<p> A character string, must be one of &quot;dataset&quot;, &quot;class.labels&quot; , &quot;miRList&quot;,&quot;p_value&quot;and &quot;p2miR&quot;.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function GetExampleData(exampleData=&quot;dataset&quot;) obtains miRNA expression dataset from the environment variable <code><a href="#topic+envData">envData</a></code>.
</p>
<p>The function GetExampleData(exampleData=&quot;class.labels&quot;) obtains class labels from the environment variable <code><a href="#topic+envData">envData</a></code>.
</p>
<p>The function GetExampleData(exampleData=&quot;miRList&quot;) obtains the drawing parameters of a miRNA List from the environment variable <code><a href="#topic+envData">envData</a></code>.
</p>
<p>The function GetExampleData(exampleData=&quot;p_value&quot;) obtains the weighting matrix from the environment variable <code><a href="#topic+envData">envData</a></code>.
</p>
<p>The function GetExampleData(exampleData=&quot;p2miR&quot;) obtains the correlation profile between kegg pathways and each human miRNA from the environment variable <code><a href="#topic+envData">envData</a></code>.
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

#obtain the gene expression dataset.
dataset&lt;-GetExampleData(exampleData="dataset")


#obtain the class labels.
class.labels&lt;-GetExampleData(exampleData="class.labels")

#obtain the drawing parameters of a miRNA List
miRList&lt;-GetExampleData(exampleData="miRList")


#obtain the weighting matrix 
p_value&lt;-GetExampleData(exampleData="p_value")

#obtain the correlation profile
p2miR&lt;-GetExampleData(exampleData="p2miR")



## End(Not run)
</code></pre>

<hr>
<h2 id='GetMiRTargetData'>Get the data of microRNA(miRNA) and target genes</h2><span id='topic+GetMiRTargetData'></span>

<h3>Description</h3>

<p>Get the data of miRNA and target genes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>   GetMiRTargetData()
</code></pre>


<h3>Details</h3>

<p>The data for target genes of miRNAs are obtained from the environment variable <code><a href="#topic+envData">envData</a></code>,which is obtained from four database(TarBaseV6.0,mir2Disease,miRecords,miRTarBase).
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>#Get the data for target genes of miRNAs
MiRTarget&lt;-GetMiRTargetData()

</code></pre>

<hr>
<h2 id='GetPathwayData'>Get the gene sets of pathways</h2><span id='topic+GetPathwayData'></span>

<h3>Description</h3>

<p>Get the gene sets of pathways for the three pathway databases (KEGG; Biocarta; Reactome)
</p>


<h3>Usage</h3>

<pre><code class='language-R'> GetPathwayData(pathway)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GetPathwayData_+3A_pathway">pathway</code></td>
<td>
<p>choose database of pathway,&quot;kegg&quot;,&quot;biocarta&quot; or&quot;reactome&quot;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The gene sets of pathways for the three pathway database (KEGG; Biocarta; Reactome)are obtained from the environment variable <code><a href="#topic+envData">envData</a></code>.
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

#obtain the gene sets of  kegg pathways.
pathway&lt;-GetPathwayData("kegg")



## End(Not run)

</code></pre>

<hr>
<h2 id='HeatMapPlot'>Plot a heatmap</h2><span id='topic+HeatMapPlot'></span>

<h3>Description</h3>

<p>Plot a heatmap of a microRNA(miRNA) expression
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HeatMapPlot(V, row.names = FALSE, col.labels, col.classes, col.names = FALSE, 
main = " ", xlab = " ", ylab = " ")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="HeatMapPlot_+3A_v">V</code></td>
<td>
<p>A miRNA expression matrix</p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_row.names">row.names</code></td>
<td>
<p>A name list of row vector,default=FALSE</p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_col.labels">col.labels</code></td>
<td>
<p>Phenotype of class disticntion of interest. A vector of binary labels having first the 1's and then the 0's </p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_col.classes">col.classes</code></td>
<td>
<p>phenotype name</p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_col.names">col.names</code></td>
<td>
<p>A name list of samples,default=FALSE</p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_main">main</code></td>
<td>
<p>a main title for the heatmap</p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_xlab">xlab</code></td>
<td>
<p>a label for the x axis, defaults to a description of x</p>
</td></tr>
<tr><td><code id="HeatMapPlot_+3A_ylab">ylab</code></td>
<td>
<p>a label for the y axis, defaults to a description of y</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a heatmap of a miRNA expression matrix including phenotype vector and miRNA, sample and phenotype labels
</p>


<h3>Note</h3>

<p>return a heatmap
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>References</h3>

<p>Andy Liaw, original,R. Gentleman, M. Maechler, W. Huber.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotHeatMap">PlotHeatMap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#example of expression profile
V&lt;-matrix(runif(200),10,20)
#example of class.labels ("0"or "1")
 a1&lt;-rep(0,20)
a1[sample(1:20,5)]=1
#plot heat map
 HeatMapPlot(V =V, row.names = FALSE, col.labels = a1, col.classes =c("a","b"), col.names =FALSE,
 main =" Heat Map for MiRs in MiR Set", xlab=" ", ylab=" ")
</code></pre>

<hr>
<h2 id='MirSEA'>Identify dysregulated pathways based on microRNA (miRNA) set enrichment analysis</h2><span id='topic+MirSEA'></span>

<h3>Description</h3>

<p>This function propose a novel method of miRNA set enrichment analysis(MiRSEA)to identify the dysregulated pathways by calculating the enrichment score of miRNA set which co-regulate a biological pathway(or prior gene set)</p>


<h3>Usage</h3>

<pre><code class='language-R'>MirSEA(input.ds, input.cls, p_value,p2miR,
reshuffling.type = "miR.labels", nperm = 1000, 
weighted.score.type = 1, ms.size.threshold.min = 10, 
ms.size.threshold.max = 500)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MirSEA_+3A_input.ds">input.ds</code></td>
<td>
<p>Input miRNA expression Affymetrix dataset file in GCT format</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_input.cls">input.cls</code></td>
<td>
<p>Input class vector (phenotype) file in CLS format</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_p_value">p_value</code></td>
<td>
<p> A weighting matrix of p value of the hypergeometric. (rows are pathway ,cols are microRNAs(miRNAs))</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_p2mir">p2miR</code></td>
<td>
<p>pathway-miRNA correlation(pmSET) profile</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_reshuffling.type">reshuffling.type</code></td>
<td>
<p>Type of permutation reshuffling: &quot;sample.labels&quot; or &quot;miR.labels&quot; (default: &quot;miR.labels&quot;)</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations (default: 1000)</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_weighted.score.type">weighted.score.type</code></td>
<td>
<p>Enrichment correlation based weighting:When weight= 0, ES reduces to the standard Kolmogorov-Smirnov statistic,when weight=1, we are weighting the miRNAs by their dw-score normalized by the sum of the dw-scores over all of the miRNAs in the miRNA set,when weight=2,it represent over weight (default: 1)</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_ms.size.threshold.min">ms.size.threshold.min</code></td>
<td>
<p>Minimum size (in miRNAs) for database miRNA sets to be considered (default: 10)</p>
</td></tr>
<tr><td><code id="MirSEA_+3A_ms.size.threshold.max">ms.size.threshold.max</code></td>
<td>
<p>Maximum size (in miRNAs) for database miRNA sets to be considered (default: 500)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>MiRSEA integrates pathway (e.g.the strength of the pathway regulated by miRNAs.) and differential expression among miRNAs in identifying dysregulated pathways.MiRSEA can order pathway by the enrichment score of miRNA set, which is co-regulated by a miRNA set.
</p>


<h3>Value</h3>

<table>
<tr><td><code>report.phen1</code></td>
<td>
<p> It is the summary of the result of the up regulated pathway</p>
</td></tr>
<tr><td><code>report.phen2</code></td>
<td>
<p>It is the summary of the result of the down regulated pathway.Each rows of the dataframe represents a pathway. Its columns include &quot;Pathway Name&quot;, &quot;SIZE&quot;, &quot;Pathway Source&quot;,
&quot;Pathway Enrichment Score&quot;, &quot;NOM p-val&quot;, &quot;FDR q-val&quot;, &quot;Tag percentage&quot;(Percent of miRNA set before running enrichment peak),&quot;MiR percentage&quot;(Percent of miRNA list before running enrichment peak),&quot;Signal strength&quot; (enrichment signal strength). </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>References</h3>

<p>Subramanian A, et al. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. Proceedings of the National Academy of Sciences of the United States of America 102(43):15545-15550.
</p>
<p>Lu M, Shi B, Wang J, Cao Q, &amp; Cui Q (2010) TAM: a method for enrichment and depletion analysis of a microRNA category in a list of microRNAs. BMC bioinformatics 11:419.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+EnrichmentScore">EnrichmentScore</a></code>, <code><a href="#topic+EnrichmentScore2">EnrichmentScore2</a></code>,<code><a href="#topic+S2N">S2N</a></code>,<code><a href="#topic+Corrp2miRfile">Corrp2miRfile</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get example of expression data
#input.ds &lt;- readLines("F:/lsy/xin data/GSE36915.gct")
#input.cls &lt;- readLines("F:/lsy/xin data/GSE36915.cls")
input.ds &lt;- GetExampleData("dataset")
input.cls &lt;- GetExampleData("class.labels")

#get example of p value matrix
p_value &lt;- GetExampleData("p_value")
#get example of correlation profile
p2miR &lt;- GetExampleData("p2miR")

#identify dysregulated pathways by using the function MirSEA
MirSEAresult &lt;- MirSEA(input.ds,input.cls,p_value,p2miR,
reshuffling.type = "miR.labels", nperm = 1000, 
weighted.score.type = 1, ms.size.threshold.min = 10, 
ms.size.threshold.max = 500)
#print the summary results of pathways to screen
 
summaryResult1 &lt;- MirSEAresult$report.phen1
summaryResult1[1:5,]
summaryResult2 &lt;- MirSEAresult$report.phen2
summaryResult2[1:5,]
 
#write the summary results of pathways to tab delimited file.
write.table(summaryResult1,file="summaryResult1.txt",sep="\t",row.names=FALSE)

write.table(summaryResult2,file="summaryResult2.txt",sep="\t",row.names=FALSE)

## End(Not run)
</code></pre>

<hr>
<h2 id='MiRSEA-internal'>MiRSEA internal functions</h2><span id='topic+initializeMiRSEA'></span><span id='topic+Getenvir'></span>

<h3>Description</h3>

<p>These functions internally called by other functions in the package <code>MiRSEA</code> and not meant to be called by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>
    initializeMiRSEA()
    Getenvir(envData)
</code></pre>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>

<hr>
<h2 id='MiRSEA-package'>MicroRNA Set Enrichment Analysis</h2><span id='topic+MiRSEA-package'></span><span id='topic+MiRSEA'></span>

<h3>Description</h3>

<p>This package can identify dysregulated pathways(or prior gene sets) regulated by microRNAs set in the context of miRNA expression data.
</p>


<h3>Introduction</h3>

<p>The package can identify dysregulated pathways(or prior gene sets) regulated by microRNAs set in the context of miRNA expression data.
(1) The MiRSEA package constructs a correlation profile of miRNAs and pathways by hypergeometric.The gene sets of pathways derived from the three public databases(KEGG;Reactome;Biocarta;).The target gene sets of miRNAs are provided by four databases(TarBaseV6.0; mir2Disease; miRecords; miRTarBase;). 
(2) The MiRSEA package can quantify the change of correlation between miRNAs for each pathway (or prior gene set) based on miRNA expression data with cases and controls.
(3) The MiRSEA package uses the weighted Kolmogorov-Smirnov statistic to calculate an enrichment score(ES)of a miRNA set that co-regulate to a pathway , which reflects the degree to which a given pathway is associated with the specific phenotype.
(4) The MiRSEA package can provide the visualization of the results.
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>

<hr>
<h2 id='MsReport'>Produce report for a microRNA (abbreviated miRNA) set </h2><span id='topic+MsReport'></span>

<h3>Description</h3>

<p> The miR.report includes miRNA names, locstion, S2N, RES and whether is core-enrichment miRNA,
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MsReport(MsNAME = "", input.ds, input.cls, p_value,
p2miR,weighted.score.type = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MsReport_+3A_msname">MsNAME</code></td>
<td>
<p>An interesting pathway name </p>
</td></tr>
<tr><td><code id="MsReport_+3A_input.ds">input.ds</code></td>
<td>
<p>Input miRNA expression Affymetrix dataset file in RES or GCT format</p>
</td></tr>
<tr><td><code id="MsReport_+3A_input.cls">input.cls</code></td>
<td>
<p>Input class vector (phenotype) file in CLS format</p>
</td></tr>
<tr><td><code id="MsReport_+3A_p_value">p_value</code></td>
<td>
<p> A weighting matrix of p value of the hypergeometric. (rows are pathway ,cols are microRNAs(miRNAs))</p>
</td></tr>
<tr><td><code id="MsReport_+3A_p2mir">p2miR</code></td>
<td>
<p>pathway-miRNA correlation(pmSET) profile</p>
</td></tr>
<tr><td><code id="MsReport_+3A_weighted.score.type">weighted.score.type</code></td>
<td>
<p>Enrichment correlation-based weighting: 0=no weight (KS), 1=standard weigth, 2 = over-weigth (default: 1)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>When users input a interesting pathway,the function MsReport can create a report for a miRNA set that coordinated regulate this pathway.
MiR : the name of miRNAs. For example the probe accession number, miRNA symbol or the miRNA identifier in the dataset.
LIST LOC : location of the miRNA in the sorted miRNA list.
S2N : correlation(tw-score) of the miRNA in the miRNA list.
RES : value of the running enrichment score at the miRNA location.
CORE_ENRICHMENT:whether this miRNA is the &quot;core enrichment&quot; section of the list, Yes or No variable specifying in the miRNA location is before (positive ES) or after (negative ES) the running enrichment peak.
</p>


<h3>Value</h3>

<p>A list. It includes two elements: Msreport and miRList.
</p>
<p>Msreport is matrix of input pathway which present the detail results . Its columns include  &quot;miRNA name&quot;, &quot;location of the miRNA in the sorted miRNA list&quot;, &quot;tw-scoe of miRNA&quot;, &quot;Running enrichment score&quot;, &quot;Property of contribution&quot;.
</p>
<p>miRList is a list of drawing parameters for function PlotHeatMap,PlotCorrelation and PlotRunEnrichment.
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+MirSEA">MirSEA</a></code>,<code><a href="#topic+S2N">S2N</a></code>,<code><a href="#topic+EnrichmentScore">EnrichmentScore</a></code>,<code><a href="#topic+PlotHeatMap">PlotHeatMap</a></code>,<code><a href="#topic+PlotCorrelation">PlotCorrelation</a></code>,<code><a href="#topic+PlotRunEnrichment">PlotRunEnrichment</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get example data
#input.ds &lt;- readLines("F:/lsy/xin data/GSE36915.gct")
#input.cls &lt;- readLines("F:/lsy/xin data/GSE36915.cls")
input.ds &lt;- GetExampleData("dataset")
input.cls &lt;- GetExampleData("class.labels")
 
#get example of p value matrix
p_value &lt;- GetExampleData("p_value")
#get example of correlation profile
p2miR &lt;- GetExampleData("p2miR")

#get a miRNA.SET report for KEGG ERBB PATHWAY
Results&lt;-MsReport(MsNAME = "KEGG_ERBB_SIGNALING_PATHWAY", input.ds, input.cls,p_value,p2miR)
# show the report of top five miRNA in the pathway
Results[[1]][1:5,]

## End(Not run)
</code></pre>

<hr>
<h2 id='PlotCorrelation'>Plot global microRNA(miRNA) correlation profile</h2><span id='topic+PlotCorrelation'></span>

<h3>Description</h3>

<p>plot global miRNA correlation profile for differential weighted scores(dw-score) of miRNAs
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotCorrelation(miRlist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotCorrelation_+3A_mirlist">miRlist</code></td>
<td>
<p>A list of miRNA LIST result obtained from the MsReport function</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+MsReport">MsReport</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get a list of miRNA list result
miRlist&lt;-GetExampleData("miRList")
#Plot global miRNA correlation profile
PlotCorrelation(miRlist)


## End(Not run)
</code></pre>

<hr>
<h2 id='PlotHeatMap'>Plot a heat map</h2><span id='topic+PlotHeatMap'></span>

<h3>Description</h3>

<p>Plot a heat map for a microRNA(miRNA) set which co-regulate pathway
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotHeatMap(miRlist,input.ds,input.cls)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotHeatMap_+3A_mirlist">miRlist</code></td>
<td>
<p>A list of miRNA LIST result obtained from the MsReport function</p>
</td></tr>
<tr><td><code id="PlotHeatMap_+3A_input.ds">input.ds</code></td>
<td>
<p>Input miRNA expression Affymetrix dataset file in GCT format</p>
</td></tr>
<tr><td><code id="PlotHeatMap_+3A_input.cls">input.cls</code></td>
<td>
<p>Input class vector (phenotype) file in CLS format</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Plots a heatmap of a miRNA set in the expression matrix including phenotype vector and miRNA, sample and phenotype labels
</p>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+PlotHeatMap">PlotHeatMap</a></code>, <code><a href="#topic+MsReport">MsReport</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get example data
#input.ds &lt;- readLines("F:/lsy/xin data/GSE36915.gct")
#input.cls &lt;- readLines("F:/lsy/xin data/GSE36915.cls")
input.ds &lt;- GetExampleData("dataset")
input.cls &lt;- GetExampleData("class.labels")

#get a list of miRNA list result
miRlist&lt;-GetExampleData("miRList")
#Plot a heat map
PlotHeatMap(miRlist,input.ds,input.cls)

## End(Not run)
</code></pre>

<hr>
<h2 id='PlotRunEnrichment'>Plot running microRNAs(miRNAs) enrichment score</h2><span id='topic+PlotRunEnrichment'></span>

<h3>Description</h3>

<p>Plot running miRNAs enrichment score for the input pathway 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>PlotRunEnrichment(miRlist)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="PlotRunEnrichment_+3A_mirlist">miRlist</code></td>
<td>
<p>A list of miRNA LIST result obtained from the MsReport function</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>See Also</h3>

<p><code><a href="#topic+MsReport">MsReport</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
#get a list of miRNA list result
miRlist&lt;-GetExampleData("miRList")
#Plot running miRNA enrichment score
PlotRunEnrichment(miRlist)

## End(Not run)
</code></pre>

<hr>
<h2 id='S2N'>
calculate signal to noise ratio for microRNAs(miRNAs)
</h2><span id='topic+S2N'></span>

<h3>Description</h3>

<p>This function calculate the signal to noise ratio for miRNAs for the actual phenotype and also random permutations 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>S2N(A, class.labels, miR.labels, nperm )
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="S2N_+3A_a">A</code></td>
<td>
<p>Matrix of miRNAs expression values (rows are miRNAs, columns are samples)</p>
</td></tr>
<tr><td><code id="S2N_+3A_class.labels">class.labels</code></td>
<td>
<p>Phenotype of class disticntion of interest. A vector of binary labels having first the 1's and then the 0's </p>
</td></tr>
<tr><td><code id="S2N_+3A_mir.labels">miR.labels</code></td>
<td>
<p> miRNA labels,Vector of probe ids or accession numbers for the rows of the expression matrix </p>
</td></tr>
<tr><td><code id="S2N_+3A_nperm">nperm</code></td>
<td>
<p>Number of random permutations to perform</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function uses matrix operations to implement the signal to noise calculation in stages and achieves fast execution speed. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>s2n.matrix</code></td>
<td>
<p>Matrix with random permuted or bootstraps signal to noise ratios (rows are miRNAs, columns are permutations or bootstrap subsamplings</p>
</td></tr>
<tr><td><code>obs.s2n.matrix</code></td>
<td>
<p> Matrix with observed signal to noise ratios (rows are miRNAs, columns are boostraps subsamplings. If fraction is set to 1.0 then all the columns have the same values</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Junwei Han<a href="mailto:hanjunwei1981@163.com">hanjunwei1981@163.com</a>,Siyao Liu <a href="mailto:liusiyao29@163.com">liusiyao29@163.com</a></p>


<h3>References</h3>

<p>Subramanian A, et al. (2005) Gene set enrichment analysis: a knowledge-based approach for interpreting genome-wide expression profiles. Proceedings of the National Academy of Sciences of the United States of America 102(43):15545-15550.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MirSEA">MirSEA</a></code></p>


<h3>Examples</h3>

<pre><code class='language-R'>##Matrix of miRNAs expression values
A&lt;-matrix(runif(200),10,20)
##class.labels("0" or "1")
a1&lt;-rep(0,20)
a1[sample(1:20,5)]=1
a1&lt;-sort(a1,decreasing=FALSE)
#calculate signal to noise ratio for example data
M1&lt;-S2N(A, class.labels=a1, miR.labels=seq(1,10), nperm=100)
#show actual results for top five in the matrix 
M1$obs.s2n.matrix[1:5,1]
#show permutation results
M1$s2n.matrix[1:5,1:5]
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
