<!DOCTYPE html><html lang="en"><head><title>Help for package miRtest</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {miRtest}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#A'><p>A</p></a></li>
<li><a href='#contingency.table'><p>Contingency table.</p></a></li>
<li><a href='#fisher.combination'><p>Fisher method of p value combination.</p></a></li>
<li><a href='#generate.A'><p>Turn a data...</p></a></li>
<li><a href='#gs.test'><p>Internal function for gene set testing.</p></a></li>
<li><a href='#inverse.normal.combination'><p>Inverse-normal method for p value combination.</p></a></li>
<li><a href='#limma.one.sided'><p>Internal algorithm: Make limma test one-sided...</p></a></li>
<li><a href='#limma.test'><p>internal algorithm for author's convenience.</p></a></li>
<li><a href='#m.combine'><p>Internal function for author's convenience and more legible code.</p></a></li>
<li><a href='#miR.test'><p>Main Function of miRtest package.</p></a></li>
<li><a href='#miRtest'>
<p>Package Description: Two-group combined miRNA- and mRNA- expression testing.</p></a></li>
<li><a href='#X'><p>X</p></a></li>
<li><a href='#Y'><p>Y</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Maintainer:</td>
<td>Stephan Artmann &lt;stephanartmann@gmx.net&gt;</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Title:</td>
<td>Combined miRNA- And mRNA-Testing</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://pubmed.ncbi.nlm.nih.gov/22723856/">https://pubmed.ncbi.nlm.nih.gov/22723856/</a></td>
</tr>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>LazyLoad:</td>
<td>yes</td>
</tr>
<tr>
<td>Author:</td>
<td>Stephan Artmann, Klaus Jung, Tim Beissbarth</td>
</tr>
<tr>
<td>Description:</td>
<td>Package for combined miRNA- and mRNA-testing.</td>
</tr>
<tr>
<td>Version:</td>
<td>2.1</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-04</td>
</tr>
<tr>
<td>Depends:</td>
<td>corpcor, MASS</td>
</tr>
<tr>
<td>Imports:</td>
<td>globaltest, GlobalAncova, limma</td>
</tr>
<tr>
<td>Collate:</td>
<td>'miRtest.R'</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-05 07:35:34 UTC; insec</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-05 12:10:03 UTC</td>
</tr>
</table>
<hr>
<h2 id='A'>A</h2><span id='topic+A'></span>

<h3>Description</h3>

<p>Part of expression data from Nielsen et al.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann taken from Nielsen et al.</p>

<hr>
<h2 id='contingency.table'>Contingency table.</h2><span id='topic+contingency.table'></span>

<h3>Description</h3>

<p>Contingency table. Necessary for Fisher test.</p>


<h3>Usage</h3>

<pre><code class='language-R'>contingency.table(gene.set, p.val, sign=0.05)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="contingency.table_+3A_gene.set">gene.set</code></td>
<td>
<p>Vector of gene sets.</p>
</td></tr>
<tr><td><code id="contingency.table_+3A_p.val">p.val</code></td>
<td>
<p>Vector with p-values.</p>
</td></tr>
<tr><td><code id="contingency.table_+3A_sign">sign</code></td>
<td>
<p>Significane threshold.</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>

<hr>
<h2 id='fisher.combination'>Fisher method of p value combination.</h2><span id='topic+fisher.combination'></span>

<h3>Description</h3>

<p>Fisher method of p value combination.</p>


<h3>Usage</h3>

<pre><code class='language-R'>fisher.combination(p1, p2, check.range=FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="fisher.combination_+3A_p1">p1</code>, <code id="fisher.combination_+3A_p2">p2</code></td>
<td>
<p>one-sided p-values that shall be combined.</p>
</td></tr>
<tr><td><code id="fisher.combination_+3A_check.range">check.range</code></td>
<td>
<p>If set to &quot;TRUE&quot; values above 1 will be set to 1.</p>
</td></tr></table>


<h3>Value</h3>

<p>Combined p-value.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>

<hr>
<h2 id='generate.A'>Turn a data...</h2><span id='topic+generate.A'></span>

<h3>Description</h3>

<p>Turn a data.frame indicating gene sets into the allocation matrix.</p>


<h3>Usage</h3>

<pre><code class='language-R'>generate.A(df, X, Y, verbose=TRUE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="generate.A_+3A_df">df</code></td>
<td>
<p>data.frame with mRNAs in its first and miRNAs in its second column.</p>
</td></tr>
<tr><td><code id="generate.A_+3A_x">X</code></td>
<td>
<p>Expression matrix of miRNAs whose row names will be used to generate the list of miRNAs.</p>
</td></tr>
<tr><td><code id="generate.A_+3A_y">Y</code></td>
<td>
<p>Expression matrix of mRNAs whose row names will be used to generate the list of mRNAs.</p>
</td></tr>
<tr><td><code id="generate.A_+3A_verbose">verbose</code></td>
<td>
<p>Logical. Shall progress be printed?</p>
</td></tr></table>


<h3>Value</h3>

<p>Allocation matrix A necessary for &quot;miR.test&quot; function.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>


<h3>Examples</h3>

<pre><code class='language-R'>#######################################
### Generate random expression data ###
#######################################
# Generate random miRNA expression data of 3 miRNAs
# with 8 replicates
set.seed(1)
X = rnorm(24);
dim(X) = c(3,8);
rownames(X) = 1:3;
# Generate random mRNA expression data with 20 mRNAs
# and 10 replicates
Y = rnorm(200);
dim(Y) = c(20,10);
rownames(Y) = 1:20;
# Let's assume that we want to compare 2 miRNA groups, each of 4 replicates:
group.miRNA = factor(c(1,1,1,1,2,2,2,2));
# ... and that the corresponding mRNA experiments had 5 replicates in each group
group.mRNA = factor(c(1,1,1,1,1,2,2,2,2,2));
####################
### Perform Test ###
####################
library(miRtest)
#Let miRNA 1 attack mRNAs 1 to 9 and miRNA 2 attack mRNAs 10 to 17.
# mRNAs 18 to 20 are not attacked. miRNA 3 has no gene set.
miR = c(rep(1,9),c(rep(2,8)));
mRNAs = 1:17;
A = data.frame(mRNAs,miR); # Note that the miRNAs MUST be in the second column!
A
set.seed(1)
P = miR.test(X,Y,A,group.miRNA,group.mRNA)
P


#####################################################
### For a faster result: use other gene set tests ###
#####################################################
# Wilcoxon two-sample test is recommended for fast results
# Note that results may vary depending on how much genes correlate

P.gsWilcox = miR.test(X,Y,A,group.miRNA,group.mRNA,gene.set.tests="W")
P.gsWilcox
############################################
### We can use an allocation matrix as A ###
############################################
A = generate.A(A,X=X,Y=Y,verbose=FALSE);
A
# Now we can test as before
set.seed(1)
P = miR.test(X,Y,A,group.miRNA,group.mRNA,allocation.matrix=TRUE)
P


#####################
### Other Designs ###
#####################

# Some more complicated designs are implemented, check the vignette "miRtest" for details.
group.miRNA = 1:8
group.mRNA = 1:10
covariable.miRNA = factor(c(1,2,3,4,1,2,3,4))    ### A covariable in miRNAs.
covariable.mRNA = factor(c(1,2,3,4,5,1,2,3,4,5)) ### A covariable in mRNAs.

library(limma)
design.miRNA = model.matrix(~group.miRNA + covariable.miRNA)
design.mRNA =  model.matrix(~group.mRNA + covariable.mRNA)

P = miR.test(X,Y,A,design.miRNA=design.miRNA,design.mRNA=design.mRNA,allocation.matrix=TRUE)
P</code></pre>

<hr>
<h2 id='gs.test'>Internal function for gene set testing.</h2><span id='topic+gs.test'></span>

<h3>Description</h3>

<p>Internal function for gene set testing.</p>


<h3>Usage</h3>

<pre><code class='language-R'>gs.test(A, X, Y, group, tests, permutation=FALSE, nrot=1000, design,
    allocation.matrix=FALSE, verbose=FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="gs.test_+3A_a">A</code></td>
<td>
<p>Allocation matrix as in &quot;miR.test&quot; function.</p>
</td></tr>
<tr><td><code id="gs.test_+3A_x">X</code></td>
<td>
<p>miRNA expression matrix as in &lsquo;miR.test&rsquo; function. Only necessary when allocation.matrix=TRUE.</p>
</td></tr>
<tr><td><code id="gs.test_+3A_y">Y</code></td>
<td>
<p>mRNA expression matrix as in &quot;miR.test&quot; function.</p>
</td></tr>
<tr><td><code id="gs.test_+3A_group">group</code></td>
<td>
<p>group as in &lsquo;miR.test&rsquo; function</p>
</td></tr>
<tr><td><code id="gs.test_+3A_tests">tests</code></td>
<td>
<p>Test applied, sie gene.set.tests</p>
</td></tr>
<tr><td><code id="gs.test_+3A_permutation">permutation</code></td>
<td>
<p>Shall permutation procedure for global tests be applied? Put 'FALSE' to use approximate results or give a number for the number of permutations.</p>
</td></tr>
<tr><td><code id="gs.test_+3A_nrot">nrot</code></td>
<td>
<p>Number of rotations of rotation tests. Defaults to 1000 to be able to show p-values as low as 10^-3.</p>
</td></tr>
<tr><td><code id="gs.test_+3A_design">design</code></td>
<td>
<p>If specified, group will be ignored. Design matrix as used in &lsquo;limma&rsquo; package. Cannot be used with global tests.</p>
</td></tr>
<tr><td><code id="gs.test_+3A_allocation.matrix">allocation.matrix</code></td>
<td>
<p>Logical, is A an allocation matrix with mRNAs in its columns and miRNAs in its rows, or is it an allocation data.frame?</p>
</td></tr>
<tr><td><code id="gs.test_+3A_verbose">verbose</code></td>
<td>
<p>Defaults to FALSE. If TRUE, progress is printed.</p>
</td></tr></table>


<h3>Value</h3>

<p>List of the following, for up- and for down-regulation: Matrix with testing results for every gene set in its rows and the applied gene set test in its columns.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>


<h3>References</h3>

<p>Artmann, Stephan and Jung, Klaus and Bleckmann, Annalen and Beissbarth, Tim (2012).
Detection of simultaneous group effects in microRNA expression and 
related functional gene sets. PLoS ONE 7(6):e38365, PMID: 22723856.
</p>
<p>Brunner, E. (2009) Repeated measures under non-sphericity.
Proceedings of the 6th St. Petersburg Workshop on Simulation,
605-609.
</p>
<p>Jelle J. Goeman, Sara A. van de Geer, Floor de Kort, Hans C. van
Houwelingen (2004) A global test for groups of genes: testing
association with a clinical outcome. Bioinformatics 20, 93-99.
</p>
<p>Jung, Klaus and Becker, Benjamin and Brunner, Edgar and Beissbarth, Tim (2011).
Comparison of Global Tests for Functinoal Gene Sets in
Two-Group Designs and Selection of Potentially
Effect-causing Genes. Bioinformatics, 27: 1377-1383.
</p>
<p>Majewski, IJ, Ritchie, ME, Phipson, B, Corbin, J, Pakusch, M,
Ebert, A, Busslinger, M, Koseki, H, Hu, Y, Smyth, GK, Alexander,
WS, Hilton, DJ, and Blewitt, ME (2010). Opposing roles of polycomb
repressive complexes in hematopoietic stem and progenitor cells.
_Blood_, published online 5 May 2010.
</p>
<p>Mansmann, U. and Meister, R., 2005, Testing differential gene
expression in functional groups, _Methods Inf Med_ 44 (3).
</p>
<p>Smyth, G. K. (2004). Linear models and empirical Bayes methods for
assessing differential expression in microarray experiments.
_Statistical Applications in Genetics and Molecular Biology_,
Volume *3*, Article 3.
</p>
<p>Wu, D, Lim, E, Francois Vaillant, F, Asselin-Labat, M-L, Visvader,
JE, and Smyth, GK (2010). ROAST: rotation gene set tests for
complex microarray experiments. _Bioinformatics_, published online
7 July 2010.</p>

<hr>
<h2 id='inverse.normal.combination'>Inverse-normal method for p value combination.</h2><span id='topic+inverse.normal.combination'></span>

<h3>Description</h3>

<p>Inverse-normal method for p value combination.</p>


<h3>Usage</h3>

<pre><code class='language-R'>inverse.normal.combination(p1, p2)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="inverse.normal.combination_+3A_p1">p1</code>, <code id="inverse.normal.combination_+3A_p2">p2</code></td>
<td>
<p>one-sided p-values that shall be combined.</p>
</td></tr></table>


<h3>Value</h3>

<p>Two-sided combined p-value.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>

<hr>
<h2 id='limma.one.sided'>Internal algorithm: Make limma test one-sided...</h2><span id='topic+limma.one.sided'></span>

<h3>Description</h3>

<p>Internal algorithm: Make limma test one-sided</p>


<h3>Usage</h3>

<pre><code class='language-R'>limma.one.sided(fit, lower=FALSE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="limma.one.sided_+3A_fit">fit</code></td>
<td>
<p>Result of &quot;lmFit&quot; and &quot;eBayes&quot; functions in &quot;limma&quot; package.</p>
</td></tr>
<tr><td><code id="limma.one.sided_+3A_lower">lower</code></td>
<td>
<p>Shall one-sided p-value indicated down-regultation?</p>
</td></tr></table>

<hr>
<h2 id='limma.test'>internal algorithm for author's convenience.</h2><span id='topic+limma.test'></span>

<h3>Description</h3>

<p>internal algorithm for author's convenience. Create a linear model with the limma package.</p>


<h3>Usage</h3>

<pre><code class='language-R'>limma.test(X, group, design)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="limma.test_+3A_x">X</code></td>
<td>
<p>Expression matrix.</p>
</td></tr>
<tr><td><code id="limma.test_+3A_group">group</code></td>
<td>
<p>Group membership of replicates.</p>
</td></tr>
<tr><td><code id="limma.test_+3A_design">design</code></td>
<td>
<p>Design as specified in limma (design matrix, see model.matrix).</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>

<hr>
<h2 id='m.combine'>Internal function for author's convenience and more legible code.</h2><span id='topic+m.combine'></span>

<h3>Description</h3>

<p>Internal function for author's convenience and more legible code. Applies a function to every column vector of a matrix and a vector.</p>


<h3>Usage</h3>

<pre><code class='language-R'>m.combine(M, v, FUN, ...)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="m.combine_+3A_m">M</code></td>
<td>
<p>The matrix for whose column vectors mapply shall be used.</p>
</td></tr>
<tr><td><code id="m.combine_+3A_v">v</code></td>
<td>
<p>The vector.</p>
</td></tr>
<tr><td><code id="m.combine_+3A_fun">FUN</code></td>
<td>
<p>The function.</p>
</td></tr>
<tr><td><code id="m.combine_+3A_...">...</code></td>
<td>
<p>Further arguments to be given to FUN.</p>
</td></tr></table>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>

<hr>
<h2 id='miR.test'>Main Function of miRtest package.</h2><span id='topic+miR.test'></span>

<h3>Description</h3>

<p>Main Function of miRtest package.</p>


<h3>Usage</h3>

<pre><code class='language-R'>miR.test(X, Y, A, group.miRNA, group.mRNA, gene.set.tests="romer",
    design.miRNA, design.mRNA, adjust="none", permutation=FALSE,
    nrot=1000, allocation.matrix=FALSE, verbose=FALSE, errors=TRUE)</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="miR.test_+3A_x">X</code></td>
<td>
<p>miRNA expression matrix with genes in rows and replicates in columns</p>
</td></tr>
<tr><td><code id="miR.test_+3A_y">Y</code></td>
<td>
<p>mRNA expression matrix with genes in rows and replicates in columns</p>
</td></tr>
<tr><td><code id="miR.test_+3A_a">A</code></td>
<td>
<p>Allocation data.frame or Allocation matrix. An allocation data.frame contains the mRNAs in its first column and the miRNAs in its second column. See vignette &lsquo;miRtest&rsquo; for information on Allocation matrices.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_group.mirna">group.miRNA</code></td>
<td>
<p>Vector of miRNA group membership, being either numeric or a factor (**this makes a difference**). E. g. if you have four replicates in a control group and three replicates in a treated group, you may choose c(1,1,1,1,2,2,2)</p>
</td></tr>
<tr><td><code id="miR.test_+3A_design.mirna">design.miRNA</code></td>
<td>
<p>If specified, group.miRNA will be ignored. Here you can specify a design matrix as it is returned from the model.matrix &lsquo;limma&rsquo; function.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_design.mrna">design.mRNA</code></td>
<td>
<p>If specified, group.mRNA will be ignored. Here you can specify a design matrix as it is returned from the model.matrix &lsquo;limma&rsquo; function.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_group.mrna">group.mRNA</code></td>
<td>
<p>Vector of mRNA group membership, being either numeric or a factor (**this makes a difference**).E. g. if you have four replicates in a control group and three replicates in a treated group, you may choose c(1,1,1,1,2,2,2)</p>
</td></tr>
<tr><td><code id="miR.test_+3A_gene.set.tests">gene.set.tests</code></td>
<td>
<p>Test to be applied for gene set testing. Can be one or more of the following: &lsquo;globaltest&rsquo;, &lsquo;GA&rsquo;, &lsquo;RHD&rsquo;, &lsquo;KS&rsquo;, &lsquo;W&rsquo;, &lsquo;Fisher&rsquo;, &lsquo;roast&rsquo;, &lsquo;romer&rsquo;, or &lsquo;all&rsquo; if you want to do all tests.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_adjust">adjust</code></td>
<td>
<p>Muliple hypothesis testing adjustment. Same options as in &quot;p.adjust&quot; function.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_permutation">permutation</code></td>
<td>
<p>Number of permutations for &lsquo;globaltest&rsquo; or &lsquo;GlobalAncova&rsquo; gene set tests. Put to &quot;FALSE&quot; to use the approximate p-values instead of permutation ones.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_nrot">nrot</code></td>
<td>
<p>Number of rotations for rotation tests &lsquo;ROAST&rsquo; and &lsquo;romer&rsquo;</p>
</td></tr>
<tr><td><code id="miR.test_+3A_allocation.matrix">allocation.matrix</code></td>
<td>
<p>Logical, is A an allocation matrix with mRNAs in its columns and miRNAs in its rows, or is it an allocation data.frame?</p>
</td></tr>
<tr><td><code id="miR.test_+3A_verbose">verbose</code></td>
<td>
<p>Defaults to FALSE. If TRUE, output on progress is printed.</p>
</td></tr>
<tr><td><code id="miR.test_+3A_errors">errors</code></td>
<td>
<p>Defaults to TRUE. If set to FALSE, some errors checking correct sizes of matrices are turned into warning messages.</p>
</td></tr></table>


<h3>Value</h3>

<p>Matrix with testing results for every miRNA in its rows and the applied gene set test in its columns. Note that result will depend on whether multiple hypothesis testing correction was applied or not.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann</p>


<h3>References</h3>

<p>Artmann, Stephan and Jung, Klaus and Bleckmann, Annalen and Beissbarth, Tim (2012).
Detection of simultaneous group effects in microRNA expression and 
related functional gene sets. PLoS ONE 7(6):e38365, PMID: 22723856.
</p>
<p>Brunner, E. (2009) Repeated measures under non-sphericity.
Proceedings of the 6th St. Petersburg Workshop on Simulation,
605-609.
</p>
<p>Jelle J. Goeman, Sara A. van de Geer, Floor de Kort, Hans C. van
Houwelingen (2004) A global test for groups of genes: testing
association with a clinical outcome. Bioinformatics 20, 93-99.
</p>
<p>Jung, Klaus and Becker, Benjamin and Brunner, Edgar and Beissbarth, Tim (2011).
Comparison of Global Tests for Functinoal Gene Sets in
Two-Group Designs and Selection of Potentially
Effect-causing Genes. Bioinformatics, 27: 1377-1383.
</p>
<p>Majewski, IJ, Ritchie, ME, Phipson, B, Corbin, J, Pakusch, M,
Ebert, A, Busslinger, M, Koseki, H, Hu, Y, Smyth, GK, Alexander,
WS, Hilton, DJ, and Blewitt, ME (2010). Opposing roles of polycomb
repressive complexes in hematopoietic stem and progenitor cells.
_Blood_, published online 5 May 2010.
</p>
<p>Mansmann, U. and Meister, R., 2005, Testing differential gene
expression in functional groups, _Methods Inf Med_ 44 (3).
</p>
<p>Smyth, G. K. (2004). Linear models and empirical Bayes methods for
assessing differential expression in microarray experiments.
_Statistical Applications in Genetics and Molecular Biology_,
Volume *3*, Article 3.
</p>
<p>Wu, D, Lim, E, Francois Vaillant, F, Asselin-Labat, M-L, Visvader,
JE, and Smyth, GK (2010). ROAST: rotation gene set tests for
complex microarray experiments. _Bioinformatics_, published online
7 July 2010.</p>


<h3>Examples</h3>

<pre><code class='language-R'>#######################################
### Generate random expression data ###
#######################################
# Generate random miRNA expression data of 3 miRNAs
# with 8 replicates
set.seed(1)
X = rnorm(24);
dim(X) = c(3,8);
rownames(X) = 1:3;
# Generate random mRNA expression data with 20 mRNAs
# and 10 replicates
Y = rnorm(200);
dim(Y) = c(20,10);
rownames(Y) = 1:20;
# Let's assume that we want to compare 2 miRNA groups, each of 4 replicates:
group.miRNA = factor(c(1,1,1,1,2,2,2,2));
# ... and that the corresponding mRNA experiments had 5 replicates in each group
group.mRNA = factor(c(1,1,1,1,1,2,2,2,2,2));
####################
### Perform Test ###
####################
library(miRtest)
#Let miRNA 1 attack mRNAs 1 to 9 and miRNA 2 attack mRNAs 10 to 17.
# mRNAs 18 to 20 are not attacked. miRNA 3 has no gene set.
miR = c(rep(1,9),c(rep(2,8)));
mRNAs = 1:17;
A = data.frame(mRNAs,miR); # Note that the miRNAs MUST be in the second column!
A
set.seed(1)
P = miR.test(X,Y,A,group.miRNA,group.mRNA)
P


#####################################################
### For a faster result: use other gene set tests ###
#####################################################
# Wilcoxon two-sample test is recommended for fast results
# Note that results may vary depending on how much genes correlate

P.gsWilcox = miR.test(X,Y,A,group.miRNA,group.mRNA,gene.set.tests="W")
P.gsWilcox
############################################
### We can use an allocation matrix as A ###
############################################
A = generate.A(A,X=X,Y=Y,verbose=FALSE);
A
# Now we can test as before
set.seed(1)
P = miR.test(X,Y,A,group.miRNA,group.mRNA,allocation.matrix=TRUE)
P


#####################
### Other Designs ###
#####################

# Some more complicated designs are implemented, check the vignette "miRtest" for details.
group.miRNA = 1:8
group.mRNA = 1:10
covariable.miRNA = factor(c(1,2,3,4,1,2,3,4))    ### A covariable in miRNAs.
covariable.mRNA = factor(c(1,2,3,4,5,1,2,3,4,5)) ### A covariable in mRNAs.

library(limma)
design.miRNA = model.matrix(~group.miRNA + covariable.miRNA)
design.mRNA =  model.matrix(~group.mRNA + covariable.mRNA)

P = miR.test(X,Y,A,design.miRNA=design.miRNA,design.mRNA=design.mRNA,allocation.matrix=TRUE)
P</code></pre>

<hr>
<h2 id='miRtest'>
Package Description: Two-group combined miRNA- and mRNA- expression testing.
</h2><span id='topic+miRtest'></span>

<h3>Description</h3>

<p>Looking for differential expression in miRNA-data can have low power. Taking their respective mRNA-gene sets on the other hand
can lead to too liberal results. In Artmann et al. we proposed a method to combine both information sources and generate p-values
that can detect either miRNA- and target gene set expression differences. 
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> miRtest</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 2.1</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2024-02-04</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL</td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
URL: </td><td style="text-align: left;"> http://www.ncbi.nlm.nih.gov/pubmed/22723856</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>

<p>For a detailed help check vignette(&quot;miRtest&quot;)
</p>
<p>You can start the test with the &quot;miR.test&quot; function, which needs the expression matrix X
of miRNAs, the expression matrix Y of mRNAs and the allocation matrix.
</p>


<h3>Author(s)</h3>

<p>Stephan Artmann &lt;stephanartmann@gmx.net&gt;,
Klaus Jung,
Tim Beissbarth
</p>
<p>Maintainer: Stephan Artmann &lt;stephanartmann@gmx.net&gt;
</p>


<h3>References</h3>

<p>Artmann, Stephan and Jung, Klaus and Bleckmann, Annalen and Beissbarth, Tim (2012).
Detection of simultaneous group effects in microRNA expression and 
related functional gene sets. Plos ONE, PMID: 22723856.
</p>
<p>Brunner, E. (2009) Repeated measures under non-sphericity.
Proceedings of the 6th St. Petersburg Workshop on Simulation,
605-609.
</p>
<p>Jelle J. Goeman, Sara A. van de Geer, Floor de Kort, Hans C. van
Houwelingen (2004) A global test for groups of genes: testing
association with a clinical outcome. Bioinformatics 20, 93-99.
</p>
<p>Jung, Klaus and Becker, Benjamin and Brunner, Edgar and Beissbarth, Tim (2011).
Comparison of Global Tests for Functinoal Gene Sets in
Two-Group Designs and Selection of Potentially
Effect-causing Genes. Bioinformatics, 27: 1377-1383.
</p>
<p>Majewski, IJ, Ritchie, ME, Phipson, B, Corbin, J, Pakusch, M,
Ebert, A, Busslinger, M, Koseki, H, Hu, Y, Smyth, GK, Alexander,
WS, Hilton, DJ, and Blewitt, ME (2010). Opposing roles of polycomb
repressive complexes in hematopoietic stem and progenitor cells.
_Blood_, published online 5 May 2010.
</p>
<p>Mansmann, U. and Meister, R., 2005, Testing differential gene
expression in functional groups, _Methods Inf Med_ 44 (3).
</p>
<p>Smyth, G. K. (2004). Linear models and empirical Bayes methods for
assessing differential expression in microarray experiments.
_Statistical Applications in Genetics and Molecular Biology_,
Volume *3*, Article 3.
</p>
<p>Wu, D, Lim, E, Francois Vaillant, F, Asselin-Labat, M-L, Visvader,
JE, and Smyth, GK (2010). ROAST: rotation gene set tests for
complex microarray experiments. _Bioinformatics_, published online
7 July 2010.
</p>


<h3>See Also</h3>

<p>Function &quot;generate.A&quot; as well as main function &quot;miR.test&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'> #######################################
 ### Generate random expression data ###
 #######################################
 # Generate random miRNA expression data of 3 miRNAs
 # with 8 replicates
 set.seed(1)
 X = rnorm(24);
 dim(X) = c(3,8);
 rownames(X) = 1:3;
 # Generate random mRNA expression data with 20 mRNAs
 # and 10 replicates
 Y = rnorm(200);
 dim(Y) = c(20,10);
 rownames(Y) = 1:20;
 # Let's assume that we want to compare 2 miRNA groups, each of 4 replicates:
 group.miRNA = factor(c(1,1,1,1,2,2,2,2));
 # ... and that the corresponding mRNA experiments had 5 replicates in each group
 group.mRNA = factor(c(1,1,1,1,1,2,2,2,2,2));
 ####################
 ### Perform Test ###
 ####################
 library(miRtest)
 #Let miRNA 1 attack mRNAs 1 to 9 and miRNA 2 attack mRNAs 10 to 17.
 # mRNAs 18 to 20 are not attacked. miRNA 3 has no gene set.
 miR = c(rep(1,9),c(rep(2,8)));
 mRNAs = 1:17;
 A = data.frame(mRNAs,miR); # Note that the miRNAs MUST be in the second column!
 A
 set.seed(1)
 P = miR.test(X,Y,A,group.miRNA,group.mRNA)
 P
 
 
 #####################################################
 ### For a faster result: use other gene set tests ###
 #####################################################
 # Wilcoxon two-sample test is recommended for fast results
 # Note that results may vary depending on how much genes correlate
 
 P.gsWilcox = miR.test(X,Y,A,group.miRNA,group.mRNA,gene.set.tests="W")
 P.gsWilcox
 ############################################
 ### We can use an allocation matrix as A ###
 ############################################
 A = generate.A(A,X=X,Y=Y,verbose=FALSE);
 A
 # Now we can test as before
 set.seed(1)
 P = miR.test(X,Y,A,group.miRNA,group.mRNA,allocation.matrix=TRUE)
 P
 
 
 #####################
 ### Other Designs ###
 #####################
 
 # Some more complicated designs are implemented, check the vignette "miRtest" for details.
 group.miRNA = 1:8
 group.mRNA = 1:10
 covariable.miRNA = factor(c(1,2,3,4,1,2,3,4))    ### A covariable in miRNAs.
 covariable.mRNA = factor(c(1,2,3,4,5,1,2,3,4,5)) ### A covariable in mRNAs.
 
 library(limma)
 design.miRNA = model.matrix(~group.miRNA + covariable.miRNA)
 design.mRNA =  model.matrix(~group.mRNA + covariable.mRNA)
 
 P = miR.test(X,Y,A,design.miRNA=design.miRNA,design.mRNA=design.mRNA,allocation.matrix=TRUE)
 P
</code></pre>

<hr>
<h2 id='X'>X</h2><span id='topic+X'></span>

<h3>Description</h3>

<p>Part of expression data from Nielsen et al.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann taken from Nielsen et al.</p>

<hr>
<h2 id='Y'>Y</h2><span id='topic+Y'></span>

<h3>Description</h3>

<p>Part of expression data from Nielsen et al.</p>


<h3>Author(s)</h3>

<p>Stephan Artmann taken from Nielsen et al.</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
