<!DOCTYPE html><html><head><title>Help for package MiscMetabar</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MiscMetabar}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MiscMetabar-package'><p><code>MiscMetabar</code> package</p></a></li>
<li><a href='#accu_plot'><p>Plot accumulation curves for <code>phyloseq-class</code> object</p></a></li>
<li><a href='#accu_plot_balanced_modality'><p>Plot accumulation curves with balanced modality and depth rarefaction</p></a></li>
<li><a href='#accu_samp_threshold'><p>Compute the number of sequence to obtain a given proportion of ASV in</p>
accumulation curves</a></li>
<li><a href='#add_blast_info'><p>Add information from <code>blast_pq()</code> to the <code>tax_table</code> slot of a <em>phyloseq</em> object</p></a></li>
<li><a href='#add_dna_to_phyloseq'><p>Add dna in <code>refseq</code> slot of a <code>physeq</code> object using taxa names and renames taxa</p>
using ASV_1, ASV_2, …</a></li>
<li><a href='#add_funguild_info'><p>Add information about Guild for FUNGI the FUNGuild databse</p></a></li>
<li><a href='#add_info_to_sam_data'><p>Add information to sample_data slot of a phyloseq-class object</p></a></li>
<li><a href='#add_new_taxonomy_pq'><p>Add new taxonomic rank to a phyloseq object.</p></a></li>
<li><a href='#adonis_pq'><p>Permanova on a phyloseq object</p></a></li>
<li><a href='#adonis_rarperm_pq'><p>Permanova (adonis) on permutations of rarefaction even depth</p></a></li>
<li><a href='#all_object_size'><p>List the size of all objects of the GlobalEnv.</p></a></li>
<li><a href='#ancombc_pq'><p>Run ANCOMBC2 on phyloseq object</p></a></li>
<li><a href='#are_modality_even_depth'><p>Test if the mean number of sequences by samples is link to the modality of</p>
a factor</a></li>
<li><a href='#as_binary_otu_table'><p>Transform the otu_table of a <code>phyloseq-class</code> object into a</p>
<code>phyloseq-class</code> object with a binary otu_table.</a></li>
<li><a href='#asv2otu'><p>Recluster sequences of an object of class <code>physeq</code></p>
or a list of DNA sequences</a></li>
<li><a href='#biplot_pq'><p>Visualization of two samples for comparison</p></a></li>
<li><a href='#blast_pq'><p>Blast all sequence of <code>refseq</code> slot of a <code>phyloseq-class</code></p>
object against a custom database.</a></li>
<li><a href='#blast_to_derep'><p>Blast some sequence against sequences from of a <code>derep-class</code></p>
object.</a></li>
<li><a href='#blast_to_phyloseq'><p>Blast some sequence against <code>refseq</code> slot of a <code>phyloseq-class</code></p>
object.</a></li>
<li><a href='#build_phytree_pq'><p>Build phylogenetic trees from refseq slot of a phyloseq object</p></a></li>
<li><a href='#chimera_detection_vs'><p>Detect for chimera taxa using <a href="https://github.com/torognes/vsearch">vsearch</a></p></a></li>
<li><a href='#chimera_removal_vs'><p>Search for a list of sequence in an object to remove chimera taxa</p>
using <a href="https://github.com/torognes/vsearch">vsearch</a></a></li>
<li><a href='#circle_pq'><p>Plot OTU circle for <code>phyloseq-class</code> object</p></a></li>
<li><a href='#clean_pq'><p>Clean phyloseq object by removing empty samples and taxa</p></a></li>
<li><a href='#compare_pairs_pq'><p>Compare samples in pairs using diversity and number of ASV including</p>
shared ASV.</a></li>
<li><a href='#count_seq'><p>Count sequences in fasta or fastq file</p></a></li>
<li><a href='#cutadapt_remove_primers'><p>Remove primers using <a href="https://github.com/marcelm/cutadapt/">cutadapt</a></p></a></li>
<li><a href='#data_fungi'><p>Fungal OTU in phyloseq format</p></a></li>
<li><a href='#data_fungi_mini'><p>Fungal OTU in phyloseq format</p></a></li>
<li><a href='#data_fungi_sp_known'><p>Fungal OTU in phyloseq format</p></a></li>
<li><a href='#diff_fct_diff_class'><p>Compute different functions for different class of vector.</p></a></li>
<li><a href='#dist_bycol'><p>Compute paired distances among matrix (e.g. otu_table)</p></a></li>
<li><a href='#dist_pos_control'><p>Calculate ecological distance among positive controls vs</p>
distance for all samples</a></li>
<li><a href='#distri_1_taxa'><p>Distribution of sequences across a factor for one taxon</p></a></li>
<li><a href='#fac2col'><p>Translates a factor into colors.</p></a></li>
<li><a href='#filter_asv_blast'><p>Filter undesirable taxa using blast against a custom database.</p></a></li>
<li><a href='#filter_trim'><p>A wrapper of the function <code>dada2::filterAndTrim()</code> to use in</p>
<a href="https://books.ropensci.org/targets/">targets</a> pipeline</a></li>
<li><a href='#formattable_pq'><p>Create an visualization table to describe taxa distribution across a modality</p></a></li>
<li><a href='#funguild_assign'><p>Assign Guilds to Organisms Based on Taxonomic Classification</p></a></li>
<li><a href='#funky_color'><p>Funky palette color</p></a></li>
<li><a href='#get_file_extension'><p>Get the extension of a file</p></a></li>
<li><a href='#get_funguild_db'><p>Retrieve the FUNGuild database</p></a></li>
<li><a href='#ggaluv_pq'><p>Alluvial plot for taxonomy and samples factor vizualisation</p></a></li>
<li><a href='#ggbetween_pq'><p>Box/Violin plots for between-subjects comparisons of Hill Number</p></a></li>
<li><a href='#ggscatt_pq'><p>Scatterplot with marginal distributions and statistical results against</p>
Hill diversity of phyloseq object</a></li>
<li><a href='#ggvenn_pq'><p>Venn diagram of <code>phyloseq-class</code> object using</p>
<code>ggVennDiagram::ggVennDiagram</code> function</a></li>
<li><a href='#glmutli_pq'><p>Automated model selection and multimodel inference with (G)LMs for phyloseq</p></a></li>
<li><a href='#graph_test_pq'><p>Performs graph-based permutation tests on phyloseq object</p></a></li>
<li><a href='#heat_tree_pq'><p>Heat tree from <code>metacoder</code> package using <code>tax_table</code> slot</p></a></li>
<li><a href='#hill_pq'><p>Graphical representation of hill number 0, 1 and 2 across a factor</p></a></li>
<li><a href='#hill_test_rarperm_pq'><p>Test multiple times effect of factor on Hill diversity</p>
with different rarefaction even depth</a></li>
<li><a href='#hill_tuckey_pq'><p>Calculate hill number and compute Tuckey post-hoc test</p></a></li>
<li><a href='#iNEXT_pq'><p>iNterpolation and EXTrapolation of Hill numbers (with iNEXT)</p></a></li>
<li><a href='#is_cutadapt_installed'><p>Test if cutadapt is installed.</p></a></li>
<li><a href='#is_falco_installed'><p>Test if falco is installed.</p></a></li>
<li><a href='#is_krona_installed'><p>Test if krona is installed.</p></a></li>
<li><a href='#is_mumu_installed'><p>Test if mumu is installed.</p></a></li>
<li><a href='#is_swarm_installed'><p>Test if swarm is installed.</p></a></li>
<li><a href='#is_vsearch_installed'><p>Test if vsearch is installed.</p></a></li>
<li><a href='#krona'><p>Make Krona files using <a href="https://github.com/marbl/Krona/wiki">KronaTools</a>.</p></a></li>
<li><a href='#LCBD_pq'><p>Compute and test local contributions to beta diversity (LCBD) of</p>
samples</a></li>
<li><a href='#list_fastq_files'><p>List fastq files</p></a></li>
<li><a href='#lulu'><p>Post Clustering Curation of Amplicon Data.</p></a></li>
<li><a href='#lulu_pq'><p>Lulu reclustering of class <code>physeq</code></p></a></li>
<li><a href='#merge_krona'><p>Merge Krona files using <a href="https://github.com/marbl/Krona/wiki">KronaTools</a>.</p></a></li>
<li><a href='#merge_samples2'><p>Merge samples by a sample variable or factor</p></a></li>
<li><a href='#merge_taxa_vec'><p>Merge taxa in groups (vectorized version)</p></a></li>
<li><a href='#MiscMetabar-deprecated'><p>Deprecated function(s) in the MiscMetabar package</p></a></li>
<li><a href='#multi_biplot_pq'><p>Visualization of a collection of couples of samples for comparison</p></a></li>
<li><a href='#multipatt_pq'><p>Test and plot multipatt result</p></a></li>
<li><a href='#multiplot'><p>Multiple plot function</p></a></li>
<li><a href='#multitax_bar_pq'><p>Plot taxonomic distribution across 3 taxonomic levels and optionally</p>
one sample factor</a></li>
<li><a href='#mumu_pq'><p>MUMU reclustering of class <code>physeq</code></p></a></li>
<li><a href='#normalize_prop_pq'><p>Normalize OTU table using samples depth</p></a></li>
<li><a href='#perc'><p>Convert a value (or a fraction x/y) in percentage</p></a></li>
<li><a href='#phyloseq_to_edgeR'><p>Convert phyloseq OTU count data into DGEList for edgeR package</p></a></li>
<li><a href='#physeq_or_string_to_dna'><p>Return a DNAStringSet object from either a character vector of DNA sequences</p>
or the <code>refseq</code> slot of a phyloseq-class object</a></li>
<li><a href='#plot_ancombc_pq'><p>Plot ANCOMBC2 result for phyloseq object</p></a></li>
<li><a href='#plot_deseq2_pq'><p>Plot DESeq2 results for a phyloseq or a DESeq2 object.</p></a></li>
<li><a href='#plot_edgeR_pq'><p>Plot edgeR results for a phyloseq or a edgeR object.</p></a></li>
<li><a href='#plot_guild_pq'><p>Plot information about Guild from tax_table slot previously</p>
created with <code>add_funguild_info()</code></a></li>
<li><a href='#plot_LCBD_pq'><p>Plot and test local contributions to beta diversity (LCBD) of samples</p></a></li>
<li><a href='#plot_mt'><p>Plot the result of a mt test <code>phyloseq::mt()</code></p></a></li>
<li><a href='#plot_SCBD_pq'><p>Plot species contributions to beta diversity (SCBD) of samples</p></a></li>
<li><a href='#plot_tax_pq'><p>Plot taxonomic distribution in function of a factor with stacked bar in %</p></a></li>
<li><a href='#plot_tsne_pq'><p>Plot a tsne low dimensional representation of a phyloseq object</p></a></li>
<li><a href='#plot_var_part_pq'><p>Plot the partition the variation of a phyloseq object</p></a></li>
<li><a href='#psmelt_samples_pq'><p>Build a sample information tibble from physeq object</p></a></li>
<li><a href='#rarefy_sample_count_by_modality'><p>Rarefy (equalize) the number of samples per modality of a factor</p></a></li>
<li><a href='#read_pq'><p>Read phyloseq object from multiple csv tables and a phylogenetic tree</p>
in Newick format.</a></li>
<li><a href='#rename_samples'><p>Rename the samples of a phyloseq slot</p></a></li>
<li><a href='#rename_samples_otu_table'><p>Rename samples of an otu_table</p></a></li>
<li><a href='#reorder_taxa_pq'><p>Reorder taxa in otu_table/tax_table/refseq slot of a phyloseq object</p></a></li>
<li><a href='#ridges_pq'><p>Ridge plot of a phyloseq object</p></a></li>
<li><a href='#rotl_pq'><p>rotl wrapper for phyloseq data</p></a></li>
<li><a href='#sample_data_with_new_names'><p>Load sample data from file and rename samples using names of samples and an</p>
optional order</a></li>
<li><a href='#sankey_pq'><p>Sankey plot of <code>phyloseq-class</code> object</p></a></li>
<li><a href='#save_pq'><p>A wrapper of write_pq to save in all three possible formats</p></a></li>
<li><a href='#search_exact_seq_pq'><p>Search for exact matching of sequences using complement,</p>
reverse and reverse-complement</a></li>
<li><a href='#select_one_sample'><p>Select one sample from a physeq object</p></a></li>
<li><a href='#select_taxa'><p>Select a subset of taxa in a specified order where possible</p></a></li>
<li><a href='#signif_ancombc'><p>Filter ancombc_pq results</p></a></li>
<li><a href='#simplify_taxo'><p>Simplify taxonomy by removing some unused characters such as &quot;k__&quot;</p></a></li>
<li><a href='#SRS_curve_pq'><p>Scaling with ranked subsampling (SRS) curve of phyloseq object</p></a></li>
<li><a href='#subsample_fastq'><p>Subsample a fastq file copying the n_seq first sequences in a given folder</p></a></li>
<li><a href='#subset_samples_pq'><p>Subset samples using a conditional boolean vector.</p></a></li>
<li><a href='#subset_taxa_pq'><p>Subset taxa using a conditional named boolean vector.</p></a></li>
<li><a href='#subset_taxa_tax_control'><p>Subset taxa using a taxa control (e.g. truffle root tips) through 3 methods.</p></a></li>
<li><a href='#summary_plot_pq'><p>Summarize a <code>phyloseq-class</code> object using a plot.</p></a></li>
<li><a href='#swarm_clustering'><p>Re-cluster sequences of an object of class <code>physeq</code></p>
or cluster a list of DNA sequences using SWARM</a></li>
<li><a href='#tax_bar_pq'><p>Plot the distribution of sequences or ASV in one taxonomic levels</p></a></li>
<li><a href='#tax_datatable'><p>Make a datatable with the taxonomy of a <code>phyloseq-class</code> object</p></a></li>
<li><a href='#taxa_as_columns'><p>Force taxa to be in columns in the otu_table of a physeq object</p></a></li>
<li><a href='#taxa_as_rows'><p>Force taxa to be in columns in the otu_table of a physeq object</p></a></li>
<li><a href='#taxa_only_in_one_level'><p>Show taxa which are present in only one given level of a modality</p></a></li>
<li><a href='#tbl_sum_samdata'><p>Summarize information from sample data in a table</p></a></li>
<li><a href='#Tengeler2020_pq'><p>This tutorial explores the dataset from Tengeler et al. (2020) available in the <code>mia</code> package.</p>
obtained using <code>mia::makePhyloseqFromTreeSE(Tengeler2020)</code></a></li>
<li><a href='#track_wkflow'><p>Track the number of reads (= sequences), samples and cluster (e.g. ASV)</p>
from various objects including dada-class and derep-class.</a></li>
<li><a href='#track_wkflow_samples'><p>Track the number of reads (= sequences), samples and cluster (e.g. ASV)</p>
for each sample</a></li>
<li><a href='#transp'><p>Adds transparency to a vector of colors</p></a></li>
<li><a href='#treemap_pq'><p>Plot treemap of 2 taxonomic levels</p></a></li>
<li><a href='#tsne_pq'><p>Compute tSNE position of samples from a phyloseq object</p></a></li>
<li><a href='#unique_or_na'><p>Get the unique value in x or NA if none</p></a></li>
<li><a href='#upset_pq'><p>Make upset plot for phyloseq object.</p></a></li>
<li><a href='#upset_test_pq'><p>Test for differences between intersections</p></a></li>
<li><a href='#var_par_pq'><p>Partition the Variation of a phyloseq object by 2, 3, or 4 Explanatory Matrices</p></a></li>
<li><a href='#var_par_rarperm_pq'><p>Partition the Variation of a phyloseq object with rarefaction permutations</p></a></li>
<li><a href='#venn_pq'><p>Venn diagram of <code>phyloseq-class</code> object</p></a></li>
<li><a href='#verify_pq'><p>Verify the validity of a phyloseq object</p></a></li>
<li><a href='#vs_search_global'><p>Search for a list of sequence in a fasta file against physeq reference</p>
sequences using <a href="https://github.com/torognes/vsearch">vsearch</a></a></li>
<li><a href='#vsearch_clustering'><p>Recluster sequences of an object of class <code>physeq</code></p>
or cluster a list of DNA sequences using vsearch software</a></li>
<li><a href='#write_pq'><p>Save phyloseq object in the form of multiple csv tables.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Miscellaneous Functions for Metabarcoding Analysis</td>
</tr>
<tr>
<td>Version:</td>
<td>0.9.1</td>
</tr>
<tr>
<td>Description:</td>
<td>Facilitate the description, transformation, exploration, and reproducibility of metabarcoding analyses. 'MiscMetabar' is mainly built on top of the 'phyloseq', 'dada2' and 'targets' R packages. It helps to build reproducible and robust bioinformatics pipelines in R. 'MiscMetabar' makes ecological analysis of alpha and beta-diversity easier, more reproducible and more powerful by integrating a large number of tools. Important features are described in Taudière A. (2023) &lt;<a href="https://doi.org/10.21105%2Fjoss.06038">doi:10.21105/joss.06038</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/AGPL-3">AGPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0), phyloseq, ggplot2 (&ge; 3.5.0), dada2, dplyr,
purrr</td>
</tr>
<tr>
<td>Suggests:</td>
<td>adespatial, ANCOMBC, BiocManager, Biostrings, circlize,
ComplexUpset, DECIPHER, DESeq2, devtools, DT, edgeR,
formattable, ggalluvial, ggfittext, gghalves, ggh4x,
ggstatsplot, ggridges, ggVennDiagram, glmulti, gtsummary,
grDevices, grid, gridExtra, here, httr, iNEXT, indicspecies,
jsonlite, knitr, magrittr, metacoder, methods, mia, mixtools,
multcompView, networkD3, pak, patchwork, permute, phangorn,
phyloseqGraphTest, pbapply, plotly, plyr, reshape2, rmarkdown,
rotl, Rtsne, scales, seqinr, SRS, stringr,
SummarizedExperiment, testthat (&ge; 3.0.0), tibble, tidyr,
treemapify, vegan, venneuler, vctrs, viridis, withr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.3.1</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/adrientaudiere/MiscMetabar">https://github.com/adrientaudiere/MiscMetabar</a>,
<a href="https://adrientaudiere.github.io/MiscMetabar/">https://adrientaudiere.github.io/MiscMetabar/</a></td>
</tr>
<tr>
<td>biocViews:</td>
<td>Sequencing, Microbiome, Metagenomics, Clustering,
Classification, Visualization</td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/adrientaudiere/MiscMetabar/issues">https://github.com/adrientaudiere/MiscMetabar/issues</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>ape, lifecycle, rlang, stats</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>Config/testthat/parallel:</td>
<td>true</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>Language:</td>
<td>en-US</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-04-28 12:49:09 UTC; adrien</td>
</tr>
<tr>
<td>Author:</td>
<td>Adrien Taudière <a href="https://orcid.org/0000-0003-1088-1182"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Adrien Taudière &lt;adrien.taudiere@zaclys.net&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-04-28 14:10:07 UTC</td>
</tr>
</table>
<hr>
<h2 id='MiscMetabar-package'><code>MiscMetabar</code> package</h2><span id='topic+MiscMetabar-package'></span>

<h3>Description</h3>

<p>Functions to help analyze and visualize metabarcoding data. Mainly based on
the phyloseq and dada2 packages.
</p>

<hr>
<h2 id='accu_plot'>Plot accumulation curves for <code>phyloseq-class</code> object</h2><span id='topic+accu_plot'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accu_plot(
  physeq,
  fact = NULL,
  add_nb_seq = TRUE,
  step = NULL,
  by.fact = FALSE,
  ci_col = NULL,
  col = NULL,
  lwd = 3,
  leg = TRUE,
  print_sam_names = FALSE,
  ci = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accu_plot_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot
different lines</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_add_nb_seq">add_nb_seq</code></td>
<td>
<p>(default: TRUE, logical)
Either plot accumulation curves using sequences or using samples</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_step">step</code></td>
<td>
<p>(Integer) distance among points calculated to plot lines. A
low value give better plot but is more time consuming.
Only used if <code>add_nb_seq</code> = TRUE.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_by.fact">by.fact</code></td>
<td>
<p>(default: FALSE, logical)
First merge the OTU table by factor to plot only one line by factor</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_ci_col">ci_col</code></td>
<td>
<p>Color vector for confidence interval.
Only use if <code>add_nb_seq</code> = FALSE.
If <code>add_nb_seq</code> = TRUE, you can use ggplot to modify the plot.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_col">col</code></td>
<td>
<p>Color vector for lines. Only use if <code>add_nb_seq</code> = FALSE.
If <code>add_nb_seq</code> = TRUE, you can use ggplot to modify the plot.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_lwd">lwd</code></td>
<td>
<p>(default: 3) thickness for lines. Only use if <code>add_nb_seq</code> = FALSE.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_leg">leg</code></td>
<td>
<p>(default: TRUE, logical) Plot legend or not. Only use if <code>add_nb_seq</code> = FALSE.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_print_sam_names">print_sam_names</code></td>
<td>
<p>(default: FALSE, logical) Print samples names or not?
Only use if <code>add_nb_seq</code> = TRUE.</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_ci">ci</code></td>
<td>
<p>(default: 2, integer) Confidence interval value used to multiply the
standard error to plot confidence interval</p>
</td></tr>
<tr><td><code id="accu_plot_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>
if <code>add_nb_seq</code> = TRUE or to <code><a href="base.html#topic+plot">plot</a></code> if <code>add_nb_seq</code> = FALSE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot representing the richness
accumulation plot if add_nb_seq = TRUE, else, if add_nb_seq = FALSE
return a base plot.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+specaccum">specaccum</a></code> <code><a href="#topic+accu_samp_threshold">accu_samp_threshold()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("GlobalPatterns", package = "phyloseq")
GP &lt;- subset_taxa(GlobalPatterns, GlobalPatterns@tax_table[, 1] == "Archaea")
GP &lt;- rarefy_even_depth(subset_samples_pq(GP, sample_sums(GP) &gt; 3000))
p &lt;- accu_plot(GP, "SampleType", add_nb_seq = TRUE, by.fact = TRUE, step = 10)
p &lt;- accu_plot(GP, "SampleType", add_nb_seq = TRUE, step = 10)

p + theme(legend.position = "none")

p + xlim(c(0, 400))

</code></pre>

<hr>
<h2 id='accu_plot_balanced_modality'>Plot accumulation curves with balanced modality and depth rarefaction</h2><span id='topic+accu_plot_balanced_modality'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This function (i) rarefy (equalize) the number of samples per modality of a
factor and (ii) rarefy the number of sequences per sample (depth). The
seed is set to 1:nperm. Thus, with exacly the same parameter, including
nperm values, results must be identical.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accu_plot_balanced_modality(
  physeq,
  fact,
  nperm = 99,
  step = 2000,
  by.fact = TRUE,
  progress_bar = TRUE,
  quantile_prob = 0.975,
  rarefy_by_sample_before_merging = TRUE,
  sample.size = 1000,
  verbose = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accu_plot_balanced_modality_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_fact">fact</code></td>
<td>
<p>(required) The variable to rarefy. Must be present in
the <code>sam_data</code> slot of the physeq object.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_nperm">nperm</code></td>
<td>
<p>(int) The number of permutations to perform.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_step">step</code></td>
<td>
<p>(int) distance among points calculated to plot lines.
A low value give better plot but is more time consuming.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_by.fact">by.fact</code></td>
<td>
<p>(logical, default TRUE)
First merge the OTU table by factor to plot only one line by factor</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_progress_bar">progress_bar</code></td>
<td>
<p>(logical, default TRUE) Do we print progress during
the calculation?</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_quantile_prob">quantile_prob</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) the value to compute the quantile.
Minimum quantile is compute using 1-quantile_prob.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_rarefy_by_sample_before_merging">rarefy_by_sample_before_merging</code></td>
<td>
<p>(logical, default TRUE):
rarefy_by_sample_before_merging = FALSE is buggy for the moment.Please
only use rarefy_by_sample_before_merging = TRUE</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_sample.size">sample.size</code></td>
<td>
<p>(int) A single integer value equal to the number of
reads being simulated, also known as the depth. See
<code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code>.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_verbose">verbose</code></td>
<td>
<p>(logical). If TRUE, print additional informations.</p>
</td></tr>
<tr><td><code id="accu_plot_balanced_modality_+3A_...">...</code></td>
<td>
<p>Other params for be passed on to <code><a href="#topic+accu_plot">accu_plot()</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 plot representing the richness accumulation plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+accu_plot">accu_plot()</a></code>, <code><a href="#topic+rarefy_sample_count_by_modality">rarefy_sample_count_by_modality()</a></code>, <code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data_fungi_woNA4Time &lt;-
  subset_samples(data_fungi, !is.na(Time))
data_fungi_woNA4Time@sam_data$Time &lt;- paste0("time-", data_fungi_woNA4Time@sam_data$Time)
accu_plot_balanced_modality(data_fungi_woNA4Time, "Time", nperm = 3)

data_fungi_woNA4Height &lt;-
  subset_samples(data_fungi, !is.na(Height))
accu_plot_balanced_modality(data_fungi_woNA4Height, "Height", nperm = 3)

</code></pre>

<hr>
<h2 id='accu_samp_threshold'>Compute the number of sequence to obtain a given proportion of ASV in
accumulation curves</h2><span id='topic+accu_samp_threshold'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>accu_samp_threshold(res_accuplot, threshold = 0.95)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="accu_samp_threshold_+3A_res_accuplot">res_accuplot</code></td>
<td>
<p>the result of the function accu_plot()</p>
</td></tr>
<tr><td><code id="accu_samp_threshold_+3A_threshold">threshold</code></td>
<td>
<p>the proportion of ASV to obtain in each samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a value for each sample of the number of sequences needed
to obtain <code>threshold</code> proportion of the ASV
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+accu_plot">accu_plot()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("GlobalPatterns", package = "phyloseq")
GP &lt;- subset_taxa(GlobalPatterns, GlobalPatterns@tax_table[, 1] == "Archaea")
GP &lt;- rarefy_even_depth(subset_samples_pq(GP, sample_sums(GP) &gt; 3000))
p &lt;- accu_plot(GP, "SampleType", add_nb_seq = TRUE, by.fact = TRUE, step = 10)

val_threshold &lt;- accu_samp_threshold(p)

summary(val_threshold)

# Plot the number of sequences needed to accumulate 0.95% of ASV in 50%, 75%
# and 100% of samples
p + geom_vline(xintercept = quantile(val_threshold, probs = c(0.50, 0.75, 1)))

</code></pre>

<hr>
<h2 id='add_blast_info'>Add information from <code><a href="#topic+blast_pq">blast_pq()</a></code> to the <code>tax_table</code> slot of a <em>phyloseq</em> object</h2><span id='topic+add_blast_info'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Basically a wrapper of <code><a href="#topic+blast_pq">blast_pq()</a></code> with option <code>unique_per_seq = TRUE</code> and
<code>score_filter = FALSE</code>.
</p>
<p>Add the information to the taxtable
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_blast_info(physeq, fasta_for_db, silent = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_blast_info_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="add_blast_info_+3A_fasta_for_db">fasta_for_db</code></td>
<td>
<p>path to a fasta file to make the blast database</p>
</td></tr>
<tr><td><code id="add_blast_info_+3A_silent">silent</code></td>
<td>
<p>(logical) If true, no message are printing.</p>
</td></tr>
<tr><td><code id="add_blast_info_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="#topic+blast_pq">blast_pq()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object with more information in tax_table based on a
blast on a given database
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='add_dna_to_phyloseq'>Add dna in <code>refseq</code> slot of a <code>physeq</code> object using taxa names and renames taxa
using ASV_1, ASV_2, …</h2><span id='topic+add_dna_to_phyloseq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_dna_to_phyloseq(physeq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_dna_to_phyloseq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object with <code>refseq</code> slot and new
taxa names
</p>

<hr>
<h2 id='add_funguild_info'>Add information about Guild for FUNGI the FUNGuild databse</h2><span id='topic+add_funguild_info'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Please cite Nguyen et al. 2016 (<a href="https://doi.org/10.1016/j.funeco.2015.06.006">doi:10.1016/j.funeco.2015.06.006</a>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_funguild_info(
  physeq,
  taxLevels = c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_funguild_info_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="add_funguild_info_+3A_taxlevels">taxLevels</code></td>
<td>
<p>Name of the 7 columns in tax_table required by funguild</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>FUNGuildR</code> package and the associate
<a href="https://www.sciencedirect.com/science/article/abs/pii/S1754504815000847">publication</a> if you
use this function.
</p>


<h3>Value</h3>

<p>A new object of class <code>physeq</code> with Guild information added to
<code>tax_table</code> slot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_guild_pq">plot_guild_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("httr")) {
  d_fung_mini &lt;- add_funguild_info(data_fungi_mini,
    taxLevels = c(
      "Domain",
      "Phylum",
      "Class",
      "Order",
      "Family",
      "Genus",
      "Species"
    )
  )
  sort(table(d_fung_mini@tax_table[, "guild"]), decreasing = TRUE)
}

</code></pre>

<hr>
<h2 id='add_info_to_sam_data'>Add information to sample_data slot of a phyloseq-class object</h2><span id='topic+add_info_to_sam_data'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Warning: The value nb_seq and nb_otu may be outdated if you transform your
phyloseq object, e.g. using the <code><a href="#topic+subset_taxa_pq">subset_taxa_pq()</a></code> function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_info_to_sam_data(
  physeq,
  df_info = NULL,
  add_nb_seq = TRUE,
  add_nb_otu = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_info_to_sam_data_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="add_info_to_sam_data_+3A_df_info">df_info</code></td>
<td>
<p>: A dataframe with rownames matching for sample names of the
phyloseq object</p>
</td></tr>
<tr><td><code id="add_info_to_sam_data_+3A_add_nb_seq">add_nb_seq</code></td>
<td>
<p>(Logical, default TRUE) Does we add a column nb_seq
collecting the number of sequences per sample?</p>
</td></tr>
<tr><td><code id="add_info_to_sam_data_+3A_add_nb_otu">add_nb_otu</code></td>
<td>
<p>(Logical, default TRUE) Does we add a column nb_otu
collecting the number of OTUs per sample?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A phyloseq object with an updated sam_data slot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data_fungi &lt;- add_info_to_sam_data(data_fungi)
boxplot(data_fungi@sam_data$nb_otu ~ data_fungi@sam_data$Time)

new_df &lt;- data.frame(
  variable_1 = runif(n = nsamples(data_fungi), min = 1, max = 20),
  variable_2 = runif(n = nsamples(data_fungi), min = 1, max = 2)
)
rownames(new_df) &lt;- sample_names(data_fungi)
data_fungi &lt;- add_info_to_sam_data(data_fungi, new_df)
plot(data_fungi@sam_data$nb_otu ~ data_fungi@sam_data$variable_1)
</code></pre>

<hr>
<h2 id='add_new_taxonomy_pq'>Add new taxonomic rank to a phyloseq object.</h2><span id='topic+add_new_taxonomy_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>One of main use of this function is to add taxonomic assignment from
a new database.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_new_taxonomy_pq(physeq, ref_fasta, suffix = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_new_taxonomy_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="add_new_taxonomy_pq_+3A_ref_fasta">ref_fasta</code></td>
<td>
<p>(required) A link to a database.
Pass on to <code>dada2::assignTaxonomy</code>.</p>
</td></tr>
<tr><td><code id="add_new_taxonomy_pq_+3A_suffix">suffix</code></td>
<td>
<p>(character) The suffix to name the new columns.
If set to NULL (the default), the basename of the file reFasta
is used.</p>
</td></tr>
<tr><td><code id="add_new_taxonomy_pq_+3A_...">...</code></td>
<td>
<p>Others arguments pass on to <code>dada2::assignTaxonomy</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object with a larger slot tax_table&quot;
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='adonis_pq'>Permanova on a phyloseq object</h2><span id='topic+adonis_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code> function in the case of <code>physeq</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adonis_pq(
  physeq,
  formula,
  dist_method = "bray",
  merge_sample_by = NULL,
  na_remove = FALSE,
  correction_for_sample_size = FALSE,
  rarefy_nb_seqs = FALSE,
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adonis_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_formula">formula</code></td>
<td>
<p>(required) the right part of a formula for <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code>.
Variables must be present in the <code>physeq@sam_data</code> slot.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_dist_method">dist_method</code></td>
<td>
<p>(default &quot;bray&quot;) the distance used. See
<code><a href="phyloseq.html#topic+distance">phyloseq::distance()</a></code> for all available distances or run
<code><a href="phyloseq.html#topic+distanceMethodList">phyloseq::distanceMethodList()</a></code>.
For aitchison and robust.aitchison distance, <code><a href="vegan.html#topic+vegdist">vegan::vegdist()</a></code>
function is directly used.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>a vector to determine
which samples to merge using the <code><a href="#topic+merge_samples2">merge_samples2()</a></code>
function. Need to be in <code>physeq@sam_data</code></p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default FALSE) If set to TRUE, remove samples with
NA in the variables set in formula.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_correction_for_sample_size">correction_for_sample_size</code></td>
<td>
<p>(logical, default FALSE) If set to TRUE,
the sample size (number of sequences by samples) is add to formula in
the form <code>y~Library_Size + Biological_Effect</code> following recommendation of
<a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-017-0237-y">Weiss et al. 2017</a>.
<code>correction_for_sample_size</code> overcome <code>rarefy_nb_seqs</code> if both are TRUE.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_rarefy_nb_seqs">rarefy_nb_seqs</code></td>
<td>
<p>(logical, default FALSE) Rarefy each sample
(before merging if merge_sample_by is set) using
<code>phyloseq::rarefy_even_depth()</code>.
if <code>correction_for_sample_size</code> is TRUE, rarefy_nb_seqs will have no
effect.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical, default TRUE) If TRUE, prompt some messages.</p>
</td></tr>
<tr><td><code id="adonis_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>vegan::adonis2()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>The function returns an anova.cca result object with a
new column for partial R^2. See help of <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code> for
more information.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(enterotype)

adonis_pq(enterotype, "SeqTech*Enterotype", na_remove = TRUE)
adonis_pq(enterotype, "SeqTech", dist_method = "jaccard")
adonis_pq(enterotype, "SeqTech", dist_method = "robust.aitchison")

</code></pre>

<hr>
<h2 id='adonis_rarperm_pq'>Permanova (adonis) on permutations of rarefaction even depth</h2><span id='topic+adonis_rarperm_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>adonis_rarperm_pq(
  physeq,
  formula,
  dist_method = "bray",
  merge_sample_by = NULL,
  na_remove = FALSE,
  rarefy_nb_seqs = FALSE,
  verbose = TRUE,
  nperm = 99,
  progress_bar = TRUE,
  quantile_prob = 0.975,
  sample.size = min(sample_sums(physeq)),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="adonis_rarperm_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_formula">formula</code></td>
<td>
<p>(required) the right part of a formula for <code><a href="vegan.html#topic+adonis">vegan::adonis2()</a></code>.
Variables must be present in the <code>physeq@sam_data</code> slot.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_dist_method">dist_method</code></td>
<td>
<p>(default &quot;bray&quot;) the distance used. See
<code><a href="phyloseq.html#topic+distance">phyloseq::distance()</a></code> for all available distances or run
<code><a href="phyloseq.html#topic+distanceMethodList">phyloseq::distanceMethodList()</a></code>.
For aitchison and robust.aitchison distance, <code><a href="vegan.html#topic+vegdist">vegan::vegdist()</a></code>
function is directly used.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>a vector to determine
which samples to merge using the <code><a href="#topic+merge_samples2">merge_samples2()</a></code>
function. Need to be in <code>physeq@sam_data</code></p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default FALSE) If set to TRUE, remove samples with
NA in the variables set in formula.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_rarefy_nb_seqs">rarefy_nb_seqs</code></td>
<td>
<p>(logical, default FALSE) Rarefy each sample
(before merging if merge_sample_by is set) using
<code>phyloseq::rarefy_even_depth()</code>.
if <code>correction_for_sample_size</code> is TRUE, rarefy_nb_seqs will have no
effect.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical, default TRUE) If TRUE, prompt some messages.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_nperm">nperm</code></td>
<td>
<p>(int) The number of permutations to perform.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_progress_bar">progress_bar</code></td>
<td>
<p>(logical, default TRUE) Do we print progress during
the calculation.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_quantile_prob">quantile_prob</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) the value to compute the quantile.
Minimum quantile is compute using 1-quantile_prob.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_sample.size">sample.size</code></td>
<td>
<p>(int) A single integer value equal to the number of
reads being simulated, also known as the depth. See
<code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code>.</p>
</td></tr>
<tr><td><code id="adonis_rarperm_pq_+3A_...">...</code></td>
<td>
<p>Other params for be passed on to <code><a href="#topic+adonis_pq">adonis_pq()</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of three dataframe representing the mean, the minimum quantile
and the maximum quantile value for adonis results. See <code><a href="#topic+adonis_pq">adonis_pq()</a></code>.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+adonis_pq">adonis_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("vegan")) {
  data_fungi_woNA &lt;-
    subset_samples(data_fungi, !is.na(Time) &amp; !is.na(Height))
  adonis_rarperm_pq(data_fungi_woNA, "Time*Height", na_remove = TRUE, nperm = 3)
}
</code></pre>

<hr>
<h2 id='all_object_size'>List the size of all objects of the GlobalEnv.</h2><span id='topic+all_object_size'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Code from https://tolstoy.newcastle.edu.au/R/e6/help/09/01/1121.html
</p>


<h3>Usage</h3>

<pre><code class='language-R'>all_object_size()
</code></pre>


<h3>Value</h3>

<p>a list of size
</p>

<hr>
<h2 id='ancombc_pq'>Run ANCOMBC2 on phyloseq object</h2><span id='topic+ancombc_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="ANCOMBC.html#topic+ancombc2">ANCOMBC::ancombc2()</a></code> function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ancombc_pq(physeq, fact, levels_fact = NULL, tax_level = "Class", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ancombc_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="ancombc_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot
different lines</p>
</td></tr>
<tr><td><code id="ancombc_pq_+3A_levels_fact">levels_fact</code></td>
<td>
<p>(default NULL) The order of the level in the factor.
Used for reorder levels and select levels (filter out levels not present
en levels_fact)</p>
</td></tr>
<tr><td><code id="ancombc_pq_+3A_tax_level">tax_level</code></td>
<td>
<p>The taxonomic level passed on to <code><a href="ANCOMBC.html#topic+ancombc2">ANCOMBC::ancombc2()</a></code></p>
</td></tr>
<tr><td><code id="ancombc_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ANCOMBC.html#topic+ancombc2">ANCOMBC::ancombc2()</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>ANCOMBC::ancombc2()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>The result of <code><a href="ANCOMBC.html#topic+ancombc2">ANCOMBC::ancombc2()</a></code> function
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("mia")) {
  data_fungi_mini@tax_table &lt;- phyloseq::tax_table(cbind(
    data_fungi_mini@tax_table,
    "taxon" = taxa_names(data_fungi_mini)
  ))
  res_height &lt;- ancombc_pq(
    data_fungi_mini,
    fact = "Height",
    levels_fact = c("Low", "High"),
    verbose = TRUE
  )

  ggplot(
    res_height$res,
    aes(
      y = reorder(taxon, lfc_HeightHigh),
      x = lfc_HeightHigh,
      color = diff_HeightHigh
    )
  ) +
    geom_vline(xintercept = 0) +
    geom_segment(aes(
      xend = 0, y = reorder(taxon, lfc_HeightHigh),
      yend = reorder(taxon, lfc_HeightHigh)
    ), color = "darkgrey") +
    geom_point()

  res_time &lt;- ancombc_pq(
    data_fungi_mini,
    fact = "Time",
    levels_fact = c("0", "15"),
    tax_level = "Family",
    verbose = TRUE
  )
}

</code></pre>

<hr>
<h2 id='are_modality_even_depth'>Test if the mean number of sequences by samples is link to the modality of
a factor</h2><span id='topic+are_modality_even_depth'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The aim of this function is to provide a warnings if samples depth significantly
vary among the modalities of a factor present in the <code>sam_data</code> slot.
</p>
<p>This function apply a Kruskal-Wallis rank sum test to the number of sequences
per samples in function of the factor <code>fact</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>are_modality_even_depth(physeq, fact, boxplot = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="are_modality_even_depth_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="are_modality_even_depth_+3A_fact">fact</code></td>
<td>
<p>(required): Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="are_modality_even_depth_+3A_boxplot">boxplot</code></td>
<td>
<p>(logical) Do you want to plot boxplot?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The result of a Kruskal-Wallis rank sum test
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
are_modality_even_depth(data_fungi_mini, "Time")$p.value
are_modality_even_depth(rarefy_even_depth(data_fungi_mini), "Time")$p.value
are_modality_even_depth(data_fungi_mini, "Height", boxplot = TRUE)
</code></pre>

<hr>
<h2 id='as_binary_otu_table'>Transform the otu_table of a <code>phyloseq-class</code> object into a
<code>phyloseq-class</code> object with a binary otu_table.</h2><span id='topic+as_binary_otu_table'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as_binary_otu_table(physeq, min_number = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="as_binary_otu_table_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="as_binary_otu_table_+3A_min_number">min_number</code></td>
<td>
<p>(int) the minimum number of sequences to put
a 1 in the OTU table.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code>physeq</code> object with only 0/1 in the OTU table
</p>


<h3>Note</h3>

<p>Useful to test if the results are not biased by sequences bias
that appended during PCR or NGS pipeline.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(enterotype)
enterotype_bin &lt;- as_binary_otu_table(enterotype)
</code></pre>

<hr>
<h2 id='asv2otu'>Recluster sequences of an object of class <code>physeq</code>
or a list of DNA sequences</h2><span id='topic+asv2otu'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>asv2otu(
  physeq = NULL,
  dna_seq = NULL,
  nproc = 1,
  method = "clusterize",
  id = 0.97,
  vsearchpath = "vsearch",
  tax_adjust = 0,
  vsearch_cluster_method = "--cluster_size",
  vsearch_args = "--strand both",
  keep_temporary_files = FALSE,
  swarmpath = "swarm",
  d = 1,
  swarm_args = "--fastidious",
  method_clusterize = "overlap",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="asv2otu_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_dna_seq">dna_seq</code></td>
<td>
<p>You may directly use a character vector of DNA sequences
in place of physeq args. When physeq is set, dna sequences take the value of
<code>physeq@refseq</code></p>
</td></tr>
<tr><td><code id="asv2otu_+3A_nproc">nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for the clustering</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_method">method</code></td>
<td>
<p>(default: clusterize)
Set the clustering method.
</p>

<ul>
<li> <p><code>clusterize</code> use the <code><a href="DECIPHER.html#topic+Clusterize">DECIPHER::Clusterize()</a></code> fonction,
</p>
</li>
<li> <p><code>vsearch</code> use the vsearch software (https://github.com/torognes/vsearch)
with arguments <code>--cluster_size</code> by default (see args <code>vsearch_cluster_method</code>)
and <code style="white-space: pre;">&#8288;-strand both&#8288;</code> (see args <code>vsearch_args</code>)
</p>
</li>
<li> <p><code>swarm</code> use the swarm
</p>
</li></ul>
</td></tr>
<tr><td><code id="asv2otu_+3A_id">id</code></td>
<td>
<p>(default: 0.97) level of identity to cluster</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_vsearchpath">vsearchpath</code></td>
<td>
<p>(default: vsearch) path to vsearch</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_tax_adjust">tax_adjust</code></td>
<td>
<p>(Default 0) See the man page
of <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code> for more details.
To conserved the taxonomic rank of the most abundant ASV,
set tax_adjust to 0 (default). For the moment only tax_adjust = 0 is
robust</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_vsearch_cluster_method">vsearch_cluster_method</code></td>
<td>
<p>(default: &quot;&ndash;cluster_size) See other possible
methods in the <a href="https://github.com/torognes/vsearch/">vsearch manual</a> (e.g. <code>--cluster_size</code> or <code>--cluster_smallmem</code>)
</p>

<ul>
<li> <p><code>--cluster_fast</code> : Clusterize the fasta sequences in filename, automatically sort by decreasing sequence length beforehand.
</p>
</li>
<li> <p><code>--cluster_size</code> : Clusterize the fasta sequences in filename, automatically sort by decreasing sequence abundance beforehand.
</p>
</li>
<li> <p><code>--cluster_smallmem</code> : Clusterize the fasta sequences in filename without automatically modifying their order beforehand. Sequence are expected to be sorted by decreasing sequence length, unless <em>&ndash;usersort</em> is used.
In that case you may set <code>vsearch_args</code> to vsearch_args = &quot;&ndash;strand both &ndash;usersort&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="asv2otu_+3A_vsearch_args">vsearch_args</code></td>
<td>
<p>(default : &quot;&ndash;strand both&quot;) a one length character element defining other parameters to
passed on to vsearch.</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files
</p>

<ul>
<li><p> temp.fasta (refseq in fasta or dna_seq sequences)
</p>
</li>
<li><p> cluster.fasta (centroid if method = &quot;vsearch&quot;)
</p>
</li>
<li><p> temp.uc (clusters if method = &quot;vsearch&quot;)
</p>
</li></ul>
</td></tr>
<tr><td><code id="asv2otu_+3A_swarmpath">swarmpath</code></td>
<td>
<p>(default: swarm) path to swarm</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_d">d</code></td>
<td>
<p>(default: 1) maximum number of differences allowed between two
amplicons, meaning that two amplicons will be grouped if they have <code>d</code>
(or less) differences</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_swarm_args">swarm_args</code></td>
<td>
<p>(default : &quot;&ndash;fastidious&quot;) a one length character
element defining other parameters to  passed on to swarm See other possible
methods in the <a href="https://github.com/torognes/swarm/blob/master/man/swarm_manual.pdf">SWARM pdf manual</a></p>
</td></tr>
<tr><td><code id="asv2otu_+3A_method_clusterize">method_clusterize</code></td>
<td>
<p>(default &quot;overlap&quot;) the method for the <code><a href="DECIPHER.html#topic+Clusterize">DECIPHER::Clusterize()</a></code> method</p>
</td></tr>
<tr><td><code id="asv2otu_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="DECIPHER.html#topic+Clusterize">DECIPHER::Clusterize()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function use the <code>merge_taxa_vec</code> function to
merge taxa into clusters. By default tax_adjust = 0. See the man page
of <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code>.
</p>


<h3>Value</h3>

<p>A new object of class <code>physeq</code> or a list of cluster if dna_seq
args was used.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>References</h3>

<p>VSEARCH can be downloaded from
<a href="https://github.com/torognes/vsearch">https://github.com/torognes/vsearch</a>.
More information in the associated publication
<a href="https://pubmed.ncbi.nlm.nih.gov/27781170">https://pubmed.ncbi.nlm.nih.gov/27781170</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+vsearch_clustering">vsearch_clustering()</a></code> and <code><a href="#topic+swarm_clustering">swarm_clustering()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("DECIPHER")) {
  asv2otu(data_fungi_mini)
}

if (requireNamespace("DECIPHER")) {
  asv2otu(data_fungi_mini, method_clusterize = "longest")

  if (MiscMetabar::is_swarm_installed()) {
    d_swarm &lt;- asv2otu(data_fungi_mini, method = "swarm")
  }
  if (MiscMetabar::is_vsearch_installed()) {
    d_vs &lt;- asv2otu(data_fungi_mini, method = "vsearch")
  }
}

</code></pre>

<hr>
<h2 id='biplot_pq'>Visualization of two samples for comparison</h2><span id='topic+biplot_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>biplot_pq(
  physeq,
  fact = NULL,
  merge_sample_by = NULL,
  rarefy_after_merging = FALSE,
  inverse_side = FALSE,
  left_name = NULL,
  left_name_col = "#4B3E1E",
  left_fill = "#4B3E1E",
  left_col = "#f3f2d9",
  right_name = NULL,
  right_name_col = "#1d2949",
  right_fill = "#1d2949",
  right_col = "#1d2949",
  log10trans = TRUE,
  nudge_y = c(0.3, 0.3),
  geom_label = FALSE,
  text_size = 3,
  size_names = 5,
  y_names = NA,
  ylim_modif = c(1, 1),
  nb_samples_info = TRUE,
  plotly_version = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="biplot_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_fact">fact</code></td>
<td>
<p>(default: NULL) Name of the factor in <code>physeq@sam_data</code>.
If left to NULL use the <code>left_name</code> and <code>right_name</code> parameter as modality.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>(default: NULL) if not <code>NULL</code> samples of
physeq are merged using the vector set by <code>merge_sample_by</code>. This
merging used the <code><a href="#topic+merge_samples2">merge_samples2()</a></code>. In the case of
<code><a href="#topic+biplot_pq">biplot_pq()</a></code> this must be a factor with two levels only.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_rarefy_after_merging">rarefy_after_merging</code></td>
<td>
<p>Rarefy each sample after merging by the
modalities merge_sample_by</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_inverse_side">inverse_side</code></td>
<td>
<p>Inverse the side (put the right modality in the left side).</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_left_name">left_name</code></td>
<td>
<p>Name fo the left sample.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_left_name_col">left_name_col</code></td>
<td>
<p>Color for the left name</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_left_fill">left_fill</code></td>
<td>
<p>Fill fo the left sample.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_left_col">left_col</code></td>
<td>
<p>Color fo the left sample.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_right_name">right_name</code></td>
<td>
<p>Name fo the right sample.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_right_name_col">right_name_col</code></td>
<td>
<p>Color for the right name</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_right_fill">right_fill</code></td>
<td>
<p>Fill fo the right sample.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_right_col">right_col</code></td>
<td>
<p>Color fo the right sample.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_log10trans">log10trans</code></td>
<td>
<p>(logical) Does abundancy is log10 transformed ?</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_nudge_y">nudge_y</code></td>
<td>
<p>A parameter to control the y position of abundancy values.
If a vector of two values are set. The first value is for the left side.
and the second value for the right one. If one value is set,
this value is used for both side.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_geom_label">geom_label</code></td>
<td>
<p>(default: FALSE, logical) if TRUE use the <code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_label()</a></code> function
instead of <code><a href="ggplot2.html#topic+geom_text">ggplot2::geom_text()</a></code> to indicate the numbers of sequences.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_text_size">text_size</code></td>
<td>
<p>size for the number of sequences</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_size_names">size_names</code></td>
<td>
<p>size for the names of the 2 samples</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_y_names">y_names</code></td>
<td>
<p>y position for the names of the 2 samples. If NA (default),
computed using the maximum abundances values.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_ylim_modif">ylim_modif</code></td>
<td>
<p>vector of two values. Modificator (by a multiplication)
of ylim. If one value is set, this value is used for both limits.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_nb_samples_info">nb_samples_info</code></td>
<td>
<p>(default: TRUE, logical) if TRUE and merge_sample_by is set,
add the number of samples merged for both levels.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_plotly_version">plotly_version</code></td>
<td>
<p>If TRUE, use <code><a href="plotly.html#topic+ggplotly">plotly::ggplotly()</a></code> to return
a interactive ggplot.</p>
</td></tr>
<tr><td><code id="biplot_pq_+3A_...">...</code></td>
<td>
<p>Other arguments for the ggplot function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_fungi_2Height &lt;- subset_samples(data_fungi_mini, Height %in% c("Low", "High"))
biplot_pq(data_fungi_2Height, "Height", merge_sample_by = "Height")
</code></pre>

<hr>
<h2 id='blast_pq'>Blast all sequence of <code>refseq</code> slot of a <code>phyloseq-class</code>
object against a custom database.</h2><span id='topic+blast_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blast_pq(
  physeq,
  fasta_for_db = NULL,
  database = NULL,
  blastpath = NULL,
  id_cut = 90,
  bit_score_cut = 50,
  min_cover_cut = 50,
  e_value_cut = 1e-30,
  unique_per_seq = FALSE,
  score_filter = TRUE,
  nproc = 1,
  args_makedb = NULL,
  args_blastn = NULL,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blast_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_fasta_for_db">fasta_for_db</code></td>
<td>
<p>path to a fasta file to make the blast database</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_database">database</code></td>
<td>
<p>path to a blast database</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_blastpath">blastpath</code></td>
<td>
<p>path to blast program</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_id_cut">id_cut</code></td>
<td>
<p>(default: 90) cut of in identity percent to keep result</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_bit_score_cut">bit_score_cut</code></td>
<td>
<p>(default: 50) cut of in bit score to keep result
The higher the bit-score, the better the sequence similarity.
The bit-score is the requires size of a sequence database in which the current
match could be found just by chance. The bit-score is a log2 scaled and
normalized raw-score. Each increase by one doubles the required database size
(2bit-score).</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_min_cover_cut">min_cover_cut</code></td>
<td>
<p>(default: 50) cut of in query cover (%) to keep result</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_e_value_cut">e_value_cut</code></td>
<td>
<p>(default: 1e-30)  cut of in e-value (%) to keep result
The BLAST E-value is the number of expected hits of similar quality (score)
that could be found just by chance.</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_unique_per_seq">unique_per_seq</code></td>
<td>
<p>(logical, default FALSE) if TRUE only return the better match
(higher <strong>bit score</strong>) for each sequence</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_score_filter">score_filter</code></td>
<td>
<p>(logical, default TRUE) does results are filter by score? If
FALSE, <code>id_cut</code>,<code>bit_score_cut</code>, <code>e_value_cut</code> and <code>min_cover_cut</code> are ignored</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_nproc">nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for blast (args -num_threads
for blastn command)</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_args_makedb">args_makedb</code></td>
<td>
<p>Additional parameters parse to makeblastdb command</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_args_blastn">args_blastn</code></td>
<td>
<p>Additional parameters parse to blastn command</p>
</td></tr>
<tr><td><code id="blast_pq_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files
</p>

<ul>
<li><p> db.fasta (refseq transformed into a database)
</p>
</li>
<li><p> dbase list of files (output of blastn)
</p>
</li>
<li><p> blast_result.txt the summary result of blastn using
<code style="white-space: pre;">&#8288;-outfmt "6 qseqid qlen sseqid slen length pident evalue bitscore qcovs"&#8288;</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>a blast table
</p>


<h3>See Also</h3>

<p><code><a href="#topic+blast_to_phyloseq">blast_to_phyloseq()</a></code> to use <code>refseq</code>
slot as a database
</p>

<hr>
<h2 id='blast_to_derep'>Blast some sequence against sequences from of a <code>derep-class</code>
object.</h2><span id='topic+blast_to_derep'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blast_to_derep(
  derep,
  seq2search,
  blastpath = NULL,
  id_cut = 90,
  bit_score_cut = 50,
  min_cover_cut = 50,
  e_value_cut = 1e-30,
  unique_per_seq = FALSE,
  score_filter = FALSE,
  list_no_output_query = FALSE,
  min_length_seq = 200,
  args_makedb = NULL,
  args_blastn = NULL,
  nproc = 1,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blast_to_derep_+3A_derep">derep</code></td>
<td>
<p>The result of <code>dada2::derepFastq()</code>. A list of <code>derep-class</code>
object.</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_seq2search">seq2search</code></td>
<td>
<p>(required) path to a fasta file defining the sequences
you want to blast against the ASV sequences from the physeq object.</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_blastpath">blastpath</code></td>
<td>
<p>path to blast program</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_id_cut">id_cut</code></td>
<td>
<p>(default: 90) cut of in identity percent to keep result</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_bit_score_cut">bit_score_cut</code></td>
<td>
<p>(default: 50) cut of in bit score to keep result
The higher the bit-score, the better the sequence similarity.
The bit-score is the requires size of a sequence database in which the current
match could be found just by chance. The bit-score is a log2 scaled and
normalized raw-score. Each increase by one doubles the required database size
(2bit-score).</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_min_cover_cut">min_cover_cut</code></td>
<td>
<p>(default: 50) cut of in query cover (%) to keep result</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_e_value_cut">e_value_cut</code></td>
<td>
<p>(default: 1e-30)  cut of in e-value (%) to keep result
The BLAST E-value is the number of expected hits of similar quality (score)
that could be found just by chance.</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_unique_per_seq">unique_per_seq</code></td>
<td>
<p>(logical, default FALSE) if TRUE only return the better match
(higher <strong>bit score</strong>) for each sequence</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_score_filter">score_filter</code></td>
<td>
<p>(logical, default TRUE) does results are filter by score? If
FALSE, <code>id_cut</code>,<code>bit_score_cut</code>, <code>e_value_cut</code> and <code>min_cover_cut</code> are ignored</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_list_no_output_query">list_no_output_query</code></td>
<td>
<p>(logical) does the result table include
query sequences for which <code>blastn</code> does not find any correspondence?</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_min_length_seq">min_length_seq</code></td>
<td>
<p>(default: 200) Removed sequences with less than
<code>min_length_seq</code> from derep before blast. Set to 0 to discard filtering
sequences by length.</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_args_makedb">args_makedb</code></td>
<td>
<p>Additional parameters parse to makeblastdb command</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_args_blastn">args_blastn</code></td>
<td>
<p>Additional parameters parse to blastn command</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_nproc">nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for blast (args -num_threads
for blastn command)</p>
</td></tr>
<tr><td><code id="blast_to_derep_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary
files :
</p>

<ul>
<li><p> db.fasta (refseq transformed into a database)
</p>
</li>
<li><p> dbase list of files (output of blastn)
</p>
</li>
<li><p> blast_result.txt the summary result of blastn using
<code style="white-space: pre;">&#8288;-outfmt "6 qseqid qlen sseqid slen length pident evalue bitscore qcovs"&#8288;</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>A blast table
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+blast_pq">blast_pq()</a></code> to use <code>refseq</code> slot as query sequences
against un custom database and <code><a href="#topic+blast_to_phyloseq">blast_to_phyloseq()</a></code>  to use
<code>refseq</code> slot as a database
</p>

<hr>
<h2 id='blast_to_phyloseq'>Blast some sequence against <code>refseq</code> slot of a <code>phyloseq-class</code>
object.</h2><span id='topic+blast_to_phyloseq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>blast_to_phyloseq(
  physeq,
  seq2search,
  blastpath = NULL,
  id_cut = 90,
  bit_score_cut = 50,
  min_cover_cut = 50,
  e_value_cut = 1e-30,
  unique_per_seq = FALSE,
  score_filter = TRUE,
  list_no_output_query = FALSE,
  args_makedb = NULL,
  args_blastn = NULL,
  nproc = 1,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="blast_to_phyloseq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_seq2search">seq2search</code></td>
<td>
<p>(required) path to a fasta file defining the sequences
you want to blast against the ASV sequences from the physeq object.</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_blastpath">blastpath</code></td>
<td>
<p>path to blast program</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_id_cut">id_cut</code></td>
<td>
<p>(default: 90) cut of in identity percent to keep result</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_bit_score_cut">bit_score_cut</code></td>
<td>
<p>(default: 50) cut of in bit score to keep result
The higher the bit-score, the better the sequence similarity.
The bit-score is the requires size of a sequence database in which the current
match could be found just by chance. The bit-score is a log2 scaled and
normalized raw-score. Each increase by one doubles the required database size
(2bit-score).</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_min_cover_cut">min_cover_cut</code></td>
<td>
<p>(default: 50) cut of in query cover (%) to keep result</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_e_value_cut">e_value_cut</code></td>
<td>
<p>(default: 1e-30)  cut of in e-value (%) to keep result
The BLAST E-value is the number of expected hits of similar quality (score)
that could be found just by chance.</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_unique_per_seq">unique_per_seq</code></td>
<td>
<p>(logical, default FALSE) if TRUE only return the better match
(higher <strong>bit score</strong>) for each sequence</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_score_filter">score_filter</code></td>
<td>
<p>(logical, default TRUE) does results are filter by score? If
FALSE, <code>id_cut</code>,<code>bit_score_cut</code>, <code>e_value_cut</code> and <code>min_cover_cut</code> are ignored</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_list_no_output_query">list_no_output_query</code></td>
<td>
<p>(logical) does the result table include
query sequences for which <code>blastn</code> does not find any correspondence?</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_args_makedb">args_makedb</code></td>
<td>
<p>Additional parameters parse to makeblastdb command</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_args_blastn">args_blastn</code></td>
<td>
<p>Additional parameters parse to blastn command</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_nproc">nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for blast (args -num_threads
for blastn command)</p>
</td></tr>
<tr><td><code id="blast_to_phyloseq_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files
</p>

<ul>
<li><p> db.fasta (refseq transformed into a database)
</p>
</li>
<li><p> dbase list of files (output of blastn)
</p>
</li>
<li><p> blast_result.txt the summary result of blastn using
<code style="white-space: pre;">&#8288;-outfmt "6 qseqid qlen sseqid slen length pident evalue bitscore qcovs"&#8288;</code>
</p>
</li></ul>
</td></tr>
</table>


<h3>Value</h3>

<p>the blast table
</p>


<h3>See Also</h3>

<p><code><a href="#topic+blast_pq">blast_pq()</a></code> to use <code>refseq</code> slot as query sequences
against un custom database.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 

blastpath &lt;- "...YOUR_PATH_TO_BLAST..."
blast_to_phyloseq(data_fungi,
  seq2search = system.file("extdata", "ex.fasta",
    package = "MiscMetabar", mustWork = TRUE
  ),
  blastpath = blastpath
)

## End(Not run)
</code></pre>

<hr>
<h2 id='build_phytree_pq'>Build phylogenetic trees from refseq slot of a phyloseq object</h2><span id='topic+build_phytree_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This function build tree phylogenetic tree and if nb_bootstrap is
set, it build also the 3 corresponding bootstrapped tree.
</p>
<p>Default parameters are based on <a href="https://doi.org/10.12688/f1000research.8986.2">doi:10.12688/f1000research.8986.2</a>
and phangorn vignette <a href="https://klausvigo.github.io/phangorn/articles/Trees.html">Estimating phylogenetic trees with phangorn</a>. You should understand your data, especially the markers,
before using this function.
</p>
<p>Note that phylogenetic reconstruction with markers used for metabarcoding are
not robust. You must verify the robustness of your phylogenetic tree using
taxonomic classification (see vignette <a href="https://adrientaudiere.github.io/MiscMetabar/articles/tree_visualization.html">Tree visualization</a>) and bootstrap or multi-tree visualization
</p>


<h3>Usage</h3>

<pre><code class='language-R'>build_phytree_pq(
  physeq,
  nb_bootstrap = 0,
  model = "GTR",
  optInv = TRUE,
  optGamma = TRUE,
  rearrangement = "NNI",
  control = phangorn::pml.control(trace = 0),
  optNni = TRUE,
  multicore = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="build_phytree_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_nb_bootstrap">nb_bootstrap</code></td>
<td>
<p>(default 0): If a positive number is set,
the function also build 3 bootstrapped trees using <code>nb_bootstrap</code>
bootstrap samples</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_model">model</code></td>
<td>
<p>allows to choose an amino acid models or nucleotide model,
see <code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_optinv">optInv</code></td>
<td>
<p>Logical value indicating whether topology gets optimized
(NNI). See <code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_optgamma">optGamma</code></td>
<td>
<p>Logical value indicating whether gamma rate parameter gets
optimized. See <code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_rearrangement">rearrangement</code></td>
<td>
<p>type of tree tree rearrangements to perform, one of
&quot;NNI&quot;, &quot;stochastic&quot; or &quot;ratchet&quot;
see <code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_control">control</code></td>
<td>
<p>A list of parameters for controlling the fitting process.
see <code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_optnni">optNni</code></td>
<td>
<p>Logical value indicating whether topology gets optimized (NNI).
see <code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_multicore">multicore</code></td>
<td>
<p>(logical) whether models should estimated in parallel.
see <code><a href="phangorn.html#topic+bootstrap.pml">phangorn::bootstrap.pml()</a></code> for more details</p>
</td></tr>
<tr><td><code id="build_phytree_pq_+3A_...">...</code></td>
<td>
<p>Other params for be passed on to
<code><a href="phangorn.html#topic+pml">phangorn::optim.pml()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>phangorn</code> package if you
use this function.
</p>


<h3>Value</h3>

<p>A list of phylogenetic tree
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (requireNamespace("phangorn")) {
  df &lt;- subset_taxa_pq(data_fungi_mini, taxa_sums(data_fungi_mini) &gt; 9000)
  df_tree &lt;- build_phytree_pq(df, nb_bootstrap = 2)
  plot(df_tree$UPGMA)
  phangorn::plotBS(df_tree$UPGMA, df_tree$UPGMA_bs, main = "UPGMA")
  plot(df_tree$NJ, "unrooted")
  plot(df_tree$ML)

  phangorn::plotBS(df_tree$ML$tree, df_tree$ML_bs, p = 20, frame = "circle")
  phangorn::plotBS(
    df_tree$ML$tree,
    df_tree$ML_bs,
    p = 20,
    frame = "circle",
    method = "TBE"
  )
  plot(phangorn::consensusNet(df_tree$ML_bs))
  plot(phangorn::consensusNet(df_tree$NJ_bs))
  ps_tree &lt;- merge_phyloseq(df, df_tree$ML$tree)
}


</code></pre>

<hr>
<h2 id='chimera_detection_vs'>Detect for chimera taxa using <a href="https://github.com/torognes/vsearch">vsearch</a></h2><span id='topic+chimera_detection_vs'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chimera_detection_vs(
  seq2search,
  nb_seq,
  vsearchpath = "vsearch",
  abskew = 2,
  min_seq_length = 100,
  vsearch_args = "--fasta_width 0",
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chimera_detection_vs_+3A_seq2search">seq2search</code></td>
<td>
<p>(required) a list of DNA sequences coercible by function
<code><a href="Biostrings.html#topic+XStringSet-class">Biostrings::DNAStringSet()</a></code></p>
</td></tr>
<tr><td><code id="chimera_detection_vs_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(required) a numeric vector giving the number of sequences for
each DNA sequences</p>
</td></tr>
<tr><td><code id="chimera_detection_vs_+3A_vsearchpath">vsearchpath</code></td>
<td>
<p>(default: vsearch) path to vsearch</p>
</td></tr>
<tr><td><code id="chimera_detection_vs_+3A_abskew">abskew</code></td>
<td>
<p>(int, default 2) The abundance skew is used to distinguish in a
three way alignment which sequence is the chimera and which are the
parents. The assumption is that chimeras appear later in the PCR
amplification process and are therefore less abundant than their parents.
The default value is 2.0, which means that the parents should be at least
2 times more abundant than their chimera. Any positive value equal or
greater than 1.0 can be used.</p>
</td></tr>
<tr><td><code id="chimera_detection_vs_+3A_min_seq_length">min_seq_length</code></td>
<td>
<p>(int, default 100)) Minimum length of sequences to
be part of the analysis</p>
</td></tr>
<tr><td><code id="chimera_detection_vs_+3A_vsearch_args">vsearch_args</code></td>
<td>
<p>(default &quot;&ndash;fasta_width 0&quot;) A list of other args for
vsearch command</p>
</td></tr>
<tr><td><code id="chimera_detection_vs_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary
files ?
</p>

<ul>
<li><p> non_chimeras.fasta
</p>
</li>
<li><p> chimeras.fasta
</p>
</li>
<li><p> borderline.fasta
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make <a href="https://github.com/torognes/vsearch">vsearch</a>.
</p>


<h3>Value</h3>

<p>A list of 3 including non-chimera taxa (<code style="white-space: pre;">&#8288;$non_chimera&#8288;</code>), chimera taxa
(<code style="white-space: pre;">&#8288;$chimera&#8288;</code>) and bordeline taxa (<code style="white-space: pre;">&#8288;$borderline&#8288;</code>)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

chimera_detection_vs(
  seq2search = data_fungi@refseq,
  nb_seq = taxa_sums(data_fungi)
)


</code></pre>

<hr>
<h2 id='chimera_removal_vs'>Search for a list of sequence in an object to remove chimera taxa
using <a href="https://github.com/torognes/vsearch">vsearch</a></h2><span id='topic+chimera_removal_vs'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chimera_removal_vs(object, type = "Discard_only_chim", clean_pq = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="chimera_removal_vs_+3A_object">object</code></td>
<td>
<p>(required) A phyloseq-class object or one of dada, derep,
data.frame or list coercible to sequences table using the
function <code><a href="dada2.html#topic+makeSequenceTable">dada2::makeSequenceTable()</a></code></p>
</td></tr>
<tr><td><code id="chimera_removal_vs_+3A_type">type</code></td>
<td>
<p>(default &quot;Discard_only_chim&quot;). The type define the type of
filtering.
</p>

<ul>
<li><p> &quot;Discard_only_chim&quot; will only discard taxa classify as chimera by vsearch
</p>
</li>
<li><p> &quot;Select_only_non_chim&quot; will only select taxa classify as non-chimera by
vsearch(after filtering taxa based on their sequence length by the
parameter <code>min_seq_length</code> from the <code><a href="#topic+chimera_detection_vs">chimera_detection_vs()</a></code> function)
</p>
</li>
<li><p> &quot;Select_only_chim&quot; will only select taxa classify as chimera by
vsearch (after filtering taxa based on their sequence length by the
parameter <code>min_seq_length</code> from the <code><a href="#topic+chimera_detection_vs">chimera_detection_vs()</a></code> function)
</p>
</li></ul>
</td></tr>
<tr><td><code id="chimera_removal_vs_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical; default FALSE) If TRUE, return the phyloseq object
after cleaning using the default parameter of <code><a href="#topic+clean_pq">clean_pq()</a></code> function.</p>
</td></tr>
<tr><td><code id="chimera_removal_vs_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="#topic+chimera_detection_vs">chimera_detection_vs()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make <a href="https://github.com/torognes/vsearch">vsearch</a>.
</p>


<h3>Value</h3>


<ul>
<li><p> I/ a sequences tables if object is of class dada, derep, data.frame or
list.
</p>
</li>
<li><p> II/ a phyloseq object without (or with if type = 'Select_only_chim')
chimeric taxa
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+chimera_detection_vs">chimera_detection_vs()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data_fungi_nochim &lt;- chimera_removal_vs(data_fungi)
data_fungi_nochim_16 &lt;- chimera_removal_vs(data_fungi,
  abskew = 16,
  min_seq_length = 10
)
data_fungi_nochim2 &lt;-
  chimera_removal_vs(data_fungi, type = "Select_only_non_chim")
data_fungi_chimera &lt;-
  chimera_removal_vs(data_fungi, type = "Select_only_chim")


</code></pre>

<hr>
<h2 id='circle_pq'>Plot OTU circle for <code>phyloseq-class</code> object</h2><span id='topic+circle_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>circle_pq(
  physeq = NULL,
  fact = NULL,
  taxa = "Order",
  nproc = 1,
  add_nb_seq = TRUE,
  rarefy = FALSE,
  min_prop_tax = 0.01,
  min_prop_mod = 0.1,
  gap_degree = NULL,
  start_degree = NULL,
  row_col = NULL,
  grid_col = NULL,
  log10trans = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="circle_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_taxa">taxa</code></td>
<td>
<p>(default: 'Order') Name of the taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_nproc">nproc</code></td>
<td>
<p>(default 1)
Set to number of cpus/processors to use for parallelization</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_add_nb_seq">add_nb_seq</code></td>
<td>
<p>(default: TRUE) Represent the number of sequences or the
number of OTUs (add_nb_seq = FALSE)</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_rarefy">rarefy</code></td>
<td>
<p>(logical) Does each samples modalities need to be rarefy in
order to compare them with the same amount of sequences?</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_min_prop_tax">min_prop_tax</code></td>
<td>
<p>(default: 0.01) The minimum proportion for taxon to be
plotted</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_min_prop_mod">min_prop_mod</code></td>
<td>
<p>(default: 0.1) The minimum proportion for modalities
to be plotted</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_gap_degree">gap_degree</code></td>
<td>
<p>Gap between two neighbour sectors.
It can be a single value or a vector. If it is a vector,
the first value corresponds to the gap after the first
sector.</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_start_degree">start_degree</code></td>
<td>
<p>The starting degree from which the circle begins to
draw. Note this degree is measured in the standard polar coordinate
which means it is always reverse-clockwise.</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_row_col">row_col</code></td>
<td>
<p>Color vector for row</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_grid_col">grid_col</code></td>
<td>
<p>Grid colors which correspond to sectors. The length of the
vector should be either 1 or the number of sectors.
It's preferred that grid_col is a named vector of which names
correspond to sectors. If it is not a named vector, the
order of grid_col corresponds to order of sectors.</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_log10trans">log10trans</code></td>
<td>
<p>(logical) Should sequence be log10 transformed
(more precisely by log10(1+x))?</p>
</td></tr>
<tr><td><code id="circle_pq_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="circlize.html#topic+chordDiagram">chordDiagram</a></code> or <code><a href="circlize.html#topic+circos.par">circos.par</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="circlize.html#topic+chordDiagram">chordDiagram</a></code> plot representing the
distribution of OTUs or sequences in the different modalities of the factor
fact
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="circlize.html#topic+chordDiagram">chordDiagram</a></code>
</p>
<p><code><a href="circlize.html#topic+circos.par">circos.par</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("pbapply")) {
  data("GlobalPatterns", package = "phyloseq")
  GP &lt;- subset_taxa(GlobalPatterns, GlobalPatterns@tax_table[, 1] == "Archaea")
  circle_pq(GP, "SampleType")
  circle_pq(GP, "SampleType", add_nb_seq = FALSE)
  circle_pq(GP, "SampleType", taxa = "Class")
}

</code></pre>

<hr>
<h2 id='clean_pq'>Clean phyloseq object by removing empty samples and taxa</h2><span id='topic+clean_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>In addition, this function check for discrepancy (and rename) between
(i) taxa names in refseq, taxonomy table and otu_table and between
(ii) sample names in sam_data and otu_table.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clean_pq(
  physeq,
  remove_empty_samples = TRUE,
  remove_empty_taxa = TRUE,
  clean_samples_names = TRUE,
  silent = FALSE,
  verbose = FALSE,
  force_taxa_as_columns = FALSE,
  force_taxa_as_rows = FALSE,
  reorder_asv = FALSE,
  rename_asv = FALSE,
  simplify_taxo = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="clean_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_remove_empty_samples">remove_empty_samples</code></td>
<td>
<p>(logical) Do you want to remove samples
without sequences (this is done after removing empty taxa)</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_remove_empty_taxa">remove_empty_taxa</code></td>
<td>
<p>(logical) Do you want to remove taxa
without sequences (this is done before removing empty samples)</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_clean_samples_names">clean_samples_names</code></td>
<td>
<p>(logical) Do you want to clean samples names?</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_silent">silent</code></td>
<td>
<p>(logical) If true, no message are printing.</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical) Additional informations in the message
the verbose parameter overwrite the silent parameter.</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_force_taxa_as_columns">force_taxa_as_columns</code></td>
<td>
<p>(logical) If true, if the taxa are rows
transpose the otu_table and set taxa_are_rows to false</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_force_taxa_as_rows">force_taxa_as_rows</code></td>
<td>
<p>(logical) If true, if the taxa are columns
transpose the otu_table and set taxa_are_rows to true</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_reorder_asv">reorder_asv</code></td>
<td>
<p>(logical) if TRUE the otu_table is ordered by the number of
sequences of ASV (descending order). Default to FALSE.</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_rename_asv">rename_asv</code></td>
<td>
<p>(logical) if TRUE, ASV are renamed by their position
in the OTU_table (asv_1, asv_2, ...). Default to FALSE. If rename ASV is true,
the ASV names in verbose information can be misleading.</p>
</td></tr>
<tr><td><code id="clean_pq_+3A_simplify_taxo">simplify_taxo</code></td>
<td>
<p>(logical) if TRUE, correct the taxonomy_table using the
<code>MiscMetabar::simplify_taxo()</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object
</p>

<hr>
<h2 id='compare_pairs_pq'>Compare samples in pairs using diversity and number of ASV including
shared ASV.</h2><span id='topic+compare_pairs_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a> #'   For the moment refseq slot need to be not Null.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compare_pairs_pq(
  physeq = NULL,
  bifactor = NULL,
  modality = NULL,
  merge_sample_by = NULL,
  nb_min_seq = 0,
  veg_index = "shannon",
  na_remove = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compare_pairs_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="compare_pairs_pq_+3A_bifactor">bifactor</code></td>
<td>
<p>(required) a factor (present in the <code>sam_data</code> slot of
the physeq object) presenting the pair names</p>
</td></tr>
<tr><td><code id="compare_pairs_pq_+3A_modality">modality</code></td>
<td>
<p>the name of the column in the <code>sam_data</code>
slot of the physeq object to split samples by pairs</p>
</td></tr>
<tr><td><code id="compare_pairs_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>a vector to determine
which samples to merge using the
<code><a href="#topic+merge_samples2">merge_samples2()</a></code> function.
Need to be in <code>physeq@sam_data</code></p>
</td></tr>
<tr><td><code id="compare_pairs_pq_+3A_nb_min_seq">nb_min_seq</code></td>
<td>
<p>minimum number of sequences per sample
to count the ASV/OTU</p>
</td></tr>
<tr><td><code id="compare_pairs_pq_+3A_veg_index">veg_index</code></td>
<td>
<p>(default: &quot;shannon&quot;) index for the <code>vegan::diversity</code> function</p>
</td></tr>
<tr><td><code id="compare_pairs_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default TRUE) If set to TRUE, remove samples with
NA in the variables set in bifactor, modality and merge_sample_by.
NA in variables are well managed even if na_remove = FALSE, so na_remove may
be useless.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with information about the number of shared ASV, shared number of sequences
and diversity
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_fungi_low_high &lt;- subset_samples(data_fungi, Height %in% c("Low", "High"))
compare_pairs_pq(data_fungi_low_high, bifactor = "Height", merge_sample_by = "Height")
compare_pairs_pq(data_fungi_low_high,
  bifactor = "Height",
  merge_sample_by = "Height", modality = "Time"
)
</code></pre>

<hr>
<h2 id='count_seq'>Count sequences in fasta or fastq file</h2><span id='topic+count_seq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
Use grep to count the number of line with only one '+' (fastq, fastq.gz)
or lines starting with a '&gt;' (fasta) to count sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>count_seq(file_path = NULL, folder_path = NULL, pattern = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="count_seq_+3A_file_path">file_path</code></td>
<td>
<p>The path to a  fasta, fastq or fastq.gz file</p>
</td></tr>
<tr><td><code id="count_seq_+3A_folder_path">folder_path</code></td>
<td>
<p>The path to a folder with fasta, fastq or fastq.gz files</p>
</td></tr>
<tr><td><code id="count_seq_+3A_pattern">pattern</code></td>
<td>
<p>A pattern to filter files in a folder. E.g. <em>R2</em></p>
</td></tr>
</table>


<h3>Value</h3>

<p>the number of sequences
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
count_seq(file_path = system.file(
  "extdata",
  "ex.fasta",
  package = "MiscMetabar",
  mustWork = TRUE
))
count_seq(
  folder_path = system.file("extdata", package = "MiscMetabar"),
  pattern = "*.fasta"
)

</code></pre>

<hr>
<h2 id='cutadapt_remove_primers'>Remove primers using <a href="https://github.com/marcelm/cutadapt/">cutadapt</a></h2><span id='topic+cutadapt_remove_primers'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>You need to install Cutadapt
</p>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cutadapt_remove_primers(
  path_to_fastq,
  primer_fw = NULL,
  primer_rev = NULL,
  folder_output = "wo_primers",
  nproc = 1,
  pattern = "fastq.gz",
  pattern_R1 = "_R1",
  pattern_R2 = "_R2",
  nb_files = Inf,
  cmd_is_run = TRUE,
  args_before_cutadapt =
    "source ~/miniconda3/etc/profile.d/conda.sh &amp;&amp; conda activate cutadaptenv &amp;&amp; "
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cutadapt_remove_primers_+3A_path_to_fastq">path_to_fastq</code></td>
<td>
<p>(Required) A path to a folder with fastq files. See
<code><a href="#topic+list_fastq_files">list_fastq_files()</a></code> for help.</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_primer_fw">primer_fw</code></td>
<td>
<p>(Required, String) The forward primer DNA sequence.</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_primer_rev">primer_rev</code></td>
<td>
<p>(String)  The reverse primer DNA sequence.</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_folder_output">folder_output</code></td>
<td>
<p>The path to a folder for output files</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_nproc">nproc</code></td>
<td>
<p>(default 1)
Set to number of cpus/processors to use for the clustering</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_pattern">pattern</code></td>
<td>
<p>a pattern to filter files (passed on to list.files function).</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_pattern_r1">pattern_R1</code></td>
<td>
<p>a pattern to filter R1 files (default &quot;<em>R1</em>&quot;)</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_pattern_r2">pattern_R2</code></td>
<td>
<p>a pattern to filter R2 files (default &quot;<em>R2</em>&quot;)</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_nb_files">nb_files</code></td>
<td>
<p>the number of fastq files to list (default FALSE)</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_cmd_is_run">cmd_is_run</code></td>
<td>
<p>(logical, default TRUE) Do the cutadapt command is run.
If set to FALSE, the only effect of the function is to return a list of
command to manually run in a terminal.</p>
</td></tr>
<tr><td><code id="cutadapt_remove_primers_+3A_args_before_cutadapt">args_before_cutadapt</code></td>
<td>
<p>(String) A one line bash command to run before
to run cutadapt. For examples, &quot;source ~/miniconda3/etc/profile.d/conda.sh &amp;&amp; conda activate cutadaptenv &amp;&amp;&quot; allow to bypass the conda init which asks to restart the shell</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please cite cutadapt (<a href="https://doi.org/10.14806/ej.17.1.200">doi:10.14806/ej.17.1.200</a>).
</p>


<h3>Value</h3>

<p>a list of command and
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
cutadapt_remove_primers("inst/extdata", "TTC", "GAA",
  folder_output = tempdir()
)


cutadapt_remove_primers(
  system.file("extdata",
    package = "dada2"
  ),
  pattern_R1 = "F.fastq.gz",
  pattern_R2 = "R.fastq.gz",
  primer_fw = "TTC",
  primer_rev = "GAA",
  folder_output = tempdir()
)

cutadapt_remove_primers(
  system.file("extdata",
    package = "dada2"
  ),
  pattern_R1 = "F.fastq.gz",
  primer_fw = "TTC",
  folder_output = tempdir(),
  cmd_is_run = FALSE
)


unlink(tempdir(), recursive = TRUE)

## End(Not run)

</code></pre>

<hr>
<h2 id='data_fungi'>Fungal OTU in phyloseq format</h2><span id='topic+data_fungi'></span>

<h3>Description</h3>

<p>Fungal OTU in phyloseq format
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_fungi)
</code></pre>


<h3>Format</h3>

<p>A physeq object containing 1420 taxa with references sequences
described by 14 taxonomic ranks and 185 samples described by 7 sample variables:
</p>

<ul>
<li> <p><em>X</em>: the name of the fastq-file
</p>
</li>
<li> <p><em>Sample_names</em>: the names of ... the samples
</p>
</li>
<li> <p><em>Treename</em>: the name of an tree
</p>
</li>
<li> <p><em>Sample_id</em>: identifier for each sample
</p>
</li>
<li> <p><em>Height</em>: height of the sample in the tree
</p>
</li>
<li> <p><em>Diameter</em>: diameter of the trunk
</p>
</li>
<li> <p><em>Time</em>: time since the dead of the tree
</p>
</li></ul>


<hr>
<h2 id='data_fungi_mini'>Fungal OTU in phyloseq format</h2><span id='topic+data_fungi_mini'></span>

<h3>Description</h3>

<p>It is a subset of the data_fungi dataset including only Basidiomycota
with more than 5000 sequences.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_fungi_mini)

data(data_fungi_mini)
</code></pre>


<h3>Format</h3>

<p>A physeq object containing 45 taxa with references sequences
described by 14 taxonomic ranks and 137 samples described by 7 sample variables:
</p>

<ul>
<li> <p><em>X</em>: the name of the fastq-file
</p>
</li>
<li> <p><em>Sample_names</em>: the names of ... the samples
</p>
</li>
<li> <p><em>Treename</em>: the name of an tree
</p>
</li>
<li> <p><em>Sample_id</em>: identifier for each sample
</p>
</li>
<li> <p><em>Height</em>: height of the sample in the tree
</p>
</li>
<li> <p><em>Diameter</em>: diameter of the trunk
</p>
</li>
<li> <p><em>Time</em>: time since the dead of the tree
</p>
</li></ul>

<p>A physeq object containing 45 taxa with references sequences
described by 14 taxonomic ranks and 137 samples described by 7 sample variables:
</p>

<ul>
<li> <p><em>X</em>: the name of the fastq-file
</p>
</li>
<li> <p><em>Sample_names</em>: the names of ... the samples
</p>
</li>
<li> <p><em>Treename</em>: the name of an tree
</p>
</li>
<li> <p><em>Sample_id</em>: identifier for each sample
</p>
</li>
<li> <p><em>Height</em>: height of the sample in the tree
</p>
</li>
<li> <p><em>Diameter</em>: diameter of the trunk
</p>
</li>
<li> <p><em>Time</em>: time since the dead of the tree
</p>
</li></ul>



<h3>Details</h3>

<p>Obtain using <code>data_fungi_mini &lt;- subset_taxa(data_fungi, Phylum == "Basidiomycota")</code>
and then <code>data_fungi_mini &lt;-   subset_taxa_pq(data_fungi_mini, colSums(data_fungi_mini@otu_table) &gt; 5000)</code>
</p>

<hr>
<h2 id='data_fungi_sp_known'>Fungal OTU in phyloseq format</h2><span id='topic+data_fungi_sp_known'></span>

<h3>Description</h3>

<p>It is a subset of the data_fungi dataset including only taxa with information
at the species level
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(data_fungi_sp_known)
</code></pre>


<h3>Format</h3>

<p>A physeq object containing 651 taxa with references sequences
described by 14 taxonomic ranks and 185 samples described by 7 sample variables:
</p>

<ul>
<li> <p><em>X</em>: the name of the fastq-file
</p>
</li>
<li> <p><em>Sample_names</em>: the names of ... the samples
</p>
</li>
<li> <p><em>Treename</em>: the name of an tree
</p>
</li>
<li> <p><em>Sample_id</em>: identifier for each sample
</p>
</li>
<li> <p><em>Height</em>: height of the sample in the tree
</p>
</li>
<li> <p><em>Diameter</em>: diameter of the trunk
</p>
</li>
<li> <p><em>Time</em>: time since the dead of the tree
</p>
</li></ul>



<h3>Details</h3>

<p>Obtain using <code>data_fungi_sp_known &lt;- subset_taxa(data_fungi, !is.na(data_fungi@tax_table[,"Species"]))</code>
</p>

<hr>
<h2 id='diff_fct_diff_class'>Compute different functions for different class of vector.</h2><span id='topic+diff_fct_diff_class'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
Mainly an internal function useful in &quot;sapply(..., tapply)&quot; methods
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diff_fct_diff_class(
  x,
  numeric_fonction = mean,
  logical_method = "TRUE_if_one",
  character_method = "unique_or_na",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="diff_fct_diff_class_+3A_x">x</code></td>
<td>
<p>: a vector</p>
</td></tr>
<tr><td><code id="diff_fct_diff_class_+3A_numeric_fonction">numeric_fonction</code></td>
<td>
<p>: a function for numeric vector. For ex. <code>sum</code> or <code>mean</code></p>
</td></tr>
<tr><td><code id="diff_fct_diff_class_+3A_logical_method">logical_method</code></td>
<td>
<p>: A method for logical vector. One of :
</p>

<ul>
<li><p> TRUE_if_one (default)
</p>
</li>
<li><p> NA_if_not_all_TRUE
</p>
</li>
<li><p> FALSE_if_not_all_TRUE
</p>
</li></ul>
</td></tr>
<tr><td><code id="diff_fct_diff_class_+3A_character_method">character_method</code></td>
<td>
<p>: A method for character vector (and factor). One of :
</p>

<ul>
<li><p> unique_or_na (default)
</p>
</li>
<li><p> more_frequent
</p>
</li>
<li><p> more_frequent_without_equality
</p>
</li></ul>
</td></tr>
<tr><td><code id="diff_fct_diff_class_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to the numeric function (ex. na.rm=TRUE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a single value
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
diff_fct_diff_class(
  data_fungi@sam_data$Sample_id,
  numeric_fonction = sum,
  na.rm = TRUE
)
diff_fct_diff_class(
  data_fungi@sam_data$Time,
  numeric_fonction = mean,
  na.rm = TRUE
)
diff_fct_diff_class(
  data_fungi@sam_data$Height == "Low",
  logical_method = "TRUE_if_one"
)
diff_fct_diff_class(
  data_fungi@sam_data$Height == "Low",
  logical_method = "NA_if_not_all_TRUE"
)
diff_fct_diff_class(
  data_fungi@sam_data$Height == "Low",
  logical_method = "FALSE_if_not_all_TRUE"
)
diff_fct_diff_class(
  data_fungi@sam_data$Height,
  character_method = "unique_or_na"
)
diff_fct_diff_class(
  c("IE", "IE"),
  character_method = "unique_or_na"
)
diff_fct_diff_class(
  c("IE", "IE", "TE", "TE"),
  character_method = "more_frequent"
)
diff_fct_diff_class(
  c("IE", "IE", "TE", "TE"),
  character_method = "more_frequent_without_equality"
)
</code></pre>

<hr>
<h2 id='dist_bycol'>Compute paired distances among matrix (e.g. otu_table)</h2><span id='topic+dist_bycol'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_bycol(x, y, method = "bray", nperm = 99, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_bycol_+3A_x">x</code></td>
<td>
<p>(required) A first matrix.</p>
</td></tr>
<tr><td><code id="dist_bycol_+3A_y">y</code></td>
<td>
<p>(required) A second matrix.</p>
</td></tr>
<tr><td><code id="dist_bycol_+3A_method">method</code></td>
<td>
<p>(default: 'bray') the method to use internally in the vegdist
function.</p>
</td></tr>
<tr><td><code id="dist_bycol_+3A_nperm">nperm</code></td>
<td>
<p>(int) The number of permutations to perform.</p>
</td></tr>
<tr><td><code id="dist_bycol_+3A_...">...</code></td>
<td>
<p>Others argument for <code>vegan::vegdist</code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of length two : (i) a vector of observed distance ($obs) and
(ii) a matrix of the distance after randomization ($null)
</p>


<h3>Note</h3>

<p>the first column of the first matrix is compare to the first column of
the second matrix, the second column of the first matrix is compare to the
second column of the second matrix and so on.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="vegan.html#topic+vegdist">vegdist</a></code>
</p>

<hr>
<h2 id='dist_pos_control'>Calculate ecological distance among positive controls vs
distance for all samples</h2><span id='topic+dist_pos_control'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Compute distance among positive controls,
i.e. samples which are duplicated
to test for variation, for example in
(i) a step in the sampling,
(ii) a step in the extraction,
(iii) a step in the sequencing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist_pos_control(physeq, samples_names, method = "bray")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="dist_pos_control_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="dist_pos_control_+3A_samples_names">samples_names</code></td>
<td>
<p>(required) a vector of names for samples with
positives controls of the same samples having the same name</p>
</td></tr>
<tr><td><code id="dist_pos_control_+3A_method">method</code></td>
<td>
<p>(default: &quot;bray&quot;) a method to calculate
the distance, parsed to <code><a href="vegan.html#topic+vegdist">vegan::vegdist()</a></code>. See ?vegdist for a list
of possible values.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of two data-frames with
(i) the distance among positive controls and
(ii) the distance among all samples
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("enterotype")
sam_name_factice &lt;- gsub("TS1_V2", "TS10_V2", sample_names(enterotype))
res_dist_cont &lt;- dist_pos_control(enterotype, sam_name_factice)
hist(unlist(res_dist_cont$distAllSamples))
abline(
  v = mean(unlist(res_dist_cont$dist_controlontrolSamples), na.rm = TRUE),
  col = "red", lwd = 3
)
</code></pre>

<hr>
<h2 id='distri_1_taxa'>Distribution of sequences across a factor for one taxon</h2><span id='topic+distri_1_taxa'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>distri_1_taxa(physeq, fact, taxa_name, digits = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="distri_1_taxa_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="distri_1_taxa_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot
different lines</p>
</td></tr>
<tr><td><code id="distri_1_taxa_+3A_taxa_name">taxa_name</code></td>
<td>
<p>(required): the name of the taxa</p>
</td></tr>
<tr><td><code id="distri_1_taxa_+3A_digits">digits</code></td>
<td>
<p>(default = 2) integer indicating the number of decimal places
to be used (see <code>?round</code> for more information)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a dataframe with levels as rows and information as column :
</p>

<ul>
<li><p> the number of sequences of the taxa (nb_seq)
</p>
</li>
<li><p> the number of samples of the taxa (nb_samp)
</p>
</li>
<li><p> the mean (mean_nb_seq) and standard deviation (sd_nb_seq) of the <em>nb_seq</em>
</p>
</li>
<li><p> the mean (mean_nb_seq_when_present) <em>nb_seq</em> excluding samples with zero
</p>
</li>
<li><p> the total number of samples (nb_total_samp)
</p>
</li>
<li><p> the proportion of samples with the taxa
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>distri_1_taxa(data_fungi, "Height", "ASV2")
distri_1_taxa(data_fungi, "Time", "ASV81", digits = 1)
</code></pre>

<hr>
<h2 id='fac2col'>Translates a factor into colors.</h2><span id='topic+fac2col'></span>

<h3>Description</h3>

<p>Translates a factor into colors.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fac2col(x, col.pal = funky_color, na.col = "grey", seed = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fac2col_+3A_x">x</code></td>
<td>
<p>a numeric vector (for num2col) or a vector converted to a factor (for fac2col).</p>
</td></tr>
<tr><td><code id="fac2col_+3A_col.pal">col.pal</code></td>
<td>
<p>(default funky_color)    a function generating colors according to a given palette.</p>
</td></tr>
<tr><td><code id="fac2col_+3A_na.col">na.col</code></td>
<td>
<p>(default grey) the color to be used for missing values (NAs)</p>
</td></tr>
<tr><td><code id="fac2col_+3A_seed">seed</code></td>
<td>
<p>(default NULL) a seed for R's random number generated, used to fix the random permutation of colors in the palette used; if NULL, no randomization is used and the colors are taken from the palette according to the ordering of the levels</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a color vector
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart in <code>adegenet</code> package
</p>


<h3>See Also</h3>

<p>The R package RColorBrewer, proposing a nice selection of color palettes. The viridis package, with many excellent palettes
</p>

<hr>
<h2 id='filter_asv_blast'>Filter undesirable taxa using blast against a custom database.</h2><span id='topic+filter_asv_blast'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_asv_blast(
  physeq,
  fasta_for_db = NULL,
  database = NULL,
  clean_pq = TRUE,
  add_info_to_taxtable = TRUE,
  id_filter = 90,
  bit_score_filter = 50,
  min_cover_filter = 50,
  e_value_filter = 1e-30,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_asv_blast_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_fasta_for_db">fasta_for_db</code></td>
<td>
<p>path to a fasta file to make the blast database</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_database">database</code></td>
<td>
<p>path to a blast database</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical)
If set to TRUE, empty samples and empty ASV are discarded
after filtering.</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_add_info_to_taxtable">add_info_to_taxtable</code></td>
<td>
<p>(logical, default TRUE) Does the blast information
are added to the taxtable ?</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_id_filter">id_filter</code></td>
<td>
<p>(default: 90) cut of in identity percent to keep result</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_bit_score_filter">bit_score_filter</code></td>
<td>
<p>(default: 50) cut of in bit score to keep result
The higher the bit-score, the better the sequence similarity.
The bit-score is the requires size of a sequence database in which the current
match could be found just by chance. The bit-score is a log2 scaled and
normalized raw-score. Each increase by one doubles the required database size
(2bit-score).</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_min_cover_filter">min_cover_filter</code></td>
<td>
<p>(default: 50) cut of in query cover (%) to keep result</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_e_value_filter">e_value_filter</code></td>
<td>
<p>(default: 1e-30)  cut of in e-value (%) to keep result
The BLAST E-value is the number of expected hits of similar quality (score)
that could be found just by chance.</p>
</td></tr>
<tr><td><code id="filter_asv_blast_+3A_...">...</code></td>
<td>
<p>Others options for the <code>blast_pq()</code> function. See <code>?blast_pq</code>.
Note that params <code>unique_per_seq</code> must be lft to TRUE and <code>score_filter</code>
must be left to FALSE.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object.
</p>

<hr>
<h2 id='filter_trim'>A wrapper of the function <code><a href="dada2.html#topic+filterAndTrim">dada2::filterAndTrim()</a></code> to use in
<a href="https://books.ropensci.org/targets/">targets</a> pipeline</h2><span id='topic+filter_trim'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>This function filter and trim (with parameters passed on to
<code><a href="dada2.html#topic+filterAndTrim">dada2::filterAndTrim()</a></code> function) forward sequences or paired end
sequence if 'rev' parameter is set. It return the list of files to
subsequent analysis in a targets pipeline.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_trim(
  fw = NULL,
  rev = NULL,
  output_fw = paste(getwd(), "/output/filterAndTrim_fwd", sep = ""),
  output_rev = paste(getwd(), "/output/filterAndTrim_rev", sep = ""),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filter_trim_+3A_fw">fw</code></td>
<td>
<p>(required) a list of forward fastq files</p>
</td></tr>
<tr><td><code id="filter_trim_+3A_rev">rev</code></td>
<td>
<p>a list of reverse fastq files for paired end trimming</p>
</td></tr>
<tr><td><code id="filter_trim_+3A_output_fw">output_fw</code></td>
<td>
<p>Path to output folder for forward files. By default,
this function will create
a folder &quot;output/filterAndTrim_fwd&quot; in the current working directory.</p>
</td></tr>
<tr><td><code id="filter_trim_+3A_output_rev">output_rev</code></td>
<td>
<p>Path to output folder for reverse files. By default,
this function will create
a folder &quot;output/filterAndTrim_fwd&quot; in the current working directory.</p>
</td></tr>
<tr><td><code id="filter_trim_+3A_...">...</code></td>
<td>
<p>Other parameters passed on to <code><a href="dada2.html#topic+filterAndTrim">dada2::filterAndTrim()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of files. If rev is set, will return a list of two lists.
The first list is a list of forward files, and the second one
is a list of reverse files.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="dada2.html#topic+filterAndTrim">dada2::filterAndTrim()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>testFastqs_fw &lt;- c(
  system.file("extdata", "sam1F.fastq.gz", package = "dada2"),
  system.file("extdata", "sam2F.fastq.gz", package = "dada2")
)
testFastqs_rev &lt;- c(
  system.file("extdata", "sam1R.fastq.gz", package = "dada2"),
  system.file("extdata", "sam2R.fastq.gz", package = "dada2")
)

filt_fastq_fw &lt;- filter_trim(testFastqs_fw, output_fw = tempdir())
derep_fw &lt;- derepFastq(filt_fastq_fw[1])
derep_fw

filt_fastq_pe &lt;- filter_trim(testFastqs_fw,
  testFastqs_rev,
  output_fw = tempdir("fw"),
  output_rev = tempdir("rev")
)
derep_fw_pe &lt;- derepFastq(filt_fastq_pe[[1]])
derep_rv_pe &lt;- derepFastq(filt_fastq_pe[[2]])
derep_fw_pe
derep_rv_pe
</code></pre>

<hr>
<h2 id='formattable_pq'>Create an visualization table to describe taxa distribution across a modality</h2><span id='topic+formattable_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>formattable_pq(
  physeq,
  modality,
  taxonomic_levels = c("Phylum", "Order", "Family", "Genus"),
  min_nb_seq_taxa = 1000,
  log10trans = FALSE,
  void_style = FALSE,
  lev_col_taxa = "Phylum",
  arrange_by = "nb_seq",
  descending_order = TRUE,
  na_remove = TRUE,
  formattable_args = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="formattable_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_modality">modality</code></td>
<td>
<p>(required) The name of a column present in the <code style="white-space: pre;">&#8288;@sam_data&#8288;</code> slot
of the physeq object. Must be a character vector or a factor.</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_taxonomic_levels">taxonomic_levels</code></td>
<td>
<p>(default = c(&quot;Phylum&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;))
The taxonomic levels (must be present in the <code style="white-space: pre;">&#8288;@sam_data&#8288;</code> slot) you want to
see and/or used (for example to compute a color) in the table.</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_min_nb_seq_taxa">min_nb_seq_taxa</code></td>
<td>
<p>(default = 1000) filter out taxa with less than <code>min_nb_seq_taxa</code>
sequences</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_log10trans">log10trans</code></td>
<td>
<p>(logical, default TRUE) Do sequences count is log10 transformed
(using log10(x + 1) to allow 0)</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_void_style">void_style</code></td>
<td>
<p>(logical, default FALSE) Do the default style is discard ?</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_lev_col_taxa">lev_col_taxa</code></td>
<td>
<p>Taxonomic level used to plot the background color of taxa names</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_arrange_by">arrange_by</code></td>
<td>
<p>The column used to sort the table.
Can take the values NULL, &quot;proportion_samp&quot;, &quot;nb_seq&quot; (default), , &quot;nb_sam&quot; &quot;OTU&quot;, or a column names
from the levels of modality or from taxonomic levels</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_descending_order">descending_order</code></td>
<td>
<p>(logical, default TRUE) Do we use descending order when sort the table
(if arrange_by is not NULL) ?</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default TRUE) if TRUE remove all the samples
with NA in the <code>split_by</code> variable of the <code>physeq@sam_data</code> slot</p>
</td></tr>
<tr><td><code id="formattable_pq_+3A_formattable_args">formattable_args</code></td>
<td>
<p>Other args to the formattable function. See examples and <code>formattable::formattable()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>formattable::formattable()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A datatable
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code>formattable::formattable()</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("formattable")) {
  ## Distribution of the nb of sequences per OTU across Height
  ##   modality (nb of sequences are log-transformed).
  ## Only OTU with more than 10000 sequences are taking into account
  ## The Phylum column is discarded
  formattable_pq(
    data_fungi,
    "Height",
    min_nb_seq_taxa = 10000,
    formattable_args = list("Phylum" = FALSE),
    log10trans = TRUE
  )

  ## Distribution of the nb of samples per OTU across Height modality
  ## Only OTU  present in more than 50 samples are taking into account
  formattable_pq(
    as_binary_otu_table(data_fungi),
    "Height",
    min_nb_seq_taxa = 50,
    formattable_args = list("nb_seq" = FALSE),
  )

  ## Distribution of the nb of sequences per OTU across Time modality
  ##  arranged by Family Name in ascending order.
  ## Only OTU with more than 10000 sequences are taking into account
  ## The Phylum column is discarded
  formattable_pq(
    data_fungi,
    "Time",
    min_nb_seq_taxa = 10000,
    taxonomic_levels = c("Order", "Family", "Genus", "Species"),
    formattable_args = list(
      Order = FALSE,
      Species = formattable::formatter(
        "span",
        style = x ~ formattable::style(
          "font-style" = "italic",
          `color` = ifelse(is.na(x), "white", "grey")
        )
      )
    ),
    arrange_by = "Family",
    descending_order = FALSE
  )
}

if (requireNamespace("formattable")) {
  ## Distribution of the nb of sequences per OTU across Height modality
  ##  (nb of sequences are log-transformed).
  ## OTU name background is light gray for Basidiomycota
  ##  and dark grey otherwise (Ascomycota)
  ## A different color is defined for each modality level
  formattable_pq(
    data_fungi,
    "Height",
    taxonomic_levels = c("Phylum", "Family", "Genus"),
    void_style = TRUE,
    formattable_args = list(
      OTU = formattable::formatter(
        "span",
        style = ~ formattable::style(
          "display" = "block",
          `border-radius` = "5px",
          `background-color` = ifelse(Phylum == "Basidiomycota", transp("gray"), "gray")
        ),
        `padding-right` = "2px"
      ),
      High = formattable::formatter(
        "span",
        style = x ~ formattable::style(
          "font-size" = "80%",
          "display" = "inline-block",
          direction = "rtl",
          `border-radius` = "0px",
          `padding-right` = "2px",
          `background-color` = formattable::csscolor(formattable::gradient(
            as.numeric(x), transp("#1a91ff"), "#1a91ff"
          )),
          width = formattable::percent(formattable::proportion(as.numeric(x), na.rm = TRUE))
        )
      ),
      Low = formattable::formatter(
        "span",
        style = x ~ formattable::style(
          "font-size" = "80%",
          "display" = "inline-block",
          direction = "rtl",
          `border-radius` = "0px",
          `padding-right` = "2px",
          `background-color` = formattable::csscolor(formattable::gradient(
            as.numeric(x),
            transp("green"), "green"
          )),
          width = formattable::percent(formattable::proportion(as.numeric(x), na.rm = TRUE))
        )
      ),
      Middle = formattable::formatter(
        "span",
        style = x ~ formattable::style(
          "font-size" = "80%",
          "display" = "inline-block",
          direction = "rtl",
          `border-radius` = "0px",
          `padding-right` = "2px",
          `background-color` = formattable::csscolor(formattable::gradient(
            as.numeric(x), transp("orange"), "orange"
          )),
          width = formattable::percent(formattable::proportion(as.numeric(x), na.rm = TRUE))
        )
      )
    )
  )
}

</code></pre>

<hr>
<h2 id='funguild_assign'>Assign Guilds to Organisms Based on Taxonomic Classification</h2><span id='topic+funguild_assign'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>The original function and documentation was written by Brendan Furneaux
in the <a href="https://github.com/brendanf/FUNGuildR/">FUNGuildR</a> package.
</p>
<p>These functions have identical behavior if supplied with a database; however
they download the database corresponding to their name by default.
</p>
<p>Taxa present in the database are matched to the taxa present in the supplied
<code>otu_table</code> by exact name.
In the case of multiple matches, the lowest (most specific) rank is chosen.
No attempt is made to check or correct the classification in
<code>otu_table$Taxonomy</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>funguild_assign(
  otu_table,
  db_funguild = get_funguild_db(),
  tax_col = "Taxonomy"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="funguild_assign_+3A_otu_table">otu_table</code></td>
<td>
<p>A <code>data.frame</code> with a <code>character</code>
column named &quot;<code>Taxonomy</code>&quot; (or another name as specified in
<code>tax_col</code>), as well as any other columns.
Each entry in &quot;<code>otu_table$Taxonomy</code>&quot; should be a comma-, colon-,
underscore-, or semicolon-delimited classification of an organism.
Rank indicators as given by Sintax (&quot;<code style="white-space: pre;">&#8288;k:&#8288;</code>&quot;, &quot;<code style="white-space: pre;">&#8288;p:&#8288;</code>&quot;...) or Unite (&quot;<code>k__</code>,
&quot;<code>p__</code>&quot;, ...) are also allowed.
A <code>character</code> vector, representing only the taxonomic classification,
is also accepted.</p>
</td></tr>
<tr><td><code id="funguild_assign_+3A_db_funguild">db_funguild</code></td>
<td>
<p>A <code>data.frame</code> representing the FUNGuild as returned by
<code><a href="#topic+get_funguild_db">get_funguild_db()</a></code>
If not supplied, the default database will be downloaded.</p>
</td></tr>
<tr><td><code id="funguild_assign_+3A_tax_col">tax_col</code></td>
<td>
<p>A <code>character</code> string, optionally giving an alternate
column name in <code>otu_table</code> to use instead of <code>otu_table$Taxonomy</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="tibble.html#topic+tibble">tibble::tibble</a></code> containing all columns of
<code>otu_table</code>, plus relevant columns of information from the FUNGuild
</p>


<h3>Author(s)</h3>

<p>Brendan Furneaux (orcid: <a href="https://orcid.org/0000-0003-3522-7363">0000-0003-3522-7363</a>),
modified by Adrien Taudière
</p>


<h3>References</h3>

<p>Nguyen NH, Song Z, Bates ST, Branco S, Tedersoo L, Menke J,
Schilling JS, Kennedy PG. 2016. <em>FUNGuild: An open annotation tool for
parsing fungal community datasets by ecological guild</em>. Fungal Ecology
20:241-248.
</p>

<hr>
<h2 id='funky_color'>Funky palette color</h2><span id='topic+funky_color'></span>

<h3>Description</h3>

<p>Funky palette color
</p>


<h3>Usage</h3>

<pre><code class='language-R'>funky_color(n)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="funky_color_+3A_n">n</code></td>
<td>
<p>a number of colors</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a color palette
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart in <code>adegenet</code> package
</p>


<h3>See Also</h3>

<p>The R package RColorBrewer, proposing a nice selection of color palettes. The viridis package, with many excellent palettes
</p>

<hr>
<h2 id='get_file_extension'>Get the extension of a file</h2><span id='topic+get_file_extension'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_file_extension(file_path)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_file_extension_+3A_file_path">file_path</code></td>
<td>
<p>(required): path to a file</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A  <code>phyloseq-class</code> object with simplified taxonomy
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='get_funguild_db'>Retrieve the FUNGuild database</h2><span id='topic+get_funguild_db'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
The original function and documentation was written by Brendan Furneaux
in the <a href="https://github.com/brendanf/FUNGuildR/">FUNGuildR</a> package.
</p>
<p>Please cite this <a href="https://www.sciencedirect.com/science/article/abs/pii/S1754504815000847">publication</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_funguild_db(db_url = "http://www.stbates.org/funguild_db_2.php")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_funguild_db_+3A_db_url">db_url</code></td>
<td>
<p>a length 1 character string giving the URL to retrieve the database
from</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code><a href="tibble.html#topic+tibble">tibble::tibble</a></code> containing the database, which can be passed
to the <code>db</code> argument of <code><a href="#topic+funguild_assign">funguild_assign()</a></code>
</p>


<h3>Author(s)</h3>

<p>Brendan Furneaux (orcid: <a href="https://orcid.org/0000-0003-3522-7363">0000-0003-3522-7363</a>),
modified by Adrien Taudière
</p>


<h3>References</h3>

<p>Nguyen NH, Song Z, Bates ST, Branco S, Tedersoo L, Menke J,
Schilling JS, Kennedy PG. 2016. <em>FUNGuild: An open annotation tool for
parsing fungal community datasets by ecological guild</em>. Fungal Ecology
20:241-248.
</p>

<hr>
<h2 id='ggaluv_pq'>Alluvial plot for taxonomy and samples factor vizualisation</h2><span id='topic+ggaluv_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Basically a wrapper of <a href="https://corybrunson.github.io/ggalluvial/index.html">ggalluvial</a>
package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggaluv_pq(
  physeq,
  taxa_ranks = c("Phylum", "Class", "Order", "Family"),
  wrap_factor = NULL,
  by_sample = FALSE,
  rarefy_by_sample = FALSE,
  fact = NULL,
  type = "nb_seq",
  width = 1.2,
  min.size = 3,
  na_remove = FALSE,
  use_ggfittext = FALSE,
  use_geom_label = FALSE,
  size_lab = 2,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggaluv_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_taxa_ranks">taxa_ranks</code></td>
<td>
<p>A vector of taxonomic ranks. For examples c(&quot;Family&quot;,&quot;Genus&quot;).
If taxa ranks is not set
(default value = c(&quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;)).</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_wrap_factor">wrap_factor</code></td>
<td>
<p>A name to determine
which samples to merge using <code><a href="#topic+merge_samples2">merge_samples2()</a></code> function.
Need to be in <code>physeq@sam_data</code>.
Need to be use when you want to wrap by factor the final plot
with the number of taxa (type=&quot;nb_asv&quot;)</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_by_sample">by_sample</code></td>
<td>
<p>(logical) If FALSE (default), sample information is not taking
into account, so the taxonomy is studied globally. If fact is not NULL, by_sample
is automatically set to TRUE.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_rarefy_by_sample">rarefy_by_sample</code></td>
<td>
<p>(logical, default FALSE) If TRUE, rarefy
samples using <code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code> function.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot  the last column</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_type">type</code></td>
<td>
<p>If &quot;nb_seq&quot; (default), the number of sequences is
used in plot. If &quot;nb_asv&quot;, the number of ASV is plotted.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_width">width</code></td>
<td>
<p>(passed on to <code><a href="ggalluvial.html#topic+geom_flow">ggalluvial::geom_flow()</a></code>) the width of each stratum,
as a proportion of the distance between axes. Defaults to 1/3.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_min.size">min.size</code></td>
<td>
<p>(passed on to <code><a href="ggfittext.html#topic+geom_fit_text">ggfittext::geom_fit_text()</a></code>) Minimum font size,
in points. Text that would need to be shrunk below this size to fit the box will
be hidden. Defaults to 4 pt.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default FALSE) If set to TRUE, remove samples with
NA in the variables set in formula.</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_use_ggfittext">use_ggfittext</code></td>
<td>
<p>(logical, default FALSE) Do we use ggfittext to plot labels?</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_use_geom_label">use_geom_label</code></td>
<td>
<p>(logical, default FALSE) Do we use geom_label to plot labels?</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_size_lab">size_lab</code></td>
<td>
<p>Size for label if use_ggfittext is FALSE</p>
</td></tr>
<tr><td><code id="ggaluv_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggalluvial.html#topic+geom_flow">ggalluvial::geom_flow()</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>ggalluvial</code> package if you
use this function.
</p>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+sankey_pq">sankey_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ggalluvial")) {
  ggaluv_pq(data_fungi_mini)
}

if (requireNamespace("ggalluvial")) {
  ggaluv_pq(data_fungi_mini, type = "nb_asv")

  ggaluv_pq(data_fungi_mini, wrap_factor = "Height", by_sample = TRUE, type = "nb_asv") +
    facet_wrap("Height")

  ggaluv_pq(data_fungi_mini,
    width = 0.9, min.size = 10,
    type = "nb_asv", taxa_ranks = c("Phylum", "Class", "Order", "Family", "Genus")
  ) +
    coord_flip() + scale_x_discrete(limits = rev)
}

</code></pre>

<hr>
<h2 id='ggbetween_pq'>Box/Violin plots for between-subjects comparisons of Hill Number</h2><span id='topic+ggbetween_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Note that contrary to <code><a href="#topic+hill_pq">hill_pq()</a></code>, this function does not take into
account for difference in the number of sequences per samples/modalities.
You may use rarefy_by_sample = TRUE if the mean number of sequences per
samples differs among modalities.
</p>
<p>Basically a wrapper of function <code><a href="ggstatsplot.html#topic+ggbetweenstats">ggstatsplot::ggbetweenstats()</a></code> for
object of class phyloseq
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggbetween_pq(physeq, fact, one_plot = FALSE, rarefy_by_sample = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggbetween_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="ggbetween_pq_+3A_fact">fact</code></td>
<td>
<p>(required): The variable to test. Must be present in
the <code>sam_data</code> slot of the physeq object.</p>
</td></tr>
<tr><td><code id="ggbetween_pq_+3A_one_plot">one_plot</code></td>
<td>
<p>(logical, default FALSE) If TRUE, return a unique
plot with the three plot inside using the patchwork package.</p>
</td></tr>
<tr><td><code id="ggbetween_pq_+3A_rarefy_by_sample">rarefy_by_sample</code></td>
<td>
<p>(logical, default FALSE) If TRUE, rarefy
samples using <code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code> function</p>
</td></tr>
<tr><td><code id="ggbetween_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggstatsplot.html#topic+ggbetweenstats">ggstatsplot::ggbetweenstats()</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>ggstatsplot::ggbetweenstats()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>Either an unique ggplot2 object (if one_plot is TRUE) or
a list of 3 ggplot2 plot:
</p>

<ul>
<li><p> plot_Hill_0 : the ggbetweenstats of Hill number 0 (= species richness)
against the variable fact
</p>
</li>
<li><p> plot_Hill_1 : the ggbetweenstats of Hill number 1 (= Shannon index)
against the variable fact
</p>
</li>
<li><p> plot_Hill_2 : the ggbetweenstats of Hill number 2 (= Simpson index)
against the variable fact
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("ggstatsplot")) {
  p &lt;- ggbetween_pq(data_fungi, fact = "Time", p.adjust.method = "BH")
  p[[1]]
  ggbetween_pq(data_fungi, fact = "Height", one_plot = TRUE)
  ggbetween_pq(data_fungi, fact = "Height", one_plot = TRUE, rarefy_by_sample = TRUE)
}

</code></pre>

<hr>
<h2 id='ggscatt_pq'>Scatterplot with marginal distributions and statistical results against
Hill diversity of phyloseq object</h2><span id='topic+ggscatt_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Basically a wrapper of function <code><a href="ggstatsplot.html#topic+ggscatterstats">ggstatsplot::ggscatterstats()</a></code> for
object of class phyloseq and Hill number.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggscatt_pq(
  physeq,
  num_modality,
  hill_scales = c(0, 1, 2),
  rarefy_by_sample = FALSE,
  one_plot = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggscatt_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="ggscatt_pq_+3A_num_modality">num_modality</code></td>
<td>
<p>(required) Name of the numeric column in
<code>physeq@sam_data</code> to plot and test against hill numberk</p>
</td></tr>
<tr><td><code id="ggscatt_pq_+3A_hill_scales">hill_scales</code></td>
<td>
<p>(a vector of integer) The list of q values to compute
the hill number H^q. If Null, no hill number are computed. Default value
compute the Hill number 0 (Species richness), the Hill number 1
(exponential of Shannon Index) and the Hill number 2 (inverse of Simpson
Index).</p>
</td></tr>
<tr><td><code id="ggscatt_pq_+3A_rarefy_by_sample">rarefy_by_sample</code></td>
<td>
<p>(logical, default FALSE) If TRUE, rarefy
samples using <code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code> function.</p>
</td></tr>
<tr><td><code id="ggscatt_pq_+3A_one_plot">one_plot</code></td>
<td>
<p>(logical, default FALSE) If TRUE, return a unique
plot with the three plot inside using the patchwork package.</p>
</td></tr>
<tr><td><code id="ggscatt_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="ggstatsplot.html#topic+ggscatterstats">ggstatsplot::ggscatterstats()</a></code>
function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>ggstatsplot::ggscatterstats()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>Either an unique ggplot2 object (if one_plot is TRUE) or
a list of ggplot2 plot for each hill_scales.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggbetween_pq">ggbetween_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ggstatsplot")) {
  ggscatt_pq(data_fungi_mini, "Time", type = "non-parametric")
  ggscatt_pq(data_fungi_mini, "Time", hill_scales = 1:4, type = "parametric")
  ggscatt_pq(data_fungi_mini, "Sample_id",
    hill_scales = c(0, 0.5),
    one_plot = FALSE
  )
}
</code></pre>

<hr>
<h2 id='ggvenn_pq'>Venn diagram of <code>phyloseq-class</code> object using
<code>ggVennDiagram::ggVennDiagram</code> function</h2><span id='topic+ggvenn_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Note that you can use ggplot2 function to customize the plot
for ex. <code>+ scale_fill_distiller(palette = "BuPu", direction = 1)</code>
and <code>+ scale_x_continuous(expand = expansion(mult = 0.5))</code>. See
examples.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ggvenn_pq(
  physeq = NULL,
  fact = NULL,
  min_nb_seq = 0,
  taxonomic_rank = NULL,
  split_by = NULL,
  add_nb_samples = TRUE,
  add_nb_seq = FALSE,
  rarefy_before_merging = FALSE,
  rarefy_after_merging = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ggvenn_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_fact">fact</code></td>
<td>
<p>(required): Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_min_nb_seq">min_nb_seq</code></td>
<td>
<p>minimum number of sequences by OTUs by
samples to take into count this OTUs in this sample. For example,
if min_nb_seq=2,each value of 2 or less in the OTU table
will not count in the venn diagram</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_taxonomic_rank">taxonomic_rank</code></td>
<td>
<p>Name (or number) of a taxonomic rank
to count. If set to Null (the default) the number of OTUs is counted.</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_split_by">split_by</code></td>
<td>
<p>Split into multiple plot using variable split_by.
The name of a variable must be present in <code>sam_data</code> slot
of the physeq object.</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_add_nb_samples">add_nb_samples</code></td>
<td>
<p>(logical, default TRUE) Add the number of samples to
levels names</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_add_nb_seq">add_nb_seq</code></td>
<td>
<p>(logical, default FALSE) Add the number of sequences to
levels names</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_rarefy_before_merging">rarefy_before_merging</code></td>
<td>
<p>Rarefy each sample before merging by the
modalities of args <code>fact</code>. Use <code>phyloseq::rarefy_even_depth()</code> function</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_rarefy_after_merging">rarefy_after_merging</code></td>
<td>
<p>Rarefy each sample after merging by the
modalities of args <code>fact</code>.</p>
</td></tr>
<tr><td><code id="ggvenn_pq_+3A_...">...</code></td>
<td>
<p>Other arguments for the <code>ggVennDiagram::ggVennDiagram</code> function
for ex. <code>category.names</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot representing Venn diagram of
modalities of the argument <code>factor</code> or if split_by is set a list
of plots.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+upset_pq">upset_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ggVennDiagram")) {
  ggvenn_pq(data_fungi, fact = "Height")
}

if (requireNamespace("ggVennDiagram")) {
  ggvenn_pq(data_fungi, fact = "Height") +
    ggplot2::scale_fill_distiller(palette = "BuPu", direction = 1)
  pl &lt;- ggvenn_pq(data_fungi, fact = "Height", split_by = "Time")
  for (i in 1:length(pl)) {
    p &lt;- pl[[i]] +
      scale_fill_distiller(palette = "BuPu", direction = 1) +
      theme(plot.title = element_text(hjust = 0.5, size = 22))
    print(p)
  }

  data_fungi2 &lt;- subset_samples(data_fungi, data_fungi@sam_data$Tree_name == "A10-005" |
    data_fungi@sam_data$Height %in% c("Low", "High"))
  ggvenn_pq(data_fungi2, fact = "Height")

  ggvenn_pq(data_fungi, fact = "Height", add_nb_seq = TRUE, set_size = 4)
  ggvenn_pq(data_fungi, fact = "Height", rarefy_before_merging = TRUE)
  ggvenn_pq(data_fungi, fact = "Height", rarefy_after_merging = TRUE) +
    scale_x_continuous(expand = expansion(mult = 0.5))
}

</code></pre>

<hr>
<h2 id='glmutli_pq'>Automated model selection and multimodel inference with (G)LMs for phyloseq</h2><span id='topic+glmutli_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>glmutli_pq(
  physeq,
  formula,
  fitfunction = "lm",
  hill_scales = c(0, 1, 2),
  aic_step = 2,
  confsetsize = 100,
  plotty = FALSE,
  level = 1,
  method = "h",
  crit = "aicc",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="glmutli_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_formula">formula</code></td>
<td>
<p>(required) a formula for <code><a href="glmulti.html#topic+glmulti">glmulti::glmulti()</a></code>
Variables must be present in the <code>physeq@sam_data</code> slot or be one
of hill number defined in hill_scales or the variable Abundance which
refer to the number of sequences per sample.</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_fitfunction">fitfunction</code></td>
<td>
<p>(default &quot;lm&quot;)</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_hill_scales">hill_scales</code></td>
<td>
<p>(a vector of integer) The list of q values to compute
the hill number H^q. If Null, no hill number are computed. Default value
compute the Hill number 0 (Species richness), the Hill number 1
(exponential of Shannon Index) and the Hill number 2 (inverse of Simpson
Index).</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_aic_step">aic_step</code></td>
<td>
<p>The value between AIC scores to cut for.</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_confsetsize">confsetsize</code></td>
<td>
<p>The number of models to be looked for, i.e. the size of the returned confidence set.</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_plotty">plotty</code></td>
<td>
<p>(logical) Whether to plot the progress of the IC profile when running.</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_level">level</code></td>
<td>
<p>If 1, only main effects (terms of order 1) are used to build
the candidate set. If 2, pairwise interactions are also used (higher order
interactions are currently ignored)</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_method">method</code></td>
<td>
<p>The method to be used to explore the candidate set of models.
If &quot;h&quot; (default) an exhaustive screening is undertaken.
If &quot;g&quot; the genetic algorithm is employed (recommended for large candidate sets).
If &quot;l&quot;, a very fast exhaustive branch-and-bound algorithm is used.
Package leaps must then be loaded, and this can only be applied to linear models
with covariates and no interactions. If &quot;d&quot;, a simple summary of the candidate set
is printed, including the number of candidate models.</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_crit">crit</code></td>
<td>
<p>The Information Criterion to be used. Default is the small-sample corrected AIC (aicc). This should be a function that accepts a fitted model as first argument. Other provided functions are the classic AIC, the Bayes IC (bic), and QAIC/QAICc (qaic and qaicc).</p>
</td></tr>
<tr><td><code id="glmutli_pq_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="glmulti.html#topic+glmulti">glmulti::glmulti()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code><a href="glmulti.html#topic+glmulti">glmulti::glmulti()</a></code> if you
use this function.
</p>


<h3>Value</h3>

<p>A data.frame summarizing the glmulti results with columns
</p>
<p>-estimates
-unconditional_interval
-nb_model&quot;
-importance
-alpha
</p>


<h3>See Also</h3>

<p><code><a href="glmulti.html#topic+glmulti">glmulti::glmulti()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("glmulti")) {
  res_glmulti &lt;-
    glmutli_pq(data_fungi, "Hill_0 ~ Hill_1 + Abundance + Time + Height", level = 1)
  res_glmulti
  res_glmulti_interaction &lt;-
    glmutli_pq(data_fungi, "Hill_0 ~ Abundance + Time + Height", level = 2)
  res_glmulti
}

</code></pre>

<hr>
<h2 id='graph_test_pq'>Performs graph-based permutation tests on phyloseq object</h2><span id='topic+graph_test_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>A wrapper of <code><a href="phyloseqGraphTest.html#topic+graph_perm_test">phyloseqGraphTest::graph_perm_test()</a></code> for quick plot with
important statistics
</p>


<h3>Usage</h3>

<pre><code class='language-R'>graph_test_pq(
  physeq,
  fact,
  merge_sample_by = NULL,
  nperm = 999,
  return_plot = TRUE,
  title = "Graph Test",
  na_remove = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="graph_test_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>. This should be a factor
with two or more levels.</p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>a vector to determine
which samples to merge using <code><a href="#topic+merge_samples2">merge_samples2()</a></code> function.
Need to be in <code>physeq@sam_data</code></p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_nperm">nperm</code></td>
<td>
<p>(int) The number of permutations to perform.</p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_return_plot">return_plot</code></td>
<td>
<p>(logical) Do we return only the result
of the test or do we plot the result?</p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_title">title</code></td>
<td>
<p>The title of the Graph.</p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default FALSE) If set to TRUE, remove samples with
NA in the variables set in formula.</p>
</td></tr>
<tr><td><code id="graph_test_pq_+3A_...">...</code></td>
<td>
<p>Other params for be passed on to
<code><a href="phyloseqGraphTest.html#topic+graph_perm_test">phyloseqGraphTest::graph_perm_test()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please cite <code>phyloseqGraphTest</code> package.
</p>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot with a subtitle indicating the pvalue
and the number of permutations
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("phyloseqGraphTest")) {
  data(enterotype)
  graph_test_pq(enterotype, fact = "SeqTech")
  graph_test_pq(enterotype, fact = "Enterotype", na_remove = TRUE)
}

</code></pre>

<hr>
<h2 id='heat_tree_pq'>Heat tree from <code>metacoder</code> package using <code>tax_table</code> slot</h2><span id='topic+heat_tree_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Note that the number of ASV is store under the name <code>n_obs</code>
and the number of sequences under the name <code>nb_sequences</code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>heat_tree_pq(physeq, taxonomic_level = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="heat_tree_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="heat_tree_pq_+3A_taxonomic_level">taxonomic_level</code></td>
<td>
<p>(default: NULL): a vector of selected
taxonomic level using
their column numbers (e.g. taxonomic_level = 1:7)</p>
</td></tr>
<tr><td><code id="heat_tree_pq_+3A_...">...</code></td>
<td>
<p>Arguments passed on to <code><a href="metacoder.html#topic+heat_tree">heat_tree</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("metacoder")) {
  library("metacoder")
  data("GlobalPatterns", package = "phyloseq")

  GPsubset &lt;- subset_taxa(
    GlobalPatterns,
    GlobalPatterns@tax_table[, 1] == "Bacteria"
  )

  GPsubset &lt;- subset_taxa(
    GPsubset,
    rowSums(GPsubset@otu_table) &gt; 5000
  )

  GPsubset &lt;- subset_taxa(
    GPsubset,
    rowSums(is.na(GPsubset@tax_table)) == 0
  )

  heat_tree_pq(GPsubset,
    node_size = n_obs,
    node_color = n_obs,
    node_label = taxon_names,
    tree_label = taxon_names,
    node_size_trans = "log10 area"
  )

  heat_tree_pq(GPsubset,
    node_size = nb_sequences,
    node_color = n_obs,
    node_label = taxon_names,
    tree_label = taxon_names,
    node_size_trans = "log10 area"
  )
}

</code></pre>

<hr>
<h2 id='hill_pq'>Graphical representation of hill number 0, 1 and 2 across a factor</h2><span id='topic+hill_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
Hill numbers are the number of equiprobable species giving the same
diversity value as the observed distribution. The Hill number 0
correspond to Species richness), the Hill number 1 to
the exponential of Shannon Index and the Hill number 2 to the inverse
of Simpson Index)
</p>
<p>Note that (if correction_for_sample_size is TRUE, default behavior)
this function use a sqrt of the read numbers in the linear
model in order to correct for uneven sampling depth. This correction
is only done before tuckey HSD plot and do not change the hill number
computed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hill_pq(
  physeq,
  fact = NULL,
  variable = NULL,
  hill_scales = c(0, 1, 2),
  color_fac = NA,
  letters = FALSE,
  add_points = FALSE,
  add_info = TRUE,
  one_plot = FALSE,
  plot_with_tuckey = TRUE,
  correction_for_sample_size = TRUE,
  na_remove = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hill_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_fact">fact</code></td>
<td>
<p>(required): The variable to test. Must be present in
the <code>sam_data</code> slot of the physeq object.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_variable">variable</code></td>
<td>
<p>: Alias for factor. Kept only for backward compatibility.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_hill_scales">hill_scales</code></td>
<td>
<p>(a vector of integer) The list of q values to compute
the hill number H^q. If Null, no hill number are computed. Default value
compute the Hill number 0 (Species richness), the Hill number 1
(exponential of Shannon Index) and the Hill number 2 (inverse of Simpson
Index).</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_color_fac">color_fac</code></td>
<td>
<p>(optional): The variable to color the barplot. For ex.
same as fact. Not very useful because ggplot2 plot colors can be
change using <code>scale_color_XXX()</code> function.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_letters">letters</code></td>
<td>
<p>(optional, default=FALSE): If set to TRUE, the plot
show letters based on p-values for comparison. Use the
<code><a href="multcompView.html#topic+multcompLetters">multcompLetters</a></code> function from the package
multcompLetters. BROKEN for the moment. Note that na values in The
variable param need to be removed (see examples) to use letters.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_add_points">add_points</code></td>
<td>
<p>(logical): add jitter point on boxplot</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_add_info">add_info</code></td>
<td>
<p>(logical, default TRUE) Do we add a subtitle with
information about the number of samples per modality ?</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_one_plot">one_plot</code></td>
<td>
<p>(logical, default FALSE) If TRUE, return a unique
plot with the four plot inside using the patchwork package.
Note that if letters and one_plot are both TRUE, tuckey HSD results
are discarded from the unique plot. In that case, use one_plot = FALSE
to see the tuckey HSD results in the fourth plot of the resulting list.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_plot_with_tuckey">plot_with_tuckey</code></td>
<td>
<p>(logical, default TRUE). If one_plot is set to
TRUE and letters to FALSE, allow to discard the tuckey plot part with
plot_with_tuckey = FALSE</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_correction_for_sample_size">correction_for_sample_size</code></td>
<td>
<p>(logical, default TRUE) This function
use a sqrt of the read numbers in the linear model in order to
correct for uneven sampling depth in the Tuckey TEST. This params
do not change value of Hill number but only the test associated
values (including the pvalues). To rarefy samples, you may use the
function <code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code>.</p>
</td></tr>
<tr><td><code id="hill_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default TRUE) Do we remove samples with NA in
the factor fact ? Note that na_remove is always TRUE when using
letters = TRUE</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either an unique ggplot2 object (if one_plot is TRUE) or
a list of 4 ggplot2 plot:
</p>

<ul>
<li><p> plot_Hill_0 : the boxplot of Hill number 0 (= species richness)
against the variable
</p>
</li>
<li><p> plot_Hill_1 : the boxplot of Hill number 1 (= Shannon index)
against the variable
</p>
</li>
<li><p> plot_Hill_2 : the boxplot of Hill number 2 (= Simpson index)
against the variable
</p>
</li>
<li><p> plot_tuckey : plot the result of the Tuckey HSD test
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+psmelt_samples_pq">psmelt_samples_pq()</a></code> and <code><a href="#topic+ggbetween_pq">ggbetween_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
p &lt;- hill_pq(data_fungi_mini, "Height", hill_scales = 1:2)
p_h1 &lt;- p[[1]] + theme(legend.position = "none")
p_h2 &lt;- p[[2]] + theme(legend.position = "none")
multiplot(plotlist = list(p_h1, p_h2, p[[3]]), cols = 4)

if (requireNamespace("multcompView")) {
  p2 &lt;- hill_pq(data_fungi, "Time",
    correction_for_sample_size = FALSE,
    letters = TRUE, add_points = TRUE, plot_with_tuckey = FALSE
  )
  if (requireNamespace("patchwork")) {
    patchwork::wrap_plots(p2, guides = "collect")
  }
  # Artificially modify data_fungi to force alpha-diversity effect
  data_fungi_modif &lt;- clean_pq(subset_samples_pq(data_fungi, !is.na(data_fungi@sam_data$Height)))
  data_fungi_modif@otu_table[data_fungi_modif@sam_data$Height == "High", ] &lt;-
    data_fungi_modif@otu_table[data_fungi_modif@sam_data$Height == "High", ] +
    sample(c(rep(0, ntaxa(data_fungi_modif) / 2), rep(100, ntaxa(data_fungi_modif) / 2)))
  p3 &lt;- hill_pq(data_fungi_modif, "Height", letters = TRUE)
  p3[[1]]
}

</code></pre>

<hr>
<h2 id='hill_test_rarperm_pq'>Test multiple times effect of factor on Hill diversity
with different rarefaction even depth</h2><span id='topic+hill_test_rarperm_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hill_test_rarperm_pq(
  physeq,
  fact,
  hill_scales = c(0, 1, 2),
  nperm = 99,
  sample.size = min(sample_sums(physeq)),
  verbose = FALSE,
  progress_bar = TRUE,
  p_val_signif = 0.05,
  type = "non-parametrique",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hill_test_rarperm_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot
different lines</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_hill_scales">hill_scales</code></td>
<td>
<p>(a vector of integer) The list of q values to compute
the hill number H^q. If Null, no hill number are computed. Default value
compute the Hill number 0 (Species richness), the Hill number 1
(exponential of Shannon Index) and the Hill number 2 (inverse of Simpson
Index).</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_nperm">nperm</code></td>
<td>
<p>(int) The number of permutations to perform.</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_sample.size">sample.size</code></td>
<td>
<p>(int) A single integer value equal to the number of
reads being simulated, also known as the depth. See
<code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code>.</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical). If TRUE, print additional informations.</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_progress_bar">progress_bar</code></td>
<td>
<p>(logical, default TRUE) Do we print progress during
the calculation?</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_p_val_signif">p_val_signif</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) The mimimum value of p-value to count a
test as significant int the <code>prop_signif</code> result.</p>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_type">type</code></td>
<td>
<p>A character specifying the type of statistical approach
(See <code><a href="ggstatsplot.html#topic+ggbetweenstats">ggstatsplot::ggbetweenstats()</a></code> for more details):
</p>

<ul>
<li><p> &quot;parametric&quot;
</p>
</li>
<li><p> &quot;nonparametric&quot;
</p>
</li>
<li><p> &quot;robust&quot;
</p>
</li>
<li><p> &quot;bayes&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="hill_test_rarperm_pq_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="ggstatsplot.html#topic+ggbetweenstats">ggstatsplot::ggbetweenstats()</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 6 components :
</p>

<ul>
<li><p> method
</p>
</li>
<li><p> expressions
</p>
</li>
<li><p> plots
</p>
</li>
<li><p> pvals
</p>
</li>
<li><p> prop_signif
</p>
</li>
<li><p> statistics
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="ggstatsplot.html#topic+ggbetweenstats">ggstatsplot::ggbetweenstats()</a></code>, <code><a href="#topic+hill_pq">hill_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("ggstatsplot")) {
  hill_test_rarperm_pq(data_fungi, "Time", nperm = 2)
  res &lt;- hill_test_rarperm_pq(data_fungi, "Height", nperm = 9, p.val = 0.9)
  patchwork::wrap_plots(res$plots[[1]])
  res$plots[[1]][[1]] + res$plots[[2]][[1]] + res$plots[[3]][[1]]
  res$prop_signif
  res_para &lt;- hill_test_rarperm_pq(data_fungi, "Height", nperm = 9, type = "parametrique")
  res_para$plots[[1]][[1]] + res_para$plots[[2]][[1]] + res_para$plots[[3]][[1]]
  res_para$pvals
  res_para$method
  res_para$expressions[[1]]
}

</code></pre>

<hr>
<h2 id='hill_tuckey_pq'>Calculate hill number and compute Tuckey post-hoc test</h2><span id='topic+hill_tuckey_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
Note that, by default, this function use a sqrt of the read numbers in the linear
model in order to correct for uneven sampling depth.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>hill_tuckey_pq(
  physeq,
  modality,
  hill_scales = c(0, 1, 2),
  silent = TRUE,
  correction_for_sample_size = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="hill_tuckey_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="hill_tuckey_pq_+3A_modality">modality</code></td>
<td>
<p>(required) the variable to test</p>
</td></tr>
<tr><td><code id="hill_tuckey_pq_+3A_hill_scales">hill_scales</code></td>
<td>
<p>(a vector of integer) The list of q values to compute
the hill number H^q. If Null, no hill number are computed. Default value
compute the Hill number 0 (Species richness), the Hill number 1
(exponential of Shannon Index) and the Hill number 2 (inverse of Simpson
Index).</p>
</td></tr>
<tr><td><code id="hill_tuckey_pq_+3A_silent">silent</code></td>
<td>
<p>(logical) If TRUE, no message are printing.</p>
</td></tr>
<tr><td><code id="hill_tuckey_pq_+3A_correction_for_sample_size">correction_for_sample_size</code></td>
<td>
<p>(logical, default TRUE) This function
use a sqrt of the read numbers in the linear model in order to
correct for uneven sampling depth.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("GlobalPatterns", package = "phyloseq")
GlobalPatterns@sam_data[, "Soil_logical"] &lt;-
  ifelse(GlobalPatterns@sam_data[, "SampleType"] == "Soil", "Soil", "Not Soil")
hill_tuckey_pq(GlobalPatterns, "Soil_logical")
hill_tuckey_pq(GlobalPatterns, "Soil_logical", hill_scales = 1:2)
</code></pre>

<hr>
<h2 id='iNEXT_pq'>iNterpolation and EXTrapolation of Hill numbers (with iNEXT)</h2><span id='topic+iNEXT_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>iNEXT_pq(physeq, merge_sample_by = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="iNEXT_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="iNEXT_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>(default: NULL) if not <code>NULL</code> samples of
physeq are merged using the vector set by <code>merge_sample_by</code>. This
merging used the <code><a href="#topic+merge_samples2">merge_samples2()</a></code>. In the case of
<code><a href="#topic+biplot_pq">biplot_pq()</a></code> this must be a factor with two levels only.</p>
</td></tr>
<tr><td><code id="iNEXT_pq_+3A_...">...</code></td>
<td>
<p>Other arguments for the <code><a href="iNEXT.html#topic+iNEXT">iNEXT::iNEXT()</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>see <code><a href="iNEXT.html#topic+iNEXT">iNEXT::iNEXT()</a></code> documentation
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
This function is mainly a wrapper of the work of others.
Please make a reference to <code>iNEXT::iNEXT()</code> if you
use this function.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("iNEXT")) {
  data("GlobalPatterns", package = "phyloseq")
  GPsubset &lt;- subset_taxa(
    GlobalPatterns,
    GlobalPatterns@tax_table[, 1] == "Bacteria"
  )
  GPsubset &lt;- subset_taxa(
    GPsubset,
    rowSums(GPsubset@otu_table) &gt; 20000
  )
  GPsubset &lt;- subset_taxa(
    GPsubset,
    rowSums(is.na(GPsubset@tax_table)) == 0
  )
  GPsubset@sam_data$human &lt;- GPsubset@sam_data$SampleType %in%
    c("Skin", "Feces", "Tong")
  res_iNEXT &lt;- iNEXT_pq(
    GPsubset,
    merge_sample_by = "human",
    q = 1,
    datatype = "abundance",
    nboot = 2
  )
  iNEXT::ggiNEXT(res_iNEXT)
  iNEXT::ggiNEXT(res_iNEXT, type = 2)
  iNEXT::ggiNEXT(res_iNEXT, type = 3)
}

</code></pre>

<hr>
<h2 id='is_cutadapt_installed'>Test if cutadapt is installed.</h2><span id='topic+is_cutadapt_installed'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Useful for testthat and examples compilation for R CMD CHECK and
test coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_cutadapt_installed(
  args_before_cutadapt =
    "source ~/miniconda3/etc/profile.d/conda.sh &amp;&amp; conda activate cutadaptenv &amp;&amp; "
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_cutadapt_installed_+3A_args_before_cutadapt">args_before_cutadapt</code></td>
<td>
<p>: (String) A one line bash command to run before
to run cutadapt. For examples, &quot;source ~/miniconda3/etc/profile.d/conda.sh &amp;&amp; conda activate cutadaptenv &amp;&amp;&quot; allow to bypass the conda init which asks to restart the shell</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical that say if cutadapt is install in
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MiscMetabar::is_cutadapt_installed()
</code></pre>

<hr>
<h2 id='is_falco_installed'>Test if falco is installed.</h2><span id='topic+is_falco_installed'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Useful for testthat and examples compilation for R CMD CHECK and
test coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_falco_installed(path = "falco")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_falco_installed_+3A_path">path</code></td>
<td>
<p>(default: falco) Path to falco</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical that say if falco is install in
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MiscMetabar::is_falco_installed()
</code></pre>

<hr>
<h2 id='is_krona_installed'>Test if krona is installed.</h2><span id='topic+is_krona_installed'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Useful for testthat and examples compilation for R CMD CHECK and
test coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_krona_installed(path = "ktImportKrona")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_krona_installed_+3A_path">path</code></td>
<td>
<p>(default: krona) Path to krona</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical that say if krona is install in
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MiscMetabar::is_krona_installed()
</code></pre>

<hr>
<h2 id='is_mumu_installed'>Test if mumu is installed.</h2><span id='topic+is_mumu_installed'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Useful for testthat and examples compilation for R CMD CHECK and
test coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_mumu_installed(path = "mumu")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_mumu_installed_+3A_path">path</code></td>
<td>
<p>(default: mumu) Path to mumu</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical that say if mumu is install in
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MiscMetabar::is_mumu_installed()
</code></pre>

<hr>
<h2 id='is_swarm_installed'>Test if swarm is installed.</h2><span id='topic+is_swarm_installed'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Useful for testthat and examples compilation for R CMD CHECK and
test coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_swarm_installed(path = "swarm")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_swarm_installed_+3A_path">path</code></td>
<td>
<p>(default: swarm) Path to falco</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical that say if swarm is install in
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MiscMetabar::is_swarm_installed()
</code></pre>

<hr>
<h2 id='is_vsearch_installed'>Test if vsearch is installed.</h2><span id='topic+is_vsearch_installed'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Useful for testthat and examples compilation for R CMD CHECK and
test coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>is_vsearch_installed(path = "vsearch")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="is_vsearch_installed_+3A_path">path</code></td>
<td>
<p>(default: vsearch) Path to vsearch</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A logical that say if vsearch is install in
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>MiscMetabar::is_vsearch_installed()
</code></pre>

<hr>
<h2 id='krona'>Make Krona files using <a href="https://github.com/marbl/Krona/wiki">KronaTools</a>.</h2><span id='topic+krona'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Need the installation of kronatools on the computer (<a href="https://github.com/marbl/Krona/wiki/Installing">installation instruction</a>).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>krona(
  physeq,
  file = "krona.html",
  nb_seq = TRUE,
  ranks = "All",
  add_unassigned_rank = 0,
  name = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="krona_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="krona_+3A_file">file</code></td>
<td>
<p>(required) the location of the html file to save</p>
</td></tr>
<tr><td><code id="krona_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(logical) If true, Krona set the distribution
of sequences in the taxonomy. If False, Krona set the distribution of ASVs
in the taxonomy.</p>
</td></tr>
<tr><td><code id="krona_+3A_ranks">ranks</code></td>
<td>
<p>Number of the taxonomic ranks to plot
(num of the column in <code>tax_table</code> slot of your <code>physeq</code> object).
Default setting plot all the ranks (argument 'All').</p>
</td></tr>
<tr><td><code id="krona_+3A_add_unassigned_rank">add_unassigned_rank</code></td>
<td>
<p>(int) Add unassigned for rank
inferior to 'add_unassigned_rank' when necessary.</p>
</td></tr>
<tr><td><code id="krona_+3A_name">name</code></td>
<td>
<p>A name for intermediary files, Useful to name
your krona result files before merging using <code><a href="#topic+merge_krona">merge_krona()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/marbl/Krona">Krona</a> if
you use this function.
</p>


<h3>Value</h3>

<p>A html file
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+merge_krona">merge_krona</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("GlobalPatterns", package = "phyloseq")
GA &lt;- subset_taxa(GlobalPatterns, Phylum == "Acidobacteria")
## Not run: 
krona(GA, "Number.of.sequences.html")
krona(GA, "Number.of.ASVs.html", nb_seq = FALSE)
merge_krona(c("Number.of.sequences.html", "Number.of.ASVs.html"))

## End(Not run)

</code></pre>

<hr>
<h2 id='LCBD_pq'>Compute and test local contributions to beta diversity (LCBD) of
samples</h2><span id='topic+LCBD_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function in the case of <code>physeq</code>
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>LCBD_pq(physeq, p_adjust_method = "BH", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="LCBD_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="LCBD_pq_+3A_p_adjust_method">p_adjust_method</code></td>
<td>
<p>(chr, default &quot;BH&quot;): the method used to adjust p-value</p>
</td></tr>
<tr><td><code id="LCBD_pq_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class <code>beta.div</code> see <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function
for more information
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
This function is mainly a wrapper of the work of others.
Please make a reference to <code>adespatial::beta.div()</code> if you
use this function.
</p>


<h3>See Also</h3>

<p><a href="#topic+plot_LCBD_pq">plot_LCBD_pq</a>, <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("adespatial")) {
  res &lt;- LCBD_pq(data_fungi_sp_known, nperm = 5)
  str(res)
  length(res$LCBD)
  length(res$SCBD)
}

if (requireNamespace("adespatial")) {
  LCBD_pq(data_fungi_sp_known, nperm = 5, method = "jaccard")
}


</code></pre>

<hr>
<h2 id='list_fastq_files'>List fastq files</h2><span id='topic+list_fastq_files'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>list_fastq_files(
  path,
  paired_end = TRUE,
  pattern = "fastq",
  pattern_R1 = "_R1_",
  pattern_R2 = "_R2_",
  nb_files = Inf
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="list_fastq_files_+3A_path">path</code></td>
<td>
<p>path to files (required)</p>
</td></tr>
<tr><td><code id="list_fastq_files_+3A_paired_end">paired_end</code></td>
<td>
<p>do you have paired_end files? (default TRUE)</p>
</td></tr>
<tr><td><code id="list_fastq_files_+3A_pattern">pattern</code></td>
<td>
<p>a pattern to filter files (passed on to list.files function).</p>
</td></tr>
<tr><td><code id="list_fastq_files_+3A_pattern_r1">pattern_R1</code></td>
<td>
<p>a pattern to filter R1 files (default &quot;<em>R1</em>&quot;)</p>
</td></tr>
<tr><td><code id="list_fastq_files_+3A_pattern_r2">pattern_R2</code></td>
<td>
<p>a pattern to filter R2 files (default &quot;<em>R2</em>&quot;)</p>
</td></tr>
<tr><td><code id="list_fastq_files_+3A_nb_files">nb_files</code></td>
<td>
<p>the number of fastq files to list (default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of one (single end) or two (paired end) list of files
files are sorted by names (default behavior of <code>list.files()</code>)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>list_fastq_files("extdata")
list_fastq_files("extdata", paired_end = FALSE, pattern_R1 = "")

</code></pre>

<hr>
<h2 id='lulu'>Post Clustering Curation of Amplicon Data.</h2><span id='topic+lulu'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>The original function and documentation was written by Tobias Guldberg Frøslev
in the <a href="https://github.com/tobiasgf/lulu">lulu</a> package.
</p>
<p>This algorithm <code>lulu</code> consumes an OTU table and a matchlist, and
evaluates cooccurence of 'daughters' (potential analytical artefacts) and
their 'parents' (~= real biological species/OTUs). The algorithm requires an
OTU table (species/site matrix), and a match list. The OTU table can be
made with various r-packages (e.g. <code>DADA2</code>) or
external pipelines (<code>VSEARCH, USEARCH, QIIME</code>, etc.), and the
match-list can be made with external bioinformatic tools like
<code>VSEARCH, USEARCH, BLASTN</code> or another algorithm
for pair-wise sequence matching.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lulu(
  otu_table,
  matchlist,
  minimum_ratio_type = "min",
  minimum_ratio = 1,
  minimum_match = 84,
  minimum_relative_cooccurence = 0.95,
  progress_bar = TRUE,
  log_conserved = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lulu_+3A_otu_table">otu_table</code></td>
<td>
<p>a data.frame with with an OTU table that has sites/samples as
columns and OTUs (unique OTU id's) as rows, and observations as read
counts.</p>
</td></tr>
<tr><td><code id="lulu_+3A_matchlist">matchlist</code></td>
<td>
<p>a data.frame containing three columns: (1) OTU id of
potential child, (2) OTU id of potential parent, (3) match - % identiti
between the sequences of the potential parent and potential child OTUs.
<strong>NB: The matchlist is the product of a mapping of OTU sequences against each other. This is
currently carried out by an external script in e.g. Blastn or VSEARCH, prior to running lulu!</strong></p>
</td></tr>
<tr><td><code id="lulu_+3A_minimum_ratio_type">minimum_ratio_type</code></td>
<td>
<p>sets whether a potential error must have lower
abundance than the parent in all samples <code>min</code> (default), or if an error
just needs to have lower abundance on average <code>avg</code>. Choosing lower
abundance on average over globally lower abundance will greatly increase
the number of designated errors. This option was introduced to make it
possible to account for non-sufficiently clustered intraspecific variation,
but is not generally recommended, as it will also increase the potential of
cluster well-separated, but co-occuring, sequence similar species.</p>
</td></tr>
<tr><td><code id="lulu_+3A_minimum_ratio">minimum_ratio</code></td>
<td>
<p>sets the minimim abundance ratio between a potential error
and a potential parent to be identified as an error. If the <code>minimum_ratio_type</code> is
set to <code>min</code> (default), the <code>minimum_ratio</code> applies to the lowest observed
ration across the samples.  If the <code>minimum_ratio_type</code> is
set to <code>avg</code> (default), the <code>minimum_ratio</code> applies to the mean of observed
ration across the samples.<code>avg</code>. (default is 1).</p>
</td></tr>
<tr><td><code id="lulu_+3A_minimum_match">minimum_match</code></td>
<td>
<p>minimum threshold of sequence similarity
for considering any OTU as an error of another can be set (default 84%).</p>
</td></tr>
<tr><td><code id="lulu_+3A_minimum_relative_cooccurence">minimum_relative_cooccurence</code></td>
<td>
<p>minimum co-occurrence rate, i.e. the
lower rate of occurrence of the potential error explained by co-occurrence
with the potential parent for considering error state.</p>
</td></tr>
<tr><td><code id="lulu_+3A_progress_bar">progress_bar</code></td>
<td>
<p>(Logical, default TRUE) print progress during the calculation or not.</p>
</td></tr>
<tr><td><code id="lulu_+3A_log_conserved">log_conserved</code></td>
<td>
<p>(Logical, default FALSE) conserved log files writed in the disk</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please cite the lulu original paper: https://www.nature.com/articles/s41467-017-01312-x
</p>


<h3>Value</h3>

<p>Function <code>lulu</code> returns a list of results based on the input OTU
table and match list.
</p>

<ul>
<li> <p><code>curated_table</code> - a curated
OTU table with daughters merged with their matching parents.
</p>
</li>
<li> <p><code>curated_count</code> - number of curated (parent) OTUs.
</p>
</li>
<li> <p><code>curated_otus</code> - ids of the OTUs that were accepted as valid OTUs.
</p>
</li>
<li> <p><code>discarded_count</code> - number of discarded (merged with parent) OTUs.
</p>
</li>
<li> <p><code>discarded_otus</code> - ids of the OTUs that were identified as
errors (daughters) and merged with respective parents.
</p>
</li>
<li> <p><code>runtime</code> - time used by the script.
</p>
</li>
<li> <p><code>minimum_match</code> - the id threshold
(minimum match \
by user).
</p>
</li>
<li> <p><code>minimum_relative_cooccurence</code> - minimum ratio of
daughter-occurences explained by co-occurence with parent (set by user).
</p>
</li>
<li> <p><code>otu_map</code> - information of which daughters were mapped to which
parents.
</p>
</li>
<li> <p><code>original_table</code> - original OTU table.
</p>
</li></ul>

<p>The matchlist is the product of a mapping of OTU sequences against each other. This is
currently carried out by an external script in e.g. BLASTN or VSEARCH, prior to running <code>lulu</code>!
Producing the match list requires a file with all the OTU sequences (centroids) - e.g. <code>OTUcentroids.fasta</code>. The matchlist can be produced by mapping all OTUs against each other with an external algorithm like VSEARCH or BLASTN. In <code>VSEARCH</code> a matchlist can be produced e.g. with the following command: <code>vsearch --usearch_global OTUcentroids.fasta --db OTUcentroids.fasta --strand plus --self --id .80 --iddef 1 --userout matchlist.txt --userfields query+target+id --maxaccepts 0 --query_cov .9 --maxhits 10</code>. In <code>BLASTN</code> a matchlist can be produces e.g. with the following commands. First we produce a blast-database from the fasta file: <code>makeblastdb -in OTUcentroids.fasta -parse_seqids -dbtype nucl</code>, then we match the centroids against that database: <code>blastn -db OTUcentoids.fasta -num_threads 10 -outfmt'6 qseqid sseqid pident' -out matchlist.txt -qcov_hsp_perc .90 -perc_identity .84 -query OTUcentroids.fasta</code>
</p>


<h3>Author(s)</h3>

<p>Tobias Guldberg Frøslev (orcid: <a href="https://orcid.org/0000-0002-3530-013X">0000-0002-3530-013X</a>),
modified by Adrien Taudière
</p>

<hr>
<h2 id='lulu_pq'>Lulu reclustering of class <code>physeq</code></h2><span id='topic+lulu_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>See https://www.nature.com/articles/s41467-017-01312-x for more information
on the method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>lulu_pq(
  physeq,
  nproc = 1,
  id = 0.84,
  vsearchpath = "vsearch",
  verbose = FALSE,
  clean_pq = FALSE,
  keep_temporary_files = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="lulu_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_nproc">nproc</code></td>
<td>
<p>(default 1)
Set to number of cpus/processors to use for the clustering</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_id">id</code></td>
<td>
<p>(default: 0.84) id for &ndash;usearch_global.</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_vsearchpath">vsearchpath</code></td>
<td>
<p>(default: vsearch) path to vsearch.</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical) if true, print some additional messages.</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical) if true, empty samples and empty ASV are discarded
before clustering.</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files</p>
</td></tr>
<tr><td><code id="lulu_pq_+3A_...">...</code></td>
<td>
<p>Others args for function <code><a href="#topic+lulu">lulu()</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>The version of LULU is a fork of Adrien Taudière (<a href="https://github.com/adrientaudiere/lulu">https://github.com/adrientaudiere/lulu</a>)
from <a href="https://github.com/tobiasgf/lulu">https://github.com/tobiasgf/lulu</a>
</p>


<h3>Value</h3>

<p>a list of for object
</p>

<ul>
<li><p> &quot;new_physeq&quot;: The new phyloseq object (class physeq)
</p>
</li>
<li><p> &quot;discrepancy_vector&quot;: A vector of discrepancy showing for each taxonomic
level the proportion of identic value before and after lulu reclustering.
A value of 0.6 stands for 60% of ASV before re-clustering have
identical value after re-clustering. In other word, 40% of ASV are assigned
to a different taxonomic
value. NA value are not counted as discrepancy.
</p>
</li>
<li><p> &quot;res_lulu&quot;: A list of the result from the lulu function
</p>
</li>
<li><p> &quot;merged_ASV&quot;: the data.frame used to merged ASV
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Tobias Guldberg Frøslev <a href="mailto:tobiasgf@snm.ku.dk">tobiasgf@snm.ku.dk</a>
&amp; Adrien Taudière <a href="mailto:adrien.taudiere@zaclys.net">adrien.taudiere@zaclys.net</a>
</p>


<h3>References</h3>


<ul>
<li><p> LULU : <a href="https://github.com/adrientaudiere/lulu">https://github.com/adrientaudiere/lulu</a>
forked from <a href="https://github.com/tobiasgf/lulu">https://github.com/tobiasgf/lulu</a>.
</p>
</li>
<li><p> VSEARCH can be downloaded from
<a href="https://github.com/torognes/vsearch">https://github.com/torognes/vsearch</a>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>

lulu_pq(data_fungi_sp_known)


</code></pre>

<hr>
<h2 id='merge_krona'>Merge Krona files using <a href="https://github.com/marbl/Krona/wiki">KronaTools</a>.</h2><span id='topic+merge_krona'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Need the installation of kronatools on the computer
(<a href="https://github.com/marbl/Krona/wiki/Installing">installation instruction</a>).
Function merge_krona allows merging multiple html files in one interactive
krona file
</p>
<p>Note that you need to use the name args in <code>krona()</code> function before <code>merge_krona()</code>
in order to give good name to each krona pie in the output.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_krona(files = NULL, output = "mergeKrona.html")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge_krona_+3A_files">files</code></td>
<td>
<p>(required) path to html files to merged</p>
</td></tr>
<tr><td><code id="merge_krona_+3A_output">output</code></td>
<td>
<p>path to the output file</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/marbl/Krona">Krona</a> if
you use this function.
</p>


<h3>Value</h3>

<p>A html file
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+krona">krona</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
data("GlobalPatterns", package = "phyloseq")
GA &lt;- subset_taxa(GlobalPatterns, Phylum == "Acidobacteria")
krona(GA, "Number.of.sequences.html", name = "Nb_seq_GP_acidobacteria")
krona(GA, "Number.of.ASVs.html", nb_seq = FALSE, name = "Nb_asv_GP_acidobacteria")
merge_krona(c("Number.of.sequences.html", "Number.of.ASVs.html"), "mergeKrona.html")
unlink(c("Number.of.sequences.html", "Number.of.ASVs.html", "mergeKrona.html"))

## End(Not run)

</code></pre>

<hr>
<h2 id='merge_samples2'>Merge samples by a sample variable or factor</h2><span id='topic+merge_samples2'></span><span id='topic+merge_samples2+2Cphyloseq-method'></span><span id='topic+merge_samples2+2Cotu_table-method'></span><span id='topic+merge_samples2+2Csample_data-method'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Firstly release in the <a href="https://github.com/mikemc/speedyseq/">speedyseq</a> R
package by Michael R. McLaren.
</p>
<p>This function provides an alternative to <code>phyloseq::merge_samples()</code> that
better handles sample variables of different types, especially categorical
sample variables. It combines the samples in <code>x</code> defined by the sample
variable or factor <code>group</code> by summing the abundances in <code>otu_table(x)</code> and
combines sample variables by the summary functions in <code>funs</code>. The default
summary function, <code>unique_or_na()</code>, collapses the values within a group to a
single unique value if it exists and otherwise returns NA. The new (merged)
samples are named by the values in <code>group</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_samples2(x, group, fun_otu = sum, funs = list(), reorder = FALSE)

## S4 method for signature 'phyloseq'
merge_samples2(x, group, fun_otu = sum, funs = list(), reorder = FALSE)

## S4 method for signature 'otu_table'
merge_samples2(x, group, fun_otu = sum, reorder = FALSE)

## S4 method for signature 'sample_data'
merge_samples2(x, group, funs = list(), reorder = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge_samples2_+3A_x">x</code></td>
<td>
<p>A <code>phyloseq</code>, <code>otu_table</code>, or <code>sample_data</code> object</p>
</td></tr>
<tr><td><code id="merge_samples2_+3A_group">group</code></td>
<td>
<p>A sample variable or a vector of length <code>nsamples(x)</code> defining
the sample grouping. A vector must be supplied if x is an otu_table</p>
</td></tr>
<tr><td><code id="merge_samples2_+3A_fun_otu">fun_otu</code></td>
<td>
<p>Function for combining abundances in the otu_table; default
is <code>sum</code>. Can be a formula to be converted to a function by
<code><a href="purrr.html#topic+as_mapper">purrr::as_mapper()</a></code></p>
</td></tr>
<tr><td><code id="merge_samples2_+3A_funs">funs</code></td>
<td>
<p>Named list of merge functions for sample variables; default is
<code>unique_or_na</code></p>
</td></tr>
<tr><td><code id="merge_samples2_+3A_reorder">reorder</code></td>
<td>
<p>Logical specifying whether to reorder the new (merged)
samples by name</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new phyloseq-class, otu_table or sam_data object depending on
the class of the x param
</p>


<h3>Author(s)</h3>

<p>Michael R. McLaren (orcid: <a href="https://orcid.org/0000-0003-1575-473X">0000-0003-1575-473X</a>) modified by Adrien Taudiere
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(enterotype)

# Merge samples with the same project and clinical status
ps &lt;- enterotype
sample_data(ps) &lt;- sample_data(ps) %&gt;%
  transform(Project.ClinicalStatus = Project:ClinicalStatus)
sample_data(ps) %&gt;% head()
ps0 &lt;- merge_samples2(ps, "Project.ClinicalStatus",
  fun_otu = mean,
  funs = list(Age = mean)
)
sample_data(ps0) %&gt;% head()
</code></pre>

<hr>
<h2 id='merge_taxa_vec'>Merge taxa in groups (vectorized version)</h2><span id='topic+merge_taxa_vec'></span><span id='topic+merge_taxa_vec+2Cphyloseq-method'></span><span id='topic+merge_taxa_vec+2Cotu_table-method'></span><span id='topic+merge_taxa_vec+2CtaxonomyTable-method'></span><span id='topic+merge_taxa_vec+2Cphylo-method'></span><span id='topic+merge_taxa_vec+2CXStringSet-method'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Firstly release in the <a href="https://github.com/mikemc/speedyseq/">speedyseq</a> R
package by Michael R. McLaren.
</p>
<p>Merge taxa in <code>x</code> into a smaller set of taxa defined by the vector <code>group</code>.
Taxa whose value in <code>group</code> is NA will be dropped. New taxa will be named
according to the most abundant taxon in each group (<code>phyloseq</code> and
<code>otu_table</code> objects) or the first taxon in each group (all other phyloseq
component objects).
</p>
<p>If <code>x</code> is a phyloseq object with a phylogenetic tree, then the new taxa will
be ordered as they are in the tree. Otherwise, the taxa order can be
controlled by the <code>reorder</code> argument, which behaves like the <code>reorder</code>
argument in <code><a href="base.html#topic+rowsum">base::rowsum()</a></code>. <code>reorder = FALSE</code> will keep taxa in
the original order determined by when the member of each group first appears
in <code>taxa_names(x)</code>; <code>reorder = TRUE</code> will order new taxa according to their
corresponding value in <code>group</code>.
</p>
<p>The <code>tax_adjust</code> argument controls the handling of taxonomic disagreements
within groups. Setting <code>tax_adjust == 0</code> causes no adjustment; the taxonomy
of the new group is set to the archetype taxon (see below). Otherwise,
disagreements within a group at a given rank cause the values at lower ranks
to be set to <code>NA</code>. If <code>tax_adjust == 1</code> (the default), then a rank where all
taxa in the group are already NA is not counted as a disagreement, and lower
ranks may be kept if the taxa agree. This corresponds to the original
phyloseq behavior. If <code>tax_adjust == 2</code>, then these NAs are treated as a
disagreement; all ranks are set to NA after the first disagreement or NA.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>merge_taxa_vec(x, group, reorder = FALSE, tax_adjust = 1L)

## S4 method for signature 'phyloseq'
merge_taxa_vec(x, group, reorder = FALSE, tax_adjust = 1L)

## S4 method for signature 'otu_table'
merge_taxa_vec(x, group, reorder = FALSE)

## S4 method for signature 'taxonomyTable'
merge_taxa_vec(x, group, reorder = FALSE, tax_adjust = 1L)

## S4 method for signature 'phylo'
merge_taxa_vec(x, group)

## S4 method for signature 'XStringSet'
merge_taxa_vec(x, group, reorder = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="merge_taxa_vec_+3A_x">x</code></td>
<td>
<p>A phyloseq object or component object</p>
</td></tr>
<tr><td><code id="merge_taxa_vec_+3A_group">group</code></td>
<td>
<p>A vector with one element for each taxon in <code>physeq</code> that
defines the new groups. see <code>base::rowsum()</code>.</p>
</td></tr>
<tr><td><code id="merge_taxa_vec_+3A_reorder">reorder</code></td>
<td>
<p>Logical specifying whether to reorder the taxa by their
<code>group</code> values. Ignored if <code>x</code> has (or is) a phylogenetic tree.</p>
</td></tr>
<tr><td><code id="merge_taxa_vec_+3A_tax_adjust">tax_adjust</code></td>
<td>
<p>0: no adjustment; 1: phyloseq-compatible adjustment; 2:
conservative adjustment</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new phyloseq-class, otu_table, tax_table, XStringset or
sam_data object depending on the class of the x param
</p>


<h3>Author(s)</h3>

<p>Michael R. McLaren (orcid: <a href="https://orcid.org/0000-0003-1575-473X">0000-0003-1575-473X</a>) modified by Adrien Taudiere
</p>


<h3>See Also</h3>

<p>Function in MiscMetabar that use this function: <code><a href="#topic+asv2otu">asv2otu()</a></code>
</p>
<p><code><a href="base.html#topic+rowsum">base::rowsum()</a></code>
</p>
<p><code><a href="phyloseq.html#topic+merge_taxa-methods">phyloseq::merge_taxa()</a></code>
</p>

<hr>
<h2 id='MiscMetabar-deprecated'>Deprecated function(s) in the MiscMetabar package</h2><span id='topic+MiscMetabar-deprecated'></span><span id='topic+physeq_graph_test'></span><span id='topic+adonis_phyloseq'></span><span id='topic+clean_physeq'></span><span id='topic+lulu_phyloseq'></span><span id='topic+otu_circle'></span><span id='topic+biplot_physeq'></span><span id='topic+read_phyloseq'></span><span id='topic+write_phyloseq'></span><span id='topic+sankey_phyloseq'></span><span id='topic+summary_plot_phyloseq'></span><span id='topic+plot_deseq2_phyloseq'></span><span id='topic+plot_edgeR_phyloseq'></span><span id='topic+venn_phyloseq'></span><span id='topic+ggVenn_phyloseq'></span><span id='topic+hill_tuckey_phyloseq'></span><span id='topic+hill_phyloseq'></span><span id='topic+physeq_heat_tree'></span>

<h3>Description</h3>

<p>These functions are provided for compatibility with older version of
the MiscMetabar package. They may eventually be completely
removed.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>physeq_graph_test(...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MiscMetabar-deprecated_+3A_...">...</code></td>
<td>
<p>Parameters to be passed on to the modern version of the function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Depend on the functions.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: right;">
<a href="#topic+graph_test_pq">graph_test_pq</a> </td><td style="text-align: left;"> now a synonym for <code>physeq_graph_test</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+adonis_pq">adonis_pq</a> </td><td style="text-align: left;"> now a synonym for <code>adonis_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+clean_pq">clean_pq</a> </td><td style="text-align: left;"> now a synonym for <code>clean_physeq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+lulu_pq">lulu_pq</a> </td><td style="text-align: left;"> now a synonym for <code>lulu_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+circle_pq">circle_pq</a> </td><td style="text-align: left;"> now a synonym for <code>otu_circle</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+biplot_pq">biplot_pq</a> </td><td style="text-align: left;"> now a synonym for <code>biplot_physeq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+read_pq">read_pq</a> </td><td style="text-align: left;"> now a synonym for <code>read_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+write_pq">write_pq</a> </td><td style="text-align: left;"> now a synonym for <code>write_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+sankey_pq">sankey_pq</a> </td><td style="text-align: left;"> now a synonym for <code>sankey_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+summary_plot_pq">summary_plot_pq</a> </td><td style="text-align: left;"> now a synonym for <code>summary_plot_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+plot_edgeR_pq">plot_edgeR_pq</a> </td><td style="text-align: left;"> now a synonym for <code>plot_edgeR_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+plot_deseq2_pq">plot_deseq2_pq</a> </td><td style="text-align: left;"> now a synonym for <code>plot_deseq2_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+venn_pq">venn_pq</a> </td><td style="text-align: left;"> now a synonym for <code>venn_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+ggvenn_pq">ggvenn_pq</a> </td><td style="text-align: left;"> now a synonym for <code>ggVenn_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+hill_tuckey_pq">hill_tuckey_pq</a> </td><td style="text-align: left;"> now a synonym for <code>hill_tuckey_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+hill_pq">hill_pq</a> </td><td style="text-align: left;"> now a synonym for <code>hill_phyloseq</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+heat_tree_pq">heat_tree_pq</a> </td><td style="text-align: left;"> now a synonym for <code>physeq_heat_tree</code></td>
</tr>
<tr>
 <td style="text-align: right;">
<a href="#topic+compare_pairs_pq">compare_pairs_pq</a> </td><td style="text-align: left;"> now a synonym for <code>multiple_share_bisamples</code></td>
</tr>
<tr>
 <td style="text-align: right;">
</td>
</tr>

</table>


<hr>
<h2 id='multi_biplot_pq'>Visualization of a collection of couples of samples for comparison</h2><span id='topic+multi_biplot_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This allow to plot all the possible <code><a href="#topic+biplot_pq">biplot_pq()</a></code> combination
using one factor.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multi_biplot_pq(physeq, split_by = NULL, pairs = NULL, na_remove = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multi_biplot_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="multi_biplot_pq_+3A_split_by">split_by</code></td>
<td>
<p>(required if pairs is NULL) the name of the factor to make all combination
of couples of values</p>
</td></tr>
<tr><td><code id="multi_biplot_pq_+3A_pairs">pairs</code></td>
<td>
<p>(required if pairs is NULL) the name of the factor in physeq@sam_data' slot
to make plot by pairs of samples. Each level must be present only two times.
Note that if you set pairs, you also must set fact arguments to pass on to <code><a href="#topic+biplot_pq">biplot_pq()</a></code>.</p>
</td></tr>
<tr><td><code id="multi_biplot_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default TRUE) if TRUE remove all the samples
with NA in the <code>split_by</code> variable of the <code>physeq@sam_data</code> slot</p>
</td></tr>
<tr><td><code id="multi_biplot_pq_+3A_...">...</code></td>
<td>
<p>Other parameters passed on to <code><a href="#topic+biplot_pq">biplot_pq()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of ggplot object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data_fungi_abun &lt;- subset_taxa_pq(data_fungi, taxa_sums(data_fungi) &gt; 10000)
p &lt;- multi_biplot_pq(data_fungi_abun, "Height")
lapply(p, print)

</code></pre>

<hr>
<h2 id='multipatt_pq'>Test and plot multipatt result</h2><span id='topic+multipatt_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="indicspecies.html#topic+multipatt">indicspecies::multipatt()</a></code> function in the case of
<code>physeq</code> object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multipatt_pq(
  physeq,
  fact,
  p_adjust_method = "BH",
  pval = 0.05,
  control = permute::how(nperm = 999),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multipatt_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="multipatt_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot
different lines</p>
</td></tr>
<tr><td><code id="multipatt_pq_+3A_p_adjust_method">p_adjust_method</code></td>
<td>
<p>(chr, default &quot;BH&quot;): the method used to adjust p-value</p>
</td></tr>
<tr><td><code id="multipatt_pq_+3A_pval">pval</code></td>
<td>
<p>(int, default 0.05): the value to determine the significance of
LCBD</p>
</td></tr>
<tr><td><code id="multipatt_pq_+3A_control">control</code></td>
<td>
<p>see <code>?indicspecies::multipatt()</code></p>
</td></tr>
<tr><td><code id="multipatt_pq_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="indicspecies.html#topic+multipatt">indicspecies::multipatt()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>indicspecies::multipatt()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("indicspecies")) {
  data(data_fungi)
  data_fungi_ab &lt;- subset_taxa_pq(data_fungi, taxa_sums(data_fungi) &gt; 10000)
  multipatt_pq(subset_samples(data_fungi_ab, !is.na(Time)), fact = "Time")
}

if (requireNamespace("indicspecies")) {
  multipatt_pq(subset_samples(data_fungi_ab, !is.na(Time)),
    fact = "Time",
    max.order = 1, control = permute::how(nperm = 99)
  )
}


</code></pre>

<hr>
<h2 id='multiplot'>Multiple plot function</h2><span id='topic+multiplot'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multiplot(..., plotlist = NULL, cols = 1, layout = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multiplot_+3A_...">...</code></td>
<td>
<p>list of ggplot objects</p>
</td></tr>
<tr><td><code id="multiplot_+3A_plotlist">plotlist</code></td>
<td>
<p>list of ggplot objects</p>
</td></tr>
<tr><td><code id="multiplot_+3A_cols">cols</code></td>
<td>
<p>number of columns</p>
</td></tr>
<tr><td><code id="multiplot_+3A_layout">layout</code></td>
<td>
<p>A matrix specifying the layout.
If present, 'cols' is ignored.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing. Print the list of ggplot objects
</p>

<hr>
<h2 id='multitax_bar_pq'>Plot taxonomic distribution across 3 taxonomic levels and optionally
one sample factor</h2><span id='topic+multitax_bar_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Note that lvl3 need to be nested in lvl2 which need to be nested
in lvl1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>multitax_bar_pq(
  physeq,
  lvl1,
  lvl2,
  lvl3,
  fact = NULL,
  nb_seq = TRUE,
  log10trans = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="multitax_bar_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="multitax_bar_pq_+3A_lvl1">lvl1</code></td>
<td>
<p>(required) Name of the first (higher) taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="multitax_bar_pq_+3A_lvl2">lvl2</code></td>
<td>
<p>(required) Name of the second (middle) taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="multitax_bar_pq_+3A_lvl3">lvl3</code></td>
<td>
<p>(required) Name of the first (lower) taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="multitax_bar_pq_+3A_fact">fact</code></td>
<td>
<p>Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>. If not set, the taxonomic
distribution is plot for all samples together.</p>
</td></tr>
<tr><td><code id="multitax_bar_pq_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(logical; default TRUE) If set to FALSE, only the number of ASV
is count. Concretely, physeq otu_table is transformed in a binary
otu_table (each value different from zero is set to one)</p>
</td></tr>
<tr><td><code id="multitax_bar_pq_+3A_log10trans">log10trans</code></td>
<td>
<p>(logical, default TRUE) If TRUE,
the number of sequences (or ASV if nb_seq = FALSE) is log10
transformed.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 graphic
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("ggh4x")) {
  multitax_bar_pq(data_fungi_sp_known, "Phylum", "Class", "Order", "Time")
  multitax_bar_pq(data_fungi_sp_known, "Phylum", "Class", "Order")
  multitax_bar_pq(data_fungi_sp_known, "Phylum", "Class", "Order",
    nb_seq = FALSE, log10trans = FALSE
  )
}

</code></pre>

<hr>
<h2 id='mumu_pq'>MUMU reclustering of class <code>physeq</code></h2><span id='topic+mumu_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>See https://www.nature.com/articles/s41467-017-01312-x for more information
on the original method LULU. This is a wrapper of
<a href="https://github.com/frederic-mahe/mumu">mumu</a> a C++ re-implementation
of LULU by Frédéric Mahé
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mumu_pq(
  physeq,
  nproc = 1,
  id = 0.84,
  vsearchpath = "vsearch",
  mumupath = "mumu",
  verbose = FALSE,
  clean_pq = TRUE,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mumu_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_nproc">nproc</code></td>
<td>
<p>(default 1)
Set to number of cpus/processors to use for the clustering</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_id">id</code></td>
<td>
<p>(default: 0.84) id for &ndash;usearch_global.</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_vsearchpath">vsearchpath</code></td>
<td>
<p>(default: vsearch) path to vsearch.</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_mumupath">mumupath</code></td>
<td>
<p>path to mumu. See <a href="https://github.com/frederic-mahe/mumu">mumu</a>
for installation instruction</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical) if true, print some additional messages.</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical) if true, empty samples and empty ASV are discarded
before clustering.</p>
</td></tr>
<tr><td><code id="mumu_pq_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/frederic-mahe/mumu/blob/main/CITATION.cff">mumu</a> and
<a href="https://www.nature.com/articles/s41467-017-01312-x">lulu</a> if you use this function
for your work.
</p>


<h3>Value</h3>

<p>a list of for object
</p>

<ul>
<li><p> &quot;new_physeq&quot;: The new phyloseq object (class physeq)
</p>
</li>
<li><p> &quot;mumu_results&quot;: The log file of the mumu software. Run <code style="white-space: pre;">&#8288;man mumu&#8288;</code> into
bash to obtain details about columns' signification.
</p>
</li></ul>



<h3>Author(s)</h3>

<p>Frédéric Mahé
&amp; Adrien Taudière <a href="mailto:adrien.taudiere@zaclys.net">adrien.taudiere@zaclys.net</a>
</p>


<h3>References</h3>


<ul>
<li><p> MUMU: <a href="https://github.com/frederic-mahe/mumu">https://github.com/frederic-mahe/mumu</a>
</p>
</li>
<li><p> VSEARCH can be downloaded from
<a href="https://github.com/torognes/vsearch">https://github.com/torognes/vsearch</a>.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>
## Not run: 
mumu_pq(data_fungi_sp_known)

## End(Not run)

</code></pre>

<hr>
<h2 id='normalize_prop_pq'>Normalize OTU table using samples depth</h2><span id='topic+normalize_prop_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This function implement the method proposed by
McKnight et al. 2018 (<a href="https://doi.org/10.5061/dryad.tn8qs35">doi:10.5061/dryad.tn8qs35</a>)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>normalize_prop_pq(physeq, base_log = 2, constante = 10000, digits = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="normalize_prop_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="normalize_prop_pq_+3A_base_log">base_log</code></td>
<td>
<p>(integer, default 2) the base for log-transformation. If
set to NULL or NA, no log-transformation is compute after normalization.</p>
</td></tr>
<tr><td><code id="normalize_prop_pq_+3A_constante">constante</code></td>
<td>
<p>a constante to multiply the otu_table values</p>
</td></tr>
<tr><td><code id="normalize_prop_pq_+3A_digits">digits</code></td>
<td>
<p>(default = 2) integer indicating the number of decimal places
to be used (see <code>?round</code> for more information)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object with otu_table count
normalize and log transformed (if base_log is an integer)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>taxa_sums(data_fungi_mini)
data_f_norm &lt;- normalize_prop_pq(data_fungi_mini)
taxa_sums(data_f_norm)
ggplot(data.frame(
  "norm" = scale(taxa_sums(data_f_norm)),
  "raw" = scale(taxa_sums(data_fungi_mini)),
  "name_otu" = taxa_names(data_f_norm)
)) +
  geom_point(aes(x = raw, y = norm))

data_f_norm &lt;- normalize_prop_pq(data_fungi_mini, base_log = NULL)
</code></pre>

<hr>
<h2 id='perc'>Convert a value (or a fraction x/y) in percentage</h2><span id='topic+perc'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>perc(x, y = NULL, accuracy = 0, add_symbol = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="perc_+3A_x">x</code></td>
<td>
<p>(required): value</p>
</td></tr>
<tr><td><code id="perc_+3A_y">y</code></td>
<td>
<p>if y is set, compute the division of x by y</p>
</td></tr>
<tr><td><code id="perc_+3A_accuracy">accuracy</code></td>
<td>
<p>number of digits (number of digits after zero)</p>
</td></tr>
<tr><td><code id="perc_+3A_add_symbol">add_symbol</code></td>
<td>
<p>if set to TRUE add the % symbol to the value</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The percentage value (number or character if add_symbol
is set to TRUE)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='phyloseq_to_edgeR'>Convert phyloseq OTU count data into DGEList for edgeR package</h2><span id='topic+phyloseq_to_edgeR'></span>

<h3>Description</h3>

<p>Convert phyloseq OTU count data into DGEList for edgeR package
</p>


<h3>Usage</h3>

<pre><code class='language-R'>phyloseq_to_edgeR(physeq, group, method = "RLE", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="phyloseq_to_edgeR_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="phyloseq_to_edgeR_+3A_group">group</code></td>
<td>
<p>(required) A character vector or factor giving the experimental
group/condition for each sample/library. Alternatively, you may provide
the name of a sample variable. This name should be among the output of
<code>sample_variables(physeq)</code>, in which case
<code>get_variable(physeq, group)</code> would return either a character vector or
factor.
This is passed on to <code><a href="edgeR.html#topic+DGEList">DGEList</a></code>,
and you may find further details or examples in its documentation.</p>
</td></tr>
<tr><td><code id="phyloseq_to_edgeR_+3A_method">method</code></td>
<td>
<p>The label of the edgeR-implemented normalization
to use.
See <code><a href="edgeR.html#topic+calcNormFactors">calcNormFactors</a></code> for supported options and details.
The default option is <code>"RLE"</code>, which is a scaling factor method
proposed by Anders and Huber (2010).
At time of writing, the <a href="edgeR.html#topic+edgeR">edgeR</a> package supported
the following options to the <code>method</code> argument:
</p>
<p><code>c("TMM", "RLE", "upperquartile", "none")</code>.</p>
</td></tr>
<tr><td><code id="phyloseq_to_edgeR_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code><a href="edgeR.html#topic+DGEList">DGEList</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A DGEList object. See <code><a href="edgeR.html#topic+estimateTagwiseDisp">edgeR::estimateTagwiseDisp()</a></code> for more details.
</p>

<hr>
<h2 id='physeq_or_string_to_dna'>Return a DNAStringSet object from either a character vector of DNA sequences
or the <code>refseq</code> slot of a phyloseq-class object</h2><span id='topic+physeq_or_string_to_dna'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#stable"><img src="../help/figures/lifecycle-stable.svg" alt='[Stable]' /></a>
</p>
<p>Internally used in <code><a href="#topic+vsearch_clustering">vsearch_clustering()</a></code>, <code><a href="#topic+swarm_clustering">swarm_clustering()</a></code> and
<code><a href="#topic+asv2otu">asv2otu()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>physeq_or_string_to_dna(physeq = NULL, dna_seq = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="physeq_or_string_to_dna_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="physeq_or_string_to_dna_+3A_dna_seq">dna_seq</code></td>
<td>
<p>You may directly use a character vector of DNA sequences
in place of physeq args. When physeq is set, dna sequences take the value
of <code>physeq@refseq</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>An object of class DNAStringSet (see the <code><a href="Biostrings.html#topic+XStringSet-class">Biostrings::DNAStringSet()</a></code>
function)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="Biostrings.html#topic+XStringSet-class">Biostrings::DNAStringSet()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
dna &lt;- physeq_or_string_to_dna(data_fungi)
dna

sequences_ex &lt;- c(
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTAATAACGAATTCATTGAATCA",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTACCTGGTAAGGCCCACTT",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTACCTGGTAGAGGTG",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACC",
  "CGGGATTTGATGGCGAATTACCTGGTATTTTAGCCCACTTACCCGGTACCATGAGGTG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACCTGG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG"
)
dna2 &lt;- physeq_or_string_to_dna(dna_seq = sequences_ex)
dna2

</code></pre>

<hr>
<h2 id='plot_ancombc_pq'>Plot ANCOMBC2 result for phyloseq object</h2><span id='topic+plot_ancombc_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_ancombc_pq(
  physeq,
  ancombc_res,
  filter_passed = TRUE,
  filter_diff = TRUE,
  min_abs_lfc = 0,
  tax_col = "Genus",
  tax_label = "Species",
  add_marginal_vioplot = TRUE,
  add_label = TRUE,
  add_hline_cut_lfc = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_ancombc_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_ancombc_res">ancombc_res</code></td>
<td>
<p>(required) the result of the ancombc_pq function
For the moment only bimodal factors are possible.</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_filter_passed">filter_passed</code></td>
<td>
<p>(logical, default TRUE) Do we filter using the column
passed_ss? The passed_ss value is  TRUE if the taxon passed the sensitivity
analysis, i.e., adding different pseudo-counts to 0s would not change the results.</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_filter_diff">filter_diff</code></td>
<td>
<p>(logical, default TRUE) Do we filter using the column
diff? The diff value is TRUE if the taxon is significant
(has q less than alpha)</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_min_abs_lfc">min_abs_lfc</code></td>
<td>
<p>(integer, default 0) Minimum absolute value to filter
results based on Log Fold Change. For ex. a value of 1 filter out taxa
for which the abundance in a given level of the modality is not at least
the double of the abundance in the other level.</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_tax_col">tax_col</code></td>
<td>
<p>The taxonomic level (must be present in <code>tax_table</code> slot)
to color the points</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_tax_label">tax_label</code></td>
<td>
<p>The taxonomic level (must be present in <code>tax_table</code> slot)
to add label</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_add_marginal_vioplot">add_marginal_vioplot</code></td>
<td>
<p>(logical, default TRUE) Do we add a marginal
vioplot representing all the taxa lfc from ancombc_res.</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_add_label">add_label</code></td>
<td>
<p>(logical, default TRUE) Do we add a label?</p>
</td></tr>
<tr><td><code id="plot_ancombc_pq_+3A_add_hline_cut_lfc">add_hline_cut_lfc</code></td>
<td>
<p>(logical, default NULL) Do we add two horizontal
lines when min_abs_lfc is set (different from zero)?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>ANCOMBC::ancombc2()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A ggplot2 object. If add_marginal_vioplot is TRUE, this is a
patchworks of plot made using <code>patchwork::plot_layout()</code>.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("mia")) {
  data_fungi_mini@tax_table &lt;- phyloseq::tax_table(cbind(
    data_fungi_mini@tax_table,
    "taxon" = taxa_names(data_fungi_mini)
  ))

  res_time &lt;- ancombc_pq(
    data_fungi_mini,
    fact = "Time",
    levels_fact = c("0", "15"),
    tax_level = "taxon",
    verbose = TRUE
  )

  plot_ancombc_pq(data_fungi_mini, res_time,
    filter_passed = FALSE,
    tax_label = "Genus", tax_col = "Order"
  )
  plot_ancombc_pq(data_fungi_mini, res_time, tax_col = "Genus")
  plot_ancombc_pq(data_fungi_mini, res_time,
    filter_passed = FALSE,
    filter_diff = FALSE, tax_col = "Family", add_label = FALSE
  )
}

</code></pre>

<hr>
<h2 id='plot_deseq2_pq'>Plot DESeq2 results for a phyloseq or a DESeq2 object.</h2><span id='topic+plot_deseq2_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_deseq2_pq(
  data,
  contrast = NULL,
  tax_table = NULL,
  pval = 0.05,
  taxolev = "Genus",
  select_taxa = NULL,
  color_tax = "Phylum",
  tax_depth = NULL,
  verbose = TRUE,
  jitter_width = 0.1,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_deseq2_pq_+3A_data">data</code></td>
<td>
<p>(required) a <code>phyloseq-class</code> or a
<code><a href="DESeq2.html#topic+DESeqDataSet-class">DESeqDataSet-class</a></code> object.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_contrast">contrast</code></td>
<td>
<p>(required) contrast specifies what comparison to extract
from the object to build a results table. See <code><a href="DESeq2.html#topic+results">results</a></code>
man page for more details.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_tax_table">tax_table</code></td>
<td>
<p>Required if data is a
<code><a href="DESeq2.html#topic+DESeqDataSet-class">DESeqDataSet-class</a></code> object.
The taxonomic table used to find the <code>taxa</code> and <code>color_taxa</code>
arguments. If data is a <code>phyloseq-class</code> object, data@tax_table
is used.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_pval">pval</code></td>
<td>
<p>(default: 0.05) the significance cutoff used for optimizing
the independent filtering. If the adjusted p-value cutoff (FDR) will be a
value other than 0.05, pval should be set to that value.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_taxolev">taxolev</code></td>
<td>
<p>taxonomic level of interest</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_select_taxa">select_taxa</code></td>
<td>
<p>Either the name of the taxa (in the form of <code><a href="DESeq2.html#topic+results">DESeq2::results()</a></code>)
or a logical vector (length of the results from <code><a href="DESeq2.html#topic+results">DESeq2::results()</a></code>) to select taxa
to plot.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_color_tax">color_tax</code></td>
<td>
<p>taxonomic level used for color or a
color vector.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_tax_depth">tax_depth</code></td>
<td>
<p>Taxonomic depth to test for differential
distribution among contrast. If Null the analysis is done at the OTU
(i.e. Species) level. If not Null, data need to be a column name in
the <code>tax_table</code> slot of the <code>phyloseq-class</code> object.</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_verbose">verbose</code></td>
<td>
<p>whether the function print some information during
the computation</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_jitter_width">jitter_width</code></td>
<td>
<p>width for the jitter positioning</p>
</td></tr>
<tr><td><code id="plot_deseq2_pq_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code><a href="DESeq2.html#topic+DESeq">DESeq</a></code>
or <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Please cite <code>DESeq2</code> package if you use chis function.
</p>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot representing DESeq2 results
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="DESeq2.html#topic+DESeq">DESeq</a></code>
</p>
<p><code><a href="DESeq2.html#topic+results">results</a></code>
</p>
<p><code><a href="#topic+plot_edgeR_pq">plot_edgeR_pq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

data("GlobalPatterns", package = "phyloseq")
GP &lt;- subset_taxa(GlobalPatterns, GlobalPatterns@tax_table[, 1] == "Archaea")
GP &lt;- subset_samples(GP, SampleType %in% c("Soil", "Skin"))
if (requireNamespace("DESeq2")) {
  res &lt;- DESeq2::DESeq(phyloseq_to_deseq2(GP, ~SampleType),
    test = "Wald", fitType = "local"
  )
  plot_deseq2_pq(res, c("SampleType", "Soil", "Skin"),
    tax_table = GP@tax_table, color_tax = "Kingdom"
  )
  plot_deseq2_pq(res, c("SampleType", "Soil", "Skin"),
    tax_table = GP@tax_table, color_tax = "Kingdom",
    pval = 0.7
  )
  plot_deseq2_pq(res, c("SampleType", "Soil", "Skin"),
    tax_table = GP@tax_table, color_tax = "Class",
    select_taxa = c("522457", "271582")
  )
}

</code></pre>

<hr>
<h2 id='plot_edgeR_pq'>Plot edgeR results for a phyloseq or a edgeR object.</h2><span id='topic+plot_edgeR_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_edgeR_pq(
  physeq,
  contrast = NULL,
  pval = 0.05,
  taxolev = "Genus",
  color_tax = "Phylum",
  verbose = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_edgeR_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_edgeR_pq_+3A_contrast">contrast</code></td>
<td>
<p>(required):This argument specifies what comparison
to extract from the object to build a results table.
See <code><a href="DESeq2.html#topic+results">results</a></code> man page for more details.</p>
</td></tr>
<tr><td><code id="plot_edgeR_pq_+3A_pval">pval</code></td>
<td>
<p>(default: 0.05): the significance cutoff used for optimizing
the independent filtering. If the adjusted p-value cutoff (FDR) will be a
value other than 0.05, pval should be set to that value.</p>
</td></tr>
<tr><td><code id="plot_edgeR_pq_+3A_taxolev">taxolev</code></td>
<td>
<p>taxonomic level of interest</p>
</td></tr>
<tr><td><code id="plot_edgeR_pq_+3A_color_tax">color_tax</code></td>
<td>
<p>taxonomic level used for
color assignation</p>
</td></tr>
<tr><td><code id="plot_edgeR_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical): whether the function print some
information during the computation</p>
</td></tr>
<tr><td><code id="plot_edgeR_pq_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to <code><a href="edgeR.html#topic+exactTest">exactTest</a></code>
or <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot representing edgeR results
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="edgeR.html#topic+exactTest">exactTest</a></code>
</p>
<p><code><a href="#topic+plot_deseq2_pq">plot_deseq2_pq</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("GlobalPatterns", package = "phyloseq")
GP_archae &lt;- subset_taxa(GlobalPatterns, GlobalPatterns@tax_table[, 1] == "Archaea")

if (requireNamespace("edgeR")) {
  plot_edgeR_pq(GP_archae, c("SampleType", "Soil", "Feces"),
    color_tax = "Kingdom"
  )

  plot_edgeR_pq(GP_archae, c("SampleType", "Soil", "Feces"),
    taxolev = "Class", color_tax = "Kingdom"
  )
}


</code></pre>

<hr>
<h2 id='plot_guild_pq'>Plot information about Guild from tax_table slot previously
created with <code><a href="#topic+add_funguild_info">add_funguild_info()</a></code></h2><span id='topic+plot_guild_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_guild_pq(physeq, levels_order = NULL, clean_pq = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_guild_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_guild_pq_+3A_levels_order">levels_order</code></td>
<td>
<p>(Default NULL) A character vector to
reorder the levels of guild. See examples.</p>
</td></tr>
<tr><td><code id="plot_guild_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical, default TRUE): Does the phyloseq
object is cleaned using the <code><a href="#topic+clean_pq">clean_pq()</a></code> function?</p>
</td></tr>
<tr><td><code id="plot_guild_pq_+3A_...">...</code></td>
<td>
<p>Other params for be passed on to
<code><a href="#topic+clean_pq">clean_pq()</a></code> function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+add_funguild_info">add_funguild_info()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("httr")) {
  d_fung_mini &lt;- add_funguild_info(data_fungi_mini,
    taxLevels = c(
      "Domain",
      "Phylum",
      "Class",
      "Order",
      "Family",
      "Genus",
      "Species"
    )
  )
  sort(table(d_fung_mini@tax_table[, "guild"]), decreasing = TRUE)

  p &lt;- plot_guild_pq(d_fung_mini)
  if (requireNamespace("patchwork")) {
    (plot_guild_pq(subset_samples(d_fung_mini, Height == "Low"),
      levels_order = p$data$Guild[order(p$data$nb_seq)]
    ) + theme(legend.position = "none")) +
      (plot_guild_pq(subset_samples(d_fung_mini, Height == "High"),
        levels_order = p$data$Guild[order(p$data$nb_seq)]
      ) + ylab("") + theme(axis.text.y = element_blank()))
  }
}

</code></pre>

<hr>
<h2 id='plot_LCBD_pq'>Plot and test local contributions to beta diversity (LCBD) of samples</h2><span id='topic+plot_LCBD_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function in the case of <code>physeq</code>
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_LCBD_pq(
  physeq,
  p_adjust_method = "BH",
  pval = 0.05,
  sam_variables = NULL,
  only_plot_significant = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_LCBD_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_LCBD_pq_+3A_p_adjust_method">p_adjust_method</code></td>
<td>
<p>(chr, default &quot;BH&quot;): the method used to adjust p-value</p>
</td></tr>
<tr><td><code id="plot_LCBD_pq_+3A_pval">pval</code></td>
<td>
<p>(int, default 0.05): the value to determine the significance of
LCBD</p>
</td></tr>
<tr><td><code id="plot_LCBD_pq_+3A_sam_variables">sam_variables</code></td>
<td>
<p>A vector of variables names present in the <code>sam_data</code>
slot to plot alongside the LCBD value</p>
</td></tr>
<tr><td><code id="plot_LCBD_pq_+3A_only_plot_significant">only_plot_significant</code></td>
<td>
<p>(logical, default TRUE) Do we plot all LCBD
values or only the significant ones</p>
</td></tr>
<tr><td><code id="plot_LCBD_pq_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>vegan::beta.div()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A ggplot2 object build with the package patchwork
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><a href="#topic+LCBD_pq">LCBD_pq</a>, <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_fungi)
if (requireNamespace("adespatial")) {
  plot_LCBD_pq(data_fungi_mini,
    nperm = 100, only_plot_significant = FALSE,
    pval = 0.2
  )
}

if (requireNamespace("adespatial")) {
  plot_LCBD_pq(data_fungi_mini,
    nperm = 100, only_plot_significant = TRUE,
    pval = 0.2
  )
  if (requireNamespace("patchwork")) {
    plot_LCBD_pq(data_fungi_mini,
      nperm = 100, only_plot_significant = FALSE,
      sam_variables = c("Time", "Height")
    )
    plot_LCBD_pq(data_fungi_mini,
      nperm = 100, only_plot_significant = TRUE, pval = 0.2,
      sam_variables = c("Time", "Height", "Tree_name")
    ) &amp;
      theme(
        legend.key.size = unit(0.4, "cm"),
        legend.text = element_text(size = 10),
        axis.title.x = element_text(size = 6)
      )
  }
}

</code></pre>

<hr>
<h2 id='plot_mt'>Plot the result of a mt test <code><a href="phyloseq.html#topic+mt-methods">phyloseq::mt()</a></code></h2><span id='topic+plot_mt'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_mt(mt = NULL, alpha = 0.05, color_tax = "Class", taxa = "Species")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_mt_+3A_mt">mt</code></td>
<td>
<p>(required) Result of a mt test from the function <code><a href="phyloseq.html#topic+mt-methods">phyloseq::mt()</a></code>.</p>
</td></tr>
<tr><td><code id="plot_mt_+3A_alpha">alpha</code></td>
<td>
<p>(default: 0.05) Choose the cut off p-value to plot taxa.</p>
</td></tr>
<tr><td><code id="plot_mt_+3A_color_tax">color_tax</code></td>
<td>
<p>(default: &quot;Class&quot;) A taxonomic level to color the points.</p>
</td></tr>
<tr><td><code id="plot_mt_+3A_taxa">taxa</code></td>
<td>
<p>(default: &quot;Species&quot;) The taxonomic level you choose for x-positioning.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot of result of a mt test
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="phyloseq.html#topic+mt-methods">phyloseq::mt()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# Filter samples that don't have Time
data_fungi_mini2 &lt;- subset_samples(data_fungi_mini, !is.na(Time))
res &lt;- mt(data_fungi_mini2, "Time", method = "fdr", test = "f", B = 300)
plot_mt(res)
plot_mt(res, taxa = "Genus", color_tax = "Order")

</code></pre>

<hr>
<h2 id='plot_SCBD_pq'>Plot species contributions to beta diversity (SCBD) of samples</h2><span id='topic+plot_SCBD_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function in the case of <code>physeq</code>
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_SCBD_pq(
  physeq,
  tax_level = "ASV",
  tax_col = "Order",
  min_SCBD = 0.01,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_SCBD_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_SCBD_pq_+3A_tax_level">tax_level</code></td>
<td>
<p>Taxonomic level to used in y axis</p>
</td></tr>
<tr><td><code id="plot_SCBD_pq_+3A_tax_col">tax_col</code></td>
<td>
<p>Taxonomic level to colored points</p>
</td></tr>
<tr><td><code id="plot_SCBD_pq_+3A_min_scbd">min_SCBD</code></td>
<td>
<p>(default 0.01) the minimum SCBD value
to plot the taxa</p>
</td></tr>
<tr><td><code id="plot_SCBD_pq_+3A_...">...</code></td>
<td>
<p>Others arguments passed on to <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code> function</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>vegan::beta.div()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A ggplot2 object build with the package patchwork
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><a href="#topic+LCBD_pq">LCBD_pq</a>, <code><a href="adespatial.html#topic+beta.div">adespatial::beta.div()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_fungi)
if (requireNamespace("adespatial")) {
  plot_SCBD_pq(data_fungi) +
    geom_text(aes(label = paste(Genus, Species)), hjust = 1, vjust = 2) +
    xlim(c(0, NA))
}

if (requireNamespace("adespatial")) {
  plot_SCBD_pq(data_fungi, tax_level = "Class", tax_col = "Phylum", min_SCBD = 0) +
    geom_jitter()
}


</code></pre>

<hr>
<h2 id='plot_tax_pq'>Plot taxonomic distribution in function of a factor with stacked bar in %</h2><span id='topic+plot_tax_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>An alternative to <code>phyloseq::plot_bar()</code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_tax_pq(
  physeq,
  fact = NULL,
  merge_sample_by = NULL,
  type = "nb_seq",
  taxa_fill = "Order",
  print_values = TRUE,
  color_border = "lightgrey",
  linewidth = 0.1,
  prop_print_value = 0.01,
  nb_print_value = NULL,
  add_info = TRUE,
  na_remove = TRUE,
  clean_pq = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_tax_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_merge_sample_by">merge_sample_by</code></td>
<td>
<p>a vector to determine
which samples to merge using the
<code><a href="#topic+merge_samples2">merge_samples2()</a></code> function.
Need to be in <code>physeq@sam_data</code></p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_type">type</code></td>
<td>
<p>If &quot;nb_seq&quot; (default), the number of sequences is
used in plot. If &quot;nb_asv&quot;, the number of ASV is plotted. If both,
return a list of two plots, one for nbSeq and one for ASV.</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_taxa_fill">taxa_fill</code></td>
<td>
<p>(default: 'Order'): Name of the taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_print_values">print_values</code></td>
<td>
<p>(logical, default TRUE): Do we print some values on plot?</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_color_border">color_border</code></td>
<td>
<p>color for the border</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_linewidth">linewidth</code></td>
<td>
<p>The line width of geom_bar</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_prop_print_value">prop_print_value</code></td>
<td>
<p>minimal proportion to print value (default 0.01)</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_nb_print_value">nb_print_value</code></td>
<td>
<p>number of higher values to print
(replace prop_print_value if both are set).</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_add_info">add_info</code></td>
<td>
<p>(logical, default TRUE) Do we add title and subtitle with
information about the total number of sequences and the number of samples
per modality.</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default TRUE) if TRUE remove all the samples
with NA in the <code>split_by</code> variable of the <code>physeq@sam_data</code> slot</p>
</td></tr>
<tr><td><code id="plot_tax_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical)
If set to TRUE, empty samples are discarded after subsetting ASV</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 graphic
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+tax_bar_pq">tax_bar_pq()</a></code> and <code><a href="#topic+multitax_bar_pq">multitax_bar_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_fungi_sp_known)
plot_tax_pq(data_fungi_sp_known,
  "Time",
  merge_sample_by = "Time",
  taxa_fill = "Class"
)

plot_tax_pq(data_fungi_sp_known,
  "Height",
  merge_sample_by = "Height",
  taxa_fill = "Class",
  na_remove = TRUE,
  color_border = rgb(0, 0, 0, 0)
)

plot_tax_pq(data_fungi_sp_known,
  "Height",
  merge_sample_by = "Height",
  taxa_fill = "Class",
  na_remove = FALSE,
  clean_pq = FALSE
)

</code></pre>

<hr>
<h2 id='plot_tsne_pq'>Plot a tsne low dimensional representation of a phyloseq object</h2><span id='topic+plot_tsne_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_tsne_pq(
  physeq,
  method = "bray",
  dims = 2,
  theta = 0,
  perplexity = 30,
  fact = NA,
  ellipse_level = 0.95,
  plot_dims = c(1, 2),
  na_remove = TRUE,
  force_factor = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_tsne_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_method">method</code></td>
<td>
<p>A method to calculate distance using <code>vegan::vegdist()</code> function (default: &quot;bray&quot;)</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_dims">dims</code></td>
<td>
<p>(Int) Output dimensionality (default: 2)</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_theta">theta</code></td>
<td>
<p>(Numeric) Speed/accuracy trade-off (increase for less accuracy), set to 0.0 for exact TSNE (default: 0.0 see details in the man page of <code>Rtsne::Rtsne</code>).</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_perplexity">perplexity</code></td>
<td>
<p>(Numeric) Perplexity parameter (should not be bigger than 3 * perplexity &lt; nrow(X) - 1, see details in the man page of <code>Rtsne::Rtsne</code>)</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_fact">fact</code></td>
<td>
<p>Name of the column in <code>physeq@sam_data</code> used to color points and compute ellipses.</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_ellipse_level">ellipse_level</code></td>
<td>
<p>The level used in stat_ellipse. Set to NULL to discard ellipse (default = 0.95)</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_plot_dims">plot_dims</code></td>
<td>
<p>A vector of 2 values defining the rank of dimension to plot (default: c(1,2))</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>(logical, default TRUE) Does the samples with NA values in fact are removed? (default: true)</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_force_factor">force_factor</code></td>
<td>
<p>(logical, default TRUE) Force the fact column to be a factor.</p>
</td></tr>
<tr><td><code id="plot_tsne_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code>Rtsne::Rtsne()</code></p>
</td></tr>
</table>


<h3>Details</h3>

<p>Partially inspired by <code>phylosmith::tsne_phyloseq()</code> function developed by Schuyler D. Smith.
</p>


<h3>Value</h3>

<p>A ggplot object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(data_fungi)
if (requireNamespace("Rtsne")) {
  plot_tsne_pq(data_fungi, fact = "Height", perplexity = 15)
}

if (requireNamespace("Rtsne")) {
  plot_tsne_pq(data_fungi, fact = "Time") + geom_label(aes(label = Sample_id, fill = Time))
  plot_tsne_pq(data_fungi, fact = "Time", na_remove = FALSE, force_factor = FALSE)
}


</code></pre>

<hr>
<h2 id='plot_var_part_pq'>Plot the partition the variation of a phyloseq object</h2><span id='topic+plot_var_part_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_var_part_pq(
  res_varpart,
  cutoff = 0,
  digits = 1,
  digits_quantile = 2,
  fill_bg = c("seagreen3", "mediumpurple", "blue", "orange"),
  show_quantiles = FALSE,
  filter_quantile_zero = TRUE,
  show_dbrda_signif = FALSE,
  show_dbrda_signif_pval = 0.05,
  alpha = 63,
  id.size = 1.2,
  min_prop_pval_signif_dbrda = 0.95
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plot_var_part_pq_+3A_res_varpart">res_varpart</code></td>
<td>
<p>(required) the result of the functions <code><a href="#topic+var_par_pq">var_par_pq()</a></code>
or <code><a href="#topic+var_par_rarperm_pq">var_par_rarperm_pq()</a></code></p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_cutoff">cutoff</code></td>
<td>
<p>The values below cutoff will not be displayed.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_digits">digits</code></td>
<td>
<p>The number of significant digits.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_digits_quantile">digits_quantile</code></td>
<td>
<p>The number of significant digits for quantile.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_fill_bg">fill_bg</code></td>
<td>
<p>Fill colours of ellipses.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_show_quantiles">show_quantiles</code></td>
<td>
<p>Do quantiles are printed ?</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_filter_quantile_zero">filter_quantile_zero</code></td>
<td>
<p>Do we filter out value with quantile encompassing
the zero value?</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_show_dbrda_signif">show_dbrda_signif</code></td>
<td>
<p>Do dbrda significance for each component is printed
using *?</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_show_dbrda_signif_pval">show_dbrda_signif_pval</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) The value under which the
dbrda is considered significant.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_alpha">alpha</code></td>
<td>
<p>(int, <code style="white-space: pre;">&#8288;[0:255]&#8288;</code>) Transparency of the fill colour.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_id.size">id.size</code></td>
<td>
<p>A numerical value giving the character expansion factor for the names of circles or ellipses.</p>
</td></tr>
<tr><td><code id="plot_var_part_pq_+3A_min_prop_pval_signif_dbrda">min_prop_pval_signif_dbrda</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) Only used if using the
result of <code><a href="#topic+var_par_rarperm_pq">var_par_rarperm_pq()</a></code> function. The * for dbrda_signif is only add if
at least <code>min_prop_pval_signif_dbrda</code> of permutations show significance.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>vegan::varpart()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+var_par_rarperm_pq">var_par_rarperm_pq()</a></code>, <code><a href="#topic+var_par_pq">var_par_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("vegan")) {
  data_fungi_woNA &lt;- subset_samples(data_fungi, !is.na(Time) &amp; !is.na(Height))
  res_var_9 &lt;- var_par_rarperm_pq(
    data_fungi_woNA,
    list_component = list(
      "Time" = c("Time"),
      "Size" = c("Height", "Diameter")
    ),
    nperm = 9,
    dbrda_computation = TRUE
  )
  res_var_2 &lt;- var_par_rarperm_pq(
    data_fungi_woNA,
    list_component = list(
      "Time" = c("Time"),
      "Size" = c("Height", "Diameter")
    ),
    nperm = 2,
    dbrda_computation = TRUE
  )
  res_var0 &lt;- var_par_pq(data_fungi_woNA,
    list_component = list(
      "Time" = c("Time"),
      "Size" = c("Height", "Diameter")
    ),
    dbrda_computation = TRUE
  )
  plot_var_part_pq(res_var0, digits_quantile = 2, show_dbrda_signif = TRUE)
  plot_var_part_pq(res_var_9,
    digits_quantile = 2, show_quantiles = TRUE,
    show_dbrda_signif = TRUE
  )
  plot_var_part_pq(
    res_var_2,
    digits = 5,
    digits_quantile = 2,
    cutoff = 0,
    show_quantiles = TRUE
  )
}

</code></pre>

<hr>
<h2 id='psmelt_samples_pq'>Build a sample information tibble from physeq object</h2><span id='topic+psmelt_samples_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Hill numbers are the number of equiprobable species giving the same diversity
value as the observed distribution.
</p>
<p>Note that contrary to <code><a href="#topic+hill_pq">hill_pq()</a></code>, this function does not take into
account for difference in the number of sequences per samples/modalities.
You may use rarefy_by_sample = TRUE if the mean number of sequences per
samples differs among modalities.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>psmelt_samples_pq(
  physeq,
  hill_scales = c(0, 1, 2),
  filter_zero = TRUE,
  rarefy_by_sample = FALSE,
  taxa_ranks = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="psmelt_samples_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="psmelt_samples_pq_+3A_hill_scales">hill_scales</code></td>
<td>
<p>(a vector of integer) The list of q values to compute
the hill number H^q. If Null, no hill number are computed. Default value
compute the Hill number 0 (Species richness), the Hill number 1
(exponential of Shannon Index) and the Hill number 2 (inverse of Simpson
Index).</p>
</td></tr>
<tr><td><code id="psmelt_samples_pq_+3A_filter_zero">filter_zero</code></td>
<td>
<p>(logical, default TRUE) Do we filter non present OTU from
samples ? For the moment, this has no effect on the result because the dataframe
is grouped by samples with abundance summed across OTU.</p>
</td></tr>
<tr><td><code id="psmelt_samples_pq_+3A_rarefy_by_sample">rarefy_by_sample</code></td>
<td>
<p>(logical, default FALSE) If TRUE, rarefy
samples using <code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code> function.</p>
</td></tr>
<tr><td><code id="psmelt_samples_pq_+3A_taxa_ranks">taxa_ranks</code></td>
<td>
<p>A vector of taxonomic ranks. For examples c(&quot;Family&quot;,&quot;Genus&quot;).
If taxa ranks is not set (default value = NULL), taxonomic information are not
present in the resulting tibble.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A tibble with a row for each sample. Columns provide information
from <code>sam_data</code> slot as well as hill numbers, Abundance (nb of sequences),
and Abundance_log10 (<em>log10(1+Abundance)</em>).
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ggstatsplot")) {
  psm_tib &lt;- psmelt_samples_pq(data_fungi_mini, hill_scales = c(0, 2, 7))
  ggstatsplot::ggbetweenstats(psm_tib, Height, Hill_0)
  ggstatsplot::ggbetweenstats(psm_tib, Height, Hill_7)

  psm_tib_tax &lt;- psmelt_samples_pq(data_fungi_mini, taxa_ranks = c("Class", "Family"))
  ggplot(filter(psm_tib_tax, Abundance &gt; 2000), aes(y = Family, x = Abundance, fill = Time)) +
    geom_bar(stat = "identity") +
    facet_wrap(~Height)
}
</code></pre>

<hr>
<h2 id='rarefy_sample_count_by_modality'>Rarefy (equalize) the number of samples per modality of a factor</h2><span id='topic+rarefy_sample_count_by_modality'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rarefy_sample_count_by_modality(physeq, fact, rngseed = FALSE, verbose = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rarefy_sample_count_by_modality_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="rarefy_sample_count_by_modality_+3A_fact">fact</code></td>
<td>
<p>(required): The variable to rarefy. Must be present in
the <code>sam_data</code> slot of the physeq object.</p>
</td></tr>
<tr><td><code id="rarefy_sample_count_by_modality_+3A_rngseed">rngseed</code></td>
<td>
<p>(Optional). A single integer value passed to set.seed,
which is used to fix a seed for reproducibly random number generation
(in this case, reproducibly random subsampling). If set to FALSE, then no
iddling with the RNG seed is performed,
and it is up to the user to appropriately call</p>
</td></tr>
<tr><td><code id="rarefy_sample_count_by_modality_+3A_verbose">verbose</code></td>
<td>
<p>(logical). If TRUE, print additional informations.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+accu_plot_balanced_modality">accu_plot_balanced_modality()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>table(data_fungi_mini@sam_data$Height)
data_fungi_mini2 &lt;- rarefy_sample_count_by_modality(data_fungi_mini, "Height")
table(data_fungi_mini2@sam_data$Height)
if (requireNamespace("patchwork")) {
  ggvenn_pq(data_fungi_mini, "Height") + ggvenn_pq(data_fungi_mini2, "Height")
}
</code></pre>

<hr>
<h2 id='read_pq'>Read phyloseq object from multiple csv tables and a phylogenetic tree
in Newick format.</h2><span id='topic+read_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_pq(
  path = NULL,
  taxa_are_rows = FALSE,
  sam_names = NULL,
  sep_csv = "\t",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="read_pq_+3A_path">path</code></td>
<td>
<p>(required) a path to the folder to read the phyloseq object</p>
</td></tr>
<tr><td><code id="read_pq_+3A_taxa_are_rows">taxa_are_rows</code></td>
<td>
<p>(default to FALSE) see ?phyloseq for details</p>
</td></tr>
<tr><td><code id="read_pq_+3A_sam_names">sam_names</code></td>
<td>
<p>The name of the variable (column) in sam_data.csv to rename
samples. Note that if you use <code><a href="#topic+write_phyloseq">write_phyloseq()</a></code> function to save your
physeq object, you may use sam_names = &quot;X&quot; to rename the samples names
as before.</p>
</td></tr>
<tr><td><code id="read_pq_+3A_sep_csv">sep_csv</code></td>
<td>
<p>(default tabulation) separator for column</p>
</td></tr>
<tr><td><code id="read_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="utils.html#topic+write.table">utils::write.table()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>One to four csv tables (refseq.csv, otu_table.csv, tax_table.csv, sam_data.csv)
and if present a phy_tree in Newick format. At least the otu_table.csv need to be present.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>write_pq(data_fungi, path = paste0(tempdir(), "/phyloseq"))
read_pq(path = paste0(tempdir(), "/phyloseq"))
unlink(paste0(tempdir(), "/phyloseq"), recursive = TRUE)
</code></pre>

<hr>
<h2 id='rename_samples'>Rename the samples of a phyloseq slot</h2><span id='topic+rename_samples'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_samples(phyloseq_component, names_of_samples, taxa_are_rows = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_samples_+3A_phyloseq_component">phyloseq_component</code></td>
<td>
<p>(required) one of otu_table or sam_data slot of a
phyloseq-class object</p>
</td></tr>
<tr><td><code id="rename_samples_+3A_names_of_samples">names_of_samples</code></td>
<td>
<p>(required) A vector of samples names</p>
</td></tr>
<tr><td><code id="rename_samples_+3A_taxa_are_rows">taxa_are_rows</code></td>
<td>
<p>(default to FALSE) see ?phyloseq for details</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The otu_table or the sam_data slot with new samples names
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>otutab &lt;- rename_samples(
  data_fungi@otu_table,
  paste0("data_f", sample_names(data_fungi))
)
otutab2 &lt;- rename_samples(
  clean_pq(data_fungi,
    force_taxa_as_rows = TRUE
  )@otu_table,
  paste0("data_f", sample_names(data_fungi))
)
samda &lt;- rename_samples(
  data_fungi@sam_data,
  paste0("data_f", sample_names(data_fungi))
)
</code></pre>

<hr>
<h2 id='rename_samples_otu_table'>Rename samples of an otu_table</h2><span id='topic+rename_samples_otu_table'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rename_samples_otu_table(physeq, names_of_samples)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rename_samples_otu_table_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="rename_samples_otu_table_+3A_names_of_samples">names_of_samples</code></td>
<td>
<p>(required) The new names of the samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>the matrix with new colnames (or rownames if <code>taxa_are_rows</code> is true)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
rename_samples_otu_table(data_fungi, as.character(seq_along(sample_names(data_fungi))))

</code></pre>

<hr>
<h2 id='reorder_taxa_pq'>Reorder taxa in otu_table/tax_table/refseq slot of a phyloseq object</h2><span id='topic+reorder_taxa_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Note that the taxa order in a physeq object with a tree is locked by
the order of leaf in the phylogenetic tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>reorder_taxa_pq(physeq, names_ordered, remove_phy_tree = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="reorder_taxa_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="reorder_taxa_pq_+3A_names_ordered">names_ordered</code></td>
<td>
<p>(required): Names of the taxa (must be the same
as taxa in <code>taxa_names(physeq)</code>) in a given order</p>
</td></tr>
<tr><td><code id="reorder_taxa_pq_+3A_remove_phy_tree">remove_phy_tree</code></td>
<td>
<p>(logical, default FALSE) If TRUE, the phylogenetic
tree is removed. It is</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A phyloseq object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data_fungi_ordered_by_genus &lt;- reorder_taxa_pq(
  data_fungi,
  taxa_names(data_fungi)[order(as.vector(data_fungi@tax_table[, "Genus"]))]
)

data_fungi_mini_asc_ordered_by_abundance &lt;- reorder_taxa_pq(
  data_fungi_mini,
  taxa_names(data_fungi_mini)[order(taxa_sums(data_fungi_mini))]
)
</code></pre>

<hr>
<h2 id='ridges_pq'>Ridge plot of a phyloseq object</h2><span id='topic+ridges_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ridges_pq(physeq, fact, nb_seq = TRUE, log10trans = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ridges_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="ridges_pq_+3A_fact">fact</code></td>
<td>
<p>(required) Name of the factor in <code>physeq@sam_data</code> used to plot
different lines</p>
</td></tr>
<tr><td><code id="ridges_pq_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(logical; default TRUE) If set to FALSE, only the number of ASV
is count. Concretely, physeq <code>otu_table</code> is transformed in a binary
<code>otu_table</code> (each value different from zero is set to one)</p>
</td></tr>
<tr><td><code id="ridges_pq_+3A_log10trans">log10trans</code></td>
<td>
<p>(logical, default TRUE) If TRUE,
the number of sequences (or ASV if nb_seq = FALSE) is log10
transformed.</p>
</td></tr>
<tr><td><code id="ridges_pq_+3A_...">...</code></td>
<td>
<p>Other params passed on to <code><a href="ggridges.html#topic+geom_density_ridges">ggridges::geom_density_ridges()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot  with bar representing the number of sequence en each
taxonomic groups
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ggridges")) {
  ridges_pq(data_fungi_mini, "Time", alpha = 0.5, log10trans = FALSE) + xlim(c(0, 1000))
}

if (requireNamespace("ggridges")) {
  ridges_pq(data_fungi_mini, "Time", alpha = 0.5, scale = 0.9)
  ridges_pq(data_fungi_mini, "Sample_names", log10trans = TRUE)

  ridges_pq(data_fungi_mini,
    "Time",
    jittered_points = TRUE,
    position = ggridges::position_points_jitter(width = 0.05, height = 0),
    point_shape = "|", point_size = 3, point_alpha = 1, alpha = 0.7,
    scale = 0.8
  )
}

</code></pre>

<hr>
<h2 id='rotl_pq'>rotl wrapper for phyloseq data</h2><span id='topic+rotl_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
Make a phylogenetic tree using the ASV names of a physeq object and the
Open Tree of Life tree.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rotl_pq(physeq, species_colnames = "Genus_species", context_name = "All life")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="rotl_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="rotl_pq_+3A_species_colnames">species_colnames</code></td>
<td>
<p>(default: &quot;Genus_species&quot;): the name of the column
where the species binominal name is stored in <code style="white-space: pre;">&#8288;@tax_table&#8288;</code> slot.
Can also be a vector of two columns names e.g. <code>c("Genus", "Species")</code></p>
</td></tr>
<tr><td><code id="rotl_pq_+3A_context_name">context_name</code></td>
<td>
<p>: can bue used to select only a part of the Open Tree
of Life. See <code>?rotl::tnrs_contexts()</code> for available values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>rotl</code> package if you
use this function.
</p>


<h3>Value</h3>

<p>A plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (requireNamespace("rotl")) {
  tr &lt;- rotl_pq(data_fungi_mini, species_colnames = "Genus_species")
  plot(tr)

  tr_Asco &lt;- rotl_pq(data_fungi, species_colnames = "Genus_species", context_name = "Ascomycetes")
  plot(tr_Asco)
}


</code></pre>

<hr>
<h2 id='sample_data_with_new_names'>Load sample data from file and rename samples using names of samples and an
optional order</h2><span id='topic+sample_data_with_new_names'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sample_data_with_new_names(
  file_path,
  names_of_samples,
  samples_order = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sample_data_with_new_names_+3A_file_path">file_path</code></td>
<td>
<p>(required) a path to the sample_data file</p>
</td></tr>
<tr><td><code id="sample_data_with_new_names_+3A_names_of_samples">names_of_samples</code></td>
<td>
<p>(required) a vector of sample names</p>
</td></tr>
<tr><td><code id="sample_data_with_new_names_+3A_samples_order">samples_order</code></td>
<td>
<p>Optional numeric vector to sort sample names</p>
</td></tr>
<tr><td><code id="sample_data_with_new_names_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="utils.html#topic+read.table">utils::read.delim()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data.frame from file_path and new names
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+rename_samples">rename_samples()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>sam_file &lt;- system.file("extdata", "sam_data.csv", package = "MiscMetabar")
sample_data_with_new_names(sam_file, paste0("Samples_", seq(1, 185)))

</code></pre>

<hr>
<h2 id='sankey_pq'>Sankey plot of <code>phyloseq-class</code> object</h2><span id='topic+sankey_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>sankey_pq(
  physeq = NULL,
  fact = NULL,
  taxa = 1:4,
  add_nb_seq = FALSE,
  min_prop_tax = 0,
  tax2remove = NULL,
  units = NULL,
  symbol2sub = c("\\.", "-"),
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="sankey_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_fact">fact</code></td>
<td>
<p>Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_taxa">taxa</code></td>
<td>
<p>a vector of taxonomic rank to plot</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_add_nb_seq">add_nb_seq</code></td>
<td>
<p>Represent the number of sequences or the
number of OTUs (add_nb_seq = FALSE). Note that plotting the number of
sequences is slower.</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_min_prop_tax">min_prop_tax</code></td>
<td>
<p>(default: 0) The minimum proportion for taxon to be
plotted. EXPERIMENTAL. For the moment each links below the min.prop.
tax is discard from the sankey network resulting in sometimes weird plot.</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_tax2remove">tax2remove</code></td>
<td>
<p>a vector of taxonomic groups to remove from the analysis
(e.g. <code>c('Incertae sedis', 'unidentified')</code>)</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_units">units</code></td>
<td>
<p>character string describing physical units (if any) for Value</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_symbol2sub">symbol2sub</code></td>
<td>
<p>(default: c('\.', '-')) vector of symbol to delete in
the taxonomy</p>
</td></tr>
<tr><td><code id="sankey_pq_+3A_...">...</code></td>
<td>
<p>Additional arguments passed on to
<code><a href="networkD3.html#topic+sankeyNetwork">sankeyNetwork</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="networkD3.html#topic+sankeyNetwork">sankeyNetwork</a></code> plot representing the
taxonomic distribution of OTUs or sequences. If <code>fact</code> is set,
represent the distribution of the last taxonomic level in the modalities
of <code>fact</code>
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="networkD3.html#topic+sankeyNetwork">sankeyNetwork</a></code>, <code><a href="#topic+ggaluv_pq">ggaluv_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("GlobalPatterns", package = "phyloseq")
GP &lt;- subset_taxa(GlobalPatterns, GlobalPatterns@tax_table[, 1] == "Archaea")
if (requireNamespace("networkD3")) {
  sankey_pq(GP, fact = "SampleType")
}

if (requireNamespace("networkD3")) {
  sankey_pq(GP, taxa = 1:4, min_prop_tax = 0.01)
  sankey_pq(GP, taxa = 1:4, min_prop_tax = 0.01, add_nb_seq = TRUE)
}

</code></pre>

<hr>
<h2 id='save_pq'>A wrapper of write_pq to save in all three possible formats</h2><span id='topic+save_pq'></span>

<h3>Description</h3>

<p>A wrapper of write_pq to save in all three possible formats
</p>


<h3>Usage</h3>

<pre><code class='language-R'>save_pq(physeq, path = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="save_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="save_pq_+3A_path">path</code></td>
<td>
<p>a path to the folder to save the phyloseq object</p>
</td></tr>
<tr><td><code id="save_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="#topic+write_pq">write_pq()</a></code> or <code><a href="utils.html#topic+write.table">utils::write.table()</a></code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Write :
</p>

<ul>
<li><p> 4 separate tables
</p>
</li>
<li><p> 1 table version
</p>
</li>
<li><p> 1 RData file
</p>
</li></ul>



<h3>Value</h3>

<p>Build a folder (in path) with four csv tables (<code>refseq.csv</code>, <code>otu_table.csv</code>, <code>tax_table.csv</code>, <code>sam_data.csv</code>) + one
table with all tables together + a rdata file (<code>physeq.RData</code>) that can be loaded using
<code><a href="base.html#topic+load">base::load()</a></code> function + if present a phylogenetic tree in Newick format (<code>phy_tree.txt</code>)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+write_pq">write_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
save_pq(data_fungi, path = paste0(tempdir(), "/phyloseq"))
unlink(paste0(tempdir(), "/phyloseq"), recursive = TRUE)

</code></pre>

<hr>
<h2 id='search_exact_seq_pq'>Search for exact matching of sequences using complement,
reverse and reverse-complement</h2><span id='topic+search_exact_seq_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>search_exact_seq_pq(physeq, seq2search)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="search_exact_seq_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="search_exact_seq_pq_+3A_seq2search">seq2search</code></td>
<td>
<p>A DNAStringSet object of sequences to search for.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of data-frames for each input sequences with the name,
the sequences and the
number of occurrences of the original sequence, the complement sequence,
the reverse sequence and the reverse-complement sequence.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data("data_fungi")
search_primers &lt;- search_exact_seq_pq(data_fungi,
  seq2search = Biostrings::DNAStringSet(c("TTGAACGCACATTGCGCC", "ATCCCTACCTGATCCGAG"))
)
</code></pre>

<hr>
<h2 id='select_one_sample'>Select one sample from a physeq object</h2><span id='topic+select_one_sample'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Mostly for internal used, for example in function <code><a href="#topic+track_wkflow_samples">track_wkflow_samples()</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_one_sample(physeq, sam_name, silent = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_one_sample_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="select_one_sample_+3A_sam_name">sam_name</code></td>
<td>
<p>(required) The sample name to select</p>
</td></tr>
<tr><td><code id="select_one_sample_+3A_silent">silent</code></td>
<td>
<p>(logical) If true, no message are printing.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object with one sample
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
A8_005 &lt;- select_one_sample(data_fungi, "A8-005_S4_MERGED.fastq.gz")
A8_005
</code></pre>

<hr>
<h2 id='select_taxa'>Select a subset of taxa in a specified order where possible</h2><span id='topic+select_taxa'></span><span id='topic+select_taxa+2Csample_data+2Ccharacter-method'></span><span id='topic+select_taxa+2Cotu_table+2Ccharacter-method'></span><span id='topic+select_taxa+2CtaxonomyTable+2Ccharacter-method'></span><span id='topic+select_taxa+2CXStringSet+2Ccharacter-method'></span><span id='topic+select_taxa+2Cphylo+2Ccharacter-method'></span><span id='topic+select_taxa+2Cphyloseq+2Ccharacter-method'></span>

<h3>Description</h3>

<p>Select (a subset of) taxa; if <code>x</code> allows taxa to be reordered, then taxa are
given in the specified order.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>select_taxa(x, taxa, reorder = TRUE)

## S4 method for signature 'sample_data,character'
select_taxa(x, taxa)

## S4 method for signature 'otu_table,character'
select_taxa(x, taxa, reorder = TRUE)

## S4 method for signature 'taxonomyTable,character'
select_taxa(x, taxa, reorder = TRUE)

## S4 method for signature 'XStringSet,character'
select_taxa(x, taxa, reorder = TRUE)

## S4 method for signature 'phylo,character'
select_taxa(x, taxa)

## S4 method for signature 'phyloseq,character'
select_taxa(x, taxa, reorder = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="select_taxa_+3A_x">x</code></td>
<td>
<p>A phyloseq object or phyloseq component object</p>
</td></tr>
<tr><td><code id="select_taxa_+3A_taxa">taxa</code></td>
<td>
<p>Character vector of taxa to select, in requested order</p>
</td></tr>
<tr><td><code id="select_taxa_+3A_reorder">reorder</code></td>
<td>
<p>Logical specifying whether to use the order in <code>taxa</code> (TRUE)
or keep the order in <code>taxa_names(x)</code> (FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This is a simple selector function that is like <code>prune_taxa(taxa, x)</code> when
<code>taxa</code> is a character vector but always gives the taxa in the order <code>taxa</code>
if possible (that is, except for phy_tree's and phyloseq objects that
contain phy_tree's).
</p>


<h3>Author(s)</h3>

<p>Michael R. McLaren (orcid: <a href="https://orcid.org/0000-0003-1575-473X">0000-0003-1575-473X</a>)
</p>

<hr>
<h2 id='signif_ancombc'>Filter ancombc_pq results</h2><span id='topic+signif_ancombc'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>signif_ancombc(
  ancombc_res,
  filter_passed = TRUE,
  filter_diff = TRUE,
  min_abs_lfc = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="signif_ancombc_+3A_ancombc_res">ancombc_res</code></td>
<td>
<p>(required) the result of the ancombc_pq function
For the moment only bimodal factors are possible.</p>
</td></tr>
<tr><td><code id="signif_ancombc_+3A_filter_passed">filter_passed</code></td>
<td>
<p>(logical, default TRUE) Do we filter using the column
passed_ss? The passed_ss value is  TRUE if the taxon passed the sensitivity
analysis, i.e., adding different pseudo-counts to 0s would not change the results.</p>
</td></tr>
<tr><td><code id="signif_ancombc_+3A_filter_diff">filter_diff</code></td>
<td>
<p>(logical, default TRUE) Do we filter using the column
diff? The diff value is TRUE if the taxon is significant
(has q less than alpha)</p>
</td></tr>
<tr><td><code id="signif_ancombc_+3A_min_abs_lfc">min_abs_lfc</code></td>
<td>
<p>(integer, default0) Minimum absolute value to filter
results based on Log Fold Change. For ex. a value of 1 filter out taxa
for which the abundance in a given level of the modality is not at least
the double of the abundance in the other level.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>ANCOMBC::ancombc2()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A data.frame with the same number of columns than the <code>ancombc_res</code>
param but with less (or equal) numbers of rows
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ancombc_pq">ancombc_pq()</a></code>, <code><a href="#topic+plot_ancombc_pq">plot_ancombc_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("mia")) {
  data_fungi_mini@tax_table &lt;- phyloseq::tax_table(cbind(
    data_fungi_mini@tax_table,
    "taxon" = taxa_names(data_fungi_mini)
  ))

  res_time &lt;- ancombc_pq(
    data_fungi_mini,
    fact = "Time",
    levels_fact = c("0", "15"),
    tax_level = "taxon",
    verbose = TRUE
  )

  signif_ancombc(res_time)
}

</code></pre>

<hr>
<h2 id='simplify_taxo'>Simplify taxonomy by removing some unused characters such as &quot;k__&quot;</h2><span id='topic+simplify_taxo'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simplify_taxo(physeq, remove_space = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="simplify_taxo_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="simplify_taxo_+3A_remove_space">remove_space</code></td>
<td>
<p>(logical; default TRUE): do we remove space?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A  <code>phyloseq-class</code> object with simplified taxonomy
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='SRS_curve_pq'>Scaling with ranked subsampling (SRS) curve of phyloseq object</h2><span id='topic+SRS_curve_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>SRS_curve_pq(physeq, clean_pq = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="SRS_curve_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="SRS_curve_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical): Does the phyloseq
object is cleaned using the <code><a href="#topic+clean_pq">clean_pq()</a></code> function?</p>
</td></tr>
<tr><td><code id="SRS_curve_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code>SRS::SRScurve()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A plot
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("SRS")) {
  SRS_curve_pq(data_fungi_mini,
    max.sample.size = 200,
    rarefy.comparison = TRUE, rarefy.repeats = 3
  )
  SRS_curve_pq(data_fungi_mini, max.sample.size = 500, metric = "shannon")
}
</code></pre>

<hr>
<h2 id='subsample_fastq'>Subsample a fastq file copying the n_seq first sequences in a given folder</h2><span id='topic+subsample_fastq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subsample_fastq(fastq_files, folder_output = "subsample", nb_seq = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subsample_fastq_+3A_fastq_files">fastq_files</code></td>
<td>
<p>The path to one fastq file or a list of fastq files
(see examples)</p>
</td></tr>
<tr><td><code id="subsample_fastq_+3A_folder_output">folder_output</code></td>
<td>
<p>The path to a folder for output files</p>
</td></tr>
<tr><td><code id="subsample_fastq_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(int; default 1000) : Number of sequences kept (every sequence
spread across 4 lines)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing, create subsampled fastq files in a folder
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
ex_file &lt;- system.file("extdata", "ex_R1_001.fastq.gz",
  package = "MiscMetabar",
  mustWork = TRUE
)
subsample_fastq(ex_file, paste0(tempdir(), "/output_fastq"))
subsample_fastq(list_fastq_files("extdata"), paste0(tempdir(), "/output_fastq"), n = 10)
unlink(paste0(tempdir(), "/output_fastq"), recursive = TRUE)

</code></pre>

<hr>
<h2 id='subset_samples_pq'>Subset samples using a conditional boolean vector.</h2><span id='topic+subset_samples_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The main objective of this function is to complete the
<code><a href="phyloseq.html#topic+subset_samples-methods">phyloseq::subset_samples()</a></code> function by propose a more easy
(but more prone to error) way of subset_samples.
It replace the subsetting expression which used the name of the variable
in the sam_data by a boolean vector.
</p>
<p>Warnings: you must verify the result of this function as the
boolean condition must match the order of samples in the <code>sam_data</code>
slot.
</p>
<p>This function is robust when you use the sam_data slot of the phyloseq object
used in physeq (see examples)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_samples_pq(physeq, condition)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_samples_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="subset_samples_pq_+3A_condition">condition</code></td>
<td>
<p>A boolean vector to subset samples. Length must fit
the number of samples</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a new phyloseq object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
cond_samp &lt;- grepl("A1", data_fungi@sam_data[["Sample_names"]])
subset_samples_pq(data_fungi, cond_samp)

subset_samples_pq(data_fungi, data_fungi@sam_data[["Height"]] == "Low")

</code></pre>

<hr>
<h2 id='subset_taxa_pq'>Subset taxa using a conditional named boolean vector.</h2><span id='topic+subset_taxa_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>The main objective of this function is to complete the
<code><a href="phyloseq.html#topic+subset_taxa-methods">phyloseq::subset_taxa()</a></code> function by propose a more easy way of
subset_taxa using a named boolean vector. Names must match taxa_names.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_taxa_pq(
  physeq,
  condition,
  verbose = TRUE,
  clean_pq = TRUE,
  taxa_names_from_physeq = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_taxa_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="subset_taxa_pq_+3A_condition">condition</code></td>
<td>
<p>A named boolean vector to subset taxa. Length must fit
the number of taxa and names must match taxa_names. Can also be a
condition using a column of the tax_table slot (see examples). If
the order of condition is the same as taxa_names(physeq),
you can use the parameter <code>taxa_names_from_physeq = TRUE</code>.</p>
</td></tr>
<tr><td><code id="subset_taxa_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical) Informations are printed</p>
</td></tr>
<tr><td><code id="subset_taxa_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical)
If set to TRUE, empty samples are discarded after subsetting ASV</p>
</td></tr>
<tr><td><code id="subset_taxa_pq_+3A_taxa_names_from_physeq">taxa_names_from_physeq</code></td>
<td>
<p>(logical) If set to TRUE, rename the
condition vector using taxa_names(physeq). Carefully check the result
of this function if you use this parameter. No effect if the condition
is of class <code>tax_table</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a new phyloseq object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
subset_taxa_pq(data_fungi, data_fungi@tax_table[, "Phylum"] == "Ascomycota")

cond_taxa &lt;- grepl("Endophyte", data_fungi@tax_table[, "Guild"])
names(cond_taxa) &lt;- taxa_names(data_fungi)
subset_taxa_pq(data_fungi, cond_taxa)

subset_taxa_pq(data_fungi, grepl("mycor", data_fungi@tax_table[, "Guild"]),
  taxa_names_from_physeq = TRUE
)

</code></pre>

<hr>
<h2 id='subset_taxa_tax_control'>Subset taxa using a taxa control (e.g. truffle root tips) through 3 methods.</h2><span id='topic+subset_taxa_tax_control'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>subset_taxa_tax_control(
  physeq,
  taxa_distri,
  method = "mean",
  min_diff_for_cutoff = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="subset_taxa_tax_control_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="subset_taxa_tax_control_+3A_taxa_distri">taxa_distri</code></td>
<td>
<p>(required) a vector of length equal to the number of
samples with the number of sequences per samples for the taxa control</p>
</td></tr>
<tr><td><code id="subset_taxa_tax_control_+3A_method">method</code></td>
<td>
<p>(default: &quot;mean&quot;) a method to calculate the cut-off value.
There is 6 available methods:
</p>

<ol>
<li> <p><code>cutoff_seq</code>: discard taxa with less than the number of sequence
than taxa control,
</p>
</li>
<li> <p><code>cutoff_mixt</code>: using mixture models,
</p>
</li>
<li> <p><code>cutoff_diff</code>: using a minimum difference threshold
(need the argument min_diff_for_cutoff)
</p>
</li>
<li> <p><code>min</code>: the minimum of the three firsts methods
</p>
</li>
<li> <p><code>max</code>: the maximum of the three firsts methods
</p>
</li>
<li> <p><code>mean</code>: the mean of the three firsts methods
</p>
</li></ol>
</td></tr>
<tr><td><code id="subset_taxa_tax_control_+3A_min_diff_for_cutoff">min_diff_for_cutoff</code></td>
<td>
<p>(int) argument for method <code>cutoff_diff</code>.
Required if method is <code>cutoff_diff</code>, <code>min</code>, <code>max</code> or <code>mean</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

subset_taxa_tax_control(data_fungi,
  as.numeric(data_fungi@otu_table[, 300]),
  min_diff_for_cutoff = 2
)

</code></pre>

<hr>
<h2 id='summary_plot_pq'>Summarize a <code>phyloseq-class</code> object using a plot.</h2><span id='topic+summary_plot_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summary_plot_pq(
  physeq,
  add_info = TRUE,
  min_seq_samples = 500,
  clean_pq = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="summary_plot_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="summary_plot_pq_+3A_add_info">add_info</code></td>
<td>
<p>Does the bottom down corner contain
extra informations?</p>
</td></tr>
<tr><td><code id="summary_plot_pq_+3A_min_seq_samples">min_seq_samples</code></td>
<td>
<p>(int): Used only when add_info is set
to true to print the number of samples with less sequences than
this number.</p>
</td></tr>
<tr><td><code id="summary_plot_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical): Does the phyloseq
object is cleaned using the <code><a href="#topic+clean_pq">clean_pq()</a></code> function?</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 object
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
summary_plot_pq(data_fungi)
summary_plot_pq(data_fungi, add_info = FALSE) + scale_fill_viridis_d()
</code></pre>

<hr>
<h2 id='swarm_clustering'>Re-cluster sequences of an object of class <code>physeq</code>
or cluster a list of DNA sequences using SWARM</h2><span id='topic+swarm_clustering'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>swarm_clustering(
  physeq = NULL,
  dna_seq = NULL,
  d = 1,
  swarmpath = "swarm",
  vsearch_path = "vsearch",
  nproc = 1,
  swarm_args = "--fastidious",
  tax_adjust = 0,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="swarm_clustering_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_dna_seq">dna_seq</code></td>
<td>
<p>NOT WORKING FOR THE MOMENT
You may directly use a character vector of DNA sequences
in place of physeq args. When physeq is set, dna sequences take the value of
<code>physeq@refseq</code></p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_d">d</code></td>
<td>
<p>(default: 1) maximum number of differences allowed between two
amplicons, meaning that two amplicons will be grouped if they have <code>d</code>
(or less) differences</p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_swarmpath">swarmpath</code></td>
<td>
<p>(default: swarm) path to swarm</p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_vsearch_path">vsearch_path</code></td>
<td>
<p>(default: vsearch) path to vsearch, used only if physeq
is NULL and dna_seq is provided.</p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_nproc">nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for the clustering</p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_swarm_args">swarm_args</code></td>
<td>
<p>(default : &quot;&ndash;fastidious&quot;) a one length character
element defining other parameters to  passed on to swarm See other possible
methods in the <a href="https://github.com/torognes/swarm/blob/master/man/swarm_manual.pdf">SWARM pdf manual</a></p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_tax_adjust">tax_adjust</code></td>
<td>
<p>(Default 0) See the man page
of <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code> for more details.
To conserved the taxonomic rank of the most abundant ASV,</p>
</td></tr>
<tr><td><code id="swarm_clustering_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary
files ?
</p>

<ul>
<li><p> temp.fasta (refseq in fasta or dna_seq sequences)
</p>
</li>
<li><p> temp_output (classical output of SWARM)
</p>
</li>
<li><p> temp_uclust (clusters output of SWARM)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function use the <code>merge_taxa_vec</code> function to
merge taxa into clusters. By default tax_adjust = 0. See the man page
of <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code>.
</p>
<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/torognes/swarm">SWARM</a>.
</p>


<h3>Value</h3>

<p>A new object of class <code>physeq</code> or a list of cluster if dna_seq
args was used.
</p>


<h3>References</h3>

<p>SWARM can be downloaded from
<a href="https://github.com/torognes/swarm/">https://github.com/torognes/swarm/</a>.
</p>
<p>SWARM can be downloaded from
<a href="https://github.com/torognes/swarm">https://github.com/torognes/swarm</a>.
More information in the associated publications
<a href="https://doi.org/10.1093/bioinformatics/btab493">doi:10.1093/bioinformatics/btab493</a> and <a href="https://doi.org/10.7717/peerj.593">doi:10.7717/peerj.593</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+asv2otu">asv2otu()</a></code>, <code><a href="#topic+vsearch_clustering">vsearch_clustering()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
summary_plot_pq(data_fungi)
system2("swarm", "-h")

data_fungi_swarm &lt;- swarm_clustering(data_fungi)
summary_plot_pq(data_fungi_swarm)

sequences_ex &lt;- c(
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTAATAACGAATTCATTGAATCA",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTACCTGGTAAGGCCCACTT",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACCCGGGATTTGATGGGGCGAATTACCTGGTAGAGGTG",
  "TACCTATGTTGCCTTGGCGGCTAAACCTACC",
  "CGGGATTTGATGGCGAATTACCTGGTATTTTAGCCCACTTACCCGGTACCATGAGGTG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACCTGG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG",
  "GCGGCTAAACCTACCCGGGATTTGATGGCGAATTACAAAG"
)

sequences_ex_swarm &lt;- swarm_clustering(
  dna_seq = sequences_ex
)

</code></pre>

<hr>
<h2 id='tax_bar_pq'>Plot the distribution of sequences or ASV in one taxonomic levels</h2><span id='topic+tax_bar_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_bar_pq(
  physeq,
  fact = "Sample",
  taxa = "Order",
  percent_bar = FALSE,
  nb_seq = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tax_bar_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="tax_bar_pq_+3A_fact">fact</code></td>
<td>
<p>Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="tax_bar_pq_+3A_taxa">taxa</code></td>
<td>
<p>(default: 'Order') Name of the taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="tax_bar_pq_+3A_percent_bar">percent_bar</code></td>
<td>
<p>(default FALSE) If TRUE, the stacked bar fill all
the space between 0 and 1. It just set position = &quot;fill&quot; in the
<code>ggplot2::geom_bar()</code> function</p>
</td></tr>
<tr><td><code id="tax_bar_pq_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(logical; default TRUE) If set to FALSE, only the number of ASV
is count. Concretely, physeq otu_table is transformed in a binary
otu_table (each value different from zero is set to one)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot  with bar representing the number of sequence en each
taxonomic groups
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+plot_tax_pq">plot_tax_pq()</a></code> and <code><a href="#topic+multitax_bar_pq">multitax_bar_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data_fungi_ab &lt;- subset_taxa_pq(data_fungi, taxa_sums(data_fungi) &gt; 10000)
tax_bar_pq(data_fungi_ab) + theme(legend.position = "none")

tax_bar_pq(data_fungi_ab, taxa = "Class")
tax_bar_pq(data_fungi_ab, taxa = "Class", percent_bar = TRUE)
tax_bar_pq(data_fungi_ab, taxa = "Class", fact = "Time")

</code></pre>

<hr>
<h2 id='tax_datatable'>Make a datatable with the taxonomy of a <code>phyloseq-class</code> object</h2><span id='topic+tax_datatable'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tax_datatable(
  physeq,
  abundance = TRUE,
  taxonomic_level = NULL,
  modality = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tax_datatable_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="tax_datatable_+3A_abundance">abundance</code></td>
<td>
<p>(default: TRUE) Does the number of sequences is print</p>
</td></tr>
<tr><td><code id="tax_datatable_+3A_taxonomic_level">taxonomic_level</code></td>
<td>
<p>(default: NULL) a vector of selected taxonomic
level using their column numbers (e.g. taxonomic_level = 1:7)</p>
</td></tr>
<tr><td><code id="tax_datatable_+3A_modality">modality</code></td>
<td>
<p>(default: NULL) A sample modality to split
OTU abundancy by level of the modality</p>
</td></tr>
<tr><td><code id="tax_datatable_+3A_...">...</code></td>
<td>
<p>Other argument for the datatable function</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A datatable
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data("GlobalPatterns", package = "phyloseq")
if (requireNamespace("DT")) {
  tax_datatable(subset_taxa(
    GlobalPatterns,
    rowSums(GlobalPatterns@otu_table) &gt; 10000
  ))

  # Using modality
  tax_datatable(GlobalPatterns,
    modality = GlobalPatterns@sam_data$SampleType
  )
}

</code></pre>

<hr>
<h2 id='taxa_as_columns'>Force taxa to be in columns in the otu_table of a physeq object</h2><span id='topic+taxa_as_columns'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_as_columns(physeq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa_as_columns_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='taxa_as_rows'>Force taxa to be in columns in the otu_table of a physeq object</h2><span id='topic+taxa_as_rows'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_as_rows(physeq)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa_as_rows_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>

<hr>
<h2 id='taxa_only_in_one_level'>Show taxa which are present in only one given level of a modality</h2><span id='topic+taxa_only_in_one_level'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>taxa_only_in_one_level(
  physeq,
  modality,
  level,
  min_nb_seq_taxa = 0,
  min_nb_samples_taxa = 0
)

taxa_only_in_one_level(
  physeq,
  modality,
  level,
  min_nb_seq_taxa = 0,
  min_nb_samples_taxa = 0
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="taxa_only_in_one_level_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="taxa_only_in_one_level_+3A_modality">modality</code></td>
<td>
<p>(required) The name of a column present in the <code style="white-space: pre;">&#8288;@sam_data&#8288;</code> slot
of the physeq object. Must be a character vector or a factor.</p>
</td></tr>
<tr><td><code id="taxa_only_in_one_level_+3A_level">level</code></td>
<td>
<p>(required) The level (must be present in modality) of interest</p>
</td></tr>
<tr><td><code id="taxa_only_in_one_level_+3A_min_nb_seq_taxa">min_nb_seq_taxa</code></td>
<td>
<p>(default 0 = no filter) The minimum number of sequences per taxa</p>
</td></tr>
<tr><td><code id="taxa_only_in_one_level_+3A_min_nb_samples_taxa">min_nb_samples_taxa</code></td>
<td>
<p>(default 0 = no filter) The minimum number of samples per taxa</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A vector of taxa names
</p>
<p>A vector of taxa names
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggvenn_pq">ggvenn_pq()</a></code> and <code><a href="#topic+upset_pq">upset_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data_fungi_mini_woNA4height &lt;- subset_samples(
  data_fungi_mini,
  !is.na(data_fungi_mini@sam_data$Height)
)
taxa_only_in_one_level(data_fungi_mini_woNA4height, "Height", "High")
# Taxa present only in low height samples
suppressMessages(suppressWarnings(taxa_only_in_one_level(data_fungi, "Height", "Low")))
# Number of taxa present only in sample of time equal to 15
suppressMessages(suppressWarnings(length(taxa_only_in_one_level(data_fungi, "Time", "15"))))
</code></pre>

<hr>
<h2 id='tbl_sum_samdata'>Summarize information from sample data in a table</h2><span id='topic+tbl_sum_samdata'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>A wrapper for the <code><a href="gtsummary.html#topic+tbl_summary">gtsummary::tbl_summary()</a></code> function in the case of <code>physeq</code>
object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tbl_sum_samdata(physeq, remove_col_unique_value = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tbl_sum_samdata_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="tbl_sum_samdata_+3A_remove_col_unique_value">remove_col_unique_value</code></td>
<td>
<p>(logical, default TRUE) Do we remove
informative columns (categorical column with one value per samples),
e.g. samples names ?</p>
</td></tr>
<tr><td><code id="tbl_sum_samdata_+3A_...">...</code></td>
<td>
<p>Others arguments pass on to <code><a href="gtsummary.html#topic+tbl_summary">gtsummary::tbl_summary()</a></code>.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>gtsummary::tbl_summary()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A new <code>phyloseq-class</code> object with a larger slot tax_table
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("gtsummary")) {
  tbl_sum_samdata(data_fungi) %&gt;%
    gtsummary::as_kable()

  summary_samdata &lt;- tbl_sum_samdata(data_fungi,
    include = c("Time", "Height"),
    type = list(Time ~ "continuous2", Height ~ "categorical"),
    statistic = list(Time ~ c("{median} ({p25}, {p75})", "{min}, {max}"))
  )
}

data(enterotype)
if (requireNamespace("gtsummary")) {
  summary_samdata &lt;- tbl_sum_samdata(enterotype)
  summary_samdata &lt;- tbl_sum_samdata(enterotype, include = !contains("SampleId"))
}

</code></pre>

<hr>
<h2 id='Tengeler2020_pq'>This tutorial explores the dataset from Tengeler et al. (2020) available in the <code>mia</code> package.
obtained using <code>mia::makePhyloseqFromTreeSE(Tengeler2020)</code></h2><span id='topic+Tengeler2020_pq'></span>

<h3>Description</h3>

<p>This is a phyloseq version of the Tengeler2020 dataset.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(Tengeler2020_pq)
</code></pre>


<h3>Format</h3>

<p>A phyloseq object
</p>


<h3>Details</h3>

<p>Tengeler2020 includes gut microbiota profiles of 27 persons with ADHD.
A standard bioinformatic and statistical analysis done to demonstrate that altered microbial
composition could be a driver of altered brain structure and function and concomitant changes
in the animals behavior. This was investigated by colonizing young, male,
germ-free C57BL/6JOlaHsd mice with microbiota from individuals with and without ADHD.
</p>
<p>Tengeler, A.C., Dam, S.A., Wiesmann, M. et al. Gut microbiota from persons with
attention-deficit/hyperactivity disorder affects the brain in mice. Microbiome 8, 44 (2020).
https://microbiomejournal.biomedcentral.com/articles/10.1186/s40168-020-00816-x
</p>

<hr>
<h2 id='track_wkflow'>Track the number of reads (= sequences), samples and cluster (e.g. ASV)
from various objects including dada-class and derep-class.</h2><span id='topic+track_wkflow'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>

<ul>
<li><p> List of fastq and fastg.gz files -&gt; nb of reads and samples
</p>
</li>
<li><p> List of dada-class -&gt; nb of reads, clusters (ASV) and samples
</p>
</li>
<li><p> List of derep-class -&gt; nb of reads, clusters (unique sequences)
and samples
</p>
</li>
<li><p> Matrix of samples x clusters (e.g. <code>otu_table</code>) -&gt; nb of reads,
clusters and samples
</p>
</li>
<li><p> Phyloseq-class -&gt; nb of reads, clusters and samples
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>track_wkflow(
  list_of_objects,
  obj_names = NULL,
  clean_pq = FALSE,
  taxonomy_rank = NULL,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_wkflow_+3A_list_of_objects">list_of_objects</code></td>
<td>
<p>(required) a list of objects</p>
</td></tr>
<tr><td><code id="track_wkflow_+3A_obj_names">obj_names</code></td>
<td>
<p>A list of names corresponding to the list of objects</p>
</td></tr>
<tr><td><code id="track_wkflow_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical)
If set to TRUE, empty samples and empty ASV are discarded
before clustering.</p>
</td></tr>
<tr><td><code id="track_wkflow_+3A_taxonomy_rank">taxonomy_rank</code></td>
<td>
<p>A vector of int. Define the column number of
taxonomic rank <code style="white-space: pre;">&#8288;in physeq@tax_table&#8288;</code> to compute the number of unique value.
Default is NULL and do not compute values for any taxonomic rank</p>
</td></tr>
<tr><td><code id="track_wkflow_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="#topic+clean_pq">clean_pq()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The number of sequences, clusters (e.g. OTUs, ASVs) and samples for
each object.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+track_wkflow_samples">track_wkflow_samples()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(enterotype)
if (requireNamespace("pbapply")) {
  track_wkflow(list(data_fungi, enterotype), taxonomy_rank = c(3, 5))
}

</code></pre>

<hr>
<h2 id='track_wkflow_samples'>Track the number of reads (= sequences), samples and cluster (e.g. ASV)
for each sample</h2><span id='topic+track_wkflow_samples'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Contrary to <code><a href="#topic+track_wkflow">track_wkflow()</a></code>, only phyloseq object are possible.
More information are available in the manual of the function <code><a href="#topic+track_wkflow">track_wkflow()</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>track_wkflow_samples(list_pq_obj, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="track_wkflow_samples_+3A_list_pq_obj">list_pq_obj</code></td>
<td>
<p>(required): a list of object passed on to <code><a href="#topic+track_wkflow">track_wkflow()</a></code>
Only phyloseq object will return value because information of sample is needed</p>
</td></tr>
<tr><td><code id="track_wkflow_samples_+3A_...">...</code></td>
<td>
<p>Other args passed on to <code><a href="#topic+track_wkflow">track_wkflow()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of dataframe. cf <code><a href="#topic+track_wkflow">track_wkflow()</a></code> for more information
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
tree_A10_005 &lt;- subset_samples(data_fungi, Tree_name == "A10-005")
if (requireNamespace("pbapply")) {
  track_wkflow_samples(tree_A10_005)
}

</code></pre>

<hr>
<h2 id='transp'>Adds transparency to a vector of colors</h2><span id='topic+transp'></span>

<h3>Description</h3>

<p>Adds transparency to a vector of colors
</p>


<h3>Usage</h3>

<pre><code class='language-R'>transp(col, alpha = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="transp_+3A_col">col</code></td>
<td>
<p>a vector of colors</p>
</td></tr>
<tr><td><code id="transp_+3A_alpha">alpha</code></td>
<td>
<p>(default 0.5) a numeric value between 0 and 1 representing the alpha coefficient; 0: total transparency; 1: no transparency.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a color vector
</p>


<h3>Author(s)</h3>

<p>Thibaut Jombart in <code>adegenet</code> package
</p>


<h3>See Also</h3>

<p>The R package RColorBrewer, proposing a nice selection of color palettes. The viridis package, with many excellent palettes
</p>

<hr>
<h2 id='treemap_pq'>Plot treemap of 2 taxonomic levels</h2><span id='topic+treemap_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Note that lvl2need to be nested in lvl1
</p>


<h3>Usage</h3>

<pre><code class='language-R'>treemap_pq(
  physeq,
  lvl1,
  lvl2,
  nb_seq = TRUE,
  log10trans = TRUE,
  plot_legend = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="treemap_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="treemap_pq_+3A_lvl1">lvl1</code></td>
<td>
<p>(required) Name of the first (higher) taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="treemap_pq_+3A_lvl2">lvl2</code></td>
<td>
<p>(required) Name of the second (lower) taxonomic rank of interest</p>
</td></tr>
<tr><td><code id="treemap_pq_+3A_nb_seq">nb_seq</code></td>
<td>
<p>(logical; default TRUE) If set to FALSE, only the number of ASV
is count. Concretely, physeq otu_table is transformed in a binary
otu_table (each value different from zero is set to one)</p>
</td></tr>
<tr><td><code id="treemap_pq_+3A_log10trans">log10trans</code></td>
<td>
<p>(logical, default TRUE) If TRUE,
the number of sequences (or ASV if nb_seq = FALSE) is log10
transformed.</p>
</td></tr>
<tr><td><code id="treemap_pq_+3A_plot_legend">plot_legend</code></td>
<td>
<p>(logical, default FALSE) If TRUE, plot che
legend of color for lvl 1</p>
</td></tr>
<tr><td><code id="treemap_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="treemapify.html#topic+geom_treemap">treemapify::geom_treemap()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A ggplot2 graphic
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_fungi_sp_known)
if (requireNamespace("treemapify")) {
  treemap_pq(
    clean_pq(subset_taxa(
      data_fungi_sp_known,
      Phylum == "Basidiomycota"
    )),
    "Order", "Class",
    plot_legend = TRUE
  )
}

if (requireNamespace("treemapify")) {
  treemap_pq(
    clean_pq(subset_taxa(
      data_fungi_sp_known,
      Phylum == "Basidiomycota"
    )),
    "Order", "Class",
    log10trans = FALSE
  )
  treemap_pq(
    clean_pq(subset_taxa(
      data_fungi_sp_known,
      Phylum == "Basidiomycota"
    )),
    "Order", "Class",
    nb_seq = FALSE, log10trans = FALSE
  )
}

</code></pre>

<hr>
<h2 id='tsne_pq'>Compute tSNE position of samples from a phyloseq object</h2><span id='topic+tsne_pq'></span>

<h3>Description</h3>

<p>Compute tSNE position of samples from a phyloseq object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>tsne_pq(physeq, method = "bray", dims = 2, theta = 0, perplexity = 30, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="tsne_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="tsne_pq_+3A_method">method</code></td>
<td>
<p>A method to calculate distance using <code>vegan::vegdist()</code> function</p>
</td></tr>
<tr><td><code id="tsne_pq_+3A_dims">dims</code></td>
<td>
<p>(Int) Output dimensionality (default: 2)</p>
</td></tr>
<tr><td><code id="tsne_pq_+3A_theta">theta</code></td>
<td>
<p>(Numeric) Speed/accuracy trade-off (increase for less accuracy), set to 0.0 for exact TSNE (default: 0.0 see details in the man page of <code>Rtsne::Rtsne</code>).</p>
</td></tr>
<tr><td><code id="tsne_pq_+3A_perplexity">perplexity</code></td>
<td>
<p>(Numeric) Perplexity parameter (should not be bigger than 3 * perplexity &lt; nrow(X) - 1, see details in the man page of <code>Rtsne::Rtsne</code>)</p>
</td></tr>
<tr><td><code id="tsne_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code>Rtsne::Rtsne()</code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of element including the matrix Y containing the new representations for the objects.
See ?Rtsne::Rtsne() for more information
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("Rtsne")) {
  res_tsne &lt;- tsne_pq(data_fungi)
}

</code></pre>

<hr>
<h2 id='unique_or_na'>Get the unique value in x or NA if none</h2><span id='topic+unique_or_na'></span>

<h3>Description</h3>

<p>If <code>unique(x)</code> is a single value, return it; otherwise, return an NA of the
same type as <code>x</code>. If <code>x</code> is a factor, then the levels and ordered status
will be kept in either case. If <code>x</code> is a non-atomic vector (i.e. a list),
then the logical <code>NA</code> will be used.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>unique_or_na(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="unique_or_na_+3A_x">x</code></td>
<td>
<p>A vector</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Either a single value (if <code>unique(x)</code> return a single value) or a NA
</p>


<h3>Author(s)</h3>

<p>Michael R. McLaren (orcid: <a href="https://orcid.org/0000-0003-1575-473X">0000-0003-1575-473X</a>)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>f &lt;- factor(c("a", "a", "b", "c"), ordered = TRUE)
unique_or_na(f)
unique_or_na(f[1:2])

x &lt;- c("a", "b", "a")
unique_or_na(x[c(1, 3)])
unique_or_na(x)
unique_or_na(x) %&gt;% typeof()
</code></pre>

<hr>
<h2 id='upset_pq'>Make upset plot for phyloseq object.</h2><span id='topic+upset_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>Alternative to venn plot.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upset_pq(
  physeq,
  fact,
  taxa_fill = NULL,
  min_nb_seq = 0,
  na_remove = TRUE,
  numeric_fonction = sum,
  rarefy_after_merging = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="upset_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_fact">fact</code></td>
<td>
<p>(required): Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_taxa_fill">taxa_fill</code></td>
<td>
<p>(default NULL) fill the ASV upset using a column in
<code>tax_table</code> slot.</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_min_nb_seq">min_nb_seq</code></td>
<td>
<p>minimum number of sequences by OTUs by
samples to take into count this OTUs in this sample. For example,
if min_nb_seq=2,each value of 2 or less in the OTU table
will not count in the venn diagram</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>: if TRUE (the default), NA values in fact are removed
if FALSE, NA values are set to &quot;NA&quot;</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_numeric_fonction">numeric_fonction</code></td>
<td>
<p>(default : sum) the function for numeric vector
useful only for complex plot (see examples)</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_rarefy_after_merging">rarefy_after_merging</code></td>
<td>
<p>Rarefy each sample after merging by the
modalities of <code>fact</code> parameter</p>
</td></tr>
<tr><td><code id="upset_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to the <code><a href="ComplexUpset.html#topic+upset">ComplexUpset::upset()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+ggvenn_pq">ggvenn_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (requireNamespace("ComplexUpset")) {
  upset_pq(data_fungi_mini,
    fact = "Height", width_ratio = 0.2,
    taxa_fill = "Class"
  )
}

if (requireNamespace("ComplexUpset")) {
  upset_pq(data_fungi_mini, fact = "Height", min_nb_seq = 1000)
  upset_pq(data_fungi_mini, fact = "Height", na_remove = FALSE)

  upset_pq(data_fungi_mini, fact = "Time", width_ratio = 0.2, rarefy_after_merging = TRUE)

  upset_pq(
    data_fungi_mini,
    fact = "Time",
    width_ratio = 0.2,
    annotations = list(
      "Sequences per ASV \n (log10)" = (
        ggplot(mapping = aes(y = log10(Abundance)))
        +
          geom_jitter(aes(
            color =
              Abundance
          ), na.rm = TRUE)
          +
          geom_violin(alpha = 0.5, na.rm = TRUE) +
          theme(legend.key.size = unit(0.2, "cm")) +
          theme(axis.text = element_text(size = 12))
      ),
      "ASV per phylum" = (
        ggplot(mapping = aes(fill = Phylum))
        +
          geom_bar() +
          ylab("ASV per phylum") +
          theme(legend.key.size = unit(0.2, "cm")) +
          theme(axis.text = element_text(size = 12))
      )
    )
  )


  upset_pq(
    data_fungi_mini,
    fact = "Time",
    width_ratio = 0.2,
    numeric_fonction = mean,
    annotations = list(
      "Sequences per ASV \n (log10)" = (
        ggplot(mapping = aes(y = log10(Abundance)))
        +
          geom_jitter(aes(
            color =
              Abundance
          ), na.rm = TRUE)
          +
          geom_violin(alpha = 0.5, na.rm = TRUE) +
          theme(legend.key.size = unit(0.2, "cm")) +
          theme(axis.text = element_text(size = 12))
      ),
      "ASV per phylum" = (
        ggplot(mapping = aes(fill = Phylum))
        +
          geom_bar() +
          ylab("ASV per phylum") +
          theme(legend.key.size = unit(0.2, "cm")) +
          theme(axis.text = element_text(size = 12))
      )
    )
  )


  upset_pq(
    subset_taxa(data_fungi_mini, Phylum == "Basidiomycota"),
    fact = "Time",
    width_ratio = 0.2,
    base_annotations = list(),
    annotations = list(
      "Sequences per ASV \n (log10)" = (
        ggplot(mapping = aes(y = log10(Abundance)))
        +
          geom_jitter(aes(
            color =
              Abundance
          ), na.rm = TRUE)
          +
          geom_violin(alpha = 0.5, na.rm = TRUE) +
          theme(legend.key.size = unit(0.2, "cm")) +
          theme(axis.text = element_text(size = 12))
      ),
      "ASV per phylum" = (
        ggplot(mapping = aes(fill = Class))
        +
          geom_bar() +
          ylab("ASV per Class") +
          theme(legend.key.size = unit(0.2, "cm")) +
          theme(axis.text = element_text(size = 12))
      )
    )
  )

  data_fungi2 &lt;- data_fungi_mini
  data_fungi2@sam_data[["Time_0"]] &lt;- data_fungi2@sam_data$Time == 0
  data_fungi2@sam_data[["Height__Time_0"]] &lt;-
    paste0(data_fungi2@sam_data[["Height"]], "__", data_fungi2@sam_data[["Time_0"]])
  data_fungi2@sam_data[["Height__Time_0"]][grepl("NA", data_fungi2@sam_data[["Height__Time_0"]])] &lt;-
    NA
  upset_pq(data_fungi2, fact = "Height__Time_0", width_ratio = 0.2, min_size = 2)
}

</code></pre>

<hr>
<h2 id='upset_test_pq'>Test for differences between intersections</h2><span id='topic+upset_test_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>upset_test_pq(
  physeq,
  fact,
  var_to_test = "OTU",
  min_nb_seq = 0,
  na_remove = TRUE,
  numeric_fonction = sum,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="upset_test_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="upset_test_pq_+3A_fact">fact</code></td>
<td>
<p>(required): Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="upset_test_pq_+3A_var_to_test">var_to_test</code></td>
<td>
<p>(default c(&quot;OTU&quot;)) : a vector of column present in
the tax_table slot from the physeq object</p>
</td></tr>
<tr><td><code id="upset_test_pq_+3A_min_nb_seq">min_nb_seq</code></td>
<td>
<p>minimum number of sequences by OTUs by
samples to take into count this OTUs in this sample. For example,
if min_nb_seq=2,each value of 2 or less in the OTU table
will not count in the venn diagram</p>
</td></tr>
<tr><td><code id="upset_test_pq_+3A_na_remove">na_remove</code></td>
<td>
<p>: if TRUE (the default), NA values in fact are removed
if FALSE, NA values are set to &quot;NA&quot;</p>
</td></tr>
<tr><td><code id="upset_test_pq_+3A_numeric_fonction">numeric_fonction</code></td>
<td>
<p>(default : sum) the function for numeric vector
useful only for complex plot (see examples)</p>
</td></tr>
<tr><td><code id="upset_test_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to the <code><a href="ComplexUpset.html#topic+upset_test">ComplexUpset::upset_test()</a></code></p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+upset_pq">upset_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data_fungi)
if (requireNamespace("ComplexUpset")) {
  upset_test_pq(data_fungi, "Height", var_to_test = c("OTU", "Class", "Guild"))
  upset_test_pq(data_fungi, "Time")
}
</code></pre>

<hr>
<h2 id='var_par_pq'>Partition the Variation of a phyloseq object by 2, 3, or 4 Explanatory Matrices</h2><span id='topic+var_par_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
The function partitions the variation in otu_table using
distance (Bray per default) with respect to two, three, or four explanatory
tables, using
adjusted R² in redundancy analysis ordination (RDA) or distance-based
redundancy analysis. If response is a single vector, partitioning is by
partial regression. Collinear variables in the explanatory tables do NOT
have to be removed prior to partitioning. See <code><a href="vegan.html#topic+varpart">vegan::varpart()</a></code> for more
information.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>var_par_pq(
  physeq,
  list_component,
  dist_method = "bray",
  dbrda_computation = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var_par_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="var_par_pq_+3A_list_component">list_component</code></td>
<td>
<p>(required) A named list of 2, 3 or four vectors with
names from the <code style="white-space: pre;">&#8288;@sam_data&#8288;</code> slot.</p>
</td></tr>
<tr><td><code id="var_par_pq_+3A_dist_method">dist_method</code></td>
<td>
<p>(default &quot;bray&quot;) the distance used. See
<code><a href="phyloseq.html#topic+distance">phyloseq::distance()</a></code> for all available distances or run
<code><a href="phyloseq.html#topic+distanceMethodList">phyloseq::distanceMethodList()</a></code>.
For &quot;aitchison&quot; and &quot;robust.aitchison&quot; distance, <code><a href="vegan.html#topic+vegdist">vegan::vegdist()</a></code>
function is directly used.</p>
</td></tr>
<tr><td><code id="var_par_pq_+3A_dbrda_computation">dbrda_computation</code></td>
<td>
<p>(logical) Do dbrda computations are runned for each
individual component (each name of the list component) ?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>vegan::varpart()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>an object of class &quot;varpart&quot;, see <code><a href="vegan.html#topic+varpart">vegan::varpart()</a></code>
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+var_par_rarperm_pq">var_par_rarperm_pq()</a></code>, <code><a href="vegan.html#topic+varpart">vegan::varpart()</a></code>, <code><a href="#topic+plot_var_part_pq">plot_var_part_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("vegan")) {
  data_fungi_woNA &lt;-
    subset_samples(data_fungi, !is.na(Time) &amp; !is.na(Height))
  res_var &lt;- var_par_pq(data_fungi_woNA,
    list_component = list(
      "Time" = c("Time"),
      "Size" = c("Height", "Diameter")
    ),
    dbrda_computation = TRUE
  )
}

</code></pre>

<hr>
<h2 id='var_par_rarperm_pq'>Partition the Variation of a phyloseq object with rarefaction permutations</h2><span id='topic+var_par_rarperm_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#experimental"><img src="../help/figures/lifecycle-experimental.svg" alt='[Experimental]' /></a>
</p>
<p>This is an extension of the function <code><a href="#topic+var_par_pq">var_par_pq()</a></code>. The main addition is
the computation of nperm permutations with rarefaction even depth by
sample. The return object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>var_par_rarperm_pq(
  physeq,
  list_component,
  dist_method = "bray",
  nperm = 99,
  quantile_prob = 0.975,
  dbrda_computation = FALSE,
  dbrda_signif_pval = 0.05,
  sample.size = min(sample_sums(physeq)),
  verbose = FALSE,
  progress_bar = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="var_par_rarperm_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_list_component">list_component</code></td>
<td>
<p>(required) A named list of 2, 3 or four vectors with
names from the <code style="white-space: pre;">&#8288;@sam_data&#8288;</code> slot.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_dist_method">dist_method</code></td>
<td>
<p>(default &quot;bray&quot;) the distance used. See
<code><a href="phyloseq.html#topic+distance">phyloseq::distance()</a></code> for all available distances or run
<code><a href="phyloseq.html#topic+distanceMethodList">phyloseq::distanceMethodList()</a></code>.
For aitchison and robust.aitchison distance, <code><a href="vegan.html#topic+vegdist">vegan::vegdist()</a></code>
function is directly used.#' @param fill_bg</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_nperm">nperm</code></td>
<td>
<p>(int) The number of permutations to perform.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_quantile_prob">quantile_prob</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) the value to compute the quantile.
Minimum quantile is compute using 1-quantile_prob.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_dbrda_computation">dbrda_computation</code></td>
<td>
<p>(logical) Do dbrda computations are runned for each
individual component (each name of the list component) ?</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_dbrda_signif_pval">dbrda_signif_pval</code></td>
<td>
<p>(float, <code style="white-space: pre;">&#8288;[0:1]&#8288;</code>) The value under which the dbrda is
considered significant.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_sample.size">sample.size</code></td>
<td>
<p>(int) A single integer value equal to the number of
reads being simulated, also known as the depth. See
<code><a href="phyloseq.html#topic+rarefy_even_depth">phyloseq::rarefy_even_depth()</a></code>.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical). If TRUE, print additional informations.</p>
</td></tr>
<tr><td><code id="var_par_rarperm_pq_+3A_progress_bar">progress_bar</code></td>
<td>
<p>(logical, default TRUE) Do we print progress during
the calculation?</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please make a reference to <code>vegan::varpart()</code> if you
use this function.
</p>


<h3>Value</h3>

<p>A list of class varpart with additional information in the
<code style="white-space: pre;">&#8288;$part$indfract&#8288;</code> part. Adj.R.square is the mean across permutation.
Adj.R.squared_quantil_min and Adj.R.squared_quantil_max represent
the quantile values of adjuste R squared
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+var_par_pq">var_par_pq()</a></code>, <code><a href="vegan.html#topic+varpart">vegan::varpart()</a></code>, <code><a href="#topic+plot_var_part_pq">plot_var_part_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("vegan")) {
  data_fungi_woNA &lt;- subset_samples(data_fungi, !is.na(Time) &amp; !is.na(Height))
  res_var_9 &lt;- var_par_rarperm_pq(
    data_fungi_woNA,
    list_component = list(
      "Time" = c("Time"),
      "Size" = c("Height", "Diameter")
    ),
    nperm = 9,
    dbrda_computation = TRUE
  )
  res_var_2 &lt;- var_par_rarperm_pq(
    data_fungi_woNA,
    list_component = list(
      "Time" = c("Time"),
      "Size" = c("Height", "Diameter")
    ),
    nperm = 2,
    dbrda_computation = TRUE
  )
}

</code></pre>

<hr>
<h2 id='venn_pq'>Venn diagram of <code>phyloseq-class</code> object</h2><span id='topic+venn_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>venn_pq(physeq, fact, min_nb_seq = 0, print_values = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="venn_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="venn_pq_+3A_fact">fact</code></td>
<td>
<p>(required): Name of the factor to cluster samples by modalities.
Need to be in <code>physeq@sam_data</code>.</p>
</td></tr>
<tr><td><code id="venn_pq_+3A_min_nb_seq">min_nb_seq</code></td>
<td>
<p>(default: 0)): minimum number of sequences by OTUs by
samples to take into count this OTUs in this sample. For example,
if min_nb_seq=2,each value of 2 or less in the OTU table
will be change into 0 for the analysis</p>
</td></tr>
<tr><td><code id="venn_pq_+3A_print_values">print_values</code></td>
<td>
<p>(logical) Print (or not) the table of number of OTUs
for each combination.
If print_values is TRUE the object is not a ggplot object.
Please use print_values = FALSE if you want to add ggplot function
(cf example).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A <code><a href="ggplot2.html#topic+ggplot">ggplot</a></code>2 plot representing Venn diagram of
modalities of the argument <code>factor</code>
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="venneuler.html#topic+venneuler">venneuler</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
if (requireNamespace("venneuler")) {
  data("enterotype")
  venn_pq(enterotype, fact = "SeqTech")
}

if (requireNamespace("venneuler")) {
  venn_pq(enterotype, fact = "ClinicalStatus")
  venn_pq(enterotype, fact = "Nationality", print_values = FALSE)
  venn_pq(enterotype, fact = "ClinicalStatus", print_values = FALSE) +
    scale_fill_hue()
  venn_pq(enterotype, fact = "ClinicalStatus", print_values = FALSE) +
    scale_fill_hue()
}


</code></pre>

<hr>
<h2 id='verify_pq'>Verify the validity of a phyloseq object</h2><span id='topic+verify_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>
<p>Mostly for internal use in MiscMetabar functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>verify_pq(
  physeq,
  verbose = FALSE,
  min_nb_seq_sample = 500,
  min_nb_seq_taxa = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="verify_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="verify_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical, default FALSE) If TRUE, prompt some warnings.</p>
</td></tr>
<tr><td><code id="verify_pq_+3A_min_nb_seq_sample">min_nb_seq_sample</code></td>
<td>
<p>(numeric) Only used if verbose = TRUE.
Minimum number of sequences per samples to not show warning.</p>
</td></tr>
<tr><td><code id="verify_pq_+3A_min_nb_seq_taxa">min_nb_seq_taxa</code></td>
<td>
<p>(numeric) Only used if verbose = TRUE.
Minimum number of sequences per taxa to not show warning.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Nothing if the phyloseq object is valid. An error in the other case.
Warnings if verbose = TRUE
</p>

<hr>
<h2 id='vs_search_global'>Search for a list of sequence in a fasta file against physeq reference
sequences using <a href="https://github.com/torognes/vsearch">vsearch</a></h2><span id='topic+vs_search_global'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vs_search_global(
  physeq,
  seq2search = NULL,
  path_to_fasta = NULL,
  vsearchpath = "vsearch",
  id = 0.8,
  iddef = 0,
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vs_search_global_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="vs_search_global_+3A_seq2search">seq2search</code></td>
<td>
<p>(required if path_to_fasta is NULL) Either (i) a DNAstringSet object
or (ii) a character vector that will be convert to DNAstringSet using
<code><a href="Biostrings.html#topic+XStringSet-class">Biostrings::DNAStringSet()</a></code></p>
</td></tr>
<tr><td><code id="vs_search_global_+3A_path_to_fasta">path_to_fasta</code></td>
<td>
<p>(required if seq2search is NULL) a path to fasta file if seq2search is est to NULL.</p>
</td></tr>
<tr><td><code id="vs_search_global_+3A_vsearchpath">vsearchpath</code></td>
<td>
<p>(default: vsearch) path to vsearch</p>
</td></tr>
<tr><td><code id="vs_search_global_+3A_id">id</code></td>
<td>
<p>(default: 0.8) id for the option <code>--usearch_global</code> of the vsearch software</p>
</td></tr>
<tr><td><code id="vs_search_global_+3A_iddef">iddef</code></td>
<td>
<p>(default: 0) iddef for the option <code>--usearch_global</code> of the vsearch software</p>
</td></tr>
<tr><td><code id="vs_search_global_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files
</p>

<ul>
<li><p> temp.fasta (refseq in fasta)
</p>
</li>
<li><p> cluster.fasta (centroid)
</p>
</li>
<li><p> temp.uc (clusters)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/torognes/vsearch">vsearch</a>.
</p>


<h3>Value</h3>

<p>A dataframe with uc results (invisible)
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

if (requireNamespace("seqinr")) {
  file_dna &lt;- tempfile("dna.fa")
  seqinr::write.fasta("GCCCATTAGTATTCTAGTGGGCATGCCTGTTCGAGCGTCATTTTCAACC",
    file = file_dna, names = "seq1"
  )

  res &lt;- vs_search_global(data_fungi, path_to_fasta = file_dna)
  unlink(file_dna)

  res[res$identity != "*", ]

  clean_pq(subset_taxa(data_fungi, res$identity != "*"))
}


</code></pre>

<hr>
<h2 id='vsearch_clustering'>Recluster sequences of an object of class <code>physeq</code>
or cluster a list of DNA sequences using vsearch software</h2><span id='topic+vsearch_clustering'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vsearch_clustering(
  physeq = NULL,
  dna_seq = NULL,
  nproc = 1,
  id = 0.97,
  vsearchpath = "vsearch",
  tax_adjust = 0,
  vsearch_cluster_method = "--cluster_size",
  vsearch_args = "--strand both",
  keep_temporary_files = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vsearch_clustering_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_dna_seq">dna_seq</code></td>
<td>
<p>You may directly use a character vector of DNA sequences
in place of physeq args. When physeq is set, dna sequences take the value of
<code>physeq@refseq</code></p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_nproc">nproc</code></td>
<td>
<p>(default: 1)
Set to number of cpus/processors to use for the clustering</p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_id">id</code></td>
<td>
<p>(default: 0.97) level of identity to cluster</p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_vsearchpath">vsearchpath</code></td>
<td>
<p>(default: vsearch) path to vsearch</p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_tax_adjust">tax_adjust</code></td>
<td>
<p>(Default 0) See the man page
of <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code> for more details.
To conserved the taxonomic rank of the most abundant ASV,
set tax_adjust to 0 (default). For the moment only tax_adjust = 0 is
robust</p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_vsearch_cluster_method">vsearch_cluster_method</code></td>
<td>
<p>(default: &quot;&ndash;cluster_size) See other possible
methods in the <a href="https://github.com/torognes/vsearch">vsearch manual</a> (e.g. <code>--cluster_size</code> or <code>--cluster_smallmem</code>)
</p>

<ul>
<li> <p><code>--cluster_fast</code> : Clusterize the fasta sequences in filename, automatically sort by decreasing sequence length beforehand.
</p>
</li>
<li> <p><code>--cluster_size</code> : Clusterize the fasta sequences in filename, automatically sort by decreasing sequence abundance beforehand.
</p>
</li>
<li> <p><code>--cluster_smallmem</code> : Clusterize the fasta sequences in filename without automatically modifying their order beforehand. Sequence are expected to be sorted by decreasing sequence length, unless <em>&ndash;usersort</em> is used.
In that case you may set <code>vsearch_args</code> to vsearch_args = &quot;&ndash;strand both &ndash;usersort&quot;
</p>
</li></ul>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_vsearch_args">vsearch_args</code></td>
<td>
<p>(default : &quot;&ndash;strand both&quot;) a one length character element defining other parameters to
passed on to vsearch.</p>
</td></tr>
<tr><td><code id="vsearch_clustering_+3A_keep_temporary_files">keep_temporary_files</code></td>
<td>
<p>(logical, default: FALSE) Do we keep temporary files ?
</p>

<ul>
<li><p> temp.fasta (refseq in fasta or dna_seq sequences)
</p>
</li>
<li><p> cluster.fasta (centroid if method = &quot;vsearch&quot;)
</p>
</li>
<li><p> temp.uc (clusters if method = &quot;vsearch&quot;)
</p>
</li></ul>
</td></tr>
</table>


<h3>Details</h3>

<p>This function use the <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code> function to
merge taxa into clusters. By default tax_adjust = 0. See the man page
of <code><a href="#topic+merge_taxa_vec">merge_taxa_vec()</a></code>.
</p>
<p>This function is mainly a wrapper of the work of others.
Please cite <a href="https://github.com/torognes/vsearch">vsearch</a>.
</p>


<h3>Value</h3>

<p>A new object of class <code>physeq</code> or a list of cluster if dna_seq
args was used.
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>References</h3>

<p>VSEARCH can be downloaded from
<a href="https://github.com/torognes/vsearch">https://github.com/torognes/vsearch</a>.
More information in the associated publication
<a href="https://pubmed.ncbi.nlm.nih.gov/27781170">https://pubmed.ncbi.nlm.nih.gov/27781170</a>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+asv2otu">asv2otu()</a></code>, <code><a href="#topic+swarm_clustering">swarm_clustering()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>

summary_plot_pq(data_fungi)
d_vs &lt;- vsearch_clustering(data_fungi)
summary_plot_pq(d_vs)


</code></pre>

<hr>
<h2 id='write_pq'>Save phyloseq object in the form of multiple csv tables.</h2><span id='topic+write_pq'></span>

<h3>Description</h3>

<p><a href="https://lifecycle.r-lib.org/articles/stages.html#maturing"><img src="../help/figures/lifecycle-maturing.svg" alt='[Maturing]' /></a>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>write_pq(
  physeq,
  path = NULL,
  rdata = FALSE,
  one_file = FALSE,
  write_sam_data = TRUE,
  sam_data_first = FALSE,
  clean_pq = TRUE,
  reorder_asv = FALSE,
  rename_asv = FALSE,
  remove_empty_samples = TRUE,
  remove_empty_taxa = TRUE,
  clean_samples_names = TRUE,
  silent = FALSE,
  verbose = FALSE,
  quote = FALSE,
  sep_csv = "\t",
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="write_pq_+3A_physeq">physeq</code></td>
<td>
<p>(required): a <code>phyloseq-class</code> object obtained
using the <code>phyloseq</code> package.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_path">path</code></td>
<td>
<p>a path to the folder to save the phyloseq object</p>
</td></tr>
<tr><td><code id="write_pq_+3A_rdata">rdata</code></td>
<td>
<p>(logical) does the phyloseq object is also saved in Rdata format?</p>
</td></tr>
<tr><td><code id="write_pq_+3A_one_file">one_file</code></td>
<td>
<p>(logical) if TRUE, combine all data in one file only</p>
</td></tr>
<tr><td><code id="write_pq_+3A_write_sam_data">write_sam_data</code></td>
<td>
<p>(logical) does the samples data are add to
the file. Only used if <code>one_file</code> is TRUE.
Note that these option result in a lot of NA values.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_sam_data_first">sam_data_first</code></td>
<td>
<p>(logical) if TRUE, put the sample data at the top of the table
Only used if <code>one_file</code> and write_sam_data are both TRUE.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_clean_pq">clean_pq</code></td>
<td>
<p>(logical)
If set to TRUE, empty samples are discarded after subsetting ASV</p>
</td></tr>
<tr><td><code id="write_pq_+3A_reorder_asv">reorder_asv</code></td>
<td>
<p>(logical) if TRUE the otu_table is ordered by the number of
sequences of ASV (descending order). Default to TRUE. Only possible if clean_pq
is set to TRUE.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_rename_asv">rename_asv</code></td>
<td>
<p>reorder_asv (logical) if TRUE, ASV are renamed by their position
in the OTU_table (asv_1, asv_2, ...). Default to FALSE. Only possible if clean_pq
is set to TRUE.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_remove_empty_samples">remove_empty_samples</code></td>
<td>
<p>(logical) Do you want to remove samples
without sequences (this is done after removing empty taxa)</p>
</td></tr>
<tr><td><code id="write_pq_+3A_remove_empty_taxa">remove_empty_taxa</code></td>
<td>
<p>(logical) Do you want to remove taxa
without sequences (this is done before removing empty samples)</p>
</td></tr>
<tr><td><code id="write_pq_+3A_clean_samples_names">clean_samples_names</code></td>
<td>
<p>(logical) Do you want to clean samples names?</p>
</td></tr>
<tr><td><code id="write_pq_+3A_silent">silent</code></td>
<td>
<p>(logical) If true, no message are printing.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_verbose">verbose</code></td>
<td>
<p>(logical) Additional informations in the message
the verbose parameter overwrite the silent parameter.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_quote">quote</code></td>
<td>
<p>a logical value (default FALSE) or a numeric vector.
If TRUE, any character or factor columns will be surrounded by
double quotes.  If a numeric vector, its elements are taken
as the indices of columns to quote.  In both cases, row and
column names are quoted if they are written. If FALSE nothing is quoted.</p>
</td></tr>
<tr><td><code id="write_pq_+3A_sep_csv">sep_csv</code></td>
<td>
<p>(default tabulation) separator for column</p>
</td></tr>
<tr><td><code id="write_pq_+3A_...">...</code></td>
<td>
<p>Other arguments passed on to <code><a href="utils.html#topic+write.table">utils::write.table()</a></code> function.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Build a folder (path) containing one to four csv tables
(refseq.csv, otu_table.csv, tax_table.csv, sam_data.csv)
and if present a phy_tree in Newick format
</p>


<h3>Author(s)</h3>

<p>Adrien Taudière
</p>


<h3>See Also</h3>

<p><code><a href="#topic+save_pq">save_pq()</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>write_pq(data_fungi, path = paste0(tempdir(), "/phyloseq"))
write_pq(data_fungi, path = paste0(tempdir(), "/phyloseq"), one_file = TRUE)
unlink(paste0(tempdir(), "/phyloseq"), recursive = TRUE)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
