<!DOCTYPE html><html lang="en"><head><title>Help for package MiSPU</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MiSPU}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MiSPU-package'>
<p>Microbiome Based Sum of Powered Score (MiSPU) Tests</p></a></li>
<li><a href='#correspLeaves'><p>Finding the correspondence OTUs for each taxa</p></a></li>
<li><a href='#Dirichlet'><p>The Dirichlet Distribution</p></a></li>
<li><a href='#DirMultOutput'>
<p>The estimate of Dirichlet-multinomial distribution</p></a></li>
<li><a href='#GUniFrac'>
<p>Generalized UniFrac distances for comparing microbial communities.</p></a></li>
<li><a href='#MiSPU'><p>microbiome based sum of powered score (MiSPU)</p></a></li>
<li><a href='#ranking'><p>ranking the OTUs</p></a></li>
<li><a href='#simulateData'><p>OTU counts simulation</p></a></li>
<li><a href='#throat.meta'>
<p>Meta data of the throat microbiome samples.</p></a></li>
<li><a href='#throat.otu.tab'>
<p>OTU count table from 16S sequencing of the throat microbiome samples.</p></a></li>
<li><a href='#throat.tree'>
<p>UPGMA tree of the OTUs from 16S sequencing of the throat microbiome samples.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Microbiome Based Sum of Powered Score (MiSPU) Tests</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-02-29</td>
</tr>
<tr>
<td>Author:</td>
<td>Chong Wu, Wei Pan</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Chong Wu &lt;wuxx0845@umn.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>There is an increasing interest in investigating how the compositions of microbial communities are associated with human health and disease. In this package, we present a novel global testing method called aMiSPU, that is highly adaptive and thus high powered across various scenarios, alleviating the issue with the choice of a phylogenetic distance. Our simulations and real data analysis demonstrated that aMiSPU test was often more powerful than several competing methods while correctly controlling type I error rates.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.1)</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.2.3), vegan, ape, aSPU,cluster</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ade4</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-03-17 18:15:15 UTC; chong</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-03-18 00:08:39</td>
</tr>
</table>
<hr>
<h2 id='MiSPU-package'>
Microbiome Based Sum of Powered Score (MiSPU) Tests 
</h2><span id='topic+MiSPU-package'></span>

<h3>Description</h3>

<p>There is an increasing interest in investigating how the compositions of microbial communities are associated with human health and disease. In this package, we present a novel global testing method called aMiSPU, that is highly adaptive and thus high powered across various scenarios, alleviating the issue with the choice of a phylogenetic distance. Our simulations and real data analysis demonstrated that aMiSPU test was often more powerful than several competing methods while correctly controlling type I error rates.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> MiSPU</td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-02-29</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL-2 </td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Chong Wu, Wei Pan
Maintainer: Chong Wu &lt;wuxx0845@umn.edu&gt;
</p>


<h3>References</h3>

<p>Pan, W., et al.(2014) A powerful and adaptive association test for rare variants, Genetics, 197(4), 1081-95
</p>
<p>Chong, W., Pan, W. (2015) An Adaptive Association Test for Microbiome Data, submitted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

Y.tmp =throat.meta[,3]
Y = rep(0,dim(throat.meta)[1])
Y[Y.tmp=="Smoker"] = 1

cov.tmp = throat.meta[,c(10,12)]
cov = matrix(1,dim(throat.meta)[1],2)
cov[cov.tmp[,1]== "None",1] = 0
cov[cov.tmp[,2]== "Male",2] = 0

start.time = proc.time()
X = as.matrix(throat.otu.tab)

out = MiSPU(Y,X, throat.tree,cov,model =  "binomial", pow = c(2:8, Inf), n.perm = 1000)
out
</code></pre>

<hr>
<h2 id='correspLeaves'>Finding the correspondence OTUs for each taxa </h2><span id='topic+correspLeaves'></span>

<h3>Description</h3>

<p>Finding the descendants for each taxa. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>correspLeaves(tree)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="correspLeaves_+3A_tree">tree</code></td>
<td>
<p>Rooted phylogenetic tree of R class &ldquo;phylo&rdquo;</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing the descendants for each taxa.
</p>


<h3>Author(s)</h3>

<p>Chong Wu
</p>


<h3>References</h3>

<p>Chong, W., Pan, W. (2015) An Adaptive Association Test for Microbiome Data, submitted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.tree)
correspLeaves(throat.tree)
</code></pre>

<hr>
<h2 id='Dirichlet'>The Dirichlet Distribution</h2><span id='topic+rdirichlet'></span>

<h3>Description</h3>

<p>Density function and random number generation for the Dirichlet distribution</p>


<h3>Usage</h3>

<pre><code class='language-R'>rdirichlet(n, alpha)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Dirichlet_+3A_n">n</code></td>
<td>
<p>number of random observations to draw</p>
</td></tr>
<tr><td><code id="Dirichlet_+3A_alpha">alpha</code></td>
<td>
<p>the Dirichlet distribution's parameters. Can be a vector (one set of parameters for all observations) or a matrix (a different set of parameters for each observation), see &ldquo;Details&rdquo;</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The Dirichlet distribution is a multidimensional generalization of the Beta distribution where each dimension is governed by an <code class="reqn">\alpha</code>-parameter.
Formally this is
</p>
<p style="text-align: center;"><code class="reqn">%
    \mathcal{D}(\alpha_i)=\left[\left.\Gamma(\sum_{i}\alpha_i)\right/\prod_i\Gamma(\alpha_i)\right]\prod_{i}y_i^{\alpha_i-1}%
  </code>
</p>

<p>Usually, <code>alpha</code> is a vector thus the same parameters will be used for all observations.
If <code>alpha</code> is a matrix, a complete set of <code class="reqn">\alpha</code>-parameters must be supplied for each observation.
</p>


<h3>Value</h3>

<p>returns a matrix with random numbers according to the supplied alpha vector or matrix.
</p>


<h3>Author(s)</h3>

<p>Chong Wu</p>


<h3>Examples</h3>

<pre><code class='language-R'>X1 &lt;- rdirichlet(100, c(5, 5, 10))
X1
</code></pre>

<hr>
<h2 id='DirMultOutput'>
The estimate of Dirichlet-multinomial distribution
</h2><span id='topic+dd'></span>

<h3>Description</h3>

<p>The estimate of Dirichlet-multinomial distribution. It just intermidates estimate.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(dd)</code></pre>


<h3>Format</h3>

<p>The format is: matrix
</p>


<h3>Details</h3>

<p>It just intermidates estimate. Not very useful.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(dd)
</code></pre>

<hr>
<h2 id='GUniFrac'>
Generalized UniFrac distances for comparing microbial communities.
</h2><span id='topic+GUniFrac'></span>

<h3>Description</h3>

<p>A generalized version of commonly used UniFrac distances. It is defined as:
</p>
<p style="text-align: center;"><code class="reqn">d^{(\alpha)} = \frac{\sum_{i=1}^m b_i (p^A_{i} + p^B_{i})^\alpha 
\left\vert \frac{ p^A_{i} - p^B_{i} }{p^A_{i} + p^B_{i}} \right\vert } { 
 \sum_{i=1}^m b_i (p^A_{i} + p^B_{i})^\alpha},</code>
</p>

<p>where <code class="reqn">m</code> is the number of branches, <code class="reqn">b_i</code> is the length of 
<code class="reqn">i</code>th branch, <code class="reqn">p^A_{i}, p^B_{i}</code> are the branch
proportion for community A and B. 
</p>
<p>Generalized UniFrac distance contains an extra parameter <code class="reqn">\alpha</code> 
controlling the weight on abundant lineages so the distance is not dominated 
by highly abundant lineages. <code class="reqn">\alpha=0.5</code> has overall the best 
power. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GUniFrac(otu.tab, tree,alpha = c(0,0.5,1))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GUniFrac_+3A_otu.tab">otu.tab</code></td>
<td>

<p>OTU count table, row - n sample, column - q OTU
</p>
</td></tr>
<tr><td><code id="GUniFrac_+3A_tree">tree</code></td>
<td>

<p>Rooted phylogenetic tree of R class &ldquo;phylo&rdquo;
</p>
</td></tr>
<tr><td><code id="GUniFrac_+3A_alpha">alpha</code></td>
<td>
<p>Parameter controlling weight on abundant lineages</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Return a list containing
</p>
<table role = "presentation">
<tr><td><code>d0</code></td>
<td>
<p>UniFrac(0)</p>
</td></tr>
<tr><td><code>d5</code></td>
<td>
<p>UniFrac(0.5)</p>
</td></tr>
<tr><td><code>d1</code></td>
<td>
<p>UniFrac(1), weighted UniFrac</p>
</td></tr>
</table>
<p>or a list containing
</p>
<table role = "presentation">
<tr><td><code>GUniFrac</code></td>
<td>
<p>The distance matrix for different alpha</p>
</td></tr>
<tr><td><code>alpha</code></td>
<td>
<p>The weight</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The time consuming part is written in C and faster than the original one. The function only accepts rooted tree. 
</p>


<h3>Author(s)</h3>

<p>Chong Wu &lt;chongwu@umn.edu&gt;
</p>


<h3>References</h3>

<p>Chen, Jun, et al (2012). &quot;Associating microbiome composition with environmental covariates using generalized UniFrac distances.&quot; Bioinformatics 28(16):2106-2113.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

groups &lt;- throat.meta$SmokingStatus

# Calculate the UniFracs
unifracs &lt;- GUniFrac(throat.otu.tab, throat.tree)
unifracs
</code></pre>

<hr>
<h2 id='MiSPU'>microbiome based sum of powered score (MiSPU)</h2><span id='topic+MiSPU'></span>

<h3>Description</h3>

<p>We propose a class of microbiome based sum of powered score (MiSPU) tests based on a newly defined generalized taxon proportion that combines observed microbial composition information with phylogenetic tree information. Different from the existing methods, a MiSPU test is based on a weighted score of the generalized taxon proportion in a general framework of regression, upweighting more likely to be associated microbial lineages. Our simulations demonstrated that one or more MiSPU tests were more powerful than MiRKAT while correctly controlling type I error rates. An adaptive MiSPU (aMiSPU) test is proposed to combine multiple MiSPU tests with various weights, approximating the most powerful MiSPU for a given scenario, consequently being highly adaptive and high powered across various scenarios.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MiSPU(y, X, tree, cov = NULL,model = c("gaussian", "binomial"),
 pow = c(2:8, Inf), n.perm = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="MiSPU_+3A_y">y</code></td>
<td>
<p>Outcome of interest. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td></tr>
<tr><td><code id="MiSPU_+3A_x">X</code></td>
<td>
<p>OTU count table, row - n sample, column - q OTU</p>
</td></tr>
<tr><td><code id="MiSPU_+3A_tree">tree</code></td>
<td>
<p>Rooted phylogenetic tree of R class &ldquo;phylo&rdquo;</p>
</td></tr>
<tr><td><code id="MiSPU_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td></tr>
<tr><td><code id="MiSPU_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
<tr><td><code id="MiSPU_+3A_pow">pow</code></td>
<td>
<p>The gamma which controls the weight. Larger pow puts more weight on the variables that have larger absolute score.</p>
</td></tr>
<tr><td><code id="MiSPU_+3A_n.perm">n.perm</code></td>
<td>
<p>number of permutations or bootstraps.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object, including the results for MiSPU_u, MiSPU_w and aMiSPU.
</p>


<h3>Author(s)</h3>

<p>Chong Wu
</p>


<h3>References</h3>

<p>Pan, W., et al.(2014) A powerful and adaptive association test for rare variants, Genetics, 197(4), 1081-95
</p>
<p>Chong, W., Pan, W. (2015) An Adaptive Association Test for Microbiome Data, submitted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

Y.tmp =throat.meta[,3]
Y = rep(0,dim(throat.meta)[1])
Y[Y.tmp=="Smoker"] = 1

cov.tmp = throat.meta[,c(10,12)]
cov = matrix(1,dim(throat.meta)[1],2)
cov[cov.tmp[,1]== "None",1] = 0
cov[cov.tmp[,2]== "Male",2] = 0

start.time = proc.time()
X = as.matrix(throat.otu.tab)

out = MiSPU(Y,X, throat.tree,cov,model =  "binomial", pow = c(2:8, Inf), n.perm = 1000)
out
</code></pre>

<hr>
<h2 id='ranking'>ranking the OTUs</h2><span id='topic+ranking'></span>

<h3>Description</h3>

<p>Ranking the importance of each taxa.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ranking(y, X, tree, cov = NULL,gamma,g.taxon.index,model = "binomial")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ranking_+3A_y">y</code></td>
<td>
<p>Outcome of interest. It can be a disease indicator; =0 for controls, =1 for cases.
Or it can be a quantitative trait. A vector with length n (number of observations).</p>
</td></tr>
<tr><td><code id="ranking_+3A_x">X</code></td>
<td>
<p>OTU count table, row - n sample, column - q OTU</p>
</td></tr>
<tr><td><code id="ranking_+3A_tree">tree</code></td>
<td>
<p>Rooted phylogenetic tree of R class &ldquo;phylo&rdquo;</p>
</td></tr>
<tr><td><code id="ranking_+3A_cov">cov</code></td>
<td>
<p>Covariates. A matrix with dimension n by p (n :number of observation, p : number of covariates).</p>
</td></tr>
<tr><td><code id="ranking_+3A_gamma">gamma</code></td>
<td>
<p>The best gamma selected by aMiSPU test.</p>
</td></tr>
<tr><td><code id="ranking_+3A_g.taxon.index">g.taxon.index</code></td>
<td>
<p>g.taxon.index = 1 stands for weigted generalized taxon proportion; otherwise means unweighted generalized taxon proportion.</p>
</td></tr>
<tr><td><code id="ranking_+3A_model">model</code></td>
<td>
<p>Use &quot;gaussian&quot; for a quantitative trait, and use &quot;binomial&quot; for a binary trait.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix containing the ranking score, the higher the more important.
</p>


<h3>Author(s)</h3>

<p>Chong Wu
</p>


<h3>References</h3>

<p>Chong, W., Pan, W. (2015) An Adaptive Association Test for Microbiome Data, submitted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
data(throat.tree)
data(throat.meta)

Y.tmp =throat.meta[,3]
Y = rep(0,dim(throat.meta)[1])
Y[Y.tmp=="Smoker"] = 1

cov.tmp = throat.meta[,c(10,12)]
cov = matrix(1,dim(throat.meta)[1],2)
cov[cov.tmp[,1]== "None",1] = 0
cov[cov.tmp[,2]== "Male",2] = 0

start.time = proc.time()
X = as.matrix(throat.otu.tab)

#out = MiSPU(Y,X, throat.tree,cov,model =  "binomial", pow = c(2:8, Inf), n.perm = 1000)
out = ranking(Y,X, throat.tree,cov,gamma = 2, g.taxon.index =1)

</code></pre>

<hr>
<h2 id='simulateData'>OTU counts simulation</h2><span id='topic+simulateData'></span>

<h3>Description</h3>

<p>We used a phylogenetic tree of OTUs from a real throat microbiome data set, which consists of 856 OTUs after discarding singleton OTUs. Then based on a complicate statistical meodel, we generated the OTU counts for each individual to simulate the feature of real microbiome data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>simulateData(nSam=100, s=12, ncluster = 20, mu = 1000, size = 25) 
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="simulateData_+3A_nsam">nSam</code></td>
<td>
<p>Sample size, the default value is 100.</p>
</td></tr>
<tr><td><code id="simulateData_+3A_s">s</code></td>
<td>
<p>The indicator of informative cluster.</p>
</td></tr>
<tr><td><code id="simulateData_+3A_ncluster">ncluster</code></td>
<td>
<p>Positive integer specifying the number of clusters of they phylogenetic tree.</p>
</td></tr>
<tr><td><code id="simulateData_+3A_mu">mu</code></td>
<td>
<p>The mean of the negative binomial distribution. </p>
</td></tr>
<tr><td><code id="simulateData_+3A_size">size</code></td>
<td>
<p>The size of the negative binomial distribution</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list object, including the informative OTU counts table and whole OTU counts table.
</p>


<h3>Author(s)</h3>

<p>Chong Wu
</p>


<h3>References</h3>

<p>Chong, W., Pan, W. (2015) An Adaptive Association Test for Microbiome Data, submitted. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>OTU = simulateData()
OTU
</code></pre>

<hr>
<h2 id='throat.meta'>
Meta data of the throat microbiome samples.
</h2><span id='topic+throat.meta'></span>

<h3>Description</h3>

<p>It is part of a microbiome data set for studying the effect of
smoking on  the upper respiratory tract microbiome. The original data 
set contains samples from both throat and  nose microbiomes, and 
from both body sides. This data set comes from the throat microbiome 
of left body side. It contains 60 subjects consisting of 32 nonsmokers
and 28 smokers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.meta)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;throat.meta&quot;
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.meta)
## maybe str(throat.meta) ; plot(throat.meta) ...
</code></pre>

<hr>
<h2 id='throat.otu.tab'>
OTU count table from 16S sequencing of the throat microbiome samples.
</h2><span id='topic+throat.otu.tab'></span>

<h3>Description</h3>

<p>It is part of a microbiome data set for studying the effect of
smoking on  the upper respiratory tract microbiome. The original data 
set contains samples from both throat and  nose microbiomes, and 
from both body sides. This data set comes from the throat microbiome 
of left body side. It contains 60 subjects consisting of 32 nonsmokers
and 28 smokers. 
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.otu.tab)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;throat.otu.tab&quot;
</p>


<h3>Details</h3>

<p>The OTU table is produced by the QIIME software. Singleton OTUs have
been discarded.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.otu.tab)
## maybe str(throat.otu.tab) ; plot(throat.otu.tab) ...
</code></pre>

<hr>
<h2 id='throat.tree'>
UPGMA tree of the OTUs from 16S sequencing of the throat microbiome samples.
</h2><span id='topic+throat.tree'></span>

<h3>Description</h3>

<p>The OTU tree is constructed using UPGMA on the K80 distance matrice of
the OTUs. It is a rooted tree of class &quot;phylo&quot;.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(throat.tree)</code></pre>


<h3>Format</h3>

<p>The format is:
chr &quot;throat.tree&quot;
</p>


<h3>Details</h3>

<p>The OTUs are produced by the QIIME software. Singleton OTUs have
been discarded.
</p>


<h3>Source</h3>

<p>Charlson ES, Chen J, Custers-Allen R, Bittinger K, Li H, et al. (2010) 
Disordered Microbial Communities in the Upper Respiratory Tract of 
Cigarette Smokers. PLoS ONE 5(12): e15216. 
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(throat.tree)
## maybe str(throat.tree) ; plot(throat.tree) ...
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
