<!DOCTYPE html><html lang="en"><head><title>Help for package MitoHEAR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MitoHEAR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#choose_features_clustering'><p>choose_features_clustering</p></a></li>
<li><a href='#clustering_angular_distance'><p>clustering_angular_distance</p></a></li>
<li><a href='#detect_insertion'><p>detect_insertion</p></a></li>
<li><a href='#dpt_test'><p>dpt_test</p></a></li>
<li><a href='#filter_bases'><p>filter_bases</p></a></li>
<li><a href='#get_distribution'><p>get_distribution</p></a></li>
<li><a href='#get_heteroplasmy'><p>get_heteroplasmy</p></a></li>
<li><a href='#get_raw_counts_allele'><p>get_raw_counts_allele</p></a></li>
<li><a href='#get_wilcox_test'><p>get_wilcox_test</p></a></li>
<li><a href='#plot_allele_frequency'><p>plot_allele_frequency</p></a></li>
<li><a href='#plot_base_coverage'><p>plot_base_coverage</p></a></li>
<li><a href='#plot_batch'><p>plot_batch</p></a></li>
<li><a href='#plot_cells_coverage'><p>plot_cells_coverage</p></a></li>
<li><a href='#plot_condition'><p>plot_condition</p></a></li>
<li><a href='#plot_coordinate_cluster'><p>plot_coordinate_cluster</p></a></li>
<li><a href='#plot_coordinate_heteroplasmy'><p>plot_coordinate_heteroplasmy</p></a></li>
<li><a href='#plot_correlation_bases'><p>plot_correlation_bases</p></a></li>
<li><a href='#plot_distance_matrix'><p>plot_distance_matrix</p></a></li>
<li><a href='#plot_distribution'><p>plot_distribution</p></a></li>
<li><a href='#plot_dpt'><p>plot_dpt</p></a></li>
<li><a href='#plot_genome_coverage'><p>plot_genome_coverage</p></a></li>
<li><a href='#plot_heatmap'><p>plot_heatmap</p></a></li>
<li><a href='#plot_heteroplasmy'><p>plot_heteroplasmy</p></a></li>
<li><a href='#plot_heteroplasmy_variability'><p>plot_heteroplasmy_variability</p></a></li>
<li><a href='#plot_spider_chart'><p>plot_spider_chart</p></a></li>
<li><a href='#vi_comparison'><p>vi_comparison</p>
We compute the variation of information (VI) between the partition provided
by <em>new_classification</em> and <em>old_classification</em>. The VI between a
random partitions (obtained with re-shuffle from original labels in
<em>old_classification</em>) and <em>old_classification</em> is also computed. A
distribution of VI values from random partitions is built. Finally, from the
comparison with this distribution, an empirical p value is given to the VI
of the unsupervised cluster analysis.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Quantification of Mitochondrial DNA Heteroplasmy</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Gabriele Lubatti</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Gabriele Lubatti &lt;gabriele.lubatti@helmholtz-muenchen.de&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Allows the estimation and downstream statistical analysis of the mitochondrial DNA Heteroplasmy calculated from single-cell datasets <a href="https://github.com/ScialdoneLab/MitoHEAR/tree/master">https://github.com/ScialdoneLab/MitoHEAR/tree/master</a>.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/Artistic-2.0">Artistic-2.0</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, circlize, ComplexHeatmap, dynamicTreeCut,
GenomicRanges, ggplot2, gridExtra, IRanges, magrittr, mcclust,
rdist, reshape2, rlist, Rsamtools,</td>
</tr>
<tr>
<td>Suggests:</td>
<td>clustree, fmsb, gam, karyoploteR, knitr, plotly, regioneR,
rmarkdown, testthat</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>biocViews:</td>
<td>software</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-02-28 13:04:23 UTC; gabriele.lubatti</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-03-01 21:20:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='choose_features_clustering'>choose_features_clustering</h2><span id='topic+choose_features_clustering'></span>

<h3>Description</h3>

<p>choose_features_clustering
</p>


<h3>Usage</h3>

<pre><code class='language-R'>choose_features_clustering(
  heteroplasmy_matrix,
  allele_matrix,
  cluster,
  top_pos,
  deepSplit_param,
  minClusterSize_param,
  min_value_vector,
  threshold = 0.2,
  index,
  max_frac = 0.7
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="choose_features_clustering_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_allele_matrix">allele_matrix</code></td>
<td>
<p>Fourth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_top_pos">top_pos</code></td>
<td>
<p>Numeric value. Number of bases sorted with decreasing values
of distance variance (see section <em>Details</em> below) among samples. If
<em>relevant_bases</em>=NULL, then the bases for performing hierarchical clustering
are the ones whose relative variance (variance of the base divided sum of
variance among <em>top_pos</em> bases) is above <em>min_value</em>.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_deepsplit_param">deepSplit_param</code></td>
<td>
<p>Integer value between 0 and 4 for the <em>deepSplit</em>
parameter of the function <em>cutreeHybrid</em>. See section <em>Details</em>
below.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_minclustersize_param">minClusterSize_param</code></td>
<td>
<p>Integer value specifying the <em>minClusterSize</em>
parameter of the function <em>cutreeHybrid</em>. See section <em>Details</em>
below.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_min_value_vector">min_value_vector</code></td>
<td>
<p>Numeric vector. For each value in the vector, the function
<em>clustering_angular_distance</em> is run with parameter <em>min_value</em> equal to
one element of the vector <em>min_value_vector</em>.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_threshold">threshold</code></td>
<td>
<p>Numeric value. If a base has heteroplasmy greater or equal
to <em>threshold</em> in more than <em>max_frac</em> of cells, then the base is
not considered for down stream analysis.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="choose_features_clustering_+3A_max_frac">max_frac</code></td>
<td>
<p>Numeric value.If a base has heteroplasmy greater or equal to
<em>threshold</em> in more than <em>max_frac</em> of cells, then the base is not
considered for down stream analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Clustree plot returned by function <em>clustree</em> from package <em>clustree</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://cran.r-project.org/package=clustree">https://cran.r-project.org/package=clustree</a>
</p>

<hr>
<h2 id='clustering_angular_distance'>clustering_angular_distance</h2><span id='topic+clustering_angular_distance'></span>

<h3>Description</h3>

<p>For each pair of samples and for each base, an angular distance matrix is
computed based on the four allele frequencies. Then only the angular
distances corresponding to the relevant_bases are kept. If relevant bases is
NULL, then only the angular distances corresponding to the bases with
relative distance variance among samples above <em>min_value</em> are kept .
Finally the distance between each pair of samples is defined as the
euclidean distance of the angular distances corresponding to the bases that
pass the previous filtering step. On this final distance matrix, a
hierarchical clustering approach is performed using the function
<em>cutreeHybrid</em> of the package <em>dynamicTreeCut</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>clustering_angular_distance(
  heteroplasmy_matrix,
  allele_matrix,
  cluster,
  top_pos,
  deepSplit_param,
  minClusterSize_param,
  threshold = 0.2,
  min_value,
  index,
  relevant_bases = NULL,
  max_frac = 0.7
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="clustering_angular_distance_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_allele_matrix">allele_matrix</code></td>
<td>
<p>Fourth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_top_pos">top_pos</code></td>
<td>
<p>Numeric value. Number of bases sorted with decreasing values
of distance variance (see section <em>Details</em> below) among samples. If
<em>relevant_bases</em>=NULL, then the bases for performing hierarchical clustering
are the ones whose relative variance (variance of the base divided sum of
variance among <em>top_pos</em> bases) is above <em>min_value</em>.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_deepsplit_param">deepSplit_param</code></td>
<td>
<p>Integer value between 0 and 4 for the <em>deepSplit</em>
parameter of the function <em>cutreeHybrid</em>. See section <em>Details</em>
below.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_minclustersize_param">minClusterSize_param</code></td>
<td>
<p>Integer value specifying the <em>minClusterSize</em>
parameter of the function <em>cutreeHybrid</em>. See section <em>Details</em>
below.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_threshold">threshold</code></td>
<td>
<p>Numeric value. If a base has heteroplasmy greater or equal
to <em>threshold</em> in more than <em>max_frac</em> of cells, then the base is
not considered for down stream analysis.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_min_value">min_value</code></td>
<td>
<p>Numeric value. If <em>relevant_bases</em>=NULL, then the bases for
performing hierarchical clustering are the ones whose relative variance
(variance of the base divided sum of variance among <em>top_pos</em> bases) is
above <em>min_value</em>.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_relevant_bases">relevant_bases</code></td>
<td>
<p>Character vector of bases to consider as features for
performing hierarchical clustering on samples.Default=NULL.</p>
</td></tr>
<tr><td><code id="clustering_angular_distance_+3A_max_frac">max_frac</code></td>
<td>
<p>Numeric value.If a base has heteroplasmy greater or equal to
<em>threshold</em> in more than <em>max_frac</em> of cells, then the base is not
considered for down stream analysis.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a list with 4 elements: </p>
<table role = "presentation">
<tr><td><code>classification</code></td>
<td>
<p> Dataframe with
two columns and n_row equal to n_row in heteroplasmy_matrix. The first
column is the old cluster annotation provided by cluster. The second columns
is the new cluster annotation obtained with hierarchical clustering on
distance matrix based on heteroplasmy values.
</p>
</td></tr>
<tr><td><code>dist_ang_matrix</code></td>
<td>
<p> Distance matrix based on heteroplasmy values as defined
in the section <em>Details</em>
</p>
</td></tr>
<tr><td><code>top_bases_dist</code></td>
<td>
<p> Vector of bases used for hierarchical clustering. If
<em>relevant_bases</em> is not NULL, then <em>top_bases_dist</em>=NULL
</p>
</td></tr>
<tr><td><code>common_idx</code></td>
<td>
<p> Vector of indices of samples for which hierarchical
clustering is performed. If <em>index</em> is NULL, then <em>common_idx</em>=NULL
</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/dynamicTreeCut/versions/1.63-1/topics/cutreeHybrid">https://www.rdocumentation.org/packages/dynamicTreeCut/versions/1.63-1/topics/cutreeHybrid</a>
</p>

<hr>
<h2 id='detect_insertion'>detect_insertion</h2><span id='topic+detect_insertion'></span>

<h3>Description</h3>

<p>detect_insertion
</p>


<h3>Usage</h3>

<pre><code class='language-R'>detect_insertion(ref_sequence, different_sequence, length_comparison = 10)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="detect_insertion_+3A_ref_sequence">ref_sequence</code></td>
<td>
<p>Character vector whose elements are the bases of a DNA
sequence to use as reference.</p>
</td></tr>
<tr><td><code id="detect_insertion_+3A_different_sequence">different_sequence</code></td>
<td>
<p>Character vector whose elements are the bases of a
DNA sequence different from the reference.</p>
</td></tr>
<tr><td><code id="detect_insertion_+3A_length_comparison">length_comparison</code></td>
<td>
<p>Integer number. Number of bases to consider for the
comparison between the two DNA sequences in order to detect and remove
insertions in the non-reference sequence.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of the different_sequence with length equal to
ref_sequence, after having removed the insertions.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='dpt_test'>dpt_test</h2><span id='topic+dpt_test'></span>

<h3>Description</h3>

<p>dpt_test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dpt_test(heteroplasmy_matrix, time, index = NULL, method = "GAM")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dpt_test_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="dpt_test_+3A_time">time</code></td>
<td>
<p>Vector of diffusion pseudo time.</p>
</td></tr>
<tr><td><code id="dpt_test_+3A_index">index</code></td>
<td>
<p>index returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="dpt_test_+3A_method">method</code></td>
<td>
<p>Character name denoting the method to choose for assigning an
adjusted p value to each of the bases. Can be one of GAM, pearson and
spearman. GAM: For each base, a GAM fit with formula z ~ lo(t) is performed
between the heteroplasmy values (z) and the time (t). The p value from the
table &quot;Anova for Parametric Effects&quot; is then assigned to the base.
pearson,spearman:for each base, a pearson or spearman correlation test is
performed between the heteroplasmy values and the time . The p value
obtained from the test is then assigned to the base. In all the three
possible methods, all the p values are then corrected with the method FDR.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with 2 columns and number of rows equal to n_col in
<em>heteroplasmy_matrix</em>. In the first column there are the names of the bases
while in the second column there are the adjusted p value.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/gam/versions/1.20/topics/gam">https://www.rdocumentation.org/packages/gam/versions/1.20/topics/gam</a>
</p>

<hr>
<h2 id='filter_bases'>filter_bases</h2><span id='topic+filter_bases'></span>

<h3>Description</h3>

<p>filter_bases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filter_bases(heteroplasmy_matrix, min_heteroplasmy, min_cells, index = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="filter_bases_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="filter_bases_+3A_min_heteroplasmy">min_heteroplasmy</code></td>
<td>
<p>Numeric value.</p>
</td></tr>
<tr><td><code id="filter_bases_+3A_min_cells">min_cells</code></td>
<td>
<p>Numeric value.</p>
</td></tr>
<tr><td><code id="filter_bases_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Character vector of bases that have an heteroplasmy greater than
<em>min_heteroplasmy</em> in more than <em>min_cells</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='get_distribution'>get_distribution</h2><span id='topic+get_distribution'></span>

<h3>Description</h3>

<p>get_distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_distribution(heteroplasmy_matrix, FUNCTION, index = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_distribution_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="get_distribution_+3A_function">FUNCTION</code></td>
<td>
<p>A character specifying the function to be applied on each
column of <em>matrix</em>. The possible values are:
<em>mean</em>,<em>max</em>,<em>min</em>,<em>median</em> and <em>sum</em>.</p>
</td></tr>
<tr><td><code id="get_distribution_+3A_index">index</code></td>
<td>
<p>index returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a numeric vector with length equal to n_col of <em>matrix</em>
where each element contains the result of the operation defined by
<em>FUNCTION</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti &lt;gabriele.lubatti@helmholtz-muenchen.de&gt;
</p>

<hr>
<h2 id='get_heteroplasmy'>get_heteroplasmy</h2><span id='topic+get_heteroplasmy'></span>

<h3>Description</h3>

<p>It is one of the two main functions of the <strong>MitoHEAR</strong> package
(together with <em>get_raw_counts_allele</em>). It computes the allele
frequencies and the heteroplasmy matrix starting from the counts matrix
obtained with <em>get_raw_counts_allele</em>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_heteroplasmy(
  raw_counts_allele,
  name_position_allele,
  name_position,
  number_reads,
  number_positions,
  filtering = 1,
  my.clusters = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_heteroplasmy_+3A_raw_counts_allele">raw_counts_allele</code></td>
<td>
<p>A raw counts matrix obtained from
<em>get_raw_counts_allele</em>.</p>
</td></tr>
<tr><td><code id="get_heteroplasmy_+3A_name_position_allele">name_position_allele</code></td>
<td>
<p>A character vector with elements specifying the
genomic coordinate of the base and the allele (obtained from
<em>get_raw_counts_allele</em>).</p>
</td></tr>
<tr><td><code id="get_heteroplasmy_+3A_name_position">name_position</code></td>
<td>
<p>A character vector with elements specifying the genomic
coordinate of the base (obtained from <em>get_raw_counts_allele</em>).</p>
</td></tr>
<tr><td><code id="get_heteroplasmy_+3A_number_reads">number_reads</code></td>
<td>
<p>Integer specifying the minimum number of counts above
which we consider the base covered by the sample.</p>
</td></tr>
<tr><td><code id="get_heteroplasmy_+3A_number_positions">number_positions</code></td>
<td>
<p>Integer specifying the minimumnumber of bases that
must be covered by the sample (with counts&gt;<em>number_reads</em>), in order to keep
the sample for down-stream analysis.</p>
</td></tr>
<tr><td><code id="get_heteroplasmy_+3A_filtering">filtering</code></td>
<td>
<p>Numeric value equal to 1 or 2. If 1 then only the bases
that are covered by all the samples are kept for the downstream analysis. If
2 then all the bases that are covered by more than 50% of the the samples
in each cluster (specified by <em>my.clusters</em>) are kept for the down-stream
analysis. Default is 1.</p>
</td></tr>
<tr><td><code id="get_heteroplasmy_+3A_my.clusters">my.clusters</code></td>
<td>
<p>Character vector specifying a partition of the samples.
It is only used when filtering is equal to 2. Default is NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Starting from <em>raw counts allele matrix</em>, the function performed two
consequentially filtering steps. The first one is on the samples, keeping
only the ones that cover a number of bases above number_positions. The
second one is on the bases, defined by the parameter filtering. The
heteroplasmy for each sample-base pair is computed as <em>1-max(f)</em>, where
<em>f</em> are the frequencies of the four alleles.
</p>


<h3>Value</h3>

<p>It returns a list with 5 elements:
</p>
<table role = "presentation">
<tr><td><code>sum_matrix</code></td>
<td>
<p>A matrix (n_row=number of sample, n_col=number of bases)
with the counts for each sample/base, for all the initial samples and bases
included in the <em>raw counts allele matrix</em>.</p>
</td></tr>
<tr><td><code>sum_matrix_qc</code></td>
<td>
<p>A matrix (n_row=number of sample, n_col=number of
bases) with the counts for each sample/base, for all the samples and bases
that pass the two consequentially filtering steps.</p>
</td></tr>
<tr><td><code>heteroplasmy_matrix</code></td>
<td>
<p>A matrix with the same dimension of <em>sum_matrix_qc</em>
where each entry (i,j) is the heteroplasmy for sample i in base j.</p>
</td></tr>
<tr><td><code>allele_matrix</code></td>
<td>
<p>A matrix (n_row=number of sample,
n_col=4*number of bases) with allele frequencies, for all the samples and
bases that pass the two consequentially filtering steps.</p>
</td></tr>
<tr><td><code>index</code></td>
<td>
<p>Indices of the samples that
cover a base, for all bases and samples that pass the two consequentially
filtering steps; if all the samples cover all the bases, then <em>index</em> is NULL </p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># Two samples and two bases whose reference allele is A and C.
# The two samples have 100 reads in the reference allele and 0 in all the others.
sample1_A &lt;- c(100, 0, 0, 0)
names_A &lt;- rep("1_A", length(sample1_A))
sample1_C &lt;- c(100, 0, 0, 0)
names_C &lt;- rep("2_C", length(sample1_C))
allele &lt;- c("A", "C", "T", "G")
names_A_allele &lt;- paste(names_A, allele, sep = " ")
names_C_allele &lt;- paste(names_C, allele, sep = " ")
sample1 &lt;- c(sample1_A, sample1_C)
sample2_A &lt;- c(100, 0, 0, 0)
sample2_C &lt;- c(100, 0, 0, 0)
sample2 &lt;- c(sample2_A, sample2_C)
test_allele &lt;- matrix(c(sample1, sample2), byrow = TRUE, ncol = 8, nrow = 2)
colnames(test_allele) &lt;- c(names_A_allele, names_C_allele)
row.names(test_allele) &lt;- c("sample1", "sample2")
name_position_allele_test &lt;- c(names_A_allele, names_C_allele)
name_position_test &lt;- c(names_A, names_C)
test &lt;- get_heteroplasmy(test_allele, name_position_allele_test, name_position_test, 50, 1, 1)
</code></pre>

<hr>
<h2 id='get_raw_counts_allele'>get_raw_counts_allele</h2><span id='topic+get_raw_counts_allele'></span>

<h3>Description</h3>

<p>It is one the two main function of the <strong>MitoHEAR</strong> package (together
with <em>get_heteroplasmy</em>). The function allows to obtain a matrix of
counts (n_row = number of sample, n_col= 4*number of bases) of the four
alleles in each base, for every sample. It takes as input a vector of sorted
bam files (one bam file for each sample) and a fasta file for the genomic
region of interest. It is based on the <em>pileup</em> function of the package
Rsamtools.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_raw_counts_allele(bam_input, path_fasta, cell_names, cores_number = 1)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_raw_counts_allele_+3A_bam_input">bam_input</code></td>
<td>
<p>Character vector of sorted bam files (full path). Each
sample is defined by one bam file. For each bam file it is needed also the
index bam file (.bai) at the same path.</p>
</td></tr>
<tr><td><code id="get_raw_counts_allele_+3A_path_fasta">path_fasta</code></td>
<td>
<p>Character string with full path to the fasta file of the
genomic region of interest.</p>
</td></tr>
<tr><td><code id="get_raw_counts_allele_+3A_cell_names">cell_names</code></td>
<td>
<p>Character vector of sample names.</p>
</td></tr>
<tr><td><code id="get_raw_counts_allele_+3A_cores_number">cores_number</code></td>
<td>
<p>Number of cores to use.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with three elements:
</p>
<table role = "presentation">
<tr><td><code>matrix_allele_counts</code></td>
<td>
<p>Matrix of counts (n_row = number of sample,
n_col= 4*number of bases) of the four alleles in each base, for every
sample. The row names is equal to cell_names. </p>
</td></tr>
<tr><td><code>name_position_allele</code></td>
<td>
<p>Character vector with length equal to n_col of
matrix_allele_counts. Each element specifies the coordinate of genomic
position for a base and the allele.</p>
</td></tr> <tr><td><code>name_position</code></td>
<td>
<p>Character vector
with length equal to n_col of matrix_allele_counts. Each element specifies
the coordinate of genomic position for a base.</p>
</td></tr>
</table>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/Rsamtools/versions/1.24.0/topics/pileup">https://www.rdocumentation.org/packages/Rsamtools/versions/1.24.0/topics/pileup</a>
</p>

<hr>
<h2 id='get_wilcox_test'>get_wilcox_test</h2><span id='topic+get_wilcox_test'></span>

<h3>Description</h3>

<p>get_wilcox_test
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_wilcox_test(heteroplasmy_matrix, cluster, label_1, label_2, index = NULL)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="get_wilcox_test_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="get_wilcox_test_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="get_wilcox_test_+3A_label_1">label_1</code></td>
<td>
<p>Character name of a first label included in cluster. It
denotes the first group used for the Wilcoxon test</p>
</td></tr>
<tr><td><code id="get_wilcox_test_+3A_label_2">label_2</code></td>
<td>
<p>Character name of a second label included in cluster and
different from label_1. it denotes the second group used for the Wilcoxon
test.</p>
</td></tr>
<tr><td><code id="get_wilcox_test_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>It returns a numeric vector of length equal to n_row in matrix. Each element
stands for a base and it contains the adjusted p-value (FDR), obtained in
unpaired two-samples Wilcoxon test from the comparison of the heteroplasmy
between the label_1 and label_2 group.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/wilcox.test">https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/wilcox.test</a>
</p>

<hr>
<h2 id='plot_allele_frequency'>plot_allele_frequency</h2><span id='topic+plot_allele_frequency'></span>

<h3>Description</h3>

<p>plot_allele_frequency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_allele_frequency(
  position,
  heteroplasmy_matrix,
  allele_matrix,
  cluster,
  names_allele_qc,
  names_position_qc,
  size_text,
  index
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_allele_frequency_+3A_position">position</code></td>
<td>
<p>Character name of the base to plot.</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_allele_matrix">allele_matrix</code></td>
<td>
<p>Fourth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_names_allele_qc">names_allele_qc</code></td>
<td>
<p>Character vector with length equal to n_col of
<em>allele_matrix</em>. Each element specifies the name of the base and the allele.</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_names_position_qc">names_position_qc</code></td>
<td>
<p>Character vector with length equal to n_col of
<em>allele_matrix</em>. Each element specifies the name of the base.</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_size_text">size_text</code></td>
<td>
<p>Character specifying the size of the text for <em>gridExtra</em> function
<em>grid.arrange</em>)</p>
</td></tr>
<tr><td><code id="plot_allele_frequency_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>grid.arrange plot of allele frequencies of a specific base
across samples divided according to cluster.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://cran.r-project.org/package=gridExtra">https://cran.r-project.org/package=gridExtra</a>
</p>

<hr>
<h2 id='plot_base_coverage'>plot_base_coverage</h2><span id='topic+plot_base_coverage'></span>

<h3>Description</h3>

<p>plot_base_coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_base_coverage(
  sum_matrix,
  sum_matrix_qc,
  selected_cells,
  interactive = FALSE,
  text_size = 10
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_base_coverage_+3A_sum_matrix">sum_matrix</code></td>
<td>
<p>First element returned by the function
<em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_base_coverage_+3A_sum_matrix_qc">sum_matrix_qc</code></td>
<td>
<p>Second element returned by the function
<em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_base_coverage_+3A_selected_cells">selected_cells</code></td>
<td>
<p>Character vector with cells used fro plotting the
coverage.</p>
</td></tr>
<tr><td><code id="plot_base_coverage_+3A_interactive">interactive</code></td>
<td>
<p>Logical. If TRUE an interactive plot is produced.</p>
</td></tr>
<tr><td><code id="plot_base_coverage_+3A_text_size">text_size</code></td>
<td>
<p>Character specifying the size of the text for ggplot2.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object (if <em>interactive</em>=FALSE) or plotly object
(if <em>interactive</em>=TRUE).
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://plotly.com/r/">https://plotly.com/r/</a>
</p>

<hr>
<h2 id='plot_batch'>plot_batch</h2><span id='topic+plot_batch'></span>

<h3>Description</h3>

<p>plot_batch
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_batch(position, heteroplasmy_matrix, batch, cluster, text_size, index)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_batch_+3A_position">position</code></td>
<td>
<p>Character name of the base to plot.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_batch">batch</code></td>
<td>
<p>Vector of batch names,with length equal to n_row of
<em>heteroplasmy_matrix</em>.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_text_size">text_size</code></td>
<td>
<p>Character specifying the size of the text for ggplot2.</p>
</td></tr>
<tr><td><code id="plot_batch_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object of the heteroplasmy level of a specific base across
samples divided according to batch.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_cells_coverage'>plot_cells_coverage</h2><span id='topic+plot_cells_coverage'></span>

<h3>Description</h3>

<p>plot_cells_coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_cells_coverage(sum_matrix, cells_selected, cluster, interactive = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_cells_coverage_+3A_sum_matrix">sum_matrix</code></td>
<td>
<p>First element returned by the function
<em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_cells_coverage_+3A_cells_selected">cells_selected</code></td>
<td>
<p>Character vector of cells for which the coverage is
computed.</p>
</td></tr>
<tr><td><code id="plot_cells_coverage_+3A_cluster">cluster</code></td>
<td>
<p>Character vector with partition information for cells
specified in <em>cells_selected</em></p>
</td></tr>
<tr><td><code id="plot_cells_coverage_+3A_interactive">interactive</code></td>
<td>
<p>Logical. If TRUE an interactive plot is produced.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object (if <em>interactive</em>=FALSE) or plotly object (if
<em>interactive</em>=TRUE).
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://plotly.com/r/">https://plotly.com/r/</a>
</p>

<hr>
<h2 id='plot_condition'>plot_condition</h2><span id='topic+plot_condition'></span>

<h3>Description</h3>

<p>plot_condition
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_condition(
  distribution_1,
  distribution_2,
  label_1,
  label_2,
  name_x,
  name_y,
  name_title
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_condition_+3A_distribution_1">distribution_1</code>, <code id="plot_condition_+3A_distribution_2">distribution_2</code></td>
<td>
<p>Numeric vector</p>
</td></tr>
<tr><td><code id="plot_condition_+3A_label_1">label_1</code></td>
<td>
<p>Character vector of length equal to distribution_1</p>
</td></tr>
<tr><td><code id="plot_condition_+3A_label_2">label_2</code></td>
<td>
<p>Character vector of length equal to distribution_2</p>
</td></tr>
<tr><td><code id="plot_condition_+3A_name_x">name_x</code></td>
<td>
<p>Character name specifying the xlab argument in <em>ggplot2</em>.</p>
</td></tr>
<tr><td><code id="plot_condition_+3A_name_y">name_y</code></td>
<td>
<p>Character name specifying the ylab argument in ggplot2.</p>
</td></tr>
<tr><td><code id="plot_condition_+3A_name_title">name_title</code></td>
<td>
<p>Character name specifying the ggtitle argument in <em>ggplot2</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 boxplot of the quantities specified by <em>distribution_1</em>
and <em>distribution_2</em>, separated by the conditions denoted by
<em>label_1</em> and <em>label_2</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_coordinate_cluster'>plot_coordinate_cluster</h2><span id='topic+plot_coordinate_cluster'></span>

<h3>Description</h3>

<p>plot_coordinate_cluster
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_coordinate_cluster(coordinate_dm, cluster)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_coordinate_cluster_+3A_coordinate_dm">coordinate_dm</code></td>
<td>
<p>Dataframe whit samples on the rows and coordinates
names on the columns.</p>
</td></tr>
<tr><td><code id="plot_coordinate_cluster_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_coordinate_heteroplasmy'>plot_coordinate_heteroplasmy</h2><span id='topic+plot_coordinate_heteroplasmy'></span>

<h3>Description</h3>

<p>plot_coordinate_heteroplasmy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_coordinate_heteroplasmy(
  coordinate_dm,
  heteroplasmy_matrix,
  index,
  name_base
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_coordinate_heteroplasmy_+3A_coordinate_dm">coordinate_dm</code></td>
<td>
<p>Dataframe whit samples on the rows and coordinates
names on the columns.</p>
</td></tr>
<tr><td><code id="plot_coordinate_heteroplasmy_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_coordinate_heteroplasmy_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_coordinate_heteroplasmy_+3A_name_base">name_base</code></td>
<td>
<p>Character name specifying the base.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_correlation_bases'>plot_correlation_bases</h2><span id='topic+plot_correlation_bases'></span>

<h3>Description</h3>

<p>plot_correlation_bases
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_correlation_bases(bases_vector, index, heteroplasmy_matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_correlation_bases_+3A_bases_vector">bases_vector</code></td>
<td>
<p>Character vector specifying the bases for which the
spearman correlation across samples is computed.</p>
</td></tr>
<tr><td><code id="plot_correlation_bases_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_correlation_bases_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Heatmap plot produced by function <em>Heatmap</em> from package <em>ComplexHeatmap</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/ComplexHeatmap/versions/1.10.2/topics/Heatmap">https://www.rdocumentation.org/packages/ComplexHeatmap/versions/1.10.2/topics/Heatmap</a>
</p>

<hr>
<h2 id='plot_distance_matrix'>plot_distance_matrix</h2><span id='topic+plot_distance_matrix'></span>

<h3>Description</h3>

<p>plot_distance_matrix
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_distance_matrix(dist_ang_matrix, cluster)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_distance_matrix_+3A_dist_ang_matrix">dist_ang_matrix</code></td>
<td>
<p>Distance matrix obtained from <em>clustering_angular_distance</em>
(second element of the output).</p>
</td></tr>
<tr><td><code id="plot_distance_matrix_+3A_cluster">cluster</code></td>
<td>
<p>Vector.Can be one of the two partitions returned by
function <em>clustering_angular_distance</em> (first element of the output).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Heatmap plot produced by function <em>Heatmap</em> from package <em>ComplexHeatmap</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/ComplexHeatmap/versions/1.10.2/topics/Heatmap">https://www.rdocumentation.org/packages/ComplexHeatmap/versions/1.10.2/topics/Heatmap</a>
</p>

<hr>
<h2 id='plot_distribution'>plot_distribution</h2><span id='topic+plot_distribution'></span>

<h3>Description</h3>

<p>plot_distribution
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_distribution(quantity_counts_cell, name_x, name_title)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_distribution_+3A_quantity_counts_cell">quantity_counts_cell</code></td>
<td>
<p>Vector returned by
<em>get_distribution</em></p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_name_x">name_x</code></td>
<td>
<p>Character name specifying the xlab argument in <em>ggplot2</em>.</p>
</td></tr>
<tr><td><code id="plot_distribution_+3A_name_title">name_title</code></td>
<td>
<p>Character name specifying the ggtitle argument in <em>ggplot2</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 density plot of the vector <em>quantity_counts_cell</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_dpt'>plot_dpt</h2><span id='topic+plot_dpt'></span>

<h3>Description</h3>

<p>plot_dpt
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_dpt(position, heteroplasmy_matrix, cluster, time, gam_fit_result, index)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_dpt_+3A_position">position</code></td>
<td>
<p>Character name of the base to plot.</p>
</td></tr>
<tr><td><code id="plot_dpt_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_dpt_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="plot_dpt_+3A_time">time</code></td>
<td>
<p>Vector of diffusion pseudo time,with length equal to n_row of
<em>heteroplasmy_matrix</em>.</p>
</td></tr>
<tr><td><code id="plot_dpt_+3A_gam_fit_result">gam_fit_result</code></td>
<td>
<p>Data frame returned by <em>dpt_test</em>.</p>
</td></tr>
<tr><td><code id="plot_dpt_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object of the heteroplasmy level of a specific base across
samples and the GAM fitted curve. The title shows the adjusted p value (FDR)
for the position obtained from <em>get_heteroplasmy</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://cran.r-project.org/package=gam">https://cran.r-project.org/package=gam</a>
</p>

<hr>
<h2 id='plot_genome_coverage'>plot_genome_coverage</h2><span id='topic+plot_genome_coverage'></span>

<h3>Description</h3>

<p>plot_genome_coverage
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_genome_coverage(biomart_file, path_fasta, chr_name, heteroplasmy_matrix)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_genome_coverage_+3A_biomart_file">biomart_file</code></td>
<td>
<p>Character string with full path to the txt file
downloaded from BioMart <a href="https://m.ensembl.org/info/data/biomart/index.html">https://m.ensembl.org/info/data/biomart/index.html</a> . It must have
the following five columns:Gene.stable.ID, Gene.name, Gene.start..bp.,
Gene.end..bp., Chromosome.scaffold.name</p>
</td></tr>
<tr><td><code id="plot_genome_coverage_+3A_path_fasta">path_fasta</code></td>
<td>
<p>Character string with full path to the fasta file of the
genomic region of interest. It should be the same file used in
<em>get_raw_counts_allele</em>.</p>
</td></tr>
<tr><td><code id="plot_genome_coverage_+3A_chr_name">chr_name</code></td>
<td>
<p>Character specifying the name of the chromosome of interest.
It must be one of the names in the <em>Chromosome.scaffold.name</em> column
from the <em>biomart_file</em>.</p>
</td></tr>
<tr><td><code id="plot_genome_coverage_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>KaryoPlot object as returned by <em>plotKaryotype</em> function from package <em>karyoploteR</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="http://bioconductor.org/packages/release/bioc/html/karyoploteR.html">http://bioconductor.org/packages/release/bioc/html/karyoploteR.html</a>
</p>

<hr>
<h2 id='plot_heatmap'>plot_heatmap</h2><span id='topic+plot_heatmap'></span>

<h3>Description</h3>

<p>plot_heatmap
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heatmap(
  new_classification,
  old_classification,
  dist_ang_matrix,
  cluster_columns = FALSE,
  cluster_rows = TRUE,
  name_legend
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heatmap_+3A_new_classification">new_classification</code></td>
<td>
<p>Character vector.Second column of the dataframe
returned by function <em>clustering_angular_distance</em> (first element of the
output).</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_old_classification">old_classification</code></td>
<td>
<p>Character vector. First column of the dataframe
returned by function <em>clustering_angular_distance</em> (first element of the
output).</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_dist_ang_matrix">dist_ang_matrix</code></td>
<td>
<p>Distance matrix obtained from <em>clustering_angular_distance</em>
(second element of the output).</p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_cluster_columns">cluster_columns</code></td>
<td>
<p>Logical. Parameter for cluster_columns argument of
the function <em>Heatmap</em> in the package <em>ComplexHeatmap</em></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_cluster_rows">cluster_rows</code></td>
<td>
<p>Logical. Parameter for cluster_rows argument of the
function <em>Heatmap</em></p>
</td></tr>
<tr><td><code id="plot_heatmap_+3A_name_legend">name_legend</code></td>
<td>
<p>Character value.Parameter for name argument of the
function <em>Heatmap</em></p>
</td></tr>
</table>


<h3>Value</h3>

<p>Heatmap plot produced by function <em>Heatmap</em> from package <em>ComplexHeatmap</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/ComplexHeatmap/versions/1.10.2/topics/Heatmap">https://www.rdocumentation.org/packages/ComplexHeatmap/versions/1.10.2/topics/Heatmap</a>
</p>

<hr>
<h2 id='plot_heteroplasmy'>plot_heteroplasmy</h2><span id='topic+plot_heteroplasmy'></span>

<h3>Description</h3>

<p>plot_heteroplasmy
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heteroplasmy(position, heteroplasmy_matrix, cluster, index)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heteroplasmy_+3A_position">position</code></td>
<td>
<p>Character name of the base to plot.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object of the heteroplasmy level of a specific base across
samples divided according to cluster.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_heteroplasmy_variability'>plot_heteroplasmy_variability</h2><span id='topic+plot_heteroplasmy_variability'></span>

<h3>Description</h3>

<p>plot_heteroplasmy_variability
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_heteroplasmy_variability(
  heteroplasmy_matrix,
  cluster,
  threshold = 0.1,
  frac = FALSE,
  index
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_heteroplasmy_variability_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_variability_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_variability_+3A_threshold">threshold</code></td>
<td>
<p>Numeric value.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_variability_+3A_frac">frac</code></td>
<td>
<p>Logical. If FALSE the absolute number of cells that have at
least one base with heteroplasmy above <em>threshold</em> are shown separated by
<em>cluster</em>. If TRUE, then the fraction of cells are shown.</p>
</td></tr>
<tr><td><code id="plot_heteroplasmy_variability_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>ggplot2 object.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>

<hr>
<h2 id='plot_spider_chart'>plot_spider_chart</h2><span id='topic+plot_spider_chart'></span>

<h3>Description</h3>

<p>plot_spider_chart
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plot_spider_chart(name_base, cluster, heteroplasmy_matrix, index)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="plot_spider_chart_+3A_name_base">name_base</code></td>
<td>
<p>Character name specifying the base.</p>
</td></tr>
<tr><td><code id="plot_spider_chart_+3A_cluster">cluster</code></td>
<td>
<p>Vector specifying a partition of the samples.</p>
</td></tr>
<tr><td><code id="plot_spider_chart_+3A_heteroplasmy_matrix">heteroplasmy_matrix</code></td>
<td>
<p>Third element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
<tr><td><code id="plot_spider_chart_+3A_index">index</code></td>
<td>
<p>Fifth element returned by <em>get_heteroplasmy</em>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>radarchart plot produced by function <em>radarchart</em> from package <em>fmsb</em>.
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://rdrr.io/cran/fmsb/man/radarchart.html">https://rdrr.io/cran/fmsb/man/radarchart.html</a>
</p>

<hr>
<h2 id='vi_comparison'>vi_comparison
We compute the variation of information (VI) between the partition provided
by <em>new_classification</em> and <em>old_classification</em>. The VI between a
random partitions (obtained with re-shuffle from original labels in
<em>old_classification</em>) and <em>old_classification</em> is also computed. A
distribution of VI values from random partitions is built. Finally, from the
comparison with this distribution, an empirical p value is given to the VI
of the unsupervised cluster analysis.</h2><span id='topic+vi_comparison'></span>

<h3>Description</h3>

<p>vi_comparison
We compute the variation of information (VI) between the partition provided
by <em>new_classification</em> and <em>old_classification</em>. The VI between a
random partitions (obtained with re-shuffle from original labels in
<em>old_classification</em>) and <em>old_classification</em> is also computed. A
distribution of VI values from random partitions is built. Finally, from the
comparison with this distribution, an empirical p value is given to the VI
of the unsupervised cluster analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vi_comparison(old_classification, new_classification, number_iter)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="vi_comparison_+3A_old_classification">old_classification</code></td>
<td>
<p>Character vector. First column of the dataframe
returned by function <em>clustering_angular_distance</em> (first element of the
output).</p>
</td></tr>
<tr><td><code id="vi_comparison_+3A_new_classification">new_classification</code></td>
<td>
<p>Character vector.Second column of the dataframe
returned by function <em>clustering_angular_distance</em> (first element of the
output).</p>
</td></tr>
<tr><td><code id="vi_comparison_+3A_number_iter">number_iter</code></td>
<td>
<p>Integer value. Specify how many random partition are
generated (starting from re-shuffle of labels in <em>old_classification</em>).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Numeric value (empirical p value).
</p>


<h3>Author(s)</h3>

<p>Gabriele Lubatti <a href="mailto:gabriele.lubatti@helmholtz-muenchen.de">gabriele.lubatti@helmholtz-muenchen.de</a>
</p>


<h3>See Also</h3>

<p><a href="https://www.rdocumentation.org/packages/mcclust/versions/1.0/topics/vi.dist">https://www.rdocumentation.org/packages/mcclust/versions/1.0/topics/vi.dist</a>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
