<!DOCTYPE html><html><head><title>Help for package mixedBayes</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mixedBayes}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mixedBayes-package'><p>Bayesian Regularized Quantile Mixed Model for G - E Interactions</p></a></li>
<li><a href='#data'><p>simulated data for demonstrating the features of mixedBayes</p></a></li>
<li><a href='#mixedBayes'><p>fit a Bayesian regularized quantile mixed model for G - E interactions</p></a></li>
<li><a href='#selection'><p>Variable selection for a mixedBayes object</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Bayesian Longitudinal Regularized Quantile Mixed Model</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Date:</td>
<td>2024-02-16</td>
</tr>
<tr>
<td>Description:</td>
<td>In longitudinal studies, the same subjects are measured repeatedly over time, leading to correlations among the repeated measurements. Properly accounting for the intra-cluster correlations in the presence of data heterogeneity and long tailed distributions of the disease phenotype is challenging, especially in the context of high dimensional regressions. Here, we aim at developing novel Bayesian regularized quantile mixed effect models to tackle these challenges. We have proposed a Bayesian variable selection in the mixed effect models for longitudinal genomics studies. To dissect important gene - environment interactions, our model can simultaneously identify important main and interaction effects on the individual and group level, which have been facilitated by imposing the spike- and -slab priors through Laplacian shrinkage in the Bayesian quantile hierarchical models. The within - subject dependence among data can be accommodated by incorporating the random effects. An efficient Gibbs sampler has been developed to facilitate fast computation. The Markov chain Monte Carlo algorithms of the proposed and alternative methods are efficiently implemented in 'C++'. The development of this software package and the associated statistical methods have been partially supported by an Innovative Research Award from Johnson Cancer Research Center, Kansas State University.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.2.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/kunfa/mixedBayes">https://github.com/kunfa/mixedBayes</a></td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp, RcppArmadillo</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-02-16 19:22:45 UTC; grimf</td>
</tr>
<tr>
<td>Author:</td>
<td>Kun Fan [aut, cre],
  Cen Wu [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kun Fan &lt;kfan@ksu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-02-16 19:40:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='mixedBayes-package'>Bayesian Regularized Quantile Mixed Model for G - E Interactions</h2><span id='topic+mixedBayes-package'></span>

<h3>Description</h3>

<p>In this package, we provide a set of Bayesian regularized variable selection methods under the mixed effect models (random intercept and slope model, random intercept model) to dissect important gene - environment interactions for longitudinal studies. A Bayesian quantile regression has been adopted to accommodate data contamination and heavy-tailed distributions in the response/ phenotype. The default method (the proposed method) conducts variable selection by accounting the group level selection on the interaction effects under random intercept and slope model. In particular, the spike&ndash;and&ndash;slab priors are imposed on both individual and group levels to identify important main and interaction effects.
In addition to the default method, users can also choose different selection structures for the interaction effects (group-level or individual-level), random intercept model, methods without spike&ndash;and&ndash;slab priors and non-robust methods. In total, <em>mixedBayes</em> provides 16 different methods (8 robust and 8 non-robust) under both mixed effects models. Among them, robust methods with spike&ndash;and&ndash;slab priors and the robust method for both individual level selection and group level selection under both mixed effects models have been developed for the first time. Please read the Details below for how to configure the method used.
</p>


<h3>Details</h3>

<p>The user friendly, integrated interface <strong>mixedBayes()</strong> allows users to flexibly choose the fitting methods by specifying the following parameter:
</p>

<table>
<tr>
 <td style="text-align: right;">
slope: </td><td style="text-align: left;"> whether to use random intercept and slope model.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
robust: </td><td style="text-align: left;"> whether to use robust methods for modelling.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
quant: </td><td style="text-align: left;"> to specify different quantiles when using robust methods.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
structure: </td><td style="text-align: left;"> structure for interaction effects.</td>
</tr>
<tr>
 <td style="text-align: right;"></td>
</tr>
<tr>
 <td style="text-align: right;">
sparse: </td><td style="text-align: left;"> whether to use the spike-and-slab priors to impose sparsity.
</td>
</tr>

</table>

<p>The function mixedBayes() returns a mixedBayes object that contains the posterior estimates of each coefficients.
S3 generic functions selection()and print() are implemented for mixedBayes objects.
selection() takes a mixedBayes object and returns the variable selection results.
</p>


<h3>References</h3>

<p>Fan, K., Wang, W. and Wu, C. (2022). Bayesian Regularized Quantile Mixed Model for Lipid - Environment Interactions in a Longitudinal Study. (to be submitted)
</p>
<p>Zhou, F., Ren, J.,  Li, G., Jiang, Y., Li, X., Wang, W. and Wu, C. (2019). Penalized Variable Selection for Lipid-Environment Interactions in a Longitudinal Lipidomics Study.
<em>Genes</em>, 10(12), 1002 <a href="https://doi.org/10.3390/genes10121002">doi:10.3390/genes10121002</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2022). Robust Bayesian variable selection for gene-environment interactions.
<em>Biometrics</em>, (in press) <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2020). roben: Robust Bayesian Variable Selection for Gene-Environment Interactions.
R package version 0.1.1. <a href="https://CRAN.R-project.org/package=roben">https://CRAN.R-project.org/package=roben</a>
</p>
<p>Wu, C., and Ma, S. (2015). A selective review of robust variable selection with applications in bioinformatics.
<em>Briefings in Bioinformatics</em>, 16(5), 873–883 <a href="https://doi.org/10.1093/bib/bbu046">doi:10.1093/bib/bbu046</a>
</p>
<p>Zhou, F., Ren, J., Lu, X., Ma, S. and Wu, C. (2021). Gene–Environment Interaction: a Variable Selection Perspective.
<em>Epistasis. Methods in Molecular Biology.</em> 2212:191–223 <a href="https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13">https://link.springer.com/protocol/10.1007/978-1-0716-0947-7_13</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Chen, Q., Zhang, H., Ma, S., Jiang, Y. and Wu, C. (2020) Semi-parametric Bayesian variable selection for gene-environment interactions.
<em>Statistics in Medicine</em>, 39: 617– 638 <a href="https://doi.org/10.1002/sim.8434">doi:10.1002/sim.8434</a>
</p>
<p>Ren, J., Zhou, F., Li, X., Wu, C. and Jiang, Y. (2019) spinBayes: Semi-Parametric Gene-Environment Interaction via Bayesian Variable Selection.
R package version 0.1.0. <a href="https://CRAN.R-project.org/package=spinBayes">https://CRAN.R-project.org/package=spinBayes</a>
</p>
<p>Wu, C., Jiang, Y., Ren, J., Cui, Y. and Ma, S. (2018). Dissecting gene-environment interactions: A penalized robust approach accounting for hierarchical structures.
<em>Statistics in Medicine</em>, 37:437–456 <a href="https://doi.org/10.1002/sim.7518">doi:10.1002/sim.7518</a>
</p>
<p>Wu, C., Cui, Y., and Ma, S. (2014). Integrative analysis of gene–environment interactions under a multi–response partially linear varying coefficient model.
<em>Statistics in Medicine</em>, 33(28), 4988–4998 <a href="https://doi.org/10.1002/sim.6287">doi:10.1002/sim.6287</a>
</p>
<p>Wu, C., Zhong, P.S. and Cui, Y. (2013). High dimensional variable selection for gene-environment interactions.
<em>Technical Report. Michigan State University.</em>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixedBayes">mixedBayes</a></code>
</p>

<hr>
<h2 id='data'>simulated data for demonstrating the features of mixedBayes</h2><span id='topic+data'></span><span id='topic+y'></span><span id='topic+e'></span><span id='topic+X'></span><span id='topic+g'></span><span id='topic+w'></span><span id='topic+k'></span><span id='topic+coeff'></span>

<h3>Description</h3>

<p>Simulated gene expression data for demonstrating the features of mixedBayes.
</p>


<h3>Format</h3>

<p>The data object consists of seven components: y, e, X, g, w ,k and coeff. coeff contains the true values of parameters used for generating Y.
</p>


<h3>Details</h3>

<p><strong>The data and model setting</strong>
</p>
<p>Consider a longitudinal study on <code class="reqn">n</code> subjects with <code class="reqn">k</code> repeated measurement for each subject. Let <code class="reqn">Y_{ij}</code> be the measurement for the <code class="reqn">i</code>th subject at each time point <code class="reqn">j</code>(<code class="reqn">1\leq i \leq n, 1\leq j \leq k</code>) .We use a <code class="reqn">m</code>-dimensional vector <code class="reqn">G_{ij}</code> to denote the genetics factors, where <code class="reqn">G_{ij} = (G_{ij1},...,G_{ijm})^{T}</code>. Also, we use <code class="reqn">p</code>-dimensional vector <code class="reqn">E_{ij}</code> to denote the environment factors, where <code class="reqn">E_{ij} = (E_{ij1},...,E_{ijp})^{T}</code>. <code class="reqn">X_{ij} = (1, T_{ij})^{T}</code>, where <code class="reqn">T_{ij}^{T}</code> is a vector of time effects . <code class="reqn">Z_{ij}</code> is a <code class="reqn">h \times 1</code> covariate associated with random effects and <code class="reqn">\alpha_{i}</code> is a <code class="reqn">h\times 1</code> vector of random effects.  At the beginning,  the interaction effects is modeled as the product of genomics features and environment factors with 4 different levels. After representing the environment factors as three dummy variables, the identification of the gene by environment interaction needs to be performed as group level.  Combing the genetics factors, environment factors and their interactions that associated with the longitudinal phenotype, we have the following mixed-effects model:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij} = X_{ij}^{T}\gamma_{0}+E_{ij}^{T}\gamma_{1}+G_{ij}^{T}\gamma_{2}+(G_{ij}\bigotimes E_{ij})^{T}\gamma_{3}+Z_{ij}^{T}\alpha_{i}+\epsilon_{ij}.</code>
</p>

<p>where <code class="reqn">\gamma_{1}</code>,<code class="reqn">\gamma_{2}</code>,<code class="reqn">\gamma_{3}</code> are <code class="reqn">p</code>,<code class="reqn">m</code> and <code class="reqn">mp</code> dimensional vectors that represent the coefficients of the environment effects, the genetics effects and interactions effects, respectively. Accommodating the Kronecker product of the <code class="reqn">m</code> - dimensional vector <code class="reqn">G_{ij}</code> and the <code class="reqn">p</code>-dimensional vector <code class="reqn">E_{ij}</code>, the interactions between genetics and environment factors can be expressed as a <code class="reqn">mp</code>-dimensional vector, denoted as the following form:
</p>
<p style="text-align: center;"><code class="reqn">G_{ij}\bigotimes E_{ij} = [E_{ij1}E_{ij1},E_{ij2}E_{ij2},...,E_{ij1}E_{ijp},E_{ij2}E_{ij1},...,E_{ijm}E_{ijp}]^{T}.</code>
</p>

<p>For random intercept and slope model, <code class="reqn">Z_{ij}^{T} = (1,j)</code> and  <code class="reqn">\alpha_{i} = (\alpha_{i1},\alpha_{i2})^{T}</code>. For random intercept model, <code class="reqn">Z_{ij}^{T} = 1</code> and <code class="reqn">\alpha_{i} = \alpha_{i1}</code>.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixedBayes">mixedBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)
dim(y)
dim(g)
dim(e)
dim(w)
print(k)
print(X)
print(coeff)

</code></pre>

<hr>
<h2 id='mixedBayes'>fit a Bayesian regularized quantile mixed model for G - E interactions</h2><span id='topic+mixedBayes'></span>

<h3>Description</h3>

<p>fit a Bayesian regularized quantile mixed model for G - E interactions
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixedBayes(
  y,
  e,
  X,
  g,
  w,
  k,
  iterations = 10000,
  burn.in = NULL,
  slope = TRUE,
  robust = TRUE,
  quant = 0.5,
  sparse = TRUE,
  structure = c("group", "individual")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mixedBayes_+3A_y">y</code></td>
<td>
<p>the matrix of response variable. The current version of mixedBayes only supports continuous response.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_e">e</code></td>
<td>
<p>the matrix of a group of dummy environmental factors variables.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_x">X</code></td>
<td>
<p>the matrix of the intercept and time effects (time effects are optional).</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_g">g</code></td>
<td>
<p>the matrix of predictors (genetic factors) without intercept. Each row should be an observation vector.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_w">w</code></td>
<td>
<p>the matrix of interactions between genetic factors and environmental factors.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_k">k</code></td>
<td>
<p>the total number of time points.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_iterations">iterations</code></td>
<td>
<p>the number of MCMC iterations.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_burn.in">burn.in</code></td>
<td>
<p>the number of iterations for burn-in.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_slope">slope</code></td>
<td>
<p>logical flag. If TRUE, random intercept and slope model will be used.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_robust">robust</code></td>
<td>
<p>logical flag. If TRUE, robust methods will be used.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_quant">quant</code></td>
<td>
<p>specify different quantiles when applying robust methods.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly.</p>
</td></tr>
<tr><td><code id="mixedBayes_+3A_structure">structure</code></td>
<td>
<p>structure for interaction effects, two choices are available. &quot;group&quot; for selection on group-level only. &quot;individual&quot; for selection on individual-level only.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Consider the data model described in &quot;<code><a href="#topic+data">data</a></code>&quot;:
</p>
<p style="text-align: center;"><code class="reqn">Y_{ij} = X_{ij}^{T}\gamma_{0}+E_{ij}^{T}\gamma_{1}+\sum_{l=1}^{p}G_{ijl}\gamma_{2l}+\sum_{l=1}^{p}W_{ijl}^{T}\gamma_{3l}+Z_{ij}^{T}\alpha_{i}+\epsilon_{ij}.</code>
</p>

<p>where <code class="reqn">\gamma_{2l}</code> is the main effect of the <code class="reqn">l</code>th genetic variant. The interaction effects is corresponding to the coefficient vector <code class="reqn">\gamma_{3l}=(\gamma_{3l1}, \gamma_{3l2},\ldots,\gamma_{3lm})^\top</code>.
</p>
<p>When structure=&quot;group&quot;, group-level selection will be conducted on <code class="reqn">||\gamma_{3l}||_{2}</code>. If structure=&quot;individual&quot;, individual-level selection will be conducted on each <code class="reqn">\gamma_{3lq}</code>, (<code class="reqn">q=1,\ldots,m</code>).
</p>
<p>When slope=TRUE (default), random intercept and slope model will be used as the mixed effects model.
</p>
<p>When sparse=TRUE (default), spike&ndash;and&ndash;slab priors are imposed on individual and/or group levels to identify important main and interaction effects. Otherwise, Laplacian shrinkage will be used.
</p>
<p>When robust=TRUE (default), the distribution of <code class="reqn">\epsilon_{ij}</code> is defined as a Laplace distribution with density.
</p>
<p><code class="reqn">
f(\epsilon_{ij}|\theta,\tau) = \theta(1-\theta)\exp\left\{-\tau\rho_{\theta}(\epsilon_{ij})\right\}
</code>, (<code class="reqn">i=1,\dots,n,j=1,\dots,k </code>), which leads to a Bayesian formulation of quantile regression. If robust=FALSE, <code class="reqn">\epsilon_{ij}</code> follows a normal distribution.
<br />
</p>
<p>Please check the references for more details about the prior distributions.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;mixedBayes&rsquo; is returned, which is a list with component:
</p>
<table>
<tr><td><code>posterior</code></td>
<td>
<p>the posteriors of coefficients.</p>
</td></tr>
<tr><td><code>coefficient</code></td>
<td>
<p>the estimated coefficients.</p>
</td></tr>
<tr><td><code>burn.in</code></td>
<td>
<p>the total number of burn-ins.</p>
</td></tr>
<tr><td><code>iterations</code></td>
<td>
<p>the total number of iterations.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+data">data</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)

## default method
fit = mixedBayes(y,e,X,g,w,k,structure=c("group"))
fit$coefficient

## Compute TP and FP
b = selection(fit,sparse=TRUE)
index = which(coeff!=0)
pos = which(b != 0)
tp = length(intersect(index, pos))
fp = length(pos) - tp
list(tp=tp, fp=fp)

## alternative: robust individual selection
fit = mixedBayes(y,e,X,g,w,k,structure=c("individual"))
fit$coefficient

## alternative: non-robust group selection
fit = mixedBayes(y,e,X,g,w,k,robust=FALSE, structure=c("group"))
fit$coefficient

## alternative: robust group selection under random intercept model
fit = mixedBayes(y,e,X,g,w,k,slope=FALSE, structure=c("group"))
fit$coefficient



</code></pre>

<hr>
<h2 id='selection'>Variable selection for a mixedBayes object</h2><span id='topic+selection'></span>

<h3>Description</h3>

<p>Variable selection for a mixedBayes object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>selection(obj, sparse)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="selection_+3A_obj">obj</code></td>
<td>
<p>mixedBayes object.</p>
</td></tr>
<tr><td><code id="selection_+3A_sparse">sparse</code></td>
<td>
<p>logical flag. If TRUE, spike-and-slab priors will be used to shrink coefficients of irrelevant covariates to zero exactly..</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If sparse, the median probability model (MPM) (Barbieri and Berger, 2004) is used to identify predictors that are significantly associated
with the response variable. Otherwise, variable selection is based on 95% credible interval.
Please check the references for more details about the variable selection.
</p>


<h3>Value</h3>

<p>an object of class &lsquo;selection&rsquo; is returned, which is a list with component:
</p>
<table>
<tr><td><code>inde</code></td>
<td>
<p>a vector of indicators of selected effects.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Ren, J., Zhou, F., Li, X., Ma, S., Jiang, Y. and Wu, C. (2022). Robust Bayesian variable selection for gene-environment interactions.
<em>Biometrics</em>, (in press) <a href="https://doi.org/10.1111/biom.13670">doi:10.1111/biom.13670</a>
</p>
<p>Barbieri, M.M. and Berger, J.O. (2004). Optimal predictive model selection. <em>Ann. Statist</em>, 32(3):870–897
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mixedBayes">mixedBayes</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(data)
## sparse
fit = mixedBayes(y,e,X,g,w,k,structure=c("group"))
selected=selection(fit,sparse=TRUE)
selected


## non-sparse
fit = mixedBayes(y,e,X,g,w,k,sparse=FALSE,structure=c("group"))
selected=selection(fit,sparse=FALSE)
selected


</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
