<!DOCTYPE html><html lang="en"><head><title>Help for package mixIndependR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mixIndependR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#AlleleFreq'><p>Calculate Allele Frequency</p></a></li>
<li><a href='#AlleleShare'><p>Calculate numbers of sharing alleles each pair at each locus</p></a></li>
<li><a href='#ComposPare_K'><p>Generate Comparison Observed and Expected No. of Heterozygous Loci.</p></a></li>
<li><a href='#ComposPare_X'><p>Generate Comparison Observed and Expected No. of Shared Alleles.</p></a></li>
<li><a href='#counta'><p>Simple count including zero###</p></a></li>
<li><a href='#Dist_SimuChisq'><p>Build a simulated distribution for Chi-Square</p></a></li>
<li><a href='#DistAlleleShare'><p>Build Expected Distribution of Numbers of Shared Alleles</p></a></li>
<li><a href='#DistHetero'><p>Build Expected Distribution of Numbers of Heterozygous Loci</p></a></li>
<li><a href='#ExpProAlleleShare'><p>Calculate the Expected Probability of 0,1 and 2 Shared Alleles###</p></a></li>
<li><a href='#FreqAlleleShare'><p>Build Observed Distribution of No. of Shared Alleles</p></a></li>
<li><a href='#FreqHetero'><p>Build Observed Distribution of No. of Heterozygous loci</p></a></li>
<li><a href='#GenotypeFreq'><p>Calculate Genotype Frequency###</p></a></li>
<li><a href='#Heterozygous'><p>Test heterozygosity at each locus</p></a></li>
<li><a href='#HWE.Chisq'><p>Test the Hardy Weinberg Equilibrium with Chi-square test####</p></a></li>
<li><a href='#mixexample'><p>Genotype Data from A Selected Mix Panel</p></a></li>
<li><a href='#mixIndependK'><p>Quick pvalue of total number of heterozygous loci</p></a></li>
<li><a href='#mixIndependX'><p>Quick pvalue of total number of shared alleles</p></a></li>
<li><a href='#read_vcf_gt'><p>Import genotype data from vcf files/</p></a></li>
<li><a href='#RealProAlleleShare'><p>Calculate the Real Probability of 0,1 and 2 Shared Alleles###</p></a></li>
<li><a href='#RxpHetero'><p>Calculate Real or Expected Average Heterozygosity at each locus</p></a></li>
<li><a href='#Simulate_DistK'><p>Generate a Bundle of Simulated distributions for No. of heterozygous loci with known heterozygosites</p></a></li>
<li><a href='#Simulate_DistX'><p>Build a simulated distribution for No. of Shared Alleles</p></a></li>
<li><a href='#splitGenotype'><p>Split Genotype Table to Duo-Allele Table</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Genetics and Independence Testing of Mixed Genetic Panels</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Author:</td>
<td>Bing Song</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Bing Song &lt;bing.song@utsouthwestern.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.6.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats (&ge; 3.3), utils(&ge; 3.6.1), data.table</td>
</tr>
<tr>
<td>Description:</td>
<td>Developed to deal with multi-locus genotype data, this package is especially designed for those panel which include different type of markers. Basic genetic parameters like allele frequency, genotype frequency, heterozygosity and Hardy-Weinberg test of mixed genetic data can be obtained.     In addition, a new test for mutual independence which is compatible for mixed genetic data is developed in this package.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, knitr, rmarkdown, ggplot2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-03-16 19:21:59 UTC; ice4p</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/ice4prince/mixIndependR">https://github.com/ice4prince/mixIndependR</a></td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-03-17 09:30:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='AlleleFreq'>Calculate Allele Frequency</h2><span id='topic+AlleleFreq'></span>

<h3>Description</h3>

<p>Calculate Allele Frequency
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AlleleFreq(x,sep)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AlleleFreq_+3A_x">x</code></td>
<td>
<p>a dataset of genotypes. Each row denotes each individual; each column contain each marker.</p>
</td></tr>
<tr><td><code id="AlleleFreq_+3A_sep">sep</code></td>
<td>
<p>the allele separator in the imported genotype data.Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;(default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the allele frequencies of one dataset.
</p>


<h3>Value</h3>

<p>a matrix of allele frequencies. Each row denotes each allele; each column denotes each marker. The order of makers follows x.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(mixIndependR)
x &lt;- data.frame(STR1=c("12|12","13|14","13|13","14|15"),
               SNP1=c("A|A","T|T","A|T","A|T"))
AlleleFreq(x,"\\|")

</code></pre>

<hr>
<h2 id='AlleleShare'>Calculate numbers of sharing alleles each pair at each locus</h2><span id='topic+AlleleShare'></span>

<h3>Description</h3>

<p>Calculate numbers of sharing alleles each pair at each locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>AlleleShare(df,sep,replacement=FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="AlleleShare_+3A_df">df</code></td>
<td>
<p>a dataframe of genotype data with rownames of sample ID and column names of markers.</p>
</td></tr>
<tr><td><code id="AlleleShare_+3A_sep">sep</code></td>
<td>
<p>allele separator in the imported genotype data. Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;(default).</p>
</td></tr>
<tr><td><code id="AlleleShare_+3A_replacement">replacement</code></td>
<td>
<p>a logical variable. If it is TRUE, the pairs are sampled with replacement; if FALSE (default), the pairs are sampled without replacement.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the numbers of shared alleles between each pair of individuals for a dataset.
</p>


<h3>Value</h3>

<p>a dataframe of numbers of shared alleles. Each row denotes each pair; Each column denotes each locus.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>df &lt;- data.frame(SNP1=c("A|A","T|T","A|T","A|T"),
                STR1=c("12|12","13|14","13|13","14|15"))
AlleleShare(df,"\\|",replacement=FALSE)

</code></pre>

<hr>
<h2 id='ComposPare_K'>Generate Comparison Observed and Expected No. of Heterozygous Loci.</h2><span id='topic+ComposPare_K'></span>

<h3>Description</h3>

<p>Generate Comparison Observed and Expected No. of Heterozygous Loci.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComposPare_K(h,Ex,trans)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ComposPare_K_+3A_h">h</code></td>
<td>
<p>a double made up of &quot;0&quot; and &quot;1&quot; where 1 means heterozygous and 0 means homozygous; Outcome of function &quot;Heterozygous&quot;; Each column denotes each locus and each row denotes each individual.</p>
</td></tr>
<tr><td><code id="ComposPare_K_+3A_ex">Ex</code></td>
<td>
<p>a dataframe of expected density, outcome of function &quot;DistHetero&quot;, on each possible total number of heterozygous loci.</p>
</td></tr>
<tr><td><code id="ComposPare_K_+3A_trans">trans</code></td>
<td>
<p>a logic variable, if True, the outcome is a dataframe of n x 2. n is the number of individuals of original imported database. First column is the observed No. of Heterozygous Loci and the second is the expected one. If False, the dataframe is 2n x 2, where n is the number of individuals of original imported database. The first column is a categorical variable denoting the frequency is observed or expected value; the second column is the frequency of No. of heterozygous loci.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a dataframe in which the observed and expected heterozygous loci for each sample are included. The observed ones are calculated from the original dataset. However, the expected ones are simulated according to the expected probability with the same sample size as observed sample.
</p>


<h3>Value</h3>

<p>a dataframe of observed and expected No. of heterozygous loci for each individual.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h&lt;-matrix(rbinom(20,1,0.5),nrow=5)
Ex &lt;- data.frame(K=c(0:5),Density=rnorm(6,mean = 0.5,sd=0.05))
ComposPare_K(h,Ex,trans = TRUE)
</code></pre>

<hr>
<h2 id='ComposPare_X'>Generate Comparison Observed and Expected No. of Shared Alleles.</h2><span id='topic+ComposPare_X'></span>

<h3>Description</h3>

<p>Generate Comparison Observed and Expected No. of Shared Alleles.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ComposPare_X(AS,Ex,trans=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ComposPare_X_+3A_as">AS</code></td>
<td>
<p>a double made up of &quot;0&quot;,&quot;1&quot; and &quot;2&quot; denoting number of shared alleles; Outcome of function &quot;AlleleShare_Table&quot;; Each column denotes each locus and each row denotes each pair of individuals.</p>
</td></tr>
<tr><td><code id="ComposPare_X_+3A_ex">Ex</code></td>
<td>
<p>a dataframe of expected density, outcome of function &quot;DistAlleleShare&quot;, on each possible total number of shared Alleles.</p>
</td></tr>
<tr><td><code id="ComposPare_X_+3A_trans">trans</code></td>
<td>
<p>a logic variable, if True, the outcome is a dataframe of n x 2. n is the number of individuals of original imported database. First column is the observed No. of Heterozygous Loci and the second is the expected one. If False, the dataframe is 2n x 2, where n is the number of individuals of original imported database. The first column is a categorical variable denoting the frequency is observed or expected value; the second column is the frequency of No. of heterozygous loci.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates a dataframe in which the observed and expected shared alleles for each pair of individuals. The observed ones are calculated from the original dataset through &quot;AlleleShare_Table&quot;. However, the expected ones are simulated according to the expected probability with the same sample size as the observed sample.
</p>


<h3>Value</h3>

<p>a dataframe of observed and expected No. of shared alleles for each pair of individuals.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AS&lt;-matrix(sample(c(0:2),20,replace=TRUE,prob=c(0.3,0.3,0.4)),nrow=5)
Ex &lt;- data.frame(X=c(0:8),Density=rnorm(9,mean = 0.5,sd=0.05))
ComposPare_X(AS,Ex,trans = TRUE)
</code></pre>

<hr>
<h2 id='counta'>Simple count including zero###</h2><span id='topic+counta'></span>

<h3>Description</h3>

<p>Simple count including zero###
</p>


<h3>Usage</h3>

<pre><code class='language-R'>counta(z, y)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="counta_+3A_z">z</code></td>
<td>
<p>a vector you would like to check</p>
</td></tr>
<tr><td><code id="counta_+3A_y">y</code></td>
<td>
<p>an element you would like to count.(Even it is not included in z)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function counts how many the assigned elements there are in one vector.
</p>


<h3>Value</h3>

<p>the times that y appears in z
</p>


<h3>Examples</h3>

<pre><code class='language-R'>z &lt;-rbinom(20,1,0.5)
counta(z,0)

</code></pre>

<hr>
<h2 id='Dist_SimuChisq'>Build a simulated distribution for Chi-Square</h2><span id='topic+Dist_SimuChisq'></span>

<h3>Description</h3>

<p>Build a simulated distribution for Chi-Square
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Dist_SimuChisq(s,prob,b)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Dist_SimuChisq_+3A_s">s</code></td>
<td>
<p>a matrix of frequencies for each simulated sample. Each row for each sample.</p>
</td></tr>
<tr><td><code id="Dist_SimuChisq_+3A_prob">prob</code></td>
<td>
<p>a vector of expected probability for each simulated sample.</p>
</td></tr>
<tr><td><code id="Dist_SimuChisq_+3A_b">b</code></td>
<td>
<p>the times of bootstrapping.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function build the distribution of Chi square statistics for simulated samples
</p>


<h3>Value</h3>

<p>a vector of Chi-square statistics, length is the times of sampling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(mixIndependR)
h&lt;-runif(10)
s&lt;-Simulate_DistK(h,500,100)
Exp &lt;- DistHetero(h)
Dist_SimuChisq(s,Exp$Density,10)
</code></pre>

<hr>
<h2 id='DistAlleleShare'>Build Expected Distribution of Numbers of Shared Alleles</h2><span id='topic+DistAlleleShare'></span>

<h3>Description</h3>

<p>Build Expected Distribution of Numbers of Shared Alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistAlleleShare(e)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DistAlleleShare_+3A_e">e</code></td>
<td>
<p>a matrix/dataframe of probability of shared alleles; outcome of &quot;ExpProAlleleShare&quot; or &quot;RealProAlleleShare&quot;. Each row denotes each locus. The first column is the case of 0 shared alleles, the second column is the case of 1 shared alleles, the third column is the case of 2 shared alleles.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function build the expected distribution of numbers of shared alleles for known shared alleles of each pair of individuals.
</p>


<h3>Value</h3>

<p>a dataframe of probabilities of each number of shared alleles(from 0 to 2*loci); the first column is No. of Shared Alleles; the Second Column is Expected Density
</p>


<h3>References</h3>

<p>Chakraborty, R., Stivers, D. N., Su, B., Zhong, Y., &amp; Budowle, B. (1999) &lt;doi:10.1002/(SICI)1522-2683(19990101)20:8&lt;1682::AID-ELPS1682&gt;3.0.CO;2-Z&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>e0&lt;-data.frame("P0"=runif(5,min = 0,max = 0.5),"P1"=runif(5,0,0.5))
e&lt;-data.frame(e0,"P2"=1-rowSums(e0))
DistAlleleShare(e)
</code></pre>

<hr>
<h2 id='DistHetero'>Build Expected Distribution of Numbers of Heterozygous Loci</h2><span id='topic+DistHetero'></span>

<h3>Description</h3>

<p>Build Expected Distribution of Numbers of Heterozygous Loci
</p>


<h3>Usage</h3>

<pre><code class='language-R'>DistHetero(H)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="DistHetero_+3A_h">H</code></td>
<td>
<p>a vector of average heterozygosity of each locus</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function build the expected distribution of numbers of heterozygous loci for known heterozygosity of each loci.
</p>


<h3>Value</h3>

<p>a dataframe of expected density on each possible total number of heterozygous loci.
</p>


<h3>References</h3>

<p>Chakraborty, R. (1981, ISSN:0016-6731)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>DistHetero(runif(10))

</code></pre>

<hr>
<h2 id='ExpProAlleleShare'>Calculate the Expected Probability of 0,1 and 2 Shared Alleles###</h2><span id='topic+ExpProAlleleShare'></span>

<h3>Description</h3>

<p>Calculate the Expected Probability of 0,1 and 2 Shared Alleles###
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ExpProAlleleShare(p)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="ExpProAlleleShare_+3A_p">p</code></td>
<td>
<p>a matrix/double of frequency of alleles; Outcome of &quot;AlleleFreq&quot;. Each column denotes each locus. Different alleles is ordered in different rows such as 11,11.3,12,12.2,13... and so on</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function Calculates the Expected Probability of 0,1 and 2 Shared Alleles for a set of loci. Usually followed by write.csv(as.data.frame(y),file = &quot;~/*.csv&quot;) to export the result of a n x3 matrix.
</p>


<h3>Value</h3>

<p>a matrix/double of expected probabilities of 0,1 and 2 shared alleles for each locus. Each row denotes each locus. The first column denotes the probability of 0 shared alleles, the second denotes 1 shared allele, the third denotes 2 shared alleles.
</p>


<h3>References</h3>

<p>Weir, B. S. (2004, ISSN:0022-1198)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>a0&lt;-matrix(runif(20),nrow=5)
a1&lt;-colSums(a0)
a&lt;-data.frame(STR1=a0[,1]/a1[1],STR2=a0[,2]/a1[2],STR3=a0[,3]/a1[3],STR4=a0[,4]/a1[4])
ExpProAlleleShare(a)

</code></pre>

<hr>
<h2 id='FreqAlleleShare'>Build Observed Distribution of No. of Shared Alleles</h2><span id='topic+FreqAlleleShare'></span>

<h3>Description</h3>

<p>Build Observed Distribution of No. of Shared Alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FreqAlleleShare(AS)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FreqAlleleShare_+3A_as">AS</code></td>
<td>
<p>a matrix of number of shared alleles, made up with 0, 1 and 2, outcome of function &quot;AlleleShare_Table&quot;. Rows for individuals, and columns for markers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function build the observed distributions from observed Allele Share table, made up of 0,1 and 2.
</p>


<h3>Value</h3>

<p>a dataframe of frequencies of each number of shared alleles(from 0 to 2*N0. of loci)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AS&lt;-matrix(sample(c(0:2),20,replace=TRUE,prob=c(0.3,0.3,0.4)),nrow=5)
FreqAlleleShare(AS)
</code></pre>

<hr>
<h2 id='FreqHetero'>Build Observed Distribution of No. of Heterozygous loci</h2><span id='topic+FreqHetero'></span>

<h3>Description</h3>

<p>Build Observed Distribution of No. of Heterozygous loci
</p>


<h3>Usage</h3>

<pre><code class='language-R'>FreqHetero(h)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="FreqHetero_+3A_h">h</code></td>
<td>
<p>a dataframe of heterozygosity, made up with 0 and 1, outcome of function &quot;Heterozygous&quot; Rows for individuals, and columns for markers.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function build the observed distributions from observed heterozygosity table, made up of 0,1.
</p>


<h3>Value</h3>

<p>a dataframe of frequencies of each number of heterozygous loci(from 0 to No. of loci)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>h&lt;-matrix(rbinom(20,1,0.5),nrow=5)
FreqHetero(h)
</code></pre>

<hr>
<h2 id='GenotypeFreq'>Calculate Genotype Frequency###</h2><span id='topic+GenotypeFreq'></span>

<h3>Description</h3>

<p>Calculate Genotype Frequency###
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GenotypeFreq(x,sep,expect=TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="GenotypeFreq_+3A_x">x</code></td>
<td>
<p>a dataframe of genotype data with rownames of sample ID and column names of markers.</p>
</td></tr>
<tr><td><code id="GenotypeFreq_+3A_sep">sep</code></td>
<td>
<p>allele separator in the imported genotype data. Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;(default).</p>
</td></tr>
<tr><td><code id="GenotypeFreq_+3A_expect">expect</code></td>
<td>
<p>a logic variable. If expect is true, the function will calculate the expected genotype probabilities. If false, calculate the observed genotype frequencies.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculates the observed or expected genotype frequency from dataset and allele frequency.#####
</p>


<h3>Value</h3>

<p>a dataframe of genotype frequencies. Each row denotes each genotype; each column denotes each loci. The order of markers follows x; the genotypes are ordered from homozygous to heterozygous.
</p>


<h3>References</h3>

<p>Chakraborty, R., Srinivasan, M. R., &amp; Daiger, S. P. (1993, ISSN:0002-9297).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(mixIndependR)
x &lt;- data.frame(SNP1=c("A|A","T|T","A|T","A|T"),
                STR1=c("12|12","13|14","13|13","14|15"))
GenotypeFreq(x,"\\|",expect=TRUE)
</code></pre>

<hr>
<h2 id='Heterozygous'>Test heterozygosity at each locus</h2><span id='topic+Heterozygous'></span>

<h3>Description</h3>

<p>Test heterozygosity at each locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Heterozygous(x,sep)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Heterozygous_+3A_x">x</code></td>
<td>
<p>a dataset of genotypes with rownames of sample ID and column names of markers.</p>
</td></tr>
<tr><td><code id="Heterozygous_+3A_sep">sep</code></td>
<td>
<p>allele separator in the imported genotype data. Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;(default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function test the heterozygosity of each individuals at each locus.Output a table and Usually followed by write.csv(as.data.frame(y),file = &quot;~/*.csv&quot;) to export the results.
</p>


<h3>Value</h3>

<p>a dataframe of heterozygosity.0 is homozygous;1 is heterozygous. Each row denotes each individual; Each column denotes each locus.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- data.frame(STR1=c("12|12","13|14","13|13","14|15"),
               SNP1=c("A|A","T|T","A|T","A|T"))
Heterozygous(x,"\\|")

</code></pre>

<hr>
<h2 id='HWE.Chisq'>Test the Hardy Weinberg Equilibrium with Chi-square test####</h2><span id='topic+HWE.Chisq'></span>

<h3>Description</h3>

<p>Test the Hardy Weinberg Equilibrium with Chi-square test####
</p>


<h3>Usage</h3>

<pre><code class='language-R'>HWE.Chisq(G,G0,rescale.p=FALSE,simulate.p.value=TRUE,B=2000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="HWE.Chisq_+3A_g">G</code></td>
<td>
<p>a dataframe of observed genotype frequencies. Each row denotes each genotype; each column denotes each marker. The order of markers follows x; the genotypes are ordered by: from 1:l-th column, the genotypes are homozygous in order as : p1p1, p2p2,p3p3,...,plpl;from ll-th to u-th column, the genotypes are heterozygous in order as:choose(l,2) like: p1p2,p1p3,...,p1pl,p2p3,p2p4,...p2pl,...p(l-1)pl</p>
</td></tr>
<tr><td><code id="HWE.Chisq_+3A_g0">G0</code></td>
<td>
<p>a dataframe of expected genotype probabilities;each row denotes each genotype; each column denotes each loci. The order of markers follows x; the genotypes are ordered by: from 1:l-th column, the genotypes are homozygous in order as : p1p1, p2p2,p3p3,...,plpl;from ll-th to u-th column, the genotypes are heterozygous in order as:choose(l,2) like: p1p2,p1p3,...,p1pl,p2p3,p2p4,...p2pl,...p(l-1)pl</p>
</td></tr>
<tr><td><code id="HWE.Chisq_+3A_rescale.p">rescale.p</code></td>
<td>
<p>a logical scalar; if TRUE then p is rescaled (if necessary) to sum to 1. If rescale.p is FALSE, and p does not sum to 1, an error is given.</p>
</td></tr>
<tr><td><code id="HWE.Chisq_+3A_simulate.p.value">simulate.p.value</code></td>
<td>
<p>a logical indicating whether to compute p-values by Monte Carlo simulation.</p>
</td></tr>
<tr><td><code id="HWE.Chisq_+3A_b">B</code></td>
<td>
<p>an integer specifying the number of replicates used in the Monte Carlo test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function check the Hardy Weinberg Equilibrium from observed and expected distribution with Chi-square test#####
</p>


<h3>Value</h3>

<p>a vector of result of p-values for chi-square test; the orders of markers follows x.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>require(mixIndependR)
x &lt;- data.frame(STR1=c("11|12","12|13","11|13","13|15"),
               STR2=c("12|12","13|14","13|13","14|15"),
               SNP1=c("A|T","A|A","T|A","A|T"),
               SNP2=c("A|A","T|T","A|T","T|A"))
G &lt;- GenotypeFreq(x,expect = FALSE)
G0 &lt;- GenotypeFreq(x,expect = TRUE)
HWE.Chisq(G,G0,rescale.p=FALSE,simulate.p.value=TRUE,B=2000)
</code></pre>

<hr>
<h2 id='mixexample'>Genotype Data from A Selected Mix Panel</h2><span id='topic+mixexample'></span>

<h3>Description</h3>

<p>This dataset is the phased genotypes for a mix panel with 100 variants.
These variants are selected from the reference haplotype data of Gymrek's lab (see Reference).
This is a sample with 2504 individuals.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>data(mixexample)
</code></pre>


<h3>Format</h3>

<p>A dataframe with 2504 observations on 100 variables. This dataframe is the phased genotype files for 100 variants (including SNPs and STRs) for 2504 individuals.
</p>


<h3>Source</h3>

<p>1000 Genomes SNP-STR Haplotype Panel &lt;https://gymreklab.com/2018/03/05/snpstr_imputation.html&gt;  The genotypes of panel after selection &lt;https://github.com/ice4prince/mixIndependR/tree/main/data&gt;
</p>


<h3>References</h3>

<p>Saini et al. (2018). A reference haplotype panel for genome-wide imputation of short tandem repeats. Nat Commun 9(1): 4397. &lt;https://pubmed.ncbi.nlm.nih.gov/30353011/&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(mixexample)
</code></pre>

<hr>
<h2 id='mixIndependK'>Quick pvalue of total number of heterozygous loci</h2><span id='topic+mixIndependK'></span>

<h3>Description</h3>

<p>Quick pvalue of total number of heterozygous loci
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixIndependK(x,sep,t,B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mixIndependK_+3A_x">x</code></td>
<td>
<p>a dataset of alleles. Each row denotes each individual.One allele in one cell.In the (2r-1)th column, there is the same locus with the 2r-th column; noted: no column for ID, make row.names=1 when importing.</p>
</td></tr>
<tr><td><code id="mixIndependK_+3A_sep">sep</code></td>
<td>
<p>allele separator in the imported genotype data. Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;.</p>
</td></tr>
<tr><td><code id="mixIndependK_+3A_t">t</code></td>
<td>
<p>times of simulation in &quot;Simulate_DistK&quot; and &quot;Simulate_DistX&quot;.</p>
</td></tr>
<tr><td><code id="mixIndependK_+3A_b">B</code></td>
<td>
<p>times of bootstrapping in Chi Squares Test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a summary of pipeline for number of heterozygous loci (K), and generates the p-value of K for the target dataset.
</p>


<h3>Value</h3>

<p>pvalue (1-cumulative probabilities) for the number of heterozygous loci(K)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- data.frame(SNP1=c("A|A","T|T","A|T","A|T"),
                STR1=c("12|12","13|14","13|13","14|15"))
mixIndependK(x,sep ="\\|",10,10)
</code></pre>

<hr>
<h2 id='mixIndependX'>Quick pvalue of total number of shared alleles</h2><span id='topic+mixIndependX'></span>

<h3>Description</h3>

<p>Quick pvalue of total number of shared alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mixIndependX(x,sep,t,B)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mixIndependX_+3A_x">x</code></td>
<td>
<p>a dataset of alleles. Each row denotes each individual.One allele in one cell.In the (2r-1)th column, there is the same locus with the 2r-th column; noted: no column for ID, make row.names=1 when importing.</p>
</td></tr>
<tr><td><code id="mixIndependX_+3A_sep">sep</code></td>
<td>
<p>allele separator in the imported genotype data. Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;.</p>
</td></tr>
<tr><td><code id="mixIndependX_+3A_t">t</code></td>
<td>
<p>times of simulation in &quot;Simulate_DistK&quot; and &quot;Simulate_DistX&quot;.</p>
</td></tr>
<tr><td><code id="mixIndependX_+3A_b">B</code></td>
<td>
<p>times of bootstrapping in Chi Squares Test.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function is a summary of pipeline for number of shared alleles(X), and generates the p-value of K for the target dataset.
</p>


<h3>Value</h3>

<p>pvalue (1-cumulative probabilities) for the number of shared alleles(K)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- data.frame(SNP1=c("A|A","T|T","A|T","A|T"),
                STR1=c("12|12","13|14","13|13","14|15"))
mixIndependX(x,sep="\\|",10,10)
</code></pre>

<hr>
<h2 id='read_vcf_gt'>Import genotype data from vcf files/</h2><span id='topic+read_vcf_gt'></span>

<h3>Description</h3>

<p>Import genotype data from vcf files/
</p>


<h3>Usage</h3>

<pre><code class='language-R'>read_vcf_gt(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="read_vcf_gt_+3A_x">x</code></td>
<td>
<p>The vcf file with its directory</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function extract the genotypes and allele status from a vcf file.
</p>


<h3>Value</h3>

<p>a list contains the genotype and allele status.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
df&lt;-read_vcf_gt("~/x.vcf")

## End(Not run)
</code></pre>

<hr>
<h2 id='RealProAlleleShare'>Calculate the Real Probability of 0,1 and 2 Shared Alleles###</h2><span id='topic+RealProAlleleShare'></span>

<h3>Description</h3>

<p>Calculate the Real Probability of 0,1 and 2 Shared Alleles###
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RealProAlleleShare(AS)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RealProAlleleShare_+3A_as">AS</code></td>
<td>
<p>a matrix/double of no. of Shared alleles, made up with 0,1 and 2; Outcome of &quot;AlleleShare_Table&quot;. Each column denotes each locus. Each row denotes each individual.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function Calculates the density of 0,1 and 2 Shared Alleles for a set of loci. Usually followed by write.csv(as.data.frame(y),file = &quot;~/*.csv&quot;) to export the result of a n x3 matrix.
</p>


<h3>Value</h3>

<p>a matrix/double of real density of 0,1 and 2 shared alleles for each locus. Each row denotes each locus. The first column denotes the probability of 0 shared alleles, the second denotes 1 shared allele, the third denotes 2 shared alleles.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>AS&lt;-matrix(sample(c(0:2),20,replace=TRUE,prob=c(0.3,0.3,0.4)),nrow=5)
RealProAlleleShare(AS)
</code></pre>

<hr>
<h2 id='RxpHetero'>Calculate Real or Expected Average Heterozygosity at each locus</h2><span id='topic+RxpHetero'></span>

<h3>Description</h3>

<p>Calculate Real or Expected Average Heterozygosity at each locus
</p>


<h3>Usage</h3>

<pre><code class='language-R'>RxpHetero(h,p,HWE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="RxpHetero_+3A_h">h</code></td>
<td>
<p>a dataset of heterozygosity, made up with 0 and 1. Output of function &quot;Heterozygous&quot;. Each row denotes each individual. Each row denotes each locus.</p>
</td></tr>
<tr><td><code id="RxpHetero_+3A_p">p</code></td>
<td>
<p>a dataset of allele frequency, Output of function &quot;AlleleFreq&quot;. Each row denotes each allele, and each column denotes each locus.</p>
</td></tr>
<tr><td><code id="RxpHetero_+3A_hwe">HWE</code></td>
<td>
<p>a logic variable. When TRUE, this function will calculate the expected heterozygosity under Hardy-Weinberg Equilibrium: H= 1-sum(q_i^2); q_i is the allele frequency; If FALSE, this function calculate the average heterozygosity from real heterozygosity table.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function calculate average heterozygosity at each locus.Output a vector of number of loci.
</p>


<h3>Value</h3>

<p>a vector of average heterozygosity on each loci.
</p>


<h3>References</h3>

<p>Chakraborty, R., &amp; Jin, L. (1992, ISSN:1432-1203) &lt;doi:10.1007/BF00197257&gt;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>x &lt;- data.frame(STR1=c(12,13,13,14,15,13,14,12,14,15),
               STR1_1=c(12,14,13,15,13,14,13,12,14,15),
               SNP1=c("A","T","A","A","T","A","A","T","T","A"),
               SNP1_1=c("A","T","T","T","A","T","A","A","T","T"))
require(mixIndependR)
h &lt;- Heterozygous(x)
p &lt;- AlleleFreq(x)
RxpHetero(h,p,HWE=TRUE)
</code></pre>

<hr>
<h2 id='Simulate_DistK'>Generate a Bundle of Simulated distributions for No. of heterozygous loci with known heterozygosites</h2><span id='topic+Simulate_DistK'></span>

<h3>Description</h3>

<p>Generate a Bundle of Simulated distributions for No. of heterozygous loci with known heterozygosites
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simulate_DistK(H,m,t)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Simulate_DistK_+3A_h">H</code></td>
<td>
<p>a vector of average heterozygosity of each loci. Length of H is the number of loci.</p>
</td></tr>
<tr><td><code id="Simulate_DistK_+3A_m">m</code></td>
<td>
<p>the sample size you want, usually similar to the real sample size.</p>
</td></tr>
<tr><td><code id="Simulate_DistK_+3A_t">t</code></td>
<td>
<p>the number of samples you want to build</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates multinomial distribution for loci known the heterozygosity and build the simulated distribution for no. of heterozygous loci.
</p>


<h3>Value</h3>

<p>a matrix of frequencies of No. of Heterozygous Loci. Each row denotes each simulated sample; Each column denotes each No. of Heterozygous loci, from 0 to length of H.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>Simulate_DistK(runif(10),500,100)

</code></pre>

<hr>
<h2 id='Simulate_DistX'>Build a simulated distribution for No. of Shared Alleles</h2><span id='topic+Simulate_DistX'></span>

<h3>Description</h3>

<p>Build a simulated distribution for No. of Shared Alleles
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Simulate_DistX(e,m,t)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Simulate_DistX_+3A_e">e</code></td>
<td>
<p>a matrix of Probability of Sharing 2,1 or 0 alleles at each loci. Each row denotes each locus. Three columns denote sharing 0,1 or 2 alleles.</p>
</td></tr>
<tr><td><code id="Simulate_DistX_+3A_m">m</code></td>
<td>
<p>the sample size you want, usually similar to the real sample size.</p>
</td></tr>
<tr><td><code id="Simulate_DistX_+3A_t">t</code></td>
<td>
<p>the number of samples you want to build/ the times to generate a sample</p>
</td></tr>
</table>


<h3>Details</h3>

<p>This function generates multinomial distribution for loci known the Allele Frequency and Expected Probability of Shared 2,1 or 0 alleles
</p>


<h3>Value</h3>

<p>a matrix of frequencies of No. of shared alleles. Each row denotes each simulated sample; Each column denotes each No. of shared alleles, from 0 to 2e length of e.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>e0&lt;-data.frame("P0"=runif(5,min = 0,max = 0.5),"P1"=runif(5,0,0.5))
e&lt;-data.frame(e0,"P2"=1-rowSums(e0))
Simulate_DistX(e,500,10)

</code></pre>

<hr>
<h2 id='splitGenotype'>Split Genotype Table to Duo-Allele Table</h2><span id='topic+splitGenotype'></span>

<h3>Description</h3>

<p>Split each column to two columns for a table of genotypes
</p>


<h3>Usage</h3>

<pre><code class='language-R'>splitGenotype(df,sep,dif,rowbind)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="splitGenotype_+3A_df">df</code></td>
<td>
<p>a dataframe of genotype data with rownames of sample ID and column names of markers.</p>
</td></tr>
<tr><td><code id="splitGenotype_+3A_sep">sep</code></td>
<td>
<p>allele separator in the imported genotype data. Note: when using the special character like &quot;|&quot;, remember to protect it as &quot;\|&quot;(default).</p>
</td></tr>
<tr><td><code id="splitGenotype_+3A_dif">dif</code></td>
<td>
<p>a symbol differentiate the one marker on each allele.</p>
</td></tr>
<tr><td><code id="splitGenotype_+3A_rowbind">rowbind</code></td>
<td>
<p>a logical variable. If rowbind is TRUE, the output is arranged with double rows but the same columns, and the table of the second allele is followed after the first allele table by rows with double individual IDs in the same order.<br /> If rowbind is false,the output is arranged by double columns and the same rows; the column names are in the order of alphabet by pairs.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The function convert a genotype data to allele data with double columns or with double rows; the rownames are sample ID in the same order but twice if the rows are doubled, and the column names are in the same order or in the order of alphabet by pairs if columns are doubled. <br /> The parameter &quot;sep&quot; is the symbol of allele separator in the imported genotype data. <br /> The parameter &quot;dif&quot; is the difference between the second and the first appearance for the same marker. For example, if &quot;dif = _1&quot;, the column names of output will be &quot;marker1&quot; &quot;marker1 _1&quot;,&quot;marker2&quot;,&quot;marker2 _1&quot;, if the original list of column names is &quot;marker1&quot;,&quot;marker2&quot;.
</p>


<h3>Value</h3>

<p>a dataframe with doubled columns of import data and alleles in different columns
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
df &lt;- data.frame(SNP1=c("A|A","T|T","A|T","A|T"),
                STR1=c("12|12","13|14","13|13","14|15"))
splitGenotype(df)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
