<!DOCTYPE html><html><head><title>Help for package MixRF</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MixRF}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MixRF'><p>Mixed Random Forest</p></a></li>
<li><a href='#MixRF-package'>
<p>A random-forest-based algorithm for imputing clustered incomplete data</p></a></li>
<li><a href='#MixRF.impute'><p>Impute a large number of genes using the MixRF algorithm with parallel computing</p></a></li>
<li><a href='#sim'>
<p>Simulated data list</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>A Random-Forest-Based Approach for Imputing Clustered Incomplete
Data</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0</td>
</tr>
<tr>
<td>Date:</td>
<td>2016-04-05</td>
</tr>
<tr>
<td>Author:</td>
<td>Jiebiao Wang and Lin S. Chen</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Jiebiao Wang &lt;randel.wang@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>It offers random-forest-based functions to impute clustered
    incomplete data. The package is tailored for but not limited to imputing
    multitissue expression data, in which a gene's expression is measured on the
    collected tissues of an individual but missing on the uncollected tissues.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL]</td>
</tr>
<tr>
<td>Depends:</td>
<td>doParallel, randomForest, lme4, foreach</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/randel/MixRF">https://github.com/randel/MixRF</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/randel/MixRF/issues">https://github.com/randel/MixRF/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>5.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2016-04-06 02:05:06 UTC; JWang</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2016-04-06 09:43:04</td>
</tr>
</table>
<hr>
<h2 id='MixRF'>Mixed Random Forest</h2><span id='topic+MixRF'></span>

<h3>Description</h3>

<p>The function to fit a random forest with random effects.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MixRF(Y, X, random, data, initialRandomEffects = 0, ErrorTolerance = 0.001, 
    MaxIterations = 1000)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MixRF_+3A_y">Y</code></td>
<td>
<p>The outcome variable.</p>
</td></tr>
<tr><td><code id="MixRF_+3A_x">X</code></td>
<td>
<p>A data frame or matrix contains the predictors.</p>
</td></tr>
<tr><td><code id="MixRF_+3A_random">random</code></td>
<td>
<p>A string in lme4 format indicates the random effect model.</p>
</td></tr>
<tr><td><code id="MixRF_+3A_data">data</code></td>
<td>
<p>The data set as a data frame.</p>
</td></tr>
<tr><td><code id="MixRF_+3A_initialrandomeffects">initialRandomEffects</code></td>
<td>
<p>The initial values for random effects.</p>
</td></tr>
<tr><td><code id="MixRF_+3A_errortolerance">ErrorTolerance</code></td>
<td>
<p>The tolerance for log-likelihood.</p>
</td></tr>
<tr><td><code id="MixRF_+3A_maxiterations">MaxIterations</code></td>
<td>
<p>The maximum iteration times.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list contains the random forest ($forest), mixed model ($MixedModel), and random effects ($RandomEffects).
See the example below for the usage.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(sleepstudy)

tmp = MixRF(Y = sleepstudy$Reaction, X = as.data.frame(sleepstudy$Days), 
    random = "(Days|Subject)", data = sleepstudy, initialRandomEffects = 0, 
    ErrorTolerance = 0.01, MaxIterations = 100)

# tmp$forest

# tmp$MixedModel

# tmp$RandomEffects
</code></pre>

<hr>
<h2 id='MixRF-package'>
A random-forest-based algorithm for imputing clustered incomplete data
</h2><span id='topic+MixRF-package'></span>

<h3>Description</h3>

<p>This package offers random-forest-based functions to impute clustered incomplete data.
The package is tailored for but not limited to imputing multitissue expression data,
in which a gene's expression is measured on the collected tissues of an individual
but missing on the uncollected tissues.
</p>


<h3>Details</h3>


<table>
<tr>
 <td style="text-align: left;">
Package: </td><td style="text-align: left;"> MixRF </td>
</tr>
<tr>
 <td style="text-align: left;">
Type: </td><td style="text-align: left;"> Package</td>
</tr>
<tr>
 <td style="text-align: left;">
Version: </td><td style="text-align: left;"> 1.0</td>
</tr>
<tr>
 <td style="text-align: left;">
Date: </td><td style="text-align: left;"> 2016-04-05</td>
</tr>
<tr>
 <td style="text-align: left;">
License: </td><td style="text-align: left;"> GPL </td>
</tr>
<tr>
 <td style="text-align: left;">
LazyLoad: </td><td style="text-align: left;"> yes</td>
</tr>
<tr>
 <td style="text-align: left;">
</td>
</tr>

</table>



<h3>Author(s)</h3>

<p>Jiebiao Wang and Lin S. Chen
</p>
<p>Maintainer: Jiebiao Wang &lt;randel.wang@gmail.com&gt;
</p>


<h3>References</h3>

<p>Wang, J., Gamazon, E.R., Pierce, B.L., Stranger, B.E., Im, H.K., Gibbons, R.D., Cox, N.J., Nicolae, D.L. and Chen, L.S.
(2016) Imputing gene expression in uncollected tissues within and beyond GTEx. http://dx.doi.org/10.1016/j.ajhg.2016.02.020
</p>


<h3>See Also</h3>

<p><code><a href="#topic+MixRF.impute">MixRF.impute</a></code>
</p>

<hr>
<h2 id='MixRF.impute'>Impute a large number of genes using the MixRF algorithm with parallel computing</h2><span id='topic+MixRF.impute'></span>

<h3>Description</h3>

<p>This function impute the expression of a large number of genes using the MixRF algorithm with parallel computing.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MixRF.impute(Ydat, eqtl.lis, snp.dat, cov = NULL, iPC = TRUE, 
    idx.selected.gene.iPC = NULL, parallel.size = 1, correlation = FALSE, 
    nCV = 3)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MixRF.impute_+3A_ydat">Ydat</code></td>
<td>
<p>An array of expression data of dimension sample-by-gene-by-tissue, nxpxT, where n is sample size.
p is the number of genes, and T is the number of tissues.  Ydat[,1,] is a matrix of the first gene expression
in T tissues for n individuals, nxT. Ydat[,,1] is a nxp matrix of the expression data of p genes in the first tissue.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_eqtl.lis">eqtl.lis</code></td>
<td>
<p>A list of eQTL names of length p. Each element in the list contains the name of the eQTLs for
the corresponding gene. The order of the list should correspond to the order of genes in Ydat. The code and example to calculate eQTLs can be found at https://github.com/randel/MixRF/blob/master/R/eqtl.r.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_snp.dat">snp.dat</code></td>
<td>
<p>A matrix of genotype. Each row is a sample and each column corresponds to one SNP. The column names should match eqtl.lis.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_cov">cov</code></td>
<td>
<p>A matrix of covariates. Each row is a sample and each column corresponds to one covariate. For example, age, gender.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_ipc">iPC</code></td>
<td>
<p>An option. When it is TRUE, the imputed PCs (iPCs) for each tissue type will be constructed
based on the combined observed and imputed data on the selected genes. The iPCs will be adjusted as covariates
in the imputation.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_idx.selected.gene.ipc">idx.selected.gene.iPC</code></td>
<td>
<p>The option is used only when iPC=TRUE. When it is, one may select a subset of genes and impute
those first to construct iPCs.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_parallel.size">parallel.size</code></td>
<td>
<p>A numerical value specifying the number of CPUs/cores/processors available for parallel computing.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_correlation">correlation</code></td>
<td>
<p>The option to calculate the imputation correlation using cross-validation or not. The default is FALSE.</p>
</td></tr>
<tr><td><code id="MixRF.impute_+3A_ncv">nCV</code></td>
<td>
<p>The option is used only when correlation=TRUE. The number of folds for cross-validation. The default is 3 folds.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An nxpxT array of imputed and observed expression data. The observed values in Ydat are still kept and the missing values in Ydat are imputed.
When the user chooses to calculate the imputation correlation using cross-validation (correlation=TRUE), the estimated imputation correlation (cor)
will also be returned in a list together with the imputed data (Yimp).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run: 
data(sim)

idx.selected.gene.iPC = which(sapply(sim$eqtl.lis, length) &gt;= 1)

Yimp = MixRF.impute(sim$Ydat, sim$eqtl.lis, sim$snp.dat, sim$cov, iPC = TRUE, 
    idx.selected.gene.iPC, parallel.size = 4)

## End(Not run)

</code></pre>

<hr>
<h2 id='sim'>
Simulated data list
</h2><span id='topic+sim'></span>

<h3>Description</h3>

<p>This simulated data list is for demonstration. 
</p>


<h3>Value</h3>

<table>
<tr><td><code>Ydat</code></td>
<td>
<p>An array of expression data of dimension sample-by-gene-by-tissue, nxpxT, where n is sample size.
p is the number of genes, and T is the number of tissues.  Ydat[,1,] is a matrix of the first gene expression
in T tissues for n individuals, nxT. Ydat[,,1] is a nxp matrix of the expression data of p genes in the first tissue.</p>
</td></tr>
<tr><td><code>eqtl.lis</code></td>
<td>
<p>A list of eQTL names of length p. Each of the element in the list contains the name of the eQTLs for
the corresponding gene. The order of the list should correspond to the order of genes in Ydat.</p>
</td></tr>
<tr><td><code>snp.dat</code></td>
<td>
<p>A matrix of genotype. Each row is a sample and each column corresponds to one SNP. The column names should match eqtl.lis.</p>
</td></tr>
<tr><td><code>cov</code></td>
<td>
<p>A matrix of covariates. Each row is a sample and each column corresponds to one covariate. For example, age, gender.</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+MixRF.impute">MixRF.impute</a></code>
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
