<!DOCTYPE html><html><head><title>Help for package MixviR</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MixviR}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#add_depths_to_ref'><p>Add Read Depths To Ref</p></a></li>
<li><a href='#call_mutations'><p>Identify Variants From A Potentially Mixed Sample</p></a></li>
<li><a href='#create_ref'><p>Create MixVir-formatted reference genome object</p></a></li>
<li><a href='#estimate_lineages'><p>Estimate Lineage Proportions In Samples</p></a></li>
<li><a href='#explore_mutations'><p>MixviR Shiny Dashboard</p></a></li>
<li><a href='#get_codons'><p>Get Codons From Gene Sequence</p></a></li>
<li><a href='#id_indels'><p>ID Indel-based Amino Acid Changes</p></a></li>
<li><a href='#id_snps'><p>ID SNV-based Amino Acid Changes</p></a></li>
<li><a href='#vcf_to_mixvir'><p>Convert Sample VCF to MixviR Input Format</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Analysis and Exploration of Mixed Microbial Genomic Samples</td>
</tr>
<tr>
<td>Version:</td>
<td>3.5.0</td>
</tr>
<tr>
<td>Description:</td>
<td>Tool for exploring DNA and amino acid variation and inferring the presence of target lineages from microbial high-throughput genomic DNA samples that potentially contain mixtures of variants/lineages. MixviR was originally created to help analyze environmental SARS-CoV-2/Covid-19 samples from environmental sources such as wastewater or dust, but can be applied to any microbial group. Inputs include reference genome information in commonly-used file formats (fasta, bed) and one or more variant call format (VCF) files, which can be generated with programs such as Illumina's DRAGEN, the Genome Analysis Toolkit, or bcftools. See DePristo et al (2011) &lt;<a href="https://doi.org/10.1038%2Fng.806">doi:10.1038/ng.806</a>&gt; and Danecek et al (2021) &lt;<a href="https://doi.org/10.1093%2Fgigascience%2Fgiab008">doi:10.1093/gigascience/giab008</a>&gt; for these tools, respectively. Available outputs include a table of mutations observed in the sample(s), estimates of proportions of target lineages in the sample(s), and an R Shiny dashboard to interactively explore the data. </td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/mikesovic/MixviR">https://github.com/mikesovic/MixviR</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://groups.google.com/g/mixvir">https://groups.google.com/g/mixvir</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.1</td>
</tr>
<tr>
<td>Imports:</td>
<td>Biostrings, dplyr, DT, ggplot2 (&ge; 3.1.0), glue, httr,
lubridate, magrittr, plotly (&ge; 4.9.4), readr (&ge; 2.0.0),
shiny, stats (&ge; 1.4.0), stringr (&ge; 1.1.0), tidyr (&ge; 0.8.0),
utils, vcfR (&ge; 1.11.0)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-10-22 02:38:40 UTC; sovic.1</td>
</tr>
<tr>
<td>Author:</td>
<td>Michael Sovic <a href="https://orcid.org/0000-0002-8556-3704"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ccp, cre],
  Francesca Savona [res],
  Zuzana Bohrerova [res, fnd],
  Seth Faith [ccp, ctb] (0000-0002-0441-9859)</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Michael Sovic &lt;sovic.1@osu.edu&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-10-22 22:17:49 UTC</td>
</tr>
</table>
<hr>
<h2 id='add_depths_to_ref'>Add Read Depths To Ref</h2><span id='topic+add_depths_to_ref'></span>

<h3>Description</h3>

<p>Add column of total read depths for a given sample to reference df. Run as part of id_mutations().
</p>


<h3>Usage</h3>

<pre><code class='language-R'>add_depths_to_ref(ref, samp.variants)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="add_depths_to_ref_+3A_ref">ref</code></td>
<td>
<p>reference genome in &quot;MixVir&quot; format (from create_ref() function)</p>
</td></tr>
<tr><td><code id="add_depths_to_ref_+3A_samp.variants">samp.variants</code></td>
<td>
<p>Data frame produced by function vcf_to_mixvir(). Contains columns &quot;CHR&quot;, &quot;POS&quot;, &quot;REF&quot;, &quot;ALT&quot;, &quot;DP&quot;, &quot;ALT_COUNT&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Original 'ref' data frame with depth (DP) column added: cols &quot;genomic_pos&quot;	&quot;ref_base&quot;	&quot;gene&quot;	&quot;ref_codon&quot;	&quot;ref_AA&quot;	&quot;GENE_AA_POS&quot;	&quot;ref_identity&quot; &quot;DP&quot;
</p>

<hr>
<h2 id='call_mutations'>Identify Variants From A Potentially Mixed Sample</h2><span id='topic+call_mutations'></span>

<h3>Description</h3>

<p>Identify full set of amino acid/SNP/indel changes from one or more samples (includes changes based on both SNVs and indels). This is generally the first function run in a MixviR analysis.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>call_mutations(
  sample.dir = NULL,
  fasta.genome,
  bed,
  reference = "custom",
  min.alt.freq = 0.01,
  name.sep = NULL,
  write.all.targets = FALSE,
  lineage.muts = NULL,
  genetic.code.num = "1",
  out.cols = c("SAMP_NAME", "CHR", "POS", "GENE", "ALT_ID", "AF", "DP"),
  write.mut.table = FALSE,
  outfile.name = "sample_mutations.csv",
  indel.format = "Fwd",
  csv.infiles = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="call_mutations_+3A_sample.dir">sample.dir</code></td>
<td>
<p><strong>Required</strong> Path to directory containing vcf files for each sample to be analyzed. VCF's need to contain &quot;DP&quot; and &quot;AD&quot; flags in the FORMAT field. This directory should not contain any other files.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_fasta.genome">fasta.genome</code></td>
<td>
<p>Path to fasta formatted reference genome file. <strong>Required unless <em>reference</em> is defined</strong>.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_bed">bed</code></td>
<td>
<p>Path to bed file defining features of interest (open reading frames to translate). Should be tab delimited and have 6 columns (no column names): chr, start, end, feature_name, score (not used), strand. <strong>Required unless <em>reference</em> is defined</strong>. See example at https://github.com/mikesovic/MixviR/blob/main/raw_files/sars_cov2_genes.bed.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_reference">reference</code></td>
<td>
<p>Optional character defining a pre-constructed MixviR reference (created with 'create_ref()&ldquo;). &quot;Wuhan&quot; uses pre-generated Sars-Cov2 ref genome. Otherwise, <em>fasta.genome</em> and <em>bed</em> are required to generate MixviR formatted reference as part of the analysis.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_min.alt.freq">min.alt.freq</code></td>
<td>
<p>Minimum frequency (0-1) for retaining alternate alleles. Default = 0.01. Extremely low values (i.e. zero) are not recommended here - see vignette for details.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_name.sep">name.sep</code></td>
<td>
<p>Optional character in input file names that separates the unique sample identifier (characters preceeding the separator) from any additional text. Only text preceeding the first instance of the character will be retained and used as the sample name.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_write.all.targets">write.all.targets</code></td>
<td>
<p>Logical that, if TRUE, reports sequencing depths for genomic positions associated with mutations of interest that are not observed in the sample, in addition to all mutations observed in the sample. If TRUE, requires columns &quot;Chr&quot; and &quot;Pos&quot; to be included in the <em>lineage.muts</em> file. Default FALSE.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_lineage.muts">lineage.muts</code></td>
<td>
<p>Path to optional csv file defining target mutations and their underlying genomic positions. Requires cols &quot;Gene&quot;, &quot;Mutation&quot;, &quot;Lineage&quot;, &quot;Chr&quot; and &quot;Pos&quot;. This is used to report the sequencing depths for relevant positions when the mutation of interest is not observed in the sample. See <em>write.all.targets</em>. This file is also used in <code>explore_mutations()</code>, where the &quot;Chr&quot; and &quot;Pos&quot; columns are optional. Only necessary here in conjunction with <em>write.all.targets</em>.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_genetic.code.num">genetic.code.num</code></td>
<td>
<p>Number (character) associated with the genetic code to be used for translation. Details can be found at https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_out.cols">out.cols</code></td>
<td>
<p>Character vector with names of columns to be returned. Choose from: CHR, POS, REF_BASE, GENE, STRAND, REF_CODON, REF_AA, GENE_AA_POS, REF_IDENT, REF, ALT, AF, ALT_COUNT, SAMP_CODON, SAMP_AA, ALT_ID, DP, SAMP_NAME, TYPE. The default columns SAMP_NAME, CHR, POS, ALT_ID, AF, DP must be included to run explore_mutations().</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_write.mut.table">write.mut.table</code></td>
<td>
<p>Logical indicating whether to write the 'samp_mutations' data frame (see &quot;Value&quot; below) to a text file (csv). Default = FALSE. See <em>outfile.name</em>.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_outfile.name">outfile.name</code></td>
<td>
<p>Path to file where (csv) output will be written if <em>write.mut.table</em> is TRUE. Default: &quot;sample_mutations.csv&quot; (written to working directory)</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_indel.format">indel.format</code></td>
<td>
<p>Defines the naming convention for indels. Default is &quot;Fwd&quot;, meaning the name would look like S_del144/144. &quot;Rev&quot; switches this to S_144/144del.</p>
</td></tr>
<tr><td><code id="call_mutations_+3A_csv.infiles">csv.infiles</code></td>
<td>
<p>Logical to indicate whether files in <em>sample.dir</em> directory are in vcf or csv format. All files must be of the same format. If csv, they must contain columns named: &quot;CHR&quot;    &quot;POS&quot;    &quot;REF&quot;    &quot;ALT&quot;    &quot;DP&quot;    &quot;ALT_COUNT&quot;. See the 'batch_vcf_to_mixvir()&ldquo; function to convert vcfs to csv format). Default is FALSE (input is in vcf format). This exists primarily for legacy reasons.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame containing variants observed for each sample, positions of the underlying mutations, and other (customizable) information. This data frame can be saved as an object in the global environment and/or written to a file (see <em>write.mut.table</em>), and in either case serves as input for the <em>MixviR</em> functions <code>explore_mutations()</code> and <code>estimate_lineages()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
 ##For SARS-CoV-2 
 #call_mutations(sample.dir = system.file("extdata", "vcf", package = "MixviR"), 
 #                name.sep = "_", reference = "Wuhan") 
   
 ##OR if defining a custom reference, follow this pattern...  
 #genome&lt;-"https://raw.githubusercontent.com/mikesovic/MixviR/main/raw_files/GCF_ASM985889v3.fa"
 #features&lt;-"https://raw.githubusercontent.com/mikesovic/MixviR/main/raw_files/sars_cov2_genes.bed"
   
 #call_mutations(sample.dir = system.file("extdata", "vcf", package = "MixviR"), 
 #                name.sep = "_", 
 #                fasta.genome = genome,
 #                bed = features)
</code></pre>

<hr>
<h2 id='create_ref'>Create MixVir-formatted reference genome object</h2><span id='topic+create_ref'></span>

<h3>Description</h3>

<p>Uses a fasta genome and bed file defining features of interest (genes/ORFs) to create a data frame that's used as a reference to translate nucleotide data to amino acids and subsequently call variants/mutations from a sample.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>create_ref(genome, feature.bed, code.num = "1", removed.genes = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="create_ref_+3A_genome">genome</code></td>
<td>
<p><em>(Required)</em> Path to fasta formatted genome file</p>
</td></tr>
<tr><td><code id="create_ref_+3A_feature.bed">feature.bed</code></td>
<td>
<p><em>(Required)</em> Path to bed file defining features of interest (open reading frames to translate). Tab delimited with 6 columns (without column names):&quot;chr&quot;, &quot;start&quot;, &quot;end&quot;, &quot;feature_name&quot;, &quot;score&quot; (not used), and &quot;strand&quot;.</p>
</td></tr>
<tr><td><code id="create_ref_+3A_code.num">code.num</code></td>
<td>
<p>Number (character) associated with the genetic code to be used for translation. Details can be found at https://www.ncbi.nlm.nih.gov/Taxonomy/Utils/wprintgc.cgi.</p>
</td></tr>
<tr><td><code id="create_ref_+3A_removed.genes">removed.genes</code></td>
<td>
<p>Character providing path/name of tab-separated file that will be written that stores names of genes (if any) in the feature.bed file that were removed because they didn't have an allowed size (not even multiples of 3). If NULL (default), file is not written.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame with columns CHR,POS,REF_BASE,GENE,STRAND,REF_CODON,REF_AA,GENE_AA_POS,REF_IDENT,GENE_BASE_NUM,CODON_POSITION
</p>


<h3>Examples</h3>

<pre><code class='language-R'>site1 &lt;- "https://raw.githubusercontent.com/mikesovic/MixviR/main/raw_files/GCF_ASM985889v3.fa"
site2 &lt;- "https://raw.githubusercontent.com/mikesovic/MixviR/main/raw_files/sars_cov2_genes.bed"

if (httr::http_error(site1) | httr::http_error(site2)) {
     message("No internet connection or data source broken.") 
     return(NULL)
} else { 
create_ref(
 genome = site1,
 feature.bed = site2,
 code.num = "1")
}
</code></pre>

<hr>
<h2 id='estimate_lineages'>Estimate Lineage Proportions In Samples</h2><span id='topic+estimate_lineages'></span>

<h3>Description</h3>

<p>Create summary tables containing data on lineages identified in samples, including estimates of relative proportions of lineages and identities of associated characteristic mutations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>estimate_lineages(
  muts.df,
  min.alt.freq = 0.01,
  dates = NULL,
  lineage.muts = NULL,
  read.muts.from = NULL,
  scale = TRUE,
  use.median = FALSE,
  outfile.name = NULL,
  presence.thresh = 0.5,
  samps.to.inc = NULL,
  locs.to.inc = NULL,
  lineages.to.inc = NULL,
  report.all = FALSE,
  depths.from = "all"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="estimate_lineages_+3A_muts.df">muts.df</code></td>
<td>
<p>A data frame (produced by <code>call_mutations()</code>) storing mutation information for samples to analyze. Must contain columns SAMP_NAME, CHR, POS, GENE, ALT_ID, AF, &amp; DP. Alternatively, the mutation data can be read in from a (comma-separated) file with the <code>read.muts.from()</code> argument. See the <em>write.mut.table</em> argument in <code>call_mutations()</code>.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_min.alt.freq">min.alt.freq</code></td>
<td>
<p>Minimum frequency (0-1) for mutation to be counted. Default = 0.01.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_dates">dates</code></td>
<td>
<p>Path to optional csv file with cols &quot;SAMP_NAME&quot;, &quot;LOCATION&quot;, and &quot;DATE&quot;. Sample names need to match those in <em>samp_mutations</em> data frame created by <code>call_mutations()</code>. Dates should be provided in the format <em>mmddyyyy</em>.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_lineage.muts">lineage.muts</code></td>
<td>
<p><em>(Required)</em> Path to csv file with cols &quot;Gene&quot;, &quot;Mutation&quot;, and  &quot;Lineage&quot; defining mutations associated with lineages of interest. See example file at &quot;https://github.com/mikesovic/MixviR/blob/main/mutation_files/outbreak_20220217.csv&quot;. Additional columns will be ignored.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_read.muts.from">read.muts.from</code></td>
<td>
<p>An alternative to <em>muts.df</em> for providing input. If a data frame generated by <code>call_mutations()</code> was previously written to a (comma-separated) file (see <em>write.mut.table</em> in <code>call_mutations()</code>), the mutation data can be read in from that file by providing its path.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_scale">scale</code></td>
<td>
<p>Logical to indicate whether estimated proportions of lineages within a sample should be scaled down to sum to 1 if the sum of the initial estimates is &gt; 1. Default = TRUE.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_use.median">use.median</code></td>
<td>
<p>Logical to define the metric used to estimate frequencies of lineages in samples. Default = FALSE (mean is used).</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_outfile.name">outfile.name</code></td>
<td>
<p>If writing output to file, a character string giving the name/path of the file (csv) to be written.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_presence.thresh">presence.thresh</code></td>
<td>
<p>Numeric (0-1) defining a proportion of characteristic mutations that must be present in the sample for a lineage to be considered present. This threshold is applied if <em>report.all</em> = FALSE (the default).</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_samps.to.inc">samps.to.inc</code></td>
<td>
<p>Character vector of one or more sample names to include. If NULL (default), all samples are included.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_locs.to.inc">locs.to.inc</code></td>
<td>
<p>Character vector of one or more locations to include. If NULL (default), all locations are included. Applies only if a dates file is provided, and these locations must match those in the 'LOCATION' column of that file.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_lineages.to.inc">lineages.to.inc</code></td>
<td>
<p>Character vector of one or more lineages to test for and report in results. If NULL (default), all lineages listed in the lineage.muts file are evaluated and reported.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_report.all">report.all</code></td>
<td>
<p>Logical indicating whether to report results for all lineages (TRUE), or just those with a proportion of mutations present that exceeds <em>presence.thresh</em>. Default FALSE.</p>
</td></tr>
<tr><td><code id="estimate_lineages_+3A_depths.from">depths.from</code></td>
<td>
<p>Character, one of &quot;all&quot; (default) or &quot;characteristic&quot;. If &quot;all&quot;, average sequencing depths are calculated based on all mutations in a sample. If &quot;characteristic&quot;, mean depths are calculated from the set of mutations that occur in only one analyzed lineage (mutations shared by two or more lineages are filtered out prior to calculating depths).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame containing estimates of proportions of each lineage in the sample.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>estimate_lineages(lineage.muts = system.file("extdata", 
                                             "example_lineage_muts.csv", 
                                             package = "MixviR"), 
                  read.muts.from = system.file("extdata", 
                                               "sample_mutations.csv", 
                                               package = "MixviR"))
</code></pre>

<hr>
<h2 id='explore_mutations'>MixviR Shiny Dashboard</h2><span id='topic+explore_mutations'></span>

<h3>Description</h3>

<p>Open dashboard to explore mutation data generated with <code>call_mutations()</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>explore_mutations(
  muts.df,
  dates = NULL,
  lineage.muts = NULL,
  read.muts.from = NULL,
  all.target.muts = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="explore_mutations_+3A_muts.df">muts.df</code></td>
<td>
<p>A data frame (produced by <code>call_mutations()</code>) storing mutation information for samples to analyze. Must contain columns SAMP_NAME, CHR, POS, GENE, ALT_ID, AF, &amp; DP. Alternatively, the mutation data can be read in from a (tab-separated) file with the <code>read.muts.from()</code> argument. See also the <em>write.mut.table</em> argument in <code>call_mutations()</code>.</p>
</td></tr>
<tr><td><code id="explore_mutations_+3A_dates">dates</code></td>
<td>
<p>Path to optional csv file with cols &quot;SAMP_NAME&quot;, &quot;LOCATION&quot;, and &quot;DATE&quot;. Sample names need to match those in the <em>muts.df</em> data frame created by <code>call_mutations()</code>. Dates should be provided in the format <em>mmddyyyy</em>.</p>
</td></tr>
<tr><td><code id="explore_mutations_+3A_lineage.muts">lineage.muts</code></td>
<td>
<p>Path to optional csv file with required cols &quot;Gene&quot;, &quot;Mutation&quot;, and  &quot;Lineage&quot; defining mutations associated with lineages of interest. See example file at &quot;https://github.com/mikesovic/MixviR/blob/main/mutation_files/outbreak_20211202.csv&quot;. Additional columns will be ignored.</p>
</td></tr>
<tr><td><code id="explore_mutations_+3A_read.muts.from">read.muts.from</code></td>
<td>
<p>An alternative to <em>muts.df</em> for providing mutation input. If a data frame generated by <code>call_mutations()</code> was previously written to a (comma separated) file (see <em>write.mut.table</em> in <code>call_mutations()</code>), the mutation data can be read in from that file by providing its path. The fields &quot;SAMP_NAME, CHR, POS, GENE, ALT_ID, AF, DP&quot; must be present (additional fields will be ignored).</p>
</td></tr>
<tr><td><code id="explore_mutations_+3A_all.target.muts">all.target.muts</code></td>
<td>
<p>Logical to indicate whether results for all target mutations were written to the output of the <code>call_mutations()</code> function. See <em>write.all.targets</em> option in <code>call_mutations()</code>. Default FALSE. If TRUE, more informative sequencing depth information can be provided in the dashboard.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Shiny Dashboard to Explore Data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>if (interactive()) {explore_mutations(read.muts.from = system.file("extdata", 
                                                                   "sample_mutations.csv", 
                                                                   package = "MixviR"),
                                      lineage.muts = system.file("extdata", 
                                                                 "example_lineage_muts.csv", 
                                                                 package = "MixviR"))}
if (interactive()) {explore_mutations(read.muts.from = system.file("extdata", 
                                                                   "sample_mutations.csv", 
                                                                   package = "MixviR"), 
                                      dates = system.file("extdata", 
                                                          "example_location_date.csv", 
                                                          package = "MixviR"), 
                                      lineage.muts = system.file("extdata", 
                                                                 "example_lineage_muts.csv", 
                                                                 package = "MixviR"))}
</code></pre>

<hr>
<h2 id='get_codons'>Get Codons From Gene Sequence</h2><span id='topic+get_codons'></span>

<h3>Description</h3>

<p>Group a gene sequence into codons (triplets) that can be used for subsequent translation. If any elements of the character vector have length &gt;1 (insertions in ALT column of VCF), they are trimmed to the first base. Used by id.snps, id.indels, and create_ref functions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_codons(gene.seq, rev = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_codons_+3A_gene.seq">gene.seq</code></td>
<td>
<p>Character vector containing gene sequence. The length of this vector should be equal to the length of sequence.</p>
</td></tr>
<tr><td><code id="get_codons_+3A_rev">rev</code></td>
<td>
<p>Logical indicating whether the reverse complement of the gene.seq should be used.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A character vector containing translated amino acids associated with each nucleotide position in the input vector.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>get_codons(gene.seq = c("A","U","G","C","A","T","T","T","A","C","A","G","T","A","A"))
</code></pre>

<hr>
<h2 id='id_indels'>ID Indel-based Amino Acid Changes</h2><span id='topic+id_indels'></span>

<h3>Description</h3>

<p>Identify amino acid changes associated with indel variation. Changes associated with SNVs are identified in separate function. Used by call_mutations() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>id_indels(variant.calls, ref)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="id_indels_+3A_variant.calls">variant.calls</code></td>
<td>
<p>Data frame with cols POS, REF, ALT, AF, DP. Additional columns will be ignored.</p>
</td></tr>
<tr><td><code id="id_indels_+3A_ref">ref</code></td>
<td>
<p>reference genome in &quot;MixVir&quot; format (from create_ref() function)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame that includes amino acid calls based on indel variants observed in sample. Contains cols &quot;POS&quot;, &quot;REF_BASE&quot;, &quot;GENE&quot;, &quot;REF_CODON&quot;, &quot;REF_AA&quot;, &quot;GENE_AA_POS&quot;, &quot;REF_IDENT&quot;, &quot;REF&quot;, &quot;ALT&quot;, &quot;ALT_freq&quot;, &quot;ALT_COUNT&quot;, &quot;samp_codon&quot;, &quot;samp_AA&quot;, &quot;samp_identity&quot;, &quot;DP&quot;
</p>

<hr>
<h2 id='id_snps'>ID SNV-based Amino Acid Changes</h2><span id='topic+id_snps'></span>

<h3>Description</h3>

<p>Identify amino acid changes associated with single nucleotide variation. Changes associated with indels are identified in separate function. Used by call_mutations() function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>id_snps(variant.calls, ref, code.num = "1")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="id_snps_+3A_variant.calls">variant.calls</code></td>
<td>
<p>Data frame with cols POS, REF, ALT, AF (alt freq), DP (total read depth).</p>
</td></tr>
<tr><td><code id="id_snps_+3A_ref">ref</code></td>
<td>
<p>reference genome in &quot;MixVir&quot; format (from create_ref() function)</p>
</td></tr>
<tr><td><code id="id_snps_+3A_code.num">code.num</code></td>
<td>
<p>Number defining the genetic code to use for translation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame that includes amino acid calls based on SNP/SNV variants observed in sample. Contains cols &quot;POS&quot;, &quot;REF_BASE&quot;, &quot;GENE&quot;, &quot;REF_CODON&quot;, &quot;REF_AA&quot;, &quot;GENE_AA_POS&quot;, &quot;REF_IDENT&quot;, &quot;REF&quot;, &quot;ALT&quot;, &quot;ALT_freq&quot;, &quot;ALT_COUNT&quot;, &quot;samp_codon&quot;, &quot;samp_AA&quot;, &quot;samp_identity&quot;, &quot;DP&quot;
</p>

<hr>
<h2 id='vcf_to_mixvir'>Convert Sample VCF to MixviR Input Format</h2><span id='topic+vcf_to_mixvir'></span>

<h3>Description</h3>

<p>Create data frame with relevant contents of VCF
</p>


<h3>Usage</h3>

<pre><code class='language-R'>vcf_to_mixvir(infile, max.vcf.size = 1e+08)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="vcf_to_mixvir_+3A_infile">infile</code></td>
<td>
<p>Path to a vcf file that must contain &quot;DP&quot; and &quot;AD&quot; flags in the FORMAT field.</p>
</td></tr>
<tr><td><code id="vcf_to_mixvir_+3A_max.vcf.size">max.vcf.size</code></td>
<td>
<p>Max memory usage (in bytes) allowed when reading in vcf file (from vcfR).</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Data frame with cols &quot;CHR&quot;	&quot;POS&quot;	&quot;REF&quot;	&quot;ALT&quot;	&quot;DP&quot; &quot;REF_COUNT&quot;	&quot;ALT_COUNT&quot;
</p>


<h3>Examples</h3>

<pre><code class='language-R'>vcf_to_mixvir(infile = system.file("extdata", "vcfs", "Sample1_04182021.vcf.gz", package="MixviR"))
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
