<!DOCTYPE html><html><head><title>Help for package mlmm.gwas</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mlmm.gwas}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#eBIC_allmodels'><p>Compute eBIC and BIC criteria</p></a></li>
<li><a href='#Estimation_allmodels'><p>Compute estimated effects</p></a></li>
<li><a href='#fromeBICtoEstimation'><p>Helper function that create the selec_XXclass argument of the Estimation_allmodels function</p></a></li>
<li><a href='#frommlmm_toebic'><p>Helper that create the selec_XX argument of eBIC_allmodels()</p></a></li>
<li><a href='#genotypes.boxplot'><p>Boxplots representation of the distributions of phenotypes according to allelic classes</p></a></li>
<li><a href='#manhattan.plot'><p>Manhattan plot</p></a></li>
<li><a href='#mlmm_allmodels'><p>Multi-Locus Mixed-Model</p></a></li>
<li><a href='#mlmm.gwas'><p>mlmm.gwas</p></a></li>
<li><a href='#mlmm.gwas.AD'><p>Dataset for examples with the mlmm.gwas package, additive and additive+dominance models</p></a></li>
<li><a href='#mlmm.gwas.FMI'><p>Dataset for examples with the mlmm.gwas package, male+female and male+female+interaction models</p></a></li>
<li><a href='#run_entire_gwas_pipeline'><p>MLMM, model selection and effects estimation</p></a></li>
<li><a href='#threshold_allmodels'><p>Select significant marker thanks to a p-value threshold</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Pipeline for GWAS Using MLMM</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.6</td>
</tr>
<tr>
<td>Author:</td>
<td>Fanny Bonnafous [aut], Alexandra Duhnen [aut], Louise Gody [aut], Olivier Guillaume[aut, cre], Brigitte Mangin [aut], Prune Pegot-Espagnet [aut], Vincent Segura [aut], Bjarni J. Vilhjalmsson [aut], Clement Mabire [aut], Timothee Flutre [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Clement Mabire &lt;clement.mabire@inra.fr&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>Pipeline for Genome-Wide Association Study using Multi-Locus Mixed Model from Segura V, Vilhjálmsson BJ et al. (2012) &lt;<a href="https://doi.org/10.1038%2Fng.2314">doi:10.1038/ng.2314</a>&gt;. The pipeline include detection of associated SNPs with MLMM, model selection by lowest eBIC and p-value threshold, estimation of the effects of the SNPs in the selected model and graphical functions.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>false</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>multcompView(&ge; 0.1-7), multcomp(&ge; 1.4-8), coxme(&ge; 2.2-5),
sommer(&ge; 3.2), Matrix(&ge; 1.2-10)</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2019-08-05 14:33:59 UTC; cmabire</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2019-08-05 15:20:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='eBIC_allmodels'>Compute eBIC and BIC criteria</h2><span id='topic+eBIC_allmodels'></span>

<h3>Description</h3>

<p>Compute log likelihood, BIC and eBIC.
</p>
<p>The model with the smallest eBIC should be selected.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>eBIC_allmodels(Y, selec_XX, KK, nb.tests, cofs = NULL, female = NULL,
  male = NULL, lambda=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="eBIC_allmodels_+3A_y">Y</code></td>
<td>
<p>A numeric named vector where the names are individuals names and the values their phenotype.</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_selec_xx">selec_XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the models. Use helper function <code><a href="#topic+frommlmm_toebic">frommlmm_toebic</a></code> to get this argument.</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_kk">KK</code></td>
<td>
<p>a list of one, two or three matrices depending on the models
</p>
<p>- additive: a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- additive+dominance: two n by n matrices, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- female+male: a n.female by n.female matrix, with rownames()=colnames()=female names and a n.male by n.male matrix, with rownames()=colnames()=male names
</p>
<p>- female+male+interaction: the same two matrices as the model female+male and a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_nb.tests">nb.tests</code></td>
<td>
<p>number of computed tests (total number of SNPs)</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_cofs">cofs</code></td>
<td>
<p>A n by q matrix, where n=number of individuals, q=number of fixed effect,</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_female">female</code></td>
<td>
<p>A factor of levels female names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_male">male</code></td>
<td>
<p>A factor of levels male names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="eBIC_allmodels_+3A_lambda">lambda</code></td>
<td>
<p>penalty used in the computation of the eBIC; if NULL, the default will be 1 - 1/(2k) with L=n^k where L=total number of SNPs (see function &quot;lambda.calc&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with a line for each mlmm step and 4 columns : BIC, ajout, eBIC_0.5 and LogL.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>

<hr>
<h2 id='Estimation_allmodels'>Compute estimated effects</h2><span id='topic+Estimation_allmodels'></span>

<h3>Description</h3>

<p>Estimate the effect of selected SNPs.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Estimation_allmodels(Y, selec_XXclass, KK, cofs = NULL, female = NULL,
  male = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="Estimation_allmodels_+3A_y">Y</code></td>
<td>
<p>A numeric named vector where the names are individuals names and the values their phenotype. The names of Y will be matched to the row names of X.</p>
</td></tr>
<tr><td><code id="Estimation_allmodels_+3A_selec_xxclass">selec_XXclass</code></td>
<td>
<p>A n by mk data.frame of factors
with rownames()=individual names, and colnames()=mk selected SNP names
additive+dominance: three levels factor
female+male+interaction: four levels factor
</p>
<p>Use function <code><a href="#topic+fromeBICtoEstimation">fromeBICtoEstimation</a></code> to get this armument.</p>
</td></tr>
<tr><td><code id="Estimation_allmodels_+3A_kk">KK</code></td>
<td>
<p>a list of one, two or three matrices depending on the models
</p>
<p>- additive: a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- additive+dominance: two n by n matrices, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- female+male: a n.female by n.female matrix, with rownames()=colnames()=female names and a n.male by n.male matrix, with rownames()=colnames()=male names
</p>
<p>- female+male+interaction: the same two matrices as the model female+male and a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names</p>
</td></tr>
<tr><td><code id="Estimation_allmodels_+3A_cofs">cofs</code></td>
<td>
<p>A n by q matrix, where n=number of individuals, q=number of fixed effect, with rownames()=individual names and with column names, forbidden head of column names for this matrix &quot;eff1_&quot; and usage of special characters as &quot;*&quot;,&quot;/&quot;,&quot;&amp;&quot;</p>
</td></tr>
<tr><td><code id="Estimation_allmodels_+3A_female">female</code></td>
<td>
<p>A factor of levels female names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="Estimation_allmodels_+3A_male">male</code></td>
<td>
<p>A factor of levels male names and length n, only for the last two models</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A dataframe with 3 colum: BLUE, Tukey.Class and Frequency. The firt line name is &quot;mu&quot;, the names of the other lines are in the form markername_allele.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>

<hr>
<h2 id='fromeBICtoEstimation'>Helper function that create the selec_XXclass argument of the Estimation_allmodels function</h2><span id='topic+fromeBICtoEstimation'></span>

<h3>Description</h3>

<p>Function that create the selec_XXclass argument of the <code><a href="#topic+Estimation_allmodels">Estimation_allmodels</a></code> function from the output of the <code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code> function or from the output of the <code><a href="#topic+threshold_allmodels">threshold_allmodels</a></code> function.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>fromeBICtoEstimation(XX, res.eBIC, res.threshold)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="fromeBICtoEstimation_+3A_xx">XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the model. Matrices are n by m matrix, where n=number of individuals, m=number of SNPs, with rownames(X)=individual names, and colnames(X)=SNP names.
</p>
<p>- additive: a single matrix
</p>
<p>- additive+dominance: two matrices
</p>
<p>- female+male: two matrices with the female one first
</p>
<p>- female+male+interaction: three matrices with the female first, the male then the interaction</p>
</td></tr>
<tr><td><code id="fromeBICtoEstimation_+3A_res.ebic">res.eBIC</code></td>
<td>
<p>output of the <code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code> function</p>
</td></tr>
<tr><td><code id="fromeBICtoEstimation_+3A_res.threshold">res.threshold</code></td>
<td>
<p>output of the <code><a href="#topic+threshold_allmodels">threshold_allmodels</a></code> function</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code> <code><a href="#topic+Estimation_allmodels">Estimation_allmodels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>

<hr>
<h2 id='frommlmm_toebic'>Helper that create the selec_XX argument of eBIC_allmodels()</h2><span id='topic+frommlmm_toebic'></span>

<h3>Description</h3>

<p>Helper function that create the selec_XX argument of <code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code> from the output of <code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>frommlmm_toebic(XX, res.mlmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="frommlmm_toebic_+3A_xx">XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the model. Matrices are n by m matrix, where n=number of individuals, m=number of SNPs, with rownames(X)=individual names, and colnames(X)=SNP names.
Use the same XX you used with the <code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code> function</p>
</td></tr>
<tr><td><code id="frommlmm_toebic_+3A_res.mlmm">res.mlmm</code></td>
<td>
<p>Output from the <code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code> function</p>
</td></tr>
</table>


<h3>See Also</h3>

<p><code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code> <code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>

<hr>
<h2 id='genotypes.boxplot'>Boxplots representation of the distributions of phenotypes according to allelic classes</h2><span id='topic+genotypes.boxplot'></span>

<h3>Description</h3>

<p>For each allele class of a given loci, display as boxplot the distributions of the phenotypes of individuals with this allele class.
</p>
<p>For instance, it can be used as a simple representation of the effects of one QTL.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>genotypes.boxplot(X, Y, markers = "all", effects = NULL,
  genotypes = c("00", "01|10", "11"), tukeyTextCol = NA, tukeyTextCex = 1,
  tukeyCol = c("#2ecc71", "#3498db", "#9b59b6", "#6c7a89", "#f2ca27",
  "#e67e22", "#e74c3c", "#c08d57"), tukeyPch = c(1, 3, 2, 4:8),
  tukeyCex = 1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="genotypes.boxplot_+3A_x">X</code></td>
<td>
<p>A matrix where rownames are individuals names, colnames are markers names, and values are genotypes. Genotypes are encoded as allelic dosage (0, 1, 2) or as any numeric values     as long as the smallest and highest values correspond to homozygous and the mean of these smallest and highest values to heterozygous. Other values (imputated genotypes) will be rounded to the nearest.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_y">Y</code></td>
<td>
<p>A numeric named vector where the names are individuals names and the values their phenotype. The names of Y will be matched to the row names of X.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_markers">markers</code></td>
<td>
<p>A vector of names of markers, a plot will be drawn for each of them. &quot;all&quot; is a special value meaning a plot will be drawn for all markers in the estimations object, or in the matrix X if the estimations object is not provided.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_effects">effects</code></td>
<td>
<p>A GWAS.EFFECTS oject, created with <code>Estimation_allmodels</code> function.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_genotypes">genotypes</code></td>
<td>
<p>A length 3 string vector, used as labels for the genotypes.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_tukeytextcol">tukeyTextCol</code></td>
<td>
<p>Colors of the letters of the Tukey classes.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_tukeytextcex">tukeyTextCex</code></td>
<td>
<p>Size of the letters of the Tukey classes.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_tukeycol">tukeyCol</code></td>
<td>
<p>Color of the symbols of the Tukey classes.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_tukeypch">tukeyPch</code></td>
<td>
<p>Symbols of the Tukey classes.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_tukeycex">tukeyCex</code></td>
<td>
<p>Size of the symbols of the Tukey classes.</p>
</td></tr>
<tr><td><code id="genotypes.boxplot_+3A_...">...</code></td>
<td>
<p>Additional arguments are passed to the <code>boxplot</code> function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>A plot is drawn for each of marker of the markers vector.
</p>
<p>In each of thes plots, a boxplot is drawn for each allelic classes. Theses boxplots represent the distribution of the phenotypes of individuals with these allelic classes.
</p>
<p>If the effects parameter is not NULL, the Tukey classes of the effects of markers will be represented as a symbol and/or a letter in the boxplot. The ordinates of these symbols is the average of the phenotype of individuals with the allele.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)
</code></pre>

<hr>
<h2 id='manhattan.plot'>Manhattan plot</h2><span id='topic+manhattan.plot'></span>

<h3>Description</h3>

<p>Draw a Manhattan plot of the association p-values of the markers.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>manhattan.plot(res.mlmm, map = NULL, steps = 1, hideCofactors = FALSE,
  chrToPlot = "all", unit = "cM", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="manhattan.plot_+3A_res.mlmm">res.mlmm</code></td>
<td>
<p>Output object from <code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code>.</p>
</td></tr>
<tr><td><code id="manhattan.plot_+3A_map">map</code></td>
<td>
<p>Dataframe with 3 columns : markers names, chromosome or scaffold names and position (any unit is allowed: cM, Mpb etc.).</p>
</td></tr>
<tr><td><code id="manhattan.plot_+3A_steps">steps</code></td>
<td>
<p>An integer. The iteration number of the forward approach. If a vector of length &gt;= 2 is passed, several plots will be drawn. By default, only step 1 is drawn.</p>
</td></tr>
<tr><td><code id="manhattan.plot_+3A_hidecofactors">hideCofactors</code></td>
<td>
<p>If TRUE, the coffactors (fixed effects) won't be drawn</p>
</td></tr>
<tr><td><code id="manhattan.plot_+3A_chrtoplot">chrToPlot</code></td>
<td>
<p>Names of the chromosomes or scaffolds to plot. Use this if you want to zoom on a particular chromosome.</p>
</td></tr>
<tr><td><code id="manhattan.plot_+3A_unit">unit</code></td>
<td>
<p>Unit of the positions in the map.</p>
</td></tr>
<tr><td><code id="manhattan.plot_+3A_...">...</code></td>
<td>
<p>additional arguments can be passed to the plot function.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Draws a manhattan plot ie. plot -log(p-value) vs marker position
</p>
<p>If a map is passed, markers position will be used as x axis.
If not, the indices of markers inside the res.mlmm object will be used instead
</p>
<p>If there are cofactors (as in all but the first step of the forward approch),
the cofactors markers will be plotted too (symbol: star).
</p>
<p>If a map is passed, markers not in the map or in the map but not assigned to a chromosome
will be assigned to a virtual chromosome 0.
</p>
<p>Markers in the map, assigned to a chromosome, but with missing position, will be
ploted at the end of the chromosome.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

## End(Not run)
</code></pre>

<hr>
<h2 id='mlmm_allmodels'>Multi-Locus Mixed-Model</h2><span id='topic+mlmm_allmodels'></span>

<h3>Description</h3>

<p>Carry GWAS correcting for population structure while including cofactors through a forward regression approach.
</p>
<p>possible models: additive, additive+dominance, female+male, female+male+interaction
</p>
<p>For additive model, look at the example below or at this <a href="../doc/gwas-manual.html">vignette</a>. For other models, read <a href="https://link.springer.com/article/10.1007/s00122-017-3003-4">Bonnafous et al. (2017)</a>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mlmm_allmodels(Y, XX, KK, nbchunks = 2, maxsteps = 20, cofs = NULL,
  female = NULL, male = NULL, threshold=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mlmm_allmodels_+3A_y">Y</code></td>
<td>
<p>A numeric named vector where the names are individuals' names and the values their phenotype. The names of Y will be matched to the row names of X.</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_xx">XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the model. Matrices are n by m matrix, where n=number of individuals, m=number of SNPs, with rownames(X)=individual names, and colnames(X)=SNP names.
</p>
<p>- additive: a single matrix
</p>
<p>- additive+dominance: two matrices
</p>
<p>- female+male: two matrices with the female one first
</p>
<p>- female+male+interaction: three matrices with the female first, the male then the interaction</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_kk">KK</code></td>
<td>
<p>a list of one, two or three matrices depending on the models
</p>
<p>- additive: a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- additive+dominance: two n by n matrices, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- female+male: a n.female by n.female matrix, with rownames()=colnames()=female names and a n.male by n.male matrix, with rownames()=colnames()=male names
</p>
<p>- female+male+interaction: the same two matrices as the model female+male and a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_nbchunks">nbchunks</code></td>
<td>
<p>An integer defining the number of chunks of matrices to run the analysis, allows to decrease the memory usage. minimum=2, increase it if you do not have enough memory</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_maxsteps">maxsteps</code></td>
<td>
<p>An integer &gt;= 3. Maximum number of steps desired in the forward approach. The forward approach breaks automatically once the pseudo-heritability is close to 0, however to avoid doing too many steps in case the pseudo-heritability does not reach a value close to 0, this parameter is also used.</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_cofs">cofs</code></td>
<td>
<p>A n by q matrix, where n=number of individuals, q=number of fixed effect, with rownames()=individual names and with column names, forbidden head of column names for this matrix &quot;eff1_&quot; and usage of special characters as &quot;*&quot;,&quot;/&quot;,&quot;&amp;&quot;</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_female">female</code></td>
<td>
<p>A factor of levels female names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_male">male</code></td>
<td>
<p>A factor of levels male names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="mlmm_allmodels_+3A_threshold">threshold</code></td>
<td>
<p>a value to declare the significant p value. The default value is Bonferroni 0.05</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each of the data arguments must be sorted in the same way, according to the individual name.
</p>


<h3>Value</h3>

<p>a list with one element per step of the forward approach. Each element of this list is a named vector of p-values, the names are the names of the markers, with &quot;selec_&quot; as prefix for the markers used as fixed effects.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+manhattan.plot">manhattan.plot</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Data for additive and additive+dominance models

data("mlmm.gwas.AD")

### Additive model ###

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC, res.threshold)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## Not run: 
### Additive + dominance model

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
#the selected model is the null model

# Marker selection 
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)

### Data for female+male and female+male+interaction

data("mlmm.gwas.FMI")

### Female+Male model

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

# Marker selection 
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)


### Female+Male+Interaction model

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Model selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateFMI, sel_XX, KK, ncol(Xf), female = female, male = male)
#the selected model is the null model

# Marker selection 
res.threshold &lt;- threshold_allmodels(threshold=NULL, res_mlmm)


## End(Not run)
</code></pre>

<hr>
<h2 id='mlmm.gwas'>mlmm.gwas</h2><span id='topic+mlmm.gwas'></span><span id='topic+mlmm.gwas-package'></span>

<h3>Description</h3>

<p>Pipeline for GWAS using Multi Locus Mixed Model (MLMM).
</p>


<h3>Details</h3>

<p>This is a fork of the MLMM / <a href="https://github.com/Gregor-Mendel-Institute/MultLocMixMod">MultLocMixMod</a> package by Vincent Segura and Bjarni J. Vilhjalmsson.
</p>
<p>The main differencies from the original package are:
</p>

<ul>
<li><p> abandon of the multi-Bonferroni model selection
</p>
</li>
<li><p> abandon of the backward model search
</p>
</li>
<li><p> eBIC modified to be adapted to the rate between number of individuals and number of markers.
</p>
</li>
<li><p> function added to select significant SNP according to a p-value threshold at each mlmm step
</p>
</li>
<li><p> new models supported: additive+dominance, male+female and male+female+interaction. These models are described in <a href="https://doi.org/10.1007/s00122-017-3003-4">Bonnafous et al. (2017)</a>.
</p>
</li>
<li><p> graphical functions: a new Manhattan plot and a boxplot representation of markers effects.
</p>
</li></ul>

<p>A <a href="../doc/gwas-manual.html">vignette</a> presents the usage of this package with the additive model.
</p>


<h3>References</h3>

<p><a href="https://doi.org/10.1007/s00122-017-3003-4">Bonnafous, F., Fievet, G., Blanchet, N. et al. Theor Appl Genet (2018) 131: 319.</a>
</p>
<p><a href="https://doi.org/10.1038/ng.2314">Segura V, Vilhjálmsson BJ, Platt A, Korte A, Seren Uuml, Long Q, et al. An efficient multi-locus mixed-model approach for genome-wide association studies in structured populations. Nat Genet. 2012;44:825–830.</a>
</p>

<hr>
<h2 id='mlmm.gwas.AD'>Dataset for examples with the mlmm.gwas package, additive and additive+dominance models</h2><span id='topic+mlmm.gwas.AD'></span><span id='topic+floweringDateAD'></span><span id='topic+Xa'></span><span id='topic+Xd'></span><span id='topic+K.add'></span><span id='topic+K.dom'></span>

<h3>Description</h3>


<ul>
<li><p> Species: Helianthus annuus
</p>
</li>
<li><p> Individuals: 125
</p>
</li>
<li><p> Markers: 500
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(mlmm.gwas.AD)
</code></pre>


<h3>Format</h3>


<ul>
<li><p> floweringDateAD: a named numeric of length 444.
</p>
</li>
<li><p> Xa: a 110x500 numeric matrix
</p>
</li>
<li><p> Xd: a 110x500 numeric matrix
</p>
</li>
<li><p> K.add: a 110x110 numeric matrix
</p>
</li>
<li><p> K.dom: a 110x110 numeric matrix
</p>
</li></ul>


<h3>Details</h3>

<p>Variables:
</p>

<ul>
<li><p> floweringDateAD: flowering dates in °C.day, time since sowing.
</p>
</li>
<li><p> Xa: genotype matrix (additive)
</p>
</li>
<li><p> Xd: genotype matrix (dominance)
</p>
</li>
<li><p> K.add: &quot;kinship&quot; matrix (additive)
</p>
</li>
<li><p> K.dom: &quot;kinship&quot; matrix (dominance)
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://link.springer.com/article/10.1007/s00122-017-3003-4">Bonnafous &amp; al. (2017)</a>
</p>

<hr>
<h2 id='mlmm.gwas.FMI'>Dataset for examples with the mlmm.gwas package, male+female and male+female+interaction models</h2><span id='topic+mlmm.gwas.FMI'></span><span id='topic+floweringDateFMI'></span><span id='topic+female'></span><span id='topic+male'></span><span id='topic+hybrid'></span><span id='topic+Xf'></span><span id='topic+Xm'></span><span id='topic+Xfm'></span><span id='topic+K.female'></span><span id='topic+K.male'></span><span id='topic+K.hybrid'></span>

<h3>Description</h3>


<ul>
<li><p> Species: Helianthus annuus
</p>
</li>
<li><p> Individuals: 303
</p>
</li>
<li><p> Markers: 500
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>data(mlmm.gwas.FMI)
</code></pre>


<h3>Format</h3>


<ul>
<li><p> floweringDateFMI: a named numeric vector of length 303.
</p>
</li>
<li><p> female: a factor of length 303
</p>
</li>
<li><p> male: a factor of length 303
</p>
</li>
<li><p> hybrid: a factor of length 303
</p>
</li>
<li><p> Xf: a 303x500 numeric matrix
</p>
</li>
<li><p> Xm: a 303x500 numeric matrix
</p>
</li>
<li><p> Xfm: a 303x500 numeric matrix
</p>
</li>
<li><p> K.female: 36x36 numeric matrix
</p>
</li>
<li><p> K.male: 36x36 numeric matrix
</p>
</li>
<li><p> K.hybrid: 303x303 numeric matrix
</p>
</li></ul>


<h3>Details</h3>

<p>Variables:
</p>

<ul>
<li><p> floweringDateFMI: flowering dates in °C.day, time since sowing.
</p>
</li>
<li><p> female: names of the female parent of the individuals
</p>
</li>
<li><p> male: names of the male parent of the individuals
</p>
</li>
<li><p> hybrid: names of the hybrids (name of female and male)
</p>
</li>
<li><p> Xf: female genotype matrix (additive)
</p>
</li>
<li><p> Xm: male genotype matrix (dominance)
</p>
</li>
<li><p> Xfm: female-male interaction genotype matrix (dominance)
</p>
</li>
<li><p> K.female: female &quot;kinship&quot; matrix (additive)
</p>
</li>
<li><p> K.male: male &quot;kinship&quot; matrix (dominance)
</p>
</li>
<li><p> K.hybrid: hybrid &quot;kinship&quot; matrix (dominance)
</p>
</li></ul>



<h3>Source</h3>

<p><a href="https://link.springer.com/article/10.1007/s00122-017-3003-4">Bonnafous &amp; al. (2017)</a>
</p>

<hr>
<h2 id='run_entire_gwas_pipeline'>MLMM, model selection and effects estimation</h2><span id='topic+run_entire_gwas_pipeline'></span>

<h3>Description</h3>

<p>Internaly run functions of the mlmm.gwas package:
</p>

<ul>
<li> <p><code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code> (GWAS)
</p>
</li>
<li> <p><code><a href="#topic+frommlmm_toebic">frommlmm_toebic</a></code>
</p>
</li>
<li> <p><code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code> (model selection)
</p>
</li>
<li> <p><code><a href="#topic+threshold_allmodels">threshold_allmodels</a></code>
</p>
</li>
<li> <p><code><a href="#topic+fromeBICtoEstimation">fromeBICtoEstimation</a></code>
</p>
</li>
<li> <p><code><a href="#topic+Estimation_allmodels">Estimation_allmodels</a></code> (effects estimation)
</p>
</li></ul>



<h3>Usage</h3>

<pre><code class='language-R'>run_entire_gwas_pipeline(Y, XX, KK, nbchunks = 2, maxsteps = 20,
  cofs = NULL, female = NULL, male = NULL,threshold=NULL,lambda=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="run_entire_gwas_pipeline_+3A_y">Y</code></td>
<td>
<p>A numeric named vector where the names are individuals' names and the values their phenotype. The names of Y will be matched to the row names of X.</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_xx">XX</code></td>
<td>
<p>A list of length one, two or three matrices depending on the model. Matrices are n by m matrix, where n=number of individuals, m=number of SNPs, with rownames(X)=individual names, and colnames(X)=SNP names.
</p>
<p>- additive: a single matrix
</p>
<p>- additive+dominance: two matrices
</p>
<p>- female+male: two matrices with the female one first
</p>
<p>- female+male+interaction: three matrices with the female first, the male then the interaction</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_kk">KK</code></td>
<td>
<p>a list of one, two or three matrices depending on the models
</p>
<p>- additive: a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- additive+dominance: two n by n matrices, where n=number of individuals, with rownames()=colnames()=individual names
</p>
<p>- female+male: a n.female by n.female matrix, with rownames()=colnames()=female names and a n.male by n.male matrix, with rownames()=colnames()=male names
</p>
<p>- female+male+interaction: the same two matrices as the model female+male and a n by n matrix, where n=number of individuals, with rownames()=colnames()=individual names</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_nbchunks">nbchunks</code></td>
<td>
<p>An integer defining the number of chunks of matrices to run the analysis, allows to decrease the memory usage. minimum=2, increase it if you do not have enough memory</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_maxsteps">maxsteps</code></td>
<td>
<p>An integer &gt;= 3. Maximum number of steps desired in the forward approach. The forward approach breaks automatically once the pseudo-heritability is close to 0, however to avoid doing too many steps in case the pseudo-heritability does not reach a value close to 0, this parameter is also used.</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_cofs">cofs</code></td>
<td>
<p>A n by q matrix, where n=number of individuals, q=number of fixed effect, with rownames()=individual names and with column names, forbidden head of column names for this matrix &quot;eff1_&quot; and usage of special characters as &quot;*&quot;,&quot;/&quot;,&quot;&amp;&quot;</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_female">female</code></td>
<td>
<p>A factor of levels female names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_male">male</code></td>
<td>
<p>A factor of levels male names and length n, only for the last two models</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_threshold">threshold</code></td>
<td>
<p>a value to declare the significant p value. The default value is Bonferroni 0.05</p>
</td></tr>
<tr><td><code id="run_entire_gwas_pipeline_+3A_lambda">lambda</code></td>
<td>
<p>penalty used in the computation of the eBIC; if NULL, the default will be 1 - 1/(2k) with L=n^k where L=total number of SNPs (see function &quot;lambda.calc&quot;)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A named list with 2 or 3 elements:
</p>

<ul>
<li><p> pval: the return value of <code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code>
</p>
</li>
<li><p> eBic: the return value of <code><a href="#topic+eBIC_allmodels">eBIC_allmodels</a></code>
</p>
</li>
<li><p> threshold: the return value of <code><a href="#topic+threshold_allmodels">threshold_allmodels</a></code>
</p>
</li>
<li><p> effects: the return value of <code><a href="#topic+Estimation_allmodels">Estimation_allmodels</a></code>, only if there is at least one marker in the model selected by lowest eBIC.
</p>
</li></ul>



<h3>Examples</h3>

<pre><code class='language-R'>data("mlmm.gwas.AD")
results &lt;- run_entire_gwas_pipeline(floweringDateAD, list(Xa), list(K.add))
</code></pre>

<hr>
<h2 id='threshold_allmodels'>Select significant marker thanks to a p-value threshold</h2><span id='topic+threshold_allmodels'></span>

<h3>Description</h3>

<p>Select significant marker at each mlmm step according to a threshold.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>threshold_allmodels(threshold=NULL, res_mlmm)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="threshold_allmodels_+3A_threshold">threshold</code></td>
<td>
<p>a value to declare the significant p value. The default value is Bonferroni 0.05</p>
</td></tr>
<tr><td><code id="threshold_allmodels_+3A_res_mlmm">res_mlmm</code></td>
<td>
<p>a list of p-value for each mlmm step. Use helper function <code><a href="#topic+mlmm_allmodels">mlmm_allmodels</a></code> to get this argument.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A matrix with a line for significant SNP at each mlmm step (according to the defined threshold) and 3 columns : SNP, p-value, step
</p>


<h3>Examples</h3>

<pre><code class='language-R'>### Additive model ###
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa)
KK = list(K.add)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Model and Marker selection
sel_XX &lt;- frommlmm_toebic(XX, res_mlmm)
res.eBIC &lt;- eBIC_allmodels(floweringDateAD, sel_XX, KK, ncol(Xa))
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

# Effects estimations with the selected model
sel_XXclass &lt;- fromeBICtoEstimation(sel_XX, res.eBIC, res.threshold)
eff.estimations &lt;- Estimation_allmodels(floweringDateAD, sel_XXclass, KK)
genotypes.boxplot(Xa, floweringDateAD, effects = eff.estimations)

## End(Not run)

### Additive + dominance model
## Not run: 
data("mlmm.gwas.AD")

XX = list(Xa, Xd)
KK = list(K.add, K.dom)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateAD, XX, KK)
manhattan.plot(res_mlmm)

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

## End(Not run)


### Female+Male model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm)
KK = list(K.female, K.male)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

## End(Not run)



### Female+Male+Interaction model
## Not run: 
data("mlmm.gwas.FMI")

XX = list(Xf, Xm, Xfm)
KK = list(K.female, K.male, K.hybrid)

# GWAS
res_mlmm &lt;- mlmm_allmodels(floweringDateFMI, XX, KK, female = female, male = male)
manhattan.plot(res_mlmm)

# Marker selection
res.threshold &lt;- threshold_allmodels(threshold, res_mlmm)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
