<!DOCTYPE html><html lang="en"><head><title>Help for package MLMOI</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MLMOI}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#corrector_numeric'><p>Removes punctuation characters and typos from data</p>
entries</a></li>
<li><a href='#corrector_string'><p>Removes punctuation characters and typos from data</p>
entries</a></li>
<li><a href='#decoder_aminoacid'><p>Translates the standard ambiguity codes for nucleotides</p>
(amino acid decoder)</a></li>
<li><a href='#decoder_codon'><p>Translates the standard ambiguity codes for nucleotides</p>
(codon decoder)</a></li>
<li><a href='#decoder_snp'><p>Converts ambeguity codes to represented bases</p></a></li>
<li><a href='#decoder_str'><p>Transforms entries to the desired coding class</p></a></li>
<li><a href='#mle'><p>Derives MLE</p></a></li>
<li><a href='#mle_fixed'><p>Derives profile-likelihood MLE of lineage frequencies</p></a></li>
<li><a href='#MLMOI'><p>MLMOI: An R Package to preprocess molecular data and</p>
derive prevalences, frequencies and multiplicity of
infection (MOI)</a></li>
<li><a href='#moi_administrator'><p>Administrator function</p></a></li>
<li><a href='#moi_duplicatefinder'><p>Finds forbidden sample ID repetitions</p></a></li>
<li><a href='#moi_empty'><p>Reports and deletes empty rows/columns</p></a></li>
<li><a href='#moi_export'><p>Exports the dataset in standard format to a new excel</p>
file.</a></li>
<li><a href='#moi_labels'><p>Finds sample IDs contained in a dataset</p></a></li>
<li><a href='#moi_marker'><p>Extracts lineages of samples at a specific marker</p></a></li>
<li><a href='#moi_mergemetadata'><p>Merges metadata</p></a></li>
<li><a href='#moi_metadata'><p>Checks the metadata entries</p></a></li>
<li><a href='#moi_nk'><p>Derives lineage prevalence counts</p></a></li>
<li><a href='#moi_prerequisite'><p>Contains different character vectors needed for importing</p>
molecular dataset</a></li>
<li><a href='#moi_separator'><p>Finds the most frequent separator</p></a></li>
<li><a href='#moi_warning'><p>Writes warnings into a file</p></a></li>
<li><a href='#moimerge'><p>Merges two molecular datasets.</p></a></li>
<li><a href='#moimle'><p>Estimates prevalences, frequency spectra and MOI parameter.</p></a></li>
<li><a href='#moimport'><p>Imports molecular data in various formats and transforms</p>
them into a standard format.</a></li>
<li><a href='#tempMLMOI'><p>tempMLMOI: An R package for deriving multiplicity of</p>
infection (MOI) from molecular data.</a></li>
<li><a href='#withWarnings'><p>Captures all the warnings</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Estimating Frequencies, Prevalence and Multiplicity of Infection</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.2</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Meraj Hashemi &lt;meraj.hashemi.esh@gmail.com&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The implemented methods reach out to scientists that seek to estimate multiplicity of infection (MOI) and lineage (allele) frequencies and prevalences at molecular markers using the maximum-likelihood method described in Schneider (2018) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0194148">doi:10.1371/journal.pone.0194148</a>&gt;, and Schneider and Escalante (2014) &lt;<a href="https://doi.org/10.1371%2Fjournal.pone.0097899">doi:10.1371/journal.pone.0097899</a>&gt;. Users can import data from Excel files in various formats, and perform maximum-likelihood estimation on the imported data by the package's moimle() function.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.3.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>openxlsx (&ge; 4.2.5.2), Rdpack (&ge; 2.6), Rmpfr (&ge; 0.9-3),</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RdMacros:</td>
<td>Rdpack</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-12-21 19:29:48 UTC; meraj</td>
</tr>
<tr>
<td>Author:</td>
<td>Meraj Hashemi [cre, aut, com],
  Kristan Schneider [aut, ths]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-12-21 22:30:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='corrector_numeric'>Removes punctuation characters and typos from data
entries</h2><span id='topic+corrector_numeric'></span>

<h3>Description</h3>

<p>This function is designed to find the
lineages (STRs) present on a microsatellite marker in a
single cell.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corrector_numeric(y, c_l, r_w, conm, cons, cha_num, rw_col, multsh)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="corrector_numeric_+3A_y">y</code></td>
<td>
<p>string; a cell entry.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_r_w">r_w</code></td>
<td>
<p>numeric; sample ID's row number in the excel
file.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_conm">conm</code></td>
<td>
<p>numeric; the multiple column per marker
identifier. For the data of format multiple columns
conm &gt; 1.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_cons">cons</code></td>
<td>
<p>numeric; the multiple row per sample
identifier. For the data of format multiple rows cons &gt;
1.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_cha_num">cha_num</code></td>
<td>
<p>string vector; the vector of punctuation
characters plus alphabets. See
<code><a href="#topic+moi_prerequisite">moi_prerequisite</a></code>.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="corrector_numeric_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of following elements: 1) a vector of
lineages found on a microsatellite marker in a single
cell. Each element corresponds to one and only one
lineage and it is free from any punctuation character.
2) an identifier whose value is 1 if a warning takes
place.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimport">moimport</a></code>,
<code><a href="#topic+moi_marker">moi_marker</a></code>.
</p>

<hr>
<h2 id='corrector_string'>Removes punctuation characters and typos from data
entries</h2><span id='topic+corrector_string'></span>

<h3>Description</h3>

<p>This function is designed to find the
lineages present on a SNP, amino-acid and codon marker
in a single cell.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>corrector_string(y, c_l, r_w, conm, cons, cha_string, rw_col, coding, multsh)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="corrector_string_+3A_y">y</code></td>
<td>
<p>string; entry of a cell.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_r_w">r_w</code></td>
<td>
<p>numeric; sample ID's row number in the excel
file.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_conm">conm</code></td>
<td>
<p>numeric; the multiple column per marker
identifier. For the data of format multiple columns
conm &gt; 1.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_cons">cons</code></td>
<td>
<p>numeric; the multiple row per sample
identifier. For the data of format multiple rows cons &gt;
1.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_cha_string">cha_string</code></td>
<td>
<p>string vector; the vector of
punctuation characters plus numerics form 1 to 9. See
<code><a href="#topic+moi_prerequisite">moi_prerequisite</a></code>.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="corrector_string_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of following elements: 1) a vector of
lineages found on a marker (SNP, amino-acid or codon)
in a single cell. Each element corresponds to one and
only one lineage and it is free from typos, 2) an
identifier whose value is 1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimport">moimport</a></code>,
<code><a href="#topic+moi_marker">moi_marker</a></code>.
</p>

<hr>
<h2 id='decoder_aminoacid'>Translates the standard ambiguity codes for nucleotides
(amino acid decoder)</h2><span id='topic+decoder_aminoacid'></span>

<h3>Description</h3>

<p>Translates the standard ambiguity codes for
nucleotides in amino acid forms from a pre-specified
coding class to 3-letter designation of amino acids.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decoder_aminoacid(
  y,
  c_l,
  r_w,
  aa_1,
  aa_2,
  let_3,
  amino_acid,
  aa_symbol,
  coding,
  rw_col,
  multsh
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="decoder_aminoacid_+3A_y">y</code></td>
<td>
<p>numeric vector; entries in a cell corresponding
to a specific sample and a specific marker.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_r_w">r_w</code></td>
<td>
<p>numeric; sample ID's row number in the excel
file.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_aa_1">aa_1</code></td>
<td>
<p>string vector; vector of different amino
acids.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_aa_2">aa_2</code></td>
<td>
<p>string vector; vector of different codons.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_let_3">let_3</code></td>
<td>
<p>string vector; vector of amino acids in 3
letter designation.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_amino_acid">amino_acid</code></td>
<td>
<p>string vector; vector of amino acids in
full name.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_aa_symbol">aa_symbol</code></td>
<td>
<p>string vector; vector of amino acids in
one letter designation.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_coding">coding</code></td>
<td>
<p>string; coding class of the molecular
marker.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="decoder_aminoacid_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two elements: 1) a vector of 3-letter
designation of amino acids on a marker corresponding to
a sample in pre-specified coding class. 2) an identifier
whose value is 1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimport">moimport</a></code>,
<code><a href="#topic+moi_marker">moi_marker</a></code> and
<code><a href="#topic+corrector_string">corrector_string</a></code>. See also the
vignette 'StandardAmbiguityCodes'.
</p>

<hr>
<h2 id='decoder_codon'>Translates the standard ambiguity codes for nucleotides
(codon decoder)</h2><span id='topic+decoder_codon'></span>

<h3>Description</h3>

<p>Translates the standard ambiguity codes for
nucleotides in codon form from a pre-specified coding
class to triplet designation of codons.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decoder_codon(
  y,
  c_l,
  r_w,
  aa_1,
  aa_2,
  compact,
  codon_s,
  coding,
  rw_col,
  multsh
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="decoder_codon_+3A_y">y</code></td>
<td>
<p>numeric vector; entries in a cell corresponding
to a specific sample and a specific marker.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_r_w">r_w</code></td>
<td>
<p>numeric; sample ID's row number in the excel
file.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_aa_1">aa_1</code></td>
<td>
<p>string vector; vector of different amino
acids.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_aa_2">aa_2</code></td>
<td>
<p>string vector; vector of different codons.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_compact">compact</code></td>
<td>
<p>string vector; vector of different codons
in compact form.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_codon_s">codon_s</code></td>
<td>
<p>string vector; vector of different codons.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_coding">coding</code></td>
<td>
<p>string; coding class of the molecular
marker.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="decoder_codon_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two elements: 1) a vector of codons in
triplet designation on a marker corresponding to a
sample in pre-specified coding class. 2) an identifier
whose value is 1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimport">moimport</a></code>,
<code><a href="#topic+moi_marker">moi_marker</a></code> and
<code><a href="#topic+corrector_string">corrector_string</a></code>.
</p>

<hr>
<h2 id='decoder_snp'>Converts ambeguity codes to represented bases</h2><span id='topic+decoder_snp'></span>

<h3>Description</h3>

<p>Translates the nucleotide ambiguity codes as
defined in DNA Sequence Assembler from a pre-specified
coding class to 4-letter codes.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decoder_snp(
  y,
  c_l,
  r_w,
  ambeguity_code,
  represented_bases,
  coding,
  rw_col,
  multsh
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="decoder_snp_+3A_y">y</code></td>
<td>
<p>numeric vector; entries in a cell corresponding
to a specific sample and a specific marker.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_r_w">r_w</code></td>
<td>
<p>numeric; sample ID's row number in the excel
file.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_ambeguity_code">ambeguity_code</code></td>
<td>
<p>string vector; ambeguity codes for
snp data.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_represented_bases">represented_bases</code></td>
<td>
<p>string vector; represented bases
for those ambeguity codes.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_coding">coding</code></td>
<td>
<p>string; coding class of the molecular
marker.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="decoder_snp_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two elements: 1) a vector of
represented bases on a marker corresponding to a sample
in pre-specified coding class. 2) an identifier whose
value is 1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimport">moimport</a></code>,
<code><a href="#topic+moi_marker">moi_marker</a></code> and
<code><a href="#topic+corrector_string">corrector_string</a></code>. See also the vignette
'StandardAmbiguityCodes'.
</p>

<hr>
<h2 id='decoder_str'>Transforms entries to the desired coding class</h2><span id='topic+decoder_str'></span>

<h3>Description</h3>

<p>Transforms the data entries in a cell to a
pre-specified coding class.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>decoder_str(y, c_l, r_w, coding, rw_col, multsh)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="decoder_str_+3A_y">y</code></td>
<td>
<p>numeric vector; entries in a cell corresponding
to a specific sample and a specific marker.</p>
</td></tr>
<tr><td><code id="decoder_str_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="decoder_str_+3A_r_w">r_w</code></td>
<td>
<p>numeric; sample ID's row number in the excel
file.</p>
</td></tr>
<tr><td><code id="decoder_str_+3A_coding">coding</code></td>
<td>
<p>string; coding class of the molecular
marker.</p>
</td></tr>
<tr><td><code id="decoder_str_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="decoder_str_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of two elements: 1) a vector of STR
entries in pre-specified coding class. 2) an identifier
whose value is 1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimport">moimport</a></code>,
<code><a href="#topic+moi_marker">moi_marker</a></code>.
</p>

<hr>
<h2 id='mle'>Derives MLE</h2><span id='topic+mle'></span>

<h3>Description</h3>

<p>Derives MLE
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle(nnk, nn)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_+3A_nnk">nnk</code></td>
<td>
<p>numeric vector; vector of lineage prevalence
counts.</p>
</td></tr>
<tr><td><code id="mle_+3A_nn">nn</code></td>
<td>
<p>numeric; sample size</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following elements: 1) log
likelihood at MLE, MLE of lambda and MLE of psi, 2) MLE
of lineage frequencies.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimle">moimle</a></code>
</p>

<hr>
<h2 id='mle_fixed'>Derives profile-likelihood MLE of lineage frequencies</h2><span id='topic+mle_fixed'></span>

<h3>Description</h3>

<p>Derives profile-likelihood MLE of lineage frequencies
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mle_fixed(lambda, nnk)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mle_fixed_+3A_lambda">lambda</code></td>
<td>
<p>numeric; MOI parameter</p>
</td></tr>
<tr><td><code id="mle_fixed_+3A_nnk">nnk</code></td>
<td>
<p>numeric vector; vector of lineage prevalence
counts.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>vector of lineage frequency estimates
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimle">moimle</a></code>
</p>

<hr>
<h2 id='MLMOI'>MLMOI: An R Package to preprocess molecular data and
derive prevalences, frequencies and multiplicity of
infection (MOI)</h2><span id='topic+MLMOI'></span>

<h3>Description</h3>

<p>The MLMOI package provides three functions: </p>

<ul>
<li><p> moimport(); </p>
</li>
<li><p> moimle(); </p>
</li>
<li><p> moimerge().</p>
</li></ul>



<h3>Details</h3>

<p>The package reaches out to scientists that seek to
estimate MOI and lineage frequencies at molecular markers
using the maximum-likelihood method described in
(Schneider 2018),
(Schneider and Escalante 2018) and
(Schneider and Escalante 2014). Users can
import data from Excel files in various formats, and
perform maximum-likelihood estimation on the imported data
by the package's <code>moimle()</code> function.
</p>


<h3>Types of molecular data</h3>

<p>Molecular data can be of
types: </p>
 <ul>
<li><p> microsatellite repeats (STRs);
</p>
</li>
<li><p> single nucleotide polymorphisms (SNPs); </p>
</li>
<li>
<p>amino acids; </p>
</li>
<li><p> codons (base triplets). </p>
</li></ul>



<h3>Import function</h3>

<p>The function <code>moimport()</code>, is
designed to import molecular data. It imports molecular
data in various formats and transforms them into a
standard format.
</p>


<h3>Merging Datasets</h3>

<p>Two datasets in standard format
can be merged with the function <code>moimerge()</code>.
</p>


<h3>Estimation MOI and frequencies</h3>

<p>The function
<code>moimle()</code> is designed to derive MLE from molecular
data in standard format.
</p>


<h3>References</h3>

<p>Schneider KA (2018).
&ldquo;Large and finite sample properties of a maximum-likelihood estimator for multiplicity of infection.&rdquo;
<em>PLOS ONE</em>, <b>13</b>(4), 1-21.
<a href="https://doi.org/10.1371/journal.pone.0194148">doi:10.1371/journal.pone.0194148</a>.
</p>
<p>Schneider KA, Escalante AA (2018).
&ldquo;Correction: A Likelihood Approach to Estimate the Number of Co-Infections.&rdquo;
<em>PLOS ONE</em>, <b>13</b>(2), 1-3.
<a href="https://doi.org/10.1371/journal.pone.0192877">doi:10.1371/journal.pone.0192877</a>.
</p>
<p>Schneider KA, Escalante AA (2014).
&ldquo;A Likelihood Approach to Estimate the Number of Co-Infections.&rdquo;
<em>PLoS ONE</em>, <b>9</b>(7), e97899.
<a href="http://dx.doi.org/10.1371%2Fjournal.pone.0097899">http://dx.doi.org/10.1371%2Fjournal.pone.0097899</a>.
</p>

<hr>
<h2 id='moi_administrator'>Administrator function</h2><span id='topic+moi_administrator'></span>

<h3>Description</h3>

<p>Administrator function
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_administrator(
  set_d,
  s_total,
  m_total,
  nummtd,
  cha,
  rw_col,
  nwsh,
  transposed,
  multsh
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_administrator_+3A_set_d">set_d</code></td>
<td>
<p>data frame; imported dataset.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_s_total">s_total</code></td>
<td>
<p>string vector; vector of sample IDs.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_m_total">m_total</code></td>
<td>
<p>string vector;vector of marker labels.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric; number of metadata columns plus 2.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_cha">cha</code></td>
<td>
<p>string vector; vector of punctuation
characters. See <code><a href="#topic+moi_prerequisite">moi_prerequisite</a></code>.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_nwsh">nwsh</code></td>
<td>
<p>numeric; worksheet number in multiple
worksheet dataset.</p>
</td></tr>
<tr><td><code id="moi_administrator_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of following elements: 1) rows in which
new samples start, 2) all sample IDs in the worksheet,
3) number of samples in the worksheet, 3) multiple row
per sample identifier, 4) another multiple row per
sample identifier, 5) columns in which new markers
start, 6) all marker labels in the worksheet, 7) number
of markers in the worksheet, 8) multiple column per
marker identifier, 9) another multiple column per
marker identifier, 10) an identifier whose value is 1
if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>.
</p>

<hr>
<h2 id='moi_duplicatefinder'>Finds forbidden sample ID repetitions</h2><span id='topic+moi_duplicatefinder'></span>

<h3>Description</h3>

<p>Sample IDs need to be uniquely assigned to
samples. This function checks if a sample ID is
assigned to two or more different samples. Similarly,
the marker labels need to be uniquely defined. This
function is also used to check forbidden marker label
repetitions.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_duplicatefinder(total, sam_mark, nummtd, rw_col)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_duplicatefinder_+3A_total">total</code></td>
<td>
<p>string vector; vector of sample IDs.</p>
</td></tr>
<tr><td><code id="moi_duplicatefinder_+3A_sam_mark">sam_mark</code></td>
<td>
<p>string; a string which is either &quot;Sample
ID&quot; or &quot;Marker&quot;.</p>
</td></tr>
<tr><td><code id="moi_duplicatefinder_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric; number of metadata columns plus 2.</p>
</td></tr>
<tr><td><code id="moi_duplicatefinder_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a warning which informs the user of forbidden
repetition of sample ID's or marker labels.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>
<code><a href="#topic+moi_labels">moi_labels</a></code>.
</p>

<hr>
<h2 id='moi_empty'>Reports and deletes empty rows/columns</h2><span id='topic+moi_empty'></span>

<h3>Description</h3>

<p>Reports and deletes empty rows/columns.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_empty(
  set_d,
  setnoempty,
  nummtd,
  rw_col,
  multsheets,
  alllabels,
  molecular,
  molid,
  n
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_empty_+3A_set_d">set_d</code></td>
<td>
<p>data frame; imported dataset.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_setnoempty">setnoempty</code></td>
<td>
<p>data frame; imported dataset.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric; number of metadata columns plus 2.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_multsheets">multsheets</code></td>
<td>
<p>logical; indicating whether data is
contained in a single or multiple worksheets. The
default value is <code> multsheets = FALSE</code>,
corresponding to data contained in a single worksheet.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_alllabels">alllabels</code></td>
<td>
<p>all the marker labels.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_molecular">molecular</code></td>
<td>
<p>molecular argument.</p>
</td></tr>
<tr><td><code id="moi_empty_+3A_molid">molid</code></td>
<td>
<p>identifier. It is greater than zero when
molecular argument is just a single value.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list of 4 elements: 1) dataset without empty
rows/columns; 2) dataset with empty rows/columns; 3)
coorected labels; 4)corrected number of metadata
columns.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>.
</p>

<hr>
<h2 id='moi_export'>Exports the dataset in standard format to a new excel
file.</h2><span id='topic+moi_export'></span>

<h3>Description</h3>

<p>This function exports the modified dataset
in standard format to a new excel file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_export(export, final)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_export_+3A_export">export</code></td>
<td>
<p>string; the path where the imported data is
stored in standardized format.</p>
</td></tr>
<tr><td><code id="moi_export_+3A_final">final</code></td>
<td>
<p>data frame; modified dataset in standard
format.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An excel file which contains dataset in standard
format.
</p>


<h3>See Also</h3>

<p>For further details, please see the following
functions: <code>moimport</code>.
</p>

<hr>
<h2 id='moi_labels'>Finds sample IDs contained in a dataset</h2><span id='topic+moi_labels'></span>

<h3>Description</h3>

<p>Each dataset consists of several number of
samples. Samples are specified by their sample ID which
are placed in the first column of the excel worksheet.
The function <code>moi_labels</code>, finds sample IDs and
the row in which they start.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_labels(s_total)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_labels_+3A_s_total">s_total</code></td>
<td>
<p>sample IDs.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list which the first element is a numeric
vector specifying the number of rows in which a new
sample starts. The second element is the name of sample
IDs.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>
and <code><a href="#topic+moi_duplicatefinder">moi_duplicatefinder</a></code>.
</p>

<hr>
<h2 id='moi_marker'>Extracts lineages of samples at a specific marker</h2><span id='topic+moi_marker'></span>

<h3>Description</h3>

<p>For a specific marker, the function goes
from one sample to another and finds lineages with the
help of the following functions:
<code><a href="#topic+corrector_numeric">corrector_numeric</a></code> along with
<code><a href="#topic+decoder_str">decoder_str</a></code>,
<code><a href="#topic+corrector_string">corrector_string</a></code> along with
<code><a href="#topic+decoder_aminoacid">decoder_aminoacid</a></code> and
<code><a href="#topic+corrector_string">corrector_string</a></code> along with
<code><a href="#topic+decoder_snp">decoder_snp</a></code>. Each of these functions are
suitable for a particular type of molecular data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_marker(
  col_j,
  c_l,
  sam,
  samorder,
  conm,
  cons,
  molecular,
  coding,
  cha_num,
  cha_string,
  ambeguity_code,
  represented_bases,
  aa_1,
  aa_2,
  let_3,
  amino_acid,
  aa_symbol,
  compact,
  codon_s,
  rw_col,
  multsh
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_marker_+3A_col_j">col_j</code></td>
<td>
<p>vector; column vector of a specific marker.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_c_l">c_l</code></td>
<td>
<p>string; marker label.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_sam">sam</code></td>
<td>
<p>numeric vector; vector which its elements
specify where a new sample starts.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_samorder">samorder</code></td>
<td>
<p>a vector which its elements specify where
a new sample starts.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_conm">conm</code></td>
<td>
<p>numeric; the multiple column identifier. For
the data of format multiple columns conm &gt; 1.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_cons">cons</code></td>
<td>
<p>numeric; the multiple row identifier. For the
data of format multiple rows conm &gt; 1.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_molecular">molecular</code></td>
<td>
<p>string; type of molecular data.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_coding">coding</code></td>
<td>
<p>string; coding class of the molecular
marker.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_cha_num">cha_num</code></td>
<td>
<p>string vector; vector of symbols (used for
microsatellite data).</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_cha_string">cha_string</code></td>
<td>
<p>string vector; vector of symbols (used
for snp and amino acid).</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_ambeguity_code">ambeguity_code</code></td>
<td>
<p>string vector; ambeguity codes for
snp data.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_represented_bases">represented_bases</code></td>
<td>
<p>string vector; represented bases
for those ambeguity codes.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_aa_1">aa_1</code></td>
<td>
<p>string vector; vector of different amino
acids.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_aa_2">aa_2</code></td>
<td>
<p>string vector; vector of different codons.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_let_3">let_3</code></td>
<td>
<p>string vector; vector of amino acids in
3-letter designation.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_amino_acid">amino_acid</code></td>
<td>
<p>string vector; vector of amino acids in
full name.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_aa_symbol">aa_symbol</code></td>
<td>
<p>string vector; vector of amino acids in
one letter designation.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_compact">compact</code></td>
<td>
<p>string vector; vector of different codons
in compact form.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_codon_s">codon_s</code></td>
<td>
<p>string vector; vector of different codons.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="moi_marker_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with the following elements: 1) a list
with elements containing lineages for a specific sample
on a specific marker. The order in which samples are
entered in dataset is preserved in the list. The
lineages are free from typos and are transformed to
pre-specified coding class, 2) an identifier whose
value is 1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details, please see the following
functions: <code>moimport</code>
</p>

<hr>
<h2 id='moi_mergemetadata'>Merges metadata</h2><span id='topic+moi_mergemetadata'></span>

<h3>Description</h3>

<p>Merges metadata
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_mergemetadata(tempmtd, mtdall, samall, multsh)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_mergemetadata_+3A_tempmtd">tempmtd</code></td>
<td>
<p>matrix; matrix of temporary metadata.</p>
</td></tr>
<tr><td><code id="moi_mergemetadata_+3A_mtdall">mtdall</code></td>
<td>
<p>string vector; vector of all metadata
labels.</p>
</td></tr>
<tr><td><code id="moi_mergemetadata_+3A_samall">samall</code></td>
<td>
<p>string vector; vector of all sample IDs in
the worksheet.</p>
</td></tr>
<tr><td><code id="moi_mergemetadata_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list of following elements: 1) unique metadata
labels, 2) unique metadata for different samples.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>.
</p>

<hr>
<h2 id='moi_metadata'>Checks the metadata entries</h2><span id='topic+moi_metadata'></span>

<h3>Description</h3>

<p>Checks if a sample has unique metadata
entries.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_metadata(
  metadata,
  mtdlabels,
  nummtd,
  samorder,
  samall,
  lsam,
  nomtdneeded,
  nomerge,
  multsh
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_metadata_+3A_metadata">metadata</code></td>
<td>
<p>matrix; matrix of metadata columns.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_mtdlabels">mtdlabels</code></td>
<td>
<p>string vector; vector of metadata
labels.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric; number of metadata columns plus 2.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_samorder">samorder</code></td>
<td>
<p>numeric vector; rows where samples start.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_samall">samall</code></td>
<td>
<p>string vector; vector of sample IDs in the
worksheet.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_lsam">lsam</code></td>
<td>
<p>numeric; number of samples in the worksheet.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_nomtdneeded">nomtdneeded</code></td>
<td>
<p>numeric vector; samples which need no
metadata.</p>
</td></tr>
<tr><td><code id="moi_metadata_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of following elements: 1) unique metadata
for different samples, 2) an identifier whose value is
1 if a warning takes place.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>.
</p>

<hr>
<h2 id='moi_nk'>Derives lineage prevalence counts</h2><span id='topic+moi_nk'></span>

<h3>Description</h3>

<p>Derives lineage prevalence counts
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_nk(datmarker, samorder)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_nk_+3A_datmarker">datmarker</code></td>
<td>
<p>vector; a column of data corresponding
to a marker from the imported dataset.</p>
</td></tr>
<tr><td><code id="moi_nk_+3A_samorder">samorder</code></td>
<td>
<p>numeric vector; row numbers in excel file
where the new samples start.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list with two elements: 1) sample size of the
marker, 2) vector of lineage prevalence counts at the
marker.
</p>


<h3>See Also</h3>

<p>For further details see: <code><a href="#topic+moimle">moimle</a></code>
</p>

<hr>
<h2 id='moi_prerequisite'>Contains different character vectors needed for importing
molecular dataset</h2><span id='topic+moi_prerequisite'></span>

<h3>Description</h3>

<p>Contains different character vectors needed for importing
molecular dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_prerequisite()
</code></pre>

<hr>
<h2 id='moi_separator'>Finds the most frequent separator</h2><span id='topic+moi_separator'></span>

<h3>Description</h3>

<p>This function is activated when the data is
of 'One row per sample, one column per marker' format.
In such a dataset, the user enters multiple information
corresponding to a sample on a marker in one cell and
separates the items with a symbol (punctuation
character). It is expected that user be consistent with
usage of symbol. Otherwise this function addresses the
inconsistencies with a warning.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_separator(set_d, nummtd, cha, rw_col, nwsh, multsh)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_separator_+3A_set_d">set_d</code></td>
<td>
<p>data frame; imported dataset.</p>
</td></tr>
<tr><td><code id="moi_separator_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric; number of metadata columns plus 2.</p>
</td></tr>
<tr><td><code id="moi_separator_+3A_cha">cha</code></td>
<td>
<p>string vector; vector of punctuation
characters. See <code><a href="#topic+moi_prerequisite">moi_prerequisite</a></code>.</p>
</td></tr>
<tr><td><code id="moi_separator_+3A_rw_col">rw_col</code></td>
<td>
<p>string vector; variable used to switch
between row and column in case of transposed data.
Namely, <code>c("rows ", "row ", "column ", "columns
")</code>.</p>
</td></tr>
<tr><td><code id="moi_separator_+3A_nwsh">nwsh</code></td>
<td>
<p>numeric; worksheet number in multiple
worksheet dataset.</p>
</td></tr>
<tr><td><code id="moi_separator_+3A_multsh">multsh</code></td>
<td>
<p>string; reports warnings for multiple
worksheet datasets.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a warning is generated reporting inconsistencies
in usage of separator.
</p>


<h3>See Also</h3>

<p>For further details see: <code>moimport</code>.
</p>

<hr>
<h2 id='moi_warning'>Writes warnings into a file</h2><span id='topic+moi_warning'></span>

<h3>Description</h3>

<p>This function exports the modified
dataset in standard format to a new excel file.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moi_warning(
  keepwarnings,
  general_warnings,
  metadata_warnings,
  marker_warnings,
  markerlabels
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moi_warning_+3A_keepwarnings">keepwarnings</code></td>
<td>
<p>string; the path where the
warnings are stored.</p>
</td></tr>
<tr><td><code id="moi_warning_+3A_general_warnings">general_warnings</code></td>
<td>
<p>list; general warnings.</p>
</td></tr>
<tr><td><code id="moi_warning_+3A_metadata_warnings">metadata_warnings</code></td>
<td>
<p>list; metadata warnings.</p>
</td></tr>
<tr><td><code id="moi_warning_+3A_marker_warnings">marker_warnings</code></td>
<td>
<p>list; marker warnings.</p>
</td></tr>
<tr><td><code id="moi_warning_+3A_markerlabels">markerlabels</code></td>
<td>
<p>string vector; marker labels.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>An excel file which contains warnings.
Each worksheet corresponds to a marker
</p>


<h3>See Also</h3>

<p>For further details, please see the following
functions: <code>moimport</code>.
</p>

<hr>
<h2 id='moimerge'>Merges two molecular datasets.</h2><span id='topic+moimerge'></span>

<h3>Description</h3>

<p>The function is designed to merge two
datasets from separate Excel files. The data in
each Excel file is placed in the first
worksheet.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moimerge(
  file1,
  file2,
  nummtd1,
  nummtd2,
  keepmtd = FALSE,
  export = NULL,
  keepwarnings = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moimerge_+3A_file1">file1</code></td>
<td>
<p>string; specifying the path of the first
dataset.</p>
</td></tr>
<tr><td><code id="moimerge_+3A_file2">file2</code></td>
<td>
<p>string; specifying the path of the second
dataset.</p>
</td></tr>
<tr><td><code id="moimerge_+3A_nummtd1">nummtd1</code></td>
<td>
<p>numeric; number of metadata columns (see
<code><a href="#topic+moimport">moimport</a>()</code>) in the first file (default as
<code>0</code>).</p>
</td></tr>
<tr><td><code id="moimerge_+3A_nummtd2">nummtd2</code></td>
<td>
<p>numeric; number of metadata columns (see
<code><a href="#topic+moimport">moimport</a>()</code>) in the second file (default as
<code>0</code>).</p>
</td></tr>
<tr><td><code id="moimerge_+3A_keepmtd">keepmtd</code></td>
<td>
<p>logical; determining whether metadata
(e.g., date) should be retained (default as
<code>TRUE</code>).</p>
</td></tr>
<tr><td><code id="moimerge_+3A_export">export</code></td>
<td>
<p>string; the path where the data is stored.</p>
</td></tr>
<tr><td><code id="moimerge_+3A_keepwarnings">keepwarnings</code></td>
<td>
<p>string; the path where the warnings
are stored.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The two datasets should be already in standard
format (see <code><a href="#topic+moimport">moimport</a>()</code>). The datasets
are placed in the first worksheet of the two different
Excel files. Notice that marker labels (=column
labels) need to be unique.
</p>


<h3>Value</h3>

<p>The output is a dataset in standard format which
constitutes of an assembly of the input datasets.
</p>


<h3>Warnings</h3>

<p>Warnings are generated if potential
inconsistencies are detected. E.g., if the same sample
occurs in both datasets and have contradicting metadata
entries. The function only prints the first 50 warnings.
If the number of warnings are more than 50, the user is
recommended to set the argument <code>keepwarnings</code>,
in order to save the warnings in an Excel file.
</p>


<h3>See Also</h3>

<p>To import and transform data into standard
format, please see the function <code><a href="#topic+moimport">moimport</a>()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#The datasets 'testDatamerge1.xlsx' and 'testDatamerge1.xlsx' are already in standard format:

infile1 &lt;- system.file("extdata", "testDatamerge1.xlsx", package = "MLMOI")
infile2 &lt;- system.file("extdata", "testDatamerge2.xlsx", package = "MLMOI")
outfile &lt;- moimerge(infile1, infile2, nummtd1 = 1, nummtd2 = 2, keepmtd = TRUE)

</code></pre>

<hr>
<h2 id='moimle'>Estimates prevalences, frequency spectra and MOI parameter.</h2><span id='topic+moimle'></span>

<h3>Description</h3>

<p><code>moimle()</code> derives the maximum-likelihood
estimate (MLE) of the MOI parameter (Poisson parameter)
and the lineage (allele) frequencies for each molecular
marker in a dataset. Additionally, the lineage
prevalence counts are derived.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moimle(file, nummtd = 0, bounds = c(NA, NA))
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moimle_+3A_file">file</code></td>
<td>
<p>string or data.frame; if file is a path it
must specify the path to the file to be imported. The
dataset can also be a data.frame object in R. The dataset
must be in standard format (see <code><a href="#topic+moimport">moimport</a>()</code>).
The first column must contain sample IDs. Adjacent columns
can contain metadata, followed by columns corresponding
to molecular markers.</p>
</td></tr>
<tr><td><code id="moimle_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric; number of metadata columns (e.g.
date, sample location, etc.) in the dataset (default
value is <code>nummtd = 0</code>).</p>
</td></tr>
<tr><td><code id="moimle_+3A_bounds">bounds</code></td>
<td>
<p>numeric vector; a vector of size 2,
specifying a lower bound (1st element) and an upper
bound (2nd element) for the MOI parameter. The function
derives lineage frequency ML estimates by profiling the
likelihood function on one of the bounds. For a marker
without sign of super-infections, the lower bound is
employed. If one allele is contained in every sample,
the upper bound is employed.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code>moimle()</code> requires a dataset in standard
format which is free of typos (e.g. incompatible and
unidentified entries). Therefore, users need to
standardize the dataset by employing the
<code><a href="#topic+moimport">moimport</a>()</code> function.
</p>
<p>If one or more molecular markers contain pathological
data, the ML estimate for the Poisson parameter is
either 0 or does not exist. Both estimates are
meaningless, however, in the former case frequency
estimates exist while they do not in the later. By
setting the option <code>bounds</code> as a range for MOI
parameter <code class="reqn">\lambda</code>. i.e., <code>bounds = </code>
c(&lt;<code class="reqn">\lambda_min</code>&gt;, &lt;<code class="reqn">\lambda_max</code>&gt;), this
problem is bypassed and the ML estimates are calculated
by profiling at <code class="reqn">\lambda_min</code> or <code class="reqn">\lambda_max</code>.
If no super-infections are observed at a marker,
<code>moimle()</code> uses <code class="reqn">\lambda_min</code> as the MOI
parameter estimate, <code class="reqn">\lambda_max</code> if one lineage is
present in all samples. For regular data, the
profile-likelihood estimate using <code class="reqn">\lambda_min</code> or
<code class="reqn">\lambda_max</code> is returned depending on whether the
ML estimate falls below <code class="reqn">\lambda_min</code> or above
<code class="reqn">\lambda_max</code>.
</p>


<h3>Value</h3>

<p><code>moimle()</code> returns a nested list,  where the
outer elements correspond to molecular markers in the
dataset. The inner elements for each molecular marker
contain the following information: </p>
<ol>
<li>
<p>sample size, </p>
</li>
<li><p> allele prevalence counts, </p>
</li>
<li>
<p>observed prevalences </p>
</li>
<li><p> log likelihood at MLE, </p>
</li>
<li>
<p>maximum-likelihood estimate of MOI parameter, </p>
</li>
<li>
<p>maximum-likelihood estimates of lineage frequencies.</p>
</li></ol>



<h3>Warnings</h3>

<p>Warnings are issued, if data is
pathological at one or multiple markers. If the option
<code>bounds</code> is set, but MLE of MOI parameter at a
molecular marker takes a lower or higher value than
<code class="reqn">\lambda_min or \lambda_max</code> respectively, a warning
is generated.
</p>


<h3>See Also</h3>

<p>To import and transform data to standard format,
please see the function <code><a href="#topic+moimport">moimport</a>()</code>.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
#basic data analysis
infile1 &lt;- system.file("extdata", "testDatamerge1.xlsx", package = "MLMOI")
mle1 &lt;- moimle(infile1, nummtd = 1)

</code></pre>

<hr>
<h2 id='moimport'>Imports molecular data in various formats and transforms
them into a standard format.</h2><span id='topic+moimport'></span>

<h3>Description</h3>

<p><code>moimport()</code> imports molecular data from
Excel workbooks. The function handles various types of
molecular data (e.g. STRs, SNPs), codings (e.g. 4-letter
vs. IUPAC format for SNPs), and detects inconsistencies
(e.g. typos, incorrect entries). <code>moimport()</code>
allows users to import data from single or multiple
worksheets.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>moimport(
  file,
  multsheets = FALSE,
  nummtd = 0,
  molecular = "str",
  coding = "integer",
  transposed = FALSE,
  keepmtd = FALSE,
  export = NULL,
  keepwarnings = NULL
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="moimport_+3A_file">file</code></td>
<td>
<p>string; specifying the path to the file to be
imported.</p>
</td></tr>
<tr><td><code id="moimport_+3A_multsheets">multsheets</code></td>
<td>
<p>logical; indicating whether data is
contained in a single or multiple worksheets. The
default value is <code>multsheets = FALSE</code>,
corresponding to data contained in a single worksheet.</p>
</td></tr>
<tr><td><code id="moimport_+3A_nummtd">nummtd</code></td>
<td>
<p>numeric number or vector; number of metadata
columns (e.g. date, sample location, etc.) in the
worksheet(s) to be imported (default value <code>nummtd
= 0</code>). In case of multiple worksheet dataset, if all
worksheets have the same number of metadata columns an
integer value is sufficient. If the numbers differ, they
have to be specified by an integer vector.</p>
</td></tr>
<tr><td><code id="moimport_+3A_molecular">molecular</code></td>
<td>
<p>string vector or list; specifies the type
of molecular data to be imported. STR, SNP, amino acid
and codon markers are specified with 'STR', 'SNP',
'amino' and 'codon' values, respectively (default value
<code>molecular = 'str'</code>). For importing single
worksheets, <code>molecular</code> is a single string or
string vector. When importing multiple worksheets,
<code>molecular</code> is a string in case the data contains
only one type of molecular data. Else it is a list, with
the k-th element being a string value or a vector
describing the data types of the k-th worksheet.</p>
</td></tr>
<tr><td><code id="moimport_+3A_coding">coding</code></td>
<td>
<p>string vector or list; specifies the coding
of each data variable (marker) depending on their type.
Admissible values for <code>coding</code> depend on molecular
data types are: 'integer', 'nearest', 'ceil' and 'floor'
for STRs; SNPs with '4let' and 'iupac' for SNPs; '3let',
'1let' and 'full' amino acids and 'triplet' and
'compact' for codons.</p>
</td></tr>
<tr><td><code id="moimport_+3A_transposed">transposed</code></td>
<td>
<p>logical or logical vector; if markers
are entered in rows and samples in columns, set
<code>transposed = TRUE</code> (default value <code>transposed
= FALSE</code>). When importing multiple worksheets,
<code>transposed</code> can be logical vector specifying for
each worksheet whether it is in transposed format.</p>
</td></tr>
<tr><td><code id="moimport_+3A_keepmtd">keepmtd</code></td>
<td>
<p>logical; determines whether metadata (e.g.,
date) should be retained during import (default value
<code>keepmtd = TRUE</code>).</p>
</td></tr>
<tr><td><code id="moimport_+3A_export">export</code></td>
<td>
<p>string; the path where the imported data is
stored in standardized format. Data is not stored if no
path is specified (default value <code>export = NULL</code>).</p>
</td></tr>
<tr><td><code id="moimport_+3A_keepwarnings">keepwarnings</code></td>
<td>
<p>string; the path where the warnings
are stored. Warnings are not stored if no path is
specified (default value <code>keepwarnings = NULL</code>).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Each worksheet of the data to be imported must
have one of the following formats: i) one row per sample
and one column per marker. Here cells can have multiple
entries, separated by a special character (separator),
e.g. a punctuation character. ii) one column per marker
and multiple rows per sample (standard format). iii) one
row per sample and multiple columns per marker.
Importantly, within one worksheet formats ii) and iii)
cannot be combined (see section Warnings and Errors).
Combinations of other formats are permitted but might
result in warnings. Additionally, Occurrence  of
different separators are reported (see section Warnings
and Errors).
</p>
<p>Users should check the following before data import:
</p>
 <ul>
<li><p> the dataset is placed in the first
worksheet of the workbook; </p>
</li>
<li><p> in case of multiple
worksheets, all worksheets contain data (additional
worksheets need to be removed); </p>
</li>
<li><p> sample IDs are
placed in the first column (first row in case of
transposed data; see section Exceptions); </p>
</li>
<li><p> marker
labels are placed in the first row (first column in case
of transposed data; see section Exceptions); </p>
</li>
<li>
<p>sample IDs and as well the marker labels are unique (the
duplication of ID/labels are allowed when sample/marker
contains data in consecutive rows/columns); </p>
</li>
<li>
<p>entries such as sentences (e.g. comments in the
worksheet) or meaningless words (e.g. 'missing' for
missing data) are removed from data; </p>
</li>
<li><p> metadata
columns (rows in case of transposed data) are placed
between sample IDs and molecular-marker columns.</p>
</li></ul>

<p>If data is contained in multiple worksheets, above
requirements need to be fulfilled for every worksheet in
the Excel workbook. Not all sample IDs must occur in
every worksheet. The sample ID must not be confused with
the patient's ID, the former refers to a particular
sample taken from a patient, the latter to a unique
patient. Several sample IDs can have the same patient's
ID. In case of multiple-worksheet datasets, all marker
labels across all worksheets need to be unique.
</p>
<p>The option <code>molecular</code> needs to be specified as a
vector, for single-worksheet data (<code>multsheets =
 FALSE</code>) containing different types of molecular markers.
A list is specified, if data spread across multiple
worksheets with different types of molecular across the
worksheets. List elements are vectors or single values,
referring to the types of molecular data of the
corresponding worksheets. Users do not need to set a
vector if all markers are of the same molecular type
(single or multiple worksheet dataset).
</p>
<p>Setting the option <code>coding</code> as vector or list is
similar to setting molecular type by <code>molecular</code>.
Every molecular data type has a pre-specified coding
class as default which users do not need to specify.
Namely, 'integer' for STRs, '4let' for SNPs, '3let' for
amino acids and 'triplet' for codons.
</p>


<h3>Value</h3>

<p>returns a data frame. <code>moimport()</code> imports
heterogeneous data formats and converts them into a
standard format which are free from typos (e.g.
incompatible and unidentified entries) appropriate for
further analyses. Metadata is retained (if <code>keepmtd
 = TRUE</code>) and, in case of data from multiple worksheets,
unified if metadata variables have the same labels
across two or more worksheets. If the argument
<code>export</code> is set, then the result is saved in the
first worksheet of the workbook of the file specified by
<code>export</code>. The imported/exported dataset will be
appropriate for other functions of the package.
</p>


<h3>Warnings and Errors</h3>

<p>Usually warnings are
generated if data is corrected pointing to suspicious
entries in the original data. Users should read warnings
carefully and check respective entries and apply manual
corrections if necessary. In case of issues an error
occurs and the function is stopped.
</p>
<p>Usually, if arguments are not set properly, errors
occur. Other cases of errors are: i) if sample IDs in a
worksheet are not uniquely defined, i.e., two samples in
non-consecutive rows have the same sample ID; ii) if
formats 'one column per marker and multiple rows per
sample' and 'one row per sample and multiple columns per
marker' are mixed.
</p>
<p>Warnings are issued in several cases. Above all, when
typos (e.g., punctuation characters) are found. Entries
which cannot be identified as a molecular type/coding
class specified by the user are also reported (e.g., '9'
is reported when marker is of type SNPs, or 'L' is
reported when coding class of an amino-acid marker is
'3let').
</p>
<p>Empty rows and columns are deleted and eventually
reported. Samples with ambiguous metadata (in a
worksheet or across worksheets in case of multiple
worksheet dataset), or missing are also reported.
</p>
<p>The function only prints the first 50 warnings.
If the number of warnings are more than 50, the user is
recommended to set the argument <code>keepwarnings</code>,
in order to save the warnings in an Excel file.
</p>


<h3>Exceptions</h3>

<p>Transposed data: usually data is
entered with samples in rows and markers in columns.
However, on the contrary some users might enter data the
opposite way. That is the case of transposed data. If
so, the argument <code>transposed = TRUE</code> is set, or a
vector in case of multiple worksheets with at least one
worksheet being transposed.
</p>


<h3>See Also</h3>

<p>For further details, see the following vignettes:
</p>
<p><code>vignette("dataimportcheck-list", package =
 "MLMOI")</code>
</p>
<p><code>vignette("StandardAmbiguityCodes", package =
 "MLMOI")</code>
</p>
<p><code>vignette("moimport-arguments", package = "MLMOI")</code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#datasets are provided by the package

#importing dataset with metadata variables:
infile &lt;- system.file("extdata", "testDatametadata.xlsx", package = "MLMOI")
moimport(infile, nummtd = 3, keepmtd = TRUE)


##more examples are included in 'examples' vignette:

#vignette("examples", package = "MLMOI")

</code></pre>

<hr>
<h2 id='tempMLMOI'>tempMLMOI: An R package for deriving multiplicity of
infection (MOI) from molecular data.</h2><span id='topic+tempMLMOI'></span>

<h3>Description</h3>

<p>The tempMLMOI package provides three functions: </p>

<ul>
<li><p> moimport; </p>
</li>
<li><p> moimle; </p>
</li>
<li><p> moimerge.</p>
</li></ul>



<h3>Details</h3>

<p>The package reaches out to scientists that seek to
estimate MOI and frequencies at molecular markers using
the maximum-likelihood method described i n Schneider and
Escalante (2018) and Schneider (2018). Users can import
data from &lsquo;<span class="file">.xlsx</span>&rsquo; files in various formats, and
perform maximum-likelihood estimation on the imported data
by the package's <code>mlmoi</code> function.
</p>


<h3>Types of molecular data</h3>

<p>Molecular data can be of
types: </p>
 <ul>
<li><p> microsatellite repeats (STRs);
</p>
</li>
<li><p> single nucleotide polymorphisms (SNPs); </p>
</li>
<li>
<p>amino acids; </p>
</li>
<li><p> codons (base triplets). </p>
</li></ul>



<h3>Import function</h3>

<p>The function <code>moimport</code>, is
designed to import molecular data. It imports molecular
data in various formats and transforms them into a
standard format.
</p>


<h3>Merging Datasets</h3>

<p>Two datasets in standard format
can be merged with the function <code>moimerge</code>.
</p>


<h3>Estimation MOI and frequencies</h3>

<p>The function
<code>moimle</code> is designed to derive MLE from molecular
data in standard format.
</p>

<hr>
<h2 id='withWarnings'>Captures all the warnings</h2><span id='topic+withWarnings'></span>

<h3>Description</h3>

<p>Captures all the warnings
</p>


<h3>Usage</h3>

<pre><code class='language-R'>withWarnings(expr)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="withWarnings_+3A_expr">expr</code></td>
<td>
<p>expression to be evaluated.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>list with two elements. First is the value of the
function. Second is the generated functions by the
expression.
</p>


<h3>See Also</h3>

<p>For further details, please see the following
functions: <code>moimport</code>.
</p>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
