<!DOCTYPE html><html lang="en"><head><title>Help for package mmcm</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mmcm}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mmcm-package'><p>The modified maximum contrast method package</p></a></li>
<li><a href='#mcm.mvt'><p>The maximum contrast method by using the randomized quasi-Monte Carlo method</p></a></li>
<li><a href='#mmcm.mvt'><p>The modified maximum contrast method by using randomized quasi-Monte Carlo method</p></a></li>
<li><a href='#mmcm.resamp'><p>The permuted modified maximum contrast method</p></a></li>
<li><a href='#print.mmcm'><p>Print function for mmcm object</p></a></li>
<li><a href='#reexports'><p>Objects exported from other packages</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Modified Maximum Contrast Method</td>
</tr>
<tr>
<td>Version:</td>
<td>1.2-8</td>
</tr>
<tr>
<td>Imports:</td>
<td>mvtnorm</td>
</tr>
<tr>
<td>Description:</td>
<td>An implementation of modified maximum contrast methods (Sato et al. (2009) &lt;<a href="https://doi.org/10.1038%2Ftpj.2008.17">doi:10.1038/tpj.2008.17</a>&gt;;
  Nagashima et al. (2011) &lt;<a href="https://doi.org/10.2202%2F1544-6115.1560">doi:10.2202/1544-6115.1560</a>&gt;) and the maximum contrast method (Yoshimura et al. (1997)
  &lt;<a href="https://doi.org/10.1177%2F009286159703100213">doi:10.1177/009286159703100213</a>&gt;): Functions mmcm.mvt() and mcm.mvt() give P-value by using randomized quasi-Monte
  Carlo method with pmvt() function of package 'mvtnorm', and mmcm.resamp() gives P-value by using a permutation method.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.0.2</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2020-03-03 19:06:21 UTC; nshi1</td>
</tr>
<tr>
<td>Author:</td>
<td>Kengo Nagashima <a href="https://orcid.org/0000-0003-4529-9045"><img alt="ORCID iD"  src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre],
  Yasunori Sato [aut]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Kengo Nagashima &lt;nshi1201@gmail.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2020-03-04 00:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='mmcm-package'>The modified maximum contrast method package</h2><span id='topic+mmcm-package'></span><span id='topic+mmcm'></span>

<h3>Description</h3>

<p>This package provides an implementation of modified maximum contrast methods and the maximum contrast method.
This version supports functions <code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code>, <code><a href="#topic+mcm.mvt">mcm.mvt</a></code> that gives P-value by using randomized quasi-Monte Carlo method from <code><a href="mvtnorm.html#topic+pmvt">pmvt</a></code> function of package <code>mvtnorm</code>, and <code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code> that gives P-value by using the permutation method.
In a one-way problem testing pattern of several factor level means, the maximum contrast statistics (Yoshimura, I., 1997) may be used. But under unequal sample size situations, denominator of the maximum contrast statistics is overestimated. Thus we propose a modified maximum contrast statistics for the unequal sample size situation. Denominetor of the modified maximum contrast statistics is not influenced under the unequal sample size situation.
</p>


<h3>References</h3>

<p>Nagashima, K., Sato, Y., Hamada, C. (2011).
A modified maximum contrast method for unequal sample sizes in pharmacogenomic studies
<em>Stat Appl Genet Mol Biol.</em> <strong>10</strong>(1): Article 41.
<a href="http://dx.doi.org/10.2202/1544-6115.1560">http://dx.doi.org/10.2202/1544-6115.1560</a>
</p>
<p>Sato, Y., Laird, N.M., Nagashima, K., et al. (2009).
A new statistical screening approach for finding pharmacokinetics-related genes in genome-wide studies.
<em>Pharmacogenomics J.</em> <strong>9</strong>(2): 137&ndash;146.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19104505">http://www.ncbi.nlm.nih.gov/pubmed/19104505</a>
</p>
<p>Yoshimura, I., Wakana, A., Hamada, C. (1997).
A performance comparison of maximum contrast methods to detect dose dependency.
<em>Drug Information J.</em> <strong>31</strong>: 423&ndash;432.
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mcm.mvt">mcm.mvt</a></code>,
<code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code>,
<code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code>
</p>

<hr>
<h2 id='mcm.mvt'>The maximum contrast method by using the randomized quasi-Monte Carlo method</h2><span id='topic+mcm.mvt'></span>

<h3>Description</h3>

<p>This function gives <code class="reqn">P</code>-value for the maximum contrast statistics by using
randomized quasi-Monte Carlo method from <code><a href="mvtnorm.html#topic+pmvt">pmvt</a></code>
function of package <code>mvtnorm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mcm.mvt(
  x,
  g,
  contrast,
  alternative = c("two.sided", "less", "greater"),
  algorithm = GenzBretz()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mcm.mvt_+3A_x">x</code></td>
<td>
<p>a numeric vector of data values</p>
</td></tr>
<tr><td><code id="mcm.mvt_+3A_g">g</code></td>
<td>
<p>a integer vector giving the group for the corresponding elements of x</p>
</td></tr>
<tr><td><code id="mcm.mvt_+3A_contrast">contrast</code></td>
<td>
<p>a numeric contrast coefficient matrix for the maximum contrast
statistics</p>
</td></tr>
<tr><td><code id="mcm.mvt_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the alternative hypothesis,
must be one of &quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;. You can specify
just the initial letter.</p>
</td></tr>
<tr><td><code id="mcm.mvt_+3A_algorithm">algorithm</code></td>
<td>
<p>an object of class <code><a href="mvtnorm.html#topic+algorithms">GenzBretz</a></code>
defining the hyper parameters of this algorithm</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+mcm.mvt">mcm.mvt</a></code> performs the maximum contrast method that is detecting
a true response pattern.
</p>
<p><code class="reqn">Y_{ij} (i=1, 2, \ldots; j=1, 2, \ldots, n_i)</code> is an observed response for <code class="reqn">j</code>-th individual in
<code class="reqn">i</code>-th group.
</p>
<p><code class="reqn">\bm{C}</code> is coefficient matrix for the maximum contrast statistics
(<code class="reqn">i \times k</code> matrix, <code class="reqn">i</code>: No. of groups, <code class="reqn">k</code>: No. of pattern).
</p>
<p style="text-align: center;"><code class="reqn">
  \bm{C}=(\bm{c}_1, \bm{c}_2, \ldots, \bm{c}_k)^{\rm{T}}
</code>
</p>

<p><code class="reqn">\bm{c}_k</code> is coefficient vector of <code class="reqn">k</code>th pattern.
</p>
<p style="text-align: center;"><code class="reqn">
  \bm{c}_k=(c_{k1}, c_{k2}, \ldots, c_{ki})^{\rm{T}} \qquad (\textstyle \sum_i c_{ki}=0)
</code>
</p>

<p><code class="reqn">T_{\max}</code> is the maximum contrast statistic.
</p>
<p style="text-align: center;"><code class="reqn">
  \bar{Y}_i=\frac{\sum_{j=1}^{n_i} Y_{ij}}{n_{i}},
  \bar{\bm{Y}}=(\bar{Y}_1, \bar{Y}_2, \ldots, \bar{Y}_i, \ldots, \bar{Y}_a)^{\rm{T}},
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  \bm{D}=diag(n_1, n_2, \ldots, n_i, \ldots, n_a),
  V=\frac{1}{\gamma}\sum_{j=1}^{n_i}\sum_{i=1}^{a} (Y_{ij}-\bar{Y}_i)^2,
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  \gamma=\sum_{i=1}^{a} (n_i-1),
  T_{k}=\frac{\bm{c}^t_k \bar{\bm{Y}}}{\sqrt{V\bm{c}^t_k \bm{D} \bm{c}_k}},
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  T_{\max}=\max(T_1, T_2, \ldots, T_k).
</code>
</p>

<p>Consider testing the overall null hypothesis <code class="reqn">H_0: \mu_1=\mu_2=\ldots=\mu_i</code>,
versus alternative hypotheses <code class="reqn">H_1</code> for response petterns
(<code class="reqn">H_1: \mu_1&lt;\mu_2&lt;\ldots&lt;\mu_i,~ \mu_1=\mu_2&lt;\ldots&lt;\mu_i,~
\mu_1&lt;\mu_2&lt;\ldots=\mu_i</code>).
The <code class="reqn">P</code>-value for the probability distribution of <code class="reqn">T_{\max}</code>
under the overall null hypothesis is
</p>
<p style="text-align: center;"><code class="reqn">
  P\mbox{-value}=\Pr(T_{\max}&gt;t_{\max} \mid H_0)
</code>
</p>

<p><code class="reqn">t_{\max}</code> is observed value of statistics.
This function gives distribution of <code class="reqn">T_{\max}</code> by using randomized
quasi-Monte Carlo method from package <code>mvtnorm</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistic</code></td>
<td>
<p>the value of the test statistic with a name describing it.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value for the test.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>a character string describing the alternative hypothesis.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>the type of test applied.</p>
</td></tr>
<tr><td><code>contrast</code></td>
<td>
<p>a character string giving the names of the data.</p>
</td></tr>
<tr><td><code>contrast.index</code></td>
<td>
<p>a suffix of coefficient vector of the <code class="reqn">k</code>th pattern
that gives maximum contrast statistics (row number of the coefficient matrix).</p>
</td></tr>
<tr><td><code>error</code></td>
<td>
<p>estimated absolute error and,</p>
</td></tr>
<tr><td><code>msg</code></td>
<td>
<p>status messages.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Yoshimura, I., Wakana, A., Hamada, C. (1997).
A performance comparison of maximum contrast methods to detect dose dependency.
<em>Drug Information J.</em> <strong>31</strong>: 423&ndash;432.
</p>


<h3>See Also</h3>

<p><code><a href="mvtnorm.html#topic+pmvt">pmvt</a></code>,
<code><a href="mvtnorm.html#topic+algorithms">GenzBretz</a></code>,
<code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 ##
#  true response pattern: dominant model c=(1, 1, -2)
set.seed(136885)
x &lt;- c(
  rnorm(130, mean =  1 / 6, sd = 1),
  rnorm( 90, mean =  1 / 6, sd = 1),
  rnorm( 10, mean = -2 / 6, sd = 1)
)
g &lt;- rep(1:3, c(130, 90, 10))
boxplot(
  x ~ g,
  width = c(length(g[g==1]), length(g[g==2]), length(g[g==3])),
  main = "Dominant model (sample data)",
  xlab = "Genotype",
  ylab = "PK parameter"
)

# coefficient matrix
# c_1: additive, c_2: recessive, c_3: dominant
contrast &lt;- rbind(
  c(-1, 0, 1), c(-2, 1, 1), c(-1, -1, 2)
)
y &lt;- mcm.mvt(x, g, contrast)
y

## Example 2 ##
#  for dataframe
#  true response pattern:
#    pos = 1 dominant  model c=( 1,  1, -2)
#          2 additive  model c=(-1,  0,  1)
#          3 recessive model c=( 2, -1, -1)
set.seed(3872435)
x &lt;- c(
  rnorm(130, mean =  1 / 6, sd = 1),
  rnorm( 90, mean =  1 / 6, sd = 1),
  rnorm( 10, mean = -2 / 6, sd = 1),
  rnorm(130, mean = -1 / 4, sd = 1),
  rnorm( 90, mean =  0 / 4, sd = 1),
  rnorm( 10, mean =  1 / 4, sd = 1),
  rnorm(130, mean =  2 / 6, sd = 1),
  rnorm( 90, mean = -1 / 6, sd = 1),
  rnorm( 10, mean = -1 / 6, sd = 1)
)
g   &lt;- rep(rep(1:3, c(130, 90, 10)), 3)
pos &lt;- rep(c("rsXXXX", "rsYYYY", "rsZZZZ"), each=230)
xx  &lt;- data.frame(pos = pos, x = x, g = g)

# coefficient matrix
# c_1: additive, c_2: recessive, c_3: dominant
contrast &lt;- rbind(
  c(-1, 0, 1), c(-2, 1, 1), c(-1, -1, 2)
)
y &lt;- by(xx, xx$pos, function(x) mmcm.mvt(x$x, x$g,
  contrast))
y &lt;- do.call(rbind, y)[,c(3,7,9)]
# miss-detection!
y
</code></pre>

<hr>
<h2 id='mmcm.mvt'>The modified maximum contrast method by using randomized quasi-Monte Carlo method</h2><span id='topic+mmcm.mvt'></span>

<h3>Description</h3>

<p>This function gives <code class="reqn">P</code>-value for the modified maximum contrast statistics
by using randomized quasi-Monte Carlo method from
<code><a href="mvtnorm.html#topic+pmvt">pmvt</a></code> function of package <code>mvtnorm</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmcm.mvt(
  x,
  g,
  contrast,
  alternative = c("two.sided", "less", "greater"),
  algorithm = GenzBretz()
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mmcm.mvt_+3A_x">x</code></td>
<td>
<p>a numeric vector of data values</p>
</td></tr>
<tr><td><code id="mmcm.mvt_+3A_g">g</code></td>
<td>
<p>a integer vector giving the group for the corresponding elements of x</p>
</td></tr>
<tr><td><code id="mmcm.mvt_+3A_contrast">contrast</code></td>
<td>
<p>a numeric contrast coefficient matrix for modified maximum
contrast statistics</p>
</td></tr>
<tr><td><code id="mmcm.mvt_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the alternative hypothesis,
must be one of &quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;. You can specify
just the initial letter.</p>
</td></tr>
<tr><td><code id="mmcm.mvt_+3A_algorithm">algorithm</code></td>
<td>
<p>an object of class <code><a href="mvtnorm.html#topic+algorithms">GenzBretz</a></code>
defining the hyper parameters of this algorithm.</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code> performs the modified maximum contrast method
that is detecting a true response pattern under the unequal sample size situation.
</p>
<p><code class="reqn">Y_{ij} (i=1, 2, \ldots; j=1, 2, \ldots, n_i)</code> is an observed response for <code class="reqn">j</code>-th individual in
<code class="reqn">i</code>-th group.
</p>
<p><code class="reqn">\bm{C}</code> is coefficient matrix for modified maximum contrast statistics
(<code class="reqn">i \times k</code> matrix, <code class="reqn">i</code>: No. of groups, <code class="reqn">k</code>: No. of pattern).
</p>
<p style="text-align: center;"><code class="reqn">
  \bm{C}=(\bm{c}_1, \bm{c}_2, \ldots, \bm{c}_k)^{\rm{T}}
</code>
</p>

<p><code class="reqn">\bm{c}_k</code> is coefficient vector of <code class="reqn">k</code>th pattern.
</p>
<p style="text-align: center;"><code class="reqn">
  \bm{c}_k=(c_{k1}, c_{k2}, \ldots, c_{ki})^{\rm{T}} \qquad (\textstyle \sum_i c_{ki}=0)
</code>
</p>

<p><code class="reqn">S_{\max}</code> is the modified maximum contrast statistic.
</p>
<p style="text-align: center;"><code class="reqn">
  \bar{Y}_i=\frac{\sum_{j=1}^{n_i} Y_{ij}}{n_{i}},
  \bar{\bm{Y}}=(\bar{Y}_1, \bar{Y}_2, \ldots, \bar{Y}_i, \ldots, \bar{Y}_a)^{\rm{T}},
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  V=\frac{1}{\gamma}\sum_{j=1}^{n_i}\sum_{i=1}^{a} (Y_{ij}-\bar{Y}_i)^2,
  \gamma=\sum_{i=1}^{a} (n_i-1),
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  S_{k}=\frac{\bm{c}^t_k \bar{\bm{Y}}}{\sqrt{V \bm{c}^t_k \bm{c}_k}},
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  S_{\max}=\max(S_1, S_2, \ldots, S_k).
</code>
</p>

<p>Consider testing the overall null hypothesis <code class="reqn">H_0: \mu_1=\mu_2=\ldots=\mu_i</code>,
versus alternative hypotheses <code class="reqn">H_1</code> for response petterns
(<code class="reqn">H_1: \mu_1&lt;\mu_2&lt;\ldots&lt;\mu_i,~ \mu_1=\mu_2&lt;\ldots&lt;\mu_i,~
\mu_1&lt;\mu_2&lt;\ldots=\mu_i</code>). The <code class="reqn">P</code>-value for the probability distribution
of <code class="reqn">S_{\max}</code> under the overall null hypothesis is
</p>
<p style="text-align: center;"><code class="reqn">
  P\mbox{-value}=\Pr(S_{\max}&gt;s_{\max} \mid H_0)
</code>
</p>

<p><code class="reqn">s_{\max}</code> is observed value of statistics.
This function gives distribution of <code class="reqn">S_{\max}</code> by using randomized
quasi-Monte Carlo method from package <code>mvtnorm</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistic</code></td>
<td>
<p>the value of the test statistic with a name describing it.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value for the test.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>a character string describing the alternative hypothesis.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>the type of test applied.</p>
</td></tr>
<tr><td><code>contrast</code></td>
<td>
<p>a character string giving the names of the data.</p>
</td></tr>
<tr><td><code>contrast.index</code></td>
<td>
<p>a suffix of coefficient vector of the <code class="reqn">k</code>th pattern
that gives modified maximum contrast statistics (row number of the coefficient
matrix).</p>
</td></tr>
<tr><td><code>error</code></td>
<td>
<p>estimated absolute error and,</p>
</td></tr>
<tr><td><code>msg</code></td>
<td>
<p>status messages.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Nagashima, K., Sato, Y., Hamada, C. (2011).
A modified maximum contrast method for unequal sample sizes in
pharmacogenomic studies
<em>Stat Appl Genet Mol Biol.</em> <strong>10</strong>(1): Article 41.
<a href="http://dx.doi.org/10.2202/1544-6115.1560">http://dx.doi.org/10.2202/1544-6115.1560</a>
</p>
<p>Sato, Y., Laird, N.M., Nagashima, K., et al. (2009).
A new statistical screening approach for finding pharmacokinetics-related
genes in genome-wide studies.
<em>Pharmacogenomics J.</em> <strong>9</strong>(2): 137&ndash;146.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19104505">http://www.ncbi.nlm.nih.gov/pubmed/19104505</a>
</p>


<h3>See Also</h3>

<p><code><a href="mvtnorm.html#topic+pmvt">pmvt</a></code>,
<code><a href="mvtnorm.html#topic+algorithms">GenzBretz</a></code>,
<code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 ##
#  true response pattern: dominant model c=(1, 1, -2)
set.seed(136885)
x &lt;- c(
  rnorm(130, mean =  1 / 6, sd = 1),
  rnorm( 90, mean =  1 / 6, sd = 1),
  rnorm( 10, mean = -2 / 6, sd = 1)
)
g &lt;- rep(1:3, c(130, 90, 10))
boxplot(
  x ~ g,
  width = c(length(g[g==1]), length(g[g==2]), length(g[g==3])),
  main  = "Dominant model (sample data)",
  xlab  = "Genotype", ylab="PK parameter"
)

# coefficient matrix
# c_1: additive, c_2: recessive, c_3: dominant
contrast &lt;- rbind(
  c(-1, 0, 1), c(-2, 1, 1), c(-1, -1, 2)
)
y &lt;- mmcm.mvt(x, g, contrast)
y

## Example 2 ##
#  for dataframe
#  true response pattern:
#    pos = 1 dominant  model c=( 1,  1, -2)
#          2 additive  model c=(-1,  0,  1)
#          3 recessive model c=( 2, -1, -1)
set.seed(3872435)
x &lt;- c(
  rnorm(130, mean =  1 / 6, sd = 1),
  rnorm( 90, mean =  1 / 6, sd = 1),
  rnorm( 10, mean = -2 / 6, sd = 1),
  rnorm(130, mean = -1 / 4, sd = 1),
  rnorm( 90, mean =  0 / 4, sd = 1),
  rnorm( 10, mean =  1 / 4, sd = 1),
  rnorm(130, mean =  2 / 6, sd = 1),
  rnorm( 90, mean = -1 / 6, sd = 1),
  rnorm( 10, mean = -1 / 6, sd = 1)
)
g   &lt;- rep(rep(1:3, c(130, 90, 10)), 3)
pos &lt;- rep(c("rsXXXX", "rsYYYY", "rsZZZZ"), each=230)
xx  &lt;- data.frame(pos = pos, x = x, g = g)

# coefficient matrix
# c_1: additive, c_2: recessive, c_3: dominant
contrast &lt;- rbind(
  c(-1, 0, 1), c(-2, 1, 1), c(-1, -1, 2)
)
y &lt;- by(xx, xx$pos, function(x) mmcm.mvt(x$x, x$g,
  contrast))
y &lt;- do.call(rbind, y)[,c(3,7,9)]
y
</code></pre>

<hr>
<h2 id='mmcm.resamp'>The permuted modified maximum contrast method</h2><span id='topic+mmcm.resamp'></span>

<h3>Description</h3>

<p>This function gives <code class="reqn">P</code>-value for the permuted modified maximum
contrast method.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmcm.resamp(
  x,
  g,
  contrast,
  alternative = c("two.sided", "less", "greater"),
  nsample = 20000,
  abseps = 0.001,
  seed = NULL,
  nthread = 2
)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="mmcm.resamp_+3A_x">x</code></td>
<td>
<p>a numeric vector of data values</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_g">g</code></td>
<td>
<p>a integer vector giving the group for the corresponding elements of x</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_contrast">contrast</code></td>
<td>
<p>a numeric contrast coefficient matrix for permuted modified
maximum contrast statistics</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_alternative">alternative</code></td>
<td>
<p>a character string specifying the alternative hypothesis,
must be one of &quot;two.sided&quot; (default), &quot;greater&quot; or &quot;less&quot;.
You can specify just the initial letter.</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_nsample">nsample</code></td>
<td>
<p>specifies the number of resamples (default: 20000)</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_abseps">abseps</code></td>
<td>
<p>specifies the absolute error tolerance (default: 0.001)</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_seed">seed</code></td>
<td>
<p>a single value, interpreted as an integer;
see <code><a href="base.html#topic+Random">set.seed()</a></code> function. (default: NULL)</p>
</td></tr>
<tr><td><code id="mmcm.resamp_+3A_nthread">nthread</code></td>
<td>
<p>sthe number of threads used in parallel computing, or FALSE 
that means single threading (default: 2)</p>
</td></tr>
</table>


<h3>Details</h3>

<p><code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code> performs the permuted modified maximum contrast
method that is detecting a true response pattern under the unequal sample size
situation.
</p>
<p><code class="reqn">Y_{ij} (i=1, 2, \ldots; j=1, 2, \ldots, n_i)</code> is an observed response for <code class="reqn">j</code>-th individual in
<code class="reqn">i</code>-th group.
</p>
<p><code class="reqn">\bm{C}</code> is coefficient matrix for permuted modified maximum contrast
statistics (<code class="reqn">i \times k</code> matrix, <code class="reqn">i</code>: No. of groups, <code class="reqn">k</code>:
No. of pattern).
</p>
<p style="text-align: center;"><code class="reqn">
  \bm{C}=(\bm{c}_1, \bm{c}_2, \ldots, \bm{c}_k)^{\rm{T}}
</code>
</p>

<p><code class="reqn">\bm{c}_k</code> is coefficient vector of <code class="reqn">k</code>-th pattern.
</p>
<p style="text-align: center;"><code class="reqn">
  \bm{c}_k=(c_{k1}, c_{k2}, \ldots, c_{ki})^{\rm{T}} \qquad (\textstyle \sum_i c_{ki}=0)
</code>
</p>

<p><code class="reqn">M_{\max}</code> is a permuted modified maximum contrast statistic.
</p>
<p style="text-align: center;"><code class="reqn">
  \bar{Y}_i=\frac{\sum_{j=1}^{n_i} Y_{ij}}{n_{i}},
  \bar{\bm{Y}}=(\bar{Y}_1, \bar{Y}_2, \ldots, \bar{Y}_i, \ldots, \bar{Y}_a)^{\rm{T}},
  M_{k}=\frac{\bm{c}^t_k \bar{\bm{Y}}}{\sqrt{\bm{c}^t_k \bm{c}_k}},
</code>
</p>

<p style="text-align: center;"><code class="reqn">
  M_{\max}=\max(M_1, M_2, \ldots, M_k).
</code>
</p>

<p>Consider testing the overall null hypothesis <code class="reqn">H_0: \mu_1=\mu_2=\ldots=\mu_i</code>,
versus alternative hypotheses <code class="reqn">H_1</code> for response petterns 
(<code class="reqn">H_1: \mu_1&lt;\mu_2&lt;\ldots&lt;\mu_i, \mu_1=\mu_2&lt;\ldots&lt;\mu_i,
\mu_1&lt;\mu_2&lt;\ldots=\mu_i</code>).
The <code class="reqn">P</code>-value for the probability distribution of <code class="reqn">M_{\max}</code>
under the overall null hypothesis is
</p>
<p style="text-align: center;"><code class="reqn">
  P\mbox{-value}=\Pr(M_{\max}&gt;m_{\max} \mid H_0)
</code>
</p>

<p><code class="reqn">m_{\max}</code> is observed value of statistics.
This function gives distribution of <code class="reqn">M_{\max}</code> by using the
permutation method, follow algorithm:
</p>
<p>1. Initialize counting variable: <code class="reqn">COUNT = 0</code>.
Input parameters: <code class="reqn">NRESAMPMIN</code> (minimum resampling count,
we set 1000), <code class="reqn">NRESAMPMAX</code> (maximum resampling count), and
<code class="reqn">\epsilon</code> (absolute error tolerance).
</p>
<p>2. Calculate <code class="reqn">m_{\max}</code> that is the observed value of the test
statistic.
</p>
<p>3. Let <code class="reqn">y_{ij}^{(r)}</code> donate data, which are sampled without
replacement, and independently, form observed value <code class="reqn">y_{ij}</code>.
Where, <code class="reqn">(r)</code> is suffix of the resampling number
<code class="reqn">(r = 1,\, 2,\, \ldots)</code>.
</p>
<p>4. Calculate <code class="reqn">m^{(r)}_{\max}</code> from <code class="reqn">y_{ij}^{(r)}</code>.
If <code class="reqn">m^{(r)}_{\max} &gt; m_{\max}</code>, then increment the
counting variable: <code class="reqn">COUNT = COUNT + 1</code>. Calculate
approximate P-value <code class="reqn">\hat{p}^{(r)}=COUNT/r</code>, and
the simulation standard error <code class="reqn">SE(\hat{p}^{(r)})=\sqrt{\hat{p}^{(r)}(1-
\hat{p}^{(r)})/r}</code>.
</p>
<p>5. Repeat 3&ndash;4, while <code class="reqn">r &gt; 1000</code> and <code class="reqn">3.5SE(\hat{p}^{(r)})
&lt; \epsilon</code> (corresponding to 99% confidence
level), or <code class="reqn">NRESAMPMAX</code> times. Output the approximate P-value
<code class="reqn">\hat{p}^{(r)}</code>.
</p>


<h3>Value</h3>

<table role = "presentation">
<tr><td><code>statistic</code></td>
<td>
<p>the value of the test statistic with a name describing it.</p>
</td></tr>
<tr><td><code>p.value</code></td>
<td>
<p>the p-value for the test.</p>
</td></tr>
<tr><td><code>alternative</code></td>
<td>
<p>a character string describing the alternative hypothesis.</p>
</td></tr>
<tr><td><code>method</code></td>
<td>
<p>the type of test applied.</p>
</td></tr>
<tr><td><code>contrast</code></td>
<td>
<p>a character string giving the names of the data.</p>
</td></tr>
<tr><td><code>contrast.index</code></td>
<td>
<p>a suffix of coefficient vector of the <code class="reqn">k</code>th pattern
that gives permuted modified maximum contrast statistics (row number of the
coefficient matrix).</p>
</td></tr>
<tr><td><code>error</code></td>
<td>
<p>estimated absolute error and,</p>
</td></tr>
<tr><td><code>msg</code></td>
<td>
<p>status messages.</p>
</td></tr>
</table>


<h3>References</h3>

<p>Nagashima, K., Sato, Y., Hamada, C. (2011).
A modified maximum contrast method for unequal sample sizes in
pharmacogenomic studies
<em>Stat Appl Genet Mol Biol.</em> <strong>10</strong>(1): Article 41.
<a href="http://dx.doi.org/10.2202/1544-6115.1560">http://dx.doi.org/10.2202/1544-6115.1560</a>
</p>
<p>Sato, Y., Laird, N.M., Nagashima, K., et al. (2009).
A new statistical screening approach for finding pharmacokinetics-related
genes in genome-wide studies.
<em>Pharmacogenomics J.</em> <strong>9</strong>(2): 137&ndash;146.
<a href="http://www.ncbi.nlm.nih.gov/pubmed/19104505">http://www.ncbi.nlm.nih.gov/pubmed/19104505</a>
</p>


<h3>See Also</h3>

<p><code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Example 1 ##
#  true response pattern: dominant model c=(1, 1, -2)
set.seed(136885)
x &lt;- c(
  rnorm(130, mean =  1 / 6, sd = 1),
  rnorm( 90, mean =  1 / 6, sd = 1),
  rnorm( 10, mean = -2 / 6, sd = 1)
)
g &lt;- rep(1:3, c(130, 90, 10))
boxplot(
  x ~ g,
  width = c(length(g[g==1]), length(g[g==2]), length(g[g==3])),
  main  = "Dominant model (sample data)",
  xlab  = "Genotype", ylab="PK parameter"
)

# coefficient matrix
# c_1: additive, c_2: recessive, c_3: dominant
contrast &lt;- rbind(
  c(-1, 0, 1), c(-2, 1, 1), c(-1, -1, 2)
)
y &lt;- mmcm.resamp(x, g, contrast, nsample = 20000,
                 abseps = 0.01, seed = 5784324)
y

## Example 2 ##
#  for dataframe
#  true response pattern:
#    pos = 1 dominant  model c=( 1,  1, -2)
#          2 additive  model c=(-1,  0,  1)
#          3 recessive model c=( 2, -1, -1)
set.seed(3872435)
x &lt;- c(
  rnorm(130, mean =  1 / 6, sd = 1),
  rnorm( 90, mean =  1 / 6, sd = 1),
  rnorm( 10, mean = -2 / 6, sd = 1),
  rnorm(130, mean = -1 / 4, sd = 1),
  rnorm( 90, mean =  0 / 4, sd = 1),
  rnorm( 10, mean =  1 / 4, sd = 1),
  rnorm(130, mean =  2 / 6, sd = 1),
  rnorm( 90, mean = -1 / 6, sd = 1),
  rnorm( 10, mean = -1 / 6, sd = 1)
)
g   &lt;- rep(rep(1:3, c(130, 90, 10)), 3)
pos &lt;- rep(c("rsXXXX", "rsYYYY", "rsZZZZ"), each = 230)
xx  &lt;- data.frame(pos = pos, x = x, g = g)

# coefficient matrix
# c_1: additive, c_2: recessive, c_3: dominant
contrast &lt;- rbind(
  c(-1, 0, 1), c(-2, 1, 1), c(-1, -1, 2)
)

y &lt;- by(xx, xx$pos, function(x) mmcm.resamp(x$x, x$g,
  contrast, abseps = 0.02, nsample = 10000))
y &lt;- do.call(rbind, y)[,c(3,7,9)]
y
</code></pre>

<hr>
<h2 id='print.mmcm'>Print function for mmcm object</h2><span id='topic+print.mmcm'></span>

<h3>Description</h3>

<p>This function print result of function <code><a href="#topic+mcm.mvt">mcm.mvt</a></code>, <code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code> and <code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code>
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mmcm'
print(x, digits = getOption("digits"), ...)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="print.mmcm_+3A_x">x</code></td>
<td>
<p>Object of class <code>mmcm</code>, which is result of function <code><a href="#topic+mcm.mvt">mcm.mvt</a></code>, <code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code> and <code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code>.</p>
</td></tr>
<tr><td><code id="print.mmcm_+3A_digits">digits</code></td>
<td>
<p>a non-null value for digits specifies the minimum number of significant digits to be printed in values. The default, NULL, uses <code><a href="base.html#topic+options">getOption</a></code>(digits). (For the interpretation for complex numbers see <code><a href="base.html#topic+Round">signif</a></code>.) Non-integer values will be rounded down, and only values greater than or equal to 1 and no greater than 22 are accepted.</p>
</td></tr>
<tr><td><code id="print.mmcm_+3A_...">...</code></td>
<td>
<p>Further arguments passed to or from other methods.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The case where printed &quot;More than 2 contrast coefficient vectors were selected&quot;, some contrast may be unsuitable.
</p>


<h3>See Also</h3>

<p><code><a href="base.html#topic+print">print.default</a></code>,
<code><a href="#topic+mmcm.mvt">mmcm.mvt</a></code>,
<code><a href="#topic+mmcm.resamp">mmcm.resamp</a></code>,
<code><a href="#topic+mcm.mvt">mcm.mvt</a></code>
</p>

<hr>
<h2 id='reexports'>Objects exported from other packages</h2><span id='topic+reexports'></span><span id='topic+GenzBretz'></span>

<h3>Description</h3>

<p>These objects are imported from other packages. Follow the links
below to see their documentation.
</p>

<dl>
<dt>mvtnorm</dt><dd><p><code><a href="mvtnorm.html#topic+GenzBretz">GenzBretz</a></code></p>
</dd>
</dl>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
