<!DOCTYPE html><html><head><title>Help for package mmconvert</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mmconvert}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mmconvert-package'><p>mmconvert: Mouse Map Converter</p></a></li>
<li><a href='#coxmap'><p>Mouse genetic map based on Cox et al. <a href="https://doi.org/10.1534/genetics.109.105486">doi:10.1534/genetics.109.105486</a>,</p>
revised for mouse genome build 39.</a></li>
<li><a href='#cross2_to_grcm39'><p>Convert a cross2 object to use mouse build GRCm39</p></a></li>
<li><a href='#grcm39_chrlen'><p>Mouse chromosome lengths in basepairs for build GRCm39</p></a></li>
<li><a href='#mmconvert'><p>Convert mouse genome positions</p></a></li>
<li><a href='#MUGAmaps'><p>Array annotation information for the mouse MUGA arrays</p>
in mouse genome build 39.</a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Version:</td>
<td>0.10</td>
</tr>
<tr>
<td>Date:</td>
<td>2023-04-25</td>
</tr>
<tr>
<td>Title:</td>
<td>Mouse Map Converter</td>
</tr>
<tr>
<td>Description:</td>
<td>Convert mouse genome positions between the build 39 physical map and the genetic map of Cox et al. (2009) &lt;<a href="https://doi.org/10.1534%2Fgenetics.109.105486">doi:10.1534/genetics.109.105486</a>&gt;.</td>
</tr>
<tr>
<td>Author:</td>
<td>Karl W Broman <a href="https://orcid.org/0000-0002-4914-6671"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Karl W Broman &lt;broman@wisc.edu&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.5.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>Rcpp (&ge; 0.12.12)</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat, devtools, roxygen2, qtl2</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/rqtl/mmconvert">https://github.com/rqtl/mmconvert</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/rqtl/mmconvert/issues">https://github.com/rqtl/mmconvert/issues</a></td>
</tr>
<tr>
<td>LinkingTo:</td>
<td>Rcpp</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>ByteCompile:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>yes</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2023-04-25 11:46:48 UTC; kbroman</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2023-04-25 12:10:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='mmconvert-package'>mmconvert: Mouse Map Converter</h2><span id='topic+mmconvert-package'></span>

<h3>Description</h3>

<p>Function to convert mouse genome positions between the build 39
physical map and the Cox genetic map
<a href="https://doi.org/10.1534/genetics.109.105486">doi:10.1534/genetics.109.105486</a>.
</p>


<h3>Useful links</h3>


<ul>
<li><p> Source at <a href="https://github.com/rqtl/mmconvert">https://github.com/rqtl/mmconvert</a>
</p>
</li>
<li><p> BugReports to <a href="https://github.com/rqtl/mmconvert/issues">https://github.com/rqtl/mmconvert/issues</a>
</p>
</li></ul>


<hr>
<h2 id='coxmap'>Mouse genetic map based on Cox et al. <a href="https://doi.org/10.1534/genetics.109.105486">doi:10.1534/genetics.109.105486</a>,
revised for mouse genome build 39.</h2><span id='topic+coxmap'></span>

<h3>Description</h3>

<p>A data frame with rows being markers and six columns: marker name, chromosome,
sex-averaged cM position, female cM position, male cM position, and build 39 basepair position.
</p>


<h3>Details</h3>

<p>Genetic maps were re-estimated after reordering markers according to their position
in mouse genome build 39. See <a href="https://github.com/kbroman/CoxMapV3">https://github.com/kbroman/CoxMapV3</a>.
Markers were shifted so that 0 cM corresponds to 3 Mbp, using the chromosome- and
sex-specific recombination rate.
The maps are smoothed slightly to avoid intervals with 0 recombination.
</p>


<h3>Source</h3>

<p><a href="https://github.com/kbroman/CoxMapV3">https://github.com/kbroman/CoxMapV3</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(coxmap)
</code></pre>

<hr>
<h2 id='cross2_to_grcm39'>Convert a cross2 object to use mouse build GRCm39</h2><span id='topic+cross2_to_grcm39'></span>

<h3>Description</h3>

<p>Convert a cross2 object (with genotypes from one of the MUGA
arrays) to use mouse build GRCm39 and the revised Cox map
positions, revising marker order and omitting markers that are not
found.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>cross2_to_grcm39(cross, array = c("guess", "gm", "mm", "mini", "muga"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="cross2_to_grcm39_+3A_cross">cross</code></td>
<td>
<p>Object of class <code>"cross2"</code>, as produced by
<code><a href="qtl2.html#topic+read_cross2">qtl2::read_cross2()</a></code>. Must have markers from just one of the MUGA arrays.</p>
</td></tr>
<tr><td><code id="cross2_to_grcm39_+3A_array">array</code></td>
<td>
<p>Character string indicating which of the MUGA arrays
was used (&quot;gm&quot; for GigaMUGA, &quot;mm&quot; for MegaMUGA, &quot;mini&quot; for
MiniMUGA, or &quot;muga&quot; for the original MUGA), or &quot;guess&quot; (the
default) to pick the array with the most matching marker names.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>The input <code>cross</code> object with markers subset to those in build GRCm39
and with <code>pmap</code> and <code>gmap</code> replaced with the GRCm39 physical map and
revised Cox genetic map, respectively.
</p>


<h3>See Also</h3>

<p><a href="#topic+MUGAmaps">MUGAmaps</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
library(qtl2)
file &lt;- paste0("https://raw.githubusercontent.com/rqtl/",
               "qtl2data/main/DOex/DOex.zip")
DOex &lt;- read_cross2(file)
DOex_rev &lt;- cross2_to_grcm39(DOex)

</code></pre>

<hr>
<h2 id='grcm39_chrlen'>Mouse chromosome lengths in basepairs for build GRCm39</h2><span id='topic+grcm39_chrlen'></span>

<h3>Description</h3>

<p>A vector of mouse chromosome lengths in basepairs for mouse genome build GRCm39.
</p>


<h3>Source</h3>

<p><a href="https://www.ncbi.nlm.nih.gov/data-hub/genome/GCF_000001635.27/">https://www.ncbi.nlm.nih.gov/data-hub/genome/GCF_000001635.27/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(grcm39_chrlen)
</code></pre>

<hr>
<h2 id='mmconvert'>Convert mouse genome positions</h2><span id='topic+mmconvert'></span>

<h3>Description</h3>

<p>Convert mouse genome positions between the build 39 physical map and the Cox genetic map.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>mmconvert(
  positions,
  input_type = c("bp", "Mbp", "ave_cM", "female_cM", "male_cM")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="mmconvert_+3A_positions">positions</code></td>
<td>
<p>A set of positions, in one of three possible formats
</p>

<ul>
<li><p> a vector of character strings with like <code>"chr:position"</code>, with names being marker names,
e.g., <code>c(rs13482072="14:6738536", rs13482231="14:67215850", gnf14.117.278="14:121955310")</code>.
</p>
</li>
<li><p> a list of marker positions, each list being positions on a given chromosome,
e.g., <code>list("14"=c(rs13482072=6738536, rs13482231=67215850, gnf14.117.278=121955310))</code>.
</p>
</li>
<li><p> a data frame with columns chromosome, position, and marker,
e.g. <code>data.frame(chr=c(14,14,14), pos=c(6738536, 67215850, 121955310), marker=c("rs13482072", "rs13482231", "gnf14.117.278"))</code>.
</p>
</li></ul>
</td></tr>
<tr><td><code id="mmconvert_+3A_input_type">input_type</code></td>
<td>
<p>Character string indicating the type of positions provided (<code>"bp"</code>, <code>"Mbp"</code>,
<code>"ave_cM"</code>, <code>"female_cM"</code>, or <code>"male_cM"</code>)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We use linear interpolation using the Cox map positions in
the object <a href="#topic+coxmap">coxmap</a>. For positions outside the range of the
markers on the Cox map, we extrapolate using the overall
recombination rate.
</p>


<h3>Value</h3>

<p>A data frame with the interpolated positions, with seven columns: marker,
chromosome, sex-averaged cM, female cM, male cM, basepairs, and mega-basepairs.
The rows are sorted by genomic position.
</p>


<h3>See Also</h3>

<p><a href="#topic+coxmap">coxmap</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'># input as character strings like chr:position
input_char &lt;- c(rs13482072="14:6738536", rs13482231="14:67215850", gnf14.117.278="14:121955310")
mmconvert(input_char)

# input as list, as in the map object for R/qtl1 and R/qtl2
input_list &lt;- list("14"=c(rs13482072=6738536, rs13482231=67215850, gnf14.117.278=121955310))
mmconvert(input_list)

# input as data frame; *must* have chr as first column and position as second
# (marker names can be third column, or can be row names)
input_df &lt;- data.frame(chr=c(14,14,14),
                       pos=c(6738536, 67215850, 121955310),
                       marker=c("rs13482072", "rs13482231", "gnf14.117.278"))
mmconvert(input_df)

# input can also be in Mbp
input_df$pos &lt;- input_df$pos / 1e6
mmconvert(input_df, input_type="Mbp")
</code></pre>

<hr>
<h2 id='MUGAmaps'>Array annotation information for the mouse MUGA arrays
in mouse genome build 39.</h2><span id='topic+MUGAmaps'></span>

<h3>Description</h3>

<p>A list of four data frames with annotation information for the four MUGA arrays,
GigaMUGA (&quot;gm&quot;), MegaMUGA (&quot;mm&quot;), MiniMUGA (&quot;mini&quot;) and the original MUGA (&quot;muga&quot;).
Each has columns marker, chromosome, build 39 basepair position, and sex-averaged cM position (in Cox Map v3).
</p>


<h3>Details</h3>

<p>SNP probes for the MUGA arrays were blasted against mouse genome
build GRCm39 and locations interpolated using revised Cox maps.
See <a href="https://github.com/kbroman/MUGAarrays">https://github.com/kbroman/MUGAarrays</a> for the array
annotations and <a href="https://github.com/kbroman/CoxMapV3">https://github.com/kbroman/CoxMapV3</a> for the
genetic maps. Note that for the genetic map locations, markers were
shifted so that 0 cM corresponds to 3 Mbp, using the chromosome-
and sex-specific recombination rate. Moreover, the Cox map
positions were smoothed slightly to avoid regions with 0
recombination.
</p>


<h3>Source</h3>

<p><a href="https://github.com/kbroman/MUGAarrays">https://github.com/kbroman/MUGAarrays</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MUGAmaps)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
