<!DOCTYPE html><html><head><title>Help for package MMD</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MMD}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#MMD_attr'><p>Attribution of individuals to sources using the MMD method</p></a></li>
<li><a href='#MMD_Entropy'><p>Loci entropies to measure allele diversity</p></a></li>
<li><a href='#MMD_Rn'><p>Loci redundancy in sequences</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Minimal Multilocus Distance (MMD) for Source Attribution and
Loci Selection</td>
</tr>
<tr>
<td>Imports:</td>
<td>e1071, plyr, bigmemory</td>
</tr>
<tr>
<td>Version:</td>
<td>1.0.0</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Francisco Perez-Reche &lt;fperez-reche@abdn.ac.uk&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>The aim of the package is two-fold: (i) To implement 
    the MMD method for attribution of individuals to sources using 
    the Hamming distance between multilocus genotypes. (ii) To select 
    informative genetic markers based on information theory concepts 
    (entropy, mutual information and redundancy). The package implements the 
    functions introduced by Perez-Reche, F. J., Rotariu, O., Lopes, B. S., 
    Forbes, K. J. and Strachan, N. J. C. Mining whole genome sequence data to 
    efficiently attribute individuals to source populations. 
    Scientific Reports 10, 12124 (2020) &lt;<a href="https://doi.org/10.1038%2Fs41598-020-68740-6">doi:10.1038/s41598-020-68740-6</a>&gt;.
    See more details and examples in the README file.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>Suggests:</td>
<td>knitr, rmarkdown</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.1.0</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2021-01-23 10:50:41 UTC; s05fp2</td>
</tr>
<tr>
<td>Author:</td>
<td>Francisco Perez-Reche [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2021-01-26 13:40:08 UTC</td>
</tr>
</table>
<hr>
<h2 id='MMD_attr'>Attribution of individuals to sources using the MMD method</h2><span id='topic+MMD_attr'></span>

<h3>Description</h3>

<p>Attribution of individuals to sources using the MMD method
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMD_attr(
  datafile,
  popfile,
  NL,
  sourcenames,
  ToAttribute,
  SelfA = "no",
  fSelfA = 0.5,
  randomSelfA = "yes",
  quantile = 0.01,
  optq = "no",
  pqmin = 0,
  pqmax = 0.5,
  np = 20,
  Nbootstrap = 10000,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMD_attr_+3A_datafile">datafile</code></td>
<td>
<p>character; Name of the file *.csv (with full path in the file system) containing the genotypes (features) of individuals.</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_popfile">popfile</code></td>
<td>
<p>character; Name of the file *.pop (with full path in the file system) containing the genotypes (features) of individuals.</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_nl">NL</code></td>
<td>
<p>integer; number of loci. If larger than the number of available loci in the data set, NL is reduced to the maximum available number of loci.</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_sourcenames">sourcenames</code></td>
<td>
<p>a character vector listing the names of the sources.</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_toattribute">ToAttribute</code></td>
<td>
<p>character giving the name of the individuals of aknown origin (i.e. those that will be attributed to source).</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_selfa">SelfA</code></td>
<td>
<p>character; if &quot;no&quot; attribution of individuals to sources is made; if &quot;yes&quot;, self-attribution of selected individuals from sources is made. (Default &quot;no&quot;)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_fselfa">fSelfA</code></td>
<td>
<p>real number in the interval (0,1). When SelfA=&quot;yes&quot;, fSelfA specifies the fraction of individuals from the source specified by ToAttribute that will be assumed to be of unknown origin. (Default 0.1)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_randomselfa">randomSelfA</code></td>
<td>
<p>character only relevant if SelfA=&quot;yes&quot;. If &quot;yes&quot;, individuals to be considered as unknown are randomly selected from the source specified by ToAttribute; if &quot;no&quot; a list of names for individuals is read from filepoplist. (Default &quot;yes&quot;)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_quantile">quantile</code></td>
<td>
<p>real number with values in (0,1) giving the q-quantile for the MMD method. Only used if the quantile is not obatined through optimisation of the probability of correct self-attribution. (Default 0.01)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_optq">optq</code></td>
<td>
<p>character; if &quot;no&quot;, the specified quantile value is used; if &quot;yes&quot;, the q-quantile is optimised (only meaningful for self-attribution so optq=&quot;no&quot; automatically if SelfA=&quot;no&quot;). (Default &quot;no&quot;)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_pqmin">pqmin</code></td>
<td>
<p>real number with values in (0,1); minimum value of q-quantile when optq=&quot;yes&quot;. (Default 0)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_pqmax">pqmax</code></td>
<td>
<p>real number with values in (0,1); maximum value of q-quantile when optq=&quot;yes&quot;. (Default 0.5)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_np">np</code></td>
<td>
<p>integer giving the number of values of q-quantile in the interval (pqmin,pqmax) when optq=&quot;yes&quot;. (Default 20)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_nbootstrap">Nbootstrap</code></td>
<td>
<p>integer giving the number of samples used for bootstrapping to estimate the uncertainty of the attribution probability $p_s$ bootstrap. (Default 10000)</p>
</td></tr>
<tr><td><code id="MMD_attr_+3A_verbose">verbose</code></td>
<td>
<p>boolean (TRUE/FALSE) for the display of a progress bar  (Default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>If optq=&quot;yes&quot;, the output is a list with seven elements:
</p>

<ol>
<li><p> Number of individuals from unknown origin.
</p>
</li>
<li><p> Number of sources.
</p>
</li>
<li><p> Statistics of the attribution probability to sources, $p_s$.
</p>
</li>
<li><p> Probability of attribution of each unknown individual to each source $p_u,s$
</p>
</li>
<li><p> Runtime of the calculation.
</p>
</li>
<li><p> Number of loci.
</p>
</li>
<li><p> Parameter q used to calculate the q-quantile of the Hamming distance in the MMD method.
</p>
</li>
<li><p> Data frame giving the probability of correct attribution vs. q-quantile.
</p>
</li></ol>

<p>If optq=&quot;no&quot;, the output list contains all the items in the list above except the last one.
</p>


<h3>Author(s)</h3>

<p>Francisco J. Perez-Reche (Univeristy of Aberdeen)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## This example uses a small dataset stored in the MMD package
datafile &lt;- system.file("extdata", "Campylobacter_10SNP_HlW.csv", package = "MMD")
popfile &lt;- system.file("extdata", "Campylobacter_10SNP_HlW.pop", package = "MMD")

NL &lt;- 100
sourcenames &lt;- c("Cattle","Chicken","Pig","Sheep","WB")

##----- Source attribution
ToAttribute &lt;- "Human"
SelfA="no"
attribution &lt;- MMD_attr(datafile,popfile,NL,sourcenames,ToAttribute)

## See more detailed examples in the vignette.

</code></pre>

<hr>
<h2 id='MMD_Entropy'>Loci entropies to measure allele diversity</h2><span id='topic+MMD_Entropy'></span>

<h3>Description</h3>

<p>Loci entropies to measure allele diversity
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMD_Entropy(datafile, popfile, NL, sourcenames, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMD_Entropy_+3A_datafile">datafile</code></td>
<td>
<p>character; Name of the file *.csv (with full path in the file system) containing the genotypes (features) of individuals.</p>
</td></tr>
<tr><td><code id="MMD_Entropy_+3A_popfile">popfile</code></td>
<td>
<p>character; Name of the file *.pop (with full path in the file system) containing the genotypes (features) of individuals.</p>
</td></tr>
<tr><td><code id="MMD_Entropy_+3A_nl">NL</code></td>
<td>
<p>integer; number of loci. If larger than the number of available loci in the data set, NL is reduced to the maximum available number of loci.</p>
</td></tr>
<tr><td><code id="MMD_Entropy_+3A_sourcenames">sourcenames</code></td>
<td>
<p>a character vector listing the names of the sources.</p>
</td></tr>
<tr><td><code id="MMD_Entropy_+3A_verbose">verbose</code></td>
<td>
<p>boolean (TRUE/FALSE) for the display of a progress bar  (Default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with
</p>

<ol>
<li><p> Number of loci.
</p>
</li>
<li><p> Number of individuals in sources.
</p>
</li>
<li><p> Table with proportional weight of each population, qs.
</p>
</li>
<li><p> Number of alleles in the dataset.
</p>
</li>
<li><p> Value of the alleles in the dataset.
</p>
</li>
<li><p> Data frame with three columns for entropies: HlT, HlW, HlB
</p>
</li>
<li><p> Runtime.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Francisco J. Perez-Reche (Univeristy of Aberdeen)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## This example uses a small dataset stored in the MMD package
datafile &lt;- system.file("extdata", "Campylobacter_10SNP_HlW.csv", package = "MMD")
popfile &lt;- system.file("extdata", "Campylobacter_10SNP_HlW.pop", package = "MMD")

NL &lt;- 100
sourcenames &lt;- c("Cattle","Chicken","Pig","Sheep","WB")

EntropyLoci &lt;- MMD_Entropy(datafile,popfile,NL,sourcenames)

## See more detailed examples in the vignette.

</code></pre>

<hr>
<h2 id='MMD_Rn'>Loci redundancy in sequences</h2><span id='topic+MMD_Rn'></span>

<h3>Description</h3>

<p>Loci redundancy in sequences
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMD_Rn(datafile, popfile, NL, sourcenames, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMD_Rn_+3A_datafile">datafile</code></td>
<td>
<p>character; Name of the file *.csv (with full path in the file system) containing the genotypes (features) of individuals.</p>
</td></tr>
<tr><td><code id="MMD_Rn_+3A_popfile">popfile</code></td>
<td>
<p>character; Name of the file *.pop (with full path in the file system) containing the genotypes (features) of individuals.</p>
</td></tr>
<tr><td><code id="MMD_Rn_+3A_nl">NL</code></td>
<td>
<p>integer; number of loci. If larger than the number of available loci in the data set, NL is reduced to the maximum available number of loci.</p>
</td></tr>
<tr><td><code id="MMD_Rn_+3A_sourcenames">sourcenames</code></td>
<td>
<p>a character vector listing the names of the sources.</p>
</td></tr>
<tr><td><code id="MMD_Rn_+3A_verbose">verbose</code></td>
<td>
<p>boolean (TRUE/FALSE) for the display of a progress bar  (Default FALSE)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list with
</p>

<ol>
<li><p> Number of loci.
</p>
</li>
<li><p> Number of individuals in sources.
</p>
</li>
<li><p> Data frame with proportional weight of each population, qs.
</p>
</li>
<li><p> Number of alleles in the dataset.
</p>
</li>
<li><p> Value of the alleles in the dataset.
</p>
</li>
<li><p> Dataframe with two columns: (1) Index of locus. (2) Rn for loci in the original dataset.
</p>
</li>
<li><p> numerical; index of loci with increasing Rn.
</p>
</li>
<li><p> Dataframe with two columns: (1) Index of loci sorted by increasing Rn. (2) Value of Rn in increasing order.
</p>
</li>
<li><p> Runtime.
</p>
</li></ol>



<h3>Author(s)</h3>

<p>Francisco J. Perez-Reche (Univeristy of Aberdeen)
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## This example uses a small dataset stored in the MMD package
datafile &lt;- system.file("extdata", "Campylobacter_10SNP_HlW.csv", package = "MMD")
popfile &lt;- system.file("extdata", "Campylobacter_10SNP_HlW.pop", package = "MMD")

NL &lt;- 100
sourcenames &lt;- c("Cattle","Chicken","Pig","Sheep","WB")

RedundancyLoci &lt;- MMD_Rn(datafile,popfile,NL,sourcenames)

## See more detailed examples in the vignette.

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
