<!DOCTYPE html><html><head><title>Help for package MMDvariance</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MMDvariance}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#gsMMD.v'><p>Gene selection based on variances by using a mixture of marginal distributions</p></a></li>
<li><a href='#plotHistDensity.v'><p> Plot of histogram and density estimate of the pooled gene expression levels.</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Type:</td>
<td>Package</td>
</tr>
<tr>
<td>Title:</td>
<td>Detecting Differentially Variable Genes Using the Mixture of
Marginal Distributions</td>
</tr>
<tr>
<td>Version:</td>
<td>0.0.9</td>
</tr>
<tr>
<td>Date:</td>
<td>2018-07-22</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Weiliang Qiu &lt;weiliang.qiu@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 3.4.0), Biobase, lawstat</td>
</tr>
<tr>
<td>Imports:</td>
<td>MASS, graphics, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>ALL</td>
</tr>
<tr>
<td>biocViews:</td>
<td>Bioinformatics, DifferentialExpression</td>
</tr>
<tr>
<td>Description:</td>
<td>Gene selection based on variance using the marginal distributions of gene profiles that characterized by a mixture of three-component multivariate distributions. Please see the reference: Li X, Fu Y, Wang X, DeMeo DL, Tantisira K, Weiss ST, Qiu W. (2018) &lt;<a href="https://doi.org/10.1155%2F2018%2F6591634">doi:10.1155/2018/6591634</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-2">GPL-2</a> | <a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a> [expanded from: GPL (&ge; 2)]</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2018-07-22 18:10:43 UTC; weiliangqiu</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Author:</td>
<td>Xuan Li [aut, ctb],
  Yuejiao Fu [aut, ctb],
  Xiaogang Wang [aut, ctb],
  Dawn L. DeMeo [aut, ctb],
  Kelan Tantisira [aut, ctb],
  Scott T. Weiss [aut, ctb],
  Weiliang Qiu [aut, cre]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2018-07-27 21:10:14 UTC</td>
</tr>
</table>
<hr>
<h2 id='gsMMD.v'>Gene selection based on variances by using a mixture of marginal distributions</h2><span id='topic+gsMMD.v'></span>

<h3>Description</h3>

<p>Gene selection based on variances by using the marginal distributions of gene profiles that characterized by a mixture of three-component multivariate distributions. The goal is to detect gene probes having different variances between
cases and controls.
Input is an object derived from the class <code>ExpressionSet</code>. The function will obtain initial gene cluster membership by its own.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>gsMMD.v(obj.eSet, 
      memSubjects, 
      maxFlag = TRUE, 
      thrshPostProb = 0.5, 
      geneNames = NULL, 
      alpha = 0.05, 
      iniGeneMethod = "myLeveneTest", 
      transformFlag = FALSE, 
      transformMethod = "boxcox", 
      scaleFlag = TRUE, 
      criterion = c("cor", "skewness", "kurtosis"), 
      minL = -10, 
      maxL = 10, 
      stepL = 0.1, 
      eps = 0.001, 
      ITMAX = 100, 
      plotFlag = FALSE,
      quiet=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="gsMMD.v_+3A_obj.eset">obj.eSet</code></td>
<td>
<p>an object derived from the class <code>ExpressionSet</code> which contains the matrix of gene expression levels.  The rows of the matrix are genes. The columns of the matrix are subjects.</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_memsubjects">memSubjects</code></td>
<td>
<p>a vector of membership of subjects. <code>memSubjects[i]=1</code> means the <code class="reqn">i</code>-th subject belongs to diseased group, <code class="reqn">0</code> otherwise. 
</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_maxflag">maxFlag</code></td>
<td>
<p>logical. Indicate how to assign gene class membership. <code>maxFlag</code>=TRUE means that a gene will be assigned
to a class in which the posterior probability of the gene belongs to this class is maximum. <code>maxFlag</code>=FALSE means that a gene will be assigned to class 1 if the posterior probability of the gene belongs to class 1 is
greater than <code>thrshPostProb</code>. Similarly, a gene will be assigned to class 1 if the posterior probability of the gene belongs to class 1 is
greater than <code>thrshPostProb</code>. If  the posterior probability is less than <code>thrshPostProb</code>, the gene
will be assigned to class 2 (non-differentially variable gene group).</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_thrshpostprob">thrshPostProb</code></td>
<td>
<p>threshold for posterior probabilities. For example, if the posterior probability that a gene belongs to cluster 1 given its gene expression levels is larger than <code>thrshPostProb</code>, then this gene will be assigned to cluster 1.</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_genenames">geneNames</code></td>
<td>
<p>an optional character vector of gene names</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_alpha">alpha</code></td>
<td>
<p>significant level which is equal to <code>1-conf.level</code>, 
<code>conf.level</code> is the argument for the function <code>t.test</code>.
</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_inigenemethod">iniGeneMethod</code></td>
<td>
<p>method to get initial 3-cluster partition of genes: 
(1) genes having higher variance in cases than in controls;
(2) genes having equal variance between cases and controls;
(3) genes having lower variance in cases than in controls.
</p>
<p>Available methods are: 
&ldquo;myAWvar&rdquo;, 
&ldquo;myBFTest&rdquo;, 
&ldquo;myFTest&rdquo;, 
&ldquo;myLeveneTest&rdquo;, 
&ldquo;myLevene.TM&rdquo;, 
&ldquo;myiAWvar.BF&rdquo;,
&ldquo;myiAWvar.Levene&rdquo;, 
&ldquo;myiAWvar.TM&rdquo;,
&ldquo;myLeveneTest&rdquo;, 
&ldquo;myLeveneTest.TM&rdquo;. 
</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_transformflag">transformFlag</code></td>
<td>
<p>logical. Indicate if data transformation is needed</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_transformmethod">transformMethod</code></td>
<td>
<p>method for transforming data. Available methods include &quot;boxcox&quot;, &quot;log2&quot;, &quot;log10&quot;, &quot;log&quot;, &quot;none&quot;.</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_scaleflag">scaleFlag</code></td>
<td>
<p>logical. Indicate if gene profiles are to be scaled to have mean zero and variance one. If <code>transformFlag=TRUE</code> and <code>scaleFlag=TRUE</code>, then scaling is performed after transformation. To avoid linear dependence of tissue samples after scaling
gene profiles, we delete one tissue sample after scaling (c.f. details).</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_criterion">criterion</code></td>
<td>
<p>if <code>transformFlag=TRUE</code>, <code>criterion</code> indicates what criterion to determine if data looks like normal.  &ldquo;cor&rdquo; means using Pearson's correlation. The idea is that the observed quantiles after transformation should be close to theoretical normal quantiles. So we can use Pearson's correlation to check if the scatter plot of theoretical normal quantiles versus observed quantiles is a straightline.  &ldquo;skewness&rdquo; means using skewness measure to check if the distribution of the transformed data are close to normal distribution; &ldquo;kurtosis&rdquo; means using kurtosis measure to check normality.</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_minl">minL</code></td>
<td>
<p>lower limit for the <code>lambda</code> parameter used in Box-Cox transformation</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_maxl">maxL</code></td>
<td>
<p>upper limit for the <code>lambda</code> parameter used in Box-Cox transformation</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_stepl">stepL</code></td>
<td>
<p>step increase when searching the optimal <code>lambda</code> parameter used in Box-Cox transformation</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_eps">eps</code></td>
<td>
<p>a small positive value. If the absolute value of a value is smaller than <code>eps</code>, this value is regarded as zero.  </p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_itmax">ITMAX</code></td>
<td>
<p>maximum iteration allowed for iterations in the EM algorithm</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_plotflag">plotFlag</code></td>
<td>
<p>logical. Indicate if the Box-Cox normality plot should be output.</p>
</td></tr>
<tr><td><code id="gsMMD.v_+3A_quiet">quiet</code></td>
<td>
<p>logical. Indicate if intermediate results should be printed out.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>We assume that the distribution of gene expression profiles is 
a mixture of 3-component multivariate normal distributions 
<code class="reqn">\sum_{k=1}^{3} \pi_k f_k(x|\theta)</code>. Each component distribution <code class="reqn">f_k</code> 
corresponds to a gene cluster. The 3 components correspond to 3 gene clusters:
(1) genes having higher variance in cases than in controls;
(2) genes having equal variance between cases and controls;
(3) genes having lower variance in cases than in controls.
The model parameter vector is <code class="reqn">\theta=(\pi_1</code>, <code class="reqn">\pi_2</code>, 
<code class="reqn">\pi_3</code>, 
<code class="reqn">\sigma^2_{c1}</code>, <code class="reqn">\sigma^2_{n1}</code>, <code class="reqn">\mu_{c1}</code>, <code class="reqn">\rho_{c1}</code>, <code class="reqn">\mu_{n1}</code>, 
<code class="reqn">\rho_{n1}</code>,
<code class="reqn">\sigma^2_2</code>, <code class="reqn">\mu_{c2}</code>, <code class="reqn">\rho_{c2}</code>, <code class="reqn">\mu_{n2}</code>, <code class="reqn">\rho_{n2}</code>, 
<code class="reqn">\sigma^2_{c3}</code>, <code class="reqn">\sigma^2_{n3}</code>, 
<code class="reqn">\mu_{c3}</code>, 
<code class="reqn">\rho_{c3}</code>, <code class="reqn">\mu_{n3}</code>, <code class="reqn">\rho_{n3}</code>.
where <code class="reqn">\pi_1</code>, <code class="reqn">\pi_2</code>, and <code class="reqn">\pi_3</code> are the mixing proportions; 
<code class="reqn">\mu_{c1}</code>, <code class="reqn">\sigma^2_{c1}</code>, and <code class="reqn">\rho_{c1}</code> are 
the marginal mean, variance, and correlation of gene expression levels 
of cluster 1 (over-variable genes) for diseased subjects; 
<code class="reqn">\mu_{n1}</code>, <code class="reqn">\sigma^2_{n1}</code>, and <code class="reqn">\rho_{n1}</code> are 
the marginal mean, variance, and correlation of gene expression levels 
of cluster 1 (over-variable genes) for non-diseased subjects; 
<code class="reqn">\sigma^2_2</code>, <code class="reqn">\mu_{c2}</code>, <code class="reqn">\rho_{c2}</code>, <code class="reqn">\mu_{n2}</code>, and <code class="reqn">\rho_{n2}</code> are the marginal mean, 
variance, and correlation of gene 
expression levels of cluster 2 (equal-variable genes); 
<code class="reqn">\mu_{c3}</code>, <code class="reqn">\sigma^2_{c3}</code>, and <code class="reqn">\rho_{c3}</code> are 
the marginal mean, variance, and correlation of gene expression levels 
of cluster 3 (under-variable genes) for diseased subjects; 
<code class="reqn">\mu_{n3}</code>, <code class="reqn">\sigma^2_{n3}</code>, and <code class="reqn">\rho_{n3}</code> are 
the marginal mean, variance, and correlation of gene expression levels 
of cluster 3 (under-variable) for non-diseased subjects. 
</p>
<p>Note that genes in cluster 2 are non-differentially variable across
abnormal and normal tissue samples. Hence there are only 5 parameters for
cluster 2.
</p>
<p>To make sure the identifiability, we set the following contraints: 
<code class="reqn">\sigma_{c1}&gt;\sigma_{n1}</code> and <code class="reqn">\sigma_{c3}&lt;\sigma_{n3}</code>.
</p>
<p>To make sure the marginal covariance matrices are poisitive definite,
we set the following contraints: 
<code class="reqn">-1/(n_c-1)&lt;\rho_{c1}&lt;1</code>,
<code class="reqn">-1/(n_n-1)&lt;\rho_{n1}&lt;1</code>,
<code class="reqn">-1/(n-1)&lt;\rho_{2}&lt;1</code>,
<code class="reqn">-1/(n_c-1)&lt;\rho_{c3}&lt;1</code>,
<code class="reqn">-1/(n_n-1)&lt;\rho_{n3}&lt;1</code>.
</p>
<p>We also has the following constraints for the mixing proportion:
<code class="reqn">\pi_3=1-\pi_1-\pi_2</code>, <code class="reqn">\pi_k&gt;0</code>, <code class="reqn">k=1,2,3</code>.
</p>
<p>We apply the EM algorithm to estimate the model parameters. 
We regard the cluster membership of genes as missing values.
</p>
<p>To facilitate the estimation of the parameters,
we reparametrize the parameter vector as
<code class="reqn">\theta^*=(\pi_1</code>, <code class="reqn">\pi_2</code>, 
<code class="reqn">s^2_{c1}</code>, <code class="reqn">\delta_{n1}</code>, <code class="reqn">\mu_{c1}</code>, 
<code class="reqn">r_{c1}</code>, <code class="reqn">\mu_{n1}</code>, <code class="reqn">r_{n1}</code>,
<code class="reqn">s^2_2</code>, <code class="reqn">\mu_{c2}</code>, <code class="reqn">r_{c2}</code>, <code class="reqn">\mu_{n2}</code>, <code class="reqn">r_{n2}</code>, 
<code class="reqn">s^2_{c3}</code>, <code class="reqn">\delta_{n3}</code>, <code class="reqn">\mu_{c3}</code>, <code class="reqn">r_{c3}</code>, <code class="reqn">\mu_{n3}</code>, 
<code class="reqn">r_{n3})</code>,
where
<code class="reqn">\sigma_{n1}=\sigma_{c1}-\exp(\delta_{n1})</code>,
<code class="reqn">\sigma_{n3}=\sigma_{c3}+\exp(\delta_{n3})</code>,
<code class="reqn">\rho_{c1}=(\exp(r_{c1})-1/(n_c-1))/(1+\exp(r_{c1}))</code>,
<code class="reqn">\rho_{n1}=(\exp(r_{n1})-1/(n_n-1))/(1+\exp(r_{n1}))</code>,
<code class="reqn">\rho_{2}=(\exp(r_{2})-1/(n-1))/(1+\exp(r_{2}))</code>,
<code class="reqn">\rho_{c3}=(\exp(r_{c3})-1/(n_c-1))/(1+\exp(r_{c3}))</code>,
<code class="reqn">\rho_{n3}=(\exp(r_{n3})-1/(n_n-1))/(1+\exp(r_{n3}))</code>.
</p>
<p>Given a gene, the expression levels of the gene are assumed independent. However, after scaling, the scaled expression levels of the gene are no longer independent and the rank <code class="reqn">r^*=r-1</code> of the covariance matrix for the scaled gene profile will be one less than the rank <code class="reqn">r</code> for the un-scaled gene profile Hence the covariance matrix of the
gene profile will no longer be positive-definite. 
To avoid this problem,
we delete a tissue sample after scaling since its information has been incorrporated by other scaled tissue samples. We arbitrarily select the tissue sample, which has the biggest label number, from the tissue sample group that has larger size than the other tissue sample group. For example, if there are 6 cancer tissue samples and 10 normal tissue samples, we delete the 10-th normal tissue sample after scaling.
</p>


<h3>Value</h3>

<p>A list contains 18 elements.
</p>
<table>
<tr><td><code>dat</code></td>
<td>
<p>the (transformed) microarray data matrix. If tranformation
performed, then <code>dat</code> will be different from the input 
microarray data matrix.</p>
</td></tr>
<tr><td><code>memSubjects</code></td>
<td>
<p>the same as the input <code>memSubjects</code>.</p>
</td></tr>
<tr><td><code>memGenes</code></td>
<td>
<p>a vector of cluster membership of genes. <code class="reqn">1</code> means over-variable gene; <code class="reqn">2</code> means non-differentially variable gene; <code class="reqn">3</code> 
means under-variable gene.</p>
</td></tr>
<tr><td><code>memGenes2</code></td>
<td>
<p>an variant of the vector of cluster membership of genes. 
<code class="reqn">1</code> means differentially variable gene; <code class="reqn">0</code> means non-differentially variable gene.</p>
</td></tr>
<tr><td><code>para</code></td>
<td>
<p>parameter estimates (c.f. details).</p>
</td></tr>
<tr><td><code>llkh</code></td>
<td>
<p>value of the loglikelihood function.</p>
</td></tr>
<tr><td><code>wiMat</code></td>
<td>
<p>posterior probability that a gene belongs to a cluster given the expression levels of this gene. Column i is for cluster i.</p>
</td></tr>
<tr><td><code>wiArray</code></td>
<td>
<p>posterior probability matrix for different initial gene selection methods.</p>
</td></tr>
<tr><td><code>memIniMat</code></td>
<td>
<p>a matrix of initial cluster membership of genes.</p>
</td></tr>
<tr><td><code>paraIniMat</code></td>
<td>
<p>a matrix of parameter estimates based on initial gene cluster membership.</p>
</td></tr>
<tr><td><code>llkhIniVec</code></td>
<td>
<p>a vector of values of loglikelihood function.</p>
</td></tr>
<tr><td><code>memMat</code></td>
<td>
<p>a matrix of cluster membership of genes based on the mixture
of marginal models with initial parameter estimates obtained initial gene cluster membership.
</p>
</td></tr>
<tr><td><code>paraMat</code></td>
<td>
<p>a matrix of parameter estimates based on the mixture
of marginal models with initial parameter estimates obtained initial gene cluster membership.
</p>
</td></tr>
<tr><td><code>llkhVec</code></td>
<td>
<p>a vector of values of loglikelihood function based on the mixture of marginal models with initial parameter estimates obtained initial gene cluster membership.</p>
</td></tr>
<tr><td><code>lambda</code></td>
<td>
<p>the parameter used to do Box-Cox transformation</p>
</td></tr>
<tr><td><code>paraRP</code></td>
<td>
<p>parameter estimates for reparametrized parameter vector (c.f. details).</p>
</td></tr>
<tr><td><code>paraIniMatRP</code></td>
<td>
<p>a matrix of parameter estimates for reparametrized parameter vector based on initial gene cluster membership.</p>
</td></tr>
<tr><td><code>paraMatRP</code></td>
<td>
<p>a matrix of parameter estimates for reparametrized parameter vector based on the mixture
of marginal models with initial parameter estimates obtained initial gene cluster membership.</p>
</td></tr>
</table>


<h3>Note</h3>

<p>The speed of the program is slow for large data sets.
</p>


<h3>Author(s)</h3>

 
<p>Xuan Li <a href="mailto:lixuan0759@gmail.com">lixuan0759@gmail.com</a>,
Yuejiao Fu <a href="mailto:yuejiao@mathstat.yorku.ca">yuejiao@mathstat.yorku.ca</a>,
Xiaogang Wang <a href="mailto:stevenw@mathstat.yorku.ca">stevenw@mathstat.yorku.ca</a>,
Dawn L. DeMeo <a href="mailto:redld@channing.harvard.edu">redld@channing.harvard.edu</a>,
Kelan Tantisira <a href="mailto:rekgt@channing.harvard.edu">rekgt@channing.harvard.edu</a>,
Scott T. Weiss <a href="mailto:restw@channing.harvard.edu">restw@channing.harvard.edu</a>,
Weiliang Qiu <a href="mailto:weiliang.qiu@gmail.com">weiliang.qiu@gmail.com</a>
</p>


<h3>References</h3>

<p>Li X, Fu Y, Wang X, DeMeo DL, Tantisira K, Weiss ST, Qiu W.
Detecting Differentially Variable MicroRNAs via Model-Based Clustering.
<em>International Journal of Genomics</em>. Article ID 6591634, Volumne 2018 (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>      t1 = proc.time()
      library(ALL)
      data(ALL)
      eSet1 &lt;- ALL[1:50, ALL$BT == "B3" | ALL$BT == "T2"]
      
      mem.str &lt;- as.character(eSet1$BT)
      nSubjects &lt;- length(mem.str)
      memSubjects &lt;- rep(0,nSubjects)
      # B3 coded as 0, T2 coded as 1
      memSubjects[mem.str == "T2"] &lt;- 1
      
      obj.gsMMD.v &lt;- gsMMD.v(eSet1, memSubjects, transformFlag = FALSE, 
        transformMethod = "boxcox", scaleFlag = FALSE, 
        eps = 1.0e-1, ITMAX = 5, quiet = TRUE)
      print(round(obj.gsMMD.v$para, 3))
      t2=proc.time()-t1
      print(t2)
      
</code></pre>

<hr>
<h2 id='plotHistDensity.v'> Plot of histogram and density estimate of the pooled gene expression levels. </h2><span id='topic+plotHistDensity.v'></span>

<h3>Description</h3>

<p>Plot of histogram of pooled gene expression levels, composited with density estimate based on the mixture of marginal distributions. The density estimate is based on the assumption that the marginal correlations between subjects are zero.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>plotHistDensity.v(obj.gsMMD,
                plotFlag="case",
                plotComponent=FALSE,
                myxlab="expression level",
                myylab="density",
                mytitle="Histogram (case)",
                x.legend=NULL,
                y.legend=NULL,
                numPoints=500,
                mycol=1:4, 
                mylty=1:4, 
                mylwd=rep(3,4), 
                cex.main=2, 
                cex.lab=1.5, 
                cex.axis=1.5, 
                cex=2,
                bty="n")

</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="plotHistDensity.v_+3A_obj.gsmmd">obj.gsMMD</code></td>
<td>
<p>an object returned by <code>gsMMD.v</code>,
<code>gsMMD.default.v</code>, <code>gsMMD2.v</code>,
or <code>gsMMD2.default.v</code>
</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_plotflag">plotFlag</code></td>
<td>
<p>logical. Indicate the plot will based on which type of subjects.</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_plotcomponent">plotComponent</code></td>
<td>
<p>logical. Indicate if components of the mixture of marginal distribution will be plotted.</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_myxlab">myxlab</code></td>
<td>
<p>label for x-axis</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_myylab">myylab</code></td>
<td>
<p>label for y-axis</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_mytitle">mytitle</code></td>
<td>
<p>title of the plot</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_x.legend">x.legend</code></td>
<td>
<p>the x-corrdiates of the legend</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_y.legend">y.legend</code></td>
<td>
<p>the y-corrdiates of the legend</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_numpoints">numPoints</code></td>
<td>
<p>logical. Indicate how many genes will be plots.</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_mycol">mycol</code></td>
<td>
<p>color for the density estimates (overall and components)</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_mylty">mylty</code></td>
<td>
<p>line styles for the density estimates (overall and components)</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_mylwd">mylwd</code></td>
<td>
<p>line width for the density estimates (overall and components)</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_cex.main">cex.main</code></td>
<td>
<p>font for main title</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_cex.lab">cex.lab</code></td>
<td>
<p>font for x- and y-axis labels</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_cex.axis">cex.axis</code></td>
<td>
<p>font for x- and y-axis</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_cex">cex</code></td>
<td>
<p>font for texts</p>
</td></tr>
<tr><td><code id="plotHistDensity.v_+3A_bty">bty</code></td>
<td>
<p>the type of box to be drawn around the legend.  The allowed
values are '&quot;o&quot;' and '&quot;n&quot;' (the default).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For a given type of subjects, we pool their expression levels together if
the marginal correlations among subjects are zero. We then draw a histogram
of the pooled expression levels. Next, we composite density estimates of gene
expression levels for the overal distribution and the 3 component distributions.
</p>


<h3>Value</h3>

<p>A list containing coordinates of the density estimates:
</p>
<table>
<tr><td><code>x</code></td>
<td>
<p>sorted pooled gene expression levels for cases or controls.</p>
</td></tr>
<tr><td><code>x2</code></td>
<td>
<p>a subset of <code>x</code> specified by the sequence:
<code>seq(from=1,to=len.x, by=delta)</code>, where <code>len.x</code> is the
length of the vector <code>x</code>, and <code>delta=floor(len.x/numpoints).</code></p>
</td></tr>
<tr><td><code>y</code></td>
<td>
<p>density estimate corresponding to <code>x2</code></p>
</td></tr>
<tr><td><code>y1</code></td>
<td>
<p>weighted density estimate for gene cluster 1</p>
</td></tr>
<tr><td><code>y2</code></td>
<td>
<p>weighted density estimate for gene cluster 2</p>
</td></tr>
<tr><td><code>y3</code></td>
<td>
<p>weighted density estimate for gene cluster 3</p>
</td></tr>
</table>


<h3>Note</h3>

<p> The density estimate is obtained based on the
assumption that the marginal correlation among
subjects is zero. If the estimated marginal correlation obtained by <code>gsMMD.v</code> is far from zero, then do not use this plot function. 
</p>


<h3>Author(s)</h3>

 
<p>Xuan Li <a href="mailto:lixuan0759@gmail.com">lixuan0759@gmail.com</a>,
Yuejiao Fu <a href="mailto:yuejiao@mathstat.yorku.ca">yuejiao@mathstat.yorku.ca</a>,
Xiaogang Wang <a href="mailto:stevenw@mathstat.yorku.ca">stevenw@mathstat.yorku.ca</a>,
Dawn L. DeMeo <a href="mailto:redld@channing.harvard.edu">redld@channing.harvard.edu</a>,
Kelan Tantisira <a href="mailto:rekgt@channing.harvard.edu">rekgt@channing.harvard.edu</a>,
Scott T. Weiss <a href="mailto:restw@channing.harvard.edu">restw@channing.harvard.edu</a>,
Weiliang Qiu <a href="mailto:weiliang.qiu@gmail.com">weiliang.qiu@gmail.com</a>
</p>


<h3>References</h3>

<p>Li X, Fu Y, Wang X, DeMeo DL, Tantisira K, Weiss ST, Qiu W.
Detecting Differentially Variable MicroRNAs via Model-Based Clustering.
<em>International Journal of Genomics</em>. Article ID 6591634, Volumne 2018 (2018).
</p>


<h3>Examples</h3>

<pre><code class='language-R'>  
      t1 = proc.time()
      library(ALL)
      data(ALL)
      eSet1 &lt;- ALL[1:50, ALL$BT == "B3" | ALL$BT == "T2"]
      
      mem.str &lt;- as.character(eSet1$BT)
      nSubjects &lt;- length(mem.str)
      memSubjects &lt;- rep(0,nSubjects)
      # B3 coded as 0, T2 coded as 1
      memSubjects[mem.str == "T2"] &lt;- 1
      
      obj.gsMMD.v &lt;- gsMMD.v(eSet1, memSubjects, transformFlag = FALSE, 
        transformMethod = "boxcox", scaleFlag = FALSE, 
        eps = 1.0e-1, ITMAX = 5, quiet = TRUE)
      print(round(obj.gsMMD.v$para, 3))
     
  
    plotHistDensity.v(obj.gsMMD.v, plotFlag = "case", 
        mytitle = "Histogram (case)", 
        plotComponent = TRUE, 
        x.legend = c(0.8, 3), 
        y.legend = c(0.3, 0.4), 
        numPoints = 50)
    t2=proc.time()-t1
    print(t2)
  

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
