<!DOCTYPE html><html><head><title>Help for package MMINP</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {MMINP}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#checkInputdata'><p>Check if input data satisfies input conditions</p></a></li>
<li><a href='#compareFeatures'><p>Compare features' abundance obtained by prediction and measurement.</p></a></li>
<li><a href='#filterFeatures'><p>Filter features of input table according to prevalence and/or</p>
abundance</a></li>
<li><a href='#get_Components'><p>Estimate components for O2-PLS method</p></a></li>
<li><a href='#get_cvo2mComponent'><p>get components number from Cross-validate procedure of O2PLS</p></a></li>
<li><a href='#MMINP_trained_model'><p>(Data) A MMINP model</p></a></li>
<li><a href='#MMINP.predict'><p>Predict metabolites from new microbiome samples using MMINP model.</p></a></li>
<li><a href='#MMINP.preprocess'><p>Data Preprocessing function for MMINP</p></a></li>
<li><a href='#MMINP.train'><p>Train MMINP model using paired microbial features and metabolites data</p></a></li>
<li><a href='#print.mminp'><p>Print function for MMINP.train</p></a></li>
<li><a href='#test_metab'><p>(Data) Normalized metabolite abundances for MMINP prediction</p></a></li>
<li><a href='#test_metag'><p>(Data) Normalized gene family abundances for MMINP prediction</p></a></li>
<li><a href='#train_metab'><p>(Data) Normalized metabolite abundances for MMINP training</p></a></li>
<li><a href='#train_metag'><p>(Data) Normalized gene family abundances for MMINP training</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Microbe-Metabolite Interactions-Based Metabolic Profiles
Predictor</td>
</tr>
<tr>
<td>Version:</td>
<td>0.1.0</td>
</tr>
<tr>
<td>Description:</td>
<td>
     Implements a computational framework to predict microbial community-based 
     metabolic profiles with 'O2PLS' model. It provides procedures of model 
     training and prediction. Paired microbiome and metabolome data are needed 
     for modeling, and the trained model can be applied to predict metabolites 
     of analogous environments using new microbial feature abundances.</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>magrittr (&ge; 2.0.1), OmicsPLS (&ge; 2.0.2), utils, stats</td>
</tr>
<tr>
<td>Suggests:</td>
<td>rmarkdown, knitr, prettydoc, testthat (&ge; 3.0.0)</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://www.r-project.org/Licenses/GPL-3">GPL (&ge; 3.0)</a></td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/YuLab-SMU/MMINP">https://github.com/YuLab-SMU/MMINP</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/YuLab-SMU/MMINP/issues">https://github.com/YuLab-SMU/MMINP/issues</a></td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>LazyData:</td>
<td>true</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.0</td>
</tr>
<tr>
<td>Config/testthat/edition:</td>
<td>3</td>
</tr>
<tr>
<td>VignetteBuilder:</td>
<td>knitr</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2022-07-04 03:45:54 UTC; Dell</td>
</tr>
<tr>
<td>Author:</td>
<td>Wenli Tang <a href="https://orcid.org/0000-0002-6141-5854"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a> [aut,
    cre],
  Guangchuang Yu <a href="https://orcid.org/0000-0002-6485-8781"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, ths]</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Wenli Tang &lt;1071429394@qq.com&gt;</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2022-07-04 18:50:05 UTC</td>
</tr>
</table>
<hr>
<h2 id='checkInputdata'>Check if input data satisfies input conditions</h2><span id='topic+checkInputdata'></span>

<h3>Description</h3>

<p>This function throws an error if <code>x</code> is not a numeric matrix or a data
frame with all numeric-alike variables, or if any elements of <code>x</code> is
<code>NA</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>checkInputdata(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="checkInputdata_+3A_x">x</code></td>
<td>
<p>A matrix or data frame.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>No return value
</p>

<hr>
<h2 id='compareFeatures'>Compare features' abundance obtained by prediction and measurement.</h2><span id='topic+compareFeatures'></span>

<h3>Description</h3>

<p>Compare features' abundance obtained by prediction and measurement.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>compareFeatures(
  predicted,
  measured,
  method = "spearman",
  adjmethod = "fdr",
  rsignif = 0.3,
  psignif = 0.05
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="compareFeatures_+3A_predicted">predicted</code></td>
<td>
<p>A matrix or data frame. The feature table obtained by
prediction.</p>
</td></tr>
<tr><td><code id="compareFeatures_+3A_measured">measured</code></td>
<td>
<p>A matrix or data frame. The feature table obtained by
measurement. The abundances are expected to be normalized (i.e. proportion)
or be preprocessed by <code><a href="#topic+MMINP.preprocess">MMINP.preprocess</a></code>.</p>
</td></tr>
<tr><td><code id="compareFeatures_+3A_method">method</code></td>
<td>
<p>A character string indicating which correlation coefficient is
to be used for the <code><a href="stats.html#topic+cor.test">cor.test</a></code>. One of &quot;pearson&quot;,
&quot;kendall&quot;, or &quot;spearman&quot;, can be abbreviated.</p>
</td></tr>
<tr><td><code id="compareFeatures_+3A_adjmethod">adjmethod</code></td>
<td>
<p>A character string indicating correction
method (<code><a href="stats.html#topic+p.adjust">p.adjust</a></code>). One of &quot;holm&quot;, &quot;hochberg&quot;, &quot;hommel&quot;,
&quot;bonferroni&quot;, &quot;BH&quot;, &quot;BY&quot;, &quot;fdr&quot;, &quot;none&quot;, can be abbreviated.</p>
</td></tr>
<tr><td><code id="compareFeatures_+3A_rsignif">rsignif</code></td>
<td>
<p>A numeric ranging from 0 to 1, the minimum correlation
coefficient of features which considered as well-predicted features.</p>
</td></tr>
<tr><td><code id="compareFeatures_+3A_psignif">psignif</code></td>
<td>
<p>A numeric ranging from 0 to 1, the maximum adjusted p value of
features which considered as well-predicted features.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing a table of correlation results and a vector of
well-predicted features.
</p>

<hr>
<h2 id='filterFeatures'>Filter features of input table according to prevalence and/or
abundance</h2><span id='topic+filterFeatures'></span>

<h3>Description</h3>

<p>Filter features of input table according to prevalence and/or abundance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>filterFeatures(x, prev = NA, abund = NA)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="filterFeatures_+3A_x">x</code></td>
<td>
<p>A matrix or data frame.</p>
</td></tr>
<tr><td><code id="filterFeatures_+3A_prev">prev</code></td>
<td>
<p>A numeric ranging from 0 to 1, the minimum prevalence of features
to be retained. If set to NA, means no need to filter prevalence.</p>
</td></tr>
<tr><td><code id="filterFeatures_+3A_abund">abund</code></td>
<td>
<p>A numeric greater than 0, the minimum abundance (mean) of
features to be retained. If set to NA, means no need to filter abundance.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A filtered feature table will be returned.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train_metag)
d &lt;- filterFeatures(train_metag, prev = 0.8)
dim(train_metag)
dim(d)
</code></pre>

<hr>
<h2 id='get_Components'>Estimate components for O2-PLS method</h2><span id='topic+get_Components'></span>

<h3>Description</h3>

<p>get components number using Cross-validate procedure of O2-PLS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_Components(
  metag,
  metab,
  compmethod = NULL,
  n = 1:10,
  nx = 0:5,
  ny = 0:5,
  seed = 1234,
  nr_folds = 3,
  nr_cores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_Components_+3A_metag">metag</code></td>
<td>
<p>Training data of sequence features' relative abundances.
Must have the exact same rows (subjects/samples) as <code>metab</code>.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_metab">metab</code></td>
<td>
<p>Training data of metabolite relative abundances.
Must have the exact same rows (subjects/samples) as <code>metag</code>.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_compmethod">compmethod</code></td>
<td>
<p>A character string indicating which Cross-validate
procedure of O2PLS is to be used for estimating components, must be one of
&quot;NULL&quot;, &quot;cvo2m&quot; or &quot;cvo2m.adj&quot;. If set to &quot;NULL&quot;, depends on the features
number.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_n">n</code></td>
<td>
<p>Integer. Number of joint PLS components. Must be positive.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_nx">nx</code></td>
<td>
<p>Integer. Number of orthogonal components in <code>metag</code>. Negative
values are interpreted as 0.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_ny">ny</code></td>
<td>
<p>Integer. Number of orthogonal components in <code>metab</code>. Negative
values are interpreted as 0.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_seed">seed</code></td>
<td>
<p>a random seed to make the analysis reproducible, default is 1234.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_nr_folds">nr_folds</code></td>
<td>
<p>Positive integer. Number of folds to consider.
Note: <code>kcv=N</code> gives leave-one-out CV. Note that CV with less than two
folds does not make sense.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="get_Components_+3A_nr_cores">nr_cores</code></td>
<td>
<p>Positive integer. Number of cores to use for CV. You might
want to use <code><a href="parallel.html#topic+detectCores">detectCores</a>()</code>. Defaults to 1.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of components number
</p>

<hr>
<h2 id='get_cvo2mComponent'>get components number from Cross-validate procedure of O2PLS</h2><span id='topic+get_cvo2mComponent'></span>

<h3>Description</h3>

<p>get components number from Cross-validate procedure of O2PLS
</p>


<h3>Usage</h3>

<pre><code class='language-R'>get_cvo2mComponent(x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="get_cvo2mComponent_+3A_x">x</code></td>
<td>
<p>List of class &quot;cvo2m&quot;, produced by
<code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code>.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A data frame of components number
</p>

<hr>
<h2 id='MMINP_trained_model'>(Data) A MMINP model</h2><span id='topic+MMINP_trained_model'></span>

<h3>Description</h3>

<p>This model was built using (<code><a href="#topic+MMINP.train">MMINP.train</a></code>) with
preprocessed values in dataset <code class="reqn">train_metag</code> and <code class="reqn">train_metab</code>.
</p>


<h3>Format</h3>

<p>A list containing an 'o2m' model, results of correlation analysis
between metabolites of training data and its predicted values, components
number, re-estimate information and iteration number of modeling.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MMINP_trained_model)
</code></pre>

<hr>
<h2 id='MMINP.predict'>Predict metabolites from new microbiome samples using MMINP model.</h2><span id='topic+MMINP.predict'></span>

<h3>Description</h3>

<p>This function aims to predict potentially metabolites in new microbial
community using trained MMINP model.
If genes in model are not appear in newdata, then this procedure will fill
them up with 0.
Note that this function does not center or scale the new microbiome matrixs,
you would better do preprocessing on newdata in advance.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMINP.predict(model, newdata, minGeneSize = 0.5)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMINP.predict_+3A_model">model</code></td>
<td>
<p>List of class <code>"mminp"</code> or <code>"o2m"</code>, produced by
<code><a href="#topic+MMINP.train">MMINP.train</a></code> or <code><a href="OmicsPLS.html#topic+o2m">o2m</a></code>.</p>
</td></tr>
<tr><td><code id="MMINP.predict_+3A_newdata">newdata</code></td>
<td>
<p>New matrix of microbial genes, each column represents a gene.</p>
</td></tr>
<tr><td><code id="MMINP.predict_+3A_mingenesize">minGeneSize</code></td>
<td>
<p>A numeric between 0-1, minimal size of genes in model
contained in newdata.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The model must be class 'mminp' or 'o2m'.
The column of newdata must be microbial genes.
</p>


<h3>Value</h3>

<p>Predicted Data
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(MMINP_trained_model)
data(test_metag)
test_metag_preprocessed &lt;- MMINP.preprocess(test_metag, normalized = FALSE)
pred_metab &lt;- MMINP.predict(model = MMINP_trained_model$model,
newdata = test_metag_preprocessed)

</code></pre>

<hr>
<h2 id='MMINP.preprocess'>Data Preprocessing function for MMINP</h2><span id='topic+MMINP.preprocess'></span>

<h3>Description</h3>

<p>Before doing MMINP analysis, abundances of both microbial features and
metabolites should be preprocessed.
Both measurements are expected to be transformed to relative abundance (i.e.
proportion) and be log-transformed.
To meet the need of O2-PLS method, data must be scaled.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMINP.preprocess(
  data,
  normalized = TRUE,
  prev = NA,
  abund = NA,
  logtransformed = TRUE,
  scaled = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMINP.preprocess_+3A_data">data</code></td>
<td>
<p>A numeric matrix or data frame containing measurements of
metabolites or microbial features.</p>
</td></tr>
<tr><td><code id="MMINP.preprocess_+3A_normalized">normalized</code></td>
<td>
<p>Logical, whether to transform measurements into relative
abundance or not.</p>
</td></tr>
<tr><td><code id="MMINP.preprocess_+3A_prev">prev</code></td>
<td>
<p>A numeric ranging from 0 to 1, the minimum prevalence of features
to be retained. If set to NA, means no need to filter prevalence.</p>
</td></tr>
<tr><td><code id="MMINP.preprocess_+3A_abund">abund</code></td>
<td>
<p>A numeric greater than 0, the minimum abundance (mean) of
features to be retained. If set to NA, means no need to filter abundance.</p>
</td></tr>
<tr><td><code id="MMINP.preprocess_+3A_logtransformed">logtransformed</code></td>
<td>
<p>Logical, whether do log transformation or not.</p>
</td></tr>
<tr><td><code id="MMINP.preprocess_+3A_scaled">scaled</code></td>
<td>
<p>Logical, whether scale the columns of data or not.</p>
</td></tr>
</table>


<h3>Details</h3>

<p>The rows of data must be samples and columns of data must be metabolites or
microbial features.
The filtering process (<code>prev</code> and <code>abund</code>) is before log
transformation and scale transformation.
</p>


<h3>Value</h3>

<p>A preprocessed numeric matrix for analysis of MMINP.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train_metag)
d &lt;- MMINP.preprocess(train_metag)
d &lt;- MMINP.preprocess(train_metag, prev = 0.3, abund = 0.001)
d[1:5, 1:5]
</code></pre>

<hr>
<h2 id='MMINP.train'>Train MMINP model using paired microbial features and metabolites data</h2><span id='topic+MMINP.train'></span>

<h3>Description</h3>

<p>This function contains three steps.
Step1, Build an O2-PLS model and use it to predict metabolites profile;
Step2, Compare predicted and measured metabolites abundances, then filter
those metabolites which predicted poorly (i.e. metabolites of which
correlation coefficient less than <code>rsignif</code> or adjusted pvalue greater
than <code>psignif</code>.);
Step3, (iteration) Re-build O2-PLS model until all reserved metabolites are
well-fitted.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>MMINP.train(
  metag,
  metab,
  n = 1:3,
  nx = 0:3,
  ny = 0:3,
  seed = 1234,
  compmethod = NULL,
  nr_folds = 3,
  nr_cores = 1,
  rsignif = 0.4,
  psignif = 0.05,
  recomponent = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="MMINP.train_+3A_metag">metag</code></td>
<td>
<p>Training data of sequence features' relative abundances.
Must have the exact same rows (subjects/samples) as <code>metab</code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_metab">metab</code></td>
<td>
<p>Training data of metabolite relative abundances.
Must have the exact same rows (subjects/samples) as <code>metag</code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_n">n</code></td>
<td>
<p>Integer. Number of joint PLS components. Must be positive.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_nx">nx</code></td>
<td>
<p>Integer. Number of orthogonal components in <code>metag</code>. Negative
values are interpreted as 0.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_ny">ny</code></td>
<td>
<p>Integer. Number of orthogonal components in <code>metab</code>. Negative
values are interpreted as 0.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_seed">seed</code></td>
<td>
<p>a random seed to make the analysis reproducible, default is 1234.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_compmethod">compmethod</code></td>
<td>
<p>A character string indicating which Cross-validate
procedure of O2PLS is to be used for estimating components, must be one of
&quot;NULL&quot;, &quot;cvo2m&quot; or &quot;cvo2m.adj&quot;. If set to &quot;NULL&quot;, depends on the features
number.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_nr_folds">nr_folds</code></td>
<td>
<p>Positive integer. Number of folds to consider.
Note: <code>kcv=N</code> gives leave-one-out CV. Note that CV with less than two
folds does not make sense.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_nr_cores">nr_cores</code></td>
<td>
<p>Positive integer. Number of cores to use for CV. You might
want to use <code><a href="parallel.html#topic+detectCores">detectCores</a>()</code>. Defaults to 1.
More details in <code><a href="OmicsPLS.html#topic+crossval_o2m">crossval_o2m</a></code> and
<code><a href="OmicsPLS.html#topic+crossval_o2m_adjR2">crossval_o2m_adjR2</a></code>.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_rsignif">rsignif</code></td>
<td>
<p>A numeric ranging from 0 to 1, the minimum correlation
coefficient of features which considered as well-predicted features.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_psignif">psignif</code></td>
<td>
<p>A numeric ranging from 0 to 1, the maximum adjusted p value of
features which considered as well-predicted features.</p>
</td></tr>
<tr><td><code id="MMINP.train_+3A_recomponent">recomponent</code></td>
<td>
<p>Logical, whether re-estimate components or not during each
iteration.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A list containing
</p>
<table>
<tr><td><code>model</code></td>
<td>
<p>O2PLS model</p>
</td></tr>
<tr><td><code>trainres</code></td>
<td>
<p>Final correlation results between predicted and measured
metabolites of training samples</p>
</td></tr>
<tr><td><code>components</code></td>
<td>
<p>Components number. If <code>recomponent = TRUE</code>, the
components number is the result of last estimation.</p>
</td></tr>
<tr><td><code>re_estimate</code></td>
<td>
<p>Re-estimate information, i.e. whether re-estimate
components or not during each iteration</p>
</td></tr>
<tr><td><code>trainnumb</code></td>
<td>
<p>Iteration number</p>
</td></tr>
</table>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_metab)
data(test_metag)
a &lt;- MMINP.preprocess(test_metag[, 1:20], normalized = FALSE)
b &lt;- MMINP.preprocess(test_metab[, 1:20], normalized = FALSE)
mminp_model &lt;- MMINP.train(metag = a,
                           metab = b,
                           n = 3:5, nx = 0:3, ny = 0:3,
                           nr_folds = 2, nr_cores = 1)
length(mminp_model$trainres$wellPredicted)
</code></pre>

<hr>
<h2 id='print.mminp'>Print function for MMINP.train</h2><span id='topic+print.mminp'></span>

<h3>Description</h3>

<p>This function is the print method for <code>MMINP.train</code>.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>## S3 method for class 'mminp'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="print.mminp_+3A_x">x</code></td>
<td>
<p>A model (an object of class &quot;mminp&quot;)</p>
</td></tr>
<tr><td><code id="print.mminp_+3A_...">...</code></td>
<td>
<p>additional parameters</p>
</td></tr>
</table>


<h3>Value</h3>

<p>Brief information about the object.
</p>

<hr>
<h2 id='test_metab'>(Data) Normalized metabolite abundances for MMINP prediction</h2><span id='topic+test_metab'></span>

<h3>Description</h3>

<p>This datasets were built from NLIBD dataset (Franzosa et al., 2019) by
converting original HMDB IDs into KEGG compound IDs and removing
unassigned and repeated features.
</p>


<h3>Format</h3>

<p>A data frame of metabolite relative abundances (i.e. proportion),
with 65 subjects in rows and 130 KEGG compound IDs in columns.
</p>


<h3>References</h3>

<p>Franzosa EA et al. (2019). Gut microbiome structure and metabolic
activity in inflammatory bowel disease. Nature Microbiology 4(2):293-305.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_metab)
</code></pre>

<hr>
<h2 id='test_metag'>(Data) Normalized gene family abundances for MMINP prediction</h2><span id='topic+test_metag'></span>

<h3>Description</h3>

<p>This datasets were built from NLIBD dataset (Franzosa et al., 2019) by
converting original UniRef90 IDs into KEGG Orthology (KO) IDs and removing
unassigned and repeated features.
</p>


<h3>Format</h3>

<p>A data frame of gene family relative abundances (i.e. proportion),
with 65 subjects in rows and 629 KEGG Orthology (KO) IDs in columns.
</p>


<h3>References</h3>

<p>Franzosa EA et al. (2019). Gut microbiome structure and metabolic
activity in inflammatory bowel disease. Nature Microbiology 4(2):293-305.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(test_metag)
</code></pre>

<hr>
<h2 id='train_metab'>(Data) Normalized metabolite abundances for MMINP training</h2><span id='topic+train_metab'></span>

<h3>Description</h3>

<p>This datasets were built from PRISM dataset (Franzosa et al., 2019) by
converting original HMDB IDs into KEGG compound IDs and removing
unassigned and repeated features.
</p>


<h3>Format</h3>

<p>A data frame of metabolite relative abundances (i.e. proportion),
with 155 subjects in rows and 135 KEGG compound IDs in columns.
</p>


<h3>References</h3>

<p>Franzosa EA et al. (2019). Gut microbiome structure and metabolic
activity in inflammatory bowel disease. Nature Microbiology 4(2):293-305.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train_metab)
</code></pre>

<hr>
<h2 id='train_metag'>(Data) Normalized gene family abundances for MMINP training</h2><span id='topic+train_metag'></span>

<h3>Description</h3>

<p>This datasets were built from PRISM dataset (Franzosa et al., 2019) by
converting original UniRef90 IDs into KEGG Orthology (KO) IDs and removing
unassigned and repeated features.
</p>


<h3>Format</h3>

<p>A data frame of gene family relative abundances (i.e. proportion),
with 155 subjects in rows and 733 KEGG Orthology (KO) IDs in columns.
</p>


<h3>References</h3>

<p>Franzosa EA et al. (2019). Gut microbiome structure and metabolic
activity in inflammatory bowel disease. Nature Microbiology 4(2):293-305.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(train_metag)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
