<!DOCTYPE html><html lang="en"><head><title>Help for package mmod</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav class="package" aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {mmod}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#mmod'><p>Modern Measures of Differentiation</p></a></li>
<li><a href='#as.genind.DNAbin'><p>as.genind.DNAbin</p></a></li>
<li><a href='#chao_bootstrap'><p>Produce bootstrap samples from each subpopulation of a genind object</p></a></li>
<li><a href='#D_Jost'><p>Calculate Jost's D</p></a></li>
<li><a href='#diff_stats'><p>Calculate differentiation statistics for a genind object</p></a></li>
<li><a href='#diff_test'><p>An exact test of population differentiation for genind objects</p></a></li>
<li><a href='#dist.codom'><p>Calculate distance between individual for co-dominant locus</p></a></li>
<li><a href='#Gst_Hedrick'><p>Calculate Nei's Gst using estimators for Hs and Ht</p></a></li>
<li><a href='#Gst_Nei'><p>Calculate Nei's Gst using estimators for Hs and Ht</p></a></li>
<li><a href='#harmonic_mean'><p>Harmonic mean</p></a></li>
<li><a href='#jacknife_populations'><p>Create jacknife samples of a genind object by population</p></a></li>
<li><a href='#pairwise_D'><p>Calculates pairwise values of Jost's D</p></a></li>
<li><a href='#pairwise_Gst_Hedrick'><p>Calculates pairwise values of Hedrick's G'st</p></a></li>
<li><a href='#pairwise_Gst_Nei'><p>Calculates pairwise values of Nei's Gst</p></a></li>
<li><a href='#Phi_st_Meirmans'><p>Calculate Phi_st from a genind object</p></a></li>
<li><a href='#rgenotypes'><p>Randomly create genotypes</p></a></li>
<li><a href='#summarise_bootstrap'><p>Apply a differentiation statistic to a bootstrap sample</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table role='presentation'>
<tr>
<td>Version:</td>
<td>1.3.3</td>
</tr>
<tr>
<td>Date:</td>
<td>2017-06-04</td>
</tr>
<tr>
<td>Title:</td>
<td>Modern Measures of Population Differentiation</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>David Winter &lt;david.winter@gmail.com&gt;</td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 2.6.0), adegenet (&ge; 2.0),</td>
</tr>
<tr>
<td>Imports:</td>
<td>stats, pegas</td>
</tr>
<tr>
<td>Suggests:</td>
<td>testthat</td>
</tr>
<tr>
<td>ZipData:</td>
<td>no</td>
</tr>
<tr>
<td>Description:</td>
<td>Provides functions for measuring
    population divergence from genotypic data.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/dwinter/mmod">https://github.com/dwinter/mmod</a></td>
</tr>
<tr>
<td>BugReports:</td>
<td><a href="https://github.com/dwinter/mmod/issues">https://github.com/dwinter/mmod/issues</a></td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>6.0.1</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2017-04-06 00:24:28 UTC; david</td>
</tr>
<tr>
<td>Author:</td>
<td>David Winter [aut, cre],
  Peter Green [ctb],
  Zhian Kamvar [ctb],
  Thierry Gosselin [ctb]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2017-04-06 06:54:15 UTC</td>
</tr>
</table>
<hr>
<h2 id='mmod'>Modern Measures of Differentiation</h2><span id='topic+mmod'></span><span id='topic+mmod-package'></span>

<h3>Description</h3>

<p>Population geneticists have traditionally used Nei's Gst (often confusingly called 
Fst...) to measure divergence between populations. Recently, it has become 
clear that simple intereptations of the value of Gst can be misleading. For
this reason several new measures differntiation have been developed. mmod is
a package that brings some of these measures to R.
</p>


<h3>Details</h3>

<p>The vignette for this package ( avaliable using 
vignette(&quot;demo&quot;, package=&quot;mmod&quot;) from within R) contains an introduction to
these methods and and example usage for this package. I strongly suggest new
users start by reading this documentation.
</p>

<hr>
<h2 id='as.genind.DNAbin'>as.genind.DNAbin</h2><span id='topic+as.genind.DNAbin'></span>

<h3>Description</h3>

<p>Convert a DNAbin object into a genind object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>as.genind.DNAbin(x, pops)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="as.genind.DNAbin_+3A_x">x</code></td>
<td>
<p>object of class DNAbin</p>
</td></tr>
<tr><td><code id="as.genind.DNAbin_+3A_pops">pops</code></td>
<td>
<p>vector of population assignemnts for each sequence</p>
</td></tr>
</table>


<h3>Value</h3>

<p>genind
</p>


<h3>Examples</h3>

<pre><code class='language-R'>library(pegas)
data(woodmouse)
wm &lt;- as.genind.DNAbin(woodmouse, rep(c("A", "B", "C"), each=5))
diff_stats(wm)
</code></pre>

<hr>
<h2 id='chao_bootstrap'>Produce bootstrap samples from each subpopulation of a genind object</h2><span id='topic+chao_bootstrap'></span>

<h3>Description</h3>

<p>This function produces bootstrap samples from a genind object, with each
subpopulation resampled according to its size. Because there are many 
statistics that you may wish to calculte from these samples, this function
returns a list of genind objects representing bootsrap samples that can then
be futher processed (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>chao_bootstrap(x, nreps = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="chao_bootstrap_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="chao_bootstrap_+3A_nreps">nreps</code></td>
<td>
<p>numeric number of bootstrap replicates to perform (default 1000)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>You should note, this is a standard (frequentist) approach to quantifying
uncertainty - effectively asking &quot;if the population was exactly like our
sample, and we repeatedly took samples like this from it, how much would 
those samples vary?&quot; The confidence intervals don't include uncertainty 
produced from any biases in the way you collected your data. 
Additionally, this boostrapping procedure displays a slight upward bias for 
some datasets. If you plan or reporting a confidence interval for your 
statistic, it is  probably a good idea to subtract the difference between 
the point estimate of the statistic and the mean of the boostrap distribution 
from the extremes of the interval (as demonstrated in the expample below)
</p>


<h3>Value</h3>

<p>A list of genind objects
</p>


<h3>References</h3>

<p>Chao, A. et al. (2008). A Two-Stage probabilistic approach to Multiple-Community similarity indices. Biometrics, 64:1178-1186
</p>


<h3>See Also</h3>

<p>Other resample: <code><a href="#topic+jacknife_populations">jacknife_populations</a></code>,
<code><a href="#topic+summarise_bootstrap">summarise_bootstrap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:   
data(nancycats)
obs.D &lt;- D_Jost(nancycats)
bs &lt;- chao_bootstrap(nancycats)
bs_D &lt;- summarise_bootstrap(bs, D_Jost)
bias &lt;- bs.D$summary.global.het[1] - obs.D$global.het
bs.D$summary.global.het - bias

## End(Not run)
</code></pre>

<hr>
<h2 id='D_Jost'>Calculate Jost's D</h2><span id='topic+D_Jost'></span>

<h3>Description</h3>

<p>This function calculates Jost's D from a genind object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>D_Jost(x, hsht_mean = "arithmetic")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="D_Jost_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="D_Jost_+3A_hsht_mean">hsht_mean</code></td>
<td>
<p>The type of mean to use to calculate values of Hs and Ht
for a global estimate. (Default is teh airthmetic mean, can also be set to
the harmonic mean).</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a genind object with population information and calculates Jost's D
Returns a list with values for each locus as well as two global estimates.
'global.het' uses the averages of Hs and Ht across all loci while 
'global.harm_mean' takes the harmonic mean of all loci.
</p>
<p>Because estimators of Hs and Ht are used, its possible to have negative
estimates of D. You should treat these as numbers close to zero.
</p>


<h3>Value</h3>

<p>per.locus values for each D for each locus in the dataset
</p>
<p>global estimtes for D based on overall heterozygosity or the harmonic
mean of values for each locus
</p>


<h3>References</h3>

<p>Jost, L. (2008), GST and its relatives do not measure differentiation. Molecular Ecology, 17: 4015-4026.
</p>


<h3>See Also</h3>

<p>Other diffstat: <code><a href="#topic+Gst_Hedrick">Gst_Hedrick</a></code>,
<code><a href="#topic+Gst_Nei">Gst_Nei</a></code>, <code><a href="#topic+Phi_st_Meirmans">Phi_st_Meirmans</a></code>,
<code><a href="#topic+diff_stats">diff_stats</a></code>
</p>
<p>Other D: <code><a href="#topic+pairwise_D">pairwise_D</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
D_Jost(nancycats)
D_Jost(nancycats, hsht_mean= "arithmetic")
</code></pre>

<hr>
<h2 id='diff_stats'>Calculate differentiation statistics for a genind object</h2><span id='topic+diff_stats'></span>

<h3>Description</h3>

<p>By default this function calculates three different statistics of 
differentiation for a genetic dataset. Nei's Gst, Hedrick's G&rdquo;st and
Jost's D. Optionally, it can also calculate Phi'st, which is not calculated 
by default as it can take somewhat more time to run.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diff_stats(x, phi_st = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diff_stats_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="diff_stats_+3A_phi_st">phi_st</code></td>
<td>
<p>Boolean Calculate Phi_st (default is FALSE)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>See individual functions (listed below) for more details.
</p>


<h3>Value</h3>

<p>per.locus values for each statistic for each locus in the dataset
</p>
<p>global estimtes for these statistics across all loci in the dataset
</p>


<h3>References</h3>

<p>Hedrick, PW. (2005), A Standardized Genetic Differentiation Measure. Evolution 59: 1633-1638.
</p>
<p>Jost, L. (2008), GST and its relatives do not measure differentiation. Molecular Ecology, 17: 4015-4026.
</p>
<p>Meirmans PG, Hedrick PW (2011), Assessing population structure: FST and related measures. Molecular Ecology Resources, 11:5-18
</p>
<p>Nei M. (1973) Analysis of gene diversity in subdivided populations. PNAS: 3321-3323.
</p>
<p>Nei M, Chesser RK. (1983). Estimation of fixation indices and gene diversities. Annals of Human Genetics. 47: 253-259.
</p>
<p>Meirmans, PW. (2005), Using the AMOVA framework to estimate a standardized genetic differentiation measure. Evolution 60: 2399-402.
</p>
<p>Excoffier, L., Smouse, P., Quattro, J. (1992), Analysis of molecular variance inferred from metric distances among DNA haplotypes: application to human mitochondrial DNA restriction data. Genetics 131: 479-91
</p>


<h3>See Also</h3>

<p>Other diffstat: <code><a href="#topic+D_Jost">D_Jost</a></code>,
<code><a href="#topic+Gst_Hedrick">Gst_Hedrick</a></code>, <code><a href="#topic+Gst_Nei">Gst_Nei</a></code>,
<code><a href="#topic+Phi_st_Meirmans">Phi_st_Meirmans</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nancycats)
diff_stats(nancycats)
</code></pre>

<hr>
<h2 id='diff_test'>An exact test of population differentiation for genind objects</h2><span id='topic+diff_test'></span>

<h3>Description</h3>

<p>This function uses Fisher's exact test to determine if alleles in 
sub-populations are drawn randomly from a larger population (i.e. a 
significance test for allelic differentiation among sub-populations).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>diff_test(x, sim = TRUE, nreps = 2000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="diff_test_+3A_x">x</code></td>
<td>
<p>a genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="diff_test_+3A_sim">sim</code></td>
<td>
<p>boolean: if TRUE simulate p-value by using an MCMC sample of 
those tables that have the same marginal totals as the observed data 
(required for all but the smallest datasets)</p>
</td></tr>
<tr><td><code id="diff_test_+3A_nreps">nreps</code></td>
<td>
<p>number of steps used to simulate p-value (default 2000)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Note, this test returns p-values for each locus in a dataset _not_  
estimates of effect size. Since most populations have some degree of 
population differentiation, very large samples are almost guaranteed to 
return significant results. Refer to estimates of the various differentiation
statistics (D, G&rdquo;ST and Phi'ST)to ascertain  how meaningful such results 
might be.
</p>


<h3>Value</h3>

<p>named vector of p-values testing the null hypothesis these samples
where drawn from a panmictic population.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+fisher.test">fisher.test</a></code>, which this function wraps
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
diff_test(seploc(nancycats)[[2]], nreps=100)

</code></pre>

<hr>
<h2 id='dist.codom'>Calculate distance between individual for co-dominant locus</h2><span id='topic+dist.codom'></span>

<h3>Description</h3>

<p>This function calculates the distance between individuals in a genind
object based on their genotypes. 
Specifically, the simple metric of Kosman and Leonard (2005) in which 
distance is calculated as a propotion of shared alleles at each locus.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>dist.codom(x, matrix = TRUE, global = TRUE, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="dist.codom_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="dist.codom_+3A_matrix">matrix</code></td>
<td>
<p>boolean: if TRUE return matrix (dist object if FALSE)</p>
</td></tr>
<tr><td><code id="dist.codom_+3A_global">global</code></td>
<td>
<p>boolean: if TRUE, return a single global estimate based on all
loci. If FALSE return a list of matrices for each locus.
if FALSE</p>
</td></tr>
<tr><td><code id="dist.codom_+3A_na.rm">na.rm</code></td>
<td>
<p>boolean: if TRUE remove individuals with NAs</p>
</td></tr>
</table>


<h3>Value</h3>

<p>either a list of distance matrices, one for each locus or a single 
matrix containing the mean distance between individuals across all loci
</p>
<p>Dropped for each distance matrix and object of class &quot;na.action&quot; 
containing indices to those indivudals in the genind object which where 
omitted due to having NAs
</p>


<h3>References</h3>

<p>Kosman E., Leonard, K.J. Similarity coefficients for molecular markers in studies of genetic relationships between individuals for haploid
diploid, and polyploid species. Molecular Ecology. 14: 415-424
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nancycats)
dm &lt;- dist.codom(nancycats[40:45], matrix=FALSE)
head(dm)
</code></pre>

<hr>
<h2 id='Gst_Hedrick'>Calculate Nei's Gst using estimators for Hs and Ht</h2><span id='topic+Gst_Hedrick'></span>

<h3>Description</h3>

<p>This function calculates Hedrick's G'st from a genind object
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gst_Hedrick(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Gst_Hedrick_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Takes a genind object with population information and calculates Hedrick's 
G&rdquo;st.
</p>
<p>Because estimators of Hs and Ht are used, it's possible to have negative
estimates of G&rdquo;st. You should treat such results as zeros (or 
an attempt to estimate a very low number with some error which might push it
below zero)
</p>


<h3>Value</h3>

<p>per.locus values for each G&rdquo;st for each locus in the dataset
</p>
<p>global estimtes for G&rdquo;st based on overall heterozygosity
</p>


<h3>References</h3>

<p>Hedrick, PW. (2005), A Standardized Genetic Differentiation Measure. Evolution 59: 1633-1638.
</p>
<p>Meirmans PG, Hedrick PW (2011), Assessing population structure: FST and related measures. Molecular Ecology Resources, 11:5-18
</p>


<h3>See Also</h3>

<p>Other diffstat: <code><a href="#topic+D_Jost">D_Jost</a></code>,
<code><a href="#topic+Gst_Nei">Gst_Nei</a></code>, <code><a href="#topic+Phi_st_Meirmans">Phi_st_Meirmans</a></code>,
<code><a href="#topic+diff_stats">diff_stats</a></code>
</p>
<p>Other Hedrick: <code><a href="#topic+pairwise_Gst_Hedrick">pairwise_Gst_Hedrick</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nancycats) 
Gst_Hedrick(nancycats)
</code></pre>

<hr>
<h2 id='Gst_Nei'>Calculate Nei's Gst using estimators for Hs and Ht</h2><span id='topic+Gst_Nei'></span>

<h3>Description</h3>

<p>This function calculates Gst following Nei's method and using
Nei and Chesser's estimators for Hs and Ht
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Gst_Nei(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Gst_Nei_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>per.locus estimates of Gst for each locus in the dataset
</p>
<p>per.locus estimates of Gst for across all loci
</p>


<h3>References</h3>

<p>Nei M. (1973) Analysis of gene diversity in subdivided populations. PNAS: 3321-3323.
</p>
<p>Nei M, Chesser RK. (1983). Estimation of fixation indices and gene diversities. Annals of Human Genetics. 47: 253-259.
</p>


<h3>See Also</h3>

<p>Other diffstat: <code><a href="#topic+D_Jost">D_Jost</a></code>,
<code><a href="#topic+Gst_Hedrick">Gst_Hedrick</a></code>, <code><a href="#topic+Phi_st_Meirmans">Phi_st_Meirmans</a></code>,
<code><a href="#topic+diff_stats">diff_stats</a></code>
</p>
<p>Other Nei: <code><a href="#topic+pairwise_Gst_Nei">pairwise_Gst_Nei</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
Gst_Nei(nancycats)
</code></pre>

<hr>
<h2 id='harmonic_mean'>Harmonic mean</h2><span id='topic+harmonic_mean'></span>

<h3>Description</h3>

<p>Calculate the harmonic mean of a numeric vector
(will return NA if there are any negative numbers in the vector)
</p>


<h3>Usage</h3>

<pre><code class='language-R'>harmonic_mean(x, na.rm = TRUE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="harmonic_mean_+3A_x">x</code></td>
<td>
<p>numeric vector</p>
</td></tr>
<tr><td><code id="harmonic_mean_+3A_na.rm">na.rm</code></td>
<td>
<p>logical remove NAs prior or calculation</p>
</td></tr>
</table>


<h3>Value</h3>

<p>harmonic mean of vector
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
pop.sizes &lt;- table(pop(nancycats))
harmonic_mean(pop.sizes)
</code></pre>

<hr>
<h2 id='jacknife_populations'>Create jacknife samples of a genind object by population</h2><span id='topic+jacknife_populations'></span>

<h3>Description</h3>

<p>Makes a series of jacknife samples across populations from a genind object.
This function returns a list of genind objects that can then be further 
processed (see examples below).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>jacknife_populations(x, sample_frac = 0.5, nreps = 1000)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="jacknife_populations_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="jacknife_populations_+3A_sample_frac">sample_frac</code></td>
<td>
<p>fraction of pops to sample in each replication (default 0.5)</p>
</td></tr>
<tr><td><code id="jacknife_populations_+3A_nreps">nreps</code></td>
<td>
<p>number of jacknife replicates to run (default 1000)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>a list of genind objects to be further processed
</p>


<h3>See Also</h3>

<p>Other resample: <code><a href="#topic+chao_bootstrap">chao_bootstrap</a></code>,
<code><a href="#topic+summarise_bootstrap">summarise_bootstrap</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:   
data(nancycats)
obs &lt;- diff_stats(nancycats)
jn &lt;- jacknife_populations(nancycats)
jn.D &lt;- summarise_bootstrap(jn, D_Jost)

## End(Not run)
</code></pre>

<hr>
<h2 id='pairwise_D'>Calculates pairwise values of Jost's D</h2><span id='topic+pairwise_D'></span>

<h3>Description</h3>

<p>This function calculates Jost's D, a measure of genetic
differentiation, between all combinations of populaitons
in a genind object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwise_D(x, linearized = FALSE, hsht_mean = "arithmetic")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairwise_D_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="pairwise_D_+3A_linearized">linearized</code></td>
<td>
<p>logical, if TRUE will turned linearized D (1/1-D)</p>
</td></tr>
<tr><td><code id="pairwise_D_+3A_hsht_mean">hsht_mean</code></td>
<td>
<p>type of mean to use for the global estimates of Hs and Ht 
default it &quot;arithmetic&quot;, can also be set to &quot;harmonic&quot;.</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A distance matrix with between-population values of D
</p>


<h3>References</h3>

<p>Jost, L. (2008), GST and its relatives do not measure differentiation. Molecular Ecology, 17: 4015-4026.
</p>


<h3>See Also</h3>

<p>Other pairwise: <code><a href="#topic+pairwise_Gst_Hedrick">pairwise_Gst_Hedrick</a></code>,
<code><a href="#topic+pairwise_Gst_Nei">pairwise_Gst_Nei</a></code>
</p>
<p>Other D: <code><a href="#topic+D_Jost">D_Jost</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
pairwise_D(nancycats[1:26,])
</code></pre>

<hr>
<h2 id='pairwise_Gst_Hedrick'>Calculates pairwise values of Hedrick's G'st</h2><span id='topic+pairwise_Gst_Hedrick'></span>

<h3>Description</h3>

<p>This function calculates Hedrick's G'st, a measure of genetic
differentiation, between all combinations of populaitons
in a genind object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwise_Gst_Hedrick(x, linearized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairwise_Gst_Hedrick_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="pairwise_Gst_Hedrick_+3A_linearized">linearized</code></td>
<td>
<p>logical, if TRUE will turned linearized G'st (1/()1-G'st))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>A distance matrix with between-population values of G&rdquo;st
</p>


<h3>References</h3>

<p>Hedrick, PW. (2005), A Standardized Genetic Differentiation Measure. Evolution 59: 1633-1638.
</p>


<h3>See Also</h3>

<p>Other pairwise: <code><a href="#topic+pairwise_D">pairwise_D</a></code>,
<code><a href="#topic+pairwise_Gst_Nei">pairwise_Gst_Nei</a></code>
</p>
<p>Other Hedrick: <code><a href="#topic+Gst_Hedrick">Gst_Hedrick</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
pairwise_Gst_Hedrick(nancycats[1:26,])
</code></pre>

<hr>
<h2 id='pairwise_Gst_Nei'>Calculates pairwise values of Nei's Gst</h2><span id='topic+pairwise_Gst_Nei'></span>

<h3>Description</h3>

<p>This function calculates Nei's Gst, a measure of genetic
differentiation, between all combinations of populaitons
in a genind object.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>pairwise_Gst_Nei(x, linearized = FALSE)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="pairwise_Gst_Nei_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
<tr><td><code id="pairwise_Gst_Nei_+3A_linearized">linearized</code></td>
<td>
<p>logical, if TRUE will turned linearized Gst (1/(1-Gst))</p>
</td></tr>
</table>


<h3>Value</h3>

<p>dist A distance matrix with between-population values of Gst
</p>


<h3>References</h3>

<p>Nei M. (1973) Analysis of gene diversity in subdivided populations. PNAS: 3321-3323.
</p>
<p>Nei M, Chesser RK. (1983). Estimation of fixation indices and gene diversities. Annals of Human Genetics. 47: 253-259.
</p>


<h3>See Also</h3>

<p>Other pairwise: <code><a href="#topic+pairwise_D">pairwise_D</a></code>,
<code><a href="#topic+pairwise_Gst_Hedrick">pairwise_Gst_Hedrick</a></code>
</p>
<p>Other Nei: <code><a href="#topic+Gst_Nei">Gst_Nei</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
pairwise_Gst_Nei(nancycats[1:26,])
</code></pre>

<hr>
<h2 id='Phi_st_Meirmans'>Calculate Phi_st from a genind object</h2><span id='topic+Phi_st_Meirmans'></span>

<h3>Description</h3>

<p>This function calculates Meirmans' corrected version of Phi_st, an Fst 
analog produced using the AMOVA framework. Note, the global estimate produced
by this function is calculated as the mean distance between individuals
across all loci, and this exlcuded individuals with one or more missing 
value.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>Phi_st_Meirmans(x)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="Phi_st_Meirmans_+3A_x">x</code></td>
<td>
<p>genind object (from package adegenet)</p>
</td></tr>
</table>


<h3>Value</h3>

<p>per.locus Phi_st estimate for each locus
</p>
<p>global Phi_st estimate across all loci
</p>


<h3>References</h3>

<p>Meirmans, PW. (2005), Using the AMOVA framework to estimate a standardized genetic differentiation measure. Evolution 60: 2399-402.
</p>
<p>Excoffier, L., Smouse, P., Quattro, J. (1992), Analysis of molecular variance inferred from metric distances among DNA haplotypes: application to human mitochondrial DNA restriction data. Genetics 131: 479-91
</p>


<h3>See Also</h3>

<p>Other diffstat: <code><a href="#topic+D_Jost">D_Jost</a></code>,
<code><a href="#topic+Gst_Hedrick">Gst_Hedrick</a></code>, <code><a href="#topic+Gst_Nei">Gst_Nei</a></code>,
<code><a href="#topic+diff_stats">diff_stats</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>data(nancycats)
Phi_st_Meirmans(nancycats[1:26,])
</code></pre>

<hr>
<h2 id='rgenotypes'>Randomly create genotypes</h2><span id='topic+rgenotypes'></span>

<h3>Description</h3>

<p>Use the multinomial distribution to randomly create genotpes for individuals
for given allele frequences. By default this function returns a matrix of 
with alleles in rows and individuals in columns. There is an option to return
a genind object representing the same data (see examples).
</p>


<h3>Usage</h3>

<pre><code class='language-R'>rgenotypes(n, ploidy, probs, genind = FALSE, pop_name = "A",
  loc_name = "L1")
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="rgenotypes_+3A_n">n</code></td>
<td>
<p>integer number of indviduals.</p>
</td></tr>
<tr><td><code id="rgenotypes_+3A_ploidy">ploidy</code></td>
<td>
<p>integer number of alleles to asign to each individual.</p>
</td></tr>
<tr><td><code id="rgenotypes_+3A_probs">probs</code></td>
<td>
<p>vector of probabilies corresponding to allele frequences.</p>
</td></tr>
<tr><td><code id="rgenotypes_+3A_genind">genind</code></td>
<td>
<p>boolean if TRUE return a genind object</p>
</td></tr>
<tr><td><code id="rgenotypes_+3A_pop_name">pop_name</code></td>
<td>
<p>charcter Name for population defined in genind object
(not required if genind is not TRUE)</p>
</td></tr>
<tr><td><code id="rgenotypes_+3A_loc_name">loc_name</code></td>
<td>
<p>character name to five locus in genind object</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Used in <code><a href="#topic+chao_bootstrap">chao_bootstrap</a></code>, also exported as it may come in handy
for other simulations.
</p>


<h3>Value</h3>

<p>Either a matrix with individuals in columns, alleles in rows or, if
genind is TRUE a genind object for one population and locus.
</p>


<h3>See Also</h3>

<p><code><a href="stats.html#topic+rmultinom">rmultinom</a></code> which this function wraps.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
data(nancycats)
obs_allele_freqs &lt;- apply(nancycats$tab[,1:16], 2,mean, na.rm=TRUE)
rgenotypes(10, 2, obs_allele_freqs)
</code></pre>

<hr>
<h2 id='summarise_bootstrap'>Apply a differentiation statistic to a bootstrap sample</h2><span id='topic+summarise_bootstrap'></span>

<h3>Description</h3>

<p>This function applies a differentiation statistic (eg, D_Jost, Gst_Hedrick or 
Gst_Nei) to a list of genind objects, possibly produced with
chao_bootsrap or jacknife_populations.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>summarise_bootstrap(bs, statistic)
</code></pre>


<h3>Arguments</h3>

<table role = "presentation">
<tr><td><code id="summarise_bootstrap_+3A_bs">bs</code></td>
<td>
<p>list of genind objects</p>
</td></tr>
<tr><td><code id="summarise_bootstrap_+3A_statistic">statistic</code></td>
<td>
<p>differentiation statistic to apply (the function itself, 
as with apply family functions)</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Two different approaches are used for calculating confidence intervals in the
results. The estimates given by <code>lower.percentile</code> and <code>upper.percentile</code>
are simply the <code>2.5</code>th and <code>97.5</code>th precentile of the statistic
across bootstrap samples. Note, the presence or rare alleles in some
populations can bias bootstrapping procedures such that these intervals
are not centered on the observed value. The mean of statistic across
samples is returned as <code>mean.bs</code> and can be used to correct biased
bootsrap samples. Alternatively, <code>lower.normal</code> and <code>upper.normal</code>
form a confidence interval centered on the observed value of the statistic
and using the standard deviation of the statistic across replicates to
generate limits (sometimes called the normal-method of obtaining a confidence
interval). The print function for objects returned by this function displays 
the normal-method confidence intervals.
</p>


<h3>Value</h3>

<p>per.locus:  <code>matirx</code> of statistics calculated for each locus (column) and each 
bootstrap replicate (row).
</p>
<p>global.het: <code>vector</code> of global estimates calculated from overall 
heterozygosity
</p>
<p>global.het: <code>vector</code> of global estimates calculated from harmonic
mean of statistic (only applied to D_Jost)
</p>
<p>summary.loci: <code>data.frame</code> summarising the distribution of the
chosen statistic across replicates. Details of the different confidence
intervals are given in details
</p>
<p>summary.global_het: A vector containing the same measures as
<code>summary.loci</code> but for a global value of the statistic calculated from
all loci
</p>
<p>summary.global_harm: As with <code>summary.global_het</code> but calculated
from the harmonic mean of the statistic across loci (only applies to D_Jost)
</p>


<h3>See Also</h3>

<p>Other resample: <code><a href="#topic+chao_bootstrap">chao_bootstrap</a></code>,
<code><a href="#topic+jacknife_populations">jacknife_populations</a></code>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>## Not run:   
data(nancycats)
bs &lt;- chao_bootstrap(nancycats)
summarise_bootstrap(bs, D_Jost)

## End(Not run)
</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
