<!DOCTYPE html><html><head><title>Help for package modACDC</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
<link href="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/themes/prism.min.css" rel="stylesheet" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.css">
<script type="text/javascript">
const macros = { "\\R": "\\textsf{R}", "\\code": "\\texttt"};
function processMathHTML() {
    var l = document.getElementsByClassName('reqn');
    for (let e of l) { katex.render(e.textContent, e, { throwOnError: false, macros }); }
    return;
}</script>
<script defer src="https://cdn.jsdelivr.net/npm/katex@0.15.3/dist/katex.min.js"
    onload="processMathHTML();"></script>
<link rel="stylesheet" type="text/css" href="R-nav.css" />
</head><body><div class="container"><nav aria-label="Topic Navigation">
<div class="dropdown-menu">
<h1>Package {modACDC}</h1>
<h2>Contents</h2>
<ul class="menu">
<li><a href='#ACDC'><p>ACDC</p></a></li>
<li><a href='#ACDChighdim'><p>ACDChighdim</p></a></li>
<li><a href='#ACDCmod'><p>ACDCmod</p></a></li>
<li><a href='#coVar'><p>coVar</p></a></li>
<li><a href='#GCTA_par'><p>GCTA_par</p></a></li>
<li><a href='#GCTA_parPlot'><p>GCTA_parPlot</p></a></li>
<li><a href='#GCTA_singleValue'><p>GCTA_singleValue</p></a></li>
<li><a href='#OSCA_par'><p>OSCA_par</p></a></li>
<li><a href='#OSCA_parPlot'><p>OSCA_parPlot</p></a></li>
<li><a href='#OSCA_singleValue'><p>OSCA_singleValue</p></a></li>
</ul>
</div>
<hr>
</nav>
<main>
<table>
<tr>
<td>Title:</td>
<td>Association of Covariance for Detecting Differential
Co-Expression</td>
</tr>
<tr>
<td>Version:</td>
<td>2.0.1</td>
</tr>
<tr>
<td>Maintainer:</td>
<td>Katelyn Queen &lt;kjqueen@usc.edu&gt;</td>
</tr>
<tr>
<td>Description:</td>
<td>A series of functions to implement association of covariance for detecting differential co-expression (ACDC), a novel approach for detection of differential co-expression that simultaneously accommodates multiple phenotypes or exposures with binary, ordinal, or continuous data types. Users can use the default method which identifies modules by Partition or may supply their own modules. Also included are functions to choose an information loss criterion (ILC) for Partition using OmicS-data-based Complex trait Analysis (OSCA) and Genome-wide Complex trait Analysis (GCTA). The manuscript describing these methods is as follows: Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. "ACDC: a general approach for detecting phenotype or exposure associated co-expression" (2023) &lt;<a href="https://doi.org/10.3389%2Ffmed.2023.1118824">doi:10.3389/fmed.2023.1118824</a>&gt;.</td>
</tr>
<tr>
<td>License:</td>
<td><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file LICENSE</td>
</tr>
<tr>
<td>URL:</td>
<td><a href="https://github.com/USCbiostats/ACDC">https://github.com/USCbiostats/ACDC</a></td>
</tr>
<tr>
<td>Depends:</td>
<td>R (&ge; 4.1.0)</td>
</tr>
<tr>
<td>Imports:</td>
<td>CCP, data.table, doParallel, foreach, genieclust, genio,
ggplot2, partition, parallel, tibble, tidyr, tools, utils</td>
</tr>
<tr>
<td>Suggests:</td>
<td>CCA</td>
</tr>
<tr>
<td>Encoding:</td>
<td>UTF-8</td>
</tr>
<tr>
<td>RoxygenNote:</td>
<td>7.2.3</td>
</tr>
<tr>
<td>NeedsCompilation:</td>
<td>no</td>
</tr>
<tr>
<td>Packaged:</td>
<td>2024-01-30 21:47:26 UTC; katelynqueen</td>
</tr>
<tr>
<td>Author:</td>
<td>Katelyn Queen <a href="https://orcid.org/0000-0002-8070-3042"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cre, cph],
  Joshua Millstein <a href="https://orcid.org/0000-0001-7961-8943"><img alt="ORCID iD"src="https://cloud.R-project.org/web/orcid.svg" style="width:16px; height:16px; margin-left:4px; margin-right:4px; vertical-align:middle" /></a>
    [aut, cph]</td>
</tr>
<tr>
<td>Repository:</td>
<td>CRAN</td>
</tr>
<tr>
<td>Date/Publication:</td>
<td>2024-01-30 22:30:02 UTC</td>
</tr>
</table>
<hr>
<h2 id='ACDC'>ACDC</h2><span id='topic+ACDC'></span>

<h3>Description</h3>

<p>ACDC detects differential co-expression between a set of genes, such as a module of co-expressed genes, and a set of external features (exposures or responses) by using canonical correlation analysis (CCA) on the external features and module co-expression values. Modules are detected via Partition.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ACDC(
  fullData,
  ILC = 0.5,
  externalVar,
  identifierList = colnames(fullData),
  numNodes = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ACDC_+3A_fulldata">fullData</code></td>
<td>
<p>data frame or matrix with samples as rows, all features as columns; each entry should be numeric gene expression or other molecular data values</p>
</td></tr>
<tr><td><code id="ACDC_+3A_ilc">ILC</code></td>
<td>
<p>information loss criterion for Partition, or the minimum intra-class correlation required for features to be condensed; 0 <code class="reqn">\leq</code> ILC <code class="reqn">\leq</code> 1; default is 0.50</p>
</td></tr>
<tr><td><code id="ACDC_+3A_externalvar">externalVar</code></td>
<td>
<p>data frame, matrix, or vector containing external variable data to be used for CCA, rows are samples; all elements must be numeric</p>
</td></tr>
<tr><td><code id="ACDC_+3A_identifierlist">identifierList</code></td>
<td>
<p>optional row vector of identifiers, of the same length and order, corresponding to columns in fullData (ex: HUGO symbols for genes); default value is the column names from fullData</p>
</td></tr>
<tr><td><code id="ACDC_+3A_numnodes">numNodes</code></td>
<td>
<p>number of available compute nodes for parallelization; default is 1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Modules are identified by Partition, an agglomerative data reduction method which performs both feature condensation and extraction based on a user provided information loss criterion (ILC). Feature condensation into modules are only accepted if the intraclass correlation between the features is at least the ILC. For more information about how the co-expression features are calculated, see the coVar documentation.
</p>
<p>Following CCA, which determines linear combinations of the co-expression and external feature vectors that maximize the cross-covariance matrix for each module, a Wilks-Lambda test is performed to determine if the correlation between these linear combinations is significant. If they are significant, that implies there is differential co-expression. If there is only one co-expression value for a module (ie two features in the module) and a single external variable, CCA reduces to a simple correlation test, and the t-distribution is used to test for significant correlation (Widmann, 2005). If the number of co-expression features in a particular module is larger than the number of samples, CCA will return correlation coefficients of 1, and p-values and BH FDR q-values will not be calculated. See ACDChighdim for our solution.
</p>


<h3>Value</h3>

<p>Tibble, sorted by ascending BH FDR value, with columns
</p>

<dl>
<dt>moduleNum</dt><dd><p>module identifier</p>
</dd>
<dt>colNames</dt><dd><p>list of column names from fullData of the features in the module</p>
</dd>
<dt>features</dt><dd><p>list of identifiers from input parameter &quot;identifierList&quot; for all features in the module</p>
</dd>
<dt>CCA_corr</dt><dd><p>list of CCA canonical correlation coefficients</p>
</dd>
<dt>CCA_pval</dt><dd><p>Wilks-Lamda F-test p-value or t-test p-value</p>
</dd>
<dt>BHFDR_qval</dt><dd><p>Benjamini-Hochberg false discovery rate q-value</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Benjamini Y, Hochberg Y. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal statistical society: series B (Methodological)</em> <strong>57</strong> (1995) 289–300.
</p>
<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824..
</p>
<p>Widmann M. One-Dimensional CCA and SVD, and Their Relationship to Regression Maps. <em>Journal of Climate</em> <strong>18</strong> (2005) 2785–2792. doi:10.1175/jcli3424.1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run function for diet and genotype
ACDC(fullData = nutrimouse$lipid,
     ILC = 0.50, 
     externalVar = data.frame(diet=as.numeric(nutrimouse$diet), 
                              genotype=as.numeric(nutrimouse$genotype)))

</code></pre>

<hr>
<h2 id='ACDChighdim'>ACDChighdim</h2><span id='topic+ACDChighdim'></span>

<h3>Description</h3>

<p>ACDC detects differential co-expression between a set of genes, such as a module of co-expressed genes, and a set of external features (exposures or responses) by using canonical correlation analysis (CCA) on the external features and module co-expression values. A high-dimensional module is supplied by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ACDChighdim(
  moduleIdentifier = 1,
  moduleCols,
  fullData,
  externalVar,
  identifierList = colnames(fullData),
  corrThreshold = 0.75
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ACDChighdim_+3A_moduleidentifier">moduleIdentifier</code></td>
<td>
<p>the module identifier given by Partition or other dimension reduction/clustering algorithm; default is 1</p>
</td></tr>
<tr><td><code id="ACDChighdim_+3A_modulecols">moduleCols</code></td>
<td>
<p>list containing indices of column locations in fullData that specify features in the module</p>
</td></tr>
<tr><td><code id="ACDChighdim_+3A_fulldata">fullData</code></td>
<td>
<p>data frame or matrix with samples as rows, all features as columns; each entry should be numeric gene expression or other molecular data values</p>
</td></tr>
<tr><td><code id="ACDChighdim_+3A_externalvar">externalVar</code></td>
<td>
<p>data frame, matrix, or vector containing external variable data to be used for CCA, rows are samples; all elements must be numeric</p>
</td></tr>
<tr><td><code id="ACDChighdim_+3A_identifierlist">identifierList</code></td>
<td>
<p>optional row vector of identifiers, of the same length and order, corresponding to columns in fullData (ex: HUGO symbols for genes); default value is the column names from fullData</p>
</td></tr>
<tr><td><code id="ACDChighdim_+3A_corrthreshold">corrThreshold</code></td>
<td>
<p>minimum correlation required between two features to be kept in the dataset; 0 <code class="reqn">\leq</code> corrThreshold <code class="reqn">\leq</code> 1; default value is 0.75</p>
</td></tr>
</table>


<h3>Details</h3>

<p>If the number of co-expression features in a particular module is larger than the number of samples, CCA will return correlation coefficients of 1, and p-values and BH FDR q-values will not be calculated. This function accepts one of these high dimension modules and reduces the dimensionality by calculating the pairwise correlations for all features and only keeping feature pairs with |correlation| &gt; the user defined corrThreshold with a maximum number of features pairs of <code class="reqn">\lfloor\frac{N}{2}\rfloor</code>. We posit that these highly correlated pairs are the skeleton structure of the full module and therefore an appropriate approximation. Once this structure is identified, co-expression values are calculated and CCA is performed as in ACDC.
</p>
<p>For more information about how the co-expression features are calculated, see the coVar documentation.
</p>
<p>Following CCA, which determines linear combinations of the co-expression and external feature vectors that maximize the cross-covariance matrix for each module, a Wilks-Lambda test is performed to determine if the correlation between these linear combinations is significant. If they are significant, that implies there is differential co-expression. If there is only one co-expression value for a module (ie two features in the module) and a single external variable, CCA reduces to a simple correlation test, and the t-distribution is used to test for significant correlation (Widmann, 2005).
</p>


<h3>Value</h3>

<p>Tibble, designed to be row binded with output from other ACDC functions after removing the final column, with columns
</p>

<dl>
<dt>moduleNum</dt><dd><p>module identifier</p>
</dd>
<dt>colNames</dt><dd><p>list of column names from fullData of the features in the module</p>
</dd>
<dt>features</dt><dd><p>list of identifiers from input parameter &quot;identifierList&quot; for all features in the module</p>
</dd>
<dt>CCA_corr</dt><dd><p>list of CCA canonical correlation coefficients</p>
</dd>
<dt>CCA_pval</dt><dd><p>Wilks-Lamda F-test p-value or t-test p-value</p>
</dd>
<dt>numPairsUsed</dt><dd><p>number of feature pairs with correlation above corrThreshold</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Benjamini Y, Hochberg Y. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal statistical society: series B (Methodological)</em> <strong>57</strong> (1995) 289–300.
</p>
<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824..
</p>
<p>Widmann M. One-Dimensional CCA and SVD, and Their Relationship to Regression Maps. <em>Journal of Climate</em> <strong>18</strong> (2005) 2785–2792. doi:10.1175/jcli3424.1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run function for diet and genotype
ACDChighdim(moduleIdentifier = 1,
            moduleCols = list(1:ncol(nutrimouse$lipid)),
            fullData = nutrimouse$lipid,
            externalVar = data.frame(diet=as.numeric(nutrimouse$diet), 
                                     genotype=as.numeric(nutrimouse$genotype)))

</code></pre>

<hr>
<h2 id='ACDCmod'>ACDCmod</h2><span id='topic+ACDCmod'></span>

<h3>Description</h3>

<p>ACDCmod detects differential co-expression between a set of genes, such as a module of co-expressed genes, and a set of external features (exposures or responses) by using canonical correlation analysis (CCA) on the external features and module co-expression values. Modules are provided by the user.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>ACDCmod(
  fullData,
  modules,
  externalVar,
  identifierList = colnames(fullData),
  numNodes = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="ACDCmod_+3A_fulldata">fullData</code></td>
<td>
<p>data frame or matrix with samples as rows, all probes as columns; each entry should be numeric gene expression or other molecular data values</p>
</td></tr>
<tr><td><code id="ACDCmod_+3A_modules">modules</code></td>
<td>
<p>vector of lists where each list contains indices of column locations in fullData that specify features in each module</p>
</td></tr>
<tr><td><code id="ACDCmod_+3A_externalvar">externalVar</code></td>
<td>
<p>data frame, matrix, or vector  containing external variable data to be used for CCA, rows are samples; all elements must be numeric</p>
</td></tr>
<tr><td><code id="ACDCmod_+3A_identifierlist">identifierList</code></td>
<td>
<p>optional row vector of identifiers, of the same length and order, corresponding to columns in fullData (ex: HUGO symbols for genes); default value is the column names from fullData</p>
</td></tr>
<tr><td><code id="ACDCmod_+3A_numnodes">numNodes</code></td>
<td>
<p>number of available compute nodes for parallelization; default is 1</p>
</td></tr>
</table>


<h3>Details</h3>

<p>For more information about how the co-expression features are calculated, see the coVar documentation.
</p>
<p>Following CCA, which determines linear combinations of the co-expression and external feature vectors that maximize the cross-covariance matrix for each module, a Wilks-Lambda test is performed to determine if the correlation between these linear combinations is significant. If they are significant, that implies there is differential co-expression. If there is only one co-expression value for a module (ie two features in the module) and a single external variable, CCA reduces to a simple correlation test, and the t-distribution is used to test for significant correlation (Widmann, 2005). If the number of co-expression features in a particular module is larger than the number of samples, CCA will return correlation coefficients of 1, and p-values and BH FDR q-values will not be calculated. See ACDChighdim for our solution.
</p>


<h3>Value</h3>

<p>Tibble, sorted by ascending BH FDR value, with columns
</p>

<dl>
<dt>moduleNum</dt><dd><p>module identifier</p>
</dd>
<dt>colNames</dt><dd><p>list of column names from fullData of the features in the module</p>
</dd>
<dt>features</dt><dd><p>list of identifiers from input parameter &quot;identifierList&quot; for all features in the module</p>
</dd>
<dt>CCA_corr</dt><dd><p>list of CCA canonical correlation coefficients</p>
</dd>
<dt>CCA_pval</dt><dd><p>Wilks-Lamda F-test p-value; t-test p-value if there are only 2 features in the module and a single external variable</p>
</dd>
<dt>BHFDR_qval</dt><dd><p>Benjamini-Hochberg false discovery rate q-value</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Benjamini Y, Hochberg Y. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal statistical society: series B (Methodological)</em> <strong>57</strong> (1995) 289–300.
</p>
<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824.
</p>
<p>Widmann M. One-Dimensional CCA and SVD, and Their Relationship to Regression Maps. <em>Journal of Climate</em> <strong>18</strong> (2005) 2785–2792. doi:10.1175/jcli3424.1.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# partition dataset and save modules
library(partition)
part &lt;- partition(nutrimouse$lipid, threshold = 0.50)
mods &lt;- part$mapping_key[which(grepl("reduced_var_", part$mapping_key$variable)), ]$mapping

# run function for diet and genotype
ACDCmod(fullData = nutrimouse$lipid,
        modules = mods,
        externalVar = data.frame(diet=as.numeric(nutrimouse$diet), 
                                  genotype=as.numeric(nutrimouse$genotype)))

</code></pre>

<hr>
<h2 id='coVar'>coVar</h2><span id='topic+coVar'></span>

<h3>Description</h3>

<p>Function to calculate ACDC covariances within a data pair for all samples
</p>


<h3>Usage</h3>

<pre><code class='language-R'>coVar(dataPair, fullData)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="coVar_+3A_datapair">dataPair</code></td>
<td>
<p>column indices of two genes to calculate covariance between</p>
</td></tr>
<tr><td><code id="coVar_+3A_fulldata">fullData</code></td>
<td>
<p>dataframe or matrix with samples as rows, all probes as columns; each entry should be numeric gene expression or other molecular data values</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Co-expression for a single sample, s, is defined as
</p>
<p style="text-align: center;"><code class="reqn">c_{s,j,k} \equiv \left(g_{s,j}-\bar{g_j}\right)\left(g_{s,k}-\bar{g_k}\right)</code>
</p>

<p>where <code class="reqn">g_{s,j}</code> denotes the expression of gene j in sample s and <code class="reqn">\bar{g_j}</code> denotes the mean expression of gene j in all samples.
</p>
<p>Denoting the sample size as N, coVar returns the co-expression profile across all samples:
</p>
<p style="text-align: center;"><code class="reqn">c_{j,k} = (c_{1,j,k}, c_{2,j,k}, ... , c_{N,j,k})</code>
</p>



<h3>Value</h3>

<p>Co-expression profile, or pairwise covariances for all samples, vector for given features
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824.
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run function with first two samples
coVar(dataPair = c(1, 2), 
      fullData = nutrimouse$lipid)

</code></pre>

<hr>
<h2 id='GCTA_par'>GCTA_par</h2><span id='topic+GCTA_par'></span>

<h3>Description</h3>

<p>GCTA_par determines the average heritability of the first principal component of either the co-expression or covariance of gene expression modules for a range of increasingly reduced datasets. Dimension reduction is done with Partition, where features are only condensed into modules if the intraclass correlation between the features is at least the user-supplied information loss criterion (ILC), 0 &lt;= ILC &lt;= 1. An ILC of one returns the full dataset with no reduction, and an ILC of zero returns one module of all input features, reducing the dataset to the mean value. For each ILC value, with the number of ILCs tested determined by input parameter ILCincrement, the function returns the point estimate and standard error of the average heritability of the first principal component of the co-expression or covariance of the gene expression modules in the reduced dataset. If input parameter permute is true, the function also returns the same values for a random permutation of the first principle component of the appropriate matrix.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GCTA_par(
  df,
  ILCincrement = 0.05,
  fileLoc,
  gctaPath,
  remlAlg = 0,
  maxRemlIt = 100,
  numCovars = NULL,
  catCovars = NULL,
  summaryType,
  permute = TRUE,
  numNodes = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GCTA_par_+3A_df">df</code></td>
<td>
<p>n x p data frame or matrix of numeric -omics values with no ID column</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_ilcincrement">ILCincrement</code></td>
<td>
<p>float between zero and one determining interval between tested ILC values; default is 0.05</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_fileloc">fileLoc</code></td>
<td>
<p>absolute file path to bed, bim, and fam files, including prefix</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_gctapath">gctaPath</code></td>
<td>
<p>absolute path to GCTA software</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_remlalg">remlAlg</code></td>
<td>
<p>algorithm to run REML iterations in GCTA; 0 = average information (AI), 1 = Fisher-scoring, 2 = EM; default is 0 (AI)</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_maxremlit">maxRemlIt</code></td>
<td>
<p>the maximum number of REML iterations; default is 100</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_numcovars">numCovars</code></td>
<td>
<p>n x c_n matrix of numerical covariates to adjust heritability model for; must be in same person order as fam file; default is NULL</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_catcovars">catCovars</code></td>
<td>
<p>n x c_c matrix of categorical covariates to adjust heritability model for; must be in same person order as fam file; default is NULL</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_summarytype">summaryType</code></td>
<td>
<p>one of &quot;coexpression&quot; or &quot;covariance&quot;; determines how to summarize each module</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_permute">permute</code></td>
<td>
<p>boolean value for whether or not to calculate values for a random permutation module summary; default is true</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_numnodes">numNodes</code></td>
<td>
<p>number of available compute nodes for parallelization; default is 1</p>
</td></tr>
<tr><td><code id="GCTA_par_+3A_verbose">verbose</code></td>
<td>
<p>logical for whether or not to display progress updates; default is TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Genome-wide Complex Trait Analysis (GCTA) is a suite of C++ functions. In order to use the GCTA functions, the user must specify the absolute path to the GCTA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/gcta/#Download">here</a>.
</p>
<p>Here, we use GCTA's Genomics REstricted Maximum Likelihood (GREML) method to estimate the heritability of an external phenotype. GREML is called 2*number of modules for each ILC tested if permutations are included.
</p>
<p>Dimension reduction is done with Partition, an agglomerative data reduction method which performs both feature condensation and extraction based on a user provided information loss criterion (ILC). Feature condensation into modules are only accepted if the intraclass correlation between the features is at least the ILC. The superPartition function is called if the gene expression dataset contains more than 4,000 features.
</p>


<h3>Value</h3>

<p>Data frame with columns
</p>

<dl>
<dt>ILC</dt><dd><p>the information loss criterion used for that iteration</p>
</dd>
<dt>InformationLost</dt><dd><p>percent information lost due to data reduction</p>
</dd>
<dt>PercentReduction</dt><dd><p>percent of variables condensed compared to unreduced data</p>
</dd>
<dt>AveVarianceExplained_Observed</dt><dd><p>average heritability estimate for PC1 of observed summary data</p>
</dd>
<dt>OverallSD_Observed</dt><dd><p>standard deviation of the heritability estimates for PC1 of observed summary data</p>
</dd>
<dt>VarianceExplained_Observed</dt><dd><p>list of heritability estimates for PC1 of observed summary for all modules</p>
</dd>
<dt>SE_Observed</dt><dd><p>list of standard errors of the heritability estimates for PC1 of observed summary data for all modules</p>
</dd>
<dt>AveVarianceExplained_Permuted</dt><dd><p>average heritability for PC1 of permuted summary data</p>
</dd>
<dt>OverallSD_Permuted</dt><dd><p>standard deviation of the heritability estimates for PC1 of permuted summary data</p>
</dd>
<dt>VarianceExplained_Permuted</dt><dd><p>list of heritability estimates for PC1 of permuted summary data for all modules</p>
</dd>
<dt>SE_Permuted</dt><dd><p>list of standard errors of the heritability estimates for PC1 of permuted summary data for all modules</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Yang J, Lee SH, Goddard ME, Visscher PM. GCTA: a tool for genome-wide complex trait analysis. Am J Hum Genet. 2011 Jan 7;88(1):76-82. doi: 10.1016/j.ajhg.2010.11.011. Epub 2010 Dec 17. PMID: 21167468; PMCID: PMC3014363.
</p>


<h3>See Also</h3>

<p>GCTA software - <a href="https://yanglab.westlake.edu.cn/software/gcta/">https://yanglab.westlake.edu.cn/software/gcta/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# run function; input absolute path to OSCA software before running
## Not run: GCTA_par(df = geneExpressionData, 
          ILCincrement = 0.25, 
          fileLoc = "pathHere",
          gctaPath = "pathHere",
          summaryType = "coexpression",
          permute = TRUE,
          numNodes = 1)
## End(Not run)

</code></pre>

<hr>
<h2 id='GCTA_parPlot'>GCTA_parPlot</h2><span id='topic+GCTA_parPlot'></span>

<h3>Description</h3>

<p>GCTA_parPlot creates a graph of the output from the GCTA_par function, plotting average heritability of the first principal component of either co-expression or covariance of gene modules against information lost/percent reduction for both observed and permuted data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GCTA_parPlot(df, dataName = "", summaryType)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GCTA_parPlot_+3A_df">df</code></td>
<td>
<p>output from GCTA_par function with permutations</p>
</td></tr>
<tr><td><code id="GCTA_parPlot_+3A_dataname">dataName</code></td>
<td>
<p>string of name of data for graph labels; default is blank</p>
</td></tr>
<tr><td><code id="GCTA_parPlot_+3A_summarytype">summaryType</code></td>
<td>
<p>one of &quot;coexpression&quot; or &quot;covariance&quot;; how modules were summarized for GCTA calculations</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Genome-wide Complex Trait Analysis (GCTA) is a suite of C++ functions. In order to use the GCTA functions, the user must specify the absolute path to the GCTA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/gcta/#Download">here</a>.
</p>
<p>In GCTA_par, we use GCTA's Genomics REstricted Maximum Likelihood (GREML) method to estimate the average heritability of the first principal component of either co-expression or covariance of gene modules. The produced plot shows these heritability estimates at varying levels of dataset reduction, calculated for observed data in blue and permuted data in red. An information loss value of 0 represents the unreduced dataset, and an information loss level of 100 represents the data being reduced to the average expression of all features.
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Yang J, Lee SH, Goddard ME, Visscher PM. GCTA: a tool for genome-wide complex trait analysis. Am J Hum Genet. 2011 Jan 7;88(1):76-82. doi: 10.1016/j.ajhg.2010.11.011. Epub 2010 Dec 17. PMID: 21167468; PMCID: PMC3014363.
</p>


<h3>See Also</h3>

<p>GCTA software - <a href="https://yanglab.westlake.edu.cn/software/gcta/#Overview">https://yanglab.westlake.edu.cn/software/gcta/#Overview</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
# run OSCA_par and save output; input absolute path to OSCA software before running
## Not run: par &lt;- GCTA_par(df = geneExpressionData, 
          ILCincrement = 0.25, 
          fileLoc = "pathHere",
          gctaPath = "pathHere",
          summaryType = "coexpression",
          permute = TRUE,
          numNodes = 1)
## End(Not run)

# run function
## Not run: GCTA_parPlot(df=par, dataName = "Example Data", summaryType = "coexpression")

</code></pre>

<hr>
<h2 id='GCTA_singleValue'>GCTA_singleValue</h2><span id='topic+GCTA_singleValue'></span>

<h3>Description</h3>

<p>Function to return the heritability of an external phenotype for a single dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>GCTA_singleValue(
  fileLoc,
  externalVar,
  gctaPath,
  remlAlg = 0,
  maxRemlIt = 100,
  numCovars = NULL,
  catCovars = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="GCTA_singleValue_+3A_fileloc">fileLoc</code></td>
<td>
<p>absolute file path to bed, bim, and fam files, including prefix</p>
</td></tr>
<tr><td><code id="GCTA_singleValue_+3A_externalvar">externalVar</code></td>
<td>
<p>vector of length n of external variable values with no ID column; must be in the same sample order as bed, bim, fam files</p>
</td></tr>
<tr><td><code id="GCTA_singleValue_+3A_gctapath">gctaPath</code></td>
<td>
<p>absolute path to GCTA software</p>
</td></tr>
<tr><td><code id="GCTA_singleValue_+3A_remlalg">remlAlg</code></td>
<td>
<p>algorithm to run REML iterations in GCTA; 0 = average information (AI), 1 = Fisher-scoring, 2 = EM; default is 0 (AI)</p>
</td></tr>
<tr><td><code id="GCTA_singleValue_+3A_maxremlit">maxRemlIt</code></td>
<td>
<p>the maximum number of REML iterations; default is 100</p>
</td></tr>
<tr><td><code id="GCTA_singleValue_+3A_numcovars">numCovars</code></td>
<td>
<p>n x c_n matrix of numerical covariates to adjust heritability model for; must be in same person order as fam file; default is NULL</p>
</td></tr>
<tr><td><code id="GCTA_singleValue_+3A_catcovars">catCovars</code></td>
<td>
<p>n x c_c matrix of categorical covariates to adjust heritability model for; must be in same person order as fam file; default is NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>Genome-wide Complex Trait Analysis (GCTA) is a suite of C++ functions. In order to use the GCTA functions, the user must specify the absolute path to the GCTA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/gcta/#Download">here</a>.
</p>
<p>Here, we use GCTA's Genomics REstricted Maximum Likelihood (GREML) method to estimate the heritability of an external phenotype.
</p>


<h3>Value</h3>

<p>Row of GREML output containing heritability point estimate of external data and standard error
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Yang J, Lee SH, Goddard ME, Visscher PM. GCTA: a tool for genome-wide complex trait analysis. Am J Hum Genet. 2011 Jan 7;88(1):76-82. doi: 10.1016/j.ajhg.2010.11.011. Epub 2010 Dec 17. PMID: 21167468; PMCID: PMC3014363.
</p>


<h3>See Also</h3>

<p>GCTA software - <a href="https://yanglab.westlake.edu.cn/software/gcta/">https://yanglab.westlake.edu.cn/software/gcta/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>
externalVar &lt;- c()

# run function; input data before running
## Not run: OSCA_singleValue(fileLoc = "pathHere", 
                  externalVar = externalVar,
                  gctaPath = "pathHere")
## End(Not run)

</code></pre>

<hr>
<h2 id='OSCA_par'>OSCA_par</h2><span id='topic+OSCA_par'></span>

<h3>Description</h3>

<p>OSCA_par determines the percent variance explained in an external variable (exposures or responses) for a range of increasingly reduced datasets. Dimension reduction is done with Partition, where features are only condensed into modules if the intraclass correlation between the features is at least the user-supplied information loss criterion (ILC), 0 &lt;= ILC &lt;= 1. An ILC of one returns the full dataset with no reduction, and an ILC of zero returns one module of all input features, reducing the dataset to the mean value. For each ILC value, with the number of ILCs tested determined by input parameter ILCincrement, the function returns the point estimate and standard error of the percent variance explained in the observed external variable by the reduced dataset. If input parameter permute is true, the function also returns the same values for a random permutation of the external variable.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OSCA_par(
  df,
  externalVar,
  ILCincrement = 0.05,
  oscaPath,
  remlAlg = 0,
  maxRemlIt = 100,
  numCovars = NULL,
  catCovars = NULL,
  permute = TRUE,
  numNodes = 1,
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OSCA_par_+3A_df">df</code></td>
<td>
<p>n x p data frame or matrix of numeric -omics values with no ID column</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_externalvar">externalVar</code></td>
<td>
<p>vector of length n of external variable values with no ID column</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_ilcincrement">ILCincrement</code></td>
<td>
<p>float between zero and one determining interval between tested ILC values; default is 0.05</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_oscapath">oscaPath</code></td>
<td>
<p>absolute path to OSCA software</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_remlalg">remlAlg</code></td>
<td>
<p>which algorithm to run REML iterations in GCTA; 0 = average information (AI), 1 = Fisher-scoring, 2 = EM; default is 0 (AI)</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_maxremlit">maxRemlIt</code></td>
<td>
<p>the maximum number of REML iterations; default is 100</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_numcovars">numCovars</code></td>
<td>
<p>n x c_n matrix of numerical covariates to adjust heritability model for; must be in same person order as externalVar; default is NULL</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_catcovars">catCovars</code></td>
<td>
<p>n x c_c matrix of categorical covariates to adjust heritability model for; must be in same person order as externalVar; default is NULL</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_permute">permute</code></td>
<td>
<p>boolean value for whether or not to calculate values for a random permutation of the external variable; default is true</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_numnodes">numNodes</code></td>
<td>
<p>number of available compute nodes for parallelization; default is 1</p>
</td></tr>
<tr><td><code id="OSCA_par_+3A_verbose">verbose</code></td>
<td>
<p>logical for whether or not to display progress updates; default is TRUE</p>
</td></tr>
</table>


<h3>Details</h3>

<p>OmicS-data-based Complex trait Analysis (OSCA) is a suite of C++ functions. In order to use the OSCA functions, the user must specify the absolute path to the OSCA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/osca/#Download">here</a>.
</p>
<p>Here, we use OSCA's Omics Restricted Maximum Likelihood (OREML) method to estimate the percent of variance in an external phenotype that can be explained by an omics profile, akin to heritability estimates in GWAS. OREML is called twice for each ILC tested if permutations are included.
</p>
<p>Dimension reduction is done with Partition, an agglomerative data reduction method which performs both feature condensation and extraction based on a user provided information loss criterion (ILC). Feature condensation into modules are only accepted if the intraclass correlation between the features is at least the ILC. The superPartition function is called if the gene expression dataset contains more than 4,000 features.
</p>


<h3>Value</h3>

<p>Data frame with columns
</p>

<dl>
<dt>ILC</dt><dd><p>the information loss criterion used for that iteration</p>
</dd>
<dt>InformationLost</dt><dd><p>percent information lost due to data reduction</p>
</dd>
<dt>PercentReduction</dt><dd><p>percent of variables condensed compared to unreduced data</p>
</dd>
<dt>VarianceExplained_Observed</dt><dd><p>percent variance explained in observed external variable by the data</p>
</dd>
<dt>SE_Observed</dt><dd><p>standard error of the percent variance estimate for observed external variable</p>
</dd>
<dt>VarianceExplained_Permuted</dt><dd><p>percent variance explained in permuted external variable by the data; only if input parameter &quot;permute&quot; is true</p>
</dd>
<dt>SE_Permuted</dt><dd><p>standard error of the percent variance estimate for permuted external variable; only if input parameter &quot;permute&quot; is true</p>
</dd>
</dl>



<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Benjamini Y, Hochberg Y. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal statistical society: series B (Methodological)</em> <strong>57</strong> (1995) 289–300.
</p>
<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824.
</p>


<h3>See Also</h3>

<p>OSCA software - <a href="https://yanglab.westlake.edu.cn/software/osca/">https://yanglab.westlake.edu.cn/software/osca/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run function; input absolute path to OSCA software before running
## Not run: OSCA_par(df = nutrimouse$gene, 
          externalVar = as.numeric(nutrimouse$diet),
          ILCincrement = 0.25, 
          oscaPath = "pathHere")
## End(Not run)

</code></pre>

<hr>
<h2 id='OSCA_parPlot'>OSCA_parPlot</h2><span id='topic+OSCA_parPlot'></span>

<h3>Description</h3>

<p>OSCA_parPlot creates a graph of the output from the OSCA_par function, plotting percent variance explained in an external variable (exposure or response) against information lost/percent reduction for both observed and permuted data.
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OSCA_parPlot(df, externalVarName = "", dataName = "")
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OSCA_parPlot_+3A_df">df</code></td>
<td>
<p>output from OSCA_par function with permutations</p>
</td></tr>
<tr><td><code id="OSCA_parPlot_+3A_externalvarname">externalVarName</code></td>
<td>
<p>string of name of external variable for graph labels; default is blank</p>
</td></tr>
<tr><td><code id="OSCA_parPlot_+3A_dataname">dataName</code></td>
<td>
<p>string of name of data for graph labels; default is blank</p>
</td></tr>
</table>


<h3>Details</h3>

<p>OmicS-data-based Complex trait Analysis (OSCA) is a suite of C++ functions. In order to use the OSCA functions, the user must specify the absolute path to the OSCA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/osca/#Download">here</a>.
</p>
<p>In OSCA_par, we use OSCA's Omics Restricted Maximum Likelihood (OREML) method to estimate the percent of variance in an external phenotype that can be explained by an omics profile, akin to heritability estimates in GWAS. The produced plot shows the percent variance explained in an external variable at varying levels of dataset reduction, calculated for observed external variables in blue and permuted external variables in red. An information loss value of 0 represents the unreduced dataset, and an information loss level of 100 represents the data being reduced to the average expression of all features.
</p>


<h3>Value</h3>

<p>ggplot object
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Benjamini Y, Hochberg Y. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal statistical society: series B (Methodological)</em> <strong>57</strong> (1995) 289–300.
</p>
<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824.
</p>


<h3>See Also</h3>

<p>OSCA software - <a href="https://yanglab.westlake.edu.cn/software/osca/">https://yanglab.westlake.edu.cn/software/osca/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run OSCA_par and save output; input absolute path to OSCA software before running
## Not run: par &lt;- OSCA_par(df = nutrimouse$gene, 
                externalVar = as.numeric(nutrimouse$diet),
                 ILCincrement = 0.25,
                 oscaPath = "pathHere")
## End(Not run)

# run function
## Not run: OSCA_parPlot(df=par, externalVarName = "Diet", dataName = "Nutritional Issue Genes")

</code></pre>

<hr>
<h2 id='OSCA_singleValue'>OSCA_singleValue</h2><span id='topic+OSCA_singleValue'></span>

<h3>Description</h3>

<p>Function to return the percent variance explained in an external phenotype for a single dataset
</p>


<h3>Usage</h3>

<pre><code class='language-R'>OSCA_singleValue(
  df,
  externalVar,
  oscaPath,
  remlAlg = 0,
  maxRemlIt = 100,
  numCovars = NULL,
  catCovars = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr><td><code id="OSCA_singleValue_+3A_df">df</code></td>
<td>
<p>n x p dataframe or matrix of numeric -omics values with no ID column</p>
</td></tr>
<tr><td><code id="OSCA_singleValue_+3A_externalvar">externalVar</code></td>
<td>
<p>vector of length n of external variable values with no ID column</p>
</td></tr>
<tr><td><code id="OSCA_singleValue_+3A_oscapath">oscaPath</code></td>
<td>
<p>absolute path to OSCA software</p>
</td></tr>
<tr><td><code id="OSCA_singleValue_+3A_remlalg">remlAlg</code></td>
<td>
<p>which algorithm to run REML iterations in GCTA; 0 = average information (AI), 1 = Fisher-scoring, 2 = EM; default is 0 (AI)</p>
</td></tr>
<tr><td><code id="OSCA_singleValue_+3A_maxremlit">maxRemlIt</code></td>
<td>
<p>the maximum number of REML iterations; default is 100</p>
</td></tr>
<tr><td><code id="OSCA_singleValue_+3A_numcovars">numCovars</code></td>
<td>
<p>n x c_n matrix of numerical covariates to adjust heritability model for; must be in same person order as fam file; default is NULL</p>
</td></tr>
<tr><td><code id="OSCA_singleValue_+3A_catcovars">catCovars</code></td>
<td>
<p>n x c_c matrix of categorical covariates to adjust heritability model for; must be in same person order as fam file; default is NULL</p>
</td></tr>
</table>


<h3>Details</h3>

<p>OmicS-data-based Complex trait Analysis (OSCA) is a suite of C++ functions. In order to use the OSCA functions, the user must specify the absolute path to the OSCA software, which can be downloaded from the Yang Lab website <a href="https://yanglab.westlake.edu.cn/software/osca/#Download">here</a>.
</p>
<p>Here, we use OSCA's Omics Restricted Maximum Likelihood (OREML) method to estimate the percent of variance in an external phenotype that can be explained by an omics profile, akin to heritability estimates in GWAS.
</p>


<h3>Value</h3>

<p>Row of OREML output containing percent variance explained in external data and standard error
</p>


<h3>Author(s)</h3>

<p>Katelyn Queen, <a href="mailto:kjqueen@usc.edu">kjqueen@usc.edu</a>
</p>


<h3>References</h3>

<p>Benjamini Y, Hochberg Y. Controlling the false discovery rate: a practical and powerful approach to multiple testing. <em>Journal of the Royal statistical society: series B (Methodological)</em> <strong>57</strong> (1995) 289–300.
</p>
<p>Martin P, et al. Novel aspects of PPARalpha-mediated regulation of lipid and xenobiotic metabolism revealed through a nutrigenomic study. <em>Hepatology</em>, in press, 2007.
</p>
<p>Millstein J, Battaglin F, Barrett M, Cao S, Zhang W, Stintzing S, et al. Partition: a surjective mapping approach for dimensionality reduction. <em>Bioinformatics</em> <strong>36</strong> (2019) 676–681. doi:10.1093/bioinformatics/ btz661.
</p>
<p>Queen K, Nguyen MN, Gilliland F, Chun S, Raby BA, Millstein J. ACDC: a general approach for detecting phenotype or exposure associated co-expression. <em>Frontiers in Medicine</em> (2023) 10. doi:10.3389/fmed.2023.1118824.
</p>


<h3>See Also</h3>

<p>OSCA software - <a href="https://yanglab.westlake.edu.cn/software/osca/">https://yanglab.westlake.edu.cn/software/osca/</a>
</p>


<h3>Examples</h3>

<pre><code class='language-R'>#load CCA package for example dataset
library(CCA)

# load dataset
data("nutrimouse")

# run function; input absolute path to OSCA software before running
## Not run: OSCA_singleValue(df = nutrimouse$gene, 
                  externalVar = as.numeric(nutrimouse$diet),
                  oscaPath = "pathHere")
## End(Not run)

</code></pre>

</main>

</div>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/prism.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/prism/1.29.0/components/prism-r.min.js"></script>
</body></html>
